<!DOCTYPE html><html><head><title>Help for package phytools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phytools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#phytools-package'>
<p>phytools: Phylogenetic Tools for comparative biology (and other things)</p></a></li>
<li><a href='#add.arrow'><p>Add an arrow pointing to a tip or node on the tree</p></a></li>
<li><a href='#add.color.bar'><p>Add color bar to a plot</p></a></li>
<li><a href='#add.everywhere'><p>Add tip to all edges in a tree</p></a></li>
<li><a href='#add.random'><p>Add tips at random to the tree</p></a></li>
<li><a href='#add.simmap.legend'><p>Add legend to stochastically mapped tree</p></a></li>
<li><a href='#add.species.to.genus'><p>Add species to genus on a phylogeny or bind simulated species subtrees to a backbone genus tree</p></a></li>
<li><a href='#aic.w'><p>Computes Akaike weights</p></a></li>
<li><a href='#allFurcTrees'><p>Generate all bi- and multifurcating unrooted trees</p></a></li>
<li><a href='#anc.Bayes'><p>Bayesian ancestral character estimation</p></a></li>
<li><a href='#anc.ML'><p>Ancestral character estimation using likelihood</p></a></li>
<li><a href='#anc.trend'><p>Ancestral character estimation with a trend</p></a></li>
<li><a href='#ancr'><p>Compute marginal or joint ancestral state estimates</p></a></li>
<li><a href='#ancThresh'><p>Ancestral character estimation under the threshold model using Bayesian MCMC</p></a></li>
<li><a href='#anoletree'><p>Phylogenetic datasets</p></a></li>
<li><a href='#ansi_phylo'><p>Compute the parsimony score</p></a></li>
<li><a href='#applyBranchLengths'><p>Applies the branch lengths of a reference tree to a target</p></a></li>
<li><a href='#as.multiPhylo'><p>Conversion to object of class <code>"multiPhylo"</code></p></a></li>
<li><a href='#as.Qmatrix'><p>Convert a fitted M<em>k</em> model to a Q-matrix</p></a></li>
<li><a href='#ave.rates'><p>Average the posterior rates</p></a></li>
<li><a href='#averageTree'><p>Compute an average tree from a set of trees and related operations</p></a></li>
<li><a href='#bd'><p>Convert object of class <code>"birthdeath"</code> to raw birth &amp; death rates</p></a></li>
<li><a href='#bind.tip'><p>Attaches a new tip to a tree</p></a></li>
<li><a href='#bind.tree.simmap'><p>Binds two trees of class <code>"simmap"</code></p></a></li>
<li><a href='#bmPlot'><p>Simulates and visualizes discrete-time Brownian evolution on a phylogeny</p></a></li>
<li><a href='#bounded_bm'><p>Fits bounded and wrapped Brownian motion models</p></a></li>
<li><a href='#branching.diffusion'><p>Animation of branching random diffusion</p></a></li>
<li><a href='#brownie.lite'><p>Likelihood test for rate variation in a continuous trait</p></a></li>
<li><a href='#brownieREML'><p>REML version of brownie.lite</p></a></li>
<li><a href='#cladelabels'><p>Add labels to subtrees of a plotted phylogeny</p></a></li>
<li><a href='#collapse.to.star'><p>Collapse a subtree to a star phylogeny</p></a></li>
<li><a href='#collapseTree'><p>Interactive tree visualizer</p></a></li>
<li><a href='#compare.chronograms'><p>Compares two chronograms with precisely matching nodes in a visual manner</p></a></li>
<li><a href='#consensus.edges'><p>Compute consensus edges for a tree under some criterion</p></a></li>
<li><a href='#contMap'><p>Map continuous trait evolution on the tree</p></a></li>
<li><a href='#cophylo'><p>Creates a co-phylogenetic plot</p></a></li>
<li><a href='#cospeciation'><p>Conducts a statistical test of cospeciation between two trees</p></a></li>
<li><a href='#countSimmap'><p>Counts the number of character changes on a object of class <code>"simmap"</code> or <code>"multiSimmap"</code></p></a></li>
<li><a href='#ctt'><p>Generates (or simulates) a 'changes through time' plot from a set of stochastic map character histories</p></a></li>
<li><a href='#density.multiSimmap'><p>Computes a posterior distribution for the number and types of changes on the tree</p></a></li>
<li><a href='#densityMap'><p>Plot posterior density of stochastic mapping on a tree</p></a></li>
<li><a href='#densityTree'><p>Plots a posterior sample of trees</p></a></li>
<li><a href='#describe.simmap'><p>Summarizes a stochastic mapped tree or set of trees</p></a></li>
<li><a href='#di2multi.simmap'><p>Collapse or resolve polytomies in a tree with a character painted on the edges</p></a></li>
<li><a href='#dotTree'><p>Creates a phylogenetic dot plot</p></a></li>
<li><a href='#drop.clade'><p>Drop a clade from a tree</p></a></li>
<li><a href='#drop.leaves'><p>Drop all the leaves (tips) from a tree</p></a></li>
<li><a href='#drop.tip.contMap'><p>Drop tip or tips from an object of class <code>"contMap"</code> or <code>"densityMap"</code></p></a></li>
<li><a href='#drop.tip.multiSimmap'><p>Drop or keep tip or tips from an object of class <code>"multiSimmap"</code></p></a></li>
<li><a href='#drop.tip.simmap'><p>Drop tips or extract clade from tree with mapped discrete character</p></a></li>
<li><a href='#Dtest'><p>Conducts correlational D-test from stochastic mapping</p></a></li>
<li><a href='#edge.widthMap'><p>Map continuous trait evolution on the tree</p></a></li>
<li><a href='#edgeProbs'><p>Compute the relative frequencies of state changes along edges</p></a></li>
<li><a href='#estDiversity'><p>Estimate diversity at each node of the tree</p></a></li>
<li><a href='#evol.rate.mcmc'><p>Bayesian MCMC method for identifying exceptional phenotypic diversification in a phylogeny</p></a></li>
<li><a href='#evol.vcv'><p>Likelihood test for variation in the evolutionary variance-covariance matrix</p></a></li>
<li><a href='#evolvcv.lite'><p>Likelihood test for a shift in the evolutionary correlation between traits</p></a></li>
<li><a href='#exhaustiveMP'><p>Exhaustive and branch &amp; bound MP optimization</p></a></li>
<li><a href='#expand.clade'><p>Expands (or contracts) the tip-spacing of a given clade or clades</p></a></li>
<li><a href='#export.as.xml'><p>Export trees &amp; data in XML format</p></a></li>
<li><a href='#fancyTree'><p>Plots special types of phylogenetic trees</p></a></li>
<li><a href='#fastAnc'><p>(Reasonably) fast estimation of ML ancestral states</p></a></li>
<li><a href='#fastBM'><p>(Reasonably) fast quantitative trait simulation on phylogenies</p></a></li>
<li><a href='#fastMRCA'><p>Get the MRCA (or height above the root of the MRCA) of a pair of tip taxa</p></a></li>
<li><a href='#findMRCA'><p>Get the MRCA of a set of taxa</p></a></li>
<li><a href='#fit.bd'><p>Fits birth-death (speciation/extinction) model to reconstructed phylogeny</p></a></li>
<li><a href='#fitBayes'><p>Evolutionary model fitting with intraspecific variability using Bayesian MCMC</p></a></li>
<li><a href='#fitDiversityModel'><p>Fit diversity-dependent phenotypic evolution model</p></a></li>
<li><a href='#fitMk'><p>Fits extended M<em>k</em> model for discrete character evolution</p></a></li>
<li><a href='#fitmultiBM'><p>Fits bounded and wrapped Brownian motion models</p></a></li>
<li><a href='#fitPagel'><p>Function to test for correlated evolution of binary traits</p></a></li>
<li><a href='#fitThresh'><p>Fits multi-state threshold model using the discrete approximation</p></a></li>
<li><a href='#force.ultrametric'><p>Coerces a phylogenetic tree to be ultrametric</p></a></li>
<li><a href='#gamma_pruning'><p>Internally used function</p></a></li>
<li><a href='#gammatest'><p>Gamma test of Pybus &amp; Harvey (2000)</p></a></li>
<li><a href='#genSeq'><p>Simulate a DNA alignment on the tree under a model</p></a></li>
<li><a href='#geo.legend'><p>Adds a geological (or other temporal) legend to a plotted tree</p></a></li>
<li><a href='#get.treepos'><p>Get position or node of a plotted tree interactively</p></a></li>
<li><a href='#getCladesofSize'><p>Get all subtrees larger than or equal to a specified size</p></a></li>
<li><a href='#getDescendants'><p>Get descendant node numbers</p></a></li>
<li><a href='#getExtant'><p>Returns a list of the extant or extinct lineages in a tree containing non-contemporaneous tips</p></a></li>
<li><a href='#getSisters'><p>Get the sister node number, label, or set of nodes for a node or tip</p></a></li>
<li><a href='#getStates'><p>Get the states at nodes or tips from a mapped tree</p></a></li>
<li><a href='#labelnodes'><p>Function to interactively label nodes of a plotted tree</p></a></li>
<li><a href='#ladderize.simmap'><p>Ladderize a tree with a mapped discrete character</p></a></li>
<li><a href='#lambda.transform'><p><code class="reqn">\lambda</code> transformation of matrix</p></a></li>
<li><a href='#likMlambda'><p>Likelihood for joint <code class="reqn">\lambda</code></p></a></li>
<li><a href='#linklabels'><p>Function to add tip labels to a plotted tree with linking lines</p></a></li>
<li><a href='#locate.fossil'><p>Locate a fossil lineage in a tree using continuous characters</p></a></li>
<li><a href='#locate.yeti'><p>Locate a cryptic, recently extinct, or missing taxon on a tree</p></a></li>
<li><a href='#ls.tree'><p>Least squares branch lengths for a given tree</p></a></li>
<li><a href='#ltt'><p>Creates lineage-through-time plot (including extinct lineages)</p></a></li>
<li><a href='#ltt95'><p>Creates a (1-<code class="reqn">\alpha</code>)% CI for a set of LTTs</p></a></li>
<li><a href='#make.era.map'><p>Create &quot;era&quot; map on a phylogenetic tree</p></a></li>
<li><a href='#make.simmap'><p>Simulate stochastic character maps on a phylogenetic tree or trees</p></a></li>
<li><a href='#map.overlap'><p>Proportional overlap between two mapped character histories on a tree</p></a></li>
<li><a href='#map.to.singleton'><p>Converts a tree without singletons to a tree with singleton nodes</p></a></li>
<li><a href='#mapped.states'><p>Returns a vector, matrix, or list of the mapped states on a tree or set of trees</p></a></li>
<li><a href='#markChanges'><p>Add marked changes to a plotted tree with mapped discrete character</p></a></li>
<li><a href='#matchNodes'><p>Matches nodes between two trees</p></a></li>
<li><a href='#mergeMappedStates'><p>Merge two or more mapped states into one state</p></a></li>
<li><a href='#midpoint_root'><p>Midpoint root a phylogeny</p></a></li>
<li><a href='#minRotate'><p>Rotates all nodes of the tree to minimize the difference in order with a vector</p></a></li>
<li><a href='#minSplit'><p>Finding the minimum (median) split in the posterior sample</p></a></li>
<li><a href='#modified.Grafen'><p>Computes modified Grafen edge lengths</p></a></li>
<li><a href='#mrp.supertree'><p>Matrix representation parsimony supertree estimation</p></a></li>
<li><a href='#multi.mantel'><p>Multiple matrix regression (partial Mantel test)</p></a></li>
<li><a href='#multiC'><p>Returns a list with phylogenetic VCV matrix for each mapped state</p></a></li>
<li><a href='#multirateBM'><p>Function to fit a multi-rate Brownian evolution model</p></a></li>
<li><a href='#multiRF'><p>Computes Robinson-Foulds distance between a set of trees</p></a></li>
<li><a href='#nodeHeights'><p>Compute the heights above the root of each node</p></a></li>
<li><a href='#nodelabels.cophylo'><p>Add labels to a plotted &quot;cophylo&quot; object</p></a></li>
<li><a href='#optim.phylo.ls'><p>Phylogeny inference using the least squares method</p></a></li>
<li><a href='#orderMappedEdge'><p>Order the columns of mapped.edge to match across trees</p></a></li>
<li><a href='#paintSubTree'><p>Paint sub-trees with a discrete character</p></a></li>
<li><a href='#paste.tree'><p>Paste two trees together</p></a></li>
<li><a href='#pbtree'><p>Simulate pure-birth or birth-death stochastic tree or trees</p></a></li>
<li><a href='#pgls.Ives'><p>Phylogenetic regression with intraspecific sampling error</p></a></li>
<li><a href='#phenogram'><p>Plot traitgram (phenogram)</p></a></li>
<li><a href='#phyl.cca'><p>Phylogenetic canonical correlation analysis</p></a></li>
<li><a href='#phyl.pairedttest'><p>Phylogenetic paired <em>t</em>-test</p></a></li>
<li><a href='#phyl.pca'><p>Phylogenetic principal components analysis</p></a></li>
<li><a href='#phyl.resid'><p>Phylogenetic size-correction via GLS regression</p></a></li>
<li><a href='#phyl.RMA'><p>Phylogenetic reduced major axis (RMA) regression</p></a></li>
<li><a href='#phyl.vcv'><p>Compute evolutionary VCV matrix for a tree &amp; dataset</p></a></li>
<li><a href='#phylANOVA'><p>Phylogenetic ANOVA and post-hoc tests</p></a></li>
<li><a href='#phylo.heatmap'><p>Creates a phylogenetic heat map</p></a></li>
<li><a href='#phylo.impute'><p>Phylogenetic imputation for multivariate continuous character data</p></a></li>
<li><a href='#phylo.to.map'><p>Plot tree with tips linked to geographic coordinates</p></a></li>
<li><a href='#phylo.toBackbone'><p>Converts tree to backbone or vice versa</p></a></li>
<li><a href='#phyloDesign'><p>Compute design matrix for least squares analyses</p></a></li>
<li><a href='#phylomorphospace'><p>Creates phylomorphospace plot</p></a></li>
<li><a href='#phylomorphospace3d'><p>Creates three-dimensional phylomorphospace plot</p></a></li>
<li><a href='#phylosig'><p>Compute phylogenetic signal with two methods</p></a></li>
<li><a href='#plot.backbonePhylo'><p>Plots backbone tree with triangles as clades</p></a></li>
<li><a href='#plotBranchbyTrait'><p>Plot branch colors by a quantitative trait or value</p></a></li>
<li><a href='#plotSimmap'><p>Plot stochastic character mapped tree</p></a></li>
<li><a href='#plotThresh'><p>Tree plotting with posterior probabilities of ancestral states from the threshold model</p></a></li>
<li><a href='#plotTree'><p>Plots rooted phylogenetic tree</p></a></li>
<li><a href='#plotTree.datamatrix'><p>Plot a tree with a discrete (or continuous) character data matrix at the tips</p></a></li>
<li><a href='#plotTree.errorbars'><p>Plot a tree with error bars around divergence dates</p></a></li>
<li><a href='#plotTree.lollipop'><p>Plot a tree with one or more matched lollipop plots</p></a></li>
<li><a href='#plotTree.wBars'><p>Plot a tree with bars at the tips</p></a></li>
<li><a href='#posterior.evolrate'><p>Analysis of the posterior sample from evol.rate.mcmc</p></a></li>
<li><a href='#posthoc'><p>Generic post-hoc test</p></a></li>
<li><a href='#print.backbonePhylo'><p>Print method for backbone phylogeny</p></a></li>
<li><a href='#pscore'><p>Compute the parsimony score</p></a></li>
<li><a href='#ratebystate'><p>Method for investigating the rate of one trait as a function of the state of another</p></a></li>
<li><a href='#ratebytree'><p>Likelihood test for rate variation among trees, clades, or traits</p></a></li>
<li><a href='#rateshift'><p>Find the temporal position of one or more rate shifts</p></a></li>
<li><a href='#read.newick'><p>Newick or Nexus style tree reader</p></a></li>
<li><a href='#read.simmap'><p>Read SIMMAP style trees from file</p></a></li>
<li><a href='#reorder.backbonePhylo'><p>Reorders a backbone phylogeny</p></a></li>
<li><a href='#reorderSimmap'><p>Reorder edges of a <code>"simmap"</code> tree</p></a></li>
<li><a href='#rep.phylo'><p>Replicate a tree or set of trees</p></a></li>
<li><a href='#reroot'><p>Re-root a tree along an edge</p></a></li>
<li><a href='#rerootingMethod'><p>Get marginal ancestral state reconstructions by re-rooting</p></a></li>
<li><a href='#rescale'><p>Rescale phylogenetic objects of different types</p></a></li>
<li><a href='#rescale.simmap'><p>Rescale object of class <code>"simmap"</code></p></a></li>
<li><a href='#resolveNode'><p>Compute all possible resolutions of a node or all nodes in a multifurcating tree</p></a></li>
<li><a href='#rotateNodes'><p>Rotates a node or set of nodes in a phylogenetic tree</p></a></li>
<li><a href='#roundBranches'><p>Rounds the branch lengths of a tree</p></a></li>
<li><a href='#roundPhylogram'><p>Plot a round, sigmoidal, or spline phylogram or cladogram</p></a></li>
<li><a href='#rstate'><p>Pick a random state according to a vector of probabilities</p></a></li>
<li><a href='#sampleFrom'><p>Sample from a set of distributions</p></a></li>
<li><a href='#setMap'><p>Set color map for various phylogenetic objects of classes</p></a></li>
<li><a href='#sim.corrs'><p>Multivariate Brownian simulation with multiple correlations and rates</p></a></li>
<li><a href='#sim.history'><p>Simulate character history or a discrete character at the tips of the tree under some model</p></a></li>
<li><a href='#sim.ratebystate'><p>Conduct simulation of state dependent rate variation</p></a></li>
<li><a href='#sim.rates'><p>Brownian or OU simulation with multiple evolutionary regimes</p></a></li>
<li><a href='#simBMphylo'><p>Creates a graphical illustration of Brownian motion evolution on a phylogeny</p></a></li>
<li><a href='#skewers'><p>Matrix comparison using the method of random skewers</p></a></li>
<li><a href='#splitEdgeColor'><p>Split edge colors when descendant edges have different mapped states</p></a></li>
<li><a href='#splitplotTree'><p>Plots a phylogeny in two columns</p></a></li>
<li><a href='#splitTree'><p>Split tree at a point</p></a></li>
<li><a href='#starTree'><p>Create star phylogeny</p></a></li>
<li><a href='#strahlerNumber'><p>Computes Strahler number for trees and nodes</p></a></li>
<li><a href='#threshBayes'><p>Threshold model using Bayesian MCMC</p></a></li>
<li><a href='#threshDIC'><p>Deviance Information Criterion from the threshold model</p></a></li>
<li><a href='#threshState'><p>Computes value for a threshold character from a liability and thresholds</p></a></li>
<li><a href='#to.matrix'><p>Convert a character vector to a binary matrix</p></a></li>
<li><a href='#tree.grow'><p>Creates an animation of a tree growing from left-to-right or upwards</p></a></li>
<li><a href='#treeSlice'><p>Slices the tree at a particular point and returns all subtrees, or the tree rootward of the point</p></a></li>
<li><a href='#untangle'><p>Attempts to untangle crossing branches for plotting</p></a></li>
<li><a href='#vcvPhylo'><p>Calculates cophenetic (i.e., phylogenetic VCV) matrix</p></a></li>
<li><a href='#write.simmap'><p>Write a stochastic character mapped tree to file</p></a></li>
<li><a href='#writeAncestors'><p>Write a tree to file with ancestral states and (optionally) CIs at nodes</p></a></li>
<li><a href='#writeNexus'><p>Write a tree to file in Nexus format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.3-0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-13</td>
</tr>
<tr>
<td>Title:</td>
<td>Phylogenetic Tools for Comparative Biology (and Other Things)</td>
</tr>
<tr>
<td>Author:</td>
<td>Liam J. Revell</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Liam J. Revell &lt;liam.revell@umb.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), ape (&ge; 5.7), maps</td>
</tr>
<tr>
<td>Imports:</td>
<td>clusterGeneration, coda, combinat, DEoptim, doParallel, expm,
foreach, graphics, grDevices, MASS, methods, mnormt, nlme,
numDeriv, optimParallel, parallel, phangorn (&ge; 2.3.1),
scatterplot3d, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>animation, geiger, plotrix, RColorBrewer, rgl</td>
</tr>
<tr>
<td>ZipData:</td>
<td>no</td>
</tr>
<tr>
<td>Description:</td>
<td>A wide range of methods for phylogenetic analysis - concentrated in phylogenetic comparative biology, but also including numerous techniques for visualizing, analyzing, manipulating, reading or writing, and even inferring phylogenetic trees. Included among the functions in phylogenetic comparative biology are various for ancestral state reconstruction, model-fitting, and simulation of phylogenies and trait data. A broad range of plotting methods for phylogenies and comparative data include (but are not restricted to) methods for mapping trait evolution on trees, for projecting trees into phenotype space or a onto a geographic map, and for visualizing correlated speciation between trees. Lastly, numerous functions are designed for reading, writing, analyzing, inferring, simulating, and manipulating phylogenetic trees and comparative data. For instance, there are functions for computing consensus phylogenies from a set, for simulating phylogenetic trees and data under a range of models, for randomly or non-randomly attaching species or clades to a tree, as well as for a wide range of other manipulations and analyses that phylogenetic biologists might find useful in their research.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/liamrevell/phytools">https://github.com/liamrevell/phytools</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-13 21:43:19 UTC; liamj</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-13 22:50:02 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='phytools-package'>
phytools: Phylogenetic Tools for comparative biology (and other things)
</h2><span id='topic+phytools-package'></span><span id='topic+phytools'></span>

<h3>Description</h3>

<p><span class="pkg">phytools</span> provides functions for phylogenetic  biology: focused on phylogenetic comparative analysis, but including methods to simulate data and trees, to visualize trees and fitted models, and to analyze and manipulate phylogenetic trees and data.
</p>
<p>The complete list of functions can be displayed with <code>library(help=phytools)</code>.
</p>
<p>The <span class="pkg">phytools</span> development page is <a href="https://github.com/liamrevell/phytools/">https://github.com/liamrevell/phytools/</a>. More information on <span class="pkg">phytools</span> can also be found at <a href="http://blog.phytools.org">http://blog.phytools.org</a> or <a href="http://www.phytools.org">http://www.phytools.org</a>.
</p>
<p>If you use <span class="pkg">phytools</span> (or other packages that depend on <span class="pkg">phytools</span>) in a publication, please <em>cite it</em>.
</p>
<p>The appropriate citation for <span class="pkg">phytools</span> is listed below or can be obtained using <code>citation("phytools")</code> with the package installed.
</p>


<h3>Author(s)</h3>

<p>Liam J. Revell
</p>
<p>Maintainer: Liam J. Revell &lt;liam.revell@umb.edu&gt;
</p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='add.arrow'>Add an arrow pointing to a tip or node on the tree</h2><span id='topic+add.arrow'></span>

<h3>Description</h3>

<p>Adds an arrow or a set of arrows to a plotted tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.arrow(tree=NULL, tip, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.arrow_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>, <code>"contMap"</code>, or <code>"densityMap"</code>. If not supplied, the function will obtain the last plotted phylogeny from the environmental variable <code>last_plot.phylo</code>.</p>
</td></tr>
<tr><td><code id="add.arrow_+3A_tip">tip</code></td>
<td>
<p>label of tip or tip or node number; or vector of such values. If <code>tree=NULL</code> then the tip label(s) or node number(s) must be supplied.</p>
</td></tr>
<tr><td><code id="add.arrow_+3A_...">...</code></td>
<td>
<p>optional arguments to control the shape and size of the arrow including: its length (<code>arrl</code>) in the units of the plot; the length of the arrowhead (<code>hedl</code>); the total angle between the wings in the arrowhead (<code>angle</code>); the line width for the plotted lines (<code>lwd</code>); the offset from the tip or end of tip label, in character widths (<code>offset</code>); and the color (<code>col</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Trees can be plotted using <span class="pkg">phytools</span> function <code>plotTree</code>, <code>plotSimmap</code>, <code>contMap</code>, <code>densityMap</code>, and <span class="pkg">ape</span> S3 method <code>plot.phylo</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+nodelabels">nodelabels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## show arrows with a black outline
data(anoletree)
plotTree(anoletree,type="fan",fsize=0.7,ftype="i")
add.arrow(anoletree,tip=c("cuvieri","krugi",
    "pulchellus","poncensis","stratulus",
    "evermanni","cooki","cristatellus",
    "gundlachi","occultus"),lwd=5,arrl=1)
add.arrow(anoletree,tip="cuvieri",col="green",
    lwd=3,arrl=1)
add.arrow(anoletree,tip=c("krugi","pulchellus",
    "poncensis"),col="#E4D96F",lwd=3,arrl=1)
add.arrow(anoletree,tip=c("stratulus","evermanni"),
    col="darkgreen",lwd=3,arrl=1)
add.arrow(anoletree,tip=c("cooki","cristatellus",
    "gundlachi"),col="brown",lwd=3,arrl=1)
add.arrow(anoletree,tip="occultus",col="darkgrey",
    lwd=3,arrl=1)
legend(x="topleft",c("crown-giant","grass-bush","trunk-crown","trunk-ground",
    "twig"),pch=22,pt.bg=c("green","#E4D96F","darkgreen",
    "brown","darkgrey"),cex=0.9,
    pt.cex=2,title="PR ecomorphs",bty="n")
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='add.color.bar'>Add color bar to a plot</h2><span id='topic+add.color.bar'></span>

<h3>Description</h3>

<p>This function adds a color gradient legend to an existing plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.color.bar(leg, cols, title=NULL, lims=c(0,1), digits=1, prompt=TRUE,
   lwd=4, outline=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.color.bar_+3A_leg">leg</code></td>
<td>
<p>numerical value for the length of the legend.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_cols">cols</code></td>
<td>
<p>colors for the legend.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_title">title</code></td>
<td>
<p>text to plot above the bar.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_lims">lims</code></td>
<td>
<p>range for the bar.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_digits">digits</code></td>
<td>
<p>digits for plotted numbers.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_prompt">prompt</code></td>
<td>
<p>logical value indicating whether the location of the legend should be obtained interactively.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_lwd">lwd</code></td>
<td>
<p>width of the plotted bar.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_outline">outline</code></td>
<td>
<p>logical value indicated whether or not to outline the plotted color bar with a 1 pt line.</p>
</td></tr>
<tr><td><code id="add.color.bar_+3A_...">...</code></td>
<td>
<p>optional arguments including: <code>x</code> x-coordinate of the legend (if <code>prompt=FALSE</code>); <code>y</code> y-coordinate of the legend; <code>subtitle</code> optional legend subtitle; <code>direction</code> direction of the color bar (i.e., increase from left to right or from right to left); and <code>fsize</code>, font size for the legend text.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function was designed to add a color bar to a plot created by <code><a href="#topic+plotBranchbyTrait">plotBranchbyTrait</a></code>. A color bar can be added by clicking on a location within the plot (when <code>prompt=TRUE</code>) or by setting <code>prompt=FALSE</code> and supplying x &amp; y coordinates for the object. 
</p>
<p>The function is also used internally by S3 methods <code><a href="#topic+plot.contMap">plot.contMap</a></code> and <code><a href="#topic+plot.densityMap">plot.densityMap</a></code>, as well as by <code><a href="#topic+errorbar.contMap">errorbar.contMap</a></code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2013) Two new graphical methods for mapping trait evolution on phylogenies. <em>Methods in Ecology and Evolution</em>, <b>4</b>, 754-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contMap">contMap</a></code>, <code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="#topic+errorbar.contMap">errorbar.contMap</a></code>, <code><a href="#topic+plotBranchbyTrait">plotBranchbyTrait</a></code>
</p>

<hr>
<h2 id='add.everywhere'>Add tip to all edges in a tree</h2><span id='topic+add.everywhere'></span>

<h3>Description</h3>

<p>This function adds a tip to all branches of the tree and returns a a list of trees as an object of class <code>"multiPhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.everywhere(tree, tip.name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.everywhere_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="add.everywhere_+3A_tip.name">tip.name</code></td>
<td>
<p>a string containing the name of the tip to add.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of trees as an object of class <code>"multiPhylo"</code>.  Since the tip can be added to any branch, the length of the list is equal to the number of edges in <code>tree</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+allFurcTrees">allFurcTrees</a></code>, <code><a href="#topic+exhaustiveMP">exhaustiveMP</a></code>
</p>

<hr>
<h2 id='add.random'>Add tips at random to the tree</h2><span id='topic+add.random'></span>

<h3>Description</h3>

<p>This function adds new tips at random to a tree with branch lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.random(tree, n=NULL, tips=NULL, edge.length=NULL, order=c("random","input"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.random_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="add.random_+3A_n">n</code></td>
<td>
<p>a number of tips to add to the tree. If <code>NULL</code>, will use <code>length(tips)</code>. If <code>tips</code> is specified, then <code>n</code> is also ignored, regardless of its value.</p>
</td></tr>
<tr><td><code id="add.random_+3A_tips">tips</code></td>
<td>
<p>a set of tip names for the added tips. If <code>NULL</code>, names will be supplied</p>
</td></tr>
<tr><td><code id="add.random_+3A_edge.length">edge.length</code></td>
<td>
<p>terminal edge length for the added tips. If <code>NULL</code>, and <code>is.ultrametric(tree)</code> <code>==TRUE</code>, then edge lengths will be assigned to keep the tree ultrametric. Note that if edge lengths are assigned and <code>n&gt;1</code>, then the assigned terminal edge lengths are not guaranteed as subsequent random tip addition could occur along the new terminal edge.</p>
</td></tr>
<tr><td><code id="add.random_+3A_order">order</code></td>
<td>
<p>addition order for the new tips.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If no edge lengths are provided, and the tree is ultrametric, then edge lengths are assigned to keep the tree ultrametric.
</p>
<p>The probability that at new tip is added along any branch is directly proportional to the length of the branch.
</p>
<p>Note that sometimes the resultant tree plotted with <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> or <code><a href="#topic+plotSimmap">plotSimmap</a></code> may display with branches crossing. If so, the tree can be 'untangled' using <code><a href="#topic+untangle">untangle</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+allFurcTrees">allFurcTrees</a></code>, <code><a href="#topic+add.everywhere">add.everywhere</a></code>
</p>

<hr>
<h2 id='add.simmap.legend'>Add legend to stochastically mapped tree</h2><span id='topic+add.simmap.legend'></span>

<h3>Description</h3>

<p>Adds a legend (by default, interactively) to a plotted stochastic character mapped tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.simmap.legend(leg=NULL, colors, prompt=TRUE, vertical=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.simmap.legend_+3A_leg">leg</code></td>
<td>
<p>states for the discrete character in the order of <code>colors</code>.</p>
</td></tr>
<tr><td><code id="add.simmap.legend_+3A_colors">colors</code></td>
<td>
<p>colors for the legend in the order of <code>leg</code>, or, if <code>leg=NULL</code>, named vector of colors in which <code>names(colors</code> are the states of the mapped discrete character.</p>
</td></tr>
<tr><td><code id="add.simmap.legend_+3A_prompt">prompt</code></td>
<td>
<p>logical value indicating whether the location of the legend should be obtained interactively (i.e., by clicking in the plotting area).</p>
</td></tr>
<tr><td><code id="add.simmap.legend_+3A_vertical">vertical</code></td>
<td>
<p>logical value indicating whether to plot the legend vertically (if <code>TRUE</code>) or horizontally.</p>
</td></tr>
<tr><td><code id="add.simmap.legend_+3A_...">...</code></td>
<td>
<p>optional arguments including: <code>x</code> x-coordinate of the legend (if <code>prompt=FALSE</code>); <code>y</code> y-coordinate of the legend; and <code>shape</code> which can be <code>shape="square"</code>, the default, or <code>shape="circle"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Redundant with <code><a href="graphics.html#topic+legend">legend</a></code>, which is much more powerful but somewhat more complicated to use.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>

<hr>
<h2 id='add.species.to.genus'>Add species to genus on a phylogeny or bind simulated species subtrees to a backbone genus tree</h2><span id='topic+add.species.to.genus'></span><span id='topic+genus.to.species.tree'></span>

<h3>Description</h3>

<p><code>add.species.to.genus</code> adds an additional species to a genus on a phylogeny. 
</p>
<p><code>genus.to.species.tree</code> simulates pure-birth subtrees and then binds them at a random height along the terminal edge leading to each corresponding genus on a genus-level backbone tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.species.to.genus(tree, species, genus=NULL, where=c("root","random"))
genus.to.species.tree(tree, species)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.species.to.genus_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>. In the case of <code>genus.to.species.tree</code> this should be a genus-level backbone tree.</p>
</td></tr>
<tr><td><code id="add.species.to.genus_+3A_species">species</code></td>
<td>
<p>string contain species name in the format <code>"Genus_species"</code> or <code>"Genus species"</code>.</p>
</td></tr>
<tr><td><code id="add.species.to.genus_+3A_genus">genus</code></td>
<td>
<p>for <code>add.species.to.genus</code>, optional argument containing the genus to which <code>species</code> is to be attached. If <code>NULL</code> then <code>genus</code> will be extracted from <code>species</code>.</p>
</td></tr>
<tr><td><code id="add.species.to.genus_+3A_where">where</code></td>
<td>
<p>for <code>add.species.to.genus</code>, the location to attach <code>species</code> to the tree. <code>where=</code> <code>"root"</code> will cause the species to be attached to the MRCA of all members of <code>genus</code>. <code>where="random"</code> will cause <code>species</code> to be attached at random to the subtree descended from the MRCA of all members of <code>genus</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>add.species.to.genus</code>, if <code>genus</code> contains only one species and <code>where="root"</code>, then <code>species</code> will be attached midway along the branch leading to the one species. If <code>where="random"</code> then <code>species</code> will be added at a random position along the edge. 
</p>
<p>If <code>genus</code> cannot be found in the tree, then the original tree is returned and a warning printed. If the tree is not ultrametric, then the resultant tree may not contain branch lengths and a warning will be printed. 
</p>
<p>Note that for some cases in which a tree is read in from file, R may initially think it is ultrametric, but then later (as tips are added) decide that it is not due to rounding of the edge lengths when it was written to file. This can most likely be resolved by using <code><a href="#topic+force.ultrametric">force.ultrametric</a></code> to coerce the tree to be exactly ultrametric (to the maximum numerical precision permitted by R) before adding tips to the tree.
</p>
<p>If <code>genus</code> is non-monophyletic then <code>species</code> will be attached to the most inclusive group containing members of <code>genus</code> and a warning will be printed.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add.random">add.random</a></code>, <code><a href="#topic+bind.tip">bind.tip</a></code>
</p>

<hr>
<h2 id='aic.w'>Computes Akaike weights</h2><span id='topic+aic.w'></span>

<h3>Description</h3>

<p>Computes Akaike weights based on a set of AIC values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aic.w(aic)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aic.w_+3A_aic">aic</code></td>
<td>
<p>vector of Akaike Information Criterion (AIC; Akaike, 1974) values for different fitted models. If the vector has names, these names will be inherited by the vector returned by the function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes Akaike weights from a set of AIC values obtained from fitted models.
</p>
<p>It returns an object of class <code>"aic.w"</code> which is just a vector which allows it to be automatically printed with a numerical precision of 8 significant digits.
</p>


<h3>Value</h3>

<p>A vector of Akaike weights.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Akaike, H. (1974) A new look at the statistical model identification. <em>IEEE Transactions on Automatic Control</em>, <b>19</b>, 716-723.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='allFurcTrees'>Generate all bi- and multifurcating unrooted trees</h2><span id='topic+allFurcTrees'></span>

<h3>Description</h3>

<p>This function creates all possible unrooted bi- and multifurcating trees and returns a list of trees as an object of class <code>"multiPhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allFurcTrees(n, tip.label=NULL, to.plot=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allFurcTrees_+3A_n">n</code></td>
<td>
<p>an integer giving the desired number of species.</p>
</td></tr>
<tr><td><code id="allFurcTrees_+3A_tip.label">tip.label</code></td>
<td>
<p>an optional vector of length n containing the tip names.</p>
</td></tr>
<tr><td><code id="allFurcTrees_+3A_to.plot">to.plot</code></td>
<td>
<p>an optional logical value indicating whether or not to plot the trees.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should be used with extreme caution for <code>n</code> greater than about 8, as in this case the number of possible trees is extremely large.
</p>


<h3>Value</h3>

<p>A list of trees as an object of class <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. 2004. <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add.everywhere">add.everywhere</a></code>, <code><a href="#topic+exhaustiveMP">exhaustiveMP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## compute &amp; plot all bi- and multifurcating
## trees for six taxa
trees&lt;-allFurcTrees(n=6)
par(mfrow=c(16,15))
nulo&lt;-sapply(trees,plot,type="unrooted",
    no.margin=TRUE)
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='anc.Bayes'>Bayesian ancestral character estimation</h2><span id='topic+anc.Bayes'></span><span id='topic+plot.anc.Bayes'></span><span id='topic+density.anc.Bayes'></span>

<h3>Description</h3>

<p>This function uses Bayesian MCMC to sample from the posterior distribution for the states at internal nodes in the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anc.Bayes(tree, x, ngen=10000, control=list(), ...)
## S3 method for class 'anc.Bayes'
plot(x, ...)
## S3 method for class 'anc.Bayes'
density(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anc.Bayes_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="anc.Bayes_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species; <code>names(x)</code> should be the species names. In the case of the <code>plot</code> and <code>density</code> methods, an object of class <code>"anc.Bayes"</code>.</p>
</td></tr>
<tr><td><code id="anc.Bayes_+3A_ngen">ngen</code></td>
<td>
<p>a integer indicating the number of generations for the MCMC.</p>
</td></tr>
<tr><td><code id="anc.Bayes_+3A_control">control</code></td>
<td>
<p>a list of control parameters containing the following elements: <code>sig2</code>: starting value for <code class="reqn">\sigma^2</code> (BM rate); <code>a</code>: starting for the state at the root node; <code>y</code>: starting values for the states at all internal nodes excluding the root (should be labeled with node numbers); <code>pr.mean</code>: means for the prior distributions in the following order - <code>sig2</code>, <code>a</code>, <code>y</code>, note that the prior probability distribution is exponential for <code>sig2</code> and normal for <code>a</code> and <code>y</code>; <code>pr.var</code>: variances on the prior distributions, same order as <code>pr.mean</code> (but the variance is not used for <code>sig2</code>); <code>prop</code>: variances on the normal proposal distributions in the same order as <code>pr.mean</code>; <code>sample</code>: sample frequency from the MCMC.</p>
</td></tr>
<tr><td><code id="anc.Bayes_+3A_...">...</code></td>
<td>
<p>optional arguments, including to be passed to <code>plot</code> and <code>density</code> methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plot.anc.Bayes</code> generates a likelihood profile plot of the MCMC by default, but can also create a profile plot for any of the sampled variables by using the optional argument <code>what</code>. For instance, <code>what=40</code> (or, equivalently, <code>what="40"</code>) will create a profile plot of the MCMC for node <code>40</code>. Additional arguments are passed to <code><a href="base.html#topic+plot">plot</a></code>.
</p>
<p><code>density.anc.Bayes</code> computes a posterior density from the MCMC sample. Like <code>plot.anc.Bayes</code> takes the optional argument <code>what</code>, but unlike <code>plot.anc.Bayes</code> computes the posterior density for the root node by default. The object computed by this function is of class <code>"density"</code> and can be visualized using <code><a href="stats.html#topic+plot.density">plot.density</a></code>. Burn-in (in generations) can be set using the optional argument <code>burnin</code>, otherwise it will be assumed to be 20
</p>
<p>The <code>print</code> and <code>summary</code> methods for this object class also return (invisibly) a vector of estimated ancestral states based on a user-supplied burn-in (or 20% of the number of generations of MCMC, if no burn-in is provided). Burn-in can be specified with the optional argument <code>burnin</code>.
</p>


<h3>Value</h3>

<p><code>anc.Bayes</code> returns an object of class <code>"anc.Bayes"</code> including at least two components: 
</p>
<table>
<tr><td><code>mcmc</code></td>
<td>
<p>a data frame with rows <code>ngen/sample+1</code> containing the posterior sample and likelihoods.  Matrix columns are labeled either <code>sig2</code> or by the node number of the internal node.</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>our input phylogeny.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+anc.ML">anc.ML</a></code>, <code><a href="#topic+anc.trend">anc.trend</a></code>, <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>, <code><a href="#topic+fastAnc">fastAnc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## set seed
set.seed(77)
## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
    rownames(mammal.data)))
## run MCMC (should be run at least 1e6 generations)
mcmc&lt;-anc.Bayes(mammal.tree,ln.bodyMass,
    ngen=50000)
print(mcmc,printlen=20) ## estimates
par(mfrow=c(2,1))
plot(mcmc,bty="l",main="Likelihood-profile from MCMC",
    font.main=3) ## likelihood-profile
plot(density(mcmc,what=Ntip(mammal.tree)+1,
    burnin=20000),bty="l",
    main="Posterior density for root state of log(body mass)",
    font.main=3)
par(mfrow=c(1,1)) ## reset par to default
</code></pre>

<hr>
<h2 id='anc.ML'>Ancestral character estimation using likelihood</h2><span id='topic+anc.ML'></span>

<h3>Description</h3>

<p>This function estimates the evolutionary parameters and ancestral states for Brownian evolution using likelihood. It is also possible (for <code>model="BM"</code>) to allow for missing data for some tip taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anc.ML(tree, x, maxit=2000, model=c("BM","OU","EB"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anc.ML_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="anc.ML_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species; <code>names(x)</code> should be the species names.</p>
</td></tr>
<tr><td><code id="anc.ML_+3A_maxit">maxit</code></td>
<td>
<p>an optional integer value indicating the maximum number of iterations for optimization.</p>
</td></tr>
<tr><td><code id="anc.ML_+3A_model">model</code></td>
<td>
<p>model of continuous character evolution on the tree. It's possible that only <code>model="BM"</code> &amp; <code>model="EB"</code> work in the present version as <code>model="OU"</code> has not be thoroughly tested &amp; some bugs were reported for an earlier version.</p>
</td></tr>
<tr><td><code id="anc.ML_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because this function relies on a high dimensional numerical optimization of the likelihood function, <code><a href="#topic+fastAnc">fastAnc</a></code> should probably be preferred for most purposes. If using <code><a href="#topic+anc.ML">anc.ML</a></code>, users should be cautious to ensure convergence. 
</p>
<p>This issue has been ameliorated in <span class="pkg">phytools</span> &gt;= 0.2-48 by seeding the ML optimization with the result from <code><a href="#topic+fastAnc">fastAnc</a></code>.
</p>
<p>For <code>model="EB"</code> this should also not be a problem as the numerical optimization is performed for only <code>sig2</code> and <code>r</code>, while the ML values of the ancestral states are obtained during every iteration of the optimization algorithmically using the re-rooting method.
</p>


<h3>Value</h3>

<p>An object of class <code>"anc.ML"</code> with at least the following four elements (if not more, depending on <code>model</code>):
</p>
<table>
<tr><td><code>sig2</code></td>
<td>
<p>the variance of the BM process.</p>
</td></tr>
<tr><td><code>ace</code></td>
<td>
<p>a vector with the ancestral states.</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>the log-likelihood.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>the value of <code>convergence</code> returned by <code><a href="stats.html#topic+optim">optim</a></code> (0 is good).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Schluter, D., Price, T., Mooers, A. O., and Ludwig, D. (1997) Likelihood of ancestor states in adaptive radiation. <em>Evolution</em> <b>51</b>, 1699-1711.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+fastAnc">fastAnc</a></code>, <code><a href="stats.html#topic+optim">optim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
    rownames(mammal.data)))
## estimate ancestral state under BM model
fit.BM&lt;-anc.ML(mammal.tree,ln.bodyMass)
print(fit.BM)
</code></pre>

<hr>
<h2 id='anc.trend'>Ancestral character estimation with a trend</h2><span id='topic+anc.trend'></span>

<h3>Description</h3>

<p>This function estimates the evolutionary parameters and ancestral states for Brownian evolution with a directional trend.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anc.trend(tree, x, maxit=2000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anc.trend_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="anc.trend_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species; <code>names(x)</code> should be the species names.</p>
</td></tr>
<tr><td><code id="anc.trend_+3A_maxit">maxit</code></td>
<td>
<p>an optional integer value indicating the maximum number of iterations for optimization.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this will generally only work and produce sensible results for a phylogeny with some non-contemporaneous tips (i.e., a tree with some fossil species).
</p>
<p>The function uses <code><a href="stats.html#topic+optim">optim</a></code> with <code>method=</code> <code>"L-BFGS-B"</code> internally; however, optimization is only constrained for the <code>sig2</code> which must be &gt; 0.
</p>


<h3>Value</h3>

<p>An object of class <code>"anc.trend"</code> with the following components:
</p>
<table>
<tr><td><code>ace</code></td>
<td>
<p>a vector with the ancestral states.</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>a trend parameter per unit time.</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>the variance of the BM process, <code class="reqn">\sigma^2</code>.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>the log-likelihood.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>the value of <code>$convergence</code> returned by <code>optim()</code> (0 is good).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+anc.ML">anc.ML</a></code>, <code><a href="stats.html#topic+optim">optim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate tree &amp; data using fastBM with a trend (m!=0)
tree&lt;-rtree(n=26,tip.label=LETTERS)
x&lt;-fastBM(tree,mu=4,internal=TRUE)
a&lt;-x[as.character(1:tree$Nnode+Ntip(tree))]
x&lt;-x[tree$tip.label]
## fit no trend model
fit.bm&lt;-anc.ML(tree,x,model="BM")
print(fit.bm)
## fit trend model
fit.trend&lt;-anc.trend(tree,x)
print(fit.trend)
## compare trend vs. no-trend models &amp; estimates
AIC(fit.bm,fit.trend)
layout(matrix(c(3,4,1,2,5,6),3,2,byrow=TRUE),
    heights=c(1.5,3,1.5),widths=c(3,3))
xlim&lt;-ylim&lt;-range(c(a,fit.bm$ace,
    fit.trend$ace))
plot(a,fit.bm$ace,pch=19,
    col=make.transparent("blue",0.5),
    xlab="true ancestral states",
    ylab="ML estimates",
    main=paste("Comparison of true and estimated",
    "\nstates under a no-trend model"),font.main=3,
    cex.main=1.2,bty="l",cex=1.5,
    xlim=xlim,ylim=ylim)
lines(xlim,ylim,lty="dotted")
plot(a,fit.trend$ace,pch=19,
    col=make.transparent("blue",0.5),
    xlab="true ancestral states",
    ylab="ML estimates",
    main=paste("Comparison of true and estimated",
    "\nstates under a trend model"),font.main=3,
    cex.main=1.2,bty="l",cex=1.5,
    xlim=xlim,ylim=ylim)
lines(xlim,ylim,lty="dotted")
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='ancr'>Compute marginal or joint ancestral state estimates</h2><span id='topic+ancr'></span><span id='topic+hide.hidden'></span><span id='topic+plot.ancr'></span>

<h3>Description</h3>

<p>By default, <code>ancr</code> computes marginal ancestral states, also known as empirical Bayes posterior probabilities, conditioning on the fitted (or set) model of <code>object</code>.
</p>
<p>Can also perform <em>joint</em> ancestral state estimation, if the optional argument <code>type</code> is set to <code>type="joint"</code>.
</p>
<p><code>hide.hidden</code> merges hidden states (if any).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancr(object, ...)
hide.hidden(object, ...)
## S3 method for class 'ancr'
plot(x, args.plotTree=list(...), args.nodelabels=list(...), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ancr_+3A_object">object</code></td>
<td>
<p>fitted M<em>k</em> model (presently object of class <code>"fitMk"</code>, <code>"fitpolyMk"</code>, and <code>"fitHRM"</code>), or a set of models in the form of a table from <code>anova</code> comparison. For <code>hide.hidden</code>, object of class <code>"ancr"</code>.</p>
</td></tr>
<tr><td><code id="ancr_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
<tr><td><code id="ancr_+3A_x">x</code></td>
<td>
<p>in the case of <code>plot.ancr</code>, an object of class <code>"ancr"</code>.</p>
</td></tr>
<tr><td><code id="ancr_+3A_args.plottree">args.plotTree</code></td>
<td>
<p>arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code>, in a list.</p>
</td></tr>
<tr><td><code id="ancr_+3A_args.nodelabels">args.nodelabels</code></td>
<td>
<p>arguments to be passed to <code><a href="ape.html#topic+nodelabels">nodelabels</a></code>, in a list.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the optional argument <code>tips=TRUE</code>, then the matrix returned contains empirical Bayes posterior probabilities (marginal scaled likelihoods) for both tips <em>and</em> internal nodes. Otherwise (the default) only node marginal states are returned.
</p>
<p>If the input object is a set of models (in the form of an <code>anova</code> table), then <code>ancr</code> will compute model-averaged marginal ancestral states (for <code>type="marginal"</code>, unless the optional argument <code>weighted=FALSE</code>, in which case only the best-supported model is used.
</p>


<h3>Value</h3>

<p>An object of class <code>"ancr"</code>. In the case of <code>type="marginal"</code>, this object consists of a matrix of marginal (empirical Bayes) probabilities and a likelihood. In the case of <code>type="joint"</code>, the object contains a set of most-likely internal node states stored in a data frame.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Pagel, M. (1999) The Maximum Likelihood approach to reconstructing ancestral character states of discrete characters on phylogenies. <em>Systematic Biology</em>, <b>3</b>, 612-622.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitMk">fitMk</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load tree and data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract discrete character (feeding mode)
fmode&lt;-setNames(sunfish.data$feeding.mode,
  rownames(sunfish.data))
## fit ARD model
ard_fmode&lt;-fitMk(sunfish.tree,fmode,model="ARD",
  pi="fitzjohn")
## compute ancestral states
anc_fmode&lt;-ancr(ard_fmode)
## plot the results
par(mfrow=c(2,1))
cols&lt;-setNames(c("blue","red"),levels(fmode))
plot(anc_fmode,
  args.plotTree=list(lwd=2,direction="upwards",
    mar=c(0.1,1.1,2.1,1.1),fsize=0.8),
  args.nodelabels=list(piecol=cols),
  args.tiplabels=list(cex=0.3),
  legend="bottomright")
mtext("a) marginal states under ARD model",adj=0)
## fit ER model
er_fmode&lt;-fitMk(sunfish.tree,fmode,model="ER",
  pi="fitzjohn")
## compare models
aov_fmode&lt;-anova(er_fmode,ard_fmode)
## compute model-averaged ancestral states
anc_fmode_model.averaged&lt;-ancr(aov_fmode)
plot(anc_fmode_model.averaged,
  args.plotTree=list(lwd=2,direction="upwards",
    mar=c(0.1,1.1,2.1,1.1),fsize=0.8),
  args.nodelabels=list(piecol=cols),
  args.tiplabels=list(cex=0.3),
  legend="bottomright")
mtext("b) marginal states model-averaging ER &amp; ARD models",
  adj=0)
par(mar=c(5.1,4.1,4.1,2.1),mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='ancThresh'>Ancestral character estimation under the threshold model using Bayesian MCMC</h2><span id='topic+ancThresh'></span>

<h3>Description</h3>

<p>This function uses Bayesian MCMC to estimate ancestral states and thresholds for a discrete character under the threshold model from quantitative genetics (Felsenstein 2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancThresh(tree, x, ngen=100000, sequence=NULL, method="mcmc", 
   model=c("BM","OU","lambda"), control=list(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ancThresh_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_x">x</code></td>
<td>
<p>a named vector containing discrete character states; or a matrix containing the tip species, in rows, and probabilities of being in each state, in columns.</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_ngen">ngen</code></td>
<td>
<p>number of generations to run the MCMC.</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_sequence">sequence</code></td>
<td>
<p>assumed ordering of the discrete character state. If not supplied and <code>x</code> is a vector then numerical/alphabetical order is assumed; if not supplied and <code>x</code> is a matrix, then the column order of <code>x</code> is used.</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_method">method</code></td>
<td>
<p>only method currently available is <code>"mcmc"</code>.</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_model">model</code></td>
<td>
<p>model for the evolution of the liability. Options are <code>"BM"</code> (Brownian motion, the default), <code>"OU"</code> (Ornstein-Uhlenbeck), or <code>"lambda"</code> (the <code class="reqn">\lambda</code> model).</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_control">control</code></td>
<td>
<p>list containing the following elements: <code>sample</code>, the sampling interval; <code>propliab</code> variance of the proposal distribution for liabilities; <code>propthresh</code> variance on the proposal distribution for the thresholds; <code>propalpha</code> variance on the proposal distribution for <code>alpha</code> (for <code>model="OU"</code>); <code>pr.anc</code> prior probability distribution on the ancestral states for each node, in a matrix - not all nodes need to be supplied; <code>pr.th</code> prior density on the thresholds; <code>burnin</code> number of generations to exclude for burn-in when plotting posterior probabilities on the tree; <code>plot</code> logical value indicating whether or not to plot the posterior probabilities; <code>print</code> logical value indicating whether or not to print the state of the MCMC; <code>piecol</code> colors for the posterior probabilities plotted as pie charts at internal nodes; and <code>tipcol</code> which indicates whether the tip colors should be based on the input data (<code>"input"</code>) or sampled tip liabilities (<code>"estimated"</code>). These will only differ if there is uncertainty in the tip states.</p>
</td></tr>
<tr><td><code id="ancThresh_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to <code><a href="#topic+plotThresh">plotThresh</a></code> (called internally).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>According to the threshold model from evolutionary quantitative genetics, values for our observed discrete character are determined by an unseen continuous trait, normally referred to as liability. Every time the value for liability crosses a threshold, the observed discrete character changes in state.
</p>
<p>Felsenstein (2012) first had the insight that this model could be used to study the evolution of discrete character traits on a reconstructed phylogenetic tree.
</p>
<p>This function uses Bayesian MCMC to sample ancestral liabilities and thresholds for a discrete character evolution under the threshold model.
</p>
<p><code>print</code> and <code>plot</code> S3 methods are now available for the object class <code>"ancThresh"</code>.
</p>


<h3>Value</h3>

<p>This function returns an object of class <code>"ancThresh"</code> containing the posterior sample from our analysis, along with other components.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (2012) A comparative method for both discrete and continuous characters using the threshold model. <em>American Naturalist</em>, <b>179</b>, 145-156.
</p>
<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+threshBayes">threshBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## load data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract character of interest
fmode&lt;-setNames(sunfish.data$feeding.mode,
    rownames(sunfish.data))
## run MCMC
mcmc&lt;-ancThresh(sunfish.tree,fmode,ngen=1000000)
## plot results
plot(mcmc,mar=c(0.1,0.1,4.1,0.1))
title(main="Posterior probabilities for node states",
    font.main=3)
## End(Not run)
</code></pre>

<hr>
<h2 id='anoletree'>Phylogenetic datasets</h2><span id='topic+anole.data'></span><span id='topic+anoletree'></span><span id='topic+ant.geog'></span><span id='topic+ant.tree'></span><span id='topic+bat.tree'></span><span id='topic+bat_virus.data'></span><span id='topic+betaCoV.tree'></span><span id='topic+bonyfish.data'></span><span id='topic+bonyfish.tree'></span><span id='topic+butterfly.data'></span><span id='topic+butterfly.tree'></span><span id='topic+cordylid.data'></span><span id='topic+cordylid.tree'></span><span id='topic+darter.tree'></span><span id='topic+eel.data'></span><span id='topic+eel.tree'></span><span id='topic+elapidae.tree'></span><span id='topic+flatworm.data'></span><span id='topic+flatworm.tree'></span><span id='topic+liolaemid.data'></span><span id='topic+liolaemid.tree'></span><span id='topic+mammal.data'></span><span id='topic+mammal.geog'></span><span id='topic+mammal.tree'></span><span id='topic+primate.data'></span><span id='topic+primate.tree'></span><span id='topic+salamanders'></span><span id='topic+sunfish.data'></span><span id='topic+sunfish.tree'></span><span id='topic+tortoise.geog'></span><span id='topic+tortoise.tree'></span><span id='topic+tropidurid.data'></span><span id='topic+tropidurid.tree'></span><span id='topic+vertebrate.data'></span><span id='topic+vertebrate.tree'></span><span id='topic+wasp.data'></span><span id='topic+wasp.trees'></span><span id='topic+whale.tree'></span>

<h3>Description</h3>

<p>Various phylogenetic datasets for comparative analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(anole.data)
data(anoletree)
data(ant.geog)
data(ant.tree)
data(bat.tree)
data(bat_virus.data)
data(betaCoV.tree)
data(bonyfish.data)
data(bonyfish.tree)
data(butterfly.data)
data(butterfly.tree)
data(cordylid.data)
data(cordylid.tree)
data(darter.tree)
data(eel.data)
data(eel.tree)
data(elapidae.tree)
data(flatworm.data)
data(flatworm.tree)
data(liolaemid.tree)
data(mammal.data)
data(mammal.geog)
data(mammal.tree)
data(primate.data)
data(primate.tree)
data(salamanders)
data(sunfish.data)
data(sunfish.tree)
data(tortoise.geog)
data(tortoise.tree)
data(tropidurid.data)
data(tropidurid.tree)
data(vertebrate.data)
data(vertebrate.tree)
data(wasp.data)
data(wasp.trees)
data(whale.tree)
</code></pre>


<h3>Format</h3>

<p><code>anoletree</code> is an object of class <code>"simmap"</code>. <code>anole.data</code> is a data frame.
</p>
<p><code>ant.tree</code> is an object of class <code>"phylo"</code>. <code>ant.geog</code> is a vector.
</p>
<p><code>bat.tree</code> and <code>betaCoV.tree</code> are objects of class <code>"phylo"</code>. <code>bat_virus.data</code> is a data frame.
</p>
<p><code>bonyfish.tree</code> is an object of class <code>"phylo"</code>. <code>bonyfish.data</code> is a data frame.
</p>
<p><code>butterfly.tree</code> is an object of class <code>"phylo"</code>. <code>butterfly.data</code> is a data frame.
</p>
<p><code>cordylid.tree</code> is an object of class <code>"phylo"</code>. <code>cordylid.data</code> is a data frame.
</p>
<p><code>darter.tree</code> is an object of class <code>"phylo"</code>.
</p>
<p><code>eel.tree</code> is an object of class <code>"phylo"</code>. <code>eel.data</code> is a data frame.
</p>
<p><code>elapidae.tree</code> is an object of class <code>"phylo"</code>.
</p>
<p><code>flatworm.tree</code> is an object of class <code>"phylo"</code>. <code>flatworm.data</code> is a data frame.
</p>
<p><code>liolaemid.tree</code> is an object of class <code>"phylo"</code>. <code>liolaemid.data</code> is a data frame.
</p>
<p><code>mammal.tree</code> is an object of class <code>"phylo"</code>. <code>mammal.data</code> is a data frame. <code>mammal.geog</code> is a numerical matrix.
</p>
<p><code>primate.tree</code> is an object of class <code>"phylo"</code>. <code>primate.data</code> is a data frame.
</p>
<p><code>salamanders</code> is an object of class <code>"phylo"</code>.
</p>
<p><code>sunfish.tree</code> is an object of class <code>"simmap"</code>. <code>sunfish.data</code> is a data frame.
</p>
<p><code>tortoise.tree</code> is an object of class <code>"phylo"</code>. <code>tortoise.geog</code> is a data frame.
</p>
<p><code>tropidurid.tree</code> is an object of class <code>"simmap"</code>. <code>tropidurid.data</code> is a data frame.
</p>
<p><code>vertebrate.tree</code> is an object of class <code>"phylo"</code>. <code>vertebrate.data</code> is a data frame.
</p>
<p><code>wasp.trees</code> is an object of class <code>"multiPhylo"</code>. <code>wasp.data</code> is a data frame.
</p>
<p><code>whale.tree</code> is an object of class <code>"phylo"</code>.
</p>


<h3>Details</h3>

<p><code>anoletree</code> is a phylogeny of Greater Antillean anole species with a mapped discrete character - <em>ecomorph class</em>. <code>anole.data</code> is a data frame of morphological characters. Data and tree are from Mahler et al. (2010).
</p>
<p><code>ant.tree</code> is a phylogeny containing 84 species of fungus farming attine ants. <code>ant.geog</code> is a vector containing biogegraphic information for the species of the tree. Both the phylogeny and biogeographic information were originally published by Branstetter et al. (2017).
</p>
<p><code>bat.tree</code> is a ML phylogeny of bats from the families Mormoopidae, Molossidae, and Phyllostomidae, based on GenBank accession numbers reported in Caraballo (2022) and estimated using <span class="pkg">phangorn</span>. <code>betaCoV.tree</code> is a ML phylogenetic tree of betacoronaviruses based on GenBank accession numbers reported in Caraballo (2022) and estimated using <span class="pkg">phangorn</span>. <code>bat_virus.data</code> is an association table of host and virus from Caraballo (2022).
</p>
<p><code>butterfly.tree</code> and <code>butterfly.data</code> are a phylogeny and dataset of habitat use in Mycalesina butterflies from Halali et al. (2020). The phylogeny of that study was adapted from one originally published by Brattstrom et al. (2020).
</p>
<p><code>bonyfish.tree</code> and <code>bonyfish.data</code> are a phylogeny and dataset of spawning mode and parental care in bony fishes from Benun Sutton and Wilson (2019). Their phylogenetic tree was adapted from a tree originally published by Betancur-R et al. (2017)
</p>
<p><code>cordylid.tree</code> and <code>cordylid.data</code> are a phylogeny and dataset of morphological traits for three different principal components axes from Broeckhoven et al. (2016).
</p>
<p><code>darter.tree</code> is a phylogeny of 201 species of darter fish (Etheostomatinae) from Near et al. (2011).
</p>
<p><code>eel.tree</code> and <code>eel.data</code> are a phylogeny and dataset of feeding mode and maximum total length from 61 species of elopomorph eels from Collar et al. (2014).
</p>
<p><code>elapidae.tree</code> is a reconstructed phylogeny containing 175 species of snakes of the family Elapidae from Lee et al. (2016).
</p>
<p><code>flatworm.tree</code> and <code>flatworm.data</code> are a phylogeny and dataset of habitat preferences for flatworms from Benitez-Alvarez et al. (2020). <code>flatworm.tree</code> has been made ultrametric using penalized likelihood.
</p>
<p><code>liolaemid.tree</code> and <code>liolaemid.data</code> are a phylogenetic tree and phenotypic trait dataset of lizards from the family Liolaemidae from Esquerre et al. (2019).
</p>
<p><code>mammal.tree</code> and <code>mammal.data</code> are the phylogeny and dataset for mammal body size and home range size from Garland et al. (1992). <code>mammal.geog</code> is a matrix containing geographic coordinates for various of the species in <code>mammal.tree</code> pulled from citizen science observations registered on the <em>iNaturalist</em> web platform.
</p>
<p><code>primate.tree</code> and <code>primate.data</code> are a phylogeny and phenotypic trait dataset from Kirk and Kay (2004).
</p>
<p><code>salamanders</code> is a phylogeny of <em>Plethodon</em> salamanders from Highton and Larson (1979). According to Wikipedia, the genus <em>Plethodon</em> contains 55 species in total.
</p>
<p><code>sunfish.tree</code> and <code>sunfish.data</code> are the phylogeny and dataset for Centrarchidae and buccal morphology (respectively) from Revell and Collar (2009).
</p>
<p><code>tortoise.tree</code> and <code>tortoise.geog</code> are a phylogeny and dataset of geographic coordinates for Galapagos tortoises. The geographic locality information was approximated from Figure 1 of Poulakakis et al. (2020), and the tree was estimated from nucleotide sequence data published with the same study.
</p>
<p><code>tropidurid.tree</code> and <code>tropidurid.data</code> are the phylogeny and bivariate quantitative trait dataset of tropidurid lizards from Revell et al. (2022).
</p>
<p><code>vertebrate.tree</code> is a time-calibrated phylogeny of vertebrates and <code>vertebrate.data</code> is a dataset of phenotypic traits. The phylogeny is from <a href="http://www.timetree.org/">http://www.timetree.org/</a> (Hedges et al. 2006).
</p>
<p><code>wasp.trees</code> and <code>wasp.data</code> are the phylogeny and host-parasite associations from Lopez-Vaamonde et al. (2001).
</p>
<p><code>whale.tree</code> is a phylogeny of cetaceans originally published in Steeman et al. (2009).
</p>


<h3>Source</h3>

<p>Benitez-Alvarez, L., A. Maria Leal-Zanchet, A. Oceguera-Figueroa, R. Lopes Ferreira, D. de Medeiros Bento, J. Braccini, R. Sluys, and M. Riutort (2020) Phylogeny and biogeography of the Cavernicola (Platyhelminthes: Tricladida): Relicts of an epigean group sheltering in caves? <em>Molecular Phylogenetics and Evolution</em>, <b>145</b>, 106709.
</p>
<p>Benun Sutton, F., and A. B. Wilson (2019) Where are all the moms? External fertilization predicts the rise of male parental care in bony fishes. <em>Evolution</em>, <b>73</b>, 2451-2460.
</p>
<p>Betancur-R, R., E. O. Wiley, G. Arratia, A. Acero, N. Bailly, M. Miya, G. Lecointre, and G. Orti (2017) Phylogenetic classification of bony fishes. <em>BMC Ecology and Evolution</em>, <b>17</b>, 162.
</p>
<p>Branstetter, M. G., A. Jesovnik, J. Sosa-Calvo, M. W. Lloyd, B. C. Faircloth, S. G. Brady, and T. R. Schultz (2017) Dry habitats were crucibles of domestication in the evolution of agriculture in ants. <em>Proceedings of the Royal Society B</em>, <b>284</b>, 20170095.
</p>
<p>Brattstrom, O., K. Aduse-Poku, E. van Bergen, V. French, and P. M. Brakefield (2020) A release from developmental bias accelerates morphological diversification in butterfly eyespots. <em>Proceedings of the National Academy of Sciences</em>, <b>177</b>, 27474-27480.
</p>
<p>Broeckhoven, C., G. Diedericks, C. Hui, B. G. Makhubo, P. le Fras N. Mouton (2016) Enemy at the gates: Rapid defensive trait diversification in an adaptive radiation of lizards. <em>Evolution</em>, <b>70</b>, 2647-2656.
</p>
<p>Caraballo, D. A. (2022) Cross-species transmission of bat coronaviruses in the Americas: Contrasting patterns between alphacoronavirus and betacoronavirus. <em>Microbiology Spectrum</em>, <b>10</b>, e01411-22.
</p>
<p>Collar, D. C., P. C. Wainwright, M. E. Alfaro, L. J. Revell, and R. S. Mehta (2014) Biting disrupts integration to spur skull evolution in eels. <em>Nature Communications</em>, <b>5</b>, 5505.
</p>
<p>Esquerre, D., D. Ramirez-Alvarez, C. J. Pavon-Vazquez, J. Troncoso-Palacios, C. F. Garin, J. S. Keogh, and A. D. Leache (2019) Speciation across mountains: Phylogenomics, species delimitation and taxonomy of the <em>Liolaemus leopardinus</em> clade (Squamata, Liolaemidae). <em>Molecular Phylogenetics and Evolution</em>, <b>139</b>, 106524.
</p>
<p>Garland, T., Jr., P. H. Harvey, and A. R. Ives (1992) Procedures for the analysis of comparative data using phylogenetically independent contrasts. <em>Systematic Biology</em>, <b>41</b>, 18-32.
</p>
<p>Kirk, E. C., and R. F. Kay (2004) The evolution of high visual acuity in the Anthropoidea. In: Ross, C. F., Kay R. F. (Eds), <em>Anthropoid Origins. Developments in Primatology: Progress and Prospects</em>, 539-602. Springer, Boston, MA.
</p>
<p>Halali, S., E. van Bergen, C. J. Breuker, P. M. Brakefield, and O. Brattstrom (2020) Seasonal environments drive convergent evolution of a faster pace-of-life in tropical butterflies. <em>Ecology Letters</em>, <b>24</b>, 102-112.
</p>
<p>Hedges, S. B., J. Dudley, and S. Kumar (2006) TimeTree: A public knowledgebase of divergence times among organisms. <em>Bioinformatics</em>, <b>22</b>, 2971-2972.
</p>
<p>Highton, R., and A. Larson (1979) The genetic relationships of the salamanders of the genus <em>Plethodon</em>. <em>Systematic Zoology</em>, <b>28</b>, 579-599.
</p>
<p>Lee, M. S. Y., K. L. Saunders, B. King, and A. Palci (2016) Diversification rates and phenotypic evolution in venomous snakes (Elapidae). <em>Royal Society Open Science</em>, <b>3</b>, 150277.
</p>
<p>Lopez-Vaamonde, C., J. Y. Rasplus, G. D. Weiblen, and J. M. Cook (2001) Molecular phylogenies of fig wasps: Partial cocladogenesis of pollinators and parasites. <em>Molecular Phylogenetics and Evolution</em>, <b>21</b>, 55-71.
</p>
<p>Mahler, D. L, L. J. Revell, R. E. Glor, and J. B. Losos (2010) Ecological opportunity and the rate of morphological evolution in the diversification of Greater Antillean anoles. <em>Evolution</em>, <b>64</b>, 2731-2745.
</p>
<p>Near, T. J., C. M. Bossu, G. S. Bradburd, R. L. Carlson, R. C. Harrington, P. R. Hollingsworth Jr., B. P. Keck, and D. A. Etnier. (2011) Phylogeny and temporal diversification of darters (Percidae: Etheostomatinae). <em>Systematic Biology</em>, <b>60</b>, 565-595.
</p>
<p>Poulakakis, N., J. M. Miller, E. L. Jensen, L. B. Beheregaray, M. A. Russello, S. Glaberman, J. Boore, and A. Caccone. (2020) Colonization history of Galapagos giant tortoises: Insights from mitogenomes support the progression rule. <em>Journal of Zoological Systematics and Evolutionary Research</em>, <b>58</b>, 1262-1275.
</p>
<p>Revell, L. J., and D. C. Collar (2009) Phylogenetic analysis of the evolutionary correlation using likelihood. <em>Evolution</em>, <b>63</b>, 1090-1100.
</p>
<p>Revell, L. J., K. S. Toyama, and D. L. Mahler (2022) A simple hierarchical model for heterogeneity in the evolutionary correlation a phylogenetic tree. <em>PeerJ</em>, <b>10</b>, e13910.
</p>
<p>Steeman, M. E., M. B. Hebsgaard, R. E. Fordyce, S. Y. W. Ho, D. L. Rabosky, R. Nielsen, C. Rahbek, H. Glenner, M. V. Sorensen, and E. Willerslev (2009) Radiation of extant cetaceans driven by restructuring of the oceans. <em>Systematic Biology</em>, <b>58</b>, 573-585.
</p>

<hr>
<h2 id='ansi_phylo'>Compute the parsimony score</h2><span id='topic+ansi_phylo'></span>

<h3>Description</h3>

<p>Plots a phylogeny in a silly, ANSI graphics style.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ansi_phylo(tree, vertical=c("|","-"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ansi_phylo_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="ansi_phylo_+3A_vertical">vertical</code></td>
<td>
<p>character for vertical lines</p>
</td></tr>
<tr><td><code id="ansi_phylo_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plotted tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotTree">plotTree</a></code>, <code><a href="#topic+splinePhylogram">splinePhylogram</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(salamanders)
ansi_phylo(salamanders)
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='applyBranchLengths'>Applies the branch lengths of a reference tree to a target</h2><span id='topic+applyBranchLengths'></span>

<h3>Description</h3>

<p>Applies the branch lengths of a reference tree to a target.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>applyBranchLengths(tree, edge.length)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyBranchLengths_+3A_tree">tree</code></td>
<td>
<p>target tree.</p>
</td></tr>
<tr><td><code id="applyBranchLengths_+3A_edge.length">edge.length</code></td>
<td>
<p>number of digits for rounding. Passed to <code><a href="base.html#topic+round">round</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function applies the set of branch lengths from a reference tree to a target tree while reconciling any mappings (as in <code><a href="#topic+read.simmap">read.simmap</a></code>) with the new branch lengths.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> or <code>"simmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='as.multiPhylo'>Conversion to object of class <code>"multiPhylo"</code></h2><span id='topic+as.multiPhylo'></span><span id='topic+as.multiPhylo.multiSimmap'></span><span id='topic+as.multiPhylo.phylo'></span>

<h3>Description</h3>

<p>This function converts between object classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.multiPhylo(x, ...)
## S3 method for class 'multiSimmap'
as.multiPhylo(x, ...)
## S3 method for class 'phylo'
as.multiPhylo(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.multiPhylo_+3A_x">x</code></td>
<td>
<p>object to be converted to <code>"multiPhylo"</code>. Presently an object of class <code>"multiSimmap"</code>, or an object of class <code>"phylo"</code>. In the latter case an object of class <code>"multiPhylo"</code> with length <code>1</code> is generated.</p>
</td></tr>
<tr><td><code id="as.multiPhylo_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='as.Qmatrix'>Convert a fitted M<em>k</em> model to a Q-matrix</h2><span id='topic+as.Qmatrix'></span><span id='topic+as.Qmatrix.fitMk'></span><span id='topic+plot.Qmatrix'></span><span id='topic+print.Qmatrix'></span>

<h3>Description</h3>

<p>Extracts a <b>Q</b>-matrix from a fitted M<em>k</em> model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.Qmatrix(x, ...)
## S3 method for class 'fitMk'
as.Qmatrix(x, ...)
## S3 method for class 'Qmatrix'
plot(x, ...)
## S3 method for class 'Qmatrix'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.Qmatrix_+3A_x">x</code></td>
<td>
<p>fitted M<em>k</em> model. (For instance, an object of class <code>"fitMk"</code>.) In the case of <code>print.Qmatrix</code>, an object of class <code>"Qmatrix"</code>.</p>
</td></tr>
<tr><td><code id="as.Qmatrix_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extracts a <b>Q</b>-matrix (in the form of an object of class <code>"Qmatrix"</code>) from a fitted M<em>k</em> model.
</p>


<h3>Value</h3>

<p>An object of class <code>"Qmatrix"</code>.
</p>
<p><code>plot.Qmatrix</code> invisibly returns the coordinates of vertices of the plotted <b>Q</b>-matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitMk">fitMk</a></code>
</p>

<hr>
<h2 id='ave.rates'>Average the posterior rates</h2><span id='topic+ave.rates'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ave.rates(tree, shift, tips, sig1, sig2, ave.shift, showTree=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ave.rates_+3A_tree">tree</code></td>
<td>
<p>a tree.</p>
</td></tr>
<tr><td><code id="ave.rates_+3A_shift">shift</code></td>
<td>
<p>the shift point for this sample.</p>
</td></tr>
<tr><td><code id="ave.rates_+3A_tips">tips</code></td>
<td>
<p>tip names tipward of <code>shift</code>.</p>
</td></tr>
<tr><td><code id="ave.rates_+3A_sig1">sig1</code></td>
<td>
<p><code class="reqn">\sigma_1^2</code>.</p>
</td></tr>
<tr><td><code id="ave.rates_+3A_sig2">sig2</code></td>
<td>
<p><code class="reqn">\sigma_2^2</code>.</p>
</td></tr>
<tr><td><code id="ave.rates_+3A_ave.shift">ave.shift</code></td>
<td>
<p>average shift from all samples.</p>
</td></tr>
<tr><td><code id="ave.rates_+3A_showtree">showTree</code></td>
<td>
<p>logical value indicating whether to plot the rate-stretched tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the rates.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>, <code><a href="#topic+minSplit">minSplit</a></code>, <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>
</p>

<hr>
<h2 id='averageTree'>Compute an average tree from a set of trees and related operations</h2><span id='topic+averageTree'></span><span id='topic+ls.consensus'></span><span id='topic+minTreeDist'></span>

<h3>Description</h3>

<p>Compute average trees or consensus trees by various criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>averageTree(trees, start=NULL, method="quadratic.path.difference",
   tol=1e-12, quiet=FALSE, ...)
ls.consensus(trees, start=NULL, tol=1e-12, quiet=FALSE, ...)
minTreeDist(tree, trees, method="quadratic.path.difference", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="averageTree_+3A_trees">trees</code></td>
<td>
<p>object of class <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="averageTree_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>. For <code>minTreeDist</code> the tree on which to find the edge lengths that minimize the distance to the phylogenies in <code>trees</code>.</p>
</td></tr>
<tr><td><code id="averageTree_+3A_start">start</code></td>
<td>
<p>starting tree for optimization.</p>
</td></tr>
<tr><td><code id="averageTree_+3A_method">method</code></td>
<td>
<p>distance criterion for minimization. Options are <code>"symmetric.difference"</code>, <code>"branch.score.difference"</code>, <code>"path.difference"</code>, and <code>"quadratic.path.difference"</code>.</p>
</td></tr>
<tr><td><code id="averageTree_+3A_tol">tol</code></td>
<td>
<p>tolerance value for optimization.</p>
</td></tr>
<tr><td><code id="averageTree_+3A_quiet">quiet</code></td>
<td>
<p>logical value indicating whether to run &quot;quietly&quot; or not.</p>
</td></tr>
<tr><td><code id="averageTree_+3A_...">...</code></td>
<td>
<p>other arguments to be passed internally.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>averageTree</code> tries to find the (hypothetical) tree topology and branch lengths that has a minimum distance to all the trees in an input set, according to some user-specified tree distance measure.
</p>
<p>The function <code>ls.consensus</code> computes the least-squares consensus tree (Lapointe &amp; Cucumel, 1997) from a set of input trees.
</p>
<p>Finally, the function <code>minTreeDist</code> finds the tree in the input set that is a minimum distance to all the other trees in the set. (This contrasts with <code>averageTree</code> which can return a tree not in the input set.)
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> with edge lengths.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Lapointe, F.-J., G. Cucumel (1997) The average consensus procedure: Combination of weighted trees containing identical or overlapping sets of taxa. <em>Systematic Biology</em>, <b>46</b>, 306-312.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='bd'>Convert object of class <code>"birthdeath"</code> to raw birth &amp; death rates</h2><span id='topic+bd'></span>

<h3>Description</h3>

<p>Converts an object of class <code>"birthdeath"</code> (from <code><a href="ape.html#topic+birthdeath">birthdeath</a></code>) to a vector with the ML birth &amp; death rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bd(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bd_+3A_x">x</code></td>
<td>
<p>object of class <code>"birthdeath"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is now somewhat obsolete as <span class="pkg">phytools</span> now contains functions to fit birth-death and pure-birth diversification models from trees (<code><a href="#topic+fit.bd">fit.bd</a></code> and <code><a href="#topic+fit.yule">fit.yule</a></code>) that also take into account incomplete sampling fraction.
</p>


<h3>Value</h3>

<p>A vector.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+birthdeath">birthdeath</a></code>, <code><a href="#topic+fit.bd">fit.bd</a></code>, <code><a href="#topic+fit.yule">fit.yule</a></code>
</p>

<hr>
<h2 id='bind.tip'>Attaches a new tip to a tree</h2><span id='topic+bind.tip'></span>

<h3>Description</h3>

<p>Adds a new tip to the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind.tip(tree, tip.label, edge.length=NULL, where=NULL, position=0,
   interactive=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bind.tip_+3A_tree">tree</code></td>
<td>
<p>receptor tree.</p>
</td></tr>
<tr><td><code id="bind.tip_+3A_tip.label">tip.label</code></td>
<td>
<p>a string containing the species name for the new tip.</p>
</td></tr>
<tr><td><code id="bind.tip_+3A_edge.length">edge.length</code></td>
<td>
<p>edge length for the new tip (a scalar).</p>
</td></tr>
<tr><td><code id="bind.tip_+3A_where">where</code></td>
<td>
<p>node number to attach new tip. If <code>position&gt;0</code> then then tip will be attached <em>rootward</em> of the specified node. Node numbers can also be tips, in which case the new tip will be added along the terminal edge. To find out the tip number for given species with name <em>&quot;species&quot;</em> type: <code>which(tree$tip.label=="species")</code>.</p>
</td></tr>
<tr><td><code id="bind.tip_+3A_position">position</code></td>
<td>
<p>distance <em>rootward</em> of the node to add the new tip.</p>
</td></tr>
<tr><td><code id="bind.tip_+3A_interactive">interactive</code></td>
<td>
<p>logical value indicating whether or not the species should be added interactively. (Defaults to <code>FALSE</code>.)</p>
</td></tr>
<tr><td><code id="bind.tip_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code>plotTree</code> (for <code>interactive=TRUE</code>.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the tree is ultrametric and no branch length is specified, then <code>edge.length</code> is scaled so that the tree remains ultrametric after the new tip is added.
</p>
<p>Wrapper function for <span class="pkg">ape</span> <code><a href="ape.html#topic+bind.tree">bind.tree</a></code>. Note that <code>interactive=TRUE</code> works only for right-facing phylograms.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
par(mfrow=c(3,1))
## generate tree
tree&lt;-pbtree(b=0.1, n=10)
## plot original tree
plotTree(tree,mar=c(2.1,0.1,2.1,0.1))
axisPhylo()
mtext("a)",adj=0)
## add an extant tip ("t_extant") sister to taxon 't5'
## with divergence time of 4.5 Ma
node &lt;- which(tree$tip.label=="t5")
tree &lt;- bind.tip(tree, tip.label="t_extant",
  where=node, position=4.5)
# plot to see the result
plotTree(tree,mar=c(2.1,0.1,2.1,0.1))
axisPhylo()
mtext("b)",adj=0)
## add an extinct tip ("t_extinct") sister to 't2' with
## divergence time of 7.8 Ma and duration (edge length) of
## 3.3 Ma
node &lt;- which(tree$tip.label=="t2")
tree &lt;- bind.tip(tree, tip.label="t_extinct", where=node,
  position=7.8, edge.length=3.3)
## plot to see the result
plotTree(tree,mar=c(2.1,0.1,2.1,0.1))
axisPhylo()
mtext("c)",adj=0)
par(mar=c(5.1,4.1,4.1,2.1),mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='bind.tree.simmap'>Binds two trees of class <code>"simmap"</code></h2><span id='topic+bind.tree.simmap'></span>

<h3>Description</h3>

<p>This function grafts tree <code>y</code> onto tree <code>x</code> at node <code>where</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind.tree.simmap(x, y, where="root")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bind.tree.simmap_+3A_x">x</code></td>
<td>
<p>an object of class <code>"simmap"</code>. (The receptor tree.)</p>
</td></tr>
<tr><td><code id="bind.tree.simmap_+3A_y">y</code></td>
<td>
<p>an object of class <code>"simmap"</code>. (The tree being grafted.)</p>
</td></tr>
<tr><td><code id="bind.tree.simmap_+3A_where">where</code></td>
<td>
<p>node number to attach new tip, or the root node if <code>where="root"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function wraps around <code><a href="ape.html#topic+bind.tree">bind.tree</a></code> for objects of class <code>"simmap"</code>; however it presently only allows <code>y</code> to be grafted at a node of <code>x</code> and it does not allow <code>y</code> to possess a root edge.
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='bmPlot'>Simulates and visualizes discrete-time Brownian evolution on a phylogeny</h2><span id='topic+bmPlot'></span>

<h3>Description</h3>

<p>Conducts a discrete-time Brownian motion simulation on an input tree, plots the outcome, and returns the tip and internal node states to the user as a named vector. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmPlot(tree, type="BM", anc=0, sig2=1/1000, ngen=1000, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bmPlot_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="bmPlot_+3A_type">type</code></td>
<td>
<p>the type of plot to create. See Description.</p>
</td></tr>
<tr><td><code id="bmPlot_+3A_anc">anc</code></td>
<td>
<p>the ancestral value for the root node.</p>
</td></tr>
<tr><td><code id="bmPlot_+3A_sig2">sig2</code></td>
<td>
<p>the BM rate (variance of the Brownian evolution process), <code class="reqn">\sigma^2</code>.</p>
</td></tr>
<tr><td><code id="bmPlot_+3A_ngen">ngen</code></td>
<td>
<p>number of generations for the simulation: will rescale the tree to this total length.</p>
</td></tr>
<tr><td><code id="bmPlot_+3A_...">...</code></td>
<td>
<p>arguments to be passed to different methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will first rescale and round the branch lengths to integer length, if they are not already in integer values. If integer branch lengths are provided, the user should also set <code>ngen</code> to be equal to the total height of the tree in generations (and thus avoid rescaling). 
</p>
<p>For <code>type="threshold"</code> the visualization is of the threshold model (Felsenstein 2012), in which the evolving character is liability and the segments of evolution are colored by their value for the threshold trait. If <code>type="threshold"</code> is used, the function requires at least one addition input: <code>thresholds</code>, a vector containing the ordered thresholds between states. The user can also provide the colors for plotting in <code>colors</code>. Note that one more color than threshold should be provided as one threshold implies two states; two thresholds, three states; etc. If no value for <code>colors</code> is provided, the function will recycle a set of four colors up to the number of times required by <code>thresholds</code>. Finally, the optional argument <code>return.tree=TRUE</code> will tell the function to return a list with the tip and note states and an object of class <code>"phylo"</code> with (for <code>type="threshold"</code>), the state for the threshold model through time mapped on the branches of the tree in discrete time.
</p>


<h3>Value</h3>

<p>This function conducts and plots discrete time Brownian simulation and returns a vector containing the simulated states at internal nodes and tips of the tree. 
</p>
<p>It also returns, by default (although this can be turned off) a tree with the branch lengths in discrete time and with a mapped discrete character (for <code>type="threshold"</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (2012) A comparative method for both discrete and continuous characters using the threshold model. <em>American Naturalist</em>, <b>179</b>, 145-156.
</p>
<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastBM">fastBM</a></code>, <code><a href="#topic+pbtree">pbtree</a></code>, <code><a href="#topic+phenogram">phenogram</a></code>, <code><a href="#topic+threshBayes">threshBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(999)
## plot BM simulation on 12 taxon tree
tree&lt;-pbtree(n=12)
par(mfrow=c(1,2),mar=c(5.1,4.1,4.1,0.1))
x&lt;-bmPlot(tree,bty="l")
plotTree(tree,direction="upwards",
    mar=c(5.1,0.1,4.1,1.1),ftype="off")
## reset par to default values
par(mfrow=c(1,1),mar=c(5.1,4.1,4.1,2.1))
## plot simulation of a threshold character
par(mfrow=c(1,2),mar=c(5.1,4.1,4.1,0.1))
tt&lt;-bmPlot(tree,type="threshold",thresholds=c(0,1,2),
    bty="l")
plot(tt$tree,direction="upwards",
    mar=c(5.1,0.1,4.1,1.1),ftype="off",
    colors=setNames(c("black","red","blue"),
    letters[1:3]),lwd=3)
## reset par to default values
par(mfrow=c(1,1),mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='bounded_bm'>Fits bounded and wrapped Brownian motion models</h2><span id='topic+bounded_bm'></span>

<h3>Description</h3>

<p>Fits a bounded Brownian motion model using the discrete approximation of Boucher &amp; Demery (2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	bounded_bm(tree, x, lims=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bounded_bm_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="bounded_bm_+3A_x">x</code></td>
<td>
<p>continuous character data in a vector.</p>
</td></tr>
<tr><td><code id="bounded_bm_+3A_lims">lims</code></td>
<td>
<p>bounds. If <code>NULL</code> will be set to three times the observed range of the trait. (This is designed to be an effectively &quot;unbounded&quot; null model.)</p>
</td></tr>
<tr><td><code id="bounded_bm_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Optional arguments.
</p>


<h3>Value</h3>

<p>An object of class <code>"bounded_bm"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Boucher, F. C., and V. Demery (2016) Inferring bounded evolution in phenotypic characters from phylogenetic comparative data. <em>Systematic Biology</em>, <b>65</b>, 651-661.
</p>
<p>Juhn, M., K. Martinet, F. Boucher, A. Schultz, L. Harmon, K. Burns, and M. Alfaro (In review) A wrapped Brownian motion model for the evolution of color.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitMk">fitMk</a></code>, <code><a href="#topic+fitmultiBM">fitmultiBM</a></code>, <code><a href="#topic+fitThresh">fitThresh</a></code>
</p>

<hr>
<h2 id='branching.diffusion'>Animation of branching random diffusion</h2><span id='topic+branching.diffusion'></span>

<h3>Description</h3>

<p>This function creates an animation of branching random diffusion (i.e., Brownian motion evolution with speciation).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>branching.diffusion(sig2=1, b=0.0023, time.stop=1000, ylim=NULL, 
   smooth=TRUE, pause=0.02, record=NULL, path=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="branching.diffusion_+3A_sig2">sig2</code></td>
<td>
<p>variance of BM process, <code class="reqn">\sigma^2</code>.</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_b">b</code></td>
<td>
<p>birthrate for branching process.</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_time.stop">time.stop</code></td>
<td>
<p>number of generations to run.</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_ylim">ylim</code></td>
<td>
<p><em>y</em> limits (for plotting).</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_smooth">smooth</code></td>
<td>
<p>no longer used.</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_pause">pause</code></td>
<td>
<p>pause (in s) between generations.</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_record">record</code></td>
<td>
<p>filename for video file output (no video if <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_path">path</code></td>
<td>
<p>full path to file for video rendering. (By default <code>branching.diffusion</code> will look for the executable <code>ffmpeg.exe</code> in the directory <code>C:/Program Files/ffmpeg/bin</code>, even though this will not make sense on non-Windows machines.)</p>
</td></tr>
<tr><td><code id="branching.diffusion_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For animation to be recorded to file, the function requires the package <em>animation</em> as well as a video renderer.
</p>


<h3>Value</h3>

<p>An animated plot and (optionally) a recorded video file.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmPlot">bmPlot</a></code>, <code><a href="#topic+fastBM">fastBM</a></code>
</p>

<hr>
<h2 id='brownie.lite'>Likelihood test for rate variation in a continuous trait</h2><span id='topic+brownie.lite'></span>

<h3>Description</h3>

<p>Fits a multi-rate Brownian motion evolution model using maximum likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brownie.lite(tree, x, maxit=2000, test="chisq", nsim=100, se=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brownie.lite_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree either as an object of class <code>"phylo"</code> or <code>"simmap"</code>. (See <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, or <code><a href="#topic+paintSubTree">paintSubTree</a></code> for more details about the latter object class.)</p>
</td></tr>
<tr><td><code id="brownie.lite_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species. <code>names(x)</code> should be the species names.</p>
</td></tr>
<tr><td><code id="brownie.lite_+3A_maxit">maxit</code></td>
<td>
<p>an optional integer value indicating the maximum number of iterations for optimization - may need to be increased for large trees.</p>
</td></tr>
<tr><td><code id="brownie.lite_+3A_test">test</code></td>
<td>
<p>an optional string indicating the method for hypothesis testing - options are <code>"chisq"</code> or <code>"simulation"</code>.</p>
</td></tr>
<tr><td><code id="brownie.lite_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations (only used if <code>test="simulation"</code>).</p>
</td></tr>
<tr><td><code id="brownie.lite_+3A_se">se</code></td>
<td>
<p>a vector containing the standard errors for each estimated mean in <code>x</code>.</p>
</td></tr>
<tr><td><code id="brownie.lite_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes an object of class <code>"phylo"</code> or class <code>"simmap"</code> with a mapped binary or multi-state trait (see <code><a href="#topic+read.simmap">read.simmap</a></code>) and data for a single continuously valued character.  It then fits the Brownian rate variation (&quot;noncensored&quot;) model of O'Meara et al. (2006; <em>Evolution</em>). This is also the basic model implemented in Brian O'Meara's <em>Brownie</em> software.
</p>
<p>Sampling error in the estimation of species means can also be accounted for by assigning the vector <code>se</code> with the species specific sampling errors for <code>x</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"brownie.lite"</code> containing the following elements:
</p>
<table>
<tr><td><code>sig2.single</code></td>
<td>
<p>is the rate, <code class="reqn">\sigma^2</code>, for a single-rate model. This is usually the &quot;null&quot; model.</p>
</td></tr>
<tr><td><code>a.single</code></td>
<td>
<p>is the estimated state at the root node for the single rate model.</p>
</td></tr>
<tr><td><code>var.single</code></td>
<td>
<p>variance on the single rate estimator - obtained from the Hessian.</p>
</td></tr>
<tr><td><code>logL1</code></td>
<td>
<p>log-likelihood of the single-rate model.</p>
</td></tr>
<tr><td><code>k1</code></td>
<td>
<p>number of parameters in the single rate model (always 2).</p>
</td></tr>
<tr><td><code>sig2.multiple</code></td>
<td>
<p>is a length <em>p</em> (for <em>p</em> rates) vector of BM rates (<code class="reqn">\sigma_1^2</code>, <code class="reqn">\sigma_2^2</code>, and so on) from the multi-rate model.</p>
</td></tr>
<tr><td><code>a.multiple</code></td>
<td>
<p>is the estimated state at the root node for the multi-rate model.</p>
</td></tr>
<tr><td><code>var.multiple</code></td>
<td>
<p><em>p</em> x <em>p</em> variance-covariance matrix for the <em>p</em> rates - the square-roots of the diagonals should give the standard error for each rate.</p>
</td></tr>
<tr><td><code>logL.multiple</code></td>
<td>
<p>log-likelihood of the multi-rate model.</p>
</td></tr>
<tr><td><code>k2</code></td>
<td>
<p>number of parameters in the multi-rate model (<em>p</em>+1).</p>
</td></tr>
<tr><td><code>P.chisq</code></td>
<td>
<p>P-value for a likelihood ratio test against the <code class="reqn">\chi^2</code> distribution; or</p>
</td></tr>
<tr><td><code>P.sim</code></td>
<td>
<p>P-value for a likelihood ratio test against a simulated null distribution.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>logical value indicating if the likelihood optimization converged.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>O'Meara, B. C., C. Ane, M. J. Sanderson, and P. C. Wainwright (2006) Testing for different rates of continuous trait evolution using likelihood. <em>Evolution</em>, <b>60</b>, 922-933.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownieREML">brownieREML</a></code>, <code><a href="#topic+evol.vcv">evol.vcv</a></code>, <code><a href="#topic+ratebytree">ratebytree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract character of interest
buccal.length&lt;-setNames(sunfish.data$buccal.length,
    rownames(sunfish.data))
## fit model
multiBM.fit&lt;-brownie.lite(sunfish.tree,
    buccal.length)
print(multiBM.fit)
</code></pre>

<hr>
<h2 id='brownieREML'>REML version of brownie.lite</h2><span id='topic+brownieREML'></span>

<h3>Description</h3>

<p>Fits a multi-rate Brownian motion evolution model using REML. (See <code><a href="#topic+brownie.lite">brownie.lite</a></code> for more details.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brownieREML(tree, x, maxit=2000, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brownieREML_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code> or <code>"simmap"</code>. (See <code><a href="#topic+read.simmap">read.simmap</a></code> and <code><a href="#topic+make.simmap">make.simmap</a></code> for more information about the latter object class.)</p>
</td></tr>
<tr><td><code id="brownieREML_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species. <code>names(x)</code> should be the species names.</p>
</td></tr>
<tr><td><code id="brownieREML_+3A_maxit">maxit</code></td>
<td>
<p>an optional integer value indicating the maximum number of iterations for optimization - may need to be increased for large trees.</p>
</td></tr>
<tr><td><code id="brownieREML_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes an object of class <code>"phylo"</code> or an object of class <code>"simmap"</code> with a mapped binary or multi-state trait (see <code><a href="#topic+read.simmap">read.simmap</a></code>) and data for a single continuously valued character.  It then uses restricted maximum likelihood (REML) to fit the Brownian rate variation (&quot;noncensored&quot;) model of O'Meara et al. (2006; <em>Evolution</em>).  This function is similar to <code><a href="#topic+brownie.lite">brownie.lite</a></code> but uses REML (which is faster and unbiased) instead of ML.  REML optimization takes advantage of Felsenstein's (1985) contrasts algorithm.
</p>


<h3>Value</h3>

<p>An object of class <code>"brownieREML"</code> containing the following components:
</p>
<table>
<tr><td><code>sig2.single</code></td>
<td>
<p>is the rate, <code class="reqn">\sigma^2</code>, for a single rate model - this is usually the &quot;null&quot; model.</p>
</td></tr>
<tr><td><code>logL1</code></td>
<td>
<p>log-likelihood of the single-rate model.</p>
</td></tr>
<tr><td><code>sig2.multiple</code></td>
<td>
<p>is a length <em>p</em> (for <em>p</em> rates) vector of BM rates (<code class="reqn">\sigma_1^2</code>, <code class="reqn">\sigma_2^2</code>, and so on) from the multi-rate model.</p>
</td></tr>
<tr><td><code>logL2</code></td>
<td>
<p>log-likelihood of the multi-rate model.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>numerical value from <code><a href="stats.html#topic+optim">optim</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (1985) Phylogenies and the comparative method. <em>American Naturalist</em>, <b>125</b>, 1-15. 
</p>
<p>O'Meara, B. C., C. Ane, M. J. Sanderson, and P. C. Wainwright (2006) Testing for different rates of continuous trait evolution using likelihood. <em>Evolution</em>, <b>60</b>, 922-933.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+evol.vcv">evol.vcv</a></code>, <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>, <code><a href="#topic+ratebytree">ratebytree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract character of interest
gape.width&lt;-setNames(sunfish.data$gape.width,
    rownames(sunfish.data))
## fit model
multiBM.reml&lt;-brownieREML(sunfish.tree,
	gape.width)
print(multiBM.reml)
</code></pre>

<hr>
<h2 id='cladelabels'>Add labels to subtrees of a plotted phylogeny</h2><span id='topic+cladelabels'></span><span id='topic+arc.cladelabels'></span>

<h3>Description</h3>

<p>This function adds clade labels to a plotted tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cladelabels(tree=NULL, text, node, offset=NULL, wing.length=NULL, cex=1,
   orientation="vertical")
arc.cladelabels(tree=NULL, text, node=NULL, ln.offset=1.02,
   lab.offset=1.06, cex=1, orientation="curved", stretch=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cladelabels_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>. If not supplied, the function will obtain the last plotted phylogeny from the environmental variable <code>last_plot.phylo</code>.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_text">text</code></td>
<td>
<p>desired clade label text.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_node">node</code></td>
<td>
<p>node number for the most recent common ancestor of members of the clade. For <code>arc.cladelabels</code> this defaults to <code>NULL</code> which means that the node of the clade to be labeled should be specified interactively (that is, by clicking on the graphical device).</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_offset">offset</code></td>
<td>
<p>offset (as a multiplier of character width) for the label. Defaults to <code>offset=1</code> if <code>tree</code> is supplied or <code>offset=8</code> otherwise.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_wing.length">wing.length</code></td>
<td>
<p>length of the wings to add to the top &amp; bottom of the label bar (in character widths).</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_cex">cex</code></td>
<td>
<p>character expansion factor.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_orientation">orientation</code></td>
<td>
<p>orientation of the text. Can be <code>orientation = "vertical"</code> (the default) or <code>"horizontal"</code>.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_ln.offset">ln.offset</code></td>
<td>
<p>line offset (as a function of total tree height) for <code>arc.cladelabels</code>.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_lab.offset">lab.offset</code></td>
<td>
<p>label offset for <code>arc.cladelabels</code>.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_stretch">stretch</code></td>
<td>
<p>argument for <code>arc.cladelabels</code> to be passed to <code>arctext</code>.</p>
</td></tr>
<tr><td><code id="cladelabels_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>arc.cladelabels</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cladelabels</code> presently works only for rightward facing plotted phylogenies - but no warning will be returned if your tree does not conform to this requirement!
</p>
<p><code>arc.cladelabels</code> is designed to do a similar thing to <code>cladelabels</code>, but for plotted fan trees. This function checks to ensure that the most recently plotted tree was plotted with <code>type="fan"</code> style.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+nodelabels">nodelabels</a></code>
</p>

<hr>
<h2 id='collapse.to.star'>Collapse a subtree to a star phylogeny</h2><span id='topic+collapse.to.star'></span>

<h3>Description</h3>

<p>Collapses a subtree to a star.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collapse.to.star(tree, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="collapse.to.star_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="collapse.to.star_+3A_node">node</code></td>
<td>
<p>node for the clade to be collapsed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the tree has edge lengths, the function will keep the tips at the same height above the root as in the original tree.
</p>
<p>If <code>node</code> is the global root of the tree a star phylogeny will be created.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+splitTree">splitTree</a></code>, <code><a href="#topic+starTree">starTree</a></code>
</p>

<hr>
<h2 id='collapseTree'>Interactive tree visualizer</h2><span id='topic+collapseTree'></span>

<h3>Description</h3>

<p>Function creates an interactive visualization of collapsing &amp; expanding clades on the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collapseTree(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="collapseTree_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="collapseTree_+3A_...">...</code></td>
<td>
<p>optional arguments. These <em>mostly</em> match the arguments of <code><a href="#topic+plotSimmap">plotSimmap</a></code>, but also include the argument <code>drop.extinct=TRUE</code> which will (if the input tree is ultrametric) drop any 'extinct' lineages from the tree that is returned by the function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function first plots a fan style tree, and then the user collapses node on the tree by clicking on them. Collapsed nodes are collapsed to the common ancestor of the clade. Nodes that have been collapsed can also be expanded by clicking. Right-click (or select Finish button in <span class="pkg">RStudio</span>) to end.
</p>
<p>Note, for the animation to work probably when run in the <span class="pkg">RStudio</span> IDE the &quot;zoom&quot; level should be set to 100%.
</p>


<h3>Value</h3>

<p>Returns the final plotted tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotTree">plotTree</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(liolaemid.tree)
pruned&lt;-collapseTree(liolaemid.tree,fsize=0.5)
## End(Not run)
</code></pre>

<hr>
<h2 id='compare.chronograms'>Compares two chronograms with precisely matching nodes in a visual manner</h2><span id='topic+compare.chronograms'></span>

<h3>Description</h3>

<p>Creates a visual comparison of two chronograms that differ in edge lengths but not topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare.chronograms(t1, t2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare.chronograms_+3A_t1">t1</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="compare.chronograms_+3A_t2">t2</code></td>
<td>
<p>object of class <code>"phylo"</code> that matches <code>t1</code> precisely in topology &amp; node rotations, but differs in edge lengths.</p>
</td></tr>
<tr><td><code id="compare.chronograms_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots two trees, with semi-transparent colors by default, &amp; uses arrows to highlight differences in depth of corresponding nodes between the trees.
</p>


<h3>Value</h3>

<p>Function creates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='consensus.edges'>Compute consensus edges for a tree under some criterion</h2><span id='topic+consensus.edges'></span>

<h3>Description</h3>

<p>Computes consensus edge lengths from a set of input trees using multiple methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consensus.edges(trees, method=c("mean.edge","least.squares"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="consensus.edges_+3A_trees">trees</code></td>
<td>
<p>object of class <code>"multiPhylo"</code> in which the trees must have edge lengths. This could be, for instance, a sample from the posterior distribution of trees in a Bayesian analysis.</p>
</td></tr>
<tr><td><code id="consensus.edges_+3A_method">method</code></td>
<td>
<p>method for computing the edge lengths. Could be the mean of all trees in which the edge is present, or it could be the least-squares edge lengths computed on the mean patristic distance matrices from the input phylogenies in <code>trees</code>. Note that in the latter case the <span class="pkg">phangorn</span> function <code>nnls.tree</code> is used and the option <code>rooted</code> will be set to <code>is.rooted(tree)</code> for the consensus tree.</p>
</td></tr>
<tr><td><code id="consensus.edges_+3A_...">...</code></td>
<td>
<p>optional arguments, the most popular of which is <code>consensus.tree</code> - a user supplied consensus tree. Another optional argument for <code>method="mean.edge"</code> is <code>if.absent</code> which tells the function how to include absent edges in the computation of average edge lengths. Possible values are <code>"zero"</code> (the default) or <code>"ignore"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> with edge lengths.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='contMap'>Map continuous trait evolution on the tree</h2><span id='topic+contMap'></span><span id='topic+plot.contMap'></span><span id='topic+errorbar.contMap'></span>

<h3>Description</h3>

<p>Projects the observed and reconstructed values of a continuous trait onto the edges of a tree using a color gradient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contMap(tree, x, res=100, fsize=NULL, ftype=NULL, lwd=4, legend=NULL,
   lims=NULL, outline=TRUE, sig=3, type="phylogram", direction="rightwards", 
   plot=TRUE, ...)
## S3 method for class 'contMap'
plot(x, ...)
errorbar.contMap(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contMap_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="contMap_+3A_x">x</code></td>
<td>
<p>a numerical vector of phenotypic trait values for species. <code>names(x)</code> should contain the species names and match <code>tree$tip.label</code>. Or, for <code>plot.contMap</code>, an object of class <code>"contMap"</code>.</p>
</td></tr>
<tr><td><code id="contMap_+3A_res">res</code></td>
<td>
<p>resolution for gradient plotting. Larger numbers (to a point) indicate a finer (smoother) gradient.</p>
</td></tr>
<tr><td><code id="contMap_+3A_fsize">fsize</code></td>
<td>
<p>relative font size - can be a vector of length 2 in which the first element gives the font size for the tip labels &amp; the second element giving the font size for the legend.</p>
</td></tr>
<tr><td><code id="contMap_+3A_ftype">ftype</code></td>
<td>
<p>font type - see options in <code><a href="#topic+plotSimmap">plotSimmap</a></code>. As with <code>fsize</code>, this can be a vector with the second element giving font type for the legend.</p>
</td></tr>
<tr><td><code id="contMap_+3A_lwd">lwd</code></td>
<td>
<p>line width for branches. Can be a single integer number or a vector. In the latter case, the second number will be taken to be the desired legend width.</p>
</td></tr>
<tr><td><code id="contMap_+3A_legend">legend</code></td>
<td>
<p>if <code>FALSE</code> no legend is plotted; if a numeric value, it gives the length of the legend in units of branch length. Default is 0.5 times the total tree length.</p>
</td></tr>
<tr><td><code id="contMap_+3A_lims">lims</code></td>
<td>
<p>range for the color map. By default, this will be <code>c(min(x),max(x))</code>, and should always include this range.</p>
</td></tr>
<tr><td><code id="contMap_+3A_outline">outline</code></td>
<td>
<p>logical value indicating whether or not to outline the branches of the tree in black.</p>
</td></tr>
<tr><td><code id="contMap_+3A_sig">sig</code></td>
<td>
<p>the number of decimal places to show on the legend limits.</p>
</td></tr>
<tr><td><code id="contMap_+3A_type">type</code></td>
<td>
<p>type of plot desired. Options are <code>"phylogram"</code> for a rightward square phylogram; and <code>"fan"</code> for a circular phylogram.</p>
</td></tr>
<tr><td><code id="contMap_+3A_direction">direction</code></td>
<td>
<p>plotting direction for <code>type="phylogram"</code>.</p>
</td></tr>
<tr><td><code id="contMap_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether or not to plot the tree. If <code>plot=FALSE</code> then an object of class <code>"contMap"</code> will be returned without plotting.</p>
</td></tr>
<tr><td><code id="contMap_+3A_obj">obj</code></td>
<td>
<p>object of class <code>"contMap"</code>.</p>
</td></tr>
<tr><td><code id="contMap_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>plot.contMap</code> which include all the arguments of <code>contMap</code> except for <code>tree</code>, <code>x</code>, <code>res</code>, and <code>lims</code>. Also <code>method</code>, <code>"fastAnc"</code>, <code>"anc.ML"</code>, or <code>"user"</code> (for user-supplied states) specifying which function to use for ancestral state estimation; <code>hold</code> specifies whether or not to hold output to graphical device before plotting (defaults to <code>hold=TRUE</code>); and <code>anc.states</code> a vector containing some or multiple ancestral user-supplied ancestral states at nodes. Some other plotting arguments, such as <code>xlim</code> and <code>ylim</code>, may also work. Optional arguments for <code>errorbar.contMap</code> include <code>x</code>, a vector containing the original trait values mapped onto the tree (otherwise these will be obtained from <code>obj</code>), <code>scale.by.ci</code>, a logical argument (defaulting to <code>TRUE</code>) that determines whether or not the length of the error bars will be scaled by the CI width, and <code>lwd</code>, which determines the line width of the plotted error bars.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function plots a tree with a mapped continuous character.
</p>
<p>The mapping is accomplished by estimating states at internal nodes using ML with <code><a href="#topic+fastAnc">fastAnc</a></code>, and then interpolating the states along each edge using equation [2] of Felsenstein (1985). This makes these interpolated values equal to the maximum likelihood estimates under a Brownian evolutionary process.
</p>
<p>The default color palette is <em>not</em> recommended as it is not colorblind friendly and does not render well into gray scale; however, this can be updated using the helper function <code><a href="#topic+setMap">setMap</a></code>.
</p>
<p><code>errorbar.contMap</code> adds error bars to an existing plot.
</p>


<h3>Value</h3>

<p>Plots a tree. An object of class <code>"contMap"</code> is returned invisibly.
</p>
<p><code>errorbar.contMap</code> adds colorful error bars to a plotted tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (1985) Phylogenies and the comparative method. <em>American Naturalist</em>, <b>125</b>, 1-15. 
</p>
<p>Revell, L. J. (2013) Two new graphical methods for mapping trait evolution on phylogenies. <em>Methods in Ecology and Evolution</em>, <b>4</b>, 754-759.
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., K. Schliep, E. Valderrama, and J. E. Richardson (2018) Graphs in phylogenetic comparative analysis: Anscombe's quartet revisited. <em>Methods in Ecology and Evolution</em>, <b>9</b>, 2145-2154.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+anc.ML">anc.ML</a></code>, <code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="#topic+fastAnc">fastAnc</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
    rownames(mammal.data)))
## create "contMap" object
mammal.contMap&lt;-contMap(mammal.tree,
    ln.bodyMass,plot=FALSE,res=200)
## change color scheme
mammal.contMap&lt;-setMap(mammal.contMap,
    c("white","#FFFFB2","#FECC5C","#FD8D3C",
    "#E31A1C"))
plot(mammal.contMap,fsize=c(0.7,0.8),
    leg.txt="log(body mass)")
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='cophylo'>Creates a co-phylogenetic plot</h2><span id='topic+cophylo'></span><span id='topic+plot.cophylo'></span><span id='topic+cotangleplot'></span>

<h3>Description</h3>

<p>Visualize co-phylogenetic trees by multiple methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophylo(tr1, tr2, assoc=NULL, rotate=TRUE, ...)
## S3 method for class 'cophylo'
plot(x, ...)
cotangleplot(tr1, tr2, type=c("cladogram","phylogram"),
   use.edge.length=TRUE, tangle=c("both","tree1","tree2"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophylo_+3A_tr1">tr1</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_tr2">tr2</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_assoc">assoc</code></td>
<td>
<p>matrix containing the tip labels in <code>tr1</code> to match to the tip labels in <code>tr2</code>. Note that not all labels in either tree need to be included; and, furthermore, one label in <code>tr1</code> can be matched with more than one label in <code>tr2</code>, or vice versa.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_rotate">rotate</code></td>
<td>
<p>logical argument indicating whether nodes on both trees should be rotated to attempt to match in vertical position.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_x">x</code></td>
<td>
<p>in the case of <code>plot.cophylo</code>, an object of class <code>"cophylo"</code> to be plotted.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_type">type</code></td>
<td>
<p>for <code>cotangleplot</code>, the tree plotting style.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_use.edge.length">use.edge.length</code></td>
<td>
<p>for <code>cotangleplot</code>, a logical value indicating whether or not to plot trees with edge lengths.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_tangle">tangle</code></td>
<td>
<p>for <code>cotangleplot</code>, whether to tangle the left tree, the right tree, or both.</p>
</td></tr>
<tr><td><code id="cophylo_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="#topic+tipRotate">tipRotate</a></code>, or, in the case of <code>plot.cophylo</code>, to the internally used tree plotting function, <code>phylogram</code>. <code>phylogram</code> takes similar arguments to <code><a href="#topic+plotSimmap">plotSimmap</a></code>, such as <code>fsize</code>, <code>ftype</code>, <code>lwd</code>, and <code>pts</code>, though not all options from <code>plotSimmap</code> and <code>plotTree</code> are available. If <code>fsize</code> is supplied as a vector, different size fonts for the left &amp; right facing trees may be used. In addition, the optional argument <code>scale.bar</code>, which should be a vector containing the lengths of the scale bars desired for the right &amp; left trees, will add scale bars to the plot when supplied to <code>plot.cophylo</code>. If either tree contains polytomies, the <code>cophylo</code> argument <code>rotate.multi</code> should be set to <code>TRUE</code>. If curved linking lines are desired, the <code>plot.cophylo</code> argument <code>link.type</code> should be set to <code>"curved"</code>. Other arguments for the <code>plot</code> method include <code>link.col</code>, <code>link.lty</code>, and <code>link.lwd</code>, which can be supplied as a scalar or a vector in which the order of the elements corresponds to the order of the associations in <code>assoc</code>. Finally, <code>edge.col</code>, a list consisting of two vectors (<code>left</code> and <code>right</code>) can be used to specify the edge colors of the two left &amp; right plotted trees. Note that the edge order is the same as in the <em>rotated</em> trees, assuming that a rotation has been performed on <code>x</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cophylo</code> an object of class <code>"cophylo"</code> or, in the case of <code>plot.cophylo</code>, plots that object. The function can (optionally) first attempt to rotate the nodes of both trees to optimize vertical matching of tips.
</p>
<p>If no matrix of associations, <code>assoc</code>, is provided, then <code>cophylo</code> will look for exact matches of tip labels between trees.
</p>
<p><code>cotangleplot</code> creates a co-phylogenetic plot in which the edges of the matched trees are crossing and is designed to be used <em>only</em> on phylogenies with matching tip labels.
</p>


<h3>Value</h3>

<p>An object of class <code>"cophylo"</code> which includes the following components or a pair of plotted facing phylogenies with links between tips as specified in <code>assoc</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+cophyloplot">cophyloplot</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Lopez-Vaamonde et al. (2001)
data(wasp.trees)
data(wasp.data)
## create co-phylogenetic object
wasp.cophylo&lt;-cophylo(wasp.trees[[1]],wasp.trees[[2]],
    assoc=wasp.data)
## plot co-phylogenies
plot(wasp.cophylo,link.type="curved",link.lwd=4,
     link.lty="solid",link.col=make.transparent("red",
     0.25))
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='cospeciation'>Conducts a statistical test of cospeciation between two trees</h2><span id='topic+cospeciation'></span><span id='topic+plot.cospeciation'></span><span id='topic+print.cospeciation'></span>

<h3>Description</h3>

<p>Test for cospeciation based on tree distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cospeciation(t1, t2, distance=c("RF","SPR"),
   method=c("simulation","permutation"), assoc=NULL,
   nsim=100, ...)
## S3 method for class 'cospeciation'
plot(x, ...)
## S3 method for class 'cospeciation'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cospeciation_+3A_t1">t1</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_t2">t2</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_distance">distance</code></td>
<td>
<p>distance method to compare trees.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_method">method</code></td>
<td>
<p>method to use (simulation of pure-birth trees, or permutation of tip labels on a fixed tree) to obtain a null distribution of tree distances via <code>distance</code>.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_assoc">assoc</code></td>
<td>
<p>matrix containing the tip labels in <code>t1</code> to match to the tip labels in <code>t2</code>. Note that not all labels in either tree need to be included; however, unlike <code>cophylo</code>, one label in <code>t1</code> cannot be matched with more than one label in <code>t2</code>, nor vice versa. If <code>NULL</code> then an exact match of tip labels will be sought.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations or permutations.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_x">x</code></td>
<td>
<p>for <code>plot</code> and <code>print</code> methods, an object of class <code>"cospeciation"</code>.</p>
</td></tr>
<tr><td><code id="cospeciation_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function conducts a test for cospeciation based on tree distance, applying a distance metric selected by the user.
</p>
<p>Note that this method should be prone to be quite liberal as the null hypothesis is <em>no similarity whatsoever</em> between the two trees!
</p>


<h3>Value</h3>

<p>An object of class <code>"cospeciation"</code>, which includes the test-statistic, the null distribution, and a p-value for the test of the null hypothesis of no topological similarity between the two trees.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophylo">cophylo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Lopez-Vaamonde et al. (2001)
data(wasp.trees)
data(wasp.data)
## test for cospeciation
wasp.cosp&lt;-cospeciation(wasp.trees[[1]],wasp.trees[[2]],
    assoc=wasp.data)
print(wasp.cosp)
plot(wasp.cosp)
title(main=paste("Simulated distribution of RF distances\n",
    "between unassociated trees"),font.main=3)
</code></pre>

<hr>
<h2 id='countSimmap'>Counts the number of character changes on a object of class <code>"simmap"</code> or <code>"multiSimmap"</code></h2><span id='topic+countSimmap'></span>

<h3>Description</h3>

<p>Count the number of changes of different types on a stochastically mapped trees or set of trees (objects of class <code>"simmap"</code> or <code>"multiSimmap"</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countSimmap(tree, states=NULL, message=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countSimmap_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code> or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="countSimmap_+3A_states">states</code></td>
<td>
<p>optional argument with the states for the mapped character. If not provided, these will be computed from the tree. This is useful if averaging across many trees, some of which may lack certain states.</p>
</td></tr>
<tr><td><code id="countSimmap_+3A_message">message</code></td>
<td>
<p>optional logical argument indicating whether or not to return an informative message about the function output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a tree or a set of trees with a mapped discrete character (that is, an object of class <code>"simmap"</code> or <code>"multiSimmap"</code>), and computes the total number of character changes as well as the number of character changes between all states.
</p>


<h3>Value</h3>

<p>A list with up to three elements: <code>N</code> is an integer value giving the total number of character changes on the tree; <code>Tr</code> gives the number of of transitions between row and column states (or a matrix containing both <code>N</code> and the transitions between states, in rows, for an object of class <code>"multiPhylo"</code>); and (optionally) <code>message</code> contains an explanatory message about the function  output.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Revell &amp; Collar (2009)
data(anoletree)
anoletree
countSimmap(anoletree)
</code></pre>

<hr>
<h2 id='ctt'>Generates (or simulates) a 'changes through time' plot from a set of stochastic map character histories</h2><span id='topic+ctt'></span><span id='topic+sim.ctt'></span><span id='topic+sim.multiCtt'></span>

<h3>Description</h3>

<p>Create a 'changes through time' plot from a <code>"multiSimmap"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctt(trees, segments=20, ...)
sim.ctt(tree, Q, anc=NULL, nmaps=100, ...)
sim.multiCtt(tree, Q, anc=NULL, nmaps=100, nsim=100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctt_+3A_trees">trees</code></td>
<td>
<p>an object of class <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="ctt_+3A_segments">segments</code></td>
<td>
<p>number of segments to break up the history of the tree.</p>
</td></tr>
<tr><td><code id="ctt_+3A_tree">tree</code></td>
<td>
<p>for <code>sim.ctt</code>, an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="ctt_+3A_q">Q</code></td>
<td>
<p>for <code>sim.ctt</code>, a transition matrix to use for simulation.</p>
</td></tr>
<tr><td><code id="ctt_+3A_anc">anc</code></td>
<td>
<p>ancestral state at the root node for simulation.</p>
</td></tr>
<tr><td><code id="ctt_+3A_nmaps">nmaps</code></td>
<td>
<p>number of stochastic maps per simulation.</p>
</td></tr>
<tr><td><code id="ctt_+3A_nsim">nsim</code></td>
<td>
<p>for <code>sim.multiCtt</code> only, the number of simulations to run.</p>
</td></tr>
<tr><td><code id="ctt_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a 'changes through time' plot in the style of a lineage-through-time (LTT) plot. It shows the mean rate or the mean number of changes per unit time from a set of stochastic character map trees.
</p>


<h3>Value</h3>

<p>An object of class <code>"ctt"</code> or <code>"multiCtt"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ltt">ltt</a></code>
</p>

<hr>
<h2 id='density.multiSimmap'>Computes a posterior distribution for the number and types of changes on the tree</h2><span id='topic+density.multiSimmap'></span><span id='topic+plot.changesMap'></span>

<h3>Description</h3>

<p>S3 methods to summarize the results of one or more stochastic maps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'multiSimmap'
density(x, ...)
## S3 method for class 'changesMap'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="density.multiSimmap_+3A_x">x</code></td>
<td>
<p>object of class <code>"multiSimmap"</code> (see <code><a href="#topic+make.simmap">make.simmap</a></code>), or, in the case <code>plot.changesMap</code>, an object of class <code>"changesMap"</code> produced via a call to <code>density.multiSimmap</code>.</p>
</td></tr>
<tr><td><code id="density.multiSimmap_+3A_...">...</code></td>
<td>
<p>optional arguments. For <code>density.multiSimmap</code> these consist of <code>bw</code> (bandwidth) &amp; <code>method</code> (<code>"changes"</code>, <code>"densityMap"</code>, or <code>"timings"</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>density.multiSimmap</code> <code>method="changes"</code> (the default) results in a posterior distribution of the number &amp; types of changes on the tree.
</p>
<p>If the package <span class="pkg">coda</span> has been installed, then the function <code>HPD.interval</code> is used to compute a 95% high probability density interval for the number of changes of each type on the tree. Otherwise, the central 95% of the posterior sample is returned as an estimate of the 95% HPD interval for each change type. 
</p>
<p>The method also computes the full posterior density for each change type using a bandwidth specified by the user. 
</p>
<p><code>method="densityMap"</code> computes a standard <code>"<a href="#topic+densityMap">densityMap</a>"</code> object, and thus only permits binary characters.
</p>
<p><code>plot.changesMap</code> plots the posterior density returned by <code>density.multiSimmap</code> for <code>method=</code> <code>"changes"</code>.
</p>


<h3>Value</h3>

<p>For <code>method="changes"</code> <code>density.multiSimmap</code> returns an object of class <code>"changesMap"</code>.
</p>
<p>For <code>method="densityMap"</code> <code>density.multiSimmap</code> returns an object of class <code>"<a href="#topic+densityMap">densityMap</a>"</code>.
</p>
<p><code>plot.changesMap</code> generates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='densityMap'>Plot posterior density of stochastic mapping on a tree</h2><span id='topic+densityMap'></span><span id='topic+plot.densityMap'></span>

<h3>Description</h3>

<p>Visualize posterior probability density from stochastic mapping using a color gradient on the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>densityMap(trees, res=100, fsize=NULL, ftype=NULL, lwd=3, check=FALSE,
   legend=NULL, outline=FALSE, type="phylogram", direction="rightwards",
   plot=TRUE, ...)
## S3 method for class 'densityMap'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="densityMap_+3A_trees">trees</code></td>
<td>
<p>set of phylogenetic trees in a modified <code>"multiPhylo"</code> object. Values for a two-state discrete character are mapped on the tree. See <code><a href="#topic+make.simmap">make.simmap</a></code> and <code><a href="#topic+read.simmap">read.simmap</a></code> for details.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_res">res</code></td>
<td>
<p>resolution for gradient plotting. Larger numbers indicate a finer (smoother) gradient.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_fsize">fsize</code></td>
<td>
<p>relative font size - can be a vector with the second element giving the font size for the legend.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_ftype">ftype</code></td>
<td>
<p>font type - see options in <code><a href="#topic+plotSimmap">plotSimmap</a></code>. As with <code>fsize</code>, can be a vector with the second element giving font type for the legend.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_lwd">lwd</code></td>
<td>
<p>line width for branches. If a vector of two elements is supplied, the second element will be taken to be the desired width of the legend bar.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_check">check</code></td>
<td>
<p>check to make sure that the topology and branch lengths of all phylogenies in <code>trees</code> are equal.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_legend">legend</code></td>
<td>
<p>if <code>FALSE</code> no legend is plotted; if a numeric value, it gives the length of the legend in units of branch length. Default is 0.5 times the total tree length.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_outline">outline</code></td>
<td>
<p>logical value indicating whether or not to outline the branches of the tree in black.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_type">type</code></td>
<td>
<p>type of plot desired. Options are <code>"phylogram"</code> for a rightward square phylogram; and <code>"fan"</code> for a circular phylogram.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether or not to plot the tree. If <code>plot=FALSE</code> then an object of class <code>"densityMap"</code> will be returned without plotting.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_direction">direction</code></td>
<td>
<p>plotting direction for <code>type="phylogram"</code>.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_x">x</code></td>
<td>
<p>for <code>plot.densityMap</code>, an object of class <code>"densityMap"</code>.</p>
</td></tr>
<tr><td><code id="densityMap_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>plot.densityMap</code>. These include all the arguments of <code>densityMap</code> except <code>trees</code> and <code>res</code>. Additional optional arguments include <code>mar</code> (margins), <code>offset</code> (tip label offset: in units of the edge length or character widths, as in <code><a href="#topic+plotSimmap">plotSimmap</a></code>), and <code>hold</code> (whether or not to use <code>dev.hold</code> to hold output to graphical device before plotting; defaults to <code>hold=TRUE</code>). Also, the argument <code>states</code> can be used to 'order' the states on the probability axis (that is, which state should correspond to a posterior probability of 0 or 1). Some other plotting arguments, such as <code>xlim</code> and <code>ylim</code>, may also work.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function plots a tree with the posterior density for a mapped character from stochastic character mapping on the tree. Since the mapped value is the probability of being in state &quot;1&quot;, only binary [0,1] characters are allowed.
</p>


<h3>Value</h3>

<p>Plots a tree and returns an object of class <code>"densityMap"</code> invisibly.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>Huelsenbeck, J. P., R. Neilsen, and J. P. Bollback (2003) Stochastic mapping of morphological characters. <em>Systematic Biology</em>, <b>52</b>, 131-138. 
</p>
<p>Revell, L. J. (2013) Two new graphical methods for mapping trait evolution on phylogenies. <em>Methods in Ecology and Evolution</em>, <b>4</b>, 754-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## load tree and data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract discrete character (feeding mode)
fmode&lt;-setNames(sunfish.data$feeding.mode,
     rownames(sunfish.data))
## do stochastic mapping
smap.trees&lt;-make.simmap(sunfish.tree,fmode,model="ER",
     nsim=100)
## compute "densityMap" object
sunfish.dmap&lt;-densityMap(smap.trees,plot=FALSE,
	res=50) ## res should be higher
## plot density map
plot(sunfish.dmap,lwd=5,outline=TRUE)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
## End(Not run)
</code></pre>

<hr>
<h2 id='densityTree'>Plots a posterior sample of trees</h2><span id='topic+densityTree'></span><span id='topic+make.transparent'></span>

<h3>Description</h3>

<p>Functions plots a posterior sample of trees, including with mapped discrete characters.
</p>
<p><code>make.transparent</code> is used internally and converts a color to transparent with a certain user-specified <code>alpha</code> level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>densityTree(trees, colors="blue", alpha=NULL, method="plotTree", 
   fix.depth=FALSE, use.edge.length=TRUE, compute.consensus=TRUE, 
   use.gradient=FALSE, show.axis=TRUE, ...)
make.transparent(color, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="densityTree_+3A_trees">trees</code></td>
<td>
<p>an object of class <code>"multiPhylo"</code> or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_colors">colors</code></td>
<td>
<p>a color or a named vector of colors in which names correspond to mapped states in an object of class <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_alpha">alpha</code></td>
<td>
<p>transparency level for plotted trees which is passed to internally used function, <code>make.transparent</code>. (<code>0</code> is fully transparent, which <code>1</code> is fully opaque.) By default will be one divided by the number of trees.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_method">method</code></td>
<td>
<p>plotting method to be used internally. Can be <code>"plotTree"</code> or <code>"plotSimmap"</code>.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_fix.depth">fix.depth</code></td>
<td>
<p>logical value indicating whether or not to plot trees with a fixed depth or to permit plotted trees to have different depths.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_use.edge.length">use.edge.length</code></td>
<td>
<p>logical value indicating whether to use the edge lengths of the input tree. Defaults to <code>use.edge.length=TRUE</code> unless any input tree edge lengths are <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_compute.consensus">compute.consensus</code></td>
<td>
<p>logical value indicating whether or not to use the tip order from a consensus tree. (Defaults to <code>compute.consensus=TRUE</code> Defaulted to <code>FALSE</code> in earlier version of this function.)</p>
</td></tr>
<tr><td><code id="densityTree_+3A_use.gradient">use.gradient</code></td>
<td>
<p>logical value indicating whether to plot all trees slightly offset using a rainbow color gradient. (Defaults to <code>use.gradient=FALSE</code>.)</p>
</td></tr>
<tr><td><code id="densityTree_+3A_show.axis">show.axis</code></td>
<td>
<p>logical value indicating whether or not to include a horizontal axis in the plot.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code>plotTree</code> or <code>plotSimmap</code>. Some may be ignored if they are incompatible with the method.</p>
</td></tr>
<tr><td><code id="densityTree_+3A_color">color</code></td>
<td>
<p>in <code>make.transparent</code>, the color (or colors in a vector) to render transparent.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function creates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='describe.simmap'>Summarizes a stochastic mapped tree or set of trees</h2><span id='topic+describe.simmap'></span><span id='topic+summary.simmap'></span><span id='topic+summary.multiSimmap'></span><span id='topic+plot.describe.simmap'></span>

<h3>Description</h3>

<p>Summarizes the result of one or more stochastic maps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>describe.simmap(tree, ...)
## S3 method for class 'simmap'
summary(object, ...)
## S3 method for class 'multiSimmap'
summary(object, ...)
## S3 method for class 'describe.simmap'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="describe.simmap_+3A_tree">tree</code></td>
<td>
<p>a single tree or a set of trees as an object of class <code>"simmap"</code> or <code>"multiSimmap"</code>, respectively.</p>
</td></tr>
<tr><td><code id="describe.simmap_+3A_object">object</code></td>
<td>
<p>object of class <code>"simmap"</code> or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="describe.simmap_+3A_x">x</code></td>
<td>
<p>for S3 <code>plot</code> method, an object of class <code>"describe.simmap"</code>.</p>
</td></tr>
<tr><td><code id="describe.simmap_+3A_...">...</code></td>
<td>
<p>optional arguments which include: <code>plot</code>, a logical value indicating whether or not to plot the posterior probabilities at nodes (default is <code>plot=FALSE</code>); <code>check.equal</code>, a logical value indicating whether or not to check if all trees are equal using <code><a href="ape.html#topic+all.equal.phylo">all.equal.phylo</a></code> (default is <code>check.equal=FALSE</code>); and <code>message</code>, a logical indicating whether or not to print an informative message to the screen (default is <code>message=TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"describe.simmap"</code> with the following elements:
</p>
<table>
<tr><td><code>count</code></td>
<td>
<p>a matrix containing the number and types of transitions for each tree, if <code>tree</code> is an object of class <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>a matrix containing the times spend in each state on each tree.</p>
</td></tr>
<tr><td><code>ace</code></td>
<td>
<p>the posterior probabilities of each node being in each state, if <code>tree</code> is an object of class <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code>legend</code></td>
<td>
<p>a vector containing the plot legend, if <code>plot=TRUE</code>.</p>
</td></tr>
</table>
<p>if <code>class(tree)="simmap"</code> then the function simply returns the results of <code><a href="#topic+countSimmap">countSimmap</a></code> combined with the states at each node of the tree and a matrix containing the total and relative times spent in each state on the tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='di2multi.simmap'>Collapse or resolve polytomies in a tree with a character painted on the edges</h2><span id='topic+di2multi.simmap'></span><span id='topic+multi2di.simmap'></span><span id='topic+di2multi.multiSimmap'></span><span id='topic+multi2di.multiSimmap'></span><span id='topic+di2multi.contMap'></span><span id='topic+multi2di.contMap'></span><span id='topic+di2multi.densityMap'></span><span id='topic+multi2di.densityMap'></span>

<h3>Description</h3>

<p>Collapses or resolves polytomies in special types of phylogenetic trees. (See <code><a href="ape.html#topic+di2multi">di2multi</a></code> and <code><a href="ape.html#topic+multi2di">multi2di</a></code>.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simmap'
di2multi(phy, ...)
## S3 method for class 'simmap'
multi2di(phy, ...)
## S3 method for class 'multiSimmap'
di2multi(phy, ...)
## S3 method for class 'multiSimmap'
multi2di(phy, ...)
## S3 method for class 'contMap'
di2multi(phy, ...)
## S3 method for class 'contMap'
multi2di(phy, ...)
## S3 method for class 'densityMap'
di2multi(phy, ...)
## S3 method for class 'densityMap'
multi2di(phy, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="di2multi.simmap_+3A_phy">phy</code></td>
<td>
<p>object of class <code>"simmap"</code>, <code>"multiSimmap"</code>, <code>"contMap"</code>, or <code>"densityMap"</code> containing a character mapped onto the edges of a tree or set of trees.</p>
</td></tr>
<tr><td><code id="di2multi.simmap_+3A_...">...</code></td>
<td>
<p>optional arguments: <code>tol</code>, length below which edges should be treated as having zero length; and <code>random</code>, specifying whether to resolve polytomies randomly (if <code>TRUE</code>) or in the order in which they are encountered.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The method <code>di2multi</code> collapses branches of zero length (or, more specifically, branches with length shorter than <code>tol</code>) to create a polytomy in a tree or set of trees.
</p>
<p>The method <code>multi2di</code> resolves polytomies by adding branches of zero length (while preserving the mappings) in a tree or set of trees.
</p>
<p>This methods should theoretically behave similarly to <code><a href="ape.html#topic+di2multi">di2multi</a></code> and <code><a href="ape.html#topic+multi2di">multi2di</a></code> from the <span class="pkg">ape</span> package.
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code>, <code>"multiSimmap"</code>, <code>"contMap"</code>, or <code>"densityMap"</code>, depending on the class of <code>phy</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contMap">contMap</a></code>, <code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="ape.html#topic+di2multi">di2multi</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="ape.html#topic+multi2di">multi2di</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>
</p>

<hr>
<h2 id='dotTree'>Creates a phylogenetic dot plot</h2><span id='topic+dotTree'></span><span id='topic+dot.legend'></span>

<h3>Description</h3>

<p>Creates a plot in which different sized dots/circles represent different tip values for a quantitative trait.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dotTree(tree, x, legend=TRUE, method="plotTree", standardize=FALSE, ...)
dot.legend(x, y, min, max, Ntip, length=5, prompt=FALSE, method="plotTree", 
   ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dotTree_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_x">x</code></td>
<td>
<p>vector of trait values; or a matrix. If <code>x</code> is a vector it must have names that correspond to the tip labels of <code>tree</code>. If <code>x</code> is a matrix (and it probably should be a <em>matrix</em>, not a data frame) then the row names of the matrix should correspond to the tip labels of the phylogeny. In the case of <code>dot.legend</code>, the <em>x</em> coordinate of the legend.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_legend">legend</code></td>
<td>
<p>logical value indicating whether or not a legend should be plotted.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_method">method</code></td>
<td>
<p>tree plotting method to be used internally. Will switch to <code>method="phylogram"</code> if the number of traits is greater than one. For <code>dot.legend</code>, it should be the method that was used for the plot.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_standardize">standardize</code></td>
<td>
<p>a logical value indicating whether or not to standardize <code>x</code>, or each column of <code>x</code>, to have a mean of zero &amp; variance of one prior to analysis.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_y">y</code></td>
<td>
<p><em>y</em> coordinate of the legend.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_min">min</code></td>
<td>
<p>minimum value for <code>dot.legend</code>.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_max">max</code></td>
<td>
<p>maximum value for <code>dot.legend</code>.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_ntip">Ntip</code></td>
<td>
<p>number of tips in the plotted tree for <code>dot.legend</code>.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_length">length</code></td>
<td>
<p>length of legend.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_prompt">prompt</code></td>
<td>
<p>logical value indicating whether or not to prompt for legend position.</p>
</td></tr>
<tr><td><code id="dotTree_+3A_...">...</code></td>
<td>
<p>optional arguments. In the case of <code>dotTree</code>, these will be passed to <code><a href="#topic+plotTree">plotTree</a></code> or a different internally used plotting function for <code>method="phylogram"</code>. See <code><a href="#topic+phylo.heatmap">phylo.heatmap</a></code> for more detail on these arguments. Other option for <code>dotTree</code> also include <code>data.type</code> (<code>"continuous"</code> or <code>"discrete"</code>), <code>colors</code>, <code>length</code>, for data type <code>"continuous"</code> the length of the legend in terms of plotted circles, <code>x.space</code>, the spacing of the columns in the plotted data matrix, and <code>leg.space</code>, the spacing of the legend dots (again, for <code>data.type="continuous"</code> only).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function creates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## log-transform trait data
log.mammal&lt;-log(mammal.data)
## plot dotTree
dotTree(mammal.tree,log.mammal,fsize=0.7,
    standardize=TRUE,length=10)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='drop.clade'>Drop a clade from a tree</h2><span id='topic+drop.clade'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drop.clade(tree, tip)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop.clade_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="drop.clade_+3A_tip">tip</code></td>
<td>
<p>set of tips in a clade.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function drops the clade containing the species in <code>tip</code>.
</p>
<p>Probably should not use unless you know what you're doing.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='drop.leaves'>Drop all the leaves (tips) from a tree</h2><span id='topic+drop.leaves'></span>

<h3>Description</h3>

<p>Drops all the leaves from a tree, leaving behind only the structure leading to internal nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drop.leaves(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop.leaves_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="drop.leaves_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently includes only the logical value <code>keep.tip.labels</code> which tells the function how to labels the tips on the reduced tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='drop.tip.contMap'>Drop tip or tips from an object of class <code>"contMap"</code> or <code>"densityMap"</code></h2><span id='topic+drop.tip.contMap'></span><span id='topic+drop.tip.densityMap'></span><span id='topic+keep.tip.contMap'></span><span id='topic+keep.tip.densityMap'></span>

<h3>Description</h3>

<p>Drops one or multiple tips from an object of class <code>"contMap"</code> or <code>"densityMap"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'contMap'
drop.tip(phy, tip, ...)
## S3 method for class 'densityMap'
drop.tip(phy, tip, ...)
## S3 method for class 'contMap'
keep.tip(phy, tip, ...)
## S3 method for class 'densityMap'
keep.tip(phy, tip, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop.tip.contMap_+3A_phy">phy</code></td>
<td>
<p>an object of class <code>"contMap"</code> or <code>"densityMap"</code>.</p>
</td></tr>
<tr><td><code id="drop.tip.contMap_+3A_tip">tip</code></td>
<td>
<p>name or names of species to be dropped or kept.</p>
</td></tr>
<tr><td><code id="drop.tip.contMap_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="#topic+drop.tip.simmap">drop.tip.simmap</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are equivalent to <code><a href="ape.html#topic+drop.tip">drop.tip</a></code> and <code><a href="ape.html#topic+keep.tip">keep.tip</a></code> in the <span class="pkg">ape</span> package, but for objects of class <code>"contMap"</code> and <code>"densityMap"</code>.
</p>
<p>For more information about objects of class <code>"contMap"</code> or <code>"densityMap"</code>, please refer to the documentation pages for <code><a href="#topic+contMap">contMap</a></code> or <code><a href="#topic+densityMap">densityMap</a></code>, respectively.
</p>


<h3>Value</h3>

<p>An object of class <code>"contMap"</code> or <code>"densityMap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contMap">contMap</a></code>, <code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="ape.html#topic+drop.tip">drop.tip</a></code>, <code><a href="#topic+drop.tip.simmap">drop.tip.simmap</a></code>, <code><a href="ape.html#topic+keep.tip">keep.tip</a></code>
</p>

<hr>
<h2 id='drop.tip.multiSimmap'>Drop or keep tip or tips from an object of class <code>"multiSimmap"</code></h2><span id='topic+drop.tip.multiSimmap'></span><span id='topic+keep.tip.multiSimmap'></span>

<h3>Description</h3>

<p>This function drops or keeps one or multiple tips from all the trees of an object of class <code>"multiSimmap"</code>.	
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'multiSimmap'
drop.tip(phy, tip, ...)
## S3 method for class 'multiSimmap'
keep.tip(phy, tip, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop.tip.multiSimmap_+3A_phy">phy</code></td>
<td>
<p>an object of class <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="drop.tip.multiSimmap_+3A_tip">tip</code></td>
<td>
<p>name or names of species to be dropped, in a vector.</p>
</td></tr>
<tr><td><code id="drop.tip.multiSimmap_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="#topic+drop.tip.simmap">drop.tip.simmap</a></code>. Most optional arguments work, with the exception of <code>interactive=TRUE</code> which will return an error.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Equivalent to <code><a href="ape.html#topic+drop.tip">drop.tip</a></code> and <code><a href="ape.html#topic+keep.tip">keep.tip</a></code> in <span class="pkg">ape</span>.
</p>
<p>This function merely wraps <code><a href="#topic+drop.tip.simmap">drop.tip.simmap</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"multiSimmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+drop.tip">drop.tip</a></code>, <code><a href="ape.html#topic+drop.tip.multiPhylo">drop.tip.multiPhylo</a></code>, <code><a href="#topic+drop.tip.simmap">drop.tip.simmap</a></code>
</p>

<hr>
<h2 id='drop.tip.simmap'>Drop tips or extract clade from tree with mapped discrete character</h2><span id='topic+drop.tip.simmap'></span><span id='topic+keep.tip.simmap'></span><span id='topic+extract.clade.simmap'></span>

<h3>Description</h3>

<p>This function drops one or multiple tips from an object of class <code>"simmap"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simmap'
drop.tip(phy, tip, ...)
## S3 method for class 'simmap'
keep.tip(phy, tip, ...)
extract.clade.simmap(tree, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop.tip.simmap_+3A_phy">phy</code></td>
<td>
<p>an object of class <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="drop.tip.simmap_+3A_tip">tip</code></td>
<td>
<p>name or names of species to be dropped (or kept).</p>
</td></tr>
<tr><td><code id="drop.tip.simmap_+3A_node">node</code></td>
<td>
<p>node number for the root node of the clade to be extracted.</p>
</td></tr>
<tr><td><code id="drop.tip.simmap_+3A_...">...</code></td>
<td>
<p>optional arguments. Currently the logical argument <code>untangle</code> which if set to <code>TRUE</code> will call <code><a href="#topic+untangle">untangle</a></code> before returning the <code>"simmap"</code> object to the user.</p>
</td></tr>
<tr><td><code id="drop.tip.simmap_+3A_tree">tree</code></td>
<td>
<p>for <code>extract.clade.simmap</code>, an object of class <code>"simmap"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Equivalent to <code><a href="ape.html#topic+drop.tip">drop.tip</a></code> and <code><a href="ape.html#topic+keep.tip">keep.tip</a></code> but for a tree with a mapped discrete character.
</p>
<p><code>extract.clade.simmap</code> is functionally equivalent to <code><a href="ape.html#topic+extract.clade">extract.clade</a></code> but preserves discrete character mappings on the tree.
</p>
<p>Following <code><a href="ape.html#topic+drop.tip">drop.tip</a></code> in <span class="pkg">ape</span>, the returned tree is always in <code>"cladewise"</code> order.
</p>


<h3>Value</h3>

<p>A modified object of class <code>"phylo"</code> containing the elements <code>maps</code> and <code>$mapped.edge</code> with the time spent in each state along each edge of the tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+drop.tip">drop.tip</a></code>, <code><a href="ape.html#topic+extract.clade">extract.clade</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+sim.history">sim.history</a></code>
</p>

<hr>
<h2 id='Dtest'>Conducts correlational D-test from stochastic mapping</h2><span id='topic+Dtest'></span>

<h3>Description</h3>

<p>Conducts the 'D-test' of Huelsenbeck et al. (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dtest(t1, t2, nsim=100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dtest_+3A_t1">t1</code></td>
<td>
<p>set of stochastic map trees (i.e., object of class <code>"multiSimmap"</code> for character 1. Note that <code>t1</code> and <code>t2</code> should be of the same length.</p>
</td></tr>
<tr><td><code id="Dtest_+3A_t2">t2</code></td>
<td>
<p>set of stochastic map trees (i.e., object of class <code>"multiSimmap"</code> for character 2. Note that <code>t1</code> and <code>t2</code> should be of the same length.</p>
</td></tr>
<tr><td><code id="Dtest_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations to use in the test.</p>
</td></tr>
<tr><td><code id="Dtest_+3A_...">...</code></td>
<td>
<p>arguments to be passed internally to <code>make.simmap</code>. Note that (for now) these must be the same for both <code>t1</code> and <code>t2</code> (that is to say, we are not able to assume different trait evolution models for each tree).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function has been included without much testing, and so the user should be wary.
</p>


<h3>Value</h3>

<p>An object of class <code>"Dtest"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Huelsenbeck, J. P., R. Neilsen, and J. P. Bollback (2003) Stochastic mapping of morphological characters. <em>Systematic Biology</em>, <b>52</b>, 131-138.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+map.overlap">map.overlap</a></code>
</p>

<hr>
<h2 id='edge.widthMap'>Map continuous trait evolution on the tree</h2><span id='topic+edge.widthMap'></span><span id='topic+plot.edge.widthMap'></span>

<h3>Description</h3>

<p>Maps a discrete character onto the edges of the tree using variable edge widths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edge.widthMap(tree, x, ...)
## S3 method for class 'edge.widthMap'
plot(x, max.width=0.9, legend="trait value", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edge.widthMap_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="edge.widthMap_+3A_x">x</code></td>
<td>
<p>a numerical vector of phenotypic trait values for species. <code>names(x)</code> should contain the species names and match <code>tree$tip.label</code>. Or, for <code>plot.edge.widthMap</code>, an object of class <code>"edge.widthMap"</code>.</p>
</td></tr>
<tr><td><code id="edge.widthMap_+3A_max.width">max.width</code></td>
<td>
<p>maximum edge width in plot units.</p>
</td></tr>
<tr><td><code id="edge.widthMap_+3A_legend">legend</code></td>
<td>
<p>label for the plot legend.</p>
</td></tr>
<tr><td><code id="edge.widthMap_+3A_...">...</code></td>
<td>
<p>optional arguments - especially for the <code>plot</code> method. Perhaps the most important of these is <code>min.width</code>, which defaults to <code>0</code> but could probably be increased for many datasets and graphical devices. Other arguments are passed internally to <code><a href="#topic+plotTree">plotTree</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>edge.widthMap</code> returns an object of class <code>"edge.widthMap"</code>.
</p>
<p><code>plot.edge.widthMap</code> can be used to plot this object.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contMap">contMap</a></code>, <code><a href="#topic+fastAnc">fastAnc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
    rownames(mammal.data)))
## create "edge.widthMap" object
mammal.ewMap&lt;-edge.widthMap(mammal.tree,ln.bodyMass,
    min.width=0.05)
## plot it
plot(mammal.ewMap,legend="log(body mass)")
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='edgeProbs'>Compute the relative frequencies of state changes along edges</h2><span id='topic+edgeProbs'></span>

<h3>Description</h3>

<p>Computes the relative frequencies of character state changes along edges from a sample of stochastically mapped character histories. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edgeProbs(trees)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edgeProbs_+3A_trees">trees</code></td>
<td>
<p>an object of class <code>"multiSimmap"</code> containing a sample of trees that are identical in topology &amp; branch lengths with different stochastically mapped character histories.</p>
</td></tr>
</table>


<h3>Details</h3>

	
<p>The function assumes that all trees in the sample differ only in their mapped histories &amp; not at all in topology or branch lengths.
</p>
<p>Note that <code>edgeProbs</code> only asks whether the starting and ending states of the edge <em>differ</em> in a particular way, and thus ignores multiple-hits along a single edge.
</p>


<h3>Value</h3>

<p>The object that is returned is a matrix with the state changes &amp; the relative frequency of each state change. Rows are in the order of the matrix <code>edge</code> for any of the mapped trees.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>

<hr>
<h2 id='estDiversity'>Estimate diversity at each node of the tree</h2><span id='topic+estDiversity'></span>

<h3>Description</h3>

<p>Estimates the lineage density at each node in the tree based on a biogeographic model (similar to Mahler et al. 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estDiversity(tree, x, method=c("asr","simulation"), model="ER", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estDiversity_+3A_tree">tree</code></td>
<td>
<p>is a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="estDiversity_+3A_x">x</code></td>
<td>
<p>a vector containing the biogeographic area for each of the tip taxa.</p>
</td></tr>
<tr><td><code id="estDiversity_+3A_method">method</code></td>
<td>
<p>method for reconstructing ancestral biogeography.</p>
</td></tr>
<tr><td><code id="estDiversity_+3A_model">model</code></td>
<td>
<p>model for ancestral character estimation. In theory, any model from <code><a href="ape.html#topic+ace">ace</a></code>; however only symmetric models permitted for <code>method="asr"</code>.</p>
</td></tr>
<tr><td><code id="estDiversity_+3A_...">...</code></td>
<td>
<p>optional arguments. So far, this includes only <code>nsim</code>, the number of stochastic mappings to conduct using <code><a href="#topic+make.simmap">make.simmap</a></code> for <code>method="simulation"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two different methods are implemented in the current version.
</p>
<p>For <code>method="asr"</code> the state at the current node, and at each position along each co-extant internal edge, is computed as the marginal (empirical Bayesian) ancestral state reconstruction using the re-rooting method of Yang (2006). The lineage density is then computed as the sum of the marginal reconstructions (posterior probabilities) times the summed marginal ancestral reconstructions across co-extant edges. 
</p>
<p>For <code>method="simulation"</code>, stochastic character mapping is used to generate optional argument <code>nsim</code> stochastic maps of ancestral biogeography. Then the lineage density at each node is computed as the number of co-existing lineages with the same biogeography as the focal node, averaged across stochastic maps.
</p>
<p>The importance of this distinction may depend on the degree to which reconstructions at internal nodes are independent, which relates to the distinction between marginal and joint reconstruction (e.g., see Yang 2006).
</p>


<h3>Value</h3>

<p>A vector containing the estimated lineage density at each node
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Mahler, D. L, L. J. Revell, R. E. Glor, and J. B. Losos. (2010) Ecological opportunity and the rate of morphological evolution in the diversification of Greater Antillean anoles. <em>Evolution</em>, <b>64</b>, 2731-2745.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Yang, Z. (2006) <em>Computational Molecular Evolution</em>. Oxford University Press.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitDiversityModel">fitDiversityModel</a></code>
</p>

<hr>
<h2 id='evol.rate.mcmc'>Bayesian MCMC method for identifying exceptional phenotypic diversification in a phylogeny</h2><span id='topic+evol.rate.mcmc'></span><span id='topic+print.evol.rate.mcmc'></span><span id='topic+summary.evol.rate.mcmc'></span><span id='topic+print.summary.evol.rate.mcmc'></span><span id='topic+plot.summary.evol.rate.mcmc'></span>

<h3>Description</h3>

<p>Estimates the phylogenetic location of a <em>single</em> rate shift on the tree using Bayesian MCMC (as described in Revell et al., 2012b).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evol.rate.mcmc(tree, x, ngen=10000, control=list(), ...)
## S3 method for class 'evol.rate.mcmc'
print(x, ...)
## S3 method for class 'evol.rate.mcmc'
summary(object, ...)
## S3 method for class 'summary.evol.rate.mcmc'
print(x, ...)
## S3 method for class 'summary.evol.rate.mcmc'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evol.rate.mcmc_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code> (a phylogenetic tree).</p>
</td></tr>
<tr><td><code id="evol.rate.mcmc_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species in which <code>names(x)</code> contains the species names of <code>tree</code>, an object of class <code>"evol.rate.mcmc"</code>, or (in the case of the S3 <code>summary</code> method) an object of class <code>"summary.evol.rate.mcmc"</code>.</p>
</td></tr>
<tr><td><code id="evol.rate.mcmc_+3A_ngen">ngen</code></td>
<td>
<p>an integer value indicating the number of generations for the MCMC.</p>
</td></tr>
<tr><td><code id="evol.rate.mcmc_+3A_control">control</code></td>
<td>
<p>a list of control parameters containing the following elements: <code>sig1</code>: starting value for <code class="reqn">\sigma_1^2</code>; <code>sig2</code>: starting value for <code class="reqn">\sigma_2^2</code>; <code>a</code>: starting value for a; <code>sd1</code>: standard deviation for the normal proposal distribution for <code class="reqn">\sigma_1^2</code>; <code>sd2</code>: standard deviation for the normal proposal distribution for <code class="reqn">\sigma_2^2</code>; <code>kloc</code>: scaling parameter for tree move proposals - <code class="reqn">1/\lambda</code> for the reflected exponential distribution; <code>sdlnr</code>: standard deviation on the log-normal prior on <code class="reqn">\sigma_1^2/\sigma_2^2</code>; <code>rand.shift</code>: probability of proposing a random shift in the tree (improves mixing); <code>print</code>: print frequency for the MCMC; <code>sample</code>: sample frequency.</p>
</td></tr>
<tr><td><code id="evol.rate.mcmc_+3A_object">object</code></td>
<td>
<p>for the S3 <code>summary</code> method, an object of class <code>"evol.rate.mcmc"</code>.</p>
</td></tr>
<tr><td><code id="evol.rate.mcmc_+3A_...">...</code></td>
<td>
<p>other optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a phylogenetic tree and data for a single continuously valued character and uses a Bayesian MCMC approach to identify the phylogenetic location of a shift in the evolutionary rate through time.
</p>
<p>Default values of <code>control</code> are given in Revell et al. (2012b).
</p>


<h3>Value</h3>

<p>An object of class <code>"evol.rate.mcmc"</code> consisting of at least the following elements:
</p>
<table>
<tr><td><code>mcmc</code></td>
<td>
<p>results from the MCMC run.</p>
</td></tr>
<tr><td><code>tips</code></td>
<td>
<p>list of tips in rate <code class="reqn">\sigma_1^2</code> for each sampled generation of MCMC (to polarize the rate shift).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., D. L. Mahler, P. Peres-Neto, and B. D. Redelings (2012) A new method for identifying exceptional phenotypic diversification. <em>Evolution</em>, <b>66</b>, 135-146.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+evol.vcv">evol.vcv</a></code>, <code><a href="#topic+minSplit">minSplit</a></code>, <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>
</p>

<hr>
<h2 id='evol.vcv'>Likelihood test for variation in the evolutionary variance-covariance matrix</h2><span id='topic+evol.vcv'></span>

<h3>Description</h3>

<p>Fits a multi-regime multivariate Brownian motion model following Revell &amp; Collar (2009).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evol.vcv(tree, X, maxit=2000, vars=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evol.vcv_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code>. If <code>tree</code> is an object of class <code>"phylo"</code> then a simple multivariate Brownian motion model will be fit to the data in <code>X</code>.</p>
</td></tr>
<tr><td><code id="evol.vcv_+3A_x">X</code></td>
<td>
<p>an <code>n</code> x <code>m</code> matrix of tip values for <code>m</code> continuously valued traits in <code>n</code> species - row names should be species names. If <code>X</code> is supplied as a data frame it will be coerced into a matrix without warning.</p>
</td></tr>
<tr><td><code id="evol.vcv_+3A_maxit">maxit</code></td>
<td>
<p>an optional integer value indicating the maximum number of iterations for optimization. This quantity may need to be increased for difficult optimizations.</p>
</td></tr>
<tr><td><code id="evol.vcv_+3A_vars">vars</code></td>
<td>
<p>an optional logical value indicating whether or not to estimate the variances of the parameter estimates from the Hessian matrix.</p>
</td></tr>
<tr><td><code id="evol.vcv_+3A_...">...</code></td>
<td>
<p>optional arguments. The most important optional argument at this time is <code>error_vcv</code> which should contain a list of matrices of sampling <em>variances</em> and covariances for (and between) the means of each species. The sampling variance for the mean is just the square of the sampling error. Sampling covariances will tend to be zero (or close to zero) if error for different traits is uncorrelated, for instance, because different specimens were used to estimate the means for different traits, and non-zero otherwise.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes an object of class <code>"simmap"</code> with a mapped binary or multi-state trait and data for an arbitrary number of continuously valued character. It then fits the multiple evolutionary variance-covariance matrix (rate matrix) model of Revell &amp; Collar (2009; <em>Evolution</em>).
</p>
<p><code>evol.vcv</code> performs optimization by maximizing the likelihood with respect to the Cholesky matrices using <code><a href="stats.html#topic+optim">optim</a></code>.	Optimization is by <code>method="Nelder-Mead"</code>. Using box constraints does not make sense here as they would be applied to the Cholesky matrix rather than the target parameters. Users may have to increase <code>maxit</code> for large trees and/or more than two traits.
</p>


<h3>Value</h3>

<p>An object of class <code>"evol.vcv"</code> with the following elements:
</p>
<table>
<tr><td><code>R.single</code></td>
<td>
<p>vcv matrix for the single rate matrix model.</p>
</td></tr>
<tr><td><code>vars.single</code></td>
<td>
<p>optionally, a matrix containing the variances of the elements of <code>R.single</code>.</p>
</td></tr>
<tr><td><code>logL1</code></td>
<td>
<p>log-likelihood for single matrix model.</p>
</td></tr>
<tr><td><code>k1</code></td>
<td>
<p>number of parameters in the single matrix model.</p>
</td></tr>
<tr><td><code>R.multiple</code></td>
<td>
<p><code>m</code> x <code>m</code> x <code>p</code> array containing the <code>p</code> estimated vcv matrices for the <code>p</code> regimes painted on the tree.</p>
</td></tr>
<tr><td><code>vars.multiple</code></td>
<td>
<p>optionally, an array containing the variances of the parameter estimates in <code>R.multiple</code>.</p>
</td></tr>
<tr><td><code>logL.multiple</code></td>
<td>
<p>log-likelihood of the multi-matrix model.</p>
</td></tr>
<tr><td><code>k2</code></td>
<td>
<p>number of parameters estimated in this model.</p>
</td></tr>
<tr><td><code>P.chisq</code></td>
<td>
<p>P-value of the <code class="reqn">\chi^2</code> test on the likelihood ratio.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>logical value indicating whether or not the optimization has converged.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., and D. C. Collar (2009) Phylogenetic analysis of the evolutionary correlation using likelihood. <em>Evolution</em>, <b>63</b>, 1090-1100.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>, <code><a href="#topic+brownie.lite">brownie.lite</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## fit multi-correlation model
sunfish.fit&lt;-evol.vcv(sunfish.tree,sunfish.data[,2:3])
print(sunfish.fit)
</code></pre>

<hr>
<h2 id='evolvcv.lite'>Likelihood test for a shift in the evolutionary correlation between traits</h2><span id='topic+evolvcv.lite'></span>

<h3>Description</h3>

<p>Fits a multi-regime model for differences in the evolutionary variance-covariance structure or correlation between two continuous traits, following Revell &amp; Collar (2009) and Revell et al. (2022).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evolvcv.lite(tree, X, maxit=2000, tol=1e-10, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evolvcv.lite_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code>. If <code>tree</code> is an object of class <code>"phylo"</code> then a simple multivariate Brownian motion model will be fit to the data in <code>X</code>.</p>
</td></tr>
<tr><td><code id="evolvcv.lite_+3A_x">X</code></td>
<td>
<p>an <code>n</code> x <code>m</code> matrix of tip values for <code>m</code> continuously valued traits in <code>n</code> species - row names should be species names. If <code>X</code> is supplied as a data frame it will be coerced into a matrix without warning.</p>
</td></tr>
<tr><td><code id="evolvcv.lite_+3A_maxit">maxit</code></td>
<td>
<p>an optional integer value indicating the maximum number of iterations for optimization - may need to be increased for large trees.</p>
</td></tr>
<tr><td><code id="evolvcv.lite_+3A_tol">tol</code></td>
<td>
<p>tolerance value for <code>"L-BFGS-B"</code> optimization.</p>
</td></tr>
<tr><td><code id="evolvcv.lite_+3A_...">...</code></td>
<td>
<p>other optional arguments. The most important optional argument is probably <code>models</code> which species the models to be fit. See <em>Description</em> for more information. A second useful argument is <code>error_vcv</code> which should be supplied as a list of matrices of sampling <em>variances</em> and covariances for (and between) the means of each species. The sampling variance for the mean is just the square of the sampling error. Sampling covariances will tend to be zero (or close to zero) if error for different traits is uncorrelated, for instance, because different specimens were used to estimate the means for different traits, and non-zero otherwise.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes an object of class <code>"simmap"</code> with a mapped binary or multi-state trait and data for two and only two continuously valued character. It then fits (by default) four different evolutionary models: common rates and correlation; different rates, common correlation; different correlations, common rates; no common structure.
</p>
<p>In addition to the four default models specified above, <code>evolvcv.lite</code> now fits an additional four additional models.
</p>
<p>The set of models to be fit can be specified using the optional argument <code>models</code> in multiple ways.
</p>
<p>First, if left unspecified, then the four models listed above will be fit.
</p>
<p>Second, if <code>models</code> is set to <code>"all models"</code> than eight models will be fit.
</p>
<p>Lastly, one or more (up to all eight) models can be fit by encoding the models to be fit into a single vector containing a subset or all of the following elements: <code>"1"</code>, <code>"2"</code>, <code>"2b"</code>, <code>"3"</code>, <code>"3b"</code>, <code>"3c"</code>, and <code>"4"</code>. These codes correspond to the following eight models: 1. common rates, common correlation; 2. different rates, common correlation; 2b. different rates for trait 1 only, common correlation; 2c. different rates for trait 2 only, common correlation; 3. common rates, different correlations; 3b. different rates for trait 1 only, different correlations; 3c. different rates for trait 2 only, different correlation; and 4. no common structure.
</p>


<h3>Value</h3>

<p>A list with the results summarized for each model.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., and D. C. Collar (2009) Phylogenetic analysis of the evolutionary correlation using likelihood. <em>Evolution</em>, <b>63</b>, 1090-1100.
</p>
<p>Revell, L. J., K. S. Toyama, and D. L. Mahler (2022) A simple hierarchical model for heterogeneity in the evolutionary correlation on a phylogenetic tree. <em>PeerJ</em>, <b>10</b>, e13910.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+evol.vcv">evol.vcv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## fit heirarchical common-structure models
sunfish.fit&lt;-evolvcv.lite(sunfish.tree,
    sunfish.data[,2:3],models="all models")
## print fitted models
print(sunfish.fit)
## compare models
anova(sunfish.fit)
</code></pre>

<hr>
<h2 id='exhaustiveMP'>Exhaustive and branch &amp; bound MP optimization</h2><span id='topic+exhaustiveMP'></span>

<h3>Description</h3>

<p>This function does exhaustive and branch &amp; bound MP searches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exhaustiveMP(data, tree=NULL, method="branch.and.bound")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exhaustiveMP_+3A_data">data</code></td>
<td>
<p>is a <code><a href="phangorn.html#topic+phyDat">phyDat</a></code> (Schliep 2011) object containing DNA or other data.</p>
</td></tr>
<tr><td><code id="exhaustiveMP_+3A_tree">tree</code></td>
<td>
<p>an optional input tree (used only with <code>method="branch.and.bound"</code>).</p>
</td></tr>
<tr><td><code id="exhaustiveMP_+3A_method">method</code></td>
<td>
<p>an optional string indicating which method to use: <code>"branch.and.bound"</code>, implementing a branch-and-bound search (obviously), or <code>"exhaustive"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Should probably not be used for more than about 8 species (and definitely not more than 10 species).  Performs parsimony calculations using <code><a href="phangorn.html#topic+parsimony">parsimony</a></code> in the <span class="pkg">phangorn</span> package (Schliep, 2011).
</p>


<h3>Value</h3>

<p>A <code>"phylo"</code> or <code>"multiPhylo"</code> object that is the MP tree or set of MP trees. It also returns the parsimony scores in <code>attr(trees,"pscore")</code> or <code>attr(trees[[i]],"pscore")</code> for the <em>i</em>th tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (2004) <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Schliep, K. P. (2011) phangorn: phylogenetic analysis in R. <em>Bioinformatics</em>, <b>27</b>, 592-593.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mrp.supertree">mrp.supertree</a></code>, <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code>, <code><a href="phangorn.html#topic+pratchet">pratchet</a></code>
</p>

<hr>
<h2 id='expand.clade'>Expands (or contracts) the tip-spacing of a given clade or clades</h2><span id='topic+expand.clade'></span><span id='topic+plot.expand.clade'></span>

<h3>Description</h3>

<p>Modify the tip-spacing of a plotted tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expand.clade(tree, node, factor=5)
## S3 method for class 'expand.clade'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expand.clade_+3A_tree">tree</code></td>
<td>
<p>tree an object of class <code>"phylo"</code> or <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="expand.clade_+3A_node">node</code></td>
<td>
<p>node index or vector of node indices.</p>
</td></tr>
<tr><td><code id="expand.clade_+3A_factor">factor</code></td>
<td>
<p>expansion factor for the tip-spacing of the taxa descended from node or nodes in <code>node</code>.</p>
</td></tr>
<tr><td><code id="expand.clade_+3A_x">x</code></td>
<td>
<p>for <code>plot</code> method, an object of class <code>"expand.clade"</code>.</p>
</td></tr>
<tr><td><code id="expand.clade_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code>plotTree</code> or <code>plotSimmap</code>, depending on the class of <code>x$tree</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The purpose of this function is to compute a custom tip-spacing for users who want to expand or contract the tip-spacing of the descendant taxa from a given node or nodes.
</p>


<h3>Value</h3>

<p>The function returns an object of class <code>"expand.clade"</code> which consists of the (possibly re-ordered) tree and a numerical vector with the calculated tip spacing based on the expansion factor specified by the user.
</p>
<p>This object can be plotted using the S3 <code>plot</code> method for the object class; or it can be plotted simply by calling a standard plotting function on the tree &amp; tip spacings.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='export.as.xml'>Export trees &amp; data in XML format</h2><span id='topic+export.as.xml'></span>

<h3>Description</h3>

<p>Exports trees &amp; character data in XML format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export.as.xml(file, trees, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export.as.xml_+3A_file">file</code></td>
<td>
<p>filename for export.</p>
</td></tr>
<tr><td><code id="export.as.xml_+3A_trees">trees</code></td>
<td>
<p>a phylogenetic tree or trees in <code>"phylo"</code> or <code>"multiPhylo"</code> format.</p>
</td></tr>
<tr><td><code id="export.as.xml_+3A_x">X</code></td>
<td>
<p>a matrix of class <code>"DNAbin"</code> or a matrix with discretely valued non-DNA character data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Can be used to create input file for the program SIMMAP v1.5 (Bollback 2006).
</p>


<h3>Value</h3>

<p>A file.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="ape.html#topic+read.nexus">read.nexus</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+write.simmap">write.simmap</a></code>
</p>

<hr>
<h2 id='fancyTree'>Plots special types of phylogenetic trees</h2><span id='topic+fancyTree'></span><span id='topic+phyloScattergram'></span><span id='topic+phenogram95'></span>

<h3>Description</h3>

<p>Plots phylogenies (or phylogenetic trees and comparative data) in a variety of different styles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fancyTree(tree, type=c("extinction","traitgram3d","droptip","densitymap",
   "contmap","phenogram95","scattergram"), ..., control=list())
phyloScattergram(tree, X=NULL, ...)
phenogram95(tree, x=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fancyTree_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fancyTree_+3A_type">type</code></td>
<td>
<p>the type of special plot to create. See Description.</p>
</td></tr>
<tr><td><code id="fancyTree_+3A_...">...</code></td>
<td>
<p>arguments to be passed to different methods. See Description.</p>
</td></tr>
<tr><td><code id="fancyTree_+3A_control">control</code></td>
<td>
<p>a list of control parameters, depending on <code>type</code>.</p>
</td></tr>
<tr><td><code id="fancyTree_+3A_x">X</code></td>
<td>
<p>in <code>phyloScattergram</code>, a matrix of continuous trait values. Row names in the matrix should correspond to species names in the tree.</p>
</td></tr>
<tr><td><code id="fancyTree_+3A_x">x</code></td>
<td>
<p>in <code>phenogram95</code>, a named vector with values for a continuously distributed trait.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots a phylogeny or phylogenetic tree and comparative data in a variety of different styles, depending on the value of <code>type</code>. In some instances, <code>fancyTree</code> is now just a wrappe for other <span class="pkg">phytools</span> functions, such as <code><a href="#topic+contMap">contMap</a></code> and <code><a href="#topic+densityMap">densityMap</a></code>.
</p>
<p>If <code>type="extinction"</code> (or any unambiguous abbreviation) the function will plot a tree in which branches preceding the MRCA of all extant taxa and branches leading only to extinct lineages are plotted with dashed red lines. 
</p>
<p>If <code>type="traitgram3d"</code> the function will plot a three dimensional traitgram (that is, a projection of the tree into three dimensional morphospace where two dimensions are the phenotypic trait and the third axis is time since the root). In this case, the additional argument <code>X</code>, a matrix containing the tip values of all species (with species IDs as row names) should be supplied. Optionally, the user can also supply the matrix <code>A</code>, which contains the ancestral states in the tree with rows labeled by node number. 
</p>
<p>If <code>type="droptip"</code> the function will create a two panel figure in which the first panel is the tree with lineages to be pruned highlighted; and the second panel is the pruned tree. In this case, the additional argument <code>tip</code>, the tip name or vector of tip names to be dropped, must be supplied. 
</p>
<p>If <code>type="densitymap"</code>, a posterior probability density &quot;heat-map&quot; is created based on a set of trees in a <code>"multiSimmap"</code> object containing a binary [0,1] mapped character. (See <code><a href="#topic+densityMap">densityMap</a></code> for additional optional arguments if <code>type="densitymap"</code>.)
</p>
<p>If <code>type="contmap"</code>, reconstructed continuous trait evolution is mapped on the tree. Again, see <code><a href="#topic+contMap">contMap</a></code> for additional arguments if <code>type="contmap"</code>.
</p>
<p>If <code>type="phenogram95"</code> a 95% traitgram (aka. &quot;phenogram&quot;) is plotted using transparency to visualize uncertainty at ancestral nodes and along branches. Most of the options of <code><a href="#topic+phenogram">phenogram</a></code> are available. 
</p>
<p>Finally, if <code>type="scattergram"</code> a phylogenetic scatter plot matrix containing <code><a href="#topic+contMap">contMap</a></code> style trees on the diagonal and <code><a href="#topic+phylomorphospace">phylomorphospace</a></code> plots in non-diagonal panels is produced. For this type a trait matrix <code>X</code> must also be supplied. The only additional arguments available for this type are <code>ftype</code>, <code>fsize</code>, <code>colors</code>, and <code>label</code>. (See <code><a href="#topic+phylomorphospace">phylomorphospace</a></code> for details on how these arguments should be used.) This function calls <code><a href="#topic+phyloScattergram">phyloScattergram</a></code> (which is also now exported to the name space) internally. In addition to creating a plot, <code>phyloScattergram</code> also returns an object of class <code>"phyloScattergram"</code> which can be replotted using different options if desired.
</p>
<p>Presently only <code>type="traitgram3d"</code> uses the list <code>control</code> which can be supplied the same set of control parameters as <code><a href="#topic+phylomorphospace3d">phylomorphospace3d</a></code>, as well as the control parameter <code>maxit</code> which will be passed to <code><a href="#topic+anc.ML">anc.ML</a></code>. 
</p>
<p>Finally, the optional argument <code>hold</code> will be passed to multiple methods if supplied. It is a logical value that indicates whether or not the output to the graphical device should be held using <code><a href="grDevices.html#topic+dev.hold">dev.hold</a></code> before plotting (defaults to <code>hold=TRUE</code>).
</p>


<h3>Value</h3>

<p>This function plots different types of phylogenetic trees. For <code>type="droptip"</code> the function also returns the pruned tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Evans, M. E. K., Smith, S. A., Flynn, R. S., Donoghue, M. J. (2009) Climate, niche evolution, and diversification of the &quot;bird-cage&quot; evening primroses (Oenothera, sections Anogra and Kleinia). <em>American Naturalist</em>, <b>173</b>, 225-240.
</p>
<p>Revell, L. J. (2013) Two new graphical methods for mapping trait evolution on phylogenies. <em>Methods in Ecology and Evolution</em>, <b>4</b>, 754-759.
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contMap">contMap</a></code>, <code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="ape.html#topic+drop.tip">drop.tip</a></code>, <code><a href="#topic+phenogram">phenogram</a></code>, <code><a href="#topic+phylomorphospace3d">phylomorphospace3d</a></code>, <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## plot tree with extinction
set.seed(10)
tree&lt;-pbtree(b=1,d=0.4,t=4)
fancyTree(tree,type="extinction")

## Not run: 
## plot 3D traitgram
## load data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
fancyTree(sunfish.tree,type="traitgram3d",
  X=sunfish.data[,2:3],
  control=list(spin=FALSE))
## End(Not run)
	
## plot with dropped tips
tree&lt;-pbtree(n=30)
tips&lt;-sample(tree$tip.label)[1:10]
pruned&lt;-fancyTree(tree,type="droptip",tip=tips)
par(mfrow=c(1,1)) ## reset mfrow to default

## Not run: 
## plot 95-percent CI phenogram
data(mammal.tree)
data(mammal.data)
bodyMass&lt;-setNames(mammal.data$bodyMass,
  rownames(mammal.data))
fancyTree(mammal.tree,type="phenogram95",x=bodyMass,
  fsize=0.7,ftype="i")
## End(Not run)

par(mar=c(5.1,4.1,4.1,2.1)) ## reset mar to defaults
</code></pre>

<hr>
<h2 id='fastAnc'>(Reasonably) fast estimation of ML ancestral states</h2><span id='topic+fastAnc'></span>

<h3>Description</h3>

<p>Estimates ancestral states for a continuous character under maximum likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastAnc(tree, x, vars=FALSE, CI=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastAnc_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fastAnc_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species; <code>names(x)</code> should be the species names.</p>
</td></tr>
<tr><td><code id="fastAnc_+3A_vars">vars</code></td>
<td>
<p>a logical value indicating whether or not to compute variances on the ancestral state estimates. Variances are based on Equation (6) of Rohlf (2001).</p>
</td></tr>
<tr><td><code id="fastAnc_+3A_ci">CI</code></td>
<td>
<p>a logical value indicating whether or not to compute 95% confidence intervals on state estimates.</p>
</td></tr>
<tr><td><code id="fastAnc_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently this consists of <code>anc.states</code>, a named vector containing ancestral states to fix. Names should correspond to node numbers in the input tree.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs (reasonably) fast estimation of the ML ancestral states for a continuous trait by taking advantage of the fact that the state computed for the root node of the tree during Felsenstein's (1985) contrasts algorithm is also the MLE of the root node. Thus, the function re-roots the tree at all internal nodes and computes the contrasts state at the root each time.
</p>
<p>The function can also (optionally) compute variances or 95% confidence intervals on the estimates.
</p>


<h3>Value</h3>

<p>An object of class <code>"fastAnc"</code> consisting of either: a named vector containing the states at internal nodes - names are node numbers; or a list containing ancestral state estimates (<code>ace</code>), variances on the estimates (<code>var</code>), and/or 95% confidence intervals (<code>CI95</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+anc.ML">anc.ML</a></code>, <code><a href="ape.html#topic+pic">pic</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
    rownames(mammal.data)))
## estimate ancestral body sizes
fit.BM&lt;-fastAnc(mammal.tree,ln.bodyMass,CI=TRUE)
print(fit.BM,printlen=10)
</code></pre>

<hr>
<h2 id='fastBM'>(Reasonably) fast quantitative trait simulation on phylogenies</h2><span id='topic+fastBM'></span>

<h3>Description</h3>

<p>Simulates one or multiple continuous traits on the tree under various evolutionary models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastBM(tree, a=0, mu=0, sig2=1, bounds=c(-Inf,Inf), internal=FALSE, nsim=1, 
   ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastBM_+3A_tree">tree</code></td>
<td>
<p>is a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_a">a</code></td>
<td>
<p>a value for ancestral state at the root node.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_mu">mu</code></td>
<td>
<p>an optional value for the mean of random normal changes along branches of the tree - can be used to simulate a trend if <code>mu!=0</code>.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_sig2">sig2</code></td>
<td>
<p>instantaneous variance of the BM process, <code class="reqn">\sigma^2</code>.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_bounds">bounds</code></td>
<td>
<p>a vector with the lower and upper bounds (respectively) for bounded Brownian simulation - by default simulation is unbounded.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_internal">internal</code></td>
<td>
<p>logical value indicating whether or not to return states for internal nodes.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations.</p>
</td></tr>
<tr><td><code id="fastBM_+3A_...">...</code></td>
<td>
<p>optional arguments <code>alpha</code> and <code>theta</code> used for OU simulation. If <code>alpha</code> is set then <code>mu</code> and <code>bounds</code> are ignored with a warning.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function conducts (reasonably) fast quantitative trait simulation on a phylogeny under several different models: Brownian motion (default), BM with a trend (for <code>mu!=0</code>), bounds (for <code>bounds!=c(-Inf,</code> <code>Inf)</code>), and OU. 
</p>


<h3>Value</h3>

<p>A vector (for <code>nsim=1</code>) or matrix containing the tip states for the <code>n</code> species in the tree, and (optionally) the ancestral states for internal nodes.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.corrs">sim.corrs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate 10 characters on the Anolis tree 
## under Brownian motion
data(anoletree)
X&lt;-fastBM(anoletree,nsim=10)
head(X)
</code></pre>

<hr>
<h2 id='fastMRCA'>Get the MRCA (or height above the root of the MRCA) of a pair of tip taxa</h2><span id='topic+fastMRCA'></span><span id='topic+fastHeight'></span><span id='topic+fastDist'></span>

<h3>Description</h3>

<p>Finds the most recent common ancestor (MRCA) for a pair of tip taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastMRCA(tree, sp1, sp2)
fastHeight(tree, sp1, sp2)
fastDist(tree, sp1, sp2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastMRCA_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fastMRCA_+3A_sp1">sp1</code></td>
<td>
<p>species one name.</p>
</td></tr>
<tr><td><code id="fastMRCA_+3A_sp2">sp2</code></td>
<td>
<p>species two name.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function (<code>fastMRCA</code>) returns the most recent common ancestor (node number) for a pair of taxa; or, in the case of <code>fastHeight</code>, the height above the root of the MRCA of a pair of taxa; or, in the case of <code>fastDist</code>, the patristic distance between a pair of taxa. 
</p>
<p>This function is mostly redundant with <code><a href="#topic+findMRCA">findMRCA</a></code> (or <code>findMRCA(...,type="height")</code> in the case of <code>fastHeight</code>), but for very large trees will be considerably faster.
</p>
<p>(Also see <code><a href="ape.html#topic+getMRCA">getMRCA</a></code> in the <span class="pkg">ape</span> package.)
</p>


<h3>Value</h3>

<p>The node number of the MRCA, the height above the root (for <code>fastHeight</code>), or the patristic distance between two taxa (for <code>fastDist</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+getMRCA">getMRCA</a></code>, <code><a href="#topic+findMRCA">findMRCA</a></code>, <code><a href="ape.html#topic+mrca">mrca</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree&lt;-pbtree(n=2000)
anc&lt;-fastMRCA(tree,"t1","t15")
</code></pre>

<hr>
<h2 id='findMRCA'>Get the MRCA of a set of taxa</h2><span id='topic+findMRCA'></span>

<h3>Description</h3>

<p>Finds the most recent common ancestor (MRCA) of a set of tips.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findMRCA(tree, tips=NULL, type=c("node","height"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findMRCA_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree as an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="findMRCA_+3A_tips">tips</code></td>
<td>
<p>a vector containing a set of tip labels.</p>
</td></tr>
<tr><td><code id="findMRCA_+3A_type">type</code></td>
<td>
<p>either <code>"node"</code> to return the node of the MRCA; or <code>"height"</code> to return the height above the root of the MRCA of <code>tips</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns node number of the most recent common ancestor of a set of taxa.
</p>
<p>If <code>tips==NULL</code> and <code>type="node"</code> (the default) it will return the result of a normal function call to <code><a href="ape.html#topic+mrca">mrca</a></code>. 
</p>
<p>If <code>tips=NULL</code> and <code>type="height"</code> it will return a matrix equal to that produced by <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>.
</p>
<p>From <span class="pkg">phytools</span> 0.5-66 forward, <code>findMRCA</code> uses <code><a href="ape.html#topic+getMRCA">getMRCA</a></code> in the <span class="pkg">ape</span> package internally, which results in a big speed-up. Even though the two functions are thus totally redundant I have left <code>findMRCA</code> in the package namespace to ensure backward compatibility.
</p>


<h3>Value</h3>

<p>The node number of the MRCA, or a matrix of node numbers (if <code>tips==NULL</code>) - for <code>type="node"</code>; or the height of the MRCA, or a matrix of heights (if <code>tips==NULL</code>) - for <code>type="height"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+findMRCA">findMRCA</a></code>, <code><a href="ape.html#topic+mrca">mrca</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anoletree)
anc&lt;-findMRCA(anoletree,c("cristatellus","cooki",
  "gundlachi"))
plotTree(anoletree,type="fan",fsize=0.7,lwd=1)
nodelabels(node=anc,frame="circle",pch=21,cex=1.5,
  bg="blue")
legend("topleft","most recent common ancestor\nof Puerto Rican TG anoles",
  pch=21,pt.cex=1.5,pt.bg="blue",cex=0.7,bty="n")
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margin to default
</code></pre>

<hr>
<h2 id='fit.bd'>Fits birth-death (speciation/extinction) model to reconstructed phylogeny</h2><span id='topic+fit.bd'></span><span id='topic+fit.yule'></span><span id='topic+lik.bd'></span><span id='topic+print.fit.bd'></span>

<h3>Description</h3>

<p>Fits a birth-death (<code>fit.bd</code>) or pure-birth (<code>fit.yule</code>) model to a reconstructed phylogenetic tree with branch lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.bd(tree, b=NULL, d=NULL, rho=1, ...)
fit.yule(tree, b=NULL, d=NULL, rho=1, ...)
lik.bd(theta, t, rho=1, N=NULL)
## S3 method for class 'fit.bd'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.bd_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_b">b</code></td>
<td>
<p>birth (speciation) rate. Presently doesn't do anything as the rate cannot be fixed.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_d">d</code></td>
<td>
<p>death (extinction) rate. Presently doesn't do anything as the rate cannot be fixed.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_rho">rho</code></td>
<td>
<p>sampling fraction.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_theta">theta</code></td>
<td>
<p>vector of <code>b</code> and <code>d</code> for likelihood function.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_t">t</code></td>
<td>
<p>branching times for calculation of the likelihood.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_n">N</code></td>
<td>
<p>number of tips in the tree.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_x">x</code></td>
<td>
<p>object of class <code>"fit.bd"</code> for <code>print</code> method.</p>
</td></tr>
<tr><td><code id="fit.bd_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>fit.bd</code> fits a birth-death model to a phylogenetic tree with edge lengths and a (potentially) incomplete sampling fraction.
</p>
<p>The function <code>fit.yule</code> fits a pure-birth model with a (potentially) incomplete sampling fraction.
</p>
<p>The function <code>lik.bd</code> computes the likelihood of a set of birth &amp; death rates given the set of branching times computed for a tree and a sampling fraction.
</p>


<h3>Value</h3>

<p><code>fit.bd</code> returns an object of class <code>"fit.bd"</code> which can be printed. This object is a list containing the fitted model parameters, likelihood, optimization conditions, a summary of the optimization, and a likelihood function.
</p>
<p><code>fit.yule</code> returns an object of class <code>"fit.yule"</code>. This object is a list containing the fitted model parameter, likelihood, optimization conditions, a summary of the optimization, and a likelihood function.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Nee, S., May, R. M. and Harvey, P. H. (1994) The reconstructed evolutionary process. <em>Philosophical Transactions of the Royal Society of London B</em>, <b>344</b>, 305-311.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Stadler, T. (2012) How can we improve the accuracy of macroevolutionary rate estimates? <em>Systematic Biology</em>, <b>62</b>, 321-329.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+birthdeath">birthdeath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(salamanders)
## compute sampling fraction based on 55 species of Plethodon
sampling.f&lt;-Ntip(salamanders)/55
## fit birth-death model
bd.fit&lt;-fit.bd(salamanders,rho=sampling.f)
print(bd.fit)
## fit Yule model
yule.fit&lt;-fit.yule(salamanders,rho=sampling.f)
print(yule.fit)
## compare b-d and yule models
anova(yule.fit,bd.fit)
## create a likelihood surface for b-d model
ngrid&lt;-100
b&lt;-seq(0.01,0.06,length.out=ngrid)
d&lt;-seq(0.005,0.03,length.out=ngrid)
logL&lt;-sapply(d,function(d,b) sapply(b,function(b,d) 
    bd.fit$lik(c(b,d)),d=d),b=b)
contour(x=b,y=d,logL,nlevels=100,
    xlab=expression(lambda),
    ylab=expression(mu),bty="l")
title(main="Likelihood surface for plethodontid diversification",
    font.main=3)
points(bd.fit$b,bd.fit$d,cex=1.5,pch=4,
    col="blue",lwd=2)
legend("bottomright","ML solution",pch=4,col="blue",
    bg="white",pt.cex=1.5,pt.lwd=2)
</code></pre>

<hr>
<h2 id='fitBayes'>Evolutionary model fitting with intraspecific variability using Bayesian MCMC</h2><span id='topic+fitBayes'></span>

<h3>Description</h3>

<p><code>fitBayes</code> uses Bayesian MCMC to sample terminal states (species means) as well as the parameters of an evolutionary model from their joint posterior distribution, following Revell &amp; Reynolds (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitBayes(tree, x, ngen=10000, model="BM", method="reduced", control=list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitBayes_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fitBayes_+3A_x">x</code></td>
<td>
<p>a vector of phenotypic values for individuals; <code>names(x)</code> should contain the species names (not individual IDs).</p>
</td></tr>
<tr><td><code id="fitBayes_+3A_ngen">ngen</code></td>
<td>
<p>a integer indicating the number of generations for the MCMC.</p>
</td></tr>
<tr><td><code id="fitBayes_+3A_model">model</code></td>
<td>
<p>an evolutionary model: either <code>"BM"</code> or <code>"lambda"</code>.</p>
</td></tr>
<tr><td><code id="fitBayes_+3A_method">method</code></td>
<td>
<p>a method: either <code>"reduced"</code> or <code>"full"</code>.</p>
</td></tr>
<tr><td><code id="fitBayes_+3A_control">control</code></td>
<td>
<p>a list of control parameters containing the following elements: <code>sig2</code>: starting value for <code class="reqn">\sigma^2</code> (BM rate); <code>lambda</code>: starting value for the <code class="reqn">\lambda</code> parameter; <code>a</code>: starting for the state at the root node; <code>xbar</code>: starting values for the states at the tips; <code>intV</code>: starting value for the intraspecific variance (reduced method); or <code>v</code>: starting value for the vector of intraspecific variances for all species (full method); <code>pr.mean</code>: means for the prior distributions in the following order - <code>sig2</code>, <code>lambda</code> (if applicable), <code>a</code>, <code>xbar</code>, <code>intV</code> or <code>v</code> (if applicable), note that the prior probability distribution is exponential for <code>sig2</code> and normal for <code>a</code> and <code>y</code>; <code>pr.var</code>: variances on the prior distributions, same order as <code>pr.mean</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"fitBayes"</code> that includes a matrix (<code>mcmc</code>) with a number of rows <code>ngen/control$sample+1</code> containing the posterior sample and likelihoods.
</p>
<p>Matrix columns are labeled by species (for species means and variances), or by the corresponding evolutionary parameter.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J. and R. G. Reynolds. (2012) A new Bayesian method for fitting evolutionary models to comparative data with intraspecific variation. <em>Evolution</em>, 66, 2697-2707.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>
</p>

<hr>
<h2 id='fitDiversityModel'>Fit diversity-dependent phenotypic evolution model</h2><span id='topic+fitDiversityModel'></span><span id='topic+logLik.fitDiversityModel'></span><span id='topic+print.fitDiversityModel'></span>

<h3>Description</h3>

<p>Fits a diversity-dependent phenotypic evolution model (similar to Mahler et al. 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitDiversityModel(tree, x, d=NULL, showTree=TRUE, tol=1e-6)
## S3 method for class 'fitDiversityModel'
logLik(object, ...)
## S3 method for class 'fitDiversityModel'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitDiversityModel_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fitDiversityModel_+3A_x">x</code></td>
<td>
<p>a vector with tip values for a continuously distributed trait. For <code>print</code> method, an object of class <code>"fitDiversityModel"</code>.</p>
</td></tr>
<tr><td><code id="fitDiversityModel_+3A_d">d</code></td>
<td>
<p>a vector containing the inferred historical diversity at each node in the tree - if <code>d=NULL</code> (the default) function will treat the diversification as if it occurred in a single geographic area.</p>
</td></tr>
<tr><td><code id="fitDiversityModel_+3A_showtree">showTree</code></td>
<td>
<p>optional logical value indicating whether to plot the tree transformation implied by the model.</p>
</td></tr>
<tr><td><code id="fitDiversityModel_+3A_tol">tol</code></td>
<td>
<p>some small value by which <code>d</code> is incremented during rescaling of <code>psi</code> for optimization. If R thinks your matrices are singular during optimization, try increasing <code>tol</code> slightly.</p>
</td></tr>
<tr><td><code id="fitDiversityModel_+3A_object">object</code></td>
<td>
<p>for <code>logLik</code> method, an object of class <code>"fitDiversityModel"</code>.</p>
</td></tr>
<tr><td><code id="fitDiversityModel_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>logLik</code> and <code>print</code> methods. Note that for the <code>logLik</code> method the number of fitted parameters (<code>"df"</code>) is assumed to be 3 for the diversity dependent model (that is, if <code>psi</code> is estimated) and 2 for the diversity independent model, unless otherwise specified (using the argument <code>df</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"fitDiversityModel"</code> consisting of the following elements:
</p>
<table>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood of the fitted model.</p>
</td></tr>
<tr><td><code>sig0</code></td>
<td>
<p>estimated starting value for the rate at the root of the tree, <code class="reqn">\sigma_0^2</code>.</p>
</td></tr>
<tr><td><code>psi</code></td>
<td>
<p>the estimated rate of change in the rate associated with the addition of a lineage.</p>
</td></tr>
<tr><td><code>vcv</code></td>
<td>
<p>a matrix with the variances and covariance of the estimated parameters (from the Hessian).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Mahler, D. L, L. J. Revell, R. E. Glor, and J. B. Losos (2010) Ecological opportunity and the rate of morphological evolution in the diversification of Greater Antillean anoles. <em>Evolution</em>, <b>64</b>, 2731-2745.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+estDiversity">estDiversity</a></code>, <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>
</p>

<hr>
<h2 id='fitMk'>Fits extended M<em>k</em> model for discrete character evolution</h2><span id='topic+fitMk'></span><span id='topic+plot.fitMk'></span><span id='topic+plot.gfit'></span><span id='topic+fitmultiMk'></span><span id='topic+fitpolyMk'></span><span id='topic+graph.polyMk'></span><span id='topic+plot.fitpolyMk'></span><span id='topic+mcmcMk'></span><span id='topic+plot.mcmcMk'></span><span id='topic+density.mcmcMk'></span><span id='topic+plot.density.mcmcMk'></span><span id='topic+fitHRM'></span><span id='topic+plot.fitHRM'></span><span id='topic+fitMk.parallel'></span><span id='topic+fitgammaMk'></span><span id='topic+fitfnMk'></span>

<h3>Description</h3>

<p>The functions <code>fitMk</code>, <code>fitmultiMk</code>, <code>fitpolyMk</code>, <code>fitHRM</code>, <code>fitMk.parallel</code>, <code>fitgammaMk</code>, <code>fitfnMk</code>, and <code>mcmcMk</code> fit various flavors of the extended M<em>k</em> model (Lewis, 2001) for discrete character evolution on a reconstructed phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitMk(tree, x, model="SYM", fixedQ=NULL, ...)
## S3 method for class 'fitMk'
plot(x, ...)
## S3 method for class 'gfit'
plot(x, ...)
fitmultiMk(tree, x, model="ER", ...)
fitpolyMk(tree, x, model="SYM", ordered=FALSE, ...)
graph.polyMk(k=2, model="SYM", ordered=FALSE, ...)
## S3 method for class 'fitpolyMk'
plot(x, ...)
mcmcMk(tree, x, model="ER", ngen=10000, ...)
## S3 method for class 'mcmcMk'
plot(x, ...)
## S3 method for class 'mcmcMk'
density(x, ...)
## S3 method for class 'density.mcmcMk'
plot(x, ...)
fitHRM(tree, x, model="ARD", ncat=2, ...)
## S3 method for class 'fitHRM'
plot(x, ...)
fitMk.parallel(tree, x, model="SYM", ncores=1, ...)
fitgammaMk(tree, x, model="ER", fixedQ=NULL, nrates=8, ...)
fitfnMk(tree, x, model="polynomial", degree=2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitMk_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>. In the case of <code>fitmultiMk</code> an object of class <code>"simmap"</code> with a mapped discrete character.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_x">x</code></td>
<td>
<p>a vector (or numeric matrix) of tip values for species; <code>names(x)</code> should be the species names. In the case of <code>plot</code> and <code>density</code> methods, an object of the appropriate class.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_model">model</code></td>
<td>
<p>model. See <code>make.simmap</code> or <code>ace</code> for details. For <code>fitfnMk</code> the only option is presently <code>model="polynomial"</code>.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_fixedq">fixedQ</code></td>
<td>
<p>fixed value of transition matrix <code>Q</code>, if one is desired.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_ordered">ordered</code></td>
<td>
<p>for <code>fitpolyMk</code>, a logical value indicating whether or not the character should be treated as ordered. For now the function assumes alphanumerical order (i.e., numbers sorted by their initial and then successive digits followed by characters or character strings in alphabetical order).</p>
</td></tr>
<tr><td><code id="fitMk_+3A_k">k</code></td>
<td>
<p>For <code>graph.polyMk</code>, the number of monomorphic states for the discrete trait.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_ngen">ngen</code></td>
<td>
<p>number of generations of MCMC for <code>mcmcMk</code>.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_ncat">ncat</code></td>
<td>
<p>number of rate categories (per level of the discrete trait) in the hidden-rate model.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_ncores">ncores</code></td>
<td>
<p>number of cores for <code>fitMk.parallel</code>.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_nrates">nrates</code></td>
<td>
<p>number of rate categories for discretized <code class="reqn">\Gamma</code> distribution.</p>
</td></tr>
<tr><td><code id="fitMk_+3A_degree">degree</code></td>
<td>
<p>the degree of the polynomial for <code>fitfnMk</code>. (Defaults to <code>degree=2</code>.)</p>
</td></tr>
<tr><td><code id="fitMk_+3A_...">...</code></td>
<td>
<p>optional arguments, including <code>pi</code>, the prior distribution at the root node (defaults to <code>pi="equal"</code>). Other options for <code>pi</code> include <code>pi="fitzjohn"</code> (which implements the prior distribution of FitzJohn et al. 2009), <code>pi="estimated"</code> (which finds the stationary distribution of state frequencies and sets that as the prior), or an arbitrary prior distribution specified by the user. For <code>plot</code> method optional arguments include (but may not be limited to): <code>signif</code>, the number of digits for the rates to be plotted; <code>main</code>, a character vector of length two with the headings for each subplot; <code>cex.main</code>, <code>cex.traits</code>, and <code>cex.rates</code>, font sizes for the various text elements of the plot; and <code>show.zeros</code>, a logical argument specifying whether or not to plot arrows with the ML estimated transition rate is not different from zero (with tolerance specified by the optional argument <code>tol</code>). Finally, for <code>fitpolyMk</code>, both <code>order</code> (an evolutionary sequence for the monomorphic condition) and <code>max.poly</code> can be set for the <code>ordered=TRUE</code> model. If not set, <code>order</code> defaults to alphanumeric order, and <code>max.poly</code> defaults to the highest level of polymorphism observed in the data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>fitMk</code> fits a so-called extended M<em>k</em> model for discrete character evolution (Lewis, 2001).
</p>
<p><code>plot.fitMk</code> plots an object of class <code>"fitMk"</code> returned by <code>fitMk</code>. <code>plot.gfit</code> plots an object of class <code>"gfit"</code> from <span class="pkg">geiger</span>'s <code>fitDiscrete</code> function. Both plots portray the fitted model using a graph of arrows connecting states.
</p>
<p>The function <code>fitmultiMk</code> fits an M<em>k</em> model in which the transition rates between character states are allowed to vary depending on the mapped state of a discrete character on the tree. It can be combined with, for example, <code><a href="#topic+paintSubTree">paintSubTree</a></code> to test hypotheses about how the process of discrete character evolution for <code>x</code> varies between different parts of the tree.
</p>
<p>The function <code>fitgammaMk</code> fits an M<em>k</em> model in which the edge rates are assumed to have been sampled randomly from a <code class="reqn">\Gamma</code> distribution with mean of 1.0 and shape parameter <code class="reqn">\alpha</code>.
</p>
<p>The function <code>fitfnMk</code> fit an ordered M<em>k</em> model in which the backward and forward transition rates between adjacent levels of the trait vary according to a functional form. Presently that function form is an <em>n</em>th degree polynomial, in which <code>degree</code> is set by the user (but defaults to <code>degree = 2</code>).
</p>
<p>The function <code>fitpolyMk</code> fits an M<em>k</em> model to data for a discrete character with intraspecific polymorphism. Polymorphic species should be coded with the name of the two or more states recorded for the species separated by a plus sign <code>+</code> (e.g., <code>A+B</code> would indicate that both states <code>A</code> and <code>B</code> are found in the corresponding taxon). Invariably it's assumed that transitions between states must occur through a polymorphic condition, whereas transitions <em>cannot</em> occur directly between two incompatible polymorphic conditions. For instance, a transition between <code>A+B</code> and <code>B+C</code> would have to occur through the monomorphic state <code>B</code>. At time of writing, this function permits the models <code>"ER"</code> (equal rates for all permitted transitions), <code>"SYM"</code> (symmetric backward &amp; forward rates for all permitted transitions), <code>"ARD"</code> (all-rates-different for permitted transitions), and a new model called <code>"transient"</code> in which the acquisition of polymorphism (e.g., <code>A -&gt; A+B</code>) is assumed to occur at a different rate than its loss (e.g., <code>A+B -&gt; B</code>). The method <code>plot.fitpolyMk</code> plots the fitted M<em>k</em> model with intraspecific polymorphism.
</p>
<p>The function <code>mcmcMk</code> runs a Bayesian MCMC version of <code>fitMk</code>. The shape of the prior distribution of the transition rates is <code class="reqn">\Gamma</code>, with <code class="reqn">\alpha</code> and <code class="reqn">\beta</code> via the argument <code>prior</code>, which takes the form of a list. The default value of <code class="reqn">\alpha</code> is 0.1, and <code class="reqn">\beta</code> defaults to a value such that <code class="reqn">\alpha/\beta</code> is equal to the parsimony score for <code>x</code> divided by the sum of the edge lengths of the tree. The shape of the proposal distribution is normal, with mean zero and a variance that can be controlled by the user via the optional argument <code>prior.var</code>. The argument <code>auto.tune</code>, if <code>TRUE</code> or <code>FALSE</code>, indicates whether or not to 'tune' the proposal variance up or down to target a particular acceptance rate (defaults to 0.5). <code>auto.tune</code> can also be a numeric value between 0 and 1, in which case this value will be the target acceptance ratio. The argument <code>plot</code> indicates whether the progress of the MCMC should be plotted (defaults to <code>TRUE</code>, but runs much faster when set to <code>FALSE</code>).
</p>
<p>The method <code>plot.mcmcMk</code> plots a log-likelihood trace and a trace of the rate parameters from the MCMC. (This the same graph that is created by setting <code>plot=TRUE</code> in <code>mcmcMk</code>.) The method <code>density.mcmcMk</code> computes a posterior density on the transition rates in the model from the posterior sample obtained in the MCMC, will import the package <span class="pkg">coda</span> if it is available, and returns an object of class <code>"density.mcmcMk"</code>. Finally, the method <code>plot.density.mcmcMk</code> creates a plot of the posterior density (or a set of plots) for the transition rates between states.
</p>
<p>Finally, the function <code>fitHRM</code> fits a hidden-rate M<em>k</em> model following Beaulieu et al. (2013). For the hidden-rate model we need to specify a number of rate categories for each level of the trait - and this can be a vector of different values for each trait. We can also choose a model (<code>"ER"</code>, <code>"SYM"</code>, or <code>"ARD"</code>), as well as whether or not to treat the character as a 'threshold' trait (<code>umbral=TRUE</code>, defaults to <code>FALSE</code>). This latter model is basically one that allows absorbing conditions for some hidden states. Since this can be a difficult optimization problem, the optional argument <code>niter</code> sets the number of optimization iterations to be run. <code>niter</code> defaults to <code>niter=10</code>. To fit the same default hidden-rates model as is implemented in <span class="pkg">corHMM</span>, one should set <code>corHMM_model=TRUE</code> and <code>ordered_hrm=FALSE</code>.
</p>
<p>Note that (by default) both <code>fitMk</code> and <code>fitmultiMk</code> recycle code from <code><a href="ape.html#topic+ace">ace</a></code> in the <span class="pkg">ape</span> package for computing the likelihood. (If the optional argument <code>pruning=TRUE</code> then alternative, slightly faster, <span class="pkg">phytools</span> code for the pruning algorithm is used.) <code>fitpolyMk</code>, <code>mcmcMk</code>, and <code>fitHRM</code> use <code>fitMk</code> internally to compute the likelihood.
</p>


<h3>Value</h3>

<p>An object of class <code>"fitMk"</code>, <code>"fitmultiMk"</code>, <code>"fitpolyMk"</code>, <code>"mcmcMk"</code>, <code>"fitHRM"</code>, <code>"fitgammaMk"</code>, or <code>"fitfnMk"</code>. In the case of <code>density.mcmcMk</code> an object of class <code>"density.mcmcMk"</code>.
</p>
<p><code>plot.fitMk</code>, <code>plot.gfit</code>, and <code>plot.HRM</code> invisibly return the coordinates of vertices of the plotted <b>Q</b>-matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Beaulieu, J. M., B. C. O'Meara, and M. J. Donoghue (2013) Identifying hidden rate changes in the evolution of a binary morphological character: The evolution of plant habit in campanulid angiosperms. <em>Systematic Biology</em>, <b>62</b>, 725-737.
</p>
<p>FitzJohn, R. G., W. P. Maddison, and S. P. Otto (2009) Estimating trait-dependent speciation and extinction rates from incompletely resolved phylogenies. <em>Systematic Biology</em>, <b>58</b>, 595-611.
</p>
<p>Lewis, P. O. (2001) A likelihood approach to estimating phylogeny from discrete morphological character data. <em>Systematic Biology</em>, <b>50</b>, 913-925.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J. and L. J. Harmon (2022) <em>Phylogenetic Comparative Methods in R</em>. Princeton University Press.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load tree and data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract discrete character (feeding mode)
fmode&lt;-setNames(sunfish.data$feeding.mode,
    rownames(sunfish.data))
## fit "ER" model
fit.ER&lt;-fitMk(sunfish.tree,fmode,model="ER")
print(fit.ER)
## fit "ARD" model
fit.ARD&lt;-fitMk(sunfish.tree,fmode,model="ARD")
print(fit.ARD)
## compare the models
AIC(fit.ER,fit.ARD)

## load tree and data from Benitez-Alvarez et al. (2000)
data(flatworm.data)
data(flatworm.tree)
## extract discrete character (habitat)
habitat&lt;-setNames(flatworm.data$Habitat,
    rownames(flatworm.data))
## fit polymorphic models "ER" and "transient"
fitpoly.ER&lt;-fitpolyMk(flatworm.tree,habitat,
    model="ER")
fitpoly.transient&lt;-fitpolyMk(flatworm.tree,habitat,
    model="transient")
## print fitted models
print(fitpoly.ER)
print(fitpoly.transient)
## compare model
AIC(fitpoly.ER,fitpoly.transient)
## plot models
par(mfrow=c(2,1))
plot(fitpoly.ER)
mtext("a) ER polymorphic model",adj=0,line=1)
plot(fitpoly.transient)
mtext("b) Transient polymorphic model",adj=0,
     line=1)
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='fitmultiBM'>Fits bounded and wrapped Brownian motion models</h2><span id='topic+fitmultiBM'></span>

<h3>Description</h3>

<p>Fits a multi-rate discrete-state-dependent Brownian motion model using the discrete approximation of Boucher &amp; Demery (2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	fitmultiBM(tree, x, y=NULL, model="ER", ncat=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitmultiBM_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fitmultiBM_+3A_x">x</code></td>
<td>
<p>continuous character data in a vector.</p>
</td></tr>
<tr><td><code id="fitmultiBM_+3A_y">y</code></td>
<td>
<p>optional discrete character, normally in the form of a factor or character vector.</p>
</td></tr>
<tr><td><code id="fitmultiBM_+3A_model">model</code></td>
<td>
<p>discrete character evolution model. (E.g., <code>"ER"</code>, <code>"SYM"</code>, <code>"ARD"</code>.)</p>
</td></tr>
<tr><td><code id="fitmultiBM_+3A_ncat">ncat</code></td>
<td>
<p>number of rate categories (if hidden rates).</p>
</td></tr>
<tr><td><code id="fitmultiBM_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Optional arguments.
</p>


<h3>Value</h3>

<p>An object of class <code>"fitmultiBM"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Boucher, F. C., and V. Demery (2016) Inferring bounded evolution in phenotypic characters from phylogenetic comparative data. <em>Systematic Biology</em>, <b>65</b>, 651-661.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bounded_bm">bounded_bm</a></code>, <code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+fitMk">fitMk</a></code>
</p>

<hr>
<h2 id='fitPagel'>Function to test for correlated evolution of binary traits</h2><span id='topic+fitPagel'></span><span id='topic+plot.fitPagel'></span>

<h3>Description</h3>

<p>Fits Pagel's (1994) model for the correlated evolution of two binary characters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitPagel(tree, x, y, method="fitMk", model="ARD", dep.var="xy", ...)
## S3 method for class 'fitPagel'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitPagel_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fitPagel_+3A_x">x</code></td>
<td>
<p>a vector of phenotypic values for a binary trait for the species in <code>tree</code>; or a matrix in which the rows of <code>x</code> give the probability of being in each column state. (The latter option is only supported for <code>method="fitMk"</code>.) For S3 <code>plot</code> method, an object of class <code>"fitPagel"</code>.</p>
</td></tr>
<tr><td><code id="fitPagel_+3A_y">y</code></td>
<td>
<p>a second binary character for the species in <code>tree</code>; or a matrix in which the rows give the probability of being in each column state.</p>
</td></tr>
<tr><td><code id="fitPagel_+3A_method">method</code></td>
<td>
<p>function to use for optimization (defaults to <code>method="fitMk"</code>). Other options are <code>"ace"</code> to use the <code><a href="ape.html#topic+ace">ace</a></code> function in <span class="pkg">ape</span> for optimization, or to <code>"fitDiscrete"</code> (if the <span class="pkg">geiger</span> package is installed) to use <span class="pkg">geiger</span>'s <code>fitDiscrete</code> for optimization.</p>
</td></tr>
<tr><td><code id="fitPagel_+3A_model">model</code></td>
<td>
<p>model of evolution for the individual characters. Can be <code>model="ER"</code>, <code>"SYM"</code> (equivalent to <code>"ER"</code> in this case),
and <code>"ARD"</code>.</p>
</td></tr>
<tr><td><code id="fitPagel_+3A_dep.var">dep.var</code></td>
<td>
<p>dependent variable. If <code>dep.var="xy"</code> than the rate of substitution in <code>x</code> depends on <code>y</code> &amp; vice versa. If <code>dep.var="x"</code> than the substitution rate in <code>x</code> depends on <code>y</code>, but not the converse. Finally, if <code>dep.var="y"</code> than the rate of substitution in <code>y</code> depends on <code>x</code>, but not the converse.</p>
</td></tr>
<tr><td><code id="fitPagel_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="#topic+fitMk">fitMk</a></code>, <code><a href="ape.html#topic+ace">ace</a></code>, or <code>fitDiscrete</code>. For <code>plot</code> method optional arguments include (but may not be limited to): <code>signif</code>, the number of digits for the rates to be plotted; <code>main</code>, a character vector of length two with the headings for each subplot; <code>cex.main</code>, <code>cex.sub</code>, <code>cex.traits</code>, and <code>cex.rates</code>, font sizes for the various text elements of the plot; and <code>lwd.by.rate</code>, a logical argument specifying whether or not to scale arrow line widths in proportion to the estimated rates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>fitPagel</code> fits both an independent evolution model, as well as Pagel's (1994) binary dependent model, and compares them with a likelihood-ratio test.
</p>
<p><code>plot.fitPagel</code> plots the fitted models using arrows.
</p>


<h3>Value</h3>

<p>An object of class <code>"fitPagel"</code> which contains the optimized matrices under an independence &amp; a dependence model, log-likelihoods, a likelihood ratio, and a P-value for the independence model based on a chi-squared test.
</p>
<p><code>plot.fitPagel</code> creates a plot showing the different fitted models with arrows.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Pagel, M. (1994) Detecting correlated evolution on phylogenies: A general method for the comparative analysis of discrete characters. <em>Proceedings of the Royal Society B</em>, <b>255</b>, 37-45.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+fitMk">fitMk</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## load data
data(bonyfish.tree)
data(bonyfish.data)
## extract discrete characters
spawning_mode&lt;-setNames(bonyfish.data$spawning_mode,
    rownames(bonyfish.data))
paternal_care&lt;-setNames(bonyfish.data$paternal_care,
    rownames(bonyfish.data))
## fit correlational model
bonyfish.pagel&lt;-fitPagel(bonyfish.tree,paternal_care,
    spawning_mode)
## test for a difference between models
anova(bonyfish.pagel)
## plot fitted models
plot(bonyfish.pagel,lwd.by.rate=TRUE)
## reset par
par(mar=c(5.1,4.1,4.1,2.1),
    mfrow=c(1,1))
## End(Not run)
</code></pre>

<hr>
<h2 id='fitThresh'>Fits multi-state threshold model using the discrete approximation</h2><span id='topic+fitThresh'></span>

<h3>Description</h3>

<p>This function uses the discrete approximation of Boucher and Demery (2016) to fit a multi-state discrete character threshold model following Felsenstein (2012; Revell 2014).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitThresh(tree, x, sequence=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitThresh_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="fitThresh_+3A_x">x</code></td>
<td>
<p>a named vector.</p>
</td></tr>
<tr><td><code id="fitThresh_+3A_sequence">sequence</code></td>
<td>
<p>assumed ordering of the discrete character state. If not supplied and then alphanumeric order is assumed.</p>
</td></tr>
<tr><td><code id="fitThresh_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"fitThresh"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Boucher, F. C., and V. Demery (2016) Inferring bounded evolution in phenotypic characters from phylogenetic comparative data. <em>Systematic Biology</em>, <b>65</b>, 651-661.
</p>
<p>Felsenstein, J. (2012) A comparative method for both discrete and continuous characters using the threshold model. <em>American Naturalist</em>, <b>179</b>, 145-156.
</p>
<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ancThresh">ancThresh</a></code>, <code><a href="#topic+bounded_bm">bounded_bm</a></code>, <code><a href="#topic+fitmultiBM">fitmultiBM</a></code>, <code><a href="#topic+threshBayes">threshBayes</a></code>
</p>

<hr>
<h2 id='force.ultrametric'>Coerces a phylogenetic tree to be ultrametric</h2><span id='topic+force.ultrametric'></span>

<h3>Description</h3>

<p>Coerces an object of class <code>"phylo"</code> to be ultrametric. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>force.ultrametric(tree, method=c("nnls","extend"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="force.ultrametric_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="force.ultrametric_+3A_method">method</code></td>
<td>
<p>the method to use to force the tree to be ultrametric. Options are <code>"nnls"</code> (which uses the <span class="pkg">phangorn</span> function <code><a href="phangorn.html#topic+nnls.tree">nnls.tree</a></code> internally), or <code>"extend"</code>.</p>
</td></tr>
<tr><td><code id="force.ultrametric_+3A_...">...</code></td>
<td>
<p>optional arguments: principally, <code>message</code>. This argument (if set to <code>FALSE</code>) can be used to suppress the default warning message that <code>force.ultrametric</code> <em>should not</em> be used as a formal statistical method to ultrametricize a tree.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>force.ultrametric</code> coerces a non-ultrametric tree to be ultrametric.
</p>
<p>This is achieved either by using <code><a href="phangorn.html#topic+nnls.tree">nnls.tree</a></code> from the <span class="pkg">phangorn</span> package to compute the set of edge lengths that result in a minimized sum-of-squares distance between the patristic distance of the output and input trees (<code>method="nnls"</code>); or by simply extending all the external edges of the tree to match the external edge with the greatest total height (<code>method="extend"</code>). 
</p>
<p>Note that neither of these should be treated as formal statistical methods for inferring an ultrametric tree. Rather, this method can be deployed when a genuinely ultrametric tree read from file fails <code><a href="ape.html#topic+is.ultrametric">is.ultrametric</a></code> for reasons of numerical precision. 
</p>


<h3>Value</h3>

<p>An ultrametric tree in an object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+is.ultrametric">is.ultrametric</a></code>, <code><a href="phangorn.html#topic+nnls.tree">nnls.tree</a></code>
</p>

<hr>
<h2 id='gamma_pruning'>Internally used function</h2><span id='topic+gamma_pruning'></span>

<h3>Description</h3>

<p>Internally used function to compute the likelihood under a <code class="reqn">\Gamma</code> model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_pruning(par, nrates=4, tree, x, model=NULL, median=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_pruning_+3A_par">par</code></td>
<td>
<p>function parameters.</p>
</td></tr>
<tr><td><code id="gamma_pruning_+3A_nrates">nrates</code></td>
<td>
<p>number of discrete rate categories.</p>
</td></tr>
<tr><td><code id="gamma_pruning_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="gamma_pruning_+3A_x">x</code></td>
<td>
<p>data in the form of a binary matrix.</p>
</td></tr>
<tr><td><code id="gamma_pruning_+3A_model">model</code></td>
<td>
<p>design matrix of fitted model.</p>
</td></tr>
<tr><td><code id="gamma_pruning_+3A_median">median</code></td>
<td>
<p>use the median of each rate category.</p>
</td></tr>
<tr><td><code id="gamma_pruning_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A value of the log-likelihood.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='gammatest'>Gamma test of Pybus &amp; Harvey (2000)</h2><span id='topic+gammatest'></span>

<h3>Description</h3>

<p>Conducts <code class="reqn">\gamma</code>-test of Pybus &amp; Harvey (2000).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gammatest(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gammatest_+3A_x">x</code></td>
<td>
<p>an object of class <code>"ltt"</code> resulting from a call of the function <code><a href="#topic+ltt">ltt</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A an object of class <code>"gammatest"</code> consisting of a list that contains:
</p>
<table>
<tr><td><code>gamma</code></td>
<td>
<p>a value for the <code class="reqn">\gamma</code>-statistic.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>two-tailed P-value for the <code class="reqn">\gamma</code>-test.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Pybus, O. G., and P. H. Harvey (2000) Testing macro-evolutionary models using incomplete molecular phylogenies. <em>Proc. R. Soc. Lond. B</em>, <b>267</b>, 2267-2272.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ltt">ltt</a></code>, <code><a href="#topic+mccr">mccr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(darter.tree)
gammatest(ltt(darter.tree,plot=FALSE))
</code></pre>

<hr>
<h2 id='genSeq'>Simulate a DNA alignment on the tree under a model</h2><span id='topic+genSeq'></span>

<h3>Description</h3>

<p>Simulates DNA sequence on <code>tree</code> under the specified model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genSeq(tree, l=1000, Q=NULL, rate=1, format="DNAbin", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genSeq_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="genSeq_+3A_l">l</code></td>
<td>
<p>length of desired sequences.</p>
</td></tr>
<tr><td><code id="genSeq_+3A_q">Q</code></td>
<td>
<p>transition matrix for the simulation. Row and column names (<code>c("a","c","g","t")</code>, although not necessarily in that order) should be provided. If <code>NULL</code>, a single rate is assumed.</p>
</td></tr>
<tr><td><code id="genSeq_+3A_rate">rate</code></td>
<td>
<p>multiplier for <code>Q</code>, or a vector for <code class="reqn">\Gamma</code> rate heterogeneity.</p>
</td></tr>
<tr><td><code id="genSeq_+3A_format">format</code></td>
<td>
<p>format of the output object. Can be <code>"DNAbin"</code>, <code>"phyDat"</code>, or <code>"matrix"</code>.</p>
</td></tr>
<tr><td><code id="genSeq_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses <code><a href="#topic+sim.Mk">sim.Mk</a></code> internally.
</p>


<h3>Value</h3>

<p>An object of class <code>"DNAbin"</code> or <code>"phyDat"</code>, or a matrix of nucleotides.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mammal.tree)
mammal.tree$edge.length&lt;-mammal.tree$edge.length/
    max(nodeHeights(mammal.tree))*0.2 ## rescale tree
## simulate gamma rate heterogeneity
gg&lt;-rgamma(n=100,shape=0.25,rate=0.25)
dna&lt;-genSeq(mammal.tree,l=100,rate=gg)
</code></pre>

<hr>
<h2 id='geo.legend'>Adds a geological (or other temporal) legend to a plotted tree</h2><span id='topic+geo.legend'></span><span id='topic+geo.palette'></span>

<h3>Description</h3>

<p>Adds a geological legend to a plotted tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geo.legend(leg=NULL, colors=NULL, alpha=0.2, ...)
geo.palette()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geo.legend_+3A_leg">leg</code></td>
<td>
<p>a matrix with the starting &amp; ending point of each plotted era in rows, &amp; names of the time periods as rownames.</p>
</td></tr>
<tr><td><code id="geo.legend_+3A_colors">colors</code></td>
<td>
<p>a vector of colors for the time periods of the rows in <code>leg</code>.</p>
</td></tr>
<tr><td><code id="geo.legend_+3A_alpha">alpha</code></td>
<td>
<p>transparency level to apply to <code>colors</code>.</p>
</td></tr>
<tr><td><code id="geo.legend_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>geo.legend</code> adds a geological (or other temporal) legend to a plotted tree.
</p>
<p>The function <code>geo.palette</code> returns a geological time color palette to the user.
</p>


<h3>Value</h3>

<p><code>geo.legend</code> adds a visual element to a plotted tree and invisible returns an object of class <code>geo.legend</code> containing the time periods and colors of the painted legend.
</p>
<p><code>geo.palette</code> simply returns a geological timescale color palette as an object of class <code>"geo.palette"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anoletree)
## rescale tree to 50 ma total depth
anoletree&lt;-rescale(anoletree,model="depth",depth=50)
## plot phylogeny
plotTree(anoletree,ylim=c(-0.16,1)*Ntip(anoletree),
  ftype="i",lwd=1,fsize=0.5)
## add geological color legend
obj&lt;-geo.legend()
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='get.treepos'>Get position or node of a plotted tree interactively</h2><span id='topic+get.treepos'></span><span id='topic+getnode'></span>

<h3>Description</h3>

<p>Return the phylogenetic position of a mouse click on a plotted tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.treepos(message=TRUE, ...)
getnode(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.treepos_+3A_message">message</code></td>
<td>
<p>for <code>get.treepos</code>, a logical value indicating whether or not to print an instructional message.</p>
</td></tr>
<tr><td><code id="get.treepos_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code>get.treepos</code> and <code>getnode</code> are primarily meant for internal use in other <span class="pkg">phytools</span> functions.
</p>
<p><code>get.treepos</code> returns the index of the node at the end of the selected edge, along with the branch distance to that node.
</p>
<p><code>getnode</code> simply returns the closest node to the user mouse click.
</p>


<h3>Value</h3>

<p>A list for <code>get.treepos</code> and a node number for <code>getnode</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='getCladesofSize'>Get all subtrees larger than or equal to a specified size</h2><span id='topic+getCladesofSize'></span>

<h3>Description</h3>

<p>This function gets all subtrees that cannot be further subdivided into two reciprocally monophyletic subtrees of size <code>&gt;= clade.size</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCladesofSize(tree, clade.size=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCladesofSize_+3A_tree">tree</code></td>
<td>
<p>is an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="getCladesofSize_+3A_clade.size">clade.size</code></td>
<td>
<p>subtree size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+extract.clade">extract.clade</a></code>, <code><a href="#topic+getDescendants">getDescendants</a></code>
</p>

<hr>
<h2 id='getDescendants'>Get descendant node numbers</h2><span id='topic+getDescendants'></span><span id='topic+getParent'></span>

<h3>Description</h3>

<p>Returns the descendants or parent of a specified node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDescendants(tree, node, curr=NULL)
getParent(tree, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDescendants_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree as an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="getDescendants_+3A_node">node</code></td>
<td>
<p>an integer specifying a node number in the tree.</p>
</td></tr>
<tr><td><code id="getDescendants_+3A_curr">curr</code></td>
<td>
<p>the set of previously stored node numbers - used in recursive function calls.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>getDescendants</code> returns the set of node &amp; tip numbers descended from <code>node</code>.
</p>
<p><code>getParent</code> returns the <em>single</em> parent node of a specified node number (or <code>NULL</code> if <code>node</code> is already the root).
</p>


<h3>Value</h3>

<p>The set of node and tip numbers for the nodes and tips descended from <code>node</code> in a vector, or for <code>getParent</code> the single node preceding <code>node</code> in the tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="phangorn.html#topic+Descendants">Descendants</a></code>, <code><a href="#topic+paintSubTree">paintSubTree</a></code>
</p>

<hr>
<h2 id='getExtant'>Returns a list of the extant or extinct lineages in a tree containing non-contemporaneous tips</h2><span id='topic+getExtant'></span><span id='topic+getExtinct'></span>

<h3>Description</h3>

<p>Computes the set of extant or extinct tips from a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExtant(tree, tol=1e-8)
getExtinct(tree, tol=1e-8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExtant_+3A_tree">tree</code></td>
<td>
<p>a phylogeny stored as an object of class <code>"phylo"</code> with some tips that are non-contemporaneous (i.e., end before the present).</p>
</td></tr>
<tr><td><code id="getExtant_+3A_tol">tol</code></td>
<td>
<p>a tolerance value to account for numerical imprecision.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>getExtant</code> takes a tree as input and returns a vector containing the names of all the tips that have a height above the root that is equal (to a degree of numerical precision determined by <code>tol</code>) to the height of the highest tip. These tips are presumed to be &quot;extant.&quot;
</p>
<p><code>getExtinct</code> returns the complement.
</p>


<h3>Value</h3>

<p>A vector with the tip names of extant or extinct species in the tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nodeHeights">nodeHeights</a></code>
</p>

<hr>
<h2 id='getSisters'>Get the sister node number, label, or set of nodes for a node or tip</h2><span id='topic+getSisters'></span>

<h3>Description</h3>

<p>Computes the sister taxon or node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSisters(tree, node, mode=c("number","label"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSisters_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="getSisters_+3A_node">node</code></td>
<td>
<p>a node number, tip number, node label, or tip label.</p>
</td></tr>
<tr><td><code id="getSisters_+3A_mode">mode</code></td>
<td>
<p>an optional string indicating whether to return the node or tip number(s) or the node or tip label(s), if applicable.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a tree and node or tip number of label and returns the number or label of the sister or sisters to that node or tip.
</p>


<h3>Value</h3>

<p>If <code>mode="number"</code> this function returns an integer or vector containing the node number of numbers of the sister node or tip. If <code>mode="label"</code> then this function returns a list containing up to two vectors: one for the node numbers of labels of sister nodes (if applicable); and the other containing the tip labels of the sister tips.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getDescendants">getDescendants</a></code>, <code><a href="phangorn.html#topic+Siblings">Siblings</a></code>
</p>

<hr>
<h2 id='getStates'>Get the states at nodes or tips from a mapped tree</h2><span id='topic+getStates'></span>

<h3>Description</h3>

<p>Gets the states from the nodes or tips of a mapped tree (e.g., <code><a href="#topic+make.simmap">make.simmap</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getStates(tree, type=c("nodes","tips","both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStates_+3A_tree">tree</code></td>
<td>
<p>is a modified object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="getStates_+3A_type">type</code></td>
<td>
<p>mode indicating whether to get states at the nodes (<code>type="nodes"</code>) or the tips (<code>type="tips"</code>) of the tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named vector (for <code>"phylo"</code>) or matrix (for <code>"multiPhylo"</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+describe.simmap">describe.simmap</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+sim.history">sim.history</a></code>
</p>

<hr>
<h2 id='labelnodes'>Function to interactively label nodes of a plotted tree</h2><span id='topic+labelnodes'></span>

<h3>Description</h3>

<p>Adds node labels to a plotted object of class <code>"phylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>labelnodes(text, node=NULL, interactive=TRUE, shape=c("circle","ellipse",
   "rect"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labelnodes_+3A_text">text</code></td>
<td>
<p>text string or vector to be used as labels.</p>
</td></tr>
<tr><td><code id="labelnodes_+3A_node">node</code></td>
<td>
<p>node numbers (indices) for the labels.</p>
</td></tr>
<tr><td><code id="labelnodes_+3A_interactive">interactive</code></td>
<td>
<p>logical value indicating whether or not nodes should be supplied interactively. (I.e., by clicking on the nodes.)</p>
</td></tr>
<tr><td><code id="labelnodes_+3A_shape">shape</code></td>
<td>
<p>shape to plot around the plotted node label(s).</p>
</td></tr>
<tr><td><code id="labelnodes_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The nodes to be labels can be selected interactively by the user (i.e., by clicking on the corresponding nodes of the plotted tree).
</p>


<h3>Value</h3>

<p>Invisibly returns a vector of the node indices for the labeled nodes.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cladelabels">cladelabels</a></code>, <code><a href="ape.html#topic+nodelabels">nodelabels</a></code>
</p>

<hr>
<h2 id='ladderize.simmap'>Ladderize a tree with a mapped discrete character</h2><span id='topic+ladderize.simmap'></span>

<h3>Description</h3>

<p>Ladderizes an object of class <code>"simmap"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ladderize.simmap(tree, right=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ladderize.simmap_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="ladderize.simmap_+3A_right">right</code></td>
<td>
<p>a logical specifying how the tree should be ladderized.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function 'ladderizes' an object of class <code>"simmap"</code> with a mapped discrete character.
</p>
<p>For more information see <code><a href="ape.html#topic+ladderize">ladderize</a></code>.
</p>


<h3>Value</h3>

<p>A ladderized object of class <code>"simmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="ape.html#topic+ladderize">ladderize</a></code>
</p>

<hr>
<h2 id='lambda.transform'><code class="reqn">\lambda</code> transformation of matrix</h2><span id='topic+lambda.transform'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+phyl.pca">phyl.pca</a></code> and others.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lambda.transform(lambda, C)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lambda.transform_+3A_lambda">lambda</code></td>
<td>
<p>scalar, usually (but not necessarily) on the interval 0,1.</p>
</td></tr>
<tr><td><code id="lambda.transform_+3A_c">C</code></td>
<td>
<p>matrix probably returned by <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multiplies the off-diagonals of a square matrix by <code>lambda</code> and returns the result.
</p>


<h3>Value</h3>

<p>Typically an among-species phylogenetic variance covariance matrix (e.g., <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>) in which the off-diagonal elements have been multiplied by <code>lambda</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='likMlambda'>Likelihood for joint <code class="reqn">\lambda</code></h2><span id='topic+likMlambda'></span>

<h3>Description</h3>

<p>Likelihood function for joint estimation of Pagel's <code class="reqn">\lambda</code> parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likMlambda(lambda, X, C)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="likMlambda_+3A_lambda">lambda</code></td>
<td>
<p>scalar, usually on the interval 0,1 (although not required to be).</p>
</td></tr>
<tr><td><code id="likMlambda_+3A_x">X</code></td>
<td>
<p>data for various continuous character, in the form of a matrix.</p>
</td></tr>
<tr><td><code id="likMlambda_+3A_c">C</code></td>
<td>
<p><em>n</em> x <em>n</em> matrix (for <em>n</em> taxa) containing the height above the root for each pair of taxa in the tree (e.g., <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generally intended to be used internally by other methods that do joint optimization of <code class="reqn">\lambda</code> (e.g., <code><a href="#topic+phyl.pca">phyl.pca</a></code>).
</p>


<h3>Value</h3>

<p>The log-likelihood.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='linklabels'>Function to add tip labels to a plotted tree with linking lines</h2><span id='topic+linklabels'></span>

<h3>Description</h3>

<p>Adds tip labels to a plotted tree by drawing curved, bent, or straight linking lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linklabels(text,tips,link.type=c("bent","curved","straight"),
   ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linklabels_+3A_text">text</code></td>
<td>
<p>text string or vector to be used as labels.</p>
</td></tr>
<tr><td><code id="linklabels_+3A_tips">tips</code></td>
<td>
<p>node numbers (indices) for the tips to be labeled.</p>
</td></tr>
<tr><td><code id="linklabels_+3A_link.type">link.type</code></td>
<td>
<p>manner in which to draw the linking lines.</p>
</td></tr>
<tr><td><code id="linklabels_+3A_...">...</code></td>
<td>
<p>optional arguments, including <code>cex</code>, <code>lty</code>, <code>lwd</code>, and <code>col</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The idea underlying this function is that the user should first plot the tree without tip labels, but set the area of the plotting device to be sufficient to accommodate the tip labels once they have been added. The function then can be called to add tip labels connected by linking lines to the tips of the plotted tree.
</p>


<h3>Value</h3>

<p>This function annotates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cladelabels">cladelabels</a></code>, <code><a href="ape.html#topic+nodelabels">nodelabels</a></code>, <code><a href="ape.html#topic+tiplabels">tiplabels</a></code>
</p>

<hr>
<h2 id='locate.fossil'>Locate a fossil lineage in a tree using continuous characters</h2><span id='topic+locate.fossil'></span>

<h3>Description</h3>

<p>Uses ML to place a fossil lineage into a tree using continuous traits following Revell et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locate.fossil(tree, X, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locate.fossil_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="locate.fossil_+3A_x">X</code></td>
<td>
<p>a matrix with continuous character data.</p>
</td></tr>
<tr><td><code id="locate.fossil_+3A_...">...</code></td>
<td>
<p>optional arguments including <code>time.constraint</code> which can be a scalar (positive height above the root of the fossil or negative time before present) or a vector (age range of fossil, either positive or negative); <code>edge.constraint</code>, which is equivalent to <code>constraint</code> in <code><a href="#topic+locate.yeti">locate.yeti</a></code>; <code>plot</code>, <code>rotate</code>, and <code>quiet</code>, which have the same interpretation (and defaults) as the equivalent arguments in <code><a href="#topic+locate.yeti">locate.yeti</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Optimized tree as an object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (1981) Maximum likelihood estimation of evolutionary trees from continuous characters. <em>American Journal of Human Genetics</em>, 25, 471-492.
</p>
<p>Felsenstein, J. (2002) Quantitative characters, phylogenies, and morphometrics. In: MacLeod, N. and P. Forey (Eds.) <em>Morphology, Shape and Phylogeny</em> (pp. 27-44). Taylor and Francis, London.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., D. L. Mahler, R. G. Reynolds, and G. J. Slater. (2015) Placing cryptic, recently extinct, or hypothesized taxa into an ultrametric phylogeny using continuous, character data: A case study with the lizard <em>Anolis roosevelti</em>. <em>Evolution</em>, <b>69</b>, 1027-1035. 
</p>

<hr>
<h2 id='locate.yeti'>Locate a cryptic, recently extinct, or missing taxon on a tree</h2><span id='topic+locate.yeti'></span>

<h3>Description</h3>

<p>Uses ML (or REML) to place a recently extinct, cryptic, or missing taxon on an ultrametric (i.e., time-calibrated) phylogeny following Revell et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locate.yeti(tree, X, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locate.yeti_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="locate.yeti_+3A_x">X</code></td>
<td>
<p>a matrix with continuous character data.</p>
</td></tr>
<tr><td><code id="locate.yeti_+3A_...">...</code></td>
<td>
<p>optional arguments including: <code>method</code> (<code>"ML"</code> or <code>"REML"</code>, defaults to <code>"ML"</code>); <code>search</code> (<code>"heuristic"</code> or <code>"exhaustive"</code>, defaults to <code>"heuristic"</code>); <code>constraint</code>, a vector containing the daughter node numbers from <code>tree$edge</code> for each edge to try; <code>plot</code> a logical argument specifying whether or not to plot the likelihood profile on edges (defaults to <code>FALSE</code>); <code>rotate</code> a logical indicating whether or not to rotate the data based on the input tree; and <code>quiet</code>, which is logical and has an obvious interpretation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Optimized tree as an object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (1981) Maximum likelihood estimation of evolutionary trees from continuous characters. <em>American Journal of Human Genetics</em>, 25, 471-492.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., D. L. Mahler, R. G. Reynolds, and G. J. Slater. (2015) Placing cryptic, recently extinct, or hypothesized taxa into an ultrametric phylogeny using continuous, character data: A case study with the lizard <em>Anolis roosevelti</em>. <em>Evolution</em>, <b>69</b>, 1027-1035.
</p>

<hr>
<h2 id='ls.tree'>Least squares branch lengths for a given tree</h2><span id='topic+ls.tree'></span>

<h3>Description</h3>

<p>Computes the least squares branch lengths conditioned on a topology and distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ls.tree(tree, D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ls.tree_+3A_tree">tree</code></td>
<td>
<p>phylogeny.</p>
</td></tr>
<tr><td><code id="ls.tree_+3A_d">D</code></td>
<td>
<p>distance matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generally intended as a function to be used internally by <code><a href="#topic+optim.phylo.ls">optim.phylo.ls</a></code>.
</p>


<h3>Value</h3>

<p>A tree with branch lengths.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='ltt'>Creates lineage-through-time plot (including extinct lineages)</h2><span id='topic+ltt'></span><span id='topic+ltt.phylo'></span><span id='topic+ltt.multiPhylo'></span><span id='topic+ltt.simmap'></span><span id='topic+ltt.multiSimmap'></span><span id='topic+gtt'></span><span id='topic+mccr'></span>

<h3>Description</h3>

<p>Computes and visualizes a lineage through time (LTT) plot, and related measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ltt(tree, ...)
## S3 method for class 'phylo'
ltt(tree, plot=TRUE, drop.extinct=FALSE, log.lineages=TRUE, gamma=TRUE, ...)
## S3 method for class 'multiPhylo'
ltt(tree, drop.extinct=FALSE, gamma=TRUE, ...)
## S3 method for class 'simmap'
ltt(tree, plot=TRUE, log.lineages=FALSE, gamma=TRUE, ...)
## S3 method for class 'multiSimmap'
ltt(tree, gamma=TRUE, ...)
gtt(tree, n=100, ...)
mccr(obj, rho=1, nsim=100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ltt_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>, <code>"multiPhylo"</code>, <code>"simmap"</code>, or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="ltt_+3A_plot">plot</code></td>
<td>
<p>a logical value indicating whether or not to create LTT plot.</p>
</td></tr>
<tr><td><code id="ltt_+3A_drop.extinct">drop.extinct</code></td>
<td>
<p>logical value indicating whether or not to drop extinct tips from the tree.</p>
</td></tr>
<tr><td><code id="ltt_+3A_log.lineages">log.lineages</code></td>
<td>
<p>logical value indicating whether LTT plot should be on log-linear (default for <code>"phylo"</code> and <code>"multiPhylo"</code> objects) or linear-linear scale.</p>
</td></tr>
<tr><td><code id="ltt_+3A_gamma">gamma</code></td>
<td>
<p>logical value indicating whether or not to compute <code class="reqn">\gamma</code> from Pybus &amp; Harvey (2000; <em>Proc. Roy. Soc. B</em>).</p>
</td></tr>
<tr><td><code id="ltt_+3A_n">n</code></td>
<td>
<p>for <code>gtt</code> the number of time intervals to use to track <code class="reqn">\gamma</code> through time.</p>
</td></tr>
<tr><td><code id="ltt_+3A_obj">obj</code></td>
<td>
<p>for <code>mccr</code> an object of class <code>"ltt"</code>.</p>
</td></tr>
<tr><td><code id="ltt_+3A_rho">rho</code></td>
<td>
<p>for <code>mccr</code> sampling fraction.</p>
</td></tr>
<tr><td><code id="ltt_+3A_nsim">nsim</code></td>
<td>
<p>for <code>mccr</code> number of simulations to use for the MCCR test.</p>
</td></tr>
<tr><td><code id="ltt_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to plotting methods. See <code><a href="graphics.html#topic+plot.default">plot.default</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>ltt</code> computes LTT plot with extant and extinct lineages, and optionally conducts <code class="reqn">\gamma</code>-test of Pybus &amp; Harvey (2000). The object returned by <code>ltt</code> can be plotted or re-plotted using <code><a href="base.html#topic+plot">plot</a></code>.
</p>
<p>For the case in which <code>tree</code> is an object of class <code>"simmap"</code> or <code>"multiSimmap"</code> then the object will contain the number of lineages through time (for each tree, in the case of <code>"multiSimmap"</code> objects) separated by mapped regimes.
</p>
<p>The function <code>gtt</code> computes the value of Pybus &amp; Harvey's <code class="reqn">\gamma</code> statistic through time by slicing the tree at various points - by default in even intervals from the time above the root at which <em>N</em> = 3 to the present day.
</p>
<p>The function <code>mccr</code> performs the MCCR test of Pybus &amp; Harvey (2000) which takes into account incomplete taxon sampling in computing a P-value of the <code class="reqn">\gamma</code> statistic.
</p>
<p>Although it is calculated here, it's unclear how to interpret the <code class="reqn">\gamma</code>-statistic if not all the tips in the tree are contemporaneous.
</p>


<h3>Value</h3>

<p><code>ltt</code> returns an object of class <code>"ltt"</code> which normally includes the following elements:
</p>
<table>
<tr><td><code>times</code></td>
<td>
<p>a vector of branching times.</p>
</td></tr>
<tr><td><code>ltt</code></td>
<td>
<p>a vector of lineages, or a matrix of lineages in each state over time for objects of class <code>"simmap"</code> and <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>optionally, a value for the <code class="reqn">\gamma</code>-statistic.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>two-tailed P-value for the <code class="reqn">\gamma</code>-test.</p>
</td></tr>
</table>
<p>If <code>tree</code> is an object of class <code>"multiPhylo"</code>, then an object of class <code>"multiLtt"</code> is returned consisting of a list of object of class <code>"ltt"</code>.
</p>
<p><code>gtt</code> returns an object of class <code>"gtt"</code>.
</p>
<p><code>mccr</code> returns of object of class <code>"mccr"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Pybus, O. G., and P. H. Harvey (2000) Testing macro-evolutionary models using incomplete molecular phylogenies. <em>Proc. R. Soc. Lond. B</em>, <b>267</b>, 2267-2272.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gammatest">gammatest</a></code>, <code><a href="#topic+ltt95">ltt95</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## LTT plots
set.seed(99)
trees&lt;-pbtree(n=100,scale=100,nsim=10)
obj&lt;-ltt(trees,plot=FALSE)
plot(obj,log="y",log.lineages=FALSE,
    bty="l")
title(main="LTT plots for 10 pure-birth trees",
    font.main=3)
tree&lt;-pbtree(b=1,d=0.25,t=4)
obj&lt;-ltt(tree,gamma=FALSE,show.tree=TRUE,
    bty="l")
title(main="LTT plot with superimposed tree",
    font.main=3)
obj
## GTT plot
data(anoletree)
anole.gtt&lt;-gtt(anoletree,n=40)
plot(anole.gtt)
</code></pre>

<hr>
<h2 id='ltt95'>Creates a (1-<code class="reqn">\alpha</code>)% CI for a set of LTTs</h2><span id='topic+ltt95'></span><span id='topic+plot.ltt95'></span>

<h3>Description</h3>

<p>This function computes LTT plots for a set of trees &amp; plots a (1-<code class="reqn">\alpha</code>)% CI by various methods. (See <code><a href="#topic+ltt">ltt</a></code> for more details.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ltt95(trees, alpha=0.05, log=FALSE, method=c("lineages","times"),
   mode=c("median","mean"), ...)
## S3 method for class 'ltt95'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ltt95_+3A_trees">trees</code></td>
<td>
<p>is an object of class <code>"multiPhylo"</code> containing a list of phylogenetic trees.</p>
</td></tr>
<tr><td><code id="ltt95_+3A_alpha">alpha</code></td>
<td>
<p>confidence level. Defaults to <code>alpha=0.05</code>. <code>alpha=0</code> will mean that the interval around <em>all</em> trees in the set will be plotted.</p>
</td></tr>
<tr><td><code id="ltt95_+3A_log">log</code></td>
<td>
<p>logical value indicating whether or not to plot on the semi-log scale.</p>
</td></tr>
<tr><td><code id="ltt95_+3A_method">method</code></td>
<td>
<p>plot the CI on the number of lineages given time (<code>"lineages"</code>); or on times given a number of lineages (<code>"times"</code>).</p>
</td></tr>
<tr><td><code id="ltt95_+3A_mode">mode</code></td>
<td>
<p>plot the median or mean LTT.</p>
</td></tr>
<tr><td><code id="ltt95_+3A_x">x</code></td>
<td>
<p>object of class <code>"ltt95"</code> for plotting method.</p>
</td></tr>
<tr><td><code id="ltt95_+3A_...">...</code></td>
<td>
<p>optional arguments to be used by <code>ltt95</code> or the plotting method. So far: <code>res</code> gives the number of time-steps (defaults to <code>res=100</code>); <code>xaxis</code> (<code>"standard"</code>, <code>"negative"</code>, or <code>"flipped"</code>) determines the scale (time from the root, time back from the present, or time from the present) of the x-axis of the plot; <code>lend</code> determines the line end type (as in <code><a href="graphics.html#topic+par">par</a></code>); <code>shaded</code> determines whether to plot the (1-<code class="reqn">\alpha</code>)% CI using dotted lines (if <code>FALSE</code>) or shading (if <code>TRUE</code>); and <code>bg</code> is the background color for shading if <code>shaded=TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates a plot and invisibly returns an object of class <code>"ltt95"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ltt">ltt</a></code>
</p>

<hr>
<h2 id='make.era.map'>Create &quot;era&quot; map on a phylogenetic tree</h2><span id='topic+make.era.map'></span>

<h3>Description</h3>

<p>Creates a temporal map on the tree based on <code>limits</code> provided by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.era.map(tree, limits, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.era.map_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="make.era.map_+3A_limits">limits</code></td>
<td>
<p>a vector containing the temporal limits, in time since the root node of the tree, for the mappings. The first number should be <code>0</code>, and each subsequent number should be the start of each subsequent regime or era to be mapped on the tree.</p>
</td></tr>
<tr><td><code id="make.era.map_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> with the specified eras mapped as different regimes.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree&lt;-pbtree(n=1000,scale=100)
tree&lt;-make.era.map(tree,c(0,25,50,75))
plot(tree,ftype="off",lwd=1)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='make.simmap'>Simulate stochastic character maps on a phylogenetic tree or trees</h2><span id='topic+make.simmap'></span><span id='topic+simmap'></span>

<h3>Description</h3>

<p>Performs stochastic character mapping (Huelsenbeck et al., 2003) using several different alternative methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.simmap(tree, x, model="SYM", nsim=1, ...)
simmap(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.simmap_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree as an object of class <code>"phylo"</code>, or a list of trees as an object of class <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="make.simmap_+3A_x">x</code></td>
<td>
<p>a vector containing the tip states for a discretely valued character, or a matrix containing the prior probabilities of tip states in rows and character states as column names. The names (if <code>x</code> is a vector) or row names (if <code>x</code> is a matrix) should match the tip labels of the tree. The vector can be of class <code>"factor"</code>, <code>"character"</code>, or <code>"numeric"</code> (although in the lattermost case its content should obviously be only integer values).</p>
</td></tr>
<tr><td><code id="make.simmap_+3A_model">model</code></td>
<td>
<p>a character string containing the model or a transition model specified in the form of a matrix. See <code><a href="ape.html#topic+ace">ace</a></code> for more details.</p>
</td></tr>
<tr><td><code id="make.simmap_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations. If <code>tree</code> is an object of class <code>"multiPhylo"</code>, then <code>nsim</code> simulations will be conducted <em>per</em> input tree.</p>
</td></tr>
<tr><td><code id="make.simmap_+3A_...">...</code></td>
<td>
<p>optional arguments. So far, <code>pi</code> gives the prior distribution on the root node of the tree. Acceptable values for <code>pi</code> are <code>"equal"</code>, <code>"estimated"</code>, or a vector with the frequencies. If <code>pi="estimated"</code> then the stationary distribution is estimated by numerically solving <code>pi*Q=0</code> for <code>pi</code>, and this is used as a prior on the root. If <code>pi="fitzjohn"</code>, then the Fitzjohn et al. (2009) root prior is used. Finally, if <code>pi</code> is a numeric vector then the root state will be sampled from this vector. The function defaults to <code>pi="equal"</code> which results in the root node being sampled from the conditional scaled likelihood distribution at the root. <code>message</code> tells whether or not to print a message containing the rate matrix, <em>Q</em> and state frequencies. <code>message</code> defaults to <code>TRUE</code>. For optional argument <code>Q="mcmc"</code> (see below) the mean value of <code>Q</code> from the posterior sample is printed. <code>tol</code> gives the tolerance for zero elements in <code>Q</code>. (Elements less then <code>tol</code> will be reset to <code>tol</code>). Optional argument <code>Q</code> can be a string (<code>"empirical"</code> or <code>"mcmc"</code>), or a fixed value of the transition matrix, <em>Q</em>. If <code>"empirical"</code> than a single value of <em>Q</em>, the most likely value, is used for all simulations. If <code>"mcmc"</code>, then <code>nsim</code> values of <em>Q</em> are first obtained from the posterior distribution for <em>Q</em> using Bayesian MCMC, then a simulated stochastic character map is generated for each sampled value of <em>Q</em>. Optional argument <code>vQ</code> can consist of a single numeric value or a vector containing the variances of the (normal) proposal distributions for the MCMC. The order of <code>vQ</code> is assumed to be in the order of the <code>index.matrix</code> in <code><a href="ape.html#topic+ace">ace</a></code> for the chosen model. <code>prior</code> is a list containing <code>alpha</code> and <code>beta</code> parameters for the <code class="reqn">\Gamma</code> prior distribution on the transition rates in <em>Q</em>. Note that <code>alpha</code> and <code>beta</code> can be single values or vectors, if different priors are desired for each value in the transition matrix <em>Q</em>. As for <code>vQ</code>, the order of <code>prior</code> is assumed to correspond with the order of <code>index.matrix</code> as in <code><a href="ape.html#topic+ace">ace</a></code>. <code>prior</code> can also be given the optional logical value <code>use.empirical</code> which tells the function whether or not to give the prior distribution the empirical mean for <em>Q</em>. If <code>TRUE</code> then only <code>prior$beta</code> is used and <code>prior$alpha</code> is set equal to <code>prior$beta</code> times the empirical mean of <em>Q</em>. <code>burnin</code> and <code>samplefreq</code> are burn-in and sample frequency for the MCMC, respectively.</p>
</td></tr>
<tr><td><code id="make.simmap_+3A_object">object</code></td>
<td>
<p>for generic <code>simmap</code> method, object of various classes: for instance, an object of class <code>"fitMk"</code> from <code><a href="#topic+fitMk">fitMk</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>Q="empirical"</code>, <code>make.simmap</code> first fits a continuous-time reversible Markov model for the evolution of <code>x</code> and then simulates stochastic character histories using that model and the tip states on the tree. This is the same procedure that is described in Bollback (2006), except that simulation is performed using a fixed value of the transition matrix, <em>Q</em>, instead of by sampling <em>Q</em> from its posterior distribution.
</p>
<p>For <code>Q="mcmc"</code>, <code>make.simmap</code> first samples <em>Q</em> <code>nsim</code> times from the posterior probability distribution of <em>Q</em> using MCMC, then it simulates <code>nsim</code> stochastic maps conditioned on each sampled value of <em>Q</em>.
</p>
<p>For <code>Q</code> set to a matrix, <code>make.simmap</code> samples stochastic mappings conditioned on the fixed input matrix.
</p>
<p><code>make.simmap</code> uses code that has been adapted from <span class="pkg">ape</span>'s function <code><a href="ape.html#topic+ace">ace</a></code> (by Paradis et al.) to perform Felsenstein's pruning algorithm to compute the likelihood.
</p>
<p>As of <span class="pkg">phytools</span> &gt;= 0.2-33 <code>x</code> can be a vector of states or a matrix containing the prior probabilities of tip states in rows. In this case the column names of <code>x</code> should contain the states, and the row names should contain the tip names.
</p>
<p>Note that there was a small (but potentially significant) bug in how node states were simulated by <code>make.simmap</code> in versions of <span class="pkg">phytools</span> &lt;= 0.2-26. Between <span class="pkg">phytools</span> 0.2-26 and 0.2-36 there was also a bug for asymmetric models of character change (e.g., <code>model="ARD"</code>). Finally, between <span class="pkg">phytools</span> 0.2-33 and <span class="pkg">phytools</span> 0.2-47 there was an error in use of the conditional likelihoods for the root node, which caused the root node of the tree to be sampled incorrectly. Giorgio Bianchini pointed out that in <span class="pkg">phytools</span> 1.0-1 (and probably prior recent versions) there was an error sampling the state at the root node of the tree based on the input prior (<code>pi</code>) supplied by a user &ndash; except for <code>pi="equal"</code> (a flat prior, the default) or for a prior distribution in which one or another state was known to be the global root state (e.g., <code>pi=c(1,0)</code>, <code>pi=c(0,1)</code>, etc.). All of these issues should be fixed in the current and all later versions.
</p>
<p>If <code>tree</code> is an object of class <code>"multiPhylo"</code> then <code>nsim</code> stochastic maps are generated for each input tree.
</p>


<h3>Value</h3>

<p>A object of class <code>"simmap"</code> or <code>"multiSimmap"</code> which consists of an object of class <code>"phylo"</code> (or a list of such objects with class <code>"multiPhylo"</code>), with the following additional elements:
</p>
<table>
<tr><td><code>maps</code></td>
<td>
<p>a list of named vectors containing the times spent in each state on each branch, in the order in which they occur.</p>
</td></tr>
<tr><td><code>mapped.edge</code></td>
<td>
<p>a matrix containing the total time spent in each state along each edge of the tree.</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>the assumed or sampled value of <code>Q</code>.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>the log-likelihood of the assumed or sampled <code>Q</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>FitzJohn, R. G., W. P. Maddison, and S. P. Otto (2009) Estimating trait-dependent speciation and extinction rates from incompletely resolved phylogenies. <em>Systematic Biology</em>, <b>58</b>, 595-611.
</p>
<p>Huelsenbeck, J. P., R. Neilsen, and J. P. Bollback (2003) Stochastic mapping of morphological characters. <em>Systematic Biology</em>, <b>52</b>, 131-138. 
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J. and L. J. Harmon (2022) <em>Phylogenetic Comparative Methods in R</em>. Princeton University Press.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+brownieREML">brownieREML</a></code>, <code><a href="#topic+countSimmap">countSimmap</a></code>, <code><a href="#topic+describe.simmap">describe.simmap</a></code>, <code><a href="#topic+evol.vcv">evol.vcv</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+write.simmap">write.simmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## load tree and data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract discrete character (feeding mode)
fmode&lt;-setNames(sunfish.data$feeding.mode,
  rownames(sunfish.data))
## fit model
er_model&lt;-fitMk(sunfish.tree,fmode,model="ER",
  pi="fitzjohn")
## do stochastic mapping
sunfish_smap&lt;-simmap(er_model)
## print a summary of the stochastic mapping
summary(sunfish_smap)
## plot a posterior probabilities of ancestral states
cols&lt;-setNames(c("blue","red"),levels(fmode))
plot(summary(sunfish_smap),colors=cols,ftype="i")
legend("topleft",c("non-piscivorous","piscivorous"),
  pch=21,pt.bg=cols,pt.cex=2)
par(mar=c(5.1,4.1,4.1,2.1),las=1)
## plot posterior density on the number of changes
plot(density(sunfish_smap),bty="l")
title(main="Posterior distribution of changes of each type",
  font.main=3)
## End(Not run)
</code></pre>

<hr>
<h2 id='map.overlap'>Proportional overlap between two mapped character histories on a tree</h2><span id='topic+map.overlap'></span><span id='topic+Map.Overlap'></span>

<h3>Description</h3>

<p>Calculates the similarity of two different stochastically mapped character histories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map.overlap(tree1, tree2, tol=1e-6, ...)
Map.Overlap(tree1, tree2, tol=1e-06, standardize=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map.overlap_+3A_tree1">tree1</code></td>
<td>
<p>an object of class <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="map.overlap_+3A_tree2">tree2</code></td>
<td>
<p>an object of class <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="map.overlap_+3A_tol">tol</code></td>
<td>
<p>an optional tolerance value.</p>
</td></tr>
<tr><td><code id="map.overlap_+3A_standardize">standardize</code></td>
<td>
<p>for <code>Map.Overlap</code>, a logical value indicating whether or not to standardize overlap by dividing by the summed branch length of the tree.</p>
</td></tr>
<tr><td><code id="map.overlap_+3A_...">...</code></td>
<td>
<p>optional arguments, such as <code>check.equal</code>, a logical value indicating whether or not to check if <code>tree1</code> and <code>tree2</code> match in underlying topology and branch lengths (they should). This value is <code>TRUE</code> by default, but can be set to <code>FALSE</code> if <code>tree1</code> and <code>tree2</code> are known to be equal to speed up calculation.</p>
</td></tr>
</table>


<h3>Details</h3>

	
<p><code>map.overlap</code> computes a single quantity giving the overall similarity of the maps, consequently this measure only makes sense of some or all of the states are shared between the two mapped tress. 
</p>
<p>In <code>Map.Overlap</code> what is computed instead is a matrix in which the rows correspond to the states observed in <code>tree1</code> and the columns give the states for <code>tree2</code>, with the numerical values of the matrix showing the total overlap between each pair of mapped states on the two trees.
</p>


<h3>Value</h3>

<p>A numerical value on the interval (0, 1), for <code>map.overlap</code>; or a matrix whose elements should sum to 1.0 (<code>Map.Overlap</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>
</p>

<hr>
<h2 id='map.to.singleton'>Converts a tree without singletons to a tree with singleton nodes</h2><span id='topic+map.to.singleton'></span><span id='topic+plotTree.singletons'></span><span id='topic+drop.tip.singleton'></span><span id='topic+rootedge.to.singleton'></span>

<h3>Description</h3>

<p><code>map.to.singleton</code> takes an object of class <code>"simmap"</code> with a mapped discrete character and converts it to a tree with singleton nodes, in which edge has only one state.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map.to.singleton(tree)
plotTree.singletons(tree)
## S3 method for class 'singleton'
drop.tip(phy, tip, ...)
rootedge.to.singleton(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map.to.singleton_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code> (for <code>map.to.singleton</code>,  or a tree with one or more singleton nodes (for <code>plotTree.singletons</code>, <code>drop.tip.singleton</code>, and <code>rootedge.to.singleton</code>).</p>
</td></tr>
<tr><td><code id="map.to.singleton_+3A_phy">phy</code></td>
<td>
<p>for <code>drop.tip.singleton</code>, an object of class <code>"singleton"</code> or <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="map.to.singleton_+3A_tip">tip</code></td>
<td>
<p>for <code>drop.tip.singleton</code>, a tip label or vector of tip labels.</p>
</td></tr>
<tr><td><code id="map.to.singleton_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>drop.tip.singleton</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

	
<p>The states for each edge are stored in <code>names(tree$edge.length)</code>. In a sense this is just an alternative way to use the general structure of the <code>"phylo"</code> object to encode a tree with a mapped character.	
</p>
<p><code>plotTree.singletons</code> plots a tree with singleton nodes. Note that <code><a href="#topic+plotTree">plotTree</a></code> and <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> now have no trouble graphing trees with singleton nodes - but they do this by just ignoring the singletons. <code>plotTree.singletons</code> marks the singletons as nodes on the plotted phylogeny.
</p>
<p><code>drop.tip.singleton</code> drops tips from the tree leaving ancestral nodes for all remaining tips as singletons. 
</p>
<p>Finally, <code>rootedge.to.singleton</code> converts a tree with a root edge to a tree with a singleton node instead.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> with singleton nodes. <code>plotTree.singletons</code> graphs a tree in which the singleton nodes are shown. 
</p>
<p>If <code>names(tree$edge.length)!=NULL</code> <code>plotTree.singletons</code> will use a different color from <code><a href="grDevices.html#topic+palette">palette</a></code> for each mapped state.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+collapse.singles">collapse.singles</a></code>, <code><a href="ape.html#topic+drop.tip">drop.tip</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>
</p>

<hr>
<h2 id='mapped.states'>Returns a vector, matrix, or list of the mapped states on a tree or set of trees</h2><span id='topic+mapped.states'></span>

<h3>Description</h3>

<p>Computes and orders a vector, matrix, or list of the unique mapped states on a tree or state of trees of class <code>"simmap"</code> or <code>"multiSimmap"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapped.states(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapped.states_+3A_tree">tree</code></td>
<td>
<p>a single tree or a set of trees as an object of class <code>"simmap"</code> or <code>"multiSimmap"</code>, respectively.</p>
</td></tr>
<tr><td><code id="mapped.states_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector, matrix, or list.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='markChanges'>Add marked changes to a plotted tree with mapped discrete character</h2><span id='topic+markChanges'></span>

<h3>Description</h3>

<p>Adds the reconstructed changes to a plotted tree with a stochastically mapped discrete character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markChanges(tree, colors=NULL, cex=1, lwd=2, plot=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="markChanges_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="markChanges_+3A_colors">colors</code></td>
<td>
<p>a named vector of colors used to plot the stochastically mapped character on the tree.</p>
</td></tr>
<tr><td><code id="markChanges_+3A_cex">cex</code></td>
<td>
<p>expansion factor for line height.</p>
</td></tr>
<tr><td><code id="markChanges_+3A_lwd">lwd</code></td>
<td>
<p>line width.</p>
</td></tr>
<tr><td><code id="markChanges_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether the changes should be plotted or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns (invisibly) a matrix containing the x &amp; y coordinates of the marked changes on the plotted tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load tree and data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract discrete character (feeding mode)
fmode&lt;-setNames(sunfish.data$feeding.mode,
  rownames(sunfish.data))
## fit model
er_model&lt;-fitMk(sunfish.tree,fmode,model="ER",
  pi="fitzjohn")
## generate single stochastic map
sunfish_smap&lt;-simmap(er_model,nsim=1)
## plot stochastic map &amp; mark changes
cols&lt;-setNames(c("blue","red"),levels(fmode))
plot(sunfish_smap,cols,ftype="i")
markChanges(sunfish_smap,colors=cols,lwd=6)
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='matchNodes'>Matches nodes between two trees</h2><span id='topic+matchNodes'></span><span id='topic+matchLabels'></span>

<h3>Description</h3>

<p>This function returns a matrix in which the first column contains <em>all</em> of the internal nodes of <code>tr1</code> and the second column contains the matching nodes from <code>tr2</code>, inasmuch as they can be identified. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchNodes(tr1, tr2, method=c("descendants","distances"), ...)
matchLabels(tr1, tr2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matchNodes_+3A_tr1">tr1</code></td>
<td>
<p>first tree.</p>
</td></tr>
<tr><td><code id="matchNodes_+3A_tr2">tr2</code></td>
<td>
<p>second tree.</p>
</td></tr>
<tr><td><code id="matchNodes_+3A_method">method</code></td>
<td>
<p>method to use to match nodes between trees. <code>"descendants"</code> uses the tip species descended from each node; <code>"distances"</code> uses the distances from the nodes to the tips. Any unambiguous shortening of <code>"descendants"</code> or <code>"distances"</code> is also permitted.</p>
</td></tr>
<tr><td><code id="matchNodes_+3A_...">...</code></td>
<td>
<p>optional arguments which may or may not be used depending on the value of <code>method</code>. <code>tol</code> is a tolerance value for the difference from exact matching that is allowed for <code>method="distances"</code>. <code>corr</code>, which is <code>FALSE</code> by default, indicates whether to match nodes under <code>method="distances"</code> using the correlation (<code>corr=TRUE</code>) or the absolute similarity of distances.</p>
</td></tr>
</table>


<h3>Details</h3>

	
<p>For <code>method="descendants"</code>, pairs of matching nodes are defined by sharing all descendant leaves in common. 
</p>
<p>For <code>method="distances"</code>, nodes are considered to matched if the share the same set of distances (or proportional distances, for optional argument <code>corr=TRUE</code>) to all tips. 
</p>
<p><code>matchLabels</code> is functionally equivalent but matches node (tip) indices based on identifying matching in the labels only.
</p>


<h3>Value</h3>

<p>A matrix in which the first column contains the nodes of <code>tr1</code> with the second column containing matching nodes in <code>tr2</code>, with the criterion for matching defined by <code>method</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='mergeMappedStates'>Merge two or more mapped states into one state</h2><span id='topic+mergeMappedStates'></span>

<h3>Description</h3>

<p>Merges two or mapped states on the tree to get one new state.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeMappedStates(tree, old.states, new.state)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergeMappedStates_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code> or <code>"multiSimmap"</code> containing one or more phylogenetic trees with a mapped discrete character.</p>
</td></tr>
<tr><td><code id="mergeMappedStates_+3A_old.states">old.states</code></td>
<td>
<p>states to merge.</p>
</td></tr>
<tr><td><code id="mergeMappedStates_+3A_new.state">new.state</code></td>
<td>
<p>name for new state.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mergeMappedStates</code> can be used to merge two or more mapped states into a single, new state. For instance, one could merge the states <code>"C"</code>, <code>"G"</code>, and <code>"T"</code> and define the new the state <code>"not-A"</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> or <code>"multiSimmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>
</p>

<hr>
<h2 id='midpoint_root'>Midpoint root a phylogeny</h2><span id='topic+midpoint_root'></span><span id='topic+midpoint.root'></span>

<h3>Description</h3>

<p>These functions midpoint root a rooted or unrooted tree (Farris 1972).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>midpoint_root(tree)
## S3 method for class 'root'
midpoint(tree, node.labels="support", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="midpoint_root_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="midpoint_root_+3A_node.labels">node.labels</code></td>
<td>
<p>argument to be passed to <code><a href="phangorn.html#topic+midpoint">midpoint</a></code>.</p>
</td></tr>
<tr><td><code id="midpoint_root_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="phangorn.html#topic+midpoint">midpoint</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Midpoint rooting involves locating the midpoint of the longest path between any two tips and putting the root in that location. 
</p>
<p>The function <code>midpoint_root</code> performs the same operation as <code><a href="phangorn.html#topic+midpoint">midpoint</a></code> in the <span class="pkg">phangorn</span> package, but uses no <span class="pkg">phangorn</span> (Schliep, 2011) code internally.
</p>
<p>The function <code>midpoint.root</code> is a pseudo S3 method for the object class <code>"root"</code> that exists because when <code><a href="phangorn.html#topic+midpoint">midpoint</a></code> was created in <span class="pkg">phangorn</span> it was not defined as a generic method. This function merely points to <code><a href="phangorn.html#topic+midpoint">midpoint</a></code> and is being deprecated out.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> containing a rooted phylogenetic tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Farris, J. (1972) Estimating phylogenetic trees from distance matrices. <em>American Naturalist</em>, <b>106</b>, 645-667.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Schliep K. P. (2011) phangorn: phylogenetic analysis in R. <em>Bioinformatics</em>, <b>27</b>, 592-593.
</p>


<h3>See Also</h3>

<p><code><a href="phangorn.html#topic+midpoint">midpoint</a></code>, <code><a href="#topic+reroot">reroot</a></code>, <code><a href="ape.html#topic+root">root</a></code>
</p>

<hr>
<h2 id='minRotate'>Rotates all nodes of the tree to minimize the difference in order with a vector</h2><span id='topic+minRotate'></span><span id='topic+tipRotate'></span>

<h3>Description</h3>

<p>Rotates all the nodes of the tree to try and minimize the different between the order of the tips and the rank-order of a numeric vector <code>x</code> or (in the case of <code>tipRotate</code>) the actual integer vector, <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minRotate(tree, x, ...)
tipRotate(tree, x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="minRotate_+3A_tree">tree</code></td>
<td>
<p>tree.</p>
</td></tr>
<tr><td><code id="minRotate_+3A_x">x</code></td>
<td>
<p>numeric vector.</p>
</td></tr>
<tr><td><code id="minRotate_+3A_...">...</code></td>
<td>
<p>optional arguments to be used by <code>tipRotate</code>. Presently optional arguments can be <code>fn</code>, function to be used to compute the distance between the order of the tip labels in <code>tree</code> and the numeric vector <code>x</code> (presently <code>fn=function(x) x^2</code> by default); <code>methods</code>, the method or methods of tree traversal (can be <code>"pre"</code>, <code>"post"</code>, or <code>c("pre","post")</code>, for pre-, post-, or both pre- and post-order tree traversal); <code>rotate.multi</code>, whether to rotate multifurcations in all possible ways using <code>rotate.multi</code> (defaults to <code>FALSE</code>); and <code>print</code>, a logical argument specifying whether to print the search progress or to behave quietly. Only the option <code>print</code> is available for <code>minRotate</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code>minRotate</code> and <code>tipRotate</code> are designed primarily to be used internally by other <span class="pkg">phytools</span> functions and particularly by <code><a href="#topic+phylo.to.map">phylo.to.map</a></code> (in the case of <code>minRotate</code>) and by <code><a href="#topic+cophylo">cophylo</a></code> (in the case of <code>tipRotate</code>).
</p>


<h3>Value</h3>

<p>A node-rotated object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='minSplit'>Finding the minimum (median) split in the posterior sample</h2><span id='topic+minSplit'></span>

<h3>Description</h3>

<p>This function takes a phylogenetic tree and a list of splits and identifies the split with the smallest summed or summed squared distances to all the other splits.	
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minSplit(tree, split.list, method="sum", printD=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="minSplit_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="minSplit_+3A_split.list">split.list</code></td>
<td>
<p>either a matrix with two named columns, <code>"node"</code> and <code>"bp"</code>; a <code>$mcmc</code> matrix from <code>evol.rate.mcmc()</code>; or the entire raw output from <code>evol.rate.mcmc()</code>.</p>
</td></tr>
<tr><td><code id="minSplit_+3A_method">method</code></td>
<td>
<p>an optional string indicating the criterion to minimize: options are <code>"sum"</code> and <code>"sumsq"</code>.</p>
</td></tr>
<tr><td><code id="minSplit_+3A_printd">printD</code></td>
<td>
<p>logical specifying whether to print distances to screen (<code>FALSE</code> by default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements:
</p>
<table>
<tr><td><code>node</code></td>
<td>
<p>node for the minimum split.</p>
</td></tr>
<tr><td><code>bp</code></td>
<td>
<p>location on the branch leading to <code>node</code> of the minimum split.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., D. L. Mahler, P. Peres-Neto, and B. D. Redelings (2012) A new method for identifying exceptional phenotypic diversification. <em>Evolution</em>, <b>66</b>, 135-146.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>, <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>
</p>

<hr>
<h2 id='modified.Grafen'>Computes modified Grafen edge lengths</h2><span id='topic+modified.Grafen'></span><span id='topic+node.paths'></span>

<h3>Description</h3>

<p>Computes modified Grafen (1989) edge lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modified.Grafen(tree, power=2)
node.paths(tree, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="modified.Grafen_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="modified.Grafen_+3A_power">power</code></td>
<td>
<p>power to raise the depths of each node (in nodes).</p>
</td></tr>
<tr><td><code id="modified.Grafen_+3A_node">node</code></td>
<td>
<p>node number for <code>node.paths</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes modified Grafen edge lengths in which the length of the edge is determined not by the number of descendant leaves, but instead by the maximum number of node lengths in the path from the node to any leaf.
</p>
<p><code>node.paths</code> is used internally by <code>modified.Grafen</code> and computes the set of paths from a node to all tips descended from that node.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> with edge lengths.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

	
<p>Grafen, A. (1989) The phylogenetic regression. <em>Philosophical Transactions of the Royal Society of London. Series B. Biological Sciences</em>, <b>326</b>, 119-157.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+compute.brlen">compute.brlen</a></code>
</p>

<hr>
<h2 id='mrp.supertree'>Matrix representation parsimony supertree estimation</h2><span id='topic+mrp.supertree'></span><span id='topic+compute.mr'></span>

<h3>Description</h3>

<p>This function estimates the MRP (matrix representation parsimony) supertree from a set of input trees (Baum 1992; Ragan 1992).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrp.supertree(trees, method=c("pratchet","optim.parsimony"), ...)
compute.mr(trees, type=c("phyDat","matrix"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrp.supertree_+3A_trees">trees</code></td>
<td>
<p>an object of class <code>"multiPhylo"</code>  that consists of a list of phylogenetic trees.</p>
</td></tr>
<tr><td><code id="mrp.supertree_+3A_method">method</code></td>
<td>
<p>an argument specifying whether to optimize the tree using the <span class="pkg">phangorn</span> parsimony optimizer <code><a href="phangorn.html#topic+pratchet">pratchet</a></code> or <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code>.</p>
</td></tr>
<tr><td><code id="mrp.supertree_+3A_type">type</code></td>
<td>
<p>for <code>compute.mr</code>, the type of object to return (e.g., <code>"phyDat"</code> or <code>"matrix"</code>).</p>
</td></tr>
<tr><td><code id="mrp.supertree_+3A_...">...</code></td>
<td>
<p>optional arguments - mostly to be passed to <code><a href="phangorn.html#topic+pratchet">pratchet</a></code> or <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mrp.supertree</code> uses <code><a href="phangorn.html#topic+pratchet">pratchet</a></code> or <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code> from the <span class="pkg">phangorn</span> package (Schliep, 2011) for optimization, and <code><a href="ape.html#topic+prop.part">prop.part</a></code> from <span class="pkg">ape</span> package (Paradis et al. 2004). 
</p>
<p>See <code><a href="phangorn.html#topic+pratchet">pratchet</a></code> or <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code> for optional arguments, which vary slightly depending on the method. All optional arguments of these methods are available to the user with one exception. The argument <code>tree</code> in <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code> is supplied instead as <code>start</code>. In addition to being an object of class <code>"phylo"</code>, <code>start</code> can also be assigned the string values of <code>"NJ"</code> or <code>"random"</code>, in which case either a neighbor-joining or random tree will be used as the starting tree for optimization.
</p>
<p>The function <code>compute.mr</code> computes the matrix-representation matrix of the input trees. It is used internally by <code>mrp.supertree</code>, but can also be used to export an object that can be written to file if desired.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> or <code>"multiPhylo"</code> that is the MP or set of MP MRP trees.
</p>
<p>In the case of <code>compute.mr</code>, an object of class <code>"phyDat"</code> or a matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Baum, B. R., (1992) Combining trees as a way of combining data sets for phylogenetic inference, and the desirability of combining gene trees. <em>Taxon</em>, <b>41</b>, 3-10.
</p>
<p>Felsenstein, J. (2004) <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Ragan, M. A. (1992) Phylogenetic inference based on matrix representation of trees. <em>Molecular Phylogenetics and Evolution</em>, <em>1</em>, 53-58.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Schliep, K. P. (2011) phangorn: phylogenetic analysis in R. <em>Bioinformatics</em>, <b>27</b>, 592-593.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exhaustiveMP">exhaustiveMP</a></code>, <code><a href="phangorn.html#topic+optim.parsimony">optim.parsimony</a></code>, <code><a href="phangorn.html#topic+pratchet">pratchet</a></code>
</p>

<hr>
<h2 id='multi.mantel'>Multiple matrix regression (partial Mantel test)</h2><span id='topic+multi.mantel'></span>

<h3>Description</h3>

<p>This function conducting a multiple matrix regression (partial Mantel test) and uses Mantel (1967) permutations to test the significance of the model and individual coefficients. It also returns the residual and predicted matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi.mantel(Y, X, nperm=1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi.mantel_+3A_y">Y</code></td>
<td>
<p>single &quot;dependent&quot; square matrix. Can be either a symmetric matrix of class <code>"matrix"</code> or a distance matrix of class <code>"dist"</code>.</p>
</td></tr>
<tr><td><code id="multi.mantel_+3A_x">X</code></td>
<td>
<p>a single independent matrix or multiple independent matrices in a list. As with <code>Y</code> can be a object of class <code>"matrix"</code> or class <code>"dist"</code>, or a list of such objects.</p>
</td></tr>
<tr><td><code id="multi.mantel_+3A_nperm">nperm</code></td>
<td>
<p>number of Mantel permutations to be used to compute a P-value of the test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Printing the object to screen will result in a summary of the analysis similar to <code>summary.lm</code>, but with p-values derived from Mantel permutations.
</p>
<p>Methods <code>residuals</code> and <code>fitted</code> can be used to return residual and fitted matrices, respectively.
</p>


<h3>Value</h3>

<p>An object of class <code>"multi.mantel"</code> consisting of the following elements:
</p>
<table>
<tr><td><code>r.squared</code></td>
<td>
<p>multiple R-squared.</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>model coefficients, including intercept.</p>
</td></tr>
<tr><td><code>tstatistic</code></td>
<td>
<p>t-statistics for model coefficients.</p>
</td></tr>
<tr><td><code>fstatistic</code></td>
<td>
<p>F-statistic for the overall model.</p>
</td></tr>
<tr><td><code>probt</code></td>
<td>
<p>vector of probabilities, based on permutations, for <code>tstatistic</code>.</p>
</td></tr>
<tr><td><code>probF</code></td>
<td>
<p>probability of F, based on Mantel permutations.</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>matrix of residuals.</p>
</td></tr>
<tr><td><code>predicted</code></td>
<td>
<p>matrix of predicted values.</p>
</td></tr>
<tr><td><code>nperm</code></td>
<td>
<p>the number of permutations used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Mantel, N. (1967) The detection of disease clustering and a generalized regression approach. <em>Cancer Research</em>, <b>27</b>, 209-220.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='multiC'>Returns a list with phylogenetic VCV matrix for each mapped state</h2><span id='topic+multiC'></span>

<h3>Description</h3>

<p>Computes phylogenetic covariance matrices from a <code>"simmap"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiC(tree, internal=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiC_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code> consisting of a phylogeny with a mapped discrete character.</p>
</td></tr>
<tr><td><code id="multiC_+3A_internal">internal</code></td>
<td>
<p>logical value indicating whether or not internal nodes should be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a modified <code>"phylo"</code> object as input and returns a set of so-called phylogenetic covariance matrices (e.g., see <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>) as a list: one for each mapped state.
</p>
<p>Used internally by multiple <span class="pkg">phytools</span> functions, such as <code><a href="#topic+brownie.lite">brownie.lite</a></code>.
</p>


<h3>Value</h3>

<p>A list of matrices.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+evolvcv.lite">evolvcv.lite</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+vcvPhylo">vcvPhylo</a></code>, <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>
</p>

<hr>
<h2 id='multirateBM'>Function to fit a multi-rate Brownian evolution model</h2><span id='topic+multirateBM'></span>

<h3>Description</h3>

<p>Fits a flexible multi-rate Brownian motion evolution model using penalized likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multirateBM(tree, x, method=c("ML","REML"), 
    optim=c("L-BFGS-B","Nelder-Mead","BFGS","CG"),
    maxit=NULL, n.iter=1, lambda=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multirateBM_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_x">x</code></td>
<td>
<p>a named numerical vector. Names should correspond to the species names of <code>tree</code>.</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_method">method</code></td>
<td>
<p>method of optimization. Currently only <code>method="ML"</code>.</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_optim">optim</code></td>
<td>
<p>optimization routine to be used by <code><a href="stats.html#topic+optim">optim</a></code>. If more than one is specified and <code>n.iter&gt;1</code> then they will be alternated. (This is recommended to improve optimization.)</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_maxit">maxit</code></td>
<td>
<p>to be passed to <code>optim</code>. If set to <code>maxit=NULL</code>, the default value of <code>maxit</code> will be used, depending on the optimization method.</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_n.iter">n.iter</code></td>
<td>
<p>number of times to reiterate failed optimization.</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_lambda">lambda</code></td>
<td>
<p>lambda penalty term. High values of <code>lambda</code> correspond to high penalty for rate heterogeneity among edges. Low values of <code>lambda</code> correspond to low penalty.</p>
</td></tr>
<tr><td><code id="multirateBM_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a flexible Brownian multi-rate model using penalized likelihood.
</p>
<p>The model that is being fit is one in which the rate of Brownian motion evolution itself evolves from edge to edge in the tree under a process of geometric Brownian evolution (i.e., Brownian motion evolution on a log scale).
</p>
<p>The penalty term, <code>lambda</code>, determines the cost of variation in the rate of evolution from branch to branch. If lambda is <em>high</em>, then the rate of evolution will vary relatively little between edges (and in the limiting case converge to the single-rate MLE estimate of the rate). By contrast, if the value of <code>lambda</code> is set to be low, then the rate of evolution can vary from edge to edge with relatively little penalty. 
</p>
<p>Decreasing the penalty term, however, is not without cost. As <code>lambda</code> is decreased towards zero, estimated rates will tend to become less and less accurate.
</p>


<h3>Value</h3>

<p>An object of class <code>"multirateBM"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2021) A variable-rate quantitative trait evolution model using penalized-likelihood. <em>PeerJ</em>, <b>9</b>, e11997.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## load data
data(sunfish.tree)
data(sunfish.data)
## convert from "simmap" to "phylo"
sunfish.tree&lt;-as.phylo(sunfish.tree)
## extract character of interest
gw&lt;-setNames(sunfish.data$gape.width,
    rownames(sunfish.data))
## run penalized-likelihood optimization
## lambda=0.1 is arbitrary
fitBM&lt;-multirateBM(sunfish.tree,gw,
    lambda=0.01)
## print and plot the results
print(fitBM)
plot(fitBM,ftype="i",fsize=0.8,lwd=6,
    outline=TRUE)
## reset par
par(mar=c(5.1,4.1,4.1,2.1))
## End(Not run)
</code></pre>

<hr>
<h2 id='multiRF'>Computes Robinson-Foulds distance between a set of trees</h2><span id='topic+multiRF'></span>

<h3>Description</h3>

<p>Computes the Robinson-Foulds (Robinson &amp; Foulds 1981) distance between a set of trees in an object of class <code>"multiPhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiRF(trees,quiet=FALSE,multi2di=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiRF_+3A_trees">trees</code></td>
<td>
<p>object of class <code>"multiPhylo"</code> consisting of two or more fully bifurcating, unrooted trees. If trees are rooted, they will be unrooted.</p>
</td></tr>
<tr><td><code id="multiRF_+3A_quiet">quiet</code></td>
<td>
<p>logical argument indicating whether or not to run quietly. (Defaults to <code>FALSE</code>.)</p>
</td></tr>
<tr><td><code id="multiRF_+3A_multi2di">multi2di</code></td>
<td>
<p>logical argument indicating whether or not to resolve multifurcating trees. (Defaults to <code>FALSE</code>.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the Robinson-Foulds distance between all phylogenies in an object of class <code>"multiPhylo"</code>. Uses <code><a href="ape.html#topic+prop.part">prop.part</a></code> internally for most of the heavy lifting.
</p>


<h3>Value</h3>

<p>A matrix containing distances.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Robinson, D. R., Foulds, L. R. (1981) Comparison of phylogenetic trees. <em>Mathematical Biosciences</em>, <b>53</b>, 131-147.
</p>

<hr>
<h2 id='nodeHeights'>Compute the heights above the root of each node</h2><span id='topic+nodeHeights'></span><span id='topic+nodeheight'></span>

<h3>Description</h3>

<p><code>nodeHeights</code> computes the height above the root for all nodes in the tree. <code>nodeheight</code> computes the height above the root for a single node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nodeHeights(tree, ...)
nodeheight(tree, node, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nodeHeights_+3A_tree">tree</code></td>
<td>
<p>a phylogeny as an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="nodeHeights_+3A_node">node</code></td>
<td>
<p>for <code>nodeheight</code>, the node for which we want to compute a height above the root (or including the root edge, for <code>root.edge=TRUE</code>).</p>
</td></tr>
<tr><td><code id="nodeHeights_+3A_...">...</code></td>
<td>
<p>optional arguments - presently only <code>root.edge</code>, a logical value indicating whether or not to include the root edge length in the calculation of node heights.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>nodeHeights</code> also gives a handy way to get the total length of the tree from the root to the highest tip which will be given by <code>max(nodeHeights(tree))</code>.
</p>
<p>Generally speaking, <code>nodeHeights</code> will be faster if the heights of all or a large proportion of nodes is needed, whereas <code>nodeheight</code> will be faster if the height of one or a small number of nodes are needed.
</p>


<h3>Value</h3>

<p>Either a matrix of the same dimensions as <code>tree$edge</code> containing the height above the root of each node in <code>edge</code> (for <code>nodeHeights</code>); or a single positive number (for <code>nodeheight</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vcvPhylo">vcvPhylo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load tree
data(vertebrate.tree)
## compute height of all nodes
H&lt;-nodeHeights(vertebrate.tree)
print(H)
## compute total tree depth
max(H)
</code></pre>

<hr>
<h2 id='nodelabels.cophylo'>Add labels to a plotted &quot;cophylo&quot; object</h2><span id='topic+nodelabels.cophylo'></span><span id='topic+tiplabels.cophylo'></span><span id='topic+edgelabels.cophylo'></span>

<h3>Description</h3>

<p>This function adds node, edge, or tip labels to the plotted trees of a <code>"cophylo"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nodelabels.cophylo(..., which=c("left","right"))
edgelabels.cophylo(..., which=c("left","right"))
tiplabels.cophylo(..., which=c("left","right"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nodelabels.cophylo_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code><a href="ape.html#topic+nodelabels">nodelabels</a></code>, <code><a href="ape.html#topic+edgelabels">edgelabels</a></code>, or <code><a href="ape.html#topic+tiplabels">tiplabels</a></code>.</p>
</td></tr>
<tr><td><code id="nodelabels.cophylo_+3A_which">which</code></td>
<td>
<p>argument indicated which of the two plotted trees (the <code>"left"</code> or <code>"right"</code> tree) to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the order of tips, edges, and nodes may be different in the object of class <code>"cophylo"</code> than they are in the original input trees, particularly if <code>cophylo(...,rotate=TRUE)</code> was used.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophylo">cophylo</a></code>, <code><a href="ape.html#topic+edgelabels">edgelabels</a></code>, <code><a href="ape.html#topic+nodelabels">nodelabels</a></code>, <code><a href="ape.html#topic+tiplabels">tiplabels</a></code>
</p>

<hr>
<h2 id='optim.phylo.ls'>Phylogeny inference using the least squares method</h2><span id='topic+optim.phylo.ls'></span>

<h3>Description</h3>

<p>Phylogenetic inference using the method of least-squares (Cavalli-Sforza &amp; Edwards, 1967).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optim.phylo.ls(D, stree=NULL, set.neg.to.zero=TRUE, fixed=FALSE,
   tol=1e-10, collapse=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optim.phylo.ls_+3A_d">D</code></td>
<td>
<p>a distance matrix.</p>
</td></tr>
<tr><td><code id="optim.phylo.ls_+3A_stree">stree</code></td>
<td>
<p>an optional starting tree for the optimization.</p>
</td></tr>
<tr><td><code id="optim.phylo.ls_+3A_set.neg.to.zero">set.neg.to.zero</code></td>
<td>
<p>a logical value indicating whether to set negative branch lengths to zero (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="optim.phylo.ls_+3A_fixed">fixed</code></td>
<td>
<p>a logical value indicating whether to estimate the topology - if <code>TRUE</code> only the branch lengths will be computed.</p>
</td></tr>
<tr><td><code id="optim.phylo.ls_+3A_tol">tol</code></td>
<td>
<p>a tolerance value used to assess whether the optimization has converged.</p>
</td></tr>
<tr><td><code id="optim.phylo.ls_+3A_collapse">collapse</code></td>
<td>
<p> a logical indicating whether to collapse branches with zero length.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function uses <code><a href="phangorn.html#topic+nni">nni</a></code> from the <span class="pkg">phangorn</span> package (Schliep 2011) to conduct NNIs for topology estimation.
</p>
<p>Since topology optimization is performed using NNIs, convergence to the true least-squares topology is not guaranteed. It is consequently probably wise to start with a very good tree - such as a NJ tree.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> that (may be) the least-squares tree with branch lengths; also returns the sum of squares in <code>attr(tree,"Q-score")</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Cavalli-Sforza, L. L., and A. W. F. Edwards. (1967) Phylogenetic analysis: Models and estimation procedures. <em>American Journal of Human Genetics</em>, <b>19</b>, 233-257.
</p>
<p>Felsenstein, J. (2004) <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer. (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Schliep, K. P. (2011) phangorn: phylogenetic analysis in R. <em>Bioinformatics</em>, <b>27</b>, 592-593.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exhaustiveMP">exhaustiveMP</a></code>, <code><a href="phangorn.html#topic+nni">nni</a></code>
</p>

<hr>
<h2 id='orderMappedEdge'>Order the columns of mapped.edge to match across trees</h2><span id='topic+orderMappedEdge'></span>

<h3>Description</h3>

<p>Orders the levels of a mapped character to match across trees in a <code>"multiSimmap"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orderMappedEdge(trees, ordering=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orderMappedEdge_+3A_trees">trees</code></td>
<td>
<p>object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="orderMappedEdge_+3A_ordering">ordering</code></td>
<td>
<p>ordering for the columns of <code>$mapped.edge</code>. If <code>NULL</code>, then an alphabetical order is assumed. Options are <code>"alphabetical"</code>, <code>"numerical"</code>, or any specific ordering of the mapped traits (e.g., <code>c("A","B","C")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a an object of class <code>"multiSimmap"</code> with a mapped discrete character (e.g., see <code><a href="#topic+make.simmap">make.simmap</a></code> and sorts the columns of each <code>tree$mapped.edge</code> element to have the same state ordering. 
</p>
<p>This is handy if we want to, for instance, run <code>brownie.lite</code> on a set of mapped trees, and then average the fitted parameter values across trees. The function also works for a single tree.
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> or (normally) <code>"multiSimmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='paintSubTree'>Paint sub-trees with a discrete character</h2><span id='topic+paintSubTree'></span><span id='topic+paintBranches'></span>

<h3>Description</h3>

<p>Paints regimes on a tree to create an object of class <code>"simmap"</code> with mapped regimes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>paintSubTree(tree, node, state, anc.state="1", stem=FALSE)
paintBranches(tree, edge, state, anc.state="1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="paintSubTree_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree as an object of class <code>"phylo"</code> or a modified object with mapped character traits.</p>
</td></tr>
<tr><td><code id="paintSubTree_+3A_node">node</code></td>
<td>
<p>an integer specifying the node number tipward of which the function should paint the derived state.</p>
</td></tr>
<tr><td><code id="paintSubTree_+3A_edge">edge</code></td>
<td>
<p>an integer or vector of integers specifying the node or tip numbers of the edges that should be painted in <code>paintBranches</code>.</p>
</td></tr>
<tr><td><code id="paintSubTree_+3A_state">state</code></td>
<td>
<p>a string (or numeric value) specifying the state to paint on the tree tipward of <code>node</code>.</p>
</td></tr>
<tr><td><code id="paintSubTree_+3A_anc.state">anc.state</code></td>
<td>
<p>the ancestral state to use; will only be applied if there are presently no character values mapped on the tree.</p>
</td></tr>
<tr><td><code id="paintSubTree_+3A_stem">stem</code></td>
<td>
<p>logical or numeric value indicating whether to use the derived state on the stem leading to <code>node</code> (or not, if <code>stem=FALSE</code>), or, alternatively, what fraction of the stem should be assigned to the derived clade. Note that for tip clades <code>stem=FALSE</code> is not allowed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions map or &quot;paint&quot; arbitrary (i.e., user-specified) discrete character histories on the tree. 
</p>
<p><code>paintSubTree</code> paints the clade downstream of <code>node</code> with a particular state; whereas <code>paintBranches</code> paints only a specified branch.
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> that contains the specified paintings as a mapped discrete character.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>, <code><a href="#topic+sim.history">sim.history</a></code>
</p>

<hr>
<h2 id='paste.tree'>Paste two trees together</h2><span id='topic+paste.tree'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>paste.tree(tr1, tr2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="paste.tree_+3A_tr1">tr1</code></td>
<td>
<p>receptor tree.</p>
</td></tr>
<tr><td><code id="paste.tree_+3A_tr2">tr2</code></td>
<td>
<p>donor clade.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Primarily designed as an internal function for <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>; however, can be used to graft a clade onto a receptor tree at the &quot;sticky tip&quot; labeled with <code>"NA"</code>.
</p>
<p>The donor clade needs to have a root edge, even if it is zero length.
</p>


<h3>Value</h3>

<p>A tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tr1&lt;-rtree(10)
tr2&lt;-rtree(10)
tr1$tip.label[1]&lt;-"NA"
tr2$root.edge&lt;-0
tr3&lt;-paste.tree(tr1,tr2)
</code></pre>

<hr>
<h2 id='pbtree'>Simulate pure-birth or birth-death stochastic tree or trees</h2><span id='topic+pbtree'></span>

<h3>Description</h3>

<p>This function simulates stochastic birth-death trees. 
</p>
<p>Simulation can be performed conditioning on <code>n</code>, on <code>t</code>, or on both simultaneously. If both, then (for optional argument <code>method="rejection"</code>) rejection sampling is performed whereby trees are simulated given <code>b</code> and <code>t</code> until a tree containing <code>n</code> taxa is found. The giving-up point can be set using the optional argument <code>max.count</code>. 
</p>
<p>Simulations can also be performed in continuous time (the default) or discrete time; the difference being that wait times in the continuous-time simulation come from the exponential distribution; whereas waiting times in discrete-time simulations come from the geometric distribution. In addition, discrete-time simulations allow for the possibility that multiple speciation events can occur at (exactly) the same time, so long as they are on separate branches. Finally, sometimes for stopping criterion <code>n</code> in discrete-time there will be a number of tips different from <code>n</code>. This indicates that the last event contained more than one speciation event, and a warning is printed.
</p>
<p><code>method="direct"</code> is presently experimental. It does not really perform direct sampling; however waiting times &amp; birth or death events are sampled first - with only wait-times consistent with <code>n</code> and <code>t</code> being retained. This rejection sampling occurs one layer earlier than for <code>method="rejection"</code>. This results in a significant (several-fold) speed-up of the code and enables sampling conditioned on <code>n</code> and <code>t</code> simultaneously for much higher <code>b</code> and <code>d</code>. At the present time, <code>extant.only=TRUE</code> does not work for this mode, nor does <code>type="discrete"</code>.
</p>
<p>Note that if <code>ape=FALSE</code>, then the function will run faster, and the tree is theoretically compatible with the ape <code>"phylo"</code> standard; however some downstream errors with functions such as <code><a href="ape.html#topic+bind.tree">bind.tree</a></code> have been observed.
</p>
<p>Lastly, under the taxon number stopping criterion (<code>n</code>) for a non-zero extinction rate (<code>d&gt;0</code>) sometimes a tree containing fewer than <code>n</code> extant tips is returned because it has gone completely extinct before the end of the simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbtree(b=1, d=0, n=NULL, t=NULL, scale=NULL, nsim=1, type=c("continuous",
   "discrete"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbtree_+3A_b">b</code></td>
<td>
<p>birth rate or speciation rate for <code>type="continuous"</code>; the probability of speciating per time-step for <code>type="discrete"</code>.</p>
</td></tr>
<tr><td><code id="pbtree_+3A_d">d</code></td>
<td>
<p>death rate or extinction rate for <code>type="continuous"</code>; the probability of going extinct per time-step for <code>type="discrete"</code>.</p>
</td></tr>
<tr><td><code id="pbtree_+3A_n">n</code></td>
<td>
<p>desired number of species (i.e., taxa-stop criterion).</p>
</td></tr>
<tr><td><code id="pbtree_+3A_t">t</code></td>
<td>
<p>total time for simulation (i.e., time-stop criterion).</p>
</td></tr>
<tr><td><code id="pbtree_+3A_scale">scale</code></td>
<td>
<p>if set, rescales tree to have total length <code>scale</code>.</p>
</td></tr>
<tr><td><code id="pbtree_+3A_nsim">nsim</code></td>
<td>
<p>number of simulated trees to return.</p>
</td></tr>
<tr><td><code id="pbtree_+3A_type">type</code></td>
<td>
<p>string to indicate whether to simulate trees in continuous or discrete time. If the former, then wait times between speciation events are drawn from an exponential distribution; whereas if the latter then wait times comes from a geometric distribution.</p>
</td></tr>
<tr><td><code id="pbtree_+3A_...">...</code></td>
<td>
<p>optional arguments including <code>ape</code>, a logical value indicating whether to return nodes in a 'ape' compatible ordering (default is <code>TRUE</code>); <code>extant.only</code> a logical value indicating whether or not to return only extant species (defaults to <code>FALSE</code>); <code>max.count</code> a numeric value indicating the maximum number of iterations to run is sampling conditioned on both <code>n</code> and <code>t</code> (defaults to <code>1e5</code>); <code>method</code> gives the method used for simultaneously conditioning on <code>n</code> and <code>t</code> - options are <code>"rejection"</code> and <code>"direct"</code>; <code>tip.label</code>, a vector of tip labels (only works for <code>n!=NULL</code>); and, finally, <code>quiet</code>, a logical value indicating whether or not to suppress certain message (defaults to <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulate stochastic birth-death trees.
</p>


<h3>Value</h3>

<p>A tree or set of trees as an object of class <code>"phylo"</code> or <code>"multiPhylo"</code>, respectively.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate a pure-birth tree with 400 tips
## scaled to a length of 1.0
tree&lt;-pbtree(n=400,scale=1)
## simulate a pure-birth tree conditioning on n &amp; t
tt&lt;-log(50)-log(2)
tree&lt;-pbtree(n=50,t=tt)
</code></pre>

<hr>
<h2 id='pgls.Ives'>Phylogenetic regression with intraspecific sampling error</h2><span id='topic+pgls.Ives'></span><span id='topic+pgls.SEy'></span>

<h3>Description</h3>

<p>Phylogenetic regression with within-species sampling error following Ives et al. (2007).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgls.Ives(tree, X, y, Vx=NULL, Vy=NULL, Cxy=NULL, lower=c(1e-8,1e-8),
    fixed.b1=NULL)
pgls.SEy(model, data, corClass=corBrownian, tree,
    se=NULL, method=c("REML","ML"), interval=c(0,1000), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pgls.Ives_+3A_tree">tree</code></td>
<td>
<p>a phylogeny as an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_x">X</code></td>
<td>
<p>a named vector containing a <em>single</em> independent variable (multiple independent variables to be added in future). <code>X</code> can contain the species means, or a single long vector containing the sample of values for each species. In the latter case the <code>names(X)</code> will be repeating - all samples from the same species should have the same name.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_y">y</code></td>
<td>
<p>vector the dependent variable. Can be species means or individual values, as for <code>X</code>.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_vx">Vx</code></td>
<td>
<p>sampling variances for <code>X</code>. If <code>NULL</code>, then the within-species variance is computed from the data assuming that individual samples, not species means, have been provided in <code>X</code>.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_vy">Vy</code></td>
<td>
<p>sampling variances for <code>y</code>. If <code>NULL</code>, then the within-species variance is computed from the data assuming that individual samples, not species means, have been provided in <code>y</code>.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_cxy">Cxy</code></td>
<td>
<p>sampling covariances between <code>X</code> and <code>y</code>. This will also be computed from the data if <code>Cxy==NULL</code>. Note than in this case - but not for the calculation of <code>Vx</code> and <code>Vy</code>, the same number of observations and the same ordering must be provided for <code>X</code> and <code>y</code>. If this is not the case, then it is assumed that different individuals have been sampled for <code>X</code> and <code>y</code> and thus <code>Cxy</code> is assumed to be zero for all species.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_lower">lower</code></td>
<td>
<p>vector specifying the lower bounds for estimation for <code class="reqn">\sigma_x^2</code> and <code class="reqn">\sigma_y^2</code>, respectively. (Must be &gt; 0.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_fixed.b1">fixed.b1</code></td>
<td>
<p>fixed regression slope, <code class="reqn">\beta</code>. Usually set to zero for null hypothesis testing.</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_model">model</code></td>
<td>
<p>model to fit. (For <code>pgls.SEy</code>.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_data">data</code></td>
<td>
<p>data frame. (For <code>pgls.SEy</code>.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_corclass">corClass</code></td>
<td>
<p>correlation structure. (For <code>pgls.SEy</code>.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_se">se</code></td>
<td>
<p>vector of standard errors in <em>y</em>. (For <code>pgls.SEy</code>.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_method">method</code></td>
<td>
<p>optimization method. (For <code>pgls.SEy</code>.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_interval">interval</code></td>
<td>
<p>interval over which to perform optimization. (For <code>pgls.SEy</code>.)</p>
</td></tr>
<tr><td><code id="pgls.Ives_+3A_...">...</code></td>
<td>
<p>optional arguments. (For <code>pgls.SEy</code>.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>pgls.Ives</code> fits the phylogenetic regression model with within-species sampling error following Ives et al. (2007).
</p>
<p><code>pgls.SEy</code> fits a simpler model in which only sampling error in <code>y</code> is taken into account. This function uses <code><a href="nlme.html#topic+gls">gls</a></code> from the <span class="pkg">nlme</span> package internally for optimization and returns an object of class <code>"gls"</code> that is compatible with all methods for that object class.
</p>
<p>In the case of <code>pgls.Ives</code>, only the bivariate regression model is implemented. Note that some problems have been reported with the optimization algorithm for this model, which is simple and thus may fail to find the ML solution.
</p>
<p>In the case of <code>pgls.SEy</code> the user can (theoretically) specify any class of linear model permitted by <code><a href="nlme.html#topic+gls">gls</a></code>.
</p>


<h3>Value</h3>

<p>In the case of <code>pgls.Ives</code>, an object of class <code>"pgls.Ives"</code> with the following elements:
</p>
<table>
<tr><td><code>beta</code></td>
<td>
<p>a vector or matrix of regression coefficients.</p>
</td></tr>
<tr><td><code>sig2x</code></td>
<td>
<p>fitted BM rate for <code>X</code>.</p>
</td></tr>
<tr><td><code>sig2y</code></td>
<td>
<p>fitted BM rate for <code>y</code>.</p>
</td></tr>
<tr><td><code>a</code></td>
<td>
<p>fitted ancestral states for <code>X</code> and <code>y</code>.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>a value for convergence. <code>convergence=0</code> is good; see <code><a href="stats.html#topic+optim">optim</a></code> for more details.</p>
</td></tr>
<tr><td><code>message</code></td>
<td>
<p>a message for convergence.</p>
</td></tr>
</table>
<p>In the case of <code>pgls.SEy</code>, an object of class <code>"gls"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Ives, A. R., P. E. Midford, and T. Garland Jr. (2007) Within-species measurement error in phylogenetic comparative methods. <em>Systematic Biology</em>, <b>56</b>, 252-270.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+phylosig">phylosig</a></code>, <code><a href="#topic+phyl.resid">phyl.resid</a></code>
</p>

<hr>
<h2 id='phenogram'>Plot traitgram (phenogram)</h2><span id='topic+phenogram'></span>

<h3>Description</h3>

<p>Plots a phylogenetic traitgram (Evans et al., 2009).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenogram(tree, x, fsize=1.0, ftype="reg", colors=NULL, axes=list(),
   add=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phenogram_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>, with or without a mapped discrete character.</p>
</td></tr>
<tr><td><code id="phenogram_+3A_x">x</code></td>
<td>
<p>a vector containing the states at the tips <em>or</em> the states at all the tips and the internal nodes of the tree.</p>
</td></tr>
<tr><td><code id="phenogram_+3A_fsize">fsize</code></td>
<td>
<p>relative font size for tip labels.</p>
</td></tr>
<tr><td><code id="phenogram_+3A_ftype">ftype</code></td>
<td>
<p>font type - options are <code>"reg"</code>, <code>"i"</code> (italics), <code>"b"</code> (bold), or <code>"bi"</code> (bold-italics).</p>
</td></tr>
<tr><td><code id="phenogram_+3A_colors">colors</code></td>
<td>
<p>colors for plotting the mapped character (if available) in <code>tree</code>. If no character is mapped on the tree, then a single color for all the branches of the tree can be provided.</p>
</td></tr>
<tr><td><code id="phenogram_+3A_axes">axes</code></td>
<td>
<p>list of axis dimensions. Items are <code>time</code> and <code>trait</code>.</p>
</td></tr>
<tr><td><code id="phenogram_+3A_add">add</code></td>
<td>
<p>optional logical value indicating whether to add to an open plot. If <code>TRUE</code>, then new axes will not be plotted.</p>
</td></tr>
<tr><td><code id="phenogram_+3A_...">...</code></td>
<td>
<p>optional arguments including <code>xlim</code>, <code>ylim</code>, <code>log</code>, <code>main</code>, <code>sub</code>, <code>xlab</code>, <code>ylab</code>, <code>asp</code>, <code>type</code>, <code>lty</code>, <code>lwd</code>, <code>offset</code>, and <code>digits</code> are as in <code><a href="graphics.html#topic+plot.default">plot.default</a></code> or <code><a href="graphics.html#topic+par">par</a></code>. Note that <code>axes</code> overrides <code>xlim</code> and <code>ylim</code>. <code>spread.labels</code> is a logical value indicating whether or not to minimize tip label overlap (default is <code>TRUE</code>); <code>spread.cost</code> is a numeric vector indicating the relative penalty to be used for label overlap and deviance, respectively (if <code>spread.labels=TRUE</code>); <code>spread.range</code> is the range over which to (potentially) spread the labels - note that if labels do not overlap, not all of that range will be used; finally, <code>link</code> is a numeric value by which to offset the tip labels, linking them to the tips with a dashed line (default is <code>0</code>, if <code>spread.labels=FALSE</code>, or 10% of the total tree length otherwise). The optional argument <code>offsetFudge</code> &quot;fudges&quot; the computation of label offset in scaling <code>xlim</code>. It is 1.37, which is the correct fudge in the Windows R GUI, but this may need to be changed in other systems. <code>hold</code> indicates whether (or not) the output to the graphical device should be held using <code><a href="grDevices.html#topic+dev.hold">dev.hold</a></code> before plotting (defaults to <code>hold=TRUE</code>). <code>quiet</code> suppresses some system messages if set to <code>quiet=TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function plots a traitgram (Evans et al. 2009), that is, a projection of the phylogenetic tree in a space defined by phenotype (on the <em>y</em> axis) and time (on the <em>x</em>). If a discrete character is mapped on the tree this will also be plotted.
</p>
<p>For <code>spread.labels=TRUE</code> numerical optimization is performed to optimize the distribution of the labels vertically, where the solution depends on the vector <code>spread.cost</code> containing the cost of overlap (first) and the cost of deviation from the vertical position of the tip. Note that because this is done via numerical optimization, plotting may hang briefly while the best solution is found (especially for large trees).
</p>


<h3>Value</h3>

<p>Plots a traitgram, optionally with a mapped discrete character, and (invisibly) returns a matrix containing the coordinates of the plotted tip labels.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Evans, M. E. K., Smith, S. A., Flynn, R. S., Donoghue, M. J. (2009) Climate, niche evolution, and diversification of the &quot;bird-cage&quot; evening primroses (Oenothera, sections Anogra and Kleinia). <em>American Naturalist</em>, <b>173</b>, 225-240.
</p>
<p>Revell, L. J. (2013) Two new graphical methods for mapping trait evolution on phylogenies. <em>Methods in Ecology and Evolution</em>, <b>4</b>, 754-759.
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., K. Schliep, E. Valderrama, and J. E. Richardson (2018) Graphs in phylogenetic comparative analysis: Anscombe's quartet revisited. <em>Methods in Ecology and Evolution</em>, <b>9</b>, 2145-2154.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
  rownames(mammal.data)))
## plot traitgram
phenogram(mammal.tree,ln.bodyMass,ftype="i",
  spread.cost=c(1,0),fsize=0.7,color=palette()[4],
  xlab="time (ma)",ylab="log(body mass)",las=1)
</code></pre>

<hr>
<h2 id='phyl.cca'>Phylogenetic canonical correlation analysis</h2><span id='topic+phyl.cca'></span>

<h3>Description</h3>

<p>Phylogenetic canonical correlation analysis, following Revell &amp; Harrison (2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyl.cca(tree, X, Y, lambda=1.0, fixed=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyl.cca_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="phyl.cca_+3A_x">X</code></td>
<td>
<p>a data matrix with traits in columns.</p>
</td></tr>
<tr><td><code id="phyl.cca_+3A_y">Y</code></td>
<td>
<p>data matrix with traits in columns, to be correlated with <code>X</code>.</p>
</td></tr>
<tr><td><code id="phyl.cca_+3A_lambda">lambda</code></td>
<td>
<p>optionally, a (fixed) value for <code class="reqn">\lambda</code>.</p>
</td></tr>
<tr><td><code id="phyl.cca_+3A_fixed">fixed</code></td>
<td>
<p>optionally, a logical value indicating whether or not to estimate <code class="reqn">\lambda</code> using likelihood.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>(Optional) joint optimization of <code class="reqn">\lambda</code> is performed using <code><a href="stats.html#topic+optimize">optimize</a></code> on the interval (0,1).
</p>


<h3>Value</h3>

<p>An object of class <code>"phyl.cca"</code> containing the following elements:
</p>
<table>
<tr><td><code>cor</code></td>
<td>
<p>canonical correlations.</p>
</td></tr>
<tr><td><code>xcoef</code></td>
<td>
<p>coefficients for the canonical variables for <code>X</code>.</p>
</td></tr>
<tr><td><code>ycoef</code></td>
<td>
<p>coefficients for the canonical variables for <code>Y</code>.</p>
</td></tr>
<tr><td><code>xscores</code></td>
<td>
<p>matrix with the canonical scores for <code>X</code>.</p>
</td></tr>
<tr><td><code>yscores</code></td>
<td>
<p>matrix with the canonical scores for <code>Y</code>.</p>
</td></tr>
<tr><td><code>chisq</code></td>
<td>
<p>vector of <code class="reqn">\chi^2</code> values.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>P-values for the hypothesis test that the <em>i</em>th and all subsequent correlations are zero.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J., Harrison, A. S. (2008) PCCA: A program for phylogenetic canonical correlation analysis. <em>Bioinformatics</em>, <b>24</b>, 1018-1020.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyl.pca">phyl.pca</a></code>
</p>

<hr>
<h2 id='phyl.pairedttest'>Phylogenetic paired <em>t</em>-test</h2><span id='topic+phyl.pairedttest'></span>

<h3>Description</h3>

<p>Phylogenetic paired <em>t</em>-test following Lindenfors et al. (2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyl.pairedttest(tree, x1, x2=NULL, se1=NULL, se2=NULL, lambda=1.0, h0=0.0,
   fixed=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyl.pairedttest_+3A_tree">tree</code></td>
<td>
<p>a phylogeny as an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_x1">x1</code></td>
<td>
<p>data vector for first trait, or matrix with two traits in columns.</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_x2">x2</code></td>
<td>
<p>data vector for second trait (or null if <code>x1</code> is a matrix).</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_se1">se1</code></td>
<td>
<p>standard errors for <code>x1</code>.</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_se2">se2</code></td>
<td>
<p>standard errors for <code>x2</code>.</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_lambda">lambda</code></td>
<td>
<p>starting value for Pagel's &lambda; (or fixed value, if <code>fixed=TRUE</code>).</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_h0">h0</code></td>
<td>
<p>null hypothesis (to be tested) for the mean difference between <code>x1</code> and <code>x2</code>.</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_fixed">fixed</code></td>
<td>
<p>logical value specifying whether or not to optimize &lambda;.</p>
</td></tr>
<tr><td><code id="phyl.pairedttest_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function conducts a phylogenetic paired <em>t</em>-test, roughly following Lindenfors et al. (2010). 
</p>
<p>This is <em>not</em> a phylogenetic ANOVA, in which we want to compare the means of different sets of species on the tree. Instead, we are interested in the difference between two characters, or two measures of a character within a species, and we want to know if this difference is significantly different from zero controlling for the phylogenetic non-independence of species. 
</p>
<p>Likelihood optimization is performed using <code><a href="stats.html#topic+optim">optim</a></code> with <code>method="L-BFGS-B"</code> with box constraints on &lambda; (0,1).
</p>


<h3>Value</h3>

<p>An object of class <code>"phyl.pairedttest"</code> with the following elements:
</p>
<table>
<tr><td><code>dbar</code></td>
<td>
<p>phylogenetic mean difference.</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>standard error of <code>dbar</code>.</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>estimated evolutionary variance (of the difference).</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>fitted (or fixed) value of &lambda;.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood of the fitted model.</p>
</td></tr>
<tr><td><code>t.dbar</code></td>
<td>
<p>t-value (<code>(dbar-h0)/se</code> where <code>se</code> is computed from the Hessian).</p>
</td></tr>
<tr><td><code>P.dbar</code></td>
<td>
<p>P-value.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>the degrees of freedom.</p>
</td></tr>
<tr><td><code>h0</code></td>
<td>
<p>the null hypothesis that was tested.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Lindenfors, P., L. J. Revell, and C. L. Nunn (2010) Sexual dimorphism in primate aerobic capacity: A phylogenetic test. <em>J. Evol. Biol.</em>, <b>23</b>, 1183-1194.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='phyl.pca'>Phylogenetic principal components analysis</h2><span id='topic+phyl.pca'></span><span id='topic+biplot.phyl.pca'></span><span id='topic+scores'></span><span id='topic+scores.phyl.pca'></span><span id='topic+as.princomp'></span><span id='topic+as.prcomp'></span>

<h3>Description</h3>

<p>This function performs phylogenetic PCA following Revell (2009).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyl.pca(tree, Y, method="BM", mode="cov", ...)
## S3 method for class 'phyl.pca'
biplot(x, ...)
scores(object, ...)
## S3 method for class 'phyl.pca'
scores(object, ...)
as.princomp(x, ...)
as.prcomp(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyl.pca_+3A_tree">tree</code></td>
<td>
<p>phylogeny as an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="phyl.pca_+3A_y">Y</code></td>
<td>
<p>data matrix with traits in columns.</p>
</td></tr>
<tr><td><code id="phyl.pca_+3A_method">method</code></td>
<td>
<p>method to obtain the correlation structure: can be <code>"BM"</code> or <code>"lambda"</code>.</p>
</td></tr>
<tr><td><code id="phyl.pca_+3A_mode">mode</code></td>
<td>
<p>is the mode for the PCA: can be <code>"cov"</code> or <code>"corr"</code>.</p>
</td></tr>
<tr><td><code id="phyl.pca_+3A_x">x</code></td>
<td>
<p>object of class <code>"phyl.pca"</code> for <code>biplot</code>, <code>as.princomp</code>, and <code>as.prcomp</code> methods.</p>
</td></tr>
<tr><td><code id="phyl.pca_+3A_object">object</code></td>
<td>
<p>object of class <code>"phyl.pca"</code> for <code>scores</code> method.</p>
</td></tr>
<tr><td><code id="phyl.pca_+3A_...">...</code></td>
<td>
<p>for S3 plotting method <code>biplot.phyl.pca</code>, other arguments to be passed to <code><a href="stats.html#topic+biplot">biplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>method="lambda"</code> then <code class="reqn">\lambda</code> will be optimized on the interval (0,1) using <code><a href="stats.html#topic+optimize">optimize</a></code>. Optimization method can be set using the option <code>opt</code> which can take values <code>"ML"</code>, <code>"REML"</code>, or <code>"fixed"</code>. If the last of these is selected than the user should also specify a value of <code class="reqn">\lambda</code> to use via the argument <code>lambda</code>. 
</p>
<p>S3 methods (<code>print</code>, <code>summary</code>, and <code>biplot</code>) are modified from code provided by Joan Maspons and are based on the same methods for objects of class <code>"prcomp"</code>. Function <code>biplot</code> now permits the argument <code>choices</code> to be supplied, which should be a vector of length two indicated the two PC axes to be plotted.
</p>
<p>S3 method <code>scores</code> extracts or computes (for a matrix of <code>newdata</code>) PC scores given an object of class <code>"phyl.pca"</code>.
</p>
<p>S3 methods <code>as.prcomp</code> and <code>as.princomp</code> convert the object of class <code>"phyl.pca"</code> to objects of class <code>"prcomp"</code> and <code>"princomp"</code>, respectively.
</p>


<h3>Value</h3>

<p>An object of class <code>"phyl.pca"</code> consisting of a list with some or all of the following elements:
</p>
<table>
<tr><td><code>Eval</code></td>
<td>
<p>diagonal matrix of eigenvalues.</p>
</td></tr>
<tr><td><code>Evec</code></td>
<td>
<p>matrix with eigenvectors in columns.</p>
</td></tr>
<tr><td><code>S</code></td>
<td>
<p>matrix with scores.</p>
</td></tr>
<tr><td><code>L</code></td>
<td>
<p>matrix with loadings.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>fitted value of <code class="reqn">\lambda</code> (<code>method="lambda"</code> only).</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood for <code class="reqn">\lambda</code> model (<code>method="logL"</code> only).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a>, Joan Maspons</p>


<h3>References</h3>

<p>Revell, L. J. (2009) Size-correction and principal components for interspecific comparative studies. <em>Evolution</em>, <b>63</b>, 3258-3268.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyl.cca">phyl.cca</a></code>, <code><a href="#topic+phyl.resid">phyl.resid</a></code>, <code><a href="stats.html#topic+prcomp">prcomp</a></code>, <code><a href="stats.html#topic+princomp">princomp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Mahler et al. (2010)
data(anoletree)
data(anole.data)
## run phylogenetic PCA
anole.pca&lt;-phyl.pca(anoletree,anole.data)
print(anole.pca)
## plot results
plot(anole.pca)
biplot(anole.pca)
</code></pre>

<hr>
<h2 id='phyl.resid'>Phylogenetic size-correction via GLS regression</h2><span id='topic+phyl.resid'></span>

<h3>Description</h3>

<p>Computes the residuals from the phylogenetic regression of multiple <em>y</em> variables (in a matrix) on a single <em>x</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyl.resid(tree, x, Y, method="BM")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyl.resid_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="phyl.resid_+3A_x">x</code></td>
<td>
<p>vector containing the single independent variable (e.g., size), or matrix with multiple independent variables in columns.</p>
</td></tr>
<tr><td><code id="phyl.resid_+3A_y">Y</code></td>
<td>
<p>vector or matrix with one or multiple dependent variables in columns.</p>
</td></tr>
<tr><td><code id="phyl.resid_+3A_method">method</code></td>
<td>
<p>method to obtain the correlation structure: can be <code>"BM"</code> or <code>"lambda"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits one or multiple phylogenetic regressions (depending on the number of columns in <code>Y</code>) and computes the residuals.  Designed for phylogenetic size correction using GLS regression (e.g., Revell 2009).
</p>
<p>Optionally fits <code class="reqn">\lambda</code> for each regression model.  Likelihood optimization of <code class="reqn">\lambda</code> is performed for <code>method=</code> <code>"lambda"</code> using <code><a href="stats.html#topic+optimize">optimize</a></code> on the interval (0,1).
</p>
<p>This function is theoretically redundant with <code>residuals</code> applied to a <code>"gls"</code> object class in which the correlation structure is based on <code>corBrownian</code> or <code>corPagel</code>; however some users may find this method simpler, and it provides a good way to cross-check your results &amp; make sure that you are using <code>gls</code> correctly.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>
<table>
<tr><td><code>beta</code></td>
<td>
<p>a vector or matrix of regression coefficients.</p>
</td></tr>
<tr><td><code>resid</code></td>
<td>
<p>a vector or matrix of residuals for species.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>a vector of <code class="reqn">\lambda</code> values (<code>method="lambda"</code> only).</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>a vector of log-likelihoods (<code>method="lambda"</code> only).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2009) Size-correction and principal components for interspecific comparative studies. <em>Evolution</em>, <b>63</b>, 3258-3268.
</p>
<p>Revell, L. J. (2010) Phylogenetic signal and linear regression on species data. <em>Methods in Ecology and Evolution</em>, <b>1</b>, 319-329.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyl.pca">phyl.pca</a></code>, <code><a href="nlme.html#topic+gls">gls</a></code>
</p>

<hr>
<h2 id='phyl.RMA'>Phylogenetic reduced major axis (RMA) regression</h2><span id='topic+phyl.RMA'></span><span id='topic+coef.phyl.RMA'></span><span id='topic+plot.phyl.RMA'></span>

<h3>Description</h3>

<p>Phylogenetic reduced major axis (RMA) regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyl.RMA(x, y, tree, method="BM", lambda=NULL, fixed=FALSE, h0=1.0)
## S3 method for class 'phyl.RMA'
coef(object, ...)
## S3 method for class 'phyl.RMA'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyl.RMA_+3A_x">x</code></td>
<td>
<p>vector with names. In the case of the S3 <code>plot</code> method <code>x</code> is an object of class <code>"phyl.RMA"</code>.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_y">y</code></td>
<td>
<p>vector with names.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_method">method</code></td>
<td>
<p>method to obtain the correlation structure: can be <code>"BM"</code> or <code>"lambda"</code>.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_lambda">lambda</code></td>
<td>
<p>value of <code>lambda</code> for fixed <code class="reqn">\lambda</code>.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_fixed">fixed</code></td>
<td>
<p>logical value indicating whether or not <code class="reqn">\lambda</code> should be optimized using likelihood.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_h0">h0</code></td>
<td>
<p>null hypothesis for <code class="reqn">\beta</code>. Defaults to 1.0. Note that a null hypothesis of 0.0 is not allowed.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_object">object</code></td>
<td>
<p>for <code>coef</code> method, an object of class <code>"phyl.RMA"</code>.</p>
</td></tr>
<tr><td><code id="phyl.RMA_+3A_...">...</code></td>
<td>
<p>optional arguments for S3 methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Optionally jointly estimates <code class="reqn">\lambda</code> if <code>method="lambda"</code>. Likelihood optimization of <code class="reqn">\lambda</code> is performed using <code><a href="stats.html#topic+optimize">optimize</a></code> on the interval (0,1).
</p>
<p>The statistical hypothesis testing is based on Clarke (1980; reviewed in McArdle 1988), which differs from some other implementations of non-phylogenetic major axis regression in R.
</p>
<p>Note that some statistician think there is <em>never</em> a condition in which a reduced-major-axis regression is appropriate.
</p>


<h3>Value</h3>

<p>An object of class <code>"phyl.RMA"</code> consisting of a list with the following elements:
</p>
<table>
<tr><td><code>RMA.beta</code></td>
<td>
<p>a vector of RMA regression coefficients.</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>a VCV matrix for the traits.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>fitted value of <code class="reqn">\lambda</code> (<code>method="lambda"</code> only).</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood (<code>method="lambda"</code> only).</p>
</td></tr>
<tr><td><code>test</code></td>
<td>
<p>a vector containing results for hypothesis tests on <code class="reqn">\beta</code>.</p>
</td></tr>
<tr><td><code>resid</code></td>
<td>
<p>a vector of residuals for <code>y</code> given <code>x</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Clarke, M. R. B. (1980) The reduced major axis of a bivariate sample. <em>Biometrika</em>, <b>67</b>, 441-446.
</p>
<p>McArdle, B. H. (1988) The structural relationship: Regression in biology. <em>Can. J. Zool.</em>, <b>66</b>, 2329-2339.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyl.cca">phyl.cca</a></code>, <code><a href="#topic+phyl.pca">phyl.pca</a></code>, <code><a href="#topic+phyl.resid">phyl.resid</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.data)
data(mammal.tree)
## pull out &amp; log transform variables
lnBodyMass&lt;-setNames(log(mammal.data$bodyMass),
  rownames(mammal.data))
lnHomeRange&lt;-setNames(log(mammal.data$homeRange),
  rownames(mammal.data))
## fit RMA regression &amp; print results
fitted.rma&lt;-phyl.RMA(lnBodyMass,lnHomeRange,
  mammal.tree)
print(fitted.rma)
## plot fitted RMA
par(las=1,bty="n")
plot(fitted.rma,las=1,bty="n")
</code></pre>

<hr>
<h2 id='phyl.vcv'>Compute evolutionary VCV matrix for a tree &amp; dataset</h2><span id='topic+phyl.vcv'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+phyl.pca">phyl.pca</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyl.vcv(X, C, lambda)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyl.vcv_+3A_lambda">lambda</code></td>
<td>
<p>value for <code class="reqn">\lambda</code> transformation.</p>
</td></tr>
<tr><td><code id="phyl.vcv_+3A_x">X</code></td>
<td>
<p>data matrix.</p>
</td></tr>
<tr><td><code id="phyl.vcv_+3A_c">C</code></td>
<td>
<p>matrix containing the height above the root of each pair of species in the tree. Typically this will have been produced by calling <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Primarily designed as an internal function for <code><a href="#topic+phyl.pca">phyl.pca</a></code>; <code>phyl.vcv</code> can be used to compute the phylogenetic trait variance-covariance matrix given a phylogenetic VCV, <code class="reqn">\lambda</code>, and a data matrix. 
</p>
<p>This function should not be confused with <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code> in the <span class="pkg">ape</span> package (although one of the objects returned is the output of <code>vcv.phylo</code>).
</p>
<p>Note that prior to <span class="pkg">phytools</span> 2.1-0 the matrix <code>X</code> was not sorted to match the rows of <code>C</code> since that was assumed to have been done in the function calling <code>phyl.vcv</code> internally; however, I recently discovered that this had caused the function to be used incorrectly resulting in a paper correction. This is now fixed such that <code>X</code> is checked for row names and (if present) <code>C</code> is sorted to match the rows of <code>X</code>. Hopefully this does not cause any problems for other functions using <code>phyl.vcv</code>!
</p>


<h3>Value</h3>

<p>A list containing three elements, as follows: <code>C</code>, the matrix <code>vcv.phylo</code> transformed by <code>lambda</code>; <code>R</code>, the among trait variance-covariance matrix for the data in <code>X</code>; and <code>alpha</code>, a vector of ancestral states at the root node of the tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='phylANOVA'>Phylogenetic ANOVA and post-hoc tests</h2><span id='topic+phylANOVA'></span>

<h3>Description</h3>

<p>Simulation based phylogenetic ANOVA following Garland et al. (1993), with post-hoc tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylANOVA(tree, x, y, nsim=1000, posthoc=TRUE, p.adj="holm")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylANOVA_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="phylANOVA_+3A_x">x</code></td>
<td>
<p>a vector containing the groups.</p>
</td></tr>
<tr><td><code id="phylANOVA_+3A_y">y</code></td>
<td>
<p>a vector containing the response variable (continuously valued).</p>
</td></tr>
<tr><td><code id="phylANOVA_+3A_nsim">nsim</code></td>
<td>
<p>an integer specifying the number of simulations (including the observed data).</p>
</td></tr>
<tr><td><code id="phylANOVA_+3A_posthoc">posthoc</code></td>
<td>
<p>a logical value indicating whether or not to conduct posthoc tests to compare the mean among groups.</p>
</td></tr>
<tr><td><code id="phylANOVA_+3A_p.adj">p.adj</code></td>
<td>
<p>method to adjust P-values for the posthoc tests to account for multiple testing. Options same as <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs the simulation-based phylogenetic ANOVA of Garland et al. (1993) and (optionally) conducts all posthoc comparisons of means among groups (also obtaining the P-values by phylogenetic simulation).
</p>
<p>This function uses a little bit of code from both <code>phy.anova</code> in the <span class="pkg">geiger</span> package and <code><a href="stats.html#topic+pairwise.t.test">pairwise.t.test</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylANOVA"</code> containing the following elements:
</p>
<table>
<tr><td><code>F</code></td>
<td>
<p>F from observed data.</p>
</td></tr>
<tr><td><code>Pf</code></td>
<td>
<p>P-value for F from simulation.</p>
</td></tr>
<tr><td><code>T</code></td>
<td>
<p>matrix of t-values.</p>
</td></tr>
<tr><td><code>Pt</code></td>
<td>
<p>matrix of multiple test corrected P-values from posthoc t-tests.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Garland, T., Jr., A. W. Dickerman, C. M. Janis, &amp; J. A. Jones (1993) Phylogenetic analysis of covariance by computer simulation. <em>Systematic Biology</em>, <b>42</b>, 265-292.
</p>
<p>Harmon, L. J., J. T. Weir, C. D. Brock, R. E. Glor, W. Challenger (2008) GEIGER: investigating evolutionary radiations. <em>Bioinformatics</em>, 24, 129-131.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+anova">anova</a></code>, <code><a href="stats.html#topic+pairwise.t.test">pairwise.t.test</a></code>
</p>

<hr>
<h2 id='phylo.heatmap'>Creates a phylogenetic heat map</h2><span id='topic+phylo.heatmap'></span>

<h3>Description</h3>

<p>Multivariate phylogenetic <code><a href="stats.html#topic+heatmap">heatmap</a></code> plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo.heatmap(tree, X, fsize=1, colors=NULL, standardize=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo.heatmap_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="phylo.heatmap_+3A_x">X</code></td>
<td>
<p>a matrix containing data for multiple continuous characters in which <code>rownames</code> correspond to the tip labels of the tree.</p>
</td></tr>
<tr><td><code id="phylo.heatmap_+3A_fsize">fsize</code></td>
<td>
<p>an integer or vector of length 3 containing the font size for the tip labels, the trait labels, and the legend text. (If a single integer is supplied, then the value will be recycled.)</p>
</td></tr>
<tr><td><code id="phylo.heatmap_+3A_colors">colors</code></td>
<td>
<p>a vector of colors to be passed to <code><a href="graphics.html#topic+image">image</a></code>. Can be a function call (e.g., <code>heat.colors(n=</code> <code>200)[200:1]</code>).</p>
</td></tr>
<tr><td><code id="phylo.heatmap_+3A_standardize">standardize</code></td>
<td>
<p>a logical value indicating whether or not to standardize each column of <code>X</code> to have the same variance &amp; mean prior to analysis.</p>
</td></tr>
<tr><td><code id="phylo.heatmap_+3A_...">...</code></td>
<td>
<p>optional arguments. So far these include: <code>legend</code>, a logical value indicating whether or not to plot a figure legend (defaults to <code>legend=TRUE</code>); <code>labels</code>, a logical value indicating whether or not to plot trait labels (defaults to <code>labels=TRUE</code>); <code>split</code>, a numeric vector indicating the fraction of the horizontal dimension to use for the tree &amp; heatmap, respectively (defaults to <code>split=c(0.5,0.5)</code>); <code>xlim</code>, <code>ylim</code>, &amp; <code>mar</code>, defined as in <code><a href="graphics.html#topic+par">par</a></code>; and <code>ftype</code>, <code>lwd</code>, and <code>pts</code> as defined in <code><a href="#topic+plotSimmap">plotSimmap</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function creates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data
data(anoletree)
data(anole.data)
## compute phylogenetic residuals
anole.data&lt;-as.matrix(anole.data)
anole.resids&lt;-cbind(anole.data[,1],
  phyl.resid(anoletree,anole.data[,1,drop=FALSE],
    anole.data[,2:ncol(anole.data)])$resid)
colnames(anole.resids)[1]&lt;-"SVL"
## plot phylogenetic heatmap
phylo.heatmap(anoletree,anole.resids,
  split=c(0.7,0.3),fsize=c(0.4,0.8,0.8),
  standardize=TRUE,pts=FALSE)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='phylo.impute'>Phylogenetic imputation for multivariate continuous character data</h2><span id='topic+phylo.impute'></span>

<h3>Description</h3>

<p>This function performs phylogenetic multiple imputation using maximum likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo.impute(tree, X, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo.impute_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="phylo.impute_+3A_x">X</code></td>
<td>
<p>data matrix with species names as row labels. Missing data to be imputed should be coded <code>NA</code>.</p>
</td></tr>
<tr><td><code id="phylo.impute_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs phylogenetic imputation in which the evolution of the characters in <code>X</code> is assumed to have occurred by correlation multivariate Brownian motion.
</p>
<p>Missing values are imputed by maximizing their likelihood jointly with the parameters of the Brownian model. The function <code><a href="#topic+evol.vcv">evol.vcv</a></code> is used internally to compute the likelihood.
</p>
<p>Note that the <span class="pkg">Rphylopars</span> package also does phylogenetic imputation for multivariate trait data and it seems to be much faster.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo.impute"</code> consisting of a complete data frame with missing values imputed.
</p>
<p>Since optimization is performed numerically using likelihood, a summary of the optimization can be seen by evaluating <code>attr(object,"optim")</code>, in which <code>object</code> is of class <code>"phylo.impute"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+evol.vcv">evol.vcv</a></code>
</p>

<hr>
<h2 id='phylo.to.map'>Plot tree with tips linked to geographic coordinates</h2><span id='topic+phylo.to.map'></span><span id='topic+plot.phylo.to.map'></span>

<h3>Description</h3>

<p>Project a phylogeny on a geographic map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo.to.map(tree, coords, rotate=TRUE, ...)
## S3 method for class 'phylo.to.map'
plot(x, type=c("phylogram","direct"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo.to.map_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="phylo.to.map_+3A_coords">coords</code></td>
<td>
<p>a matrix containing the latitude (in column 1) and the longitude of all tip species in the tree. The row names should be the same as <code>tree$tip.label</code>; however, more than one set of coordinates per species can be supplied by duplicating some row names.</p>
</td></tr>
<tr><td><code id="phylo.to.map_+3A_rotate">rotate</code></td>
<td>
<p>a logical value indicating whether or not to rotate nodes of the tree to better match longitudinal positions.</p>
</td></tr>
<tr><td><code id="phylo.to.map_+3A_x">x</code></td>
<td>
<p>for <code>plot.phylo.to.map</code>, an object of class <code>"phylo.to.map"</code>.</p>
</td></tr>
<tr><td><code id="phylo.to.map_+3A_type">type</code></td>
<td>
<p>a string indicating whether to map the tips of the tree onto a geographic map from a square phylogram (<code>type="phylogram"</code>) or to project the tree directly onto the map (<code>type="direct"</code>).</p>
</td></tr>
<tr><td><code id="phylo.to.map_+3A_...">...</code></td>
<td>
<p>various optional arguments. For the function <code>phylo.to.map</code>, which first creates an object of the special class <code>"phylo.to.map"</code> and then (optionally) plots this object, arguments include: <code>database</code> and <code>regions</code> (see <code><a href="maps.html#topic+map">map</a></code>), as well as any arguments that should be passed to <code>plot.phylo.to.map</code> internally. For <code>phylo.to.map</code>, optional arguments <code>xlim</code> and <code>ylim</code>, which control the plot area for the map; <code>fsize</code> for the font size of plot labels and <code>ftype</code> for the font type (following <code><a href="#topic+plotSimmap">plotSimmap</a></code>; <code>split</code> which controls the proportion of vertical (or horizontal) space for the tree (first) and map, in a vector; <code>psize</code> the size of the plotted points on the map - or <code>cex.points</code>, a vector contain the size of the tip points and geographic coordinate points, respectively; <code>from.tip</code> a logical value indicating whether to plot the linking lines from the tips (if <code>TRUE</code>) or from the end of the tip label, the default; <code>colors</code>, a single value or a vector of colors for the points and the linking lines; <code>pch</code> a single value or a vector of point types; <code>lwd</code> and <code>lty</code> for the linking lines; and <code>pts</code> a logical value indicating whether or not to plot points at the tips of the tree. <code>mar</code> and <code>asp</code> are as in <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>phylo.to.map</code> creates an object of class <code>"phylo.to.map"</code> and (optionally) plots that object.
</p>
<p><code>plot.phylo.to.map</code> plots an object of class <code>"phylo.to.map"</code> in which the tips of the tree point to coordinates on a geographic map.
</p>


<h3>Value</h3>

<p><code>phylo.to.map</code> creates an object of class <code>"phylo.to.map"</code> and  (if <code>plot=TRUE</code>) plots a phylogeny projected onto a geographic map.
</p>
<p><code>plot.phylo.to.map</code> plots on object of class <code>"phylo.to.map"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## generally recommend using higher resolution map
## e.g., from mapdata package
data(tortoise.tree)
data(tortoise.geog)
tortoise.phymap&lt;-phylo.to.map(tortoise.tree,
  tortoise.geog,plot=FALSE,direction="rightwards",
  regions="Ecuador")
plot(tortoise.phymap,direction="rightwards",pts=FALSE,
  xlim=c(-92.25,-89.25),ylim=c(-1.8,0.75),ftype="i",
  fsize=0.8,lty="dashed",map.bg="lightgreen",
  colors="slategrey")
## reset margins
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='phylo.toBackbone'>Converts tree to backbone or vice versa</h2><span id='topic+phylo.toBackbone'></span><span id='topic+backbone.toPhylo'></span>

<h3>Description</h3>

<p>Converts between <code>"phylo"</code> and <code>"backbonePhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo.toBackbone(x, trans, ...)
backbone.toPhylo(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo.toBackbone_+3A_x">x</code></td>
<td>
<p>an object of class <code>"phylo"</code> (for the function <code>phylo.toBackbone</code>), or an object of the special class <code>"backbonePhylo"</code> (for <code>backbone.toPhylo</code>).</p>
</td></tr>
<tr><td><code id="phylo.toBackbone_+3A_trans">trans</code></td>
<td>
<p>data frame containing the attributes necessary to translate a backbone tree to an object of class <code>"backbonePhylo"</code>. The data frame should contain the following variables: <code>tip.label</code>: the tip labels in the input tree (not all need be included); <code>clade.label</code>: labels for the unobserved subtrees; <code>N</code>: number of species in each subtree; and <code>depth</code>: desired depth of each subtree. <code>depth</code> for each terminal taxon in <code>x</code> cannot be greater than the terminal edge length for that taxon.</p>
</td></tr>
<tr><td><code id="phylo.toBackbone_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either an object of class <code>"phylo"</code> or an object of class <code>"backbonePhylo"</code>, depending on the method.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.backbonePhylo">plot.backbonePhylo</a></code>
</p>

<hr>
<h2 id='phyloDesign'>Compute design matrix for least squares analyses</h2><span id='topic+phyloDesign'></span>

<h3>Description</h3>

<p>Primarily an internal function for <code><a href="#topic+optim.phylo.ls">optim.phylo.ls</a></code>, this function creates a design matrix for least squares phylogenetic analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloDesign(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloDesign_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a matrix containing the edges in the tree (in columns) and pairs of tip node numbers (in rows). Values in the matrix are either <code>1</code> if the edge is on the shortest path between the two tips; and <code>0</code> otherwise. Probably do not use unless you know what you're doing.
</p>


<h3>Value</h3>

<p>A matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='phylomorphospace'>Creates phylomorphospace plot</h2><span id='topic+phylomorphospace'></span><span id='topic+project.phylomorphospace'></span>

<h3>Description</h3>

<p>Project a phylogeny into morphospace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylomorphospace(tree, X, A=NULL, label=c("radial","horizontal","off"),
    control=list(), ...)
project.phylomorphospace(tree, X, nsteps=200, sleep=0,
    direction=c("to","from","both"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylomorphospace_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format, or a modified <code>"phylo"</code> object with a mapped discrete character.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_x">X</code></td>
<td>
<p>an <code>n</code> x 2 matrix of tip values for two characters in <code>n</code> species.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_a">A</code></td>
<td>
<p>an optional <code>m</code> x 2 matrix (for <code>m</code> nodes) of values for two traits at internal nodes in the tree - if not supplied, these values will be estimated using <code><a href="#topic+fastAnc">fastAnc</a></code>.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_label">label</code></td>
<td>
<p>string indicating whether to plot the tip labels in the same direction as the terminal edge (<code>label="radial"</code>), horizontally <code>label="horizontal"</code>, or not at all <code>"off"</code>. <code>label=TRUE</code> and <code>label=FALSE</code> are also acceptable, for compatibility with phytools &lt;= 0.3-03.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_control">control</code></td>
<td>
<p>a list containing the following optional control parameters: <code>col.edge</code>: a vector of edge colors; and <code>col.node</code>: a vector of node colors.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_nsteps">nsteps</code></td>
<td>
<p>for <code>project.phylomorphospace</code> the number of frames in the animation between the phylogeny &amp; the phylomorphospace or <em>vice versa</em>.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_sleep">sleep</code></td>
<td>
<p>for <code>project.phylomorphospace</code> the time between frames.</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_direction">direction</code></td>
<td>
<p>for <code>project.phylomorphospace</code> whether to morph <code>"to"</code> a phylomorphospace, <code>"from"</code> a phylomorphospace, or there &amp; back again (<code>"both"</code>).</p>
</td></tr>
<tr><td><code id="phylomorphospace_+3A_...">...</code></td>
<td>
<p>optional arguments for plotting, including <code>xlim</code>, <code>ylim</code>, <code>xlab</code>, <code>ylab</code>, <code>lwd</code>, <code>colors</code>, <code>fsize</code>, and <code>node.by.map</code>. <code>colors</code> is only used when there is a mapped discrete character on the tree, in which case <code>control$col.edge</code> is ignored. <code>fsize</code> is relative to the default, which is <code>textxy(...,cx=0.75)</code>. <code>node.by.map</code> is a logical value (defaults to <code>FALSE</code> which tells the function whether or not to plot the node colors using the colors of the mapped discrete character. Setting this option to <code>TRUE</code> will cause <code>control$col.node</code> to be ignored. <code>node.size</code> is a vector containing the point size relative to the default (see <code><a href="graphics.html#topic+par">par</a></code> for plotted internal nodes and tips, respectively. Defaults to <code>node.size=c(1,1.3)</code>. If only one number is provided it will be recycled. <code>axes</code> is a logical value indicating whether or not axes should be plotted (see <code><a href="graphics.html#topic+plot.default">plot.default</a></code>. Finally, <code>add</code> indicates whether to add the phylomorphospace to the current plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates a phylomorphospace plot (a projection of the tree into morphospace) for two characters following Sidlauskas (2008; <em>Evolution</em>). It will also plot a discrete character mapped on tree.
</p>
<p><code>project.phylomorphospace</code> animates the phylomorphospace projection.
</p>


<h3>Value</h3>

<p>This function creates a phylomorphospace plot
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-90.
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Sidlauskas, B. (2008) Continuous and arrested morphological diversification in sister clades of characiform fishes: A phylomorphospace approach. <em>Evolution</em>, <b>62</b>, 3135-3156.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load tree &amp; data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## set colors for mapped discrete character
cols&lt;-setNames(c("blue","red"),
    levels(sunfish.data$feeding.mode))
phylomorphospace(sunfish.tree,sunfish.data[,3:2],
    colors=cols,bty="l",ftype="off",node.by.map=TRUE,
    node.size=c(0,1.2),xlab="relative buccal length",
    ylab="relative gape width")
title(main="Phylomorphospace of buccal morphology in Centrarchidae",
    font.main=3)	
</code></pre>

<hr>
<h2 id='phylomorphospace3d'>Creates three-dimensional phylomorphospace plot</h2><span id='topic+phylomorphospace3d'></span>

<h3>Description</h3>

<p>Creates a phylomorphospace plot in three dimensions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylomorphospace3d(tree, X, A=NULL, label=TRUE, control=list(), 
   method=c("dynamic","static"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylomorphospace3d_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="phylomorphospace3d_+3A_x">X</code></td>
<td>
<p>an <code>n</code> x 3 matrix of tip values for two characters in <code>n</code> species.</p>
</td></tr>
<tr><td><code id="phylomorphospace3d_+3A_a">A</code></td>
<td>
<p>an optional <code>m</code> x 3 matrix (for <code>m</code> nodes) of values for two traits at internal nodes in the tree - if not supplied, these values will be estimated using <code><a href="#topic+anc.ML">anc.ML</a></code>.</p>
</td></tr>
<tr><td><code id="phylomorphospace3d_+3A_label">label</code></td>
<td>
<p>logical value indicating whether to print tip labels next to terminal nodes in the plot (presently doesn't do anything, but labels can be dropped using <code>control</code>).</p>
</td></tr>
<tr><td><code id="phylomorphospace3d_+3A_control">control</code></td>
<td>
<p>a list containing the following optional control parameters: <code>spin</code>: a logical value indicating whether to animate the plot when created; <code>axes</code>: a logical indicating whether to plot the axes; <code>box</code>: a logical value indicating whether to plot in box; <code>simple.axes</code>: logical value indicating whether to replace <code>box</code> and <code>axes</code> with simpler axes; <code>lwd</code>: line widths; <code>ftype</code>: font type (&quot;off&quot; turns off labels altogether); <code>col.edge</code> a vector of colors of length <code>nrow(tree$edge)</code>.</p>
</td></tr>
<tr><td><code id="phylomorphospace3d_+3A_method">method</code></td>
<td>
<p>a string either <code>"dynamic"</code> for a dynamic (animated) plot created using <span class="pkg">rgl</span>; or <code>"static"</code> for a flat 3D plot created using <code>scatterplot3d</code> and base graphics. The latter has the advantage of being very easy to export in standard format.</p>
</td></tr>
<tr><td><code id="phylomorphospace3d_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code>scatterplot3d</code>. Most options not available. <code>angle</code> is an important option that does work here.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates a phylomorphospace plot for three characters using the 3D visualization package, <span class="pkg">rgl</span> (if available) or statically, by simulating 3D on a flat graphical device.
</p>


<h3>Value</h3>

<p>This function creates a three dimensional phylomorphospace plot. The function returns a function from <code>spin3d</code> (for <code>method="dynamic"</code>); or a series of functions from <code><a href="scatterplot3d.html#topic+scatterplot3d">scatterplot3d</a></code> (for <code>method="static"</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Sidlauskas, B. (2008) Continuous and arrested morphological diversification in sister clades of characiform fishes: A phylomorphospace approach. <em>Evolution</em>, <b>62</b>, 3135-3156.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fancyTree">fancyTree</a></code>, <code><a href="#topic+phenogram">phenogram</a></code>, <code><a href="#topic+phylomorphospace">phylomorphospace</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anoletree)
data(anole.data)
anole.pca&lt;-phyl.pca(anoletree,anole.data)
## Not run: 
phylomorphospace3d(anoletree,scores(anole.pca)[,1:3],
    control=list(spin=FALSE))
## End(Not run)
par(cex=0.5)
phylomorphospace3d(anoletree,scores(anole.pca)[,1:3],
    method="static",angle=-30)
par(cex=1)
</code></pre>

<hr>
<h2 id='phylosig'>Compute phylogenetic signal with two methods</h2><span id='topic+phylosig'></span><span id='topic+plot.phylosig'></span>

<h3>Description</h3>

<p>Calculate phylogenetic signal using two different methods (Pagel, 1999; Blomberg et al., 2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylosig(tree, x, method="K", test=FALSE, nsim=1000, se=NULL, start=NULL,
   control=list(), niter=10)
## S3 method for class 'phylosig'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylosig_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_x">x</code></td>
<td>
<p>vector containing values for a single continuously distributed trait. In the case of the <code>plot</code> method, <code>x</code> is an object of class <code>"phylosig"</code>.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_method">method</code></td>
<td>
<p>method to compute signal: can be <code>"K"</code> or <code>"lambda"</code>.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_test">test</code></td>
<td>
<p>logical indicating whether or not to conduct a hypothesis test of <code>"K"</code> or <code>"lambda"</code>.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_nsim">nsim</code></td>
<td>
<p>for <code>method="K"</code>, number of simulations in randomization test.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_se">se</code></td>
<td>
<p>named vector containing the standard errors for each species.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_start">start</code></td>
<td>
<p>vector of starting values for optimization of (respectively) <code class="reqn">\sigma^2</code> and <code class="reqn">\lambda</code>. Only used in <code>method="lambda"</code> and <code>se!=NULL</code>.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_control">control</code></td>
<td>
<p>list of control parameters for multidimensional optimization, implemented in <code><a href="stats.html#topic+optim">optim</a></code>. Only used in <code>method="lambda"</code> and <code>se!=NULL</code>.</p>
</td></tr>
<tr><td><code id="phylosig_+3A_niter">niter</code></td>
<td>
<p>number of iterations for likelihood optimization of <code class="reqn">\lambda</code> (if <code>se!=NULL</code>), or the number of <em>intervals</em> between 0 and the maximum possible value of <code class="reqn">\lambda</code> for univariate optimization of <code class="reqn">\lambda</code> (if <code>se==NULL</code>).</p>
</td></tr>
<tr><td><code id="phylosig_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>plot</code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes phylogenetic signal using two different methods. It can also conduct the hypothesis tests for significant phylogenetic signal, and estimate phylogenetic signal incorporating sampling error following Ives et al. (2007).
</p>
<p><code class="reqn">\lambda</code> optimization is performed using <code><a href="stats.html#topic+optimize">optimize</a></code> with the range of <code class="reqn">\lambda</code> set between 0 and the theoretical upper limit of <code class="reqn">\lambda</code> (determined by the relative height of the most recent internal node on the tree).
</p>
<p><code>plot.phylosig</code> creates either a plot of the null distribution of <em>K</em> or a likelihood surface, depending on the value of <code>method</code>.
</p>


<h3>Value</h3>

<p>The function returns an object of class <code>"phylosig"</code>. With default arguments (<code>method="K"</code>, <code>test=FALSE</code>, and <code>se=NULL</code>), this will be a single numeric value. Otherwise, if <code>(method="K")</code>, it will consist of a list with up to the following elements:
</p>
<table>
<tr><td><code>K</code></td>
<td>
<p>value of the K-statistic.</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>rate of evolution, <code class="reqn">\sigma^2</code>, for estimation with sampling error.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood, for estimation with sampling error.</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>optionally, the P-value from the randomization test.</p>
</td></tr>
</table>
<p>If <code>(method="lambda")</code>, it will be a list with up to the following elements:
</p>
<table>
<tr><td><code>lambda</code></td>
<td>
<p>fitted value of <code class="reqn">\lambda</code>.</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>rate of evolution, <code class="reqn">\sigma^2</code>, for estimation with sampling error.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood.</p>
</td></tr>
<tr><td><code>logL0</code></td>
<td>
<p>log-likelihood for <code class="reqn">\lambda=0</code>.</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>P-value of the likelihood ratio test.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>value for convergence, for estimation with sampling error only. (See <code><a href="stats.html#topic+optim">optim</a></code>).</p>
</td></tr>
<tr><td><code>message</code></td>
<td>
<p>message from <code><a href="stats.html#topic+optim">optim</a></code>, for estimation with sampling error only.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Blomberg, S. P., T. Garland Jr., and A. R. Ives (2003) Testing for phylogenetic signal in comparative data: Behavioral traits are more labile. <em>Evolution</em>, <b>57</b>, 717-745.
</p>
<p>Ives, A. R., P. E. Midford, and T. Garland Jr. (2007) Within-species variation and measurement error in phylogenetic comparative biology. <em>Systematic Biology</em>, <b>56</b>, 252-270.
</p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. <em>Nature</em>, <b>401</b>, 877-884.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract characters of interest
ln.bodyMass&lt;-log(setNames(mammal.data$bodyMass,
    rownames(mammal.data)))
ln.homeRange&lt;-log(setNames(mammal.data$homeRange,
    rownames(mammal.data)))
## compute phylogenetic signal K
K.bodyMass&lt;-phylosig(mammal.tree,ln.bodyMass,
    test=TRUE)
print(K.bodyMass)
plot(K.bodyMass)
K.homeRange&lt;-phylosig(mammal.tree,ln.homeRange,
    test=TRUE)
print(K.homeRange)
plot(K.homeRange)
## compute phylogenetic signal lambda
lambda.bodyMass&lt;-phylosig(mammal.tree,ln.bodyMass,
    method="lambda",test=TRUE)
print(lambda.bodyMass)
plot(lambda.bodyMass)
lambda.homeRange&lt;-phylosig(mammal.tree,ln.homeRange,
    method="lambda",test=TRUE)
print(lambda.homeRange)
plot(lambda.homeRange)
</code></pre>

<hr>
<h2 id='plot.backbonePhylo'>Plots backbone tree with triangles as clades</h2><span id='topic+plot.backbonePhylo'></span>

<h3>Description</h3>

<p>Plots a backbone tree (stored as an object of class <code>"backbonePhylo"</code>) with triangles as subtrees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'backbonePhylo'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.backbonePhylo_+3A_x">x</code></td>
<td>
<p>an object of class <code>"backbonePhylo"</code>.</p>
</td></tr>
<tr><td><code id="plot.backbonePhylo_+3A_...">...</code></td>
<td>
<p>optional arguments. Includes <code>vscale</code> (to rescale the vertical dimension in plotting), <code>fixed.height</code> (logical value to fix the plotted height of subtree triangles), <code>print.clade.size</code> (logical), <code>fixed.n1</code> (logical value indicating whether or not to use the fixed triangle height for subtrees containing only one taxon, or to plot as a leaf - defaults to <code>FALSE</code>), and <code>col</code> (a single value, or a vector with names, giving the clade colors), as well as <code>xlim</code>, <code>ylim</code>, and <code>lwd</code> (as well as perhaps other standard plotting arguments).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylo.toBackbone">phylo.toBackbone</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## first create our backbone tree with
## random subtree diversities
tree&lt;-phytools:::lambdaTree(pbtree(n=10),lambda=0.5)
## create a translation table
## leaving a couple of single-taxon clades for fun
tip.label&lt;-sample(tree$tip.label,8)
clade.label&lt;-LETTERS[1:8]
N&lt;-ceiling(runif(n=8,min=1,max=20))
## set crown node depth to 1/2 the maximum depth
depth&lt;-sapply(tip.label,function(x,y) 
    0.5*y$edge.length[which(tree$edge[,2]==
    which(y$tip.label==x))],y=tree)
trans&lt;-data.frame(tip.label,clade.label,N,depth)
rownames(trans)&lt;-NULL
## here's what trans looks like
print(trans)
## convert
obj&lt;-phylo.toBackbone(tree,trans)
## plot
plot(obj)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset par
</code></pre>

<hr>
<h2 id='plotBranchbyTrait'>Plot branch colors by a quantitative trait or value</h2><span id='topic+plotBranchbyTrait'></span>

<h3>Description</h3>

<p>Function plots a tree with branches colored by the value for a quantitative trait or probability, by various methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBranchbyTrait(tree, x, mode=c("edges","tips","nodes"), palette="rainbow", 
   legend=TRUE, xlims=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBranchbyTrait_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="plotBranchbyTrait_+3A_x">x</code></td>
<td>
<p>either a vector of states for the edges, tips, or nodes of the tree (for <code>mode="edges"</code>, <code>"tips"</code>, and <code>"nodes"</code>, respectively).</p>
</td></tr>
<tr><td><code id="plotBranchbyTrait_+3A_mode">mode</code></td>
<td>
<p>string indicating plotting mode. <code>mode="edges"</code>, the default, requires that the mapping state of each edge in the tree should be provided. <code>mode="tips"</code> takes the tip values and estimates the state at each internal node. The mapped character value along each branch is the average of the nodes subtending that branch. <code>mode="nodes"</code> similar to <code>"tips"</code>, except that the node values are provided instead of estimated.</p>
</td></tr>
<tr><td><code id="plotBranchbyTrait_+3A_palette">palette</code></td>
<td>
<p>color palette to translate character values to color. Options are presently <code>"rainbow"</code> (the default), <code>"heat.colors"</code>, and <code>"gray"</code>. <code>palette</code> can also be a function produced by <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.</p>
</td></tr>
<tr><td><code id="plotBranchbyTrait_+3A_legend">legend</code></td>
<td>
<p>can be a logical value (<code>TRUE</code> or <code>FALSE</code>) or a numeric value greater than 0. In the latter case the numeric value gives the length of the plotted legend, which also acts as a scale bar for the branch lengths of the tree.</p>
</td></tr>
<tr><td><code id="plotBranchbyTrait_+3A_xlims">xlims</code></td>
<td>
<p>range for the translation map between trait values and the color map. Should be inclusive of all the values in <code>x</code>.</p>
</td></tr>
<tr><td><code id="plotBranchbyTrait_+3A_...">...</code></td>
<td>
<p>other optional arguments to be passed to <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> - pretty much all arguments are available. In addition, there <code>plotBranchbyTrait</code> has the following additional optional arguments: <code>tol</code> a small tolerance value to be added to the range of <code>x</code>; <code>prompt</code> for <code>legend=TRUE</code>, a logical value indicating whether to prompt for the position of the legend (or not) - the default is to put the legend in the lower left hand size of the plot; <code>title</code> for <code>legend=TRUE</code>, the title of the legend; and <code>digits</code> for <code>legend=TRUE</code>, the number of digits in the quantitative scale of the legend. Finally, <code>cex</code> can be supplied as either a single numeric value, or as a vector of two different values. If the latter is true than the second element of <code>cex</code> will be passed internally to the function <code><a href="#topic+add.color.bar">add.color.bar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Unlike most other tree plotting functions in <span class="pkg">phytools</span>, this function calls <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> (not <code>plotSimmap</code>) internally.
</p>
<p>Note that if <code>prompt=TRUE</code>, the function will prompt for the position of the legend.
</p>


<h3>Value</h3>

<p>Plots a phylogeny.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='plotSimmap'>Plot stochastic character mapped tree</h2><span id='topic+plotSimmap'></span><span id='topic+plot.simmap'></span><span id='topic+plot.multiSimmap'></span>

<h3>Description</h3>

<p>Plots one or multiple stochastic character mapped trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSimmap(tree, colors=NULL, fsize=1.0, ftype="reg", lwd=2, pts=FALSE, 
   node.numbers=FALSE, mar=NULL, add=FALSE, offset=NULL,
   direction="rightwards", type="phylogram", setEnv=TRUE, 
   part=if(type=="arc") 0.5 else 1.0, xlim=NULL, ylim=NULL, 
   nodes="intermediate", tips=NULL, maxY=NULL, hold=TRUE, 
   split.vertical=FALSE, lend=2, asp=NA, outline=FALSE, 
   plot=TRUE, underscore=FALSE, arc_height=2)
## S3 method for class 'simmap'
plot(x, ...)
## S3 method for class 'multiSimmap'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSimmap_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code> or <code>"multiSimmap"</code> containing a stochastic mapping or set of mappings (e.g., see <code><a href="#topic+read.simmap">read.simmap</a></code> &amp; <code><a href="#topic+make.simmap">make.simmap</a></code>).</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_colors">colors</code></td>
<td>
<p>a vector with names translating the mapped states to colors - see <code>Examples</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_fsize">fsize</code></td>
<td>
<p>relative font size for tip labels.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_ftype">ftype</code></td>
<td>
<p>font type - options are <code>"reg"</code>, <code>"i"</code> (italics), <code>"b"</code> (bold), or <code>"bi"</code> (bold-italics).</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_lwd">lwd</code></td>
<td>
<p>line width for plotting.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_pts">pts</code></td>
<td>
<p>logical value indicating whether or not to plot filled circles at each vertex of the tree, as well as at transition points between mapped states.  Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_node.numbers">node.numbers</code></td>
<td>
<p>a logical value indicating whether or not node numbers should be plotted.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_mar">mar</code></td>
<td>
<p>vector containing the margins for the plot to be passed to <code><a href="graphics.html#topic+par">par</a></code>. If not specified, the default margins are [0.1,0.1,0.1,0.1].</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_add">add</code></td>
<td>
<p>a logical value indicating whether or not to add the plotted tree to the current plot (<code>TRUE</code>) or create a new plot (<code>FALSE</code>, the default).</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_offset">offset</code></td>
<td>
<p>offset for the tip labels in character widths.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_direction">direction</code></td>
<td>
<p>plotting direction. Options are <code>"rightwards"</code> (the default), <code>"leftwards"</code>, <code>"upwards"</code> or <code>"downwards"</code>. For <code>method="fan"</code> <code>direction</code> is ignored.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_type">type</code></td>
<td>
<p>plot type. Can be <code>"phylogram"</code>, <code>"fan"</code>, or <code>"cladogram"</code>. Only a subset of options are presently available for <code>type="fan"</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_setenv">setEnv</code></td>
<td>
<p>logical value indicating whether or not to set the environment <code>.PlotPhyloEnv</code>. Setting this to <code>TRUE</code> (the default) will allow compatibility with <span class="pkg">ape</span> labeling functions such as <code><a href="ape.html#topic+nodelabels">nodelabels</a></code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_part">part</code></td>
<td>
<p>value between 0 and 1 for <code>type="fan"</code> indicating what fraction of the full circular tree to use as plotting area. For instance, <code>part=0.5</code> will plot a half fan phylogeny. It also affects the axis scaling used.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_xlim">xlim</code></td>
<td>
<p>x-limits for the plot.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_ylim">ylim</code></td>
<td>
<p>y-limits for the plot.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_nodes">nodes</code></td>
<td>
<p>node placement following Felsenstein (2004; pp. 574-576). Can be <code>"intermediate"</code>, <code>"centered"</code>, <code>"weighted"</code>, or <code>"inner"</code>. So far only works for <code>type="phylogram"</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_tips">tips</code></td>
<td>
<p>labeled vector containing the vertical position of tips. Normally this will be <code>1:N</code> for <code>N</code> tips in the tree.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_maxy">maxY</code></td>
<td>
<p>maximum value of y to use before rotating a tree into fan configuration. This will only make a difference if different from <code>Ntip(tree)</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_hold">hold</code></td>
<td>
<p>logical argument indicating whether or not to hold the output to the graphical device before plotting. Defaults to <code>hold=TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_split.vertical">split.vertical</code></td>
<td>
<p>split the color of the vertically plotted edges by the state of the daughter edges. Only applies if the edge state changes exactly at a node.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_lend">lend</code></td>
<td>
<p>line end style. See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_asp">asp</code></td>
<td>
<p>aspect ratio. See <code><a href="graphics.html#topic+plot.window">plot.window</a></code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_outline">outline</code></td>
<td>
<p>logical value indicating whether or not to draw a black outline around the plotted edges of the tree.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether or not to actually plot the tree. (See equivalent argument in <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>.)</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_underscore">underscore</code></td>
<td>
<p>logical value indicating whether to plot the underscore character, <code>"_"</code> (if <code>underscore=TRUE</code>) or substitute for a space. Defaults to <code>underscore=FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_arc_height">arc_height</code></td>
<td>
<p>for <code>type="arc"</code> trees, the height of the arc in units of total tree depth.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_x">x</code></td>
<td>
<p>for S3 plotting method, object of class <code>"simmap"</code> or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="plotSimmap_+3A_...">...</code></td>
<td>
<p>for S3 plotting method, other arguments to be passed to <code><a href="#topic+plotSimmap">plotSimmap</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>Felsenstein, J. (2004) <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Huelsenbeck, J. P., R. Neilsen, and J. P. Bollback (2003) Stochastic mapping of morphological characters. <em>Systematic Biology</em>, <b>52</b>, 131-138. 
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anoletree)
cols&lt;-setNames(c("green","#E4D96F","darkgreen",
    "brown","black","darkgrey"),
    c("CG","GB","TC","TG","Tr","Tw"))
plot(anoletree,cols,fsize=0.5,ftype="i",outline=TRUE,
    lwd=3,ylim=c(0,Ntip(anoletree)),
    mar=c(0.1,0.1,1.1,0.1))
add.simmap.legend(colors=cols,prompt=FALSE,x=0,y=-0.5,
    vertical=FALSE)
title(main="Caribbean ecomorphs of anoles",font.main=3,
    line=0)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='plotThresh'>Tree plotting with posterior probabilities of ancestral states from the threshold model</h2><span id='topic+plotThresh'></span>

<h3>Description</h3>

<p>Plots estimated posterior probabilities at nodes under the threshold model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotThresh(tree, x, mcmc, burnin=NULL, piecol, tipcol="input", legend=TRUE, 
   ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotThresh_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_x">x</code></td>
<td>
<p>a named vector containing discrete character states; or a matrix containing the tip species, in rows, and probabilities of being in each state, in columns.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_mcmc">mcmc</code></td>
<td>
<p>list object returned by <code><a href="#topic+ancThresh">ancThresh</a></code>.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_burnin">burnin</code></td>
<td>
<p>number of generations (not samples) to exclude as burn in; if <code>NULL</code> then 20% of generations are excluded as burn-in.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_piecol">piecol</code></td>
<td>
<p>a named vector containing the colors for the posterior probabilities plotted as pie charts at internal nodes.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_tipcol">tipcol</code></td>
<td>
<p>a string indicating  whether the tip colors should be based on the input data (<code>"input"</code>) or sampled tip liabilities (<code>"estimated"</code>). These will only differ if there is uncertainty in the tip states.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_legend">legend</code></td>
<td>
<p>logical value or text to be plotted in the legend.</p>
</td></tr>
<tr><td><code id="plotThresh_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> - <code>label.offset</code> should be &gt;0 so that tip labels and species names do not overlap.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the object returned by <code><a href="#topic+ancThresh">ancThresh</a></code> to plot the posterior probabilities of ancestral states under the threshold model.
</p>
<p>It is also called internally by <code><a href="#topic+ancThresh">ancThresh</a></code>.
</p>


<h3>Value</h3>

<p>Plots a tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ancThresh">ancThresh</a></code>, <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>
</p>

<hr>
<h2 id='plotTree'>Plots rooted phylogenetic tree</h2><span id='topic+plotTree'></span>

<h3>Description</h3>

<p>Essentially a wrapper for <code><a href="#topic+plotSimmap">plotSimmap</a></code>. Arguments in <code>...</code> are passed to <code><a href="#topic+plotSimmap">plotSimmap</a></code>, with the exception of optional argument <code>color</code> which is used to determine the plotted color of the branch lengths of the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTree(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTree_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format; or multiple trees as an object of class <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a rooted phylogram or cladogram.
</p>


<h3>Value</h3>

<p>This function plots a rooted phylogram or cladogram.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(vertebrate.tree)
plotTree(vertebrate.tree,fsize=1.2,ftype="i")
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='plotTree.datamatrix'>Plot a tree with a discrete (or continuous) character data matrix at the tips</h2><span id='topic+plotTree.datamatrix'></span><span id='topic+plotFanTree.wTraits'></span>

<h3>Description</h3>

<p><code>plotTree.datamatrix</code> plots a phylogeny next to a matrix of discrete characters. <code>plotFanTree.wTraits</code> plots an arc or fan style tree with discrete or continuous data at the tips.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTree.datamatrix(tree, X, ...)
plotFanTree.wTraits(tree, X, type=c("arc","fan"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTree.datamatrix_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="plotTree.datamatrix_+3A_x">X</code></td>
<td>
<p>a data frame with columns as factors for <code>plotTree.datamatrix</code>, or a matrix of data frame containing continuous and/or discrete characters for <code>plotFanTree.wTraits</code>.</p>
</td></tr>
<tr><td><code id="plotTree.datamatrix_+3A_type">type</code></td>
<td>
<p>plot type for <code>plotFanTree.wTraits</code>.</p>
</td></tr>
<tr><td><code id="plotTree.datamatrix_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>plotTree.datamatrix</code> invisibly returns a list containing the font size, a list of the colors used for each column of the plotted data matrix, and the x-coordinate of the rightmost edge of the matrix.
</p>
<p><code>plotFanTree.wTraits</code> invisibly returns the color palette used for plotting.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dotTree">dotTree</a></code>, <code><a href="#topic+phylo.heatmap">phylo.heatmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("liolaemid.data")
data("liolaemid.tree")
liolaemid.data[liolaemid.tree$tip.label,]-&gt;liolaemid.data
colors&lt;-list(
  c("blue","white","red"),
  terrain.colors(n=10),
  setNames(c("#F0EAD6","#DF536B"),c("O","V")))
cols&lt;-plotFanTree.wTraits(liolaemid.tree,
  liolaemid.data[,3:1],lwd=12,colors=colors,ftype="off",
  spacer=0)
legend(x=0,y=0.7*max(nodeHeights(liolaemid.tree)),
  names(colors[[3]]),lwd=8,col=colors[[3]],
  title="parity mode",bty="n",xjust=0.5,yjust=0.5)
add.color.bar(1.5*max(nodeHeights(liolaemid.tree)),cols[[2]],
  title="maximum altitude (m)",
  lims=range(liolaemid.data[,2]),digits=2,prompt=FALSE,
  x=-0.75*max(nodeHeights(liolaemid.tree)),
  y=0.2*max(nodeHeights(liolaemid.tree)),subtitle="",
  lwd=8,outline=FALSE)
add.color.bar(1.5*max(nodeHeights(liolaemid.tree)),cols[[1]],
  title="environmental temp.",
  lims=range(liolaemid.data[,3]),digits=2,prompt=FALSE,
  x=-0.75*max(nodeHeights(liolaemid.tree)),
  y=-0.15*max(nodeHeights(liolaemid.tree)),subtitle="",
  lwd=8,outline=FALSE)
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margins to default
</code></pre>

<hr>
<h2 id='plotTree.errorbars'>Plot a tree with error bars around divergence dates</h2><span id='topic+plotTree.errorbars'></span>

<h3>Description</h3>

<p>Plots a tree with error bars around divergence times (nodes).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTree.errorbars(tree, CI, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTree.errorbars_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="plotTree.errorbars_+3A_ci">CI</code></td>
<td>
<p>confidence intervals around internal nodes of the tree, measured in time since the present.</p>
</td></tr>
<tr><td><code id="plotTree.errorbars_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The matrix CI show contain (in rows) the lower &amp; upper confidence bounds in time since the present.
</p>
<p>Optional arguments specific to the error bar plot include <code>gridlines</code>, <code>bar.lwd</code>,<code>cex</code> (for the points plotted at nodes), and <code>bar.col</code>.
</p>


<h3>Value</h3>

<p>Plots a tree with error bars around internal nodes..
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ltt">ltt</a></code>, <code><a href="#topic+plotTree">plotTree</a></code>
</p>

<hr>
<h2 id='plotTree.lollipop'>Plot a tree with one or more matched lollipop plots</h2><span id='topic+plotTree.lollipop'></span>

<h3>Description</h3>

<p>Plots an upward facing phylogenetic tree with one or more matched lollipop plots. Inspired by a plotting style used by Title et al. (2024).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTree.lollipop(tree, x, args.plotTree=list(), args.lollipop=list(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTree.lollipop_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="plotTree.lollipop_+3A_x">x</code></td>
<td>
<p>a matrix, data frame, or vector of numeric values.</p>
</td></tr>
<tr><td><code id="plotTree.lollipop_+3A_args.plottree">args.plotTree</code></td>
<td>
<p>list of arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code>.</p>
</td></tr>
<tr><td><code id="plotTree.lollipop_+3A_args.lollipop">args.lollipop</code></td>
<td>
<p>list of arguments for the lollipop plot(s).</p>
</td></tr>
<tr><td><code id="plotTree.lollipop_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Title, P. O., Singhal, S., Grundler, M. C., Costa, G. C., Pyron, R. A., Colston, T. J., Grundler, M. R., Prates, I., Stepanova, N., Jones, M. E. H., Cavalcanti, L. B. Q., Colli, G. R., Di-Poi, N., Donnellan, S. C., Moritz, C., Mesquita, D. O., Pianka, E. R., Smith, S. A., Vitt, L. J., and Rabosky, D. L. (2024) The macroevolutionary singularity of snakes. <em>Science</em>, <b>383</b>, 918-923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylo.heatmap">phylo.heatmap</a></code>, <code><a href="#topic+plotTree.barplot">plotTree.barplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anoletree)
anole_tree&lt;-as.phylo(anoletree)
data(anole.data)
anole_data&lt;-cbind(phyl.resid(anole_tree,
  x=as.matrix(anole.data[,"SVL",drop=FALSE]),
  Y=as.matrix(anole.data[,c(6,4,2)]))$resid,
  exp(anole.data[,"SVL",drop=FALSE]))
plotTree.lollipop(anole_tree,anole_data,
  ylab=c("relative (TL)","relative (FLL)",
    "relative (HL)","SVL"))
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='plotTree.wBars'>Plot a tree with bars at the tips</h2><span id='topic+plotTree.wBars'></span><span id='topic+plotTree.barplot'></span><span id='topic+plotTree.boxplot'></span>

<h3>Description</h3>

<p>Plots a phylogenetic tree with adjacent boxplot or barplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTree.wBars(tree, x, scale=NULL, width=NULL, type="phylogram", 
    method="plotTree", tip.labels=FALSE, col="grey", border=NULL, 
    ...)
plotTree.barplot(tree, x, args.plotTree=list(), args.barplot=list(), 
    ...)
plotTree.boxplot(tree, x, args.plotTree=list(), args.boxplot=list(),
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTree.wBars_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_x">x</code></td>
<td>
<p>a named vector or matrix of trait values. For <code>plotTree.boxplot</code>, the names should repeat for multiple observations per species. For <code>plotTree.boxplot</code> <code>x</code> can also be supplied as a formula, though in that case the factor levels need to be provided in a valid cladewise order of the tips in <code>tree</code>. This order doesn't need to correspond with the current order of the tip labels. For <code>plotTree.barplot</code> <code>x</code> can be a matrix (or a data frame) in which columns are the values of multiple traits to be simultaneously plotted on the tree.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_scale">scale</code></td>
<td>
<p>scaling factor for the tip bars (relative to the total tree height). If left as <code>NULL</code> a reasonable scaling factor is computed automatically.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_width">width</code></td>
<td>
<p>width of the tip bars.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_type">type</code></td>
<td>
<p>plot type. Can be <code>"phylogram"</code> or <code>"fan"</code>.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_method">method</code></td>
<td>
<p>plotting method to use. Can be <code>"plotTree"</code> (for <code><a href="#topic+plotTree">plotTree</a></code>) or <code>"plotSimmap"</code> (for <code><a href="#topic+plotSimmap">plotSimmap</a></code>).</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_tip.labels">tip.labels</code></td>
<td>
<p>argument indicating whether or not tip labels should be plotted. Defaults to <code>tip.labels=FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_col">col</code></td>
<td>
<p>colors of the plotted bars. Can be a single value or a vector with length equal to the number of tips in the tree.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_border">border</code></td>
<td>
<p>single value specifying the color of the border for the plotted bars. Defaults to <code>border=NULL</code>, which means that black borders will be plotted.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_args.plottree">args.plotTree</code></td>
<td>
<p>in <code>plotTree.barplot</code>, arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code>.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_args.barplot">args.barplot</code></td>
<td>
<p>in <code>plotTree.barplot</code>, arguments to be passed to <code><a href="graphics.html#topic+barplot">barplot</a></code>.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_args.boxplot">args.boxplot</code></td>
<td>
<p>in <code>plotTree.boxplot</code>, arguments to be passed to <code><a href="graphics.html#topic+boxplot">boxplot</a></code>.</p>
</td></tr>
<tr><td><code id="plotTree.wBars_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code> or <code><a href="#topic+plotSimmap">plotSimmap</a></code> in the case of <code>plotTree.wBars</code>. For <code>plotTree.barplot</code>, the only optional arguments are <code>add</code> and <code>ylim</code>. Generally <code>add</code> should not be used; however it can be employed to tell the function to draw the tree &amp; barplot, respectively, in the next two open plotting devices - rather than creating a table of figures in the current plotting device. <code>ylim</code> (which is also an optional argument for <code>plotTree.boxplot</code> should be supplied here rather than using <code>args.plotTree</code>, <code>args.boxplot</code>, or <code>args.barplot</code> because <em>y</em> axis limits must match exactly between the two plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plotTree.wbars</code> plots a phylogeny in phylogram or fan style with bars at the tips representing the values for a phenotypic trait.
</p>
<p><code>plotTree.barplot</code> creates a split plot in which a right-facing phylogram is on the left, and a bar plot is shown on the right.
</p>
<p><code>plotTree.boxplot</code> creates a split plot in which a right-facing phylogram is on the left, and a box plot is shown on the right.
</p>


<h3>Value</h3>

<p>Plots a tree with an associated bar plot for a continuously valued character at the tips.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+barplot">barplot</a></code>, <code><a href="#topic+dotTree">dotTree</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>, <code><a href="#topic+plotTree">plotTree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data from Mahler et al. (2010)
data(anoletree)
data(anole.data)
## extract overall body size (SVL)
svl&lt;-setNames(anole.data$SVL,rownames(anole.data))
## plotTree.wBars
plotTree.wBars(anoletree,svl,type="fan",scal=0.5)
par(mar=c(5.1,4.1,4.1,2.1))
## plotTree.barplot
plotTree.barplot(anoletree,exp(svl),
    args.plotTree=list(fsize=0.5),
    args.barplot=list(xlab="SVL (mm)"))
	
## load vertebrate tree and data
data(vertebrate.tree)
data(vertebrate.data)
## plotTree.barplot
options(scipen=4) ## change sci-notation
par(cex.axis=0.8)
plotTree.barplot(vertebrate.tree,
    setNames(vertebrate.data$Mass,
    rownames(vertebrate.data)),
    args.barplot=list(
    log="x",
    xlab="mass (kg)",
    xlim=c(0.01,500000),
    col=palette()[4]))
options(scipen=0)

## reset par to defaults
par(mfrow=c(1,1),mar=c(5.1,4.1,4.1,2.1),cex.axis=1)
</code></pre>

<hr>
<h2 id='posterior.evolrate'>Analysis of the posterior sample from evol.rate.mcmc</h2><span id='topic+posterior.evolrate'></span>

<h3>Description</h3>

<p>Analyzes posterior sample from <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posterior.evolrate(tree, ave.shift, mcmc, tips, showTree=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="posterior.evolrate_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="posterior.evolrate_+3A_ave.shift">ave.shift</code></td>
<td>
<p>mean or median shift-point from the posterior sample (see <code><a href="#topic+minSplit">minSplit</a></code>.</p>
</td></tr>
<tr><td><code id="posterior.evolrate_+3A_mcmc">mcmc</code></td>
<td>
<p>matrix <code>$mcmc</code> from <code>evol.rate.mcmc</code> (probably with burn-in excluded).</p>
</td></tr>
<tr><td><code id="posterior.evolrate_+3A_tips">tips</code></td>
<td>
<p>list of tips in state <code class="reqn">\sigma_1^2</code> for each sampled generation of MCMC.</p>
</td></tr>
<tr><td><code id="posterior.evolrate_+3A_showtree">showTree</code></td>
<td>
<p>optional logical value indicating whether or not to plot the stretched and shrunken tree generated by the pre-processing algorithm implemented in this function (default is <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a phylogenetic tree, an average split position, and a raw MCMC output from <code>evol.rate.mcmc</code> and returns a posterior sample of evolutionary rates rootward (<code class="reqn">\sigma_1^2</code>) and tipward (<code class="reqn">\sigma_2^2</code>) from the average split.
</p>


<h3>Value</h3>

<p>A matrix containing the posterior sample of evolutionary rates and shift-points between rates.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., D. L. Mahler, P. Peres-Neto, and B. D. Redelings (2012) A new method for identifying exceptional phenotypic diversification. <em>Evolution</em>, <b>66</b>, 135-146.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>, <code><a href="#topic+minSplit">minSplit</a></code>
</p>

<hr>
<h2 id='posthoc'>Generic post-hoc test</h2><span id='topic+posthoc'></span>

<h3>Description</h3>

<p>Conducts posthoc test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="posthoc_+3A_x">x</code></td>
<td>
<p>an object on which to conduct a post-hoc test.</p>
</td></tr>
<tr><td><code id="posthoc_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>So far is only implemented for object class <code>"ratebytree"</code>.
</p>


<h3>Value</h3>

<p>An object of the appropriate class containing the results of a posthoc test.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ratebytree">ratebytree</a></code>
</p>

<hr>
<h2 id='print.backbonePhylo'>Print method for backbone phylogeny</h2><span id='topic+print.backbonePhylo'></span>

<h3>Description</h3>

<p>Print method for an object of class <code>"backbonePhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'backbonePhylo'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.backbonePhylo_+3A_x">x</code></td>
<td>
<p>an object of class <code>"backbonePhylo"</code>.</p>
</td></tr>
<tr><td><code id="print.backbonePhylo_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints to screen.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylo.toBackbone">phylo.toBackbone</a></code>
</p>

<hr>
<h2 id='pscore'>Compute the parsimony score</h2><span id='topic+pscore'></span>

<h3>Description</h3>

<p>Calculates the parsimony score using the Fitch algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pscore(tree, x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pscore_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="pscore_+3A_x">x</code></td>
<td>
<p>vector (e.g., factor vector), matrix, or data frame. Should contain names or row names.</p>
</td></tr>
<tr><td><code id="pscore_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mostly for diagnostic purposes. Users interested in using Maximum Parsimony for phylogeny inference or ancestral state reconstruction should refer to the <span class="pkg">phangorn</span> package.
</p>


<h3>Value</h3>

<p>A numerical value or vector of values.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (2004) <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitMk">fitMk</a></code>, <code><a href="#topic+sim.Mk">sim.Mk</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load tree and data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## extract discrete character (feeding mode)
fmode&lt;-setNames(sunfish.data$feeding.mode,
    rownames(sunfish.data))
## compute the parsimony score
pscore(sunfish.tree,fmode)
</code></pre>

<hr>
<h2 id='ratebystate'>Method for investigating the rate of one trait as a function of the state of another</h2><span id='topic+ratebystate'></span>

<h3>Description</h3>

<p>Statistical test of whether the rate of a continuous character might be influenced by the state of another.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ratebystate(tree, x, y, nsim=100, corr=c("pearson","spearman"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ratebystate_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
<tr><td><code id="ratebystate_+3A_x">x</code></td>
<td>
<p>a continuous character - the dependent variable in the model.</p>
</td></tr>
<tr><td><code id="ratebystate_+3A_y">y</code></td>
<td>
<p>a second continuous trait - the response variable.</p>
</td></tr>
<tr><td><code id="ratebystate_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations for hypothesis testing.</p>
</td></tr>
<tr><td><code id="ratebystate_+3A_corr">corr</code></td>
<td>
<p>correlation method to use. Same as in <code><a href="stats.html#topic+cor">cor</a></code>.</p>
</td></tr>
<tr><td><code id="ratebystate_+3A_...">...</code></td>
<td>
<p>optional arguments which include <code>sim.method</code> (<code>"fastBM"</code> or <code>"sim.corrs"</code>; see <code><a href="#topic+fastBM">fastBM</a></code> and <code><a href="#topic+sim.corrs">sim.corrs</a></code>); <code>method</code> (<code>"by.node"</code> or <code>"by.branch"</code> indicating whether to assume the rate varies as a function of the node state or the mean branch state); <code>message</code> - a logical value indicating whether or not to return <code>corr</code> and <code>method</code>; finally <code>logarithm</code> - indicating whether or not to fit a model in which the variance of Brownian evolution in <code>y</code> changes as a multiplicative function of <code>x</code>. The default is <code>logarithm=FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function attempts to ask if the rate of a continuous character, <code>y</code>, depends on the state of a separate continuous trait, <code>x</code>. This is accomplished by regressing the squared contrasts in <code>y</code> on the branch or node ancestral estimates of <code>x</code>.
</p>


<h3>Value</h3>

<p>This function returns an object of class <code>"ratebystate"</code> with up to the following four elements:
</p>
<table>
<tr><td><code>beta</code></td>
<td>
<p>value of the regression coefficient for square of the contrasts in <code>y</code> regressed on the ancestral or branch-wise estimated states for <code>x</code>.</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>correlation coefficient for <code>corr=corr</code>.</p>
</td></tr>
<tr><td><code>corr</code></td>
<td>
<p>string giving the value of <code>corr</code>.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>string giving the value of <code>method</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastAnc">fastAnc</a></code>, <code><a href="ape.html#topic+pic">pic</a></code>
</p>

<hr>
<h2 id='ratebytree'>Likelihood test for rate variation among trees, clades, or traits</h2><span id='topic+ratebytree'></span><span id='topic+posthoc.ratebytree'></span>

<h3>Description</h3>

<p>Multiple methods for comparing the rate or process of evolution between trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ratebytree(trees, x, ...)
## S3 method for class 'ratebytree'
posthoc(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ratebytree_+3A_trees">trees</code></td>
<td>
<p>an object of class <code>"multiPhylo"</code>. If <code>x</code> consists of a list of different traits to be compared, then <code>trees</code> could also be a simple set of duplicates of the same tree, e.g., <code>rep(tree,length(x))</code>.</p>
</td></tr>
<tr><td><code id="ratebytree_+3A_x">x</code></td>
<td>
<p>a list of trait vectors for a continuous trait in which the names of each vectors correspond to the tip labels of <code>trees</code>. This is not used if <code>type="diversification"</code>. In the case of <code>posthoc.ratebytree</code>, an object of class <code>"ratebytree"</code>.</p>
</td></tr>
<tr><td><code id="ratebytree_+3A_...">...</code></td>
<td>
<p>optional arguments, including the argument <code>type</code> (<code>"continuous"</code>, <code>"discrete"</code>, or <code>"diversification"</code>), which, if not specified, the function will attempt to ascertain. See <code>Details</code> for more information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function essentially implements three different methods for comparing the rate or process of evolution between trees: one for continuously-valued traits, a second for discrete characters, and a third for the rate of diversification (speciation &amp; extinction).
</p>
<p>In all cases, the function takes an object of class <code>"multiPhylo"</code> containing two or more phylogenies (<code>trees</code>), and (for the first two analyses) a list of trait vectors (<code>x</code>).
</p>
<p>For continuous traits, the function then proceeds to fit two models: one in which the rate (or regime, for models <code>"OU"</code> and <code>"EB"</code>) of trait evolution is equal among all trees; and a second in which the rates or regimes can differ between trees.
</p>
<p>The latter model corresponds to an extension of the <em>censored</em> approach of O'Meara et al. (2006; Revell et al. 2018) and should also be related to the method of Adams (2012) for comparing rates among traits. See <code><a href="#topic+brownie.lite">brownie.lite</a></code> for a different implementation of the <em>noncensored</em> approach of O'Meara et al. (2006).
</p>
<p>For discrete traits, the function instead proceeds to fit two variants of the M<em>k</em> model (Lewis 2001): one in which the parameters values (transition rates) of the process are free to vary between trees, and a second in which they are fixed to be the same. 
</p>
<p>For diversification alone, the function fits two different diversification (speciation &amp; extinction) models (Nee et al. 1994; Stadler 2012): one in which the birth (speciation) and death (extinction) rates are identical between the trees, and a second in which they are permitted to differ in various ways depending on the value of <code>"model"</code> (Revell 2018).
</p>
<p>The method <code>posthoc</code> conducts a post-hoc comparison of parameter estimates between trees in the multi-rate or multi-process model. The parameter that is compared depends on the fitted model. For instance, in <code>model="BM"</code> posthoc comparison is made of <code>sig2</code>; if <code>model="OU"</code> fitted values of <code>alpha</code> are compared; and so on. The argument <code>p.adjust.method</code> can be used to specify a method for adjusting P-values for multiple tests following <code>p.adjust</code> (defaults to <code>p.adjust.method="none"</code>.
</p>
<p>At present it is not possible to specify different models to fit for the different trees - although if (for instance) character evolution on tree 1 proceeded by a strong <em>OU</em> process while character evolution on tree 2 was by <em>BM</em>, we would probably reject a constant-process model and tree 2 should show a very low value of <code>alpha</code>.
</p>
<p>To compute the standard errors for each fitted parameter value, the function computes the negative inverse of the Hessian matrix at the MLEs; however, if this matrix is computationally singular the generalized inverse (<code><a href="MASS.html#topic+ginv">ginv</a></code>) will be used instead without warning.
</p>
<p>The function also conducts a likelihood-ratio test to compare the two models.
</p>
<p>For continuous character, optional arguments presently include the following: <code>model</code>, the model of continuous trait evolution (options are <code>"BM"</code>, the default, <code>"OU"</code>, and <code>"EB"</code>). <code>tol</code>, used as a minimum value for the fitting rates, to prevent problems in optimization. <code>trace</code>, a logical value indicating whether or not to report progress in the optimization. <code>test</code>, the method for hypothesis testing (options are <code>"chisq"</code> and <code>"simulation"</code>). <code>quiet</code>, a logical value indicating whether or not to run perfectly quietly. Finally, <code>se</code>, a list of vectors containing the standard errors for each value of <code>x</code>. 
</p>
<p>For <code>type="discrete"</code> the optional arguments are slightly different. The argument <code>model</code> can be used, but it must assume the values <code>"ER"</code>, <code>"SYM"</code>, <code>"ARD"</code>, or a numeric matrix following <code><a href="ape.html#topic+ace">ace</a></code>. 
</p>
<p>Finally, for <code>type= "diversification"</code> models are so far <code>"birth-death"</code>, <code>"equal-extinction"</code>, and <code>"equal-specation"</code>, and <code>"Yule"</code>. It is also important to consider supplying the sampling fractions, <code>rho</code>, which is a vector of values between 0 and 1 of the same length as <code>trees</code>. If not provided the method will assume a sampling fraction of 1.0 for all trees - which is seldom true of empirical studies.
</p>


<h3>Value</h3>

<p>An object of class <code>"ratebytree"</code> or an object of class <code>"posthoc.ratebytree"</code> in the case of the method <code>posthoc</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Adams, D. C. (2012) Comparing evolutionary rates for different phenotypic traits on a phylogeny using likelihood. <em>Syst. Biol.</em>, <b>62</b>, 181-192.
</p>
<p>Lewis, P. O. (2001) A likelihood approach to estimating phylogeny from discrete morphological character data. <em>Systematic Biology</em>, <b>50</b>, 913-925.
</p>
<p>Nee, S., May, R. M. and Harvey, P. H. (1994) The reconstructed evolutionary process. <em>Philosophical Transactions of the Royal Society of London B</em>, <b>344</b>, 305-311.
</p>
<p>O'Meara, B. C., C. Ane, M. J. Sanderson, and P. C. Wainwright. (2006) Testing for different rates of continuous trait evolution using likelihood. <em>Evolution</em>, <b>60</b>, 922-933.
</p>
<p>Revell, L. J. (2018) Comparing the rates of speciation and extinction between phylogenetic trees. <em>Ecology and Evolution</em>, <b>8</b>, 5303-5312.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J., Gonzalez-Valenzuela, L. E., Alfonso, A., Castellanos-Garcia, L. A., Guarnizo, C. E., and Crawford, A. J. (2018) Comparing evolutionary rates between trees, clades, &amp; traits. <em>Methods Ecol. Evol.</em>, <b>9</b>, 994-1005.
</p>
<p>Stadler, T. (2012) How can we improve the accuracy of macroevolutionary rate estimates? <em>Systematic Biology</em>, <b>62</b>, 321-329.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+fitMk">fitMk</a></code>
</p>

<hr>
<h2 id='rateshift'>Find the temporal position of one or more rate shifts</h2><span id='topic+rateshift'></span><span id='topic+plot.rateshift'></span><span id='topic+likSurface.rateshift'></span>

<h3>Description</h3>

<p>Fits a model with one or more temporal rate shifts for a continuous trait on the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rateshift(tree, x, nrates=1, niter=10, method="ML", ...)
## S3 method for class 'rateshift'
plot(x, ...)
likSurface.rateshift(tree, x, nrates=2, shift.range=NULL,
   density=20, plot=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rateshift_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_x">x</code></td>
<td>
<p>vector of phenotypic trait values for species. <code>names(x)</code> should contain the species names and match <code>tree$tip.label</code>. For <code>plot</code> method, <code>x</code> is an object of class <code>"rateshift"</code>.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_nrates">nrates</code></td>
<td>
<p>number of rates.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_niter">niter</code></td>
<td>
<p>number of iterations of optimization routine to ensure convergence.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_method">method</code></td>
<td>
<p>optimization method. Can be <code>"ML"</code> (maximum likelihood) or <code>"REML"</code> (restricted maximum likelihood).</p>
</td></tr>
<tr><td><code id="rateshift_+3A_...">...</code></td>
<td>
<p>optional arguments. In the case of the <code>plot</code> method, these will be passed to <code><a href="#topic+plotSimmap">plotSimmap</a></code>. For <code>rateshift</code>, optional arguments include: <code>tol</code>, tolerance; <code>plot</code> &amp; <code>print</code>, logical values indicating whether to plot or print the progress of the optimization (default to <code>FALSE</code>); <code>quiet</code>, logical argument indicating whether to suppress all notifications (defaults to <code>FALSE</code>); <code>minL</code>, numeric value; and <code>fixed.shift</code>, either a vector of fixed shift points, or a logical value.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_shift.range">shift.range</code></td>
<td>
<p>for <code>likSurface.rateshift</code>.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_density">density</code></td>
<td>
<p>for <code>likSurface.rateshift</code>.</p>
</td></tr>
<tr><td><code id="rateshift_+3A_plot">plot</code></td>
<td>
<p>logical argument for <code>likSurface.rateshift</code>. If <code>plot=FALSE</code> then the surface is returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>rateshift</code> attempts to find the location of one or more rate shifts. This model is quite easy to compute the likelihood for, but quite difficult to optimize as the likelihood surface is often rugged.
</p>
<p><code>likSurface.rateshift</code> plots the likelihood surface.
</p>


<h3>Value</h3>

<p>A fitted object of class <code>"rateshift"</code>, or, in the case of <code>likSurface.rateshift</code>, a likelihood surface for the shift points.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Revell, L. J. and L. J. Harmon (2022) <em>Phylogenetic Comparative Methods in R</em>. Princeton University Press.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>
</p>

<hr>
<h2 id='read.newick'>Newick or Nexus style tree reader</h2><span id='topic+read.newick'></span><span id='topic+readNexus'></span>

<h3>Description</h3>

<p>Reads a phylogenetic tree from file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.newick(file="", text, ...)
readNexus(file="", format=c("standard","raxml"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.newick_+3A_file">file</code></td>
<td>
<p>name of text file with single Newick style tree or multiple trees, one per line. For <code>readNexus</code> this should be a Nexus format tree.</p>
</td></tr>
<tr><td><code id="read.newick_+3A_text">text</code></td>
<td>
<p>character string containing tree.</p>
</td></tr>
<tr><td><code id="read.newick_+3A_format">format</code></td>
<td>
<p>file format (source) for <code>readNexus</code>. In the case of <code>format="standard"</code>, <code><a href="ape.html#topic+read.nexus">read.nexus</a></code> from <span class="pkg">ape</span> will be used internally. For <code>format="raxml"</code>, the parser assumes that bootstrap values have been stored as node labels in the format <code>[&amp;label=bootstrap]</code>.</p>
</td></tr>
<tr><td><code id="read.newick_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to <code><a href="base.html#topic+scan">scan</a></code>. Note that if the arguments <code>sep</code> or <code>what</code> are supplied this could generate an error. Useful optional arguments might include <code>skip</code> (number of lines to skip) and <code>nlines</code> (number of lines to read).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>read.newick</code> reads a simple Newick style tree from file. This function is now almost completely redundant with <code><a href="ape.html#topic+read.tree">read.tree</a></code>. At the time of development, however, it was more 'robust' than <code>read.tree</code> in that it didn't fail if the tree contained so-called 'singles' (nodes with only one descendant); however, <code>read.tree</code> can now handle singleton nodes without difficulty.
</p>
<p>The function <code>readNexus</code> reads a Nexus formatted tree, optionally with bootstrap values as node labels. This function can read a simple Nexus formatted tree from file (like <code><a href="ape.html#topic+read.nexus">read.nexus</a></code>); however, it can also parse the node labels as bootstrap values. This is the output format from the software <em>RAxML</em>. For Nexus tree files with complex node labels (e.g., from the software <em>MrBayes</em>) it will probably fail to parse node labels correctly, if at all.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>, possibly containing singletons (see <code><a href="ape.html#topic+collapse.singles">collapse.singles</a></code>); or an object of class <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+read.tree">read.tree</a></code>, <code><a href="ape.html#topic+read.nexus">read.nexus</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree&lt;-"((Human,Chimp),Gorilla),Monkey);"
phy&lt;-read.newick(text=tree)
</code></pre>

<hr>
<h2 id='read.simmap'>Read SIMMAP style trees from file</h2><span id='topic+read.simmap'></span>

<h3>Description</h3>

<p>This reads one or multiple SIMMAP style trees from file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.simmap(file="", text, format="nexus", rev.order=TRUE, version=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.simmap_+3A_file">file</code></td>
<td>
<p>name of text file with one or multiple SIMMAP v1.0 or v1.5 style trees containing the mapped history of a discrete character.</p>
</td></tr>
<tr><td><code id="read.simmap_+3A_text">text</code></td>
<td>
<p>character string containing the tree. If <code>version=1.5</code> this argument is ignored. (This format tree can only be read from file in the present version.)</p>
</td></tr>
<tr><td><code id="read.simmap_+3A_format">format</code></td>
<td>
<p>format of the trees: either <code>"phylip"</code> or <code>"nexus"</code> - the latter is the default output from SIMMAP. If <code>version=1.5</code> this argument is ignored.</p>
</td></tr>
<tr><td><code id="read.simmap_+3A_rev.order">rev.order</code></td>
<td>
<p>a logical value indicating whether the states and times along each branch is given (from root to tip) in right-to-left order (if TRUE) or in left-to-right order. If <code>version=1.5</code> this argument is ignored.</p>
</td></tr>
<tr><td><code id="read.simmap_+3A_version">version</code></td>
<td>
<p>version of SIMMAP for input tree. If the tree(s) was/were simulated in SIMMAP v1.0 or written to file by <code>link{make.simmap}</code> then <code>version=1.0</code>; if the tree(s) was/were simulated using SIMMAP v1.5 then <code>version=1.5</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function now accepts trees in both SIMMAP v1.0 and SIMMAP v1.5 format. In addition, it can read a more flexible format than is produced by SIMMAP (for instance, multi-character mapped states and more than 7 mapped states).
</p>
<p>The function uses some modified code from <code><a href="ape.html#topic+read.nexus">read.nexus</a></code> from the <span class="pkg">ape</span> package to read the NEXUS block created by SIMMAP. Also creates the attribute <code>"map.order"</code> which indicates whether the stochastic map was read in from left to right or right to left.  This attribute is used by default by <code><a href="#topic+write.simmap">write.simmap</a></code> to write the tree in the same order. 
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> (or list of such objects with class <code>"multiSimmap"</code>), consisting of a modified object of class <code>"phylo"</code> with at least the following additional elements:
</p>
<table>
<tr><td><code>maps</code></td>
<td>
<p>a list of named vectors containing the times spent in each state on each branch, in the order in which they occur.</p>
</td></tr>
<tr><td><code>mapped.edge</code></td>
<td>
<p>a matrix containing the total time spent in each state along each edge of the tree.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+brownie.lite">brownie.lite</a></code>, <code><a href="#topic+evol.vcv">evol.vcv</a></code>, <code><a href="ape.html#topic+read.tree">read.tree</a></code>, <code><a href="ape.html#topic+read.nexus">read.nexus</a></code>
</p>

<hr>
<h2 id='reorder.backbonePhylo'>Reorders a backbone phylogeny</h2><span id='topic+reorder.backbonePhylo'></span>

<h3>Description</h3>

<p>Function reorders an object of class <code>"backbonePhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'backbonePhylo'
reorder(x, order="cladewise", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reorder.backbonePhylo_+3A_x">x</code></td>
<td>
<p>an object of class <code>"backbonePhylo"</code>.</p>
</td></tr>
<tr><td><code id="reorder.backbonePhylo_+3A_order">order</code></td>
<td>
<p>order. See <code><a href="ape.html#topic+reorder.phylo">reorder.phylo</a></code> for possible orderings.</p>
</td></tr>
<tr><td><code id="reorder.backbonePhylo_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"backbonePhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylo.toBackbone">phylo.toBackbone</a></code>
</p>

<hr>
<h2 id='reorderSimmap'>Reorder edges of a <code>"simmap"</code> tree</h2><span id='topic+reorderSimmap'></span>

<h3>Description</h3>

<p>Function returns a reordered modified <code>"phylo"</code> object by using <code>reorder.phylo</code> but then sorting the additional elements <code>$mapped.edge</code> and <code>$maps</code> to have the same order as <code>$edge</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reorderSimmap(tree, order="cladewise", index.only=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reorderSimmap_+3A_tree">tree</code></td>
<td>
<p>a modified object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="reorderSimmap_+3A_order">order</code></td>
<td>
<p><code>"cladewise"</code>, <code>"pruningwise"</code>, or any other allowable order permitted by <code><a href="ape.html#topic+reorder.phylo">reorder.phylo</a></code>.</p>
</td></tr>
<tr><td><code id="reorderSimmap_+3A_index.only">index.only</code></td>
<td>
<p>logical value indicating whether only an index should be returned.</p>
</td></tr>
<tr><td><code id="reorderSimmap_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+reorder.phylo">reorder.phylo</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>

<hr>
<h2 id='rep.phylo'>Replicate a tree or set of trees</h2><span id='topic+rep.phylo'></span><span id='topic+rep.multiPhylo'></span><span id='topic+repPhylo'></span>

<h3>Description</h3>

<p><code><a href="base.html#topic+rep">rep</a></code> method for object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo'
rep(x, ...)
## S3 method for class 'multiPhylo'
rep(x, ...)
repPhylo(tree, times)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rep.phylo_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="rep.phylo_+3A_times">times</code></td>
<td>
<p>number of times to replicate tree.</p>
</td></tr>
<tr><td><code id="rep.phylo_+3A_x">x</code></td>
<td>
<p>for S3 method an object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="rep.phylo_+3A_...">...</code></td>
<td>
<p>other arguments for <code>rep</code> (specifically, <code>times</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>repPhylo</code> is just an alias for <code>rep.phylo</code> and <code>rep.multiPhylo</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+c.phylo">c.phylo</a></code>, <code><a href="base.html#topic+rep">rep</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree&lt;-pbtree(n=100)
trees&lt;-rep(tree,100)
</code></pre>

<hr>
<h2 id='reroot'>Re-root a tree along an edge</h2><span id='topic+reroot'></span>

<h3>Description</h3>

<p>Re-roots a phylogenetic tree at an arbitrary position along an edge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reroot(tree, node.number, position=NULL, interactive=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reroot_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="reroot_+3A_node.number">node.number</code></td>
<td>
<p> number of the node descending from the target branch in <code>tree$edge</code> - this can also be a tip in which case the node number is the index number of the tip in <code>tree$tip.label</code>.</p>
</td></tr>
<tr><td><code id="reroot_+3A_position">position</code></td>
<td>
<p>position along the target edge at which to re-root the tree. If not supplied, then the tree will be re-rooted <em>at</em> the node or tip.</p>
</td></tr>
<tr><td><code id="reroot_+3A_interactive">interactive</code></td>
<td>
<p>logical value indicating whether to use interactive mode (defaults to <code>interactive=</code> <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="reroot_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code>plotTree</code> for <code>interactive=TRUE</code> only.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function had an error for rootings along edges descended from the root node for <span class="pkg">phytools</span>&lt;=0.2-47. This should be fixed in the present version. Now uses <code><a href="#topic+paste.tree">paste.tree</a></code>, <code><a href="ape.html#topic+root">root</a></code>, and <code><a href="#topic+splitTree">splitTree</a></code> internally. Earlier versions also had an error related to node labels. This should be fixed in <span class="pkg">phytools</span>&gt;=0.4-47.
</p>


<h3>Value</h3>

<p>A phylogenetic tree in <code>"phylo"</code> format.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+splitTree">splitTree</a></code>, <code><a href="#topic+paste.tree">paste.tree</a></code>, <code><a href="ape.html#topic+root">root</a></code>
</p>

<hr>
<h2 id='rerootingMethod'>Get marginal ancestral state reconstructions by re-rooting</h2><span id='topic+rerootingMethod'></span>

<h3>Description</h3>

<p>Computes marginal ancestral states for a discrete character using the re-rooting method of Yang et al. (1995).
</p>
<p>In general, this function is <b>redundant</b> with <code><a href="#topic+ancr">ancr</a></code> for circumstances in which it is valid (i.e., symmetric <b>Q</b> matrices), and improper otherwise. In general <code><a href="#topic+ancr">ancr</a></code> should be preferred.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rerootingMethod(tree, x, model=c("ER","SYM"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rerootingMethod_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="rerootingMethod_+3A_x">x</code></td>
<td>
<p>a vector of tip values for species, or a matrix containing the prior probability that the tip is in each state. If <code>x</code> is a vector, then <code>names(x)</code> should be the species names. If <code>x</code> is a matrix of prior probabilities, then rownames should be species names, column names should be states for the discrete character, and rows of the matrix should sum to 1.0.</p>
</td></tr>
<tr><td><code id="rerootingMethod_+3A_model">model</code></td>
<td>
<p>any reversible model. <code>model=c("ER","SYM")</code> recommended.</p>
</td></tr>
<tr><td><code id="rerootingMethod_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently the logical argument <code>tips</code>. If <code>tips=TRUE</code>, then the function will also compute the empirical Bayes posterior probabilities of the tips following Yang (2006). Note that <code>...</code> is passed internally to <code>fitMk</code>, but should be used in this way with caution because any arguments that conflict with the default arguments of the method will cause the function execution to fail. The most practical use of this would be to force a particular value of the transition matrix, <code>Q</code>, via the argument <code>fixedQ</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the re-rooting method of Yang et al. (1995) to get the marginal ancestral state estimates for each internal node of the tree using likelihood. This method get the conditional scaled likelihoods for the root node (which is the same as the marginal ancestral state reconstruction for that node) and successively moves the root to each node in the tree. The function can also return the posterior probabilities for the tip nodes of the tree.
</p>
<p><code>rerootingMethod</code> calls <code><a href="#topic+fitMk">fitMk</a></code> internally. <code>fitMk</code> uses some code adapted from <code>ace</code> in the <span class="pkg">ape</span> package.
</p>


<h3>Value</h3>

<p>An object of class <code>"rerootingMethod"</code> containing at least the following elements:
</p>
<table>
<tr><td><code>loglik</code></td>
<td>
<p>the log-likelihood.</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>the fitted transition matrix between states.</p>
</td></tr>
<tr><td><code>marginal.anc</code></td>
<td>
<p>the marginal ancestral state reconstructions for each node (and, optionally, each tip).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Yang, Z., Kumar, S., Nei, M. (1995) A new method of inference of ancestral nucleotide and amino acid sequences. <em>Genetics</em>, <b>141</b>, 1641-1650.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+ancr">ancr</a></code>, <code><a href="#topic+fitMk">fitMk</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>
</p>

<hr>
<h2 id='rescale'>Rescale phylogenetic objects of different types</h2><span id='topic+rescale'></span>

<h3>Description</h3>

<p>Generic method for rescaling different types of phylogenetic trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rescale(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rescale_+3A_x">x</code></td>
<td>
<p>phylogenetic tree object to be rescaled: e.g., object of class <code>"phylo"</code> or <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="rescale_+3A_...">...</code></td>
<td>
<p>other arguments to be used in rescaling, depending on the object class. (E.g., see <code><a href="geiger.html#topic+rescale.phylo">rescale.phylo</a></code> in <span class="pkg">geiger</span> and <code><a href="#topic+rescale.simmap">rescale.simmap</a></code>.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code><a href="geiger.html#topic+rescale.phylo">rescale.phylo</a></code> in <span class="pkg">geiger</span> and <code><a href="#topic+rescale.simmap">rescale.simmap</a></code> for details.
</p>


<h3>Value</h3>

<p>A rescaled phylogenetic tree object.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Pennell, M.W., J. M. Eastman, G. J. Slater, J. W. Brown, J. C. Uyeda, R. G. FitzJohn, M. E. Alfaro, and L. J. Harmon (2014) geiger v2.0: an expanded suite of methods for fitting macroevolutionary models to phylogenetic trees. <em>Bioinformatics</em>, <b>30</b>, 2216-2218.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+rescale.phylo">rescale.phylo</a></code>, <code><a href="#topic+rescale.simmap">rescale.simmap</a></code>
</p>

<hr>
<h2 id='rescale.simmap'>Rescale object of class <code>"simmap"</code></h2><span id='topic+rescale.simmap'></span><span id='topic+rescale.multiSimmap'></span><span id='topic+rescaleSimmap'></span>

<h3>Description</h3>

<p>Scales a tree with a mapped discrete character (<code>"simmap"</code> object), or a set of such trees, to an arbitrary total height, preserving the relative time spent in each state along each edge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simmap'
rescale(x, model="depth", ...)
## S3 method for class 'multiSimmap'
rescale(x, model="depth", ...)
rescaleSimmap(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rescale.simmap_+3A_x">x</code></td>
<td>
<p>object of class <code>"simmap"</code> or <code>"multiSimmap"</code> to be rescaled.</p>
</td></tr>
<tr><td><code id="rescale.simmap_+3A_model">model</code></td>
<td>
<p>model to use to rescale the tree. Currently the only option is <code>"depth"</code>.</p>
</td></tr>
<tr><td><code id="rescale.simmap_+3A_...">...</code></td>
<td>
<p>parameter of the model to use in rescaling. Currently the only parameter is <code>depth</code> for <code>model="depth"</code>.</p>
</td></tr>
<tr><td><code id="rescale.simmap_+3A_tree">tree</code></td>
<td>
<p>for <code>rescaleSimmap</code>, object of class <code>"simmap"</code> to be rescaled.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Replaces <code>rescaleTree</code> (now <code>rescale.phylo</code>) in the <span class="pkg">geiger</span> package for the <code>"simmap"</code> object class. <code>rescaleSimmap</code> is now a redundant alias for the method <code>rescale.simmap</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> or <code>"multiSimmap"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load anoletree
data(anoletree)
## rescale to have total depth of 50
rescaled_anoletree&lt;-rescale(anoletree,depth=50)
## plot rescaled tree
plot(rescaled_anoletree,ftype="i",fsize=0.6,
	mar=c(5.1,1.1,1.1,1.1))
axis(1,at=seq(0,50,by=10))
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margin to default
</code></pre>

<hr>
<h2 id='resolveNode'>Compute all possible resolutions of a node or all nodes in a multifurcating tree</h2><span id='topic+resolveNode'></span><span id='topic+resolveAllNodes'></span>

<h3>Description</h3>

<p>Resolves a single multifurcation or all multifurcations in all possible ways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resolveNode(tree,node)
resolveAllNodes(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resolveNode_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="resolveNode_+3A_node">node</code></td>
<td>
<p>for <code>resolveNode</code>, the node with a polytomy to resolve.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions resolves a single multifurcation or all multifurcations in a tree in all possible ways. If the input tree has edge lengths, then the resolutions will use internal edges of zero length.
</p>
<p>For <code>resolveNode</code> applied to a multifurcation with <em>n</em> descendants, the number of resolved trees will be equal to the number of possible rooted trees of <em>n</em> taxa. (For instance, three for a trifurcation, 15 for a quadrifurcation, and so on.)
</p>
<p>For <code>resolveAllNodes</code> the number of fully resolved trees will be equal to the product of numbers for <code>resolveNode</code> applied to each multifurcation separately. (For instance, 45 for a tree containing one trifurcation and one quadrifurcation.)
</p>


<h3>Value</h3>

<p>An object of class <code>"multiPhylo"</code> - or, if the input tree is already fully resolved, an object of class <code>"phylo"</code> identical to <code>tree</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+multi2di">multi2di</a></code>
</p>

<hr>
<h2 id='rotateNodes'>Rotates a node or set of nodes in a phylogenetic tree</h2><span id='topic+rotateNodes'></span><span id='topic+rotate.multi'></span><span id='topic+allRotations'></span>

<h3>Description</h3>

<p>The function <code>rotateNodes</code> is a simple wrapper for <code><a href="ape.html#topic+rotate">rotate</a></code> which rotates a set of nodes or all nodes.
</p>
<p>The function <code>rotate.multi</code> finds all possible rotations around a multifurcating node, given by <code>node</code>. This will be an object of class <code>"multiPhylo"</code>, assuming that the node specified is indeed a multifurcation.
</p>
<p>The function <code>allRotations</code> computes all possible rotated trees for a given input phylogeny. For a binary tree, this is generally two raised to the power of the number of internal nodes (so a very large number, if <em>N</em> is even modest in size).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotateNodes(tree, nodes, polytom=c(1,2), ...)
rotate.multi(tree, node)
allRotations(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rotateNodes_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="rotateNodes_+3A_nodes">nodes</code></td>
<td>
<p>either a single node number to rotate, a vector of node numbers, or the string <code>"all"</code>.</p>
</td></tr>
<tr><td><code id="rotateNodes_+3A_polytom">polytom</code></td>
<td>
<p>a vector of mode numeric and length two specifying the two clades that should be exchanged in a polytomy (see <code><a href="ape.html#topic+rotate">rotate</a></code>).</p>
</td></tr>
<tr><td><code id="rotateNodes_+3A_node">node</code></td>
<td>
<p>a single node to rotate (in the case of <code>rotate.multi</code>).</p>
</td></tr>
<tr><td><code id="rotateNodes_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All three functions also address the problem that the product of multiple rotations from <code><a href="ape.html#topic+rotate">rotate</a></code> can be non-compliant with the implicit <code>"phylo"</code> standard because the tip numbers in <code>tree$edge</code> are not in numerical order <code>1:n</code> for <code>n</code> tips.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> (i.e., a phylogenetic tree), in the case of <code>rotateNodes</code>, or an object of class <code>"multiPhylo"</code> for <code>rotate.multi</code> or <code>allRotations</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='roundBranches'>Rounds the branch lengths of a tree</h2><span id='topic+roundBranches'></span>

<h3>Description</h3>

<p>Rounds the branch lengths of a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roundBranches(tree, digits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roundBranches_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>, <code>"multiPhylo"</code>, <code>"simmap"</code>, or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="roundBranches_+3A_digits">digits</code></td>
<td>
<p>number of digits for rounding. Passed to <code><a href="base.html#topic+round">round</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function rounds the branch lengths of a tree or trees to a precision indicated by <code>digits</code>, and reconciles any mappings for objects of class <code>"simmap"</code> or <code>"multiSimmap"</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>, <code>"multiPhylo"</code>, <code>"simmap"</code>, or <code>"multiSimmap"</code>, with rounded edge lengths.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='roundPhylogram'>Plot a round, sigmoidal, or spline phylogram or cladogram</h2><span id='topic+roundPhylogram'></span><span id='topic+sigmoidPhylogram'></span><span id='topic+splinePhylogram'></span>

<h3>Description</h3>

<p>Plots one or multiple round phylograms, a sigmoidal phylogram or cladogram, or a phylogram draw using cubic splines through the nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roundPhylogram(tree, fsize=1.0, ftype="reg", lwd=2, mar=NULL, offset=NULL, 
   direction="rightwards", type="phylogram", xlim=NULL, ylim=NULL, ...)
sigmoidPhylogram(tree, ...)
splinePhylogram(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roundPhylogram_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code> or <code>"multiPhylo"</code> containing one or multiple phylogenies.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_fsize">fsize</code></td>
<td>
<p>relative font size for tip labels.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_ftype">ftype</code></td>
<td>
<p>font type - options are <code>"reg"</code>, <code>"i"</code> (italics), <code>"b"</code> (bold), or <code>"bi"</code> (bold-italics).</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_lwd">lwd</code></td>
<td>
<p>line width for plotting.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_mar">mar</code></td>
<td>
<p>vector containing the margins for the plot to be passed to <code><a href="graphics.html#topic+par">par</a></code>. If not specified, the default margins are [0.1,0.1,0.1,0.1].</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_offset">offset</code></td>
<td>
<p>offset for the tip labels.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_direction">direction</code></td>
<td>
<p>plotting direction. Only the option <code>direction="rightwards"</code> is presently supported.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_type">type</code></td>
<td>
<p>plot type. Can be <code>"phylogram"</code> or <code>"cladogram"</code>. If <code>type="cladogram"</code> then the branch lengths are not necessary (and, indeed, are not used).</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_xlim">xlim</code></td>
<td>
<p>x-limits for the plot.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_ylim">ylim</code></td>
<td>
<p>y-limits for the plot.</p>
</td></tr>
<tr><td><code id="roundPhylogram_+3A_...">...</code></td>
<td>
<p>optional arguments. In the case of <code>roundPhylogram</code>, these are mostly as in <code><a href="#topic+plotTree">plotTree</a></code> and <code><a href="#topic+plotSimmap">plotSimmap</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The underscore character <code>"_"</code> is automatically swapped for a space in tip labels, as in <code><a href="#topic+plotSimmap">plotSimmap</a></code>.
</p>


<h3>Value</h3>

<p>Plots a tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimmap">plotSimmap</a></code>, <code><a href="#topic+plotTree">plotTree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load mammal.tree
data(mammal.tree)
## plot using roundPhylogram
roundPhylogram(compute.brlen(mammal.tree),
    fsize=0.8,ftype="i")
## load anoletree
data(anoletree)
## plot using sigmoidPhylogram
sigmoidPhylogram(anoletree,fsize=0.6,
    ftype="i",direction="upwards")
par(mar=c(5.1,4.1,4.1,2.1)) ## reset margin to default
</code></pre>

<hr>
<h2 id='rstate'>Pick a random state according to a vector of probabilities</h2><span id='topic+rstate'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+make.simmap">make.simmap</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rstate(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rstate_+3A_y">y</code></td>
<td>
<p>vector of probabilities. Must have names &amp; should probably add to <code>1.0</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function picks a random element in a vector according to the probability assigned that element.  It returns the name. Uses <code><a href="stats.html#topic+rmultinom">rmultinom</a></code>.
</p>


<h3>Value</h3>

<p>A character or string.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='sampleFrom'>Sample from a set of distributions</h2><span id='topic+sampleFrom'></span>

<h3>Description</h3>

<p>Samples from a set of normal distributions with parameters given in <code>xbar</code> and <code>xvar</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleFrom(xbar=0, xvar=1, n=1, randn=NULL, type="norm")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampleFrom_+3A_xbar">xbar</code></td>
<td>
<p>a named vector of means.</p>
</td></tr>
<tr><td><code id="sampleFrom_+3A_xvar">xvar</code></td>
<td>
<p>a named vector of variances.</p>
</td></tr>
<tr><td><code id="sampleFrom_+3A_n">n</code></td>
<td>
<p>a vector containing the sample sizes of each species.</p>
</td></tr>
<tr><td><code id="sampleFrom_+3A_randn">randn</code></td>
<td>
<p>a range of sample sizes are to be random.</p>
</td></tr>
<tr><td><code id="sampleFrom_+3A_type">type</code></td>
<td>
<p><code>"norm"</code> is the only distribution implemented so far.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector, with labels.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='setMap'>Set color map for various phylogenetic objects of classes</h2><span id='topic+setMap'></span><span id='topic+setMap.contMap'></span><span id='topic+setMap.densityMap'></span><span id='topic+setMap.phyloScattergram'></span><span id='topic+setMap.multirateBM_plot'></span>

<h3>Description</h3>

<p>Changes the color map (ramp) in an object of class <code>"contMap"</code>, <code>"densityMap"</code>, <code>"phyloScattergram"</code>, or <code>"multirateBM_plot"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setMap(x, ...)
## S3 method for class 'contMap'
setMap(x, ...)
## S3 method for class 'densityMap'
setMap(x, ...)
## S3 method for class 'phyloScattergram'
setMap(x, ...)
## S3 method for class 'multirateBM_plot'
setMap(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setMap_+3A_x">x</code></td>
<td>
<p>an object of class <code>"contMap"</code>, <code>"densityMap"</code>, <code>"phyloScattergram"</code>, or <code>"multirateBM_plot"</code>.</p>
</td></tr>
<tr><td><code id="setMap_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>. Also, the argument <code>invert</code> which (if <code>invert=TRUE</code>) will just flip the current color ramp.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"contMap"</code>, <code>"densityMap"</code>, <code>"phyloScattergram"</code>, or <code>"multirateBM_plot"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). <em>Methods Ecol. Evol.</em>, <b>3</b>, 217-223.
</p>
<p>Revell, L. J. (2013) Two new graphical methods for mapping trait evolution on phylogenies. <em>Methods in Ecology and Evolution</em>, <b>4</b>, 754-759.
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contMap">contMap</a></code>, <code><a href="#topic+densityMap">densityMap</a></code>, <code><a href="#topic+multirateBM">multirateBM</a></code>
</p>

<hr>
<h2 id='sim.corrs'>Multivariate Brownian simulation with multiple correlations and rates</h2><span id='topic+sim.corrs'></span>

<h3>Description</h3>

<p>Simulates multivariate Brownian motion evolution on a tree with multiple evolutionary correlation/covariance matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.corrs(tree, vcv, anc=NULL, internal=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.corrs_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code> or <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="sim.corrs_+3A_vcv">vcv</code></td>
<td>
<p>is a square covariance matrix or named list of matrices (one for each mapped state on the tree).</p>
</td></tr>
<tr><td><code id="sim.corrs_+3A_anc">anc</code></td>
<td>
<p>optional vector of values for the root state.</p>
</td></tr>
<tr><td><code id="sim.corrs_+3A_internal">internal</code></td>
<td>
<p>logical value indicating whether to return states at internal nodes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function conducts BM simulation on a tree with multiple rates and/or multiple evolutionary correlations between characters.
</p>
<p>If <code>vcv</code> is a single matrix, instead of a list of matrices, <code>sim.corrs</code> will simulate multivariate BM with a single rate matrix.
</p>


<h3>Value</h3>

<p>A matrix containing the multivariate tip states for the <code>n</code> species in the tree (and nodes if <code>internal=</code> <code>TRUE</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J., and D. C. Collar (2009) Phylogenetic analysis of the evolutionary correlation using likelihood. <em>Evolution</em>, <b>63</b>, 1090-1100.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastBM">fastBM</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+sim.history">sim.history</a></code>, <code><a href="#topic+sim.rates">sim.rates</a></code>
</p>

<hr>
<h2 id='sim.history'>Simulate character history or a discrete character at the tips of the tree under some model</h2><span id='topic+sim.history'></span><span id='topic+sim.Mk'></span><span id='topic+sim.multiMk'></span>

<h3>Description</h3>

<p>Simulates discrete character evolution on a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.history(tree, Q, anc=NULL, nsim=1, 
    direction=c("column_to_row","row_to_column"), ...)
sim.Mk(tree, Q, anc=NULL, nsim=1, ...)
sim.multiMk(tree, Q, anc=NULL, nsim=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.history_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree as an object of class <code>"phylo"</code>. For the case of <code>sim.multiMk</code> <code>tree</code> should be an object of class <code>"simmap"</code> in which the regimes for simulation have been mapped onto the tree.</p>
</td></tr>
<tr><td><code id="sim.history_+3A_q">Q</code></td>
<td>
<p>a matrix containing the instantaneous transition rates between states. Note that for <code>sim.history</code> <em>by default</em> (i.e., when <code>direction="column_to_row"</code>, see below) normally this is the <em>transpose</em> of the matrix produced by <code>fitDiscrete</code> in the <span class="pkg">geiger</span> package or <code><a href="#topic+make.simmap">make.simmap</a></code> in <span class="pkg">phytools</span>; that is to say the transition rate from <code>i -&gt; j</code> should be given by <code>Q[j,i]</code>. However, if your matrix is properly conformed (i.e., rows <em>or</em> columns sum to 0), then <code>sim.history</code> will attempt to transpose your matrix correctly &amp; will return an informative message (if <code>message=TRUE</code>, see below). For <code>sim.Mk</code> and <code>sim.multiMk</code> this matrix has the same conformation as in <code>fitContinuous</code> and <code>make.simmap</code>. For <code>sim.multiMk</code> <code>Q</code> should be a list of transition matrices with names that correspond to the states mapped onto the tree.</p>
</td></tr>
<tr><td><code id="sim.history_+3A_anc">anc</code></td>
<td>
<p>an optional value for the state at the root node; if <code>NULL</code>, a random state will be assigned. <code>anc</code> can be a vector of states, in which one of the states will be chosen randomly for each simulation. For <code>sim.history</code> <code>anc</code> can be a vector of probabilities with names, in which case a state will be chosen in proportion to the given probabilities.</p>
</td></tr>
<tr><td><code id="sim.history_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations.</p>
</td></tr>
<tr><td><code id="sim.history_+3A_direction">direction</code></td>
<td>
<p>row/column direction of the input transition matrix, <code>Q</code>. <code>"column_to_row"</code> indicates that the transition rate from <code>i -&gt; j</code> should be given by <code>Q[j,i]</code>, while <code>"row_to_column"</code> indicates the converse.</p>
</td></tr>
<tr><td><code id="sim.history_+3A_...">...</code></td>
<td>
<p>other optional arguments. Currently only <code>internal</code>, a logical value indicating whether or not to return internal node states (defaults to <code>internal=FALSE</code>; and <code>message</code>, a logical indicating whether or not to turn on informational messages (defaults to <code>message=TRUE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>sim.history</code> simulates a stochastic character history for a discretely valued character trait on the tree. The resultant tree is stored as a modified <code>"phylo"</code> object in stochastic character map (e.g., <code><a href="#topic+make.simmap">make.simmap</a></code>) format.
</p>
<p>The function <code>sim.Mk</code> simulates the states for a discrete character at the tips of the tree only.
</p>
<p>Finally, the function <code>sim.multiMk</code> is the same as <code>sim.Mk</code> except that it permits the user to simulate under different values of <code>Q</code> in different parts of the tree.
</p>


<h3>Value</h3>

<p><code>sim.history</code> returns an object of class <code>"simmap"</code> (a tree with a mapped discrete character) or <code>"multiSimmap"</code> for <code>nsim</code> greater than one.
</p>
<p><code>sim.Mk</code> and <code>sim.multiMk</code> return a factor with the states of our discrete character at the tips of the tree only.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitMk">fitMk</a></code>, <code><a href="#topic+fitmultiMk">fitmultiMk</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>, <code><a href="#topic+sim.rates">sim.rates</a></code>
</p>

<hr>
<h2 id='sim.ratebystate'>Conduct simulation of state dependent rate variation</h2><span id='topic+sim.ratebystate'></span>

<h3>Description</h3>

<p>Simulates two characters under a model in which the rate of one depends on the state of the other.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.ratebystate(tree, sig2x=1, sig2y=1, beta=c(0,1), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.ratebystate_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
<tr><td><code id="sim.ratebystate_+3A_sig2x">sig2x</code></td>
<td>
<p>variance of the Brownian process of evolution for <em>x</em>, <code class="reqn">\sigma_x^2</code>.</p>
</td></tr>
<tr><td><code id="sim.ratebystate_+3A_sig2y">sig2y</code></td>
<td>
<p>variance of the Brownian process of evolution for <em>y</em> when <code>x-min(x)==1</code> (for <code>logarithm=FALSE</code>) or <code>x==0</code> (for <code>logarithm=TRUE</code>).</p>
</td></tr>
<tr><td><code id="sim.ratebystate_+3A_beta">beta</code></td>
<td>
<p>intercept and slope of the relationship between the value of <em>x</em> and the Brownian rate in <em>y</em>.</p>
</td></tr>	
<tr><td><code id="sim.ratebystate_+3A_...">...</code></td>
<td>
<p>optional arguments which include <code>method</code> (<code>"by.node"</code> or <code>"by.branch"</code> indicating whether to assume the rate varies as a function of the node state or the mean branch state); <code>plot</code>, a logical value indicating whether or not to plot a traitgram (&quot;phenogram&quot;) with the branches used for simulation of <code>y</code> after rescaling by the state of <code>x</code>; and <code>logarithm</code>, a logical value indicating whether or not simulate changes in the variance of Brownian evolution for <code>y</code> as an additive <code>logarithm=FALSE</code> or multiplicative function of <code>x</code>. The default is <code>logarithm=FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function attempts to simulate two characters under a model in which the rate of evolution for the second (<em>y</em>) depends on the states for the first (<em>x</em>). 
</p>
<p>See <code><a href="#topic+ratebystate">ratebystate</a></code> for more details.
</p>


<h3>Value</h3>

<p>This function returns a matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastBM">fastBM</a></code>, <code><a href="#topic+ratebystate">ratebystate</a></code>
</p>

<hr>
<h2 id='sim.rates'>Brownian or OU simulation with multiple evolutionary regimes</h2><span id='topic+sim.rates'></span><span id='topic+multiOU'></span>

<h3>Description</h3>

<p>Simulates multi-rate or multi-regime continuous trait evolution on a phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.rates(tree, sig2, anc=0, nsim=1, internal=FALSE, plot=FALSE)
multiOU(tree, alpha, sig2, theta=NULL, a0=NULL, nsim=1, internal=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.rates_+3A_tree">tree</code></td>
<td>
<p>is a stochastic map format phylogenetic tree in modified <code>"phylo"</code> format (e.g., see <code><a href="#topic+make.simmap">make.simmap</a></code>).</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_sig2">sig2</code></td>
<td>
<p>a named vector containing the rates for each state; names should be states in <code>mtree</code>.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_anc">anc</code></td>
<td>
<p>optional value for the root state.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_nsim">nsim</code></td>
<td>
<p>number of simulations.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_internal">internal</code></td>
<td>
<p>logical value indicating whether to return states at internal nodes.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether or not to visual the rate heterogeneity (default value is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_alpha">alpha</code></td>
<td>
<p>single value or vector of values of the OU <code class="reqn">\alpha</code> parameter.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_theta">theta</code></td>
<td>
<p>single value or vector of values of the OU <code class="reqn">\theta</code> parameter.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_a0">a0</code></td>
<td>
<p>optional value of the root state. Defaults to zero.</p>
</td></tr>
<tr><td><code id="sim.rates_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>sim.rates</code> conducts BM simulation on a tree with multiple rates.
</p>
<p>The function <code>multiOU</code> conducts multi-regime OU simulations on the tree under a range of conditions. <code>multiOU</code> uses a difference equation approximation of the OU process.
</p>


<h3>Value</h3>

<p>A vector (for <code>nsim=1</code>) or matrix containing the tip states for the <code>n</code> species in the tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastBM">fastBM</a></code>, <code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+sim.history">sim.history</a></code>
</p>

<hr>
<h2 id='simBMphylo'>Creates a graphical illustration of Brownian motion evolution on a phylogeny</h2><span id='topic+simBMphylo'></span><span id='topic+plot.simBMphylo'></span>

<h3>Description</h3>

<p>Simulates a discrete time phylogeny and Brownian motion trait, and generates a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simBMphylo(n, t, sig2, plot=TRUE, ...)
## S3 method for class 'simBMphylo'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simBMphylo_+3A_n">n</code></td>
<td>
<p>number of taxa to simulate in the output tree.</p>
</td></tr>
<tr><td><code id="simBMphylo_+3A_t">t</code></td>
<td>
<p>total time for the simulation.</p>
</td></tr>
<tr><td><code id="simBMphylo_+3A_sig2">sig2</code></td>
<td>
<p>the rate of evolution under Brownian motion, <code class="reqn">\sigma^2</code>, or a vector of rates. If the latter the length of the vector must exactly match <code>t</code>, otherwise the first element of <code>sig2</code> will just be duplicated <code>t</code> times.</p>
</td></tr>
<tr><td><code id="simBMphylo_+3A_plot">plot</code></td>
<td>
<p>optional logical value indicating whether or not the simulated object should be plotted.</p>
</td></tr>
<tr><td><code id="simBMphylo_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to the <code>plot</code> method.</p>
</td></tr>
<tr><td><code id="simBMphylo_+3A_x">x</code></td>
<td>
<p>in <code>plot</code> method, object of class <code>"simBMphylo"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function simulates a discrete-time pure-birth phylogeny (for fixed <em>N</em> and <em>t</em> using rejection sampling) and then discrete-time Brownian motion on that tree. It then proceeds to generating a plot of the results.
</p>


<h3>Value</h3>

<p>An object of class <code>"simBMphylo"</code> or a plot.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmPlot">bmPlot</a></code>, <code><a href="#topic+fastBM">fastBM</a></code>, <code><a href="#topic+pbtree">pbtree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(777)
simBMphylo(n=6,t=100,sig2=0.01)
## reset par to defaults
par(mfrow=c(1,1))
par(mar=c(5.1,4.1,4.1,2.1))
</code></pre>

<hr>
<h2 id='skewers'>Matrix comparison using the method of random skewers</h2><span id='topic+skewers'></span>

<h3>Description</h3>

<p>Performs the random skewers matrix comparison method of Cheverud (1996).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skewers(X, Y, nsim=100, method=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="skewers_+3A_x">X</code></td>
<td>
<p>covariance matrix.</p>
</td></tr>
<tr><td><code id="skewers_+3A_y">Y</code></td>
<td>
<p>covariance matrix.</p>
</td></tr>
<tr><td><code id="skewers_+3A_nsim">nsim</code></td>
<td>
<p>number of random vectors.</p>
</td></tr>
<tr><td><code id="skewers_+3A_method">method</code></td>
<td>
<p>method to generate a null distribution of the random skewers correlation between matrices. If <code>method=NULL</code> then the correlation will be compared to the correlation between random vectors; however this test has type I error substantially above the nominal level for ostensibly random matrices. Other values of <code>method</code> will be passed as <code>covMethod</code> to <code>genPositiveDefMat</code> for a more robust hypothesis test (see below). Recommended values include <code>"unifcorrmat"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs the random skewers matrix comparison method of Cheverud (1996; also see Cheverud &amp; Marroig 2007 for more details). In addition, it includes a more robust hypothesis test in which random covariance matrices are simulated under a variety of models, and then the mean correlation between response vectors to random skewers are computed.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>r</code></td>
<td>
<p>mean random skewers correlation.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p-value from simulation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Cheverud, J. M. (1996) Quantitative genetic analysis of cranial morphology in the cotton-top (<em>Saguinus oedipus</em>) and saddle-back (<em>S. fuscicollis</em>) tamarins. <em>J. Evol. Biol.</em>, <b>9</b>, 5&ndash;42.
</p>
<p>Cheverud, J. M. and G. Marroig (2007) Comparing covariance matrices: Random skewers method compared to the common principal components model. <em>Genetics &amp; Molecular Biology</em>, <b>30</b>, 461&ndash;469.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='splitEdgeColor'>Split edge colors when descendant edges have different mapped states</h2><span id='topic+splitEdgeColor'></span>

<h3>Description</h3>

<p>Splits the vertical linking-line color on a plotted tree to match the daughter edges.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitEdgeColor(tree,colors,lwd=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitEdgeColor_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="splitEdgeColor_+3A_colors">colors</code></td>
<td>
<p>named vector of colors to be used for plotting.</p>
</td></tr>
<tr><td><code id="splitEdgeColor_+3A_lwd">lwd</code></td>
<td>
<p>width of the plotted lines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function splits the vertical line colors to match the daughter edges when the daughters have different states. Mostly to be used with trees generated using <code><a href="#topic+paintBranches">paintBranches</a></code> or <code><a href="#topic+paintSubTree">paintSubTree</a></code>. Also used internally by <code><a href="#topic+plotSimmap">plotSimmap</a></code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>

<hr>
<h2 id='splitplotTree'>Plots a phylogeny in two columns</h2><span id='topic+splitplotTree'></span><span id='topic+plotTree.splits'></span>

<h3>Description</h3>

<p>Plots a tree in two columns or windows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitplotTree(tree, fsize=1.0, ftype="reg", lwd=2, split=NULL, new.window=FALSE)
plotTree.splits(tree, splits=NULL, file=NULL, fn=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitplotTree_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_fsize">fsize</code></td>
<td>
<p>relative font size for tip labels.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_ftype">ftype</code></td>
<td>
<p>font type - options are <code>"reg"</code>, <code>"i"</code> (italics), <code>"b"</code> (bold), or <code>"bi"</code> (bold-italics).</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_lwd">lwd</code></td>
<td>
<p>line width for plotting.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_split">split</code></td>
<td>
<p>relative vertical position for splitting the tree (between 0 &amp; 1).</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_new.window">new.window</code></td>
<td>
<p>whether or not to plot the split tree in a new window. If <code>FALSE</code> then the tree will be plotted in two columns within the same plotting window.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_splits">splits</code></td>
<td>
<p>for <code>plotTree.splits</code> relative positions (from 0 to 1) to split the tree across pages or devices.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_file">file</code></td>
<td>
<p>filename if saving to a PDF file is desired. Otherwise will plot to the default plotting device.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_fn">fn</code></td>
<td>
<p>function to be executed on each plotted page. For instance, might be: <code>function()</code> <code>cladelabels()</code> if clade labels are desired.</p>
</td></tr>
<tr><td><code id="splitplotTree_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a tree.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotTree">plotTree</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code> 
</p>

<hr>
<h2 id='splitTree'>Split tree at a point</h2><span id='topic+splitTree'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+posterior.evolrate">posterior.evolrate</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitTree(tree, split)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitTree_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
<tr><td><code id="splitTree_+3A_split">split</code></td>
<td>
<p>split encoded as a list with two elements: <code>node</code>: the node number tipward of the split; and <code>bp</code>: the position along the branch to break the tree, measured from the rootward end of the edge.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function splits the tree at a given point, and returns the two subtrees as an object of class <code>"multiPhylo"</code>.
</p>
<p>Probably do not use this unless you can figure out what you are doing.
</p>


<h3>Value</h3>

<p>Two trees in a list.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='starTree'>Create star phylogeny</h2><span id='topic+starTree'></span>

<h3>Description</h3>

<p>Creates a star phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>starTree(species, branch.lengths=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="starTree_+3A_species">species</code></td>
<td>
<p>a list of species.</p>
</td></tr>
<tr><td><code id="starTree_+3A_branch.lengths">branch.lengths</code></td>
<td>
<p>an optional list of branch lengths in the same order as <code>species</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a star phylogeny with (optionally) user specified branch lengths.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+stree">stree</a></code>
</p>

<hr>
<h2 id='strahlerNumber'>Computes Strahler number for trees and nodes</h2><span id='topic+strahlerNumber'></span><span id='topic+extract.strahlerNumber'></span>

<h3>Description</h3>

<p>Computes the Strahler number of all nodes and tips in a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strahlerNumber(tree, plot=TRUE)
extract.strahlerNumber(tree, i, plot=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strahlerNumber_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="strahlerNumber_+3A_i">i</code></td>
<td>
<p>order of Strahler number to extract for <code>extract.strahlerNumber</code>.</p>
</td></tr>
<tr><td><code id="strahlerNumber_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether to plot the tree with Strahler numbers for node labels.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>strahlerNumber</code> computes the Strahler number of all nodes and tips in the tree. For more information about Strahler numbers see <a href="https://en.wikipedia.org/wiki/Strahler_number">https://en.wikipedia.org/wiki/Strahler_number</a>. The function <code>extract.strahlerNumber</code> extracts all of the most inclusive clades of Strahler number <code>i</code>.
</p>


<h3>Value</h3>

<p>Either a vector with the Strahler number for each tip and internal node; or (for <code>extract.strahlerNumber</code> the set of (most inclusive) subtrees with Strahler number <code>i</code> as an object of class <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='threshBayes'>Threshold model using Bayesian MCMC</h2><span id='topic+threshBayes'></span>

<h3>Description</h3>

<p>Fits the threshold model from quantitative genetics to data for two discrete characters or one discrete and one continuous trait, following Felsenstein (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshBayes(tree, X, types=NULL, ngen=10000, control=list(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threshBayes_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="threshBayes_+3A_x">X</code></td>
<td>
<p>a matrix or data frame containing values for a discrete character and a continuous character; or two discrete characters. The row names of <code>X</code> should be species names.</p>
</td></tr>
<tr><td><code id="threshBayes_+3A_types">types</code></td>
<td>
<p>a vector of length <code>ncol(X)</code> containing the data types for each column of <code>X</code>, for instance <code>c("discrete","continuous")</code>.</p>
</td></tr>
<tr><td><code id="threshBayes_+3A_ngen">ngen</code></td>
<td>
<p>a integer indicating the number of generations for the MCMC.</p>
</td></tr>
<tr><td><code id="threshBayes_+3A_control">control</code></td>
<td>
<p>a list of control parameters for the MCMC. Control parameters include: <code>sample</code>, the sampling interval for the MCMC; <code>propvar</code>, a vector containing (in this order) proposal variances for the two rates (if the type is <code>"discrete"</code> this will be ignored), the two ancestral states, and the correlation; <code>propliab</code>, a single proposal variance for the liabilities; <code>pr.mean</code>, a vector for the mean of the prior probability distributions for each parameter, in the same order as <code>propvar</code>; <code>pr.liab</code>, currently ignored; <code>pr.var</code>, a vector with variances for the prior densities for each parameter, in the same order as <code>pr.mean</code> - note that for the rates we use an exponential distribution so the first two means are currently ignored; and <code>pr.vliab</code> currently ignored.</p>
</td></tr>
<tr><td><code id="threshBayes_+3A_...">...</code></td>
<td>
<p>other optional arguments. The argument <code>auto.tune</code> adjusts the proposal distribution to target an proposal acceptance rate (of 0.234, by default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses Bayesian MCMC to fit the quantitative genetics threshold model (Felsenstein 2012) to data for two discrete characters or one discrete and one continuous character.
</p>
<p>The <code>plot</code> method for the object class can be used to generate a three panel plot showing the likelihood profile, the mean acceptance rates (using a sliding window), and a profile plot for the correlation coefficient, <em>r</em>.
</p>
<p>The <code>density</code> method for the object can be used to plot a posterior density of the correlation coefficient, <em>r</em>. This posterior density is of class <code>"density.threshBayes"</code> which can also be plotted using an <code>plot</code> method.
</p>
<p>Discrete characters must be binary, but can be coded as any factor.
</p>


<h3>Value</h3>

<p>This function returns an object of class <code>"threshBayes"</code> consisting of a list with at least the following two elements: <code>par</code> a matrix containing the posterior sample for the model parameters (evolutionary rates, ancestral states, and correlation); <code>liab</code> a matrix containing the posterior sample of the liabilities. For continuous characters, the liabilities are treated as known and so the posterior samples are just the observed values.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Felsenstein, J. (2012) A comparative method for both discrete and continuous characters using the threshold model. <em>American Naturalist</em>, <b>179</b>, 145-156.
</p>
<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+anc.Bayes">anc.Bayes</a></code>, <code><a href="#topic+bmPlot">bmPlot</a></code>, <code><a href="#topic+evol.rate.mcmc">evol.rate.mcmc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## load data
data(bonyfish.tree)
data(bonyfish.data)
## run MCMC
mcmc&lt;-threshBayes(bonyfish.tree,bonyfish.data,
    ngen=100000,plot=FALSE)
## visualize summary of MCMC
plot(mcmc)
## reset par
par(mfrow=c(1,1),mar=c(5.1,4.1,4.1,2.1))
## graph posterior density of correlation
plot(density(mcmc,bw=0.1))
## End(Not run)
</code></pre>

<hr>
<h2 id='threshDIC'>Deviance Information Criterion from the threshold model</h2><span id='topic+threshDIC'></span>

<h3>Description</h3>

<p>Computes Deviance Information Criterion from the MCMC object returned by <code><a href="#topic+ancThresh">ancThresh</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshDIC(tree, x, mcmc, burnin=NULL, sequence=NULL, method="pD")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threshDIC_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree.</p>
</td></tr>
<tr><td><code id="threshDIC_+3A_x">x</code></td>
<td>
<p>a named vector containing discrete character states; or a matrix containing the tip species, in rows, and probabilities of being in each state, in columns.</p>
</td></tr>
<tr><td><code id="threshDIC_+3A_mcmc">mcmc</code></td>
<td>
<p>list object returned by <code><a href="#topic+ancThresh">ancThresh</a></code>.</p>
</td></tr>
<tr><td><code id="threshDIC_+3A_burnin">burnin</code></td>
<td>
<p>number of generations (not samples) to exclude as burn in; if not supplied then 20% of generations are excluded.</p>
</td></tr>
<tr><td><code id="threshDIC_+3A_sequence">sequence</code></td>
<td>
<p>assumed ordering of the discrete character state. If not supplied and <code>x</code> is a vector then numerical-alphabetical order is assumed; if not supplied and <code>x</code> is a matrix, then the column order of <code>x</code> is used.</p>
</td></tr>
<tr><td><code id="threshDIC_+3A_method">method</code></td>
<td>
<p>method for computing the effective number of parameters (options are <code>"pD"</code> and <code>"pV"</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the Deviance Information Criterion from the MCMC object returned by <code><a href="#topic+ancThresh">ancThresh</a></code>.
</p>


<h3>Value</h3>

<p>A vector containing the mean deviance and deviance for the parameter means, the effective number of parameters, and the DIC.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Spiegelhalter, D. J., N. G. Best, B. P. Carlin, and A. Van Der Linde (2002) Bayesian measures of model complexity and fit. <em>Journal of the Royal Statistical Society Series B: Statistical Methodology</em>, <b>64</b>, 583-639.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ancThresh">ancThresh</a></code>
</p>

<hr>
<h2 id='threshState'>Computes value for a threshold character from a liability and thresholds</h2><span id='topic+threshState'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+ancThresh">ancThresh</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshState(x, thresholds)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threshState_+3A_x">x</code></td>
<td>
<p>liability.</p>
</td></tr>
<tr><td><code id="threshState_+3A_thresholds">thresholds</code></td>
<td>
<p>a named vector containing the thresholds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>threshState</code> can also be used to simulate threshold traits.
</p>


<h3>Value</h3>

<p>A discrete character value.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2014) Ancestral character estimation under the threshold model from quantitative genetics. <em>Evolution</em>, <b>68</b>, 743-759.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ancThresh">ancThresh</a></code>, <code><a href="#topic+threshDIC">threshDIC</a></code>
</p>

<hr>
<h2 id='to.matrix'>Convert a character vector to a binary matrix</h2><span id='topic+to.matrix'></span>

<h3>Description</h3>

<p>Creates a binary matrix, normally from a factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to.matrix(x, seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to.matrix_+3A_x">x</code></td>
<td>
<p>a vector of characters.</p>
</td></tr>
<tr><td><code id="to.matrix_+3A_seq">seq</code></td>
<td>
<p>the sequence for the columns in the output matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a vector of characters or a factor and computes a binary matrix.
</p>
<p>Primarily designed to be used internally by <code><a href="#topic+make.simmap">make.simmap</a></code> and <code><a href="#topic+rerootingMethod">rerootingMethod</a></code>.
</p>


<h3>Value</h3>

<p>A binary matrix of dimensions <code>length(x)</code> by <code>length(seq)</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='tree.grow'>Creates an animation of a tree growing from left-to-right or upwards</h2><span id='topic+tree.grow'></span>

<h3>Description</h3>

<p>Animates a birth-death tree simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree.grow(..., res=200, direction="rightwards", ladderize=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree.grow_+3A_...">...</code></td>
<td>
<p>arguments to pass to <code><a href="#topic+pbtree">pbtree</a></code>.</p>
</td></tr>
<tr><td><code id="tree.grow_+3A_res">res</code></td>
<td>
<p>number of steps (the resolution of the animation). This also corresponds to the number of frames that will be created if the animation is to be converted to a .gif file.</p>
</td></tr>
<tr><td><code id="tree.grow_+3A_direction">direction</code></td>
<td>
<p>the direction to plot the tree. Only <code>direction="rightwards"</code> (the default) and <code>direction="upwards"</code> are supported.</p>
</td></tr>
<tr><td><code id="tree.grow_+3A_ladderize">ladderize</code></td>
<td>
<p>logical value indicating whether or not to 'ladderize' the plotted tree. (Defaults to <code>TRUE</code>.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simulates a birth-death tree under user-defined conditions and then creates an animation of that tree growing from left-to-right in the plotting device, or upwards.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pbtree">pbtree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## to create a .gif with ImageMagick installed
png(file="pbtree-
tree.grow(b=0.06,d=0.02,t=100)
dev.off()
system("ImageMagick convert -delay 5 -loop 0 *.png pbtree-anim.gif")
## End(Not run)
</code></pre>

<hr>
<h2 id='treeSlice'>Slices the tree at a particular point and returns all subtrees, or the tree rootward of the point</h2><span id='topic+treeSlice'></span>

<h3>Description</h3>

<p>Cut (or slice) a phylogenetic tree at a particular time point.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treeSlice(tree, slice, trivial=FALSE, prompt=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="treeSlice_+3A_tree">tree</code></td>
<td>
<p>is a phylogenetic tree in <code>"phylo"</code> format.</p>
</td></tr>
<tr><td><code id="treeSlice_+3A_slice">slice</code></td>
<td>
<p>a real number indicating the height above the root at which to slice the tree.</p>
</td></tr>
<tr><td><code id="treeSlice_+3A_trivial">trivial</code></td>
<td>
<p>a logical value indicating whether or not to return subtrees with a number of tips less than two (default is <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="treeSlice_+3A_prompt">prompt</code></td>
<td>
<p>logical value indicating whether or not the height of the slice should be given interactively.</p>
</td></tr>
<tr><td><code id="treeSlice_+3A_...">...</code></td>
<td>
<p>for <code>prompt=TRUE</code>, other arguments to be passed to <code><a href="#topic+plotTree">plotTree</a></code>. In addition, the argument <code>orientation</code> can be used to specify whether the <code>"tipwards"</code> subtrees or the <code>"rootwards"</code> phylogeny are/is to be returned by the function call (using those two argument values, respectively).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function slices a tree at a particular height above the root and returns all subtrees or all non-trivial subtrees (i.e., subtrees with more than 1 taxon). Uses <code><a href="ape.html#topic+extract.clade">extract.clade</a></code> in the <span class="pkg">ape</span> package.
</p>
<p><code>treeSlice</code> can also be used to crop the terminal fraction of a tree for <code>orientation="rootwards"</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+extract.clade">extract.clade</a></code>
</p>

<hr>
<h2 id='untangle'>Attempts to untangle crossing branches for plotting</h2><span id='topic+untangle'></span>

<h3>Description</h3>

<p>Attempts to &quot;untangle&quot; the branches of a tree that are tangled when plotting with <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>, <code><a href="#topic+plotTree">plotTree</a></code>, or <code><a href="#topic+plotSimmap">plotSimmap</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>untangle(tree, method=c("reorder","read.tree"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="untangle_+3A_tree">tree</code></td>
<td>
<p>tree as an object of class <code>"phylo"</code> or <code>"simmap"</code>.</p>
</td></tr>
<tr><td><code id="untangle_+3A_method">method</code></td>
<td>
<p>method to use to attempt to untangle branches. <code>method="reorder"</code> uses two calls of <code><a href="ape.html#topic+reorder.phylo">reorder.phylo</a></code> or <code><a href="#topic+reorderSimmap">reorderSimmap</a></code>; <code>method="read.tree"</code> writes the tree to a text string and then reads it back into memory using <code><a href="ape.html#topic+read.tree">read.tree</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generally speaking, this function is wraps several different internal functions that might be use to fix a badly conformed <code>"phylo"</code> or <code>"simmap"</code> object.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> or <code>"simmap"</code>, depending on the class of <code>tree</code>.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='vcvPhylo'>Calculates cophenetic (i.e., phylogenetic VCV) matrix</h2><span id='topic+vcvPhylo'></span>

<h3>Description</h3>

<p>Computes a phylogenetic variance-covariance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcvPhylo(tree, anc.nodes=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcvPhylo_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td></tr>
<tr><td><code id="vcvPhylo_+3A_anc.nodes">anc.nodes</code></td>
<td>
<p>logical value indicating whether or not to include ancestral nodes.</p>
</td></tr>
<tr><td><code id="vcvPhylo_+3A_...">...</code></td>
<td>
<p>optional arguments including <code>internal</code> (synonym of <code>anc.nodes</code>) and <code>model</code> (can be <code>"BM"</code>, <code>"OU"</code>, or <code>"lambda"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a so-called <em>phylogenetic variance covariance matrix</em> (e.g., see <code><a href="ape.html#topic+vcv.phylo">vcv.phylo</a></code>), but (optionally) including ancestral nodes, as well as under multiple evolutionary models.
</p>
<p><code>vcvPhylo</code> is designed primarily for internal use by other <em>phytools</em> functions.
</p>


<h3>Value</h3>

<p>A matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>

<hr>
<h2 id='write.simmap'>Write a stochastic character mapped tree to file</h2><span id='topic+write.simmap'></span>

<h3>Description</h3>

<p>Writes one or multiple stochastic character mapped trees to file in several formats (following Bollback, 2006).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.simmap(tree, file=NULL, append=FALSE, map.order=NULL, quiet=FALSE, 
    format="phylip", version=1.0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.simmap_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"simmap"</code> or <code>"multiSimmap"</code>.</p>
</td></tr>
<tr><td><code id="write.simmap_+3A_file">file</code></td>
<td>
<p>an optional filename.</p>
</td></tr>
<tr><td><code id="write.simmap_+3A_append">append</code></td>
<td>
<p>a logical value indicating whether to append to file.</p>
</td></tr>
<tr><td><code id="write.simmap_+3A_map.order">map.order</code></td>
<td>
<p>a optional value specifying whether to write the map in left-to-right or right-to-left order.  Acceptable values are &quot;left-to-right&quot; or &quot;right-to-left&quot; or some abbreviation of either.  If not provided, <code>write.simmap</code> will use <code>attr(tree,"map.order")</code> if available.</p>
</td></tr>
<tr><td><code id="write.simmap_+3A_quiet">quiet</code></td>
<td>
<p>logical value indicating whether or not to print a warning message when <code>map.order</code> is neither specified by a function argument or as an attribute of <code>tree</code>.</p>
</td></tr>
<tr><td><code id="write.simmap_+3A_format">format</code></td>
<td>
<p>file format for output.</p>
</td></tr>
<tr><td><code id="write.simmap_+3A_version">version</code></td>
<td>
<p>version of SIMMAP. Note that the options are <code>1.0</code> and <code>1.5</code>. <code>version=1.5</code> is generally recommended because in this format the tree can also be parsed by typical tree readers, but absent its mapped trait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A file.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>Huelsenbeck, J. P., R. Neilsen, and J. P. Bollback (2003) Stochastic mapping of morphological characters. <em>Systematic Biology</em>, <b>52</b>, 131-138. 
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.simmap">make.simmap</a></code>, <code><a href="#topic+read.simmap">read.simmap</a></code>, <code><a href="#topic+plotSimmap">plotSimmap</a></code>
</p>

<hr>
<h2 id='writeAncestors'>Write a tree to file with ancestral states and (optionally) CIs at nodes</h2><span id='topic+writeAncestors'></span>

<h3>Description</h3>

<p>This function writes a tree to file with ancestral character states and (optionally) 95% confidence intervals stored as node value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeAncestors(tree, Anc=NULL, file="", digits=6, format=c("phylip","nexus"), 
   ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeAncestors_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree or set of trees as an object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="writeAncestors_+3A_anc">Anc</code></td>
<td>
<p>a vector of ancestral states, a list containing the ancestral states and 95% confidence intervals (as from <code><a href="#topic+fastAnc">fastAnc</a></code> or <code><a href="ape.html#topic+ace">ace</a></code>, or a list of such results.</p>
</td></tr>
<tr><td><code id="writeAncestors_+3A_file">file</code></td>
<td>
<p>an optional string with the filename for output.</p>
</td></tr>
<tr><td><code id="writeAncestors_+3A_digits">digits</code></td>
<td>
<p>an integer indicating the number of digits to print for branch lengths and ancestral character values.</p>
</td></tr>
<tr><td><code id="writeAncestors_+3A_format">format</code></td>
<td>
<p>a string indicating whether to output the result in simple Newick (i.e., <code>"phylip"</code>) or Nexus format.</p>
</td></tr>
<tr><td><code id="writeAncestors_+3A_...">...</code></td>
<td>
<p>additional arguments including <code>x</code>: a vector of character values, in which case ancestral states are estimated internally using <code>fastAnc</code>; and <code>CI</code>: a logical value indicating whether or not to estimate 95% confidence intervals.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A file, string, or vector of strings.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>, <code><a href="#topic+fastAnc">fastAnc</a></code>, <code><a href="ape.html#topic+write.tree">write.tree</a></code>
</p>

<hr>
<h2 id='writeNexus'>Write a tree to file in Nexus format</h2><span id='topic+writeNexus'></span>

<h3>Description</h3>

<p>Writes one or multiple phylogenetic trees to file in NEXUS format. 
</p>
<p>Somewhat redundant with <code><a href="ape.html#topic+write.nexus">write.nexus</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeNexus(tree, file="")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeNexus_+3A_tree">tree</code></td>
<td>
<p>object of class <code>"phylo"</code> or <code>"multiPhylo"</code>.</p>
</td></tr>
<tr><td><code id="writeNexus_+3A_file">file</code></td>
<td>
<p>file name for output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Trees written to file.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+write.simmap">write.simmap</a></code>, <code><a href="ape.html#topic+write.nexus">write.nexus</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
