<!DOCTYPE html><html><head><title>Help for package picante</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {picante}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#picante-package'>
<p>picante: Integrating Phylogenies and Ecology</p></a></li>
<li><a href='#color.plot.phylo'><p>Color tip labels based on trait</p></a></li>
<li><a href='#comdist'><p> Calculates inter-community mean pairwise distance</p></a></li>
<li><a href='#comdistnt'><p> Calculates inter-community mean nearest taxon distance</p></a></li>
<li><a href='#comm.phylo.cor'><p> Correlations between species co-occurrence and phylogenetic distances</p></a></li>
<li><a href='#comm.phylo.qr'><p> Quantile regression slopes between species co-occurrence and phylogenetic distances</p></a></li>
<li><a href='#cor.table'><p> Table of correlations and P-values</p></a></li>
<li><a href='#evol.distinct'>
<p>Species' evolutionary distinctiveness</p></a></li>
<li><a href='#expected.pd'><p> Expected PD, PD Variance, and Edge Abundance Distribution of a phylogeny</p></a></li>
<li><a href='#IvesGodfray'><p> Host-parasitoid food web data</p></a></li>
<li><a href='#Kcalc'><p> K statistic of phylogenetic signal</p></a></li>
<li><a href='#match.phylo.data'><p> Match taxa in phylogeny and data</p></a></li>
<li><a href='#matrix2sample'><p> Convert community data matrix to Phylocom sample</p></a></li>
<li><a href='#mntd'><p> Mean nearest taxon distance</p></a></li>
<li><a href='#mpd'><p> Mean pairwise distance</p></a></li>
<li><a href='#multiPhylosignal'><p> Calculates phylogenetic signal for data.frame of traits</p></a></li>
<li><a href='#pblm'><p> Phylogenetic Bipartite Linear Model</p></a></li>
<li><a href='#pcd'><p>Phylogenetic Community Dissimilarity</p></a></li>
<li><a href='#pd'><p> Calculate Faith's Phylogenetic Diversity</p></a></li>
<li><a href='#phyEstimate'><p> Phylogenetic estimation of traits for unobserved taxa</p></a></li>
<li><a href='#phylocom'><p> Phylocom default data</p></a></li>
<li><a href='#phylosignal'><p> Measure phylogenetic signal</p></a></li>
<li><a href='#phylosor'><p>Phylogenetic index of beta-diversity PhyloSor</p></a></li>
<li><a href='#phylosor.rnd'><p> Null PhyloSor values of phylogenetic beta-diversity</p></a></li>
<li><a href='#phylostruct'><p> Permutations to Test for Phylogenetic Signal in Community Composition</p></a></li>
<li><a href='#prune.sample'><p> Prune tree to match community data or trait data</p></a></li>
<li><a href='#psd'><p> Phylogenetic Species Diversity Metrics</p></a></li>
<li><a href='#randomizeMatrix'><p> Null models for community data matrix randomization</p></a></li>
<li><a href='#raoD'><p> Rao's quadratic entropy</p></a></li>
<li><a href='#readsample'><p> Read Phylocom sample</p></a></li>
<li><a href='#sample2matrix'><p> Convert Phylocom sample to community data matrix</p></a></li>
<li><a href='#ses.mntd'><p> Standardized effect size of MNTD</p></a></li>
<li><a href='#ses.mpd'><p> Standardized effect size of MPD</p></a></li>
<li><a href='#ses.pd'><p> Standardized effect size of PD</p></a></li>
<li><a href='#specaccum.psr'><p> Phylogenetic Species Richness Sample-Based Rarefaction Curve</p></a></li>
<li><a href='#species.dist'><p> Species co-occurrence distances</p></a></li>
<li><a href='#sppregs'><p> Regressions to Separate Phylogenetic Attraction and Repulsion</p></a></li>
<li><a href='#tax.distinctiveness'>
<p>Taxonomic distinctiveness sensu Vane-Wright or May</p></a></li>
<li><a href='#traitgram'><p> Draw phylogeny with nodes at trait positions</p></a></li>
<li><a href='#unifrac'><p> Unweighted UniFrac distance between communities</p></a></li>
<li><a href='#utility'><p> Picante utility functions</p></a></li>
<li><a href='#writesample'><p> Write a Phylocom community sample file</p></a></li>
<li><a href='#writetraits'><p> Write a Phylocom traits formatted file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrating Phylogenies and Ecology</td>
</tr>
<tr>
<td>Version:</td>
<td>1.8.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-06-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Steven W. Kembel &lt;steve.kembel@gmail.com&gt;, David D. Ackerly &lt;dackerly@berkeley.edu&gt;, Simon P. Blomberg &lt;s.blomberg1@uq.edu.au&gt;, Will K. Cornwell &lt;cornwell@zoology.ubc.ca&gt;, Peter D. Cowan &lt;pdc@berkeley.edu&gt;, Matthew R. Helmus &lt;mrhelmus@wisc.edu&gt;, Helene Morlon &lt;morlon.helene@gmail.com&gt;, Campbell O. Webb &lt;cwebb@oeb.harvard.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Steven W. Kembel &lt;steve.kembel@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>ape, vegan, nlme, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>brglm, circular, corpcor, quantreg</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for phylocom integration, community analyses, null-models, traits and evolution. Implements numerous ecophylogenetic approaches including measures of community phylogenetic and trait diversity, phylogenetic signal, estimation of trait values for unobserved taxa, null models for community and phylogeny randomizations, and utility functions for data input/output and phylogeny plotting. A full description of package functionality and methods are provided by Kembel et al. (2010) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtq166">doi:10.1093/bioinformatics/btq166</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-10 14:53:27 UTC; steve</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-10 15:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='picante-package'>
picante: Integrating Phylogenies and Ecology
</h2><span id='topic+picante-package'></span><span id='topic+picante'></span>

<h3>Description</h3>

<p>Functions for phylocom integration, community analyses, null-models, traits and evolution. Implements numerous ecophylogenetic approaches including measures of community phylogenetic and trait diversity, phylogenetic signal, estimation of trait values for unobserved taxa, null models for community and phylogeny randomizations, and utility functions for data input/output and phylogeny plotting. A full description of package functionality and methods are provided by Kembel et al. (2010) &lt;doi:10.1093/bioinformatics/btq166&gt;.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> picante</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.8.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-06-08</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Author: Steven W. Kembel &lt;steve.kembel@gmail.com&gt;, David D. Ackerly &lt;dackerly@berkeley.edu&gt;, Simon P. Blomberg &lt;s.blomberg1@uq.edu.au&gt;, Will K. Cornwell &lt;cornwell@zoology.ubc.ca&gt;, Peter D. Cowan &lt;pdc@berkeley.edu&gt;, Matthew R. Helmus &lt;mrhelmus@wisc.edu&gt;, Helene Morlon &lt;morlon.helene@gmail.com&gt;, Campbell O. Webb &lt;cwebb@oeb.harvard.edu&gt;
Maintainer: Steven W. Kembel &lt;steve.kembel@gmail.com&gt;
</p>

<hr>
<h2 id='color.plot.phylo'>Color tip labels based on trait</h2><span id='topic+color.plot.phylo'></span>

<h3>Description</h3>

<p>Plots a phylogeny with tip labels colored to indicate continuous or discrete trait values</p>


<h3>Usage</h3>

<pre><code class='language-R'>  color.plot.phylo(phylo, df, trait, taxa.names,
                    num.breaks = ifelse(is.factor(df[,trait]),
                        length(levels(df[,trait])), 12),
                    col.names = rainbow(ifelse(length(num.breaks) &gt; 1,
                        length(num.breaks) - 1, num.breaks)),
                    cut.labs = NULL,
                    leg.title = NULL,
                    main = trait,
                    leg.cex = 1,
                    tip.labs = NULL,
                    ...)
        </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="color.plot.phylo_+3A_phylo">phylo</code></td>
<td>
<p>An object of class <code>phylo</code></p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_df">df</code></td>
<td>
<p>A dataframe containing the traits to be plotted</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_trait">trait</code></td>
<td>
<p>A string representing the name of column in the dataframe to be plotted</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_taxa.names">taxa.names</code></td>
<td>
<p>A string representing the name of column in the dataframe that contains the names of the taxa</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_num.breaks">num.breaks</code></td>
<td>
<p>For continuous traits, the number of bins to separate the data into</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_col.names">col.names</code></td>
<td>
<p>A vector of colors to use for tip labels</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_leg.title">leg.title</code></td>
<td>
<p>A title for the tip color legend</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_main">main</code></td>
<td>
<p>A main title for the plot</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_cut.labs">cut.labs</code></td>
<td>
<p>A main title for the plot</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_leg.cex">leg.cex</code></td>
<td>
<p>A main title for the plot</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_tip.labs">tip.labs</code></td>
<td>
<p>A main title for the plot</p>
</td></tr>
<tr><td><code id="color.plot.phylo_+3A_...">...</code></td>
<td>
<p>Additional argument to pass to the <code>plot.phylo</code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If if <code>trait</code> is a factor then each level of the factor is plotted with the corresponding <code>col.names</code> value (if <code>length(num.breaks) &gt; length(col.names)</code> colors are recycled.) If <code>trait</code> is not a factor then it is assumed to be continuous and <code>trait</code> is evenly divided into <code>num.breaks</code> levels.
</p>


<h3>Value</h3>

<p>The command is invoked for its side effect, a plot of the <code>phylo</code> with tips colored based on <code>trait</code>
</p>


<h3>Author(s)</h3>

<p> Peter Cowan &lt;pdc@berkeley.edu&gt; </p>

<hr>
<h2 id='comdist'> Calculates inter-community mean pairwise distance </h2><span id='topic+comdist'></span>

<h3>Description</h3>

<p>Calculates MPD (mean pairwise distance) separating taxa in two communities, a measure of phylogenetic beta diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comdist(comm, dis, abundance.weighted = FALSE)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comdist_+3A_comm">comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="comdist_+3A_dis">dis</code></td>
<td>
<p> Interspecific distance matrix </p>
</td></tr>
<tr><td><code id="comdist_+3A_abundance.weighted">abundance.weighted</code></td>
<td>
<p> Should mean pairwise distances separating species in two communities be weighted by species abundances? (default = FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates a measure of phylogenetic beta diversity: the expected phylogenetic distance separating two individuals or taxa drawn randomly from different communities.
</p>


<h3>Value</h3>

<p>Distance object of MPD values separating each pair of communities.
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> C.O. Webb, D.D. Ackerly, and S.W. Kembel. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Bioinformatics 18:2098-2100. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+mpd">mpd</a></code>, <code><a href="#topic+ses.mpd">ses.mpd</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
comdist(phylocom$sample, cophenetic(phylocom$phylo), abundance.weighted=TRUE)</code></pre>

<hr>
<h2 id='comdistnt'> Calculates inter-community mean nearest taxon distance </h2><span id='topic+comdistnt'></span><span id='topic+comdistnn'></span>

<h3>Description</h3>

<p>Calculates MNTD (mean nearest taxon distance) separating taxa in two communities, a measure of phylogenetic beta diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comdistnt(comm, dis, abundance.weighted = FALSE, exclude.conspecifics = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comdistnt_+3A_comm">comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="comdistnt_+3A_dis">dis</code></td>
<td>
<p> Interspecific distance matrix </p>
</td></tr>
<tr><td><code id="comdistnt_+3A_abundance.weighted">abundance.weighted</code></td>
<td>
<p> Should mean nearest taxon distances from each species to species in the other community be weighted by species abundance? (default = FALSE)</p>
</td></tr>
<tr><td><code id="comdistnt_+3A_exclude.conspecifics">exclude.conspecifics</code></td>
<td>
<p> Should conspecific taxa in different communities be exclude from MNTD calculations? (default = FALSE) </p>
</td></tr>  
</table>


<h3>Details</h3>

<p>This metric has also been referred to as MNND (mean nearest neighbour distance).
</p>
<p>This function calculates a measure of phylogenetic beta diversity: the  average phylogenetic distance to the most similar taxon or individual in the other community for taxa or individuals in two communities.
</p>


<h3>Value</h3>

<p>Distance object of MNTD values separating each pair of communities.
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> C.O. Webb, D.D. Ackerly, and S.W. Kembel. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Bioinformatics 18:2098-2100. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+mntd">mntd</a></code>, <code><a href="#topic+ses.mntd">ses.mntd</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
comdistnt(phylocom$sample, cophenetic(phylocom$phylo), abundance.weighted=FALSE)</code></pre>

<hr>
<h2 id='comm.phylo.cor'> Correlations between species co-occurrence and phylogenetic distances </h2><span id='topic+comm.phylo.cor'></span>

<h3>Description</h3>

<p>Calculates measures of community phylogenetic structure (correlation between co-occurrence and phylogenetic distance) to patterns expected under various null models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comm.phylo.cor(samp, phylo, metric = c("cij", "checkerboard", "jaccard", "doij"),
    null.model = c("sample.taxa.labels", "pool.taxa.labels",
        "frequency", "richness", "independentswap","trialswap"), runs = 999, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comm.phylo.cor_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="comm.phylo.cor_+3A_phylo">phylo</code></td>
<td>
<p> Phylogenetic tree </p>
</td></tr>
<tr><td><code id="comm.phylo.cor_+3A_metric">metric</code></td>
<td>
<p> Metric of co-occurrence to use (see <code><a href="#topic+species.dist">species.dist</a></code>)</p>
</td></tr>
<tr><td><code id="comm.phylo.cor_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section for description) </p>
</td></tr>
<tr><td><code id="comm.phylo.cor_+3A_runs">runs</code></td>
<td>
<p> Number of runs (randomizations) </p>
</td></tr>
<tr><td><code id="comm.phylo.cor_+3A_...">...</code></td>
<td>
<p> Additional arguments to <a href="#topic+randomizeMatrix">randomizeMatrix</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>sample.taxa.labels</dt><dd><p>Shuffle phylogeny tip labels (only within set of taxa present in community data)</p>
</dd>
<dt>pool.taxa.labels</dt><dd><p>Shuffle phylogeny tip labels (across all taxa included in phylogenetic tree)</p>
</dd>
<dt>frequency</dt><dd><p>Randomize community data matrix abundances within species (maintains species occurence frequency)</p>
</dd>
<dt>richness</dt><dd><p>Randomize community data matrix abundances within samples (maintains sample species richness)</p>
</dd>
<dt>independentswap</dt><dd><p>Randomize community data matrix maintaining species occurrence frequency and site richnessing using independent swap</p>
</dd>
<dt>trialswap</dt><dd><p>Randomize community data matrix maintaining species occurrence frequency and site richnessing using trial swap</p>
</dd>     
</dl>



<h3>Value</h3>

<p>A list with elements:
</p>
<table>
<tr><td><code>obs.corr</code></td>
<td>
<p> Observed co-occurrence/phylogenetic distance correlation</p>
</td></tr>
<tr><td><code>obs.corr.p</code></td>
<td>
<p> P-value of observed correlation (standard P-value for correlation coefficient, not based on comparison with randomizations)</p>
</td></tr>
<tr><td><code>obs.rank</code></td>
<td>
<p> Rank of observed correlation vs. random</p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p> Number of runs (randomizations) </p>
</td></tr>
<tr><td><code>obs.rand.p</code></td>
<td>
<p> P-value of observed correlation vs. randomizations (= obs.rank / (runs + 1))</p>
</td></tr>
<tr><td><code>random.corrs</code></td>
<td>
<p>A vector of random correlation calculated for each run</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Cavender-Bares J., D.A. Ackerly, D. Baum and F.A. Bazzaz. 2004. Phylogenetic overdispersion in Floridian oak communities, American Naturalist, 163(6):823-843. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
comm.phylo.cor(phylocom$sample, phylocom$phylo, metric="cij",null.model="sample.taxa.labels")</code></pre>

<hr>
<h2 id='comm.phylo.qr'> Quantile regression slopes between species co-occurrence and phylogenetic distances </h2><span id='topic+comm.phylo.qr'></span>

<h3>Description</h3>

<p>Calculates measures of community phylogenetic structure (quantile regression  between co-occurrence and phylogenetic distance) to patterns expected under various null models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comm.phylo.qr(samp, phylo, metric = c("cij", "checkerboard", "jaccard", "doij"),
    null.model = c("sample.taxa.labels", "pool.taxa.labels",
        "frequency", "richness", "independentswap","trialswap"), 
        quant = 0.75, runs = 999, show.plot = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comm.phylo.qr_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_phylo">phylo</code></td>
<td>
<p> Phylogenetic tree </p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_metric">metric</code></td>
<td>
<p> Metric of co-occurrence to use (see <code><a href="#topic+species.dist">species.dist</a></code>)</p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section for description) </p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_quant">quant</code></td>
<td>
<p> Quantile of slope to be fit (using <code><a href="quantreg.html#topic+rq">rq</a></code>) </p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_runs">runs</code></td>
<td>
<p> Number of runs (randomizations) </p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_show.plot">show.plot</code></td>
<td>
<p> Option to display a plot of co-occurrence versus phylogenetic distance with quantile regression slope fit </p>
</td></tr>
<tr><td><code id="comm.phylo.qr_+3A_...">...</code></td>
<td>
<p> Additional arguments to <a href="#topic+randomizeMatrix">randomizeMatrix</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a quantile regression of co-occurrence versus phylogenetic distances separating species, and compares observed patterns to the patterns expected under some null model. The quantile regressions are fit using the <code><a href="quantreg.html#topic+rq">rq</a></code> function from the <code>quantreg</code> package.
</p>
<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>sample.taxa.labels</dt><dd><p>Shuffle phylogeny tip labels (only within set of taxa present in community data)</p>
</dd>
<dt>pool.taxa.labels</dt><dd><p>Shuffle phylogeny tip labels (across all taxa included in phylogenetic tree)</p>
</dd>
<dt>frequency</dt><dd><p>Randomize community data matrix abundances within species (maintains species occurence frequency)</p>
</dd>
<dt>richness</dt><dd><p>Randomize community data matrix abundances within samples (maintains sample species richness)</p>
</dd>
<dt>independentswap</dt><dd><p>Randomize community data matrix maintaining species occurrence frequency and site richnessing using independent swap</p>
</dd>
<dt>trialswap</dt><dd><p>Randomize community data matrix maintaining species occurrence frequency and site richnessing using trial swap</p>
</dd>     
</dl>



<h3>Value</h3>

<p>A list with elements:
</p>
<table>
<tr><td><code>obs.qr.intercept</code></td>
<td>
<p> Observed co-occurrence/phylogenetic distance quantile regression intercept</p>
</td></tr>
<tr><td><code>obs.qr.slope</code></td>
<td>
<p> Observed co-occurrence/phylogenetic distance quantile regression slope</p>
</td></tr>    
<tr><td><code>obs.qr.slope.p</code></td>
<td>
<p> P-value of observed quantile regression slope significance versus null model (calculated based on comparison with randomizations)</p>
</td></tr>
<tr><td><code>obs.rank</code></td>
<td>
<p> Rank of observed quantile regression slope vs. random</p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p> Number of runs (randomizations) </p>
</td></tr>
<tr><td><code>random.qr.slopes</code></td>
<td>
<p>A vector of quantile regression slopes calculated for each randomization</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Cavender-Bares J., D.A. Ackerly, D. Baum and F.A. Bazzaz. 2004. Phylogenetic overdispersion in Floridian oak communities, American Naturalist, 163(6):823-843.
Slingsby, J. A. and G. A. Verboom. 2006. Phylogenetic relatedness limits coexistence at fine spatial scales: evidence from the schoenoid sedges (Cyperaceae: Schoeneae) of the Cape Floristic Region, South Africa. The American Naturalist 168:14-27.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
comm.phylo.qr(phylocom$sample, phylocom$phylo, metric="cij",
  null.model="sample.taxa.labels", runs=99)</code></pre>

<hr>
<h2 id='cor.table'> Table of correlations and P-values </h2><span id='topic+cor.table'></span>

<h3>Description</h3>

<p>Table of correlations with associated P-values and df, can be used with regular or independent contrast data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor.table(x, cor.method = c("pearson","spearman"),
    cor.type=c("standard","contrast")) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cor.table_+3A_x">x</code></td>
<td>
<p> Data frame of data points or contrasts at nodes </p>
</td></tr>
<tr><td><code id="cor.table_+3A_cor.method">cor.method</code></td>
<td>
<p> Correlation method (as <code><a href="stats.html#topic+cor">cor</a></code>) </p>
</td></tr>
<tr><td><code id="cor.table_+3A_cor.type">cor.type</code></td>
<td>
<p> Are data <code>standard</code> or independent <code>contrast</code> values?</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>r</code></td>
<td>
<p>Correlation values</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>P-values</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Garland, T., Jr., P. H. Harvey, and A. R. Ives. 1992. Procedures for the analysis of comparative data using phylogenetically independent contrasts. Systematic Biology 41:18-32. </p>

<hr>
<h2 id='evol.distinct'>
Species' evolutionary distinctiveness
</h2><span id='topic+evol.distinct'></span>

<h3>Description</h3>

<p>Calculates evolutionary distinctiveness measures for a suite of species by: a) equal splits (Redding and Mooers 2006) b) fair proportions (Isaac et al., 2007). Returns a datafram with species identifiers and species scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evol.distinct(tree, type = c("equal.splits", "fair.proportion"),
    scale = FALSE, use.branch.lengths = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evol.distinct_+3A_tree">tree</code></td>
<td>

<p>an object of class phylo</p>
</td></tr>
<tr><td><code id="evol.distinct_+3A_type">type</code></td>
<td>

<p>a) equal splits (Redding and Mooers 2006) or b) fair proportions (Isaac et al., 2007)
</p>
</td></tr>
<tr><td><code id="evol.distinct_+3A_scale">scale</code></td>
<td>

<p>The scale option refers to whether or not the phylogeny should be scaled to a depth of 1 or, in the case of an ultrametric tree,  scaled such that branch lengths are relative.
</p>
</td></tr>
<tr><td><code id="evol.distinct_+3A_use.branch.lengths">use.branch.lengths</code></td>
<td>

<p>If use.branch.lengths=FALSE, then all branch lengths are changed to 1.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function will return a vector of evolutionary distinctivenss for every species in the given tree.
If only a subset of values are needed there are two, concetually distinct options: either prune the tree first and then pass the tree in or subset the resulting vector.  These two options will provide very different outputs.  
</p>


<h3>Author(s)</h3>

<p>Karen Magnuson-Ford, Will Cornwell, Arne Mooers, Mark Vellend
</p>


<h3>References</h3>

<p>Redding, D.W. and Mooers, A.O. (2006). Incorporating evolutionary measures into conservation prioritisation. Conservation Biology, 20, 1670-1678.
</p>
<p>Isaac, N.J.B., Turvey, S.T., Collen, B., Waterman, C. and Baillie, J.E.M. (2007). Mammals on the EDGE: conservation priorities based on threat and phylogeny. PLoS ONE, 2, e296.
</p>
<p>Mark Vellend, William K. Cornwell, Karen Magnuson-Ford, and Arne Mooers. In press. Measuring phylogenetic biodiversity. In: Biological diversity: frontiers in measurement and assessment.  Edited by Anne Magurran and Brian McGill.
</p>

<hr>
<h2 id='expected.pd'> Expected PD, PD Variance, and Edge Abundance Distribution of a phylogeny </h2><span id='topic+expected.pd'></span><span id='topic+variance.pd'></span><span id='topic+ead'></span>

<h3>Description</h3>

<p> Calculates the expected phylogenetic diversity (Faith's PD) and variance of PD under binomial sampling with a fixed probability of each tip being sampled, and the Edge-length Abundance Distribution of a phylogeny. </p>


<h3>Usage</h3>

<pre><code class='language-R'>expected.pd(phy)
variance.pd(phy, upper.bound=TRUE)
ead(phy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expected.pd_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="expected.pd_+3A_upper.bound">upper.bound</code></td>
<td>
<p> Calculate upper bound of PD variance? (default = TRUE) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>expected.pd</code> calculates the expected phylogenetic diversity (Faith's PD - total branch length) for all subsets of a phylogeny, based on an analytic solution for expected PD.
</p>
<p>The function <code>variance.pd</code> additionally calculates the variance of expected PD for all subsets of a phylogeny, based on an analytic solution for expected PD. If argument upper.bound=TRUE, a fast solution for the upper bound of the variance is returned. Otherwise, the exact solution for the variance is returned. Note that the exact solution is much slower than the upper bound solution.
</p>
<p>The function <code>ead</code> calculates the edge abundance distribution (EAD), the length of edges with different numbers of descendant tips.
</p>


<h3>Value</h3>

<table>
<tr><td><code>n</code></td>
<td>
<p> Expected Number of tips sampled </p>
</td></tr>
<tr><td><code>expected.pd</code></td>
<td>
<p> Expected PD for a given n </p>
</td></tr>
<tr><td><code>variance.pd</code></td>
<td>
<p> Variance of PD for a given n </p>
</td></tr>
<tr><td><code>num.children</code></td>
<td>
<p> Number of tips descended from an edge </p>
</td></tr>
<tr><td><code>edge.length</code></td>
<td>
<p> Total phylogenetic edge length for a given number of tips descended from an edge </p>
</td></tr>      
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; and James O'Dwyer &lt;jodwyer@santafe.edu&gt; </p>


<h3>References</h3>

<p> J.P. O'Dwyer, S.W. Kembel, and J.L. Green. 2012. Phylogenetic Diversity Theory Sheds Light on the Structure of Microbial Communities. PLoS Comput Biol 8(12): e1002832. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+pd">pd</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>randtree &lt;- rcoal(300)
randtree.pd.ub &lt;- variance.pd(randtree, upper.bound=TRUE)
randtree.pd.exact &lt;- variance.pd(randtree, upper.bound=FALSE)
plot(expected.pd(randtree), xlab="Number of tips",
    ylab="Phylogenetic diversity (PD)", type="l", log="xy")
lines(randtree.pd.exact$expected.pd+1.96*sqrt(randtree.pd.exact$variance.pd), lty=2)
lines(randtree.pd.exact$expected.pd-1.96*sqrt(randtree.pd.exact$variance.pd), lty=2)
lines(randtree.pd.ub$expected.pd+1.96*sqrt(randtree.pd.ub$variance.pd), lty=3)
lines(randtree.pd.ub$expected.pd-1.96*sqrt(randtree.pd.ub$variance.pd), lty=3)
legend("bottomright", lty=c(1,2,3), legend=c("Expected PD",
    "95 percent CI (exact)","95 percent CI (upper bound)"))
</code></pre>

<hr>
<h2 id='IvesGodfray'> Host-parasitoid food web data </h2><span id='topic+IvesGodfray'></span>

<h3>Description</h3>

<p>Data on the structure of a host-parasitoid food web from Ives &amp; Godfray (2006). Includes information on phylogenetic covariances among 12 leaf-mining moth hosts and 27 species of parasitoid wasps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(IvesGodfray)</code></pre>


<h3>Format</h3>

<p>A list with three elements:
</p>

<ul>
<li><p> host  Phylogenetic variance/covariance matrix for 12 leaf-mining moth hosts 
</p>
</li>
<li><p> parasitoid  Phylogenetic variance/covariance matrix for 27 species of parasitoid wasps 
</p>
</li>
<li><p> interactions  Matrix describing interactions between hosts and parasitoids 
</p>
</li></ul>



<h3>Source</h3>

<p>Ives A.R. &amp; Godfray H.C. (2006) Phylogenetic analysis of trophic associations. The American Naturalist, 168, E1-E14
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+pblm">pblm</a></code> </p>

<hr>
<h2 id='Kcalc'> K statistic of phylogenetic signal </h2><span id='topic+Kcalc'></span>

<h3>Description</h3>

<p>Calculates K statistic of phylogenetic signal
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Kcalc(x, phy, checkdata=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Kcalc_+3A_x">x</code></td>
<td>
<p> Vector or data.frame of trait data (in phylo\$tip.label order) </p>
</td></tr>
<tr><td><code id="Kcalc_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="Kcalc_+3A_checkdata">checkdata</code></td>
<td>
<p> Check for match between trait and phylogeny taxa labels using <code><a href="#topic+match.phylo.data">match.phylo.data</a></code>? (default=TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>K</code></td>
<td>
<p>K statistic</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Simon Blomberg &lt;s.blomberg1@uq.edu.au&gt; and David Ackerly &lt;dackerly@berkeley.edu&gt; </p>


<h3>References</h3>

<p> Blomberg, S. P., and T. Garland, Jr. 2002. Tempo and mode in evolution: phylogenetic inertia, adaptation and comparative methods. Journal of Evolutionary Biology 15:899-910.
</p>
<p>Blomberg, S. P., T. Garland, Jr., and A. R. Ives. 2003. Testing for phylogenetic signal in comparative data: behavioral traits are more labile. Evolution 57:717-745.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+phylosignal">phylosignal</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
randtree &lt;- rcoal(20)
randtraits &lt;- rTraitCont(randtree)
Kcalc(randtraits[randtree$tip.label],randtree)

</code></pre>

<hr>
<h2 id='match.phylo.data'> Match taxa in phylogeny and data </h2><span id='topic+match.phylo.data'></span><span id='topic+match.phylo.comm'></span><span id='topic+match.comm.dist'></span>

<h3>Description</h3>

<p>These functions compare taxa present in phylogenies with community or trait data sets, pruning and sorting the two kinds of data to match one another for subsequent analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match.phylo.comm(phy, comm)
match.phylo.data(phy, data)
match.comm.dist(comm, dis)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match.phylo.data_+3A_phy">phy</code></td>
<td>
<p> A phylogeny object of class phylo </p>
</td></tr>
<tr><td><code id="match.phylo.data_+3A_comm">comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="match.phylo.data_+3A_data">data</code></td>
<td>
<p> A data object - a vector (with names matching phy) or a data.frame or matrix (with row names matching phy) </p>
</td></tr>
<tr><td><code id="match.phylo.data_+3A_dis">dis</code></td>
<td>
<p> A distance matrix - a dist or matrix object </p>
</td></tr>
</table>


<h3>Details</h3>

<p>A common pitfall in comparative analyses in R is that taxa labels are assumed to match between phylogenetic and other data sets. These functions prune a phylogeny and community or trait data set to match one another, reporting taxa that are missing from one data set or the other.
</p>
<p>Taxa names for phylogeny objects are taken from the phylogeny's tip labels. Taxa names for community data are taken from the column names. Taxa names for trait data are taken from the element names (vector) or row names (data.frame or matrix). Taxa names for distance data are taken from column/row names of the distance matrix/dist object.
</p>
<p>If community data lack taxa names, the function will issue a warning and no result will be returned, since the community-phylogenetic analyses in <code>picante</code> require named taxa in the community data set.
</p>
<p>If trait data or distance matrix lack names, a warning is issued and the data are assumed to be sorted in the same order as the phylogeny's tip labels or community's column labels.
</p>
<p>These utility functions are used by several functions that assume taxa labels in phylogeny and data match, including <code><a href="#topic+Kcalc">Kcalc</a></code>, <code><a href="#topic+phylosignal">phylosignal</a></code>, and <code><a href="#topic+raoD">raoD</a></code>.
</p>


<h3>Value</h3>

<p>A list containing the following elements, pruned and sorted to match one another:
</p>
<table>
<tr><td><code>phy</code></td>
<td>
<p> A phylogeny object of class phylo </p>
</td></tr>
<tr><td><code>comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p> A data object (vector, data.frame or matrix) </p>
</td></tr>
<tr><td><code>dist</code></td>
<td>
<p> A distance matrix - a dist or matrix object </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>See Also</h3>

<p><code><a href="#topic+prune.missing">prune.missing</a></code>, <code><a href="#topic+prune.sample">prune.sample</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
match.phylo.comm(phylocom$phylo, phylocom$sample)
match.phylo.data(phylocom$phylo, phylocom$traits[1:10,])
</code></pre>

<hr>
<h2 id='matrix2sample'> Convert community data matrix to Phylocom sample </h2><span id='topic+matrix2sample'></span>

<h3>Description</h3>

<p>Converts a community data matrix to a Phylocom database-format community sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrix2sample(z)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrix2sample_+3A_z">z</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Phylocom database-format community sample
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; and Cam Webb &lt;cwebb@oeb.harvard.edu&gt; </p>


<h3>References</h3>

<p> Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
matrix2sample(phylocom$sample)
</code></pre>

<hr>
<h2 id='mntd'> Mean nearest taxon distance </h2><span id='topic+mntd'></span><span id='topic+mnnd'></span>

<h3>Description</h3>

<p>Calculates MNTD (mean nearest taxon distance) for taxa in a community
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mntd(samp, dis, abundance.weighted=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mntd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="mntd_+3A_dis">dis</code></td>
<td>
<p> Interspecific distance matrix </p>
</td></tr>
<tr><td><code id="mntd_+3A_abundance.weighted">abundance.weighted</code></td>
<td>
<p> Should mean nearest taxon distances for each species be weighted by species abundance? (default = FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This metric has also been referred to as MNND (mean nearest neighbour distance), and the function was named <code>mnnd</code> in picante versions &lt; 0.7.
</p>


<h3>Value</h3>

<p>Vector of MNTD values for each community.
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Webb, C., D. Ackerly, M. McPeek, and M. Donoghue. 2002. Phylogenies and community ecology. Annual Review of Ecology and Systematics 33:475-505. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+ses.mntd">ses.mntd</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
mntd(phylocom$sample, cophenetic(phylocom$phylo), abundance.weighted=TRUE)
</code></pre>

<hr>
<h2 id='mpd'> Mean pairwise distance </h2><span id='topic+mpd'></span>

<h3>Description</h3>

<p>Calculates mean pairwise distance separating taxa in a community
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mpd(samp, dis, abundance.weighted=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mpd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="mpd_+3A_dis">dis</code></td>
<td>
<p> Interspecific distance matrix </p>
</td></tr>
<tr><td><code id="mpd_+3A_abundance.weighted">abundance.weighted</code></td>
<td>
<p> Should mean pairwise distances be weighted by species abundance? (default = FALSE)</p>
</td></tr>  
</table>


<h3>Value</h3>

<p>Vector of MPD values for each community
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Webb, C., D. Ackerly, M. McPeek, and M. Donoghue. 2002. Phylogenies and community ecology. Annual Review of Ecology and Systematics 33:475-505. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+ses.mpd">ses.mpd</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
mpd(phylocom$sample, cophenetic(phylocom$phylo), abundance.weighted=TRUE)
</code></pre>

<hr>
<h2 id='multiPhylosignal'> Calculates phylogenetic signal for data.frame of traits </h2><span id='topic+multiPhylosignal'></span>

<h3>Description</h3>

<p>Calculates phylogenetic signal for data.frame of traits. Traits may have missing values in which case the tree will be pruned prior to calculating phylogenetic signal for each trait.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiPhylosignal(x, phy, checkdata=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiPhylosignal_+3A_x">x</code></td>
<td>
<p> Data frame of trait data (traits in columns) with row names corresponding to tip.labels</p>
</td></tr>
<tr><td><code id="multiPhylosignal_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="multiPhylosignal_+3A_checkdata">checkdata</code></td>
<td>
<p> Check for match between trait and phylogeny taxa labels using <code><a href="#topic+match.phylo.data">match.phylo.data</a></code>? (default=TRUE)</p>
</td></tr>    
<tr><td><code id="multiPhylosignal_+3A_...">...</code></td>
<td>
<p> Additional arguments to <a href="#topic+phylosignal">phylosignal</a> </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with phylogenetic signal results for each trait
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>

<hr>
<h2 id='pblm'> Phylogenetic Bipartite Linear Model </h2><span id='topic+pblm'></span><span id='topic+pblmpredict'></span>

<h3>Description</h3>

<p>Fits a linear model to the association strengths of a bipartite data set with or without phylogenetic correlation among the interacting species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pblm(assocs, tree1=NULL, tree2=NULL, covars1=NULL, covars2=NULL, bootstrap=FALSE,
    nreps=10, maxit=10000, pstart=c(.5,.5))
pblmpredict(x, tree1.w.novel=NULL, tree2.w.novel=NULL, predict.originals=FALSE)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pblm_+3A_assocs">assocs</code></td>
<td>
<p> A matrix of association strengths among two sets of interacting species </p>
</td></tr>
<tr><td><code id="pblm_+3A_tree1">tree1</code></td>
<td>
<p> A phylo tree object or a phylogenetic covariance matrix for the rows of <code>assocs</code> </p>
</td></tr>
<tr><td><code id="pblm_+3A_tree2">tree2</code></td>
<td>
<p> A phylo tree object or a phylogenetic covariance matrix for the columns of <code>assocs</code></p>
</td></tr>
<tr><td><code id="pblm_+3A_covars1">covars1</code></td>
<td>
<p> A matrix of covariates (e.g., traits) for the row species of <code>assocs</code> </p>
</td></tr>
<tr><td><code id="pblm_+3A_covars2">covars2</code></td>
<td>
<p> A matrix of covariates (e.g., traits) for the column species of <code>assocs</code> </p>
</td></tr>
<tr><td><code id="pblm_+3A_bootstrap">bootstrap</code></td>
<td>
<p> logical, bootstrap confidence intervals of the parameter estimates </p>
</td></tr>
<tr><td><code id="pblm_+3A_nreps">nreps</code></td>
<td>
<p> Number of bootstrap replicated data sets to estimate parameter CIs </p>
</td></tr>
<tr><td><code id="pblm_+3A_maxit">maxit</code></td>
<td>
<p> as in <code><a href="stats.html#topic+optim">optim</a></code> </p>
</td></tr>
<tr><td><code id="pblm_+3A_pstart">pstart</code></td>
<td>
<p> starting values of the two phylogenetic signal strength parameters passed to <code><a href="stats.html#topic+optim">optim</a></code> </p>
</td></tr>
<tr><td><code id="pblm_+3A_x">x</code></td>
<td>
<p> object of class <code>pblm</code> </p>
</td></tr>
<tr><td><code id="pblm_+3A_tree1.w.novel">tree1.w.novel</code></td>
<td>
<p> A phylo tree object or a phylogenetic covariance matrix which corresponds to <code>tree1</code> of <code>x</code> with species to predict associations </p>
</td></tr>
<tr><td><code id="pblm_+3A_tree2.w.novel">tree2.w.novel</code></td>
<td>
<p> A phylo tree object or a phylogenetic covariance matrix which corresponds to <code>tree2</code> of <code>x</code> with species to predict associations </p>
</td></tr> 
<tr><td><code id="pblm_+3A_predict.originals">predict.originals</code></td>
<td>
<p> if <code>TRUE</code> then the associations of each original species in the two phylogenies is predicted </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>Fit a linear model with covariates using estimated generalized least squares to the association strengths between two sets of interacting species. 
Associations can be either binary or continuous. If phylogenies of the two sets of interacting species are supplied, 
two <em>phyogenetic signal strength</em> parameters (<em>d1</em> and <em>d2</em>), one for each species set, based on an Ornstein-Uhlenbeck model of 
evolution with stabilizing selection are estimated. Values of <em>d=1</em> indicate no stabilizing selection and correspond to the Brownian motion model of 
evolution; <em>0&lt;d&lt;1</em> represents stabilizing selection; <em>d=0</em> depicts the absence of phylogenetic correlation (i.e., a star phylogeny); and <em>d&gt;1</em> corresponds 
to disruptive selection where phylogenetic signal is amplified. Confidence intervals for these and the other parameters can be estimated with 
bootstrapping.
<br />
<br />
The function <code>pblmpredict</code> predicts the associations of novel species following the methods given in appendix B of Ives and Godfray (2006). 
</p>


<h3>Value</h3>

<table>
<tr><td><code>MSE</code></td>
<td>
<p> total, full (each <em>d</em> estimated), star (<em>d=0</em>), and base (<em>d=1</em>) mean squared errors </p>
</td></tr>
<tr><td><code>signal.strength</code></td>
<td>
<p> two estimates of phylogenetic signal strength </p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p> estimated intercept and covariate coefficients with approximate 95 percent CIs for the three model types (full, star, base) </p>
</td></tr>
<tr><td><code>CI.boot</code></td>
<td>
<p> 95 percent CIs for all parameters </p>
</td></tr>
<tr><td><code>variates</code></td>
<td>
<p> matrix of model variates (can be used for plotting) </p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p> matrix of residuals from the three models (full, star and base) </p>
</td></tr>
<tr><td><code>predicted</code></td>
<td>
<p> predicted associations </p>
</td></tr>
<tr><td><code>bootvalues</code></td>
<td>
<p> matrix of parameters estimated from the <code>nreps</code> bootstrap replicated data sets used to calculate CIs </p>
</td></tr>
<tr><td><code>phylocovs</code></td>
<td>
<p> phylogenetic covariance matricies scaled by the estimated <code>d1</code> and <code>d2</code> </p>
</td></tr>
<tr><td><code>cors.1</code></td>
<td>
<p> correlations among predicted and observed associations for species of <code>tree1</code>, <code>NA</code> if <code>predict.originals=FALSE</code> </p>
</td></tr>
<tr><td><code>cors.2</code></td>
<td>
<p> correlations among predicted and observed associations for species of <code>tree2</code>,  <code>NA</code> if <code>predict.originals=FALSE</code> </p>
</td></tr>
<tr><td><code>pred.novels1</code></td>
<td>
<p> predicted associations for the novel speices of <code>tree1</code> </p>
</td></tr>
<tr><td><code>pred.novels2</code></td>
<td>
<p> predicted associations for the novel speices of <code>tree2</code> </p>
</td></tr>
</table>


<h3>Note</h3>

<p>Covariates that apply to both species sets (e.g., sampling site) should be supplied in the covariate matrix of the set with the most species.
<br />
<br />
Bootstrapping CIs is slow due to the function <code><a href="stats.html#topic+optim">optim</a></code> used to estimate the model parameters. See appendix A in Ives and Godfray (2006) 
for a discussion about this boostrapping procedure
<br />
<br />
If <code>pblmpredict=TRUE</code> the function does not first remove each species in turn when predicting the associations of the original species as 
is done in Ives and Godfray (2006).</p>


<h3>Author(s)</h3>

<p> Matthew Helmus <a href="mailto:mrhelmus@gmail.com">mrhelmus@gmail.com</a> </p>


<h3>References</h3>

<p>Ives A.R. &amp; Godfray H.C. (2006) Phylogenetic analysis of trophic associations. The American Naturalist, 168, E1-E14 <br />
<br />
Blomberg S.P., Garland T.J. &amp; Ives A.R. (2003) Testing for phylogenetic signal in comparative data: Behavioral traits are more labile. Evolution, 57, 717-745
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#load example data from Ives &amp; Godfray (2006)
data(IvesGodfray)

#net attack rate of parasitoid on host eq.4 in Ives and Godfray
A&lt;-(-1*log(1-IvesGodfray$interactions[,-28]/t(IvesGodfray$interactions[28])))

# Make tips of the phylogenetic trees contemporaneous by extending tips
p&lt;-dim(IvesGodfray$host)[1]
q&lt;-dim(IvesGodfray$parasitoid)[1]
host.cov.scaled&lt;-IvesGodfray$host
para.cov.scaled&lt;-IvesGodfray$parasitoid
for (i in 1:p)
{
  host.cov.scaled[i,i]&lt;-max(host.cov.scaled)
}
for (i in 1:q)
{
  para.cov.scaled[i,i]&lt;-max(para.cov.scaled)
}

# scale covariance matrices (this reduces numerical problems caused by
# determinants going to infinity or zero)
  host.cov.scaled&lt;-host.cov.scaled/(det(as.matrix(host.cov.scaled))^(1/p))
  para.cov.scaled&lt;-para.cov.scaled/(det(as.matrix(para.cov.scaled))^(1/q))

pblm.A &lt;- pblm(sqrt(A),tree1=host.cov.scaled,tree2=para.cov.scaled)
pblm.A$signal.strength    #compare to Ives and Godfray (2006) Table 1 Line 1
pblm.A$MSE

</code></pre>

<hr>
<h2 id='pcd'>Phylogenetic Community Dissimilarity</h2><span id='topic+pcd'></span><span id='topic+PCD'></span>

<h3>Description</h3>

<p>Pairwise dissimilarity in phylogenetic community composition that is partitioned into a nonphylogenetic and a phylogenetic component.</p>


<h3>Usage</h3>

<pre><code class='language-R'>    pcd(comm, tree, PSVmncd=NULL, PSVpool=NULL, reps=10^4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcd_+3A_comm">comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="pcd_+3A_tree">tree</code></td>
<td>
<p> Object of class phylo or a phylogenetic covariance matrix </p>
</td></tr>
<tr><td><code id="pcd_+3A_psvmncd">PSVmncd</code></td>
<td>
<p> Vector of null mean conditional <em>phylogenetic species variability (<a href="#topic+psv">PSV</a>)</em> values </p>
</td></tr>
<tr><td><code id="pcd_+3A_psvpool">PSVpool</code></td>
<td>
<p> The standard, unconditional <em>PSV</em> calculated for the species pool </p>
</td></tr>
<tr><td><code id="pcd_+3A_reps">reps</code></td>
<td>
<p> The number of random draws from the species pool used to produce <code>PSVmncd</code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Phylogenetic community dissimilarity (PCD)</em> is the pairwise differences between communities derived by asking how much of the variance 
among species in the values of a hypothetical nonselected trait in one community can be predicted by the known trait values of species in another community. 
<em>PCD</em> is partitioned into a nonphylogenetic component that reflects shared species between communities (<em>PCDc</em>) 
and a phylogenetic component that reflects the evolutionary relationships among nonshared species (<em>PCDp</em>). In order to compare communities that vary 
in species richness, the metric is standardized under the assumption that the species in communities are selected at random from the species pool. The   
analyses here define the species pool as the list of all species in the set of communities in <code>comm</code>, but the species pool can be defined under 
any hypothesis of community assembly either by manipulating the code or inputting a user defined <code>PSVmncd</code> and <code>PSVpool</code>.
</p>


<h3>Value</h3>

<p>The function returns a list with items:
</p>
<table>
<tr><td><code>PCD</code></td>
<td>
<p> A square matrix of <em>PCD</em> values </p>
</td></tr>
<tr><td><code>PCDc</code></td>
<td>
<p> A square matrix of <em>PCDc</em> values </p>
</td></tr>
<tr><td><code>PCDp</code></td>
<td>
<p> A square matrix of <em>PCDp</em> values </p>
</td></tr>
<tr><td><code>PSVmncd</code></td>
<td>
<p> A vector of null mean conditional <em>PSV</em> values used to calculate <em>PCD</em> </p>
</td></tr>
<tr><td><code>PSVpool</code></td>
<td>
<p> The unconditional PSV of the species pool used to calculate <em>PCD</em> </p>
</td></tr>
</table>


<h3>Note</h3>

<p>The sampling procedure used to standardize <em>PCD</em> and produce <code>PSVmncd</code> and <code>PSVpool</code> can be slow.
</p>


<h3>Author(s)</h3>

<p> Anthony Ives &lt;arives@wisc.edu&gt; and Matthew Helmus &lt;mrhelmus@gmail.com&gt; </p>


<h3>References</h3>

<p>Ives A.R. &amp; Helmus M.R. (2010). Phylogenetic metrics of community similarity. The American Naturalist, 176, E128-E142.</p>


<h3>See Also</h3>

<p><code><a href="#topic+psv">psv</a></code>, <code><a href="#topic+phylosor">phylosor</a></code>, <code><a href="#topic+unifrac">unifrac</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(phylocom)
pcd(phylocom$sample, phylocom$phylo)

</code></pre>

<hr>
<h2 id='pd'> Calculate Faith's Phylogenetic Diversity </h2><span id='topic+pd'></span>

<h3>Description</h3>

<p>Calculate the sum of the total phylogenetic branch length for one or multiple samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pd(samp, tree, include.root=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="pd_+3A_tree">tree</code></td>
<td>
<p> A phylo tree object </p>
</td></tr>
<tr><td><code id="pd_+3A_include.root">include.root</code></td>
<td>
<p> Should the root node be included in all PD calculations (default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a dataframe of the PD and species richness (SR) values for all samples
</p>


<h3>Warning </h3>

<p>If the root is to be included in all calculations (<code>include.root=TRUE</code>), the PD of all samples will include the branch length connecting taxa in those samples and the root node of the supplied tree. The root of the supplied tree may not be spanned by any taxa in the sample. If you want the root of your tree to correspond to the most recent ancestor of the taxa actually present in your sample, you should prune the tree before running <code>pd</code>:
</p>
<p><code>prunedTree &lt;- prune.sample(sample,tree)</code>
</p>


<h3>Note</h3>

<p>The data sets need not be species-community data sets but may be any sample data set with an associated phylogeny. PD is not statistically independent of species richness, it positively correlates with species richness across samples. The function <code><a href="#topic+ses.pd">ses.pd</a></code> compares observed PD to the values expected under various randomizations and allows a way to standardize for unequal richness across samples.
</p>
<p>If the root is to be included in all calculations of PD (<code>include.root=TRUE</code>), the tree must be rooted. Single-species samples will be assigned a PD value equal to the distance from the root to the present.
</p>
<p>If the root is not included in all calculations by default (<code>include.root=FALSE</code>), the tree need not rooted, but in the case of single-species samples the PD will be equal to NA and a warning will be issued.
</p>


<h3>Author(s)</h3>

<p> Matthew Helmus <a href="mailto:mrhelmus@gmail.com">mrhelmus@gmail.com</a>, Jonathan Davies <a href="mailto:davies@nceas.ucsb.edu">davies@nceas.ucsb.edu</a>, Steven Kembel <a href="mailto:steve.kembel@gmail.com">steve.kembel@gmail.com</a></p>


<h3>References</h3>

<p> Faith D.P. (1992) Conservation evaluation and phylogenetic diversity. Biological Conservation, 61, 1-10. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+psr">psr</a></code>, <code><a href="#topic+ses.pd">ses.pd</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
pd(phylocom$sample, phylocom$phylo)</code></pre>

<hr>
<h2 id='phyEstimate'> Phylogenetic estimation of traits for unobserved taxa </h2><span id='topic+phyEstimate'></span><span id='topic+phyEstimateDisc'></span>

<h3>Description</h3>

<p>Uses phylogenetic ancestral state reconstruction to estimate trait values for unobserved taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyEstimate(phy, trait, method="pic", ...)
phyEstimateDisc(phy, trait, best.state=TRUE, cutoff=0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyEstimate_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="phyEstimate_+3A_trait">trait</code></td>
<td>
<p> vector or data.frame containing trait values </p>
</td></tr>
<tr><td><code id="phyEstimate_+3A_method">method</code></td>
<td>
<p> ancestral state estimation method used by <code>ace</code> (default=&quot;pic&quot;)</p>
</td></tr>
<tr><td><code id="phyEstimate_+3A_best.state">best.state</code></td>
<td>
<p> estimate best-supported trait state for discrete variables? (default=TRUE)</p>
</td></tr>  
<tr><td><code id="phyEstimate_+3A_cutoff">cutoff</code></td>
<td>
<p> support cutoff required to declare a best.state </p>
</td></tr>  
<tr><td><code id="phyEstimate_+3A_...">...</code></td>
<td>
<p> Additional arguments passed to <code>ace</code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions use phylogenetic ancestral state estimation to infer trait values for novel taxa on a phylogenetic tree, for continuous (<code>phyEstimate</code>) and discrete (<code>phyEstimateDisc</code>) traits.
</p>
<p>The required input is a phylogenetic tree object plus a vector or data.frame containing estimated trait values for a subset of the taxa in the phylogenetic tree. Trait values for taxa that are present in the tree but not the trait data will be estimated using ancestral state estimation (Garland and Ives 2000). Briefly, for each taxon present in the tree but not the trait data, the phylogeny is rerooted at the most recent common ancestor of the novel taxon and the rest of the phylogeny, and the trait value of the novel taxon is estimated from the reconstructed trait value at the root of the rerooted phylogeny.
</p>
<p>For <code>phyEstimateDisc</code>, the state with the highest support will be reported if argument <code>best.state=TRUE</code>. If the best-supported state's support is less than the specified <code>cutoff</code>, no best state is reported and a <code>NA</code> value will be returned.
</p>


<h3>Value</h3>

<p>phyEstimate produces a data frame with columns:
</p>
<table>
<tr><td><code>est</code></td>
<td>
<p> Estimated trait value </p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p> Standard error of estimated trait value </p>
</td></tr>
</table>
<p>phyEstimateDisc produces a data frame with columns:
</p>
<table>
<tr><td><code>states 1..N</code></td>
<td>
<p> A column with statistical support is produced for each discrete trait state </p>
</td></tr>
<tr><td><code>estimated.state</code></td>
<td>
<p> If best.state=TRUE, a column with the state with the highest support </p>
</td></tr>
<tr><td><code>estimated.state.support</code></td>
<td>
<p> Statistical support for the state with the highest support </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

 
<p>T. Garland Jr., and A.R. Ives. 2000. Using the past to predict the present: confidence intervals for regression equations in phylogenetic comparative methods. American Naturalist 155:346364.
</p>
<p>S.W. Kembel, M. Wu, J.A. Eisen, and J.L. Green. 2012. Incorporating 16S gene copy number information improves estimates of microbial diversity and abundance. PLoS Computational Biology 8(10):e1002743.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#generate random phylogeny
randtree &lt;- rcoal(50)
#simulate trait evolution for a subset of taxa on phylogeny
randtraits &lt;- sample(rTraitCont(randtree, sigma=10, root.value=100), 40)
#estimate trait values for "missing" taxa using PIC method
phyEstimate(randtree, randtraits, method="pic")
</code></pre>

<hr>
<h2 id='phylocom'> Phylocom default data </h2><span id='topic+phylocom'></span>

<h3>Description</h3>

<p>Phylogeny, community and trait data from the Phylocom 4.0 distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phylocom)</code></pre>


<h3>Format</h3>

<p>A list with three elements:
</p>

<ul>
<li><p> phylo  Phylogenetic tree (an object of class phylo) 
</p>
</li>
<li><p> sample  Community data (a data.frame with samples in rows and species in columns 
</p>
</li>
<li><p> traits  Trait data (a data.frame with species in rows and traits in columns 
</p>
</li></ul>



<h3>Source</h3>

<p>Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1.
<a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>.
</p>

<hr>
<h2 id='phylosignal'> Measure phylogenetic signal </h2><span id='topic+phylosignal'></span>

<h3>Description</h3>

<p>Calculates K statistic of phylogenetic signal as well as P-value based on variance of phylogenetically independent contrasts relative to tip shuffling randomization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylosignal(x, phy, reps = 999, checkdata=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylosignal_+3A_x">x</code></td>
<td>
<p> Trait vector (same order as phy\$tip.label) </p>
</td></tr>
<tr><td><code id="phylosignal_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="phylosignal_+3A_reps">reps</code></td>
<td>
<p> Number of randomizations </p>
</td></tr>
<tr><td><code id="phylosignal_+3A_checkdata">checkdata</code></td>
<td>
<p> Check for match between trait and phylogeny taxa labels using <code><a href="#topic+match.phylo.data">match.phylo.data</a></code>? (default=TRUE)</p>
</td></tr>  
<tr><td><code id="phylosignal_+3A_...">...</code></td>
<td>
<p> Additional arguments passed to pic </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with columns:
</p>
<table>
<tr><td><code>K</code></td>
<td>
<p>K statistic</p>
</td></tr>
<tr><td><code>PIC.variance</code></td>
<td>
<p>Mean observed PIC variance</p>
</td></tr>
<tr><td><code>PIC.variance.P</code></td>
<td>
<p>P-value of observed vs. random variance of PICs</p>
</td></tr>
<tr><td><code>PIC.variance.z</code></td>
<td>
<p>Z-score of observed vs. random variance of PICs</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

 
<p>Blomberg, S. P., and T. Garland, Jr. 2002. Tempo and mode in evolution: phylogenetic inertia, adaptation and comparative methods. Journal of Evolutionary Biology 15:899-910.
</p>
<p>Blomberg, S. P., T. Garland, Jr., and A. R. Ives. 2003. Testing for phylogenetic signal in comparative data: behavioral traits are more labile. Evolution 57:717-745. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+Kcalc">Kcalc</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
randtree &lt;- rcoal(20)
randtraits &lt;- rTraitCont(randtree)
phylosignal(randtraits[randtree$tip.label],randtree)

</code></pre>

<hr>
<h2 id='phylosor'>Phylogenetic index of beta-diversity PhyloSor</h2><span id='topic+phylosor'></span>

<h3>Description</h3>

<p>Fraction of branch-length shared between two communities</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylosor(samp, tree)    
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylosor_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="phylosor_+3A_tree">tree</code></td>
<td>
<p> Object of class phylo - a rooted phylogeny</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A distance object of the PhyloSor index of similarity between communities, the fraction of PD (branch-length) shared between two samples</p>


<h3>Warning </h3>

<p>The phylosor of all samples will include the branch length connecting taxa in those samples and the root of the supplied tree. The root of the supplied tree may not be spanned by any taxa in the sample. If you want the root of your tree to correspond to the most recent ancestor of the taxa actually present in your sample, you should prune the tree before running <code>phylosor</code>:
</p>
<p><code>prunedTree &lt;- prune.sample(sample,tree)</code>
</p>


<h3>Note</h3>

<p>The root of the supplied tree is included in calculations of PhyloSor. The supplied tree must be rooted. Single-species samples will be assigned a PD value equal to the distance from the root to the present.
</p>


<h3>Author(s)</h3>

<p> Helene Morlon &lt;morlon.helene@gmail.com&gt; and Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Bryant, J.B., Lamanna, C., Morlon, H., Kerkhoff, A.J., Enquist, B.J., Green, J.L. 2008. Microbes on mountainsides: Contrasting elevational patterns of bacterial and plant diversity. Proceedings of the National Academy of Sciences 105 Supplement 1: 11505-11511</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylosor.rnd">phylosor.rnd</a></code>, <code><a href="#topic+pd">pd</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
phylosor(phylocom$sample, phylocom$phylo)</code></pre>

<hr>
<h2 id='phylosor.rnd'> Null PhyloSor values of phylogenetic beta-diversity </h2><span id='topic+phylosor.rnd'></span>

<h3>Description</h3>

<p>PhyloSor values obtained by randomization for different choices of null models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylosor.rnd(samp,tree, cstSor=TRUE, null.model=c("taxa.labels",
    "frequency","richness","independentswap","trialswap"),
    runs=999, iterations=1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylosor.rnd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="phylosor.rnd_+3A_tree">tree</code></td>
<td>
<p> Object of class phylo - a rooted phylogeny</p>
</td></tr>
<tr><td><code id="phylosor.rnd_+3A_cstsor">cstSor</code></td>
<td>
<p> TRUE if the Sorensen similarity should be kept constant across communities. FALSE otherwise </p>
</td></tr>
<tr><td><code id="phylosor.rnd_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section) </p>
</td></tr>
<tr><td><code id="phylosor.rnd_+3A_runs">runs</code></td>
<td>
<p> Number of randomizations </p>
</td></tr>
<tr><td><code id="phylosor.rnd_+3A_iterations">iterations</code></td>
<td>
<p> Number of iterations to use for each randomization (for independent swap and trial null models) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>taxa.labels</dt><dd><p> Shuffle community data matrix labels. Maintains species richness in each community and species shared between communities. Should be used with cstSor=TRUE</p>
</dd>
<dt>frequency</dt><dd><p> Randomize community data matrix abundances within species (maintains species occurence frequency). Does not maintain species richness in communities nor species shared between communities. Can only be used with cstSor=FALSE</p>
</dd>
<dt>richness</dt><dd><p> With cstSor=TRUE: For each pair of community, maintains species richness in each community and species shared between communities. Sample in the species pool with equal probability;
With cstSor=FALSE: Maintains species richness in each community, does not maintain species shared between communities. Sample in the species pool with equal probability</p>
</dd>
<dt>independentswap</dt><dd><p> Randomize community data matrix with the independent swap algorithm (Gotelli 2000) maintaining species occurrence frequency and sample species richness. Can only be used with cstSor=FALSE</p>
</dd>
<dt>trialswap</dt><dd><p> Randomize community data matrix with the trial-swap algorithm (Miklos &amp; Podani 2004) maintaining species occurrence frequency and sample species richness. Can only be used with cstSor=FALSE</p>
</dd>
</dl>



<h3>Value</h3>

<p>A list of length the number of runs. Each element of the list is a distance matrix containing the PhyloSor values of phylogenetic beta-diversity obtained by randomization
</p>


<h3>Author(s)</h3>

<p> Helene Morlon &lt;morlon.helene@gmail.com&gt; and Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Bryant, J.B., Lamanna, C., Morlon, H., Kerkhoff, A.J., Enquist, B.J., Green, J.L. 2008. Microbes on mountainsides: Contrasting elevational patterns of bacterial and plant diversity. Proceedings of the National Academy of Sciences 105 Supplement 1: 11505-11511</p>


<h3>See Also</h3>

 <p><code><a href="#topic+phylosor">phylosor</a></code>, <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
phylosor.rnd(phylocom$sample,phylocom$phylo,cstSor=TRUE,null.model="richness",runs=5)
</code></pre>

<hr>
<h2 id='phylostruct'> Permutations to Test for Phylogenetic Signal in Community Composition </h2><span id='topic+phylostruct'></span>

<h3>Description</h3>

<p> Randomize sample/community data matrices to create null distributions of given metrics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylostruct(samp, tree, env=NULL, metric=c("psv","psr","pse","psc","sppregs"),
    null.model=c("frequency", "richness","independentswap","trialswap"),
    runs=100, it=1000, alpha=0.05, fam="binomial")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylostruct_+3A_samp">samp</code></td>
<td>
<p> community data matrix, species as columns, communities as rows </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_tree">tree</code></td>
<td>
<p> phylo tree object or a phylogenetic covariance matrix </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_env">env</code></td>
<td>
<p> environmental data matrix </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_metric">metric</code></td>
<td>
<p> if <code>metric="psv"</code>, <code>"psr"</code>, <code>"pse"</code>, or <code>"psc"</code> compares the observed mean of the respective metric to a null distribution at a given alpha; if <code>metric="sppregs"</code> compares the three correlations produced by <code><a href="#topic+sppregs">sppregs</a></code> to null distributions </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_null.model">null.model</code></td>
<td>
<p> permutation procedure used to create the null distribution, see <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code></p>
</td></tr>
<tr><td><code id="phylostruct_+3A_runs">runs</code></td>
<td>
<p> the number of permutations to create the distribution, a rule of thumb is (number of communities)/alpha  </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_it">it</code></td>
<td>
<p> the number of swaps for the independent and trial-swap null models, see <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code> </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_alpha">alpha</code></td>
<td>
<p> probability value to compare the observed mean/correlations to a null distribution </p>
</td></tr>
<tr><td><code id="phylostruct_+3A_fam">fam</code></td>
<td>
<p> as in <code><a href="#topic+sppregs">sppregs</a></code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function creates null distributions for the <code><a href="#topic+psd">psd</a></code> set of metrics and for the correlations of <code><a href="#topic+sppregs">sppregs</a></code> from observed community data sets.</p>


<h3>Value</h3>

<table>
<tr><td><code>metric</code></td>
<td>
<p> metric used </p>
</td></tr>
<tr><td><code>null.model</code></td>
<td>
<p> permutation used </p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p> number of permutations </p>
</td></tr>
<tr><td><code>it</code></td>
<td>
<p> number of swaps if applicable </p>
</td></tr>
<tr><td><code>obs</code></td>
<td>
<p> observed mean value of a particular metric or the three observed correlations from <code><a href="#topic+sppregs">sppregs</a></code> </p>
</td></tr>
<tr><td><code>mean.null</code></td>
<td>
<p> mean(s) of the null distribution(s) </p>
</td></tr>
<tr><td><code>quantiles.null</code></td>
<td>
<p>  quantiles of the null distribution(s) to compare to <code>obs</code>; determined by <code>alpha</code></p>
</td></tr>
<tr><td><code>phylo.structure</code></td>
<td>
<p> if <code>obs</code> less than (alpha/2), <code>phylo.structure="underdispersed"</code>; if <code>obs</code> greater than (1-alpha/2), <code>phylo.structure="overdispersed"</code>; otherwise <code>phylo.structure="random"</code> and NULL if <code>metric="sppregs"</code></p>
</td></tr>
<tr><td><code>nulls</code></td>
<td>
<p> null values of the distribution(s)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Matthew Helmus <a href="mailto:mrhelmus@gmail.com">mrhelmus@gmail.com</a> </p>


<h3>References</h3>

<p> Helmus M.R., Bland T.J., Williams C.K. &amp; Ives A.R. (2007a) Phylogenetic measures of biodiversity. American Naturalist, 169, E68-E83
<br />
<br />
Helmus M.R., Savage K., Diebel M.W., Maxted J.T. &amp; Ives A.R. (2007b) Separating the determinants of phylogenetic community structure. Ecology Letters, 10, 917-925
<br />
<br />
Gotelli N.J. (2000) Null model analysis of species co-occurrence patterns. Ecology, 81, 2606-2621</p>


<h3>See Also</h3>

 <p><code><a href="#topic+psd">psd</a></code> ,<code><a href="#topic+sppregs">sppregs</a></code>, <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code> </p>

<hr>
<h2 id='prune.sample'> Prune tree to match community data or trait data </h2><span id='topic+prune.sample'></span><span id='topic+prune.missing'></span>

<h3>Description</h3>

<p>Prune a phylogenetic tree to include only species present in a community data set or with non-missing trait data</p>


<h3>Usage</h3>

<pre><code class='language-R'>prune.sample(samp, phylo)
prune.missing(x, phylo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prune.sample_+3A_phylo">phylo</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="prune.sample_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="prune.sample_+3A_x">x</code></td>
<td>
<p> Vector of trait data </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a pruned phylo object
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>

<hr>
<h2 id='psd'> Phylogenetic Species Diversity Metrics </h2><span id='topic+psd'></span><span id='topic+psv.spp'></span><span id='topic+psv'></span><span id='topic+psr'></span><span id='topic+pse'></span><span id='topic+psc'></span>

<h3>Description</h3>

<p>Calculate the bounded phylogenetic biodiversity metrics: phylogenetic species variability, richness, evenness and clustering for one or multiple samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>psv(samp,tree,compute.var=TRUE,scale.vcv=TRUE)
psr(samp,tree,compute.var=TRUE,scale.vcv=TRUE)
pse(samp,tree,scale.vcv=TRUE)
psc(samp,tree,scale.vcv=TRUE)
psd(samp,tree,compute.var=TRUE,scale.vcv=TRUE)
psv.spp(samp,tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="psd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="psd_+3A_tree">tree</code></td>
<td>
<p> A phylo tree object or a phylogenetic covariance matrix </p>
</td></tr>
<tr><td><code id="psd_+3A_compute.var">compute.var</code></td>
<td>
<p> Computes the expected variances for PSV and PSR for each community </p>
</td></tr>
<tr><td><code id="psd_+3A_scale.vcv">scale.vcv</code></td>
<td>
<p> Scale the phylogenetic covariance matrix to bound the metric between 0 and 1 </p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Phylogenetic species variability (PSV)</em> quantifies how phylogenetic relatedness decreases the variance of a hypothetical unselected/neutral trait
shared by all species in a community. The expected value of PSV is statistically independent of species richness, is one when all species in a sample
are unrelated (i.e., a star phylogeny) and approaches zero as species become more related. PSV is directly related to mean phylogenetic distance, except
except calculated on a scaled phylogenetic covariance matrix. The expected variance around PSV for any sample of a particular species richness can be approximated. 
To address how individual species contribute to the mean PSV of a data set, the function <code>psv.spp</code> gives signed proportions of the total deviation 
from the mean PSV that occurs when all species are removed from the data set one at a time. The absolute values of these &ldquo;species effects&rdquo; 
tend to positively correlate with species prevalence. 
<br />
<br />
<em>Phylogenetic species richness (PSR)</em> is the number of species in a sample multiplied by PSV. It can be considered the species richness of a sample
after discounting by species relatedness. The value is maximum at the species richness of the sample, and decreases towards zero as relatedness 
increases. The expected variance around PSR for any sample of a particular species richness can be approximated.
<br />
<br />
<em>Phylogenetic species evenness (PSE)</em> is the metric PSV modified to incorporate relative species abundances. The maximum attainable value of PSE (i.e., 1)
occurs only if species abundances are equal and species phylogeny is a star. PSE essentially grafts each individual of a species onto the tip of the 
phylogeny of its species with branch lengths of zero.
<br />
<br />
<em>Phylogenetic species clustering (PSC)</em> is a metric of the branch tip clustering of species across a sample's phylogeny. As PSC increases to 1, species
are less related to one another the tips of the phylogeny. PSC is directly related to mean nearest neighbor distance.
</p>


<h3>Value</h3>

<p>Returns a dataframe of the respective phylogenetic species diversity metric values  
</p>


<h3>Note</h3>

<p>These metrics are bounded either between zero and one (PSV, PSE, PSC) or zero and species richness (PSR); but the metrics asymptotically 
approach zero as relatedness increases. Zero can be assigned to communities with less than two species, but conclusions drawn from assigning 
communities zero values need be carefully explored for any data set. The data sets need not be species-community data sets but may be any sample 
data set with an associated phylogeny. 
</p>


<h3>Author(s)</h3>

<p> Matthew Helmus <a href="mailto:mrhelmus@gmail.com">mrhelmus@gmail.com</a> </p>


<h3>References</h3>

<p> Helmus M.R., Bland T.J., Williams C.K. &amp; Ives A.R. (2007) Phylogenetic measures of biodiversity. American Naturalist, 169, E68-E83 </p>


<h3>See Also</h3>

 <p><code><a href="#topic+mpd">mpd</a></code> ,<code><a href="#topic+mnnd">mnnd</a></code>, <code><a href="#topic+specaccum.psr">specaccum.psr</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
psd(phylocom$sample, phylocom$phylo)
</code></pre>

<hr>
<h2 id='randomizeMatrix'> Null models for community data matrix randomization</h2><span id='topic+randomizeMatrix'></span>

<h3>Description</h3>

<p>Various null models for randomizing community data matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>randomizeMatrix(samp, null.model = c("frequency", "richness",
    "independentswap", "trialswap"), iterations = 1000)    
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="randomizeMatrix_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="randomizeMatrix_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section for description) </p>
</td></tr>
<tr><td><code id="randomizeMatrix_+3A_iterations">iterations</code></td>
<td>
<p> Number of independent or trial-swaps to perform </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>frequency</dt><dd><p> Randomize community data matrix abundances within species (maintains species occurence frequency)</p>
</dd>
<dt>richness</dt><dd><p> Randomize community data matrix abundances within samples (maintains sample species richness)</p>
</dd>
<dt>independentswap</dt><dd><p> Randomize community data matrix with the independent swap algorithm (Gotelli 2000) maintaining species occurrence frequency and sample species richness </p>
</dd>
<dt>trialswap</dt><dd><p> Randomize community data matrix with the trial-swap algorithm (Miklos &amp; Podani 2004) maintaining species occurrence frequency and sample species richness </p>
</dd>
</dl>



<h3>Value</h3>

<p>Randomized community data matrix
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Gotelli, N.J. 2000. Null model analysis of species co-occurrence patterns. Ecology 81: 2606-2621
</p>
<p>Miklos I. &amp; Podani J. 2004. Randomization of presence-absence matrices: Comments and new algorithms. Ecology 85: 86-92.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
randomizeMatrix(phylocom$sample, null.model="richness")</code></pre>

<hr>
<h2 id='raoD'> Rao's quadratic entropy </h2><span id='topic+raoD'></span>

<h3>Description</h3>

<p>Calculates Rao's quadratic entropy, a measure of within- and among-community diversity taking species dissimilarities into account
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  raoD(comm, phy=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="raoD_+3A_comm">comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="raoD_+3A_phy">phy</code></td>
<td>
<p> Object of class phylo - an ultrametric phylogenetic tree (optional) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Rao's quadratic entropy (Rao 1982) is a measure of diversity in ecological communities that can optionally take species differences (e.g. phylogenetic dissimilarity) into account. This method is conceptually similar to analyses of genetic diversity among populations (Nei 1973), but instead of diversity of alleles among populations, it measures diversity of species among communities. 
</p>
<p>If no phylogeny is supplied, Dkk is equivalent to Simpson's diversity (probability that two individuals drawn from a community are from different taxa), Dkl is a beta-diversity equivalent of Simpson's diversity (probability that individuals drawn from each of two communities belong to different taxa), and H is Dkl standardized to account for within-community diversity.
</p>
<p>If an ultrametric phylogeny is supplied, Dkk is equivalent to the mean pairwise phylogenetic distance (distance to MRCA) between two individuals drawn from a community, Dkl is the mean pairwise phylogenetic distance between individuals drawn from each of two communities, and H is Dkl standardized to account for within-community diversity.
</p>
<p><em>D[kl] = sum(t[ij] * x[ki] * x[lj])</em>
</p>
<p>where <em>x[ki]</em> is the relative abundance of taxon <em>i</em> in community <em>k</em> and <em>t[ij]</em> is a matrix of weights for all pairs of taxa <em>i,j</em>. Without a phylogeny, when <em>i=j</em>, <em>t[ij]=0</em>, otherwise <em>t[ij]=1</em>. With a phylogeny, <em>t[ij]</em> is the phylogenetic distance to MRCA for taxa <em>i,j</em>.
</p>
<p><em>H[kl] = D[kl] - (D[kk] + D[ll])/2</em>
</p>
<p>Alpha, beta and total measure the average diversity within, among, and across all communities based on Dkk and H values taking variation in number of individuals per community into account. A Fst-like measure is calculated by dividing beta by the total diversity across all samples.
</p>


<h3>Value</h3>

<p>A list of results
</p>
<table>
<tr><td><code>Dkk</code></td>
<td>
<p> Within-community diversity </p>
</td></tr>
<tr><td><code>Dkl</code></td>
<td>
<p> Among-community diversity </p>
</td></tr>
<tr><td><code>H</code></td>
<td>
<p> Among-community diversities excluding within-community diversity </p>
</td></tr>
<tr><td><code>total</code></td>
<td>
<p> Total diversity across all samples </p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p> Alpha diversity - average within-community diversity </p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p> Beta diversity - average among-community diversity </p>
</td></tr>
<tr><td><code>Fst</code></td>
<td>
<p> Beta diversity / total diversity </p>
</td></tr>
</table>


<h3> Warning </h3>

<p>Alpha, beta, and total diversity components and Fst should not be interpreted as a measure of relative differentiation among versus within communities. See Jost (2007) for a detailed description of this problem. Hardy and Jost (2008) suggest Fst can be interpreted as 'local species identity excess' or 'local phylogenetic similarity excess' rather than as a measure of among-community differentiation.
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Hardy, O.J., and Jost. L. 2008. Interpreting and estimating measures of community phylogenetic structuring. J. Ecol. 96:849-852.
</p>
<p>Jost, L. 2007. Partitioning diversity into independent alpha and beta components. Ecology 88: 24272439.
</p>
<p>Nei, M. 1973. Analysis of gene diversity in sub-divided populations. Proceedings of the National Academy of Sciences of the USA 70:3321-3323.
</p>
<p>Rao, C.R. 1982. Diversity and dissimilarity coefficients: a unified approach. Theoretical Population Biology 21:2443.
</p>
<p>Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+mpd">mpd</a></code>, <code><a href="#topic+comdist">comdist</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
raoD(phylocom$sample)
raoD(phylocom$sample, phylocom$phylo)
</code></pre>

<hr>
<h2 id='readsample'> Read Phylocom sample </h2><span id='topic+readsample'></span>

<h3>Description</h3>

<p>Reads a Phylocom sample file and converts to a community data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readsample(filename = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readsample_+3A_filename">filename</code></td>
<td>
<p> Phylocom sample file path </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Community data matrix
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;skembel&gt; and Cam Webb &lt;cwebb@oeb.harvard.edu&gt; </p>


<h3>References</h3>

<p>   Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>. </p>

<hr>
<h2 id='sample2matrix'> Convert Phylocom sample to community data matrix </h2><span id='topic+sample2matrix'></span>

<h3>Description</h3>

<p>Convert a Phylocom database-format sample to community data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample2matrix(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample2matrix_+3A_x">x</code></td>
<td>
<p> Phylocom sample formatted data frame, a data frame with three columns:
</p>

<ul>
<li><p>Column 1    Community name
</p>
</li>
<li><p>Column 2    Species abundance
</p>
</li>
<li><p>Column 3    Species name
</p>
</li></ul>

</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; and Cam Webb &lt;cwebb@oeb.harvard.edu&gt; </p>


<h3>References</h3>

<p>   Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>.</p>

<hr>
<h2 id='ses.mntd'> Standardized effect size of MNTD </h2><span id='topic+ses.mntd'></span><span id='topic+ses.mnnd'></span>

<h3>Description</h3>

<p>Standardized effect size of mean nearest taxon distances in communities. When used with a phylogenetic distance matrix, equivalent to -1 times the Nearest Taxon Index (NTI).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ses.mntd(samp, dis, null.model = c("taxa.labels", "richness", "frequency",
    "sample.pool", "phylogeny.pool", "independentswap", "trialswap"),
    abundance.weighted=FALSE, runs = 999, iterations = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ses.mntd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="ses.mntd_+3A_dis">dis</code></td>
<td>
<p> Distance matrix (generally a phylogenetic distance matrix)</p>
</td></tr>
<tr><td><code id="ses.mntd_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section for description) </p>
</td></tr>
<tr><td><code id="ses.mntd_+3A_abundance.weighted">abundance.weighted</code></td>
<td>
<p> Should mean nearest taxon distances for each species be weighted by species abundance? (default = FALSE) </p>
</td></tr>
<tr><td><code id="ses.mntd_+3A_runs">runs</code></td>
<td>
<p> Number of randomizations </p>
</td></tr>
<tr><td><code id="ses.mntd_+3A_iterations">iterations</code></td>
<td>
<p> Number of iterations to use for each randomization (for independent swap and trial null models) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The metric used by this function has also been referred to as MNND (mean nearest neighbour distance), and the function was named <code>ses.mnnd</code> in picante versions &lt; 0.7.
</p>
<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>taxa.labels</dt><dd><p> Shuffle distance matrix labels (across all taxa included in distance matrix)</p>
</dd>
<dt>richness</dt><dd><p> Randomize community data matrix abundances within samples (maintains sample species richness)</p>
</dd>    
<dt>frequency</dt><dd><p> Randomize community data matrix abundances within species (maintains species occurence frequency)</p>
</dd>
<dt>sample.pool</dt><dd><p> Randomize community data matrix by drawing species from pool of species occurring in at least one community (sample pool) with equal probability</p>
</dd>  
<dt>phylogeny.pool</dt><dd><p> Randomize community data matrix by drawing species from pool of species occurring in the distance matrix (phylogeny pool) with equal probability</p>
</dd>  
<dt>independentswap</dt><dd><p> Randomize community data matrix with the independent swap algorithm (Gotelli 2000) maintaining species occurrence frequency and sample species richness </p>
</dd>
<dt>trialswap</dt><dd><p> Randomize community data matrix with the trial-swap algorithm (Miklos &amp; Podani 2004) maintaining species occurrence frequency and sample species richness </p>
</dd>
</dl>



<h3>Value</h3>

<p>A data frame of results for each community
</p>
<table>
<tr><td><code>ntaxa</code></td>
<td>
<p>Number of taxa in community</p>
</td></tr>
<tr><td><code>mntd.obs</code></td>
<td>
<p>Observed MNTD in community</p>
</td></tr>
<tr><td><code>mntd.rand.mean</code></td>
<td>
<p>Mean MNTD in null communities</p>
</td></tr>
<tr><td><code>mntd.rand.sd</code></td>
<td>
<p>Standard deviation of MNTD in null communities</p>
</td></tr>
<tr><td><code>mntd.obs.rank</code></td>
<td>
<p>Rank of observed MNTD vs. null communities</p>
</td></tr>
<tr><td><code>mntd.obs.z</code></td>
<td>
<p>Standardized effect size of MNTD vs. null communities (= (mntd.obs - mntd.rand.mean) / mntd.rand.sd, equivalent to -NTI)</p>
</td></tr>
<tr><td><code>mntd.obs.p</code></td>
<td>
<p>P-value (quantile) of observed MNTD vs. null communities (= mntd.obs.rank / runs + 1)</p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p>Number of randomizations</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+mntd">mntd</a></code>, <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
ses.mntd(phylocom$sample, cophenetic(phylocom$phylo),null.model="taxa.labels")</code></pre>

<hr>
<h2 id='ses.mpd'> Standardized effect size of MPD </h2><span id='topic+ses.mpd'></span>

<h3>Description</h3>

<p>Standardized effect size of mean pairwise distances in communities. When used with a phylogenetic distance matrix, equivalent to -1 times the Nearest Relative Index (NRI).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ses.mpd(samp, dis, null.model = c("taxa.labels", "richness", "frequency", "sample.pool",
            "phylogeny.pool", "independentswap", "trialswap"),
            abundance.weighted = FALSE, runs = 999, iterations = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ses.mpd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="ses.mpd_+3A_dis">dis</code></td>
<td>
<p> Distance matrix (generally a phylogenetic distance matrix)</p>
</td></tr>
<tr><td><code id="ses.mpd_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section for description) </p>
</td></tr>
<tr><td><code id="ses.mpd_+3A_abundance.weighted">abundance.weighted</code></td>
<td>
<p> Should mean nearest taxon distances for each species be weighted by species abundance? (default = FALSE) </p>
</td></tr>  
<tr><td><code id="ses.mpd_+3A_runs">runs</code></td>
<td>
<p> Number of randomizations </p>
</td></tr>
<tr><td><code id="ses.mpd_+3A_iterations">iterations</code></td>
<td>
<p> Number of iterations to use for each randomization (for independent swap and trial null models) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>taxa.labels</dt><dd><p> Shuffle distance matrix labels (across all taxa included in distance matrix)</p>
</dd>
<dt>richness</dt><dd><p> Randomize community data matrix abundances within samples (maintains sample species richness)</p>
</dd>
<dt>frequency</dt><dd><p> Randomize community data matrix abundances within species (maintains species occurence frequency)</p>
</dd>
<dt>sample.pool</dt><dd><p> Randomize community data matrix by drawing species from pool of species occurring in at least one community (sample pool) with equal probability</p>
</dd>  
<dt>phylogeny.pool</dt><dd><p> Randomize community data matrix by drawing species from pool of species occurring in the distance matrix (phylogeny pool) with equal probability</p>
</dd>  
<dt>independentswap</dt><dd><p> Randomize community data matrix with the independent swap algorithm (Gotelli 2000) maintaining species occurrence frequency and sample species richness </p>
</dd>
<dt>trialswap</dt><dd><p> Randomize community data matrix with the trial-swap algorithm (Miklos &amp; Podani 2004) maintaining species occurrence frequency and sample species richness </p>
</dd>
</dl>



<h3>Value</h3>

<p>A data frame of results for each community
</p>
<table>
<tr><td><code>ntaxa</code></td>
<td>
<p>Number of taxa in community</p>
</td></tr>
<tr><td><code>mpd.obs</code></td>
<td>
<p>Observed mpd in community</p>
</td></tr>
<tr><td><code>mpd.rand.mean</code></td>
<td>
<p>Mean mpd in null communities</p>
</td></tr>
<tr><td><code>mpd.rand.sd</code></td>
<td>
<p>Standard deviation of mpd in null communities</p>
</td></tr>
<tr><td><code>mpd.obs.rank</code></td>
<td>
<p>Rank of observed mpd vs. null communities</p>
</td></tr>
<tr><td><code>mpd.obs.z</code></td>
<td>
<p>Standardized effect size of mpd vs. null communities (= (mpd.obs - mpd.rand.mean) / mpd.rand.sd, equivalent to -NRI)</p>
</td></tr>
<tr><td><code>mpd.obs.p</code></td>
<td>
<p>P-value (quantile) of observed mpd vs. null communities (= mpd.obs.rank / runs + 1)</p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p>Number of randomizations</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>. </p>


<h3>See Also</h3>

 <p><code><a href="#topic+mpd">mpd</a></code>, <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
ses.mpd(phylocom$sample, cophenetic(phylocom$phylo),null.model="taxa.labels")</code></pre>

<hr>
<h2 id='ses.pd'> Standardized effect size of PD </h2><span id='topic+ses.pd'></span>

<h3>Description</h3>

<p>Standardized effect size of phylogenetic diversity (Faith's PD) in communities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ses.pd(samp, tree, null.model = c("taxa.labels", "richness", "frequency",
    "sample.pool", "phylogeny.pool", "independentswap", "trialswap"),
    runs = 999, iterations = 1000, include.root=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ses.pd_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="ses.pd_+3A_tree">tree</code></td>
<td>
<p> Phylogeny (phylo object) </p>
</td></tr>
<tr><td><code id="ses.pd_+3A_null.model">null.model</code></td>
<td>
<p> Null model to use (see Details section for description) </p>
</td></tr>
<tr><td><code id="ses.pd_+3A_runs">runs</code></td>
<td>
<p> Number of randomizations </p>
</td></tr>
<tr><td><code id="ses.pd_+3A_iterations">iterations</code></td>
<td>
<p> Number of iterations to use for each randomization (for independent swap and trial null models) </p>
</td></tr>
<tr><td><code id="ses.pd_+3A_include.root">include.root</code></td>
<td>
<p> Include distance to root node in calculation of PD (see documentation in <code><a href="#topic+pd">pd</a></code> function </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently implemented null models (arguments to null.model):
</p>

<dl>
<dt>taxa.labels</dt><dd><p> Shuffle taxa labels across tips of phylogeny (across all taxa included in phylogeny)</p>
</dd>
<dt>richness</dt><dd><p> Randomize community data matrix abundances within samples (maintains sample species richness)</p>
</dd>    
<dt>frequency</dt><dd><p> Randomize community data matrix abundances within species (maintains species occurence frequency)</p>
</dd>    
<dt>sample.pool</dt><dd><p> Randomize community data matrix by drawing species from pool of species occurring in at least one community (sample pool) with equal probability</p>
</dd>  
<dt>phylogeny.pool</dt><dd><p> Randomize community data matrix by drawing species from pool of species occurring in the phylogeny (phylogeny pool) with equal probability</p>
</dd>  
<dt>independentswap</dt><dd><p> Randomize community data matrix with the independent swap algorithm (Gotelli 2000) maintaining species occurrence frequency and sample species richness </p>
</dd>
<dt>trialswap</dt><dd><p> Randomize community data matrix with the trial-swap algorithm (Miklos &amp; Podani 2004) maintaining species occurrence frequency and sample species richness </p>
</dd>
</dl>



<h3>Value</h3>

<p>A data frame of results for each community
</p>
<table>
<tr><td><code>ntaxa</code></td>
<td>
<p>Number of taxa in community</p>
</td></tr>
<tr><td><code>pd.obs</code></td>
<td>
<p>Observed PD in community</p>
</td></tr>
<tr><td><code>pd.rand.mean</code></td>
<td>
<p>Mean PD in null communities</p>
</td></tr>
<tr><td><code>pd.rand.sd</code></td>
<td>
<p>Standard deviation of PD in null communities</p>
</td></tr>
<tr><td><code>pd.obs.rank</code></td>
<td>
<p>Rank of observed PD vs. null communities</p>
</td></tr>
<tr><td><code>pd.obs.z</code></td>
<td>
<p>Standardized effect size of PD vs. null communities (= (pd.obs - pd.rand.mean) / pd.rand.sd)</p>
</td></tr>
<tr><td><code>pd.obs.p</code></td>
<td>
<p>P-value (quantile) of observed PD vs. null communities (= mpd.obs.rank / runs + 1)</p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p>Number of randomizations</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>.
</p>
<p>Proches, S., Wilson, J.R.U. and Cowling, R.M. 2006. How much evolutionary history in a 10 x 10m plot? Proceedings of Royal Society of London B, Biological Sciences 273:1143-1148.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+pd">pd</a></code>, <code><a href="#topic+randomizeMatrix">randomizeMatrix</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
ses.pd(phylocom$sample, phylocom$phylo, null.model="taxa.labels", runs=99)</code></pre>

<hr>
<h2 id='specaccum.psr'> Phylogenetic Species Richness Sample-Based Rarefaction Curve </h2><span id='topic+specaccum.psr'></span>

<h3>Description</h3>

<p>Finds a sample-based rarefaction curve for phylogentic species richness for a set of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>specaccum.psr(samp, tree, permutations = 100, method = "random", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="specaccum.psr_+3A_samp">samp</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="specaccum.psr_+3A_tree">tree</code></td>
<td>
<p> A phylo tree object or a phylogenetic covariance matrix</p>
</td></tr>
<tr><td><code id="specaccum.psr_+3A_permutations">permutations</code></td>
<td>
<p> Number of permutations with method <code>method= "random"</code> </p>
</td></tr>
<tr><td><code id="specaccum.psr_+3A_method">method</code></td>
<td>
<p> Species accumulation method, currently only <code>"random"</code> is supported which adds samples in random order. </p>
</td></tr>
<tr><td><code id="specaccum.psr_+3A_...">...</code></td>
<td>
<p> Other parameters to functions </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns an object of class <code>"specaccum"</code> with items:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p> Function call. </p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p> Accumulator method. </p>
</td></tr>
<tr><td><code>sites</code></td>
<td>
<p> Number of sites/samples. </p>
</td></tr>
<tr><td><code>richness</code></td>
<td>
<p> The mean phylogenetic species richness corresponding to number of sites/samples. </p>
</td></tr>
<tr><td><code>sd</code></td>
<td>
<p> The standard deviation of phylogenetic apecies accumulation curve (or its standard error) estimated from permutations in <code>method = "random"</code>. </p>
</td></tr>
<tr><td><code>perm</code></td>
<td>
<p> Permutation results with <code>method = "random"</code> and NULL in other cases. Each column in perm holds one permutation. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Matthew Helmus <a href="mailto:mrhelmus@gmail.com">mrhelmus@gmail.com</a> based on the <code>vegan</code> package <a href="vegan.html#topic+specaccum">specaccum</a> function by Roeland Kindt and Jari Oksanen. </p>


<h3>References</h3>

<p> Gotelli N.J. &amp; Colwell R.K. (2001) Quantifying biodiversity: procedures and pitfalls in the measurement and comparison of species richness. Ecology Letters, 4, 379-391<br />
<br /> Helmus M.R., Bland T.J., Williams C.K. &amp; Ives A.R. (2007) Phylogenetic measures of biodiversity. American Naturalist, 169, E68-E83 </p>


<h3>See Also</h3>

 <p><code><a href="#topic+psr">psr</a></code>, <code><a href="vegan.html#topic+specaccum">specaccum</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
accum.sr&lt;-specaccum(phylocom$sample, permutations = 100, method = "random")
plot(accum.sr, col="blue")
points(accum.sr$sites, accum.sr$richness, pch=19, col="blue")

accum.psr&lt;-specaccum.psr(phylocom$sample, phylocom$phylo, permutations = 100, method = "random")
plot(accum.psr, add=TRUE, col = "red")
points(accum.psr$sites, accum.psr$richness, pch=19, col="red")

legend(5,5,legend=c("SR","PSR"),pch=c(19,19),col=c("blue","red"))
</code></pre>

<hr>
<h2 id='species.dist'> Species co-occurrence distances</h2><span id='topic+species.dist'></span>

<h3>Description</h3>

<p>Compute interspecific distances based on patterns of species co-occurrence in communities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>species.dist(x, metric = c("cij", "jaccard", "checkerboard", "doij"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="species.dist_+3A_x">x</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="species.dist_+3A_metric">metric</code></td>
<td>
<p> Co-occurrence metric to use (see Details section for description) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently implemented co-occurrence measures (arguments to metric):
</p>

<dl>
<dt>cij</dt><dd><p> Schoener's index of co-occurrence </p>
</dd>
<dt>jaccard</dt><dd><p> Jaccard index of co-occurrence </p>
</dd>
<dt>checkerboard</dt><dd><p> Checkerboard index of co-occurrence </p>
</dd>
<dt>doij</dt><dd><p> DOij index of co-occurrence </p>
</dd>    
</dl>



<h3>Value</h3>

<p>A <code>dist</code> object with co-occurrences among all species pairs
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Hardy, O.J. 2008. Testing the spatial phylogenetic structure of local communities: statistical performances of different null models and test statistics on a locally neutral community. Journal of Ecology 96:914-926.
</p>


<h3>See Also</h3>

 <p><code><a href="vegan.html#topic+vegdist">vegdist</a></code> </p>

<hr>
<h2 id='sppregs'> Regressions to Separate Phylogenetic Attraction and Repulsion </h2><span id='topic+sppregs'></span><span id='topic+sppregs.plot'></span>

<h3>Description</h3>

<p>Fit regressions on species abundance or presence/absence across communities and calculate phylogenetic correlations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sppregs(samp, env, tree=NULL, fam="gaussian")
sppregs.plot(sppreg, rows=c(1,3), cex.mag=1, x.label="phylogenetic correlations", 
    y.label=c("occurrence correlations w/ env", "occurrence correlations wo/ env",
    "change in correlations"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sppregs_+3A_samp">samp</code></td>
<td>
<p> community data matrix, species as columns, communities as rows </p>
</td></tr>
<tr><td><code id="sppregs_+3A_env">env</code></td>
<td>
<p> environmental data matrix </p>
</td></tr>
<tr><td><code id="sppregs_+3A_tree">tree</code></td>
<td>
<p>  phylo tree object or a phylogenetic covariance matrix </p>
</td></tr>
<tr><td><code id="sppregs_+3A_fam">fam</code></td>
<td>
<p> with <code>fam = "gaussian"</code> fits with <code><a href="stats.html#topic+glm">glm</a></code>; with <code>fam = "binomial"</code> fit logistic regressions with Firth's bias-reduction using <code><a href="brglm.html#topic+brglm">brglm</a></code>  </p>
</td></tr>
<tr><td><code id="sppregs_+3A_sppreg">sppreg</code></td>
<td>
<p> object from function <code><a href="#topic+sppregs">sppregs</a></code> </p>
</td></tr>
<tr><td><code id="sppregs_+3A_rows">rows</code></td>
<td>
 <p><code>rows = c(1,3)</code> plots in a row; <code>rows = c(3,1)</code> in a column </p>
</td></tr>
<tr><td><code id="sppregs_+3A_cex.mag">cex.mag</code></td>
<td>
<p> value for <code>cex</code> in <code>par</code> </p>
</td></tr>
<tr><td><code id="sppregs_+3A_x.label">x.label</code></td>
<td>
<p> x axis labels </p>
</td></tr>
<tr><td><code id="sppregs_+3A_y.label">y.label</code></td>
<td>
<p> y axis labels </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each species in <code>samp</code>, the function fits regressions of species presence/absence or abundances
on the environmental variables supplied in <code>env</code>; and calculates the <code>(n^2-n)/2</code> pairwise species correlations
between the residuals of these fits and pairwise species phylogenetic correlations. The residuals can be
thought of as the presence/absence of species across sites/communities after accounting for how species respond
to environmental variation across sites. Each set of coefficients can be tested for phylogenetic signal with, for example, 
the function <code><a href="#topic+phylosignal">phylosignal</a></code>. 
<br />
<br />
The function <code>sppregs.plot</code> produces a set of three plots of the correlations of pairwise species phylogenetic correlations versus: 
the observed pairwise correlations of species across communities, the residual correlations, and the pairwise differences between (i.e., the 
change in species co-occurrence once the environmental variables are taken into account). The significance of these correlations can be tested
via permutation with the function <code><a href="#topic+phylostruct">phylostruct</a></code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>family</code></td>
<td>
<p> the regression error distribution </p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p> the residuals from each species regression </p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p> the estimated coefficients from each species regression </p>
</td></tr>
<tr><td><code>std.errors</code></td>
<td>
<p> the standard errors of the coefficients </p>
</td></tr>
<tr><td><code>correlations</code></td>
<td>
<p> correlations of pairwise species phylogenetic correlations between: the observed pairwise correlations of species across communities, the residual correlations, and the pairwise differences between the two </p>
</td></tr>
<tr><td><code>cors.pa</code></td>
<td>
<p> the observed pairwise correlations of species across communities </p>
</td></tr>
<tr><td><code>cors.resid</code></td>
<td>
<p> the residual pairwise correlations of species across communities </p>
</td></tr>
<tr><td><code>cors.phylo</code></td>
<td>
<p> the phylogenetic pairwise correlations among species </p>
</td></tr>
</table>


<h3>Note</h3>

<p>The function requires the library <code><a href="brglm.html#topic+brglm">brglm</a></code> to perform logistic regressions</p>


<h3>Author(s)</h3>

<p> Matthew Helmus <a href="mailto:mrhelmus@gmail.com">mrhelmus@gmail.com</a> </p>


<h3>References</h3>

<p> Helmus M.R., Savage K., Diebel M.W., Maxted J.T. &amp; Ives A.R. (2007) Separating the determinants of phylogenetic community structure. Ecology Letters, 10, 917-925 </p>


<h3>See Also</h3>

<p><code><a href="#topic+phylostruct">phylostruct</a></code>, <code><a href="#topic+phylosignal">phylosignal</a></code></p>

<hr>
<h2 id='tax.distinctiveness'>
Taxonomic distinctiveness sensu Vane-Wright or May
</h2><span id='topic+tax.distinctiveness'></span>

<h3>Description</h3>

<p>Taxic diversity: Vane-Wright et al., 1991 and May 1990 which accounts for polytomies by counting the number of branches descending from each node that lies on the path from a spp tip to the root (not just counting the number of nodes).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax.distinctiveness(tree, type = c("Vane-Wright", "May"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tax.distinctiveness_+3A_tree">tree</code></td>
<td>

<p>an object of class phylo
</p>
</td></tr>
<tr><td><code id="tax.distinctiveness_+3A_type">type</code></td>
<td>

<p>specify &quot;Vane-Wright&quot; or &quot;May&quot;
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Karen Magnuson-Ford, Will Cornwell, Arne Mooers, Mark Vellend
</p>


<h3>References</h3>

<p>Vane-Wright, R.I., Humphries, C.J. and Williams, P.H. (1991). What to protect? - Systematics and the agony of choice. Biological Conservation, 55, 235-254.
</p>
<p>May, R.M. (1990). Taxonomy as destiny. Nature, 347, 129-130.
</p>
<p>Mark Vellend, William K. Cornwell, Karen Magnuson-Ford, and Arne Mooers. In press. Measuring phylogenetic biodiversity In: Biological diversity: frontiers in measurement and assessment. Edited by Anne Magurran and Brian McGill.
</p>

<hr>
<h2 id='traitgram'> Draw phylogeny with nodes at trait positions </h2><span id='topic+traitgram'></span>

<h3>Description</h3>

<p>Draws a phylogeny where x position of nodes and tips corresponds to value of a continuous trait variable, and y position corresponds to node depth (i.e. age).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traitgram(x, phy, xaxt = 's', underscore = FALSE, show.names = TRUE,
            show.xaxis.values = TRUE, method = c('ML','pic'), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traitgram_+3A_x">x</code></td>
<td>
<p> Trait vector (same order as phy\$tip.label, or with taxon names in names) </p>
</td></tr>
<tr><td><code id="traitgram_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="traitgram_+3A_xaxt">xaxt</code></td>
<td>
<p> x axis default type </p>
</td></tr>
<tr><td><code id="traitgram_+3A_underscore">underscore</code></td>
<td>
<p> if FALSE remove underscore from taxonomic names </p>
</td></tr>
<tr><td><code id="traitgram_+3A_show.names">show.names</code></td>
<td>
<p> if TRUE show taxon names across tips of phylogeny </p>
</td></tr>
<tr><td><code id="traitgram_+3A_show.xaxis.values">show.xaxis.values</code></td>
<td>
<p> if TRUE show values for trait on x=axis </p>
</td></tr>
<tr><td><code id="traitgram_+3A_method">method</code></td>
<td>
<p> method for calculation of internal trait values. 'ML' = maximum likelihood method; 'pic' = independent contrasts method. pic option can be used when ML fails to converge or otherwise seems to fail to correctly reconstruct ancestral values</p>
</td></tr>
<tr><td><code id="traitgram_+3A_...">...</code></td>
<td>
<p> Additional arguments passed to plot </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a traitgram, no values returned.
</p>


<h3>Author(s)</h3>

<p> David Ackerly &lt;dackerly@berkeley.edu&gt; </p>


<h3>References</h3>

 
<p>Ackerly, D. D. 2009. Conservatism and diversification of plant functional traits: Evolutionary rates versus phylogenetic signal. Proceedings of the National Academy of Sciences USA 106:19699-19706. doi: 10.1073/pnas.0901635106. 
</p>
<p>Evans, M. E. K., S. A. Smith, R. S. Flynn, and M. J. Donoghue. 2009. Climate, Niche Evolution, and Diversification of the &quot;bird-cage&quot; Evening Primroses (Oenothera, Sections Anogra and Kleinia). American Naturalist 173:225-240. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>randtree &lt;- rcoal(20)
randtraits &lt;- rTraitCont(randtree)
traitgram(randtraits,randtree)
traitgram(randtraits,randtree,method='pic')
</code></pre>

<hr>
<h2 id='unifrac'> Unweighted UniFrac distance between communities </h2><span id='topic+unifrac'></span>

<h3>Description</h3>

<p> Calculates unweighted UniFrac, a phylogenetic beta diversity metric of the the unique (non-shared) fraction of total phylogenetic diversity (branch-length) between two communities. </p>


<h3>Usage</h3>

<pre><code class='language-R'>unifrac(comm, tree)    
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unifrac_+3A_comm">comm</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="unifrac_+3A_tree">tree</code></td>
<td>
<p> Object of class phylo - a rooted phylogeny</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dist object of the unweighted UniFrac distances between communities (the unique (non-shared) fraction of total phylogenetic diversity (branch-length) between two communities).</p>


<h3>Warning </h3>

<p>The UniFrac distance between samples will include the branch length connecting taxa in those samples and the root of the supplied tree. The root of the supplied tree may not be spanned by any taxa in the sample. If you want the root of your tree to correspond to the most recent ancestor of the taxa actually present in your samples, you should prune the tree before running <code>unifrac</code>:
<code>prunedTree &lt;- prune.sample(sample,tree)</code>
</p>


<h3>Note</h3>

<p>The supplied tree must be rooted. Single-species samples will be assigned a PD value equal to the distance from the root to the present.
</p>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p>Lozupone, C., Hamady, M., and Knight, R. 2006. UniFrac - an online tool for comparing microbial community diversity in a phylogenetic context. BMC Bioinformatics 7:371.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pd">pd</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylocom)
unifrac(phylocom$sample, phylocom$phylo)</code></pre>

<hr>
<h2 id='utility'> Picante utility functions </h2><span id='topic+df2vec'></span><span id='topic+internal2tips'></span><span id='topic+node.age'></span><span id='topic+pic.variance'></span><span id='topic+sortColumns'></span><span id='topic+sortRows'></span><span id='topic+taxaShuffle'></span><span id='topic+tipShuffle'></span>

<h3>Description</h3>

<p>Picante utility functions for tree and data manipulation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df2vec(x, colID=1)
internal2tips(phy, int.node, return.names = FALSE)
node.age(phy)
pic.variance(x, phy, scaled = TRUE)
sortColumns(x)
sortRows(x)
taxaShuffle(x)
tipShuffle(phy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utility_+3A_phy">phy</code></td>
<td>
<p> phylo object </p>
</td></tr>
<tr><td><code id="utility_+3A_x">x</code></td>
<td>
<p> A data.frame, matrix or dist object </p>
</td></tr>
<tr><td><code id="utility_+3A_colid">colID</code></td>
<td>
<p> Numeric or character ID of column to include </p>
</td></tr>
<tr><td><code id="utility_+3A_int.node">int.node</code></td>
<td>
<p> internal node number </p>
</td></tr>
<tr><td><code id="utility_+3A_return.names">return.names</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="utility_+3A_scaled">scaled</code></td>
<td>
<p> Scale contrasts by branch length </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Various utility functions for manipulating trees, data, etc.
</p>


<h3>Value</h3>

<table>
<tr><td><code>df2vec</code></td>
<td>
<p>A named vector</p>
</td></tr>
<tr><td><code>internal2tips</code></td>
<td>
<p>Vector of tips descended from a node</p>
</td></tr>
<tr><td><code>node.age</code></td>
<td>
<p>Phylo object with phylo\$ages vector of node ages corresponding to phylo\$edge</p>
</td></tr>
<tr><td><code>pic.variance</code></td>
<td>
<p>Variance of independent contrasts</p>
</td></tr>
<tr><td><code>sortColumns</code></td>
<td>
<p>A data.frame or matrix with columns sorted by name</p>
</td></tr>
<tr><td><code>sortRows</code></td>
<td>
<p>A data.frame or matrix with rows sorted by name</p>
</td></tr>
<tr><td><code>taxaShuffle</code></td>
<td>
<p>Matrix with taxa names shuffled</p>
</td></tr>
<tr><td><code>tipShuffle</code></td>
<td>
<p>Phylo object with taxa names shuffled</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt;, Peter Cowan &lt;pdc@berkeley.edu&gt;, David Ackerly &lt;dackerly@berkeley.edu&gt; </p>

<hr>
<h2 id='writesample'> Write a Phylocom community sample file </h2><span id='topic+writesample'></span>

<h3>Description</h3>

<p>Write a community data matrix to a Phylocom community sample file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writesample(community, filename = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writesample_+3A_community">community</code></td>
<td>
<p> Community data matrix </p>
</td></tr>
<tr><td><code id="writesample_+3A_filename">filename</code></td>
<td>
<p> Filename path </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Steven Kembel &lt;steve.kembel@gmail.com&gt; and Cam Webb &lt;cwebb@oeb.harvard.edu&gt; </p>


<h3>References</h3>

<p> Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>. </p>

<hr>
<h2 id='writetraits'> Write a Phylocom traits formatted file</h2><span id='topic+writetraits'></span>

<h3>Description</h3>

<p>Write a Phylocom traits formatted file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writetraits(trt, file = "", bin = NULL, sigd = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writetraits_+3A_trt">trt</code></td>
<td>
<p> Data frame containing trait data </p>
</td></tr>
<tr><td><code id="writetraits_+3A_file">file</code></td>
<td>
<p> Filename path </p>
</td></tr>
<tr><td><code id="writetraits_+3A_bin">bin</code></td>
<td>
<p> Vector index of trait columns to be treated as binary </p>
</td></tr>
<tr><td><code id="writetraits_+3A_sigd">sigd</code></td>
<td>
<p> Significant digits for output </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> David Ackerly &lt;dackerly@berkeley.edu&gt; and Steven Kembel &lt;steve.kembel@gmail.com&gt; </p>


<h3>References</h3>

<p> Webb, C.O., Ackerly, D.D., and Kembel, S.W. 2008. Phylocom: software for the analysis of phylogenetic community structure and trait evolution. Version 4.0.1. <a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a>. </p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
