<!DOCTYPE html><html><head><title>Help for package PlasmaMutationDetector</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PlasmaMutationDetector}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#background_error_rate'><p>The package provide the SNV and INDEL PERs computed for the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 from 29 controls in a table available in the data file <code>background_error_rate.txt</code>.</p></a></li>
<li><a href='#BuildCtrlErrorRate'><p>function BuildCtrlErrorRate</p></a></li>
<li><a href='#DetectPlasmaMutation'><p>function DetectPlasmaMutation</p></a></li>
<li><a href='#hotspot'><p>The package provide a list of known hotspot positions located on the amplicons of the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 as a txt file <code>hotspot.txt</code> which contains a vector/variable &mdash;named chrpos (first row)&mdash; of chars, of the form chrN:XXXXXXXXX defining genomic positions.</p></a></li>
<li><a href='#LoadBackgroundErrorRate'><p>function LoadBackgroundErrorRate</p></a></li>
<li><a href='#MAF_from_BAM'><p>function MAF_from_BAM</p></a></li>
<li><a href='#positions_ranges'><p>The package provide the positions and ranges computed for the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 as a Rdata file <code>positions_ranges.rda</code>.</p></a></li>
<li><a href='#PrepareLibrary'><p>function PrepareLibrary</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tumor Mutation Detection in Plasma</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-05-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Yves Rozenholc, Nicolas Pécuchet, Pierre Laurent-Puig</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yves Rozenholc &lt;yves.rozenholc@parisdescartes.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Aims at detecting single nucleotide variation
    (SNV) and insertion/deletion (INDEL) in circulating tumor DNA (ctDNA), used
    as a surrogate marker for tumor, at each base position of an Next Generation
    Sequencing (NGS) analysis. Mutations are assessed by comparing the minor-allele
    frequency at each position to the measured PER in control samples.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>S4Vectors (&ge; 0.16.0), Rsamtools (&ge; 1.30.0), rtracklayer (&ge;
1.38.0), robustbase (&ge; 0.92-8), SummarizedExperiment (&ge;
1.8.0)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), ggplot2 (&ge; 2.2.0), grid (&ge; 3.4.0),
GenomicRanges (&ge; 1.30.0), VariantAnnotation (&ge; 1.24.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-11 02:11:42 UTC; rozen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-11 07:43:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='background_error_rate'>The package provide the SNV and INDEL PERs computed for the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 from 29 controls in a table available in the data file <code>background_error_rate.txt</code>.</h2><span id='topic+background_error_rate'></span><span id='topic+background_error_rate.txt'></span>

<h3>Description</h3>

<p>This table contains 9 variables for each genomic position
</p>

<ul>
<li> <p><code>chrpos</code>, char, of the form chrN:XXXXXXXXX defining genomic position
</p>
</li>
<li> <p><code>N0</code>, integer, the coverture in the controls
</p>
</li>
<li> <p><code>E0</code>, integer, the number of errors in the controls
</p>
</li>
<li> <p><code>p.sain</code>, numeric,  the ratio E0/N0
</p>
</li>
<li> <p><code>up.sain</code>, numeric, the 95th quantile of the Binomial with parameter N0 and E0/N0
</p>
</li>
<li> <p><code>E0indel</code>, integer, the amount of indel
</p>
</li>
<li> <p><code>indel.p.sain</code>, numeric, the ration E0indel/N0
</p>
</li>
<li> <p><code>indel.up.sain</code>, numeric, the 95th quantile of the Binomial with parameter N0 and E0indel/N0
</p>
</li>
<li> <p><code>hotspot</code>, char, either 'Non-hotspot' or 'Hotspot' depending if the genomic position is known as hotspot or not.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(background_error_rate)
</code></pre>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>See Also</h3>

<p><code>BuildCtrlErrorRate</code>
</p>

<hr>
<h2 id='BuildCtrlErrorRate'>function BuildCtrlErrorRate</h2><span id='topic+BuildCtrlErrorRate'></span>

<h3>Description</h3>

<p>Compute the SNV Position-Error Rates and INDEL Position-Error Rates from control samples (available in the control directory <code>ctrl.dir</code>).
This function requires MAF files, that will be automatically generated if not present in the specified control folder.
SNV PER is computed as the sum in control samples of SNV background counts / sum in control samples of depths where SNV background counts = depth - major allele count.
INDEL PER is computed as sum in control samples of INDEL background counts / sum in control samples of depths where INDEL background counts = sum of insertion and deletion counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BuildCtrlErrorRate(ctrl.dir = "Plasma ctrl/", bai.ext = ".bai",
  pos_ranges.file = NULL, hotspot.file = NULL, force = FALSE,
  output.dir = ctrl.dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildCtrlErrorRate_+3A_ctrl.dir">ctrl.dir</code></td>
<td>
<p>char, foldername containing the control files (default 'Plasma ctrl/'). The typical folder hierarchy will consist of 'Plasma ctrl/rBAM'</p>
</td></tr>
<tr><td><code id="BuildCtrlErrorRate_+3A_bai.ext">bai.ext</code></td>
<td>
<p>char, filename extension of the bai files (default '.bai')</p>
</td></tr>
<tr><td><code id="BuildCtrlErrorRate_+3A_pos_ranges.file">pos_ranges.file</code></td>
<td>
<p>char, name of the Rdata file containing the three variables <code>pos_ind</code>, <code>pos_snp</code> and <code>pos_ranges</code> as build by the function <code>PrepareLibrary</code>. Default NULL, use the position_ranges.rda provided, used for our analysis.</p>
</td></tr>
<tr><td><code id="BuildCtrlErrorRate_+3A_hotspot.file">hotspot.file</code></td>
<td>
<p>char, name of the text file containing a list of the genomic positions of the hotspots (default NULL, read the provide hotspot.txt, see <code>hotspot</code>)</p>
</td></tr>
<tr><td><code id="BuildCtrlErrorRate_+3A_force">force</code></td>
<td>
<p>boolean, (default FALSE) if TRUE force all computations to all files including already processed ones</p>
</td></tr>
<tr><td><code id="BuildCtrlErrorRate_+3A_output.dir">output.dir</code></td>
<td>
<p>char, name of the folder to save results (default <code>ctrl.dir</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of processed files
</p>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   ctrl.dir = system.file("extdata", "4test_only/ctrl/", package = "PlasmaMutationDetector")
   if (substr(ctrl.dir,nchar(ctrl.dir),nchar(ctrl.dir))!='/')
     ctrl.dir = paste0(ctrl.dir,'/') # TO RUN UNDER WINDOWS
   BuildCtrlErrorRate(ctrl.dir,output.dir=paste0(tempdir(),'/'))
   
## End(Not run)

</code></pre>

<hr>
<h2 id='DetectPlasmaMutation'>function DetectPlasmaMutation</h2><span id='topic+DetectPlasmaMutation'></span>

<h3>Description</h3>

<p>This is the main function of the package that calls mutations by comparing at each genomic position the SNV or INDEL frequencies computed in one tested sample to
the SNV or INDEL Position-Error Rates computed from several control samples by a binomial test. An outlier detection is performed among all intra-sample p-values
to call a mutation.
For users wishing to develop their own analysis for other sequencing panel, it requires recalibrated BAM files control samples to be processed to compute the
Position-Error Rates stored in a file specified in <code>ber.ctrl.file</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectPlasmaMutation(patient.dir = "./", patient.name = NULL,
  pos_ranges.file = NULL, ber.ctrl.file = NULL, bai.ext = ".bai",
  n.trim = 8, cov.min = 0, force = FALSE, show.more = FALSE,
  qcutoff.snv = 0.95, qcutoff.indel = 0.99, cutoff.sb.ref = 0.1,
  cutoff.sb.hotspot = 3.1, cutoff.sb.nonhotspot = 2.5,
  hotspot.indel = "chr7:55227950:55249171", output.dir = patient.dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectPlasmaMutation_+3A_patient.dir">patient.dir</code></td>
<td>
<p>char, foldername containing the rBAM folder of the patients. The typical folder hierarchy will consist of 'Plasma/rBAM'</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_patient.name">patient.name</code></td>
<td>
<p>char, filename of the patient .bam file(s) (default NULL read all patients in folder <code>patient.dir</code>)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_pos_ranges.file">pos_ranges.file</code></td>
<td>
<p>char, name of the Rdata file containing the three variables <code>pos_ind</code>, <code>pos_snp</code>, <code>pos_ranges</code> as build by the function <code>PrepareLibrary</code>. Default NULL, use the position_ranges.rda provides that we used for our analysis.</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_ber.ctrl.file">ber.ctrl.file</code></td>
<td>
<p>char, pathname of the file providing the background error rates obtained from the controls (default NULL use the provided background error rates obtained from our 29 controls). See <code>background_error_rate.txt</code> data and <code>BuildCtrlErrorRate</code> function.</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_bai.ext">bai.ext</code></td>
<td>
<p>char, filename extension of the bai files (default '.bai')</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_n.trim">n.trim</code></td>
<td>
<p>integer, number of base positions trimmed at the ends of each amplicon (default 8)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_cov.min">cov.min</code></td>
<td>
<p>integer, minimal coverture required at each position (default 0)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_force">force</code></td>
<td>
<p>boolean, (default FALSE) if TRUE force all computations to all files including already processed ones</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_show.more">show.more</code></td>
<td>
<p>boolean, (default FALSE show only detected positions) if TRUE additional annotations on result plots are given for non-significant mutations</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_qcutoff.snv">qcutoff.snv</code></td>
<td>
<p>numeric, proportion of kept base positions ranged by increasing 95th percentile SNV PER in control samples (default 0.95)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_qcutoff.indel">qcutoff.indel</code></td>
<td>
<p>numeric, proportion of kept base positions ranged by increasing 95th percentile INDEL PER in control samples (default 0.99)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_cutoff.sb.ref">cutoff.sb.ref</code></td>
<td>
<p>numeric, exclude reference positions without cutoff &lt; strand bias &lt; 1-cutoff (default 0.1) (see Supplementary Materials in References)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_cutoff.sb.hotspot">cutoff.sb.hotspot</code></td>
<td>
<p>numeric, exclude hotspot positions with Symmetric Odds Ratio test &gt; cutoff (default 3.1) (see Supplementary Materials in References)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_cutoff.sb.nonhotspot">cutoff.sb.nonhotspot</code></td>
<td>
<p>numeric, exclude non-hotspot positions with Symmetric Odds Ratio test &gt; cutoff (default 2.5) (see Supplementary Materials in References)</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_hotspot.indel">hotspot.indel</code></td>
<td>
<p>char, a vector containing the known positions of hotspot deletion/insertion defined as chrX:start:end (default 'chr7:55227950:55249171')</p>
</td></tr>
<tr><td><code id="DetectPlasmaMutation_+3A_output.dir">output.dir</code></td>
<td>
<p>char, name of the folder to save results  (default <code>patient.dir</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of processed patients
</p>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     patient.dir=system.file("extdata","4test_only/case/",package="PlasmaMutationDetector")
     if (substr(patient.dir,nchar(patient.dir),nchar(patient.dir))!='/')
       patient.dir = paste0(patient.dir,'/') # TO RUN UNDER WINDOWS
     DetectPlasmaMutation(patient.dir,output.dir=paste0(tempdir(),'/'))


</code></pre>

<hr>
<h2 id='hotspot'>The package provide a list of known hotspot positions located on the amplicons of the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 as a txt file <code>hotspot.txt</code> which contains a vector/variable &mdash;named chrpos (first row)&mdash; of chars, of the form chrN:XXXXXXXXX defining genomic positions.</h2><span id='topic+hotspot'></span><span id='topic+hotspot.txt'></span>

<h3>Description</h3>

<p>The package provide a list of known hotspot positions located on the amplicons of the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 as a txt file <code>hotspot.txt</code> which contains a vector/variable &mdash;named chrpos (first row)&mdash; of chars, of the form chrN:XXXXXXXXX defining genomic positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hotspot)
</code></pre>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>

<hr>
<h2 id='LoadBackgroundErrorRate'>function LoadBackgroundErrorRate</h2><span id='topic+LoadBackgroundErrorRate'></span>

<h3>Description</h3>

<p>This function will load the background error rates created from the controls using the function BuildCtrlErrorRate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LoadBackgroundErrorRate(pos_ranges.file, ber.ctrl.file, n.trim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LoadBackgroundErrorRate_+3A_pos_ranges.file">pos_ranges.file</code></td>
<td>
<p>char, name of the Rdata file containing the three variables <code>pos_ind</code>, <code>pos_snp</code>, <code>pos_ranges</code> as build by the function <code>PrepareLibrary</code>. Default NULL, use the position_ranges.rda provides that we used for our analysis.</p>
</td></tr>
<tr><td><code id="LoadBackgroundErrorRate_+3A_ber.ctrl.file">ber.ctrl.file</code></td>
<td>
<p>char, pathname of the file providing the background error rates obtained from the controls (default NULL use the provided background error rates obtained from our 29 controls). See <code>background_error_rate.txt</code> data and <code>BuildCtrlErrorRate</code> function.</p>
</td></tr>
<tr><td><code id="LoadBackgroundErrorRate_+3A_n.trim">n.trim</code></td>
<td>
<p>integer, number of base positions trimmed at the ends of each amplicon (default 8)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the adapted background error rate
</p>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>

<hr>
<h2 id='MAF_from_BAM'>function MAF_from_BAM</h2><span id='topic+MAF_from_BAM'></span>

<h3>Description</h3>

<p>Read BAM files  and create MAF file. BAMfiles are stored in a sub-folder '/rBAM'.
MAF files are intermediate files stored in a sub-folder '/BER'.
MAF files contain the raw counts of A,T,C,G, insertion, deletion, insertion&gt;2bp, deletion &gt;2bp for strand plus and stand minus.
Note : we strongly recommand to externally recalibrate BAM files using tools like GATK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAF_from_BAM(study.dir = "Plasma/", input.filenames = NULL,
  bai.ext = ".bai", pos_ranges.file = NULL, force = FALSE,
  output.dir = study.dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAF_from_BAM_+3A_study.dir">study.dir</code></td>
<td>
<p>char, name of the folder containing the rBAM directory  (default 'Plasma/'). The typical folder hierarchy will consist of 'Plasma/rBAM'</p>
</td></tr>
<tr><td><code id="MAF_from_BAM_+3A_input.filenames">input.filenames</code></td>
<td>
<p>a vector of char (default NULL), the names of the BAM files to process. If NULL all BAM files in the rBAM folder will be processed</p>
</td></tr>
<tr><td><code id="MAF_from_BAM_+3A_bai.ext">bai.ext</code></td>
<td>
<p>char, filename extension of the bai files (default '.bai')</p>
</td></tr>
<tr><td><code id="MAF_from_BAM_+3A_pos_ranges.file">pos_ranges.file</code></td>
<td>
<p>char, name of the Rdata file containing the three variables <code>pos_ind</code>, <code>pos_snp</code> and <code>pos_ranges</code> as build by the function <code>PrepareLibrary</code>. Default NULL, use the position_ranges.rda provided, used for our analysis.</p>
</td></tr>
<tr><td><code id="MAF_from_BAM_+3A_force">force</code></td>
<td>
<p>boolean, (default FALSE) if TRUE force all computations to all files including already processed ones</p>
</td></tr>
<tr><td><code id="MAF_from_BAM_+3A_output.dir">output.dir</code></td>
<td>
<p>char, name of the folder to save results  (default <code>study.dir</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the path/names of the MAF files
</p>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
     ctrl.dir = system.file("extdata", "4test_only/ctrl/", package = "PlasmaMutationDetector")
     if (substr(ctrl.dir,nchar(ctrl.dir),nchar(ctrl.dir))!='/')
       ctrl.dir = paste0(ctrl.dir,'/') # TO RUN UNDER WINDOWS
     MAF_from_BAM(ctrl.dir,force=TRUE,output.dir=paste0(tempdir(),'/'))
   
## End(Not run)

</code></pre>

<hr>
<h2 id='positions_ranges'>The package provide the positions and ranges computed for the Ion AmpliSeq™ Colon and Lung Cancer Panel v2 as a Rdata file <code>positions_ranges.rda</code>.</h2><span id='topic+positions_ranges'></span><span id='topic+positions_ranges.rda'></span><span id='topic+pos_ind'></span><span id='topic+pos_snp'></span><span id='topic+pos_ranges'></span>

<h3>Description</h3>

<p>This file contains 4 variables
</p>

<ul>
<li> <p><code>pos_ind</code>, vector of chars, of the form chrN:XXXXXXXXX defining genomic positions of the Ion AmpliSeq™ Colon and Lung Cancer Panel v2
</p>
</li>
<li> <p><code>pos_snp</code>, vector of chars, of the form chrN:XXXXXXXXX defining the known snp genomic positions
</p>
</li>
<li> <p><code>pos_ranges</code>, GRanges object, describing the 92 amplicons of the Ion AmpliSeq™ Colon and Lung Cancer Panel v2
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(positions_ranges)
</code></pre>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>See Also</h3>

<p><code>Prepare_Library</code>
</p>

<hr>
<h2 id='PrepareLibrary'>function PrepareLibrary</h2><span id='topic+PrepareLibrary'></span>

<h3>Description</h3>

<p>Define the Genomic Ranges and Genomic Positions covered by the AmpliSeq™ Panel to include in the study and define SNP positions to exclude from the study.
Trimming amplicon ends is performed if specified. This function is mostly useful if you want to add some SNP positions which are not existing in the
positions_ranges.rda file provided within the package. It is provided to be able to reconstruct <code>positions_ranges.rda</code> data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrepareLibrary(info.dir = "Info/", bed.filename = "lungcolonV2.bed.txt",
  snp.filename = "ExAC.r0.3.sites.vep.vcf.gz",
  snp.extra = c("chr2:212812097", "chr4:1807909", "chr7:140481511",
  "chr14:105246474", "chr18:48586344", "chr19:1223055"),
  output.name = "positions_ranges.rda", output.dir = info.dir,
  load.from.broad.insitute = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrepareLibrary_+3A_info.dir">info.dir</code></td>
<td>
<p>char, name of the folder containing the library information files (default 'Info/')</p>
</td></tr>
<tr><td><code id="PrepareLibrary_+3A_bed.filename">bed.filename</code></td>
<td>
<p>char, name of a BED table (tab-delimited) describing the Panel (with first 3 columns: &quot;chr&quot; (ex:chr1), &quot;start position&quot; (ex:115252190), &quot;end position&quot; (ex:115252305), i.e. the Ion AmpliSeq™ Colon and Lung Cancer Research Panel v2 (default 'lungcolonV2.bed.txt' as provided in the inst/extdata/Info folder of the package).</p>
</td></tr>
<tr><td><code id="PrepareLibrary_+3A_snp.filename">snp.filename</code></td>
<td>
<p>char, name of the vcf file describing known SNP positions, obtained from ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3/ExAC.r0.3.sites.vep.vcf.gz (default 'ExAC.r0.3.sites.vep.vcf.gz'). It requires a corresponding TBI file to be in the same folder (obtained from ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3/ExAC.r0.3.sites.vep.vcf.gz.tbi)</p>
</td></tr>
<tr><td><code id="PrepareLibrary_+3A_snp.extra">snp.extra</code></td>
<td>
<p>a vector of char, a vector of extra known snp positions manually curated (ex:&quot;chrN:XXXXXXXXX&quot;)</p>
</td></tr>
<tr><td><code id="PrepareLibrary_+3A_output.name">output.name</code></td>
<td>
<p>char, filename to save <code>pos_ind</code> and <code>pos_snp</code> (default 'positions_ranges.rda')</p>
</td></tr>
<tr><td><code id="PrepareLibrary_+3A_output.dir">output.dir</code></td>
<td>
<p>char, directory where to save <code>pos_ind</code> and <code>pos_snp</code> (default <code>info.dir</code>)</p>
</td></tr>
<tr><td><code id="PrepareLibrary_+3A_load.from.broad.insitute">load.from.broad.insitute</code></td>
<td>
<p>boolean, if TRUE load <code>snp.filename</code> from Broad Institute ftp server otherwise use the file positions_ranges_broad.rda (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Save the following variables in a .rda file defined by <code>output.name</code> in the folder defined by <code>output.dir</code>:
</p>

<ul>
<li> <p><code>pos_ranges</code>, a GRanges descriptor of amplicon positions
</p>
</li>
<li> <p><code>pos_ind</code>, a vector of char &quot;chrN:XXXXXXXXX&quot;, defining ALL index positions
</p>
</li>
<li> <p><code>pos_snp</code>, a vector of char &quot;chrN:XXXXXXXXX&quot;, defining SNP positions
</p>
</li></ul>



<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>See Also</h3>

<p>positions_ranges,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   bad.pos = "chr7:15478"
   PrepareLibrary(info.dir='./',snp.extra=bad.pos,output.dir=paste0(tempdir(),'/'))


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
