<!DOCTYPE html><html><head><title>Help for package Plasmidprofiler</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Plasmidprofiler}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#amr_positives'><p>Identify Antimicrobial Resistance Positive Plasmids from Blast Results</p></a></li>
<li><a href='#amr_presence'><p>Adds the AMR_gene column to report</p></a></li>
<li><a href='#blast_parser'><p>Blast Results Parser Function</p></a></li>
<li><a href='#blastdata'><p>Example Table of Blast Results</p></a></li>
<li><a href='#combine_results'><p>Combines SRST2 and Blast results into a single dataframe</p></a></li>
<li><a href='#create_grob'><p>Create Heatmap Graphical Object</p></a></li>
<li><a href='#create_plotly'><p>Create Plotly Object</p></a></li>
<li><a href='#define_colours'><p>Defining Colours Based on a Column of Data</p></a></li>
<li><a href='#file_cacher'><p>Filecacher</p></a></li>
<li><a href='#main'><p>Main: Run everything</p></a></li>
<li><a href='#minmax'><p>Minmax</p></a></li>
<li><a href='#normalize'><p>Normalize</p></a></li>
<li><a href='#order_report'><p>Order the Report</p></a></li>
<li><a href='#plot_heatmap'><p>Create GGPLOT Heatmap</p></a></li>
<li><a href='#read_blast'><p>Blast file import function</p></a></li>
<li><a href='#read_srst2'><p>SRST2 file import function</p></a></li>
<li><a href='#report'><p>Example Complete Report after the following steps.</p>
Blast data from attached blastdata table
SRST2 data from attached srst2data table</a></li>
<li><a href='#save_files'><p>Save Files</p></a></li>
<li><a href='#srst2data'><p>Example Table of SRST2 Results</p></a></li>
<li><a href='#subsampler'><p>Subsetting Results</p></a></li>
<li><a href='#tree_maker'><p>Create Dendrogram Based on Plasmid Content</p></a></li>
<li><a href='#zetner_score'><p>Adds the Zetner Score column to report</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization of Plasmid Profile Results</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.6</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Adrian Zetner &lt;adrian.zetner@phac-aspc.gc.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains functions developed to combine the results of querying a plasmid database using
    short-read sequence typing with the results of a blast analysis against the query results.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Pandoc (&gt;= 1.15)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>lintr</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, dplyr, gdata, ggdendro, ggplot2, grid, gridExtra, gtable,
htmlwidgets, magrittr, plotly, plyr, RColorBrewer, reshape2,
stringr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-01-05 23:40:35 UTC; adrian</td>
</tr>
<tr>
<td>Author:</td>
<td>Adrian Zetner [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-01-06 01:10:47</td>
</tr>
</table>
<hr>
<h2 id='amr_positives'>Identify Antimicrobial Resistance Positive Plasmids from Blast Results</h2><span id='topic+amr_positives'></span>

<h3>Description</h3>

<p>This function loads the imported blast results, identifies which plasmids
carry AMR genes at highest identity. May have issues with multiple genes per
plasmid, currently optimized for identifying one of two genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>amr_positives(blast.results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amr_positives_+3A_blast.results">blast.results</code></td>
<td>
<p>Blast results loaded from read_blast or from Global Env</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Two column DF of plasmid names and genes present
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
amr_positives(blastdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='amr_presence'>Adds the AMR_gene column to report</h2><span id='topic+amr_presence'></span>

<h3>Description</h3>

<p>Appends the results of amr_positives to the report
in column AMR_gene, missing have &quot;-&quot; instead
</p>


<h3>Usage</h3>

<pre><code class='language-R'>amr_presence(report, pos.samples)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amr_presence_+3A_report">report</code></td>
<td>
<p>Dataframe of results produced
by <code><a href="#topic+subsampler">subsampler</a></code> or <code><a href="#topic+combine_results">combine_results</a></code></p>
</td></tr>
<tr><td><code id="amr_presence_+3A_pos.samples">pos.samples</code></td>
<td>
<p>Two column DF of plasmid names and genes present produced
by <code><a href="#topic+amr_positives">amr_positives</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Report with AMR_genes added
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsampler">subsampler</a></code>, <code><a href="#topic+combine_results">combine_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
amr_presence(report, pos.samples)

## End(Not run)
</code></pre>

<hr>
<h2 id='blast_parser'>Blast Results Parser Function</h2><span id='topic+blast_parser'></span>

<h3>Description</h3>

<p>Loads the imported blast results, extracts desired columns, Create new column of ratio between hit
length to query length - higher as denominator, adjusts pID by this ratio. Any AMR results are removed from the returned df.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blast_parser(blast.results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blast_parser_+3A_blast.results">blast.results</code></td>
<td>
<p>Blast results loaded from read_blast or Global Env</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Blast table with pID adjusted by ratio of hit length to query length (larger as denominator)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
blast_parser(blastdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='blastdata'>Example Table of Blast Results</h2><span id='topic+blastdata'></span>

<h3>Description</h3>

<p>Example Table of Blast Results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(blastdata)
</code></pre>


<h3>Format</h3>

<p>Dataframe.</p>


<h3>Source</h3>

<p>Strains graciously provided by the authors of the following papers:
Complete Genome and Plasmid Sequences of Three Canadian Isolates of
Salmonella enterica subsp. enterica Serovar Heidelberg from Human
and Food Sources. 2016 Labbe et al.
PMID: 26769926
</p>
<p>Complete Sequence of Four Multidrug-Resistant MOBQ1 Plasmids Harboring
blaGES-5 Isolated from Escherichia coli and Serratia marcescens
Persisting in a Hospital in Canada. 2015 Boyd et al.
PMID: 25545311
</p>
<p>Colistin-Nonsusceptible Pseudomonas aeruginosa Sequence Type 654 with
blaNDM-1 Arrives in North America. 2016 Mataseje et al.
PMID: 26824951
</p>


<h3>References</h3>

<p>None Yet
(<a href="http://www.ncbi.nlm.nih.gov/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(blastdata)
</code></pre>

<hr>
<h2 id='combine_results'>Combines SRST2 and Blast results into a single dataframe</h2><span id='topic+combine_results'></span>

<h3>Description</h3>

<p>Combines blast and SRST2 results, cuts to desired columns (Sample,
Plasmid, Inc_group, Coverage, Divergence, Length, Clusterid), matches
plasmids to BR and appends simplified INC names, all future
modifications are done to this dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_results(sr, br)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_results_+3A_sr">sr</code></td>
<td>
<p>SRST2 results loaded from read_srst2</p>
</td></tr>
<tr><td><code id="combine_results_+3A_br">br</code></td>
<td>
<p>Blast results parsed by blast_parser</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seven column dataframe of SRST2 results now including INC groups
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
combine_results(example_srst2_results, example_blast_results)

## End(Not run)
</code></pre>

<hr>
<h2 id='create_grob'>Create Heatmap Graphical Object</h2><span id='topic+create_grob'></span>

<h3>Description</h3>

<p>Combines the tree, heatmap, and titles to create final heatmap image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_grob(report, grob.title = "Plasmid Profiles")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_grob_+3A_report">report</code></td>
<td>
<p>Dataframe of results</p>
</td></tr>
<tr><td><code id="create_grob_+3A_grob.title">grob.title</code></td>
<td>
<p>Title of heatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Composite image
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
create_grob(report, grob.title="Plasmid Profiles")

## End(Not run)
</code></pre>

<hr>
<h2 id='create_plotly'>Create Plotly Object</h2><span id='topic+create_plotly'></span>

<h3>Description</h3>

<p>Builds the heatmap, creates final interactive plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_plotly(report, user, api.key, post = NA, title = "Plasmid Profiles",
  len.highlight = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_plotly_+3A_report">report</code></td>
<td>
<p>Dataframe of results</p>
</td></tr>
<tr><td><code id="create_plotly_+3A_user">user</code></td>
<td>
<p>User ID for plotly web publishing</p>
</td></tr>
<tr><td><code id="create_plotly_+3A_api.key">api.key</code></td>
<td>
<p>API key for plotly web publishing</p>
</td></tr>
<tr><td><code id="create_plotly_+3A_post">post</code></td>
<td>
<p>Flag determines whether or not to post to plotly (default NA, no post)</p>
</td></tr>
<tr><td><code id="create_plotly_+3A_title">title</code></td>
<td>
<p>Title of heatmap</p>
</td></tr>
<tr><td><code id="create_plotly_+3A_len.highlight">len.highlight</code></td>
<td>
<p>If anything but NA will highlight the largest plasmid hit per incompatibility group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plotly object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
create_plotly(report, title="Plasmid Profiles")

## End(Not run)
</code></pre>

<hr>
<h2 id='define_colours'>Defining Colours Based on a Column of Data</h2><span id='topic+define_colours'></span>

<h3>Description</h3>

<pre>
This function uses RColorBrewer to produce palettes based on the factor levels of the identified column in a report.
   </pre>


<h3>Usage</h3>

<pre><code class='language-R'>define_colours(report, column)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="define_colours_+3A_report">report</code></td>
<td>
<p>Dataframe of results produced by <code><a href="#topic+subsampler">subsampler</a></code> or <code><a href="#topic+combine_results">combine_results</a></code></p>
</td></tr>
<tr><td><code id="define_colours_+3A_column">column</code></td>
<td>
<p>Specify a column by name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named vector of colours, names are factor levels of column supplied
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
define_colours(report, "AMR_gene")

## End(Not run)
</code></pre>

<hr>
<h2 id='file_cacher'>Filecacher</h2><span id='topic+file_cacher'></span>

<h3>Description</h3>

<p>Creates filecache environment if needed for transferring variables
between functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>file_cacher()
</code></pre>

<hr>
<h2 id='main'>Main: Run everything</h2><span id='topic+main'></span>

<h3>Description</h3>

<p>Run all the interim functions to produce outputs.
Can be run in order individually if desired.
</p>

<ol>
<li> <p><code><a href="#topic+read_blast">read_blast</a></code> Import the blast file, add column names
</p>
</li>
<li> <p><code><a href="#topic+blast_parser">blast_parser</a></code> Parse imported file
</p>
</li>
<li> <p><code><a href="#topic+amr_positives">amr_positives</a></code> Detect AMR positive plasmids
</p>
</li>
<li> <p><code><a href="#topic+read_srst2">read_srst2</a></code> Import SRST2 file
</p>
</li>
<li> <p><code><a href="#topic+combine_results">combine_results</a></code> Combine SRST2 and Blast
</p>
</li>
<li> <p><code><a href="#topic+zetner_score">zetner_score</a></code> Add Sureness value
</p>
</li>
<li> <p><code><a href="#topic+amr_presence">amr_presence</a></code> Add detected AMR to report
</p>
</li>
<li> <p><code><a href="#topic+subsampler">subsampler</a></code> Apply filters to report
</p>
</li>
<li> <p><code><a href="#topic+order_report">order_report</a></code> Arrange report
</p>
</li>
<li> <p><code><a href="#topic+save_files">save_files</a></code> Save JPG and CSV
</p>
</li>
<li> <p><code><a href="#topic+create_plotly">create_plotly</a></code> Creates plot
</p>
</li>
<li> <p><code><a href="#topic+save_files">save_files</a></code> Save HTML plot
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>main(blast.file, srst2.file, coverage.filter = NA, sureness.filter = NA,
  length.filter = NA, combine.inc = NA, plotly.user, plotly.api,
  post.plotly = NA, anonymize = NA, main.title = "Plasmid Profiles")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="main_+3A_blast.file">blast.file</code></td>
<td>
<p>Either system location of blast results (tsv) or dataframe</p>
</td></tr>
<tr><td><code id="main_+3A_srst2.file">srst2.file</code></td>
<td>
<p>Either system location of srst2 results (tsv) or dataframe</p>
</td></tr>
<tr><td><code id="main_+3A_coverage.filter">coverage.filter</code></td>
<td>
<p>Filters results below percent read coverage specified (eg. 80)</p>
</td></tr>
<tr><td><code id="main_+3A_sureness.filter">sureness.filter</code></td>
<td>
<p>Filters results below sureness specified (eg. 0.75)</p>
</td></tr>
<tr><td><code id="main_+3A_length.filter">length.filter</code></td>
<td>
<p>Filters plasmid sequences shorter than length specified (eg. 10000)</p>
</td></tr>
<tr><td><code id="main_+3A_combine.inc">combine.inc</code></td>
<td>
<p>Flag to combine incompatibility sub-groups into their main type (set to 1)</p>
</td></tr>
<tr><td><code id="main_+3A_plotly.user">plotly.user</code></td>
<td>
<p>Enter your plotly info to upload to (<a href="https://plot.ly/feed/">Plotly</a>)</p>
</td></tr>
<tr><td><code id="main_+3A_plotly.api">plotly.api</code></td>
<td>
<p>Enter your plotly info to upload to (<a href="https://plot.ly/feed/">Plotly</a>)</p>
</td></tr>
<tr><td><code id="main_+3A_post.plotly">post.plotly</code></td>
<td>
<p>Flag to post to (<a href="https://plot.ly/feed/">Plotly</a>)</p>
</td></tr>
<tr><td><code id="main_+3A_anonymize">anonymize</code></td>
<td>
<p>Flag to post to anonymize plasmids and samples (set to 1)</p>
</td></tr>
<tr><td><code id="main_+3A_main.title">main.title</code></td>
<td>
<p>A title for the figure</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saves output files in working directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>main(blastdata,
srst2data,
coverage.filter=NA,
sureness.filter=0.75,
length.filter=10000,
main.title="Example Results")
</code></pre>

<hr>
<h2 id='minmax'>Minmax</h2><span id='topic+minmax'></span>

<h3>Description</h3>

<p>Takes two columns of numerical data,
normalizes it to ranges from 0 to 1 (0 to -1 for minimums),
sums them, arranges by sum, then returns the sorted dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minmax(df, maxcol, mincol)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="minmax_+3A_df">df</code></td>
<td>
<p>Dataframe</p>
</td></tr>
<tr><td><code id="minmax_+3A_maxcol">maxcol</code></td>
<td>
<p>Column to normalize from 0 to 1</p>
</td></tr>
<tr><td><code id="minmax_+3A_mincol">mincol</code></td>
<td>
<p>Column to normalize from 0 to -1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe sorted by sum of maxcol and mincol
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 minmax(report, "Length", "Coverage")
 
## End(Not run)
</code></pre>

<hr>
<h2 id='normalize'>Normalize</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>Normalizes a vector of values to a range of 0-1
x - min(x)) / (max(x) - min(x)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_+3A_x">x</code></td>
<td>
<p>Vector of values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Normalized vector of values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 normalize(x)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='order_report'>Order the Report</h2><span id='topic+order_report'></span>

<h3>Description</h3>

<p>Order the report first by sample order (tree), then by incompatibility group, then by sureness on each plasmid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order_report(report, anonymize = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="order_report_+3A_report">report</code></td>
<td>
<p>Dataframe of results produced by <code><a href="#topic+subsampler">subsampler</a></code> or <code><a href="#topic+combine_results">combine_results</a></code></p>
</td></tr>
<tr><td><code id="order_report_+3A_anonymize">anonymize</code></td>
<td>
<p>Flag to anything other than NA to replace plasmid and sample names with generic names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Ordered report
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsampler">subsampler</a></code>, <code><a href="#topic+combine_results">combine_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
order_report(report)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot_heatmap'>Create GGPLOT Heatmap</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>Using a ggplot2 tile geometry this function will create a heatmap of values in the report
coloured by incompatibility group, with alpha values from the sureness score. The order of
samples is determined by order_report and plasmids by incompatibility group and sureness score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(report, len.highlight = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_heatmap_+3A_report">report</code></td>
<td>
<p>Dataframe of results</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_len.highlight">len.highlight</code></td>
<td>
<p>If anything but NA will highlight the largest plasmid hit per incompatibility group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>GGPLOT plotted heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plot_heatmap(report)

## End(Not run)
</code></pre>

<hr>
<h2 id='read_blast'>Blast file import function</h2><span id='topic+read_blast'></span>

<h3>Description</h3>

<p>This function imports the 25 column blast file and adds column headers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_blast(br.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_blast_+3A_br.file">br.file</code></td>
<td>
<p>System location of the blast file, no default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of blast data with correct column headers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
read_blast("/data/blast_results.tsv")

## End(Not run)
</code></pre>

<hr>
<h2 id='read_srst2'>SRST2 file import function</h2><span id='topic+read_srst2'></span>

<h3>Description</h3>

<p>This function imports the 14 column SRST2 file. Kind of superfluous
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_srst2(srst2.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_srst2_+3A_srst2.file">srst2.file</code></td>
<td>
<p>System location of the srst2 file, no default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of srst2 data with correct column headers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
read_srst2("/data/srst2_results.tsv")

## End(Not run)
</code></pre>

<hr>
<h2 id='report'>Example Complete Report after the following steps.
Blast data from attached blastdata table
SRST2 data from attached srst2data table</h2><span id='topic+report'></span>

<h3>Description</h3>

<p>read_blast Import the blast file, add column names
blast_parser Parse imported file
amr_positives Detect AMR positive plasmids
read_srst2 Import SRST2 file
combine_results Combine SRST2 and Blast
zetner_score Add Sureness value
amr_presence Add detected AMR to report
order_report Arrange report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(report)
</code></pre>


<h3>Format</h3>

<p>Dataframe.</p>


<h3>Source</h3>

<p>Strains graciously provided by the authors of the following papers:
Complete Genome and Plasmid Sequences of Three Canadian Isolates of
Salmonella enterica subsp. enterica Serovar Heidelberg from Human
and Food Sources. 2016 Labbe et al.
PMID: 26769926
</p>
<p>Complete Sequence of Four Multidrug-Resistant MOBQ1 Plasmids Harboring
blaGES-5 Isolated from Escherichia coli and Serratia marcescens
Persisting in a Hospital in Canada. 2015 Boyd et al.
PMID: 25545311
</p>
<p>Colistin-Nonsusceptible Pseudomonas aeruginosa Sequence Type 654 with
blaNDM-1 Arrives in North America. 2016 Mataseje et al.
PMID: 26824951
</p>


<h3>References</h3>

<p>None Yet
(<a href="http://www.ncbi.nlm.nih.gov/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(report)
</code></pre>

<hr>
<h2 id='save_files'>Save Files</h2><span id='topic+save_files'></span>

<h3>Description</h3>

<p>Save various files: JPG, CSV, HTML depending on parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_files(report, plot.png = NA, report.csv = NA, webpage = NA,
  title = "Plasmid Profiles")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save_files_+3A_report">report</code></td>
<td>
<p>Dataframe of results</p>
</td></tr>
<tr><td><code id="save_files_+3A_plot.png">plot.png</code></td>
<td>
<p>Do you want to save a png? (Anything but NA)</p>
</td></tr>
<tr><td><code id="save_files_+3A_report.csv">report.csv</code></td>
<td>
<p>Do you want to save a text report? (Anything but NA)</p>
</td></tr>
<tr><td><code id="save_files_+3A_webpage">webpage</code></td>
<td>
<p>Do you want to save an interactive heatmap as html? (Anything but NA)</p>
</td></tr>
<tr><td><code id="save_files_+3A_title">title</code></td>
<td>
<p>Enter a title for the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named vector of colours, names are factor levels of column supplied
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 save_files(report, plot.png=1, report.csv=1, webpage=NA)

## End(Not run)
</code></pre>

<hr>
<h2 id='srst2data'>Example Table of SRST2 Results</h2><span id='topic+srst2data'></span>

<h3>Description</h3>

<p>Example Table of SRST2 Results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(srst2data)
</code></pre>


<h3>Format</h3>

<p>Dataframe.</p>


<h3>Source</h3>

<p>Strains graciously provided by the authors of the following papers:
Complete Genome and Plasmid Sequences of Three Canadian Isolates of
Salmonella enterica subsp. enterica Serovar Heidelberg from Human
and Food Sources. 2016 Labbe et al.
PMID: 26769926
</p>
<p>Complete Sequence of Four Multidrug-Resistant MOBQ1 Plasmids Harboring
blaGES-5 Isolated from Escherichia coli and Serratia marcescens
Persisting in a Hospital in Canada. 2015 Boyd et al.
PMID: 25545311
</p>
<p>Colistin-Nonsusceptible Pseudomonas aeruginosa Sequence Type 654 with
blaNDM-1 Arrives in North America. 2016 Mataseje et al.
PMID: 26824951
</p>


<h3>References</h3>

<p>None Yet
(<a href="http://www.ncbi.nlm.nih.gov/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(srst2data)
</code></pre>

<hr>
<h2 id='subsampler'>Subsetting Results</h2><span id='topic+subsampler'></span>

<h3>Description</h3>

<pre>Several filters can be applied:
   Coverage: Filters results below percent read coverage specified
               eg. 95.9 cuts results where reads covered less than 95.9% of the total length
   Sureness: Filters results below sureness specified
               eg. 0.9 cuts results where the sureness falls below 0.9
   Length:   Filters plasmid sequences shorter than length specified
               eg. 10000 cuts out results where the plasmid was less than 10kb
   Incompatibility groups can also be combined (eg. Fii(S) and Fii(K) are combined into Fii)</pre>


<h3>Usage</h3>

<pre><code class='language-R'>subsampler(report, cov.filter = NA, sure.filter = NA, len.filter = NA,
  inc.combine = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsampler_+3A_report">report</code></td>
<td>
<p>Dataframe of results produced
by <code><a href="#topic+subsampler">subsampler</a></code> or <code><a href="#topic+combine_results">combine_results</a></code></p>
</td></tr>
<tr><td><code id="subsampler_+3A_cov.filter">cov.filter</code></td>
<td>
<p>Filters results below percent read coverage specified (eg. 80)</p>
</td></tr>
<tr><td><code id="subsampler_+3A_sure.filter">sure.filter</code></td>
<td>
<p>Filters results below sureness specified (eg. 0.75)</p>
</td></tr>
<tr><td><code id="subsampler_+3A_len.filter">len.filter</code></td>
<td>
<p>Filters plasmid sequences shorter than length specified (eg. 10000)</p>
</td></tr>
<tr><td><code id="subsampler_+3A_inc.combine">inc.combine</code></td>
<td>
<p>Flag to ombine incompatibility sub-groups into their main type (set to 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Report with filters applied
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsampler">subsampler</a></code>, <code><a href="#topic+combine_results">combine_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
subsampler(report, sureness.filter = 0.75, len.filter = 10000)

## End(Not run)
</code></pre>

<hr>
<h2 id='tree_maker'>Create Dendrogram Based on Plasmid Content</h2><span id='topic+tree_maker'></span>

<h3>Description</h3>

<p>Reads report, converts to matrix of Sample ~ Plasmid with Sureness as cell values.
Performs a hierarchical cluster analysis on a set of dissimilarities derived from the matrix.
Creates a dendrogram from this data. Returns either the HC data or the dendrogram plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_maker(report, hc.only = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_maker_+3A_report">report</code></td>
<td>
<p>Dataframe of results produced
by <code><a href="#topic+subsampler">subsampler</a></code> or <code><a href="#topic+combine_results">combine_results</a></code></p>
</td></tr>
<tr><td><code id="tree_maker_+3A_hc.only">hc.only</code></td>
<td>
<p>Flag to return only hierarchical clustering
results instead of dendrogram plot (set to 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dendrogram object or hierarchical clustering results
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsampler">subsampler</a></code>, <code><a href="#topic+combine_results">combine_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tree_maker(report)

## End(Not run)
</code></pre>

<hr>
<h2 id='zetner_score'>Adds the Zetner Score column to report</h2><span id='topic+zetner_score'></span>

<h3>Description</h3>

<p>Runs mimmax function on Coverage and Divergence, returns sum of normalized Coverage with negative normalized Divergence
a value which is then normalized from 0 to 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zetner_score(report)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zetner_score_+3A_report">report</code></td>
<td>
<p>Dataframe of results produced by <code><a href="#topic+subsampler">subsampler</a></code> or <code><a href="#topic+combine_results">combine_results</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Report with zetner score added
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsampler">subsampler</a></code>, <code><a href="#topic+combine_results">combine_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
zetner_score(report)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
