<!DOCTYPE html><html><head><title>Help for package plasma</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {plasma}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CombinedWeights-class'><p>Class <code>"CombinedWeights"</code></p></a></li>
<li><a href='#Contribution-class'><p>Class <code>"Contribution"</code></p></a></li>
<li><a href='#esca-type-data'><p>ESCA type data</p></a></li>
<li><a href='#Imputation'><p>Imputation</p></a></li>
<li><a href='#MultiOmics-class'><p>Class <code>"MultiOmics"</code></p></a></li>
<li><a href='#MultiplePLSCoxModels-class'><p>Class <code>"MultiplePLSCoxModels"</code></p></a></li>
<li><a href='#plasma-class'><p>Class <code>"plasma"</code></p></a></li>
<li><a href='#plasmaPredictions-class'><p>Class <code>"plasmaPredictions"</code></p></a></li>
<li><a href='#SingleModel-class'><p>Class <code>"SingleModel"</code></p></a></li>
<li><a href='#TCGA-ESCA'>
<p>Esophageal carcinoma (ESCA) data or lung squamous cell carcinoma (LUSC)</p>
data from The Cancer Genome Atlas (TCGA).</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Partial LeAst Squares for Multiomic Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-03-10</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains tools for supervised analyses of incomplete, overlapping
  multiomics datasets. Applies partial least squares in multiple steps to find
  models that predict survival outcomes. See Yamaguchi et al. (2023)
  &lt;<a href="https://doi.org/10.1101%2F2023.03.10.532096">doi:10.1101/2023.03.10.532096</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, stats, graphics, survival, pls, plsRcox, Polychrome
(&ge; 1.5.0), viridisLite, beanplot, oompaBase</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.rsp, tidyr, ClassDiscovery</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-13 13:27:20 UTC; KRC</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin R. Coombes [cre, aut],
  Kyoko Yamaguchi [aut],
  Salma Abdelbaky [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-13 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CombinedWeights-class'>Class <code>"CombinedWeights"</code></h2><span id='topic+CombinedWeights'></span><span id='topic+CombinedWeights-class'></span><span id='topic+summary+2CCombinedWeights-method'></span><span id='topic+image+2CCombinedWeights-method'></span><span id='topic+combineAllWeights'></span><span id='topic+stdize'></span><span id='topic+interpret'></span>

<h3>Description</h3>

<p>The <code>CombinedWeights</code> object class merges the weight matrices for
all data sets in a plasma object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineAllWeights(pl)
## S4 method for signature 'CombinedWeights'
summary(object, ...)
## S4 method for signature 'CombinedWeights'
image(x, ...)
stdize(object, type = c("standard", "robust"))
interpret(object, component, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CombinedWeights-class_+3A_pl">pl</code></td>
<td>
<p>An object of the <code>plasma</code> class.</p>
</td></tr>
<tr><td><code id="CombinedWeights-class_+3A_object">object</code></td>
<td>
<p>An object of the <code>CombinedWeightss</code> class.</p>
</td></tr>
<tr><td><code id="CombinedWeights-class_+3A_x">x</code></td>
<td>
<p>An object of the <code>CombinedWeightss</code> class.</p>
</td></tr>
<tr><td><code id="CombinedWeights-class_+3A_type">type</code></td>
<td>
<p>A single character string indicating how to standardize
the object. Legal value are &quot;standard&quot; or &quot;robust&quot;.</p>
</td></tr>
<tr><td><code id="CombinedWeights-class_+3A_component">component</code></td>
<td>
<p>A single chaaracter string; which componen should be
interpreted.</p>
</td></tr>
<tr><td><code id="CombinedWeights-class_+3A_alpha">alpha</code></td>
<td>
<p>A single numerical value between 0 and 1; what
signfiicance value should be used to select important features.</p>
</td></tr>
<tr><td><code id="CombinedWeights-class_+3A_...">...</code></td>
<td>
<p>Ignored; potentially, extra arguments to the summary or
image methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>combineAllWeights</code> function returns a newly constructed object of the
<code>CombinedWeights</code> class. The summary method returna list
containing four matrices. Each matrix has one row for each omics data
set and one column for each model component. Each amtric contains
different summary statistics, including the Mean, SD, Median, and MAD.
</p>


<h3>Objects from the Class</h3>

<p>Objects are defined using the <code>combineAllWeights</code> functions.
Simply supply an object of class <code>plasma</code>. 
</p>


<h3>Slots</h3>


<dl>
<dt><code>combined</code>:</dt><dd><p>a matrix of the original variables in
dataset <code>N</code> as rows and the PLS components <code>M</code> as
columns.</p>
</dd>
<dt><code>featureSize</code>:</dt><dd><p>a numeric (usually integer) vector that
stores the number of features in each omics data set.</p>
</dd>
<dt><code>dataSource</code>:</dt><dd><p>a factor indicating which omics data set
each feature came from.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+summary">summary</a></code>:</dt><dd><p>outputs summary statistics for the contributions of dataset <code>N</code> to components from all datasets in the case of <code>getAllWeights</code> or dataset <code>M</code> in the case of <code>getCompositeWeights</code>.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
# restrict data set size
MO &lt;- with(plasmaEnv, prepareMultiOmics(
    assemble[c("ClinicalBin", "ClinicalCont", "RPPA")], Outcome))

splitVec &lt;- with(plasmaEnv, rbinom(nrow(Outcome), 1, 0.6))
trainD &lt;- MO[, splitVec == 1]
testD &lt;- MO[, splitVec == 0]

firstPass &lt;- fitCoxModels(trainD, "Days", "vital_status", "dead")
pl &lt;- plasma(object = trainD, multi = firstPass)

getCompositeWeights(object = pl, N = "ClinicalBin", M = "RPPA")

cbin &lt;- getAllWeights(object = pl, N = "ClinicalBin")
summary(cbin)
image(cbin)
heat(cbin, cexCol = 0.5)

cbin01 &lt;- pickSignificant(object = cbin, alpha = 0.01)
image(cbin01)
heat(cbin01, cexCol = 0.5)

getTop(object = cbin01, N = 3)
</code></pre>

<hr>
<h2 id='Contribution-class'>Class <code>"Contribution"</code></h2><span id='topic+Contribution'></span><span id='topic+Contribution-class'></span><span id='topic++5B+2CContribution+2CANY+2CANY+2CANY-method'></span><span id='topic+summary+2CContribution-method'></span><span id='topic+image+2CContribution-method'></span><span id='topic+heat'></span><span id='topic+heat+2CContribution-method'></span><span id='topic+getCompositeWeights'></span><span id='topic+getAllWeights'></span><span id='topic+getFinalWeights'></span><span id='topic+getTop'></span><span id='topic+pickSignificant'></span>

<h3>Description</h3>

<p>The <code>Contribution</code> object class contains the weight matrix between variables and the PLS components. The values in the weight matrix are a numeric representation of how much a variable from the omics datasets contributed to defining the final PLS components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCompositeWeights(object, N, M)
getAllWeights(object, N)
getFinalWeights(object)
getTop(object, N = 1)
pickSignificant(object, alpha)
## S4 method for signature 'Contribution'
summary(object, ...)
## S4 method for signature 'Contribution'
image(x, col = viridis(64), mai = c(1.82, 1.52, 0.32, 0.32), ...)
## S4 method for signature 'Contribution'
heat(object, main = "Contributions", col = viridis(64),
                                           mai = c(1.52, 0.32, 0.82, 1.82), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Contribution-class_+3A_object">object</code></td>
<td>
<p>In the first four functions, an object of the
<code>plasma</code> class. In the methods described here, an object
of the <code>Contributions</code> class.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_n">N</code></td>
<td>
<p>in the function <code>getCompositeWeights</code>, the name of the dataset being modeled. <br /> in the function <code>getTop</code>, the number of significant components you want to print.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_m">M</code></td>
<td>
<p>name of the dataset being modeled pairwise with dataset <code>N</code> in the <code>getCompositeWeights</code> function.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_alpha">alpha</code></td>
<td>
<p>level of significance used in the <code>pickSignificant</code> function.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_...">...</code></td>
<td>
<p>other graphical parameters.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_x">x</code></td>
<td>
<p>an object of the <code>Contributions</code> class.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_main">main</code></td>
<td>
<p>A character vector of length one; the main plot title.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_col">col</code></td>
<td>
<p>A vector of color descriptors.</p>
</td></tr>
<tr><td><code id="Contribution-class_+3A_mai">mai</code></td>
<td>
<p>A vector of four nonnegative numbers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>plasma</code> function returns a newly constructed object of the
<code>plasma</code> class. 
</p>


<h3>Objects from the Class</h3>

<p>Objects are defined using the <code>getAllWeights</code>, <code>getCompositeWeights</code>, <code>getTop</code>, or <code>pickSignificant</code> functions. In the simplest scenario, one would enter an object of class <code>plasma</code> and any specific parameters associated with the function (see arguments section for more info).
</p>


<h3>Slots</h3>


<dl>
<dt><code>contrib</code>:</dt><dd><p>a matrix of the original variables in dataset <code>N</code> as rows and the PLS components <code>M</code> as columns.</p>
</dd>
<dt><code>datasets</code>:</dt><dd><p>a character vector that stores the names of the datasets that were specified for the function.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+summary">summary</a></code>:</dt><dd><p>outputs summary statistics for the contributions of dataset <code>N</code> to components from all datasets in the case of <code>getAllWeights</code> or dataset <code>M</code> in the case of <code>getCompositeWeights</code>.</p>
</dd>
<dt><code><a href="graphics.html#topic+image">image</a></code>:</dt><dd><p>outputs a heatmap of the transposed <code>contrib</code> matrix.</p>
</dd>
<dt><code>heat</code>:</dt><dd><p>outputs a clustered heatmap of the <code>contrib</code> matrix.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
# restrict data set size
MO &lt;- with(plasmaEnv, prepareMultiOmics(
    assemble[c("ClinicalBin", "ClinicalCont", "RPPA")], Outcome))

splitVec &lt;- with(plasmaEnv, rbinom(nrow(Outcome), 1, 0.6))
trainD &lt;- MO[, splitVec == 1]
testD &lt;- MO[, splitVec == 0]

firstPass &lt;- fitCoxModels(trainD, "Days", "vital_status", "dead")
pl &lt;- plasma(object = trainD, multi = firstPass)

getCompositeWeights(object = pl, N = "ClinicalBin", M = "RPPA")

cbin &lt;- getAllWeights(object = pl, N = "ClinicalBin")
summary(cbin)
image(cbin)
heat(cbin, cexCol = 0.5)

cbin01 &lt;- pickSignificant(object = cbin, alpha = 0.01)
image(cbin01)
heat(cbin01, cexCol = 0.5)

getTop(object = cbin01, N = 3)
</code></pre>

<hr>
<h2 id='esca-type-data'>ESCA type data</h2><span id='topic+tfESCA'></span><span id='topic+mirESCA'></span>

<h3>Description</h3>

<p>The <code>CombinedWeights</code> object class merges the weight matrices for
all data sets in a plasma object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tfESCA)
data(mirESCA)
</code></pre>


<h3>Format</h3>

<p>Both <code>tfData</code> and <code>mirESCA</code> are data frames containng two
columns. The first column is and <code>ID</code> column containing the TCGA
sample barcode for an esophagela cancer sample. The second column,
called <code>Type</code> identifies the sample as either &quot;squamous&quot; (for
likely squamous cell carcinomas that cluster near head and neck
cancers) or &quot;adeno&quot; (for likley adenocarcinomas that cluster near
stomach cancers).
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Source</h3>

<p>All data supplied here are based upon esophageal cancer data generated
by the TCGA Research Network (https://www.cancer.gov/tcga).
</p>
<p>The transcription factor classifications of 196 esophageal cancer
into squamous cell carcinoma or adenocarcinoma are taken from work
published by Abrams and colleaagues in BMC Genomics.
</p>
<p>The microRNA classifications of 195 esophageal cancer samples
into squamous cell carcinoma or adenocarcinoma are taken from work
published by Asiaee and colleaagues in J Comput Biol.
</p>


<h3>References</h3>

<p>Abrams ZB, Zucker M, Wang M, Asiaee Taheri A, Abruzzo LV, Coombes KR.<br />
<em>Thirty biologically interpretable clusters of transcription
factors distinguish cancer type.</em><br />
BMC Genomics. 2018 Oct 11;19(1):738. doi: 10.1186/s12864-018-5093-z.
</p>
<p>Asiaee A, Abrams ZB, Nakayiza S, Sampath D, Coombes KR.<br />
<em>Explaining Gene Expression Using Twenty-One MicroRNAs.</em><br />
J Comput Biol. 2020 Jul;27(7):1157-1170. doi: 10.1089/cmb.2019.0321.
</p>

<hr>
<h2 id='Imputation'>Imputation</h2><span id='topic+meanModeImputer'></span><span id='topic+samplingImputer'></span>

<h3>Description</h3>

<p>Functions to impute missing data in omics data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meanModeImputer(X)
samplingImputer(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Imputation_+3A_x">X</code></td>
<td>
<p>A numeric matrix, where the columns represent independent
observations (patients or samples) and the columns represent measured
features (genes, proteins, clinical variables, etc).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We recommend imputing small amounts of missing data in the input data
sets when using the <code>plasma</code> package. The underlying issue is
that the PLS models we use for individual omics data sets will not be
able to make predictions on a sample if even one data point is
missing. As a result, if a sample is missing at least one data point in
every omics data set, then it will be impossible to use that sample at
all.
</p>
<p>For a range of available imputation methods and R packages, consult
the <a href="https://CRAN.R-project.org/view=MissingData">CRAN Task
View on Missing Data</a>.  We also recommend the
<a href="https://rmisstastic.netlify.app/">R-miss-tastic web site on
missing data</a>.  Their simulations suggest that, for purposes of
producing predictive models from omics data, the imputation method is
not particularly important. Because of the latter finding, we have
only implemented two simple imputation methods in the <code>plasma</code>
package:
</p>

<ol>
<li><p>  The <code>meanModeImputer</code> function will replace any missing data by the
mean value of the observed data if there are more than five
distinct values; otherwise, it will replace missing data by the
mode.  This approach works relatively well for both continuous
data and for binary or small categorical data.
</p>
</li>
<li><p> The <code>samplingImpute</code> function replaces missing values by sampling
randomly from the observed data distribution.
</p>
</li></ol>



<h3>Value</h3>

<p>Both functions return a numeric matrix of the same size and with the
same row and column names as the input variable
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>loadESCAdata()
imputed &lt;- with(plasmaEnv, lapply(assemble, samplingImputer) )
imputed &lt;- with(plasmaEnv, lapply(assemble, meanModeImputer))
</code></pre>

<hr>
<h2 id='MultiOmics-class'>Class <code>"MultiOmics"</code></h2><span id='topic+MultiOmics'></span><span id='topic+MultiOmics-class'></span><span id='topic+summary+2CMultiOmics-method'></span><span id='topic++5B+2CMultiOmics+2CANY+2CANY+2CANY-method'></span><span id='topic+plot+2CMultiOmics+2Cmissing-method'></span><span id='topic+validMultiOmics'></span><span id='topic+prepareMultiOmics'></span>

<h3>Description</h3>

<p>The <code>prepareMultiOmics</code> function returns a new object of <code>MultiOmics</code> class for use in <code>fitCoxModel</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepareMultiOmics(datalist, outcome)
## S4 method for signature 'MultiOmics'
summary(object, ...)
## S4 method for signature 'MultiOmics,missing'
plot(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiOmics-class_+3A_datalist">datalist</code></td>
<td>
<p>a list of dataframes formatted to have variables as rows (dimension D) and samples as columns (dimension N).</p>
</td></tr>
<tr><td><code id="MultiOmics-class_+3A_outcome">outcome</code></td>
<td>
<p>a dataframe of clinical outcomes formatted to have
sample names as row indexes and variable names as column indexes</p>
</td></tr>
<tr><td><code id="MultiOmics-class_+3A_object">object</code></td>
<td>
<p>An object of the <code>MultiOmics</code> class.</p>
</td></tr>
<tr><td><code id="MultiOmics-class_+3A_x">x</code></td>
<td>
<p>An object of the <code>MultiOmics</code> class.</p>
</td></tr>
<tr><td><code id="MultiOmics-class_+3A_y">y</code></td>
<td>
<p>Nothing; ignored.</p>
</td></tr>
<tr><td><code id="MultiOmics-class_+3A_...">...</code></td>
<td>
<p>Extra graphical or other parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>prepareMultiOmics</code> function returns a new object of the <code>MultiOmics</code> class.
</p>


<h3>Objects from the Class</h3>

<p>Objects should be defined using the <code>prepareMultiOmics</code> constructor. In
the simplest case, you enter two objects: a list of dataframes and a dataframe of clinical outcomes.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code>:</dt><dd><p>A list of dataframes with variables as rows or varying length and samples as columns of uniform length N, where N is the maximum value of non-missing samples in any given dataset. Note that <code>NA</code>s have been added to &ldquo;pad&rdquo; to make the column length uniform across data types.</p>
</dd>
<dt><code>outcome</code>:</dt><dd><p>A dataframe of clinical outcomes with variables as columns and samples as rows.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+plot">plot</a></code>:</dt><dd><p>Produces a visual representation of the  dimensionalities of each dataframe in datalist. D corresponds to the number of variables in each omics dataframe, and N corresponds to samples (or members) whose variable is not entirely missing. Gray areas correspond to missing samples.</p>
</dd>
<dt><code><a href="base.html#topic+summary">summary</a></code>:</dt><dd><p>Produces summary tables corresponding to datasets and outcomes.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
MO &lt;- with(plasmaEnv,
           prepareMultiOmics(datalist = assemble, outcome = Outcome))
plot(MO)
summary(MO)
</code></pre>

<hr>
<h2 id='MultiplePLSCoxModels-class'>Class <code>"MultiplePLSCoxModels"</code></h2><span id='topic+MultiplePLSCoxModels'></span><span id='topic+MultiplePLSCoxModels-class'></span><span id='topic+summary+2CMultiplePLSCoxModels-method'></span><span id='topic++5B+2CMultiplePLSCoxModels+2CANY+2CANY+2CANY-method'></span><span id='topic+plot+2CMultiplePLSCoxModels+2Cmissing-method'></span><span id='topic+predict+2CMultiplePLSCoxModels-method'></span><span id='topic+validMultipleCoxModels'></span><span id='topic+fitCoxModels'></span><span id='topic+getSizes'></span>

<h3>Description</h3>

<p>The <code>MultiplePLSCoxModels</code> object class ...
The <code>validMultipleCoxModels</code> function checks if each data set contains the same set of samples.
The <code>fitCoxModels</code> function fits many plsRcoxmodels and returns an S4 object of class <code>MultiplePLSCoxModels</code>.
The <code>getSizes</code> function returns a matrix with the list of dataframes of the <code>MultiOmics</code> object as rownames and columns with NT, cNT, and p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitCoxModels(multi, timevar, eventvar, eventvalue, verbose)
## S4 method for signature 'MultiplePLSCoxModels'
summary(object, ...)
## S4 method for signature 'MultiplePLSCoxModels,missing'
plot(x, y, col = c("blue", "red"),
      lwd = 2, xlab = "", ylab = "Fraction Surviving",
       mark.time = TRUE, legloc = "topright", ...)
## S4 method for signature 'MultiplePLSCoxModels'
predict(object, newdata, type = c("components", "risk",
                 "split", "survfit"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_multi">multi</code></td>
<td>
<p>an object of class <code>MultiOmics</code> for fitting the model.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_timevar">timevar</code></td>
<td>
<p>a column in the <code>MultiOmics</code> object in the
<code>outcome</code> dataframe containing the time-to-event.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_eventvar">eventvar</code></td>
<td>
<p>a column in the <code>MultiOmics</code> object in the
<code>outcome</code> dataframe containing the event.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_eventvalue">eventvalue</code></td>
<td>
<p>a character string specifying the value of the event in <code>eventvar</code>.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_verbose">verbose</code></td>
<td>
<p>logical; should the function report progress.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_object">object</code></td>
<td>
<p>an object of class <code>MultiplePLSCoxModels</code> for outputting the summary.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_x">x</code></td>
<td>
<p>an object of class <code>MultiplePLSCoxModels</code> for plotting the Kaplan-Meier curves.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_y">y</code></td>
<td>
<p>An ignored argrument for the plot method.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_col">col</code></td>
<td>
<p>A vector of color specifications. Default is c(&ldquo;blue&rdquo;, &ldquo;red&rdquo;).</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_lwd">lwd</code></td>
<td>
<p>A vector specifying the line width. Default is &ldquo;2&rdquo;.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_xlab">xlab</code></td>
<td>
<p>A character string to label the x-axis. Default is &ldquo;&rdquo;.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_ylab">ylab</code></td>
<td>
<p>A character string to label the y-axis. Default is &ldquo;Fraction Surviving&rdquo;.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_mark.time">mark.time</code></td>
<td>
<p>A logical value; should tickmarks indicate censored
data? Default is TRUE.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_legloc">legloc</code></td>
<td>
<p>A character string indicating where to put the legend. Default is &ldquo;topright&rdquo;.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_...">...</code></td>
<td>
<p>Other graphical parameters.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_newdata">newdata</code></td>
<td>
<p>A <code>MultiOmics</code> object with the same structure as
the training data.</p>
</td></tr>
<tr><td><code id="MultiplePLSCoxModels-class_+3A_type">type</code></td>
<td>
<p>An enumerated character value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>fitCoxModels</code> function retuns a newly constructed object of
the <code>MultiplePLSCoxModels</code> class. The <code>plot</code> method
invisibly returns the object on which it was invoked. The
<code>summary</code> method returns no value. The predict method returns a
list of prediction results, each of which comes from the
<code>predict</code> method for the <code><a href="#topic+SingleModel-class">SingleModel-class</a>.</code>
</p>


<h3>Slots</h3>


<dl>
<dt><code>models</code>:</dt><dd><p>A list of <code>SingleModel</code> objects, one for each assay.</p>
</dd>
<dt><code>timevar</code>:</dt><dd><p>A character matching the name of the column containing the time-to-event.</p>
</dd>
<dt><code>eventvar</code>:</dt><dd><p>A character matching the name of the column containing the event.</p>
</dd>
<dt><code>eventvalue</code>:</dt><dd><p>A character specifying the event in <code>eventvar.</code> </p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+plot">plot</a></code>:</dt><dd><p>Plots Kaplan-Meier curves for each
omics dataset split into Low Risk and High Risk groups.</p>
</dd>
<dt><code><a href="base.html#topic+summary">summary</a></code>:</dt><dd><p>Returns a description of the
<code>MultiplePLSCoxModels</code> object and the names of the omics
datasets used to build the model.</p>
</dd>
<dt><code><a href="stats.html#topic+predict">predict</a></code>:</dt><dd><p>usually returns a list of numeric
vectors of predicted risk per data type. When <code>type = "survfit"</code>,
retuns a list of <code>survfit</code> objects.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>See Also</h3>

<p>fitSingleModel</p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
# restrict data set size
MO &lt;- with(plasmaEnv, prepareMultiOmics(
   assemble[c("ClinicalBin", "ClinicalCont", "RPPA")], Outcome))

splitVec &lt;- with(plasmaEnv, rbinom(nrow(Outcome), 1, 0.6))
trainD &lt;- MO[, splitVec == 1]
testD &lt;- MO[, splitVec == 0]

firstPass &lt;- fitCoxModels(trainD, "Days", "vital_status", "dead")

summary(firstPass)
plot(firstPass)
getSizes(firstPass)
pre1 &lt;- predict(firstPass, testD)
</code></pre>

<hr>
<h2 id='plasma-class'>Class <code>"plasma"</code></h2><span id='topic+plasma-class'></span><span id='topic+plot+2Cplasma+2Cmissing-method'></span><span id='topic+barplot+2Cplasma-method'></span><span id='topic+summary+2Cplasma-method'></span><span id='topic+predict+2Cplasma-method'></span><span id='topic+plasma'></span>

<h3>Description</h3>

<p>The <code>plasma</code> object class is returned after running the <code>plasma</code> function.
The <code>plasma</code> function uses the <code>PLSRCox</code> components from one
dataset as the predictor variables and the <code>PLSRCox</code> components
of another dataset as the response variables to fit a partial least
squares regression (<code>plsr</code>) model. Then, we take the mean of the
predictions to create a final matrix of samples versus components.
</p>
<p>The matrix of components described earlier is then used to fit a Cox
Proportional Hazards (<code>coxph</code>) model with AIC stepwise variable
selection to return a final object of class <code>plasma</code> which
includes a <code>coxph</code> model with a reduced number of predictors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plasma(object, multi)
## S4 method for signature 'plasma,missing'
plot(x, y, ...)
## S4 method for signature 'plasma'
barplot(height, source, n, direction = c("both", "up","down"),
                lhcol = c("cyan", "red"), ...)
## S4 method for signature 'plasma'
predict(object, newdata = NULL, type = c("components", "risk",
                 "split"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plasma-class_+3A_multi">multi</code></td>
<td>
<p>an object of the <code>MultiplePLSCoxModels</code> class.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_object">object</code></td>
<td>
<p>an object of the <code>plasma</code> class.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_height">height</code></td>
<td>
<p>an object of the <code>plasma</code> class for the <code>barplot</code> method.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_x">x</code></td>
<td>
<p>an object of class <code>plasma</code> for plotting the Kaplan-Meier curves.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_y">y</code></td>
<td>
<p>An ignored argrument for the plot method.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_source">source</code></td>
<td>
<p>A length-one character vector; the name of a data set in
a <code>plasma</code> object.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_n">n</code></td>
<td>
<p>A length-one integer vector; the number of high-weight
features to display.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_direction">direction</code></td>
<td>
<p>A length-one character vector; show features with
positive weights (up), negative (down), or both.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_lhcol">lhcol</code></td>
<td>
<p>A chaacter vector of length 2, indicating the preferred
colors for low (negative) or high (positive) weights.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_newdata">newdata</code></td>
<td>
<p>A <code>MultiOmics</code> object with the same structure as
the training data.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_type">type</code></td>
<td>
<p>An enumerated character value.</p>
</td></tr>
<tr><td><code id="plasma-class_+3A_...">...</code></td>
<td>
<p>Additional graphical parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>plasma</code> function returns a newly constructed object of the <code>plasma</code> class. The <code>plot</code> method invisibly returns the object on which it was invoked. The <code>predict</code> method returns an object of the <code><a href="#topic+plasmaPredictions">plasmaPredictions</a></code> class.
</p>


<h3>Objects from the Class</h3>

<p>Objects should be defined using the <code>plasma</code> function.
</p>


<h3>Slots</h3>


<dl>
<dt><code>traindata</code>:</dt><dd><p>An object of class <code>MultiOmics</code> used for training the model.</p>
</dd>
<dt><code>compModels</code>:</dt><dd><p>A list containing objects in the form of <code>plsr</code>.</p>
</dd>
<dt><code>fullModel</code>:</dt><dd><p>A coxph object with variables (components) selected via AIC stepwise selection.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+plot">plot</a></code>:</dt><dd><p>Plots a Kaplan-Meier curve of the final <code>coxph</code> model that has been categorized into &ldquo;low risk&rdquo; and &ldquo;high risk&rdquo; based whether it is higher or lower, respectively, than the median value of risk.</p>
</dd>
<dt><code><a href="stats.html#topic+predict">predict</a></code>:</dt><dd><p>creates an object of class <code>plasmaPredictions</code>.</p>
</dd>
<dt><code><a href="graphics.html#topic+barplot">barplot</a></code>:</dt><dd><p>Produces a barplot of the <code>n</code> largest
weights assigned to features from the appropriate data <code>source</code>.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>See Also</h3>

<p>plasmaPredictions, plsr</p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
# restrict data set size
MO &lt;- with(plasmaEnv, prepareMultiOmics(
    assemble[c("ClinicalBin", "ClinicalCont", "RPPA")], Outcome) )

splitVec &lt;- with(plasmaEnv, rbinom(nrow(Outcome), 1, 0.6))
trainD &lt;- MO[, splitVec == 1]
testD &lt;- MO[, splitVec == 0]

firstPass &lt;- fitCoxModels(trainD, "Days", "vital_status", "dead")
pl &lt;- plasma(object = trainD, multi = firstPass)

plot(pl, legloc = "topright", main = "Training Data")
barplot(pl, "RPPA", 6)
barplot(pl, "RPPA", 10, "up")
</code></pre>

<hr>
<h2 id='plasmaPredictions-class'>Class <code>"plasmaPredictions"</code></h2><span id='topic+plasmaPredictions'></span><span id='topic+plasmaPredictions-class'></span><span id='topic+plot+2CplasmaPredictions+2Cmissing-method'></span>

<h3>Description</h3>

<p>The <code>plasmaPredictions</code> object class is returned when running the
<code>predict</code> method on an object of class <code>plasma</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'plasmaPredictions,missing'
plot(x, y, col = c("blue", "red"),
                 lwd = 2, xlab = "", ylab = "Fraction Surviving",
                 mark.time = TRUE, legloc = "topright", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plasmaPredictions-class_+3A_x">x</code></td>
<td>
<p>An object of the <code>plasmaPredictions</code> class for plotting the Kaplan-Meier curves.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_y">y</code></td>
<td>
<p>An ignored argument for the plot method.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_col">col</code></td>
<td>
<p>A vector of color specifications. Default is c(&ldquo;blue&rdquo;, &ldquo;red&rdquo;).</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_lwd">lwd</code></td>
<td>
<p>A vactor specifying the line width. Default is &ldquo;2&rdquo;.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_xlab">xlab</code></td>
<td>
<p>A character string to label the x-axis. Default is &ldquo;&rdquo;.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_ylab">ylab</code></td>
<td>
<p>A character string to label the y-axis. Default is &ldquo;Fraction Surviving&rdquo;.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_mark.time">mark.time</code></td>
<td>
<p>A logical value; should tickmarks indicate censored
data? Default is TRUE.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_legloc">legloc</code></td>
<td>
<p>A character string indicating where to put the legend. Default is &ldquo;topright&rdquo;.</p>
</td></tr>
<tr><td><code id="plasmaPredictions-class_+3A_...">...</code></td>
<td>
<p>Other graphical parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>predict</code> method on an object of the <code><a href="#topic+plasma">plasma</a></code>
class returns an object of the <code><a href="#topic+plasmaPredictions">plasmaPredictions</a></code>
class. The <code>plot</code> method invisibly returns the value on which it
was invoked.
</p>


<h3>Objects from the Class</h3>

<p>Users shold not create objects of this class directly. They will be
automatically created when you apply the <code>predict</code> method to a
fully worked out <code>plasma</code> model.
</p>


<h3>Slots</h3>


<dl>
<dt><code>meanPredictions</code>:</dt><dd><p>A matrix with samples as rows and
factors as columns that is a result of taking the mean of the
PLS component predictions from each dataset.</p>
</dd>
<dt><code>riskDF</code>:</dt><dd><p>Object of type <code>data.frame</code> containing
the original <code>outcome</code> dataframe and additional columns for
&quot;Risk&quot;, and &quot;Split&quot;, corresponding to the risk of the event
calculated by the model, and patient assignment to low versus
high-risk groups, respectively.</p>
</dd>
<dt><code>riskModel</code>:</dt><dd><p>Object of type <code>coxph</code> that uses
predicted Risk (continuous) as the predictor variable and
survival as the response variable. See documentation for
<code>link{coxph}</code>.</p>
</dd> 
<dt><code>splitModel</code>:</dt><dd><p>Object of type <code>coxph</code> that uses
predicted Split (predicted Risk categorized into &ldquo;high&rdquo;
and &ldquo;low&rdquo; risk by the median predicted Risk) as the
predictor variable and survival as the response variable. See
documentation for <code>link{coxph}</code>.</p>
</dd> 
<dt><code>SF</code>:</dt><dd><p>Object of type <code>survfit</code> which is used by
the <code>plot</code> method to plot Kaplan-Meier curves grouped by
predicted Split. See documentation for <code>link{survfit}</code>.</p>
</dd> 
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+plot">plot</a></code>:</dt><dd><p>Produces Kaplan-Meier curves for the
low risk and high risk groups.</p>
</dd>
</dl>



<h3>Note</h3>

<p>An object of <code>plasmaPredictions</code> class contains many models that
are similar to an object of <code><a href="#topic+MultiplePLSCoxModels">MultiplePLSCoxModels</a></code> class.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>See Also</h3>

<p>plasma</p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
# restrict data set size
MO &lt;- with(plasmaEnv, prepareMultiOmics(
    assemble[c("ClinicalBin", "ClinicalCont", "RPPA")], Outcome))

splitVec &lt;- with(plasmaEnv, rbinom(nrow(Outcome), 1, 0.6))
trainD &lt;- MO[, splitVec == 1]
testD &lt;- MO[, splitVec == 0]

firstPass &lt;- fitCoxModels(trainD, "Days", "vital_status", "dead")
pl &lt;- plasma(object = trainD, multi = firstPass)

testpred &lt;- predict(pl, testD)
plot(testpred, main = "Testing", xlab = "Time (Days)")
</code></pre>

<hr>
<h2 id='SingleModel-class'>Class <code>"SingleModel"</code></h2><span id='topic+fitSingleModel'></span><span id='topic+SingleModel-class'></span><span id='topic+predict+2CSingleModel-method'></span><span id='topic+summary+2CSingleModel-method'></span><span id='topic+plot+2CSingleModel+2Cmissing-method'></span><span id='topic+fitSingleModel'></span>

<h3>Description</h3>

<p>The <code>fitSingleModel</code> function takes in an object of
<code>MultiOmics</code> class and returns a new object of
<code>SingleModel</code> class. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitSingleModel(multi, N, timevar, eventvar, eventvalue)
## S4 method for signature 'SingleModel'
summary(object, ...)
## S4 method for signature 'SingleModel,missing'
plot(x, y, col = c("blue", "red"),
      lwd = 2, xlab = "", ylab = "Fraction Surviving",
       mark.time = TRUE, legloc = "topright", ...)
## S4 method for signature 'SingleModel'
predict(object, newdata, type = c("components", "risk",
                 "split", "survfit"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SingleModel-class_+3A_multi">multi</code></td>
<td>
<p>an object of class <code>MultiOmics</code> for fitting the model.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_n">N</code></td>
<td>
<p>A character string identifying the data set being modeled.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_timevar">timevar</code></td>
<td>
<p>a column in the <code>MultiOmics</code> object in the
<code>outcome</code> dataframe containing the time-to-event.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_eventvar">eventvar</code></td>
<td>
<p>a column in the <code>MultiOmics</code> object in the
<code>outcome</code> dataframe containing the event.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_eventvalue">eventvalue</code></td>
<td>
<p>a character string specifying the value of the event.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_x">x</code></td>
<td>
<p>an object of class <code>plsRcoxmodel</code> for plotting the Kaplan-Meier curves.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_y">y</code></td>
<td>
<p>An ignored argrument for the plot method.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_col">col</code></td>
<td>
<p>A vector of color specifications.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_lwd">lwd</code></td>
<td>
<p>A vactor specifying the line width.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_xlab">xlab</code></td>
<td>
<p>A character string to label the x-axis.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_ylab">ylab</code></td>
<td>
<p>A character string to label the y-axis.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_mark.time">mark.time</code></td>
<td>
<p>A logical value; should tickmarks indicate censored
data?</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_legloc">legloc</code></td>
<td>
<p>A character string indicating where to put the legend.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_object">object</code></td>
<td>
<p>an object of class <code>SingleModel</code>.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_newdata">newdata</code></td>
<td>
<p>A <code>MultiOmics</code> object with the same structure as
the training data.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_type">type</code></td>
<td>
<p>An enumerated character value.</p>
</td></tr>
<tr><td><code id="SingleModel-class_+3A_...">...</code></td>
<td>
<p>other parameters used in graphing or prediction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>fitSingleModel</code> function returns a newly constructed object
of the <code>SingleModel</code> class. The <code>plot</code> method invisibly
returns the value on which it was invoked. The summary method
returns an object summarizing the final model produced by PLS R cox
regression. The <code>predict</code> method returns either a vector or
matrix depending on the type of predictions requested.
</p>


<h3>Slots</h3>


<dl>
<dt><code>plsmod</code>:</dt><dd><p>Object of class <code>plsRcoxmodel</code>
containing the fitted model.</p>
</dd>
<dt><code>Xout</code>:</dt><dd><p>Object of type <code>data.frame</code> containing
the original <code>outcome</code> dataframe and additional columns for
&quot;Risk&quot;, and &quot;Split&quot;, corresponding to the risk of the event
calculated by the model, and patient assignment to low versus
high-risk groups, respectively.</p>
</dd>
<dt><code>dsname</code>:</dt><dd><p>A character vector of length one; the name
of the data set being modeled from a <code>MultiOmics</code> object.</p>
</dd>
<dt><code>SF</code>:</dt><dd><p>Object of type <code>survfit</code> which is used by the <code>plot</code> method to plot Kaplan-Meier curves grouped by predicted Split. See documentation for <code>link{survfit}</code>.</p>
</dd>
<dt><code>riskModel</code>:</dt><dd><p>Object of type <code>coxph</code> that uses predicted Risk (continuous) as the predictor variable and survival as the response variable. See documentation for <code>link{coxph}</code>.</p>
</dd>
<dt><code>splitModel</code>:</dt><dd><p>Object of type <code>coxph</code> that uses predicted Split (predicted Risk categorized into &ldquo;high&rdquo; and &ldquo;low&rdquo; risk by the median predicted Risk) as the predictor variable and survival as the response variable. See documentation for <code>link{coxph}</code>.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code><a href="base.html#topic+plot">plot</a></code>:</dt><dd><p>Plots Kaplan-Meier curves for each
omics dataset split into Low Risk and High Risk groups.</p>
</dd>
<dt><code><a href="base.html#topic+summary">summary</a></code>:</dt><dd><p>Returns a description of the
<code>MultiplePLSCoxModels</code> object and the names of the omics
datasets used to build the model.</p>
</dd>
<dt><code><a href="stats.html#topic+predict">predict</a></code>:</dt><dd><p>Usually, a numeric vector containing
the predicted risk values. However, when using <code>type =
	"survfit"</code>, tghe return value is a <code>survfit</code> object from
the<code>survival</code> package.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+getSizes">getSizes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
MO &lt;- with(plasmaEnv, prepareMultiOmics(assemble, Outcome) )
MO &lt;- MO[c("ClinicalBin", "ClinicalCont", "RPPA"),]
set.seed(98765)
splitVec &lt;- with(plasmaEnv, rbinom(nrow(Outcome), 1, 0.6))
trainD &lt;- MO[, splitVec == 1]
testD &lt;- MO[, splitVec == 0]

zerothPass &lt;- fitSingleModel(trainD, N = "RPPA",
                           timevar = "Days", eventvar = "vital_status",
                           eventvalue = "dead")
summary(zerothPass)
plot(zerothPass)
pre0 &lt;- predict(zerothPass, testD)
</code></pre>

<hr>
<h2 id='TCGA-ESCA'>
Esophageal carcinoma (ESCA) data or lung squamous cell carcinoma (LUSC)
data from The Cancer Genome Atlas (TCGA).
</h2><span id='topic+TCGA-ESCA'></span><span id='topic+loadESCAdata'></span><span id='topic+TCGA-LUSC'></span><span id='topic+loadLUSCdata'></span><span id='topic+plasmaEnv'></span><span id='topic+assemble'></span><span id='topic+Outcome'></span><span id='topic+m450info'></span>

<h3>Description</h3>

<p>The <code>TCGA-ESCA</code> dataset contains the objects <code>assemble</code>,
<code>Outcome</code>, and <code>m450info</code> for building the <code>MultiOmics</code>
object. Because its size exceeds the CRAN limits, the data is stored on
a remote server and must be loaded using the function
<code>loadESCAdata</code>.
</p>
<p>The <code>TCGA-LUSC1</code>dataset is a parallel object for lung
squamous cell carcinoma (LUSC) data, whihc must be loaded using the
<code>loadLUSCdata</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadESCAdata(env = plasmaEnv)
loadLUSCdata(env = plasmaEnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TCGA-ESCA_+3A_env">env</code></td>
<td>
<p>an environment in which to load the data. The default
value is a private environment in the package, accessible as
<code>plasmaEnv</code>. To make access easier, you can use
<code>globalenv()</code> or <code>.GlobalEnv</code>.</p>
</td></tr>
</table>


<h3>Format</h3>

<p>The &ldquo;TCGA-ESCA&rdquo; dataset contains the following:
</p>

<dl>
<dt><code>assemble</code></dt><dd><p>A list of 7 different omics dataframes with
varying numbers of features as rows (D) and varying number of
patients as columns (N). Note that some of these omics dataframes
had been manipulated to contain NAs, where these may be complete
on the GDC Dat Portal from which these data originally came. This
was done to illustrate the capability of the <code>plasma</code> package
on working with missing data.</p>
</dd>
</dl>

<ol>
<li><p><code>ClinicalBin</code>a dataframe (53x185) of clinical binary values.
</p>
</li>
<li><p><code>ClinicalCont</code>a dataframe (6x185) of clinical continuous values.
</p>
</li>
<li><p><code>MAF</code>a dataframe (566x184) of minor allele frequencies
(MAF) that have been converted to binary based on whether they had
a MAF greater than 0.03 (1) or not (0).
</p>
</li>
<li><p><code>Meth450</code>a dataframe (1454x185) of continuous beta
values from the Illumina Infinium HumanMethylation450 arrays. The
features in this dataframe have been filtered on mean greater than
0.15 and a standard deviation greater than 0.3.
</p>
</li>
<li><p><code>miRSeq</code>a dataframe (926x166) of continuous counts
values from microRNA (miRNA) sequencing. The features in this
dataframe have been filtered on a standard deviation of 0.05.
</p>
</li>
<li><p><code>mRNASeq</code>a dataframe (2520x157) of continuous counts
values from mRNA sequencing data. The features in this dataframe
have been filtered on a mean greater than 4 and a standard
deviation greater than 0.7.
</p>
</li>
<li><p><code>RPPA</code>a dataframe (192x126) of continuous protein
expression values from reverse phase protein array (RPPA) assays.
</p>
</li></ol>

<dl>
<dt><code>Outcome</code></dt><dd><p>a dataframe (185x5) containing the survival
outcomes for the patients in <code>assemble</code>.</p>
</dd>
<dt><code>m450info</code></dt><dd><p>a dataframe (1454x3) containing gene symbol,
chromosome number, and genomic coordinate IDs corresponding to the
features (or &ldquo;probes&rdquo;) in <code>Meth450</code>.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Source</h3>

<p><a href="https://portal.gdc.cancer.gov/projects/TCGA-ESCA">https://portal.gdc.cancer.gov/projects/TCGA-ESCA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
