<!DOCTYPE html><html><head><title>Help for package pleio</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pleio}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pleio.demo'>
<p>Demonstration dataset for pleiotropy tests with a mixture of trait</p>
types and covariates</a></li>
<li><a href='#pleio.glm.fit'>
<p>Compute Intermediate Statisics for Testing Pleiotropy</p></a></li>
<li><a href='#pleio.glm.sequential'>
<p>Perform sequential tests of pleiotropy</p></a></li>
<li><a href='#pleio.glm.test'>
<p>Single test of the number of traits associated with genotype</p></a></li>
<li><a href='#pleio.q.fit'>
<p>Compute Intermediate Statisics for Testing Pleiotropy</p></a></li>
<li><a href='#pleio.q.sequential'>
<p>Perform sequential tests of pleiotropy</p></a></li>
<li><a href='#pleio.q.test'>
<p>Single test of the number of traits associated with genotype</p></a></li>
<li><a href='#pleio.qdemo'>
<p>Demonstration dataset for quantitative pleiotropy tests</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pleiotropy Test for Multiple Traits on a Genetic Marker</td>
</tr>
<tr>
<td>Version:</td>
<td>1.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-12-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel J Schaid</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jason Sinnwell &lt;sinnwell.jason@mayo.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform tests for pleiotropy of multiple traits of various variable types on genotypes for a genetic marker.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), rms, Matrix, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bioinformaticstools.mayo.edu/research/pleio/">https://bioinformaticstools.mayo.edu/research/pleio/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-10 12:37:35 UTC; sinnwell</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-11 08:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='pleio.demo'>
Demonstration dataset for pleiotropy tests with a mixture of trait
types and covariates
</h2><span id='topic+pleio.demo'></span><span id='topic+geno'></span><span id='topic+y'></span><span id='topic+x'></span>

<h3>Description</h3>

<p>A simulated dataset with 3 y variables simulated from multivariate
normal with common correlation structure, correlation of 0.5. The traits
are (1) gaussian, (2) binomial, and (3) ordinal. Genotypes were simulated based
on minor allele frequency of 0.2, and assume that the 3 traits 
are not asssociated with dose of minor allele (all 3 betas = 0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pleio.demo")</code></pre>


<h3>Format</h3>

<p>y is a data.frame with 500 observations and 3 traits, whose
distribution families are gaussian, binomial, and ordinal, respectively;  
x is a matrix of 5 covariates,
geno is a single SNP of dosage of a minor allele simulated from frequency of 0.2. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pleio.demo)
str(y)
table(geno)
x[1:5,]
</code></pre>

<hr>
<h2 id='pleio.glm.fit'>
Compute Intermediate Statisics for Testing Pleiotropy
</h2><span id='topic+pleio.glm.fit'></span>

<h3>Description</h3>


<p>Pre-compute statisics for testing pleiotropy, to be used in
pleio.glm.test and pleio.glm.sequential 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio.glm.fit(y, g, glm.family, x.all=NULL, x.index.list=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pleio.glm.fit_+3A_y">y</code></td>
<td>
<p> matrix of multiple traits to test for
pleiotropy on genotype vector. Rows are for subjects and columns are for traits.
</p>
</td></tr>
<tr><td><code id="pleio.glm.fit_+3A_g">g</code></td>
<td>
<p>vector of genotypes coded as dosage of a minor allele, 0,1,
and 2. Allows imputed genotype dosage.
</p>
</td></tr>
<tr><td><code id="pleio.glm.fit_+3A_glm.family">glm.family</code></td>
<td>
<p>vector of strings corresponding to the type of
phenotype for the columns of y. Types accepted are gaussian,
binomial, and ordinal.
</p>
</td></tr>
<tr><td><code id="pleio.glm.fit_+3A_x.all">x.all</code></td>
<td>
<p> matrix of all covariates traits to adjust for the traits.
</p>
</td></tr>
<tr><td><code id="pleio.glm.fit_+3A_x.index.list">x.index.list</code></td>
<td>
<p> a list of vectors containing the index of the
columns of x.all for which to be adjusted for each trait. The list
must have the same number of vectors as there are traits in y. If
one trait has no covarites, specify a vector containing 0 for that trait.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>The matrix y of traits is expected to have  rows for subjects and
columns for traits. The number of rows of y  should match the length of
vector g. If adjusting traits by covarites in x.all, the x.index.list
must be a list containing a vector of indices of the columns of x.all
for each trait. If one trait does not have covariates and others do,
then the vector for the trait with no covariate must be a vector containing
the number 0. This function computes intermediate statistics that are
subsequently used to test various hypotheses about pleiotropy.
For technical details,  Schaid et al. (2016, 2017)
</p>


<h3>Value</h3>

<p>A list containing
</p>
<table>
<tr><td><code>theta</code></td>
<td>
  </td></tr>
<tr><td><code>n.intercepts</code></td>
<td>
 </td></tr>
<tr><td><code>n.coef.covar</code></td>
<td>
 </td></tr>
<tr><td><code>n.parm</code></td>
<td>
 </td></tr>
<tr><td><code>an.mat</code></td>
<td>
 </td></tr>
<tr><td><code>n.traits</code></td>
<td>
<p>number of traits (columns) in y</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Schaid, Jason Sinnwell.
</p>


<h3>References</h3>


<p>Schaid DJ, Tong X, Larrabee B, Kennedy RB, Poland GA, Sinnwell JP.
Statistical Methods for Testing Genetic Pleiotropy. Genetics. 2016
Oct;204(2):483-497. 
</p>
<p>Schaid DJ, Tong X, Batzler A, Sinnwell JP, Qing J, Biernacka
JM. Multivariate Generalized Linear Model for Genetic
Pleiotropy. Under review. 
</p>

<hr>
<h2 id='pleio.glm.sequential'>
Perform sequential tests of pleiotropy
</h2><span id='topic+pleio.glm.sequential'></span>

<h3>Description</h3>

<p>Perform sequential tests of pleiotropy to determine the number of
traits, and which traits, are associatd with a vector of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio.glm.sequential(obj.pleio.glm.fit, pval.threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pleio.glm.sequential_+3A_obj.pleio.glm.fit">obj.pleio.glm.fit</code></td>
<td>
<p> result of pleio.glm.fit </p>
</td></tr>
<tr><td><code id="pleio.glm.sequential_+3A_pval.threshold">pval.threshold</code></td>
<td>
<p>p-value for rejecting the null hypothesis of the
specified number of coefficients constrained to be zero.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>Perform sequential tests of pleiotropy, starting at the usual
multivarite null hypothesis that all coefficients = 0. If this test
rejects because the p-value &lt; pval.threshold, then allow one coefficient to
be non-zero in order to test whether the remaining coefficients = 0.
If the test of one non-zero coefficient rejects, then allow two
non-zero coefficients, considering all possible combinations of two 
non-zero coefficients and test whether the remaining coefficients =
0. Continue this sequential testing until the p-value for a test is
greater than the specific pval.threshold. The step at which the
p-value &gt; pval.threshold determines which traits are associated with
the genotype. If there are m traits, the sequential testing stops
either when p-value &gt; pval.threshold, or when (m-1) traits are tested.
If the p-value remains less than pval.threshold when testing (m-1)
traits, this implies that all m traits are associated with the
genotype.
</p>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the final test from the sequential testing</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>the number of nonzero coefficients</p>
</td></tr>
<tr><td><code>index.nonzero.coef</code></td>
<td>
<p>index of column(s) of y that have non-zero coefficients.
These indices indicate which traits are associated with the genotype,
accounting for the correlations among the traits.</p>
</td></tr>
</table>


<h3>Author(s)</h3>


<p>Dan Schaid and Jason Sinnwell
</p>


<h3>References</h3>

<p>Schaid DJ, Tong X, Larrabee B, Kennedy RB, Poland GA, Sinnwell JP.
Statistical Methods for Testing Genetic Pleiotropy. Genetics. 2016
Oct;204(2):483-497. 
</p>
<p>Schaid DJ, Tong X, Batzler A, Sinnwell JP, Qing J, Biernacka
JM. Multivariate Generalized Linear Model for Genetic
Pleiotropy. Under review. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pleio.demo)

## test without covars
fams &lt;- c("gaussian","binomial","ordinal")
obj &lt;- pleio.glm.fit(y, geno, glm.family=fams)
stat &lt;- pleio.glm.test(obj, count.nonzero.coef = 0)
stat$stat
stat$pval
pseq &lt;- pleio.glm.sequential(obj, pval.threshold=.5)
pseq
</code></pre>

<hr>
<h2 id='pleio.glm.test'>
Single test of the number of traits associated with genotype
</h2><span id='topic+pleio.glm.test'></span>

<h3>Description</h3>

<p>Perform single test of the number of traits associated with a
genotype, by allowing a specified number of regression beta's to
be unconstrained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio.glm.test(obj.pleio.glm.fit, count.nonzero.coef = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pleio.glm.test_+3A_obj.pleio.glm.fit">obj.pleio.glm.fit</code></td>
<td>
<p> result of pleio.glm.fit </p>
</td></tr>
<tr><td><code id="pleio.glm.test_+3A_count.nonzero.coef">count.nonzero.coef</code></td>
<td>
<p>Number of non-zero coefficients (betas). A non-zero beta is
allowed to be unconstrained, while all other beta's are constrained to be zero.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>By specifying the number of non-zero coefficients, the algorithm evaluates
all possible ways of choosing unconstrained and constrained betas, and
for each configuration a statistic (tk) is computed. This tk statistic
can be considered a measure of fit of a model. The minimum tk over all
possible configurations provides a global test of whether one of the
models fits well.
</p>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>stat</code></td>
<td>
<p>global test statistic</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>degrees of freedom of the statistic</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value for the test</p>
</td></tr>
<tr><td><code>index.nonzero.coef</code></td>
<td>
<p>index of the non-zero coefficients that provide
the minimum tk goodness of fit statistic - this configuration is
assumed to have coefficients for all other indices equal to zero.</p>
</td></tr>
<tr><td><code>tk</code></td>
<td>
<p>test testistic for the tests performed on trait combinations
in vk.set</p>
</td></tr>
<tr><td><code>vk.set</code></td>
<td>
<p>data.frame containing the tests performed.
For m traits, and k = count.nonzero.coef, there are
m-choose-k tests considered in the null hypothesis. The data.frame
provides the indices of the unconstrained coefficients for the
corresponding tk test statistic for the configuration. Rows are the
indices for each configuration, and the columns are for the
different configurations tested.</p>
</td></tr>
</table>


<h3>Author(s)</h3>


<p>Dan Schaid and Jason Sinnwell
</p>


<h3>References</h3>

<p>Schaid DJ, Tong X, Larrabee B, Kennedy RB, Poland GA, Sinnwell JP.
Statistical Methods for Testing Genetic Pleiotropy. Genetics. 2016
Oct;204(2):483-497. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pleio.demo)
obj &lt;- pleio.glm.fit(y, geno, glm.family=c("gaussian","binomial","ordinal"))
test1 &lt;- pleio.glm.test(obj, count.nonzero.coef = 0)
test1
test2 &lt;- pleio.glm.test(obj, count.nonzero.coef = 1)
test2
</code></pre>

<hr>
<h2 id='pleio.q.fit'>
Compute Intermediate Statisics for Testing Pleiotropy
</h2><span id='topic+pleio.q.fit'></span>

<h3>Description</h3>


<p>Pre-compute statisics for testing pleiotropy, to be used in
pleio.q.test and pleio.q.sequential. This is an earlier function for
gaussian traits without covariates. The function pleio.glm.fit is more
general, allowing for gaussian traits, and possibly adjusting covariates. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio.q.fit(y, geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pleio.q.fit_+3A_y">y</code></td>
<td>
<p> matrix of multiple quantitative traits to test for
pleiotropy on genotype vector. Rows are for subjects and columns are for traits.
</p>
</td></tr>
<tr><td><code id="pleio.q.fit_+3A_geno">geno</code></td>
<td>
<p>vector of genotypes coded as dosage of a minor allele,
0,1,2. Allows imputed genotype dosage.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>The matrix y of traits is expected to have  rows for subjects and
columns for traits. The number of rows of y  should match the length of
vector geno.  This function computes intermediate statistics that are
subsequently used to test various hypotheses about pleiotropy.
For technical details,  Schaid et al. (Genetics, 2016)
</p>


<h3>Value</h3>

<p>A list containing
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>the de-correlated genotype design matrix</p>
</td></tr>
<tr><td><code>xx.inv</code></td>
<td>
<p>inverse of x'x</p>
</td></tr>
<tr><td><code>beta.ols</code></td>
<td>
<p>estimated unconstrained beta coefficients for y traits based on
ordinary least squares</p>
</td></tr>
<tr><td><code>n.traits</code></td>
<td>
<p>number of traits (columns) in y</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Schaid, Jason Sinnwell.
</p>


<h3>References</h3>


<p>Schaid DJ, Tong X, Larrabee B, Kennedy RB, Poland GA, Sinnwell JP.
Statistical Methods for Testing Genetic Pleiotropy. Genetics. 2016
Oct;204(2):483-497. 
</p>

<hr>
<h2 id='pleio.q.sequential'>
Perform sequential tests of pleiotropy
</h2><span id='topic+pleio.q.sequential'></span>

<h3>Description</h3>

<p>Perform sequential tests of pleiotropy to determine the number of
traits, and which traits, are associatd with a vector of
genotypes. For gaussian traits without covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio.q.sequential(obj.fit, pval.threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pleio.q.sequential_+3A_obj.fit">obj.fit</code></td>
<td>
<p> result of pleio.q.fit </p>
</td></tr>
<tr><td><code id="pleio.q.sequential_+3A_pval.threshold">pval.threshold</code></td>
<td>
<p>p-value for rejecting the null hypothesis of the
specified number of non-zero betas.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>Perform sequential tests of pleiotropy, starting at the usual
multivarite null hypothesis that all coefficients = 0. If this test
rejects because the p-value &lt; pval.threshold, then allow one coefficient to
be non-zero in order to test whether the remaining coefficients = 0.
If the test of one non-zero coefficient rejects, then allow two
non-zero coefficients, considering all possible combinations of two 
non-zero coefficients and test whether the remaining coefficients =
0. Continue this sequential testing until the p-value for a test is
greater than the specific pval.threshold. The step at which the
p-value &gt; pval.threshold determines which traits are associated with
the genotype. If there are m traits, the sequential testing stops
either when p-value &gt; pval.threshold, or when (m-1) traits are tested.
If the p-value remains less than pval.threshold when testing (m-1)
traits, this implies that all m traits are associated with the
genotype.
</p>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the final test from the sequential testing</p>
</td></tr>
<tr><td><code>index.beta</code></td>
<td>
<p>index of columns of y that have non-zero betas.
These indices indicate which traits are associated with the genotype,
accounting for the correlations among the traits.</p>
</td></tr>
</table>


<h3>Author(s)</h3>


<p>Dan Schaid and Jason Sinnwell
</p>


<h3>References</h3>


<p>Schaid DJ, Tong X, Larrabee B, Kennedy RB, Poland GA, Sinnwell JP.
Statistical Methods for Testing Genetic Pleiotropy. To appear in Genetics, 2016.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pleio.qdemo)
fit &lt;- pleio.q.fit(y, geno)
test.seq &lt;- pleio.q.sequential(fit, pval.threshold=.05)
test.seq
</code></pre>

<hr>
<h2 id='pleio.q.test'>
Single test of the number of traits associated with genotype
</h2><span id='topic+pleio.q.test'></span>

<h3>Description</h3>

<p>Perform single test of the number of traits associated with a
genotype, by allowing a specified number of regression beta's to
be unconstrained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio.q.test(obj.fit, count.nonzero.beta = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pleio.q.test_+3A_obj.fit">obj.fit</code></td>
<td>
<p> result of pleio.q.fit </p>
</td></tr>
<tr><td><code id="pleio.q.test_+3A_count.nonzero.beta">count.nonzero.beta</code></td>
<td>
<p>Number of non-zero betas. A non-zero beta is
allowed to be unconstrained, while all other beta's are constrained
to be zero.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>By specifying the number of non-zero beta's, the algorithm evaluates
all possible ways of choosing unconstrained and constrained beta's, and
for each configuration a statistic (tk) is computed. This tk statistic
can be considered a measure of fit of a model. The minimum tk over all
possible configurations provides a global test of whether one of the
models fits well.
</p>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>stat</code></td>
<td>
<p>global test statistict</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>degrees of freedom of the statistic</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value for the test</p>
</td></tr>
<tr><td><code>index.nonzero.beta</code></td>
<td>
<p>index of the non-zero beta(s) that provide(s)
the minimum tk goodness of fit statistic - this configuration is
assumed to have beta's for all other indices equal to zero.</p>
</td></tr>
<tr><td><code>tests</code></td>
<td>
<p>data.frame containing the tests performed.
For m traits, and k = count.nonzero.beta, there are
m-choose-k tests considered in the null hypothesis. The data.frame
provides the indices of the unconstrained betas and the
corresponding tk test statistic for the configuration.</p>
</td></tr>
</table>


<h3>Author(s)</h3>


<p>Dan Schaid and Jason Sinnwell
</p>


<h3>References</h3>


<p>Schaid DJ, Tong X, Larrabee B, Kennedy RB, Poland GA, Sinnwell JP.
Statistical Methods for Testing Genetic Pleiotropy. Genetics. 2016
Oct;204(2):483-497.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pleio.qdemo)
fit &lt;- pleio.q.fit(y, geno)
## usual multivariate test of whether all betas = 0
test0 &lt;- pleio.q.test(fit, count.nonzero.beta = 0)
test0
## test whether allowing 2 betas to be non-zero fits data
test2 &lt;- pleio.q.test(fit, count.nonzero.beta = 2)
test2
</code></pre>

<hr>
<h2 id='pleio.qdemo'>
Demonstration dataset for quantitative pleiotropy tests
</h2><span id='topic+pleio.qdemo'></span>

<h3>Description</h3>

<p>A simulated dataset with 6 y variables simulated from multivariate
normal distribution with common correlation structure, correlation of
0.5, and genotypes simulated based on minor allele frequency of 0.2, and
assume that betas for traits 2 and 3 have non-zero values, while all
other traits are not asssociated with dose of minor allele. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pleio.qdemo")</code></pre>


<h3>Format</h3>

<p>y is a data.frame with 500 observations (rows) and 6 quantitative traits (columns),
while geno is a single SNP of dosage of a minor allele,
simulated from frequency of 0.2. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pleio.qdemo)
str(y)
table(geno)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
