<!DOCTYPE html><html lang="en"><head><title>Help for package pleiotest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pleiotest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#identify_subsets'><p>Internal function to identify sub-sets of data and return a list with IDs.</p></a></li>
<li><a href='#manhattan_plot'><p>Single Trait Manhattan plot</p></a></li>
<li><a href='#mt_gwas'><p>Multi-trait Genome wide association model.</p></a></li>
<li><a href='#pleio_ideogram'><p>Plot ideogram from pleio_test results</p></a></li>
<li><a href='#pleio_plot'><p>Pleiotropic manhattan plot</p></a></li>
<li><a href='#pleio_simulate'><p>Create simulations</p></a></li>
<li><a href='#pleio_test'><p>Sequential Wald test for pleiotropy</p></a></li>
<li><a href='#pleioR'><p>Fit a multi-trait model to test for genetic pleiotropy</p></a></li>
<li><a href='#xrsx_xrsy'><p>Calculate XRsX and XRsY</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Fast Sequential Pleiotropy Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>It performs a fast multi-trait genome-wide association analysis based on seemingly unrelated regressions. It tests for pleiotropic effects based on a series of Intersection-Union Wald tests. The package can handle large and unbalanced data and plot results.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, RColorBrewer</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/FerAguate/pleiotest">https://github.com/FerAguate/pleiotest</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-16 21:10:26 UTC; quantgen</td>
</tr>
<tr>
<td>Author:</td>
<td>Fernando Aguate <a href="https://orcid.org/0000-0002-3608-8425"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gustavo de los Campos
    <a href="https://orcid.org/0000-0001-5692-7129"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Alexander Grueneberg [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fernando Aguate &lt;fmaguate@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-18 10:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='identify_subsets'>Internal function to identify sub-sets of data and return a list with IDs.</h2><span id='topic+identify_subsets'></span>

<h3>Description</h3>

<p>This function is used internally in pleioR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>identify_subsets(trait, id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identify_subsets_+3A_trait">trait</code></td>
<td>
<p>character indicating traits.</p>
</td></tr>
<tr><td><code id="identify_subsets_+3A_id">id</code></td>
<td>
<p>character indicating IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with an ID matrix and ID subsets.
</p>


<h3>Author(s)</h3>

<p>Original code by Fernando Aguate.
</p>

<hr>
<h2 id='manhattan_plot'>Single Trait Manhattan plot</h2><span id='topic+manhattan_plot'></span>

<h3>Description</h3>

<p>Manhattan plot of results from mt_gwas function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan_plot(mt_gwas_results, trait, bp_positions, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="manhattan_plot_+3A_mt_gwas_results">mt_gwas_results</code></td>
<td>
<p>Object returned by mt_gwas</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_trait">trait</code></td>
<td>
<p>integer indicating the position of the trait (see: names(mt_gwas_results)) to be plotted.</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_bp_positions">bp_positions</code></td>
<td>
<p>dataframe with SNPs base pair positions. colnames msut be 'chr' and 'position', rownames must be SNP identifiers matching names in mt_gwas.</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_...">...</code></td>
<td>
<p>further graphical parameters. Options include: title=, bty=, pch=, cex.lab=, and cex.main=.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Manhattan plot
</p>

<hr>
<h2 id='mt_gwas'>Multi-trait Genome wide association model.</h2><span id='topic+mt_gwas'></span>

<h3>Description</h3>

<p>Performs a multi-trait model with correlated errors (seemingly unrelated regressions), and generates results by trait in a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mt_gwas(pleio_results, save_at = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mt_gwas_+3A_pleio_results">pleio_results</code></td>
<td>
<p>object of class pleio_class (returned by pleioR() function).</p>
</td></tr>
<tr><td><code id="mt_gwas_+3A_save_at">save_at</code></td>
<td>
<p>character with directory and/or file name (.rdata) to save the results. This is useful when handling multiple results such as in parallel jobs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with by trait dataframes that contain results of the multi-trait model.
</p>

<hr>
<h2 id='pleio_ideogram'>Plot ideogram from pleio_test results</h2><span id='topic+pleio_ideogram'></span>

<h3>Description</h3>

<p>Plots genomic segments that contain significant pleiotropic SNPs using results of pleio_test(). It also returns a dataframe with segment information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio_ideogram(
  pleio_res,
  alpha = "bonferroni05",
  n_traits = 2,
  bp_positions,
  window_size = 1e+06,
  centromeres = NULL,
  color_bias = 1,
  set_plot = T,
  set_legend = T,
  set_ylim_prop = 1.1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pleio_ideogram_+3A_pleio_res">pleio_res</code></td>
<td>
<p>list returned by pleio_test().</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_alpha">alpha</code></td>
<td>
<p>numeric threshold for significance level (Bonferroni correction by default).</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_n_traits">n_traits</code></td>
<td>
<p>integer indicating the level of pleiotropy to test (a.k.a. number of traits).</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_bp_positions">bp_positions</code></td>
<td>
<p>dataframe with colnames 'chr' and 'pos' indicating the chromosome and position for each SNP. Rownames must contain SNP names matching results of pleio_test.</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_window_size">window_size</code></td>
<td>
<p>numeric value indicating the minimum size (in base pairs) of the genomic region that contains significant SNPs.</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_centromeres">centromeres</code></td>
<td>
<p>string 'human' or dataframe (or matrix) with chromosome and position (in mbp) of the centromeres in the first and second columns. If NULL (default) does not plot the centromeres.</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_color_bias">color_bias</code></td>
<td>
<p>number for bias of the color scale. See help(colorRampPalette). By default color_bias = 1</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_set_plot">set_plot</code></td>
<td>
<p>logical indicating whether to plot the ideogram (TRUE by default).</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_set_legend">set_legend</code></td>
<td>
<p>logical indicating whether to plot a legend (TRUE by default).</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_set_ylim_prop">set_ylim_prop</code></td>
<td>
<p>numeric proportion of upper margin to fit the legend (no margin by default). 1 = no margin, 1.1 = 10% left for margin, etc.</p>
</td></tr>
<tr><td><code id="pleio_ideogram_+3A_...">...</code></td>
<td>
<p>more plot arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Ideogram plot and a dataframe with genomic segments information.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pleio_plot">pleio_plot</a></code>
</p>

<hr>
<h2 id='pleio_plot'>Pleiotropic manhattan plot</h2><span id='topic+pleio_plot'></span>

<h3>Description</h3>

<p>Plots the p-values that test the hypothesis of pleiotropic effects on n_traits. This function also returns a dataframe with information of the significant SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio_plot(
  pleio_res,
  alpha = "bonferroni05",
  n_traits = 2,
  bp_positions = NULL,
  set_colors = NULL,
  set_text = NULL,
  set_plot = TRUE,
  chr_spacing = 1e+05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pleio_plot_+3A_pleio_res">pleio_res</code></td>
<td>
<p>object returned by pleio_test().</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_alpha">alpha</code></td>
<td>
<p>numeric threshold for significance level (Bonferroni correction by default).</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_n_traits">n_traits</code></td>
<td>
<p>integer indicating the level of pleiotropy to test (a.k.a. number of traits).</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_bp_positions">bp_positions</code></td>
<td>
<p>dataframe with colnames 'chr' and 'pos' indicating the chromosome and position for each SNP. Rownames must contain SNP names matching results of pleio_test.</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_set_colors">set_colors</code></td>
<td>
<p>string with 3 colors to use in the plot (by default: c('goldenrod4', 'brown4', 'royalblue2')).</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_set_text">set_text</code></td>
<td>
<p>dataframe or matrix with strings to add as text to identify SNPs or genes. Rownames must be SNP names matching results of pleio_test. The first column of the dataframe must have strings to plot as text.</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_set_plot">set_plot</code></td>
<td>
<p>logical indicating whether to return the manhattan plot (TRUE by default).</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_chr_spacing">chr_spacing</code></td>
<td>
<p>integer indicating the spacing (in base pair positions) between chromosomes. 1e5 by default.</p>
</td></tr>
<tr><td><code id="pleio_plot_+3A_...">...</code></td>
<td>
<p>additional graphic parameters for the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Manhattan plot and dataframe with information related to significant SNPs.
</p>

<hr>
<h2 id='pleio_simulate'>Create simulations</h2><span id='topic+pleio_simulate'></span>

<h3>Description</h3>

<p>Example function to create simulations with no effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio_simulate(n_traits, n_individuals, n_snp, percentage_mv = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pleio_simulate_+3A_n_traits">n_traits</code></td>
<td>
<p>number of traits to simulate.</p>
</td></tr>
<tr><td><code id="pleio_simulate_+3A_n_individuals">n_individuals</code></td>
<td>
<p>number of individuals to simulate.</p>
</td></tr>
<tr><td><code id="pleio_simulate_+3A_n_snp">n_snp</code></td>
<td>
<p>number of SNPs to simulate.</p>
</td></tr>
<tr><td><code id="pleio_simulate_+3A_percentage_mv">percentage_mv</code></td>
<td>
<p>proportion of missing values. By default = 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with pheno and geno to test the pleioR function.
</p>


<h3>Author(s)</h3>

<p>Original code by Fernando Aguate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim1 &lt;- pleio_simulate(n_traits = 3, n_individuals = 1e4, n_snp = 1e3, percentage_mv = 0.1)
</code></pre>

<hr>
<h2 id='pleio_test'>Sequential Wald test for pleiotropy</h2><span id='topic+pleio_test'></span>

<h3>Description</h3>

<p>Performs the sequential test of pleiotropic effects using results of pleioR().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleio_test(
  pleio_results,
  loop_breaker = 1,
  save_at = NULL,
  contrast_matrices_list = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pleio_test_+3A_pleio_results">pleio_results</code></td>
<td>
<p>pleio_class object returned by pleioR().</p>
</td></tr>
<tr><td><code id="pleio_test_+3A_loop_breaker">loop_breaker</code></td>
<td>
<p>numeric value for a maximum p-value used to stop the sequence if a higher p-value is obtained. This saves computation time if there are many tests to perform.</p>
</td></tr>
<tr><td><code id="pleio_test_+3A_save_at">save_at</code></td>
<td>
<p>character with directory and/or file name (.rdata) to save the results. This is useful when handling multiple results such as in parallel jobs.</p>
</td></tr>
<tr><td><code id="pleio_test_+3A_contrast_matrices_list">contrast_matrices_list</code></td>
<td>
<p>user-specified contrast matrices within a list of lists, or a single contrast matrix (see example). Each matrix must have the same number of columns, and must be equal to the number of traits.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of p-values, indices, and trait numeric identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of user-specified contrast matrices with 3 traits
cm1 &lt;- matrix(c(-1, 0, 1), ncol = 3)
cm2 &lt;- matrix(c(0, -1, 1), ncol = 3)
contrast_matrices &lt;- list('1vs3' = list(cm1), '2vs3' = list(cm2))
# or a single contrast matrix as:
contrast_matrices &lt;- cm1
</code></pre>

<hr>
<h2 id='pleioR'>Fit a multi-trait model to test for genetic pleiotropy</h2><span id='topic+pleioR'></span>

<h3>Description</h3>

<p>Fits a seemingly unrelated regression with, possibly unbalanced data, and/or covariates. It returns a pleio_class object to perform the sequential test with pleio_test() or to obtain by-trait estimates with mt_gwas().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pleioR(pheno, geno, i = NULL, j = NULL, covariates = NULL, drop_subsets = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pleioR_+3A_pheno">pheno</code></td>
<td>
<p>dataframe with phenotypic data. Must have columns 'id', 'trait', and 'y'. Column 'y' must contain the observations for the corresponding 'trait' and 'id'. See function melt() in the 'reshape2' package for a simple formatting of your data.</p>
</td></tr>
<tr><td><code id="pleioR_+3A_geno">geno</code></td>
<td>
<p>matrix with SNPs in columns and IDs in rownames. This can also be a memory-mapped matrix returned by BEDMatrix() in the 'BEDMatrix' package.</p>
</td></tr>
<tr><td><code id="pleioR_+3A_i">i</code></td>
<td>
<p>integers indexing rows from geno to use in the model.</p>
</td></tr>
<tr><td><code id="pleioR_+3A_j">j</code></td>
<td>
<p>integers indexing columns from geno to use in the model. Useful when working with multiple jobs in parallel.</p>
</td></tr>
<tr><td><code id="pleioR_+3A_covariates">covariates</code></td>
<td>
<p>(optional) dataframe or matrix containing covariates in columns and IDs as rownames. These IDs must match those in geno.</p>
</td></tr>
<tr><td><code id="pleioR_+3A_drop_subsets">drop_subsets</code></td>
<td>
<p>minimum sample size of sub-data sets to consider for analysis, 10 by default. When working with unbalanced data (a.k.a. fragmented data), save computation time by dropping small fragments of data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pleio_class list of left and right hand side solutions of the model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Random generated example with 3 traits, 1e4 individuals, 1000 SNPs and 10% missing values.
sim1 &lt;- pleio_simulate(n_traits = 3, n_individuals = 1e4, n_snp = 1e3, percentage_mv = 0.1)
pleio_model &lt;- pleioR(pheno = sim1$pheno, geno = sim1$geno)
pleio_model_test &lt;- pleio_test(pleio_model)
</code></pre>

<hr>
<h2 id='xrsx_xrsy'>Calculate XRsX and XRsY</h2><span id='topic+xrsx_xrsy'></span>

<h3>Description</h3>

<p>internal function to calculate crossproducts within pleioR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xrsx_xrsy(id_matrix, sets_rs, xx, xy)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="xrsx_xrsy_+3A_id_matrix">id_matrix</code></td>
<td>
<p>matrix of IDs</p>
</td></tr>
<tr><td><code id="xrsx_xrsy_+3A_sets_rs">sets_rs</code></td>
<td>
<p>list of inverses of matrix R</p>
</td></tr>
<tr><td><code id="xrsx_xrsy_+3A_xx">xx</code></td>
<td>
<p>numeric vector with crossproducts of the X matrix</p>
</td></tr>
<tr><td><code id="xrsx_xrsy_+3A_xy">xy</code></td>
<td>
<p>matrix with X transpose Y products.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
