<!DOCTYPE html><html><head><title>Help for package plinkQC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {plinkQC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check_ancestry'><p>Identification of individuals of divergent ancestry</p></a></li>
<li><a href='#check_het_and_miss'><p>Identification of individuals with outlying missing genotype or</p>
heterozygosity rates</a></li>
<li><a href='#check_hwe'><p>Identification of SNPs showing a significant deviation from Hardy-Weinberg-</p>
equilibrium (HWE)</a></li>
<li><a href='#check_maf'><p>Identification of SNPs with low minor allele frequency</p></a></li>
<li><a href='#check_relatedness'><p>Identification of related individuals</p></a></li>
<li><a href='#check_sex'><p>Identification of individuals with discordant sex information</p></a></li>
<li><a href='#check_snp_missingness'><p>Identification of SNPs with high missingness rate</p></a></li>
<li><a href='#checkFiltering'><p>Check and construct PLINK sample and marker filters</p></a></li>
<li><a href='#checkPlink'><p>Check PLINK software access</p></a></li>
<li><a href='#checkRemoveIDs'><p>Check and construct individual IDs to be removed</p></a></li>
<li><a href='#cleanData'><p>Create plink dataset with individuals and markers passing quality control</p></a></li>
<li><a href='#evaluate_check_ancestry'><p>Evaluate results from PLINK PCA on combined study and reference data</p></a></li>
<li><a href='#evaluate_check_het_and_miss'><p>Evaluate results from PLINK missing genotype and heterozygosity rate check.</p></a></li>
<li><a href='#evaluate_check_relatedness'><p>Evaluate results from PLINK IBD estimation.</p></a></li>
<li><a href='#evaluate_check_sex'><p>Evaluate results from PLINK sex check.</p></a></li>
<li><a href='#overviewPerIndividualQC'><p>Overview of per sample QC</p></a></li>
<li><a href='#overviewPerMarkerQC'><p>Overview of per marker QC</p></a></li>
<li><a href='#perIndividualQC'><p>Quality control for all individuals in plink-dataset</p></a></li>
<li><a href='#perMarkerQC'><p>Quality control for all markers in plink-dataset</p></a></li>
<li><a href='#plinkQC'><p>plinkQC</p></a></li>
<li><a href='#relatednessFilter'><p>Remove related individuals while keeping maximum number of individuals</p></a></li>
<li><a href='#run_check_ancestry'><p>Run PLINK principal component analysis</p></a></li>
<li><a href='#run_check_heterozygosity'><p>Run PLINK heterozygosity rate calculation</p></a></li>
<li><a href='#run_check_missingness'><p>Run PLINK missingness rate calculation</p></a></li>
<li><a href='#run_check_relatedness'><p>Run PLINK IBD estimation</p></a></li>
<li><a href='#run_check_sex'><p>Run PLINK sexcheck</p></a></li>
<li><a href='#testNumerics'><p>Test lists for different properties of numerics</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genotype Quality Control with 'PLINK'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://meyer-lab-cshl.github.io/plinkQC/">https://meyer-lab-cshl.github.io/plinkQC/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/meyer-lab-cshl/plinkQC/issues">https://github.com/meyer-lab-cshl/plinkQC/issues</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hannah Meyer &lt;hannah.v.meyer@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genotyping arrays enable the direct measurement of an individuals
    genotype at thousands of markers. 'plinkQC' facilitates genotype quality
    control for genetic association studies as described by Anderson and
    colleagues (2010) &lt;<a href="https://doi.org/10.1038%2Fnprot.2010.116">doi:10.1038/nprot.2010.116</a>&gt;. It makes 'PLINK' basic
    statistics (e.g. missing genotyping rates per individual, allele frequencies
    per genetic marker) and relationship functions accessible from 'R' and
    generates a per-individual and per-marker quality control report.
    Individuals and markers that fail the quality control can subsequently be
    removed to generate a new, clean dataset. Removal of individuals based on
    relationship status is optimised to retain as many individuals as possible
    in the study.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, optparse, data.table (&ge; 1.11.0), R.utils, ggplot2,
ggforce, ggrepel, cowplot, UpSetR, dplyr, igraph (&ge; 1.2.4),
sys</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, mockery, formatR, knitr, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>plink (1.9)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-07-15 15:22:27 UTC; hannah</td>
</tr>
<tr>
<td>Author:</td>
<td>Hannah Meyer <a href="https://orcid.org/0000-0003-4564-0899"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-15 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='check_ancestry'>Identification of individuals of divergent ancestry</h2><span id='topic+check_ancestry'></span>

<h3>Description</h3>

<p>Runs and evaluates results of plink &ndash;pca on merged genotypes from
individuals to be QCed and individuals of reference population of known
genotypes. Currently, check ancestry only supports automatic selection of
individuals of European descent. It uses information from principal
components 1 and 2 returned by plink &ndash;pca to find the center of the European
reference samples (mean(PC1_europeanRef), mean(PC2_europeanRef). It then
computes the maximum Euclidean distance (maxDist) of the European reference
samples from this centre. All study samples whose Euclidean distance from the
centre falls outside the circle described by the radius r=europeanTh* maxDist
are considered non-European and their IDs are returned as failing the
ancestry check.
<code>check_ancestry</code> creates a scatter plot of PC1 versus PC2 colour-coded
for samples of the reference populations and the study population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_ancestry(
  indir,
  name,
  qcdir = indir,
  prefixMergedDataset,
  europeanTh = 1.5,
  defaultRefSamples = c("HapMap", "1000Genomes"),
  refPopulation = c("CEU", "TSI"),
  refSamples = NULL,
  refColors = NULL,
  refSamplesFile = NULL,
  refColorsFile = NULL,
  refSamplesIID = "IID",
  refSamplesPop = "Pop",
  refColorsColor = "Color",
  refColorsPop = "Pop",
  studyColor = "#2c7bb6",
  legend_labels_per_row = 6,
  run.check_ancestry = TRUE,
  interactive = FALSE,
  verbose = verbose,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  highlight_legend = FALSE,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  path2plink = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_ancestry_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_name">name</code></td>
<td>
<p>[character] prefix of plink files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where
prefixMergedDataset.eigenvec results as returned by plink &ndash;pca should be
saved. Per default qcdir=indir. If run.check_ancestry is FALSE, it is assumed
that plink &ndash;pca prefixMergedDataset has been run and
qcdir/prefixMergedDataset.eigenvec exists.User needs writing permission to
qcdir.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_prefixmergeddataset">prefixMergedDataset</code></td>
<td>
<p>[character] Prefix of merged dataset (study and
reference samples) used in plink &ndash;pca, resulting in
prefixMergedDataset.eigenvec.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_europeanth">europeanTh</code></td>
<td>
<p>[double] Scaling factor of radius to be drawn around center
of European reference samples, with study samples inside this radius
considered to be of European descent and samples outside this radius of
non-European descent. The radius is computed as the maximum Euclidean
distance of European reference samples to the centre of European reference
samples.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_defaultrefsamples">defaultRefSamples</code></td>
<td>
<p>[character] Option to use pre-downloaded individual
and population identifiers from either the 1000Genomes or HapMap project.
If refSamples and refSamplesFile are not provided, the HapMap identifiers (or
1000Genomes is specified) will be used as default and the function will fail
if the reference samples in the prefixMergedDataset do not match these
reference samples. If refColors and refColorsFile are not provided, this also
sets default colors for the reference populations.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refpopulation">refPopulation</code></td>
<td>
<p>[vector] Vector with population identifiers of European
reference population. Identifiers have to be corresponding to population IDs
[refColorsPop] in refColorsfile/refColors.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refsamples">refSamples</code></td>
<td>
<p>[data.frame] Dataframe with sample identifiers
[refSamplesIID] corresponding to IIDs in prefixMergedDataset.eigenvec and
population identifier [refSamplesPop] corresponding to population IDs
[refColorsPop] in refColorsfile/refColors. Either refSamples or
refSamplesFile have to be specified.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refcolors">refColors</code></td>
<td>
<p>[data.frame, optional] Dataframe with population IDs in
column [refColorsPop] and corresponding colour-code for PCA plot in column
[refColorsColor]. If not provided and is.null(refColorsFile) default colors
are used.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refsamplesfile">refSamplesFile</code></td>
<td>
<p>[character] /path/to/File/with/reference samples. Needs
columns with sample identifiers [refSamplesIID] corresponding to IIDs in
prefixMergedDataset.eigenvec and population identifier [refSamplesPop]
corresponding to population IDs [refColorsPop] in refColorsfile/refColors.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refcolorsfile">refColorsFile</code></td>
<td>
<p>[character, optional]
/path/to/File/with/Population/Colors containing population IDs in column
[refColorsPop] and corresponding colour-code for PCA plot in column
[refColorsColor].If not provided and is.null(refColors) default colors for
are used.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refsamplesiid">refSamplesIID</code></td>
<td>
<p>[character] Column name of reference sample IDs in
refSamples/refSamplesFile.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refsamplespop">refSamplesPop</code></td>
<td>
<p>[character] Column name of reference sample population
IDs in refSamples/refSamplesFile.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refcolorscolor">refColorsColor</code></td>
<td>
<p>[character] Column name of population colors in
refColors/refColorsFile</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_refcolorspop">refColorsPop</code></td>
<td>
<p>[character] Column name of reference sample population
IDs in refColors/refColorsFile.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_studycolor">studyColor</code></td>
<td>
<p>[character] Colour to be used for study population.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_legend_labels_per_row">legend_labels_per_row</code></td>
<td>
<p>[integer] Number of population names per row in
PCA plot.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_run.check_ancestry">run.check_ancestry</code></td>
<td>
<p>[logical] Should plink &ndash;pca be run to
determine principal components of merged dataset; if FALSE, it is assumed
that plink &ndash;pca has been run successfully and
qcdir/prefixMergedDataset.eigenvec is present;
<code><a href="#topic+check_ancestry">check_ancestry</a></code> will fail with missing file error otherwise.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_ancestry) via ggplot2::ggsave(p=p_ancestry,
other_arguments) or pdf(outfile) print(p_ancestry) dev.off().</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_ancestry); all highlight_samples IIDs have to
be present in the IIDs of the prefixMergedDataset.fam file.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified.Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_ancestry_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named [list] with i) fail_ancestry, containing a [data.frame] with
FID and IID of non-European individuals and ii) p_ancestry, a ggplot2-object
'containing' a scatter plot of PC1 versus PC2 colour-coded for samples of the
reference populations and the study population, which can be shown by
print(p_ancestry).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- "data"
fail_ancestry &lt;- check_ancestry(indir=indir, name=name,
refSamplesFile=paste(indir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(indir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE,
run.check_ancestry=FALSE)

# highlight samples
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
fail_ancestry &lt;- check_ancestry(indir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE,
highlight_samples = highlight_samples[,2],
run.check_ancestry=FALSE,
highlight_type = c("text", "shape"))

## End(Not run)
</code></pre>

<hr>
<h2 id='check_het_and_miss'>Identification of individuals with outlying missing genotype or
heterozygosity rates</h2><span id='topic+check_het_and_miss'></span>

<h3>Description</h3>

<p>Runs and evaluates results from plink &ndash;missing (missing genotype rates
per individual) and plink &ndash;het (heterozygosity rates per individual).
Non-systematic failures in genotyping and outlying heterozygosity (hz) rates
per individual are often proxies for DNA sample quality. Larger than expected
heterozygosity can indicate possible DNA contamination.
The mean heterozygosity in PLINK is computed as hz_mean = (N-O)/N, where
N: number of non-missing genotypes and O:observed number of homozygous
genotypes for a given individual.
Mean heterozygosity can differ between populations and SNP genotyping panels.
Within a population and genotyping panel, a reduced heterozygosity rate can
indicate inbreeding - these individuals will then likely be returned by
<code><a href="#topic+check_relatedness">check_relatedness</a></code> as individuals that fail the relatedness
filters. <code>check_het_and_miss</code> creates a scatter plot with the
individuals' missingness rates on x-axis and their heterozygosity rates on
the y-axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_het_and_miss(
  indir,
  name,
  qcdir = indir,
  imissTh = 0.03,
  hetTh = 3,
  run.check_het_and_miss = TRUE,
  label_fail = TRUE,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  highlight_legend = FALSE,
  interactive = FALSE,
  verbose = FALSE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  path2plink = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_het_and_miss_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam, name.het and name.imiss.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where name.het as returned by
plink &ndash;het and name.imiss as returned by plink &ndash;missing will be saved. Per
default qcdir=indir. If run.check_het_and_miss is FALSE, it is assumed that
plink &ndash;missing and plink &ndash;het have been run and qcdir/name.imiss and
qcdir/name.het are present. User needs writing permission to qcdir.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_imissth">imissTh</code></td>
<td>
<p>[double] Threshold for acceptable missing genotype rate per
individual; has to be proportion between (0,1)</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_hetth">hetTh</code></td>
<td>
<p>[double] Threshold for acceptable deviation from mean
heterozygosity per individual. Expressed as multiples of standard
deviation of heterozygosity (het), i.e. individuals outside mean(het) +/-
hetTh*sd(het) will be returned as failing heterozygosity check; has to be
larger than 0.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_run.check_het_and_miss">run.check_het_and_miss</code></td>
<td>
<p>[logical] Should plink &ndash;missing and plink
&ndash;het be run to determine genotype missingness and heterozygosity rates; if
FALSE, it is assumed that plink &ndash;missing and plink &ndash;het have been run and
qcdir/name.imiss and qcdir/name.het are present;
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code> will fail with missing file error otherwise.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_label_fail">label_fail</code></td>
<td>
<p>[logical] Set TRUE, to add fail IDs as text labels in
scatter plot.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_het_imiss); all highlight_samples IIDs have to
be present in the IIDs of the name.fam file.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified.Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_het_imiss) via ggplot2::ggsave(p=p_het_imiss
, other_arguments) or pdf(outfile) print(p_het_imiss) dev.off().</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_het_and_miss_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code> wraps around
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code>,
<code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code> and
<code><a href="#topic+evaluate_check_het_and_miss">evaluate_check_het_and_miss</a></code>.
If run.check_het_and_miss is TRUE, <code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code> and
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code> are executed ; otherwise it is assumed
that plink &ndash;missing and plink &ndash;het have been run externally and
qcdir/name.het and qcdir/name.imiss exist. <code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>
will fail with missing file error otherwise.
</p>
<p>For details on the output data.frame fail_imiss and fail_het, check the
original description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#imiss">https://www.cog-genomics.org/plink/1.9/formats#imiss</a> and
<a href="https://www.cog-genomics.org/plink/1.9/formats#het">https://www.cog-genomics.org/plink/1.9/formats#het</a>
</p>


<h3>Value</h3>

<p>Named [list] with i) fail_imiss [data.frame] containing FID (Family
ID), IID (Within-family ID), MISS_PHENO (Phenotype missing? (Y/N)), N_MISS
(Number of missing genotype call(s), not including obligatory missings),
N_GENO (Number of potentially valid call(s)), F_MISS (Missing call rate) of
individuals failing missing genotype check and ii) fail_het [data.frame]
containing FID (Family ID), IID (Within-family ID), O(HOM) (Observed number
of homozygotes), E(HOM) (Expected number of homozygotes), N(NM) (Number of
non-missing autosomal genotypes), F (Method-of-moments F coefficient
estimate) of individuals failing  outlying heterozygosity check and iii)
p_het_imiss, a ggplot2-object 'containing' a scatter plot with the samples'
missingness rates on x-axis and their heterozygosity rates on the y-axis,
which can be shown by print(p_het_imiss).
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- "data"
path2plink &lt;- "path/to/plink"

# whole dataset
fail_het_miss &lt;- check_het_and_miss(indir=indir, name=name,
interactive=FALSE,path2plink=path2plink)

# subset of dataset with sample highlighting
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
fail_het_miss &lt;- check_het_and_miss(indir=indir, name=name,
interactive=FALSE,path2plink=path2plink,
remove_individuals=remove_individuals_file,
highlight_samples=highlight_samples[,2], highlight_type = c("text", "shape"))

## End(Not run)
</code></pre>

<hr>
<h2 id='check_hwe'>Identification of SNPs showing a significant deviation from Hardy-Weinberg-
equilibrium (HWE)</h2><span id='topic+check_hwe'></span>

<h3>Description</h3>

<p>Runs and evaluates results from plink &ndash;hardy. It calculates the observed and
expected heterozygote frequencies for all variants in the individuals that
passed the <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> and computes the deviation of the
frequencies from Hardy-Weinberg equilibrium (HWE) by HWE exact test. The
p-values of the HWE exact test are displayed as histograms (stratified by
all and low p-values), where the hweTh is used to depict the quality control
cut-off for SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_hwe(
  indir,
  name,
  qcdir = indir,
  hweTh = 1e-05,
  interactive = FALSE,
  path2plink = NULL,
  verbose = FALSE,
  showPlinkOutput = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_hwe_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> was conducted, this directory should be the
same as qcdir specified in <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_hweth">hweTh</code></td>
<td>
<p>[double] Significance threshold for deviation from HWE.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_hwe) via ggplot2::ggsave(p=p_hwe,
other_arguments) or pdf(outfile) print(p_hwe) dev.off().</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out
and specifically, if TRUE, plink log will be displayed.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_hwe_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>check_hwe</code> uses plink &ndash;remove name.fail.IDs &ndash;hardy to
calculate the observed and expected heterozygote frequencies per SNP in the
individuals that passed the <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>. It does so
without generating a new dataset but simply removes the IDs when calculating
the statistics.
</p>
<p>For details on the output data.frame fail_hwe, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#hwe">https://www.cog-genomics.org/plink/1.9/formats#hwe</a>.
</p>


<h3>Value</h3>

<p>Named list with i) fail_hwe containing a [data.frame] with CHR
(Chromosome code), SNP (Variant identifier), TEST (Type of test: one of
'ALL', 'AFF', 'UNAFF', 'ALL(QT)', 'ALL(NP)'), A1 (Allele 1; usually minor),
A2 (Allele 2; usually major), GENO ('/'-separated genotype counts: A1 hom,
het, A2 hom), O(HET) (Observed heterozygote frequency E(HET) (Expected
heterozygote frequency), P (Hardy-Weinberg equilibrium exact test p-value)
for all SNPs that failed the hweTh and ii) p_hwe, a ggplot2-object
'containing' the HWE p-value distribution histogram which can be shown by
(print(p_hwe)).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# run on all individuals and markers
fail_hwe &lt;- check_hwe(indir=indir, qcdir=qcdir, name=name, interactive=FALSE,
verbose=TRUE, path2plink=path2plink)

# run on subset of individuals and markers
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
extract_markers_file &lt;- system.file("extdata", "extract_markers",
package="plinkQC")
fail_hwe &lt;- check_hwe(qcdir=qcdir, indir=indir,
name=name, interactive=FALSE, verbose=TRUE, path2plink=path2plink,
remove_individuals=remove_individuals_file,
extract_markers=extract_markers_file)

## End(Not run)
</code></pre>

<hr>
<h2 id='check_maf'>Identification of SNPs with low minor allele frequency</h2><span id='topic+check_maf'></span>

<h3>Description</h3>

<p>Runs and evaluates results from plink &ndash;freq. It calculates the minor allele
frequencies for all variants in the individuals that passed the
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code>. The minor allele frequency distributions is
plotted as a histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_maf(
  indir,
  name,
  qcdir = indir,
  macTh = 20,
  mafTh = NULL,
  verbose = FALSE,
  interactive = FALSE,
  path2plink = NULL,
  showPlinkOutput = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_maf_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> was conducted, this directory should be the
same as qcdir specified in <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_macth">macTh</code></td>
<td>
<p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples).</p>
</td></tr>
<tr><td><code id="check_maf_+3A_mafth">mafTh</code></td>
<td>
<p>[double] Threshold for minor allele frequency cut-off.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out
and specifically, if TRUE, plink log will be displayed.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_hwe) via ggplot2::ggsave(p=p_maf,
other_arguments) or pdf(outfile) print(p_maf) dev.off().</p>
</td></tr>
<tr><td><code id="check_maf_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_maf_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_maf_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>check_maf</code> uses plink &ndash;remove name.fail.IDs &ndash;freq to calculate the
minor allele frequencies for all variants in the individuals that passed the
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code>. It does so without generating a new dataset
but simply removes the IDs when calculating the statistics.
</p>
<p>For details on the output data.frame fail_maf, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#frq">https://www.cog-genomics.org/plink/1.9/formats#frq</a>.
</p>


<h3>Value</h3>

<p>Named list with i) fail_maf containing a [data.frame] with CHR
(Chromosome code), SNP (Variant identifier), A1 (Allele 1; usually minor), A2
(Allele 2; usually major), MAF (Allele 1 frequency), NCHROBS (Number of
allele observations) for all SNPs that failed the mafTh/macTh and ii) p_maf,
a ggplot2-object 'containing' the MAF distribution histogram which can be
shown by (print(p_maf)).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# run on all individuals and markers
fail_maf &lt;- check_maf(indir=indir, qcdir=qcdir, name=name, macTh=15,
interactive=FALSE, verbose=TRUE, path2plink=path2plink)

# run on subset of individuals and markers
keep_individuals_file &lt;- system.file("extdata", "keep_individuals",
package="plinkQC")
exclude_markers_file &lt;- system.file("extdata", "exclude_markers",
package="plinkQC")
fail_maf &lt;- check_maf(qcdir=qcdir, indir=indir,
name=name, interactive=FALSE, verbose=TRUE, path2plink=path2plink,
keep_individuals=keep_individuals_file, exclude_markers=exclude_markers_file)

## End(Not run)
</code></pre>

<hr>
<h2 id='check_relatedness'>Identification of related individuals</h2><span id='topic+check_relatedness'></span>

<h3>Description</h3>

<p>Runs and evaluates results from plink &ndash;genome.
plink &ndash;genome calculates identity by state (IBS) for each pair of
individuals based on the average proportion of alleles shared at genotyped
SNPs. The degree of recent shared ancestry, i.e. the identity by descent
(IBD) can be estimated from the genome-wide IBS. The proportion of IBD
between two individuals is returned by plink &ndash;genome as PI_HAT.
check_relatedness finds pairs of samples whose proportion of IBD is larger
than the specified highIBDTh. Subsequently, for pairs of individuals that do
not have additional relatives in the dataset, the individual with the greater
genotype missingness rate is selected and returned as the individual failing
the relatedness check. For more complex family structures, the unrelated
individuals per family are selected (e.g. in a parents-offspring trio, the
offspring will be marked as fail, while the parents will be kept in the
analysis).
<code>check_relatedness</code> depicts all pair-wise IBD-estimates as histograms
stratified by value of PI_HAT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_relatedness(
  indir,
  name,
  qcdir = indir,
  highIBDTh = 0.1875,
  filter_high_ldregion = TRUE,
  high_ldregion_file = NULL,
  genomebuild = "hg19",
  imissTh = 0.03,
  run.check_relatedness = TRUE,
  interactive = FALSE,
  verbose = FALSE,
  mafThRelatedness = 0.1,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_relatedness_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam, name.genome and name.imiss.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to where name.genome as returned
by plink &ndash;genome will be saved.  Per default qcdir=indir. If
run.check_relatedness is FALSE, it is assumed that plink
&ndash;missing and plink &ndash;genome have been run and qcdir/name.imiss and
qcdir/name.genome exist. User needs writing permission to qcdir.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_highibdth">highIBDTh</code></td>
<td>
<p>[double] Threshold for acceptable proportion of IBD between
pair of individuals.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_filter_high_ldregion">filter_high_ldregion</code></td>
<td>
<p>[logical] Should high LD regions be filtered
before IBD estimation; carried out per default with high LD regions for
hg19 provided as default via <code>genomebuild</code>. For alternative genome
builds not provided or non-human data, high LD regions files can be
provided via <code>high_ldregion_file</code>.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_high_ldregion_file">high_ldregion_file</code></td>
<td>
<p>[character] Path to file with high LD regions used
for filtering before IBD estimation if <code>filter_high_ldregion</code> == TRUE,
otherwise ignored; for human genome data, high LD region files are provided
and can simply be chosen via <code>genomebuild</code>. Files have to be
space-delimited, no column names with the following columns: chromosome,
region-start, region-end, region number. Chromosomes are specified without
'chr' prefix. For instance:
1 48000000 52000000 1
2 86000000 100500000 2</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_genomebuild">genomebuild</code></td>
<td>
<p>[character] Name of the genome build of the PLINK file
annotations, ie mappings in the name.bim file. Will be used to remove
high-LD regions based on the coordinates of the respective build. Options
are hg18, hg19 and hg38. See @details.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_imissth">imissTh</code></td>
<td>
<p>[double] Threshold for acceptable missing genotype rate in any
individual; has to be proportion between (0,1)</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_run.check_relatedness">run.check_relatedness</code></td>
<td>
<p>[logical] Should plink &ndash;genome be run to
determine pairwise IBD of individuals; if FALSE, it is assumed that
plink &ndash;genome and plink &ndash;missing have been run and qcdir/name.imiss and
qcdir/name.genome are present;
<code><a href="#topic+check_relatedness">check_relatedness</a></code> will fail with missing file error otherwise.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_IBD() via ggplot2::ggsave(p=p_IBD,
other_arguments) or pdf(outfile) print(p_IBD) dev.off().</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_mafthrelatedness">mafThRelatedness</code></td>
<td>
<p>[double] Threshold of minor allele frequency filter
for selecting variants for IBD estimation.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="check_relatedness_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+check_relatedness">check_relatedness</a></code> wraps around
<code><a href="#topic+run_check_relatedness">run_check_relatedness</a></code> and
<code><a href="#topic+evaluate_check_relatedness">evaluate_check_relatedness</a></code>. If run.check_relatedness is TRUE,
<code><a href="#topic+run_check_relatedness">run_check_relatedness</a></code> is executed ; otherwise it is assumed that
plink &ndash;genome has been run externally and qcdir/name.genome exists.
<code><a href="#topic+check_relatedness">check_relatedness</a></code>  will fail with missing file error otherwise.
</p>
<p>For details on the output data.frame fail_high_IBD, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#genome">https://www.cog-genomics.org/plink/1.9/formats#genome</a>.
</p>


<h3>Value</h3>

<p>Named [list] with i) fail_high_IBD containing a [data.frame] of
IIDs and FIDs of individuals who fail the IBDTh in columns
FID1 and IID1. In addition, the following columns are returned (as originally
obtained by plink &ndash;genome):
FID2 (Family ID for second sample), IID2 (Individual ID for second sample),
RT (Relationship type inferred from .fam/.ped file), EZ (IBD sharing expected
value, based on just .fam/.ped relationship), Z0 (P(IBD=0)), Z1 (P(IBD=1)),
Z2 (P(IBD=2)), PI_HAT (Proportion IBD, i.e. P(IBD=2) + 0.5*P(IBD=1)), PHE
(Pairwise phenotypic code (1, 0, -1 = AA, AU, and UU pairs, respectively)),
DST (IBS distance, i.e. (IBS2 + 0.5*IBS1) / (IBS0 + IBS1 + IBS2)), PPC (IBS
binomial test), RATIO (HETHET : IBS0 SNP ratio (expected value 2)).
and ii) failIDs containing a [data.frame] with individual IDs [IID] and
family IDs [FID] of individuals failing the highIBDTh iii) p_IBD, a
ggplot2-object 'containing' all pair-wise IBD-estimates as histograms
stratified by value of PI_HAT, which can be
shown by print(p_IBD).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- 'data'
path2plink &lt;- "path/to/plink"

# whole dataset
relatednessQC &lt;- check_relatedness(indir=indir, name=name, interactive=FALSE,
run.check_relatedness=FALSE, path2plink=path2plink)

# subset of dataset
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
fail_relatedness &lt;- check_relatedness(indir=qcdir, name=name,
remove_individuals=remove_individuals_file, path2plink=path2plink)

## End(Not run)
</code></pre>

<hr>
<h2 id='check_sex'>Identification of individuals with discordant sex information</h2><span id='topic+check_sex'></span>

<h3>Description</h3>

<p>Runs and evaluates results from plink &ndash;check-sex.
<code><a href="#topic+check_sex">check_sex</a></code> returns IIDs for individuals whose SNPSEX != PEDSEX
(where the SNPSEX is determined by the heterozygosity rate across
X-chromosomal variants).
Mismatching SNPSEX and PEDSEX IDs can indicate plating errors, sample-mixup
or generally samples with poor genotyping. In the latter case, these IDs are
likely to fail other QC steps as well.
Optionally, an extra data.frame (externalSex) with sample IDs and sex can be
provided to double check if external and PEDSEX data (often processed at
different centers) match. If a mismatch between PEDSEX and SNPSEX was
detected, while SNPSEX == Sex, PEDSEX of these individuals can optionally be
updated (fixMixup=TRUE).
<code><a href="#topic+check_sex">check_sex</a></code> depicts the X-chromosomal heterozygosity (SNPSEX) of
the individuals split by their (PEDSEX).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_sex(
  indir,
  name,
  qcdir = indir,
  maleTh = 0.8,
  femaleTh = 0.2,
  run.check_sex = TRUE,
  externalSex = NULL,
  externalFemale = "F",
  externalMale = "M",
  externalSexSex = "Sex",
  externalSexID = "IID",
  fixMixup = FALSE,
  interactive = FALSE,
  verbose = FALSE,
  label_fail = TRUE,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  highlight_legend = FALSE,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_sex_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam and name.sexcheck.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to save name.sexcheck as returned
by plink &ndash;check-sex. Per default qcdir=indir. If run.check_sex is FALSE, it
is assumed that plink &ndash;check-sex has been run and qcdir/name.sexcheck is
present. User needs writing permission to qcdir.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_maleth">maleTh</code></td>
<td>
<p>[double] Threshold of X-chromosomal heterozygosity rate for
males.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_femaleth">femaleTh</code></td>
<td>
<p>[double] Threshold of X-chromosomal heterozygosity rate for
females.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_run.check_sex">run.check_sex</code></td>
<td>
<p>[logical] Should plink &ndash;check-sex be run? if set to
FALSE, it is assumed that plink &ndash;check-sex has been run and
qcdir/name.sexcheck is present; <code><a href="#topic+check_sex">check_sex</a></code>  will fail with
missing file error otherwise.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_externalsex">externalSex</code></td>
<td>
<p>[data.frame, optional] Dataframe with sample IDs
[externalSexID] and sex [externalSexSex] to double check if external and
PEDSEX data (often processed at different centers) match.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_externalfemale">externalFemale</code></td>
<td>
<p>[integer/character] Identifier for 'female' in
externalSex.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_externalmale">externalMale</code></td>
<td>
<p>[integer/character] Identifier for 'male' in externalSex.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_externalsexsex">externalSexSex</code></td>
<td>
<p>[character] Column identifier for column containing sex
information in externalSex.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_externalsexid">externalSexID</code></td>
<td>
<p>[character] Column identifier for column containing ID
information in externalSex.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_fixmixup">fixMixup</code></td>
<td>
<p>[logical] Should PEDSEX of individuals with mismatch between
PEDSEX and Sex while Sex==SNPSEX automatically corrected: this will directly
change the name.bim/.bed/.fam files!</p>
</td></tr>
<tr><td><code id="check_sex_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_sexcheck) via ggplot2::ggsave(p=p_sexcheck,
other_arguments) or pdf(outfile) print(p_sexcheck) dev.off().</p>
</td></tr>
<tr><td><code id="check_sex_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_label_fail">label_fail</code></td>
<td>
<p>[logical] Set TRUE, to add fail IDs as text labels in
scatter plot.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_sexcheck); all highlight_samples IIDs have to
be present in the IIDs of the name.fam file.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified. Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="check_sex_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="check_sex_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="check_sex_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_sex_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="check_sex_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+check_sex">check_sex</a></code> wraps around <code><a href="#topic+run_check_sex">run_check_sex</a></code>  and
<code><a href="#topic+evaluate_check_sex">evaluate_check_sex</a></code>. If run.check_sex is TRUE,
<code><a href="#topic+run_check_sex">run_check_sex</a></code> is executed ; otherwise it is assumed that plink
&ndash;check-sex has been run externally and qcdir/name.sexcheck exists.
<code><a href="#topic+check_sex">check_sex</a></code>  will fail with missing file error otherwise.
</p>
<p>For details on the output data.frame fail_sex, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#sexcheck">https://www.cog-genomics.org/plink/1.9/formats#sexcheck</a>.
</p>


<h3>Value</h3>

<p>Named list with i) fail_sex: [data.frame] with FID, IID, PEDSEX,
SNPSEX and Sex (if externalSex was provided) of individuals failing sex
check, ii) mixup: dataframe with FID, IID, PEDSEX, SNPSEX and Sex (if
externalSex was provided) of individuals whose PEDSEX != Sex and Sex ==
SNPSEX and iii) p_sexcheck, a ggplot2-object 'containing' a scatter plot of
the X-chromosomal heterozygosity (SNPSEX) of the sample split by their
(PEDSEX), which can be shown by print(p_sexcheck).
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- "data"

# whole dataset
fail_sex &lt;- check_sex(indir=indir, name=name, run.check_sex=FALSE,
interactive=FALSE, verbose=FALSE)

# subset of dataset with sample highlighting
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
fail_sex &lt;- check_sex(indir=indir, name=name,
interactive=FALSE, path2plink=path2plink,
remove_individuals=remove_individuals_file,
highlight_samples=highlight_samples[,2], highlight_type = c("text", "shape"))

## End(Not run)
</code></pre>

<hr>
<h2 id='check_snp_missingness'>Identification of SNPs with high missingness rate</h2><span id='topic+check_snp_missingness'></span>

<h3>Description</h3>

<p>Runs and evaluates results from plink &ndash;missing &ndash;freq. It calculate the
rates of missing genotype calls and frequency for all variants in the
individuals that passed the <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>. The SNP
missingness rates (stratified by minor allele frequency) are depicted as
histograms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_snp_missingness(
  indir,
  name,
  qcdir = indir,
  lmissTh = 0.01,
  interactive = FALSE,
  path2plink = NULL,
  verbose = FALSE,
  showPlinkOutput = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_snp_missingness_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> was conducted, this directory should be the
same as qcdir specified in <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_lmissth">lmissTh</code></td>
<td>
<p>[double] Threshold for acceptable variant missing rate across
samples.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_lmiss) via ggplot2::ggsave(p=p_lmiss,
other_arguments) or pdf(outfile) print(p_lmiss) dev.off().</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out
and specifically, if TRUE, plink log will be displayed.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="check_snp_missingness_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>check_snp_missingness</code> uses plink &ndash;remove name.fail.IDs &ndash;missing
&ndash;freq to calculate rates of missing genotype calls and frequency per SNP in
the individuals that passed the <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>. It does so
without generating a new dataset but simply removes the IDs when calculating
the statistics.
</p>
<p>For details on the output data.frame fail_missingness, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#lmiss">https://www.cog-genomics.org/plink/1.9/formats#lmiss</a>.
</p>


<h3>Value</h3>

<p>Named list with i) fail_missingness containing a [data.frame] with
CHR (Chromosome code), SNP (Variant identifier), CLST (Cluster identifier.
Only present with &ndash;within/&ndash;family), N_MISS (Number of missing genotype
call(s), not counting obligatory missings), N_CLST  (Cluster size; does not
include nonmales on Ychr; Only present with &ndash;within/&ndash;family), N_GENO
(Number of potentially valid call(s)), F_MISS (Missing call rate) for all
SNPs failing the lmissTh and ii) p_lmiss, a ggplot2-object 'containing' the
SNP missingness histogram which can be shown by (print(p_lmiss)).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# run on all individuals and markers
fail_snp_missingness &lt;- check_snp_missingness(qcdir=qcdir, indir=indir,
name=name, interactive=FALSE, verbose=TRUE, path2plink=path2plink)

# run on subset of individuals and markers
keep_individuals_file &lt;- system.file("extdata", "keep_individuals",
package="plinkQC")
extract_markers_file &lt;- system.file("extdata", "extract_markers",
package="plinkQC")
fail_snp_missingness &lt;- check_snp_missingness(qcdir=qcdir, indir=indir,
name=name, interactive=FALSE, verbose=TRUE, path2plink=path2plink,
keep_individuals=keep_individuals_file, extract_markers=extract_markers_file)

## End(Not run)
</code></pre>

<hr>
<h2 id='checkFiltering'>Check and construct PLINK sample and marker filters</h2><span id='topic+checkFiltering'></span>

<h3>Description</h3>

<p>checkFiltering checks that the file names with the individuals and markers to
be filtered can be found. If so, it constructs the command for filtering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkFiltering(
  keep_individuals = NULL,
  remove_individuals = NULL,
  extract_markers = NULL,
  exclude_markers = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkFiltering_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="checkFiltering_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="checkFiltering_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="checkFiltering_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector containing args in sys::exec_wait format to enable filtering
on individuals and/or markers.
</p>

<hr>
<h2 id='checkPlink'>Check PLINK software access</h2><span id='topic+checkPlink'></span>

<h3>Description</h3>

<p>checkPlink checks that the PLINK software
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>)
can be found from system call.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkPlink(path2plink = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkPlink_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to PLINK executable.
</p>

<hr>
<h2 id='checkRemoveIDs'>Check and construct individual IDs to be removed</h2><span id='topic+checkRemoveIDs'></span>

<h3>Description</h3>

<p>checkRemoveIDs checks that the file names with the individuals to be filtered
can be found. It reads the corresponding files, combines the selected
individuals into one data.frame and compares these to all individuals in
the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkRemoveIDs(prefix, remove_individuals = NULL, keep_individuals)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkRemoveIDs_+3A_prefix">prefix</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. path/2/name.bed,
path/2/name.bim and path/2/name.fam.</p>
</td></tr>
<tr><td><code id="checkRemoveIDs_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="checkRemoveIDs_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame containing family (FID) and individual (IID) IDs of
individuals to be removed from analysis.
</p>

<hr>
<h2 id='cleanData'>Create plink dataset with individuals and markers passing quality control</h2><span id='topic+cleanData'></span>

<h3>Description</h3>

<p>Individuals that fail per-individual QC and markers that fail
per-marker QC are removed from indir/name.bim/.bed/.fam and a new, dataset
with the remaining individuals and markers is created as
qcdir/name.clean.bim/.bed/.fam.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cleanData(
  indir,
  name,
  qcdir = indir,
  filterSex = TRUE,
  filterHeterozygosity = TRUE,
  filterSampleMissingness = TRUE,
  filterAncestry = TRUE,
  filterRelated = TRUE,
  filterSNPMissingness = TRUE,
  lmissTh = 0.01,
  filterHWE = TRUE,
  hweTh = 1e-05,
  filterMAF = TRUE,
  macTh = 20,
  mafTh = NULL,
  path2plink = NULL,
  verbose = FALSE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cleanData_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> was conducted, this directory should be the
same as qcdir specified in <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filtersex">filterSex</code></td>
<td>
<p>[logical] Set to exclude samples that failed the sex
check (via <code><a href="#topic+check_sex">check_sex</a></code> or <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>).
Requires file qcdir/name.fail-sexcheck.IDs (automatically created by
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code> if do.evaluate_check_sex set to TRUE).</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filterheterozygosity">filterHeterozygosity</code></td>
<td>
<p>[logical] Set to exclude samples that failed
check for outlying heterozygosity rates (via
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code> or
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code>). Requires file qcdir/name.fail-het.IDs
(automatically created by <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> if
do.evaluate_check_het_and_miss set to TRUE).</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filtersamplemissingness">filterSampleMissingness</code></td>
<td>
<p>[logical] Set to exclude samples that failed
check for excessive missing genotype rates (via
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code> or
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code>). Requires file qcdir/name.fail-imiss.IDs
(automatically created by <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> if
do.evaluate_check_het_and_miss set to TRUE).</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filterancestry">filterAncestry</code></td>
<td>
<p>[logical] Set to exclude samples that failed ancestry
check (via <code><a href="#topic+check_ancestry">check_ancestry</a></code> or <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>).
Requires file qcdir/name.fail-ancestry.IDs (automatically created by
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code> if do.check_ancestry set to TRUE).</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filterrelated">filterRelated</code></td>
<td>
<p>[logical] Set to exclude samples that failed relatedness
check (via <code><a href="#topic+check_relatedness">check_relatedness</a></code> or <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>).
Requires file qcdir/name.fail-IBD.IDs (automatically created by
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code> if do.evaluate_check_relatedness set to TRUE).</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filtersnpmissingness">filterSNPMissingness</code></td>
<td>
<p>[logical] Set to exclude markers that have
excessive missing rates across samples (via
<code><a href="#topic+check_snp_missingness">check_snp_missingness</a></code> or <code><a href="#topic+perMarkerQC">perMarkerQC</a></code>). Requires
lmissTh to be set.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_lmissth">lmissTh</code></td>
<td>
<p>[double] Threshold for acceptable variant missing rate across
samples.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filterhwe">filterHWE</code></td>
<td>
<p>[logical] Set to exclude markers that fail HWE exact test
(via <code><a href="#topic+check_hwe">check_hwe</a></code> or <code><a href="#topic+perMarkerQC">perMarkerQC</a></code>). Requires hweTh to
be set.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_hweth">hweTh</code></td>
<td>
<p>[double] Significance threshold for deviation from HWE.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_filtermaf">filterMAF</code></td>
<td>
<p>[logical] Set to exclude markers that fail minor allele
frequency or minor allele count threshold (via <code><a href="#topic+check_maf">check_maf</a></code> or
<code><a href="#topic+perMarkerQC">perMarkerQC</a></code>). Requires mafTh or macTh to be set.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_macth">macTh</code></td>
<td>
<p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples).</p>
</td></tr>
<tr><td><code id="cleanData_+3A_mafth">mafTh</code></td>
<td>
<p>[double] Threshold for minor allele frequency cut-off.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="cleanData_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="cleanData_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>names [list] with i) passIDs, containing a [data.frame] with family
[FID] and individual [IID] IDs of samples that pass the QC, ii) failIDs,
containing a [data.frame] with family [FID] and individual [IID] IDs of
samples that fail the QC.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>package.dir &lt;- find.package('plinkQC')
indir &lt;- file.path(package.dir, 'extdata')
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# Run qc on all samples and markers in the dataset
## Run individual QC checks
fail_individuals &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE, verbose=FALSE,
path2plink=path2plink)

## Run marker QC checks
fail_markers &lt;- perMarkerQC(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

## Create new dataset of individuals and markers passing QC
ids_all &lt;- cleanData(indir=indir, qcdir=qcdir, name=name, macTh=15,
verbose=TRUE, path2plink=path2plink, filterAncestry=FALSE,
filterRelated=TRUE)

# Run qc on subset of samples and markers in the dataset
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")

fail_individuals &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
dont.check_ancestry = TRUE, interactive=FALSE, verbose=FALSE,
highlight_samples = highlight_samples[,2], highlight_type = "label",
remove_individuals = remove_individuals_file, path2plink=path2plink)

## Run marker QC checks
fail_markers &lt;- perMarkerQC(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

## Create new dataset of individuals and markers passing QC
ids_all &lt;- cleanData(indir=indir, qcdir=qcdir, name=name, macTh=15,
verbose=TRUE, path2plink=path2plink, filterAncestry=FALSE,
remove_individuals = remove_individuals_file)

## End(Not run)
</code></pre>

<hr>
<h2 id='evaluate_check_ancestry'>Evaluate results from PLINK PCA on combined study and reference data</h2><span id='topic+evaluate_check_ancestry'></span>

<h3>Description</h3>

<p>Evaluates and depicts results of plink &ndash;pca (via
<code><a href="#topic+run_check_ancestry">run_check_ancestry</a></code> or externally conducted pca) on merged
genotypes from individuals to be QCed and individuals of reference population
of known genotypes. Currently, check ancestry only supports automatic
selection of individuals of European descent. It uses information from
principal components 1 and 2 returned by plink &ndash;pca to find the center of
the European reference samples (mean(PC1_europeanRef), mean(PC2_europeanRef).
It computes the maximum Euclidean distance (maxDist) of the European
reference samples from this centre. All study samples whose Euclidean
distance from the centre falls outside the circle described by the radius
r=europeanTh* maxDist are considered non-European and their IDs are returned
as failing the ancestry check.
check_ancestry creates a scatter plot of PC1 versus PC2 colour-coded for
samples of the reference populations and the study population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluate_check_ancestry(
  indir,
  name,
  prefixMergedDataset,
  qcdir = indir,
  europeanTh = 1.5,
  defaultRefSamples = c("HapMap", "1000Genomes"),
  refSamples = NULL,
  refColors = NULL,
  refSamplesFile = NULL,
  refColorsFile = NULL,
  refSamplesIID = "IID",
  refSamplesPop = "Pop",
  refColorsColor = "Color",
  refColorsPop = "Pop",
  studyColor = "#2c7bb6",
  refPopulation = c("CEU", "TSI"),
  legend_labels_per_row = 6,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  highlight_legend = FALSE,
  interactive = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluate_check_ancestry_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_prefixmergeddataset">prefixMergedDataset</code></td>
<td>
<p>[character] Prefix of merged dataset (study and
reference samples) used in plink &ndash;pca, resulting in
prefixMergedDataset.eigenvec.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory/with/QC/results containing
prefixMergedDataset.eigenvec results as returned by plink &ndash;pca. Per default
qcdir=indir.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_europeanth">europeanTh</code></td>
<td>
<p>[double] Scaling factor of radius to be drawn around center
of European reference samples, with study samples inside this radius
considered to be of European descent and samples outside this radius of
non-European descent. The radius is computed as the maximum Euclidean distance
of European reference samples to the centre of European reference samples.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_defaultrefsamples">defaultRefSamples</code></td>
<td>
<p>[character] Option to use pre-downloaded individual
and population identifiers from either the 1000Genomes or HapMap project.
If refSamples and refSamplesFile are not provided, the HapMap identifiers (or
1000Genomes is specified) will be used as default and the function will fail
if the reference samples in the prefixMergedDataset do not match these
reference samples. If refColors and refColorsFile are not provided, this also
sets default colors for the reference populations.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refsamples">refSamples</code></td>
<td>
<p>[data.frame] Dataframe with sample identifiers
[refSamplesIID] corresponding to IIDs in prefixMergedDataset.eigenvec and
population identifier [refSamplesPop] corresponding to population IDs
[refColorsPop] in refColorsfile/refColors. If refSamples and
refSamplesFile are not specified, defaultRefSamples will be used as
reference.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refcolors">refColors</code></td>
<td>
<p>[data.frame, optional] Dataframe with population IDs in
column [refColorsPop] and corresponding colour-code for PCA plot in column
[refColorsColor].  If refColors and refColorsFile are not specified and
refSamples and refSamplesFile are not specified, default colors will be
determined from the defaultRefSamples option. If refColors and refColorsFile
are not specified and but refSamples or refSamplesFile are given, ggplot
default colors will be used.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refsamplesfile">refSamplesFile</code></td>
<td>
<p>[character] /path/to/File/with/reference samples. Needs
columns with sample identifiers [refSamplesIID] corresponding to IIDs in
prefixMergedDataset.eigenvec and population identifier [refSamplesPop]
corresponding to population IDs [refColorsPop] in refColorsfile/refColors. If
both refSamplesFile and refSamples are not NULL, defaultRefSamples information
is used.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refcolorsfile">refColorsFile</code></td>
<td>
<p>[character, optional]
/path/to/File/with/Population/Colors containing population IDs in column
[refColorsPop] and corresponding colour-code for PCA plot in column
[refColorsColor]. If refColors and refColorsFile are not specified and
refSamples and refSamplesFile are not specified, default colors will be
determined from the defaultRefSamples option. If refColors and refColorsFile
are not specified and but refSamples or refSamplesFile are given, ggplot
default colors will be used.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refsamplesiid">refSamplesIID</code></td>
<td>
<p>[character] Column name of reference sample IDs in
refSamples/refSamplesFile.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refsamplespop">refSamplesPop</code></td>
<td>
<p>[character] Column name of reference sample population
IDs in refSamples/refSamplesFile.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refcolorscolor">refColorsColor</code></td>
<td>
<p>[character] Column name of population colors in
refColors/refColorsFile</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refcolorspop">refColorsPop</code></td>
<td>
<p>[character] Column name of reference sample population
IDs in refColors/refColorsFile.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_studycolor">studyColor</code></td>
<td>
<p>[character] Colour to be used for study population if plot
is TRUE.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_refpopulation">refPopulation</code></td>
<td>
<p>[vector] Vector with population identifiers of European
reference population. Identifiers have to be corresponding to population IDs
[refColorsPop] in refColorsfile/refColors.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_legend_labels_per_row">legend_labels_per_row</code></td>
<td>
<p>[integer] Number of population names per row in
PCA plot.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_ancestry); all highlight_samples IIDs have to
be present in the IIDs of the prefixMergedDataset.fam file.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified.Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_ancestry) via ggplot2::ggsave(p=p_ancestry,
other_arguments) or pdf(outfile) print(p_ancestry) dev.off().</p>
</td></tr>
<tr><td><code id="evaluate_check_ancestry_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code><a href="#topic+run_check_ancestry">run_check_ancestry</a></code> and
<code><a href="#topic+evaluate_check_ancestry">evaluate_check_ancestry</a></code> can simply be invoked by
<code><a href="#topic+check_ancestry">check_ancestry</a></code>.
</p>
<p>1000 Genomes samples were downloaded from
<a href="https://www.internationalgenome.org/category/sample/">https://www.internationalgenome.org/category/sample/</a>, HapMap Phase 3
samples were downloaded from
<a href="https://www.broadinstitute.org/medical-and-population-genetics/hapmap-3">https://www.broadinstitute.org/medical-and-population-genetics/hapmap-3</a>.
</p>


<h3>Value</h3>

<p>Named [list] with i) fail_ancestry, containing a [data.frame] with
FID and IID of non-European individuals and ii) p_ancestry, a ggplot2-object
'containing' a scatter plot of PC1 versus PC2 colour-coded for samples of the
reference populations and the study population, which can be shown by
print(p_ancestry) and iii) plot_data, a data.frame with the data
visualised in p_ancestry (ii).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
qcdir &lt;- system.file("extdata", package="plinkQC")
name &lt;- "data"

# whole dataset
fail_ancestry &lt;- evaluate_check_ancestry(indir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE)

# highlight samples
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
fail_ancestry &lt;- evaluate_check_ancestry(indir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE,
highlight_samples = highlight_samples[,2],
highlight_type = c("text", "shape"))

## End(Not run)
</code></pre>

<hr>
<h2 id='evaluate_check_het_and_miss'>Evaluate results from PLINK missing genotype and heterozygosity rate check.</h2><span id='topic+evaluate_check_het_and_miss'></span>

<h3>Description</h3>

<p>Evaluates and depicts results from plink &ndash;missing (missing genotype rates
per individual) and plink &ndash;het (heterozygosity rates per individual) via
<code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code> and
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code> or externally conducted check.)
Non-systematic failures in genotyping and outlying heterozygosity (hz) rates
per individual are often proxies for DNA sample quality. Larger than expected
heterozygosity can indicate possible DNA contamination.
The mean heterozygosity in PLINK is computed as hz_mean = (N-O)/N, where
N: number of non-missing genotypes and O:observed number of homozygous
genotypes for a given individual.
Mean heterozygosity can differ between populations and SNP genotyping panels.
Within a population and genotyping panel, a reduced heterozygosity rate can
indicate inbreeding - these individuals will then be returned by
check_relatedness as individuals that fail the relatedness filters.
<code>evaluate_check_het_and_miss</code> creates a scatter plot with the
individuals' missingness rates on x-axis and their heterozygosity rates on
the y-axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluate_check_het_and_miss(
  qcdir,
  name,
  imissTh = 0.03,
  hetTh = 3,
  label_fail = TRUE,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  highlight_legend = FALSE,
  interactive = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluate_check_het_and_miss_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] path/to/directory/with/QC/results containing
name.imiss and name.het results as returned by plink &ndash;missing and plink
&ndash;het.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam, name.het and name.imiss.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_imissth">imissTh</code></td>
<td>
<p>[double] Threshold for acceptable missing genotype rate in any
individual; has to be proportion between (0,1)</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_hetth">hetTh</code></td>
<td>
<p>[double] Threshold for acceptable deviation from mean
heterozygosity in any individual. Expressed as multiples of standard
deviation of heterozygosity (het), i.e. individuals outside mean(het) +/-
hetTh*sd(het) will be returned as failing heterozygosity check; has to be
larger than 0.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_label_fail">label_fail</code></td>
<td>
<p>[logical] Set TRUE, to add fail IDs as text labels in
scatter plot.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_het_imiss); all highlight_samples IIDs have to
be present in the IIDs of the name.fam file.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified.Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="evaluate_check_het_and_miss_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_het_imiss) via ggplot2::ggsave(p=p_het_imiss
, other_arguments) or pdf(outfile) print(p_het_imiss) dev.off().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All, <code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code>,
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code> and
<code><a href="#topic+evaluate_check_het_and_miss">evaluate_check_het_and_miss</a></code> can simply be invoked by
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>.
</p>
<p>For details on the output data.frame fail_imiss and fail_het, check the
original description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#imiss">https://www.cog-genomics.org/plink/1.9/formats#imiss</a> and
<a href="https://www.cog-genomics.org/plink/1.9/formats#het">https://www.cog-genomics.org/plink/1.9/formats#het</a>
</p>


<h3>Value</h3>

<p>named [list] with i) fail_imiss dataframe containing FID (Family ID),
IID (Within-family ID), MISS_PHENO (Phenotype missing? (Y/N)), N_MISS (Number
of missing genotype call(s), not including obligatory missings), N_GENO (
Number of potentially valid call(s)), F_MISS (Missing call rate) of
individuals failing missing genotype check and ii) fail_het dataframe
containing FID (Family ID), IID (Within-family ID), O(HOM) (Observed number
of homozygotes), E(HOM) (Expected number of homozygotes), N(NM) (Number of
non-missing autosomal genotypes), F (Method-of-moments F coefficient
estimate) of individuals failing  outlying heterozygosity check; iii)
p_het_imiss, a ggplot2-object 'containing' a scatter plot with the samples'
missingness rates on x-axis and their heterozygosity rates on the y-axis,
which can be shown by print(p_het_imiss) and iv) plot_data, a data.frame with
the data visualised in p_het_imiss (iii).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qcdir &lt;- system.file("extdata", package="plinkQC")
name &lt;- "data"
## Not run: 
fail_het_miss &lt;- evaluate_check_het_and_miss(qcdir=qcdir, name=name,
interactive=FALSE)

#' # highlight samples
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
fail_het_miss &lt;- evaluate_check_het_and_miss(qcdir=qcdir, name=name,
interactive=FALSE, highlight_samples = highlight_samples[,2],
highlight_type = c("text", "color"))

## End(Not run)
</code></pre>

<hr>
<h2 id='evaluate_check_relatedness'>Evaluate results from PLINK IBD estimation.</h2><span id='topic+evaluate_check_relatedness'></span>

<h3>Description</h3>

<p>Evaluates and depicts results from plink &ndash;genome on the LD pruned dataset
(via <code><a href="#topic+run_check_relatedness">run_check_relatedness</a></code> or externally conducted IBD
estimation). plink &ndash;genome calculates identity by state (IBS) for each pair
of individuals based on the average proportion of alleles shared at genotyped
SNPs. The degree of recent shared ancestry, i.e. the identity by descent
(IBD) can be estimated from the genome-wide IBS. The proportion of IBD
between two individuals is returned by &ndash;genome as PI_HAT.
<code>evaluate_check_relatedness</code> finds pairs of samples whose proportion of
IBD is larger than the specified highIBDTh. Subsequently, for pairs of
individual that do not have additional relatives in the dataset, the
individual with the greater genotype missingness rate is selected and
returned as the individual failing the relatedness check. For more complex
family structures, the unrelated individuals per family are selected (e.g. in
a parents-offspring trio, the offspring will be marked as fail, while the
parents will be kept in the analysis).
<code>evaluate_check_relatedness</code> depicts all pair-wise IBD-estimates as
histograms stratified by value of PI_HAT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluate_check_relatedness(
  qcdir,
  name,
  highIBDTh = 0.1875,
  imissTh = 0.03,
  interactive = FALSE,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluate_check_relatedness_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] path/to/directory/with/QC/results containing
name.imiss and name.genome results as returned by plink &ndash;missing and plink
&ndash;genome.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam, name.genome and name.imiss.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_highibdth">highIBDTh</code></td>
<td>
<p>[double] Threshold for acceptable proportion of IBD between
pair of individuals.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_imissth">imissTh</code></td>
<td>
<p>[double] Threshold for acceptable missing genotype rate in any
individual; has to be proportion between (0,1)</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_IBD() via ggplot2::ggsave(p=p_IBD,
other_arguments) or pdf(outfile) print(p_IBD) dev.off().</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="evaluate_check_relatedness_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code><a href="#topic+run_check_relatedness">run_check_relatedness</a></code> and
<code><a href="#topic+evaluate_check_relatedness">evaluate_check_relatedness</a></code> can simply be invoked by
<code><a href="#topic+check_relatedness">check_relatedness</a></code>.
</p>
<p>For details on the output data.frame fail_high_IBD, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#genome">https://www.cog-genomics.org/plink/1.9/formats#genome</a>.
</p>


<h3>Value</h3>

<p>a named [list] with i) fail_high_IBD containing a [data.frame] of
IIDs and FIDs of individuals who fail the IBDTh in columns
FID1 and IID1. In addition, the following columns are returned (as originally
obtained by plink &ndash;genome):
FID2 (Family ID for second sample), IID2 (Individual ID for second sample),
RT (Relationship type inferred from .fam/.ped file), EZ (IBD sharing expected
value, based on just .fam/.ped relationship), Z0 (P(IBD=0)), Z1 (P(IBD=1)),
Z2 (P(IBD=2)), PI_HAT (Proportion IBD, i.e. P(IBD=2) + 0.5*P(IBD=1)), PHE
(Pairwise phenotypic code (1, 0, -1 = AA, AU, and UU pairs, respectively)),
DST (IBS distance, i.e. (IBS2 + 0.5*IBS1) / (IBS0 + IBS1 + IBS2)), PPC (IBS
binomial test), RATIO (HETHET : IBS0 SNP ratio (expected value 2)).
and ii) failIDs containing a [data.frame] with individual IDs [IID] and
family IDs [FID] of individuals failing the highIBDTh; iii) p_IBD, a
ggplot2-object 'containing' all pair-wise IBD-estimates as histograms
stratified by value of PI_HAT, which can be shown by print(p_IBD and iv)
plot_data, a data.frame with the data visualised in p_IBD (iii).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qcdir &lt;- system.file("extdata", package="plinkQC")
name &lt;- 'data'
## Not run: 
relatednessQC &lt;- evaluate_check_relatedness(qcdir=qcdir, name=name,
interactive=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='evaluate_check_sex'>Evaluate results from PLINK sex check.</h2><span id='topic+evaluate_check_sex'></span>

<h3>Description</h3>

<p>Evaluates and depicts results from plink &ndash;check-sex (via
<code><a href="#topic+run_check_sex">run_check_sex</a></code> or externally conducted sex check).
Takes file qcdir/name.sexcheck and returns IIDs for samples whose
SNPSEX != PEDSEX (where the SNPSEX is determined by the heterozygosity rate
across X-chromosomal variants).
Mismatching SNPSEX and PEDSEX IDs can indicate plating errors, sample-mixup
or generally samples with poor genotyping. In the latter case, these IDs are
likely to fail other QC steps as well.
Optionally, an extra data.frame (externalSex) with sample IDs and sex can be
provided to double check if external and PEDSEX data (often processed at
different centers) match. If a mismatch between PEDSEX and SNPSEX was
detected while SNPSEX == Sex, PEDSEX of these individuals can optionally be
updated (fixMixup=TRUE).
<code>evaluate_check_sex</code> depicts the X-chromosomal heterozygosity (SNPSEX)
of the samples split by their (PEDSEX).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluate_check_sex(
  qcdir,
  name,
  maleTh = 0.8,
  femaleTh = 0.2,
  externalSex = NULL,
  fixMixup = FALSE,
  indir = qcdir,
  externalFemale = "F",
  externalMale = "M",
  externalSexSex = "Sex",
  externalSexID = "IID",
  verbose = FALSE,
  label_fail = TRUE,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  highlight_legend = FALSE,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE,
  interactive = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluate_check_sex_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory containing name.sexcheck as
returned by plink &ndash;check-sex.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam and name.sexcheck.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_maleth">maleTh</code></td>
<td>
<p>[double] Threshold of X-chromosomal heterozygosity rate for
males.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_femaleth">femaleTh</code></td>
<td>
<p>[double] Threshold of X-chromosomal heterozygosity rate for
females.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_externalsex">externalSex</code></td>
<td>
<p>[data.frame, optional] with sample IDs [externalSexID] and
sex [externalSexSex] to double check if external and PEDSEX data (often
processed at different centers) match.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_fixmixup">fixMixup</code></td>
<td>
<p>[logical] Should PEDSEX of individuals with mismatch between
PEDSEX and Sex, with Sex==SNPSEX automatically corrected: this will directly
change the name.bim/.bed/.fam files!</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files; only required of fixMixup==TRUE.
User needs writing permission to indir.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_externalfemale">externalFemale</code></td>
<td>
<p>[integer/character] Identifier for 'female' in
externalSex.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_externalmale">externalMale</code></td>
<td>
<p>[integer/character] Identifier for 'male' in externalSex.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_externalsexsex">externalSexSex</code></td>
<td>
<p>[character] Column identifier for column containing sex
information in externalSex.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_externalsexid">externalSexID</code></td>
<td>
<p>[character] Column identifier for column containing ID
information in externalSex.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_label_fail">label_fail</code></td>
<td>
<p>[logical] Set TRUE, to add fail IDs as text labels in
scatter plot.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_sexcheck); all highlight_samples IIDs have to
be present in the IIDs of the name.fam file.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified. Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
<tr><td><code id="evaluate_check_sex_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_sexcheck) via ggplot2::ggsave(p=p_sexcheck,
other_arguments) or pdf(outfile) print(p_sexcheck) dev.off().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code><a href="#topic+run_check_sex">run_check_sex</a></code> and <code><a href="#topic+evaluate_check_sex">evaluate_check_sex</a></code> can
simply be invoked by <code><a href="#topic+check_sex">check_sex</a></code>.
</p>
<p>For details on the output data.frame fail_sex, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#sexcheck">https://www.cog-genomics.org/plink/1.9/formats#sexcheck</a>.
</p>


<h3>Value</h3>

<p>named list with i) fail_sex: dataframe with FID, IID, PEDSEX, SNPSEX
and Sex (if externalSex was provided) of individuals failing sex check;
ii) mixup: dataframe with FID, IID, PEDSEX, SNPSEX and Sex (if externalSex
was provided) of individuals whose PEDSEX != Sex and Sex == SNPSEX; iii)
p_sexcheck, a ggplot2-object 'containing' a scatter plot of the X-chromosomal
heterozygosity (SNPSEX) of the individuals split by their (PEDSEX), which can
be shown by print(p_sexcheck) and iv) plot_data, a data.frame with the data
visualised in p_sexcheck (iii).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qcdir &lt;- system.file("extdata", package="plinkQC")
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
## Not run: 
fail_sex &lt;- evaluate_check_sex(qcdir=qcdir, name=name, interactive=FALSE,
verbose=FALSE, path2plink=path2plink)

# highlight samples
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
fail_sex &lt;- evaluate_check_sex(qcdir=qcdir, name=name, interactive=FALSE,
verbose=FALSE, path2plink=path2plink,
highlight_samples = highlight_samples[,2],
highlight_type = c("label", "color"), highlight_color = "darkgreen")

## End(Not run)
</code></pre>

<hr>
<h2 id='overviewPerIndividualQC'>Overview of per sample QC</h2><span id='topic+overviewPerIndividualQC'></span>

<h3>Description</h3>

<p><code>overviewPerIndividualQC</code> depicts results of
<code><a href="#topic+perIndividualQC">perIndividualQC</a></code> as intersection plots
(via <code><a href="UpSetR.html#topic+upset">upset</a></code>) and returns dataframes indicating
which QC checks individuals failed or passed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overviewPerIndividualQC(results_perIndividualQC, interactive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overviewPerIndividualQC_+3A_results_perindividualqc">results_perIndividualQC</code></td>
<td>
<p>[list] Output of <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>
i.e. named [list]
with i) sample_missingness containing a [vector] with sample IIDs failing
the selected missingness threshold imissTh, ii) highIBD containing
a [vector] with sample IIDs failing the selected relatedness threshold
highIBDTh, iii) outlying_heterozygosity containing a [vector] with sample
IIDs failing selected the heterozygosity threshold hetTh, iv) mismatched_sex
containing a [vector] with the sample IIDs failing the sexcheck based on
SNPSEX and selected femaleTh/maleTh, v) ancestry containing a vector with
sample IIDs failing the ancestry check based on the selected europeanTh and
vi) p_sampleQC, a ggplot2-object 'containing' a sub-paneled plot with the
QC-plots of <code><a href="#topic+check_sex">check_sex</a></code>,
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>,
<code><a href="#topic+check_relatedness">check_relatedness</a></code> and <code><a href="#topic+check_ancestry">check_ancestry</a></code>.</p>
</td></tr>
<tr><td><code id="overviewPerIndividualQC_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_overview) via ggplot2::ggsave(p=p_overview,
other_arguments) or pdf(outfile) print(p_overview) dev.off().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named [list] with i) nr_fail_samples: total number of samples
[integer] failing perIndividualQC, ii) fail_QC containing a [data.frame] with
samples that failed QC steps (excluding ancestry) with IID, FID,
all QC steps applied by perIndividualQC (max=4), with entries=0
if passing the QC and entries=1 if failing that particular QC and iii)
fail_QC_and_ancestry containing a [data.frame] with samples that failed
ancestry and QC checks with IID, FID, QC_fail and
Ancestry_fail, with entries=0 if passing and entries=1 if failing that check,
iii) p_overview, a ggplot2-object 'containing' a sub-paneled plot with the
QC-plots.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
## Not run: 
fail_individuals &lt;- perIndividualQC(qcdir=qcdir, indir=indir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE, verbose=FALSE,
do.run_check_het_and_miss=FALSE, do.run_check_relatedness=FALSE,
do.run_check_sex=FALSE, do.run_check_ancestry=FALSE)

overview &lt;- overviewPerIndividualQC(fail_individuals)

## End(Not run)
</code></pre>

<hr>
<h2 id='overviewPerMarkerQC'>Overview of per marker QC</h2><span id='topic+overviewPerMarkerQC'></span>

<h3>Description</h3>

<p><code>overviewPerMarkerQC</code> depicts results of <code><a href="#topic+perMarkerQC">perMarkerQC</a></code> as
an intersection plot (via <code><a href="UpSetR.html#topic+upset">upset</a></code>) and returns a
dataframe indicating which QC checks were failed or passed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overviewPerMarkerQC(results_perMarkerQC, interactive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overviewPerMarkerQC_+3A_results_permarkerqc">results_perMarkerQC</code></td>
<td>
<p>[list] Output of <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>
i.e. named [list] with i) fail_list, a named [list] with 1.
SNP_missingness, containing SNP IDs failing the missingness threshold
lmissTh, 2. hwe, containing SNP IDs failing the HWE exact test threshold
hweTh and 3. maf, containing SNPs failing the MAF threshold mafTh/MAC
threshold macTh and ii) p_markerQC, a ggplot2-object 'containing' a
sub-paneled plot with the QC-plots of <code><a href="#topic+check_snp_missingness">check_snp_missingness</a></code>,
<code><a href="#topic+check_hwe">check_hwe</a></code> and <code><a href="#topic+check_maf">check_maf</a></code></p>
</td></tr>
<tr><td><code id="overviewPerMarkerQC_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_overview) via ggplot2::ggsave(p=p_overview,
other_arguments) or pdf(outfile) print(p_overview) dev.off().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named [list] with i) nr_fail_markers: total number of markers
[integer] failing perMarkerQC, ii) fail_QC containing a [data.frame] with
markers that failed QC steps: marker rsIDs in rows,
columns are all QC steps applied by perMarkerQC (max=3), with entries=0 if
passing the QC and entries=1 if failing that particular QC.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
# All quality control checks
## Not run: 
fail_markers &lt;- perMarkerQC(qcdir=qcdir, indir=indir, name=name,
interactive=FALSE, verbose=TRUE, path2plink=path2plink)
overview &lt;- overviewPerMarkerQC(fail_markers)

## End(Not run)
</code></pre>

<hr>
<h2 id='perIndividualQC'>Quality control for all individuals in plink-dataset</h2><span id='topic+perIndividualQC'></span>

<h3>Description</h3>

<p>perIndividualQC checks the samples in the plink dataset for their total
missingness and heterozygosity rates, the concordance of their assigned sex
to their SNP sex, their relatedness to other study individuals and their
genetic ancestry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perIndividualQC(
  indir,
  name,
  qcdir = indir,
  dont.check_sex = FALSE,
  do.run_check_sex = TRUE,
  do.evaluate_check_sex = TRUE,
  maleTh = 0.8,
  femaleTh = 0.2,
  externalSex = NULL,
  externalMale = "M",
  externalSexSex = "Sex",
  externalSexID = "IID",
  externalFemale = "F",
  fixMixup = FALSE,
  dont.check_het_and_miss = FALSE,
  do.run_check_het_and_miss = TRUE,
  do.evaluate_check_het_and_miss = TRUE,
  imissTh = 0.03,
  hetTh = 3,
  dont.check_relatedness = FALSE,
  do.run_check_relatedness = TRUE,
  do.evaluate_check_relatedness = TRUE,
  highIBDTh = 0.1875,
  mafThRelatedness = 0.1,
  filter_high_ldregion = TRUE,
  high_ldregion_file = NULL,
  genomebuild = "hg19",
  dont.check_ancestry = FALSE,
  do.run_check_ancestry = TRUE,
  do.evaluate_check_ancestry = TRUE,
  prefixMergedDataset,
  europeanTh = 1.5,
  defaultRefSamples = c("HapMap", "1000Genomes"),
  refSamples = NULL,
  refColors = NULL,
  refSamplesFile = NULL,
  refColorsFile = NULL,
  refSamplesIID = "IID",
  refSamplesPop = "Pop",
  refColorsColor = "Color",
  refColorsPop = "Pop",
  studyColor = "#2c7bb6",
  label_fail = TRUE,
  highlight_samples = NULL,
  highlight_type = c("text", "label", "color", "shape"),
  highlight_text_size = 3,
  highlight_color = "#c51b8a",
  highlight_shape = 17,
  highlight_legend = FALSE,
  interactive = FALSE,
  verbose = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  subplot_label_size = 9,
  title_size = 9,
  path2plink = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perIndividualQC_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be saved.
Per default, qcdir=indir. If do.evaluate_[analysis] is set to TRUE and
do.run_[analysis] is FALSE, <code>perIndividualQC</code> expects the
analysis-specific plink output files in qcdir i.e. do.check_sex expects
name.sexcheck, do.evaluate_check_het_and_miss expects name.het and name.imiss,
do.evaluate_check_relatedness expects name.genome and name.imiss and
do.evaluate_check_ancestry expects prefixMergeData.eigenvec. If these files
are not present <code>perIndividualQC</code> will fail with missing file error.
Setting do.run_[analysis] TRUE will execute the checks and create the
required files. User needs writing permission to qcdir.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_dont.check_sex">dont.check_sex</code></td>
<td>
<p>[logical] If TRUE, no sex check will be conducted;
short for do.run_check_sex=FALSE and do.evaluate_check_sex=FALSE.
Takes precedence over do.run_check_sex and do.evaluate_check_sex.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.run_check_sex">do.run_check_sex</code></td>
<td>
<p>[logical] If TRUE, run <code><a href="#topic+run_check_sex">run_check_sex</a></code></p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.evaluate_check_sex">do.evaluate_check_sex</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+evaluate_check_sex">evaluate_check_sex</a></code></p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_maleth">maleTh</code></td>
<td>
<p>[double] Threshold of X-chromosomal heterozygosity rate for
males.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_femaleth">femaleTh</code></td>
<td>
<p>[double] Threshold of X-chromosomal heterozygosity rate for
females.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_externalsex">externalSex</code></td>
<td>
<p>[data.frame, optional] Dataframe with sample IDs
[externalSexID] and sex [externalSexSex] to double check if external and
PEDSEX data (often processed at different centers) match.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_externalmale">externalMale</code></td>
<td>
<p>[integer/character] Identifier for 'male' in externalSex.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_externalsexsex">externalSexSex</code></td>
<td>
<p>[character] Column identifier for column containing sex
information in externalSex.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_externalsexid">externalSexID</code></td>
<td>
<p>[character] Column identifier for column containing ID
information in externalSex.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_externalfemale">externalFemale</code></td>
<td>
<p>[integer/character] Identifier for 'female' in
externalSex.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_fixmixup">fixMixup</code></td>
<td>
<p>[logical] Should PEDSEX of individuals with mismatch between
PEDSEX and Sex while Sex==SNPSEX automatically corrected: this will directly
change the name.bim/.bed/.fam files!</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_dont.check_het_and_miss">dont.check_het_and_miss</code></td>
<td>
<p>[logical] If TRUE, no heterozygosity and
missingness check will be conducted; short for
do.run_check_heterozygosity=FALSE, do.run_check_missingness=FALSE and
do.evaluate_check_het_and_miss=FALSE. Takes precedence over
do.run_check_heterozygosity, do.run_check_missingness and
do.evaluate_check_het_and_miss.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.run_check_het_and_miss">do.run_check_het_and_miss</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code> and
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code></p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.evaluate_check_het_and_miss">do.evaluate_check_het_and_miss</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+evaluate_check_het_and_miss">evaluate_check_het_and_miss</a></code>.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_imissth">imissTh</code></td>
<td>
<p>[double] Threshold for acceptable missing genotype rate in any
individual; has to be proportion between (0,1)</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_hetth">hetTh</code></td>
<td>
<p>[double] Threshold for acceptable deviation from mean
heterozygosity per individual. Expressed as multiples of standard
deviation of heterozygosity (het), i.e. individuals outside mean(het) +/-
hetTh*sd(het) will be returned as failing heterozygosity check; has to be
larger than 0.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_dont.check_relatedness">dont.check_relatedness</code></td>
<td>
<p>[logical] If TRUE, no relatedness check will be
conducted; short for do.run_check_relatedness=FALSE and
do.evaluate_check_relatedness=FALSE. Takes precedence over
do.run_check_relatedness and do.evaluate_check_relatedness.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.run_check_relatedness">do.run_check_relatedness</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+run_check_relatedness">run_check_relatedness</a></code>.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.evaluate_check_relatedness">do.evaluate_check_relatedness</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+evaluate_check_relatedness">evaluate_check_relatedness</a></code>.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highibdth">highIBDTh</code></td>
<td>
<p>[double] Threshold for acceptable proportion of IBD between
pair of individuals.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_mafthrelatedness">mafThRelatedness</code></td>
<td>
<p>[double] Threshold of minor allele frequency filter
for selecting variants for IBD estimation.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_filter_high_ldregion">filter_high_ldregion</code></td>
<td>
<p>[logical] Should high LD regions be filtered
before IBD estimation; carried out per default with high LD regions for
hg19 provided as default via <code>genomebuild</code>. For alternative genome
builds not provided or non-human data, high LD regions files can be
provided via <code>high_ldregion_file</code>.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_high_ldregion_file">high_ldregion_file</code></td>
<td>
<p>[character] Path to file with high LD regions used
for filtering before IBD estimation if <code>filter_high_ldregion</code> == TRUE,
otherwise ignored; for human genome data, high LD region files are provided
and can simply be chosen via <code>genomebuild</code>. Files have to be
space-delimited, no column names with the following columns: chromosome,
region-start, region-end, region number. Chromosomes are specified without
'chr' prefix. For instance:
1 48000000 52000000 1
2 86000000 100500000 2</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_genomebuild">genomebuild</code></td>
<td>
<p>[character] Name of the genome build of the PLINK file
annotations, ie mappings in the name.bim file. Will be used to remove
high-LD regions based on the coordinates of the respective build. Options
are hg18, hg19 and hg38. See @details.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_dont.check_ancestry">dont.check_ancestry</code></td>
<td>
<p>[logical] If TRUE, no ancestry check will be
conducted; short for do.run_check_ancestry=FALSE and
do.evaluate_check_ancestry=FALSE. Takes precedence over
do.run_check_ancestry and do.evaluate_check_ancestry.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.run_check_ancestry">do.run_check_ancestry</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+run_check_ancestry">run_check_ancestry</a></code>.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_do.evaluate_check_ancestry">do.evaluate_check_ancestry</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+evaluate_check_ancestry">evaluate_check_ancestry</a></code>.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_prefixmergeddataset">prefixMergedDataset</code></td>
<td>
<p>[character] Prefix of merged dataset (study and
reference samples) used in plink &ndash;pca, resulting in
prefixMergedDataset.eigenvec.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_europeanth">europeanTh</code></td>
<td>
<p>[double] Scaling factor of radius to be drawn around center
of European reference samples, with study samples inside this radius
considered to be of European descent and samples outside this radius of
non-European descent. The radius is computed as the maximum Euclidean
distance of European reference samples to the centre of European reference
samples.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_defaultrefsamples">defaultRefSamples</code></td>
<td>
<p>[character] Option to use pre-downloaded individual
and population identifiers from either the 1000Genomes or HapMap project.
If refSamples and refSamplesFile are not provided, the HapMap identifiers (or
1000Genomes is specified) will be used as default and the function will fail
if the reference samples in the prefixMergedDataset do not match these
reference samples. If refColors and refColorsFile are not provided, this also
sets default colors for the reference populations.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refsamples">refSamples</code></td>
<td>
<p>[data.frame] Dataframe with sample identifiers
[refSamplesIID] corresponding to IIDs in prefixMergedDataset.eigenvec and
population identifier [refSamplesPop] corresponding to population IDs
[refColorsPop] in refColorsfile/refColors. Either refSamples or
refSamplesFile have to be specified.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refcolors">refColors</code></td>
<td>
<p>[data.frame, optional] Dataframe with population IDs in
column [refColorsPop] and corresponding colour-code for PCA plot in column
[refColorsColor]. If not provided and is.null(refColorsFile) default colors
are used.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refsamplesfile">refSamplesFile</code></td>
<td>
<p>[character] /path/to/File/with/reference samples. Needs
columns with sample identifiers [refSamplesIID] corresponding to IIDs in
prefixMergedDataset.eigenvec and population identifier [refSamplesPop]
corresponding to population IDs [refColorsPop] in refColorsfile/refColors.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refcolorsfile">refColorsFile</code></td>
<td>
<p>[character, optional]
/path/to/File/with/Population/Colors containing population IDs in column
[refColorsPop] and corresponding colour-code for PCA plot in column
[refColorsColor].If not provided and is.null(refColors) default colors for
are used.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refsamplesiid">refSamplesIID</code></td>
<td>
<p>[character] Column name of reference sample IDs in
refSamples/refSamplesFile.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refsamplespop">refSamplesPop</code></td>
<td>
<p>[character] Column name of reference sample population
IDs in refSamples/refSamplesFile.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refcolorscolor">refColorsColor</code></td>
<td>
<p>[character] Column name of population colors in
refColors/refColorsFile</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_refcolorspop">refColorsPop</code></td>
<td>
<p>[character] Column name of reference sample population
IDs in refColors/refColorsFile.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_studycolor">studyColor</code></td>
<td>
<p>[character] Colour to be used for study population.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_label_fail">label_fail</code></td>
<td>
<p>[logical] Set TRUE, to add fail IDs as text labels in
scatter plot.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highlight_samples">highlight_samples</code></td>
<td>
<p>[character vector] Vector of sample IIDs to
highlight in the plot (p_sexcheck); all highlight_samples IIDs have to
be present in the IIDs of the name.fam file.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highlight_type">highlight_type</code></td>
<td>
<p>[character]  Type of sample highlight, labeling
by IID (&quot;text&quot;/&quot;label&quot;) and/or highlighting data points in different &quot;color&quot;
and/or &quot;shape&quot;. &quot;text&quot; and &quot;label&quot; use ggrepel for minimal
overlap of text labels (&quot;text) or label boxes (&quot;label&quot;).  Only one of &quot;text&quot;
and &quot;label&quot; can be specified. Text/Label size can be specified with
highlight_text_size, highlight color with highlight_color,
or highlight shape with highlight_shape.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highlight_text_size">highlight_text_size</code></td>
<td>
<p>[integer] Text/Label size for samples specified to
be highlighted (highlight_samples) by &quot;text&quot; or &quot;label&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highlight_color">highlight_color</code></td>
<td>
<p>[character] Color for samples specified to
be highlighted (highlight_samples) by &quot;color&quot; (highlight_type).</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>[integer] Shape for samples specified to
be highlighted (highlight_samples) by &quot;shape&quot; (highlight_type). Possible
shapes and their encoding can be found at:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#sec:shape-spec</a></p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_highlight_legend">highlight_legend</code></td>
<td>
<p>[logical] Should a separate legend for the
highlighted samples be provided; only relevant for highlight_type == &quot;color&quot;
or highlight_type == &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_sampleQC) via ggplot2::ggsave(p=p_sampleQC,
other_arguments) or pdf(outfile) print(p_sampleQC) dev.off().
If TRUE, i) depicts the X-chromosomal heterozygosity (SNPSEX) of the samples
split by their PEDSEX (if do.evaluate_check_sex is TRUE), ii) creates a
scatter plot
with samples' missingness rates on x-axis and their heterozygosity rates on
the y-axis (if do.evaluate_check_het_and_miss is TRUE), iii) depicts
all pair-wise IBD-estimates as histogram (if do.evaluate_check_relatedness is
TRUE)
and iv) creates a scatter plot of PC1 versus PC2 color-coded for samples of
reference populations and study population (if do.check_ancestry is set to
TRUE).</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_subplot_label_size">subplot_label_size</code></td>
<td>
<p>[integer] Size of the subplot labeling.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="perIndividualQC_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>perIndividualQC wraps around the individual QC functions
<code><a href="#topic+check_sex">check_sex</a></code>, <code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>,
<code><a href="#topic+check_relatedness">check_relatedness</a></code> and <code><a href="#topic+check_ancestry">check_ancestry</a></code>. For details
on the parameters and outputs, check these function documentations. For
detailed output for fail IIDs (instead of simple IID lists), run each
function individually.
</p>


<h3>Value</h3>

<p>Named [list] with i) fail_list, a named [list] with 1.
sample_missingness containing a [vector] with sample IIDs failing the
missingness threshold imissTh, 2. highIBD containing a [vector] with sample
IIDs failing the relatedness threshold highIBDTh, 3. outlying_heterozygosity
containing a [vector] with sample IIDs failing the heterozygosity threshold
hetTh, 4. mismatched_sex containing a [vector] with the sample IIDs failing
the sexcheck based on SNPSEX and femaleTh/maleTh and 5. ancestry containing
a vector with sample IIDs failing the ancestry check based on europeanTh and
ii) p_sampleQC, a ggplot2-object 'containing' a sub-paneled plot with the
QC-plots of <code><a href="#topic+check_sex">check_sex</a></code>,
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>,
<code><a href="#topic+check_relatedness">check_relatedness</a></code> and <code><a href="#topic+check_ancestry">check_ancestry</a></code>, which can
be shown by print(p_sampleQC).
List entries contain NULL if that specific check was not chosen.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
# All quality control checks
## Not run: 
# whole dataset
fail_individuals &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE, verbose=FALSE,
do.run_check_het_and_miss=FALSE, do.run_check_relatedness=FALSE,
do.run_check_sex=FALSE, do.run_check_ancestry=FALSE)

# Only check sex and missingness/heterozygosity
fail_sex_het_miss &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
dont.check_ancestry=TRUE, dont.check_relatedness=TRUE,
interactive=FALSE, verbose=FALSE)

# subset of dataset with sample highlighting
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
individual_qc &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE, verbose=FALSE,
do.run_check_ancestry=FALSE, do.evaluate_check_ancestry=TRUE,
path2plink=path2plink,
remove_individuals=remove_individuals_file,
highlight_samples=highlight_samples[,2],
highlight_type = c("text", "color"), highlight_color="goldenrod")

## End(Not run)
</code></pre>

<hr>
<h2 id='perMarkerQC'>Quality control for all markers in plink-dataset</h2><span id='topic+perMarkerQC'></span>

<h3>Description</h3>

<p>perMarkerQC checks the markers in the plink dataset for their missingness
rates across samples, their deviation from Hardy-Weinberg-Equilibrium (HWE)
and their minor allele frequencies (MAF). Per default, it assumes that IDs of
individuals that have failed <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> have been written
to qcdir/name.fail.IDs and removes these individuals when computing
missingness rates, HWE p-values and MAF. If the qcdir/name.fail.IDs file does
not exist, a message is written to stdout but the analyses will continue for
all samples in the name.fam/name.bed/name.bim dataset.
Depicts i) SNP missingness rates (stratified by minor allele
frequency) as histograms, ii) p-values of HWE exact test (stratified by all
and low p-values) as histograms and iii) the minor allele frequency
distribution as a histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perMarkerQC(
  indir,
  qcdir = indir,
  name,
  do.check_snp_missingness = TRUE,
  lmissTh = 0.01,
  do.check_hwe = TRUE,
  hweTh = 1e-05,
  do.check_maf = TRUE,
  macTh = 20,
  mafTh = NULL,
  interactive = FALSE,
  verbose = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  subplot_label_size = 9,
  path2plink = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perMarkerQC_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code><a href="#topic+perIndividualQC">perIndividualQC</a></code> was conducted, this directory should be the
same as qcdir specified in <code><a href="#topic+perIndividualQC">perIndividualQC</a></code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_do.check_snp_missingness">do.check_snp_missingness</code></td>
<td>
<p>[logical] If TRUE, run
<code><a href="#topic+check_snp_missingness">check_snp_missingness</a></code>.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_lmissth">lmissTh</code></td>
<td>
<p>[double] Threshold for acceptable variant missing rate across
samples.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_do.check_hwe">do.check_hwe</code></td>
<td>
<p>[logical] If TRUE, run <code><a href="#topic+check_hwe">check_hwe</a></code>.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_hweth">hweTh</code></td>
<td>
<p>[double] Significance threshold for deviation from HWE.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_do.check_maf">do.check_maf</code></td>
<td>
<p>[logical] If TRUE, run <code><a href="#topic+check_maf">check_maf</a></code>.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_macth">macTh</code></td>
<td>
<p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples).</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_mafth">mafTh</code></td>
<td>
<p>[double] Threshold for minor allele frequency cut-off.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_interactive">interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_marker) via ggplot2::ggsave(p=p_marker,
other_arguments) or pdf(outfile) print(p_marker) dev.off().</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_title_size">title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_subplot_label_size">subplot_label_size</code></td>
<td>
<p>[integer] Size of the subplot labeling.</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="perMarkerQC_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>perMarkerQC wraps around the marker QC functions
<code><a href="#topic+check_snp_missingness">check_snp_missingness</a></code>, <code><a href="#topic+check_hwe">check_hwe</a></code> and
<code><a href="#topic+check_maf">check_maf</a></code>. For details on the parameters and outputs, check
these function documentations.
</p>


<h3>Value</h3>

<p>Named [list] with i) fail_list, a named [list] with 1.
SNP_missingness, containing SNP IDs [vector] failing the missingness
threshold lmissTh, 2. hwe, containing SNP IDs [vector] failing the HWE exact
test threshold hweTh and 3. maf, containing SNPs Ids [vector] failing the MAF
threshold mafTh/MAC threshold macTh and ii) p_markerQC, a ggplot2-object
'containing' a sub-paneled plot with the QC-plots of
<code><a href="#topic+check_snp_missingness">check_snp_missingness</a></code>, <code><a href="#topic+check_hwe">check_hwe</a></code> and
<code><a href="#topic+check_maf">check_maf</a></code>, which can be shown by print(p_markerQC).
List entries contain NULL if that specific check was not chosen.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
# All quality control checks
## Not run: 
# run on all markers and individuals
fail_markers &lt;- perMarkerQC(indir=indir, qcdir=qcdir, name=name,
interactive=FALSE, verbose=TRUE, path2plink=path2plink)

# run on subset of individuals and markers
keep_individuals_file &lt;- system.file("extdata", "keep_individuals",
package="plinkQC")
extract_markers_file &lt;- system.file("extdata", "extract_markers",
package="plinkQC")
fail_markers &lt;- perMarkerQC(qcdir=qcdir, indir=indir,
name=name, interactive=FALSE, verbose=TRUE, path2plink=path2plink,
keep_individuals=keep_individuals_file, extract_markers=extract_markers_file)

## End(Not run)
</code></pre>

<hr>
<h2 id='plinkQC'>plinkQC</h2><span id='topic+plinkQC'></span>

<h3>Description</h3>

<p>Genotype Quality Control with 'PLINK'
</p>

<hr>
<h2 id='relatednessFilter'>Remove related individuals while keeping maximum number of individuals</h2><span id='topic+relatednessFilter'></span>

<h3>Description</h3>

<p><code>relatednessFilter</code> takes a data.frame with pair-wise relatedness
measures of samples and returns pairs of individual IDs that are related as
well as a list of suggested individual IDs to remove.
<code>relatednessFilter</code> finds pairs of samples whose relatedness estimate
is larger than the specified relatednessTh. Subsequently, for pairs of
individual that do not have additional relatives in the dataset, the
individual with the worse otherCriterionMeasure (if provided) or arbitrarily
individual 1 of that pair is selected and returned as the individual failing
the relatedness check. For more complex family structures, the unrelated
individuals per family are selected (e.g. in a simple case of a
parents-offspring trio, the offspring will be marked as fail, while the
parents will be kept in the analysis). Selection is achieved by constructing
subgraphs of clusters of individuals that are related.
<code>relatednessFilter</code> then finds the maximum independent set of vertices
in the subgraphs of related individuals. If all individuals are related (i.e.
all maximum independent sets are 0), one individual of that cluster will be
kept and all others listed as failIDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relatednessFilter(
  relatedness,
  otherCriterion = NULL,
  relatednessTh,
  otherCriterionTh = NULL,
  otherCriterionThDirection = c("gt", "ge", "lt", "le", "eq"),
  relatednessIID1 = "IID1",
  relatednessIID2 = "IID2",
  relatednessFID1 = NULL,
  relatednessFID2 = NULL,
  relatednessRelatedness = "PI_HAT",
  otherCriterionIID = "IID",
  otherCriterionMeasure = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relatednessFilter_+3A_relatedness">relatedness</code></td>
<td>
<p>[data.frame] containing pair-wise relatedness estimates
(in column [relatednessRelatedness]) for individual 1 (in column
[relatednessIID1] and individual 2 (in column [relatednessIID1]). Columns
relatednessIID1, relatednessIID2 and relatednessRelatedness have to present,
while additional columns such as family IDs can be present. Default column
names correspond to column names in output of plink &ndash;genome
(<a href="https://www.cog-genomics.org/plink/1.9/ibd">https://www.cog-genomics.org/plink/1.9/ibd</a>). All original
columns for pair-wise highIBDTh fails will be returned in fail_IBD.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_othercriterion">otherCriterion</code></td>
<td>
<p>[data.frame] containing a QC measure (in column
[otherCriterionMeasure]) per individual (in column [otherCriterionIID]).
otherCriterionMeasure and otherCriterionIID have to present, while additional
columns such as family IDs can be present. IIDs in relatednessIID1 have to
be present in otherCriterionIID.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_relatednessth">relatednessTh</code></td>
<td>
<p>[double] Threshold for filtering related individuals.
Individuals, whose pair-wise relatedness estimates are greater than this
threshold are considered related.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_othercriterionth">otherCriterionTh</code></td>
<td>
<p>[double] Threshold for filtering individuals based on
otherCriterionMeasure. If related individuals fail this threshold they will
automatically be excluded.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_othercriterionthdirection">otherCriterionThDirection</code></td>
<td>
<p>[character] Used to determine the direction
for failing the otherCriterionTh. If 'gt', individuals whose
otherCriterionMeasure &gt; otherCriterionTh will automatically be excluded.
For pairs of individuals that have no other related samples in the cohort:
if both otherCriterionMeasure &lt; otherCriterionTh, the individual with the
larger otherCriterionMeasure will be excluded.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_relatednessiid1">relatednessIID1</code></td>
<td>
<p>[character] Column name of column containing the IDs
of the first individual.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_relatednessiid2">relatednessIID2</code></td>
<td>
<p>[character] Column name of column containing the IDs
of the second individual.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_relatednessfid1">relatednessFID1</code></td>
<td>
<p>[character, optional] Column name of column containing
the family IDs of the first individual; if only relatednessFID1 but not
relatednessFID2 provided, or none provided even though present in
relatedness, FIDs will not be returned.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_relatednessfid2">relatednessFID2</code></td>
<td>
<p>[character, optional] Column name of column containing
the family IDs of the second individual; if only relatednessFID2 but not
relatednessFID1 provided, or none provided even though present in
relatedness, FIDs will not be returned.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_relatednessrelatedness">relatednessRelatedness</code></td>
<td>
<p>[character] Column name of column containing
the relatedness estimate.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_othercriterioniid">otherCriterionIID</code></td>
<td>
<p>[character] Column name of column containing the
individual IDs.</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_othercriterionmeasure">otherCriterionMeasure</code></td>
<td>
<p>[character] Column name of the column containing
the measure of the otherCriterion (for instance SNP missingness rate).</p>
</td></tr>
<tr><td><code id="relatednessFilter_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named [list] with i) relatednessFails, a [data.frame] containing the
data.frame relatedness after filtering for pairs of individuals in
relatednessIID1 and relatednessIID2, that fail the relatedness
QC; the data.frame is reordered with the fail individuals in column 1 and
their related individuals in column 2 and ii) failIDs, a [data.frame] with
the [IID]s (and [FID]s if provided) of the individuals that fail the
relatednessTh.
</p>

<hr>
<h2 id='run_check_ancestry'>Run PLINK principal component analysis</h2><span id='topic+run_check_ancestry'></span>

<h3>Description</h3>

<p>Run plink &ndash;pca to calculate the principal components on merged genotypes
of the study and reference dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_check_ancestry(
  indir,
  prefixMergedDataset,
  qcdir = indir,
  verbose = FALSE,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_check_ancestry_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files prefixMergedDataset.bim,prefixMergedDataset.fam and
prefixMergedDataset.bed.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_prefixmergeddataset">prefixMergedDataset</code></td>
<td>
<p>[character] Prefix of merged study and
reference data files, i.e. prefixMergedDataset.bed, prefixMergedDataset.bim,
prefixMergedDataset.fam.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to save
prefixMergedDataset.eigenvec as returned by plink &ndash;pca. User needs writing
permission to qcdir. Per default qcdir=indir.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_ancestry_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both, <code><a href="#topic+run_check_ancestry">run_check_ancestry</a></code> and its evaluation by
<code><a href="#topic+evaluate_check_ancestry">evaluate_check_ancestry</a></code> can simply be invoked by
<code><a href="#topic+check_ancestry">check_ancestry</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
prefixMergedDataset &lt;- 'data.HapMapIII'
path2plink &lt;- 'path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# ancestry check on all individuals in dataset
run &lt;- run_check_ancestry(indir=indir, qcdir=qcdir, prefixMergedDataset,
path2plink=path2plink)

# ancestry check on subset of dataset
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
run &lt;- run_check_ancestry(indir=indir, qcdir=qcdir, name=name,
remove_individuals=remove_individuals_file, path2plink=path2plink)

## End(Not run)
</code></pre>

<hr>
<h2 id='run_check_heterozygosity'>Run PLINK heterozygosity rate calculation</h2><span id='topic+run_check_heterozygosity'></span>

<h3>Description</h3>

<p>Run plink &ndash;het to calculate heterozygosity rates per individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_check_heterozygosity(
  indir,
  name,
  qcdir = indir,
  verbose = FALSE,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_check_heterozygosity_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to save name.het as returned by
plink &ndash;het. User needs writing permission to qcdir. Per default
qcdir=indir.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_heterozygosity_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All, <code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code>,
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code> and their evaluation by
<code><a href="#topic+evaluate_check_het_and_miss">evaluate_check_het_and_miss</a></code> can simply be invoked by
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- 'data'
qcdir &lt;- tempdir()
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# heterozygosity check on all individuals in dataset
run &lt;- run_check_heterozygosity(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

#' # heterozygosity on subset of dataset
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
run &lt;- run_check_heterozygosity(indir=indir, qcdir=qcdir, name=name,
remove_individuals=remove_individuals_file,path2plink=path2plink)

## End(Not run)
</code></pre>

<hr>
<h2 id='run_check_missingness'>Run PLINK missingness rate calculation</h2><span id='topic+run_check_missingness'></span>

<h3>Description</h3>

<p>Run  plink &ndash;missing to calculate missing genotype rates
per individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_check_missingness(
  indir,
  name,
  qcdir = indir,
  verbose = FALSE,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_check_missingness_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to save name.imiss as returned
by plink &ndash;missing. User needs writing permission to qcdir. Per default
qcdir=indir.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_missingness_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All, <code><a href="#topic+run_check_heterozygosity">run_check_heterozygosity</a></code>,
<code><a href="#topic+run_check_missingness">run_check_missingness</a></code> and their evaluation by
<code><a href="#topic+evaluate_check_het_and_miss">evaluate_check_het_and_miss</a></code> can simply be invoked by
<code><a href="#topic+check_het_and_miss">check_het_and_miss</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- 'data'
qcdir &lt;- tempdir()
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# missingness check on all individuals in dataset
run &lt;- run_check_missingness(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

# missingness on subset of dataset
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
run &lt;- run_check_missingness(indir=indir, qcdir=qcdir, name=name,
remove_individuals=remove_individuals_file, path2plink=path2plink)

## End(Not run)
</code></pre>

<hr>
<h2 id='run_check_relatedness'>Run PLINK IBD estimation</h2><span id='topic+run_check_relatedness'></span>

<h3>Description</h3>

<p>Run LD pruning on dataset with plink &ndash;exclude range highldfile
&ndash;indep-pairwise 50 5 0.2, where highldfile contains regions of high LD as
provided by Anderson et (2010) Nature Protocols. Subsequently, plink
&ndash;genome is run on the LD pruned, maf-filtered data. plink &ndash;genome
calculates identity by state (IBS) for each pair of individuals based on the
average proportion of alleles shared at genotyped SNPs. The degree of recent
shared ancestry,i.e. the identity by descent (IBD) can be estimated from the
genome-wide IBS. The proportion of IBD between two individuals is returned by
&ndash;genome as PI_HAT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_check_relatedness(
  indir,
  name,
  qcdir = indir,
  highIBDTh = 0.185,
  mafThRelatedness = 0.1,
  path2plink = NULL,
  filter_high_ldregion = TRUE,
  high_ldregion_file = NULL,
  genomebuild = "hg19",
  showPlinkOutput = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_check_relatedness_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to save name.genome as returned
by plink &ndash;genome. User needs writing permission to qcdir. Per default
qcdir=indir.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_highibdth">highIBDTh</code></td>
<td>
<p>[double] Threshold for acceptable proportion of IBD between
pair of individuals; only pairwise relationship estimates larger than this
threshold will be recorded.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_mafthrelatedness">mafThRelatedness</code></td>
<td>
<p>[double] Threshold of minor allele frequency filter
for selecting variants for IBD estimation.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_filter_high_ldregion">filter_high_ldregion</code></td>
<td>
<p>[logical] Should high LD regions be filtered
before IBD estimation; carried out per default with high LD regions for
hg19 provided as default via <code>genomebuild</code>. For alternative genome
builds not provided or non-human data, high LD regions files can be
provided via <code>high_ldregion_file</code>.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_high_ldregion_file">high_ldregion_file</code></td>
<td>
<p>[character] Path to file with high LD regions used
for filtering before IBD estimation if <code>filter_high_ldregion</code> == TRUE,
otherwise ignored; for human genome data, high LD region files are provided
and can simply be chosen via <code>genomebuild</code>. Files have to be
space-delimited, no column names with the following columns: chromosome,
region-start, region-end, region number. Chromosomes are specified without
'chr' prefix. For instance:
1 48000000 52000000 1
2 86000000 100500000 2</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_genomebuild">genomebuild</code></td>
<td>
<p>[character] Name of the genome build of the PLINK file
annotations, ie mappings in the name.bim file. Will be used to remove
high-LD regions based on the coordinates of the respective build. Options
are hg18, hg19 and hg38. See @details.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_relatedness_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code><a href="#topic+run_check_relatedness">run_check_relatedness</a></code> and its evaluation via
<code><a href="#topic+evaluate_check_relatedness">evaluate_check_relatedness</a></code> can simply be invoked by
<code><a href="#topic+check_relatedness">check_relatedness</a></code>.
</p>
<p>The IBD estimation is conducted on LD pruned data and in a first
step, high LD regions are excluded. The regions were derived from the
high-LD-regions file provided by Anderson et (2010) Nature Protocols. These
regions are in NCBI36 (hg18) coordinates and were lifted to GRCh37 (hg19)
and GRC38 (hg38) coordinates using the liftOver tool available here:
<a href="https://genome.ucsc.edu/cgi-bin/hgLiftOver">https://genome.ucsc.edu/cgi-bin/hgLiftOver</a>. The 'Minimum ratio of bases
that must remap' which was set to 0.5 and the 'Allow multiple output regions'
box ticked; for all other parameters, the default options were selected.
LiftOver files were generated on July 9,2019. The commands for formatting
the files are provided in system.file(&quot;extdata&quot;, 'liftOver.cmd',
package=&quot;plinkQC&quot;).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- 'data'
qcdir &lt;- tempdir()
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# Relatedness estimation based in all markers in dataset
run &lt;- run_check_relatedness(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

# relatedness estimation on subset of dataset
keep_individuals_file &lt;- system.file("extdata", "keep_individuals",
package="plinkQC")
run &lt;- run_check_relatedness(indir=indir, qcdir=qcdir, name=name,
keep_individuals=keep_individuals_file, path2plink=path2plink)

## End(Not run)
</code></pre>

<hr>
<h2 id='run_check_sex'>Run PLINK sexcheck</h2><span id='topic+run_check_sex'></span>

<h3>Description</h3>

<p>Run  plink &ndash;sexcheck to calculate the heterozygosity rate
across X-chromosomal variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_check_sex(
  indir,
  name,
  qcdir = indir,
  verbose = FALSE,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_check_sex_+3A_indir">indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_name">name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_qcdir">qcdir</code></td>
<td>
<p>[character] /path/to/directory to save name.sexcheck as returned
by plink &ndash;check-sex. User needs writing permission to qcdir. Per default
qcdir=indir.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_verbose">verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_path2plink">path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href="sys.html#topic+exec">exec</a></code>('plink').</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_keep_individuals">keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_remove_individuals">remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_exclude_markers">exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_extract_markers">extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td></tr>
<tr><td><code id="run_check_sex_+3A_showplinkoutput">showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both <code><a href="#topic+run_check_sex">run_check_sex</a></code> and its evaluation
<code><a href="#topic+evaluate_check_sex">evaluate_check_sex</a></code> can simply be invoked by
<code><a href="#topic+check_sex">check_sex</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indir &lt;- system.file("extdata", package="plinkQC")
name &lt;- 'data'
qcdir &lt;- tempdir()
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# simple sexcheck on all individuals in dataset
run &lt;- run_check_sex(indir=indir, qcdir=qcdir, name=name)

# sexcheck on subset of dataset
keep_individuals_file &lt;- system.file("extdata", "keep_individuals",
package="plinkQC")
run &lt;- run_check_sex(indir=indir, qcdir=qcdir, name=name,
keep_individuals=keep_individuals_file, path2plink=path2plink)

## End(Not run)
</code></pre>

<hr>
<h2 id='testNumerics'>Test lists for different properties of numerics</h2><span id='topic+testNumerics'></span>

<h3>Description</h3>

<p>Test all elements of a list if they are numeric, positive numbers, integers
or proportions (range 0-1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testNumerics(numbers, positives = NULL, integers = NULL, proportions = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="testNumerics_+3A_numbers">numbers</code></td>
<td>
<p>[list] whose elements are tested for being numeric.</p>
</td></tr>
<tr><td><code id="testNumerics_+3A_positives">positives</code></td>
<td>
<p>[list] whose elements are tested for being positive numbers.</p>
</td></tr>
<tr><td><code id="testNumerics_+3A_integers">integers</code></td>
<td>
<p>[list] whose elements are tested for being integers.</p>
</td></tr>
<tr><td><code id="testNumerics_+3A_proportions">proportions</code></td>
<td>
<p>[list] whose elements are tested for being proportions.
between 0 and 1.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
