<!DOCTYPE html><html><head><title>Help for package plinkFile</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {plinkFile}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CHR'><p>A dictionary to map chromosome names to integers.</p></a></li>
<li><a href='#DBT'><p>A decoding byte table to map raw intergers to genotype dosages.</p></a></li>
<li><a href='#readBED'><p>Read BED file</p></a></li>
<li><a href='#readBIM'><p>Read BIM file</p></a></li>
<li><a href='#readBSM'><p>Read Binary Symmetric Matrix (BSM)</p></a></li>
<li><a href='#readFAM'><p>Read FAM file</p></a></li>
<li><a href='#readGRM'><p>Read Genetic Related Matrix (GRM) of GCTA</p></a></li>
<li><a href='#readIBS'><p>Read PLINK Binary IBS matrix</p></a></li>
<li><a href='#readIID'><p>read individual ID</p></a></li>
<li><a href='#readREL'><p>Read PLINK Binary REL matrix</p></a></li>
<li><a href='#readVCM'><p>Read Variant Count Matrix (VCM) accompanying a GCTA GRM</p></a></li>
<li><a href='#readVID'><p>read variant ID</p></a></li>
<li><a href='#saveBED'><p>Save BED file</p></a></li>
<li><a href='#saveBSM'><p>Save Symmetric Matrix to Binary</p></a></li>
<li><a href='#saveGRM'><p>Save symmetic matrix to GCTA GRM format.</p></a></li>
<li><a href='#scanBED'><p>travers variants in a PLINK1 BED fileset</p></a></li>
<li><a href='#testReadBED'><p>Test BED Reader</p></a></li>
<li><a href='#testReadBSM'><p>Test Genetic Relatedness Matrix Reader</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>'PLINK' (and 'GCTA') File Helpers</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Reads/write binary genotype file compatible with 'PLINK' <a href="https://www.cog-genomics.org/plink/1.9/input#bed">https://www.cog-genomics.org/plink/1.9/input#bed</a> into/from a R matrix; traverse genotype data one windows of variants at a time, like apply() or a for loop; reads/writes genotype relatedness/kinship matrices created by 'PLINK' <a href="https://www.cog-genomics.org/plink/1.9/distance#make_rel">https://www.cog-genomics.org/plink/1.9/distance#make_rel</a> or 'GCTA' <a href="https://cnsgenomics.com/software/gcta/#MakingaGRM">https://cnsgenomics.com/software/gcta/#MakingaGRM</a> into/from a R square matrix. It is best used for bringing data produced by 'PLINK' and 'GCTA' into R workflow.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-24 00:17:02 UTC; xt</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaoran Tong [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaoran Tong &lt;xiaoran.tong.cn@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-24 12:00:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='CHR'>A dictionary to map chromosome names to integers.</h2><span id='topic+CHR'></span>

<h3>Description</h3>

<p>A dictionary to map chromosome names to integers.
</p>

<hr>
<h2 id='DBT'>A decoding byte table to map raw intergers to genotype dosages.</h2><span id='topic+DBT'></span>

<h3>Description</h3>

<p>The table avoids bit shifting and may speed up the reading of plink BED.
</p>


<h3>Details</h3>

<p>The decoding table approach is still experimental.
</p>
<p>Actually, DBT == dbd(as.raw(seq(0x00, 0xFF)), 4L)
</p>

<hr>
<h2 id='readBED'>Read BED file</h2><span id='topic+readBED'></span>

<h3>Description</h3>

<p>Read a PLINK BED file into a R matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBED(pfx, iid = 1, vid = 1, vfr = NULL, vto = NULL, quiet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readBED_+3A_pfx">pfx</code></td>
<td>
<p>prefix of PLINK file set, or the fullname of a BED file.</p>
</td></tr>
<tr><td><code id="readBED_+3A_iid">iid</code></td>
<td>
<p>option to read <code>N</code> IID as row names. (def=1, see <code><a href="#topic+readIID">readIID()</a></code>)</p>
</td></tr>
<tr><td><code id="readBED_+3A_vid">vid</code></td>
<td>
<p>option to read <code>P</code> VID as col names. (def=1, see <code><a href="#topic+readVID">readVID()</a></code>)</p>
</td></tr>
<tr><td><code id="readBED_+3A_vfr">vfr</code></td>
<td>
<p>variant-wise, from where to read? (number/proportion, def=1)</p>
</td></tr>
<tr><td><code id="readBED_+3A_vto">vto</code></td>
<td>
<p>varinat-wise, to where then stop? (number/proportion, def=P)</p>
</td></tr>
<tr><td><code id="readBED_+3A_quiet">quiet</code></td>
<td>
<p>suppress screen printing? (def=TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is  meant for genotype that  can fit into  system memory; the size  of R
matrix is 16 times  the BED file.  To traverse a huge  BED several varants at
time  without  loading it  entirely  into  the  memoty, see  <code><a href="#topic+scanBED">scanBED()</a></code>  and
<code><a href="#topic+loopBED">loopBED()</a></code>.
</p>
<p>A PLINK1 binary fileset has three files,
</p>

<dl>
<dt><code>pfx.fam</code>:</dt><dd><p>text table of <code>N</code> individuals.</p>
</dd>
<dt><code>pfx.bim</code>:</dt><dd><p>text table of <code>P</code> genomic variants (i.e., SNPs).</p>
</dd>
<dt><code>pfx.bed</code>:</dt><dd><p><code>N</code> x <code>P</code> genotype matrix in condensed binary format.</p>
</dd>
</dl>

<p>The three  files comprising a genotype  data are typically referred  by their
common  prefix,  for  example,  the  X  chromosome  genotype  represented  by
<code>chrX.bed</code>, <code>chrX.fam</code>, and <code>chrX.bim</code> are jointly refered by <code>chrX</code>.
</p>


<h3>Value</h3>

<p>genotype matrix with <code>N</code> row individuals and <code>P</code> column variants.
</p>


<h3>See Also</h3>

<p>readBED
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## read an entire small data
bed &lt;- system.file("extdata", 'm20.bed', package="plinkFile")
gmx &lt;- readBED(bed, quiet=FALSE)

## read part of a large data
bed &lt;- system.file("extdata", '000.bed', package="plinkFile")
U &lt;- readBED(bed, vfr=01, vto=10, quiet=FALSE)
V &lt;- readBED(bed, vfr=11, vto=20, quiet=FALSE)
W &lt;- cbind(U, V)
X &lt;- readBED(bed, vfr=01, vto=20, quiet=FALSE)
all.equal(W, X)

</code></pre>

<hr>
<h2 id='readBIM'>Read BIM file</h2><span id='topic+readBIM'></span>

<h3>Description</h3>

<p>Get variant meta-data form the <em>bim</em> file of a PLINK1 BED fileset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBIM(bim, vfr = NULL, vto = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readBIM_+3A_bim">bim</code></td>
<td>
<p>prefix or name of a PLINK file.</p>
</td></tr>
<tr><td><code id="readBIM_+3A_vfr">vfr</code></td>
<td>
<p>variant-wise, from where to read? (index/proportion, def=1).</p>
</td></tr>
<tr><td><code id="readBIM_+3A_vto">vto</code></td>
<td>
<p>varinat-wise, to where then stop? (index/proportion, def=P).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are six columns in a <em>bim</em> file
</p>

<ul>
<li><p> chr: chromosme of the variant
</p>
</li>
<li><p> vid: variant id, such as an RS number;
</p>
</li>
<li><p> cmg: position by centimorgan;
</p>
</li>
<li><p> bps: position by basepairs;
</p>
</li>
<li><p> al1: allele 1, the one counted as dosage.
</p>
</li>
<li><p> al2: allele 2.
</p>
</li></ul>



<h3>Value</h3>

<p>data frame of variants, loaded from BIM.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bed &lt;- file.path(system.file("extdata", package="plinkFile"), "000.bed")
bim &lt;- readBIM(bed, 20, 30)
bim
</code></pre>

<hr>
<h2 id='readBSM'>Read Binary Symmetric Matrix (BSM)</h2><span id='topic+readBSM'></span>

<h3>Description</h3>

<p>Read BSM represented by a pair of files suffixed by &quot;.bin&quot; and &quot;.id&quot;, usually
produced by PLINK and GCTA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBSM(pfx, dgv = 1, fid = NULL, id = NULL, bin = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readBSM_+3A_pfx">pfx</code></td>
<td>
<p>prefix of data files <code>pfx.id</code> and <code>pfx.bin</code></p>
</td></tr>
<tr><td><code id="readBSM_+3A_dgv">dgv</code></td>
<td>
<p>diagonal value for matrix without a diagonal (def=1.0)</p>
</td></tr>
<tr><td><code id="readBSM_+3A_fid">fid</code></td>
<td>
<p>separator between FID and IID (def=NULL, use IID only)</p>
</td></tr>
<tr><td><code id="readBSM_+3A_id">id</code></td>
<td>
<p>use id file instead of the default <code>pfx.id</code></p>
</td></tr>
<tr><td><code id="readBSM_+3A_bin">bin</code></td>
<td>
<p>use bin file instead of the default <code>pfx.bin</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The &quot;.bin&quot; is a binary file storing the matrix entries, which can be
</p>

<dl>
<dt>SQR</dt><dd><p>the <code>N</code> x <code>N</code> symmetric matrix in full</p>
</dd>
<dt>LWD</dt><dd><p>the lower triangle with diagonal</p>
</dd>
<dt>LND</dt><dd><p>the lower triangle without diagonal</p>
</dd>
</dl>
<p>,
saved as either single or double precision.
</p>
<p>The &quot;.id&quot; is  a text file of <code>N</code>  family ID (FID) and individual  ID (IID) in
two columns. By default, IID is used as matix row and column names.
</p>
<p>PLINK option  <code>--make-red bin</code>,  <code>--distance bin</code>, and  GCTA option
<code>--make-grm</code> all creats binary symmetric matrices, widely used in linear
mixed model or kernel based models for genetics.
</p>


<h3>Value</h3>

<p>symmetric matrix loaded, with sample ID as both row and column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20.rel")
(readBSM(pfx, fid=":"))

</code></pre>

<hr>
<h2 id='readFAM'>Read FAM file</h2><span id='topic+readFAM'></span>

<h3>Description</h3>

<p>Read sample meta-data form the <em>fam</em> file of a PLINK1 BED fileset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readFAM(fam)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readFAM_+3A_fam">fam</code></td>
<td>
<p>prefix or name of a PLINK file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are six columns in a <em>bim</em> file
</p>

<ul>
<li><p> fid: family ID;
</p>
</li>
<li><p> iid: individual ID, default row name used by <code style="white-space: pre;">&#8288;[readBED]&#8288;</code>;
</p>
</li>
<li><p> mom: maternal ID;
</p>
</li>
<li><p> dad: paternal ID;
</p>
</li>
<li><p> sex: individual sex.
</p>
</li>
<li><p> phe: phenotype, not often used;
</p>
</li></ul>

<p>The PLINK1 <em>bim</em> file has no header line, this is changed in PLINK2.
</p>
<p>The columns &quot;sex&quot; and &quot;phe&quot; are  mostly the legency of early GWAS, nowerdays
it is common to provide sex, among other covariates, and multiple phenotypes
in a separate file.
</p>


<h3>Value</h3>

<p>data frame of individuals, loaded from FAM.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
fam &lt;- readFAM(pfx)
fam
</code></pre>

<hr>
<h2 id='readGRM'>Read Genetic Related Matrix (GRM) of GCTA</h2><span id='topic+readGRM'></span>

<h3>Description</h3>

<p>GRM is the  core formt of GCTA,  which is an binary symmetric  matrix with an
extra variant count  matrix (VCM), this function reads  the binary sysmmetric
matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readGRM(pfx, fid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readGRM_+3A_pfx">pfx</code></td>
<td>
<p>prefix of GRM file set</p>
</td></tr>
<tr><td><code id="readGRM_+3A_fid">fid</code></td>
<td>
<p>separator after family ID (def=NULL, use IID only)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GCTA GRM is represented by a set of three files:
</p>

<dl>
<dt>.grm.bin   :</dt><dd><p>GRM matrix in binary</p>
</dd>
<dt>.grm.id    :</dt><dd><p>sample FID and IID in text</p>
</dd>
<dt>.grm.N.bin :</dt><dd><p>number of valid variants for each GRM entry</p>
</dd></dl>

<p>and it always  uses single precision (4  bytes per entry).
</p>
<p>To read the extra the extra VCM (grm.N.bin), use <code><a href="#topic+readVCM">readVCM</a></code>.
</p>


<h3>Value</h3>

<p>matrix of relatedness with sample ID in row and column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
(readGRM(pfx))

</code></pre>

<hr>
<h2 id='readIBS'>Read PLINK Binary IBS matrix</h2><span id='topic+readIBS'></span>

<h3>Description</h3>

<p>A PLINK IBS (Identity by State) matrix is represented by
</p>

<dl>
<dt>.mibs.bin:</dt><dd><p>IBS matrix in binary</p>
</dd>
<dt>.mibs.id :</dt><dd><p>FID and IID in text</p>
</dd>
</dl>

<p>A binary IBS matrix is the result of PLINK <code>--distance ibs bin</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readIBS(pfx, fid = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readIBS_+3A_pfx">pfx</code></td>
<td>
<p>prefix of the IBS file set.</p>
</td></tr>
<tr><td><code id="readIBS_+3A_fid">fid</code></td>
<td>
<p>seperate after family ID (def=NULL, use IID only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>IBS matrix with row and column names set to sample ID.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
(readIBS(pfx))

</code></pre>

<hr>
<h2 id='readIID'>read individual ID</h2><span id='topic+readIID'></span>

<h3>Description</h3>

<p>Generate individual ID automatically, or based on a <em>fam</em> file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readIID(fam, opt = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readIID_+3A_fam">fam</code></td>
<td>
<p>prefix or name of a PLINK file, or data fram from a FAM file.</p>
</td></tr>
<tr><td><code id="readIID_+3A_opt">opt</code></td>
<td>
<p>option (def=1: the 2nd column in FAM file).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The option (<code>opt</code>) can be:
</p>

<ul>
<li><p> 1 = the <em>iid</em> column in FAM (default),
</p>
</li>
<li><p> 2 = formated as <em>fid</em>.<em>iid</em>,
</p>
</li>
<li><p> 0 = nothing
</p>
</li>
<li><p> -1 = numbering of individuals, decimal
</p>
</li>
<li><p> -2 = numbering of individuals, zero-padded fix-length decimal
</p>
</li>
<li><p> -3 = numbering of individuals, zero-padded fix-length hexedemical
or, a vector of IDs to use.
</p>
</li></ul>



<h3>Value</h3>

<p>a vector of individual ID
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
readIID(pfx,  1) # opt= 1: IID
readIID(pfx,  2) # opt= 2: FID.IID
readIID(pfx, -1) # opt=-1: number sequence
readIID(pfx, -2) # opt=-2: number sequence, fixed length, decimal
readIID(pfx, -3) # opt=-3: number sequence, fixed length, hexidemical

</code></pre>

<hr>
<h2 id='readREL'>Read PLINK Binary REL matrix</h2><span id='topic+readREL'></span>

<h3>Description</h3>

<p>A PLINK REL (Relatedness) matrix is represented by
</p>

<dl>
<dt>.rel.bin:</dt><dd><p>REL matrix in binary</p>
</dd>
<dt>.rel.id :</dt><dd><p>FID and IID in text</p>
</dd>
</dl>

<p>A binary REL matrix is the result of PLINK <code>--make-rel bin</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readREL(pfx, fid = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readREL_+3A_pfx">pfx</code></td>
<td>
<p>prefix of the REL file set</p>
</td></tr>
<tr><td><code id="readREL_+3A_fid">fid</code></td>
<td>
<p>separate after family ID. (def=NULL, use IID only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>relatedness matrix with row and column names set to sample ID.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
(readREL(pfx))

</code></pre>

<hr>
<h2 id='readVCM'>Read Variant Count Matrix (VCM) accompanying a GCTA GRM</h2><span id='topic+readVCM'></span>

<h3>Description</h3>

<p>GRM (Genetic Relatedness Matrix) is the core formt of GCTA, which is a PLINK
binary  symmetric matrix  with  an  extra variant  count  matrix (VCM),  this
function reads the VCM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVCM(pfx, fid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readVCM_+3A_pfx">pfx</code></td>
<td>
<p>prefix of GRM file set</p>
</td></tr>
<tr><td><code id="readVCM_+3A_fid">fid</code></td>
<td>
<p>seperate after family ID (def=NULL, use IID only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of variant count with sample ID in row and column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
(readVCM(pfx))

</code></pre>

<hr>
<h2 id='readVID'>read variant ID</h2><span id='topic+readVID'></span>

<h3>Description</h3>

<p>Generate variant ID automatically, or based on a <em>bim</em> file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVID(bim, opt = NULL, vfr = NULL, vto = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readVID_+3A_bim">bim</code></td>
<td>
<p>prefix or name of a PLINK file, or data frame from a BIM file.</p>
</td></tr>
<tr><td><code id="readVID_+3A_opt">opt</code></td>
<td>
<p>option (def=1: the 2nd column in BIM file).</p>
</td></tr>
<tr><td><code id="readVID_+3A_vfr">vfr</code></td>
<td>
<p>variant-wise, from where to read? (index/proportion, def=1).</p>
</td></tr>
<tr><td><code id="readVID_+3A_vto">vto</code></td>
<td>
<p>varinat-wise, to where then stop? (index/proportion, def=P).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The option (<code>opt</code>) can be:
</p>

<ul>
<li><p> 1 = the 2nd column in <em>pfx</em>.bim (default),
</p>
</li>
<li><p> 2 = formated as %CHR(02d):%BPS(09d),
</p>
</li>
<li><p> 3 = formated as %CHR(02d):%BPS(09d)_AL1(s)_AL2(s)
</p>
</li>
<li><p> 0 = nothing
</p>
</li>
<li><p> -1 = numbering of variants, decimal
</p>
</li>
<li><p> -2 = numbering of variants, zero-padded, fixed length decimal
</p>
</li>
<li><p> -3 = numbering of variants, zero-padded, fixed length hexedemical
</p>
</li>
<li><p> or, a vector of IDs to use.
</p>
</li></ul>



<h3>Value</h3>

<p>a vector of variant ID
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read variant ID
pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")

# opt=1: 2nd column in the BED file (default)
vid &lt;- readVID(pfx, 1); head(vid); tail(vid)

# opt=2: format by position
vid &lt;- readVID(pfx, 2); head(vid); tail(vid)

# opt=3: format by position and alleles
vid &lt;- readVID(pfx, 3); head(vid); tail(vid)

# opt=-1: number sequence
vid &lt;- readVID(pfx, -1); head(vid); tail(vid)

# opt=-2: number sequence, fixed length, decimal
vid &lt;- readVID(pfx, -2); head(vid); tail(vid)

# opt=-3: number sequence, fixed length, hexidemical
vid &lt;- readVID(pfx, -3); head(vid); tail(vid)
</code></pre>

<hr>
<h2 id='saveBED'>Save BED file</h2><span id='topic+saveBED'></span>

<h3>Description</h3>

<p>Save a R matrix into a PLINK BED file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveBED(pfx, bed, quiet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saveBED_+3A_pfx">pfx</code></td>
<td>
<p>prefix of the output file set, in PLINK1 BED format.</p>
</td></tr>
<tr><td><code id="saveBED_+3A_bed">bed</code></td>
<td>
<p>N x P genotype matrix</p>
</td></tr>
<tr><td><code id="saveBED_+3A_quiet">quiet</code></td>
<td>
<p>do not report (def=TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is meant for genotype small enough  to fit into system memory. The size
of R matrix is 16  times the size of the BED file.
</p>

<hr>
<h2 id='saveBSM'>Save Symmetric Matrix to Binary</h2><span id='topic+saveBSM'></span>

<h3>Description</h3>

<p>Save symmetric matrix  to a binary core  file (.bin), and a text  file of IDs
(.id), recognizable by PLINK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveBSM(pfx, x, ltr = TRUE, diag = TRUE, unit = 4L, fid = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saveBSM_+3A_pfx">pfx</code></td>
<td>
<p>prefix of output files</p>
</td></tr>
<tr><td><code id="saveBSM_+3A_x">x</code></td>
<td>
<p>symmetric matrix to save</p>
</td></tr>
<tr><td><code id="saveBSM_+3A_ltr">ltr</code></td>
<td>
<p>store the lower triangle only? (def=TRUE)</p>
</td></tr>
<tr><td><code id="saveBSM_+3A_diag">diag</code></td>
<td>
<p>save diagnal? (def=TRUE) ignored if <code>ltr</code> is FALSE.</p>
</td></tr>
<tr><td><code id="saveBSM_+3A_unit">unit</code></td>
<td>
<p>numerical unit, (def=4, single precision)</p>
</td></tr>
<tr><td><code id="saveBSM_+3A_fid">fid</code></td>
<td>
<p>separator between FID and IID (def=&quot;.&quot;).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20.rel")
rel &lt;- readBSM(pfx)  # relatedness kernel matrix
re2 &lt;- rel^2         # 2nd order polynomial kernel

tmp &lt;- tempdir()
dir.create(tmp, FALSE)
out &lt;- file.path(tmp, 'm20.re2')
saveBSM(out, re2)    # save the polynomial kernel
dir(tmp)             # show new files, then clean up
unlink(tmp, recursive=TRUE)

</code></pre>

<hr>
<h2 id='saveGRM'>Save symmetic matrix to GCTA GRM format.</h2><span id='topic+saveGRM'></span>

<h3>Description</h3>

<p>GRM (Genetic  Relatedness Matrix) is  the core  formt of GCTA,  this function
saves a R symmetric matrix to a file set recgnizable by GCTA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveGRM(pfx, grm, vcm = NULL, fid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saveGRM_+3A_pfx">pfx</code></td>
<td>
<p>prefix of data files</p>
</td></tr>
<tr><td><code id="saveGRM_+3A_grm">grm</code></td>
<td>
<p>genome relatedness matrix to save</p>
</td></tr>
<tr><td><code id="saveGRM_+3A_vcm">vcm</code></td>
<td>
<p>variant counts matrix to save (def=1).</p>
</td></tr>
<tr><td><code id="saveGRM_+3A_fid">fid</code></td>
<td>
<p>separator after family ID. (def=NULL)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Three files will be saved:
</p>

<dl>
<dt>.grm.bin   :</dt><dd><p>genetic relatedness matrix in binary</p>
</dd>
<dt>.grm.id    :</dt><dd><p>FID and IID for N individuals in text</p>
</dd>
<dt>.grm.N.bin :</dt><dd><p>variant count matrix (VCM) in binary</p>
</dd>
</dl>

<p>FID and IID will be generated if the <code>grm</code> to be saved has no row names.
</p>
<p>When save the  <code>vcm</code>, if a single  number is given, this  number is used
as the variant count for all entries in the GRM.
</p>
<p><code>saveGRM</code> is useful in exporting  customized kinship matrices (such as a
Gaussian or a  Laplacian kernel) to a  GRM acceptable by GCTA,  which are not
supported by GCTA's own GRM builder.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "m20")
gmx &lt;- readBED(pfx)  # read genotype matrix from PLINK BED.
gmx &lt;- scale(gmx)    # standardize
tmp &lt;- tempdir()     # for example outputs
dir.create(tmp, FALSE)

# kinship matrix as Gaussian kernel, built from the first 10 variants
gmx.gau &lt;- gmx[, +(1:10)]                 # the first 10 variants
not.na.gau &lt;- tcrossprod(!is.na(gmx.gau)) # variant count matrix
kin.gau &lt;- exp(as.matrix(-dist(gmx.gau, "euc")) / not.na.gau)
print(kin.gau)                            # the Gaussian kernel
out.gau &lt;- file.path(tmp, "m20.gau")
saveGRM(out.gau, kin.gau, not.na.gau)     # gau.grm.* should appear

# kinship matrix as Laplacian kernel, built without the first 10 variants
gmx.lap &lt;- gmx[, -(1:10)]                 # drop the first 10 variants
not.na.lap &lt;- tcrossprod(!is.na(gmx.lap)) # variant count matrix
kin.lap &lt;- exp(as.matrix(-dist(gmx.lap, "man")) / not.na.lap)
out.lap &lt;- file.path(tmp, "m20.lap")
print(kin.lap)                            # the Laplacian kernel
saveGRM(out.lap, kin.lap, not.na.lap)     # lap.grm.* should appear

# merge kinship in R language for a radius based function kernel matrix
not.na.rbf &lt;- not.na.gau + not.na.lap
kin.rbf &lt;- (kin.gau * not.na.gau + kin.lap * not.na.lap) / not.na.rbf
print(kin.rbf)
out.rbf &lt;- file.path(tmp, "m20.rbf")
saveGRM(out.rbf, kin.rbf, not.na.rbf)     # rbf.grm.* should appear

# show saved matrices, then clean up
dir(tmp, "(gau|lap|rbf)")
unlink(tmp, recursive=TRUE)

</code></pre>

<hr>
<h2 id='scanBED'>travers variants in a PLINK1 BED fileset</h2><span id='topic+scanBED'></span><span id='topic+loopBED'></span><span id='topic+bed'></span>

<h3>Description</h3>

<p>Sequentially visits variants in a PLINK1  BED fileset with a stepping window
matrix, and process each window matrix  with user scripts either in function
or expression form, meant for data to big to fit in the memory.
</p>
<p>To read the entire BED into a R matrix, use <code style="white-space: pre;">&#8288;[readBED]()&#8288;</code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanBED(
  pfx,
  FUN,
  ...,
  win = 1,
  iid = 1,
  vid = 1,
  vfr = NULL,
  vto = NULL,
  buf = 2^24,
  simplify = TRUE
)

loopBED(
  pfx,
  EXP,
  GVR = "g",
  win = 1,
  iid = 1,
  vid = 1,
  vfr = NULL,
  vto = NULL,
  buf = 2^24,
  simplify = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scanBED_+3A_pfx">pfx</code></td>
<td>
<p>prefix of PLINK BED.</p>
</td></tr>
<tr><td><code id="scanBED_+3A_fun">FUN</code></td>
<td>
<p>a function to process each window of variants;</p>
</td></tr>
<tr><td><code id="scanBED_+3A_...">...</code></td>
<td>
<p>additional argument for <em><code>FUN</code></em> when <code>scanBED</code> is used.</p>
</td></tr>
<tr><td><code id="scanBED_+3A_win">win</code></td>
<td>
<p>reading window size (def=100 variants per window)</p>
</td></tr>
<tr><td><code id="scanBED_+3A_iid">iid</code></td>
<td>
<p>option to read <code>N</code> IID as row names (def=1, see <code><a href="#topic+readIID">readIID()</a></code>).</p>
</td></tr>
<tr><td><code id="scanBED_+3A_vid">vid</code></td>
<td>
<p>option to read <code>P</code> VID as col names (def=1, see <code><a href="#topic+readVID">readVID()</a></code>).</p>
</td></tr>
<tr><td><code id="scanBED_+3A_vfr">vfr</code></td>
<td>
<p>variant-wise, from where to read (number/proportion, def=1)?</p>
</td></tr>
<tr><td><code id="scanBED_+3A_vto">vto</code></td>
<td>
<p>varinat-wise, to where then stop (number/proportion, def=P)?</p>
</td></tr>
<tr><td><code id="scanBED_+3A_buf">buf</code></td>
<td>
<p>buffer size in byptes (def=2^24, or 16 MB).</p>
</td></tr>
<tr><td><code id="scanBED_+3A_simplify">simplify</code></td>
<td>
<p>try simplifying the results  into an array, or leave them in
a list, or specify a function to simplify the said list.</p>
</td></tr>
<tr><td><code id="scanBED_+3A_exp">EXP</code></td>
<td>
<p>a R expression to evaluate with each window of variants;</p>
</td></tr>
<tr><td><code id="scanBED_+3A_gvr">GVR</code></td>
<td>
<p>a R variable name to assign the window to (def=&quot;g&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>results of all windows processed by the user script.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>scanBED()</code>: apply a function to variants in a PLINK1 BED fileset
</p>
<p>Travers <code>P</code>  variants via a sliding  window while calling a  function on each
window of variants without side effects on the calling environment, mimicking
various R <code>apply</code> utilities.
</p>
</li>
<li> <p><code>loopBED()</code>: evaluate an expression on variants in a PLINK1 BED
</p>
<p>Travers <code>P</code> variants via a sliding  window and evaluate an R expression given
each  window of  variants,  with  side effects  on  the calling  environment,
mimicking the syntax of R <code>for</code> loop.
</p>
</li></ul>


<h3>BED PLINK1 Binary Pedigree fileset</h3>

<p>A popular format to store biallelic dosage genotype, with three files,
</p>

<ul>
<li> <p><em>pfx</em>.fam: text table for <code>N</code> individuals, detailed in <a href="#topic+readFAM">readFAM</a>;
</p>
</li>
<li> <p><em>pfx</em>.bim: text table for <code>P</code> variants, detailed in <a href="#topic+readBIM">readBIM</a>;
</p>
</li>
<li> <p><em>pfx</em>.bed: transposed genotype matrix (<code>P</code> x <code>N</code> ) in binary format.
</p>
</li></ul>

<p>The triplets are commonly referred by the shared prefix (<code>pfx</code>), e.g., the X
chromosome represented by &quot;chrX.bed&quot;, &quot;chrX.fam&quot;, and &quot;chrX.bim&quot; are refered
by <code>"chrX"</code>.
</p>
<p>The binary file &quot;<em>pfx</em>.bed&quot; represent each dosage value with two bits - just
enough to encode all four possiblities: 0, 1, or 2 alleles, or missing.
</p>
<p>The number of variants (<code>P</code>) and samples (<code>N</code>) equals to the number of lines
in text file &quot;<em>pfx</em>.bim&quot; and &quot;<em>pfx</em>.fam&quot;, respectively.
</p>
<p>For the detailed specification of PLINK1 BED genotype format, see the lagecy
PLINK v1.07 page at: \
<a href="https://zzz.bwh.harvard.edu/plink/binary.shtml">https://zzz.bwh.harvard.edu/plink/binary.shtml</a>. \
For the modern use and management of PLINK1 BED, see the PLINK v1.9 page: \
<a href="https://www.cog-genomics.org/plink/1.9/input#bed">https://www.cog-genomics.org/plink/1.9/input#bed</a>.
</p>


<h3>detailed arguments</h3>


<ul>
<li> <p><code>win</code>: visiting window size.
</p>
<p>the number of variants per window, that  is, the number of columns in each
window matrix passed to the user script.
</p>
<p>For example, a size one window means  the user script will be dealing with
only one variant at  a time, received from in a matrix  of a single column
&ndash; a manner similar to  genome wide association analysis (GWAS).  However,
a larger, multi-variant window coupled with R language's vector and matrix
syntax can significantly boost efficiency.
</p>
<p>The default size is 1000 variants / columns per window.
</p>
</li>
<li> <p><code>buf</code>: buffer size in bytes
</p>
<p>a large buffer reduces the frequency of hard disk visits when traversing a
PLINK1 BED file, which in turn reduces non-computation overhead.
</p>
<p>The default size is <code>2^24</code> bytes, or 16 MB.
</p>
</li>
<li> <p><code>simplify</code>:
</p>
<p>when FALSE: resuts  of user script processing each window  of variants are
returned in a list;
</p>
<p>when TRUE,  use <code>simplify2array</code> to put  the results into an  array, if it
fails, fallback and return a list.
</p>
<p>when a function is specified, it is then used to simplify the results, if
an execption is thrown, fallback and return a list.
</p>
<p>e.g., the window script returns a  data frame of estimate, standard error,
t-statistic, and p-value  for each variant, <code>simplify =  rbind</code> to combine
results of all windows into one data frame of <code>P</code> rows and four columns of
statistics.
</p>
</li></ul>



<h3>genotype context</h3>

<p>context infomation  such the number of  variants and samples are  updated in
the window processing environment to ease user scripting, which includes:
</p>

<ul>
<li> <p><code>.i</code>: indies of variants in the current visiting window;
</p>
</li>
<li> <p><code>.p</code>: number of variants in the current visiting window.
</p>
</li>
<li> <p><code>.P</code>: total number of variants;
</p>
</li>
<li> <p><code>.w</code>: index of the current window;
</p>
</li>
<li> <p><code>.W</code>: total number of windows to go through;
</p>
</li>
<li> <p><code>.N</code>: number of individuals.
</p>
</li>
<li> <p><code>.b</code>: index of the current buffer.
</p>
</li>
<li> <p><code>.B</code>: number of buffers to be swapped.
</p>
</li></ul>

<p>e.g. (1) print percentage progress with <code>print(.w / .W * 100)</code>; \
e.g. (2) use <code>inf &lt;- readBIM(pfx)</code> to  read the table of variants before the
window visits,  later use <code>inf[.i,  ]</code> to  access meta-data for  variants in
each window.
</p>


<h3>See Also</h3>

<p><code style="white-space: pre;">&#8288;[readBED]&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## traverse genotype, apply R function without side effects
pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "000")
ret &lt;- scanBED(pfx, function(g)
{
    .af &lt;- colMeans(g, na.rm=TRUE) / 2
    maf &lt;- pmin(.af, 1 - .af)
    mis &lt;- colSums(is.na(g)) / .N
    pct &lt;- round(.w / .W * 100, 2)
    cbind(buf=.b, wnd=.w, idx=.i, MAF=maf, MIS=mis, PCT=pct)
},
vfr=NULL, vto=NULL, win=13, simplify=rbind, buf=2^18)
head(ret)
tail(ret)

## traversing genotype, evaluate R expression with side effects
pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "000.bed")
ret &lt;- list() # use side effect to keep the result of each window.
loopBED(pfx,
{
    af &lt;- colMeans(gt, na.rm=TRUE) / 2
    sg &lt;- af * (1 - af)
    ret[[.w]] &lt;- cbind(wnd=.w, alf=af, var=sg)
},
win=13, GVR="gt", vid=3, buf=2^18)
head(ret)
tail(ret)

</code></pre>

<hr>
<h2 id='testReadBED'>Test BED Reader</h2><span id='topic+testReadBED'></span>

<h3>Description</h3>

<p>Read m20 (bed, bim, and fam) under  &quot;extdata&quot; and compare with the content in
text file &quot;i10.txt&quot; converted from m20 by PLINK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testReadBED()
</code></pre>

<hr>
<h2 id='testReadBSM'>Test Genetic Relatedness Matrix Reader</h2><span id='topic+testReadBSM'></span>

<h3>Description</h3>

<p>Compare the read from genetic relatedness matrix created from the same genome
segment but stored in different shapes and types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testReadBSM()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
