<!DOCTYPE html><html><head><title>Help for package PLIS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PLIS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bwfw.hmm'>
<p>backward and forward inferences</p></a></li>
<li><a href='#bwfw1.hmm'>
<p>backward and forward inferences</p></a></li>
<li><a href='#em.hmm'>
<p>EM algorithm for HMM to estimate LIS statistic</p></a></li>
<li><a href='#GWAS.SampleData'>
<p>Sample GWAS Dataset</p></a></li>
<li><a href='#plis'>
<p>A multiple testing procedure based on pooled LIS statistics</p></a></li>
<li><a href='#PLIS-package'>
<p>PLIS</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multiplicity Control using Pooled LIS Statistic</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-10-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhi Wei &amp; Wenguang Sun</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhi Wei &lt;zhiwei04@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A multiple testing procedure for testing several groups of 
        hypotheses is implemented. Linear dependency among the hypotheses 
        within the same group is modeled by using hidden Markov Models. 
        It is noted that a smaller p value does not necessarily imply 
        more significance due to the dependency. A typical application is 
        to analyze genome wide association studies datasets, where SNPs 
        from the same chromosome are treated as a group and exhibit 
        strong linear genomic dependency. See Wei Z, Sun W, Wang K, 
        Hakonarson H (2009) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtp476">doi:10.1093/bioinformatics/btp476</a>&gt; for more details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-01 23:58:59 UTC; xl456</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-02 15:40:02 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='bwfw.hmm'>
backward and forward inferences
</h2><span id='topic+bwfw.hmm'></span>

<h3>Description</h3>

<p>When L&gt;1, calculate values for backward, forward variables, probabilities of hidden states. A supporting function called by em.hmm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bwfw.hmm(x, pii, A, pc, f0, f1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bwfw.hmm_+3A_x">x</code></td>
<td>

<p>the observed Z values
</p>
</td></tr>
<tr><td><code id="bwfw.hmm_+3A_pii">pii</code></td>
<td>

<p>(prob. of being 0, prob. of being 1), the initial state distribution 
</p>
</td></tr>
<tr><td><code id="bwfw.hmm_+3A_a">A</code></td>
<td>

<p>A=(a00 a01\\ a10 a11), transition matrix
</p>
</td></tr>
<tr><td><code id="bwfw.hmm_+3A_pc">pc</code></td>
<td>

<p>(c[1], ..., c[L])&ndash;the probability weights in the mixture for each component
</p>
</td></tr>
<tr><td><code id="bwfw.hmm_+3A_f0">f0</code></td>
<td>

<p>(mu, sigma), the parameters for null distribution
</p>
</td></tr>
<tr><td><code id="bwfw.hmm_+3A_f1">f1</code></td>
<td>
<p>(mu[1], sigma[1]\\...\\mu[L], sigma[L])&ndash;an L by 2 matrix, the parameter set for the non-null distribution
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>calculates values for backward, forward variables, probabilities of hidden states,<br />
&ndash;the lfdr variables and etc. <br />
&ndash;using the forward-backward procedure (Rabiner 89) <br />
&ndash;based on a sequence of observations for a given hidden markov model M=(pii, A, f) <br />
&ndash;see Sun and Cai (2009) for a detailed instruction on the coding of this algorithm
</p>


<h3>Value</h3>





<table>
<tr><td><code>alpha</code></td>
<td>
<p>rescaled backward variables</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>rescaled forward variables</p>
</td></tr>
<tr><td><code>lfdr</code></td>
<td>
<p>lfdr variables</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>probabilities of hidden states</p>
</td></tr>
<tr><td><code>dgamma</code></td>
<td>
<p>rescaled transition variables</p>
</td></tr>
<tr><td><code>omega</code></td>
<td>
<p>rescaled weight variables</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wei Z, Sun W, Wang K and Hakonarson H
</p>


<h3>References</h3>

<p>Multiple Testing in Genome-Wide Association Studies via Hidden Markov Models, Bioinformatics, 2009<br />
Large-scale multiple testing under dependence, Sun W and Cai T (2009), JRSSB, 71, 393-424<br />
A Tutorial on Hidden Markov Models and Selected Applications in Speech Recognition, Rabiner L (1989), Procedings of the IEEE, 77, 257-286.
</p>

<hr>
<h2 id='bwfw1.hmm'>
backward and forward inferences
</h2><span id='topic+bwfw1.hmm'></span>

<h3>Description</h3>

<p>When L=1, calculate values for backward, forward variables, probabilities of hidden states. A supporting function called by em.hmm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bwfw1.hmm(x, pii, A, f0, f1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bwfw1.hmm_+3A_x">x</code></td>
<td>

<p>the observed Z values
</p>
</td></tr>
<tr><td><code id="bwfw1.hmm_+3A_pii">pii</code></td>
<td>

<p>(prob. of being 0, prob. of being 1), the initial state distribution 
</p>
</td></tr>
<tr><td><code id="bwfw1.hmm_+3A_a">A</code></td>
<td>

<p>A=(a00 a01\\a10 a11), transition matrix
</p>
</td></tr>
<tr><td><code id="bwfw1.hmm_+3A_f0">f0</code></td>
<td>

<p>(mu, sigma), the parameters for null distribution
</p>
</td></tr>
<tr><td><code id="bwfw1.hmm_+3A_f1">f1</code></td>
<td>
<p>(mu[1], sigma[1]\\...\\mu[L], sigma[L])&ndash;an L by 2 matrix, the parameter set for the non-null distribution
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>calculates values for backward, forward variables, probabilities of hidden states,<br />
&ndash;the lfdr variables and etc. <br />
&ndash;using the forward-backward procedure (Rabiner 89) <br />
&ndash;based on a sequence of observations for a given hidden markov model M=(pii, A, f) <br />
&ndash;see Sun and Cai (2009) for a detailed instruction on the coding of this algorithm
</p>


<h3>Value</h3>





<table>
<tr><td><code>alpha</code></td>
<td>
<p>rescaled backward variables</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>rescaled forward variables</p>
</td></tr>
<tr><td><code>lfdr</code></td>
<td>
<p>lfdr variables</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>probabilities of hidden states</p>
</td></tr>
<tr><td><code>dgamma</code></td>
<td>
<p>rescaled transition variables</p>
</td></tr>
<tr><td><code>omega</code></td>
<td>
<p>rescaled weight variables</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wei Z, Sun W, Wang K and Hakonarson H
</p>


<h3>References</h3>

<p>Multiple Testing in Genome-Wide Association Studies via Hidden Markov Models, Bioinformatics, 2009<br />
Large-scale multiple testing under dependence, Sun W and Cai T (2009), JRSSB, 71, 393-424<br />
A Tutorial on Hidden Markov Models and Selected Applications in Speech Recognition, Rabiner L (1989), Procedings of the IEEE, 77, 257-286.
</p>

<hr>
<h2 id='em.hmm'>
EM algorithm for HMM to estimate LIS statistic
</h2><span id='topic+em.hmm'></span>

<h3>Description</h3>

<p>em.hmm calculates the MLE for a HMM model with hidden states being 0/1.
the distribution of observed Z values given state 0 is assumed to be normal and 
gvien state 1, is assumed to be a normal mixture with L components
</p>


<h3>Usage</h3>

<pre><code class='language-R'>em.hmm(x, L=2, maxiter = 1000, est.null = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="em.hmm_+3A_x">x</code></td>
<td>

<p>the observed Z values
</p>
</td></tr>
<tr><td><code id="em.hmm_+3A_l">L</code></td>
<td>

<p>the number of components in the non-null mixture, default value=2
</p>
</td></tr>
<tr><td><code id="em.hmm_+3A_maxiter">maxiter</code></td>
<td>

<p>the maximum number of iterations, default value=1000
</p>
</td></tr>
<tr><td><code id="em.hmm_+3A_est.null">est.null</code></td>
<td>

<p>logical. If FALSE (the default) set the null distribution as N(0,1), otherwise will estimate the null distribution.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<table>
<tr><td><code>pii</code></td>
<td>
<p>the initial state distribution, pii=(prob. of being 0, prob. of being 1)</p>
</td></tr>
<tr><td><code>A</code></td>
<td>
<p>transition matrix, A=(a00 a01\ a10 a11)</p>
</td></tr>
<tr><td><code>f0</code></td>
<td>
<p>the null distribution</p>
</td></tr>
<tr><td><code>pc</code></td>
<td>
<p>probability weights of each component in the non-null mixture</p>
</td></tr> 
<tr><td><code>f1</code></td>
<td>
<p>an L by 2 matrix, specifying the dist. of each component in the non-null mixture</p>
</td></tr> 
<tr><td><code>LIS</code></td>
<td>
<p>the LIS statistics</p>
</td></tr> 
<tr><td><code>ni</code></td>
<td>
<p>the number of iterations excecuted</p>
</td></tr> 
<tr><td><code>logL</code></td>
<td>
<p>log likelihood</p>
</td></tr> 
<tr><td><code>BIC</code></td>
<td>
<p>BIC score for the estimated model</p>
</td></tr> 
<tr><td><code>converged</code></td>
<td>
<p>Logic, Convergence indicator of the EM procedure</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wei Z, Sun W, Wang K and Hakonarson H
</p>


<h3>References</h3>

<p>Multiple Testing in Genome-Wide Association Studies via Hidden Markov Models, Bioinformatics, 2009
</p>


<h3>See Also</h3>

<p>plis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##(1) Example for analyzing simulated data
grp1.nonNull.loci=c(21:30, 51:60); grp2.nonNull.loci=c(41:60)
grp1.theta&lt;-grp2.theta&lt;-rep(0,200)
grp1.theta[grp1.nonNull.loci]=2; grp2.theta[grp2.nonNull.loci]=2

grp1.zval=rnorm(n=length(grp1.theta),mean=grp1.theta)
grp2.zval=rnorm(n=length(grp2.theta),mean=grp2.theta)
##Group 1
#Use default L=2
grp1.L2rlts=em.hmm(grp1.zval)
#Use true value L=1
grp1.L1rlts=em.hmm(grp1.zval,L=1)
#Choose L by BIC criteria
grp1.Allrlts=sapply(1:3, function(k) em.hmm(grp1.zval,L=k))
BICs=c()
for(i in 1:3) {
  BICs=c(BICs,grp1.Allrlts[[i]]$BIC)
}
grp1.BICrlts=grp1.Allrlts[[which(BICs==max(BICs))]]

rank(grp1.BICrlts$LIS)[grp1.nonNull.loci]
rank(-abs(grp1.zval))[grp1.nonNull.loci]

##Group 2
grp2.Allrlts=sapply(1:3, function(k) em.hmm(grp2.zval,L=k))
BICs=c()
for(i in 1:3) {
  BICs=c(BICs,grp2.Allrlts[[i]]$BIC)
}
grp2.BICrlts=grp2.Allrlts[[which(BICs==max(BICs))]]

rank(grp2.BICrlts$LIS)[grp2.nonNull.loci]
rank(-abs(grp2.zval))[grp2.nonNull.loci]

##PLIS: control global FDR
states=plis(c(grp1.BICrlts$LIS,grp2.BICrlts$LIS),fdr=0.1,adjust=FALSE)$States
#0 accept; 1 reject under fdr level 0.1

##(2) Example for analyzing Genome-Wide Association Studies (GWAS) data
#Information in GWAS.SampleData can be obtained by using PLINK
#http://pngu.mgh.harvard.edu/~purcell/plink/

#not running
#please uncomment to run
#
#data(GWAS.SampleData)
#
#chr1.data=GWAS.SampleData[which(GWAS.SampleData[,"CHR"]==1),]
#chr6.data=GWAS.SampleData[which(GWAS.SampleData[,"CHR"]==6),]
#
##Make sure SNPs in the linear physical order
#chr1.data&lt;-chr1.data[order(chr1.data[,"BP"]),]
#chr6.data&lt;-chr6.data[order(chr6.data[,"BP"]),]
#
##convert p values by chi_sq test to z values; odds ratio (OR) is needed.
#chr1.zval&lt;-rep(0, nrow(chr1.data))
#chr1.ors=(chr1.data[,"OR"]&gt;1)
#chr1.zval[chr1.ors]&lt;-qnorm(chr1.data[chr1.ors, "P"]/2, 0, 1, lower.tail=FALSE)
#chr1.zval[!chr1.ors]&lt;-qnorm(chr1.data[!chr1.ors, "P"]/2, 0, 1, lower.tail=TRUE)
#chr1.L2rlts=em.hmm(chr1.zval)
#
#chr6.zval&lt;-rep(0, nrow(chr6.data))
#chr6.ors=(chr6.data[,"OR"]&gt;1)
#chr6.zval[chr6.ors]&lt;-qnorm(chr6.data[chr6.ors, "P"]/2, 0, 1, lower.tail=FALSE)
#chr6.zval[!chr6.ors]&lt;-qnorm(chr6.data[!chr6.ors, "P"]/2, 0, 1, lower.tail=TRUE)
#chr6.L2rlts=em.hmm(chr6.zval)
#
##Note that for analyzing a chromosome in real GWAS dataset, em.hmm can take as long as 10+ hrs
##L=2 or 3 is recommended for GWAS based on our experience
##em.hmm can be run in parallel for different chromosomes before applying the PLIS procedure
#plis.rlts=plis(c(chr1.L2rlts$LIS,chr6.L2rlts$LIS),fdr=0.01)
#all.Rlts=cbind(rbind(chr1.data,chr6.data), LIS=c(chr1.L2rlts$LIS,chr6.L2rlts$LIS), 
#gFDR=plis.rlts$aLIS, fdr001state=plis.rlts$States)
#all.Rlts[order(all.Rlts[,"LIS"])[1:10],]

</code></pre>

<hr>
<h2 id='GWAS.SampleData'>
Sample GWAS Dataset
</h2><span id='topic+GWAS.SampleData'></span>

<h3>Description</h3>

<p>Sample GWAS Dataset with 400 SNPs from Chromosome 1 and 6 (200 SNPs each).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GWAS.SampleData)</code></pre>


<h3>Format</h3>

<p>A data frame with 400 observations on the following 6 variables.
</p>

<dl>
<dt><code>CHR</code></dt><dd><p>Chromosome ID</p>
</dd>
<dt><code>SNP</code></dt><dd><p>rs Id</p>
</dd>
<dt><code>BP</code></dt><dd><p>Phisical Position</p>
</dd>
<dt><code>OR</code></dt><dd><p>Odds Ratio</p>
</dd>
<dt><code>CHISQ</code></dt><dd><p>1 d.f. Chi Square test Statistic</p>
</dd>
<dt><code>P</code></dt><dd><p>P value of 1 d.f. Chi Square test Statistic</p>
</dd>
</dl>



<h3>Details</h3>

<p>The required values (Odds ratio and P value) can be calculated by using PLINK
</p>


<h3>References</h3>

<p>Supplementary Material of Multiple Testing in Genome-Wide Association Studies via Hidden Markov Models, Bioinformatics, 2009
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWAS.SampleData)
</code></pre>

<hr>
<h2 id='plis'>
A multiple testing procedure based on pooled LIS statistics
</h2><span id='topic+plis'></span>

<h3>Description</h3>

<p>It controls the global FDR for the pooled hypotheses from different groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plis(lis, fdr = 0.001, adjust = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plis_+3A_lis">lis</code></td>
<td>

<p>pooled LIS statistics estimated from different groups
</p>
</td></tr>
<tr><td><code id="plis_+3A_fdr">fdr</code></td>
<td>

<p>nominal fdr level you want to control 
</p>
</td></tr>
<tr><td><code id="plis_+3A_adjust">adjust</code></td>
<td>

<p>logical. If TRUE (the default), will calculate and return &quot;adjusted&quot; LIS value&ndash; the corresponding global FDR if using the LIS statistic as the significance cutoff. It may take hours if you have hundreds of thousands LISs to adjust.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>States</code></td>
<td>
<p>state sequence indicating if the hypotheses should be rejected or not: 0 accepted , 1 rejected</p>
</td></tr>
<tr><td><code>aLIS</code></td>
<td>
<p>the corresponding global FDR if using the LIS statistic as the significance cutoff </p>
</td></tr>	
</table>


<h3>Author(s)</h3>

<p>Wei Z, Sun W, Wang K and Hakonarson H
</p>


<h3>References</h3>

<p>Multiple Testing in Genome-Wide Association Studies via Hidden Markov Models, Bioinformatics, 2009
</p>


<h3>See Also</h3>

<p>see em.hmm for examples
</p>

<hr>
<h2 id='PLIS-package'>
PLIS
</h2><span id='topic+PLIS-package'></span><span id='topic+PLIS'></span>

<h3>Description</h3>

<p>PLIS is a multiple testing procedure for testing several groups of hypotheses. Linear dependency is expected from the hypotheses within the same group and is modeled by hidden Markov Models.
It is noted that, for PLIS, a smaller p value does not necessarily imply more significance because of dependency among the hypotheses. A typical applicaiton of PLIS is to analyze genome wide association studies datasets, where SNPs from the same chromosome are treated as a group and exhibit strong linear genomic dependency. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PLIS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2012-08-08</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>main functions: em.hmm &amp; plis
</p>


<h3>Author(s)</h3>

<p>Wei Z, Sun W, Wang K and Hakonarson H<br />
Maintainer: Zhi Wei &lt;zhiwei04@gmail.com&gt;
</p>


<h3>References</h3>

<p>Wei Z, Sun W, Wang K and Hakonarson H, Multiple Testing in Genome-Wide Association Studies via Hidden Markov Models, Bioinformatics, 2009
</p>


<h3>See Also</h3>

<p>p.adjust(), in which the traditional procedures are implemented. The adjustment made by p.adjust will not change the original ranking based on the given p values. However, taking into account dependency, PLIS may generate a ranking different from that by p value.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
