<!DOCTYPE html><html lang="en"><head><title>Help for package plotthis</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {plotthis}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.anno_ggcat'><p>Heatmap annotation function for categorical data</p></a></li>
<li><a href='#.anno_ggseries'><p>Heatmap annotation functions</p></a></li>
<li><a href='#adjcolors'><p>Convert a color with arbitrary transparency to a fixed color</p></a></li>
<li><a href='#adjust_network_layout'><p>Adjust_network_layout</p></a></li>
<li><a href='#AreaPlot'><p>Area plot</p></a></li>
<li><a href='#AreaPlotAtomic'><p>Atomic area plot</p></a></li>
<li><a href='#BarPlot'><p>Bar Plot</p></a></li>
<li><a href='#BarPlotAtomic'><p>Atomic bar plot</p></a></li>
<li><a href='#BarPlotGrouped'><p>Bar plot with groups</p></a></li>
<li><a href='#BarPlotSingle'><p>BarPlotSingle</p></a></li>
<li><a href='#bg_layer'><p>Get a ggplot layer for background</p></a></li>
<li><a href='#blend_colors'><p>Blend colors</p></a></li>
<li><a href='#blend_rgblist'><p>Blend a list of colors</p></a></li>
<li><a href='#blend_to_color'><p>Blend two colors</p></a></li>
<li><a href='#BoxPlot'><p>Box / Violin Plot</p></a></li>
<li><a href='#BoxViolinPlot'><p>Box/Violin plot</p></a></li>
<li><a href='#BoxViolinPlotAtomic'><p>Atomic Box/Violin plot</p></a></li>
<li><a href='#calc_just'><p>Calculate hjust and vjust based on angle</p></a></li>
<li><a href='#check_columns'><p>Check the columns if columns found in the data</p></a></li>
<li><a href='#check_palcolor'><p>check_palcolor</p>
Check if the palcolor can be properly used</a></li>
<li><a href='#check_palette'><p>check_palette</p>
Check if the palette can be properly used</a></li>
<li><a href='#ChordPlot'><p>Chord / Circos plot</p></a></li>
<li><a href='#ChordPlotAtomic'><p>Atomic chord plot</p></a></li>
<li><a href='#ClustreePlot'><p>Clustree Plot</p></a></li>
<li><a href='#ClustreePlotAtomic'><p>Atomic function for clustree plot</p></a></li>
<li><a href='#combine_plots'><p>Combine plots into one</p></a></li>
<li><a href='#common_args'><p>Common arguments for plots</p></a></li>
<li><a href='#CorPairsPlot'><p>CorPairsPlot</p></a></li>
<li><a href='#CorPairsPlotAtomic'><p>Atomic Correlation Pairs Plot</p></a></li>
<li><a href='#CorPlot'><p>CorPlot</p></a></li>
<li><a href='#CorPlotAtomic'><p>Atomic Correlation Plot</p></a></li>
<li><a href='#DensityHistoPlotAtomic'><p>Atomic density/histogram plot</p></a></li>
<li><a href='#DensityPlot'><p>Density Plot / Histogram</p></a></li>
<li><a href='#detect_upset_datatype'><p>Detect the type of the input data for Upset plot</p></a></li>
<li><a href='#detect_venn_datatype'><p>Detect the datatype of the input data of Venn diagram</p></a></li>
<li><a href='#DimPlot'><p>DimPLot / FeatureDimPlot</p></a></li>
<li><a href='#DimPlotAtomic'><p>Atomic Dimension Reduction Plot without splitting the data</p></a></li>
<li><a href='#DotPlot'><p>Dot Plot / Scatter Plot / Lollipop Plot</p></a></li>
<li><a href='#DotPlotAtomic'><p>Dot Plot without splitting the data</p></a></li>
<li><a href='#element_textbox'><p>Theme element that add a box to the text</p></a></li>
<li><a href='#enrich_example'><p>A example of clusterProfiler enrichment result</p></a></li>
<li><a href='#enrich_multidb_example'><p>A example of clusterProfiler enrichment result with multiple databases</p></a></li>
<li><a href='#EnrichMapAtomic'><p>Atomic Enrichment Map</p></a></li>
<li><a href='#EnrichNetworkAtomic'><p>Atomic Enrichment Network</p></a></li>
<li><a href='#facet_plot'><p>Facetting a plot</p></a></li>
<li><a href='#get_cutoffs_data'><p>Prepare the cutoff data for the ROC curve</p></a></li>
<li><a href='#gggrob'><p>Get the grid.draw-able ggplot grob</p>
The output from ggplotGrob can not be directly used in grid.draw, the position can not be set.
This function extracts the gTree from the ggplot grob.</a></li>
<li><a href='#gsea_example'><p>A example of GSEA result from fgsea package</p></a></li>
<li><a href='#gsea_running_score'><p>Get the running enrichment score of a gene set</p></a></li>
<li><a href='#GSEAPlotAtomic'><p>GSEA plot for a single term</p></a></li>
<li><a href='#Heatmap'><p>Heatmap</p></a></li>
<li><a href='#HeatmapAtomic'><p>Atomic heatmap</p></a></li>
<li><a href='#layer_white_bg'><p>Heatmap layer functions used to draw on the heatmap cells</p></a></li>
<li><a href='#LinePlot'><p>Line Plot</p></a></li>
<li><a href='#LinePlotAtomic'><p>LinePlotAtomic</p></a></li>
<li><a href='#LinePlotGrouped'><p>LinePlotGrouped</p></a></li>
<li><a href='#LinePlotSingle'><p>LinePlotSingle</p></a></li>
<li><a href='#Network'><p>Network</p></a></li>
<li><a href='#NetworkAtomic'><p>NetworkAtomic</p></a></li>
<li><a href='#norm_expansion'><p>Expand the plot area with CSS-like padding</p></a></li>
<li><a href='#palette_list'><p>A list of palettes for use in data visualization</p></a></li>
<li><a href='#palette_this'><p>Color palettes collected in plotthis.</p></a></li>
<li><a href='#PieChart'><p>Pie Chart</p></a></li>
<li><a href='#PieChartAtomic'><p>Pie chart without data splitting</p></a></li>
<li><a href='#PrepareEnrichrResult'><p>Enrichment Map/Network</p></a></li>
<li><a href='#PrepareFGSEAResult'><p>GSEA plots</p></a></li>
<li><a href='#PrepareUpsetData'><p>Upset Plot</p></a></li>
<li><a href='#PrepareVennData'><p>VennDiagram</p></a></li>
<li><a href='#process_theme'><p>Process theme to allow 'ggplot2::theme_minimal' to work</p></a></li>
<li><a href='#QQPlot'><p>QQ plot</p></a></li>
<li><a href='#QQPlotAtomic'><p>Atomic QQ plot</p></a></li>
<li><a href='#RadarPlot'><p>Radar plot / Spider plot</p></a></li>
<li><a href='#RadarPlotAtomic'><p>Atomic Radar plot</p></a></li>
<li><a href='#RarefactionPlot'><p>RarefactionPlot</p></a></li>
<li><a href='#RarefactionPlotAtomic'><p>RarefactionPlotAtomic</p></a></li>
<li><a href='#rgba_to_rgb'><p>Convert RGBA to RGB</p></a></li>
<li><a href='#RidgePlot'><p>Ridge Plot</p></a></li>
<li><a href='#RidgePlotAtomic'><p>Atomic ridge plot</p></a></li>
<li><a href='#RingPlot'><p>Ring Plot</p></a></li>
<li><a href='#RingPlotAtomic'><p>RingPlotAtomic</p></a></li>
<li><a href='#ROCCurve'><p>ROC curve</p></a></li>
<li><a href='#ROCCurveAtomic'><p>Atomic ROC curve</p></a></li>
<li><a href='#SankeyPlot'><p>Sankey / Alluvial Plot</p></a></li>
<li><a href='#SankeyPlotAtomic'><p>Atomic Sankey plot</p></a></li>
<li><a href='#ScatterPlot'><p>Scatter Plot</p></a></li>
<li><a href='#ScatterPlotAtomic'><p>Scatter Plot Atomic</p></a></li>
<li><a href='#show_palettes'><p>Show the color palettes</p></a></li>
<li><a href='#SplitBarPlotAtomic'><p>SplitBarPlotAtomic</p></a></li>
<li><a href='#theme_blank'><p>Blank theme</p></a></li>
<li><a href='#theme_this'><p>A ggplot2 theme and palettes for plotthis</p>
Borrowed from the <code>theme_this</code> function in the <code>SCP</code> pipeline</a></li>
<li><a href='#TrendPlot'><p>Trend plot</p></a></li>
<li><a href='#TrendPlotAtomic'><p>Atomic trend plot</p></a></li>
<li><a href='#UpsetPlotAtomic'><p>Atomic Upset plot</p></a></li>
<li><a href='#validate_common_args'><p>Validate common arguments</p></a></li>
<li><a href='#VennDiagramAtomic'><p>Atomic Venn diagram</p></a></li>
<li><a href='#VolcanoPlot'><p>Volcano plot</p></a></li>
<li><a href='#VolcanoPlotAtomic'><p>Atomic volcano plot</p></a></li>
<li><a href='#WordCloudPlot'><p>Word Cloud Plot</p></a></li>
<li><a href='#WordCloudPlotAtomic'><p>Word cloud without data splitting</p></a></li>
<li><a href='#words_excluded'><p>Excluded words in keyword enrichment analysis and extraction</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>High-Level Plotting Built Upon 'ggplot2' and Other Plotting
Packages</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides high-level API and a wide range of options to create stunning, publication-quality plots effortlessly.
    It is built upon 'ggplot2' and other plotting packages, and is designed to be easy to use and to work seamlessly with 'ggplot2' objects.
    It is particularly useful for creating complex plots with multiple layers, facets, and annotations.
    It also provides a set of functions to create plots for specific types of data, such as Venn diagrams, alluvial diagrams, and phylogenetic trees.
    The package is designed to be flexible and customizable, and to work well with the 'ggplot2' ecosystem.
    The API can be found at <a href="https://pwwang.github.io/plotthis/reference/index.html">https://pwwang.github.io/plotthis/reference/index.html</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/pwwang/plotthis">https://github.com/pwwang/plotthis</a>
<a href="https://pwwang.github.io/plotthis/">https://pwwang.github.io/plotthis/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/pwwang/plotthis/issues">https://github.com/pwwang/plotthis/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>circlize, ggplot2, rlang, dplyr, tidyr, glue, forcats, gtable,
reshape2, stringr, scales, gridtext, methods, patchwork,
ggrepel, ggnewscale, cowplot, zoo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, alluvial, datasets, ComplexHeatmap, clustree,
gglogger, ggwordcloud, ggalluvial, ggVennDiagram (&ge; 1.5.0),
ggupset, ggpubr, ggforce, ggraph, ggridges, qqplotr, hexbin,
igraph, iNEXT, scattermore, concaveman, plotROC,
OptimalCutpoints</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-16 07:54:28 UTC; pwwang</td>
</tr>
<tr>
<td>Author:</td>
<td>Panwen Wang <a href="https://orcid.org/0000-0002-4614-8970"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Panwen Wang &lt;pwwang@pwwang.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-16 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.anno_ggcat'>Heatmap annotation function for categorical data</h2><span id='topic+.anno_ggcat'></span>

<h3>Description</h3>

<p>Heatmap annotation function for categorical data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.anno_ggcat(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  .plotting,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".anno_ggcat_+3A_x">x</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_split_by">split_by</code></td>
<td>
<p>A character string of the column name to split the data</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_group_by">group_by</code></td>
<td>
<p>A character string of the column name to group the data</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_column">column</code></td>
<td>
<p>A character string of the column name to plot</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_title">title</code></td>
<td>
<p>A character string to name the legend</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_which">which</code></td>
<td>
<p>A character string specifying the direction of the annotation. Default is &quot;row&quot;.
Other options are &quot;column&quot;.</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette of the annotation</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_palcolor">palcolor</code></td>
<td>
<p>A character vector of colors to override the palette</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_border">border</code></td>
<td>
<p>A logical value indicating whether to draw the border of the annotation</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend. Default is &quot;vertical&quot;.
Other options are &quot;horizontal&quot;.</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_show_legend">show_legend</code></td>
<td>
<p>A logical value indicating whether to show the legend</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_.plotting">.plotting</code></td>
<td>
<p>A function to create the plot for each split and each group</p>
</td></tr>
<tr><td><code id=".anno_ggcat_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>ComplexHeatmap::AnnotationFunction</code></p>
</td></tr>
</table>

<hr>
<h2 id='.anno_ggseries'>Heatmap annotation functions</h2><span id='topic+.anno_ggseries'></span><span id='topic+anno_pie'></span><span id='topic+anno_ring'></span><span id='topic+anno_bar'></span><span id='topic+anno_violin'></span><span id='topic+anno_boxplot'></span><span id='topic+anno_density'></span><span id='topic+anno_simple'></span><span id='topic+anno_points'></span><span id='topic+anno_lines'></span>

<h3>Description</h3>

<p>Heatmap annotation functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.anno_ggseries(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  .plotting,
  ...
)

anno_pie(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  ...
)

anno_ring(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  ...
)

anno_bar(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  ...
)

anno_violin(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  ...
)

anno_boxplot(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  ...
)

anno_density(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  ...
)

anno_simple(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  alpha = 1,
  ...
)

anno_points(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  alpha = 1,
  ...
)

anno_lines(
  x,
  split_by = NULL,
  group_by,
  column,
  title,
  which = "row",
  palette,
  palcolor = NULL,
  border = TRUE,
  legend.direction,
  show_legend = TRUE,
  alpha = 1,
  add_points = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".anno_ggseries_+3A_x">x</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_split_by">split_by</code></td>
<td>
<p>A character string of the column name to split the data (heatmap)</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_group_by">group_by</code></td>
<td>
<p>A character string of the column name to group the data (rows or columns of the heatmap)</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_column">column</code></td>
<td>
<p>A character string of the column name of the data <code>x</code> to plot</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_title">title</code></td>
<td>
<p>A character string to name the legend</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_which">which</code></td>
<td>
<p>A character string specifying the direction of the annotation. Default is &quot;row&quot;.
Other options are &quot;column&quot;.</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette of the annotation</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_palcolor">palcolor</code></td>
<td>
<p>A character vector of colors to override the palette</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_border">border</code></td>
<td>
<p>A logical value indicating whether to draw the border of the annotation</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend. Default is &quot;vertical&quot;.
Other options are &quot;horizontal&quot;.</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_show_legend">show_legend</code></td>
<td>
<p>A logical value indicating whether to show the legend</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_.plotting">.plotting</code></td>
<td>
<p>A function to create the plot for each split and each group</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>ComplexHeatmap::AnnotationFunction</code>
The parameters passed to <code>row_annotation_params</code> and <code>column_annotation_params</code> will be passed here.</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value between 0 and 1 specifying the transparency of the annotation</p>
</td></tr>
<tr><td><code id=".anno_ggseries_+3A_add_points">add_points</code></td>
<td>
<p>A logical value indicating whether to add points to the annotation</p>
</td></tr>
</table>

<hr>
<h2 id='adjcolors'>Convert a color with arbitrary transparency to a fixed color</h2><span id='topic+adjcolors'></span>

<h3>Description</h3>

<p>This function takes a vector of colors and an alpha level and converts the colors
to fixed colors with the specified alpha level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjcolors(colors, alpha)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjcolors_+3A_colors">colors</code></td>
<td>
<p>Color vectors.</p>
</td></tr>
<tr><td><code id="adjcolors_+3A_alpha">alpha</code></td>
<td>
<p>Alpha level ranging from 0 to 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The colors with the specified alpha level.
</p>

<hr>
<h2 id='adjust_network_layout'>Adjust_network_layout</h2><span id='topic+adjust_network_layout'></span>

<h3>Description</h3>

<p>Adjust_network_layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_network_layout(
  graph,
  layout,
  width,
  height = 2,
  scale = 100,
  iter = 100
)
</code></pre>

<hr>
<h2 id='AreaPlot'>Area plot</h2><span id='topic+AreaPlot'></span>

<h3>Description</h3>

<p>A plot showing how one or more groups' numeric values change over the
progression of a another variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AreaPlot(
  data,
  x,
  y = NULL,
  x_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  scale_y = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AreaPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of fill.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_scale_y">scale_y</code></td>
<td>
<p>A logical value to scale the y-axis by the total number in each x-axis group.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="AreaPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(
    x = rep(c("A", "B", "C", "D"), 2),
    y = c(1, 3, 6, 4, 2, 5, 7, 8),
    group = rep(c("F1", "F2"), each = 4)
)
AreaPlot(data, x = "x", y = "y", group_by = "group")
AreaPlot(data, x = "x", y = "y", group_by = "group",
         scale_y = TRUE)
AreaPlot(data, x = "x", y = "y", split_by = "group")
AreaPlot(data, x = "x", y = "y", split_by = "group", palette = c(F1 = "Blues", F2 = "Reds"))
</code></pre>

<hr>
<h2 id='AreaPlotAtomic'>Atomic area plot</h2><span id='topic+AreaPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic area plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AreaPlotAtomic(
  data,
  x,
  y = NULL,
  x_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  scale_y = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AreaPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string of the column name to plot on the x-axis.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis.
A numeric column is expected.
If NULL, the count of the x-axis column will be used.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character vector of column names to fill the area plot by.
If NULL, the plot will be filled by the first color of the palette.
If multiple columns are provided, the columns will be concatenated with
<code>group_by_sep</code> and used as the fill column.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to separate the columns in <code>group_by</code>.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of fill.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_scale_y">scale_y</code></td>
<td>
<p>A logical value to scale the y-axis by the total number in each x-axis group.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="AreaPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='BarPlot'>Bar Plot</h2><span id='topic+BarPlot'></span><span id='topic+SplitBarPlot'></span><span id='topic+WaterfallPlot'></span>

<h3>Description</h3>


<ul>
<li> <p><code>BarPlot</code> is used to create a bar plot.
</p>
</li>
<li> <p><code>SplitBarPlot</code> (a.k.a <code>WaterfallPlot</code>) is used to create a bar plot with splitting the bars on the two sides.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>BarPlot(
  data,
  x,
  x_sep = "_",
  y = NULL,
  flip = FALSE,
  fill_by_x_if_no_group = TRUE,
  line_name = NULL,
  label_nudge = 0,
  label = NULL,
  label_fg = "black",
  label_size = 4,
  label_bg = "white",
  label_bg_r = 0.1,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  split_by = NULL,
  split_by_sep = "_",
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  facet_args = list(),
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  add_trend = FALSE,
  trend_color = "black",
  trend_linewidth = 1,
  trend_ptsize = 2,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  x_text_angle = 0,
  aspect.ratio = 1,
  y_min = NULL,
  y_max = NULL,
  position = "auto",
  position_dodge_preserve = "total",
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  expand = waiver(),
  width = waiver(),
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)

SplitBarPlot(
  data,
  x,
  y,
  y_sep = "_",
  flip = FALSE,
  split_by = NULL,
  split_by_sep = "_",
  alpha_by = NULL,
  alpha_reverse = FALSE,
  alpha_name = NULL,
  order_y = list(`+` = c("x_desc", "alpha_desc"), `-` = c("x_desc", "alpha_asc")),
  bar_height = 0.9,
  lineheight = 0.5,
  max_charwidth = 80,
  fill_by = NULL,
  fill_by_sep = "_",
  fill_name = NULL,
  direction_pos_name = "positive",
  direction_neg_name = "negative",
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "free_y",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  x_min = NULL,
  x_max = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)

WaterfallPlot(
  data,
  x,
  y,
  y_sep = "_",
  flip = FALSE,
  split_by = NULL,
  split_by_sep = "_",
  alpha_by = NULL,
  alpha_reverse = FALSE,
  alpha_name = NULL,
  order_y = list(`+` = c("x_desc", "alpha_desc"), `-` = c("x_desc", "alpha_asc")),
  bar_height = 0.9,
  lineheight = 0.5,
  max_charwidth = 80,
  fill_by = NULL,
  fill_by_sep = "_",
  fill_name = NULL,
  direction_pos_name = "positive",
  direction_neg_name = "negative",
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "free_y",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  x_min = NULL,
  x_max = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BarPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_fill_by_x_if_no_group">fill_by_x_if_no_group</code></td>
<td>
<p>A logical value indicating whether to fill the bars by the x-axis values if there is no group_by.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_line_name">line_name</code></td>
<td>
<p>A character string indicating the name of the line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_label_nudge">label_nudge</code></td>
<td>
<p>A numeric value to nudge the labels (the distance between the label and the top of the bar).</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_label">label</code></td>
<td>
<p>A column name for the values to be displayed on the top of the bars.
If TRUE, the y values will be displayed.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string indicating the color of the label.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value indicating the size of the label.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string indicating the background color of the label.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value indicating the radius of the background.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="BarPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to specify the name of the group_by in the legend.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="BarPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BarPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BarPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_facet_args">facet_args</code></td>
<td>
<p>A list of arguments to pass to <a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid</a> or <a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap</a>.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character string indicating the palette to use for the background.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character string indicating the color to use for the background.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value indicating the alpha of the background.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_add_line">add_line</code></td>
<td>
<p>A numeric value indicating the y value to add a horizontal line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_line_color">line_color</code></td>
<td>
<p>A character string indicating the color of the line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value indicating the size of the line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value indicating the type of the line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value to specify the minimum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value to specify the maximum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_position">position</code></td>
<td>
<p>A character string indicating the position of the bars.
If &quot;auto&quot;, the position will be &quot;stack&quot; if group_by has more than 5 levels, otherwise &quot;dodge&quot;.
&quot;fill&quot; is also a valid option. Only works when group_by is not NULL.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_position_dodge_preserve">position_dodge_preserve</code></td>
<td>
<p>Should dodging preserve the &quot;total&quot; width of all elements at a position, or the width of a &quot;single&quot; element?</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="BarPlot_+3A_width">width</code></td>
<td>
<p>A numeric value specifying the width of the bars.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_y_sep">y_sep</code></td>
<td>
<p>A character string to concatenate the x columns if there are multiple.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_alpha_by">alpha_by</code></td>
<td>
<p>A character string indicating the column name to use for the transparency of the bars.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_alpha_reverse">alpha_reverse</code></td>
<td>
<p>A logical value indicating whether to reverse the transparency.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_alpha_name">alpha_name</code></td>
<td>
<p>A character string indicating the legend name of the transparency.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_order_y">order_y</code></td>
<td>
<p>A list of character strings indicating the order of the y axis.
The keys are &quot;+&quot;, &quot;-&quot;, or &quot;<em>&quot;. However, &quot;+/-&quot; should not be mixed with &quot;</em>&quot;.
The values are &quot;x_asc&quot;, &quot;x_desc&quot;, &quot;alpha_asc&quot;, or &quot;alpha_desc&quot;, indicating how to order the y axis.
The default is <code>list("+" = c("x_desc", "alpha_desc"), "-" = c("x_desc", "alpha_asc"))</code>, meaning
the positive values are ordered by the x-axis values in descending order and the alpha values in descending order,
and the negative values are ordered by the x-axis values in descending order and the alpha values in ascending order.
The &quot;*&quot; key is used to order the y axis without considering the direction.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_bar_height">bar_height</code></td>
<td>
<p>A numeric value indicating the height of the bars.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_lineheight">lineheight</code></td>
<td>
<p>A numeric value indicating the height of the text.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_max_charwidth">max_charwidth</code></td>
<td>
<p>A numeric value indicating the maximum width of the text.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_fill_by">fill_by</code></td>
<td>
<p>A character string indicating the column name to use for the fill of the bars.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_fill_by_sep">fill_by_sep</code></td>
<td>
<p>A character string to concatenate the fill columns if there are multiple.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_fill_name">fill_name</code></td>
<td>
<p>A character string indicating the legend name of the fill.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_direction_pos_name">direction_pos_name</code></td>
<td>
<p>A character string indicating the name of the positive direction.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_direction_neg_name">direction_neg_name</code></td>
<td>
<p>A character string indicating the name of the negative direction.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_x_min">x_min</code></td>
<td>
<p>A numeric value indicating the minimum value of the x axis.</p>
</td></tr>
<tr><td><code id="BarPlot_+3A_x_max">x_max</code></td>
<td>
<p>A numeric value indicating the maximum value of the x axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(
    x = c("A", "B", "C", "D", "E", "F", "G", "H"),
    y = c(10, 8, 16, 4, 6, 12, 14, 2),
    group = c("G1", "G1", "G2", "G2", "G3", "G3", "G4", "G4"),
    facet = c("F1", "F2", "F3", "F4", "F1", "F2", "F3", "F4")
)

BarPlot(data, x = "x", y = "y")
BarPlot(data, x = "x", y = "y", fill_by_x_if_no_group = FALSE)
BarPlot(data, x = "x", y = "y", label = TRUE)
BarPlot(data, x = "x", y = "y", label = "facet", label_nudge = 1)
BarPlot(data, x = "group", y = "y", group_by = "x")
BarPlot(data,
    x = "group", y = "y", group_by = "x",
    position = "dodge", add_bg = TRUE
)
BarPlot(data,
    x = "x", y = "y", split_by = "group",
    facet_by = "facet", position = "dodge", facet_ncol = 1
)
BarPlot(data,
    x = "x", y = "y", split_by = "group",
    palette = list(G1 = "Reds", G2 = "Blues", G3 = "Greens", G4 = "Purp"),
    facet_by = "facet", position = "dodge", facet_ncol = 1
)
BarPlot(data,
    x = "group", y = "y", group_by = "x",
    position = "dodge", add_bg = TRUE, bg_palette = "Spectral"
)
# use the count
BarPlot(data, x = "group", ylab = "count")
# flip the plot
BarPlot(data, x = "group", flip = TRUE, ylab = "count")
data &lt;- data.frame(
    word = c("apple", "banana", "cherry", "date", "elderberry",
             "It is a very long term with a lot of words"),
    count = c(-10, 20, -30, 40, 50, 34),
    score = c(1, 2, 3, 4, 5, 3.2),
    group = c("A", "A", "B", "B", "C", "C")
)
SplitBarPlot(data, x = "count", y = "word", alpha_by = "score")
SplitBarPlot(data, x = "count", y = "word", alpha_by = "score",
             max_charwidth = 30, lineheight = 1.1)
SplitBarPlot(data, x = "count", y = "word", fill_by = "group")
SplitBarPlot(data, x = "count", y = "word", facet_by = "group",
             fill_name = "Direction")
SplitBarPlot(data, x = "count", y = "word", alpha_by = "score", split_by="group",
             palette = c(A = "Reds", B = "Blues", C = "Greens"))
</code></pre>

<hr>
<h2 id='BarPlotAtomic'>Atomic bar plot</h2><span id='topic+BarPlotAtomic'></span>

<h3>Description</h3>

<p>Create a bar plot with or without groups. This function does not handle splitting but only facetting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BarPlotAtomic(
  data,
  x,
  x_sep = "_",
  y = NULL,
  scale_y = FALSE,
  flip = FALSE,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  fill_by_x_if_no_group = TRUE,
  label_nudge = 0,
  label = NULL,
  label_fg = "black",
  label_size = 4,
  label_bg = "white",
  label_bg_r = 0.1,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  x_text_angle = 0,
  aspect.ratio = 1,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  line_name = NULL,
  add_trend = FALSE,
  trend_color = "black",
  trend_linewidth = 1,
  trend_ptsize = 2,
  position = "auto",
  position_dodge_preserve = "total",
  y_min = NULL,
  y_max = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  expand = waiver(),
  width = waiver(),
  facet_by = NULL,
  facet_scales = "fixed",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  facet_args = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BarPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_scale_y">scale_y</code></td>
<td>
<p>A logical value indicating whether to scale the total y values in each group to 100%.
Only works when group_by is specified.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to specify the name of the group_by in the legend.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_fill_by_x_if_no_group">fill_by_x_if_no_group</code></td>
<td>
<p>A logical value indicating whether to fill the bars by the x-axis values if there is no group_by.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_label_nudge">label_nudge</code></td>
<td>
<p>A numeric value to nudge the labels (the distance between the label and the top of the bar).</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_label">label</code></td>
<td>
<p>A column name for the values to be displayed on the top of the bars.
If TRUE, the y values will be displayed.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string indicating the color of the label.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value indicating the size of the label.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string indicating the background color of the label.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value indicating the radius of the background.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character string indicating the palette to use for the background.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character string indicating the color to use for the background.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value indicating the alpha of the background.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_add_line">add_line</code></td>
<td>
<p>A numeric value indicating the y value to add a horizontal line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_line_color">line_color</code></td>
<td>
<p>A character string indicating the color of the line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value indicating the size of the line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value indicating the type of the line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_line_name">line_name</code></td>
<td>
<p>A character string indicating the name of the line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_position">position</code></td>
<td>
<p>A character string indicating the position of the bars.
If &quot;auto&quot;, the position will be &quot;stack&quot; if group_by has more than 5 levels, otherwise &quot;dodge&quot;.
&quot;fill&quot; is also a valid option. Only works when group_by is not NULL.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_position_dodge_preserve">position_dodge_preserve</code></td>
<td>
<p>Should dodging preserve the &quot;total&quot; width of all elements at a position, or the width of a &quot;single&quot; element?</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value to specify the minimum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value to specify the maximum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_width">width</code></td>
<td>
<p>A numeric value specifying the width of the bars.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_facet_args">facet_args</code></td>
<td>
<p>A list of arguments to pass to <a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid</a> or <a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap</a>.</p>
</td></tr>
<tr><td><code id="BarPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='BarPlotGrouped'>Bar plot with groups</h2><span id='topic+BarPlotGrouped'></span>

<h3>Description</h3>

<p>Create a bar plot with groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BarPlotGrouped(
  data,
  x,
  x_sep = "_",
  y = NULL,
  scale_y = FALSE,
  flip = FALSE,
  group_by,
  group_by_sep = "_",
  group_name = NULL,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  alpha = 1,
  x_text_angle = 0,
  aspect.ratio = 1,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  line_name = NULL,
  add_trend = FALSE,
  trend_color = "black",
  trend_linewidth = 1,
  trend_ptsize = 2.5,
  position = "auto",
  position_dodge_preserve = "total",
  y_min = NULL,
  y_max = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  expand = waiver(),
  width = 0.8,
  facet_by = NULL,
  facet_scales = "fixed",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BarPlotGrouped_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_scale_y">scale_y</code></td>
<td>
<p>A logical value indicating whether to scale the total y values in each group to 100%.
Only works when group_by is specified.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_group_by">group_by</code></td>
<td>
<p>A character vector specifying the column as the group_by of the plot.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_group_name">group_name</code></td>
<td>
<p>A character string to specify the name of the group_by in the legend.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character string indicating the palette to use for the background.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character string indicating the color to use for the background.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value indicating the alpha of the background.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_add_line">add_line</code></td>
<td>
<p>A numeric value indicating the y value to add a horizontal line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_line_color">line_color</code></td>
<td>
<p>A character string indicating the color of the line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value indicating the size of the line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value indicating the type of the line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_line_name">line_name</code></td>
<td>
<p>A character string indicating the name of the line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_position">position</code></td>
<td>
<p>A character string indicating the position of the bars.
If &quot;auto&quot;, the position will be &quot;stack&quot; if group_by has more than 5 levels, otherwise &quot;dodge&quot;.
&quot;fill&quot; is also a valid option. Only works when group_by is not NULL.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_position_dodge_preserve">position_dodge_preserve</code></td>
<td>
<p>Should dodging preserve the &quot;total&quot; width of all elements at a position, or the width of a &quot;single&quot; element?</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value to specify the minimum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value to specify the maximum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_width">width</code></td>
<td>
<p>A numeric value specifying the width of the bars.</p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BarPlotGrouped_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='BarPlotSingle'>BarPlotSingle</h2><span id='topic+BarPlotSingle'></span>

<h3>Description</h3>

<p>Create a bar plot without groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BarPlotSingle(
  data,
  x,
  x_sep = "_",
  y = NULL,
  flip = FALSE,
  facet_by = NULL,
  facet_scales = "fixed",
  label = NULL,
  label_nudge = 0,
  label_fg = "black",
  label_size = 4,
  label_bg = "white",
  label_bg_r = 0.1,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  x_text_angle = 0,
  aspect.ratio = 1,
  y_min = NULL,
  y_max = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  line_name = NULL,
  add_trend = FALSE,
  trend_color = "black",
  trend_linewidth = 1,
  trend_ptsize = 2.5,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  expand = waiver(),
  fill_by_x = TRUE,
  width = 0.9,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BarPlotSingle_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_x">x</code></td>
<td>
<p>A character vector specifying the column as the x axis of the plot.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_y">y</code></td>
<td>
<p>A character vector specifying the column as the y axis of the plot.
Default is NULL, meaning the y axis is the count of the data.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_label">label</code></td>
<td>
<p>A column name for the values to be displayed on the top of the bars.
If TRUE, the y values will be displayed.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_label_nudge">label_nudge</code></td>
<td>
<p>A numeric value to nudge the labels (the distance between the label and the top of the bar).</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string indicating the color of the label.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value indicating the size of the label.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string indicating the background color of the label.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value indicating the radius of the background.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value to specify the minimum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value to specify the maximum value of the y axis.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_add_line">add_line</code></td>
<td>
<p>A numeric value indicating the y value to add a horizontal line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_line_color">line_color</code></td>
<td>
<p>A character string indicating the color of the line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value indicating the size of the line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value indicating the type of the line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_line_name">line_name</code></td>
<td>
<p>A character string indicating the name of the line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_fill_by_x">fill_by_x</code></td>
<td>
<p>A logical value indicating whether to fill the bars by the x-axis values.
If FALSE, the bars will be filled a single color (the first color in the palette).</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_width">width</code></td>
<td>
<p>A numeric value specifying the width of the bars.</p>
</td></tr>
<tr><td><code id="BarPlotSingle_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='bg_layer'>Get a ggplot layer for background</h2><span id='topic+bg_layer'></span>

<h3>Description</h3>

<p>Get a ggplot layer for background
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bg_layer(
  data,
  x,
  palette,
  palcolor,
  alpha,
  keep_empty,
  facet_by,
  direction = "vertical"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bg_layer_+3A_data">data</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name(s) of the data frame to facet the plot</p>
</td></tr>
<tr><td><code id="bg_layer_+3A_direction">direction</code></td>
<td>
<p>A character string specifying the direction for the background</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot layer for background
</p>

<hr>
<h2 id='blend_colors'>Blend colors</h2><span id='topic+blend_colors'></span>

<h3>Description</h3>

<p>This function blends a list of colors using the specified blend mode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blend_colors(colors, mode = c("blend", "average", "screen", "multiply"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="blend_colors_+3A_colors">colors</code></td>
<td>
<p>Color vectors.</p>
</td></tr>
<tr><td><code id="blend_colors_+3A_mode">mode</code></td>
<td>
<p>Blend mode. One of &quot;blend&quot;, &quot;average&quot;, &quot;screen&quot;, or &quot;multiply&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The blended color.
</p>

<hr>
<h2 id='blend_rgblist'>Blend a list of colors</h2><span id='topic+blend_rgblist'></span>

<h3>Description</h3>

<p>Blend a list of colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blend_rgblist(Clist, mode = "blend", RGB_BackGround = c(1, 1, 1))
</code></pre>

<hr>
<h2 id='blend_to_color'>Blend two colors</h2><span id='topic+blend_to_color'></span>

<h3>Description</h3>

<p>Blend two colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blend_to_color(C1, C2, mode = "blend")
</code></pre>

<hr>
<h2 id='BoxPlot'>Box / Violin Plot</h2><span id='topic+BoxPlot'></span><span id='topic+ViolinPlot'></span>

<h3>Description</h3>

<p>Box plot or violin plot with optional jitter points, trend line, statistical test, background, line, and highlight.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxPlot(
  data,
  x,
  x_sep = "_",
  y = NULL,
  in_form = c("long", "wide"),
  split_by = NULL,
  split_by_sep = "_",
  sort_x = c("none", "mean_asc", "mean_desc", "mean", "median_asc", "median_desc",
    "median"),
  flip = FALSE,
  keep_empty = FALSE,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  x_text_angle = ifelse(isTRUE(flip) &amp;&amp; isTRUE(stack), 90, 45),
  fill_mode = ifelse(!is.null(group_by), "dodge", "x"),
  fill_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  add_point = FALSE,
  pt_color = "grey30",
  pt_size = NULL,
  pt_alpha = 1,
  jitter_width = 0.5,
  jitter_height = 0.1,
  stack = FALSE,
  y_max = NULL,
  y_min = NULL,
  add_trend = FALSE,
  trend_color = NULL,
  trend_linewidth = 1,
  trend_ptsize = 2,
  add_stat = NULL,
  stat_name = NULL,
  stat_color = "black",
  stat_size = 1,
  stat_stroke = 1,
  stat_shape = 25,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  highlight = NULL,
  highlight_color = "red2",
  highlight_size = 1,
  highlight_alpha = 1,
  comparisons = NULL,
  ref_group = NULL,
  pairwise_method = "wilcox.test",
  multiplegroup_comparisons = FALSE,
  multiple_method = "kruskal.test",
  sig_label = c("p.signif", "p.format"),
  sig_labelsize = 3.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

ViolinPlot(
  data,
  x,
  x_sep = "_",
  y = NULL,
  in_form = c("long", "wide"),
  split_by = NULL,
  split_by_sep = "_",
  sort_x = c("none", "mean_asc", "mean_desc", "mean", "median_asc", "median_desc",
    "median"),
  flip = FALSE,
  keep_empty = FALSE,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  x_text_angle = ifelse(isTRUE(flip) &amp;&amp; isTRUE(stack), 90, 45),
  fill_mode = ifelse(!is.null(group_by), "dodge", "x"),
  fill_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  add_point = FALSE,
  pt_color = "grey30",
  pt_size = NULL,
  pt_alpha = 1,
  jitter_width = 0.5,
  jitter_height = 0.1,
  stack = FALSE,
  y_max = NULL,
  y_min = NULL,
  add_box = FALSE,
  box_color = "black",
  box_width = 0.1,
  box_ptsize = 2.5,
  add_trend = FALSE,
  trend_color = NULL,
  trend_linewidth = 1,
  trend_ptsize = 2,
  add_stat = NULL,
  stat_name = NULL,
  stat_color = "black",
  stat_size = 1,
  stat_stroke = 1,
  stat_shape = 25,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  highlight = NULL,
  highlight_color = "red2",
  highlight_size = 1,
  highlight_alpha = 1,
  comparisons = NULL,
  ref_group = NULL,
  pairwise_method = "wilcox.test",
  multiplegroup_comparisons = FALSE,
  multiple_method = "kruskal.test",
  sig_label = c("p.signif", "p.format"),
  sig_labelsize = 3.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BoxPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.
When <code>in_form</code> is &quot;wide&quot;, <code>x</code> columns will not be concatenated.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_in_form">in_form</code></td>
<td>
<p>A character string to specify the input data type. Either &quot;long&quot; or &quot;wide&quot;.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_sort_x">sort_x</code></td>
<td>
<p>A character string to specify the sorting of x-axis. Either &quot;none&quot;, &quot;mean_asc&quot;, &quot;mean_desc&quot;, &quot;mean&quot;, &quot;median_asc&quot;, &quot;median_desc&quot;, &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of dodge.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_fill_mode">fill_mode</code></td>
<td>
<p>A character string to specify the fill mode. Either &quot;dodge&quot;, &quot;x&quot;, &quot;mean&quot;, &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_fill_reverse">fill_reverse</code></td>
<td>
<p>A logical value to reverse the fill colors for gradient fill (mean/median).</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_add_point">add_point</code></td>
<td>
<p>A logical value to add (jitter) points to the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_pt_color">pt_color</code></td>
<td>
<p>A character string to specify the color of the points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to specify the size of the points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_jitter_width">jitter_width</code></td>
<td>
<p>A numeric value to specify the width of the jitter.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_jitter_height">jitter_height</code></td>
<td>
<p>A numeric value to specify the height of the jitter.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_stack">stack</code></td>
<td>
<p>A logical value whether to stack the facetted plot by 'facet_by'.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value or a character string to specify the maximum value of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value or a character string to specify the minimum value of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_add_stat">add_stat</code></td>
<td>
<p>A character string to add statistical test to the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_stat_name">stat_name</code></td>
<td>
<p>A character string to specify the name of the stat legend.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_stat_color">stat_color</code></td>
<td>
<p>A character string to specify the color of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_stat_size">stat_size</code></td>
<td>
<p>A numeric value to specify the size of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_stat_stroke">stat_stroke</code></td>
<td>
<p>A numeric value to specify the stroke of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_stat_shape">stat_shape</code></td>
<td>
<p>A numeric value to specify the shape of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value to add background to the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character string to specify the palette of the background.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character vector to specify the colors of the background.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the background.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_add_line">add_line</code></td>
<td>
<p>A character string to add a line to the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_line_color">line_color</code></td>
<td>
<p>A character string to specify the color of the line.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value to specify the size of the line.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value to specify the type of the line.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_highlight">highlight</code></td>
<td>
<p>A vector of character strings to highlight the points.
It should be a subset of the row names of the data.
If TRUE, it will highlight all points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string to specify the color of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value to specify the size of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_comparisons">comparisons</code></td>
<td>
<p>A logical value or a list of vectors to perform pairwise comparisons.
If <code>TRUE</code>, it will perform pairwise comparisons for all pairs.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_ref_group">ref_group</code></td>
<td>
<p>A character string to specify the reference group for comparisons.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_pairwise_method">pairwise_method</code></td>
<td>
<p>A character string to specify the pairwise comparison method.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_multiplegroup_comparisons">multiplegroup_comparisons</code></td>
<td>
<p>A logical value to perform multiple group comparisons.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_multiple_method">multiple_method</code></td>
<td>
<p>A character string to specify the multiple group comparison method.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_sig_label">sig_label</code></td>
<td>
<p>A character string to specify the label of the significance test.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_sig_labelsize">sig_labelsize</code></td>
<td>
<p>A numeric value to specify the size of the significance test label.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_add_box">add_box</code></td>
<td>
<p>A logical value to add box plot to the plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_box_color">box_color</code></td>
<td>
<p>A character string to specify the color of the box plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_box_width">box_width</code></td>
<td>
<p>A numeric value to specify the width of the box plot.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_box_ptsize">box_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the box plot points in the middle.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Box / Violin plot(s).
When <code>split_by</code> is not provided, it returns a ggplot object.
When <code>split_by</code> is provided, it returns a object of plots wrapped by <code>patchwork::wrap_plots</code> if <code>combine = TRUE</code>;
otherwise, it returns a list of ggplot objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
data &lt;- data.frame(
    x = rep(LETTERS[1:8], 40),
    y = rnorm(320),
    group1 = sample(c("g1", "g2"), 320, replace = TRUE),
    group2 = sample(c("h1", "h2", "h3", "h4"), 320, replace = TRUE)
)

BoxPlot(data, x = "x", y = "y")
BoxPlot(data,
    x = "x", y = "y",
    stack = TRUE, flip = TRUE, facet_by = "group1",
    add_bg = TRUE, bg_palette = "Paired"
)
BoxPlot(data,
    x = "x", y = "y",
    stack = TRUE, flip = TRUE, split_by = "group1",
    add_bg = TRUE, bg_palette = "Paired",
    palcolor = list(g1 = c("red", "blue"), g2 = c("blue", "red"))
)

# wide form data
data_wide &lt;- data.frame(
    A = rnorm(100),
    B = rnorm(100),
    C = rnorm(100)
)
BoxPlot(data_wide, x = c("A", "B", "C"), in_form = "wide")


ViolinPlot(data, x = "x", y = "y")
ViolinPlot(data, x = "x", y = "y", add_box = TRUE)
ViolinPlot(data, x = "x", y = "y", add_point = TRUE)
ViolinPlot(data, x = "x", y = "y", add_trend = TRUE)
ViolinPlot(data, x = "x", y = "y", add_stat = mean)
ViolinPlot(data, x = "x", y = "y", add_bg = TRUE)
ViolinPlot(data, x = "x", y = "y", add_line = 0)
ViolinPlot(data, x = "x", y = "y", group_by = "group1")
ViolinPlot(data,
    x = "x", y = "y", group_by = "group1",
    facet_by = "group2", add_box = TRUE
)
ViolinPlot(data, x = "x", y = "y", add_point = TRUE, highlight = 'group1 == "g1"',
    alpha = 0.8, highlight_size = 1.5, pt_size = 1, add_box = TRUE)
ViolinPlot(data,
    x = "x", y = "y", group_by = "group1",
    comparisons = TRUE
)
ViolinPlot(data,
    x = "x", y = "y", sig_label = "p.format",
    facet_by = "group2", comparisons = list(c("A", "B"))
)
ViolinPlot(data,
    x = "x", y = "y", fill_mode = "mean",
    facet_by = "group2", palette = "Blues"
)
ViolinPlot(data,
    x = "x", y = "y", fill_mode = "mean",
    split_by = "group1", palette = c(g1 = "Blues", g2 = "Reds")
)
ViolinPlot(data,
    x = "x", y = "y", stack = TRUE,
    facet_by = "group2", add_box = TRUE, add_bg = TRUE,
    bg_palette = "Paired"
)

</code></pre>

<hr>
<h2 id='BoxViolinPlot'>Box/Violin plot</h2><span id='topic+BoxViolinPlot'></span>

<h3>Description</h3>

<p>Box/Violin plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxViolinPlot(
  data,
  x,
  x_sep = "_",
  y = NULL,
  base = c("box", "violin"),
  in_form = c("long", "wide"),
  split_by = NULL,
  split_by_sep = "_",
  sort_x = c("none", "mean_asc", "mean_desc", "mean", "median_asc", "median_desc",
    "median"),
  flip = FALSE,
  keep_empty = FALSE,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  x_text_angle = ifelse(isTRUE(flip) &amp;&amp; isTRUE(stack), 90, 45),
  fill_mode = ifelse(!is.null(group_by), "dodge", "x"),
  fill_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  add_point = FALSE,
  pt_color = "grey30",
  pt_size = NULL,
  pt_alpha = 1,
  jitter_width = 0.5,
  jitter_height = 0.1,
  stack = FALSE,
  y_max = NULL,
  y_min = NULL,
  add_box = FALSE,
  box_color = "black",
  box_width = 0.1,
  box_ptsize = 2.5,
  add_trend = FALSE,
  trend_color = NULL,
  trend_linewidth = 1,
  trend_ptsize = 2,
  add_stat = NULL,
  stat_name = NULL,
  stat_color = "black",
  stat_size = 1,
  stat_stroke = 1,
  stat_shape = 25,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  highlight = NULL,
  highlight_color = "red2",
  highlight_size = 1,
  highlight_alpha = 1,
  comparisons = NULL,
  ref_group = NULL,
  pairwise_method = "wilcox.test",
  multiplegroup_comparisons = FALSE,
  multiple_method = "kruskal.test",
  sig_label = c("p.signif", "p.format"),
  sig_labelsize = 3.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BoxViolinPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.
When <code>in_form</code> is &quot;wide&quot;, <code>x</code> columns will not be concatenated.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_base">base</code></td>
<td>
<p>A character string to specify the base plot type. Either &quot;box&quot; or &quot;violin&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_in_form">in_form</code></td>
<td>
<p>A character string to specify the input data type. Either &quot;long&quot; or &quot;wide&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_sort_x">sort_x</code></td>
<td>
<p>A character string to specify the sorting of x-axis. Either &quot;none&quot;, &quot;mean_asc&quot;, &quot;mean_desc&quot;, &quot;mean&quot;, &quot;median_asc&quot;, &quot;median_desc&quot;, &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of dodge.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_fill_mode">fill_mode</code></td>
<td>
<p>A character string to specify the fill mode. Either &quot;dodge&quot;, &quot;x&quot;, &quot;mean&quot;, &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_fill_reverse">fill_reverse</code></td>
<td>
<p>A logical value to reverse the fill colors for gradient fill (mean/median).</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_add_point">add_point</code></td>
<td>
<p>A logical value to add (jitter) points to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_pt_color">pt_color</code></td>
<td>
<p>A character string to specify the color of the points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to specify the size of the points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_jitter_width">jitter_width</code></td>
<td>
<p>A numeric value to specify the width of the jitter.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_jitter_height">jitter_height</code></td>
<td>
<p>A numeric value to specify the height of the jitter.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_stack">stack</code></td>
<td>
<p>A logical value whether to stack the facetted plot by 'facet_by'.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value or a character string to specify the maximum value of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value or a character string to specify the minimum value of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_add_box">add_box</code></td>
<td>
<p>A logical value to add box plot to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_box_color">box_color</code></td>
<td>
<p>A character string to specify the color of the box plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_box_width">box_width</code></td>
<td>
<p>A numeric value to specify the width of the box plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_box_ptsize">box_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the box plot points in the middle.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_add_stat">add_stat</code></td>
<td>
<p>A character string to add statistical test to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_stat_name">stat_name</code></td>
<td>
<p>A character string to specify the name of the stat legend.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_stat_color">stat_color</code></td>
<td>
<p>A character string to specify the color of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_stat_size">stat_size</code></td>
<td>
<p>A numeric value to specify the size of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_stat_stroke">stat_stroke</code></td>
<td>
<p>A numeric value to specify the stroke of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_stat_shape">stat_shape</code></td>
<td>
<p>A numeric value to specify the shape of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value to add background to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character string to specify the palette of the background.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character vector to specify the colors of the background.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the background.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_add_line">add_line</code></td>
<td>
<p>A character string to add a line to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_line_color">line_color</code></td>
<td>
<p>A character string to specify the color of the line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value to specify the size of the line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value to specify the type of the line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_highlight">highlight</code></td>
<td>
<p>A vector of character strings to highlight the points.
It should be a subset of the row names of the data.
If TRUE, it will highlight all points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string to specify the color of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value to specify the size of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_comparisons">comparisons</code></td>
<td>
<p>A logical value or a list of vectors to perform pairwise comparisons.
If <code>TRUE</code>, it will perform pairwise comparisons for all pairs.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_ref_group">ref_group</code></td>
<td>
<p>A character string to specify the reference group for comparisons.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_pairwise_method">pairwise_method</code></td>
<td>
<p>A character string to specify the pairwise comparison method.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_multiplegroup_comparisons">multiplegroup_comparisons</code></td>
<td>
<p>A logical value to perform multiple group comparisons.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_multiple_method">multiple_method</code></td>
<td>
<p>A character string to specify the multiple group comparison method.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_sig_label">sig_label</code></td>
<td>
<p>A character string to specify the label of the significance test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_sig_labelsize">sig_labelsize</code></td>
<td>
<p>A numeric value to specify the size of the significance test label.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="BoxViolinPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A combined ggplot object or wrap_plots object or a list of ggplot objects
</p>

<hr>
<h2 id='BoxViolinPlotAtomic'>Atomic Box/Violin plot</h2><span id='topic+BoxViolinPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic Box/Violin plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxViolinPlotAtomic(
  data,
  x,
  x_sep = "_",
  y = NULL,
  base = c("box", "violin"),
  in_form = c("long", "wide"),
  sort_x = c("none", "mean_asc", "mean_desc", "mean", "median_asc", "median_desc",
    "median"),
  flip = FALSE,
  keep_empty = FALSE,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  x_text_angle = ifelse(isTRUE(flip) &amp;&amp; isTRUE(stack), 90, 45),
  fill_mode = ifelse(!is.null(group_by), "dodge", "x"),
  fill_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  add_point = FALSE,
  pt_color = "grey30",
  pt_size = NULL,
  pt_alpha = 1,
  y_nbreaks = 4,
  jitter_width = 0.5,
  jitter_height = 0.1,
  stack = FALSE,
  y_max = NULL,
  y_min = NULL,
  y_trans = "identity",
  add_box = FALSE,
  box_color = "black",
  box_width = 0.1,
  box_ptsize = 2.5,
  add_trend = FALSE,
  trend_color = NULL,
  trend_linewidth = 1,
  trend_ptsize = 2,
  add_stat = NULL,
  stat_name = NULL,
  stat_color = "black",
  stat_size = 1,
  stat_stroke = 1,
  stat_shape = 25,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_line = NULL,
  line_color = "red2",
  line_width = 0.6,
  line_type = 2,
  highlight = NULL,
  highlight_color = "red2",
  highlight_size = 1,
  highlight_alpha = 1,
  comparisons = NULL,
  ref_group = NULL,
  pairwise_method = "wilcox.test",
  multiplegroup_comparisons = FALSE,
  multiple_method = "kruskal.test",
  sig_label = c("p.signif", "p.format"),
  sig_labelsize = 3.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BoxViolinPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string of the column name to plot on the x-axis.
A character/factor column is expected. If multiple columns are provided, the columns will be concatenated with <code>x_sep</code>.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.
When <code>in_form</code> is &quot;wide&quot;, <code>x</code> columns will not be concatenated.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis. A numeric column is expected.
When <code>in_form</code> is &quot;wide&quot;, <code>y</code> is not required. The values under <code>x</code> columns will be used as y-values.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_base">base</code></td>
<td>
<p>A character string to specify the base plot type. Either &quot;box&quot; or &quot;violin&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_in_form">in_form</code></td>
<td>
<p>A character string to specify the input data type. Either &quot;long&quot; or &quot;wide&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_sort_x">sort_x</code></td>
<td>
<p>A character string to specify the sorting of x-axis. Either &quot;none&quot;, &quot;mean_asc&quot;, &quot;mean_desc&quot;, &quot;mean&quot;, &quot;median_asc&quot;, &quot;median_desc&quot;, &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value to keep the empty levels in the x-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string of the column name to dodge the boxes/violins</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of dodge.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_fill_mode">fill_mode</code></td>
<td>
<p>A character string to specify the fill mode. Either &quot;dodge&quot;, &quot;x&quot;, &quot;mean&quot;, &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_fill_reverse">fill_reverse</code></td>
<td>
<p>A logical value to reverse the fill colors for gradient fill (mean/median).</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_add_point">add_point</code></td>
<td>
<p>A logical value to add (jitter) points to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_pt_color">pt_color</code></td>
<td>
<p>A character string to specify the color of the points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to specify the size of the points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_y_nbreaks">y_nbreaks</code></td>
<td>
<p>A numeric value to specify the number of breaks in the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_jitter_width">jitter_width</code></td>
<td>
<p>A numeric value to specify the width of the jitter.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_jitter_height">jitter_height</code></td>
<td>
<p>A numeric value to specify the height of the jitter.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_stack">stack</code></td>
<td>
<p>A logical value whether to stack the facetted plot by 'facet_by'.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value or a character string to specify the maximum value of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value or a character string to specify the minimum value of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_y_trans">y_trans</code></td>
<td>
<p>A character string to specify the transformation of the y-axis.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_add_box">add_box</code></td>
<td>
<p>A logical value to add box plot to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_box_color">box_color</code></td>
<td>
<p>A character string to specify the color of the box plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_box_width">box_width</code></td>
<td>
<p>A numeric value to specify the width of the box plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_box_ptsize">box_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the box plot points in the middle.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value to add trend line to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_trend_color">trend_color</code></td>
<td>
<p>A character string to specify the color of the trend line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value to specify the width of the trend line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_trend_ptsize">trend_ptsize</code></td>
<td>
<p>A numeric value to specify the size of the trend line points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_add_stat">add_stat</code></td>
<td>
<p>A character string to add statistical test to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_stat_name">stat_name</code></td>
<td>
<p>A character string to specify the name of the stat legend.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_stat_color">stat_color</code></td>
<td>
<p>A character string to specify the color of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_stat_size">stat_size</code></td>
<td>
<p>A numeric value to specify the size of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_stat_stroke">stat_stroke</code></td>
<td>
<p>A numeric value to specify the stroke of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_stat_shape">stat_shape</code></td>
<td>
<p>A numeric value to specify the shape of the statistical test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value to add background to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character string to specify the palette of the background.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character vector to specify the colors of the background.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the background.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_add_line">add_line</code></td>
<td>
<p>A character string to add a line to the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_line_color">line_color</code></td>
<td>
<p>A character string to specify the color of the line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value to specify the size of the line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_line_type">line_type</code></td>
<td>
<p>A numeric value to specify the type of the line.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_highlight">highlight</code></td>
<td>
<p>A vector of character strings to highlight the points.
It should be a subset of the row names of the data.
If TRUE, it will highlight all points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string to specify the color of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value to specify the size of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of the highlighted points.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_comparisons">comparisons</code></td>
<td>
<p>A logical value or a list of vectors to perform pairwise comparisons.
If <code>TRUE</code>, it will perform pairwise comparisons for all pairs.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_ref_group">ref_group</code></td>
<td>
<p>A character string to specify the reference group for comparisons.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_pairwise_method">pairwise_method</code></td>
<td>
<p>A character string to specify the pairwise comparison method.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_multiplegroup_comparisons">multiplegroup_comparisons</code></td>
<td>
<p>A logical value to perform multiple group comparisons.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_multiple_method">multiple_method</code></td>
<td>
<p>A character string to specify the multiple group comparison method.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_sig_label">sig_label</code></td>
<td>
<p>A character string to specify the label of the significance test.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_sig_labelsize">sig_labelsize</code></td>
<td>
<p>A numeric value to specify the size of the significance test label.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="BoxViolinPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='calc_just'>Calculate hjust and vjust based on angle</h2><span id='topic+calc_just'></span>

<h3>Description</h3>

<p>Calculate hjust and vjust based on angle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_just(angle)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_just_+3A_angle">angle</code></td>
<td>
<p>A numeric value of the angle</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with h and v values
</p>

<hr>
<h2 id='check_columns'>Check the columns if columns found in the data</h2><span id='topic+check_columns'></span>

<h3>Description</h3>

<p>Check the columns if columns found in the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_columns(
  df,
  columns,
  force_factor = FALSE,
  allow_multi = FALSE,
  concat_multi = FALSE,
  concat_sep = "_"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_columns_+3A_df">df</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id="check_columns_+3A_columns">columns</code></td>
<td>
<p>A character vector of column names</p>
</td></tr>
<tr><td><code id="check_columns_+3A_force_factor">force_factor</code></td>
<td>
<p>Whether to force the columns to be factors</p>
</td></tr>
<tr><td><code id="check_columns_+3A_allow_multi">allow_multi</code></td>
<td>
<p>Whether to allow multiple columns</p>
</td></tr>
<tr><td><code id="check_columns_+3A_concat_multi">concat_multi</code></td>
<td>
<p>Whether to concatenate multiple columns</p>
</td></tr>
<tr><td><code id="check_columns_+3A_concat_sep">concat_sep</code></td>
<td>
<p>The separator to use for concatenation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string of the valid column
</p>

<hr>
<h2 id='check_palcolor'>check_palcolor
Check if the palcolor can be properly used</h2><span id='topic+check_palcolor'></span>

<h3>Description</h3>

<p>check_palcolor
Check if the palcolor can be properly used
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_palcolor(palcolor, datas_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_palcolor_+3A_palcolor">palcolor</code></td>
<td>
<p>palcolor</p>
</td></tr>
<tr><td><code id="check_palcolor_+3A_datas_name">datas_name</code></td>
<td>
<p>names of the split data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named list containing color names
</p>

<hr>
<h2 id='check_palette'>check_palette
Check if the palette can be properly used</h2><span id='topic+check_palette'></span>

<h3>Description</h3>

<p>check_palette
Check if the palette can be properly used
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_palette(palette, datas_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_palette_+3A_palette">palette</code></td>
<td>
<p>palette</p>
</td></tr>
<tr><td><code id="check_palette_+3A_datas_name">datas_name</code></td>
<td>
<p>names of the split data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named list containing palette names
</p>

<hr>
<h2 id='ChordPlot'>Chord / Circos plot</h2><span id='topic+ChordPlot'></span><span id='topic+CircosPlot'></span>

<h3>Description</h3>

<p><code>ChordPlot</code> is used to create a chord plot to visualize the relationships between two categorical variables.
<code>CircosPlot</code> is an alias of <code>ChordPlot</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChordPlot(
  data,
  y = NULL,
  from = NULL,
  from_sep = "_",
  to = NULL,
  to_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  flip = FALSE,
  links_color = c("from", "to"),
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 0.5,
  labels_rot = FALSE,
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

CircosPlot(
  data,
  y = NULL,
  from = NULL,
  from_sep = "_",
  to = NULL,
  to_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  flip = FALSE,
  links_color = c("from", "to"),
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 0.5,
  labels_rot = FALSE,
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChordPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_from">from</code></td>
<td>
<p>A character string of the column name to plot for the source.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_from_sep">from_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>from</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_to">to</code></td>
<td>
<p>A character string of the column name to plot for the target.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_to_sep">to_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>to</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the source and target.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_links_color">links_color</code></td>
<td>
<p>A character string to specify the color of the links.
Either &quot;from&quot; or &quot;to&quot;.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_labels_rot">labels_rot</code></td>
<td>
<p>A logical value to rotate the labels by 90 degrees.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="ChordPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A combined plot or a list of plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
data &lt;- data.frame(
    nodes1 = sample(c("Soure1", "Source2", "Source3"), 10, replace = TRUE),
    nodes2 = sample(letters[1:3], 10, replace = TRUE),
    y = sample(1:5, 10, replace = TRUE)
)

ChordPlot(data, from = "nodes1", to = "nodes2")
ChordPlot(data, from = "nodes1", to = "nodes2",
          links_color = "to", labels_rot = TRUE)
ChordPlot(data, from = "nodes1", to = "nodes2", y = "y")
ChordPlot(data, from = "nodes1", to = "nodes2", split_by = "y")
ChordPlot(data, from = "nodes1", to = "nodes2", split_by = "y",
          palette = c("1" = "Reds", "2" = "Blues", "3" = "Greens", "4" = "Purp"))
ChordPlot(data, from = "nodes1", to = "nodes2", flip = TRUE)

</code></pre>

<hr>
<h2 id='ChordPlotAtomic'>Atomic chord plot</h2><span id='topic+ChordPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic chord plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChordPlotAtomic(
  data,
  y = NULL,
  from = NULL,
  from_sep = "_",
  to = NULL,
  to_sep = "_",
  flip = FALSE,
  links_color = c("from", "to"),
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 0.5,
  labels_rot = FALSE,
  title = NULL,
  subtitle = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChordPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_from">from</code></td>
<td>
<p>A character string of the column name to plot for the source.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_from_sep">from_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>from</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_to">to</code></td>
<td>
<p>A character string of the column name to plot for the target.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_to_sep">to_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>to</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the source and target.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_links_color">links_color</code></td>
<td>
<p>A character string to specify the color of the links.
Either &quot;from&quot; or &quot;to&quot;.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_labels_rot">labels_rot</code></td>
<td>
<p>A logical value to rotate the labels by 90 degrees.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ChordPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A wrapped element of chord plot
</p>

<hr>
<h2 id='ClustreePlot'>Clustree Plot</h2><span id='topic+ClustreePlot'></span>

<h3>Description</h3>

<p>A plot visualizing Clusterings at Different Resolutions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClustreePlot(
  data,
  prefix,
  flip = FALSE,
  split_by = NULL,
  split_by_sep = "_",
  palette = "Paired",
  palcolor = NULL,
  edge_palette = "Spectral",
  edge_palcolor = NULL,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  expand = c(0.1, 0.1),
  theme = "theme_this",
  theme_args = list(),
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ClustreePlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the prefix of the columns to plot.
The columns with the prefix will be used to plot the tree.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the tree.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_edge_palette">edge_palette</code></td>
<td>
<p>A character string of the palette name to color the edges.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_edge_palcolor">edge_palcolor</code></td>
<td>
<p>A character vector of colors to color the edges.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="ClustreePlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(8525)
N = 100
data &lt;- data.frame(
    p.0.4 = sample(LETTERS[1:5], N, replace = TRUE),
    p.0.5 = sample(LETTERS[1:6], N, replace = TRUE),
    p.0.6 = sample(LETTERS[1:7], N, replace = TRUE),
    p.0.7 = sample(LETTERS[1:8], N, replace = TRUE),
    p.0.8 = sample(LETTERS[1:9], N, replace = TRUE),
    p.0.9 = sample(LETTERS[1:10], N, replace = TRUE),
    p.1 = sample(LETTERS[1:30], N, replace = TRUE),
    split = sample(1:2, N, replace = TRUE)
)

ClustreePlot(data, prefix = "p")
ClustreePlot(data, prefix = "p", flip = TRUE)
ClustreePlot(data, prefix = "p", split_by = "split")
ClustreePlot(data, prefix = "p", split_by = "split",
             palette = c("1" = "Set1", "2" = "Paired"))
</code></pre>

<hr>
<h2 id='ClustreePlotAtomic'>Atomic function for clustree plot</h2><span id='topic+ClustreePlotAtomic'></span>

<h3>Description</h3>

<p>Atomic function for clustree plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClustreePlotAtomic(
  data,
  prefix,
  flip = FALSE,
  alpha = 0.85,
  palette = "Paired",
  palcolor = NULL,
  edge_palette = "Spectral",
  edge_palcolor = NULL,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  expand = c(0.1, 0.1),
  theme = "theme_this",
  theme_args = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ClustreePlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the prefix of the columns to plot.
The columns with the prefix will be used to plot the tree.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the tree.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value of the transparency of the nodes.
Only used when <code>node_alpha</code> is not provided in <code>...</code>.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_edge_palette">edge_palette</code></td>
<td>
<p>A character string of the palette name to color the edges.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_edge_palcolor">edge_palcolor</code></td>
<td>
<p>A character vector of colors to color the edges.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ClustreePlotAtomic_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>clustree::clustree</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='combine_plots'>Combine plots into one</h2><span id='topic+combine_plots'></span>

<h3>Description</h3>

<p>Combine plots into one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_plots(plots, combine, nrow, ncol, byrow, recalc_size = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine_plots_+3A_plots">plots</code></td>
<td>
<p>A list of plots</p>
</td></tr>
<tr><td><code id="combine_plots_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one</p>
</td></tr>
<tr><td><code id="combine_plots_+3A_nrow">nrow</code></td>
<td>
<p>The number of rows in the combined plot</p>
</td></tr>
<tr><td><code id="combine_plots_+3A_ncol">ncol</code></td>
<td>
<p>The number of columns in the combined plot</p>
</td></tr>
<tr><td><code id="combine_plots_+3A_byrow">byrow</code></td>
<td>
<p>Whether to fill the plots by row</p>
</td></tr>
<tr><td><code id="combine_plots_+3A_recalc_size">recalc_size</code></td>
<td>
<p>Whether to re-calculate the size of the combined plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The faceted plot. If guess_size is TRUE, attr(p, &quot;height&quot;) and attr(p, &quot;width&quot;) will be set
</p>

<hr>
<h2 id='common_args'>Common arguments for plots</h2><span id='topic+common_args'></span>

<h3>Description</h3>

<p>Common arguments for plots
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="common_args_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="common_args_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="common_args_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="common_args_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="common_args_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="common_args_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="common_args_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="common_args_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="common_args_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="common_args_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="common_args_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="common_args_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="common_args_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="common_args_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="common_args_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="common_args_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="common_args_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="common_args_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="common_args_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="common_args_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="common_args_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="common_args_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="common_args_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="common_args_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="common_args_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="common_args_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="common_args_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="common_args_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="common_args_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="common_args_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="common_args_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="common_args_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="common_args_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='CorPairsPlot'>CorPairsPlot</h2><span id='topic+CorPairsPlot'></span>

<h3>Description</h3>

<p>Generate a grid of scatter correlation plots for all pairs of variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CorPairsPlot(
  data,
  columns = NULL,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  split_by = NULL,
  split_by_sep = "_",
  diag_type = NULL,
  diag_args = list(),
  layout = c(".\\", "\\.", "/.", "./"),
  cor_method = c("pearson", "spearman", "kendall"),
  cor_palette = "RdBu",
  cor_palcolor = NULL,
  cor_size = 3,
  cor_format = "corr: {round(corr, 2)}",
  cor_fg = "black",
  cor_bg = "white",
  cor_bg_r = 0.1,
  theme = "theme_this",
  theme_args = list(),
  palette = ifelse(is.null(group_by), "Spectral", "Paired"),
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  facet_by = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CorPairsPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_columns">columns</code></td>
<td>
<p>The column names of the data to be plotted.
If NULL, all columns, except <code>group_by</code>, will be used.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_group_name">group_name</code></td>
<td>
<p>The name of the group in the legend.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_diag_type">diag_type</code></td>
<td>
<p>The type of the diagonal plots.
Available types: &quot;density&quot;, &quot;violin&quot;, &quot;histogram&quot;, &quot;box&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_diag_args">diag_args</code></td>
<td>
<p>A list of additional arguments to be passed to the diagonal plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_layout">layout</code></td>
<td>
<p>The layout of the plots.
Available layouts: &quot;.\&quot;, &quot;\.&quot;, &quot;/.&quot;, &quot;./&quot;.
</p>

<ul>
<li><p> '\' or '/' means the diagonal plots are on the top-left to bottom-right diagonal.
</p>
</li>
<li><p> '.' means where the scatter plots are.
</p>
</li></ul>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_method">cor_method</code></td>
<td>
<p>The method to calculate the correlation.
Available methods: &quot;pearson&quot;, &quot;spearman&quot;, &quot;kendall&quot;.
The correlation will be shown in the other triangle of the scatter plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_palette">cor_palette</code></td>
<td>
<p>The color palette for the correlation tile plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_palcolor">cor_palcolor</code></td>
<td>
<p>Custom colors used to create a color palette for the correlation tile plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_size">cor_size</code></td>
<td>
<p>The size of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_format">cor_format</code></td>
<td>
<p>The format of the correlation text. Default is &quot;corr: %.2f&quot;.
It will be formatted using <code>sprintf(cor_format, corr)</code>.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_fg">cor_fg</code></td>
<td>
<p>The color of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_bg">cor_bg</code></td>
<td>
<p>The background color of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_cor_bg_r">cor_bg_r</code></td>
<td>
<p>The radius of the background of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="CorPairsPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>patch_work::wrap_plots</code> object or a list of them if <code>combine</code> is <code>FALSE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
data &lt;- data.frame(x = rnorm(100))
data$y &lt;- rnorm(100, 10, sd = 0.5)
data$z &lt;- -data$x + data$y + rnorm(100, 20, 1)
data$g &lt;- sample(1:4, 100, replace = TRUE)

CorPairsPlot(data, diag_type = "histogram", diag_args = list(bins = 30, palette = "Paired"),
 layout = "/.")

CorPairsPlot(data, group_by = "g", diag_type = "none", layout = "./",
 theme_args = list(axis.title = element_textbox(
     color = "black", box.color = "grey20", size = 16, halign = 0.5, fill = "grey90",
     linetype = 1, width = grid::unit(1, "npc"), padding = ggplot2::margin(5, 5, 5, 5))))

CorPairsPlot(data, group_by = "g", diag_type = "violin", layout = "\\.",
  cor_format = "{x}\n{y}\ncorr: {round(corr, 2)}")

CorPairsPlot(data, split_by = "g", diag_type = "none", layout = ".\\",
 legend.position = "bottom", legend.direction = "horizontal", group_name = "group")

CorPairsPlot(data, split_by = "g",
 palcolor = list("1" = "red", "2" = "blue", "3" = "green", "4" = "yellow"))

</code></pre>

<hr>
<h2 id='CorPairsPlotAtomic'>Atomic Correlation Pairs Plot</h2><span id='topic+CorPairsPlotAtomic'></span>

<h3>Description</h3>

<p>Generate a grid of scatter correlation plots for all pairs of variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CorPairsPlotAtomic(
  data,
  columns = NULL,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  diag_type = NULL,
  diag_args = list(),
  layout = c(".\\", "\\.", "/.", "./"),
  cor_method = c("pearson", "spearman", "kendall"),
  cor_palette = "RdBu",
  cor_palcolor = NULL,
  cor_size = 3,
  cor_format = "corr: {round(corr, 2)}",
  cor_fg = "black",
  cor_bg = "white",
  cor_bg_r = 0.1,
  theme = "theme_this",
  theme_args = list(),
  palette = ifelse(is.null(group_by), "Spectral", "Paired"),
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  facet_by = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CorPairsPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_columns">columns</code></td>
<td>
<p>The column names of the data to be plotted.
If NULL, all columns, except <code>group_by</code>, will be used.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>The column name of the data to be used for grouping.
Different groups will be plotted in different colors.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator used to concatenate multiple columns in <code>group_by</code>.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>The name of the group in the legend.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_diag_type">diag_type</code></td>
<td>
<p>The type of the diagonal plots.
Available types: &quot;density&quot;, &quot;violin&quot;, &quot;histogram&quot;, &quot;box&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_diag_args">diag_args</code></td>
<td>
<p>A list of additional arguments to be passed to the diagonal plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_layout">layout</code></td>
<td>
<p>The layout of the plots.
Available layouts: &quot;.\&quot;, &quot;\.&quot;, &quot;/.&quot;, &quot;./&quot;.
</p>

<ul>
<li><p> '\' or '/' means the diagonal plots are on the top-left to bottom-right diagonal.
</p>
</li>
<li><p> '.' means where the scatter plots are.
</p>
</li></ul>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_method">cor_method</code></td>
<td>
<p>The method to calculate the correlation.
Available methods: &quot;pearson&quot;, &quot;spearman&quot;, &quot;kendall&quot;.
The correlation will be shown in the other triangle of the scatter plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_palette">cor_palette</code></td>
<td>
<p>The color palette for the correlation tile plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_palcolor">cor_palcolor</code></td>
<td>
<p>Custom colors used to create a color palette for the correlation tile plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_size">cor_size</code></td>
<td>
<p>The size of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_format">cor_format</code></td>
<td>
<p>The format of the correlation text. Default is &quot;corr: %.2f&quot;.
It will be formatted using <code>sprintf(cor_format, corr)</code>.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_fg">cor_fg</code></td>
<td>
<p>The color of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_bg">cor_bg</code></td>
<td>
<p>The background color of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_cor_bg_r">cor_bg_r</code></td>
<td>
<p>The radius of the background of the correlation text.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>The color palette for the scatter plots and default palette for the diagonal plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>Custom colors used to create a color palette for the scatter plots and diagonal plots.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="CorPairsPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code><a href="#topic+CorPlot">CorPlot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>theme</code> and <code>theme_args</code> are also supported, they will be passed to each individual plot.
</p>


<h3>Value</h3>

<p>A <code>patch_work::wrap_plots</code> object.
</p>

<hr>
<h2 id='CorPlot'>CorPlot</h2><span id='topic+CorPlot'></span>

<h3>Description</h3>

<p>Generate scatter correlation plot for two variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CorPlot(
  data,
  x,
  y,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  split_by = NULL,
  split_by_sep = "_",
  pt_size = 2,
  pt_shape = 16,
  raster = FALSE,
  alpha = 1,
  raster_dpi = c(512, 512),
  highlight = NULL,
  highlight_color = "black",
  highlight_size = 1,
  highlight_alpha = 1,
  highlight_stroke = 0.8,
  anno_items = c("eq", "r2", "p"),
  anno_size = 3,
  anno_fg = "black",
  anno_bg = "white",
  anno_bg_r = 0.1,
  anno_position = c("topleft", "topright", "bottomleft", "bottomright", "tl", "tr", "bl",
    "br"),
  add_smooth = TRUE,
  smooth_color = "red2",
  smooth_width = 1.5,
  smooth_se = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = ifelse(is.null(group_by), "Spectral", "Paired"),
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CorPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="CorPlot_+3A_group_name">group_name</code></td>
<td>
<p>The name of the group in the legend.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="CorPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>The size of the points.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_pt_shape">pt_shape</code></td>
<td>
<p>The shape of the points.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_raster">raster</code></td>
<td>
<p>Whether to use raster graphics for plotting.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_raster_dpi">raster_dpi</code></td>
<td>
<p>The DPI of the raster graphics.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_highlight">highlight</code></td>
<td>
<p>The items to be highlighted.
Could be either a vector of rownames if data has rownames, or a vector of indices, or
An expression that can be evaluated by <code>dplyr::filter</code> to get the highlighted items.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>The color of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>The size of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>The alpha of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_highlight_stroke">highlight_stroke</code></td>
<td>
<p>The stroke of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_anno_items">anno_items</code></td>
<td>
<p>The items to be annotated on the plot.
Available items: &quot;eq&quot;, &quot;r2&quot;, &quot;p&quot;, &quot;spearman&quot;, &quot;pearson&quot;, &quot;kendall&quot;, &quot;n&quot;.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_anno_size">anno_size</code></td>
<td>
<p>The size of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_anno_fg">anno_fg</code></td>
<td>
<p>The color of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_anno_bg">anno_bg</code></td>
<td>
<p>The background color of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_anno_bg_r">anno_bg_r</code></td>
<td>
<p>The radius of the background of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_anno_position">anno_position</code></td>
<td>
<p>The position of the annotation text.
Available positions: &quot;topleft&quot;, &quot;topright&quot;, &quot;bottomleft&quot;, &quot;bottomright&quot;.
Shortcuts: &quot;tl&quot;, &quot;tr&quot;, &quot;bl&quot;, &quot;br&quot;.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_add_smooth">add_smooth</code></td>
<td>
<p>Whether to add a linear regression line.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_smooth_color">smooth_color</code></td>
<td>
<p>The color of the regression line.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_smooth_width">smooth_width</code></td>
<td>
<p>The width of the regression line.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_smooth_se">smooth_se</code></td>
<td>
<p>Whether to add the standard error band to the regression line.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="CorPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="CorPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="CorPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or a list of ggplot objects if <code>combine</code> is <code>FALSE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
CorPlot(iris, "Sepal.Length", "Sepal.Width", group_by = "Species")
CorPlot(iris, "Sepal.Length", "Sepal.Width", group_by = "Species",
 highlight = 'Species == "setosa"', highlight_stroke = 1.5,
 anno_items = c("eq", "pearson"), anno_position = "bottomright")
CorPlot(iris, "Sepal.Length", "Sepal.Width", facet_by = "Species", facet_scales = "free")
CorPlot(iris, "Sepal.Length", "Sepal.Width", split_by = "Species",
        palette = c(setosa = "Set1", versicolor = "Dark2", virginica = "Paired"))
</code></pre>

<hr>
<h2 id='CorPlotAtomic'>Atomic Correlation Plot</h2><span id='topic+CorPlotAtomic'></span>

<h3>Description</h3>

<p>Generate scatter correlation plot for two variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CorPlotAtomic(
  data,
  x,
  y,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  pt_size = 2,
  pt_shape = 16,
  alpha = 1,
  raster = FALSE,
  raster_dpi = c(512, 512),
  highlight = NULL,
  highlight_color = "black",
  highlight_size = 1,
  highlight_alpha = 1,
  highlight_stroke = 0.8,
  anno_items = c("eq", "r2", "p"),
  anno_size = 3,
  anno_fg = "black",
  anno_bg = "white",
  anno_bg_r = 0.1,
  anno_position = c("topleft", "topright", "bottomleft", "bottomright", "tl", "tr", "bl",
    "br"),
  add_smooth = TRUE,
  smooth_color = "red2",
  smooth_width = 1.5,
  smooth_se = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = ifelse(is.null(group_by), "Spectral", "Paired"),
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CorPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_x">x</code></td>
<td>
<p>y The column names of the data to be plotted.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>The column name of the data to be used for grouping.
Different groups will be plotted in different colors.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator used to concatenate multiple columns in <code>group_by</code>.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>The name of the group in the legend.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>The size of the points.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_pt_shape">pt_shape</code></td>
<td>
<p>The shape of the points.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_raster">raster</code></td>
<td>
<p>Whether to use raster graphics for plotting.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_raster_dpi">raster_dpi</code></td>
<td>
<p>The DPI of the raster graphics.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_highlight">highlight</code></td>
<td>
<p>The items to be highlighted.
Could be either a vector of rownames if data has rownames, or a vector of indices, or
An expression that can be evaluated by <code>dplyr::filter</code> to get the highlighted items.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_highlight_color">highlight_color</code></td>
<td>
<p>The color of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_highlight_size">highlight_size</code></td>
<td>
<p>The size of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>The alpha of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_highlight_stroke">highlight_stroke</code></td>
<td>
<p>The stroke of the highlighted points.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_anno_items">anno_items</code></td>
<td>
<p>The items to be annotated on the plot.
Available items: &quot;eq&quot;, &quot;r2&quot;, &quot;p&quot;, &quot;spearman&quot;, &quot;pearson&quot;, &quot;kendall&quot;, &quot;n&quot;.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_anno_size">anno_size</code></td>
<td>
<p>The size of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_anno_fg">anno_fg</code></td>
<td>
<p>The color of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_anno_bg">anno_bg</code></td>
<td>
<p>The background color of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_anno_bg_r">anno_bg_r</code></td>
<td>
<p>The radius of the background of the annotation text.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_anno_position">anno_position</code></td>
<td>
<p>The position of the annotation text.
Available positions: &quot;topleft&quot;, &quot;topright&quot;, &quot;bottomleft&quot;, &quot;bottomright&quot;.
Shortcuts: &quot;tl&quot;, &quot;tr&quot;, &quot;bl&quot;, &quot;br&quot;.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_add_smooth">add_smooth</code></td>
<td>
<p>Whether to add a linear regression line.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_smooth_color">smooth_color</code></td>
<td>
<p>The color of the regression line.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_smooth_width">smooth_width</code></td>
<td>
<p>The width of the regression line.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_smooth_se">smooth_se</code></td>
<td>
<p>Whether to add the standard error band to the regression line.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="CorPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='DensityHistoPlotAtomic'>Atomic density/histogram plot</h2><span id='topic+DensityHistoPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic density/histogram plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DensityHistoPlotAtomic(
  data,
  x,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  xtrans = "identity",
  ytrans = "identity",
  type = c("density", "histogram"),
  bins = NULL,
  binwidth = NULL,
  flip = FALSE,
  add_bars = FALSE,
  bar_height = 0.025,
  bar_alpha = 1,
  bar_width = 0.1,
  position = "identity",
  use_trend = FALSE,
  add_trend = FALSE,
  trend_alpha = 1,
  trend_linewidth = 0.8,
  trend_pt_size = 1.5,
  trend_skip_zero = FALSE,
  palette = "Paired",
  palcolor = NULL,
  alpha = 0.5,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  expand = c(bottom = 0, left = 0, right = 0),
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  legend.position = ifelse(is.null(group_by), "none", "right"),
  legend.direction = "vertical",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DensityHistoPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name for the values
A numeric column is expected.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name to group the data</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code> if multiple columns are provided</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of group_by</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_xtrans">xtrans</code></td>
<td>
<p>A character string specifying the transformation of the x-axis. Default is &quot;identity&quot;.
Other options see transform of <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code>.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_ytrans">ytrans</code></td>
<td>
<p>A character string specifying the transformation of the y-axis. Default is &quot;identity&quot;.
Other options see transform of <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code>.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_type">type</code></td>
<td>
<p>A character string specifying the type of plot. Default is &quot;density&quot;.
Other options are &quot;histogram&quot;.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_bins">bins</code></td>
<td>
<p>A numeric value specifying the number of bins for the histogram.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_binwidth">binwidth</code></td>
<td>
<p>A numeric value specifying the width of the bins for the histogram.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value. If TRUE, the plot will be flipped.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_add_bars">add_bars</code></td>
<td>
<p>A logical value. If TRUE, add lines to the plot to show the data distribution on the bottom.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_bar_height">bar_height</code></td>
<td>
<p>A numeric value specifying the height of the bars. The actual height will be calculated based on the maximum density or count.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_bar_alpha">bar_alpha</code></td>
<td>
<p>A numeric value specifying the alpha of the bars.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_bar_width">bar_width</code></td>
<td>
<p>A numeric value specifying the width of the bars.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_position">position</code></td>
<td>
<p>How should we position the values in each bin? Default is &quot;identity&quot;.
Unlike the default position = &quot;stack&quot; in <a href="ggplot2.html#topic+geom_histogram">ggplot2::geom_histogram</a> or <a href="ggplot2.html#topic+geom_density">ggplot2::geom_density</a>,
the default position is &quot;identity&quot; to show the actual count or density for each group.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_use_trend">use_trend</code></td>
<td>
<p>A logical value. If TRUE, use trend line instead of histogram.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value. If TRUE, add trend line to the histogram.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_trend_alpha">trend_alpha</code></td>
<td>
<p>A numeric value specifying the alpha of the trend line and points</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value specifying the width of the trend line</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_trend_pt_size">trend_pt_size</code></td>
<td>
<p>A numeric value specifying the size of the trend points</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_trend_skip_zero">trend_skip_zero</code></td>
<td>
<p>A logical value. If TRUE, skip the zero count when drawing the trend line.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="DensityHistoPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='DensityPlot'>Density Plot / Histogram</h2><span id='topic+DensityPlot'></span><span id='topic+Histogram'></span>

<h3>Description</h3>

<p>Density plot and histogram to illustrate the distribution of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DensityPlot(
  data,
  x,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  xtrans = "identity",
  ytrans = "identity",
  split_by = NULL,
  split_by_sep = "_",
  flip = FALSE,
  position = "identity",
  palette = "Paired",
  palcolor = NULL,
  alpha = 0.5,
  theme = "theme_this",
  theme_args = list(),
  add_bars = FALSE,
  bar_height = 0.025,
  bar_alpha = 1,
  bar_width = 0.1,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  expand = c(bottom = 0, left = 0, right = 0),
  facet_by = NULL,
  facet_scales = "free_y",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  legend.position = ifelse(is.null(group_by), "none", "right"),
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

Histogram(
  data,
  x,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  xtrans = "identity",
  ytrans = "identity",
  split_by = NULL,
  split_by_sep = "_",
  flip = FALSE,
  bins = NULL,
  binwidth = NULL,
  trend_skip_zero = FALSE,
  add_bars = FALSE,
  bar_height = 0.025,
  bar_alpha = 1,
  bar_width = 0.1,
  position = "identity",
  use_trend = FALSE,
  add_trend = FALSE,
  trend_alpha = 1,
  trend_linewidth = 0.8,
  trend_pt_size = 1.5,
  palette = "Paired",
  palcolor = NULL,
  alpha = 0.5,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  expand = c(bottom = 0, left = 0, right = 0),
  facet_by = NULL,
  facet_scales = "free_y",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  legend.position = ifelse(is.null(group_by), "none", "right"),
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DensityPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of group_by</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_xtrans">xtrans</code></td>
<td>
<p>A character string specifying the transformation of the x-axis. Default is &quot;identity&quot;.
Other options see transform of <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code>.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_ytrans">ytrans</code></td>
<td>
<p>A character string specifying the transformation of the y-axis. Default is &quot;identity&quot;.
Other options see transform of <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code>.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value. If TRUE, the plot will be flipped.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_position">position</code></td>
<td>
<p>How should we position the values in each bin? Default is &quot;identity&quot;.
Unlike the default position = &quot;stack&quot; in <a href="ggplot2.html#topic+geom_histogram">ggplot2::geom_histogram</a> or <a href="ggplot2.html#topic+geom_density">ggplot2::geom_density</a>,
the default position is &quot;identity&quot; to show the actual count or density for each group.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_add_bars">add_bars</code></td>
<td>
<p>A logical value. If TRUE, add lines to the plot to show the data distribution on the bottom.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_bar_height">bar_height</code></td>
<td>
<p>A numeric value specifying the height of the bars. The actual height will be calculated based on the maximum density or count.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_bar_alpha">bar_alpha</code></td>
<td>
<p>A numeric value specifying the alpha of the bars.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_bar_width">bar_width</code></td>
<td>
<p>A numeric value specifying the width of the bars.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_bins">bins</code></td>
<td>
<p>A numeric value specifying the number of bins for the histogram.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_binwidth">binwidth</code></td>
<td>
<p>A numeric value specifying the width of the bins for the histogram.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_trend_skip_zero">trend_skip_zero</code></td>
<td>
<p>A logical value. If TRUE, skip the zero count when drawing the trend line.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_use_trend">use_trend</code></td>
<td>
<p>A logical value. If TRUE, use trend line instead of histogram.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_add_trend">add_trend</code></td>
<td>
<p>A logical value. If TRUE, add trend line to the histogram.</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_trend_alpha">trend_alpha</code></td>
<td>
<p>A numeric value specifying the alpha of the trend line and points</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_trend_linewidth">trend_linewidth</code></td>
<td>
<p>A numeric value specifying the width of the trend line</p>
</td></tr>
<tr><td><code id="DensityPlot_+3A_trend_pt_size">trend_pt_size</code></td>
<td>
<p>A numeric value specifying the size of the trend points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(8525)
data &lt;- data.frame(
    x = c(rnorm(500, -1), rnorm(500, 1)),
    group = rep(c("A", "B"), each = 500),
    facet = sample(c("F1", "F2"), 1000, replace = TRUE)
)

DensityPlot(data, x = "x")
DensityPlot(data, x = "x", group_by = "group", facet_by = "facet")
DensityPlot(data, x = "x", split_by = "facet", add_bars = TRUE)
DensityPlot(data, x = "x", split_by = "facet", add_bars = TRUE,
    palette = c(F1 = "Set1", F2 = "Set2"))
set.seed(8525)
data &lt;- data.frame(
    x = sample(setdiff(1:100, c(30:36, 50:55, 70:77)), 1000, replace = TRUE),
    group = factor(rep(c("A", "B"), each = 500), levels = c("A", "B")),
    facet = sample(c("F1", "F2"), 1000, replace = TRUE)
)

Histogram(data, x = "x")
Histogram(data, x = "x", group_by = "group")
Histogram(data, x = "x", split_by = "facet", add_bars = TRUE)
Histogram(data, x = "x", group_by = "group", add_trend = TRUE)
Histogram(data, x = "x", group_by = "group", add_trend = TRUE, trend_skip_zero = TRUE)
Histogram(data, x = "x", group_by = "group", split_by = "facet",
 use_trend = TRUE, trend_pt_size = 3)
Histogram(data, x = "x", group_by = "group", split_by = "facet",
 palette = c(F1 = "Paired", F2 = "Spectral"))
</code></pre>

<hr>
<h2 id='detect_upset_datatype'>Detect the type of the input data for Upset plot</h2><span id='topic+detect_upset_datatype'></span>

<h3>Description</h3>

<p>Detect the type of the input data for Upset plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detect_upset_datatype(data, group_by = NULL, id_by = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="detect_upset_datatype_+3A_data">data</code></td>
<td>
<p>A data frame or a list</p>
</td></tr>
<tr><td><code id="detect_upset_datatype_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to group the data.</p>
</td></tr>
<tr><td><code id="detect_upset_datatype_+3A_id_by">id_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to identify the instances.
Required when <code>group_by</code> is a single column and data is a data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string indicating the datatype of the input data or error message if invalid.
Possible values are &quot;long&quot;, &quot;wide&quot;, &quot;list&quot; and &quot;upset&quot;.
&quot;long&quot; indicates the data is in long format.
&quot;wide&quot; indicates the data is in wide format.
&quot;list&quot; indicates the data is a list.
&quot;upset&quot; indicates the data is a UpsetPlotData object.
</p>

<hr>
<h2 id='detect_venn_datatype'>Detect the datatype of the input data of Venn diagram</h2><span id='topic+detect_venn_datatype'></span>

<h3>Description</h3>

<p>Detect the datatype of the input data of Venn diagram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detect_venn_datatype(data, group_by = NULL, id_by = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="detect_venn_datatype_+3A_data">data</code></td>
<td>
<p>A data frame or a list or a VennPlotData object.</p>
</td></tr>
<tr><td><code id="detect_venn_datatype_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to group the data.</p>
</td></tr>
<tr><td><code id="detect_venn_datatype_+3A_id_by">id_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to identify the instances.
Required when <code>group_by</code> is a single column and data is a data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string indicating the datatype of the input data or error message if invalid.
Possible values are &quot;long&quot;, &quot;wide&quot;, &quot;list&quot; and &quot;venn&quot;.
&quot;long&quot; indicates the data is in long format.
&quot;wide&quot; indicates the data is in wide format.
&quot;list&quot; indicates the data is a list.
&quot;venn&quot; indicates the data is a VennPlotData object.
</p>

<hr>
<h2 id='DimPlot'>DimPLot / FeatureDimPlot</h2><span id='topic+DimPlot'></span><span id='topic+FeatureDimPlot'></span>

<h3>Description</h3>

<p>Visualizing the dimension reduction data.
<code>FeatureDimPlot</code> is used to plot the feature numeric values on the dimension reduction plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DimPlot(
  data,
  dims = 1:2,
  group_by,
  group_by_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  pt_size = NULL,
  pt_alpha = 1,
  bg_color = "grey80",
  label_insitu = FALSE,
  show_stat = !identical(theme, "theme_blank"),
  label = FALSE,
  label_size = 4,
  label_fg = "white",
  label_bg = "black",
  label_bg_r = 0.1,
  label_repel = FALSE,
  label_repulsion = 20,
  label_pt_size = 1,
  label_pt_color = "black",
  label_segment_color = "black",
  highlight = NULL,
  highlight_alpha = 1,
  highlight_size = 1,
  highlight_color = "black",
  highlight_stroke = 0.8,
  add_mark = FALSE,
  mark_type = c("hull", "ellipse", "rect", "circle"),
  mark_expand = unit(3, "mm"),
  mark_alpha = 0.1,
  mark_linetype = 1,
  stat_by = NULL,
  stat_plot_type = c("pie", "ring", "bar", "line"),
  stat_plot_size = 0.1,
  stat_args = list(palette = "Set1"),
  graph = NULL,
  edge_size = c(0.05, 0.5),
  edge_alpha = 0.1,
  edge_color = "grey40",
  add_density = FALSE,
  density_color = "grey80",
  density_filled = FALSE,
  density_filled_palette = "Greys",
  density_filled_palcolor = NULL,
  lineages = NULL,
  lineages_trim = c(0.01, 0.99),
  lineages_span = 0.75,
  lineages_palette = "Dark2",
  lineages_palcolor = NULL,
  lineages_arrow = arrow(length = unit(0.1, "inches")),
  lineages_linewidth = 1,
  lineages_line_bg = "white",
  lineages_line_bg_stroke = 0.5,
  lineages_whiskers = FALSE,
  lineages_whiskers_linewidth = 0.5,
  lineages_whiskers_alpha = 0.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  theme = "theme_this",
  theme_args = list(),
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  raster = NULL,
  raster_dpi = c(512, 512),
  hex = FALSE,
  hex_linewidth = 0.5,
  hex_count = TRUE,
  hex_bins = 50,
  hex_binwidth = NULL,
  palette = "Paired",
  palcolor = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

FeatureDimPlot(
  data,
  dims = 1:2,
  features,
  split_by = NULL,
  split_by_sep = "_",
  lower_quantile = 0,
  upper_quantile = 0.99,
  lower_cutoff = NULL,
  upper_cutoff = NULL,
  pt_size = NULL,
  pt_alpha = 1,
  bg_color = "grey80",
  bg_cutoff = NULL,
  label_insitu = FALSE,
  show_stat = !identical(theme, "theme_blank"),
  color_name = "",
  label = FALSE,
  label_size = 4,
  label_fg = "white",
  label_bg = "black",
  label_bg_r = 0.1,
  label_repel = FALSE,
  label_repulsion = 20,
  label_pt_size = 1,
  label_pt_color = "black",
  label_segment_color = "black",
  highlight = NULL,
  highlight_alpha = 1,
  highlight_size = 1,
  highlight_color = "black",
  highlight_stroke = 0.8,
  add_mark = FALSE,
  mark_type = c("hull", "ellipse", "rect", "circle"),
  mark_expand = unit(3, "mm"),
  mark_alpha = 0.1,
  mark_linetype = 1,
  stat_by = NULL,
  stat_plot_type = c("pie", "ring", "bar", "line"),
  stat_plot_size = 0.1,
  stat_args = list(palette = "Set1"),
  graph = NULL,
  edge_size = c(0.05, 0.5),
  edge_alpha = 0.1,
  edge_color = "grey40",
  add_density = FALSE,
  density_color = "grey80",
  density_filled = FALSE,
  density_filled_palette = "Greys",
  density_filled_palcolor = NULL,
  lineages = NULL,
  lineages_trim = c(0.01, 0.99),
  lineages_span = 0.75,
  lineages_palette = "Dark2",
  lineages_palcolor = NULL,
  lineages_arrow = arrow(length = unit(0.1, "inches")),
  lineages_linewidth = 1,
  lineages_line_bg = "white",
  lineages_line_bg_stroke = 0.5,
  lineages_whiskers = FALSE,
  lineages_whiskers_linewidth = 0.5,
  lineages_whiskers_alpha = 0.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  theme = "theme_this",
  theme_args = list(),
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  raster = NULL,
  raster_dpi = c(512, 512),
  hex = FALSE,
  hex_linewidth = 0.5,
  hex_count = FALSE,
  hex_bins = 50,
  hex_binwidth = NULL,
  palette = "Spectral",
  palcolor = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DimPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_dims">dims</code></td>
<td>
<p>A character vector of the column names to plot on the x and y axes or a numeric vector of the column indices.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="DimPlot_+3A_split_by">split_by</code></td>
<td>
<p>A character vector of column names to split the data and plot separately
If TRUE, we will split the data by the <code>features</code>. Each feature will be plotted separately.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="DimPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value of the point size. If NULL, the point size will be calculated based on the number of data points.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value of the point transparency. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_bg_color">bg_color</code></td>
<td>
<p>A character string of the background or NA points. Default is &quot;grey80&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_insitu">label_insitu</code></td>
<td>
<p>Whether to place the raw labels (group names) in the center of the points with the corresponding group. Default is FALSE, which using numbers instead of raw labels.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_show_stat">show_stat</code></td>
<td>
<p>Whether to show the number of points in the subtitle. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label">label</code></td>
<td>
<p>Whether to show the labels of groups. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value of the label size. Default is 4.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string of the label foreground color. Default is &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string of the label background color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value of the background ratio of the labels. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_repel">label_repel</code></td>
<td>
<p>Whether to repel the labels. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_repulsion">label_repulsion</code></td>
<td>
<p>A numeric value of the label repulsion. Default is 20.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_pt_size">label_pt_size</code></td>
<td>
<p>A numeric value of the label point size. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_pt_color">label_pt_color</code></td>
<td>
<p>A character string of the label point color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_label_segment_color">label_segment_color</code></td>
<td>
<p>A character string of the label segment color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of the row names to highlight. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value of the highlight transparency. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value of the highlight size. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string of the highlight color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_highlight_stroke">highlight_stroke</code></td>
<td>
<p>A numeric value of the highlight stroke. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_add_mark">add_mark</code></td>
<td>
<p>Whether to add mark to the plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_mark_type">mark_type</code></td>
<td>
<p>A character string of the mark type. Default is &quot;hull&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_mark_expand">mark_expand</code></td>
<td>
<p>A unit value of the mark expand. Default is 3mm.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_mark_alpha">mark_alpha</code></td>
<td>
<p>A numeric value of the mark transparency. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_mark_linetype">mark_linetype</code></td>
<td>
<p>A numeric value of the mark line type. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_stat_by">stat_by</code></td>
<td>
<p>A character string of the column name to calculate the statistics. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_stat_plot_type">stat_plot_type</code></td>
<td>
<p>A character string of the statistic plot type. Default is &quot;pie&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_stat_plot_size">stat_plot_size</code></td>
<td>
<p>A numeric value of the statistic plot size. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_stat_args">stat_args</code></td>
<td>
<p>A list of additional arguments to the statistic plot. Default is list(palette = &quot;Set1&quot;).</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_graph">graph</code></td>
<td>
<p>A character string of column names or the indexes in the data for the graph data. Default is NULL.
If &quot;@graph&quot; is provided, the graph data will be extracted from the data attribute 'graph'.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_edge_size">edge_size</code></td>
<td>
<p>A numeric vector of the edge size range. Default is c(0.05, 0.5).</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_edge_alpha">edge_alpha</code></td>
<td>
<p>A numeric value of the edge transparency. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_edge_color">edge_color</code></td>
<td>
<p>A character string of the edge color. Default is &quot;grey40&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_add_density">add_density</code></td>
<td>
<p>Whether to add density plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_density_color">density_color</code></td>
<td>
<p>A character string of the density color. Default is &quot;grey80&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_density_filled">density_filled</code></td>
<td>
<p>Whether to fill the density plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_density_filled_palette">density_filled_palette</code></td>
<td>
<p>A character string of the filled density palette. Default is &quot;Greys&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_density_filled_palcolor">density_filled_palcolor</code></td>
<td>
<p>A character vector of the filled density palette colors. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages">lineages</code></td>
<td>
<p>A character vector of the column names for lineages. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_trim">lineages_trim</code></td>
<td>
<p>A numeric vector of the trim range for lineages. Default is c(0.01, 0.99).</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_span">lineages_span</code></td>
<td>
<p>A numeric value of the lineages span. Default is 0.75.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_palette">lineages_palette</code></td>
<td>
<p>A character string of the lineages palette. Default is &quot;Dark2&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_palcolor">lineages_palcolor</code></td>
<td>
<p>A character vector of the lineages palette colors. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_arrow">lineages_arrow</code></td>
<td>
<p>An arrow object for the lineages. Default is arrow(length = unit(0.1, &quot;inches&quot;)).</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_linewidth">lineages_linewidth</code></td>
<td>
<p>A numeric value of the lineages line width. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_line_bg">lineages_line_bg</code></td>
<td>
<p>A character string of the lineages line background color. Default is &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_line_bg_stroke">lineages_line_bg_stroke</code></td>
<td>
<p>A numeric value of the lineages line background stroke. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_whiskers">lineages_whiskers</code></td>
<td>
<p>Whether to add whiskers to the lineages. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_whiskers_linewidth">lineages_whiskers_linewidth</code></td>
<td>
<p>A numeric value of the lineages whiskers line width. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lineages_whiskers_alpha">lineages_whiskers_alpha</code></td>
<td>
<p>A numeric value of the lineages whiskers transparency. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="DimPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="DimPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_raster">raster</code></td>
<td>
<p>Whether to raster the plot. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_raster_dpi">raster_dpi</code></td>
<td>
<p>A numeric vector of the raster dpi. Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_hex">hex</code></td>
<td>
<p>Whether to use hex plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_hex_linewidth">hex_linewidth</code></td>
<td>
<p>A numeric value of the hex line width. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_hex_count">hex_count</code></td>
<td>
<p>Whether to count the hex.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_hex_bins">hex_bins</code></td>
<td>
<p>A numeric value of the hex bins. Default is 50.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_hex_binwidth">hex_binwidth</code></td>
<td>
<p>A numeric value of the hex bin width. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_features">features</code></td>
<td>
<p>A character vector of the column names to plot as features.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_lower_quantile">lower_quantile</code>, <code id="DimPlot_+3A_upper_quantile">upper_quantile</code>, <code id="DimPlot_+3A_lower_cutoff">lower_cutoff</code>, <code id="DimPlot_+3A_upper_cutoff">upper_cutoff</code></td>
<td>
<p>Vector of minimum and maximum cutoff values or quantile values for each feature.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_bg_cutoff">bg_cutoff</code></td>
<td>
<p>A numeric value to be used a cutoff to set the feature values to NA. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_+3A_color_name">color_name</code></td>
<td>
<p>A character string of the color legend name. Default is &quot;&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>
<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# generate a PCA dimension data for DimPlot
set.seed(8525)
df &lt;- matrix(c(rnorm(333), rnorm(334, 0.2), rnorm(333, .4)), ncol = 10)
# run PCA
pca &lt;- prcomp(df)
# get coordinates
data &lt;- pca$x[, 1:2]
# kmeans clustering
km &lt;- kmeans(data, 3)
data &lt;- as.data.frame(data)
data$cluster &lt;- factor(paste0("C", km$cluster))
data$group &lt;- sample(c("A", "B"), nrow(data), replace = TRUE)

graph &lt;- rnorm(nrow(data) * nrow(data))
graph[sample(1:(nrow(data) * nrow(data)), 5000)] &lt;- NA
graph &lt;- matrix(graph, nrow = nrow(data))
rownames(graph) &lt;- colnames(graph) &lt;- rownames(data)

attr(data, "graph") &lt;- graph

data$L1 &lt;- rnorm(nrow(data), 0, 0.1)
data$L2 &lt;- rnorm(nrow(data), 1, 0.2)
data$L3 &lt;- rnorm(nrow(data), 2, 0.3)

DimPlot(data, group_by = "cluster")
DimPlot(data, group_by = "cluster", theme = "theme_blank")
DimPlot(data, group_by = "cluster", theme = ggplot2::theme_classic,
        theme_args = list(base_size = 16), palette = "seurat")
DimPlot(data, group_by = "cluster", raster = TRUE, raster_dpi = 30)
DimPlot(data, group_by = "cluster", highlight = 1:20,
        highlight_color = "red2", highlight_stroke = 0.8)
DimPlot(data, group_by = "cluster", highlight = TRUE, facet_by = "group",
        theme = "theme_blank")
DimPlot(data, group_by = "cluster", label = TRUE)
DimPlot(data, group_by = "cluster", label = TRUE, label_fg = "red",
        label_bg = "yellow", label_size = 5)
DimPlot(data, group_by = "cluster", label = TRUE, label_insitu = TRUE)
DimPlot(data, group_by = "cluster", add_mark = TRUE)
DimPlot(data, group_by = "cluster", add_mark = TRUE, mark_linetype = 2)
DimPlot(data, group_by = "cluster", add_mark = TRUE, mark_type = "ellipse")
DimPlot(data, group_by = "cluster", add_density = TRUE)
DimPlot(data, group_by = "cluster", add_density = TRUE, density_filled = TRUE)
DimPlot(data, group_by = "cluster", add_density = TRUE, density_filled = TRUE,
        density_filled_palette = "Blues", highlight = TRUE)
DimPlot(data, group_by = "cluster", stat_by = "group")
DimPlot(data, group_by = "cluster", stat_by = "group", stat_plot_type = "bar")
DimPlot(data, group_by = "cluster", hex = TRUE)
DimPlot(data, group_by = "cluster", hex = TRUE, hex_bins = 20)
DimPlot(data, group_by = "cluster", hex = TRUE, hex_count = FALSE)
DimPlot(data, group_by = "cluster", graph = "@graph", edge_color = "grey80")
DimPlot(data, group_by = "cluster", lineages = c("L1", "L2", "L3"))
DimPlot(data, group_by = "cluster", lineages = c("L1", "L2", "L3"),
        lineages_whiskers = TRUE)
DimPlot(data, group_by = "cluster", lineages = c("L1", "L2", "L3"),
        lineages_span = 1)
DimPlot(data, group_by = "cluster",  split_by = "cluster",
        palcolor = list(C1 = "red", C2 = "blue", C3 = "green"))


# Feature Dim Plot
FeatureDimPlot(data, features = "L1", pt_size = 2)
FeatureDimPlot(data, features = "L1", pt_size = 2, bg_cutoff = -Inf)
FeatureDimPlot(data, features = "L1", raster = TRUE, raster_dpi = 30)
FeatureDimPlot(data, features = c("L1", "L2"), pt_size = 2)
FeatureDimPlot(data, features = c("L1"), pt_size = 2, facet_by = "group")
# Can't facet multiple features
FeatureDimPlot(data, features = c("L1", "L2", "L3"), pt_size = 2)
# We can use split_by
FeatureDimPlot(data, features = c("L1", "L2", "L3"), split_by = "group", nrow = 2)
FeatureDimPlot(data, features = c("L1", "L2", "L3"), highlight = TRUE)
FeatureDimPlot(data, features = c("L1", "L2", "L3"), hex = TRUE, hex_bins = 15)
FeatureDimPlot(data, features = c("L1", "L2", "L3"), hex = TRUE, hex_bins = 15,
  split_by = "cluster", palette = list(C1 = "Reds", C2 = "Blues", C3 = "Greens"))

</code></pre>

<hr>
<h2 id='DimPlotAtomic'>Atomic Dimension Reduction Plot without splitting the data</h2><span id='topic+DimPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic Dimension Reduction Plot without splitting the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DimPlotAtomic(
  data,
  dims = 1:2,
  group_by = NULL,
  group_by_sep = "_",
  features = NULL,
  lower_quantile = 0,
  upper_quantile = 0.99,
  lower_cutoff = NULL,
  upper_cutoff = NULL,
  pt_size = NULL,
  pt_alpha = 1,
  bg_color = "grey80",
  bg_cutoff = NULL,
  color_name = "",
  label_insitu = FALSE,
  show_stat = !identical(theme, "theme_blank"),
  label = FALSE,
  label_size = 4,
  label_fg = "white",
  label_bg = "black",
  label_bg_r = 0.1,
  label_repel = FALSE,
  label_repulsion = 20,
  label_pt_size = 1,
  label_pt_color = "black",
  label_segment_color = "black",
  highlight = NULL,
  highlight_alpha = 1,
  highlight_size = 1,
  highlight_color = "black",
  highlight_stroke = 0.8,
  add_mark = FALSE,
  mark_type = c("hull", "ellipse", "rect", "circle"),
  mark_expand = unit(3, "mm"),
  mark_alpha = 0.1,
  mark_linetype = 1,
  stat_by = NULL,
  stat_plot_type = c("pie", "ring", "bar", "line"),
  stat_plot_size = 0.1,
  stat_palette = "Set1",
  stat_args = list(),
  graph = NULL,
  edge_size = c(0.05, 0.5),
  edge_alpha = 0.1,
  edge_color = "grey40",
  add_density = FALSE,
  density_color = "grey80",
  density_filled = FALSE,
  density_filled_palette = "Greys",
  density_filled_palcolor = NULL,
  lineages = NULL,
  lineages_trim = c(0.01, 0.99),
  lineages_span = 0.75,
  lineages_palette = "Dark2",
  lineages_palcolor = NULL,
  lineages_arrow = ggplot2::arrow(length = unit(0.1, "inches")),
  lineages_linewidth = 1,
  lineages_line_bg = "white",
  lineages_line_bg_stroke = 0.5,
  lineages_whiskers = FALSE,
  lineages_whiskers_linewidth = 0.5,
  lineages_whiskers_alpha = 0.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  theme = "theme_this",
  theme_args = list(),
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  raster = NULL,
  raster_dpi = c(512, 512),
  hex = FALSE,
  hex_linewidth = 0.5,
  hex_count = !is.null(group_by),
  hex_bins = 50,
  hex_binwidth = NULL,
  palette = ifelse(is.null(features), "Paired", "Spectral"),
  palcolor = NULL,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DimPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_dims">dims</code></td>
<td>
<p>A character vector of the column names to plot on the x and y axes or a numeric vector of the column indices.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string of the column name to group the data.
A character/factor column is expected. If multiple columns are provided, the columns will be concatenated with <code>group_by_sep</code>.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_features">features</code></td>
<td>
<p>A character vector of the column names to plot as features.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lower_quantile">lower_quantile</code>, <code id="DimPlotAtomic_+3A_upper_quantile">upper_quantile</code>, <code id="DimPlotAtomic_+3A_lower_cutoff">lower_cutoff</code>, <code id="DimPlotAtomic_+3A_upper_cutoff">upper_cutoff</code></td>
<td>
<p>Vector of minimum and maximum cutoff values or quantile values for each feature.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value of the point size. If NULL, the point size will be calculated based on the number of data points.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value of the point transparency. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_bg_color">bg_color</code></td>
<td>
<p>A character string of the background or NA points. Default is &quot;grey80&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_bg_cutoff">bg_cutoff</code></td>
<td>
<p>A numeric value to be used a cutoff to set the feature values to NA. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_color_name">color_name</code></td>
<td>
<p>A character string of the color legend name. Default is &quot;&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_insitu">label_insitu</code></td>
<td>
<p>Whether to place the raw labels (group names) in the center of the points with the corresponding group. Default is FALSE, which using numbers instead of raw labels.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_show_stat">show_stat</code></td>
<td>
<p>Whether to show the number of points in the subtitle. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label">label</code></td>
<td>
<p>Whether to show the labels of groups. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value of the label size. Default is 4.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string of the label foreground color. Default is &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string of the label background color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value of the background ratio of the labels. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_repel">label_repel</code></td>
<td>
<p>Whether to repel the labels. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_repulsion">label_repulsion</code></td>
<td>
<p>A numeric value of the label repulsion. Default is 20.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_pt_size">label_pt_size</code></td>
<td>
<p>A numeric value of the label point size. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_pt_color">label_pt_color</code></td>
<td>
<p>A character string of the label point color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_label_segment_color">label_segment_color</code></td>
<td>
<p>A character string of the label segment color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of the row names to highlight. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value of the highlight transparency. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value of the highlight size. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string of the highlight color. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_highlight_stroke">highlight_stroke</code></td>
<td>
<p>A numeric value of the highlight stroke. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_add_mark">add_mark</code></td>
<td>
<p>Whether to add mark to the plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_mark_type">mark_type</code></td>
<td>
<p>A character string of the mark type. Default is &quot;hull&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_mark_expand">mark_expand</code></td>
<td>
<p>A unit value of the mark expand. Default is 3mm.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_mark_alpha">mark_alpha</code></td>
<td>
<p>A numeric value of the mark transparency. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_mark_linetype">mark_linetype</code></td>
<td>
<p>A numeric value of the mark line type. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_stat_by">stat_by</code></td>
<td>
<p>A character string of the column name to calculate the statistics. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_stat_plot_type">stat_plot_type</code></td>
<td>
<p>A character string of the statistic plot type. Default is &quot;pie&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_stat_plot_size">stat_plot_size</code></td>
<td>
<p>A numeric value of the statistic plot size. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_stat_args">stat_args</code></td>
<td>
<p>A list of additional arguments to the statistic plot. Default is list(palette = &quot;Set1&quot;).</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_graph">graph</code></td>
<td>
<p>A character string of column names or the indexes in the data for the graph data. Default is NULL.
If &quot;@graph&quot; is provided, the graph data will be extracted from the data attribute 'graph'.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_edge_size">edge_size</code></td>
<td>
<p>A numeric vector of the edge size range. Default is c(0.05, 0.5).</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_edge_alpha">edge_alpha</code></td>
<td>
<p>A numeric value of the edge transparency. Default is 0.1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_edge_color">edge_color</code></td>
<td>
<p>A character string of the edge color. Default is &quot;grey40&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_add_density">add_density</code></td>
<td>
<p>Whether to add density plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_density_color">density_color</code></td>
<td>
<p>A character string of the density color. Default is &quot;grey80&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_density_filled">density_filled</code></td>
<td>
<p>Whether to fill the density plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_density_filled_palette">density_filled_palette</code></td>
<td>
<p>A character string of the filled density palette. Default is &quot;Greys&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_density_filled_palcolor">density_filled_palcolor</code></td>
<td>
<p>A character vector of the filled density palette colors. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages">lineages</code></td>
<td>
<p>A character vector of the column names for lineages. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_trim">lineages_trim</code></td>
<td>
<p>A numeric vector of the trim range for lineages. Default is c(0.01, 0.99).</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_span">lineages_span</code></td>
<td>
<p>A numeric value of the lineages span. Default is 0.75.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_palette">lineages_palette</code></td>
<td>
<p>A character string of the lineages palette. Default is &quot;Dark2&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_palcolor">lineages_palcolor</code></td>
<td>
<p>A character vector of the lineages palette colors. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_arrow">lineages_arrow</code></td>
<td>
<p>An arrow object for the lineages. Default is arrow(length = unit(0.1, &quot;inches&quot;)).</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_linewidth">lineages_linewidth</code></td>
<td>
<p>A numeric value of the lineages line width. Default is 1.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_line_bg">lineages_line_bg</code></td>
<td>
<p>A character string of the lineages line background color. Default is &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_line_bg_stroke">lineages_line_bg_stroke</code></td>
<td>
<p>A numeric value of the lineages line background stroke. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_whiskers">lineages_whiskers</code></td>
<td>
<p>Whether to add whiskers to the lineages. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_whiskers_linewidth">lineages_whiskers_linewidth</code></td>
<td>
<p>A numeric value of the lineages whiskers line width. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_lineages_whiskers_alpha">lineages_whiskers_alpha</code></td>
<td>
<p>A numeric value of the lineages whiskers transparency. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_raster">raster</code></td>
<td>
<p>Whether to raster the plot. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_raster_dpi">raster_dpi</code></td>
<td>
<p>A numeric vector of the raster dpi. Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_hex">hex</code></td>
<td>
<p>Whether to use hex plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_hex_linewidth">hex_linewidth</code></td>
<td>
<p>A numeric value of the hex line width. Default is 0.5.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_hex_count">hex_count</code></td>
<td>
<p>Whether to count the hex.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_hex_bins">hex_bins</code></td>
<td>
<p>A numeric value of the hex bins. Default is 50.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_hex_binwidth">hex_binwidth</code></td>
<td>
<p>A numeric value of the hex bin width. Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="DimPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='DotPlot'>Dot Plot / Scatter Plot / Lollipop Plot</h2><span id='topic+DotPlot'></span><span id='topic+LollipopPlot'></span>

<h3>Description</h3>

<p>For <code>DotPlot</code>, X-axis and Y-axis could be either numeric or factor/character.
When x-axis and y-axis are both numeric, the plot works as a scatter plot.
<code>LollipopPlot</code> is an alias of <code>DotPlot</code> when <code>lollipop</code> = TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DotPlot(
  data,
  x,
  y,
  x_sep = "_",
  y_sep = "_",
  flip = FALSE,
  split_by = NULL,
  split_by_sep = "_",
  size_name = NULL,
  fill_name = NULL,
  fill_cutoff_name = NULL,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  bg_direction = c("vertical", "horizontal", "v", "h"),
  size_by = NULL,
  fill_by = NULL,
  fill_cutoff = NULL,
  fill_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  seed = 8525,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

LollipopPlot(
  data,
  x,
  y,
  y_sep = NULL,
  flip = FALSE,
  split_by = NULL,
  split_by_sep = "_",
  size_name = NULL,
  fill_name = NULL,
  fill_cutoff_name = NULL,
  size_by = NULL,
  fill_by = NULL,
  fill_cutoff = NULL,
  fill_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  seed = 8525,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DotPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_x">x</code></td>
<td>
<p>A character vector specifying the column to use for the x-axis.
A numeric column is expected.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_y">y</code></td>
<td>
<p>A character vector specifying the column to use for the y-axis.
A factor/character column is expected.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character vector to concatenate multiple columns in x. Default is &quot;_&quot;.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_y_sep">y_sep</code></td>
<td>
<p>A character vector to concatenate multiple columns in y. Default is &quot;_&quot;.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="DotPlot_+3A_size_name">size_name</code></td>
<td>
<p>A character vector specifying the name for the size legend.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_fill_name">fill_name</code></td>
<td>
<p>A character vector specifying the name for the fill legend.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_fill_cutoff_name">fill_cutoff_name</code></td>
<td>
<p>A character vector specifying the name for the fill cutoff legend.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background color to the plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character vector specifying the palette for the background color. Default is &quot;stripe&quot;.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character vector specifying the color for the background color.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value specifying the alpha for the background color. Default is 0.2.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_bg_direction">bg_direction</code></td>
<td>
<p>A character vector specifying the direction for the background color. Default is &quot;vertical&quot;.
Other options are &quot;horizontal&quot;. &quot;h&quot; and &quot;v&quot; are also accepted.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_size_by">size_by</code></td>
<td>
<p>Which column to use as the size of the dots. It must be a numeric column.
If not provided, the size will be the count of the instances for each 'y' in 'x'.
For 'ScatterPlot', it can be a single numeric value to specify the size of the dots.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_fill_by">fill_by</code></td>
<td>
<p>Which column to use as the fill the dots. It must be a numeric column.
If not provided, all dots will be filled with the same color at the middle of the palette.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_fill_cutoff">fill_cutoff</code></td>
<td>
<p>A numeric value specifying the cutoff for the fill column.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_fill_reverse">fill_reverse</code></td>
<td>
<p>A logical value indicating whether to reverse the fill direction. Default is FALSE.
By default, the fill direction is &quot;up&quot;. If TRUE, the fill direction is &quot;down&quot;.
When the direction is &quot;up&quot;, the values less than the cutoff will be filled with grey.
When the direction is &quot;down&quot;, the values greater than the cutoff will be filled with grey.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="DotPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="DotPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="DotPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>
<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mtcars &lt;- datasets::mtcars
mtcars$carb &lt;- factor(mtcars$carb)
mtcars$gear &lt;- factor(mtcars$gear)
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18)
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, add_bg = TRUE)
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, add_bg = TRUE,
        bg_direction = "h")
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, facet_by = "cyl")
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, facet_by = "cyl",
        facet_scales = "free_x")
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, split_by = "cyl")
DotPlot(mtcars, x = "carb", y = "gear", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, split_by = "cyl",
        palette = list("4" = "Set1", "6" = "Paired", "8" = "Reds"))
# works as a scatter plot
DotPlot(mtcars, x = "qsec", y = "drat", size_by = "wt",
        fill_by = "mpg", fill_cutoff = 18, fill_cutoff_name = "Small mpgs")
LollipopPlot(mtcars, x = "qsec", y = "drat", size_by = "wt",
             fill_by = "mpg")
LollipopPlot(mtcars, x = "qsec", y = "drat", size_by = "wt",
             fill_by = "mpg", fill_cutoff = 18, facet_by = "cyl",
             facet_scales = "free_y")
LollipopPlot(mtcars, x = "qsec", y = "drat", size_by = "wt",
             split_by = "vs", palette = list("0" = "Reds", "1" = "Blues"))
</code></pre>

<hr>
<h2 id='DotPlotAtomic'>Dot Plot without splitting the data</h2><span id='topic+DotPlotAtomic'></span>

<h3>Description</h3>

<p>Dot Plot without splitting the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DotPlotAtomic(
  data,
  x,
  y,
  x_sep = "_",
  y_sep = "_",
  flip = FALSE,
  lollipop = FALSE,
  size_by = NULL,
  fill_by = NULL,
  fill_cutoff = NULL,
  fill_reverse = FALSE,
  size_name = NULL,
  fill_name = NULL,
  fill_cutoff_name = NULL,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  bg_direction = c("vertical", "horizontal", "v", "h"),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DotPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character vector specifying the column to use for the x-axis.
Could be either numeric or factor/character. When multiple columns are provided, they will be concatenated with 'x_sep'.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character vector specifying the column to use for the y-axis.
Could be either numeric or factor/character. When multiple columns are provided, they will be concatenated with 'y_sep'.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character vector to concatenate multiple columns in x. Default is &quot;_&quot;.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_y_sep">y_sep</code></td>
<td>
<p>A character vector to concatenate multiple columns in y. Default is &quot;_&quot;.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_lollipop">lollipop</code></td>
<td>
<p>A logical value indicating whether to make it a lolipop plot. Default is FALSE.
When TRUE, 'x' should be a numeric column and 'y' should be a factor/character column.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_size_by">size_by</code></td>
<td>
<p>Which column to use as the size of the dots. It must be a numeric column.
If not provided, the size will be the count of the instances for each 'y' in 'x'.
For 'ScatterPlot', it can be a single numeric value to specify the size of the dots.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_fill_by">fill_by</code></td>
<td>
<p>Which column to use as the fill the dots. It must be a numeric column.
If not provided, all dots will be filled with the same color at the middle of the palette.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_fill_cutoff">fill_cutoff</code></td>
<td>
<p>A numeric value specifying the cutoff for the fill column.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_fill_reverse">fill_reverse</code></td>
<td>
<p>A logical value indicating whether to reverse the fill direction. Default is FALSE.
By default, the fill direction is &quot;up&quot;. If TRUE, the fill direction is &quot;down&quot;.
When the direction is &quot;up&quot;, the values less than the cutoff will be filled with grey.
When the direction is &quot;down&quot;, the values greater than the cutoff will be filled with grey.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_size_name">size_name</code></td>
<td>
<p>A character vector specifying the name for the size legend.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_fill_name">fill_name</code></td>
<td>
<p>A character vector specifying the name for the fill legend.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_fill_cutoff_name">fill_cutoff_name</code></td>
<td>
<p>A character vector specifying the name for the fill cutoff legend.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background color to the plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_bg_palette">bg_palette</code></td>
<td>
<p>A character vector specifying the palette for the background color. Default is &quot;stripe&quot;.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>A character vector specifying the color for the background color.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value specifying the alpha for the background color. Default is 0.2.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_bg_direction">bg_direction</code></td>
<td>
<p>A character vector specifying the direction for the background color. Default is &quot;vertical&quot;.
Other options are &quot;horizontal&quot;. &quot;h&quot; and &quot;v&quot; are also accepted.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="DotPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='element_textbox'>Theme element that add a box to the text</h2><span id='topic+element_textbox'></span><span id='topic+element_grob.element_textbox'></span>

<h3>Description</h3>

<p>Code grabbed from the <code>ggtext</code> package. See the original code at:
https://github.com/wilkelab/ggtext
This is used to create a text box around the text, primarily to be used in <code>CorPairsPlot</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>element_textbox(
  family = NULL,
  face = NULL,
  size = NULL,
  colour = NULL,
  fill = NULL,
  box.colour = NULL,
  linetype = NULL,
  linewidth = NULL,
  hjust = NULL,
  vjust = NULL,
  halign = NULL,
  valign = NULL,
  lineheight = NULL,
  margin = NULL,
  padding = NULL,
  width = NULL,
  height = NULL,
  minwidth = NULL,
  maxwidth = NULL,
  minheight = NULL,
  maxheight = NULL,
  r = NULL,
  orientation = NULL,
  color = NULL,
  box.color = NULL,
  debug = FALSE,
  inherit.blank = FALSE
)

## S3 method for class 'element_textbox'
element_grob(
  element,
  label = "",
  x = NULL,
  y = NULL,
  family = NULL,
  face = NULL,
  colour = NULL,
  size = NULL,
  hjust = NULL,
  vjust = NULL,
  lineheight = NULL,
  margin = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="element_textbox_+3A_family">family</code></td>
<td>
<p>Font family</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_face">face</code></td>
<td>
<p>Font face</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_size">size</code></td>
<td>
<p>Font size (in pt)</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_colour">colour</code>, <code id="element_textbox_+3A_color">color</code></td>
<td>
<p>Text color</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_fill">fill</code></td>
<td>
<p>Fill color of the enclosing box</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_box.colour">box.colour</code>, <code id="element_textbox_+3A_box.color">box.color</code></td>
<td>
<p>Line color of the enclosing box (if different from the text color)</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_linetype">linetype</code></td>
<td>
<p>Line type of the enclosing box (like <code>lty</code> in base R)</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_linewidth">linewidth</code></td>
<td>
<p>Line width of the enclosing box (measured in mm, just like <code>size</code> in
<code><a href="ggplot2.html#topic+element">ggplot2::element_line()</a></code>).</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_hjust">hjust</code></td>
<td>
<p>Horizontal justification</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_vjust">vjust</code></td>
<td>
<p>Vertical justification</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_halign">halign</code></td>
<td>
<p>Horizontal justification</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_valign">valign</code></td>
<td>
<p>Vertical justification</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_lineheight">lineheight</code></td>
<td>
<p>Line height, in multiples of the font size</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_padding">padding</code>, <code id="element_textbox_+3A_margin">margin</code></td>
<td>
<p>Padding and margins around the text box.
See <code><a href="gridtext.html#topic+textbox_grob">gridtext::textbox_grob()</a></code> for details.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_width">width</code>, <code id="element_textbox_+3A_height">height</code></td>
<td>
<p>Unit objects specifying the width and height
of the textbox, as in <code><a href="gridtext.html#topic+textbox_grob">gridtext::textbox_grob()</a></code>.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_minwidth">minwidth</code>, <code id="element_textbox_+3A_minheight">minheight</code>, <code id="element_textbox_+3A_maxwidth">maxwidth</code>, <code id="element_textbox_+3A_maxheight">maxheight</code></td>
<td>
<p>Min and max values for width and height.
Set to NULL to impose neither a minimum nor a maximum.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_r">r</code></td>
<td>
<p>Unit value specifying the corner radius of the box</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_orientation">orientation</code></td>
<td>
<p>Orientation of the text box. See <code><a href="gridtext.html#topic+textbox_grob">gridtext::textbox_grob()</a></code> for details.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_debug">debug</code></td>
<td>
<p>Not implemented.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_inherit.blank">inherit.blank</code></td>
<td>
<p>See <code><a href="ggplot2.html#topic+element">ggplot2::margin()</a></code> for details.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_element">element</code></td>
<td>
<p>A theme element created by <code><a href="#topic+element_textbox">element_textbox()</a></code>.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_label">label</code></td>
<td>
<p>Text to display in the textbox.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_x">x</code>, <code id="element_textbox_+3A_y">y</code></td>
<td>
<p>Position of the textbox.</p>
</td></tr>
<tr><td><code id="element_textbox_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="gridtext.html#topic+textbox_grob">gridtext::textbox_grob()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 theme element that can be used inside a <code><a href="ggplot2.html#topic+theme">ggplot2::theme()</a></code> call.
</p>

<hr>
<h2 id='enrich_example'>A example of clusterProfiler enrichment result</h2><span id='topic+enrich_example'></span>

<h3>Description</h3>

<p>A example of clusterProfiler enrichment result
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (interactive()) {
  data(geneList, package="DOSE")
  de &lt;- names(geneList)[abs(geneList) &gt; 1.5]
  enrich_example &lt;- clusterProfiler::enrichPathway(gene=de, pvalueCutoff = 0.05, readable=TRUE)
  enrich_example &lt;- as.data.frame(enrich_example)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='enrich_multidb_example'>A example of clusterProfiler enrichment result with multiple databases</h2><span id='topic+enrich_multidb_example'></span>

<h3>Description</h3>

<p>A example of clusterProfiler enrichment result with multiple databases
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (interactive()) {
  data(enrich_example, package="plotthis")
  enrich_example$Database &lt;- "DB1"
  enrich_example2 &lt;- enrich_example
  enrich_example2$Database &lt;- "DB2"
  enrich_example2$ID &lt;- paste0(enrich_example2$ID, "_DB2")
  enrich_example2$Description &lt;- paste0(enrich_example2$Description, " (DB2)")
  enrich_multidb_example &lt;- rbind(enrich_example, enrich_example2)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='EnrichMapAtomic'>Atomic Enrichment Map</h2><span id='topic+EnrichMapAtomic'></span>

<h3>Description</h3>

<p>Atomic Enrichment Map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EnrichMapAtomic(
  data,
  top_term = 100,
  metric = "p.adjust",
  layout = "fr",
  minchar = 2,
  cluster = "fast_greedy",
  show_keyword = FALSE,
  nlabel = 4,
  character_width = 50,
  words_excluded = plotthis::words_excluded,
  mark = "ellipse",
  label = c("term", "feature"),
  labelsize = 5,
  expand = c(0.4, 0.4),
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EnrichMapAtomic_+3A_data">data</code></td>
<td>
<p>A data frame containing the data to be plotted.
It should be in the format of clusterProfiler enrichment result,
which includes the columns: ID, Description, GeneRatio, BgRatio, pvalue, p.adjust,
qvalue, geneID and Count.
</p>

<ul>
<li><p> The <code>ID</code>, <code>qvalue</code>, <code>BgRatio</code>, and <code>Count</code> columns are optional.
</p>
</li>
<li><p> The <code>Description</code> is the description of the term.
</p>
</li>
<li><p> The <code>GeneRatio</code> is the number of genes in the term divided by the total number of genes in the input list.
</p>
</li>
<li><p> The <code>BgRatio</code> is the number of genes in the term divided by the total number of genes in the background list (all terms).
</p>
</li>
<li><p> The <code>Count</code> column, if given, should be the same as the first number in GeneRatio.
</p>
</li></ul>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_top_term">top_term</code></td>
<td>
<p>An integer specifying the number of top terms to show.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_metric">metric</code></td>
<td>
<p>A character string specifying the metric to use for the size of the nodes.
It is also used to order the terms when selected the top terms.
Either &quot;pvalue&quot; or &quot;p.adjust&quot;. The default is &quot;p.adjust&quot;.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_layout">layout</code></td>
<td>
<p>A character string specifying the layout of the graph.
Either &quot;circle&quot;, &quot;tree&quot;, &quot;grid&quot; or other layout functions in <code>igraph</code>.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_minchar">minchar</code></td>
<td>
<p>An integer specifying the minimum number of characters to show in the keyword.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_cluster">cluster</code></td>
<td>
<p>A character string specifying the clustering method.
Either &quot;fast_greedy&quot;, &quot;walktrap&quot;, &quot;edge_betweenness&quot;, &quot;infomap&quot; or other clustering functions in <code>igraph</code>.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_show_keyword">show_keyword</code></td>
<td>
<p>A logical value specifying whether to show the keyword instead of Description/Term in the plot.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_nlabel">nlabel</code></td>
<td>
<p>An integer specifying the number of labels to show in each cluster.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_character_width">character_width</code></td>
<td>
<p>The width of the characters used to wrap the keyword.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_words_excluded">words_excluded</code></td>
<td>
<p>A character vector specifying the words to exclude in the keyword.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_mark">mark</code></td>
<td>
<p>A character string specifying the mark to use for the nodes.
Either &quot;ellipse&quot;, &quot;rect&quot;, &quot;circle&quot;, &quot;text&quot; or other mark functions in <code>ggforce</code>.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_label">label</code></td>
<td>
<p>A character string specifying the label to show in the legend.
Either &quot;term&quot; or &quot;feature&quot;. The default is &quot;term&quot;.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_labelsize">labelsize</code></td>
<td>
<p>A numeric value specifying the size of the label.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_expand">expand</code></td>
<td>
<p>A numeric vector of length 2 specifying the expansion of the x and y axis.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="EnrichMapAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='EnrichNetworkAtomic'>Atomic Enrichment Network</h2><span id='topic+EnrichNetworkAtomic'></span>

<h3>Description</h3>

<p>Atomic Enrichment Network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EnrichNetworkAtomic(
  data,
  top_term = 6,
  metric = "p.adjust",
  character_width = 50,
  layout = "fr",
  layoutadjust = TRUE,
  adjscale = 60,
  adjiter = 100,
  blendmode = "blend",
  labelsize = 5,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EnrichNetworkAtomic_+3A_data">data</code></td>
<td>
<p>A data frame containing the data to be plotted.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_layoutadjust">layoutadjust</code></td>
<td>
<p>A logical value specifying whether to adjust the layout of the network.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_adjscale">adjscale</code></td>
<td>
<p>A numeric value specifying the scale of the adjustment.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_adjiter">adjiter</code></td>
<td>
<p>A numeric value specifying the number of iterations for the adjustment.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_blendmode">blendmode</code></td>
<td>
<p>A character string specifying the blend mode of the colors.
Either &quot;blend&quot;, &quot;average&quot;, &quot;multiply&quot; and &quot;screen&quot;.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="EnrichNetworkAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='facet_plot'>Facetting a plot</h2><span id='topic+facet_plot'></span>

<h3>Description</h3>

<p>Facetting a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_plot(
  plot,
  facet_by,
  facet_scales,
  nrow,
  ncol,
  byrow,
  legend.position = "right",
  legend.direction = "vertical",
  recalc_size = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_plot_+3A_plot">plot</code></td>
<td>
<p>The plot to facet or a list list(plot, height, width) if guess_size is TRUE</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_facet_by">facet_by</code></td>
<td>
<p>The column(s) to split data by and plot separately or facet by
If NULL, no faceting will be done</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets.</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_nrow">nrow</code></td>
<td>
<p>The number of rows in facet_wrap</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_ncol">ncol</code></td>
<td>
<p>The number of columns in facet_wrap</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_byrow">byrow</code></td>
<td>
<p>Whether to fill the plots by row</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_legend.position">legend.position</code></td>
<td>
<p>The position of the legend</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>The direction of the legend</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_recalc_size">recalc_size</code></td>
<td>
<p>Whether to re-calculate the size of the plot</p>
</td></tr>
<tr><td><code id="facet_plot_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to facet_wrap or facet_grid</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The faceted plot. If guess_size is TRUE, attr(p, &quot;height&quot;) and attr(p, &quot;width&quot;) will be set
</p>

<hr>
<h2 id='get_cutoffs_data'>Prepare the cutoff data for the ROC curve</h2><span id='topic+get_cutoffs_data'></span>

<h3>Description</h3>

<p>Prepare the cutoff data for the ROC curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cutoffs_data(
  data,
  truth_by,
  score_by,
  cat_by,
  cutoffs_at = NULL,
  cutoffs_labels = NULL,
  cutoffs_accuracy = 0.01,
  n_cuts = 0,
  increasing = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_cutoffs_data_+3A_data">data</code></td>
<td>
<p>A data frame with the truth and score columns.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_truth_by">truth_by</code></td>
<td>
<p>A character string of the column name that contains the true class labels.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_score_by">score_by</code></td>
<td>
<p>A character string of the column name that contains the predicted scores.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_cat_by">cat_by</code></td>
<td>
<p>A character string of the column name to categorize/group the data.
If specified, the cutoffs will be calculated for each category.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_cutoffs_at">cutoffs_at</code></td>
<td>
<p>Vector of user supplied cutoffs to plot as points. If non-NULL,
it will override the values of n_cuts and plot the observed cutoffs closest to the user-supplied ones.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_cutoffs_labels">cutoffs_labels</code></td>
<td>
<p>vector of user-supplied labels for the cutoffs. Must be a character vector of the
same length as cutoffs_at.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_n_cuts">n_cuts</code></td>
<td>
<p>An integer to specify the number of cuts on the ROC curve.</p>
</td></tr>
<tr><td><code id="get_cutoffs_data_+3A_increasing">increasing</code></td>
<td>
<p>TRUE if the score is increasing with the truth (1), FALSE otherwise.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the cutoffs and the corresponding x and y values.
</p>

<hr>
<h2 id='gggrob'>Get the grid.draw-able ggplot grob
The output from ggplotGrob can not be directly used in grid.draw, the position can not be set.
This function extracts the gTree from the ggplot grob.</h2><span id='topic+gggrob'></span>

<h3>Description</h3>

<p>Get the grid.draw-able ggplot grob
The output from ggplotGrob can not be directly used in grid.draw, the position can not be set.
This function extracts the gTree from the ggplot grob.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gggrob(p, void = TRUE, nolegend = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gggrob_+3A_p">p</code></td>
<td>
<p>A ggplot object</p>
</td></tr>
<tr><td><code id="gggrob_+3A_void">void</code></td>
<td>
<p>If TRUE, the theme_void will be added to the ggplot object</p>
</td></tr>
<tr><td><code id="gggrob_+3A_nolegend">nolegend</code></td>
<td>
<p>If TRUE, the legend will be removed from the ggplot object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gTree object
</p>

<hr>
<h2 id='gsea_example'>A example of GSEA result from fgsea package</h2><span id='topic+gsea_example'></span>

<h3>Description</h3>

<p>A example of GSEA result from fgsea package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (interactive()) {
 set.seed(1234)
 data(geneList, package="DOSE")
 gsea_example &lt;- DOSE::gseDO(geneList)
 gene_ranks &lt;- gsea_example@geneList
 gene_sets &lt;- gsea_example@geneSets
 gsea_example_pos &lt;- gsea_example[gsea_example$p.adjust &lt; 0.05 &amp; gsea_example$NES &gt; 0, ]
 gsea_example_neg &lt;- gsea_example[gsea_example$p.adjust &lt; 0.05 &amp; gsea_example$NES &lt; 0, ]
 gsea_example &lt;- rbind(
     gsea_example_pos[sample(1:nrow(gsea_example_pos), 5), ],
     gsea_example_pos[sample(1:nrow(gsea_example_pos), 5), ]
 )

 attr(gsea_example, "gene_ranks") &lt;- gene_ranks
 attr(gsea_example, "gene_sets") &lt;- gene_sets[gsea_example$ID]
}

## End(Not run)
</code></pre>

<hr>
<h2 id='gsea_running_score'>Get the running enrichment score of a gene set</h2><span id='topic+gsea_running_score'></span>

<h3>Description</h3>

<p>Get the running enrichment score of a gene set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsea_running_score(genes, gene_ranks, exponent = 1, hits_only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsea_running_score_+3A_genes">genes</code></td>
<td>
<p>A vector of genes</p>
</td></tr>
<tr><td><code id="gsea_running_score_+3A_gene_ranks">gene_ranks</code></td>
<td>
<p>A numeric vector of gene ranks with names</p>
</td></tr>
<tr><td><code id="gsea_running_score_+3A_exponent">exponent</code></td>
<td>
<p>A numeric value to raise the gene ranks to</p>
</td></tr>
<tr><td><code id="gsea_running_score_+3A_hits_only">hits_only</code></td>
<td>
<p>A logical value to return only the running enrichment score of the hits</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of the running enrichment score
</p>

<hr>
<h2 id='GSEAPlotAtomic'>GSEA plot for a single term</h2><span id='topic+GSEAPlotAtomic'></span>

<h3>Description</h3>

<p>GSEA plot for a single term
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSEAPlotAtomic(
  data,
  gene_ranks = "@gene_ranks",
  gs,
  genes,
  metric = "p.adjust",
  sample_coregenes = FALSE,
  line_width = 1.5,
  line_alpha = 1,
  line_color = "#6BB82D",
  n_coregenes = 10,
  genes_label = NULL,
  label_fg = "black",
  label_bg = "white",
  label_bg_r = 0.1,
  label_size = 4,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GSEAPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_gene_ranks">gene_ranks</code></td>
<td>
<p>A numeric vector of gene ranks with genes as names
The gene ranks are used to plot the gene sets.
If <code>gene_ranks</code> is a character vector starting with <code>@</code>, the gene ranks will be taken from the attribute of <code>data</code>.</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_gs">gs</code></td>
<td>
<p>The name of the gene set</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_genes">genes</code></td>
<td>
<p>The genes in the gene set</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_metric">metric</code></td>
<td>
<p>The metric to show in the subtitle</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_sample_coregenes">sample_coregenes</code></td>
<td>
<p>A logical value to sample the core genes from the core_enrichment; if <code>FALSE</code>, the first <code>n_coregenes</code> will be used</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_line_width">line_width</code></td>
<td>
<p>The width of the line in the running score plot</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_line_alpha">line_alpha</code></td>
<td>
<p>The alpha of the line in the running score plot</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_line_color">line_color</code></td>
<td>
<p>The color of the line in the running score plot</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_n_coregenes">n_coregenes</code></td>
<td>
<p>The number of core genes to label</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_genes_label">genes_label</code></td>
<td>
<p>The genes to label. If set, <code>n_coregenes</code> will be ignored</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>The color of the label text</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>The background color of the label</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>The radius of the background color of the label</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_label_size">label_size</code></td>
<td>
<p>The size of the label text</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>The label of the y-axis, will be shown on the right side</p>
</td></tr>
<tr><td><code id="GSEAPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='Heatmap'>Heatmap</h2><span id='topic+Heatmap'></span>

<h3>Description</h3>

<p>Heatmap is a popular way to visualize data in matrix format. It is widely used in biology to visualize gene expression data in microarray and RNA-seq data. The heatmap is a matrix where rows represent the samples and columns represent the features. The color of each cell represents the value of the feature in the sample. The color can be continuous or discrete. The heatmap can be split by the columns or rows to show the subgroups in the data. The heatmap can also be annotated by the columns or rows to show the additional information of the samples or features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Heatmap(
  data,
  rows,
  columns_by,
  rows_name = "rows",
  columns_name = "columns",
  split_by = NULL,
  split_by_sep = "_",
  split_rows_data = FALSE,
  name = "value",
  border = TRUE,
  rows_palette = "Paired",
  rows_palcolor = NULL,
  title = NULL,
  pie_group_by = NULL,
  pie_group_by_sep = "_",
  pie_palette = "Spectral",
  pie_palcolor = NULL,
  pie_size = NULL,
  pie_name = NULL,
  pie_size_name = "size",
  pie_values = "count",
  legend_items = NULL,
  legend_discrete = FALSE,
  lower_quantile = 0,
  upper_quantile = 0.99,
  lower_cutoff = NULL,
  upper_cutoff = NULL,
  columns_by_sep = "_",
  columns_split_by = NULL,
  columns_split_name = NULL,
  columns_palette = "Paired",
  columns_palcolor = NULL,
  columns_split_by_sep = "_",
  columns_split_palette = "simspec",
  columns_split_palcolor = NULL,
  rows_data = NULL,
  rows_split_by = NULL,
  rows_split_by_sep = "_",
  rows_split_palette = "simspec",
  rows_split_palcolor = NULL,
  column_name_annotation = TRUE,
  column_name_legend = isFALSE(show_column_names) &amp;&amp; !identical(legend.position, "none"),
  row_name_annotation = TRUE,
  row_name_legend = isFALSE(show_row_names) &amp;&amp; !identical(legend.position, "none"),
  cluster_columns = TRUE,
  cluster_rows = TRUE,
  show_row_names = !row_name_annotation,
  show_column_names = !column_name_annotation,
  column_title = character(0),
  row_title = character(0),
  na_col = "grey85",
  row_names_side = "right",
  column_names_side = "bottom",
  bars_sample = 100,
  flip = FALSE,
  label_size = 10,
  label_cutoff = NULL,
  label_accuracy = 0.01,
  layer_fun_callback = NULL,
  cell_type = c("tile", "bars", "label", "dot", "violin", "boxplot", "pie"),
  cell_agg = mean,
  add_bg = FALSE,
  bg_alpha = 0.5,
  violin_fill = NULL,
  boxplot_fill = NULL,
  dot_size = 8,
  dot_size_name = "size",
  column_annotation = NULL,
  column_annotation_side = "top",
  column_annotation_palette = "Paired",
  column_annotation_palcolor = NULL,
  column_annotation_type = "auto",
  column_annotation_params = list(),
  column_annotation_agg = NULL,
  row_annotation = NULL,
  row_annotation_side = "left",
  row_annotation_palette = "Paired",
  row_annotation_palcolor = NULL,
  row_annotation_type = "auto",
  row_annotation_params = list(),
  row_annotation_agg = NULL,
  add_reticle = FALSE,
  reticle_color = "grey",
  palette = "RdBu",
  palcolor = NULL,
  alpha = 1,
  legend.position = "right",
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Heatmap_+3A_data">data</code></td>
<td>
<p>A data frame used to create the heatmap.
The data should be in a long form where each row represents a instance in the heatmap.
The <code>rows</code> should be multiple columns if you want to plot as rows, which you can refer as &quot;features&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows">rows</code></td>
<td>
<p>A character string/vector of the column name(s) to plot for the rows
Multiple columns in the data frame can be used as the rows.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_by">columns_by</code></td>
<td>
<p>A character string of the column name to plot for the columns
A character/factor column is expected.
Multiple columns are also supported. If so, <code>NA</code>s can be used to exclude rows in that group.
If there is a single value in each group (other than <code>NA</code>), the group name, rather than the value,
will be shown in the column group annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_name">rows_name</code></td>
<td>
<p>A character string specifying the name of rows, which will be shown in the
row group annotation and the legend of it.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_name">columns_name</code></td>
<td>
<p>A character string specifying the name of columns, which will be shown in the
column group annotation and the legend of it.
Only used when <code>columns_by</code> has multiple columns.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="Heatmap_+3A_split_rows_data">split_rows_data</code></td>
<td>
<p>A logical value indicating whether to split the rows data as well using 'split_by' and 'split_by_sep'.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_name">name</code></td>
<td>
<p>A character string specifying the name of the main legend of the heatmap</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_border">border</code></td>
<td>
<p>A logical value indicating whether to draw the border of the heatmap.
If TRUE, the borders of the slices will be also drawn.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_palette">rows_palette</code></td>
<td>
<p>A character string specifying the palette of the row group annotation.
The default is &quot;Paired&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_palcolor">rows_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the row group annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_group_by">pie_group_by</code></td>
<td>
<p>A character string of the column name to group the data for the pie chart.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_group_by_sep">pie_group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>pie_group_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_palette">pie_palette</code></td>
<td>
<p>A character string specifying the palette of the pie chart.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_palcolor">pie_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the pie chart.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_size">pie_size</code></td>
<td>
<p>A numeric value or a function specifying the size of the pie chart.
If it is a function, the function should take <code>count</code> as the argument and return the size.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_name">pie_name</code></td>
<td>
<p>A character string specifying the name of the legend for the pie chart.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_size_name">pie_size_name</code></td>
<td>
<p>A character string specifying the name of the legend for the pie size.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_pie_values">pie_values</code></td>
<td>
<p>How to calculate the values for the pie chart. Default is &quot;count&quot;.
Other options are &quot;value&quot; and a function.
</p>

<ul>
<li><p> &quot;count&quot;: Count the number of instances in <code>pie_group_by</code>.
</p>
</li>
<li><p> &quot;value&quot;: Use the values in the columns specified by <code>rows</code> as the values.
If there are multiple values for a single group in <code>pie_group_by</code>, a function should be provided to aggregate the values.
If not, the sum of the values will be used and a warning will be shown.
</p>
</li>
<li><p> A function: The function should take a vector of values as the argument and return a single value, for each
group in <code>pie_group_by</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="Heatmap_+3A_legend_items">legend_items</code></td>
<td>
<p>A numeric vector with names to specifiy the items in the main legend.
The names will be working as the labels of the legend items.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_legend_discrete">legend_discrete</code></td>
<td>
<p>A logical value indicating whether the main legend is discrete.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_lower_quantile">lower_quantile</code>, <code id="Heatmap_+3A_upper_quantile">upper_quantile</code>, <code id="Heatmap_+3A_lower_cutoff">lower_cutoff</code>, <code id="Heatmap_+3A_upper_cutoff">upper_cutoff</code></td>
<td>
<p>Vector of minimum and maximum cutoff values or quantile values for each feature.
It's applied to aggregated values when aggregated values are used (e.g. plot_type tile, label, etc).
It's applied to raw values when raw values are used (e.g. plot_type bars, etc).</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_by_sep">columns_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>columns_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_split_by">columns_split_by</code></td>
<td>
<p>A character string of the column name to split the heatmap columns into slices.
A character/factor column or multiple columns are expected.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_split_name">columns_split_name</code></td>
<td>
<p>A character string specifying the name of the column split annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_palette">columns_palette</code></td>
<td>
<p>A character string specifying the palette of the column group annotation.
The default is &quot;Paired&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_palcolor">columns_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the column group annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_split_by_sep">columns_split_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>columns_split_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_split_palette">columns_split_palette</code></td>
<td>
<p>A character string specifying the palette of the column split annotation.
The default is &quot;simspec&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_columns_split_palcolor">columns_split_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the column split annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_data">rows_data</code></td>
<td>
<p>A character string of the column name to use as the data for the row group annotation.
If it starts with &quot;@&quot;, it will be treated as an attribute of the data.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_split_by">rows_split_by</code></td>
<td>
<p>A character string of the column name to split the heatmap rows into slices.
A character/factor column or multiple columns are expected.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_split_by_sep">rows_split_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>rows_split_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_split_palette">rows_split_palette</code></td>
<td>
<p>A character string specifying the palette of the row split annotation.
The default is &quot;simspec&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_rows_split_palcolor">rows_split_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the row split annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_name_annotation">column_name_annotation</code></td>
<td>
<p>A logical value indicating whether to add the column annotation for the column names.
which is a simple annotaion indicating the column names.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_name_legend">column_name_legend</code></td>
<td>
<p>A logical value indicating whether to show the legend of the column name annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_name_annotation">row_name_annotation</code></td>
<td>
<p>A logical value indicating whether to add the row annotation for the row names.
which is a simple annotaion indicating the row names.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_name_legend">row_name_legend</code></td>
<td>
<p>A logical value indicating whether to show the legend of the row name annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_cluster_columns">cluster_columns</code></td>
<td>
<p>A logical value indicating whether to cluster the columns.
If TRUE and columns_split_by is provided, the clustering will only be applied to the columns within the same split.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>A logical value indicating whether to cluster the rows.
If TRUE and rows_split_by is provided, the clustering will only be applied to the rows within the same split.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_show_row_names">show_row_names</code></td>
<td>
<p>A logical value indicating whether to show the row names.
If TRUE, the legend of the row group annotation will be hidden.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_show_column_names">show_column_names</code></td>
<td>
<p>A logical value indicating whether to show the column names.
If TRUE, the legend of the column group annotation will be hidden.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_title">column_title</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the title of the column group annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_title">row_title</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the title of the row group annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_na_col">na_col</code></td>
<td>
<p>A character string specifying the color for missing values.
The default is &quot;grey85&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_names_side">row_names_side</code></td>
<td>
<p>A character string specifying the side of the row names.
The default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_names_side">column_names_side</code></td>
<td>
<p>A character string specifying the side of the column names.
The default is &quot;bottom&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_bars_sample">bars_sample</code></td>
<td>
<p>An integer specifying the number of samples to draw the bars.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the heatmap.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the labels when <code>cell_type = "label"</code>.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_label_cutoff">label_cutoff</code></td>
<td>
<p>A numeric value specifying the cutoff to show the labels when <code>cell_type = "label"</code>.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_label_accuracy">label_accuracy</code></td>
<td>
<p>A numeric value specifying the accuracy of the labels when <code>cell_type = "label"</code>.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_layer_fun_callback">layer_fun_callback</code></td>
<td>
<p>A function to add additional layers to the heatmap.
The function should have the following arguments: <code>j</code>, <code>i</code>, <code>x</code>, <code>y</code>, <code>w</code>, <code>h</code>, <code>fill</code>, <code>sr</code> and <code>sc</code>.
Please also refer to the <code>layer_fun</code> argument in <code>ComplexHeatmap::Heatmap</code>.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_cell_type">cell_type</code></td>
<td>
<p>A character string specifying the type of the heatmap cells.
The default is values. Other options are &quot;bars&quot;, &quot;label&quot;, &quot;dot&quot;, &quot;violin&quot;, &quot;boxplot&quot;.
Note that for pie chart, the values under columns specified by <code>rows</code> will not be used directly. Instead, the values
will just be counted in different <code>pie_group_by</code> groups. <code>NA</code> values will not be counted.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_cell_agg">cell_agg</code></td>
<td>
<p>A function to aggregate the values in the cell, for the cell type &quot;tile&quot; and &quot;label&quot;.
The default is <code>mean</code>.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the heatmap.
Does not work with <code>cell_type = "bars"</code> or <code>cell_type = "tile"</code>.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value between 0 and 1 specifying the transparency of the background.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_violin_fill">violin_fill</code></td>
<td>
<p>A character vector of colors to override the fill color of the violin plot.
If NULL, the fill color will be the same as the annotion.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_boxplot_fill">boxplot_fill</code></td>
<td>
<p>A character vector of colors to override the fill color of the boxplot.
If NULL, the fill color will be the same as the annotion.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_dot_size">dot_size</code></td>
<td>
<p>A numeric value specifying the size of the dot or a function to calculate the size
from the values in the cell.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_dot_size_name">dot_size_name</code></td>
<td>
<p>A character string specifying the name of the legend for the dot size.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation">column_annotation</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the column annotation.
Or a list with the keys as the names of the annotation and the values as the column names.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation_side">column_annotation_side</code></td>
<td>
<p>A character string specifying the side of the column annotation.
Could be a list with the keys as the names of the annotation and the values as the sides.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation_palette">column_annotation_palette</code></td>
<td>
<p>A character string specifying the palette of the column annotation.
The default is &quot;Paired&quot;.
Could be a list with the keys as the names of the annotation and the values as the palettes.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation_palcolor">column_annotation_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the column annotation.
Could be a list with the keys as the names of the annotation and the values as the palcolors.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation_type">column_annotation_type</code></td>
<td>
<p>A character string specifying the type of the column annotation.
The default is &quot;auto&quot;. Other options are &quot;simple&quot;, &quot;pie&quot;, &quot;ring&quot;, &quot;bar&quot;, &quot;violin&quot;, &quot;boxplot&quot;, &quot;density&quot;.
Could be a list with the keys as the names of the annotation and the values as the types.
If the type is &quot;auto&quot;, the type will be determined by the type and number of the column data.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation_params">column_annotation_params</code></td>
<td>
<p>A list of parameters passed to the annotation function.
Could be a list with the keys as the names of the annotation and the values as the parameters.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_column_annotation_agg">column_annotation_agg</code></td>
<td>
<p>A function to aggregate the values in the column annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation">row_annotation</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the row annotation.
Or a list with the keys as the names of the annotation and the values as the column names.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation_side">row_annotation_side</code></td>
<td>
<p>A character string specifying the side of the row annotation.
Could be a list with the keys as the names of the annotation and the values as the sides.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation_palette">row_annotation_palette</code></td>
<td>
<p>A character string specifying the palette of the row annotation.
The default is &quot;Paired&quot;.
Could be a list with the keys as the names of the annotation and the values as the palettes.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation_palcolor">row_annotation_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the row annotation.
Could be a list with the keys as the names of the annotation and the values as the palcolors.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation_type">row_annotation_type</code></td>
<td>
<p>A character string specifying the type of the row annotation.
The default is &quot;auto&quot;. Other options are &quot;simple&quot;, &quot;pie&quot;, &quot;ring&quot;, &quot;bar&quot;, &quot;violin&quot;, &quot;boxplot&quot;, &quot;density&quot;.
Could be a list with the keys as the names of the annotation and the values as the types.
If the type is &quot;auto&quot;, the type will be determined by the type and number of the row data.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation_params">row_annotation_params</code></td>
<td>
<p>A list of parameters passed to the annotation function.
Could be a list with the keys as the names of the annotation and the values as the parameters.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_row_annotation_agg">row_annotation_agg</code></td>
<td>
<p>A function to aggregate the values in the row annotation.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_add_reticle">add_reticle</code></td>
<td>
<p>A logical value indicating whether to add a reticle to the heatmap.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_reticle_color">reticle_color</code></td>
<td>
<p>A character string specifying the color of the reticle.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The heatmap(s).
When <code>split_by</code> is not provided, the function returns a single heatmap (<a href="ComplexHeatmap.html#topic+Heatmap">ComplexHeatmap::Heatmap</a>-class object).
When <code>split_by</code> is provided, the function returns a combined plot of multiple heatmaps wrapped by <a href="patchwork.html#topic+wrap_plots">patchwork::wrap_plots</a>
if <code>combine = TRUE</code>, otherwise, it returns a list of heatmaps.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+anno_simple">anno_simple</a></code>, <code><a href="#topic+anno_points">anno_points</a></code>, <code><a href="#topic+anno_lines">anno_lines</a></code>, <code><a href="#topic+anno_pie">anno_pie</a></code>, <code><a href="#topic+anno_violin">anno_violin</a></code>, <code><a href="#topic+anno_boxplot">anno_boxplot</a></code>, <code><a href="#topic+anno_density">anno_density</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
data &lt;- data.frame(
   F1 = rnorm(100, 0.1),
   F2 = rnorm(100, 0.2),
   F3 = rnorm(100),
   F4 = rnorm(100, 0.3),
   F5 = rnorm(100, -0.1),
   F6 = rnorm(100, -0.2),
   c = sample(letters[1:8], 100, replace = TRUE),
   s = sample(LETTERS[1:2], 100, replace = TRUE),
   p = sample(c("X", "Y", "Z"), 100, replace = TRUE),
   a = sample(1:5, 100, replace = TRUE),
   p1 = c(sample(c(1, NA), 100, replace = TRUE)),
   p2 = c(sample(c(1, NA), 100, replace = TRUE)),
   p3 = c(sample(c(1, NA), 100, replace = TRUE))
)
rows &lt;- c("F1", "F2", "F3", "F4", "F5", "F6")
rows_data &lt;- data.frame(
   rows = rep(c("F1", "F2", "F3", "F4", "F5", "F6"), each = 10),
   rows1 = rep(c("F1", "F2", "F3", "F4", "F5", "F6"), each = 10),
   rs = rep(letters[1:2], each = 30),
   rp = sample(c("X", "Y", "Z"), 60, replace = TRUE),
   rv = rnorm(60, 0.5)
)

Heatmap(data, rows = rows, columns_by = "c")
Heatmap(data, rows = list(RG1 = c("F1", "F2", "F3"), RG2 = c("F4", "F5", "F6")),
   columns_by = "c")
# Multiple columns_by, each as a split
Heatmap(data, rows = rows, columns_by = c("c", "s"), columns_by_sep = "/")
Heatmap(data, rows = rows, columns_by = c("p1", "p2", "p3"))
Heatmap(data, rows = rows, columns_by = "c", split_by = "s")
Heatmap(data, rows = rows, columns_by = "c", columns_split_by = "s")
Heatmap(data, rows = rows, columns_by = "c", columns_split_by = "s",
        rows_data = rows_data, rows_split_by = "rs")
Heatmap(data, rows = rows, columns_by = "c", columns_split_by = "s",
        rows_data = rows_data, rows_split_by = "rs", flip = TRUE)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "bars")
Heatmap(data, rows = rows, columns_by = "c", cell_type = "bars",
        bars_sample = 3)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "label")
Heatmap(data, rows = rows, columns_by = "c", cell_type = "label",
        label_cutoff = 0)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "dot")
Heatmap(data, rows = rows, columns_by = "c", cell_type = "dot",
        dot_size = mean)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "dot",
        dot_size = mean, row_name_annotation = FALSE, column_name_annotation = FALSE,
        row_names_side = "left", cluster_rows = FALSE, cluster_columns = FALSE)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "dot",
        dot_size = mean, add_bg = TRUE)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "dot",
        dot_size = mean, add_reticle = TRUE)
Heatmap(data, cluster_rows = FALSE, cluster_columns = FALSE, columns_by = "p",
   rows = c("p1", "p2", "p3"), name = "Category", pie_group_by = "c",
   cell_type = "pie")
Heatmap(data, cluster_rows = FALSE, cluster_columns = FALSE, columns_by = "p",
   rows = c("p1", "p2", "p3"), name = "Category", pie_group_by = "c",
   cell_type = "pie", pie_size = sqrt, add_bg = TRUE)
Heatmap(data, rows = rows, columns_by = "c", cell_type = "violin")
Heatmap(data, rows = rows, columns_by = "c", cell_type = "boxplot")
Heatmap(data, rows = rows, columns_by = "c", rows_data = rows_data,
  column_annotation = list(p1 = "p", p2 = "p", F1 = "F1"),
  column_annotation_type = list(p1 = "ring", p2 = "bar", F1 = "violin"),
  column_annotation_params = list(
     p1 = list(height = grid::unit(10, "mm"), show_legend = FALSE),
     F1 = list(height = grid::unit(18, "mm"))),
  rows_split_by = "rs",
  row_annotation = c("rp", "rv", "rows1"),
  row_annotation_side = "right",
  row_annotation_type = list(rp = "pie", rv = "density", rows1 = "simple"),
  row_annotation_params = list(rp = list(width = grid::unit(12, "mm"))),
  show_row_names = TRUE, show_column_names = TRUE)
Heatmap(data, rows = rows, columns_by = "c", rows_data = rows_data,
  column_annotation = list(p1 = "p", p2 = "p", F1 = "F1"),
  column_annotation_type = list(p1 = "ring", p2 = "bar", F1 = "violin"),
  column_annotation_params = list(
     p1 = list(height = grid::unit(10, "mm"), show_legend = FALSE),
     F1 = list(height = grid::unit(18, "mm"))),
  rows_split_by = "rs",
  row_annotation = c("rp", "rv", "rows1"),
  row_annotation_side = "right",
  row_annotation_type = list(rp = "pie", rv = "density", rows1 = "simple"),
  row_annotation_params = list(rp = list(width = grid::unit(12, "mm"))),
  show_row_names = TRUE, show_column_names = TRUE, flip = TRUE)
Heatmap(data, rows = rows, columns_by = "c", split_by = "p",
        palette = list(X = "Reds", Y = "Blues", Z = "Purp"))

</code></pre>

<hr>
<h2 id='HeatmapAtomic'>Atomic heatmap</h2><span id='topic+HeatmapAtomic'></span>

<h3>Description</h3>

<p>Atomic heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HeatmapAtomic(
  data,
  rows,
  columns_by,
  rows_name = "rows",
  rows_split_name = "rows_split",
  columns_name = "columns",
  name = "value",
  border = TRUE,
  rows_palette = "Paired",
  rows_palcolor = NULL,
  pie_group_by = NULL,
  pie_group_by_sep = "_",
  pie_palette = "Spectral",
  pie_palcolor = NULL,
  pie_size = NULL,
  pie_name = NULL,
  pie_size_name = "size",
  pie_values = "count",
  columns_by_sep = "_",
  columns_split_by = NULL,
  columns_split_name = NULL,
  columns_palette = "Paired",
  columns_palcolor = NULL,
  columns_split_by_sep = "_",
  columns_split_palette = "simspec",
  columns_split_palcolor = NULL,
  lower_quantile = 0,
  upper_quantile = 0.99,
  lower_cutoff = NULL,
  upper_cutoff = NULL,
  rows_data = NULL,
  rows_split_by = NULL,
  rows_split_by_sep = "_",
  rows_split_palette = "simspec",
  rows_split_palcolor = NULL,
  column_name_annotation = TRUE,
  column_name_legend = isFALSE(show_column_names) &amp;&amp; !identical(legend.position, "none"),
  row_name_annotation = TRUE,
  row_name_legend = isFALSE(show_row_names) &amp;&amp; !identical(legend.position, "none"),
  cluster_columns = TRUE,
  cluster_rows = TRUE,
  legend_items = NULL,
  legend_discrete = FALSE,
  show_row_names = !row_name_annotation,
  show_column_names = !column_name_annotation,
  column_title = character(0),
  row_title = character(0),
  na_col = "grey85",
  title = NULL,
  row_names_side = "right",
  column_names_side = "bottom",
  bars_sample = 100,
  flip = FALSE,
  label_size = 10,
  label_cutoff = NULL,
  label_accuracy = 0.01,
  layer_fun_callback = NULL,
  cell_type = c("tile", "bars", "label", "dot", "violin", "boxplot", "pie"),
  cell_agg = mean,
  add_bg = FALSE,
  bg_alpha = 0.5,
  violin_fill = NULL,
  boxplot_fill = NULL,
  dot_size = 8,
  dot_size_name = "size",
  column_annotation = NULL,
  column_annotation_side = "top",
  column_annotation_palette = "Paired",
  column_annotation_palcolor = NULL,
  column_annotation_type = "auto",
  column_annotation_params = list(),
  column_annotation_agg = NULL,
  row_annotation = NULL,
  row_annotation_side = "left",
  row_annotation_palette = "Paired",
  row_annotation_palcolor = NULL,
  row_annotation_type = "auto",
  row_annotation_params = list(),
  row_annotation_agg = NULL,
  add_reticle = FALSE,
  reticle_color = "grey",
  return_grob = FALSE,
  palette = "RdBu",
  palcolor = NULL,
  alpha = 1,
  legend.position = "right",
  legend.direction = "vertical",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HeatmapAtomic_+3A_data">data</code></td>
<td>
<p>A data frame used to create the heatmap.
The data should be in a long form where each row represents a instance in the heatmap.
The <code>rows</code> should be multiple columns if you want to plot as rows, which you can refer as &quot;features&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows">rows</code></td>
<td>
<p>A character string/vector of the column name(s) to plot for the rows
Multiple columns in the data frame can be used as the rows.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_by">columns_by</code></td>
<td>
<p>A character string of the column name to plot for the columns
A character/factor column is expected.
Multiple columns are also supported. If so, <code>NA</code>s can be used to exclude rows in that group.
If there is a single value in each group (other than <code>NA</code>), the group name, rather than the value,
will be shown in the column group annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_name">rows_name</code></td>
<td>
<p>A character string specifying the name of rows, which will be shown in the
row group annotation and the legend of it.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_name">columns_name</code></td>
<td>
<p>A character string specifying the name of columns, which will be shown in the
column group annotation and the legend of it.
Only used when <code>columns_by</code> has multiple columns.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_name">name</code></td>
<td>
<p>A character string specifying the name of the main legend of the heatmap</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_border">border</code></td>
<td>
<p>A logical value indicating whether to draw the border of the heatmap.
If TRUE, the borders of the slices will be also drawn.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_palette">rows_palette</code></td>
<td>
<p>A character string specifying the palette of the row group annotation.
The default is &quot;Paired&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_palcolor">rows_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the row group annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_group_by">pie_group_by</code></td>
<td>
<p>A character string of the column name to group the data for the pie chart.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_group_by_sep">pie_group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>pie_group_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_palette">pie_palette</code></td>
<td>
<p>A character string specifying the palette of the pie chart.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_palcolor">pie_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the pie chart.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_size">pie_size</code></td>
<td>
<p>A numeric value or a function specifying the size of the pie chart.
If it is a function, the function should take <code>count</code> as the argument and return the size.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_name">pie_name</code></td>
<td>
<p>A character string specifying the name of the legend for the pie chart.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_size_name">pie_size_name</code></td>
<td>
<p>A character string specifying the name of the legend for the pie size.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_pie_values">pie_values</code></td>
<td>
<p>How to calculate the values for the pie chart. Default is &quot;count&quot;.
Other options are &quot;value&quot; and a function.
</p>

<ul>
<li><p> &quot;count&quot;: Count the number of instances in <code>pie_group_by</code>.
</p>
</li>
<li><p> &quot;value&quot;: Use the values in the columns specified by <code>rows</code> as the values.
If there are multiple values for a single group in <code>pie_group_by</code>, a function should be provided to aggregate the values.
If not, the sum of the values will be used and a warning will be shown.
</p>
</li>
<li><p> A function: The function should take a vector of values as the argument and return a single value, for each
group in <code>pie_group_by</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_by_sep">columns_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>columns_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_split_by">columns_split_by</code></td>
<td>
<p>A character string of the column name to split the heatmap columns into slices.
A character/factor column or multiple columns are expected.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_split_name">columns_split_name</code></td>
<td>
<p>A character string specifying the name of the column split annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_palette">columns_palette</code></td>
<td>
<p>A character string specifying the palette of the column group annotation.
The default is &quot;Paired&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_palcolor">columns_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the column group annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_split_by_sep">columns_split_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>columns_split_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_split_palette">columns_split_palette</code></td>
<td>
<p>A character string specifying the palette of the column split annotation.
The default is &quot;simspec&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_columns_split_palcolor">columns_split_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the column split annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_lower_quantile">lower_quantile</code>, <code id="HeatmapAtomic_+3A_upper_quantile">upper_quantile</code>, <code id="HeatmapAtomic_+3A_lower_cutoff">lower_cutoff</code>, <code id="HeatmapAtomic_+3A_upper_cutoff">upper_cutoff</code></td>
<td>
<p>Vector of minimum and maximum cutoff values or quantile values for each feature.
It's applied to aggregated values when aggregated values are used (e.g. plot_type tile, label, etc).
It's applied to raw values when raw values are used (e.g. plot_type bars, etc).</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_data">rows_data</code></td>
<td>
<p>A character string of the column name to use as the data for the row group annotation.
If it starts with &quot;@&quot;, it will be treated as an attribute of the data.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_split_by">rows_split_by</code></td>
<td>
<p>A character string of the column name to split the heatmap rows into slices.
A character/factor column or multiple columns are expected.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_split_by_sep">rows_split_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>rows_split_by</code> if there are multiple columns.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_split_palette">rows_split_palette</code></td>
<td>
<p>A character string specifying the palette of the row split annotation.
The default is &quot;simspec&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_rows_split_palcolor">rows_split_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the row split annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_name_annotation">column_name_annotation</code></td>
<td>
<p>A logical value indicating whether to add the column annotation for the column names.
which is a simple annotaion indicating the column names.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_name_legend">column_name_legend</code></td>
<td>
<p>A logical value indicating whether to show the legend of the column name annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_name_annotation">row_name_annotation</code></td>
<td>
<p>A logical value indicating whether to add the row annotation for the row names.
which is a simple annotaion indicating the row names.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_name_legend">row_name_legend</code></td>
<td>
<p>A logical value indicating whether to show the legend of the row name annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_cluster_columns">cluster_columns</code></td>
<td>
<p>A logical value indicating whether to cluster the columns.
If TRUE and columns_split_by is provided, the clustering will only be applied to the columns within the same split.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>A logical value indicating whether to cluster the rows.
If TRUE and rows_split_by is provided, the clustering will only be applied to the rows within the same split.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_legend_items">legend_items</code></td>
<td>
<p>A numeric vector with names to specifiy the items in the main legend.
The names will be working as the labels of the legend items.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_legend_discrete">legend_discrete</code></td>
<td>
<p>A logical value indicating whether the main legend is discrete.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_show_row_names">show_row_names</code></td>
<td>
<p>A logical value indicating whether to show the row names.
If TRUE, the legend of the row group annotation will be hidden.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_show_column_names">show_column_names</code></td>
<td>
<p>A logical value indicating whether to show the column names.
If TRUE, the legend of the column group annotation will be hidden.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_title">column_title</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the title of the column group annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_title">row_title</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the title of the row group annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_na_col">na_col</code></td>
<td>
<p>A character string specifying the color for missing values.
The default is &quot;grey85&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_title">title</code></td>
<td>
<p>The global (column) title of the heatmap</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_names_side">row_names_side</code></td>
<td>
<p>A character string specifying the side of the row names.
The default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_names_side">column_names_side</code></td>
<td>
<p>A character string specifying the side of the column names.
The default is &quot;bottom&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_bars_sample">bars_sample</code></td>
<td>
<p>An integer specifying the number of samples to draw the bars.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the heatmap.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the labels when <code>cell_type = "label"</code>.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_label_cutoff">label_cutoff</code></td>
<td>
<p>A numeric value specifying the cutoff to show the labels when <code>cell_type = "label"</code>.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_label_accuracy">label_accuracy</code></td>
<td>
<p>A numeric value specifying the accuracy of the labels when <code>cell_type = "label"</code>.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_layer_fun_callback">layer_fun_callback</code></td>
<td>
<p>A function to add additional layers to the heatmap.
The function should have the following arguments: <code>j</code>, <code>i</code>, <code>x</code>, <code>y</code>, <code>w</code>, <code>h</code>, <code>fill</code>, <code>sr</code> and <code>sc</code>.
Please also refer to the <code>layer_fun</code> argument in <code>ComplexHeatmap::Heatmap</code>.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_cell_type">cell_type</code></td>
<td>
<p>A character string specifying the type of the heatmap cells.
The default is values. Other options are &quot;bars&quot;, &quot;label&quot;, &quot;dot&quot;, &quot;violin&quot;, &quot;boxplot&quot;.
Note that for pie chart, the values under columns specified by <code>rows</code> will not be used directly. Instead, the values
will just be counted in different <code>pie_group_by</code> groups. <code>NA</code> values will not be counted.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_cell_agg">cell_agg</code></td>
<td>
<p>A function to aggregate the values in the cell, for the cell type &quot;tile&quot; and &quot;label&quot;.
The default is <code>mean</code>.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the heatmap.
Does not work with <code>cell_type = "bars"</code> or <code>cell_type = "tile"</code>.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>A numeric value between 0 and 1 specifying the transparency of the background.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_violin_fill">violin_fill</code></td>
<td>
<p>A character vector of colors to override the fill color of the violin plot.
If NULL, the fill color will be the same as the annotion.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_boxplot_fill">boxplot_fill</code></td>
<td>
<p>A character vector of colors to override the fill color of the boxplot.
If NULL, the fill color will be the same as the annotion.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_dot_size">dot_size</code></td>
<td>
<p>A numeric value specifying the size of the dot or a function to calculate the size
from the values in the cell.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_dot_size_name">dot_size_name</code></td>
<td>
<p>A character string specifying the name of the legend for the dot size.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation">column_annotation</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the column annotation.
Or a list with the keys as the names of the annotation and the values as the column names.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation_side">column_annotation_side</code></td>
<td>
<p>A character string specifying the side of the column annotation.
Could be a list with the keys as the names of the annotation and the values as the sides.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation_palette">column_annotation_palette</code></td>
<td>
<p>A character string specifying the palette of the column annotation.
The default is &quot;Paired&quot;.
Could be a list with the keys as the names of the annotation and the values as the palettes.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation_palcolor">column_annotation_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the column annotation.
Could be a list with the keys as the names of the annotation and the values as the palcolors.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation_type">column_annotation_type</code></td>
<td>
<p>A character string specifying the type of the column annotation.
The default is &quot;auto&quot;. Other options are &quot;simple&quot;, &quot;pie&quot;, &quot;ring&quot;, &quot;bar&quot;, &quot;violin&quot;, &quot;boxplot&quot;, &quot;density&quot;.
Could be a list with the keys as the names of the annotation and the values as the types.
If the type is &quot;auto&quot;, the type will be determined by the type and number of the column data.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation_params">column_annotation_params</code></td>
<td>
<p>A list of parameters passed to the annotation function.
Could be a list with the keys as the names of the annotation and the values as the parameters.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_column_annotation_agg">column_annotation_agg</code></td>
<td>
<p>A function to aggregate the values in the column annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation">row_annotation</code></td>
<td>
<p>A character string/vector of the column name(s) to use as the row annotation.
Or a list with the keys as the names of the annotation and the values as the column names.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation_side">row_annotation_side</code></td>
<td>
<p>A character string specifying the side of the row annotation.
Could be a list with the keys as the names of the annotation and the values as the sides.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation_palette">row_annotation_palette</code></td>
<td>
<p>A character string specifying the palette of the row annotation.
The default is &quot;Paired&quot;.
Could be a list with the keys as the names of the annotation and the values as the palettes.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation_palcolor">row_annotation_palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the row annotation.
Could be a list with the keys as the names of the annotation and the values as the palcolors.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation_type">row_annotation_type</code></td>
<td>
<p>A character string specifying the type of the row annotation.
The default is &quot;auto&quot;. Other options are &quot;simple&quot;, &quot;pie&quot;, &quot;ring&quot;, &quot;bar&quot;, &quot;violin&quot;, &quot;boxplot&quot;, &quot;density&quot;.
Could be a list with the keys as the names of the annotation and the values as the types.
If the type is &quot;auto&quot;, the type will be determined by the type and number of the row data.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation_params">row_annotation_params</code></td>
<td>
<p>A list of parameters passed to the annotation function.
Could be a list with the keys as the names of the annotation and the values as the parameters.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_row_annotation_agg">row_annotation_agg</code></td>
<td>
<p>A function to aggregate the values in the row annotation.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_add_reticle">add_reticle</code></td>
<td>
<p>A logical value indicating whether to add a reticle to the heatmap.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_reticle_color">reticle_color</code></td>
<td>
<p>A character string specifying the color of the reticle.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_return_grob">return_grob</code></td>
<td>
<p>A logical value indicating whether to return the grob object of the heatmap.
This is useful when merging multiple heatmaps using patchwork.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette of the heatmap cells.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character vector of colors to override the palette of the heatmap cells.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value between 0 and 1 specifying the transparency of the heatmap cells.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="HeatmapAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A drawn HeatmapList object if <code>return_grob = FALSE</code>. Otherwise, a grob/gTree object.
</p>

<hr>
<h2 id='layer_white_bg'>Heatmap layer functions used to draw on the heatmap cells</h2><span id='topic+layer_white_bg'></span><span id='topic+layer_bg'></span><span id='topic+layer_reticle'></span><span id='topic+layer_dot'></span><span id='topic+layer_bars'></span><span id='topic+layer_pie'></span><span id='topic+layer_violin'></span><span id='topic+layer_boxplot'></span>

<h3>Description</h3>

<p>Heatmap layer functions used to draw on the heatmap cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layer_white_bg(j, i, x, y, w, h, fill)

layer_bg(j, i, x, y, w, h, fill, alpha)

layer_reticle(j, i, x, y, w, h, fill, color)

layer_dot(j, i, x, y, w, h, fill, hmdf, dot_size, alpha)

layer_bars(j, i, x, y, w, h, fill, col_fun, hmdf, alpha)

layer_pie(j, i, x, y, w, h, fill, palette, palcolor, hmdf, pie_size)

layer_violin(j, i, x, y, w, h, fill, hmdf, violin_fill)

layer_boxplot(j, i, x, y, w, h, fill, hmdf, boxplot_fill)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layer_white_bg_+3A_j">j</code></td>
<td>
<p>An integer specifying the column index</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_i">i</code></td>
<td>
<p>An integer specifying the row index</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_x">x</code></td>
<td>
<p>A numeric vector specifying the x position</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_y">y</code></td>
<td>
<p>A numeric vector specifying the y position</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_w">w</code></td>
<td>
<p>A numeric vector specifying the width</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_h">h</code></td>
<td>
<p>A numeric vector specifying the height</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_fill">fill</code></td>
<td>
<p>A character vector specifying the fill color</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value between 0 and 1 specifying the transparency of the fill color</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_color">color</code></td>
<td>
<p>A character vector specifying the color of the reticle</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_hmdf">hmdf</code></td>
<td>
<p>A dataframe used to create the annotation. Different from the data used to
create the heatmap itself, which is aggregated data. This dataframe is the original data,
where each cell could have multiple values.</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_dot_size">dot_size</code></td>
<td>
<p>A numeric value specifying the size of the dot or a function to calculate the size
from the values in the cell</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_col_fun">col_fun</code></td>
<td>
<p>A function to calculate the color of the bars</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_violin_fill">violin_fill</code></td>
<td>
<p>A character vector specifying the fill color of the violin plot.
If not provided, the fill color of row/column annotation will be used</p>
</td></tr>
<tr><td><code id="layer_white_bg_+3A_boxplot_fill">boxplot_fill</code></td>
<td>
<p>A character vector specifying the fill color of the boxplot.
If not provided, the fill color of row/column annotation will be used</p>
</td></tr>
</table>

<hr>
<h2 id='LinePlot'>Line Plot</h2><span id='topic+LinePlot'></span>

<h3>Description</h3>

<p>Visualizing the change of a numeric value over the progression of a categorical variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinePlot(
  data,
  x,
  y = NULL,
  group_by = NULL,
  group_by_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  fill_point_by_x_if_no_group = TRUE,
  color_line_by_x_if_no_group = TRUE,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_errorbars = FALSE,
  errorbar_width = 0.1,
  errorbar_alpha = 1,
  errorbar_color = "grey30",
  errorbar_linewidth = 0.75,
  errorbar_min = NULL,
  errorbar_max = NULL,
  errorbar_sd = NULL,
  highlight = NULL,
  highlight_size = pt_size - 0.75,
  highlight_color = "red2",
  highlight_alpha = 0.8,
  pt_alpha = 1,
  pt_size = 5,
  line_type = "solid",
  line_width = 1,
  line_alpha = 0.8,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  facet_by = NULL,
  facet_scales = "fixed",
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  facet_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LinePlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string specifying the separator to use when concatenating multiple columns.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="LinePlot_+3A_fill_point_by_x_if_no_group">fill_point_by_x_if_no_group</code></td>
<td>
<p>A logical value indicating whether to color the points by the x-axis values
when there is no group_by column.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_color_line_by_x_if_no_group">color_line_by_x_if_no_group</code></td>
<td>
<p>A logical value indicating whether to color the lines by the x-axis values</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_bg_palette">bg_palette</code></td>
<td>
<p>The palette to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>The color to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>The alpha value of the background.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_add_errorbars">add_errorbars</code></td>
<td>
<p>A logical value indicating whether to add error bars to the plot.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_width">errorbar_width</code></td>
<td>
<p>The width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_alpha">errorbar_alpha</code></td>
<td>
<p>The alpha value of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_color">errorbar_color</code></td>
<td>
<p>The color to use for the error bars.
If &quot;line&quot;, the error bars will be colored the same as the lines.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_linewidth">errorbar_linewidth</code></td>
<td>
<p>The line width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_min">errorbar_min</code></td>
<td>
<p>The column in the data frame containing the lower bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_max">errorbar_max</code></td>
<td>
<p>The column in the data frame containing the upper bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_errorbar_sd">errorbar_sd</code></td>
<td>
<p>The column in the data frame containing the standard deviation of the error bars.
If errorbar_min and errorbar_max are not provided, this column will be used to calculate the error bars.
errorbar_min = y - errorbar_sd, errorbar_max = y + errorbar_sd.
If errorbar_min and errorbar_max are provided, this column will be ignored.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_highlight">highlight</code></td>
<td>
<p>A vector of indexes or rownames to select the points to highlight.
It could also be an expression (in string) to filter the data.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>The size of the highlighted points.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character vector specifying the color of the highlighted points. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the highlighted points. Default is 1.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>The alpha value of the points.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_pt_size">pt_size</code></td>
<td>
<p>The size of the points.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_line_type">line_type</code></td>
<td>
<p>The type of line to draw.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_line_width">line_width</code></td>
<td>
<p>The width of the line.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_line_alpha">line_alpha</code></td>
<td>
<p>The alpha value of the line.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="LinePlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="LinePlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_facet_args">facet_args</code></td>
<td>
<p>A list of arguments to pass to <code><a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap()</a></code> or <code><a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid()</a></code>.
when there is no group_by column.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(
   x = factor(c("A", "B", "C", "D", "A", "B", "C", "D"), levels = LETTERS[1:6]),
   y = c(10, 8, 16, 4, 6, 12, 14, 2),
   group = c("G1", "G1", "G1", "G1", "G2", "G2", "G2", "G2"),
   facet = c("F1", "F1", "F2", "F2", "F3", "F3", "F4", "F4")
)

LinePlot(data, x = "x", y = "y")
LinePlot(data, x = "x", y = "y", highlight = "group == 'G1'",
   fill_point_by_x_if_no_group = FALSE, color_line_by_x_if_no_group = FALSE)
LinePlot(data, x = "x", y = "y", group_by = "group")
LinePlot(data, x = "x", y = "y", group_by = "group", add_bg = TRUE,
   highlight = "y &gt; 10")
LinePlot(data, x = "x", y = "y", group_by = "group", facet_by = "facet")
LinePlot(data, x = "x", y = "y", group_by = "group", split_by = "facet")
LinePlot(data, x = "x", y = "y", split_by = "group",
         palcolor = list(G1 = c("red", "blue"), G2 = c("green", "black")))
</code></pre>

<hr>
<h2 id='LinePlotAtomic'>LinePlotAtomic</h2><span id='topic+LinePlotAtomic'></span>

<h3>Description</h3>

<p>Line plot with atomic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinePlotAtomic(
  data,
  x,
  y = NULL,
  group_by = NULL,
  fill_point_by_x_if_no_group = TRUE,
  color_line_by_x_if_no_group = TRUE,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_errorbars = FALSE,
  errorbar_width = 0.1,
  errorbar_alpha = 1,
  errorbar_color = "grey30",
  errorbar_linewidth = 0.75,
  errorbar_min = NULL,
  errorbar_max = NULL,
  errorbar_sd = NULL,
  highlight = NULL,
  highlight_size = pt_size - 0.75,
  highlight_color = "red2",
  highlight_alpha = 0.8,
  pt_alpha = 1,
  pt_size = 5,
  line_type = "solid",
  line_width = 1,
  line_alpha = 0.8,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_args = list(),
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LinePlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_fill_point_by_x_if_no_group">fill_point_by_x_if_no_group</code></td>
<td>
<p>A logical value indicating whether to color the points by the x-axis values
when there is no group_by column.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_color_line_by_x_if_no_group">color_line_by_x_if_no_group</code></td>
<td>
<p>A logical value indicating whether to color the lines by the x-axis values</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_bg_palette">bg_palette</code></td>
<td>
<p>The palette to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>The color to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>The alpha value of the background.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_add_errorbars">add_errorbars</code></td>
<td>
<p>A logical value indicating whether to add error bars to the plot.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_width">errorbar_width</code></td>
<td>
<p>The width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_alpha">errorbar_alpha</code></td>
<td>
<p>The alpha value of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_color">errorbar_color</code></td>
<td>
<p>The color to use for the error bars.
If &quot;line&quot;, the error bars will be colored the same as the lines.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_linewidth">errorbar_linewidth</code></td>
<td>
<p>The line width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_min">errorbar_min</code></td>
<td>
<p>The column in the data frame containing the lower bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_max">errorbar_max</code></td>
<td>
<p>The column in the data frame containing the upper bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_errorbar_sd">errorbar_sd</code></td>
<td>
<p>The column in the data frame containing the standard deviation of the error bars.
If errorbar_min and errorbar_max are not provided, this column will be used to calculate the error bars.
errorbar_min = y - errorbar_sd, errorbar_max = y + errorbar_sd.
If errorbar_min and errorbar_max are provided, this column will be ignored.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_highlight">highlight</code></td>
<td>
<p>A vector of indexes or rownames to select the points to highlight.
It could also be an expression (in string) to filter the data.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_highlight_size">highlight_size</code></td>
<td>
<p>The size of the highlighted points.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character vector specifying the color of the highlighted points. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the highlighted points. Default is 1.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>The alpha value of the points.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>The size of the points.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_line_type">line_type</code></td>
<td>
<p>The type of line to draw.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_line_width">line_width</code></td>
<td>
<p>The width of the line.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_line_alpha">line_alpha</code></td>
<td>
<p>The alpha value of the line.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_facet_args">facet_args</code></td>
<td>
<p>A list of arguments to pass to <code><a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap()</a></code> or <code><a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid()</a></code>.
when there is no group_by column.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="LinePlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='LinePlotGrouped'>LinePlotGrouped</h2><span id='topic+LinePlotGrouped'></span>

<h3>Description</h3>

<p>Line plot with groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinePlotGrouped(
  data,
  x,
  y = NULL,
  group_by,
  group_by_sep = "_",
  facet_by = NULL,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_errorbars = FALSE,
  errorbar_width = 0.1,
  errorbar_alpha = 1,
  errorbar_color = "grey30",
  errorbar_linewidth = 0.75,
  errorbar_min = NULL,
  errorbar_max = NULL,
  errorbar_sd = NULL,
  highlight = NULL,
  highlight_size = pt_size - 0.75,
  highlight_color = "red2",
  highlight_alpha = 0.8,
  pt_alpha = 1,
  pt_size = 5,
  line_type = "solid",
  line_width = 1,
  line_alpha = 0.8,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LinePlotGrouped_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to group the plot.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string specifying the separator to use when concatenating multiple columns.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_bg_palette">bg_palette</code></td>
<td>
<p>The palette to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>The color to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>The alpha value of the background.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_add_errorbars">add_errorbars</code></td>
<td>
<p>A logical value indicating whether to add error bars to the plot.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_width">errorbar_width</code></td>
<td>
<p>The width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_alpha">errorbar_alpha</code></td>
<td>
<p>The alpha value of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_color">errorbar_color</code></td>
<td>
<p>The color to use for the error bars.
If &quot;line&quot;, the error bars will be colored the same as the lines.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_linewidth">errorbar_linewidth</code></td>
<td>
<p>The line width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_min">errorbar_min</code></td>
<td>
<p>The column in the data frame containing the lower bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_max">errorbar_max</code></td>
<td>
<p>The column in the data frame containing the upper bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_errorbar_sd">errorbar_sd</code></td>
<td>
<p>The column in the data frame containing the standard deviation of the error bars.
If errorbar_min and errorbar_max are not provided, this column will be used to calculate the error bars.
errorbar_min = y - errorbar_sd, errorbar_max = y + errorbar_sd.
If errorbar_min and errorbar_max are provided, this column will be ignored.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_highlight">highlight</code></td>
<td>
<p>A vector of indexes or rownames to select the points to highlight.
It could also be an expression (in string) to filter the data.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_highlight_size">highlight_size</code></td>
<td>
<p>The size of the highlighted points.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character vector specifying the color of the highlighted points. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the highlighted points. Default is 1.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>The alpha value of the points.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_pt_size">pt_size</code></td>
<td>
<p>The size of the points.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_line_type">line_type</code></td>
<td>
<p>The type of line to draw.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_line_width">line_width</code></td>
<td>
<p>The width of the line.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_line_alpha">line_alpha</code></td>
<td>
<p>The alpha value of the line.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="LinePlotGrouped_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='LinePlotSingle'>LinePlotSingle</h2><span id='topic+LinePlotSingle'></span>

<h3>Description</h3>

<p>Line plot without groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinePlotSingle(
  data,
  x,
  y = NULL,
  fill_point_by_x = TRUE,
  color_line_by_x = TRUE,
  facet_by = NULL,
  add_bg = FALSE,
  bg_palette = "stripe",
  bg_palcolor = NULL,
  bg_alpha = 0.2,
  add_errorbars = FALSE,
  errorbar_width = 0.1,
  errorbar_alpha = 1,
  errorbar_color = "grey30",
  errorbar_linewidth = 0.75,
  errorbar_min = NULL,
  errorbar_max = NULL,
  errorbar_sd = NULL,
  highlight = NULL,
  highlight_size = pt_size - 0.75,
  highlight_color = "red2",
  highlight_alpha = 0.8,
  pt_alpha = 1,
  pt_size = 5,
  line_type = "solid",
  line_width = 1,
  line_alpha = 0.8,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LinePlotSingle_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_fill_point_by_x">fill_point_by_x</code></td>
<td>
<p>A logical value indicating whether to color the points by the x-axis values.
If FALSE, the lines will be colored a single color (the first color in the palette).</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_color_line_by_x">color_line_by_x</code></td>
<td>
<p>A logical value indicating whether to color the lines by the x-axis values.
If FALSE, the lines will be colored a single color (the first color in the palette).</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_add_bg">add_bg</code></td>
<td>
<p>A logical value indicating whether to add a background to the plot.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_bg_palette">bg_palette</code></td>
<td>
<p>The palette to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_bg_palcolor">bg_palcolor</code></td>
<td>
<p>The color to use for the background.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_bg_alpha">bg_alpha</code></td>
<td>
<p>The alpha value of the background.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_add_errorbars">add_errorbars</code></td>
<td>
<p>A logical value indicating whether to add error bars to the plot.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_width">errorbar_width</code></td>
<td>
<p>The width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_alpha">errorbar_alpha</code></td>
<td>
<p>The alpha value of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_color">errorbar_color</code></td>
<td>
<p>The color to use for the error bars.
If &quot;line&quot;, the error bars will be colored the same as the lines.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_linewidth">errorbar_linewidth</code></td>
<td>
<p>The line width of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_min">errorbar_min</code></td>
<td>
<p>The column in the data frame containing the lower bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_max">errorbar_max</code></td>
<td>
<p>The column in the data frame containing the upper bound of the error bars.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_errorbar_sd">errorbar_sd</code></td>
<td>
<p>The column in the data frame containing the standard deviation of the error bars.
If errorbar_min and errorbar_max are not provided, this column will be used to calculate the error bars.
errorbar_min = y - errorbar_sd, errorbar_max = y + errorbar_sd.
If errorbar_min and errorbar_max are provided, this column will be ignored.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_highlight">highlight</code></td>
<td>
<p>A vector of indexes or rownames to select the points to highlight.
It could also be an expression (in string) to filter the data.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_highlight_size">highlight_size</code></td>
<td>
<p>The size of the highlighted points.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character vector specifying the color of the highlighted points. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the highlighted points. Default is 1.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>The alpha value of the points.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_pt_size">pt_size</code></td>
<td>
<p>The size of the points.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_line_type">line_type</code></td>
<td>
<p>The type of line to draw.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_line_width">line_width</code></td>
<td>
<p>The width of the line.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_line_alpha">line_alpha</code></td>
<td>
<p>The alpha value of the line.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="LinePlotSingle_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='Network'>Network</h2><span id='topic+Network'></span>

<h3>Description</h3>

<p>Plot a network graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Network(
  links,
  nodes = NULL,
  split_by = NULL,
  split_by_sep = "_",
  split_nodes = FALSE,
  from = NULL,
  from_sep = "_",
  to = NULL,
  to_sep = "_",
  node_by = NULL,
  node_by_sep = "_",
  link_weight_by = 2,
  link_weight_name = NULL,
  link_type_by = "solid",
  link_type_name = NULL,
  node_size_by = 15,
  node_size_name = NULL,
  node_color_by = "black",
  node_color_name = NULL,
  node_shape_by = 21,
  node_shape_name = NULL,
  node_fill_by = "grey20",
  node_fill_name = NULL,
  link_alpha = 1,
  node_alpha = 0.95,
  node_stroke = 1.5,
  cluster_scale = c("fill", "color", "shape"),
  node_size_range = c(5, 20),
  link_weight_range = c(0.5, 5),
  link_arrow_offset = 20,
  link_curvature = 0,
  link_color_by = "from",
  link_color_name = NULL,
  palette = "Paired",
  palcolor = NULL,
  link_palette = ifelse(link_color_by %in% c("from", "to"), palette, "Set1"),
  link_palcolor = if (link_color_by %in% c("from", "to")) palcolor else NULL,
  directed = TRUE,
  layout = "circle",
  cluster = "none",
  add_mark = FALSE,
  mark_expand = ggplot2::unit(10, "mm"),
  mark_type = c("hull", "ellipse", "rect", "circle"),
  mark_alpha = 0.1,
  mark_linetype = 1,
  add_label = TRUE,
  label_size = 3,
  label_fg = "white",
  label_bg = "black",
  label_bg_r = 0.1,
  arrow = ggplot2::arrow(type = "closed", length = ggplot2::unit(0.1, "inches")),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  aspect.ratio = 1,
  theme = "theme_this",
  theme_args = list(),
  legend.position = "right",
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Network_+3A_links">links</code></td>
<td>
<p>A data frame containing the links between nodes.</p>
</td></tr>
<tr><td><code id="Network_+3A_nodes">nodes</code></td>
<td>
<p>A data frame containing the nodes.
This is optional. The names of the nodes are extracted from the links data frame.
If <code>"@nodes"</code> is provided, the nodes data frame will be extracted from the attribute <code>nodes</code> of the links data frame.</p>
</td></tr>
<tr><td><code id="Network_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="Network_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="Network_+3A_split_nodes">split_nodes</code></td>
<td>
<p>A logical value specifying whether to split the nodes data.
If TRUE, the nodes data will also be split by the <code>split_by</code> column.</p>
</td></tr>
<tr><td><code id="Network_+3A_from">from</code></td>
<td>
<p>A character string specifying the column name of the links data frame for the source nodes.
Default is the first column of the links data frame.</p>
</td></tr>
<tr><td><code id="Network_+3A_from_sep">from_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>from</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="Network_+3A_to">to</code></td>
<td>
<p>A character string specifying the column name of the links data frame for the target nodes.
Default is the second column of the links data frame.</p>
</td></tr>
<tr><td><code id="Network_+3A_to_sep">to_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>to</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_by">node_by</code></td>
<td>
<p>A character string specifying the column name of the nodes data frame for the node names.
Default is the first column of the nodes data frame.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_by_sep">node_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>node_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_weight_by">link_weight_by</code></td>
<td>
<p>A numeric value or a character string specifying the column name of the links data frame for the link weight.
If a numeric value is provided, all links will have the same weight.
This determines the width of the links.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_weight_name">link_weight_name</code></td>
<td>
<p>A character string specifying the name of the link weight in the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_type_by">link_type_by</code></td>
<td>
<p>A character string specifying the type of the links.
This can be &quot;solid&quot;, &quot;dashed&quot;, &quot;dotted&quot;, or a column name from the links data frame.
It has higher priority when it is a column name.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_type_name">link_type_name</code></td>
<td>
<p>A character string specifying the name of the link type in the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_size_by">node_size_by</code></td>
<td>
<p>A numeric value or a character string specifying the column name of the nodes data frame for the node size.
If a numeric value is provided, all nodes will have the same size.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_size_name">node_size_name</code></td>
<td>
<p>A character string specifying the name of the node size in the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_color_by">node_color_by</code></td>
<td>
<p>A character string specifying the color of the nodes.
This can be a color name, a hex code, or a column name from the nodes data frame.
It has higher priority when it is a column name.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_color_name">node_color_name</code></td>
<td>
<p>A character string specifying the name of the node color in the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_shape_by">node_shape_by</code></td>
<td>
<p>A numeric value or a character string specifying the column name of the nodes data frame for the node shape.
If a numeric value is provided, all nodes will have the same shape.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_shape_name">node_shape_name</code></td>
<td>
<p>A character string specifying the name of the node shape in the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_fill_by">node_fill_by</code></td>
<td>
<p>A character string specifying the fill color of the nodes.
This can be a color name, a hex code, or a column name from the nodes data frame.
It has higher priority when it is a column name.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_fill_name">node_fill_name</code></td>
<td>
<p>A character string specifying the name of the node fill in the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_alpha">link_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the links.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_alpha">node_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the nodes.
It only works when the nodes are filled.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_stroke">node_stroke</code></td>
<td>
<p>A numeric value specifying the stroke of the nodes.</p>
</td></tr>
<tr><td><code id="Network_+3A_cluster_scale">cluster_scale</code></td>
<td>
<p>A character string specifying how to scale the clusters.
It can be &quot;fill&quot;, &quot;color&quot;, or &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="Network_+3A_node_size_range">node_size_range</code></td>
<td>
<p>A numeric vector specifying the range of the node size.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_weight_range">link_weight_range</code></td>
<td>
<p>A numeric vector specifying the range of the link weight.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_arrow_offset">link_arrow_offset</code></td>
<td>
<p>A numeric value specifying the offset of the link arrows.
So that they won't overlap with the nodes.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_curvature">link_curvature</code></td>
<td>
<p>A numeric value specifying the curvature of the links.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_color_by">link_color_by</code></td>
<td>
<p>A character string specifying the colors of the link. It can be:
</p>

<ul>
<li><p> &quot;from&quot; means the color of the link is determined by the source node.
</p>
</li>
<li><p> &quot;to&quot; means the color of the link is determined by the target node.
</p>
</li>
<li><p> Otherwise, the color of the link is determined by the column name from the links data frame.
</p>
</li></ul>
</td></tr>
<tr><td><code id="Network_+3A_link_color_name">link_color_name</code></td>
<td>
<p>A character string specifying the name of the link color in the legend.
Only used when <code>link_color_by</code> is a column name.</p>
</td></tr>
<tr><td><code id="Network_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="Network_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="Network_+3A_link_palette">link_palette</code></td>
<td>
<p>A character string specifying the palette of the links.
When <code>link_color_by</code> is &quot;from&quot; or &quot;to&quot;, the palette of the links defaults to the palette of the nodes.</p>
</td></tr>
<tr><td><code id="Network_+3A_link_palcolor">link_palcolor</code></td>
<td>
<p>A character vector specifying the colors of the link palette.
When <code>link_color_by</code> is &quot;from&quot; or &quot;to&quot;, the colors of the link palette defaults to the colors of the node palette.</p>
</td></tr>
<tr><td><code id="Network_+3A_directed">directed</code></td>
<td>
<p>A logical value specifying whether the graph is directed.</p>
</td></tr>
<tr><td><code id="Network_+3A_layout">layout</code></td>
<td>
<p>A character string specifying the layout of the graph.
It can be &quot;circle&quot;, &quot;tree&quot;, &quot;grid&quot;, or a layout function from igraph.</p>
</td></tr>
<tr><td><code id="Network_+3A_cluster">cluster</code></td>
<td>
<p>A character string specifying the clustering method.
It can be &quot;none&quot;, &quot;fast_greedy&quot;, &quot;walktrap&quot;, &quot;edge_betweenness&quot;, &quot;infomap&quot;, or a clustering function from igraph.</p>
</td></tr>
<tr><td><code id="Network_+3A_add_mark">add_mark</code></td>
<td>
<p>A logical value specifying whether to add mark for the clusters to the plot.</p>
</td></tr>
<tr><td><code id="Network_+3A_mark_expand">mark_expand</code></td>
<td>
<p>A unit value specifying the expansion of the mark.</p>
</td></tr>
<tr><td><code id="Network_+3A_mark_type">mark_type</code></td>
<td>
<p>A character string specifying the type of the mark.
It can be &quot;hull&quot;, &quot;ellipse&quot;, &quot;rect&quot;, &quot;circle&quot;, or a mark function from ggforce.</p>
</td></tr>
<tr><td><code id="Network_+3A_mark_alpha">mark_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the mark.</p>
</td></tr>
<tr><td><code id="Network_+3A_mark_linetype">mark_linetype</code></td>
<td>
<p>A numeric value specifying the line type of the mark.</p>
</td></tr>
<tr><td><code id="Network_+3A_add_label">add_label</code></td>
<td>
<p>A logical value specifying whether to add label to the nodes to the plot.</p>
</td></tr>
<tr><td><code id="Network_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the label.</p>
</td></tr>
<tr><td><code id="Network_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string specifying the foreground color of the label.</p>
</td></tr>
<tr><td><code id="Network_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string specifying the background color of the label.</p>
</td></tr>
<tr><td><code id="Network_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the background ratio of the label.</p>
</td></tr>
<tr><td><code id="Network_+3A_arrow">arrow</code></td>
<td>
<p>An arrow object for the links.</p>
</td></tr>
<tr><td><code id="Network_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="Network_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="Network_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="Network_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="Network_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="Network_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="Network_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="Network_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="Network_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="Network_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="Network_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="Network_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="Network_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="Network_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="Network_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
actors &lt;- data.frame(
  name = c("Alice", "Bob", "Cecil", "David", "Esmeralda"),
  age = c(48, 33, 45, 34, 21),
  shape = c(21, 22, 21, 22, 23),
  gender = c("F", "M", "F", "M", "F")
)
relations &lt;- data.frame(
  from = c("Bob", "Cecil", "Cecil", "David", "David", "Esmeralda", "Bob", "Alice",
     "Cecil", "David"),
  to = c("Alice", "Bob", "Alice", "Alice", "Bob", "Alice", "Bob", "Alice", "Cecil",
     "David"),
  friendship = c(4, 5, 5, 2, 1, 1, 2, 1, 3, 4),
  type = c(1, 1, 1, 1, 1, 2, 2, 2, 2, 2)
)
Network(relations, actors)
Network(relations, actors, theme = "theme_blank", theme_args = list(add_coord = FALSE))
Network(relations, actors, link_weight_by = "friendship", node_size_by = "age",
 link_weight_name = "FRIENDSHIP", node_fill_by = "gender", link_color_by = "to",
 link_type_by = "type", node_color_by = "black", layout = "circle", link_curvature = 0.2)
Network(relations, actors, layout = "tree", directed = FALSE, cluster = "fast_greedy",
 add_mark = TRUE)
Network(relations, actors, split_by = "type")

</code></pre>

<hr>
<h2 id='NetworkAtomic'>NetworkAtomic</h2><span id='topic+NetworkAtomic'></span>

<h3>Description</h3>

<p>Plot a network graph without splitting the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NetworkAtomic(
  links,
  nodes = NULL,
  from = NULL,
  from_sep = "_",
  to = NULL,
  to_sep = "_",
  node_by = NULL,
  node_by_sep = "_",
  link_weight_by = 2,
  link_weight_name = NULL,
  link_type_by = "solid",
  link_type_name = NULL,
  node_size_by = 15,
  node_size_name = NULL,
  node_color_by = "black",
  node_color_name = NULL,
  node_shape_by = 21,
  node_shape_name = NULL,
  node_fill_by = "grey20",
  node_fill_name = NULL,
  link_alpha = 1,
  node_alpha = 0.95,
  node_stroke = 1.5,
  cluster_scale = c("fill", "color", "shape"),
  node_size_range = c(5, 20),
  link_weight_range = c(0.5, 5),
  link_arrow_offset = 20,
  link_curvature = 0,
  link_color_by = "from",
  link_color_name = NULL,
  palette = "Paired",
  palcolor = NULL,
  link_palette = ifelse(link_color_by %in% c("from", "to"), palette, "Set1"),
  link_palcolor = if (link_color_by %in% c("from", "to")) palcolor else NULL,
  directed = TRUE,
  layout = "circle",
  cluster = "none",
  add_mark = FALSE,
  mark_expand = ggplot2::unit(10, "mm"),
  mark_type = c("hull", "ellipse", "rect", "circle"),
  mark_alpha = 0.1,
  mark_linetype = 1,
  add_label = TRUE,
  label_size = 3,
  label_fg = "white",
  label_bg = "black",
  label_bg_r = 0.1,
  arrow = ggplot2::arrow(type = "closed", length = ggplot2::unit(0.1, "inches")),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  aspect.ratio = 1,
  theme = "theme_this",
  theme_args = list(),
  legend.position = "right",
  legend.direction = "vertical",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NetworkAtomic_+3A_links">links</code></td>
<td>
<p>A data frame containing the links between nodes.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_nodes">nodes</code></td>
<td>
<p>A data frame containing the nodes.
This is optional. The names of the nodes are extracted from the links data frame.
If <code>"@nodes"</code> is provided, the nodes data frame will be extracted from the attribute <code>nodes</code> of the links data frame.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_from">from</code></td>
<td>
<p>A character string specifying the column name of the links data frame for the source nodes.
Default is the first column of the links data frame.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_from_sep">from_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>from</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_to">to</code></td>
<td>
<p>A character string specifying the column name of the links data frame for the target nodes.
Default is the second column of the links data frame.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_to_sep">to_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>to</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_by">node_by</code></td>
<td>
<p>A character string specifying the column name of the nodes data frame for the node names.
Default is the first column of the nodes data frame.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_by_sep">node_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>node_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_weight_by">link_weight_by</code></td>
<td>
<p>A numeric value or a character string specifying the column name of the links data frame for the link weight.
If a numeric value is provided, all links will have the same weight.
This determines the width of the links.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_weight_name">link_weight_name</code></td>
<td>
<p>A character string specifying the name of the link weight in the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_type_by">link_type_by</code></td>
<td>
<p>A character string specifying the type of the links.
This can be &quot;solid&quot;, &quot;dashed&quot;, &quot;dotted&quot;, or a column name from the links data frame.
It has higher priority when it is a column name.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_type_name">link_type_name</code></td>
<td>
<p>A character string specifying the name of the link type in the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_size_by">node_size_by</code></td>
<td>
<p>A numeric value or a character string specifying the column name of the nodes data frame for the node size.
If a numeric value is provided, all nodes will have the same size.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_size_name">node_size_name</code></td>
<td>
<p>A character string specifying the name of the node size in the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_color_by">node_color_by</code></td>
<td>
<p>A character string specifying the color of the nodes.
This can be a color name, a hex code, or a column name from the nodes data frame.
It has higher priority when it is a column name.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_color_name">node_color_name</code></td>
<td>
<p>A character string specifying the name of the node color in the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_shape_by">node_shape_by</code></td>
<td>
<p>A numeric value or a character string specifying the column name of the nodes data frame for the node shape.
If a numeric value is provided, all nodes will have the same shape.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_shape_name">node_shape_name</code></td>
<td>
<p>A character string specifying the name of the node shape in the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_fill_by">node_fill_by</code></td>
<td>
<p>A character string specifying the fill color of the nodes.
This can be a color name, a hex code, or a column name from the nodes data frame.
It has higher priority when it is a column name.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_fill_name">node_fill_name</code></td>
<td>
<p>A character string specifying the name of the node fill in the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_alpha">link_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the links.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_alpha">node_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the nodes.
It only works when the nodes are filled.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_stroke">node_stroke</code></td>
<td>
<p>A numeric value specifying the stroke of the nodes.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_cluster_scale">cluster_scale</code></td>
<td>
<p>A character string specifying how to scale the clusters.
It can be &quot;fill&quot;, &quot;color&quot;, or &quot;shape&quot;.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_node_size_range">node_size_range</code></td>
<td>
<p>A numeric vector specifying the range of the node size.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_weight_range">link_weight_range</code></td>
<td>
<p>A numeric vector specifying the range of the link weight.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_arrow_offset">link_arrow_offset</code></td>
<td>
<p>A numeric value specifying the offset of the link arrows.
So that they won't overlap with the nodes.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_curvature">link_curvature</code></td>
<td>
<p>A numeric value specifying the curvature of the links.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_color_by">link_color_by</code></td>
<td>
<p>A character string specifying the colors of the link. It can be:
</p>

<ul>
<li><p> &quot;from&quot; means the color of the link is determined by the source node.
</p>
</li>
<li><p> &quot;to&quot; means the color of the link is determined by the target node.
</p>
</li>
<li><p> Otherwise, the color of the link is determined by the column name from the links data frame.
</p>
</li></ul>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_color_name">link_color_name</code></td>
<td>
<p>A character string specifying the name of the link color in the legend.
Only used when <code>link_color_by</code> is a column name.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette of the nodes.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character vector specifying the colors of the node palette.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_palette">link_palette</code></td>
<td>
<p>A character string specifying the palette of the links.
When <code>link_color_by</code> is &quot;from&quot; or &quot;to&quot;, the palette of the links defaults to the palette of the nodes.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_link_palcolor">link_palcolor</code></td>
<td>
<p>A character vector specifying the colors of the link palette.
When <code>link_color_by</code> is &quot;from&quot; or &quot;to&quot;, the colors of the link palette defaults to the colors of the node palette.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_directed">directed</code></td>
<td>
<p>A logical value specifying whether the graph is directed.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_layout">layout</code></td>
<td>
<p>A character string specifying the layout of the graph.
It can be &quot;circle&quot;, &quot;tree&quot;, &quot;grid&quot;, or a layout function from igraph.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_cluster">cluster</code></td>
<td>
<p>A character string specifying the clustering method.
It can be &quot;none&quot;, &quot;fast_greedy&quot;, &quot;walktrap&quot;, &quot;edge_betweenness&quot;, &quot;infomap&quot;, or a clustering function from igraph.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_add_mark">add_mark</code></td>
<td>
<p>A logical value specifying whether to add mark for the clusters to the plot.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_mark_expand">mark_expand</code></td>
<td>
<p>A unit value specifying the expansion of the mark.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_mark_type">mark_type</code></td>
<td>
<p>A character string specifying the type of the mark.
It can be &quot;hull&quot;, &quot;ellipse&quot;, &quot;rect&quot;, &quot;circle&quot;, or a mark function from ggforce.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_mark_alpha">mark_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the mark.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_mark_linetype">mark_linetype</code></td>
<td>
<p>A numeric value specifying the line type of the mark.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_add_label">add_label</code></td>
<td>
<p>A logical value specifying whether to add label to the nodes to the plot.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the label.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string specifying the foreground color of the label.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string specifying the background color of the label.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the background ratio of the label.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_arrow">arrow</code></td>
<td>
<p>An arrow object for the links.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="NetworkAtomic_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='norm_expansion'>Expand the plot area with CSS-like padding</h2><span id='topic+norm_expansion'></span>

<h3>Description</h3>

<p>Expand the plot area with CSS-like padding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>norm_expansion(
  expand,
  x_type,
  y_type,
  continuous_default = c(0.05, 0),
  discrete_default = c(0, 0.6)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="norm_expansion_+3A_expand">expand</code></td>
<td>
<p>A numeric vector of length 1, 2, 3, or 4
The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="norm_expansion_+3A_x_type">x_type</code></td>
<td>
<p>The type of x-axis, either &quot;continuous&quot; or &quot;discrete&quot;</p>
</td></tr>
<tr><td><code id="norm_expansion_+3A_y_type">y_type</code></td>
<td>
<p>The type of y-axis, either &quot;continuous&quot; or &quot;discrete&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with x and y values for expand
</p>

<hr>
<h2 id='palette_list'>A list of palettes for use in data visualization</h2><span id='topic+palette_list'></span>

<h3>Description</h3>

<p>A list of palettes for use in data visualization
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (interactive()) {
  library(stringr)
  library(RColorBrewer)
  library(Redmonder)
  library(rcartocolor)
  library(nord)
  library(viridis)
  library(pals)
  library(dichromat)
  library(jcolors)
  library(scales)
  syspals &lt;- utils::getFromNamespace("syspals", "pals")
  brewer.pal.info &lt;- RColorBrewer::brewer.pal.info
  ggsci_db &lt;- utils::getFromNamespace("ggsci_db", "ggsci")
  redmonder.pal.info &lt;- Redmonder::redmonder.pal.info
  metacartocolors &lt;- rcartocolor::metacartocolors
  rownames(metacartocolors) &lt;- metacartocolors$Name
  nord_palettes &lt;- nord::nord_palettes
  viridis_names &lt;- c("magma", "inferno", "plasma", "viridis", "cividis", "rocket",
     "mako", "turbo")
  viridis_palettes &lt;- lapply(setNames(viridis_names, viridis_names),
     function(x) viridis::viridis(100, option = x))
  ocean_names &lt;- names(syspals)[grep("ocean", names(syspals))]
  ocean_palettes &lt;- syspals[ocean_names]
  dichromat_palettes &lt;- dichromat::colorschemes
  jcolors_names &lt;- paste0("jcolors-", c("default", "pal2", "pal3", "pal4", "pal5",
     "pal6", "pal7", "pal8", "pal9", "pal10", "pal11", "pal12", "rainbow"))
  custom_names &lt;- c("jet", "simspec", "GdRd")
  custom_palettes &lt;- list(
    oompaBase::jetColors(N = 100),
    c("#c22b86", "#f769a1", "#fcc5c1", "#253777", "#1d92c0", "#9ec9e1", "#015b33",
        "#42aa5e", "#d9f0a2", "#E66F00", "#f18c28", "#FFBB61"),
    c("gold", "red3")
  )
  names(custom_palettes) &lt;- custom_names
  seurat_discrete_palettes &lt;- list(
     alphabet = c(
       "#F0A0FF", "#0075DC", "#993F00", "#4C005C", "#191919", "#005C31",
       "#2BCE48", "#FFCC99", "#808080", "#94FFB5", "#8F7C00", "#9DCC00",
       "#C20088", "#003380", "#FFA405", "#FFA8BB", "#426600", "#FF0010",
       "#5EF1F2", "#00998F", "#E0FF66", "#740AFF", "#990000", "#FFFF80",
       "#FFE100", "#FF5005"
     ),
     alphabet2 = c(
       "#AA0DFE", "#3283FE", "#85660D", "#782AB6", "#565656", "#1C8356",
       "#16FF32", "#F7E1A0", "#E2E2E2", "#1CBE4F", "#C4451C", "#DEA0FD",
       "#FE00FA", "#325A9B", "#FEAF16", "#F8A19F", "#90AD1C", "#F6222E",
       "#1CFFCE", "#2ED9FF", "#B10DA1", "#C075A6", "#FC1CBF", "#B00068",
       "#FBE426", "#FA0087"
     ),
     glasbey = c(
       "#0000FF", "#FF0000", "#00FF00", "#000033", "#FF00B6", "#005300",
       "#FFD300", "#009FFF", "#9A4D42", "#00FFBE", "#783FC1", "#1F9698",
       "#FFACFD", "#B1CC71", "#F1085C", "#FE8F42", "#DD00FF", "#201A01",
       "#720055", "#766C95", "#02AD24", "#C8FF00", "#886C00", "#FFB79F",
       "#858567", "#A10300", "#14F9FF", "#00479E", "#DC5E93", "#93D4FF",
       "#004CFF", "#F2F318"
     ),
     polychrome = c(
       "#5A5156", "#E4E1E3", "#F6222E", "#FE00FA", "#16FF32", "#3283FE",
       "#FEAF16", "#B00068", "#1CFFCE", "#90AD1C", "#2ED9FF", "#DEA0FD",
       "#AA0DFE", "#F8A19F", "#325A9B", "#C4451C", "#1C8356", "#85660D",
       "#B10DA1", "#FBE426", "#1CBE4F", "#FA0087", "#FC1CBF", "#F7E1A0",
       "#C075A6", "#782AB6", "#AAF400", "#BDCDFF", "#822E1C", "#B5EFB5",
       "#7ED7D1", "#1C7F93", "#D85FF7", "#683B79", "#66B0FF", "#3B00FB"
     ),
     stepped = c(
       "#990F26", "#B33E52", "#CC7A88", "#E6B8BF", "#99600F", "#B3823E",
       "#CCAA7A", "#E6D2B8", "#54990F", "#78B33E", "#A3CC7A", "#CFE6B8",
       "#0F8299", "#3E9FB3", "#7ABECC", "#B8DEE6", "#3D0F99", "#653EB3",
       "#967ACC", "#C7B8E6", "#333333", "#666666", "#999999", "#CCCCCC"
     ),
     parade = c(
       '#ff6969', '#9b37ff', '#cd3737', '#69cdff', '#ffff69', '#69cdcd',
       '#9b379b', '#3737cd', '#ffff9b', '#cdff69', '#ff9b37', '#37ffff',
       '#9b69ff', '#37cd69', '#ff3769', '#ff3737', '#37ff9b', '#cdcd37',
       '#3769cd', '#37cdff', '#9b3737', '#ff699b', '#9b9bff', '#cd9b37',
       '#69ff37', '#cd3769', '#cd69cd', '#cd6937', '#3737ff', '#cdcd69',
       '#ff9b69', '#cd37cd', '#9bff37', '#cd379b', '#cd6969', '#69ff9b',
       '#ff379b', '#9bff9b', '#6937ff', '#69cd37', '#cdff37', '#9bff69',
       '#9b37cd', '#ff37ff', '#ff37cd', '#ffff37', '#37cd9b', '#379bff',
       '#ffcd37', '#379b37', '#ff9bff', '#379b9b', '#69ffcd', '#379bcd',
       '#ff69ff', '#ff9b9b', '#37ff69', '#ff6937', '#6969ff', '#699bff',
       '#ffcd69', '#69ffff', '#37ff37', '#6937cd', '#37cd37', '#3769ff',
       '#cd69ff', '#6969cd', '#9bcd37', '#69ff69', '#37cdcd', '#cd37ff',
       '#37379b', '#37ffcd', '#69cd69'
     )
  )
  seurat_continuous_palettes &lt;- list(
    seurat.16 = hue_pal()(16),
    seurat.32 = hue_pal()(32),
    seurat.64 = hue_pal()(64)
  )

  palette_list &lt;- list()
  all_colors &lt;- c(
    rownames(brewer.pal.info), names(ggsci_db), rownames(redmonder.pal.info),
    rownames(metacartocolors), names(nord_palettes), names(viridis_palettes),
    ocean_names, names(dichromat_palettes), jcolors_names, names(seurat_palettes),
    names(seurat_continuous_palettes), custom_names
  )
  for (pal in all_colors) {
    if (!pal %in% all_colors) {
      stop(paste0("Invalid pal Must be one of ", paste0(all_colors, collapse = ",")))
    }
    if (pal %in% rownames(brewer.pal.info)) {
      pal_n &lt;- brewer.pal.info[pal, "maxcolors"]
      pal_category &lt;- brewer.pal.info[pal, "category"]
      if (pal_category == "div") {
        palcolor &lt;- rev(brewer.pal(name = pal, n = pal_n))
      } else {
        if (pal == "Paired") {
          palcolor &lt;- brewer.pal(12, "Paired")[c(1:4, 7, 8, 5, 6, 9, 10, 11, 12)]
        } else {
          palcolor &lt;- brewer.pal(name = pal, n = pal_n)
        }
      }
      if (pal_category == "qual") {
        attr(palcolor, "type") &lt;- "discrete"
      } else {
        attr(palcolor, "type") &lt;- "continuous"
      }
    } else if (pal %in% names(ggsci_db)) {
      if (pal %in% c("d3", "uchicago", "material")) {
        for (subpal in names(ggsci_db[[pal]])) {
          palcolor &lt;- ggsci_db[[pal]][[subpal]]
          if (pal == "material") {
            attr(palcolor, "type") &lt;- "continuous"
          } else {
            attr(palcolor, "type") &lt;- "discrete"
          }
          palette_list[[paste0(pal, "-", subpal)]] &lt;- palcolor
        }
        next
      } else {
        palcolor &lt;- ggsci_db[[pal]][[1]]
        if (pal == "gsea") {
          attr(palcolor, "type") &lt;- "continuous"
        } else {
          attr(palcolor, "type") &lt;- "discrete"
        }
      }
    } else if (pal %in% rownames(redmonder.pal.info)) {
      pal_n &lt;- redmonder.pal.info[pal, "maxcolors"]
      pal_category &lt;- redmonder.pal.info[pal, "category"]
      if (pal_category == "div") {
        palcolor &lt;- rev(redmonder.pal(name = pal, n = pal_n))
      } else {
        palcolor &lt;- redmonder.pal(name = pal, n = pal_n)
      }
      if (pal_category == "qual") {
        attr(palcolor, "type") &lt;- "discrete"
      } else {
        attr(palcolor, "type") &lt;- "continuous"
      }
    } else if (pal %in% rownames(metacartocolors)) {
      pal_n &lt;- metacartocolors[pal, "Max_n"]
      palcolor &lt;- carto_pal(name = pal, n = pal_n)
      if (pal_category == "qualitative") {
        attr(palcolor, "type") &lt;- "discrete"
      } else {
        attr(palcolor, "type") &lt;- "continuous"
      }
    } else if (pal %in% names(nord_palettes)) {
      palcolor &lt;- nord_palettes[[pal]]
      attr(palcolor, "type") &lt;- "discrete"
    } else if (pal %in% names(viridis_palettes)) {
      palcolor &lt;- viridis_palettes[[pal]]
      attr(palcolor, "type") &lt;- "continuous"
    } else if (pal %in% names(ocean_palettes)) {
      palcolor &lt;- ocean_palettes[[pal]]
      attr(palcolor, "type") &lt;- "continuous"
    } else if (pal %in% names(dichromat_palettes)) {
      palcolor &lt;- dichromat_palettes[[pal]]
      if (pal %in% c("Categorical.12", "SteppedSequential.5")) {
        attr(palcolor, "type") &lt;- "discrete"
      } else {
        attr(palcolor, "type") &lt;- "continuous"
      }
    } else if (pal %in% jcolors_names) {
      palcolor &lt;- jcolors(palette = gsub("jcolors-", "", pal))
      if (pal %in% paste0("jcolors-", c("pal10", "pal11", "pal12", "rainbow"))) {
        attr(palcolor, "type") &lt;- "continuous"
      } else {
        attr(palcolor, "type") &lt;- "discrete"
      }
    } else if (pal %in% custom_names) {
      palcolor &lt;- custom_palettes[[pal]]
      if (pal %in% c("jet")) {
        attr(palcolor, "type") &lt;- "continuous"
      } else {
        attr(palcolor, "type") &lt;- "discrete"
      }
    } else if (pal %in% names(seurat_discrete_palettes)) {
      palcolor &lt;- seurat_discrete_palettes[[pal]]
      attr(palcolor, "type") &lt;- "discrete"
    } else if (pal %in% names(seurat_continuous_palettes)) {
      palcolor &lt;- seurat_continuous_palettes[[pal]]
      attr(palcolor, "type") &lt;- "continuous"
    }
    palette_list[[pal]] &lt;- palcolor
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='palette_this'>Color palettes collected in plotthis.</h2><span id='topic+palette_this'></span>

<h3>Description</h3>

<p>Color palettes collected in plotthis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>palette_this(
  x,
  n = 100,
  palette = "Paired",
  palcolor = NULL,
  type = "auto",
  keep_names = TRUE,
  alpha = 1,
  matched = FALSE,
  reverse = FALSE,
  NA_keep = FALSE,
  NA_color = "grey80",
  transparent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="palette_this_+3A_x">x</code></td>
<td>
<p>A vector of character/factor or numeric values. If missing, numeric values 1:n will be used as x.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_n">n</code></td>
<td>
<p>The number of colors to return for numeric values.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_palette">palette</code></td>
<td>
<p>Palette name. All available palette names can be queried with <code>show_palettes()</code>.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_palcolor">palcolor</code></td>
<td>
<p>Custom colors used to create a color palette.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_type">type</code></td>
<td>
<p>Type of <code>x</code>. Can be one of &quot;auto&quot;, &quot;discrete&quot; or &quot;continuous&quot;. The default is &quot;auto&quot;, which automatically detects if <code>x</code> is a numeric value.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_keep_names">keep_names</code></td>
<td>
<p>Whether to keep the names of the color vector.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_alpha">alpha</code></td>
<td>
<p>The alpha value of the colors. Default is 1.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_matched">matched</code></td>
<td>
<p>If <code>TRUE</code>, will return a color vector of the same length as <code>x</code>.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_reverse">reverse</code></td>
<td>
<p>Whether to invert the colors.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_na_keep">NA_keep</code></td>
<td>
<p>Whether to keep the color assignment to NA in <code>x</code>.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_na_color">NA_color</code></td>
<td>
<p>Color assigned to NA if NA_keep is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="palette_this_+3A_transparent">transparent</code></td>
<td>
<p>Whether to make the colors transparent when alpha &lt; 1.
When <code>TRUE</code>, <code><a href="ggplot2.html#topic+reexports">ggplot2::alpha()</a></code> is used to make the colors transparent.
Otherwise, <code>adjcolors</code> is used to adjust the colors based on the alpha. The color will be not be actually transparent.
For example, <code>ggplot2::alpha("red", 0.5) == "#FF000080"</code>; while <code>adjcolors("red", 0.5) == "#FF8080"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of colors.
</p>

<hr>
<h2 id='PieChart'>Pie Chart</h2><span id='topic+PieChart'></span>

<h3>Description</h3>

<p>Pie chart to illustrate numerical proportion of each group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PieChart(
  data,
  x,
  y = NULL,
  label = y,
  split_by = NULL,
  split_by_sep = "_",
  clockwise = TRUE,
  facet_by = NULL,
  facet_scales = "free_y",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PieChart_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis.
A numeric column is expected.
If NULL, the count of each x column will be used.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_label">label</code></td>
<td>
<p>Which column to use as the label. NULL means no label.
Default is the same as y. If y is NULL, you should use &quot;.y&quot; to specify the count as the label.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="PieChart_+3A_clockwise">clockwise</code></td>
<td>
<p>A logical value to draw the pie chart clockwise or not.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="PieChart_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="PieChart_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="PieChart_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="PieChart_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(
   x = c("A", "B", "C", "D", "E", "F", "G", "H"),
   y = c(10, 8, 16, 4, 6, 12, 14, 2),
   group = c("G1", "G1", "G2", "G2", "G3", "G3", "G4", "G4"),
   facet = c("F1", "F2", "F3", "F4", "F1", "F2", "F3", "F4")
)

PieChart(data, x = "x", y = "y")
PieChart(data, x = "x", y = "y", clockwise = FALSE)
PieChart(data, x = "x", y = "y", label = "group")
PieChart(data, x = "x", y = "y", facet_by = "facet")
PieChart(data, x = "x", y = "y", split_by = "group")
PieChart(data, x = "x", y = "y", split_by = "group",
         palette = list(G1 = "Reds", G2 = "Blues", G3 = "Greens", G4 = "Purp"))

# y from count
PieChart(data, x = "group")
# add label
PieChart(data, x = "group", label = ".y")
</code></pre>

<hr>
<h2 id='PieChartAtomic'>Pie chart without data splitting</h2><span id='topic+PieChartAtomic'></span>

<h3>Description</h3>

<p>Pie chart without data splitting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PieChartAtomic(
  data,
  x,
  y = NULL,
  label = y,
  clockwise = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "free_y",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PieChartAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis.
A numeric column is expected.
If NULL, the count of each x column will be used.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_label">label</code></td>
<td>
<p>Which column to use as the label. NULL means no label.
Default is the same as y. If y is NULL, you should use &quot;.y&quot; to specify the count as the label.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_clockwise">clockwise</code></td>
<td>
<p>A logical value to draw the pie chart clockwise or not.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="PieChartAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='PrepareEnrichrResult'>Enrichment Map/Network</h2><span id='topic+PrepareEnrichrResult'></span><span id='topic+EnrichMap'></span><span id='topic+EnrichNetwork'></span>

<h3>Description</h3>

<p><code>PrepareEnrichrResult</code> is a function to process the enrichment results from Enrichr.
<code>EnrichMap</code> is a function to plot the enrichment map.
<code>EnrichNetwork</code> is a function to plot the enrichment network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrepareEnrichrResult(data, n_input = NULL)

EnrichMap(
  data,
  split_by = NULL,
  split_by_sep = "_",
  top_term = 10,
  metric = "p.adjust",
  layout = "fr",
  minchar = 2,
  cluster = "fast_greedy",
  show_keyword = FALSE,
  nlabel = 4,
  character_width = 50,
  mark = "ellipse",
  label = c("term", "feature"),
  labelsize = 5,
  expand = c(0.4, 0.4),
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

EnrichNetwork(
  data,
  split_by = NULL,
  split_by_sep = "_",
  top_term = 10,
  metric = "p.adjust",
  character_width = 50,
  layout = "fr",
  layoutadjust = TRUE,
  adjscale = 60,
  adjiter = 100,
  blendmode = "blend",
  labelsize = 5,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PrepareEnrichrResult_+3A_data">data</code></td>
<td>
<p>A data frame containing the data to be plotted.
It should be in the format of clusterProfiler enrichment result,
which includes the columns: ID, Description, GeneRatio, BgRatio, pvalue, p.adjust,
qvalue, geneID and Count.
</p>

<ul>
<li><p> The <code>ID</code>, <code>qvalue</code> and <code>Count</code> columns are optional.
</p>
</li>
<li><p> The <code>Description</code> is the description of the term.
</p>
</li>
<li><p> The <code>GeneRatio</code> is the number of genes in the term divided by the total number of genes in the input list.
</p>
</li>
<li><p> The <code>BgRatio</code> is the number of genes in the term divided by the total number of genes in the background list (all terms).
</p>
</li>
<li><p> The <code>Count</code> column, if given, should be the same as the first number in GeneRatio.
</p>
</li></ul>

<p>If you have enrichment results from multiple databases, you can combine them into one data frame and add a column (e.g. Database)
to indicate the database.
You can plot them in a single plot using the <code>split_by</code> argument (e.g. <code>split_by = "Database"</code>).</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_n_input">n_input</code></td>
<td>
<p>An integer specifying the number of input genes.
Enrichr result doesn't ship with the number of input genes.
You can either provide the number directly or we will infer it. See details.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_top_term">top_term</code></td>
<td>
<p>An integer specifying the number of top terms to show.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_metric">metric</code></td>
<td>
<p>A character string specifying the metric to use for the size of the nodes.
It is also used to order the terms when selected the top terms.
Either &quot;pvalue&quot; or &quot;p.adjust&quot;. The default is &quot;p.adjust&quot;.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_layout">layout</code></td>
<td>
<p>A character string specifying the layout of the graph.
Either &quot;circle&quot;, &quot;tree&quot;, &quot;grid&quot; or other layout functions in <code>igraph</code>.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_minchar">minchar</code></td>
<td>
<p>An integer specifying the minimum number of characters to show in the keyword.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_cluster">cluster</code></td>
<td>
<p>A character string specifying the clustering method.
Either &quot;fast_greedy&quot;, &quot;walktrap&quot;, &quot;edge_betweenness&quot;, &quot;infomap&quot; or other clustering functions in <code>igraph</code>.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_show_keyword">show_keyword</code></td>
<td>
<p>A logical value specifying whether to show the keyword instead of Description/Term in the plot.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_nlabel">nlabel</code></td>
<td>
<p>An integer specifying the number of labels to show in each cluster.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_character_width">character_width</code></td>
<td>
<p>The width of the characters used to wrap the keyword.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_mark">mark</code></td>
<td>
<p>A character string specifying the mark to use for the nodes.
Either &quot;ellipse&quot;, &quot;rect&quot;, &quot;circle&quot;, &quot;text&quot; or other mark functions in <code>ggforce</code>.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_label">label</code></td>
<td>
<p>A character string specifying the label to show in the legend.
Either &quot;term&quot; or &quot;feature&quot;. The default is &quot;term&quot;.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_labelsize">labelsize</code></td>
<td>
<p>A numeric value specifying the size of the label.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_layoutadjust">layoutadjust</code></td>
<td>
<p>A logical value specifying whether to adjust the layout of the network.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_adjscale">adjscale</code></td>
<td>
<p>A numeric value specifying the scale of the adjustment.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_adjiter">adjiter</code></td>
<td>
<p>A numeric value specifying the number of iterations for the adjustment.</p>
</td></tr>
<tr><td><code id="PrepareEnrichrResult_+3A_blendmode">blendmode</code></td>
<td>
<p>A character string specifying the blend mode of the colors.
Either &quot;blend&quot;, &quot;average&quot;, &quot;multiply&quot; and &quot;screen&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to use the <code>EnrichMap</code> and <code>EnrichNetwork</code> functions and other visualization functions in <code>plotthis</code>,
the enrichment results from Enrichr need to be processed by the <code>PrepareEnrichrResult</code> function.
The following columns are renamed:
</p>

<ul>
<li> <p><code>Term</code> -&gt; <code>Description</code>
</p>
</li>
<li> <p><code>Genes</code> -&gt; <code>geneID</code> (separated replaced by <code>/</code>)
</p>
</li>
<li> <p><code>P.value</code> -&gt; <code>pvalue</code>
</p>
</li>
<li> <p><code>Adjusted.P.value</code> -&gt; <code>p.adjust</code>
Additionally, GeneRatio and BgRatio columns are inferred.
From <a href="https://maayanlab.cloud/Enrichr/help#background">enrichr's documentation</a>, the oddsRatio is defined as:
<code style="white-space: pre;">&#8288;oddsRatio = (A * (D - B - C + A) / max((B - A) * (C - A), 1)&#8288;</code>, where A is the overlapping genes; B is the total genes in the gene set;
C (n_input) is the genes in input list; D is the total genes in the background.
D is not provided by Enrichr. To infer it, <code>D = oddsRatio * max((B - A) * (C - A), 1) / A + B + C - A</code>.
</p>
</li>
<li> <p><code>Overlap = A / B</code> (from Enrichr)
</p>
</li>
<li> <p><code>GeneRatio = A / C</code> (from ClusterProfiler)
</p>
</li>
<li> <p><code>BgRatio = B / D</code> (from ClusterProfiler)
<code>C (n_input)</code>, if not provided, will be inferred when <code>D</code> for all terms are equal.
When starting inferrence, the minimum value to try will be unique genes in <code>data$Genes</code>/<code>data$geneID</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A data frame that can be used in <code>EnrichMap</code>.
</p>
<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(enrich_example)
EnrichMap(enrich_example)
EnrichMap(enrich_example, label = "feature")
EnrichMap(enrich_example, show_keyword = TRUE, label = "term")
EnrichMap(enrich_example, show_keyword = TRUE, label = "feature")

data(enrich_multidb_example)
EnrichMap(enrich_multidb_example, split_by = "Database")
EnrichMap(enrich_multidb_example, split_by = "Database",
          palette = list(DB1 = "Paired", DB2 = "Set1"))
EnrichNetwork(enrich_example, top_term = 5)
</code></pre>

<hr>
<h2 id='PrepareFGSEAResult'>GSEA plots</h2><span id='topic+PrepareFGSEAResult'></span><span id='topic+GSEASummaryPlot'></span><span id='topic+GSEAPlot'></span>

<h3>Description</h3>


<ul>
<li> <p><code>GSEASummaryPlot</code> is used to plot a summary of the results of a GSEA analysis.
</p>
</li>
<li> <p><code>GSEAPlot</code> is used to plot the results of a GSEA analysis.
</p>
</li>
<li> <p><code>PrepareFGSEA</code> is used to prepare the result of GSEA from the <code>fgsea</code> package for plotting.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>PrepareFGSEAResult(data, gene_ranks, gene_sets)

GSEASummaryPlot(
  data,
  gene_ranks = "@gene_ranks",
  gene_sets = "@gene_sets",
  top_term = 10,
  metric = "p.adjust",
  cutoff = 0.05,
  character_width = 50,
  line_plot_size = 0.25,
  metric_name = paste0("-log10(", metric, ")"),
  nonsig_name = "Non-significant",
  linewidth = 0.2,
  line_by = c("prerank", "running_score"),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  alpha = 0.6,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  seed = 8525,
  ...
)

GSEAPlot(
  data,
  gene_ranks = "@gene_ranks",
  gene_sets = "@gene_sets",
  sample_coregenes = FALSE,
  line_width = 1.5,
  line_alpha = 1,
  line_color = "#6BB82D",
  n_coregenes = 10,
  genes_label = NULL,
  label_fg = "black",
  label_bg = "white",
  label_bg_r = 0.1,
  label_size = 4,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PrepareFGSEAResult_+3A_data">data</code></td>
<td>
<p>A data frame of GSEA results
For example, from <code>DOSE::gseDO()</code>.
Required columns are <code>ID</code>, <code>Description</code>, <code>NES</code>, <code>p.adjust</code>, <code>pvalue</code>.
The <code>ID</code> column is used to match the gene sets.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_gene_ranks">gene_ranks</code></td>
<td>
<p>A numeric vector of gene ranks with genes as names
The gene ranks are used to plot the gene sets.
If <code>gene_ranks</code> is a character vector starting with <code>@</code>, the gene ranks will be taken from the attribute of <code>data</code>.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_gene_sets">gene_sets</code></td>
<td>
<p>A list of gene sets, typically from a record of a GMT file
The names of the list should match the <code>ID</code> column of <code>data</code>.
If <code>gene_sets</code> is a character vector starting with <code>@</code>, the gene sets will be taken from the attribute of <code>data</code>.
The GSEA plots will be plotted for each gene set. So, the number of plots will be the number of gene sets.
If you only want to plot a subset of gene sets, you can subset the <code>gene_sets</code> before passing it to this function.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_top_term">top_term</code></td>
<td>
<p>An integer to select the top terms</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_metric">metric</code></td>
<td>
<p>The metric to use for the significance of the terms</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff for the significance of the terms
The terms will not be filtered with this cutoff; they are only filtered by the <code>top_term</code> ranked by the <code>metric</code>.
The cutoff here is used to show the significance of the terms on the plot.
For the terms that are not significant, the color will be grey.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_character_width">character_width</code></td>
<td>
<p>The width of the characters in the y-axis</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_line_plot_size">line_plot_size</code></td>
<td>
<p>The size of the line plots</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_metric_name">metric_name</code></td>
<td>
<p>The name of the metric to show in the color bar</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_nonsig_name">nonsig_name</code></td>
<td>
<p>The name of the legend for the nonsignificant terms</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_linewidth">linewidth</code></td>
<td>
<p>The width of the lines in the line plots</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_line_by">line_by</code></td>
<td>
<p>The method to calculate the line plots.
</p>

<ul>
<li> <p><code>prerank</code>: Use the gene ranks as heights to plot the line plots.
</p>
</li>
<li> <p><code>running_score</code>: Use the running score to plot the line plots.
</p>
</li></ul>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_sample_coregenes">sample_coregenes</code></td>
<td>
<p>A logical value to sample the core genes from the core_enrichment; if <code>FALSE</code>, the first <code>n_coregenes</code> will be used</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_line_width">line_width</code></td>
<td>
<p>The width of the line in the running score plot</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_line_alpha">line_alpha</code></td>
<td>
<p>The alpha of the line in the running score plot</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_line_color">line_color</code></td>
<td>
<p>The color of the line in the running score plot</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_n_coregenes">n_coregenes</code></td>
<td>
<p>The number of core genes to label</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_genes_label">genes_label</code></td>
<td>
<p>The genes to label. If set, <code>n_coregenes</code> will be ignored</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_label_fg">label_fg</code></td>
<td>
<p>The color of the label text</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_label_bg">label_bg</code></td>
<td>
<p>The background color of the label</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>The radius of the background color of the label</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_label_size">label_size</code></td>
<td>
<p>The size of the label text</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="PrepareFGSEAResult_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the desired columns for plotting and the gene ranks and gene sets as attributes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(gsea_example)
GSEASummaryPlot(gsea_example)
GSEASummaryPlot(gsea_example, line_by = "running_score")
GSEASummaryPlot(gsea_example, cutoff = 0.01)


GSEAPlot(gsea_example, gene_sets = attr(gsea_example, "gene_sets")[1])
GSEAPlot(gsea_example, gene_sets = attr(gsea_example, "gene_sets")[1:4])

</code></pre>

<hr>
<h2 id='PrepareUpsetData'>Upset Plot</h2><span id='topic+PrepareUpsetData'></span><span id='topic+UpsetPlot'></span>

<h3>Description</h3>

<p><code>PrepareUpsetData</code> is used to process the input data for Upset plot.
<code>UpsetPlot</code> is used to plot the processed data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrepareUpsetData(
  data,
  in_form = NULL,
  group_by = NULL,
  group_by_sep = "_",
  id_by = NULL,
  specific = TRUE
)

UpsetPlot(
  data,
  in_form = NULL,
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  id_by = NULL,
  label = TRUE,
  label_fg = "black",
  label_size = NULL,
  label_bg = "white",
  label_bg_r = 0.1,
  palette = "material-indigo",
  palcolor = NULL,
  alpha = 1,
  specific = TRUE,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  aspect.ratio = 0.6,
  legend.position = "right",
  legend.direction = "vertical",
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PrepareUpsetData_+3A_data">data</code></td>
<td>
<p>A data frame or a list or an UpsetPlotData object.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_in_form">in_form</code></td>
<td>
<p>A character string indicating the datatype of the input data.
Possible values are &quot;long&quot;, &quot;wide&quot;, &quot;list&quot;, &quot;upset&quot; or NULL.
&quot;long&quot; indicates the data is in long format.
&quot;wide&quot; indicates the data is in wide format.
&quot;list&quot; indicates the data is a list.
&quot;upset&quot; indicates the data is a UpsetPlotData object.
NULL indicates the function will detect the datatype of the input data.
</p>
<p>A long format data would look like:
</p>
<pre>
group_by id_by
A        a1
A        a2
B        a1
B        a3
...
</pre>
<p>A wide format data would look like:
</p>
<pre>
A    B
TRUE TRUE
TRUE FALSE
FALSE TRUE
...
</pre>
<p>A list format data would look like:
</p>
<pre>
list(A = c("a1", "a2"), B = c("a1", "a3"))
</pre>
<p>An UpsetPlotData object is generated by PrepareUpsetData() would look like:
</p>
<pre>
group_by
--------
list("A")  # a2
list("B")  # a3
list(c("A", "B"))  # a1
...
</pre></td></tr>
<tr><td><code id="PrepareUpsetData_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to group the data.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>,
if multiple columns are provided and the in_form is &quot;long&quot;.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_id_by">id_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to identify the instances.
Required when <code>group_by</code> is a single column and data is a data frame.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_specific">specific</code></td>
<td>
<p>A logical value to show the specific intersections only.
ggVennDiagram, by default, only return the specific subsets of a region. However,
sometimes, we want to show all the overlapping items for two or more sets.
See <a href="https://github.com/gaospecial/ggVennDiagram/issues/64">https://github.com/gaospecial/ggVennDiagram/issues/64</a> for more details.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_label">label</code></td>
<td>
<p>A logical value to show the labels on the bars.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string specifying the color of the label text.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the label text.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string specifying the background color of the label.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the radius of the background of the label.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="PrepareUpsetData_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UpsetPlotData object
</p>
<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- list(
    A = 1:5,
    B = 2:6,
    C = 3:7,
    D = 4:8
)
UpsetPlot(data)
UpsetPlot(data, label = FALSE)
UpsetPlot(data, palette = "Reds", specific = FALSE)
</code></pre>

<hr>
<h2 id='PrepareVennData'>VennDiagram</h2><span id='topic+PrepareVennData'></span><span id='topic+VennDiagram'></span>

<h3>Description</h3>

<p><code>PrepareVennData</code> is helper function to process the input data for Venn diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrepareVennData(
  data,
  in_form = NULL,
  group_by = NULL,
  group_by_sep = "_",
  id_by = NULL
)

VennDiagram(
  data,
  in_form = NULL,
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  id_by = NULL,
  label = "count",
  label_fg = "black",
  label_size = NULL,
  label_bg = "white",
  label_bg_r = 0.1,
  fill_mode = "count",
  fill_name = NULL,
  palette = ifelse(fill_mode == "set", "Paired", "Spectral"),
  palcolor = NULL,
  alpha = 1,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PrepareVennData_+3A_data">data</code></td>
<td>
<p>A data frame or a list or a VennPlotData object.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_in_form">in_form</code></td>
<td>
<p>A character string indicating the datatype of the input data.
Possible values are &quot;long&quot;, &quot;wide&quot;, &quot;list&quot;, &quot;venn&quot; or NULL.
&quot;long&quot; indicates the data is in long format.
&quot;wide&quot; indicates the data is in wide format.
&quot;list&quot; indicates the data is a list.
&quot;venn&quot; indicates the data is a VennPlotData object.
NULL indicates the function will detect the datatype of the input data.
</p>
<p>A long format data would look like:
</p>
<pre>
group_by id_by
A        a1
A        a2
B        a1
B        a3
...
</pre>
<p>A wide format data would look like:
</p>
<pre>
A    B
TRUE TRUE
TRUE FALSE
FALSE TRUE
...
</pre>
<p>A list format data would look like:
</p>
<pre>
list(A = c("a1", "a2"), B = c("a1", "a3"))
</pre></td></tr>
<tr><td><code id="PrepareVennData_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to group the data.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>,
if multiple columns are provided and the in_form is &quot;long&quot;.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_id_by">id_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to identify the instances.
Required when <code>group_by</code> is a single column and data is a data frame.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_label">label</code></td>
<td>
<p>A character string specifying the label to show on the Venn diagram.
Possible values are &quot;count&quot;, &quot;percent&quot;, &quot;both&quot;, &quot;none&quot; and a function.
&quot;count&quot; indicates the count of the intersection. &quot;percent&quot; indicates the percentage of the intersection.
&quot;both&quot; indicates both the count and the percentage of the intersection. &quot;none&quot; indicates no label.
If it is a function, if takes a data frame as input and returns a character vector as label.
The data frame has columns &quot;id&quot;, &quot;X&quot;, &quot;Y&quot;, &quot;name&quot;, &quot;item&quot; and &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string specifying the color of the label text.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the label text.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string specifying the background color of the label.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the radius of the background of the label.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_fill_mode">fill_mode</code></td>
<td>
<p>A character string specifying the fill mode of the Venn diagram.
Possible values are &quot;count&quot;, &quot;set&quot;, &quot;count_rev&quot;.
&quot;count&quot; indicates the fill color is based on the count of the intersection.
&quot;set&quot; indicates the fill color is based on the set of the intersection.
&quot;count_rev&quot; indicates the fill color is based on the count of the intersection in reverse order.
The palette will be continuous for &quot;count&quot; and &quot;count_rev&quot;. The palette will be discrete for &quot;set&quot;.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_fill_name">fill_name</code></td>
<td>
<p>A character string to name the legend of colorbar.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="PrepareVennData_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A VennPlotData object
</p>
<p>A combined ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
data = list(
    A = sort(sample(letters, 8)),
    B = sort(sample(letters, 8)),
    C = sort(sample(letters, 8)),
    D = sort(sample(letters, 8))
)

VennDiagram(data)
VennDiagram(data, fill_mode = "set")
VennDiagram(data, label = "both")
# label with a function
VennDiagram(data, label = function(df) df$name)
VennDiagram(data, palette = "material-indigo", alpha = 0.6)

</code></pre>

<hr>
<h2 id='process_theme'>Process theme to allow 'ggplot2::theme_minimal' to work</h2><span id='topic+process_theme'></span>

<h3>Description</h3>

<p>Process theme to allow 'ggplot2::theme_minimal' to work
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_theme(theme)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process_theme_+3A_theme">theme</code></td>
<td>
<p>The theme to process</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The processed theme
</p>

<hr>
<h2 id='QQPlot'>QQ plot</h2><span id='topic+QQPlot'></span>

<h3>Description</h3>

<p>QQ plot is a graphical tool to compare two distributions by plotting their quantiles against each other.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QQPlot(
  data,
  val,
  val_trans = NULL,
  type = c("qq", "pp"),
  split_by = NULL,
  split_by_sep = "_",
  band = NULL,
  line = list(),
  point = list(),
  fill_name = "Bands",
  band_alpha = 0.5,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlim = NULL,
  ylim = NULL,
  xlab = ifelse(type == "qq", "Theoretical Quantiles", "Probability Points"),
  ylab = ifelse(type == "qq", "Sample Quantiles", "Cumulative Probability"),
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QQPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_val">val</code></td>
<td>
<p>A character string of the column name for the values to plot.
A numeric column is expected.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_val_trans">val_trans</code></td>
<td>
<p>A function to transform the values before plotting.
Default is NULL, which means no transformation.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_type">type</code></td>
<td>
<p>A character string to specify the type of plot.
Default is &quot;qq&quot;, which means QQ plot.
Other options are &quot;pp&quot;, which means PP plot.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="QQPlot_+3A_band">band</code></td>
<td>
<p>A list of arguments to pass to <code><a href="qqplotr.html#topic+stat_qq_band">qqplotr::stat_qq_band()</a></code> or
<code><a href="qqplotr.html#topic+stat_pp_band">qqplotr::stat_pp_band()</a></code>, depending on the value of <code>type</code>.
Default is NULL, which means no band.
If an empty list or TRUE is provided, the default arguments will be used.
Multiple bands can be added by providing a list of lists.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_line">line</code></td>
<td>
<p>A list of arguments to pass to <code><a href="qqplotr.html#topic+stat_qq_line">qqplotr::stat_qq_line()</a></code> or
<code><a href="qqplotr.html#topic+stat_pp_line">qqplotr::stat_pp_line()</a></code>, depending on the value of <code>type</code>.
Default is <code>list()</code>, which means to add a line with default arguments.
If <code>NULL</code> is provided, no line will be added.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_point">point</code></td>
<td>
<p>A list of arguments to pass to <code><a href="qqplotr.html#topic+stat_qq_point">qqplotr::stat_qq_point()</a></code> or
<code><a href="qqplotr.html#topic+stat_pp_point">qqplotr::stat_pp_point()</a></code>, depending on the value of <code>type</code>.
Default is <code>list()</code>, which means to add points with default arguments.
If <code>NULL</code> is provided, no points will be added (not recommended).</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_fill_name">fill_name</code></td>
<td>
<p>A character string to name the legend of fill.
Default is &quot;Band Type&quot;.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_band_alpha">band_alpha</code></td>
<td>
<p>A numeric value to set the alpha of all bands.
Default is 0.5.
It is a shortcut for setting alpha of all bands. You can override it by
setting <code>alpha</code> in <code>band</code> argument.
For example, <code>band = list(list(alpha = 0.3), list(alpha = 0.7))</code>.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="QQPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="QQPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_xlim">xlim</code></td>
<td>
<p>A numeric vector of length 2 to set the x-axis limits.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_ylim">ylim</code></td>
<td>
<p>A numeric vector of length 2 to set the y-axis limits.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="QQPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(8525)
data &lt;- data.frame(norm = rnorm(100))

QQPlot(data, val = "norm", band = TRUE)
QQPlot(data, val = "norm", band = list(
    list(bandType = "ks", mapping = ggplot2::aes(fill = "KS"), alpha = 0.3),
    list(bandType = "ts", mapping = ggplot2::aes(fill = "TS")),
    list(bandType = "pointwise", mapping = ggplot2::aes(fill = "Normal")),
    list(bandType = "boot", mapping = ggplot2::aes(fill = "Bootstrap"))
), band_alpha = 0.6)

data(airquality, package = "datasets")
di &lt;- "exp" # exponential distribution
dp &lt;- list(rate = 2) # exponential rate parameter
QQPlot(airquality, val = "Ozone",
    band = list(distribution = di, dparams = dp),
    line = list(distribution = di, dparams = dp),
    point = list(distribution = di, dparams = dp)
)

de &lt;- TRUE # enabling the detrend option
QQPlot(airquality, val = "Ozone",
    band = list(distribution = di, dparams = dp, detrend = de),
    line = list(distribution = di, dparams = dp, detrend = de),
    point = list(distribution = di, dparams = dp, detrend = de)
)

QQPlot(data, val = "norm", type = "pp", band = TRUE)

dp &lt;- list(mean = 2, sd = 2) # shifted and rescaled Normal parameters
QQPlot(data, val = "norm", type = "pp",
    band = list(dparams = dp),
    point = list(dparams = dp))

QQPlot(data, val = "norm", type = "pp", band = TRUE,
    line = list(ab = c(.2, .5)))

di &lt;- "exp"
dp &lt;- list(rate = .022) # value is based on some empirical tests
de &lt;- TRUE
QQPlot(airquality, val = "Ozone", type = "pp",
   band = list(distribution = di, detrend = de, dparams = dp),
   line = list(detrend = de),
   point = list(distribution = di, detrend = de, dparams = dp),
   ylim = c(-.5, .5)
)
</code></pre>

<hr>
<h2 id='QQPlotAtomic'>Atomic QQ plot</h2><span id='topic+QQPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic QQ plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QQPlotAtomic(
  data,
  val,
  val_trans = NULL,
  type = c("qq", "pp"),
  band = NULL,
  line = list(),
  point = list(),
  fill_name = "Bands",
  band_alpha = 0.5,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  xlim = NULL,
  ylim = NULL,
  xlab = ifelse(type == "qq", "Theoretical Quantiles", "Probability Points"),
  ylab = ifelse(type == "qq", "Sample Quantiles", "Cumulative Probability"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QQPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_val">val</code></td>
<td>
<p>A character string of the column name for the values to plot.
A numeric column is expected.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_val_trans">val_trans</code></td>
<td>
<p>A function to transform the values before plotting.
Default is NULL, which means no transformation.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_type">type</code></td>
<td>
<p>A character string to specify the type of plot.
Default is &quot;qq&quot;, which means QQ plot.
Other options are &quot;pp&quot;, which means PP plot.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_band">band</code></td>
<td>
<p>A list of arguments to pass to <code><a href="qqplotr.html#topic+stat_qq_band">qqplotr::stat_qq_band()</a></code> or
<code><a href="qqplotr.html#topic+stat_pp_band">qqplotr::stat_pp_band()</a></code>, depending on the value of <code>type</code>.
Default is NULL, which means no band.
If an empty list or TRUE is provided, the default arguments will be used.
Multiple bands can be added by providing a list of lists.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_line">line</code></td>
<td>
<p>A list of arguments to pass to <code><a href="qqplotr.html#topic+stat_qq_line">qqplotr::stat_qq_line()</a></code> or
<code><a href="qqplotr.html#topic+stat_pp_line">qqplotr::stat_pp_line()</a></code>, depending on the value of <code>type</code>.
Default is <code>list()</code>, which means to add a line with default arguments.
If <code>NULL</code> is provided, no line will be added.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_point">point</code></td>
<td>
<p>A list of arguments to pass to <code><a href="qqplotr.html#topic+stat_qq_point">qqplotr::stat_qq_point()</a></code> or
<code><a href="qqplotr.html#topic+stat_pp_point">qqplotr::stat_pp_point()</a></code>, depending on the value of <code>type</code>.
Default is <code>list()</code>, which means to add points with default arguments.
If <code>NULL</code> is provided, no points will be added (not recommended).</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_fill_name">fill_name</code></td>
<td>
<p>A character string to name the legend of fill.
Default is &quot;Band Type&quot;.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_band_alpha">band_alpha</code></td>
<td>
<p>A numeric value to set the alpha of all bands.
Default is 0.5.
It is a shortcut for setting alpha of all bands. You can override it by
setting <code>alpha</code> in <code>band</code> argument.
For example, <code>band = list(list(alpha = 0.3), list(alpha = 0.7))</code>.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>A numeric value to set the seed for random number generation.
Default is 8525.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_xlim">xlim</code></td>
<td>
<p>A numeric vector of length 2 to set the x-axis limits.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_ylim">ylim</code></td>
<td>
<p>A numeric vector of length 2 to set the y-axis limits.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="QQPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='RadarPlot'>Radar plot / Spider plot</h2><span id='topic+RadarPlot'></span><span id='topic+SpiderPlot'></span>

<h3>Description</h3>

<p>Create a radar plot or spider plot for a series of data.
Radar plot uses circles as the plot grid and Spider plot uses polygons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RadarPlot(
  data,
  x,
  x_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  y = NULL,
  group_name = NULL,
  scale_y = c("group", "global", "x", "none"),
  y_min = 0,
  y_max = NULL,
  y_nbreaks = 4,
  fill = TRUE,
  linewidth = 1,
  pt_size = 4,
  max_charwidth = 16,
  split_by = NULL,
  split_by_sep = "_",
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  alpha = 0.2,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

SpiderPlot(
  data,
  x,
  x_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  y = NULL,
  group_name = NULL,
  scale_y = c("group", "global", "x", "none"),
  y_min = 0,
  y_max = NULL,
  y_nbreaks = 4,
  fill = TRUE,
  linewidth = 1,
  pt_size = 4,
  max_charwidth = 16,
  split_by = NULL,
  split_by_sep = "_",
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  alpha = 0.2,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RadarPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of group.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_scale_y">scale_y</code></td>
<td>
<p>How should the y-axis be scaled? Default is &quot;group&quot;.
Other options are &quot;global&quot;, &quot;x&quot; and &quot;none&quot;.
</p>

<ul>
<li><p> If &quot;group&quot;, the y-axis will be scaled to the fraction within each group.
</p>
</li>
<li><p> If &quot;global&quot;, the y-axis will be scaled to the fraction of the total.
</p>
</li>
<li><p> If &quot;x&quot;, the y-axis will be scaled to the fraction of the total within each x-axis group.
</p>
</li>
<li><p> If &quot;none&quot;, the y-axis will be scaled to the count of each x-axis group.
</p>
</li></ul>
</td></tr>
<tr><td><code id="RadarPlot_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value to set the minimum value of the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value to set the maximum value of the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_y_nbreaks">y_nbreaks</code></td>
<td>
<p>A numeric value to set the number of breaks in the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_fill">fill</code></td>
<td>
<p>A logical value to fill the polygons with colors.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_linewidth">linewidth</code></td>
<td>
<p>A numeric value to set the width of the lines.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to set the size of the points.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_max_charwidth">max_charwidth</code></td>
<td>
<p>A numeric value to set the maximum character width for the x labels.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="RadarPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'># use the count
data &lt;- data.frame(
   x = c(rep("A", 2), rep("B", 3), rep("C", 3), rep("D", 4), rep("E", 5)),
   group = sample(paste0("G", 1:4), 17, replace = TRUE)
)
RadarPlot(data, x = "x")
RadarPlot(data, x = "x", scale_y = "none")
RadarPlot(data, x = "x", group_by = "group")
SpiderPlot(data, x = "x")
SpiderPlot(data, x = "x", group_by = "group")

# use the y value
data &lt;- data.frame(
   x = rep(LETTERS[1:5], 2),
   y = c(1, 3, 6, 4, 2, 5, 7, 8, 9, 10),
   group = rep(c("G1", "G2"), each = 5)
)
RadarPlot(data, x = "x", y = "y", scale_y = "none", group_by = "group")
RadarPlot(data, x = "x", y = "y", facet_by = "group")
RadarPlot(data, x = "x", y = "y", split_by = "group")
RadarPlot(data, x = "x", y = "y", split_by = "group",
          palette = c(G1 = "Set1", G2 = "Paired"))
</code></pre>

<hr>
<h2 id='RadarPlotAtomic'>Atomic Radar plot</h2><span id='topic+RadarPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic Radar plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RadarPlotAtomic(
  data,
  x,
  x_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  y = NULL,
  group_name = NULL,
  scale_y = c("group", "global", "x", "none"),
  y_min = 0,
  y_max = NULL,
  y_nbreaks = 4,
  polygon = FALSE,
  fill = TRUE,
  linewidth = 1,
  pt_size = 4,
  max_charwidth = 16,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  alpha = 0.2,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RadarPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string of the column name to plot on the x-axis/circles.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string of the column name(s) to group the data (the lines) by.
Character/factor column(s) is expected.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis.
A numeric column is expected.
If NULL, the count of the x-axis column in each group will be used.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of group.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_scale_y">scale_y</code></td>
<td>
<p>How should the y-axis be scaled? Default is &quot;group&quot;.
Other options are &quot;global&quot;, &quot;x&quot; and &quot;none&quot;.
</p>

<ul>
<li><p> If &quot;group&quot;, the y-axis will be scaled to the fraction within each group.
</p>
</li>
<li><p> If &quot;global&quot;, the y-axis will be scaled to the fraction of the total.
</p>
</li>
<li><p> If &quot;x&quot;, the y-axis will be scaled to the fraction of the total within each x-axis group.
</p>
</li>
<li><p> If &quot;none&quot;, the y-axis will be scaled to the count of each x-axis group.
</p>
</li></ul>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_y_min">y_min</code></td>
<td>
<p>A numeric value to set the minimum value of the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_y_max">y_max</code></td>
<td>
<p>A numeric value to set the maximum value of the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_y_nbreaks">y_nbreaks</code></td>
<td>
<p>A numeric value to set the number of breaks in the y-axis.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_polygon">polygon</code></td>
<td>
<p>A logical value to draw the polygons instead of the circles as panel grid.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_fill">fill</code></td>
<td>
<p>A logical value to fill the polygons with colors.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_linewidth">linewidth</code></td>
<td>
<p>A numeric value to set the width of the lines.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to set the size of the points.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_max_charwidth">max_charwidth</code></td>
<td>
<p>A numeric value to set the maximum character width for the x labels.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RadarPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='RarefactionPlot'>RarefactionPlot</h2><span id='topic+RarefactionPlot'></span>

<h3>Description</h3>

<p>This function generates a rarefraction plot for a given dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RarefactionPlot(
  data,
  type = 1,
  se = NULL,
  group_by = "group",
  group_by_sep = "_",
  group_name = NULL,
  split_by = NULL,
  split_by_sep = "_",
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 0.2,
  pt_size = 3,
  line_width = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RarefactionPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_type">type</code></td>
<td>
<p>three types of plots: sample-size-based rarefaction/extrapolation curve (<code>type = 1</code>);
sample completeness curve (<code>type = 2</code>); coverage-based rarefaction/extrapolation curve (<code>type = 3</code>).</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_se">se</code></td>
<td>
<p>a logical variable to display confidence interval around the estimated sampling curve.
Default to <code>NULL</code> which means TRUE if the data has the lower and upper bounds.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string indicating how to separate the group_by column if both &quot;q&quot; and &quot;group&quot; are used.
for 'group_by'. Default to &quot;_&quot;.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string indicating the name of the group, showing as the legend title.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_split_by">split_by</code></td>
<td>
<p>A character string indicating how to split the data and plots
Possible values are &quot;q&quot; and &quot;group&quot;</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value specifying the size of the points.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value specifying the width of the lines.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="RarefactionPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
spider &lt;- list(
   Girdled = c(46, 22, 17, 15, 15, 9, 8, 6, 6, 4, rep(2, 4), rep(1, 12)),
   Logged = c(88, 22, 16, 15, 13, 10, 8, 8, 7, 7, 7, 5, 4, 4, 4, 3, 3, 3, 3,
     2, 2, 2, 2, rep(1, 14))
)

RarefactionPlot(spider)
RarefactionPlot(spider, q = c(0, 1, 2), facet_by = "q")
RarefactionPlot(spider, q = c(0, 1, 2), split_by = "q")
RarefactionPlot(spider, q = c(0, 1, 2), split_by = "q",
                palette = c("0" = "Paired", "1" = "Set1", "2" = "Dark2"))
RarefactionPlot(spider, q = c(0, 1, 2), group_by = "q",
 facet_by = "group", palette = "Set1", type = 3)

</code></pre>

<hr>
<h2 id='RarefactionPlotAtomic'>RarefactionPlotAtomic</h2><span id='topic+RarefactionPlotAtomic'></span>

<h3>Description</h3>

<p>This function generates a rarefraction plot for a given dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RarefactionPlotAtomic(
  data,
  type = 1,
  se = TRUE,
  group_by = "group",
  group_name = NULL,
  pt_size = 3,
  line_width = 1,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 0.2,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RarefactionPlotAtomic_+3A_data">data</code></td>
<td>
<p>An iNEXT object or a list of data that will be handled by <a href="iNEXT.html#topic+iNEXT">iNEXT::iNEXT</a>.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_type">type</code></td>
<td>
<p>three types of plots: sample-size-based rarefaction/extrapolation curve (<code>type = 1</code>);
sample completeness curve (<code>type = 2</code>); coverage-based rarefaction/extrapolation curve (<code>type = 3</code>).</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_se">se</code></td>
<td>
<p>a logical variable to display confidence interval around the estimated sampling curve.
Default to <code>NULL</code> which means TRUE if the data has the lower and upper bounds.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string indicating how to group the data (color the lines).
Possible values are &quot;q&quot; and &quot;group&quot;</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string indicating the name of the group, showing as the legend title.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value specifying the size of the points.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_line_width">line_width</code></td>
<td>
<p>A numeric value specifying the width of the lines.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string indicating how to facet the data and plots
Possible values are &quot;q&quot; and &quot;group&quot;</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="RarefactionPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <a href="iNEXT.html#topic+iNEXT">iNEXT::iNEXT</a> when <code>data</code> is not an iNEXT object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='rgba_to_rgb'>Convert RGBA to RGB</h2><span id='topic+rgba_to_rgb'></span>

<h3>Description</h3>

<p>Convert RGBA to RGB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rgba_to_rgb(RGBA, BackGround = c(1, 1, 1))
</code></pre>

<hr>
<h2 id='RidgePlot'>Ridge Plot</h2><span id='topic+RidgePlot'></span>

<h3>Description</h3>

<p>Ridge plot to illustrate the distribution of the data in different groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgePlot(
  data,
  x = NULL,
  in_form = c("long", "wide"),
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  scale = NULL,
  flip = FALSE,
  alpha = 1,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  x_text_angle = 90,
  keep_empty = FALSE,
  reverse = FALSE,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "none",
  legend.direction = "vertical",
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RidgePlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_in_form">in_form</code></td>
<td>
<p>A character string specifying the form of the data. Default is &quot;long&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of 'group_by', if 'legend.position' is not &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_scale">scale</code></td>
<td>
<p>A numeric value to scale the ridges.
See also <code><a href="ggridges.html#topic+geom_density_ridges">geom_density_ridges</a></code>.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_flip">flip</code></td>
<td>
<p>A logical value. If TRUE, the plot will be flipped.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_reverse">reverse</code></td>
<td>
<p>A logical value. If TRUE, reverse the order of the groups on the y-axis.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="RidgePlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects.
If no <code>split_by</code> is provided, a single plot (ggplot object) will be returned.
If 'combine' is TRUE, a wrap_plots object will be returned.
If 'combine' is FALSE, a list of ggplot objects will be returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(8525)
data &lt;- data.frame(
   x = c(rnorm(250, -1), rnorm(250, 1)),
   group = rep(LETTERS[1:5], each = 100)
)
RidgePlot(data, x = "x", group_by = "group")
RidgePlot(data, x = "x", group_by = "group", reverse = TRUE)

# wide form
data_wide &lt;- data.frame(
   A = rnorm(100),
   B = rnorm(100),
   C = rnorm(100),
   D = rnorm(100),
   E = rnorm(100),
   group = sample(letters[1:4], 100, replace = TRUE)
)
RidgePlot(data_wide, group_by = LETTERS[1:5], in_form = "wide")
RidgePlot(data_wide, group_by = LETTERS[1:5], in_form = "wide", facet_by = "group")
RidgePlot(data_wide, group_by = LETTERS[1:5], in_form = "wide", split_by = "group",
   palette = list(a = "Reds", b = "Blues", c = "Greens", d = "Purples"))
</code></pre>

<hr>
<h2 id='RidgePlotAtomic'>Atomic ridge plot</h2><span id='topic+RidgePlotAtomic'></span>

<h3>Description</h3>

<p>Atomic ridge plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgePlotAtomic(
  data,
  x = NULL,
  in_form = c("long", "wide"),
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  flip = FALSE,
  alpha = 1,
  scale = NULL,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  x_text_angle = 90,
  keep_empty = FALSE,
  reverse = FALSE,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "none",
  legend.direction = "vertical",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RidgePlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame
It has two forms: wide and long.
For the wide form, the values should under different 'group_by' columns.
For the long form, the values should be under the 'x' column and the 'group_by' column should be provided,
which should be a single column with the group names.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name for the values
A numeric column is expected.
If 'data' is in the wide form, 'x' should be NULL. The values will be taken from the data under 'group_by' columns.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_in_form">in_form</code></td>
<td>
<p>A character string specifying the form of the data. Default is &quot;long&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name to group the data
These groups will be shown on the y-axis.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code> if multiple columns are provided
If 'data' is in the wide form, the columns will not be concatenated.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of 'group_by', if 'legend.position' is not &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value. If TRUE, the plot will be flipped.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the alpha of the ridges.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_scale">scale</code></td>
<td>
<p>A numeric value to scale the ridges.
See also <code><a href="ggridges.html#topic+geom_density_ridges">geom_density_ridges</a></code>.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value. If TRUE, keep the empty groups on the y-axis.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_reverse">reverse</code></td>
<td>
<p>A logical value. If TRUE, reverse the order of the groups on the y-axis.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RidgePlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='RingPlot'>Ring Plot</h2><span id='topic+RingPlot'></span>

<h3>Description</h3>

<p>A ring plot is like pie chart but with multiple rings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RingPlot(
  data,
  x = NULL,
  y = NULL,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  label = NULL,
  split_by = NULL,
  split_by_sep = "_",
  facet_by = NULL,
  facet_scales = "free_y",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RingPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_x">x</code></td>
<td>
<p>A character vector specifying the column as the rings of the plot.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_y">y</code></td>
<td>
<p>A character vector specifying the column as the y axis of the plot.
Default is NULL, meaning the y axis is the count of the data.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_group_by">group_by</code></td>
<td>
<p>A character vector specifying the column as the group_by of the plot.
How the ring is divided.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to specify the name of the group_by in the legend.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_label">label</code></td>
<td>
<p>A logical value indicating whether to show the labels on the rings.
The labels should be the values of group_by. Default is NULL, meaning no labels for one ring and
showing the labels for multiple rings.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="RingPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RingPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RingPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="RingPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PieChart">PieChart</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>RingPlot(datasets::iris, group_by = "Species")

data &lt;- data.frame(
  x = c("A", "B", "C", "A", "B", "C"),
  y = c(1, 2, 3, 4, 5, 6),
  group = c("a", "a", "a", "b", "b", "b")
)
RingPlot(data, x = "x", y = "y", group_by = "group")
RingPlot(datasets::mtcars, x = "cyl", group_by = "carb", facet_by = "vs")
RingPlot(datasets::mtcars, x = "cyl", group_by = "carb", split_by = "vs",
        palette = c("0" = "Set1", "1" = "Paired"))
</code></pre>

<hr>
<h2 id='RingPlotAtomic'>RingPlotAtomic</h2><span id='topic+RingPlotAtomic'></span>

<h3>Description</h3>

<p>Ring plot for atomic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RingPlotAtomic(
  data,
  x = NULL,
  y = NULL,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  label = NULL,
  clockwise = TRUE,
  facet_by = NULL,
  facet_scales = "free_y",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RingPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character vector specifying the column as the rings of the plot.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character vector specifying the column as the y axis of the plot.
Default is NULL, meaning the y axis is the count of the data.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character vector specifying the column as the group_by of the plot.
How the ring is divided.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>group_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to specify the name of the group_by in the legend.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_label">label</code></td>
<td>
<p>A logical value indicating whether to show the labels on the rings.
The labels should be the values of group_by. Default is NULL, meaning no labels for one ring and
showing the labels for multiple rings.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_clockwise">clockwise</code></td>
<td>
<p>A logical value to draw the ring plot clockwise or not.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="RingPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='ROCCurve'>ROC curve</h2><span id='topic+ROCCurve'></span>

<h3>Description</h3>

<p>A wrapped function around <code>plotROC</code> package to create ROC curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ROCCurve(
  data,
  truth_by,
  score_by,
  pos_label = NULL,
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  x_axis_reverse = FALSE,
  percent = FALSE,
  ci = NULL,
  n_cuts = 0,
  cutoffs_at = NULL,
  cutoffs_labels = NULL,
  cutoffs_accuracy = 0.01,
  cutoffs_pt_size = 5,
  cutoffs_pt_shape = 4,
  cutoffs_pt_stroke = 1,
  cutoffs_labal_fg = "black",
  cutoffs_label_size = 4,
  cutoffs_label_bg = "white",
  cutoffs_label_bg_r = 0.1,
  show_auc = c("auto", "none", "legend", "plot"),
  auc_accuracy = 0.01,
  auc_size = 4,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = ifelse(x_axis_reverse, "Specificity", "1 - Specificity"),
  ylab = "Sensitivity",
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ROCCurve_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_truth_by">truth_by</code></td>
<td>
<p>A character string of the column name that contains the true class labels.
(a.k.a. the binary outcome, 1/0 or TRUE/FALSE.)</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_score_by">score_by</code></td>
<td>
<p>character strings of the column names that contains the predicted scores.
When multiple columns are provided, the ROC curve is plotted for each column.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_pos_label">pos_label</code></td>
<td>
<p>A character string of the positive class label.
When NULL, the labels will be handled by the <code>plotROC</code> package.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of the ROC curve groups.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_x_axis_reverse">x_axis_reverse</code></td>
<td>
<p>A logical to reverse the x-axis, that is from 1 to 0.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_percent">percent</code></td>
<td>
<p>A logical to display the x and y axis as percentages.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_ci">ci</code></td>
<td>
<p>A list of arguments to pass to <code><a href="plotROC.html#topic+geom_rocci">plotROC::geom_rocci()</a></code> to add confidence intervals.
When NULL, no confidence intervals are added.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_n_cuts">n_cuts</code></td>
<td>
<p>An integer to specify the number of cutpoints on the ROC curve.
It will be the quantiles of the predicted scores.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_at">cutoffs_at</code></td>
<td>
<p>Vector of user supplied cutoffs to plot as points. If non-NULL,
it will override the values of n_cuts and plot the observed cutoffs closest to the user-supplied ones.
Both <code>cutoffs_at</code> and <code>cutoffs.labels</code> will be passed to <code><a href="plotROC.html#topic+geom_roc">plotROC::geom_roc()</a></code>.
Other than numeric values, the following special values are allowed. These values are the methods of
<code><a href="OptimalCutpoints.html#topic+optimal.cutpoints">OptimalCutpoints::optimal.cutpoints()</a></code>, they are literally:
</p>

<ul>
<li><p> &quot;CB&quot; (cost-benefit method);
</p>
</li>
<li><p> &quot;MCT&quot; (minimizes Misclassification Cost Term);
</p>
</li>
<li><p> &quot;MinValueSp&quot; (a minimum value set for Specificity);
</p>
</li>
<li><p> &quot;MinValueSe&quot; (a minimum value set for Sensitivity);
</p>
</li>
<li><p> &quot;ValueSe&quot; (a value set for Sensitivity);
</p>
</li>
<li><p> &quot;MinValueSpSe&quot; (a minimum value set for Specificity and Sensitivity);
</p>
</li>
<li><p> &quot;MaxSp&quot; (maximizes Specificity);
</p>
</li>
<li><p> &quot;MaxSe&quot; (maximizes Sensitivity);
</p>
</li>
<li><p> &quot;MaxSpSe&quot; (maximizes Sensitivity and Specificity simultaneously);
</p>
</li>
<li><p> &quot;MaxProdSpSe&quot; (maximizes the product of Sensitivity and Specificity or Accuracy Area);
</p>
</li>
<li><p> &quot;ROC01&quot; (minimizes distance between ROC plot and point (0,1));
</p>
</li>
<li><p> &quot;SpEqualSe&quot; (Sensitivity = Specificity);
</p>
</li>
<li><p> &quot;Youden&quot; (Youden Index);
</p>
</li>
<li><p> &quot;MaxEfficiency&quot; (maximizes Efficiency or Accuracy, similar to minimize Error Rate);
</p>
</li>
<li><p> &quot;Minimax&quot; (minimizes the most frequent error);
</p>
</li>
<li><p> &quot;MaxDOR&quot; (maximizes Diagnostic Odds Ratio);
</p>
</li>
<li><p> &quot;MaxKappa&quot; (maximizes Kappa Index);
</p>
</li>
<li><p> &quot;MinValueNPV&quot; (a minimum value set for Negative Predictive Value);
</p>
</li>
<li><p> &quot;MinValuePPV&quot; (a minimum value set for Positive Predictive Value);
</p>
</li>
<li><p> &quot;ValueNPV&quot; (a value set for Negative Predictive Value);
</p>
</li>
<li><p> &quot;ValuePPV&quot; (a value set for Positive Predictive Value);
</p>
</li>
<li><p> &quot;MinValueNPVPPV&quot; (a minimum value set for Predictive Values);
</p>
</li>
<li><p> &quot;PROC01&quot; (minimizes distance between PROC plot and point (0,1));
</p>
</li>
<li><p> &quot;NPVEqualPPV&quot; (Negative Predictive Value = Positive Predictive Value);
</p>
</li>
<li><p> &quot;MaxNPVPPV&quot; (maximizes Positive Predictive Value and Negative Predictive Value simultaneously);
</p>
</li>
<li><p> &quot;MaxSumNPVPPV&quot; (maximizes the sum of the Predictive Values);
</p>
</li>
<li><p> &quot;MaxProdNPVPPV&quot; (maximizes the product of Predictive Values);
</p>
</li>
<li><p> &quot;ValueDLR.Negative&quot; (a value set for Negative Diagnostic Likelihood Ratio);
</p>
</li>
<li><p> &quot;ValueDLR.Positive&quot; (a value set for Positive Diagnostic Likelihood Ratio);
</p>
</li>
<li><p> &quot;MinPvalue&quot; (minimizes p-value associated with the statistical Chi-squared test which measures the association between
the marker and the binary result obtained on using the cutpoint);
</p>
</li>
<li><p> &quot;ObservedPrev&quot; (The closest value to observed prevalence);
</p>
</li>
<li><p> &quot;MeanPrev&quot; (The closest value to the mean of the diagnostic test values);
</p>
</li>
<li><p> &quot;PrevalenceMatching&quot; (The value for which predicted prevalence is practically equal to observed prevalence).
</p>
</li></ul>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_labels">cutoffs_labels</code></td>
<td>
<p>vector of user-supplied labels for the cutoffs. Must be a character vector of the
same length as cutoffs_at.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_accuracy">cutoffs_accuracy</code></td>
<td>
<p>A numeric to specify the accuracy of the cutoff values to show.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_pt_size">cutoffs_pt_size</code></td>
<td>
<p>A numeric to specify the size of the cutoff points.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_pt_shape">cutoffs_pt_shape</code></td>
<td>
<p>A numeric to specify the shape of the cutoff points.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_pt_stroke">cutoffs_pt_stroke</code></td>
<td>
<p>A numeric to specify the stroke of the cutoff points.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_labal_fg">cutoffs_labal_fg</code></td>
<td>
<p>A character string to specify the color of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_label_size">cutoffs_label_size</code></td>
<td>
<p>A numeric to specify the size of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_label_bg">cutoffs_label_bg</code></td>
<td>
<p>A character string to specify the background color of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_cutoffs_label_bg_r">cutoffs_label_bg_r</code></td>
<td>
<p>A numeric to specify the radius of the background of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_show_auc">show_auc</code></td>
<td>
<p>A character string to specify the position of the AUC values.
</p>

<ul>
<li><p> &quot;auto&quot; (default): Automatically determine the position based on the plot.
When there is a single group or 'facet_by' is provided, the AUC is placed on the plot.
Otherwise, the AUC is placed in the legend.
</p>
</li>
<li><p> &quot;none&quot;: Do not display the AUC values.
</p>
</li>
<li><p> &quot;legend&quot;: Display the AUC values in the legend.
</p>
</li>
<li><p> &quot;plot&quot;: Display the AUC values on the plot (left/right bottom corner).
</p>
</li></ul>
</td></tr>
<tr><td><code id="ROCCurve_+3A_auc_accuracy">auc_accuracy</code></td>
<td>
<p>A numeric to specify the accuracy of the AUC values.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_auc_size">auc_size</code></td>
<td>
<p>A numeric to specify the size of the AUC values when they are displayed on the plot.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="ROCCurve_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>patch_work::wrap_plots</code> object or a list of them if <code>combine</code> is <code>FALSE</code>.
You can retrieve the AUC values using <code>attr(p, "auc")</code> if <code>combine</code> is <code>TRUE</code>.
If <code>combine</code> is <code>FALSE</code>, The AUC value of each plot can be retrieved using <code>attr(p[[i]], "auc")</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(8525)

D.ex &lt;- rbinom(200, size = 1, prob = .5)
M1 &lt;- rnorm(200, mean = D.ex, sd = .65)
M2 &lt;- rnorm(200, mean = D.ex, sd = 1.5)
gender &lt;- c("Male", "Female")[rbinom(200, 1, .49) + 1]

data &lt;- data.frame(D = D.ex, D.str = c("Healthy", "Ill")[D.ex + 1],
  gender = gender, M1 = M1, M2 = M2)

ROCCurve(data, truth_by = "D", score_by = "M1")
# will warn about the positive label
ROCCurve(data, truth_by = "D.str", score_by = "M1")
ROCCurve(data, truth_by = "D", score_by = "M1", increasing = FALSE)
# Multiple ROC curves
ROCCurve(data, truth_by = "D", score_by = c("M1", "M2"), group_name = "Method")
ROCCurve(data, truth_by = "D", score_by = "M1", group_by = "gender", show_auc = "plot")
# Reverse the x-axis and display the axes as percentages
ROCCurve(data, truth_by = "D", score_by = "M1", x_axis_reverse = TRUE, percent = TRUE)
# Pass additional arguments to geom_roc and make the curve black
ROCCurve(data, truth_by = "D", score_by = "M1", n_cuts = 10, palcolor = "black")
# Add confidence intervals
ROCCurve(data, truth_by = "D", score_by = "M1", ci = list(sig.level = .01))
# Facet by a column
ROCCurve(data, truth_by = "D", score_by = "M1", facet_by = "gender")
# Show cutoffs
ROCCurve(data, truth_by = "D", score_by = "M1", cutoffs_at = c(0, "ROC01", "SpEqualSe"))
# Split by a column
p &lt;- ROCCurve(data, truth_by = "D", score_by = "M1", split_by = "gender")
p
# Retrieve the AUC values
attr(p, "auc")
</code></pre>

<hr>
<h2 id='ROCCurveAtomic'>Atomic ROC curve</h2><span id='topic+ROCCurveAtomic'></span>

<h3>Description</h3>

<p>Atomic ROC curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ROCCurveAtomic(
  data,
  truth_by,
  score_by,
  pos_label = NULL,
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  x_axis_reverse = FALSE,
  percent = FALSE,
  ci = NULL,
  n_cuts = 0,
  cutoffs_at = NULL,
  cutoffs_labels = NULL,
  cutoffs_accuracy = 0.01,
  cutoffs_pt_size = 5,
  cutoffs_pt_shape = 4,
  cutoffs_pt_stroke = 1,
  cutoffs_labal_fg = "black",
  cutoffs_label_size = 4,
  cutoffs_label_bg = "white",
  cutoffs_label_bg_r = 0.1,
  show_auc = c("auto", "none", "legend", "plot"),
  auc_accuracy = 0.01,
  auc_size = 4,
  increasing = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = ifelse(x_axis_reverse, "Specificity", "1 - Specificity"),
  ylab = "Sensitivity",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ROCCurveAtomic_+3A_data">data</code></td>
<td>
<p>A data frame with the truth and score columns.
See also https://cran.r-project.org/web/packages/plotROC/vignettes/examples.html.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_truth_by">truth_by</code></td>
<td>
<p>A character string of the column name that contains the true class labels.
(a.k.a. the binary outcome, 1/0 or TRUE/FALSE.)</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_score_by">score_by</code></td>
<td>
<p>character strings of the column names that contains the predicted scores.
When multiple columns are provided, the ROC curve is plotted for each column.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_pos_label">pos_label</code></td>
<td>
<p>A character string of the positive class label.
When NULL, the labels will be handled by the <code>plotROC</code> package.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character vector of column names to group the ROC curve by.
When <code>score_by</code> contains multiple columns, <code>group_by</code> should be NULL.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to separate the columns in <code>group_by</code>.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of the ROC curve groups.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_x_axis_reverse">x_axis_reverse</code></td>
<td>
<p>A logical to reverse the x-axis, that is from 1 to 0.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_percent">percent</code></td>
<td>
<p>A logical to display the x and y axis as percentages.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_ci">ci</code></td>
<td>
<p>A list of arguments to pass to <code><a href="plotROC.html#topic+geom_rocci">plotROC::geom_rocci()</a></code> to add confidence intervals.
When NULL, no confidence intervals are added.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_n_cuts">n_cuts</code></td>
<td>
<p>An integer to specify the number of cutpoints on the ROC curve.
It will be the quantiles of the predicted scores.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_at">cutoffs_at</code></td>
<td>
<p>Vector of user supplied cutoffs to plot as points. If non-NULL,
it will override the values of n_cuts and plot the observed cutoffs closest to the user-supplied ones.
Both <code>cutoffs_at</code> and <code>cutoffs.labels</code> will be passed to <code><a href="plotROC.html#topic+geom_roc">plotROC::geom_roc()</a></code>.
Other than numeric values, the following special values are allowed. These values are the methods of
<code><a href="OptimalCutpoints.html#topic+optimal.cutpoints">OptimalCutpoints::optimal.cutpoints()</a></code>, they are literally:
</p>

<ul>
<li><p> &quot;CB&quot; (cost-benefit method);
</p>
</li>
<li><p> &quot;MCT&quot; (minimizes Misclassification Cost Term);
</p>
</li>
<li><p> &quot;MinValueSp&quot; (a minimum value set for Specificity);
</p>
</li>
<li><p> &quot;MinValueSe&quot; (a minimum value set for Sensitivity);
</p>
</li>
<li><p> &quot;ValueSe&quot; (a value set for Sensitivity);
</p>
</li>
<li><p> &quot;MinValueSpSe&quot; (a minimum value set for Specificity and Sensitivity);
</p>
</li>
<li><p> &quot;MaxSp&quot; (maximizes Specificity);
</p>
</li>
<li><p> &quot;MaxSe&quot; (maximizes Sensitivity);
</p>
</li>
<li><p> &quot;MaxSpSe&quot; (maximizes Sensitivity and Specificity simultaneously);
</p>
</li>
<li><p> &quot;MaxProdSpSe&quot; (maximizes the product of Sensitivity and Specificity or Accuracy Area);
</p>
</li>
<li><p> &quot;ROC01&quot; (minimizes distance between ROC plot and point (0,1));
</p>
</li>
<li><p> &quot;SpEqualSe&quot; (Sensitivity = Specificity);
</p>
</li>
<li><p> &quot;Youden&quot; (Youden Index);
</p>
</li>
<li><p> &quot;MaxEfficiency&quot; (maximizes Efficiency or Accuracy, similar to minimize Error Rate);
</p>
</li>
<li><p> &quot;Minimax&quot; (minimizes the most frequent error);
</p>
</li>
<li><p> &quot;MaxDOR&quot; (maximizes Diagnostic Odds Ratio);
</p>
</li>
<li><p> &quot;MaxKappa&quot; (maximizes Kappa Index);
</p>
</li>
<li><p> &quot;MinValueNPV&quot; (a minimum value set for Negative Predictive Value);
</p>
</li>
<li><p> &quot;MinValuePPV&quot; (a minimum value set for Positive Predictive Value);
</p>
</li>
<li><p> &quot;ValueNPV&quot; (a value set for Negative Predictive Value);
</p>
</li>
<li><p> &quot;ValuePPV&quot; (a value set for Positive Predictive Value);
</p>
</li>
<li><p> &quot;MinValueNPVPPV&quot; (a minimum value set for Predictive Values);
</p>
</li>
<li><p> &quot;PROC01&quot; (minimizes distance between PROC plot and point (0,1));
</p>
</li>
<li><p> &quot;NPVEqualPPV&quot; (Negative Predictive Value = Positive Predictive Value);
</p>
</li>
<li><p> &quot;MaxNPVPPV&quot; (maximizes Positive Predictive Value and Negative Predictive Value simultaneously);
</p>
</li>
<li><p> &quot;MaxSumNPVPPV&quot; (maximizes the sum of the Predictive Values);
</p>
</li>
<li><p> &quot;MaxProdNPVPPV&quot; (maximizes the product of Predictive Values);
</p>
</li>
<li><p> &quot;ValueDLR.Negative&quot; (a value set for Negative Diagnostic Likelihood Ratio);
</p>
</li>
<li><p> &quot;ValueDLR.Positive&quot; (a value set for Positive Diagnostic Likelihood Ratio);
</p>
</li>
<li><p> &quot;MinPvalue&quot; (minimizes p-value associated with the statistical Chi-squared test which measures the association between
the marker and the binary result obtained on using the cutpoint);
</p>
</li>
<li><p> &quot;ObservedPrev&quot; (The closest value to observed prevalence);
</p>
</li>
<li><p> &quot;MeanPrev&quot; (The closest value to the mean of the diagnostic test values);
</p>
</li>
<li><p> &quot;PrevalenceMatching&quot; (The value for which predicted prevalence is practically equal to observed prevalence).
</p>
</li></ul>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_labels">cutoffs_labels</code></td>
<td>
<p>vector of user-supplied labels for the cutoffs. Must be a character vector of the
same length as cutoffs_at.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_accuracy">cutoffs_accuracy</code></td>
<td>
<p>A numeric to specify the accuracy of the cutoff values to show.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_pt_size">cutoffs_pt_size</code></td>
<td>
<p>A numeric to specify the size of the cutoff points.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_pt_shape">cutoffs_pt_shape</code></td>
<td>
<p>A numeric to specify the shape of the cutoff points.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_pt_stroke">cutoffs_pt_stroke</code></td>
<td>
<p>A numeric to specify the stroke of the cutoff points.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_labal_fg">cutoffs_labal_fg</code></td>
<td>
<p>A character string to specify the color of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_label_size">cutoffs_label_size</code></td>
<td>
<p>A numeric to specify the size of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_label_bg">cutoffs_label_bg</code></td>
<td>
<p>A character string to specify the background color of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_cutoffs_label_bg_r">cutoffs_label_bg_r</code></td>
<td>
<p>A numeric to specify the radius of the background of the cutoff labels.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_show_auc">show_auc</code></td>
<td>
<p>A character string to specify the position of the AUC values.
</p>

<ul>
<li><p> &quot;auto&quot; (default): Automatically determine the position based on the plot.
When there is a single group or 'facet_by' is provided, the AUC is placed on the plot.
Otherwise, the AUC is placed in the legend.
</p>
</li>
<li><p> &quot;none&quot;: Do not display the AUC values.
</p>
</li>
<li><p> &quot;legend&quot;: Display the AUC values in the legend.
</p>
</li>
<li><p> &quot;plot&quot;: Display the AUC values on the plot (left/right bottom corner).
</p>
</li></ul>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_auc_accuracy">auc_accuracy</code></td>
<td>
<p>A numeric to specify the accuracy of the AUC values.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_auc_size">auc_size</code></td>
<td>
<p>A numeric to specify the size of the AUC values when they are displayed on the plot.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="ROCCurveAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='SankeyPlot'>Sankey / Alluvial Plot</h2><span id='topic+SankeyPlot'></span><span id='topic+AlluvialPlot'></span>

<h3>Description</h3>

<p>A plot visualizing flow/movement/change from one state to another or one time to another.
<code>AlluvialPlot</code> is an alias of <code>SankeyPlot</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SankeyPlot(
  data,
  in_form = c("auto", "long", "lodes", "wide", "alluvia", "counts"),
  x,
  x_sep = "_",
  y = NULL,
  stratum = NULL,
  stratum_sep = "_",
  alluvium = NULL,
  alluvium_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  keep_empty = TRUE,
  flow = FALSE,
  expand = c(0, 0, 0, 0),
  nodes_legend = c("auto", "separate", "merge", "none"),
  nodes_color = "grey30",
  links_fill_by = NULL,
  links_fill_by_sep = "_",
  links_name = NULL,
  links_color = "gray80",
  nodes_palette = "Paired",
  nodes_palcolor = NULL,
  nodes_alpha = 1,
  nodes_label = FALSE,
  nodes_label_miny = 0,
  nodes_width = 0.25,
  links_palette = "Paired",
  links_palcolor = NULL,
  links_alpha = 0.6,
  legend.box = "vertical",
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  flip = FALSE,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)

AlluvialPlot(
  data,
  in_form = c("auto", "long", "lodes", "wide", "alluvia", "counts"),
  x,
  x_sep = "_",
  y = NULL,
  stratum = NULL,
  stratum_sep = "_",
  alluvium = NULL,
  alluvium_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  keep_empty = TRUE,
  flow = FALSE,
  expand = c(0, 0, 0, 0),
  nodes_legend = c("auto", "separate", "merge", "none"),
  nodes_color = "grey30",
  links_fill_by = NULL,
  links_fill_by_sep = "_",
  links_name = NULL,
  links_color = "gray80",
  nodes_palette = "Paired",
  nodes_palcolor = NULL,
  nodes_alpha = 1,
  nodes_label = FALSE,
  nodes_label_miny = 0,
  nodes_width = 0.25,
  links_palette = "Paired",
  links_palcolor = NULL,
  links_alpha = 0.6,
  legend.box = "vertical",
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  flip = FALSE,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SankeyPlot_+3A_data">data</code></td>
<td>
<p>A data frame in following possible formats:
</p>

<ul>
<li><p> &quot;long&quot; or &quot;lodes&quot;: A long format with columns for <code>x</code>, <code>stratum</code>, <code>alluvium</code>, and <code>y</code>.
<code>x</code> (required, single columns or concatenated by <code>x_sep</code>) is the column name to plot on the x-axis,
<code>stratum</code> (defaults to <code>links_fill_by</code>) is the column name
to group the nodes for each <code>x</code>, <code>alluvium</code> (required) is the column name to define the links, and <code>y</code>
is the frequency of each <code>x</code>, <code>stratum</code>, and <code>alluvium</code>.
</p>
</li>
<li><p> &quot;wide&quot; or &quot;alluvia&quot;: A wide format with columns for <code>x</code>.
<code>x</code> (required, multiple columns, <code>x_sep</code> won't be used) are the columns to plot on the x-axis,
<code>stratum</code> and <code>alluvium</code> will be ignored.
See <a href="ggalluvial.html#topic+alluvial-data">ggalluvial::to_lodes_form</a> for more details.
</p>
</li>
<li><p> &quot;counts&quot;: A format with counts being provides under each <code>x</code>.
<code>x</code> (required, multiple columns, <code>x_sep</code> won't be used) are the columns to plot on the x-axis.
When the first element of <code>x</code> is &quot;.&quot;, values of <code>links_fill_by</code> (required) will be added to the plot as the first column of nodes.
It is useful to show how the links are flowed from the source to the targets.
</p>
</li>
<li><p> &quot;auto&quot; (default): Automatically determine the format based on the columns provided.
When the length of <code>x</code> is greater than 1 and all <code>x</code> columns are numeric, &quot;counts&quot; format will be used.
When the length of <code>x</code> is greater than 1 and <a href="ggalluvial.html#topic+alluvial-data">ggalluvial::is_alluvia_form</a> returns TRUE, &quot;alluvia&quot; format will be used.
Otherwise, &quot;lodes&quot; format will be tried.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_in_form">in_form</code></td>
<td>
<p>A character string to specify the format of the data.
Possible values are &quot;auto&quot;, &quot;long&quot;, &quot;lodes&quot;, &quot;wide&quot;, &quot;alluvia&quot;, and &quot;counts&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_stratum">stratum</code></td>
<td>
<p>A character string of the column name to group the nodes for each <code>x</code>.
See <code>data</code> for more details.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_stratum_sep">stratum_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>stratum</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_alluvium">alluvium</code></td>
<td>
<p>A character string of the column name to define the links.
See <code>data</code> for more details.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_alluvium_sep">alluvium_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>alluvium</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_flow">flow</code></td>
<td>
<p>A logical value to use <a href="ggalluvial.html#topic+geom_flow">ggalluvial::geom_flow</a> instead of <a href="ggalluvial.html#topic+geom_alluvium">ggalluvial::geom_alluvium</a>.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_legend">nodes_legend</code></td>
<td>
<p>Controls how the legend of nodes will be shown. Possible values are:
</p>

<ul>
<li><p> &quot;merge&quot;: Merge the legends of nodes. That is only one legend will be shown for all nodes.
</p>
</li>
<li><p> &quot;separate&quot;: Show the legends of nodes separately. That is, nodes on each <code>x</code> will have their own legend.
</p>
</li>
<li><p> &quot;none&quot;: Do not show the legend of nodes.
</p>
</li>
<li><p> &quot;auto&quot;: Automatically determine how to show the legend.
When <code>nodes_label</code> is TRUE, &quot;none&quot; will apply.
When <code>nodes_label</code> is FALSE, and if stratum is the same as links_fill_by, &quot;none&quot; will apply.
If there is any overlapping values between the nodes on different <code>x</code>,
&quot;merge&quot; will apply. Otherwise, &quot;separate&quot; will apply.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_color">nodes_color</code></td>
<td>
<p>A character string to color the nodes.
Use a special value &quot;.fill&quot; to use the same color as the fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_fill_by">links_fill_by</code></td>
<td>
<p>A character string of the column name to fill the links.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_fill_by_sep">links_fill_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>links_fill_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_name">links_name</code></td>
<td>
<p>A character string to name the legend of links.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_color">links_color</code></td>
<td>
<p>A character string to color the borders of links.
Use a special value &quot;.fill&quot; to use the same color as the fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_palette">nodes_palette</code></td>
<td>
<p>A character string to specify the palette of nodes fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_palcolor">nodes_palcolor</code></td>
<td>
<p>A character vector to specify the colors of nodes fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_alpha">nodes_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of nodes fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_label">nodes_label</code></td>
<td>
<p>A logical value to show the labels on the nodes.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_label_miny">nodes_label_miny</code></td>
<td>
<p>A numeric value to specify the minimum y (frequency) to show the labels.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nodes_width">nodes_width</code></td>
<td>
<p>A numeric value to specify the width of nodes.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_palette">links_palette</code></td>
<td>
<p>A character string to specify the palette of links fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_palcolor">links_palcolor</code></td>
<td>
<p>A character vector to specify the colors of links fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_links_alpha">links_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of links fill.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_legend.box">legend.box</code></td>
<td>
<p>A character string to specify the box of the legend, either &quot;vertical&quot; or &quot;horizontal&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the plot.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="SankeyPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Reproduce the examples in ggalluvial
set.seed(8525)

data(UCBAdmissions, package = "datasets")
UCBAdmissions &lt;- as.data.frame(UCBAdmissions)
SankeyPlot(as.data.frame(UCBAdmissions), x = c("Gender", "Dept"),
    y = "Freq", nodes_width = 1/12, links_fill_by = "Admit", nodes_label = TRUE,
    nodes_palette = "simspec", links_palette = "Set1", links_alpha = 0.5,
    nodes_palcolor = "black", links_color = "transparent")

data(HairEyeColor, package = "datasets")
SankeyPlot(as.data.frame(HairEyeColor), x = c("Hair", "Eye", "Sex"),
    y = "Freq", links_fill_by = "Eye", nodes_width = 1/8, nodes_alpha = 0.4,
    flip = TRUE, reverse = FALSE, knot.pos = 0, links_color = "transparent",
    ylab = "Freq", links_alpha = 0.5, links_name = "Eye (links)", links_palcolor = c(
        Brown = "#70493D", Hazel = "#E2AC76", Green = "#3F752B", Blue = "#81B0E4"))

data(Refugees, package = "alluvial")
country_regions &lt;- c(
    Afghanistan = "Middle East",
    Burundi = "Central Africa",
    `Congo DRC` = "Central Africa",
    Iraq = "Middle East",
    Myanmar = "Southeast Asia",
    Palestine = "Middle East",
    Somalia = "Horn of Africa",
    Sudan = "Central Africa",
    Syria = "Middle East",
    Vietnam = "Southeast Asia"
)
Refugees$region &lt;- country_regions[Refugees$country]
SankeyPlot(Refugees, x = "year", y = "refugees", alluvium = "country",
    links_fill_by = "country", links_color = ".fill", links_alpha = 0.75,
    links_palette = "Set3", facet_by = "region", x_text_angle = -45, nodes_legend = "none",
    theme_args = list(strip.background = ggplot2::element_rect(fill="grey80")),
    decreasing = FALSE, nodes_width = 0, nodes_color = "transparent", ylab = "refugees",
    title = "Refugee volume by country and region of origin")

data(majors, package = "ggalluvial")
majors$curriculum &lt;- as.factor(majors$curriculum)
SankeyPlot(majors, x = "semester", stratum = "curriculum", alluvium = "student",
    links_fill_by = "curriculum", flow = TRUE, stat = "alluvium", nodes_palette = "Set2",
    links_palette = "Set2")

data(vaccinations, package = "ggalluvial")
vaccinations &lt;- transform(vaccinations,
    response = factor(response, rev(levels(response))))
SankeyPlot(vaccinations, x = "survey", stratum = "response", alluvium = "subject",
    y = "freq", links_fill_by = "response", nodes_label = TRUE, nodes_alpha = 0.5,
    nodes_palette = "seurat", links_palette = "seurat", links_alpha = 0.5,
    legend.position = "none", flow = TRUE, expand = c(0, 0, 0, .15), stat = "alluvium",
    title = "vaccination survey responses at three points in time")

data(Titanic, package = "datasets")
SankeyPlot(as.data.frame(Titanic), x = c("Class", "Sex"), y = "Freq",
    links_fill_by = "Survived", flow = TRUE, facet_by = "Age", facet_scales = "free_y",
    nodes_label = TRUE, expand = c(0.05, 0), xlab = "", links_palette = "Set1",
    nodes_palcolor = "white", nodes_label_miny = 10)

# Simulated examples
df &lt;- data.frame(
    Clone = paste0("clone", 1:10),
    Timepoint1 = sample(c(rep(0, 30), 1:100), 10),
    Timepoint2 = sample(c(rep(0, 30), 1:100), 10)
)
SankeyPlot(df, x = c("Timepoint1", "Timepoint2"), alluvium = "Clone",
    links_color = ".fill")

df &lt;- data.frame(
    Clone = rep(paste0("clone", 1:6), each = 2),
    Timepoint1 = sample(c(rep(0, 30), 1:100), 6),
    Timepoint2 = sample(c(rep(0, 30), 1:100), 6),
    Group = rep(c("A", "B"), 6)
)
SankeyPlot(df, x = c(".", "Timepoint1", "Timepoint2"),
    stratum = "Group", links_fill_by = "Clone", links_color = ".fill")

</code></pre>

<hr>
<h2 id='SankeyPlotAtomic'>Atomic Sankey plot</h2><span id='topic+SankeyPlotAtomic'></span>

<h3>Description</h3>

<p>Plot a Sankey plot without splitting the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SankeyPlotAtomic(
  data,
  in_form = c("auto", "long", "lodes", "wide", "alluvia", "counts"),
  x,
  x_sep = "_",
  y = NULL,
  stratum = NULL,
  stratum_sep = "_",
  alluvium = NULL,
  alluvium_sep = "_",
  flow = FALSE,
  nodes_color = "grey30",
  links_fill_by = NULL,
  links_fill_by_sep = "_",
  links_name = NULL,
  links_color = "gray80",
  nodes_palette = "Paired",
  nodes_palcolor = NULL,
  nodes_alpha = 1,
  nodes_label = FALSE,
  nodes_width = 0.25,
  nodes_label_miny = 0,
  nodes_legend = c("auto", "separate", "merge", "none"),
  expand = c(0, 0, 0, 0),
  links_palette = "Paired",
  links_palcolor = NULL,
  links_alpha = 0.6,
  legend.box = "vertical",
  keep_empty = TRUE,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  flip = FALSE,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SankeyPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_in_form">in_form</code></td>
<td>
<p>A character string to specify the format of the data.
Possible values are &quot;auto&quot;, &quot;long&quot;, &quot;lodes&quot;, &quot;wide&quot;, &quot;alluvia&quot;, and &quot;counts&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string of the column name to plot on the x-axis.
See <code>data</code> for more details.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis.
When <code>in_form</code> is &quot;counts&quot;, <code>y</code> will be ignored. Otherwise, it defaults to the count of each <code>x</code>, <code>stratum</code>, <code>alluvium</code> and <code>links_fill_by</code>.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_stratum">stratum</code></td>
<td>
<p>A character string of the column name to group the nodes for each <code>x</code>.
See <code>data</code> for more details.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_stratum_sep">stratum_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>stratum</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_alluvium">alluvium</code></td>
<td>
<p>A character string of the column name to define the links.
See <code>data</code> for more details.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_alluvium_sep">alluvium_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>alluvium</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_flow">flow</code></td>
<td>
<p>A logical value to use <a href="ggalluvial.html#topic+geom_flow">ggalluvial::geom_flow</a> instead of <a href="ggalluvial.html#topic+geom_alluvium">ggalluvial::geom_alluvium</a>.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_color">nodes_color</code></td>
<td>
<p>A character string to color the nodes.
Use a special value &quot;.fill&quot; to use the same color as the fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_fill_by">links_fill_by</code></td>
<td>
<p>A character string of the column name to fill the links.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_fill_by_sep">links_fill_by_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>links_fill_by</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_name">links_name</code></td>
<td>
<p>A character string to name the legend of links.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_color">links_color</code></td>
<td>
<p>A character string to color the borders of links.
Use a special value &quot;.fill&quot; to use the same color as the fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_palette">nodes_palette</code></td>
<td>
<p>A character string to specify the palette of nodes fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_palcolor">nodes_palcolor</code></td>
<td>
<p>A character vector to specify the colors of nodes fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_alpha">nodes_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of nodes fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_label">nodes_label</code></td>
<td>
<p>A logical value to show the labels on the nodes.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_width">nodes_width</code></td>
<td>
<p>A numeric value to specify the width of nodes.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_label_miny">nodes_label_miny</code></td>
<td>
<p>A numeric value to specify the minimum y (frequency) to show the labels.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_nodes_legend">nodes_legend</code></td>
<td>
<p>Controls how the legend of nodes will be shown. Possible values are:
</p>

<ul>
<li><p> &quot;merge&quot;: Merge the legends of nodes. That is only one legend will be shown for all nodes.
</p>
</li>
<li><p> &quot;separate&quot;: Show the legends of nodes separately. That is, nodes on each <code>x</code> will have their own legend.
</p>
</li>
<li><p> &quot;none&quot;: Do not show the legend of nodes.
</p>
</li>
<li><p> &quot;auto&quot;: Automatically determine how to show the legend.
When <code>nodes_label</code> is TRUE, &quot;none&quot; will apply.
When <code>nodes_label</code> is FALSE, and if stratum is the same as links_fill_by, &quot;none&quot; will apply.
If there is any overlapping values between the nodes on different <code>x</code>,
&quot;merge&quot; will apply. Otherwise, &quot;separate&quot; will apply.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_palette">links_palette</code></td>
<td>
<p>A character string to specify the palette of links fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_palcolor">links_palcolor</code></td>
<td>
<p>A character vector to specify the colors of links fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_links_alpha">links_alpha</code></td>
<td>
<p>A numeric value to specify the transparency of links fill.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_legend.box">legend.box</code></td>
<td>
<p>A character string to specify the box of the legend, either &quot;vertical&quot; or &quot;horizontal&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value to keep the empty nodes.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value to flip the plot.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="SankeyPlotAtomic_+3A_...">...</code></td>
<td>
<p>Other arguments to pass to <a href="ggalluvial.html#topic+geom_alluvium">ggalluvial::geom_alluvium</a> or <a href="ggalluvial.html#topic+geom_flow">ggalluvial::geom_flow</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='ScatterPlot'>Scatter Plot</h2><span id='topic+ScatterPlot'></span>

<h3>Description</h3>

<p>Scatter Plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScatterPlot(
  data,
  x,
  y,
  size_by = 2,
  size_name = NULL,
  color_by = NULL,
  color_name = NULL,
  color_reverse = FALSE,
  split_by = NULL,
  split_by_sep = "_",
  shape = 21,
  alpha = ifelse(shape %in% 21:25, 0.65, 1),
  border_color = "black",
  highlight = NULL,
  highlight_shape = 16,
  highlight_size = 3,
  highlight_color = "red",
  highlight_alpha = 1,
  theme = "theme_this",
  theme_args = list(),
  palette = ifelse(!is.null(color_by) &amp;&amp; !is.numeric(data[[color_by]]), "Paired",
    "Spectral"),
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ScatterPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_size_by">size_by</code></td>
<td>
<p>Which column to use as the size of the dots. It must be a numeric column.
Or it can be a numeric value to specify the size of the dots.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_size_name">size_name</code></td>
<td>
<p>A character vector specifying the name for the size legend.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_color_by">color_by</code></td>
<td>
<p>Which column to use as the color of the dots.
It could be a numeric column or a factor/character column.
For shapes 21-25, the color is applied to the fill color.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_color_name">color_name</code></td>
<td>
<p>A character vector specifying the name for the color legend.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_color_reverse">color_reverse</code></td>
<td>
<p>A logical value indicating whether to reverse the color direction. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_shape">shape</code></td>
<td>
<p>A numeric value specifying the shape of the points. Default is 21.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_border_color">border_color</code></td>
<td>
<p>A character vector specifying the color for the border of the points.
Or TRUE to use the fill color as the border color.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_highlight">highlight</code></td>
<td>
<p>A vector of indexes or rownames to select the points to highlight.
It could also be an expression (in string) to filter the data.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>A numeric value specifying the shape of the highlighted points. Default is 16.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value specifying the size of the highlighted points. Default is 3.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character vector specifying the color of the highlighted points. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the highlighted points. Default is 1.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="ScatterPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(8525)

data &lt;- data.frame(
   x = rnorm(20),
   y = rnorm(20),
   w = abs(rnorm(20)),
   t = sample(c("A", "B"), 20, replace = TRUE)
)
ScatterPlot(data, x = "x", y = "y")

# highlight points
ScatterPlot(data, x = "x", y = "y", highlight = 'x &gt; 0')

# size_by is a numeric column
ScatterPlot(data, x = "x", y = "y", size_by = "w")

# color_by is a numeric column
ScatterPlot(data, x = "x", y = "y", color_by = "w")

# color_by is a factor/character column and set a border_color
ScatterPlot(data, x = "x", y = "y", size_by = "w", color_by = "t",
 border_color = "red")

# Same border_color as the fill color
ScatterPlot(data, x = "x", y = "y", size_by = "w", color_by = "t",
 border_color = TRUE)

# Shape doesn't have fill color
ScatterPlot(data, x = "x", y = "y", size_by = "w", color_by = "t",
 shape = 1, palette = "Set1")

# Change color per plot
ScatterPlot(data, x = "x", y = "y", split_by = "t",
            palcolor = list(A = "blue", B = "red"))
</code></pre>

<hr>
<h2 id='ScatterPlotAtomic'>Scatter Plot Atomic</h2><span id='topic+ScatterPlotAtomic'></span>

<h3>Description</h3>

<p>Scatter Plot Atomic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScatterPlotAtomic(
  data,
  x,
  y,
  size_by = 2,
  size_name = NULL,
  color_by = NULL,
  color_name = NULL,
  color_reverse = FALSE,
  theme = "theme_this",
  theme_args = list(),
  alpha = ifelse(shape %in% 21:25, 0.65, 1),
  shape = 21,
  border_color = "black",
  xtrans = "identity",
  ytrans = "identity",
  highlight = NULL,
  highlight_shape = 16,
  highlight_size = 3,
  highlight_color = "red",
  highlight_alpha = 1,
  palette = ifelse(!is.null(color_by) &amp;&amp; !is.numeric(data[[color_by]]), "Paired",
    "Spectral"),
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ScatterPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character vector specifying the column to use for the x-axis.
A numeric column is expected.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character vector specifying the column to use for the y-axis.
A numeric column is expected.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_size_by">size_by</code></td>
<td>
<p>Which column to use as the size of the dots. It must be a numeric column.
Or it can be a numeric value to specify the size of the dots.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_size_name">size_name</code></td>
<td>
<p>A character vector specifying the name for the size legend.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_color_by">color_by</code></td>
<td>
<p>Which column to use as the color of the dots.
It could be a numeric column or a factor/character column.
For shapes 21-25, the color is applied to the fill color.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_color_name">color_name</code></td>
<td>
<p>A character vector specifying the name for the color legend.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_color_reverse">color_reverse</code></td>
<td>
<p>A logical value indicating whether to reverse the color direction. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the dots. Default is 1.
For shapes 21-25, the transparency is applied to the fill color.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_shape">shape</code></td>
<td>
<p>A numeric value specifying the shape of the points. Default is 21.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_border_color">border_color</code></td>
<td>
<p>A character vector specifying the color for the border of the points.
Or TRUE to use the fill color as the border color.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_xtrans">xtrans</code></td>
<td>
<p>A character vector specifying the transformation of the x-axis. Default is &quot;identity&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_ytrans">ytrans</code></td>
<td>
<p>A character vector specifying the transformation of the y-axis. Default is &quot;identity&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_highlight">highlight</code></td>
<td>
<p>A vector of indexes or rownames to select the points to highlight.
It could also be an expression (in string) to filter the data.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_highlight_shape">highlight_shape</code></td>
<td>
<p>A numeric value specifying the shape of the highlighted points. Default is 16.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value specifying the size of the highlighted points. Default is 3.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character vector specifying the color of the highlighted points. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the highlighted points. Default is 1.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="ScatterPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='show_palettes'>Show the color palettes</h2><span id='topic+show_palettes'></span>

<h3>Description</h3>

<p>This function displays color palettes using ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_palettes(
  palettes = NULL,
  type = c("discrete", "continuous"),
  index = NULL,
  palette_names = NULL,
  return_names = TRUE,
  return_palettes = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="show_palettes_+3A_palettes">palettes</code></td>
<td>
<p>A list of color palettes. If <code>NULL</code>, uses default palettes.</p>
</td></tr>
<tr><td><code id="show_palettes_+3A_type">type</code></td>
<td>
<p>A character vector specifying the type of palettes to include. Default is &quot;discrete&quot;.</p>
</td></tr>
<tr><td><code id="show_palettes_+3A_index">index</code></td>
<td>
<p>A numeric vector specifying the indices of the palettes to include. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="show_palettes_+3A_palette_names">palette_names</code></td>
<td>
<p>A character vector specifying the names of the SCP palettes to include. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="show_palettes_+3A_return_names">return_names</code></td>
<td>
<p>A logical value indicating whether to return the names of the selected palettes. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="show_palettes_+3A_return_palettes">return_palettes</code></td>
<td>
<p>A logical value indicating whether to return the colors of selected palettes. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of palette names or a list of palettes.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+palette_list">palette_list</a></code>
</p>
<p><a href="../articles/all-available-palettes.html">All available palettes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>show_palettes(palettes = list(c("red", "blue", "green"), c("yellow", "purple", "orange")))
all_palettes &lt;- show_palettes(return_palettes = TRUE)
names(all_palettes)
all_palettes[["simspec"]]
show_palettes(index = 1:10)
show_palettes(type = "discrete", index = 1:10)
show_palettes(type = "continuous", index = 1:10)
show_palettes(
    palette_names = c("Paired", "nejm", "simspec", "Spectral", "jet"),
    return_palettes = TRUE
)

</code></pre>

<hr>
<h2 id='SplitBarPlotAtomic'>SplitBarPlotAtomic</h2><span id='topic+SplitBarPlotAtomic'></span>

<h3>Description</h3>

<p>Create a split bar plot without splitting the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SplitBarPlotAtomic(
  data,
  x,
  y,
  y_sep = "_",
  flip = FALSE,
  alpha_by = NULL,
  alpha_reverse = FALSE,
  alpha_name = NULL,
  order_y = list(`+` = c("x_desc", "alpha_desc"), `-` = c("x_desc", "alpha_asc")),
  bar_height = 0.9,
  lineheight = 0.5,
  max_charwidth = 80,
  fill_by = NULL,
  fill_by_sep = "_",
  fill_name = NULL,
  direction_pos_name = "positive",
  direction_neg_name = "negative",
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  facet_by = NULL,
  facet_scales = "free_y",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  aspect.ratio = 1,
  x_min = NULL,
  x_max = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  keep_empty = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SplitBarPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_x">x</code></td>
<td>
<p>The column name of the terms on the x axis. There should be both negative and positive values.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_y">y</code></td>
<td>
<p>The column name(s) of the values. If there are multiple columns, they will be concatenated.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_y_sep">y_sep</code></td>
<td>
<p>A character string to concatenate the x columns if there are multiple.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the x and y axes.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_alpha_by">alpha_by</code></td>
<td>
<p>A character string indicating the column name to use for the transparency of the bars.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_alpha_reverse">alpha_reverse</code></td>
<td>
<p>A logical value indicating whether to reverse the transparency.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_alpha_name">alpha_name</code></td>
<td>
<p>A character string indicating the legend name of the transparency.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_order_y">order_y</code></td>
<td>
<p>A list of character strings indicating the order of the y axis.
The keys are &quot;+&quot;, &quot;-&quot;, or &quot;<em>&quot;. However, &quot;+/-&quot; should not be mixed with &quot;</em>&quot;.
The values are &quot;x_asc&quot;, &quot;x_desc&quot;, &quot;alpha_asc&quot;, or &quot;alpha_desc&quot;, indicating how to order the y axis.
The default is <code>list("+" = c("x_desc", "alpha_desc"), "-" = c("x_desc", "alpha_asc"))</code>, meaning
the positive values are ordered by the x-axis values in descending order and the alpha values in descending order,
and the negative values are ordered by the x-axis values in descending order and the alpha values in ascending order.
The &quot;*&quot; key is used to order the y axis without considering the direction.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_bar_height">bar_height</code></td>
<td>
<p>A numeric value indicating the height of the bars.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_lineheight">lineheight</code></td>
<td>
<p>A numeric value indicating the height of the text.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_max_charwidth">max_charwidth</code></td>
<td>
<p>A numeric value indicating the maximum width of the text.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_fill_by">fill_by</code></td>
<td>
<p>A character string indicating the column name to use for the fill of the bars.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_fill_by_sep">fill_by_sep</code></td>
<td>
<p>A character string to concatenate the fill columns if there are multiple.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_fill_name">fill_name</code></td>
<td>
<p>A character string indicating the legend name of the fill.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_direction_pos_name">direction_pos_name</code></td>
<td>
<p>A character string indicating the name of the positive direction.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_direction_neg_name">direction_neg_name</code></td>
<td>
<p>A character string indicating the name of the negative direction.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_x_min">x_min</code></td>
<td>
<p>A numeric value indicating the minimum value of the x axis.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_x_max">x_max</code></td>
<td>
<p>A numeric value indicating the maximum value of the x axis.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="SplitBarPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='theme_blank'>Blank theme</h2><span id='topic+theme_blank'></span>

<h3>Description</h3>

<p>This function creates a theme with all elements blank except for axis lines and labels.
It can optionally add coordinate axes in the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_blank(
  add_coord = TRUE,
  xlen_npc = 0.15,
  ylen_npc = 0.15,
  xlab = "",
  ylab = "",
  lab_size = 12,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_blank_+3A_add_coord">add_coord</code></td>
<td>
<p>Whether to add coordinate arrows. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="theme_blank_+3A_xlen_npc">xlen_npc</code></td>
<td>
<p>The length of the x-axis arrow in &quot;npc&quot;.</p>
</td></tr>
<tr><td><code id="theme_blank_+3A_ylen_npc">ylen_npc</code></td>
<td>
<p>The length of the y-axis arrow in &quot;npc&quot;.</p>
</td></tr>
<tr><td><code id="theme_blank_+3A_xlab">xlab</code></td>
<td>
<p>x-axis label.</p>
</td></tr>
<tr><td><code id="theme_blank_+3A_ylab">ylab</code></td>
<td>
<p>y-axis label.</p>
</td></tr>
<tr><td><code id="theme_blank_+3A_lab_size">lab_size</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="theme_blank_+3A_...">...</code></td>
<td>
<p>Arguments passed to the <code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 theme.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
p &lt;- ggplot(mtcars, aes(x = wt, y = mpg, colour = factor(cyl))) +
    geom_point()
p + theme_blank()
p + theme_blank(xlab = "x-axis", ylab = "y-axis", lab_size = 16)
</code></pre>

<hr>
<h2 id='theme_this'>A ggplot2 theme and palettes for plotthis
Borrowed from the <code>theme_this</code> function in the <code>SCP</code> pipeline</h2><span id='topic+theme_this'></span>

<h3>Description</h3>

<p>A ggplot2 theme and palettes for plotthis
Borrowed from the <code>theme_this</code> function in the <code>SCP</code> pipeline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_this(aspect.ratio = NULL, base_size = 12, font_family = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_this_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>The aspect ratio of the plot</p>
</td></tr>
<tr><td><code id="theme_this_+3A_base_size">base_size</code></td>
<td>
<p>The base size of the text</p>
</td></tr>
<tr><td><code id="theme_this_+3A_font_family">font_family</code></td>
<td>
<p>The font family of the text</p>
</td></tr>
<tr><td><code id="theme_this_+3A_...">...</code></td>
<td>
<p>Other arguments for <code>theme()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 theme
</p>


<h3>See Also</h3>

<p><a href="https://github.com/zhanghao-njmu/SCP">https://github.com/zhanghao-njmu/SCP</a>
</p>

<hr>
<h2 id='TrendPlot'>Trend plot</h2><span id='topic+TrendPlot'></span>

<h3>Description</h3>

<p>A trend plot is like an area plot but with gaps between the bars.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TrendPlot(
  data,
  x,
  y = NULL,
  x_sep = "_",
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  scale_y = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TrendPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of fill.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_scale_y">scale_y</code></td>
<td>
<p>A logical value to scale the y-axis by the total number in each x-axis group.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="TrendPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AreaPlot">AreaPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(
    x = rep(c("A", "B", "C", "D"), 2),
    y = c(1, 3, 6, 4, 2, 5, 7, 8),
    group = rep(c("F1", "F2"), each = 4)
)
TrendPlot(data, x = "x", y = "y", group_by = "group")
TrendPlot(data, x = "x", y = "y", group_by = "group",
         scale_y = TRUE)
TrendPlot(data, x = "x", y = "y", split_by = "group")
TrendPlot(data, x = "x", y = "y", split_by = "group",
          palette = c(F1 = "Set1", F2 = "Paired"))
</code></pre>

<hr>
<h2 id='TrendPlotAtomic'>Atomic trend plot</h2><span id='topic+TrendPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic trend plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TrendPlotAtomic(
  data,
  x,
  y = NULL,
  x_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  group_name = NULL,
  scale_y = FALSE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = waiver(),
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TrendPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string of the column name to plot on the x-axis.
A character/factor column is expected.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string of the column name to plot on the y-axis.
A numeric column is expected.
If NULL, the count of the x-axis column will be used.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_x_sep">x_sep</code></td>
<td>
<p>A character string to concatenate the columns in <code>x</code>, if multiple columns are provided.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character vector of column names to fill the area plot by.
If NULL, the plot will be filled by the first color of the palette.
If multiple columns are provided, the columns will be concatenated with
<code>group_by_sep</code> and used as the fill column.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>A character string to separate the columns in <code>group_by</code>.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_group_name">group_name</code></td>
<td>
<p>A character string to name the legend of fill.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_scale_y">scale_y</code></td>
<td>
<p>A logical value to scale the y-axis by the total number in each x-axis group.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="TrendPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='UpsetPlotAtomic'>Atomic Upset plot</h2><span id='topic+UpsetPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic Upset plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpsetPlotAtomic(
  data,
  in_form = NULL,
  group_by = NULL,
  group_by_sep = "_",
  id_by = NULL,
  label = TRUE,
  label_fg = "black",
  label_size = NULL,
  label_bg = "white",
  label_bg_r = 0.1,
  palette = "material-indigo",
  palcolor = NULL,
  alpha = 1,
  specific = TRUE,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  aspect.ratio = 0.6,
  legend.position = "right",
  legend.direction = "vertical",
  levels = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UpsetPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_in_form">in_form</code></td>
<td>
<p>A character string indicating the datatype of the input data.
Possible values are &quot;long&quot;, &quot;wide&quot;, &quot;list&quot;, &quot;upset&quot; or NULL.
&quot;long&quot; indicates the data is in long format.
&quot;wide&quot; indicates the data is in wide format.
&quot;list&quot; indicates the data is a list.
&quot;upset&quot; indicates the data is a UpsetPlotData object.
NULL indicates the function will detect the datatype of the input data.
</p>
<p>A long format data would look like:
</p>
<pre>
group_by id_by
A        a1
A        a2
B        a1
B        a3
...
</pre>
<p>A wide format data would look like:
</p>
<pre>
A    B
TRUE TRUE
TRUE FALSE
FALSE TRUE
...
</pre>
<p>A list format data would look like:
</p>
<pre>
list(A = c("a1", "a2"), B = c("a1", "a3"))
</pre>
<p>An UpsetPlotData object is generated by PrepareUpsetData() would look like:
</p>
<pre>
group_by
--------
list("A")  # a2
list("B")  # a3
list(c("A", "B"))  # a1
...
</pre></td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_id_by">id_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to identify the instances.
Required when <code>group_by</code> is a single column and data is a data frame.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_label">label</code></td>
<td>
<p>A logical value to show the labels on the bars.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string specifying the color of the label text.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the label text.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string specifying the background color of the label.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the radius of the background of the label.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_specific">specific</code></td>
<td>
<p>A logical value to show the specific intersections only.
ggVennDiagram, by default, only return the specific subsets of a region. However,
sometimes, we want to show all the overlapping items for two or more sets.
See <a href="https://github.com/gaospecial/ggVennDiagram/issues/64">https://github.com/gaospecial/ggVennDiagram/issues/64</a> for more details.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="UpsetPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <a href="ggupset.html#topic+scale_x_upset">ggupset::scale_x_upset</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object with Upset plot
</p>

<hr>
<h2 id='validate_common_args'>Validate common arguments</h2><span id='topic+validate_common_args'></span>

<h3>Description</h3>

<p>Validate common arguments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_common_args(
  seed,
  facet_by = NULL,
  plot_type = NULL,
  split_by = NULL,
  split_by_sep = "_",
  group_by = NULL,
  group_by_sep = "_",
  facet_scales = "fixed",
  facet_nrow = NULL,
  facet_ncol = NULL,
  facet_byrow = TRUE,
  theme = "theme_scp",
  theme_args = list(),
  palette = NULL,
  palcolor = NULL,
  expand = NULL,
  keep_empty = FALSE,
  alpha = 1,
  x_text_angle = 0,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate_common_args_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_group_by">group_by</code></td>
<td>
<p>Columns to group the data for plotting
For those plotting functions that do not support multiple groups,
They will be concatenated into one column, using <code>group_by_sep</code> as the separator</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_expand">expand</code></td>
<td>
<p>The values to expand the x and y axes. It is like CSS padding.
When a single value is provided, it is used for both axes on both sides.
When two values are provided, the first value is used for the top/bottom side and the second value is used for the left/right side.
When three values are provided, the first value is used for the top side, the second value is used for the left/right side, and the third value is used for the bottom side.
When four values are provided, the values are used for the top, right, bottom, and left sides, respectively.
You can also use a named vector to specify the values for each side.
When the axis is discrete, the values will be applied as 'add' to the 'expansion' function.
When the axis is continuous, the values will be applied as 'mult' to the 'expansion' function.
See also <a href="https://ggplot2.tidyverse.org/reference/expansion.html">https://ggplot2.tidyverse.org/reference/expansion.html</a></p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_keep_empty">keep_empty</code></td>
<td>
<p>A logical value indicating whether to keep empty groups.
If FALSE, empty groups will be removed.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_x_text_angle">x_text_angle</code></td>
<td>
<p>A numeric value specifying the angle of the x-axis text.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="validate_common_args_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>

<hr>
<h2 id='VennDiagramAtomic'>Atomic Venn diagram</h2><span id='topic+VennDiagramAtomic'></span>

<h3>Description</h3>

<p>Atomic Venn diagram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VennDiagramAtomic(
  data,
  in_form = NULL,
  group_by = NULL,
  group_by_sep = "_",
  id_by = NULL,
  label = "count",
  label_fg = "black",
  label_size = NULL,
  label_bg = "white",
  label_bg_r = 0.1,
  fill_mode = "count",
  fill_name = NULL,
  palette = ifelse(fill_mode == "set", "Paired", "Spectral"),
  palcolor = NULL,
  alpha = 1,
  theme = "theme_this",
  theme_args = list(),
  title = NULL,
  subtitle = NULL,
  legend.position = "right",
  legend.direction = "vertical",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VennDiagramAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_in_form">in_form</code></td>
<td>
<p>A character string indicating the datatype of the input data.
Possible values are &quot;long&quot;, &quot;wide&quot;, &quot;list&quot;, &quot;venn&quot; or NULL.
&quot;long&quot; indicates the data is in long format.
&quot;wide&quot; indicates the data is in wide format.
&quot;list&quot; indicates the data is a list.
&quot;venn&quot; indicates the data is a VennPlotData object.
NULL indicates the function will detect the datatype of the input data.
</p>
<p>A long format data would look like:
</p>
<pre>
group_by id_by
A        a1
A        a2
B        a1
B        a3
...
</pre>
<p>A wide format data would look like:
</p>
<pre>
A    B
TRUE TRUE
TRUE FALSE
FALSE TRUE
...
</pre>
<p>A list format data would look like:
</p>
<pre>
list(A = c("a1", "a2"), B = c("a1", "a3"))
</pre></td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_group_by">group_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to group the data.
When in_form is &quot;wide&quot;, it should be the columns for the groups.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_group_by_sep">group_by_sep</code></td>
<td>
<p>The separator for multiple group_by columns. See <code>group_by</code></p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_id_by">id_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to identify the instances.
Required when <code>group_by</code> is a single column and data is a data frame.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_label">label</code></td>
<td>
<p>A character string specifying the label to show on the Venn diagram.
Possible values are &quot;count&quot;, &quot;percent&quot;, &quot;both&quot;, &quot;none&quot; and a function.
&quot;count&quot; indicates the count of the intersection. &quot;percent&quot; indicates the percentage of the intersection.
&quot;both&quot; indicates both the count and the percentage of the intersection. &quot;none&quot; indicates no label.
If it is a function, if takes a data frame as input and returns a character vector as label.
The data frame has columns &quot;id&quot;, &quot;X&quot;, &quot;Y&quot;, &quot;name&quot;, &quot;item&quot; and &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string specifying the color of the label text.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value specifying the size of the label text.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string specifying the background color of the label.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the radius of the background of the label.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_fill_mode">fill_mode</code></td>
<td>
<p>A character string specifying the fill mode of the Venn diagram.
Possible values are &quot;count&quot;, &quot;set&quot;, &quot;count_rev&quot;.
&quot;count&quot; indicates the fill color is based on the count of the intersection.
&quot;set&quot; indicates the fill color is based on the set of the intersection.
&quot;count_rev&quot; indicates the fill color is based on the count of the intersection in reverse order.
The palette will be continuous for &quot;count&quot; and &quot;count_rev&quot;. The palette will be discrete for &quot;set&quot;.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_fill_name">fill_name</code></td>
<td>
<p>A character string to name the legend of colorbar.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="VennDiagramAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object with Venn diagram
</p>

<hr>
<h2 id='VolcanoPlot'>Volcano plot</h2><span id='topic+VolcanoPlot'></span>

<h3>Description</h3>

<p>A volcano plot is a type of scatter plot that shows statistical significance (usually on the y-axis) versus magnitude of change (usually on the x-axis).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VolcanoPlot(
  data,
  x,
  y,
  ytrans = function(n) -log10(n),
  color_by = NULL,
  color_name = NULL,
  xlim = NULL,
  flip_negatives = FALSE,
  x_cutoff = NULL,
  y_cutoff = 0.05,
  split_by = NULL,
  split_by_sep = "_",
  x_cutoff_name = NULL,
  y_cutoff_name = NULL,
  x_cutoff_color = "red2",
  y_cutoff_color = "blue2",
  x_cutoff_linetype = "dashed",
  y_cutoff_linetype = "dashed",
  x_cutoff_linewidth = 0.5,
  y_cutoff_linewidth = 0.5,
  pt_size = 2,
  pt_alpha = 0.5,
  nlabel = 5,
  labels = NULL,
  label_size = 3,
  label_fg = "black",
  label_bg = "white",
  label_bg_r = 0.1,
  highlight = NULL,
  highlight_color = "red",
  highlight_size = 2,
  highlight_alpha = 1,
  highlight_stroke = 0.5,
  trim = c(0, 1),
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VolcanoPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_ytrans">ytrans</code></td>
<td>
<p>A function to transform the y-axis values.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_color_by">color_by</code></td>
<td>
<p>A character vector of column names to color the points by.
If NULL, the points will be filled by the x and y cutoff value.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_color_name">color_name</code></td>
<td>
<p>A character string to name the legend of color.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_xlim">xlim</code></td>
<td>
<p>A numeric vector of length 2 to set the x-axis limits.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_flip_negatives">flip_negatives</code></td>
<td>
<p>A logical value to flip the y-axis for negative x values.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_x_cutoff">x_cutoff</code></td>
<td>
<p>A numeric value to set the x-axis cutoff.
Both negative and positive of this value will be used.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_y_cutoff">y_cutoff</code></td>
<td>
<p>A numeric value to set the y-axis cutoff.
Note that the y-axis cutoff will be transformed by <code>ytrans</code>.
So you should provide the original value.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_x_cutoff_name">x_cutoff_name</code></td>
<td>
<p>A character string to name the x-axis cutoff.
If &quot;none&quot;, the legend for the x-axis cutoff will not be shown.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_y_cutoff_name">y_cutoff_name</code></td>
<td>
<p>A character string to name the y-axis cutoff.
If &quot;none&quot;, the legend for the y-axis cutoff will not be shown.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_x_cutoff_color">x_cutoff_color</code></td>
<td>
<p>A character string to color the x-axis cutoff line.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_y_cutoff_color">y_cutoff_color</code></td>
<td>
<p>A character string to color the y-axis cutoff line.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_x_cutoff_linetype">x_cutoff_linetype</code></td>
<td>
<p>A character string to set the x-axis cutoff line type.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_y_cutoff_linetype">y_cutoff_linetype</code></td>
<td>
<p>A character string to set the y-axis cutoff line type.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_x_cutoff_linewidth">x_cutoff_linewidth</code></td>
<td>
<p>A numeric value to set the x-axis cutoff line size.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_y_cutoff_linewidth">y_cutoff_linewidth</code></td>
<td>
<p>A numeric value to set the y-axis cutoff line size.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to set the point size.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value to set the point transparency.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_nlabel">nlabel</code></td>
<td>
<p>A numeric value to set the number of labels to show.
The points will be ordered by the distance to the origin. Top <code>nlabel</code> points will be labeled.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_labels">labels</code></td>
<td>
<p>A character vector of row names or indexes to label the points.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value to set the label size.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string to set the label color.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string to set the label background color.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the radius of the background of the label.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of row names or indexes to highlight the points.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string to set the highlight color.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value to set the highlight size.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value to set the highlight transparency.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_highlight_stroke">highlight_stroke</code></td>
<td>
<p>A numeric value to set the highlight stroke size.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_trim">trim</code></td>
<td>
<p>A numeric vector of length 2 to trim the x-axis values.
The values must be in the range from 0 to 1, which works as quantile to trim the x-axis values.
For example, c(0.01, 0.99) will trim the 1% and 99% quantile of the x-axis values.
If the values are less then 1% or greater than 99% quantile, the values will be set to the 1% or 99% quantile.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="VolcanoPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot objects or a wrap_plots object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(8525)
# Obtained by Seurat::FindMakers for the first cluster of pbmc_small
data &lt;- data.frame(
   avg_log2FC = c(
     -3.69, -4.10, -2.68, -3.51, -3.09, -2.52, -3.53, -3.35, -2.82, -2.71, -3.16, -2.24,
     -5.62, -3.10, -3.42, -2.72, -3.23, -3.25, -4.68, 3.67, -2.66, 4.79, -2.99, 10.14,
     -1.78, -2.67, -2.26, -2.59, -3.39, 5.36, 4.56, 4.62, -2.94, -9.47, -9.12, -1.63,
     -2.77, 3.31, -1.53, -3.89, -4.21, 4.72, -2.98, -2.29, -1.41, -9.48, -4.30, 3.01,
     -1.19, -4.83, -1.35, -1.68, -1.63, -2.70, 3.86, 3.81, 7.23, -1.45, -0.92, -2.45,
     3.91, -4.45, -9.33, 3.56, 2.27, -1.60, -1.15, 11.40, -9.77, -8.32, 2.61, -1.25,
     -1.72, 10.61, 11.34, 10.02, 2.78, -3.48, -1.98, 5.86, 5.57, 4.57, 9.75, 9.97,
     10.90, 9.19, 2.93, 5.10, -1.52, -3.93, -1.95, -2.46, -0.64, 4.60, -1.82, -0.80,
     9.34, 7.51, 6.45, 5.23, 4.41, 3.60, -1.94, -1.15),
   p_val_adj = c(
     3.82e-09, 1.52e-07, 1.79e-07, 4.68e-07, 4.83e-07, 6.26e-07, 2.61e-06, 1.33e-05,
     1.79e-05, 3.71e-05, 5.21e-05, 5.36e-05, 5.83e-05, 6.66e-05, 8.22e-05, 2.89e-04,
     3.00e-04, 4.94e-04, 7.62e-04, 8.93e-04, 9.55e-04, 9.61e-04, 1.12e-03, 1.47e-03,
     1.66e-03, 1.95e-03, 2.06e-03, 3.01e-03, 3.26e-03, 4.35e-03, 4.85e-03, 5.12e-03,
     5.40e-03, 7.18e-03, 7.18e-03, 1.04e-02, 1.24e-02, 1.90e-02, 1.94e-02, 1.97e-02,
     2.09e-02, 2.13e-02, 2.25e-02, 2.61e-02, 3.18e-02, 3.27e-02, 3.69e-02, 3.80e-02,
     4.95e-02, 5.73e-02, 5.77e-02, 6.10e-02, 6.22e-02, 6.31e-02, 6.72e-02, 9.23e-02,
     9.85e-02, 1.06e-01, 1.07e-01, 1.11e-01, 1.31e-01, 1.38e-01, 1.40e-01, 1.43e-01,
     2.00e-01, 2.39e-01, 2.49e-01, 2.57e-01, 2.86e-01, 2.86e-01, 2.98e-01, 3.32e-01,
     4.15e-01, 4.91e-01, 4.91e-01, 4.91e-01, 5.97e-01, 7.11e-01, 7.59e-01, 8.38e-01,
     9.20e-01, 9.20e-01, 9.29e-01, 9.29e-01, 9.29e-01, 9.29e-01, 9.34e-01, 9.68e-01,
     1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00,
     1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00, 1.00e+00),
   gene = c(
     "HLA-DPB1", "LYZ", "HLA-DRA", "TYMP", "HLA-DPA1", "HLA-DRB1", "CST3", "HLA-DQB1",
     "HLA-DRB5", "LST1", "HLA-DQA1", "AIF1", "S100A8", "IFITM3", "HLA-DMB", "FCGRT",
     "SERPINA1", "IFI30", "S100A9", "CCL5", "GRN", "LCK", "HLA-DMA", "MS4A6A", "CTSS",
     "CFP", "FCN1", "BID", "CFD", "CD3D", "CD7", "CD3E", "LGALS2", "CD14", "SMCO4",
     "LINC00936", "HCK", "CTSW", "LGALS1", "HLA-DQA2", "LRRC25", "GZMM", "RNF130",
     "LGALS3", "S100A11", "C5AR1", "IL1B", "GZMA", "FCER1G", "MPEG1", "TYROBP", "TSPO",
     "GSTP1", "CTSB", "IL32", "CD247", "GNLY", "COTL1", "NFKBIA", "NUP214", "LAMP1",
     "FPR1", "CLEC10A", "CST7", "PRF1", "BLVRA", "PSAP", "GZMH", "EAF2", "ASGR1",
     "RARRES3", "SAT1", "LY86", "GP9", "TUBB1", "NGFRAP1", "XBP1", "SCO2", "RGS2", "GZMB",
     "HIST1H2AC", "KLRD1", "PGRMC1", "AKR1C3", "PTGDR", "IL2RB", "GYPC", "CCL4", "CD68",
     "FCER1A", "CD79B", "MS4A7", "CARD16", "ACAP1", "CD79A", "ANXA2", "TMEM40", "PF4",
     "GNG11", "CLU", "CD9", "FGFBP2", "TNFRSF1B", "IFI6"),
  pct_diff = c(
     -0.752, -0.457, -0.460, -0.671, -0.626, -0.701, -0.502, -0.619, -0.623, -0.598,
     -0.566, -0.626, -0.543, -0.566, -0.541, -0.542, -0.515, -0.489, -0.444, 0.428,
     -0.517, 0.461, -0.491, -0.410, -0.480, -0.491, -0.521, -0.491, -0.438, 0.411,
     0.411, 0.409, -0.438, -0.359, -0.359, -0.440, -0.386, 0.385, -0.332, -0.361, -0.361,
     0.364, -0.387, -0.415, -0.454, -0.308, -0.335, 0.364, -0.454, -0.309, -0.379, -0.427,
     -0.377, -0.389, 0.335, 0.315, 0.313, -0.284, -0.502, -0.309, 0.313, -0.284, -0.256,
     0.309, 0.313, -0.364, -0.406, 0.244, -0.231, -0.231, 0.281, -0.311, -0.312, 0.220,
     0.220, 0.220, 0.261, -0.232, -0.367, 0.240, 0.218, 0.218, 0.195, 0.195, 0.195, 0.195,
     0.262, 0.218, -0.288, -0.207, -0.290, -0.233, -0.367, 0.217, -0.233, -0.403, 0.171,
     0.194, 0.194, 0.194, 0.194, 0.213, -0.235, -0.292),
  group = sample(LETTERS[1:2], 104, replace = TRUE)
)
rownames(data) &lt;- data$gene

VolcanoPlot(data, x = "avg_log2FC", y = "p_val_adj", color_by = "pct_diff",
   y_cutoff_name = "-log10(0.05)")
VolcanoPlot(data, x = "avg_log2FC", y = "p_val_adj", y_cutoff_name = "none",
   flip_negatives = TRUE)
VolcanoPlot(data, x = "avg_log2FC", y = "p_val_adj", y_cutoff_name = "none",
   flip_negatives = TRUE, facet_by = "group")
VolcanoPlot(data, x = "avg_log2FC", y = "p_val_adj", y_cutoff_name = "none",
   flip_negatives = TRUE, split_by = "group")
VolcanoPlot(data, x = "avg_log2FC", y = "p_val_adj", y_cutoff_name = "none",
   highlight = c("ANXA2", "TMEM40", "PF4", "GNG11", "CLU", "CD9", "FGFBP2",
   "TNFRSF1B", "IFI6"))
VolcanoPlot(data, x = "avg_log2FC", y = "p_val_adj", color_by = "pct_diff",
   y_cutoff_name = "-log10(0.05)", split_by = "group",
   palette = c(A = "Set1", B = "Dark2"))

</code></pre>

<hr>
<h2 id='VolcanoPlotAtomic'>Atomic volcano plot</h2><span id='topic+VolcanoPlotAtomic'></span>

<h3>Description</h3>

<p>Atomic volcano plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VolcanoPlotAtomic(
  data,
  x,
  y,
  ytrans = function(n) -log10(n),
  color_by = NULL,
  color_name = NULL,
  flip_negatives = FALSE,
  x_cutoff = NULL,
  y_cutoff = 0.05,
  trim = c(0, 1),
  xlim = NULL,
  x_cutoff_name = NULL,
  y_cutoff_name = NULL,
  x_cutoff_color = "red2",
  y_cutoff_color = "blue2",
  x_cutoff_linetype = "dashed",
  y_cutoff_linetype = "dashed",
  x_cutoff_linewidth = 0.5,
  y_cutoff_linewidth = 0.5,
  pt_size = 2,
  pt_alpha = 0.5,
  nlabel = 5,
  labels = NULL,
  label_size = 3,
  label_fg = "black",
  label_bg = "white",
  label_bg_r = 0.1,
  highlight = NULL,
  highlight_color = "red",
  highlight_size = 2,
  highlight_alpha = 1,
  highlight_stroke = 0.5,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  title = NULL,
  subtitle = NULL,
  xlab = NULL,
  ylab = NULL,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VolcanoPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_x">x</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the x-axis.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_y">y</code></td>
<td>
<p>A character string specifying the column name of the data frame to plot for the y-axis.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_ytrans">ytrans</code></td>
<td>
<p>A function to transform the y-axis values.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_color_by">color_by</code></td>
<td>
<p>A character vector of column names to color the points by.
If NULL, the points will be filled by the x and y cutoff value.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_color_name">color_name</code></td>
<td>
<p>A character string to name the legend of color.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_flip_negatives">flip_negatives</code></td>
<td>
<p>A logical value to flip the y-axis for negative x values.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_x_cutoff">x_cutoff</code></td>
<td>
<p>A numeric value to set the x-axis cutoff.
Both negative and positive of this value will be used.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_y_cutoff">y_cutoff</code></td>
<td>
<p>A numeric value to set the y-axis cutoff.
Note that the y-axis cutoff will be transformed by <code>ytrans</code>.
So you should provide the original value.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_trim">trim</code></td>
<td>
<p>A numeric vector of length 2 to trim the x-axis values.
The values must be in the range from 0 to 1, which works as quantile to trim the x-axis values.
For example, c(0.01, 0.99) will trim the 1% and 99% quantile of the x-axis values.
If the values are less then 1% or greater than 99% quantile, the values will be set to the 1% or 99% quantile.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_xlim">xlim</code></td>
<td>
<p>A numeric vector of length 2 to set the x-axis limits.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_x_cutoff_name">x_cutoff_name</code></td>
<td>
<p>A character string to name the x-axis cutoff.
If &quot;none&quot;, the legend for the x-axis cutoff will not be shown.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_y_cutoff_name">y_cutoff_name</code></td>
<td>
<p>A character string to name the y-axis cutoff.
If &quot;none&quot;, the legend for the y-axis cutoff will not be shown.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_x_cutoff_color">x_cutoff_color</code></td>
<td>
<p>A character string to color the x-axis cutoff line.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_y_cutoff_color">y_cutoff_color</code></td>
<td>
<p>A character string to color the y-axis cutoff line.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_x_cutoff_linetype">x_cutoff_linetype</code></td>
<td>
<p>A character string to set the x-axis cutoff line type.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_y_cutoff_linetype">y_cutoff_linetype</code></td>
<td>
<p>A character string to set the y-axis cutoff line type.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_x_cutoff_linewidth">x_cutoff_linewidth</code></td>
<td>
<p>A numeric value to set the x-axis cutoff line size.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_y_cutoff_linewidth">y_cutoff_linewidth</code></td>
<td>
<p>A numeric value to set the y-axis cutoff line size.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_pt_size">pt_size</code></td>
<td>
<p>A numeric value to set the point size.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_pt_alpha">pt_alpha</code></td>
<td>
<p>A numeric value to set the point transparency.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_nlabel">nlabel</code></td>
<td>
<p>A numeric value to set the number of labels to show.
The points will be ordered by the distance to the origin. Top <code>nlabel</code> points will be labeled.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_labels">labels</code></td>
<td>
<p>A character vector of row names or indexes to label the points.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_label_size">label_size</code></td>
<td>
<p>A numeric value to set the label size.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_label_fg">label_fg</code></td>
<td>
<p>A character string to set the label color.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_label_bg">label_bg</code></td>
<td>
<p>A character string to set the label background color.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_label_bg_r">label_bg_r</code></td>
<td>
<p>A numeric value specifying the radius of the background of the label.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of row names or indexes to highlight the points.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_highlight_color">highlight_color</code></td>
<td>
<p>A character string to set the highlight color.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_highlight_size">highlight_size</code></td>
<td>
<p>A numeric value to set the highlight size.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_highlight_alpha">highlight_alpha</code></td>
<td>
<p>A numeric value to set the highlight transparency.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_highlight_stroke">highlight_stroke</code></td>
<td>
<p>A numeric value to set the highlight stroke size.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_xlab">xlab</code></td>
<td>
<p>A character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_ylab">ylab</code></td>
<td>
<p>A character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="VolcanoPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='WordCloudPlot'>Word Cloud Plot</h2><span id='topic+WordCloudPlot'></span>

<h3>Description</h3>

<p>Word cloud plot to illustrate the count/frequency of words.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WordCloudPlot(
  data,
  word_by = NULL,
  sentence_by = NULL,
  count_by = NULL,
  score_by = NULL,
  count_name = NULL,
  score_name = NULL,
  split_by = NULL,
  split_by_sep = "_",
  words_excluded = plotthis::words_excluded,
  score_agg = mean,
  minchar = 2,
  word_size = c(2, 8),
  top_words = 100,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Spectral",
  palcolor = NULL,
  alpha = 1,
  palreverse = FALSE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  combine = TRUE,
  nrow = NULL,
  ncol = NULL,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WordCloudPlot_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_word_by">word_by</code></td>
<td>
<p>A character string of the column name to use as the word.
A character column is expected.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_sentence_by">sentence_by</code></td>
<td>
<p>A character string of the column name to split the sentence.
A character column is expected.
Either <code>word_by</code> or <code>sentence_by</code> should be specified.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_count_by">count_by</code></td>
<td>
<p>A character string of the column name for the count of the word/sentence.
A numeric column is expected.
If NULL, the count of the word/sentence will be used.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_score_by">score_by</code></td>
<td>
<p>A character string of the column name for the score of the word/sentence.
A numeric column is expected, used for the color of the word cloud.
If NULL, the score will be set to 1.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_count_name">count_name</code></td>
<td>
<p>A character string to name the legend of count.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_score_name">score_name</code></td>
<td>
<p>A character string to name the legend of score.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_split_by">split_by</code></td>
<td>
<p>The column(s) to split data by and plot separately.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_split_by_sep">split_by_sep</code></td>
<td>
<p>The separator for multiple split_by columns. See <code>split_by</code></p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_words_excluded">words_excluded</code></td>
<td>
<p>A character vector of words to exclude from the word cloud.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_score_agg">score_agg</code></td>
<td>
<p>A function to aggregate the scores. Default is <code>mean</code>.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_minchar">minchar</code></td>
<td>
<p>A numeric value specifying the minimum number of characters for the word.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_word_size">word_size</code></td>
<td>
<p>A numeric vector specifying the range of the word size.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_top_words">top_words</code></td>
<td>
<p>A numeric value specifying the number of top words to show.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_palreverse">palreverse</code></td>
<td>
<p>A logical value to reverse the palette colors.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_combine">combine</code></td>
<td>
<p>Whether to combine the plots into one when facet is FALSE. Default is TRUE.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_nrow">nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_ncol">ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_byrow">byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row.</p>
</td></tr>
<tr><td><code id="WordCloudPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object or wrap_plots object or a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(
   word = c("apple", "banana", "cherry", "date", "elderberry"),
   count = c(10, 20, 30, 40, 50),
   score = c(1, 2, 3, 4, 5)
)
WordCloudPlot(data, word_by = "word", count_by = "count", score_by = "score")
</code></pre>

<hr>
<h2 id='WordCloudPlotAtomic'>Word cloud without data splitting</h2><span id='topic+WordCloudPlotAtomic'></span>

<h3>Description</h3>

<p>Word cloud without data splitting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WordCloudPlotAtomic(
  data,
  word_by = NULL,
  sentence_by = NULL,
  count_by = NULL,
  score_by = NULL,
  count_name = NULL,
  score_name = NULL,
  words_excluded = plotthis::words_excluded,
  score_agg = mean,
  minchar = 2,
  word_size = c(2, 8),
  top_words = 100,
  facet_by = NULL,
  facet_scales = "fixed",
  facet_ncol = NULL,
  facet_nrow = NULL,
  facet_byrow = TRUE,
  theme = "theme_this",
  theme_args = list(),
  palette = "Paired",
  palcolor = NULL,
  alpha = 1,
  palreverse = FALSE,
  aspect.ratio = 1,
  legend.position = "right",
  legend.direction = "vertical",
  title = NULL,
  subtitle = NULL,
  seed = 8525,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WordCloudPlotAtomic_+3A_data">data</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_word_by">word_by</code></td>
<td>
<p>A character string of the column name to use as the word.
A character column is expected.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_sentence_by">sentence_by</code></td>
<td>
<p>A character string of the column name to split the sentence.
A character column is expected.
Either <code>word_by</code> or <code>sentence_by</code> should be specified.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_count_by">count_by</code></td>
<td>
<p>A character string of the column name for the count of the word/sentence.
A numeric column is expected.
If NULL, the count of the word/sentence will be used.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_score_by">score_by</code></td>
<td>
<p>A character string of the column name for the score of the word/sentence.
A numeric column is expected, used for the color of the word cloud.
If NULL, the score will be set to 1.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_count_name">count_name</code></td>
<td>
<p>A character string to name the legend of count.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_score_name">score_name</code></td>
<td>
<p>A character string to name the legend of score.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_words_excluded">words_excluded</code></td>
<td>
<p>A character vector of words to exclude from the word cloud.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_score_agg">score_agg</code></td>
<td>
<p>A function to aggregate the scores. Default is <code>mean</code>.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_minchar">minchar</code></td>
<td>
<p>A numeric value specifying the minimum number of characters for the word.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_word_size">word_size</code></td>
<td>
<p>A numeric vector specifying the range of the word size.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_top_words">top_words</code></td>
<td>
<p>A numeric value specifying the number of top words to show.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_facet_by">facet_by</code></td>
<td>
<p>A character string specifying the column name of the data frame to facet the plot.
Otherwise, the data will be split by <code>split_by</code> and generate multiple plots
and combine them into one using <code>patchwork::wrap_plots</code></p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_facet_scales">facet_scales</code></td>
<td>
<p>Whether to scale the axes of facets. Default is &quot;fixed&quot;
Other options are &quot;free&quot;, &quot;free_x&quot;, &quot;free_y&quot;. See <code>ggplot2::facet_wrap</code></p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>A numeric value specifying the number of columns in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_facet_nrow">facet_nrow</code></td>
<td>
<p>A numeric value specifying the number of rows in the facet.
When facet_by is a single column and facet_wrap is used.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_facet_byrow">facet_byrow</code></td>
<td>
<p>A logical value indicating whether to fill the plots by row. Default is TRUE.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_theme">theme</code></td>
<td>
<p>A character string or a theme class (i.e. ggplot2::theme_classic) specifying the theme to use.
Default is &quot;theme_this&quot;.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_theme_args">theme_args</code></td>
<td>
<p>A list of arguments to pass to the theme function.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_palette">palette</code></td>
<td>
<p>A character string specifying the palette to use.
A named list or vector can be used to specify the palettes for different <code>split_by</code> values.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_palcolor">palcolor</code></td>
<td>
<p>A character string specifying the color to use in the palette.
A named list can be used to specify the colors for different <code>split_by</code> values.
If some values are missing, the values from the palette will be used (palcolor will be NULL for those values).</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the transparency of the plot.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_palreverse">palreverse</code></td>
<td>
<p>A logical value to reverse the palette colors.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>A numeric value specifying the aspect ratio of the plot.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_legend.position">legend.position</code></td>
<td>
<p>A character string specifying the position of the legend.
if <code>waiver()</code>, for single groups, the legend will be &quot;none&quot;, otherwise &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_legend.direction">legend.direction</code></td>
<td>
<p>A character string specifying the direction of the legend.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_title">title</code></td>
<td>
<p>A character string specifying the title of the plot.
A function can be used to generate the title based on the default title.
This is useful when split_by is used and the title needs to be dynamic.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_subtitle">subtitle</code></td>
<td>
<p>A character string specifying the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_seed">seed</code></td>
<td>
<p>The random seed to use. Default is 8525.</p>
</td></tr>
<tr><td><code id="WordCloudPlotAtomic_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='words_excluded'>Excluded words in keyword enrichment analysis and extraction</h2><span id='topic+words_excluded'></span>

<h3>Description</h3>

<p>The variable &quot;words_excluded&quot; represents the words that are excluded during keyword enrichment analysis or keyword extraction process. These mainly include words that are excessively redundant or of little value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (interactive()) {
  words_excluded &lt;- c(
    "the", "is", "and", "or", "a", "in", "on", "under", "between", "of", "through",
    "via", "along", "that", "for", "with", "within", "without", "cell", "cellular",
    "dna", "rna", "protein", "peptide", "amino", "acid", "development", "involved",
    "organization", "system", "regulation", "regulated", "positive", "negative",
    "response", "process", "processing", "small", "large", "change", "disease"
  )
}

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
