<!DOCTYPE html><html lang="en"><head><title>Help for package plotprotein</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {plotprotein}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#plotprotein-package'>
<p>Development of Visualization Tools for Protein Sequence</p></a></li>
<li><a href='#conservation'>
<p>conservation</p></a></li>
<li><a href='#domain_data'>
<p>downloading protein length</p></a></li>
<li><a href='#length_data'>
<p>downloading protein length</p></a></li>
<li><a href='#plotdomain'>
<p>ploting domain</p></a></li>
<li><a href='#plotmutagensis'>
<p>ploting mutagensis</p></a></li>
<li><a href='#plotprotein'>
<p>ploting protein</p></a></li>
<li><a href='#plotsite'>
<p>ploting site</p></a></li>
<li><a href='#site_data'>
<p>downloading protein site</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Development of Visualization Tools for Protein Sequence</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaoyu Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yao Geng &lt;gengyao0103521@qq.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The image of the amino acid transform on the protein level is drawn, and the automatic routing of the functional elements such as the domain and the mutation site is completed.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>XML, plyr, plotrix, seqinr, ade4</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-05-09 00:05:37 UTC; Administrator</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-05-09 07:06:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='plotprotein-package'>
Development of Visualization Tools for Protein Sequence
</h2><span id='topic+plotprotein-package'></span>

<h3>Description</h3>

<p>The image of the amino acid transform on the protein level is drawn, and the automatic routing of the functional elements such as the domain and the mutation site is completed.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> plotprotein</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Development of Visualization Tools for Protein Sequence</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-04-17</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Xiaoyu Zhang</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Yao Geng &lt;gengyao0103521@qq.com&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> The image of the amino acid transform on the protein level is drawn, and the automatic routing of the functional elements such as the domain and the mutation site is completed.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> 
XML,
plyr,
plotrix,
seqinr,
ade4</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Index of help topics:
</p>
<pre>
conservation            conservation
domain_data             downloading protein length
length_data             downloading protein length
plotdomain              ploting domain
plotmutagensis          ploting mutagensis
plotprotein             ploting protein
plotprotein-package     Development of Visualization Tools for Protein
                        Sequence
plotsite                ploting site
site_data               downloading protein site
</pre>

<h3>Author(s)</h3>

<p>Xiaoyu Zhang
Maintainer: Yao Geng &lt;gengyao0103521@qq.com&gt;</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>

<hr>
<h2 id='conservation'>
conservation
</h2><span id='topic+conservation'></span>

<h3>Description</h3>

<p>Draw a conservative curve, calculate the conservative score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conservation()
</code></pre>


<h3>Details</h3>

<p>The tool ennable visualization of amino acid changes at the protein level,The scale of a protein domain and the position of a functional motif/site will be precisely defined. The features available includeting conservation, conservation score
</p>


<h3>Value</h3>

<p>The returned value is a conservative score
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p><code><a href="utils.html#topic+help">help</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    domain = read.table("Domain.txt", sep = "\t", stringsAsFactors = F)
    length = read.table("Length.txt", sep = "\t", stringsAsFactors = F)
    site = read.table("Site.txt", sep = "\t", stringsAsFactors = F)
    muta = read.table("Mutagenesis.txt", sep = "\t", stringsAsFactors = F)
    option = read.table("Option.txt", sep = "\t", stringsAsFactors = F)
    zoomin = read.table("ZoomIn.txt", sep = "\t", stringsAsFactors = F)
    nameOfYourQuery = option[2, 1]
    additionalOptions = option[2, 2]
    showReferenceSequence = option[2, 3]
    showConservationScore = option[2, 4]
    showGridlinesAtTicks = option[2, 5]
    conservation = option[2, 6]
    zoomIn = zoomin[2, 1]
    zoomStart = zoomin[2, 2]
    zoomEnd = zoomin[2, 3]
    tickSize = as.numeric(zoomin[2, 4])
    referenceSequencePositionInFile = option[2, 7]
    option = read.table("Option.txt", sep = "\t", stringsAsFactors = F)
    a &lt;- read.fasta(file = "alignmentFile.fasta")
    seq &lt;- list()
    for (i in 1:length(a)) {
        seq[[i]] &lt;- a[[i]][1:length(a[[i]])]
    }
    numberOfSeq &lt;- length(seq)
    mat &lt;- matrix(0, nrow = length(a), ncol = length(a[[1]]))
    for (i in 1:length(seq)) {
        mat[i, ] &lt;- seq[[i]]
    }
    df &lt;- as.data.frame(mat)
    tdf &lt;- t(df)
    referenceSequencePositionInFile = option[2, 7]
    referenceSeq &lt;- tdf[which(tdf[, as.numeric(referenceSequencePositionInFile)] != 
        "-"), ]
    referenceSeq &lt;- as.data.frame(referenceSeq)
    write.table(referenceSeq, file = "alignment_table", sep = "\t", 
        quote = F, row.names = F, col.names = F)
    counter &lt;- rep(0, nrow(referenceSeq))
    a &lt;- read.table("alignment_table", sep = "\t")
    a &lt;- data.frame(lapply(a, as.character), stringsAsFactors = FALSE)
    for (i in 1:nrow(a)) {
        a[i, "consensus"] &lt;- paste(as.character(a[i, ]), collapse = "")
    }
    countBases &lt;- function(string) {
        table(strsplit(string, "")[[1]])
    }
    c &lt;- as.character(a[, "consensus"])
    tab &lt;- list()
    for (i in 1:length(c)) {
        tab[[i]] &lt;- countBases(c[i])
    }
    score &lt;- rep(0, nrow(a))
    for (i in 1:length(tab)) {
        for (j in 1:length(tab[[i]])) {
            if ((names(tab[[i]][j])) == a[i, ][as.numeric(referenceSequencePositionInFile)]) 
                score[i] &lt;- tab[[i]][j]
        }
    }
    scorePlot &lt;- -(((score/numberOfSeq)))
    a &lt;- read.fasta(file = "alignmentFile.fasta")
    seqForPlot &lt;- a[[as.numeric(referenceSequencePositionInFile)]][
which(a[[as.numeric(referenceSequencePositionInFile)]] != 
        "-")]
    if (additionalOptions == "yes") {
        if (conservation == "yes") {
            lines(scorePlot, col = "purple3")
        }
    }
    if (additionalOptions == "yes") {
        if (showReferenceSequence == "yes") {
            rect(0, -4.75, length(scorePlot), -5.05, col = "white", 
                border = NA)
            for (i in 1:length(seqForPlot)) {
                text(i, -4.9, toupper(seqForPlot[i]), font = 2, 
                  cex = 1)
            }
        }
    }
    if (additionalOptions == "yes") {
        if (showConservationScore == "yes") {
            rect(0, 0.3, length(scorePlot), 0.7, col = "white", 
                border = NA)
            for (i in 1:length(seqForPlot)) {
                text(i, 0.5, toupper(abs(round(scorePlot[i], 
                  1))), font = 2, cex = 0.8, srt = 90, col = "purple3")
            }
        }
    }
  }
</code></pre>

<hr>
<h2 id='domain_data'>
downloading protein length
</h2><span id='topic+domain_data'></span>

<h3>Description</h3>

<p>Load the start and end positions of the domain
</p>


<h3>Usage</h3>

<pre><code class='language-R'>domain_data()
</code></pre>


<h3>Details</h3>

<p>The tool ennable visualization of amino acid changes at the protein level,The scale of a protein domain and the position of a functional motif/site will be precisely defined. The features available include domains
</p>


<h3>Value</h3>

<p>The start and end positions of the domain
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    library(XML)
    library(plyr)
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    name = protein[2]
    url_p = "http://www.uniprot.org/uniprot/"
    url_s = "#showFeatures"
    url_w = paste(url_p, name, url_s, sep = "")
    url = url_w
    doc &lt;- htmlParse(url)
    position_d = xpathSApply (doc, "//table[@id= 'domainsAnno_section'] 
  /tr/td/  a[@class = 'position tooltipped']",  
        xmlValue)
    name_d = xpathSApply (doc, "//table[@id= 'domainsAnno_section']/tr/td/span[@property='text']", 
        xmlValue)
    s_d = c()
    for (i in 1:length(position_d)) {
        s_d[i] &lt;- gsub(pattern = "//D", replacement = "x", position_d[i])
    }
    s_d &lt;- strsplit(s_d, "xxx")
    d1_d &lt;- laply(s_d, function(x) x[1])
    d2_d &lt;- laply(s_d, function(x) x[2])
    r1_d = d1_d
    r2_d = d2_d
    r3_d = name_d
    dfrm_d = data.frame(r1_d, r2_d, r3_d)
    write.table(dfrm_d, file = "Domain.txt", sep = "/t", quote = FALSE, 
        row.names = F, col.names = F)
  }
</code></pre>

<hr>
<h2 id='length_data'>
downloading protein length
</h2><span id='topic+length_data'></span>

<h3>Description</h3>

<p>Download the length of the protein, including the starting and ending positions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>length_data()
</code></pre>


<h3>Details</h3>

<p>Download the length of the protein, including the starting and ending positions
</p>


<h3>Value</h3>

<p>The length of the protein
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    library(XML)
    library(plyr)
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    name = protein[2]
    url_p = "http://www.uniprot.org/uniprot/"
    url_s = "#showFeatures"
    url_w = paste(url_p, name, url_s, sep = "")
    url = url_w
    doc &lt;- htmlParse(url)
    position_l = xpathSApply (doc, "//table[@id= 'peptides_section'] 
  /tr/td/  a[@class = 'position tooltipped']",  
        xmlValue)
    s_l &lt;- c()
    for (i in 1:length(position_l)) {
        s_l[i] &lt;- gsub(pattern = "//D", replacement = "x", position_l[i])
    }
    s_l &lt;- strsplit(s_l, "xxx")
    d2_l &lt;- laply(s_l, function(x) x[2])
    r1_l &lt;- 0
    r2_l &lt;- d2_l
    dfrm_l &lt;- data.frame(r1_l, r2_l)
    write.table(dfrm_l, file = "Length.txt", sep = "/t", quote = FALSE, 
        row.names = F, col.names = F)
  }
</code></pre>

<hr>
<h2 id='plotdomain'>
ploting domain
</h2><span id='topic+plotdomain'></span>

<h3>Description</h3>

<p>Draw the domain of the protein
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotdomain()
</code></pre>


<h3>Details</h3>

<p>The tool ennable visualization of amino acid changes at the protein level,The scale of a protein domain and the position of a functional motif/site will be precisely defined. The features available include domains
</p>


<h3>Value</h3>

<p>The starting position, end position and name of the protein domain
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    domain = read.table("Domain.txt", sep = "\t", stringsAsFactors = F)
    length = read.table("Length.txt", sep = "\t", stringsAsFactors = F)
    site = read.table("Site.txt", sep = "\t", stringsAsFactors = F)
    muta = read.table("Mutagenesis.txt", sep = "\t", stringsAsFactors = F)
    option = read.table("Option.txt", sep = "\t", stringsAsFactors = F)
    zoomin = read.table("ZoomIn.txt", sep = "\t", stringsAsFactors = F)
    Domain = function(start, end, name, height = -0.3, color = "orange", 
        face = "stereoscopic", protein_width, x_y) {
        h1 = -2.8
        h2 = -3.1
        dec = 2 * nchar(name) * protein_width/100
        if (face == "stereoscopic") {
            cylindrect(start, h1, end, h2, col = color, gradient = "y")
        }
        else {
            rect(start, h1, end, h2, col = color)
        }
        if (end - start &gt;= dec) {
            par(srt = 0)
            text((end + start)/2, h1 + height/2, name, cex = 0.7)
            isContain = TRUE
        }
        else {
            isContain = FALSE
        }
        isContain
    }
    Domain_w = function(domain_pos, domain_name, protein_width) {
        dec = 1.4 * protein_width/100
        position2 = 1:length(domain_pos)
        position2[1] = domain_pos[1]
        if (length(domain_pos) &gt; 1) {
            for (i in 2:length(domain_pos)) {
                if (domain_pos[i] - domain_pos[i - 1] &lt;= dec) {
                  if (domain_pos[i] != domain_pos[i - 1]) {
                    position2[i] = position2[i - 1] + dec
                  }
                  else {
                    position2[i] = position2[i - 1]
                  }
                }
                else {
                  position2[i] = domain_pos[i]
                }
            }
        }
        return(position2)
    }
    Domain_h = function(position, position2, name, height = -0.3, 
        x_y, up_down) {
        h1 = -0.1
        h2 = -0.2
        h = -0.4
        hh1 = -2.8
        if (up_down == "up") {
            if (position == position2) {
                segments(position, hh1 + height, position, hh1 + 
                  height + h)
            }
            else {
                segments(position, hh1 + height, position, hh1 + 
                  height + h1)
                segments(position2, hh1 + height + h - h2, position2, 
                  hh1 + height + h)
                segments(position, hh1 + height + h1, position2, 
                  hh1 + height + h - h2)
            }
            text(position2, hh1 + height + h - 0.02, name, srt = 90, 
                adj = c(0, 0.5), cex = 0.8)
        }
        else {
            if (position == position2) {
                segments(position, hh1, position, hh1 - h)
            }
            else {
                segments(position, hh1, position, hh1 - h1)
                segments(position2, hh1 - h + h2, position2, 
                  hh1 - h)
                segments(position, hh1 - h1, position2, hh1 - 
                  h + h2)
            }
            text(position2, hh1 - h + 0.02, name, srt = 270, 
                adj = c(0, 0.5), cex = 0.8)
        }
    }
    if (!is.na(domain[1, 1])) {
        domainn = domain
        count = 0
        for (i in 1:nrow(domainn)) {
            isContain = Domain(start = as.numeric(domainn[i, 
                1]), end = as.numeric(domainn[i, 2]), name = as.character(domainn[i, 
                3]), height = as.numeric(protein[4]), color = i + 
                1, face = protein[6], protein_width = as.numeric(length[2]), 
                x_y = flag)
            if (isContain == TRUE) {
                domain = domain[-i + count, ]
                count = count + 1
            }
        }
        domain2 = (domain[, 1] + domain[, 2])/2
        if (length(domain2) != 0) {
            flag = TRUE
            if (flag == TRUE) {
                position3 = Domain_w(domain2, domain[, 3], as.numeric(length[2]))
            }
            for (i in 1:nrow(domain)) {
                position1 = (as.numeric(domain[i, 1]) + as.numeric(domain[i, 
                  2]))/2
                Domain_h(position = position1, position2 = position3[i], 
                  name = as.character(domain[i, 3]), height = as.numeric(protein[4]), 
                  x_y = flag, up_down = "down")
            }
        }
    }
  }
</code></pre>

<hr>
<h2 id='plotmutagensis'>
ploting mutagensis
</h2><span id='topic+plotmutagensis'></span>

<h3>Description</h3>

<p>Draw the mutagensis of the protein
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotmutagensis()
</code></pre>


<h3>Details</h3>

<p>The tool ennable visualization of amino acid changes at the protein level,The scale of a protein domain and the position of a functional motif/site will be precisely defined. The features available include mutagensis
</p>


<h3>Value</h3>

<p>The location, height and name of the transition point
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    domain = read.table("Domain.txt", sep = "\t", stringsAsFactors = F)
    length = read.table("Length.txt", sep = "\t", stringsAsFactors = F)
    site = read.table("Site.txt", sep = "\t", stringsAsFactors = F)
    muta = read.table("Mutagenesis.txt", sep = "\t", stringsAsFactors = F)
    option = read.table("Option.txt", sep = "\t", stringsAsFactors = F)
    zoomin = read.table("ZoomIn.txt", sep = "\t", stringsAsFactors = F)
    Mutagenesis = function(position, position2, color, height2, 
        height, up_down, start, end, pc, cex1) {
        h1 = -0.1
        h2 = -1.4
        h = -1.6
        hh1 = -2.8
        if (up_down == "up") {
            if (position == position2) {
                segments(position, hh1 + height, position, hh1 + 
                  height + h)
            }
            else {
                segments(position, hh1 + height, position, hh1 + 
                  height + h1)
                segments(position2, hh1 + height + h - h2, position2, 
                  hh1 + height + h)
                segments(position, hh1 + height + h1, position2, 
                  hh1 + height + h - h2)
            }
        }
        x = 0
        kong1 = (round(log(start, 10)) + 1) * start/50
        kong2 = (round(log(end, 10)) + 1) * end/50
        if (round(log(end, 10)) + 1 &lt;= 5) {
            kong2 = (round(log(end, 10)) + 1) * end/50
        }
        else {
            kong2 = 5 * end/50
        }
        boxplot(x, xlim = c(start - kong1, end + kong2), ylim = c(1, 
            -5.5), axes = FALSE, add = TRUE, border = FALSE)
        points(position2, height2, pch = pc, col = color, cex = cex1)
    }
    Change_h = function(muta_pos, muta_name, protein_h) {
        d = 0.1
        d1 = 0.26
        hh1 = -2.8
        height2 = 1:length(muta_pos)
        height2[1] = hh1 + protein_h - d1
        position_h = muta_pos
        position_h[1] = muta_pos[1]
        if (length(muta_pos) &gt; 1) {
            for (i in 2:length(muta_pos)) {
                if (muta_pos[i] == position_h[i - 1]) {
                  height2[i] = height2[i - 1] - d
                }
                else {
                  height2[i] = hh1 + protein_h - d1
                }
            }
        }
        height2
    }
    Change_m = function(muta, protein_width) {
        dec = 1.4 * protein_width/100
        position3 = 1:length(muta)
        position3[1] = muta[1]
        if (length(muta) &gt; 1) {
            for (i in 2:length(muta)) {
                if (muta[i] - muta[i - 1] &lt;= dec) {
                  if (muta[i] != muta[i - 1]) {
                    position3[i] = position3[i - 1] + dec
                  }
                  else {
                    position3[i] = position3[i - 1]
                  }
                }
                else {
                  position3[i] = muta[i]
                }
            }
        }
        position3
    }
    if (!is.na(muta[1, 1])) {
        position3 = Change_m(muta[, 1], as.numeric(length[2]))
        height2 = Change_h(muta[, 1], muta[, 2], as.numeric(protein[4]))
        for (i in 1:nrow(muta)) {
            Mutagenesis(position = as.numeric(muta[i, 1]), position2 = position3[i], 
                color = as.character(muta[i, 2]), height2 = height2[i], 
                height = as.numeric(protein[4]), up_down = "up", 
                start = as.numeric(length[1]), end = as.numeric(length[2]), 
                pc = as.numeric(protein[7]), cex1 = as.numeric(protein[8]))
        }
    }
  }
</code></pre>

<hr>
<h2 id='plotprotein'>
ploting protein
</h2><span id='topic+plotprotein'></span>

<h3>Description</h3>

<p>Draw a visualized structure of the protein
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotprotein()
</code></pre>


<h3>Details</h3>

<p>The tool ennable visualization of amino acid changes at the protein level,The scale of a protein domain and the position of a functional motif/site will be precisely defined
</p>


<h3>Value</h3>

<p>Visualization of protein structure
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    library("ade4")
    library("seqinr")
    library("plotrix")
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    domain = read.table("Domain.txt", sep = "\t", stringsAsFactors = F)
    length = read.table("Length.txt", sep = "\t", stringsAsFactors = F)
    site = read.table("Site.txt", sep = "\t", stringsAsFactors = F)
    muta = read.table("Mutagenesis.txt", sep = "\t", stringsAsFactors = F)
    option = read.table("Option.txt", sep = "\t", stringsAsFactors = F)
    zoomin = read.table("ZoomIn.txt", sep = "\t", stringsAsFactors = F)
    path = protein[1]
    pdf(as.character(path), height = 10.5, width = 11)
    layout(matrix(c(1, 2), nrow = 1), widths = c(1, 3))
    par(oma = c(3, 0, 2, 0), mar = c(4, 0, 2, 0) + 0.4)
    nameOfYourQuery = option[2, 1]
    additionalOptions = option[2, 2]
    showReferenceSequence = option[2, 3]
    showConservationScore = option[2, 4]
    showGridlinesAtTicks = option[2, 5]
    conservation = option[2, 6]
    zoomIn = zoomin[2, 1]
    zoomStart = zoomin[2, 2]
    zoomEnd = zoomin[2, 3]
    tickSize = as.numeric(zoomin[2, 4])
    plot((-30:-15), rep(-1, 16), col = "white", type = "l", ann = FALSE, 
        bty = "n", xaxt = "n", yaxt = "n", xlim = c(-160, -15), 
        ylim = c(1, -5.5))
    if (additionalOptions == "yes") {
        if (conservation == "yes") {
            lines((-30:-15), rep(0, 16), col = "purple3")
            lines((-30:-15), rep(-0.5, 16), col = "purple3")
            lines((-30:-15), rep(-1, 16), col = "purple3")
            text(-100, -0.5, "Conservation", col = "purple3", 
                cex = 0.9, font = 2)
            text(-45, -1, "1", col = "purple3", cex = 0.9)
            text(-45, -0.5, "0.5", col = "purple3", cex = 0.9)
            text(-45, 0, "0", col = "purple3", cex = 0.9)
        }
    }
    if (additionalOptions == "yes") {
        if (showReferenceSequence == "yes") {
            text(-100, -4.9, "Reference", col = "black", cex = 0.9, 
                font = 2)
        }
    }
    if (additionalOptions == "yes") {
        if (showConservationScore == "yes") {
            text(-100, 0.5, "Score", col = "purple3", cex = 0.9, 
                font = 2)
        }
    }
    text(-100, -2.95, nameOfYourQuery, col = "blue", cex = 0.9, 
        font = 2)
    Protein = function(start = 1, end, height = -0.3, color = "green", 
        face = "stereoscopic") {
        x = 0
        kong1 = (round(log(start, 10)) + 1) * start/50
        kong2 = (round(log(end, 10)) + 1) * end/50
        if (round(log(end, 10)) + 1 &lt;= 5) {
            kong2 = (round(log(end, 10)) + 1) * end/50
        }
        else {
            kong2 = 5 * end/50
        }
        h1 = -2.8
        h2 = -3.1
        boxplot((1:as.numeric(end)), rep(h1, as.numeric(end)), 
            xlab = "Amino Acid Position", ylab = "", xlim = c(0, 
                as.numeric(end)), ylim = c(1, -5.5), axes = FALSE)
        if (face == "stereoscopic") {
            cylindrect(start, h1, end, h2, col = color, gradient = "y")
        }
        else {
            rect(start, h1, end, h2, col = color)
        }
        text(0, h1 - height/2, start, adj = 1)
        text(end - 17, h1 - height/2, end, adj = 0)
    }
    ZoomIn = function(start = 1, end, height = -0.3, color = "green", 
        face = "stereoscopic", zoomstart, zoomend) {
        x = 0
        kong1 = (round(log(start, 10)) + 1) * start/50
        kong2 = (round(log(end, 10)) + 1) * end/50
        if (round(log(end, 10)) + 1 &lt;= 5) {
            kong2 = (round(log(end, 10)) + 1) * end/50
        }
        else {
            kong2 = 5 * end/50
        }
        h1 = -2.8
        h2 = -3.1
        boxplot((as.numeric(zoomstart):as.numeric(zoomend)), 
            rep(h1, as.numeric(zoomend)), xlab = "Amino Acid Position", 
            ylab = "", xlim = c(as.numeric(zoomstart), as.numeric(zoomend)), 
            ylim = c(1, -5.5), axes = FALSE)
        if (face == "stereoscopic") {
            cylindrect(start, h1, end, h2, col = color, gradient = "y")
        }
        else {
            rect(start, h1, end, h2, col = color)
        }
        text(start, h1 + height/2, start, adj = 1)
        text(end, h1 + height/2, end, adj = 0)
    }
    if (zoomIn == "yes") {
        ZoomIn(start = as.numeric(length[1]), end = as.numeric(length[2]), 
            height = as.numeric(protein[4]), color = as.character(protein[5]), 
            face = protein[6], zoomstart = zoomin[2, 2], zoomend = zoomin[2, 
                3])
    }
    else {
        Protein(start = as.numeric(length[1]), end = as.numeric(length[2]), 
            height = as.numeric(protein[4]), color = as.character(protein[5]), 
            face = protein[6])
    }
    legend("topleft", legend = c("mutation", "Protein Domain"), 
        pch = c(19, 15), col = c("lightseagreen", "deeppink"), 
        box.col = "white", bg = "white", pt.cex = 1.5, text.width = 1)
    ticks = seq(0, as.numeric(length[2]), by = tickSize)
    axis(side = 1, at = ticks, las = 3)
    if (additionalOptions == "yes") {
        if (showGridlinesAtTicks == "yes") {
            len = array(rep(1:as.numeric(length[2])))
            for (i in 1:length(len)) {
                abline(v = ticks[i], lty = 3, lwd = 0.5, col = "lightgray")
            }
        }
    }
  }
</code></pre>

<hr>
<h2 id='plotsite'>
ploting site
</h2><span id='topic+plotsite'></span>

<h3>Description</h3>

<p>Draw the protein site
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotsite()
</code></pre>


<h3>Details</h3>

<p>The tool ennable visualization of amino acid changes at the protein level,The scale of a protein domain and the position of a functional motif/site will be precisely defined. The features available include site
</p>


<h3>Value</h3>

<p>Location of the site in the protein
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    domain = read.table("Domain.txt", sep = "\t", stringsAsFactors = F)
    length = read.table("Length.txt", sep = "\t", stringsAsFactors = F)
    site = read.table("Site.txt", sep = "\t", stringsAsFactors = F)
    muta = read.table("Mutagenesis.txt", sep = "\t", stringsAsFactors = F)
    option = read.table("Option.txt", sep = "\t", stringsAsFactors = F)
    zoomin = read.table("ZoomIn.txt", sep = "\t", stringsAsFactors = F)
    Site = function(position, position2, name, height = -0.3, 
        x_y, up_down) {
        h1 = -0.1
        h2 = -0.2
        h = -0.4
        hh1 = -2.8
        if (up_down == "up") {
            if (position == position2) {
                segments(position, hh1 + height, position, hh1 + 
                  height + h)
            }
            else {
                segments(position, hh1 + height, position, hh1 + 
                  height + h1)
                segments(position2, hh1 + height + h - h2, position2, 
                  hh1 + height + h)
                segments(position, hh1 + height + h1, position2, 
                  hh1 + height + h - h2)
            }
            text(position2, hh1 + height + h - 0.02, name, srt = 90, 
                adj = c(0, 0.5), cex = 0.8)
        }
        else {
            if (position == position2) {
                segments(position, hh1, position, hh1 - h)
            }
            else {
                segments(position, hh1, position, hh1 - h1)
                segments(position2, hh1 - h + h2, position2, 
                  hh1 - h)
                segments(position, hh1 - h1, position2, hh1 - 
                  h + h2)
            }
            text(position2, hh1 - h + 0.02, name, srt = 270, 
                adj = c(0, 0.5), cex = 0.8)
        }
    }
    Change_x = function(site_pos, site_name, protein_width) {
        dec = 1.4 * protein_width/100
        position2 = 1:length(site_pos)
        position2[1] = site_pos[1]
        if (length(site_pos) &gt; 1) {
            for (i in 2:length(site_pos)) {
                if (site_pos[i] - site_pos[i - 1] &lt;= dec) {
                  if (site_pos[i] != site_pos[i - 1]) {
                    position2[i] = position2[i - 1] + dec
                  }
                  else {
                    position2[i] = position2[i - 1]
                  }
                }
                else {
                  position2[i] = site_pos[i]
                }
            }
        }
        return(position2)
    }
    if (!is.na(site[1, 1])) {
        position2 = Change_x(site[, 1], site[, 2], as.numeric(length[2]))
        for (i in 1:nrow(site)) {
            Site(position = as.numeric(site[i, 1]), position2 = position2[i], 
                name = as.character(site[i, 2]), height = as.numeric(protein[4]), 
                x_y = flag, up_down = "up")
        }
    }
  }
</code></pre>

<hr>
<h2 id='site_data'>
downloading protein site
</h2><span id='topic+site_data'></span>

<h3>Description</h3>

<p>Download the site of the protein, including the name 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_data()
</code></pre>


<h3>Details</h3>

<p>Download the site of the protein, including the distribution of the locus of the marker space
</p>


<h3>Value</h3>

<p>The location of the marker line
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Zhang
</p>


<h3>References</h3>

<p>https://cran.r-project.org/doc/manuals/R-exts.html
</p>


<h3>See Also</h3>

<p>code<a href="utils.html#topic+help">help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function () 
{
    library(XML)
    library(plyr)
    protein = read.table("Protein.txt", sep = "\t", stringsAsFactors = F)
    name = protein[2]
    url_p = "http://www.uniprot.org/uniprot/"
    url_s = "#showFeatures"
    url_w = paste(url_p, name, url_s, sep = "")
    url = url_w
    doc &lt;- htmlParse(url)
 position_s = xpathSApply (doc, "//table[@id= 'sitesAnno_section'] 
  /tr/td/  a[@class = 'position tooltipped']",  
        xmlValue)
    name_s = xpathSApply (doc, "//table[@id= 'sitesAnno_section']/tr/td/span[@property='text']", 
        xmlValue)
    s_s &lt;- c()
    for (i in 1:length(position_s)) {
        s_s[i] &lt;- gsub(pattern = "//D", replacement = "x", position_s[i])
    }
    s_s &lt;- strsplit(s_s, "xxx")
    d1_s &lt;- laply(s_s, function(x) x[1])
    d2_s &lt;- laply(s_s, function(x) x[2])
    r1_site = d1_s
    r2_site = name_s
    dfrm_site = data.frame(r1_site, r2_site)
    write.table(dfrm_site, file = "Site.txt", sep = "/t", quote = FALSE, 
        row.names = F, col.names = F)
  }
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
