<!DOCTYPE html><html><head><title>Help for package PMAPscore</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PMAPscore}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#final_signature'><p>final_signature, the final prognosis-related pathways</p></a></li>
<li><a href='#gene_symbol_Entrez'><p>gene_symbol_Entrez, the genes' symbol and ENTREZID</p></a></li>
<li><a href='#gene_Ucox'><p>gene_Ucox</p></a></li>
<li><a href='#gene_Ucox_res'><p>gene_Ucox_res, the univariate Cox regression result of candidate genes.</p></a></li>
<li><a href='#get_Entrez_ID'><p>Convert gene symbol to Entrez_Gene_ID</p></a></li>
<li><a href='#get_final_signature'><p>Identify the candidate prognosis-related pathways</p></a></li>
<li><a href='#get_km_survival_curve'><p>Plot Kaplan-Meier survival curve.</p></a></li>
<li><a href='#get_MultivariateCox_result'><p>Perform the multivariate Cox regression</p></a></li>
<li><a href='#get_mut_status'><p>Converts MAF file into mutation matrix</p></a></li>
<li><a href='#get_Oncoplots'><p>draw an GenePathwayOncoplots</p></a></li>
<li><a href='#get_pfs_score'><p>Calculates the pathway-based mutation accumulate perturbation score</p></a></li>
<li><a href='#get_response_plot'><p>Plot the response column diagram</p></a></li>
<li><a href='#get_risk_score'><p>Calculates the risk score for patients</p></a></li>
<li><a href='#get_roc_curve'><p>Plot the ROC curve</p></a></li>
<li><a href='#get_sam_cla'><p>get_sam_cla</p></a></li>
<li><a href='#get_univarCox_result'><p>Perform the univariate Cox regression analysis.</p></a></li>
<li><a href='#km_data'><p>km_data</p></a></li>
<li><a href='#maf_data'><p>maf_data</p></a></li>
<li><a href='#maffile'><p>maffile</p></a></li>
<li><a href='#mut_num'><p>mut_num</p></a></li>
<li><a href='#mut_sam'><p>mut_sam</p></a></li>
<li><a href='#mut_sample'><p>mut_sample</p></a></li>
<li><a href='#mut_status'><p>mut_status</p></a></li>
<li><a href='#newspia'><p>newspia</p></a></li>
<li><a href='#path_cox_data'><p>path_cox_data</p></a></li>
<li><a href='#path_gene'><p>path_gene</p></a></li>
<li><a href='#path_Ucox_mul'><p>path_Ucox_mul</p></a></li>
<li><a href='#path_Ucox_mul_res'><p>path_Ucox_mul_res</p></a></li>
<li><a href='#pfs_score'><p>pfs_score</p></a></li>
<li><a href='#response'><p>response</p></a></li>
<li><a href='#roc_data'><p>roc_data, the data frame use for ploting ROC curve</p></a></li>
<li><a href='#sig'><p>sig</p></a></li>
<li><a href='#sur'><p>sur</p></a></li>
<li><a href='#symbol_Entrez'><p>symbol_Entrez</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identify Prognosis-Related Pathways Altered by Somatic Mutation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We innovatively defined a pathway mutation accumulate perturbation score (PMAPscore) to reflect the position and the cumulative effect of the genetic mutations at the pathway level. Based on the PMAPscore of pathways, identified prognosis-related pathways altered by somatic mutation and predict immunotherapy efficacy by constructing a multiple-pathway-based risk model (Tarca, Adi Laurentiu et al (2008) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn577">doi:10.1093/bioinformatics/btn577</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>base, clusterProfiler, glmnet, graphics, grDevices, maftools,
org.Hs.eg.db, pROC, stats, survival, survminer, utils</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-12 07:34:51 UTC; DELL</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [aut, cre, cph],
  Yalan He [aut],
  Xiangmei Li [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-12 08:12:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='final_signature'>final_signature, the final prognosis-related pathways</h2><span id='topic+final_signature'></span>

<h3>Description</h3>

<p>The final prognosis-related pathways identified by our approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>final_signature
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 7.
</p>

<hr>
<h2 id='gene_symbol_Entrez'>gene_symbol_Entrez, the genes' symbol and ENTREZID</h2><span id='topic+gene_symbol_Entrez'></span>

<h3>Description</h3>

<p>The genes' symbol and ENTREZID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_symbol_Entrez
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 54245 rows and 2 columns.
</p>

<hr>
<h2 id='gene_Ucox'>gene_Ucox</h2><span id='topic+gene_Ucox'></span>

<h3>Description</h3>

<p>gene_Ucox
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_Ucox
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4287 rows and 5 columns.
</p>

<hr>
<h2 id='gene_Ucox_res'>gene_Ucox_res, the univariate Cox regression result of candidate genes.</h2><span id='topic+gene_Ucox_res'></span>

<h3>Description</h3>

<p>The univariate Cox regression result of candidate genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_Ucox_res
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4287 rows and 5 columns.
</p>

<hr>
<h2 id='get_Entrez_ID'>Convert gene symbol to Entrez_Gene_ID</h2><span id='topic+get_Entrez_ID'></span>

<h3>Description</h3>

<p>The function 'get_Entrez_ID' is used to convert gene symbol to Entrez_Gene_ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Entrez_ID(mut_status, gene_symbol_Entrez, Entrez_ID = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_Entrez_ID_+3A_mut_status">mut_status</code></td>
<td>
<p>A binary matrix that contains the mutation state of genes in each sample and its row name is the gene symbol. Noted the matrix can be generated by the function 'get_mut_status'.</p>
</td></tr>
<tr><td><code id="get_Entrez_ID_+3A_gene_symbol_entrez">gene_symbol_Entrez</code></td>
<td>
<p>A data table containing gene symbol and the corresponding gene Entrez ID.</p>
</td></tr>
<tr><td><code id="get_Entrez_ID_+3A_entrez_id">Entrez_ID</code></td>
<td>
<p>Logical,tell whether there are Entrez IDs corresponding to gene symbol in the gene_symbol_Entrez.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A binary matrix that contains the mutation state of genes in each sample and its row name is Entrez_Gene_ID.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data.
data(mut_status,gene_symbol_Entrez)
#perform function `get_Entrez_ID`.
mut_status&lt;-get_Entrez_ID(mut_status,gene_symbol_Entrez,Entrez_ID=TRUE)
</code></pre>

<hr>
<h2 id='get_final_signature'>Identify the candidate prognosis-related pathways</h2><span id='topic+get_final_signature'></span>

<h3>Description</h3>

<p>The function 'get_final_signature' uses to identify the candidate prognosis-related pathways based on the PMAPscore.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_final_signature(pfs_score, sur, wilcox_p = 0.05, uni_cox_p = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_final_signature_+3A_pfs_score">pfs_score</code></td>
<td>
<p>A 2 X n matrix that contains the pfs_score in each sample of the signal pathways. Noted the matrix can be generated by the function 'get_pfs_score'.</p>
</td></tr>
<tr><td><code id="get_final_signature_+3A_sur">sur</code></td>
<td>
<p>This data contains survival status and survival time of each sample.</p>
</td></tr>
<tr><td><code id="get_final_signature_+3A_wilcox_p">wilcox_p</code></td>
<td>
<p>The threshold of p value for Wilcoxon rank-sum test.</p>
</td></tr>
<tr><td><code id="get_final_signature_+3A_uni_cox_p">uni_cox_p</code></td>
<td>
<p>The threshold of p value for univariate Cox regression analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return the candidate prognosis-related pathways
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data.
data(pfs_score,sur)
#perform function `get_final_signature`.
final_signature&lt;-get_final_signature(pfs_score,sur)
</code></pre>

<hr>
<h2 id='get_km_survival_curve'>Plot Kaplan-Meier survival curve.</h2><span id='topic+get_km_survival_curve'></span>

<h3>Description</h3>

<p>The function 'get_km_survival_curve' uses to draw the Kaplan-Meier survival curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_km_survival_curve(km_data, cut_point, TRAIN = TRUE, risk.table = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_km_survival_curve_+3A_km_data">km_data</code></td>
<td>
<p>A data frame, including survival status, survival time, and risk score of each sample. The data frame can be generated by the function 'get_risk_score'.</p>
</td></tr>
<tr><td><code id="get_km_survival_curve_+3A_cut_point">cut_point</code></td>
<td>
<p>The threshold uses to classify patients into two subgroups with different OS.</p>
</td></tr>
<tr><td><code id="get_km_survival_curve_+3A_train">TRAIN</code></td>
<td>
<p>Logical,if set to TRUE,the 'cut_point' is generated by the median of the risk score; Otherwise,'cut_point' can be customized.</p>
</td></tr>
<tr><td><code id="get_km_survival_curve_+3A_risk.table">risk.table</code></td>
<td>
<p>Allowed values include:TRUE or FALSE specifying whether to show or not the risk table. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return, plot the Kaplan-Meier survival curve.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data.
data(km_data)
#perform the function `get_km_survival_curve`.
get_km_survival_curve(km_data,cut_point,TRAIN = TRUE,risk.table=TRUE)
</code></pre>

<hr>
<h2 id='get_MultivariateCox_result'>Perform the multivariate Cox regression</h2><span id='topic+get_MultivariateCox_result'></span>

<h3>Description</h3>

<p>The function 'get_MultivariateCox_result' uses to perform multivariate Cox regression analysis on the cancer-specific dysregulated signaling pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_MultivariateCox_result(DE_path_sur)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_MultivariateCox_result_+3A_de_path_sur">DE_path_sur</code></td>
<td>
<p>A binary metadata table containing sample survival status and survival time.Note that the column names of survival time and survival status must be &quot;survival&quot; and &quot;event&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return the multivariate Cox regression results of cancer-specific dysregulated signaling pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load the data.
data(path_cox_data)
#perform function `get_MultivariateCox_result`.
res&lt;-get_MultivariateCox_result(path_cox_data)
</code></pre>

<hr>
<h2 id='get_mut_status'>Converts MAF file into mutation matrix</h2><span id='topic+get_mut_status'></span>

<h3>Description</h3>

<p>The function 'get_mut_status' uses to convert MAF file into mutation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mut_status(maf_data, nonsynonymous = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mut_status_+3A_maf_data">maf_data</code></td>
<td>
<p>The patients' somatic mutation data, which in MAF format.</p>
</td></tr>
<tr><td><code id="get_mut_status_+3A_nonsynonymous">nonsynonymous</code></td>
<td>
<p>Logical, tell if extract the non-silent somatic mutations (nonsense mutation, missense mutation, frame-shif indels, splice site, nonstop mutation, translation start site, inframe indels).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A binary mutations matrix, in which 1 represents that a particular gene has mutated in a particular sample, and 0 represents that gene has no mutation in a particular sample .
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
data(maf_data)
#perform the function `get_mut_status`.
mutmatrix.example&lt;-get_mut_status(maf_data,nonsynonymous = TRUE)
</code></pre>

<hr>
<h2 id='get_Oncoplots'>draw an GenePathwayOncoplots</h2><span id='topic+get_Oncoplots'></span>

<h3>Description</h3>

<p>Load the data in MAF format and draws an GenePathwayOncoplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Oncoplots(
  maffile,
  path_gene,
  mut_status,
  risk_score,
  cut_off,
  final_signature,
  pathway_name,
  isTCGA = FALSE,
  top = 20,
  clinicalFeatures = "sample_group",
  annotationColor = c("red", "green"),
  sortByAnnotation = TRUE,
  removeNonMutated = FALSE,
  drawRowBar = TRUE,
  drawColBar = TRUE,
  leftBarData = NULL,
  leftBarLims = NULL,
  rightBarData = NULL,
  rightBarLims = NULL,
  topBarData = NULL,
  logColBar = FALSE,
  draw_titv = FALSE,
  showTumorSampleBarcodes = FALSE,
  fill = TRUE,
  showTitle = TRUE,
  titleText = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_Oncoplots_+3A_maffile">maffile</code></td>
<td>
<p>A data of MAF format.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_path_gene">path_gene</code></td>
<td>
<p>User input pathways geneset list.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_mut_status">mut_status</code></td>
<td>
<p>The mutations matrix,generated by 'get_mut_matrix'.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_risk_score">risk_score</code></td>
<td>
<p>Samples' PTMB-related risk score,which could be a biomarker for survival analysis and immunotherapy prediction.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_cut_off">cut_off</code></td>
<td>
<p>A threshold value(the median risk score as the default value).Using this value to divide the sample into high and low risk groups with different overall survival.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_final_signature">final_signature</code></td>
<td>
<p>The pathway signature,use to map gene in the GenePathwayOncoplots.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_pathway_name">pathway_name</code></td>
<td>
<p>The name of the pathway that you want to visualize.For example &quot;Gap junction&quot;</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_istcga">isTCGA</code></td>
<td>
<p>Is input MAF file from TCGA source. If TRUE uses only first 12 characters from Tumor_Sample_Barcode.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_top">top</code></td>
<td>
<p>How many top genes to be drawn,genes are arranged from high to low depending on the frequency of mutations. defaults to 20.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_clinicalfeatures">clinicalFeatures</code></td>
<td>
<p>Columns names from 'clinical.data' slot of MAF to be drawn in the plot. Dafault &quot;sample_group&quot;.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_annotationcolor">annotationColor</code></td>
<td>
<p>Custom colors to use for sample annotation-&quot;sample_group&quot;. Must be a named list containing a named vector of colors. Default &quot;red&quot; and &quot;green&quot;.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_sortbyannotation">sortByAnnotation</code></td>
<td>
<p>Logical sort oncomatrix (samples) by provided 'clinicalFeatures'. Sorts based on first 'clinicalFeatures'. Defaults to TRUE. column-sort.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_removenonmutated">removeNonMutated</code></td>
<td>
<p>Logical. If TRUE removes samples with no mutations in the GenePathwayOncoplots for better visualization. Default FALSE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_drawrowbar">drawRowBar</code></td>
<td>
<p>Logical. Plots righ barplot for each gene. Default TRUE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_drawcolbar">drawColBar</code></td>
<td>
<p>Logical plots top barplot for each sample. Default TRUE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_leftbardata">leftBarData</code></td>
<td>
<p>Data for leftside barplot. Must be a data.frame with two columns containing gene names and values. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_leftbarlims">leftBarLims</code></td>
<td>
<p>Limits for 'leftBarData'. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_rightbardata">rightBarData</code></td>
<td>
<p>Data for rightside barplot. Must be a data.frame with two columns containing to gene names and values. Default 'NULL' which draws distibution by variant classification. This option is applicable when only 'drawRowBar' is TRUE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_rightbarlims">rightBarLims</code></td>
<td>
<p>Limits for 'rightBarData'. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_topbardata">topBarData</code></td>
<td>
<p>Default 'NULL' which draws absolute number of mutation load for each sample. Can be overridden by choosing one clinical indicator(Numeric) or by providing a two column data.frame contaning sample names and values for each sample. This option is applicable when only 'drawColBar' is TRUE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_logcolbar">logColBar</code></td>
<td>
<p>Plot top bar plot on log10 scale. Default FALSE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_draw_titv">draw_titv</code></td>
<td>
<p>Logical Includes TiTv plot. Default FALSE</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_showtumorsamplebarcodes">showTumorSampleBarcodes</code></td>
<td>
<p>Logical to include sample names.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_fill">fill</code></td>
<td>
<p>Logical. If TRUE draws genes and samples as blank grids even when they are not altered.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_showtitle">showTitle</code></td>
<td>
<p>Default TRUE.</p>
</td></tr>
<tr><td><code id="get_Oncoplots_+3A_titletext">titleText</code></td>
<td>
<p>Custom title. Default 'NULL'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#obtain the risksciore
data(km_data)
risk_score&lt;-km_data$multiple_score
names(risk_score)&lt;-rownames(km_data)
cut_off&lt;-median(risk_score)
#load the dtata
data(final_signature,path_gene,mut_status,maffile)
##draw an GenePathwayOncoplots
get_Oncoplots(maffile,path_gene,mut_status,risk_score,cut_off,final_signature,"Gap junction")

</code></pre>

<hr>
<h2 id='get_pfs_score'>Calculates the pathway-based mutation accumulate perturbation score</h2><span id='topic+get_pfs_score'></span>

<h3>Description</h3>

<p>The function 'get_pfs_score' uses to calculate the pathway-based mutation accumulate perturbation score using the matrix of gene mutation state and pathway information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pfs_score(
  mut_status,
  percent,
  gene_Ucox_res,
  gene_symbol_Entrez,
  data.dir = NULL,
  organism = "hsa",
  verbose = TRUE,
  Entrez_ID = TRUE,
  gene_set = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pfs_score_+3A_mut_status">mut_status</code></td>
<td>
<p>Mutation status of a particular gene in a particular sample. The file can be generated by the function 'get_mut_status'.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_percent">percent</code></td>
<td>
<p>This parameter is used to control the mutation rate of gene. Genes less than this value will be deleted</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_gene_ucox_res">gene_Ucox_res</code></td>
<td>
<p>Results of gene univariate Cox regression.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_gene_symbol_entrez">gene_symbol_Entrez</code></td>
<td>
<p>A data table containing gene symbol and gene Entrez ID.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_data.dir">data.dir</code></td>
<td>
<p>Location of the &quot;organism&quot;SPIA.RData file containing the pathways data.If set to NULL will look for this file in the extdata folder of the PFS library.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_organism">organism</code></td>
<td>
<p>A three letter character designating the organism. See a full list at ftp://ftp.genome.jp/pub/kegg/xml/organisms.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_verbose">verbose</code></td>
<td>
<p>If set to TRUE, displays the number of pathways already analyzed.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_entrez_id">Entrez_ID</code></td>
<td>
<p>Logical,tell whether there are Entrez IDs corresponding to gene symbol in the gene_symbol_Entrez.</p>
</td></tr>
<tr><td><code id="get_pfs_score_+3A_gene_set">gene_set</code></td>
<td>
<p>A group of cancer specific gene symbols obtained from the training set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A binary mutations matrix, which column names is sample and the row name is the pathway.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file.
data(mut_status,gene_Ucox_res,gene_symbol_Entrez)
#perform the function `get_pfs_score`.
pfs_score&lt;-get_pfs_score(mut_status[,1:2],percent=0.03,gene_Ucox_res,gene_symbol_Entrez)
</code></pre>

<hr>
<h2 id='get_response_plot'>Plot the response column diagram</h2><span id='topic+get_response_plot'></span>

<h3>Description</h3>

<p>The function 'get_response_plot' uses to plot the column diagram of drug response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_response_plot(km_data, response, cut_point, TRAIN = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_response_plot_+3A_km_data">km_data</code></td>
<td>
<p>A data frame, including survival status, survival time, and risk score of each sample. The data frame can be generated by the function 'get_risk_score'.</p>
</td></tr>
<tr><td><code id="get_response_plot_+3A_response">response</code></td>
<td>
<p>Response status of the sample to the drug.</p>
</td></tr>
<tr><td><code id="get_response_plot_+3A_cut_point">cut_point</code></td>
<td>
<p>The threshold uses to classify patients into two subgroups with different OS.</p>
</td></tr>
<tr><td><code id="get_response_plot_+3A_train">TRAIN</code></td>
<td>
<p>Logical,if set to TRUE,the 'cut_point' is generated by the median of the risk score; Otherwise,'cut_point' can be customized.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Comparison of the objective response rate between the high-risk and low-risk groups, plot the bar graph and return the p value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load the data.
data(km_data,response)
#perform the function `get_response_plot`.
get_response_plot(km_data,response,cut_point,TRAIN=TRUE)
</code></pre>

<hr>
<h2 id='get_risk_score'>Calculates the risk score for patients</h2><span id='topic+get_risk_score'></span>

<h3>Description</h3>

<p>The function 'get_risk_score' uses to calculate the risk score for patients based on cancer-specific dysregulated signaling pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_risk_score(
  final_signature,
  pfs_score,
  path_Ucox_mul_res,
  sur,
  TRAIN = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_risk_score_+3A_final_signature">final_signature</code></td>
<td>
<p>Cancer-specific dysregulated signal pathways. It can be generated by the function 'get_final_signature'.</p>
</td></tr>
<tr><td><code id="get_risk_score_+3A_pfs_score">pfs_score</code></td>
<td>
<p>A matrix that contains the pfs_score in each sample of the signal pathways. Noted the matrix can be generated by the function 'get_pfs_score'.</p>
</td></tr>
<tr><td><code id="get_risk_score_+3A_path_ucox_mul_res">path_Ucox_mul_res</code></td>
<td>
<p>Results of multivariate Cox regression of cancer specific pathway in training set.</p>
</td></tr>
<tr><td><code id="get_risk_score_+3A_sur">sur</code></td>
<td>
<p>This data contains survival status and survival time of each sample.</p>
</td></tr>
<tr><td><code id="get_risk_score_+3A_train">TRAIN</code></td>
<td>
<p>Logical,if set FLASE,we need to load the result of multivariate Cox regression of cancer specific pathways into the training set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data set with the risk score for each sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load the data.
data(final_signature,pfs_score,sur,path_Ucox_mul_res)
#perform the function `get_risk_score`.
km_data&lt;-get_risk_score(final_signature,pfs_score,path_Ucox_mul_res,sur,TRAIN=TRUE)
</code></pre>

<hr>
<h2 id='get_roc_curve'>Plot the ROC curve</h2><span id='topic+get_roc_curve'></span>

<h3>Description</h3>

<p>The function 'get_roc_curve' uses to plot the ROC curve for predicting immunotherapy response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_roc_curve(roc_data, print.auc = TRUE, main = "Objective Response")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_roc_curve_+3A_roc_data">roc_data</code></td>
<td>
<p>A 2 X n data fram, which contain the immunotherapy response and risk score (generated by the function 'get_risk_score') for patients.</p>
</td></tr>
<tr><td><code id="get_roc_curve_+3A_print.auc">print.auc</code></td>
<td>
<p>Boolean. Should the numeric value of AUC be printed on the plot?</p>
</td></tr>
<tr><td><code id="get_roc_curve_+3A_main">main</code></td>
<td>
<p>A main title for the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return, plot the ROC curve for immunotherapy response prediction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load the data.
data(roc_data)
#perform the function `get_roc_curve`.
get_roc_curve(roc_data,print.auc=TRUE,main="Objective Response")
</code></pre>

<hr>
<h2 id='get_sam_cla'>get_sam_cla</h2><span id='topic+get_sam_cla'></span>

<h3>Description</h3>

<p>Function 'get_sample_classification' This function is used to judge the classification of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sam_cla(
  mut_sam,
  gene_Ucox,
  symbol_Entrez,
  path_cox_data,
  sur,
  path_Ucox_mul,
  sig,
  cut_off = -0.986,
  data.dir = NULL,
  organism = "hsa",
  TRAIN = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_sam_cla_+3A_mut_sam">mut_sam</code></td>
<td>
<p>The sample somatic mutation data.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_gene_ucox">gene_Ucox</code></td>
<td>
<p>Results of gene univariate Cox regression.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_symbol_entrez">symbol_Entrez</code></td>
<td>
<p>A data table containing gene symbol and gene Entrez ID.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_path_cox_data">path_cox_data</code></td>
<td>
<p>Pathways of Cancer-specifical obtained from the training set.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_sur">sur</code></td>
<td>
<p>This data contains survival status and survival time of each sample.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_path_ucox_mul">path_Ucox_mul</code></td>
<td>
<p>Multivariate Cox regression results of  Cancer-specifical pathways.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_sig">sig</code></td>
<td>
<p>Cancer-specific dysregulated signal pathways. It can be generated by the function 'get_final_signature'.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_cut_off">cut_off</code></td>
<td>
<p>Threshold of classification.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_data.dir">data.dir</code></td>
<td>
<p>Location of the &quot;organism&quot;SPIA.RData file containing the pathways data. If set to NULL will look for this file in the extdata folder of the PMAPscore library.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_organism">organism</code></td>
<td>
<p>A three letter character designating the organism. See a full list at ftp://ftp.genome.jp/pub/kegg/xml/organisms.</p>
</td></tr>
<tr><td><code id="get_sam_cla_+3A_train">TRAIN</code></td>
<td>
<p>Logical,if set FLASE,we need to load the result of multivariate Cox regression of cancer specific pathways into the training set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a data frame, the sample's risk score and the sample's risk group.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load the data.
data(mut_sam,gene_Ucox,symbol_Entrez,path_cox_data,sur,path_Ucox_mul)
#perform function `get_sample_cla`.
get_sam_cla(mut_sam,gene_Ucox,symbol_Entrez,path_cox_data,sur,path_Ucox_mul,sig,cut_off=-0.986)
</code></pre>

<hr>
<h2 id='get_univarCox_result'>Perform the univariate Cox regression analysis.</h2><span id='topic+get_univarCox_result'></span>

<h3>Description</h3>

<p>The function 'get_univarCox_result' uses to perform the univariate Cox regression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_univarCox_result(DE_path_sur)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_univarCox_result_+3A_de_path_sur">DE_path_sur</code></td>
<td>
<p>A binary metadata table containing  survival status and survival time of each sample.Note that the column names of survival time and survival status must be &quot;survival&quot; and &quot;event&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a data frame, the univariate Cox regression analysis results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get path of the mutation annotation file.
data(path_cox_data)
#perform function `get_univarCox_result`.
res&lt;-get_univarCox_result(path_cox_data)
</code></pre>

<hr>
<h2 id='km_data'>km_data</h2><span id='topic+km_data'></span>

<h3>Description</h3>

<p>The data use for drawing K-M survival curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>km_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 105 rows and 10 columns.
</p>

<hr>
<h2 id='maf_data'>maf_data</h2><span id='topic+maf_data'></span>

<h3>Description</h3>

<p>The mutation data of patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maf_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 24461 rows and 4 columns.
</p>

<hr>
<h2 id='maffile'>maffile</h2><span id='topic+maffile'></span>

<h3>Description</h3>

<p>The mutation data of patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maffile
</code></pre>


<h3>Format</h3>

<p>An object of class <code>MAF</code> of length 1.
</p>

<hr>
<h2 id='mut_num'>mut_num</h2><span id='topic+mut_num'></span>

<h3>Description</h3>

<p>mut_num
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_num
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 13858 rows and 105 columns.
</p>

<hr>
<h2 id='mut_sam'>mut_sam</h2><span id='topic+mut_sam'></span>

<h3>Description</h3>

<p>mut_sam.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_sam
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 13858 rows and 2 columns.
</p>

<hr>
<h2 id='mut_sample'>mut_sample</h2><span id='topic+mut_sample'></span>

<h3>Description</h3>

<p>mut_sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_sample
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 13858 rows and 2 columns.
</p>

<hr>
<h2 id='mut_status'>mut_status</h2><span id='topic+mut_status'></span>

<h3>Description</h3>

<p>mut_status.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_status
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 13858 rows and 105 columns.
</p>

<hr>
<h2 id='newspia'>newspia</h2><span id='topic+newspia'></span>

<h3>Description</h3>

<p>Function 'newspia' This function is based on SPIA algorithm to analyse KEGG signal pathway for single sample..
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newspia(
  de = NULL,
  all = NULL,
  organism = "hsa",
  data.dir = NULL,
  pathids = NULL,
  verbose = TRUE,
  beta = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newspia_+3A_de">de</code></td>
<td>
<p>A named vector containing the statue of particular genes in a particular sample.The names of this numeric vector are Entrez gene IDs.</p>
</td></tr>
<tr><td><code id="newspia_+3A_all">all</code></td>
<td>
<p>A vector with the Entrez IDs in the reference set. If the data was obtained from a microarray experiment,this set will contain all genes present on the specific array used for the experiment.This vector should contain all names of the de argument.</p>
</td></tr>
<tr><td><code id="newspia_+3A_organism">organism</code></td>
<td>
<p>A three letter character designating the organism. See a full list at ftp://ftp.genome.jp/pub/kegg/xml/organisms.</p>
</td></tr>
<tr><td><code id="newspia_+3A_data.dir">data.dir</code></td>
<td>
<p>Location of the &quot;organism&quot;SPIA.RData file containing the pathways data .If set to NULL will look for this file in the extdata folder of the PMAPscore library.</p>
</td></tr>
<tr><td><code id="newspia_+3A_pathids">pathids</code></td>
<td>
<p>A character vector with the names of the pathways to be analyzed.If left NULL all pathways available will be tested.</p>
</td></tr>
<tr><td><code id="newspia_+3A_verbose">verbose</code></td>
<td>
<p>If set to TRUE, displays the number of pathways already analyzed.</p>
</td></tr>
<tr><td><code id="newspia_+3A_beta">beta</code></td>
<td>
<p>Weights to be assigned to each type of gene/protein relation type.
It should be a named numeric vector of length 23, whose names must be: c(&quot;activation&quot;,&quot;compound&quot;,&quot;binding/association&quot;,&quot;expression&quot;,
&quot;inhibition&quot;,&quot;activation_phosphorylation&quot;,&quot;phosphorylation&quot;, &quot;indirect&quot;,&quot;inhibition_phosphorylation&quot;,&quot;dephosphorylation_inhibition&quot;,
&quot;dissociation&quot;,&quot;dephosphorylation&quot;,&quot;activation_dephosphorylation&quot;, &quot;state&quot;,&quot;activation_indirect&quot;,&quot;inhibition_ubiquination&quot;,&quot;ubiquination&quot;,
&quot;expression_indirect&quot;,&quot;indirect_inhibition&quot;,&quot;repression&quot;, &quot;binding/association_phosphorylation&quot;,&quot;dissociation_phosphorylation&quot;,&quot;indirect_phosphorylation&quot;)
If set to null, beta will be by default chosen as: c(1,0,0,1,1,1,0,0,1,1,0,0,1,0,1,1,0,1,1,1,0,0,0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Get one Data in data frame format,which cotains pathway's id,pathway's name and PFS_score.
</p>

<hr>
<h2 id='path_cox_data'>path_cox_data</h2><span id='topic+path_cox_data'></span>

<h3>Description</h3>

<p>path_cox_data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>path_cox_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 105 rows and 9 columns.
</p>

<hr>
<h2 id='path_gene'>path_gene</h2><span id='topic+path_gene'></span>

<h3>Description</h3>

<p>path_gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>path_gene
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 7.
</p>

<hr>
<h2 id='path_Ucox_mul'>path_Ucox_mul</h2><span id='topic+path_Ucox_mul'></span>

<h3>Description</h3>

<p>path_Ucox_mul
</p>


<h3>Usage</h3>

<pre><code class='language-R'>path_Ucox_mul
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 7 rows and 5 columns.
</p>

<hr>
<h2 id='path_Ucox_mul_res'>path_Ucox_mul_res</h2><span id='topic+path_Ucox_mul_res'></span>

<h3>Description</h3>

<p>path_Ucox_mul_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>path_Ucox_mul_res
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 7 rows and 5 columns.
</p>

<hr>
<h2 id='pfs_score'>pfs_score</h2><span id='topic+pfs_score'></span>

<h3>Description</h3>

<p>pfs_score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pfs_score
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 123 rows and 105 columns.
</p>

<hr>
<h2 id='response'>response</h2><span id='topic+response'></span>

<h3>Description</h3>

<p>response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>response
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 110 rows and 2 columns.
</p>

<hr>
<h2 id='roc_data'>roc_data, the data frame use for ploting ROC curve</h2><span id='topic+roc_data'></span>

<h3>Description</h3>

<p>The roc_data is used to generate ROC curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roc_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 105 rows and 4 columns.
</p>

<hr>
<h2 id='sig'>sig</h2><span id='topic+sig'></span>

<h3>Description</h3>

<p>sig
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sig
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 7.
</p>

<hr>
<h2 id='sur'>sur</h2><span id='topic+sur'></span>

<h3>Description</h3>

<p>sur
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sur
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 110 rows and 2 columns.
</p>

<hr>
<h2 id='symbol_Entrez'>symbol_Entrez</h2><span id='topic+symbol_Entrez'></span>

<h3>Description</h3>

<p>symbol_Entrez
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symbol_Entrez
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 54245 rows and 2 columns.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
