<!DOCTYPE html><html><head><title>Help for package pmartR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pmartR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.is_edata'><p>Test if a file is an edata file</p></a></li>
<li><a href='#all_subset'><p>Identify All Biomolecules for Use in Normalization</p></a></li>
<li><a href='#anova_filter'><p>Identifies biomolecules to be filtered in preparation for IMD-ANOVA.</p></a></li>
<li><a href='#anova_test'><p>Tests for a quantiative difference between groups (aka factors, aka main</p>
effects)</a></li>
<li><a href='#applyFilt'><p>Apply a S3 filter object to a pmartR S3 object</p></a></li>
<li><a href='#as.isobaricpepData'><p>Create pmartR Object of Class isobaricpepData</p></a></li>
<li><a href='#as.lipidData'><p>Create pmartR Object of Class lipidData</p></a></li>
<li><a href='#as.metabData'><p>Create pmartR Object of Class metabData</p></a></li>
<li><a href='#as.multiData'><p>Create a 'multiData' object from multiple omicsData objects</p></a></li>
<li><a href='#as.nmrData'><p>Create pmartR Object of Class nmrData</p></a></li>
<li><a href='#as.pepData'><p>Create pmartR Object of Class pepData</p></a></li>
<li><a href='#as.proData'><p>Create pmartR Object of Class proData</p></a></li>
<li><a href='#as.seqData'><p>Create pmartR Object of Class seqData</p></a></li>
<li><a href='#as.trelliData'><p>Generate an object from omicsData and/or statRes objects to pass to</p>
trelliscope building functions</a></li>
<li><a href='#as.trelliData.edata'><p>Generate an object from edata to pass to trelliscope building</p>
functions</a></li>
<li><a href='#bpquant'><p>Runs BP-Quant</p></a></li>
<li><a href='#bpquant_mod'><p>bpquant_mod function</p></a></li>
<li><a href='#column_matches_exact'><p>Helper to find the names of columns of a data.frame that contain exactly all</p>
the elements of an input column</a></li>
<li><a href='#combine_lipidData'><p>Combines two omicsData objects with identical sample information.</p></a></li>
<li><a href='#combine_techreps'><p>Combine technical replicates of an omicsData object</p></a></li>
<li><a href='#complete_mols'><p>Identify biomolecules with no missing values across samples</p></a></li>
<li><a href='#cor_result'><p>Compute Correlation matrix of biomolecule data</p></a></li>
<li><a href='#create_comparisonDF'><p>Returns data frame with comparisons to be made</p></a></li>
<li><a href='#custom_filter'><p>Custom Filter Object</p></a></li>
<li><a href='#custom_sampnames'><p>Creates custom sample names to be used in plots</p></a></li>
<li><a href='#cv_filter'><p>Pooled Coefficient of Variation (CV) Filter Object</p></a></li>
<li><a href='#DESeq2_wrapper'><p>Wrapper for DESeq2 workflow</p></a></li>
<li><a href='#diffexp_seq'><p>Differential Expression for seqData</p></a></li>
<li><a href='#dim_reduction'><p>Reduce Dimension of Data for Exploratory Data Analysis</p></a></li>
<li><a href='#dispersion_est'><p>Diagnostic plot for seqData</p></a></li>
<li><a href='#edata_replace'><p>Replace Values Equal to x with y</p></a></li>
<li><a href='#edata_summary'><p>Creates a list of six Data Frames, one for each summarizing metric</p></a></li>
<li><a href='#edata_transform'><p>Apply a Transformation to the Data</p></a></li>
<li><a href='#edgeR_wrapper'><p>Wrapper for edgeR workflow</p></a></li>
<li><a href='#find_fmeta_cnames'><p>Find column names in f_meta for each object.  May return in two objects</p>
sharing the same column name.</a></li>
<li><a href='#fit_surv'><p>Basic survival analysis function</p></a></li>
<li><a href='#fmeta_matches'><p>Check that the f_meta file contains a column aligned to each omicsData objects</p></a></li>
<li><a href='#get_check_names'><p><b>DEPRECATED</b>: Fetch the check.names attribute</p></a></li>
<li><a href='#get_comparisons'><p>Return comparisons of statRes object</p></a></li>
<li><a href='#get_data_class'><p>Return data_class of statRes or trelliData object</p></a></li>
<li><a href='#get_data_info'><p>Fetch the data_info attribute</p></a></li>
<li><a href='#get_data_norm'><p>Fetch the normalization status of the data</p></a></li>
<li><a href='#get_data_scale'><p>Fetch the current data scale</p></a></li>
<li><a href='#get_data_scale_orig'><p>Fetch the original data scale</p></a></li>
<li><a href='#get_edata_cname'><p>Fetch the e_data column name</p></a></li>
<li><a href='#get_emeta_cname'><p>Fetch the e_meta column name</p></a></li>
<li><a href='#get_fdata_cname'><p>Fetch the f_data column name</p></a></li>
<li><a href='#get_filter_type'><p>Extracts the types of filters that have been applied. This function will be</p>
used at the beginning of the applyFilt function to give a warning if the
same type of filter has already been applied.</a></li>
<li><a href='#get_filters'><p>Fetch the filters attribute</p></a></li>
<li><a href='#get_group_DF'><p>Fetch the group_DF attribute</p></a></li>
<li><a href='#get_group_formula'><p>Get formula for group design</p></a></li>
<li><a href='#get_group_table'><p>Get group table</p></a></li>
<li><a href='#get_isobaric_info'><p>Fetch the isobaric_info attribute</p></a></li>
<li><a href='#get_isobaric_norm'><p>Fetch the isobaric normalization info</p></a></li>
<li><a href='#get_lsmeans'><p>Compute the least squares means from a prediction grid and estimated coefficients</p></a></li>
<li><a href='#get_meta_info'><p>Fetch the meta_info attribute</p></a></li>
<li><a href='#get_nmr_info'><p>Fetch the nmr_info attribute</p></a></li>
<li><a href='#get_nmr_norm'><p>Fetch the NMR normalization info</p></a></li>
<li><a href='#get_pred_grid'><p>Build the prediction grid to compute least squares means.</p></a></li>
<li><a href='#get_spans_params'><p>Gets the parameters for the highest ranked methods from spans.</p></a></li>
<li><a href='#group_comparison_anova'><p>Group comparisons for the anova test</p></a></li>
<li><a href='#group_comparison_imd'><p>Group comparisons for the g-test</p></a></li>
<li><a href='#group_designation'><p>Creates Attribute of omicsData Object for Group Membership</p></a></li>
<li><a href='#gtest_filter'><p>Identifies peptides to be filtered out in preparation for IMD-ANOVA.</p></a></li>
<li><a href='#gtest_heatmap'><p>Plot a heatmap for the g-test results of imd-anova</p></a></li>
<li><a href='#imd_anova'><p>Test for a qualitative and quantitative difference between groups using IMD</p>
and ANOVA, respectively</a></li>
<li><a href='#imd_test'><p>Tests for the independence of missing data across groups (aka factors, aka</p>
main effects)</a></li>
<li><a href='#imdanova_filter'><p>IMD-ANOVA Filter Object</p></a></li>
<li><a href='#los'><p>Identify Biomolecules from the Top L Order Statistics for Use in</p>
Normalization</a></li>
<li><a href='#mad_transform'><p>Median Absolute Deviation Transformation</p></a></li>
<li><a href='#make_volcano_plot_df'><p>Create a plotting dataframe for volcano plots and heatmaps</p></a></li>
<li><a href='#mean_center'><p>Mean Center Transformation</p></a></li>
<li><a href='#median_center'><p>Median Center Transformation</p></a></li>
<li><a href='#missingval_result'><p>Creates an object of class naRes (NA Result)</p></a></li>
<li><a href='#molecule_filter'><p>Molecule Filter Object</p></a></li>
<li><a href='#nonmissing_per_group'><p>Computes the Number of Non-Missing Data Points by Group</p></a></li>
<li><a href='#normalize_global'><p>Calculate Normalization Parameters and Apply Global Normalization</p></a></li>
<li><a href='#normalize_global_basic'><p>Normalize e_data within SPANS</p></a></li>
<li><a href='#normalize_isobaric'><p>Examine and Apply Isobaric Normalization</p></a></li>
<li><a href='#normalize_loess'><p>Loess Normalization</p></a></li>
<li><a href='#normalize_nmr'><p>Normalize an Object of Class nmrData</p></a></li>
<li><a href='#normalize_quantile'><p>Quantile Normalization</p></a></li>
<li><a href='#normRes_tests'><p>Test the location and scale parameters from a normalization procedure</p></a></li>
<li><a href='#p_adjustment_anova'><p>Adjust p-values for multiple comparisons</p></a></li>
<li><a href='#plot_km'><p>Basic survival analysis plot</p></a></li>
<li><a href='#plot.corRes'><p>Plot corRes Object</p></a></li>
<li><a href='#plot.customFilt'><p>Plot customFilt Object</p></a></li>
<li><a href='#plot.cvFilt'><p>Plot cvFilt Object</p></a></li>
<li><a href='#plot.dataRes'><p>Plot dataRes object</p></a></li>
<li><a href='#plot.dimRes'><p>Plot dimRes Object</p></a></li>
<li><a href='#plot.imdanovaFilt'><p>Plot imdanovaFilt Object</p></a></li>
<li><a href='#plot.isobaricnormRes'><p>Plot isobaricnormRes object</p></a></li>
<li><a href='#plot.isobaricpepData'><p>Plot isobaricpepData Object</p></a></li>
<li><a href='#plot.lipidData'><p>Plot lipidData Object</p></a></li>
<li><a href='#plot.metabData'><p>Plot metabData Object</p></a></li>
<li><a href='#plot.moleculeFilt'><p>Plot moleculeFilt Object</p></a></li>
<li><a href='#plot.naRes'><p>Plot naRes Object</p></a></li>
<li><a href='#plot.nmrData'><p>Plot nmrData Object</p></a></li>
<li><a href='#plot.nmrnormRes'><p>Plot nmrnormRes Object</p></a></li>
<li><a href='#plot.normRes'><p>Plot normRes Object</p></a></li>
<li><a href='#plot.pepData'><p>Plot pepData Object</p></a></li>
<li><a href='#plot.proData'><p>Plot proData Object</p></a></li>
<li><a href='#plot.proteomicsFilt'><p>Plot proteomicsFilt Object</p></a></li>
<li><a href='#plot.rmdFilt'><p>Plot rmdFilt Object</p></a></li>
<li><a href='#plot.RNAFilt'><p>Plot RNAFilt Object</p></a></li>
<li><a href='#plot.seqData'><p>Plot seqData Object</p></a></li>
<li><a href='#plot.SPANSRes'><p>Plot SPANSRes Object</p></a></li>
<li><a href='#plot.statRes'><p>Plot statRes Object</p></a></li>
<li><a href='#plot.totalCountFilt'><p>Plot totalCountFilt Object</p></a></li>
<li><a href='#pmartR'><p>Panomics Marketplace - Quality Control and Statistical Analysis for Panomics Data</p></a></li>
<li><a href='#pmartR_filter_worker'><p>Remove items that need to be filtered out</p></a></li>
<li><a href='#ppp'><p>Identify Biomolecules from the Proportion Present (PPP) for Use in</p>
Normalization</a></li>
<li><a href='#ppp_rip'><p>Identify Proportion of Peptides Present (PPP) and Rank Invariant Peptides</p>
(RIP) for Use in Normalization</a></li>
<li><a href='#pquant'><p>Protein Quantitation using Mean or Median Peptide Abundances</p></a></li>
<li><a href='#pre_imdanova_melt'><p>Create a Melted and Grouped Version of e_data for IMD_ANOVA filter</p></a></li>
<li><a href='#prep_flags'><p>Extract flag columns from a statRes object</p></a></li>
<li><a href='#print.customFilt'><p>print.customFilt</p></a></li>
<li><a href='#print.customFilterSummary'><p>Custom Filter Print Method</p></a></li>
<li><a href='#print.cvFilt'><p>print.cvFilt</p></a></li>
<li><a href='#print.cvFilterSummary'><p>CV Filter Print Method</p></a></li>
<li><a href='#print.dataRes'><p>print.dataRes</p></a></li>
<li><a href='#print.imdanovaFilt'><p>print.imdanovaFilt</p></a></li>
<li><a href='#print.imdanovaFilterSummary'><p>IMD-ANOVA Filter Print Method</p></a></li>
<li><a href='#print.lipidData'><p>print.lipidData</p></a></li>
<li><a href='#print.metabData'><p>print.metabData</p></a></li>
<li><a href='#print.moleculeFilt'><p>print.moleculeFilt</p></a></li>
<li><a href='#print.moleculeFilterSummary'><p>Molecule Filter Print Method</p></a></li>
<li><a href='#print.normRes'><p>print.normRes</p></a></li>
<li><a href='#print.pepData'><p>print.pepData</p></a></li>
<li><a href='#print.proData'><p>print.proData</p></a></li>
<li><a href='#print.proteomicsFilt'><p>print.proteomicsFilt</p></a></li>
<li><a href='#print.proteomicsFilterSummary'><p>Proteomics Filter Print Method</p></a></li>
<li><a href='#print.rmdFilt'><p>print.rmdFilt</p></a></li>
<li><a href='#print.rmdFilterSummary'><p>RMD Filter Print Method</p></a></li>
<li><a href='#print.RNAFilt'><p>print.RNAFilt</p></a></li>
<li><a href='#print.RNAFiltSummary'><p>RNA Filter Print Method</p></a></li>
<li><a href='#print.seqData'><p>print.seqData</p></a></li>
<li><a href='#print.totalCountFilt'><p>print.totalCountFilt</p></a></li>
<li><a href='#print.totalCountFiltSummary'><p>Total Count Filter Print Method</p></a></li>
<li><a href='#protein_quant'><p>Protein Quantification</p></a></li>
<li><a href='#proteomics_filter'><p>Proteomics Filter Object</p></a></li>
<li><a href='#qrollup'><p>Applies qrollup function</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#replace_nas'><p>Replace NA with 0</p></a></li>
<li><a href='#replace_zeros'><p>Replace 0 with NA</p></a></li>
<li><a href='#report_dataRes'><p>Creates a data frame displaying multiple metrics</p></a></li>
<li><a href='#rip'><p>Identify Rank-Invariant Biomolcules for Use in Normalization</p></a></li>
<li><a href='#rmd_conversion'><p>Conversion between log2(RMD) and p-value</p></a></li>
<li><a href='#rmd_filter'><p>Robust Mahalanobis Distance (RMD) Filter Object</p></a></li>
<li><a href='#RNA_filter'><p>RNA Filter Object</p></a></li>
<li><a href='#rrollup'><p>Applies rrollup function</p></a></li>
<li><a href='#run_group_meancor'><p>Calculate the Mean Correlation of a Sample with Respect to Group</p></a></li>
<li><a href='#run_kurtosis'><p>Calculate the Kurtosis of Sample Runs</p></a></li>
<li><a href='#run_mad'><p>Calculate the Median Absolute Deviance (MAD) of Sample Runs</p></a></li>
<li><a href='#run_prop_missing'><p>Calculate the Fraction of Missing Data of Sample Runs</p></a></li>
<li><a href='#run_skewness'><p>Calculate the Skewness of Sample Runs</p></a></li>
<li><a href='#set_check_names'><p><b>DEPRECATED</b>: Set check.names attribute of omicsData object</p></a></li>
<li><a href='#set_data_info'><p>Functions to set omicsData attributes &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-</p></a></li>
<li><a href='#set_filter'><p>Sets/updates the filters attribute with a filter class object.</p></a></li>
<li><a href='#set_isobaric_info'><p>Sets/updates the values in the isobaric_info attribute.</p></a></li>
<li><a href='#set_meta_info'><p>Sets/updates the values in the meta_info attribute</p></a></li>
<li><a href='#set_nmr_info'><p>Sets/updates the values in the nmr_info attribute.</p></a></li>
<li><a href='#spans_make_distribution'><p>Creates the list of median p-values used to make the background distribution</p>
used to compute the SPANS score in step 2.</a></li>
<li><a href='#spans_procedure'><p>Calculate SPANS Score for a Number of Normalization Methods</p></a></li>
<li><a href='#statres_barplot'><p>Fold change barplots for statres objects</p></a></li>
<li><a href='#statRes_output'><p>Function to take raw output of 'imd_anova' and create output for 'statRes' object</p></a></li>
<li><a href='#statres_volcano_plot'><p>Volcano plot for the anova results of imd-anova</p></a></li>
<li><a href='#statRes-class'><p>Summary of statRes Object</p></a></li>
<li><a href='#summary_km'><p>Basic survival analysis summary</p></a></li>
<li><a href='#summary-isobaricnormRes'><p>Summary for isobaricnormRes Object</p></a></li>
<li><a href='#summary-nmrnormRes'><p>Summary of nmrnormRes Object</p></a></li>
<li><a href='#summary-omicsData'><p>Produce a basic summary of a pmartR omicsData S3 Object</p></a></li>
<li><a href='#summary-pmartR-results'><p>Summary of pmartR Analysis Functions</p></a></li>
<li><a href='#summary-trelliData'><p>Summarizes potential plotting options for a trelliData object</p></a></li>
<li><a href='#summary.customFilt'><p>Custom Filter Summary</p></a></li>
<li><a href='#summary.cvFilt'><p>Coefficient of Variation (CV) Filter Summary</p></a></li>
<li><a href='#summary.imdanovaFilt'><p>IMD-ANOVA Filter Summary</p></a></li>
<li><a href='#summary.moleculeFilt'><p>Molecule Filter Summary</p></a></li>
<li><a href='#summary.proteomicsFilt'><p>Proteomics Filter Summary</p></a></li>
<li><a href='#summary.rmdFilt'><p>RMD Filter Summary</p></a></li>
<li><a href='#summary.RNAFilt'><p>RNA Filter Summary</p></a></li>
<li><a href='#summary.totalCountFilt'><p>Total Count Filter Summary</p></a></li>
<li><a href='#surv_designation'><p>Create a &quot;surv_DF&quot; attribute so that survival analysis can be implemented.</p></a></li>
<li><a href='#take_diff'><p>Compute pairwise differences</p></a></li>
<li><a href='#total_count_filter'><p>Total Count Filter Object</p></a></li>
<li><a href='#trelli_abundance_boxplot'><p>Boxplot trelliscope building function for abundance data</p></a></li>
<li><a href='#trelli_abundance_heatmap'><p>Heatmap trelliscope building function for abundance data</p></a></li>
<li><a href='#trelli_abundance_histogram'><p>Histogram trelliscope building function for abundance data</p></a></li>
<li><a href='#trelli_foldchange_bar'><p>Bar chart trelliscope building function for fold_change</p></a></li>
<li><a href='#trelli_foldchange_boxplot'><p>Boxplot trelliscope building function for fold_changes</p></a></li>
<li><a href='#trelli_foldchange_heatmap'><p>Heatmap trelliscope building function for fold_change</p></a></li>
<li><a href='#trelli_foldchange_volcano'><p>Volcano trelliscope building function for fold_change</p></a></li>
<li><a href='#trelli_missingness_bar'><p>Bar chart trelliscope building function for missing data</p></a></li>
<li><a href='#trelli_panel_by'><p>Set the &quot;panel_by&quot; variable for a trelliData object</p></a></li>
<li><a href='#trelli_precheck'><p>Performs initial checks for trelliData objects</p></a></li>
<li><a href='#trelli_pvalue_filter'><p>Filter a paneled trelliData object by a p-value</p></a></li>
<li><a href='#trelli_rnaseq_boxplot'><p>Boxplot trelliscope building function for RNA-seq data</p></a></li>
<li><a href='#trelli_rnaseq_heatmap'><p>Heatmap trelliscope building function for RNA-seq data</p></a></li>
<li><a href='#trelli_rnaseq_histogram'><p>Histogram trelliscope building function for RNA-Seq data</p></a></li>
<li><a href='#trelli_rnaseq_nonzero_bar'><p>Bar chart trelliscope building function for Non-Zero counts in RNA-seq data</p></a></li>
<li><a href='#vector_replace'><p>Replace x with y for a single vector</p></a></li>
<li><a href='#voom_wrapper'><p>Wrapper for limma-voom workflow</p></a></li>
<li><a href='#zrollup'><p>Applies zrollup function</p></a></li>
<li><a href='#zscore_transform'><p>Z-Score Transformation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Panomics Marketplace - Quality Control and Statistical Analysis
for Panomics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-2-27</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functionality for quality control processing and statistical analysis of mass spectrometry (MS) omics data, in particular proteomic (either at the peptide or the protein level), lipidomic, and metabolomic data, as well as RNA-seq based count data and nuclear magnetic resonance (NMR) data. This includes data transformation, specification of groups that are to be compared against each other, filtering of features and/or samples, data normalization, data summarization (correlation, PCA), and statistical comparisons between defined groups.  Implements methods described in:  Webb-Robertson et al. (2014) &lt;<a href="https://doi.org/10.1074%2Fmcp.M113.030932">doi:10.1074/mcp.M113.030932</a>&gt;.  Webb-Robertson et al. (2011) &lt;<a href="https://doi.org/10.1002%2Fpmic.201100078">doi:10.1002/pmic.201100078</a>&gt;.  Matzke et al. (2011) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtr479">doi:10.1093/bioinformatics/btr479</a>&gt;.  Matzke et al. (2013) &lt;<a href="https://doi.org/10.1002%2Fpmic.201200269">doi:10.1002/pmic.201200269</a>&gt;.  Polpitiya et al. (2008) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn217">doi:10.1093/bioinformatics/btn217</a>&gt;.  Webb-Robertson et al. (2010) &lt;<a href="https://doi.org/10.1021%2Fpr1005247">doi:10.1021/pr1005247</a>&gt;.  </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-2-Clause">BSD_2_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://pmartr.github.io/pmartR/">https://pmartr.github.io/pmartR/</a>, <a href="https://github.com/pmartR/pmartR">https://github.com/pmartR/pmartR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/pmartR/pmartR/issues">https://github.com/pmartR/pmartR/issues</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, BH</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.8), data.table, doParallel, dplyr, ggplot2,
e1071, foreach, methods, mvtnorm, pcaMethods, purrr, rrcov,
stringr, tidyr (&ge; 1.3.0), RColorBrewer, magrittr, parallelly,
patchwork, glmpca,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, limma, rmarkdown, edgeR, DESeq2, plotly, scales,
S4Vectors, survival, testthat, trelliscopejs, pmartRdata,
vdiffr</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://pmartR.github.io/drat">https://pmartR.github.io/drat</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-27 19:47:09 UTC; clab683</td>
</tr>
<tr>
<td>Author:</td>
<td>Lisa Bramer [aut, cre],
  Kelly Stratton [aut],
  Daniel Claborne [aut],
  Evan Glasscock [ctb],
  Rachel Richardson [ctb],
  David Degnan [ctb],
  Evan Martin [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lisa Bramer &lt;lisa.bramer@pnnl.gov&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-27 21:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.is_edata'>Test if a file is an edata file</h2><span id='topic+.is_edata'></span>

<h3>Description</h3>

<p>Test if a file is an edata file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.is_edata(edata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".is_edata_+3A_edata">edata</code></td>
<td>
<p>Must be a dataframe. Required.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean where TRUE means the file is an acceptable edata file.
</p>

<hr>
<h2 id='all_subset'>Identify All Biomolecules for Use in Normalization</h2><span id='topic+all_subset'></span>

<h3>Description</h3>

<p>Selects biomolecules for normalization via choosing all biomolecules
currently in the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>all_subset(e_data, edata_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="all_subset_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, proteins, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="all_subset_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns the subset of all biomolecules. These will be
used for normalization.
</p>


<h3>Value</h3>

<p>Character vector containing all biomolecules.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>

<hr>
<h2 id='anova_filter'>Identifies biomolecules to be filtered in preparation for IMD-ANOVA.</h2><span id='topic+anova_filter'></span>

<h3>Description</h3>

<p>The method identifies biomolecules to be filtered specifically according data
requirements for running an ANOVA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anova_filter(nonmiss_per_group, min_nonmiss_anova, comparisons = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anova_filter_+3A_nonmiss_per_group">nonmiss_per_group</code></td>
<td>
<p>a list of length two. The first element giving the
total number of possible samples for each group. The second element giving
a data.frame with the first column giving the biomolecule identifier and
the second through kth columns giving the number of non-missing
observations for each of the <code>k</code> groups. Usually the result of
<code><a href="#topic+nonmissing_per_group">nonmissing_per_group</a></code></p>
</td></tr>
<tr><td><code id="anova_filter_+3A_min_nonmiss_anova">min_nonmiss_anova</code></td>
<td>
<p>the minimum number of nonmissing biomolecule values
required, in each group, in order for the biomolecule to not be filtered.
Must be greater than or equal to 2; default value is 2.</p>
</td></tr>
<tr><td><code id="anova_filter_+3A_comparisons">comparisons</code></td>
<td>
<p>data.frame with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control. If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function filters biomolecules that do not have at least
<code>min.nonmiss.allowed</code> values per group, where groups are from
<code>group_designation</code>.
</p>


<h3>Value</h3>

<p>filter.peps a character vector of the biomolecules to be filtered out
prior to ANOVA or IMD-ANOVA
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nonmissing_per_group">nonmissing_per_group</a></code>
</p>

<hr>
<h2 id='anova_test'>Tests for a quantiative difference between groups (aka factors, aka main
effects)</h2><span id='topic+anova_test'></span>

<h3>Description</h3>

<p>This is the ANOVA part of the IMD-ANOVA test proposed in Webb-Robertson et
al. (2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anova_test(
  omicsData,
  groupData,
  comparisons,
  pval_adjust_multcomp,
  pval_adjust_fdr,
  pval_thresh,
  covariates,
  paired,
  equal_var,
  parallel
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anova_test_+3A_omicsdata">omicsData</code></td>
<td>
<p>A pmartR data object of any class</p>
</td></tr>
<tr><td><code id="anova_test_+3A_groupdata">groupData</code></td>
<td>
<p>'data.frame' that assigns sample names to groups</p>
</td></tr>
<tr><td><code id="anova_test_+3A_comparisons">comparisons</code></td>
<td>
<p>'data.frame' with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="anova_test_+3A_pval_adjust_multcomp">pval_adjust_multcomp</code></td>
<td>
<p>character string specifying the type of multiple
comparisons adjustment to implement. The default, &quot;none&quot;, corresponds to no
adjustment. Valid options include: &quot;bonferroni&quot;, &quot;holm&quot;, &quot;tukey&quot;, and
&quot;dunnett&quot;.</p>
</td></tr>
<tr><td><code id="anova_test_+3A_pval_adjust_fdr">pval_adjust_fdr</code></td>
<td>
<p>character string specifying the type of FDR adjustment
to implement. The default, &quot;none&quot;, corresponds to no adjustment. Valid
options include: &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, and &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="anova_test_+3A_pval_thresh">pval_thresh</code></td>
<td>
<p>numeric p-value threshold, below or equal to which
peptides are considered differentially expressed. Defaults to 0.05</p>
</td></tr>
<tr><td><code id="anova_test_+3A_covariates">covariates</code></td>
<td>
<p>A character vector with no more than two variable names
that will be used as covariates in the IMD-ANOVA analysis.</p>
</td></tr>
<tr><td><code id="anova_test_+3A_paired">paired</code></td>
<td>
<p>logical; should the data be paired or not? if TRUE then the
'f_data' element of 'omicsData' is checked for a &quot;Pair&quot; column, an error is
returned if none is found</p>
</td></tr>
<tr><td><code id="anova_test_+3A_equal_var">equal_var</code></td>
<td>
<p>logical; should the variance across groups be assumed equal?</p>
</td></tr>
<tr><td><code id="anova_test_+3A_parallel">parallel</code></td>
<td>
<p>A logical value indicating if the t test should be run in
parallel.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The order in which different scenarios are handeled: </p>
 <ol>
<li><p> If the
data are paired, then the pairing is accounted for first then each of the
next steps is carried out on the new variable that is the difference in the
paired individuals.&lt;br&gt; </p>
</li>
<li><p> If covariates are provided, their effect is
removed before testing for group differences though mathematically covariates
and grouping effects are accounted for simultaneously </p>
</li>
<li><p> ANOVA is executed
to assess the effect of each main effects, results in a vector of group means
for each biomolecule and variance estimate </p>
</li>
<li><p> Group comparisons defined by
'comaprison' argument are implemented use parameter vector and variance
estimates in ANOVA step </p>
</li></ol>



<h3>Value</h3>

<p>a list of 'data.frame's
</p>

<table>
<tr>
 <td style="text-align: left;">
  Results  </td><td style="text-align: left;"> Edata cname,
  Variance Estimate, ANOVA F-Statistic, ANOVA p-value, Group means</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;">
  </td>
</tr>
<tr>
 <td style="text-align: left;"> Fold_changes  </td><td style="text-align: left;"> Estimated fold-changes for each comparison </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;">
  </td>
</tr>
<tr>
 <td style="text-align: left;"> Fold_changes_pvalues  </td><td style="text-align: left;"> P-values corresponding to the fold-changes
  for each comparison </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> Fold_change_flags  </td><td style="text-align: left;"> Indicator of
  statistical significance (0/+-2 to if adjusted p-value&gt;=pval_thresh or
  p-value&lt;pval_thresh) </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Bryan Stanfill, Daniel Claborne
</p>


<h3>References</h3>

<p>Webb-Robertson, Bobbie-Jo M., et al. &quot;Combined statistical
analyses of peptide intensities and peptide occurrences improves
identification of significant peptides from MS-based proteomics data.&quot;
Journal of proteome research 9.11 (2010): 5748-5756.
</p>

<hr>
<h2 id='applyFilt'>Apply a S3 filter object to a pmartR S3 object</h2><span id='topic+applyFilt'></span><span id='topic+applyFilt.moleculeFilt'></span><span id='topic+applyFilt.totalCountFilt'></span><span id='topic+applyFilt.RNAFilt'></span><span id='topic+applyFilt.cvFilt'></span><span id='topic+applyFilt.rmdFilt'></span><span id='topic+applyFilt.proteomicsFilt'></span><span id='topic+applyFilt.imdanovaFilt'></span><span id='topic+applyFilt.customFilt'></span>

<h3>Description</h3>

<p>This function takes a filter object of class 'cvFilt', 'rmdFilt',
'moleculeFilt', 'proteomicsFilt', 'imdanovaFilt', 'RNAFilt', 'totalCountFilt',
or 'customFilt' and  applies the filter to a dataset of <code>pepData</code>,
<code>proData</code>, <code>lipidData</code>, <code>metabData</code>, <code>nmrData</code> or
<code>seqData</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>applyFilt(filter_object, omicsData, ...)

## S3 method for class 'moleculeFilt'
applyFilt(filter_object, omicsData, min_num = 2, ...)

## S3 method for class 'totalCountFilt'
applyFilt(filter_object, omicsData, min_count, ...)

## S3 method for class 'RNAFilt'
applyFilt(
  filter_object,
  omicsData,
  min_nonzero = NULL,
  size_library = NULL,
  ...
)

## S3 method for class 'cvFilt'
applyFilt(filter_object, omicsData, cv_threshold = 150, ...)

## S3 method for class 'rmdFilt'
applyFilt(
  filter_object,
  omicsData,
  pvalue_threshold = 1e-04,
  min_num_biomolecules = 50,
  ...
)

## S3 method for class 'proteomicsFilt'
applyFilt(
  filter_object,
  omicsData,
  min_num_peps = NULL,
  redundancy = FALSE,
  ...
)

## S3 method for class 'imdanovaFilt'
applyFilt(
  filter_object,
  omicsData,
  comparisons = NULL,
  min_nonmiss_anova = NULL,
  min_nonmiss_gtest = NULL,
  remove_singleton_groups = TRUE,
  ...
)

## S3 method for class 'customFilt'
applyFilt(filter_object, omicsData, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyFilt_+3A_filter_object">filter_object</code></td>
<td>
<p>an object of the class 'cvFilt', 'proteomicsFilt',
'rmdFilt', 'moleculeFilt', 'imdanovaFilt', 'customFilt', 'RNAFilt', or
'totalCountFilt' created by <code>cv_filter</code>, <code>proteomics_filter</code>,
<code>rmd_filter</code>, <code>molecule_filter</code>, <code>imdanova_filter</code>,
<code>custom_filter</code>, <code>RNA_filter</code>, or <code>total_count_filter</code>,
respectively.</p>
</td></tr>
<tr><td><code id="applyFilt_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class <code>pepData</code>, <code>proData</code>,
<code>lipidData</code>, <code>metabData</code>, <code>nmrData</code>, or <code>seqData</code>
usually created by <code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="applyFilt_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
<tr><td><code id="applyFilt_+3A_min_num">min_num</code>, <code id="applyFilt_+3A_min_count">min_count</code>, <code id="applyFilt_+3A_min_nonzero">min_nonzero</code>, <code id="applyFilt_+3A_size_library">size_library</code>, <code id="applyFilt_+3A_cv_threshold">cv_threshold</code>, <code id="applyFilt_+3A_pvalue_threshold">pvalue_threshold</code>, <code id="applyFilt_+3A_min_num_biomolecules">min_num_biomolecules</code>, <code id="applyFilt_+3A_min_num_peps">min_num_peps</code>, <code id="applyFilt_+3A_redundancy">redundancy</code>, <code id="applyFilt_+3A_comparisons">comparisons</code>, <code id="applyFilt_+3A_min_nonmiss_anova">min_nonmiss_anova</code>, <code id="applyFilt_+3A_min_nonmiss_gtest">min_nonmiss_gtest</code>, <code id="applyFilt_+3A_remove_singleton_groups">remove_singleton_groups</code></td>
<td>
<p>Arguments that depend on the class of <code>filter_object</code>, see details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Further arguments can be specified depending on the class of the
<code>filter_object</code> being applied.
</p>
<p>For a <code>filter_object</code> of type 'moleculeFilt': </p>

<table>
<tr>
 <td style="text-align: left;">
  <code>min_num</code> </td><td style="text-align: left;"> an integer value specifying the minimum number of times
  each biomolecule must be observed across all samples in order to retain the
  biomolecule. Default value is 2. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>For a <code>filter_object</code> of type 'cvFilt': </p>

<table>
<tr>
 <td style="text-align: left;">
  <code>cv_threshold</code> </td><td style="text-align: left;"> an integer value specifying the maximum
  coefficient of variation (CV) threshold for the biomolecules. Biomolecules
  with CV greater than this threshold will be filtered. Default threshold is
  150. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>For a <code>filter_object</code> of type 'rmdFilt': </p>

<table>
<tr>
 <td style="text-align: left;">
  <code>pvalue_threshold</code> </td><td style="text-align: left;"> numeric value between 0 and 1, specifying the
  p-value below which samples will be removed from the dataset. Default is
  0.001. </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>min_num_biomolecules</code> </td><td style="text-align: left;"> numeric value greater than 10
  (preferably greater than 50) that specifies the minimum number of
  biomolecules that must be present in order to create an rmdFilt object.
  Using values less than 50 is not advised. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>For a <code>filter_object</code> of type 'proteomicsFilt' either or both of the
following can be applied: </p>

<table>
<tr>
 <td style="text-align: left;"> <code>min_num_peps</code> </td><td style="text-align: left;"> an
  optional integer value between 1 and the maximum number of peptides that
  map to a protein in omicsData. The value specifies the minimum number of
  peptides that must map to a protein. Any protein with less than
  <code>min_num_peps</code> mapping to it will be removed from the dataset. Default
  value is NULL, meaning that this filter is not applied. </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>redundancy</code> </td><td style="text-align: left;"> logical indicator of whether to filter out
  degenerate/redundant peptides (peptides that map to more than one protein).
  Default value is FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>For a <code>filter_object</code> of type 'imdanovaFilt': </p>

<table>
<tr>
 <td style="text-align: left;">
  <code>min_nonmiss_anova</code> </td><td style="text-align: left;"> integer value specifying the minimum number
  of non-missing feature values allowed per group for <code>anova_filter</code>.
  Default value is 2. </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>min_nonmiss_gtest</code> </td><td style="text-align: left;"> integer value
  specifying the minimum number of non-missing feature values allowed per
  group for <code>gtest_filter</code>. Default value is 3.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>For a <code>filter_object</code> of type 'totalCountFilt': </p>

<table>
<tr>
 <td style="text-align: left;">
  <code>min_count</code> </td><td style="text-align: left;"> integer value specifying the minimum number of
  biomolecule counts observed across all samples in order for the biomolecule
  to be retained in the dataset. This filter is only applicable for
  <code>seqData</code> objects. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>For a <code>filter_object</code> of type 'RNAFilt' either or both of the
following can be applied: </p>

<table>
<tr>
 <td style="text-align: left;"> <code>min_nonzero</code> </td><td style="text-align: left;"> integer
  value specifying the minimum number of non-zero feature values per sample.
  </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>size_library</code> </td><td style="text-align: left;"> integer value or fraction between 0 and 1
  specifying the minimum number of total reads per sample. This filter is
  only applicable for <code>seqData</code> objects. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>

<p>There are no further arguments for a <code>filter_object</code> of type '
customFilt'.
</p>


<h3>Value</h3>

<p>An object of the class <code>pepData</code>, <code>proData</code>,
<code>lipidData</code>, <code>metabData</code>, <code>nmrData</code>, or <code>seqData</code> with
specified cname_ids, edata_cnames, and emeta_cnames filtered out of the
appropriate datasets.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+molecule_filter">molecule_filter</a></code>
</p>
<p><code><a href="#topic+imdanova_filter">imdanova_filter</a></code>
</p>
<p><code><a href="#topic+rmd_filter">rmd_filter</a></code>
</p>
<p><code><a href="#topic+cv_filter">cv_filter</a></code>
</p>
<p><code><a href="#topic+proteomics_filter">proteomics_filter</a></code>
</p>
<p><code><a href="#topic+custom_filter">custom_filter</a></code>
</p>
<p><code><a href="#topic+RNA_filter">RNA_filter</a></code>
</p>
<p><code><a href="#topic+total_count_filter">total_count_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
to_filter &lt;- molecule_filter(omicsData = pep_object)
summary(to_filter, min_num = 2)
pep_object2 &lt;- applyFilt(
  filter_object = to_filter,
  omicsData = pep_object, min_num = 2
)
summary(pep_object2) # number of Peptides is as expected based on summary of 
                     # the filter object that was applied
pep_object2 &lt;- group_designation(omicsData = pep_object2,
                                 main_effects = "Phenotype")
to_filter2 &lt;- imdanova_filter(omicsData = pep_object2)
pep_object3 &lt;- applyFilt(
  filter_object = to_filter2,
  omicsData = pep_object2,
  min_nonmiss_anova = 3
)

</code></pre>

<hr>
<h2 id='as.isobaricpepData'>Create pmartR Object of Class isobaricpepData</h2><span id='topic+as.isobaricpepData'></span>

<h3>Description</h3>

<p>Converts several data frames of isobaric peptide data
to an object of the class 'isobaricpepData'. Objects of the class
'isobaricpepData' are lists with two obligatory components, <code>e_data</code> and
<code>f_data</code>. An optional list component, <code>e_meta</code>, is used if analysis
or visualization at other levels (e.g. protein) is also desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.isobaricpepData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.isobaricpepData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of peptides observed and <code class="reqn">n</code> is the number of
samples (an additional peptide identifier/name column should also be
present in the data frame). Each row corresponds to data for one
peptide. One column specifying a unique identifier for each peptide (row)
must be present.</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample.</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with at least <code class="reqn">p</code> rows. Each row
corresponds to a peptide with one column giving peptide names (must be
named the same as the column in <code>e_data</code>) and other columns giving biomolecule
meta information (e.g. mappings of peptides to proteins).</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the peptide identifiers in <code>e_data</code> and <code>e_meta</code> (if
applicable).</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the protein identifiers (or other mapping variable) in
<code>e_meta</code> (if applicable). Defaults to NULL. If <code>e_meta</code> is NULL,
then either do not specify <code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> containing the identifiers for the biological samples if the
observations represent technical replicates.  This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.isobaricpepData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The class 'isobaricpepData' is meant to deal with labeled peptide data
generated on instruments (e.g. TMT, iTRAQ) where a reference pool sample will be utilized for normalization.
</p>
<p>If your data has already undergone normalization to the reference pool, you
should specify <code>isobaric_norm = T</code>.
</p>
<p>Objects of class 'isobaricpepData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
 Scale of the data provided in <code>e_data</code>. Acceptable values are 'log2',
 'log10', 'log', and 'abundance', which indicate data is log base 2, base 10,
 natural log transformed, and raw abundance, respectively. Default is
 'abundance'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument, specifying
 whether the data has been normalized or not (this normalization refers to a statistical normalization, such as median centering or other methods). Default value is FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> isobaric_norm </td><td style="text-align: left;"> A logical argument, specifying whether the data
 has been normalized to the appropriate reference pool sample or not.
 Default value is FALSE </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an empty
 list, which will be populated with a single named element
 <code>is_normalized = is_normalized</code>. When a normalization is applied to the
 data, this becomes populated with a list containing the normalization
 function, normalization subset and subset parameters, the location and scale
 parameters used to normalize the data, and the location and scale parameters
 used to backtransform the data (if applicable). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> data_types </td><td style="text-align: left;">
 Character string describing the type of data, most commonly used for lipidomic data (lipidData objects) or NMR data (nmrData objects) but available for other data classes as well. Default
 value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed values included in the <code>data_info</code>
attribute are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> num_edata </td><td style="text-align: left;"> The number of unique
 <code>edata_cname</code> entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_miss_obs </td><td style="text-align: left;"> The number of
 missing observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_zero_obs </td><td style="text-align: left;"> For seqData only: The
 number of zero observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_emeta </td><td style="text-align: left;"> The number of unique
 <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_missing </td><td style="text-align: left;"> The proportion of
 <code>e_data</code> values that are NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_zeros </td><td style="text-align: left;"> For seqData
 only: the proportion of zero counts observed in <code>e_data</code> values. </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_samps </td><td style="text-align: left;"> The number of samples that make up the columns of
 <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> meta_info </td><td style="text-align: left;"> A logical argument, specifying
 whether <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class isobaricpepData and pepData.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.pepData">as.pepData</a></code>
</p>
<p><code><a href="#topic+normalize_isobaric">normalize_isobaric</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- as.isobaricpepData(
  e_data = isobaric_edata,
  e_meta = isobaric_emeta,
  f_data = isobaric_fdata,
  edata_cname = "Peptide",
  fdata_cname = "SampleID",
  emeta_cname = "Protein"
)

</code></pre>

<hr>
<h2 id='as.lipidData'>Create pmartR Object of Class lipidData</h2><span id='topic+as.lipidData'></span>

<h3>Description</h3>

<p>Converts several data frames of lipid data to an
object of the class 'lipidData'. Objects of the class 'lipidData' are lists
with two obligatory components, <code>e_data</code> and <code>f_data</code>. An optional
list component, <code>e_meta</code>, is used if analysis or visualization at other
levels (e.g. lipid class) is also desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.lipidData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.lipidData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of lipids observed and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for one lipid. One column specifying
a unique identifier for each lipid (row) must be present.</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample.</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with <code class="reqn">p</code> rows. Each row corresponds
to a lipid with one column giving lipid names (must be named the same as
the column in <code>e_data</code>) and other columns giving biomolecule meta information.</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the lipid identifiers in <code>e_data</code> and <code>e_meta</code> (if
applicable).</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the mapped identifiers in <code>e_meta</code> (if applicable). Can be the same as edata_cname, if desired.
Defaults to NULL. If <code>e_meta</code> is NULL, then either do not specify
<code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> that specifies which samples are technical replicates. This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.lipidData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects of class 'lipidData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
  Scale of the data provided in <code>e_data</code>. Acceptable values are 'log2',
  'log10', 'log', and 'abundance', which indicate data is log base 2, base
  10, natural log, or raw abundance, respectively. Default
  values is 'abundance'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument,
  specifying whether the data has been normalized or not. Default value is
  FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an empty list, which
  will be populated with a single named element <code>is_normalized =
  is_normalized</code>. When a normalization is applied to the data, this becomes
  populated with a list containing the normalization function, normalization
  subset and subset parameters, the location and scale parameters used to
  normalize the data, and the location and scale parameters used to
  backtransform the data (if applicable). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> data_types </td><td style="text-align: left;">
  Character string describing the type of data (e.g. 'Positive ion' or Negative ion for lipid data). Default
  value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed values included in the <code>data_info</code>
attribute are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> <code>num_edata</code> </td><td style="text-align: left;"> The number of
  unique <code>edata_cname</code> entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>num_miss_obs</code> </td><td style="text-align: left;"> The
  number of missing observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>num_emeta</code> </td><td style="text-align: left;"> The
  number of unique <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>prop_missing</code> </td><td style="text-align: left;"> The proportion of <code>e_data</code> values that are
  NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>num_samps</code> </td><td style="text-align: left;"> The number of samples that make up
  the columns of <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>meta_info</code> </td><td style="text-align: left;"> A logical
  argument, specifying where the <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class lipidData
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.metabData">as.metabData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mylipid &lt;- as.lipidData(
  e_data = lipid_neg_edata,
  f_data = lipid_neg_fdata,
  edata_cname = "Lipid",
  fdata_cname = "SampleID"
)

</code></pre>

<hr>
<h2 id='as.metabData'>Create pmartR Object of Class metabData</h2><span id='topic+as.metabData'></span>

<h3>Description</h3>

<p>Converts several data frames of metabolomic data to an
object of the class 'metabData'. Objects of the class 'metabData' are lists
with two obligatory components, <code>e_data</code> and <code>f_data</code>. An optional
list component, <code>e_meta</code>, is used if analysis or visualization at other
levels (e.g. metabolite identification) is also desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.metabData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.metabData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of metabolites observed and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for one metabolite. One column
specifying a unique identifier for each metabolite (row) must be present.</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample.</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with <code class="reqn">p</code> rows. Each row corresponds
to a metabolite with one column giving metabolite names (must be named the
same as the column in <code>e_data</code>) and other columns giving meta
information.</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the metabolite identifiers in <code>e_data</code> and <code>e_meta</code>
(if applicable).</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the mapped identifiers in <code>e_meta</code> (if applicable).
Defaults to NULL. Can be the same as edata_cname, if desired. If <code>e_meta</code> is NULL, then either do not specify
<code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> that specifies which samples are technical replicates. This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.metabData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects of class 'metabData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
  Scale of the data provided in <code>e_data</code>. Acceptable values are 'log2',
  'log10', 'log', and 'abundance', which indicate data is log base 2, base
  10, natural log, or raw abundance, respectively. Default is
  'abundance'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument, specifying
  whether the data has been normalized or not. Default value is FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an empty list, which will be
  populated with a single named element <code>is_normalized = is_normalized</code>.
  When a normalization is applied to the data, this becomes populated with a
  list containing the normalization function, normalization subset and subset
  parameters, the location and scale parameters used to normalize the data,
  and the location and scale parameters used to backtransform the data (if
  applicable). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> data_types </td><td style="text-align: left;"> Character string describing the type of data, most commonly used for lipidomic data (lipidData objects) or NMR data (nmrData objects) but available for other data classes as well. Default value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed
values included in the <code>data_info</code> attribute are as follows:
</p>

<table>
<tr>
 <td style="text-align: left;"> num_edata </td><td style="text-align: left;"> The number of unique <code>edata_cname</code>
  entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_miss_obs </td><td style="text-align: left;"> The number of missing
  observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_emeta </td><td style="text-align: left;"> The number of unique
  <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_missing </td><td style="text-align: left;"> The proportion
  of <code>e_data</code> values that are NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_samps </td><td style="text-align: left;"> The number
  of samples that make up the columns of <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> meta_info
  </td><td style="text-align: left;"> A logical argument, specifying where the <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class metabData
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.lipidData">as.lipidData</a></code>
</p>
<p><code><a href="#topic+as.nmrData">as.nmrData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetabData &lt;- as.metabData(
  e_data = metab_edata,
  f_data = metab_fdata,
  edata_cname = "Metabolite",
  fdata_cname = "SampleID"
)

</code></pre>

<hr>
<h2 id='as.multiData'>Create a 'multiData' object from multiple omicsData objects</h2><span id='topic+as.multiData'></span>

<h3>Description</h3>

<p>Create a 'multiData' object from multiple omicsData objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.multiData(
  ...,
  f_meta = NULL,
  sample_intersect = FALSE,
  match_samples = TRUE,
  keep_sample_info = FALSE,
  auto_fmeta = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.multiData_+3A_...">...</code></td>
<td>
<p>two or more objects of type 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData', created by <code><a href="#topic+as.pepData">as.pepData</a></code></p>
</td></tr>
<tr><td><code id="as.multiData_+3A_f_meta">f_meta</code></td>
<td>
<p>A data.frame containing sample and group information for all
omicsData objects supplied to the function.</p>
</td></tr>
<tr><td><code id="as.multiData_+3A_sample_intersect">sample_intersect</code></td>
<td>
<p>logical indicator for whether only the samples that
are common across all datasets be kept in f_meta. See details for how
samples will be dropped.</p>
</td></tr>
<tr><td><code id="as.multiData_+3A_match_samples">match_samples</code></td>
<td>
<p>logical indicator. If auto_fmeta = TRUE, whether to 
attempt to match the names in the sample columns in f_data across all
objects in an attempt to align them in f_meta. Defaults to TRUE.</p>
</td></tr>
<tr><td><code id="as.multiData_+3A_keep_sample_info">keep_sample_info</code></td>
<td>
<p>logical indicator for whether to attempt to append
sample information contained in the objects' f_data to the final f_meta via
a series of left joins. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="as.multiData_+3A_auto_fmeta">auto_fmeta</code></td>
<td>
<p>logical indicator for whether to attempt to automatically
construct f_meta from the objects' sample information. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Object limits: Currently, as.multiData accepts at most one object
from each of classes 'pepData/proData', 'metabData', 'nmrData', and at most
two objects of class 'lipidData'.
</p>
<p><code>sample_intersect</code> will auto-align samples that occur in all datasets.
Specifically, it creates a vector of all samples that are common across all
datasets, and simply creates an f_meta by copying this vector for each dataset
and column-binding them.
</p>


<h3>Value</h3>

<p>Object of class 'multiData' containing the omicsData objects, and the
sample alignment information f_meta.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+combine_lipidData">combine_lipidData</a></code> if you want to combine lipidData
objects before providing them to as.multiData.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

# Combine metabolomics and protein object into multidata, both must be log2
# and normalized.
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
mymetab &lt;- normalize_global(omicsData = mymetab, subset_fn = "all",
                            norm_fn = "median", apply_norm = TRUE)

mypro &lt;- pro_object

# Combine without specifically supplying f_meta, either directly, or as one
# of the f_datas in any object.
mymultidata &lt;- as.multiData(mymetab, mypro, auto_fmeta = TRUE, sample_intersect = TRUE)

# Manually supply an f_meta
f_meta &lt;- data.frame(
  "Proteins" = mypro$f_data$SampleID[match(mymetab$f_data$SampleID, mypro$f_data$SampleID)],
  "Metabolites" = mymetab$f_data$SampleID,
  "Condition" = mymetab$f_data$Phenotype[match(mymetab$f_data$SampleID, mypro$f_data$SampleID)]
)

mymultidata &lt;- as.multiData(mymetab, mypro, f_meta = f_meta)
# remove samples that are not common across all data.
mymultidata &lt;- as.multiData(mymetab, mypro, f_meta = f_meta, sample_intersect = TRUE)

</code></pre>

<hr>
<h2 id='as.nmrData'>Create pmartR Object of Class nmrData</h2><span id='topic+as.nmrData'></span>

<h3>Description</h3>

<p>Converts several data frames of NMR-generated
metabolomic data to an object of the class 'nmrData'. Objects of the
class 'nmrData' are lists with two obligatory components, <code>e_data</code> and
<code>f_data</code>. An optional list component, <code>e_meta</code>, is used if analysis
or visualization at other levels (e.g. metabolite identification) is also
desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.nmrData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.nmrData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of metabolites observed and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for one metabolite. One column
specifying a unique identifier for each metabolite (row) must be present.</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample.</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with <code class="reqn">p</code> rows. Each row corresponds
to a metabolite with one column giving metabolite names (must be named the
same as the column in <code>e_data</code>) and other columns giving meta
information.</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the metabolite identifiers in <code>e_data</code> and <code>e_meta</code>
(if applicable).</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the mapped identifiers in <code>e_meta</code> (if applicable).
Defaults to NULL. Can be the same as edata_cname, if desired. If <code>e_meta</code> is NULL, then either do not specify
<code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> that specifies which samples are technical replicates. This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.nmrData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects of class 'nmrData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
  Scale of the data provided in <code>e_data</code>. Acceptable values are 'log2',
  'log10', 'log', and 'abundance', which indicate data is log base 2, base
  10, natural log, or raw abundance, respectively. Default is
  'abundance'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument, specifying
  whether the data has been normalized or not. Default value is FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> nmr_norm </td><td style="text-align: left;"> A logical argument, specifying whether the data has
  been normalized either to a spiked in metabolite or to a property taking
  sample-specific values </td>
</tr>
<tr>
 <td style="text-align: left;"> #' </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an
  empty list, which will be populated with a single named element
  <code>is_normalized = is_normalized</code>. When a normalization is applied to
  the data, this becomes populated with a list containing the normalization
  function, normalization subset and subset parameters, the location and
  scale parameters used to normalize the data, and the location and scale
  parameters used to backtransform the data (if applicable). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
  data_types </td><td style="text-align: left;"> Character string describing the type of data (e.g.'binned'
  or 'identified', for NMR data). Default value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed
values included in the <code>data_info</code> attribute are as follows:
</p>

<table>
<tr>
 <td style="text-align: left;"> num_edata </td><td style="text-align: left;"> The number of unique <code>edata_cname</code>
  entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_miss_obs </td><td style="text-align: left;"> The number of missing
  observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_emeta </td><td style="text-align: left;"> The number of unique
  <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_missing </td><td style="text-align: left;"> The proportion
  of <code>e_data</code> values that are NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_samps </td><td style="text-align: left;"> The number
  of samples that make up the columns of <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> meta_info
  </td><td style="text-align: left;"> A logical argument, specifying where the <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class nmrData
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.metabData">as.metabData</a></code>
</p>
<p><code><a href="#topic+normalize_nmr">normalize_nmr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mynmrData &lt;- as.nmrData(
  e_data = nmr_identified_edata,
  f_data = nmr_identified_fdata,
  edata_cname = "Metabolite",
  fdata_cname = "SampleID",
  data_type = "identified"
)

</code></pre>

<hr>
<h2 id='as.pepData'>Create pmartR Object of Class pepData</h2><span id='topic+as.pepData'></span>

<h3>Description</h3>

<p>Converts several data frames of (unlabeled or global, as opposed to labeled) peptide data to an
object of the class 'pepData'. Objects of the class 'pepData' are lists with
two obligatory components, <code>e_data</code> and <code>f_data</code>. An optional list
component, <code>e_meta</code>, is used if analysis or visualization at other levels
(e.g. protein) is also desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.pepData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.pepData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of peptides observed and <code class="reqn">n</code> is the number of
samples (an additional peptide identifier/name column should also be
present somewhere in the data frame). Each row corresponds to data for one
peptide. One column specifying a unique identifier for each peptide (row)
must be present.</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample.</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with at least <code class="reqn">p</code> rows. Each row
corresponds to a peptide with one column giving peptide names (must be
named the same as the column in <code>e_data</code>) and other columns giving biomolecule
meta information (e.g. mappings of peptides to proteins).</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the peptide identifiers in <code>e_data</code> and <code>e_meta</code> (if
applicable).</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the protein identifiers (or other mapping variable) in
<code>e_meta</code> (if applicable). Defaults to NULL. Can be the same as edata_cname, if desired. If <code>e_meta</code> is NULL,
then either do not specify <code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> that specifies which samples are technical replicates. This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.pepData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects of class 'pepData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
  Scale of the data provided in <code>e_data</code>. Acceptable values are 'log2',
  'log10', 'log', and 'abundance', which indicate data is log base 2, base
  10, natural log, or raw abundance, respectively. Default is
  'abundance'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument, specifying
  whether the data has been normalized or not. Default value is FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an empty list, which will be
  populated with a single named element <code>is_normalized = is_normalized</code>.
  When a normalization is applied to the data, this becomes populated with a
  list containing the normalization function, normalization subset and subset
  parameters, the location and scale parameters used to normalize the data,
  and the location and scale parameters used to backtransform the data (if
  applicable). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> data_types </td><td style="text-align: left;"> Character string describing the type of data, most commonly used for lipidomic data (lipidData objects) or NMR data (nmrData objects) but available for other data classes as well. Default value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed
values included in the <code>data_info</code> attribute are as follows:
</p>

<table>
<tr>
 <td style="text-align: left;"> num_edata </td><td style="text-align: left;"> The number of unique <code>edata_cname</code>
  entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_miss_obs </td><td style="text-align: left;"> The number of missing
  observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_emeta </td><td style="text-align: left;"> The number of unique
  <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_missing </td><td style="text-align: left;"> The proportion
  of <code>e_data</code> values that are NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_samps </td><td style="text-align: left;"> The number
  of samples that make up the columns of <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> meta_info
  </td><td style="text-align: left;"> A logical argument, specifying whether <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class pepData
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton, Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.proData">as.proData</a></code>
</p>
<p><code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypepData &lt;- as.pepData(
  e_data = pep_edata,
  e_meta = pep_emeta,
  f_data = pep_fdata,
  edata_cname = "Peptide",
  fdata_cname = "SampleID",
  emeta_cname = "RazorProtein"
)

</code></pre>

<hr>
<h2 id='as.proData'>Create pmartR Object of Class proData</h2><span id='topic+as.proData'></span>

<h3>Description</h3>

<p>Converts several data frames of protein data to an
object of the class 'proData'. Objects of the class 'proData' are lists with
two obligatory components, <code>e_data</code> and <code>f_data</code>. An optional list
component, <code>e_meta</code>, is used if analysis or visualization at other levels
(e.g. gene) is also desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.proData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.proData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of proteins observed and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for one protein. One column
specifying a unique identifier for each protein (row) must be present.</p>
</td></tr>
<tr><td><code id="as.proData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample.</p>
</td></tr>
<tr><td><code id="as.proData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with <code class="reqn">p</code> rows. Each row corresponds
to a protein with one column giving protein names (must be named the same
as the column in <code>e_data</code>) and other columns giving biomolecule meta information
(e.g. mappings of proteins to genes).</p>
</td></tr>
<tr><td><code id="as.proData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the protein identifiers in <code>e_data</code> and <code>e_meta</code> (if
applicable).</p>
</td></tr>
<tr><td><code id="as.proData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.proData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the gene identifiers (or other mapping variable) in
<code>e_meta</code> (if applicable). Defaults to NULL. Can be the same as edata_cname, if desired. If <code>e_meta</code> is NULL,
then either do not specify <code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.proData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> that specifies which samples are technical replicates. This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.proData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects of class 'proData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
  Scale of the data provided in <code>e_data</code>. Acceptable values are 'log2',
  'log10', 'log', and 'abundance', which indicate data is log base 2, base
  10, natural log, or raw abundance, respectively. Default
  values is 'abundance'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument,
  specifying whether the data has been normalized or not. Default value is
  FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an empty list, which
  will be populated with a single named element <code>is_normalized =
  is_normalized</code>. When a normalization is applied to the data, this becomes
  populated with a list containing the normalization function, normalization
  subset and subset parameters, the location and scale parameters used to
  normalize the data, and the location and scale parameters used to
  backtransform the data (if applicable). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> data_types </td><td style="text-align: left;">
  Character string describing the type of data, most commonly used for lipidomic data (lipidData objects) or NMR data (nmrData objects) but available for other data classes as well. Default
  value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed values included in the <code>data_info</code>
attribute are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> num_edata </td><td style="text-align: left;"> The number of unique
  <code>edata_cname</code> entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_miss_obs </td><td style="text-align: left;"> The number of
  missing observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_emeta </td><td style="text-align: left;"> The number of unique
  <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_missing </td><td style="text-align: left;"> The proportion
  of <code>e_data</code> values that are NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_samps </td><td style="text-align: left;"> The number
  of samples that make up the columns of <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> meta_info
  </td><td style="text-align: left;"> A logical argument, specifying whether <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class proData
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton, Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.pepData">as.pepData</a></code>
</p>
<p><code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myproData &lt;- as.proData(
  e_data = pro_edata,
  f_data = pro_fdata,
  edata_cname = "RazorProtein",
  fdata_cname = "SampleID",
  is_normalized = TRUE
)

</code></pre>

<hr>
<h2 id='as.seqData'>Create pmartR Object of Class seqData</h2><span id='topic+as.seqData'></span>

<h3>Description</h3>

<p>Converts several data frames of RNA-seq transcript data to
an object of the class 'seqData'. Objects of the class 'seqData' are lists
with two obligatory components, <code>e_data</code> and <code>f_data</code>. An optional
list component, <code>e_meta</code>, is used if analysis or visualization at other
levels (e.g. gene, protein, pathway) is also desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.seqData(
  e_data,
  f_data,
  e_meta = NULL,
  edata_cname,
  fdata_cname,
  emeta_cname = NULL,
  techrep_cname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.seqData_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of RNA transcripts observed and <code class="reqn">n</code> is the number
of samples (an additional transcript identifier/name column should also be
present somewhere in the data frame). Each row corresponds to data for one
transcript. One column specifying a unique identifier for each transcript
(row) must be present. All counts are required to be raw for processing.</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_f_data">f_data</code></td>
<td>
<p>a data frame with <code class="reqn">n</code> rows. Each row corresponds to a
sample with one column giving the unique sample identifiers found in e_data
column names and other columns providing qualitative and/or quantitative
traits of each sample. For library size normalization, this can be provided as part of f_data
or calculated from columns in e_data.</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_e_meta">e_meta</code></td>
<td>
<p>an optional data frame with at least <code class="reqn">p</code> rows. Each row
corresponds to a transcript with one column giving transcript names (must be
named the same as the column in <code>e_data</code>) and other columns giving biomolecule
meta information (e.g. mappings of transcripts to genes or proteins). Can be the same as edata_cname, if desired.</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the transcript identifiers in <code>e_data</code> and <code>e_meta</code>
(if applicable).</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_fdata_cname">fdata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the sample identifiers in <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the gene identifiers (or other mapping variable) in
<code>e_meta</code> (if applicable). Defaults to NULL. If <code>e_meta</code> is NULL,
then either do not specify <code>emeta_cname</code> or specify it as NULL.</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_techrep_cname">techrep_cname</code></td>
<td>
<p>character string specifying the name of the column in
<code>f_data</code> that specifies which samples are technical replicates. This column is used to
collapse the data when <code>combine_techreps</code> is called on this object.
Defaults to NULL (no technical replicates).</p>
</td></tr>
<tr><td><code id="as.seqData_+3A_...">...</code></td>
<td>
<p>further arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects of class 'seqData' contain some attributes that are
referenced by downstream functions. These attributes can be changed from
their default value by manual specification. A list of these attributes as
well as their default values are as follows: </p>

<table>
<tr>
 <td style="text-align: left;"> data_scale </td><td style="text-align: left;">
  Scale of the data provided in <code>e_data</code>. Only 'counts' is valid for
  'seqData'. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> is_normalized </td><td style="text-align: left;"> A logical argument, specifying
  whether the data has been normalized or not. Default value is FALSE. </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> norm_info </td><td style="text-align: left;"> Default value is an empty list, which will be
  populated with a single named element <code>is_normalized = is_normalized</code>.
  </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> data_types </td><td style="text-align: left;"> Character string describing the type of data, most commonly used for lipidomic data (lipidData objects) or NMR data (nmrData objects) but available for other data classes as well. Default value is NULL. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> Computed
values included in the <code>data_info</code> attribute are as follows:
</p>

<table>
<tr>
 <td style="text-align: left;"> num_edata </td><td style="text-align: left;"> The number of unique <code>edata_cname</code>
  entries.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_zero_obs </td><td style="text-align: left;"> The number of zero-value
  observations.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_emeta </td><td style="text-align: left;"> The number of unique
  <code>emeta_cname</code> entries. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> prop_missing </td><td style="text-align: left;"> The proportion
  of <code>e_data</code> values that are NA. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> num_samps </td><td style="text-align: left;"> The number
  of samples that make up the columns of <code>e_data</code>.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> meta_info
  </td><td style="text-align: left;"> A logical argument, specifying whether <code>e_meta</code> is provided.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>Object of class seqData
</p>


<h3>Author(s)</h3>

<p>Rachel Richardson, Kelly Stratton, Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.proData">as.proData</a></code>
</p>
<p><code><a href="#topic+as.pepData">as.pepData</a></code>
</p>
<p><code><a href="#topic+as.lipidData">as.lipidData</a></code>
</p>
<p><code><a href="#topic+as.metabData">as.metabData</a></code>
</p>
<p><code><a href="#topic+as.nmrData">as.nmrData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myseq &lt;- as.seqData(
  e_data = rnaseq_edata,
  e_meta = rnaseq_emeta,
  f_data = rnaseq_fdata,
  edata_cname = "Transcript",
  fdata_cname = "SampleName",
  emeta_cname = "Transcript"
)

</code></pre>

<hr>
<h2 id='as.trelliData'>Generate an object from omicsData and/or statRes objects to pass to
trelliscope building functions</h2><span id='topic+as.trelliData'></span>

<h3>Description</h3>

<p>Either an omicData and/or a statRes object are accepted.
omicData must be transformed and normalized, unless the data is isobaric
protein or NMR data. If group_designation() has been run on the omicData
object to add &quot;main_effects&quot;, the resulting plots will include groups. The
main effects group_designation and e_meta columns are merged to the e_data
in long format to create the trelliData.omics dataframe, and e_meta is
merged to statRes in long format to create trelliData.stat dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.trelliData(omicsData = NULL, statRes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.trelliData_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'isobaricpepData',
proData', 'metabData', 'lipidData', or 'nmrData', created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="as.trelliData_+3A_statres">statRes</code></td>
<td>
<p>statRes an object of the class 'statRes', created by
<code><a href="#topic+imd_anova">imd_anova</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'trelliData' containing the raw data and optionally, statRes.  
To be passed to trelliscope building functions.
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

###########################
## MS/NMR OMICS EXAMPLES ##
###########################

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData2 &lt;- as.trelliData(omicsData = omicsData)
trelliData3 &lt;- as.trelliData(statRes = statRes)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

######################
## RNA-SEQ EXAMPLES ##  
######################

# Group data by condition
omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq2 &lt;- as.trelliData(omicsData = omicsData_seq)
trelliData_seq3 &lt;- as.trelliData(statRes = statRes_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)


</code></pre>

<hr>
<h2 id='as.trelliData.edata'>Generate an object from edata to pass to trelliscope building
functions</h2><span id='topic+as.trelliData.edata'></span>

<h3>Description</h3>

<p>The only acceptable input file type is a single edata file.
Transformation and normalization must be specified. Isobaric protein or NMR
data does not need to be normalized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.trelliData.edata(
  e_data,
  edata_cname,
  omics_type,
  data_scale_original = "abundance",
  data_scale = "log2",
  normalization_fun = "global",
  normalization_params = list(subset_fn = "all", norm_fn = "median", apply_norm = TRUE,
    backtransform = TRUE),
  is_normalized = FALSE,
  force_normalization = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.trelliData.edata_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p * (n + 1)</code> data.frame of expression data, where
<code class="reqn">p</code> is the number of biomolecules observed and <code class="reqn">n</code> is the number of
samples (an additional biomolecule identifier/name column should also be
present anywhere in the data.frame). Each row corresponds to data for one
biomolecule. One column specifying a unique identifier for each biomolecule
(row) must be present. We do not recommend passing data that requires
reference normalization (isobaric, nmr, etc.)</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_edata_cname">edata_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the biomolecule identifiers. It should be the only non-numeric
colummn in edata.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_omics_type">omics_type</code></td>
<td>
<p>A string specifying the data type. Acceptable options are
&quot;pepData&quot;, &quot;isobaricpepData&quot;, &quot;proData&quot;, &quot;metabData&quot;, &quot;lipidData&quot;, 
&quot;nmrData&quot;, or &quot;seqData&quot;.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_data_scale_original">data_scale_original</code></td>
<td>
<p>A character string indicating original scale
of the data. Valid values are: 'log2', 'log', 'log10', or 'abundance'.
Default is abundance. This parameter is ignored if the data is &quot;seqData&quot;.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_data_scale">data_scale</code></td>
<td>
<p>A character string indicating the scale to transform the
data to. Valid values are: 'log2', 'log', 'log10', or 'abundance'. If the
value is the same as data_scale_original, then transformation is not
applied. Default is log2. This parameter is ignored if the data is &quot;seqData&quot;.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_normalization_fun">normalization_fun</code></td>
<td>
<p>A character string indicating the pmartR
normalization function to use on the data, if is_normalized is FALSE.
Acceptable choices are 'global', 'loess', and 'quantile'. This parameter is
ignored if the data is &quot;seqData&quot;.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_normalization_params">normalization_params</code></td>
<td>
<p>A vector or list where the normalization
parameters are the names, and the parameter values are the list values. For
example, an acceptable entry for 'normalize_global' would be
list(&quot;subset_fn&quot; = &quot;all&quot;, &quot;norm_fn&quot; = &quot;median&quot;, &quot;apply_norm&quot; = TRUE,
&quot;backtransform&quot; = TRUE). This parameter is ignored if the data is &quot;seqData&quot;.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_is_normalized">is_normalized</code></td>
<td>
<p>A logical indicator of whether the data is already
normalized (and will therefore skip the normalization step). This parameter is
ignored if the data is &quot;seqData&quot;.</p>
</td></tr>
<tr><td><code id="as.trelliData.edata_+3A_force_normalization">force_normalization</code></td>
<td>
<p>A logical indicator to force normalization that is
not required for both isobaric protein and NMR data. This parameter is ignored
if the data is &quot;seqData.&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'trelliData' containing the raw data.
To be passed to trelliscope building functions.
</p>


<h3>Author(s)</h3>

<p>David Degnan, Daniel Claborne, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

###########################
## MS/NMR OMICS EXAMPLES ##
###########################

# Simple MS/NMR Example 
trelliData1 &lt;- as.trelliData.edata(e_data = pep_edata,
                                   edata_cname = "Peptide",
                                   omics_type = "pepData")
 
######################
## RNA-SEQ EXAMPLES ##  
######################
                                    
# RNA-seq Example
trelliData_seq1 &lt;- as.trelliData.edata(e_data = rnaseq_edata, 
                                      edata_cname = "Transcript",
                                      omics_type = "seqData")
                                  
          

</code></pre>

<hr>
<h2 id='bpquant'>Runs BP-Quant</h2><span id='topic+bpquant'></span>

<h3>Description</h3>

<p>Applies BP-Quant to a pepData object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bpquant(statRes, pepData, pi_not = 0.9, max_proteoforms = 5, parallel = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bpquant_+3A_statres">statRes</code></td>
<td>
<p>an object of the class 'statRes'</p>
</td></tr>
<tr><td><code id="bpquant_+3A_pepdata">pepData</code></td>
<td>
<p>an omicsData object of the class 'pepData' that includes the
e_meta component</p>
</td></tr>
<tr><td><code id="bpquant_+3A_pi_not">pi_not</code></td>
<td>
<p>numeric value between 0 and 1 indicating the background
probability/frequency of a zero signature</p>
</td></tr>
<tr><td><code id="bpquant_+3A_max_proteoforms">max_proteoforms</code></td>
<td>
<p>a numeric value corresponding to the maximum threshold
for the number of possible proteoforms</p>
</td></tr>
<tr><td><code id="bpquant_+3A_parallel">parallel</code></td>
<td>
<p>a logical indicator of whether the calculation will be
parallelized</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The result of this function can be used as one the <code>isoformRes</code>
input argument to <code><a href="#topic+protein_quant">protein_quant</a></code>. The <code>bpquant</code> function
itself operates as follows: The statRes object contains the signatures data
frame, the pepData object is used for its e_meta data frame. Next the
signatures data frame and e_meta are merged by their edata_cname (e.g.
peptide identifier) columns, this new data frame called protein_sig_data
will be input to bpquant_mod in a foreach statement. Foreach will
subset protein_sig_data for each unique protein and apply bpquant_mod to
each subset and store the results.
</p>


<h3>Value</h3>

<p>a list of data frames, one for each unique protein. The data frames
have three columns, a protein identifier, a peptide identifier, and a
&quot;ProteoformID&quot;. The class of this list is 'isoformRes'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

mypepData &lt;- group_designation(
  omicsData = pep_object,
  main_effects = c("Phenotype")
)
mypepData = edata_transform(omicsData = mypepData, data_scale = "log2")

imdanova_Filt &lt;- imdanova_filter(omicsData = mypepData)
mypepData &lt;- applyFilt(
  filter_object = imdanova_Filt,
  omicsData = mypepData,
  min_nonmiss_anova = 2
)

imd_anova_res &lt;- imd_anova(
  omicsData = mypepData,
  test_method = 'combined',
  pval_adjust_a_multcomp = 'bon',
  pval_adjust_g_multcomp = 'bon'
)

result = bpquant(statRes = imd_anova_res, pepData = mypepData)


</code></pre>

<hr>
<h2 id='bpquant_mod'>bpquant_mod function</h2><span id='topic+bpquant_mod'></span>

<h3>Description</h3>

<p>The function is written to take input from one protein at a time and requires
three inputs: protein_sig, pi_not and max_proteforms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bpquant_mod(protein_sig, pi_not, max_proteoforms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bpquant_mod_+3A_protein_sig">protein_sig</code></td>
<td>
<p>is a matrix or data.frame with p rows and n columns,
where p is the number of peptides mapped to the protein of interest and n
is the number of tests conducted to generate signatures made up of values
0, 1, and -1.</p>
</td></tr>
<tr><td><code id="bpquant_mod_+3A_pi_not">pi_not</code></td>
<td>
<p>is a numeric value between 0 and 1 indicating the background
probability/frequency of a zero signature.</p>
</td></tr>
<tr><td><code id="bpquant_mod_+3A_max_proteoforms">max_proteoforms</code></td>
<td>
<p>a numeric value, a maximum threshold for the number of
possible proteoforms.</p>
</td></tr>
</table>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;"> num_proteoforms </td><td style="text-align: left;"> the number of proteoforms as
identified by bpquant </td>
</tr>
<tr>
 <td style="text-align: left;"> unique_sigs </td><td style="text-align: left;"> matrix of unique signatures
observed  </td>
</tr>
<tr>
 <td style="text-align: left;"> proteoform_configs </td><td style="text-align: left;"> matrix of 0/1 values indicating
scenarios of proteoform absence/presence scenarios </td>
</tr>
<tr>
 <td style="text-align: left;"> post_prob </td><td style="text-align: left;"> vector
of posterior probabilities corresponding to each proteoform configuration in
"proteoform_configs" </td>
</tr>
<tr>
 <td style="text-align: left;"> peptide_idx </td><td style="text-align: left;"> vector of 0, 1, 2, . . . values
indicating which proteoform each peptide belongs to </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>a list of five items: num_proteoforms, unique_sigs,
proteoform_configs, post_prob and peptide_idx
</p>

<hr>
<h2 id='column_matches_exact'>Helper to find the names of columns of a data.frame that contain exactly all
the elements of an input column</h2><span id='topic+column_matches_exact'></span>

<h3>Description</h3>

<p>Helper to find the names of columns of a data.frame that contain exactly all
the elements of an input column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>column_matches_exact(df, col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="column_matches_exact_+3A_df">df</code></td>
<td>
<p>A data.frame whose columns we want to match to some query column.</p>
</td></tr>
<tr><td><code id="column_matches_exact_+3A_col">col</code></td>
<td>
<p>Vector of values which will be compared to a column in df.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of column names of df that contain exactly all the elements of
the input column
</p>

<hr>
<h2 id='combine_lipidData'>Combines two omicsData objects with identical sample information.</h2><span id='topic+combine_lipidData'></span>

<h3>Description</h3>

<p>Combines two omicsData objects with identical sample information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_lipidData(
  obj_1,
  obj_2,
  retain_groups = FALSE,
  retain_filters = FALSE,
  drop_duplicate_emeta = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_lipidData_+3A_obj_1">obj_1</code></td>
<td>
<p>omicsData object of the same supported type as obj_2, currently
&quot;lipidData&quot;. See details for more requirements.</p>
</td></tr>
<tr><td><code id="combine_lipidData_+3A_obj_2">obj_2</code></td>
<td>
<p>omicsData object of the same supported type as obj_1, currently
&quot;lipidData&quot;. See details for more requirements.</p>
</td></tr>
<tr><td><code id="combine_lipidData_+3A_retain_groups">retain_groups</code></td>
<td>
<p>logical indicator of whether to attempt to apply
existing group information to the new object. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="combine_lipidData_+3A_retain_filters">retain_filters</code></td>
<td>
<p>Whether to retain filter information in the new object
(defaults to FALSE).</p>
</td></tr>
<tr><td><code id="combine_lipidData_+3A_drop_duplicate_emeta">drop_duplicate_emeta</code></td>
<td>
<p>a logical indicator of whether duplicate molecule
identifiers in e_meta should be dropped</p>
</td></tr>
<tr><td><code id="combine_lipidData_+3A_...">...</code></td>
<td>
<p>Extra arguments, not one of 'omicsData', 'main_effects', or
'covariates' to be passed to 'pmartR::group_designation'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>General requirements:
</p>
<p>* sample names: These must be identical for both objects (column names of
e_data, and sample identifiers in f_data)
* data attributes: Objects must be on the same scale and both be either
normalized or unnormalized
* group designation: Objects must have the same grouping structure if
retain_groups = T
</p>


<h3>Value</h3>

<p>An object of the same type as the two input objects, with their
combined data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

obj_1 &lt;- lipid_neg_object
obj_2 &lt;- lipid_pos_object

# de-duplicate any duplicate edata identifiers
all(obj_2$e_data[, get_edata_cname(obj_2)] == obj_2$e_meta[, get_edata_cname(obj_2)])
obj_2$e_data[, get_edata_cname(obj_2)] &lt;- paste0("obj_2_", obj_2$e_data[, get_edata_cname(obj_2)])
obj_2$e_meta[, get_edata_cname(obj_2)] &lt;- obj_2$e_data[, get_edata_cname(obj_2)]

combine_object &lt;- combine_lipidData(obj_1 = obj_1, obj_2 = obj_2)

# preprocess and group the data and keep filters/grouping structure

obj_1 &lt;- edata_transform(omicsData = obj_1, data_scale = "log2")
obj_1 &lt;- normalize_global(omicsData = obj_1, subset_fn = "all",
                          norm_fn = "median", apply_norm = TRUE)
obj_2 &lt;- edata_transform(omicsData = obj_2, data_scale = "log2")
obj_2 &lt;- normalize_global(omicsData = obj_2, subset_fn = "all",
                          norm_fn = "median", apply_norm = TRUE)

obj_1 &lt;- group_designation(omicsData = obj_1, main_effects = "Virus")
obj_2 &lt;- group_designation(omicsData = obj_2, main_effects = "Virus")

obj_1 &lt;- applyFilt(filter_object = molecule_filter(omicsData = obj_1),
                   omicsData = obj_1, min_num = 2)
obj_2 &lt;- applyFilt(filter_object = cv_filter(omicsData = obj_2), obj_2, cv_thresh = 60)

combine_object_later &lt;- combine_lipidData(
                          obj_1 = obj_1,
                          obj_2 = obj_2,
                          retain_groups = TRUE,
                          retain_filters = TRUE
                        )

</code></pre>

<hr>
<h2 id='combine_techreps'>Combine technical replicates of an omicsData object</h2><span id='topic+combine_techreps'></span>

<h3>Description</h3>

<p>For each biomolecule, this function aggregates the technical replicates of
the biological samples using a specified aggregation method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_techreps(omicsData, combine_fn = NULL, bio_sample_names = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_techreps_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'lipidData', 'metabData', 'pepData',
'proData', 'nmrData', or 'seqData', created by
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>, respectively. The
parameter techrep_cnames must have been specified when creating this
object.</p>
</td></tr>
<tr><td><code id="combine_techreps_+3A_combine_fn">combine_fn</code></td>
<td>
<p>a character string specifying the function used to
aggregate across technical replicates. Currently supported functions are
'sum' and 'mean'. Defaults to 'sum' for seqData and 'mean' for all other
omicsData.</p>
</td></tr>
<tr><td><code id="combine_techreps_+3A_bio_sample_names">bio_sample_names</code></td>
<td>
<p>a character string specifying the column in
<code>f_data</code> which contains names by which to label aggregated samples in
<code>omicsData$e_data</code> and <code>omicsData$f_data</code> OR a character vector
with number of elements equal to the number of biological samples. If a
column name is specified, it should have a one-to-one correspondence with
the technical replicate ID column in <code>f_data</code>. Defaults to NULL, in
which case default names are used according to the technical replicate ID
column, which was specified at data object creation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Loss of information after aggregation </p>

<table>
<tr>
 <td style="text-align: left;"> f_data: </td><td style="text-align: left;">
  If there are columns in f_data that have more than 1 value per biological
  sample, then for each biological sample, only the first value in that
  column will be retained. Technical replicate specific information will be
  lost.</td>
</tr>
<tr>
 <td style="text-align: left;"> group information: </td><td style="text-align: left;"> If a grouping structure has been set
  using a main effect from f_data that has more than 1 level within any given
  biological sample, that grouping structure will be removed. Call
  <code>group_designation</code> again on the aggregated data to assign a grouping
  structure. </td>
</tr>
<tr>
 <td style="text-align: left;"> sample names: </td><td style="text-align: left;"> Identifiers for each biological sample
  will replace the identifiers for technical replicates as column names in
  e_data as well as the identifier column <code>attr(omicsData,
  'fdata_cname')</code> in f_data. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>An object with the same class as omicsData that has been aggregated
to the biological sample level
</p>


<h3>Author(s)</h3>

<p>Daniel Claborne
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

pep_object_averaged &lt;- combine_techreps(omicsData = pep_techrep_object)

</code></pre>

<hr>
<h2 id='complete_mols'>Identify biomolecules with no missing values across samples</h2><span id='topic+complete_mols'></span>

<h3>Description</h3>

<p>Selects biomolecules that have complete rows in e_data, equivalent to 'ppp'
with proportion = 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>complete_mols(e_data, edata_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="complete_mols_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, proteins, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="complete_mols_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector containing the biomolecules with no missing values
across all samples.
</p>

<hr>
<h2 id='cor_result'>Compute Correlation matrix of biomolecule data</h2><span id='topic+cor_result'></span>

<h3>Description</h3>

<p>This function returns an object of class corRes (correlation Result)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor_result(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cor_result_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'lipidData', 'metabData', 'pepData',
'proData', 'nmrData', or 'seqData', created by
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The pairwise correlations between samples are calculated based on
biomolecules that are observed in both samples. For seqData objects,
Spearman correlation is used. For all other data types, Pearson correlation
is used and data must be log transformed. See <code><a href="stats.html#topic+cor">cor</a></code> for further details.
</p>


<h3>Value</h3>

<p>An <code class="reqn">n \times n</code> matrix of class corRes giving the correlation
between samples.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton, Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+edata_transform">edata_transform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
my_correlation &lt;- cor_result(omicsData = mymetab)


myseq_correlation &lt;- cor_result(omicsData = rnaseq_object)


</code></pre>

<hr>
<h2 id='create_comparisonDF'>Returns data frame with comparisons to be made</h2><span id='topic+create_comparisonDF'></span>

<h3>Description</h3>

<p>The method creates a data frame containing the comparisons to be made when
performing differential statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_comparisonDF(comp_type, omicsData, control_group = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_comparisonDF_+3A_comp_type">comp_type</code></td>
<td>
<p>string specifying either &quot;control&quot; or
&quot;pairwise&quot;. Specifying &quot;control&quot; indicates that all other groups are
to be compared to a single control group. Specifying &quot;pairwise&quot;
indicates that all pairwise comparisons are to be made.</p>
</td></tr>
<tr><td><code id="create_comparisonDF_+3A_omicsdata">omicsData</code></td>
<td>
<p>A pmartR data object of any class, which has a 'group_df'
attribute created by the 'group_designation()' function</p>
</td></tr>
<tr><td><code id="create_comparisonDF_+3A_control_group">control_group</code></td>
<td>
<p>string indicating the group to use for the control
group. Only required when comp_type=&quot;control&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes in the omicsData and type of comparison, and
returns a data frame where each row corresponds to a comparison of
interest.
</p>


<h3>Value</h3>

<p>data frame with columns for Test and Control. Each row corresponds to
a comparison of interest.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+group_designation">group_designation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- group_designation(omicsData = metab_object, main_effects = "Phenotype")
create_comparisonDF(comp_type = "pairwise", omicsData = mymetab)

create_comparisonDF(comp_type = "control", omicsData = mymetab, control_group = "Phenotype1")

</code></pre>

<hr>
<h2 id='custom_filter'>Custom Filter Object</h2><span id='topic+custom_filter'></span>

<h3>Description</h3>

<p>This function creates a customFilt S3 object based on user-specified items to
filter out of the dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_filter(
  omicsData,
  e_data_remove = NULL,
  f_data_remove = NULL,
  e_meta_remove = NULL,
  e_data_keep = NULL,
  f_data_keep = NULL,
  e_meta_keep = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', or 'seqData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>,
<code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="custom_filter_+3A_e_data_remove">e_data_remove</code></td>
<td>
<p>character vector specifying the names of the e_data
identifiers to remove from the data. This argument can only be specified
with other 'remove' arguments.</p>
</td></tr>
<tr><td><code id="custom_filter_+3A_f_data_remove">f_data_remove</code></td>
<td>
<p>character vector specifying the names of f_data
identifiers to remove from the data. This argument can only be specified
with other 'remove' arguments.</p>
</td></tr>
<tr><td><code id="custom_filter_+3A_e_meta_remove">e_meta_remove</code></td>
<td>
<p>character vector specifying the names of the e_meta
identifiers to remove from the data. This argument can only be specified
with other 'remove' arguments.</p>
</td></tr>
<tr><td><code id="custom_filter_+3A_e_data_keep">e_data_keep</code></td>
<td>
<p>character vector specifying the names of the e_data
identifiers to keep from the data. This argument can only be specified with
other 'keep' arguments.</p>
</td></tr>
<tr><td><code id="custom_filter_+3A_f_data_keep">f_data_keep</code></td>
<td>
<p>character vector specifying the names of f_data
identifiers to keep from the data. This argument can only be specified with
other 'keep' arguments.</p>
</td></tr>
<tr><td><code id="custom_filter_+3A_e_meta_keep">e_meta_keep</code></td>
<td>
<p>character vector specifying the names of the e_meta
identifiers to keep from the data. This argument can only be specified with
other 'keep' arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An S3 object of class 'customFilt', which is a list with 3 elements
for e_data, f_data, and e_meta, specifying which entries should be either
kept or removed
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
to_filter &lt;- custom_filter(omicsData = metab_object, 
                           e_data_remove = "fumaric acid",
                           f_data_remove = "Sample_1_Phenotype2_B")
summary(to_filter)

to_filter2 &lt;- custom_filter(omicsData = metab_object, 
                            f_data_keep = metab_object$f_data$SampleID[1:10])
summary(to_filter2)

</code></pre>

<hr>
<h2 id='custom_sampnames'>Creates custom sample names to be used in plots</h2><span id='topic+custom_sampnames'></span>

<h3>Description</h3>

<p>This helper function creates custom sample names for plot data object
functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_sampnames(
  omicsData,
  firstn = NULL,
  from = NULL,
  to = NULL,
  delim = NULL,
  components = NULL,
  pattern = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_sampnames_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData',
'metabData', 'lipidData', 'nmrData', or 'seqData', created by <code>as.pepData</code>,
<code>as.proData</code>, <code>as.metabData</code>, <code>as.lipidData</code>, <code>as.nmrData</code>,
or <code>as.seqData</code>, respectively.</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_firstn">firstn</code></td>
<td>
<p>an integer specifying the first n characters to keep as the
sample name. This argument is optional.</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_from">from</code></td>
<td>
<p>an integer specifying the start of the range of characters to
keep as the sample name. This argument is optional. If this argument is
specified, 'to' must also be specified.</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_to">to</code></td>
<td>
<p>an integer specifying the end of the range of characters to keep as
the sample name. This argument is optional. If this argument is specified,
'from' must also be specified.</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_delim">delim</code></td>
<td>
<p>character delimiter by which to separate sample name components.
This argument is optional. If this argument is specified, 'components' must
also be specified.</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_components">components</code></td>
<td>
<p>integer vector specifying which components separated by the
delimiter should be kept as the custom sample name. This argument is
optional. If this argument is specified, 'delim' must also be specified.</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_pattern">pattern</code></td>
<td>
<p>character string specifying the regex pattern to use to
extract substrings from the sample names</p>
</td></tr>
<tr><td><code id="custom_sampnames_+3A_...">...</code></td>
<td>
<p>extra arguments passed to regexpr if pattern is specified</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to create custom (and shorter) sample
names to be used when plotting so that axis labels are not so long that
they interfere with the graph itself. To use the custom sample names when
plotting, specify the optional argument 'use_VizSampNames = TRUE'.
</p>


<h3>Value</h3>

<p>Object of same class as omicsData, with added column in f_data named
'VizSampNames'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

mypep &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")
plot(mypep)

# specify new names using firstn argument
results &lt;- custom_sampnames(omicsData = mypep, firstn = 9)
plot(results, use_VizSampNames = TRUE)

# specify new names using from and to arguments
results &lt;- custom_sampnames(omicsData = mypep, from = 1, to = 9)
plot(results, use_VizSampNames = TRUE)

# specify new names using delim and components arguments
results &lt;- custom_sampnames(omicsData = mypep, delim = "_", components = c(1, 2))
plot(results, use_VizSampNames = TRUE)

## specify new names using pattern arguments (regex)

# match everything after "Sample_"
pattern1 &lt;- "[0-9]+_[0-9A-Za-z]+_[A-Z]"

results &lt;- custom_sampnames(omicsData = mypep, pattern = pattern1)
plot(results, use_VizSampNames = TRUE)

# match "Sample_" and the number after it
pattern2 &lt;- "^Sample_[0-9]+"

results &lt;- custom_sampnames(omicsData = mypep, pattern = pattern2)
plot(results, use_VizSampNames = TRUE)

</code></pre>

<hr>
<h2 id='cv_filter'>Pooled Coefficient of Variation (CV) Filter Object</h2><span id='topic+cv_filter'></span>

<h3>Description</h3>

<p>A pooled CV is calculated for each biomolecule.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv_filter(omicsData, use_groups = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData' created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively. Note,
if <code><a href="#topic+group_designation">group_designation</a></code> has not been run, the CV is calculated
based on all samples for each biomolecule.</p>
</td></tr>
<tr><td><code id="cv_filter_+3A_use_groups">use_groups</code></td>
<td>
<p>logical indicator for whether to utilize group information
from <code><a href="#topic+group_designation">group_designation</a></code> when calculating the CV. Defaults to
TRUE. If use_groups is set to TRUE but <code><a href="#topic+group_designation">group_designation</a></code> has
not been run on the omicsData object, use_groups will be treated as FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each biomolecule, the CV of each group is calculated as the
standard deviation divided by the mean, excluding missing values. A pooled
CV estimate is then calculated based on the methods of Ahmed (1995). Any
groups consisting of a single sample are excluded from the CV calculation,
and thus, from the cv_filter result. If group_designation has not been run
on the omicsData object, all samples are considered to belong to the same
group.
</p>


<h3>Value</h3>

<p>An S3 object of class 'cvFilt' giving the pooled CV for each
biomolecule and additional information used for plotting a data.frame with a
column giving the biomolecule name and a column giving the pooled CV value.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>References</h3>

<p>Ahmed, S.E. (1995). <em>A pooling methodology for coefficient
of variation</em>. The Indian Journal of Statistics. 57: 57-75.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- group_designation(omicsData = pep_object, 
                           main_effects = "Phenotype")
to_filter &lt;- cv_filter(omicsData = mypep, use_groups = TRUE)
summary(to_filter, cv_threshold = 30)

</code></pre>

<hr>
<h2 id='DESeq2_wrapper'>Wrapper for DESeq2 workflow</h2><span id='topic+DESeq2_wrapper'></span>

<h3>Description</h3>

<p>For generating statistics for 'seqData' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DESeq2_wrapper(
  omicsData,
  test = "Wald",
  p_adjust = "BH",
  comparisons = NULL,
  p_cutoff = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DESeq2_wrapper_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of type 'seqData', created by <code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
<tr><td><code id="DESeq2_wrapper_+3A_test">test</code></td>
<td>
<p>either &quot;Wald&quot; or &quot;LRT&quot;, which will then use either Wald
significance tests, or the likelihood ratio test on the difference in
deviance between a full and reduced model formula</p>
</td></tr>
<tr><td><code id="DESeq2_wrapper_+3A_p_adjust">p_adjust</code></td>
<td>
<p>Character string for p-value correction method, refer to
?p.adjust() for valid options. Defaults to &quot;BH&quot; (Benjamini &amp; Hochberg)</p>
</td></tr>
<tr><td><code id="DESeq2_wrapper_+3A_comparisons">comparisons</code></td>
<td>
<p>'data.frame' with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="DESeq2_wrapper_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>Numeric value between 0 and 1 for setting p-value
significance threshold</p>
</td></tr>
<tr><td><code id="DESeq2_wrapper_+3A_...">...</code></td>
<td>
<p>additional arguments passed to function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Runs default DESeq workflow. Defaults to Wald test, no independent
filtering, and running in parallel. Additional arguments can be passed for
use in the function, refer to DESeq() and results() in DESeq2 package.
Requires 'survival' package to run.
</p>
<p>Flags (signatures) -	Indicator of statistical significance for computed
test. Zeros indicate no significance, while +/- 1 indicates direction of
significance.
</p>


<h3>Value</h3>

<p>statRes object
</p>


<h3>References</h3>

<p>Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and
dispersion for RNA-seq data with DESeq2 Genome Biology 15(12):550 (2014)
</p>

<hr>
<h2 id='diffexp_seq'>Differential Expression for seqData</h2><span id='topic+diffexp_seq'></span>

<h3>Description</h3>

<p>Performs statistical analysis for differential expression of seqData objects,
using methods from one of: edgeR, DESeq2, or limma-voom
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diffexp_seq(
  omicsData,
  method = "edgeR",
  p_adjust = "BH",
  comparisons = NULL,
  p_cutoff = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diffexp_seq_+3A_omicsdata">omicsData</code></td>
<td>
<p>object of type 'seqData' created by
<code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
<tr><td><code id="diffexp_seq_+3A_method">method</code></td>
<td>
<p>character string of length one specifying which wrapper to
use. Can be 'edgeR', 'DESeq2', or 'voom'</p>
</td></tr>
<tr><td><code id="diffexp_seq_+3A_p_adjust">p_adjust</code></td>
<td>
<p>character string for p-value correction method, refer to
?p.adjust() for valid options. Defaults to &quot;BH&quot; (Benjamini &amp; Hochberg).</p>
</td></tr>
<tr><td><code id="diffexp_seq_+3A_comparisons">comparisons</code></td>
<td>
<p>data frame with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="diffexp_seq_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>numeric value between 0 and 1 for setting p-value
significance threshold</p>
</td></tr>
<tr><td><code id="diffexp_seq_+3A_...">...</code></td>
<td>
<p>additional arguments passed to methods functions. Note, formatting
option changes will interfere with wrapping functionality.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Runs default differential expression workflows.
</p>
<p>Flags (signatures) -	Indicator of statistical significance.
Zeroes indicate no significance, while +/- 1 indicates direction of
significance.
</p>
<p>Method &quot;edgeR&quot; - Runs default edgeR workflow with empirical Bayes
quasi-likelihood F-tests. Additional arguments can be passed for use in the
function. Refer to calcNormFactors() and glmQLFit() in edgeR package.
Requires the 'edgeR' and 'limma' packages to run.
</p>
<p>Method &quot;DESeq2&quot; - Runs default DESeq workflow. Defaults to Wald test, no
independent filtering, and running in parallel. Additional arguments can be
passed for use in the function. Refer to DESeq() and results() in DESeq2
package. Requires 'survival' package to run.
</p>
<p>Method &quot;voom&quot; - Runs default limma-voom workflow using empirical Bayes
moderated t-statistics. Additional arguments can be passed for use in the
function. Refer to calcNormFactors() in edgeR package.
Requires the 'edgeR' and 'limma' packages to run.
</p>


<h3>Value</h3>

<p>object of class statRes
</p>


<h3>References</h3>

<p>Robinson MD, McCarthy DJ, Smyth GK (2010). edgeR: a Bioconductor package
for differential expression analysis of digital gene expression data.
Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.
</p>
<p>Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and
dispersion for RNA-seq data with DESeq2 Genome Biology 15(12):550 (2014)
</p>
<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth,
G.K. (2015). limma powers differential expression analyses for
RNA-sequencing and microarray studies. Nucleic Acids Research 43(7), e47.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)
myseqData &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
edger_results &lt;- diffexp_seq(omicsData = myseqData, method = "edgeR")
deseq_results &lt;- diffexp_seq(omicsData = myseqData, method = "DESeq2")
voom_results &lt;- diffexp_seq(omicsData = myseqData, method = "voom")


</code></pre>

<hr>
<h2 id='dim_reduction'>Reduce Dimension of Data for Exploratory Data Analysis</h2><span id='topic+dim_reduction'></span>

<h3>Description</h3>

<p>For data types other than seqData, this function calculates principal
components using projection pursuit estimation, which implements an
expectation-maximization (EM) estimation algorithm when data is missing. For
seqData counts, a generalized version of principal components analysis for
non-normally distributed data is calculated under the assumption of a
negative binomial distribution with global dispersion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dim_reduction(omicsData, k = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim_reduction_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepdata', 'prodata', 'metabData',
'lipidData', 'nmrData', or 'seqData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or
<code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="dim_reduction_+3A_k">k</code></td>
<td>
<p>integer number of principal components to return. Defaults to 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Any biomolecules seen in only one sample or with a variance less
than 1E-6 across all samples are not included in the PCA calculations. This
function leverages code from <code><a href="pcaMethods.html#topic+pca">pca</a></code> and
<code><a href="glmpca.html#topic+glmpca">glmpca</a></code> .
</p>


<h3>Value</h3>

<p>a data.frame with first <code>k</code> principal component scores, sample
identifiers, and group membership for each sample (if group designation was
previously run on the data). The object is of class dimRes (dimension
reduction Result).
</p>


<h3>References</h3>

<p>Redestig H, Stacklies W, Scholz M, Selbig J, &amp; Walther
D (2007). <em>pcaMethods - a bioconductor package providing PCA methods
for incomplete data</em>. Bioinformatics. 23(9): 1164-7.
</p>
<p>Townes FW, Hicks SC, Aryee MJ, Irizarry RA (2019). <em>Feature selection
and dimension reduction for single-cell RNA-seq based on a multinomial
model.</em> Genome Biol. 20, 116.
</p>
<p>Huang H, Wang Y, Rudin C, Browne EP (2022). <em>Towards a comprehensive
evaluation of dimension reduction methods for transcriptomic data
visualization.</em> Communications Biology 5, 719.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

mylipid &lt;- edata_transform(omicsData = lipid_neg_object, data_scale = "log2")
mylipid &lt;- group_designation(omicsData = mylipid, main_effects = "Virus")
pca_lipids &lt;- dim_reduction(omicsData = mylipid)


myseq &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
pca_seq &lt;- dim_reduction(omicsData = myseq)


</code></pre>

<hr>
<h2 id='dispersion_est'>Diagnostic plot for seqData</h2><span id='topic+dispersion_est'></span>

<h3>Description</h3>

<p>For generating statistics for 'seqData' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dispersion_est(
  omicsData,
  method,
  interactive = FALSE,
  x_lab = NULL,
  x_lab_size = 11,
  x_lab_angle = NULL,
  y_lab = NULL,
  y_lab_size = 11,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  bw_theme = TRUE,
  palette = NULL,
  point_size = 0.2,
  custom_theme = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dispersion_est_+3A_omicsdata">omicsData</code></td>
<td>
<p>seqData object used to terst dispersions</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_method">method</code></td>
<td>
<p>either &quot;DESeq2&quot;, &quot;edgeR&quot;, or &quot;voom&quot; for testing dispersion</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_interactive">interactive</code></td>
<td>
<p>Logical. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_x_lab">x_lab</code></td>
<td>
<p>A character string specifying the x-axis label when the metric
argument is NULL. The default is NULL in which case the x-axis label will
be &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>An integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>An integer value indicating the angle of x-axis labels.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_y_lab">y_lab</code></td>
<td>
<p>A character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>An integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_title_lab">title_lab</code></td>
<td>
<p>A character string specifying the plot title when the
<code>metric</code> argument is NULL.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>An integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_legend_lab">legend_lab</code></td>
<td>
<p>A character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_legend_position">legend_position</code></td>
<td>
<p>A character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, or &quot;bottom&quot;. The default is
&quot;right&quot;.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_bw_theme">bw_theme</code></td>
<td>
<p>Logical. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_palette">palette</code></td>
<td>
<p>A character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_point_size">point_size</code></td>
<td>
<p>An integer specifying the size of the points. The default
is 0.2.</p>
</td></tr>
<tr><td><code id="dispersion_est_+3A_custom_theme">custom_theme</code></td>
<td>
<p>a ggplot 'theme' object to be applied to non-interactive
plots, or those converted by plotly::ggplotly().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DESeq2 option requires package &quot;survival&quot; to be available.
</p>


<h3>Value</h3>

<p>plot result
</p>


<h3>References</h3>

<p>Robinson MD, McCarthy DJ, Smyth GK (2010). edgeR: a Bioconductor package
for differential expression analysis of digital gene expression data.
Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.
</p>
<p>Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and
dispersion for RNA-seq data with DESeq2 Genome Biology 15(12):550 (2014)
</p>
<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth,
G.K. (2015). limma powers differential expression analyses for
RNA-sequencing and microarray studies. Nucleic Acids Research 43(7), e47.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)
myseqData &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
dispersion_est(omicsData = myseqData, method = "edgeR")
dispersion_est(omicsData = myseqData, method = "DESeq2")
dispersion_est(omicsData = myseqData, method = "voom")


</code></pre>

<hr>
<h2 id='edata_replace'>Replace Values Equal to x with y</h2><span id='topic+edata_replace'></span>

<h3>Description</h3>

<p>This function finds all values of x in the e_data element of omicsData and
replaces them with y
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edata_replace(omicsData, x, y, threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edata_replace_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData' created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="edata_replace_+3A_x">x</code></td>
<td>
<p>value to be replaced, usually numeric or NA</p>
</td></tr>
<tr><td><code id="edata_replace_+3A_y">y</code></td>
<td>
<p>replacement value, usually numeric or NA</p>
</td></tr>
<tr><td><code id="edata_replace_+3A_threshold">threshold</code></td>
<td>
<p>Positive numeric value.  Observed values below this
threshold will be replaced by 'y' (in addition to all 'x' values).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is often used to replace any 0 values in peptide,
protein, metabolite, or lipid data with NA's. For omicsData on the
abundance scale, when the omicsData object is created, any 0's in e_data
are automatically converted to NA's. For omicsData on the count scale (e.g.
seqData objects), when the omicsData object is created, any NA's in e_data
are automatically converted to 0's.
</p>


<h3>Value</h3>

<p>data object of the same class as omicsData
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_replace(omicsData = metab_object, x = 0, y = NA)

</code></pre>

<hr>
<h2 id='edata_summary'>Creates a list of six Data Frames, one for each summarizing metric</h2><span id='topic+edata_summary'></span>

<h3>Description</h3>

<p>This function takes in an omicsData object and returns a summary of the
e_data component. The six summarizing metrics include the mean, standard
deviation, median, percent observed, minimum, and maximum.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edata_summary(omicsData, by = "sample", groupvar = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edata_summary_+3A_omicsdata">omicsData</code></td>
<td>
<p>object of the class 'lipidData', 'metabData', 'pepData',
'proData', or 'nmrData' created by <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="edata_summary_+3A_by">by</code></td>
<td>
<p>character string indicating whether summarizing metrics will be
applied by 'sample' or by 'molecule'. Defaults to 'sample'.</p>
</td></tr>
<tr><td><code id="edata_summary_+3A_groupvar">groupvar</code></td>
<td>
<p>a character vector with no more than two variable names that
should be used to determine group membership of samples. The variable name
must match a column name from <code>f_data</code>. Defaults to NULL, in which
case group_DF attribute will be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If groupvar is NULL and group_designation has not been applied to
omicsData, then the metrics will be applied to each column of e_data (when
by = 'sample) or to each row of e_data (when by = 'molecule'). When
groupvar is provided, it must match a column name from <code>f_data</code>, this
column of f_data is used to group e_data in order to apply the metrics.
</p>


<h3>Value</h3>

<p>A list of six data frames, of class 'dataRes' (data Result), which
are the results of applying the metrics (mean, standard deviation, median,
percent observed, minimum and maximum) to omicsData$e_data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

mylipid &lt;- edata_transform(omicsData = lipid_pos_object, data_scale = "log2")
mylipid &lt;- group_designation(omicsData = mylipid, main_effects = "Virus")
result &lt;- edata_summary(omicsData = mylipid, by = "sample", groupvar = NULL)

</code></pre>

<hr>
<h2 id='edata_transform'>Apply a Transformation to the Data</h2><span id='topic+edata_transform'></span>

<h3>Description</h3>

<p>This function applies a transformation to the e_data element of omicsData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edata_transform(omicsData, data_scale)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edata_transform_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData', created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>, or
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="edata_transform_+3A_data_scale">data_scale</code></td>
<td>
<p>a character string indicating the type of transformation to
be applied to the data. Valid values for 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData': 'log2', 'log', 'log10', or 'abundance'. A value
of 'abundance' indicates the data has previously undergone one of the log
transformations and should be transformed back to raw values with no
transformation applied.
</p>
<p>Valid values for 'seqData': 'upper', 'median',
'lcpm'. For 'seqData', 'lcpm' transforms by log2 counts per million,
'upper' transforms by the upper quartile of non-zero counts, and 'median'
transforms by the median of non-zero counts.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For all but seqData, this function is intended to be used before
analysis of the data begins, and data are typically analyzed on a log
scale. This function is not applicable to seqData objects, as any
transformations needed e.g. to allow more meaningful visualization of
seqData objects are performed within the pertinent functions.
</p>


<h3>Value</h3>

<p>data object of the same class as omicsData
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton, Natalie Heller
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
attr(mymetab, "data_info")$data_scale

</code></pre>

<hr>
<h2 id='edgeR_wrapper'>Wrapper for edgeR workflow</h2><span id='topic+edgeR_wrapper'></span>

<h3>Description</h3>

<p>For generating statistics for 'seqData' objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edgeR_wrapper(
  omicsData,
  p_adjust = "BH",
  comparisons = NULL,
  p_cutoff = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edgeR_wrapper_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of type 'seqData', created by
<code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
<tr><td><code id="edgeR_wrapper_+3A_p_adjust">p_adjust</code></td>
<td>
<p>Character string for p-value correction method, refer to
?p.adjust() for valid options. Defaults to &quot;BH&quot; (Benjamini &amp; Hochberg).</p>
</td></tr>
<tr><td><code id="edgeR_wrapper_+3A_comparisons">comparisons</code></td>
<td>
<p>'data.frame' with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="edgeR_wrapper_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>Numeric value between 0 and 1 for setting p-value
significance threshold</p>
</td></tr>
<tr><td><code id="edgeR_wrapper_+3A_...">...</code></td>
<td>
<p>additional arguments passed to methods functions. Note, formatting
option changes will interfere with wrapping functionality.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Requires the 'edgeR' and 'limma' packages. Runs default edgeR workflow with 
empirical Bayes quasi-likelihood F-tests. Additional arguments can be
passed for use in the function, refer to calcNormFactors() and glmQLFit()
in edgeR package.
</p>
<p>Flags (signatures) -	Indicator of statistical significance for computed test.
Zeroes indicate no significance, while +/- 1 indicates direction of significance.
</p>


<h3>Value</h3>

<p>statRes object
</p>


<h3>References</h3>

<p>Robinson MD, McCarthy DJ, Smyth GK (2010). edgeR: a Bioconductor package
for differential expression analysis of digital gene expression data.
Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.
</p>

<hr>
<h2 id='find_fmeta_cnames'>Find column names in f_meta for each object.  May return in two objects
sharing the same column name.</h2><span id='topic+find_fmeta_cnames'></span>

<h3>Description</h3>

<p>Find column names in f_meta for each object.  May return in two objects
sharing the same column name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_fmeta_cnames(res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_fmeta_cnames_+3A_res">res</code></td>
<td>
<p>The output of <code>fmeta_matches</code>, A list, the i-th element of
which contains a character vector of column name matches in f_meta for the
i-th omicsData object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector containing a column name in f_meta that the i-th
object matches to
</p>

<hr>
<h2 id='fit_surv'>Basic survival analysis function</h2><span id='topic+fit_surv'></span>

<h3>Description</h3>

<p>Implements overall survival analysis or progression-free survival analysis, depending upon the datatypes supplied to
surv_designation, and return the &quot;survfit&quot; object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_surv(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_surv_+3A_omicsdata">omicsData</code></td>
<td>
<p>A pmartR data object of any class, which has a 'group_df' attribute that is usually created by the 'group_designation()' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if fitted survival analysis object is returned
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(MSomicsSTAT)
library(OvarianPepdataBP)

# Basic analysis without covariates
attr(tcga_ovarian_pepdata_bp, "survDF") &lt;- list(t_death = "survival_time",
                                                ind_death = "vital_status")
sfit &lt;- fit_surv(tcga_ovarian_pepdata_bp)
plot(sfit)

# Add some covariate information
attr(tcga_ovarian_pepdata_bp, "survDF") &lt;- list(
  t_death = "survival_time",
  ind_death = "vital_status",
  covariates = "g__initial_pathologic_diagnosis_method_g1"
)
sfit &lt;- fit_surv(tcga_ovarian_pepdata_bp)
plot(sfit, col = c(1, 2))

## End(Not run)

</code></pre>

<hr>
<h2 id='fmeta_matches'>Check that the f_meta file contains a column aligned to each omicsData objects</h2><span id='topic+fmeta_matches'></span>

<h3>Description</h3>

<p>Check that the f_meta file contains a column aligned to each omicsData objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fmeta_matches(omicsData_objects, f_meta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fmeta_matches_+3A_omicsdata_objects">omicsData_objects</code></td>
<td>
<p>A list of omicsdata objects containing sample
information matching that in f_meta</p>
</td></tr>
<tr><td><code id="fmeta_matches_+3A_f_meta">f_meta</code></td>
<td>
<p>passed from <code>as.multiData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, each element of which contains a character vector of column
name matches in f_meta for each omicsData object.
</p>

<hr>
<h2 id='get_check_names'><b>DEPRECATED</b>: Fetch the check.names attribute</h2><span id='topic+get_check_names'></span>

<h3>Description</h3>

<p><em>Retrieves the value in check.names attribute from an omicsData object.</em>
<b>This function has been deprecated in favor of handling checking names
externally, and will always return FALSE.</b>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_check_names(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_check_names_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value indicating if the syntax of the column names in a
data frame should be checked. See <code><a href="base.html#topic+data.frame">data.frame</a></code> for
more details.
</p>

<hr>
<h2 id='get_comparisons'>Return comparisons of statRes object</h2><span id='topic+get_comparisons'></span>

<h3>Description</h3>

<p>This function returns comparisons from statRes or trelliData object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_comparisons(compObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_comparisons_+3A_compobj">compObj</code></td>
<td>
<p>is an object with the comparison attribute; specifically
objects of class 'statRes' and 'trelliData' objects derived from 'statRes'
objects in <code><a href="#topic+as.trelliData">as.trelliData</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a data frame with comparisons and their indices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

my_prodata = group_designation(
  omicsData = pro_object,
  main_effects = c("Phenotype")
)

imdanova_Filt = imdanova_filter(omicsData = my_prodata)

my_prodata = applyFilt(
  filter_object = imdanova_Filt,
  omicsData = my_prodata,
  min_nonmiss_anova = 2
)

imd_anova_res = imd_anova(
  omicsData = my_prodata,
  test_method = 'comb',
  pval_adjust_a_multcomp = 'bon',
  pval_adjust_g_multcomp = 'bon'
)

result = get_comparisons(imd_anova_res)

</code></pre>

<hr>
<h2 id='get_data_class'>Return data_class of statRes or trelliData object</h2><span id='topic+get_data_class'></span>

<h3>Description</h3>

<p>This function returns data_class attribute from statRes or trelliData object,
inherited from the omicsData used in <code><a href="#topic+imd_anova">imd_anova</a></code> or
<code><a href="#topic+as.trelliData">as.trelliData</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_data_class(dcObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_data_class_+3A_dcobj">dcObj</code></td>
<td>
<p>an object of class 'statRes' or 'trelliData'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the data_class attribute from a 'statRes' or 'trelliData'
object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

my_prodata = group_designation(
  omicsData = pro_object,
  main_effects = c("Phenotype")
)

imdanova_Filt = imdanova_filter(omicsData = my_prodata)

my_prodata = applyFilt(
  filter_object = imdanova_Filt,
  omicsData = my_prodata,
  min_nonmiss_anova = 2
)

imd_anova_res = imd_anova(
  omicsData = my_prodata,
  test_method = 'comb',
  pval_adjust_a_multcomp = 'bon',
  pval_adjust_g_multcomp = 'bon'
)

result = get_data_class(imd_anova_res)

</code></pre>

<hr>
<h2 id='get_data_info'>Fetch the data_info attribute</h2><span id='topic+get_data_info'></span>

<h3>Description</h3>

<p>Retrieves the values in the data_info attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_data_info(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_data_info_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing seven elements:
</p>

<ul>
<li><p> data_scale &ndash; A Character string indicating the scale of the
data in <code>e_data</code>.
</p>
</li>
<li><p> norm_info &ndash; A list containing a single element indicating
whether the data in <code>e_data</code> have been normalized.
</p>
</li>
<li><p> num_edata &ndash; The number of unique entries present in the
<code>edata_cname</code> column in <code>e_data</code>.
</p>
</li>
<li><p> num_miss_obs &ndash; An integer. The number of missing
observations in <code>e_data</code>.
</p>
</li>
<li><p> num_zero_obs &ndash; An integer. The number of zeros
in <code>e_data</code> for seqData objects.
</p>
</li>
<li><p> prop_missing &ndash; A number between 0 and 1. The proportion of
missing observations in <code>e_data</code>.
</p>
</li>
<li><p> num_samps &ndash; An integer indicating the number of samples or
columns (excluding the identifier column <code>edata_cname</code>) in
<code>e_data</code>.
</p>
</li>
<li><p> data_types &ndash; A character string describing the type of data
in <code>e_data</code>.
</p>
</li></ul>


<hr>
<h2 id='get_data_norm'>Fetch the normalization status of the data</h2><span id='topic+get_data_norm'></span>

<h3>Description</h3>

<p>This function returns the norm_info element of the data_info attribute
indicating whether the data have been normalized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_data_norm(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_data_norm_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', 'statRes', or 'trelliData', usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, <code><a href="#topic+imd_anova">imd_anova</a></code>, or
<code><a href="#topic+as.trelliData">as.trelliData</a></code> respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value indicating whether the data have been normalized.
</p>

<hr>
<h2 id='get_data_scale'>Fetch the current data scale</h2><span id='topic+get_data_scale'></span>

<h3>Description</h3>

<p>This function returns current data scale which may be different from the
original data scale (if <code>edata_transform</code> was used).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_data_scale(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_data_scale_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', 'statRes', or 'trelliData', usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, <code><a href="#topic+imd_anova">imd_anova</a></code>, or
<code><a href="#topic+as.trelliData">as.trelliData</a></code> respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character string describing data scale
</p>

<hr>
<h2 id='get_data_scale_orig'>Fetch the original data scale</h2><span id='topic+get_data_scale_orig'></span>

<h3>Description</h3>

<p>Retrieves the character string indicating the scale the data was originally
on when read into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_data_scale_orig(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_data_scale_orig_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>

<hr>
<h2 id='get_edata_cname'>Fetch the e_data column name</h2><span id='topic+get_edata_cname'></span>

<h3>Description</h3>

<p>This function returns the name of the column in e_data that contains the
biomolecule IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_edata_cname(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_edata_cname_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', 'statRes', or 'trelliData', usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, <code><a href="#topic+imd_anova">imd_anova</a></code>, or
<code><a href="#topic+as.trelliData">as.trelliData</a></code> respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character string describing e_data cname
</p>

<hr>
<h2 id='get_emeta_cname'>Fetch the e_meta column name</h2><span id='topic+get_emeta_cname'></span>

<h3>Description</h3>

<p>This function returns the name of the column in e_meta that contains the
mapping variable IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_emeta_cname(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_emeta_cname_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', 'statRes', or 'trelliData', usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, <code><a href="#topic+imd_anova">imd_anova</a></code>, or
<code><a href="#topic+as.trelliData">as.trelliData</a></code> respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character string describing e_meta cname
</p>

<hr>
<h2 id='get_fdata_cname'>Fetch the f_data column name</h2><span id='topic+get_fdata_cname'></span>

<h3>Description</h3>

<p>This function returns the name of the column in f_data that contains the
names of the samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fdata_cname(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fdata_cname_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', 'statRes', or 'trelliData', usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, <code><a href="#topic+imd_anova">imd_anova</a></code>, or
<code><a href="#topic+as.trelliData">as.trelliData</a></code> respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character string describing f_data cname
</p>

<hr>
<h2 id='get_filter_type'>Extracts the types of filters that have been applied. This function will be
used at the beginning of the applyFilt function to give a warning if the
same type of filter has already been applied.</h2><span id='topic+get_filter_type'></span>

<h3>Description</h3>

<p>Retrieves the values in the filters attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_filter_type(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_filter_type_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
seqData, or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of filters used on omicsData
</p>

<hr>
<h2 id='get_filters'>Fetch the filters attribute</h2><span id='topic+get_filters'></span>

<h3>Description</h3>

<p>Retrieves the values in the filters attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_filters(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_filters_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing filter class objects. Each element in this list
corresponds to a filter applied to the data. The filters will be
listed in the order they were applied. A filter object contains two
elements:
</p>

<ul>
<li><p> threshold &ndash; The threshold used to filter <code>e_data</code>. This
value depends on the type of filter applied.
</p>
</li>
<li><p> filtered &ndash; A vector containing the identifiers from the
<code>edata_cname</code> column that will be filtered.
</p>
</li>
<li><p> method &ndash; A character string indicating the type of method
used to filter. This only applies when imdanova_filter is used.
</p>
</li></ul>


<hr>
<h2 id='get_group_DF'>Fetch the group_DF attribute</h2><span id='topic+get_group_DF'></span>

<h3>Description</h3>

<p>Retrieves the values in the group_DF attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_group_DF(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_group_DF_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with columns for sample ID and group. If two main
effects are provided the original main effect levels for each sample
are returned as the third and fourth columns of the data frame.
Additionally, the covariates provided will be listed as attributes of
this data frame.
</p>

<hr>
<h2 id='get_group_formula'>Get formula for group design</h2><span id='topic+get_group_formula'></span>

<h3>Description</h3>

<p>For generating group design formulas and correctly ordered group data for
seqData statistical functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_group_formula(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_group_formula_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of type 'seqData', created by <code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements:
</p>

<ul>
<li><p> grouping_info: A data.frame with the grouping information used in
the statistical analysis
</p>
</li>
<li><p> formula_string: A character string with the formula used in the
statistical analysis
</p>
</li></ul>


<hr>
<h2 id='get_group_table'>Get group table</h2><span id='topic+get_group_table'></span>

<h3>Description</h3>

<p>This function returns a table with number of samples per group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_group_table(omicsObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_group_table_+3A_omicsobject">omicsObject</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', 'statRes', or 'trelliData', usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, <code><a href="#topic+imd_anova">imd_anova</a></code>, or
<code><a href="#topic+as.trelliData">as.trelliData</a></code> respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table containing number of samples per group
</p>

<hr>
<h2 id='get_isobaric_info'>Fetch the isobaric_info attribute</h2><span id='topic+get_isobaric_info'></span>

<h3>Description</h3>

<p>Retrieves the values in the isobaric_info attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_isobaric_info(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_isobaric_info_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following six elements:
</p>

<ul>
<li><p> exp_cname &ndash;
</p>
</li>
<li><p> channel_cname &ndash;
</p>
</li>
<li><p> refpool_channel &ndash;
</p>
</li>
<li><p> refpool_cname &ndash;
</p>
</li>
<li><p> refpool_notation &ndash;
</p>
</li>
<li><p> norm_info &ndash; A list containing a single logical element that
indicates whether the data have been normalized to a reference
pool.
</p>
</li></ul>


<hr>
<h2 id='get_isobaric_norm'>Fetch the isobaric normalization info</h2><span id='topic+get_isobaric_norm'></span>

<h3>Description</h3>

<p>This function returns the norm_info element of the isobaric_info attribute
which indicates if the data have been isobaric normalized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_isobaric_norm(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_isobaric_norm_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'isobaricpepData' or
'proData', usually created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value indicating whether the data have been isobaric
normalized.
</p>

<hr>
<h2 id='get_lsmeans'>Compute the least squares means from a prediction grid and estimated coefficients</h2><span id='topic+get_lsmeans'></span>

<h3>Description</h3>

<p>Compute the least squares means from a prediction grid and estimated coefficients
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lsmeans(data, xmatrix, pred_grid, Betas, continuous_covar_inds = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_lsmeans_+3A_data">data</code></td>
<td>
<p>The raw data from which the estimates were computed</p>
</td></tr>
<tr><td><code id="get_lsmeans_+3A_xmatrix">xmatrix</code></td>
<td>
<p>The design matrix from which the prediction grid was constructed</p>
</td></tr>
<tr><td><code id="get_lsmeans_+3A_pred_grid">pred_grid</code></td>
<td>
<p>The prediction grid, obtained from <code><a href="#topic+get_pred_grid">get_pred_grid</a></code></p>
</td></tr>
<tr><td><code id="get_lsmeans_+3A_betas">Betas</code></td>
<td>
<p>The estimated coefficients</p>
</td></tr>
<tr><td><code id="get_lsmeans_+3A_continuous_covar_inds">continuous_covar_inds</code></td>
<td>
<p>The column indices of xmatrix corresponding to continuous covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the least squares means
</p>


<h3>Author(s)</h3>

<p>Daniel Claborne
</p>

<hr>
<h2 id='get_meta_info'>Fetch the meta_info attribute</h2><span id='topic+get_meta_info'></span>

<h3>Description</h3>

<p>Retrieves the values in the meta_info attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_meta_info(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_meta_info_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li><p> meta_data &ndash; Logical. Indicates if the <code>e_meta</code> data
frame was provided.
</p>
</li>
<li><p> num_emeta &ndash; The number of unique entries present in the
<code>emeta_cname</code> column in <code>e_meta</code>.
</p>
</li></ul>


<hr>
<h2 id='get_nmr_info'>Fetch the nmr_info attribute</h2><span id='topic+get_nmr_info'></span>

<h3>Description</h3>

<p>Retrieves the values in the nmr_info attribute from an omicsData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_nmr_info(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_nmr_info_+3A_omicsdata">omicsData</code></td>
<td>
<p>An object of class pepData, proData, metabData, lipidData,
or nmrData.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following three elements:
</p>

<ul>
<li><p> metabolite_name &ndash;
</p>
</li>
<li><p> sample_property_cname &ndash;
</p>
</li>
<li><p> norm_info &ndash; A list containing two logical elements that
indicate i) whether the data have been normalized to a spiked in
metabolite or to a property taking sample-specific values and ii)
whether the data have been back transformed so the values are on a
similar scale to the raw values before normalization.
</p>
</li></ul>


<hr>
<h2 id='get_nmr_norm'>Fetch the NMR normalization info</h2><span id='topic+get_nmr_norm'></span>

<h3>Description</h3>

<p>This function returns the norm_info element of the nmr_info attribute which
indicates if the data have been NMR normalized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_nmr_norm(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_nmr_norm_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'nmrData'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value indicating whether the data have been NMR normalized.
</p>

<hr>
<h2 id='get_pred_grid'>Build the prediction grid to compute least squares means.</h2><span id='topic+get_pred_grid'></span>

<h3>Description</h3>

<p>Build the prediction grid to compute least squares means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pred_grid(
  group_df,
  main_effect_names,
  covariate_names = NULL,
  fspec = as.formula("~.")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pred_grid_+3A_group_df">group_df</code></td>
<td>
<p>A dataframe with the reserved 'Group' column, and columns for main effects and covariates.</p>
</td></tr>
<tr><td><code id="get_pred_grid_+3A_main_effect_names">main_effect_names</code></td>
<td>
<p>Character vector with the column names of the main effects in group_df.</p>
</td></tr>
<tr><td><code id="get_pred_grid_+3A_covariate_names">covariate_names</code></td>
<td>
<p>Character vector with the column names of the covariates in group_df.</p>
</td></tr>
<tr><td><code id="get_pred_grid_+3A_fspec">fspec</code></td>
<td>
<p>A formula specification to be passed to <code><a href="stats.html#topic+model.matrix">model.matrix</a></code> to construct the prediction grid in model matrix form.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of the prediction grid
</p>


<h3>Author(s)</h3>

<p>Daniel Claborne
</p>

<hr>
<h2 id='get_spans_params'>Gets the parameters for the highest ranked methods from spans.</h2><span id='topic+get_spans_params'></span>

<h3>Description</h3>

<p>Gets the parameters for the highest ranked methods from spans.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_spans_params(SPANSRes_obj, sort_by_nmols = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_spans_params_+3A_spansres_obj">SPANSRes_obj</code></td>
<td>
<p>an object of the class SPANSRes obtained by calling
<code>spans_procedure()</code></p>
</td></tr>
<tr><td><code id="get_spans_params_+3A_sort_by_nmols">sort_by_nmols</code></td>
<td>
<p>a logical indicator of whether to sort by number of
molecules used in the normalization (see <code><a href="#topic+spans_procedure">spans_procedure</a></code> for
info about the 'mols_used_in_norm' column)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of lists, where there are multiple sublists only if there were
ties for the top SPANS score.  Each sublist contains named elements for the
subset and normalization methods, and the parameters used for the subset
method. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

# data must be log transformed and grouped
myobject &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")
myobject &lt;- group_designation(omicsData = myobject, main_effects = "Phenotype")

spans_result &lt;- spans_procedure(omicsData = myobject)

# a list of the parameters for any normalization procedure with the best SPANS score
best_params &lt;- get_spans_params(spans_result)

# extract the arguments from the first list element
subset_fn = best_params[[1]]$subset_fn
norm_fn = best_params[[1]]$norm_fn
params = best_params[[1]]$params
if (is.null(params[[1]])) {
  params = NULL
}

# pass arguments to normalize global
norm_object &lt;- normalize_global(omicsData = myobject, subset_fn = subset_fn,
                                norm_fn = norm_fn, params = params)


</code></pre>

<hr>
<h2 id='group_comparison_anova'>Group comparisons for the anova test</h2><span id='topic+group_comparison_anova'></span>

<h3>Description</h3>

<p>Takes the results of anova_test() and returns group comparison p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_comparison_anova(
  data,
  groupData,
  comparisons,
  Xfull,
  Xred,
  anova_results_full,
  beta_to_mu_full,
  beta_to_mu_red
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_comparison_anova_+3A_data">data</code></td>
<td>
<p>The expression values without the id column</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_groupdata">groupData</code></td>
<td>
<p>data frame that assigns sample names to groups</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_comparisons">comparisons</code></td>
<td>
<p>dataframe that defines the comparsions of interest</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_xfull">Xfull</code></td>
<td>
<p>design matrix for the full model with interaction terms between the main effects</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_xred">Xred</code></td>
<td>
<p>design matrix for the reduced model with no interaction terms between the main effects</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_anova_results_full">anova_results_full</code></td>
<td>
<p>results of the <code>pmartR::anova_test()</code> function</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_beta_to_mu_full">beta_to_mu_full</code></td>
<td>
<p>matrix that maps the beta coefficients to the group means for the full model</p>
</td></tr>
<tr><td><code id="group_comparison_anova_+3A_beta_to_mu_red">beta_to_mu_red</code></td>
<td>
<p>matrix that maps the beta coefficients to the group means for the reduced model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing the p-values from the group comparisons.
</p>


<h3>Author(s)</h3>

<p>Bryan Stanfill, Daniel Claborne
</p>

<hr>
<h2 id='group_comparison_imd'>Group comparisons for the g-test</h2><span id='topic+group_comparison_imd'></span>

<h3>Description</h3>

<p>Takes the results of imd_test() and returns group comparison p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_comparison_imd(groupData, comparisons, observed, absent)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_comparison_imd_+3A_groupdata">groupData</code></td>
<td>
<p>data frame that assigns sample names to groups</p>
</td></tr>
<tr><td><code id="group_comparison_imd_+3A_comparisons">comparisons</code></td>
<td>
<p>dataframe that defiens the comparsions of interest</p>
</td></tr>
<tr><td><code id="group_comparison_imd_+3A_observed">observed</code></td>
<td>
<p>matrix of number of observed counts</p>
</td></tr>
<tr><td><code id="group_comparison_imd_+3A_absent">absent</code></td>
<td>
<p>matrix of number of observed counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing the p-values from the group comparisons.
</p>


<h3>Author(s)</h3>

<p>Bryan Stanfill
</p>

<hr>
<h2 id='group_designation'>Creates Attribute of omicsData Object for Group Membership</h2><span id='topic+group_designation'></span>

<h3>Description</h3>

<p>The method assigns each sample to a group, for use in future analyses, based on the
variable(s) specified as main effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_designation(
  omicsData,
  main_effects = NULL,
  covariates = NULL,
  cov_type = NULL,
  pair_id = NULL,
  pair_group = NULL,
  pair_denom = NULL,
  batch_id = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_designation_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'lipidData', 'metabData', 'pepData',
'proData', 'isobaricpepData', 'nmrData', or 'seqData', usually created by
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>,
respectively.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_main_effects">main_effects</code></td>
<td>
<p>a character vector with no more than two variable names
that should be used as main effects to determine group membership of
samples. The variable name must match a column name from <code>f_data</code>.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_covariates">covariates</code></td>
<td>
<p>a character vector of no more than two variable names that
should be used as covariates in downstream analyses. Covariates are
typically variables that a user wants to account for in the analysis but
quantifying/examining the effect of the variable is not of interest.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_cov_type">cov_type</code></td>
<td>
<p>An optional character vector (must be the same length as
<code>covariates</code> if used) indicating the class or type of each covariate.
For example, &quot;numeric&quot;, &quot;character&quot;, or &quot;factor&quot;. Partial matching (&quot;num&quot;
for &quot;numeric&quot;) is NOT used and the entire class/type must be typed out. If
the class of a covariate does not match the input to <code>cov_type</code> the
covariate will be coerced to that type. For example, if the covariate is a
numeric vector of 0s and 1s (indicating two categories) and the input to
<code>cov_type</code> is a class other than numeric this vector will be coerced
to a character vector. The default value is NULL. In this case the class of
the covariates is neither checked nor altered.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_pair_id">pair_id</code></td>
<td>
<p>A character string indicating the column in <code>f_data</code> that
contains the IDs for each pair. This string must match the column name
exactly.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_pair_group">pair_group</code></td>
<td>
<p>A character string specifying the column in <code>f_data</code>
that indicates which group each pair belongs to. This variable must contain
just two levels or values (e.g., &quot;before&quot; and &quot;after&quot;). Numeric values can
be used (e.g., 0 and 1). However, they will be converted to character
strings.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_pair_denom">pair_denom</code></td>
<td>
<p>A character string specifying which pair group is the
&quot;control&quot;. When taking the difference, the value for the control group will
be subtracted from the non-control group value.</p>
</td></tr>
<tr><td><code id="group_designation_+3A_batch_id">batch_id</code></td>
<td>
<p>an optional character vector of no more than one variable that
should be used as batch information for downstream analyses. Batch ID is
similar to covariates but unlike covariates it is specific to that of specific
batch effects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups are formed based on the levels of the main effect variables.
One or two main effect variables are allowed. In the case of two main
effect variables, groups are formed based on unique combinations of the
levels of the two main effect variables. Any samples with level NA for a
main effect variable will be removed from the data and will not be included
in the final group designation results. Groups with a single sample are
allowed, as is a single group.
</p>


<h3>Value</h3>

<p>An object of the same class as the input <code>omicsData</code> object -
the provided object with the samples filtered out, if any NAs were produced
in designating groups. An attribute 'group_DF', a data.frame with columns
for sample id and group, is added to the object. If two main effects are
provided the original main effect levels for each sample are returned as
the third and fourth columns of the data.frame. Additionally, the
covariates provided will be listed as attributes of this data.frame.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mylipid &lt;- group_designation(
  omicsData = lipid_pos_object,
  main_effects = "Virus"
)
attr(mylipid, "group_DF")

</code></pre>

<hr>
<h2 id='gtest_filter'>Identifies peptides to be filtered out in preparation for IMD-ANOVA.</h2><span id='topic+gtest_filter'></span>

<h3>Description</h3>

<p>The method identifies peptides, proteins, lipids, or metabolites to be
filtered specifically according to the G-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtest_filter(nonmiss_per_group, min_nonmiss_gtest, comparisons = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtest_filter_+3A_nonmiss_per_group">nonmiss_per_group</code></td>
<td>
<p>list created by <code><a href="#topic+nonmissing_per_group">nonmissing_per_group</a></code>.
The first element giving the total number of possible samples for each
group. The second element giving a data.frame with the first column giving
the biomolecule and the second through kth columns giving the number of
non-missing observations for each of the <code>k</code> groups.</p>
</td></tr>
<tr><td><code id="gtest_filter_+3A_min_nonmiss_gtest">min_nonmiss_gtest</code></td>
<td>
<p>the minimum number of non-missing peptide values
allowed in a minimum of one group. Default value is 3.</p>
</td></tr>
<tr><td><code id="gtest_filter_+3A_comparisons">comparisons</code></td>
<td>
<p>data.frame with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control. If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two methods are available for determining the peptides to be
filtered. The naive approach is based on <code>min.nonmiss.allowed</code>, and
looks for peptides that do not have at least <code>min.nonmiss.allowed</code>
values per group. The other approach also looks for peptides that do not
have at least a minimum number of values per group, but this minimum number
is determined using the G-test and a p-value threshold supplied by the
user. The G-test is a test of independence, used here to test the null
hypothesis of independence between the number of missing values across
groups.
</p>


<h3>Value</h3>

<p>filter.peps a character vector of the peptides to be filtered out
prior to the G-test or IMD-ANOVA
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>

<hr>
<h2 id='gtest_heatmap'>Plot a heatmap for the g-test results of imd-anova</h2><span id='topic+gtest_heatmap'></span>

<h3>Description</h3>

<p>Plots a heatmap showing bins for combinations of # of biomolecules present
across groups.  Bins are colored by the number of biomolecules falling into
each bin, and have indicators for significance by g-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtest_heatmap(
  volcano,
  pval_thresh,
  show_sig,
  interactive,
  color_low,
  color_high,
  plotly_layout,
  text_size,
  display_count,
  x_lab,
  y_lab,
  title_lab,
  legend_lab
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtest_heatmap_+3A_volcano">volcano</code></td>
<td>
<p>'data.frame' produced by pmartR:::<a href="#topic+make_volcano_plot_df">make_volcano_plot_df</a></p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_pval_thresh">pval_thresh</code></td>
<td>
<p>numeric value indicating the p-value threshold for
significance.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_show_sig">show_sig</code></td>
<td>
<p>Boolean whether to show the visual indicator that a certain
bin combination is significant by the g-test</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_interactive">interactive</code></td>
<td>
<p>passed from
<code><a href="#topic+plot.statRes">pmartR::plot.statRes()</a></code>. If T, will build
a plotly version of the plot.  Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_color_low">color_low</code></td>
<td>
<p>character string specifying the color of the gradient for
low count values.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_color_high">color_high</code></td>
<td>
<p>character string specifying the color of the gradient for
high count values.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_plotly_layout">plotly_layout</code></td>
<td>
<p>A list of arguments, not including the plot, to be
passed to plotly::layout if interactive = T.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_text_size">text_size</code></td>
<td>
<p>An integer specifying the size of the text (number of
non-missing values) within the plot. The default is 3.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the non-missing counts
will be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="gtest_heatmap_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'ggplot or plotly' object, depending on if <code>interactive</code> is set
to TRUE or FALSE respectively. A g-test heatmap
</p>

<hr>
<h2 id='imd_anova'>Test for a qualitative and quantitative difference between groups using IMD
and ANOVA, respectively</h2><span id='topic+imd_anova'></span>

<h3>Description</h3>

<p>This is the IMD-ANOVA test defined in Webb-Robertson et al. (2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imd_anova(
  omicsData,
  comparisons = NULL,
  test_method,
  pval_adjust_a_multcomp = "none",
  pval_adjust_g_multcomp = "none",
  pval_adjust_a_fdr = "none",
  pval_adjust_g_fdr = "none",
  pval_thresh = 0.05,
  equal_var = TRUE,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imd_anova_+3A_omicsdata">omicsData</code></td>
<td>
<p>pmartR data object of any class, which has a 'group_df'
attribute created by the 'group_designation()' function</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_comparisons">comparisons</code></td>
<td>
<p>data frame with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control. If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_test_method">test_method</code></td>
<td>
<p>character string specifying the filter method to use:
&quot;combined&quot;, &quot;gtest&quot;, or &quot;anova&quot;. Specifying &quot;combined&quot; implements both the gtest and
anova filters.</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_pval_adjust_a_multcomp">pval_adjust_a_multcomp</code></td>
<td>
<p>character string specifying the type of multiple
comparison adjustment to implement for ANOVA tests. Valid options include:
&quot;bonferroni&quot;, &quot;holm&quot;, &quot;tukey&quot;, and &quot;dunnett&quot;. The default is &quot;none&quot; which
corresponds to no p-value adjustment.</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_pval_adjust_g_multcomp">pval_adjust_g_multcomp</code></td>
<td>
<p>character string specifying the type of multiple
comparison adjustment to implement for G-test tests. Valid options include:
&quot;bonferroni&quot; and &quot;holm&quot;. The default is &quot;none&quot; which corresponds to no
p-value adjustment.</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_pval_adjust_a_fdr">pval_adjust_a_fdr</code></td>
<td>
<p>character string specifying the type of FDR
adjustment to implement for ANOVA tests. Valid options include:
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, and &quot;fdr&quot;. The default is &quot;none&quot; which corresponds
to no p-value adjustment.</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_pval_adjust_g_fdr">pval_adjust_g_fdr</code></td>
<td>
<p>character string specifying the type of FDR
adjustment to implement for G-test tests. Valid options include:
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, and &quot;fdr&quot;. The default is &quot;none&quot; which corresponds
to no p-value adjustment.</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_pval_thresh">pval_thresh</code></td>
<td>
<p>numeric p-value threshold, below or equal to which
biomolecules are considered differentially expressed. Defaults to 0.05</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_equal_var">equal_var</code></td>
<td>
<p>logical; should the variance across groups be assumed equal?</p>
</td></tr>
<tr><td><code id="imd_anova_+3A_parallel">parallel</code></td>
<td>
<p>logical value indicating whether or not to use a
&quot;doParallel&quot; loop when running the G-Test with covariates. Defaults to
TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'statRes', which is a data frame containing
columns (when relevant based on the test(s) performed) for: e_data cname,
group counts, group means, ANOVA p-values, IMD p-values, fold change
estimates on the same scale as the data (e.g. log2, log10, etc.), and fold
change significance flags (0 = not significant; +1 = significant and
positive fold change (ANOVA) or more observations in test group relative to
reference group (IMD); -1 = significant and negative fold change (ANOVA) or
fewer observations in test group relative to reference group (IMD))
</p>


<h3>Author(s)</h3>

<p>Bryan Stanfill, Kelly Stratton
</p>


<h3>References</h3>

<p>Webb-Robertson, Bobbie-Jo M., et al. &quot;Combined statistical
analyses of peptide intensities and peptide occurrences improves
identification of significant peptides from MS-based proteomics data.&quot;
Journal of proteome research 9.11 (2010): 5748-5756.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
# Transform the data
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")

# Group the data by condition
mymetab &lt;- group_designation(omicsData = mymetab, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = mymetab)
mymetab &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = mymetab, min_nonmiss_anova = 2)

# Implement IMD ANOVA and compute all pairwise comparisons 
# (i.e. leave the comparisons argument NULL), with FDR adjustment
anova_res &lt;- imd_anova(omicsData = mymetab, test_method = "anova",
                       pval_adjust_a_multcomp = "Holm", pval_adjust_a_fdr = "BY")
imd_res &lt;- imd_anova(omicsData = mymetab, test_method = "gtest",
                     pval_adjust_g_multcomp = "bon", pval_adjust_g_fdr = "BY")
imd_anova_res &lt;- imd_anova(omicsData = mymetab, test_method = "combined",
                           pval_adjust_a_fdr = "BY", pval_adjust_g_fdr = "BY")
imd_anova_res &lt;- imd_anova(omicsData = mymetab, test_method = "combined",
                           pval_adjust_a_multcomp = "bon", pval_adjust_g_multcomp = "bon",
                           pval_adjust_a_fdr = "BY", pval_adjust_g_fdr = "BY")

# Two main effects and a covariate
mymetab &lt;- group_designation(omicsData = mymetab, main_effects = c("Phenotype", "SecondPhenotype"),
                             covariates = "Characteristic")
imd_anova_res &lt;- imd_anova(omicsData = mymetab, test_method = 'comb')

# Same but with custom comparisons
comp_df &lt;- data.frame(Control = c("Phenotype1", "A"), Test = c("Phenotype2", "B"))
custom_comps_res &lt;- imd_anova(omicsData = mymetab, comparisons = comp_df, test_method = "combined")

</code></pre>

<hr>
<h2 id='imd_test'>Tests for the independence of missing data across groups (aka factors, aka
main effects)</h2><span id='topic+imd_test'></span>

<h3>Description</h3>

<p>Tests the null hypothesis that the number of missing observations is
independent of the groups.  A g-test is used to test this null hypothese
against the alternative that the groups and missing data are related.  This
is usually performed in conjuction with an ANOVA which tests if the mean
response (which varies with data type) is the same across groups; this
combination is called IMD_ANOVA.  It's probably a good idea to first filter
the data with 'imd_anova_filter' to see if there is enough infomration to
even do this test.  See Webb-Robertson et al. (2010) for more.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imd_test(
  omicsData,
  groupData,
  comparisons,
  pval_adjust_multcomp,
  pval_adjust_fdr,
  pval_thresh,
  covariates,
  paired,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imd_test_+3A_omicsdata">omicsData</code></td>
<td>
<p>A pmartR data object of any class</p>
</td></tr>
<tr><td><code id="imd_test_+3A_groupdata">groupData</code></td>
<td>
<p>'data.frame' that assigns sample names to groups</p>
</td></tr>
<tr><td><code id="imd_test_+3A_comparisons">comparisons</code></td>
<td>
<p>'data.frame' with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="imd_test_+3A_pval_adjust_multcomp">pval_adjust_multcomp</code></td>
<td>
<p>A character string specifying the type of multiple
comparisons adjustment to implement. The default setting, &quot;none&quot;, is to not
apply an adjustment. Valid options include: &quot;bonferroni&quot; and &quot;holm&quot;.</p>
</td></tr>
<tr><td><code id="imd_test_+3A_pval_adjust_fdr">pval_adjust_fdr</code></td>
<td>
<p>A character string specifying the type of FDR
adjustment to implement. The default setting, &quot;none&quot;, is to not
apply an adjustment. Valid options include: &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, and
&quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="imd_test_+3A_pval_thresh">pval_thresh</code></td>
<td>
<p>numeric p-value threshold, below or equal to which
peptides are considered differentially expressed. Defaults to 0.05</p>
</td></tr>
<tr><td><code id="imd_test_+3A_covariates">covariates</code></td>
<td>
<p>A character vector with no more than two variable names
that will be used as covariates in the IMD-ANOVA analysis.</p>
</td></tr>
<tr><td><code id="imd_test_+3A_paired">paired</code></td>
<td>
<p>A logical value that determines whether paired data should be
accounted for</p>
</td></tr>
<tr><td><code id="imd_test_+3A_parallel">parallel</code></td>
<td>
<p>A logical value indicating whether or not to use a
&quot;doParallel&quot; loop when running the G-Test with covariates. The default is
TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of 'data.frame's
</p>

<table>
<tr>
 <td style="text-align: left;">
  Results  </td><td style="text-align: left;"> e_data cname,
  Count of non-missing data for each group, Global G-test statistic and
  p-value</td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> Gstats  </td><td style="text-align: left;"> Value of the g statistics for each of the
  pairwise comparisons specified by the `comparisons` argument </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
  Pvalues  </td><td style="text-align: left;"> p-values for each of the pairwise comparisons specified by
  `comparisons` argument </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> Flags  </td><td style="text-align: left;"> Indicator of statistical
  significance where the sign of the flag reflects the difference in the
  ratio of non-missing observations (0/+-2 to if adjusted
  p-value&gt;=pval_thresh or p-value&lt;pval_thresh) </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Bryan Stanfill
</p>


<h3>References</h3>

<p>Webb-Robertson, Bobbie-Jo M., et al. &quot;Combined statistical
analyses of peptide intensities and peptide occurrences improves
identification of significant peptides from MS-based proteomics data.&quot;
Journal of proteome research 9.11 (2010): 5748-5756.
</p>

<hr>
<h2 id='imdanova_filter'>IMD-ANOVA Filter Object</h2><span id='topic+imdanova_filter'></span>

<h3>Description</h3>

<p>This function returns an imdanovaFilt object for use with
<code><a href="#topic+applyFilt">applyFilt</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imdanova_filter(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imdanova_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>object of one of the classes &quot;pepData&quot;, &quot;isobaricpepData&quot;,
&quot;proData&quot;, &quot;lipidData&quot;, &quot;metabData&quot;, or &quot;nmrData&quot;, created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.
Groups (more than one) must have been specified using the
<code><a href="#topic+group_designation">group_designation</a></code> function prior to using the imdanova_filter
function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output from this function can be used in conjunction with
<code><a href="#topic+applyFilt">applyFilt</a></code> to filter out molecules that are not present in
enough samples to do statistical comparisons. If any singleton groups are
present in the omicsData object, those groups are not part of the filter
object that is returned.
</p>


<h3>Value</h3>

<p>An S3 object of class imdanovaFilt (also a data.frame) containing the
molecule identifier and number of samples in each group with non-missing
values for that molecule.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- group_designation(omicsData = pep_object, main_effects = "Phenotype")
to_filter &lt;- imdanova_filter(omicsData = mypep)
summary(to_filter, min_nonmiss_anova = 2)

</code></pre>

<hr>
<h2 id='los'>Identify Biomolecules from the Top L Order Statistics for Use in
Normalization</h2><span id='topic+los'></span>

<h3>Description</h3>

<p>Select biomolecules for normalization via the method of the top L order
statistics (LOS)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>los(e_data, edata_id, L = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="los_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, proteins, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="los_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the column giving the
peptide, protein, lipid, or metabolite identifier. Usually obtained by
calling <code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="los_+3A_l">L</code></td>
<td>
<p>numeric value between 0 and 1, indicating the top proportion of
biomolecules to be retained (default value 0.05)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The biomolecule abundances of the top <code>L</code> order statistics are
identified and returned. Specifically, for each sample, the biomolecules with
the top <code>L</code> proportion of highest absolute abundance are retained, and
the union of these biomolecules is taken as the subset identified.
</p>


<h3>Value</h3>

<p>Character vector containing the biomolecules belonging to the subset.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton, Lisa Bramer
</p>

<hr>
<h2 id='mad_transform'>Median Absolute Deviation Transformation</h2><span id='topic+mad_transform'></span>

<h3>Description</h3>

<p>Calculate normalization parameters for the data via median absolute deviation
(MAD) transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mad_transform(
  e_data,
  edata_id,
  subset_fn,
  feature_subset,
  backtransform = FALSE,
  apply_norm = FALSE,
  check.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mad_transform_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, lipids, or metabolites and <code class="reqn">n</code> is the number of samples.
Each row corresponds to data for a peptide, lipid, or metabolite, with one
column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="mad_transform_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="mad_transform_+3A_subset_fn">subset_fn</code></td>
<td>
<p>character string indicating the subset function to use for
normalization.</p>
</td></tr>
<tr><td><code id="mad_transform_+3A_feature_subset">feature_subset</code></td>
<td>
<p>character vector containing the feature names in the
subset to be used for normalization</p>
</td></tr>
<tr><td><code id="mad_transform_+3A_backtransform">backtransform</code></td>
<td>
<p>logical argument. If TRUE, the parameters for
backtransforming the data after normalization will be calculated so that
the values are on a scale similar to their raw values. See details for more
information. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="mad_transform_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical argument. If TRUE, the normalization will be
applied to the data. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="mad_transform_+3A_check.names">check.names</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each feature is scaled by subtracting the median of the feature
subset specified for normalization and then dividing the result by the
median absolute deviation (MAD) of the feature subset specified for
normalization to get the normalized data. The location estimates are the
subset medians for each sample. The scale estimates are the subset MADs for
each sample. Medians are taken ignoring any NA values. If backtransform is
TRUE, the normalized feature values are multiplied by a pooled MAD
(estimated from all samples) and a global median of the subset data (across
all samples) is added back to the normalized values.
</p>


<h3>Value</h3>

<p>List containing two elements: <code>norm_params</code> is list with two
elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> numeric vector of length <code>n</code> median absolute deviations (MAD) for each sample </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric vector of length <code>n</code> medians for each sample
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>backtransform_params</code> is a list with two elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> numeric value giving pooled MAD </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric value giving global median across all samples
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If <code>backtransform</code> is set to TRUE then each list item under
<code>backtransform_params</code> will be NULL.
</p>
<p>If <code>apply_norm</code> is TRUE, the transformed data is returned as a third
list item.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>

<hr>
<h2 id='make_volcano_plot_df'>Create a plotting dataframe for volcano plots and heatmaps</h2><span id='topic+make_volcano_plot_df'></span>

<h3>Description</h3>

<p>A function internal to pmartR:::<a href="#topic+plot.statRes">plot.statRes</a> which creates the
dataframe necessary to construct volcano plots and heatmaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_volcano_plot_df(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_volcano_plot_df_+3A_x">x</code></td>
<td>
<p>'statRes' object to be plotted, usually the result of 'imd_anova'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'data.frame' object with plotting information about each biomolecule
such as missing counts per group, and p-values for t and g-tests.
</p>

<hr>
<h2 id='mean_center'>Mean Center Transformation</h2><span id='topic+mean_center'></span>

<h3>Description</h3>

<p>Calculate normalization parameters for the data via via mean centering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean_center(
  e_data,
  edata_id,
  subset_fn,
  feature_subset,
  backtransform = FALSE,
  apply_norm = FALSE,
  check.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mean_center_+3A_e_data">e_data</code></td>
<td>
<p>e_data a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the
number of peptides, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="mean_center_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="mean_center_+3A_subset_fn">subset_fn</code></td>
<td>
<p>character string indicating the subset function to use for
normalization.</p>
</td></tr>
<tr><td><code id="mean_center_+3A_feature_subset">feature_subset</code></td>
<td>
<p>character vector containing the feature names in the
subset to be used for normalization</p>
</td></tr>
<tr><td><code id="mean_center_+3A_backtransform">backtransform</code></td>
<td>
<p>logical argument. If TRUE, the data will be back
transformed after normalization so that the values are on a scale similar
to their raw values. See details for more information. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="mean_center_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical argument. If TRUE, the normalization will be
applied to the data. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="mean_center_+3A_check.names">check.names</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sample-wise mean of the feature subset specified for
normalization is subtracted from each feature in e_data to get the
normalized data. The location estimates are the sample-wise means of the
subset data. There are no scale estimates for mean centering, though the
function returns a NULL list element as a placeholdfer for a scale
estimate. If backtransform is TRUE, the global median of the subset data
(across all samples) is added back to the normalized values. Medians are
taken ignoring any NA values.
</p>


<h3>Value</h3>

<p>List containing two elements: <code>norm_params</code> is list with two
elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> NULL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric vector of length <code>n</code> means for each sample
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>backtransform_params</code> is a list with two elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> NULL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric value giving global median across all samples
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If <code>backtransform</code> is set to TRUE then each list item under
<code>backtransform_params</code> will be NULL.
</p>
<p>If <code>apply_norm</code> is TRUE, the transformed data is returned as a third
list item.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>

<hr>
<h2 id='median_center'>Median Center Transformation</h2><span id='topic+median_center'></span>

<h3>Description</h3>

<p>Calculate normalization parameters for the data via median centering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>median_center(
  e_data,
  edata_id,
  subset_fn,
  feature_subset,
  backtransform = FALSE,
  apply_norm = FALSE,
  check.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="median_center_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, lipids, or metabolites and <code class="reqn">n</code> is the number of samples.
Each row corresponds to data for a peptide, protein, lipid, or metabolite,
with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="median_center_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="median_center_+3A_subset_fn">subset_fn</code></td>
<td>
<p>character string indicating the subset function to use for
normalization.</p>
</td></tr>
<tr><td><code id="median_center_+3A_feature_subset">feature_subset</code></td>
<td>
<p>character vector containing the feature names in the
subset to be used for normalization</p>
</td></tr>
<tr><td><code id="median_center_+3A_backtransform">backtransform</code></td>
<td>
<p>logical argument. If TRUE, the data will be back
transformed after normalization so that the values are on a scale similar
to their raw values. See details for more information. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="median_center_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical argument. If TRUE, the normalization will be
applied to the data. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="median_center_+3A_check.names">check.names</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sample-wise median of the feature subset specified for
normalization is subtracted from each feature in e_data to get the
normalized data. The location estimates are the sample-wise medians of the
subset data. There are no scale estimates for median centering, though the
function returns a NULL list element as a placeholder for a scale estimate.
If backtransform is TRUE, the global median of the subset data (across all
samples) is added back to the normalized values. Medians are taken ignoring
any NA values.
</p>


<h3>Value</h3>

<p>List containing two elements: <code>norm_params</code> is list with two
elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> NULL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric vector of length <code>n</code> medians for each sample
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>backtransform_params</code> is a list with two elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> NULL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric value giving global median across all samples
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If <code>backtransform</code> is set to TRUE then each list item under
<code>backtransform_params</code> will be NULL.
</p>
<p>If <code>apply_norm</code> is TRUE, the transformed data is returned as a third
list item.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>

<hr>
<h2 id='missingval_result'>Creates an object of class naRes (NA Result)</h2><span id='topic+missingval_result'></span>

<h3>Description</h3>

<p>This function takes in an omicsData object, and outputs a list of two data
frames, one containing the number of missing values by sample, and the other
containing the number of missing values by molecule
</p>


<h3>Usage</h3>

<pre><code class='language-R'>missingval_result(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="missingval_result_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of class &quot;pepData&quot;, &quot;proData&quot;, &quot;metabData&quot;,
&quot;lipidData&quot;, &quot;nmrData&quot;, or &quot;seqData&quot;, created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or
<code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S3 object of class naRes, which is a list of two data frames, one
containing the number of missing values per sample, and the other
containing the number of missing values per molecule. For count data,
zeroes represent missing values; for abundance data, NA's represent missing
values. This object can be used with 'plot' and 'summary' methods to
examine the missing values in the dataset.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
result1 = missingval_result(omicsData = lipid_neg_object)
result2 = missingval_result(omicsData = metab_object)

</code></pre>

<hr>
<h2 id='molecule_filter'>Molecule Filter Object</h2><span id='topic+molecule_filter'></span>

<h3>Description</h3>

<p>This function returns a moleculeFilt object for use with
<code><a href="#topic+applyFilt">applyFilt</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>molecule_filter(omicsData, use_groups = FALSE, use_batch = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="molecule_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', or 'seqData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="molecule_filter_+3A_use_groups">use_groups</code></td>
<td>
<p>logical indicator for whether to utilize group information
from <code><a href="#topic+group_designation">group_designation</a></code> when calculating the molecule filter.
Defaults to FALSE. When group information is used to calculate the molecule
filter, the minimum number of observations is required within each group,
as opposed to across all samples regardless of group membership.</p>
</td></tr>
<tr><td><code id="molecule_filter_+3A_use_batch">use_batch</code></td>
<td>
<p>logical indicator for whether to utilize batch information
from <code><a href="#topic+group_designation">group_designation</a></code> when calculating the molecule filter.
Defaults to FALSE. When batch information is used to calculate the molecule
filter, the minimum number of observations is required within each batch,
as opposed to across all samples regardless of batch. If ComBat or similar
method will be used for downstream batch effect correction, this argument
should be set to TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Attribute of molecule_filt object is &quot;total_poss_obs&quot;, the number of
total possible observations for each feature (same as the number of
samples)
</p>


<h3>Value</h3>

<p>An S3 object of class 'moleculeFilt' (also a data.frame) that contains the
molecule identifier and the number of samples for which the molecule was
observed (i.e. not NA)
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
to_filter &lt;- molecule_filter(omicsData = pep_object)
summary(to_filter, min_num = 2)

</code></pre>

<hr>
<h2 id='nonmissing_per_group'>Computes the Number of Non-Missing Data Points by Group</h2><span id='topic+nonmissing_per_group'></span>

<h3>Description</h3>

<p>This function computes the number of non-missing observations for samples,
based on a group designation, for every biomolecule in the dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nonmissing_per_group(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nonmissing_per_group_+3A_omicsdata">omicsData</code></td>
<td>
<p>an optional object of one of the classes &quot;pepData&quot;,
&quot;proData&quot;, &quot;metabData&quot;,  &quot;lipidData&quot;, or &quot;nmrData&quot;, usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>, or
<code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively. Either omicsData or all of
e_data, groupDF, cname_id, and samp_id must be provided.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of length two. The first element giving the total number of
possible samples for each group. The second element giving a
data.frame with the first column giving the peptide and the second
through kth columns giving the number of non-missing observations for
each of the <code>k</code> groups.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>

<hr>
<h2 id='normalize_global'>Calculate Normalization Parameters and Apply Global Normalization</h2><span id='topic+normalize_global'></span>

<h3>Description</h3>

<p>Calculates normalization parameters based on the data using the specified
subset and normalization functions with option to apply the normalization to
the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_global(
  omicsData,
  subset_fn,
  norm_fn,
  params = NULL,
  apply_norm = FALSE,
  backtransform = FALSE,
  min_prop = NULL,
  check.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_global_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively. The
function <code><a href="#topic+group_designation">group_designation</a></code> must have been run on omicsData to
use several of the subset functions (i.e. rip and ppp_rip).</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_subset_fn">subset_fn</code></td>
<td>
<p>character string indicating the subset function to use for
normalization. See details for the current offerings.</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_norm_fn">norm_fn</code></td>
<td>
<p>character string indicating the normalization function to use
for normalization. See details for the current offerings.</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_params">params</code></td>
<td>
<p>additional arguments passed to the specified subset function.
See details for parameter specification and default values.</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical argument indicating if the normalization should be
applied to the data. Defaults to FALSE. If TRUE, the normalization is
applied to the data and an S3 object of the same class as <code>omicsData</code>
(e.g. 'pepData') with normalized values in <code>e_data</code> is returned. If
FALSE, the normalization is not applied to the data and an S3 object of
class 'normRes' is returned.</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_backtransform">backtransform</code></td>
<td>
<p>logical argument indicating if parameters for back
transforming the data, after normalization, should be calculated. Defaults
to FALSE. If TRUE, the parameters for back transforming the data after
normalization will be calculated, and subsequently included in the data
normalization if <code>apply_norm</code> is TRUE. See the details section for an 
explanation of how these factors are calculated.</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_min_prop">min_prop</code></td>
<td>
<p>numeric threshold between 0 and 1 giving the minimum value
for the proportion of biomolecules subset (rows of <code>e_data</code>)</p>
</td></tr>
<tr><td><code id="normalize_global_+3A_check.names">check.names</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Below are details for specifying function and parameter options.
</p>


<h3>Value</h3>

<p>If apply_norm is FALSE, an S3 object of type 'normRes' is returned.
This object contains a list with: subset method, normalization method,
normalization parameters, number of biomolecules used in normalization, and
proportion of biomolecules used in normalization. plot() and summary() methods
are available for this object. If apply_norm is TRUE, then the normalized
data is returned in an object of the appropriate S3 class (e.g. pepData).
</p>


<h3>Subset Functions</h3>

<p>Specifying a subset function indicates the subset
of biomolecules (rows of <code>e_data</code>) that should be used for computing
normalization factors. The following are valid options: &quot;all&quot;, &quot;los&quot;,
&quot;ppp&quot;, &quot;complete&quot;, &quot;rip&quot;, and &quot;ppp_rip&quot;. The option &quot;all&quot; is the subset
that includes all biomolecules (i.e. no subsetting is done). The option 
&quot;los&quot; identifies the subset of the biomolecules associated with the top 
<code>L</code> order statistics, where <code>L</code> is a proportion between 0 and 1.
Specifically, the biomolecules falling within the top <code>L</code> proportion of highest
absolute abundance are retained for each sample, and the union of these
biomolecules is taken as the subset identified (Wang et al., 2006). The option
&quot;ppp&quot; (originally stands for percentage of peptides present) identifies the
subset of biomolecules that are present/non-missing for a minimum
<code>proportion</code> of samples (Karpievitch et al., 2009; Kultima et al.,
2009). The option &quot;complete&quot; retains molecules with no missing data across
all samples, equivalent to &quot;ppp&quot; with proportion = 1. The option &quot;rip&quot;
identifies biomolecules with complete data that have a p-value greater than a
defined threshold <code>alpha</code> (common values include 0.1 or 0.25) when
subjected to a Kruskal-Wallis test based (non-parametric one-way ANOVA) on
group membership (Webb-Robertson et al., 2011). The option &quot;ppp_rip&quot; is
equivalent to &quot;rip&quot; however rather than requiring biomolecules with complete
data, biomolecules with at least a <code>proportion</code> of non-missing values are
subject to the Kruskal-Wallis test.
</p>


<h3>Normalization Functions</h3>

<p>Specifying a normalization function
indicates how normalization scale and location parameters should be
calculated. The following are valid options: &quot;median&quot;, &quot;mean&quot;, &quot;zscore&quot;,
and &quot;mad&quot;. For median centering, the location estimates are the sample-wise
medians of the subset data and there are no scale estimates. For mean
centering, the location estimates are the sample-wise means of the subset
data and there are no scale estimates. For z-score transformation, the
location estimates are the subset means for each sample and the scale
estimates are the subset standard deviations for each sample. For median
absolute deviation (MAD) transformation, the location estimates are the
subset medians for each sample and the scale estimates are the subset MADs
for each sample.
</p>


<h3>Specifying Subset Parameters Using the <code>params</code> Argument</h3>

<p>Parameters for the chosen subset function should be specified in a list
with the function specification followed by an equal sign and the desired
parameter value. For example, if LOS with 0.1 is desired, one should use
<code>params = list(los = 0.1)</code>. ppp_rip can be specified in one of two
ways: specify the parameters with each separate function or combine using a
nested list (e.g. <code>params = list(ppp_rip = list(ppp = 0.5, rip =
  0.2))</code>).
</p>
<p>The following functions have parameters that can be specified:
</p>

<table>
<tr>
 <td style="text-align: left;"> los </td><td style="text-align: left;"> a value between 0 and 1 indicating the top
  proportion of order statistics. Defaults to 0.05 if unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;">
  </td>
</tr>
<tr>
 <td style="text-align: left;"> ppp </td><td style="text-align: left;"> a value between 0 and 1 specifying the proportion of samples
  that must have non-missing values for a biomolecule to be retained. Defaults to
  0.5 if unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> rip </td><td style="text-align: left;"> a value between 0 and 1
  specifying the p-value threshold for determining rank invariance. Defaults
  to 0.2 if unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> ppp_rip </td><td style="text-align: left;"> two values corresponding
  to the RIP and PPP parameters above. Defaults to 0.5 and 0.2, respectively.
  </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Backtransform</h3>

<p>The purpose of back transforming data is to ensure
values are on a scale similar to their raw values before normaliztion. The
following values are calculated and/or applied for backtransformation
purposes: </p>

<table>
<tr>
 <td style="text-align: left;"> <code>median</code> </td><td style="text-align: left;"> scale is NULL and location
  parameter is a global median across all samples </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>mean</code>
  </td><td style="text-align: left;"> scale is NULL and location parameter is a global median across all
  samples </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>zscore</code> </td><td style="text-align: left;"> scale is pooled standard deviation
  and location is global mean across all samples </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> <code>mad</code> </td><td style="text-align: left;">
  scale is pooled median absolute deviation and location is global median
  across all samples </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>References</h3>

<p>Webb-Robertson BJ, Matzke MM, Jacobs JM, Pounds JG, Waters KM. A
statistical selection strategy for normalization procedures in LC-MS
proteomics experiments through dataset-dependent ranking of normalization
scaling factors. Proteomics. 2011;11(24):4736-41.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

mymetab &lt;- edata_transform(
  omicsData = metab_object,
  data_scale = "log2"
)
mymetab &lt;- group_designation(
  omicsData = mymetab,
  main_effects = "Phenotype"
)
norm_object &lt;- normalize_global(
  omicsData = mymetab,
  subset_fn = "all",
  norm_fn = "median"
)
norm_data &lt;- normalize_global(
  omicsData = mymetab,
  subset_fn = "all",
  norm_fn = "median",
  apply_norm = TRUE,
  backtransform = TRUE
)

</code></pre>

<hr>
<h2 id='normalize_global_basic'>Normalize e_data within SPANS</h2><span id='topic+normalize_global_basic'></span>

<h3>Description</h3>

<p>This function is intended to be used in SPANS only. It is a VERY trimmed down
version of normalize_global. It is trimmed down because within SPANS we only
need the norm_params element from the output of the normalize_global
function. All of the other options and output can be ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_global_basic(edata, norm_fn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_global_basic_+3A_edata">edata</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, lipids, or metabolites and <code class="reqn">n</code> is the number of samples.
Each row corresponds to data for a peptide, protein, lipid, or metabolite,
with a column giving the identifer name.</p>
</td></tr>
<tr><td><code id="normalize_global_basic_+3A_norm_fn">norm_fn</code></td>
<td>
<p>character string indicating the normalization function to use
for normalization. See details for the current offerings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the location and scale parameters for normalizing
the data.
</p>

<hr>
<h2 id='normalize_isobaric'>Examine and Apply Isobaric Normalization</h2><span id='topic+normalize_isobaric'></span>

<h3>Description</h3>

<p>Examine reference pool samples and apply normalization of study samples to
their corresponding reference pool sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_isobaric(
  omicsData,
  exp_cname = NULL,
  apply_norm = FALSE,
  channel_cname = NULL,
  refpool_channel = NULL,
  refpool_cname = NULL,
  refpool_notation = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_isobaric_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'isobaricpepData'</p>
</td></tr>
<tr><td><code id="normalize_isobaric_+3A_exp_cname">exp_cname</code></td>
<td>
<p>character string specifying the name of the column
containing the experiment/plate information in <code>f_data</code></p>
</td></tr>
<tr><td><code id="normalize_isobaric_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical, indicates whether normalization should be applied
to omicsData$e_data</p>
</td></tr>
<tr><td><code id="normalize_isobaric_+3A_channel_cname">channel_cname</code></td>
<td>
<p>optional character string specifying the name of the
column containing the instrument channel a sample was run on in
<code>f_data</code>. This argument is optional. See Details for how to specify
information regarding reference pool samples. If using this argument, the
'refpool_channel' argument must also be specified; in this case,
'refpool_cname' and 'refpool_notation' should not be specified.</p>
</td></tr>
<tr><td><code id="normalize_isobaric_+3A_refpool_channel">refpool_channel</code></td>
<td>
<p>optional character string specifying which channel
contains the reference pool sample. Only used when this is the same
from experiment to experiment. This argument is optional. See Details for
how to specify information regarding reference pool samples. If using this
argument, the 'channel_cname' argument must also be specified; in this
case, 'refpool_cname' and 'refpool_notation' should not be specified.</p>
</td></tr>
<tr><td><code id="normalize_isobaric_+3A_refpool_cname">refpool_cname</code></td>
<td>
<p>optional character string specifying the name of the
column containing information about which samples are reference samples in
<code>f_data</code>. This argument is optional. see Details for how to specify
information regarding reference pool samples. If using this argument, the
'refpool_notation' argument must also be specified; in this case,
'channel_cname' and 'refpool_channel' should not be specified.</p>
</td></tr>
<tr><td><code id="normalize_isobaric_+3A_refpool_notation">refpool_notation</code></td>
<td>
<p>optional character string specifying the value in the
refpool_channel column which denotes that a sample is a reference sample.
This argument is optional. See Details for how to specify information
regarding reference pool samples. If using this argument, the
'refpool_cname' argument must also be specified; in this case,
'channel_cname' and 'refpool_channel' should not be specified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two ways to specify the information needed for identifying
reference samples which should be used for normalization: </p>
 <ol>
<li>
<p>specify <code>channel_cname</code> and <code>refpool_channel</code>. This should be
used when the reference sample for each experiment/plate was always located
in the same channel. Here <code>channel_cname</code> gives the column name for
the column in <code>f_data</code> which gives information about which channel
each sample was run on, and <code>refpool_channel</code> is a character string
specifying the value in <code>channel_colname</code> that corresponds to the
reference sample channel. </p>
</li>
<li><p> specify <code>refpool_cname</code> and
<code>refpool_notation</code>. This should be used when the reference sample is
not in a consistent channel across experiments/plates. Here,
<code>refpool_cname</code> gives the name of the column in <code>f_data</code> which
indicates whether a sample is a reference or not, and
<code>refpool_notation</code> is a character string giving the value used to
denote a reference sample in that column. </p>
</li></ol>
<p> In both cases you must specify
<code>exp_cname</code> which gives the column name for the column in
<code>f_data</code> containing information about which experiment/plate a sample
was run on.
</p>


<h3>Value</h3>

<p>If apply_norm = TRUE, an object of class 'isobaricpepData', normalized to reference pool,
and with the attribute 'isobaric_info' updated to include information about
the reference pool samples and the normalization procedure.  Otherwise an
object of class 'isobaricnormRes' containing similar information about the 
normalization process
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

myiso &lt;- edata_transform(isobaric_object, "log2")

# Don't apply the normalization quite yet;
# can use summary() and plot() to view reference pool samples
myiso_refpools &lt;- normalize_isobaric(
  omicsData = myiso, exp_cname = "Plex",
  apply_norm = FALSE,
  refpool_cname = "Virus",
  refpool_notation = "Pool"
)
summary(myiso_refpools)

# Now apply the normalization;
# can use plot() to view the study samples after reference pool normalization
myiso_norm &lt;- normalize_isobaric(
  omicsData = myiso, exp_cname = "Plex",
  apply_norm = TRUE,
  refpool_cname = "Virus",
  refpool_notation = "Pool"
)

</code></pre>

<hr>
<h2 id='normalize_loess'>Loess Normalization</h2><span id='topic+normalize_loess'></span>

<h3>Description</h3>

<p>Perform Loess normalization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_loess(omicsData, method = "fast", span = 0.4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_loess_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively. The
function <code><a href="#topic+group_designation">group_designation</a></code> must have been run on omicsData to
use several of the subset functions (i.e. rip and ppp_rip).</p>
</td></tr>
<tr><td><code id="normalize_loess_+3A_method">method</code></td>
<td>
<p>character string specifying which variant of the cyclic loess
method to use. Options are &quot;fast&quot; (default), &quot;affy&quot;, or &quot;pairs&quot;</p>
</td></tr>
<tr><td><code id="normalize_loess_+3A_span">span</code></td>
<td>
<p>span of loess smoothing window, between 0 and 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A wrapper for the normalizeCyclicLoess function from the limma
package.
</p>


<h3>Value</h3>

<p>The normalized data is returned in an object of the appropriate S3
class (e.g. pepData), on the same scale as omicsData (e.g. if omicsData
contains log2 transformed data, the normalization will be performed on the
non-log2 scale and then re-scaled after normalization to be returned on the
log2 scale).
</p>


<h3>References</h3>

<p>Bolstad, B. M., Irizarry R. A., Astrand, M., and Speed, T. P.
(2003). <em>A comparison of normalization methods for high density
oligonucleotide array data based on bias and variance.</em> Bioinformatics 19,
185-193.
</p>
<p>Ballman, KV Grill, DE, Oberg, AL and Therneau, TM (2004).
<em>Faster cyclic loess: normalizing RNA arrays via linear models.</em>
Bioinformatics 20, 2778-2786.
</p>


<h3>See Also</h3>

<p><code><a href="limma.html#topic+normalizeCyclicLoess">normalizeCyclicLoess</a></code> in the <code>limma</code> package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- edata_transform(pep_object, "log2")
result &lt;- normalize_loess(mypep)

</code></pre>

<hr>
<h2 id='normalize_nmr'>Normalize an Object of Class nmrData</h2><span id='topic+normalize_nmr'></span>

<h3>Description</h3>

<p>The data is normalized either to a spiked-in metabolite or to a
sample-specific property
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_nmr(
  omicsData,
  apply_norm = FALSE,
  backtransform = FALSE,
  metabolite_name = NULL,
  sample_property_cname = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_nmr_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'nmrData'</p>
</td></tr>
<tr><td><code id="normalize_nmr_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical, indicates whether normalization should be applied
to omicsData$e_data. Defaults to FALSE. If TRUE, the normalization is
applied to the data and an S3 object of the same class as <code>omicsData</code>
(e.g. 'nmrData') with normalized values in <code>e_data</code> is returned. If
FALSE, the normalization is not applied and an S3 object of the class
<code>nmrnormRes</code> is returned, allowing some exploratory data analysis
prior to subsequently applying the normalization.</p>
</td></tr>
<tr><td><code id="normalize_nmr_+3A_backtransform">backtransform</code></td>
<td>
<p>logical argument indicating if parameters for back
transforming the data, after normalization, should be calculated. Defaults
to FALSE. If TRUE, the parameters for back transforming the data after
normalization will be calculated, and subsequently included in the data
normalization if <code>apply_norm</code> is TRUE. See details for an explanation 
of how these factors are calculated.</p>
</td></tr>
<tr><td><code id="normalize_nmr_+3A_metabolite_name">metabolite_name</code></td>
<td>
<p>optional character string specifying the name of the
(spiked in) metabolite in <code>e_data</code> to use for instrument normalization
of the nmrData object. These values will be used to divide the raw
abundance of the corresponding sample in e_data (if e_data is log
transformed, this function accounts for that and returns normalized data on
the same scale it was provided). If using this argument, the
'sample_property_cname' argument should not be specified.</p>
</td></tr>
<tr><td><code id="normalize_nmr_+3A_sample_property_cname">sample_property_cname</code></td>
<td>
<p>optional character string specifying the name of
the column in f_data containing information to use for instrument
normalization of the nmrData object, such as a concentration. These values
will be used to divide the raw abundance of the corresponding sample in
e_data (if e_data is log transformed, this function accounts for that by
temporarily un-log transforming the data and then returning normalized data
on the same scale it was provided). If using this argument, the
'metabolite_name' argument should not be specified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two ways to specify the information needed for
performing instrument normalization on an nmrData object: </p>
 <ol>
<li>
<p>specify <code>metabolite_name</code>. This should be used when normalization to a
spiked in standard is desired. Here <code>metabolite_name</code> gives the name of
the metabolite in e_data (and e_meta, if present) corresponding to the spiked
in standard. If any samples have a missing value for this metabolite, an
error is returned. </p>
</li>
<li><p> specify <code>sample_property_cname</code>. This should be
used when normalizing to a sample property, such as concentration, is
desired. Here, <code>sample_property_cname</code> gives the name of the column in
<code>f_data</code> which contains the property to use for normalization. If any
samples have a missing value for this column, and error is returned.</p>
</li></ol>



<h3>Value</h3>

<p>If <code>apply_norm</code> is TRUE, an object of class 'nmrData', normalized
and with information about the normalization process in 'nmr_info'.  Otherwise,
an object of class 'nmrnormRes' is returned, with the same info about
normalization in attribute 'nmr_info' to be passed to plotting and summary functions.
</p>


<h3>Backtransform</h3>

<p>The purpose of back transforming data is to ensure
values are on a scale similar to their raw values before normalization. The
following values are calculated and/or applied for backtransformation
purposes: </p>

<table>
<tr>
 <td style="text-align: left;"> If normalization using a metabolite in
<code>e_data</code> is specified </td><td style="text-align: left;"> location parameter is the median of the
values for <code>metabolite_name</code> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> If normalization using a
sample property in <code>f_data</code> is specified </td><td style="text-align: left;"> location parameter is
the median of the values in <code>sample_property</code> </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> See examples below.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

# Normalize using a metabolite (this is merely an example of how to use this specification;
# the metabolite used was not actually spiked-in for the purpose of normalization)
mynmr &lt;- edata_transform(
  omicsData = nmr_identified_object,
  data_scale = "log2"
)
nmr_norm &lt;- normalize_nmr(
  omicsData = mynmr, apply_norm = TRUE,
  metabolite_name = "unkm1.53",
  backtransform = TRUE
)

# Normalization using a sample property
mynmr &lt;- edata_transform(
  omicsData = nmr_identified_object,
  data_scale = "log2"
)
nmr_norm &lt;- normalize_nmr(
  omicsData = mynmr, apply_norm = TRUE,
  sample_property_cname = "Concentration",
  backtransform = TRUE
)

</code></pre>

<hr>
<h2 id='normalize_quantile'>Quantile Normalization</h2><span id='topic+normalize_quantile'></span>

<h3>Description</h3>

<p>Perform quantile normalization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_quantile(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_quantile_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', 'nmrData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Quantile normalization is an algorithm for normalizing a set of data
vectors by giving them the same distribution. It is applied to data on the
abundance scale (e.g. not a log scale). It is often used for microarray
data.
</p>
<p>The method is implemented as described in Bolstad et al. (2003).
</p>


<h3>Value</h3>

<p>The normalized data is returned in an object of the appropriate S3
class (e.g. pepData), on the same scale as omicsData (e.g. if omicsData
contains log2 transformed data, the normalization will be performed on the
non-log2 scale and then re-scaled after normalization to be returned on the
log2 scale).
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>


<h3>References</h3>

<p>Bolstad, B. M., Irizarry, R. A., strand, M., &amp; Speed, T. P.
(2003). A comparison of normalization methods for high density
oligonucleotide array data based on variance and bias. Bioinformatics,
19(2), 185-193.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myfilt &lt;- molecule_filter(omicsData = metab_object)
# quantile normalization requires complete data
# summary(myfilt, min_num = 50)
mymetab &lt;- applyFilt(filter_object = myfilt, omicsData = metab_object, min_num = 50)
norm_data &lt;- normalize_quantile(omicsData = mymetab)

</code></pre>

<hr>
<h2 id='normRes_tests'>Test the location and scale parameters from a normalization procedure</h2><span id='topic+normRes_tests'></span>

<h3>Description</h3>

<p>Computes p-values from a test of dependence between normalization parameters
and group assignment of a normalized omicsData or normRes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normRes_tests(norm_obj, test_fn = "kw")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normRes_tests_+3A_norm_obj">norm_obj</code></td>
<td>
<p>object of class 'pepData', 'proData', 'lipidData',
'metabData', 'isobaricpepData', or 'nmrData' that has had <code>normalize_global()</code> run
on it, or a 'normRes' object</p>
</td></tr>
<tr><td><code id="normRes_tests_+3A_test_fn">test_fn</code></td>
<td>
<p>character string indicating the statistical test to use.
Current options are &quot;anova&quot; and &quot;kw&quot; for a Kruskal-Wallis test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 2 entries containing the p_value of the test performed on
the location and scale (if it exists) parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
mymetab &lt;- group_designation(omicsData = mymetab, main_effects = "Phenotype")

# provide the normRes object
mynorm &lt;- normalize_global(omicsData = mymetab, subset_fn = "all",
                           norm_fn = "median", apply_norm = FALSE)
norm_pvals &lt;- normRes_tests(norm_obj = mynorm)

# provide normalized omicsData object
mymetab &lt;- normalize_global(omicsData = mymetab, subset_fn = "all",
                            norm_fn = "median", apply_norm = TRUE)
norm_pvals &lt;- normRes_tests(norm_obj = mymetab)

# NMR data object
mynmr &lt;- edata_transform(omicsData = nmr_identified_object, data_scale = "log2")
mynmr &lt;- group_designation(mynmr, main_effects = "Condition")
mynmrnorm &lt;- normalize_nmr(
  omicsData = mynmr,
  apply_norm = TRUE,
  sample_property_cname = "Concentration"
)
mynmrnorm &lt;- normalize_global(omicsData = mynmrnorm, subset_fn = "all",
                              norm_fn = "median", apply_norm = TRUE, backtransform = TRUE)
norm_pvals &lt;- normRes_tests(norm_obj = mynmrnorm)

</code></pre>

<hr>
<h2 id='p_adjustment_anova'>Adjust p-values for multiple comparisons</h2><span id='topic+p_adjustment_anova'></span>

<h3>Description</h3>

<p>Depending upon the <code>pval_adjust</code> method selected, the supplied p_values are compared against an adjusted <code>pval_thresh</code> value or the provided
means are used to compute new statistics, p-values are computed and compared against the provided <code>pval_thresh</code>.  A <code>data.frame</code> that indicates which
of the tests are significant, 1 if significant or 0 if insignificant.  If <code>means</code> is also provided and the p-value is signficant then the direction
of the change is indicated by the sign on 1, i.e., means&lt;0 and p_value&lt;pval_thresh will return -1, similarly for means&gt;0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_adjustment_anova(
  p_values,
  diff_mean,
  t_stats,
  sizes,
  pval_adjust_multcomp,
  pval_adjust_fdr
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="p_adjustment_anova_+3A_p_values">p_values</code></td>
<td>
<p>A matrix (or <code>data.frame</code>) of p-values to be adjusted.</p>
</td></tr>
<tr><td><code id="p_adjustment_anova_+3A_diff_mean">diff_mean</code></td>
<td>
<p>A matrix (or <code>data.frame</code>) of groups means that are to be compared</p>
</td></tr>
<tr><td><code id="p_adjustment_anova_+3A_t_stats">t_stats</code></td>
<td>
<p>A matrix (or <code>data.frame</code>) of t-test statistics resulting from from standard procedures</p>
</td></tr>
<tr><td><code id="p_adjustment_anova_+3A_sizes">sizes</code></td>
<td>
<p>A matrix (or <code>data.frame</code>) of group sizes</p>
</td></tr>
<tr><td><code id="p_adjustment_anova_+3A_pval_adjust_multcomp">pval_adjust_multcomp</code></td>
<td>
<p>character vector specifying the type of multiple comparisons adjustment to implement. A NULL value corresponds to no adjustment. Valid options include: holm, bonferroni, dunnett, tukey or none.</p>
</td></tr>
<tr><td><code id="p_adjustment_anova_+3A_pval_adjust_fdr">pval_adjust_fdr</code></td>
<td>
<p>character vector specifying the type of FDR adjustment to implement. A NULL value corresponds to no adjustment. Valid options include: bonferroni, BH, BY, fdr, or none.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the following columns: group means, global G-test statistic and corresponding p-value
</p>


<h3>Author(s)</h3>

<p>Bryan Stanfill
</p>

<hr>
<h2 id='plot_km'>Basic survival analysis plot</h2><span id='topic+plot_km'></span>

<h3>Description</h3>

<p>Implements overall survival analysis or progression-free survival analysis, depending upon the datatypes supplied to
surv_designation, and plot the resulting Kaplan-Meier curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_km(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_km_+3A_omicsdata">omicsData</code></td>
<td>
<p>A pmartR data object of any class, which has a 'group_df' attribute that is usually created by the 'group_designation()' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Kaplan-Meier curve
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(MSomicsSTAT)
library(OvarianPepdataBP)
attr(tcga_ovarian_pepdata_bp, "survDF") &lt;- list(t_death = "survival_time",
                                                ind_death = "vital_status")
plot_km(omicsData = tcga_ovarian_pepdata_bp)

# Add covariates to "survDF" attribute
attr(tcga_ovarian_pepdata_bp, "survDF") &lt;- list(
  t_death = "survival_time",
  ind_death = "vital_status",
  covariates = "age_at_initial_pathologic_diagnosis"
)
plot_km(omicsData = tcga_ovarian_pepdata_bp)

## End(Not run)

</code></pre>

<hr>
<h2 id='plot.corRes'>Plot corRes Object</h2><span id='topic+plot.corRes'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'corRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'corRes'
plot(
  x,
  omicsData = NULL,
  order_by = NULL,
  colorbar_lim = c(NA, NA),
  x_text = TRUE,
  y_text = TRUE,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  color_low = NULL,
  color_high = NULL,
  bw_theme = TRUE,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.corRes_+3A_x">x</code></td>
<td>
<p>An object of class &quot;corRes&quot; created via <code>cor_result</code></p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'isobaricpepData',
'proData', 'lipidData', 'metabData', 'nmrData' or 'seqData' created via
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or
<code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_order_by">order_by</code></td>
<td>
<p>A character string specifying a column in f_data by which to
order the samples.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_colorbar_lim">colorbar_lim</code></td>
<td>
<p>A pair of numeric values specifying the minimum and
maximum values to use in the heat map color bar. Defaults to 'c(NA, NA)',
in which case ggplot2 automatically sets the minimum and maximum values
based on the correlation values in the data.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_x_text">x_text</code></td>
<td>
<p>logical value. Indicates whether the x-axis will be labeled
with the sample names. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_y_text">y_text</code></td>
<td>
<p>logical value. Indicates whether the y-axis will be labeled
with the sample names. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 90.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_color_low">color_low</code></td>
<td>
<p>character string specifying the color of the gradient for
low values</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_color_high">color_high</code></td>
<td>
<p>character string specifying the color of the gradient for
high values</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.corRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
mymetab &lt;- group_designation(omicsData = mymetab, main_effects = "Phenotype")
my_correlation &lt;- cor_result(omicsData = mymetab)
plot(my_correlation, omicsData = mymetab, order_by = "Phenotype")


myseq_correlation &lt;- cor_result(omicsData = rnaseq_object)
plot(myseq_correlation)


</code></pre>

<hr>
<h2 id='plot.customFilt'>Plot customFilt Object</h2><span id='topic+plot.customFilt'></span>

<h3>Description</h3>

<p>Currently plotting a customFilt object is not supported
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'customFilt'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.customFilt_+3A_x">x</code></td>
<td>
<p>An object of class customFilt.</p>
</td></tr>
<tr><td><code id="plot.customFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, implemented to provide information to user.
</p>

<hr>
<h2 id='plot.cvFilt'>Plot cvFilt Object</h2><span id='topic+plot.cvFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'cvFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cvFilt'
plot(
  x,
  cv_threshold = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 0,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  log_scale = TRUE,
  n_breaks = 15,
  n_bins = 30,
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.cvFilt_+3A_x">x</code></td>
<td>
<p>object of class cvFilt generated via
<code><a href="#topic+cv_filter">cv_filter</a></code></p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_cv_threshold">cv_threshold</code></td>
<td>
<p>numeric value for cv threshold above which to remove the
corresponding biomolecules</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_log_scale">log_scale</code></td>
<td>
<p>logical value. Indicates whether to use a log2 transformed x-axis.
The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_n_breaks">n_breaks</code></td>
<td>
<p>integer value specifying the number of breaks to use. You
may get less breaks if rounding causes certain values to become non-unique.
The default is 15.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_n_bins">n_bins</code></td>
<td>
<p>integer value specifying the number of bins to draw in the
histogram.  The default is 30.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.cvFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
data(pep_object)
mypep &lt;- group_designation(
  omicsData = pep_object,
  main_effects = "Phenotype"
)

cvfilt &lt;- cv_filter(omicsData = mypep)

plot(cvfilt, cv_threshold = 20)
plot(cvfilt, cv_threshold = 10, log_scale = FALSE)

</code></pre>

<hr>
<h2 id='plot.dataRes'>Plot dataRes object</h2><span id='topic+plot.dataRes'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type dataRes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dataRes'
plot(
  x,
  metric = NULL,
  density = FALSE,
  ncols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  x_lab_sd = NULL,
  x_lab_median = NULL,
  y_lab = NULL,
  y_lab_sd = NULL,
  y_lab_median = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = NULL,
  title_lab = NULL,
  title_lab_sd = NULL,
  title_lab_median = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  point_size = 2,
  bin_width = 1,
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.dataRes_+3A_x">x</code></td>
<td>
<p>object of class dataRes, created by the
<code><a href="#topic+edata_summary">edata_summary</a></code> function</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_metric">metric</code></td>
<td>
<p>character string indicating which metric to use in plot:
'mean', 'median', 'sd, 'pct_obs', 'min', or 'max'</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_density">density</code></td>
<td>
<p>logical value, defaults to FALSE. If TRUE, a density plot of
the specified metric is returned.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_ncols">ncols</code></td>
<td>
<p>integer value specifying the number columns for the histogram
facet_wrap. This argument is used when <code>metric</code> is not null. The
default is NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE, produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label when the metric
argument is NULL. The default is NULL in which case the x-axis label will
be &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_x_lab_sd">x_lab_sd</code></td>
<td>
<p>character string used for the x-axis label for the
mean/standard deviation plot when the <code>metric</code> argument is not NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_x_lab_median">x_lab_median</code></td>
<td>
<p>character string used for the x-axis label for the
mean/median plot when the <code>metric</code> argument is not NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_y_lab_sd">y_lab_sd</code></td>
<td>
<p>character string used for the y-axis label for the
mean/standard deviation plot when the <code>metric</code> argument is not NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_y_lab_median">y_lab_median</code></td>
<td>
<p>character string used for the y-axis label for the
mean/median plot when the <code>metric</code> argument is not NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title when the
<code>metric</code> argument is NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_title_lab_sd">title_lab_sd</code></td>
<td>
<p>character string used for the plot title for the
mean/standard deviation plot when the <code>metric</code> argument is not NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_title_lab_median">title_lab_median</code></td>
<td>
<p>character string used for the plot title for the
mean/median plot when the <code>metric</code> argument is not NULL.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, or &quot;bottom&quot;. The default is
&quot;right&quot;.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_point_size">point_size</code></td>
<td>
<p>integer specifying the size of the points. The default is
2.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_bin_width">bin_width</code></td>
<td>
<p>integer indicating the bin width in a histogram. The default
is 0.5.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE, uses the ggplot2 black and white
theme.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.dataRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can only create plots for dataRes objects whose 'by' =
'molecule' and 'groupvar' attribute is non NULL
</p>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mylipid &lt;- edata_transform(omicsData = lipid_pos_object, data_scale = "log2")
result &lt;- edata_summary(
  omicsData = mylipid,
  by = "molecule",
  groupvar = "Virus"
)
plot(result)

</code></pre>

<hr>
<h2 id='plot.dimRes'>Plot dimRes Object</h2><span id='topic+plot.dimRes'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'dimRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dimRes'
plot(
  x,
  omicsData = NULL,
  color_by = NULL,
  shape_by = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 0,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  point_size = 4,
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.dimRes_+3A_x">x</code></td>
<td>
<p>object of class dimRes created by the <code>dim_reduction</code>
function</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_omicsdata">omicsData</code></td>
<td>
<p>optional omicsData for use in specifying a column name in
fdata when using <code>color_by</code> or <code>shape_by</code>.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying which column to use to control 
the color for plotting. NULL indicates the default value of the main effect 
levels (if present). &quot;Group&quot; uses the &quot;Group&quot; column of group_DF. NA 
indicates no column will be used, and will use the default theme color. If 
an omicsData object is passed, any other value will use the specified 
column of f_data.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_shape_by">shape_by</code></td>
<td>
<p>character string specifying which column to use to control 
the shape for plotting. NULL indicates the default value of the second main
effect levels (if present). &quot;Group&quot; uses the &quot;Group&quot; column of group_DF. NA 
indicates no column will be used, and will use the default theme shape. If 
an omicsData object is passed, any other value will use the specified 
column of f_data.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_point_size">point_size</code></td>
<td>
<p>An integer specifying the size of the points. The default
is 4.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.dimRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)

mylipid &lt;- edata_transform(omicsData = lipid_neg_object, data_scale = "log2")
mylipid &lt;- group_designation(omicsData = mylipid, main_effects = "Virus")
pca_lipids &lt;- dim_reduction(omicsData = mylipid)
plot(pca_lipids)


myseq &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
pca_seq &lt;- dim_reduction(omicsData = myseq)
plot(pca_seq)


</code></pre>

<hr>
<h2 id='plot.imdanovaFilt'>Plot imdanovaFilt Object</h2><span id='topic+plot.imdanovaFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'imdanovaFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'imdanovaFilt'
plot(
  x,
  min_nonmiss_anova = NULL,
  min_nonmiss_gtest = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 0,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  point_size = 3,
  line_size = 0.75,
  text_size = 3,
  bw_theme = TRUE,
  palette = NULL,
  display_count = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.imdanovaFilt_+3A_x">x</code></td>
<td>
<p>Object of class imdanovaFilt (also a data frame) containing
the molecule identifier and number of samples in each group with
non-missing values for that molecule</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_min_nonmiss_anova">min_nonmiss_anova</code></td>
<td>
<p>An integer indicating the minimum number of
non-missing feature values allowed per group for <code>anova_filter</code>.
Suggested value is 2.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_min_nonmiss_gtest">min_nonmiss_gtest</code></td>
<td>
<p>An integer indicating the minimum number of
non-missing feature values allowed per group for <code>gtest_filter</code>.
Suggested value is 3.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_point_size">point_size</code></td>
<td>
<p>integer specifying the size of the points. The default
is 3.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_line_size">line_size</code></td>
<td>
<p>integer specifying the thickness of the line. The default
is 0.75.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_text_size">text_size</code></td>
<td>
<p>integer specifying the size of the text (number of
biomolecules per group). The default is 3.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the missing value counts by
sample will be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.imdanovaFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
data(pep_object)
mypep &lt;- group_designation(omicsData = pep_object, main_effects = "Phenotype")
to_filter &lt;- imdanova_filter(omicsData = mypep)
plot(to_filter, min_nonmiss_anova = 2, min_nonmiss_gtest = 3)

</code></pre>

<hr>
<h2 id='plot.isobaricnormRes'>Plot isobaricnormRes object</h2><span id='topic+plot.isobaricnormRes'></span>

<h3>Description</h3>

<p>Creates box plots for an S3 object of type 'isobaricnormRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'isobaricnormRes'
plot(
  x,
  order = FALSE,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = NULL,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "none",
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.isobaricnormRes_+3A_x">x</code></td>
<td>
<p>an object of type isobaricnormRes, created by
<code><a href="#topic+normalize_isobaric">normalize_isobaric</a></code> with apply_norm = FALSE</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_order">order</code></td>
<td>
<p>logical value. If TRUE the samples will be ordered by the column
of f_data containing the experiment/plate information.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white
theme.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.isobaricnormRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myiso &lt;- edata_transform(omicsData = isobaric_object, data_scale = "log2")
result &lt;- normalize_isobaric(myiso,
  exp_cname = "Plex",
  apply_norm = FALSE,
  refpool_cname = "Virus",
  refpool_notation = "Pool"
)
plot(result)

</code></pre>

<hr>
<h2 id='plot.isobaricpepData'>Plot isobaricpepData Object</h2><span id='topic+plot.isobaricpepData'></span>

<h3>Description</h3>

<p>For plotting isobaricpepData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'isobaricpepData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.isobaricpepData_+3A_x">x</code></td>
<td>
<p>An isobaricpepData object</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.isobaricpepData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myiso &lt;- edata_transform(omicsData = isobaric_object, data_scale = "log2")
plot(myiso)

</code></pre>

<hr>
<h2 id='plot.lipidData'>Plot lipidData Object</h2><span id='topic+plot.lipidData'></span>

<h3>Description</h3>

<p>For plotting lipidData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lipidData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.lipidData_+3A_x">x</code></td>
<td>
<p>lipidData object</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.lipidData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mylipid &lt;- edata_transform(omicsData = lipid_pos_object, data_scale = "log2")
plot(mylipid, order_by = "Virus", color_by = "Virus")

</code></pre>

<hr>
<h2 id='plot.metabData'>Plot metabData Object</h2><span id='topic+plot.metabData'></span>

<h3>Description</h3>

<p>For plotting metabData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metabData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.metabData_+3A_x">x</code></td>
<td>
<p>metabData object</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.metabData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
plot(mymetab, order_by = "Phenotype", color_by = "Phenotype")

</code></pre>

<hr>
<h2 id='plot.moleculeFilt'>Plot moleculeFilt Object</h2><span id='topic+plot.moleculeFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'moleculeFilt':
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'moleculeFilt'
plot(
  x,
  min_num = NULL,
  cumulative = TRUE,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 0,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  text_size = 3,
  bar_width = 0.8,
  bw_theme = TRUE,
  palette = NULL,
  display_count = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.moleculeFilt_+3A_x">x</code></td>
<td>
<p>object of class moleculeFilt that contains the molecule
identifier and the number of samples for which the molecule was measured
(not NA)</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_min_num">min_num</code></td>
<td>
<p>An integer specifying the minimum number of samples in which a
biomolecule must appear. If a value is specified, a horizontal line will be
drawn when <code>cumulative=TRUE</code>, and bars will be colored appropriately
if <code>cumulative=FALSE</code>. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_cumulative">cumulative</code></td>
<td>
<p>logical indicating whether the number of biomolecules
observed in <em>at least</em> (TRUE) x number of samples or <em>exactly</em>
(FALSE) x number of samples should be plotted.  Defaults to TRUE.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_text_size">text_size</code></td>
<td>
<p>integer specifying the size of the text (number of
biomolecules by sample) within the bar plot. The default is 3.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_bar_width">bar_width</code></td>
<td>
<p>integer indicating the width of the bars in the bar plot.
The default is 0.8.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the missing value counts by
sample will be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.moleculeFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
data(pep_object)
molfilt &lt;- molecule_filter(omicsData = pep_object)
plot(molfilt, min_num = 5)
plot(molfilt, min_num = 3, cumulative = FALSE)

</code></pre>

<hr>
<h2 id='plot.naRes'>Plot naRes Object</h2><span id='topic+plot.naRes'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'naRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'naRes'
plot(
  x,
  omicsData,
  plot_type = "bar",
  nonmissing = FALSE,
  proportion = FALSE,
  order_by = NULL,
  color_by = NULL,
  interactive = FALSE,
  x_lab_bar = NULL,
  x_lab_scatter = NULL,
  y_lab_bar = NULL,
  y_lab_scatter = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 60,
  title_lab_bar = NULL,
  title_lab_scatter = NULL,
  title_lab_size = 14,
  legend_lab_bar = NULL,
  legend_lab_scatter = NULL,
  legend_position = "right",
  point_size = 2,
  text_size = 3,
  bar_width = 0.8,
  bw_theme = TRUE,
  palette = NULL,
  display_count = TRUE,
  coordinate_flip = FALSE,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.naRes_+3A_x">x</code></td>
<td>
<p>list of two data frames, one containing the number of
missing values by sample, and the other containing missing values by
molecule</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_omicsdata">omicsData</code></td>
<td>
<p>object of class 'pepData', 'proData', 'metabData',
'lipidData', nmrData', or 'seqData', created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_plot_type">plot_type</code></td>
<td>
<p>character string specifying which type of plot to produce.
The two options are 'bar' or 'scatter'.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_nonmissing">nonmissing</code></td>
<td>
<p>logical value. When FALSE, plots missing values. When TRUE,
plots non-missing values.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_proportion">proportion</code></td>
<td>
<p>logical value. When TRUE, plots the proportion of missing 
(or non-missing if <code>nonmissing</code> is TRUE) to the total number of 
values. Only works with a plot type of 'bar'.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_order_by">order_by</code></td>
<td>
<p>A character string specifying a column in f_data by which to
order the samples. Specifying &quot;Group&quot; will use the &quot;Group&quot; column of the
object's <code>group_DF</code> attribute to order the samples. Only works with a 
plot type of 'bar'.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_color_by">color_by</code></td>
<td>
<p>A character string specifying a column in f_data by which to
color the bars or the points depending on the <code>plot_type</code>. Specifying
&quot;Group&quot; will use the &quot;Group&quot; column of the object's <code>group_DF</code>
attribute to color the samples. Only works with a plot type of 'bar'.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_x_lab_bar">x_lab_bar</code></td>
<td>
<p>character string used for the x-axis label for the bar
plot</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_x_lab_scatter">x_lab_scatter</code></td>
<td>
<p>character string used for the x-axis label for the
scatter plot</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_y_lab_bar">y_lab_bar</code></td>
<td>
<p>character string used for the y-axis label for the bar
plot</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_y_lab_scatter">y_lab_scatter</code></td>
<td>
<p>character string used for the y-axis label for the
scatter plot</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_title_lab_bar">title_lab_bar</code></td>
<td>
<p>character string used for the plot title when
<code>plot_type</code> is 'bar'.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_title_lab_scatter">title_lab_scatter</code></td>
<td>
<p>character string used for the plot title when
<code>plot_type</code> is 'scatter'.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_legend_lab_bar">legend_lab_bar</code></td>
<td>
<p>character string specifying the legend title when
creating a bar plot.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_legend_lab_scatter">legend_lab_scatter</code></td>
<td>
<p>character string specifying the legend title when
creating a scatter plot.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, or &quot;bottom&quot;. The default is
&quot;right&quot;.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_point_size">point_size</code></td>
<td>
<p>An integer specifying the size of the points. The default
is 2.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_text_size">text_size</code></td>
<td>
<p>An integer specifying the size of the text (number of
missing values by sample) within the bar plot. The default is 3.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_bar_width">bar_width</code></td>
<td>
<p>An integer indicating the width of the bars in the bar plot.
The default is 0.8.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the missing value
counts by sample will be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_coordinate_flip">coordinate_flip</code></td>
<td>
<p>logical value. Indicates whether the x and y axes will
be flipped. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.naRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes in an object of class naRes and creates either a
bar or scatter plot of missing values. When plot_type = 'bar', a sample
name by missing values count bar chart is returned. When plot_type =
'scatter' a mean intensity vs number of missing values (per molecule)
scatter plot is returned. Note: If the omicsData object has had
<code><a href="#topic+group_designation">group_designation</a></code> applied to it, the points in the plot will
be colored by group.
</p>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mylipid &lt;- group_designation(omicsData = lipid_neg_object, main_effects = "Virus")
result &lt;- missingval_result(omicsData = mylipid)
plot(result, omicsData = mylipid, plot_type = "bar",
     x_lab_angle = 50, order_by = "Virus", color_by = "Virus")
plot(result, omicsData = mylipid, plot_type = "scatter",
     x_lab_angle = 50, color_by = "Virus")

result &lt;- missingval_result(omicsData = rnaseq_object)
plot(result, omicsData = rnaseq_object, plot_type = "bar")

</code></pre>

<hr>
<h2 id='plot.nmrData'>Plot nmrData Object</h2><span id='topic+plot.nmrData'></span>

<h3>Description</h3>

<p>For plotting nmrData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nmrData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.nmrData_+3A_x">x</code></td>
<td>
<p>nmrData object</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>An optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_ylimit">ylimit</code></td>
<td>
<p>A numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.nmrData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mynmr &lt;- edata_transform(omicsData = nmr_identified_object, data_scale = "log2")
plot(mynmr)

</code></pre>

<hr>
<h2 id='plot.nmrnormRes'>Plot nmrnormRes Object</h2><span id='topic+plot.nmrnormRes'></span>

<h3>Description</h3>

<p>Creates a scatter plot for an S3 object of type 'nmrnormRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nmrnormRes'
plot(
  x,
  nmrData = NULL,
  order_by = NULL,
  color_by = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "none",
  point_size = 2,
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.nmrnormRes_+3A_x">x</code></td>
<td>
<p>an object of type nmrnormRes, created by
<code><a href="#topic+normalize_nmr">normalize_nmr</a></code></p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_nmrdata">nmrData</code></td>
<td>
<p>An nmrData object.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_order_by">order_by</code></td>
<td>
<p>A character string specifying a column in f_data by which to
order the samples.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_color_by">color_by</code></td>
<td>
<p>A character string specifying a column in f_data by which to
color the points.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_point_size">point_size</code></td>
<td>
<p>integer specifying the size of the points. The default is
2.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white
theme.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.nmrnormRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mynmr &lt;- edata_transform(omicsData = nmr_identified_object, data_scale = "log2")
mynmrnorm &lt;- normalize_nmr(
  omicsData = mynmr,
  apply_norm = FALSE,
  metabolite_name = "unkm1.53"
)
plot(mynmrnorm)

mynmrnorm2 &lt;- normalize_nmr(
  omicsData = mynmr,
  apply_norm = FALSE,
  sample_property_cname = "Concentration"
)
plot(mynmrnorm2)

</code></pre>

<hr>
<h2 id='plot.normRes'>Plot normRes Object</h2><span id='topic+plot.normRes'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'normRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'normRes'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.normRes_+3A_x">x</code></td>
<td>
<p>normRes object created by the normalize_global function</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_facet_cols">facet_cols</code></td>
<td>
<p>optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.normRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(
  omicsData = metab_object,
  data_scale = "log2"
)
mymetab &lt;- group_designation(
  omicsData = mymetab,
  main_effects = "Phenotype"
)
norm_object &lt;- normalize_global(
  omicsData = mymetab,
  subset_fn = "all",
  norm_fn = "median"
)
plot(norm_object, order_by = "Phenotype", color_by = "Phenotype")

</code></pre>

<hr>
<h2 id='plot.pepData'>Plot pepData Object</h2><span id='topic+plot.pepData'></span>

<h3>Description</h3>

<p>For plotting pepData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pepData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.pepData_+3A_x">x</code></td>
<td>
<p>pepData object</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>An optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.pepData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
data(pep_object)
mypep &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")
plot(mypep, order_by = "Phenotype", color_by = "Phenotype")

</code></pre>

<hr>
<h2 id='plot.proData'>Plot proData Object</h2><span id='topic+plot.proData'></span>

<h3>Description</h3>

<p>For plotting proData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'proData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.proData_+3A_x">x</code></td>
<td>
<p>proData object</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.proData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
plot(pro_object, order_by = "Phenotype", color_by = "Phenotype")

</code></pre>

<hr>
<h2 id='plot.proteomicsFilt'>Plot proteomicsFilt Object</h2><span id='topic+plot.proteomicsFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'proteomicsFilt':
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'proteomicsFilt'
plot(
  x,
  plot_type = "num_peps",
  min_num_peps = NULL,
  interactive = FALSE,
  x_lab_pep = NULL,
  x_lab_pro = NULL,
  y_lab_pep = NULL,
  y_lab_pro = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 0,
  title_lab_pep = NULL,
  title_lab_pro = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  text_size = 3,
  bar_width = 0.8,
  bw_theme = TRUE,
  palette = NULL,
  display_count = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.proteomicsFilt_+3A_x">x</code></td>
<td>
<p>object of class proteomicsFilt, which is a list with two
elements. The first element is a data frame of counts for each unique
peptide. The second element is a data frame with the counts for the number
of peptides that map to each unique protein.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_plot_type">plot_type</code></td>
<td>
<p>character string specifying the type of plot to be
displayed. The available options are &quot;num_peps&quot; or &quot;redundancy&quot;. If
&quot;num_peps&quot; the plot is displayed that shows the counts of proteins that
have a specific number of peptides mapping to them. If &quot;redundancy&quot; the
plot showing the counts of peptides that map to a specific number of
proteins is displayed.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_min_num_peps">min_num_peps</code></td>
<td>
<p>an optional integer value between 1 and the maximum
number of peptides that map to a protein in the data. The value specifies
the minimum number of peptides that must map to a protein. Any protein with
less than <code>min_num_peps</code> mapping to it will be returned as a protein
that should be filtered. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_x_lab_pep">x_lab_pep</code></td>
<td>
<p>character string used for the x-axis label for the
num_peps plot. The default is NULL in which case the default x-axis label
will be used.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_x_lab_pro">x_lab_pro</code></td>
<td>
<p>character string used for the x-axis label for the
redundancy plot. The default is NULL in which case the default x-axis label
will be used.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_y_lab_pep">y_lab_pep</code></td>
<td>
<p>character string used for the y-axis label for the
num_peps plot. The default is NULL in which case the default y-axis label
will be used.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_y_lab_pro">y_lab_pro</code></td>
<td>
<p>character string used for the y-axis label for the
redundancy plot. The default is NULL in which case the default y-axis label
will be used.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_title_lab_pep">title_lab_pep</code></td>
<td>
<p>character string specifying the num_peps plot title.
The default is NULL in which case the default title will be used.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_title_lab_pro">title_lab_pro</code></td>
<td>
<p>character string specifying the redundancy plot title.
The default is NULL in which case the default title will be used.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_text_size">text_size</code></td>
<td>
<p>An integer specifying the size of the text (number of
peptides or proteins depending on the plot) within the bar plot. The
default is 3.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_bar_width">bar_width</code></td>
<td>
<p>An integer indicating the width of the bars in the bar plot.
The default is 0.8.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the peptide or protein counts
will be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.proteomicsFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
data(pep_object)
my_filter &lt;- proteomics_filter(omicsData = pep_object)
plot(my_filter, min_num_peps = 3)
plot(my_filter, plot_type = "redundancy")

</code></pre>

<hr>
<h2 id='plot.rmdFilt'>Plot rmdFilt Object</h2><span id='topic+plot.rmdFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'rmdFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rmdFilt'
plot(
  x,
  pvalue_threshold = NULL,
  sampleID = NULL,
  order_by = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  point_size = 3,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.rmdFilt_+3A_x">x</code></td>
<td>
<p>object of class rmdFilt created via <code><a href="#topic+rmd_filter">rmd_filter</a></code></p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_pvalue_threshold">pvalue_threshold</code></td>
<td>
<p>numeric threshold for the Robust Mahalanobis Distance (RMD)
p-value. If <code>sampleID</code> is NULL (see <code>sampleID</code> below), a
horizontal line is plotted at the RMD value that corresponds with the
threshold, and all samples above the line have a p-value below the
threshold. If <code>sampleID</code> is not NULL, <code>pvalue_threshold</code> will do
nothing. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_sampleid">sampleID</code></td>
<td>
<p>character vector specifying the sample names to be plotted.
If specified, the plot function produces a boxplot instead of a
scatterplot. A point, colored by sample, will be placed on each boxplot for
that sample's value for the given metric. The default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying a variable by which to order
the samples in the plot. This variable must be found in the column names of
fdata from the rmdFilt object. If <code>order_by</code> is &quot;Group&quot;, the plot will
be ordered by the group variable from the group_designation function. If
NULL (default), the samples will be displayed in the order in which they
first appear.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 90.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_point_size">point_size</code></td>
<td>
<p>An integer specifying the size of the points. The default
is 3.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.rmdFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
mymetab &lt;- group_designation(omicsData = mymetab, main_effects = "Phenotype")
rmd_results &lt;- rmd_filter(omicsData = mymetab, metrics = c("MAD", "Skewness", "Correlation"))
plot(rmd_results, pvalue_threshold = 0.0001, order_by = "Phenotype")

</code></pre>

<hr>
<h2 id='plot.RNAFilt'>Plot RNAFilt Object</h2><span id='topic+plot.RNAFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'RNAFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RNAFilt'
plot(
  x,
  plot_type = "library",
  size_library = NULL,
  min_nonzero = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = "",
  legend_position = "right",
  text_size = 3,
  bar_width = 0.8,
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.RNAFilt_+3A_x">x</code></td>
<td>
<p>object of class RNAFilt that contains the sample
identifier, library size, number of non-zero biomolecules, and proportion
of non-zero biomolecules</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_plot_type">plot_type</code></td>
<td>
<p>character string, specified as &quot;library&quot; or &quot;biomolecule&quot;.
&quot;library&quot; displays library size for each sample, &quot;biomolecule&quot; displays the
number of unique biomolecules with non-zero counts per sample.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_size_library">size_library</code></td>
<td>
<p>integer cut-off for sample library size (i.e. number
of reads). Defaults to NULL.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_min_nonzero">min_nonzero</code></td>
<td>
<p>integer or float between 0 and 1. Cut-off for number of
unique biomolecules with non-zero counts or as a proportion of total
biomolecules. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_text_size">text_size</code></td>
<td>
<p>An integer specifying the size of the text (number of
biomolecules by sample) within the bar plot. The default is 3.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_bar_width">bar_width</code></td>
<td>
<p>An integer indicating the width of the bars in the bar plot.
The default is 0.8.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.RNAFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
seqfilt &lt;- RNA_filter(omicsData = rnaseq_object)
plot(seqfilt)

</code></pre>

<hr>
<h2 id='plot.seqData'>Plot seqData Object</h2><span id='topic+plot.seqData'></span>

<h3>Description</h3>

<p>For plotting seqData S3 objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'seqData'
plot(
  x,
  order_by = NULL,
  color_by = NULL,
  facet_by = NULL,
  facet_cols = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 90,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  ylimit = NULL,
  bw_theme = TRUE,
  palette = NULL,
  use_VizSampNames = FALSE,
  transformation = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.seqData_+3A_x">x</code></td>
<td>
<p>seqData object</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_order_by">order_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to order the boxplots. If <code>order_by</code> is &quot;Group&quot;, the boxplots
will be ordered by the group variable from the group_designation function.
If NULL (default), the boxplots will be displayed in the order they appear
in the data.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_color_by">color_by</code></td>
<td>
<p>character string specifying the column name of f_data by
which to color the boxplots. If <code>color_by</code> is &quot;Group&quot;, the boxplots
will be colored by the group variable from the group_designation function.
If NULL (default), the boxplots will have one default color.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_facet_by">facet_by</code></td>
<td>
<p>character string specifying the column name of f_data with
which to create a facet plot. Default value is NULL.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_facet_cols">facet_cols</code></td>
<td>
<p>optional integer specifying the number of columns to
show in the facet plot.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_ylimit">ylimit</code></td>
<td>
<p>numeric vector of length 2 specifying y-axis lower and upper
limits.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_use_vizsampnames">use_VizSampNames</code></td>
<td>
<p>logical value. Indicates whether to use custom sample
names. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_transformation">transformation</code></td>
<td>
<p>character string. String of length 1 defining a
transformation for visualizing count data. Valid options are 'lcpm',
'upper', and 'median'. 'lcpm' - For each column: scale column intensities by
(total column sum/1 million), then log2 transform. 'median' - For each
column: scale column intensities by median column intensities, then
back-transform to original scale. 'upper' - For each column: scale column
intensities by 75th quantile column intensities, then back-transform to
original scale. For 'median' and 'upper' options, all zeros are converted
to NAs.</p>
</td></tr>
<tr><td><code id="plot.seqData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
plot(rnaseq_object, transformation = "lcpm")

</code></pre>

<hr>
<h2 id='plot.SPANSRes'>Plot SPANSRes Object</h2><span id='topic+plot.SPANSRes'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'SPANSRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SPANSRes'
plot(
  x,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = NULL,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  color_low = NULL,
  color_high = NULL,
  bw_theme = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.SPANSRes_+3A_x">x</code></td>
<td>
<p>an object of the class 'SPANSRes', created by
<code><a href="#topic+spans_procedure">spans_procedure</a></code></p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is NULL.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_color_low">color_low</code></td>
<td>
<p>character string specifying the color of the gradient for
low values.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_color_high">color_high</code></td>
<td>
<p>character string specifying the color of the gradient for
high values</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.SPANSRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)
data(pep_object)
mypep &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")
mypep &lt;- group_designation(omicsData = mypep, main_effects = "Phenotype")
myspans &lt;- spans_procedure(omicsData = mypep)
plot(myspans)


</code></pre>

<hr>
<h2 id='plot.statRes'>Plot statRes Object</h2><span id='topic+plot.statRes'></span>

<h3>Description</h3>

<p>Produces plots that summarize the results contained in a 'statRes' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'statRes'
plot(
  x,
  plot_type = "bar",
  fc_threshold = NULL,
  fc_colors = c("red", "black", "green"),
  stacked = TRUE,
  show_sig = TRUE,
  color_low = NULL,
  color_high = NULL,
  plotly_layout = NULL,
  interactive = FALSE,
  x_lab = NULL,
  x_lab_size = 11,
  x_lab_angle = NULL,
  y_lab = NULL,
  y_lab_size = 11,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  text_size = 3,
  bw_theme = TRUE,
  display_count = TRUE,
  custom_theme = NULL,
  cluster = FALSE,
  free_y_axis = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.statRes_+3A_x">x</code></td>
<td>
<p>'statRes' object to be plotted, usually the result of 'imd_anova'</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_plot_type">plot_type</code></td>
<td>
<p>defines which plots to be produced, options are &quot;bar&quot;,
&quot;volcano&quot;, &quot;gheatmap&quot;, &quot;fcheatmap&quot;; defaults to &quot;bar&quot;.  See details for
plot descriptions.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_fc_threshold">fc_threshold</code></td>
<td>
<p>optional threshold value for fold change estimates.
Modifies the volcano plot as follows:  Vertical lines are added at
(+/-)<code>fc_threshold</code> and all observations that have absolute fold
change less than <code>abs(fc_threshold)</code> are colored as 'non-significant'
(as specified by <code>fc_colors</code>).</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_fc_colors">fc_colors</code></td>
<td>
<p>vector of length three with character color values
interpretable by ggplot. i.e. c(&quot;orange&quot;, &quot;black&quot;, &quot;blue&quot;) with the values
being used to color negative, non-significant, and positive fold changes
respectively</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_stacked">stacked</code></td>
<td>
<p>TRUE/FALSE for whether to stack positive and negative fold
change sections in the barplot, defaults to TRUE</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_show_sig">show_sig</code></td>
<td>
<p>This input is used when <code>plot_type = "gheatmap"</code>. A
logical value. If TRUE a visual indicator that a certain bin combination is
significant by the g-test is shown.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_color_low">color_low</code></td>
<td>
<p>This input is used when <code>plot_type = "gheatmap"</code>. A
character string specifying the color of the gradient for low count values.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_color_high">color_high</code></td>
<td>
<p>This input is used when <code>plot_type = "gheatmap"</code>. A
character string specifying the color of the gradient for high count
values.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_plotly_layout">plotly_layout</code></td>
<td>
<p>This input is used when <code>plot_type = "gheatmap"</code>. A
list of arguments, not including the plot, to be passed to
<code>plotly::layout</code> if <code>interactive = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_interactive">interactive</code></td>
<td>
<p>TRUE/FALSE for whether to create an interactive plot using
plotly. Not valid for all plots.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis. The
default is 11.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_text_size">text_size</code></td>
<td>
<p>integer specifying the size of the text (number of
non-missing values) within the plot. The default is 3.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white
theme.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the non-missing counts
will be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_custom_theme">custom_theme</code></td>
<td>
<p>a ggplot 'theme' object to be applied to non-interactive
plots, or those converted by plotly::ggplotly().</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_cluster">cluster</code></td>
<td>
<p>logical for heatmaps; TRUE will cluster biomolecules on X
axis. defaults to TRUE for seqData statistics and FALSE for all others.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_free_y_axis">free_y_axis</code></td>
<td>
<p>Logical. If TRUE the y axis for each bar plot can have its
own range. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.statRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot types:
</p>

<ul>
<li><p>&quot;bar&quot; <code>?pmartR::statres_barplot</code> Bar-chart with bar heights
indicating the number of significant biomolecules, grouped by test type and
fold change direction.
</p>
</li>
<li><p>&quot;volcano&quot; <code>?pmartR::statres_volcano_plot</code> Scatter plot showing
negative-log-pvalues against fold change. Colored by statistical
significance and fold change.
</p>
</li>
<li><p>&quot;gheatmap&quot; <code>?pmartR::gtest_heatmap</code> Heatmap with x and y axes
indicating the number of nonmissing values for two groups. Colored by
number of biomolecules that fall into that combination of nonmissing values.
</p>
</li>
<li><p>&quot;fcheatmap&quot; Heatmap showing all biomolecules across comparisons,
colored by fold change.
</p>
</li></ul>



<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
# Group the data by condition
mypro &lt;- group_designation(
  omicsData = pro_object,
  main_effects = c("Phenotype")
)

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = mypro)
mypro &lt;- applyFilt(
  filter_object = imdanova_Filt,
  omicsData = mypro,
  min_nonmiss_anova = 2
)

# Implement the IMD ANOVA method and compuate all pairwise comparisons
# (i.e. leave the `comparisons` argument NULL)
anova_res &lt;- imd_anova(omicsData = mypro, test_method = 'anova')
plot(anova_res)
plot(anova_res, plot_type = "volcano")

imd_res &lt;- imd_anova(omicsData = mypro, test_method = 'gtest')
plot(imd_res)

imd_anova_res &lt;- imd_anova(
  omicsData = mypro,
  test_method = 'comb',
  pval_adjust_a_multcomp = 'bon',
  pval_adjust_g_multcomp = 'bon'
)
plot(imd_anova_res, bw_theme = TRUE)
plot(imd_anova_res, plot_type = "volcano", bw_theme = TRUE)

</code></pre>

<hr>
<h2 id='plot.totalCountFilt'>Plot totalCountFilt Object</h2><span id='topic+plot.totalCountFilt'></span>

<h3>Description</h3>

<p>For plotting an S3 object of type 'totalCountFilt':
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'totalCountFilt'
plot(
  x,
  min_count = NULL,
  interactive = FALSE,
  x_lab = NULL,
  y_lab = NULL,
  x_lab_size = 11,
  y_lab_size = 11,
  x_lab_angle = 0,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = "",
  legend_position = "right",
  text_size = 3,
  bar_width = 0.8,
  bw_theme = TRUE,
  palette = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.totalCountFilt_+3A_x">x</code></td>
<td>
<p>object of class totalCountFilt that contains the molecule
identifier and the number of total counts for which the molecule was measured
(not NA).</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_min_count">min_count</code></td>
<td>
<p>integer specifying the minimum number of samples in which a
biomolecule must appear. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_interactive">interactive</code></td>
<td>
<p>logical value. If TRUE produces an interactive plot.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_x_lab_size">x_lab_size</code></td>
<td>
<p>integer value indicating the font size for the x-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_y_lab_size">y_lab_size</code></td>
<td>
<p>integer value indicating the font size for the y-axis.
The default is 11.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_x_lab_angle">x_lab_angle</code></td>
<td>
<p>integer value indicating the angle of x-axis labels.
The default is 0.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_title_lab_size">title_lab_size</code></td>
<td>
<p>integer value indicating the font size of the plot
title. The default is 14.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_legend_position">legend_position</code></td>
<td>
<p>character string specifying the position of the
legend. Can be one of &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;, or &quot;none&quot;. The
default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_text_size">text_size</code></td>
<td>
<p>integer specifying the size of the text (number of
biomolecules by sample) within the bar plot. The default is 3.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_bar_width">bar_width</code></td>
<td>
<p>integer indicating the width of the bars in the bar plot.
The default is 0.8.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_bw_theme">bw_theme</code></td>
<td>
<p>logical value. If TRUE uses the ggplot2 black and white theme.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_palette">palette</code></td>
<td>
<p>character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code><a href="RColorBrewer.html#topic+RColorBrewer">RColorBrewer</a></code>.</p>
</td></tr>
<tr><td><code id="plot.totalCountFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot object if interactive is FALSE, or plotly plot object if
interactive is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)
seqfilt &lt;- total_count_filter(omicsData = rnaseq_object)
plot(seqfilt, min_count = 15)


</code></pre>

<hr>
<h2 id='pmartR'>Panomics Marketplace - Quality Control and Statistical Analysis for Panomics Data</h2><span id='topic+pmartR'></span>

<h3>Description</h3>

<p>Provides functionality for quality control processing and
statistical analysis of mass spectrometry (MS) omics data, in particular 
proteomic (either at the peptide or the protein level), lipidomic, and 
metabolomic data, as well as RNA-seq based count data and nuclear magnetic
resonance (NMR) data. This includes data transformation, specification of 
groups that are to be compared against each other, filtering of features 
and/or samples, data normalization, data summarization (correlation, PCA), 
and statistical comparisons between defined groups.
</p>


<h3>Value</h3>

<p>No return value, used to appease R CMD check.
</p>

<hr>
<h2 id='pmartR_filter_worker'>Remove items that need to be filtered out</h2><span id='topic+pmartR_filter_worker'></span>

<h3>Description</h3>

<p>This function removes rows and columns in e_data, f_data, and e_meta based
on either remove or keep criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pmartR_filter_worker(filter_object, omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pmartR_filter_worker_+3A_filter_object">filter_object</code></td>
<td>
<p>A list of three elements. Each element contains a set of
names to either remove or keep from e_data, f_data, and e_meta.</p>
</td></tr>
<tr><td><code id="pmartR_filter_worker_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class <code>pepData</code>, <code>proData</code>,
<code>lipidData</code>, or <code>metabData</code>, usually created by
<code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.metabData">as.metabData</a></code>, respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three elements: first is the filtered e_data object,
second is the filtered f_data object, and third is the filtered
e_meta object.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton, Lisa Bramer
</p>

<hr>
<h2 id='ppp'>Identify Biomolecules from the Proportion Present (PPP) for Use in
Normalization</h2><span id='topic+ppp'></span>

<h3>Description</h3>

<p>Selects biomolecules for normalization via the method of percentage of the
peptides (or proteins, metabolites, etc.) present (PPP)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppp(e_data, edata_id, proportion = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ppp_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, proteins, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="ppp_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="ppp_+3A_proportion">proportion</code></td>
<td>
<p>numeric value between 0 and 1, indicating the proportion at
or above which a biomolecule must be present across all samples in order to be
retained (default value 0.5)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biomolecules present across <code>proportion</code> samples are designated as
PPP.
</p>


<h3>Value</h3>

<p>Character vector containing the biomolecules belonging to the PPP
subset.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>

<hr>
<h2 id='ppp_rip'>Identify Proportion of Peptides Present (PPP) and Rank Invariant Peptides
(RIP) for Use in Normalization</h2><span id='topic+ppp_rip'></span>

<h3>Description</h3>

<p>Selects biomolecules for normalization via the method of proportion of
biomolecules present and rank invariant biomolecules (ppp_rip)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppp_rip(e_data, edata_id, fdata_id, groupDF, alpha = 0.2, proportion = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ppp_rip_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, proteins, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="ppp_rip_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="ppp_rip_+3A_fdata_id">fdata_id</code></td>
<td>
<p>character string indicating the name of the sample column
name in f_data.</p>
</td></tr>
<tr><td><code id="ppp_rip_+3A_groupdf">groupDF</code></td>
<td>
<p>data.frame created by <code>group_designation</code> with columns
for sample.id and group. If two main effects are provided the original main
effect levels for each sample are returned as the third and fourth columns
of the data.frame.</p>
</td></tr>
<tr><td><code id="ppp_rip_+3A_alpha">alpha</code></td>
<td>
<p>numeric p-value threshold, above which the biomolecules are retained
as rank invariant (default value 0.25)</p>
</td></tr>
<tr><td><code id="ppp_rip_+3A_proportion">proportion</code></td>
<td>
<p>numeric value between 0 and 1, indicating the percentage at
or above which a biomolecule must be present across all samples in order to be
retained (default value 0.5)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biomolecules present across <code>proportion</code> samples are subjected to a
Kruskal-Wallis test (non-parametric one-way ANOVA, where NAs are ignored)
on group membership, and those biomolecules with p-value greater than a
defined threshold <code>alpha</code> (common values include 0.1 or 0.25) are
retained as rank-invariant biomolecules.
</p>


<h3>Value</h3>

<p>Character vector containing the biomolecules belonging to the ppp_rip subset.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>

<hr>
<h2 id='pquant'>Protein Quantitation using Mean or Median Peptide Abundances</h2><span id='topic+pquant'></span>

<h3>Description</h3>

<p>This function takes in a pepData object and returns a proData object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pquant(pepData, combine_fn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pquant_+3A_pepdata">pepData</code></td>
<td>
<p>omicsData object of class 'pepData'</p>
</td></tr>
<tr><td><code id="pquant_+3A_combine_fn">combine_fn</code></td>
<td>
<p>A character string that can either be 'mean' or 'median'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An omicsData object of class 'proData'
</p>

<hr>
<h2 id='pre_imdanova_melt'>Create a Melted and Grouped Version of e_data for IMD_ANOVA filter</h2><span id='topic+pre_imdanova_melt'></span>

<h3>Description</h3>

<p>This function creates a melted version of e_data, grouped by edata_id and
group designation, for future use of implementing a IMD_ANOVA filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_imdanova_melt(e_data, groupDF, samp_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre_imdanova_melt_+3A_e_data">e_data</code></td>
<td>
<p><code class="reqn">p \times n</code> data.frame, where <code class="reqn">p</code> is the number of
peptides, proteins, lipids, metabolites, or accessions and <code class="reqn">n</code> is the
number of samples</p>
</td></tr>
<tr><td><code id="pre_imdanova_melt_+3A_groupdf">groupDF</code></td>
<td>
<p>data frame created by <code>group_designation</code> with columns
for the sample identifier and the designated group.</p>
</td></tr>
<tr><td><code id="pre_imdanova_melt_+3A_samp_id">samp_id</code></td>
<td>
<p>character string specifying the name of the column containing
the sample identifiers in <code>groupDF</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame of class &quot;grouped_dt&quot; which is compatible with functions
in the dplyr package
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>

<hr>
<h2 id='prep_flags'>Extract flag columns from a statRes object</h2><span id='topic+prep_flags'></span>

<h3>Description</h3>

<p>Changes the flags columns from a statRes object into a format that the
statRes plot funcitons can handle. pmartR is an unruly beast that cannot be
tamed!!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_flags(x, test)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prep_flags_+3A_x">x</code></td>
<td>
<p>A statRes object.</p>
</td></tr>
<tr><td><code id="prep_flags_+3A_test">test</code></td>
<td>
<p>character string indicating the type of test run.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the sample IDs and significance flags from a statistical test.
</p>

<hr>
<h2 id='print.customFilt'>print.customFilt</h2><span id='topic+print.customFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'customFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'customFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.customFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'customFilt'</p>
</td></tr>
<tr><td><code id="print.customFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.customFilterSummary'>Custom Filter Print Method</h2><span id='topic+print.customFilterSummary'></span>

<h3>Description</h3>

<p>Print method for summary of custom filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'customFilterSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.customFilterSummary_+3A_x">x</code></td>
<td>
<p>the custom filter summary to print</p>
</td></tr>
<tr><td><code id="print.customFilterSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.cvFilt'>print.cvFilt</h2><span id='topic+print.cvFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'cvFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cvFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.cvFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'cvFilt'</p>
</td></tr>
<tr><td><code id="print.cvFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.cvFilterSummary'>CV Filter Print Method</h2><span id='topic+print.cvFilterSummary'></span>

<h3>Description</h3>

<p>Print method for summary of CV filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cvFilterSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.cvFilterSummary_+3A_x">x</code></td>
<td>
<p>the CV filter summary to print</p>
</td></tr>
<tr><td><code id="print.cvFilterSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.dataRes'>print.dataRes</h2><span id='topic+print.dataRes'></span>

<h3>Description</h3>

<p>For printing an S3 object of class 'dataRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dataRes'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.dataRes_+3A_x">x</code></td>
<td>
<p>An object of class 'dataRes'</p>
</td></tr>
<tr><td><code id="print.dataRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x.
</p>

<hr>
<h2 id='print.imdanovaFilt'>print.imdanovaFilt</h2><span id='topic+print.imdanovaFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'imdanovaFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'imdanovaFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.imdanovaFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'imdanovaFilt'</p>
</td></tr>
<tr><td><code id="print.imdanovaFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.imdanovaFilterSummary'>IMD-ANOVA Filter Print Method</h2><span id='topic+print.imdanovaFilterSummary'></span>

<h3>Description</h3>

<p>Print method for summary of imdanova filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'imdanovaFilterSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.imdanovaFilterSummary_+3A_x">x</code></td>
<td>
<p>the imdanova filter summary to print</p>
</td></tr>
<tr><td><code id="print.imdanovaFilterSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.lipidData'>print.lipidData</h2><span id='topic+print.lipidData'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'lipidData'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lipidData'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.lipidData_+3A_x">x</code></td>
<td>
<p>An object of type 'lipidData'</p>
</td></tr>
<tr><td><code id="print.lipidData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.metabData'>print.metabData</h2><span id='topic+print.metabData'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'metabData'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metabData'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.metabData_+3A_x">x</code></td>
<td>
<p>An object of type 'metabData'</p>
</td></tr>
<tr><td><code id="print.metabData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.moleculeFilt'>print.moleculeFilt</h2><span id='topic+print.moleculeFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'moleculeFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'moleculeFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.moleculeFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'moleculeFilt'</p>
</td></tr>
<tr><td><code id="print.moleculeFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.moleculeFilterSummary'>Molecule Filter Print Method</h2><span id='topic+print.moleculeFilterSummary'></span>

<h3>Description</h3>

<p>Print method for moleculeFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'moleculeFilterSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.moleculeFilterSummary_+3A_x">x</code></td>
<td>
<p>the moleculeFilt summary to print</p>
</td></tr>
<tr><td><code id="print.moleculeFilterSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.normRes'>print.normRes</h2><span id='topic+print.normRes'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'normRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'normRes'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.normRes_+3A_x">x</code></td>
<td>
<p>An object of type 'normRes'</p>
</td></tr>
<tr><td><code id="print.normRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.pepData'>print.pepData</h2><span id='topic+print.pepData'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'pepData'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pepData'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.pepData_+3A_x">x</code></td>
<td>
<p>An object of type 'pepData'</p>
</td></tr>
<tr><td><code id="print.pepData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.proData'>print.proData</h2><span id='topic+print.proData'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'proData'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'proData'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.proData_+3A_x">x</code></td>
<td>
<p>An object of type 'proData'</p>
</td></tr>
<tr><td><code id="print.proData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.proteomicsFilt'>print.proteomicsFilt</h2><span id='topic+print.proteomicsFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'proteomicsFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'proteomicsFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.proteomicsFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'proteomicsFilt'</p>
</td></tr>
<tr><td><code id="print.proteomicsFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.proteomicsFilterSummary'>Proteomics Filter Print Method</h2><span id='topic+print.proteomicsFilterSummary'></span>

<h3>Description</h3>

<p>Print method for summary of proteomics filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'proteomicsFilterSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.proteomicsFilterSummary_+3A_x">x</code></td>
<td>
<p>the proteomics filter summary to print</p>
</td></tr>
<tr><td><code id="print.proteomicsFilterSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.rmdFilt'>print.rmdFilt</h2><span id='topic+print.rmdFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'rmdFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rmdFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.rmdFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'rmdFilt'</p>
</td></tr>
<tr><td><code id="print.rmdFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.rmdFilterSummary'>RMD Filter Print Method</h2><span id='topic+print.rmdFilterSummary'></span>

<h3>Description</h3>

<p>Print method for summary of RMD filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rmdFilterSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.rmdFilterSummary_+3A_x">x</code></td>
<td>
<p>the RMD filter summary to print</p>
</td></tr>
<tr><td><code id="print.rmdFilterSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.RNAFilt'>print.RNAFilt</h2><span id='topic+print.RNAFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'RNAFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RNAFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.RNAFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'RNAFilt'</p>
</td></tr>
<tr><td><code id="print.RNAFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.RNAFiltSummary'>RNA Filter Print Method</h2><span id='topic+print.RNAFiltSummary'></span>

<h3>Description</h3>

<p>Print method for summary of RNAFilt
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RNAFiltSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.RNAFiltSummary_+3A_x">x</code></td>
<td>
<p>the RNAFilt summary to print</p>
</td></tr>
<tr><td><code id="print.RNAFiltSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.seqData'>print.seqData</h2><span id='topic+print.seqData'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'seqData'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'seqData'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.seqData_+3A_x">x</code></td>
<td>
<p>An object of type 'seqData'</p>
</td></tr>
<tr><td><code id="print.seqData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.totalCountFilt'>print.totalCountFilt</h2><span id='topic+print.totalCountFilt'></span>

<h3>Description</h3>

<p>For printing an S3 object of type 'totalCountFilt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'totalCountFilt'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.totalCountFilt_+3A_x">x</code></td>
<td>
<p>An object of type 'totalCountFilt'</p>
</td></tr>
<tr><td><code id="print.totalCountFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='print.totalCountFiltSummary'>Total Count Filter Print Method</h2><span id='topic+print.totalCountFiltSummary'></span>

<h3>Description</h3>

<p>Print method for summary of Total Count filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'totalCountFiltSummary'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.totalCountFiltSummary_+3A_x">x</code></td>
<td>
<p>the Total Count filter summary to print</p>
</td></tr>
<tr><td><code id="print.totalCountFiltSummary_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, prints details about x
</p>

<hr>
<h2 id='protein_quant'>Protein Quantification</h2><span id='topic+protein_quant'></span>

<h3>Description</h3>

<p>This function takes in a pepData object, method (quantification method, mean,
median or rrollup), and the optional argument isoformRes (defaults to NULL).
An object of the class 'proData' is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protein_quant(
  pepData,
  method,
  isoformRes = NULL,
  qrollup_thresh = NULL,
  single_pep = FALSE,
  single_observation = FALSE,
  combine_fn = "median",
  parallel = TRUE,
  emeta_cols = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protein_quant_+3A_pepdata">pepData</code></td>
<td>
<p>an omicsData object of the class 'pepData'</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_method">method</code></td>
<td>
<p>character string specifying one of four protein quantification methods, 'rollup',
'rrollup', 'qrollup' and 'zrollup'</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_isoformres">isoformRes</code></td>
<td>
<p>list of data frames, the result of applying the
'bpquant' function to original pepData object. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_qrollup_thresh">qrollup_thresh</code></td>
<td>
<p>numeric value; is the peptide abundance cutoff
value. Is an argument to qrollup function.</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_single_pep">single_pep</code></td>
<td>
<p>logical indicating whether or not to remove proteins that
have just a single peptide mapping to them, defaults to FALSE.</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_single_observation">single_observation</code></td>
<td>
<p>logical indicating whether or not to remove
peptides that have just a single observation, defaults to FALSE.</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_combine_fn">combine_fn</code></td>
<td>
<p>character string specifying either be 'mean' or 'median'</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_parallel">parallel</code></td>
<td>
<p>logical indicating whether or not to use &quot;doParallel&quot;
loop in applying rollup functions. Defaults to TRUE. Is an argument of
rrollup, qrollup and zrollup functions.</p>
</td></tr>
<tr><td><code id="protein_quant_+3A_emeta_cols">emeta_cols</code></td>
<td>
<p>character vector indicating additional columns of e_meta
that should be kept after rolling up to the protein level. The default,
NULL, only keeps the column containing the mapping variable along with the
new columns created (peps_per_pro and n_peps_used).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If isoformRes is provided then, a temporary pepData object is formed
using the isoformRes information as the e_meta component and the original
pepData object will be used for e_data and f_data components. The
emeta_cname for the temporary pepData object will be the 'protein_isoform'
column of isoformRes. Then one of the three 'method' functions can be
applied to the temporary pepData object to return a proData object. If
isofromRes is left NULL, then depending on the input for 'method', the
correct 'method' function is applied directly to the input pepData object
and a proData object is returned.
</p>


<h3>Value</h3>

<p>omicsData object of the class 'proData'
</p>


<h3>References</h3>

<p>Webb-Robertson, B.-J. M., Matzke, M. M., Datta, S., Payne, S. H.,
Kang, J., Bramer, L. M., ... Waters, K. M. (2014). <em>Bayesian
Proteoform Modeling Improves Protein Quantification of Global Proteomic
Measurements</em>. Molecular &amp; Cellular Proteomics.: MCP, 13(12), 3639-3646.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

mypepData &lt;- group_designation(omicsData = pep_object, main_effects = c("Phenotype"))
mypepData = edata_transform(omicsData = mypepData, "log2")

imdanova_Filt &lt;- imdanova_filter(omicsData = mypepData)
mypepData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = mypepData, min_nonmiss_anova = 2)

imd_anova_res &lt;- imd_anova(omicsData = mypepData, test_method = 'comb',
                           pval_adjust_a_multcomp = 'bon', pval_adjust_g_multcomp = 'bon')

isoformRes = bpquant(statRes = imd_anova_res, pepData = mypepData)

# case where isoformRes is NULL:
results &lt;- protein_quant(pepData = mypepData, method = 'rollup',
                         combine_fn = 'median', isoformRes = NULL)

# case where isoformRes is provided:
# results2 = protein_quant(pepData = mypepData, method = 'rollup',
#                           combine_fn = 'mean', isoformRes = isoformRes)


</code></pre>

<hr>
<h2 id='proteomics_filter'>Proteomics Filter Object</h2><span id='topic+proteomics_filter'></span>

<h3>Description</h3>

<p>This function counts the number of peptides that map to each protein and/or
the number of proteins to which each individual peptide maps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proteomics_filter(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proteomics_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of class &quot;pepData&quot;, the a result of
<code><a href="#topic+as.pepData">as.pepData</a></code> or <code><a href="#topic+as.isobaricpepData">as.isobaricpepData</a></code>. The e_meta
component of omicsData must be nonempty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An S3 object of class proteomicsFilt, which is a list with two
elements. The first element is a data frame of counts for each unique
peptide. The second element is a data frame with the counts for the number
of peptides that map to each unique protein.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
my_filter &lt;- proteomics_filter(omicsData = pep_object)
summary(my_filter, min_num_peps = 3)

</code></pre>

<hr>
<h2 id='qrollup'>Applies qrollup function</h2><span id='topic+qrollup'></span>

<h3>Description</h3>

<p>This function applies the qrollup method to a pepData object for each unique
protein and returns a proData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qrollup(pepData, qrollup_thresh, combine_fn, parallel = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qrollup_+3A_pepdata">pepData</code></td>
<td>
<p>an omicsData object of class 'pepData'</p>
</td></tr>
<tr><td><code id="qrollup_+3A_qrollup_thresh">qrollup_thresh</code></td>
<td>
<p>numeric value between 0 and 1 inclusive. Peptides above
this threshold are used to roll up to the protein level</p>
</td></tr>
<tr><td><code id="qrollup_+3A_combine_fn">combine_fn</code></td>
<td>
<p>logical indicating what combine_fn to use, defaults to
median, other option is mean</p>
</td></tr>
<tr><td><code id="qrollup_+3A_parallel">parallel</code></td>
<td>
<p>logical indicating whether or not to use &quot;doParallel&quot; loop in
applying qrollup function. Defaults to TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the qrollup method, peptides are selected according to a user
selected abundance cutoff value (qrollup_thresh), and protein abundance is
set as the mean of these selected peptides.
</p>


<h3>Value</h3>

<p>an omicsData object of class 'proData'
</p>


<h3>References</h3>

<p>Polpitiya, A. D., Qian, W.-J., Jaitly, N., Petyuk, V. A., Adkins,
J. N., Camp, D. G., ... Smith, R. D. (2008). <em>DAnTE: a statistical
tool for quantitative analysis of -omics data</em>. Bioinformatics (Oxford,
England), 24(13), 1556-1558.
</p>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25dopar+25'></span><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>foreach</dt><dd><p><code><a href="foreach.html#topic+foreach">%dopar%</a></code></p>
</dd>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+pipe">%&gt;%</a></code></p>
</dd>
</dl>


<h3>Value</h3>

<p>No return, used solely for roxygen2 documentation
</p>
<p>No return, used solely for roxygen2 documentation
</p>

<hr>
<h2 id='replace_nas'>Replace NA with 0</h2><span id='topic+replace_nas'></span>

<h3>Description</h3>

<p>This function finds all instances of NA in e_data and replaces them with 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replace_nas(edata, edata_cname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="replace_nas_+3A_edata">edata</code></td>
<td>
<p>A <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of xxx observed and <code class="reqn">n</code> is the
number of samples.</p>
</td></tr>
<tr><td><code id="replace_nas_+3A_edata_cname">edata_cname</code></td>
<td>
<p>A character string specifying the name of the ID column in
the e_data data frame.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used in the as.seqData functions to
replace any NA values with 0s.
</p>


<h3>Value</h3>

<p>An updated e_data data frame where all instances of NA have been
replaced with 0.
</p>

<hr>
<h2 id='replace_zeros'>Replace 0 with NA</h2><span id='topic+replace_zeros'></span>

<h3>Description</h3>

<p>This function finds all instances of 0 in e_data and replaces them with NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replace_zeros(e_data, edata_cname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="replace_zeros_+3A_e_data">e_data</code></td>
<td>
<p>A <code class="reqn">p \times n + 1</code> data frame of expression data, where
<code class="reqn">p</code> is the number of xxx observed and <code class="reqn">n</code> is the
number of samples.</p>
</td></tr>
<tr><td><code id="replace_zeros_+3A_edata_cname">edata_cname</code></td>
<td>
<p>A character string specifying the name of the ID column in
the e_data data frame.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used in the as.pepData, as.proData, as.lipidData,
as.metabData, as.isobaricpepData, and as.nmrData functions to
replace any 0 values with NAs.
</p>


<h3>Value</h3>

<p>An updated e_data data frame where all instances of 0 have been
replaced with NA.
</p>

<hr>
<h2 id='report_dataRes'>Creates a data frame displaying multiple metrics</h2><span id='topic+report_dataRes'></span>

<h3>Description</h3>

<p>This function takes in an object of class 'dataRes' and returns a data frame
displaying a combination of metrics. The six summarizing metrics include,
mean, standard deviation, median, percent observed, minimum, and maximum.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_dataRes(dataRes, minmax = FALSE, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report_dataRes_+3A_datares">dataRes</code></td>
<td>
<p>an object of the class 'dataRes', created by
<code><a href="#topic+edata_summary">edata_summary</a></code>.</p>
</td></tr>
<tr><td><code id="report_dataRes_+3A_minmax">minmax</code></td>
<td>
<p>logical specifying whether or not to include minimum and
maximum data in the returned data frame. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="report_dataRes_+3A_digits">digits</code></td>
<td>
<p>integer indicating the number of decimal places to round</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When creating the 'dataRes' object via <code><a href="#topic+edata_summary">edata_summary</a></code>,
if the 'by' argument is set to 'sample', then the 'groupvar' argument must
be NULL
</p>


<h3>Value</h3>

<p>prints a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mylipid &lt;- edata_transform(omicsData = lipid_neg_object, data_scale = "log2")

dataRes_sample &lt;- edata_summary(omicsData = mylipid, groupvar = NULL, by = "sample")
my_output &lt;- report_dataRes(dataRes_sample)

</code></pre>

<hr>
<h2 id='rip'>Identify Rank-Invariant Biomolcules for Use in Normalization</h2><span id='topic+rip'></span>

<h3>Description</h3>

<p>Selects biomolecules for normalization via the method of rank-invariant
biomolcules (RIP)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rip(e_data, edata_id, fdata_id, groupDF, alpha = 0.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rip_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n</code> data.frame, where <code class="reqn">p</code> is the number of
peptides, proteins, lipids, or metabolites and <code class="reqn">n</code> is the number of
samples. Each row corresponds to data for a peptide, protein, lipid, or
metabolite, with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="rip_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="rip_+3A_fdata_id">fdata_id</code></td>
<td>
<p>character string indicating the name of the sample column
name in f_data.</p>
</td></tr>
<tr><td><code id="rip_+3A_groupdf">groupDF</code></td>
<td>
<p>data.frame created by <code>group_designation</code> with columns
for sample.id and group. If two main effects are provided the original main
effect levels for each sample are returned as the third and fourth columns
of the data.frame.</p>
</td></tr>
<tr><td><code id="rip_+3A_alpha">alpha</code></td>
<td>
<p>numeric p-value threshold, above which the biomolecules are
retained as rank invariant (default value 0.25)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biomolecules with complete data are subjected to a Kruskal-Wallis
test (non-parametric one-way ANOVA) on group membership, and those
biomolecules with p-value greater than a defined threshold <code>alpha</code>
(common values include 0.1 or 0.25) are retained as rank-invariant
biomolecules.
</p>


<h3>Value</h3>

<p>Character vector containing the biomolecules belonging to the RIP
subset.
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>

<hr>
<h2 id='rmd_conversion'>Conversion between log2(RMD) and p-value</h2><span id='topic+rmd_conversion'></span>

<h3>Description</h3>

<p>This function provides a conversion between the log base 2 robust Mahalanobis
distance value and p-value for output from the <code>rmd_runs</code> function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmd_conversion(log2rmd = NULL, pval = NULL, df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmd_conversion_+3A_log2rmd">log2rmd</code></td>
<td>
<p>numeric log base 2 transformed robust Mahalanobis distance
value</p>
</td></tr>
<tr><td><code id="rmd_conversion_+3A_pval">pval</code></td>
<td>
<p>numeric p-value associated with rmd_runs algorithm</p>
</td></tr>
<tr><td><code id="rmd_conversion_+3A_df">df</code></td>
<td>
<p>integer value specifying the degrees of freedom associated with the
test, which should be equal to the number of metrics used in rmd_runs</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Only one of <code>log2rmd</code> and <code>pval</code> should be provided. The
input not provided will be solved for based on the provided input.
</p>


<h3>Value</h3>

<p>The function returns the corresponding p-value or log base 2 robust
Mahalanobis when the other parameter is specified.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(
  omicsData = metab_object,
  data_scale = "log2"
)
mymetab &lt;- group_designation(
  omicsData = mymetab,
  main_effects = "Phenotype"
)
rmd_results &lt;- rmd_filter(
  omicsData = mymetab,
  metrics = c("MAD", "Skewness", "Correlation")
)
rmd_conversion(log2rmd = rmd_results$Log2.md, df = 3)

rmd_conversion(pval = .0001, df = 3)
rmd_conversion(log2rmd = 4.5, df = 3)

</code></pre>

<hr>
<h2 id='rmd_filter'>Robust Mahalanobis Distance (RMD) Filter Object</h2><span id='topic+rmd_filter'></span>

<h3>Description</h3>

<p>The method computes a robust Mahalanobis distance that can be mapped to a
p-value and used to identify outlying samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmd_filter(omicsData, ignore_singleton_groups = TRUE, metrics = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmd_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData' created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="rmd_filter_+3A_ignore_singleton_groups">ignore_singleton_groups</code></td>
<td>
<p>logical indicator of whether to remove
singleton groups or not; defaults to TRUE. A singleton group is a group
consisting of just a single sample. If TRUE, rmd_filter results are
returned only for samples in groups of size greater than 1. This is used
when calculating the correlation.</p>
</td></tr>
<tr><td><code id="rmd_filter_+3A_metrics">metrics</code></td>
<td>
<p>A character vector indicating which metrics should be used
when calculating the robust Mahalanobis distance. This vector must contain
between two and five of the following options: &quot;MAD&quot; (Median Absolute
Deviation), &quot;Kurtosis&quot;, &quot;Skewness&quot;, &quot;Correlation&quot;, and
&quot;Proportion_Missing&quot;. The default is NULL. When NULL a combination of
metrics will be chosen depending on the class of omicsData.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The metrics on which the log2 robust Mahalanobis distance is based
can be specified using the <code>metrics</code> argument. 
</p>

<table>
<tr>
 <td style="text-align: left;"> pepData, proData </td><td style="text-align: left;"> For pepData and proData objects, all 
 five of the metrics "MAD", "Kurtosis", "Skewness", "Correlation", 
 "Proportion_Missing" may be used (this is the default). </td>
</tr>
<tr>
 <td style="text-align: left;"> metabData, 
 lipidData, nmrData </td><td style="text-align: left;"> The use of "Proportion_Missing" is discouraged due 
 to the general lack of missing data in these datasets (the default behavior 
 omits "Proportion_Missing" from the metrics). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Value</h3>

<p>An S3 object of class 'rmdFilt' containing columns for the sample
identifier, log2 robust Mahalanobis distance, p-values, and robust
Mahalanobis distance
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>References</h3>

<p>Matzke, M., Waters, K., Metz, T., Jacobs, J., Sims, A., Baric, 
R., Pounds, J., and Webb-Robertson, B.J. (2011), <em>Improved quality 
control processing of peptide-centric LC-MS proteomics data</em>. 
Bioinformatics. 27(20): 2866-2872.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- edata_transform(omicsData = metab_object, data_scale = "log2")
mymetab &lt;- group_designation(omicsData = mymetab, main_effects = "Phenotype")
rmd_results &lt;- rmd_filter(omicsData = mymetab, 
                          metrics = c("MAD", "Skewness", "Correlation"))
rmd_results &lt;- rmd_filter(omicsData = mymetab)

mypep &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")
mypep &lt;- group_designation(omicsData = mypep, main_effects = "Phenotype")
rmd_results &lt;- rmd_filter(omicsData = mypep)

</code></pre>

<hr>
<h2 id='RNA_filter'>RNA Filter Object</h2><span id='topic+RNA_filter'></span>

<h3>Description</h3>

<p>This function returns a RNAFilt object for use with
<code><a href="#topic+applyFilt">applyFilt</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNA_filter(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RNA_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'seqData', created by
<code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Filter omicsData samples by library size (number of reads) or number
of unique non-zero biomolecules per sample. Useful for visualizing if a
sample contains lower than expected number of reads.
</p>


<h3>Value</h3>

<p>An S3 object of class 'RNAFilt' (data.frame) that contains the
sample identifiers, library size, the number of unique biomolecules with
non-zero observations per sample, and the proportion of non-zero
observations over the total number of biomolecules.
</p>


<h3>Author(s)</h3>

<p>Rachel Richardson
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
to_filter &lt;- RNA_filter(omicsData = rnaseq_object)
summary(to_filter, size_library = 10000)
summary(to_filter, min_nonzero = 5000)
summary(to_filter, min_nonzero = .2)

</code></pre>

<hr>
<h2 id='rrollup'>Applies rrollup function</h2><span id='topic+rrollup'></span>

<h3>Description</h3>

<p>This function applies the rrollup method to a pepData object for each unique
protein and returns a proData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rrollup(pepData, combine_fn, parallel = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rrollup_+3A_pepdata">pepData</code></td>
<td>
<p>an omicsData object of class 'pepData'</p>
</td></tr>
<tr><td><code id="rrollup_+3A_combine_fn">combine_fn</code></td>
<td>
<p>logical indicating what combine_fn to use, defaults to
median, other option is mean</p>
</td></tr>
<tr><td><code id="rrollup_+3A_parallel">parallel</code></td>
<td>
<p>logical indicating whether or not to use &quot;doParallel&quot; loop in
applying rrollup function. Defaults to TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the rrollup method, peptides are scaled based on a reference
peptide and protein abundance is set as the mean of these scaled peptides.
</p>


<h3>Value</h3>

<p>an omicsData object of class 'proData'
</p>


<h3>References</h3>

<p>Matzke, M. M., Brown, J. N., Gritsenko, M. A., Metz, T. O.,
Pounds, J. G., Rodland, K. D., ... Webb-Robertson, B.-J. (2013). <em>A
comparative analysis of computational approaches to relative protein
quantification using peptide peak intensities in label-free LC-MS
proteomics experiments</em>. Proteomics, 13(0), 493-503.
</p>
<p>Polpitiya, A. D., Qian, W.-J., Jaitly, N., Petyuk, V. A., Adkins, J. N.,
Camp, D. G., ... Smith, R. D. (2008). <em>DAnTE: a statistical tool for
quantitative analysis of -omics data</em>. Bioinformatics (Oxford, England),
24(13), 1556-1558.
</p>

<hr>
<h2 id='run_group_meancor'>Calculate the Mean Correlation of a Sample with Respect to Group</h2><span id='topic+run_group_meancor'></span>

<h3>Description</h3>

<p>This function calculates the mean correlation of a sample with all other
samples that have the same group membership
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_group_meancor(omicsData, mintR_groupDF, ignore_singleton_groups = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_group_meancor_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData', or
'lipidData' usually created
by <code><a href="#topic+as.pepData">as.pepData</a></code>, <code><a href="#topic+as.proData">as.proData</a></code>, 
<code><a href="#topic+as.metabData">as.metabData</a></code>, or <code><a href="#topic+as.lipidData">as.lipidData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="run_group_meancor_+3A_mintr_groupdf">mintR_groupDF</code></td>
<td>
<p>data.frame created by <code><a href="#topic+group_designation">group_designation</a></code>
with columns for sample.id and group.</p>
</td></tr>
<tr><td><code id="run_group_meancor_+3A_ignore_singleton_groups">ignore_singleton_groups</code></td>
<td>
<p>logical indicator of whether to remove
singleton groups or not; defaults to TRUE. A singleton group is a group
consisting of just a single sample. If TRUE, rmd_filter results are
returned only for samples in groups of size greater than 1. This is used
when calculating the correlation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Correlation calculations use only complete pairwise observations.
</p>


<h3>Value</h3>

<p>data.frame with two elements: Sample.ID, a character vector giving
the sample names; and Mean_Correlation, a numeric vector giving the mean
correlation values
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>

<hr>
<h2 id='run_kurtosis'>Calculate the Kurtosis of Sample Runs</h2><span id='topic+run_kurtosis'></span>

<h3>Description</h3>

<p>This function calculates the kurtosis across data for each sample run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_kurtosis(data_only)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_kurtosis_+3A_data_only">data_only</code></td>
<td>
<p>a <code class="reqn">p \times n</code> data.frame, where <code class="reqn">p</code> is the number
of peptides and <code class="reqn">n</code> is the number of samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Kurtosis is calculated by method 2 in the <code>e1071</code> package,
which is unbiased under normality. Within a sample NA values are ignorned
in the kurtosis calculation. If all peptide abundance values are missing
within a sample, the kurtosis is replaced by the overall mean of nonmissing
kurtosis values for the data.
</p>


<h3>Value</h3>

<p>data.frame with two elements: Sample, a character vector giving the
sample names; and Kurtosis, a numeric vector giving the kurtosis
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>

<hr>
<h2 id='run_mad'>Calculate the Median Absolute Deviance (MAD) of Sample Runs</h2><span id='topic+run_mad'></span>

<h3>Description</h3>

<p>This function calculates the median absolute deviance across data for each
sample run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_mad(data_only)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_mad_+3A_data_only">data_only</code></td>
<td>
<p>a <code class="reqn">p \times n</code> data.frame, where <code class="reqn">p</code> is the number
of peptides and <code class="reqn">n</code> is the number of samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When calculating the MAD within a sample NA values are ignored. If
all peptide abundance values are missing within a sample, the MAD is
replaced by the overall mean MAD values for the data.
</p>


<h3>Value</h3>

<p>data.frame with two elements: Sample, a character vector giving the
sample names; and MAD, a numeric vector giving the MAD values
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>

<hr>
<h2 id='run_prop_missing'>Calculate the Fraction of Missing Data of Sample Runs</h2><span id='topic+run_prop_missing'></span>

<h3>Description</h3>

<p>This function calculates the fraction of missing data for each sample run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_prop_missing(data_only)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_prop_missing_+3A_data_only">data_only</code></td>
<td>
<p>a <code class="reqn">p \times n</code> data.frame, where <code class="reqn">p</code> is the number
of peptides and <code class="reqn">n</code> is the number of samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with two elements: Sample, a character vector giving the
sample names; and Prop_missing, a numeric vector giving the fraction of
missing values per run
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>

<hr>
<h2 id='run_skewness'>Calculate the Skewness of Sample Runs</h2><span id='topic+run_skewness'></span>

<h3>Description</h3>

<p>This function calculates the skewness across data for each sample run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_skewness(data_only)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_skewness_+3A_data_only">data_only</code></td>
<td>
<p>a <code class="reqn">p \times n</code> data.frame, where <code class="reqn">p</code> is the number
of peptides and <code class="reqn">n</code> is the number of samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Skewness is calculated as a bias-corrected calculation given by
method 2 in the <code>e1071</code> package. Within a sample NA values are
ignorned in the skewness calculation. If all peptide abundance values are
missing within a sample, the skewness is replaced by the overall mean of
nonmissing skewness values for the data.
</p>


<h3>Value</h3>

<p>data.frame with two elements: Sample, a character vector giving the
sample names; and Skewness, a numeric vector giving the skewness values
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>

<hr>
<h2 id='set_check_names'><b>DEPRECATED</b>: Set check.names attribute of omicsData object</h2><span id='topic+set_check_names'></span>

<h3>Description</h3>

<p><em>This function sets the check.names attribute of an omicsData object.</em>
<b>This function has been deprecated in favor of handling checking names
externally and will return an unmodified omicsData.</b>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_check_names(omicsData, set_to = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_check_names_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData', 'proData', 'metabData',
'lipidData', or 'nmrData', usually created by <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.metabData">as.metabData</a></code>,
<code><a href="#topic+as.lipidData">as.lipidData</a></code>, or <code><a href="#topic+as.nmrData">as.nmrData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="set_check_names_+3A_set_to">set_to</code></td>
<td>
<p>logical indicating what to set check.names attribute to.
Defaults to TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>omicsData object with updated check.names attribute
</p>

<hr>
<h2 id='set_data_info'>Functions to set omicsData attributes &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-</h2><span id='topic+set_data_info'></span>

<h3>Description</h3>

<p>Create a set function that will return the value for each attribute. For
example, set_data_info will perform all of the calculations to fill in the
data_info attribute. These functions will be called in the as.xxx functions
to create an xxxData object but can be used individually to update any one of
the attributes at a later time.
Sets/updates the values in the data_info attribute
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_data_info(
  e_data,
  edata_cname,
  data_scale_orig,
  data_scale,
  data_types,
  norm_info,
  is_normalized,
  batch_info,
  is_bc
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_data_info_+3A_e_data">e_data</code>, <code id="set_data_info_+3A_edata_cname">edata_cname</code>, <code id="set_data_info_+3A_data_scale_orig">data_scale_orig</code>, <code id="set_data_info_+3A_data_scale">data_scale</code>, <code id="set_data_info_+3A_data_types">data_types</code>, <code id="set_data_info_+3A_norm_info">norm_info</code>, <code id="set_data_info_+3A_is_normalized">is_normalized</code>, <code id="set_data_info_+3A_batch_info">batch_info</code>, <code id="set_data_info_+3A_is_bc">is_bc</code></td>
<td>
<p>See the <code>as.omicsData</code> functions, e.g. <code><a href="#topic+as.lipidData">as.lipidData</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of edited data_info to replace the attribute in the
omicsData object.
</p>

<hr>
<h2 id='set_filter'>Sets/updates the filters attribute with a filter class object.</h2><span id='topic+set_filter'></span>

<h3>Description</h3>

<p>This function will create a filter class object. The output will always have
the same elements but not all of them will be used for every filter type. This
object will be appended to the list in the filters attribute for an omicsData
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_filter(type, threshold, filtered, method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_filter_+3A_type">type</code></td>
<td>
<p>The type of filter, see <code>filter_objects.R</code></p>
</td></tr>
<tr><td><code id="set_filter_+3A_threshold">threshold</code></td>
<td>
<p>The threshold for filtering applied, if applicable.</p>
</td></tr>
<tr><td><code id="set_filter_+3A_filtered">filtered</code></td>
<td>
<p>The samples/biomolecules that are filtered.</p>
</td></tr>
<tr><td><code id="set_filter_+3A_method">method</code></td>
<td>
<p>A list of metadata about what method was used in applying the 
filter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the elements in the filters attribute:
(list all attributes)
</p>

<hr>
<h2 id='set_isobaric_info'>Sets/updates the values in the isobaric_info attribute.</h2><span id='topic+set_isobaric_info'></span>

<h3>Description</h3>

<p>Sets/updates the values in the isobaric_info attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_isobaric_info(
  exp_cname,
  channel_cname,
  refpool_channel,
  refpool_cname,
  refpool_notation,
  norm_info,
  isobaric_norm
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_isobaric_info_+3A_exp_cname">exp_cname</code>, <code id="set_isobaric_info_+3A_channel_cname">channel_cname</code>, <code id="set_isobaric_info_+3A_refpool_channel">refpool_channel</code>, <code id="set_isobaric_info_+3A_refpool_cname">refpool_cname</code>, <code id="set_isobaric_info_+3A_refpool_notation">refpool_notation</code>, <code id="set_isobaric_info_+3A_norm_info">norm_info</code>, <code id="set_isobaric_info_+3A_isobaric_norm">isobaric_norm</code></td>
<td>
<p>See <code><a href="#topic+normalize_isobaric">normalize_isobaric</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the elements in the isobaric_info attribute:
(list all attributes)
</p>

<hr>
<h2 id='set_meta_info'>Sets/updates the values in the meta_info attribute</h2><span id='topic+set_meta_info'></span>

<h3>Description</h3>

<p>Sets/updates the values in the meta_info attribute
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_meta_info(e_meta, emeta_cname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_meta_info_+3A_e_meta">e_meta</code>, <code id="set_meta_info_+3A_emeta_cname">emeta_cname</code></td>
<td>
<p>See the <code>as.omicsData</code> functions, e.g. 
<code><a href="#topic+as.lipidData">as.lipidData</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the elements in the meta_info attribute:
(list all attributes)
</p>

<hr>
<h2 id='set_nmr_info'>Sets/updates the values in the nmr_info attribute.</h2><span id='topic+set_nmr_info'></span>

<h3>Description</h3>

<p>Sets/updates the values in the nmr_info attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_nmr_info(
  metabolite_name,
  sample_property_cname,
  norm_info,
  nmr_norm,
  backtransform
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_nmr_info_+3A_metabolite_name">metabolite_name</code></td>
<td>
</td></tr>
<tr><td><code id="set_nmr_info_+3A_sample_property_cname">sample_property_cname</code></td>
<td>
</td></tr>
<tr><td><code id="set_nmr_info_+3A_norm_info">norm_info</code></td>
<td>
</td></tr>
<tr><td><code id="set_nmr_info_+3A_nmr_norm">nmr_norm</code></td>
<td>
</td></tr>
<tr><td><code id="set_nmr_info_+3A_backtransform">backtransform</code></td>
<td>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the elements in the nmr_info attribute:
(list all attributes)
</p>

<hr>
<h2 id='spans_make_distribution'>Creates the list of median p-values used to make the background distribution
used to compute the SPANS score in step 2.</h2><span id='topic+spans_make_distribution'></span>

<h3>Description</h3>

<p>Creates the list of median p-values used to make the background distribution
used to compute the SPANS score in step 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spans_make_distribution(
  omicsData,
  group_vector,
  norm_fn,
  sig_inds,
  nonsig_inds,
  select_n
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spans_make_distribution_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'pepData' or 'proData' created by
<code><a href="#topic+as.pepData">as.pepData</a></code> or <code><a href="#topic+as.proData">as.proData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="spans_make_distribution_+3A_group_vector">group_vector</code></td>
<td>
<p>A character vector from the group_DF attribute specifying
the order of the samples. This order is the same as the order of the
samples (columns) in e_data.</p>
</td></tr>
<tr><td><code id="spans_make_distribution_+3A_norm_fn">norm_fn</code></td>
<td>
<p>a character vector of normalization methods to choose from.
Current options are 'mean', 'median', 'zscore', and 'mad'.</p>
</td></tr>
<tr><td><code id="spans_make_distribution_+3A_sig_inds">sig_inds</code></td>
<td>
<p>significant peptide indices (row indices) based on a
Kruskal-Wallis test on the un-normalized data</p>
</td></tr>
<tr><td><code id="spans_make_distribution_+3A_nonsig_inds">nonsig_inds</code></td>
<td>
<p>non-significant peptide indices (row indices) based on a
Kruskal-Wallis test on the un-normalized data</p>
</td></tr>
<tr><td><code id="spans_make_distribution_+3A_select_n">select_n</code></td>
<td>
<p>number of peptide by sample indices in the data to randomly
select to determine normalization parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with 2 elements. The median of highly significant p-values,
and the median of nonsignificant p-values. These are obtained from a SINGLE
Kruskal-Wallis test on data normalized by scale/location factors determined
from a randomly selected subset of peptides and normalization method
</p>

<hr>
<h2 id='spans_procedure'>Calculate SPANS Score for a Number of Normalization Methods</h2><span id='topic+spans_procedure'></span>

<h3>Description</h3>

<p>Ranks different combinations of subset and normalization methods based on
a score that captures how much bias a particular normalization procedure
introduces into the data. Higher score implies less bias.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spans_procedure(
  omicsData,
  norm_fn = c("median", "mean", "zscore", "mad"),
  subset_fn = c("all", "los", "ppp", "rip", "ppp_rip"),
  params = NULL,
  group = NULL,
  n_iter = 1000,
  sig_thresh = 1e-04,
  nonsig_thresh = 0.5,
  min_nonsig = 20,
  min_sig = 20,
  max_nonsig = NULL,
  max_sig = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spans_procedure_+3A_omicsdata">omicsData</code></td>
<td>
<p>aobject of the class 'pepData' or 'proData' created by
<code><a href="#topic+as.pepData">as.pepData</a></code> or <code><a href="#topic+as.proData">as.proData</a></code> respectively. The data
must be log transformed (using edata_transform()) and have a grouping
structure, usually set by calling group_designation() on the object.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_norm_fn">norm_fn</code></td>
<td>
<p>character vector indicating the normalization functions to
test. See details for the current offerings.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_subset_fn">subset_fn</code></td>
<td>
<p>character vector indicating which subset functions to test.
See details for the current offerings.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_params">params</code></td>
<td>
<p>list of additional arguments passed to the chosen subset
functions. See details for parameter specification and default values.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_group">group</code></td>
<td>
<p>character specifying a column name in f_data that gives the
group assignment of the samples. Defaults to NULL, in which case the
grouping structure given in <code>attr(omicsData, 'group_DF')</code> is used.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_n_iter">n_iter</code></td>
<td>
<p>number of iterations used in calculating the background
distribution in step 0 of SPANS. Defaults to 1000.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_sig_thresh">sig_thresh</code></td>
<td>
<p>numeric value that specifies the maximum p-value for which
a biomolecule can be considered highly significant based on a
Kruskal-Wallis test. Defaults to 0.0001.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_nonsig_thresh">nonsig_thresh</code></td>
<td>
<p>numeric value that specifies the minimum p-value for
which a biomolecule can be considered non-significant based on a
Kruskal-Wallis test. Defaults to 0.5.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_min_nonsig">min_nonsig</code></td>
<td>
<p>integer value specifying the minimum number of
non-significant biomolecules identified in step 0 of SPANS in order to
proceed.  nonsig_thresh will be adjusted to the maximum value that gives
this many biomolecules.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_min_sig">min_sig</code></td>
<td>
<p>integer value specifying the minimum number of highly
significant biomolecules identified in step 0 of SPANS in order to proceed.
sig_thresh will be adjusted to the minimum value that gives this many
biomolecules.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_max_nonsig">max_nonsig</code></td>
<td>
<p>integer value specifying the maximum number of
non-significant biomolecules identified in step 0 if SPANS in order to
proceed.  Excesses of non-significant biomolecules will be randomly sampled
down to these values.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_max_sig">max_sig</code></td>
<td>
<p>integer value specifying the maximum number of
highly significant biomolecules identified in step 0 if SPANS in order to
proceed.  Excesses of highly significant biomolecules will be randomly
sampled down to these values.</p>
</td></tr>
<tr><td><code id="spans_procedure_+3A_...">...</code></td>
<td>
<p>Additional arguments </p>

<table>
<tr>
 <td style="text-align: left;"> <code>location_thresh,
scale_thresh</code> The minimum p-value resulting from a Kruskal-Wallis test on
the location and scale parameters resulting from a normalization method in
order for that method to be considered a candidate for scoring.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>verbose</code> Logical specifying whether to print the completion of SPANS
procedure steps to console. Defaults to TRUE.</td>
</tr>
<tr>
 <td style="text-align: left;"> <code>parallel</code> Logical
specifying whether to use a parallel backend.  Depending on the size of
your data, setting this to FALSE can cause the algorithm to be very slow.
Defaults to TRUE. </td>
</tr>

</table>
</td></tr>
</table>


<h3>Details</h3>

<p>Below are details for specifying function and parameter options.
</p>


<h3>Value</h3>

<p>An object of class 'SPANSRes', which is a dataframe containing
columns for the subset method and normalization used, the parameters used
in the subset method, and the corresponding SPANS score.  <br />
</p>
<p>The column 'mols_used_in_norm' contains the number of molecules that were
selected by the subset method and subsequently used to determine the
location/scale parameters for normalization.  The column 'passed selection'
is <code>TRUE</code> if the subset+normalization procedure was selected for
scoring.<br />
</p>
<p>The attribute 'method_selection_pvals' is a dataframe containing
information on the p values used to determine if a method was selected for
scoring (location_p_value, scale_p_value) as well as the probabilities
(F_log_HSmPV, F_log_NSmPV) given by the empirical cdfs generated in the
first step of SPANS.
</p>


<h3>Subset Functions</h3>

<p>Specifying a subset function indicates the subset
of features (rows of <code>e_data</code>) that should be used for computing
normalization factors. The following are valid options: &quot;all&quot;, &quot;los&quot;,
&quot;ppp&quot;, &quot;rip&quot;, and &quot;ppp_rip&quot;. <br /> </p>

<table>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> "all" is the subset
  that includes all features (i.e. no subsetting is done). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> "los"
  identifies the subset of the features associated with the top <code>L</code>,
  where <code>L</code> is a proportion between 0 and 1, order statistics.
  Specifically, the features with the top <code>L</code> proportion of highest
  absolute abundance are retained for each sample, and the union of these
  features is taken as the subset identified (Wang et al., 2006). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;">
  "ppp" (orignally stands for percentage of peptides present) identifies the
  subset of features that are present/non-missing for a minimum
  <code>proportion</code> of samples (Karpievitch et al., 2009; Kultima et al.,
  2009). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> "complete" subset of features that have no missing data
  across all samples.  Equivalent to "ppp" with proportion = 1. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;">
  "rip" identifies features with complete data that have a p-value greater
  than a defined threshold <code>alpha</code> (common values include 0.1 or 0.25)
  when subjected to a Kruskal-Wallis test based (non-parametric one-way
  ANOVA) on group membership (Webb-Robertson et al., 2011). </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;">
  "ppp_rip" is equivalent to "rip" however rather than requiring features
  with complete data, features with at least a <code>proportion</code> of
  non-missing values are subject to the Kruskal-Wallis test.</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Normalization Functions</h3>

<p>Specifying a normalization function
indicates how normalization scale and location parameters should be
calculated. The following are valid options: &quot;median&quot;, &quot;mean&quot;, &quot;zscore&quot;,
and &quot;mad&quot;. Parameters for median centering are calculated if &quot;median&quot; is
specified. The location estimates are the sample-wise medians of the subset
data. There are no scale estimates for median centering. Parameters for
mean centering are calculated if &quot;mean&quot; is specified. The location
estimates are the sample-wise means of the subset data. There are no scale
estimates for median centering. Parameters for z-score transformation are
calculated if &quot;zscore&quot; is specified. The location estimates are the subset
means for each sample. The scale estimates are the subset standard
deviations for each sample. Parameters for median absolute deviation (MAD)
transformation are calculated if &quot;mad&quot; is specified.
</p>


<h3>Specifying Subset Parameters Using the <code>params</code> argument</h3>

<p>Parameters for the chosen subset function should be specified in a list.
The list elements should have names corresponding to the subset function
inputs and contain a <em>list</em> of numeric values.  The elements of
ppp_rip will be length 2 numeric vectors, corresponding to the parameters
for ppp and rip. See examples.
</p>
<p>The following subset functions have parameters that can be specified:
</p>

<table>
<tr>
 <td style="text-align: left;"> los </td><td style="text-align: left;"> list of values between 0 and 1 indicating the top
  proportion of order statistics. Defaults to list(0.05,0.1,0.2,0.3) if
  unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> ppp </td><td style="text-align: left;"> list of values between 0 and 1
  specifying the proportion of samples that must have non-missing values for
  a feature to be retained. Defaults to list(0.1,0.25,0.50,0.75) if
  unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> rip </td><td style="text-align: left;"> list of values between 0 and 1
  specifying the p-value threshold for determining rank invariance. Defaults
  to list(0.1,0.15,0.2,0.25) if unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> ppp_rip </td><td style="text-align: left;"> list
  of length 2 numeric vectors corresponding to the RIP and PPP parameters
  above. Defaults list(c(0.1,0.1), c(0.25, 0.15), c(0.5, 0.2), c(0.75,0.25))
  if unspecified. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Daniel Claborne
</p>


<h3>References</h3>

<p>Webb-Robertson BJ, Matzke MM, Jacobs JM, Pounds JG, Waters KM. A
statistical selection strategy for normalization procedures in LC-MS
proteomics experiments through dataset-dependent ranking of normalization
scaling factors. Proteomics. 2011;11(24):4736-41.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

pep_object &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")
pep_object &lt;- group_designation(omicsData = pep_object, main_effects = "Phenotype")

## default parameters
spans_res &lt;- spans_procedure(omicsData = pep_object)

## specify only certain subset and normalization functions
spans_res &lt;- spans_procedure(omicsData = pep_object, 
                             norm_fn = c("median", "zscore"), 
                             subset_fn = c("all", "los", "ppp"))

## specify parameters for supplied subset functions, 
## notice ppp_rip takes a vector of two numeric arguments.
spans_res &lt;- spans_procedure(omicsData = pep_object, 
                             subset_fn = c("all", "los", "ppp"), 
                             params = list(los = list(0.25, 0.5), 
                             ppp = list(0.15, 0.25)))
spans_res &lt;- spans_procedure(omicsData = pep_object, 
                             subset_fn = c("all", "rip", "ppp_rip"), 
                             params = list(rip = list(0.3, 0.4), 
                             ppp_rip = list(c(0.15, 0.5), c(0.25, 0.5))))


</code></pre>

<hr>
<h2 id='statres_barplot'>Fold change barplots for statres objects</h2><span id='topic+statres_barplot'></span>

<h3>Description</h3>

<p>Plots a bar-chart with bar heights indicating the number of significant
biomolecules, grouped by test type and fold change direction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>statres_barplot(
  x,
  stacked,
  fc_colors,
  text_size,
  display_count,
  x_lab,
  y_lab,
  title_lab,
  legend_lab,
  free_y_axis
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="statres_barplot_+3A_x">x</code>, <code id="statres_barplot_+3A_stacked">stacked</code>, <code id="statres_barplot_+3A_fc_colors">fc_colors</code></td>
<td>
<p>passed from
<code><a href="#topic+plot.statRes">pmartR::plot.statRes()</a></code></p>
</td></tr>
<tr><td><code id="statres_barplot_+3A_text_size">text_size</code></td>
<td>
<p>An integer specifying the size of the text (number of
non-missing values) within the plot. The default is 3.</p>
</td></tr>
<tr><td><code id="statres_barplot_+3A_display_count">display_count</code></td>
<td>
<p>logical value. Indicates whether the non-missing counts will
be displayed on the bar plot. The default is TRUE.</p>
</td></tr>
<tr><td><code id="statres_barplot_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="statres_barplot_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="statres_barplot_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="statres_barplot_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'ggplot' object - barplot.
</p>

<hr>
<h2 id='statRes_output'>Function to take raw output of 'imd_anova' and create output for 'statRes' object</h2><span id='topic+statRes_output'></span>

<h3>Description</h3>

<p>Function to take raw output of 'imd_anova' and create output for 'statRes' object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>statRes_output(
  imd_anova_out,
  omicsData,
  comparisons,
  test_method,
  pval_adjust_a_multcomp,
  pval_adjust_g_multcomp,
  pval_adjust_a_fdr,
  pval_adjust_g_fdr,
  pval_thresh
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="statRes_output_+3A_imd_anova_out">imd_anova_out</code></td>
<td>
<p>data frame containing the results of the
<code>imd_anova</code> call.</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_omicsdata">omicsData</code></td>
<td>
<p>pmartR data object of any class, which has a 'group_df'
attribute that is usually created by the 'group_designation()' function</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_comparisons">comparisons</code></td>
<td>
<p>character vector of comparison names, e.g. c(&quot;A_vs_B&quot;, &quot;B_vs_C&quot;, ...)</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_test_method">test_method</code></td>
<td>
<p>test method used (&quot;anova&quot;, &quot;gtest&quot;, or &quot;combined&quot;)</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_pval_adjust_a_multcomp">pval_adjust_a_multcomp</code></td>
<td>
<p>character string specifying which type of multiple
comparison adjustment was implemented for ANOVA tests. Valid options include:
&quot;bonferroni&quot;, &quot;holm&quot;, &quot;tukey&quot;, and &quot;dunnett&quot;.</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_pval_adjust_g_multcomp">pval_adjust_g_multcomp</code></td>
<td>
<p>character string specifying which type of multiple
comparison adjustment was implemented for G-tests. Valid options include:
&quot;bonferroni&quot; and &quot;holm&quot;.</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_pval_adjust_a_fdr">pval_adjust_a_fdr</code></td>
<td>
<p>character string specifying which type of FDR
adjustment was implemented for ANOVA tests. Valid options include:
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, and &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_pval_adjust_g_fdr">pval_adjust_g_fdr</code></td>
<td>
<p>character string specifying which type of FDR
adjustment was implemented for G-tests. Valid options include:
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, and &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="statRes_output_+3A_pval_thresh">pval_thresh</code></td>
<td>
<p>numeric p-value threshold value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class statRes
</p>

<hr>
<h2 id='statres_volcano_plot'>Volcano plot for the anova results of imd-anova</h2><span id='topic+statres_volcano_plot'></span>

<h3>Description</h3>

<p>Plots a volcano plot showing negative log10 p-values on the y axis and fold
change on the x axis. Each point is colored by fold change direction and
whether or not it was significant by ANOVA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>statres_volcano_plot(
  volcano,
  data_scale,
  pval_thresh,
  fc_colors,
  fc_threshold,
  interactive,
  x_lab,
  y_lab,
  title_lab,
  legend_lab
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="statres_volcano_plot_+3A_volcano">volcano</code></td>
<td>
<p>data frame produced by pmartR:::<a href="#topic+make_volcano_plot_df">make_volcano_plot_df</a></p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_data_scale">data_scale</code></td>
<td>
<p>One of c(&quot;log2&quot;,&quot;log&quot;,&quot;log10&quot;), for labeling purposes.</p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_pval_thresh">pval_thresh</code></td>
<td>
<p>numeric value between 0 and 1 for the alpha level to
determine significance. Any values that are significant at this level will
be colored based on fc_colors.</p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_fc_colors">fc_colors</code>, <code id="statres_volcano_plot_+3A_fc_threshold">fc_threshold</code>, <code id="statres_volcano_plot_+3A_interactive">interactive</code></td>
<td>
<p>passed from
<code><a href="#topic+plot.statRes">pmartR::plot.statRes()</a></code></p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_x_lab">x_lab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_y_lab">y_lab</code></td>
<td>
<p>character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_title_lab">title_lab</code></td>
<td>
<p>character string specifying the plot title.</p>
</td></tr>
<tr><td><code id="statres_volcano_plot_+3A_legend_lab">legend_lab</code></td>
<td>
<p>character string specifying the legend title.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'ggplot' object.  A volcano plot.
</p>

<hr>
<h2 id='statRes-class'>Summary of statRes Object</h2><span id='topic+statRes-class'></span>

<h3>Description</h3>

<p>Provide summary information about statRes objects
</p>


<h3>Value</h3>

<p>No return value, prints details about the statres object.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+imd_anova">imd_anova</a></code>
</p>

<hr>
<h2 id='summary_km'>Basic survival analysis summary</h2><span id='topic+summary_km'></span>

<h3>Description</h3>

<p>Implements overall survival analysis or progression-free survival analysis, depending upon the datatypes supplied to
surv_designation, and gives a summary of the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_km(omicsData, percent = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_km_+3A_omicsdata">omicsData</code></td>
<td>
<p>A pmartR data object of any class, which has a 'group_df' attribute that is usually created by the 'group_designation()' function</p>
</td></tr>
<tr><td><code id="summary_km_+3A_percent">percent</code></td>
<td>
<p>The percentile</p>
</td></tr>
<tr><td><code id="summary_km_+3A_...">...</code></td>
<td>
<p>extra arguments passed to regexpr if pattern is specified</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if 'percent' is provided then the time at which that probability of death is returned; else, the summary of the 'survival' object is returned
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(OvarianPepdataBP)
attr(tcga_ovarian_pepdata_bp, "survDF") &lt;- list(t_death = "survival_time",
                                                ind_death = "vital_status")
# No percent is provided so the entire object is returned
summary_km(tcga_ovarian_pepdata_bp)

# Percent is provided so corresponding time point is returned
summary_km(tcga_ovarian_pepdata_bp, .4)

## End(Not run)

</code></pre>

<hr>
<h2 id='summary-isobaricnormRes'>Summary for isobaricnormRes Object</h2><span id='topic+summary-isobaricnormRes'></span><span id='topic+summary.isobaricnormRes'></span>

<h3>Description</h3>

<p>For creating a summary of an S3 object of type 'isobaricnormRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'isobaricnormRes'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary-isobaricnormRes_+3A_object">object</code></td>
<td>
<p>object of type isobaricnormRes, created by
<code><a href="#topic+normalize_isobaric">normalize_isobaric</a></code></p>
</td></tr>
<tr><td><code id="summary-isobaricnormRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myiso &lt;- edata_transform(omicsData = isobaric_object, data_scale = "log2")
myiso_norm &lt;- normalize_isobaric(
  omicsData = myiso, exp_cname = "Plex",
  apply_norm = FALSE,
  refpool_cname = "Virus",
  refpool_notation = "Pool"
)
mysummary &lt;- summary(myiso_norm)

</code></pre>

<hr>
<h2 id='summary-nmrnormRes'>Summary of nmrnormRes Object</h2><span id='topic+summary-nmrnormRes'></span><span id='topic+summary.nmrnormRes'></span>

<h3>Description</h3>

<p>For creating a summary of an S3 object of type 'nmrnormRes'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nmrnormRes'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary-nmrnormRes_+3A_object">object</code></td>
<td>
<p>object of type nmrnormRes, created by
<code><a href="#topic+normalize_nmr">normalize_nmr</a></code></p>
</td></tr>
<tr><td><code id="summary-nmrnormRes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mynmr &lt;- edata_transform(
  omicsData = nmr_identified_object,
  data_scale = "log2"
)
nmr_norm &lt;- normalize_nmr(
  omicsData = mynmr, apply_norm = FALSE,
  sample_property_cname = "Concentration"
)
mysummary &lt;- summary(nmr_norm)

</code></pre>

<hr>
<h2 id='summary-omicsData'>Produce a basic summary of a pmartR omicsData S3 Object</h2><span id='topic+summary-omicsData'></span><span id='topic+summary.pepData'></span><span id='topic+summary.proData'></span><span id='topic+summary.lipidData'></span><span id='topic+summary.metabData'></span><span id='topic+summary.nmrData'></span><span id='topic+summary.seqData'></span>

<h3>Description</h3>

<p>This function will provide basic summary statistics for omicsData objects
from the pmartR package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pepData'
summary(object, ...)

## S3 method for class 'proData'
summary(object, ...)

## S3 method for class 'lipidData'
summary(object, ...)

## S3 method for class 'metabData'
summary(object, ...)

## S3 method for class 'nmrData'
summary(object, ...)

## S3 method for class 'seqData'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary-omicsData_+3A_object">object</code></td>
<td>
<p>an object of the class 'lipidData', 'metabData', 'pepData',
'proData', nmrData', or 'seqData' usually created by <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.pepData">as.pepData</a></code>,
<code><a href="#topic+as.proData">as.proData</a></code>, <code><a href="#topic+as.nmrData">as.nmrData</a></code>, or <code><a href="#topic+as.seqData">as.seqData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="summary-omicsData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary table for the pmartR omicsData object. If assigned to a
variable, the elements of the summary table are saved in a list format.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton, Thomas Johansen
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
pep_summary &lt;- summary(pep_object)
iso_summary &lt;- summary(isobaric_object)
pro_summary &lt;- summary(pro_object)
metab_summary &lt;- summary(metab_object)
lipid_summary &lt;- summary(lipid_neg_object)
nmr_summary &lt;- summary(nmr_identified_object)
rnaseq_summary &lt;- summary(rnaseq_object)

</code></pre>

<hr>
<h2 id='summary-pmartR-results'>Summary of pmartR Analysis Functions</h2><span id='topic+summary-pmartR-results'></span><span id='topic+summary.normRes'></span><span id='topic+summary.SPANSRes'></span><span id='topic+summary.dimRes'></span><span id='topic+summary.corRes'></span>

<h3>Description</h3>

<p>Provide basic summaries for results objects from the pmartR package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'normRes'
summary(object, ...)

## S3 method for class 'SPANSRes'
summary(object, ...)

## S3 method for class 'dimRes'
summary(object, ...)

## S3 method for class 'corRes'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary-pmartR-results_+3A_object">object</code></td>
<td>
<p>object of class corRes</p>
</td></tr>
<tr><td><code id="summary-pmartR-results_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary table or list for the pmartR results object
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton, Thomas Johansen
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- group_designation(omicsData = pep_object, main_effects = "Phenotype")
mypep &lt;- edata_transform(omicsData = mypep, data_scale = "log2")

norm_result &lt;- normalize_global(omicsData = mypep, norm_fn = "median", subset_fn = "all")
summary(norm_result)


spans_results &lt;- spans_procedure(omicsData = mypep)
summary(spans_results)


dim_results &lt;- dim_reduction(omicsData = mypep)
summary(dim_results)

cor_results &lt;- cor_result(omicsData = mypep)
summary(cor_results)

</code></pre>

<hr>
<h2 id='summary-trelliData'>Summarizes potential plotting options for a trelliData object</h2><span id='topic+summary-trelliData'></span><span id='topic+summary.trelliData'></span>

<h3>Description</h3>

<p>Summarizes potential plotting options for a trelliData object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'trelliData'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary-trelliData_+3A_object">object</code></td>
<td>
<p>An object from the as.trelliData.edata or as.trelliData
functions</p>
</td></tr>
<tr><td><code id="summary-trelliData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing panel plot options for this trelliData object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


library(dplyr)
library(pmartRdata)

trelliData &lt;- as.trelliData.edata(e_data = pep_edata,
                                   edata_cname = "Peptide",
                                   omics_type = "pepData")

# Use an edata example. Build with as.trelliData.edata.
summary(trelliData)
summary(trelliData %&gt;% trelli_panel_by("Peptide"))
summary(trelliData %&gt;% trelli_panel_by("Sample"))



</code></pre>

<hr>
<h2 id='summary.customFilt'>Custom Filter Summary</h2><span id='topic+summary.customFilt'></span>

<h3>Description</h3>

<p>Provide summary of a customFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'customFilt'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.customFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'customFilt' created by
<code><a href="#topic+custom_filter">custom_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.customFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary of the items in e_data, f_data, and e_meta that will be
removed as a result of applying the custom filter.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+custom_filter">custom_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
to_filter &lt;- custom_filter(omicsData = metab_object, e_data_remove = "fumaric acid",
                           f_data_remove = "Sample_1_Phenotype2_B")
summary(to_filter)

to_filter2 &lt;- custom_filter(omicsData = metab_object, 
                            f_data_keep = metab_object$f_data$SampleID[1:10])
summary(to_filter2)

</code></pre>

<hr>
<h2 id='summary.cvFilt'>Coefficient of Variation (CV) Filter Summary</h2><span id='topic+summary.cvFilt'></span>

<h3>Description</h3>

<p>Provide summary of a cvFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cvFilt'
summary(object, cv_threshold = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.cvFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'cvFilt' created by
<code><a href="#topic+cv_filter">cv_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.cvFilt_+3A_cv_threshold">cv_threshold</code></td>
<td>
<p>numeric value greater than 1 and less than the value
given by filter_object$CV. CV values above cv_threshold are filtered out.
Default value is NULL.</p>
</td></tr>
<tr><td><code id="summary.cvFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary of the CV values, number of NA values, and non-NA values.
If a CV threshold is provided, the biomolecules that would be filtered
based on this threshold are reported.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cv_filter">cv_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- group_designation(omicsData = pep_object, main_effects = "Phenotype")
to_filter &lt;- cv_filter(omicsData = mypep, use_groups = TRUE)
summary(to_filter, cv_threshold = 30)

</code></pre>

<hr>
<h2 id='summary.imdanovaFilt'>IMD-ANOVA Filter Summary</h2><span id='topic+summary.imdanovaFilt'></span>

<h3>Description</h3>

<p>Provide summary of a imdanovaFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'imdanovaFilt'
summary(
  object,
  min_nonmiss_anova = NULL,
  min_nonmiss_gtest = NULL,
  comparisons = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.imdanovaFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'imdanovaFilt' created by
<code><a href="#topic+imdanova_filter">imdanova_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.imdanovaFilt_+3A_min_nonmiss_anova">min_nonmiss_anova</code></td>
<td>
<p>integer value specifying the minimum number of
non-missing feature values allowed per group for <code>anova_filter</code>.
Defaults to NULL. Suggested value is 2.</p>
</td></tr>
<tr><td><code id="summary.imdanovaFilt_+3A_min_nonmiss_gtest">min_nonmiss_gtest</code></td>
<td>
<p>integer value specifying the minimum number of
non-missing feature values allowed per group for <code>gtest_filter</code>.
Defaults to NULL. Suggested value is 3.</p>
</td></tr>
<tr><td><code id="summary.imdanovaFilt_+3A_comparisons">comparisons</code></td>
<td>
<p>data frame with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control (e.g. Control is the
reference group).</p>
</td></tr>
<tr><td><code id="summary.imdanovaFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If min_nonmiss_gtest or min_nonmiss_anova is specified, the number of
biomolecules to be filtered with the specified threshold are reported.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+imdanova_filter">imdanova_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mypep &lt;- group_designation(omicsData = pep_object, main_effects = "Phenotype")
myfilt &lt;- imdanova_filter(omicsData = mypep)
summary(myfilt, min_nonmiss_anova = 2, min_nonmiss_gtest = 3)

</code></pre>

<hr>
<h2 id='summary.moleculeFilt'>Molecule Filter Summary</h2><span id='topic+summary.moleculeFilt'></span>

<h3>Description</h3>

<p>Provide summary of a moleculeFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'moleculeFilt'
summary(object, min_num = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.moleculeFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'moleculeFilt' created by
<code><a href="#topic+molecule_filter">molecule_filter</a></code></p>
</td></tr>
<tr><td><code id="summary.moleculeFilt_+3A_min_num">min_num</code></td>
<td>
<p>integer value specifying the minimum number of times each
feature must be observed across all samples. Default value is NULL.</p>
</td></tr>
<tr><td><code id="summary.moleculeFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary table giving the number of biomolecules by number of
observed values across all samples. If min_num is specified, the numbers of
biomolecules to be filtered and to be retained based on the specified
threshold are reported. If, upon creation of moleculeFilt object,
<code>use_groups = TRUE</code> or <code>use_batches = TRUE</code> were specified, the
numbers reported by the summary are based on groups and/or batches.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+molecule_filter">molecule_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myfilter &lt;- molecule_filter(omicsData = pep_object)
summary(myfilter)
summary(myfilter, min_num = 2)

</code></pre>

<hr>
<h2 id='summary.proteomicsFilt'>Proteomics Filter Summary</h2><span id='topic+summary.proteomicsFilt'></span>

<h3>Description</h3>

<p>Provide summary of a proteomicsFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'proteomicsFilt'
summary(object, min_num_peps = NULL, degen_peps = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.proteomicsFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'proteomicsFilt' created by
<code><a href="#topic+proteomics_filter">proteomics_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.proteomicsFilt_+3A_min_num_peps">min_num_peps</code></td>
<td>
<p>optional integer value between 1 and the maximum
number of peptides that map to a protein in the data. The value specifies
the minimum number of peptides that must map to a protein. Any protein with
less than <code>min_num_peps</code> mapping to it will be returned as a protein
that should be filtered. Default value is NULL.</p>
</td></tr>
<tr><td><code id="summary.proteomicsFilt_+3A_degen_peps">degen_peps</code></td>
<td>
<p>logical indicator of whether to filter out 'degenerate' or 'redundant'
peptides (i.e. peptides mapping to multiple proteins) (TRUE) or not
(FALSE). Default value is FALSE.</p>
</td></tr>
<tr><td><code id="summary.proteomicsFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary table giving the number of Observed Proteins per Peptide
and number of Observed Peptides per Protein. If min_num_peps is specified
and/or degen_peps is TRUE, the number of biomolecules to be filtered with
the specified threshold(s) are reported.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+proteomics_filter">proteomics_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myfilt &lt;- proteomics_filter(omicsData = pep_object)
summary(myfilt, degen_peps = TRUE) # there are no degenerate peptides to filter out
summary(myfilt, min_num_peps = 2)

</code></pre>

<hr>
<h2 id='summary.rmdFilt'>RMD Filter Summary</h2><span id='topic+summary.rmdFilt'></span>

<h3>Description</h3>

<p>Provide summary of a rmdFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rmdFilt'
summary(object, pvalue_threshold = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.rmdFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'rmdFilt' created by
<code><a href="#topic+rmd_filter">rmd_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.rmdFilt_+3A_pvalue_threshold">pvalue_threshold</code></td>
<td>
<p>A threshold for the Robust Mahalanobis Distance (RMD)
p-value. All samples with p-values below the threshold will be filtered
out. Default value is NULL. Suggested value is 0.0001</p>
</td></tr>
<tr><td><code id="summary.rmdFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary of the p-values associated with running RMD-PAV across all
samples. If a p-value threshold is provided the samples that would be
filtered at this threshold are reported.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rmd_filter">rmd_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
mymetab &lt;- group_designation(omicsData = metab_object, main_effects = "Phenotype")
mymetab &lt;- edata_transform(omicsData = mymetab, data_scale = "log2")
myfilt &lt;- rmd_filter(omicsData = mymetab)
summary(myfilt, pvalue_threshold = 0.001)

</code></pre>

<hr>
<h2 id='summary.RNAFilt'>RNA Filter Summary</h2><span id='topic+summary.RNAFilt'></span>

<h3>Description</h3>

<p>Provide summary of a RNAFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RNAFilt'
summary(object, size_library = NULL, min_nonzero = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.RNAFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'RNAFilt' created by
<code><a href="#topic+RNA_filter">RNA_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.RNAFilt_+3A_size_library">size_library</code></td>
<td>
<p>integer cut-off for sample library size (i.e. number of
reads). Defaults to NULL.</p>
</td></tr>
<tr><td><code id="summary.RNAFilt_+3A_min_nonzero">min_nonzero</code></td>
<td>
<p>integer or float between 0 and 1. Cut-off for number of
unique biomolecules with non-zero counts or as a proportion of total
biomolecules. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="summary.RNAFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary table giving the minimum, maximum, 1st and 3rd quartiles,
mean and standard deviation for library size (the number of unique
biomolecules with non-zero observations per sample), and the proportion of
non-zero observations over the total number of biomolecules.
</p>


<h3>Author(s)</h3>

<p>Rachel Richardson
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RNA_filter">RNA_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pmartRdata)
myfilter &lt;- RNA_filter(omicsData = rnaseq_object)
summary(myfilter)
summary(myfilter, min_nonzero = 2)

</code></pre>

<hr>
<h2 id='summary.totalCountFilt'>Total Count Filter Summary</h2><span id='topic+summary.totalCountFilt'></span>

<h3>Description</h3>

<p>Provide summary of a totalCountFilt S3 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'totalCountFilt'
summary(object, min_count = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.totalCountFilt_+3A_object">object</code></td>
<td>
<p>S3 object of class 'totalCountFilt' created by
<code><a href="#topic+total_count_filter">total_count_filter</a></code>.</p>
</td></tr>
<tr><td><code id="summary.totalCountFilt_+3A_min_count">min_count</code></td>
<td>
<p>numeric value greater than 1 and less than the value
given by filter_object$Total_Count. Values below min_count are filtered out.
Default value is NULL.</p>
</td></tr>
<tr><td><code id="summary.totalCountFilt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a summary of the Total Count values, number of zero values, and
non-zero values. If a min_count is provided the biomolecules that would be
filtered at this threshold are reported.
</p>


<h3>Author(s)</h3>

<p>Rachel Richardson
</p>


<h3>See Also</h3>

<p><code><a href="#topic+total_count_filter">total_count_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)
myfilt &lt;- total_count_filter(omicsData = rnaseq_object)
summary(myfilt, min_count = 15)


</code></pre>

<hr>
<h2 id='surv_designation'>Create a &quot;surv_DF&quot; attribute so that survival analysis can be implemented.</h2><span id='topic+surv_designation'></span>

<h3>Description</h3>

<p>This function will add the necessary information to omicsData such that
survival analysis can be applied to it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_designation(
  omicsData,
  t_death,
  t_progress = NULL,
  ind_death,
  ind_progress = NULL,
  covariates = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_designation_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'lipidData', 'metabData', 'pepData',
or 'proData' usually created by <code><a href="#topic+as.lipidData">as.lipidData</a></code>,
<code><a href="#topic+as.metabData">as.metabData</a></code>, <code><a href="#topic+as.pepData">as.pepData</a></code>, or
<code><a href="#topic+as.proData">as.proData</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="surv_designation_+3A_t_death">t_death</code></td>
<td>
<p>the column in 'f_data' that corresponds to the subjects' time
of death</p>
</td></tr>
<tr><td><code id="surv_designation_+3A_t_progress">t_progress</code></td>
<td>
<p>the column in 'f_data' that corresponds to the subjects'
time of progression</p>
</td></tr>
<tr><td><code id="surv_designation_+3A_ind_death">ind_death</code></td>
<td>
<p>the column in 'f_data' that corresponds to the subjects'
status, e.g. alive/dead</p>
</td></tr>
<tr><td><code id="surv_designation_+3A_ind_progress">ind_progress</code></td>
<td>
<p>the column in 'f_data' that corresponds to the subjects'
progression status</p>
</td></tr>
<tr><td><code id="surv_designation_+3A_covariates">covariates</code></td>
<td>
<p>the column(s) in 'f_data' that correspond to covariates to
be included in the survivial analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>omicsData is returned with the additional attribute
</p>


<h3>Author(s)</h3>

<p>Bryan Stanfill
</p>

<hr>
<h2 id='take_diff'>Compute pairwise differences</h2><span id='topic+take_diff'></span>

<h3>Description</h3>

<p>Computes the differences for paired data according to the information in the
pairing column of f_data. This variable name is also an attribute of the
group_DF attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>take_diff(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="take_diff_+3A_omicsdata">omicsData</code></td>
<td>
<p>Any one of the omicsData objects (pepData, metabData, ...).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing the differences between paired samples.
</p>


<h3>Author(s)</h3>

<p>Evan A Martin
</p>

<hr>
<h2 id='total_count_filter'>Total Count Filter Object</h2><span id='topic+total_count_filter'></span>

<h3>Description</h3>

<p>This function returns a totalcountFilt object for use with
<code><a href="#topic+applyFilt">applyFilt</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>total_count_filter(omicsData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="total_count_filter_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of the class 'seqData', created by
<code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Filter is based off of recommendations in edgeR processing, where
the low-observed biomolecules are removed from processing. Default
recommendation in edgeR is at least 15 total counts observed across samples
(i.e., if the sum of counts in a row of e_data is &lt; 15, default edgeR
filtering would remove this biomolecule).
</p>


<h3>Value</h3>

<p>An S3 object of class 'totalcountFilt' (data.frame) that contains the
molecule identifier and the total count of observed reads for that molecule
across all samples.
</p>


<h3>Author(s)</h3>

<p>Rachel Richardson
</p>


<h3>References</h3>

<p>Chen Y, Lun ATL, and Smyth, GK (2016). From reads to genes to pathways:
differential expression analysis of RNA-Seq experiments using Rsubread and
the edgeR quasi-likelihood pipeline. F1000Research 5, 1438.
http://f1000research.com/articles/5-1438
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)
to_filter &lt;- total_count_filter(omicsData = rnaseq_object)
summary(to_filter, min_count = 15)


</code></pre>

<hr>
<h2 id='trelli_abundance_boxplot'>Boxplot trelliscope building function for abundance data</h2><span id='topic+trelli_abundance_boxplot'></span>

<h3>Description</h3>

<p>Specify a boxplot design and cognostics for the abundance
boxplot trelliscope. Each boxplot will have its own groups as specified by
the first main effect in group_designation. Use &quot;trelli_rnaseq_boxplot&quot;
for RNA-Seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_abundance_boxplot(
  trelliData,
  cognostics = c("count", "mean abundance"),
  ggplot_params = NULL,
  interactive = FALSE,
  include_points = TRUE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_abundance_boxplot_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData or
as.trelliData.edata, and grouped by trelli_panel_by. Required.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries
are &quot;count&quot;, &quot;mean abundance&quot;, &quot;median abundance&quot;, and &quot;cv abundance&quot;. 
If data are paneled by a biomolecule, the count will be &quot;sample count&quot;.
If data are paneled by a sample or a biomolecule class, the count will be &quot;biomolecule count&quot;. 
If statRes data is included, &quot;anova p-value&quot; and &quot;fold change&quot; data per comparisons
may be added. If grouping information is included, only &quot;sample count&quot; and 
&quot;mean abundance&quot; will be calculated, along with &quot;anova p-value&quot; and &quot;fold change&quot;
if specified. &quot;anova p-value&quot; will not be included if paneling a trelliscope
display by a biomolecule class. Default is &quot;sample count&quot; and &quot;mean abundance&quot;.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;ylim(c(2,20))&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_include_points">include_points</code></td>
<td>
<p>Add points as a geom_jitter. Default is TRUE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_boxplot_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of boxplots that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

trelliData1 &lt;- as.trelliData.edata(e_data = pep_edata,
                                   edata_cname = "Peptide",
                                   omics_type = "pepData")
# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData2 &lt;- as.trelliData(omicsData = omicsData)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

# Build the abundance boxplot with an edata file where each panel is a biomolecule. 
trelli_panel_by(trelliData = trelliData1, panel = "Peptide") %&gt;% 
   trelli_abundance_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the abundance boxplot wher each panel is a sample.
# Include all applicable cognostics. Remove points. 
trelli_panel_by(trelliData = trelliData1, panel = "Sample") %&gt;% 
   trelli_abundance_boxplot(test_mode = TRUE, test_example = 1:10, 
                            include_points = FALSE,
                            cognostics = c("count", 
                                           "mean abundance", 
                                           "median abundance", 
                                           "cv abundance"),
                             path = tempdir()
                           )

# Build the abundance boxplot with an omicsData object.
# Let the panels be biomolecules. Here, grouping information is included.
trelli_panel_by(trelliData = trelliData2, panel = "Peptide") %&gt;% 
   trelli_abundance_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the abundance boxplot with an omicsData object. The panel is a biomolecule class,
# which is proteins in this case.
trelli_panel_by(trelliData = trelliData2, panel = "RazorProtein") %&gt;% 
   trelli_abundance_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the abundance boxplot with an omicsData and statRes object.
# Panel by a biomolecule, and add statistics data to the cognostics
trelli_panel_by(trelliData = trelliData4, panel = "Peptide") %&gt;%
   trelli_abundance_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir(),
                            cognostics = c("mean abundance", "anova p-value", "fold change"))

# Other options include modifying the ggplot  
trelli_panel_by(trelliData = trelliData1, panel = "Peptide") %&gt;% 
   trelli_abundance_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir(),
     ggplot_params = c("ylab('')", "ylim(c(20,30))"))

# Or making the plot interactive 
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein") %&gt;% 
    trelli_abundance_boxplot(
     interactive = TRUE, test_mode = TRUE, test_example = 1:10, path = tempdir())


}


</code></pre>

<hr>
<h2 id='trelli_abundance_heatmap'>Heatmap trelliscope building function for abundance data</h2><span id='topic+trelli_abundance_heatmap'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the abundance heatmap
trelliscope. Data must be grouped by an e_meta column. Main_effects order
the y-variables. All statRes data is ignored. For RNA-Seq data, use &quot;trelli_rnaseq_heatmap&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_abundance_heatmap(
  trelliData,
  cognostics = c("sample count", "mean abundance", "biomolecule count"),
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_abundance_heatmap_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData, and
grouped by an emeta variable. Required.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options. Defaults are &quot;sample count&quot;, 
&quot;mean abundance&quot; and &quot;biomolecule count&quot;. &quot;sample count&quot; and &quot;mean abundance&quot;
are reported per group, and &quot;biomolecule count&quot; is the total number of biomolecules
in the biomolecule class (e_meta column).</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of heatmaps that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData2 &lt;- as.trelliData(omicsData = omicsData)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

# Build the abundance heatmap with an omicsData object with emeta variables. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData2, panel = "RazorProtein") %&gt;%
   trelli_abundance_heatmap(test_mode = TRUE, test_example = 1:3, path = tempdir())
   
# Users can modify the plotting function with ggplot parameters and interactivity, 
# and can also select certain cognostics.     
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein") %&gt;% 
   trelli_abundance_heatmap(
     test_mode = TRUE, test_example = 1:5, 
     ggplot_params = c("ylab('')", "xlab('')"), 
     interactive = TRUE, cognostics = c("biomolecule count"),
     path = tempdir()
   )


}


</code></pre>

<hr>
<h2 id='trelli_abundance_histogram'>Histogram trelliscope building function for abundance data</h2><span id='topic+trelli_abundance_histogram'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the abundance histogram
trelliscope. Main_effects grouping are ignored. Data must be grouped by
edata_cname. For RNA-Seq data, use &quot;trelli_rnaseq_histogram&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_abundance_histogram(
  trelliData,
  cognostics = c("sample count", "mean abundance", "median abundance", "cv abundance",
    "skew abundance"),
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_abundance_histogram_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData or
as.trelliData.edata, and grouped by edata_cname in trelli_panel_by.
Required.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries
are &quot;sample count&quot;, &quot;mean abundance&quot;, &quot;median abundance&quot;, &quot;cv abundance&quot;, 
and &quot;skew abundance&quot;. All are included by default.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;ylim(c(1,2))&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_abundance_histogram_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of histograms that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

trelliData1 &lt;- as.trelliData.edata(e_data = pep_edata,
                                   edata_cname = "Peptide",
                                   omics_type = "pepData")
# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData2 &lt;- as.trelliData(omicsData = omicsData)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

# Build the abundance histogram with an edata file. 
# Generate trelliData in as.trelliData.edata
trelli_panel_by(trelliData = trelliData1, panel = "Peptide") %&gt;% 
   trelli_abundance_histogram(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the abundance histogram with an omicsData object. 
# Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData2, panel = "Peptide") %&gt;% 
   trelli_abundance_histogram(test_mode = TRUE, test_example = 1:10, path = tempdir())
    
# Build the abundance histogram with an omicsData and statRes object. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData4, panel = "Peptide") %&gt;%
   trelli_abundance_histogram(
     test_mode = TRUE, test_example = 1:10, cognostics = "sample count", path = tempdir())
   
# Users can modify the plotting function with ggplot parameters and interactivity, 
# and can also select certain cognostics.     
trelli_panel_by(trelliData = trelliData1, panel = "Peptide") %&gt;% 
   trelli_abundance_histogram(test_mode = TRUE, test_example = 1:10, 
     ggplot_params = c("ylab('')", "xlab('Abundance')"), interactive = TRUE,
     cognostics = c("mean abundance", "median abundance"), path = tempdir())  
 

}


</code></pre>

<hr>
<h2 id='trelli_foldchange_bar'>Bar chart trelliscope building function for fold_change</h2><span id='topic+trelli_foldchange_bar'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the fold_change
barchart trelliscope. Fold change must be grouped by edata_cname.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_foldchange_bar(
  trelliData,
  cognostics = c("fold change", "p-value"),
  p_value_thresh = 0.05,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_foldchange_bar_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object with statRes results. Required.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries
and the defaults are &quot;fold change&quot; and &quot;p-value&quot;. If the omics data is MS/NMR,
p-value will be the results from the ANOVA test. If the omics data is sedData,
the p-value will be the results from the function &quot;diffexp_seq&quot;.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_p_value_thresh">p_value_thresh</code></td>
<td>
<p>A value between 0 and 1 to indicate significant
biomolecules for p_value_test. Default is 0.05.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_bar_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of fold_change bar plots that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData3 &lt;- as.trelliData(statRes = statRes)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

# Build fold_change bar plot with statRes data grouped by edata_colname.
trelli_panel_by(trelliData = trelliData3, panel = "Peptide") %&gt;% 
  trelli_foldchange_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())


}


</code></pre>

<hr>
<h2 id='trelli_foldchange_boxplot'>Boxplot trelliscope building function for fold_changes</h2><span id='topic+trelli_foldchange_boxplot'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the fold_change boxplot
trelliscope. Fold change must be grouped by an emeta column, which means
both an omicsData object and statRes are required to make this plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_foldchange_boxplot(
  trelliData,
  cognostics = "biomolecule count",
  p_value_thresh = 0.05,
  include_points = TRUE,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_foldchange_boxplot_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object with omicsData and statRes
results. Required.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries are 
&quot;biomolecule count&quot;, &quot;proportion significant&quot;, &quot;mean fold change&quot;,
and &quot;sd fold change&quot;. Default is &quot;biomolecule count&quot;.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_p_value_thresh">p_value_thresh</code></td>
<td>
<p>A value between 0 and 1 to indicate significant
biomolecules for the anova (MS/NMR) or diffexp_seq (RNA-seq) test. Default is 0.05.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_include_points">include_points</code></td>
<td>
<p>Add points. Default is TRUE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_boxplot_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of fold_change boxplots that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 
if (interactive()) {
library(pmartRdata)

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

# Build fold_change box plot with statRes data grouped by edata_colname.
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein") %&gt;% 
  trelli_foldchange_boxplot(test_mode = TRUE, 
                            test_example = 1:10,
                            cognostics = c("biomolecule count", 
                                           "proportion significant", 
                                           "mean fold change",
                                           "sd fold change"),
                            path = tempdir()
                           )
                           
                           
#####################
## RNA-SEQ EXAMPLE ##                            
#####################

# Build fold_change box plot with statRes data grouped by edata_colname.
trelli_panel_by(trelliData = trelliData_seq4, panel = "Gene") %&gt;% 
  trelli_foldchange_boxplot(test_mode = TRUE, 
                            test_example = c(16823, 16890, 17680, 17976, 17981, 19281),
                            cognostics = c("biomolecule count", 
                                           "proportion significant", 
                                           "mean fold change",
                                           "sd fold change"),
                            path = tempdir()
                           )


}


</code></pre>

<hr>
<h2 id='trelli_foldchange_heatmap'>Heatmap trelliscope building function for fold_change</h2><span id='topic+trelli_foldchange_heatmap'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the fold_change heatmap
trelliscope. Fold change must be grouped by an emeta column, which means
both an omicsData object and statRes are required to make this plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_foldchange_heatmap(
  trelliData,
  cognostics = "biomolecule count",
  p_value_thresh = 0.05,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_foldchange_heatmap_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object with omicsData and statRes
results. Required.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries are 
&quot;biomolecule count&quot;, &quot;proportion significant&quot;, &quot;mean fold change&quot;,
and &quot;sd fold change&quot;. Default is &quot;biomolecule count&quot;.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_p_value_thresh">p_value_thresh</code></td>
<td>
<p>A value between 0 and 1 to indicate significant biomolecules 
for the anova (MS/NMR) or diffexp_seq (RNA-seq) test. Default is 0.05.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of fold-change heatmaps that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

##########################
## MS/NMR OMICS EXAMPLE ##
##########################

# Build fold_change bar plot with statRes data grouped by edata_colname.
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein") %&gt;% 
  trelli_foldchange_heatmap(test_mode = TRUE, 
                            test_example = 1:10,
                            path = tempdir())


}


</code></pre>

<hr>
<h2 id='trelli_foldchange_volcano'>Volcano trelliscope building function for fold_change</h2><span id='topic+trelli_foldchange_volcano'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the fold_change volcano
trelliscope. Fold change must be grouped by an emeta column, which means
both an omicsData object and statRes are required to make this plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_foldchange_volcano(
  trelliData,
  comparison = "all",
  cognostics = "biomolecule count",
  p_value_thresh = 0.05,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_foldchange_volcano_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object with omicsData and statRes
results. Required.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_comparison">comparison</code></td>
<td>
<p>The specific comparison to visualize in the fold_change
volcano. See attr(statRes, &quot;comparisons&quot;) for the available options.
If all comparisons are desired, the word &quot;all&quot; can be used, which is the
default. Required.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries are 
&quot;biomolecule count&quot;, &quot;proportion significant&quot;, &quot;proportion significant up&quot;,
and &quot;proportion significant down&quot;. Default is &quot;biomolecule count&quot;.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_p_value_thresh">p_value_thresh</code></td>
<td>
<p>A value between 0 and 1 to indicate significant
biomolecules for p_value_test. Default is 0.05.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_foldchange_volcano_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of fold-change volcano plots that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)
## Build fold_change bar plot with statRes data grouped by edata_colname.
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein") %&gt;% 
  trelli_foldchange_volcano(comparison = "all", test_mode = TRUE, test_example = 1:10,
                            cognostics = c("biomolecule count", "proportion significant"),
                            path = tempdir())


}


</code></pre>

<hr>
<h2 id='trelli_missingness_bar'>Bar chart trelliscope building function for missing data</h2><span id='topic+trelli_missingness_bar'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the missing barchart
trelliscope. Missingness is displayed per panel_by variable. Main_effects
data is used to split samples when applicable. For RNA-Seq data, use 
&quot;trelli rnaseq nonzero bar&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_missingness_bar(
  trelliData,
  cognostics = c("total count", "observed count", "observed proportion"),
  proportion = TRUE,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_missingness_bar_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData.edata or
as.trelliData. Required.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Defaults are &quot;total count&quot;,
&quot;observed count&quot;, and &quot;observed proportion&quot;. If grouping
data is included, all cognostics will be reported per group. If the 
trelliData is paneled by a biomolecule, the counts and proportion we be 
samples. If paneled by a sample or biomolecule class, the counts and proportions
will be biomolecules. If statRes data is included, &quot;g-test p-value&quot; may 
be included.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_proportion">proportion</code></td>
<td>
<p>A logical to determine whether plots should display counts
or proportions. Default is TRUE.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_missingness_bar_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of missingness bar charts that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (interactive()) {
library(pmartRdata)

trelliData1 &lt;- as.trelliData.edata(e_data = pep_edata,
                                   edata_cname = "Peptide",
                                   omics_type = "pepData")
# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData2 &lt;- as.trelliData(omicsData = omicsData)
trelliData3 &lt;- as.trelliData(statRes = statRes)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

# Build the missingness bar plot with an edata file. Generate trelliData in as.trelliData.edata
trelli_panel_by(trelliData = trelliData1, panel = "Peptide") %&gt;% 
  trelli_missingness_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())
trelli_panel_by(trelliData = trelliData1, panel = "Sample") %&gt;% 
  trelli_missingness_bar(test_mode = TRUE, test_example = 1:10, 
   cognostics = "observed proportion", path = tempdir())

# Build the missingness bar plot with an omicsData object. Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData2, panel = "Peptide") %&gt;% 
  trelli_missingness_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the missingness bar plot with a statRes object. Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData3, panel = "Peptide") %&gt;%
  trelli_missingness_bar(test_mode = TRUE, test_example = 1:10, path = tempdir(),
                         cognostics = c("observed proportion", "g-test p-value"))

# Build the missingness bar plot with an omicsData and statRes object. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein") %&gt;%
  trelli_missingness_bar(test_mode = TRUE, test_example = 1:10, path = tempdir()) 

# Or making the plot interactive 
trelli_panel_by(trelliData = trelliData2, panel = "Peptide") %&gt;% 
   trelli_missingness_bar(
     test_mode = TRUE, test_example = 1:5, interactive = TRUE, path = tempdir())
   
# Or visualize only count data 
trelli_panel_by(trelliData = trelliData2, panel = "Peptide") %&gt;% 
   trelli_missingness_bar(
     test_mode = TRUE, test_example = 1:5, 
     cognostics = "observed count", proportion = FALSE,
     path = tempdir()
   )


}


</code></pre>

<hr>
<h2 id='trelli_panel_by'>Set the &quot;panel_by&quot; variable for a trelliData object</h2><span id='topic+trelli_panel_by'></span>

<h3>Description</h3>

<p>Allows for grouping omics or stats data for downstream plotting
and cognostic functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_panel_by(trelliData, panel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_panel_by_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData or as.trelliData.edata. Required.</p>
</td></tr>
<tr><td><code id="trelli_panel_by_+3A_panel">panel</code></td>
<td>
<p>The name of a column in trelliData to panel the data by. Required.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trelliData object with attributes &quot;panel_by_omics&quot; or &quot;panel_by_stat&quot; to determine 
which columns to divide the data by.
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

trelliData1 &lt;- as.trelliData.edata(e_data = pep_edata,
                                   edata_cname = "Peptide",
                                   omics_type = "pepData")
# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData2 &lt;- as.trelliData(omicsData = omicsData)
trelliData3 &lt;- as.trelliData(statRes = statRes)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

## "panel_by" with an edata file. 
trelli_panel_by(trelliData = trelliData1, panel = "Peptide")
trelli_panel_by(trelliData = trelliData1, panel = "Sample")

## "panel_by" with trelliData containing omicsData. 
## Generate trelliData2 using the example code for as.trelliData
trelli_panel_by(trelliData = trelliData2, panel = "Peptide")
trelli_panel_by(trelliData = trelliData2, panel = "RazorProtein")

## "panel_by" with trelliData containing statRes. 
## Generate trelliData3 using the example code for as.trelliData
trelli_panel_by(trelliData = trelliData3, panel = "Peptide")

## "panel_by" with trelliData containing both omicsData and statRes. 
## Generate trelliData4 using the example code for as.trelliData
trelli_panel_by(trelliData = trelliData4, panel = "Peptide")
trelli_panel_by(trelliData = trelliData4, panel = "RazorProtein")
trelli_panel_by(trelliData = trelliData4, panel = "SampleID")


</code></pre>

<hr>
<h2 id='trelli_precheck'>Performs initial checks for trelliData objects</h2><span id='topic+trelli_precheck'></span>

<h3>Description</h3>

<p>This function runs necessary checks for pmartR trelliscope plotting
functions. It cleans any parameters (rounding numerics to integers, etc.),
and returns them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_precheck(
  trelliData,
  trelliCheck,
  cognostics,
  acceptable_cognostics,
  ggplot_params,
  interactive,
  test_mode,
  test_example,
  single_plot,
  seqDataCheck,
  seqText = NULL,
  p_value_skip = FALSE,
  p_value_thresh = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_precheck_+3A_trellidata">trelliData</code></td>
<td>
<p>trelliData object the user passed to a plotting
function</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_trellicheck">trelliCheck</code></td>
<td>
<p>Check if the object type is supposed to be &quot;omics&quot;,
&quot;statRes&quot; or put a vector of both</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of the user provided cognstics</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_acceptable_cognostics">acceptable_cognostics</code></td>
<td>
<p>The acceptable cognostics for this plot</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>The vector of user provided ggplots</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_interactive">interactive</code></td>
<td>
<p>The user provided logical for whether the plot should be
interactive</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_test_mode">test_mode</code></td>
<td>
<p>The user provided logical for whether a smaller trelliscope
should be returned</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_test_example">test_example</code></td>
<td>
<p>The user provided vector of plot indices</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_single_plot">single_plot</code></td>
<td>
<p>The user provided logical for whether a single plot should
be returned</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_seqdatacheck">seqDataCheck</code></td>
<td>
<p>Whether seqData is permitted for this plot. &quot;no&quot; means that 
seqData cannot be used at all, &quot;permissible&quot; means that seqData can be used, 
and &quot;required&quot; means that seqData is required for the plotting function.</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_seqtext">seqText</code></td>
<td>
<p>Text that should appear when seqDataCheck is violated.</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_p_value_skip">p_value_skip</code></td>
<td>
<p>Whether to skip specific p_value checks.</p>
</td></tr>
<tr><td><code id="trelli_precheck_+3A_p_value_thresh">p_value_thresh</code></td>
<td>
<p>The user provided threshold for plotting significant
p-values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, validates a trelliData object before passing it to builder functions.
</p>

<hr>
<h2 id='trelli_pvalue_filter'>Filter a paneled trelliData object by a p-value</h2><span id='topic+trelli_pvalue_filter'></span>

<h3>Description</h3>

<p>This use-case-specific function allows users to filter down their plots to a 
specified p-value IF statistics data has been included. This function is mostly
relevant to the MODE application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_pvalue_filter(
  trelliData,
  p_value_test = "anova",
  p_value_thresh = 0.05,
  comparison = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_pvalue_filter_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliData object with statistics results (statRes). Required.</p>
</td></tr>
<tr><td><code id="trelli_pvalue_filter_+3A_p_value_test">p_value_test</code></td>
<td>
<p>A string to indicate which p_values to plot. Acceptable
entries are &quot;anova&quot; or &quot;gtest&quot;. Default is &quot;anova&quot;. Unlike the
plotting functions, here p_value_test cannot be null. Required unless
the data is seqData, when this parameter will be ignored.</p>
</td></tr>
<tr><td><code id="trelli_pvalue_filter_+3A_p_value_thresh">p_value_thresh</code></td>
<td>
<p>A value between 0 and 1 to indicate the p-value threshold 
at which to keep plots. Default is 0.05. Required.</p>
</td></tr>
<tr><td><code id="trelli_pvalue_filter_+3A_comparison">comparison</code></td>
<td>
<p>The specific comparison to filter significant values to. Can
be null. See attr(statRes, &quot;comparisons&quot;) for the available options. Optional.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A paneled trelliData object with only plots corresponding to significant
p-values from a statistical test.
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(pmartRdata)

# Transform the data
omicsData &lt;- edata_transform(omicsData = pep_object, data_scale = "log2")

# Group the data by condition
omicsData &lt;- group_designation(omicsData = omicsData, main_effects = c("Phenotype"))

# Apply the IMD ANOVA filter
imdanova_Filt &lt;- imdanova_filter(omicsData = omicsData)
omicsData &lt;- applyFilt(filter_object = imdanova_Filt, omicsData = omicsData,
                       min_nonmiss_anova = 2)

# Normalize my pepData
omicsData &lt;- normalize_global(omicsData, "subset_fn" = "all", "norm_fn" = "median",
                             "apply_norm" = TRUE, "backtransform" = TRUE)

# Implement the IMD ANOVA method and compute all pairwise comparisons 
# (i.e. leave the `comparisons` argument NULL)
statRes &lt;- imd_anova(omicsData = omicsData, test_method = 'combined')

# Generate the trelliData object
trelliData3 &lt;- as.trelliData(statRes = statRes)
trelliData4 &lt;- as.trelliData(omicsData = omicsData, statRes = statRes)

###########################
## MS/NMR OMICS EXAMPLES ##
###########################

# Filter a trelliData object with only statistics results, while not caring about a comparison
trelli_pvalue_filter(trelliData3, p_value_test = "anova", p_value_thresh = 0.1)

# Filter a trelliData object with only statistics results, while caring about a specific comparison
trelli_pvalue_filter(
 trelliData3, p_value_test = "anova", p_value_thresh = 0.1, comparison = "Phenotype3_vs_Phenotype2")

# Filter both a omicsData and statRes object, while not caring about a specific comparison
trelli_pvalue_filter(trelliData4, p_value_test = "anova", p_value_thresh = 0.001)

# Filter both a omicsData and statRes object, while caring about a specific comparison
trelli_pvalue_filter(
 trelliData4, p_value_test = "gtest", p_value_thresh = 0.25, 
 comparison = "Phenotype3_vs_Phenotype2"
)

######################
## RNA-SEQ EXAMPLES ##  
######################

#' # Group data by condition
omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(
 filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15
)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq3 &lt;- as.trelliData(statRes = statRes_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)

# Filter a trelliData seqData object with only statistics results, while not 
# caring about a comparison
trelliData_seq3_filt &lt;- trelli_pvalue_filter(trelliData_seq3, p_value_thresh = 0.05)

# Filter both a omicsData and statRes object, while caring about a specific comparison
trelliData_seq4_filt &lt;- trelli_pvalue_filter(trelliData_seq4, p_value_thresh = 0.05, 
 comparison = "StrainA_vs_StrainB")



</code></pre>

<hr>
<h2 id='trelli_rnaseq_boxplot'>Boxplot trelliscope building function for RNA-seq data</h2><span id='topic+trelli_rnaseq_boxplot'></span>

<h3>Description</h3>

<p>Specify a boxplot design and cognostics for the RNA-Seq
boxplot trelliscope. Each boxplot will have its own groups as specified by
the first main effect in group_designation. Use &quot;trelli_abundance_boxplot&quot;
for MS/NMR-based omics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_rnaseq_boxplot(
  trelliData,
  cognostics = c("count", "mean lcpm"),
  ggplot_params = NULL,
  interactive = FALSE,
  include_points = TRUE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData or
as.trelliData.edata, and grouped by trelli_panel_by. Must be built using 
seqData. Required.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries
are &quot;count&quot;, &quot;mean lcpm&quot;, &quot;median lcpm&quot;, and &quot;cv lcpm&quot;. 
If data are paneled by a biomolecule, the count will be &quot;sample count&quot;.
If data are paneled by a sample or a biomolecule class, the count will be &quot;biomolecule count&quot;. 
If statRes data is included, &quot;p-value&quot; and &quot;fold change&quot; data per comparisons
may be added. If grouping information is included, only &quot;sample count&quot; and 
&quot;mean lcpm&quot; will be calculated, along with &quot;p-value&quot; and &quot;fold change&quot;
if specified. &quot;p-value&quot; will not be included if paneling a trelliscope
display by a biomolecule class. Default is &quot;sample count&quot; and &quot;mean lcpm&quot;.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;ylim(c(2,20))&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_include_points">include_points</code></td>
<td>
<p>Add points as a geom_jitter. Default is TRUE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_boxplot_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of boxplots that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(pmartRdata)

trelliData_seq1 &lt;- as.trelliData.edata(e_data = rnaseq_edata,
                                      edata_cname = "Transcript",
                                      omics_type = "seqData")
omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq2 &lt;- as.trelliData(omicsData = omicsData_seq)
trelliData_seq3 &lt;- as.trelliData(statRes = statRes_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)

## Generate trelliData objects using the as.trelliData.edata example code.

# Build the RNA-seq boxplot with an edata file where each panel is a biomolecule. 
trelli_panel_by(trelliData = trelliData_seq1, panel = "Transcript") %&gt;% 
   trelli_rnaseq_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir())
   
# Build the RNA-seq boxplot where each panel is a sample.
# Include all applicable cognostics. Remove points. 
trelli_panel_by(trelliData = trelliData_seq1, panel = "Sample") %&gt;% 
   trelli_rnaseq_boxplot(test_mode = TRUE, test_example = 1:10, 
                            include_points = FALSE,
                            cognostics = c("count", 
                                           "mean lcpm", 
                                           "median lcpm", 
                                           "cv lcpm"),
                            path = tempdir()
                           )

# Build the RNA-seq boxplot with an omicsData object.
# Let the panels be biomolecules. Here, grouping information is included.
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
   trelli_rnaseq_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir())
   
# Build the RNA-seq boxplot with an omicsData object. The panel is a biomolecule class,
# which is proteins in this case.
trelli_panel_by(trelliData = trelliData_seq2, panel = "Gene") %&gt;% 
   trelli_rnaseq_boxplot(test_mode = TRUE, test_example = 1:10, path = tempdir())
    
# Build the RNA-seq boxplot with an omicsData and statRes object.
# Panel by a biomolecule, and add statistics data to the cognostics
trelli_panel_by(trelliData = trelliData_seq4, panel = "Transcript") %&gt;%
   trelli_rnaseq_boxplot(test_mode = TRUE, test_example = 1:10,
     cognostics = c("mean lcpm", "p-value", "fold change"), path = tempdir())
 
# Other options include modifying the ggplot  
trelli_panel_by(trelliData = trelliData_seq1, panel = "Transcript") %&gt;%
   trelli_rnaseq_boxplot(test_mode = TRUE, test_example = 1:10,
     ggplot_params = c("ylab('')", "xlab('')"), path = tempdir())

# Or making the plot interactive 
trelli_panel_by(trelliData = trelliData_seq4, panel = "Gene") %&gt;%
    trelli_rnaseq_boxplot(interactive = TRUE, test_mode = TRUE, 
     test_example = 1:10, path = tempdir())

\dontshow{closeAllConnections()}

## End(Not run)

</code></pre>

<hr>
<h2 id='trelli_rnaseq_heatmap'>Heatmap trelliscope building function for RNA-seq data</h2><span id='topic+trelli_rnaseq_heatmap'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the RNA-seq heatmap
trelliscope. Data must be grouped by an e_meta column. Main_effects order
the y-variables. All statRes data is ignored. 
For MS/NMR data, use &quot;trelli_abundance_heatmap&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_rnaseq_heatmap(
  trelliData,
  cognostics = c("sample count", "mean LCPM", "biomolecule count"),
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData, and
grouped by an emeta variable. Must be built using seqData. Required.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options. Defaults are &quot;sample count&quot;, 
&quot;mean LCPM&quot; and &quot;biomolecule count&quot;. &quot;sample count&quot; and &quot;mean LCPM&quot;
are reported per group, and &quot;biomolecule count&quot; is the total number of biomolecules
in the biomolecule class (e_meta column).</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of heatmaps that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(pmartRdata)

omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq2 &lt;- as.trelliData(omicsData = omicsData_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)

# Build the RNA-seq heatmap with an omicsData object with emeta variables. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData_seq2, panel = "Gene") %&gt;% 
   trelli_rnaseq_heatmap(test_mode = TRUE, test_example = c(1532, 1905, 6134), path = tempdir())
   
# Users can modify the plotting function with ggplot parameters and interactivity, 
# and can also select certain cognostics.     
trelli_panel_by(trelliData = trelliData_seq4, panel = "Gene") %&gt;% 
   trelli_rnaseq_heatmap(test_mode = TRUE, test_example = c(1532, 1905, 6134), 
     ggplot_params = c("ylab('')", "xlab('')"), 
     interactive = TRUE, cognostics = c("biomolecule count"), path = tempdir())  

\dontshow{closeAllConnections()}

## End(Not run)

</code></pre>

<hr>
<h2 id='trelli_rnaseq_histogram'>Histogram trelliscope building function for RNA-Seq data</h2><span id='topic+trelli_rnaseq_histogram'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the abundance histogram
trelliscope. Main_effects grouping are ignored. Data must be grouped by
edata_cname. For MS/NMR data, use &quot;trelli_abundance_histogram&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_rnaseq_histogram(
  trelliData,
  cognostics = c("sample count", "mean lcpm", "median lcpm", "cv lcpm", "skew lcpm"),
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_rnaseq_histogram_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData or
as.trelliData.edata, and grouped by edata_cname in trelli_panel_by.
Must be built using seqData. Required.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Valid entries
are &quot;sample count&quot;, &quot;mean lcpm&quot;, &quot;median lcpm&quot;, &quot;cv lcpm&quot;, 
and &quot;skew lcpm&quot;. All are included by default.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;ylim(c(1,2))&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_histogram_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of histograms that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(pmartRdata)

trelliData_seq1 &lt;- as.trelliData.edata(e_data = rnaseq_edata, 
                                      edata_cname = "Transcript",
                                      omics_type = "seqData")
omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq2 &lt;- as.trelliData(omicsData = omicsData_seq)
trelliData_seq3 &lt;- as.trelliData(statRes = statRes_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)

# Build the RNA-seq histogram with an edata file. 
# Generate trelliData in as.trelliData.edata
trelli_panel_by(trelliData = trelliData_seq1, panel = "Transcript") %&gt;% 
   trelli_rnaseq_histogram(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the RNA-seq histogram with an omicsData object. 
# Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
   trelli_rnaseq_histogram(test_mode = TRUE, test_example = 1:10, path = tempdir())
    
# Build the RNA-seq histogram with an omicsData and statRes object. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData_seq4, panel = "Transcript") %&gt;%
   trelli_rnaseq_histogram(test_mode = TRUE, test_example = 1:10, 
     cognostics = "sample count", path = tempdir())
   
# Users can modify the plotting function with ggplot parameters and interactivity, 
# and can also select certain cognostics.     
trelli_panel_by(trelliData = trelliData_seq1, panel = "Transcript") %&gt;% 
   trelli_rnaseq_histogram(test_mode = TRUE, test_example = 1:10, 
     ggplot_params = c("ylab('')", "xlab('')"), interactive = TRUE,
     cognostics = c("mean lcpm", "median lcpm"), path = tempdir())  

\dontshow{closeAllConnections()}

## End(Not run)

</code></pre>

<hr>
<h2 id='trelli_rnaseq_nonzero_bar'>Bar chart trelliscope building function for Non-Zero counts in RNA-seq data</h2><span id='topic+trelli_rnaseq_nonzero_bar'></span>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the Non-Zero barchart
trelliscope. Non-Zeroes are displayed per panel_by variable. Main_effects
data is used to split samples when applicable. For MS/NMR data, use 
&quot;trelli missingness bar&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trelli_rnaseq_nonzero_bar(
  trelliData,
  cognostics = c("total count", "non-zero count", "non-zero proportion"),
  proportion = TRUE,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_trellidata">trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData.edata or
as.trelliData. Must be built using seqData. Required.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_cognostics">cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Defaults are &quot;total count&quot;,
&quot;non-zero count&quot;, and &quot;non-zero proportion&quot;. If grouping
data is included, all cognostics will be reported per group. If the 
trelliData is paneled by a biomolecule, the counts and proportion we be 
samples. If paneled by a sample or biomolecule class, the counts and proportions
will be biomolecules.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_proportion">proportion</code></td>
<td>
<p>A logical to determine whether plots should display counts
or proportions. Default is TRUE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_ggplot_params">ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c(&quot;ylab(&rdquo;)&quot;, &quot;xlab(&rdquo;)&quot;).
Default is NULL.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_interactive">interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_path">path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_name">name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_test_mode">test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_test_example">test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_single_plot">single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td></tr>
<tr><td><code id="trelli_rnaseq_nonzero_bar_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, builds a trelliscope display of bar charts that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(pmartRdata)

trelliData_seq1 &lt;- as.trelliData.edata(e_data = rnaseq_edata, 
                                      edata_cname = "Transcript",
                                      omics_type = "seqData")
omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq2 &lt;- as.trelliData(omicsData = omicsData_seq)
trelliData_seq3 &lt;- as.trelliData(statRes = statRes_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)

# Build the non-zero bar plot with an edata file. Generate trelliData in as.trelliData.edata
trelli_panel_by(trelliData = trelliData_seq1, panel = "Transcript") %&gt;% 
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())
trelli_panel_by(trelliData = trelliData_seq1, panel = "Sample") %&gt;% 
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, 
   cognostics = "non-zero proportion", path = tempdir())

# Build the non-zero bar plot with an omicsData object. Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the non-zero bar plot with a statRes object. Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData_seq3, panel = "Transcript") %&gt;%
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10,
                         cognostics = c("non-zero proportion"), path = tempdir())

# Build the non-zero bar plot with an omicsData and statRes object. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData_seq4, panel = "Gene") %&gt;%
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, path = tempdir()) 

# Or making the plot interactive 
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
   trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:5, 
     interactive = TRUE, path = tempdir())
   
# Or visualize only count data 
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
   trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:5, 
     cognostics = "non-zero count", proportion = FALSE, path = tempdir())

\dontshow{closeAllConnections()}   

## End(Not run)

</code></pre>

<hr>
<h2 id='vector_replace'>Replace x with y for a single vector</h2><span id='topic+vector_replace'></span>

<h3>Description</h3>

<p>Replace x with y for a single vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vector_replace(one_vector, x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vector_replace_+3A_one_vector">one_vector</code></td>
<td>
<p>numeric vector</p>
</td></tr>
<tr><td><code id="vector_replace_+3A_x">x</code></td>
<td>
<p>value to be replaced</p>
</td></tr>
<tr><td><code id="vector_replace_+3A_y">y</code></td>
<td>
<p>replacement value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector
</p>


<h3>Author(s)</h3>

<p>Kelly Stratton
</p>

<hr>
<h2 id='voom_wrapper'>Wrapper for limma-voom workflow</h2><span id='topic+voom_wrapper'></span>

<h3>Description</h3>

<p>For generating statistics for 'seqData' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>voom_wrapper(
  omicsData,
  p_adjust = "BH",
  comparisons = NULL,
  p_cutoff = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="voom_wrapper_+3A_omicsdata">omicsData</code></td>
<td>
<p>an object of type 'seqData', created by
<code><a href="#topic+as.seqData">as.seqData</a></code></p>
</td></tr>
<tr><td><code id="voom_wrapper_+3A_p_adjust">p_adjust</code></td>
<td>
<p>Character string for p-value correction method, refer to
?p.adjust() for valid options</p>
</td></tr>
<tr><td><code id="voom_wrapper_+3A_comparisons">comparisons</code></td>
<td>
<p>'data.frame' with columns for &quot;Control&quot; and &quot;Test&quot;
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td></tr>
<tr><td><code id="voom_wrapper_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>Numeric value between 0 and 1 for setting p-value
significance threshold</p>
</td></tr>
<tr><td><code id="voom_wrapper_+3A_...">...</code></td>
<td>
<p>additional arguments passed to methods functions. Note, formatting
option changes will interfere with wrapping functionality.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Runs default limma-voom workflow using empirical Bayes moderated
t-statistics. Additional arguments can be passed for use in the function,
refer to calcNormFactors() in edgeR package.
</p>


<h3>Value</h3>

<p>statRes object
</p>


<h3>References</h3>

<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth,
G.K. (2015). limma powers differential expression analyses for
RNA-sequencing and microarray studies. Nucleic Acids Research 43(7), e47.
</p>

<hr>
<h2 id='zrollup'>Applies zrollup function</h2><span id='topic+zrollup'></span>

<h3>Description</h3>

<p>This function applies the zrollup method to a pepData object for each unique
protein and returns a proData object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zrollup(pepData, combine_fn, parallel = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zrollup_+3A_pepdata">pepData</code></td>
<td>
<p>an omicsData object of class 'pepData'</p>
</td></tr>
<tr><td><code id="zrollup_+3A_combine_fn">combine_fn</code></td>
<td>
<p>logical indicating what combine_fn to use, defaults to
median, other option is mean</p>
</td></tr>
<tr><td><code id="zrollup_+3A_parallel">parallel</code></td>
<td>
<p>logical indicating whether or not to use &quot;doParallel&quot; loop in
applying zrollup function. Defaults to TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the zrollup method, peptides are scaled as, pep_scaled = (pep -
median)/sd, and protein abundance is set as the mean of these scaled
peptides.
</p>


<h3>Value</h3>

<p>an omicsData object of class 'proData'
</p>


<h3>References</h3>

<p>Polpitiya, A. D., Qian, W.-J., Jaitly, N., Petyuk, V. A., Adkins,
J. N., Camp, D. G., ... Smith, R. D. (2008). <em>DAnTE: a statistical
tool for quantitative analysis of -omics data</em>. Bioinformatics (Oxford,
England), 24(13), 1556-1558.
</p>

<hr>
<h2 id='zscore_transform'>Z-Score Transformation</h2><span id='topic+zscore_transform'></span>

<h3>Description</h3>

<p>Calculate normalization parameters for the data via via z-score
transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zscore_transform(
  e_data,
  edata_id,
  subset_fn,
  feature_subset,
  backtransform = FALSE,
  apply_norm = FALSE,
  check.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zscore_transform_+3A_e_data">e_data</code></td>
<td>
<p>a <code class="reqn">p \times n + 1</code> data.frame, where <code class="reqn">p</code> is the number
of peptides, lipids, or metabolites and <code class="reqn">n</code> is the number of samples.
Each row corresponds to data for a peptide, protein, lipid, or metabolite,
with one column giving the biomolecule identifier name.</p>
</td></tr>
<tr><td><code id="zscore_transform_+3A_edata_id">edata_id</code></td>
<td>
<p>character string indicating the name of the peptide, protein,
lipid, or metabolite identifier. Usually obtained by calling
<code>attr(omicsData, "cnames")$edata_cname</code>.</p>
</td></tr>
<tr><td><code id="zscore_transform_+3A_subset_fn">subset_fn</code></td>
<td>
<p>character string indicating the subset function to use for
normalization</p>
</td></tr>
<tr><td><code id="zscore_transform_+3A_feature_subset">feature_subset</code></td>
<td>
<p>character vector containing the feature names in the
subset to be used for normalization</p>
</td></tr>
<tr><td><code id="zscore_transform_+3A_backtransform">backtransform</code></td>
<td>
<p>logical argument. If TRUE, the data will be back
transformed after normalization so that the values are on a scale similar
to their raw values. See details for more information. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="zscore_transform_+3A_apply_norm">apply_norm</code></td>
<td>
<p>logical argument. If TRUE, the normalization will be
applied to the data. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="zscore_transform_+3A_check.names">check.names</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each feature is scaled by subtracting the mean of the feature subset
specified for normalization and then dividing the result by the standard
deviation (SD) of the feature subset specified for normalization to get the
normalized data. The location estimates are the subset means for each
sample. The scale estimates are the subset SDs for each sample. If
backtransform is TRUE, the normalized feature values are multiplied by a
pooled standard deviation (estimated across all samples) and a global mean
of the subset data (across all samples) is added back to the normalized
values. Means are taken ignoring any NA values.
</p>


<h3>Value</h3>

<p>List containing two elements: <code>norm_params</code> is list with two
elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> numeric vector of length <code>n</code> standard deviations for each sample </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric vector of length <code>n</code> means for each sample
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>backtransform_params</code> is a list with two elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
scale </td><td style="text-align: left;"> numeric value giving the pooled standard deviation across all samples </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
location </td><td style="text-align: left;"> numeric value giving global mean across all samples
</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If <code>backtransform</code> is set to TRUE then each list item under
<code>backtransform_params</code> will be NULL.
</p>
<p>If <code>apply_norm</code> is TRUE, the transformed data is returned as a third
list item.
</p>


<h3>Author(s)</h3>

<p>Lisa Bramer, Kelly Stratton, Bryan Stanfill
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
