<!DOCTYPE html><html><head><title>Help for package PoiClaClu</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PoiClaClu}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PoiClaClu-package'>
<p>Classification and clustering of RNA sequencing data using a Poisson model</p></a></li>
<li><a href='#Classify'>
<p>Classify observations using a Poisson model.</p></a></li>
<li><a href='#Classify.cv'>
<p>Function to do cross-validation for Poisson classification.</p></a></li>
<li><a href='#ColorDendrogram'><p>Color the leaves in a hierarchical clustering dendrogram</p></a></li>
<li><a href='#CountDataSet'>
<p>Generate a simulated sequencing data set using a negative binomial model.</p></a></li>
<li><a href='#FindBestTransform'>
<p>Find the power transformation that makes a data set approximately Poisson.</p></a></li>
<li><a href='#PoiClaClu-internal'><p>Internal PoiClaClu functions</p></a></li>
<li><a href='#PoissonDistance'>
<p>Given a n-by-p data matrix, compute the corresponding n-by-n Poisson dissimilarity matrix.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Classification and Clustering of Sequencing Data Based on a
Poisson Model</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2013-01-02</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniela Witten</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniela Witten &lt;dwitten@u.washington.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the methods described in the paper, Witten (2011) Classification and Clustering of Sequencing Data using a Poisson Model, Annals of Applied Statistics 5(4) 2493-2518.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-01-04 17:04:11 UTC; hornik</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-01-04 17:05:57 UTC</td>
</tr>
</table>
<hr>
<h2 id='PoiClaClu-package'>
Classification and clustering of RNA sequencing data using a Poisson model
</h2><span id='topic+PoiClaClu-package'></span><span id='topic+PoiClaClu'></span>

<h3>Description</h3>

<p>A simple approach for performing classification and clustering of
samples for which RNA sequencing data is available. Based upon a
simple Poisson model proposed by a number of authors (e.g. Marioni et
al Genome Research 2008, Bullard et al BMC Bioinformatics 2010, and
others).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PoiClaClu</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2013-01-02</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Daniela Witten
</p>
<p>Maintainer: Daniela Witten &lt;dwitten@u.washington.edu&gt;
</p>


<h3>References</h3>

<p>D. Witten (2011) Classification and clustering of sequencing data
using a Poisson model. Annals of Applied Statistics 5(4): 2493-2518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Poisson clustering #
set.seed(1)
dat &lt;- CountDataSet(n=20,p=100,sdsignal=.5,K=4,param=10)
dd &lt;- PoissonDistance(dat$x, type="mle")
print(dd)
ColorDendrogram(hclust(dd$dd), y=dat$y)

# Poisson classification #
set.seed(1)
dat &lt;- CountDataSet(n=20,p=100,sdsignal=.1,K=4,param=10)
out &lt;- Classify(x=dat$x,y=dat$y,xte=dat$xte,rhos=c(0,5,10))
print(out)

</code></pre>

<hr>
<h2 id='Classify'>
Classify observations using a Poisson model.
</h2><span id='topic+Classify'></span>

<h3>Description</h3>

<p>Classify observations using a simple Poisson model. This function
implements the &quot;sparse Poisson linear discriminant analysis classifier&quot;,
which is similar to linear discriminant analysis but assumes a Poisson
model rather than a Gaussian model for the data. The classifier
soft-thresholds the estimated effect of each feature in order to achieve sparsity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Classify(x, y, xte=NULL, rho = 0, beta = 1, rhos = NULL, type=c("mle","deseq","quantile"),
prior = NULL, transform=TRUE, alpha=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Classify_+3A_x">x</code></td>
<td>

<p>A n-by-p training data matrix; n observations and p features. Used to train the
classifier.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_y">y</code></td>
<td>

<p>A numeric vector of class labels of length n: 1, 2, ...., K if there
are K classes.
Each element of y corresponds to a row of x; i.e. these are the
class labels for the observations in x.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_xte">xte</code></td>
<td>

<p>A m-by-p data matrix: m test observations and p features. The classifier
fit on the training data set x will be tested on this data set. If NULL,
then testing will be performed on the training set.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_rho">rho</code></td>
<td>

<p>Tuning parameter controlling the amount of soft thresholding performed,
i.e. the level of sparsity, i.e. number of nonzero features in
classifier. Rho=0 means that there is no soft-thresolding, i.e. all
features used in classifier. Larger rho means that fewer features will
be used.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_beta">beta</code></td>
<td>

<p>A smoothing term. A Gamma(beta,beta) prior is used to fit the
Poisson model.
Recommendation is to just leave it at 1, the default value.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_rhos">rhos</code></td>
<td>

<p>A vector of tuning parameters that control the amount of soft thresholding
performed. If &quot;rhos&quot; is provided then a number of
models will be fit (one for each element of &quot;rhos&quot;), and a number of
predicted class labels will be output (one for each element of &quot;rhos&quot;).
</p>
</td></tr>
<tr><td><code id="Classify_+3A_type">type</code></td>
<td>

<p>How should the observations be normalized within the
Poisson model, i.e. how should the size factors be estimated?
Options are &quot;quantile&quot; or &quot;deseq&quot; (more robust) or &quot;mle&quot; (less
robust).
</p>
<p>In greater detail: &quot;quantile&quot; is quantile normalization approach
of Bullard et al 2010 BMC Bioinformatics, &quot;deseq&quot;  is median of the
ratio of an observation to a pseudoreference obtained by taking the
geometric mean, described in Anders and Huber 2010 Genome Biology and
implemented in Bioconductor package &quot;DESeq&quot;, and &quot;mle&quot; is the sum of
counts for each sample; this is the maximum likelihood estimate
under a simple Poisson model.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_prior">prior</code></td>
<td>

<p>Vector of length equal to the number of classes, representing prior
probabilities
for each class. If NULL then uniform priors are used (i.e. each
class is equally likely).
</p>
</td></tr>
<tr><td><code id="Classify_+3A_transform">transform</code></td>
<td>

<p>Should data matrices x and xte first be power transformed so that it more
closely fits
the Poisson model? TRUE or FALSE. Power transformation is
especially
useful if the data are overdispersed relative to the Poisson
model.
</p>
</td></tr>
<tr><td><code id="Classify_+3A_alpha">alpha</code></td>
<td>

<p>If transform=TRUE, this determines the power to which the data
matrices x and xte are transformed.
If alpha=NULL then the transformation that
makes the Poisson model best fit the data matrix x is
computed. (Note that alpha is computed based on x, not based on
xte). Or a value of
alpha, 0&lt;alpha&lt;=1, can be entered by the user.
</p>
</td></tr>              
</table>


<h3>Value</h3>

<table>
<tr><td><code>ytehat</code></td>
<td>
<p>The predicted class labels for each of the test
observations (rows of xte). </p>
</td></tr>
<tr><td><code>discriminant</code></td>
<td>
<p>A m-by-K matrix, where K is the number of classes.
The (i,k) element is large if the ith element of
xte belongs to class k.</p>
</td></tr>
<tr><td><code>ds</code></td>
<td>
<p>A K-by-p matrix indicating the extent to which each feature
is under- or over-expressed in each class. The (k,j) element is &gt;1
if feature j is over-expressed in class k, and is &lt;1 if feature j is
under-expressed in class k. When rho is large then many of the elemtns
of this matrix are shrunken towards 1 (no over- or under-expression).</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Power transformation used (if transform=TRUE).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniela Witten
</p>


<h3>References</h3>

<p>D Witten (2011) Classification and clustering of sequencing data using a
Poisson model. To appear in Annals of Applied Statistics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Classify.cv">Classify.cv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
dat &lt;- CountDataSet(n=40,p=500,sdsignal=.1,K=3,param=10)
cv.out &lt;- Classify.cv(dat$x,dat$y)
print(cv.out)
out &lt;- Classify(dat$x,dat$y,dat$xte,rho=cv.out$bestrho)
print(out)
cat("Confusion matrix for predicted and true test class labels:", fill=TRUE)
print(table(out$ytehat,dat$yte))
</code></pre>

<hr>
<h2 id='Classify.cv'>
Function to do cross-validation for Poisson classification.
</h2><span id='topic+Classify.cv'></span>

<h3>Description</h3>

<p>Perform cross-validation for the function that 
implements the &quot;sparse Poisson linear discriminant analysis
classifier&quot;,
which is similar to linear discriminant analysis but assumes a Poisson
model rather than a Gaussian model for the data. The classifier
soft-thresholds the estimated effect of each feature in order to
achieve sparsity. This cross-validation function selects the proper
value of the tuning parameter that controls the level of soft-thresholding.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Classify.cv(x, y, rhos = NULL, beta = 1, nfolds = 5, type=c("mle","deseq","quantile"),
folds = NULL, transform=TRUE, alpha=NULL, prior=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Classify.cv_+3A_x">x</code></td>
<td>
<p>A n-by-p training data matrix; n observations and p features.
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_y">y</code></td>
<td>

<p>A numeric vector of class labels of length n: 1, 2, ...., K if there
are K classes.
Each element of y corresponds to a row of x; i.e. these are the
class labels for the observations in x.
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_rhos">rhos</code></td>
<td>

<p>A vector of tuning parameters to try out in cross-validation. Rho
controls the level of shrinkage performed, i.e. the number of features
that are not involved in the classifier. When rho=0 then all features
are involved in the classifier, and when rho is very large no features
are involved.
If rhos=NULL then a vector of rho values will be chosen automatically.
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_beta">beta</code></td>
<td>

<p>A smoothing term. A Gamma(beta,beta) prior is used to fit the
Poisson model.
Recommendation is to leave it at 1, the default value.
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_nfolds">nfolds</code></td>
<td>

<p>The number of folds in the cross-validation; default is 5-fold cross-validation.
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_type">type</code></td>
<td>

<p>How should the observations be normalized within the
Poisson model, i.e. how should the size factors be estimated?
Options are &quot;quantile&quot; or &quot;deseq&quot; (more robust) or &quot;mle&quot; (less
robust).
</p>
<p>In greater detail: &quot;quantile&quot; is quantile normalization approach
of Bullard et al 2010 BMC Bioinformatics, &quot;deseq&quot;  is median of the
ratio of an observation to a pseudoreference obtained by taking the
geometric mean, described in Anders and Huber 2010 Genome
Biology and implemented in Bioconductor package &quot;DESeq&quot;, and &quot;mle&quot; is
the sum of counts for each sample; this is the maximum likelihood
estimate under a simple Poisson model.            
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_prior">prior</code></td>
<td>

<p>Vector of length equal to the number of classes, representing prior
probabilities
for each class. If NULL then uniform priors are used (i.e. each
class is equally likely).
</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_transform">transform</code></td>
<td>

<p>Should data matrices x and xte first be power transformed so  that it
more  closely fits the Poisson model? TRUE or FALSE. Power
transformation is especially  useful if the data are overdispersed
relative to the Poisson model.</p>
</td></tr>
<tr><td><code id="Classify.cv_+3A_alpha">alpha</code></td>
<td>

<p>If transform=TRUE, this determines the power to which the data
matrices x and xte are transformed. If alpha=NULL then the
transformation that makes the Poisson model best fit the data matrix x
is computed. (Note that alpha is computed based on x, not based on xte). Or a
value of alpha, 0&lt;alpha&lt;=1, can be  entered by the user.
</p>
</td></tr>                                                          
<tr><td><code id="Classify.cv_+3A_folds">folds</code></td>
<td>

<p>Instead of specifying the number of folds in cross-validation, one can
explicitly specify the folds. To do this, input a list of length r (to
perform r-fold cross-validation). The rth element of the list should be
a vector containing the indices of the test observations in the rth fold.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>errs</code></td>
<td>
<p>A matrix of dimension (number of folds)-by-(length of
rhos). The (i,j) element is the number of errors occurring in the ith cross-validation
fold for the jth value of the tuning parameter, i.e. rhos[j].</p>
</td></tr>
<tr><td><code>bestrho</code></td>
<td>
<p>The tuning parameter value resulting in the lowest
overall cross-validation error rate.</p>
</td></tr>
<tr><td><code>rhos</code></td>
<td>
<p>The vector of rho values used in cross-validation.</p>
</td></tr>
<tr><td><code>nnonzero</code></td>
<td>
<p>A matrix of dimension (number of folds)-by-(length of
rhos). The (i,j) element is the number of features included in the
classifier occurring in the ith cross-validation fold for the jth
value of the tuning paramer.</p>
</td></tr>
<tr><td><code>folds</code></td>
<td>
<p>Cross-validation folds used.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Power transformation used (if transform=TRUE).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniela Witten
</p>


<h3>References</h3>

<p>D Witten (2011) Classification and clustering of sequencing data using a
Poisson model. To appear in Annals of Applied Statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
dat &lt;- CountDataSet(n=40,p=500,sdsignal=.1,K=3,param=10)
cv.out &lt;- Classify.cv(dat$x,dat$y)
print(cv.out)
out &lt;- Classify(dat$x,dat$y,dat$xte,rho=cv.out$bestrho)
print(out)
cat("Confusion matrix comparing predicted class labels to true class
labels for training observations:", fill=TRUE)
print(table(out$ytehat,dat$yte))
</code></pre>

<hr>
<h2 id='ColorDendrogram'>Color the leaves in a hierarchical clustering dendrogram</h2><span id='topic+ColorDendrogram'></span>

<h3>Description</h3>

<p>Pass in the output of &quot;hclust&quot; and a class label for each
observation. A colored dendrogram will result, with the leaf colors indicating
the classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ColorDendrogram(hc, y, main = "", branchlength = 0.7, labels = NULL,
xlab = NULL, sub = NULL, ylab = "", cex.main = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ColorDendrogram_+3A_hc">hc</code></td>
<td>
<p>The output of running &quot;hclust&quot; on a nxn dissimilarity matrix</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_y">y</code></td>
<td>
<p>A vector of n class labels for the observations that were clustered
using &quot;hclust&quot;. If labels are numeric from 1 to K, then colors will
be determine automatically. Otherwise the labels can take the form
of colors (e.g. c(&quot;red&quot;, &quot;red&quot;, &quot;orange&quot;, &quot;orange&quot;)).</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_main">main</code></td>
<td>
<p>The main title for the dendrogram.</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_branchlength">branchlength</code></td>
<td>
<p>How long to make the colored part of the
branches. Adjustment will be needed for each dissimilarity matrix</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_labels">labels</code></td>
<td>
<p>The labels for the n observations.</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_xlab">xlab</code></td>
<td>
<p>X-axis label.</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_sub">sub</code></td>
<td>
<p>Sub-x-axis label.</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_ylab">ylab</code></td>
<td>
<p>Y-axis label.</p>
</td></tr>
<tr><td><code id="ColorDendrogram_+3A_cex.main">cex.main</code></td>
<td>
<p>The amount by which to enlarge the main title for the figure.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniela Witten</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
dat &lt;- CountDataSet(n=20,p=100,sdsignal=2,K=4,param=10)
dd &lt;- PoissonDistance(dat$x,type="mle")
ColorDendrogram(hclust(dd$dd), y=dat$y, branchlength=10)

</code></pre>

<hr>
<h2 id='CountDataSet'>
Generate a simulated sequencing data set using a negative binomial model.  
</h2><span id='topic+CountDataSet'></span>

<h3>Description</h3>

<p>Generate two nxp data sets: a training set and a test set, as well as
outcome vectors y and yte of length n indicating the class labels of the
training and test observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CountDataSet(n, p, K, param, sdsignal)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CountDataSet_+3A_n">n</code></td>
<td>

<p>Number of observations desired.
</p>
</td></tr>
<tr><td><code id="CountDataSet_+3A_p">p</code></td>
<td>

<p>Number of features desired. Note that 30% of the features will differ
between classes, though some of those differences may be small.
</p>
</td></tr>
<tr><td><code id="CountDataSet_+3A_k">K</code></td>
<td>

<p>Number of classes desired. Note that the function requires that n be at
least equal to 4K &ndash; i.e. there must be at least 4 observations per
class on average.
</p>
</td></tr>
<tr><td><code id="CountDataSet_+3A_param">param</code></td>
<td>

<p>The dispersion parameter for the negative binomial distribution. The
negative binomial distribution is parameterized using &quot;mu&quot; and &quot;size&quot; in
the R function &quot;rnbinom&quot;. That is, Y ~ NB(mu, param) means that E(Y)=mu
and Var(Y) = mu+mu^2/param.
So when param is very large this is essentially a Poisson distribution,
and when param is smaller then there is a lot of overdispersion relative
to the Poisson distribution.
</p>
</td></tr>
<tr><td><code id="CountDataSet_+3A_sdsignal">sdsignal</code></td>
<td>

<p>The extent to which the classes are different. If this equals zero then
there are no class differences and if this is large then the classes are
very different.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is based in part on a function in the DESeq Bioconductor package
(Anders and Huber 2010 Genome Biology) for generating a simulated RNA
sequencing data set.
</p>


<h3>Value</h3>

<table>
<tr><td><code>x</code></td>
<td>
<p>nxq data matrix. May have q&lt;p because features with 0 total
counts are removed.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>class labels for the n observations in x.</p>
</td></tr>
<tr><td><code>xte</code></td>
<td>
<p>nxq data matrix of test observations; the q features are
those with &gt;0 total counts in x. So q&lt;=p.</p>
</td></tr>
<tr><td><code>yte</code></td>
<td>
<p>class labels for the n observation in xte.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniela Witten, based on software written by Anders and Huber in the
DESeq Bioconductor package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
dat &lt;- CountDataSet(n=20,p=100,sdsignal=2,K=4,param=10)
dd &lt;- PoissonDistance(dat$x,type="mle", transform=TRUE)
</code></pre>

<hr>
<h2 id='FindBestTransform'>
Find the power transformation that makes a data set approximately Poisson.
</h2><span id='topic+FindBestTransform'></span>

<h3>Description</h3>

<p>Find a constant alpha, 0&lt;alpha&lt;=1, such that x raised to the power alpha
approximately follows the simple Poisson log linear model that says that
the (i,j) element of x is Poisson with mean si times gj, where si is a
sample-specific term and gj is a feature-specific term.
Alpha is selected via a grid search.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindBestTransform(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindBestTransform_+3A_x">x</code></td>
<td>
<p>A n-by-p matrix of sequencing data, with n observations and p features.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns alpha, the power to which x should be raised.
</p>


<h3>Author(s)</h3>

<p>Daniela Witten
</p>


<h3>References</h3>

<p>D Witten (2011) Classification and clustering of sequencing data using a
Poisson model. To appear in Annals of Applied Statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
dat &lt;- CountDataSet(n=20,p=100,sdsignal=2,K=4,param=10)
alpha &lt;- FindBestTransform(dat$x)
# This is the best transformation!
dd &lt;- PoissonDistance(dat$x^alpha,type="mle", transform=FALSE)
# OR we could get the samething automatically:
dd2 &lt;- PoissonDistance(dat$x,type="mle",transform=TRUE)
# or like this:
dd3 &lt;- PoissonDistance(dat$x,type="mle",transform=TRUE,alpha=alpha)
ColorDendrogram(hclust(dd$dd), y=dat$y, branchlength=10)
</code></pre>

<hr>
<h2 id='PoiClaClu-internal'>Internal PoiClaClu functions</h2><span id='topic+NullModel'></span><span id='topic+NullModelTest'></span><span id='topic+print.poicla'></span><span id='topic+print.poiclacv'></span><span id='topic+print.poidist'></span>

<h3>Description</h3>

<p>Internal PoiClaClu functions</p>


<h3>Usage</h3>

<pre><code class='language-R'>NullModel(x,type=c("mle","deseq","quantile"))
NullModelTest(null.out,x,xte,type=c("mle","deseq","quantile"))
</code></pre>


<h3>Author(s)</h3>

<p>Daniela Witten</p>

<hr>
<h2 id='PoissonDistance'>
Given a n-by-p data matrix, compute the corresponding n-by-n Poisson dissimilarity matrix.
</h2><span id='topic+PoissonDistance'></span>

<h3>Description</h3>

<p>This function computes a Poisson dissimilarity matrix as described in
the paper referenced below, and is intended to be applied to a data
matrix of counts resulting from a sequencing experiment.
The (i,k) element of the Poisson dissimilarity matrix is the
dissimilarity between observations i and k of the data matrix x: that is, the log
likelihood ratio statistic under a simple Poisson model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PoissonDistance(x, beta = 1, type=c("mle","deseq","quantile"),
transform=TRUE, alpha=NULL, perfeature=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PoissonDistance_+3A_x">x</code></td>
<td>

<p>A n-by-p data matrix with observations on the rows, and p features on
the columns. The (i,j) element of x is the number of reads in
observation i that mapped to feature (e.g. gene or exon) j.
</p>
</td></tr>
<tr><td><code id="PoissonDistance_+3A_beta">beta</code></td>
<td>

<p>A smoothing term; essentially the parameter beta in a Gamma(beta,beta)
prior used to estimate the log likelihood ratio statistic for computing
the dissimilarity between a pair of observations. Recommended to leave
it at 1, the default value.
</p>
</td></tr>
<tr><td><code id="PoissonDistance_+3A_type">type</code></td>
<td>
<p>  How should the observations be normalized within the
Poisson model, i.e. how should the size factors be estimated?
Options are &quot;quantile&quot; or &quot;deseq&quot; (more robust) or &quot;mle&quot; (less robust).
</p>
<p>In greater detail: &quot;quantile&quot; is quantile normalization approach of Bullard
et al 2010 BMC Bioinformatics, &quot;deseq&quot;  is median of the ratio of an
observation to a pseudoreference obtained by taking the geometric
mean, described in Anders and Huber 2010 Genome Biology and
implemented in Bioconductor package &quot;DESeq&quot;, and &quot;mle&quot; is the sum of
counts for each sample; this is the maximum likelihood estimate
under a simple Poisson model.
</p>
</td></tr>
<tr><td><code id="PoissonDistance_+3A_transform">transform</code></td>
<td>

<p>Should data matrix x first be power transformed so that it more closely fits
the Poisson model? TRUE or FALSE. Power transformation is especially
useful if the data are overdispersed relative to the Poisson model.
</p>
</td></tr>
<tr><td><code id="PoissonDistance_+3A_alpha">alpha</code></td>
<td>

<p>If transform=TRUE, this determines the power to which the data
matrix x is transformed. If alpha=NULL then the transformation that
makes the Poisson model best fit the data is computed. Or a value of
alpha, 0&lt;alpha&lt;=1, can be entered by the user.
</p>
</td></tr>
<tr><td><code id="PoissonDistance_+3A_perfeature">perfeature</code></td>
<td>

<p>If perfeature=TRUE, then in addition to the nxn dissimilarity
matrix, a nxnxp array will be returned. Its elements will be the
contributions of each of the p features to the nxn dissimilarity
matrix; summing over the 3rd index will simply give back the nxn
dissimilarity matrix.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>More details can be found in the paper referenced below.
</p>


<h3>Value</h3>

<table>
<tr><td><code>dd</code></td>
<td>
<p>A nxn Poisson dissimilarity matrix, containing pairwise
dissimilarities between observations based on the original nxp data
matrix x input by the user.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Power to which data was transformed before computing
dissimilarity matrix, if transform was TRUE. This was either input by
the user, or computed automatically if not specified.</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>Data used to compute dissimilarity matrix, this will be x
raised to the power alpha.</p>
</td></tr>
<tr><td><code>ddd</code></td>
<td>
<p>If perfeature=TRUE, then this is the nxnxp array containing
the contribution of each feature to the nxn dissimilarity matrix.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniela Witten
</p>


<h3>References</h3>

<p>D Witten (2011) Classification and clustering of sequencing data using a
Poisson model. To appear in Annals of Applied Statistics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FindBestTransform">FindBestTransform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
dat &lt;- CountDataSet(n=20,p=100,sdsignal=2,K=4,param=10)
dd &lt;- PoissonDistance(dat$x,type="mle")
print(dd)
ColorDendrogram(hclust(dd$dd), y=dat$y, branchlength=10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
