<!DOCTYPE html><html><head><title>Help for package POINT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {POINT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#point'><p>Protein Structure Guided Local Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Protein Structure Guided Local Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Rachel Marceau West, Shannon T. Holloway</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shannon T. Holloway &lt;shannon.t.holloway@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides an implementation of a rare variant association test that utilizes protein tertiary structure to increase signal and to identify likely causal variants. Performs structure-guided collapsing, which leads to local tests that borrow information from neighboring variants on a protein and that provide association information on a variant-specific level. For details of the implemented method see West, R. M.,  Lu, W., Rotroff, D. M., Kuenemann, M., Chang, S-M., Wagner M. J., Buse, J. B., Motsinger-Reif, A., Fourches, D., and Tzeng, J-Y. (2019) &lt;<a href="https://doi.org/10.1371%2Fjournal.pcbi.1006722">doi:10.1371/journal.pcbi.1006722</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>methods, stats, rARPACK, Matrix, CompQuadForm</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Collate:</td>
<td>'A_Kernel.R' 'A_BurdenKernel.R' 'A_LinearKernel.R'
'A_PolyKernel.R' 'B_PvMethod.R' 'B_PvMethod_Davies.R'
'B_PvMethod_Liu.R' 'C_BinomialTrait.R' 'C_GaussianTrait.R'
'D_NullResult.R' 'calcLocalKernel.R' 'distanceMatrix.R'
'mainCode.R' 'point.R' 'pvResamp.R'</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-01 17:33:27 UTC; 19194</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-01 18:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='point'>Protein Structure Guided Local Test</h2><span id='topic+point'></span>

<h3>Description</h3>

<p>A rare variant association test that utilizes protein tertiary structure to 
increase signal and to identify likely causal variants. Performs 
structure-guided collapsing, which leads to local tests that borrow 
information from neighboring variants on a protein and that provide 
association information on a variant-specific level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>point(
  yy,
  X,
  snp,
  proteinCoord,
  ...,
  trait = "binomial",
  cValues = c(0, 0.1, 0.2, 0.3, 0.4, 0.5),
  weighted = TRUE,
  weight = NULL,
  kernel = "linear",
  d = NULL,
  pvMethod = "davies",
  nperturb = 1000,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="point_+3A_yy">yy</code></td>
<td>
<p>numeric vector; phenotype values.</p>
</td></tr>
<tr><td><code id="point_+3A_x">X</code></td>
<td>
<p>numeric matrix; non-genetic covariates.</p>
</td></tr>
<tr><td><code id="point_+3A_snp">snp</code></td>
<td>
<p>numeric matrix; genotype snp matrix (count of minor alleles).
Matrix cannot contain missing values.</p>
</td></tr>
<tr><td><code id="point_+3A_proteincoord">proteinCoord</code></td>
<td>
<p>numeric matrix; columns correspond to 3 dimensional
coordinates (x,y,z) of each variant in the protein tertiary structure.</p>
</td></tr>
<tr><td><code id="point_+3A_...">...</code></td>
<td>
<p>optional additional arguments for p-value methods 
CompQuadForm::davies and CompQuadForm::liu.</p>
</td></tr>
<tr><td><code id="point_+3A_trait">trait</code></td>
<td>
<p>character; type of phenotype data.
Must be one of {'gaussian','binomial'}
quantitative or case control data, respectively.</p>
</td></tr>
<tr><td><code id="point_+3A_cvalues">cValues</code></td>
<td>
<p>numeric vector; c values from which to choose the
optimal neighborhood size for borrowing significant information.</p>
</td></tr>
<tr><td><code id="point_+3A_weighted">weighted</code></td>
<td>
<p>logical; whether or not to weight the local kernel
test using (non-distance based) weights.</p>
</td></tr>
<tr><td><code id="point_+3A_weight">weight</code></td>
<td>
<p>numeric vector (optional) If NULL and weighted is TRUE
(1.0-MAF)^24. Ignored if weighted is FALSE.</p>
</td></tr>
<tr><td><code id="point_+3A_kernel">kernel</code></td>
<td>
<p>character; type of local kernel to use;
Must be one of {'burden', 'linear', 'polynomial'}.</p>
</td></tr>
<tr><td><code id="point_+3A_d">d</code></td>
<td>
<p>numeric; If kernel = 'poly', d is the order of the
polynomial kernel.</p>
</td></tr>
<tr><td><code id="point_+3A_pvmethod">pvMethod</code></td>
<td>
<p>character; method of calculating the p-value of each
single marker test for fixed c values.
Must be one of {'davies', 'liu'}.</p>
</td></tr>
<tr><td><code id="point_+3A_nperturb">nperturb</code></td>
<td>
<p>numeric, number of perturbations/resamples
(perturbed test statistics) to calculate p-value of minP statistic.</p>
</td></tr>
<tr><td><code id="point_+3A_verbose">verbose</code></td>
<td>
<p>logical; generate progress screen prints.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix the rows of which correspond to individual markers.
Columns correspond to: <br /> 
(1) minP statistic; <br /> 
(2) local kernel test p-value; <br />
(3) optimal scale value from input cValues; <br />
(4) minor allele frequency; and <br />
(5) single variant score test p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # number of subjects
  nsubj &lt;- 1000

  # number of markers
  nm &lt;- 5

  # generate coordinates for proteins
  protein &lt;- cbind( stats::rnorm(n = nm, mean = 17.6, sd =  6.6),
                    stats::rnorm(n = nm, mean =  1.6, sd = 13.6),
                    stats::rnorm(n = nm, mean = 22.9, sd = 10.4) )

  # generate snp matrix
  snp &lt;- matrix(data = rbinom(n = nsubj*nm, size = 1, p = 0.02), 
                nrow = nsubj, ncol = nm)
  colnames(snp) = paste0("m",1:nm)

  # generate binmoial response
  MAF &lt;- colMeans(x = snp)/2
  causal &lt;- numeric(nm)
  causal[c(2,4)] &lt;- 1.0
  betaG &lt;- 0.4*abs(log10(x = MAF))*causal

  #no non-genetic covariates
  X &lt;- NULL
  mu &lt;- -0.05 + snp %*% betaG

  pryy &lt;- exp(mu)/(1+exp(mu))
  yy &lt;- sapply(X = pryy, FUN = stats::rbinom, n = 1, size = 1)

  res &lt;- point(yy = yy, X = X, snp = snp, proteinCoord = protein,
               trait = 'binomial', cValues = c(0.1,0.2),
               weighted = TRUE, weight = NULL, kernel = 'linear',
               pvMethod = 'liu', nperturb = 100,
               verbose = FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
