<!DOCTYPE html><html><head><title>Help for package polySegratioMM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {polySegratioMM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#polySegratioMM-package'>
<p>Bayesian Mixture Models for Marker Dosage in Autopolyploids</p></a></li>
<li><a href='#calculateDIC'><p>Compute DIC for fitted mixture model</p></a></li>
<li><a href='#diagnosticsJagsMix'><p>MCMC diagnostics for polyploid segregation ratio mixture models</p></a></li>
<li><a href='#DistributionPlotBinomial'><p>Distribution Plot</p></a></li>
<li><a href='#dosagesJagsMix'><p>Compute dosages under specified Bayesian mixture model</p></a></li>
<li><a href='#dumpData'><p>Dumps segregation ratio data to file for subsequent JAGS run</p></a></li>
<li><a href='#hexmarkers'><p>Simulated autopolyploid dominant markers from 200 hexaploid individuals</p></a></li>
<li><a href='#hexmarkers.overdisp'><p>Simulated overdispersed autopolyploid dominant markers from 200 hexaploid individuals</p></a></li>
<li><a href='#mcmcHexRun'><p>Results of MCMC estimation for simulated overdispersed markers</p></a></li>
<li><a href='#plot.segratioMCMC'><p> MCMC plots for segregation ratio mixture models</p></a></li>
<li><a href='#plotFitted'><p>Plot observed segregation ratios and fitted and theoretical models</p></a></li>
<li><a href='#print.dosagesMCMC'><p>Doses from Bayesian mixture model</p></a></li>
<li><a href='#print.runJags'><p>Running JAGS</p></a></li>
<li><a href='#readJags'><p>Read MCMC sample(s) from a JAGS run</p></a></li>
<li><a href='#runJags'><p>Run JAGS to create MCMC sample for segregation ratio mixture model</p></a></li>
<li><a href='#runSegratioMM'><p>Run a Bayesian mixture model for marker dosage with minimal effort</p></a></li>
<li><a href='#setControl'><p>Set up controls for a JAGS segregation ratio model run</p></a></li>
<li><a href='#setInits'><p>Set up and dump initial values given the model and prior</p></a></li>
<li><a href='#setModel'><p>Set characteristics of the Bayesian mixture model for dosages</p></a></li>
<li><a href='#setPriors'><p>Set prior distributions for parameters of Bayesian mixture model for dosages</p></a></li>
<li><a href='#summary.segratioMCMC'><p> Summary statistics for an segratioMCMC object</p></a></li>
<li><a href='#writeControlFile'><p>Write JAGS .cmd file for running JAGS</p></a></li>
<li><a href='#writeJagsFile'><p>Writes BUGS file for processing by JAGS</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Mixture Models for Marker Dosage in Autopolyploids</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6-4</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-03-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter Baker [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peter Baker &lt;p.baker1@uq.edu.au&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.12.0), polySegratio</td>
</tr>
<tr>
<td>Imports:</td>
<td>gtools, coda, lattice</td>
</tr>
<tr>
<td>Description:</td>
<td>Fits Bayesian mixture models to estimate marker dosage for dominant markers in autopolyploids using JAGS (1.0 or greater) as outlined in Baker et al "Bayesian estimation of marker dosage in sugarcane and other autopolyploids" (2010, &lt;<a href="https://doi.org/10.1007%2Fs00122-010-1283-z">doi:10.1007/s00122-010-1283-z</a>&gt;). May be used in conjunction with polySegratio for simulation studies and comparison with standard methods.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/petebaker/polysegratiomm">https://github.com/petebaker/polysegratiomm</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/petebaker/polysegratiomm/issues">https://github.com/petebaker/polysegratiomm/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-03-23 05:04:53 UTC; pete</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-03-23 08:49:42 UTC</td>
</tr>
</table>
<hr>
<h2 id='polySegratioMM-package'>
Bayesian Mixture Models for Marker Dosage in Autopolyploids
</h2><span id='topic+polySegratioMM-package'></span><span id='topic+polySegratioMM'></span>

<h3>Description</h3>

<p>These functions provide tools for estimating marker dosage for
dominant markers in regular autopolyploids via Bayesian mixture
model. Wrappers are provided for generating MCMC samples
using the <code>JAGS</code> software. Convergence diagnostics and posterior
distribution densities are provided by the <code>coda</code>
package. 


</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> polySegratioMM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.6-4</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-03-22</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;">  GPL-3
</td>
</tr>

</table>

<p>The simplest way to fit a model is to use <code><a href="#topic+runSegratioMM">runSegratioMM</a></code>.
Given segregation ratios and a ploidy level, a mixture model is
constructed with default priors and initial values and <code>JAGS</code> run
to produce an MCMC sample for statistical inference.
</p>
<p>A standard model may be set up with <code><a href="#topic+setModel">setModel</a></code> where two
parameters are set, namely <code>ploidy.level</code> or the number of
homologous chromosomes set either as a numeric or as a character
string and also <code>n.components</code> or the number of components for
mixture model (less than or equal to maximum number of possible
dosages).
</p>
<p>Vague or strong priors may be constructed automatically using
<code><a href="#topic+setPriors">setPriors</a></code>. Plots of standard conjugate distributions may
be obtained using <code><a href="#topic+DistributionPlotBinomial">DistributionPlotBinomial</a></code>
<code><a href="#topic+DistributionPlotGamma">DistributionPlotGamma</a></code> and
<code><a href="#topic+DistributionPlotNorm">DistributionPlotNorm</a></code>.
</p>
<p>If necessary, other operations like setting up initial values or the
control files for <code>JAGS</code> may be set using <code><a href="#topic+setInits">setInits</a></code>
<code><a href="#topic+setControl">setControl</a></code> <code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code>
<code><a href="#topic+writeControlFile">writeControlFile</a></code> <code><a href="#topic+writeJagsFile">writeJagsFile</a></code>. Once the
<code>BUGS</code> files and <code>JAGS</code> control files are set up then
<code>JAGS</code> may be run using <code><a href="#topic+runJags">runJags</a></code> and results read
using <code><a href="#topic+readJags">readJags</a></code>.
</p>
<p>Convergence diagnostics may be carried out using <code>coda</code> or the
convenience wrapper <code><a href="#topic+diagnosticsJagsMix">diagnosticsJagsMix</a></code>.
</p>
<p>Dose allocation can be carried out using <code><a href="#topic+dosagesJagsMix">dosagesJagsMix</a></code>.
</p>
<p>Plots may be produced and objects printed and summarised using standard
<code><a href="base.html#topic+print">print</a></code> and <code><a href="graphics.html#topic+plot">plot</a></code> methods. Plots of theoretical
binomial distributions with different ploidy levels and sample sizes may
be obtained with <code><a href="#topic+plotFitted">plotFitted</a></code>. In addition,
<code><a href="#topic+plotFitted">plotFitted</a></code> produces a lattice plot of the observed
segregation ratios and fitted mixture model on the logit scale.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li><p>Baker P, Jackson P, and Aitken K. (2010) Bayesian estimation
of marker dosage in sugarcane and other autopolyploids.
<em>TAG Theoretical and Applied Genetics</em> <b>120</b> (8): 1653-1672.
</p>
</li>
<li><p>J B S Haldane   (1930) Theoretical genetics of
autopolyploids. <em>Journal of genetics</em> <b>22</b> 359&ndash;372
</p>
</li>
<li><p>Ripol, M I et al   (1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31&ndash;41 
</p>
</li>
<li><p>JAGS  <a href="http://mcmc-jags.sourceforge.net/">http://mcmc-jags.sourceforge.net/</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set of 100 markers for 50 individuals
## with %70 Single, %20 Double and %10 Triple Dose markers 
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=400,n.individuals=275)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)  # autooctapolid mode with 3 components

## Not run: 
## fit simple model in one hit with default priors, inits etc
## warning: this is too small an MCMC sample so should give inaccurate
## answers but it could still take quite a while
x.run &lt;- runSegratioMM(sr, x, burn.in=2000, sample=5000)
print(x.run)

## plot observed segregation ratios, fitted model and expected distribution
plot(x.run, theoretical=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='calculateDIC'>Compute DIC for fitted mixture model</h2><span id='topic+calculateDIC'></span>

<h3>Description</h3>

<p> Computes and returns the Deviance Information Critereon
(DIC) as suggested by Celeaux et al
(2006) as their  DIC$_4$ for Bayesian mixture models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateDIC(mcmc.mixture, model, priors, seg.ratios, chain=1, print.DIC=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateDIC_+3A_mcmc.mixture">mcmc.mixture</code></td>
<td>
<p>Object of type <code><a href="#topic+segratioMCMC">segratioMCMC</a></code>
produced by <code>coda</code> usually by using <code><a href="#topic+readJags">readJags</a></code></p>
</td></tr>
<tr><td><code id="calculateDIC_+3A_model">model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code id="calculateDIC_+3A_priors">priors</code></td>
<td>
<p>Object of class <code>priorsSegratioMM</code> </p>
</td></tr>
<tr><td><code id="calculateDIC_+3A_seg.ratios">seg.ratios</code></td>
<td>
<p>Object of class <code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
contains the segregation ratios for dominant markers and other
information such as the number of dominant markers per individual</p>
</td></tr>
<tr><td><code id="calculateDIC_+3A_chain">chain</code></td>
<td>
<p>Which chain to use when compute dosages (Default: 1)</p>
</td></tr>
<tr><td><code id="calculateDIC_+3A_print.dic">print.DIC</code></td>
<td>
<p>Whether to print DIC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scalar DIC is returned
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li><p>G Celeaux et. al. (2006) Deviance Information Criteria for
Missing Data Models <em>Bayesian Analysis</em> <b>4</b> 23pp
</p>
</li>
<li><p>D Spiegelhalter et. el. (2002) Bayesian measures of model
complexity and fit <em>JRSS B</em> <b>64</b> 583&ndash;640
</p>
</li></ul>



<h3>See Also</h3>

 <p><code><a href="#topic+dosagesMCMC">dosagesMCMC</a></code>  <code><a href="#topic+readJags">readJags</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## compute segregation ratios
sr &lt;-  segregationRatios(a1$markers)

## set up model, priors, inits etc and write files for JAGS
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
dumpData(sr, x)
inits &lt;- setInits(x,x2)
dumpInits(inits)
writeJagsFile(x, x2, stem="test")

## Not run: 
## run JAGS
small &lt;- setControl(x, burn.in=200, sample=500)
writeControlFile(small)
rj &lt;- runJags(small)  ## just run it
print(rj)

## read mcmc chains and print DIC
xj &lt;- readJags(rj)
print(calculateDIC(xj, x, x2, sr))

## End(Not run)
</code></pre>

<hr>
<h2 id='diagnosticsJagsMix'>MCMC diagnostics for polyploid segregation ratio mixture models</h2><span id='topic+diagnosticsJagsMix'></span>

<h3>Description</h3>

<p>Produce and/or plot various diagnostic measures from <code>coda</code>
package for Bayesian mixture models for assessing marker dosage in
autopolyploids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnosticsJagsMix(mcmc.mixture, diagnostics = TRUE, plots = FALSE,
 index = -c( grep("T\\[",varnames(mcmc.mixture$mcmc.list)),
             grep("b\\[",varnames(mcmc.mixture$mcmc.list)) ),
 trace.plots = FALSE, auto.corrs = FALSE, density.plots = FALSE,
 xy.plots = FALSE, hpd.intervals = FALSE, hdp.prob = 0.95,
 return.results = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnosticsJagsMix_+3A_mcmc.mixture">mcmc.mixture</code></td>
<td>
<p>Object of class <code><a href="#topic+segratioMCMC">segratioMCMC</a></code> or
<code><a href="#topic+runJagsWrapper">runJagsWrapper</a></code> after <code>JAGS</code> run
produced by <code>coda</code> </p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_diagnostics">diagnostics</code></td>
<td>
<p>if TRUE then print several <code>coda</code> dignostic tests</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_plots">plots</code></td>
<td>
<p>if TRUE then produce several <code>coda</code> dignostic plots</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_index">index</code></td>
<td>
<p>index of parameters for disgnostic tests/plots (Default:
mixture model (and random effects) parameters)</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_trace.plots">trace.plots</code></td>
<td>
<p>if TRUE plot mcmc traces (default: FALSE)</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_auto.corrs">auto.corrs</code></td>
<td>
<p>if TRUE produce autocorrelations of mcmc's (default: FALSE)</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_density.plots">density.plots</code></td>
<td>
<p>if TRUE plot parameter densities  (default: FALSE)</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_xy.plots">xy.plots</code></td>
<td>
<p>if TRUE plot traces using 'lattice' (default: FALSE)</p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_hpd.intervals">hpd.intervals</code></td>
<td>
<p>if TRUE print and return highest posterior density
intervals for parameters specified by <code>index</code></p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_hdp.prob">hdp.prob</code></td>
<td>
<p>probability for <code>hpd.intervals</code></p>
</td></tr>
<tr><td><code id="diagnosticsJagsMix_+3A_return.results">return.results</code></td>
<td>
<p>if TRUE return results as list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>return.results</code> is TRUE then a list is returned with
components depending on various settings of arguments
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="coda.html#topic+mcmc">mcmc</a></code> <code><a href="coda.html#topic+autocorr.diag">autocorr.diag</a></code>
<code><a href="coda.html#topic+raftery.diag">raftery.diag</a></code>  <code><a href="coda.html#topic+geweke.diag">geweke.diag</a></code>
<code><a href="coda.html#topic+gelman.diag">gelman.diag</a></code> <code><a href="coda.html#topic+trellisplots">trellisplots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## Not run: 
## fit simple model in one hit

x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)
print(x.run)
diagnosticsJagsMix(x.run)
diagnosticsJagsMix(x.run, plot=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='DistributionPlotBinomial'>Distribution Plot</h2><span id='topic+DistributionPlotBinomial'></span><span id='topic+DistributionPlotGamma'></span><span id='topic+DistributionPlotNorm'></span>

<h3>Description</h3>

<p>Plots probability density function given the parameters. May be useful when
investigating parameter choice for prior distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistributionPlotBinomial(size = 200, prob = 0.5,
xlab = "Number of Successes", ylab = "Probability Mass", signif.digits = 3,
main = paste("Binomial Distribution: n =", size, "p =",
signif(prob, digits = signif.digits)))

DistributionPlotGamma(shape = 1, rate = 1, length = 100, xlab = "x",
ylab = "Density", main = bquote(paste("Gamma Distribution: ", alpha,
"=", .(signif(shape, digits = signif.digits)), ",", beta, "=",
.(signif(rate, digits = signif.digits)))), signif.digits = 3)

DistributionPlotNorm(mean = 0, sd = 1, length = 100, xlab = "x", ylab =
"Density", main = bquote(paste("Normal Distribution: ", mu, "=",
.(signif(mean, digits = signif.digits)), ",", sigma, "=", .(signif(sd,
digits = signif.digits)))), signif.digits = 3)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DistributionPlotBinomial_+3A_size">size</code></td>
<td>
<p>number of trials (Binomial)</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_prob">prob</code></td>
<td>
<p>probability of success (Binomial)</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_shape">shape</code></td>
<td>
<p>shape parameter. Must be strictly positive. (Gamma)</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_rate">rate</code></td>
<td>
<p>an alternative way to specify the scale (Gamma)</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_mean">mean</code></td>
<td>
<p>mean (Normal)</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_sd">sd</code></td>
<td>
<p>standard deviation (Normal)</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_xlab">xlab</code></td>
<td>
<p>x-axis label</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_ylab">ylab</code></td>
<td>
<p>y-axis label</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_signif.digits">signif.digits</code></td>
<td>
<p>number of significant digits for default
<code>main</code> title</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_main">main</code></td>
<td>
<p>title for plot</p>
</td></tr>
<tr><td><code id="DistributionPlotBinomial_+3A_length">length</code></td>
<td>
<p>Number of points to use for obtaining a smooth curve</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on functions in package <code>Rcmdr</code>
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="Rcmdr.html#topic+Rcmdr">Rcmdr</a></code> <code><a href="stats.html#topic+Binomial">Binomial</a></code>
<code><a href="stats.html#topic+Normal">Normal</a></code> <code><a href="stats.html#topic+GammaDist">GammaDist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Binomial distribution
DistributionPlotBinomial()
DistributionPlotBinomial(size=20, prob=0.2)

## Gamma distribution
DistributionPlotGamma()

## Normal distribution
DistributionPlotNorm()
</code></pre>

<hr>
<h2 id='dosagesJagsMix'>Compute dosages under specified Bayesian mixture model</h2><span id='topic+dosagesJagsMix'></span><span id='topic+dosagesMCMC'></span>

<h3>Description</h3>

<p> Computes and returns estimated dosages under specified
model using posterior probabilities derived from mcmc chains by
the proportion of samples in each dosage class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dosagesJagsMix(mcmc.mixture, jags.control, seg.ratio, chain = 1,
max.post.prob = TRUE, thresholds = c(0.5, 0.6, 0.7, 0.8, 0.9, 0.95,
0.99), print = FALSE, print.warning = TRUE, index.sample = 20)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dosagesJagsMix_+3A_mcmc.mixture">mcmc.mixture</code></td>
<td>
<p>Object of type <code><a href="#topic+segratioMCMC">segratioMCMC</a></code>
produced by <code>coda</code> usually by using <code><a href="#topic+readJags">readJags</a></code></p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_jags.control">jags.control</code></td>
<td>
<p>Object of class <code>jagsControl</code> for setting up
<code>JAGS</code> command file </p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_seg.ratio">seg.ratio</code></td>
<td>
<p>Object of class <code><a href="polySegratio.html#topic+segRatio">segRatio</a></code> contains the
segregation ratios for dominant markers and other information
such as the number of dominant markers per individual</p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_chain">chain</code></td>
<td>
<p>Which chain to use when compute dosages (Default: 1)</p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_max.post.prob">max.post.prob</code></td>
<td>
<p>Logical for producing dose allocations based on the
maximum posterior probability (Default: <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_thresholds">thresholds</code></td>
<td>
<p>Numeric vector of thresholds for  allocating
dosages when the posterior probabilty to a particular dosage class
is above the threshold </p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_print">print</code></td>
<td>
<p>Logical indicating whether or not to print intermediate
results (Default: <code>FALSE</code>) </p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_print.warning">print.warning</code></td>
<td>
<p>Logical to print warnings if there is more than
one marker with the maximum posterior probability  </p>
</td></tr>
<tr><td><code id="dosagesJagsMix_+3A_index.sample">index.sample</code></td>
<td>
<p>Numeric vector indicating which markers to print
if <code>print</code> is <code>TRUE</code>. If <code>index.sample</code> is of length
1 then a random sample of size <code>index.sample</code> is selected </p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+dosagesMCMC">dosagesMCMC</a></code> is returned with components:
</p>
<table>
<tr><td><code>p.dosage</code></td>
<td>
<p>Matrix of posterior probabilities of dosages for each
marker dosage</p>
</td></tr>
<tr><td><code>dosage</code></td>
<td>
<p>Matrix of allocated dosages based on posterior probabilities.
The columns correspond to different 'thresholds' and if requested,
the last column is allocated on basis of <code>max.post</code></p>
</td></tr>
<tr><td><code>thresholds</code></td>
<td>
<p>vector of cutoff probabilities for dosage class</p>
</td></tr>
<tr><td><code>chain</code></td>
<td>
<p>Chain used to compute dosages</p>
</td></tr>
<tr><td><code>max.post</code></td>
<td>
<p>maximum dosage posterior probabilties for each marker</p>
</td></tr>
<tr><td><code>index.sample</code></td>
<td>
<p>Numeric vector indicating which markers to print
if <code>print</code> is <code>TRUE</code>. If <code>index.sample</code> is of length
1 then a random sample of size <code>index.sample</code> is selected </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+dosagesMCMC">dosagesMCMC</a></code>  <code><a href="#topic+readJags">readJags</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## compute segregation ratios
sr &lt;-  segregationRatios(a1$markers)

## set up model, priors, inits etc and write files for JAGS
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
dumpData(sr, x)
inits &lt;- setInits(x,x2)
dumpInits(inits)
writeJagsFile(x, x2, stem="test")

## Not run: 
## run JAGS
small &lt;- setControl(x, burn.in=200, sample=500)
writeControlFile(small)
rj &lt;- runJags(small)  ## just run it
print(rj)

## read mcmc chains and produce dosage allocations
xj &lt;- readJags(rj)
dd &lt;- dosagesJagsMix(xj, small, sr)
print(dd)

## End(Not run)
</code></pre>

<hr>
<h2 id='dumpData'>Dumps segregation ratio data to file for subsequent JAGS run</h2><span id='topic+dumpData'></span>

<h3>Description</h3>

<p>Given segregation ratio data provided as an object of class
<code>segRatio</code>, data are dumped in <span class="rlang"><b>R</b></span> format for use by <code>JAGS</code> 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dumpData(seg.ratio, model, stem = "test", fix.one = TRUE,
 data.file = paste(stem, "-data.R", sep = ""))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dumpData_+3A_seg.ratio">seg.ratio</code></td>
<td>
<p>Object of class <code><a href="polySegratio.html#topic+segRatio">segRatio</a></code> contains the
segregation ratios for dominant markers and other information
such as the number of dominant markers per individual</p>
</td></tr>
<tr><td><code id="dumpData_+3A_model">model</code></td>
<td>
<p>Object of class <code><a href="#topic+modelSegratioMM">modelSegratioMM</a></code> containing
mixture model information</p>
</td></tr>
<tr><td><code id="dumpData_+3A_stem">stem</code></td>
<td>
<p>File name stem for data file (default &ldquo;test&rdquo;) </p>
</td></tr>
<tr><td><code id="dumpData_+3A_fix.one">fix.one</code></td>
<td>
<p>Logical to fix the dosage of the observation closest to
the centre of each component on the logit scale. This can greatly
assist with convergence (Default: <code>TRUE</code>) </p>
</td></tr>
<tr><td><code id="dumpData_+3A_data.file">data.file</code></td>
<td>
<p>Data file name which is automatically generated from
<code>stem</code> if not specified</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

  <p><code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>  <code><a href="base.html#topic+dump">dump</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## compute segregation ratios
sr &lt;-  segregationRatios(a1$markers)

## set up model for 3 components of autooctoploid
x &lt;- setModel(3,8)

dumpData(sr, x)

</code></pre>

<hr>
<h2 id='hexmarkers'>Simulated autopolyploid dominant markers from 200 hexaploid individuals</h2><span id='topic+hexmarkers'></span>

<h3>Description</h3>

<p>These data were simulated as 500 markers for 200
&ldquo;auto&ndash;hexaploid individuals&rdquo; exhibiting no
overdispersion. The underlying percentages of single double and
triple dose markers are 70%, 20% and 10%, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hexmarkers</code></pre>


<h3>Format</h3>

<p>An object of S3 class <code>sim.autoMarkers</code> containing 500
simulated dominant markers for 200 auto&ndash;hexaploid individuals.</p>


<h3>References</h3>

<p>Haldane, J B S. 1930. Theoretical genetics of autopolyploids.
<em>Journal of Genetics</em> 22: 359-372.
</p>
<p>Baker P, Jackson P, and Aitken K. 2010. Bayesian estimation of marker
dosage in sugarcane and other autopolyploids. <em>TAG Theoretical and
Applied Genetics</em> 120 (8): 1653-1672.
</p>

<hr>
<h2 id='hexmarkers.overdisp'>Simulated overdispersed autopolyploid dominant markers from 200 hexaploid individuals</h2><span id='topic+hexmarkers.overdisp'></span>

<h3>Description</h3>

<p>These data are simulated as 500 markers for 200 &ldquo;auto&ndash;hexaploid
individuals&rdquo; exhibiting overdispersion with the parameter shape1 =
25. The underlying percentages of single double and triple dose
markers are 70%, 20% and 10%, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hexmarkers.overdisp</code></pre>


<h3>Format</h3>

<p>An object of S3 class <code>sim.autoMarkers</code> containing 500
simulated dominant markers for 200 auto&ndash;hexaploid individuals.</p>


<h3>References</h3>

<p>Haldane, J B S. 1930. Theoretical genetics of autopolyploids.
<em>Journal of Genetics</em> 22: 359-372.
</p>
<p>Baker P, Jackson P, and Aitken K. 2010. Bayesian estimation of marker
dosage in sugarcane and other autopolyploids. <em>TAG Theoretical and
Applied Genetics</em> 120 (8): 1653-1672.
</p>

<hr>
<h2 id='mcmcHexRun'>Results of MCMC estimation for simulated overdispersed markers</h2><span id='topic+mcmcHexRun'></span>

<h3>Description</h3>

<p> MCMC was performed using the wrapper function
<code>runSegratioMM</code> to
run JAGS for a Bayesian mixture model on the segregation ratios
obtained using the simulated data <code>hexmarkers.overdisp</code>. These
data were simulated as 500 markers for 200 &ldquo;auto&ndash;hexaploid
individuals&rdquo; exhibiting overdispersion with shape1=25. The underlying
percentages of single double and triple dose markers are 70%, 20% and
10%, respectively. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmcHexRun</code></pre>


<h3>Format</h3>

<p>An object of S3 class <code>runJagsWrapper</code> with various
components including summaries and diagnostics.</p>


<h3>References</h3>

<p>Baker P, Jackson P, and Aitken K. 2010. Bayesian estimation of marker
dosage in sugarcane and other autopolyploids. <em>TAG Theoretical and
Applied Genetics</em> 120 (8): 1653-1672.
</p>

<hr>
<h2 id='plot.segratioMCMC'> MCMC plots for segregation ratio mixture models </h2><span id='topic+plot.segratioMCMC'></span>

<h3>Description</h3>

<p>Standard MCMC trace and density plots for specified mixure model
parameters and posterior probability distributions for specified markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'segratioMCMC'
plot(x, ..., row.index = c(1:10), var.index = c(1:6),
marker.index = c(1:8))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.segratioMCMC_+3A_x">x</code></td>
<td>
<p>object of class <code>segratioMCMC</code></p>
</td></tr>
<tr><td><code id="plot.segratioMCMC_+3A_...">...</code></td>
<td>
<p>extra options for printing</p>
</td></tr>
<tr><td><code id="plot.segratioMCMC_+3A_row.index">row.index</code></td>
<td>
<p>which rows to print (Default: first 10)</p>
</td></tr>
<tr><td><code id="plot.segratioMCMC_+3A_var.index">var.index</code></td>
<td>
<p>which mixture model variable to summarise (Default:
all)</p>
</td></tr>
<tr><td><code id="plot.segratioMCMC_+3A_marker.index">marker.index</code></td>
<td>
<p>which markers to summarise (Default: 1:8)</p>
</td></tr>
</table>


<h3>Value</h3>

<p> None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+dosagesMCMC">dosagesMCMC</a></code>  <code><a href="#topic+readJags">readJags</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## Not run: 
## fit simple model in one hit and summarise

x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)
plot(x.run$mcmc.mixture)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotFitted'>Plot observed segregation ratios and fitted and theoretical models</h2><span id='topic+plotFitted'></span><span id='topic+plotTheoretical'></span><span id='topic+plot.runJagsWrapper'></span>

<h3>Description</h3>

<p>Plots  histogram of observed segregation ratios on
logit scale along with scaled density of fitted
components corresponding to dosage classes. Plots of expected
theoretical distributions can be plotted with or without segregation
ratio data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'runJagsWrapper'
plot(x, theoretical=FALSE, ...)

plotFitted(seg.ratios, summary.mixture, add.random.effect=TRUE,
  theoretical=FALSE, model=NULL, theory.col="red",
  xaxis=c("logit","raw"), ylim=NULL, NCLASS=NULL, n.seq=100,
  xlab="logit(Segregation Ratio)", ylab="Density", density.plot=FALSE,
  fitted.lwd=2, fitted.col="blue", bar.col="lightgreen", cex=1,
  warnings = FALSE, main=NULL, ...)

plotTheoretical(ploidy.level=8, seg.ratios=NULL, n.components=NULL,
  expected.segratio=NULL, proportions=c(0.65,0.2,0.1,0.03,0.01,0.01, 0, 0),
  n.individuals=200, xaxis=c("raw","logit"),
  type.parents=c("heterogeneous","homozygous"), xlim=c(0,1),
  NCLASS=NULL, xlab="Segregation Ratio", ylab="Density",
  density.plot=FALSE, fitted.lwd=2, fitted.col="blue", cex=1,
  warnings = TRUE, main=NULL, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotFitted_+3A_x">x</code></td>
<td>
<p>object of class <code>runJagsWrapper</code> produced by using
<code>runSegratioMM</code> to set up and fit mixture model</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_seg.ratios">seg.ratios</code></td>
<td>
<p>segregation ratios as class <code>segRatio</code> </p>
</td></tr>
<tr><td><code id="plotFitted_+3A_summary.mixture">summary.mixture</code></td>
<td>
<p>mcmc summary data produce by
<code>summary.segratioMCMC</code></p>
</td></tr>
<tr><td><code id="plotFitted_+3A_add.random.effect">add.random.effect</code></td>
<td>
<p> add random variance component to fitted
distribution plot if model includes a random effect (default: TRUE)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_theoretical">theoretical</code></td>
<td>
<p>whether to plot the expected theoretical
distribution under the fitted model (default: FALSE)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_model">model</code></td>
<td>
<p>object of class <code>modelSegratioMM</code> specifying model
if plotting expected theoretical distribution</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_theory.col">theory.col</code></td>
<td>
<p>colour for expected theoretical distribution
(default: &quot;red&quot;)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_ploidy.level">ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_n.components">n.components</code></td>
<td>
<p>number of components for mixture model</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_expected.segratio">expected.segratio</code></td>
<td>
<p>may be specified or automatically calculated
from ploidy level etc</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_xaxis">xaxis</code></td>
<td>
<p>whether to plot on &quot;logit&quot; or &quot;raw&quot; scale. Defaults to
&quot;logit&quot; if plotting segregation ratios or &quot;raw&quot; for theoretical
distributions</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_proportions">proportions</code></td>
<td>
<p>for no. of markers in each component of theoretical
distribution plot</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_n.individuals">n.individuals</code></td>
<td>
<p>for theoretical distribution plot - taken from
segregation ratios if supplied</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_type.parents">type.parents</code></td>
<td>
<p> &quot;heterogeneous&quot; if parental markers are 0,1 or
&quot;homogeneous&quot; if parental markers are both 1</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_ylim">ylim</code></td>
<td>
<p> c(lower,upper) yaxis limits for histogram of segregation ratios</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_xlim">xlim</code></td>
<td>
<p> c(lower,upper) xaxis limits for segregation ratios</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_nclass">NCLASS</code></td>
<td>
<p>number of classes for histogram (Default: 100)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_n.seq">n.seq</code></td>
<td>
<p>number of points to use for plotting fitted mixture</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_xlab">xlab</code></td>
<td>
<p>x-axis label</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_ylab">ylab</code></td>
<td>
<p>y-axis label</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_density.plot">density.plot</code></td>
<td>
<p>whether to plot a smoothed density as well as
segregation data and fitted and/or theoretical distributions (default:
FALSE)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_main">main</code></td>
<td>
<p>title for plot</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_fitted.lwd">fitted.lwd</code></td>
<td>
<p>width for fitted line</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_fitted.col">fitted.col</code></td>
<td>
<p>colour for fitted line</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_bar.col">bar.col</code></td>
<td>
<p>colour for histogram</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_cex">cex</code></td>
<td>
<p>character expansion for text (see <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_warnings">warnings</code></td>
<td>
<p>print warnings like number of components etc (Default: FALSE)</p>
</td></tr>
<tr><td><code id="plotFitted_+3A_...">...</code></td>
<td>
<p>extra options for <code>plot</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plotFitted</code> plot histogram of observed segregation ratios on
logit scale along with scaled density of fitted
components corresponding to dosage classes using trellis
</p>
<p><code>plotTheoretical</code> plot expected distribution of
autopolyploid dominant markers on probability (0,1)
scale. Segregation ratios may also be plotted
</p>
<p><code>plot.runJagsWrapper</code> plots the fitted values of object of class
<code>runJagsWrapper</code> which has been produced by using
<code>runSegratioMM</code> to set up and fit mixture model
</p>
<p>Note that since trellis graphics are employed, plots may need to be
printed in order to see them
</p>


<h3>Value</h3>

<p> None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="coda.html#topic+summary.mcmc">summary.mcmc</a></code> <code><a href="coda.html#topic+mcmc">mcmc</a></code>
<code><a href="#topic+segratioMCMC">segratioMCMC</a></code> <code><a href="#topic+readJags">readJags</a></code>
<code><a href="#topic+diagnosticsJagsMix">diagnosticsJagsMix</a></code> <code><a href="#topic+runSegratioMM">runSegratioMM</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
plotTheoretical(8, proportion=c(0.7,0.2,0.1),n.individuals=50)
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## fit simple model in one hit and summarise
## Not run: 
x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)
print(x.run)

## plot fitted model using 'plotFitted'
plotFitted(sr, x.run$summary)
a.plot &lt;- plotFitted(sr, x.run$summary, density.plot=TRUE)
print(a.plot)
## or the easier way
plot(x.run, theoretical=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='print.dosagesMCMC'>Doses from Bayesian mixture model</h2><span id='topic+print.dosagesMCMC'></span><span id='topic+print.segratioMCMC'></span>

<h3>Description</h3>

<p>Prints objects of S3 class <code>dosagesMCMC</code> or <code>segratioMCMC</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dosagesMCMC'
print(x, ..., index.sample = 20)

## S3 method for class 'segratioMCMC'
print(x, ..., row.index = c(1:10), var.index = c(1:6), marker.index
 = c(1:8), chain = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.dosagesMCMC_+3A_x">x</code></td>
<td>
<p>object of class <code>dosagesMCMC</code> or <code>segratioMCMC</code></p>
</td></tr>
<tr><td><code id="print.dosagesMCMC_+3A_...">...</code></td>
<td>
<p>extra options for printing</p>
</td></tr>
<tr><td><code id="print.dosagesMCMC_+3A_index.sample">index.sample</code></td>
<td>
<p>which markers to print (Default: 20 markers at random)</p>
</td></tr>
<tr><td><code id="print.dosagesMCMC_+3A_row.index">row.index</code></td>
<td>
<p>which rows to print (Default: first 10)</p>
</td></tr>
<tr><td><code id="print.dosagesMCMC_+3A_var.index">var.index</code></td>
<td>
<p>which mixture model variable to summarise (Default:
all)</p>
</td></tr>
<tr><td><code id="print.dosagesMCMC_+3A_marker.index">marker.index</code></td>
<td>
<p>which markers to summarise (Default: 1:8)</p>
</td></tr>
<tr><td><code id="print.dosagesMCMC_+3A_chain">chain</code></td>
<td>
<p>which chain to print (Default: 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+dosagesMCMC">dosagesMCMC</a></code>  <code><a href="#topic+readJags">readJags</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## fit simple model in one hit

## Not run: 
x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)

print(x.run$doses)

## End(Not run)
</code></pre>

<hr>
<h2 id='print.runJags'>Running JAGS</h2><span id='topic+print.runJags'></span><span id='topic+print.runJagsWrapper'></span>

<h3>Description</h3>

<p>Print details and timing of <code>JAGS</code> run and summaries of results 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'runJags'
print(x, ...)
## S3 method for class 'runJagsWrapper'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.runJags_+3A_x">x</code></td>
<td>
<p>Objects of class <code>runJags</code> or <code>runJagsWrapper</code>  </p>
</td></tr>
<tr><td><code id="print.runJags_+3A_...">...</code></td>
<td>
<p>extra printing options</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>print.runJags</code> can be employed when <code>runJags</code> is called
directly and reports timimgs and dates while
<code>print.runJagsWrapper</code> provides summary statistics when
<code>runSegratioMM</code> is used. 
</p>


<h3>Value</h3>

<p> None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+runJags">runJags</a></code> <code><a href="#topic+runSegratioMM">runSegratioMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## Not run: 
## fit simple model in one hit

x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)
print(x.run)

## End(Not run)
</code></pre>

<hr>
<h2 id='readJags'>Read MCMC sample(s) from a JAGS run</h2><span id='topic+readJags'></span><span id='topic+segratioMCMC'></span>

<h3>Description</h3>

<p> wrapper to <code>read.openbugs</code>  which
returns object of class <code>mcmc.list</code> and so can be used to
specify the start and end iterations for the MCMC sample(s) and also
specify thinning
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readJags(run.jags, quiet = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readJags_+3A_run.jags">run.jags</code></td>
<td>
<p>object of class <code>runJAGS</code> produced by running JAGS</p>
</td></tr>
<tr><td><code id="readJags_+3A_quiet">quiet</code></td>
<td>
<p>logical to return program output (Default: TRUE)</p>
</td></tr>
<tr><td><code id="readJags_+3A_...">...</code></td>
<td>
<p>other options for <code>read.openbugs</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p> Returns object of class <code>segratioMCMC</code> with components
</p>
<table>
<tr><td><code>run.jags</code></td>
<td>
<p>object of class <code>runJAGS</code> produced by running JAGS</p>
</td></tr>
<tr><td><code>mcmc.list</code></td>
<td>
<p>object of class <code>mcmc.list</code> containing the MCMC
sample(s)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

  <p><code><a href="coda.html#topic+mcmc.list">mcmc.list</a></code> <code><a href="#topic+setPriors">setPriors</a></code>
<code><a href="#topic+setInits">setInits</a></code>
<code><a href="polySegratio.html#topic+expected.segRatio">expected.segRatio</a></code>
<code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
<code><a href="#topic+setControl">setControl</a></code>
<code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code> or for an easier way to
run a  segregation ratio mixture model  see
<code><a href="#topic+runSegratioMM">runSegratioMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(polySegratio)

## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
cat(x$bugs.code,x2$bugs.code,sep="\n")


x3 &lt;- setModel(3,8, random.effect = TRUE)
x4 &lt;- setPriors(x3, type="strong")

dumpData(sr, x3)
inits &lt;- setInits(x,x2)
dumpInits(inits)
##x.priors &lt;- setPriors(x, "vague")
writeJagsFile(x, x2, stem="test")

small &lt;- setControl(x, burn.in=20, sample=50)
writeControlFile(small)
## Not run: 
rj &lt;- runJags(small)  ## just run it

xj &lt;- readJags(rj)
print(xj)

## End(Not run)
</code></pre>

<hr>
<h2 id='runJags'>Run JAGS to create MCMC sample for segregation ratio mixture model</h2><span id='topic+runJags'></span>

<h3>Description</h3>

<p>Runs external program <code>JAGS</code> and returns MCMC list for processing
by <code>coda</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runJags(jags.control, jags = "jags", quiet = FALSE,
 cmd.file = paste(jags.control$stem, ".cmd", sep = ""), timing = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runJags_+3A_jags.control">jags.control</code></td>
<td>
<p>Object of class <code>jagsControl</code> containing MCMC
burn in, sample and thinning as well as relavant files for BUGS
commands, inits and data </p>
</td></tr>
<tr><td><code id="runJags_+3A_jags">jags</code></td>
<td>
<p>Name of <code>JAGS</code> program assumed to be in
PATH. However, jags may explicitly set here to include the full path
name</p>
</td></tr>
<tr><td><code id="runJags_+3A_quiet">quiet</code></td>
<td>
<p>Locial to return program output (Default: FALSE)</p>
</td></tr>
<tr><td><code id="runJags_+3A_cmd.file">cmd.file</code></td>
<td>
<p><code>JAGS</code> .cmd commad file (Default: deduced from
<code>jags.control</code>)</p>
</td></tr>
<tr><td><code id="runJags_+3A_timing">timing</code></td>
<td>
<p>Logical to return timing information such as date
started and ended and elapsed user and system time </p>
</td></tr>
</table>


<h3>Value</h3>

<p> Returns object of class <code>runJAGS</code> with components
</p>
<table>
<tr><td><code>jags.control</code></td>
<td>
<p>Object of class <code>jagsControl</code> </p>
</td></tr>
<tr><td><code>exit</code></td>
<td>
<p>integer indicating return error (0 if no errors)</p>
</td></tr>
<tr><td><code>cmd.file</code></td>
<td>
<p><code>JAGS</code> command file</p>
</td></tr>
<tr><td><code>start.time</code></td>
<td>
<p>time JAGS run started</p>
</td></tr>
<tr><td><code>end.time</code></td>
<td>
<p>time JAGS run finished</p>
</td></tr>
<tr><td><code>elapsed.time</code></td>
<td>
<p>elapsed user and system time</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>function call</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setPriors">setPriors</a></code> <code><a href="#topic+setInits">setInits</a></code>
<code><a href="polySegratio.html#topic+expected.segRatio">expected.segRatio</a></code>
<code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
<code><a href="#topic+setControl">setControl</a></code>
<code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code> or for an easier way to
run a  segregation ratio mixture model  see
<code><a href="#topic+runSegratioMM">runSegratioMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
sr &lt;-  segregationRatios(a1$markers)

## set up model with 3 components
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
dumpData(sr, x)
inits &lt;- setInits(x,x2)
dumpInits(inits)
##x.priors &lt;- setPriors(x, "vague")
writeJagsFile(x, x2, stem="test")

## Not run: 
small &lt;- setControl(x, burn.in=20, sample=50)
writeControlFile(small)
rj &lt;- runJags(small)  ## just run it
print(rj)

## End(Not run)
</code></pre>

<hr>
<h2 id='runSegratioMM'>Run a Bayesian mixture model for marker dosage with minimal effort</h2><span id='topic+runSegratioMM'></span><span id='topic+runJagsWrapper'></span>

<h3>Description</h3>

<p>Given segregation ratios and a ploidy level, a mixture model is
constructed with default priors and initial values and <code>JAGS</code> run
to produce an MCMC sample for statistical inference. Returns an object
of S3 class <code>runJagsWrapper</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runSegratioMM(seg.ratios, model, priors = setPriors(model),
 inits = setInits(model, priors), jags.control =
 setControl(model, stem, burn.in = burn.in, sample = sample, thin = thin),
 burn.in = 2000, sample = 5000, thin = 1, stem = "test", fix.one = TRUE,
 print = TRUE, plots = TRUE, print.diagnostics = TRUE,
 plot.diagnostics = TRUE, run.diagnostics.later=FALSE )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runSegratioMM_+3A_seg.ratios">seg.ratios</code></td>
<td>
<p>Object of class <code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
contains the
segregation ratios for dominant markers and other information
such as the number of dominant markers per individual</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_model">model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_priors">priors</code></td>
<td>
<p>object of class <code>priorsSegratioMM</code> indicating
priors that are &ldquo;vague&rdquo;, &ldquo;strong&rdquo; or &ldquo;specified&rdquo;</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_inits">inits</code></td>
<td>
<p>A list of initial values usually produced by <code>setInits</code></p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_jags.control">jags.control</code></td>
<td>
<p>Object of class <code>jagsControl</code> containing MCMC
burn in, sample and thinning as well as relavant files for BUGS
commands, inits and data </p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_burn.in">burn.in</code></td>
<td>
<p>size of MCMC burn in (Default: 2000)</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_sample">sample</code></td>
<td>
<p> size of MCMC sample (default: 5000) </p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_thin">thin</code></td>
<td>
<p>thinning interval between consecutive observations
(default: 1 or no thinning)</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_stem">stem</code></td>
<td>
<p>text to be used as part of <code>JAGS</code> .cmd file name</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_fix.one">fix.one</code></td>
<td>
<p>Logical to fix the dosage of the observation closest to
the centre of each component on the logit scale. This can greatly
assist with convergence (Default: <code>TRUE</code>) </p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_print">print</code></td>
<td>
<p>logical for printing monitoring and summary information
(default: TRUE)</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_plots">plots</code></td>
<td>
<p>logical to plotting MCMC posterior distributions 
(default: TRUE)</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_print.diagnostics">print.diagnostics</code></td>
<td>
<p>logical for printing disagnostic statistics
(default: TRUE)</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_plot.diagnostics">plot.diagnostics</code></td>
<td>
<p>logical for diagnostic plots
(default: TRUE)</p>
</td></tr>
<tr><td><code id="runSegratioMM_+3A_run.diagnostics.later">run.diagnostics.later</code></td>
<td>
<p>should diagnostics be run later which may
help if there are convergence problems (Default: FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p> Returns object of class <code>runJagsWrapper</code> with components
</p>
<table>
<tr><td><code>seg.ratios</code></td>
<td>
<p>Object of class <code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
contains the
segregation ratios for dominant markers</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code>priors</code></td>
<td>
<p>Object of class <code>priorsSegratioMM</code> specifying prior
distributions</p>
</td></tr>
<tr><td><code>inits</code></td>
<td>
<p>A list of initial values usually produced by <code>setInits</code></p>
</td></tr>
<tr><td><code>jags.control</code></td>
<td>
<p>Object of class <code>jagsControl</code> containing MCMC
burn in, sample and thinning as well as relavant files for BUGS
commands, inits and data </p>
</td></tr>
<tr><td><code>stem</code></td>
<td>
<p>text to be used as part of <code>JAGS</code> .cmd file name and
other files</p>
</td></tr>
<tr><td><code>fix.one</code></td>
<td>
<p>Logical to fix the dosage of the observation closest to
the centre of each component on the logit scale. This can greatly
assist with convergence (Default: <code>TRUE</code>) </p>
</td></tr>
<tr><td><code>run.jags</code></td>
<td>
<p>object of class <code>runJAGS</code> produced by running JAGS</p>
</td></tr>
<tr><td><code>mcmc.mixture</code></td>
<td>
<p>Object of type <code><a href="#topic+segratioMCMC">segratioMCMC</a></code>
produced by <code>coda</code> usually by using <code><a href="#topic+readJags">readJags</a></code></p>
</td></tr>
<tr><td><code>diagnostics</code></td>
<td>
<p>list containing various diagnostic summaries and
statistics produced by <code>coda</code></p>
</td></tr>
<tr><td><code>summary</code></td>
<td>
<p>summaries of posterior distributions of model parameters</p>
</td></tr>
<tr><td><code>doses</code></td>
<td>
<p>object of class <code><a href="#topic+dosagesMCMC">dosagesMCMC</a></code> containing
posterior probabilities of dosages for each
marker dosage and allocated dosages</p>
</td></tr>
<tr><td><code>DIC</code></td>
<td>
<p>Deviance Information Critereon</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setPriors">setPriors</a></code> <code><a href="#topic+setInits">setInits</a></code>
<code><a href="polySegratio.html#topic+expected.segRatio">expected.segRatio</a></code>
<code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
<code><a href="#topic+setControl">setControl</a></code>
<code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code> and 
<code><a href="#topic+diagnosticsJagsMix">diagnosticsJagsMix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## Not run: 
## fit simple model in one hit

x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)
print(x.run)

## End(Not run)
</code></pre>

<hr>
<h2 id='setControl'>Set up controls for a JAGS segregation ratio model run </h2><span id='topic+setControl'></span>

<h3>Description</h3>

<p>Sets up directives for running <code>JAGS</code> which are subsequently put
into a .cmd file. MCMC attributes such as the size of burn in,
length of MCMC and thinning may be specified
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setControl(model, stem = "test", burn.in = 2000, sample = 5000, thin = 1,
 bugs.file = paste(stem, ".bug", sep = ""),
 data.file = paste(stem, "-data.R", sep = ""),
 inits.file = paste(stem, "-inits.R", sep = ""),
 monitor.var = model$monitor.var, seed=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setControl_+3A_model">model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code id="setControl_+3A_stem">stem</code></td>
<td>
<p>text to be used as part of <code>JAGS</code> .cmd file name</p>
</td></tr>
<tr><td><code id="setControl_+3A_burn.in">burn.in</code></td>
<td>
<p>size of MCMC burn in (Default: 2000)</p>
</td></tr>
<tr><td><code id="setControl_+3A_sample">sample</code></td>
<td>
<p> size of MCMC sample (default: 5000) </p>
</td></tr>
<tr><td><code id="setControl_+3A_thin">thin</code></td>
<td>
<p>thinning interval between consecutive
observations. Thinning may be a scalar or specified for each
variable set by specifying a vector (default: 1 or no thinning)</p>
</td></tr>
<tr><td><code id="setControl_+3A_bugs.file">bugs.file</code></td>
<td>
<p>name of .bug file</p>
</td></tr>
<tr><td><code id="setControl_+3A_data.file">data.file</code></td>
<td>
<p>name of <span class="rlang"><b>R</b></span> data file </p>
</td></tr>
<tr><td><code id="setControl_+3A_inits.file">inits.file</code></td>
<td>
<p>name of <span class="rlang"><b>R</b></span> inits file </p>
</td></tr>
<tr><td><code id="setControl_+3A_monitor.var">monitor.var</code></td>
<td>
<p>which variables to be monitored (Default: as per
model)</p>
</td></tr>
<tr><td><code id="setControl_+3A_seed">seed</code></td>
<td>
<p>seed for JAGS run for Windows only (for unix set seed in
<code>setInits</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>jagsControl</code> which is a list
with components
</p>
<table>
<tr><td><code>jags.code</code></td>
<td>
<p>Text containing control statements for <code>JAGS</code>
.cmd file</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code>stem</code></td>
<td>
<p>text to be used as part of <code>JAGS</code> .cmd file name</p>
</td></tr>
<tr><td><code>burn.in</code></td>
<td>
<p>size of MCMC burn in (Default: 2000)</p>
</td></tr>
<tr><td><code>sample</code></td>
<td>
<p> size of MCMC sample (default: 5000) </p>
</td></tr>
<tr><td><code>thin</code></td>
<td>
<p>thinning interval between consecutive observations</p>
</td></tr>
<tr><td><code>bugs.file</code></td>
<td>
<p>name of .bug file</p>
</td></tr>
<tr><td><code>data.file</code></td>
<td>
<p>name of <span class="rlang"><b>R</b></span> data file </p>
</td></tr>
<tr><td><code>inits.file</code></td>
<td>
<p>name of <span class="rlang"><b>R</b></span> inits file </p>
</td></tr>
<tr><td><code>monitor.var</code></td>
<td>
<p>which variables to be monitored</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>function call</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setModel">setModel</a></code> <code><a href="#topic+setInits">setInits</a></code>
<code><a href="polySegratio.html#topic+expected.segRatio">expected.segRatio</a></code>
<code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
<code><a href="#topic+setControl">setControl</a></code>
<code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code> or for an easier way to
run a  segregation ratio mixture model  see
<code><a href="#topic+runSegratioMM">runSegratioMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## set up model with 3 components
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)

jc &lt;- setControl(x)
print(jc)
</code></pre>

<hr>
<h2 id='setInits'>Set up and dump initial values given the model and prior</h2><span id='topic+setInits'></span><span id='topic+dumpInits'></span>

<h3>Description</h3>

<p>Given a model of class <code>modelSegratioMM</code> and priors of class
<code>priorsSegratioMM</code>, initial values are computed using approximate
expected values by <code>setInits</code> and then written to file by
<code>dumpInits</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setInits(model, priors, seed = 1)

dumpInits(inits, stem = "test", inits.file = paste(stem, "-inits.R",
  sep = ""))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setInits_+3A_model">model</code></td>
<td>
<p> Object of class <code>modelSegratioMM</code> providing model
attributes like the number of components and ploidy level</p>
</td></tr>
<tr><td><code id="setInits_+3A_priors">priors</code></td>
<td>
<p>Object of class <code>priorsSegratioMM</code> </p>
</td></tr>
<tr><td><code id="setInits_+3A_seed">seed</code></td>
<td>
<p> Seed to be used for <code>JAGS</code> runs. If a number of
chains are to be run a vector of starting values may be
specified. However, see note below. </p>
</td></tr>
<tr><td><code id="setInits_+3A_inits">inits</code></td>
<td>
<p>A list of initial values usually produced by <code>setInits</code></p>
</td></tr>
<tr><td><code id="setInits_+3A_stem">stem</code></td>
<td>
<p>File name stem for inits file (default &ldquo;test&rdquo;)</p>
</td></tr>
<tr><td><code id="setInits_+3A_inits.file">inits.file</code></td>
<td>
<p>Inits file name which is automatically generated from
<code>stem</code> if not specified</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the following initial values:
</p>
<table>
<tr><td><code>mu</code></td>
<td>
<p>Mean of dosage classes on logit scale: usually c(0,NA,NA,...,NA)</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>Initial value for proportion in each dosage class</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>Precision of means which depends on whether priors are
strong or weak</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>Differences in means (for parameterisation employed for
better convergence)</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>Sets seed for each MCMC chain (Default:1) </p>
</td></tr>
<tr><td><code>taub</code></td>
<td>
<p>If the model contains a random effect then sets 
initial value of precision of random effect <em>b</em> which is normally
distributed with mean 0 and precision <code>taub</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p><em>Warning</em>: If a number of chains are to be produced then
several seeds may be specified. Currently, this is largely untested
and so it is highly unlikely that this will actually
work for all functions in this package.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setModel">setModel</a></code> <code><a href="#topic+setPriors">setPriors</a></code>
<code><a href="#topic+setControl">setControl</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## set up model, priors, inits etc and write files for JAGS
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
inits &lt;- setInits(x,x2)
dumpInits(inits)

</code></pre>

<hr>
<h2 id='setModel'>Set characteristics of the Bayesian mixture model for dosages</h2><span id='topic+setModel'></span><span id='topic+modelSegratioMM'></span>

<h3>Description</h3>

<p>Used to automatically set up Bayesian finite mixture models
for dosage allocation of dominant markers in autopolyploids given the
number of components and ploidy level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setModel(n.components, ploidy.level, random.effect = FALSE, seg.ratios =NULL,
 ploidy.name = NULL, equal.variances=TRUE, 
 type.parents = c("heterogeneous", "homozygous"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setModel_+3A_n.components">n.components</code></td>
<td>
<p>number of components for mixture model (less than
or equal to maximum number of possible dosages)</p>
</td></tr>
<tr><td><code id="setModel_+3A_ploidy.level">ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric or as a character string</p>
</td></tr>
<tr><td><code id="setModel_+3A_random.effect">random.effect</code></td>
<td>
<p>Logical indicating whether model contains random
effect (Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="setModel_+3A_seg.ratios">seg.ratios</code></td>
<td>
<p>segregation proportions for each marker provided as
S3 class <code>segRatio</code></p>
</td></tr>
<tr><td><code id="setModel_+3A_ploidy.name">ploidy.name</code></td>
<td>
<p>Can overide ploidy name here or allow it to be
determined from  <code>ploidy.level</code> </p>
</td></tr>
<tr><td><code id="setModel_+3A_equal.variances">equal.variances</code></td>
<td>
<p>Logical indicating whether model contains
separate or common variances for each component (Default: <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="setModel_+3A_type.parents">type.parents</code></td>
<td>
<p> &quot;heterogeneous&quot; if parental markers are 0,1 or
&quot;homogeneous&quot; if parental markers are both 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns object of class <code>modelSegratioMM</code> with components
</p>
<table>
<tr><td><code>bugs.code</code></td>
<td>
<p>text to be used by <code>JAGS</code> in the .bug file but
without statements pertaining to priors</p>
</td></tr>
<tr><td><code>n.components</code></td>
<td>
<p>number of components for mixture model</p>
</td></tr>
<tr><td><code>monitor.var</code></td>
<td>
<p>names of variables to be monitored in <code>JAGS</code> run</p>
</td></tr>
<tr><td><code>ploidy.level</code></td>
<td>
<p>ploidy level</p>
</td></tr>
<tr><td><code>random.effect</code></td>
<td>
<p>Logical indicating whether model contains random
effect (Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code>equal.variances</code></td>
<td>
<p>Logical indicating equal or separate variances
for each component</p>
</td></tr>
<tr><td><code>E.segRatio</code></td>
<td>
<p>Expected segregation ratios</p>
</td></tr>
<tr><td><code>type.parents</code></td>
<td>
<p> &quot;heterogeneous&quot; if parental markers are 0,1 or
&quot;homogeneous&quot; if parental markers are both 1</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>function call</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setPriors">setPriors</a></code> <code><a href="#topic+setInits">setInits</a></code>
<code><a href="polySegratio.html#topic+expected.segRatio">expected.segRatio</a></code>
<code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
<code><a href="#topic+setControl">setControl</a></code>
<code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code> or for an easier way to
run a  segregation ratio mixture model  see
<code><a href="#topic+runSegratioMM">runSegratioMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## set up model with 3 components
x &lt;- setModel(3,8)
print(x)
</code></pre>

<hr>
<h2 id='setPriors'>Set prior distributions for parameters of Bayesian mixture model for dosages</h2><span id='topic+setPriors'></span>

<h3>Description</h3>

<p>May be used to automatically set up vague or strong priors or
explicitly set them for Bayesian finite mixture model specified as an
object of class <code>modelSegratioMM</code> using
<code>setModel</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setPriors(model, type.prior = c("strong",
              "vague", "strong.tau","strong.s", "specified"),
mean.vague = 0.1, prec.vague = 0.1, A.vague = 0.1, B.vague = 0.1,
prec.strong=400, n.individuals=200, reffect.A = 44, reffect.B = 0.8,
M.sd = 0.025, STRONG.PREC=c(0.025, 0.975), UPPER = 0.995,  PREC.INT=0.2,
params = NULL, segRatio = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setPriors_+3A_model">model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code id="setPriors_+3A_type.prior">type.prior</code></td>
<td>
<p>The type of prior required being one of
&ldquo;strong&rdquo;, &ldquo;vague&rdquo;, &ldquo;strong.tau&rdquo;
&ldquo;strong.s&rdquo; or &ldquo;specified&rdquo;. The first
four prior types will automatically set prior distributions whereas
for the last, namely &ldquo;specified&rdquo;, the prior distribution
parameters must be set explicitly. Note that strong priors get
progressively stronger from &ldquo;strong&rdquo; to &ldquo;strong.s&rdquo; </p>
</td></tr>
<tr><td><code id="setPriors_+3A_mean.vague">mean.vague</code></td>
<td>
<p>The mean of Normal priors for a &ldquo;vague&rdquo; prior</p>
</td></tr>
<tr><td><code id="setPriors_+3A_prec.vague">prec.vague</code></td>
<td>
<p>The precision of Normal priors for a &ldquo;vague&rdquo; prior</p>
</td></tr>
<tr><td><code id="setPriors_+3A_a.vague">A.vague</code></td>
<td>
<p>The shape parameter of the Gamma prior for the
precision parameters for a &ldquo;vague&rdquo; prior </p>
</td></tr>
<tr><td><code id="setPriors_+3A_b.vague">B.vague</code></td>
<td>
<p> The rate (scale) parameter of the Gamma prior for the
precision parameters for a &ldquo;vague&rdquo; prior </p>
</td></tr>
<tr><td><code id="setPriors_+3A_prec.strong">prec.strong</code></td>
<td>
<p>Precision for Normal mean parameters when
<code>type.prior</code> is &ldquo;strong&rdquo;. Note that on logit scale
default is equivalent to having a 95%CI as +/- 0.1</p>
</td></tr>
<tr><td><code id="setPriors_+3A_n.individuals">n.individuals</code></td>
<td>
<p>Used for Binomial calculations to set prior
precision parameters when <code>type.prior</code> is &ldquo;strong&rdquo;.</p>
</td></tr>
<tr><td><code id="setPriors_+3A_reffect.a">reffect.A</code></td>
<td>
<p> The shape parameter of the Gamma prior for the
precision parameter of the random.effect for a &ldquo;vague&rdquo; prior </p>
</td></tr>
<tr><td><code id="setPriors_+3A_reffect.b">reffect.B</code></td>
<td>
<p>The rate (scale) parameter of the Gamma prior for the
precision parameter of the random.effect for a &ldquo;vague&rdquo; prior</p>
</td></tr>
<tr><td><code id="setPriors_+3A_m.sd">M.sd</code></td>
<td>
<p>Approximate standard deviation for the mean segregation
ratios on raw probability scale - this is set to 0.025 which would
give an approximate 95% interval of 0.1 for the segregation ratio</p>
</td></tr>
<tr><td><code id="setPriors_+3A_upper">UPPER</code></td>
<td>
<p>Cutoff for guessing parameters on logit scale noting that
logit(1) is undefined</p>
</td></tr>
<tr><td><code id="setPriors_+3A_strong.prec">STRONG.PREC</code></td>
<td>
<p>Interval on raw probabilty scale used to set strong
priors on the the precision distribution parameters of the
segregation ratios by using a 95% interval on the theoretical
distribution and equating this on the logit scale (Default: c(0.025,
0.975))</p>
</td></tr>
<tr><td><code id="setPriors_+3A_prec.int">PREC.INT</code></td>
<td>
<p>Multiplier or setting prior for precision on logit
scale corresponding to approx confidence region being precision*(1 -
PREC.INT, 1 + PREC.INT) Default:0.2</p>
</td></tr> 
<tr><td><code id="setPriors_+3A_params">params</code></td>
<td>
<p>if <code>type.prior</code> is &ldquo;specified&rdquo; then a list
of priors parameters must be set containing components M for means,
A and B for gamma prior parameters and if the model contains a
random.effect then reffect.A, and reffect.B for the gamma prior for
the precision of random effect <code>taub</code>. Note that the lengths of
M, prec, A and B should be equal to the number of components</p>
</td></tr>
<tr><td><code id="setPriors_+3A_segratio">segRatio</code></td>
<td>
<p>If specified, this value overides the automatically
generated value which is set as the expected segregation ratio given
the ploidy level</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>priorsSegratioMM</code> which is a list
with components
</p>
<table>
<tr><td><code>type</code></td>
<td>
<p>Type of prior: one of &ldquo;vague&rdquo;, &ldquo;strong&rdquo;
or &ldquo;specified&rdquo;</p>
</td></tr>
<tr><td><code>bugs.code</code></td>
<td>
<p>Text containing prior statements for <code>BUGS</code> file</p>
</td></tr>
<tr><td><code>random.effect</code></td>
<td>
<p>Logical indicating whether model contains random
effect (Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code>equal.variances</code></td>
<td>
<p>Logical indicating equal or separate variances
for each component</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>List containing Normal means on logit scale
<code>logit.means</code>, precision on logit scale <code>logit.prec</code>, and
Gamma parameters <code>A</code> and <code>B</code> and finally <code>reffect.A</code> and
<code>reffect.B</code> if the model contains a random effect</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>function call</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setModel">setModel</a></code> <code><a href="#topic+setInits">setInits</a></code>
<code><a href="polySegratio.html#topic+expected.segRatio">expected.segRatio</a></code>
<code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>
<code><a href="#topic+setControl">setControl</a></code>
<code><a href="#topic+dumpData">dumpData</a></code> <code><a href="#topic+dumpInits">dumpInits</a></code> or for an easier way to
run a  segregation ratio mixture model  see
<code><a href="#topic+runSegratioMM">runSegratioMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## set up model with 3 components
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
print(x2)

x2b &lt;- setPriors(x, "strong")
print(x2b)

</code></pre>

<hr>
<h2 id='summary.segratioMCMC'> Summary statistics for an segratioMCMC object</h2><span id='topic+summary.segratioMCMC'></span><span id='topic+summarySegratioMCMC'></span>

<h3>Description</h3>

<p>Wrapper for <code>summary.mcmc</code> processing only mixture model parameters
although markers may also easily be summarised. The  mean, standard
deviation, naive standard error of the mean (ignoring autocorrelation
of the chain) and time-series standard error based on an estimate of
the spectral density at 0. For details see <code>summary.mcmc</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'segratioMCMC'
summary(object, ..., row.index = c(1:10),
 var.index = NULL,
 marker.index = c(1:8))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.segratioMCMC_+3A_object">object</code></td>
<td>
<p>object of class <code>segratioMCMC</code></p>
</td></tr>
<tr><td><code id="summary.segratioMCMC_+3A_...">...</code></td>
<td>
<p>extra options for <code>summary.mcmc</code></p>
</td></tr>
<tr><td><code id="summary.segratioMCMC_+3A_row.index">row.index</code></td>
<td>
<p>which rows to print (Default: first 10)</p>
</td></tr>
<tr><td><code id="summary.segratioMCMC_+3A_var.index">var.index</code></td>
<td>
<p>which mixture model variable to summarise (Default:
all)</p>
</td></tr>
<tr><td><code id="summary.segratioMCMC_+3A_marker.index">marker.index</code></td>
<td>
<p>which markers to summarise (Default: 1:8)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>summarySegratioMCMC</code> is returned which
contains summary statistics for parameters and some markers. For
details see <code><a href="coda.html#topic+summary.mcmc">summary.mcmc</a></code>
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="coda.html#topic+summary.mcmc">summary.mcmc</a></code> <code><a href="coda.html#topic+mcmc">mcmc</a></code>
<code><a href="#topic+segratioMCMC">segratioMCMC</a></code> <code><a href="#topic+readJags">readJags</a></code>
<code><a href="#topic+diagnosticsJagsMix">diagnosticsJagsMix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)

## Not run: 
## fit simple model in one hit and summarise

x.run &lt;- runSegratioMM(sr, x, burn.in=200, sample=500)
print(summary(x.run$mcmc.mixture))
print(summary(x.run$mcmc.mixture, var.index=c(1:3), marker.index=c(1:4)))

## End(Not run)
</code></pre>

<hr>
<h2 id='writeControlFile'>Write JAGS .cmd file for running JAGS</h2><span id='topic+writeControlFile'></span>

<h3>Description</h3>

<p>Write JAGS .cmd file to disk
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeControlFile(jags.control,
     file = paste(jags.control$stem, ".cmd", sep = ""))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeControlFile_+3A_jags.control">jags.control</code></td>
<td>
<p>Object of class <code>jagsControl</code> containing MCMC
burn in, sample and thinning as well as relavant files for BUGS
commands, inits and data </p>
</td></tr>
<tr><td><code id="writeControlFile_+3A_file">file</code></td>
<td>
<p>JAGS .cmd file name </p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+setControl">setControl</a></code>  <code><a href="#topic+runJags">runJags</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)
sr &lt;-  segregationRatios(a1$markers)

## set up model with 3 components
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)
dumpData(sr, x)
inits &lt;- setInits(x,x2)
dumpInits(inits)
##x.priors &lt;- setPriors(x, "vague")
writeJagsFile(x, x2, stem="test")

small &lt;- setControl(x, burn.in=20, sample=50)
writeControlFile(small)
</code></pre>

<hr>
<h2 id='writeJagsFile'>Writes BUGS file for processing by JAGS</h2><span id='topic+writeJagsFile'></span>

<h3>Description</h3>

<p>Given the model and priors a file is written to disk for subsequent
<code>JAGS</code> run. BUGS code contained in the model and priors objects
is combined and alterered if necessary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeJagsFile(model, priors, stem = "test")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeJagsFile_+3A_model">model</code></td>
<td>
<p> object of class <code>modelSegratioMM</code> specifying model
parameters, ploidy etc</p>
</td></tr>
<tr><td><code id="writeJagsFile_+3A_priors">priors</code></td>
<td>
<p>Object of class <code>priorsSegratioMM</code> specifying priors</p>
</td></tr>
<tr><td><code id="writeJagsFile_+3A_stem">stem</code></td>
<td>
<p>File name stem for BUGS file (default &ldquo;test&rdquo;) </p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

  <p><code><a href="polySegratio.html#topic+segRatio">segRatio</a></code>  <code><a href="base.html#topic+dump">dump</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate small autooctaploid data set
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=100,n.individuals=50)

## compute segregation ratios
sr &lt;-  segregationRatios(a1$markers)

## set up model for 3 components of autooctoploid
x &lt;- setModel(3,8)
x2 &lt;- setPriors(x)

dumpData(sr, x)
inits &lt;- setInits(x,x2)
dumpInits(inits)
##x.priors &lt;- setPriors(x, "vague")
writeJagsFile(x, x2, stem="test")


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
