<!DOCTYPE html><html><head><title>Help for package polySegratio</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {polySegratio}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addMisclass'><p>Misclassifies marker data in objects of class autoMarker or autoCross</p></a></li>
<li><a href='#addMissing'><p> Adds missing data to objects of class autoMarker or autoCross</p></a></li>
<li><a href='#autoFill'><p> Automatically fill out blanks of a vector with the preceding label</p></a></li>
<li><a href='#divide.autoMarkers'><p> Divide markers by parental type</p></a></li>
<li><a href='#divideAutoMarkers'><p> S3 class divideAutoMarkers</p></a></li>
<li><a href='#expected.segRatio'><p> Compute theoretical segregation proportions for regular autopolyploids</p></a></li>
<li><a href='#makeLabel'><p> Generate labels from two columns where blanks in first column</p>
are replaced by preceding non&ndash;blank label</a></li>
<li><a href='#plot.segRatio'><p> Plot segregation ratios for either observed or simulated marker data</p></a></li>
<li><a href='#polySegratio-package'>
<p>Segregation Ratios for Autoployploids</p></a></li>
<li><a href='#print.segRatio'><p> Print segregation ratios</p></a></li>
<li><a href='#print.simAutoMarkers'><p> Print objects of class  simAutoMarkers</p></a></li>
<li><a href='#print.testSegRatio'><p> Print objects of class testSegRatio</p></a></li>
<li><a href='#segRatio'><p>S3 class segRatio</p></a></li>
<li><a href='#segregationRatios'><p> Compute observed segregation proportions for dominant markers in</p>
autopolyploids</a></li>
<li><a href='#sim.autoCross'><p>Simulate dominant markers for an autopolyploid cross for all</p>
parental types</a></li>
<li><a href='#sim.autoMarkers'><p>Simulates dominant markers from an autopolyploid cross</p></a></li>
<li><a href='#simAutoCross'><p>S3 class simAutoCross</p></a></li>
<li><a href='#simAutoMarkers'><p>S3 class simAutoMarkers</p></a></li>
<li><a href='#test.segRatio'><p> Classic tests for assessing marker dosage in autopolyploids</p></a></li>
<li><a href='#testSegRatio'><p>S3 class testSegRatio</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulate and Test Marker Dosage for Dominant Markers in
Autopolyploids</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2-5</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-03-22</td>
</tr>
<tr>
<td>Imports:</td>
<td>gdata</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter Baker</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peter Baker &lt;p.baker1@uq.edu.au&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform classic chi-squared tests and Ripol et al(1999)
             binomial confidence interval approach for autopolyploid
             dominant markers. Also, dominant markers may be generated
             for families of offspring where either one or both of the
             parents possess the marker. Missing values and
             misclassified markers may be generated at random.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-03-22 15:38:45 UTC; pete</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-03-22 20:16:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='addMisclass'>Misclassifies marker data in objects of class autoMarker or autoCross</h2><span id='topic+addMisclass'></span>

<h3>Description</h3>

<p>Marker data are misclassified at a specified rate for  objects
of class <code>simAutoMarkers</code> or <code>simAutoCross</code>. The rate may be
specified either as a proportion of missing at random or a proportion
of columns and rows with specified proportions of missings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMisclass(x, misclass = 0, bands.missed=0, parents = FALSE,
parent.cols = c(1, 2), seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMisclass_+3A_x">x</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> or <code>simAutoCross</code>,
or a matrix with dominant markers scored as 0 or 1</p>
</td></tr>
<tr><td><code id="addMisclass_+3A_misclass">misclass</code></td>
<td>
<p> proportion misclassified specified as for
na.proportion (Default: 0)</p>
</td></tr>
<tr><td><code id="addMisclass_+3A_bands.missed">bands.missed</code></td>
<td>
<p>proportion of bands that are not scored when they
are actually present. Note this is applied to correctly specified
markers after markers are misclassified (Default: 0)</p>
</td></tr>
<tr><td><code id="addMisclass_+3A_parents">parents</code></td>
<td>
<p>if TRUE then misclassify parental alleles, otherwise
misclassify offspring marker alleles</p>
</td></tr>
<tr><td><code id="addMisclass_+3A_parent.cols">parent.cols</code></td>
<td>
<p>for object of simAutoClass the columns containg
parental markers</p>
</td></tr>
<tr><td><code id="addMisclass_+3A_seed">seed</code></td>
<td>
<p> random number generator (RNG) state for random number
which will be set at start to reproduce results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns object of class <code>simAutoMarkers</code> or <code>simAutoCross</code>,
or a matrix with
dominant markers scored as 0 or 1 with extra components
</p>
<table>
<tr><td><code>misclass.info</code></td>
<td>
<p>list with components
</p>

<ul>
<li><p><code>proportion</code>numeric proportion misclassified
</p>
</li>
<li><p><code>index</code>indicates which markers were set as misclassified
</p>
</li>
<li><p><code>bands.proportion</code>numeric proportion marker bands missed
</p>
</li>
<li><p><code>bands.index</code>indicates which markers bands were missed
</p>
</li>
<li><p><code>call</code>matches arguments when function called
</p>
</li>
<li><p><code>time.generated</code>time/date when misclassifieds added
</p>
</li>
<li><p><code>seed</code> seed for random number generation
</p>
</li></ul>

</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+addMissing">addMissing</a></code> add missing  markers at random,
<code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code> simulate autopolyploid markers,
<code><a href="#topic+sim.autoCross">sim.autoCross</a></code> simulate autopolyploid markers for a cross
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## simulate autopolyploid markers
p1 &lt;- sim.autoCross(4, dose.proportion=c(0.7,0.3), n.markers=20, n.indiv=10)
p2 &lt;- sim.autoCross(4, dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),p11=c(
0.6,0.2,0.2)))

## add misclassified for a whopping 20% of markers
print(addMisclass(p1, 0.2, parents=TRUE), row=1:20)
addMisclass(p2, 0.1)
                    
</code></pre>

<hr>
<h2 id='addMissing'> Adds missing data to objects of class autoMarker or autoCross</h2><span id='topic+addMissing'></span>

<h3>Description</h3>

<p>Adds missing data to objects of class <code>simAutoMarkers</code> or
<code>simAutoCross</code> as specified either as a proportion of missing at
random or a proportion  of columns and rows with specified proportions
of missings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMissing(x, na.proportion = 0, parent.cols = c(1, 2), seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMissing_+3A_x">x</code></td>
<td>
<p> object of class <code>simAutoMarkers</code> or <code>simAutoCross</code>,
or a <code>matrix</code> with dominant markers scored as 0 or 1</p>
</td></tr>
<tr><td><code id="addMissing_+3A_na.proportion">na.proportion</code></td>
<td>
<p>proportion missing at random or a list with
two components indiv and marker each containing
c(prop. markers missing, prop. missing) (Default: 0)</p>
</td></tr>
<tr><td><code id="addMissing_+3A_parent.cols">parent.cols</code></td>
<td>
<p>columns containing parental markers (etc) not
altered only used if object of class <code>simAutoCross</code></p>
</td></tr>
<tr><td><code id="addMissing_+3A_seed">seed</code></td>
<td>
<p>random number generator (RNG) state for random number
which will be set at start to reproduce results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns object of class <code>simAutoMarkers</code> or <code>simAutoCross</code>, or a
<code>matrix</code> with dominant markers scored as 0 or 1 with extra
component na.proportion
which has the following elements
</p>
<table>
<tr><td><code>na.proportion</code></td>
<td>
<p>proportion missing at random or a list
with two components indiv and marker each containing
c(prop. markers missing, prop. missing)</p>
</td></tr>
<tr><td><code>time.generated</code></td>
<td>
<p>time/date when data set generated + when missing added</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>random number generator seed which could be used to
reproduce results (I hope)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>matches arguments when function called</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+addMisclass">addMisclass</a></code> misclassifies markers at random,
<code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code> simulate autopolyploid markers,
<code><a href="#topic+sim.autoCross">sim.autoCross</a></code> simulate autopolyploid markers for a cross
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate autopolyploid markers
p1 &lt;- sim.autoCross(4, dose.proportion=c(0.7,0.3), n.markers=20, n.indiv=10)
p2 &lt;- sim.autoCross(4, dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),p11=c(
0.6,0.2,0.2)))

## add missings
addMissing(p2, 0.1)
</code></pre>

<hr>
<h2 id='autoFill'> Automatically fill out blanks of a vector with the preceding label </h2><span id='topic+autoFill'></span>

<h3>Description</h3>

<p><code>autoFill</code> is commonly used to generate labels from columns of a
spreadsheet when many cells are left blank in order to save a lot of
typing. For instance, c(&quot;a&quot;,&quot;&quot;,&quot;&quot;,&quot;b&quot;,&quot;&quot;) becomes
c(&quot;a&quot;,&quot;a&quot;,&quot;a&quot;,&quot;b&quot;,&quot;b&quot;)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoFill(x, squash = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoFill_+3A_x">x</code></td>
<td>
<p>a vector of character strings</p>
</td></tr>
<tr><td><code id="autoFill_+3A_squash">squash</code></td>
<td>
<p> If set to TRUE then 
leading and trailing spaces are removed which is useful if spaces are
inadvertantly typed because these may be hard to track down. Default: FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>x</code></td>
<td>
<p>a vector of character strings with blank strings replaced by
preceding non&ndash;blank strings</p>
</td></tr>
</table>


<h3>Note</h3>

<p> While this function may be called directly, it is more often
called by <code>makeLabel</code>
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+makeLabel">makeLabel</a></code> uses <code>autoFill</code> to create labels
from two columns of marker names </p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## description: fill out blanks of a vector with preceeding label
  label.1 &lt;- c("a","","","b","")
  print(autoFill(label.1))

  label.2 &lt;- c("agc","","","","gct5","","ccc","","")
  print(autoFill(label.2))
</code></pre>

<hr>
<h2 id='divide.autoMarkers'> Divide markers by parental type</h2><span id='topic+divide.autoMarkers'></span>

<h3>Description</h3>

<p>Given  markers (or more correctly dominant 1,0) marker data
and return list object of containing markers data split
according to parental alleles, namely 1,0 for each parent and
1,1 for both parents
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divide.autoMarkers(markers, description = paste("Markers split for",
deparse(substitute(markers))), parent.cols = c(1, 2),
extra.cols = NULL, cols.drop = c(parent.cols, extra.cols))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="divide.autoMarkers_+3A_markers">markers</code></td>
<td>
<p> matrix of 1, 0, NA indicating marker alleles where
rownames are markernames, column names are progeny names</p>
</td></tr>
<tr><td><code id="divide.autoMarkers_+3A_description">description</code></td>
<td>
<p> text containing a description for printing</p>
</td></tr>
<tr><td><code id="divide.autoMarkers_+3A_parent.cols">parent.cols</code></td>
<td>
<p> column(s) for parental markers (default: 1,2)</p>
</td></tr>
<tr><td><code id="divide.autoMarkers_+3A_extra.cols">extra.cols</code></td>
<td>
<p> extra column(s) to be subsetted (default: NULL)</p>
</td></tr>
<tr><td><code id="divide.autoMarkers_+3A_cols.drop">cols.drop</code></td>
<td>
<p> columns to be dropped from markers before splitting
data which can be set to NULL if no columns are to be dropped
(Default: c(parent.cols,extra.cols))
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns S3 class <code>divideAutoMarkers</code> containing
</p>
<table>
<tr><td><code>p10</code>, <code>p01</code>, <code>p11</code></td>
<td>
<p>lists for where the first, second components are
heterozygous for parents 1, 2 and both resp. Each list contains
</p>

<ul>
<li><p><code>description</code>text containing a description for
printing
</p>
</li>
<li><p><code>parent</code>label for parent
</p>
</li>
<li><p><code>markers</code>markers for specified parental type
(including parents etc)
</p>
</li>
<li><p><code>extras</code>extra columns subsetted (if specified)
</p>
</li>
<li><p><code>seg.ratios</code>segregation ratios as class <code>segRatio</code>
</p>
</li></ul>

</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+sim.autoCross">sim.autoCross</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
p2 &lt;- sim.autoCross(4,
dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),
                     p11=c(0.6,0.2,0.2)))
print(p2)

ss &lt;- divide.autoMarkers(p2$markers)

print(ss)
</code></pre>

<hr>
<h2 id='divideAutoMarkers'> S3 class divideAutoMarkers</h2><span id='topic+divideAutoMarkers'></span>

<h3>Description</h3>

<p>An S3 class which contains marker data and segregation proportions
split into three groups corresponding to parents with &lsquo;01&rsquo;, &lsquo;10&rsquo; and &lsquo;11&rsquo; markers
</p>


<h3>Value</h3>

<table>
<tr><td><code>p10</code>, <code>p01</code>, <code>p11</code></td>
<td>
<p>lists for where the first, second components are
heterozygous for parents 1, 2 and both resp. Each list contains
</p>

<ul>
<li><p><code>description</code>text containing a description for printing
</p>
</li>
<li><p><code>parent</code>label for parent
</p>
</li>
<li><p><code>markers</code>markers for specified parental type
(including parents etc)
</p>
</li>
<li><p><code>extras</code>extra columns subsetted (if specified)
</p>
</li>
<li><p><code>seg.ratios</code>segregation ratios as class <code>segRatio</code>
</p>
</li></ul>

</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+sim.autoCross">sim.autoCross</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
p2 &lt;- sim.autoCross(4,
dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),
                     p11=c(0.6,0.2,0.2)))
print(p2)

ss &lt;- divide.autoMarkers(p2$markers)

print(ss)
</code></pre>

<hr>
<h2 id='expected.segRatio'> Compute theoretical segregation proportions for regular autopolyploids</h2><span id='topic+expected.segRatio'></span>

<h3>Description</h3>

<p>Expected segregation proportions for various dosages of dominant
markers for regular autopolyploids are calculated
using the formula of Ripol et al (1999) based on Haldane (1930) for
single dose and multiple dose parents cross nulliplex
(&quot;homozygous&quot;) and an unpublished formula where both parents
possess at least single dose markers (&quot;heterogeneous&quot;) 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expected.segRatio(ploidy.level = stop("No ploidy level set"),
type.parents = c("heterogeneous", "homozygous"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expected.segRatio_+3A_ploidy.level">ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric or as a character string</p>
</td></tr>
<tr><td><code id="expected.segRatio_+3A_type.parents">type.parents</code></td>
<td>
<p> &quot;heterogeneous&quot; if parental markers are 0,1 or
&quot;homogeneous&quot; if parental markers are both 1</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ratio</code></td>
<td>
<p>vector of proportions for each dosage</p>
</td></tr>
<tr><td><code>ploidy.level</code></td>
<td>
<p>numeric value of ploidy level 2,4,6,8, ...</p>
</td></tr>
<tr><td><code>ploidy.name</code></td>
<td>
<p>name of ploidy</p>
</td></tr>
</table>


<h3>Warning </h3>

<p>While results will be returned if the ploidy level is set as an odd
number, the formula used are only for even numbers. 
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li><p>J B S Haldane (1930) Theoretical genetics of
autopolyploids. <em>Journal of genetics</em> <b>22</b> 359&ndash;372
</p>
</li>
<li><p>Ripol, M I et al(1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31&ndash;41 
</p>
</li></ul>



<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+test.segRatio">test.segRatio</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## heterogeneous parents

expected.segRatio(2)
expected.segRatio("Tetraploid")
expected.segRatio("tEtR")
expected.segRatio("octo")
expected.segRatio("Octa")
expected.segRatio(14)
## warning
expected.segRatio(9)

## errors - not run
## expected.segRatio("abcd")
## expected.segRatio(-1)

## homogeneous parents

expected.segRatio("Octa", type.parents="heter")
expected.segRatio("Octa", type.parents="homo")
expected.segRatio("tetra", type.parents="homo")
expected.segRatio(6, type.parents="homo")
expected.segRatio(9, type.parents="homo")

</code></pre>

<hr>
<h2 id='makeLabel'> Generate labels from two columns where blanks in first column
are replaced by preceding non&ndash;blank label</h2><span id='topic+makeLabel'></span>

<h3>Description</h3>

<p>Primarily used to generate marker labels from two columns where the
first column is a nucleotide sequence which is mainly blank in
that it is the same as the previous one while the second column
is increasing numbers (fragment size) for each nucleotide
combination
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeLabel(x, columns = c(1, 2), squash = TRUE, sep = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeLabel_+3A_x">x</code></td>
<td>
<p>data frame of markers including labels</p>
</td></tr>
<tr><td><code id="makeLabel_+3A_columns">columns</code></td>
<td>
<p> the column numbers containing labels (default: c(1,2))</p>
</td></tr>
<tr><td><code id="makeLabel_+3A_squash">squash</code></td>
<td>
<p> remove trailing/leading blanks in 1st column (default:TRUE) </p>
</td></tr>
<tr><td><code id="makeLabel_+3A_sep">sep</code></td>
<td>
<p> separator when combining two label columns (default: &quot;&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns vector of marker names
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+autoFill">autoFill</a></code> is used to replace blanks in first column</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## imaginary data frame representing ceq marker names read in from
  ## spreadsheet
  x &lt;- data.frame( col1 = c("agc","","","","gct5","","ccc","",""),
                   col2 = c(1,3,4,5,1,2,2,4,6))
  print(x)
  print(makeLabel(x))
  print(cbind(x,lab=makeLabel(x, sep=".")))
</code></pre>

<hr>
<h2 id='plot.segRatio'> Plot segregation ratios for either observed or simulated marker data</h2><span id='topic+plot.segRatio'></span><span id='topic+plot.simAutoMarkers'></span><span id='topic+plot.simAutoCross'></span>

<h3>Description</h3>

<p>Plots an object of S3 class <code>segRatio</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'segRatio'
plot(x, main =
deparse(substitute(x)), xlab="", xlab.segRatio = "Segregation ratio",
xlab.nobs = "Number of dominant markers",
xlab.miss = "Number of missing markers per individual",
NCLASS = 100, type = c("seg.ratio", "all","no","missing"), ...)

## S3 method for class 'simAutoMarkers'
plot(x, main = deparse(substitute(x)), xlab = "Segregation ratio",...)

## S3 method for class 'simAutoCross'
plot(x, main = deparse(substitute(x)), xlab = "Segregation ratio",
...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.segRatio_+3A_x">x</code></td>
<td>
<p> An object of class <code>segRatio</code> </p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_xlab">xlab</code></td>
<td>
<p>label for x axis: not usually set</p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_main">main</code></td>
<td>
<p> Title for plot </p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_xlab.segratio">xlab.segRatio</code></td>
<td>
<p> x&ndash;axis label when plotting segregation proportions </p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_xlab.nobs">xlab.nobs</code></td>
<td>
<p> x axis label when plotting no. of 1's </p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_xlab.miss">xlab.miss</code></td>
<td>
<p>  x axis label when plotting number of missing
individuals per marker</p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_nclass">NCLASS</code></td>
<td>
<p> number of classes for histograms (Default: 100)</p>
</td></tr>
<tr><td><code id="plot.segRatio_+3A_type">type</code></td>
<td>
<p>type of plot may be set to
</p>

<ul>
<li><p><code>seg.ratio</code>Histogram of segregation proportions (Default)
</p>
</li>
<li><p><code>no</code>Histogram of the number of 1s
</p>
</li>
<li><p><code>missing</code>Histogram of the numbers of missing values
per marker
</p>
</li>
<li><p><code>all</code>Produce all plots on one page
</p>
</li></ul>

</td></tr>
<tr><td><code id="plot.segRatio_+3A_...">...</code></td>
<td>
<p> other parameters passed to plot function </p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the histograms are produced of the segregation proportions.
Other histograms that may be produced are numbers of observed
dominant markers (recorded as a 1) and the number of
individuals missing a particular marker.
</p>


<h3>Value</h3>

<p>Used for its side-effects
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+segregationRatios">segregationRatios</a></code>,
<code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code>, <code><a href="#topic+sim.autoCross">sim.autoCross</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## generate some autooctoploid data
a &lt;- sim.autoMarkers(8,c(0.7,0.2,0.09,0.01))

## print markers and plot segratios
print(a)
plot(a$seg.ratios)  # plot the segregation ratios directly
plot(a)             # plot the simAutoMarkers object

## add some missing values and plot all histograms
plot(addMissing(a,0.2)$seg.ratios, type="all")
</code></pre>

<hr>
<h2 id='polySegratio-package'>
Segregation Ratios for Autoployploids
</h2><span id='topic+polySegratio-package'></span><span id='topic+polySegratio'></span>

<h3>Description</h3>

<p>These functions provide tools for computing expected segregation ratios
(or more correctly segregation proportions) for dominant markers in
regular autopolyploids  and simulating such marker data as well as
conducting standard Chi squared tests and Binomial confidance
intervals for assigning marker dosage. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> polySegratio</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.2-5</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-03-22</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Use <code><a href="#topic+expected.segRatio">expected.segRatio</a></code> to compute expected segregation
proportions for regular autopolyploids
</p>
<p>Use <code><a href="#topic+segregationRatios">segregationRatios</a></code> to compute segregation ratios for
a matrix of markers
</p>
<p>Use <code><a href="#topic+test.segRatio">test.segRatio</a></code> to assignmarker dosage via Chi squared
tests or Binomial CIs
</p>
<p>Use <code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code> and <code><a href="#topic+sim.autoCross">sim.autoCross</a></code> to
simulate marker data under various scenarios
</p>
<p>Use <code><a href="#topic+addMisclass">addMisclass</a></code> and <code><a href="#topic+addMissing">addMissing</a></code> make some
markers misclassified or missing at random
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li><p>J B S Haldane (1930) Theoretical genetics of
autopolyploids. <em>Journal of genetics</em> <b>22</b> 359&ndash;372
</p>
</li>
<li><p>Ripol, M I et al(1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31&ndash;41 
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## expected segregation proportions heterogeneous parents
expected.segRatio(4)
expected.segRatio("Tetraploid")
expected.segRatio("Octa")

## expected segregation proportions homogeneous parents
expected.segRatio("Octa",type="heter")

## generate dominant markers for autotetraploids
a1 &lt;- sim.autoMarkers(4,c(0.8,0.2))
print(a1)
plot(a1)

## generate crosses for different parental types
p2 &lt;- sim.autoCross(4, dose.proportion=list(p01=c(0.7,0.3),
                     p10=c(0.7,0.3),p11=c(0.6,0.2,0.2)))
print(p2)
plot(p2)

## simulate and test some markers, printing out a summary table of
## no.s of correct marker dosages 

a &lt;- sim.autoMarkers(ploidy = 8, c(0.7,0.2,0.09,0.01),
                     type="hetero", n.markers=500,n.individuals=100)
a &lt;- addMissing(a, 0.07)  # make seven percent missing at random
at &lt;- test.segRatio(a$seg.ratios, ploidy=8, type.parents="het",
                      method="bin")
print(addmargins(table(a$true.doses$dosage, at$dosage, exclude=NULL)))

</code></pre>

<hr>
<h2 id='print.segRatio'> Print segregation ratios </h2><span id='topic+print.segRatio'></span>

<h3>Description</h3>

<p>Prints an object of S3 class <code>segRatio</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'segRatio'
print(x, digits=3, ..., index = c(1:min(10,length(x$r))) )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.segRatio_+3A_x">x</code></td>
<td>
<p>object of class <code>segRatio</code> </p>
</td></tr>
<tr><td><code id="print.segRatio_+3A_digits">digits</code></td>
<td>
<p>minimal number of <code>significant</code> digits, see
<code><a href="base.html#topic+print.default">print.default</a></code></p>
</td></tr>
<tr><td><code id="print.segRatio_+3A_index">index</code></td>
<td>
<p>which rows of the marker matrix and segregation
proportions to print. (Default: c(1:10))  </p>
</td></tr>
<tr><td><code id="print.segRatio_+3A_...">...</code></td>
<td>
<p>extra parameters passed on to <code>print</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Note</h3>

<p> Objects of class <code><a href="#topic+segRatio">segRatio</a></code> may be produced from
a <code>matrix</code> of markers by
employing the function <code><a href="#topic+segregationRatios">segregationRatios</a></code>
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+segregationRatios">segregationRatios</a></code>,
<code><a href="base.html#topic+print">print</a></code>,<code><a href="base.html#topic+print.default">print.default</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
 ## generate autopolyploid markers
 a1 &lt;- sim.autoMarkers(4,c(0.8,0.2),n.markers=20,n.individuals=10)

 print(class(a1$seg.ratios))
 print(a1$seg.ratios)

</code></pre>

<hr>
<h2 id='print.simAutoMarkers'> Print objects of class  simAutoMarkers </h2><span id='topic+print.simAutoMarkers'></span><span id='topic+print.simAutoCross'></span><span id='topic+print.divideAutoMarkers'></span>

<h3>Description</h3>

<p>Prints an object of S3 class <code>simAutoMarkers</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simAutoMarkers'
print(x, ..., row.index = c(1:min(10,
nrow(x$markers))), col.index = c(1:min(10, ncol(x$markers))) )

## S3 method for class 'simAutoCross'
print(x, ..., row.index = c(1:min(10,
nrow(x$markers))), col.index = c(1:min(10, ncol(x$markers))))

## S3 method for class 'divideAutoMarkers'
print(x, ..., row.index = c(1:10),
  col.index = c(1:10), tabulate.extras = FALSE )

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.simAutoMarkers_+3A_x">x</code></td>
<td>
<p> object of class <code>simAutoMarkers</code> </p>
</td></tr>
<tr><td><code id="print.simAutoMarkers_+3A_row.index">row.index</code></td>
<td>
<p> which rows to print (Default: first 10)</p>
</td></tr>
<tr><td><code id="print.simAutoMarkers_+3A_col.index">col.index</code></td>
<td>
<p> which columns to print (Default: first 10)</p>
</td></tr>
<tr><td><code id="print.simAutoMarkers_+3A_tabulate.extras">tabulate.extras</code></td>
<td>
<p>If TRUE then cross&ndash;tabulate any axtra columns
(Default: FALSE)</p>
</td></tr>
<tr><td><code id="print.simAutoMarkers_+3A_...">...</code></td>
<td>
<p>extra options for printing</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Note</h3>

<p> Objects of class <code><a href="#topic+simAutoMarkers">simAutoMarkers</a></code> may be produced from
by employing the function <code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code> and the same
for <code><a href="#topic+sim.autoCross">sim.autoCross</a></code> and <code><a href="#topic+divide.autoMarkers">divide.autoMarkers</a> </code>
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+segregationRatios">segregationRatios</a></code>,
<code><a href="#topic+sim.autoCross">sim.autoCross</a></code>, <code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code>,
<code><a href="#topic+divide.autoMarkers">divide.autoMarkers</a></code>, <code><a href="base.html#topic+print">print</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## generate data sets
a1 &lt;- sim.autoMarkers(4,c(0.8,0.2))
a2 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.09,0.01),type="homo",n.markers=20,n.individuals=10)

print(a1)
print(a2)

## datasets from crosses
p1 &lt;- sim.autoCross(4, dose.proportion=c(0.7,0.3), n.markers=20, n.indiv=10)
print(p1)
p2 &lt;- sim.autoCross(4, dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),p11=c(0.6,0.2,0.2)))
print(p2)

## divide up data from crosses
ss &lt;- divide.autoMarkers(p2$markers)
print(ss)



</code></pre>

<hr>
<h2 id='print.testSegRatio'> Print objects of class testSegRatio</h2><span id='topic+print.testSegRatio'></span>

<h3>Description</h3>

<p>Prints an object of S3 class <code>testSegRatio</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'testSegRatio'
print(x, ..., last = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.testSegRatio_+3A_x">x</code></td>
<td>
<p> object of class <code>testSegRatio</code></p>
</td></tr>
<tr><td><code id="print.testSegRatio_+3A_last">last</code></td>
<td>
<p> prints from 1 to <code>last</code> segregation ratio tests
(Default: 10) </p>
</td></tr>
<tr><td><code id="print.testSegRatio_+3A_...">...</code></td>
<td>
<p> extra printing options</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+segregationRatios">segregationRatios</a></code>,
<code><a href="#topic+test.segRatio">test.segRatio</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## simulated data
  a &lt;- sim.autoMarkers(ploidy = 8, c(0.7,0.2,0.09,0.01))
  ac &lt;- test.segRatio(a$seg.ratios, ploidy=8, method="chi.squared")
  print(ac)
</code></pre>

<hr>
<h2 id='segRatio'>S3 class segRatio</h2><span id='topic+segRatio'></span>

<h3>Description</h3>

<p>An S3 class which contains the segregation
ratios for dominant markers and other information such as the number
of dominant markers per individual
</p>


<h3>Value</h3>

<table>
<tr><td><code>r</code></td>
<td>
<p>no. of 1's for each individual</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>total no. of markers present for each individual</p>
</td></tr>
<tr><td><code>seg.ratio</code></td>
<td>
<p>segregation proportion for each individual</p>
</td></tr>
<tr><td><code>n.individuals</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+segregationRatios">segregationRatios</a></code>: computing segregation
ratios,  <code><a href="#topic+testSegRatio">testSegRatio</a></code>: chi squared <code class="reqn">\chi^2</code> and
tests and Binomial
confidence intervals for assigning marker dosage,
<code><a href="#topic+expected.segRatio">expected.segRatio</a></code>: compute expected segregation
proportions for various dosages
for dominant markers in regular autopolyploids
</p>

<hr>
<h2 id='segregationRatios'> Compute observed segregation proportions for dominant markers in
autopolyploids </h2><span id='topic+segregationRatios'></span>

<h3>Description</h3>

<p>Computes segregation ratios for a matrix of markers
where the rows are markers and the columns are individuals and the
markers are recorded as 0's and 1's
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segregationRatios(x, drop.cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="segregationRatios_+3A_x">x</code></td>
<td>
 <p><code>matrix</code> of 0's, 1,s and NA's representing scores of
dominant markers where the rows are markers and the columns are
individuals
</p>
</td></tr>
<tr><td><code id="segregationRatios_+3A_drop.cols">drop.cols</code></td>
<td>
 <p><code>numeric</code> columns to drop when calculating
segregation ratios </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>segRatio</code> containing
</p>
<table>
<tr><td><code>r</code></td>
<td>
<p>no. of 1's for each individual</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>total no. of markers present for each individual</p>
</td></tr>
<tr><td><code>seg.ratio</code></td>
<td>
<p>segregation proportion for each individual</p>
</td></tr>
<tr><td><code>n.individuals</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+testSegRatio">testSegRatio</a></code>: chi squared <code class="reqn">\chi^2</code> and
tests and Binomial
confidence intervals for assigning marker dosage, 
<code><a href="#topic+expected.segRatio">expected.segRatio</a></code>: compute expected segregation
proportions for various dosages
for dominant markers in regular autopolyploids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## simulate small autotetraplid data set
a1 &lt;- sim.autoMarkers(4,c(0.8,0.2),n.markers=20,n.individuals=10)
print(a1)

print(segregationRatios(a1$markers))

</code></pre>

<hr>
<h2 id='sim.autoCross'>Simulate dominant markers for an autopolyploid cross for all
parental types </h2><span id='topic+sim.autoCross'></span>

<h3>Description</h3>

<p>Simulates dominant markers from an autopolyploid cross given the ploidy
level and/or expected segregation ratios and the proportions in each
dosage marker class. This is a wrapper to <code>sim.autoMarkers</code> to
generate markers for &lsquo;10&rsquo;, &lsquo;01&rsquo; and &lsquo;11&rsquo; parents
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.autoCross(ploidy.level, prop.par.type = structure(c(0.4, 0.4, 0.2),
names = c("p10", "p01", "p11")), n.markers = 500, n.individuals = 200,
dose.proportion, true.seg.ratios, no.dosage.classes,
marker.names = paste("M", 1:n.markers, sep = "."),
individual.names = paste("X", 1:n.individuals, sep = "."),
parent.names = c("P.1", "P.2"), seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.autoCross_+3A_ploidy.level">ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric (single value) or as a character string containing
type tetraploid, hexaploid, octoploid, ...</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_prop.par.type">prop.par.type</code></td>
<td>
<p> the proportion of markers generated from each
parental type '10', '01' and '11'. Note that the exact number will
be randomly generated from the multinomial distribution
(Default: c(0.4,0.4,0.2))</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_n.markers">n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_n.individuals">n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_dose.proportion">dose.proportion</code></td>
<td>
<p> the proportion of
markers to be simulated in each dosage class. Note that the exact
number will be randomly generated from the multinomial distribution
NB: If a vector is supplied the dose.proportion is same for each
parental type otherwise as list with components &lsquo;p01&rsquo;,
&lsquo;p10&rsquo; and &lsquo;p11&rsquo;</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_true.seg.ratios">true.seg.ratios</code></td>
<td>
<p> numeric vector containing segregation
proportion to be supplied if you wish to overide automatic
calculations using ploidy.level</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_no.dosage.classes">no.dosage.classes</code></td>
<td>
<p>numeric vector containing the number of
dosage classes</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_marker.names">marker.names</code></td>
<td>
<p> labels for markers (Default: M.1 ... M.n.markers)</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_individual.names">individual.names</code></td>
<td>
<p> labels for offspring (Default: ... X.j ... )</p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_parent.names">parent.names</code></td>
<td>
<p> numeric vector of length 2 containing columns
of marker matrix containing parental markers
(Default: first 2 columns) </p>
</td></tr>
<tr><td><code id="sim.autoCross_+3A_seed">seed</code></td>
<td>
<p> integer used to set seed for random number generator
(RNG) which (if set) may be used to reproduce results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+simAutoCross">simAutoCross</a></code> containing
</p>
<table>
<tr><td><code>markers</code></td>
<td>
<p> matrix of 0,1 dominant markers with individuals as
cols and rows as markers</p>
</td></tr>
<tr><td><code>true.dosage</code></td>
<td>
<p><em>true</em> doses for each marker</p>
</td></tr>
<tr><td><code>name.true.dose</code></td>
<td>
<p>names of <em>true</em> doses for each marker</p>
</td></tr>
<tr><td><code>p10</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type &lsquo;10&rsquo; </p>
</td></tr>
<tr><td><code>p01</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type &lsquo;01&rsquo; </p>
</td></tr>
<tr><td><code>p11</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type &lsquo;11&rsquo; </p>
</td></tr>
<tr><td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes as numeric
(single value)</p>
</td></tr>
<tr><td><code>prop.par.type</code></td>
<td>
<p>proportion of markers for each parental type
&lsquo;p01&rsquo;, &lsquo;p10&rsquo; and &lsquo;p11&rsquo;</p>
</td></tr>
<tr><td><code>n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td></tr>
<tr><td><code>n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td></tr>
<tr><td><code>dose.proportion</code></td>
<td>
<p>proportion in each dose &ndash; if numeric vector is
the same for &lsquo;p01&rsquo;, &lsquo;p10&rsquo; and &lsquo;p11&rsquo; else a list
with components &lsquo;p01&rsquo;, &lsquo;p10&rsquo; and &lsquo;p11&rsquo;</p>
</td></tr>
<tr><td><code>no.dosage.classes</code></td>
<td>
<p>number in each dosage class</p>
</td></tr>
<tr><td><code>no.parType</code></td>
<td>
<p>number in each parental type</p>
</td></tr>
<tr><td><code>time.generated</code></td>
<td>
<p>time/date when data set generated</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>seed for random number generator seed which could be used to
reproduce results (I hope)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>matches arguments when function called</p>
</td></tr>
</table>


<h3>Note</h3>

<p>All parameters
except the proportions of marker dosage types can be left at the
default. If only one value is set, then individual list
components will be assumed to be equal. The marker matrix is
prepended with parental marker alleles. An alternative is to
simply create each group using sim.automarkers and cbind them.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+simAutoCross">simAutoCross</a></code>,<code><a href="#topic+simAutoMarkers">simAutoMarkers</a></code>,
<code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- sim.autoCross(4, dose.proportion=c(0.7,0.3), n.markers=20, n.indiv=10)
print(p1)

p2 &lt;- sim.autoCross(4, dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),p11=c(0.6,0.2,0.2)))
print(p2)
</code></pre>

<hr>
<h2 id='sim.autoMarkers'>Simulates dominant markers from an autopolyploid cross</h2><span id='topic+sim.autoMarkers'></span>

<h3>Description</h3>

<p>Dominant markers are simulated from an autopolyploid cross given the
ploidy level, expected segregation ratios and the proportions in
each dosage marker class. This may be chosen from tetraploid to
heccaidecaploid and the segregation ratios may be specified
explicitly or generated automatically.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.autoMarkers(ploidy.level, dose.proportion, n.markers = 500,
n.individuals = 200, seg.ratios, no.dosage.classes,
type.parents = c("heterogeneous", "homozygous"),
marker.names = paste("M", 1:n.markers, sep = "."),
individual.names = paste("X", 1:n.individuals, sep = "."),
overdispersion=FALSE, shape1=50, seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.autoMarkers_+3A_ploidy.level">ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric (single value) or as a character string containing
type tetraploid, hexaploid, octoploid, ...</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_dose.proportion">dose.proportion</code></td>
<td>
<p> the proportion of markers to be simulated in
each dosage class. Note that the exact number will be randomly
generated from the multinomial distribution</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_n.markers">n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_n.individuals">n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_seg.ratios">seg.ratios</code></td>
<td>
<p> numeric vector containing segregation proportion to
be supplied if you wish to override automatic calculations using
ploidy.level</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_no.dosage.classes">no.dosage.classes</code></td>
<td>
<p>only generate markers for the first
<code>no.dosage.classes</code> classes (if set)</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_type.parents">type.parents</code></td>
<td>
<p>heterogeneous for (1,0) or (0,1) homozygous for
(1,1) (default: heterogeneous)</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_marker.names">marker.names</code></td>
<td>
<p> labels for markers (Default: M.1 ... M.n.markers)</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_individual.names">individual.names</code></td>
<td>
<p> labels for offspring (Default: ... X.j ... )</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_overdispersion">overdispersion</code></td>
<td>
<p>logical indicating overdispersion (Default:
FALSE)</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_shape1">shape1</code></td>
<td>
<p>shape1 parameter(s) for the beta distribution used to
generate the Binomial probability p, either of length 1 or
no.dosage.classes. Default: 50 which implies very little
overdispersion. NB: 'shape2' is calculated from shape 1 and
expected segregation ratios</p>
</td></tr>
<tr><td><code id="sim.autoMarkers_+3A_seed">seed</code></td>
<td>
<p> integer used to set seed for random number generator
(RNG) which (if set) may be used to reproduce results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+simAutoMarkers">simAutoMarkers</a></code> containing
</p>
<table>
<tr><td><code>markers</code></td>
<td>
<p> matrix of 0,1 dominant markers with individuals as cols and
rows as markers</p>
</td></tr>
<tr><td><code>E.segRatio</code></td>
<td>
<p>expected segregation proportions, list with
components
</p>

<ul>
<li><p><code>ratio</code>segregation proportions,
</p>
</li>
<li><p><code>ploidy.level</code>level of ploidy 4,6,8, ...
</p>
</li>
<li><p><code>ploidy.name</code>tetraploid, ... , unknown
</p>
</li></ul>

</td></tr>
<tr><td><code>type.parents</code></td>
<td>
<p>heterogeneous for (1,0) or (0,1) homozygous for (1,1)</p>
</td></tr>
<tr><td><code>dose.proportion</code></td>
<td>
<p>proportions of markers set for each dosage class</p>
</td></tr>
<tr><td><code>n.markers</code></td>
<td>
<p>number of markers (Default: 500)</p>
</td></tr>
<tr><td><code>n.individuals</code></td>
<td>
<p>number of individuals in the cross (Default: 200)</p>
</td></tr>
<tr><td><code>true.doses</code></td>
<td>
<p>list containing
</p>

<ul>
<li><p><code>dosage</code> doses generated for each marker for simulation
</p>
</li>
<li><p><code>table.dosages</code>summary of no.s in each dosage
</p>
</li>
<li><p><code>names</code>names for each dosage such as SD (single dose),
DD (double dose), SDxSD etc
</p>
</li></ul>

</td></tr>
<tr><td><code>seg.ratios</code></td>
<td>
<p>object of class segRatio containing segregation ratios</p>
</td></tr>
<tr><td><code>time.generated</code></td>
<td>
<p>time/date when data set generated</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>seed for random number generator seed which could be used to
reproduce results (I hope)</p>
</td></tr>
<tr><td><code>overdispersion</code></td>
<td>
<p>either a list with components 'overdispersion':
logical for whether overdispersion is set or not and if TRUE then
two extra components 'shape1' and 'shape2' contain parameters for
the beta distribution employed to generate Binomial probabilities</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>matches arguments when function called</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For  use in simulation studies, other parameters such as the true dosage
of each marker are also returned. Also, if extra binomial variation or
overdispersion is requested then a beta-binomial distribution is
employed to simulate marker data.Note that as the 'shape1' parameter
becomes larger, the resulting marker data are less overdispersed.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+simAutoMarkers">simAutoMarkers</a></code>,
<code><a href="#topic+print.simAutoMarkers">print.simAutoMarkers</a></code>,  
<code><a href="#topic+plot.simAutoMarkers">plot.simAutoMarkers</a></code>,
<code><a href="#topic+segRatio">segRatio</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## generate autopolyploid markers
a1 &lt;- sim.autoMarkers(4,c(0.8,0.2),n.markers=20,n.individuals=10)
print(a1)

a2 &lt;-
sim.autoMarkers(8,c(0.7,0.2,0.09,0.01),type.parents="homo",n.markers=20,n.individuals=10)
print(a2)

</code></pre>

<hr>
<h2 id='simAutoCross'>S3 class simAutoCross</h2><span id='topic+simAutoCross'></span>

<h3>Description</h3>

<p>An S3 class which contains  simulated dominant marker data for
autopolyploids and other data of interest such as segregation
proportions as well as parameters set for the generating given
parents with &lsquo;01&rsquo;, &lsquo;10&rsquo; and &lsquo;11&rsquo; markers
</p>


<h3>Value</h3>

<table>
<tr><td><code>markers</code></td>
<td>
<p> matrix of 0,1 dominant markers with individuals as
cols and rows as markers</p>
</td></tr>
<tr><td><code>true.dosage</code></td>
<td>
<p><em>true</em> doses for each marker</p>
</td></tr>
<tr><td><code>name.true.dose</code></td>
<td>
<p>names of <em>true</em> doses for each marker</p>
</td></tr>
<tr><td><code>p10</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type &lsquo;10&rsquo; </p>
</td></tr>
<tr><td><code>p01</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type &lsquo;01&rsquo; </p>
</td></tr>
<tr><td><code>p11</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type &lsquo;11&rsquo; </p>
</td></tr>
<tr><td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes as numeric
(single value)</p>
</td></tr>
<tr><td><code>prop.par.type</code></td>
<td>
<p>proportion of markers for each parental type
&lsquo;p01&rsquo;, &lsquo;p10&rsquo; and &lsquo;p11&rsquo;</p>
</td></tr>
<tr><td><code>n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td></tr>
<tr><td><code>n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td></tr>
<tr><td><code>dose.proportion</code></td>
<td>
<p>proportion in each dose &ndash; if numeric vector is
the same for &lsquo;p01&rsquo;, &lsquo;p10&rsquo; and &lsquo;p11&rsquo; else a list
with components sQuotep01, &lsquo;p10&rsquo; and &lsquo;p11&rsquo;</p>
</td></tr>
<tr><td><code>no.dosage.classes</code></td>
<td>
<p>number in each dosage class</p>
</td></tr>
<tr><td><code>no.parType</code></td>
<td>
<p>number in each parental type</p>
</td></tr>
<tr><td><code>time.generated</code></td>
<td>
<p>time/date when data set generated</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>seed for random number generator seed which could be used to
reproduce results (I hope)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>matches arguments when function called</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.autoCross">sim.autoCross</a></code>,<code><a href="#topic+simAutoMarkers">simAutoMarkers</a></code>,
<code><a href="#topic+sim.autoMarkers">sim.autoMarkers</a></code> </p>

<hr>
<h2 id='simAutoMarkers'>S3 class simAutoMarkers</h2><span id='topic+simAutoMarkers'></span>

<h3>Description</h3>

<p>An S3 class which contains the simulated dominant marker data for
autopolyploids and other data of interest such as segregation
proportions as well as parameters set for the generating
</p>


<h3>Value</h3>

<table>
<tr><td><code>markers</code></td>
<td>
<p> matrix of 0,1 dominant markers with individuals as cols and
rows as markers</p>
</td></tr>
<tr><td><code>E.segRatio</code></td>
<td>
<p>expected segregation porportions, list with components
<code>ratio</code>: segregation proportions, <code>ploidy.level</code>: level of
ploidy 4,6,8, ..., <code>ploidy.name</code>: tetraploid, ... , unknown
</p>
</td></tr>
<tr><td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric (single value) or as a character string containing
type tetraploid, hexaploid, octoploid, ...</p>
</td></tr>
<tr><td><code>n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td></tr>
<tr><td><code>n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td></tr>
<tr><td><code>dose.proportion</code></td>
<td>
<p> the proportion of markers to be simulated in
each dosage class. Note that the exact number will be randomly
generated from the multinomial distribution</p>
</td></tr>
<tr><td><code>true.doses</code></td>
<td>
<p>list containing
</p>

<ul>
<li><p><code>dosage</code>doses generated for each marker for simulation
</p>
</li>
<li><p><code>table.dosages</code>summary of no.s in each dosage
</p>
</li>
<li><p><code>names</code>names for each dosage such as (SD) single dose,
(DD) double dose, SDxSD etc
</p>
</li></ul>

</td></tr>
<tr><td><code>seg.ratios</code></td>
<td>
<p>segregation proportions as class
<code><a href="#topic+segRatio">segRatio</a></code></p>
</td></tr>
<tr><td><code>time.generated</code></td>
<td>
<p>date and time data set generated</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p> function call used to generate data set</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+expected.segRatio">expected.segRatio</a></code>,
<code><a href="#topic+segRatio">segRatio</a></code>,
<code><a href="#topic+print.simAutoMarkers">print.simAutoMarkers</a></code>,  
<code><a href="#topic+plot.simAutoMarkers">plot.simAutoMarkers</a></code> </p>

<hr>
<h2 id='test.segRatio'> Classic tests for assessing marker dosage in autopolyploids</h2><span id='topic+test.segRatio'></span>

<h3>Description</h3>

<p>Perform chi&ndash;squared tests or binomial CIs to obtain expected marker dosage
in autopolyploids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test.segRatio(seg.ratio, ploidy.level = 4,
  type.parents = c("heterogeneous", "homozygous"),
  method = c("chi.squared", "binomial"), alpha = 0.05, expected.ratio)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test.segRatio_+3A_seg.ratio">seg.ratio</code></td>
<td>
<p> object of class <code>segRatio</code> containing
segregation proportions</p>
</td></tr>
<tr><td><code id="test.segRatio_+3A_ploidy.level">ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric or as a character string</p>
</td></tr>
<tr><td><code id="test.segRatio_+3A_type.parents">type.parents</code></td>
<td>
<p> &quot;heterogeneous&quot; if parental markers are 0,1 or
&quot;homozygous&quot; if parental markers are both 1</p>
</td></tr>
<tr><td><code id="test.segRatio_+3A_method">method</code></td>
<td>
<p> specify which method &lsquo;chi.squared&rsquo; or
&lsquo;binomial&rsquo; </p>
</td></tr>
<tr><td><code id="test.segRatio_+3A_alpha">alpha</code></td>
<td>
<p> significance level for tests/CIs</p>
</td></tr>
<tr><td><code id="test.segRatio_+3A_expected.ratio">expected.ratio</code></td>
<td>
<p> vector of expected segregation proportions
Default: determined by using function <code>expected.segRatio</code> given
the <code>ploidy.level</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns object of class <code>testSegRatio</code> with components
</p>
<table>
<tr><td><code>probability</code></td>
<td>
<p>matrix of probabilities under the test for each
dosage where columns are doses  and rows are markers</p>
</td></tr>
<tr><td><code>dosage</code></td>
<td>
<p> vector of allocated dosages where allocation unique
otherwise <code>NA</code></p>
</td></tr>
<tr><td><code>allocated</code></td>
<td>
<p> matrix of 0's and 1's where 1 indicates dosage
allocation where columns are doses  and rows are markers</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>alpha level for significance test or CI construction</p>
</td></tr>
<tr><td><code>expected.ratios</code></td>
<td>
<p>expected segregation ratios under null hypotheses</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>call to test.segRatio</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li><p>K Mather(1951) The measurement of linkage in heredity.
<em>Methuen</em> London
</p>
</li>
<li><p>Ripol, M I et al(1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31&ndash;41 
</p>
</li></ul>



<h3>See Also</h3>

 <p><code><a href="#topic+segregationRatios">segregationRatios</a></code> for computing segregation
ratios and <code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+expected.segRatio">expected.segRatio</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## simulated data
  a &lt;- sim.autoMarkers(ploidy = 8, c(0.7,0.2,0.09,0.01))
  print(a)

  ## summarise chi-squared test vs true
  ac &lt;- test.segRatio(a$seg.ratios, ploidy=8, method="chi.squared")
  print(addmargins(table(a$true.doses$dosage, ac$dosage, exclude=NULL)))

  ## summarise binomial CI vs true
  ab &lt;- test.segRatio(a$seg.ratios, ploidy=8, method="bin")
  print(addmargins(table(a$true.doses$dosage, ab$dosage, exclude=NULL)))
</code></pre>

<hr>
<h2 id='testSegRatio'>S3 class testSegRatio</h2><span id='topic+testSegRatio'></span>

<h3>Description</h3>

<p> An S3 class which contains  results of classic tests for
assessing marker dosage in autopolyploids using chi-squared tests or
binomial confidence intervals
</p>


<h3>Value</h3>

<p>Returns object of class <code>testSegRatio</code> with components
</p>
<table>
<tr><td><code>probability</code></td>
<td>
<p>matrix of probabilities under the test for each
dosage where columns are doses  and rows are markers</p>
</td></tr>
<tr><td><code>dosage</code></td>
<td>
<p> vector of allocated dosages where allocation unique
otherwise <code>NA</code></p>
</td></tr>
<tr><td><code>allocated</code></td>
<td>
<p> matrix of 0's and 1's where 1 indicates dosage
allocation where columns are doses  and rows are markers</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>alpha level for significance test or CI construction</p>
</td></tr>
<tr><td><code>expected.ratios</code></td>
<td>
<p>expected segregation ratios under null hypotheses</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>call to test.segRatio</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li><p>K Mather(1951) The measurement of linkage in heredity.
<em>Methuen</em> London
</p>
</li>
<li><p>Ripol, M I et al(1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31&ndash;41 
</p>
</li></ul>



<h3>See Also</h3>

 <p><code><a href="#topic+segRatio">segRatio</a></code>, <code><a href="#topic+expected.segRatio">expected.segRatio</a></code>,
<code><a href="#topic+test.segRatio">test.segRatio</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
