<!DOCTYPE html><html><head><title>Help for package poolVIM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {poolVIM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fisher'><p>fisher</p></a></li>
<li><a href='#gaussianize'><p>gaussianize null variable importances</p></a></li>
<li><a href='#m_effective'><p>meff.</p></a></li>
<li><a href='#poolVIM'><p>after the Actual Impurity Reduction Importance is computed with a Random Forest, pvalues from different probes or SNPs belonging to the same gene can be aggregated in order to obtain a single pvalue for that gene. Correlation between probes can also be taken into account.</p></a></li>
<li><a href='#tippett'><p>tippett.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene-Based Association Tests using the Actual Impurity Reduction
(AIR) Variable Importance</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-07-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Stefano Nembrini &lt;stefanonembrini@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stefano Nembrini &lt;stefanonembrini@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene-based association tests using the actual impurity reduction (AIR) variable importance. The function aggregates AIR importance measures from a group of SNPs or probes and outputs a p-value for each gene. The procedures builds upon the method described in &lt;<a href="https://doi.org/10.1093%2FBioinformatics%2FBty373">doi:10.1093/Bioinformatics/Bty373</a>&gt; and will be published soon.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, ranger</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.3.1), EmpiricalBrownsMethod(&ge; 1.6.0), Hmisc(&ge; 4.1)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-30 07:18:24 UTC; S</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-30 17:30:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='fisher'>fisher</h2><span id='topic+fisher'></span>

<h3>Description</h3>

<p>fisher
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher(p, adjust, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisher_+3A_p">p</code></td>
<td>
<p>vector of pvalues</p>
</td></tr>
<tr><td><code id="fisher_+3A_adjust">adjust</code></td>
<td>
<p>if correlation has to be taken into account</p>
</td></tr>
<tr><td><code id="fisher_+3A_r">R</code></td>
<td>
<p>correlation matrix</p>
</td></tr>
</table>

<hr>
<h2 id='gaussianize'>gaussianize null variable importances</h2><span id='topic+gaussianize'></span>

<h3>Description</h3>

<p>gaussianize null variable importances
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaussianize(x, a)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gaussianize_+3A_x">x</code></td>
<td>
<p>distr</p>
</td></tr>
<tr><td><code id="gaussianize_+3A_a">a</code></td>
<td>
<p>value to interpolate/extrapolate</p>
</td></tr>
</table>

<hr>
<h2 id='m_effective'>meff.</h2><span id='topic+m_effective'></span>

<h3>Description</h3>

<p>meff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>m_effective(R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="m_effective_+3A_r">R</code></td>
<td>
<p>R</p>
</td></tr>
</table>

<hr>
<h2 id='poolVIM'>after the Actual Impurity Reduction Importance is computed with a Random Forest, pvalues from different probes or SNPs belonging to the same gene can be aggregated in order to obtain a single pvalue for that gene. Correlation between probes can also be taken into account.</h2><span id='topic+poolVIM'></span>

<h3>Description</h3>

<p>after the Actual Impurity Reduction Importance is computed with a Random Forest, pvalues from different probes or SNPs belonging to the same gene can be aggregated in order to obtain a single pvalue for that gene. Correlation between probes can also be taken into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poolVIM(rf, genenames, x, method = "Tippett", adjust)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="poolVIM_+3A_rf">rf</code></td>
<td>
<p>a ranger object with &quot;importance=&quot;impurity_corrected&quot;</p>
</td></tr>
<tr><td><code id="poolVIM_+3A_genenames">genenames</code></td>
<td>
<p>a vector of the name of the gene to which each probe or SNP belongs, it has to be of size dim(x)[1]</p>
</td></tr>
<tr><td><code id="poolVIM_+3A_x">x</code></td>
<td>
<p>design matrix used by the random forest</p>
</td></tr>
<tr><td><code id="poolVIM_+3A_method">method</code></td>
<td>
<p>one of Tippett, Fisher, Kost, EBM</p>
</td></tr>
<tr><td><code id="poolVIM_+3A_adjust">adjust</code></td>
<td>
<p>&quot;no&quot; / &quot;yes&quot; depending if correlation has to be taken into account</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 250
x=replicate(50, runif(n))
dat &lt;- data.frame(y = factor(rbinom(n, 1, .5)), x)
library(ranger)
rf &lt;- ranger(y ~ ., dat, importance = "impurity_corrected",num.trees=100)
genenames=colnames(x)=rep(c("G1","G2"),50/2)
poolVIM(rf,genenames,x,method="Fisher",adjust="no")

</code></pre>

<hr>
<h2 id='tippett'>tippett.</h2><span id='topic+tippett'></span>

<h3>Description</h3>

<p>tippett.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tippett(p, adjust, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tippett_+3A_p">p</code></td>
<td>
<p>vector of pvalues</p>
</td></tr>
<tr><td><code id="tippett_+3A_adjust">adjust</code></td>
<td>
<p>if correlation has to be taken into account</p>
</td></tr>
<tr><td><code id="tippett_+3A_r">R</code></td>
<td>
<p>correlation matrix</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
