<!DOCTYPE html><html lang="en"><head><title>Help for package popgenr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {popgenr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#coal'><p>Simulate and visualize a coalescent process</p></a></li>
<li><a href='#Dcalc'><p>Calculate and visualize LD</p></a></li>
<li><a href='#fly'><p>Drosphila melanogaster bw75 data</p></a></li>
<li><a href='#genotypes'><p>Genotype data from Aleppo Pines</p></a></li>
<li><a href='#moth'><p>Temporal allele frequency shifts</p></a></li>
<li><a href='#snp'><p>SNP information for 25 loci</p></a></li>
<li><a href='#thirteen'><p>Genotypes across 13 CODIS loci</p></a></li>
<li><a href='#whale'><p>Genotypes of 246 South Pacific Blue Whales</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Accompaniment to Population Genetics with R: An Introduction for
Life Scientists</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Aki Jarl Laruson; Floyd Reed</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aki Jarl Laruson &lt;akijarl@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides several data sets and functions to accompany the book "Population Genetics with R: An Introduction for Life Scientists" (2021, ISBN:9780198829546).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/publicdomain/zero/1.0/legalcode">CC0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-24 23:09:15 UTC; root</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://global.oup.com/academic/product/population-genetics-with-r-9780198829546?cc=gb&amp;lang=en#">https://global.oup.com/academic/product/population-genetics-with-r-9780198829546?cc=gb&amp;lang=en#</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-25 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='coal'>Simulate and visualize a coalescent process</h2><span id='topic+coal'></span>

<h3>Description</h3>

<p>A function that provides a simple starting off point to simulate a coalescent process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coal(length, number, muscale, reps, prnt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coal_+3A_length">length</code></td>
<td>
<p>Length of sequence to simulate.</p>
</td></tr>
<tr><td><code id="coal_+3A_number">number</code></td>
<td>
<p>Number of starting lineages.</p>
</td></tr>
<tr><td><code id="coal_+3A_muscale">muscale</code></td>
<td>
<p>Mutation rate, scaled by 4Neu.</p>
</td></tr>
<tr><td><code id="coal_+3A_reps">reps</code></td>
<td>
<p>Number of replicates to run.</p>
</td></tr>
<tr><td><code id="coal_+3A_prnt">prnt</code></td>
<td>
<p>Whether to print calculation output to the terminal (prnt=1), or not (prnt=0).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#assign genotype counts
length=250

number=8

muscale=6.25 # theta = 4Neu = 25, theta/4 = 6.25

reps=100

prnt=0

#run function
coal(length, number, muscale, reps, prnt)
</code></pre>

<hr>
<h2 id='Dcalc'>Calculate and visualize LD</h2><span id='topic+Dcalc'></span>

<h3>Description</h3>

<p>Calculates the degree of linkage disequilibirum between two biallelic diploid loci and plots the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dcalc(AABB, AaBB, aaBB, AABb, AaBb, aaBb, AAbb, Aabb, aabb)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Dcalc_+3A_aabb">AABB</code></td>
<td>
<p>Count of A and B double homozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">AaBB</code></td>
<td>
<p>Count of A heterozygotes, B homozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">aaBB</code></td>
<td>
<p>Count of a and B double homozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">AABb</code></td>
<td>
<p>Count of A homozygotes, B heterozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">AaBb</code></td>
<td>
<p>Count of double heterozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">aaBb</code></td>
<td>
<p>Count of a homozygotes, B heterozygote.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">AAbb</code></td>
<td>
<p>Count of A and b double homozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">Aabb</code></td>
<td>
<p>Count of A heterozygotes, b homozygotes.</p>
</td></tr>
<tr><td><code id="Dcalc_+3A_aabb">aabb</code></td>
<td>
<p>Count of a and b double homozygotes.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#assign genotype counts
AABB=2
AaBB=0
aaBB=0
AABb=0
AaBb=1
aaBb=0
AAbb=1
Aabb=0
aabb=0

#run function
Dcalc(AABB, AaBB, aaBB, AABb, AaBb, aaBb, AAbb, Aabb, aabb)
</code></pre>

<hr>
<h2 id='fly'>Drosphila melanogaster bw75 data</h2><span id='topic+fly'></span>

<h3>Description</h3>

<p>A matrix of observed allele counts from Buri (1956). Each row contains observations starting at generation 1 in the first row, across 107 replicates of the number of bw75 allele.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fly
</code></pre>


<h3>Source</h3>

<p>Buri, P. (1956). Gene frequency in small populations of mutant Drosophila.
Evolution, 10, 367-402.
</p>

<hr>
<h2 id='genotypes'>Genotype data from Aleppo Pines</h2><span id='topic+genotypes'></span>

<h3>Description</h3>

<p>Multiple sampled alleles from Aleppo Pine (Pinus halepensis) in the Eastern Mediterranean. Adapted from Gershberg et al. 2016.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypes
</code></pre>


<h3>Source</h3>

<p>Gershberg, A., Ne'eman, G., Ben-Shlomo, R. (2016). Genetic structure of a
naturally regenerating post-fire seedling population: Pinus halepensis as
a case study. Frontiers in Plant Science 7: 549.
</p>

<hr>
<h2 id='moth'>Temporal allele frequency shifts</h2><span id='topic+moth'></span>

<h3>Description</h3>

<p>Change in allele frequency over eight generations in the scarlet tiger moth (Callimorpha
dominula).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moth
</code></pre>


<h3>Source</h3>

<p>Fisher, R. A. and E. B. Ford (1947). The Spread of a Gene in Natural Conditions
in a Colony of the Moth Panaxia dominula. Heredity 1: 143-174.
</p>

<hr>
<h2 id='snp'>SNP information for 25 loci</h2><span id='topic+snp'></span>

<h3>Description</h3>

<p>Single Nucleotide Polymorphism data for 25 loci. Includes information on allele frequency, homozygosity, heterozygosity, chromosome location, and functional location of SNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snp
</code></pre>


<h3>Source</h3>

<p>1,000 Genomes Project (http://www.internationalgenome.org)
</p>

<hr>
<h2 id='thirteen'>Genotypes across 13 CODIS loci</h2><span id='topic+thirteen'></span>

<h3>Description</h3>

<p>Genotypes of 1,036 individuals across 13 core CODIS (Combined DNA Index System) loci, collected in the USA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>thirteen
</code></pre>


<h3>Source</h3>

<p>Hill, C.R., Duewer, D.L., Kline, M.C., Coble, M.D., and Butler, J.M. (2013) U.S. population data for 29 autosomal STR loci. Forensic Sci. Int. Genet. 7: e82-e83.
Accessed from: https://strbase.nist.gov/fbicore.htm
</p>

<hr>
<h2 id='whale'>Genotypes of 246 South Pacific Blue Whales</h2><span id='topic+whale'></span>

<h3>Description</h3>

<p>Data frame of genotypes collected from 264 individual blue Whales (Balaenoptera musculus) across seven distinct loci. Adapted from Attard et al. 2012.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>whale
</code></pre>


<h3>Source</h3>

<p>Attard, C. R., Beheregaray, L. B., Jenner, K. C. S., Gill, P. C., Jenner, M. N.,
Morrice, M. G., Robertson, K.M. and Moller, L. M. (2012). Hybridization
of Southern Hemisphere blue whale subspecies and a sympatric area off
Antarctica: impacts of whaling or climate change? Molecular Ecology,
21(23), 5715-5727.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
