<!DOCTYPE html><html lang="en"><head><title>Help for package PopGenHelpR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PopGenHelpR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Ancestry_barchart'><p>Plot an ancestry matrix for individuals and(or) populations.</p></a></li>
<li><a href='#Differentiation'><p>A function to estimate three measures of genetic differentiation using geno files, vcf files, or vcfR objects. Data is assumed to be bi-allelic.</p></a></li>
<li><a href='#Fst_dat'><p>A genetic differentiation matrix and locality information for each population. This data was generated</p>
by subsetting data of Farleigh et al., 2021.</a></li>
<li><a href='#Het_dat'><p>A data frame of hypothetical heterozygosity data produced by Heterozygosity.</p></a></li>
<li><a href='#Heterozygosity'><p>A function to estimate seven measures of heterozygosity using geno files, vcf files, or vcfR objects. Data is assumed to be bi-allelic.</p></a></li>
<li><a href='#HornedLizard_Pop'><p>A population assignment data frame to be used in <code>Heterozygosity</code> and <code>Differentiation</code>.</p></a></li>
<li><a href='#HornedLizard_VCF'><p>A vcfR object to be used in <code>Heterozygosity</code> and <code>Differentiation</code>.</p></a></li>
<li><a href='#Network_map'><p>A function to map statistics (i.e., genetic differentiation) between points as a network on a map.</p></a></li>
<li><a href='#Pairwise_heatmap'><p>A function to plot a heatmap from a symmetric matrix.</p></a></li>
<li><a href='#PCA'><p>A function to perform principal component analysis (PCA) on genetic data. Loci with missing data will be removed prior to PCA.</p></a></li>
<li><a href='#Piechart_map'><p>Plot a map of ancestry pie charts.</p></a></li>
<li><a href='#Plot_coordinates'><p>A function to plot coordinates on a map.</p></a></li>
<li><a href='#Point_map'><p>A function to map statistics as colored points on a map.</p></a></li>
<li><a href='#Private.alleles'><p>A function to estimate the number of private alleles in each population.</p></a></li>
<li><a href='#Q_dat'><p>A list representing a q-matrix and the locality information associated with the qmatrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Streamline Population Genomic and Genetic Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimate commonly used population genomic statistics and generate publication quality figures. 'PopGenHelpR' uses vcf, 'geno' (012), and csv files to generate output.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://kfarleigh.github.io/PopGenHelpR/">https://kfarleigh.github.io/PopGenHelpR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kfarleigh/PopGenHelpR/issues">https://github.com/kfarleigh/PopGenHelpR/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, magrittr, methods, reshape2, rlang,
scatterpie, stats, spData, spdep, utils, vcfR</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-31 16:58:42 UTC; Keaka</td>
</tr>
<tr>
<td>Author:</td>
<td>Keaka Farleigh <a href="https://orcid.org/0000-0002-9195-121X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, cre],
  Mason Murphy <a href="https://orcid.org/0000-0002-5317-1386"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, ctb],
  Christopher Blair <a href="https://orcid.org/0000-0002-4782-6788"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, ctb],
  Tereza Jezkova <a href="https://orcid.org/0000-0003-4114-4564"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Keaka Farleigh &lt;keakafarleigh@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-02 10:20:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='Ancestry_barchart'>Plot an ancestry matrix for individuals and(or) populations.</h2><span id='topic+Ancestry_barchart'></span>

<h3>Description</h3>

<p>Plot an ancestry matrix for individuals and(or) populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ancestry_barchart(
  anc.mat,
  pops,
  K,
  plot.type = "all",
  col,
  ind.order = NULL,
  pop.order = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Ancestry_barchart_+3A_anc.mat">anc.mat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The first column should be the names of each sample/population, followed by the estimated contribution of each cluster to that individual/pop.</p>
</td></tr>
<tr><td><code id="Ancestry_barchart_+3A_pops">pops</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The columns should be named Sample, containing the sample IDs; Population indicating the population assignment of the individual, population and sample names must be the same type (i.e., both numeric or both characters); Long, indicating the longitude of the sample; Lat, indicating the latitude of the sample.</p>
</td></tr>
<tr><td><code id="Ancestry_barchart_+3A_k">K</code></td>
<td>
<p>Numeric.The number of genetic clusters in your data set, please contact the package authors if you need help doing this.</p>
</td></tr>
<tr><td><code id="Ancestry_barchart_+3A_plot.type">plot.type</code></td>
<td>
<p>Character string. Options are all, individual, and population. All is default and recommended, this will plot a barchart for both the individuals and populations.</p>
</td></tr>
<tr><td><code id="Ancestry_barchart_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting.</p>
</td></tr>
<tr><td><code id="Ancestry_barchart_+3A_ind.order">ind.order</code></td>
<td>
<p>Character vector indicating the order to plot the individuals in the individual ancestry bar chart.</p>
</td></tr>
<tr><td><code id="Ancestry_barchart_+3A_pop.order">pop.order</code></td>
<td>
<p>Chracter vector indicating the order to plot the populations in the population ancesyry bar chart.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing your plots and the data frames used to generate the plots.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Q_dat)
Qmat &lt;- Q_dat[[1]]
rownames(Qmat) &lt;- Qmat[,1]
Loc &lt;- Q_dat[[2]]
Test_all &lt;- Ancestry_barchart(anc.mat = Qmat, pops = Loc, K = 5,
plot.type = 'all',col = c('#d73027', '#fc8d59', '#e0f3f8', '#91bfdb', '#4575b4'))
</code></pre>

<hr>
<h2 id='Differentiation'>A function to estimate three measures of genetic differentiation using geno files, vcf files, or vcfR objects. Data is assumed to be bi-allelic.</h2><span id='topic+Differentiation'></span>

<h3>Description</h3>

<p>A function to estimate three measures of genetic differentiation using geno files, vcf files, or vcfR objects. Data is assumed to be bi-allelic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Differentiation(
  data,
  pops,
  statistic = "all",
  missing_value = NA,
  write = FALSE,
  prefix = NULL,
  population_col = NULL,
  individual_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Differentiation_+3A_data">data</code></td>
<td>
<p>Character. String indicating the name of the vcf file, geno file or vcfR object to be used in the analysis.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_pops">pops</code></td>
<td>
<p>Character. String indicating the name of the population assignment file or dataframe containing the population assignment information for each individual in the data. This file must be in the same order as the vcf file and include columns specifying the individual and the population that individual belongs to. The first column should contain individual names and the second column should indicate the population assignment of each individual. Alternatively, you can indicate the column containing the individual and population information using the individual_col and population_col arguments.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_statistic">statistic</code></td>
<td>
<p>Character. String or vector indicating the statistic to calculate. Options are any of: all; all of the statistics; Fst, Weir and Cockerham (1984) Fst; NeisD, Nei's D statistic; JostsD, Jost's D.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_missing_value">missing_value</code></td>
<td>
<p>Character. String indicating missing data in the input data. It is assumed to be NA, but that may not be true (is likely not) in the case of geno files.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_write">write</code></td>
<td>
<p>Boolean. Whether or not to write the output to files in the current working directory. There will be one or two files for each statistic. Files will be named based on their statistic such as Fst_perpop.csv.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_prefix">prefix</code></td>
<td>
<p>Character. Optional argument. String that will be appended to file output. Please provide a prefix if write is set to TRUE.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_population_col">population_col</code></td>
<td>
<p>Numeric. Optional argument (a number) indicating the column that contains the population assignment information.</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_individual_col">individual_col</code></td>
<td>
<p>Numeric. Optional argument (a number) indicating the column that contains the individuals (i.e., sample name) in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimated heterozygosity statistics. The per pop values are calculated by taking the average of the per locus estimates.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>References</h3>

<p><b>Fst:</b>
</p>
<p>Pembleton, L. W., Cogan, N. O., &amp; Forster, J. W. (2013). StAMPP: An R package for calculation of genetic differentiation and structure of mixed‐ploidy level populations. Molecular ecology resources, 13(5), 946-952.<a href="https://doi.org/10.1111/1755-0998.12129">doi:10.1111/1755-0998.12129</a>
</p>
<p><a href="https://www.jstor.org/stable/2408641">Weir, B. S., &amp; Cockerham, C. C. (1984)</a>. Estimating F-statistics for the analysis of population structure. evolution, 1358-1370.
</p>
<p><b>Nei's D:</b>
</p>
<p>Nei, M. (1972). Genetic distance between populations. The American Naturalist, 106(949), 283-292.<a href="https://doi.org/10.1086/282771">doi:10.1086/282771</a>
</p>
<p><a href="https://doi.org/10.1111/1755-0998.12129">doi:10.1111/1755-0998.12129</a> Pembleton, L. W., Cogan, N. O., &amp; Forster, J. W. (2013). StAMPP: An R package for calculation of genetic differentiation and structure of mixed‐ploidy level populations. Molecular ecology resources, 13(5), 946-952.
</p>
<p><b>Jost's D:</b>
</p>
<p>Jost L (2008). GST and its relatives do not measure differentiation. Molecular Ecology, 17, 4015–4026.<a href="https://doi.org/10.1111/j.1365-294X.2008.03887.x">doi:10.1111/j.1365-294X.2008.03887.x</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Differentiation(data = HornedLizard_VCF, pops = HornedLizard_Pop, write = FALSE)
</code></pre>

<hr>
<h2 id='Fst_dat'>A genetic differentiation matrix and locality information for each population. This data was generated
by subsetting data of Farleigh et al., 2021.</h2><span id='topic+Fst_dat'></span>

<h3>Description</h3>

<p>A symmetric matrix with estimated genetic differentiation (Fst) between 3 populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fst_dat)
</code></pre>


<h3>Format</h3>

<p>A list with two elements:
</p>

<dl>
<dt>Fst_dat</dt><dd><p>Data frame with three rows and three columns</p>
</dd>
<dt>Loc_dat</dt><dd><p>Data frame containing the locality information for each population</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fst_dat)
Fst &lt;- Fst_dat[[1]]
Loc &lt;- Fst_dat[[2]]

 Test &lt;- Network_map(dat = Fst, pops = Loc,
neighbors = 2,col = c('#4575b4', '#91bfdb', '#e0f3f8','#fd8d3c','#fc4e2a'),
statistic = "Fst", Lat_buffer = 1, Long_buffer = 1)

Fstat_plot &lt;- Pairwise_heatmap(dat = Fst, statistic = 'FST')

</code></pre>

<hr>
<h2 id='Het_dat'>A data frame of hypothetical heterozygosity data produced by Heterozygosity.</h2><span id='topic+Het_dat'></span>

<h3>Description</h3>

<p>Data frame containing 5 columns and 3 rows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Het_dat)
</code></pre>


<h3>Format</h3>

<p>A data frame with 5 columns and 3 rows:
</p>

<dl>
<dt>Heterozygosity</dt><dd><p>Estimated heterozygosity</p>
</dd>
<dt>Pop</dt><dd><p>Population assignment</p>
</dd>
<dt>Standard.Deviation</dt><dd><p>standard deviation</p>
</dd>
<dt>Longitude</dt><dd><p>Longitude</p>
</dd>
<dt>Latitude</dt><dd><p>Latitude</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Coordinates and population names taken from Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Het_dat)
Test &lt;- Point_map(Het_dat, statistic = "Heterozygosity")

</code></pre>

<hr>
<h2 id='Heterozygosity'>A function to estimate seven measures of heterozygosity using geno files, vcf files, or vcfR objects. Data is assumed to be bi-allelic.</h2><span id='topic+Heterozygosity'></span>

<h3>Description</h3>

<p>A function to estimate seven measures of heterozygosity using geno files, vcf files, or vcfR objects. Data is assumed to be bi-allelic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Heterozygosity(
  data,
  pops,
  statistic = "all",
  missing_value = NA,
  write = FALSE,
  prefix = NULL,
  population_col = NULL,
  individual_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Heterozygosity_+3A_data">data</code></td>
<td>
<p>Character. String indicating the name of the vcf file, geno file or vcfR object to be used in the analysis.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_pops">pops</code></td>
<td>
<p>Character. String indicating the name of the population assignment file or dataframe containing the population assignment information for each individual in the data. This file must be in the same order as the vcf file and include columns specifying the individual and the population that individual belongs to. The first column should contain individual names and the second column should indicate the population assignment of each individual. Alternatively, you can indicate the column containing the individual and population information using the individual_col and population_col arguments.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_statistic">statistic</code></td>
<td>
<p>Character. String or vector indicating the statistic to calculate. Options are any of: all; all of the statistics; Ho, observed heterozygosity; He, expected heterozygosity; PHt, proportion of heterozygous loci; Hs_exp, heterozygosity standardized by the average expected heterozygosity; Hs_obs, heterozygosity standardized by the average observed heterozygosity; IR, internal relatedness; HL, homozygosity by locus.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_missing_value">missing_value</code></td>
<td>
<p>Character. String indicating missing data in the input data. It is assumed to be NA, but that may not be true (is likely not) in the case of geno files.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_write">write</code></td>
<td>
<p>Boolean. Whether or not to write the output to files in the current working directory. There will be one or two files for each statistic. Files will be named based on their statistic such as Ho_perpop.csv or Ho_perloc.csv.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_prefix">prefix</code></td>
<td>
<p>Character. Optional argument. String that will be appended to file output. Please provide a prefix if write is set to TRUE.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_population_col">population_col</code></td>
<td>
<p>Numeric. Optional argument (a number) indicating the column that contains the population assignment information.</p>
</td></tr>
<tr><td><code id="Heterozygosity_+3A_individual_col">individual_col</code></td>
<td>
<p>Numeric. Optional argument (a number) indicating the column that contains the individuals (i.e., sample name) in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimated heterozygosity statistics. The per pop values are calculated by taking the average of the per locus estimates.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>References</h3>

<p><b>Expected (He) and observed heterozygosity (Ho):</b>
</p>
<p>Nei, M. (1987) Molecular Evolutionary Genetics. Columbia University Press
</p>
<p><b>Homozygosity by locus (HL) and internal relatedness (IR):</b>
</p>
<p><a href="https://onlinelibrary.wiley.com/doi/full/10.1111/j.1755-0998.2010.02830.x?casa_token=QiNcMSJyunkAAAAA%3Agv-CK7GrUn1bHSgz4qZSOcB2nyHDeR8B1Wtm9bM7q7vZCAcJhNkhTWnpM0EfkSCb2EvkRrr2ArMzC7v7">Alho, J. S., Välimäki, K., &amp; Merilä, J. (2010)</a>. Rhh: an R extension for estimating multilocus heterozygosity and heterozygosity–heterozygosity correlation. Molecular ecology resources, 10(4), 720-722.
</p>
<p>Amos, W., Worthington Wilmer, J., Fullard, K., Burg, T. M., Croxall, J. P., Bloch, D., &amp; Coulson, T. (2001). The influence of parental relatedness on reproductive success. Proceedings of the Royal Society of London. Series B: Biological Sciences, 268(1480), 2021-2027.<a href="https://doi.org/10.1098/rspb.2001.1751">doi:10.1098/rspb.2001.1751</a>
</p>
<p><a href="https://onlinelibrary.wiley.com/doi/10.1111/j.1365-294X.2006.03111.x">Aparicio, J. M., Ortego, J., &amp; Cordero, P. J. (2006)</a>. What should we weigh to estimate heterozygosity, alleles or loci?. Molecular Ecology, 15(14), 4659-4665.
</p>
<p><b>Heterozygosity standardized by expected (Hs_exp) and observed heterozygosity (Hs_obs):</b>
</p>
<p><a href="https://academic.oup.com/evolut/article/53/4/1259/6757148">Coltman, D. W., Pilkington, J. G., Smith, J. A., &amp; Pemberton, J. M. (1999)</a>. Parasite‐mediated selection against Inbred Soay sheep in a free‐living island populaton. Evolution, 53(4), 1259-1267.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Heterozygosity(data = HornedLizard_VCF, pops = HornedLizard_Pop, write = FALSE)
</code></pre>

<hr>
<h2 id='HornedLizard_Pop'>A population assignment data frame to be used in <code>Heterozygosity</code> and <code>Differentiation</code>.</h2><span id='topic+HornedLizard_Pop'></span>

<h3>Description</h3>

<p>Data frame containing 4 columns and 72 rows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HornedLizard_Pop)
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 columns and 72 rows:
</p>

<dl>
<dt>Sample</dt><dd><p>Sample Name</p>
</dd>
<dt>Population</dt><dd><p>Population assignment according to sNMF results (see citation)</p>
</dd>
<dt>Longitude</dt><dd><p>Longitude</p>
</dd>
<dt>Latitude</dt><dd><p>Latitude</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Coordinates and population names taken from Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Differentiation(data = HornedLizard_VCF, pops = HornedLizard_Pop, write = FALSE)


</code></pre>

<hr>
<h2 id='HornedLizard_VCF'>A vcfR object to be used in <code>Heterozygosity</code> and <code>Differentiation</code>.</h2><span id='topic+HornedLizard_VCF'></span>

<h3>Description</h3>

<p>Data frame containing 4 columns and 72 rows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HornedLizard_Pop)
</code></pre>


<h3>Format</h3>

<p>A vcfR object
</p>

<dl>
<dt>vcfR object</dt><dd><p>A vcfR object containing genotype and sample informaiton for 72 individuals.</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Heterozygosity(data = HornedLizard_VCF, pops = HornedLizard_Pop, write = FALSE)


</code></pre>

<hr>
<h2 id='Network_map'>A function to map statistics (i.e., genetic differentiation) between points as a network on a map.</h2><span id='topic+Network_map'></span>

<h3>Description</h3>

<p>A function to map statistics (i.e., genetic differentiation) between points as a network on a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Network_map(
  dat,
  pops,
  neighbors,
  col,
  statistic = NULL,
  breaks = NULL,
  Lat_buffer = 1,
  Long_buffer = 1,
  Latitude_col = NULL,
  Longitude_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Network_map_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. If it is a csv, the 1st row should contain the individual/population names. The columns should also be named in this fashion.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_pops">pops</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The columns should be named Sample, containing the sample IDs; Population indicating the population assignment of the individual; Long, indicating the longitude of the sample; Lat, indicating the latitude of the sample. Alternatively, see the Longitude_col and Latitude_col arguments.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_neighbors">neighbors</code></td>
<td>
<p>Numeric or character. The number of neighbors to plot connections with, or the specific relationship that you want to visualize. Names should match those in the population assignment file and be seperated by an underscore. If I want to visualize the relationship between East and West, for example, I would set neighbors = &quot;East_West&quot;.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_statistic">statistic</code></td>
<td>
<p>Character indicating the statistic being plotted. This will be used to title the legend. The legend title will be blank if left as NULL.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_breaks">breaks</code></td>
<td>
<p>Numeric. The breaks used to generate the color ramp when plotting. Users should supply 3 values if custom breaks are desired.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_latitude_col">Latitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the latitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Lat column.</p>
</td></tr>
<tr><td><code id="Network_map_+3A_longitude_col">Longitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the longitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Long column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the map and the matrix used to plot the map.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Fst_dat)
Fst &lt;- Fst_dat[[1]]
Loc &lt;- Fst_dat[[2]]
Test &lt;- Network_map(dat = Fst, pops = Loc,
neighbors = 2,col = c('#4575b4', '#91bfdb', '#e0f3f8','#fd8d3c','#fc4e2a'),
statistic = "Fst", Lat_buffer = 1, Long_buffer = 1)
</code></pre>

<hr>
<h2 id='Pairwise_heatmap'>A function to plot a heatmap from a symmetric matrix.</h2><span id='topic+Pairwise_heatmap'></span>

<h3>Description</h3>

<p>A function to plot a heatmap from a symmetric matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pairwise_heatmap(dat, statistic, col = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Pairwise_heatmap_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. If it is a csv, the 1st row should contain the individual/population names. The columns should also be named in this fashion.</p>
</td></tr>
<tr><td><code id="Pairwise_heatmap_+3A_statistic">statistic</code></td>
<td>
<p>Character indicating the statistic represented in the matrix, this will be used to label the plot.</p>
</td></tr>
<tr><td><code id="Pairwise_heatmap_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors to be used in plotting. The vector should contain two colors, the first will be the low value, the second will be the high value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' data(Fst_dat)
Fst &lt;- Fst_dat[[1]]
Fstat_plot &lt;- Pairwise_heatmap(dat = Fst, statistic = 'FST')
</code></pre>

<hr>
<h2 id='PCA'>A function to perform principal component analysis (PCA) on genetic data. Loci with missing data will be removed prior to PCA.</h2><span id='topic+PCA'></span>

<h3>Description</h3>

<p>A function to perform principal component analysis (PCA) on genetic data. Loci with missing data will be removed prior to PCA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA(
  data,
  center = TRUE,
  scale = FALSE,
  missing_value = NA,
  write = FALSE,
  prefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PCA_+3A_data">data</code></td>
<td>
<p>Character. String indicating the name of the vcf file, geno file or vcfR object to be used in the analysis.</p>
</td></tr>
<tr><td><code id="PCA_+3A_center">center</code></td>
<td>
<p>Boolean. Whether or not to center the data before principal component analysis.</p>
</td></tr>
<tr><td><code id="PCA_+3A_scale">scale</code></td>
<td>
<p>Boolean. Whether or not to scale the data before principal component analysis.</p>
</td></tr>
<tr><td><code id="PCA_+3A_missing_value">missing_value</code></td>
<td>
<p>Character. String indicating missing data in the input data. It is assumed to be NA, but that may not be true (is likely not) in the case of geno files.</p>
</td></tr>
<tr><td><code id="PCA_+3A_write">write</code></td>
<td>
<p>Boolean. Whether or not to write the output to files in the current working directory. There will be two files, one for the individual loadings and the other for the percent variance explained by each axis.</p>
</td></tr>
<tr><td><code id="PCA_+3A_prefix">prefix</code></td>
<td>
<p>Character. Optional argument. String that will be appended to file output. Please provide a prefix if write is set to TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements: the loadings of individuals on each principal component and the variance explained by each principal component.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_VCF")
Test &lt;- PCA(data = HornedLizard_VCF)
</code></pre>

<hr>
<h2 id='Piechart_map'>Plot a map of ancestry pie charts.</h2><span id='topic+Piechart_map'></span>

<h3>Description</h3>

<p>Plot a map of ancestry pie charts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Piechart_map(
  anc.mat,
  pops,
  K,
  plot.type = "all",
  col,
  piesize = 0.35,
  Lat_buffer,
  Long_buffer,
  Latitude_col = NULL,
  Longitude_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Piechart_map_+3A_anc.mat">anc.mat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The first column should be the names of each sample/population, followed by the estimated contribution of each cluster to that individual/pop.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_pops">pops</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The columns should be named Sample, containing the sample IDs; Population indicating the population assignment of the individual, population and sample names must be the same type (i.e., both numeric or both characters); Long, indicating the longitude of the sample; Lat, indicating the latitude of the sample. Alternatively, see the Longitude_col and Latitude_col arguments.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_k">K</code></td>
<td>
<p>Numeric.The number of genetic clusters in your data set, please contact the package authors if you need help doing this.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_plot.type">plot.type</code></td>
<td>
<p>Character string. Options are all, individual, and population. All is default and recommended, this will plot a piechart map for both the individuals and populations.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_piesize">piesize</code></td>
<td>
<p>Numeric. The radius of the pie chart for ancestry mapping.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_latitude_col">Latitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the latitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Lat column.</p>
</td></tr>
<tr><td><code id="Piechart_map_+3A_longitude_col">Longitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the longitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Long column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing your plots and the data frames used to generate the plots.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Q_dat)
Qmat &lt;- Q_dat[[1]]
rownames(Qmat) &lt;- Qmat[,1]
Loc &lt;- Q_dat[[2]]
Test_all &lt;- Piechart_map(anc.mat = Qmat, pops = Loc, K = 5,
plot.type = 'all', col = c('#d73027', '#fc8d59', '#e0f3f8', '#91bfdb', '#4575b4'), piesize = 0.35,
Lat_buffer = 1, Long_buffer = 1)
</code></pre>

<hr>
<h2 id='Plot_coordinates'>A function to plot coordinates on a map.</h2><span id='topic+Plot_coordinates'></span>

<h3>Description</h3>

<p>A function to plot coordinates on a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_coordinates(
  dat,
  col = c("#A9A9A9", "#000000"),
  size = 3,
  Lat_buffer = 1,
  Long_buffer = 1,
  Latitude_col = NULL,
  Longitude_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Plot_coordinates_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The coordinates of each row should be indicated by columns named Longitude and Latitude. Alternatively, see the Latitude_col and Longitude_col arugments.</p>
</td></tr>
<tr><td><code id="Plot_coordinates_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting, two colors are allowed. The first color will be the fill color, the second is the outline color. For example, if I want red points with a black outline I would set col to col = c(&quot;#FF0000&quot;, &quot;#000000&quot;).</p>
</td></tr>
<tr><td><code id="Plot_coordinates_+3A_size">size</code></td>
<td>
<p>Numeric. The size of the points to plot.</p>
</td></tr>
<tr><td><code id="Plot_coordinates_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Plot_coordinates_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Plot_coordinates_+3A_latitude_col">Latitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the latitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Latitude column.</p>
</td></tr>
<tr><td><code id="Plot_coordinates_+3A_longitude_col">Longitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the longitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Longitude column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_Pop")
Test &lt;- Plot_coordinates(HornedLizard_Pop)
</code></pre>

<hr>
<h2 id='Point_map'>A function to map statistics as colored points on a map.</h2><span id='topic+Point_map'></span>

<h3>Description</h3>

<p>A function to map statistics as colored points on a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Point_map(
  dat,
  statistic,
  size = 3,
  breaks = NULL,
  col,
  out.col = NULL,
  Lat_buffer = 1,
  Long_buffer = 1,
  Latitude_col = NULL,
  Longitude_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Point_map_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The first column should be the statistic to be plotted. The coordinates of each row should be indicated by columns named Longitude and Latitude. Alternatively, see the Longitude_col and Latitude_col arguments.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_statistic">statistic</code></td>
<td>
<p>Character string. The statistic to be plotted.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_size">size</code></td>
<td>
<p>Numeric. The size of the points to plot.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_breaks">breaks</code></td>
<td>
<p>Numeric. The breaks used to generate the color ramp when plotting. Users should supply 3 values if custom breaks are desired.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting, three colors are allowed (low, mid, high). The first color will be the low color, the second the middle, the third the high.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_out.col">out.col</code></td>
<td>
<p>Character. A color for outlining points on the map. There will be no visible outline if left as NULL.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_latitude_col">Latitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the latitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Latitude column.</p>
</td></tr>
<tr><td><code id="Point_map_+3A_longitude_col">Longitude_col</code></td>
<td>
<p>Numeric. The number of the column indicating the longitude for each sample. If this is not null, PopGenHelpR will use this column instead of looking for the Longitude column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing maps and the data frames used to generate them.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Het_dat)
Test &lt;- Point_map(Het_dat, statistic = "Heterozygosity")
</code></pre>

<hr>
<h2 id='Private.alleles'>A function to estimate the number of private alleles in each population.</h2><span id='topic+Private.alleles'></span>

<h3>Description</h3>

<p>A function to estimate the number of private alleles in each population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Private.alleles(
  data,
  pops,
  write = FALSE,
  prefix = NULL,
  population_col = NULL,
  individual_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Private.alleles_+3A_data">data</code></td>
<td>
<p>Character. String indicating the name of the vcf file or vcfR object to be used in the analysis.</p>
</td></tr>
<tr><td><code id="Private.alleles_+3A_pops">pops</code></td>
<td>
<p>Character. String indicating the name of the population assignment file or dataframe containing the population assignment information for each individual in the data. This file must be in the same order as the vcf file and include columns specifying the individual and the population that individual belongs to. The first column should contain individual names and the second column should indicate the population assignment of each individual. Alternatively, you can indicate the column containing the individual and population information using the individual_col and population_col arguments.</p>
</td></tr>
<tr><td><code id="Private.alleles_+3A_write">write</code></td>
<td>
<p>Boolean. Optional argument indicating Whether or not to write the output to a file in the current working directory. This will output to files; 1) the table of private allele counts per population (named prefix_PrivateAlleles_countperpop) and 2) metadata associated with the private alleles (named prefix_PrivateAlleles_metadata). Please supply a prefix it you write files to your working directory as a best practice.</p>
</td></tr>
<tr><td><code id="Private.alleles_+3A_prefix">prefix</code></td>
<td>
<p>Character. Optional argument indicating a string that will be appended to file output. Please set a prefix if write is TRUE.</p>
</td></tr>
<tr><td><code id="Private.alleles_+3A_population_col">population_col</code></td>
<td>
<p>Numeric. Optional argument (a number) indicating the column that contains the population assignment information.</p>
</td></tr>
<tr><td><code id="Private.alleles_+3A_individual_col">individual_col</code></td>
<td>
<p>Numeric. Optional argument (a number) indicating the column that contains the individuals (i.e., sample name) in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the count of private alleles in each population and the metadata for those alleles. The metadata is a list that contains the private allele and locus name for each population.
</p>


<h3>Author(s)</h3>

<p>Keaka Farleigh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Private.alleles(data = HornedLizard_VCF, pops = HornedLizard_Pop, write = FALSE)
</code></pre>

<hr>
<h2 id='Q_dat'>A list representing a q-matrix and the locality information associated with the qmatrix</h2><span id='topic+Q_dat'></span>

<h3>Description</h3>

<p>List with two elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Q_dat)
</code></pre>


<h3>Format</h3>

<p>A list with two elements:
</p>

<dl>
<dt>Qmat</dt><dd><p>A q-matrix with 6 columns and 30 rows, the first column lists the sample name and the remaining 5 represent the contribution a genetic cluster to that individuals ancestry</p>
</dd>
<dt>Loc_dat</dt><dd><p>The locality information for each individual in the q-matrix</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Data was generated by package authors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Q_dat)
Qmat &lt;- Q_dat[[1]]
rownames(Qmat) &lt;- Qmat[,1]
Loc &lt;- Q_dat[[2]]
Test_all &lt;- Ancestry_barchart(anc.mat = Qmat, pops = Loc, K = 5,
plot.type = 'all',col = c('#d73027', '#fc8d59', '#e0f3f8', '#91bfdb', '#4575b4'))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
