<!DOCTYPE html><html><head><title>Help for package PopGenHelpR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PopGenHelpR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Dif_stats'><p>A function to calculate differentiation statistics and perform significance testing with a vcf file.</p></a></li>
<li><a href='#Dif_Stats_Map'><p>A function to map differentiation statistics.</p></a></li>
<li><a href='#Div_stats'><p>A function to estimate heterozygosity and the number of private alleles from a vcf file.</p></a></li>
<li><a href='#Div_Stats_Map'><p>A function to map diversity statistics.</p></a></li>
<li><a href='#Fst_dat'><p>A genetic differentiation matrix and locality information for each population. This data was generated</p>
by subsetting data of Farleigh et al., 2021.</a></li>
<li><a href='#Het_dat'><p>A data frame of hypothetical heterozygosity data produced by Div_Stats.</p></a></li>
<li><a href='#HornedLizard_Pop'><p>A population assignment data frame to be used in <code>Div_stats</code> and <code>Dif_stats</code>.</p></a></li>
<li><a href='#HornedLizard_VCF'><p>A vcfR object to be used in <code>Div_stats</code> and <code>Dif_stats</code>.</p></a></li>
<li><a href='#Pairwise_heatmap'><p>A function to plot a heatmap from a symmetric matrix.</p></a></li>
<li><a href='#Plot_ancestry'><p>Plot an ancestry matrix and map of ancestry pie charts.</p></a></li>
<li><a href='#Q_dat'><p>A list representing a q-matrix and the locality information associated with the qmatrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Streamline Population Genomic and Genetic Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimate commonly used population genomic statistics and generate publication quality figures. The current version of 'PopGenHelpR' uses vcf and csv files to generate output, however, future implementations will expand the input file type options.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://kfarleigh.github.io/PopGenHelpR/">https://kfarleigh.github.io/PopGenHelpR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kfarleigh/PopGenHelpR/issues">https://github.com/kfarleigh/PopGenHelpR/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>adegenet, dartR, dplyr, ggplot2, gstat, hierfstat, magrittr,
methods, poppr, raster, reshape2, rlang, scatterpie, StAMPP,
stats, sp, spData, spdep, utils, vcfR</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-15 16:09:22 UTC; Keaka</td>
</tr>
<tr>
<td>Author:</td>
<td>Keaka Farleigh <a href="https://orcid.org/0000-0002-9195-121X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, cre],
  Mason Murphy <a href="https://orcid.org/0000-0002-5317-1386"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, ctb],
  Christopher Blair <a href="https://orcid.org/0000-0002-4782-6788"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, ctb],
  Tereza Jezkova <a href="https://orcid.org/0000-0003-4114-4564"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Keaka Farleigh &lt;keakafarleigh@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-16 09:32:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='Dif_stats'>A function to calculate differentiation statistics and perform significance testing with a vcf file.</h2><span id='topic+Dif_stats'></span>

<h3>Description</h3>

<p>A function to calculate differentiation statistics and perform significance testing with a vcf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dif_stats(
  VCF,
  pops,
  ploidy,
  statistic = "both",
  boots,
  write = FALSE,
  prefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dif_stats_+3A_vcf">VCF</code></td>
<td>
<p>Character string indicating the name of the vcf file to be used in analysis.</p>
</td></tr>
<tr><td><code id="Dif_stats_+3A_pops">pops</code></td>
<td>
<p>Character string indicating the name of the population assignment file. This file should have four columns and be in the same order as your vcf file. The first column named Sample indicates the sample name. The second column named Population indicates the population assignment of each individual. The third column named Long indicates the longitude of the sample.  The fourth column named Lat indicates the latitude of the sample.</p>
</td></tr>
<tr><td><code id="Dif_stats_+3A_ploidy">ploidy</code></td>
<td>
<p>Numeric. The ploidy of the data.</p>
</td></tr>
<tr><td><code id="Dif_stats_+3A_statistic">statistic</code></td>
<td>
<p>Character string. Options are both, FST, and NeisD.</p>
</td></tr>
<tr><td><code id="Dif_stats_+3A_boots">boots</code></td>
<td>
<p>Numeric. The number of boostraps to use to evaluate statistical significance. Only relevant for FST estimation.</p>
</td></tr>
<tr><td><code id="Dif_stats_+3A_write">write</code></td>
<td>
<p>Boolean. Whether or not to write the output to a file in the current working directory.</p>
</td></tr>
<tr><td><code id="Dif_stats_+3A_prefix">prefix</code></td>
<td>
<p>Character string that will be appended to file output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contianing data frames for the requested statistic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Dif_stats(VCF = HornedLizard_VCF, pops = HornedLizard_Pop,
ploidy = 2, statistic = "both", boots = 10, write = FALSE)
</code></pre>

<hr>
<h2 id='Dif_Stats_Map'>A function to map differentiation statistics.</h2><span id='topic+Dif_Stats_Map'></span>

<h3>Description</h3>

<p>A function to map differentiation statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dif_Stats_Map(
  dat,
  pops,
  neighbors,
  col,
  breaks = NULL,
  Lat_buffer,
  Long_buffer
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dif_Stats_Map_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. If it is a csv, the 1st row should contain the individual/population names. The columns should also be named in this fashion.</p>
</td></tr>
<tr><td><code id="Dif_Stats_Map_+3A_pops">pops</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The columns should be named Sample, containing the sample IDs; Population indicating the population assignment of the individual; Long, indicating the longitude of the sample; Lat, indicating the latitude of the sample.</p>
</td></tr>
<tr><td><code id="Dif_Stats_Map_+3A_neighbors">neighbors</code></td>
<td>
<p>Numeric. The number of neighbors to plot connections with.</p>
</td></tr>
<tr><td><code id="Dif_Stats_Map_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting.</p>
</td></tr>
<tr><td><code id="Dif_Stats_Map_+3A_breaks">breaks</code></td>
<td>
<p>Numeric. The breaks used to generate the color ramp when plotting. Users should supply 3 values if custom breaks are desired.</p>
</td></tr>
<tr><td><code id="Dif_Stats_Map_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Dif_Stats_Map_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the map and the matrix used to plot the map.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Fst_dat)
Fst &lt;- Fst_dat[[1]]
Loc &lt;- Fst_dat[[2]]
Test &lt;- Dif_Stats_Map(dat = Fst, pops = Loc,
neighbors = 2,
col = c('#fd8d3c','#fc4e2a','#e31a1c','#bd0026','#800026'),Lat_buffer = 1, Long_buffer = 1)
</code></pre>

<hr>
<h2 id='Div_stats'>A function to estimate heterozygosity and the number of private alleles from a vcf file.</h2><span id='topic+Div_stats'></span>

<h3>Description</h3>

<p>A function to estimate heterozygosity and the number of private alleles from a vcf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Div_stats(VCF, pops, ploidy, write = FALSE, prefix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Div_stats_+3A_vcf">VCF</code></td>
<td>
<p>Character string indicating the name of the vcf file to be used in analysis.</p>
</td></tr>
<tr><td><code id="Div_stats_+3A_pops">pops</code></td>
<td>
<p>Character string indicating the name of the population assignment file. This file should have four columns and be in the same order as your vcf file. The first column named Sample indicates the sample name. The second column named Population indicates the population assignment of each individual. The third column named Longitude indicates the longitude of the sample.  The fourth column named Latitude indicates the latitude of the sample.</p>
</td></tr>
<tr><td><code id="Div_stats_+3A_ploidy">ploidy</code></td>
<td>
<p>Numeric. The ploidy of the data.</p>
</td></tr>
<tr><td><code id="Div_stats_+3A_write">write</code></td>
<td>
<p>Boolean. Whether or not to write the output to a file in the current working directory.</p>
</td></tr>
<tr><td><code id="Div_stats_+3A_prefix">prefix</code></td>
<td>
<p>Character string that will be appended to file output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimated diversity statistics, model output from linear regression of these statistics against latitude, and model plots.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Div_stats(VCF = HornedLizard_VCF, pops = HornedLizard_Pop,
ploidy = 2, write = FALSE)
</code></pre>

<hr>
<h2 id='Div_Stats_Map'>A function to map diversity statistics.</h2><span id='topic+Div_Stats_Map'></span>

<h3>Description</h3>

<p>A function to map diversity statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Div_Stats_Map(
  dat,
  plot.type = "all",
  statistic,
  breaks = NULL,
  col,
  Lat_buffer = 1,
  Long_buffer = 1,
  write = FALSE,
  prefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Div_Stats_Map_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The first column should be the statistic to be plotted and named the same as the statistic argument. The second column is Population indicating which population each row belongs to. The third column is the standard deviation, the fourth column is Long indicating the longitude, and the fifth column is Lat, indicating the latitude.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_plot.type">plot.type</code></td>
<td>
<p>Character string. Options are all, point, or interpolated. All is recommended and will generate a map with points colored according to heterozygosity as well as a rater of interpolated heterozygosity values.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_statistic">statistic</code></td>
<td>
<p>Character string. The statistic to be plotted.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_breaks">breaks</code></td>
<td>
<p>Numeric. The breaks used to generate the color ramp when plotting. Users should supply 3 values if custom breaks are desired.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting, three colors are allowed (low, mid, high). The first color will be the low color, the second the middle, the third the high.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_write">write</code></td>
<td>
<p>Boolean. Whether or not to write the output to a file in the current working directory.</p>
</td></tr>
<tr><td><code id="Div_Stats_Map_+3A_prefix">prefix</code></td>
<td>
<p>Character string that will be appended to file output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing maps and the data frames used to generate them.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Het_dat)
Test_het &lt;- Div_Stats_Map(dat = Het_dat, plot.type = 'all',
statistic = "Heterozygosity",
Lat_buffer = 1, Long_buffer = 1, write = FALSE, prefix = 'Test_het')
</code></pre>

<hr>
<h2 id='Fst_dat'>A genetic differentiation matrix and locality information for each population. This data was generated
by subsetting data of Farleigh et al., 2021.</h2><span id='topic+Fst_dat'></span>

<h3>Description</h3>

<p>A symmetric matrix with estimated genetic differentiation (Fst) between 3 populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fst_dat)
</code></pre>


<h3>Format</h3>

<p>A list with two elements:
</p>

<dl>
<dt>Fst_dat</dt><dd><p>Data frame with three rows and three columns</p>
</dd>
<dt>Loc_dat</dt><dd><p>Data frame containing the locality information for each population</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fst_dat)
Fst &lt;- Fst_dat[[1]]
Loc &lt;- Fst_dat[[2]]

Test &lt;- Dif_Stats_Map(dat = Fst, pops = Loc,
neighbors = 2,
col = c('#fd8d3c','#fc4e2a','#e31a1c','#bd0026','#800026'),Lat_buffer = 1, Long_buffer = 1)

Fstat_plot &lt;- Pairwise_heatmap(dat = Fst, statistic = 'FST')

</code></pre>

<hr>
<h2 id='Het_dat'>A data frame of hypothetical heterozygosity data produced by Div_Stats.</h2><span id='topic+Het_dat'></span>

<h3>Description</h3>

<p>Data frame containing 5 columns and 3 rows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Het_dat)
</code></pre>


<h3>Format</h3>

<p>A data frame with 5 columns and 3 rows:
</p>

<dl>
<dt>Heterozygosity</dt><dd><p>Estimated heterozygosity</p>
</dd>
<dt>Pop</dt><dd><p>Population assignment</p>
</dd>
<dt>Standard.Deviation</dt><dd><p>standard deviation</p>
</dd>
<dt>Longitude</dt><dd><p>Longitude</p>
</dd>
<dt>Latitude</dt><dd><p>Latitude</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Coordinates and population names taken from Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Het_dat)
Test_het &lt;- Div_Stats_Map(dat = Het_dat, plot.type = 'all',
statistic = "Heterozygosity",
Lat_buffer = 1, Long_buffer = 1, write = FALSE, prefix = 'Test_het')

</code></pre>

<hr>
<h2 id='HornedLizard_Pop'>A population assignment data frame to be used in <code>Div_stats</code> and <code>Dif_stats</code>.</h2><span id='topic+HornedLizard_Pop'></span>

<h3>Description</h3>

<p>Data frame containing 4 columns and 72 rows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HornedLizard_Pop)
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 columns and 72 rows:
</p>

<dl>
<dt>Sample</dt><dd><p>Sample Name</p>
</dd>
<dt>Population</dt><dd><p>Population assignment according to sNMF results (see citation)</p>
</dd>
<dt>Longitude</dt><dd><p>Longitude</p>
</dd>
<dt>Latitude</dt><dd><p>Latitude</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Coordinates and population names taken from Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Div_stats(VCF = HornedLizard_VCF, pops = HornedLizard_Pop,
ploidy = 2, write = FALSE)


</code></pre>

<hr>
<h2 id='HornedLizard_VCF'>A vcfR object to be used in <code>Div_stats</code> and <code>Dif_stats</code>.</h2><span id='topic+HornedLizard_VCF'></span>

<h3>Description</h3>

<p>Data frame containing 4 columns and 72 rows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HornedLizard_Pop)
</code></pre>


<h3>Format</h3>

<p>A vcfR object
</p>

<dl>
<dt>vcfR object</dt><dd><p>A vcfR object containing genotype and sample informaiton for 72 individuals.</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T., Koochekian, N., Schield, D. R., ... &amp; Jezkova, T. (2021). The effects of climate and demographic history in shaping genomic variation across populations of the Desert Horned Lizard (Phrynosoma platyrhinos). Molecular Ecology, 30(18), 4481-4496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data("HornedLizard_Pop")
data("HornedLizard_VCF")
Test &lt;- Div_stats(VCF = HornedLizard_VCF, pops = HornedLizard_Pop,
ploidy = 2, write = FALSE)


</code></pre>

<hr>
<h2 id='Pairwise_heatmap'>A function to plot a heatmap from a symmetric matrix.</h2><span id='topic+Pairwise_heatmap'></span>

<h3>Description</h3>

<p>A function to plot a heatmap from a symmetric matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pairwise_heatmap(dat, statistic, col = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Pairwise_heatmap_+3A_dat">dat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. If it is a csv, the 1st row should contain the individual/population names. The columns should also be named in this fashion.</p>
</td></tr>
<tr><td><code id="Pairwise_heatmap_+3A_statistic">statistic</code></td>
<td>
<p>Character indicating the statistic represented in the matrix, this will be used to label the plot.</p>
</td></tr>
<tr><td><code id="Pairwise_heatmap_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors to be used in plotting. The vector should contain two colors, the first will be the low value, the second will be the high value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' data(Fst_dat)
Fst &lt;- Fst_dat[[1]]
Fstat_plot &lt;- Pairwise_heatmap(dat = Fst, statistic = 'FST')
</code></pre>

<hr>
<h2 id='Plot_ancestry'>Plot an ancestry matrix and map of ancestry pie charts.</h2><span id='topic+Plot_ancestry'></span>

<h3>Description</h3>

<p>Plot an ancestry matrix and map of ancestry pie charts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_ancestry(
  anc.mat,
  pops,
  K,
  plot.type = "all",
  col,
  Lat_buffer,
  Long_buffer
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_ancestry_+3A_anc.mat">anc.mat</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The first column should be the names of each sample/population, followed by the estimated contribution of each cluster to that individual/pop.</p>
</td></tr>
<tr><td><code id="Plot_ancestry_+3A_pops">pops</code></td>
<td>
<p>Data frame or character string that supplies the input data. If it is a character string, the file should be a csv. The columns should be named Sample, containing the sample IDs; Population indicating the population assignment of the individual, population and sample names must be the same type (i.e., both numeric or both characters); Long, indicating the longitude of the sample; Lat, indicating the latitude of the sample.</p>
</td></tr>
<tr><td><code id="Plot_ancestry_+3A_k">K</code></td>
<td>
<p>Numeric.The number of genetic clusters in your data set, please contact the package authors if you need help doing this.</p>
</td></tr>
<tr><td><code id="Plot_ancestry_+3A_plot.type">plot.type</code></td>
<td>
<p>Character string. Options are all, individual, and population. All is default and recommended, this will plot a barchart and piechart map for both the individuals and populations.</p>
</td></tr>
<tr><td><code id="Plot_ancestry_+3A_col">col</code></td>
<td>
<p>Character vector indicating the colors you wish to use for plotting.</p>
</td></tr>
<tr><td><code id="Plot_ancestry_+3A_lat_buffer">Lat_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
<tr><td><code id="Plot_ancestry_+3A_long_buffer">Long_buffer</code></td>
<td>
<p>Numeric. A buffer to customize visualization.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing your plots and the data frames used to generate the plots.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Q_dat)
Qmat &lt;- Q_dat[[1]]
rownames(Qmat) &lt;- Qmat[,1]
Loc &lt;- Q_dat[[2]]
Test_all &lt;- Plot_ancestry(anc.mat = Qmat, pops = Loc, K = 5,
plot.type = 'all', col &lt;- c('red', 'maroon', 'navy', 'cyan', 'blue'),
Lat_buffer = 1, Long_buffer = 1)
</code></pre>

<hr>
<h2 id='Q_dat'>A list representing a q-matrix and the locality information associated with the qmatrix</h2><span id='topic+Q_dat'></span>

<h3>Description</h3>

<p>List with two elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Q_dat)
</code></pre>


<h3>Format</h3>

<p>A list with two elements:
</p>

<dl>
<dt>Qmat</dt><dd><p>A q-matrix with 6 columns and 30 rows, the first column lists the sample name and the remaining 5 represent the contribution a genetic cluster to that individuals ancestry</p>
</dd>
<dt>Loc_dat</dt><dd><p>The locality information for each individual in the q-matrix</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Data was generated by package authors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Q_dat)
Qmat &lt;- Q_dat[[1]]
rownames(Qmat) &lt;- Qmat[,1]
Loc &lt;- Q_dat[[2]]
Test_all &lt;- Plot_ancestry(anc.mat = Qmat, pops = Loc, K = 5,
plot.type = 'all', col &lt;- c('red', 'maroon', 'navy', 'cyan', 'blue'),
Lat_buffer = 1, Long_buffer = 1)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
