<!DOCTYPE html><html lang="en"><head><title>Help for package POSTm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {POSTm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#p.adjust'><p>Adjust P-values for Multiple Comparisons</p></a></li>
<li><a href='#plot'><p>Plot Phylogenies with Significant OTU</p></a></li>
<li><a href='#post'><p>Phylogeney-Guided OTU-Specific Association Test for Microbiome Data</p></a></li>
<li><a href='#POSTmData'><p>Example Dataset For Illustration</p></a></li>
<li><a href='#print'><p>Print the Primary Results of a post() Analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Phylogeny-Guided OTU-Specific Association Test for Microbiome
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-04-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Caizhi Huang [aut],
  Jung-Ying Tzeng [aut],
  Shannon T. Holloway [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shannon T. Holloway &lt;shannon.t.holloway@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the Phylogeny-Guided Microbiome OTU-Specific Association 
    Test method, which boosts the testing power by adaptively borrowing 
    information from phylogenetically close OTUs (operational taxonomic units)
    of the target OTU. This method
    is built on a kernel machine regression framework and allows for flexible 
    modeling of complex microbiome effects, adjustments for covariates, and 
    can accommodate both continuous and binary outcomes. </td>
</tr>
<tr>
<td>Depends:</td>
<td>methods</td>
</tr>
<tr>
<td>Imports:</td>
<td>CompQuadForm, ape</td>
</tr>
<tr>
<td>Suggests:</td>
<td>multtest, rmarkdown, knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Collate:</td>
<td>'ACAT.R' 'CKAT.R' 'POSTmData.R' 'distanceMatrix.R'
'getKernel.R' 'glmFit.R' 'lmFit.R' 'p.adjust.POST.R'
'plot.POST.R' 'post.R' 'print.POST.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-17 22:52:04 UTC; 19194</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-17 23:10:18 UTC</td>
</tr>
</table>
<hr>
<h2 id='p.adjust'>Adjust P-values for Multiple Comparisons</h2><span id='topic+p.adjust'></span><span id='topic+p.adjust.POST'></span>

<h3>Description</h3>

<p>Given a set of p-values, returns p-values adjusted using one of
several methods.
</p>
<p>Given an object of class POST, returns the POST and SO p-values
adjusted using the specified method(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p.adjust(p, ...)

## S3 method for class 'POST'
p.adjust(p, ..., method = p.adjust.methods, n = length(x = p), alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p.adjust_+3A_p">p</code></td>
<td>
<p>An object of class POST. The value object returned by post().</p>
</td></tr>
<tr><td><code id="p.adjust_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
<tr><td><code id="p.adjust_+3A_method">method</code></td>
<td>
<p>A character vector object. One or more correction methods.
must be one or more from {&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,
&quot;fdr&quot;, &quot;none&quot;, &quot;TSBH&quot;}.</p>
</td></tr>
<tr><td><code id="p.adjust_+3A_n">n</code></td>
<td>
<p>number of comparisons, must be at least ‘length(p)’; 
Per ?stats::p.adjust &ndash; only set this (to non-default) when you know 
what you are doing!</p>
</td></tr>
<tr><td><code id="p.adjust_+3A_alpha">alpha</code></td>
<td>
<p>A numeric object. A nominal type I error rate, or a vector of 
error rates, used for estimating the number of true null hypotheses in 
the two-stage Benjamini &amp; Hochberg procedure (&quot;TSBH&quot;). Default is 0.05.
NOTE: This definition does not conform to stats::p.adjust.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extends the p.adjust() method of stats and incorporates the TSBH
method available through multtest::mt.raw2adjp.
</p>


<h3>Value</h3>

<p>A list, the contents of which will depend on the type of analysis.
If a tree object was provided for the analysis, the list will contain
</p>
<table role = "presentation">
<tr><td><code>adjPOST</code></td>
<td>
<p>A matrix object. The adjusted p-values for the POST p-values.
The rows correspond to each OTU; the
columns to each specified adjustment method.</p>
</td></tr>
<tr><td><code>adjSO</code></td>
<td>
<p>A matrix object. The adjusted p-values for the single OTU p-values.
The rows correspond to each OTU; the
columns to each specified adjustment method.</p>
</td></tr>
</table>
<p>If no tree object was provided, only adjSO as described above will be returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("POSTmData")

y &lt;- as.integer(x = metadata[,"GC"] == "BV")
X &lt;- metadata[,"mRace"]

result &lt;- post(y = y, 
               X = X, 
               OTU = otu[,1:20], 
               tree = otutree,
               cValues = seq(0,0.05,by=0.01))

 p.adjust(p = result, method = c("BH","BY"))

</code></pre>

<hr>
<h2 id='plot'>Plot Phylogenies with Significant OTU</h2><span id='topic+plot'></span><span id='topic+plot.POST'></span>

<h3>Description</h3>

<p>Plot phylogenetic tree on the current graphical device as defined by
package ape with tip labels highlighted to show the significant OTU.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'POST'
plot(x, ..., siglevel = 0.05, method = "none", alpha = 0.05, subTree = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_+3A_x">x</code></td>
<td>
<p>A POST object. The value object returned by post(). Note that
a tree of class &quot;phylo&quot; must have been provided to the original
analysis.</p>
</td></tr>
<tr><td><code id="plot_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to plot.phylo.</p>
</td></tr>
<tr><td><code id="plot_+3A_siglevel">siglevel</code></td>
<td>
<p>A numeric object. The significance level; OTU's with 
(adjusted or raw) p-values below this level will be highlighted.
Default is 0.05.</p>
</td></tr>
<tr><td><code id="plot_+3A_method">method</code></td>
<td>
<p>A character object. Must be one of {&quot;holm&quot;, &quot;hochberg&quot;, 
&quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;TSBH&quot;, &quot;none&quot;}, where
'none' indicates the raw POST p-values. &quot;TSBH&quot; is the
two-stage Benjamini &amp; Hochberg procedure. See ?stats::p.adjust for 
descriptions of all other possible values.</p>
</td></tr>
<tr><td><code id="plot_+3A_alpha">alpha</code></td>
<td>
<p>A numeric object. A nominal type I error rate, or a vector of 
error rates, used for estimating the number of true null hypotheses in 
the two-stage Benjamini &amp; Hochberg procedure (&quot;TSBH&quot;). Default is 0.05.
Ignored if method is not TSBH.</p>
</td></tr>
<tr><td><code id="plot_+3A_subtree">subTree</code></td>
<td>
<p>A logical object. If TRUE, only the OTU used in the
analysis are included in the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called to produce graphical elements.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("POSTmData")

y &lt;- as.integer(x = metadata[,"GC"] == "BV")
X &lt;- metadata[,"mRace"]

result &lt;- post(y = y, 
               X = X, 
               OTU = otu[,1:20], 
               tree = otutree,
               cValues = seq(0,0.05,by=0.01))

plot(x = result)

</code></pre>

<hr>
<h2 id='post'>Phylogeney-Guided OTU-Specific Association Test for Microbiome Data</h2><span id='topic+post'></span>

<h3>Description</h3>

<p>The POST implements a
phylogeny-guided OTU-specific association test for microbiome data. This
method boosts the 
testing power by adaptively borrowing information from phylogenetically 
close OTUs of the target OTU. Whether or not borrowing information or the 
amount of information from the neighboring OTUs is data adaptive and 
supervised by phylogenetic distance and the outcome variable. POST is 
built on a kernel machine regression framework and inherited the 
advantages including flexibly model complex microbiome effects (e.g., 
effects from opposite direction), easily adjust for covariates, and 
accommodate both continuous and binary outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>post(
  y,
  OTU,
  tree = NULL,
  X = NULL,
  cValues = seq(from = 0, to = 0.05, by = 0.01)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="post_+3A_y">y</code></td>
<td>
<p>A numerical vector. The outcome of interest. 
Data can be binary or continuous.</p>
</td></tr>
<tr><td><code id="post_+3A_otu">OTU</code></td>
<td>
<p>A matrix object. The operational taxonomic units (OTU).
Data can be provided as counts or as proportions.
Each row indicates a single sample; each column a single OTU.
NA/0 values are allowed, but their presence will trigger a shift of all
data by a small internally defined value. 
The matrix must include column headers providing unique
identification for each OTU; these identifiers are expected to be included
in the tip labels of the input tree object. Any identifiers that are not
included as tip labels are removed from the analysis.</p>
</td></tr>
<tr><td><code id="post_+3A_tree">tree</code></td>
<td>
<p>An object of class &quot;phylo&quot;, &quot;hclust&quot;, &quot;phylog&quot;, a matrix object
or NULL. If NULL, only the single OTU test will be estimated.
Objects of class &quot;phylo&quot;, &quot;hclust&quot;, and &quot;phylog&quot; are phylogenetic trees,
the tip labels of which must include all of the
identifiers used as column headers of OTU. If a matrix, a
square symmetric matrix containing the pairwise distances between OTUs
as defined by the branch lengths. Note that the full tree should be
provided/used and should not be subset or truncated, even if OTU does
not contain all tips.
See details for further information.</p>
</td></tr>
<tr><td><code id="post_+3A_x">X</code></td>
<td>
<p>A data.frame object, matrix object or NULL. The covariates data.
If NULL, an intercept only model is assumed. Factor covariates are allowed.</p>
</td></tr>
<tr><td><code id="post_+3A_cvalues">cValues</code></td>
<td>
<p>A numeric vector. The c values at which p-values are to be
estimated. The default is a vector of evenly spaced values between
zero and the recommended maximum value for OTUs defined at
97% sequence similiary, 
c_max = 0.05.
If no tree is provided, cValues will be set to 0.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is assumed that the OTU table is defined by a 97% sequence similarity. 
Though this
threshold is not (cannot be) enforced in the implementation, the
recommended maximum c-value may not be appropriate for other thresholds.
</p>
<p>There are numerous packages available for generating phylogenetic trees.
The sole purpose of the tree input is to enable the calculation of 
pairwise distances between the pairs of tips using branch lengths.
Because it is not feasible to support this functionality for all 
packages that generate such trees, the package allows for the specification
of the distance matrix as an alternative to providing a specifically
formatted tree object. For objects of class &quot;phylo&quot;, &quot;hclust&quot;, and &quot;phylog&quot;, 
the ape package provides tools to obtain the distance matrix, and these
tools are used in this implementation. For all others, the distance
matrix (defined by branch lengths) must be provided by the user through 
input tree.
</p>


<h3>Value</h3>

<p>Returns a POST object. Analysis results are provided as
a matrix, the contents of which will depend
on the inputs selected for the analysis. Possible columns include
</p>
<table role = "presentation">
<tr><td><code>POST_pvalue</code></td>
<td>
<p>A numeric object. The POST p-value.</p>
</td></tr>
<tr><td><code>SO_pvalue</code></td>
<td>
<p>A numeric object. The single OTU test p-value.</p>
</td></tr>
<tr><td><code>BEST_C</code></td>
<td>
<p>A numeric object. The c value corresponding to the minimum
POST p-value.</p>
</td></tr>
</table>
<p>Row names indicate the OTU to which the results pertain.
Results are ordered according to the POST (if tree provided) or SO (if tree
not provided) p-values.
</p>


<h3>References</h3>

<p>Huang, C., Callahan, B., Wu, M. C., Holloway, S. T., 
Brochu, H., Lu, W., Peng, X., and Tzeng, J-Y. (2021). 
Phylogeny-guided microbiome OTU-specific association test (POST). 
Bioinformatics, under revision.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("POSTmData")

y &lt;- as.integer(x = metadata[,"GC"] == "BV")
X &lt;- metadata[,"mRace"]

result &lt;- post(y = y, 
               X = X, 
               OTU = otu[,1:20], 
               tree = otutree,
               cValues = seq(0,0.05,by=0.01))

</code></pre>

<hr>
<h2 id='POSTmData'>Example Dataset For Illustration</h2><span id='topic+POSTmData'></span><span id='topic+metadata'></span><span id='topic+otu'></span><span id='topic+otutree'></span><span id='topic+otuseq'></span>

<h3>Description</h3>

<p>This data set is provided for the purposes of illustrating the use of
the software. It was adapted from the vaginal microbiome dataset from 
Subramaniam et al. (2016). The original dataset consists of 39 
individuals with 19 bacterial vaginosis (BV) patients and 20 healthy controls.
The sequencing data and metadata are publicly available at NCBI SRA database
(PRJNA600021). Initial data processing, lead to 2711 OTUs formed at 97%
sequence similarity. FastTree (Price et al, 2010) was used to construct the 
phylogenetic tree. The OTUs data have been filitered to exclude those with 
abundance &lt; 0.005% and prevalence &lt; 10%.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(POSTmData)
</code></pre>


<h3>Format</h3>

<p>POSTmData contains 4 objects.
</p>

<dl>
<dt>metadata</dt><dd><p>a data.frame containing 30 participants. The
data.frame contains 3 columns</p>
</dd> 
</dl>

<dl>
<dt>SampleID</dt><dd><p>Character. Unique participant identifiers.</p>
</dd>
<dt>mRace</dt><dd><p>A dichotomous covariate describing the maternity race.</p>
</dd>
<dt>GC</dt><dd><p>Gestational complication indicator. &quot;BV&quot; indicates that
individual was diagnosed with bacterial vaginosis; &quot;Normal&quot;
otherwise</p>
</dd>
</dl>

<dl>
<dt>otu</dt><dd><p>a 39 x 189 matrix containing OTU count data</p>
</dd>
<dt>otutree</dt><dd><p>a phylogenetic tree constructed using all 2711 OTU sequences</p>
</dd>
<dt>otuseq</dt><dd><p>a character vector of OTU sequencing data used to construct 
the phylogenetic tree</p>
</dd>
</dl>


<hr>
<h2 id='print'>Print the Primary Results of a post() Analysis</h2><span id='topic+print'></span><span id='topic+print.POST'></span>

<h3>Description</h3>

<p>Print the primary results of a post() analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'POST'
print(x, ..., siglevel = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print_+3A_x">x</code></td>
<td>
<p>A POST object. The value object returned by a call to post()</p>
</td></tr>
<tr><td><code id="print_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods</p>
</td></tr>
<tr><td><code id="print_+3A_siglevel">siglevel</code></td>
<td>
<p>A numeric object. If &lt; 1, shows only the OTUs for which
the p-value is below the specified siglevel.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called to display key results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("POSTmData")

y &lt;- as.integer(x = metadata[,"GC"] == "BV")
X &lt;- metadata[,"mRace"]

result &lt;- post(y = y, 
               X = X, 
               OTU = otu[,1:20], 
               tree = otutree,
               cValues = seq(0,0.05,by=0.01))

print(x = result)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
