<!DOCTYPE html><html><head><title>Help for package pould</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pould}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cALD'><p>Calculation of the <code class="reqn">D'</code>, <code class="reqn">Wn</code>, and conditional Asymmetric LD Measures</p></a></li>
<li><a href='#drb1.dqb1.demo'><p>Example HLA Genotype Data for DRB1 and DQB1</p></a></li>
<li><a href='#extractLoci'><p>Extract Locus Information from Supplied Haplotype Data</p></a></li>
<li><a href='#hla.hap.demo'><p>Example Six-Locus HLA Haplotype Data in GL String Format</p></a></li>
<li><a href='#LD.heat.map'><p>Generates heat-maps for four linkage disequilibrium (LD) values (D', Wn, WLoc1/Loc2 and WLoc2/Loc1) generated for all pairs of phased and unphased two-locus haplotypes by LDWrap().</p></a></li>
<li><a href='#LD.sign.test'><p>Perform the sign test on paired LD values for phased and unphased haplotypes</p></a></li>
<li><a href='#LDWrap'><p>Parser for CSV-formatted GL String Haplotype Data</p></a></li>
<li><a href='#parseGenotypes'><p>Reformat columnnar genotype data to GL String format</p></a></li>
<li><a href='#trimAlleles'><p>Truncate allele names in haplotypes to the specified number of fields.</p></a></li>
<li><a href='#writeVector'><p>Exporting Haplotype Vectors</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Phased or Unphased Linkage Disequilibrium</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Computes the D', Wn, and conditional asymmetric linkage disequilibrium (ALD) measures for pairs of genetic loci. Performs these linkage disequilibrium (LD) calculations on phased genotype data recorded using Genotype List (GL) String or columnar formats. Alternatively, generates expectation-maximization (EM) estimated haplotypes from phased data, or performs LD calculations on EM estimated haplotypes. Performs sign tests comparing LD values for phased and unphased datasets, and generates heat-maps for each LD measure. Described by Osoegawa et al. (2019a) &lt;<a href="https://doi.org/10.1016%2Fj.humimm.2019.01.010">doi:10.1016/j.humimm.2019.01.010</a>&gt;, and Osoegawa et. al. (2019b) &lt;<a href="https://doi.org/10.1016%2Fj.humimm.2019.05.018">doi:10.1016/j.humimm.2019.05.018</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>haplo.stats,gap,stats,utils,ggplot2,reshape2,BIGDAWG,graphics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-08 17:34:11 UTC; sjmack</td>
</tr>
<tr>
<td>Author:</td>
<td>Steven Mack [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Steven Mack &lt;steven.mack@ucsf.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-16 13:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='cALD'>Calculation of the <code class="reqn">D'</code>, <code class="reqn">Wn</code>, and conditional Asymmetric LD Measures</h2><span id='topic+cALD'></span>

<h3>Description</h3>

<p>Calculates <code class="reqn">D'</code>, <code class="reqn">Wn</code> (Cramer's V) and Thomson and Single's conditional asymmetric LD (<code class="reqn">ALD</code>) measures for pairs of loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cALD(
  dataSet,
  inPhase = FALSE,
  verbose = TRUE,
  saveVector = FALSE,
  vectorName = "",
  vectorPrefix = "",
  vecDir = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cALD_+3A_dataset">dataSet</code></td>
<td>
<p>A data frame or tab delimited file consisting of four columns of genotype data named, e.g. locus1_1 locus1_2 locus2_1 locus2_2, with 1 row per sample. The columns must be organized in this exact order, but the column names should not have _1 or _2 appended; use the same locus name for each column of a given locus. For phased data, locus1_1 is in phase with locus2_1, and locus1_2 is in phase with locus_2_2. Because this funciton operates on locus pairs, any rows with missing data should be excluded from the input genotype data.</p>
</td></tr>
<tr><td><code id="cALD_+3A_inphase">inPhase</code></td>
<td>
<p>A boolean identifying the genotyping data as phased or unphased (TRUE = phased; FALSE = unphased); default is unphased.</p>
</td></tr>
<tr><td><code id="cALD_+3A_verbose">verbose</code></td>
<td>
<p>A boolean identifying if results should be printed to the console (verbose = TRUE), or returned in a vector of (<code class="reqn">D'</code>, <code class="reqn">Wn</code>, <code class="reqn">WLocus2/Locus1</code>, <code class="reqn">WLocus1/Locus2</code>, number of haplotypes) (verbose = FALSE)</p>
</td></tr>
<tr><td><code id="cALD_+3A_savevector">saveVector</code></td>
<td>
<p>A boolean identifying if the vector of all haplotypes should be exported as a text file (saveVector = TRUE), or not (saveVector = FALSE).</p>
</td></tr>
<tr><td><code id="cALD_+3A_vectorname">vectorName</code></td>
<td>
<p>A name for the exported haplotype vector file; this name is not used if saveVector = FALSE. If a name is unspecified, then a filename including the locus-pair and a timestamp is generated.</p>
</td></tr>
<tr><td><code id="cALD_+3A_vectorprefix">vectorPrefix</code></td>
<td>
<p>An optional prefix for the haplotpe vector to be used if saveVector = TRUE. This prefix will be appended, along with the phase status, before the locus name and timestamp. LDWrap() uses this parameter to identify the dataset and haplotype information passed to cALD().</p>
</td></tr>
<tr><td><code id="cALD_+3A_vecdir">vecDir</code></td>
<td>
<p>The directory into which the haplotype vector should be written if saveVector = TRUE. The default is the directory specified by tempdir().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>LD results can be directed to the console or to a data file or data frame object. This function can generate a haplotype vector file for each locus pair analyzed, and will return the LD results eiher in the console, or as a data frame object.
The implementation of ALD applied here is calculated using individual <code class="reqn">Dij</code> LD values and allele frequencies.
</p>


<h3>Value</h3>

<p>A vector of D', Wn, WLocus2/Locus1, WLocus1/Locus2 values, and the number of haplotypes evaluated
</p>


<h3>References</h3>

<p>Thomson G. &amp; Single R.M. GENETICS 2014;198(1):321-31. https://doi.org/10.1534/genetics.114.165266
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Analyze the first 10 rows of the included drb1.dqb1.demo genotype dataset
# and report LD results to the console.
cALD(drb1.dqb1.demo[1:10,])
# Alternatively, return a vector of LD results.
LDvec &lt;- cALD(drb1.dqb1.demo[1:10,],verbose=FALSE)
</code></pre>

<hr>
<h2 id='drb1.dqb1.demo'>Example HLA Genotype Data for DRB1 and DQB1</h2><span id='topic+drb1.dqb1.demo'></span>

<h3>Description</h3>

<p>A data frame of phased two-field HLA-DRB1 and HLA-DQB1 genotypes for 419 European American control subjects included
in a study of Multiple Sclerosis.
Genotype data for each locus is shown in two columns; the alleles in columns 1 and 2 are phased with the alleles in columns 3 and 4, respectively.
These genotypes were extracted from the hla.hap.demo dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(drb1.dqb1.demo)
</code></pre>


<h3>Format</h3>

<p>A data frame with 419 rows and four columns
</p>

<ul>
<li><p>DRB1: the first of two two-field HLA-DRB1 alleles
</p>
</li>
<li><p>DRB1: the second of two two-field HLA-DRB1 alleles
</p>
</li>
<li><p>DQB1: the first of two two-field HLA-DQB1 alleles
</p>
</li>
<li><p>DQB1: the second of two two-field HLA-DRB1 alleles
</p>
</li></ul>



<h3>Source</h3>

<p>immport.org study SDY1045 doi:10.21430/M3QW34U2SG
</p>


<h3>References</h3>

<p>Mack et al. Genes Immun. 2019;20(4):308-326. doi: 10.1038/s41435-017-0006-8.
</p>

<hr>
<h2 id='extractLoci'>Extract Locus Information from Supplied Haplotype Data</h2><span id='topic+extractLoci'></span>

<h3>Description</h3>

<p>This function extracts locus information from the haplotype data, and structures it for LDWrap().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractLoci(dataSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractLoci_+3A_dataset">dataSet</code></td>
<td>
<p>Data frame of two haplotypes extracted from the famData provided to LDWrap()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of two vector elements; the locus prefix (if any), e.g. &quot;HLA-&quot;, and the interleaved unsuffixed and suffixed locus names (e.g., locus, locus_1)
</p>


<h3>Note</h3>

<p>This function is for internal POULD use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
</code></pre>

<hr>
<h2 id='hla.hap.demo'>Example Six-Locus HLA Haplotype Data in GL String Format</h2><span id='topic+hla.hap.demo'></span>

<h3>Description</h3>

<p>A data frame of experimentally phased genotype data for the HLA-A, -C, -B, -DRB1, -DQB1 and DQB1 loci. These haplotypes are
for 419 unrelated European American control subjects included in a study of Multiple Sclerosis. These haplotypes were experimentally phased using the EM algorithm; low-frequency haplotypes (with counts &lt; 3) are assigned stochasically by the EM method, so the phase &quot;quality&quot; of these haplotypes should be considered lower than that for haplotypes determined via family segregation analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hla.hap.demo)
</code></pre>


<h3>Format</h3>

<p>A data frame with 419 rows and two columns
</p>

<ul>
<li><p>Relation: the genotyped individual's status in a family
</p>
</li>
<li><p>Gl String: GL String formatted multilocus phased HLA genotype
</p>
</li></ul>



<h3>Note</h3>

<p>This data is formatted as example input for the LDWrap() function, but is not actual family data. Actual family data would include 'mother', 'father' and 'child' in the &quot;Relation&quot; field. Nevertheless, including 'Subject' in this field is sufficient for analysis using LDWrap().
</p>


<h3>Source</h3>

<p>immport.org study SDY1045 doi:10.21430/M3QW34U2SG
</p>


<h3>References</h3>


<ul>
<li><p>Published study: Mack et al. Genes Immun. 2019;20(4):308-326. doi: 10.1038/s41435-017-0006-8.
</p>
</li>
<li><p>GL String format: Milius et al. Tissue Antigens. 2013;82(2):106-12. doi: 10.1111/tan.12150.
</p>
</li></ul>


<hr>
<h2 id='LD.heat.map'>Generates heat-maps for four linkage disequilibrium (LD) values (D', Wn, WLoc1/Loc2 and WLoc2/Loc1) generated for all pairs of phased and unphased two-locus haplotypes by LDWrap().</h2><span id='topic+LD.heat.map'></span>

<h3>Description</h3>

<p>This function accepts *_LD_results.csv files generated by LDWrap() as input, and generates a PNG-formatted heat-map plot file for each LD measure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LD.heat.map(
  dataName = "",
  phasedData = "",
  unphasedData = "",
  phasedLabel = "Phased",
  unphasedLabel = "EM-estimated",
  color = TRUE,
  writePlot = FALSE,
  writeDir = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LD.heat.map_+3A_dataname">dataName</code></td>
<td>
<p>The &quot;base&quot; name of the _LD_result.csv files generated by LDWrap() without the &quot;_Phased_LD_results.csv&quot; or &quot;_Unphased_LD_results.csv&quot; suffixes. See Examples, below. If both corresponding &quot;&lt;dataName&gt;_Phased_LD_results.csv&quot; and &quot;&lt;dataName&gt;_Unphased_LD_results.csv&quot; files are not found, the funciton will halt with a notification. However, if only one of those files is found in the working director, half-matrix heat map plots will be generated.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_phaseddata">phasedData</code></td>
<td>
<p>The complete name of a file of phased LD results generated by LDWrap(). Provide this filename if no base name is provided for dataName and you want to generate heat-maps for a specific set of phased LD values.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_unphaseddata">unphasedData</code></td>
<td>
<p>The complete name of a file of unphased LD results generated by LDWrap(). Provide this filename if no base name is provided for dataName and you want to generate heat-maps for a specific set of unphased LD values.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_phasedlabel">phasedLabel</code></td>
<td>
<p>The label that should appear on the heat-map plots for the upper, phased half of the plot. The default option is 'Phased'.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_unphasedlabel">unphasedLabel</code></td>
<td>
<p>The label that should appear on the heat-map plots for the lower, unphased half of the plot. The default option is 'EM-estimated'.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_color">color</code></td>
<td>
<p>A logical parameter that identifies if the heat-maps should be plotted in color (TRUE) or greyscale (FALSE). The default option is TRUE.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_writeplot">writePlot</code></td>
<td>
<p>A logical parameter that identifies if the heat-map plots should be automatically saved after they are generated. The default is 'writePlot=FALSE'.</p>
</td></tr>
<tr><td><code id="LD.heat.map_+3A_writedir">writeDir</code></td>
<td>
<p>The directory into which the heat-map plots should be saved when 'writePlot=TRUE'. The default is the directory specified by tempdir().</p>
</td></tr>
</table>


<h3>References</h3>

<p>Osoegawa et al. Hum Immunol. 2019;80(9):644 https://doi.org/10.1016/j.humimm.2019.05.018
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Using the results of LDWrap() for the first 10 rows of the drb1.dqb1.demo dataset.
# Results are saved in the temporary directory as 
# "hla-family-data_Phased_LD_results.csv" and 
# "hla-family-data_Unphased_LD_results.csv", respectively.
LDWrap(drb1.dqb1.demo[1:10,])
LDWrap(drb1.dqb1.demo[1:10,],phased=FALSE)
exampleData &lt;- paste(tempdir(),"hla-family-data",sep=.Platform$file.sep)
LD.heat.map(exampleData)
# Alternatively, these files can be sepcified individually to generate a half-matrix. 
LD.heat.map(phasedData=paste(exampleData,"_Phased_LD_results.csv",sep=""),unphasedLabel="")
# Further, two different sets of results for the same loci can be plotted; e.g., using
# phasedData="my_Phased_LD_results.csv" and unphasedData="your_Phased_LD_results.csv".
</code></pre>

<hr>
<h2 id='LD.sign.test'>Perform the sign test on paired LD values for phased and unphased haplotypes</h2><span id='topic+LD.sign.test'></span>

<h3>Description</h3>

<p>Applies binom.test() to pairs of D', Wn, WLoc1/Loc2, WLoc2/Loc1 and number of haplotypes values in the *_LD_results.csv files generated by LDWrap().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LD.sign.test(
  dataName,
  verbose = TRUE,
  returnFrame = TRUE,
  resultDir = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LD.sign.test_+3A_dataname">dataName</code></td>
<td>
<p>The &quot;base&quot; name of the _LD_result.csv files generated by LDWrap() without the &quot;_Phased_LD_results.csv&quot; or &quot;_Unphased_LD_results.csv&quot; suffixes. See Examples, below. If the corresponding &quot;&lt;dataName&gt;_Phased_LD_results.csv&quot; or &quot;&lt;dataName&gt;_Unphased_LD_results.csv&quot; files are not found, the funciton will halt with a notification.</p>
</td></tr>
<tr><td><code id="LD.sign.test_+3A_verbose">verbose</code></td>
<td>
<p>A boolean identifying if messages about function progress and results should be displayed in the console (verbose=TRUE) or not (verbose=FALSE). The default is verbose=TRUE.</p>
</td></tr>
<tr><td><code id="LD.sign.test_+3A_returnframe">returnFrame</code></td>
<td>
<p>A boolean identifying if a data frame of results should be returned (returnFrame=TRUE). If 'returnFrame=FALSE', a CSV file of results named &lt;dataName&gt;_LD-sign-test_results.csv is written in the directory specified by the 'resultDir' parameter. The default is returnFrame=TRUE.</p>
</td></tr>
<tr><td><code id="LD.sign.test_+3A_resultdir">resultDir</code></td>
<td>
<p>The directory into which the CSV file of results should be written when 'returnFrame=FALSE'. The default is the directory specified by tempdir().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns p-values for the sign test comparing the phased and unphased values of each LD measure, as well as the number of haplotypes, for each locus pair in a dataset tested using LDWrap(). It also returns the number of locus pairs for which the value in quesiton is higher in unphased haplotypes than phased haplotypes, the number of locus pairs in which the values are the same, and the total number of locus pairs asssessed. This function writes a results file to the working directory for each dataset, and will optionally display those results in the console.
</p>


<h3>Value</h3>

<p>A data frame of five columns (D', Wn, WLoc1/Loc2, WLoc2/Loc1 and N_Haplotypes) and four rows (#unphased &gt; phased, #unphased = phased, #locus pairs and p-values).
</p>


<h3>Note</h3>

<p>When verbose=TRUE, LD.sign.test() writes a table of results to the console with the column headers &quot;Measure&quot;, &quot;#U &gt; P&quot;, &quot;#U = P&quot; and &quot;p-value&quot;. Column &quot;#U &gt; P&quot; identifies the number locus pairs for which that measure for unphased haplotypes (U) was greater than that measure for phased haplotypes (P). Similarly, column &quot;#U = P&quot; identiifes the number of locus pairs for each meausure where the value of that measure was the same in phased and unphased haplotypes.
</p>
<p>Only the significance of the sign test is reported; when a significant trend is indicated, the directionality of the trend is not reported.
</p>


<h3>References</h3>

<p>Osoegawa et al. Hum Immunol. 2019;80(9):633 (https://doi.org/10.1016/j.humimm.2019.01.010)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Using LDWrap() to analyze the first 10 rows of the drb1.dqb1.demo dataset.
# LDWrap() results are saved in the temporary directory as 
# "hla-family-data_Phased_LD_results.csv" and 
# "hla-family-data_Unphased_LD_results.csv", respectively. 
LDWrap(drb1.dqb1.demo[1:10,])
LDWrap(drb1.dqb1.demo[1:10,],phased=FALSE)
exampleData &lt;- paste(tempdir(),"hla-family-data",sep=.Platform$file.sep)
LD.res &lt;- LD.sign.test(exampleData)
# Return only a data frame of results.
LD.res &lt;- LD.sign.test(exampleData,verbose=FALSE)
</code></pre>

<hr>
<h2 id='LDWrap'>Parser for CSV-formatted GL String Haplotype Data</h2><span id='topic+LDWrap'></span>

<h3>Description</h3>

<p>A wrapper for parsing phased haplotype data recorded in GL String format. Extracts all pairs of loci from GL String formatted haplotypes or column formatted genotypes, passes paired-genotype data to the cALD function, and generates files consumed by the LD.sign.test() and LD.heat.map() functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDWrap(
  famData,
  threshold = 10,
  phased = TRUE,
  frameName = "hla-family-data",
  trunc = 0,
  writeTo = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDWrap_+3A_famdata">famData</code></td>
<td>
<p>A data frame or CSV formatted file (with a .csv filename suffix) that contains the two columns named &quot;Gl String&quot; and &quot;Relation&quot;. Other columns can be included (in any order), but will not impact the analysis. The Relation column can contain any data; however anything other than &quot;Relation=child&quot; will be included in the LD analyses. The Gl String column should consist of two tilde (~) delimited haplotypes conneced by a plus (+) sign (GL String format). Allele names should be recorded using the LOCUS*VARIANT structure used for HLA and KIR alleles. A locus prefix (e.g., 'HLA-') is not required, but if a locus prefix is included, all allele names must include the same locus prefix. Alternatively, LDWrap() will consume genotype data in a data frame or headered tab-delimited text file (TXT or TSV), with two columns per locus. See the parseGenotype() documentation for additional requirements. The name of the file provided will serve as the basis for the name of the LD result files.</p>
</td></tr>
<tr><td><code id="LDWrap_+3A_threshold">threshold</code></td>
<td>
<p>An integer that specifies the minimnum number of subjects allowed for the analysis of a locus-pair. The default value is 10. If the number of subjects with haplotypes for a locus pair is less than the threshold, the *_LD_results.csv file will contain 'Not Calculated' 'Subject Threshold=##' 'Complete subjects=#' '.' in columns 2-5 for that locus pair, where ## is the set threshold and # is the number of subjects. Column 6 will be empty.</p>
</td></tr>
<tr><td><code id="LDWrap_+3A_phased">phased</code></td>
<td>
<p>A boolean that determines if the LD calculations should be performed for phased data (TRUE) or unphased data (FALSE). If phased=FALSE, the EM algorithm is used to estimate haplotypes for the data in the Gl String column of family haplotype datasets.</p>
</td></tr>
<tr><td><code id="LDWrap_+3A_framename">frameName</code></td>
<td>
<p>A descriptor for the data frame of family data provided. The default value is &quot;hla-family-data&quot;. This value is not used if a CSV file is provided.</p>
</td></tr>
<tr><td><code id="LDWrap_+3A_trunc">trunc</code></td>
<td>
<p>An integer that specifies the number of fields to which colon-delimited allele names in famdData should be truncated. The default value of 0 indicates no truncation. A value higher than the number of fields in the supplied allele data will result in no truncation. When a positive value of trunc is provided, the names of the output files will include the specified truncation level.</p>
</td></tr>
<tr><td><code id="LDWrap_+3A_writeto">writeTo</code></td>
<td>
<p>The directory into which the LDWrap() output files should be written. The default is the directory specified by tempdir().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function coerces cALD() to generate a haplotype vector file for each locus pair analyzed, and generates a single LD results file containing LD values for all locus pairs, along with the number of haplotypes tested, one locus pair per row. The LD results file will contain six columns (&quot;Loc1~Loc2&quot;,&quot;D'&quot;,&quot;Wn&quot;,&quot;WLoc1/Loc2&quot;,&quot;WLoc2/Loc1&quot;,&quot;N_Haplotypes&quot;), and will be named &quot;&lt;filename prefix&gt;_&lt;Phased/Unphased&gt;_LD_results.csv&quot;.
</p>


<h3>Note</h3>

<p>When at least one locus in a locus pair is monomorphic, no LD calculations will be performed, and column 5 of the results for that locus pair will identify the monomorphic loci.
</p>
<p>This function does not validate HLA allele names. Unusual allele names (e.g., 'HLA-A*NULL', 'HLA-DRB1*NoMatch', 'HLA-DPB1*NT') and truncated versions of allele names (e.g., 'HLA-A*01', 'HLA-A*01:01', 'HLA-A*01:01:01', etc.) will be analyzed as distinct alleles. Including unusual allele names or different truncated versions of the same allele name in a dataset will likely skew the analytic results. In the latter case, the trunc parameter can be used to specify analysis at a specific number of fields.
</p>
<p>Column-formatted genotype data are generally unphased; unless genotype data have been structured so that all alleles in the first column for each locus are in one haplotype, and all of the alleles in the second column in each locus are in the other haplotype, phased should be set to FALSE for column-formatted genotype datasets.
</p>


<h3>References</h3>

<p>Osoegawa et al. Hum Immunol. 2019;80(9):633 (https://doi.org/10.1016/j.humimm.2019.01.010)
</p>
<p>Osoegawa et al. Hum Immunol. 2019;80(9):644 (https://doi.org/10.1016/j.humimm.2019.05.018)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Analyze the first 10 rows of the drb1.dqb1.demo genotype dataset.
LDWrap(drb1.dqb1.demo[1:10,],frameName="DRDQDemo")
# Analyze the includeed example genotype data with all alleles truncated to one field.
LDWrap(drb1.dqb1.demo[1:10,],frameName="DRDQDemoTrunc",trunc=1)
</code></pre>

<hr>
<h2 id='parseGenotypes'>Reformat columnnar genotype data to GL String format</h2><span id='topic+parseGenotypes'></span>

<h3>Description</h3>

<p>This function accepts genotype data organized in locus-column pairs, and returns GL String-formatted data structured for LDWrap(). Of the resulting multilocus haplotype pair, the first haplotype is constructed from the first column for each locus, and the second haplotype is constructed from the second column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseGenotypes(dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parseGenotypes_+3A_dataset">dataset</code></td>
<td>
<p>A tab-delimited text file (with a .txt or .tsv filename suffix) with a header row or a data frame. Each row corresponds to a subject, with two columns per locus. Allele names can include a locus name (e.g., locus*allele) or can can exclude the locus, but all allele names in the dataset must either include or exclude the locus. Missing (untyped) allele data can be identified with an empty cell or a set of four asterisks in files, and with NA values in data frames. Column names for each locus pair must be adjacent, but can be either identical (e.g., &quot;locus&quot; and &quot;locus&quot;), or suffixed (e.g., &quot;locus_1&quot; and &quot;locus_2&quot;, where &quot;locus_1&quot; always precedes &quot;locus_2&quot;). A optional column of sample identifiers can be included, but must be named &quot;SampleID&quot;.  A column named &quot;Disease&quot; can be included, but will be ignored. No other non-locus columns are permitted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns. The &quot;Relation&quot; column includes sample identifiers if provided, or numbers from 1 to the number of subjects. The &quot;GL String&quot; column contains the GL String formatted genotypes.
</p>


<h3>Note</h3>

<p>This function is for internal POULD use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
</code></pre>

<hr>
<h2 id='trimAlleles'>Truncate allele names in haplotypes to the specified number of fields.</h2><span id='topic+trimAlleles'></span>

<h3>Description</h3>

<p>This function accepts a dataframe of tilde-delimited haplotypes and trims colon-delimited names to the number of fields specified by 'reso'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trimAlleles(haplotypes, reso)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trimAlleles_+3A_haplotypes">haplotypes</code></td>
<td>
<p>Data frame of tilde-delimited haplotypes extracted from the famData provided to LDWrap()</p>
</td></tr>
<tr><td><code id="trimAlleles_+3A_reso">reso</code></td>
<td>
<p>An integer that specifies the number of fields to which colon-delimited allele names in famData should be truncated. The default value of 0 indicates no truncation. A value higher than the number of fields in the supplied allele data will result in no truncation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two sets of tilde-delimited haplotypes.
</p>


<h3>Note</h3>

<p>This function is for internal POULD use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
</code></pre>

<hr>
<h2 id='writeVector'>Exporting Haplotype Vectors</h2><span id='topic+writeVector'></span>

<h3>Description</h3>

<p>This function writes the haplotype vector and any accessory information to a text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeVector(nLoc1, Res, hapVec, numSamp, writeName, genPhase, Prefix, vDir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeVector_+3A_nloc1">nLoc1</code></td>
<td>
<p>Number of alleles at locus 1</p>
</td></tr>
<tr><td><code id="writeVector_+3A_res">Res</code></td>
<td>
<p>Data frame consisting of Locus Number, Locus Name, Allele Name, Allele Count, Allele Frequency and Frequency^2</p>
</td></tr>
<tr><td><code id="writeVector_+3A_hapvec">hapVec</code></td>
<td>
<p>Haplotype vector</p>
</td></tr>
<tr><td><code id="writeVector_+3A_numsamp">numSamp</code></td>
<td>
<p>The dataset's 2N value</p>
</td></tr>
<tr><td><code id="writeVector_+3A_writename">writeName</code></td>
<td>
<p>The specified name for the vector</p>
</td></tr>
<tr><td><code id="writeVector_+3A_genphase">genPhase</code></td>
<td>
<p>Boolean describing phased (TRUE) or unphased (FALSE) analysis</p>
</td></tr>
<tr><td><code id="writeVector_+3A_prefix">Prefix</code></td>
<td>
<p>The specified prefix for the written vector file</p>
</td></tr>
<tr><td><code id="writeVector_+3A_vdir">vDir</code></td>
<td>
<p>The directory into which the vector file should be written.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal POULD use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
