<!DOCTYPE html><html lang="en"><head><title>Help for package PoweREST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PoweREST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#fit_powerest'><p>Fit the power surface</p></a></li>
<li><a href='#fit_XGBoost'><p>Fit with XGBoost</p></a></li>
<li><a href='#plotly_powerest'><p>3D interactive visualization</p></a></li>
<li><a href='#power_example'><p>An example of power results with multiple replicates number</p></a></li>
<li><a href='#PoweREST'><p>Bootstrap resampling and power calculation upon ST data</p></a></li>
<li><a href='#PoweREST_gene'><p>Bootstrap resampling and power estimation for one single gene</p></a></li>
<li><a href='#PoweREST_subset'><p>Bootstrap resampling and power calculation for a subset of genes</p></a></li>
<li><a href='#pred_powerest'><p>Power value prediction</p></a></li>
<li><a href='#pred_XGBoost'><p>Prediction results from XGBoost</p></a></li>
<li><a href='#result_example'><p>An example of power results from PoweREST</p></a></li>
<li><a href='#vis_powerest'><p>Visualization of the power surface</p></a></li>
<li><a href='#vis_XGBoost'><p>Visualization of the power estimations from XGBoost</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A Bootstrap-Based Power Estimation Tool for Spatial
Transcriptomics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>scam, Seurat, dplyr, plotly, resample, xgboost, magrittr,
rayshader, ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>patchwork, boot, knitr, rmarkdown, fields, rayrender, tidyr</td>
</tr>
<tr>
<td>Description:</td>
<td>Power estimation and sample size calculation for 10X Visium Spatial Transcriptomics data to detect differential expressed genes between two conditions based on bootstrap resampling. See Shui et al. (2024) &lt;<a href="https://doi.org/10.1101%2F2024.08.30.610564">doi:10.1101/2024.08.30.610564</a>&gt; for method details.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-04 13:17:57 UTC; shuilan</td>
</tr>
<tr>
<td>Author:</td>
<td>Lan Shui <a href="https://orcid.org/0000-0002-2719-842X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lan Shui &lt;Lan.Shui@uth.tmc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-09 09:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='fit_powerest'>Fit the power surface</h2><span id='topic+fit_powerest'></span>

<h3>Description</h3>

<p>This function loads the power values with corresponding avg_log2FC and avg_PCT derived from bootstrap sampling
and utilizes the <a href="https://CRAN.R-project.org/package=scam">scam</a> package to fit two dimensional
smoothing splines under monotone constraints: 1.positive relationship between power and avg_log2FC;
2.positive relationship between power and avg_PCT. The values of avg_log2FC and avg_PCT can be either
from the averages of the bootstrap samples or from the original spatial transcriptomics data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_powerest(power,avg_log2FC,avg_PCT,filter_zero=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_powerest_+3A_power">power</code></td>
<td>
<p>The raw power values.</p>
</td></tr>
<tr><td><code id="fit_powerest_+3A_avg_log2fc">avg_log2FC</code></td>
<td>
<p>The corresponding log2FC values.</p>
</td></tr>
<tr><td><code id="fit_powerest_+3A_avg_pct">avg_PCT</code></td>
<td>
<p>The corresponding PCT values.</p>
</td></tr>
<tr><td><code id="fit_powerest_+3A_filter_zero">filter_zero</code></td>
<td>
<p>Whether the user would like to filter to remove the power values being 0, default=TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'scam' object is the result of scam function. More information about the content of a 'scam' object can be found
at the document of R package <a href="https://CRAN.R-project.org/package=scam">scam</a>.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(result_example)
 b&lt;-fit_powerest(result_example$power,result_example$avg_logFC,result_example$avg_PCT)
</code></pre>

<hr>
<h2 id='fit_XGBoost'>Fit with XGBoost</h2><span id='topic+fit_XGBoost'></span>

<h3>Description</h3>

<p>This function estimates the power values based on XGBoost under 3-dimensional monotone constraints upon avg_log2FC, avg_PCT
and replicates. This function is recommended when there exist crossings between power surfaces fitted by 'fit_powerest' and
used for estimating local power values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_XGBoost(power,avg_log2FC,avg_PCT,replicates,filter_zero=TRUE,
max.depth=6,eta=0.3,nround=100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_XGBoost_+3A_power">power</code></td>
<td>
<p>The raw power values.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_avg_log2fc">avg_log2FC</code></td>
<td>
<p>The corresponding log2FC values.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_avg_pct">avg_PCT</code></td>
<td>
<p>The corresponding PCT values.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_replicates">replicates</code></td>
<td>
<p>The corresponding replicates number.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_filter_zero">filter_zero</code></td>
<td>
<p>Whether the user would like to filter to remove the power values being 0. Default=TRUE.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_max.depth">max.depth</code></td>
<td>
<p>Maximum depth of a tree. Default=6.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_eta">eta</code></td>
<td>
<p>control the learning rate: scale the contribution of each tree by a factor of 0 &lt; eta &lt; 1 when it is added to the current approximation. Used to prevent overfitting by making the boosting process more conservative. Default=0.3.</p>
</td></tr>
<tr><td><code id="fit_XGBoost_+3A_nround">nround</code></td>
<td>
<p>Max number of boosting iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A object of class 'xgb.Booster'. More information about the content of a 'xgb.Booster' object can be found
at the document of R package <a href="https://CRAN.R-project.org/package=xgboost">xgboost</a>.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(power_example)
# Fit the local power surface of avg_log2FC_abs between 1 and 2
avg_log2FC_abs_1_2&lt;-dplyr::filter(power_example,avg_log2FC_abs&gt;1 &amp; avg_log2FC_abs&lt;2)
# Fit the model
bst&lt;-fit_XGBoost(power_example$power,avg_log2FC=power_example$avg_log2FC_abs,
avg_PCT=power_example$mean_pct,replicates=power_example$sample_size)

</code></pre>

<hr>
<h2 id='plotly_powerest'>3D interactive visualization</h2><span id='topic+plotly_powerest'></span>

<h3>Description</h3>

<p>This function creates 3d interactive plot of the power against other parameters based on 'plot_ly'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotly_powerest(pred,opacity=0.8,colors='BrBG',fig_title=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotly_powerest_+3A_pred">pred</code></td>
<td>
<p>The result from 'pred_powerest'.</p>
</td></tr>
<tr><td><code id="plotly_powerest_+3A_opacity">opacity</code></td>
<td>
<p>The opacity of the graph, default=0.8.</p>
</td></tr>
<tr><td><code id="plotly_powerest_+3A_colors">colors</code></td>
<td>
<p>The color for the graph, default='BrBG'.</p>
</td></tr>
<tr><td><code id="plotly_powerest_+3A_fig_title">fig_title</code></td>
<td>
<p>The title of the graph, default=NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 3d interactive plot of the power surface. Users can also plot multiple surfaces together to compare them.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(result_example)
 b&lt;-fit_powerest(result_example$power,result_example$avg_logFC,result_example$avg_PCT)
 pred &lt;- pred_powerest(b,xlim= c(0,6),ylim=c(0,1))
 plotly_powerest(pred,fig_title='Power estimation result')

</code></pre>

<hr>
<h2 id='power_example'>An example of power results with multiple replicates number</h2><span id='topic+power_example'></span>

<h3>Description</h3>

<p>A subset of power results with multiple
replicates number from PoweREST
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_example
</code></pre>


<h3>Format</h3>



<h4><code>power_example</code></h4>

<p>A data frame with 844 rows and 5 columns:
</p>

<dl>
<dt>avg_logFC</dt><dd><p>average log2FC</p>
</dd>
<dt>mean_PCT</dt><dd><p>percentage of spots detecting the gene</p>
</dd>
<dt>sample_size</dt><dd><p>number of replicates</p>
</dd>
<dt>power</dt><dd><p>power values</p>
</dd>
<dt>avg_log2FC_abs</dt><dd><p>the absolute value of average log2FC</p>
</dd>
</dl>



<hr>
<h2 id='PoweREST'>Bootstrap resampling and power calculation upon ST data</h2><span id='topic+PoweREST'></span>

<h3>Description</h3>

<p>This function performs bootstrap resampling upon a Seurat subject under each condition
to resemble the real dataset which allows the exact power calculation, and perform DE analysis.
Users can specify the test they would like to perform for the DE analysis in '...' which should not contain
min.pct and logfc.threshold or other parameters attempt to pre-filter genes, as we specify min.pct and logfc.threshold as 0s
to calculate power for all the genes available. Therefore it may take one night to run if the ST data owns over thousands of genes.
To speed up this process, one may want to try function 'PoweREST_subset' where the pre-filter of genes are included in this process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PoweREST(Seurat_obj,cond,replicates=1,spots_num,
iteration=100,random_seed=1,pvalue=0.05,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PoweREST_+3A_seurat_obj">Seurat_obj</code></td>
<td>
<p>A <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a> object.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_cond">cond</code></td>
<td>
<p>The name of the variable that indicates different conditions which is also stored in the
meta.data of the Seurat_obj and should be in character type.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_replicates">replicates</code></td>
<td>
<p>The number of sample replicates per group.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_spots_num">spots_num</code></td>
<td>
<p>The number of spots per replicate.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_iteration">iteration</code></td>
<td>
<p>The number of iterations of the resampling.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_random_seed">random_seed</code></td>
<td>
<p>To set a random seed.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_pvalue">pvalue</code></td>
<td>
<p>The pvalue that will be considered significant.</p>
</td></tr>
<tr><td><code id="PoweREST_+3A_...">...</code></td>
<td>
<p>DE test to use other than the default Wilcoxon test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of values containing the power, average log2FC and percentage of spots detecting the gene among
the resampling data, the replicate value and the spots number per slice specified by the user and corresponding genes' name.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>

<hr>
<h2 id='PoweREST_gene'>Bootstrap resampling and power estimation for one single gene</h2><span id='topic+PoweREST_gene'></span>

<h3>Description</h3>

<p>This function performs bootstrap resampling upon a Seurat subject under each condition
to resemble the real dataset which allows the exact power calculation, and perform DE
analysis upon one gene specified by the user. Users can specify the test they would like
to perform for the DE analysis in '...'. Note that the results are not multiple testing
corrected, therefore should be interpreted carefully.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PoweREST_gene(Seurat_obj,cond,replicates=1,spots_num,
gene_name,iteration=100,random_seed=1,pvalue=0.05,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PoweREST_gene_+3A_seurat_obj">Seurat_obj</code></td>
<td>
<p>A <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a> object.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_cond">cond</code></td>
<td>
<p>The name of the variable that indicates different conditions which is also stored in the
meta.data of the Seurat_obj and should be in character type.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_replicates">replicates</code></td>
<td>
<p>The number of sample replicates per group.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_spots_num">spots_num</code></td>
<td>
<p>The number of spots per replicate.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_gene_name">gene_name</code></td>
<td>
<p>Specify the name of gene for power calculation.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_iteration">iteration</code></td>
<td>
<p>The number of iterations of the resampling.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_random_seed">random_seed</code></td>
<td>
<p>To set a random seed.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_pvalue">pvalue</code></td>
<td>
<p>The pvalue that will be considered significant.</p>
</td></tr>
<tr><td><code id="PoweREST_gene_+3A_...">...</code></td>
<td>
<p>DE Test to use other than the default Wilcoxon test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of values containing the power, average log2FC and percentage of spots detecting the gene among
the resampling data, the replicate value and the spots number per slice specified by the user and corresponding gene's name.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>

<hr>
<h2 id='PoweREST_subset'>Bootstrap resampling and power calculation for a subset of genes</h2><span id='topic+PoweREST_subset'></span>

<h3>Description</h3>

<p>This function performs bootstrap resampling upon a Seurat subject under each condition
to resemble the real dataset which allows the exact power calculation, and perform DE analysis.
Similar to 'PoweREST', users can specify the test they would like to perform for the DE analysis
in '...' (more test options can be refered to <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a>.
Different to 'PoweREST', users can specify the values of 'min.pct' and 'logfc.threshold'
to pre-filter the genes based on their minimum detection rate 'min.pct' and at least X-fold difference (log-scale)
('logfc.threshold') across both groups. But this kind of filtering can miss weaker signals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PoweREST_subset(Seurat_obj,cond,replicates=1,spots_num,
iteration=100,random_seed=1,pvalue=0.05,logfc.threshold = 0.1,
min.pct = 0.01,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PoweREST_subset_+3A_seurat_obj">Seurat_obj</code></td>
<td>
<p>A <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a> object.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_cond">cond</code></td>
<td>
<p>The name of the variable that indicates different conditions which is also stored in the
meta.data of the Seurat_obj and should be in character type.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_replicates">replicates</code></td>
<td>
<p>The number of sample replicates per group.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_spots_num">spots_num</code></td>
<td>
<p>The number of spots per replicate.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_iteration">iteration</code></td>
<td>
<p>The number of iterations of the resampling.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_random_seed">random_seed</code></td>
<td>
<p>To set a random seed.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_pvalue">pvalue</code></td>
<td>
<p>The pvalue that will be considered significant.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_logfc.threshold">logfc.threshold</code></td>
<td>
<p>For every resampling, limit testing to genes which show, on average, at least X-fold difference (log-scale) between the two groups.
Default is 0.1 Increasing logfc.threshold speeds up the function, but can miss weaker signals.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_min.pct">min.pct</code></td>
<td>
<p>For every resampling, only test genes that are detected in a minimum fraction of min.pct spots in either of the two populations.
Meant to speed up the function by not testing genes that are very infrequently expressed. Default is 0.01.</p>
</td></tr>
<tr><td><code id="PoweREST_subset_+3A_...">...</code></td>
<td>
<p>DE test to use other than the default Wilcoxon test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of values containing the power, average log2FC and percentage of spots detecting the gene among
the resampling data, the replicate value and the spots number per slice specified by the user and the filtered.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>

<hr>
<h2 id='pred_powerest'>Power value prediction</h2><span id='topic+pred_powerest'></span>

<h3>Description</h3>

<p>This function provides the prediction from the <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a>
object which could be used for visualization by 'plotly_powerest' and 'vis_powerest' or
the power result for your proposal or research. And it is a modified version of the scam library code predict.scam.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_powerest(x,n.grid=30,xlim=NULL,ylim=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_powerest_+3A_x">x</code></td>
<td>
<p>A <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a> object.</p>
</td></tr>
<tr><td><code id="pred_powerest_+3A_n.grid">n.grid</code></td>
<td>
<p>The grid note number within 'xlim' and 'ylim', default=30.</p>
</td></tr>
<tr><td><code id="pred_powerest_+3A_xlim">xlim</code></td>
<td>
<p>The range of the absolute value of log2FC used for prediction, default=NULL which means the original range.</p>
</td></tr>
<tr><td><code id="pred_powerest_+3A_ylim">ylim</code></td>
<td>
<p>The range of the avg_pct used for prediction, default=NULL which means the original range.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The prediction values of the power.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a> based partly on 'scam' by Natalya Pya
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(result_example)
 b&lt;-fit_powerest(result_example$power,result_example$avg_logFC,result_example$avg_PCT)
 pred &lt;- pred_powerest(b,xlim= c(0,6),ylim=c(0,1))

</code></pre>

<hr>
<h2 id='pred_XGBoost'>Prediction results from XGBoost</h2><span id='topic+pred_XGBoost'></span>

<h3>Description</h3>

<p>This function takes the result from 'fit_XGBoost' and make predictions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_XGBoost(x,n.grid=30,xlim,ylim,replicates)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_XGBoost_+3A_x">x</code></td>
<td>
<p>A object of class 'xgb.Booster'.</p>
</td></tr>
<tr><td><code id="pred_XGBoost_+3A_n.grid">n.grid</code></td>
<td>
<p>The grid note number within 'xlim' and 'ylim', default=30.</p>
</td></tr>
<tr><td><code id="pred_XGBoost_+3A_xlim">xlim</code></td>
<td>
<p>The range of the absolute value of avg_log2FC used for prediction.</p>
</td></tr>
<tr><td><code id="pred_XGBoost_+3A_ylim">ylim</code></td>
<td>
<p>The range of the avg_pct used for prediction.</p>
</td></tr>
<tr><td><code id="pred_XGBoost_+3A_replicates">replicates</code></td>
<td>
<p>The replicates number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The power estimations from XGBoost.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(power_example)
# Fit the local power surface of avg_log2FC_abs between 1 and 2
avg_log2FC_abs_1_2&lt;-dplyr::filter(power_example,avg_log2FC_abs&gt;1 &amp; avg_log2FC_abs&lt;2)
# Fit the model
bst&lt;-fit_XGBoost(power_example$power,avg_log2FC=power_example$avg_log2FC_abs,
avg_PCT=power_example$mean_pct,replicates=power_example$sample_size)
pred&lt;-pred_XGBoost(bst,n.grid=30,xlim=c(0,1.5),ylim=c(0,0.1),replicates=3)

</code></pre>

<hr>
<h2 id='result_example'>An example of power results from PoweREST</h2><span id='topic+result_example'></span>

<h3>Description</h3>

<p>A subset of power results from PoweREST
by running PoweREST(Peri,cond='Condition',
replicates=5,spots_num=80,iteration=2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>result_example
</code></pre>


<h3>Format</h3>



<h4><code>result_example</code></h4>

<p>A data frame with ~20,000 rows and 3 columns:
</p>

<dl>
<dt>power</dt><dd><p>power values</p>
</dd>
<dt>avg_logFC</dt><dd><p>average log2FC</p>
</dd>
<dt>avg_PCT</dt><dd><p>percentage of spots detecting the gene</p>
</dd>
</dl>



<hr>
<h2 id='vis_powerest'>Visualization of the power surface</h2><span id='topic+vis_powerest'></span>

<h3>Description</h3>

<p>This function takes the result from 'pred_powerest' and plots 2D views of it,
supply ticktype=&quot;detailed&quot; to get proper axis annotation and is a modified version of the 'scam' library code 'vis.scam'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_powerest(x,color="heat",contour.col=NULL,
se=-1,zlim=NULL,n.grid=30,col=NA,plot.type="persp",
nCol=50,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vis_powerest_+3A_x">x</code></td>
<td>
<p>A <a href="https://CRAN.R-project.org/package=scam">scam</a> object.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_color">color</code></td>
<td>
<p>The color of the plot which can be one of the &quot;heat&quot;, &quot;topo&quot;, &quot;cm&quot;, &quot;terrain&quot;, &quot;gray&quot; or &quot;bw&quot;.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_contour.col">contour.col</code></td>
<td>
<p>The color of the contour plot when using plot.type=&quot;contour&quot;.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_se">se</code></td>
<td>
<p>If less than or equal to zero then only the predicted surface is plotted, but if
greater than zero, then 3 surfaces are plotted, one at the predicted values minus se standard errors,
one at the predicted values and one at the predicted values plus se standard errors.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_zlim">zlim</code></td>
<td>
<p>The range of power value the user want to show.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_n.grid">n.grid</code></td>
<td>
<p>The number of grid nodes in each direction used for calculating the plotted surface.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_col">col</code></td>
<td>
<p>The colors for the facets of the plot. If this is NA then if se&gt;0 the facets are transparent,
otherwise the color scheme specified in color is used. If col is not NA then it is used as the facet color.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_plot.type">plot.type</code></td>
<td>
<p>One of &quot;contour&quot; or &quot;persp&quot;.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_ncol">nCol</code></td>
<td>
<p>The number of colors to use in color schemes.</p>
</td></tr>
<tr><td><code id="vis_powerest_+3A_...">...</code></td>
<td>
<p>Other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2d plot of the power surface. More details can be seen at <a href="https://CRAN.R-project.org/package=scam">scam</a>.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a> based partly on 'scam' by Natalya Pya
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(result_example)
 b&lt;-fit_powerest(result_example$power,result_example$avg_logFC,result_example$avg_PCT)
 pred &lt;- pred_powerest(b,xlim= c(0,6),ylim=c(0,1))
 vis_powerest(pred,theta=-30,phi=30,color='heat',ticktype = "detailed",xlim=c(0,6),nticks=5)

</code></pre>

<hr>
<h2 id='vis_XGBoost'>Visualization of the power estimations from XGBoost</h2><span id='topic+vis_XGBoost'></span>

<h3>Description</h3>

<p>This function takes the result from 'pred_XGboost' and plots 2D/3D views of it,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_XGBoost(x,view='2D',legend_name='Power',
xlab='avg_log2FC_abs',ylab='mean_pct')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vis_XGBoost_+3A_x">x</code></td>
<td>
<p>The result dataframe from 'pred_XGboost'.</p>
</td></tr>
<tr><td><code id="vis_XGBoost_+3A_view">view</code></td>
<td>
<p>determines plot 2D/3D view, default='2D'.</p>
</td></tr>
<tr><td><code id="vis_XGBoost_+3A_legend_name">legend_name</code></td>
<td>
<p>The name of legend, default='Power'.</p>
</td></tr>
<tr><td><code id="vis_XGBoost_+3A_xlab">xlab</code></td>
<td>
<p>The name of xlab, default='avg_log2FC_abs'.</p>
</td></tr>
<tr><td><code id="vis_XGBoost_+3A_ylab">ylab</code></td>
<td>
<p>The name of ylab, default='mean_pct'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2D/3D plot of the power results from XGBoost.
</p>


<h3>Author(s)</h3>

<p>Lan Shui <a href="mailto:lshui@mdanderson.org">lshui@mdanderson.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(power_example)
# Fit the local power surface of avg_log2FC_abs between 1 and 2
avg_log2FC_abs_1_2&lt;-dplyr::filter(power_example,avg_log2FC_abs&gt;1 &amp; avg_log2FC_abs&lt;2)
# Fit the model
bst&lt;-fit_XGBoost(power_example$power,avg_log2FC=power_example$avg_log2FC_abs,
avg_PCT=power_example$mean_pct,replicates=power_example$sample_size)
pred&lt;-pred_XGBoost(bst,n.grid=30,xlim=c(0,1.5),ylim=c(0,0.1),replicates=3)
vis_XGBoost(pred,view='2D',legend_name='Power',xlab='avg_log2FC_abs',ylab='mean_pct')


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
