<!DOCTYPE html><html><head><title>Help for package powerpkg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {powerpkg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fnfam'><p> Internal function called by ntdt</p></a></li>
<li><a href='#fx'><p> Internal function for the ptdt function</p></a></li>
<li><a href='#normplot'><p> Power plotting demonstration</p></a></li>
<li><a href='#nsibs'><p> A function to compute power of an affected sib pair design.</p></a></li>
<li><a href='#ntdt'><p> Computes number of trios needed for a TDT study.</p></a></li>
<li><a href='#ntdt.q'><p> Computes power of a TDT study as a function of the susceptibility</p>
allele frequency q.</a></li>
<li><a href='#plotDom'><p> plotDom creates power graphs for an affected sibpair linkage study.</p></a></li>
<li><a href='#plotNsibs'>
<p>plotNsibs: Create power plots for an affected sib pair linkage</p>
study</a></li>
<li><a href='#plotNtdt'><p> A function for plotting the results from ntdt.q</p></a></li>
<li><a href='#ptb'><p> A power table for the TDT design</p></a></li>
<li><a href='#ptdt'><p> Power for a TDT design</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Power Analyses for the Affected Sib Pair and the TDT Design</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel E. Weeks</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel E. Weeks &lt;weeks@pitt.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>There are two main functions: (1) To estimate the power of testing for linkage using an
        affected sib pair design, as a function of the recurrence risk
        ratios. We will use analytical power formulae as implemented in
        R. These are based on a Mathematica notebook created by Martin
        Farrall. (2) To examine how the power of the transmission
        disequilibrium test (TDT) depends on the disease allele
        frequency, the marker allele frequency, the strength of the
        linkage disequilibrium, and the magnitude of the genetic
        effect. We will use an R program that implements the power
        formulae of Abel and Muller-Myhsok (1998). These formulae allow
        one to quickly compute power of the TDT approach under a
        variety of different conditions. This R program was modeled on
        Martin Farrall's Mathematica notebook.</td>
</tr>
<tr>
<td>Depends:</td>
<td>tcltk</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-18 22:52:28 UTC; dweeks</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-19 06:50:02 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='fnfam'> Internal function called by ntdt </h2><span id='topic+fnfam'></span>

<h3>Description</h3>

<p>An internal function called by the 'ntdt' function based on formulae from Abel and Muller-Myhsok (1998)
Am J Hum Genet 63:664-667
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fnfam(n, power, alpha, p1, h)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fnfam_+3A_n">n</code></td>
<td>
<p> number of families. </p>
</td></tr>
<tr><td><code id="fnfam_+3A_power">power</code></td>
<td>
<p> the power of the test. </p>
</td></tr>
<tr><td><code id="fnfam_+3A_alpha">alpha</code></td>
<td>
<p> the desired significance level. </p>
</td></tr>
<tr><td><code id="fnfam_+3A_p1">p1</code></td>
<td>
<p> defined by Abel and Muller-Myhsok. </p>
</td></tr>
<tr><td><code id="fnfam_+3A_h">h</code></td>
<td>
<p> defined by Abel and Muller-Myhsok. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Abel and Muller-Myhsok
</p>


<h3>Value</h3>

<table>
<tr><td><code>n</code></td>
<td>
<p> the number of tdt trios needed for the desired power.</p>
</td></tr>
</table>


<h3>Note</h3>

<p> These are based on a Mathematica notebook created by Martin Farrall. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNtdt">plotNtdt</a></code>, <code><a href="#topic+ntdt">ntdt</a></code>, <code><a href="#topic+ntdt.q">ntdt.q</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## NONE: Internal function
</code></pre>

<hr>
<h2 id='fx'> Internal function for the ptdt function </h2><span id='topic+fx'></span>

<h3>Description</h3>

<p>Internal function for the ptdt function, based on formulae from Abel and Muller-Myhsok (1998)
Am J Hum Genet 63:664-667
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fx(bta, n, alpha, p1, s)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fx_+3A_bta">bta</code></td>
<td>
<p> defined by Abel and Muller-Myhsok.  </p>
</td></tr>
<tr><td><code id="fx_+3A_n">n</code></td>
<td>
<p> defined by Abel and Muller-Myhsok.  </p>
</td></tr>
<tr><td><code id="fx_+3A_alpha">alpha</code></td>
<td>
<p> the significance level of the test. </p>
</td></tr>
<tr><td><code id="fx_+3A_p1">p1</code></td>
<td>
<p> defined by Abel and Muller-Myhsok.  </p>
</td></tr>
<tr><td><code id="fx_+3A_s">s</code></td>
<td>
<p> defined by Abel and Muller-Myhsok.  </p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Abel and Muller-Myhsok
</p>


<h3>Value</h3>

<table>
<tr><td><code>f</code></td>
<td>
<p> See Abel and Muller-Myhsok </p>
</td></tr>
</table>


<h3>Note</h3>

<p> These are based on a Mathematica notebook created by Martin Farrall. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNtdt">plotNtdt</a></code>, <code><a href="#topic+ptdt">ptdt</a></code>, <code><a href="#topic+ntdt">ntdt</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## NONE: Internal function
</code></pre>

<hr>
<h2 id='normplot'> Power plotting demonstration </h2><span id='topic+normplot'></span>

<h3>Description</h3>

<p> This is an interactive program designed to illustrate
the basic principles of what controls power.  Here, we assume that
under the null, the statistic is distributed as a standard normal
with mean zero and variance one, while under the alternative, the
statistic is distributed as a standard normal with a shifted mean.
Power is a function of the mean under the alternative hypothesis
and the desired significance level.  The 'Options' box
allows the user to select the desired significance level using the radio
buttons, and to set
the mean under the alternative hypothesis using the slider.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normplot()
</code></pre>


<h3>Details</h3>

<p>This is a modified version of the 'tkdensity' demo from the 'tcltk' package.
</p>


<h3>Value</h3>

<p>A graph is drawn which reflects the user's choices.  The resulting power
is displayed in the title line of the graph, as well as the mean under
the alternative hypothesis.  The red line indicates the density under
the null hypothesis (mean = 0), and the black line indicates the density
under the alternative hypothesis.  The vertical blue line indicates the
significance threshold.
</p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>

<hr>
<h2 id='nsibs'> A function to compute power of an affected sib pair design. </h2><span id='topic+nsibs'></span>

<h3>Description</h3>

<p>The nsibs() function will compute the number of affected sibpairs
needed to detect linkage to a susceptibility gene attributed with a
user-specified lambda sib (ls) and lambda off (lo). Lambda sib is the
recurrence risk ratio for full- sibs (includes dominance variance),
lambda off is the recurrence risk ratio for parent- offspring pairs (so
does not include dominance variance). In these calculations we assume that
parents and sibs are completely genotyped with markers that perfectly
define the IBD configurations. alpha and beta correspond to the type 1
and 1 - type 2 error rates respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nsibs(ls, lo, alpha, beta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nsibs_+3A_ls">ls</code></td>
<td>
<p> the recurrence risk ratio for full-sibs (includes dominance variance). </p>
</td></tr>
<tr><td><code id="nsibs_+3A_lo">lo</code></td>
<td>
<p> the recurrence risk ratio for parent- offspring pairs (so does not include dominance variance) </p>
</td></tr>
<tr><td><code id="nsibs_+3A_alpha">alpha</code></td>
<td>
<p> the Type 1 error rate. </p>
</td></tr>
<tr><td><code id="nsibs_+3A_beta">beta</code></td>
<td>
<p>  one minus the Type 2 error rate, so (1-beta)=Power.  </p>
</td></tr>
</table>


<h3>Details</h3>

<p>In these calculations we assume that parents and sibs are completely
genotyped with markers that perfectly define the IBD configurations.
</p>


<h3>Value</h3>

<table>
<tr><td><code>n</code></td>
<td>
<p>The number of sib pairs needed to obtain the desired power
under the chosen study design. </p>
</td></tr>
</table>


<h3>Warning </h3>

<p> Makes the simplifying assumptions that (1) all
families will have exactly two affected sibpairs; and that (2) the
identity-by-descent status in the ASPs can be determined unambiguously
</p>


<h3>Note</h3>

<p> These are based on a Mathematica notebook created by Martin Farrall. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNsibs">plotNsibs</a></code>, <code><a href="#topic+plotDom">plotDom</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Use the nsibs() function to calculate the number of sibpairs required
## when lambda sib = 2 and lambda off = 2, in order to have 85 percent 
## power to detect linkage at an alpha level of 0.05.

nsibs(ls=2,lo=2,alpha=0.05,beta=0.15)

</code></pre>

<hr>
<h2 id='ntdt'> Computes number of trios needed for a TDT study. </h2><span id='topic+ntdt'></span>

<h3>Description</h3>

<p>Calculation of how many TDT trios need to be studied in order to have
an adequately powered TDT study. The user specifies the susceptibility
gene allele frequency (q), the frequency of the marker allele (m) in
linkage disequilibrium with the susceptibility allele, the strength of
the linkage disequilibrium between marker and susceptibility allele (ld),
the genotype risk ratio for the susceptibility gene (g), the desired power
(power) and the type 1 error rate (alpha) allowing for multiple testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ntdt(g, q, m, ld, power, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ntdt_+3A_g">g</code></td>
<td>
<p> the genotype risk ratio for the susceptibility gene </p>
</td></tr>
<tr><td><code id="ntdt_+3A_q">q</code></td>
<td>
<p> the susceptibility allele frequency </p>
</td></tr>
<tr><td><code id="ntdt_+3A_m">m</code></td>
<td>
<p> the frequency of the marker allele in linkage disequilibrium
with the susceptibility allele. </p>
</td></tr>
<tr><td><code id="ntdt_+3A_ld">ld</code></td>
<td>
<p> the strength ofthe linkage disequilibrium between marker
and susceptibility allele, on a scale of 0 to 1. </p>
</td></tr>
<tr><td><code id="ntdt_+3A_power">power</code></td>
<td>
<p> the desired power </p>
</td></tr>
<tr><td><code id="ntdt_+3A_alpha">alpha</code></td>
<td>
<p>  the Type 1 error rate </p>
</td></tr>
</table>


<h3>Details</h3>

<p>We will use an R program that implements the power formulae of Abel
and Muller-Myhsok (1998). These formulae allow one to quickly compute
power of the TDT approach under a variety of different conditions. This
R program was modeled on Martin Farrall's Mathematica notebook.
</p>
<p>The power computations here use a simple genetic model with several
aspects: (1) The disease locus has two alleles, A and a, with allele
frequencies q and 1-q. The risk of disease follows a multiplicative
model with genotype relative risks of g and g*g for the A/a and A/A
subjects. (2) There is a perfectly linked marker with two alleles,
with allele frequencies m and 1-m.
</p>


<h3>Value</h3>

<table>
<tr><td><code>nfam</code></td>
<td>
<p> The number of trios needed to obtain the desired power. </p>
</td></tr>
</table>


<h3>Note</h3>

<p> This R program was modeled on Martin
Farrall's Mathematica notebook. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>
<p>Chen WM, Deng HW. A general and accurate approach for computing the
statistical power of the transmission disequilibrium test for complex
disease genes. Genet Epidemiol. 2001 Jul;21(1):53-67.
</p>
<p>Iles MM. On calculating the power of a TDT study&ndash;comparison of
methods. Ann Hum Genet. 2002 Jul;66(Pt 4):323-8.
</p>
<p>Purcell S, Cherny SS, Sham PC. Genetic Power Calculator: design
of linkage and association genetic mapping studies of complex
traits. Bioinformatics. 2003 Jan;19(1):149-50.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNtdt">plotNtdt</a></code>, <code><a href="#topic+ntdt.q">ntdt.q</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute the number of TDT trios would be needed under these conditions:
## q = 0.15; m = 0.25;ld= 0.45;g= 1.5;power =0.85;alpha =0.05;

ntdt(q=0.15,m=0.25,ld=0.45,g=1.5,power=0.85,alpha = 0.05)
</code></pre>

<hr>
<h2 id='ntdt.q'> Computes power of a TDT study as a function of the susceptibility
allele frequency q. </h2><span id='topic+ntdt.q'></span>

<h3>Description</h3>

<p>Power for a TDT study will be highest when the frequency of the
susceptibility allele (q) matches the frequency of the associated allele
(m). We can now examine this by using the ntdt.q() function, which
returns a table give the required sample sizes (and log sample sizes) for
a range of values of q, at three different levels of ld. These levels are
(1) the maximum (dmax), (2) 75% of the maximum (dmax.75), and (3) 50%
of the maximum (dmax.50).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ntdt.q(g, m, alpha = 5e-08, power = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ntdt.q_+3A_g">g</code></td>
<td>
<p> the genotype risk ratio for the susceptibility gene </p>
</td></tr>
<tr><td><code id="ntdt.q_+3A_m">m</code></td>
<td>
<p> the frequency of the marker allele in linkage disequilibrium
with the susceptibility allele. </p>
</td></tr>
<tr><td><code id="ntdt.q_+3A_alpha">alpha</code></td>
<td>
<p>  the Type 1 error rate </p>
</td></tr>
<tr><td><code id="ntdt.q_+3A_power">power</code></td>
<td>
<p> the desired power </p>
</td></tr>
</table>


<h3>Details</h3>

<p>We will use an R program that implements the power formulae of Abel
and Muller-Myhsok (1998). These formulae allow one to quickly compute
power of the TDT approach under a variety of different conditions. This
R program was modeled on Martin Farrall's Mathematica notebook.
</p>
<p>The power computations here use a simple genetic model with several
aspects: (1) The disease locus has two alleles, A and a, with allele
frequencies q and 1-q. The risk of disease follows a multiplicative
model with genotype relative risks of g and g*g for the A/a and A/A
subjects. (2) There is a perfectly linked marker with two alleles,
with allele frequencies m and 1-m.
</p>


<h3>Value</h3>

<p>This function returns a table give the required sample sizes (and log
sample sizes) for a range of values of q, at three different levels
of ld. These levels are (1) the maximum (dmax), (2) 75% of the maximum
(dmax.75), and (3) 50% of the maximum (dmax.50).
</p>
<p>The results can be plotted using the plotNtdt function.
</p>


<h3>Note</h3>

<p> This R program was modeled on Martin
Farrall's Mathematica notebook. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>
<p>Chen WM, Deng HW. A general and accurate approach for computing the
statistical power of the transmission disequilibrium test for complex
disease genes. Genet Epidemiol. 2001 Jul;21(1):53-67.
</p>
<p>Iles MM. On calculating the power of a TDT study&ndash;comparison of
methods. Ann Hum Genet. 2002 Jul;66(Pt 4):323-8.
</p>
<p>Purcell S, Cherny SS, Sham PC. Genetic Power Calculator: design
of linkage and association genetic mapping studies of complex
traits. Bioinformatics. 2003 Jan;19(1):149-50.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNtdt">plotNtdt</a></code>, <code><a href="#topic+ntdt">ntdt</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
ntdt.q(g=2,m=0.5,alpha=0.00000005,power=0.80)

</code></pre>

<hr>
<h2 id='plotDom'> plotDom creates power graphs for an affected sibpair linkage study. </h2><span id='topic+plotDom'></span>

<h3>Description</h3>

<p>plotDom plots the effects of dominance on sample size in an  affected 
sibpair linkage study. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDom(lsib = 1.2, alpha = 0.001, power = 0.85,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDom_+3A_lsib">lsib</code></td>
<td>
<p> is the relative risk ratio to a sibling. </p>
</td></tr>
<tr><td><code id="plotDom_+3A_alpha">alpha</code></td>
<td>
<p> Type 1 error rate (significane level)  </p>
</td></tr>
<tr><td><code id="plotDom_+3A_power">power</code></td>
<td>
<p> is the power of the study to detect inkage
under the (ideal) chosen conditions and parameters. </p>
</td></tr>
<tr><td><code id="plotDom_+3A_...">...</code></td>
<td>
<p>additional arguments for plot </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot, and returns a table of the data 
that generated the plot.  In this table, the first column
gives the lambda off value, while the second column gives
the corresponding number of affected sib pairs needed to have
the desired power.
</p>


<h3>Warning </h3>

<p> Makes the simplifying assumptions that (1) all
families will have exactly two affected sibpairs; and that (2) the
identity-by-descent status in the ASPs can be determined unambiguously
</p>


<h3>Note</h3>

<p> These are based on a Mathematica notebook created by Martin Farrall. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>See Also</h3>

 <p><code><a href="#topic+nsibs">nsibs</a></code>, <code><a href="#topic+plotNsibs">plotNsibs</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Plot the effects of dominance on sample size in an affected sibpair
## linkage study. Do this under the assumption that lambda sib is 1.2,
## and the desired power is 85 percent at an alpha level of 0.001. 

plotDom(lsib=1.2,alpha=0.001,power=0.85)
</code></pre>

<hr>
<h2 id='plotNsibs'> 
plotNsibs: Create power plots for an affected sib pair linkage
study </h2><span id='topic+plotNsibs'></span>

<h3>Description</h3>

<p>Plots the required sample sizes needed to detect linkage
in an affected sib pair linakge study as a function of lambda sib,
under the assumption that lambda sib = lambda off. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNsibs(alpha = 0.001, power = 0.85,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotNsibs_+3A_alpha">alpha</code></td>
<td>
<p> the Type 1 error rate (significance level). </p>
</td></tr>
<tr><td><code id="plotNsibs_+3A_power">power</code></td>
<td>
<p> the power to detect linkage. </p>
</td></tr>
<tr><td><code id="plotNsibs_+3A_...">...</code></td>
<td>
<p>additional arguments for plot </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that lambda sib = lambda off.  
</p>


<h3>Value</h3>

<p>Creates a plot, and returns a table of the data 
that generated the plot.  In this table, the first column
gives the lambda sib value, while the second column gives
the corresponding number of affected sib pairs needed to have
the desired power.
</p>


<h3>Warning </h3>

<p> Makes the simplifying assumptions that (1) all
families will have exactly two affected sibpairs; and that (2) the
identity-by-descent status in the ASPs can be determined unambiguously
</p>


<h3>Note</h3>

<p> These are based on a Mathematica notebook created by Martin Farrall. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>See Also</h3>

 <p><code><a href="#topic+nsibs">nsibs</a></code>, <code><a href="#topic+plotDom">plotDom</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Plot the required sample sizes in an affected sib pair linkage study as
## a function of lambda sib required to have 85 percent power at an alpha 
## level of 0.001, under the assumption that lambda sib = lambda off.

plotNsibs(alpha=0.001,power=0.85)

</code></pre>

<hr>
<h2 id='plotNtdt'> A function for plotting the results from ntdt.q </h2><span id='topic+plotNtdt'></span>

<h3>Description</h3>

<p> This function plots the results from ntdt.q
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNtdt(g, m, alpha = 5e-08, power = 0.8,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotNtdt_+3A_g">g</code></td>
<td>
<p> the genotype risk ratio for the susceptibility gene </p>
</td></tr>
<tr><td><code id="plotNtdt_+3A_m">m</code></td>
<td>
<p> the frequency of the marker allele in linkage disequilibrium
with the susceptibility allele. </p>
</td></tr>
<tr><td><code id="plotNtdt_+3A_alpha">alpha</code></td>
<td>
<p>  the Type 1 error rate </p>
</td></tr>
<tr><td><code id="plotNtdt_+3A_power">power</code></td>
<td>
<p> the desired power </p>
</td></tr>
<tr><td><code id="plotNtdt_+3A_...">...</code></td>
<td>
<p>additional arguments for plot </p>
</td></tr>
</table>


<h3>Details</h3>

<p>We will use an R program that implements the power formulae of Abel
and Muller-Myhsok (1998). These formulae allow one to quickly compute
power of the TDT approach under a variety of different conditions. This
R program was modeled on Martin Farrall's Mathematica notebook.
</p>
<p>The power computations here use a simple genetic model with several
aspects: (1) The disease locus has two alleles, A and a, with allele
frequencies q and 1-q. The risk of disease follows a multiplicative
model with genotype relative risks of g and g*g for the A/a and A/A
subjects. (2) There is a perfectly linked marker with two alleles,
with allele frequencies m and 1-m.
</p>


<h3>Value</h3>

<p>A plot of the results generated by a call to the ntdt.q() function.
</p>


<h3>Note</h3>

<p> This R program was modeled on Martin
Farrall's Mathematica notebook. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>
<p>Chen WM, Deng HW. A general and accurate approach for computing the
statistical power of the transmission disequilibrium test for complex
disease genes. Genet Epidemiol. 2001 Jul;21(1):53-67.
</p>
<p>Iles MM. On calculating the power of a TDT study&ndash;comparison of
methods. Ann Hum Genet. 2002 Jul;66(Pt 4):323-8.
</p>
<p>Purcell S, Cherny SS, Sham PC. Genetic Power Calculator: design
of linkage and association genetic mapping studies of complex
traits. Bioinformatics. 2003 Jan;19(1):149-50.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+ntdt">ntdt</a></code>, <code><a href="#topic+ntdt.q">ntdt.q</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Duplicate Figures 1 and 2 from Abel and Muller-Myhsok (1998)
## Am J Hum Genet 63:664-667
plotNtdt(g=2,m=0.10) 
## gives Figure 1A
plotNtdt(g=2,m=0.50) 
## gives Figure 1B
</code></pre>

<hr>
<h2 id='ptb'> A power table for the TDT design </h2><span id='topic+ptb'></span>

<h3>Description</h3>

<p>Create a power table for different values of q.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptb(g, m, ld, nfam, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptb_+3A_g">g</code></td>
<td>
<p> the genotype risk ratio for the susceptibility gene. </p>
</td></tr> 
<tr><td><code id="ptb_+3A_m">m</code></td>
<td>
<p> the frequency of the marker allele in linkage disequilibrium
with the susceptibility allele. </p>
</td></tr>
<tr><td><code id="ptb_+3A_ld">ld</code></td>
<td>
<p> the strength ofthe linkage disequilibrium between marker
and susceptibility allele, on a scale of 0 to 1. </p>
</td></tr>
<tr><td><code id="ptb_+3A_nfam">nfam</code></td>
<td>
<p> the number of families. </p>
</td></tr>
<tr><td><code id="ptb_+3A_alpha">alpha</code></td>
<td>
<p> the significance level of the test. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>We will use an R program that implements the power formulae of Abel
and Muller-Myhsok (1998). These formulae allow one to quickly compute
power of the TDT approach under a variety of different conditions. This
R program was modeled on Martin Farrall's Mathematica notebook.
</p>
<p>The power computations here use a simple genetic model with several
aspects: (1) The disease locus has two alleles, A and a, with allele
frequencies q and 1-q. The risk of disease follows a multiplicative
model with genotype relative risks of g and g*g for the A/a and A/A
subjects. (2) There is a perfectly linked marker with two alleles,
with allele frequencies m and 1-m.
</p>


<h3>Value</h3>

<table>
<tr><td><code>tb</code></td>
<td>
<p> A table of power results as a function of q</p>
</td></tr>
</table>


<h3>Note</h3>

<p> This R program was modeled on Martin
Farrall's Mathematica notebook. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>
<p>Chen WM, Deng HW. A general and accurate approach for computing the
statistical power of the transmission disequilibrium test for complex
disease genes. Genet Epidemiol. 2001 Jul;21(1):53-67.
</p>
<p>Iles MM. On calculating the power of a TDT study&ndash;comparison of
methods. Ann Hum Genet. 2002 Jul;66(Pt 4):323-8.
</p>
<p>Purcell S, Cherny SS, Sham PC. Genetic Power Calculator: design
of linkage and association genetic mapping studies of complex
traits. Bioinformatics. 2003 Jan;19(1):149-50.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNtdt">plotNtdt</a></code>, <code><a href="#topic+ntdt.q">ntdt.q</a></code>, <code><a href="#topic+ntdt">ntdt</a></code>, <code><a href="#topic+ptdt">ptdt</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>  ptb(g=1.5,m=0.5,ld=0.75,nfam=2000,alpha=0.00000005)
</code></pre>

<hr>
<h2 id='ptdt'> Power for a TDT design </h2><span id='topic+ptdt'></span>

<h3>Description</h3>

<p>Function to compute power for a TDT design
using formulae from Abel and Muller-Myhsok (1998)
Am J Hum Genet 63:664-667
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptdt(g, q, m, ld, nfam, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptdt_+3A_g">g</code></td>
<td>
<p> the genotype risk ratio for the susceptibility gene. </p>
</td></tr>
<tr><td><code id="ptdt_+3A_q">q</code></td>
<td>
<p> the susceptibility allele frequency </p>
</td></tr>
<tr><td><code id="ptdt_+3A_m">m</code></td>
<td>
<p> the frequency of the marker allele in linkage disequilibrium
with the susceptibility allele. </p>
</td></tr>
<tr><td><code id="ptdt_+3A_ld">ld</code></td>
<td>
<p> the strength ofthe linkage disequilibrium between marker
and susceptibility allele, on a scale of 0 to 1. </p>
</td></tr>
<tr><td><code id="ptdt_+3A_nfam">nfam</code></td>
<td>
<p> the number of families. </p>
</td></tr>
<tr><td><code id="ptdt_+3A_alpha">alpha</code></td>
<td>
<p> the significance level of the test. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>We will use an R program that implements the power formulae of Abel
and Muller-Myhsok (1998). These formulae allow one to quickly compute
power of the TDT approach under a variety of different conditions. This
R program was modeled on Martin Farrall's Mathematica notebook.
</p>
<p>The power computations here use a simple genetic model with several
aspects: (1) The disease locus has two alleles, A and a, with allele
frequencies q and 1-q. The risk of disease follows a multiplicative
model with genotype relative risks of g and g*g for the A/a and A/A
subjects. (2) There is a perfectly linked marker with two alleles,
with allele frequencies m and 1-m.
</p>


<h3>Value</h3>

<table>
<tr><td><code>power</code></td>
<td>
<p> power of the statistical test under the chosen conditions.</p>
</td></tr>
</table>


<h3>Note</h3>

<p> This R program was modeled on Martin
Farrall's Mathematica notebook. </p>


<h3>Author(s)</h3>

<p> Daniel E. Weeks </p>


<h3>References</h3>

 
<p>Abel L, Muller-Myhsok B. Maximum-likelihood expression of the
transmission/disequilibrium test and power considerations. Am J Hum
Genet. 1998 Aug;63(2):664-7.
</p>
<p>Chen WM, Deng HW. A general and accurate approach for computing the
statistical power of the transmission disequilibrium test for complex
disease genes. Genet Epidemiol. 2001 Jul;21(1):53-67.
</p>
<p>Iles MM. On calculating the power of a TDT study&ndash;comparison of
methods. Ann Hum Genet. 2002 Jul;66(Pt 4):323-8.
</p>
<p>Purcell S, Cherny SS, Sham PC. Genetic Power Calculator: design
of linkage and association genetic mapping studies of complex
traits. Bioinformatics. 2003 Jan;19(1):149-50.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotNtdt">plotNtdt</a></code>, <code><a href="#topic+ntdt.q">ntdt.q</a></code>, <code><a href="#topic+ntdt">ntdt</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>ptdt(q=0.15,m=0.25,ld=0.45,g=2.5,alpha = 0.05,nfam=300)
ptdt(q=0.15,m=0.25,ld=0.45,g=2.5,alpha = 0.05,nfam=400)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
