<!DOCTYPE html><html lang="en"><head><title>Help for package powerGWASinteraction</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {powerGWASinteraction}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#powerGE'><p>Power for GxE interactions in genetic association studies</p></a></li>
<li><a href='#powerGG'><p>Power for GxG interactions in genetic association studies</p></a></li>
<li><a href='#powerGWASinteraction'><p>Depreciated</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-07-02</td>
</tr>
<tr>
<td>Title:</td>
<td>Power Calculations for GxE and GxG Interactions for GWAS</td>
</tr>
<tr>
<td>Author:</td>
<td>Charles Kooperberg &lt;clk@fredhutch.org&gt; and Li Hsu &lt;lih@fredhutch.org&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Charles Kooperberg &lt;clk@fredhutch.org&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), mvtnorm, pwr</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Description:</td>
<td>Analytical power calculations for GxE and GxG interactions for case-control studies of candidate genes and genome-wide association studies (GWAS). This includes power calculation for four two-step screening and testing procedures.  It can also calculate power for GxE and GxG without any screening.    </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-07-02 20:15:15 UTC; clk</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-07-03 01:34:42</td>
</tr>
</table>
<hr>
<h2 id='powerGE'>Power for GxE interactions in genetic association studies</h2><span id='topic+powerGE'></span>

<h3>Description</h3>

<p>This routine carries out (analytical, approximate) power calculations for identifying Gene-Environment interactions in Genome Wide Association Studies</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerGE(n, power, model, caco, alpha, alpha1, maintain.alpha)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="powerGE_+3A_n">n</code></td>
<td>
<p>Sample size: combined number of cases and controls. Note: exactly one of <code>n</code> and <code>power</code> should be specified.</p>
</td></tr>
<tr><td><code id="powerGE_+3A_power">power</code></td>
<td>
<p>Power: targeted power. Note: exactly one of <code>n</code> and <code>power</code> should be specified.</p>
</td></tr>
<tr><td><code id="powerGE_+3A_model">model</code></td>
<td>
<p>List specifying the genetic model. This list contains the following objects:
</p>

<ul>
<li> <p><code>prev</code> Prevalence of the outcome in the <em>population</em>. Note that for case-only and empirical Bayes estimators to be valid,
the prevalence needs to be low.
</p>
</li>
<li> <p><code>pGene</code> Probability that a <em>binary</em> SNP is 1 (i.e. not the minor allele frequency for a three level SNP).
</p>
</li>
<li> <p><code>pEnv</code> Frequency of the binary environmental variable.
</p>
</li>
<li> <p><code>orGE</code> Odds ratio between the binary SNP and binary environmental variable.
</p>
</li>
<li> <p><code>beta.LOR</code> Vector of length three with the odds ratios of the genetic, environmental, and GxE interaction effect, respectively.
</p>
</li>
<li> <p><code>nSNP</code> Number of SNPs (genes) being tested.</p>
</li></ul>
</td></tr>
<tr><td><code id="powerGE_+3A_caco">caco</code></td>
<td>
<p>Fraction of the sample that are cases (default = 0.5).</p>
</td></tr>
<tr><td><code id="powerGE_+3A_alpha">alpha</code></td>
<td>
<p>Overall (family-wise) Type 1 error (default = 0.05).</p>
</td></tr>
<tr><td><code id="powerGE_+3A_alpha1">alpha1</code></td>
<td>
<p>Significance level at which testing during the first stage (screening) takes place. If alpha1 = 1, there is no screening.</p>
</td></tr>
<tr><td><code id="powerGE_+3A_maintain.alpha">maintain.alpha</code></td>
<td>
<p>Some combinations of screening and GxE testing methods do not maintain the proper Type 1 error. Default is <code>True</code>: combinations that do not maintain the Type 1 error are not computed. If <code>maintain.alpha</code> is <code>False</code> all combinations are computed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The routine computes power for a variety of two-stage procedures. Five different screening procedures are used:
</p>

<ul>
<li> <p><b>No screening</b> All SNPs are tested for interaction
</p>
</li>
<li> <p><b>Marginal screening</b> Only SNPs that are marginally significant at level alpha1 are screened for interaction. See Kooperberg and LeBlanc (2010).
</p>
</li>
<li> <p><b>Correlation screening</b> Only SNPs that are, combined over all cases and controls, associated with the environmental variable at level alpha 1
are screened for interaction. See Murcray et al. (2012).
</p>
</li>
<li> <p><b>Cocktail screening</b> SNPs are screened on the most significant of marginal and correlation screening. See Hsu et al. (2012).
</p>
</li>
<li> <p><b>Chi-square screening</b> SNPs are screened using a chi-square combination of correlation and marginal screening. See Gauderman et al. (2013).</p>
</li></ul>

<p>After screening, the SNPs that pass the screen can be tested using
</p>

<ul>
<li> <p><b>Case-control</b> The standard case-control estimator.
</p>
</li>
<li> <p><b>Case-only</b> The case-only estimator.
</p>
</li>
<li> <p><b>Empirical Bayes</b> The empirical Bayes estimator of Mukherjee and Chatterjee (2010).</p>
</li></ul>

<p>If screening took place using the correlation or chi-square screening, the Type 1 error won't be
maintained if the final GxE testing is carried out using either the case-only or empirical Bayes estimator.
See Dai et al. (2012). The cocktail screening maintains the Type 1 family wise error rate, since only those
SNPs that pass on to the second stage using marginal screening will use the case-only or empirical Bayes estimator, the
SNPs that pass on to the second stage using correlation screening will always use the case-control estimator.
</p>
<p>When SNP and environment are correlated in the population (i.e. <code>model$orGE</code> does not equal 1) the case-only estimator does not maintain the Type 1 error.
The empirical Bayes estimator may also have a moderately inflated Type 1 error. When the disease is common either the case-only
estimator or the empirical Bayes estimator also may not estimate the GxE interaction.
</p>
<p>Power calculations are described in Kooperberg, Dai, and Hsu (2014). Briefly, for a given genetic model we compute the expected p-values for all
screening statistics. We then use a normal approximation to compute the probability that this SNP passes the screening (e.g., if <code>alpha1</code> equaled this expected p-value this probability would be exactly 0.5), and combine this with power calculations for the second stage of GxE testing.</p>


<h3>Value</h3>

<p>A list with three components.
</p>
<table role = "presentation">
<tr><td><code>power</code></td>
<td>
<p>A 5x3 matrix with estimated power for all testing approaches, only if <code>n</code> was specified.</p>
</td></tr>
<tr><td><code>samplesize</code></td>
<td>
<p>A 5x3 matrix with required sample sizes for all testing approaches, only if <code>power</code> was specified.</p>
</td></tr>
<tr><td><code>expected.p</code></td>
<td>
<p>A 5x3 matrix with the expected p value for the SNP to pass screening. This p-value depends on the sample size, but not on the second stage testing.</p>
</td></tr>
<tr><td><code>prob.select</code></td>
<td>
<p>A 5x3 matrix with the probability that the interacting SNP would pass the screening stage. This probability depends on the sample size, but not on the second stage testing.</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Li Hsu <a href="mailto:lih@fredhutch.org">lih@fredhutch.org</a> and
Charles Kooperberg <a href="mailto:clk@fredhutch.org">clk@fredhutch.org</a>.
</p>


<h3>References</h3>

<p>Dai JY, Kooperberg C, LeBlanc M, Prentice RL (2012).
Two-stage testing procedures with  independent filtering for genome-wide
gene-environment interaction. <em>Biometrika</em>, <b>99</b>, 929-944.
</p>
<p>Gauderman WJ, Zhang P, Morrison JL, Lewinger JP (2013).
Finding novel genes by testing GxE interactions in a genome-wide association study.
<em>Genetic Epidemiology</em>, <b>37</b>, 603-613.
</p>
<p>Hsu L, Jiao S, Dai JY, Hutter C, Peters U, Kooperberg C (2012).
Powerful cocktail methods for detecting genome-wide gene-environment interaction.
<em>Genetic Epidemiology</em>, <b>36</b>, 183-194. 
</p>
<p>Kooperberg C, Dai, JY, Hsu L (2014).
Two-stage procedures for the identification of gene x
environment and gene x gene interactions in
genome-wide association studies. <em>To appear</em>.
</p>
<p>Kooperberg C, LeBlanc ML (2008). Increasing the power of
identifying gene x gene interactions in genome-wide
association studies. <em>Genetic Epidemiology</em>, <b>32</b>, 255-263.
</p>
<p>Mukherjee B, Chatterjee N (2008). Exploiting gene-environment inde-
pendence for analysis of case-control studies: an empirical
Bayes-type shrinkage estimator to trade-off between bias and efficiency
<em>Biometrics</em>, <b>64</b>, 685-694.
</p>
<p>Murcray CE, Lewinger JP, Gauderman WJ (2009).
Gene-environment interaction in genome-wide association studies.
<em>American Journalk of Epidemiology</em>, <b>169</b>, 219-226.
</p>


<h3>See Also</h3>

<p>powerGG</p>


<h3>Examples</h3>

<pre><code class='language-R'>mod1 &lt;- list(prev=0.01,pGene=0.2,pEnv=0.2,beta.LOR=log(c(1.0,1.2,1.4)),orGE=1.2,nSNP=10^6)
results &lt;- powerGE(n=20000, model=mod1,alpha1=.01)
print(results)

mod2 &lt;- list(prev=0.01,pGene=0.2,pEnv=0.2,beta.LOR=log(c(1.0,1.0,1.4)),orGE=1,nSNP=10^6)
results &lt;- powerGE(power=0.8, model=mod2,alpha1=.01)
print(results)
</code></pre>

<hr>
<h2 id='powerGG'>Power for GxG interactions in genetic association studies</h2><span id='topic+powerGG'></span>

<h3>Description</h3>

<p>This routine carries out (analytical, approximate) power calculations for identifying Gene-Gene interactions in Genome Wide Association Studies</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerGG(n, power, model, caco, alpha, alpha1)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="powerGG_+3A_n">n</code></td>
<td>
<p>Sample size: combined number of cases and controls. Note: exactly one of <code>n</code> and <code>power</code> should be specified.</p>
</td></tr>
<tr><td><code id="powerGG_+3A_power">power</code></td>
<td>
<p>Power: targeted power. Note: exactly one of <code>n</code> and <code>power</code> should be specified.</p>
</td></tr>
<tr><td><code id="powerGG_+3A_model">model</code></td>
<td>
<p>List specifying the genetic model. This list contains the following objects:
</p>

<ul>
<li> <p><code>prev</code> Prevalence of the outcome in the <em>population</em>. Note that for case-only and empirical Bayes estimators to be valid,
the prevalence needs to be low.
</p>
</li>
<li> <p><code>pGene1</code> Probability that the first <em>binary</em> SNP is 1 (i.e. not the minor allele frequency for a three level SNP).
</p>
</li>
<li> <p><code>pGene2</code> Probability that the first <em>binary</em> SNP is 1 (i.e. not the minor allele frequency for a three level SNP).
</p>
</li>
<li> <p><code>beta.LOR</code> Vector of length three with the odds ratios of the first genetic, second genetic, and GxG interaction effect, respectively.
</p>
</li>
<li> <p><code>nSNP</code> Number of SNPs (genes) being tested.</p>
</li></ul>
</td></tr>
<tr><td><code id="powerGG_+3A_caco">caco</code></td>
<td>
<p>Fraction of the sample that are cases (default = 0.5).</p>
</td></tr>
<tr><td><code id="powerGG_+3A_alpha">alpha</code></td>
<td>
<p>Overall (family-wise) Type 1 error (default = 0.05).</p>
</td></tr>
<tr><td><code id="powerGG_+3A_alpha1">alpha1</code></td>
<td>
<p>Significance level at which testing during the first stage (screening) takes place. If alpha1 = 1, there is no screening.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The routine computes power calculations for a two-stage procedure with marginal screening followed by either case-control or case-only testing.</p>


<h3>Value</h3>

<p>A data frame consisting of two numbers: the power for the case-control and case-only approaches if <code>n</code> is specified or the required combined sample size 
for the case-control and case-only approaches if <code>power</code> is specified.</p>


<h3>Author(s)</h3>

<p>Charles Kooperberg, <a href="mailto:clk@fredhutch.org">clk@fredhutch.org</a></p>


<h3>References</h3>

<p>Kooperberg C, LeBlanc M (2008). Increasing the power of identifying gene x gene interactions in genome-wide association studies.
<em>Genetic Epidemiology</em>, <b>32</b>, 255-263.</p>


<h3>See Also</h3>

<p>powerGG</p>


<h3>Examples</h3>

<pre><code class='language-R'>mod1 &lt;- list(prev=0.05, pGene1=0.3, pGene2=0.3, beta.LOR=c(0,0,.6),nSNP=500000)
powerGG(n=10000,mod=mod1,caco=0.5,alpha=.05,alpha1=.001)
powerGG(power=0.8,mod=mod1,caco=0.5,alpha=.05,alpha1=.001)
</code></pre>

<hr>
<h2 id='powerGWASinteraction'>Depreciated</h2><span id='topic+powerGWASinteraction'></span>

<h3>Description</h3>

<p>This function is depreciated and has been replaced by <code>powerGG</code> and <code>powerGE</code></p>


<h3>Usage</h3>

<pre><code class='language-R'>powerGWASinteraction()</code></pre>


<h3>Value</h3>

<p>An error message is printed</p>


<h3>Author(s)</h3>

<p>Charles Kooperberg, <a href="mailto:clk@fredhutch.org">clk@fredhutch.org</a></p>


<h3>See Also</h3>

<p>powerGG,powerGE</p>


<h3>Examples</h3>

<pre><code class='language-R'> powerGWASinteraction()</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
