<!DOCTYPE html><html><head><title>Help for package powerEQTL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {powerEQTL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#powerEQTL.ANOVA'>
<p>Power Calculation for EQTL Analysis Based on Un-Balanced One-Way ANOVA</p></a></li>
<li><a href='#powerEQTL.scRNAseq'>
<p>Power Calculation for Association Between Genotype and Gene Expression Based on Single Cell RNAseq Data</p></a></li>
<li><a href='#powerEQTL.scRNAseq.sim'>
<p>Power Calculation for Association Between Genotype and Gene Expression Based on Single Cell RNAseq Data via Simulation from ZINB Mixed Effects Regression Model</p></a></li>
<li><a href='#powerEQTL.SLR'>
<p>Power Calculation for EQTL Analysis Based on Simple Linear Regression</p></a></li>
<li><a href='#powerLME'>
<p>Power Calculation for Simple Linear Mixed Effects Model</p></a></li>
<li><a href='#powerLMEnoCov'>
<p>Power Calculation for Simple Linear Mixed Effects Model Without Covariate</p></a></li>
<li><a href='#simDat.eQTL.scRNAseq'>
<p>Generate Gene Expression Levels Of One Gene And Genotypes Of One SNP For Subjects With Multiple Cells Based On ZINB Mixed Effects Regression Model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Power and Sample Size Calculation for Bulk Tissue and
Single-Cell eQTL Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-07-21</td>
</tr>
<tr>
<td>Author:</td>
<td>Xianjun Dong [aut, ctb],
  Xiaoqi Li [aut, ctb],
  Tzuu-Wang Chang [aut, ctb],
  Scott T. Weiss [aut, ctb],
  Weiliang Qiu [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, nlme, GLMMadaptive, parallel</td>
</tr>
<tr>
<td>Description:</td>
<td>Power and sample size calculation for bulk tissue and single-cell eQTL analysis
             based on ANOVA, simple linear regression, or linear mixed effects model. It can also calculate power/sample size 
             for testing the association of a SNP to a continuous type phenotype.
             Please see the reference: Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, Qiu W. (2021) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtab385">doi:10.1093/bioinformatics/btab385</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sterding/powerEQTL">https://github.com/sterding/powerEQTL</a> and
<a href="https://bwhbioinfo.shinyapps.io/powerEQTL/">https://bwhbioinfo.shinyapps.io/powerEQTL/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-07-21 16:58:53 UTC; i0388414</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-22 08:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='powerEQTL.ANOVA'>
Power Calculation for EQTL Analysis Based on Un-Balanced One-Way ANOVA
</h2><span id='topic+powerEQTL.ANOVA'></span>

<h3>Description</h3>

<p>Power calculation for eQTL analysis that tests if a SNP is associated to a gene probe by using un-balanced one-way ANOVA. This function can be used to calculate one of the 3 parameters (power, sample size, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 2 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerEQTL.ANOVA(MAF,
                deltaVec=c(-0.13, 0.13),
                n=200,
                power = NULL,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000,
                n.lower = 4,
                n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerEQTL.ANOVA_+3A_maf">MAF</code></td>
<td>

<p>numeric. Minor allele frequency.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_deltavec">deltaVec</code></td>
<td>

<p>numeric. A vector having 2 elements. The first element is equal to
<code class="reqn">\mu_2-\mu_1</code> and the second element is equal to
<code class="reqn">\mu_3-\mu_2</code>, where <code class="reqn">\mu_1</code> is
the mean gene expression level for the mutation homozygotes,
<code class="reqn">\mu_2</code> is the mean gene expression level for the heterozygotes,
and <code class="reqn">\mu_3</code> is the mean gene expression level for the 
wild-type gene expression level.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_n">n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_power">power</code></td>
<td>

<p>numeric. Power for testing if 3 genotypes have the same mean 
gene expression levels.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_sigma">sigma</code></td>
<td>

<p>numeric. Standard deviation of the random error.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_fwer">FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_ntests">nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td></tr>
<tr><td><code id="powerEQTL.ANOVA_+3A_n.lower">n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr> <tr><td><code id="powerEQTL.ANOVA_+3A_n.upper">n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If we would like to test potential non-linear relationship between genotype of a SNP and expression of a gene, we can use un-balanced one-way ANOVA. Actually, an article published by the GTEx Consortium in 2013 used this approach. 
</p>
<p>Suppose there are <code class="reqn">k=3</code> groups of subjects: (1) mutation homozygotes; (2) heterozygotes; and (3) wildtype homozygotes. We would like to test if the mean expression <code class="reqn">\mu_i</code>, <code class="reqn">i=1, \ldots, k</code>, of the gene is the same among the <code class="reqn">k</code> groups of subjects. We can use the following one-way ANOVA model to characterize the relationship between observed gene expression level <code class="reqn">y_{ij}</code> and the population mean expression level <code class="reqn">\mu_i</code>:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \mu_i + \epsilon_{ij}, \quad \epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p>where <code class="reqn">i=1,\ldots, k</code>, <code class="reqn">j= 1, \ldots, n_i</code>,
<code class="reqn">y_{ij}</code> is the observed gene expression level for the <code class="reqn">j</code>-th subject in the <code class="reqn">i</code>-th group, <code class="reqn">\mu_i</code> is the mean gene expression level of the <code class="reqn">i</code>-th group, <code class="reqn">\epsilon_{ij}</code> is the random error, <code class="reqn">k</code> is the number of groups, <code class="reqn">n_i</code> is the number of subjects in the <code class="reqn">i</code>-th group. 
Denote the total number of subjects as <code class="reqn">N = \sum_{i=1}^{k} n_i</code>.
That is, we have <code class="reqn">n_1</code> mutation homozygotes, <code class="reqn">n_2</code> heterozygotes, and <code class="reqn">n_3</code> wildtype homozygotes.
</p>
<p>We would like to test the null hypothesis <code class="reqn">H_0</code> and alternative hypothesis <code class="reqn">H_1</code>:
</p>
<p style="text-align: center;"><code class="reqn">H_0:  \mu_1 = \mu_2 = \mu_3,</code>
</p>

<p style="text-align: center;"><code class="reqn">H_1:  \mbox{not all means are the same}.</code>
</p>

<p>According to O'Brien and Muller (1993), the power calculation formula for unbalanced one-way ANOVA is
</p>
<p style="text-align: center;"><code class="reqn">power=Pr\left(\left.F\geq F_{1-\alpha}\left(k-1, N-k\right)\right| F\sim
F_{k-1, N-k, \lambda}\right),</code>
</p>

<p>where <code class="reqn">k=3</code> is the number of groups of subjects, <code class="reqn">N</code> is the total number
of subjects, <code class="reqn">F_{1-\alpha}\left(k-1, N-k\right)</code> is the 
<code class="reqn">100(1-\alpha)</code>-th percentile of central F distribution with degrees of freedoms <code class="reqn">k-1</code> and <code class="reqn">N-k</code>, and <code class="reqn">F_{k-1, N-k, \lambda}</code> is the non-central F distribution
with degrees of freedoms <code class="reqn">k-1</code> and <code class="reqn">N-k</code> and non-central parameter (ncp)
<code class="reqn">\lambda</code>. The ncp <code class="reqn">\lambda</code> is equal to
</p>
<p style="text-align: center;"><code class="reqn">\lambda=\frac{N}{\sigma^2}\sum_{i=1}^{k} w_i \left(\mu_i-\mu\right)^2,</code>
</p>
<p> where <code class="reqn">\mu_i</code> is the mean gene expression level
for the <code class="reqn">i</code>-th group of subjects, <code class="reqn">w_i</code> is the weight for the <code class="reqn">i</code>-th group of subjects, <code class="reqn">\sigma^2</code> is the variance of the random errors in ANOVA (assuming each group has equal variance), and
<code class="reqn">\mu</code> is the weighted mean gene expression level
</p>
<p style="text-align: center;"><code class="reqn">\mu=\sum_{i=1}^{k}w_i \mu_i.</code>
</p>

<p>The weights <code class="reqn">w_i=n_i/N</code> are the sample proportions for the 3 groups of subjects, where <code class="reqn">N=n_1+n_2+n_3</code> is the total number of subjects. Hence, 
<code class="reqn">\sum_{i=1}^{3}w_i = 1</code>.
Based on Hardy-Weinberg Equilibrium, we have
<code class="reqn">w_1 = \theta^2</code>, <code class="reqn">w_2 = 2\theta(1-\theta)</code>, and <code class="reqn">w_3 = (1-\theta)^2</code>, 
where <code class="reqn">\theta</code> is MAF.
</p>
<p>Without loss of generality, we set <code class="reqn">\mu_1 = -\delta_1</code>, <code class="reqn">\mu_2=0</code>,
and <code class="reqn">\mu_3 = \delta_2</code>.
</p>
<p>We adopted the parameters from the GTEx cohort (see the Power analysis&quot; section of Nature Genetics, 2013; https://www.nature.com/articles/ng.2653), where they modeled the expression data as having a log-normal distribution with a log standard deviation of 0.13 within each genotype class (AA, AB, BB). This level of noise is based on estimates from initial GTEx data. In their power analysis, they assumed the across-genotype difference delta = 0.13 (i.e., equivalent to detecting a log expression change similar to the standard deviation within a single genotype class). 
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>The GTEx Consortium. 
The Genotype-Tissue Expression (GTEx) project. 
Nature Genetics, 45:580-585, 2013.
</p>
<p>O'Brien, RG and Muller, KE. 
Unified power analysis for t-tests through multivariate hypotheses. 
In LK Edwards, editor, Applied Analysis of Variance in Behavioral Science,
pages 297-344. New York: Dekker, 1993.
</p>
<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# calculate power
powerEQTL.ANOVA(MAF = 0.1,
                deltaVec = c(-0.13, 0.13),
                n = 282,
                power = NULL,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000)
                
# calculate sample size (total number of subjects)
powerEQTL.ANOVA(MAF = 0.1,
                deltaVec = c(-0.13, 0.13),
                n = NULL,
                power = 0.8,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000) 
                
# calculate minimum allowable MAF
powerEQTL.ANOVA(MAF = NULL,
                deltaVec = c(-0.13, 0.13),
                n = 282,
                power = 0.8,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000)                 
                       
                       
</code></pre>

<hr>
<h2 id='powerEQTL.scRNAseq'>
Power Calculation for Association Between Genotype and Gene Expression Based on Single Cell RNAseq Data
</h2><span id='topic+powerEQTL.scRNAseq'></span>

<h3>Description</h3>

<p>Power calculation for association between genotype and gene expression based on single cell RNAseq data. This function can be used to calculate one of the 4 parameters (power, sample size, minimum detectable slope, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 3 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerEQTL.scRNAseq(
  slope, 
  n, 
  m, 
  power = NULL,
  sigma.y, 
  MAF = 0.2, 
  rho = 0.8, 
  FWER = 0.05,
  nTests = 1,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerEQTL.scRNAseq_+3A_slope">slope</code></td>
<td>

<p>numeric. Slope under alternative hypothesis.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_n">n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_m">m</code></td>
<td>

<p>integer. Number of cells per subject.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_power">power</code></td>
<td>

<p>numeric. Power for testing if the slope is equal to zero.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_sigma.y">sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the gene expression.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_maf">MAF</code></td>
<td>

<p>numeric. Minor allele frequency (between 0 and 0.5).
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_rho">rho</code></td>
<td>

<p>numeric. Intra-class correlation (i.e., correlation between 
<code class="reqn">y_{ij}</code> and <code class="reqn">y_{ik}</code> for the <code class="reqn">j</code>-th  and <code class="reqn">k</code>-th cells of the <code class="reqn">i</code>-th subject).
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_fwer">FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate for one pair (SNP, gene).
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_ntests">nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq_+3A_n.lower">n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr> <tr><td><code id="powerEQTL.scRNAseq_+3A_n.upper">n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We assume the following simple linear mixed effects model for each
(SNP, gene) pair to characterize the association between
genotype and gene expression:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \beta_{0i} + \beta_1 * x_i + \epsilon_{ij},</code>
</p>

<p>where 
</p>
<p style="text-align: center;"><code class="reqn">\beta_{0i} \sim N\left(\beta_0, \sigma^2_{\beta}\right),</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">\epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p><code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>, <code class="reqn">n</code> is the number of
subjects, <code class="reqn">m</code> is the number of cells per subject,
<code class="reqn">y_{ij}</code> is the gene expression level for the <code class="reqn">j</code>-th cell
of the <code class="reqn">i</code>-th subject, <code class="reqn">x_i</code> is the genotype for the
<code class="reqn">i</code>-th subject using additive coding. That is, <code class="reqn">x_i=0</code> indicates the <code class="reqn">i</code>-th subject is a wildtype homozygote,
<code class="reqn">x_i=1</code> indicates the <code class="reqn">i</code>-th subject is a heterozygote,
and <code class="reqn">x_i=2</code> indicates the <code class="reqn">i</code>-th subject is a mutation
homozygote.
</p>
<p>We would like to test the following hypotheses:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \beta_1=0,</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">H_1: \beta_1 = \delta,</code>
</p>

<p>where <code class="reqn">\delta\neq 0</code>.
</p>
<p>For a given SNP, we assume Hardy-Weinberg Equilibrium and denote the minor allele
frequency of the SNP as <code class="reqn">\theta</code>.
</p>
<p>We can derive the power calculation formula is
</p>
<p style="text-align: center;"><code class="reqn">power=1- \Phi\left(z_{\alpha^{*}/2}-a\times b\right)
+\Phi\left(-z_{\alpha^{*}/2} - a\times b\right),</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">a=
\frac{\sqrt{2\theta\left(1-\theta\right)}}{\sigma_y}</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">
b=\frac{\delta\sqrt{m(n-1)}}{\sqrt{1+(m-1)\rho}}
</code>
</p>

<p>and <code class="reqn">z_{\alpha^{*}/2}</code> is the upper <code class="reqn">100\alpha^{*}/2</code>
percentile of the standard normal distribution,
<code class="reqn">\alpha^{*}=FWER/nTests</code>, nTests is the number of
(SNP, gene) pairs,
<code class="reqn">\sigma_y=\sqrt{\sigma^2_{\beta}+\sigma^2}</code>,
and <code class="reqn">\rho=\sigma^2_{\beta}/\left(\sigma^2_{\beta}+\sigma^2\right)</code> is the intra-class correlation.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>;
</p>
<p>minimum allowable MAF if the input parameter <code>MAF = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  n = 102
  m = 227868
  
  # calculate power
  power = powerEQTL.scRNAseq(
    slope = 0.6, 
    n = n, 
    m = m,
    power = NULL,
    sigma.y = 0.29, 
    MAF = 0.05, 
    rho = 0.8, 
    nTests = 1e+6)

  print(power)
  
  # calculate sample size (total number of subjects)
  n = powerEQTL.scRNAseq(
    slope = 0.6, 
    n = NULL, 
    m = m,
    power = 0.9567288,
    sigma.y = 0.29, 
    MAF = 0.05, 
    rho = 0.8, 
    nTests = 1e+6)

  print(n)

  # calculate slope
  slope = powerEQTL.scRNAseq(
    slope = NULL, 
    n = n, 
    m = m,
    power = 0.9567288,
    sigma.y = 0.29, 
    MAF = 0.05, 
    rho = 0.8, 
    nTests = 1e+6)

  print(slope)
  
  # calculate MAF
  MAF = powerEQTL.scRNAseq(
    slope = 0.6, 
    n = n, 
    m = m,
    power = 0.9567288,
    sigma.y = 0.29, 
    MAF = NULL, 
    rho = 0.8, 
    nTests = 1e+6)
  print(MAF)

</code></pre>

<hr>
<h2 id='powerEQTL.scRNAseq.sim'>
Power Calculation for Association Between Genotype and Gene Expression Based on Single Cell RNAseq Data via Simulation from ZINB Mixed Effects Regression Model
</h2><span id='topic+powerEQTL.scRNAseq.sim'></span>

<h3>Description</h3>

<p>Power calculation for association between genotype and gene expression based on single cell RNAseq data via simulation from ZINB mixed effects regression model. This function can be used to calculate one of the 4 parameters (power, sample size, minimum detectable slope, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 3 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerEQTL.scRNAseq.sim(slope, 
		       n, 
		       m, 
                       power = NULL,
		       m.int = -1, 
		       sigma.int = 1, 
		       zero.p = 0.1, 
		       theta = 1, 
		       MAF = 0.2, 
		       FWER = 0.05, 
		       nTests = 1, 
		       nSim = 1000, 
		       estMethod = "GLMMadaptive", 
		       nCores = 1,
                       n.lower = 2.01,
                       n.upper = 1e+4,
                       slope.lower =  1e-6,
                       slope.upper = log(1.0e+6),
                       MAF.lower = 0.05,
                       MAF.upper = 0.49
)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_slope">slope</code></td>
<td>

<p>numeric. Slope (see details).
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_n">n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_m">m</code></td>
<td>

<p>integer. Number of cells per subject.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_power">power</code></td>
<td>

<p>numeric. Power for testing if the slope is equal to zero.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_m.int">m.int</code></td>
<td>

<p>numeric. Mean of random intercept (see details).
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_sigma.int">sigma.int</code></td>
<td>

<p>numeric. Standard deviation of random intercept (see details).
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_zero.p">zero.p</code></td>
<td>

<p>numeric. Probability that an excess zero occurs.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_theta">theta</code></td>
<td>

<p>numeric. Dispersion parameter of negative binomial distribution.
The smaller <code>theta</code> is, the larger variance of NB random variable is.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_maf">MAF</code></td>
<td>

<p>numeric. Minor allele frequency of the SNP.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_fwer">FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_ntests">nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_nsim">nSim</code></td>
<td>

<p>integer. Number of simulated datasets to be generated.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_estmethod">estMethod</code></td>
<td>

<p>character. Indicates which method would be used to fit zero inflated 
negative binomial mixed effects model.
Currently, the possible choice is &ldquo;GLMMadaptive&rdquo;.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_ncores">nCores</code></td>
<td>

<p>integer. Number of computer cores used by <code>mclapply</code> for parallel computing. For Windows, <code>nCores=1</code>.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_n.lower">n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr> <tr><td><code id="powerEQTL.scRNAseq.sim_+3A_n.upper">n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_slope.lower">slope.lower</code></td>
<td>

<p>numeric. Lower bound of the slope. Only used when calculating minimum slope.
</p>
</td></tr> <tr><td><code id="powerEQTL.scRNAseq.sim_+3A_slope.upper">slope.upper</code></td>
<td>

<p>numeric. Upper bound of the slope Only used when calculating minimum slope.
</p>
</td></tr>
<tr><td><code id="powerEQTL.scRNAseq.sim_+3A_maf.lower">MAF.lower</code></td>
<td>

<p>numeric. Lower bound of the MAF. Only used when calculating minimum MAF.
</p>
</td></tr> <tr><td><code id="powerEQTL.scRNAseq.sim_+3A_maf.upper">MAF.upper</code></td>
<td>

<p>numeric. Upper bound of the MAF Only used when calculating minimum MAF.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the power for testing if genotypes of a SNP is associated with the expression of a gene via <code>nSim</code> datasets generated from zero-inflated negative binomial (ZINB) regression model.
</p>
<p>Each dataset is generated from zero-inflated negative binomial mixed effects regression model with only one covariate: genotype. That is, the read counts of a gene follows a mixture of 2-component distributions. One component takes only one value: zero. The other component is negative binomial distribution, which takes non-negative values 0, 1, 2, .... The log mean of the negative binomial distribution is linear function of the genotype.
</p>
<p>For each dataset, the p-value for testing if the slope for genotype is equal to zero will be calculated.
</p>
<p>The proportion of p-values <code class="reqn">&lt; \alpha</code> is the estimated power, where <code class="reqn">\alpha = FWER/nTests</code>.
</p>
<p>Each simulated dataset contains gene expression levels of one gene and genotypes of one SNP for subjects with multiple cells.
The gene expression levels (read counts) follow zero-inflated negative binomial distribution. Denote <code class="reqn">Y_{ij}</code> as the read counts for the <code class="reqn">j</code>-th cell of
the <code class="reqn">i</code>-th subject, <code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>,
<code class="reqn">n</code> is the number of subjects, and <code class="reqn">m</code> is the number of cells per subject.
Denote <code class="reqn">p</code> as the probability that <code class="reqn">Y_{ij}=0</code> is an excess zero.
With probability <code class="reqn">1-p</code>, <code class="reqn">Y_{ij}</code> follows a negative binomial distribution <code class="reqn">NB(\mu, \theta)</code>, where <code class="reqn">\mu</code> is the mean (i.e., <code class="reqn">\mu=E(Y_{ij})</code>) and <code class="reqn">\theta</code> is the dispersion parameter.
The variance of the NB distribution is <code class="reqn">\mu + \mu^2/\theta</code>.
The relationship between gene expression and genotype for the <code class="reqn">i</code>-th subject is characterized by the equation 
</p>
<p style="text-align: center;"><code class="reqn">\mu_i = \exp(\beta_{0i} + \beta_{1} x_i),</code>
</p>

<p>where <code class="reqn">\beta_{0i}</code> is the random intercept following a normal
distribution <code class="reqn">N(\beta_0, \sigma^2)</code> to account for within-subject correlation of gene expression, <code class="reqn">\beta_0</code> is the mean of the random intercept, <code class="reqn">\sigma</code> is the standard deviation of the random intercept, <code class="reqn">\beta_1</code> is the slope, and <code class="reqn">x_i</code> is the additive-coded genotype for the SNP with minor allele frequency <code class="reqn">MAF</code>.
</p>
<p>We assume that the SNP satisfies the Hardy-Weinberg Equilibrium. That is, the
probabilities of the 3 genotypes <code class="reqn">(0, 1, 2)</code> are <code class="reqn">(1-MAF)^2, 2 MAF (1-MAF), MAF^2</code>, respectively.
</p>
<p>For simplicity, we assume that excess zeros are caused by technical issues, hence are not related to genotypes.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>;
</p>
<p>minimum allowable MAF if the input parameter <code>MAF = NULL</code>.
</p>


<h3>Note</h3>

<p>The speed of simulation approach is slow. It is recommended to run the function <code>powerEQTL.scRNAseq.sim</code> in parallel computing environment (e.g., Unix/Linux) and set <code>nCores &gt; 1</code>.
Also, it is important to set appropriate ranges to search sample size, minimum detectable slope, or minimum allowable MAF. If the function (e.g., GLMMadaptive) to fit data returns an error message 
for a simulated dataset, then the function <code>powerEQTL.scRNAseq.sim</code> will try a few more runs with different simulated datasets. If all the runs failed, zero (for power estimation) or <code>NA</code> or <code>NaN</code> (for estimating slope, sample size, or MAF) would be output.  It is recommended to set <code>nSim &gt;= 1000</code> to get stable results.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
nSubj = 10
nCellPerSubj = 10

# calculate power
power = powerEQTL.scRNAseq.sim(
  slope = 1.62, # slope
  n = nSubj, # total number of subjects
  m = nCellPerSubj, # number of cells per subject 
  power = NULL, # power to be estimated
  m.int = -1, # mean of random intercept
  sigma.int = 1, # SD of the random intercept
  zero.p = 0.01, # probability that an excess zero occurs
  theta = 1, # dispersion parameter of NB distribution NB(mu, theta)
  MAF = 0.45, 
  FWER = 0.05,
  nTests = 1,
  nSim  =  5, # number of simulations
  estMethod = "GLMMadaptive", # parameter estimation method for ZINB
  nCores = 1 # number of computer cores used by 'mclapply'
)

print(power)


</code></pre>

<hr>
<h2 id='powerEQTL.SLR'>
Power Calculation for EQTL Analysis Based on Simple Linear Regression
</h2><span id='topic+powerEQTL.SLR'></span>

<h3>Description</h3>

<p>Power calculation for eQTL analysis that tests if a SNP is associated to a gene probe by using simple linear regression. This function can be used to calculate one of the 4 parameters (power, sample size, minimum detectable slope, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 3 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerEQTL.SLR(
  MAF,
  slope = 0.13,
  n = 200,
  power = NULL,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerEQTL.SLR_+3A_maf">MAF</code></td>
<td>

<p>numeric. Minor allele frequency.
</p>
</td></tr>
<tr><td><code id="powerEQTL.SLR_+3A_slope">slope</code></td>
<td>

<p>numeric. Slope of the simple linear regression.
</p>
</td></tr>
<tr><td><code id="powerEQTL.SLR_+3A_n">n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerEQTL.SLR_+3A_power">power</code></td>
<td>

<p>numeric. Power for testing if the slope is equal to zero.
</p>
</td></tr>
<tr><td><code id="powerEQTL.SLR_+3A_sigma.y">sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the outcome <code class="reqn">y_i</code> in simple linear regression.
</p>
</td></tr>
<tr><td><code id="powerEQTL.SLR_+3A_fwer">FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate.
</p>
</td></tr>
<tr><td><code id="powerEQTL.SLR_+3A_ntests">nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td></tr> 
<tr><td><code id="powerEQTL.SLR_+3A_n.lower">n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr> <tr><td><code id="powerEQTL.SLR_+3A_n.upper">n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To test if a SNP is associated with a gene probe, we use the simple linear regression
</p>
<p style="text-align: center;"><code class="reqn">y_i = \beta_0+\beta_1 x_i + \epsilon_i,</code>
</p>

<p>where <code class="reqn">y_i</code> is the gene expression level of the <code class="reqn">i</code>-th subject,
<code class="reqn">x_i</code> is the genotype of the <code class="reqn">i</code>-th subject, and
<code class="reqn">\epsilon_i</code> is the random error term with mean zero and standard deviation <code class="reqn">\sigma</code>. Additive coding for genotype is used. That is,
<code class="reqn">x_i=0</code> indicates wildtype homozygotes;
<code class="reqn">x_i=1</code> indicates heterozygotes; and <code class="reqn">x_i=2</code> indicates mutation heterozygotes.
</p>
<p>To test if the SNP is associated with the gene probe, we test the null hypothesis <code class="reqn">H_0: \beta_1=0</code> versus the alternative hypothesis <code class="reqn">H_1: \beta_1 = \delta</code>, where <code class="reqn">\delta\neq 0</code>.
</p>
<p>Denote <code class="reqn">\theta</code> as the minor allele frequency (MAF) of the SNP. Under Hardy-Weinberg equilibrium, we can calculate the variance of genotype of the SNP: 
<code class="reqn">\sigma^2_x=2 \theta (1-\theta)</code>, 
where <code class="reqn">\sigma^2_x</code> is the variance of the
predictor (i.e. the SNP) <code class="reqn">x_i</code>.
</p>
<p>The exact power calculation formula can be derived as
</p>
<p style="text-align: center;"><code class="reqn">1-T_{n-2, \lambda}(t_{n-2}(\alpha/2)) + T_{n-2, \lambda}(-t_{n-2}(\alpha/2)),</code>
</p>

<p>where <code class="reqn">T_{n-2, \lambda}(a)</code> is the value at <code class="reqn">a</code> of cumulative distribution function of non-central t distribution with <code class="reqn">n-2</code> degrees of freedom
and non-centrality parameter <code class="reqn">\lambda=\delta/\sqrt{\sigma^2/[(n-1)\tilde{\sigma}^2_{x}]}</code>. And <code class="reqn">\tilde{\sigma}^2_{x}=\sum_{i=1}^n(x_i - \bar{x})^2/(n-1)</code>. 
</p>
<p>Dupont and Plummer (1998) mentioned the following relationship:
</p>
<p style="text-align: center;"><code class="reqn">\sigma^2 = \sigma^2_y - \beta_1^2 \sigma^2_x.</code>
</p>

<p>So we can plug in the above equation to the power calculation formula.
</p>
<p>Under Hardy-Weinberg equilibrium, we have <code class="reqn">\sigma_x^2=2\theta(1-\theta)</code>,
where <code class="reqn">\theta</code> is the minor allele frequency (MAF).
</p>
<p>Hence, the non-centrality parameter can be rewritten as
</p>
<p style="text-align: center;"><code class="reqn">\lambda=\frac{\delta}{\sqrt{
\left(\sigma_y^2 - \delta^2 2\left(1-\hat{\theta}\right)\hat{\theta}\right)/
\left[(n-1)2\left(1-\hat{\theta}\right)\hat{\theta}\right]
}}</code>
</p>

<p>We adopted the parameters from the GTEx cohort (see the Power analysis&quot; section of Nature Genetics, 2013; https://www.nature.com/articles/ng.2653), where they modeled the expression data as having a log-normal distribution with a log standard deviation of 0.13 within each genotype class (AA, AB, BB). This level of noise is based on estimates from initial GTEx data. In their power analysis, they assumed the across-genotype difference delta = 0.13 (i.e., equivalent to detecting a log expression change similar to the standard deviation within a single genotype class). 
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>;
</p>
<p>minimum allowable MAF if the input parameter <code>MAF = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dupont, W.D. and Plummer, W.D.. 
Power and Sample Size Calculations for Studies Involving Linear Regression.
Controlled Clinical Trials. 1998;19:589-601.
</p>
<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculate power
powerEQTL.SLR(
  MAF = 0.1,
  slope = 0.13,
  n = 179,
  power = NULL,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05)
  
# calculate sample size (total number of subjects)
powerEQTL.SLR(
  MAF = 0.1,
  slope = 0.13,
  n = NULL,
  power = 0.8,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05)
  
# calculate minimum detectable slope
powerEQTL.SLR(
  MAF = 0.1,
  slope = NULL,
  n = 179,
  power = 0.8,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05)  
  
# calculate minimum allowable MAF
powerEQTL.SLR(
  MAF = NULL,
  slope = 0.13,
  n = 179,
  power = 0.8,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05) 

</code></pre>

<hr>
<h2 id='powerLME'>
Power Calculation for Simple Linear Mixed Effects Model
</h2><span id='topic+powerLME'></span>

<h3>Description</h3>

<p>Power calculation for simple linear mixed effects model.
This function can be used to calculate one of the 3 parameters (power, sample size, and minimum detectable slope) by setting the corresponding parameter
as NULL and providing values for the other 2 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerLME(
  slope, 
  n, 
  m, 
  sigma.y, 
  sigma.x,
  power = NULL,
  rho = 0.8, 
  FWER = 0.05,
  nTests = 1,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerLME_+3A_slope">slope</code></td>
<td>

<p>numeric. Slope under alternative hypothesis.
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_n">n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_m">m</code></td>
<td>

<p>integer. Number of observations per subject.
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_sigma.y">sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the outcome y.
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_sigma.x">sigma.x</code></td>
<td>

<p>numeric. Standard deviation of the predictor x.
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_power">power</code></td>
<td>

<p>numeric. Desired power.
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_rho">rho</code></td>
<td>

<p>numeric. Intra-class correlation (i.e., correlation between 
<code class="reqn">y_{ij}</code> and <code class="reqn">y_{ik}</code> for the <code class="reqn">j</code>-th and <code class="reqn">k</code>-th observations of the <code class="reqn">i</code>-th subject).
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_fwer">FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate. 
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_ntests">nTests</code></td>
<td>

<p>integer. Number of tests (e.g., number of genes in differential expression analysis based on scRNAseq to compare gene expression between diseased subjects and healthy subjects).
</p>
</td></tr>
<tr><td><code id="powerLME_+3A_n.lower">n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating toal number of subjects.
</p>
</td></tr> <tr><td><code id="powerLME_+3A_n.upper">n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We assume the following simple linear mixed effects model to characterize the association between
the predictor x and the outcome y:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \beta_{0i} + \beta_1 * x_i + \epsilon_{ij},</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">\beta_{0i} \sim N\left(\beta_0, \sigma^2_{\beta}\right),</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">\epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p><code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>, <code class="reqn">n</code> is the number of
subjects, <code class="reqn">m</code> is the number of observations per subject,
<code class="reqn">y_{ij}</code> is the outcome value for the <code class="reqn">j</code>-th observation
of the <code class="reqn">i</code>-th subject, <code class="reqn">x_i</code> is the predictor value for the
<code class="reqn">i</code>-th subject. For example, <code class="reqn">x_i</code> is the binary variable indicating if the <code class="reqn">i</code>-th subject is a diseased subject or not.
</p>
<p>We would like to test the following hypotheses:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \beta_1=0,</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">H_1: \beta_1 = \delta,</code>
</p>

<p>where <code class="reqn">\delta\neq 0</code>.
</p>
<p>We can derive the power calculation formula is
</p>
<p style="text-align: center;"><code class="reqn">power=1- \Phi\left(z_{\alpha^{*}/2}-a\times b\right)
+\Phi\left(-z_{\alpha^{*}/2} - a\times b\right),</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">a= 
\frac{\hat{\sigma}_x
}{\sigma_y}</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">
b=\frac{\delta\sqrt{m(n-1)}}{\sqrt{1+(m-1)\rho}}
</code>
</p>

<p>and <code class="reqn">z_{\alpha^{*}/2}</code> is the upper <code class="reqn">100\alpha^{*}/2</code>
percentile of the standard normal distribution,
<code class="reqn">\alpha^{*}=\alpha/nTests</code>, nTests is the number of 
tests,
<code class="reqn">\sigma_y=\sqrt{\sigma^2_{\beta}+\sigma^2}</code>,
<code class="reqn">\hat{\sigma}_x=\sqrt{\sum_{i=1}^n\left(x_i-\bar{x}\right)^2/(n-1)}</code>,
and <code class="reqn">\rho=\sigma^2_{\beta}/\left(\sigma^2_{\beta}+\sigma^2\right)</code> is the intra-class correlation.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  n = 102
  m = 227868
  
  # calculate power
  power = powerLME(
    slope = 0.6, 
    n = n, 
    m = m,
    sigma.y = 0.29, 
    sigma.x = 0.308,
    power = NULL,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 1e+6)

  print(power)
  
  # calculate sample size (total number of subjects)
  n = powerLME(
    slope = 0.6, 
    n = NULL, 
    m = m,
    sigma.y = 0.29, 
    sigma.x = 0.308,
    power = 0.9562555,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 1e+6)

  print(n)
  
  # calculate slope
  slope = powerLME(
    slope = NULL, 
    n = n, 
    m = m,
    sigma.y = 0.29, 
    sigma.x = 0.308,
    power = 0.9562555,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 1e+6)

  print(slope)
  


</code></pre>

<hr>
<h2 id='powerLMEnoCov'>
Power Calculation for Simple Linear Mixed Effects Model Without Covariate
</h2><span id='topic+powerLMEnoCov'></span>

<h3>Description</h3>

<p>Power calculation for simple linear mixed effects model without covariate.
This function can be used to calculate one of the 3 parameters (power, sample size, and minimum detectable slope) by setting the corresponding parameter
as NULL and providing values for the other 2 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerLMEnoCov(
  slope, 
  n, 
  m, 
  sigma.y,
  power = NULL,
  rho = 0.8, 
  FWER = 0.05,
  nTests = 1,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerLMEnoCov_+3A_slope">slope</code></td>
<td>

<p>numeric. Slope under alternative hypothesis.
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_n">n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_m">m</code></td>
<td>

<p>integer. Number of pairs of replicates per subject.
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_sigma.y">sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the outcome y.
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_power">power</code></td>
<td>

<p>numeric. Desired power.
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_rho">rho</code></td>
<td>

<p>numeric. Intra-class correlation (i.e., correlation between 
<code class="reqn">y_{ij}</code> and <code class="reqn">y_{ik}</code> for the <code class="reqn">j</code>-th and <code class="reqn">k</code>-th observations of the <code class="reqn">i</code>-th subject).
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_fwer">FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate. 
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_ntests">nTests</code></td>
<td>

<p>integer. Number of tests (e.g., number of genes in differential expression analysis based on scRNAseq to compare gene expression before and after treatment).
</p>
</td></tr>
<tr><td><code id="powerLMEnoCov_+3A_n.lower">n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr> <tr><td><code id="powerLMEnoCov_+3A_n.upper">n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In an experiment, there are <code class="reqn">n</code> samples. For each sample, we get 
<code class="reqn">m</code> pairs of replicates. For each pair, one replicate will receive no treatment. The other
replicate will receive treatment. The outcome is the expression of a gene. The overall goal of the experiment is to check if the treatment affects gene expression level or not. Or equivalently, the overall goal of the experiment is to test if the mean within-pair difference of gene expression is equal to zero or not. In the design stage, we would like to calculate the power/sample size of the experiment for testing if the mean within-pair difference of gene expression is equal to zero or not. 
</p>
<p>We assume the following linear mixed effects model to characterize the 
relationship between the within-pair difference of gene expression <code class="reqn">y_{ij}</code> and
the mean of the within-pair difference <code class="reqn">\beta_{0i}</code>:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \beta_{0i} + \epsilon_{ij},</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">\beta_{0i} \sim N\left(\beta_0, \sigma^2_{\beta}\right),</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">\epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p><code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>, <code class="reqn">n</code> is the number of
subjects, <code class="reqn">m</code> is the number of pairs of replicates per subject,
<code class="reqn">y_{ij}</code> is the within-pair difference of outcome value for the <code class="reqn">j</code>-th pair of the <code class="reqn">i</code>-th subject. 
</p>
<p>We would like to test the following hypotheses:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \beta_0=0,</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">H_1: \beta_0 = \delta,</code>
</p>

<p>where <code class="reqn">\delta\neq 0</code>. If we reject the null hypothesis <code class="reqn">H_0</code>
based on a sample, we then get evidence that the treatment affects
the gene expression level.
</p>
<p>We can derive the power calculation formula:
</p>
<p style="text-align: center;"><code class="reqn">power=1- \Phi\left(z_{\alpha^{*}/2}-a\times b\right)
+\Phi\left(-z_{\alpha^{*}/2} - a\times b\right),</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">a= 
\frac{1
}{\sigma_y}</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">
b=\frac{\delta\sqrt{mn}}{\sqrt{1+(m-1)\rho}}
</code>
</p>

<p>and <code class="reqn">z_{\alpha^{*}/2}</code> is the upper <code class="reqn">100\alpha^{*}/2</code>
percentile of the standard normal distribution,
<code class="reqn">\alpha^{*}=\alpha/nTests</code>, nTests is the number of 
tests,
<code class="reqn">\sigma_y=\sqrt{\sigma^2_{\beta}+\sigma^2}</code>
and <code class="reqn">\rho=\sigma^2_{\beta}/\left(\sigma^2_{\beta}+\sigma^2\right)</code> is the intra-class correlation.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  n = 17
  m = 5
  sigma.y = 0.68
  slope = 1.3*sigma.y
  print(slope)
  
  # estimate power
  power = powerLMEnoCov(
    slope = slope, 
    n = n, 
    m = m,
    sigma.y = sigma.y,
    power = NULL,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 20345)

  print(power)

  # estimate sample size (total number of subjects)
  n = powerLMEnoCov(
    slope = slope, 
    n = NULL, 
    m = m,
    sigma.y = sigma.y,
    power = 0.8721607,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 20345)

  print(n)
  
  # estimate slope
  slope = powerLMEnoCov(
    slope = NULL, 
    n = n, 
    m = m,
    sigma.y = sigma.y,
    power = 0.8721607,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 20345)

  print(slope)
</code></pre>

<hr>
<h2 id='simDat.eQTL.scRNAseq'>
Generate Gene Expression Levels Of One Gene And Genotypes Of One SNP For Subjects With Multiple Cells Based On ZINB Mixed Effects Regression Model
</h2><span id='topic+simDat.eQTL.scRNAseq'></span>

<h3>Description</h3>

<p>Generate gene expression levels of one gene and genotypes of one SNP for subjects with multiple cells based on ZINB mixed effects regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simDat.eQTL.scRNAseq(nSubj = 50, 
		     nCellPerSubj = 100, 
		     zero.p = 0.01, 
		     m.int = 0, 
		     sigma.int = 1, 
		     slope = 1, 
		     theta = 1, 
		     MAF = 0.45)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_nsubj">nSubj</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_ncellpersubj">nCellPerSubj</code></td>
<td>

<p>integer. Number of cells per subject.
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_zero.p">zero.p</code></td>
<td>

<p>numeric. Probability that an excess zero occurs.
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_m.int">m.int</code></td>
<td>

<p>numeric. Mean of random intercept (see details).
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_sigma.int">sigma.int</code></td>
<td>

<p>numeric. Standard deviation of random intercept (see details).
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_slope">slope</code></td>
<td>

<p>numeric. Slope (see details).
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_theta">theta</code></td>
<td>

<p>numeric. dispersion parameter of negative binomial distribution.
The smaller <code>theta</code> is, the larger variance of NB random variable is.
</p>
</td></tr>
<tr><td><code id="simDat.eQTL.scRNAseq_+3A_maf">MAF</code></td>
<td>

<p>numeric. Minor allele frequency of the SNP.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simulates gene expression levels of one gene and genotypes of one SNP for subjects with multiple cells based on zero-inflated negative binomial (ZINB) regression model with only one 
covariate: genotype.  
That is, the read counts of a gene follows a mixture of 2-component distributions. One component takes only one value: zero. The other component is negative binomial distribution, which takes non-negative values 0, 1, 2, .... The log mean of the negative binomial distribution is linear function of the genotype.
</p>
<p>Denote <code class="reqn">Y_{ij}</code> as the read counts for the <code class="reqn">j</code>-th cell of
the <code class="reqn">i</code>-th subject, <code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>,
<code class="reqn">n</code> is the number of subjects, and <code class="reqn">m</code> is the number of cells per subject.
</p>
<p>Denote <code class="reqn">p</code> as the probability that <code class="reqn">Y_{ij}=0</code> is an excess zero.
With probability <code class="reqn">1-p</code>, <code class="reqn">Y_{ij}</code> follows a negative binomial distribution <code class="reqn">NB(\mu, \theta)</code>, where <code class="reqn">\mu</code> is the mean (i.e., <code class="reqn">\mu=E(Y_{ij})</code>) and <code class="reqn">\theta</code> is the dispersion parameter.
The variance of the NB distribution is <code class="reqn">\mu + \mu^2/\theta</code>.
The relationship between gene expression and genotype for the <code class="reqn">i</code>-th subject is characterized by the equation 
</p>
<p style="text-align: center;"><code class="reqn">\mu_i = \exp(\beta_{0i} + \beta_{1} x_i),</code>
</p>

<p>where <code class="reqn">\beta_{0i}</code> is the random intercept following a normal
distribution <code class="reqn">N(\beta_0, \sigma^2)</code> to account for within-subject correlation of gene expression, <code class="reqn">\beta_0</code> is the mean of the random intercept, <code class="reqn">\sigma</code> is the standard deviation of the random intercept, <code class="reqn">\beta_1</code> is the slope, and <code class="reqn">x_i</code> is the additive-coded genotype for the SNP with minor allele frequency <code class="reqn">MAF</code>.
</p>
<p>We assume that the SNP satisfies the Hardy-Weinberg Equilibrium. That is, the
probabilities of the 3 genotypes <code class="reqn">(0, 1, 2)</code> are <code class="reqn">(1-MAF)^2,
		2 MAF (1-MAF), MAF^2</code>, respectively.
</p>
<p>For simplicity, we assume that excess zeros are caused by technical issues, hence are not related to genotypes.
</p>


<h3>Value</h3>

<p>A data frame with 3 columns:
</p>
<table>
<tr><td><code>id</code></td>
<td>
<p>subject id</p>
</td></tr>
<tr><td><code>geno</code></td>
<td>
<p>additive-coded genotype of the SNP</p>
</td></tr>
<tr><td><code>counts</code></td>
<td>
<p>gene expression of the gene</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class='language-R'>frame = simDat.eQTL.scRNAseq(nSubj = 5, 
		     nCellPerSubj = 3, 
		     zero.p = 0.01, 
		     m.int = 0, 
		     sigma.int = 1, 
		     slope = 1, 
		     theta = 1, 
		     MAF = 0.45)
print(dim(frame))
print(frame[1:10,])

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
