<!DOCTYPE html><html lang="en"><head><title>Help for package pPCA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pPCA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pPCA'><p>Performs a principal component analysis on a large sparse matrices or a list of large sparse</p>
matrices and returns the results as an object compatible to class prcomp</a></li>
<li><a href='#print.pPCA'><p>Print the Output of Principal Component Analysis (pPCA)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Partial Principal Component Analysis of Partitioned Large Sparse
Matrices</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-10-22</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Srika Raja &lt;sri1919@iastate.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs partial principal component analysis of a large sparse matrix. 
  The matrix may be stored as a list of matrices to be concatenated (implicitly) horizontally.
  Useful application includes cases where the number of total nonzero entries exceed the capacity of 
  32 bit integers (e.g., with large Single Nucleotide Polymorphism data).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.2), methods, RSpectra (&ge; 0.16-1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix (&ge; 1.1-0), Rcpp (&ge; 0.11.5)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggbiplot</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-22 06:14:10 UTC; srika</td>
</tr>
<tr>
<td>Author:</td>
<td>Srika Raja [aut, cre],
  Somak Dutta [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-22 06:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='pPCA'>Performs a principal component analysis on a large sparse matrices or a list of large sparse
matrices and returns the results as an object compatible to class prcomp</h2><span id='topic+pPCA'></span>

<h3>Description</h3>

<p>Performs a partial principal component analysis on a large sparse matrices or a list of large sparse
matrices and returns the results as an object compatible to class prcomp. Uses RSpectra library
to compute the largest eigenvalues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pPCA(x, rank, retX = TRUE, scale. = TRUE, normalize = FALSE, sd.tol = 1e-05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pPCA_+3A_x">x</code></td>
<td>
<p>A matrix, sparse matrix (Matrix::dgCMatrix), or a list of these. When a list
is supplied, the entries are concatenated horizontally (implicitly). See description.</p>
</td></tr>
<tr><td><code id="pPCA_+3A_rank">rank</code></td>
<td>
<p>An integer specifying the number of principal components to compute.</p>
</td></tr>
<tr><td><code id="pPCA_+3A_retx">retX</code></td>
<td>
<p>A logical value indicating whether the rotated variables (PC scores) should be returned.</p>
</td></tr>
<tr><td><code id="pPCA_+3A_scale.">scale.</code></td>
<td>
<p>A logical value indicating whether the variables should be scaled to have
unit variance before the analysis takes place.</p>
</td></tr>
<tr><td><code id="pPCA_+3A_normalize">normalize</code></td>
<td>
<p>A logical value indicating whether the principal component scores should be normalized.</p>
</td></tr>
<tr><td><code id="pPCA_+3A_sd.tol">sd.tol</code></td>
<td>
<p>A positive number, warnings are printed if the standard deviation of any
column is less than this threshold.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the input argument is a matrix (of class &quot;matrix&quot; or &quot;dgCMatrix&quot;), principal component analysis
is performed to extract a few largest components. When a list of matrices is passed, the partial PCA
is performed on the horizontally concatenated matrix, i.e., if <code>x = list(X1,X2,X3)</code> then the
partial PCA is done on the matrix [X1 X2 X3], without concatenating the matrices explicitly. This can be
useful when the matrix is so high-dimensional that the total number of non-zero entries
exceed 2^31-1 (roughly 9.33e10), the capacity of a 32 bit integer. For example, in PCA with very
high-dimensional SNP data, the sparse matrices can be stored for each chromosome within the capacity
of 32 bit integers.
</p>


<h3>Value</h3>

<p>pPCA returns a list with class &quot;pPCA&quot; (compatible with &quot;prcomp&quot;) containing the following
components:
</p>
<table role = "presentation">
<tr><td><code>sdev</code></td>
<td>
<p>A vector of the singular values (standard deviations of the principal components).</p>
</td></tr>
<tr><td><code>rotation</code></td>
<td>
<p>A matrix whose columns contain the eigenvectors (loadings).</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>A matrix of the principal component scores, returned if retX is true. This is
the centred (and scaled if requested) data multiplied by the rotation matrix.</p>
</td></tr>
<tr><td><code>center</code></td>
<td>
<p>column means.</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>column standard deviations, if scale. is true. Otherwise, FALSE.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The partial SVD is computed through the RSpectra package. All elements in the first row of the rotation
matrix are positive.
</p>


<h3>Author(s)</h3>

<p>Srika Raja and Somak Dutta
</p>


<h3>References</h3>

<p>Raja, S. and Dutta, S. (2024). Matrix-free partial PCA of partitioned genetic data.
REU project 2024, Iowa State University.
</p>
<p>Dai, F., Dutta, S., and, Maitra, R. (2020). A Matrix-Free Likelihood Method for
Exploratory Factor Analysis of High-Dimensional Gaussian Data. Journal of Computational and
Graphical Statistics, 29(3), 675&ndash;680.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+biplot">biplot</a>,<a href="stats.html#topic+prcomp">prcomp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Matrix)
set.seed(20190329)
m &lt;- rsparsematrix(50,100,density = 0.35)
results &lt;- pPCA(m, rank = 2)
biplot(results)
data &lt;- list(rsparsematrix(nrow = 50,ncol = 10,density = 0.35),
             rsparsematrix(nrow = 50,ncol = 40,density = 0.35)) # Using a list of matrices
result &lt;- pPCA(data, rank = 3)
print(result)
biplot(result)


</code></pre>

<hr>
<h2 id='print.pPCA'>Print the Output of Principal Component Analysis (pPCA)</h2><span id='topic+print.pPCA'></span>

<h3>Description</h3>

<p>Prints the output of the <code>pPCA</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pPCA'
print(x, digits = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.pPCA_+3A_x">x</code></td>
<td>
<p>An object of class <code>pPCA</code> that contains the results of a partial principal component analysis.</p>
</td></tr>
<tr><td><code id="print.pPCA_+3A_digits">digits</code></td>
<td>
<p>The number of decimal places to use in printing results such as variance explained and PC scores. Defaults to 3.</p>
</td></tr>
<tr><td><code id="print.pPCA_+3A_...">...</code></td>
<td>
<p>Further arguments passed to <code>print</code> for additional control over the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
