<!DOCTYPE html><html lang="en"><head><title>Help for package PPforest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PPforest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#baggtree'><p>For each bootstrap sample grow a projection pursuit tree (PPtree object).</p></a></li>
<li><a href='#crab'><p>Astralian crabs</p></a></li>
<li><a href='#fishcatch'><p>Fish catch data set</p></a></li>
<li><a href='#glass'><p>Glass data set</p></a></li>
<li><a href='#image'><p>The image data set</p></a></li>
<li><a href='#leukemia'><p>Leukemia data set</p></a></li>
<li><a href='#lymphoma'><p>Lymphoma data set</p></a></li>
<li><a href='#NCI60'><p>NCI60 data set</p></a></li>
<li><a href='#node_data'><p>Data structure with the  projected and boundary by node and class.</p></a></li>
<li><a href='#olive'><p>The olive data set</p></a></li>
<li><a href='#parkinson'><p>Parkinson data set</p></a></li>
<li><a href='#permute_importance'><p>Obtain the permuted importance variable measure</p></a></li>
<li><a href='#PPclassify2'><p>Predict class for the test set and calculate prediction error after finding the PPtree structure, .</p></a></li>
<li><a href='#ppf_avg_imp'><p>Global importance measure for a PPforest object as the average IMP PPtree measure over all the trees</p>
in the forest</a></li>
<li><a href='#ppf_global_imp'><p>Global importance measure for a PPforest object</p></a></li>
<li><a href='#PPforest'><p>Projection Pursuit Random Forest</p></a></li>
<li><a href='#PPtree_split'><p>Projection pursuit classification tree with random variable selection in each split</p></a></li>
<li><a href='#print.PPforest'><p>Print PPforest object</p></a></li>
<li><a href='#ternary_str'><p>Data structure with the  projected and boundary by node and class.</p></a></li>
<li><a href='#trees_pred'><p>Obtain predicted class for new data from baggtree function or PPforest</p></a></li>
<li><a href='#wine'><p>Wine data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Title:</td>
<td>Projection Pursuit Classification Forest</td>
</tr>
<tr>
<td>Author:</td>
<td>Natalia da Silva, Eun-Kyung Lee, Di Cook</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Natalia da Silva &lt;natalia.dasilva@fcea.edu.uy&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements projection pursuit forest algorithm for supervised classification.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/natydasilva/PPforest">https://github.com/natydasilva/PPforest</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.7), magrittr, plyr, dplyr (&ge; 0.7.5), tidyr,
doParallel, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, gridExtra, GGally, ggplot2, RColorBrewer, roxygen2 (&ge;
3.0.0), PPtreeViz, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp,RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-09 23:14:21 UTC; natydasilva</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-09 23:32:55 UTC</td>
</tr>
</table>
<hr>
<h2 id='baggtree'>For each bootstrap sample grow a projection pursuit tree (PPtree object).</h2><span id='topic+baggtree'></span>

<h3>Description</h3>

<p>For each bootstrap sample grow a projection pursuit tree (PPtree object).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baggtree(
  data,
  class,
  m = 500,
  PPmethod = "LDA",
  lambda = 0.1,
  size.p = 1,
  parallel = FALSE,
  cores = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="baggtree_+3A_data">data</code></td>
<td>
<p>Data frame with the complete data set.</p>
</td></tr>
<tr><td><code id="baggtree_+3A_class">class</code></td>
<td>
<p>A character with the name of the class variable.</p>
</td></tr>
<tr><td><code id="baggtree_+3A_m">m</code></td>
<td>
<p>is the number of bootstrap replicates, this corresponds with the number of trees to grow. To ensure that each observation is predicted a few times we have to select this number no too small. <code>m = 500</code> is by default.</p>
</td></tr>
<tr><td><code id="baggtree_+3A_ppmethod">PPmethod</code></td>
<td>
<p>is the projection pursuit index to be optimized, options LDA or PDA, by default it is LDA.</p>
</td></tr>
<tr><td><code id="baggtree_+3A_lambda">lambda</code></td>
<td>
<p>a parameter for PDA index</p>
</td></tr>
<tr><td><code id="baggtree_+3A_size.p">size.p</code></td>
<td>
<p>proportion of random sample variables in each split.</p>
</td></tr>
<tr><td><code id="baggtree_+3A_parallel">parallel</code></td>
<td>
<p>logical condition, if it is TRUE then  parallelize the function</p>
</td></tr>
<tr><td><code id="baggtree_+3A_cores">cores</code></td>
<td>
<p>number of cores used in the parallelization</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with trees_pp output for all the bootstraps samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set
crab.trees &lt;- baggtree(data = crab, class = 'Type',
m =  200, PPmethod = 'LDA', lambda = .1, size.p = 0.5 , parallel = TRUE, cores = 2)
str(crab.trees, max.level = 1)

</code></pre>

<hr>
<h2 id='crab'>Astralian crabs</h2><span id='topic+crab'></span>

<h3>Description</h3>

<p>Measurements on rock crabs of the genus Leptograpsus. The data set contains 200 observations
from two species of crab (blue and orange), there are 50 specimens of each sex of each species,
collected on site at Fremantle, Western Australia. 
</p>

<ul>
<li><p>Type is the class variable and has 4 classes with the combinations of specie and sex (BlueMale, BlueFemale, OrangeMale and OrangeFemale). 
</p>
</li>
<li><p>FLthe size of the frontal lobe length, in mm
</p>
</li>
<li><p>RWrear width, in mm
</p>
</li>
<li><p>CLlength of midline of the carapace, in mm
</p>
</li>
<li><p>CWmaximum width of carapace, in mm
</p>
</li>
<li><p>BDdepth of the body; for females, measured after displacement of the abdomen, in mm
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(crab)
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 6 variables
</p>


<h3>Source</h3>

<p>Campbell, N. A. &amp; Mahon, R. J. (1974), A Multivariate Study of Variation in Two Species of Rock Crab of genus Leptograpsus, Australian Journal of Zoology 22(3), 417 - 425.
</p>

<hr>
<h2 id='fishcatch'>Fish catch data set</h2><span id='topic+fishcatch'></span>

<h3>Description</h3>

<p>There are 159 fishes of 7 species are caught and measured. Altogether there are
7 variables.  All the fishes are caught from the same lake(Laengelmavesi) near Tampere in Finland.
</p>

<ul>
<li><p>Type has 7 fish classes, with 35 cases of Bream, 11 cases of Parkki, 56 cases of Perch 17 cases of Pike, 20 cases of Roach, 14 cases of Smelt and 6 cases of Whitewish. 
</p>
</li>
<li><p>weight Weight of the fish (in grams)
</p>
</li>
<li><p>length1 Length from the nose to the beginning of the tail (in cm)
</p>
</li>
<li><p>length2 Length from the nose to the notch of the tail (in cm)
</p>
</li>
<li><p>length3 Length from the nose to the end of the tail (in cm)
</p>
</li>
<li><p>height Maximal height as % of Length3
</p>
</li>
<li><p>width Maximal width as % of Length3
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(fishcatch)
</code></pre>


<h3>Format</h3>

<p>A data frame with 159 rows and 7 variables
</p>


<h3>Source</h3>

<p>urlhttp://www.amstat.org/publications/jse/jse_data_archive.htm
</p>

<hr>
<h2 id='glass'>Glass data set</h2><span id='topic+glass'></span>

<h3>Description</h3>

<p>Contains measurements 214 observations of 6 types of glass; defined in terms of their oxide content.
</p>

<ul>
<li><p>Type has 6 types of glasses
</p>
</li>
<li><p>X1 refractive index 
</p>
</li>
<li><p>X2 Sodium (unit measurement: weight percent in corresponding oxide). 
</p>
</li>
<li><p>X3 Magnesium
</p>
</li>
<li><p>X4 Aluminum
</p>
</li>
<li><p>X5 Silicon
</p>
</li>
<li><p>X6 Potassium
</p>
</li>
<li><p>X7 Calcium
</p>
</li>
<li><p>X8 Barium
</p>
</li>
<li><p>X9 Iron
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(glass)
</code></pre>


<h3>Format</h3>

<p>A data frame with 214 rows and 10 variables
</p>

<hr>
<h2 id='image'>The image data set</h2><span id='topic+image'></span>

<h3>Description</h3>

<p>contains  2310 observations of instances from 7 outdoor images
</p>

<ul>
<li><p>Type has 7 types of outdoor images, brickface, cement,  foliage, grass, path, sky, and window.
</p>
</li>
<li><p>X1 the column of the center pixel of the region 
</p>
</li>
<li><p>X2 the row of the center pixel of the region. 
</p>
</li>
<li><p>X3 the number of pixels in a region = 9. 
</p>
</li>
<li><p>X4 the results of a line extraction algorithm that counts how many lines of length 5 (any orientation) with low contrast, less than or equal to 5, go through the region.
</p>
</li>
<li><p>X5 measure the contrast of horizontally adjacent pixels in the region. There are 6, the mean and standard deviation are given. This attribute is used as a vertical edge detector.
</p>
</li>
<li><p>X6 X5 sd
</p>
</li>
<li><p>X7 measures the contrast of vertically adjacent pixels. Used for horizontal line detection. 
</p>
</li>
<li><p>X8 sd X7
</p>
</li>
<li><p>X9 the average over the region of (R + G + B)/3
</p>
</li>
<li><p>X10 the average over the region of the R value.
</p>
</li>
<li><p>X11 the average over the region of the B value.
</p>
</li>
<li><p>X12 the average over the region of the G value.
</p>
</li>
<li><p>X13 measure the excess red: (2R - (G + B))
</p>
</li>
<li><p>X14 measure the excess blue: (2B - (G + R))
</p>
</li>
<li><p>X15 measure the excess green: (2G - (R + B))
</p>
</li>
<li><p>X16 3-d nonlinear transformation of RGB. (Algorithm can be found in Foley and VanDam, Fundamentals of Interactive Computer Graphics)
</p>
</li>
<li><p>X17 mean of X16
</p>
</li>
<li><p>X18 hue  mean
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(image)
</code></pre>


<h3>Format</h3>

<p>A data frame contains 2310 observations and 19 variables
</p>

<hr>
<h2 id='leukemia'>Leukemia data set</h2><span id='topic+leukemia'></span>

<h3>Description</h3>

<p>This dataset comes from a study of gene expression in two types of acute leukemias, acute lymphoblastic leukemia (ALL) and acute myeloid leukemia (AML). Gene expression levels were measured using Affymetrix high density oligonucleotide arrays containing 6817 human genes. A data set containing 72 observations from 3 leukemia types classes. 
</p>

<ul>
<li><p>Type has 3 classes with 38 cases of B-cell ALL, 25 cases of AML and 9 cases of T-cell ALL. 
</p>
</li>
<li><p>Gene1 to Gen 40 gene expression levels
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(leukemia)
</code></pre>


<h3>Format</h3>

<p>A data frame with 72 rows and 41 variables
</p>


<h3>Source</h3>

<p>Dudoit, S., Fridlyand, J. and Speed, T. P. (2002). Comparison of Discrimination Methods for the Classification of Tumors Using Gene Expression Data. Journal of the American statistical Association 97 77-87.
</p>

<hr>
<h2 id='lymphoma'>Lymphoma data set</h2><span id='topic+lymphoma'></span>

<h3>Description</h3>

<p>Gene expression in the three most prevalent adult lymphoid malignancies: B-cell chronic lymphocytic leukemia (B-CLL), follicular lymphoma (FL), and diffuse large B-cell lym- phoma (DLBCL). Gene expression levels were measured using a specialized cDNA microarray, the Lymphochip, containing genes that are preferentially expressed in lymphoid cells or that are of known immunologic or oncologic importance. This data set contain 80 observations from 3 lymphoma types.
</p>

<ul>
<li><p>Type Class variable has 3 classes with 29 cases of B-cell ALL (B-CLL), 42 cases of diffuse large B-cell lymphoma (DLBCL) and 9 cases of follicular lymphoma (FL). 
</p>
</li>
<li><p>Gene1 to Gen 50gene expression
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(lymphoma)
</code></pre>


<h3>Format</h3>

<p>A data frame with 80 rows and 51 variables
</p>


<h3>Source</h3>

<p>Dudoit, S., Fridlyand, J. and Speed, T. P. (2002). Comparison of Discrimination Methods for the Classification of Tumors Using Gene Ex- pression Data. Journal of the American statistical Association 97 77-87.
</p>

<hr>
<h2 id='NCI60'>NCI60 data set</h2><span id='topic+NCI60'></span>

<h3>Description</h3>

<p>cDNA microarrays were used to examine the variation in gene expression among the 60 cell lines.  The cell lines are derived from tumors with different sites of origin. This data set contain 61 observations and 30 feature variables from 8 different tissue types.
</p>

<ul>
<li><p>Type has 8 different tissue types, 9 cases of breast, 5 cases of central nervous system (CNS), 7 cases pf colon, 8 cases of leukemia, 8 cases of melanoma, 9 cases of  non-small-cell lung carcinoma (NSCLC), 6 cases of ovarian and 9 cases of renal. 
</p>
</li>
<li><p>Gene1 to Gen 30 gene expression information
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(NCI60)
</code></pre>


<h3>Format</h3>

<p>A data frame with 61 rows and 31 variables
</p>


<h3>Source</h3>

<p>Dudoit, S., Fridlyand, J. and Speed, T. P. (2002). Comparison of Discrimination Methods for the Classification of Tumors Using Gene Expression Data. Journal of the American statistical Association 97 77-87.
</p>

<hr>
<h2 id='node_data'>Data structure with the  projected and boundary by node and class.</h2><span id='topic+node_data'></span>

<h3>Description</h3>

<p>Data structure with the  projected and boundary by node and class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>node_data(ppf, tr, Rule = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="node_data_+3A_ppf">ppf</code></td>
<td>
<p>is a PPforest object</p>
</td></tr>
<tr><td><code id="node_data_+3A_tr">tr</code></td>
<td>
<p>numerical value to  identify a tree</p>
</td></tr>
<tr><td><code id="node_data_+3A_rule">Rule</code></td>
<td>
<p>split rule 1:mean of two group means, 2:weighted mean, 3: mean of max(left group) and min(right group), 4: weighted mean of max(left group) and min(right group)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with projected data for each class and node id and the boundaries
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set with all the observations used as training

pprf.crab &lt;- PPforest(data = crab, std =TRUE, class = 'Type',
 size.tr = 1, m = 200, size.p = .5, PPmethod = 'LDA')
node_data(ppf = pprf.crab, tr = 1) 

</code></pre>

<hr>
<h2 id='olive'>The olive data set</h2><span id='topic+olive'></span>

<h3>Description</h3>

<p>contains  572 observations and 10 variables
</p>

<ul>
<li><p>RegionThree super-classes of Italy: North, South and the island of Sardinia 
</p>
</li>
<li><p>area Nine collection areas: three from North, four from South and 2 from Sardinia 
</p>
</li>
<li><p>palmitic fatty acids percent x 100
</p>
</li>
<li><p>palmitoleic fatty acids percent x 100 
</p>
</li>
<li><p>stearicfatty acids percent x 100
</p>
</li>
<li><p>oleicfatty acids percent x 100 
</p>
</li>
<li><p>linoleicfatty acids percent x 100
</p>
</li>
<li><p>linolenicfatty acids percent x 100  
</p>
</li>
<li><p>arachidic fatty acids percent x 100
</p>
</li>
<li><p>eicosenoic fatty acids percent x 100
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(olive)
</code></pre>


<h3>Format</h3>

<p>A data frame contains 573 observations and 10 variables
</p>

<hr>
<h2 id='parkinson'>Parkinson data set</h2><span id='topic+parkinson'></span>

<h3>Description</h3>

<p>A data set containing 195 observations from 2 parkinson types.
</p>

<ul>
<li><p>Type Class variable has 2 classes, there are 48 cases of healthy people and 147 cases with Parkinson. The feature variables are biomedical voice measures. 
</p>
</li>
<li><p>X1 Average vocal fundamental frequency
</p>
</li>
<li><p>X2 Maximum vocal fundamental frequency 
</p>
</li>
<li><p>X3 Minimum vocal fundamental frequency
</p>
</li>
<li><p>X4 MDVP:Jitter(%) measures of variation in fundamental frequency
</p>
</li>
<li><p>X5 MDVP:Jitter(Abs) measures of variation in fundamental frequency
</p>
</li>
<li><p>X6 MDVP:RAP measures of variation in fundamental frequency
</p>
</li>
<li><p>X7 MDVP:PPQ measures of variation in fundamental frequency
</p>
</li>
<li><p>X8 Jitter:DDP measures of variation in fundamental frequency
</p>
</li>
<li><p>X9 MDVP:Shimmer measures of variation in amplitude
</p>
</li>
<li><p>X10 MDVP:Shimmer(dB) measures of variation in amplitude
</p>
</li>
<li><p>X11 Shimmer:APQ3 measures of variation in amplitude
</p>
</li>
<li><p>X12 Shimmer:APQ5 measures of variation in amplitude
</p>
</li>
<li><p>X13 MDVP:APQ measures of variation in amplitude
</p>
</li>
<li><p>X14 Shimmer:DDA measures of variation in amplitude
</p>
</li>
<li><p>X15 NHR measures of ratio of noise to tonal components in the voice
</p>
</li>
<li><p>X16 HNR measures of ratio of noise to tonal components in the voice
</p>
</li>
<li><p>X17 RPDE nonlinear dynamical complexity measures
</p>
</li>
<li><p>X18 D2 nonlinear dynamical complexity measures
</p>
</li>
<li><p>X19 DFA - Signal fractal scaling exponent
</p>
</li>
<li><p>X20 spread1 Nonlinear measures of fundamental frequency variation
</p>
</li>
<li><p>X21 spread2 Nonlinear measures of fundamental frequency variation
</p>
</li>
<li><p>X22 PPE Nonlinear measures of fundamental frequency variation
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(parkinson)
</code></pre>


<h3>Format</h3>

<p>A data frame with 195 rows and 23 variables
</p>


<h3>Source</h3>

<p>urlhttps://archive.ics.uci.edu/ml/datasets/Parkinsons
</p>

<hr>
<h2 id='permute_importance'>Obtain the permuted importance variable measure</h2><span id='topic+permute_importance'></span>

<h3>Description</h3>

<p>Obtain the permuted importance variable measure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permute_importance(ppf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permute_importance_+3A_ppf">ppf</code></td>
<td>
<p>is a PPforest object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with permuted importance measures, imp is the permuted importance measure defined in Brieman paper,
imp2 is the permuted importance measure defined in randomForest package, the standard deviation (sd.im and sd.imp2) for each measure is computed and the 
also the standardized mesure.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pprf.crab &lt;- PPforest(data = crab, class = 'Type',
std = TRUE, size.tr = 1, m = 100, size.p = .4, PPmethod = 'LDA', parallel = TRUE, core = 2)
permute_importance(ppf = pprf.crab) 

</code></pre>

<hr>
<h2 id='PPclassify2'>Predict class for the test set and calculate prediction error after finding the PPtree structure, .</h2><span id='topic+PPclassify2'></span>

<h3>Description</h3>

<p>Predict class for the test set and calculate prediction error after finding the PPtree structure, .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PPclassify2( Tree.result, test.data = NULL, Rule = 1, true.class = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PPclassify2_+3A_tree.result">Tree.result</code></td>
<td>
<p>the result of PP.Tree</p>
</td></tr>
<tr><td><code id="PPclassify2_+3A_test.data">test.data</code></td>
<td>
<p>the test dataset</p>
</td></tr>
<tr><td><code id="PPclassify2_+3A_rule">Rule</code></td>
<td>
<p>split rule 1:mean of two group means, 2:weighted mean, 3: mean of max(left group) and min(right group), 4: weighted mean of max(left group) and min(right group)</p>
</td></tr>
<tr><td><code id="PPclassify2_+3A_true.class">true.class</code></td>
<td>
<p>true class of test dataset if available</p>
</td></tr>
</table>


<h3>Value</h3>

<p>predict.class predicted class
</p>
<p>predict.error prediction error
</p>


<h3>References</h3>

<p>Lee, YD, Cook, D., Park JW, and Lee, EK(2013) 
PPtree: Projection pursuit classification tree, 
Electronic Journal of Statistics, 7:1369-1386.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set

Tree.crab &lt;- PPtree_split('Type~.', data = crab, PPmethod = 'LDA', size.p = 0.5)
Tree.crab

PPclassify2(Tree.crab)

</code></pre>

<hr>
<h2 id='ppf_avg_imp'>Global importance measure for a PPforest object as the average IMP PPtree measure over all the trees 
in the forest</h2><span id='topic+ppf_avg_imp'></span>

<h3>Description</h3>

<p>Global importance measure for a PPforest object as the average IMP PPtree measure over all the trees 
in the forest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppf_avg_imp(ppf, class)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppf_avg_imp_+3A_ppf">ppf</code></td>
<td>
<p>is a PPforest object</p>
</td></tr>
<tr><td><code id="ppf_avg_imp_+3A_class">class</code></td>
<td>
<p>A character with the name of the class variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with the global importance measure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set with all the observations used as training

pprf.crab &lt;- PPforest(data = crab, std =TRUE, class = 'Type',
 size.tr = 1, m = 100, size.p = .5, PPmethod = 'LDA')
 ppf_avg_imp(pprf.crab, 'Type') 
 
</code></pre>

<hr>
<h2 id='ppf_global_imp'>Global importance measure for a PPforest object</h2><span id='topic+ppf_global_imp'></span>

<h3>Description</h3>

<p>Global importance measure for a PPforest object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppf_global_imp(data, class, ppf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppf_global_imp_+3A_data">data</code></td>
<td>
<p>Data frame with the complete data set.</p>
</td></tr>
<tr><td><code id="ppf_global_imp_+3A_class">class</code></td>
<td>
<p>A character with the name of the class variable.</p>
</td></tr>
<tr><td><code id="ppf_global_imp_+3A_ppf">ppf</code></td>
<td>
<p>is a PPforest object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with the global importance measure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set with all the observations used as training

pprf.crab &lt;- PPforest(data = crab, std = TRUE, class = 'Type',
 size.tr = 1, m = 200, size.p = .5, PPmethod = 'LDA', parallel = TRUE, cores = 2)
 
ppf_global_imp(data = crab, class = 'Type', pprf.crab) 

</code></pre>

<hr>
<h2 id='PPforest'>Projection Pursuit Random Forest</h2><span id='topic+PPforest'></span>

<h3>Description</h3>

<p><code>PPforest</code> implements a random forest using projection pursuit trees algorithm (based on PPtreeViz package).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PPforest(data, class, std = TRUE, size.tr, m, PPmethod, size.p,
 lambda = .1, parallel = FALSE, cores = 2, rule = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PPforest_+3A_data">data</code></td>
<td>
<p>Data frame with the complete data set.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_class">class</code></td>
<td>
<p>A character with the name of the class variable.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_std">std</code></td>
<td>
<p>if TRUE standardize the data set, needed to compute global importance measure.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_size.tr">size.tr</code></td>
<td>
<p>is the size proportion of the training if we want to split the data in training and test.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_m">m</code></td>
<td>
<p>is the number of bootstrap replicates, this corresponds with the number of trees to grow. To ensure that each observation is predicted a few times we have to select this number no too small. <code>m = 500</code> is by default.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_ppmethod">PPmethod</code></td>
<td>
<p>is the projection pursuit index to optimize in each classification tree. The options are <code>LDA</code> and <code>PDA</code>, linear discriminant and penalized linear discriminant. By default it is <code>LDA</code>.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_size.p">size.p</code></td>
<td>
<p>proportion of variables randomly sampled in each split.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_lambda">lambda</code></td>
<td>
<p>penalty parameter in PDA index and is between 0 to 1 . If <code>lambda = 0</code>, no penalty parameter is added and the PDA index is the same as LDA index. If <code>lambda = 1</code> all variables are treated as uncorrelated. The default value is <code>lambda = 0.1</code>.</p>
</td></tr>
<tr><td><code id="PPforest_+3A_parallel">parallel</code></td>
<td>
<p>logical condition, if it is TRUE then  parallelize the function</p>
</td></tr>
<tr><td><code id="PPforest_+3A_cores">cores</code></td>
<td>
<p>number of cores used in the parallelization</p>
</td></tr>
<tr><td><code id="PPforest_+3A_rule">rule</code></td>
<td>
<p>split rule 1: mean of two group means 2: weighted mean of two group means - weight with group size 3: weighted mean of two group means - weight with group sd 4: weighted mean of two group means - weight with group se 5: mean of two group medians 6: weighted mean of two group medians - weight with group size 7: weighted mean of two group median - weight with group IQR 8: weighted mean of two group median - weight with group IQR and size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>PPforest</code> with components.
</p>
<table role = "presentation">
<tr><td><code>prediction.training</code></td>
<td>
<p>predicted values for training data set.</p>
</td></tr>
<tr><td><code>training.error</code></td>
<td>
<p>error of the training data set.</p>
</td></tr>
<tr><td><code>prediction.test</code></td>
<td>
<p>predicted values for the test data set if <code>testap = TRUE</code>(default).</p>
</td></tr>
<tr><td><code>error.test</code></td>
<td>
<p>error of the test data set if <code>testap = TRUE</code>(default).</p>
</td></tr>
<tr><td><code>oob.error.forest</code></td>
<td>
<p>out of bag error in the forest.</p>
</td></tr>
<tr><td><code>oob.error.tree</code></td>
<td>
<p>out of bag error for each tree in the forest.</p>
</td></tr>
<tr><td><code>boot.samp</code></td>
<td>
<p>information of bootrap samples.</p>
</td></tr>
<tr><td><code>output.trees</code></td>
<td>
<p>output from a <code>trees_pp</code> for each bootrap sample.</p>
</td></tr>
<tr><td><code>proximity</code></td>
<td>
<p>Proximity matrix, if two cases are classified in the same terminal node then the proximity matrix is increased by one in <code>PPforest</code> there are one terminal node per class.</p>
</td></tr>
<tr><td><code>votes</code></td>
<td>
<p> a matrix with one row for each input data point and one column for each class, giving the fraction of (OOB) votes from the <code>PPforest</code>.</p>
</td></tr>
<tr><td><code>n.tree</code></td>
<td>
<p>number of trees grown in <code>PPforest</code>.</p>
</td></tr>
<tr><td><code>n.var</code></td>
<td>
<p>number of predictor variables selected to use for spliting at each node.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>classification.</p>
</td></tr>
<tr><td><code>confusion</code></td>
<td>
<p>confusion matrix of the prediction (based on OOB data).</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the original call to <code>PPforest</code>.</p>
</td></tr>
<tr><td><code>train</code></td>
<td>
<p>is the training data based on <code>size.tr</code> sample proportion</p>
</td></tr>
<tr><td><code>test</code></td>
<td>
<p>is the test data based on <code>1-size.tr</code> sample proportion</p>
</td></tr>
</table>


<h3>References</h3>

<p>Natalia da Silva, Dianne Cook &amp; Eun-Kyung Lee (2021) 
A Projection Pursuit Forest Algorithm for Supervised Classification, 
Journal of Computational and Graphical Statistics,
DOI: 10.1080/10618600.2020.1870480
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab example with all the observations used as training

pprf.crab &lt;- PPforest(data = crab, class = 'Type',
 std = FALSE, size.tr = 1, m = 200, size.p = .5, 
 PPmethod = 'LDA' , parallel = TRUE, cores = 2, rule=1)
pprf.crab

</code></pre>

<hr>
<h2 id='PPtree_split'>Projection pursuit classification tree with random variable selection in each split</h2><span id='topic+PPtree_split'></span>

<h3>Description</h3>

<p>Find tree structure using various projection pursuit indices of classification in each split.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PPtree_split(form, data, PPmethod='LDA', 
size.p=1,  lambda = 0.1,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PPtree_split_+3A_form">form</code></td>
<td>
<p>A character with the name of the class variable.</p>
</td></tr>
<tr><td><code id="PPtree_split_+3A_data">data</code></td>
<td>
<p>Data frame with the complete data set.</p>
</td></tr>
<tr><td><code id="PPtree_split_+3A_ppmethod">PPmethod</code></td>
<td>
<p>index to use for projection pursuit: 'LDA', 'PDA'</p>
</td></tr>
<tr><td><code id="PPtree_split_+3A_size.p">size.p</code></td>
<td>
<p>proportion of variables randomly sampled in each split, default is 1, returns a PPtree.</p>
</td></tr>
<tr><td><code id="PPtree_split_+3A_lambda">lambda</code></td>
<td>
<p>penalty parameter in PDA index and is between 0 to 1 . If <code>lambda = 0</code>, no penalty parameter is added and the PDA index is the same as LDA index. If <code>lambda = 1</code> all variables are treated as uncorrelated. The default value is <code>lambda = 0.1</code>.</p>
</td></tr>
<tr><td><code id="PPtree_split_+3A_...">...</code></td>
<td>
<p>arguments to be passed to methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>PPtreeclass</code> with components
</p>
<table role = "presentation">
<tr><td><code>Tree.Struct</code></td>
<td>
<p>Tree structure of projection pursuit classification tree</p>
</td></tr>
<tr><td><code>projbest.node</code></td>
<td>
<p>1-dim optimal projections of each split node</p>
</td></tr>
<tr><td><code>splitCutoff.node</code></td>
<td>
<p>cutoff values of each split node</p>
</td></tr>
<tr><td><code>origclass</code></td>
<td>
<p>original class</p>
</td></tr> 
<tr><td><code>origdata</code></td>
<td>
<p>original data</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lee, YD, Cook, D., Park JW, and Lee, EK (2013) 
PPtree: Projection pursuit classification tree, 
Electronic Journal of Statistics, 7:1369-1386.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set

Tree.crab &lt;- PPtree_split('Type~.', data = crab, PPmethod = 'LDA', size.p = 0.5)
Tree.crab

</code></pre>

<hr>
<h2 id='print.PPforest'>Print PPforest object</h2><span id='topic+print.PPforest'></span>

<h3>Description</h3>

<p>Print PPforest object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PPforest'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.PPforest_+3A_x">x</code></td>
<td>
<p>is a PPforest class object</p>
</td></tr>
<tr><td><code id="print.PPforest_+3A_...">...</code></td>
<td>
<p>additional parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>printed results for PPforest object
</p>

<hr>
<h2 id='ternary_str'>Data structure with the  projected and boundary by node and class.</h2><span id='topic+ternary_str'></span>

<h3>Description</h3>

<p>Data structure with the  projected and boundary by node and class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ternary_str(ppf, id, sp, dx, dy)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ternary_str_+3A_ppf">ppf</code></td>
<td>
<p>is a PPforest object</p>
</td></tr>
<tr><td><code id="ternary_str_+3A_id">id</code></td>
<td>
<p>is a vector with the selected projection directions</p>
</td></tr>
<tr><td><code id="ternary_str_+3A_sp">sp</code></td>
<td>
<p>is the simplex dimensions, if k is the number of classes sp = k - 1</p>
</td></tr>
<tr><td><code id="ternary_str_+3A_dx">dx</code></td>
<td>
<p>first direction included in id</p>
</td></tr>
<tr><td><code id="ternary_str_+3A_dy">dy</code></td>
<td>
<p>second direction included in id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame needed to visualize a ternary plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#crab data set with all the observations used as training
pprf.crab &lt;- PPforest(data = crab, std =TRUE, class = "Type",
 size.tr = 1, m = 100, size.p = .5, PPmethod = 'LDA')
 require(dplyr)
pl_ter &lt;- function(dat, dx, dy ){
  p1  &lt;- dat[[1]] %&gt;% dplyr::filter(pair %in% paste(dx, dy, sep = "-") ) %&gt;%
    dplyr::select(Class, x, y) %&gt;%
    ggplot2::ggplot(ggplot2::aes(x, y, color = Class)) +
    ggplot2::geom_segment(data = dat[[2]], ggplot2::aes(x = x1, xend = x2,
                                               y = y1, yend = y2), color = "black" ) +
    ggplot2::geom_point(size = I(3), alpha = .5) +
    ggplot2::labs(y = " ",  x = " ") +
    ggplot2::theme(legend.position = "none", aspect.ratio = 1) +
    ggplot2::scale_colour_brewer(type = "qual", palette = "Dark2") +
    ggplot2::labs(x = paste0("T", dx, " "), y = paste0("T", dy, " ")) +
    ggplot2::theme(aspect.ratio = 1)
  p1
}
#ternary plot in tree different selected dierections
 pl_ter(ternary_str(pprf.crab, id = c(1, 2, 3), sp = 3, dx = 1, dy = 2), 1, 2 )

</code></pre>

<hr>
<h2 id='trees_pred'>Obtain predicted class for new data from baggtree function or PPforest</h2><span id='topic+trees_pred'></span>

<h3>Description</h3>

<p>Obtain predicted class for new data from baggtree function or PPforest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trees_pred(object, xnew, parallel = FALSE, cores = 2, rule = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trees_pred_+3A_object">object</code></td>
<td>
<p>Projection pursuit classification forest structure from PPforest or baggtree</p>
</td></tr>
<tr><td><code id="trees_pred_+3A_xnew">xnew</code></td>
<td>
<p>data frame with explicative variables used to get new predicted values.</p>
</td></tr>
<tr><td><code id="trees_pred_+3A_parallel">parallel</code></td>
<td>
<p>logical condition, if it is TRUE then  parallelize the function</p>
</td></tr>
<tr><td><code id="trees_pred_+3A_cores">cores</code></td>
<td>
<p>number of cores used in the parallelization</p>
</td></tr>
<tr><td><code id="trees_pred_+3A_rule">rule</code></td>
<td>
<p>split rule 1: mean of two group means 2: weighted mean of two group means - weight with group size 3: weighted mean of two group means - weight with group sd 4: weighted mean of two group means - weight with group se 5: mean of two group medians 6: weighted mean of two group medians - weight with group size 7: weighted mean of two group median - weight with group IQR 8: weighted mean of two group median - weight with group IQR and size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>predicted values from PPforest or baggtree
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
crab.trees &lt;- baggtree(data = crab, class = 'Type', 
m =  200, PPmethod = 'LDA', lambda = .1, size.p = 0.4 )
 
pr &lt;- trees_pred(  crab.trees,xnew = crab[, -1], parallel= FALSE, cores = 2)

pprf.crab &lt;- PPforest(data = crab, class = 'Type',
 std = FALSE, size.tr = 2/3, m = 100, size.p = .4, PPmethod = 'LDA', parallel = TRUE )
 
trees_pred(pprf.crab, xnew = pprf.crab$test ,parallel = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='wine'>Wine data set</h2><span id='topic+wine'></span>

<h3>Description</h3>

<p>A data set containing 178 observations from 3 wine grown cultivares in Italy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(wine)
</code></pre>


<h3>Format</h3>

<p>A data frame with 178 rows and 14 variables
</p>


<h3>Details</h3>


<ul>
<li><p>Type Class variable has 3 classes that are 3 different wine grown cultivares in Italy.  
</p>
</li>
<li><p>X1 to X13Check vbles
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
