<!DOCTYPE html><html><head><title>Help for package PPMR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PPMR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Exampleindividual'><p>Individual level dataset</p></a></li>
<li><a href='#Examplesummary'><p>Summary level dataset</p></a></li>
<li><a href='#PMR_individual'><p>PMR model with individual level data</p></a></li>
<li><a href='#PMR_summary'><p>PMR model with summary data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Probabilistic Two Sample Mendelian Randomization</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Efficient statistical inference of two-sample MR (Mendelian Randomization) analysis. 
    It can account for the correlated instruments and the horizontal pleiotropy, 
    and can provide the accurate estimates of both causal effect and horizontal 
    pleiotropy effect as well as the two corresponding p-values. There are two main 
    functions in the 'PPMR' package. One is PMR_individual() for individual level data, 
    the other is PMR_summary() for summary data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1.9000</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/umich-biostatistics/PPMR/issues">https://github.com/umich-biostatistics/PPMR/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-08-08 13:46:00 UTC; mkleinsa</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhongshang Yuan [aut],
  Xiang Zhou [aut],
  Michael Kleinsasser [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Kleinsasser &lt;mkleinsa@umich.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-08-09 10:20:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='Exampleindividual'>Individual level dataset</h2><span id='topic+Exampleindividual'></span>

<h3>Description</h3>

<p>A simulated individual level dataset for PMR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Exampleindividual
</code></pre>


<h3>Format</h3>

<p>A list contains the following objects:
</p>

<dl>
<dt>zx</dt><dd><p>the standardized genotype matrix for 465 individuals and 50 cis-SNPs in eQTL data.</p>
</dd>
<dt>zy</dt><dd><p>the standardized genotype matrix for 2000 individuals and 50 cis-SNPs in GWAS data.</p>
</dd>
<dt>x</dt><dd><p>the standarized gene expression vector for 465 individuals in eQL data.</p>
</dd>
<dt>y</dt><dd><p>the standarized complex trait vector for 2000 individuals in GWAS data.</p>
</dd>
</dl>

<hr>
<h2 id='Examplesummary'>Summary level dataset</h2><span id='topic+Examplesummary'></span>

<h3>Description</h3>

<p>A simulated summary level dataset for PMR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Examplesummary
</code></pre>


<h3>Format</h3>

<p>A list contains the following objects:
</p>

<dl>
<dt>betax</dt><dd><p>the cis-SNP effect size vector for one specific gene in eQTL data.</p>
</dd>
<dt>betay</dt><dd><p>the cis-SNP effect size vector for one specific gene in GWAS data.</p>
</dd>
<dt>Sigma1</dt><dd><p>the LD matrix in eQTL data.</p>
</dd>
<dt>Sigma2</dt><dd><p>the LD matrix in GWAS data.</p>
</dd>
<dt>n1</dt><dd><p>the sample size of eQTL data.</p>
</dd>
<dt>n2</dt><dd><p>the sample size of GWAS data.</p>
</dd>
</dl>

<hr>
<h2 id='PMR_individual'>PMR model with individual level data</h2><span id='topic+PMR_individual'></span>

<h3>Description</h3>

<p>Fit the probabilistic MR model with individual level data while accounting for 
the correlated instruments and horizontal pleiotropy in TWAS framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PMR_individual(yin, zin, x1in, x2in, gammain, alphain, max_iterin, epsin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PMR_individual_+3A_yin">yin</code></td>
<td>
<p>standardized exposure vector (e.g. gene expression in TWAS).</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_zin">zin</code></td>
<td>
<p>standardized complex trait vector.</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_x1in">x1in</code></td>
<td>
<p>standardized cis-genotype matrix in eQTL data.</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_x2in">x2in</code></td>
<td>
<p>standardized cis-genotype matrix in GWAS data.</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_gammain">gammain</code></td>
<td>
<p>indicator variable for constrained PMR model, with 1 for the null 
model that there is no horizontal pleiotropy.</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_alphain">alphain</code></td>
<td>
<p>indicator variable for constrained PMR model, with 1 for the null 
model that there is no causal effect.</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_max_iterin">max_iterin</code></td>
<td>
<p>The maximum iteration.</p>
</td></tr>
<tr><td><code id="PMR_individual_+3A_epsin">epsin</code></td>
<td>
<p>The convergence tolerance of the absolute value of the difference 
between the nth and (n+1)th log likelihood.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of estimates of model parameters, including the causal effect 
<code>alpha</code>, the horizontal pleiotropy effect <code>gamma</code>, and the two 
corresponding p values
</p>


<h3>Author(s)</h3>

<p>Zhongshang Yuan, Xiang Zhou.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Exampleindividual)
attach(Exampleindividual)
fmH1 = PMR_individual(yin=x, zin=y, x1in=zx, x2in=zy,
                      gammain=0,alphain = 0,max_iterin =1000,epsin=1e-5)
alpha&lt;-fmH1$alpha
gamma&lt;-fmH1$gamma
fmH0gamma = PMR_individual(yin=x,zin= y, x1in=zx, x2in=zy,gammain=1, 
                           alphain = 0,max_iterin =1000,epsin=1e-5)
fmH0alpha = PMR_individual(yin=x, zin=y, x1in=zx, x2in=zy,gammain=0,
                           alphain = 1,max_iterin =1000, epsin=1e-5)
loglikH1=max(fmH1$loglik,na.rm=TRUE)
loglikH0gamma=max(fmH0gamma$loglik,na.rm=TRUE)
loglikH0alpha=max(fmH0alpha$loglik,na.rm=TRUE)
stat_alpha = 2 * (loglikH1 - loglikH0alpha)
pvalue_alpha = pchisq(stat_alpha,1,lower.tail=FALSE)
stat_gamma = 2 * (loglikH1 - loglikH0gamma)
pvalue_gamma = pchisq(stat_gamma,1,lower.tail=FALSE)

</code></pre>

<hr>
<h2 id='PMR_summary'>PMR model with summary data</h2><span id='topic+PMR_summary'></span>

<h3>Description</h3>

<p>Fit the probabilistic MR model with summary data while accounting for the 
correlated instruments and horizontal pleiotropy in TWAS framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PMR_summary(betaxin, betayin, Sigma1sin, Sigma2sin, samplen1, samplen2,
  gammain, alphain, max_iterin, epsin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PMR_summary_+3A_betaxin">betaxin</code></td>
<td>
<p>the cis-SNP effect size vector for one specific gene in eQTL 
data, which must be calculated based on both the standardized gene expression 
value and the standardized cis-genotype matrix.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_betayin">betayin</code></td>
<td>
<p>the cis-SNP effect size vector for one specific gene in GWAS 
data, which be calculated based on both the standardized complex trait value 
and the standardized cis-genotype matrix.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_sigma1sin">Sigma1sin</code></td>
<td>
<p>the LD matrix in eQTL data.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_sigma2sin">Sigma2sin</code></td>
<td>
<p>the LD matrix in GWAS data. Both <code>Sigma2sin</code> and 
<code>sigma1sin</code> are often the same from the reference panel.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_samplen1">samplen1</code></td>
<td>
<p>the sample size of eQTL data.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_samplen2">samplen2</code></td>
<td>
<p>the sample size of GWAS data.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_gammain">gammain</code></td>
<td>
<p>indicator variable for constrained model, with 1 for the null 
model that there is no horizontal pleiotropy.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_alphain">alphain</code></td>
<td>
<p>indicator variable for constrained model, with 1 for the null 
model that there is no causal effect.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_max_iterin">max_iterin</code></td>
<td>
<p>The maximum iteration.</p>
</td></tr>
<tr><td><code id="PMR_summary_+3A_epsin">epsin</code></td>
<td>
<p>The convergence tolerance of the absolute value of the difference 
between the nth and (n+1)th log likelihood.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of estimates of model parameters, including the causal effect 
<code>alpha</code>, the horizontal pleiotropy effect <code>gamma</code>, and the two corresponding p values.
</p>


<h3>Author(s)</h3>

<p>Zhongshang Yuan, Xiang Zhou.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Examplesummary)
attach(Examplesummary)
fmH1=PMR_summary(betaxin=betax,betayin=betay,Sigma1sin=Sigma1,Sigma2sin=Sigma2,
      samplen1=n1,samplen2=n2,gammain=0,alphain=0,max_iterin =1000, epsin=1e-5)
fmH0alpha=PMR_summary(betaxin=betax,betayin=betay,Sigma1sin=Sigma1,Sigma2sin=Sigma2,
           samplen1=n1,samplen2=n2,gammain=0,alphain=1,max_iterin =1000, epsin=1e-5)
fmH0gamma=PMR_summary(betaxin=betax,betayin=betay,Sigma1sin=Sigma1,Sigma2sin=Sigma2,
           samplen1=n1,samplen2=n2,gammain=1,alphain=0,max_iterin =1000, epsin=1e-5)
loglikH1=max(fmH1$loglik,na.rm=TRUE)
loglikH0alpha=max(fmH0alpha$loglik,na.rm=TRUE)
loglikH0gamma=max(fmH0gamma$loglik,na.rm=TRUE)
stat_alpha = 2 * (loglikH1 - loglikH0alpha)
pvalue_alpha = pchisq(stat_alpha,1,lower.tail=FALSE)
stat_gamma = 2 * (loglikH1 - loglikH0gamma)
pvalue_gamma = pchisq(stat_gamma,1,lower.tail=FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
