<!DOCTYPE html><html><head><title>Help for package PQLseq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PQLseq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PQLseq-package'><p>Efficient Mixed Model Analysis of Count Data in Large-Scale Genomic Sequencing Studies</p></a></li>
<li><a href='#ExampleBSseq'><p>BSseq example dataset</p></a></li>
<li><a href='#ExampleRNAseq'><p>RNAseq example dataset</p></a></li>
<li><a href='#pqlseq'><p>Fit Generalized Linear Mixed Model with Known Kinship Matrices Through Penalized-quasi Likelihood</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Efficient Mixed Model Analysis of Count Data in Large-Scale
Genomic Sequencing Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-06-06</td>
</tr>
<tr>
<td>Author:</td>
<td>Shiquan Sun, Jiaqiang Zhu, Xiang Zhou</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiaqiang Zhu &lt;jiaqiang@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An efficient tool designed for differential analysis of large-scale RNA sequencing (RNAseq) data and Bisulfite sequencing (BSseq) data in the presence of individual relatedness and population structure. 'PQLseq' first fits a Generalized Linear Mixed Model (GLMM) with adjusted covariates, predictor of interest and random effects to account for population structure and individual relatedness, and then performs Wald tests for each gene in RNAseq or site in BSseq. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.14),foreach,doParallel,parallel,Matrix,methods</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp,RcppArmadillo</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-06 13:10:04 UTC; jiaqiang</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-06 13:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='PQLseq-package'>Efficient Mixed Model Analysis of Count Data in Large-Scale Genomic Sequencing Studies</h2><span id='topic+PQLseq-package'></span><span id='topic+PQLseq'></span>

<h3>Description</h3>

<p>An efficient tool designed for differential analysis of large-scale RNA sequencing (RNAseq) data and Bisulfite sequencing (BSseq) data in the presence of individual relatedness and population structure. 'PQLseq' first fits a Generalized Linear Mixed Model (GLMM) with adjusted covariates, predictor of interest and random effects to account for population structure and individual relatedness, and then performs Wald tests for each gene in RNAseq or site in BSseq. PQLseq is an R package for efficient differential analysis of large-scale RNA sequencing data and bisulfite sequencing data in the presence of individual relatedness and population structure. It first fits a Generalized linear mixed model with adjusted covariates, predictor of interest and random effects to account for population structure and individual relatedness, and then performs Wald test for each gene in RNA sequencing data or site in bisulfite sequencing data.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PQLseq</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.10</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-06-02</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Shiquan Sun, Jiaqiang Zhu, Xiang Zhou
Maintainer: Shiquan Sun &lt;shiquans@umich.edu&gt;
</p>


<h3>References</h3>

<p>Breslow, N.E. and Clayton, D.G. (1993) Approximate Inference in Generalized Linear Mixed Models. Journal of the American Statistical Association 88, 9-25.
</p>
<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celedon, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X (2016). Control for population structure and relatedness for binary traits in genetic association studies using logistic mixed models. The American Journal of Human Genetics, 98, 653-666.
</p>
<p>Gilmour, A.R., Thompson, R. and Cullis, B.R. (1995) Average Information REML: An Efficient Algorithm for Variance Parameter Estimation in Linear Mixed Models. Biometrics 51, 1440-1450.
</p>
<p>Lea,A., Tung, J. and Zhou,X. (2015) A flexible, effcient binomial mixed model for identifying differential DNA methylation in bisulfite sequencing data. PLoS Genetics. 11: e1005650.
</p>
<p>Rue H., Martino S. , and Chopin N.(2009) Approximate Bayesian inference for latent Gaussian models using integrated nested Laplace approximations (with discussion). Journal of the Royal Statistical Society, Series B,71(2):319-392. 
</p>
<p>Yang, J., Lee, S.H., Goddard, M.E. and Visscher, P.M. (2011) GCTA: A Tool for Genome-wide Complex Trait Analysis. The American Journal of Human Genetics 88, 76-82.
</p>
<p>Zhou, X., Carbonetto, P. and Stephens,M. (2013) Polygenic modeling with Bayesian sparse linear mixed models. PLoS Genetics. 9(2): e1003264.
</p>
<p>Zhou, X. and Stephens, M. (2012) Genome-wide efficient mixed-model analysis for association studies. Nature Genetics 44, 821-824.
</p>
<p>Zhou, X. and Stephens, M.(2014) Efficient multivariate linear mixed model algorithms for genome-wide association studies. Nature Methods.
</p>

<hr>
<h2 id='ExampleBSseq'>BSseq example dataset</h2><span id='topic+ExampleBSseq'></span>

<h3>Description</h3>

<p>A simulated example dataset of BSseq for PQLseq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleBSseq)</code></pre>


<h3>Format</h3>

<p>Contains the following objects:
</p>

<dl>
<dt>mcount</dt><dd><p>a data frame containing the read counts for 5 sites.</p>
</dd>
<dt>predictor</dt><dd><p>a vector of 100 observations on a continuous variable.</p>
</dd>
<dt>relatednessmatrix</dt><dd><p>a genetic relationship matrix for 100 individuals.</p>
</dd>
<dt>totalcount</dt><dd><p>a data frame containing the total read counts for 5 sites.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleBSseq)
attach(ExampleBSseq)
model_DNA=pqlseq(RawCountDataSet=mcount, Phenotypes=predictor, 
  RelatednessMatrix=relatednessmatrix, LibSize=totalcount,
  fit.model="BMM",numCore=1)
head(model_DNA)
detach(ExampleBSseq)
</code></pre>

<hr>
<h2 id='ExampleRNAseq'>RNAseq example dataset</h2><span id='topic+ExampleRNAseq'></span>

<h3>Description</h3>

<p>A simulated example dataset of RNAseq for PQLseq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleRNAseq)</code></pre>


<h3>Format</h3>

<p>Contains the following objects:
</p>

<dl>
<dt>count</dt><dd><p>a data frame containing the read counts for 5 genes.</p>
</dd>
<dt>predictor</dt><dd><p>a vector of 100 observations on a continuous variable.</p>
</dd>
<dt>relatednessmatrix</dt><dd><p>a genetic relationship matrix for 100 individuals.</p>
</dd>
<dt>totalcount</dt><dd><p>a data frame containing the total read counts for 5 genes.</p>
</dd>
</dl>


<hr>
<h2 id='pqlseq'>Fit Generalized Linear Mixed Model with Known Kinship Matrices Through Penalized-quasi Likelihood</h2><span id='topic+pqlseq'></span>

<h3>Description</h3>

<p>Fit a generalized linear mixed model with a random intercept. The covariance matrix of the random intercept is proportional to a known kinship matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
  pqlseq(RawCountDataSet, Phenotypes, Covariates=NULL,
  RelatednessMatrix=NULL, LibSize=NULL, fit.model="PMM",
  fit.method = "AI.REML", fit.maxiter=500, fit.tol=1e-5, 
  numCore=1, filtering=TRUE, verbose=FALSE,...)  
</code></pre>


<h3>Arguments</h3>

   
<table>
<tr><td><code id="pqlseq_+3A_rawcountdataset">RawCountDataSet</code></td>
<td>
<p>a data frame containing the read count.</p>
</td></tr>   
<tr><td><code id="pqlseq_+3A_phenotypes">Phenotypes</code></td>
<td>
<p>a vector containing the predictor of interest.</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_covariates">Covariates</code></td>
<td>
<p>a data frame containing the covariates subject to adjustment (Default = NULL).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_relatednessmatrix">RelatednessMatrix</code></td>
<td>
<p>a known relationship matrix (e.g. kinship matrix in genetic studies). When supplied with a matrix, this matrix should be a positive semi-definite matrix with dimensions equal to the
sample size in count data, and the order of subjects in this matrix should also match the order of subjects in count data. Currently there is no ID checking feature implemented, and it is the user's responsibility to match the orders. </p>
</td></tr> 
<tr><td><code id="pqlseq_+3A_libsize">LibSize</code></td>
<td>
<p>a data frame containing the total read count. For
possion mixed model, it will be calculated automatically if users do not provide. For binomial mixed model, it is required.  </p>
</td></tr>
<tr><td><code id="pqlseq_+3A_fit.model">fit.model</code></td>
<td>
<p>a description of the error distribution and link function to be used in the model. Either &quot;PMM&quot; for possion model, or &quot;BMM&quot; for binomial model (default = &quot;PMM&quot;).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_fit.method">fit.method</code></td>
<td>
<p>method of fitting the generalized linear mixed model, currently only &quot;REML&quot; version is available.</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_fit.maxiter">fit.maxiter</code></td>
<td>
<p>a positive integer specifying the maximum number of iterations when fitting the generalized linear mixed model (default = 500).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_fit.tol">fit.tol</code></td>
<td>
<p>a positive number specifying tolerance, the difference threshold for parameter estimates below which iterations should be stopped (default = 1e-5).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_numcore">numCore</code></td>
<td>
<p>a positive integer specifying the number of cores for parallel computing (default = 1).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_filtering">filtering</code></td>
<td>
<p>a logical switch for RNAseq data. By default, for each gene, at least two individuals should have read counts greater than 5. Otherwise, the gene is filtered (default = TRUE).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_verbose">verbose</code></td>
<td>
<p>a logical switch for printing detailed information (parameter estimates in each iteration) for testing and debugging purpose (default = FALSE).</p>
</td></tr>
<tr><td><code id="pqlseq_+3A_...">...</code></td>
<td>
<p>additional arguments that could be passed to glm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generalized linear mixed models (GLMM) are fitted using the penalized quasi-likelihood (PQL) method proposed by Breslow and Clayton (1993). Statistical inference in GLMM is notoriously difficult because of an intractable high-dimensional integral in the likelihood (Chen, 2016 and Lea, 2015), and by default we use the Average Information REML algorithm (Gilmour, Thompson and Cullis, 1995; Yang et al., 2011) to fit the model. An eigen-decomposition is performed in each outer iteration and the estimate of the variance component parameter <code class="reqn">\tau</code> is obtained by maximizing the profiled log restricted likelihood. When the Average Information REML algorithm fails to converge, a warning message is given and the algorithm is default to INLA approaches (Rue, 2009).
</p>


<h3>Value</h3>

<table>
<tr><td><code>numIDV</code></td>
<td>
<p>number of individuals with data being analyzed</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>the fixed effect parameter estimate for the predictor of interest.</p>
</td></tr>
<tr><td><code>se_beta</code></td>
<td>
<p>the standard deviation of fixed effect.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>P value for the fixed effect, based on the wald test.</p>
</td></tr>
<tr><td><code>h2</code></td>
<td>
<p>heritability of the transformed rate.</p>
</td></tr>
<tr><td><code>sigma2</code></td>
<td>
<p>total variance component.</p>
</td></tr>
<tr><td><code>overdisp</code></td>
<td>
<p>dispersion parameter estimate</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>a logical indicator for convergence.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Shiquan Sun, Jiaqiang Zhu, Xiang Zhou
</p>


<h3>References</h3>

<p>Breslow, N.E. and Clayton, D.G. (1993) Approximate Inference in Generalized Linear Mixed Models. Journal of the American Statistical Association 88, 9-25.
</p>
<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celedon, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. Control for population structure and relatedness for binary traits in genetic association studies using logistic mixed models. The American Journal of Human Genetics, 98, 653-666.
</p>
<p>Gilmour, A.R., Thompson, R. and Cullis, B.R. (1995) Average Information REML: An Efficient Algorithm for Variance Parameter Estimation in Linear Mixed Models. Biometrics 51, 1440-1450.
</p>
<p>Lea,A., Tung, J. and Zhou,X. (2015) A flexible, effcient binomial mixed model for identifying differential DNA methylation in bisulfite sequencing data. PLoS Genetics. 11: e1005650.
</p>
<p>Rue, H., Martino, S. , and Chopin, N.(2009) Approximate Bayesian inference for latent Gaussian models using integrated nested Laplace approximations (with discussion). Journal of the Royal Statistical Society, Series B, 71(2):319-392
</p>
<p>Yang, J., Lee, S.H., Goddard, M.E. and Visscher, P.M. (2011) GCTA: A Tool for Genome-wide Complex Trait Analysis. The American Journal of Human Genetics 88, 76-82.
</p>
<p>Zhou, X. and Stephens, M. (2012) Genome-wide efficient mixed-model analysis for association studies. Nature Genetics 44, 821-824.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleRNAseq)
attach(ExampleRNAseq)
model_RNA=pqlseq(RawCountDataSet=count, Phenotypes=predictor, 
  RelatednessMatrix=relatednessmatrix, LibSize=totalcount,
  fit.model="PMM",numCore=1)
head(model_RNA)
detach(ExampleRNAseq)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
