<!DOCTYPE html><html><head><title>Help for package pqrBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pqrBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data'><p>simulated data for demonstrating the features of pqrBayes</p></a></li>
<li><a href='#pqrBayes'><p>fit a regularized Bayesian quantile varying coefficient model</p></a></li>
<li><a href='#pqrBayes-package'><p>Regularized Bayesian Quantile Varying Coefficient Model</p></a></li>
<li><a href='#predict.pqrBayes'><p>make predictions from a pqrBayes object</p></a></li>
<li><a href='#print.pqrBayes'><p>print a pqrBayes result</p></a></li>
<li><a href='#print.pqrBayes.pred'><p>print a pqrBayes.pred object</p></a></li>
<li><a href='#print.VCselect'><p>print a select.VC object</p></a></li>
<li><a href='#VCselect'><p>Variable selection for a pqrBayes object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Penalized Quantile Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-13</td>
</tr>
<tr>
<td>Description:</td>
<td>The quantile varying coefficient model is robust to data heterogeneity, 
    outliers and heavy-tailed distributions in the response variable due to the check
    loss function in quantile regression. In addition, it can flexibly model the dynamic
    pattern of regression coefficients through nonparametric varying coefficient 
    functions. Although high dimensional quantile varying coefficient model has been 
    examined extensively in the frequentist framework, the corresponding Bayesian variable 
    selection methods have rarely been developed. In this package, we have implemented 
    the Gibbs samplers of the penalized Bayesian quantile varying coefficient model with 
    the spike-and-slab priors [Zhou et al.(2023)]&lt;<a href="https://doi.org/10.1016%2Fj.csda.2023.107808">doi:10.1016/j.csda.2023.107808</a>&gt;.
    The Markov Chain Monte Carlo (MCMC) algorithms of the proposed
    and alternative models can be efficiently performed by using the package.    </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cenwu/pqrBayes">https://github.com/cenwu/pqrBayes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cenwu/pqrBayes/issues">https://github.com/cenwu/pqrBayes/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp,glmnet</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-13 22:30:11 UTC; wucen</td>
</tr>
<tr>
<td>Author:</td>
<td>Cen Wu [aut, cre],
  Fei Zhou [aut],
  Jie Ren [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cen Wu &lt;wucen@ksu.edu&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-14 18:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='data'>simulated data for demonstrating the features of pqrBayes</h2><span id='topic+data'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of pqrBayes.
</p>


<h3>Format</h3>

<p>The data object consists of five components: g, y, u, e and coeff. coeff contains the true values of parameters used for generating the response variable <code class="reqn">y</code>.
</p>


<h3>Details</h3>

<p><strong>The model for generating Y</strong>
</p>
<p>Use subscript <code class="reqn">i</code> to denote the <code class="reqn">i</code>th subject. Let <code class="reqn">(\boldsymbol X_{i}, Y_{i}, V_{i}, \boldsymbol E_{i})</code>, (<code class="reqn">i=1,\ldots,n</code>) be
independent and identically distributed random vectors. <code class="reqn">Y_{i}</code> is a continuous response variable representing the
disease phenotype. <code class="reqn">\boldsymbol X_{i}=(X_{i0},...,X_{ip})^\top</code> denotes a <code class="reqn">(1+p)</code>&ndash;dimensional vector of predictors (e.g. genetic factors) with the first element <code class="reqn">X_{i0}=1</code>. 
The environmental factor <code class="reqn">V_i \in \rm I\!R^1</code> is a univariate index variable. <code class="reqn">\boldsymbol E_{i}=(E_{i1},...,E_{iq})^\top</code> is the <code class="reqn">q</code>-dimensional vector 
of clinical covariates. At a given quantile level <code class="reqn">\tau</code>,
considering the following quantile varying coefficient model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i}=\sum_{k=1}^{q} E_{ik} \beta_{k,\tau} +\sum_{j=0}^{p}\gamma_{j,\tau}(V_i)X_{ij} +\epsilon_{i,\tau},</code>
</p>

<p>where <code class="reqn">\beta_{k,\tau}</code>'s are the regression coefficients for the clinical covariates and <code class="reqn">\gamma_{j,\tau}(\cdot)</code>'s are unknown smooth varying-coefficient functions. 
The regression coefficients of <code class="reqn">\boldsymbol X</code> vary with the univariate index variable <code class="reqn">\boldsymbol v=(v_1,...,v_n)^\top</code>. 
The <code class="reqn">\epsilon_{i,\tau}</code> is the random error. For simplicity of notation, the quantile level <code class="reqn">\tau</code> has been suppressed hereafter.
</p>
<p>The true model that we used to generate Y:
</p>
<p style="text-align: center;"><code class="reqn">Y_i=\gamma_0(v_i)+\gamma_1(v_i)X_{i1}+\gamma_2(v_i)X_{i2}+\gamma_3(v_i)X_{i3}+\epsilon_i,</code>
</p>

<p>where <code class="reqn">\epsilon_i\sim N(0,1)</code>, <code class="reqn">\gamma_{0}=1.5\sin(0.2\pi*v_i</code>), <code class="reqn">\gamma_{1}=2\exp(0.2v_i-1)-1.5 </code>, <code class="reqn">\gamma_{2}=2-2v_i </code> and <code class="reqn">\gamma_3=-4+(v_i-2)^3/6</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)
g=data$g
dim(g)
coeff=data$coeff
print(coeff)


</code></pre>

<hr>
<h2 id='pqrBayes'>fit a regularized Bayesian quantile varying coefficient model</h2><span id='topic+pqrBayes'></span>

<h3>Description</h3>

<p>fit a regularized Bayesian quantile varying coefficient model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pqrBayes(
  g,
  y,
  u,
  e = NULL,
  quant = 0.5,
  iterations = 10000,
  kn = 2,
  degree = 2,
  sparse = TRUE,
  hyper = NULL,
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pqrBayes_+3A_g">g</code></td>
<td>
<p>the matrix of predictors (subject to selection) without intercept.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_y">y</code></td>
<td>
<p>the response variable. The current version only supports the continuous response.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_u">u</code></td>
<td>
<p>a vector of effect modifying variable of the quantile varying coefficient model.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_e">e</code></td>
<td>
<p>a matrix of clinical covariates not subject to selection.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_quant">quant</code></td>
<td>
<p>the quantile level specified by users. The default value is 0.5.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_kn">kn</code></td>
<td>
<p>the number of interior knots for B-spline.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_degree">degree</code></td>
<td>
<p>the degree of B-spline basis.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_hyper">hyper</code></td>
<td>
<p>a named list of hyperparameters.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_debugging">debugging</code></td>
<td>
<p>logical flag. If TRUE, progress will be output to the console and extra information will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model described in &quot;<code><a href="#topic+data">data</a></code>&quot; is:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i}=\sum_{k=1}^{q} E_{ik} \beta_k +\sum_{j=0}^{p}\gamma_j(V_i)X_{ij} +\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\beta_k</code>'s are the regression coefficients for the clinical covariates and <code class="reqn">\gamma_j</code>'s are the varying coefficients for the intercept and predictors (e.g. genetic factors).
</p>
<p>When sparse=TRUE (default), spike&ndash;and&ndash;slab priors are adopted. Otherwise, Laplacian shrinkage will be used.
Users can modify the hyper-parameters by providing a named list of hyper-parameters via the argument &lsquo;hyper&rsquo;.
The list can have the following named components
</p>

<ul>
<li><p>a0, b0:  shape parameters of the Beta priors (<code class="reqn">\pi^{a_{0}-1}(1-\pi)^{b_{0}-1}</code>) on <code class="reqn">\pi_{0}</code>.
</p>
</li>
<li><p>c1, c2:  the shape parameter and the rate parameter of the Gamma prior on <code class="reqn">\nu</code>.
</p>
</li></ul>

<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &quot;pqrBayes&quot; is returned, which is a list with components:
</p>
<table>
<tr><td><code>posterior</code></td>
<td>
<p>posterior samples from the MCMC</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>a list of posterior estimates of coefficients</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhou, F., Ren, J., Ma, S. and Wu, C. (2023). The Bayesian regularized quantile varying coefficient model.
<em>Computational Statistics &amp; Data Analysis</em>, 107808 <a href="https://doi.org/10.1016/j.csda.2023.107808">doi:10.1016/j.csda.2023.107808</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions. 
<em>Biometrics</em>, 79(2), 684-694 <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y. and Wu, C. (2020) Semi-parametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39: 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)
g=data$g
y=data$y
u=data$u
e=data$e

## default method
fit1=pqrBayes(g,y,u,e,quant=0.5)
fit1



## non-sparse
sparse=FALSE
fit2=pqrBayes(g,y,u,e,quant=0.5,sparse = sparse)
fit2

</code></pre>

<hr>
<h2 id='pqrBayes-package'>Regularized Bayesian Quantile Varying Coefficient Model</h2><span id='topic+pqrBayes-package'></span>

<h3>Description</h3>

<p>In this package, we implement a Bayesian quantile varying coefficient model for non-linear gene-environment interaction analysis. The varying 
coefficient functions capture the possible non-linear gene-environment interactions and they are approximated using linear combinations of B-spline basis. 
Quantile regression is adopted as it's robust to long-tailed distributions in the response/phenotype and provides the capability of describing the relationship 
between the response variable and predictors at different quantiles of the response variable. The default method (the proposed method) conducts variable 
selection by accounting for sparsity. In particular, the spike&ndash;and&ndash;slab priors are adopted to shrink the coefficients of unimportant effects to exactly zero. 
In addition to the default method, users can also choose the method without spike&ndash;and&ndash;slab priors.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface <strong>pqrBayes()</strong> allows users to flexibly choose the fitting methods by specifying the following parameter:
</p>

<table>
<tr>
 <td style="text-align: right;">
sparse: </td><td style="text-align: left;"> whether to use the spike-and-slab priors to impose sparsity. 
</td>
</tr>

</table>

<p>The function pqrBayes() returns a pqrBayes object that contains the posterior estimates of each coefficients.
</p>


<h3>References</h3>

<p>Zhou, F., Ren, J., Ma, S. and Wu, C. (2023). The Bayesian regularized quantile varying coefficient model.
<em>Computational Statistics &amp; Data Analysis</em>, 107808 <a href="https://doi.org/10.1016/j.csda.2023.107808">doi:10.1016/j.csda.2023.107808</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions. 
<em>Biometrics</em>, 79(2), 684-694 <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021). Gene–Environment Interaction: a Variable Selection Perspective. 
<em>Epistasis. Methods in Molecular Biology.</em> 2212:191–223 <a href="https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13">https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y. and Wu, C. (2020) Semi-parametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39: 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Wu, C. and Jiang, Y. (2019) spinBayes: Semi-Parametric Gene-Environment Interaction via Bayesian Variable Selection.
R package version 0.1.0. <a href="https://CRAN.R-project.org/package=spinBayes">https://CRAN.R-project.org/package=spinBayes</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y. and Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37:437–456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Wu, C., Shi, X., Cui, Y. and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34 (30): 4016–4030 <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2018). Additive varying–coefficient model for nonlinear gene–environment interactions.
<em>Statistical Applications in Genetics and Molecular Biology</em>, 17(2) <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>

<hr>
<h2 id='predict.pqrBayes'>make predictions from a pqrBayes object</h2><span id='topic+predict.pqrBayes'></span>

<h3>Description</h3>

<p>make predictions from a pqrBayes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pqrBayes'
predict(
  object,
  g.new,
  u.new,
  e.new = NULL,
  y.new = NULL,
  quant = 0.5,
  kn = 2,
  degree = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.pqrBayes_+3A_object">object</code></td>
<td>
<p>pqrBayes object.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_g.new">g.new</code></td>
<td>
<p>a matrix of new predictors (e.g. genetic factors) at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_u.new">u.new</code></td>
<td>
<p>a vector of new environmental factor at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_e.new">e.new</code></td>
<td>
<p>a vector or matrix of new clinic covariates at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_y.new">y.new</code></td>
<td>
<p>a vector of the response of new observations. If provided, the prediction error will be computed based on Y.new.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_quant">quant</code></td>
<td>
<p>the quantile for the response variable.  The default is 0.5.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_kn">kn</code></td>
<td>
<p>the number of interior knots for B-spline.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_degree">degree</code></td>
<td>
<p>the degree of B-spline basis.</p>
</td></tr>
<tr><td><code id="predict.pqrBayes_+3A_...">...</code></td>
<td>
<p>other predict arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>g.new (u.new) must have the same number of columns as g (u) used for fitting the model. By default, the clinic covariates are NULL unless 
provided. The predictions are made based on the posterior estimates of coefficients in the pqrBayes object.
</p>
<p>If y.new is provided, the prediction error will be computed based on the check loss.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;pqrBayes.pred&rsquo; is returned, which is a list with components:
</p>
<table>
<tr><td><code>error</code></td>
<td>
<p>prediction error. error is NULL if y.new=NULL.</p>
</td></tr>
<tr><td><code>y.pred</code></td>
<td>
<p>predicted values of the new observations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>

<hr>
<h2 id='print.pqrBayes'>print a pqrBayes result</h2><span id='topic+print.pqrBayes'></span>

<h3>Description</h3>

<p>Print a pqrBayes result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pqrBayes'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.pqrBayes_+3A_x">x</code></td>
<td>
<p>pqrBayes result</p>
</td></tr>
<tr><td><code id="print.pqrBayes_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.pqrBayes_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>

<hr>
<h2 id='print.pqrBayes.pred'>print a pqrBayes.pred object</h2><span id='topic+print.pqrBayes.pred'></span>

<h3>Description</h3>

<p>Print a summary of a pqrBayes.pred object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pqrBayes.pred'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.pqrBayes.pred_+3A_x">x</code></td>
<td>
<p>pqrBayes.pred object.</p>
</td></tr>
<tr><td><code id="print.pqrBayes.pred_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.pqrBayes.pred_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict.pqrBayes">predict.pqrBayes</a></code>
</p>

<hr>
<h2 id='print.VCselect'>print a select.VC object</h2><span id='topic+print.VCselect'></span>

<h3>Description</h3>

<p>Print a summary of a select.VC object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'VCselect'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.VCselect_+3A_x">x</code></td>
<td>
<p>VCselect object.</p>
</td></tr>
<tr><td><code id="print.VCselect_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.VCselect_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+VCselect">VCselect</a></code>
</p>

<hr>
<h2 id='VCselect'>Variable selection for a pqrBayes object</h2><span id='topic+VCselect'></span>

<h3>Description</h3>

<p>Variable selection for a pqrBayes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VCselect(obj, sparse, iterations = 10000, kn = 2, degree = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VCselect_+3A_obj">obj</code></td>
<td>
<p>pqrBayes object.</p>
</td></tr>
<tr><td><code id="VCselect_+3A_sparse">sparse</code></td>
<td>
<p>logical flag.</p>
</td></tr>
<tr><td><code id="VCselect_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="VCselect_+3A_kn">kn</code></td>
<td>
<p>the number of interior knots for B-spline.</p>
</td></tr>
<tr><td><code id="VCselect_+3A_degree">degree</code></td>
<td>
<p>the degree of B-spline basis.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For class &lsquo;Sparse&rsquo;, the median probability model (MPM) (Barbieri and Berger, 2004) is used to identify predictors that are significantly associated
with the response variable. For class &lsquo;NonSparse&rsquo;, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;VCselect&rsquo; is returned, which includes the indices of the selected predictors (e.g. genetic factors).
</p>


<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2022). Robust Bayesian variable selection for gene-environment interactions. 
<em>Biometrics</em>, (in press) <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Barbieri, M.M. and Berger, J.O. (2004). Optimal predictive model selection. <em>Ann. Statist</em>, 32(3):870–897
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
