<!DOCTYPE html><html lang="en"><head><title>Help for package pqrBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pqrBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pqrBayes-package'><p>Bayesian penalized quantile regression for linear and varying coefficient models</p></a></li>
<li><a href='#coverage'><p>95% empirical coverage probability for a pqrBayes object</p></a></li>
<li><a href='#data'><p>simulated data under sparse linear and quantile varying coefficient models</p></a></li>
<li><a href='#estimation.pqrBayes'><p>Estimation and estimation accuracy for a pqrBayes object</p></a></li>
<li><a href='#pqrBayes'><p>fit Bayesian penalized quantile regression for linear and varying coefficient models</p></a></li>
<li><a href='#pqrBayes.select'><p>Variable selection for a pqrBayes object</p></a></li>
<li><a href='#predict_pqrBayes'><p>Make predictions from a pqrBayes object</p></a></li>
<li><a href='#print.pqrBayes'><p>print a pqrBayes result</p></a></li>
<li><a href='#print.pqrBayes.pred'><p>print a pqrBayes.pred object</p></a></li>
<li><a href='#print.pqrBayes.select'><p>print a select.pqrBayes object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Penalized Quantile Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-02-22</td>
</tr>
<tr>
<td>Description:</td>
<td>Bayesian regularized quantile regression utilizing sparse priors to 
    impose exact sparsity leads to efficient Bayesian shrinkage estimation, variable 
    selection and statistical inference. In this package, we have implemented robust
    Bayesian variable selection with spike-and-slab priors under high-dimensional
    linear regression models (Fan et al. (2024) &lt;<a href="https://doi.org/10.3390%2Fe26090794">doi:10.3390/e26090794</a>&gt; and  
    Ren et al. (2023) &lt;<a href="https://doi.org/10.1111%2Fbiom.13670">doi:10.1111/biom.13670</a>&gt;), and regularized quantile varying
    coefficient models (Zhou et al.(2023) &lt;<a href="https://doi.org/10.1016%2Fj.csda.2023.107808">doi:10.1016/j.csda.2023.107808</a>&gt;). In particular, 
    valid robust Bayesian inferences under both models in the presence of heavy-tailed errors
    can be validated on finite samples. The Markov Chain Monte Carlo (MCMC) algorithms 
    of the proposed and alternative models are implemented in C++.   </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cenwu/pqrBayes">https://github.com/cenwu/pqrBayes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cenwu/pqrBayes/issues">https://github.com/cenwu/pqrBayes/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp,glmnet,splines, stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-22 13:32:38 UTC; cenwu</td>
</tr>
<tr>
<td>Author:</td>
<td>Kun Fan [aut],
  Cen Wu [aut, cre],
  Jie Ren [aut],
  Fei Zhou [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cen Wu &lt;wucen@ksu.edu&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-23 23:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='pqrBayes-package'>Bayesian penalized quantile regression for linear and varying coefficient models</h2><span id='topic+pqrBayes-package'></span>

<h3>Description</h3>

<p>In this package, we implement Bayesian penalized quantile regression for the linear regression model and quantile varying coefficient (VC) model. 
Point-mass spike-and-slab priors have been incorporated in the Bayesian hierarchical models to facilitate Bayesian shrinkage 
estimation with exact sparsity in both models. The two default methods are Bayesian regularized quantile regression with spike-and-slab priors
under the linear model and VC model, correspondingly. In addition to default methods, users can also choose methods without robustness and/or spike&ndash;and&ndash;slab priors.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface <strong>pqrBayes()</strong> allows users to flexibly choose fitting models by specifying the following parameters:
</p>

<table>
<tr>
 <td style="text-align: right;">
robust: </td><td style="text-align: left;"> whether to fit a robust sparse quantile regression model or a quantile varying coefficient </td>
</tr>
<tr>
 <td style="text-align: right;"></td><td style="text-align: left;"> model or their non-robust counterparts. </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
sparse: </td><td style="text-align: left;"> whether to use the spike-and-slab priors to impose exact sparsity. </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
model: </td><td style="text-align: left;"> whether to fit a linear model or a varying coefficient model.
</td>
</tr>

</table>

<p>The function pqrBayes() returns a pqrBayes object that stores the posterior estimates of regression coefficients.
</p>


<h3>References</h3>

<p>Fan, K., Subedi, S., Yang, G., Lu, X., Ren, J. and Wu, C. (2024). Is Seeing Believing? A Practitioner's Perspective on High-dimensional Statistical Inference in Cancer Genomics Studies. <em>Entropy</em>, 26(9).794 <a href="https://doi.org/10.3390/e26090794">doi:10.3390/e26090794</a>  
</p>
<p>Zhou, F., Ren, J., Ma, S. and Wu, C. (2023). The Bayesian regularized quantile varying coefficient model.
<em>Computational Statistics &amp; Data Analysis</em>, 107808 <a href="https://doi.org/10.1016/j.csda.2023.107808">doi:10.1016/j.csda.2023.107808</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions. 
<em>Biometrics</em>, 79(2), 684-694 <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021). Gene–Environment Interaction: a Variable Selection Perspective. 
<em>Epistasis. Methods in Molecular Biology.</em> 2212:191–223 <a href="https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13">https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y. and Wu, C. (2020) Semi-parametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39: 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Wu, C. and Jiang, Y. (2019) spinBayes: Semi-Parametric Gene-Environment Interaction via Bayesian Variable Selection.
R package version 0.1.0. <a href="https://CRAN.R-project.org/package=spinBayes">https://CRAN.R-project.org/package=spinBayes</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y. and Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37:437–456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Wu, C., Shi, X., Cui, Y. and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34 (30): 4016–4030 <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2018). Additive varying–coefficient model for nonlinear gene–environment interactions.
<em>Statistical Applications in Genetics and Molecular Biology</em>, 17(2) <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>

<hr>
<h2 id='coverage'>95% empirical coverage probability for a pqrBayes object</h2><span id='topic+coverage'></span>

<h3>Description</h3>

<p>Calculate 95% empirical coverage probabilities for regression coefficients under linear and VC models, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coverage(object,coefficient,u.grid=NULL,model="linear")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coverage_+3A_object">object</code></td>
<td>
<p>the pqrBayes object.</p>
</td></tr>
<tr><td><code id="coverage_+3A_coefficient">coefficient</code></td>
<td>
<p>the vector of true regression coefficients under a linear model or the matrix of true varying coefficients evaluated on the grid points under a varying coefficient model.</p>
</td></tr>
<tr><td><code id="coverage_+3A_u.grid">u.grid</code></td>
<td>
<p>the vector of grid points under a varying coefficient model. When assessing empirical coverage probabilities under a linear model, u.grid = NULL.</p>
</td></tr>
<tr><td><code id="coverage_+3A_model">model</code></td>
<td>
<p>the model to be fitted. Users can also choose &quot;linear&quot; for a linear model or &quot;VC&quot; for a varying coefficient model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>c
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The quantile regression model
data(data)
data = data$data_linear
g=data$g
y=data$y
e=data$e
coeff = data$coeff
fit1=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,model="linear")
coverage=coverage(fit1,coeff,model="linear")
</code></pre>

<hr>
<h2 id='data'>simulated data under sparse linear and quantile varying coefficient models</h2><span id='topic+data'></span>

<h3>Description</h3>

<p>Simulated data under sparse linear and quantile varying coefficient models
</p>


<h3>Format</h3>

<p>The data_linear object consists of 4 components: g, y, e and coeff. coeff contains the true values of parameters used for generating the response variable <code class="reqn">y</code>.
The data_varying object consists of five components: g, y, u, e and coeff. coeff contains the true values of parameters used for generating the response variable <code class="reqn">y</code>.
</p>


<h3>Details</h3>

<p><strong>Generating Y using a sparse linear (quantile) regression model</strong>
</p>
<p>The true data generating model under sparse linear regression:
</p>
<p style="text-align: center;"><code class="reqn">Y_i=\beta_0+\beta_{1}X_{i1}+\beta_{2}X_{i2}+\beta_{3}X_{i3}+\epsilon_i,</code>
</p>

<p>where <code class="reqn">\epsilon_i\sim N(0,1)</code>, <code class="reqn">\beta_{0}=0</code>, <code class="reqn">\beta_{1}=1 </code>, <code class="reqn">\beta_{2}=1.5</code> and <code class="reqn">\beta_3=2</code>.
</p>
<p><strong>Generating Y using a (quantile) varying coefficient model</strong>
</p>
<p>Data generation under sparse (quantile) VC model:
</p>
<p style="text-align: center;"><code class="reqn">Y_i=\gamma_0(v_i)+\gamma_1(v_i)X_{i1}+\gamma_2(v_i)X_{i2}+\gamma_3(v_i)X_{i3}+\epsilon_i,</code>
</p>

<p>where <code class="reqn">\epsilon_i\sim N(0,1)</code>, <code class="reqn">\gamma_{0}(v_i)=1.5\sin(0.2\pi*v_i</code>), <code class="reqn">\gamma_{1}(v_i)=2\exp(0.2v_i-1)-1.5 </code>, <code class="reqn">\gamma_{2}(v_i)=2-2v_i </code> and <code class="reqn">\gamma_3(v_i)=-4+(v_i-2)^3/6</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)
data = data$data_linear
g=data$g
dim(g)
y=data$y
coeff=data$coeff
print(coeff)

data = data$data_varying
g=data$g
dim(g)
coeff=data$coeff
print(coeff)


</code></pre>

<hr>
<h2 id='estimation.pqrBayes'>Estimation and estimation accuracy for a pqrBayes object</h2><span id='topic+estimation.pqrBayes'></span>

<h3>Description</h3>

<p>Calculate estimated regression coefficients with estimation accuracy from linear and quantile VC models, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimation.pqrBayes(object,coefficient,u.grid=NULL,model="linear")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimation.pqrBayes_+3A_object">object</code></td>
<td>
<p>an object of class &lsquo;pqrBayes&rsquo;.</p>
</td></tr>
<tr><td><code id="estimation.pqrBayes_+3A_coefficient">coefficient</code></td>
<td>
<p>the vector of quantile regression coefficients under a linear model or the matrix of true varying coefficients evaluated on the grid points under a varying coefficient model.</p>
</td></tr>
<tr><td><code id="estimation.pqrBayes_+3A_u.grid">u.grid</code></td>
<td>
<p>the vector of grid points under a varying coefficient model. When fitting a linear (quantile) regression model, u.grid = NULL.</p>
</td></tr>
<tr><td><code id="estimation.pqrBayes_+3A_model">model</code></td>
<td>
<p>the model to be fitted. Users can choose &quot;linear&quot; for a linear model or &quot;VC&quot; for a varying coefficient model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class &lsquo;pqrBayes.est&rsquo; is returned, which is a list with components:
</p>
<table role = "presentation">
<tr><td><code>error</code></td>
<td>
<p>mean square error or integrated mean square errors and total integrated mean square error.</p>
</td></tr>
<tr><td><code>coeff.est</code></td>
<td>
<p>estimated values of the regression coefficients or the varying coefficients.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The quantile regression model
data(data)
data = data$data_linear
g=data$g
y=data$y
e=data$e
coeff = data$coeff
fit1=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,model="linear")
estimation=estimation.pqrBayes(fit1,coeff,model="linear")

</code></pre>

<hr>
<h2 id='pqrBayes'>fit Bayesian penalized quantile regression for linear and varying coefficient models</h2><span id='topic+pqrBayes'></span>

<h3>Description</h3>

<p>fit Bayesian penalized quantile regression for linear and varying coefficient models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pqrBayes(
  g,
  y,
  u = NULL,
  e,
  quant = 0.5,
  iterations = 10000,
  burn.in = NULL,
  spline,
  robust = TRUE,
  sparse = TRUE,
  model = "linear",
  hyper = NULL,
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pqrBayes_+3A_g">g</code></td>
<td>
<p>the matrix of predictors (subject to selection). Users do not need to specify an intercept which will be automatically included.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_y">y</code></td>
<td>
<p>the response variable. The current version only supports the continuous response.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_u">u</code></td>
<td>
<p>a vector of effect modifying variable of the quantile varying coefficient model. When fitting a linear model, u = NULL.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_e">e</code></td>
<td>
<p>a matrix of clinical covariates not subject to selection.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_quant">quant</code></td>
<td>
<p>the quantile level specified by users. The default value is 0.5.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations. The default value is 10,000.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_burn.in">burn.in</code></td>
<td>
<p>the number of burn-in iterations. If NULL, the first half of MCMC iterations will be used as burn-ins.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_spline">spline</code></td>
<td>
<p>a list of the number of interior knots (kn) for B-spline and the degree of B-spline basis (degree). When fitting a linear model, spline = NULL.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_robust">robust</code></td>
<td>
<p>logical flag. If TRUE, robust methods will be used. Otherwise, non-robust methods will be used. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be adopted to impose exact sparsity on regression coefficients. Otherwise, Laplacian shrinkage will be adopted. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_model">model</code></td>
<td>
<p>the model to be fitted. Users can specify &quot;linear&quot; for a linear model and &quot;VC&quot; for a varying coefficient model.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_hyper">hyper</code></td>
<td>
<p>a named list of hyper-parameters. The default value is NULL.</p>
</td></tr>
<tr><td><code id="pqrBayes_+3A_debugging">debugging</code></td>
<td>
<p>logical flag. If TRUE, progress will be output to the console and extra information will be returned. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The quantile regression model described in &quot;<code><a href="#topic+data">data</a></code>&quot; is:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i}=\sum_{k=1}^{q} E_{ik} \beta_k +\sum_{j=0}^{p}X_{ij}\gamma_j +\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\beta_k</code>'s are the regression coefficients for clinical covariates and <code class="reqn">\gamma_j</code>'s are the regression coefficients of <code class="reqn">\boldsymbol X</code>.
</p>
<p>The quantile varying coefficient model described in &quot;<code><a href="#topic+data">data</a></code>&quot; is:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i}=\sum_{k=1}^{q} E_{ik} \beta_k +\sum_{j=0}^{p}\gamma_j(V_i)X_{ij} +\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\beta_k</code>'s are the regression coefficients for the clinical covariates, and <code class="reqn">\gamma_j</code>'s are the varying intercept and varying coefficients for predictors (e.g. genetic factors), respectively.
</p>
<p>Users can modify the hyper-parameters by providing a named list of hyper-parameters via the argument &lsquo;hyper&rsquo;.
The list can have the following named components
</p>

<dl>
<dt>a0, b0: </dt><dd><p> shape parameters of the Beta priors (<code class="reqn">\pi^{a_{0}-1}(1-\pi)^{b_{0}-1}</code>) on <code class="reqn">\pi_{0}</code>.</p>
</dd>
<dt>c1, c2: </dt><dd><p> the shape parameter and the rate parameter of the Gamma prior on <code class="reqn">\nu</code>.</p>
</dd>
</dl>

<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &quot;pqrBayes&quot; is returned, which is a list with components:
</p>
<table role = "presentation">
<tr><td><code>obj</code></td>
<td>
<p>a list of posterior samples from the MCMC and other parameters</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>a list of posterior estimates of coefficients</p>
</td></tr>
</table>


<h3>References</h3>

<p>Fan, K., Subedi, S., Yang, G., Lu, X., Ren, J. and Wu, C. (2024). Is Seeing Believing? A Practitioner's Perspective on High-dimensional Statistical Inference in Cancer Genomics Studies. <em>Entropy</em>, 26(9).794 <a href="https://doi.org/10.3390/e26090794">doi:10.3390/e26090794</a>
</p>
<p>Zhou, F., Ren, J., Ma, S. and Wu, C. (2023). The Bayesian regularized quantile varying coefficient model.
<em>Computational Statistics &amp; Data Analysis</em>, 107808 <a href="https://doi.org/10.1016/j.csda.2023.107808">doi:10.1016/j.csda.2023.107808</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions. 
<em>Biometrics</em>, 79(2), 684-694 <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y. and Wu, C. (2020) Semi-parametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39: 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The quantile regression model
data(data)
data = data$data_linear
g=data$g
y=data$y
e=data$e

fit1=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,model="linear")


## non-sparse
sparse=FALSE
fit2=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,sparse = sparse,model="linear")

## non-robust
robust = FALSE
fit3=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,robust = robust,model="linear")



## The quantile varying coefficient model
data(data)
data = data$data_varying
g=data$g
y=data$y
u=data$u
e=data$e

spline = list(kn=2,degree=2)
fit1=pqrBayes(g,y,u,e,quant=0.5,spline = spline, model="VC")



## non-sparse
sparse=FALSE
fit2=pqrBayes(g,y,u,e,quant=0.5,spline = spline,sparse = sparse,model="VC")

## non-robust
robust = FALSE
fit3=pqrBayes(g,y,u,e,quant=0.5,spline = spline,robust = robust,model="VC")


</code></pre>

<hr>
<h2 id='pqrBayes.select'>Variable selection for a pqrBayes object</h2><span id='topic+pqrBayes.select'></span>

<h3>Description</h3>

<p>Variable selection for a pqrBayes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pqrBayes.select(object,sparse=T,model="linear")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pqrBayes.select_+3A_object">object</code></td>
<td>
<p>a pqrBayes object.</p>
</td></tr>
<tr><td><code id="pqrBayes.select_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, the sparse model is used for variable selection. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="pqrBayes.select_+3A_model">model</code></td>
<td>
<p>the model to be fitted. Users can also choose &quot;linear&quot; for a linear model or &quot;VC&quot; for a varying coefficient model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For class &lsquo;Sparse&rsquo;, the median probability model (MPM) (Barbieri and Berger, 2004) is used to identify predictors that are significantly associated
with the response variable. For class &lsquo;NonSparse&rsquo;, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;select&rsquo; is returned, which includes the indices of the selected predictors (e.g. genetic factors).
</p>


<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions. 
<em>Biometrics</em>, 79(2), 684-694 <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Barbieri, M.M. and Berger, J.O. (2004). Optimal predictive model selection. <em>Ann. Statist</em>, 32(3):870–897
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The quantile regression model
data(data)
data = data$data_linear
g=data$g
y=data$y
e=data$e

fit1=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,model="linear")
sparse=TRUE
select=pqrBayes.select(obj = fit1,sparse = sparse,model="linear")

## The quantile varying coefficient model
data(data)
data = data$data_varying
g=data$g
y=data$y
u=data$u
e=data$e
spline = list(kn=2,degree=2)
fit1=pqrBayes(g,y,u,e,quant=0.5,spline = spline, model="VC")
sparse=TRUE
select=pqrBayes.select(obj = fit1,sparse = sparse,model="VC")
select


## non-sparse
sparse=FALSE
spline = list(kn=2,degree=2)
fit2=pqrBayes(g,y,u,e,quant=0.5,spline = spline,sparse = sparse,model="VC")
select=pqrBayes.select(obj=fit2,sparse=FALSE,model="VC")
select


</code></pre>

<hr>
<h2 id='predict_pqrBayes'>Make predictions from a pqrBayes object</h2><span id='topic+predict_pqrBayes'></span>

<h3>Description</h3>

<p>Make predictions from a pqrBayes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_pqrBayes(object, g.new, u.new, e.new, y.new, quant, model, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict_pqrBayes_+3A_object">object</code></td>
<td>
<p>a pqrBayes object.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_g.new">g.new</code></td>
<td>
<p>a matrix of new predictors (e.g. genetic factors) at which predictions are to be made. When being applied to the linear model, g.new = g.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_u.new">u.new</code></td>
<td>
<p>a vector of new environmental factor at which predictions are to be made. When being applied to the linear model, u.new = NULL.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_e.new">e.new</code></td>
<td>
<p>a vector or matrix of new clinical covariates at which predictions are to be made. When being applied to the linear model, e.new = e.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_y.new">y.new</code></td>
<td>
<p>a vector of the response of new observations. When being applied to the linear model, y.new = y.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_quant">quant</code></td>
<td>
<p>the quantile level.  The default is 0.5.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_model">model</code></td>
<td>
<p>the model to be fitted. The default is &quot;VC&quot; for a quantile varying coefficient model. Users can also specify &quot;linear&quot; for a linear model.</p>
</td></tr>
<tr><td><code id="predict_pqrBayes_+3A_...">...</code></td>
<td>
<p>other predict arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>g.new (u.new) must have the same number of columns as g (u) used for fitting the model. By default, the clinical covariates are NULL unless 
provided. The predictions are made based on the posterior estimates of coefficients in the pqrBayes object.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;pqrBayes.pred&rsquo; is returned, which is a list with components:
</p>
<table role = "presentation">
<tr><td><code>error</code></td>
<td>
<p>prediction error.</p>
</td></tr>
<tr><td><code>y.pred</code></td>
<td>
<p>predicted values of the new observations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The quantile regression model
data(data)
data = data$data_linear
g=data$g
y=data$y
e=data$e
fit1=pqrBayes(g,y,u=NULL,e,quant=0.5,spline=NULL,model="linear")
prediction=predict_pqrBayes(fit1,g,u.new=NULL,e.new = e, y.new = y,model="linear")
</code></pre>

<hr>
<h2 id='print.pqrBayes'>print a pqrBayes result</h2><span id='topic+print.pqrBayes'></span>

<h3>Description</h3>

<p>Print a pqrBayes result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pqrBayes'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.pqrBayes_+3A_x">x</code></td>
<td>
<p>pqrBayes result.</p>
</td></tr>
<tr><td><code id="print.pqrBayes_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.pqrBayes_+3A_...">...</code></td>
<td>
<p>other print arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes">pqrBayes</a></code>
</p>

<hr>
<h2 id='print.pqrBayes.pred'>print a pqrBayes.pred object</h2><span id='topic+print.pqrBayes.pred'></span>

<h3>Description</h3>

<p>Print a summary of a pqrBayes.pred object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pqrBayes.pred'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.pqrBayes.pred_+3A_x">x</code></td>
<td>
<p>pqrBayes.pred object.</p>
</td></tr>
<tr><td><code id="print.pqrBayes.pred_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.pqrBayes.pred_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_pqrBayes">predict_pqrBayes</a></code>
</p>

<hr>
<h2 id='print.pqrBayes.select'>print a select.pqrBayes object</h2><span id='topic+print.pqrBayes.select'></span>

<h3>Description</h3>

<p>Print a summary of a select.pqrBayes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pqrBayes.select'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.pqrBayes.select_+3A_x">x</code></td>
<td>
<p>pqrBayes.select object.</p>
</td></tr>
<tr><td><code id="print.pqrBayes.select_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.pqrBayes.select_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pqrBayes.select">pqrBayes.select</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
