<!DOCTYPE html><html lang="en"><head><title>Help for package pQTLdata</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pQTLdata}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pQTLdata'><p>A summary of datasets</p></a></li>
<li><a href='#caprion'><p>Caprion panel</p></a></li>
<li><a href='#inf1'><p>Olink/INF1 panel</p></a></li>
<li><a href='#Olink_Explore_1536'><p>Olink/Explore 1536 panel</p></a></li>
<li><a href='#Olink_Explore_3072'><p>Olink/Explore 3072 panels</p></a></li>
<li><a href='#Olink_Explore_HT'><p>Olink/Explore HT panels</p></a></li>
<li><a href='#Olink_qPCR'><p>Olink/qPCR panels</p></a></li>
<li><a href='#Olink_Target_96'><p>Olink/Target 96 panels</p></a></li>
<li><a href='#scallop_inf1'><p>Supplementary table 3</p></a></li>
<li><a href='#seer1980'><p>Seer 1980 panel</p></a></li>
<li><a href='#SomaScan11k'><p>SomaScan 11k</p></a></li>
<li><a href='#SomaScan160410'><p>Somascan panel</p></a></li>
<li><a href='#SomaScanV4.1'><p>SomaScan v4.1</p></a></li>
<li><a href='#swath_ms'><p>SWATH-MS panel</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A Collection of Proteome Panels and Meta-Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-3-5</td>
</tr>
<tr>
<td>Description:</td>
<td>It aggregates protein panel data and metadata for protein quantitative
 trait locus (pQTL) analysis using 'pQTLtools' (<a href="https://jinghuazhao.github.io/pQTLtools/">https://jinghuazhao.github.io/pQTLtools/</a>).
 The package includes data from affinity-based panels such as 'Olink' (<a href="https://olink.com/">https://olink.com/</a>) 
 and 'SomaScan' (<a href="https://somalogic.com/">https://somalogic.com/</a>), as well as mass spectrometry-based panels from
 'CellCarta' (<a href="https://cellcarta.com/">https://cellcarta.com/</a>) and 'Seer' (<a href="https://seer.bio/">https://seer.bio/</a>). The metadata
 encompasses updated annotations and publication details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://jinghuazhao.github.io/pQTLdata/">https://jinghuazhao.github.io/pQTLdata/</a>,
<a href="https://jinghuazhao.github.io/pQTLdata/">https://jinghuazhao.github.io/pQTLdata/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>knitr, Rdpack</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Suggests:</td>
<td>dplyr, grid, EnsDb.Hsapiens.v75, ensembldb, IRanges,
org.Hs.eg.db, S4Vectors, VennDiagram</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>Yes</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-05 16:14:38 UTC; jhz22</td>
</tr>
<tr>
<td>Author:</td>
<td>Jing Hua Zhao <a href="https://orcid.org/0000-0002-1463-5870"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre] (0000-0003-4930-3582),
  Uwe Ligges [ctb],
  Benjamin Altmann [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jing Hua Zhao &lt;jinghuazhao@hotmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-07 11:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='pQTLdata'>A summary of datasets</h2><span id='topic+pQTLdata'></span><span id='topic+pQTLdata-package'></span>

<h3>Description</h3>

<p>It aggregates protein panel data and metadata for protein quantitative trait locus (pQTL) analysis using 'pQTLtools' (<a href="https://jinghuazhao.github.io/pQTLtools/">https://jinghuazhao.github.io/pQTLtools/</a>). The package includes data from affinity-based panels such as 'Olink' (<a href="https://olink.com/">https://olink.com/</a>) and 'SomaScan' (<a href="https://somalogic.com/">https://somalogic.com/</a>), as well as mass spectrometry-based panels from 'CellCarta' (<a href="https://cellcarta.com/">https://cellcarta.com/</a>) and 'Seer' (<a href="https://seer.bio/">https://seer.bio/</a>). The metadata encompasses updated annotations and publication details.
</p>


<h3>Details</h3>

<p>Available data are listed in the following table.</p>

<table>
<tr>
 <td style="text-align: left;">
   Objects </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>Datasets</strong> </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+caprion">caprion</a></code> </td><td style="text-align: left;"> Caprion panel </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+inf1">inf1</a></code> </td><td style="text-align: left;"> Olink/INF panel </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+Olink_Explore_1536">Olink_Explore_1536</a></code> </td><td style="text-align: left;"> Olink/NGS 1472 panels </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+Olink_Explore_3072">Olink_Explore_3072</a></code> </td><td style="text-align: left;"> Olink/Explore 3072 panels </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+Olink_Explore_HT">Olink_Explore_HT</a></code> </td><td style="text-align: left;"> Olink/Explore HT panels </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+Olink_Target_96">Olink_Target_96</a></code> </td><td style="text-align: left;"> Olink/Target 96 panels </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+Olink_qPCR">Olink_qPCR</a></code> </td><td style="text-align: left;"> Olink/qPCR panels </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+SomaScan160410">SomaScan160410</a></code> </td><td style="text-align: left;"> SomaScan panel </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+SomaScanV4.1">SomaScanV4.1</a></code> </td><td style="text-align: left;"> SomaScan v4.1 panel </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+SomaScan11k">SomaScan11k</a></code> </td><td style="text-align: left;"> SomaScan 11k panel </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+scallop_inf1">scallop_inf1</a></code> </td><td style="text-align: left;"> SCALLOP/INF meta-analysis results </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+seer1980">seer1980</a></code> </td><td style="text-align: left;"> ST1 from Suhre et al. (2024) bioRxiv </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+swath_ms">swath_ms</a></code> </td><td style="text-align: left;"> SWATH-MS panel </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>Installations</strong> </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
   EndNote/ </td><td style="text-align: left;"> Proteogenomics references </td>
</tr>
<tr>
 <td style="text-align: left;">
   Olink/ </td><td style="text-align: left;"> Olink-COVID analysis by MGH </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Some generic description for the datasets are as follows.
</p>

<ul>
<li><p> chr Chromosome.
</p>
</li>
<li><p> start Start position.
</p>
</li>
<li><p> end End position.
</p>
</li>
<li><p> gene Gene name.
</p>
</li>
<li><p> UniProt UniProt ID.
</p>
</li></ul>



<h3>Usage</h3>

<p>Vignettes on package usage:
</p>

<ul>
<li><p> An Overview of pQTLdata. <code>vignette("pQTLdata")</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Jing Hua Zhao in collaboration with other colleagues.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://jinghuazhao.github.io/pQTLdata/">https://jinghuazhao.github.io/pQTLdata/</a>
</p>
</li>
<li> <p><a href="https://jinghuazhao.github.io/pQTLdata/">https://jinghuazhao.github.io/pQTLdata/</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
# Olink-SomaScan panel overlap
p &lt;- list(setdiff(inf1$uniprot,"P23560"),
          setdiff(SomaScan160410$UniProt[!is.na(SomaScan160410$UniProt)],"P23560"))
cnames &lt;- c("INF1","SomaScan")
os &lt;- VennDiagram::venn.diagram(x = p, category.names=cnames, filename=NULL,
                                disable.logging = TRUE,height=8,width=8,units="in")
grid::grid.newpage()
grid::grid.draw(os)
m &lt;- merge(inf1,SomaScan160410,by.x="uniprot",by.y="UniProt")
u &lt;- setdiff(with(m,unique(uniprot)),"P23560")
o &lt;- subset(inf1,uniprot %in% u)
dim(o)
vars &lt;- c("UniProt","chr","start","end","extGene","Target","TargetFullName")
s &lt;- subset(SomaScan160410[vars], UniProt %in% u)
dim(s)
us &lt;- s[!duplicated(s),]
dim(us)
us

</code></pre>

<hr>
<h2 id='caprion'>Caprion panel</h2><span id='topic+caprion'></span>

<h3>Description</h3>

<p>Information based on Caprion pilot studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>caprion
</code></pre>


<h3>Format</h3>

<p>A data frame with 987 rows and 12 variables:
</p>

<dl>
<dt><code>Gene</code></dt><dd><p>HGNC symbols simplified in four instances</p>
</dd>
<dt><code>Gene.orig</code></dt><dd><p>HGNC symbol</p>
</dd>
<dt><code>Protein</code></dt><dd><p>Protein name as in UniProt</p>
</dd>
<dt><code>Accession</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Protein.Description</code></dt><dd><p>Detailed information on protein</p>
</dd>
<dt><code>GO.Cellular.Component</code></dt><dd><p>GO Ceullular component</p>
</dd>
<dt><code>GO.Function</code></dt><dd><p>GO function</p>
</dd>
<dt><code>GO.Process</code></dt><dd><p>GO process</p>
</dd>
<dt><code>ensGenes</code></dt><dd><p>Ensembl genes</p>
</dd>
<dt><code>chrom</code></dt><dd><p>chromosome</p>
</dd>
<dt><code>chr</code></dt><dd><p>chromosome</p>
</dd>
<dt><code>starts</code></dt><dd><p>start positions</p>
</dd>
<dt><code>ends</code></dt><dd><p>end positions</p>
</dd>
<dt><code>start</code></dt><dd><p>minimum start</p>
</dd>
<dt><code>end</code></dt><dd><p>maximum end</p>
</dd>
</dl>



<h3>Details</h3>

<p>See the Caprion repository involving its use.
</p>

<hr>
<h2 id='inf1'>Olink/INF1 panel</h2><span id='topic+inf1'></span>

<h3>Description</h3>

<p>The panel is based on SCALLOP-INF Zhao et al. (2023).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inf1
</code></pre>


<h3>Format</h3>

<p>A data frame with 92 rows and 9 variables:
</p>

<dl>
<dt><code>uniprot</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>prot</code></dt><dd><p>Protein</p>
</dd>
<dt><code>target</code></dt><dd><p>Protein target name</p>
</dd>
<dt><code>target.short</code></dt><dd><p>Protein target short name</p>
</dd>
<dt><code>gene</code></dt><dd><p>HGNC symbol</p>
</dd>
<dt><code>chr</code></dt><dd><p>chromosome (1-13,16-17,19-22)</p>
</dd>
<dt><code>start</code></dt><dd><p>start</p>
</dd>
<dt><code>end</code></dt><dd><p>end</p>
</dd>
<dt><code>chromosome</code></dt><dd><p>updated chromosomes</p>
</dd>
<dt><code>start38</code></dt><dd><p>start position under build 38</p>
</dd>
<dt><code>end38</code></dt><dd><p>end position under build 38</p>
</dd>
<dt><code>ensGene</code></dt><dd><p>Ensembl gene name</p>
</dd>
<dt><code>ensembl_gene_id</code></dt><dd><p>ENSEMBL gene</p>
</dd>
<dt><code>alt_name</code></dt><dd><p>recent name from www.uniprot.org</p>
</dd>
</dl>



<h3>Details</h3>

<p>Assembled for SCALLOP-INF
</p>


<h3>References</h3>

<p>Zhao JH, Stacey D, Eriksson N, Macdonald-Dunlop E, Hedman ÅK, Kalnapenkis A, Enroth S, Cozzetto D, Digby-Bell J, Marten J, Folkersen L, Herder C, Jonsson L, Bergen SE, Gieger C, Needham EJ, Surendran P, Team EBR, Paul DS, Polasek O, Thorand B, Grallert H, Roden M, Võsa U, Esko T, Hayward C, Johansson Å, Gyllensten U, Powell N, Hansson O, Mattsson-Carlgren N, Joshi PK, Danesh J, Padyukov L, Klareskog L, Landén M, Wilson JF, Siegbahn A, Wallentin L, Mälarstig A, Butterworth AS, Peters JE (2023).
&ldquo;Genetics of circulating inflammatory proteins identifies drivers of immune-mediated disease risk and therapeutic targets.&rdquo;
<em>Nature Immunology</em>, <b>24</b>(9), 1540-1551.
<a href="https://doi.org/10.1038/s41590-023-01588-w">doi:10.1038/s41590-023-01588-w</a>.
</p>

<hr>
<h2 id='Olink_Explore_1536'>Olink/Explore 1536 panel</h2><span id='topic+Olink_Explore_1536'></span>

<h3>Description</h3>

<p>Information based on pilot studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Olink_Explore_1536
</code></pre>


<h3>Format</h3>

<p>A data frame with 1,472 rows and 3 variables:
</p>

<dl>
<dt><code>UniProt</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Assay</code></dt><dd><p>Experimental assay</p>
</dd>
<dt><code>Panel</code></dt><dd><p>Olink panel</p>
</dd>
</dl>



<h3>Details</h3>

<p>Curated from R.
</p>

<hr>
<h2 id='Olink_Explore_3072'>Olink/Explore 3072 panels</h2><span id='topic+Olink_Explore_3072'></span>

<h3>Description</h3>

<p>Information on all qPCR panels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Olink_Explore_3072
</code></pre>


<h3>Format</h3>

<p>A data frame with 2,945 rows and 4 variables:
</p>

<dl>
<dt><code>UniProt.ID</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Protein.name</code></dt><dd><p>Protein name</p>
</dd>
<dt><code>Gene.name</code></dt><dd><p>Gene name</p>
</dd>
<dt><code>Explore.384.panel</code></dt><dd><p>Explore 384 panel</p>
</dd>
</dl>



<h3>Details</h3>

<p>Curated from Excel.
</p>

<hr>
<h2 id='Olink_Explore_HT'>Olink/Explore HT panels</h2><span id='topic+Olink_Explore_HT'></span>

<h3>Description</h3>

<p>Information on all qPCR panels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Olink_Explore_HT
</code></pre>


<h3>Format</h3>

<p>A data frame with 5,416 rows and 4 variables:
</p>

<dl>
<dt><code>Olink.ID</code></dt><dd><p>Olink id</p>
</dd>
<dt><code>UniProt.ID</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Protein.name</code></dt><dd><p>Protein name</p>
</dd>
<dt><code>Gene.name</code></dt><dd><p>Gene name</p>
</dd>
</dl>



<h3>Details</h3>

<p>Curated from Excel.
</p>

<hr>
<h2 id='Olink_qPCR'>Olink/qPCR panels</h2><span id='topic+Olink_qPCR'></span>

<h3>Description</h3>

<p>Information on all qPCR panels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Olink_qPCR
</code></pre>


<h3>Format</h3>

<p>A data frame with 1,112 rows and 7 variables:
</p>

<dl>
<dt><code>UniProt</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Panel</code></dt><dd><p>Panels</p>
</dd>
<dt><code>Target</code></dt><dd><p>Protein</p>
</dd>
<dt><code>gene</code></dt><dd><p>HGNC symbol</p>
</dd>
<dt><code>chr</code></dt><dd><p>Chromosome</p>
</dd>
<dt><code>start</code></dt><dd><p>start</p>
</dd>
<dt><code>end</code></dt><dd><p>end</p>
</dd>
</dl>



<h3>Details</h3>

<p>Curated from Excel.
</p>

<hr>
<h2 id='Olink_Target_96'>Olink/Target 96 panels</h2><span id='topic+Olink_Target_96'></span>

<h3>Description</h3>

<p>Information on all Target 96 panels. Individual panels are also available from the companion xlsx in the Olink/ directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Olink_Target_96
</code></pre>


<h3>Format</h3>

<p>A data frame with 1,116 rows and 3 variables:
</p>

<dl>
<dt><code>UniProt</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Protein</code></dt><dd><p>Protein</p>
</dd>
<dt><code>Panel</code></dt><dd><p>Panel</p>
</dd>
</dl>



<h3>Details</h3>

<p>Curated from Excel.
</p>

<hr>
<h2 id='scallop_inf1'>Supplementary table 3</h2><span id='topic+scallop_inf1'></span>

<h3>Description</h3>

<p>Supplementary information for Zhao et al. (2023).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scallop_inf1
</code></pre>


<h3>Format</h3>

<p>A data frame with 180 rows and 19 variables:
</p>

<dl>
<dt>UniProt</dt><dd><p>UnitProt ID</p>
</dd>
<dt>Protein</dt><dd><p>Protein name</p>
</dd>
<dt>Protein_gene_symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>Chromosome</dt><dd><p>Chromosome</p>
</dd>
<dt>Position</dt><dd><p>Position</p>
</dd>
<dt>cistrans</dt><dd><p>cis/trans</p>
</dd>
<dt>rsid</dt><dd><p>reference sequence ID</p>
</dd>
<dt>Effect_allele</dt><dd><p>Effect allele</p>
</dd>
<dt>Other_allele</dt><dd><p>Eeference allele</p>
</dd>
<dt>EAF</dt><dd><p>Effect allele frequency</p>
</dd>
<dt>b</dt><dd><p>b</p>
</dd>
<dt>SE</dt><dd><p>SE</p>
</dd>
<dt>log10P</dt><dd><p>log10(P)</p>
</dd>
<dt>Direction</dt><dd><p>Direction field in METAL output</p>
</dd>
<dt>HetISq</dt><dd><p><em>I</em><code class="reqn">^2</code></p>
</dd>
<dt>HetChiSq</dt><dd><p>Heterogeneity chi-square</p>
</dd>
<dt>HetDf</dt><dd><p>degrees of freedom</p>
</dd>
<dt>logHetP</dt><dd><p>Heterogeneity log10(P)</p>
</dd>
<dt>N</dt><dd><p>N</p>
</dd>
</dl>



<h3>References</h3>

<p>Zhao JH, Stacey D, Eriksson N, Macdonald-Dunlop E, Hedman ÅK, Kalnapenkis A, Enroth S, Cozzetto D, Digby-Bell J, Marten J, Folkersen L, Herder C, Jonsson L, Bergen SE, Gieger C, Needham EJ, Surendran P, Team EBR, Paul DS, Polasek O, Thorand B, Grallert H, Roden M, Võsa U, Esko T, Hayward C, Johansson Å, Gyllensten U, Powell N, Hansson O, Mattsson-Carlgren N, Joshi PK, Danesh J, Padyukov L, Klareskog L, Landén M, Wilson JF, Siegbahn A, Wallentin L, Mälarstig A, Butterworth AS, Peters JE (2023).
&ldquo;Genetics of circulating inflammatory proteins identifies drivers of immune-mediated disease risk and therapeutic targets.&rdquo;
<em>Nature Immunology</em>, <b>24</b>(9), 1540-1551.
<a href="https://doi.org/10.1038/s41590-023-01588-w">doi:10.1038/s41590-023-01588-w</a>.
</p>

<hr>
<h2 id='seer1980'>Seer 1980 panel</h2><span id='topic+seer1980'></span>

<h3>Description</h3>

<p>ST1 from Suhre et al. (2024).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seer1980
</code></pre>


<h3>Format</h3>

<p>A data frame with 1,980 rows:
</p>

<dl>
<dt><code>PID.NP</code></dt><dd><p>PID.NP</p>
</dd>
<dt><code>protein_ids</code></dt><dd><p>protein_ids</p>
</dd>
<dt><code>protein_names</code></dt><dd><p>protein_names</p>
</dd>
<dt><code>mapped.UniProtID</code></dt><dd><p>mapped.UniProtID</p>
</dd>
<dt><code>mapped_gene_id</code></dt><dd><p>mapped_gene_id</p>
</dd>
<dt><code>gene_name</code></dt><dd><p>gene_name</p>
</dd>
<dt><code>description</code></dt><dd><p>description</p>
</dd>
<dt><code>chr</code></dt><dd><p>chr</p>
</dd>
<dt><code>start</code></dt><dd><p>start</p>
</dd>
<dt><code>end</code></dt><dd><p>end</p>
</dd>
</dl>



<h3>Details</h3>

<p>As above.
</p>


<h3>References</h3>

<p>Suhre K, Chen Q, Halama A, Mendez K, Dahlin A, Stephan N, Thareja G, Sarwath H, Guturu H, Dwaraka VB, Batzoglou S, Schmidt F, Lasky-Su JA (2024).
&ldquo;A genome-wide association study of mass spectrometry proteomics using the Seer Proteograph platform.&rdquo;
<em>BioRxiv</em>.
<a href="https://doi.org/10.1101/2024.05.27.596028">doi:10.1101/2024.05.27.596028</a>.
</p>

<hr>
<h2 id='SomaScan11k'>SomaScan 11k</h2><span id='topic+SomaScan11k'></span>

<h3>Description</h3>

<p>This is also the latest panel
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SomaScan11k
</code></pre>


<h3>Format</h3>

<p>A data frame with 10,776 rows and 5 variables:
</p>

<dl>
<dt><code>Sequence.ID</code></dt><dd><p>Sequence ID</p>
</dd>
<dt><code>Full.Name</code></dt><dd><p>Full name</p>
</dd>
<dt><code>Target.Name</code></dt><dd><p>Target name</p>
</dd>
<dt><code>UniProt.ID</code></dt><dd><p>UniProt ID</p>
</dd>
<dt><code>Entrez.Gene.Name</code></dt><dd><p>Entrez gene name</p>
</dd>
</dl>



<h3>Details</h3>

<p>curated from SomaLogic website.
</p>


<h3>Source</h3>

<p><a href="https://somalogic.com/somascan-11k-assay/">https://somalogic.com/somascan-11k-assay/</a>
</p>

<hr>
<h2 id='SomaScan160410'>Somascan panel</h2><span id='topic+SomaScan160410'></span>

<h3>Description</h3>

<p>This is based on panel used in Sun et al. (2018).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SomaScan160410
</code></pre>


<h3>Format</h3>

<p>A data frame with 5,178 rows and 10 variables:
</p>

<dl>
<dt><code>SOMAMER_ID</code></dt><dd><p>Somamer id</p>
</dd>
<dt><code>UniProt</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>Target</code></dt><dd><p>Protein target</p>
</dd>
<dt><code>TargetFullName</code></dt><dd><p>Protein target full name</p>
</dd>
<dt><code>chr</code></dt><dd><p>chromosome (1-22,X,Y)</p>
</dd>
<dt><code>start</code></dt><dd><p>start</p>
</dd>
<dt><code>end</code></dt><dd><p>end</p>
</dd>
<dt><code>entGene</code></dt><dd><p>entrez gene</p>
</dd>
<dt><code>ensGene</code></dt><dd><p>ENSEMBL gene</p>
</dd>
<dt><code>extGene</code></dt><dd><p>external gene</p>
</dd>
</dl>



<h3>Details</h3>

<p>from the INTERVAL study.
</p>


<h3>References</h3>

<p>Sun BB, Maranville JC, Peters JE, Stacey D, Staley JR, Blackshaw J, Burgess S, Jiang T, Paige E, Surendran P, Oliver-Williams C, Kamat MA, Prins BP, Wilcox SK, Zimmerman ES, Chi A, Bansal N, Spain SL, Wood AM, Morrell NW, Bradley JR, Janjic N, Roberts DJ, Ouwehand WH, Todd JA, Soranzo N, Suhre K, Paul DS, Fox CS, Plenge RM, Danesh J, Runz H, Butterworth AS (2018).
&ldquo;Genomic atlas of the human plasma proteome.&rdquo;
<em>Nature</em>, <b>558</b>(7708), 73-79.
ISSN 1476-4687 (Electronic) 0028-0836 (Linking), <a href="https://doi.org/10.1038/s41586-018-0175-2">doi:10.1038/s41586-018-0175-2</a>.
</p>

<hr>
<h2 id='SomaScanV4.1'>SomaScan v4.1</h2><span id='topic+SomaScanV4.1'></span>

<h3>Description</h3>

<p>This is the 7k panel
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SomaScanV4.1
</code></pre>


<h3>Format</h3>

<p>A data frame with 7,288 rows and 6 variables:
</p>

<dl>
<dt><code>#</code></dt><dd><p>A serial number</p>
</dd>
<dt><code>SeqID</code></dt><dd><p>SeqID</p>
</dd>
<dt><code>Human.Target.or.Analyte</code></dt><dd><p>Human target/analyte</p>
</dd>
<dt><code>UniProt.ID</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>GeneID</code></dt><dd><p>HGNC symbol</p>
</dd>
<dt><code>Type</code></dt><dd><p>&quot;Protein&quot;</p>
</dd>
</dl>



<h3>Details</h3>

<p>obtained directly from SomaLogic.
</p>

<hr>
<h2 id='swath_ms'>SWATH-MS panel</h2><span id='topic+swath_ms'></span>

<h3>Description</h3>

<p>Curated during INTERVAL pilot study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swath_ms
</code></pre>


<h3>Format</h3>

<p>A data frame with 684 rows and 5 variables:
</p>

<dl>
<dt><code>Accession</code></dt><dd><p>UniProt id</p>
</dd>
<dt><code>accList</code></dt><dd><p>List of UniProt ids</p>
</dd>
<dt><code>uniprotName</code></dt><dd><p>Protein</p>
</dd>
<dt><code>ensGene</code></dt><dd><p>ENSEMBL gene</p>
</dd>
<dt><code>geneName</code></dt><dd><p>HGNC symbol</p>
</dd>
</dl>



<h3>Details</h3>

<p>As above.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
