<!DOCTYPE html><html><head><title>Help for package PRANA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PRANA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjpval'><p>adjpval</p></a></li>
<li><a href='#adjpval_specific_var'><p>adjpval_specific_var</p></a></li>
<li><a href='#coeff'><p>coeff</p></a></li>
<li><a href='#coeff_specific_var'><p>coeff_specific_var</p></a></li>
<li><a href='#combinedCOPDdat_RGO'><p>Chronic Obstructive Pulmonary Disease (COPD) data</p></a></li>
<li><a href='#EBS'><p>EBS</p></a></li>
<li><a href='#PRANA'><p>PRANA</p></a></li>
<li><a href='#sigDCGnames'><p>sigDCGnames</p></a></li>
<li><a href='#sigDCGtab'><p>sigDCGtab</p></a></li>
<li><a href='#thetahats'><p>thetahats</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pseudo-Value Regression Approach for Network Analysis (PRANA)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>A novel pseudo-value regression approach for the differential co-expression network analysis in expression data, which can incorporate additional clinical variables in the model. This is a direct regression modeling for the differential network analysis, and it is therefore computationally amenable for the most users. The full methodological details can be found in Ahn S et al (2023) &lt;<a href="https://doi.org/10.1186%2Fs12859-022-05123-w">doi:10.1186/s12859-022-05123-w</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dnapath, dplyr, parallel, robustbase, stats, minet</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-16 17:08:13 UTC; seungjunahn</td>
</tr>
<tr>
<td>Author:</td>
<td>Seungjun Ahn <a href="https://orcid.org/0000-0002-4816-8924"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut, trl],
  Somnath Datta [ctb, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Seungjun Ahn &lt;seungjun.ahn@mountsinai.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-17 00:00:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjpval'>adjpval</h2><span id='topic+adjpval'></span>

<h3>Description</h3>

<p>A function to retrieve a table with adjusted p-values after running PRANA.
The table includes all variables that were included in the pseudo-value regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjpval(PRANAres)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjpval_+3A_pranares">PRANAres</code></td>
<td>
<p>An object called after running PRANA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table that includes adjusted p-values for all variables included in the fitted model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
 groupA = index_grpA, groupB = index_grpB)

# Now, we want to keep the table with adjusted p-values only.
adjpval(PRANAres)
</code></pre>

<hr>
<h2 id='adjpval_specific_var'>adjpval_specific_var</h2><span id='topic+adjpval_specific_var'></span>

<h3>Description</h3>

<p>A function to retrieve a vector of adjusted p-values after running PRANA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjpval_specific_var(adjptab, varname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjpval_specific_var_+3A_adjptab">adjptab</code></td>
<td>
<p>A table that includes adjusted p-values for a specific variable.</p>
</td></tr>
<tr><td><code id="adjpval_specific_var_+3A_varname">varname</code></td>
<td>
<p>Specify the name of the variable of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of adjusted p-values for a single variable from the model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
groupA = index_grpA, groupB = index_grpB)

# Create an object to keep the table with adjusted p-values using adjpval() function.
adjpvaltab &lt;- adjpval(PRANAres)

# Retrieve a vector of adjusted p-values for a single variable of interest.
adjpval_specific_var(adjptab = adjpvaltab, varname = "currentsmoking")
</code></pre>

<hr>
<h2 id='coeff'>coeff</h2><span id='topic+coeff'></span>

<h3>Description</h3>

<p>A function to retrieve a table with coefficient estimates after running PRANA.
The table includes all variables that were included in the pseudo-value regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coeff(PRANAres)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coeff_+3A_pranares">PRANAres</code></td>
<td>
<p>An object called after running PRANA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table that includes coefficient estimates for all variables included in the fitted model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
# Then, create an object called `res` to call results later.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
 groupA = index_grpA, groupB = index_grpB)

# Now, we want to keep the table with coefficient estimates only.
coeff(PRANAres)
</code></pre>

<hr>
<h2 id='coeff_specific_var'>coeff_specific_var</h2><span id='topic+coeff_specific_var'></span>

<h3>Description</h3>

<p>A function to retrieve a vector of coefficient estimates for a specific variable of interest after running PRANA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coeff_specific_var(coefftab, varname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coeff_specific_var_+3A_coefftab">coefftab</code></td>
<td>
<p>A table that includes adjusted p-values for a specific variable.</p>
</td></tr>
<tr><td><code id="coeff_specific_var_+3A_varname">varname</code></td>
<td>
<p>Specify the name of the variable of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of coefficient estimates for a single variable from the model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
# Then, create an object called `res` to call results later.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
 groupA = index_grpA, groupB = index_grpB)

# Create an object to keep the table with coefficient estimates using coeff() function.
coefftab &lt;- coeff(PRANAres)

# Lastly, we use coeff_specific_var function to retrieve
# adjusted p-values for a single variable of interest.
coeff_specific_var(coefftab = coefftab, varname = "currentsmoking")
</code></pre>

<hr>
<h2 id='combinedCOPDdat_RGO'>Chronic Obstructive Pulmonary Disease (COPD) data</h2><span id='topic+combinedCOPDdat_RGO'></span>

<h3>Description</h3>

<p>A subset of the COPDGene study data, obtained from the Gene Expression Omnibus database. This contains expression and clinical covariates data.
All of the 28 genes contained in this data are identified as COPD-related by Sakornsakolpat et al. Nat Genet, 2019.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(combinedCOPDdat_RGO)
</code></pre>


<h3>Format</h3>

<p>combinedCOPDdat_RGO
</p>


<h3>References</h3>

<p>Sakornsakolpat P, Prokopenko D, Lamontagne M, et al. Genetic landscape of chronic obstructive pulmonary disease identifies heterogeneous cell-type and phenotype associations. Nat Genet. 2019;51(3):494–505.
(<a href="https://pubmed.ncbi.nlm.nih.gov/30804561/">PubMed</a>)
Ahn S, Grimes T, Datta S. A pseudo-value regression approach for differential network analysis of co-expression data. BMC Bioinformatics, 2023;24(1):8.
(<a href="https://pubmed.ncbi.nlm.nih.gov/36624383/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(combinedCOPDdat_RGO)
combinedCOPDdat_RGO
</code></pre>

<hr>
<h2 id='EBS'>EBS</h2><span id='topic+EBS'></span>

<h3>Description</h3>

<p>A function to calculate the adjusted p-value for each gene (Datta and Datta, 2005)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBS(pvo, alpha = 0.05, B = 500, h = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBS_+3A_pvo">pvo</code></td>
<td>
<p>An object with p-values estimated from the user-provided expression data</p>
</td></tr>
<tr><td><code id="EBS_+3A_alpha">alpha</code></td>
<td>
<p>a level of significance (default is 0.05)</p>
</td></tr>
<tr><td><code id="EBS_+3A_b">B</code></td>
<td>
<p>size of bootstrapping (default is 500)</p>
</td></tr>
<tr><td><code id="EBS_+3A_h">h</code></td>
<td>
<p>a bandwidth (default is 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of adjusted p-values for each gene.
</p>


<h3>References</h3>

<p>Datta, S and Datta, S (2005). Empirical Bayes screening of many p-values with applications to microarray studies. Bioinformatics, 21(9), 1987-1994
</p>

<hr>
<h2 id='PRANA'>PRANA</h2><span id='topic+PRANA'></span>

<h3>Description</h3>

<p>A pseudo-value regression approach for differential network analysis
that adjusts for additional covariates (PRANA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRANA(RNASeqdat, clindat, groupA, groupB)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PRANA_+3A_rnaseqdat">RNASeqdat</code></td>
<td>
<p>An RNA-Seq data with subjects in rows and genes in columns.</p>
</td></tr>
<tr><td><code id="PRANA_+3A_clindat">clindat</code></td>
<td>
<p>A data with clinical variables to be included in the regression
(e.g., binary group variable indicating
current smoking status, continuous age, ...)</p>
</td></tr>
<tr><td><code id="PRANA_+3A_groupa">groupA</code></td>
<td>
<p>Indices of the subjects in the first category (e.g., non-current smoker)
of binary group variable.</p>
</td></tr>
<tr><td><code id="PRANA_+3A_groupb">groupB</code></td>
<td>
<p>Indices of the subjects in the second category (e.g., current smoker)
of binary group variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing three data frame objects that summarize the results of PRANA. This includes
beta coefficients, p-values, and adjusted p-values via the empirical Bayes approach
for each predictor variables that are included in the regression model.
</p>


<h3>References</h3>

<p>Ahn S, Grimes T, Datta S. A pseudo-value regression approach for differential network analysis of co-expression data. BMC Bioinformatics, 2023;24(1):8
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
groupA = index_grpA, groupB = index_grpB)
</code></pre>

<hr>
<h2 id='sigDCGnames'>sigDCGnames</h2><span id='topic+sigDCGnames'></span>

<h3>Description</h3>

<p>A function to retrieve the name of genes that are significantly differentially connected (DC).
between two biological/clinical states (aka the main binary indicator) with the presence of additional covariate information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigDCGnames(adjptab, groupvar, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sigDCGnames_+3A_adjptab">adjptab</code></td>
<td>
<p>A table with adjusted p-values for all variables that were included in the pseudo-value regression model.</p>
</td></tr>
<tr><td><code id="sigDCGnames_+3A_groupvar">groupvar</code></td>
<td>
<p>Specify the name of binary indicator variable.</p>
</td></tr>
<tr><td><code id="sigDCGnames_+3A_alpha">alpha</code></td>
<td>
<p>A level of significance (e.g. 0.05).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Names of significantly DC genes (e.g. gene IDs) from PRANA.
If you need both adjusted p-values and names, please use sigDCGtab() instead.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
# Then, create an object called PRANA_Results to call results.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
groupA = index_grpA, groupB = index_grpB)

# Next, we want to call the table with adjusted p-values only.
adjptab &lt;- adjpval(PRANAres)

# Please specify the name of binary group indicator in sigDCGnames(groupvar = ).
sigDCGnames &lt;- sigDCGnames(adjptab = adjptab, groupvar = "currentsmoking", alpha = 0.05)
sigDCGnames
</code></pre>

<hr>
<h2 id='sigDCGtab'>sigDCGtab</h2><span id='topic+sigDCGtab'></span>

<h3>Description</h3>

<p>A function to retrieve the data frame that are significantly differentially connected (DC).
between two biological/clinical states (aka the main binary indicator) with the presence of additional covariate information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigDCGtab(adjptab, groupvar, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sigDCGtab_+3A_adjptab">adjptab</code></td>
<td>
<p>A table with adjusted p-values and names for the variable that the user specifies in the groupvar.</p>
</td></tr>
<tr><td><code id="sigDCGtab_+3A_groupvar">groupvar</code></td>
<td>
<p>Specify the name of binary indicator variable.</p>
</td></tr>
<tr><td><code id="sigDCGtab_+3A_alpha">alpha</code></td>
<td>
<p>A level of significance (e.g. 0.05).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Adjusted p-values and names of significantly DC genes (e.g. gene IDs) from PRANA.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
# Then, create an object called PRANA_Results to call results.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
 groupA = index_grpA, groupB = index_grpB)

# Next, we want to call the table with adjusted p-values.
adjptab &lt;- adjpval(PRANAres)

# Please specify the name of variable in sigDCGtab(groupvar = ).
sigDCGtab &lt;- sigDCGtab(adjptab = adjptab, groupvar = "currentsmoking", alpha = 0.05)
sigDCGtab
</code></pre>

<hr>
<h2 id='thetahats'>thetahats</h2><span id='topic+thetahats'></span>

<h3>Description</h3>

<p>A function to compute the total connectivity of each gene from the association matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>thetahats(asso.matinput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="thetahats_+3A_asso.matinput">asso.matinput</code></td>
<td>
<p>An association matrix that is estimated from the user-provided expression data is used
as an input to compute the total connectivity of each gene.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing total connectivity of each gene (i.e. continuous version of centrality measure of a network)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
