<!DOCTYPE html><html lang="en"><head><title>Help for package PRECAST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PRECAST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Add_embed'><p>Add embeddings for a Seurat object</p></a></li>
<li><a href='#AddAdjList'><p>Add adjacency matrix list for a PRECASTObj object</p></a></li>
<li><a href='#AddParSetting'><p>Add model settings  for a PRECASTObj object</p></a></li>
<li><a href='#AddTSNE'><p>Add tSNE embeddings for a Seurat object</p></a></li>
<li><a href='#AddUMAP'><p>Add UMAP embeddings for a Seurat object</p></a></li>
<li><a href='#boxPlot'><p>Boxplot for a matrix</p></a></li>
<li><a href='#chooseColors'><p>Choose color schema from a palette</p></a></li>
<li><a href='#coordinate_rotate'><p>Coordinates rotation for visualization</p></a></li>
<li><a href='#CreatePRECASTObject'><p>Create the PRECAST object with preprocessing step.</p></a></li>
<li><a href='#dimPlot'><p>Low-dimensional embeddings' plot</p></a></li>
<li><a href='#doHeatmap'><p>Heatmap for spots-by-feature matrix</p></a></li>
<li><a href='#drawFigs'><p>Draw a figure using a group of ggplot objects</p></a></li>
<li><a href='#featurePlot'><p>Spatial expression heatmap</p></a></li>
<li><a href='#firstup'><p>Set the first letter of a string vector to captial</p></a></li>
<li><a href='#getAdj_fixedNumber'><p>Calculate adjacency matrix by user-specified number of neighbors</p></a></li>
<li><a href='#getAdj_reg'><p>Calculate adjacency matrix for regular spatial coordinates.</p></a></li>
<li><a href='#Human_HK_genes'><p>Human housekeeping genes database</p></a></li>
<li><a href='#ICM.EM'><p>ICM-EM algorithm implementation</p></a></li>
<li><a href='#ICM.EM_structure'><p>ICM-EM algorithm implementation with organized paramters</p></a></li>
<li><a href='#IntegrateSpaData'><p>Integrate multiple SRT data</p></a></li>
<li><a href='#model_set'><p>PRECAST model setting</p></a></li>
<li><a href='#Mouse_HK_genes'><p>Mouse housekeeping genes database</p></a></li>
<li><a href='#plot_RGB'><p>Spatial RGB heatmap</p></a></li>
<li><a href='#plot_scatter'><p>Scatter plot for two-dimensional embeddings</p></a></li>
<li><a href='#PRECAST'><p>Fit a PRECAST model</p></a></li>
<li><a href='#PRECASTObj'><p>A simple PRECASTObj for example</p></a></li>
<li><a href='#PRECASTObj-class'><p>Each PRECASTObj object has a number of slots which store information.</p></a></li>
<li><a href='#selectIntFeatures'><p>Select common genes for multiple data batches</p></a></li>
<li><a href='#SelectModel'><p>Select best PRECAST model from candidated models</p></a></li>
<li><a href='#SpaPlot'><p>Spatial heatmap</p></a></li>
<li><a href='#volinPlot'><p>Volin/boxplot plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Embedding and Clustering with Alignment for Spatial Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Liu [aut, cre],
  Yi Yang [aut],
  Jin Liu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Liu &lt;liuweideng@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An efficient data integration method is provided for multiple spatial transcriptomics data with non-cluster-relevant effects such as the complex batch effects. It unifies spatial factor analysis simultaneously with spatial clustering and embedding alignment, requiring only partially shared cell/domain clusters across datasets. More details can be referred to Wei Liu, et al. (2023) &lt;<a href="https://doi.org/10.1038%2Fs41467-023-35947-w">doi:10.1038/s41467-023-35947-w</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>parallel, gtools, R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>GiRaF, MASS, Matrix, mclust, methods, purrr, utils, Seurat,
cowplot, patchwork, scater, pbapply, ggthemes, dplyr, ggplot2,
stats, DR.SC, scales, ggpubr, graphics, colorspace, Rcpp (&ge;
1.0.5)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/feiyoung/PRECAST">https://github.com/feiyoung/PRECAST</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/feiyoung/PRECAST/issues">https://github.com/feiyoung/PRECAST/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-19 00:44:14 UTC; Liuxianju</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-19 08:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Add_embed'>Add embeddings for a Seurat object</h2><span id='topic+Add_embed'></span>

<h3>Description</h3>

<p>Add embeddings for a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  Add_embed(embed, seu, embed_name='tSNE' , assay = "RNA")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Add_embed_+3A_embed">embed</code></td>
<td>
<p>an embedding matrix.</p>
</td></tr>
<tr><td><code id="Add_embed_+3A_seu">seu</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="Add_embed_+3A_embed_name">embed_name</code></td>
<td>
<p>an optional string, the name of embeddings.</p>
</td></tr>
<tr><td><code id="Add_embed_+3A_assay">assay</code></td>
<td>
<p>Name of assay that that embed is being put</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a revised Seurat object by adding a embedding matrix to the Reduc slot in Seurat object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>

<hr>
<h2 id='AddAdjList'>Add adjacency matrix list for a PRECASTObj object</h2><span id='topic+AddAdjList'></span>

<h3>Description</h3>

<p>Add adjacency matrix list for a PRECASTObj object to prepare for PRECAST model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  AddAdjList(PRECASTObj, type="fixed_distance", platform="Visium", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AddAdjList_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <a href="#topic+CreatePRECASTObject">CreatePRECASTObject</a>.</p>
</td></tr>
<tr><td><code id="AddAdjList_+3A_type">type</code></td>
<td>
<p>an optional string, specify which type of neighbors' definition. Here we provide two definition: one is &quot;fixed_distance&quot;, the other is &quot;fixed_number&quot;.</p>
</td></tr>
<tr><td><code id="AddAdjList_+3A_platform">platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as &quot;Visium&quot;. There are more platforms to be chosen, including &quot;Visuim&quot;, &quot;ST&quot; and &quot;Other_SRT&quot; (&quot;Other_SRT&quot; represents the other SRT platforms except for 'Visium' and 'ST'), which means there are spatial coordinates information in the metadata of PRECASTObj. The platform helps to calculate the adjacency matrix by defining the neighborhoods when type=&quot;fixed_distance&quot; is chosen.</p>
</td></tr>
<tr><td><code id="AddAdjList_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <a href="DR.SC.html#topic+getAdj">getAdj</a>, <a href="DR.SC.html#topic+getAdj_auto">getAdj_auto</a>  and <a href="#topic+getAdj_fixedNumber">getAdj_fixedNumber</a> funciton.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the type = &quot;fixed_distance&quot;, then the spots within the Euclidean distance cutoffs from one spot are regarded as the neighbors of this spot. When the type = &quot;fixed_number&quot;,  the K-nearest spots are regarded as the neighbors of each spot. 
</p>


<h3>Value</h3>

<p>Return a revised PRECASTObj object by adding the adjacency matrix list.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p><a href="#topic+AddParSetting">AddParSetting</a>.
</p>

<hr>
<h2 id='AddParSetting'>Add model settings  for a PRECASTObj object</h2><span id='topic+AddParSetting'></span>

<h3>Description</h3>

<p>The main interface function provides serveral PRECAST submodels, so a model setting is required to specified in advance  for a PRECASTObj object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  AddParSetting(PRECASTObj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AddParSetting_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <a href="#topic+CreatePRECASTObject">CreatePRECASTObject</a>.</p>
</td></tr>
<tr><td><code id="AddParSetting_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <a href="#topic+model_set">model_set</a> funciton.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing 
</p>


<h3>Value</h3>

<p>Return a revised PRECASTObj object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(PRECASTObj)
  PRECASTObj &lt;-AddParSetting(PRECASTObj)
  PRECASTObj@parameterList
</code></pre>

<hr>
<h2 id='AddTSNE'>Add tSNE embeddings for a Seurat object</h2><span id='topic+AddTSNE'></span>

<h3>Description</h3>

<p>Run t-SNE dimensionality reduction on selected features. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  AddTSNE(seuInt, n_comp=3, reduction='PRECAST', assay='PRE_CAST', seed=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AddTSNE_+3A_seuint">seuInt</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="AddTSNE_+3A_n_comp">n_comp</code></td>
<td>
<p>an optional positive integer, specify the number of features to be extracted.</p>
</td></tr>
<tr><td><code id="AddTSNE_+3A_reduction">reduction</code></td>
<td>
<p>an optional string, means which dimensional reduction (e.g. PRECAST, PCA) to use for the tSNE. Default is PRECAST.</p>
</td></tr>
<tr><td><code id="AddTSNE_+3A_assay">assay</code></td>
<td>
<p>Name of assay that that t-SNE is being run on.</p>
</td></tr>
<tr><td><code id="AddTSNE_+3A_seed">seed</code></td>
<td>
<p>an  optional integer, the random seed to evaluate tSNE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a revised Seurat object by adding tSNE reduction object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>

<hr>
<h2 id='AddUMAP'>Add UMAP embeddings for a Seurat object</h2><span id='topic+AddUMAP'></span>

<h3>Description</h3>

<p>Run UMAP dimensionality reduction on selected features. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  AddUMAP(seuInt, n_comp=3, reduction='PRECAST', assay='PRE_CAST', seed=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AddUMAP_+3A_seuint">seuInt</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="AddUMAP_+3A_n_comp">n_comp</code></td>
<td>
<p>an optional positive integer, specify the number of features to be extracted.</p>
</td></tr>
<tr><td><code id="AddUMAP_+3A_reduction">reduction</code></td>
<td>
<p>an optional string, means which dimensional reduction (e.g. PRECAST, PCA) to use for the UMAP. Default is PRECAST.</p>
</td></tr>
<tr><td><code id="AddUMAP_+3A_assay">assay</code></td>
<td>
<p>Name of assay that that t-SNE is being run on.</p>
</td></tr>
<tr><td><code id="AddUMAP_+3A_seed">seed</code></td>
<td>
<p>an  optional integer, the random seed to evaluate UMAP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a revised Seurat object by adding UMAP reduction object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>

<hr>
<h2 id='boxPlot'>Boxplot for a matrix</h2><span id='topic+boxPlot'></span>

<h3>Description</h3>

<p>Boxplot for a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  boxPlot(mat, ylabel='ARI', cols=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="boxPlot_+3A_mat">mat</code></td>
<td>
<p>a matrix with columns.</p>
</td></tr>
<tr><td><code id="boxPlot_+3A_ylabel">ylabel</code></td>
<td>
<p>an optional string, the name of ylabel.</p>
</td></tr>
<tr><td><code id="boxPlot_+3A_cols">cols</code></td>
<td>
<p>colors used in the plot</p>
</td></tr>
<tr><td><code id="boxPlot_+3A_...">...</code></td>
<td>
<p>Other parameters passed to geom_boxplot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   mat &lt;- matrix(runif(100*3, 0.6, 1), 100, 3)
   colnames(mat) &lt;- paste0("Method", 1:3)
   boxPlot(mat)
</code></pre>

<hr>
<h2 id='chooseColors'>Choose color schema from a palette</h2><span id='topic+chooseColors'></span>

<h3>Description</h3>

<p>Choose color schema from a palette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chooseColors(
  palettes_name = c("Nature 10", "Light 13", "Classic 20", "Blink 23", "Hue n"),
  n_colors = 7,
  alpha = 1,
  plot_colors = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chooseColors_+3A_palettes_name">palettes_name</code></td>
<td>
<p>a string, the palette name, one of &quot;Nature 10&quot;, &quot;Light 13&quot;, &quot;Classic 20&quot;, &quot;Blink 23&quot; and &quot;Hue n&quot;, default as 'Nature 10'.</p>
</td></tr>
<tr><td><code id="chooseColors_+3A_n_colors">n_colors</code></td>
<td>
<p>a positive integer, the number of colors.</p>
</td></tr>
<tr><td><code id="chooseColors_+3A_alpha">alpha</code></td>
<td>
<p>a positive real, the transparency of the color.</p>
</td></tr>
<tr><td><code id="chooseColors_+3A_plot_colors">plot_colors</code></td>
<td>
<p>a logical value, whether plot the selected colors.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>chooseColors()

</code></pre>

<hr>
<h2 id='coordinate_rotate'>Coordinates rotation for visualization</h2><span id='topic+coordinate_rotate'></span>

<h3>Description</h3>

<p>Coordinates rotation for visualization. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  coordinate_rotate(pos, theta=0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coordinate_rotate_+3A_pos">pos</code></td>
<td>
<p>a matrix, the n-by-d coordinates, where n is the number of coordinates, d is the dimension of coordinates.</p>
</td></tr>
<tr><td><code id="coordinate_rotate_+3A_theta">theta</code></td>
<td>
<p>a real number, the angle for counter-clock-wise rotation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a rotated coordinate matrix.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    x &lt;- 1:100
    pos &lt;- cbind(x, sin(pi/4*x))
    oldpar &lt;- par(mfrow = c(1,2))
    plot(pos)
    plot(coordinate_rotate(pos, 40))
    par(oldpar)
    
</code></pre>

<hr>
<h2 id='CreatePRECASTObject'>Create the PRECAST object with preprocessing step.</h2><span id='topic+CreatePRECASTObject'></span>

<h3>Description</h3>

<p>Create the PRECAST object with preprocessing step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreatePRECASTObject(seuList,  project = "PRECAST",  gene.number=2000, 
                    selectGenesMethod='SPARK-X',numCores_sparkx=1,  
                    customGenelist=NULL, premin.spots = 20,  
                    premin.features=20, postmin.spots=15, postmin.features=15,
                              rawData.preserve=FALSE,verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreatePRECASTObject_+3A_seulist">seuList</code></td>
<td>
<p>a list consisting of Seurat objects, where each object is a SRT data batch. The default assay of each Seurat object will be used for data preprocessing and followed model fitting. The specified format about seuList argument can be referred to the details and example.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_project">project</code></td>
<td>
<p>An optional string, name of the project, default as &quot;PRECAST&quot;.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_gene.number">gene.number</code></td>
<td>
<p>an optional integer, the number of top spatially variable genes (SVGs) or highly variable  genes (HVGs) to be chosen.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_selectgenesmethod">selectGenesMethod</code></td>
<td>
<p>an optional integer, the method to select genes for each sample. It supports 'SPARK-X' and 'HVGs' to select genes now. Users can provide self-selected genes using customGenelist argument.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_numcores_sparkx">numCores_sparkx</code></td>
<td>
<p>an optional integer, specify the number of CPU cores in SPARK package to use when selecting spatial genes.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_customgenelist">customGenelist</code></td>
<td>
<p>an optional string vector, the list of user specified genes to be used for PRECAST model fitting. If this argument is given, SVGs/HVGs will not be selected.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_premin.spots">premin.spots</code></td>
<td>
<p>An optional integer, the features (genes) are retained in raw data filtering step with at least premin.spots number of spots, default is 20.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_premin.features">premin.features</code></td>
<td>
<p>An optional integer, the locations are retained in raw data filtering step with at least premin.features number of  nonzero-count features (genes), default is 20.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_postmin.spots">postmin.spots</code></td>
<td>
<p>An optional integer, the features (genes) are retained in filtering step after common genes selected among all data batches with at least postmin.spots number of spots, default is 15.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_postmin.features">postmin.features</code></td>
<td>
<p>An optional integer, the locations are retained in filtering step after common genes selected among all data batches  with at least postmin.features number of  nonzero-count features (genes), default is 15.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_rawdata.preserve">rawData.preserve</code></td>
<td>
<p>An optional logical value, whether preserve the raw seuList data.</p>
</td></tr>
<tr><td><code id="CreatePRECASTObject_+3A_verbose">verbose</code></td>
<td>
<p>whether display the message in the creating process.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>seuList is a <a href="base.html#topic+list">list</a>  with Seurat object as component, and each Seurat object includes the raw expression count matrix, spatial coordinates and meta data for each data batch, where the spatial coordinates information must be saved in the metadata of Seurat, named &quot;row&quot; and &quot;col&quot; for each data batch.
</p>


<h3>Value</h3>

<p>Returns PRECAST object prepared for PRECAST model fitting. See <a href="#topic+PRECASTObj-class">PRECASTObj-class</a> for more details.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(PRECASTObj)
  library(Seurat)
  seuList &lt;- PRECASTObj@seulist
  ## Check the input of seuList for create PRECAST object.
  ## Check the default assay for each data batch
  lapply(seuList, DefaultAssay)
  ## Check the spatial coordinates in the meta data named "row" and "col".
  head(seuList[[1]]@meta.data)
  ## Then create PRECAST object using this seuList.
  ## For convenience, we show the  user-specified genes' list for creating PRECAST object.
  ## Users can use SVGs from SPARK-X or HVGs.
  PRECASTObj2 &lt;- CreatePRECASTObject(seuList, 
   customGenelist= row.names(seuList[[1]]), verbose=FALSE)


</code></pre>

<hr>
<h2 id='dimPlot'>Low-dimensional embeddings' plot</h2><span id='topic+dimPlot'></span>

<h3>Description</h3>

<p>Low-dimensional embeddings' plot colored by a specified meta data in the Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  dimPlot(seuInt, item=NULL, reduction=NULL, point_size=1,text_size=16, 
                    cols=NULL,font_family='', border_col="gray10",
                    fill_col="white", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dimPlot_+3A_seuint">seuInt</code></td>
<td>
<p>an object named &quot;Seurat&quot;. </p>
</td></tr>
<tr><td><code id="dimPlot_+3A_item">item</code></td>
<td>
<p>the item used for coloring the plot in the meta data of seuInt object.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_reduction">reduction</code></td>
<td>
<p>the reduction used for plot in the seuInt object. If reduction is null, the last added one is used for plotting.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_point_size">point_size</code></td>
<td>
<p>the size of point in the scatter plot.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_text_size">text_size</code></td>
<td>
<p>the text size in the plot.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_cols">cols</code></td>
<td>
<p>colors used in the plot</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_font_family">font_family</code></td>
<td>
<p>the font family used for the plot.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_border_col">border_col</code></td>
<td>
<p>the border color in the plot.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_fill_col">fill_col</code></td>
<td>
<p>the  color used in backgroup.</p>
</td></tr>
<tr><td><code id="dimPlot_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="#topic+plot_scatter">plot_scatter</a></code></p>
</td></tr></table>
<p>.
</p>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(PRECASTObj)
  PRECASTObj &lt;- SelectModel(PRECASTObj)
  seuInt &lt;- IntegrateSpaData(PRECASTObj, species='unknown')
  dimPlot(seuInt, reduction = 'PRECAST')
  ## or use the Seurat::DimPlot(seuInt, reduction = 'PRECAST')
  
</code></pre>

<hr>
<h2 id='doHeatmap'>Heatmap for spots-by-feature matrix</h2><span id='topic+doHeatmap'></span>

<h3>Description</h3>

<p>Plot  heatmap for a Seurat object with expressioin data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  doHeatmap(seu, features=NULL, cell_label='Cell type', grp_label = FALSE,
                      pt_size=4, grp_color=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="doHeatmap_+3A_seu">seu</code></td>
<td>
<p>an object named &quot;Seurat&quot;. The object of class &quot;Seurat&quot; must include slot &quot;scale.data&quot;.</p>
</td></tr>
<tr><td><code id="doHeatmap_+3A_features">features</code></td>
<td>
<p>an optional string vector, the features to be plotted.</p>
</td></tr>
<tr><td><code id="doHeatmap_+3A_cell_label">cell_label</code></td>
<td>
<p>an optional string, the name of legend.</p>
</td></tr>
<tr><td><code id="doHeatmap_+3A_grp_label">grp_label</code></td>
<td>
<p>an optional logical value, whether display the group names.</p>
</td></tr>
<tr><td><code id="doHeatmap_+3A_pt_size">pt_size</code></td>
<td>
<p>the point size used in the plot</p>
</td></tr>
<tr><td><code id="doHeatmap_+3A_grp_color">grp_color</code></td>
<td>
<p>the colors to use for the group color bar.</p>
</td></tr>
<tr><td><code id="doHeatmap_+3A_...">...</code></td>
<td>
<p>Other paramters passed to <a href="Seurat.html#topic+DoHeatmap">DoHeatmap</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p><a href="#topic+featurePlot">featurePlot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  library(Seurat)
  data(PRECASTObj)
  PRECASTObj &lt;- SelectModel(PRECASTObj)
  seuInt &lt;- IntegrateSpaData(PRECASTObj, species='unknown')
  seuInt &lt;- ScaleData(seuInt)
  doHeatmap(seuInt, features=row.names(seuInt)[1:5])
  
</code></pre>

<hr>
<h2 id='drawFigs'>Draw a figure using a group of ggplot objects</h2><span id='topic+drawFigs'></span>

<h3>Description</h3>

<p>Draw a figure using a group of ggplot objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawFigs(
  pList,
  layout.dim = NULL,
  common.legend = FALSE,
  legend.position = "right",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="drawFigs_+3A_plist">pList</code></td>
<td>
<p>a list with component ggplot objects.</p>
</td></tr>
<tr><td><code id="drawFigs_+3A_layout.dim">layout.dim</code></td>
<td>
<p>a integer vector with length 2, the layout of subplots in rows and columns.</p>
</td></tr>
<tr><td><code id="drawFigs_+3A_common.legend">common.legend</code></td>
<td>
<p>a logical value, whether use common legend for all subplots.</p>
</td></tr>
<tr><td><code id="drawFigs_+3A_legend.position">legend.position</code></td>
<td>
<p>a string, the position of legend.</p>
</td></tr>
<tr><td><code id="drawFigs_+3A_...">...</code></td>
<td>
<p>other arguments that pass to <code><a href="ggpubr.html#topic+ggarrange">ggarrange</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return a new ggplot object.
</p>

<hr>
<h2 id='featurePlot'>Spatial expression heatmap</h2><span id='topic+featurePlot'></span>

<h3>Description</h3>

<p>Plot spatial heatmap for a feature of Seurat object with spatial transcriptomics data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  featurePlot(seu, feature=NULL, cols=NULL, pt_size=1, title_size =16, quant=0.5, 
  assay='RNA' , reduction="position")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="featurePlot_+3A_seu">seu</code></td>
<td>
<p>an object named &quot;Seurat&quot;. The object of class &quot;Seurat&quot; must include slot &quot;scale.data&quot;.</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_feature">feature</code></td>
<td>
<p>an optional string, specify the name of feature to be plotted. If it is null, the first feature will be plotted.</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_cols">cols</code></td>
<td>
<p>colors used in the plot</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_pt_size">pt_size</code></td>
<td>
<p>the size of point in the spatial heatmap plot.</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_title_size">title_size</code></td>
<td>
<p>the title size used for the plot.</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_quant">quant</code></td>
<td>
<p>the quantile value to generate the gradient color map.</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_assay">assay</code></td>
<td>
<p>the assay selected for plot.</p>
</td></tr>
<tr><td><code id="featurePlot_+3A_reduction">reduction</code></td>
<td>
<p>the Reduc object for plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  library(Seurat)
  data(PRECASTObj)
  PRECASTObj &lt;- SelectModel(PRECASTObj)
  seuInt &lt;- IntegrateSpaData(PRECASTObj, species='unknown')
  seuInt &lt;- ScaleData(seuInt)
  featurePlot(seuInt, assay='PRE_CAST')
  
</code></pre>

<hr>
<h2 id='firstup'>Set the first letter of a string vector to captial</h2><span id='topic+firstup'></span>

<h3>Description</h3>

<p>Set the first letter of a string vector to captial.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  firstup(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="firstup_+3A_x">x</code></td>
<td>
<p>a string vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a string vector with first letter capital.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x &lt;- c("good", "Morning")
  firstup(x)
</code></pre>

<hr>
<h2 id='getAdj_fixedNumber'>Calculate adjacency matrix by user-specified number of neighbors</h2><span id='topic+getAdj_fixedNumber'></span>

<h3>Description</h3>

<p>an efficient function to find the neighborhood based on the matrix of position and a  user-specified number of neighbors of each spot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAdj_fixedNumber(pos, number=6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAdj_fixedNumber_+3A_pos">pos</code></td>
<td>
<p>is a n-by-d matrix of position, where n is the number of spots, and d is the dimension of coordinates.</p>
</td></tr>
<tr><td><code id="getAdj_fixedNumber_+3A_number">number</code></td>
<td>
<p>is the number of neighbors of each spot. Euclidean distance to decide whether a spot is an neighborhood of another spot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse matrix containing the neighbourhood.
</p>


<h3>See Also</h3>

<p><a href="DR.SC.html#topic+getAdj_auto">getAdj_auto</a>, <a href="DR.SC.html#topic+getAdj">getAdj</a>.
</p>

<hr>
<h2 id='getAdj_reg'>Calculate adjacency matrix for regular spatial coordinates.</h2><span id='topic+getAdj_reg'></span>

<h3>Description</h3>

<p>Calculate adjacency matrix for regular spatial coordinates from ST or Visium platform.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAdj_reg(pos, platform= "Visium")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAdj_reg_+3A_pos">pos</code></td>
<td>
<p>is a n-by-d matrix of position, where n is the number of spots, and d is the dimension of coordinates.</p>
</td></tr>
<tr><td><code id="getAdj_reg_+3A_platform">platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as &quot;Visium&quot;, and only support &quot;ST&quot; and &quot;Visium&quot; platform. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse matrix containing the neighbourhood.
</p>


<h3>See Also</h3>

<p><a href="DR.SC.html#topic+getAdj_auto">getAdj_auto</a>, <a href="DR.SC.html#topic+getAdj">getAdj</a>, <a href="#topic+getAdj_fixedNumber">getAdj_fixedNumber</a>.
</p>

<hr>
<h2 id='Human_HK_genes'>Human housekeeping genes database</h2><span id='topic+Human_HK_genes'></span>

<h3>Description</h3>

<p>Human housekeeping genes database.
</p>


<h3>Details</h3>

<p>This data is a <a href="base.html#topic+data.frame">data.frame</a> and include the Human housekeeping genes information in the columns named &quot;Gene&quot;  and  &quot;Ensembl&quot;.
</p>

<hr>
<h2 id='ICM.EM'>ICM-EM algorithm implementation</h2><span id='topic+ICM.EM'></span>

<h3>Description</h3>

<p>ICM-EM algorithm for fitting PRECAST model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ICM.EM(XList, q, K, AdjList=NULL,  Adjlist_car=NULL, posList = NULL, 
      platform = "ST", beta_grid=seq(0.2,4, by=0.2),maxIter_ICM=6,
      maxIter=20, epsLogLik=1e-5, verbose=TRUE,mix_prop_heter=TRUE, 
      Sigma_equal=FALSE, Sigma_diag=TRUE,error_heter=TRUE, Sp2=TRUE,
      wpca_int=FALSE, int.model='EEE', seed=1,coreNum = 1, coreNum_int=coreNum)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ICM.EM_+3A_xlist">XList</code></td>
<td>
<p>an M-length list consisting of multiple matrices with class <code>dgCMatrix</code> or <code>matrix</code> that specify the log-normalization gene expression matrix for each data sample used for PRECAST model.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_q">q</code></td>
<td>
<p>a positive integer, specify the number of latent features to be extracted, default as 15.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_k">K</code></td>
<td>
<p>a positive  integer allowing scalar or vector, specify the number of clusters in model fitting.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_adjlist">AdjList</code></td>
<td>
<p>an M-length list of sparse matrices with class <code>dgCMatrix</code>, specify the adjacency matrix used for Potts model in PRECAST. We provide this interface for those users  who would like to define the adjacency matrix by their own.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_adjlist_car">Adjlist_car</code></td>
<td>
<p>an M-length list of sparse matrices with class <code>dgCMatrix</code>, specify the adjacency matrix used for CAR model in PRECAST, default as AdjList in the Potts model. We provide this interface for those users  who would like to use the different adjacency matrix in CAR model.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_poslist">posList</code></td>
<td>
<p>an M-length list composed by spatial coordinate matrix for each data sample.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_platform">platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as &quot;Visium&quot;.
There are many platforms to be supported, including (&quot;Visuim&quot;, &quot;ST&quot;, &quot;SeqFISH&quot;, 'merFISH', 'slide-seqv2', 'seqscope', &quot;HDST&quot;). If AdjList is not given, the The platform helps to calculate the adjacency matrix by defining the neighbors.</p>
</td></tr> 
<tr><td><code id="ICM.EM_+3A_beta_grid">beta_grid</code></td>
<td>
<p>an optional vector of positive value, the candidate set of the smoothing parameter to be searched by the grid-search optimization approach.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_maxiter_icm">maxIter_ICM</code></td>
<td>
<p>an optional positive value, represents the maximum iterations of ICM.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_maxiter">maxIter</code></td>
<td>
<p>an optional positive value, represents the maximum iterations of EM.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_epsloglik">epsLogLik</code></td>
<td>
<p>an optional positive vlaue, tolerance vlaue of relative variation rate of the observed pseudo log-loglikelihood value, defualt as '1e-5'.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_verbose">verbose</code></td>
<td>
<p>an optional logical value, whether output the information of the ICM-EM algorithm.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_mix_prop_heter">mix_prop_heter</code></td>
<td>
<p>an optional logical value, specify whether betar are distict, default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_sigma_equal">Sigma_equal</code></td>
<td>
<p>an optional logical value, specify whether Sigmaks are equal, default as FALSE.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_sigma_diag">Sigma_diag</code></td>
<td>
<p>an optional logical value, specify whether Sigmaks are diagonal matrices, default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_error_heter">error_heter</code></td>
<td>
<p>an optional logical value, whether use the heterogenous error for DR-SC model, default as <code>TRUE</code>. If <code>error_heter=FALSE</code>, then the homogenuous error is used for probabilistic PCA model in PRECAST.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_sp2">Sp2</code></td>
<td>
<p>an optional logical value, whether add the ICAR model component in the model, default as TRUE. We provide this interface for those users  who don't want to include the  ICAR model.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_wpca_int">wpca_int</code></td>
<td>
<p>an optional logical value, means whether use the weighted PCA to obtain the initial values of loadings and other paramters, default as <code>FALSE</code> which means the ordinary PCA  is used.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_int.model">int.model</code></td>
<td>
<p>an optional string, specify which Gaussian mixture model is used in evaluting the initial values for PRECAST, default as &quot;EEE&quot;; and see <code>Mclust</code> for more models' names.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_seed">seed</code></td>
<td>
<p>an optional integer, the random seed in fitting PRECAST model.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_corenum">coreNum</code></td>
<td>
<p>an optional positive integer, means the number of thread used in parallel computating.</p>
</td></tr>
<tr><td><code id="ICM.EM_+3A_corenum_int">coreNum_int</code></td>
<td>
<p>an optional positive integer, means the number of cores used in parallel computation for initial values when <code>K</code> is a vector, default as same as <code>coreNum</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>ICM.EM returns a <a href="base.html#topic+list">list</a> with class &quot;SeqKiDRSC_Object&quot; with the number of components equal to the length of <code>K</code>, where each component includes the model fitting results for one number of cluster and is a list consisting of following components:
</p>
<table role = "presentation">
<tr><td><code>cluster</code></td>
<td>
<p>an M-length list that includes the inferred class labels for each data sample.</p>
</td></tr>
<tr><td><code>hZ</code></td>
<td>
<p>an M-length list that includes the batch corrected low-dimensional embeddings for each data sample.</p>
</td></tr>
<tr><td><code>hV</code></td>
<td>
<p>an M-length list that includes the estimate the ICAR component for each sample.</p>
</td></tr>
<tr><td><code>Rf</code></td>
<td>
<p>an M-length list that includes the posterior probability of domain clusters for each sample.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>an M-length vector that includes the estimated smoothing parameters for each sample.</p>
</td></tr>
<tr><td><code>Mu</code></td>
<td>
<p>mean vectors of mixtures components.</p>
</td></tr>
<tr><td><code>Sigma</code></td>
<td>
<p>covariance matrix of mixtures components.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>estimated loading matrix</p>
</td></tr>
<tr><td><code>Lam</code></td>
<td>
<p>estimated variance of errors in probabilistic PCA model</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>
<p>pseudo observed log-likelihood.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://www.nature.com/articles/s41467-023-35947-w">Wei Liu, Liao, X., Luo, Z. et al, Jin Liu* (2023). Probabilistic embedding, clustering, and alignment for integrating spatial transcriptomics data with PRECAST. Nature Communications, 14, 296</a>
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
  library(Matrix)
  q &lt;- 10; K &lt;- 4
  data(PRECASTObj)
  posList &lt;- lapply(PRECASTObj@seulist, function(x) cbind(x$row, x$col))
  AdjList &lt;- lapply(posList, getAdj_reg, platform='ST')
  XList &lt;- lapply(PRECASTObj@seulist, function(x) t(x[['RNA']]@data))
  XList &lt;- lapply(XList, scale, scale=FALSE)
  ## For illustration, maxIter is set to 4
  resList &lt;- ICM.EM(XList,AdjList = AdjList, maxIter=4,
                   q=q, K=K, verbose=TRUE)

</code></pre>

<hr>
<h2 id='ICM.EM_structure'>ICM-EM algorithm implementation with organized paramters</h2><span id='topic+ICM.EM_structure'></span>

<h3>Description</h3>

<p>Efficient data integration as well as  spatial clustering for multiple spatial transcriptomics data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ICM.EM_structure(XList,  K, AdjList, q=15,parameterList=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ICM.EM_structure_+3A_xlist">XList</code></td>
<td>
<p>an M-length list consisting of multiple matrices with class <code>dgCMatrix</code> or <code>matrix</code> that specify the log-normalization gene expression matrix for each data sample used for PRECAST model.</p>
</td></tr>
<tr><td><code id="ICM.EM_structure_+3A_k">K</code></td>
<td>
<p>a positive  integer allowing scalar or vector, specify the number of clusters in model fitting.</p>
</td></tr>
<tr><td><code id="ICM.EM_structure_+3A_adjlist">AdjList</code></td>
<td>
<p>an M-length list of sparse matrices with class <code>dgCMatrix</code>, specify the adjacency matrix used for Potts model and Intrisic CAR model in PRECAST model. We provide this interface for those users  who would like to define the adjacency matrix by their own.</p>
</td></tr>
<tr><td><code id="ICM.EM_structure_+3A_q">q</code></td>
<td>
<p>a positive integer, specify the number of latent features to be extracted, default as 15.</p>
</td></tr>
<tr><td><code id="ICM.EM_structure_+3A_parameterlist">parameterList</code></td>
<td>
<p>Other arguments in PRECAST model, it can be set by <a href="#topic+model_set">model_set</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>ICM.EM_structure returns a <a href="base.html#topic+list">list</a> with class &quot;SeqK_PRECAST_Object&quot; with the number of components equal to the length of <code>K</code>, where each component includes the model fitting results for one number of cluster and is a list consisting of following components:
</p>
<table role = "presentation">
<tr><td><code>cluster</code></td>
<td>
<p>an M-length list that includes the inferred class labels for each data sample.</p>
</td></tr>
<tr><td><code>hZ</code></td>
<td>
<p>an M-length list that includes the batch corrected low-dimensional embeddings for each data sample.</p>
</td></tr>
<tr><td><code>hV</code></td>
<td>
<p>an M-length list that includes the estimate the ICAR component for each sample.</p>
</td></tr>
<tr><td><code>Rf</code></td>
<td>
<p>an M-length list that includes the posterior probability of domain clusters for each sample.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>an M-length vector that includes the estimated smoothing parameters for each sample.</p>
</td></tr>
<tr><td><code>Mu</code></td>
<td>
<p>mean vectors of mixtures components.</p>
</td></tr>
<tr><td><code>Sigma</code></td>
<td>
<p>covariance matrix of mixtures components.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>estimated loading matrix</p>
</td></tr>
<tr><td><code>Lam</code></td>
<td>
<p>estimated variance of errors in probabilistic PCA model</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>
<p>pseudo observed log-likelihood.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://www.nature.com/articles/s41467-023-35947-w">Wei Liu, Liao, X., Luo, Z. et al, Jin Liu* (2023). Probabilistic embedding, clustering, and alignment for integrating spatial transcriptomics data with PRECAST. Nature Communications, 14, 296</a>
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
  library(Matrix)
  q &lt;- 10; K &lt;- 4
  data(PRECASTObj)
  posList &lt;- lapply(PRECASTObj@seulist, function(x) cbind(x$row, x$col))
  AdjList &lt;- lapply(posList, getAdj_reg, platform='ST')
  XList &lt;- lapply(PRECASTObj@seulist, function(x) t(x[['RNA']]@data))
  XList &lt;- lapply(XList, scale, scale=FALSE)
  parList &lt;- model_set(maxIter=4)
  resList &lt;- ICM.EM_structure(XList,  AdjList = AdjList, 
                   q=q, K=K, parameterList=parList)

</code></pre>

<hr>
<h2 id='IntegrateSpaData'>Integrate multiple SRT data</h2><span id='topic+IntegrateSpaData'></span>

<h3>Description</h3>

<p>Integrate multiple SRT data based on the PRECASTObj by PRECAST model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  IntegrateSpaData(PRECASTObj, species="Human", 
                 custom_housekeep=NULL, covariates_use=NULL,
                 seuList=NULL, subsample_rate=1, sample_seed=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IntegrateSpaData_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object after finishing the PRECAST model fitting and model selection.</p>
</td></tr>
<tr><td><code id="IntegrateSpaData_+3A_species">species</code></td>
<td>
<p>an optional string, one of 'Human', 'Mouse' and 'Unknown', specify the species of the SRT data to help choose the housekeeping genes. 'Unknown' means only using the PRECAST results reconstruct the alligned gene expression.</p>
</td></tr>
<tr><td><code id="IntegrateSpaData_+3A_custom_housekeep">custom_housekeep</code></td>
<td>
<p>user-specified housekeeping genes.</p>
</td></tr>
<tr><td><code id="IntegrateSpaData_+3A_covariates_use">covariates_use</code></td>
<td>
<p>a string vector, the colnames  in 'PRECASTObj@seulist[[1]]@meta.data', representing other biological covariates to considered when removing batch effects.  This is achieved by adding additional covariates for biological conditions in the regression, such as case or control. Default as 'NULL', denoting no other covariates to be considered.</p>
</td></tr>
<tr><td><code id="IntegrateSpaData_+3A_seulist">seuList</code></td>
<td>
<p>an optional Seurat list object, 'seuList' plays a crucial role in the integration process. If 'seuList' is set to 'NULL' and 'PRECASTObj@seuList' is not NULL, then 'seuList' will adopt the values of 'PRECASTObj@seuList'. Subsequently, the genes within 'seuList' will be utilized for integration. Conversely, if 'seuList' is not NULL, the integration will directly employ the genes specified within 'seuList'. In the event that both 'seuList' and 'PRECASTObj@seuList' are set to NULL, integration will proceed using the genes outlined in 'PRECASTObj@seulist', i.e., the variable genes. To preserve the 'seuList' not NULL in 'PRECASTObj@seuList', user can set 'rawData.preserve=TRUE' when running 'CreatePRECASTObject'. This parameter empowers users to integrate the entire set of genes in 'seuList' when implementing the integration, as opposed to exclusively considering the variable genes within 'PRECASTObj@seuList'.</p>
</td></tr>
<tr><td><code id="IntegrateSpaData_+3A_subsample_rate">subsample_rate</code></td>
<td>
<p>an optional real number ranging from zero to one, this parameter specifies the subsampling rate during integration to enhance computational efficiency, default as 1 (without subsampling).</p>
</td></tr>
<tr><td><code id="IntegrateSpaData_+3A_sample_seed">sample_seed</code></td>
<td>
<p>an optional integer, with a default value of 1, serves to designate the random seed when 'subsample_rate' is set to a value less than one, ensuring reproducibility in the sampling process.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a Seurat object by integrating all SRT data batches into a SRT data, where the column &quot;batch&quot; in the meta.data represents the batch ID, and the column &quot;cluster&quot; represents the clusters obtained by PRECAST.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://www.nature.com/articles/s41467-023-35947-w">Wei Liu, Liao, X., Luo, Z. et al, Jin Liu* (2023). Probabilistic embedding, clustering, and alignment for integrating spatial transcriptomics data with PRECAST. Nature Communications, 14, 296</a>
</p>
<p>Gagnon-Bartsch, J. A., Jacob, L., &amp; Speed, T. P. (2013). Removing unwanted variation from high dimensional data with negative controls. Berkeley: Tech Reports from Dep Stat Univ California, 1-112.
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(PRECASTObj)
  PRECASTObj &lt;- SelectModel(PRECASTObj)
  seuInt &lt;- IntegrateSpaData(PRECASTObj, species='unknown')
</code></pre>

<hr>
<h2 id='model_set'>PRECAST model setting</h2><span id='topic+model_set'></span>

<h3>Description</h3>

<p>Set the PRECAST model structure and paramters in the algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  model_set(Sigma_equal=FALSE, Sigma_diag=TRUE,mix_prop_heter=TRUE,
                      error_heter=TRUE, Sp2=TRUE, wpca_int=FALSE,int.model='EEE',
                      coreNum = 1, coreNum_int=coreNum,
                      beta_grid=seq(0.2,4, by=0.2),
                      maxIter_ICM=6,maxIter=20, epsLogLik=1e-5, verbose=TRUE, seed=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="model_set_+3A_sigma_equal">Sigma_equal</code></td>
<td>
<p>an optional logical value, specify whether Sigmaks are equal, default as FALSE.</p>
</td></tr>
<tr><td><code id="model_set_+3A_sigma_diag">Sigma_diag</code></td>
<td>
<p>an optional logical value, specify whether Sigmaks are diagonal matrices, default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="model_set_+3A_mix_prop_heter">mix_prop_heter</code></td>
<td>
<p>an optional logical value, specify whether betar are distict, default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="model_set_+3A_error_heter">error_heter</code></td>
<td>
<p>an optional logical value, whether use the heterogenous error i.e. lambdarj != lambdark for each sample r, default as <code>TRUE</code>. If <code>error_heter=FALSE</code>, then the homogenuous error is used for probabilistic PCA model.</p>
</td></tr>
<tr><td><code id="model_set_+3A_sp2">Sp2</code></td>
<td>
<p>an optional logical value, whether add the ICAR model component in the model, default as TRUE. We provide this interface for those users  who don't want to include the  ICAR model.</p>
</td></tr>
<tr><td><code id="model_set_+3A_wpca_int">wpca_int</code></td>
<td>
<p>an optional logical value, means whether use the weighted PCA to obtain the initial values of loadings and other paramters, default as <code>FALSE</code> which means the ordinary PCA  is used.</p>
</td></tr>
<tr><td><code id="model_set_+3A_int.model">int.model</code></td>
<td>
<p>an optional string, specify which Gaussian mixture model is used in evaluting the initial values for PRECAST, default as &quot;EEE&quot;; and see <code>Mclust</code> for more models' names.</p>
</td></tr>
<tr><td><code id="model_set_+3A_corenum">coreNum</code></td>
<td>
<p>an optional positive integer, means the number of thread used in parallel computating.</p>
</td></tr>
<tr><td><code id="model_set_+3A_corenum_int">coreNum_int</code></td>
<td>
<p>an optional positive integer, means the number of cores used in parallel computation for initial values when <code>K</code> is a vector, default as same as <code>coreNum</code>.</p>
</td></tr>
<tr><td><code id="model_set_+3A_beta_grid">beta_grid</code></td>
<td>
<p>an optional vector of positive value, the candidate set of the smoothing parameter to be searched by the grid-search optimization approach.</p>
</td></tr>
<tr><td><code id="model_set_+3A_maxiter_icm">maxIter_ICM</code></td>
<td>
<p>an optional positive value, represents the maximum iterations of ICM.</p>
</td></tr>
<tr><td><code id="model_set_+3A_maxiter">maxIter</code></td>
<td>
<p>an optional positive value, represents the maximum iterations of EM.</p>
</td></tr>
<tr><td><code id="model_set_+3A_epsloglik">epsLogLik</code></td>
<td>
<p>an optional positive vlaue, tolerance vlaue of relative variation rate of the observed pseudo log-loglikelihood value, defualt as '1e-5'.</p>
</td></tr>
<tr><td><code id="model_set_+3A_verbose">verbose</code></td>
<td>
<p>an optional logical value, whether output the information of the ICM-EM algorithm.</p>
</td></tr>
<tr><td><code id="model_set_+3A_seed">seed</code></td>
<td>
<p>an optional integer, the random seed in fitting PRECAST model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing 
</p>


<h3>Value</h3>

<p>Return a <a href="base.html#topic+list">list</a> including all paramters' setting.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  model_set()
</code></pre>

<hr>
<h2 id='Mouse_HK_genes'>Mouse housekeeping genes database</h2><span id='topic+Mouse_HK_genes'></span>

<h3>Description</h3>

<p>Mouse housekeeping genes database.
</p>


<h3>Details</h3>

<p>This data is a <a href="base.html#topic+data.frame">data.frame</a> and include the mouse housekeeping genes information in the columns named &quot;Gene&quot;  and  &quot;Ensembl&quot;.
</p>

<hr>
<h2 id='plot_RGB'>Spatial RGB heatmap</h2><span id='topic+plot_RGB'></span>

<h3>Description</h3>

<p>Plot spatial RGB heatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  plot_RGB(position, embed_3d, pointsize=2,textsize=15)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RGB_+3A_position">position</code></td>
<td>
<p>a coordinates matrix with two columns: x-coordinate and y-coordinate.</p>
</td></tr>
<tr><td><code id="plot_RGB_+3A_embed_3d">embed_3d</code></td>
<td>
<p>a embedding matrix with three columns: x, y and z embeddings.</p>
</td></tr>
<tr><td><code id="plot_RGB_+3A_pointsize">pointsize</code></td>
<td>
<p>the size of point in the scatter plot.</p>
</td></tr>
<tr><td><code id="plot_RGB_+3A_textsize">textsize</code></td>
<td>
<p>the text size in the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>

<hr>
<h2 id='plot_scatter'>Scatter plot for two-dimensional embeddings</h2><span id='topic+plot_scatter'></span>

<h3>Description</h3>

<p>Scatter plot for two-dimensional embeddings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  plot_scatter(embed_use, meta_data, label_name, 
    xy_names=c('tSNE1', 'tSNE2'), no_guides = FALSE, 
    cols = NULL, 
    point_size = 0.5, point_alpha=1, 
    base_size = 12, do_points = TRUE, do_density = FALSE, border_col='gray',
    legend_pos='right', legend_dir='vertical', nrow.legend=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_scatter_+3A_embed_use">embed_use</code></td>
<td>
<p>an object named &quot;Seurat&quot;, &quot;maxtrix&quot; or &quot;dgCMatrix&quot;. The object of class &quot;Seurat&quot; must include slot &quot;scale.data&quot;.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_meta_data">meta_data</code></td>
<td>
<p>an optional positive integer, specify the number of features to be extracted.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_label_name">label_name</code></td>
<td>
<p>the size of point in the scatter plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_xy_names">xy_names</code></td>
<td>
<p>the text size in the plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_no_guides">no_guides</code></td>
<td>
<p>whether display the legend.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_cols">cols</code></td>
<td>
<p>colors used in the plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_point_size">point_size</code></td>
<td>
<p>the point size of scatter plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_point_alpha">point_alpha</code></td>
<td>
<p>the transparency of the plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_base_size">base_size</code></td>
<td>
<p>the base text size.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_do_points">do_points</code></td>
<td>
<p>Plot point plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_do_density">do_density</code></td>
<td>
<p>Plot density plot</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_border_col">border_col</code></td>
<td>
<p>the border color in the plot.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_legend_pos">legend_pos</code></td>
<td>
<p>the position of legend.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_legend_dir">legend_dir</code></td>
<td>
<p>the direction of legend.</p>
</td></tr>
<tr><td><code id="plot_scatter_+3A_nrow.legend">nrow.legend</code></td>
<td>
<p>the number of rows of legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  embed_use &lt;- cbind(1:100, sin((1:100)*pi/2))
  meta_data &lt;- data.frame(cluster=factor(rep(1:2, each=50)))
  plot_scatter(embed_use, meta_data, label_name='cluster')
</code></pre>

<hr>
<h2 id='PRECAST'>Fit a PRECAST model</h2><span id='topic+PRECAST'></span>

<h3>Description</h3>

<p>Fit a PRECAST model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  PRECAST(PRECASTObj, K=NULL, q= 15)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PRECAST_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>an object named &quot;PRECASTObj&quot;. The object PRECASTObj is created by <a href="#topic+CreatePRECASTObject">CreatePRECASTObject</a>.</p>
</td></tr>
<tr><td><code id="PRECAST_+3A_k">K</code></td>
<td>
<p>An optional integer or integer vector, specify the candidates of number of clusters. if <code>K=NULL</code>, it will be set to 4~12.</p>
</td></tr>
<tr><td><code id="PRECAST_+3A_q">q</code></td>
<td>
<p>An optional integer, specify the number of low-dimensional embeddings to extract in  PRECAST.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model fitting results are saved in the slot of resList.
</p>


<h3>Value</h3>

<p>Return a revised PRECASTObj object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://www.nature.com/articles/s41467-023-35947-w">Wei Liu, Liao, X., Luo, Z. et al, Jin Liu* (2023). Probabilistic embedding, clustering, and alignment for integrating spatial transcriptomics data with PRECAST. Nature Communications, 14, 296</a>
</p>


<h3>See Also</h3>

<p>None
</p>

<hr>
<h2 id='PRECASTObj'>A simple PRECASTObj for example</h2><span id='topic+PRECASTObj'></span>

<h3>Description</h3>

<p>A simple PRECASTObj for example.
</p>


<h3>Details</h3>

<p>This PRECASTObj include the basic slots in PRECAST object; see <a href="#topic+PRECASTObj-class">PRECASTObj-class</a> for more details.
</p>

<hr>
<h2 id='PRECASTObj-class'>Each PRECASTObj object has a number of slots which store information. </h2><span id='topic+PRECASTObj-class'></span>

<h3>Description</h3>

<p>Each PRECASTObj object has a number of slots which store information. Key slots to access
are listed below.
</p>


<h3>Slots</h3>


<dl>
<dt><code>seuList</code></dt><dd><p>A list with Seurat object as component, representing the raw expression count matrix, spatial coordinates and meta data for each data batch, where the spatial coordinates information is saved in the metadata of Seurat, named &quot;row&quot; and &quot;col&quot; for eahc data batch.</p>
</dd>
<dt><code>seulist</code></dt><dd><p>A Seurat list after the preprocessing step in preparation for PRECAST model.</p>
</dd>
<dt><code>AdjList</code></dt><dd><p>The adjacency matrix list for a PRECASTObj object.</p>
</dd>
<dt><code>parameterList</code></dt><dd><p>The model parameter settings  for a PRECASTObj object</p>
</dd>
<dt><code>resList</code></dt><dd><p>The results after fitting PRECAST models.</p>
</dd>
<dt><code>project</code></dt><dd><p>Name of the project.</p>
</dd>
</dl>

<hr>
<h2 id='selectIntFeatures'>Select common genes for multiple data batches</h2><span id='topic+selectIntFeatures'></span>

<h3>Description</h3>

<p>selectIntFeatures prioritizes genes based on the number of times they were selected as HVGs/SVGs in all data baches, and chose the top genes as the input for the analysis.  We broke ties by examining the ranks of the tied genes in each original dataset and taking those with the highest median rank.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  selectIntFeatures(seulist, spaFeatureList, IntFeatures=2000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="selectIntFeatures_+3A_seulist">seulist</code></td>
<td>
<p>a list consisting of Seurat objects, where each object is a SRT data batch.</p>
</td></tr>
<tr><td><code id="selectIntFeatures_+3A_spafeaturelist">spaFeatureList</code></td>
<td>
<p>an list consisting of SVGs vectors, where each vector is the top HVGs/SVGs for each SRT data batch.</p>
</td></tr>
<tr><td><code id="selectIntFeatures_+3A_intfeatures">IntFeatures</code></td>
<td>
<p>the number of common HVGs/SVGs genes to be chosen.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a string vector, the selected gene list for integration in PRECAST.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://www.nature.com/articles/s41467-023-35947-w">Wei Liu, Liao, X., Luo, Z. et al, Jin Liu* (2023). Probabilistic embedding, clustering, and alignment for integrating spatial transcriptomics data with PRECAST. Nature Communications, 14, 296</a>
</p>


<h3>See Also</h3>

<p>None
</p>

<hr>
<h2 id='SelectModel'>Select best PRECAST model from candidated models</h2><span id='topic+SelectModel'></span><span id='topic+SelectModel.SeqK_PRECAST_Object'></span><span id='topic+SelectModel.PRECASTObj'></span>

<h3>Description</h3>

<p>Select best PRECAST model from candidated models with different number of clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'SeqK_PRECAST_Object'
SelectModel(obj, criteria = 'MBIC',pen_const=1, return_para_est=FALSE)
  ## S3 method for class 'PRECASTObj'
SelectModel(obj, criteria = 'MBIC',pen_const=1, return_para_est=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SelectModel_+3A_obj">obj</code></td>
<td>
<p>a SeqK_PRECAST_Object or  PRECASTObj object after PRECAST model fitting.</p>
</td></tr>
<tr><td><code id="SelectModel_+3A_criteria">criteria</code></td>
<td>
<p>a string, specify the criteria used for selecting the number of clusters, supporting &quot;MBIC&quot;, &quot;BIC&quot; and &quot;AIC&quot;.</p>
</td></tr>
<tr><td><code id="SelectModel_+3A_pen_const">pen_const</code></td>
<td>
<p>an optional positive value, the adjusted constant used in the MBIC criteria. </p>
</td></tr>
<tr><td><code id="SelectModel_+3A_return_para_est">return_para_est</code></td>
<td>
<p>an optional logical value, whether return the other paramters' estimators in PRECAST.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a revised PRECASTObj object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  data(PRECASTObj)
  PRECASTObj &lt;- SelectModel(PRECASTObj)
  
</code></pre>

<hr>
<h2 id='SpaPlot'>Spatial heatmap</h2><span id='topic+SpaPlot'></span>

<h3>Description</h3>

<p>Plot spatial heatmap for a Seurat object with spatial transcriptomics data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  SpaPlot(seuInt, batch=NULL, item=NULL, point_size=2,text_size=12, 
                    cols=NULL,font_family='', border_col="gray10",
                    fill_col='white', ncol=2, combine = TRUE,
                    title_name="Sample", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SpaPlot_+3A_seuint">seuInt</code></td>
<td>
<p>an object named &quot;Seurat&quot;.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_batch">batch</code></td>
<td>
<p>an optional positive integer or integer vector, specify the batches to be extracted. Users can check the batches' names by <code>unique(seuInt$batch)</code>.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_item">item</code></td>
<td>
<p>an optional string, which column is plotted in the meta data of seuInt. Users can check the meta data by <code>head(seuInt@meta.data)</code>. If <code>item</code> takes value from (&quot;RGB_UMAP&quot;, &quot;RGB_tSNE&quot;), this function will plot the RGB plot.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_point_size">point_size</code></td>
<td>
<p>the size of point in the scatter plot.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_text_size">text_size</code></td>
<td>
<p>the text size in the plot.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_cols">cols</code></td>
<td>
<p>colors used in the plot</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_font_family">font_family</code></td>
<td>
<p>the font family used for the plot, default as Times New Roman.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_border_col">border_col</code></td>
<td>
<p>the border color in the plot.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_fill_col">fill_col</code></td>
<td>
<p>the  color used in backgroup.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_ncol">ncol</code></td>
<td>
<p>the number of columns in the layout of plots.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_combine">combine</code></td>
<td>
<p>an optional logical value, whether plot all on a figure. If TRUE, all figures are plotted; otherwise, return a list with each plot as component.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_title_name">title_name</code></td>
<td>
<p>an optional string, title name in the plot.</p>
</td></tr>
<tr><td><code id="SpaPlot_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="#topic+plot_scatter">plot_scatter</a></code></p>
</td></tr></table>
<p>.
</p>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object or list of ggplots objects.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  data(PRECASTObj)
  PRECASTObj &lt;- SelectModel(PRECASTObj)
  seuInt &lt;- IntegrateSpaData(PRECASTObj, species='unknown')
  SpaPlot(seuInt)
  
</code></pre>

<hr>
<h2 id='volinPlot'>Volin/boxplot plot</h2><span id='topic+volinPlot'></span>

<h3>Description</h3>

<p>Plot volin/boxplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  volinPlot(mat, ylabel='ARI', cols=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="volinPlot_+3A_mat">mat</code></td>
<td>
<p>a matrix with columns.</p>
</td></tr>
<tr><td><code id="volinPlot_+3A_ylabel">ylabel</code></td>
<td>
<p>an optional string, the name of ylabel.</p>
</td></tr>
<tr><td><code id="volinPlot_+3A_cols">cols</code></td>
<td>
<p>colors used in the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   mat &lt;- matrix(runif(100*3, 0.6, 1), 100, 3)
   colnames(mat) &lt;- paste0("Method", 1:3)
   volinPlot(mat)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
