<!DOCTYPE html><html><head><title>Help for package PredCRG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PredCRG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#model1'>
<p>Trained model with the Q1 dataset.</p></a></li>
<li><a href='#model2'>
<p>Trained model with the Q2 dataset.</p></a></li>
<li><a href='#model3'>
<p>Trained model with the Q3 dataset.</p></a></li>
<li><a href='#model4'>
<p>Trained model with the Q4 dataset.</p></a></li>
<li><a href='#PredCRG'>
<p>Prediction of circadian proteins using the proposed PredCRG model.</p></a></li>
<li><a href='#PredCRG_data'>
<p>Training dataset of the PredCRG model.</p></a></li>
<li><a href='#PredCRG_Enc'>
<p>Encoding of protein sequence data in to numeric feature vector based on PredCRG features.</p></a></li>
<li><a href='#PredCRG_training'>
<p>Training of the PredCRG model using the user supplied sequence dataset.</p></a></li>
<li><a href='#test'>
<p>Test dataset.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Computational Prediction of Proteins Encoded by Circadian Genes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-12-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Prabina Kumar Meher &lt;meherprabin@yahoo.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Prabina Kumar Meher &lt;meherprabin@yahoo.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.3.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, protr,Peptides, kernlab, e1071</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Description:</td>
<td>A computational model for predicting proteins encoded by circadian genes. The support vector machine has been employed with Laplace kernel for prediction of circadian proteins, where compositional, transitional and physico-chemical features were utilized as numeric features. User can predict for the test dataset using the proposed computational model. Besides, the user can also build their own training model using their training dataset, followed by prediction for the test set. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-12-11 15:29:46 UTC; USER</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-12-11 16:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='model1'>
Trained model with the Q1 dataset.
</h2><span id='topic+model1'></span>

<h3>Description</h3>

<p>The <code>model1</code> is the trained model with the Q1 dataset using the developed approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("model1")</code></pre>


<h3>Details</h3>

<p>Here, 1558 sequences of pos_Q1 and neg_Q1 datasets were used for training. For prediction, support vector machine with Laplace kernel has been trained in which compositionsl, transitional and physico-chemical features are utilized.  
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(kernlab)
data(test)
nam &lt;- names(test)

#encoding of test set using compositional, transitional and physico-chemical features
enc &lt;- PredCRG_Enc(test)

#predicting test set using model1 as CRG or non-CRG
pred &lt;- predict(model1, newdata=enc[1:10,], type="response") 

#predicting probabilities of the test sequences using model1
pred1 &lt;- predict(model1, newdata=enc[1:10,], type="probabilities") 

#combining predicted labels and probabilities
result &lt;- data.frame(seq_name=nam[1:10], predicted_label=as.character(pred)
,predicted_probability=pred1[,"CRG"])

print(result)
</code></pre>

<hr>
<h2 id='model2'>
Trained model with the Q2 dataset.
</h2><span id='topic+model2'></span>

<h3>Description</h3>

<p>The <code>model2</code> is the trained model with the Q2 dataset using the developed approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("model2")</code></pre>


<h3>Details</h3>

<p>Here, 1596 sequences of pos_Q2 and neg_Q2 datasets were used for training. For prediction, support vector machine with Laplace kernel has been trained in which compositionsl, transitional and physico-chemical features are utilized.  
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(kernlab)
data(test)
nam &lt;- names(test)

#encoding of test set using compositional, transitional and physico-chemical features
enc &lt;- PredCRG_Enc(test)

#predicting test set using model2 as CRG or non-CRG
pred &lt;- predict(model2, newdata=enc[1:10,], type="response") 

#predicting probabilities of the test sequences using model2
pred1 &lt;- predict(model2, newdata=enc[1:10,], type="probabilities") 

#combining predicted labels and probabilities
result &lt;- data.frame(seq_name=nam[1:10], predicted_label=as.character(pred)
,predicted_probability=pred1[,"CRG"])

print(result)
</code></pre>

<hr>
<h2 id='model3'>
Trained model with the Q3 dataset.
</h2><span id='topic+model3'></span>

<h3>Description</h3>

<p>The <code>model3</code> is the trained model with the Q3 dataset using the developed approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("model3")</code></pre>


<h3>Details</h3>

<p>Here, 1593 sequences of pos_Q3 and neg_Q3 datasets were used for training. For prediction, support vector machine with Laplace kernel has been trained in which compositionsl, transitional and physico-chemical features are utilized.  
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(kernlab)
data(test)
nam &lt;- names(test)

#encoding of test set using compositional, transitional and physico-chemical features
enc &lt;- PredCRG_Enc(test)

#predicting test set using model3 as CRG or non-CRG
pred &lt;- predict(model3, newdata=enc[1:10,], type="response") 

#predicting probabilities of the test sequences using model3
pred1 &lt;- predict(model3, newdata=enc[1:10,], type="probabilities") 

#combining predicted labels and probabilities
result &lt;- data.frame(seq_name=nam[1:10], predicted_label=as.character(pred)
,predicted_probability=pred1[,"CRG"])

print(result)
</code></pre>

<hr>
<h2 id='model4'>
Trained model with the Q4 dataset.
</h2><span id='topic+model4'></span>

<h3>Description</h3>

<p>The <code>model4</code> is the trained model with the Q4 dataset using the developed approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("model4")</code></pre>


<h3>Details</h3>

<p>Here, 1365 sequences of pos_Q4 and neg_Q4 datasets were used for training. For prediction, support vector machine with Laplace kernel has been trained in which compositionsl, transitional and physico-chemical features are utilized.  
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(kernlab)
data(test)
nam &lt;- names(test)

#encoding of test set using compositional, transitional and physico-chemical features
enc &lt;- PredCRG_Enc(test)

#predicting test set using model4 as CRG or non-CRG
pred &lt;- predict(model4, newdata=enc[1:10,], type="response") 

#predicting probabilities of the test sequences using model4
pred1 &lt;- predict(model4, newdata=enc[1:10,], type="probabilities") 

#combining predicted labels and probabilities
result &lt;- data.frame(seq_name=nam[1:10], predicted_label=as.character(pred)
,predicted_probability=pred1[,"CRG"])

print(result)
</code></pre>

<hr>
<h2 id='PredCRG'>
Prediction of circadian proteins using the proposed PredCRG model.
</h2><span id='topic+PredCRG'></span>

<h3>Description</h3>

<p>The user can predict the protein sequences as CRG (circadian protein) or non-CRG (non-circadian protein) with certain probability by supplying the test sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PredCRG(seq_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PredCRG_+3A_seq_data">seq_data</code></td>
<td>

<p>Sequence dataset in FASTA format consisting of protein sequences with standard amino acid residues only. It must be an object of class <code>AAStringSet</code> which can be obtained by reading sequences with <code>readAAStringSet</code> available in <b><code>Biostrings</code></b> package.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply only the <code>seq_data</code> for which the prediction is to be made. 
</p>


<h3>Value</h3>

<p>A dataframe with three columns consisting of sequence name, predicted labels of sequences (CRG or non-CRG) and probabilities of prediction.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, ICAR-Indian Agricultural Statsitics Research Institute, New Delhi-110012, INDIA
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>,<a href="#topic+model1">model1</a>, <a href="#topic+model2">model2</a>,<a href="#topic+model3">model3</a>,<a href="#topic+model4">model4</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(test)
tst &lt;- test[1:10]
PredCRG(seq_data=tst)
</code></pre>

<hr>
<h2 id='PredCRG_data'>
Training dataset of the PredCRG model.
</h2><span id='topic+PredCRG_data'></span>

<h3>Description</h3>

<p>The dataset that has been used to train the PredCRG model contains four sub-datasets (Q1, Q2, Q3 and Q4) which are prepared based on the homogeneity of sequence length. The positive sets of the sub-datasets are denoted as pos_Q1, pos_Q2, pos_Q3 and pos_Q4 respectively, whereas the negative sets as neg_Q1, neg_Q2, neg_Q3 and neq_Q4 respectively. Further, same number of sequences are there in both positive and negative sets in each sub-dataset. More clearly, 1588, 1596, 1593 and 1365 sequences are present for both positive and negative sets for Q1, Q2, Q3 and Q4 sub-datasets respectively. Further, the range of the length of the sequences for pos_Q1, pos_Q2, pos_Q3 and pos_Q4 are 39-221, 221-363, 363-538, 538-1000 amino acids respectively, and the range of the length of the sequences for neg_Q1, neg_Q2, neg_Q3 and neg_Q4 are 43-407, 407-485, 485-607 and 607-1000 amino acids respectively. In this dataset, only the Q1 sub-dataset is available due to constraint of space in CRAN. However, one can get all the four sub-datasets from GitHub repository (<a href="https://github.com/meher861982/PredCRG_dataset">https://github.com/meher861982/PredCRG_dataset</a> ).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("PredCRG_data")</code></pre>


<h3>Format</h3>

<p>The datasets are in <code>AAStringSet</code> format, which can be obtained by reading the FASTA file using <code>readAAStringSet</code> function availbale in <b><code>Biostrings</code></b> package.
</p>


<h3>Details</h3>

<p>The protein sequences encoded by the circadian genes contitutes the positive datasets, whereas a randomly selected dataset from the <b>Uniprot</b> for the clad <em>Viridi plantae</em> constitutes the negative dataset.
</p>


<h3>Source</h3>

<p>The circadian gene sequecnces are collected from the circadian gene database accessible at <a href="http://cgdb.biocuckoo.org/">http://cgdb.biocuckoo.org/</a> .
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>,<a href="#topic+model1">model1</a>, <a href="#topic+model2">model2</a>,<a href="#topic+model3">model3</a>,<a href="#topic+model4">model4</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PredCRG_data)

pos_Q1 &lt;- PredCRG_data$pos_Q1 #positive set of Q1 dataset
neg_Q1 &lt;- PredCRG_data$neg_Q1 #negative set of Q1 dataset

</code></pre>

<hr>
<h2 id='PredCRG_Enc'>
Encoding of protein sequence data in to numeric feature vector based on PredCRG features.
</h2><span id='topic+PredCRG_Enc'></span>

<h3>Description</h3>

<p>Before using the protein sequences for prediction using the proposed model, the sequences must be transformed into numeric feature vectors. The function <code>PredCRG_Enc</code> will transform each protein sequnces to a numeric vector of 62 observations, based on the compositional, physico-chemical and transitional features used in the <code>PredCRG</code> model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PredCRG_Enc(prot_seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PredCRG_Enc_+3A_prot_seq">prot_seq</code></td>
<td>

<p>Sequence dataset to be supplied as input, must be an object of class <code>AAStringSet</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The dataset must contains the protein sequences having standard amino acid residues only. The clas <code>AAStringSet</code> can be obtained by reading the FASTA file using <code>readAAStringSet</code> available in bioconductor package <b><code>Biostrings</code></b>.
</p>


<h3>Value</h3>

<p>A matrix of dimension n*62, for n number of sequences.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, ICAR-Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_training">PredCRG_training</a>, <a href="#topic+model1">model1</a>, <a href="#topic+model2">model2</a>,<a href="#topic+model3">model3</a>,<a href="#topic+model4">model4</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(test)
enc &lt;- PredCRG_Enc(test)#encoding of test sequence data
enc[1:5,1:5]
  </code></pre>

<hr>
<h2 id='PredCRG_training'>
Training of the PredCRG model using the user supplied sequence dataset.
</h2><span id='topic+PredCRG_training'></span>

<h3>Description</h3>

<p>User can build their own PredCRG model by using their own training dataset. User has to supply the protein sequence dataset of both positive and negative classes having standard amino acid residues only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PredCRG_training(pos_seq, neg_seq, kern)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PredCRG_training_+3A_pos_seq">pos_seq</code></td>
<td>

<p>circadian protein sequence dataset (also called positive dataset), must be an object of class <code>AAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="PredCRG_training_+3A_neg_seq">neg_seq</code></td>
<td>

<p>non-circadian protein sequence dataset (also called negative dataset), must be an object of class <code>AAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="PredCRG_training_+3A_kern">kern</code></td>
<td>

<p>Type of kernel to be used. It may be <code>laplace</code>, <code>linear</code>, <code>polynomial</code> or <code>RBF</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sequences must of <code>AAStringSet</code> type can be obtained by reading the FASTA file of the sequences using function <code>readAAStringSet</code> available in <b><code>Biostrings</code></b> package.
</p>


<h3>Value</h3>

<p>Support Vector Machine object of class <code>ksvm</code>
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, ICAR-Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+model1">model1</a>, <a href="#topic+model2">model2</a>,<a href="#topic+model3">model3</a>,<a href="#topic+model4">model4</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(kernlab)
pos_Q1 &lt;- PredCRG_data$pos_Q1
neg_Q1 &lt;- PredCRG_data$neg_Q1

#training of the model using laplace kernel.
user_model &lt;- PredCRG_training(pos_seq=pos_Q1[1:100], neg_seq=neg_Q1[1:100], kern="laplace")

data(test)
tst_enc &lt;- PredCRG_Enc(test[1:10])#encoding of the test set
predict(user_model, tst_enc, type="response") #predicting the label of the test instances
predict(user_model, tst_enc, type="probabilities")#predicting the probability of the test instances


library(e1071)
#training of the model using RBF kernel.
user_model &lt;- PredCRG_training(pos_seq=pos_Q1[1:100], neg_seq=neg_Q1[1:100], kern="RBF")
predict(user_model, tst_enc, probability=TRUE) #Predicting probability
predict(user_model, tst_enc) #Predicting labels
</code></pre>

<hr>
<h2 id='test'>
Test dataset. 
</h2><span id='topic+test'></span>

<h3>Description</h3>

<p>A test dataset containing 54 circadian protein sequences collected from literature. This dataset has been used as an independent test dataset for assessing the predition accuracy of <code>PredCRG</code> model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("test")</code></pre>


<h3>See Also</h3>

<p><a href="#topic+PredCRG">PredCRG</a>, <a href="#topic+PredCRG_Enc">PredCRG_Enc</a>, <a href="#topic+PredCRG_data">PredCRG_data</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(test)
PredCRG(test[1:10])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
