<!DOCTYPE html><html><head><title>Help for package predhy.GUI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {predhy.GUI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#hybrid_phe'><p>Phenotypic data of hybrids</p></a></li>
<li><a href='#input_geno'><p>Genotype in Hapmap Format</p></a></li>
<li><a href='#input_geno1'><p>Genotype in Numeric Format</p></a></li>
<li><a href='#predhy.GUI'><p>Graphical User Interface for R package predhy</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genomic Prediction of Hybrid Performance with Graphical User
Interface</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs genomic prediction of hybrid performance using eight GS methods including GBLUP, BayesB, RKHS, PLS, LASSO, Elastic net, Random forest and XGBoost.
             GBLUP: genomic best liner unbiased prediction, RKHS: reproducing kernel Hilbert space, PLS: partial least squares regression, LASSO: least absolute shrinkage and selection operator, XGBoost: extreme gradient boosting.
             It also provides fast cross-validation and mating design scheme for training population (Xu S et al (2016) &lt;<a href="https://doi.org/10.1111%2Ftpj.13242">doi:10.1111/tpj.13242</a>&gt;; Xu S (2017) &lt;<a href="https://doi.org/10.1534%2Fg3.116.038059">doi:10.1534/g3.116.038059</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>shiny, data.table, DT, predhy(&ge; 1.2.1), BGLR, pls, glmnet,
randomForest, xgboost, foreach, doParallel, parallel, htmltools</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-20 01:34:02 UTC; BUQING</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-21 11:40:03 UTC</td>
</tr>
<tr>
<td>Author:</td>
<td>Yang Xu [aut],
  Guangning Yu [aut],
  Yuxiang Zhang [aut, cre],
  Yanru Cui [ctb],
  Shizhong Xu [ctb],
  Chenwu Xu [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuxiang Zhang &lt;yuxiangzhang_99@foxmail.com&gt;</td>
</tr>
</table>
<hr>
<h2 id='hybrid_phe'>Phenotypic data of hybrids</h2><span id='topic+hybrid_phe'></span>

<h3>Description</h3>

<p>This dataset contains phenotypic data of 410 hybrids for grain yield in maize.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hybrid_phe
</code></pre>


<h3>Format</h3>

<p>A data frame with 410 rows and 3 variables:
</p>

<dl>
<dt><code>M</code></dt><dd><p>The names of male parents.</p>
</dd>
<dt><code>F</code></dt><dd><p>The names of female parents.</p>
</dd>
<dt><code>GY</code></dt><dd><p>The grain yield of hybrids.</p>
</dd>
</dl>


<hr>
<h2 id='input_geno'>Genotype in Hapmap Format</h2><span id='topic+input_geno'></span>

<h3>Description</h3>

<p>Genotypic data of 348 maize inbred lines in Hapmap format with double bit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_geno
</code></pre>


<h3>Format</h3>

<p>A data frame with 4979 rows and 359 columns.
</p>

<hr>
<h2 id='input_geno1'>Genotype in Numeric Format</h2><span id='topic+input_geno1'></span>

<h3>Description</h3>

<p>Genotypic data of 50 rice inbred lines with 1000 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_geno1
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows and 50 variables.
</p>

<hr>
<h2 id='predhy.GUI'>Graphical User Interface for R package predhy</h2><span id='topic+predhy.GUI'></span>

<h3>Description</h3>

<p>Graphical User Interface for cross validation, genotype conversion and hybrid performance prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predhy.GUI()
</code></pre>


<h3>Value</h3>

<p>No return value, called for Graphical User Interface
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
predhy.GUI()}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
