<!DOCTYPE html><html><head><title>Help for package PredictABEL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PredictABEL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PredictABEL-package'><p>An R package for the analysis of (genetic) risk prediction studies.</p></a></li>
<li><a href='#ExampleData'><p>A hypothetical dataset that is used to demonstrate all functions.</p></a></li>
<li><a href='#ExampleModels'><p>An example code to construct a risk model using logistic regression analysis.</p></a></li>
<li><a href='#fitLogRegModel'><p>Function to fit a logistic regression model.</p></a></li>
<li><a href='#ORmultivariate'><p>Function to obtain multivariate odds ratios from a logistic regression model.</p></a></li>
<li><a href='#plotCalibration'><p>Function for calibration plot and Hosmer-Lemeshow goodness of fit test.</p></a></li>
<li><a href='#plotDiscriminationBox'><p>Function for box plots of predicted risks separately for individuals with and without the outcome of interest.</p></a></li>
<li><a href='#plotPredictivenessCurve'><p>Function for predictiveness curve.</p></a></li>
<li><a href='#plotPriorPosteriorRisk'><p>Function to plot posterior risks against prior risks.</p></a></li>
<li><a href='#plotRiskDistribution'><p>Function to plot histogram of risks separated for individuals with and without the outcome of interest.</p></a></li>
<li><a href='#plotRiskscorePredrisk'><p>Function to plot predicted risks against risk scores.</p></a></li>
<li><a href='#plotROC'><p>Function for a receiver operating characteristic curve (ROC) plot and area under the ROC curve (AUC) value.</p></a></li>
<li><a href='#predRisk'><p>Function to compute predicted risks for all individuals in the dataset.</p></a></li>
<li><a href='#reclassification'><p>Function for reclassification table and statistics.</p></a></li>
<li><a href='#riskScore'><p>Function to compute genetic risk scores.</p></a></li>
<li><a href='#simulatedDataset'><p>Function to construct a simulated dataset containing individual genotype</p>
data, genetic risks and disease status for a hypothetical population.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Assessment of Risk Prediction Models</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2-4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-02-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Suman Kundu, Yurii S. Aulchenko, A. Cecile J.W. Janssens</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Suman Kundu &lt;suman_math@yahoo.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.12.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Hmisc, ROCR, PBSmodelling, lazyeval, methods</td>
</tr>
<tr>
<td>Description:</td>
<td>We included functions to assess the performance of
 risk models. The package contains functions for the various measures that are
 used in empirical studies, including univariate and multivariate odds ratios
 (OR) of the predictors, the c-statistic (or area under the receiver operating
 characteristic (ROC) curve (AUC)), Hosmer-Lemeshow goodness of fit test,
 reclassification table, net reclassification improvement (NRI) and
 integrated discrimination improvement (IDI). Also included are functions
 to create plots, such as risk distributions, ROC curves, calibration plot,
 discrimination box plot and predictiveness curves. In addition to functions
 to assess the performance of risk models, the package includes functions to
 obtain weighted and unweighted risk scores as well as predicted risks using
 logistic regression analysis. These logistic regression functions are
 specifically written for models that include genetic variables, but they
 can also be applied to models that are based on non-genetic risk factors only.
 Finally, the package includes function to construct a simulated dataset with 
 genotypes, genetic risks, and disease status for a hypothetical population, which 
 is used for the evaluation of genetic risk models.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-03-09 20:50:19 UTC; kundus</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-03-09 21:20:02 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
</table>
<hr>
<h2 id='PredictABEL-package'>An R package for the analysis of (genetic) risk prediction studies.</h2><span id='topic+PredictABEL-package'></span>

<h3>Description</h3>

<p>An R package for the analysis of (genetic) risk prediction studies.</p>


<h3>Details</h3>

<p>Fueled by the substantial gene discoveries from genome-wide association 
studies, there is increasing interest in investigating the predictive 
ability of genetic risk models. To assess the performance of genetic risk 
models, PredictABEL includes functions for the various measures and plots 
that have been used in empirical studies, including univariate and 
multivariate odds ratios (ORs) of the predictors, the c-statistic (or AUC), 
Hosmer-Lemeshow goodness of fit test, reclassification table, net 
reclassification improvement (NRI) and integrated discrimination 
improvement (IDI). The plots included are the ROC plot, calibration plot, 
discrimination box plot, predictiveness curve, and several risk distributions. 
</p>
<p>These functions can be applied to predicted risks that are obtained using 
logistic regression analysis, to weighted or unweighted risk scores, for 
which the functions are included in this package. The functions can also be 
used to assess risks or risk scores that are constructed using other methods, e.g., Cox Proportional 
Hazards regression analysis, which are not included in the current version. 
Risks obtained from other methods can be imported into R for assessment 
of the predictive performance.
</p>
<p>The functions to construct the risk models using logistic regression analyses 
are specifically written for models that include genetic variables, 
eventually in addition to non-genetic factors, but they can also be applied 
to construct models that are based on non-genetic risk factors only. <br /> 
</p>
<p>Before using the functions <code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> for constructing 
a risk model or <code><a href="#topic+riskScore">riskScore</a></code> for computing risk 
scores, the following checks on the dataset are advisable to be done: 
</p>
<p>(1) Missing values: The logistic regression analyses and computation of 
the risk score are done only for subjects that have no missing data. In case 
of missing values, individuals with missing data can be removed from the 
dataset or imputation strategies can be used to fill in missing data. 
Subjects with missing data can be removed with the R function <code>na.omit</code> 
(available in <code>stats</code> package). 
Example: <code>DataFileNew &lt;- na.omit(DataFile)</code> 
will make a new dataset (<code>DataFileNew</code>) with no missing values;
</p>
<p>(2) Multicollinearity: When there is strong correlation between the 
predictor variables, regression coefficients may be estimated imprecisely 
and risks scores may be biased because the assumption of independent effects 
is violated. In genetic risk prediction studies, problems with 
multicollinearity should be expected when single nucleotide polymorphisms 
(SNPs) located in the same gene are                                                    
in strong linkage disequilibrium (LD). For SNPs in LD it is common to select 
the variant with the lowest p-value in the model;  
</p>
<p>(3) Outliers: When the data contain significant outliers, either clinical 
variables with extreme values of the outcomes or extreme values resulting 
from errors in the data entry, these may impact the construction of the risk models and 
computation of the risks scores. Data should be carefully checked and outliers 
need to be removed or replaced, if justified;
</p>
<p>(4) Recoding of data: In the computation of unweighted risk scores, it is assumed 
that the genetic variants are coded <code>0,1,2</code>  
representing the number of alleles carried. When variants 
are coded <code>0,1</code> representing a dominant or recessive effect of the alleles, 
the variables need to be recoded before unweighted risk scores can be computed. <br />
</p>
<p>To import data into R several alternative strategies can be used. Use the 
<code>Hmisc</code> package for importing SPSS and SAS data into R. 
Use &quot;<code>ExampleData &lt;- read.table("DataName.txt", header=T, sep="\t")</code>&quot; for text 
files where variable names are included as column headers and data are 
separated by tabs.  
Use &quot;<code>ExampleData &lt;- read.table("Name.csv", sep=",", header=T)</code>&quot; 
for comma-separated files with variable names as column headers. 
Use <code>"setwd(dir)"</code> to set the working directory to &quot;dir&quot;. The datafile 
needs to be present in the working directory. <br /> 
</p>
<p>To export datafiles from R tables to a tab-delimited textfile with the first row as
the name of the variables, 
use &quot;<code>write.table(R_Table, file="Name.txt", row.names=FALSE, sep="\t")</code>&quot;  and 
when a comma-separated textfile is requested and variable names are provided in the first row,
use &quot;<code>write.table(R_Table, file="Name.csv", row.names=FALSE, sep=",")</code>&quot;.  
When the directory is not specified, the file will be
saved in the working directory. For exporting R data into SPSS, SAS and 
Stata data, use functions in the the <code>foreign</code> package. <br />
</p>
<p>Several functions in this package depend on other R packages:
</p>
<p>(1) <code>Hmisc</code>, is used to compute NRI and IDI;
</p>
<p>(2) <code>ROCR</code>, is used to produce ROC plots;
</p>
<p>(3) <code>epitools</code>, is used to compute  univariate odds ratios;
</p>
<p>(4) <code>PBSmodelling</code>, is used to produce predictiveness curve.</p>


<h3>Acknowledgements</h3>

<p>The authors would like to acknowledge Lennart Karssen, 
Maksim Struchalin and Linda Broer from the Department of Epidemiology, 
Erasmus Medical Center, Rotterdam for their valuable comments and 
suggestions to make this package.
</p>


<h3>Note</h3>

<p>The current version of the package includes the basic measures 
and plots that are used in the assessment of (genetic) risk prediction models and the
function to construct a simulated dataset that contains individual genotype 
data, estimated genetic risk and disease status, used for the evaluation of 
genetic risk models (see Janssens et al, Genet Med 2006). 
Planned extensions of the package include functions to construct risk 
models using Cox Proportional Hazards analysis for prospective data and 
assess the performance of risk models for time-to-event data.</p>


<h3>Author(s)</h3>

<p>Suman Kundu  
</p>
<p>Yurii S. Aulchenko
</p>
<p>A. Cecile J.W. Janssens</p>


<h3>References</h3>

<p>S Kundu, YS Aulchenko, CM van Duijn, ACJW Janssens. PredictABEL:
an R package for the assessment of risk prediction models.
Eur J Epidemiol. 2011;26:261-4. <br />
</p>
<p>ACJW Janssens, JPA Ioannidis, CM van Duijn, J Little, MJ Khoury.
Strengthening the Reporting of Genetic Risk Prediction Studies: The GRIPS
Statement Proposal. Eur J Epidemiol. 2011;26:255-9. <br />
</p>
<p>ACJW Janssens, JPA Ioannidis, S Bedrosian, P Boffetta, SM Dolan, N Dowling,
I Fortier, AN. Freedman, JM Grimshaw, J Gulcher, M Gwinn, MA Hlatky, H Janes,
P Kraft, S Melillo, CJ O'Donnell, MJ Pencina, D Ransohoff, SD Schully,
D Seminara, DM Winn, CF Wright, CM van Duijn, J Little, MJ Khoury.
Strengthening the reporting of genetic risk prediction studies
(GRIPS)-Elaboration and explanation. Eur J Epidemiol. 2011;26:313-37. <br />   
</p>
<p>Aulchenko YS, Ripke S, Isaacs A, van Duijn CM. GenABEL: an R package for genome-wide
association analysis. Bioinformatics 2007;23(10):1294-6.</p>

<hr>
<h2 id='ExampleData'>A hypothetical dataset that is used to demonstrate all functions.</h2><span id='topic+ExampleData'></span>

<h3>Description</h3>

<p>ExampleData is a hypothetical dataset constructed to demonstrate all functions 
in the package. ExampleData is a data frame containing a binary outcome variable 
(e.g., disease present/absent) and genetic and non-genetic predictor variables 
for 10,000 persons. In this dataset, column 1 is the ID number, column 2 is the
outcome variable, columns 3-10 are non-genetic variables and columns 11-16 are  
genetic variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleData)</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ExampleData)
# show first 5 records (rows) of the dataset
head(ExampleData,5)
</code></pre>

<hr>
<h2 id='ExampleModels'>An example code to construct a risk model using logistic regression analysis.</h2><span id='topic+ExampleModels'></span>

<h3>Description</h3>

<p><code>ExampleModels</code> constructs two risk models using logistic regression analysis. 
Most of the functions in this package require a logistic regression model as an input and 
estimate predicted risks from this fitted model. 
To illustrate these functions without repeating the construction of a 
logistic regression model, this example code has been created.
The function returns two different risk models, riskModel1 which is based 
on non-genetic predictors and riskModel2 which includes genetic and non-genetic predictors.</p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of the outcome variable
cOutcome &lt;- 2
# specify column numbers of non-genetic predictors
cNonGenPred1 &lt;- c(3:10)
cNonGenPred2 &lt;- c(3:10)
# specify column numbers of non-genetic predictors that are categorical
cNonGenPredCat1 &lt;- c(6:8)
cNonGenPredCat2 &lt;- c(6:8)
# specify column numbers of genetic predictors
cGenPred1 &lt;- c(0)
cGenPred2 &lt;- c(11:16)
# specify column numbers of genetic predictors that are categorical
cGenPredsCat1 &lt;- c(0)
cGenPredsCat2 &lt;- c(0)

# fit logistic regression models
riskmodel1 &lt;- fitLogRegModel(data=ExampleData, cOutcome=cOutcome,
cNonGenPreds=cNonGenPred1, cNonGenPredsCat=cNonGenPredCat1,
cGenPreds=cGenPred1, cGenPredsCat=cGenPredsCat1)
riskmodel2 &lt;- fitLogRegModel(data=ExampleData, cOutcome=cOutcome,
cNonGenPreds=cNonGenPred2, cNonGenPredsCat=cNonGenPredCat2,
cGenPreds=cGenPred2, cGenPredsCat=cGenPredsCat2)

# combine output in a list 
ExampleModels &lt;- list(riskModel1=riskmodel1, riskModel2=riskmodel2)</code></pre>

<hr>
<h2 id='fitLogRegModel'>Function to fit a logistic regression model.</h2><span id='topic+fitLogRegModel'></span>

<h3>Description</h3>

<p>The function fits a standard GLM function for the logistic regression model.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitLogRegModel(data, cOutcome, cNonGenPreds, cNonGenPredsCat, 
cGenPreds, cGenPredsCat)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitLogRegModel_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome and 
predictor variables.</p>
</td></tr>
<tr><td><code id="fitLogRegModel_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable. <code>cOutcome=2</code>
means that the second column of the dataset is the outcome variable.
To fit the logistic regression model, the outcome variable needs to be 
(re)coded as <code>1</code> for the presence and <code>0</code> for the absence of the
outcome of interest.</p>
</td></tr>
<tr><td><code id="fitLogRegModel_+3A_cnongenpreds">cNonGenPreds</code></td>
<td>
<p>Column numbers of the non-genetic predictors that are  
included in the model. An example to denote column numbers is  
<code>c(3,6:8,10)</code>. Choose <code>c(0)</code> when no non-genetic predictors 
are considered.</p>
</td></tr>
<tr><td><code id="fitLogRegModel_+3A_cnongenpredscat">cNonGenPredsCat</code></td>
<td>
<p>Column numbers of the non-genetic predictors that  
are entered as categorical variables in the model. When non-genetic 
predictors are not specified as being categorical they are treated as     
continuous variables in the model. If no non-genetic predictors are 
categorical, denote <code>c(0)</code>.</p>
</td></tr>
<tr><td><code id="fitLogRegModel_+3A_cgenpreds">cGenPreds</code></td>
<td>
<p>Column numbers of the genetic predictors or genetic risk score.
Denote <code>c(0)</code>
when the prediction model does not consider 
genetic predictors or genetic risk score.</p>
</td></tr>
<tr><td><code id="fitLogRegModel_+3A_cgenpredscat">cGenPredsCat</code></td>
<td>
<p>Column numbers of the genetic predictors that are 
entered as categorical variables in the model. When SNPs are considered as 
categorical, the model  
will estimate effects per genotype. Otherwise, SNPs are considered as  
continuous variables for which  the model will estimate an allelic effect.  
Choose c(0) when no genetic predictors are considered as categorical 
or when genetic predictors are entered as a risk score into the model.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function fits a standard GLM function for the logistic regression model.
This function can be used to construct a logistic regression model based on genetic and non-genetic
predictors. The function also allows to enter the genetic predictors 
as a single risk score. For that purpose, the function requires that 
the dataset additionally includes the risk score.
A new dataset can be constructed using
&quot;<code>NewExampleData &lt;- cbind(ExampleData,riskScore)</code>&quot;. 
The genetic risk scores can be obtained
using the function <code><a href="#topic+riskScore">riskScore</a></code> in this package or be
imported from other methods.</p>


<h3>Value</h3>

<p>No value returned.</p>


<h3>See Also</h3>

<p><code><a href="#topic+predRisk">predRisk</a></code>, <code><a href="#topic+ORmultivariate">ORmultivariate</a></code>, <code><a href="#topic+riskScore">riskScore</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables 
data(ExampleData)
# specify column number of outcome variable
cOutcome &lt;- 2 
# specify column numbers of non-genetic predictors
cNonGenPred &lt;- c(3:10)
# specify column numbers of non-genetic predictors that are categorical      
cNonGenPredCat &lt;- c(6:8)
# specify column numbers of genetic predictors
cGenPred &lt;- c(11,13:16)
# specify column numbers of genetic predictors that are categorical
cGenPredCat &lt;- c(0) 

# fit logistic regression model
riskmodel &lt;- fitLogRegModel(data=ExampleData, cOutcome=cOutcome, 
cNonGenPreds=cNonGenPred, cNonGenPredsCat=cNonGenPredCat, 
cGenPreds=cGenPred, cGenPredsCat=cGenPredCat)

# show summary details for the fitted risk model
summary(riskmodel)</code></pre>

<hr>
<h2 id='ORmultivariate'>Function to obtain multivariate odds ratios from a logistic regression model.</h2><span id='topic+ORmultivariate'></span>

<h3>Description</h3>

<p>The function estimates multivariate (adjusted) odds ratios (ORs) with    
95% confidence intervals (CIs) for all the genetic and non-genetic variables 
in the risk model.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ORmultivariate(riskModel, filename)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ORmultivariate_+3A_riskmodel">riskModel</code></td>
<td>
<p>Name of logistic regression model that can be fitted using  
the function <code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code>.</p>
</td></tr>
<tr><td><code id="ORmultivariate_+3A_filename">filename</code></td>
<td>
<p>Name of the output file in which the multivariate    
ORs will be saved. If no directory is specified, the file is 
saved in the working directory as a txt file.   
When <code>filename</code> is not specified, the output is not saved.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function requires that first a logistic regression  
model is fitted either by using <code>GLM</code> function or the function 
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code>. In addition to the multivariate ORs, 
the function returns summary statistics of model performance, namely the Brier 
score and the Nagelkerke's <code class="reqn">R^2</code> value.  
The Brier score quantifies the accuracy of risk predictions by comparing 
predicted risks with observed outcomes at individual level (where outcome 
values are either 0 or 1). The Nagelkerke's <code class="reqn">R^2</code> value indicates the percentage of variation   
of the outcome explained by the predictors in the model.</p>


<h3>Value</h3>

<p>The function returns: 
</p>
<table>
<tr><td><code>Predictors Summary</code></td>
<td>
<p>OR with 95% CI and corresponding  p-values for 
each predictor in the model</p>
</td></tr>
<tr><td><code>Brier Score</code></td>
<td>
<p>Brier score</p>
</td></tr>
<tr><td><code>Nagelkerke Index</code></td>
<td>
<p>Nagelkerke's <code class="reqn">R^2</code> value</p>
</td></tr></table>


<h3>References</h3>

<p>Brier GW. Verification of forecasts expressed in terms of probability. 
Monthly weather review 1950;78:1-3.   
</p>
<p>Nagelkerke NJ. A note on a general definition of the coefficient
of determination. Biometrika 1991;78:691-692.</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of outcome variable
cOutcome &lt;- 2
# specify column numbers of non-genetic predictors
cNonGenPred &lt;- c(3:10)
# specify column numbers of non-genetic predictors that are categorical
cNonGenPredCat &lt;- c(6:8)
# specify column numbers of genetic predictors
cGenPred &lt;- c(11,13:16)
# specify column numbers of genetic predictors that are categorical
cGenPredCat &lt;- c(0)

# fit logistic regression model
riskmodel &lt;- fitLogRegModel(data=ExampleData, cOutcome=cOutcome,
cNonGenPreds=cNonGenPred, cNonGenPredsCat=cNonGenPredCat,
cGenPreds=cGenPred, cGenPredsCat=cGenPredCat)

# obtain multivariate OR(95% CI) for all predictors of the fitted model
ORmultivariate(riskModel=riskmodel)</code></pre>

<hr>
<h2 id='plotCalibration'>Function for calibration plot and Hosmer-Lemeshow goodness of fit test.</h2><span id='topic+plotCalibration'></span>

<h3>Description</h3>

<p>The function produces a calibration plot and provides Hosmer-Lemeshow 
goodness of fit test statistics.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCalibration(data, cOutcome, predRisk, groups, rangeaxis, 
plottitle, xlabel, ylabel, filename, fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCalibration_+3A_data">data</code></td>
<td>
<p>Data frame or numeric matrix that includes the outcome and 
predictor variables.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_predrisk">predRisk</code></td>
<td>
<p>Vector of predicted risks of all individuals in the dataset.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_groups">groups</code></td>
<td>
<p>Number of groups considered in 
Hosmer-Lemeshow test. Specification of <code>groups</code> is optional (default <code>groups</code> is 10).</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_rangeaxis">rangeaxis</code></td>
<td>
<p>Range of x-axis and y-axis. Specification of <code>rangeaxis</code> is optional. Default is <code>c(0,1)</code>.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> is optional. Default is &quot;Calibration plot&quot;.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_xlabel">xlabel</code></td>
<td>
<p>Label of x-axis Default. Specification of <code>xlabel</code> is optional. Default is &quot;Predicted risk&quot;.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> is optional. Default is &quot;Observed risk&quot;.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_filename">filename</code></td>
<td>
<p>Name of the output file in which the calibration table is saved. 
The file is saved as a txt file in the working directory. When  no 
<code>filename</code> is specified, the output is not saved. Example: filename=&quot;calibration.txt&quot;</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the file that contains the calibation plot.
The file is saved in the working directory in the format specified under <code>plottype</code>. Example:
<code>fileplot="plotname"</code>. Note that the extension is not specified here.
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotCalibration_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. Foe example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Details</h3>

<p>Hosmer-Lemeshow test statistic is a measure of the fit 
of the model, comparing observed and predicted risks across subgroups of 
the population. The default number of groups is 10. 
</p>
<p>The function requires the outcome of interest and predicted risks of 
all individuals. Predicted risks can be obtained from the 
functions <code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code> or 
be imported from other packages or methods.</p>


<h3>Value</h3>

<p>The function creates a calibration plot and returns the following measures: 
</p>
<table>
<tr><td><code>Chi_square</code></td>
<td>
<p>Chi square value of  Hosmer-Lemeshow test</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom, which is <code>(groups-2)</code> where <code>groups:</code> number 
of groups</p>
</td></tr>
<tr><td><code>p_value</code></td>
<td>
<p>p-value of Hosmer-Lemeshow test for goodness of fit</p>
</td></tr></table>


<h3>References</h3>

<p>Hosmer DW, Hosmer T, Le Cessie S, Lemeshow S. A comparison of
goodness-of-fit tests for the logistic regression model. Stat Med 1997;
16:965-980.</p>


<h3>See Also</h3>

<p><code><a href="#topic+predRisk">predRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables 
data(ExampleData)
# specify column number of the outcome variable
cOutcome &lt;- 2

# fit a logistic regression model
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk &lt;- predRisk(riskmodel)

# specify range of x-axis and y-axis  
rangeaxis &lt;- c(0,1) 
# specify number of groups for Hosmer-Lemeshow test            
groups &lt;- 10 

# compute calibration measures and produce calibration plot                          
plotCalibration(data=ExampleData, cOutcome=cOutcome, predRisk=predRisk, 
groups=groups, rangeaxis=rangeaxis)</code></pre>

<hr>
<h2 id='plotDiscriminationBox'>Function for box plots of predicted risks separately for individuals with and without the outcome of interest.</h2><span id='topic+plotDiscriminationBox'></span>

<h3>Description</h3>

<p>The function produces box plots of predicted risks for individuals with 
and without the outcome of interest and calculates the discrimination slope.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDiscriminationBox(data, cOutcome, predrisk, labels, plottitle, 
ylabel, fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDiscriminationBox_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome and 
predictors variables.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_predrisk">predrisk</code></td>
<td>
<p>Vector of predicted risks.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_labels">labels</code></td>
<td>
<p>Labels given to the groups of individuals without and with 
the outcome of interest. Specification of <code>label</code> is optional. 
Default is <code>c("Without disease", "With disease")</code>.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> 
is optional. Default is &quot;Box plot&quot;.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> 
is optional. Default is &quot;Predicted risks&quot;.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the file that contains the plot. The file is
saved in the working directory in the format specified under <code>plottype</code>. 
Example: <code>fileplot="name"</code>. Note that the extension is not specified here. 
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotDiscriminationBox_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. Foe example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Details</h3>

<p>The discrimination slope is the difference between the mean predicted risks 
of individuals with and without the outcome of interest. Predicted risks 
can be obtained using the 
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code> or be 
imported from other programs. The difference between discrimination 
slopes of two separate risk models is equivalent 
to (<code>IDI</code>) which is discussed 
in details in the <code><a href="#topic+reclassification">reclassification</a></code> function.</p>


<h3>Value</h3>

<p>The function creates a box plots of predicted risks for  
individuals with and without the outcome of interest and returns the discrimination slope.</p>


<h3>References</h3>

<p>Yates JF. External correspondence: decomposition of the mean 
probability score. Organizational Behavior and Human Performance 1982;30:132-156.</p>


<h3>See Also</h3>

<p><code><a href="#topic+reclassification">reclassification</a></code>, <code><a href="#topic+predRisk">predRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of outcome variable
cOutcome &lt;- 2

# fit a logistic regression model
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk &lt;- predRisk(riskmodel)
# specify labels for the groups without and with the outcome of interest 
labels &lt;- c("Without disease", "With disease")    

# produce discrimination box plot     
plotDiscriminationBox(data=ExampleData, cOutcome=cOutcome, predrisk=predRisk, 
labels=labels)</code></pre>

<hr>
<h2 id='plotPredictivenessCurve'>Function for predictiveness curve.</h2><span id='topic+plotPredictivenessCurve'></span>

<h3>Description</h3>

<p>The function creates a plot of cumulative percentage 
of individuals to the predicted risks.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPredictivenessCurve(predrisk, rangeyaxis, labels, plottitle, 
xlabel, ylabel, fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPredictivenessCurve_+3A_predrisk">predrisk</code></td>
<td>
<p>Vector of predicted risk. When multiple curves need to 
be presented in one plot, specify multiple vectors of predicted 
risks as <code>predrisk=cbind(predrisk1, predrisk2,...,predriskn)</code>.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_rangeyaxis">rangeyaxis</code></td>
<td>
<p>Range of the y axis. Default <code>rangeyaxis</code> is <code>c(0,1)</code>.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_labels">labels</code></td>
<td>
<p>Label(s) given to the predictiveness curve(s). Specification of <code>labels</code> is optional.
When specified, the <code>labels</code> should be in the same order as specified in <code>predrisk</code>.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> is optional. Default is &quot;Predictiveness curve&quot;.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_xlabel">xlabel</code></td>
<td>
<p>Label of x-axis. Specification of <code>xlabel</code> is optional. Default is &quot;Cumulative percentage&quot;.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> is optional. Default is &quot;Predicted risks&quot;.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the output file that contains the plot. The file is
saved in the working directory in the format specified under <code>plottype</code>. Example:
<code>fileplot="plotname"</code>. Note that the extension is not specified here. 
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotPredictivenessCurve_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. For example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Details</h3>

<p>The Predictiveness curve is a plot of cumulative percentage 
of individuals to the predicted risks. Cumulative percentage indicates 
the percentage of individual that has a predicted risk equal or lower 
than the risk value. 
Predicted risks can be obtained using the functions 
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code> 
or be imported from other methods or packages.</p>


<h3>Value</h3>

<p>The function creates a predictiveness curve.</p>


<h3>References</h3>

<p>Pepe MS, Feng Z, Huang Y, et al. Integrating the predictiveness   
of a marker with its performance as a classifier. 
Am J Epidemiol 2008;167:362-368.</p>


<h3>See Also</h3>

<p><code><a href="#topic+predRisk">predRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables 
data(ExampleData)

# fit logistic regression models
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel1 &lt;- ExampleModels()$riskModel1
riskmodel2 &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk1 &lt;- predRisk(riskmodel1)
predRisk2 &lt;- predRisk(riskmodel2)

# specify range of y-axis
rangeyaxis &lt;- c(0,1) 
# specify labels of the predictiveness curves
labels &lt;- c("without genetic factors", "with genetic factors")

# produce predictiveness curves
plotPredictivenessCurve(predrisk=cbind(predRisk1,predRisk2),
rangeyaxis=rangeyaxis, labels=labels)</code></pre>

<hr>
<h2 id='plotPriorPosteriorRisk'>Function to plot posterior risks against prior risks.</h2><span id='topic+plotPriorPosteriorRisk'></span>

<h3>Description</h3>

<p>Function to plot posterior risks against prior risks.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPriorPosteriorRisk(data, priorrisk, posteriorrisk, cOutcome, plottitle, 
xlabel, ylabel, rangeaxis, plotAll=TRUE, labels, filename, fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPriorPosteriorRisk_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome and 
predictors variables.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_priorrisk">priorrisk</code></td>
<td>
<p>Vector of predicted risks based on initial model.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_posteriorrisk">posteriorrisk</code></td>
<td>
<p>Vector of predicted risks based on updated model.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> is optional. Default is &quot;PriorPosteriorRisk plot&quot;.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_xlabel">xlabel</code></td>
<td>
<p>Label of x-axis. Specification of <code>xlabel</code> is optional. Default is &quot;Prior risk&quot;.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> is optional. Default is &quot;Posterior risk&quot;.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_rangeaxis">rangeaxis</code></td>
<td>
<p>Range of x-axis and y-axis. Specification of <code>rangeaxis</code> is optional. Default is <code>c(0,1)</code>.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_plotall">plotAll</code></td>
<td>
<p><code>plotAll=TRUE</code> will create one plot for the 
total population. When  <code>plotAll=FALSE</code> separate plots will be created 
for individuals with and without the outcome of interest.   
means two separate plots for with and without outcome of interest.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_labels">labels</code></td>
<td>
<p>Labels given to the groups of individuals without and with 
the outcome of interest. Default <code>labels</code> is
<code>c("without outcome", "with outcome")</code>. Note that when 
<code>plotAll=TRUE</code>, specification of <code>labels</code> is not necessary.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_filename">filename</code></td>
<td>
<p>Name of the output file in which prior and posterior 
risks for each individual with the outcome will be saved. If no directory is 
specified, the file is saved in the working directory as a txt file. 
When no <code>filename</code> is specified, the output is not saved.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the output file that contains the plot. The file is
saved in the working directory in the format specified under <code>plottype</code>. Example:
<code>fileplot="plotname"</code>. Note that the extension is not specified here. 
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotPriorPosteriorRisk_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. For example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function creates a plot of posterior risks (predicted risks using 
the updated model) against prior risks (predicted risks using the initial 
model). Predicted risks can be obtained using the functions 
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code> or be             
imported from other packages or methods.</p>


<h3>Value</h3>

<p>The function creates a plot of posterior risks against prior risks.</p>


<h3>See Also</h3>

<p><code><a href="#topic+predRisk">predRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of outcome variable
cOutcome &lt;- 2

# fit logistic regression models
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel1 &lt;- ExampleModels()$riskModel1
riskmodel2 &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk1 &lt;- predRisk(riskmodel1)
predRisk2 &lt;- predRisk(riskmodel2)

# specify label of x-axis 
xlabel &lt;- "Prior risk"                   
# specify label of y-axis
ylabel &lt;- "Posterior risk"               
# specify title for the plot
titleplot &lt;- "Prior versus posterior risk" 
# specify range of the x-axis and y-axis 
rangeaxis &lt;- c(0,1)                       
# labels given to the groups without and with the outcome of interest
labels&lt;- c("without outcome", "with outcome")           

# produce prior risks and posterior risks plot
plotPriorPosteriorRisk(data=ExampleData, priorrisk=predRisk1,
posteriorrisk=predRisk2, cOutcome=cOutcome, xlabel=xlabel, ylabel=ylabel,   
rangeaxis=rangeaxis, plotAll=TRUE, plottitle=titleplot, labels=labels)</code></pre>

<hr>
<h2 id='plotRiskDistribution'>Function to plot histogram of risks separated for individuals with and without the outcome of interest.</h2><span id='topic+plotRiskDistribution'></span>

<h3>Description</h3>

<p>Function to plot histogram of risks separated for individuals with and without the outcome of interest.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRiskDistribution(data, cOutcome, risks, interval, rangexaxis, 
rangeyaxis, plottitle, xlabel, ylabel, labels, fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRiskDistribution_+3A_data">data</code></td>
<td>
<p>Data frame or numeric matrix that includes the outcome and 
predictor variables.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_risks">risks</code></td>
<td>
<p>Risk of each individual. It
is specified by either a vector of risk scores or a vector of predicted risks.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_interval">interval</code></td>
<td>
<p>Size of the risk intervals. For example, <code>interval=.1</code> 
will construct the following intervals for predicted risks: 
<code>0-0.1, 0.1-0.2,..., 0.9-1</code>.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_rangexaxis">rangexaxis</code></td>
<td>
<p>Range of the x-axis. Specification of <code>rangexaxis</code> is optional.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_rangeyaxis">rangeyaxis</code></td>
<td>
<p>Range of the y-axis.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> is optional. Default is &quot;Histogram of risks&quot;.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_xlabel">xlabel</code></td>
<td>
<p>Label of x-axis. Specification of <code>xlabel</code> is optional. Default is &quot;Risk score&quot;.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> is optional. Default is &quot;Percentage&quot;.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_labels">labels</code></td>
<td>
<p>Labels given to the groups of individuals without and 
with the outcome of interest. Specification of <code>labels</code> is optional. Default is
<code>c("Without outcome", "With outcome")</code>.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the output file that contains the plot. The file is
saved in the working directory in the format specified under <code>plottype</code>. Example:
<code>fileplot="plotname"</code>. Note that the extension is not specified here. 
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotRiskDistribution_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. For example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Value</h3>

<p>The function creates the histogram of risks separated for individuals 
with and without the outcome of interest.</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotROC">plotROC</a></code>, <code><a href="#topic+riskScore">riskScore</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of the outcome variable
cOutcome &lt;- 2

# fit a logistic regression model
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk &lt;- predRisk(riskmodel)

# specify the size of each interval
interval &lt;- .05
# specify label of x-axis
xlabel &lt;- "Predicted risk"
# specify label of y-axis
ylabel &lt;- "Percentage"
# specify range of x-axis
xrange &lt;- c(0,1)
# specify range of y-axis
yrange &lt;- c(0,40)
# specify title for the plot
maintitle &lt;- "Distribution of predicted risks"
# specify labels
labels &lt;- c("Without outcome", "With outcome")

# produce risk distribution plot
plotRiskDistribution(data=ExampleData, cOutcome=cOutcome,
risks=predRisk, interval=interval, plottitle=maintitle, rangexaxis=xrange,
rangeyaxis=yrange, xlabel=xlabel, ylabel=ylabel, labels=labels)</code></pre>

<hr>
<h2 id='plotRiskscorePredrisk'>Function to plot predicted risks against risk scores.</h2><span id='topic+plotRiskscorePredrisk'></span>

<h3>Description</h3>

<p>This function is used to make a plot of predicted risks against risk scores.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRiskscorePredrisk(data, riskScore, predRisk, plottitle, xlabel, 
ylabel, rangexaxis, rangeyaxis, filename, fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRiskscorePredrisk_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome and 
predictors variables.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_riskscore">riskScore</code></td>
<td>
<p>Vector of (weighted or unweighted) genetic risk scores.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_predrisk">predRisk</code></td>
<td>
<p>Vector of predicted risks.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> is optional. Default is &quot;Risk score predicted risk plot&quot;.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_xlabel">xlabel</code></td>
<td>
<p>Label of x-axis. Specification of <code>xlabel</code> is optional. Default is &quot;Risk score&quot;.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> is optional. Default is &quot;Predicted risk&quot;.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_rangexaxis">rangexaxis</code></td>
<td>
<p>Range of the x axis. Specification of <code>rangexaxis</code> is optional.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_rangeyaxis">rangeyaxis</code></td>
<td>
<p>Range of the y axis. Specification of <code>rangeyaxis</code> is optional. Default is <code>c(0,1)</code>.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_filename">filename</code></td>
<td>
<p>Name of the output file in which risk scores and
predicted risks for each individual will be saved. If no directory is 
specified, the file is saved in the working directory as a txt file. 
When no <code>filename</code> is specified, the output is not saved.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the output file that contains the plot. The file is
saved in the working directory in the format specified under <code>plottype</code>. Example:
<code>fileplot="plotname"</code>. Note that the extension is not specified here. 
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotRiskscorePredrisk_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. For example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function creates a plot of predicted risks against risk scores. 
Predicted risks can be obtained using the functions
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code>
or be imported from other methods or packages. 
The function <code><a href="#topic+riskScore">riskScore</a></code> can be
used to compute unweighted or weighted risk scores.</p>


<h3>Value</h3>

<p>The function creates a plot of predicted risks against risk scores.</p>


<h3>See Also</h3>

<p><code><a href="#topic+riskScore">riskScore</a></code>, <code><a href="#topic+predRisk">predRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)

# fit a logistic regression  model
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk &lt;- predRisk(riskmodel)

# specify column numbers of genetic predictors
cGenPred &lt;- c(11:16)

# function to compute unweighted genetic risk scores
riskScore &lt;- riskScore(weights=riskmodel, data=ExampleData, 
cGenPreds=cGenPred, Type="unweighted")

# specify range of x-axis
rangexaxis &lt;- c(0,12)   
# specify range of y-axis
rangeyaxis &lt;- c(0,1)     
# specify label of x-axis
xlabel &lt;- "Risk score"     
# specify label of y-axis
ylabel &lt;- "Predicted risk" 
# specify title for the plot
plottitle &lt;- "Risk score versus predicted risk"

# produce risk score-predicted risk plot
plotRiskscorePredrisk(data=ExampleData, riskScore=riskScore, predRisk=predRisk, 
plottitle=plottitle, xlabel=xlabel, ylabel=ylabel, rangexaxis=rangexaxis, 
rangeyaxis=rangeyaxis)</code></pre>

<hr>
<h2 id='plotROC'>Function for a receiver operating characteristic curve (ROC) plot and area under the ROC curve (AUC) value.</h2><span id='topic+plotROC'></span>

<h3>Description</h3>

<p>The function produces ROC curve and corresponding AUC value with 95% CI. 
The function can plot one or multiple ROC curves in a single plot.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotROC(data, cOutcome, predrisk, labels, plottitle, xlabel, ylabel, 
fileplot, plottype)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotROC_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome and 
predictors variables.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_predrisk">predrisk</code></td>
<td>
<p>Vector of predicted risk. When multiple curves need to
be presented in one plot, specify multiple vectors of predicted
risks as <code>predrisk=cbind(predrisk1, predrisk2,...,predriskn)</code>.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_labels">labels</code></td>
<td>
<p>Label(s) given to the ROC curve(s). Specification of <code>labels</code> is optional.
When specified, the <code>labels</code> should be in the same order as specified in <code>predrisk</code>.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_plottitle">plottitle</code></td>
<td>
<p>Title of the plot. Specification of <code>plottitle</code> is optional. Default is &quot;ROC plot&quot;.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_xlabel">xlabel</code></td>
<td>
<p>Label of x-axis. Specification of <code>xlabel</code> is optional. Default is &quot;1- Specificity&quot;.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_ylabel">ylabel</code></td>
<td>
<p>Label of y-axis. Specification of <code>ylabel</code> is optional. Default is &quot;Sensitivity&quot;.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_fileplot">fileplot</code></td>
<td>
<p>Name of the output file that contains the plot. The file is
saved in the working directory in the format specified under <code>plottype</code>. Example:
<code>fileplot="plotname"</code>. Note that the extension is not specified here.
When <code>fileplot</code> is not specified, the plot is not saved.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_plottype">plottype</code></td>
<td>
<p>The format in which the plot is saved. Available formats are
wmf, emf, png, jpg, jpeg, bmp, tif, tiff, ps,
eps or pdf. For example, <code>plottype="eps"</code> will save the plot in eps format.
When <code>plottype</code> is not specified, the plot will be saved in jpg format.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function requirs predicted risks or risk scores and the outcome of 
interest for all individuals. 
Predicted risks can be obtained using the functions
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code>
or be imported from other methods or packages.</p>


<h3>Value</h3>

<p>The function creates ROC plot and returns AUC value with 95% CI.</p>


<h3>References</h3>

<p>Hanley JA, McNeil BJ. The meaning and use of the area under a  
receiver operating characteristic (ROC) curve. Radiology 1982;143:29-36.
</p>
<p>Tobias Sing, Oliver Sander, Niko Beerenwinkel, Thomas Lengauer.
ROCR: visualizing classifier performance in R.
Bioinformatics 2005;21(20):3940-3941.</p>


<h3>See Also</h3>

<p><code><a href="#topic+predRisk">predRisk</a></code>, <code><a href="#topic+plotRiskDistribution">plotRiskDistribution</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify the arguments in the function to produce ROC plot
# specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of the outcome variable
cOutcome &lt;- 2

# fit logistic regression models
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel1 &lt;- ExampleModels()$riskModel1
riskmodel2 &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk1 &lt;- predRisk(riskmodel1)
predRisk2 &lt;- predRisk(riskmodel2)

# specify label of the ROC curve
labels &lt;- c("without genetic factors", "with genetic factors")

# produce ROC curve
plotROC(data=ExampleData, cOutcome=cOutcome, 
predrisk=cbind(predRisk1,predRisk2), labels=labels)</code></pre>

<hr>
<h2 id='predRisk'>Function to compute predicted risks for all individuals in the dataset.</h2><span id='topic+predRisk'></span>

<h3>Description</h3>

<p>Function to compute predicted risks for all individuals in the (new)dataset.</p>


<h3>Usage</h3>

<pre><code class='language-R'>predRisk(riskModel, data, cID, filename)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predRisk_+3A_riskmodel">riskModel</code></td>
<td>
<p>Name of logistic regression model that can be fitted using  
the function <code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code>.</p>
</td></tr>
<tr><td><code id="predRisk_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the ID number and 
predictor variables.</p>
</td></tr>
<tr><td><code id="predRisk_+3A_cid">cID</code></td>
<td>
<p>Column number of ID variable. The ID number and predicted risks 
will be saved under <code>filename</code>. When <code>cID</code> is not specified, the output is not saved.</p>
</td></tr>
<tr><td><code id="predRisk_+3A_filename">filename</code></td>
<td>
<p>Name of the output file in which the ID number and 
estimated predicted risks will be saved. The file is saved in the working  
directory as a txt file. Example: filename=&quot;name.txt&quot;. When no <code>filename</code> 
is specified, the output is not saved.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function computes predicted risks from a specified logistic regression model.   
The function <code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> can be used to construct such a model.</p>


<h3>Value</h3>

<p>The function returns a vector of predicted risks.</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code>, <code><a href="#topic+plotCalibration">plotCalibration</a></code>, 
<code><a href="#topic+plotROC">plotROC</a></code>, <code><a href="#topic+plotPriorPosteriorRisk">plotPriorPosteriorRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData)
# specify column number of the outcome variable
cOutcome &lt;- 2
# specify column number of ID variable
cID &lt;- 1
# specify column numbers of non-genetic predictors
cNonGenPred &lt;- c(3:10)
# specify column numbers of non-genetic predictors that are categorical
cNonGenPredCat &lt;- c(6:8)
# specify column numbers of genetic predictors
cGenPred &lt;- c(11,13:16)
# specify column numbers of genetic predictors that are categorical
cGenPredCat &lt;- c(0)

# fit logistic regression model
riskmodel &lt;- fitLogRegModel(data=ExampleData, cOutcome=cOutcome,
cNonGenPreds=cNonGenPred, cNonGenPredsCat=cNonGenPredCat,
cGenPreds=cGenPred, cGenPredsCat=cGenPredCat)

# obtain predicted risks 
predRisk &lt;- predRisk(riskModel=riskmodel)</code></pre>

<hr>
<h2 id='reclassification'>Function for reclassification table and statistics.</h2><span id='topic+reclassification'></span>

<h3>Description</h3>

<p>The function creates a reclassification table and provides statistics.</p>


<h3>Usage</h3>

<pre><code class='language-R'>reclassification(data, cOutcome, predrisk1, predrisk2, cutoff)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reclassification_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome and 
predictors variables.</p>
</td></tr>
<tr><td><code id="reclassification_+3A_coutcome">cOutcome</code></td>
<td>
<p>Column number of the outcome variable.</p>
</td></tr>
<tr><td><code id="reclassification_+3A_predrisk1">predrisk1</code></td>
<td>
<p>Vector of predicted risks of all individuals using initial
model.</p>
</td></tr>
<tr><td><code id="reclassification_+3A_predrisk2">predrisk2</code></td>
<td>
<p>Vector of predicted risks of all individuals using updated
model.</p>
</td></tr>
<tr><td><code id="reclassification_+3A_cutoff">cutoff</code></td>
<td>
<p>Cutoff values for risk categories. 
Define the cut-off values as <code>c(0,...,1)</code>. 
Multiple values can be defined and always specify <code>0</code> and <code>1</code>. 
Example: <code>c(0,.20,.30,1)</code></p>
</td></tr></table>


<h3>Details</h3>

<p>The function creates a reclassification table and computes the 
categorical and continuous net reclassification improvement (<code>NRI</code>) and 
integrated discrimination improvement (<code>IDI</code>). A reclassification table 
indicates the number of individuals who move to another risk category or remain 
in the same risk category as a result of updating the risk model. Categorical <code>NRI</code> equal to 
<code>x%</code> means that compared with individuals without outcome, 
individuals with outcome were almost <code>x%</code> more likely to move up a category than down.
The function also computes continuous <code>NRI</code>, which does not require any discrete 
risk categories and relies on the proportions of individuals with outcome 
correctly assigned a higher probability and individuals without outcome 
correctly assigned a lower probability by an updated model compared with the 
initial model.
<code>IDI</code> equal to <code>x%</code> means that the difference in average 
predicted risks between the individuals with and without the outcome  
increased by <code>x%</code> in the updated model.
The function requires predicted risks estimated by using two separate risk 
models. Predicted risks can be obtained using the functions
<code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code> and <code><a href="#topic+predRisk">predRisk</a></code>
or be imported from other methods or packages. p-values in NRI and IDI were rounded upto five decimal places.</p>


<h3>Value</h3>

<p>The function returns the reclassification table, separately  
for individuals with and without the outcome of interest and the following measures: 
</p>
<table>
<tr><td><code>NRI (Categorical)</code></td>
<td>
<p>Categorical Net Reclassification Improvement with 95% CI and <code>p-value</code> of the test</p>
</td></tr>
<tr><td><code>NRI (Continuous)</code></td>
<td>
<p>Continuous Net Reclassification Improvement with 95% CI and <code>p-value</code> of the test</p>
</td></tr>
<tr><td><code>IDI</code></td>
<td>
<p>Integrated Discrimination Improvement with 95% CI and <code>p-value</code> 
of the test</p>
</td></tr></table>


<h3>References</h3>

<p>Cook NR. Use and misuse of the receiver operating characteristic
curve in risk prediction. Circulation 2007;115(7):928-935.      
</p>
<p>Pencina MJ, D'Agostino RB Sr, D'Agostino RB Jr, Vasan RS.
Evaluating the added predictive ability of a new marker: from
area under the ROC curve to reclassification and beyond. Stat
Med 2008;27(2):157-172; discussion 207-212.</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotDiscriminationBox">plotDiscriminationBox</a></code>, <code><a href="#topic+predRisk">predRisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables 
data(ExampleData)
# specify column number of the outcome variable
cOutcome &lt;- 2

# fit logistic regression models
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel1 &lt;- ExampleModels()$riskModel1
riskmodel2 &lt;- ExampleModels()$riskModel2

# obtain predicted risks
predRisk1 &lt;- predRisk(riskmodel1)
predRisk2 &lt;- predRisk(riskmodel2)
# specify cutoff values for risk categories
cutoff &lt;- c(0,.10,.30,1)    

# compute reclassification measures
reclassification(data=ExampleData, cOutcome=cOutcome, 
predrisk1=predRisk1, predrisk2=predRisk2, cutoff)</code></pre>

<hr>
<h2 id='riskScore'>Function to compute genetic risk scores.</h2><span id='topic+riskScore'></span>

<h3>Description</h3>

<p>The function computes unweighted 
or weighted genetic risk scores. The relative effects (or weights) of 
genetic variants can either come from beta coefficients of a risk model 
or from a vector of beta coefficients imported into R, e.g., when beta cofficients are obtained from meta-analysis.</p>


<h3>Usage</h3>

<pre><code class='language-R'>riskScore(weights, data, cGenPreds, Type)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="riskScore_+3A_weights">weights</code></td>
<td>
<p>The vector that includes the weights given to the genetic 
variants. See details for more informations.</p>
</td></tr>
<tr><td><code id="riskScore_+3A_data">data</code></td>
<td>
<p>Data frame or matrix that includes the outcome  
and predictors variables.</p>
</td></tr>
<tr><td><code id="riskScore_+3A_cgenpreds">cGenPreds</code></td>
<td>
<p>Column numbers of the genetic variables on the basis of   
which the risk score is computed.</p>
</td></tr>
<tr><td><code id="riskScore_+3A_type">Type</code></td>
<td>
<p>Specification of the type of risk scores that will be computed.
Type can be weighted (<code>Type="weighted"</code>) or 
unweighted (<code>Type="unweighted"</code>).</p>
</td></tr></table>


<h3>Details</h3>

<p>The function calculates unweighted 
or weighted genetic risk scores. The unweighted genetic risk score is a simple
risk allele count assuming that all alleles have the same effect. For this
calculation, it is required that the genetic variables are coded as the number of risk
alleles. Beta coefficients are used to determine which allele is the risk
allele. When the sign of the beta coefficient is negative, the allele coding
is reversed. The weighted risk score is a sum of the number of risk alleles
multiplied by their beta coefficients.
</p>
<p>The beta coefficients can come from two different sources, either beta coefficients of a risk model 
or a vector of beta coefficients imported into R, e.g., when beta cofficients are obtained from meta-analysis.
This vector of beta coefficients 
should be a named vector containing the same names as mentioned in genetic variants.
A logistic regression model can be constructed using <code><a href="#topic+fitLogRegModel">fitLogRegModel</a></code>
from this package.</p>


<h3>Value</h3>

<p>The function returns a vector of risk scores.</p>


<h3>Note</h3>

<p>When a vector of beta coefficients is imported, it should be checked 
whether the DNA strands and the coding of the risk alleles are the same 
as in the study data. The functions are available in the package <code>GenABEL</code>
to accurately compute risk scores when the DNA strands are different or the risk 
alleles are coded differently in the study data and the data used in meta-analysis.</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotRiskDistribution">plotRiskDistribution</a></code>, <code><a href="#topic+plotRiskscorePredrisk">plotRiskscorePredrisk</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># specify dataset with outcome and predictor variables
data(ExampleData) 
# specify column numbers of genetic predictors
cGenPred &lt;- c(11:16)

# fit a logistic regression model
# all steps needed to construct a logistic regression model are written in a function
# called 'ExampleModels', which is described on page 4-5
riskmodel &lt;- ExampleModels()$riskModel2

# compute unweighted risk scores 
riskScore &lt;- riskScore(weights=riskmodel, data=ExampleData, 
cGenPreds=cGenPred, Type="unweighted")</code></pre>

<hr>
<h2 id='simulatedDataset'>Function to construct a simulated dataset containing individual genotype  
data, genetic risks and disease status for a hypothetical population.</h2><span id='topic+simulatedDataset'></span>

<h3>Description</h3>

<p>Construct a dataset that contains individual genotype data, genetic risk,   
and disease status for a hypothetical population. 
The dataset is constructed using simulation in such a way that the frequencies 
and odds ratios (ORs) of the genetic variants and the population disease risk 
computed from this dataset are the same as specified by the input parameters.</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulatedDataset(ORfreq, poprisk, popsize, filename)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulatedDataset_+3A_orfreq">ORfreq</code></td>
<td>
<p>Matrix with ORs and frequencies of the genetic variants. 
The matrix contains four columns in which the first two describe ORs and the 
last two describe the corresponding frequencies. The number of rows in this 
matrix is same as the number of genetic variants included. Genetic variants 
can be specified as per genotype, per allele, or as dominant/ recessive 
effect of the risk allele. When per genotype data are used, OR of the 
heterozygous and homozygous risk genotypes are mentioned in the first two 
columns and the corresponding genotype frequencies are mentioned in the last 
two columns. When per allele data are used, the OR and frequency of the risk 
allele are specified in the first and third column and the remaining two cells 
are coded as '1'.  Similarly, when dominant/ recessive effects of the risk 
alleles are used, the 
OR and frequency of the dominant/ recessive variant are specified in the first 
and third column, and the remaining two cells are coded as '0'.</p>
</td></tr>
<tr><td><code id="simulatedDataset_+3A_poprisk">poprisk</code></td>
<td>
<p>Population disease risk (expressed in proportion).</p>
</td></tr>
<tr><td><code id="simulatedDataset_+3A_popsize">popsize</code></td>
<td>
<p>Total number of individuals included in the dataset.</p>
</td></tr>
<tr><td><code id="simulatedDataset_+3A_filename">filename</code></td>
<td>
<p>Name of the file in which the dataset will be saved. 
The file is saved in the working directory as a txt file. When no filename 
is specified, the output is not saved.</p>
</td></tr></table>


<h3>Details</h3>

<p>The function will execute when the matrix with odds ratios and frequencies, 
population disease risk and the number of individuals are specified. <br /> 
</p>
<p>The simulation method is described in detail in the references. <br /> 
</p>
<p>The method assumes that (i) the combined effect of the genetic variants 
on disease risk follows a multiplicative (log additive) risk model; 
(ii) genetic variants inherit independently, that is no linkage disequilibrium 
between the variants; (iii) genetic variants have independent effects on the 
disease risk, which indicates no interaction among variants; and (iv) all 
genotypes and allele proportions are in Hardy-Weinberg equilibrium. 
Assumption (ii) and (iv) are used to generate the genotype data, and assumption 
(ii) and (iii) are used to calculate disease risk.
</p>
<p>Simulating the dataset involves three steps: (1) modelling genotype data, 
(2) modelling disease risks, and (3) modelling disease status. Brief 
descriptions of these steps are as follows:
</p>
<p>(1) Modelling genotype data: For each variant the genotype 
frequencies are either specified or calculated from the allele frequencies 
using Hardy-Weinberg equilibrium. Then, the genotypes for each genetic 
variant are randomly distributed without replacement over all individuals.
</p>
<p>(2) Modelling disease risks: For the calculation of the individual disease 
risk, Bayes' theorem is used, which states that the posterior odds of disease 
are obtained by multiplying the prior odds by the likelihood ratio (LR) of 
the individual genotype data. The prior odds are calculated from the 
population disease risk or disease prevalence 
(prior odds= prior risk/ (1- prior risk)) and the posterior odds are converted 
back into disease risk (disease risk= posterior odds/ (1+ posterior odds)). 
Under the no linkage disequilibrium (LD) assumption, the LR of a genetic profile  
is obtained by multiplying the LRs of the single genotypes that are included in 
the risk model. The LR of a single genotype is calculated using frequencies 
and ORs of genetic variants and population disease risk. See references 
for more details.
</p>
<p>(3) Modelling disease status: To model disease status, we used a procedure 
that compares the estimated disease risk of each subject to a randomly drawn value 
between 0 and 1 from a uniform distribution. A subject is assigned to the 
group who will develop the disease when the disease risk is higher than the 
random value and to the group who will not develop the disease when the risk 
is lower than the random value. 
</p>
<p>This procedure ensures that for each genomic profile, the percentage of 
people who will develop the disease equals the population disease risk 
associated with that profile, when the subgroup of individuals with that 
profile is sufficiently large.</p>


<h3>Value</h3>

<p>The function returns:
</p>
<table>
<tr><td><code>Dataset</code></td>
<td>
<p>A data frame or matrix that includes genotype data, 
genetic risk and disease status for a hypothetical population. 
The dataset contains (4 + number of genetic variants included) columns,  
in which the first column is the un-weighted risk score, which is the sum 
of the number of risk alleles for each individual, the third column is the 
estimated genetic risk, the forth column is the individual disease status expressed 
as '0' or '1', indicating without or with the outcome of interest, and the fifth until 
the end column are genotype data for the variants expressed as '0', '1' or '2', 
which indicate the number of risk alleles present in each individual for the genetic variants.</p>
</td></tr></table>


<h3>References</h3>

<p>Janssens AC, Aulchenko YS, Elefante S, Borsboom GJ, Steyerberg EW, 
van Duijn CM. Predictive testing for complex diseases using multiple genes: 
fact or fiction? Genet Med. 2006;8:395-400.
</p>
<p>Kundu S, Karssen LC, Janssens AC: Analytical and simulation methods for 
estimating the potential predictive ability of genetic profiling: a comparison 
of methods and results. Eur J Hum Genet. 2012 May 30.
</p>
<p>van Zitteren M, van der Net JB, Kundu S, Freedman AN, van Duijn CM, 
Janssens AC. Genome-based prediction of breast cancer risk in the general 
population: a modeling study based on meta-analyses of genetic associations. 
Cancer Epidemiol Biomarkers Prev. 2011;20:9-22.
</p>
<p>van der Net JB, Janssens AC, Sijbrands EJ, Steyerberg EW. Value of genetic 
profiling for the prediction of coronary heart disease. 
Am Heart J. 2009;158:105-10.
</p>
<p>Janssens AC, Moonesinghe R, Yang Q, Steyerberg EW, van Duijn CM, Khoury MJ. 
The impact of genotype frequencies on the clinical validity of genomic 
profiling for predicting common chronic diseases. Genet Med. 2007;9:528-35.</p>


<h3>Examples</h3>

<pre><code class='language-R'># specify the matrix containing the ORs and frequencies of genetic variants 
# In this example we used per allele effects of the risk variants
ORfreq&lt;-cbind(c(1.35,1.20,1.24,1.16), rep(1,4), c(.41,.29,.28,.51),rep(1,4))

# specify the population disease risk
popRisk &lt;- 0.3
# specify size of hypothetical population
popSize &lt;- 10000

# Obtain the simulated dataset
Data &lt;- simulatedDataset(ORfreq=ORfreq, poprisk=popRisk, popsize=popSize)

# Obtain the AUC and produce ROC curve
plotROC(data=Data, cOutcome=4, predrisk=Data[,3])</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
