<!DOCTYPE html><html><head><title>Help for package predhy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {predhy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#convertgen'><p>Convert Genotype</p></a></li>
<li><a href='#crodesign'><p>Generate Mating Design</p></a></li>
<li><a href='#cv'><p>Evaluate Trait Predictability via Cross Validation</p></a></li>
<li><a href='#cv_fast'><p>Evaluate Trait Predictability via the HAT Method</p></a></li>
<li><a href='#hybrid_phe'><p>Phenotypic data of hybrids</p></a></li>
<li><a href='#infergen'><p>Infer Genotype of Hybrids</p></a></li>
<li><a href='#input_geno'><p>Genotype in Hapmap Format</p></a></li>
<li><a href='#input_geno1'><p>Genotype in Numeric Format</p></a></li>
<li><a href='#kin'><p>Calculate Kinship Matrix</p></a></li>
<li><a href='#mixed'><p>Solve Mixed Model</p></a></li>
<li><a href='#predhy.predict'><p>Predict the Performance of Hybrids</p></a></li>
<li><a href='#predhy.predict_NCII'><p>Predict the Performance of Hybrids</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genomic Prediction of Hybrid Performance</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Yang Xu, Guangning Yu, Yanru Cui, Shizhong Xu, Chenwu Xu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yang Xu &lt;xuyang_89@126.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs genomic prediction of hybrid performance using eight GS methods including GBLUP, BayesB, RKHS, PLS, LASSO, Elastic net, LightGBM and XGBoost. It also provides fast cross-validation and mating design scheme for training population (Xu S et al (2016) &lt;<a href="https://doi.org/10.1111%2Ftpj.13242">doi:10.1111/tpj.13242</a>&gt;; Xu S (2017) &lt;<a href="https://doi.org/10.1534%2Fg3.116.038059">doi:10.1534/g3.116.038059</a>&gt;). </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>BGLR, pls, glmnet, xgboost, lightgbm, foreach, doParallel,
parallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-29 04:11:16 UTC; Administrator</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-29 05:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='convertgen'>Convert Genotype</h2><span id='topic+convertgen'></span>

<h3>Description</h3>

<p>Convert genotypes in HapMap format or in numeric format for hypred package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertgen(
  input_geno,
  type = c("hmp1", "hmp2", "num"),
  missingrate = 0.2,
  maf = 0.05,
  impute = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convertgen_+3A_input_geno">input_geno</code></td>
<td>
<p>genotype in HapMap format or in numeric format. The names of individuals should be provided. Missing (NA) values are allowed.</p>
</td></tr>
<tr><td><code id="convertgen_+3A_type">type</code></td>
<td>
<p>the type of genotype. There are three options: &quot;hmp1&quot; for genotypes in HapMap format with single bit, &quot;hmp2&quot; for genotypes in HapMap format with double bit, and &quot;num&quot; for genotypes in numeric format.</p>
</td></tr>
<tr><td><code id="convertgen_+3A_missingrate">missingrate</code></td>
<td>
<p>max missing percentage for each SNP, default is 0.2.</p>
</td></tr>
<tr><td><code id="convertgen_+3A_maf">maf</code></td>
<td>
<p>minor allele frequency for each SNP, default is 0.05.</p>
</td></tr>
<tr><td><code id="convertgen_+3A_impute">impute</code></td>
<td>
<p>logical. If TRUE, imputation. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of genotypes in numeric format, coded as 1, 0, -1 for AA, Aa, aa. Each row represents an individual and each column represents a marker. The rownames of the matrix are the names of individuals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load genotype in HapMap format with double bit
data(input_geno)

## convert genotype for hypred package
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")


## load genotype in numeric format
data(input_geno1)
head(input_geno1)

## convert genotype for hypred package
inbred_gen1 &lt;- convertgen(input_geno1, type = "num")


</code></pre>

<hr>
<h2 id='crodesign'>Generate Mating Design</h2><span id='topic+crodesign'></span>

<h3>Description</h3>

<p>Generate a mating design for a subset of crosses based on a balanced random partial rectangle cross-design (BRPRCD) (Xu et al. 2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crodesign(d, male_name, female_name, seed = 123)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crodesign_+3A_d">d</code></td>
<td>
<p>an integer denoting 1/d percentage of crosses to be evaluated in the field.</p>
</td></tr>
<tr><td><code id="crodesign_+3A_male_name">male_name</code></td>
<td>
<p>a character string for the names of male parents.</p>
</td></tr>
<tr><td><code id="crodesign_+3A_female_name">female_name</code></td>
<td>
<p>a character string for the names of male parents.</p>
</td></tr>
<tr><td><code id="crodesign_+3A_seed">seed</code></td>
<td>
<p>the random number, default is 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of mating design result with three columns. The first column is &quot;crossID&quot;, the second column is the &quot;male_Name&quot; and the third column is the &quot;female_Name&quot;.
</p>


<h3>References</h3>

<p>Xu S, Xu Y, Gong L and Zhang Q. (2016) Metabolomic prediction of yield in hybrid rice. Plant J. 88, 219-227.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## generate a mating design with 100 male parents and 150 female parents
## for 1/d = 1/50 percentage of crosses to be evaluated in the field.
## the total number of potential crosses is 100*150 = 15000.
## The number of crosses to be field evaluated is 15000*(1/50) = 300.

male_name &lt;- paste("m", 1:100, sep = "")
female_name &lt;- paste("f", 1:150, sep = "")
design &lt;- crodesign(d = 50, male_name, female_name)

</code></pre>

<hr>
<h2 id='cv'>Evaluate Trait Predictability via Cross Validation</h2><span id='topic+cv'></span>

<h3>Description</h3>

<p>The cv function evaluates trait predictability based on eight GS methods via k-fold cross validation.
The trait predictability is defined as the squared Pearson correlation coefficient between the observed and the predicted trait values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv(
  fix = NULL,
  gena,
  gend = NULL,
  inbred_phe = NULL,
  y,
  method = "GBLUP",
  drawplot = TRUE,
  nfold = 5,
  nTimes = 1,
  seed = 1234,
  CPU = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_+3A_fix">fix</code></td>
<td>
<p>a design matrix of the fixed effects.</p>
</td></tr>
<tr><td><code id="cv_+3A_gena">gena</code></td>
<td>
<p>a matrix (n x m) of additive genotypes for the training population.</p>
</td></tr>
<tr><td><code id="cv_+3A_gend">gend</code></td>
<td>
<p>a matrix (n x m) of domiance genotypes for the training population. Default is NULL.</p>
</td></tr>
<tr><td><code id="cv_+3A_inbred_phe">inbred_phe</code></td>
<td>
<p>a matrix (n x 2) of inbred_phe phenotypic.Default is NULL.</p>
</td></tr>
<tr><td><code id="cv_+3A_y">y</code></td>
<td>
<p>a vector(n x 1) of the phenotypic values.</p>
</td></tr>
<tr><td><code id="cv_+3A_method">method</code></td>
<td>
<p>eight GS methods including &quot;GBLUP&quot;, &quot;BayesB&quot;, &quot;RKHS&quot;, &quot;PLS&quot;, &quot;LASSO&quot;, &quot;EN&quot;, &quot;XGBoost&quot;,&quot;LightGBM&quot;.
Users may select one of these methods or all of them simultaneously with &quot;ALL&quot;. Default is &quot;GBLUP&quot;.</p>
</td></tr>
<tr><td><code id="cv_+3A_drawplot">drawplot</code></td>
<td>
<p>when method =&quot;ALL&quot;, user may select TRUE for a barplot about eight GS methods. Default is TRUE.</p>
</td></tr>
<tr><td><code id="cv_+3A_nfold">nfold</code></td>
<td>
<p>the number of folds. Default is 5.</p>
</td></tr>
<tr><td><code id="cv_+3A_ntimes">nTimes</code></td>
<td>
<p>the number of independent replicates for the cross-validation. Default is 1.</p>
</td></tr>
<tr><td><code id="cv_+3A_seed">seed</code></td>
<td>
<p>the random number. Default is 1234.</p>
</td></tr>
<tr><td><code id="cv_+3A_cpu">CPU</code></td>
<td>
<p>the number of CPU.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Trait predictability
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from hypred package
data(hybrid_phe)
data(input_geno)

## convert original genotype
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")

##additive model infer the additive and dominance genotypes of hybrids
gena &lt;- infergen(inbred_gen, hybrid_phe)$add
gend &lt;- infergen(inbred_gen, hybrid_phe)$dom

##additive model
R2&lt;-cv(fix=NULL,gena,gend=NULL,y=hybrid_phe[,3],inbred_phe=NULL,method ="GBLUP")

##additive-dominance model
R2&lt;-cv(fix=NULL,gena,gend,y=hybrid_phe[,3],inbred_phe=NULL,method ="GBLUP")
</code></pre>

<hr>
<h2 id='cv_fast'>Evaluate Trait Predictability via the HAT Method</h2><span id='topic+cv_fast'></span>

<h3>Description</h3>

<p>The HAT method is a fast algorithm for the ordinary cross validation. It is highly recommended for large dataset (Xu et al. 2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv_fast(fix = NULL, y, kk, nfold = 5, seed = 123)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_fast_+3A_fix">fix</code></td>
<td>
<p>a design matrix of the fixed effects. If not passed, a vector of ones is added for the intercept.</p>
</td></tr>
<tr><td><code id="cv_fast_+3A_y">y</code></td>
<td>
<p>a vector of the phenotypic values.</p>
</td></tr>
<tr><td><code id="cv_fast_+3A_kk">kk</code></td>
<td>
<p>a list of one or multiple kinship matrices.</p>
</td></tr>
<tr><td><code id="cv_fast_+3A_nfold">nfold</code></td>
<td>
<p>the number of folds, default is 5. For the HAT Method, nfold can be set as the sample size (leave-one-out CV) to avoid
variation caused by random partitioning of the samples, but it is not recommended for <code><a href="#topic+cv">cv</a></code>.</p>
</td></tr>
<tr><td><code id="cv_fast_+3A_seed">seed</code></td>
<td>
<p>the random number, default is 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Trait predictability
</p>


<h3>References</h3>

<p>Xu S. (2017) Predicted residual error sum of squares of mixed models: an application for genomic prediction. G3 (Bethesda) 7, 895-909.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from hypred package
data(hybrid_phe)
data(input_geno)

## convert original genotype
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")

## infer the additive and dominance genotypes of hybrids
gena &lt;- infergen(inbred_gen, hybrid_phe)$add
gend &lt;- infergen(inbred_gen, hybrid_phe)$dom

## calculate the additive and dominance kinship matrix
ka &lt;- kin(gena)
kd &lt;- kin(gend)

##for the additive model
predictability &lt;- cv_fast(y = hybrid_phe[,3], kk = list(ka))

##for the additive-dominance model
predictability &lt;- cv_fast(y = hybrid_phe[,3], kk = list(ka,kd))


</code></pre>

<hr>
<h2 id='hybrid_phe'>Phenotypic data of hybrids</h2><span id='topic+hybrid_phe'></span>

<h3>Description</h3>

<p>This dataset contains phenotypic data of 410 hybrids for grain yield in maize.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hybrid_phe
</code></pre>


<h3>Format</h3>

<p>A data frame with 410 rows and 3 variables:
</p>

<dl>
<dt><code>M</code></dt><dd><p>The names of male parents.</p>
</dd>
<dt><code>F</code></dt><dd><p>The names of female parents.</p>
</dd>
<dt><code>GY</code></dt><dd><p>The grain yield of hybrids.</p>
</dd>
</dl>


<hr>
<h2 id='infergen'>Infer Genotype of Hybrids</h2><span id='topic+infergen'></span>

<h3>Description</h3>

<p>Infer additive and dominance genotypes of hybrids based on their parental genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infergen(inbred_gen, hybrid_phe)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infergen_+3A_inbred_gen">inbred_gen</code></td>
<td>
<p>a matrix for genotypes of parental lines in numeric format, coded as 1, 0 and -1. The row.names of inbred_gen must be provied. It can be obtained from the original genotype using <code><a href="#topic+convertgen">convertgen</a></code> function.</p>
</td></tr>
<tr><td><code id="infergen_+3A_hybrid_phe">hybrid_phe</code></td>
<td>
<p>a data frame with three columns. The first column and the second column are the names of male and female parents of the corresponding hybrids, respectively; the third column is the phenotypic values of hybrids.
The names of male and female parents must match the rownames of inbred_gen. Missing (NA) values are not allowed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with following information is returned:
</p>
<p>$add  additive genotypes of hybrids
</p>
<p>$dom  dominance genotypes of hybrids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from hypred package
data(hybrid_phe)
head(hybrid_phe)
data(input_geno)

## convert original genotype
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")

gena &lt;- infergen(inbred_gen, hybrid_phe)$add
gend &lt;- infergen(inbred_gen, hybrid_phe)$dom

</code></pre>

<hr>
<h2 id='input_geno'>Genotype in Hapmap Format</h2><span id='topic+input_geno'></span>

<h3>Description</h3>

<p>Genotypic data of 348 maize inbred lines in Hapmap format with double bit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_geno
</code></pre>


<h3>Format</h3>

<p>A data frame with 4979 rows and 359 columns.
</p>

<hr>
<h2 id='input_geno1'>Genotype in Numeric Format</h2><span id='topic+input_geno1'></span>

<h3>Description</h3>

<p>Genotypic data of 50 rice inbred lines with 1000 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_geno1
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows and 50 variables.
</p>

<hr>
<h2 id='kin'>Calculate Kinship Matrix</h2><span id='topic+kin'></span>

<h3>Description</h3>

<p>Calculate the additive and dominance kinship matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kin(gen)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kin_+3A_gen">gen</code></td>
<td>
<p>a matrix for genotypes, coded as 1, 0, -1 for AA, Aa, aa. Each row represents an individual and each column represents a marker.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a kinship matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## random population with 100 lines and 1000 markers
gen &lt;- matrix(rep(0,100*1000),100,1000)
gen &lt;- apply(gen,2,function(x){x &lt;- sample(c(-1,0,1), 100, replace = TRUE)})

## generate 100*100 kinship matrix
k &lt;- kin(gen)

</code></pre>

<hr>
<h2 id='mixed'>Solve Mixed Model</h2><span id='topic+mixed'></span>

<h3>Description</h3>

<p>Solve linear mixed model using restricted maximum likelihood (REML). Multiple variance components can be estimated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixed(fix = NULL, y, kk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixed_+3A_fix">fix</code></td>
<td>
<p>a design matrix of the fixed effects. If not passed, a vector of ones is added for the intercept.</p>
</td></tr>
<tr><td><code id="mixed_+3A_y">y</code></td>
<td>
<p>a vector of the phenotypic values.</p>
</td></tr>
<tr><td><code id="mixed_+3A_kk">kk</code></td>
<td>
<p>a list of one or multiple kinship matrices.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with following information is returned:
</p>
<p>$v_i  the inverse of the phenotypic variance-covariance matrix
</p>
<p>$var  estimated variance components of genetic effects
</p>
<p>$ve   estimated residual variance
</p>
<p>$beta estimated fixed effects
</p>


<h3>References</h3>

<p>Xu S, Zhu D and Zhang Q. (2014) Predicting hybrid performance in rice using genomic best linear unbiased prediction. Proc. Natl. Acad. Sci. USA 111, 12456-12461.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from hypred package
data(hybrid_phe)
data(input_geno)

## convert original genotype
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")

## infer the additive and dominance genotypes of hybrids
gena &lt;- infergen(inbred_gen, hybrid_phe)$add
gend &lt;- infergen(inbred_gen, hybrid_phe)$dom

## calculate the additive and dominance kinship matrix
ka &lt;- kin(gena)
kd &lt;- kin(gend)

## for the additive model
parm &lt;- mixed(y = hybrid_phe[,3], kk = list(ka))

## for the additive-dominance model
parm &lt;- mixed(y = hybrid_phe[,3], kk = list(ka, kd))


</code></pre>

<hr>
<h2 id='predhy.predict'>Predict the Performance of Hybrids</h2><span id='topic+predhy.predict'></span>

<h3>Description</h3>

<p>Predict all potential crosses of a given set of parents using a subset of crosses as the training sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predhy.predict(
  inbred_gen,
  hybrid_phe,
  inbred_phe = NULL,
  method = "GBLUP",
  model = "A",
  select = "top",
  number = "100"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predhy.predict_+3A_inbred_gen">inbred_gen</code></td>
<td>
<p>a matrix for genotypes of parental lines in numeric format, coded as 1, 0 and -1. The row.names of inbred_gen must be provied. It can be obtained from the original genotype using  <code><a href="#topic+convertgen">convertgen</a></code> function.</p>
</td></tr>
<tr><td><code id="predhy.predict_+3A_hybrid_phe">hybrid_phe</code></td>
<td>
<p>a data frame with three columns. The first column and the second column are the names of male and female parents of the corresponding hybrids, respectively; the third column is the phenotypic values of hybrids.
The names of male and female parents must match the rownames of inbred_gen. Missing (NA) values are not allowed.</p>
</td></tr>
<tr><td><code id="predhy.predict_+3A_inbred_phe">inbred_phe</code></td>
<td>
<p>a matrix (n x 2) of inbred_phe phenotypic.Default is NULL.</p>
</td></tr>
<tr><td><code id="predhy.predict_+3A_method">method</code></td>
<td>
<p>eight GS methods including &quot;GBLUP&quot;, &quot;BayesB&quot;, &quot;RKHS&quot;, &quot;PLS&quot;, &quot;LASSO&quot;, &quot;EN&quot;, &quot;XGBoost&quot;, &quot;LightGBM&quot;.
Users may select P of these methods. Default is &quot;GBLUP&quot;.</p>
</td></tr>
<tr><td><code id="predhy.predict_+3A_model">model</code></td>
<td>
<p>the prediction model. There are four options: model = &quot;A&quot; for the additive model, model = &quot;AD&quot; for the additive-dominance model,model = &quot;A-P&quot; for the additive-phenotypic model,model = &quot;AD-P&quot; for the additive-dominance-phenotypic model. Default is model = &quot;A&quot;.</p>
</td></tr>
<tr><td><code id="predhy.predict_+3A_select">select</code></td>
<td>
<p>the selection of hybrids based on the prediction results. There are three options: select = &quot;all&quot;, which selects all potential crosses. select = &quot;top&quot;, which selects the top n crosses. select = &quot;bottom&quot;, which selects the bottom n crosses. The n is determined by the param number.</p>
</td></tr>
<tr><td><code id="predhy.predict_+3A_number">number</code></td>
<td>
<p>the number of selected top or bottom hybrids, only when select = &quot;top&quot; or select = &quot;bottom&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame of prediction results with two columns. The first column denotes the names of male and female parents of the predicted hybrids, and the second column denotes the phenotypic values of the predicted hybrids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from hypred package
data(hybrid_phe)
data(input_geno)
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")

## infer the additive and dominance genotypes of hybrids
gena &lt;- infergen(inbred_gen, hybrid_phe)$add
gend &lt;- infergen(inbred_gen, hybrid_phe)$dom

pred&lt;-predhy.predict(inbred_gen,hybrid_phe,method="LASSO",model="A",select="top",number="100")
pred&lt;-predhy.predict(inbred_gen,hybrid_phe,method="LASSO",model="AD",select="all")
 
</code></pre>

<hr>
<h2 id='predhy.predict_NCII'>Predict the Performance of Hybrids</h2><span id='topic+predhy.predict_NCII'></span>

<h3>Description</h3>

<p>Predict all potential crosses of a given set of parents using a subset of crosses as the training sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predhy.predict_NCII(
  inbred_gen,
  hybrid_phe,
  inbred_phe = NULL,
  male_name = hybrid_phe[, 1],
  female_name = hybrid_phe[, 2],
  method = "GBLUP",
  model = "A",
  select = "top",
  number = "100"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predhy.predict_NCII_+3A_inbred_gen">inbred_gen</code></td>
<td>
<p>a matrix for genotypes of parental lines in numeric format, coded as 1, 0 and -1. The row.names of inbred_gen must be provied. It can be obtained from the original genotype using  <code><a href="#topic+convertgen">convertgen</a></code> function.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_hybrid_phe">hybrid_phe</code></td>
<td>
<p>a data frame with three columns. The first column and the second column are the names of male and female parents of the corresponding hybrids, respectively; the third column is the phenotypic values of hybrids.
The names of male and female parents must match the rownames of inbred_gen. Missing (NA) values are not allowed.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_inbred_phe">inbred_phe</code></td>
<td>
<p>a matrix (n x 2) of inbred_phe phenotypic.Default is NULL.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_male_name">male_name</code></td>
<td>
<p>a vector of the names of male parents.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_female_name">female_name</code></td>
<td>
<p>a vector of the names of female parents.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_method">method</code></td>
<td>
<p>eight GS methods including &quot;GBLUP&quot;, &quot;BayesB&quot;, &quot;RKHS&quot;, &quot;PLS&quot;, &quot;LASSO&quot;, &quot;EN&quot;, &quot;XGBoost&quot;, &quot;LightGBM&quot;.
Users may select P of these methods. Default is &quot;GBLUP&quot;.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_model">model</code></td>
<td>
<p>the prediction model. There are two options: model = &quot;A&quot; for the additive model, model = &quot;AD&quot; for the additive-dominance model,model = &quot;A-P&quot; for the additive-phenotypic model,model = &quot;AD-P&quot; for the additive-dominance-phenotypic model. Default is model = &quot;A&quot;.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_select">select</code></td>
<td>
<p>the selection of hybrids based on the prediction results. There are three options: select = &quot;all&quot;, which selects all potential crosses. select = &quot;top&quot;, which selects the top n crosses. select = &quot;bottom&quot;, which selects the bottom n crosses. The n is determined by the param number.</p>
</td></tr>
<tr><td><code id="predhy.predict_NCII_+3A_number">number</code></td>
<td>
<p>the number of selected top or bottom hybrids, only when select = &quot;top&quot; or select = &quot;bottom&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame of prediction results with two columns. The first column denotes the names of male and female parents of the predicted hybrids, and the second column denotes the phenotypic values of the predicted hybrids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from hypred package
data(hybrid_phe)
data(input_geno)
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")

## infer the additive and dominance genotypes of hybrids
gena &lt;- infergen(inbred_gen, hybrid_phe)$add
gend &lt;- infergen(inbred_gen, hybrid_phe)$dom

pred&lt;-predhy.predict_NCII(inbred_gen,hybrid_phe,method="LASSO",model="A")
pred&lt;-predhy.predict_NCII(inbred_gen,hybrid_phe,method="LASSO",model = "AD",select="all")
 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
