<!DOCTYPE html><html><head><title>Help for package preseqR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {preseqR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bbc.rSAC'>
<p>BBC estimator</p></a></li>
<li><a href='#cs.rSAC'>
<p>CS estimator</p></a></li>
<li><a href='#Dickens'><p>Dickens' vocabulary</p></a></li>
<li><a href='#ds.rSAC'>
<p>RFA estimator</p></a></li>
<li><a href='#ds.rSAC.bootstrap'>
<p>RFA estimator with bootstrap</p></a></li>
<li><a href='#fisher.alpha'>
<p>Parameter alpha in the logseries estimator</p></a></li>
<li><a href='#fisher.rSAC'>
<p>Logseries estimator</p></a></li>
<li><a href='#FisherButterfly'><p>Fisher's butterfly data</p></a></li>
<li><a href='#kmer.frac.curve'>
<p>Fraction of <code class="reqn">k</code>-mers observed at least <code class="reqn">r</code> times</p></a></li>
<li><a href='#kmer.frac.curve.bootstrap'>
<p>Fraction of <code class="reqn">k</code>-mers observed at least <code class="reqn">r</code> times with bootstrap</p></a></li>
<li><a href='#preseqR-package'>
<p>Predicting <code class="reqn">r</code>-species accumulation curves</p></a></li>
<li><a href='#preseqR.interpolate.rSAC'>
<p>Interpolation</p></a></li>
<li><a href='#preseqR.nonreplace.sampling'>
<p>Sampling</p></a></li>
<li><a href='#preseqR.optimal.sequencing'>
<p>Optimal amount of sequencing for scWGS</p></a></li>
<li><a href='#preseqR.rSAC'>
<p>Best practice for <code class="reqn">r</code>-SAC &ndash; a fast version</p></a></li>
<li><a href='#preseqR.rSAC.bootstrap'>
<p>Best practice for <code class="reqn">r</code>-SAC</p></a></li>
<li><a href='#preseqR.rSAC.sequencing.rmdup'>
<p>Predicting <code class="reqn">r</code>-SAC in WES/WGS</p></a></li>
<li><a href='#preseqR.sample.cov'>
<p>Predicting generalized sample coverage</p></a></li>
<li><a href='#preseqR.sample.cov.bootstrap'>
<p>Predicting generalized sample coverage with bootstrap</p></a></li>
<li><a href='#preseqR.simu.hist'>
<p>Simulation</p></a></li>
<li><a href='#preseqR.ztnb.em'>
<p>Fitting a zero-truncated negative binomial distribution</p></a></li>
<li><a href='#Shakespeare'><p>Shakespeare's word type frequencies</p></a></li>
<li><a href='#SRR061157_k31'><p><code class="reqn">k</code>-mer counts of a metagenomic data</p></a></li>
<li><a href='#SRR1301329_1M_base'><p>Coverage histogram of a WES data</p></a></li>
<li><a href='#SRR1301329_1M_read'><p>Read counts of a WES data</p></a></li>
<li><a href='#SRR1301329_base'><p>Coverage histogram of a WES data</p></a></li>
<li><a href='#SRR1301329_read'><p>Read counts of a WES data</p></a></li>
<li><a href='#SRR611492'><p>Coverage histogram of a scWGS data</p></a></li>
<li><a href='#SRR611492_5M'><p>Coverage histogram of a scWGS data</p></a></li>
<li><a href='#Twitter'><p>Social network</p></a></li>
<li><a href='#WillButterfly'><p>Fisher's butterfly data</p></a></li>
<li><a href='#ztnb.rSAC'>
<p>ZTNB estimator</p></a></li>
<li><a href='#ztp.rSAC'>
<p>ZTP estimator</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Predicting Species Accumulation Curves</td>
</tr>
<tr>
<td>Version:</td>
<td>4.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-06-27</td>
</tr>
<tr>
<td>Author:</td>
<td>Chao Deng, Timothy Daley and Andrew D. Smith</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chao Deng &lt;chaodeng@usc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Originally as an R version of Preseq &lt;<a href="https://doi.org/10.1038%2Fnmeth.2375">doi:10.1038/nmeth.2375</a>&gt;, the package has extended its functionality to predict the r-species accumulation curve (r-SAC), which is the number of species represented at least r times as a function of the sampling effort. When r = 1, the curve is known as the species accumulation curve, or the library complexity curve in high-throughput genomic sequencing. The package includes both parametric and nonparametric methods, as described by Deng C, et al. (2018) &lt;<a href="https://arxiv.org/abs/1607.02804v3">arXiv:1607.02804v3</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>polynom, graphics, stats</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-27 10:02:27 UTC; JAKE</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-27 10:52:56 UTC</td>
</tr>
</table>
<hr>
<h2 id='bbc.rSAC'>
BBC estimator
</h2><span id='topic+bbc.rSAC'></span>

<h3>Description</h3>

<p><code>bbc.rSAC</code> predicts the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample, based on the initial sample. 
The estimator was originally proposed by Boneh et al. (1998) for estimating
the SAC. We generalize this estimator for predicting the <code class="reqn">r</code>-SAC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbc.rSAC(n, r=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbc.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="bbc.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Boneh, S., Boneh, A., &amp; Caron, R. J. (1998). Estimating the prediction function
and the number of unseen species in sampling with replacement.
Journal of the American Statistical Association, 93(441), 372-379.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for SAC
bbc1 &lt;- bbc.rSAC(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
bbc1(c(10, 20))

## construct the estimator for r-SAC
bbc2 &lt;- bbc.rSAC(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
bbc2(c(50, 100))
</code></pre>

<hr>
<h2 id='cs.rSAC'>
CS estimator
</h2><span id='topic+cs.rSAC'></span>

<h3>Description</h3>

<p><code>cs.rSAC</code> predicts the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample, based on the initial sample. 
The estimator was originally proposed by Chao and Shen (2004) for estimating
the SAC. We generalize this estimator for predicting the <code class="reqn">r</code>-SAC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs.rSAC(n, r=1, k=10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cs.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="cs.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="cs.rSAC_+3A_k">k</code></td>
<td>

<p>A cutoff for common species. Default is 10.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Chao, A., &amp; Shen, T. J. (2004). Nonparametric prediction in species sampling.       
Journal of agricultural, biological, and environmental statistics, 9(3), 253-269.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for SAC
chao1 &lt;- cs.rSAC(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
chao1(c(10, 20))

## construct the estimator for r-SAC
chao2 &lt;- cs.rSAC(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
chao2(c(50, 100))
</code></pre>

<hr>
<h2 id='Dickens'>Dickens' vocabulary</h2><span id='topic+Dickens'></span>

<h3>Description</h3>

<p>Words frequencies of a collection of Charles Dickens from Project Gutenberg</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of unique words appeared exactly <code class="reqn">j</code>
times in a collection of Charles Dickens.
</p>


<h3>References</h3>

<p><a href="http://zipfr.r-forge.r-project.org/">http://zipfr.r-forge.r-project.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(Dickens)
</code></pre>

<hr>
<h2 id='ds.rSAC'>
RFA estimator
</h2><span id='topic+ds.rSAC'></span>

<h3>Description</h3>

<p><code>ds.rSAC</code> predicts the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample, based on the initial sample. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ds.rSAC(n, r=1, mt=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ds.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="ds.rSAC_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="ds.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The estimator is based on an empirical Bayes approach using rational
function approximation (RFA), as described in the paper in the 
references section.
</p>
<p><code>ds.rSAC</code> is the fast version of <code><a href="#topic+ds.rSAC.bootstrap">ds.rSAC.bootstrap</a></code>.
The function does not provide the confidence interval. To obtain the
confidence interval along with the estimates, one should use the function
<code><a href="#topic+ds.rSAC.bootstrap">ds.rSAC.bootstrap</a></code>.
</p>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for SAC
ds1 &lt;- ds.rSAC(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
ds1(c(10, 20))

## construct the estimator for r-SAC
ds2 &lt;- ds.rSAC(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
ds2(c(50, 100))
</code></pre>

<hr>
<h2 id='ds.rSAC.bootstrap'>
RFA estimator with bootstrap
</h2><span id='topic+ds.rSAC.bootstrap'></span>

<h3>Description</h3>

<p><code>ds.rSAC.bootstrap</code> predicts the expected number of species 
represented at least <code class="reqn">r</code> times in a random sample, 
based on the initial sample. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ds.rSAC.bootstrap(n, r=1, mt=20, times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ds.rSAC.bootstrap_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="ds.rSAC.bootstrap_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="ds.rSAC.bootstrap_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="ds.rSAC.bootstrap_+3A_times">times</code></td>
<td>

<p>The number of bootstrap samples. Default is 30.
</p>
</td></tr>
<tr><td><code id="ds.rSAC.bootstrap_+3A_conf">conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the bootstrap version of <code><a href="#topic+ds.rSAC">ds.rSAC</a></code>. The bootstrap
sample is generated by randomly sampling the initial sample with replacement.
For each bootstrap sample, we construct an estimator. The median of
estimates is used as the prediction for the number of species
represented at least <code class="reqn">r</code> times in a random sample. 
</p>
<p>The confidence interval is constructed based on a lognormal distribution.
</p>


<h3>Value</h3>

<table>
<tr><td><code>f</code></td>
<td>

<p>The estimator for the number of species represented at least <code class="reqn">r</code> times in 
a random sample. The input of the estimator is a vector of sampling 
efforts t, i.e. the relative sample sizes comparing with the initial sample.
For example, t = 2 means a random sample that is twice the size of the 
initial sample.
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>The standard error for the estimator. The input is a vector of sampling 
efforts <code class="reqn">t</code>.
</p>
</td></tr>
<tr><td><code>lb</code></td>
<td>

<p>The lower bound of the confidence interval.The input is a vector of sampling 
efforts <code class="reqn">t</code>.
</p>
</td></tr>
<tr><td><code>ub</code></td>
<td>

<p>The upper bound of the confidence interval.The input is a vector of sampling 
efforts <code class="reqn">t</code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Efron, B., &amp; Tibshirani, R. J. (1994). An introduction to the bootstrap. CRC press.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
# library(preseqR)

## import data
# data(FisherButterfly)

## construct the estimator for SAC
# ds1 &lt;- ds.rSAC.bootstrap(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
# ds1$f(c(10, 20))
## The standard error of the estiamtes
# ds1$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- ds1$lb(c(10, 20))
# ub &lt;- ds1$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

## construct the estimator for SAC
# ds2 &lt;- ds.rSAC.bootstrap(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
# ds2$f(c(50, 100))
## The standard error of the estiamtes
# ds2$se(c(50, 100))
## The confidence interval of the estimates
# lb &lt;- ds2$lb(c(50, 100))
# ub &lt;- ds2$ub(c(50, 100))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)
</code></pre>

<hr>
<h2 id='fisher.alpha'>
Parameter alpha in the logseries estimator
</h2><span id='topic+fisher.alpha'></span>

<h3>Description</h3>

<p><code>fisher.alpha</code> estimates the parameter alpha in the logseries estimator by
Fisher, R. A., et al. (1943) based on an initial sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher.alpha(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisher.alpha_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A double, the estimated value of the parameter alpha
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Fisher, R., Corbet, A., &amp; Williams, C. (1943). The Relation Between the Number 
of Species and the Number of Individuals in a Random Sample of an Animal 
Population. Journal of Animal Ecology, 12(1), 42-58. doi:10.2307/1411  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(WillButterfly)

## estimating alpha
fisher.alpha &lt;- fisher.alpha(WillButterfly)
</code></pre>

<hr>
<h2 id='fisher.rSAC'>
Logseries estimator
</h2><span id='topic+fisher.rSAC'></span>

<h3>Description</h3>

<p><code>fisher.rSAC</code> estimates the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample, based on the initial sample.
The estimator was originally proposed by Fisher et al. (1943) for estimating
the SAC. We generalize this estimator for predicting the <code class="reqn">r</code>-SAC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher.rSAC(n, r=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisher.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="fisher.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Fisher, R., Corbet, A., &amp; Williams, C. (1943). The Relation Between the Number 
of Species and the Number of Individuals in a Random Sample of an Animal 
Population. Journal of Animal Ecology, 12(1), 42-58. doi:10.2307/1411  
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(WillButterfly)

## construct the estimator for SAC
fisher1 &lt;- fisher.rSAC(WillButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
fisher1(c(10, 20))

## construct the estimator for r-SAC
fisher2 &lt;- fisher.rSAC(WillButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
fisher2(c(50, 100))
</code></pre>

<hr>
<h2 id='FisherButterfly'>Fisher's butterfly data</h2><span id='topic+FisherButterfly'></span>

<h3>Description</h3>

<p>Frequencies data of butterflies collected in the Malay peninsula
was from Fisher, R. A., Corbet, A. S., &amp; Williams, C. B. (1943).</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">n_j</code>, the number of butterflies captured <code class="reqn">j</code>
times in the sample.
</p>


<h3>References</h3>

<p>Fisher, R. A., Corbet, A. S., and Williams, C. B. ,1943, The Relation Between 
the Number of Species and the Number of Individuals in a Random Sample of an 
Animal Population, Journal of Animal Ecology, 12, 42-58, Table 1,2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(FisherButterfly)
</code></pre>

<hr>
<h2 id='kmer.frac.curve'>
Fraction of <code class="reqn">k</code>-mers observed at least <code class="reqn">r</code> times 
</h2><span id='topic+kmer.frac.curve'></span>

<h3>Description</h3>

<p><code>kmer.frac.curve</code> predicts the expected fraction of <code class="reqn">k</code>-mers observed at
least <code class="reqn">r</code> times in a high-throughput sequencing experiment given the 
amount of sequencing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  kmer.frac.curve(n, k, read.len, seq, r=2, mt=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmer.frac.curve_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of <code class="reqn">k</code>-mers observed exactly <code class="reqn">j</code>
times in the initial experiment. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve_+3A_k">k</code></td>
<td>

<p>The number of nucleotides in a <code class="reqn">k</code>-mer.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve_+3A_read.len">read.len</code></td>
<td>

<p>The average length of a read.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve_+3A_seq">seq</code></td>
<td>

<p>The amount of nucleotides sequenced..
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>kmer.frac.curve</code> is mainly designed for metagenomics to evaluate how 
saturated a metagenomic data is.
</p>
<p><code>kmer.frac.curve</code> is the fast version of <code><a href="#topic+kmer.frac.curve.bootstrap">kmer.frac.curve.bootstrap</a></code>.
The function does not provide the confidence interval. To obtain the
confidence interval along with the estimates, one should use the function
<code><a href="#topic+kmer.frac.curve.bootstrap">kmer.frac.curve.bootstrap</a></code>.
</p>


<h3>Value</h3>

<p>A two-column matrix. The first column is the amount of sequencing in an 
experiment. The second column is the estimate of the fraction of <code class="reqn">k</code>-mers observed at least
<code class="reqn">r</code> times in the experiment.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C and Smith, AD (2016). Estimating the number of species to attain 
sufficient representation in a random sample. arXiv preprint arXiv:1607.02804
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(SRR061157_k31)

## the fraction of 31-mers represented at least 10 times in an experiment when
## sequencing 1M, 10M, 100M, 1G, 10G, 100G, 1T nucleotides
kmer.frac.curve(n=SRR061157_k31, k=31, read.len=100, seq=10^(6:12), r=10, mt=20) 
</code></pre>

<hr>
<h2 id='kmer.frac.curve.bootstrap'>
Fraction of <code class="reqn">k</code>-mers observed at least <code class="reqn">r</code> times with bootstrap
</h2><span id='topic+kmer.frac.curve.bootstrap'></span>

<h3>Description</h3>

<p><code>kmer.frac.curve</code> predicts the expected fraction of <code class="reqn">k</code>-mers observed at
least <code class="reqn">r</code> times in a high-throughput sequencing experiment given the 
amount of sequencing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmer.frac.curve.bootstrap(n, k, read.len, seq, r=2, mt=20, times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of <code class="reqn">k</code>-mers observed exactly <code class="reqn">j</code>
times in the initial experiment. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_k">k</code></td>
<td>

<p>The number of nucleotides in a <code class="reqn">k</code>-mer.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_read.len">read.len</code></td>
<td>

<p>The average length of a read.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_seq">seq</code></td>
<td>

<p>The amount of nucleotides sequenced.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_times">times</code></td>
<td>

<p>The number of bootstrap samples.
</p>
</td></tr>
<tr><td><code id="kmer.frac.curve.bootstrap_+3A_conf">conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the bootstrap version of <code><a href="#topic+kmer.frac.curve">kmer.frac.curve</a></code>. The bootstrap
sample is generated by randomly sampling the initial sample with replacement.
For each bootstrap sample, we construct an estimator. The median of
estimates is used as the prediction for the number of species
represented at least <code class="reqn">r</code> times in a random sample. 
</p>
<p>The confidence interval is constructed based on a lognormal distribution.
</p>


<h3>Value</h3>

<p>A four-column matrix. The first column is the amount of sequencing in an 
experiment.
The second column is the estimate of the fraction of <code class="reqn">k</code>-mers observed at least
<code class="reqn">r</code> times in the experiment. The third and fourth columns are the lower
bounds and the upper bounds of the confidence intervals.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Efron, B., &amp; Tibshirani, R. J. (1994). An introduction to the bootstrap. CRC press.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
# library(preseqR)

## import data
# data(SRR061157_k31)

## the fraction of 31-mers represented at least 10 times in an experiment when
## sequencing 1M, 10M, 100M, 1G, 10G, 100G, 1T nucleotides
# kmer.frac.curve.bootstrap(n=SRR061157_k31, k=31, read.len=100, 
#                          seq=10^(6:12), r=10, mt=20) 
</code></pre>

<hr>
<h2 id='preseqR-package'>
Predicting <code class="reqn">r</code>-species accumulation curves
</h2><span id='topic+preseqR-package'></span>

<h3>Description</h3>

<p>The functionality of this package is to predict <code class="reqn">r</code>-species
accumulaiton curves. The method is based on a nonparametric empirical Bayes approach
with rational function approximation. The estimator is
excellent in accuracy for both large values of <code class="reqn">r</code> and long-range
extrapolations, which are essential to large-scale applications. Some
examples are predicting the molecular complexity of sequencing
libraries, estimating the minimum sufficient sequencing depths for
whole-exome sequencing experiments and optimizing depths for single-cell
whole-genome sequencing experiments. 
</p>


<h3>Details</h3>

<p>main functions: 
</p>
<p>preseqR.rSAC
</p>
<p>preseqR.rSAC.bootstrap
</p>
<p>preseqR.optimal.sequencing
</p>
<p>preseqR.rSAC.sequencing.rmdup
</p>
<p>preseqR.sample.cov
</p>
<p>preseqR.sample.cov.bootstrap
</p>


<h3>Author(s)</h3>

<p>Chao Deng, Timothy Daley, and Andrew D. Smith
</p>
<p>Maintainer: Chao Deng &lt;chaodeng@usc.edu&gt;
</p>


<h3>References</h3>

<p>Baker, G. A., &amp; Graves-Morris, P. (1996). Pade approximants
(Encyclopedia of Mathematics and its Applications vol 59).
</p>
<p>Boneh, S., Boneh, A., &amp; Caron, R. J. (1998). Estimating the prediction function
and the number of unseen species in sampling with replacement.
Journal of the American Statistical Association, 93(441), 372-379.
</p>
<p>Chao, A., &amp; Shen, T. J. (2004). Nonparametric prediction in species sampling.
Journal of agricultural, biological, and environmental statistics, 9(3), 253-269.
</p>
<p>Cohen Jr, A. C. (1960). Estimating the parameters of a modified Poisson distribution.
Journal of the American Statistical Association, 55(289), 139-143.
</p>
<p>Daley, T., &amp; Smith, A. D. (2013). Predicting the molecular complexity of
sequencing libraries. Nature methods, 10(4), 325-327.
</p>
<p>Deng C, Daley T &amp; Smith AD (2015). Applications of species accumulation curves
in large-scale biological data analysis. Quantitative Biology, 3(3), 135-144. 
URL http://dx.doi.org/10.1007/s40484-015-0049-7.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>
<p>Efron, B., &amp; Thisted, R. (1976). Estimating the number of unseen species:
How many words did Shakespeare know?. Biometrika, 63(3), 435-447.
</p>
<p>Efron, B. (1979). Bootstrap methods: another look at the jackknife.
The annals of Statistics, 1-26.
</p>
<p>Efron, B., &amp; Tibshirani, R. J. (1994). An introduction to the bootstrap. CRC press.
</p>
<p>Fisher, R. A., Corbet, A. S., and Williams, C. B. ,1943, The Relation Between 
the Number of Species and the Number of Individuals in a Random Sample of an  
Animal Population, Journal of Animal Ecology, 12, 42-58.
</p>
<p>Good, I. J., &amp; Toulmin, G. H. (1956). The number of new species, and the
increase in population coverage, when a sample is increased.
Biometrika, 43(1-2), 45-63.
</p>
<p>Heck Jr, K. L., van Belle, G., &amp; Simberloff, D. (1975). 
Explicit calculation of the rarefaction diversity measurement and the 
determination of sufficient sample size. Ecology, 1459-1461.
</p>
<p>Kalinin V (1965). Functionals related to the poisson distribution and               
statistical structure of a text. Articles on Mathematical Statistics and            
the Theory of Probability pp. 202-220.
</p>

<hr>
<h2 id='preseqR.interpolate.rSAC'>
Interpolation
</h2><span id='topic+preseqR.interpolate.rSAC'></span>

<h3>Description</h3>

<p>Interpolating the number of species represented at least <code class="reqn">r</code> times
in a subsample given an initial sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preseqR.interpolate.rSAC(n, ss, r=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.interpolate.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.interpolate.rSAC_+3A_ss">ss</code></td>
<td>

<p>A positive double equal to the step size between subsamples.
</p>
</td></tr>
<tr><td><code id="preseqR.interpolate.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The expected number of species represented at least <code class="reqn">r</code>
times in the subsample is estimated based on an expended version of the
formula by Heck Jr, KL. et al. (1975).
</p>


<h3>Value</h3>

<p>A two-column matrix for the number of species represented at least <code class="reqn">r</code>
times in a random sample. The first column is the size of the random sample; 
the second column is the expected number of species represented at least <code class="reqn">r</code>
times in the sample.
</p>
<p>NULL if failed.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Heck Jr, K. L., van Belle, G., &amp; Simberloff, D. (1975). 
Explicit calculation of the rarefaction diversity measurement and the
determination of sufficient sample size. Ecology, 1459-1461.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(Shakespeare)

## The expected number of distinct words represented twice or more in the
## subsample 
preseqR.interpolate.rSAC(n=Shakespeare, ss=1e5, r=2)
</code></pre>

<hr>
<h2 id='preseqR.nonreplace.sampling'>
Sampling
</h2><span id='topic+preseqR.nonreplace.sampling'></span>

<h3>Description</h3>

<p>Generating a histogram by subsampling without replacement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  preseqR.nonreplace.sampling(n, size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.nonreplace.sampling_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species represented exactly <code class="reqn">j</code> times in the initial
sample. The first column must be sorted in an ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.nonreplace.sampling_+3A_size">size</code></td>
<td>

<p>An positive integer representing the size of the subsample.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>preseqR.nonreplace.sampling</code> generates a subsample by sampling the
initial sample without replacement.
<code>sample</code> in <code class="reqn">R</code> is used to implement the function. We wrap up
this function in such a way that both the input and the output are histograms. 
</p>


<h3>Value</h3>

<p>A two-column matrix as a subsample.
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species represented <code class="reqn">j</code> times in the subsample. 
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p><a href="https://stat.ethz.ch/R-manual/R-patched/library/base/html/sample.html">https://stat.ethz.ch/R-manual/R-patched/library/base/html/sample.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)
## import data
data(FisherButterfly)
## generate a subsample of size 1000.
preseqR.nonreplace.sampling(n=FisherButterfly, size=1000)
</code></pre>

<hr>
<h2 id='preseqR.optimal.sequencing'>
Optimal amount of sequencing for scWGS
</h2><span id='topic+preseqR.optimal.sequencing'></span>

<h3>Description</h3>

<p><code>preseqR.optimal.sequencing</code> predicts the optimal amount of sequencing in
a single-cell whole-genome sequencing (scWGS) experiment based on a shallow sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preseqR.optimal.sequencing(n, efficiency=0.05, bin=1e8, r=1, mt=20, 
                           times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.optimal.sequencing_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">n_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.optimal.sequencing_+3A_efficiency">efficiency</code></td>
<td>

<p>The minimum benefit-cost ratio
</p>
</td></tr>
<tr><td><code id="preseqR.optimal.sequencing_+3A_bin">bin</code></td>
<td>

<p>One unit of sequencing effort. Default is 1e8.
</p>
</td></tr>
<tr><td><code id="preseqR.optimal.sequencing_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.optimal.sequencing_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="preseqR.optimal.sequencing_+3A_times">times</code></td>
<td>

<p>The number of bootstrap samples.
</p>
</td></tr>
<tr><td><code id="preseqR.optimal.sequencing_+3A_conf">conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>preseqR.optimal.sequencing</code> predicts the optimal amount of sequencing
in a scWGS experiment. The term optimal is interpreted as the maximum 
amount of sequencing with its benefit-cost ratio greater than a given threshold.
The benefit-cost ratio is defined as the probability of a new nucleotide in the
genome represented at least <code class="reqn">r</code> times when one more base is sequenced. 
In order to improve the numeric stability, we use the mean of new nucleotdies
with coverage at least <code class="reqn">r</code> in one unit of sequencing effort to approximate the
ratio. The amount of sequences in one unit of sequencing effort is defined by
the variable <code>bin</code>.
</p>
<p>Note that the benefit-cost ratio is not monotonic. The ratio first increases
and then decrease as the amount of sequencing increase. To predicte the
optimal amount of sequencing, we consider only the areas after the peak, where
the ratio starts to decrease.
</p>


<h3>Value</h3>

<p>A vector of three dimensions. The first coordinate is the optimal amount of
sequencing. The second and the third coordinates are the lower and upper bound
of the confidence interval.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
#library(preseqR)

## import data
# data(SRR611492_5M)
## the optimal amount of sequencing with the benefit-cost ratio greater than 
## 0.05 for r = 4
# preseqR.optimal.sequencing(n=SRR611492_5M, efficiency=0.05, bin=1e8, r=4)
## the optimal amount of sequencing with the benefit-cost ratio greater than 
## 0.05 for r = 10
# preseqR.optimal.sequencing(n=SRR611492_5M, efficiency=0.05, bin=1e8, r=10)
</code></pre>

<hr>
<h2 id='preseqR.rSAC'>
Best practice for <code class="reqn">r</code>-SAC &ndash; a fast version
</h2><span id='topic+preseqR.rSAC'></span>

<h3>Description</h3>

<p><code>preseqR.rSAC</code> predicts the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample based on the initial sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preseqR.rSAC(n, r=1, mt=20, size=SIZE.INIT, mu=MU.INIT)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC_+3A_mt">mt</code></td>
<td>

<p>A positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC_+3A_size">size</code></td>
<td>

<p>A positive double, the initial value of the parameter <code>size</code> in
the negative binomial distribution for the EM algorithm. Default value is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC_+3A_mu">mu</code></td>
<td>

<p>A positive double, the initial value of the parameter <code>mu</code> in the
negative binomial distribution for the EM algorithm. Default value is 0.5.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>preseqR.rSAC</code> combines the nonparametric approach using the rational
function approximation and the parametric approach using the
zero-truncated negative binomial (ZTNB). For a given initial sample, if the sample
is from a heterogeneous population, the function calls
<code><a href="#topic+ds.rSAC">ds.rSAC</a></code>; otherwise it calls <code><a href="#topic+ztnb.rSAC">ztnb.rSAC</a></code>. The degree
of heterogeneity is measured by the coefficient of variation, which is
estimated by the ZTNB approach.
</p>
<p><code>preseqR.rSAC</code> is the fast version of <code><a href="#topic+preseqR.rSAC.bootstrap">preseqR.rSAC.bootstrap</a></code>.
The function does not provide the confidence interval. To obtain the
confidence interval along with the estimates, one should use the function
<code><a href="#topic+preseqR.rSAC.bootstrap">preseqR.rSAC.bootstrap</a></code>.
</p>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for SAC
estimator1 &lt;- preseqR.rSAC(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
estimator1(c(10, 20))

## construct the estimator for r-SAC
estimator2 &lt;- preseqR.rSAC(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
estimator2(c(50, 100))
</code></pre>

<hr>
<h2 id='preseqR.rSAC.bootstrap'>
Best practice for <code class="reqn">r</code>-SAC
</h2><span id='topic+preseqR.rSAC.bootstrap'></span>

<h3>Description</h3>

<p><code class="reqn">preseqR.rSAC.bootstrap</code> predicts the expected number of species 
represented at least <code class="reqn">r</code> times in a random sample based on the initial sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preseqR.rSAC.bootstrap(n, r=1, mt=20, size=SIZE.INIT, mu=MU.INIT, times=30,
                       conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_times">times</code></td>
<td>

<p>The number of bootstrap samples.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_size">size</code></td>
<td>

<p>A positive double, the initial value of the parameter <code>size</code> in
the negative binomial distribution for the EM algorithm. Default value is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_mu">mu</code></td>
<td>

<p>A positive double, the initial value of the parameter <code>mu</code> in the
negative binomial distribution for the EM algorithm. Default value is 0.5.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.bootstrap_+3A_conf">conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the bootstrap version of <code><a href="#topic+preseqR.rSAC">preseqR.rSAC</a></code>. The bootstrap
sample is generated by randomly sampling the initial sample with replacement.
For each bootstrap sample, we construct an estimator. The median of
estimates is used as the prediction for the number of species
represented at least <code class="reqn">r</code> times in a random sample. 
</p>
<p>The confidence interval is constructed based on a lognormal distribution.
</p>


<h3>Value</h3>

<table>
<tr><td><code>f</code></td>
<td>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>The standard error for the estimator. The input is a vector of sampling 
efforts t.
</p>
</td></tr>
<tr><td><code>lb</code></td>
<td>

<p>The lower bound of the confidence interval.The input is a vector of sampling 
efforts t.
</p>
</td></tr>
<tr><td><code>ub</code></td>
<td>

<p>The upper bound of the confidence interval.The input is a vector of sampling 
efforts t.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Efron, B., &amp; Tibshirani, R. J. (1994). An introduction to the bootstrap. CRC press.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
# library(preseqR)

## import data
# data(FisherButterfly)

## construct estimator for SAC
# estimator1 &lt;- preseqR.rSAC.bootstrap(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
# estimator1$f(c(10, 20))
## The standard error of the estiamtes
# estimator1$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- estimator1$lb(c(10, 20))
# ub &lt;- estimator1$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

## construct estimator for r-SAC
# estimator2 &lt;- preseqR.rSAC.bootstrap(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
# estimator2$f(c(50, 100))
## The standard error of the estiamtes
# estimator2$se(c(50, 100))
## The confidence interval of the estimates
# lb &lt;- estimator2$lb(c(50, 100))
# ub &lt;- estimator2$ub(c(50, 100))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)
</code></pre>

<hr>
<h2 id='preseqR.rSAC.sequencing.rmdup'>
Predicting <code class="reqn">r</code>-SAC in WES/WGS
</h2><span id='topic+preseqR.rSAC.sequencing.rmdup'></span>

<h3>Description</h3>

<p><code>preseqR.rSAC.sequencing.rmdup</code> predicts the expected number of 
nucleotides in the genome sequenced at least r times in a sequencing
experiment, based on a shallow sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preseqR.rSAC.sequencing.rmdup(n_base, n_read, r=1, mt=20, times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.rSAC.sequencing.rmdup_+3A_n_base">n_base</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of nucleotides in the genome sequenced exactly 
<code class="reqn">j</code> times in the initial experiment. The first column must be sorted in
an ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.sequencing.rmdup_+3A_n_read">n_read</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N'_j</code>, the number of distinct reads with exactly <code class="reqn">j</code> duplicates
in the initial experiment. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.sequencing.rmdup_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.sequencing.rmdup_+3A_mt">mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.sequencing.rmdup_+3A_times">times</code></td>
<td>

<p>The number of bootstrap samples. Default is 30.
</p>
</td></tr>
<tr><td><code id="preseqR.rSAC.sequencing.rmdup_+3A_conf">conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>preseqR.rSAC.sequencing.rmdup</code> is designed for sequencing experiments, 
where duplicate reads are removed. The procedure is commonly used in 
whole-exome sequencing experiments and sometimes appeared in WGS as well.
To use the function, one must have two histograms. The first histogram
is the coverage histogram, which is based on distinct reads.
The second histogram is the counts of reads with exactly <code class="reqn">j</code> duplicates.
</p>


<h3>Value</h3>

<table>
<tr><td><code>f</code></td>
<td>

<p>The estimator for the expected number of nucleotides in the genome
sequenced at least <code class="reqn">r</code> times given the amount of sequencing.
The input of the estimator is a vector of sequencing efforts t, i.e. 
the relative amount of sequencing comparing with the amount in the initial
experiment. For example, t = 2 means sequencing twice the amount of the initial
experiment.
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>The standard error for the estimator. The input is a vector of sequencing
efforts t.
</p>
</td></tr>
<tr><td><code>lb</code></td>
<td>

<p>The lower bound of the confidence interval.The input is a vector of
sequencing efforts t.
</p>
</td></tr>
<tr><td><code>ub</code></td>
<td>

<p>The upper bound of the confidence interval.The input is a vector of
sequencing efforts t.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
#library(preseqR)

## import data
# data(SRR1301329_1M_base)
# data(SRR1301329_1M_read)
# construct the estimator
# estimator1 &lt;- preseqR.rSAC.sequencing.rmdup(
#                 n_base=SRR1301329_1M_base, n_read=SRR5365359_5M_read, 
#                 r=4, mt=20, times=100, conf=0.95)
## The number of nucleotides in the genome covered at least 4 times, when the
## amount of sequencing is 10 or 20 times of the intial experiment
## 10 or 20 times of the initial sample
# estimator1$f(c(10, 20))
## The standard error of the estiamtes
# estimator1$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- estimator1$lb(c(10, 20))
# ub &lt;- estimator1$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

# construct the estimator
# estimator2 &lt;- preseqR.rSAC.sequencing.rmdup(
#                 n_base=SRR1301329_1M_base, n_read=SRR5365359_5M_read, 
#                 r=10, mt=20, times=100, conf=0.95)
## The number of nucleotides in the genome covered at least 10 times, when the
## amount of sequencing is 10 or 20 times of the intial experiment
## 10 or 20 times of the initial sample
# estimator2$f(c(10, 20))
## The standard error of the estiamtes
# estimator2$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- estimator2$lb(c(10, 20))
# ub &lt;- estimator2$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

</code></pre>

<hr>
<h2 id='preseqR.sample.cov'>
Predicting generalized sample coverage
</h2><span id='topic+preseqR.sample.cov'></span>

<h3>Description</h3>

<p><code>preseqR.sample.cov</code> predicts the probability of observing a species
represented at least <code class="reqn">r</code> times in a random sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  preseqR.sample.cov(n, r=1, mt=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.sample.cov_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.sample.cov_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.sample.cov_+3A_mt">mt</code></td>
<td>

<p>A positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose a sample is given and one more individual is randomly drawn from the
population. <code>preseqR.sample.cov</code> estimates the probability of the
species, which represents the individual, has been observed at least
<code class="reqn">r</code> times in the
sample. When <code class="reqn">r = 1</code>, the probability is called the sample coverage.
</p>
<p>Let <code class="reqn">N_j</code> be the number of species represented exactly <code class="reqn">j</code> times in 
a sample. The probability of observing a species represented at
least <code class="reqn">r</code> times in the sample is estimated as 
<code class="reqn">\sum_{j=r+1}^\infty jN_j / \sum_{j=1}^\infty jN_j</code>. The theory is
described by Mao and Lindsay (2002). For a random sample
where <code class="reqn">N_j</code> is unknown, a modified rational function approximation is
first used to predict the value of <code class="reqn">N_j</code>. Then the estimates are
substituted to obtain an estimator for the probability of observing a species
represented at least <code class="reqn">r</code> times in the sample.
</p>
<p>This function is the fast version of <code><a href="#topic+preseqR.sample.cov.bootstrap">preseqR.sample.cov.bootstrap</a></code>.
The function does not provide the confidence interval. To obtain the
confidence interval along with the estimates, one should use the function
<code><a href="#topic+preseqR.sample.cov.bootstrap">preseqR.sample.cov.bootstrap</a></code>.
</p>


<h3>Value</h3>

<p>The estimator for the probability of observing a species represented at least
<code class="reqn">r</code> times in a random sample.
The input of the estimator is a vector of sampling efforts <code class="reqn">t</code>, i.e.,
the relative sample sizes comparing with the initial sample. 
For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Good, I. J. (1953). The population frequencies of species and the estimation of
population parameters. Biometrika, 40(3-4), 237-264.
</p>
<p>Mao, C. X. and Lindsay, B. G. (2002). A Poisson model for the coverage problem
with a genomic application. Biometrika, 89(3), 669-682.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for the sample coverage
estimator1 &lt;- preseqR.sample.cov(FisherButterfly, r=1) 
## Given a sample that is 10 times or 20 times the size of an initial samples,
## suppose one randomly draws one more individual from the population. The
## value of the function is the probability that the representing species 
## has been observed in the sample
estimator1(c(10, 20))

## construct the estimator
estimator2 &lt;- preseqR.sample.cov(FisherButterfly, r=2)
## the probability a species represented at least twice when the sample size
## is 50 times or 100 times of the initial sample
estimator2(c(50, 100))
</code></pre>

<hr>
<h2 id='preseqR.sample.cov.bootstrap'>
Predicting generalized sample coverage with bootstrap
</h2><span id='topic+preseqR.sample.cov.bootstrap'></span>

<h3>Description</h3>

<p><code>preseqR.sample.cov.bootstrap</code> predicts the probability of observing a species
represented at least <code class="reqn">r</code> times in a random sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  preseqR.sample.cov.bootstrap(n, r=1, mt=20, times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.sample.cov.bootstrap_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.sample.cov.bootstrap_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.sample.cov.bootstrap_+3A_mt">mt</code></td>
<td>

<p>A positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td></tr>
<tr><td><code id="preseqR.sample.cov.bootstrap_+3A_times">times</code></td>
<td>

<p>The number of bootstrap samples. Default is 30.
</p>
</td></tr>
<tr><td><code id="preseqR.sample.cov.bootstrap_+3A_conf">conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the bootstrap version of <code><a href="#topic+preseqR.sample.cov">preseqR.sample.cov</a></code>. The bootstrap
sample is generated by randomly sampling the initial sample with replacement.
For each bootstrap sample, we construct an estimator. The median of
estimates is used as the prediction for the number of species
represented at least <code class="reqn">r</code> times in a random sample. 
</p>
<p>The confidence interval is constructed based on a lognormal distribution.
</p>


<h3>Value</h3>

<table>
<tr><td><code>f</code></td>
<td>

<p>The estimator for the probability of observing a species represented at least
<code class="reqn">r</code> times in a sample as a function of the sample size.
The input of the estimator is a vector of sampling 
efforts t, i.e. the relative sample sizes comparing with the initial sample.
For example, t = 2 means a random sample that is twice the size of the 
initial sample.
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>The standard error for the estimator. The input is a vector of sampling 
efforts t.
</p>
</td></tr>
<tr><td><code>lb</code></td>
<td>

<p>The lower bound of the confidence interval.The input is a vector of sampling 
efforts t.
</p>
</td></tr>
<tr><td><code>ub</code></td>
<td>

<p>The upper bound of the confidence interval.The input is a vector of sampling 
efforts t.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Efron, B., &amp; Tibshirani, R. J. (1994). An introduction to the bootstrap. CRC press.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
#library(preseqR)

## import data
#data(FisherButterfly)

## construct the estimator for the sample coverage
# estimator1 &lt;- preseqR.sample.cov.bootstrap(FisherButterfly, r=1)
## Given a sample that is 10 times or 20 times the size of an initial samples,
## suppose one randomly draws one more individual from the population. The
## value of the function is the probability that the representing species 
## has been observed in the sample
# estimator1$f(c(10, 20))
## The standard error of the estiamtes
# estimator1$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- estimator1$lb(c(10, 20))
# ub &lt;- estimator1$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

## construct the estimator
# estimator2 &lt;- preseqR.rSAC.bootstrap(FisherButterfly, r=2)
## the probability when the sample size is 50 times or 100 times of the initial
## sample
# estimator2$f(c(50, 100))
## The standard error of the estiamtes
# estimator2$se(c(50, 100))
## The confidence interval of the estimates
# lb &lt;- estimator2$lb(c(50, 100))
# ub &lt;- estimator2$ub(c(50, 100))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)
</code></pre>

<hr>
<h2 id='preseqR.simu.hist'>
Simulation
</h2><span id='topic+preseqR.simu.hist'></span>

<h3>Description</h3>

<p>Generating a histogram based on a Poisson mixture model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  preseqR.simu.hist(L=1e8, N, FUN)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.simu.hist_+3A_l">L</code></td>
<td>

<p>A positive integer, the number of species in a population.
</p>
</td></tr>
<tr><td><code id="preseqR.simu.hist_+3A_n">N</code></td>
<td>

<p>A positive interger, the simulated sample size.
</p>
</td></tr>
<tr><td><code id="preseqR.simu.hist_+3A_fun">FUN</code></td>
<td>

<p>An RNG generating non negative real number.
</p>
</td></tr> 
</table>


<h3>Details</h3>

<p><code>preseqR.simu.hist</code> uses a mixture of Poisson distributions to generate
a sample, which size is defined by the variable <code class="reqn">N</code>. 
The statistical assumption is that for each species the number of individuals
captured in a sample follows a Poisson process. 
The Poisson rates among species are generated by a given
function <code>FUN</code> per unit of sampling effort.
</p>
<p><code>FUN</code> must take an argument indicating the number of random
numbers generated and return a vector of generated numbers. 
</p>


<h3>Value</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)
## construct a RNG
f &lt;- function(n) {
  rgamma(n, shape=0.5, scale=1)
}
## sample 10,000 individuals
preseqR.simu.hist(L=1e5, N=10000, f)
</code></pre>

<hr>
<h2 id='preseqR.ztnb.em'>
Fitting a zero-truncated negative binomial distribution
</h2><span id='topic+preseqR.ztnb.em'></span>

<h3>Description</h3>

<p><code>preseqR.ztnb.em</code> fits a zero-truncated negative binomial (ZTNB)
distribution to the initial sample.
Since the species with zero observations are missed in the sample, an
EM algorithm is used to estimate the parameters assuming the number of
individuals for each species follows a Negative Binomial distribution
with the zero counts as a missing latent data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preseqR.ztnb.em(n, size = SIZE.INIT, mu = MU.INIT)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preseqR.ztnb.em_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="preseqR.ztnb.em_+3A_size">size</code></td>
<td>

<p>A positive double setting the initial value of the parameter <code>size</code> in
a negative binomial distribution for the EM algorithm. Default value is 1.
</p>
</td></tr>
<tr><td><code id="preseqR.ztnb.em_+3A_mu">mu</code></td>
<td>

<p>A positive double setting the initial value of the parameter <code>mu</code> in a
negative binomial distribution for the EM algorithm. Default value is 0.5.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the supplement of Daley and Smith (2013).
</p>


<h3>Value</h3>

<table>
<tr><td><code>size</code></td>
<td>

<p>The estimate of the parameter <code>size</code> in the negative binomial.
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>

<p>The estimate of the parameter <code>mu</code> in the negative binomial.
</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>

<p>Log-likelihood under estimated ZTNB.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## print the parameters of a fitting negative binomial distribution
preseqR.ztnb.em(FisherButterfly)
</code></pre>

<hr>
<h2 id='Shakespeare'>Shakespeare's word type frequencies</h2><span id='topic+Shakespeare'></span>

<h3>Description</h3>

<p>The Shakespeare's word type frequencies data was from 
Efron, B., &amp; Thisted, R. (1976).</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">n_j</code>, the number of unique words appeared <code class="reqn">j</code>
times in Shakespeare's work.
</p>


<h3>References</h3>

<p>Efron, B., &amp; Thisted, R. (1976). Estimating the number of unseen species:
How many words did Shakespeare know?. Biometrika, 63(3), 435-447.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(Shakespeare)
</code></pre>

<hr>
<h2 id='SRR061157_k31'><code class="reqn">k</code>-mer counts of a metagenomic data</h2><span id='topic+SRR061157_k31'></span>

<h3>Description</h3>

<p>The <code class="reqn">k</code>-mer counts are based on a metagenome sequencing data from
Human Microbiome Project with the accession number
SRR061157. Only forward reads are used to generate the <code class="reqn">k</code>-mer counts.</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of 31-mers observed exactly <code class="reqn">j</code> times.
</p>


<h3>References</h3>

<p>Human Microbiome Project (<a href="https://hmpdacc.org/">https://hmpdacc.org/</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR061157_k31)
</code></pre>

<hr>
<h2 id='SRR1301329_1M_base'>Coverage histogram of a WES data</h2><span id='topic+SRR1301329_1M_base'></span>

<h3>Description</h3>

<p>The coverage histogram is based on a whole-exome sequencing (WES) 
data from Simons Foundation Autism Research Initiative with the accession number
SRR1301329. One million reads are randomly sampled from the raw data to
generate this coverage histogram.</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of nucleotides in the genome covered exactly
<code class="reqn">j</code> times.
</p>


<h3>References</h3>

<p>Simons Foundation Autism Research Initiative (<a href="https://www.sfari.org/">https://www.sfari.org/</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR1301329_1M_base)
</code></pre>

<hr>
<h2 id='SRR1301329_1M_read'>Read counts of a WES data</h2><span id='topic+SRR1301329_1M_read'></span>

<h3>Description</h3>

<p>The read counts are based on a whole-exome sequencing (WES) data
from Simons Foundation Autism Research Initiative with the accession number
SRR1301329. One million reads are randomly sampled from the raw data to
generate the read counts.</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of reads observed exactly <code class="reqn">j</code> times in the
data.
</p>


<h3>References</h3>

<p>Simons Foundation Autism Research Initiative (<a href="https://www.sfari.org/">https://www.sfari.org/</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR1301329_1M_read)
</code></pre>

<hr>
<h2 id='SRR1301329_base'>Coverage histogram of a WES data</h2><span id='topic+SRR1301329_base'></span>

<h3>Description</h3>

<p>The coverage histogram is based on a whole-exome sequencing (WES) 
data from Simons Foundation Autism Research Initiative with the accession number
SRR1301329. Only forward reads are used to generate the coverage histogram.
</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of nucleotides in the genome covered exactly
<code class="reqn">j</code> times.
</p>


<h3>References</h3>

<p>Simons Foundation Autism Research Initiative (<a href="https://www.sfari.org/">https://www.sfari.org/</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR1301329_base)
</code></pre>

<hr>
<h2 id='SRR1301329_read'>Read counts of a WES data</h2><span id='topic+SRR1301329_read'></span>

<h3>Description</h3>

<p>The read counts are based on a whole-exome sequencing data from
Simons Foundation Autism Research Initiative with the accession number
SRR1301329. Only forward reads are used to generate the read counts.</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of reads observed exactly <code class="reqn">j</code> times in the
data.
</p>


<h3>References</h3>

<p>Simons Foundation Autism Research Initiative (<a href="https://www.sfari.org/">https://www.sfari.org/</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR1301329_read)
</code></pre>

<hr>
<h2 id='SRR611492'>Coverage histogram of a scWGS data</h2><span id='topic+SRR611492'></span>

<h3>Description</h3>

<p>The coverage histogram is based on a single-cell whole-genome
sequencing data (scWGS) through MALBAK protocol. The accession number of the raw
data is SRR1301329. Only forward reads are used to generate the coverage
histogram.
</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of nucleotides in the genome covered exactly
<code class="reqn">j</code> times.
</p>


<h3>References</h3>

<p>Zong, C., Lu, S., Chapman, A. R., &amp; Xie, X. S. (2012). Genome-wide detection
of single-nucleotide and copy-number variations of a single human cell.
Science, 338(6114), 1622-1626.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR611492)
</code></pre>

<hr>
<h2 id='SRR611492_5M'>Coverage histogram of a scWGS data</h2><span id='topic+SRR611492_5M'></span>

<h3>Description</h3>

<p>The coverage histogram is based on a single-cell whole-genome
sequencing (scWGS) data through MALBAK protocol. The accession number of the
raw data is SRR1301329. Five million reads are randomly sampled from the raw
data to generate this coverage histogram.
</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of nucleotides in the genome covered exactly
<code class="reqn">j</code> times.
</p>


<h3>References</h3>

<p>Zong, C., Lu, S., Chapman, A. R., &amp; Xie, X. S. (2012). Genome-wide detection
of single-nucleotide and copy-number variations of a single human cell.
Science, 338(6114), 1622-1626.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(SRR1301329_5M)
</code></pre>

<hr>
<h2 id='Twitter'>Social network</h2><span id='topic+Twitter'></span>

<h3>Description</h3>

<p>Following relationships of Twitter's social network</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">n_j</code>, the number of users with exactly <code class="reqn">j</code> followers.
</p>


<h3>References</h3>

<p>Zafarani R, Liu H (2009) Social computing data repository at ASU.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(Twitter)
</code></pre>

<hr>
<h2 id='WillButterfly'>Fisher's butterfly data</h2><span id='topic+WillButterfly'></span>

<h3>Description</h3>

<p>Frequencies data of butterflies collected in the Malay peninsula
was from Fisher, R. A., Corbet, A. S., &amp; Williams, C. B. (1943).</p>


<h3>Details</h3>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">n_j</code>, the number of butterflies captured exactly <code class="reqn">j</code>
times in the sample.
</p>


<h3>References</h3>

<p>Fisher, R. A., Corbet, A. S., and Williams, C. B. ,1943, The Relation Between 
the Number of Species and the Number of Individuals in a Random Sample of an 
Animal Population, Journal of Animal Ecology, 12, 42-58, Table 3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load library
library(preseqR)

##load data
data(WillButterfly)
</code></pre>

<hr>
<h2 id='ztnb.rSAC'>
ZTNB estimator
</h2><span id='topic+ztnb.rSAC'></span>

<h3>Description</h3>

<p><code>ztnb.rSAC</code> predicts the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample, based on the initial sample. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ztnb.rSAC(n, r=1, size=SIZE.INIT, mu=MU.INIT)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ztnb.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="ztnb.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
<tr><td><code id="ztnb.rSAC_+3A_size">size</code></td>
<td>

<p>A positive double, the initial value of the parameter <code>size</code> in
the negative binomial distribution for the EM algorithm. Default value is 1.
</p>
</td></tr>
<tr><td><code id="ztnb.rSAC_+3A_mu">mu</code></td>
<td>

<p>A positive double, the initial value of the parameter <code>mu</code> in the
negative binomial distribution for the EM algorithm. Default value is 0.5.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The statistical assumption is that for each species the number of individuals
in a sample follows a Poisson distribution. The Poisson rate <code>lambda</code>
are numbers generated from a gamma distribution. So the random variable
<code>X</code>, which is the number of species represented x (x &gt; 0) times in the
sample, follows a zero-truncated negative binomial distribution. The
unknown parameters are estimated by the function <code>preseqR.ztnb.em</code> based
on the initial sample. Using the estimated distribution, we calculate the 
expected number of species represented at least r times in a random sample.
Details of the estimation procedure can be found in the supplement of 
Daley T. and Smith AD. (2013).
</p>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Daley, T., &amp; Smith, A. D. (2013). Predicting the molecular complexity of
sequencing libraries. Nature methods, 10(4), 325-327.
</p>
<p>Deng C, Daley T &amp; Smith AD (2015). Applications of species accumulation curves
in large-scale biological data analysis. Quantitative Biology, 3(3), 135-144. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+preseqR.ztnb.em">preseqR.ztnb.em</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for SAC
ztnb1 &lt;- ztnb.rSAC(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
ztnb1(c(10, 20))

## construct the estimator for r-SAC
ztnb2 &lt;- ztnb.rSAC(FisherButterfly, r=2)
## The number of species represented at least twice in a sample, 
## when the sample size is 50 or 100 times of the initial sample
ztnb2(c(50, 100))
</code></pre>

<hr>
<h2 id='ztp.rSAC'>
ZTP estimator
</h2><span id='topic+ztp.rSAC'></span>

<h3>Description</h3>

<p><code>ztp.rSAC</code> predicts the expected number of species represented at least 
<code class="reqn">r</code> times in a random sample, based on the initial sample. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ztp.rSAC(n, r=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ztp.rSAC_+3A_n">n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td></tr>
<tr><td><code id="ztp.rSAC_+3A_r">r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The statistical assumption is that for each species the number of individuals
in a sample follows a Poisson distribution. The Poisson rate <code>lambda</code>
is the same among all species. So the random variable <code>X</code>, which is 
the number of species represented x (x &gt; 0) times, follows a zero-truncated 
Poisson distribution. The unknown parameters are estimated by 
Cohen (1960). Based on the estimated distribution,
we calculate the expected number of species in a random sample. 
</p>


<h3>Value</h3>

<p>The estimator for the <code class="reqn">r</code>-SAC. The input of the estimator is a vector of
sampling efforts <code class="reqn">t</code>, i.e., the relative sample sizes comparing with the initial
sample. For example, <code class="reqn">t = 2</code> means a random sample that is twice the size of
the initial sample.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Cohen, A. Clifford. &quot;Estimating the parameter in a conditional Poisson 
distribution.&quot; Biometrics 16, no. 2 (1960): 203-211.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load library
library(preseqR)

## import data
data(FisherButterfly)

## construct the estimator for SAC
ztp1 &lt;- ztp.rSAC(FisherButterfly, r=1)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
ztp1(c(10, 20))

## construct the estimator for r-SAC
ztp2 &lt;- ztp.rSAC(FisherButterfly, r=2)
## The number of species represented at least once in a sample, 
## when the sample size is 10 or 20 times of the initial sample
ztp2(c(50, 100))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
