<!DOCTYPE html><html lang="en"><head><title>Help for package primerTree</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {primerTree}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#accession2taxid'><p>Maps a nucleotide database accession to a taxonomy database taxId</p></a></li>
<li><a href='#bryophytes_trnL'><p>PrimerTree results for the bryophyte trnL primers</p></a></li>
<li><a href='#calc_rank_dist_ave'><p>Summarize pairwise differences.</p></a></li>
<li><a href='#clustalo'><p>Multiple sequence alignment with clustal omega</p></a></li>
<li><a href='#filter_seqs'><p>Filter out sequences retrieved by search_primer_pair() that are either too</p>
short or too long. The alignment and tree will be recalculated after removing
unwanted reads.</a></li>
<li><a href='#get_sequence'><p>Retrieves a fasta sequence from NCBI nucleotide database.</p></a></li>
<li><a href='#get_sequences'><p>Retrieves fasta sequences from NCBI nucleotide database.</p></a></li>
<li><a href='#get_taxonomy'><p>Retrieve the taxonomy information from NCBI for a set of nucleotide gis.</p></a></li>
<li><a href='#identify.primerTree_plot'><p>identify the point closest to the mouse click</p>
only works on single ranks</a></li>
<li><a href='#layout_tree_ape'><p>layout a tree using ape, return an object to be plotted by</p>
<code>plot_tree</code></a></li>
<li><a href='#mammals_16S'><p>PrimerTree results for the mammalian 16S primers</p></a></li>
<li><a href='#parse_primer_hits'><p>Parse the primer hits</p></a></li>
<li><a href='#plot_tree'><p>plots a tree, optionally with colored and labeled points by taxonomic rank</p></a></li>
<li><a href='#plot_tree_ranks'><p>plots a tree along with a series of taxonomic ranks</p></a></li>
<li><a href='#plot.primerTree'><p>plot function for a primerTree object, calls plot_tree_ranks</p></a></li>
<li><a href='#primer_search'><p>Query a pair of primers using ncbi's Primer-BLAST, if primers contain iupac</p></a></li>
<li><a href='#primerTree'><p><span class="pkg">primerTree</span> Visually Assessing the Specificity and Informativeness of Primer Pairs</p></a></li>
<li><a href='#search_primer_pair'><p>Automatic primer searching Search a given primer pair, retrieving the alignment</p>
results, their product sequences, the taxonomic information for the sequences,
a multiple alignment of the products</a></li>
<li><a href='#seq_lengths'><p>Get a summary of sequence lengths from a primerTree object</p></a></li>
<li><a href='#seq_lengths.primerTree'><p>Method for primerTree objects</p></a></li>
<li><a href='#summary.primerTree'><p>Summarize a primerTree result, printing quantiles of sequence length and</p>
pairwise differences.</a></li>
<li><a href='#tree_from_alignment'><p>Construct a neighbor joining tree from a dna alignment</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Visually Assessing the Specificity and Informativeness of Primer
Pairs</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.6</td>
</tr>
<tr>
<td>Description:</td>
<td>Identifies potential target sequences for a given
    set of primers and generates phylogenetic trees annotated with the
    taxonomies of the predicted amplification products.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), directlabels, gridExtra</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, foreach, ggplot2, grid, httr, lubridate, plyr, reshape2,
scales, stringr, XML</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-04 23:28:09 UTC; MattDellLaptop</td>
</tr>
<tr>
<td>Author:</td>
<td>Jim Hester [aut],
  Matt Cannon [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matt Cannon &lt;matthewvc1@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-05 14:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='accession2taxid'>Maps a nucleotide database accession to a taxonomy database taxId</h2><span id='topic+accession2taxid'></span>

<h3>Description</h3>

<p>Maps a nucleotide database accession to a taxonomy database taxId
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accession2taxid(accessions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="accession2taxid_+3A_accessions">accessions</code></td>
<td>
<p>accessions character vector to lookup.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named vector of taxIds.
</p>

<hr>
<h2 id='bryophytes_trnL'>PrimerTree results for the bryophyte trnL primers</h2><span id='topic+bryophytes_trnL'></span>

<h3>Description</h3>

<p>PrimerTree results for the bryophyte trnL primers
</p>

<hr>
<h2 id='calc_rank_dist_ave'>Summarize pairwise differences.</h2><span id='topic+calc_rank_dist_ave'></span>

<h3>Description</h3>

<p>Summarize pairwise differences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_rank_dist_ave(x, ranks = common_ranks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_rank_dist_ave_+3A_x">x</code></td>
<td>
<p>a primerTree object</p>
</td></tr>
<tr><td><code id="calc_rank_dist_ave_+3A_ranks">ranks</code></td>
<td>
<p>ranks to show unique counts for, defaults to the common ranks</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The purpose of this function is to calculate the average number
of nucleotide differences between species within each taxa of given taxonomic
level.
</p>
<p>For example, at the genus level, the function calculates the average number
of nucleotide differences between all species within each genus and reports
the mean of those values.
</p>
<p>There are several key assumptions and calculations made in this
function.
</p>
<p>First, the function randomly selects one sequence from each species
in the primerTree results. This is to keep any one species (e.g.
human, cow, etc.) with many hits from skewing the results.
</p>
<p>Second, for each taxonomic level tested, the function divides the
sequences by each taxon at that level and calculates the mean
number of nucleotide differences within that taxa, then returns the
mean of those values.
</p>
<p>Third, when calculating the average distance, any taxa for which
there is only one species is omitted, as the number of nucleotide
differences will always be 0.
</p>


<h3>Value</h3>

<p>returns a data frame of results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
calc_rank_dist_ave(mammals_16S)

calc_rank_dist_ave(bryophytes_trnL)

# Note that the differences between the results from these two primers
# the mean nucleotide differences is much higher for the mammal primers
# than the byrophyte primers. This suggests that the mammal primers have
# better resolution to distinguish individual species.

## End(Not run)
</code></pre>

<hr>
<h2 id='clustalo'>Multiple sequence alignment with clustal omega</h2><span id='topic+clustalo'></span>

<h3>Description</h3>

<p>Calls clustal omega to align a set of sequences of class DNAbin.  Run
without any arguments to see all the options you can pass to the command
line clustal omega.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustalo(x, exec = "clustalo", quiet = TRUE, original.ordering = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clustalo_+3A_x">x</code></td>
<td>
<p>an object of class 'DNAbin'</p>
</td></tr>
<tr><td><code id="clustalo_+3A_exec">exec</code></td>
<td>
<p>a character string with the name or path to the program</p>
</td></tr>
<tr><td><code id="clustalo_+3A_quiet">quiet</code></td>
<td>
<p>whether to supress output to stderr or stdout</p>
</td></tr>
<tr><td><code id="clustalo_+3A_original.ordering">original.ordering</code></td>
<td>
<p>use the original ordering of the sequences</p>
</td></tr>
<tr><td><code id="clustalo_+3A_...">...</code></td>
<td>
<p>additional arguments passed to the command line clustalo</p>
</td></tr>
</table>

<hr>
<h2 id='filter_seqs'>Filter out sequences retrieved by search_primer_pair() that are either too
short or too long. The alignment and tree will be recalculated after removing
unwanted reads.</h2><span id='topic+filter_seqs'></span><span id='topic+filter_seqs.primerTree'></span>

<h3>Description</h3>

<p>Filter out sequences retrieved by search_primer_pair() that are either too
short or too long. The alignment and tree will be recalculated after removing
unwanted reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_seqs(x, ...)

## S3 method for class 'primerTree'
filter_seqs(x, min_length = 0, max_length = Inf, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_seqs_+3A_x">x</code></td>
<td>
<p>a primerTree object</p>
</td></tr>
<tr><td><code id="filter_seqs_+3A_...">...</code></td>
<td>
<p>additional arguments passed to methods.</p>
</td></tr>
<tr><td><code id="filter_seqs_+3A_min_length">min_length</code></td>
<td>
<p>the minimum sequence length to keep</p>
</td></tr>
<tr><td><code id="filter_seqs_+3A_max_length">max_length</code></td>
<td>
<p>the maximum sequence length to keep</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a primerTree object
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>primerTree</code>: Method for primerTree objects
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# filter out sequences longer or shorter than desired:
mammals_16S_filtered &lt;- filter_seqs(mammals_16S, min_length=131, max_length=156)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_sequence'>Retrieves a fasta sequence from NCBI nucleotide database.</h2><span id='topic+get_sequence'></span>

<h3>Description</h3>

<p>Retrieves a fasta sequence from NCBI nucleotide database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sequence(
  accession,
  start = NULL,
  stop = NULL,
  api_key = Sys.getenv("NCBI_API_KEY")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sequence_+3A_accession">accession</code></td>
<td>
<p>nucleotide accession to retrieve.</p>
</td></tr>
<tr><td><code id="get_sequence_+3A_start">start</code></td>
<td>
<p>start base to retrieve, numbered beginning at 1.  If NULL the
beginning of the sequence.</p>
</td></tr>
<tr><td><code id="get_sequence_+3A_stop">stop</code></td>
<td>
<p>last base to retrieve, numbered beginning at 1. if NULL the end of
the sequence.</p>
</td></tr>
<tr><td><code id="get_sequence_+3A_api_key">api_key</code></td>
<td>
<p>NCBI api-key to allow faster sequence retrieval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an DNAbin object.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+DNAbin">DNAbin</a></code>
</p>

<hr>
<h2 id='get_sequences'>Retrieves fasta sequences from NCBI nucleotide database.</h2><span id='topic+get_sequences'></span>

<h3>Description</h3>

<p>Retrieves fasta sequences from NCBI nucleotide database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sequences(
  accession,
  start = NULL,
  stop = NULL,
  api_key = Sys.getenv("NCBI_API_KEY"),
  simplify = TRUE,
  .parallel = FALSE,
  .progress = "none"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sequences_+3A_accession">accession</code></td>
<td>
<p>the accession number of the sequence to retrieve</p>
</td></tr>
<tr><td><code id="get_sequences_+3A_start">start</code></td>
<td>
<p>start bases to retrieve, numbered beginning at 1.  If NULL the
beginning of the sequence.</p>
</td></tr>
<tr><td><code id="get_sequences_+3A_stop">stop</code></td>
<td>
<p>stop bases to retrieve, numbered beginning at 1. if NULL the stop of
the sequence.</p>
</td></tr>
<tr><td><code id="get_sequences_+3A_api_key">api_key</code></td>
<td>
<p>NCBI api-key to allow faster sequence retrieval.</p>
</td></tr>
<tr><td><code id="get_sequences_+3A_simplify">simplify</code></td>
<td>
<p>simplify the FASTA headers to include only the genbank
accession.</p>
</td></tr>
<tr><td><code id="get_sequences_+3A_.parallel">.parallel</code></td>
<td>
<p>if 'TRUE', perform in parallel, using parallel backend
provided by foreach</p>
</td></tr>
<tr><td><code id="get_sequences_+3A_.progress">.progress</code></td>
<td>
<p>name of the progress bar to use, see 'create_progress_bar'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an DNAbin object.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+DNAbin">DNAbin</a></code>
</p>

<hr>
<h2 id='get_taxonomy'>Retrieve the taxonomy information from NCBI for a set of nucleotide gis.</h2><span id='topic+get_taxonomy'></span>

<h3>Description</h3>

<p>Retrieve the taxonomy information from NCBI for a set of nucleotide gis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_taxonomy(accessions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_taxonomy_+3A_accessions">accessions</code></td>
<td>
<p>a character vector of the accessions to retrieve</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame of the 'accessions, taxIds, and taxonomy
</p>

<hr>
<h2 id='identify.primerTree_plot'>identify the point closest to the mouse click
only works on single ranks</h2><span id='topic+identify.primerTree_plot'></span>

<h3>Description</h3>

<p>identify the point closest to the mouse click
only works on single ranks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'primerTree_plot'
identify(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identify.primerTree_plot_+3A_x">x</code></td>
<td>
<p>the plot to identify</p>
</td></tr>
<tr><td><code id="identify.primerTree_plot_+3A_...">...</code></td>
<td>
<p>additional arguments passed to annotate</p>
</td></tr>
</table>

<hr>
<h2 id='layout_tree_ape'>layout a tree using ape, return an object to be plotted by
<code><a href="#topic+plot_tree">plot_tree</a></code></h2><span id='topic+layout_tree_ape'></span>

<h3>Description</h3>

<p>layout a tree using ape, return an object to be plotted by
<code><a href="#topic+plot_tree">plot_tree</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layout_tree_ape(tree, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layout_tree_ape_+3A_tree">tree</code></td>
<td>
<p>The <code><a href="ape.html#topic+phylo">phylo</a></code> tree to be plotted</p>
</td></tr>
<tr><td><code id="layout_tree_ape_+3A_...">...</code></td>
<td>
<p>additional arguments to <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>edge</code></td>
<td>
<p>list of x, y and xend, yend coordinates
as well as ids for the edges</p>
</td></tr>
<tr><td><code>tips</code></td>
<td>
<p>list of x, y, label and id for the tips</p>
</td></tr>
<tr><td><code>nodes</code></td>
<td>
<p>list of x, y and id for the nodes</p>
</td></tr>
</table>

<hr>
<h2 id='mammals_16S'>PrimerTree results for the mammalian 16S primers</h2><span id='topic+mammals_16S'></span>

<h3>Description</h3>

<p>PrimerTree results for the mammalian 16S primers
</p>

<hr>
<h2 id='parse_primer_hits'>Parse the primer hits</h2><span id='topic+parse_primer_hits'></span>

<h3>Description</h3>

<p>Parse the primer hits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_primer_hits(response)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_primer_hits_+3A_response">response</code></td>
<td>
<p>a httr response object obtained from <code><a href="#topic+primer_search">primer_search</a></code></p>
</td></tr>
</table>

<hr>
<h2 id='plot_tree'>plots a tree, optionally with colored and labeled points by taxonomic rank</h2><span id='topic+plot_tree'></span>

<h3>Description</h3>

<p>plots a tree, optionally with colored and labeled points by taxonomic rank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_tree(
  tree,
  type = "unrooted",
  main = NULL,
  guide_size = NULL,
  rank = NULL,
  taxonomy = NULL,
  size = 2,
  legend_cutoff = 25,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_tree_+3A_tree">tree</code></td>
<td>
<p>to be plotted, use layout_tree to layout tree.</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_type">type</code></td>
<td>
<p>The type of tree to plot, default unrooted.</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_main">main</code></td>
<td>
<p>An optional title for the plot</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_guide_size">guide_size</code></td>
<td>
<p>The size of the length guide.  If NULL auto detects a
reasonable size.</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_rank">rank</code></td>
<td>
<p>The rank to include, if null only the tree is plotted</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_taxonomy">taxonomy</code></td>
<td>
<p>A data.frame with an accession field corresponding to the
tree tip labels.</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_size">size</code></td>
<td>
<p>The size of the colored points</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_legend_cutoff">legend_cutoff</code></td>
<td>
<p>The number of different taxa names after which the
names are no longer printed.</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="#topic+layout_tree_ape">layout_tree_ape</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot to be printed.
</p>

<hr>
<h2 id='plot_tree_ranks'>plots a tree along with a series of taxonomic ranks</h2><span id='topic+plot_tree_ranks'></span>

<h3>Description</h3>

<p>plots a tree along with a series of taxonomic ranks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_tree_ranks(
  tree,
  taxonomy,
  main = NULL,
  type = "unrooted",
  ranks = common_ranks,
  size = 2,
  guide_size = NULL,
  legend_cutoff = 25,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_tree_ranks_+3A_tree">tree</code></td>
<td>
<p>to be plotted, use layout_tree to layout tree.</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_taxonomy">taxonomy</code></td>
<td>
<p>A data.frame with an accession field corresponding to the
tree tip labels.</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_main">main</code></td>
<td>
<p>An optional title for the plot</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_type">type</code></td>
<td>
<p>The type of tree to plot, default unrooted.</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_ranks">ranks</code></td>
<td>
<p>The ranks to include, defaults to all common ranks, if null
print all ranks.</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_size">size</code></td>
<td>
<p>The size of the colored points</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_guide_size">guide_size</code></td>
<td>
<p>The size of the length guide.  If NULL auto detects a
reasonable size.</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_legend_cutoff">legend_cutoff</code></td>
<td>
<p>The number of different taxa names after which the
names are no longer printed.</p>
</td></tr>
<tr><td><code id="plot_tree_ranks_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="#topic+layout_tree_ape">layout_tree_ape</a></code></p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+plot_tree">plot_tree</a></code> to plot only a single rank or the just the
tree layout.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(gridExtra)
library(directlabels)
#plot all the common ranks
plot_tree_ranks(mammals_16S$tree, mammals_16S$taxonomy)
#plot specific ranks, with a larger dot size
plot_tree_ranks(mammals_16S$tree, mammals_16S$taxonomy,
  ranks=c('kingdom', 'class', 'family'), size=3)
</code></pre>

<hr>
<h2 id='plot.primerTree'>plot function for a primerTree object, calls plot_tree_ranks</h2><span id='topic+plot.primerTree'></span>

<h3>Description</h3>

<p>plot function for a primerTree object, calls plot_tree_ranks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'primerTree'
plot(x, ranks = NULL, main = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.primerTree_+3A_x">x</code></td>
<td>
<p>primerTree object to plot</p>
</td></tr>
<tr><td><code id="plot.primerTree_+3A_ranks">ranks</code></td>
<td>
<p>The ranks to include, defaults to all common ranks, if NULL
print all ranks.  If 'none' just print the layout.</p>
</td></tr>
<tr><td><code id="plot.primerTree_+3A_main">main</code></td>
<td>
<p>an optional title to display, if NULL displays the name as the title</p>
</td></tr>
<tr><td><code id="plot.primerTree_+3A_...">...</code></td>
<td>
<p>additional arguments passed to plot_tree_ranks</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+plot_tree_ranks">plot_tree_ranks</a></code>, <code><a href="#topic+plot_tree">plot_tree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(gridExtra)
library(directlabels)
#plot with all common ranks
plot(mammals_16S)

#plot only the class
plot(mammals_16S, 'class')

#plot the layout only
plot(mammals_16S, 'none')
</code></pre>

<hr>
<h2 id='primer_search'>Query a pair of primers using ncbi's Primer-BLAST, if primers contain iupac</h2><span id='topic+primer_search'></span>

<h3>Description</h3>

<p>ambiguity codes, enumerate all possible combinations and combine the
results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>primer_search(
  forward,
  reverse,
  num_aligns = 500,
  num_permutations = 25,
  ...,
  .parallel = FALSE,
  .progress = "none"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="primer_search_+3A_forward">forward</code></td>
<td>
<p>forward primer to search by 5'-3' on plus strand</p>
</td></tr>
<tr><td><code id="primer_search_+3A_reverse">reverse</code></td>
<td>
<p>reverse primer to search by 5'-3' on minus strand</p>
</td></tr>
<tr><td><code id="primer_search_+3A_num_aligns">num_aligns</code></td>
<td>
<p>number of alignment results to keep</p>
</td></tr>
<tr><td><code id="primer_search_+3A_num_permutations">num_permutations</code></td>
<td>
<p>the number of primer permutations to search, if the degenerate bases
cause more than this number of permutations to exist, this number will be
sampled from all possible permutations.</p>
</td></tr>
<tr><td><code id="primer_search_+3A_...">...</code></td>
<td>
<p>additional arguments passed to Primer-Blast</p>
</td></tr>
<tr><td><code id="primer_search_+3A_.parallel">.parallel</code></td>
<td>
<p>if 'TRUE', perform in parallel, using parallel backend
provided by foreach</p>
</td></tr>
<tr><td><code id="primer_search_+3A_.progress">.progress</code></td>
<td>
<p>name of the progress bar to use, see 'create_progress_bar'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>httr response object of the query, pass to <code><a href="#topic+parse_primer_hits">parse_primer_hits</a></code> to parse the results.
</p>

<hr>
<h2 id='primerTree'><span class="pkg">primerTree</span> Visually Assessing the Specificity and Informativeness of Primer Pairs</h2><span id='topic+primerTree'></span>

<h3>Description</h3>

<p><code>primerTree</code> has two main commands:
<code><a href="#topic+search_primer_pair">search_primer_pair</a></code> which takes a primer pair and returns an
primerTree object of the search results
<code><a href="#topic+plot.primerTree">plot.primerTree</a></code> a S3 method for plotting the primerTree object
obtained using <code><a href="#topic+search_primer_pair">search_primer_pair</a></code>
</p>

<hr>
<h2 id='search_primer_pair'>Automatic primer searching Search a given primer pair, retrieving the alignment
results, their product sequences, the taxonomic information for the sequences,
a multiple alignment of the products</h2><span id='topic+search_primer_pair'></span>

<h3>Description</h3>

<p>Automatic primer searching Search a given primer pair, retrieving the alignment
results, their product sequences, the taxonomic information for the sequences,
a multiple alignment of the products
</p>


<h3>Usage</h3>

<pre><code class='language-R'>search_primer_pair(
  forward,
  reverse,
  name = NULL,
  num_aligns = 500,
  num_permutations = 25,
  simplify = TRUE,
  clustal_options = list(),
  distance_options = list(model = "N", pairwise.deletion = T),
  api_key = Sys.getenv("NCBI_API_KEY"),
  ...,
  .parallel = FALSE,
  .progress = "none"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="search_primer_pair_+3A_forward">forward</code></td>
<td>
<p>forward primer to search by 5'-3' on plus strand</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_reverse">reverse</code></td>
<td>
<p>reverse primer to search by 5'-3' on minus strand</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_name">name</code></td>
<td>
<p>name to give to the primer pair</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_num_aligns">num_aligns</code></td>
<td>
<p>number of alignment results to keep</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_num_permutations">num_permutations</code></td>
<td>
<p>the number of primer permutations to search, if the degenerate bases
cause more than this number of permutations to exist, this number will be
sampled from all possible permutations.</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_simplify">simplify</code></td>
<td>
<p>use simple names for primer hit results or complex</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_clustal_options">clustal_options</code></td>
<td>
<p>a list of options to pass to clustal omega, see
<code>link{clustalo}</code> for a list of options</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_distance_options">distance_options</code></td>
<td>
<p>a list of options to pass to dist.dna, see
<code>link{dist.dna}</code> for a list of options</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_api_key">api_key</code></td>
<td>
<p>NCBI api-key to allow faster sequence retrieval</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_...">...</code></td>
<td>
<p>additional arguments passed to Primer-Blast</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_.parallel">.parallel</code></td>
<td>
<p>if 'TRUE', perform in parallel, using parallel backend
provided by foreach</p>
</td></tr>
<tr><td><code id="search_primer_pair_+3A_.progress">.progress</code></td>
<td>
<p>name of the progress bar to use, see
<code><a href="plyr.html#topic+create_progress_bar">create_progress_bar</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements,
</p>
<table role = "presentation">
<tr><td><code>name</code></td>
<td>
<p>name of the primer pair</p>
</td></tr>
<tr><td><code>BLAST_result</code></td>
<td>
<p>html blast results from Primer-BLAST as
'a <code><a href="httr.html#topic+response">response</a></code></p>
</td></tr></table>
<p> object.
</p>
<table role = "presentation">
<tr><td><code>taxonomy</code></td>
<td>
<p>taxonomy for the primer products from NCBI</p>
</td></tr>
<tr><td><code>sequence</code></td>
<td>
<p>sequence of the primer products</p>
</td></tr>
<tr><td><code>alignment</code></td>
<td>
<p>multiple alignment of the primer products</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>phylogenetic tree of the reconstructed from the
'multiple alignment</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+primer_search">primer_search</a></code>, <code><a href="#topic+clustalo">clustalo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#simple search
mammals_16S = search_primer_pair(name='Mammals 16S',
 'CGGTTGGGGTGACCTCGGA', 'GCTGTTATCCCTAGGGTAACT')
#returning 1000 alignments, allow up to 3 mismatches in primer
mammals_16S = search_primer_pair(name='Mammals 16S',
 'CGGTTGGGGTGACCTCGGA', 'GCTGTTATCCCTAGGGTAACT',
 num_aligns=1000, total_primer_specificity_mismatch=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='seq_lengths'>Get a summary of sequence lengths from a primerTree object</h2><span id='topic+seq_lengths'></span>

<h3>Description</h3>

<p>Get a summary of sequence lengths from a primerTree object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seq_lengths(x, summarize = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seq_lengths_+3A_x">x</code></td>
<td>
<p>a primerTree object.</p>
</td></tr>
<tr><td><code id="seq_lengths_+3A_summarize">summarize</code></td>
<td>
<p>a logical indicating if a summary should be displayed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table of sequence length frequencies
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Show the counts for each length
seq_lengths(mammals_16S)

# Plot the distribution of lengths
seqLengths &lt;- seq_lengths(mammals_16S)
barplot(seqLengths, 
 main = "Frequency of sequence lengths for 16S mammal primers", 
 xlab="Amplicon length (in bp)", 
 ylab=("Frequency"))
</code></pre>

<hr>
<h2 id='seq_lengths.primerTree'>Method for primerTree objects</h2><span id='topic+seq_lengths.primerTree'></span>

<h3>Description</h3>

<p>Method for primerTree objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'primerTree'
seq_lengths(x, summarize = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seq_lengths.primerTree_+3A_x">x</code></td>
<td>
<p>a primerTree object.</p>
</td></tr>
<tr><td><code id="seq_lengths.primerTree_+3A_summarize">summarize</code></td>
<td>
<p>a logical indicating if a summary should be displayed</p>
</td></tr>
</table>

<hr>
<h2 id='summary.primerTree'>Summarize a primerTree result, printing quantiles of sequence length and
pairwise differences.</h2><span id='topic+summary.primerTree'></span>

<h3>Description</h3>

<p>Summarize a primerTree result, printing quantiles of sequence length and
pairwise differences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'primerTree'
summary(object, ..., probs = c(0, 0.05, 0.5, 0.95, 1), ranks = common_ranks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.primerTree_+3A_object">object</code></td>
<td>
<p>the primerTree object to summarise</p>
</td></tr>
<tr><td><code id="summary.primerTree_+3A_...">...</code></td>
<td>
<p>Ignored options</p>
</td></tr>
<tr><td><code id="summary.primerTree_+3A_probs">probs</code></td>
<td>
<p>quantile probabilities to compute, defaults to 0, 5, 50, 95,
and 100 probabilities.</p>
</td></tr>
<tr><td><code id="summary.primerTree_+3A_ranks">ranks</code></td>
<td>
<p>ranks to show unique counts for, defaults to the common ranks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisibly returns a list containing the printed results
</p>

<hr>
<h2 id='tree_from_alignment'>Construct a neighbor joining tree from a dna alignment</h2><span id='topic+tree_from_alignment'></span>

<h3>Description</h3>

<p>Construct a neighbor joining tree from a dna alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_from_alignment(dna, pairwise.deletion = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_from_alignment_+3A_dna">dna</code></td>
<td>
<p>fasta dna object the tree is to be constructed from</p>
</td></tr>
<tr><td><code id="tree_from_alignment_+3A_pairwise.deletion">pairwise.deletion</code></td>
<td>
<p>a logical indicating if the distance matrix should 
be constructed using pairwise deletion</p>
</td></tr>
<tr><td><code id="tree_from_alignment_+3A_...">...</code></td>
<td>
<p>furthur arguments to dist.dna</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+dist.dna">dist.dna</a></code>, <code><a href="ape.html#topic+nj">nj</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
