<!DOCTYPE html><html><head><title>Help for package prioGene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {prioGene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#deal_net'><p>Title deal with network</p></a></li>
<li><a href='#dise_gene'><p>a vector of disease related genes</p></a></li>
<li><a href='#edge_weight'><p>weights of edges of a net</p></a></li>
<li><a href='#genes_mat'><p>a one-to-many matrix of GO term and gene</p></a></li>
<li><a href='#get_edge_weight'><p>Title weight edge</p></a></li>
<li><a href='#get_gene_mat'><p>Get a one-to-many matrix of gene and GO term</p></a></li>
<li><a href='#get_neighbor'><p>Title get neighbor of a node</p></a></li>
<li><a href='#get_net_disease_term'><p>Title Get the GO terms for each pair of nodes in the network</p></a></li>
<li><a href='#get_node_weight'><p>Title weight node</p></a></li>
<li><a href='#get_Q'><p>Title get the disease risk transition probability matrix</p></a></li>
<li><a href='#get_R'><p>Title get the final genetic disease risk scores</p></a></li>
<li><a href='#get_R_0'><p>Title get the vector of initial disease risk scores for all genes</p></a></li>
<li><a href='#get_term_mat'><p>Get a one-to-many matrix of GO term and gene</p></a></li>
<li><a href='#get_W'><p>Title</p></a></li>
<li><a href='#metabolic_net'><p>a matrix, Human metabolic network</p></a></li>
<li><a href='#net'><p>a network of genes</p></a></li>
<li><a href='#net_disease'><p>a network of disease related genes</p></a></li>
<li><a href='#net_disease_term'><p>GO terms for each pair of nodes in the network</p></a></li>
<li><a href='#node_weight'><p>a matrix, genes and their weights</p></a></li>
<li><a href='#R_0'><p>the vector of initial disease risk scores for all genes</p></a></li>
<li><a href='#terms_mat'><p>a matrix, GO terms and GO genes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Candidate Gene Prioritization for Non-Communicable Diseases
Based on Functional Information</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GraphAndNetwork, FunctionalGenomics, Genetics, Network</td>
</tr>
<tr>
<td>Description:</td>
<td>In gene sequencing methods, the topological features of protein-protein interaction (PPI) networks are 
    often used, such as ToppNet <a href="https://toppgene.cchmc.org">https://toppgene.cchmc.org</a>. In this study, a candidate gene prioritization method
    was proposed for non-communicable diseases considering disease risks transferred between genes in weighted disease 
    PPI networks with weights for nodes and edges based on functional information.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>AnnotationDbi, org.Hs.eg.db</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-02-01 16:39:53 UTC; Administrator</td>
</tr>
<tr>
<td>Author:</td>
<td>Erqiang Hu [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Erqiang Hu &lt;13766876214@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-02-01 17:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='deal_net'>Title deal with network</h2><span id='topic+deal_net'></span>

<h3>Description</h3>

<p>Title deal with network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deal_net(net, dise_gene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deal_net_+3A_net">net</code></td>
<td>
<p>a network</p>
</td></tr>
<tr><td><code id="deal_net_+3A_dise_gene">dise_gene</code></td>
<td>
<p>a matrix with one column of genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>deal_net(net,dise_gene)
</code></pre>

<hr>
<h2 id='dise_gene'>a vector of disease related genes</h2><span id='topic+dise_gene'></span>

<h3>Description</h3>

<p>some genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dise_gene
</code></pre>


<h3>Format</h3>

<p>A matrix with 79 rows and 1 column</p>

<hr>
<h2 id='edge_weight'>weights of edges of a net</h2><span id='topic+edge_weight'></span>

<h3>Description</h3>

<p>the first two columns are a net, and third column is their weight
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edge_weight
</code></pre>


<h3>Format</h3>

<p>A matrix with 25 rows and 3 columns</p>

<hr>
<h2 id='genes_mat'>a one-to-many matrix of GO term and gene</h2><span id='topic+genes_mat'></span>

<h3>Description</h3>

<p>the first column is the gene symbol, the second column is the go terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genes_mat
</code></pre>


<h3>Format</h3>

<p>A matrix with 45 rows and 3 columns</p>


<h3>Details</h3>

<p>the third column is the number of go terms
</p>

<hr>
<h2 id='get_edge_weight'>Title weight edge</h2><span id='topic+get_edge_weight'></span>

<h3>Description</h3>

<p>Title weight edge
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_edge_weight(net_disease_term, terms_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_edge_weight_+3A_net_disease_term">net_disease_term</code></td>
<td>
<p>GO terms for each pair of nodes in the network</p>
</td></tr>
<tr><td><code id="get_edge_weight_+3A_terms_mat">terms_mat</code></td>
<td>
<p>result of get_term_mat()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_edge_weight(net_disease_term,terms_mat)
</code></pre>

<hr>
<h2 id='get_gene_mat'>Get a one-to-many matrix of gene and GO term</h2><span id='topic+get_gene_mat'></span>

<h3>Description</h3>

<p>Get a one-to-many matrix of gene and GO term
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_mat(net_disease)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_mat_+3A_net_disease">net_disease</code></td>
<td>
<p>a disease related network, matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_gene_mat(net_disease)
</code></pre>

<hr>
<h2 id='get_neighbor'>Title get neighbor of a node</h2><span id='topic+get_neighbor'></span>

<h3>Description</h3>

<p>Title get neighbor of a node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_neighbor(node, net)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_neighbor_+3A_node">node</code></td>
<td>
<p>a gene</p>
</td></tr>
<tr><td><code id="get_neighbor_+3A_net">net</code></td>
<td>
<p>a network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of gene
</p>

<hr>
<h2 id='get_net_disease_term'>Title Get the GO terms for each pair of nodes in the network</h2><span id='topic+get_net_disease_term'></span>

<h3>Description</h3>

<p>Title Get the GO terms for each pair of nodes in the network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_net_disease_term(genes_mat, net_disease)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_net_disease_term_+3A_genes_mat">genes_mat</code></td>
<td>
<p>a one-to-many matrix of GO term and gene</p>
</td></tr>
<tr><td><code id="get_net_disease_term_+3A_net_disease">net_disease</code></td>
<td>
<p>a disease related network, matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_net_disease_term(genes_mat,net_disease)
</code></pre>

<hr>
<h2 id='get_node_weight'>Title weight node</h2><span id='topic+get_node_weight'></span>

<h3>Description</h3>

<p>Title weight node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_node_weight(genes_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_node_weight_+3A_genes_mat">genes_mat</code></td>
<td>
<p>a one-to-many matrix of GO term and gene</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_node_weight(genes_mat)
</code></pre>

<hr>
<h2 id='get_Q'>Title get the disease risk transition probability matrix</h2><span id='topic+get_Q'></span>

<h3>Description</h3>

<p>Title get the disease risk transition probability matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Q(node_weight, net_disease_term)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_Q_+3A_node_weight">node_weight</code></td>
<td>
<p>a matrix, genes and their weights</p>
</td></tr>
<tr><td><code id="get_Q_+3A_net_disease_term">net_disease_term</code></td>
<td>
<p>GO terms for each pair of nodes in the network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>

<hr>
<h2 id='get_R'>Title get the final genetic disease risk scores</h2><span id='topic+get_R'></span>

<h3>Description</h3>

<p>Title get the final genetic disease risk scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_R(node_weight, net_disease_term, bet, R_0, threshold = 10^(-9))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_R_+3A_node_weight">node_weight</code></td>
<td>
<p>a matrix, genes and their weights</p>
</td></tr>
<tr><td><code id="get_R_+3A_net_disease_term">net_disease_term</code></td>
<td>
<p>GO terms for each pair of nodes in the network</p>
</td></tr>
<tr><td><code id="get_R_+3A_bet">bet</code></td>
<td>
<p>a parameter to measure the importance of genes and interactions</p>
</td></tr>
<tr><td><code id="get_R_+3A_r_0">R_0</code></td>
<td>
<p>the vector of initial disease risk scores for all genes</p>
</td></tr>
<tr><td><code id="get_R_+3A_threshold">threshold</code></td>
<td>
<p>a threshold for terminating iterations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>net_disease &lt;- deal_net(net,dise_gene)
genes_mat &lt;- get_gene_mat(net_disease)
node_weight &lt;- get_node_weight(genes_mat)
net_disease_term &lt;- get_net_disease_term(genes_mat,net_disease)
R_0&lt;- get_R_0(dise_gene,node_weight,f=1)
result &lt;- get_R(node_weight, net_disease_term, bet = 0.5, R_0 = R_0, threshold = 10^(-9))
</code></pre>

<hr>
<h2 id='get_R_0'>Title get the vector of initial disease risk scores for all genes</h2><span id='topic+get_R_0'></span>

<h3>Description</h3>

<p>Title get the vector of initial disease risk scores for all genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_R_0(disease_gene, node_weight, f = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_R_0_+3A_disease_gene">disease_gene</code></td>
<td>
<p>a matrix of a column of genes</p>
</td></tr>
<tr><td><code id="get_R_0_+3A_node_weight">node_weight</code></td>
<td>
<p>a matrix, genes and their weights</p>
</td></tr>
<tr><td><code id="get_R_0_+3A_f">f</code></td>
<td>
<p>an integer parameter to measure the significance of disease
genes and candidate genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_R_0(dise_gene,node_weight,1)
</code></pre>

<hr>
<h2 id='get_term_mat'>Get a one-to-many matrix of GO term and gene</h2><span id='topic+get_term_mat'></span>

<h3>Description</h3>

<p>Get a one-to-many matrix of GO term and gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_mat(net_disease)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_term_mat_+3A_net_disease">net_disease</code></td>
<td>
<p>a disease related network, matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_term_mat(net_disease)
</code></pre>

<hr>
<h2 id='get_W'>Title</h2><span id='topic+get_W'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_W(node1, node2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_W_+3A_node1">node1</code></td>
<td>
<p>a gene</p>
</td></tr>
<tr><td><code id="get_W_+3A_node2">node2</code></td>
<td>
<p>a gene</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a number
</p>

<hr>
<h2 id='metabolic_net'>a matrix, Human metabolic network</h2><span id='topic+metabolic_net'></span>

<h3>Description</h3>

<p>a matrix, Human metabolic network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metabolic_net
</code></pre>


<h3>Format</h3>

<p>A matrix with 589,199 rows and 2 columns</p>

<hr>
<h2 id='net'>a network of genes</h2><span id='topic+net'></span>

<h3>Description</h3>

<p>a network of genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>net
</code></pre>


<h3>Format</h3>

<p>A matrix with 2000 rows and 2 columns</p>

<hr>
<h2 id='net_disease'>a network of disease related genes</h2><span id='topic+net_disease'></span>

<h3>Description</h3>

<p>a network of disease related genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>net_disease
</code></pre>


<h3>Format</h3>

<p>A matrix with 26 rows and 2 columns</p>

<hr>
<h2 id='net_disease_term'>GO terms for each pair of nodes in the network</h2><span id='topic+net_disease_term'></span>

<h3>Description</h3>

<p>the first two columns is the network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>net_disease_term
</code></pre>


<h3>Format</h3>

<p>A matrix with 25 rows and 4 columns</p>


<h3>Details</h3>

<p>the third column is the go terms,the fourth column is the number of go terms
</p>
<p>the fourth column is the number of go terms
</p>

<hr>
<h2 id='node_weight'>a matrix, genes and their weights</h2><span id='topic+node_weight'></span>

<h3>Description</h3>

<p>a matrix, genes and their weights
</p>


<h3>Usage</h3>

<pre><code class='language-R'>node_weight
</code></pre>


<h3>Format</h3>

<p>A matrix with 45 rows and 2 columns</p>

<hr>
<h2 id='R_0'>the vector of initial disease risk scores for all genes</h2><span id='topic+R_0'></span>

<h3>Description</h3>

<p>the vector of initial disease risk scores for all genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>R_0
</code></pre>


<h3>Format</h3>

<p>A vector of 45 number</p>

<hr>
<h2 id='terms_mat'>a matrix, GO terms and GO genes</h2><span id='topic+terms_mat'></span>

<h3>Description</h3>

<p>a one-to-many matrix of GO term and gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>terms_mat
</code></pre>


<h3>Format</h3>

<p>A matrix with 1172 rows and 3 columns</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
