<!DOCTYPE html><html><head><title>Help for package prioritylasso</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {prioritylasso}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_offsets'><p>Calculates the offsets for the current block</p></a></li>
<li><a href='#coef.prioritylasso'><p>Extract coefficients from a prioritylasso object</p></a></li>
<li><a href='#compare_boolean'><p>Compare the rows of a matrix with a pattern</p></a></li>
<li><a href='#cvm_prioritylasso'><p>prioritylasso with several block specifications</p></a></li>
<li><a href='#missing.control'><p>Construct control structures for handling of missing data for <code>prioritylasso</code></p></a></li>
<li><a href='#pl_data'><p>Simulated AML data with binary outcome</p></a></li>
<li><a href='#predict.prioritylasso'><p>Predictions from prioritylasso</p></a></li>
<li><a href='#prioritylasso'><p>Patient outcome prediction based on multi-omics data taking practitioners' preferences into account</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analyzing Multiple Omics Data with an Offset Approach</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Simon Klau, Roman Hornung, Alina Bauer, Jonas Hagenberg</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Roman Hornung &lt;hornung@ibe.med.uni-muenchen.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fits successive Lasso models for several blocks of (omics) data with different priorities and takes the predicted values as an offset for the next block. Also offers options to deal with block-wise missingness in multi-omics data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival, glmnet, utils, checkmate</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, pROC, bookdown, ipflasso</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-30 14:07:05 UTC; hornung</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-10 21:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_offsets'>Calculates the offsets for the current block</h2><span id='topic+calculate_offsets'></span>

<h3>Description</h3>

<p>Calculates the offsets for the current block
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_offsets(
  current_missings,
  current_observations,
  mcontrol,
  current_block,
  pred,
  liste,
  X,
  blocks,
  current_intercept
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_offsets_+3A_current_missings">current_missings</code></td>
<td>
<p>index vector (indices) of current missing observations</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_current_observations">current_observations</code></td>
<td>
<p>index vector (indices) of current used observations</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_mcontrol">mcontrol</code></td>
<td>
<p>control for missing data handling</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_current_block">current_block</code></td>
<td>
<p>index of current block</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_pred">pred</code></td>
<td>
<p>predictions of current block</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_liste">liste</code></td>
<td>
<p>list with offsets</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_x">X</code></td>
<td>
<p>original data</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_blocks">blocks</code></td>
<td>
<p>information which variable belongs to which block</p>
</td></tr>
<tr><td><code id="calculate_offsets_+3A_current_intercept">current_intercept</code></td>
<td>
<p>the intercept estimated for the current block</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with offsets, used imputation model and the blocks used for the imputation model (if applicable)
</p>

<hr>
<h2 id='coef.prioritylasso'>Extract coefficients from a prioritylasso object</h2><span id='topic+coef.prioritylasso'></span>

<h3>Description</h3>

<p>Extract coefficients from a prioritylasso object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'prioritylasso'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.prioritylasso_+3A_object">object</code></td>
<td>
<p>model of type prioritylasso</p>
</td></tr>
<tr><td><code id="coef.prioritylasso_+3A_...">...</code></td>
<td>
<p>additional arguments, currently not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the coefficients and the intercepts
</p>

<hr>
<h2 id='compare_boolean'>Compare the rows of a matrix with a pattern</h2><span id='topic+compare_boolean'></span>

<h3>Description</h3>

<p>Compare the rows of a matrix with a pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_boolean(object, pattern)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_boolean_+3A_object">object</code></td>
<td>
<p>matrix</p>
</td></tr>
<tr><td><code id="compare_boolean_+3A_pattern">pattern</code></td>
<td>
<p>pattern which is compared against the rows of the matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logical vector if the pattern matches the rows
</p>

<hr>
<h2 id='cvm_prioritylasso'>prioritylasso with several block specifications</h2><span id='topic+cvm_prioritylasso'></span>

<h3>Description</h3>

<p>Runs prioritylasso for a list of block specifications and gives the best results
in terms of cv error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cvm_prioritylasso(
  X,
  Y,
  weights,
  family,
  type.measure,
  blocks.list,
  max.coef.list = NULL,
  block1.penalization = TRUE,
  lambda.type = "lambda.min",
  standardize = TRUE,
  nfolds = 10,
  foldid,
  cvoffset = FALSE,
  cvoffsetnfolds = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cvm_prioritylasso_+3A_x">X</code></td>
<td>
<p>a (nxp) matrix of predictors with observations in rows and predictors in columns.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_y">Y</code></td>
<td>
<p>n-vector giving the value of the response (either continuous, numeric-binary 0/1, or <code>Surv</code> object).</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_weights">weights</code></td>
<td>
<p>observation weights. Default is 1 for each observation.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_family">family</code></td>
<td>
<p>should be &quot;gaussian&quot; for continuous <code>Y</code>, &quot;binomial&quot; for binary <code>Y</code>, &quot;cox&quot; for <code>Y</code> of type <code>Surv</code>.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_type.measure">type.measure</code></td>
<td>
<p>accuracy/error measure computed in cross-validation. It should be &quot;class&quot; (classification error) or &quot;auc&quot; (area under the ROC curve) if <code>family="binomial"</code>, &quot;mse&quot; (mean squared error) if <code>family="gaussian"</code> and &quot;deviance&quot; if <code>family="cox"</code> which uses the partial-likelihood.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_blocks.list">blocks.list</code></td>
<td>
<p>list of the format <code>list(list(bp1=...,bp2=...,), list(bp1=,...,bp2=...,), ...)</code>. For the specification of the entries, see <code><a href="#topic+prioritylasso">prioritylasso</a></code>.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_max.coef.list">max.coef.list</code></td>
<td>
<p>list of <code>max.coef</code> vectors. The first entries are omitted if <code>block1.penalization = FALSE</code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_block1.penalization">block1.penalization</code></td>
<td>
<p>whether the first block should be penalized. Default is TRUE.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_lambda.type">lambda.type</code></td>
<td>
<p>specifies the value of lambda used for the predictions. <code>lambda.min</code> gives lambda with minimum cross-validated errors. <code>lambda.1se</code> gives the largest value of lambda such that the error is within 1 standard error of the minimum. Note that <code>lambda.1se</code> can only be chosen without restrictions of <code>max.coef</code>.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_standardize">standardize</code></td>
<td>
<p>logical, whether the predictors should be standardized or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_nfolds">nfolds</code></td>
<td>
<p>the number of CV procedure folds.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_foldid">foldid</code></td>
<td>
<p>an optional vector of values between 1 and nfold identifying what fold each observation is in.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_cvoffset">cvoffset</code></td>
<td>
<p>logical, whether CV should be used to estimate the offsets. Default is FALSE.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_cvoffsetnfolds">cvoffsetnfolds</code></td>
<td>
<p>the number of folds in the CV procedure that is performed to estimate the offsets. Default is 10. Only relevant if <code>cvoffset=TRUE</code>.</p>
</td></tr>
<tr><td><code id="cvm_prioritylasso_+3A_...">...</code></td>
<td>
<p>other arguments that can be passed to the function <code>prioritylasso</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>cvm_prioritylasso</code> with the following elements. If these elements are lists, they contain the results for each penalized block of the best result.
</p>

<dl>
<dt><code>lambda.ind</code></dt><dd><p>list with indices of lambda for <code>lambda.type</code>.</p>
</dd>
<dt><code>lambda.type</code></dt><dd><p>type of lambda which is used for the predictions.</p>
</dd>
<dt><code>lambda.min</code></dt><dd><p>list with values of lambda for <code>lambda.type</code>.</p>
</dd>
<dt><code>min.cvm</code></dt><dd><p>list with the mean cross-validated errors for <code>lambda.type</code>.</p>
</dd>
<dt><code>nzero</code></dt><dd><p>list with numbers of non-zero coefficients for <code>lambda.type</code>.</p>
</dd>
<dt><code>glmnet.fit</code></dt><dd><p>list of fitted <code>glmnet</code> objects.</p>
</dd>
<dt><code>name</code></dt><dd><p>a text string indicating type of measure.</p>
</dd>
<dt><code>block1unpen</code></dt><dd><p>if <code>block1.penalization = FALSE</code>, the results of either the fitted <code>glm</code> or <code>coxph</code> object.</p>
</dd>
<dt><code>best.blocks</code></dt><dd><p>character vector with the indices of the best block specification.</p>
</dd>
<dt><code>best.blocks.indices</code></dt><dd><p>list with the indices of the best block specification ordered by best to worst.</p>
</dd>
<dt><code>best.max.coef</code></dt><dd><p>vector with the number of maximal coefficients corresponding to <code>best.blocks</code>.</p>
</dd>
<dt><code>best.model</code></dt><dd><p>complete <code>prioritylasso</code> model of the best solution.</p>
</dd>
<dt><code>coefficients</code></dt><dd><p>coefficients according to the results obtained with <code>best.blocks</code>.</p>
</dd>
<dt><code>call</code></dt><dd><p>the function call.</p>
</dd>
</dl>



<h3>Note</h3>

<p>The function description and the first example are based on the R package <code>ipflasso</code>.
</p>


<h3>Author(s)</h3>

<p>Simon Klau <br />
Maintainer: Roman Hornung (<a href="mailto:hornung@ibe.med.uni-muenchen.de">hornung@ibe.med.uni-muenchen.de</a>)
</p>


<h3>References</h3>

<p>Klau, S., Jurinovic, V., Hornung, R., Herold, T., Boulesteix, A.-L. (2018). Priority-Lasso: a simple hierarchical approach to the prediction of clinical outcome using multi-omics data. BMC Bioinformatics 19, 322
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pl_data">pl_data</a></code>, <code><a href="#topic+prioritylasso">prioritylasso</a></code>, <code><a href="ipflasso.html#topic+cvr2.ipflasso">cvr2.ipflasso</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cvm_prioritylasso(X = matrix(rnorm(50*500),50,500), Y = rnorm(50), family = "gaussian",
                  type.measure = "mse", lambda.type = "lambda.min", nfolds = 5,
                  blocks.list = list(list(bp1=1:75, bp2=76:200, bp3=201:500),
                                     list(bp1=1:75, bp2=201:500, bp3=76:200)))
## Not run: 
cvm_prioritylasso(X = pl_data[,1:1028], Y = pl_data[,1029], family = "binomial",
                  type.measure = "auc", standardize = FALSE, block1.penalization = FALSE,
                  blocks.list = list(list(1:4, 5:9, 10:28, 29:1028),
                                     list(1:4, 5:9, 29:1028, 10:28)),
                  max.coef.list = list(c(Inf, Inf, Inf, 10), c(Inf, Inf, 10, Inf)))
## End(Not run)

</code></pre>

<hr>
<h2 id='missing.control'>Construct control structures for handling of missing data for <code>prioritylasso</code></h2><span id='topic+missing.control'></span>

<h3>Description</h3>

<p>Construct control structures for handling of missing data for <code>prioritylasso</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>missing.control(
  handle.missingdata = c("none", "ignore", "impute.offset"),
  offset.firstblock = c("zero", "intercept"),
  impute.offset.cases = c("complete.cases", "available.cases"),
  nfolds.imputation = 10,
  lambda.imputation = c("lambda.min", "lambda.1se"),
  perc.comp.cases.warning = 0.3,
  threshold.available.cases = 30,
  select.available.cases = c("maximise.blocks", "max")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="missing.control_+3A_handle.missingdata">handle.missingdata</code></td>
<td>
<p>how blockwise missing data should be treated. Default is <code>none</code> which does nothing, <code>ignore</code> ignores the observations with missing data for the current block, <code>impute.offset</code> imputes the offset for the missing values.</p>
</td></tr>
<tr><td><code id="missing.control_+3A_offset.firstblock">offset.firstblock</code></td>
<td>
<p>determines if the offset of the first block for missing observations is zero or the intercept of the observed values for <code>handle.missingdata = ignore</code></p>
</td></tr>
<tr><td><code id="missing.control_+3A_impute.offset.cases">impute.offset.cases</code></td>
<td>
<p>which cases/observations should be used for the imputation model to impute missing offsets. Supported are complete cases (additional constraint is that every observation can only contain one missing block) and all available observations which have an overlap with the current block.</p>
</td></tr>
<tr><td><code id="missing.control_+3A_nfolds.imputation">nfolds.imputation</code></td>
<td>
<p>nfolds for the glmnet of the imputation model</p>
</td></tr>
<tr><td><code id="missing.control_+3A_lambda.imputation">lambda.imputation</code></td>
<td>
<p>which lambda-value should be used for predicting the imputed offsets in cv.glmnet</p>
</td></tr>
<tr><td><code id="missing.control_+3A_perc.comp.cases.warning">perc.comp.cases.warning</code></td>
<td>
<p>percentage of complete cases when a warning is issued of too few cases for the imputation model</p>
</td></tr>
<tr><td><code id="missing.control_+3A_threshold.available.cases">threshold.available.cases</code></td>
<td>
<p>if the number of available cases for <code>impute.offset.cases = available.cases</code> is below this threshold, <code>prioritylasso</code> tries to reduce the number of blocks taken into account for the imputation model to increase the number of observations used for the imputation model.</p>
</td></tr>
<tr><td><code id="missing.control_+3A_select.available.cases">select.available.cases</code></td>
<td>
<p>determines how the blocks which are used for the imputation model are selected when <code>impute.offset.cases = available.cases</code>. <code>max</code> selects the blocks that maximise the number of observations, <code>maximise.blocks</code> tries to include as many blocks as possible, starting with the blocks with the hightes priority</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with control parameters
</p>

<hr>
<h2 id='pl_data'>Simulated AML data with binary outcome</h2><span id='topic+pl_data'></span>

<h3>Description</h3>

<p>A data set containing the binary outcome and 1028 predictor variables of 400 artificial AML patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pl_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 400 rows and 1029 variables:
</p>

<dl>
<dt>pl_out: (<code>pl_data[,1029]</code>)</dt><dd><p>binary outcome representing refractory status.</p>
</dd>
<dt>b1: (<code>pl_data[,1:4]</code>)</dt><dd><p>4 binary variables representing variables with a known influence on the outcome.</p>
</dd>
<dt>b2: (<code>pl_data[,5:9]</code>)</dt><dd><p>5 continuous variables representing clinical variables.</p>
</dd>
<dt>b3: (<code>pl_data[,10:28]</code>)</dt><dd><p>19 binary variables representing mutations.</p>
</dd>
<dt>b4: (<code>pl_data[,29:1028]</code>)</dt><dd><p>1000 continuous variables representing gene expression data.</p>
</dd>
</dl>



<h3>Details</h3>

<p>We generated the data in the following way: We took the empirical correlation of 1028 variables related to
315 AML patients. This correlation served as a correlation matrix when generating 1028 multivariate
normally distributed variables with the R function <code><a href="mvtnorm.html#topic+rmvnorm">rmvnorm</a></code>. Because we didn't have a positive
definite matrix, we took the nearest positive definite matrix according to the function <code><a href="Matrix.html#topic+nearPD">nearPD</a></code>.
The variables that should be binary were dichotomized, so that their marginal probabilities corresponded to
the marginal probabilities they were based on.
The coefficients were defined by
</p>

<ul>
<li><p><code>beta_b1 &lt;- c(0.8, 0.8, 0.6, 0.6)</code>
</p>
</li>
<li><p><code>beta_b2 &lt;- c(rep(0.5,3), rep(0,2))</code>
</p>
</li>
<li><p><code>beta_b3 &lt;- c(rep(0.4, 4), rep(0,15))</code>
</p>
</li>
<li><p><code>beta_b4 &lt;- c(rep(0.5, 5), rep(0.3, 5), rep(0,990))</code>.
</p>
</li></ul>

<p>We included them in the vector <code>beta &lt;- c(beta_b1, beta_b2, beta_b3, beta_b4)</code> and calculated
the probability through </p>
<p style="text-align: center;"><code class="reqn">pi = exp(\beta*x) / (1 + exp(\beta*x))</code>
</p>
<p> where x denotes our data matrix
with 1028 predictor variables. Finally we got the outcome through
<code>pl_out &lt;- rbinom(400, size = 1, p = pi)</code>.
</p>

<hr>
<h2 id='predict.prioritylasso'>Predictions from prioritylasso</h2><span id='topic+predict.prioritylasso'></span>

<h3>Description</h3>

<p>Makes predictions for a <code>prioritylasso</code> object. It can be chosen between linear predictors or fitted values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'prioritylasso'
predict(
  object,
  newdata = NULL,
  type = c("link", "response"),
  handle.missingtestdata = c("none", "omit.prediction", "set.zero", "impute.block"),
  include.allintercepts = FALSE,
  use.blocks = "all",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.prioritylasso_+3A_object">object</code></td>
<td>
<p>An object of class <code>prioritylasso</code>.</p>
</td></tr>
<tr><td><code id="predict.prioritylasso_+3A_newdata">newdata</code></td>
<td>
<p>(nnew <code>x</code> p) matrix or data frame with new values.</p>
</td></tr>
<tr><td><code id="predict.prioritylasso_+3A_type">type</code></td>
<td>
<p>Specifies the type of predictions. <code>link</code> gives the linear predictors for all types of response and <code>response</code> gives the fitted values.</p>
</td></tr>
<tr><td><code id="predict.prioritylasso_+3A_handle.missingtestdata">handle.missingtestdata</code></td>
<td>
<p>Specifies how to deal with missing data in the test data; possibilities are <code>none</code>, <code>omit.prediction</code>, <code>set.zero</code> and <code>impute.block</code></p>
</td></tr>
<tr><td><code id="predict.prioritylasso_+3A_include.allintercepts">include.allintercepts</code></td>
<td>
<p>should the intercepts from all blocks included in the prediction? If <code>FALSE</code>, only the intercept from the first block is included (default in the past).</p>
</td></tr>
<tr><td><code id="predict.prioritylasso_+3A_use.blocks">use.blocks</code></td>
<td>
<p>determines which blocks are used for the prediction, the default is all. Otherwise one can specify the number of blocks which are used in a vector</p>
</td></tr>
<tr><td><code id="predict.prioritylasso_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>handle.missingtestdata</code> specifies how to deal with missing data.
The default <code>none</code> cannot handle missing data, <code>omit.prediction</code> does not make a prediction for observations with missing values and return <code>NA</code>. <code>set.zero</code> ignores
the missing data for the calculation of the prediction (the missing value is set to zero).
<code>impute.block</code> uses an imputation model to impute the offset of a missing block. This only works if the prioritylasso object was fitted with <code>handle.missingdata = "impute.offset"</code>.
If <code>impute.offset.cases = "complete.cases"</code> was used, then every observation can have only one missing block. For observations with more than one missing block, <code>NA</code> is returned.
If <code>impute.offset.cases = "available.cases"</code> was used, the missingness pattern in the test data has to be the same as in the train data. For observations with an unknown missingness pattern, <code>NA</code> is returned.
</p>


<h3>Value</h3>

<p>Predictions that depend on <code>type</code>.
</p>


<h3>Author(s)</h3>

<p>Simon Klau
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pl_data">pl_data</a></code>, <code><a href="#topic+prioritylasso">prioritylasso</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pl_bin &lt;- prioritylasso(X = matrix(rnorm(50*190),50,190), Y = rbinom(50,1,0.5),
                       family = "binomial", type.measure = "auc",
                       blocks = list(block1=1:13,block2=14:80, block3=81:190),
                       block1.penalization = TRUE, lambda.type = "lambda.min",
                       standardize = FALSE, nfolds = 3)

newdata_bin &lt;- matrix(rnorm(10*190),10,190)

predict(object = pl_bin, newdata = newdata_bin, type = "response")
</code></pre>

<hr>
<h2 id='prioritylasso'>Patient outcome prediction based on multi-omics data taking practitioners' preferences into account</h2><span id='topic+prioritylasso'></span>

<h3>Description</h3>

<p>Fits successive Lasso models for several ordered blocks of (omics) data and takes the predicted values as an offset for the next block.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prioritylasso(
  X,
  Y,
  weights,
  family = c("gaussian", "binomial", "cox"),
  type.measure,
  blocks,
  max.coef = NULL,
  block1.penalization = TRUE,
  lambda.type = "lambda.min",
  standardize = TRUE,
  nfolds = 10,
  foldid,
  cvoffset = FALSE,
  cvoffsetnfolds = 10,
  mcontrol = missing.control(),
  scale.y = FALSE,
  return.x = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prioritylasso_+3A_x">X</code></td>
<td>
<p>a (nxp) matrix of predictors with observations in rows and predictors in columns.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_y">Y</code></td>
<td>
<p>n-vector giving the value of the response (either continuous, numeric-binary 0/1, or <code>Surv</code> object).</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_weights">weights</code></td>
<td>
<p>observation weights. Default is 1 for each observation.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_family">family</code></td>
<td>
<p>should be &quot;gaussian&quot; for continuous <code>Y</code>, &quot;binomial&quot; for binary <code>Y</code>, &quot;cox&quot; for <code>Y</code> of type <code>Surv</code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_type.measure">type.measure</code></td>
<td>
<p>accuracy/error measure computed in cross-validation. It should be &quot;class&quot; (classification error) or &quot;auc&quot; (area under the ROC curve) if <code>family="binomial"</code>, &quot;mse&quot; (mean squared error) if <code>family="gaussian"</code> and &quot;deviance&quot; if <code>family="cox"</code> which uses the partial-likelihood.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_blocks">blocks</code></td>
<td>
<p>list of the format <code>list(bp1=...,bp2=...,)</code>, where the dots should be replaced by the indices of the predictors included in this block. The blocks should form a partition of 1:p.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_max.coef">max.coef</code></td>
<td>
<p>vector with integer values which specify the number of maximal coefficients for each block. The first entry is omitted if <code>block1.penalization = FALSE</code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_block1.penalization">block1.penalization</code></td>
<td>
<p>whether the first block should be penalized. Default is TRUE.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_lambda.type">lambda.type</code></td>
<td>
<p>specifies the value of lambda used for the predictions. <code>lambda.min</code> gives lambda with minimum cross-validated errors. <code>lambda.1se</code> gives the largest value of lambda such that the error is within 1 standard error of the minimum. Note that <code>lambda.1se</code> can only be chosen without restrictions of <code>max.coef</code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_standardize">standardize</code></td>
<td>
<p>logical, whether the predictors should be standardized or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_nfolds">nfolds</code></td>
<td>
<p>the number of CV procedure folds.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_foldid">foldid</code></td>
<td>
<p>an optional vector of values between 1 and nfold identifying what fold each observation is in.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_cvoffset">cvoffset</code></td>
<td>
<p>logical, whether CV should be used to estimate the offsets. Default is FALSE.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_cvoffsetnfolds">cvoffsetnfolds</code></td>
<td>
<p>the number of folds in the CV procedure that is performed to estimate the offsets. Default is 10. Only relevant if <code>cvoffset=TRUE</code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_mcontrol">mcontrol</code></td>
<td>
<p>controls how to deal with blockwise missing data. For details see below or <code><a href="#topic+missing.control">missing.control</a></code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_scale.y">scale.y</code></td>
<td>
<p>determines if y gets scaled before passed to glmnet. Can only be used for <code>family = 'gaussian'</code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_return.x">return.x</code></td>
<td>
<p>logical, determines if the input data should be returned by <code>prioritylasso</code>. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="prioritylasso_+3A_...">...</code></td>
<td>
<p>other arguments that can be passed to the function <code>cv.glmnet</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>block1.penalization = TRUE</code>, the function fits a Lasso model for each block. First, a standard Lasso for the first entry of <code>blocks</code> (block of priority 1) is fitted.
The predictions are then taken as an offset in the Lasso fit of the block of priority 2, etc.
For <code>block1.penalization = FALSE</code>, the function fits a model without penalty to the block of priority 1 (recommended as a block with clinical predictors where <code>p &lt; n</code>).
This is either a generalized linear model for family &quot;gaussian&quot; or &quot;binomial&quot;, or a Cox model. The predicted values are then taken as an offset in the following Lasso fit of the block with priority 2, etc. <br />
</p>
<p>The first entry of <code>blocks</code> contains the indices of variables of the block with priority 1 (first block included in the model).
Assume that <code>blocks = list(1:100, 101:200, 201:300)</code> then the block with priority 1 consists of the first 100 variables of the data matrix.
Analogously, the block with priority 2 consists of the variables 101 to 200 and the block with priority 3 of the variables 201 to 300.
</p>
<p><code>standardize = TRUE</code> leads to a standardisation of the covariables (<code>X</code>) in <code>glmnet</code> which is recommend by <code>glmnet</code>.
In case of an unpenalized first block, the covariables for the first block are not standardized.
Please note that the returned coefficients are rescaled to the original scale of the covariates as provided in <code>X</code>.
Therefore, new data in <code>predict.prioritylasso</code> should be on the same scale as <code>X</code>.
</p>
<p>To use the method with blockwise missing data, one can set <code>handle.missingdata = ignore</code>.
Then, to calculate the coefficients for a given block only the observations with values for this blocks are used.
For the observations with missing values, the result from the previous block is used as the offset for the next block.
Crossvalidated offsets are not supported with <code>handle.missingdata = ignore</code>.
Please note that dealing with single missing values is not supported.
Normally, every observation gets a unique foldid which stays the same across all blocks for the call to <code>cv.glmnet</code>.
However when <code>handle.missingdata != none</code>, the foldid is set new for every block.
</p>


<h3>Value</h3>

<p>object of class <code>prioritylasso</code> with the following elements. If these elements are lists, they contain the results for each penalized block.
</p>

<dl>
<dt><code>lambda.ind</code></dt><dd><p>list with indices of lambda for <code>lambda.type</code>.</p>
</dd>
<dt><code>lambda.type</code></dt><dd><p>type of lambda which is used for the predictions.</p>
</dd>
<dt><code>lambda.min</code></dt><dd><p>list with values of lambda for <code>lambda.type</code>.</p>
</dd>
<dt><code>min.cvm</code></dt><dd><p>list with the mean cross-validated errors for <code>lambda.type</code>.</p>
</dd>
<dt><code>nzero</code></dt><dd><p>list with numbers of non-zero coefficients for <code>lambda.type</code>.</p>
</dd>
<dt><code>glmnet.fit</code></dt><dd><p>list of fitted <code>glmnet</code> objects.</p>
</dd>
<dt><code>name</code></dt><dd><p>a text string indicating type of measure.</p>
</dd>
<dt><code>block1unpen</code></dt><dd><p>if <code>block1.penalization = FALSE</code>, the results of either the fitted <code>glm</code> or <code>coxph</code> object corresponding to <code>best.blocks</code>.</p>
</dd>
<dt><code>coefficients</code></dt><dd><p>vector of estimated coefficients. If <code>block1.penalization = FALSE</code> and <code>family = gaussian</code> or <code>binomial</code>, the first entry contains an intercept.</p>
</dd>
<dt><code>call</code></dt><dd><p>the function call.</p>
</dd>
<dt><code>X</code></dt><dd><p>the original data used for the calculation or <code>NA</code> if <code>return.x = FALSE</code></p>
</dd>
<dt><code>missing.data</code></dt><dd><p>list with logical entries for every block which observation is missing (<code>TRUE</code> means missing)</p>
</dd>
<dt><code>imputation.models</code></dt><dd><p>if <code>handle.missingdata = "impute.offsets"</code>, it contains the used imputation models</p>
</dd>
<dt><code>blocks.used.for.imputation</code></dt><dd><p>if <code>handle.missingdata = "impute.offsets"</code>, it contains the blocks which were used for the imputation model for every block</p>
</dd>
<dt><code>y.scale.param</code></dt><dd><p>if <code>scale.y = TRUE</code>, then it contains the mean and sd used for scaling.</p>
</dd>
<dt><code>blocks</code></dt><dd><p>list with the description which variables belong to which block</p>
</dd>
<dt><code>mcontrol</code></dt><dd><p>the missing control settings used</p>
</dd>
<dt><code>family</code></dt><dd><p>the family of the fitted data</p>
</dd>
<dt><code>dim.x</code></dt><dd><p>the dimension of the used training data</p>
</dd>
</dl>



<h3>Note</h3>

<p>The function description and the first example are based on the R package <code>ipflasso</code>. The second example is inspired by the example of <code><a href="glmnet.html#topic+cv.glmnet">cv.glmnet</a></code> from the <code>glmnet</code> package.
</p>


<h3>Author(s)</h3>

<p>Simon Klau, Roman Hornung, Alina Bauer <br />
Maintainer: Roman Hornung (<a href="mailto:hornung@ibe.med.uni-muenchen.de">hornung@ibe.med.uni-muenchen.de</a>)
</p>


<h3>References</h3>

<p>Klau, S., Jurinovic, V., Hornung, R., Herold, T., Boulesteix, A.-L. (2018). Priority-Lasso: a simple hierarchical approach to the prediction of clinical outcome using multi-omics data. BMC Bioinformatics 19, 322
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pl_data">pl_data</a></code>, <code><a href="#topic+cvm_prioritylasso">cvm_prioritylasso</a></code>, <code><a href="ipflasso.html#topic+cvr.ipflasso">cvr.ipflasso</a></code>, <code><a href="ipflasso.html#topic+cvr2.ipflasso">cvr2.ipflasso</a></code>, <code><a href="#topic+missing.control">missing.control</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># gaussian
  prioritylasso(X = matrix(rnorm(50*500),50,500), Y = rnorm(50), family = "gaussian",
                type.measure = "mse", blocks = list(bp1=1:75, bp2=76:200, bp3=201:500),
                max.coef = c(Inf,8,5), block1.penalization = TRUE,
                lambda.type = "lambda.min", standardize = TRUE, nfolds = 5, cvoffset = FALSE)
## Not run: 
  # cox
  # simulation of survival data:
  n &lt;- 50;p &lt;- 300
  nzc &lt;- trunc(p/10)
  x &lt;- matrix(rnorm(n*p), n, p)
  beta &lt;- rnorm(nzc)
  fx &lt;- x[, seq(nzc)]%*%beta/3
  hx &lt;- exp(fx)
  # survival times:
  ty &lt;- rexp(n,hx)
  # censoring indicator:
  tcens &lt;- rbinom(n = n,prob = .3,size = 1)
  library(survival)
  y &lt;- Surv(ty, 1-tcens)
  blocks &lt;- list(bp1=1:20, bp2=21:200, bp3=201:300)
  # run prioritylasso:
  prioritylasso(x, y, family = "cox", type.measure = "deviance", blocks = blocks,
                block1.penalization = TRUE, lambda.type = "lambda.min", standardize = TRUE,
                nfolds = 5)

  # binomial
  # using pl_data:
  prioritylasso(X = pl_data[,1:1028], Y = pl_data[,1029], family = "binomial", type.measure = "auc",
                blocks = list(bp1=1:4, bp2=5:9, bp3=10:28, bp4=29:1028), standardize = FALSE)
## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
