<!DOCTYPE html><html lang="en"><head><title>Help for package ProActive</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ProActive}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ProActive-package'><p><b>ProActive</b></p></a></li>
<li><a href='#changewindowSize'><p>Change the pileup window size</p></a></li>
<li><a href='#classifSumm'><p>Summarizes pattern-matching results</p></a></li>
<li><a href='#collectBestMatchInfo'><p>Collect information regarding the pattern-match</p></a></li>
<li><a href='#contigChunks'><p>'chunk' long contigs</p></a></li>
<li><a href='#elevOrGapClassif'><p>Classifies partial elevation/gap pattern-matches</p></a></li>
<li><a href='#fullElevGap'><p>Controller function for full elevation/gap pattern-matching</p></a></li>
<li><a href='#fullElevGapShrink'><p>Shrink the width of full elevation and gap patterns</p></a></li>
<li><a href='#geneAnnotationPlot'><p>Gene annotation plot</p></a></li>
<li><a href='#geneAnnotationSearch'><p>Search for gene annotations on classified contigs/chunks</p></a></li>
<li><a href='#genomeChunks'><p>'chunk' genomes</p></a></li>
<li><a href='#GPsInElevGaps'><p>Detect gene predictions in elevations and gaps</p></a></li>
<li><a href='#linkChunks'><p>Link pattern-matches on contig/genome chunks</p></a></li>
<li><a href='#noPattern'><p>No read coverage pattern</p></a></li>
<li><a href='#partialElevGap'><p>Controller function for partial elevation/gap pattern-matching</p></a></li>
<li><a href='#partialElevGapShrink'><p>Shrink the width of partial elevation and gap patterns</p></a></li>
<li><a href='#patternBuilder'><p>Builds pattern-match vectors</p></a></li>
<li><a href='#patternMatcher'><p>Controller function for pattern-matching</p></a></li>
<li><a href='#patternTranslator'><p>Full elevation/gap pattern translator</p></a></li>
<li><a href='#pileupFormatter'><p>Reformat input pileup file</p></a></li>
<li><a href='#plotProActiveResults'><p>Plot results of 'ProActive()' pattern-matching</p></a></li>
<li><a href='#ProActiveDetect'><p>Detect elevations and gaps in mapped read coverage patterns.</p></a></li>
<li><a href='#removeNoPatterns'><p>Removes 'NoPattern' classifications from best match list</p></a></li>
<li><a href='#sampleGenomegffTSV'><p>sampleGenomegffTSV</p></a></li>
<li><a href='#sampleGenomePileup'><p>sampleGenomePileup</p></a></li>
<li><a href='#sampleMetagenomegffTSV'><p>sampleMetagenomegffTSV</p></a></li>
<li><a href='#sampleMetagenomePileup'><p>sampleMetagenomePileup</p></a></li>
<li><a href='#sampleMetagenomeResults'><p>sampleMetagenomeResults</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Detect Elevations and Gaps in Mapped Sequencing Read Coverage</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jessie Maier &lt;jlmaier@ncsu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
  Automate the detection of gaps and elevations in mapped sequencing 
  read coverage using a 2D pattern-matching algorithm. 'ProActive' detects, characterizes 
  and visualizes read coverage patterns in both genomes and metagenomes. Optionally, 
  users may provide gene annotations associated with their genome or metagenome 
  in the form of a .gff file. In this case, 'ProActive' will generate an additional 
  output table containing the gene annotations found within the detected regions of gapped 
  and elevated read coverage. Additionally, users can search for gene 
  annotations of interest in the output read coverage plots.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jlmaier12/ProActive">https://github.com/jlmaier12/ProActive</a>,
<a href="https://jlmaier12.github.io/ProActive/">https://jlmaier12.github.io/ProActive/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jlmaier12/ProActive/issues">https://github.com/jlmaier12/ProActive/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, dplyr, ggplot2, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0), kableExtra</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-20 20:39:54 UTC; jlmaier</td>
</tr>
<tr>
<td>Author:</td>
<td>Jessie Maier <a href="https://orcid.org/0009-0001-8575-5386"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Manuel Kleiner <a href="https://orcid.org/0000-0001-6904-0287"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ths]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-21 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ProActive-package'><b>ProActive</b></h2><span id='topic+ProActive'></span><span id='topic+ProActive-package'></span>

<h3>Description</h3>

<p>'ProActive' automatically detects regions of gapped and elevated read coverage
using a 2D pattern-matching algorithm. 'ProActive' detects, characterizes and
visualizes read coverage patterns in both genomes and metagenomes. Optionally,
users may provide gene annotations associated with their genome or metagenome
in the form of a .gff file. In this case, 'ProActive' will generate an additional
output table containing the gene annotations found within the detected regions of
gapped and elevated read coverage. Additionally, users can search for gene
annotations of interest in the output read coverage plots.
</p>


<h3>Details</h3>

<p>The three main functions in 'ProActive' are:
</p>

<ol>
<li> <p><code><a href="#topic+ProActiveDetect">ProActiveDetect</a></code> performs the pattern-matching and characterization of read coverage patterns.
</p>
</li>
<li> <p><code><a href="#topic+plotProActiveResults">plotProActiveResults</a></code> plots the results from
<code>ProActiveDetect()</code>
</p>
</li>
<li> <p><code><a href="#topic+geneAnnotationSearch">geneAnnotationSearch</a></code> searches classified contigs/chunks for gene annotations that match
user-provided keywords.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Jessie Maier <a href="mailto:jlmaier@ncsu.edu">jlmaier@ncsu.edu</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/jlmaier12/ProActive">https://github.com/jlmaier12/ProActive</a>
</p>
</li>
<li> <p><a href="https://jlmaier12.github.io/ProActive/">https://jlmaier12.github.io/ProActive/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/jlmaier12/ProActive/issues">https://github.com/jlmaier12/ProActive/issues</a>
</p>
</li></ul>


<hr>
<h2 id='changewindowSize'>Change the pileup window size</h2><span id='topic+changewindowSize'></span>

<h3>Description</h3>

<p>Re-averages windows of pileup files with 100bp windows to reduce pileup size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>changewindowSize(pileupSubset, windowSize, mode)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="changewindowSize_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="changewindowSize_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="changewindowSize_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;</p>
</td></tr>
</table>

<hr>
<h2 id='classifSumm'>Summarizes pattern-matching results</h2><span id='topic+classifSumm'></span>

<h3>Description</h3>

<p>Summarizes the list of pattern-matching classifications into a table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classifSumm(pileup, bestMatchList, windowSize, mode, chunkSize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="classifSumm_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="classifSumm_+3A_bestmatchlist">bestMatchList</code></td>
<td>
<p>A list containing pattern-match information associated with
all contigs/chunks classified by 'ProActive()' pattern-matching</p>
</td></tr>
<tr><td><code id="classifSumm_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="classifSumm_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;</p>
</td></tr>
<tr><td><code id="classifSumm_+3A_chunksize">chunkSize</code></td>
<td>
<p>If 'mode'=&quot;genome&quot; OR if 'mode'=&quot;metagenome&quot; and 'chunkContigs'=TRUE,
chunk the genome or contigs, respectively, into smaller subsets for pattern-matching.</p>
</td></tr>
</table>

<hr>
<h2 id='collectBestMatchInfo'>Collect information regarding the pattern-match</h2><span id='topic+collectBestMatchInfo'></span>

<h3>Description</h3>

<p>Make a list containing the match-score, min and max pattern-match values,
the start and stop positions of the elevated or gapped region, the elevation
ratio and the classification
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collectBestMatchInfo(pattern, pileupSubset, elevOrGap, leftRightFull)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="collectBestMatchInfo_+3A_pattern">pattern</code></td>
<td>
<p>A vector containing the values associated with the pattern-match</p>
</td></tr>
<tr><td><code id="collectBestMatchInfo_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="collectBestMatchInfo_+3A_elevorgap">elevOrGap</code></td>
<td>
<p>Pattern-matching on 'elevation' or 'gap' pattern.</p>
</td></tr>
<tr><td><code id="collectBestMatchInfo_+3A_leftrightfull">leftRightFull</code></td>
<td>
<p>'Left' or'Right' partial gap/elevation pattern or full
elevation/gap pattern.</p>
</td></tr>
</table>

<hr>
<h2 id='contigChunks'>'chunk' long contigs</h2><span id='topic+contigChunks'></span>

<h3>Description</h3>

<p>Subset long contigs in metagenome pileup into chunks for pattern-matching
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contigChunks(pileup, chunkSize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contigChunks_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="contigChunks_+3A_chunksize">chunkSize</code></td>
<td>
<p>If 'mode'=&quot;genome&quot; OR if 'mode'=&quot;metagenome&quot; and 'chunkContigs'=TRUE,
chunk the genome or contigs, respectively, into smaller subsets for pattern-matching.
&lsquo;chunkSize' determines the size (in bp) of each &rsquo;chunk'. Default is 100000.</p>
</td></tr>
</table>

<hr>
<h2 id='elevOrGapClassif'>Classifies partial elevation/gap pattern-matches</h2><span id='topic+elevOrGapClassif'></span>

<h3>Description</h3>

<p>classify the contig/chunk as 'gap' if the elevated region is less than 50
the length of the contig/chunk and otherwise classify as 'elevation'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>elevOrGapClassif(bestMatchList, pileupSubset)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="elevOrGapClassif_+3A_bestmatchlist">bestMatchList</code></td>
<td>
<p>A list containing pattern-match information associated with
all contigs/chunks classified by 'ProActive()' pattern-matching</p>
</td></tr>
<tr><td><code id="elevOrGapClassif_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
</table>

<hr>
<h2 id='fullElevGap'>Controller function for full elevation/gap pattern-matching</h2><span id='topic+fullElevGap'></span>

<h3>Description</h3>

<p>Builds full elevation/gap pattern-matches, shrinks the width, and collects
best match information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fullElevGap(pileupSubset, windowSize, minSize, maxSize, elevOrGap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fullElevGap_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="fullElevGap_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="fullElevGap_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size (in bp) of elevation or gap patterns. Default is 10000.</p>
</td></tr>
<tr><td><code id="fullElevGap_+3A_maxsize">maxSize</code></td>
<td>
<p>The maximum size (in bp) of elevation or gap patterns. Default is NA
(i.e. no maximum).</p>
</td></tr>
<tr><td><code id="fullElevGap_+3A_elevorgap">elevOrGap</code></td>
<td>
<p>Pattern-matching on 'elevation' or 'gap' pattern.</p>
</td></tr>
</table>

<hr>
<h2 id='fullElevGapShrink'>Shrink the width of full elevation and gap patterns</h2><span id='topic+fullElevGapShrink'></span>

<h3>Description</h3>

<p>Remove values from gapped/elevated region in the pattern-match vector
until it reaches the 'minSize'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fullElevGapShrink(
  minCov,
  windowSize,
  maxCov,
  elevLength,
  nonElev,
  bestMatchInfo,
  pileupSubset,
  minSize,
  elevOrGap
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fullElevGapShrink_+3A_mincov">minCov</code></td>
<td>
<p>The minimum value of the pattern-match vector.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_maxcov">maxCov</code></td>
<td>
<p>The maximum value of the pattern-match vector.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_elevlength">elevLength</code></td>
<td>
<p>Length of the elevated/gapped pattern-match region.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_nonelev">nonElev</code></td>
<td>
<p>Length of the non-elevated/gapped pattern-match region.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_bestmatchinfo">bestMatchInfo</code></td>
<td>
<p>The information associated with the current best pattern-match
for the contig/chunk being assessed.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size (in bp) of elevation or gap patterns. Default is 10000.</p>
</td></tr>
<tr><td><code id="fullElevGapShrink_+3A_elevorgap">elevOrGap</code></td>
<td>
<p>Pattern-matching on 'elevation' or 'gap' pattern.</p>
</td></tr>
</table>

<hr>
<h2 id='geneAnnotationPlot'>Gene annotation plot</h2><span id='topic+geneAnnotationPlot'></span>

<h3>Description</h3>

<p>Plot read coverage and location of gene annotations that match the keywords and
search criteria for contig/chunk currently being assessed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneAnnotationPlot(
  geneAnnotSubset,
  keywords,
  pileupSubset,
  colIdx,
  startbpRange,
  endbpRange,
  elevRatio,
  pattern,
  windowSize,
  chunkSize,
  mode
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geneAnnotationPlot_+3A_geneannotsubset">geneAnnotSubset</code></td>
<td>
<p>Subset of gene annotations to be plotted</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_keywords">keywords</code></td>
<td>
<p>The key-word(s) used for the search.</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup associated with the contig/chunk being assessed</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_colidx">colIdx</code></td>
<td>
<p>The column index 'gene' or 'product' column</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_startbprange">startbpRange</code></td>
<td>
<p>The basepair at which the search is started if a 'specific' search is used</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_endbprange">endbpRange</code></td>
<td>
<p>The basepair at which the search is ended if a 'specific' search is used</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_elevratio">elevRatio</code></td>
<td>
<p>The maximum/minimum values of the pattern-match</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_pattern">pattern</code></td>
<td>
<p>The pattern-match information associated with the contig/chunk being assessed</p>
</td></tr>
<tr><td><code id="geneAnnotationPlot_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.</p>
</td></tr>
</table>

<hr>
<h2 id='geneAnnotationSearch'>Search for gene annotations on classified contigs/chunks</h2><span id='topic+geneAnnotationSearch'></span>

<h3>Description</h3>

<p>Search contigs classified with ProActive for gene-annotations that match a provided
key-word(s). Outputs read coverage plots for contigs/chunks with matching annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneAnnotationSearch(
  ProActiveResults,
  pileup,
  gffTSV,
  geneOrProduct,
  keyWords,
  inGapOrElev = FALSE,
  bpRange = 0,
  elevFilter,
  saveFilesTo,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geneAnnotationSearch_+3A_proactiveresults">ProActiveResults</code></td>
<td>
<p>The output from 'ProActive()'.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_gfftsv">gffTSV</code></td>
<td>
<p>A .gff file (TSV) containing gene predictions associated with the .fasta
file used to generate the pileup.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_geneorproduct">geneOrProduct</code></td>
<td>
<p>&quot;gene&quot; or &quot;product&quot;. Search for keyWords associated with genes or gene products.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_keywords">keyWords</code></td>
<td>
<p>The keyWord(s) to search for. Case independent. Searches will return the string
that contains the matching keyWord. KeyWord(s) must be in quotes, comma-separated, and surrounded by
c() i.e( c(&quot;antibiotic&quot;, &quot;resistance&quot;, &quot;drug&quot;) )</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_ingaporelev">inGapOrElev</code></td>
<td>
<p>TRUE or FALSE. If TRUE, only search for gene-annotations in
the gap/elevation region of the pattern-match. Default is FALSE (i.e search the
entire contig/chunk for the gene annotation key-words)</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_bprange">bpRange</code></td>
<td>
<p>If 'inGapOrElev' = TRUE, the user may specify the region (in base pairs) that should
be searched to the left and right of the gap/elevation region. Default is 0.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_elevfilter">elevFilter</code></td>
<td>
<p>Optional, only plot results with pattern-matches that achieved an
elevation ratio (max/min) greater than the specified values. Default is no filter.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_savefilesto">saveFilesTo</code></td>
<td>
<p>Optional, Provide a path to the directory you wish to save
output to. A folder will be made within the provided directory to store
results.</p>
</td></tr>
<tr><td><code id="geneAnnotationSearch_+3A_verbose">verbose</code></td>
<td>
<p>TRUE or FALSE. Print progress messages to console. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geneAnnotMatches &lt;- geneAnnotationSearch(sampleMetagenomeResults, sampleMetagenomePileup,
                                          sampleMetagenomegffTSV, geneOrProduct="product",
                                          keyWords=c("toxin", "drug", "resistance", "phage"))
</code></pre>

<hr>
<h2 id='genomeChunks'>'chunk' genomes</h2><span id='topic+genomeChunks'></span>

<h3>Description</h3>

<p>Subset genome pileup into chunks for pattern-matching
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomeChunks(pileup, chunkSize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomeChunks_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="genomeChunks_+3A_chunksize">chunkSize</code></td>
<td>
<p>If 'mode'=&quot;genome&quot; OR if 'mode'=&quot;metagenome&quot; and 'chunkContigs'=TRUE,
chunk the genome or contigs, respectively, into smaller subsets for pattern-matching.
&lsquo;chunkSize' determines the size (in bp) of each &rsquo;chunk'. Default is 50000.</p>
</td></tr>
</table>

<hr>
<h2 id='GPsInElevGaps'>Detect gene predictions in elevations and gaps</h2><span id='topic+GPsInElevGaps'></span>

<h3>Description</h3>

<p>Extracts subsets of the gffTSV associated with gene predictions that fall within regions
of detected gapped or elevated read coverage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GPsInElevGaps(
  elevGapSummList,
  windowSize,
  gffTSV,
  mode,
  chunkContigs,
  chunkSize
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GPsInElevGaps_+3A_elevgapsummlist">elevGapSummList</code></td>
<td>
<p>A list containing pattern-match information associated with all
elevation and gap classifications. (i.e. no NoPattern classifications)</p>
</td></tr>
<tr><td><code id="GPsInElevGaps_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="GPsInElevGaps_+3A_gfftsv">gffTSV</code></td>
<td>
<p>Optional, a .gff file (TSV) containing gene predictions associated with the .fasta
file used to generate the pileup.</p>
</td></tr>
<tr><td><code id="GPsInElevGaps_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;</p>
</td></tr>
<tr><td><code id="GPsInElevGaps_+3A_chunkcontigs">chunkContigs</code></td>
<td>
<p>TRUE or FALSE, If TRUE and 'mode'=&quot;metagenome&quot;, contigs longer
than the &lsquo;chunkSize' will be &rsquo;chunked' into smaller subsets and pattern-matching
will be performed on each subset. Default is FALSE.</p>
</td></tr>
<tr><td><code id="GPsInElevGaps_+3A_chunksize">chunkSize</code></td>
<td>
<p>If 'mode'=&quot;genome&quot; OR if 'mode'=&quot;metagenome&quot; and 'chunkContigs'=TRUE,
chunk the genome or contigs, respectively, into smaller subsets for pattern-matching.</p>
</td></tr>
</table>

<hr>
<h2 id='linkChunks'>Link pattern-matches on contig/genome chunks</h2><span id='topic+linkChunks'></span>

<h3>Description</h3>

<p>Detect partial gap/elevation pattern matches that fall on the edges of chunked
genomes/contigs that may be part of the same pattern prior to chunking
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linkChunks(bestMatchList, pileup, windowSize, mode, verbose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="linkChunks_+3A_bestmatchlist">bestMatchList</code></td>
<td>
<p>A list containing pattern-match information associated with
all contigs/chunks classified by 'ProActive()' pattern-matching</p>
</td></tr>
<tr><td><code id="linkChunks_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="linkChunks_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.</p>
</td></tr>
<tr><td><code id="linkChunks_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;</p>
</td></tr>
<tr><td><code id="linkChunks_+3A_verbose">verbose</code></td>
<td>
<p>TRUE or FALSE. Print progress messages to console. Default is TRUE.</p>
</td></tr>
</table>

<hr>
<h2 id='noPattern'>No read coverage pattern</h2><span id='topic+noPattern'></span>

<h3>Description</h3>

<p>Assess whether a contig/chunk does not have an elevated/gapped read coverage pattern.
A horizontal line at the mean or median coverage should be the best
match if the contig/chunk read coverage is not gapped or elevated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noPattern(pileupSubset)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="noPattern_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the read coverage dataset that pertains only to
the contig currently being assessed</p>
</td></tr>
</table>

<hr>
<h2 id='partialElevGap'>Controller function for partial elevation/gap pattern-matching</h2><span id='topic+partialElevGap'></span>

<h3>Description</h3>

<p>Builds partial elevation/gap pattern-match for patterns going off both the
left and right sides of the contig/chunk, shrinks the width, and collects
best match information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partialElevGap(pileupSubset, windowSize, minSize, maxSize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="partialElevGap_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="partialElevGap_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="partialElevGap_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size (in bp) of elevation or gap patterns. Default is 10000.</p>
</td></tr>
<tr><td><code id="partialElevGap_+3A_maxsize">maxSize</code></td>
<td>
<p>The maximum size (in bp) of elevation or gap patterns. Default is NA
(i.e. no maximum).</p>
</td></tr>
</table>

<hr>
<h2 id='partialElevGapShrink'>Shrink the width of partial elevation and gap patterns</h2><span id='topic+partialElevGapShrink'></span>

<h3>Description</h3>

<p>Remove values from gapped/elevated region in the pattern-match vector
until it reaches the 'minSize'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partialElevGapShrink(
  minCov,
  windowSize,
  maxCov,
  elevLength,
  nonElev,
  bestMatchInfo,
  pileupSubset,
  minSize,
  leftOrRight
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="partialElevGapShrink_+3A_mincov">minCov</code></td>
<td>
<p>The minimum value of the pattern-match vector.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_maxcov">maxCov</code></td>
<td>
<p>The maximum value of the pattern-match vector.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_elevlength">elevLength</code></td>
<td>
<p>Length of the elevated/gapped pattern-match region.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_nonelev">nonElev</code></td>
<td>
<p>Length of the non-elevated/gapped pattern-match region.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_bestmatchinfo">bestMatchInfo</code></td>
<td>
<p>The information associated with the current best pattern-match
for the contig/chunk being assessed.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size (in bp) of elevation or gap patterns. Default is 10000.</p>
</td></tr>
<tr><td><code id="partialElevGapShrink_+3A_leftorright">leftOrRight</code></td>
<td>
<p>'Left' or 'Right' partial gap/elevation pattern.</p>
</td></tr>
</table>

<hr>
<h2 id='patternBuilder'>Builds pattern-match vectors</h2><span id='topic+patternBuilder'></span>

<h3>Description</h3>

<p>Builds the pattern-match (vector) associated with each contig/chunk for
visualization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>patternBuilder(pileupSubset, bestMatchInfo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patternBuilder_+3A_pileupsubset">pileupSubset</code></td>
<td>
<p>A subset of the pileup that pertains only to the contig/chunk
currently being assessed.</p>
</td></tr>
<tr><td><code id="patternBuilder_+3A_bestmatchinfo">bestMatchInfo</code></td>
<td>
<p>The information associated with the current best pattern-match
for the contig/chunk being assessed.</p>
</td></tr>
</table>

<hr>
<h2 id='patternMatcher'>Controller function for pattern-matching</h2><span id='topic+patternMatcher'></span>

<h3>Description</h3>

<p>Creates the pileupSubset, representative of one contig/chunk, used as input for
each individual pattern-matching function. After the information associated with
the best match for each pattern is obtained, the pattern-match with the lowest
mean absolute difference (match-score) is used for classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>patternMatcher(
  pileup,
  windowSize,
  minSize,
  maxSize,
  mode,
  minContigLength,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patternMatcher_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="patternMatcher_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.</p>
</td></tr>
<tr><td><code id="patternMatcher_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size (in bp) of elevation or gap patterns. Default is 10000.</p>
</td></tr>
<tr><td><code id="patternMatcher_+3A_maxsize">maxSize</code></td>
<td>
<p>The maximum size (in bp) of elevation or gap patterns. Default is NA
(i.e. no maximum).</p>
</td></tr>
<tr><td><code id="patternMatcher_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;.</p>
</td></tr>
<tr><td><code id="patternMatcher_+3A_mincontiglength">minContigLength</code></td>
<td>
<p>The minimum contig/chunk size (in bp) to perform pattern-matching
on. Default is 25000.</p>
</td></tr>
<tr><td><code id="patternMatcher_+3A_verbose">verbose</code></td>
<td>
<p>TRUE or FALSE. Print progress messages to console. Default is TRUE.</p>
</td></tr>
</table>

<hr>
<h2 id='patternTranslator'>Full elevation/gap pattern translator</h2><span id='topic+patternTranslator'></span>

<h3>Description</h3>

<p>Translates full elevation/gap patterns across contigs/chunks 1000bp at a time.
Translation stops when the elevation pattern is 5000bp from the end of the contig/chunk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>patternTranslator(contigCov, bestMatchInfo, windowSize, pattern, elevOrGap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patternTranslator_+3A_contigcov">contigCov</code></td>
<td>
<p>The read coverages that pertain to the pileupSubset</p>
</td></tr>
<tr><td><code id="patternTranslator_+3A_bestmatchinfo">bestMatchInfo</code></td>
<td>
<p>The information associated with the current best pattern-match
for the contig/chunk being assessed.</p>
</td></tr>
<tr><td><code id="patternTranslator_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="patternTranslator_+3A_pattern">pattern</code></td>
<td>
<p>A vector containing the values associated with the pattern-match</p>
</td></tr>
<tr><td><code id="patternTranslator_+3A_elevorgap">elevOrGap</code></td>
<td>
<p>Pattern-matching on 'elevation' or 'gap' pattern.</p>
</td></tr>
</table>

<hr>
<h2 id='pileupFormatter'>Reformat input pileup file</h2><span id='topic+pileupFormatter'></span>

<h3>Description</h3>

<p>Place columns in correct order, clean accessions by removing text after white
space, and name columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pileupFormatter(pileup, mode)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pileupFormatter_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="pileupFormatter_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;</p>
</td></tr>
</table>

<hr>
<h2 id='plotProActiveResults'>Plot results of 'ProActive()' pattern-matching</h2><span id='topic+plotProActiveResults'></span>

<h3>Description</h3>

<p>Plot read coverage of contigs/chunks with detected gaps and elevations and their
associated pattern-match.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotProActiveResults(pileup, ProActiveResults, elevFilter, saveFilesTo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotProActiveResults_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="plotProActiveResults_+3A_proactiveresults">ProActiveResults</code></td>
<td>
<p>The output from 'ProActive()'.</p>
</td></tr>
<tr><td><code id="plotProActiveResults_+3A_elevfilter">elevFilter</code></td>
<td>
<p>Optional, only plot results with pattern-matches that achieved an
elevation ratio (max/min) greater than the specified values. Default is no filter.</p>
</td></tr>
<tr><td><code id="plotProActiveResults_+3A_savefilesto">saveFilesTo</code></td>
<td>
<p>Optional, Provide a path to the directory you wish to save
output to. A folder will be made within the provided directory to store
results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing ggplot objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ProActivePlots &lt;- plotProActiveResults(sampleMetagenomePileup,
                                       sampleMetagenomeResults)
</code></pre>

<hr>
<h2 id='ProActiveDetect'>Detect elevations and gaps in mapped read coverage patterns.</h2><span id='topic+ProActiveDetect'></span>

<h3>Description</h3>

<p>Performs read coverage pattern-matching and summarizes the results into a list.
The first list item summarizes the pattern-matching results. The second list
item is the 'cleaned' version of the summary table with all the 'noPattern'
classifications removed. (i.e were not filtered out). The third list item contains
the pattern-match information needed for pattern-match visualization with
'plotProActiveResults()'. The fourth list item is a table containing all the contigs that
were filtered out prior to pattern-matching. The fifth list item contains arguments used during
pattern-matching (windowSize, mode, chunkSize, chunkContigs). If the user provides a
gffTSV files, then the last list is a table consisting of ORFs found within
the detected gaps and elevations in read coverage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProActiveDetect(
  pileup,
  mode,
  gffTSV,
  windowSize = 1000,
  chunkContigs = FALSE,
  minSize = 10000,
  maxSize = Inf,
  minContigLength = 30000,
  chunkSize = 1e+05,
  IncludeNoPatterns = FALSE,
  verbose = TRUE,
  saveFilesTo
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ProActiveDetect_+3A_pileup">pileup</code></td>
<td>
<p>A .txt file containing mapped sequencing read coverages averaged over
100 bp windows/bins.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_mode">mode</code></td>
<td>
<p>Either &quot;genome&quot; or &quot;metagenome&quot;</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_gfftsv">gffTSV</code></td>
<td>
<p>Optional, a .gff file (TSV) containing gene predictions associated with the .fasta
file used to generate the pileup.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_windowsize">windowSize</code></td>
<td>
<p>The number of basepairs to average read coverage values over.
Options are 100, 200, 500, 1000 ONLY. Default is 1000.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_chunkcontigs">chunkContigs</code></td>
<td>
<p>TRUE or FALSE, If TRUE and 'mode'=&quot;metagenome&quot;, contigs longer
than the &lsquo;chunkSize' will be &rsquo;chunked' into smaller subsets and pattern-matching
will be performed on each subset. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size (in bp) of elevation or gap patterns. Default is 10000.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_maxsize">maxSize</code></td>
<td>
<p>The maximum size (in bp) of elevation or gap patterns. Default is NA
(i.e. no maximum).</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_mincontiglength">minContigLength</code></td>
<td>
<p>The minimum contig/chunk size (in bp) to perform pattern-matching
on. Default is 25000.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_chunksize">chunkSize</code></td>
<td>
<p>If 'mode'=&quot;genome&quot; OR if 'mode'=&quot;metagenome&quot; and 'chunkContigs'=TRUE,
chunk the genome or contigs, respectively, into smaller subsets for pattern-matching.
&lsquo;chunkSize' determines the size (in bp) of each &rsquo;chunk'. Default is 100000.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_includenopatterns">IncludeNoPatterns</code></td>
<td>
<p>TRUE or FALSE, If TRUE the noPattern pattern-matches will
be included in the ProActive PatternMatches output list. If you would like to visualize
the noPattern pattern-matches in 'plotProActiveResults()', this should be set to TRUE.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_verbose">verbose</code></td>
<td>
<p>TRUE or FALSE. Print progress messages to console. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ProActiveDetect_+3A_savefilesto">saveFilesTo</code></td>
<td>
<p>Optional, Provide a path to the directory you wish to save
output to. A folder will be made within the provided directory to store
results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing 6 objects described in the function description.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>metagenome_results &lt;- ProActiveDetect(
  pileup = sampleMetagenomePileup,
  mode = "metagenome",
  gffTSV = sampleMetagenomegffTSV
)
</code></pre>

<hr>
<h2 id='removeNoPatterns'>Removes 'NoPattern' classifications from best match list</h2><span id='topic+removeNoPatterns'></span>

<h3>Description</h3>

<p>Removes 'NoPattern' classifications from the list of pattern-match information
associated with the best pattern-matches for each contig/chunk
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeNoPatterns(bestMatchList)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="removeNoPatterns_+3A_bestmatchlist">bestMatchList</code></td>
<td>
<p>A list containing pattern-match information associated with
all contigs/chunks classified by 'ProActive()' pattern-matching</p>
</td></tr>
</table>

<hr>
<h2 id='sampleGenomegffTSV'>sampleGenomegffTSV</h2><span id='topic+sampleGenomegffTSV'></span>

<h3>Description</h3>

<p>Gene annotations associated with the genome in the sampleGenomePileup
Report...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleGenomegffTSV
</code></pre>


<h3>Format</h3>

<p>## 'sampleGenomegffTSV'
A data frame with 85,575 rows and 9 columns:
</p>

<dl>
<dt>V1</dt><dd><p>seqid</p>
</dd>
<dt>V2</dt><dd><p>source</p>
</dd>
<dt>V3</dt><dd><p>type</p>
</dd>
<dt>V4</dt><dd><p>start</p>
</dd>
<dt>V5</dt><dd><p>end</p>
</dd>
<dt>V6</dt><dd><p>score</p>
</dd>
<dt>V7</dt><dd><p>strand</p>
</dd>
<dt>V8</dt><dd><p>phase</p>
</dd>
<dt>V9</dt><dd><p>attributes</p>
</dd>
</dl>



<h3>Details</h3>

<p>This is a standard .gff file format. The .gff file was generated by
running PROKKA with default parameters on the *Salmonella enterica* LT2 genome
sequence  (NCBI RefSeq NC_003197.2) associated with the sampleGenomePileup in
the ProActive package.
</p>

<hr>
<h2 id='sampleGenomePileup'>sampleGenomePileup</h2><span id='topic+sampleGenomePileup'></span>

<h3>Description</h3>

<p>A pileup file generated during read mapping to the *Salmonella enterica* LT2 genome.
Report...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleGenomePileup
</code></pre>


<h3>Format</h3>

<p>## 'sampleGenomePileup'
A data frame with 48,575 rows and 4 columns:
</p>

<dl>
<dt>V1</dt><dd><p>Accession</p>
</dd>
<dt>V2</dt><dd><p>Mapped read coverage averaged over a 100 bp window size</p>
</dd>
<dt>V3</dt><dd><p>Starting position (bp) of each 100 bp window. Starts from 100.</p>
</dd>
<dt>V4</dt><dd><p>Starting position (bp) of each 100 bp window. Starts from 0.</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset was generated by extracting DNA from a culture of
*Salmonella enterica* LT2 (LT2) infected with phage P22. The DNA was shotgun
sequenced with Illumina (paired-end mode, 150 bp reads). The sequencing reads
were mapped to the LT2 reference genome (NCBI RefSeq NC_003197.2). The
bbmap.sh bincov parameter with covbinsize=100 was used to create a pileup file
with 100 bp windows.
</p>


<h3>Source</h3>

<p>&lt;https://pubmed.ncbi.nlm.nih.gov/25608871/&gt;
</p>

<hr>
<h2 id='sampleMetagenomegffTSV'>sampleMetagenomegffTSV</h2><span id='topic+sampleMetagenomegffTSV'></span>

<h3>Description</h3>

<p>A subset of gene annotations associated with the metagenome in the sampleMetagenomePileup
Report...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleMetagenomegffTSV
</code></pre>


<h3>Format</h3>

<p>## 'sampleMetagenomegffTSV'
A data frame with 467 rows and 9 columns:
</p>

<dl>
<dt>V1</dt><dd><p>seqid</p>
</dd>
<dt>V2</dt><dd><p>source</p>
</dd>
<dt>V3</dt><dd><p>type</p>
</dd>
<dt>V4</dt><dd><p>start</p>
</dd>
<dt>V5</dt><dd><p>end</p>
</dd>
<dt>V6</dt><dd><p>score</p>
</dd>
<dt>V7</dt><dd><p>strand</p>
</dd>
<dt>V8</dt><dd><p>phase</p>
</dd>
<dt>V9</dt><dd><p>attributes</p>
</dd>
</dl>



<h3>Details</h3>

<p>This is a standard .gff file format. The .gff file was generated by
running PROKKA with default parameters on the metagenome assembly
associated with the sampleMetagenomePileup in the ProActive package.
The gff was subset to only include the data associated with the
contigs in the sample data subset.
</p>


<h3>Source</h3>

<p>&lt;https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00935-5&gt;
</p>

<hr>
<h2 id='sampleMetagenomePileup'>sampleMetagenomePileup</h2><span id='topic+sampleMetagenomePileup'></span>

<h3>Description</h3>

<p>A subset of contigs from the raw whole-community fraction read coverage
pileup file generated during read mapping.
Report...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleMetagenomePileup
</code></pre>


<h3>Format</h3>

<p>## 'sampleMetagenomePileup'
A data frame with 4,604 rows and 4 columns:
</p>

<dl>
<dt>V1</dt><dd><p>Contig accession</p>
</dd>
<dt>V2</dt><dd><p>Mapped read coverage averaged over a 100 bp window size</p>
</dd>
<dt>V3</dt><dd><p>Starting position (bp) of each 100 bp window. Restarts from 0 at
the start of each new contig.</p>
</dd>
<dt>V4</dt><dd><p>Starting position (bp) of each 100 bp window. Does NOT restart at
the start of each new contig.</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset was generated from a conventional mouse fecal
homogenate. The whole-community extracted DNA was sequenced with Illumina
(paired-end mode, 150 bp reads) after which the metagenome was assembled. The
sequencing reads were mapped to the assembled contigs using BBMap. The
bbmap.sh bincov parameter with covbinsize=100 was used to create a pileup
file with 100 bp windows. A subset of 10 contigs from the pileup file were
selected for this sample dataset. The contigs were chosen because their
associated read coverage patterns exemplify ProActive's
pattern-matching and characterization functionality across classifications:
NODE_1911: elevation off left
NODE_1583: elevation off right
NODE_1884: gap off right
NODE_1255: gap off left
NODE_368: full gap
NODE_617: elevation full
NODE_1625: no pattern
</p>


<h3>Source</h3>

<p>&lt;https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00935-5&gt;
</p>

<hr>
<h2 id='sampleMetagenomeResults'>sampleMetagenomeResults</h2><span id='topic+sampleMetagenomeResults'></span>

<h3>Description</h3>

<p>Output of 'ProActiveDetect()'
Report...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleMetagenomeResults
</code></pre>


<h3>Format</h3>

<p>## 'sampleMetagenomeResults'
A list with 6 objects:
</p>

<dl>
<dt>SummaryTable</dt><dd><p>A table containing all pattern-matching classifications</p>
</dd>
<dt>CleanSummaryTable</dt><dd><p>A table containing only gap and elevation pattern-match
classifications (i.e. noPattern classifications removed)</p>
</dd>
<dt>PatternMatches</dt><dd><p>A list object containing information needed to visualize the
pattern-matches in 'plotProActiveResults()'</p>
</dd>
<dt>FilteredOut</dt><dd><p>A table containing contigs/chunks that were filtered out for being
too small or having too low read coverage</p>
</dd>
<dt>Arguments</dt><dd><p>A list object containing arguments used for pattern-matching (windowSize,
mode, chunkSize, chunkContigs)</p>
</dd>
<dt>GeneAnnotTable</dt><dd><p>A table containing gene predictions associated with elevated
or gapped regions in pattern-matches</p>
</dd>
</dl>



<h3>Details</h3>

<p>This data was generated by running 'ProActiveDetect()' on the sampleMetagenomePileup
and sampleMetagenomegffTSV with default parameters.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
