<!DOCTYPE html><html lang="en"><head><title>Help for package ProbeDeveloper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ProbeDeveloper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ProbeMake'><p>Make probes</p></a></li>
<li><a href='#samplefa'>
<p>sample data for target sequence region with class 'DNAStringSet'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Develop Hybridization Probes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-01-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Junhui Li</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junhui Li &lt;junhuili@cau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Hybridization probes for target sequences can be made based on melting temperature value calculated by R package 'TmCalculator' <a href="https://CRAN.R-project.org/package=TmCalculator">https://CRAN.R-project.org/package=TmCalculator</a> and methods extended from Beliveau, B. J.,(2018) &lt;<a href="https://doi.org/10.1073%2Fpnas.1714530115">doi:10.1073/pnas.1714530115</a>&gt;, and those hybridization probes can be used to capture specific target regions in fluorescence in situ hybridization and next generation sequence experiments.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>TmCalculator (&ge; 1.0.2),Biostrings(&ge; 2.12.0)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-30 23:47:20 UTC; junhuili</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-31 00:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ProbeMake'>Make probes</h2><span id='topic+ProbeMake'></span>

<h3>Description</h3>

<p>Probes are made with a FASTA-formatted input file containing the target sequence. User can specify the allowable ranges of probe length, percent GC content, and adjust melting temperature calculated using nearest neighbor thermodynamics or empirical formulas based on GC content. Candidate probe sequences passing all checks output in BED format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProbeMake(
  fafile,
  LN = 90,
  ln = 60,
  TM = 80,
  tm = 60,
  CG = 70,
  cg = 30,
  gap = 0,
  method = c("S2L", "L2S"),
  direction = c("3to5", "5to3"),
  prohibitseq = NULL,
  TmMethod = c("Tm_GC", "Tm_NN"),
  variant = c("Primer3Plus", "Chester1993", "QuikChange", "Schildkraut1965",
    "Wetmur1991_MELTING", "Wetmur1991_RNA", "Wetmur1991_RNA/DNA", "vonAhsen2001"),
  nn_table = c("DNA_NN4", "DNA_NN1", "DNA_NN2", "DNA_NN3", "RNA_NN1", "RNA_NN2",
    "RNA_NN3", "R_DNA_NN1"),
  tmm_table = "DNA_TMM1",
  imm_table = "DNA_IMM1",
  de_table = c("DNA_DE1", "RNA_DE1"),
  dnac1 = 25,
  dnac2 = 25,
  Na = 0,
  K = 0,
  Tris = 0,
  Mg = 0,
  dNTPs = 0,
  saltcorr = c("Schildkraut2010", "Wetmur1991", "SantaLucia1996", "SantaLucia1998-1",
    "SantaLucia1998-2", "Owczarzy2004", "Owczarzy2008"),
  DMSO = 0,
  fmd = 0,
  DMSOfactor = 0.75,
  fmdfactor = 0.65,
  fmdmethod = c("concentration", "molar")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ProbeMake_+3A_fafile">fafile</code></td>
<td>
<p>Input file with a FASTA format read by function readDNAStringSet in R package 'Biostrings'</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_ln">LN</code></td>
<td>
<p>The maximum allowed probe length, default is 90</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_ln">ln</code></td>
<td>
<p>The minimum allowed probe length, default is 60</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_tm">TM</code></td>
<td>
<p>The maximum allowed melting temperature, default is 80</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_tm">tm</code></td>
<td>
<p>The minimum allowed melting temperature, default is 60</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_cg">CG</code></td>
<td>
<p>The maximum allowed percent GC content, default is 70</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_cg">cg</code></td>
<td>
<p>The minimum allowed percent GC content, default is 30</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_gap">gap</code></td>
<td>
<p>The minimum gap between adjacent probes, default is 0</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_method">method</code></td>
<td>
<p>'S2L' is used to design probe extending from minimal length to the maximum until passing all checks, conversely 'L2S' make probe from maximal length to the minimum. Default is 'S2L'</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_direction">direction</code></td>
<td>
<p>Design probes from 3 to 5 end or from 5 to 3 end of target sequence, default is '3to5'</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_prohibitseq">prohibitseq</code></td>
<td>
<p>Prohibited sequence list, e.g prohibitseq=c(&quot;GGGGG&quot;,&quot;CCCCC&quot;), default is NULL</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_tmmethod">TmMethod</code></td>
<td>
<p>The method used to calculate Tm, 'Tm_NN' and 'Tm_GC' can be seleted</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_variant">variant</code></td>
<td>
<p>Empirical constants coefficient with 8 variant for 'Tm_GC' method: Chester1993, QuikChange, Schildkraut1965, Wetmur1991_MELTING, Wetmur1991_RNA, Wetmur1991_RNA/DNA, Primer3Plus and vonAhsen2001</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_nn_table">nn_table</code></td>
<td>
<p>Thermodynamic NN values, eight tables are implemented.
</p>
<p>For DNA/DNA hybridizations:
DNA_NN1,DNA_NN2,DNA_NN3,DNA_NN4
</p>
<p>For RNA/RNA hybridizations:
RNA_NN1,RNA_NN2,RNA_NN3
</p>
<p>For RNA/DNA hybridizations:
R_DNA_NN1</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_tmm_table">tmm_table</code></td>
<td>
<p>Thermodynamic values for terminal mismatches. Default: DNA_TMM1</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_imm_table">imm_table</code></td>
<td>
<p>Thermodynamic values for internal mismatches, may include insosine mismatches. Default: DNA_IMM1</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_de_table">de_table</code></td>
<td>
<p>Thermodynamic values for dangling ends: DNA_DE1(default),RNA_DE1</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_dnac1">dnac1</code></td>
<td>
<p>Concentration of the higher concentrated strand [nM]. Typically this will be the primer (for PCR) or the probe. Default: 25</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_dnac2">dnac2</code></td>
<td>
<p>Concentration of the lower concentrated strand [nM]. Default: 25</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_na">Na</code></td>
<td>
<p>Millimolar concentration of Na, default is 0</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_k">K</code></td>
<td>
<p>Millimolar concentration of K, default is 0</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_tris">Tris</code></td>
<td>
<p>Millimolar concentration of Tris, default is 0</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_mg">Mg</code></td>
<td>
<p>Millimolar concentration of Mg, default is 0</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_dntps">dNTPs</code></td>
<td>
<p>Millimolar concentration of dNTPs, default is 0</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_saltcorr">saltcorr</code></td>
<td>
<p>Salt correction method. Options are &quot;Schildkraut2010&quot;, &quot;Wetmur1991&quot;,&quot;SantaLucia1996&quot;,&quot;SantaLucia1998-1&quot;,&quot;Owczarzy2004&quot;,&quot;Owczarzy2008&quot;. Note that &quot;SantaLucia1998-2&quot; is not available for this function.</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_dmso">DMSO</code></td>
<td>
<p>Percent of DMSO</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_fmd">fmd</code></td>
<td>
<p>Formamide concentration in percentage (fmdmethod=&quot;concentration&quot;) or molar (fmdmethod=&quot;molar&quot;)</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_dmsofactor">DMSOfactor</code></td>
<td>
<p>Coeffecient of Tm decreases per percent DMSO. Default=0.75 von Ahsen N (2001) &lt;PMID:11673362&gt;. Other published values are 0.5, 0.6 and 0.675.</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_fmdfactor">fmdfactor</code></td>
<td>
<p>Coeffecient of Tm decrease per percent formamide. Default=0.65. Several papers report factors between 0.6 and 0.72.</p>
</td></tr>
<tr><td><code id="ProbeMake_+3A_fmdmethod">fmdmethod</code></td>
<td>
<p>&quot;concentration&quot; method for formamide concentration in percentage and &quot;molar&quot; for formamide concentration in molar</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a bed file in the format TargetID &lt;tab&gt; Chr &lt;tab&gt; Start &lt;tab&gt; End &lt;tab&gt; Sequence &lt;tab&gt; Tm &lt;tab&gt; GC
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>Beliveau B J, Kishi J Y, Nir G, et al. (2017). OligoMiner: A rapid, flexible environment for the design of genome-scale oligonucleotide in situ hybridization probes. bioRxiv.
</p>
<p>Breslauer K J , Frank R , Blocker H , et al. Predicting DNA duplex stability from the base sequence.[J]. Proceedings of the National Academy of Sciences, 1986, 83(11):3746-3750.
</p>
<p>Sugimoto N , Nakano S , Yoneyama M , et al. Improved Thermodynamic Parameters and Helix Initiation Factor to Predict Stability of DNA Duplexes[J]. Nucleic Acids Research, 1996, 24(22):4501-5.
</p>
<p>Allawi, H. Thermodynamics of internal C.T mismatches in DNA[J]. Nucleic Acids Research, 1998, 26(11):2694-2701.
</p>
<p>Hicks L D , Santalucia J . The thermodynamics of DNA structural motifs.[J]. Annual Review of Biophysics &amp; Biomolecular Structure, 2004, 33(1):415-440.
</p>
<p>Freier S M , Kierzek R , Jaeger J A , et al. Improved free-energy parameters for predictions of RNA duplex stability.[J]. Proceedings of the National Academy of Sciences, 1986, 83(24):9373-9377.
</p>
<p>Xia T , Santalucia , J , Burkard M E , et al. Thermodynamic Parameters for an Expanded Nearest-Neighbor Model for Formation of RNA Duplexes with Watson-Crick Base Pairs,[J]. Biochemistry, 1998, 37(42):14719-14735.
</p>
<p>Chen J L , Dishler A L , Kennedy S D , et al. Testing the Nearest Neighbor Model for Canonical RNA Base Pairs: Revision of GU Parameters[J]. Biochemistry, 2012, 51(16):3508-3522.
</p>
<p>Bommarito S, Peyret N, Jr S L. Thermodynamic parameters for DNA sequences with dangling ends[J]. Nucleic Acids Research, 2000, 28(9):1929-1934.
</p>
<p>Turner D H , Mathews D H . NNDB: the nearest neighbor parameter database for predicting stability of nucleic acid secondary structure[J]. Nucleic Acids Research, 2010, 38(Database issue):D280-D282.
</p>
<p>Sugimoto N , Nakano S I , Katoh M , et al. Thermodynamic Parameters To Predict Stability of RNA/DNA Hybrid Duplexes[J]. Biochemistry, 1995, 34(35):11211-11216.
</p>
<p>Allawi H, SantaLucia J: Thermodynamics and NMR of internal G-T mismatches in DNA. Biochemistry 1997, 36:10581-10594.
</p>
<p>Santalucia N E W J . Nearest-neighbor thermodynamics of deoxyinosine pairs in DNA duplexes[J]. Nucleic Acids Research, 2005, 33(19):6258-67.
</p>
<p>Peyret N , Seneviratne P A , Allawi H T , et al. Nearest-Neighbor Thermodynamics and NMR of DNA Sequences with Internal A-A, C-C, G-G, and T-T Mismatches, [J]. Biochemistry, 1999, 38(12):3468-3477.
</p>
<p>Marmur J , Doty P . Determination of the base composition of deoxyribonucleic acid from its thermal denaturation temperature.[J]. Journal of Molecular Biology, 1962, 5(1):109-118.
</p>
<p>Schildkraut C . Dependence of the melting temperature of DNA on salt concentration[J]. Biopolymers, 2010, 3(2):195-208.
</p>
<p>Wetmur J G . DNA Probes: Applications of the Principles of Nucleic Acid Hybridization[J]. CRC Critical Reviews in Biochemistry, 1991, 26(3-4):33.
</p>
<p>Untergasser A , Cutcutache I , Koressaar T , et al. Primer3&ndash;new capabilities and interfaces[J]. Nucleic Acids Research, 2012, 40(15):e115-e115.
</p>
<p>von Ahsen N, Wittwer CT, Schutz E , et al. Oligonucleotide melting temperatures under PCR conditions: deoxynucleotide Triphosphate and Dimethyl sulfoxide concentrations with comparison to alternative empirical formulas. Clin Chem 2001, 47:1956-1961.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(samplefa)
ProbeMake(samplefa,LN=90,ln=60,TM=80,tm=70,CG=80,cg=20,TmMethod="Tm_NN",Na=50)

</code></pre>

<hr>
<h2 id='samplefa'>
sample data for target sequence region with class 'DNAStringSet'
</h2><span id='topic+samplefa'></span>

<h3>Description</h3>

<p>sample data read by function readDNAStringSet in R package 'Biostrings' from fasta format file, there are two target sequence resion in this data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("samplefa")</code></pre>


<h3>Format</h3>

<p>Formal class 'DNAStringSet' [package &quot;Biostrings&quot;] with 5 slots
</p>

<p>sample data read by function readDNAStringSet in R package 'Biostrings' from fasta format file, which is from ncbiRefSeq database for Homo Sapiens with referece genome version hg19

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(samplefa)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
