<!DOCTYPE html><html><head><title>Help for package ProFAST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ProFAST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AddAdj'><p>Calculate the adjacency matrix given a spatial coordinate matrix</p></a></li>
<li><a href='#AddParSettingFAST'><p>Add FAST model settings for a PRECASTObj object</p></a></li>
<li><a href='#coembed_plot'><p>Coembedding dimensional reduction plot</p></a></li>
<li><a href='#coembedding_umap'><p>Calculate UMAP projections for coembedding of cells and features</p></a></li>
<li><a href='#CosMx_subset'><p>A Seurat object including spatial transcriptomics dataset from CosMx platform</p></a></li>
<li><a href='#diagnostic.cor.eigs'><p>Determine the dimension of low dimensional embedding</p></a></li>
<li><a href='#FAST'><p>Run FAST model for a PRECASTObj object</p></a></li>
<li><a href='#FAST_run'><p>(Varitional) ICM-EM algorithm for implementing FAST model</p></a></li>
<li><a href='#FAST_single'><p>Fit FAST model for single-section SRT data</p></a></li>
<li><a href='#FAST_structure'><p>(Varitional) ICM-EM algorithm for implementing FAST model with structurized parameters</p></a></li>
<li><a href='#find.signature.genes'><p>Find the signature genes for each group of cell/spots</p></a></li>
<li><a href='#get_r2_mcfadden'><p>Calcuate the the adjusted McFadden's pseudo R-square</p></a></li>
<li><a href='#get.top.signature.dat'><p>Obtain the top signature genes and related information</p></a></li>
<li><a href='#IntegrateSRTData'><p>Integrate multiple SRT data into a Seurat object</p></a></li>
<li><a href='#iscmeb_run'><p>Fit an iSC-MEB model using specified multi-section embeddings</p></a></li>
<li><a href='#model_set_FAST'><p>Set parameters for FAST model</p></a></li>
<li><a href='#NCFM'><p>Cell-feature coembedding for scRNA-seq data</p></a></li>
<li><a href='#NCFM_fast'><p>Cell-feature coembedding for SRT data</p></a></li>
<li><a href='#pbmc3k_subset'><p>A Seurat object including scRNA-seq PBMC dataset</p></a></li>
<li><a href='#pdistance'><p>Calculate the cell-feature distance matrix</p></a></li>
<li><a href='#RunHarmonyLouvain'><p>Embedding alignment and clustering based on the embeddings from FAST</p></a></li>
<li><a href='#RuniSCMEB'><p>Fit an iSC-MEB model using the embeddings from FAST</p></a></li>
<li><a href='#SelectHKgenes'><p>Select housekeeping genes</p></a></li>
<li><a href='#top5_signatures'><p>A data.frame object including top five signature genes in scRNA-seq PBMC dataset</p></a></li>
<li><a href='#transferGeneNames'><p>Transfer gene names from one fortmat to the other format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Probabilistic Factor Analysis for Spatially-Aware Dimension
Reduction</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Liu [aut, cre],
  Xiao Zhang [aut],
  Jin Liu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Liu &lt;liuweideng@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Probabilistic factor analysis for spatially-aware dimension reduction across multi-section spatial transcriptomics data with millions of spatial locations.
    More details can be referred to Wei Liu, et al. (2023) &lt;<a href="https://doi.org/10.1101%2F2023.07.11.548486">doi:10.1101/2023.07.11.548486</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), gtools</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.10), furrr, future, ggplot2, AnnotationDbi,
DR.SC, Matrix, mclust, PRECAST, pbapply, biomaRt, irlba,
org.Hs.eg.db, org.Mm.eg.db, Seurat, parallel, harmony, methods,
stats, utils</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/feiyoung/ProFAST">https://github.com/feiyoung/ProFAST</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/feiyoung/ProFAST/issues">https://github.com/feiyoung/ProFAST/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, performance, nnet, scater, ggrepel, RANN,
grDevices</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-18 01:11:07 UTC; Liuxianju</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-18 08:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='AddAdj'>Calculate the adjacency matrix given a spatial coordinate matrix</h2><span id='topic+AddAdj'></span>

<h3>Description</h3>

<p>Calculate the adjacency matrix given a spatial coordinate matrix with 2-dimension or 3-dimension or more.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddAdj(
  pos,
  type = "fixed_distance",
  platform = c("Others", "Visium", "ST"),
  neighbors = 6,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddAdj_+3A_pos">pos</code></td>
<td>
<p>a matrix object, with columns representing the spatial coordinates that can be any diemsion, i.e., 2, 3 and &gt;3.</p>
</td></tr>
<tr><td><code id="AddAdj_+3A_type">type</code></td>
<td>
<p>an optional string, specify which type of neighbors' definition. Here we provide two definition: one is &quot;fixed_distance&quot;, the other is &quot;fixed_number&quot;.</p>
</td></tr>
<tr><td><code id="AddAdj_+3A_platform">platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as &quot;Others&quot;. There are more platforms to be chosen, including &quot;Visuim&quot;, &quot;ST&quot; and &quot;Others&quot; (&quot;Others&quot; represents the other SRT platforms except for 'Visium' and 'ST')
The platform helps to calculate the adjacency matrix by defining the neighborhoods when type=&quot;fixed_distance&quot; is chosen.</p>
</td></tr>
<tr><td><code id="AddAdj_+3A_neighbors">neighbors</code></td>
<td>
<p>an optional postive integer,  specify how many neighbors used in calculation, default as 6.</p>
</td></tr>
<tr><td><code id="AddAdj_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>getAdj_auto</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the type = &quot;fixed_distance&quot;, then the spots within the Euclidean distance cutoffs from one spot are regarded as the neighbors of this spot. When the type = &quot;fixed_number&quot;, the K-nearest spots are regarded as the neighbors of each spot.
</p>


<h3>Value</h3>

<p>return a sparse matrix, representing the adjacency matrix.
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(CosMx_subset)
pos &lt;- as.matrix(CosMx_subset@meta.data[,c("x", "y")])
Adj_sp &lt;- AddAdj(pos)

</code></pre>

<hr>
<h2 id='AddParSettingFAST'>Add FAST model settings for a PRECASTObj object</h2><span id='topic+AddParSettingFAST'></span>

<h3>Description</h3>

<p>Add FAST model settings for a PRECASTObj object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddParSettingFAST(PRECASTObj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddParSettingFAST_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <code>CreatePRECASTObject</code>.</p>
</td></tr>
<tr><td><code id="AddParSettingFAST_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="#topic+model_set_FAST">model_set_FAST</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a revised PRECASTObj object with slot <code>parameterList</code> changed.
</p>


<h3>References</h3>

<p>None
</p>

<hr>
<h2 id='coembed_plot'>Coembedding dimensional reduction plot</h2><span id='topic+coembed_plot'></span>

<h3>Description</h3>

<p>Graph  output of a dimensional reduction technique on a 2D scatter plot where each point is a cell or feature and it's positioned based on the  coembeddings determined by the reduction technique. By default, cells and their signature features are colored by their identity class (can be changed with the group.by parameter).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coembed_plot(
  seu,
  reduction,
  gene_txtdata = NULL,
  cell_label = NULL,
  xy_name = reduction,
  dims = c(1, 2),
  cols = NULL,
  shape_cg = c(1, 5),
  pt_size = 1,
  pt_text_size = 5,
  base_size = 16,
  base_family = "serif",
  legend.point.size = 5,
  legend.key.size = 1.5,
  alpha = 0.3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coembed_plot_+3A_seu">seu</code></td>
<td>
<p>a Seurat object with coembedding in the reductions slot wiht component name reduction.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_reduction">reduction</code></td>
<td>
<p>a string, specify the reduction component that denotes coembedding.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_gene_txtdata">gene_txtdata</code></td>
<td>
<p>a data.frame object with columns indcluding 'gene' and 'label', specify the cell type/spatial domain and signature genes. Default as NULL, all features will be used in comebeddings.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_cell_label">cell_label</code></td>
<td>
<p>an optional character in columns of metadata,  specify the group of cells/spots. Default as NULL, use Idents as the group.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_xy_name">xy_name</code></td>
<td>
<p>an optional character,  specify the names of x and y-axis, default as the same as reduction.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_dims">dims</code></td>
<td>
<p>a postive integer vector with length 2, specify the two components for visualization.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_cols">cols</code></td>
<td>
<p>an optional string vector, specify the colors for cell group in visualization.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_shape_cg">shape_cg</code></td>
<td>
<p>a positive integers with length 2, specify the shapes of cell/spot and feature in plot.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_pt_size">pt_size</code></td>
<td>
<p>an optional integer, specify the point size, default as 1.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_pt_text_size">pt_text_size</code></td>
<td>
<p>an optional integer, specify the point size of text, default as 5.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_base_size">base_size</code></td>
<td>
<p>an optional integer, specify the basic size.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_base_family">base_family</code></td>
<td>
<p>an optional character, specify the font.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_legend.point.size">legend.point.size</code></td>
<td>
<p>an optional integer, specify the point size of legend.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_legend.key.size">legend.key.size</code></td>
<td>
<p>an optional integer, specify the size of legend key.</p>
</td></tr>
<tr><td><code id="coembed_plot_+3A_alpha">alpha</code></td>
<td>
<p>an optional positive real, range from 0 to 1, specify the transparancy of points.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>return a ggplot object
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coembedding_umap">coembedding_umap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pbmc3k_subset)
data(top5_signatures)
coembed_plot(pbmc3k_subset, reduction = "UMAPsig",
 gene_txtdata = top5_signatures,  pt_text_size = 3, alpha=0.3)

</code></pre>

<hr>
<h2 id='coembedding_umap'>Calculate UMAP projections for coembedding of cells and features</h2><span id='topic+coembedding_umap'></span>

<h3>Description</h3>

<p>Calculate UMAP projections for coembedding of cells and features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coembedding_umap(
  seu,
  reduction,
  reduction.name,
  gene.set = NULL,
  slot = "data",
  assay = "RNA",
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coembedding_umap_+3A_seu">seu</code></td>
<td>
<p>a Seurat object with coembedding in the reductions slot wiht component name reduction.</p>
</td></tr>
<tr><td><code id="coembedding_umap_+3A_reduction">reduction</code></td>
<td>
<p>a string, specify the reduction component that denotes coembedding.</p>
</td></tr>
<tr><td><code id="coembedding_umap_+3A_reduction.name">reduction.name</code></td>
<td>
<p>a string, specify the reduction name for the obtained UMAP projection.</p>
</td></tr>
<tr><td><code id="coembedding_umap_+3A_gene.set">gene.set</code></td>
<td>
<p>a string vector,  specify the features (genes) in calculating the UMAP projection, default as all features.</p>
</td></tr>
<tr><td><code id="coembedding_umap_+3A_slot">slot</code></td>
<td>
<p>an optional string,  specify the slot in the assay, default as 'data'.</p>
</td></tr>
<tr><td><code id="coembedding_umap_+3A_assay">assay</code></td>
<td>
<p>an optional string,  specify the assay name in the Seurat object when adding the UMAP projection.</p>
</td></tr>
<tr><td><code id="coembedding_umap_+3A_seed">seed</code></td>
<td>
<p>an optional integer, specify the random seed for reproducibility.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>return a revised Seurat object by adding a new reduction component named 'reduction.name'.
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pbmc3k_subset)
data(top5_signatures)

pbmc3k_subset &lt;- coembedding_umap(
  pbmc3k_subset, reduction = "ncfm", reduction.name = "UMAPsig",
  gene.set = top5_signatures$gene
)



</code></pre>

<hr>
<h2 id='CosMx_subset'>A Seurat object including spatial transcriptomics dataset from CosMx platform</h2><span id='topic+CosMx_subset'></span>

<h3>Description</h3>

  
<p>This data is a subset of  SCLC CosMx spatial transcriptomics  dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  
  data(CosMx_subset)  
</code></pre>


<h3>Format</h3>

  
<p>A Seurat object, including count matrix, sptial coordinates, and manual annotation.
</p>


<h3>Source</h3>

  
<p>The data is from the CosMx SRT sequencing platform.
</p>


<h3>References</h3>

  
<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  # Show some examples of how to use the dataset.  
  data(CosMx_subset)  
  library(Seurat)
  CosMx_subset
</code></pre>

<hr>
<h2 id='diagnostic.cor.eigs'>Determine the dimension of low dimensional embedding</h2><span id='topic+diagnostic.cor.eigs'></span><span id='topic+diagnostic.cor.eigs.default'></span><span id='topic+diagnostic.cor.eigs.Seurat'></span>

<h3>Description</h3>

<p>This function estimate the dimension of low dimensional embedding for a given cell by gene expression matrix. For more details, see Franklin et al. (1995) and Crawford et al. (2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnostic.cor.eigs(object, ...)

## Default S3 method:
diagnostic.cor.eigs(
  object,
  q_max = 50,
  plot = TRUE,
  n.sims = 10,
  parallel = TRUE,
  ncores = 10,
  seed = 1,
  ...
)

## S3 method for class 'Seurat'
diagnostic.cor.eigs(
  object,
  assay = NULL,
  slot = "data",
  nfeatures = 2000,
  q_max = 50,
  seed = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnostic.cor.eigs_+3A_object">object</code></td>
<td>
<p>A Seurat or matrix object</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+diagnostic.cor.eigs.default">diagnostic.cor.eigs.default</a></code>.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_q_max">q_max</code></td>
<td>
<p>the upper bound of low dimensional embedding. Default is 50.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_plot">plot</code></td>
<td>
<p>a indicator of whether plot eigen values.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_n.sims">n.sims</code></td>
<td>
<p>number of simulaton times. Default is 10.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_parallel">parallel</code></td>
<td>
<p>a indicator of whether use parallel analysis.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_ncores">ncores</code></td>
<td>
<p>the number of cores used in parallel analysis. Default is 10.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_seed">seed</code></td>
<td>
<p>a postive integer, specify the random seed for reproducibility</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_assay">assay</code></td>
<td>
<p>an optional string, specify the name of assay in the Seurat object to be used.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_slot">slot</code></td>
<td>
<p>an optional string, specify the name of slot.</p>
</td></tr>
<tr><td><code id="diagnostic.cor.eigs_+3A_nfeatures">nfeatures</code></td>
<td>
<p>an optional integer, specify the number of features to select as top variable features. Default is 2000.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with attribute 'q_est' and 'plot', which is the estimated dimension of low dimensional embedding. In addition, this data.frame containing the following components:
</p>

<ul>
<li><p> q - The index of eigen values.
</p>
</li>
<li><p> eig_value - The eigen values on observed data.
</p>
</li>
<li><p> eig_sim - The mean value of eigen values of n.sims simulated data.
</p>
</li>
<li><p> q_est - The selected dimension in attr(obj, 'q_est').
</p>
</li>
<li><p> plot - The plot saved in attr(obj, 'plot').
</p>
</li></ul>



<h3>References</h3>

<p>1. Franklin, S. B., Gibson, D. J., Robertson, P. A., Pohlmann, J. T., &amp; Fralish, J. S. (1995). Parallel analysis: a method for determining significant principal components. Journal of Vegetation Science, 6(1), 99-106.
</p>
<p>2. Crawford, A. V., Green, S. B., Levy, R., Lo, W. J., Scott, L., Svetina, D., &amp; Thompson, M. S. (2010). Evaluation of parallel analysis methods for determining the number of factors.Educational and Psychological Measurement, 70(6), 885-901.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 100
p &lt;- 50
d &lt;- 15
object &lt;- matrix(rnorm(n*d), n, d) %*% matrix(rnorm(d*p), d, p)
diagnostic.cor.eigs(object, n.sims=2)
</code></pre>

<hr>
<h2 id='FAST'>Run FAST model for a PRECASTObj object</h2><span id='topic+FAST'></span>

<h3>Description</h3>

<p>Run FAST model for a PRECASTObj object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FAST(PRECASTObj, q = 15, fit.model = c("poisson", "gaussian"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FAST_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <code>CreatePRECASTObject</code>.</p>
</td></tr>
<tr><td><code id="FAST_+3A_q">q</code></td>
<td>
<p>an optional integer, specify the number of low-dimensional embeddings to extract in FAST</p>
</td></tr>
<tr><td><code id="FAST_+3A_fit.model">fit.model</code></td>
<td>
<p>an optional string, specify the version of FAST to be fitted. The Gaussian version models the log-count matrices while the Poisson verions models the count matrices; default as poisson.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a revised PRECASTObj object with slot <code>PRECASTObj@resList</code> added by a <code>FAST</code> compoonent.
</p>


<h3>References</h3>

<p>None
</p>

<hr>
<h2 id='FAST_run'>(Varitional) ICM-EM algorithm for implementing FAST model</h2><span id='topic+FAST_run'></span>

<h3>Description</h3>

<p>(Varitional) ICM-EM algorithm for implementing FAST model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FAST_run(
  XList,
  AdjList,
  q = 15,
  fit.model = c("gaussian", "poisson"),
  AList = NULL,
  maxIter = 25,
  epsLogLik = 1e-05,
  verbose = TRUE,
  seed = 1,
  error_heter = TRUE,
  Psi_diag = FALSE,
  Vint_zero = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FAST_run_+3A_xlist">XList</code></td>
<td>
<p>an M-length list consisting of multiple matrices with class <code>dgCMatrix</code> or <code>matrix</code> that specifies the count/log-count gene expression matrix for each data batch used for FAST model.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_adjlist">AdjList</code></td>
<td>
<p>an M-length list of sparse matrices with class <code>dgCMatrix</code>, specify the adjacency matrix used for intrisic CAR model in FAST. We provide this interface for those users who would like to define the adjacency matrix by themselves.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_q">q</code></td>
<td>
<p>an optional integer, specify the number of low-dimensional embeddings to extract in FAST. Larger q means more information extracted.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_fit.model">fit.model</code></td>
<td>
<p>an optional string, specify the version of FAST to be fitted. The Gaussian version models the log-count matrices while the Poisson verions models the count matrices; default as <code>gaussian</code> due to fastter computation.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_alist">AList</code></td>
<td>
<p>an optional list with each component being a vector whose length is equal to the rows of component in <code>XList</code>, specify the normalization factor in FAST. The default is <code>NULL</code> that means the normalization factor equal to 1.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_maxiter">maxIter</code></td>
<td>
<p>the maximum iteration of ICM-EM algorithm. The default is 30.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_epsloglik">epsLogLik</code></td>
<td>
<p>an optional positive vlaue, tolerance of relative variation rate of the observed pseudo loglikelihood value, defualt as '1e-5'.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_verbose">verbose</code></td>
<td>
<p>a logical value, whether output the information in iteration.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_seed">seed</code></td>
<td>
<p>a postive integer, the random seed to be set in initialization.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_error_heter">error_heter</code></td>
<td>
<p>a logical value, whether use the heterogenous error for FAST model, default as <code>TRUE</code>. If <code>error.heter=FALSE</code>, then the homogenuous error is used.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_psi_diag">Psi_diag</code></td>
<td>
<p>a logical value, whether set the conditional covariance matrix of the intrisic CAR to diagonal, default as <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="FAST_run_+3A_vint_zero">Vint_zero</code></td>
<td>
<p>an optional logical value, specify whether the intial value of intrisic CAR component is set to zero; default as <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>return a list including the following components: (1) hV: an M-length list consisting of spatial embeddings in FAST; (2) nu: the estimated intercept vector; (3) Psi: the estimated covariance matrix; (4) W: the estimated shared loading matrix; (5) Lam: the estimated covariance matrix of error term; (6): ELBO: the ELBO value when algorithm convergence; (7) ELBO_seq: the ELBO values for all itrations.
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FAST_structure">FAST_structure</a></code>, <code><a href="#topic+FAST">FAST</a></code>, <code><a href="#topic+model_set_FAST">model_set_FAST</a></code>
</p>

<hr>
<h2 id='FAST_single'>Fit FAST model for single-section SRT data</h2><span id='topic+FAST_single'></span>

<h3>Description</h3>

<p>Fit FAST model for single-section SRT data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FAST_single(
  seu,
  Adj_sp,
  q = 15,
  fit.model = c("poisson", "gaussian"),
  slot = "data",
  assay = NULL,
  reduction.name = "fast",
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FAST_single_+3A_seu">seu</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_adj_sp">Adj_sp</code></td>
<td>
<p>a sparse matrix, specify the adjacency matrix among spots.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_q">q</code></td>
<td>
<p>an optional integer, specify the number of low-dimensional embeddings to extract in FAST. Larger q means more information extracted.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_fit.model">fit.model</code></td>
<td>
<p>an optional string, specify the version of FAST to be fitted. The Gaussian version models the log-count matrices while the Poisson verions models the count matrices; default as possion model.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_slot">slot</code></td>
<td>
<p>an optional string, specify the slot in Seurat object as the input of FAST model, default as 'data'.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_assay">assay</code></td>
<td>
<p>an optional string, specify the assay in Seurat object, default as 'NULL' that means the default assay in Seurat object.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_reduction.name">reduction.name</code></td>
<td>
<p>an optional string, specify the reduction name for the fast embedding, default as 'fast'.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_verbose">verbose</code></td>
<td>
<p>a logical value, whether output the information in iteration.</p>
</td></tr>
<tr><td><code id="FAST_single_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="#topic+FAST_run">FAST_run</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return a list including the parameters set in the arguments.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FAST_run">FAST_run</a></code>
</p>

<hr>
<h2 id='FAST_structure'>(Varitional) ICM-EM algorithm for implementing FAST model with structurized parameters</h2><span id='topic+FAST_structure'></span>

<h3>Description</h3>

<p>(Varitional) ICM-EM algorithm for implementing FAST model with structurized parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FAST_structure(
  XList,
  AdjList,
  q = 15,
  fit.model = c("poisson", "gaussian"),
  parameterList = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FAST_structure_+3A_xlist">XList</code></td>
<td>
<p>an M-length list consisting of multiple matrices with class dgCMatrix or matrix that specify the count/log-count gene expression matrix for each data batch used for FAST model.</p>
</td></tr>
<tr><td><code id="FAST_structure_+3A_adjlist">AdjList</code></td>
<td>
<p>an M-length list of sparse matrices with class dgCMatrix, specify the adjacency matrix used for intrisic CAR model in FAST. We provide this interface for those users who would like to define the adjacency matrix by themselves.</p>
</td></tr>
<tr><td><code id="FAST_structure_+3A_q">q</code></td>
<td>
<p>an optional integer, specify the number of low-dimensional embeddings to extract in FAST</p>
</td></tr>
<tr><td><code id="FAST_structure_+3A_fit.model">fit.model</code></td>
<td>
<p>an optional string, specify the version of FAST to be fitted. The Gaussian version models the log-count matrices while the Poisson verions models the count matrices; default as gaussian due to fastter computation.</p>
</td></tr>
<tr><td><code id="FAST_structure_+3A_parameterlist">parameterList</code></td>
<td>
<p>an optional list, specify other parameters in FAST model; see <code><a href="#topic+model_set_FAST">model_set_FAST</a></code> for other paramters. The default is <code>NULL</code> that means the default parameters produced by <code><a href="#topic+model_set_FAST">model_set_FAST</a></code> is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>return a list including the following components: (1) hV: an M-length list consisting of spatial embeddings in FAST; (2) nu: the estimated intercept vector; (3) Psi: the estimated covariance matrix; (4) W: the estimated shared loading matrix; (5) Lam: the estimated covariance matrix of error term; (6): ELBO: the ELBO value when algorithm convergence; (7) ELBO_seq: the ELBO values for all itrations.
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FAST_run">FAST_run</a></code>, <code><a href="#topic+FAST">FAST</a></code>, <code><a href="#topic+model_set_FAST">model_set_FAST</a></code>
</p>

<hr>
<h2 id='find.signature.genes'>Find the signature genes for each group of cell/spots</h2><span id='topic+find.signature.genes'></span>

<h3>Description</h3>

<p>Find the signature genes for each group of cell/spots based on coembedding distance and expression ratio.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.signature.genes(
  seu,
  distce.assay = "distce",
  ident = NULL,
  expr.prop.cutoff = 0.1,
  assay = NULL,
  genes.use = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.signature.genes_+3A_seu">seu</code></td>
<td>
<p>a Seurat object with coembedding in the reductions slot wiht component name reduction.</p>
</td></tr>
<tr><td><code id="find.signature.genes_+3A_distce.assay">distce.assay</code></td>
<td>
<p>an optional character, specify the assay name that constains distance matrix beween cells/spots and features, default as 'distce' (distance of coembeddings).</p>
</td></tr>
<tr><td><code id="find.signature.genes_+3A_ident">ident</code></td>
<td>
<p>an optional character in columns of metadata,  specify the group of cells/spots. Default as NULL, use Idents as the group.</p>
</td></tr>
<tr><td><code id="find.signature.genes_+3A_expr.prop.cutoff">expr.prop.cutoff</code></td>
<td>
<p>an optional postive real ranging from 0 to 1,  specify cutoff of expression proportion of  features, default as 0.1.</p>
</td></tr>
<tr><td><code id="find.signature.genes_+3A_assay">assay</code></td>
<td>
<p>an optional character,  specify the assay in seu, default as NULL, representing the default assay in seu.</p>
</td></tr>
<tr><td><code id="find.signature.genes_+3A_genes.use">genes.use</code></td>
<td>
<p>an optional string vector, specify genes as the signature candidates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In each data.frame object of the returned value, the row.names are gene names, and these genes are sorted by decreasing order of 'distance'. User can define the signature genes as top n genes in distance and that the 'expr.prop' larger than a cutoff. We set the cutoff as 0.1.
</p>


<h3>Value</h3>

<p>return a list with each component a data.frame object having two columns: 'distance' and 'expr.prop'.
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(pbmc3k_subset)
pbmc3k_subset &lt;- pdistance(pbmc3k_subset, reduction='ncfm')
df_list_rna &lt;- find.signature.genes(pbmc3k_subset)

</code></pre>

<hr>
<h2 id='get_r2_mcfadden'>Calcuate the the adjusted McFadden's pseudo R-square</h2><span id='topic+get_r2_mcfadden'></span>

<h3>Description</h3>

<p>Calcuate the the adjusted McFadden's pseudo R-square  between the  embeddings and the  labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_r2_mcfadden(embeds, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_r2_mcfadden_+3A_embeds">embeds</code></td>
<td>
<p>a n-by-q matrix, specify the embedding matrix.</p>
</td></tr>
<tr><td><code id="get_r2_mcfadden_+3A_y">y</code></td>
<td>
<p>a n-length vector, specify the labels.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>return the adjusted McFadden's pseudo R-square.
</p>


<h3>References</h3>

<p>McFadden, D. (1987). Regression-based specification tests for the multinomial logit model. Journal of econometrics, 34(1-2), 63-82.
</p>

<hr>
<h2 id='get.top.signature.dat'>Obtain the top signature genes and related information</h2><span id='topic+get.top.signature.dat'></span>

<h3>Description</h3>

<p>Obtain the top signature genes and related information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.top.signature.dat(df.list, ntop = 5, expr.prop.cutoff = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.top.signature.dat_+3A_df.list">df.list</code></td>
<td>
<p>a list that is obtained by the function <code><a href="#topic+find.signature.genes">find.signature.genes</a></code>.</p>
</td></tr>
<tr><td><code id="get.top.signature.dat_+3A_ntop">ntop</code></td>
<td>
<p>an optional positive integer, specify the how many top signature genes extracted, default as 5.</p>
</td></tr>
<tr><td><code id="get.top.signature.dat_+3A_expr.prop.cutoff">expr.prop.cutoff</code></td>
<td>
<p>an optional postive real ranging from 0 to 1,  specify cutoff of expression proportion of  features, default as 0.1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Using this funciton, we obtain the top signature genes and organize them into a data.frame. The 'row.names' are gene names.
The colname 'distance' means the distance between gene (i.e., VPREB3) and cells with the specific cell type (i.e., B cell),
which is calculated based on the coembedding of genes and cells in the coembedding space. The distance is smaller, the association between gene and the cell type is stronger. 
The colname 'expr.prop' represents the expression proportion of the gene (i.e., VPREB3) within the cell type (i.e., B cell). 
The colname 'label' means the cell types and colname 'gene' denotes the gene name. 
By the data.frame object, we know 'VPREB3' is the one of the top signature gene of B cell.
</p>


<h3>Value</h3>

<p>return  a 'data.frame' object with four columns: 'distance','expr.prop', 'label' and 'gene'.
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(pbmc3k_subset)
pbmc3k_subset &lt;- pdistance(pbmc3k_subset, reduction='ncfm')
df_list_rna &lt;- find.signature.genes(pbmc3k_subset)
dat.sig &lt;- get.top.signature.dat(df_list_rna, ntop=5)
head(dat.sig)
</code></pre>

<hr>
<h2 id='IntegrateSRTData'>Integrate multiple SRT data into a Seurat object</h2><span id='topic+IntegrateSRTData'></span>

<h3>Description</h3>

<p>Integrate multiple SRT data based on the <code>PRECASTObj</code> object by FAST and other model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IntegrateSRTData(
  PRECASTObj,
  seulist_HK,
  Method = c("iSC-MEB", "HarmonyLouvain"),
  seuList_raw = NULL,
  covariates_use = NULL,
  Tm = NULL,
  subsample_rate = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IntegrateSRTData_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <code>CreatePRECASTObject</code>.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_seulist_hk">seulist_HK</code></td>
<td>
<p>a list with Seurat object as component including only the housekeeping genes.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_method">Method</code></td>
<td>
<p>a string, specify the method to be used and two methods are supprted: <code>iSC-MEB</code> and <code>HarmonyLouvain</code>. The default is <code>iSC-MEB</code>.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_seulist_raw">seuList_raw</code></td>
<td>
<p>an optional list with Seurat object, the raw data.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_covariates_use">covariates_use</code></td>
<td>
<p>a string vector, the colnames in <code>PRECASTObj@seulist[[1]]@meta.data</code>, representing other biological covariates to considered when removing batch effects. This is achieved by adding additional covariates for biological conditions in the regression, such as case or control. Default as 'NULL', denoting no other covariates to be considered.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_tm">Tm</code></td>
<td>
<p>an optional numeric vector with the length equal to <code>PRECASTObj@seulist</code>, the time point information if the data include the temporal information. Default as <code>NULL</code> that means there is no temporal information.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_subsample_rate">subsample_rate</code></td>
<td>
<p>a real ranging in (0,1], specify the rate of spot drawing for speeding up the computation when the number of spots is very large. Default is 1, meaing using all spots.</p>
</td></tr>
<tr><td><code id="IntegrateSRTData_+3A_verbose">verbose</code></td>
<td>
<p>an optional logical value, default as <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>seuList_raw</code> is not equal <code>NULL</code> or <code>PRECASTObj@seuList</code> is not <code>NULL</code>, this function will remove the unwanted variations for all genes in <code>seuList_raw</code> object. Otherwise, only the the unwanted variation of genes in <code>PRECASTObj@seulist</code> will be removed. The former requires a big memory to be run, while the latter not. To speed up the computation when the number of spots is very large, we also provide a subsampling schema controlled by the arugment <code>subsample_rate</code>. When the total number of spots is larger than 80,000, this function will automatically draws 50,000 spots to calculate the paramters in the spatial linear model for removing unwanted variations.
</p>


<h3>Value</h3>

<p>Return a Seurat object by integrating all SRT data batches into a SRT data, where the column &quot;batch&quot; in the meta.data represents the batch ID, and the column &quot;cluster&quot; represents the clusters. The embeddings are put in <code>seu@reductions</code> slot and <code>Idents(seu)</code> is set to cluster label. Note that only the normalized expression is valid in the data slot while count is invalid.
</p>

<hr>
<h2 id='iscmeb_run'>Fit an iSC-MEB model using specified multi-section embeddings</h2><span id='topic+iscmeb_run'></span>

<h3>Description</h3>

<p>Integrate multiple SRT data based on the PRECASTObj by FAST and iSC-MEB model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iscmeb_run(
  VList,
  AdjList,
  K,
  beta_grid = seq(0, 5, by = 0.2),
  maxIter = 25,
  epsLogLik = 1e-05,
  verbose = TRUE,
  int.model = "EEE",
  init.start = 1,
  Sigma_equal = FALSE,
  Sigma_diag = TRUE,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iscmeb_run_+3A_vlist">VList</code></td>
<td>
<p>a M-length list of embeddings. The i-th element is a ni * q matrtix, where ni is the number of spots of sample i, and q is the number of embeddings. We provide this interface for those users who would like to define the embeddings by themselves.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_adjlist">AdjList</code></td>
<td>
<p>an M-length list of sparse matrices with class <code>dgCMatrix</code>, specify the adjacency matrix used for intrisic CAR model in FAST. We provide this interface for those users who would like to define the adjacency matrix by themselves.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_k">K</code></td>
<td>
<p>an integer, specify the number of clusters.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_beta_grid">beta_grid</code></td>
<td>
<p>an optional vector of positive value, the candidate set of the smoothing parameter to be searched by the grid-search optimization approach, defualt as a sequence starts from 0, ends with 5, increase by 0.2.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_maxiter">maxIter</code></td>
<td>
<p>the maximum iteration of ICM-EM algorithm. The default is 25.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_epsloglik">epsLogLik</code></td>
<td>
<p>a string, the species, one of 'Human' and 'Mouse'.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_verbose">verbose</code></td>
<td>
<p>an optional intger, spcify the number of housekeeping genes to be selected.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_int.model">int.model</code></td>
<td>
<p>an optional string, specify which Gaussian mixture model is used in evaluting the initial values for iSC.MEB, default as &quot;EEE&quot;; and see <code>Mclust</code> for more models' names.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_init.start">init.start</code></td>
<td>
<p>an optional number of times to calculate the initial value (1 by default). When init.start is larger than 1, initial value will be determined by log likelihood of mclust results.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_sigma_equal">Sigma_equal</code></td>
<td>
<p>an optional logical value, specify whether Sigmaks are equal, default as <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_sigma_diag">Sigma_diag</code></td>
<td>
<p>an optional logical value, specify whether Sigmaks are diagonal matrices, default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="iscmeb_run_+3A_seed">seed</code></td>
<td>
<p>an optional integer, the random seed in fitting iSC-MEB model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a iSCMEBResObj object which contains all model results.
</p>

<hr>
<h2 id='model_set_FAST'>Set parameters for FAST model</h2><span id='topic+model_set_FAST'></span>

<h3>Description</h3>

<p>Prepare parameters setup for FAST model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model_set_FAST(
  maxIter = 30,
  epsLogLik = 1e-05,
  error_heter = TRUE,
  Psi_diag = FALSE,
  verbose = TRUE,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model_set_FAST_+3A_maxiter">maxIter</code></td>
<td>
<p>the maximum iteration of ICM-EM algorithm. The default is 30.</p>
</td></tr>
<tr><td><code id="model_set_FAST_+3A_epsloglik">epsLogLik</code></td>
<td>
<p>an optional positive vlaue, tolerance of relative variation rate of the observed pseudo loglikelihood value, defualt as '1e-5'.</p>
</td></tr>
<tr><td><code id="model_set_FAST_+3A_error_heter">error_heter</code></td>
<td>
<p>a logical value, whether use the heterogenous error for FAST model, default as <code>TRUE</code>. If <code>error.heter=FALSE</code>, then the homogenuous error is used.</p>
</td></tr>
<tr><td><code id="model_set_FAST_+3A_psi_diag">Psi_diag</code></td>
<td>
<p>a logical value, whether set the conditional covariance matrices of intrisic CAR to diagonal, default as <code>FALSE</code></p>
</td></tr>
<tr><td><code id="model_set_FAST_+3A_verbose">verbose</code></td>
<td>
<p>a logical value, whether output the information in iteration.</p>
</td></tr>
<tr><td><code id="model_set_FAST_+3A_seed">seed</code></td>
<td>
<p>a postive integer, the random seed to be set in initialization.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return a Seurat object with new reduction  (named reduction.name) added to the 'reductions' slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>model_set_FAST(maxIter = 30, epsLogLik = 1e-5,
  error_heter=TRUE, Psi_diag=FALSE, verbose=TRUE, seed=2023)

</code></pre>

<hr>
<h2 id='NCFM'>Cell-feature coembedding for scRNA-seq data</h2><span id='topic+NCFM'></span>

<h3>Description</h3>

<p>Cell-feature coembedding for scRNA-seq data based on FAST model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NCFM(
  object,
  assay = NULL,
  slot = "data",
  nfeatures = 2000,
  q = 10,
  reduction.name = "ncfm",
  weighted = FALSE,
  var.features = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NCFM_+3A_object">object</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_assay">assay</code></td>
<td>
<p>an optional string, specify the name of assay in the Seurat object to be used, 'NULL' means default assay in seu.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_slot">slot</code></td>
<td>
<p>an optional string, specify the name of slot.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_nfeatures">nfeatures</code></td>
<td>
<p>an optional integer, specify the number of features to select as top variable features. Default is 2000.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_q">q</code></td>
<td>
<p>an optional positive integer, specify the dimension of low dimensional embeddings to compute and store. Default is 10.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_reduction.name">reduction.name</code></td>
<td>
<p>an optional string, specify the dimensional reduction name, 'ncfm' by default.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_weighted">weighted</code></td>
<td>
<p>an optional logical value, specify whether use weighted method.</p>
</td></tr>
<tr><td><code id="NCFM_+3A_var.features">var.features</code></td>
<td>
<p>an optional string vector, specify the variable features used to calculate cell embedding.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(pbmc3k_subset)
pbmc3k_subset &lt;- NCFM(pbmc3k_subset)
</code></pre>

<hr>
<h2 id='NCFM_fast'>Cell-feature coembedding for SRT data</h2><span id='topic+NCFM_fast'></span>

<h3>Description</h3>

<p>Run cell-feature coembedding for SRT data based on FAST model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NCFM_fast(
  object,
  Adj_sp,
  assay = NULL,
  slot = "data",
  nfeatures = 2000,
  q = 10,
  reduction.name = "fast",
  var.features = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NCFM_fast_+3A_object">object</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_adj_sp">Adj_sp</code></td>
<td>
<p>a sparse matrix, specify the adjacency matrix among spots.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_assay">assay</code></td>
<td>
<p>an optional string, the name of assay used.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_slot">slot</code></td>
<td>
<p>an optional string, the name of slot used.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_nfeatures">nfeatures</code></td>
<td>
<p>an optional postive integer, the number of features to select as top variable features. Default is 2000.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_q">q</code></td>
<td>
<p>an optional positive integer, specify the dimension of low dimensional embeddings to compute and store. Default is 10.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_reduction.name">reduction.name</code></td>
<td>
<p>an optional string, dimensional reduction name, 'fast' by default.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_var.features">var.features</code></td>
<td>
<p>an optional string vector, specify the variable features, used to calculate cell embedding.</p>
</td></tr>
<tr><td><code id="NCFM_fast_+3A_...">...</code></td>
<td>
<p>Other argument passed  to the <code><a href="#topic+FAST_run">FAST_run</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(CosMx_subset)
pos &lt;- as.matrix(CosMx_subset@meta.data[,c("x", "y")])
Adj_sp &lt;- AddAdj(pos)
# Here, we set maxIter = 3 for fast computation and demonstration.
CosMx_subset &lt;- NCFM_fast(CosMx_subset, Adj_sp = Adj_sp, maxIter=3)

</code></pre>

<hr>
<h2 id='pbmc3k_subset'>A Seurat object including scRNA-seq PBMC dataset</h2><span id='topic+pbmc3k_subset'></span>

<h3>Description</h3>

  
<p>This data is a subset of  PBMC3k scRNA-seq data in SeuratData package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  
  data(pbmc3k_subset)  
</code></pre>


<h3>Format</h3>

  
<p>A Seurat object, including count matrix, and manual annotation.
</p>


<h3>Source</h3>

  
<p>The data is from the scRNA-seq sequencing platform.
</p>


<h3>References</h3>

  
<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  # Show  examples of how to use the dataset.  
  data(pbmc3k_subset)  
  library(Seurat)
  pbmc3k_subset
</code></pre>

<hr>
<h2 id='pdistance'>Calculate the cell-feature distance matrix</h2><span id='topic+pdistance'></span>

<h3>Description</h3>

<p>Calculate the cell-feature distance matrix based on coembeddings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdistance(object, reduction = "fast", assay.name = "distce", eta = 1e-10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pdistance_+3A_object">object</code></td>
<td>
<p>a Seurat object.</p>
</td></tr>
<tr><td><code id="pdistance_+3A_reduction">reduction</code></td>
<td>
<p>a opstional string, dimensional reduction name, 'fast' by default.</p>
</td></tr>
<tr><td><code id="pdistance_+3A_assay.name">assay.name</code></td>
<td>
<p>a opstional string, specify the new generated assay name, 'distce' by default.</p>
</td></tr>
<tr><td><code id="pdistance_+3A_eta">eta</code></td>
<td>
<p>an optional postive real, a quantity to avoid numerical errors. 1e-10 by default.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculate the distance matrix between cells/spots and features, and then put the distance matrix in a new generated assay. This 
distance matrix will be used in the siganture gene identification.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pbmc3k_subset)
pbmc3k_subset &lt;- NCFM(pbmc3k_subset)
pbmc3k_subset &lt;- pdistance(pbmc3k_subset, "ncfm")
</code></pre>

<hr>
<h2 id='RunHarmonyLouvain'>Embedding alignment and clustering based on the embeddings from FAST</h2><span id='topic+RunHarmonyLouvain'></span>

<h3>Description</h3>

<p>Embedding alignment and clustering using the Harmony and Louvain based on the ebmeddings from FAST as well as determining the number of clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunHarmonyLouvain(PRECASTObj, resolution = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunHarmonyLouvain_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <code>CreatePRECASTObject</code>.</p>
</td></tr>
<tr><td><code id="RunHarmonyLouvain_+3A_resolution">resolution</code></td>
<td>
<p>an optional real, the value of the resolution parameter, use a value above (below) 1.0 if you want to obtain a larger (smaller) number of communities.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a revised <code>PRECASTObj</code> object with slot <code>PRECASTObj@resList</code> added by a <code>Harmony</code> compoonent (including the aligned embeddings and embeddings of batch effects) and a <code>Louvain</code> component (including the clusters).
</p>

<hr>
<h2 id='RuniSCMEB'>Fit an iSC-MEB model using the embeddings from FAST</h2><span id='topic+RuniSCMEB'></span>

<h3>Description</h3>

<p>Fit an iSC-MEB model using the embeddings from FAST and the number of clusters obtained by Louvain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RuniSCMEB(PRECASTObj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RuniSCMEB_+3A_precastobj">PRECASTObj</code></td>
<td>
<p>a PRECASTObj object created by <code>CreatePRECASTObject</code>.</p>
</td></tr>
<tr><td><code id="RuniSCMEB_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="#topic+iscmeb_run">iscmeb_run</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a revised PRECASTObj object with an added component <code>iSCMEB</code> in the slot <code>PRECASTObj@resList</code> (including the aligned embeddings, clusters and posterior probability matrix of clusters).
</p>

<hr>
<h2 id='SelectHKgenes'>Select housekeeping genes</h2><span id='topic+SelectHKgenes'></span>

<h3>Description</h3>

<p>Select housekeeping genes for preparation of removing unwanted variations in expression matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SelectHKgenes(seuList, species = c("Human", "Mouse"), HK.number = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SelectHKgenes_+3A_seulist">seuList</code></td>
<td>
<p>an M-length  list consisting of Seurat object, include the information of expression matrix and spatial coordinates (named <code>row</code> and <code>col</code>) in the slot <code>meta.data</code>.</p>
</td></tr>
<tr><td><code id="SelectHKgenes_+3A_species">species</code></td>
<td>
<p>a string, the species, one of 'Human' and 'Mouse'.</p>
</td></tr>
<tr><td><code id="SelectHKgenes_+3A_hk.number">HK.number</code></td>
<td>
<p>an optional integer, specify the number of housekeeping genes to be selected.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a string vector of the selected gene names.
</p>

<hr>
<h2 id='top5_signatures'>A data.frame object including top five signature genes in scRNA-seq PBMC dataset</h2><span id='topic+top5_signatures'></span>

<h3>Description</h3>

  
<p>This data is a data.frame object that includes top five signature genes in scRNA-seq PBMC dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  
  data(top5_signatures)  
</code></pre>


<h3>Format</h3>

  
<p>A data.frame object, including signature genes, distance, and manual annotation.
</p>


<h3>Source</h3>

  
<p>None
</p>


<h3>References</h3>

  
<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  # Show  examples of how to use the dataset.  
  data(top5_signatures)  
  head(top5_signatures)
</code></pre>

<hr>
<h2 id='transferGeneNames'>Transfer gene names from one fortmat to the other format</h2><span id='topic+transferGeneNames'></span>

<h3>Description</h3>

<p>Transfer gene names from one fortmat to the other format for two species: human and mouse.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transferGeneNames(
  genelist,
  now_name = "ensembl",
  to_name = "symbol",
  species = c("Human", "Mouse"),
  Method = c("eg.db", "biomart")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transferGeneNames_+3A_genelist">genelist</code></td>
<td>
<p>a string vector, the gene list to be transferred.</p>
</td></tr>
<tr><td><code id="transferGeneNames_+3A_now_name">now_name</code></td>
<td>
<p>a string, the current format of gene names, one of 'ensembl', 'symbol'.</p>
</td></tr>
<tr><td><code id="transferGeneNames_+3A_to_name">to_name</code></td>
<td>
<p>a string, the  format of gene names to transfer, one of 'ensembl', 'symbol'.</p>
</td></tr>
<tr><td><code id="transferGeneNames_+3A_species">species</code></td>
<td>
<p>a string, the species, one of 'Human' and 'Mouse'.</p>
</td></tr>
<tr><td><code id="transferGeneNames_+3A_method">Method</code></td>
<td>
<p>a string, the method to use, one of 'biomaRt' and 'eg.db', default as 'eg.db'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a string vector of transferred gene names. The gene names not matched in the database will not change.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geneNames &lt;- c("ENSG00000171885", "ENSG00000115756")
transferGeneNames(geneNames, now_name = "ensembl", to_name="symbol",species="Human", Method='eg.db')


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
