<!DOCTYPE html><html><head><title>Help for package ProliferativeIndex</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ProliferativeIndex}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculatePI'><p>A function for calculating proliferative index from variance stabalized RNA-seq data in the ProliferativeIndex package</p></a></li>
<li><a href='#compareModeltoPI'><p>A function for comparing proliferative indices from variance stabalized RNA-seq data to model principal componenets in the ProliferativeIndex package</p></a></li>
<li><a href='#comparePI'><p>A function for comparing proliferative indices from variance stabalized RNA-seq data in the ProliferativeIndex package</p></a></li>
<li><a href='#exReadDataObj'><p>TCGA ACC data set output from readDataForPI function</p></a></li>
<li><a href='#exVSTPI'><p>TCGA ACC data set output from calculatePI function</p></a></li>
<li><a href='#readDataForPI'><p>A function for reading data in for use with the ProliferativeIndex package</p></a></li>
<li><a href='#vstTCGA_ACCData_sub'><p>TCGA ACC data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Calculates and Analyzes the Proliferative Index</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions for calculating and analyzing the proliferative 
    index (PI) from an RNA-seq dataset. As described in Ramaker &amp; Lasseigne, 
    et al. bioRxiv, 2016 &lt;<a href="https://doi.org/10.1101%2F063057">doi:10.1101/063057</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-08-14 17:06:10 UTC; brittany</td>
</tr>
<tr>
<td>Author:</td>
<td>Brittany Lasseigne [aut, cre],
  Ryne Ramaker [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Brittany Lasseigne &lt;brittany.lasseigne@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-08-14 17:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculatePI'>A function for calculating proliferative index from variance stabalized RNA-seq data in the ProliferativeIndex package</h2><span id='topic+calculatePI'></span>

<h3>Description</h3>

<p>This function allows the user to read in data for subsequent proliferative index calculation and analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculatePI(userObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculatePI_+3A_userobject">userObject</code></td>
<td>
<p>Output from ProliferativeIndex readDataForPI function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>calculatePI(exReadDataObj)
</code></pre>

<hr>
<h2 id='compareModeltoPI'>A function for comparing proliferative indices from variance stabalized RNA-seq data to model principal componenets in the ProliferativeIndex package</h2><span id='topic+compareModeltoPI'></span>

<h3>Description</h3>

<p>This function allows the user to calculate a correlation between their model and PI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareModeltoPI(userObject, vstPI)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareModeltoPI_+3A_userobject">userObject</code></td>
<td>
<p>Output from ProliferativeIndex readDataForPI function (user data)</p>
</td></tr>
<tr><td><code id="compareModeltoPI_+3A_vstpi">vstPI</code></td>
<td>
<p>Output from ProliferativeIndex calculatePI function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>compareModeltoPI(exReadDataObj, exVSTPI)
</code></pre>

<hr>
<h2 id='comparePI'>A function for comparing proliferative indices from variance stabalized RNA-seq data in the ProliferativeIndex package</h2><span id='topic+comparePI'></span>

<h3>Description</h3>

<p>This function allows the user to examine a summary and plot of their PIs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comparePI(vstPIinput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comparePI_+3A_vstpiinput">vstPIinput</code></td>
<td>
<p>Output from ProliferativeIndex calculatePI function (PI of user vst data)////////////////////////////////////////////////////////////////////////</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>comparePI(exVSTPI)
</code></pre>

<hr>
<h2 id='exReadDataObj'>TCGA ACC data set output from readDataForPI function</h2><span id='topic+exReadDataObj'></span>

<h3>Description</h3>

<p>This object is the output from readDataForPI taking vstTCGA_ACCData_sub and this model:
c(&quot;AIFM3&quot;, &quot;ATP9B&quot;, &quot;CTRC&quot;, &quot;MCL1&quot;, &quot;MGAT4B&quot;, &quot;ODF2L&quot;, &quot;SNORA65&quot;, &quot;TPPP2&quot;))
as inputs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exReadDataObj)
</code></pre>


<h3>Format</h3>

<p>A list of the vstData (vstTCGA_ACCData_sub data frame with 20501 rows and 10 columns) and the modelIDs (character vector of &quot;AIFM3&quot;, &quot;ATP9B&quot;, &quot;CTRC&quot;, &quot;MCL1&quot;, &quot;MGAT4B&quot;, &quot;ODF2L&quot;, &quot;SNORA65&quot;, &quot;TPPP2&quot;)</p>

<hr>
<h2 id='exVSTPI'>TCGA ACC data set output from calculatePI function</h2><span id='topic+exVSTPI'></span>

<h3>Description</h3>

<p>This object is the output from calculatePI from the exReadDataObj as input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exVSTPI)
</code></pre>


<h3>Format</h3>

<p>A numeric vector of the calculated Proliferation Indices for the vstTCGA_ACCData_sub data</p>

<hr>
<h2 id='readDataForPI'>A function for reading data in for use with the ProliferativeIndex package</h2><span id='topic+readDataForPI'></span>

<h3>Description</h3>

<p>This function allows the user to read in variance stabalized RNA-seq data and gene model names for subsequent proliferative index calculation and analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readDataForPI(vstData, modelIDs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readDataForPI_+3A_vstdata">vstData</code></td>
<td>
<p>Dataframe of user variance stabalized count data (from DESeq2) with samples in columns and genes in rows. Rownames must be genes.</p>
</td></tr>
<tr><td><code id="readDataForPI_+3A_modelids">modelIDs</code></td>
<td>
<p>Genes in user identified model for comparison to proliferative index</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>readDataForPI(vstTCGA_ACCData_sub, c("AIFM3", "ATP9B", "CTRC", "MCL1", 
"MGAT4B", "ODF2L", "SNORA65", "TPPP2"))
</code></pre>

<hr>
<h2 id='vstTCGA_ACCData_sub'>TCGA ACC data set</h2><span id='topic+vstTCGA_ACCData_sub'></span>

<h3>Description</h3>

<p>A dataset containing data from the The Cancer Genome Atlas (TCGA) Adrenocortical Carcinoma (ACC) dataset. This data was obtained from the TCGA data portal (tcga-data.nci.nih.gov) in June 2015. Level 3 RNASeqV2 raw count data was variance stabalized with the DESeq2 v1.8.2 varianceStabilizingTransformation:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(vstTCGA_ACCData_sub)
</code></pre>


<h3>Format</h3>

<p>A data frame with 20501 rows and 10 variables</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
