<!DOCTYPE html><html><head><title>Help for package promor</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {promor}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aver_techreps'><p>Compute average intensity</p></a></li>
<li><a href='#corr_plot'><p>Correlation between technical replicates</p></a></li>
<li><a href='#covid_fit_df'><p>Suvarna et al 2021 LFQ data (fit object)</p></a></li>
<li><a href='#covid_norm_df'><p>Suvarna et al 2021 LFQ data (normalized)</p></a></li>
<li><a href='#create_df'><p>Create a data frame of protein intensities</p></a></li>
<li><a href='#ecoli_fit_df'><p>Cox et al 2014 LFQ data (fit object)</p></a></li>
<li><a href='#ecoli_norm_df'><p>Cox et al 2014 LFQ data (normalized)</p></a></li>
<li><a href='#feature_plot'><p>Visualize feature (protein) variation among conditions</p></a></li>
<li><a href='#filterbygroup_na'><p>Filter proteins by group level missing data</p></a></li>
<li><a href='#find_dep'><p>Identify differentially expressed proteins between groups</p></a></li>
<li><a href='#heatmap_de'><p>Heatmap of differentially expressed proteins</p></a></li>
<li><a href='#heatmap_na'><p>Visualize missing data</p></a></li>
<li><a href='#impute_na'><p>Impute missing values</p></a></li>
<li><a href='#impute_plot'><p>Visualize the impact of imputation</p></a></li>
<li><a href='#norm_plot'><p>Visualize the effect of normalization</p></a></li>
<li><a href='#normalize_data'><p>Normalize intensity data</p></a></li>
<li><a href='#onegroup_only'><p>Proteins that are only expressed in a given group</p></a></li>
<li><a href='#performance_plot'><p>Model performance plot</p></a></li>
<li><a href='#pre_process'><p>Pre-process protein intensity data for modeling</p></a></li>
<li><a href='#rem_feature'><p>Remove user-specified proteins (features) from a data frame</p></a></li>
<li><a href='#rem_sample'><p>Remove user-specified samples</p></a></li>
<li><a href='#roc_plot'><p>ROC plot</p></a></li>
<li><a href='#split_data'><p>Split the data frame to create training and test data</p></a></li>
<li><a href='#test_models'><p>Test machine learning models on test data</p></a></li>
<li><a href='#train_models'><p>Train machine learning models on training data</p></a></li>
<li><a href='#varimp_plot'><p>Variable importance plot</p></a></li>
<li><a href='#volcano_plot'><p>Volcano plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Proteomics Data Analysis and Modeling Tools</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A comprehensive, user-friendly package for label-free proteomics data analysis and machine learning-based modeling. Data generated from 'MaxQuant' can be easily used to conduct differential expression analysis, build predictive models with top protein candidates, and assess model performance. promor includes a suite of tools for quality control, visualization, missing data imputation (Lazar et. al. (2016) &lt;<a href="https://doi.org/10.1021%2Facs.jproteome.5b00981">doi:10.1021/acs.jproteome.5b00981</a>&gt;), differential expression analysis (Ritchie et. al. (2015) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkv007">doi:10.1093/nar/gkv007</a>&gt;), and machine learning-based modeling (Kuhn (2008) &lt;<a href="https://doi.org/10.18637%2Fjss.v028.i05">doi:10.18637/jss.v028.i05</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL (&ge; 2.1)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/caranathunge/promor">https://github.com/caranathunge/promor</a>,
<a href="https://caranathunge.github.io/promor/">https://caranathunge.github.io/promor/</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2, ggplot2, ggrepel, gridExtra, limma, statmod,
pcaMethods, VIM, missForest, caret, kernlab, xgboost,
naivebayes, viridis, pROC</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/caranathunge/promor/issues">https://github.com/caranathunge/promor/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-17 22:04:15 UTC; caran</td>
</tr>
<tr>
<td>Author:</td>
<td>Chathurani Ranathunge
    <a href="https://orcid.org/0000-0003-1901-2119"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre,
    cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chathurani Ranathunge &lt;caranathunge86@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-17 22:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aver_techreps'>Compute average intensity</h2><span id='topic+aver_techreps'></span>

<h3>Description</h3>

<p>This function computes average intensities across
technical replicates for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aver_techreps(raw_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aver_techreps_+3A_raw_df">raw_df</code></td>
<td>
<p>A <code>raw_df</code> object containing technical replicates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>aver_techreps</code> assumes that column names in the data frame
follow the &quot;Group_UniqueSampleID_TechnicalReplicate&quot; notation. (Use
<code>head(raw_df)</code> to see the structure of the <code>raw_df</code> object.)
</p>


<h3>Value</h3>

<p>A <code>raw_df</code> object of averaged intensities.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<ul>
<li><p><code><a href="limma.html#topic+avearrays">avearrays</a></code> from <code><a href="limma.html#topic+limma">limma</a></code>
package.</p>
</li>
<li> <p><code><a href="#topic+create_df">create_df</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Use a data set containing technical replicates to create a raw_df object
raw_df &lt;- create_df(
prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg2.txt",
exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed2.txt",
tech_reps = TRUE
)

# Compute average intensities across technical replicates.
rawdf_ave &lt;- aver_techreps(raw_df)

</code></pre>

<hr>
<h2 id='corr_plot'>Correlation between technical replicates</h2><span id='topic+corr_plot'></span>

<h3>Description</h3>

<p>This function generates scatter plots to visualize the
correlation between a given pair of technical replicates (Eg: 1 vs 2)
for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr_plot(
  raw_df,
  rep_1,
  rep_2,
  save = FALSE,
  file_type = "pdf",
  palette = "viridis",
  text_size = 5,
  n_row = 4,
  n_col = 4,
  dpi = 80,
  file_path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corr_plot_+3A_raw_df">raw_df</code></td>
<td>
<p>A <code>raw_df</code> object (output of <code><a href="#topic+create_df">create_df</a></code>)
containing technical replicates.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_rep_1">rep_1</code></td>
<td>
<p>Numerical. Technical replicate number.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_rep_2">rep_2</code></td>
<td>
<p>Numerical. Number of the second technical replicate to compare
to <code>rep1</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the scatter plots.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_n_row">n_row</code></td>
<td>
<p>Numerical. Number of plots to print in a row in a single page.
Default is <code>4</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_n_col">n_col</code></td>
<td>
<p>Numerical. Number of plots to print in a column in a single
page. Default is <code>4</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p> Given a data frame of log-transformed intensities
(a <code>raw_df</code> object) and a pair of numbers referring to the technical
replicates, <code>corr_plot</code> produces a list of scatter plots showing
correlation between the given pair of technical replicates for all the
samples provided in the data frame.
</p>
</li>
<li><p> Note: <code>n_row</code> * <code>n_col</code> should be equal to the number of
samples to display in a single page.
</p>
</li></ul>



<h3>Value</h3>

<p>A list of <code>ggplot2</code> plot objects.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<p><code>create_df</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Use a data set containing technical replicates to create a raw_df object
raw_df &lt;- create_df(
prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg2.txt",
exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed2.txt",
tech_reps = TRUE
)



## Compare technical replicates 1 vs. 2 for all samples
corr_plot(raw_df, rep_1 = 1, rep_2 = 2)


</code></pre>

<hr>
<h2 id='covid_fit_df'>Suvarna et al 2021 LFQ data (fit object)</h2><span id='topic+covid_fit_df'></span>

<h3>Description</h3>

<p>An object of class &quot;MArrayLM&quot; from running find_dep on covid_norm_df
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(covid_fit_df)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;MArrayLM&quot;
</p>


<h3>References</h3>

<p><a href="https://www.frontiersin.org/articles/10.3389/fphys.2021.652799/full#h3">https://www.frontiersin.org/articles/10.3389/fphys.2021.652799/full#h3</a>
</p>

<hr>
<h2 id='covid_norm_df'>Suvarna et al 2021 LFQ data (normalized)</h2><span id='topic+covid_norm_df'></span>

<h3>Description</h3>

<p>A dataframe containing normalized LFQ protein intensity data for 230
proteins in 35 samples (a subset of the original data set)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(covid_norm_df)
</code></pre>


<h3>Format</h3>

<p>A data frame with 230 rows (proteins) and 35 columns (samples)
</p>


<h3>References</h3>

<p><a href="https://www.frontiersin.org/articles/10.3389/fphys.2021.652799/full#h3">https://www.frontiersin.org/articles/10.3389/fphys.2021.652799/full#h3</a>
</p>

<hr>
<h2 id='create_df'>Create a data frame of protein intensities</h2><span id='topic+create_df'></span>

<h3>Description</h3>

<p>This function creates a data frame of protein intensities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_df(
  prot_groups,
  exp_design,
  input_type = "MaxQuant",
  data_type = "LFQ",
  filter_na = TRUE,
  filter_prot = TRUE,
  uniq_pep = 2,
  tech_reps = FALSE,
  zero_na = TRUE,
  log_tr = TRUE,
  base = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_df_+3A_prot_groups">prot_groups</code></td>
<td>
<p>File path to a proteinGroups.txt file produced by MaxQuant
or a standard input file containing a quantitative matrix
where the proteins or protein groups are indicated by rows and the
samples by columns.</p>
</td></tr>
<tr><td><code id="create_df_+3A_exp_design">exp_design</code></td>
<td>
<p>File path to a text file containing the experimental
design.</p>
</td></tr>
<tr><td><code id="create_df_+3A_input_type">input_type</code></td>
<td>
<p>Type of input file indicated by <code>prot_groups</code>.
Available options are: &quot;MaxQuant&quot;, if a proteinGroups.txt file is used, or
&quot;standard&quot; if a standard input file is used. Default is &quot;MaxQuant.&quot;</p>
</td></tr>
<tr><td><code id="create_df_+3A_data_type">data_type</code></td>
<td>
<p>Type of sample protein intensity data columns to use from
the proteinGroups.txt file. Some available options are &quot;LFQ&quot;, &quot;iBAQ&quot;,
&quot;Intensity&quot;. Default is &quot;LFQ.&quot; User-defined prefixes in the proteinGroups.txt
file are also allowed. The <code>data_type</code> argument is case-sensitive, and
only applies when <code>input_type = "MaxQuant"</code>.</p>
</td></tr>
<tr><td><code id="create_df_+3A_filter_na">filter_na</code></td>
<td>
<p>Logical. If <code>TRUE</code>(default), filters out empty rows and
columns from the data frame.</p>
</td></tr>
<tr><td><code id="create_df_+3A_filter_prot">filter_prot</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), filters out
reverse proteins, proteins only identified by site, potential contaminants,
and proteins identified with less than the minimum number of unique peptides
indicated by <code>uniq_pep</code>. Only applies when
<code>input_type = "MaxQuant"</code>.</p>
</td></tr>
<tr><td><code id="create_df_+3A_uniq_pep">uniq_pep</code></td>
<td>
<p>Numerical. Proteins that are identified by this number or
fewer number of unique peptides are filtered out (default is 2).Only applies
when <code>input_type = "MaxQuant"</code>.</p>
</td></tr>
<tr><td><code id="create_df_+3A_tech_reps">tech_reps</code></td>
<td>
<p>Logical. Indicate as <code>TRUE</code> if technical replicates
are present in the data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="create_df_+3A_zero_na">zero_na</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), zeros are considered
missing values and replaced with NAs.</p>
</td></tr>
<tr><td><code id="create_df_+3A_log_tr">log_tr</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), intensity values are log
transformed to the base indicated by <code>base</code>.</p>
</td></tr>
<tr><td><code id="create_df_+3A_base">base</code></td>
<td>
<p>Numerical. Logarithm base. Default is 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p> This function first reads in the proteinGroups.txt file
produced by MaxQuant or a standard input file containing a quantitative
matrix where the proteins or protein groups are indicated by rows and the
samples by columns.
</p>
</li>
<li><p> It then reads in the expDesign.txt file provided as
<code>exp_design</code> and extracts relevant information from it to add to the
data frame. an example of the expDesign.txt is provided here:
<a href="https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt">https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt</a>.
</p>
</li>
<li><p> First, empty rows and columns are removed from the data frame.
</p>
</li>
<li><p> Next, if a proteinGroups.txt file is used, it filters out reverse
proteins, proteins that were only identified by site, and potential
contaminants.Then it removes proteins identified with less than
the number of unique peptides indicated by <code>uniq_pep</code> from the
data frame.
</p>
</li>
<li><p> Next, it extracts the intensity columns indicated by <code>data type</code>
and the selected protein rows from the data frame.
</p>
</li>
<li><p> Converts missing values (zeros) to NAs.
</p>
</li>
<li><p> Finally, the function log transforms the intensity values.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>raw_df</code> object which is a data frame containing protein
intensities. Proteins or protein groups are indicated by rows and samples
by columns.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

### Using a proteinGroups.txt file produced by MaxQuant as input.
## Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt",
  input_type = "MaxQuant"
)

## Data containing technical replicates
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg2.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed2.txt",
  input_type = "MaxQuant",
  tech_reps = TRUE
)

## Alter the number of unique peptides needed to retain a protein
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt",
  input_type = "MaxQuant",
  uniq_pep = 1
)

## Use "iBAQ" values instead of "LFQ" values
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt",
  input_type = "MaxQuant",
  data_type = "iBAQ"
)

### Using a universal standard input file instead of MaxQuant output.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/st.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt",
  input_type = "standard"
)

</code></pre>

<hr>
<h2 id='ecoli_fit_df'>Cox et al 2014 LFQ data (fit object)</h2><span id='topic+ecoli_fit_df'></span>

<h3>Description</h3>

<p>An object of class &quot;MArrayLM&quot; from running find_dep on ecoli_norm_df
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ecoli_fit_df)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;MArrayLM&quot;
</p>


<h3>References</h3>

<p><a href="https://europepmc.org/article/MED/24942700#id609082">https://europepmc.org/article/MED/24942700#id609082</a>
</p>

<hr>
<h2 id='ecoli_norm_df'>Cox et al 2014 LFQ data (normalized)</h2><span id='topic+ecoli_norm_df'></span>

<h3>Description</h3>

<p>A dataframe containing normalized LFQ protein intensity data for 4360
proteins in 6 samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ecoli_norm_df)
</code></pre>


<h3>Format</h3>

<p>A data frame with 4360 rows (proteins) and 6 columns (samples)
</p>


<h3>References</h3>

<p><a href="https://europepmc.org/article/MED/24942700#id609082">https://europepmc.org/article/MED/24942700#id609082</a>
</p>

<hr>
<h2 id='feature_plot'>Visualize feature (protein) variation among conditions</h2><span id='topic+feature_plot'></span>

<h3>Description</h3>

<p>This function visualizes protein intensity differences among
conditions (classes) using box plots or density distribution plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature_plot(
  model_df,
  type = "box",
  text_size = 10,
  palette = "viridis",
  n_row,
  n_col,
  save = FALSE,
  file_path = NULL,
  file_name = "Feature_plot",
  file_type = "pdf",
  dpi = 80,
  plot_width = 7,
  plot_height = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="feature_plot_+3A_model_df">model_df</code></td>
<td>
<p>A <code>model_df</code> object from performing <code>pre_process</code>.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_type">type</code></td>
<td>
<p>Type of plot to generate. Choices are &quot;box&quot; or &quot;density.&quot; Default
is <code>"box."</code></p>
</td></tr>
<tr><td><code id="feature_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_n_row">n_row</code></td>
<td>
<p>Number of rows to print the plots.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_n_col">n_col</code></td>
<td>
<p>Number of columns to print the plots.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot.
Default is <code>"Feature_plot."</code></p>
</td></tr>
<tr><td><code id="feature_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="feature_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>7</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes condition-wise differences in protein
intensity using boxplots and/or density plots.
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> object
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>pre_process</code>, <code>rem_feature</code></p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Create a model_df object with default settings.
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Feature variation - box plots
feature_plot(covid_model_df, type = "box", n_row = 4, n_col = 2)

## Density plots
feature_plot(covid_model_df, type = "density")

## Change color palette
feature_plot(covid_model_df, type = "density", n_row = 4, n_col = 2, palette = "rocket")

</code></pre>

<hr>
<h2 id='filterbygroup_na'>Filter proteins by group level missing data</h2><span id='topic+filterbygroup_na'></span>

<h3>Description</h3>

<p>This function filters out proteins based on missing data
at the group level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterbygroup_na(raw_df, set_na = 0.34, filter_condition = "either")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterbygroup_na_+3A_raw_df">raw_df</code></td>
<td>
<p>A <code>raw_df</code> object (output of <code><a href="#topic+create_df">create_df</a></code>)</p>
</td></tr>
<tr><td><code id="filterbygroup_na_+3A_set_na">set_na</code></td>
<td>
<p>The proportion of missing data allowed.
Default is 0.34 (one third of the samples in the group).</p>
</td></tr>
<tr><td><code id="filterbygroup_na_+3A_filter_condition">filter_condition</code></td>
<td>
<p>If set to <code>"each"</code>, proteins that exceed
the missing value proportion threshold set by <code>set_na</code> in each group
will be removed (lenient).
If set to <code>"either"</code>(default), proteins that exceed the missing value
proportion threshold set by <code>set_na</code> in at least one group will be
removed (stringent).</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p> This function first
extracts group or condition information from the <code>raw_df</code> object and
assigns samples to their groups.
</p>
</li>
<li><p> If <code>filter_condition = "each"</code>, it then removes proteins (rows)
from the data frame if the proportion of NAs in <strong>each</strong> group exceeds the
threshold indicated by <code>set_na</code> (default is 0.34). This option is
more lenient in comparison to <code>filter_condition = "either"</code>, where
proteins that exceeds the missing data threshold in <strong>either</strong> group gets
removed from the data frame.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>raw_df</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_df">create_df</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Remove proteins that exceed 34% NAs in either group (default)
rawdf_filt1 &lt;- filterbygroup_na(raw_df)

## Remove proteins that exceed 34% NAs in each group
rawdf_filt2 &lt;- filterbygroup_na(raw_df, filter_condition = "each")

## Proportion of samples with NAs allowed in each group = 0.5
rawdf_filt3 &lt;- filterbygroup_na(raw_df, set_na = 0.5, filter_condition = "each")


</code></pre>

<hr>
<h2 id='find_dep'>Identify differentially expressed proteins between groups</h2><span id='topic+find_dep'></span>

<h3>Description</h3>

<p>This function performs differential expression analysis
on protein intensity data with limma.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_dep(
  df,
  save_output = FALSE,
  save_tophits = FALSE,
  file_path = NULL,
  adj_method = "BH",
  cutoff = 0.05,
  lfc = 1,
  n_top = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_dep_+3A_df">df</code></td>
<td>
<p>A <code>norm_df</code> object or an <code>imp_df</code> object.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_save_output">save_output</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves results from the
differential expression analysis in a text file labeled &quot;limma_output.txt&quot;
in the directory specified by <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_save_tophits">save_tophits</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves <code>n_top</code>
number of top hits from the differential expression analysis in a text file
labeled &quot;TopHits.txt&quot; in the directory specified by <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_adj_method">adj_method</code></td>
<td>
<p>Method used for adjusting the p-values for multiple
testing. Default is <code>"BH"</code> for &quot;Benjamini-Hochberg&quot; method.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_cutoff">cutoff</code></td>
<td>
<p>Cutoff value for p-values and adjusted p-values. Default is
0.05.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_lfc">lfc</code></td>
<td>
<p>Minimum absolute log2-fold change to use as threshold for
differential expression.</p>
</td></tr>
<tr><td><code id="find_dep_+3A_n_top">n_top</code></td>
<td>
<p>The number of top differentially expressed proteins to save in
the &quot;TopHits.txt&quot; file. Default is <code>20</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li>
<p>It is important that the data is first log-transformed, ideally,
imputed, and normalized before performing differential expression analysis.
</p>
</li>
<li> <p><code>save_output</code> saves the complete results table from the
differential expression analysis.
</p>
</li>
<li> <p><code>save_tophits</code> first subsets the results to those with absolute
log fold change of more than 1, performs multiple correction with
the method specified in <code>adj_method</code> and outputs the top <code>n_top</code>
results based on lowest p-value and adjusted p-value.
</p>
</li>
<li><p> If the number of hits with absolute log fold change of more than 1 is
less than <code>n_top</code>, <code>find_dep</code> prints only those with
log-fold change &gt; 1 to &quot;TopHits.txt&quot;.
</p>
</li>
<li><p> If the <code>file_path</code> is not specified, text files will be saved in
a temporary directory.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>fit_df</code> object, which is similar to a <code>limma</code>
<code>fit</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>References</h3>

<p>Ritchie, Matthew E., et al. &quot;limma powers differential expression
analyses for RNA-sequencing and microarray studies.&quot; Nucleic acids research
43.7 (2015): e47-e47.
</p>


<h3>See Also</h3>

<ul>
<li><p><code>normalize_data</code>
</p>
</li>
<li><p><code><a href="limma.html#topic+lmFit">lmFit</a></code>,
<code><a href="limma.html#topic+eBayes">eBayes</a></code>,
<code><a href="limma.html#topic+topTable">topTable</a></code>, and
<code><a href="limma.html#topic+write.fit">write.fit</a></code> functions from the
<code><a href="limma.html#topic+limma">limma</a></code> package.</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Perform differential expression analysis using default settings
fit_df1 &lt;- find_dep(ecoli_norm_df)

## Change p-value and adjusted p-value cutoff
fit_df2 &lt;- find_dep(ecoli_norm_df, cutoff = 0.1)

</code></pre>

<hr>
<h2 id='heatmap_de'>Heatmap of differentially expressed proteins</h2><span id='topic+heatmap_de'></span>

<h3>Description</h3>

<p>This function generates a heatmap to visualize differentially
expressed proteins between groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_de(
  fit_df,
  df,
  adj_method = "BH",
  cutoff = 0.05,
  lfc = 1,
  sig = "adjP",
  n_top = 20,
  palette = "viridis",
  text_size = 10,
  save = FALSE,
  file_path = NULL,
  file_name = "HeatmapDE",
  file_type = "pdf",
  dpi = 80,
  plot_height = 7,
  plot_width = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_de_+3A_fit_df">fit_df</code></td>
<td>
<p>A <code>fit_df</code> object from performing <code>find_dep</code>.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_df">df</code></td>
<td>
<p>The <code>norm_df</code> object or the <code>imp_df</code> object from which
the <code>fit_df</code> object was obtained.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_adj_method">adj_method</code></td>
<td>
<p>Method used for adjusting the p-values for multiple
testing. Default is <code>"BH"</code>.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_cutoff">cutoff</code></td>
<td>
<p>Cutoff value for p-values and adjusted p-values. Default is
0.05.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_lfc">lfc</code></td>
<td>
<p>Minimum absolute log2-fold change to use as threshold for
differential expression. Default is 1.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_sig">sig</code></td>
<td>
<p>Criteria to denote significance. Choices are <code>"adjP"</code>
(default) for adjusted p-value or <code>"P"</code> for p-value.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_n_top">n_top</code></td>
<td>
<p>Number of top hits to include in the heat map.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_text_size">text_size</code></td>
<td>
<p>Text size for axis text, labels etc.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot. Default is &quot;HeatmapDE.&quot;</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot. Default is <code>"pdf".</code></p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80.</code></p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is 7.</p>
</td></tr>
<tr><td><code id="heatmap_de_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is 7.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the tiles in the heatmap are reordered by intensity values
along both axes (x axis = samples, y axis = proteins).
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>find_dep</code>
</p>
</li>
<li> <p><code><a href="limma.html#topic+topTable">topTable</a></code> and
<code><a href="limma.html#topic+lmFit">lmFit</a></code> functions from the
<code><a href="limma.html#topic+limma">limma</a></code> package.</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Build a heatmap of differentially expressed proteins using the provided
## example fit_df and norm_df data objects
heatmap_de(covid_fit_df, covid_norm_df)

## Create a heatmap with P-value of 0.05 and log fold change of 1 as
## significance criteria.
heatmap_de(covid_fit_df, covid_norm_df, cutoff = 0.05, sig = "P")

## Visualize the top 30 differentially expressed proteins in the heatmap and
## change the color palette
heatmap_de(covid_fit_df, covid_norm_df,
  cutoff = 0.05, sig = "P", n_top = 30,
  palette = "magma"
)

</code></pre>

<hr>
<h2 id='heatmap_na'>Visualize missing data</h2><span id='topic+heatmap_na'></span>

<h3>Description</h3>

<p>This function visualizes the patterns of missing value
occurrence using a heatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_na(
  raw_df,
  protein_range,
  sample_range,
  reorder_x = FALSE,
  reorder_y = FALSE,
  x_fun = mean,
  y_fun = mean,
  palette = "viridis",
  label_proteins = FALSE,
  text_size = 10,
  save = FALSE,
  file_type = "pdf",
  file_path = NULL,
  file_name = "Missing_data_heatmap",
  plot_width = 15,
  plot_height = 15,
  dpi = 80
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_na_+3A_raw_df">raw_df</code></td>
<td>
<p>A <code>raw_df</code> object (output from <code><a href="#topic+create_df">create_df</a></code>).</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_protein_range">protein_range</code></td>
<td>
<p>The range or subset of proteins (rows) to plot. If not
provided, all the proteins (rows) in the data frame will be used.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_sample_range">sample_range</code></td>
<td>
<p>The range of samples to plot. If not
provided, all the samples (columns) in the data frame will be used.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_reorder_x">reorder_x</code></td>
<td>
<p>Logical. If <code>TRUE</code> samples on the x axis are reordered
using the function given in <code>x_fun</code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_reorder_y">reorder_y</code></td>
<td>
<p>Logical. If <code>TRUE</code> proteins in the y axis are reordered
using the function given in <code>y_fun</code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_x_fun">x_fun</code></td>
<td>
<p>Function to reorder samples along the x axis. Possible options
are <code>mean</code> and <code>sum</code>. Default is <code>mean</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_y_fun">y_fun</code></td>
<td>
<p>Function to reorder proteins along the y axis. Possible options
are <code>mean</code> and <code>sum</code>. Default is <code>mean</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_label_proteins">label_proteins</code></td>
<td>
<p>If <code>TRUE</code> proteins on the y axis
will be labeled with their Majority Protein IDs. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_text_size">text_size</code></td>
<td>
<p>Text size for axis labels. Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the heatmap. Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the heatmap. Default is
<code>"Missing_data_heatmap"</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>15</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>15</code>.</p>
</td></tr>
<tr><td><code id="heatmap_na_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes patterns of missing value occurrence using a
heatmap. The user can choose to reorder the axes using the available functions
(<code>x_fun</code>, <code>y_fun</code>) to better understand the underlying cause of
missing data.
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_df">create_df</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Missing data heatmap with default settings.
heatmap_na(raw_df)

## Missing data heatmap with x and y axes reordered by the mean (default) of
## protein intensity.
heatmap_na(raw_df,
  reorder_x = TRUE, reorder_y = TRUE
)

## Missing data heatmap with x and y axes reordered by the sum of
## protein intensity.
heatmap_na(raw_df,
  reorder_x = TRUE, reorder_y = TRUE, x_fun = sum,
  y_fun = sum
)

## Missing data heatmap for a subset of the proteins with x and y axes
## reordered by the mean (default) of protein intensity and the y axis
## labeled with protein IDs.
heatmap_na(raw_df,
  protein_range = 1:30,
  reorder_x = TRUE, reorder_y = TRUE,
  label_proteins = TRUE
)

</code></pre>

<hr>
<h2 id='impute_na'>Impute missing values</h2><span id='topic+impute_na'></span>

<h3>Description</h3>

<p>This function imputes missing values using a user-specified
imputation method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute_na(
  df,
  method = "minProb",
  tune_sigma = 1,
  q = 0.01,
  maxiter = 10,
  ntree = 20,
  n_pcs = 2,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="impute_na_+3A_df">df</code></td>
<td>
<p>A <code>raw_df</code> object (output of <code><a href="#topic+create_df">create_df</a></code>)
containing missing values or a <code>norm_df</code> object after performing
normalization.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_method">method</code></td>
<td>
<p>Imputation method to use. Default is <code>"minProb"</code>.
Available methods: <code>"minDet", "RF", "kNN", and "SVD"</code>.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_tune_sigma">tune_sigma</code></td>
<td>
<p>A scalar used in the <code>"minProb"</code> method for
controlling the standard deviation of the Gaussian distribution
from which random values are drawn for imputation.<br />
Default is 1.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_q">q</code></td>
<td>
<p>A scalar used in <code>"minProb"</code> and <code>"minDet"</code> methods
to obtain a low intensity value for imputation. <code>q</code> should be set to a
very low value. Default is 0.01.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_maxiter">maxiter</code></td>
<td>
<p>Maximum number of iterations to be performed when using the
<code>"RF"</code> method. Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_ntree">ntree</code></td>
<td>
<p>Number of trees to grow in each forest when using the
<code>"RF"</code> method. Default is <code>20</code>.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_n_pcs">n_pcs</code></td>
<td>
<p>Number of principal components to calculate when using the
<code>"SVD"</code> method. Default is 2.</p>
</td></tr>
<tr><td><code id="impute_na_+3A_seed">seed</code></td>
<td>
<p>Numerical. Random number seed. Default is <code>NULL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p> Ideally, you should first remove proteins with
high levels of missing data using the <code>filterbygroup_na</code> function
before running <code>impute_na</code> on the <code>raw_df</code> object or the
<code>norm_df</code> object.
</p>
</li>
<li> <p><code>impute_na</code> function imputes missing values using a
user-specified imputation method from the available options, <code>minProb</code>,
<code>minDet</code>, <code>kNN</code>, <code>RF</code>, and <code>SVD</code>.
</p>
</li>
<li> <p><strong>Note: Some imputation methods may require that the data be normalized
prior to imputation.</strong>
</p>
</li>
<li><p> Make sure to fix the random number seed with <code>seed</code> for reproducibility</p>
</li></ul>
<p>.
</p>


<h3>Value</h3>

<p>An <code>imp_df</code> object, which is a data frame of protein intensities
with no missing values.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>References</h3>

<p>Lazar, Cosmin, et al. &quot;Accounting for the multiple natures of
missing values in label-free quantitative proteomics data sets to compare
imputation strategies.&quot; Journal of proteome research 15.4 (2016): 1116-1125.
</p>


<h3>See Also</h3>

<p>More information on the available imputation methods can be found
in their respective packages.
</p>
<ul>
<li> <p><code><a href="#topic+create_df">create_df</a></code>
</p>
</li>
<li><p> For <code>minProb</code> and
<code>minDet</code> methods, see
<code>imputeLCMD</code> package.
</p>
</li>
<li><p> For Random Forest (<code>RF</code>) method, see
<code><a href="missForest.html#topic+missForest">missForest</a></code>.
</p>
</li>
<li><p> For <code>kNN</code> method, see <code><a href="VIM.html#topic+kNN">kNN</a></code> from the
<code><a href="VIM.html#topic+VIM">VIM</a></code> package.
</p>
</li>
<li><p> For <code>SVD</code> method, see <code><a href="pcaMethods.html#topic+pca">pca</a></code> from the
<code><a href="pcaMethods.html#topic+pcaMethods">pcaMethods</a></code> package.</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Impute missing values in the data frame using the default minProb
## method.
imp_df1 &lt;- impute_na(raw_df, seed = 3312)


## Impute using the RF method with the number of iterations set at 5
## and number of trees set at 100.
imp_df2 &lt;- impute_na(raw_df,
  method = "RF",
  maxiter = 5, ntree = 100,
  seed = 3312
)


## Using the kNN method.
imp_df3 &lt;- impute_na(raw_df, method = "kNN", seed = 3312)



## Using the SVD method with n_pcs set to 3.
imp_df4 &lt;- impute_na(raw_df, method = "SVD", n_pcs = 3, seed = 3312)

## Using the minDet method with q set at 0.001.
imp_df5 &lt;- impute_na(raw_df, method = "minDet", q = 0.001, seed = 3312)

## Impute a normalized data set using the kNN method
imp_df6 &lt;- impute_na(ecoli_norm_df, method = "kNN")

</code></pre>

<hr>
<h2 id='impute_plot'>Visualize the impact of imputation</h2><span id='topic+impute_plot'></span>

<h3>Description</h3>

<p>This function generates density plots to visualize the impact of
missing data imputation on the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute_plot(
  original,
  imputed,
  global = TRUE,
  text_size = 10,
  palette = "viridis",
  n_row,
  n_col,
  save = FALSE,
  file_path = NULL,
  file_name = "Impute_plot",
  file_type = "pdf",
  plot_width = 7,
  plot_height = 7,
  dpi = 80
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="impute_plot_+3A_original">original</code></td>
<td>
<p>A <code>raw_df</code> object (output of <code><a href="#topic+create_df">create_df</a></code>)
containing missing values or a <code>norm_df</code> object containing normalized
protein intensity data.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_imputed">imputed</code></td>
<td>
<p>An <code>imp_df</code> object obtained from running <code>impute_na</code>
on the same data frame provided as <code>original</code>.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_global">global</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), a global density plot is
produced. If <code>FALSE</code>, sample-wise density plots are produced.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_n_row">n_row</code></td>
<td>
<p>Used if <code>global = FALSE</code> to indicate the number of rows
to print the plots.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_n_col">n_col</code></td>
<td>
<p>Used if <code>global = FALSE</code> to indicate the number of
columns to print the plots.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the density plot/s.
Default is <code>"Impute_plot."</code></p>
</td></tr>
<tr><td><code id="impute_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the density plot/s.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="impute_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p> Given two data frames, one with missing values
and the other, an imputed data frame (<code>imp_df</code> object) of the same
data set, <code>impute_plot</code> generates global or sample-wise density plots
to visualize the impact of imputation on the data set.
</p>
</li>
<li><p> Note, when sample-wise option is selected (<code>global = FALSE</code>),
<code>n_col</code> and <code>n_row</code> can be used to specify the number of columns
and rows to print the plots.
</p>
</li>
<li><p> If you choose to specify <code>n_row</code> and <code>n_col</code>, make sure that
<code>n_row</code> * <code>n_col</code> matches the total number of samples in the
data frame.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>ggplot2</code> plot object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Impute missing values in the data frame using the default minProb
## method.
imp_df &lt;- impute_na(raw_df)

## Visualize the impact of missing data imputation with a global density
## plot.
impute_plot(original = raw_df, imputed = imp_df)

## Make sample-wise density plots
impute_plot(raw_df, imp_df, global = FALSE)

## Print plots in user-specified numbers of rows and columns
impute_plot(raw_df, imp_df, global = FALSE, n_col = 2, n_row = 3)

</code></pre>

<hr>
<h2 id='norm_plot'>Visualize the effect of normalization</h2><span id='topic+norm_plot'></span>

<h3>Description</h3>

<p>This function visualizes the impact of normalization on
the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>norm_plot(
  original,
  normalized,
  type = "box",
  text_size = 10,
  palette = "viridis",
  save = FALSE,
  file_path = NULL,
  file_name = "Norm_plot",
  file_type = "pdf",
  dpi = 80,
  plot_width = 10,
  plot_height = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="norm_plot_+3A_original">original</code></td>
<td>
<p>A <code>raw_df</code> object (output of <code><a href="#topic+create_df">create_df</a></code>)
containing missing values, or an <code>imp_df</code> object after
imputing the missing values with <code>impute_na</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_normalized">normalized</code></td>
<td>
<p>A <code>norm_df</code> object after normalizing the data frame
provided as <code>original</code> using <code>normalize_data</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_type">type</code></td>
<td>
<p>Type of plot to generate. Choices are &quot;box&quot; or &quot;density.&quot; Default
is <code>"box."</code></p>
</td></tr>
<tr><td><code id="norm_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot.
Default is <code>"Norm_plot."</code></p>
</td></tr>
<tr><td><code id="norm_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="norm_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>7</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given two data frames, one with data prior to normalization
(<code>original</code>), and the other, after normalization (<code>normalized</code>),
<code>norm_plot</code> generates side-by-side plots to visualize the effect of
normalization on the protein intensity data.
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<ul>
<li> <p><code><a href="#topic+normalize_data">normalize_data</a></code>
</p>
</li>
<li> <p><code>create_df</code>
</p>
</li>
<li> <p><code>impute_na</code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Impute missing values in the data frame using the default minProb
## method.
imp_df &lt;- impute_na(raw_df)

## Normalize the imp_df object using the default quantile method
norm_df &lt;- normalize_data(imp_df)

## Visualize normalization using box plots
norm_plot(original = imp_df, normalized = norm_df)

## Visualize normalization using density plots
norm_plot(imp_df, norm_df, type = "density")


</code></pre>

<hr>
<h2 id='normalize_data'>Normalize intensity data</h2><span id='topic+normalize_data'></span>

<h3>Description</h3>

<p>This function normalizes data using a user-specified
normalization method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_data(df, method = "quantile")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_data_+3A_df">df</code></td>
<td>
<p>An <code>imp_df</code> object with missing values imputed using
<code>impute_na</code> or a <code>raw_df</code> object containing missing values.</p>
</td></tr>
<tr><td><code id="normalize_data_+3A_method">method</code></td>
<td>
<p>Name of the normalization method to use. Choices are
<code>"none", "scale", "quantile" or "cyclicloess."</code>
Default is <code>"quantile."</code></p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p><code>normalize_data</code> is a wrapper function around
the <code><a href="limma.html#topic+normalizeBetweenArrays">normalizeBetweenArrays</a></code> function from the
<code>limma</code> package. </p>
</li>
<li><p> This function normalizes
intensity values to achieve consistency among samples.
</p>
</li>
<li><p> It assumes that the intensities in the
data frame have been log-transformed, therefore, it is important to make sure
that <code>create_df</code> was run with <code>log_tr = TRUE</code>(default) when
creating the <code>raw_df</code> object.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>norm_df</code> object, which is a data frame of
normalized protein intensities.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<ul>
<li> <p><code>create_df</code>
</p>
</li>
<li> <p><code>impute_na</code>
</p>
</li>
<li><p> See <code><a href="limma.html#topic+normalizeBetweenArrays">normalizeBetweenArrays</a></code> in the R package
<code>limma</code> for more information on the different normalization methods
available.</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
  prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
  exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Impute missing values in the data frame using the default minProb
## method prioir to normalization.
imp_df &lt;- impute_na(raw_df)

## Normalize the imp_df object using the default quantile method
norm_df1 &lt;- normalize_data(imp_df)

## Use the cyclicloess method
norm_df2 &lt;- normalize_data(imp_df, method = "cyclicloess")

## Normalize data in the raw_df object prior to imputation.
norm_df3 &lt;- normalize_data(raw_df)

</code></pre>

<hr>
<h2 id='onegroup_only'>Proteins that are only expressed in a given group</h2><span id='topic+onegroup_only'></span>

<h3>Description</h3>

<p>This function outputs a list of proteins that are only
expressed (present) in one user-specified group while not expressed
(completely absent) in another user-specified group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onegroup_only(
  raw_df,
  abs_group,
  pres_group,
  set_na = 0.34,
  save = FALSE,
  file_path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="onegroup_only_+3A_raw_df">raw_df</code></td>
<td>
<p>A <code>raw_df</code> object (output of <code><a href="#topic+create_df">create_df</a></code>)</p>
</td></tr>
<tr><td><code id="onegroup_only_+3A_abs_group">abs_group</code></td>
<td>
<p>Name of the group in which proteins are not expressed.</p>
</td></tr>
<tr><td><code id="onegroup_only_+3A_pres_group">pres_group</code></td>
<td>
<p>Name of the group in which proteins are expressed.</p>
</td></tr>
<tr><td><code id="onegroup_only_+3A_set_na">set_na</code></td>
<td>
<p>The percentage of missing data allowed in <code>pres_group</code>.
Default is 0.34 (one third of the samples in the group).</p>
</td></tr>
<tr><td><code id="onegroup_only_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), it saves the output in a text
file named &quot;Group_<code>pres_group</code>_only.txt.&quot;</p>
</td></tr>
<tr><td><code id="onegroup_only_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: <code>onegroup_only</code> function assumes that column names in
the <code>raw_df</code> object provided as <code>df</code> follow &quot;Group_UniqueSampleID&quot;
notation. (Use <code>head(raw_df)</code> to check the structure of your
<code>raw_df</code> object.)
</p>
<ul>
<li><p> Given a pair of groups, <code>onegroup_only</code>
function finds proteins that are only expressed in <code>pres_group</code> while
completely absent or not expressed in <code>abs_group</code>.
</p>
</li>
<li><p> A text file containing majority protein IDs will be saved in a
temporary directory if <code>file_path</code> is not specified.</p>
</li></ul>



<h3>Value</h3>

<p>A list of majority protein IDs.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a raw_df object with default settings. No technical replicates.
raw_df &lt;- create_df(
prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg1.txt",
exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed1.txt"
)

## Find the proteins only expressed in group L, but absent in group H.
onegroup_only(raw_df, abs_group = "H",
pres_group = "L")

</code></pre>

<hr>
<h2 id='performance_plot'>Model performance plot</h2><span id='topic+performance_plot'></span>

<h3>Description</h3>

<p>This function generates plots to visualize model performance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>performance_plot(
  model_list,
  type = "box",
  text_size = 10,
  palette = "viridis",
  save = FALSE,
  file_path = NULL,
  file_name = "Performance_plot",
  file_type = "pdf",
  plot_width = 7,
  plot_height = 7,
  dpi = 80
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="performance_plot_+3A_model_list">model_list</code></td>
<td>
<p>A <code>model_list</code> object from performing
<code>train_models</code>.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_type">type</code></td>
<td>
<p>Type of plot to generate. Choices are &quot;box&quot; or &quot;dot.&quot;
Default is <code>"box."</code> for boxplots.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot.
Default is <code>"Performance_plot."</code></p>
</td></tr>
<tr><td><code id="performance_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="performance_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li> <p><code>performance_plot</code> uses resampling results from
models included in the <code>model_list</code> to generate plots showing model
performance.
</p>
</li>
<li><p> The default metrics used for classification based models are &quot;Accuracy&quot;
and &quot;Kappa.&quot;
</p>
</li>
<li><p> These metric types can be changed by providing additional arguments to
the <code>train_models</code> function. See <code><a href="caret.html#topic+train">train</a></code> and
<code><a href="caret.html#topic+trainControl">trainControl</a></code> for more information.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>ggplot2</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>train_models</code>
</p>
</li>
<li><p><code><a href="caret.html#topic+resamples">resamples</a></code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+train">train</a></code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+trainControl">trainControl</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>

## Create a model_df object
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Split the data frame into training and test data sets
covid_split_df &lt;- split_data(covid_model_df)

## Fit models based on the default list of machine learning (ML) algorithms
covid_model_list &lt;- train_models(covid_split_df)

## Generate box plots to visualize performance of different ML algorithms
performance_plot(covid_model_list)

## Generate dot plots
performance_plot(covid_model_list, type = "dot")

## Change color palette
performance_plot(covid_model_list, type = "dot", palette = "inferno")

</code></pre>

<hr>
<h2 id='pre_process'>Pre-process protein intensity data for modeling</h2><span id='topic+pre_process'></span>

<h3>Description</h3>

<p>This function pre-processes protein intensity data from
the top differentially expressed proteins identified with <code>find_dep</code> for
modeling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_process(
  fit_df,
  norm_df,
  sig = "adjP",
  sig_cutoff = 0.05,
  fc = 1,
  n_top = 20,
  find_highcorr = TRUE,
  corr_cutoff = 0.9,
  save_corrmatrix = FALSE,
  file_path = NULL,
  rem_highcorr = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre_process_+3A_fit_df">fit_df</code></td>
<td>
<p>A <code>fit_df</code> object from performing <code>find_dep</code>.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_norm_df">norm_df</code></td>
<td>
<p>The <code>norm_df</code> object from which the <code>fit_df</code> object
was obtained.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_sig">sig</code></td>
<td>
<p>Criteria to denote significance in differential expression.
Choices are <code>"adjP"</code> (default) for adjusted p-value or <code>"P"</code>
for p-value.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_sig_cutoff">sig_cutoff</code></td>
<td>
<p>Cutoff value for p-values and adjusted p-values in
differential expression. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_fc">fc</code></td>
<td>
<p>Minimum absolute log-fold change to use as threshold for
differential expression. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_n_top">n_top</code></td>
<td>
<p>The number of top hits from <code>find_dep</code> to be used in
modeling. Default is <code>20</code>.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_find_highcorr">find_highcorr</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), finds highly
correlated proteins.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_corr_cutoff">corr_cutoff</code></td>
<td>
<p>A numeric value specifying the correlation cutoff.
Default is <code>0.90</code>.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_save_corrmatrix">save_corrmatrix</code></td>
<td>
<p>Logical. If <code>TRUE</code>, saves a copy of the
protein correlation matrix in a tab-delimited text file labeled
&quot;Protein_correlation.txt&quot; in the directory specified by <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="pre_process_+3A_rem_highcorr">rem_highcorr</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), removes highly
correlated proteins (predictors or features).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates a data frame that contains protein intensities
for a user-specified number of top differentially expressed proteins.
</p>
<ul>
<li><p> Using <code>find_highcorr = TRUE</code>, highly correlated
proteins can be identified, and can be removed with
<code>rem_highcorr = TRUE</code>.
</p>
</li>
<li><p> Note: Most models will benefit from reducing correlation between
proteins (predictors or features), therefore we recommend removing those
proteins at this stage to reduce pairwise-correlation.
</p>
</li>
<li><p> If no or few proteins meet the significance threshold for differential
expression, you may adjust <code>sig</code>, <code>fc</code>, and/or <code>sig_cutoff</code>
accordingly to obtain more proteins for modeling.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>model_df</code> object, which is a data frame of protein
intensities with proteins indicated by columns.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>find_dep</code>, <code>normalize_data</code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+findCorrelation">caret: findCorrelation</a></code></p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Create a model_df object with default settings.
covid_model_df1 &lt;- pre_process(fit_df = covid_fit_df, norm_df = covid_norm_df)

## Change the correlation cutoff.
covid_model_df2 &lt;- pre_process(covid_fit_df, covid_norm_df, corr_cutoff = 0.95)

## Change the significance criteria to include more proteins
covid_model_df3 &lt;- pre_process(covid_fit_df, covid_norm_df, sig = "P")

## Change the number of top differentially expressed proteins to include
covid_model_df4 &lt;- pre_process(covid_fit_df, covid_norm_df, sig = "P", n_top = 24)

</code></pre>

<hr>
<h2 id='rem_feature'>Remove user-specified proteins (features) from a data frame</h2><span id='topic+rem_feature'></span>

<h3>Description</h3>

<p>This function removes user-specified proteins from a <code>model_df</code>
object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rem_feature(model_df, rem_protein)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rem_feature_+3A_model_df">model_df</code></td>
<td>
<p>A <code>model_df</code> object.</p>
</td></tr>
<tr><td><code id="rem_feature_+3A_rem_protein">rem_protein</code></td>
<td>
<p>Name of the protein to remove.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p> After visualizing protein intensity variation
among conditions with <code>feature_plot</code> or after assessing the importance
of each protein in models using <code>varimp_plot</code>, you can choose to remove
specific proteins (features) from the data frame. </p>
</li>
<li><p> For example, you can
choose to remove a protein from the <code>model_df</code> object if the protein
does not show distinct patterns of variation among conditions. This protein
may show mostly overlapping distributions in the feature plots.
</p>
</li>
<li><p> Another incidence would be removing a protein that is very low in
variable importance in the models built using <code>train_models</code>. You can
visualize variable importance using <code>varimp_plot</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>model_df</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_plot">feature_plot</a></code>, <code><a href="#topic+pre_process">pre_process</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
covid_model_df &lt;- pre_process(fit_df = covid_fit_df, norm_df = covid_norm_df)

## Remove sp|P22352|GPX3_HUMAN protein from the model_df object
covid_model_df1 &lt;- rem_feature(covid_model_df, rem_protein = "sp|P22352|GPX3_HUMAN")

</code></pre>

<hr>
<h2 id='rem_sample'>Remove user-specified samples</h2><span id='topic+rem_sample'></span>

<h3>Description</h3>

<p>This function removes user-specified samples from the
data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rem_sample(raw_df, rem)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rem_sample_+3A_raw_df">raw_df</code></td>
<td>
<p>A <code>raw_df</code> object.</p>
</td></tr>
<tr><td><code id="rem_sample_+3A_rem">rem</code></td>
<td>
<p>Name of the sample to remove.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li><p><code>rem_sample</code> assumes that sample names follow the
&quot;Group_UniqueSampleID_TechnicalReplicate&quot; notation (Use <code>head(raw_df)</code>
to see the structure of the <code>raw_df</code> object.)
</p>
</li>
<li><p> If all the technical replicates representing a sample needs to be
removed, provide &quot;Group_UniqueSampleID&quot; as <code>rem</code>.
</p>
</li>
<li><p> If a specific technical replicate needs to be removed in case it
shows weak correlation with other technical replicates for example, you can
remove that particular technical replicate by providing
&quot;Group_UniqueSampleID_TechnicalReplicate&quot; as <code>rem</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>raw_df</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>

<p><code><a href="#topic+corr_plot">corr_plot</a></code>, <code><a href="#topic+create_df">create_df</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Use a data set containing technical replicates to create a raw_df object
raw_df &lt;- create_df(
prot_groups = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/pg2.txt",
exp_design = "https://raw.githubusercontent.com/caranathunge/promor_example_data/main/ed2.txt",
tech_reps = TRUE
)
# Check the first few rows of the raw_df object
head(raw_df)

## Remove all technical replicates of "WT_4"
raw_df1 &lt;- rem_sample(raw_df, "WT_4")

## Remove only technical replicate number 2 of "WT_4"
raw_df2 &lt;- rem_sample(raw_df, "WT_4_2")

</code></pre>

<hr>
<h2 id='roc_plot'>ROC plot</h2><span id='topic+roc_plot'></span>

<h3>Description</h3>

<p>This function generates Receiver Operating Characteristic (ROC)
curves to evaluate models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roc_plot(
  probability_list,
  split_df,
  ...,
  multiple_plots = TRUE,
  text_size = 10,
  palette = "viridis",
  save = FALSE,
  file_path = NULL,
  file_name = "ROC_plot",
  file_type = "pdf",
  plot_width = 7,
  plot_height = 7,
  dpi = 80
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roc_plot_+3A_probability_list">probability_list</code></td>
<td>
<p>A <code>probability_list</code> object from performing
<code>test_models</code> with <code>type = "prob"</code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_split_df">split_df</code></td>
<td>
<p>A <code>split_df</code> object from performing <code>split_data</code></p>
</td></tr>
<tr><td><code id="roc_plot_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to
<code><a href="pROC.html#topic+roc">roc</a></code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_multiple_plots">multiple_plots</code></td>
<td>
<p>Logical. If <code>FALSE</code> plots all ROC curves
representing algorithms included in the <code>probability_list</code> in a single
plot.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot.
Default is <code>"ROC_plot."</code></p>
</td></tr>
<tr><td><code id="roc_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="roc_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li> <p><code>roc_plot</code> first uses probabilities generated
during <code>test_models</code> to build a ROC object.
</p>
</li>
<li><p> Next, relevant information is extracted from the ROC object to
plot the ROC curves.</p>
</li></ul>



<h3>Value</h3>

<p>A <code>ggplot2</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>test_models</code>
</p>
</li>
<li><p><code><a href="pROC.html#topic+roc">roc</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Create a model_df object
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Split the data frame into training and test data sets
covid_split_df &lt;- split_data(covid_model_df)

## Fit models using the default list of machine learning (ML) algorithms
covid_model_list &lt;- train_models(covid_split_df)

# Test a list of models on a test data set and output class probabilities,
covid_prob_list &lt;- test_models(covid_model_list, covid_split_df, type = "prob")

## Plot ROC curves separately for each ML algorithm
roc_plot(covid_prob_list, covid_split_df)

## Plot all ROC curves in one plot
roc_plot(covid_prob_list, covid_split_df, multiple_plots = FALSE)

## Change color palette
roc_plot(covid_prob_list, covid_split_df, palette = "plasma")

</code></pre>

<hr>
<h2 id='split_data'>Split the data frame to create training and test data</h2><span id='topic+split_data'></span>

<h3>Description</h3>

<p>This function can be used to create balanced splits of the
protein intensity data in a <code>model_df</code> object to create training and test data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_data(model_df, train_size = 0.8, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split_data_+3A_model_df">model_df</code></td>
<td>
<p>A <code>model_df</code> object from performing <code>pre_process</code>.</p>
</td></tr>
<tr><td><code id="split_data_+3A_train_size">train_size</code></td>
<td>
<p>The size of the training data set as a proportion of the
complete data set. Default is 0.8.</p>
</td></tr>
<tr><td><code id="split_data_+3A_seed">seed</code></td>
<td>
<p>Numerical. Random number seed. Default is <code>NULL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function splits the <code>model_df</code> object in to training and
test data sets using random sampling while preserving the original
class distribution of the data. Make sure to fix the random number seed with
<code>seed</code> for reproducibility
</p>


<h3>Value</h3>

<p>A list of data frames.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>pre_process</code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+createDataPartition">createDataPartition</a></code></p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Create a model_df object
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Split the data frame into training and test data sets using default settings
covid_split_df1 &lt;- split_data(covid_model_df, seed = 8314)

## Split the data frame into training and test data sets with 70% of the
## data in training and 30% in test data sets
covid_split_df2 &lt;- split_data(covid_model_df, train_size = 0.7, seed = 8314)

## Access training data set
covid_split_df1$training

## Access test data set
covid_split_df1$test

</code></pre>

<hr>
<h2 id='test_models'>Test machine learning models on test data</h2><span id='topic+test_models'></span>

<h3>Description</h3>

<p>This function can be used to predict test data using models
generated by different machine learning algorithms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_models(
  model_list,
  split_df,
  type = "prob",
  save_confusionmatrix = FALSE,
  file_path = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test_models_+3A_model_list">model_list</code></td>
<td>
<p>A <code>model_list</code> object from performing
<code>train_models</code>.</p>
</td></tr>
<tr><td><code id="test_models_+3A_split_df">split_df</code></td>
<td>
<p>A <code>split_df</code> object from performing <code>split_data</code>.</p>
</td></tr>
<tr><td><code id="test_models_+3A_type">type</code></td>
<td>
<p>Type of output. Set <code>type</code> as &quot;prob&quot; (default) to output
class probabilities, and &quot;raw&quot; to output class predictions.</p>
</td></tr>
<tr><td><code id="test_models_+3A_save_confusionmatrix">save_confusionmatrix</code></td>
<td>
<p>Logical. If <code>TRUE</code>, a tab-delimited
text file (&quot;Confusion_matrices.txt&quot;) with confusion matrices in the
long-form data format will be saved in the directory specified by
<code>file_path</code>.
See below for more details.</p>
</td></tr>
<tr><td><code id="test_models_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="test_models_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to
<code><a href="stats.html#topic+predict">predict</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li> <p><code>test_models</code> function uses
models obtained from <code>train_models</code> to predict a given test data set.
</p>
</li>
<li><p> Setting <code>type = "raw"</code> is required to obtain confusion matrices.
</p>
</li>
<li><p> Setting <code>type = "prob"</code> (default) will output a list of
probabilities that can be used to generate ROC curves using <code>roc_plot</code>.</p>
</li></ul>



<h3>Value</h3>

<ul>
<li> <p><code>probability_list</code>: If <code>type = "prob"</code>, a list of
data frames containing class probabilities for each method in the
<code>model_list</code> will be returned.
</p>
</li>
<li> <p><code>prediction_list</code>: If <code>type = "raw"</code>, a list of factors
containing class predictions for each method will be returned.</p>
</li></ul>



<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>split_df</code>
</p>
</li>
<li> <p><code>train_models</code>
</p>
</li>
<li> <p><code><a href="stats.html#topic+predict">predict</a></code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+confusionMatrix">confusionMatrix</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Create a model_df object
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Split the data frame into training and test data sets
covid_split_df &lt;- split_data(covid_model_df)

## Fit models using the default list of machine learning (ML) algorithms
covid_model_list &lt;- train_models(covid_split_df)

# Test a list of models on a test data set and output class probabilities,
covid_prob_list &lt;- test_models(model_list = covid_model_list, split_df = covid_split_df)


## Not run: 
# Save confusion matrices in the working directory and output class predictions
covid_pred_list &lt;- test_models(
  model_list = covid_model_list,
  split_df = covid_split_df,
  type = "raw",
  save_confusionmatrix = TRUE,
  file_path = "."
)

## End(Not run)

</code></pre>

<hr>
<h2 id='train_models'>Train machine learning models on training data</h2><span id='topic+train_models'></span>

<h3>Description</h3>

<p>This function can be used to train models on protein intensity
data using different machine learning algorithms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>train_models(
  split_df,
  resample_method = "repeatedcv",
  resample_iterations = 10,
  num_repeats = 3,
  algorithm_list,
  seed = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="train_models_+3A_split_df">split_df</code></td>
<td>
<p>A <code>split_df</code> object from performing <code>split_data</code>.</p>
</td></tr>
<tr><td><code id="train_models_+3A_resample_method">resample_method</code></td>
<td>
<p>The resampling method to use. Default is
<code>"repeatedcv"</code> for repeated cross validation.
See <code><a href="caret.html#topic+trainControl">trainControl</a></code> for
details on other available methods.</p>
</td></tr>
<tr><td><code id="train_models_+3A_resample_iterations">resample_iterations</code></td>
<td>
<p>Number of resampling iterations. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="train_models_+3A_num_repeats">num_repeats</code></td>
<td>
<p>The number of complete sets of folds to compute (For
<code>resampling method = "repeatedcv"</code> only).</p>
</td></tr>
<tr><td><code id="train_models_+3A_algorithm_list">algorithm_list</code></td>
<td>
<p>A list of classification or regression algorithms to
use.
A full list of machine learning algorithms available through
the <code>caret</code> package can be found here:
<a href="http://topepo.github.io/caret/train-models-by-tag.html">http://topepo.github.io/caret/train-models-by-tag.html</a>. See below for
default options.</p>
</td></tr>
<tr><td><code id="train_models_+3A_seed">seed</code></td>
<td>
<p>Numerical. Random number seed. Default is <code>NULL</code></p>
</td></tr>
<tr><td><code id="train_models_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to
<code><a href="caret.html#topic+train">train</a></code> function in the <code>caret</code> package.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li> <p><code>train_models</code> function can be used to first
define the control parameters to be used in training models, calculate
resampling-based performance measures for models based on a given set of
machine-learning algorithms, and output the best model for each algorithm.
</p>
</li>
<li><p> In the event that <code>algorithm_list</code> is not provided, a default
list of four classification-based machine-learning algorithms will be used
for building and training models. Default <code>algorithm_list</code>:
&quot;svmRadial&quot;, &quot;rf&quot;, &quot;glm&quot;, &quot;xgbLinear, and &quot;naive_bayes.&quot;
</p>
</li>
<li><p> Note: Models that fail to build are removed from the output.
</p>
</li>
<li><p> Make sure to fix the random number seed with
<code>seed</code> for reproducibility</p>
</li></ul>



<h3>Value</h3>

<p>A list of class <code>train</code> for each machine-learning algorithm.
See <code><a href="caret.html#topic+train">train</a></code> for more information on accessing
different elements of this list.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>References</h3>

<p>Kuhn, Max. &quot;Building predictive models in R using the caret
package.&quot; Journal of statistical software 28 (2008): 1-26.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>pre_process</code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+trainControl">trainControl</a></code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+train">train</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>

## Create a model_df object
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Split the data frame into training and test data sets
covid_split_df &lt;- split_data(covid_model_df, seed = 8314)

## Fit models based on the default list of machine learning (ML) algorithms
covid_model_list1 &lt;- train_models(split_df = covid_split_df, seed = 351)

## Fit models using a user-specified list of ML algorithms.
covid_model_list2 &lt;- train_models(
  covid_split_df,
  algorithm_list = c("svmRadial", "glmboost"),
  seed = 351
)

## Change resampling method and resampling iterations.
covid_model_list3 &lt;- train_models(
  covid_split_df,
  resample_method = "cv",
  resample_iterations = 50,
  seed = 351
)


</code></pre>

<hr>
<h2 id='varimp_plot'>Variable importance plot</h2><span id='topic+varimp_plot'></span>

<h3>Description</h3>

<p>This function visualizes variable importance in models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varimp_plot(
  model_list,
  ...,
  type = "lollipop",
  text_size = 10,
  palette = "viridis",
  n_row,
  n_col,
  save = FALSE,
  file_path = NULL,
  file_name = "VarImp_plot",
  file_type = "pdf",
  dpi = 80,
  plot_width = 7,
  plot_height = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varimp_plot_+3A_model_list">model_list</code></td>
<td>
<p>A <code>model_list</code> object from performing
<code>train_models</code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed on to
<code><a href="caret.html#topic+varImp">varImp</a></code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_type">type</code></td>
<td>
<p>Type of plot to generate. Choices are &quot;bar&quot; or &quot;lollipop.&quot;
Default is <code>"lollipop."</code></p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for plot labels, axis labels etc. Default is
<code>10</code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_n_row">n_row</code></td>
<td>
<p>Number of rows to print the plots.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_n_col">n_col</code></td>
<td>
<p>Number of columns to print the plots.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot.
Default is <code>"VarImp_plot."</code></p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot.
Default is <code>"pdf"</code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80</code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is <code>7</code>.</p>
</td></tr>
<tr><td><code id="varimp_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is <code>7</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<ul>
<li> <p><code>varimp_plot</code> produces a list of plots showing
variable importance measures calculated from models generated with different
machine-learning algorithms.
</p>
</li>
<li><p> Note: Variables are ordered by variable importance in
descending order, and by default, importance values are scaled to 0 and 100.
This can be changed by specifying <code>scale = FALSE</code>. See
<code><a href="caret.html#topic+varImp">varImp</a></code> for more information.</p>
</li></ul>



<h3>Value</h3>

<p>A list of <code>ggplot2</code> objects.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>train_models</code>, <code>rem_feature</code>
</p>
</li>
<li> <p><code><a href="caret.html#topic+varImp">varImp</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>

## Create a model_df object
covid_model_df &lt;- pre_process(covid_fit_df, covid_norm_df)

## Split the data frame into training and test data sets
covid_split_df &lt;- split_data(covid_model_df)

## Fit models based on the default list of machine learning (ML) algorithms
covid_model_list &lt;- train_models(covid_split_df)

## Variable importance - lollipop plots
varimp_plot(covid_model_list)

## Bar plots
varimp_plot(covid_model_list, type = "bar")

## Do not scale variable importance values
varimp_plot(covid_model_list, scale = FALSE)

## Change color palette
varimp_plot(covid_model_list, palette = "magma")

</code></pre>

<hr>
<h2 id='volcano_plot'>Volcano plot</h2><span id='topic+volcano_plot'></span>

<h3>Description</h3>

<p>This function generates volcano plots to visualize
differentially expressed proteins between groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcano_plot(
  fit_df,
  adj_method = "BH",
  sig = "adjP",
  cutoff = 0.05,
  lfc = 1,
  line_fc = TRUE,
  line_p = TRUE,
  palette = "viridis",
  text_size = 10,
  label_top = FALSE,
  n_top = 10,
  save = FALSE,
  file_path = NULL,
  file_name = "Volcano_plot",
  file_type = "pdf",
  plot_height = 7,
  plot_width = 7,
  dpi = 80
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcano_plot_+3A_fit_df">fit_df</code></td>
<td>
<p>A <code>fit_df</code> object from performing <code>find_dep</code>.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_adj_method">adj_method</code></td>
<td>
<p>Method used for adjusting the p-values for multiple
testing. Default is <code>"BH"</code>.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_sig">sig</code></td>
<td>
<p>Criteria to denote significance. Choices are <code>"adjP"</code>
(default) for adjusted p-value or <code>"P"</code> for p-value.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_cutoff">cutoff</code></td>
<td>
<p>Cutoff value for p-values and adjusted p-values. Default is
0.05.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_lfc">lfc</code></td>
<td>
<p>Minimum absolute log2-fold change to use as threshold for
differential expression.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_line_fc">line_fc</code></td>
<td>
<p>Logical. If <code>TRUE</code>(default), a dotted line will be shown
to indicate the <code>lfc</code> threshold in the plot.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_line_p">line_p</code></td>
<td>
<p>Logical. If <code>TRUE</code>(default), a dotted line will be shown
to indicate the p-value or adjusted p-value <code>cutoff.</code></p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_palette">palette</code></td>
<td>
<p>Viridis color palette option for plots. Default is
<code>"viridis"</code>. See
<code><a href="viridis.html#topic+viridis">viridis</a></code>
for available options.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_text_size">text_size</code></td>
<td>
<p>Text size for axis text, labels etc.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_label_top">label_top</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default), labels are added to the
dots to indicate protein names.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_n_top">n_top</code></td>
<td>
<p>The number of top hits to label with protein name when
<code>label_top = TRUE.</code> Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_save">save</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves a copy of the plot in the
directory provided in <code>file_path</code>.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_file_path">file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_file_name">file_name</code></td>
<td>
<p>File name to save the plot. Default is &quot;Volcano_plot.&quot;</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save the plot. Default is <code>"pdf".</code></p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_plot_height">plot_height</code></td>
<td>
<p>Height of the plot. Default is 7.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_plot_width">plot_width</code></td>
<td>
<p>Width of the plot. Default is 7.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_dpi">dpi</code></td>
<td>
<p>Plot resolution. Default is <code>80.</code></p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> Volcano plots show log-2-fold change on the x-axis,
and based on the significance criteria chosen, either -log10(p-value) or
-log10(adjusted p-value) on the y-axis.
</p>
</li>
<li> <p><code>volcano_plot</code> requires a <code>fit_df</code> object from performing
differential expression analysis with <code>find_dep.</code>
</p>
</li>
<li><p> User has the option to choose criteria that denote significance.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>ggplot2</code> plot object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>find_dep</code>
</p>
</li>
<li> <p><code><a href="limma.html#topic+topTable">topTable</a></code> and
<code><a href="limma.html#topic+lmFit">lmFit</a></code> functions from the
<code><a href="limma.html#topic+limma">limma</a></code> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Create a volcano plot with default settings.
volcano_plot(ecoli_fit_df)

## Change significance criteria and cutoff
volcano_plot(ecoli_fit_df, cutoff = 0.1, sig = "P")

## Label top 30 differentially expressed proteins and
## change the color palette of the plot
volcano_plot(ecoli_fit_df, label_top = TRUE, n_top = 30, palette = "mako")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
