<!DOCTYPE html><html><head><title>Help for package propOverlap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {propOverlap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CI.emprical'>
<p>Computing the Core Intervals for Both Classes.</p></a></li>
<li><a href='#GMask'>
<p>Producing Gene Masks.</p></a></li>
<li><a href='#leukaemia'>
<p>Leukaemia data set.</p></a></li>
<li><a href='#lung'>
<p>Lung cancer data set.</p></a></li>
<li><a href='#POS'>
<p>Calculating the proportional Overlapping Scores.</p></a></li>
<li><a href='#propOverlap-internal'><p>Internal <code>propOverlap</code> functions.</p></a></li>
<li><a href='#propOverlap-package'><p>Feature (gene) selection based on the Proportional Overlapping Scores.</p></a></li>
<li><a href='#RDC'>
<p>Assiging the Relative Dominant Class.</p></a></li>
<li><a href='#Sel.Features'>
<p>Gene (Feature) Selection.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Feature (gene) selection based on the Proportional Overlapping
Scores</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2014-09-15</td>
</tr>
<tr>
<td>Author:</td>
<td>Osama Mahmoud, Andrew Harrison, Aris Perperoglou, Asma Gul, Zardad Khan, Berthold Lausen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Osama Mahmoud &lt;ofamah@essex.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A package for selecting the most relevant features (genes) in the high-dimensional binary classification problems. The discriminative features are identified using analyzing the overlap between the expression values across both classes. The package includes functions for measuring the proportional overlapping score for each gene avoiding the outliers effect. The used measure for the overlap is the one defined in the "Proportional Overlapping Score (POS)" technique for feature selection. A gene mask which represents a gene's classification power can also be produced for each gene (feature). The set size of the selected genes might be set by the user. The minimum set of genes that correctly classify the maximum number of the given tissue samples (observations) can be also produced.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), Biobase</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2014-09-15 12:28:08 UTC; osama</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2014-09-15 17:06:03</td>
</tr>
</table>
<hr>
<h2 id='CI.emprical'>
Computing the Core Intervals for Both Classes.
</h2><span id='topic+CI.emprical'></span>

<h3>Description</h3>

<p><code>CI.emprical</code> is used to compute the core interval boundaries for each class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CI.emprical(ES, Y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CI.emprical_+3A_es">ES</code></td>
<td>

<p>gene (feature) matrix: P, number of genes, by N, number of samples(observations).
</p>
</td></tr>
<tr><td><code id="CI.emprical_+3A_y">Y</code></td>
<td>

<p>a vector of length N for samples' class label.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>CI.emprical</code> returns an object of class &ldquo;data.frame&rdquo; which has P rows and 4 columns. The first two columns represent a1, the minimum boundary of the first class, and b1, the maximum boundary of the first class, respectively. Whereas, the last two columns represent a2, the minimum boundary of the second class, and b2, the maximum boundary of the second class, respectively.
</p>


<h3>Author(s)</h3>

<p>Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
GenesExpression &lt;- lung[1:12533,]   #define the features matrix
Class           &lt;- lung[12534,]     #define the observations' class labels
CoreIntervals   &lt;- CI.emprical(GenesExpression, Class)
CoreIntervals[1:10,]                #show classes' core interval for the first 10 features
</code></pre>

<hr>
<h2 id='GMask'>
Producing Gene Masks.
</h2><span id='topic+GMask'></span>

<h3>Description</h3>

<p><code>GMask</code> produces the masks of features (genes). Each gene mask reports the samples that can unambiguously be assigned to their correct target classes by this gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GMask(ES, Core, Y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GMask_+3A_es">ES</code></td>
<td>

<p>gene (feature) matrix: P, number of genes, by N, number of samples(observations).
</p>
</td></tr>
<tr><td><code id="GMask_+3A_core">Core</code></td>
<td>

<p>a <code>data.frame</code> of the core interval boundaries for both classes. It should have the same number of rows as <code>ES</code> and 4 columns (the minimum and the maximum of the first class's core interval followed by the minimum and the maximum of the second class's core interval). See the returned value of the <code><a href="#topic+CI.emprical">CI.emprical</a></code>.
</p>
</td></tr>
<tr><td><code id="GMask_+3A_y">Y</code></td>
<td>

<p>a vector of length N for samples' class label.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>GMask</code> gives the gene masks that can represent the capability of genes to correctly classify each sample. Such a mask represents a gene's classification power. Each element of a mask is set either to 1 or 0 based on whether the corresponding sample (observation) could be unambiguously assign to its correct target class by the considered gene or not respectively.
</p>


<h3>Value</h3>

<p>It returns a P by N matrix with elements of zeros and ones.
</p>


<h3>Author(s)</h3>

<p>Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CI.emprical">CI.emprical</a></code> for the core interval boundaries.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukaemia)
GenesExpression &lt;- leukaemia[1:7129,] #define the features matrix
Class           &lt;- leukaemia[7130,]   #define the observations' class labels
Gene.Masks      &lt;- GMask(GenesExpression, CI.emprical(GenesExpression, Class), Class)
Gene.Masks[1:100,]                    #show the masks of the first 100 features
</code></pre>

<hr>
<h2 id='leukaemia'>
Leukaemia data set.
</h2><span id='topic+leukaemia'></span>

<h3>Description</h3>

<p>The leukemia dataset was taken from a collection of leukemia patient samples reported by Golub et. al., (1999). This dataset often serves as a benchmark for microarray analysis methods. It contains gene expressions corresponding to acute lymphoblast leukemia (ALL) and acute myeloid leukemia (AML) samples from bone marrow and peripheral blood. The dataset consisted of 72 samples: 49 samples of ALL; 23 samples of AML. Each sample is measured over 7,129 genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leukaemia)</code></pre>


<h3>Format</h3>

<p>A matrix with 7130 rows (7129 rows show the gene expressions while the last row reports the corresponding sample's class label), and 72 columns represent the samples. The samples class's label coded as follows:
</p>

<dl>
<dt><code>1</code></dt><dd><p>acute lymphoblast leukemia sample (ALL).</p>
</dd>
<dt><code>2</code></dt><dd><p>acute myeloid leukemia sample (AML).</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://cilab.ujn.edu.cn/datasets.htm">http://cilab.ujn.edu.cn/datasets.htm</a>
</p>


<h3>References</h3>

<p>Golub TR, Slonim DK, Tamayo P, Huard C, Gaasenbeek M, Mesirov JP, Coller H, Loh ML, Downing JR, Caligiuri MA, Bloomfield CD, Lander ES. (1999) <em>Molecular classification of cancer: class discovery and class prediction by gene expression monitoring</em>. Science: 286 (5439), 531-537.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukaemia)
str(leukaemia)
</code></pre>

<hr>
<h2 id='lung'>
Lung cancer data set.
</h2><span id='topic+lung'></span>

<h3>Description</h3>

<p>Gene expression data for lung cancer classification between two classes: adenocarcinoma (ADCA); malignant pleural mesothe-lioma (MPM). The lung data set contains 181 tissue samples (150 ADCA and 31 MPM). Each sample is described by 12533 genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lung)</code></pre>


<h3>Format</h3>

<p>A matrix with 12534 rows (12533 rows show the gene expressions for 181 tissue samples, reported in columns, while the last row reports the corresponding sample's class label). The samples class's label coded as follows:
</p>

<dl>
<dt><code>1</code></dt><dd><p>adenocarcinoma sample (ADCA).</p>
</dd>
<dt><code>2</code></dt><dd><p>malignant pleural mesothe-lioma sample (MPM).</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://cilab.ujn.edu.cn/datasets.htm">http://cilab.ujn.edu.cn/datasets.htm</a>
</p>


<h3>References</h3>

<p>Gordon GJ, Jensen RV, Hsiao L-L, Gullans SR, Blumenstock JE, Ramaswamy S, Richards WG, Sugarbaker DJ, Bueno R. (2002) <em>Translation of microarray data into clinically relevant cancer diagnostic tests using gene expression ratios in lung cancer and mesothelioma</em>. Cancer research: 62(17), 4963-4967.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
str(lung)
</code></pre>

<hr>
<h2 id='POS'>
Calculating the proportional Overlapping Scores.
</h2><span id='topic+POS'></span>

<h3>Description</h3>

<p><code>POS</code> computes the proportional overlapping scores of the given genes (features). This score measures the overlap degree between gene expression values across various classes. It produces a value lies in the interval [0,1]. A lower score denotes gene with higher discriminative power for the considered classification problem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>POS(ES, Core, Y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="POS_+3A_es">ES</code></td>
<td>

<p>gene (feature) matrix: P, number of genes, by N, number of samples(observations).
</p>
</td></tr>
<tr><td><code id="POS_+3A_core">Core</code></td>
<td>

<p>a <code>data.frame</code> of the core interval boundaries for both classes. It should have the same number of rows as <code>ES</code> and 4 columns (the minimum and the maximum of the first class's core interval followed by the minimum and the maximum of the second class's core interval). See the returned value of the <code><a href="#topic+CI.emprical">CI.emprical</a></code>.
</p>
</td></tr>
<tr><td><code id="POS_+3A_y">Y</code></td>
<td>

<p>a vector of length N for samples' class label.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each gene, <code>POS</code> computes a measure that estimates the overlapping degree between the expression intervals of different classes. For estimating the overlap, POS measure takes into account three factors: the length of the overlapping region; number of the overlapped samples (observations); the proportion of each class's overlapped samples to the total number of overlapping samples.
</p>


<h3>Value</h3>

<p>It returns a vector of length P for &lsquo;POS&rsquo; measures of all genes (features).
</p>


<h3>Author(s)</h3>

<p>Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CI.emprical">CI.emprical</a></code> for the core interval boundaries and <code><a href="#topic+GMask">GMask</a></code> for the gene masks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukaemia)
Score &lt;- POS(leukaemia[1:7129,], CI.emprical(leukaemia[1:7129,],
leukaemia[7130,]), leukaemia[7130,])
Score[1:5]      #show the proportional overlapping scores for the first 5 features
summary(Score)  #show the the summary of the scores of all features.
</code></pre>

<hr>
<h2 id='propOverlap-internal'>Internal <code>propOverlap</code> functions.</h2><span id='topic+block'></span><span id='topic+Min.Set'></span>

<h3>Description</h3>

<p>Internal <code>propOverlap</code> functions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>block(...)
Min.Set(GMask, Scores)
</code></pre>


<h3>Details</h3>

<p>These are not to be called by the user</p>

<hr>
<h2 id='propOverlap-package'>Feature (gene) selection based on the Proportional Overlapping Scores.</h2><span id='topic+propOverlap-package'></span><span id='topic+propOverlap'></span>

<h3>Description</h3>

<p>A package for selecting the most relevant features (genes) in the high-dimensional binary classification problems. The discriminative features are identified using analyzing the overlap between the expression values across both classes. The package includes functions for measuring the proportional overlapping score for each gene avoiding the outliers effect. The used measure of the overlap is the one defined in the &ldquo;Proportional Overlapping Score (<b>POS</b>)&rdquo; technique for feature selection, see &lsquo;References&rsquo; section below. A gene mask which represents a gene's classification power can also be produced for each gene (feature). The set size of the selected genes might be set by the user. The minimum set of genes that correctly classify the maximum number of the given tissue samples (observations) can be also produced.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> propOverlap</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2014-09-15</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Osama Mahmoud, Andrew Harrison, Aris Perperoglou, Asma Gul, Zardad Khan, Berthold Lausen
Maintainer: Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274</p>

<hr>
<h2 id='RDC'>
Assiging the Relative Dominant Class.
</h2><span id='topic+RDC'></span>

<h3>Description</h3>

<p><code>RDC</code> associates genes (features) with the class which it is more able to distingish. For each gene, a class that has the highest proportion, relative to classes' size, of correctly assigned samples (observations) is reported as the relative dominant class for the considered gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RDC(GMask, Y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDC_+3A_gmask">GMask</code></td>
<td>

<p>gene (feature) mask matrix: P, number of genes, by N, number of samples(observations) with elements of zeros and ones. See the returned value of the <code><a href="#topic+GMask">GMask</a></code>.
</p>
</td></tr>
<tr><td><code id="RDC_+3A_y">Y</code></td>
<td>

<p>a vector of length N for samples' class label.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>RDC</code> returns a vector of length P. Each element's value is either 1 or 2 indicating which class label is reported as the relative dominant class for the corresponding gene (feature).
</p>


<h3>Author(s)</h3>

<p>Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GMask">GMask</a></code> for gene (feature) mask matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
Class           &lt;- lung[12534,]   #define the observations' class labels
Gene.Masks      &lt;- GMask(lung[1:12533,], CI.emprical(lung[1:12533,], Class), Class)
RelativeDC      &lt;- RDC(Gene.Masks, Class)
RelativeDC[1:10]                  #show the relative dominant classes for the first 10 features
table(RelativeDC)                 #show the number of assignments for each class
</code></pre>

<hr>
<h2 id='Sel.Features'>
Gene (Feature) Selection.
</h2><span id='topic+Sel.Features'></span>

<h3>Description</h3>

<p><code>Sel.Feature</code> selects the most discriminative genes (features) among the given ones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sel.Features(ES, Y, K = "Min", Verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Sel.Features_+3A_es">ES</code></td>
<td>

<p>gene (feature) matrix: P, number of genes, by N, number of samples (observations).
</p>
</td></tr>
<tr><td><code id="Sel.Features_+3A_y">Y</code></td>
<td>

<p>a vector of length N for samples' class label.
</p>
</td></tr>
<tr><td><code id="Sel.Features_+3A_k">K</code></td>
<td>

<p>the number of genes to be selected. The default is to give the minimum subset of genes that correctly classify the maximum number of the given tissue samples (observations). Alternatively, <code>K</code> should be a positive integer.
</p>
</td></tr>
<tr><td><code id="Sel.Features_+3A_verbose">Verbose</code></td>
<td>

<p>logical. If <code>TRUE</code>, more information about the selected genes are returned. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Sel.Feature</code> selects the most relevant genes (features) in the high-dimensional binary classification problems. The discriminative genes are identified using analyzing the overlap between the expression values across both classes. The &ldquo;<b>POS</b>&rdquo; technique has been applied to produce the selected set of genes. A proportional overlapping score measures the overlapping degree avoiding the outliers effect for each gene. Each gene is described by a robust mask that represents its discriminative power. The constructed masks along with the gene scores are exploited to produce the selected subset of genes.
</p>


<h3>Value</h3>

<p>If <code>K</code> is specified as &lsquo;Min&rsquo; (the default), a list containing the following components is returned:
</p>
<table>
<tr><td><code>Features</code></td>
<td>

<p>A matrix of the indices of selected genes with their POS measures. See <code><a href="#topic+POS">POS</a></code>.
</p>
</td></tr>
<tr><td><code>Covered.Obs</code></td>
<td>

<p>A vector showing the indices of the observations that have been covered by the returned minimum subset of genes.
</p>
</td></tr>
</table>
<p>If <code>K</code> is specified as a positive integer, a list containing the following components is returned:
</p>
<table>
<tr><td><code>features</code></td>
<td>

<p>A vector of the indices of the selected genes.
</p>
</td></tr>
<tr><td><code>nMin.Features</code></td>
<td>

<p>The number of genes included in the minimum subset.
</p>
</td></tr>
<tr><td><code>Measures</code></td>
<td>

<p>Available only when <code>Verbose</code> is <code>TRUE</code>. It is an object with class &ldquo;<code>data.frame</code>&rdquo; which contains 3 columns: the indices of the selected genes; the POS measures of the selected genes (see <code><a href="#topic+POS">POS</a></code>); the status that reports on which basis a gene is selected (&ldquo;Min.Set&rdquo;: the gene is a member of the selected minimum subset, 1: the gene has a low POS score and its relative dominant class is the first class or 2: the gene has a low POS score and its relative dominant class is the second class), see <code><a href="#topic+RDC">RDC</a></code>.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p><code>Verbose</code> is only needed when <code>K</code> is specified. If <code>K</code> is set to &ldquo;Min&rdquo; (default), all information are automatically returned.
</p>


<h3>Author(s)</h3>

<p>Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+POS">POS</a></code> for calculating the proportional overlapping scores and <code><a href="#topic+RDC">RDC</a></code> for assigning the relative dominant class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukaemia)
GenesExpression &lt;- leukaemia[1:7129,] #define the features matrix
Class           &lt;- leukaemia[7130,]   #define the observations' class labels
## select the minimum subset of features
Selection       &lt;- Sel.Features(GenesExpression, Class)
attributes(Selection)
(Candidates      &lt;- Selection$Features)   #return the selected features
(Covered.observations &lt;- Selection$Covered.Obs) #return the covered observations by the selection
## select a specific number of features
Selection.k      &lt;- Sel.Features(GenesExpression, Class, K=10, Verbose=TRUE)
Selection.k$Features
Selection.k$nMin.Features   #return the size of the minimum subset of genes
Selection.k$Measures        #return the selected features' information
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
