<!DOCTYPE html><html><head><title>Help for package PROSPER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PROSPER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gen_check'><p>Checking the plausibility and conversion of dom and af</p>
<code>link{struc_preparation}</code></a></li>
<li><a href='#gen_diploid'><p>Diploid genetics: recombination of alleles</p></a></li>
<li><a href='#gen_freq'><p>generating the start values for PROSPER models</p></a></li>
<li><a href='#herb_check'><p>Checking the main parameters</p>
<code>link{sel_herb}</code></a></li>
<li><a href='#intern_herbicide'><p>Surviving the Herbicide</p></a></li>
<li><a href='#intern_SimData'><p>Extraction Simulation Data</p></a></li>
<li><a href='#intern_SimData&lt;-'><p>replacement of Simulation Data</p></a></li>
<li><a href='#mod_check'><p>Checking the main parameters</p>
<code>struc_preparation2</code></a></li>
<li><a href='#plot'><p>Simple plot for prosper simulation results</p></a></li>
<li><a href='#pop_germc'><p>Germination</p></a></li>
<li><a href='#pop_reprod'><p>Seed production and crop yield</p></a></li>
<li><a href='#pop_step'><p>Surviving a non-selective process</p></a></li>
<li><a href='#prosper'><p>Creation of a PROSPER object</p></a></li>
<li><a href='#prosper-class'><p>Class for PROSPER results</p></a></li>
<li><a href='#prosper-models'><p>Population dynamic models - Examples</p></a></li>
<li><a href='#PROSPER-pkg'><p>PROSPER-pkg</p></a></li>
<li><a href='#quanti'><p>Step by step: the model develops driven by field data</p></a></li>
<li><a href='#sel_herb'><p>Surviving the herbicide</p></a></li>
<li><a href='#sel_resist'><p>Calculating resistance values</p></a></li>
<li><a href='#struc_addSimData'><p>Wrapper for <code>intern_SimData&lt;-</code>  generic</p></a></li>
<li><a href='#struc_endSim'><p>Check for some stop conditions.</p></a></li>
<li><a href='#struc_preparation2'><p>Generating data input and output structures</p></a></li>
<li><a href='#struc_saveSimData'><p>Saving simulation data</p></a></li>
<li><a href='#summary'><p>Summary for prosper objects</p></a></li>
<li><a href='#weed-parameters'><p>Parameterisations for population dynamic models - Examples</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulation of Weed Population Dynamics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-06-23</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christoph v. Redwitz &lt;christoph.redwitz@posteo.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An environment to simulate the development of annual plant populations with regard to population dynamics and genetics, especially herbicide resistance. It combines genetics on the individual level (Renton et al. 2011) with a stochastic development on the population level (Daedlow, 2015).
	Renton, M, Diggle, A, Manalil, S and Powles, S (2011) &lt;<a href="https://doi.org/10.1016%2Fj.jtbi.2011.05.010">doi:10.1016/j.jtbi.2011.05.010</a>&gt; 
	Daedlow, Daniel (2015, doctoral dissertation: University of Rostock, Faculty of Agriculture and Environmental Sciences.)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, data.table</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-27 13:06:44 UTC; christoph.redwitz</td>
</tr>
<tr>
<td>Author:</td>
<td>Christoph v. Redwitz [aut, cre],
  Friederike de Mol [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-27 14:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gen_check'>Checking the plausibility and conversion of dom and af
<code>link{struc_preparation}</code></h2><span id='topic+gen_check'></span>

<h3>Description</h3>

<p>Before calling <code>struc_preparation</code> it is necessary to bring the genetics into the correct form, which is mainly the correct dimensions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_check(Rmx, af, dom, epis)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_check_+3A_rmx">Rmx</code></td>
<td>
<p>maximum resistance value, if all gene loci under consideration are homozygous resistant. <code class="reqn">\code{numeric}, \ge 1</code>.</p>
</td></tr>
<tr><td><code id="gen_check_+3A_af">af</code></td>
<td>
<p>initial frequency of resistance alleles in the population. <code>numeric vector</code> with length of <code>ng</code> (number of genes) and elements in [0,1].</p>
</td></tr>
<tr><td><code id="gen_check_+3A_dom">dom</code></td>
<td>
<p>dominance of resistance alleles. <code>numeric vector</code> with length of <code>n_loci</code> (number of genes) and elements in [0,1].</p>
</td></tr>
<tr><td><code id="gen_check_+3A_epis">epis</code></td>
<td>
<p>epistasis value, describing the interaction between resistance alleles. <code>numeric</code>. <code>epis</code> = 0: no interaction i.e. additive effects of resistance alleles, <code>epis</code> &lt; 0: effect of resistance alleles is smaller than additive, <code>epis</code> &gt; 0: effect of resistance alleles is higher than additive.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If no genetics are required <code>af</code> has to be set to <code>NA</code>. The value of <code>dom</code> is adjusted to the length of <code>af</code>. If there are mismatches, the <code>dom</code> is cut or the first given value is repeated to fit the number of given alleles in <code>af</code>. The variables af and <code>dom</code> are corrected and <code>n_loci</code> is created, which is 0 when no genetics is included. If no value is given to <code>epis</code> it is set to 0.
</p>

<hr>
<h2 id='gen_diploid'>Diploid genetics: recombination of alleles</h2><span id='topic+gen_diploid'></span>

<h3>Description</h3>

<p><code>gen_diploid</code> recombines the alleles of weeds with diploid genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_diploid(
  start,
  start_comb = NA,
  result,
  newseeds = get0("newseeds", envir = parent.frame(n = 1)),
  max_vec_length = 1e+07
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_diploid_+3A_start">start</code></td>
<td>
<p>column names of parental cohorts. <code>character vector</code>.</p>
</td></tr>
<tr><td><code id="gen_diploid_+3A_start_comb">start_comb</code></td>
<td>
<p>all named columns in <code>start</code> are joined and returned as a column with the name of <code>start_comb</code> (<code>character</code>). Only used when <code>start</code> longer than 1.</p>
</td></tr>
<tr><td><code id="gen_diploid_+3A_result">result</code></td>
<td>
<p>name of the results column. <code>character</code>.</p>
</td></tr>
<tr><td><code id="gen_diploid_+3A_newseeds">newseeds</code></td>
<td>
<p>number of new seeds calculated by <code><a href="#topic+pop_reprod">pop_reprod</a></code>. <code>positive</code> <code>integer</code>.</p>
</td></tr>
<tr><td><code id="gen_diploid_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>gen_diploid</code> assumes independent allele recombination.
</p>


<h3>Value</h3>

<p>A column with the name of <code>result</code> and, if necessary, a column with the name of <code>start_comb</code> is added to the <code>data.frame</code> <code>dfgenotype</code>.
</p>


<h3>warning</h3>

<p>newseeds are always coerced to a whole number.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pop_reprod">pop_reprod</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# generate a 'dfgenotype' data.frame:
struc_preparation2(Rmx=10, af=c(0.01, 0.2), epis=0, dom=1)
#Distribute 10000 individuals of the starting population across the genotypes. 
#The two gene loci have initial frequencies of 0.01 and 0.8.
gen_freq(af=c(0.01,0.8), n_seeds=10000)
# The column "initialSB" represents the parent generation, which recombines and 
#therefore defines the genetics of the new seeds
newseeds &lt;- 10000
gen_diploid(start="initialSB", result="followingSB")
# If a second cohort is reprodusing in the same time
gen_diploid(start=c("initialSB", "followingSB"), 
                  start_comb="two_cohorts_combind", result="followingSB2")
rm(dfgenotype, mf, newseeds, xprobab)

## End(Not run)
</code></pre>

<hr>
<h2 id='gen_freq'>generating the start values for PROSPER models</h2><span id='topic+gen_freq'></span>

<h3>Description</h3>

<p><code>gen_freq</code> generates the numbers of genotypes at the beginning of the first simulated year.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_freq(
  af,
  n_seeds,
  result = "initialSB",
  distribution = NA,
  max_vec_length = 1e+07
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_freq_+3A_af">af</code></td>
<td>
<p>initial frequency of resistance alleles in the population. <code>numeric vector</code> with length of <code>ng</code> (number of genes) and elements in [0,1].</p>
</td></tr>
<tr><td><code id="gen_freq_+3A_n_seeds">n_seeds</code></td>
<td>
<p>initial number of weed seeds. They will be allocated to the genotypes. <code>integer</code>.</p>
</td></tr>
<tr><td><code id="gen_freq_+3A_result">result</code></td>
<td>
<p>names of the results columns. <code>character</code>.</p>
</td></tr>
<tr><td><code id="gen_freq_+3A_distribution">distribution</code></td>
<td>
<p>defines the proportion with <code>n_seeds</code> is distributed among multiple cloumns, if result defines multiple columns. Has to be of the same length as <code>result</code>. <code>numeric</code> between 0 and 1 or &quot;equal&quot;. See details.</p>
</td></tr>
<tr><td><code id="gen_freq_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The start values for a model include the initial frequencies of alleles and the initial seedbank, i.e. the number of seeds in the soil. <code>gen_freq</code> allocates the <code>n_seeds</code> individuals to the different genotypes as provided by <code>dfgenotype</code>, which is created by <code>struc_preparation2()</code>. If <code>af == NULL</code> the function gives all seed to the only &quot;genotype&quot;. The distribution can be splitted to multiple cohorts using <code>distribution</code>. 
If <code>distribution</code> is &quot;equal&quot;, n_seeds are distributed equaly amoung the cohorts defindes with <code>result</code>.
</p>


<h3>Value</h3>

<p>Returns a <code>data.frame</code> containing the genotypes and their frequencies provided by <code>dfgenotype</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sel_resist">sel_resist</a></code> <code><a href="#topic+struc_preparation2">struc_preparation2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate a 'dfgenotype' data.frame:
struc_preparation2(Rmx=10, af=c(0.01,0.8), epis=0, dom=1)
#Distribute 10000 individuals of the starting population across the genotypes. 
#The two gene loci have initial frequencies of 0.01 and 0.8.
gen_freq(af=c(0.01,0.8), n_seeds=10000)
rm(dfgenotype, mf, xprobab)
</code></pre>

<hr>
<h2 id='herb_check'>Checking the main parameters
<code>link{sel_herb}</code></h2><span id='topic+herb_check'></span>

<h3>Description</h3>

<p>Is called within <code>sel_herb()</code> to check for the existence of the necessary data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>herb_check(put, sdrate, thresh, rate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="herb_check_+3A_put">put</code></td>
<td>
<p>probability of a weed to be untouched by the herbicide. <code>numeric</code>, <code class="reqn">0 \le \code{put} \le 1</code>.</p>
</td></tr>
<tr><td><code id="herb_check_+3A_sdrate">sdrate</code></td>
<td>
<p>variance of the herbicide rate reaching the weed. <code>positive numeric</code>, 1 = 1 unit standard deviation.</p>
</td></tr>
<tr><td><code id="herb_check_+3A_thresh">thresh</code></td>
<td>
<p>threshold herbicide rate to kill weeds without resistance. <code>numeric</code>, <code class="reqn">0 \le \code{thresh} \le 1</code>.</p>
</td></tr>
<tr><td><code id="herb_check_+3A_rate">rate</code></td>
<td>
<p>percentage (%) of the registered herbicide dose. <code>positive numeric</code>, can exceed 100 %.</p>
</td></tr>
</table>

<hr>
<h2 id='intern_herbicide'>Surviving the Herbicide</h2><span id='topic+intern_herbicide'></span>

<h3>Description</h3>

<p>Utility function internally used. It's used only by <code>sel_herb</code>, and usually there is no reason to change it. Calculates the surviving number of weeds according to the specific genotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intern_herbicide(resist, n_samples, put, rate, sdrate, thresh)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intern_herbicide_+3A_resist">resist</code></td>
<td>
<p>numeric, value of resistance for the genotype (defined by Renton et al. 2011). Shall be <code class="reqn">\le 1</code>.</p>
</td></tr>
<tr><td><code id="intern_herbicide_+3A_n_samples">n_samples</code></td>
<td>
<p>integer, the number of weeds with one specific genotype.</p>
</td></tr>
<tr><td><code id="intern_herbicide_+3A_put">put</code></td>
<td>
<p>probability of a weed to be untouched by the herbicide. <code>numeric</code>, <code class="reqn">0 \le \code{put} \le 1</code>.</p>
</td></tr>
<tr><td><code id="intern_herbicide_+3A_rate">rate</code></td>
<td>
<p>percentage (%) of the registered herbicide dose. <code>positive numeric</code>, can exceed 100 %.</p>
</td></tr>
<tr><td><code id="intern_herbicide_+3A_sdrate">sdrate</code></td>
<td>
<p>variance of the herbicide rate reaching the weed. <code>positive numeric</code>, 1 = 1 unit standard deviation.</p>
</td></tr>
<tr><td><code id="intern_herbicide_+3A_thresh">thresh</code></td>
<td>
<p>threshold herbicide rate to kill weeds without resistance. <code>numeric</code>, <code class="reqn">0 \le \code{thresh} \le 1</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>intern_herbicide</code> is used in <code><a href="#topic+sel_herb">sel_herb</a></code>. Firstly, it calculates the number of weeds that are untouched by the herbicide by chance (probability=<code>put</code>). In the second step, the herbicide rate that reached an individual is compared with the resistance value (calculated after Renton et al. 2011, <code><a href="#topic+sel_resist">sel_resist</a></code>). If the resistance value is lower than the dose, the weed dies. All surviving weeds are summed up.
</p>


<h3>Value</h3>

<p>The number of weeds surviving the herbicide.
</p>


<h3>References</h3>

<p>Renton, M.; Diggle, A.; Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, 283, 14-27.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sel_herb">sel_herb</a></code> <code><a href="#topic+sel_resist">sel_resist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#How many of 1000 weeds of a genotype with resistance value 5.5 survive a herbicide application 
#with full dose? 'It is assumed that weeds reseaving less than 20 \% of the full dose survive
#independently of their resistant value.
intern_herbicide(resist=5.5, n_samples=1000, put=0.04, rate=100, sdrate=0.4, thresh=20)
</code></pre>

<hr>
<h2 id='intern_SimData'>Extraction Simulation Data</h2><span id='topic+intern_SimData'></span><span id='topic+intern_SimData+2Cprosper-method'></span>

<h3>Description</h3>

<p><code>intern_SimData</code> extracts the simulation data from a <code>prosper</code>-object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intern_SimData(object)

## S4 method for signature 'prosper'
intern_SimData(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intern_SimData_+3A_object">object</code></td>
<td>
<p><code>prosper</code>, the simulation data of this object will be extracted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.table</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
struc_preparation2(Rmx=10, af=c(0.01,0.8), epis=0, dom=c(1,0.3))
   #simdata[,"repetition":=1]
   #simdata[,"simcycle":=1]
   #setcolorder(simdata, c(c("repetition", "simcycle"),
   #colnames(simdata)[!(colnames(simdata) %in%
   #c("simulation_run", "repetition", "simcycle"))])) 
#prosp_ob &lt;- prosper(simstruc = c(5,10), loci=2, simData = simdata)
#intern_SimData(prosp_ob)
</code></pre>

<hr>
<h2 id='intern_SimData+26lt+3B-'>replacement of Simulation Data</h2><span id='topic+intern_SimData+3C-'></span>

<h3>Description</h3>

<p><code>intern_SimData&lt;-</code> replaces the simulation data of a <code>prosper</code>-object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intern_SimData(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intern_SimData+2B26lt+2B3B-_+3A_object">object</code></td>
<td>
<p><code>prosper</code>, the simulation data of this object replaced.</p>
</td></tr>
<tr><td><code id="intern_SimData+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>object fitting the requirements for simulation data of the <code>class prosper</code>, this replaces the existing simulation data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>prosper</code>
</p>

<hr>
<h2 id='mod_check'>Checking the main parameters
<code><a href="#topic+struc_preparation2">struc_preparation2</a></code></h2><span id='topic+mod_check'></span>

<h3>Description</h3>

<p>Before the preparation (<code><a href="#topic+struc_preparation2">struc_preparation2</a></code>) it is necessary to check, whether the necessary information is provided to principally conduct a simulation run. Furthermore the crop.list is brought to the correct form.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mod_check(
  param.weed = NA,
  area = NA,
  duration = NA,
  repetitions = NA,
  crop_list = NA,
  max_vec_length = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mod_check_+3A_param.weed">param.weed</code></td>
<td>
<p>A <code>data.frame</code> with population dynamic parameters with or without stochasticity. The structure of <code>param.weed</code> is essential (see details). The easiest way to create the <code>data.frame</code> is to adopt an example (<a href="#topic+param.ECHCG">param.ECHCG</a>).</p>
</td></tr>
<tr><td><code id="mod_check_+3A_area">area</code></td>
<td>
<p>number of area units. <code>positive numeric</code>.</p>
</td></tr>
<tr><td><code id="mod_check_+3A_duration">duration</code></td>
<td>
<p>maximum number of simulation loops in the simulation. <code>positive integer</code>.</p>
</td></tr>
<tr><td><code id="mod_check_+3A_repetitions">repetitions</code></td>
<td>
<p>number of repetitions of the simulation. <code>positive integer</code></p>
</td></tr>
<tr><td><code id="mod_check_+3A_crop_list">crop_list</code></td>
<td>
<p>crop rotation. <code>character vector</code>, elements must fit to the names in the <code>data.frame</code> weed.</p>
</td></tr>
<tr><td><code id="mod_check_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If no genetics are required <code>af</code> has to be set to <code>NA</code>. The value of <code>dom</code> is adjusted to the length of <code>af</code>. The <code>dom</code> is cut or the first given value is repeated to fit the number of given alleles in <code>af</code>. The variables af and <code>dom</code> are corrected and <code>n_loci</code> is created, which is 0 when no genetics is included.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mod_check(param.weed=param.GALAP, area=100, duration=2,
     repetitions=1, crop_list=c("corn"), max_vec_length=1000000)
</code></pre>

<hr>
<h2 id='plot'>Simple plot for prosper simulation results</h2><span id='topic+plot'></span><span id='topic+plot+2Cprosper+2Cmissing-method'></span>

<h3>Description</h3>

<p>This function draws three figures. i)Numerics: mean number of individual plants of a specified development stage for each simulation cycle,  summing up the genotypes to one number per cycle. ii)Raw counts: same as i) using the results of the repetitions instead of the mean. iii) Mean of the proportion of plants with only alleles for sensitivity, proportion of plants with only alleles for resistance, proportion of plants with mixed alleles for resistance and sensitivity, proportion of R alleles in the population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'prosper,missing'
plot(x, y, plot_var = "SB_autumn", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_+3A_x">x</code></td>
<td>
<p><code>prosper</code>, the result of a prosper simulation.</p>
</td></tr>
<tr><td><code id="plot_+3A_y">y</code></td>
<td>
<p>not used.</p>
</td></tr>
<tr><td><code id="plot_+3A_plot_var">plot_var</code></td>
<td>
<p>variable, i.e. a column name of the result data.frame, to be plotted. <code>character</code>.</p>
</td></tr>
<tr><td><code id="plot_+3A_...">...</code></td>
<td>
<p>other graphical parameters.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+summary">summary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(param.LOLRI)
mod_lolri    &lt;- prosper.LOLRI(param.weed=param.LOLRI, area=20,  af = c(0.005, 0.01),
                                       duration=3, repetitions=2)
plot(mod_lolri)
</code></pre>

<hr>
<h2 id='pop_germc'>Germination</h2><span id='topic+pop_germc'></span>

<h3>Description</h3>

<p><code>pop_germc</code> describes germination as random event for individual seeds. The function considers different cohorts and dormancy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pop_germc(init_sb, germ, max_vec_length = 1e+07)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pop_germc_+3A_init_sb">init_sb</code></td>
<td>
<p>column name of the initial seed bank in the data.frame dfgenotype that is delivered by struc_preparation. <code>character</code>.</p>
</td></tr>
<tr><td><code id="pop_germc_+3A_germ">germ</code></td>
<td>
<p>germination probabilities for the different cohorts. See details.</p>
</td></tr>
<tr><td><code id="pop_germc_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each individual has a chance to germinate or to stay dormant. In case of germination, it emerges in one of the cohorts. The distribution of individual seeds to cohorts or dormancy is random. The function uses the columns <code>init_SB</code> of the data.frame <code>dfgenotype</code> as input. The output values are the numbers of seedlings of each genotype and each cohort.
</p>
<p><code>germ</code> must be given as a numeric vector or - in case of multiple columns in init_sb - data.frame or matrix. In that case the row number must fit to the length of <code>init_sb</code>. If the columns in init_sb represent cohorts, the rows of <code>germ</code> give the germination probabilities for these specific cohorts. The sum of one row of <code>germ</code> shall be <code class="reqn">\ge 0</code> and <code class="reqn">\le 1</code>. The difference of 1 and the sum of on row <code>germ</code> is the probability of dormancy.
</p>


<h3>Value</h3>

<p>Columns are added to <code>dfgenotype</code>: &quot;germ_dorm&quot; contains the numbers of each genotype that remain dormant, &quot;germ1&quot; to &quot;germX&quot; contain the numbers for each of X cohorts.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pop_reprod">pop_reprod</a></code> <code><a href="#topic+pop_step">pop_step</a></code> <code><a href="#topic+quanti">quanti</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>struc_preparation2(Rmx=10, af=c(0.01,0.8), epis=0, dom=1)
ls()
gen_freq( af=c(0.01,0.8), n_seeds=10000)
#Distribute the individuals to three cohorts with the germination 
#probabilities 0.2, 0.4 and 0.4.
pop_germc( init_sb="initialSB", germ=c(0.2,0.4,0.4))
rm(mf, dfgenotype, xprobab) 
</code></pre>

<hr>
<h2 id='pop_reprod'>Seed production and crop yield</h2><span id='topic+pop_reprod'></span>

<h3>Description</h3>

<p>Calculates the produced seeds and optionally the proportion of crop yield that is realized relative to weed free yield (Renton at al. 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pop_reprod(start, area, kw, pen_co, kc, dc, crop_inr, SSmax, yield = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pop_reprod_+3A_start">start</code></td>
<td>
<p>column names of parental cohorts. <code>character vector</code>.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_area">area</code></td>
<td>
<p>number of area units. <code>positive numeric</code>.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_kw">kw</code></td>
<td>
<p>dimensionless weed competition parameter. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_pen_co">pen_co</code></td>
<td>
<p>penalty values for different weed cohorts. <code>numeric vector</code>, each element in [0,1].</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_kc">kc</code></td>
<td>
<p>dimensionless crop competition parameter. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_dc">dc</code></td>
<td>
<p>crop sowing density, seeds per unit area. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_crop_inr">crop_inr</code></td>
<td>
<p>position of the crop in the crop rotation. <code>positive integer</code> <code class="reqn">\le</code> length of the crop rotation.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_ssmax">SSmax</code></td>
<td>
<p>maximum of weed seed production per unit area. <code>positive integer</code>.</p>
</td></tr>
<tr><td><code id="pop_reprod_+3A_yield">yield</code></td>
<td>
<p><code>logical</code>, whether the percentage of yield gained should be calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The number of produced seeds is calculated for 1m^2 by the formula:
<br />
</p>
<p style="text-align: center;"><code class="reqn">over\_all  = 1 + kc*dc[crop\_inr] + sum(kw*dw*pen\_co)</code>
</p>
 
<p style="text-align: center;"><code class="reqn">producedseeds  = round(sum((SSmax[crop\_inr] * kw * dw * pen_co)/over\_all),digits=0)</code>
</p>

<p style="text-align: center;"><code class="reqn">propyield = (1 + kc*dc[crop\_inr])/over\_all</code>
</p>

<p><br />
The weed density <code>dw</code> is calculated for each squaremeter derived from the current simulation run (<code>start</code>). The used parameters values apply to wheat and ryegrass (Pannell et al. 2004, cited in Renton et al. 2011).
</p>


<h3>References</h3>

<p>Renton, M.; Diggle, A.; Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, 283, 14-27.
Pannell, D. J.; Stewart, V.; Bennett, A.; Monjardino, M.; Schmidt, C. &amp; Powles, S. B. (2004): RIM: a bioeconomic model for integrated weed management of Lolium rigidum in Western Australia Agricultural Systems, 79, 305-325
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pop_reprod">pop_reprod</a></code> <code><a href="#topic+pop_step">pop_step</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>struc_preparation2(Rmx=10, af=c(0.01,0.8), epis=0, dom=1)
#Distribute 10000 individuals of the starting population across the genotypes provided by tmp. 
#The two gene loci have initial frequencies of 0.01 and 0.8.
gen_freq(af=c(0.01,0.8), n_seeds=10000, max_vec_length=1e+07)
pop_reprod("initialSB",  area=100, kw=0.5, pen_co=1, kc=0.05, dc=100, 
                         crop_inr="wheat", SSmax=3000, yield=TRUE)
rm(producedseeds, dfgenotype, mf, xprobab, propyield)
</code></pre>

<hr>
<h2 id='pop_step'>Surviving a non-selective process</h2><span id='topic+pop_step'></span>

<h3>Description</h3>

<p><code>pop_step</code> picks the individuals that will pass to the next development stage. This is a random process for every individual, which does not exert any selection pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pop_step(
  start,
  start_comb = NA,
  result = NA,
  stepname = NA,
  surv_prob,
  max_vec_length = 1e+07
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pop_step_+3A_start">start</code></td>
<td>
<p>column names of parental cohorts. <code>character vector</code>.</p>
</td></tr>
<tr><td><code id="pop_step_+3A_start_comb">start_comb</code></td>
<td>
<p>all named columns in <code>start</code> are joined and returned as a column with the name of <code>start_comb</code> (<code>character</code>). Only used when <code>start</code> longer than 1.</p>
</td></tr>
<tr><td><code id="pop_step_+3A_result">result</code></td>
<td>
<p>name of the results column. <code>character</code>.</p>
</td></tr>
<tr><td><code id="pop_step_+3A_stepname">stepname</code></td>
<td>
<p>name of the new column of dfgenotype added by this function. <code>character</code>.</p>
</td></tr>
<tr><td><code id="pop_step_+3A_surv_prob">surv_prob</code></td>
<td>
<p>probability to survive this step and reach the next growth stage. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="pop_step_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Individuals that reach the next growth stage are picked by using <code><a href="stats.html#topic+rbinom">rbinom</a></code>. In contrast to <code><a href="#topic+sel_herb">sel_herb</a></code>, <code>pop_step</code> does not exert any evolutionary selection pressure.
When more than one column is selected with <code>start</code>, they are summed and the result is passed to the picking process. By setting <code>start_comb</code> the sum is added as a column to <code>dfgenotype</code>.
</p>


<h3>Value</h3>

<p>A new column is added to <code>dfgenotype</code> containing the surviving individuals of the different genotypes.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+quanti">quanti</a></code> <code><a href="#topic+pop_germc">pop_germc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>struc_preparation2(Rmx=10, af=c(0.01,0.8), epis=0, dom=1)
gen_freq(af=c(0.01,0.8), n_seeds=10000)
#How many individuals of each genotype will reach the next growth stage?
pop_step(start="initialSB",  stepname="survivingthewinter",
                            surv_prob=0.4)
</code></pre>

<hr>
<h2 id='prosper'>Creation of a PROSPER object</h2><span id='topic+prosper'></span>

<h3>Description</h3>

<p>The creation method for PROSPER objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prosper(
  Call = match.call(definition = sys.function(sys.parent(n = 2)), call =
    sys.call(sys.parent(n = 2))),
  simstruc = numeric(),
  loci = numeric(),
  simData = as.data.table()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prosper_+3A_call">Call</code></td>
<td>
<p><code>character</code>, the model call of the simulation.</p>
</td></tr>
<tr><td><code id="prosper_+3A_simstruc">simstruc</code></td>
<td>
<p><code>numeric vector</code> with two numbers, the first defines the number of repetitions for the simulation, the second defines the number of simulation cycles in each repetition of the simulation.</p>
</td></tr>
<tr><td><code id="prosper_+3A_loci">loci</code></td>
<td>
<p><code>numeric</code>, the number of loci used (up to 4), 0 when no genetic is included.</p>
</td></tr>
<tr><td><code id="prosper_+3A_simdata">simData</code></td>
<td>
<p><code>data.table</code>, all simulated data is stored here. The first column is treated as the number of the simulation run, the second column the number of the repetition, the third as the simulation cycle/year, the forth as the type of data (genotype based or only year based), the fifth as the genotype (<code>character</code>) (NA when data is year based). All following columns are up to specific the model design.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genotypes are described by their relevant loci (up to 4). Each locus can have 0, 1 or 2 alleles in a diploid genome. So a genotype for two loci is coded like that: 00 (no relevant alleles), 21 (2 alleles at the first locus, and 1 at the second). For this coding <code>loci</code> must be &quot;2&quot;.
</p>


<h3>Value</h3>

<p>An object of class &quot;prosper&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(data.table)
struc_preparation2(Rmx=10, af=c(0.01,0.8), epis=0, dom=c(1,0.3))
simdata &lt;- as.data.table(dfgenotype)
   simdata[,"repetition":=1]
   simdata[,"simcycle":=1]
   setcolorder(simdata, c(c("repetition", "simcycle"),
   colnames(simdata)[!(colnames(simdata) %in%
   c("simulation_run", "repetition", "simcycle"))])) 
prosper(simstruc = c(5,10), loci=2, simData = simdata)  

</code></pre>

<hr>
<h2 id='prosper-class'>Class for PROSPER results</h2><span id='topic+prosper-class'></span>

<h3>Description</h3>

<p>This <code>class</code> simplifies the analysis of PROSPER results, by providing a fixed structure with some basic summary and plot methods.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Call</code></dt><dd><p><code>character</code>, the model call of the simulation.</p>
</dd>
<dt><code>simstruc</code></dt><dd><p><code>numeric vector</code> with 2 numbers, the first defines the number of repetitions for the simulation, the second defines the number of simulation cycles in each repetition of the simulation.</p>
</dd>
<dt><code>loci</code></dt><dd><p><code>numeric</code>, the number of loci used, 0 when no genetic is included.</p>
</dd>
<dt><code>simData</code></dt><dd><p><code>data.table</code>, all simulated data is stored here. The first column is treated as the number of the simulation run, the second column the number of the repetition, the third as the simulation cycle/year, the forth as the type of data (genotype based or only year based), the fifth as the genotype (<code>character</code>) (NA when data is year based). All following columns are up to specific the model design.</p>
</dd>
</dl>

<hr>
<h2 id='prosper-models'>Population dynamic models - Examples</h2><span id='topic+prosper-models'></span><span id='topic+param.ECHCG'></span><span id='topic+param.GALAP'></span><span id='topic+param.LOLRI'></span><span id='topic+prosper.ECHCG'></span><span id='topic+prosper.GALAP'></span><span id='topic+prosper.LOLRI'></span>

<h3>Description</h3>

<p>PROSPER entails full parameterized models, which are described here. 
Population dynamic models adress different purposes the models differ. The functions <code>prosper.*</code> present different adaptations.
</p>
<p><code>prosper.ECHCG</code> provides the setting for a simulation of the population dynamic of <em>Echinochloa crus-galli</em>.
</p>
<p><code>prosper.GALAP</code> provides the setting for a simulation of the population dynamic of <em>Galium aparine</em>. No selection process is used.
</p>
<p>prosper.LOLRI performs a simulation of PROSPER using the setting presented by Renton at al. (2011). To manipulate the parameters see Details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prosper.ECHCG(
  param.weed = PROSPER::param.ECHCG,
  area = NA,
  af = NA,
  dom = NA,
  epis = 0,
  put = 0.05,
  sdrate = 0.4,
  thresh = 20,
  Rmx = 10,
  rate = 100,
  duration = NA,
  repetitions = NA,
  crop_list = "corn",
  max_vec_length = 1e+07,
  undersowing = NA
)

prosper.GALAP(
  param.weed = PROSPER::param.GALAP,
  sens_seeds = 400,
  area = 100,
  af = c(0.03, 0.08, 0.02),
  dom = c(0.5, 0.5, 0.5),
  epis = 0,
  put = 0.05,
  thresh = 20,
  Rmx = 10,
  rate = 100,
  sdrate = 0.4,
  duration = 15,
  repetitions = 1,
  crop_list = "wheat",
  max_vec_length = 1e+07
)

prosper.LOLRI(
  param.weed = PROSPER::param.LOLRI,
  area = 100,
  af = c(0.005, 0.01, 0.015, 0.02),
  dom = 0.5,
  epis = 0,
  put = 0.05,
  sdrate = 0.4,
  thresh = 20,
  Rmx = 10,
  dc = 150,
  kc = 1/11,
  kw = 1/33,
  SSmax = 30000,
  rate = 100,
  pen_co = c(1, 0.5),
  duration = 10,
  repetitions = 1,
  crop_list = c("wheat"),
  max_vec_length = 1e+07
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prosper-models_+3A_param.weed">param.weed</code></td>
<td>
<p>A <code>data.frame</code> with population dynamic parameters with or without stochasticity. The structure of <code>param.weed</code> is essential (see details). The easiest way to create the <code>data.frame</code> is to adopt an example (<a href="#topic+param.ECHCG">param.ECHCG</a>).</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_area">area</code></td>
<td>
<p>number of area units. <code>positive numeric</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_af">af</code></td>
<td>
<p>initial frequency of resistance alleles in the population. <code>numeric vector</code> with length of <code>ng</code> (number of genes) and elements in [0,1].</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_dom">dom</code></td>
<td>
<p>dominance of resistance alleles. <code>numeric vector</code> with length of <code>n_loci</code> (number of genes) and elements in [0,1].</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_epis">epis</code></td>
<td>
<p>epistasis value, describing the interaction between resistance alleles. <code>numeric</code>. <code>epis</code> = 0: no interaction i.e. additive effects of resistance alleles, <code>epis</code> &lt; 0: effect of resistance alleles is smaller than additive, <code>epis</code> &gt; 0: effect of resistance alleles is higher than additive.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_put">put</code></td>
<td>
<p>probability of a weed to be untouched by the herbicide. <code>numeric</code>, <code class="reqn">0 \le \code{put} \le 1</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_sdrate">sdrate</code></td>
<td>
<p>variance of the herbicide rate reaching the weed. <code>positive numeric</code>, 1 = 1 unit standard deviation.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_thresh">thresh</code></td>
<td>
<p>threshold herbicide rate to kill weeds without resistance. <code>numeric</code>, <code class="reqn">0 \le \code{thresh} \le 1</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_rmx">Rmx</code></td>
<td>
<p>maximum resistance value, if all gene loci under consideration are homozygous resistant. <code class="reqn">\code{numeric}, \ge 1</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_rate">rate</code></td>
<td>
<p>percentage (%) of the registered herbicide dose. <code>positive numeric</code>, can exceed 100 %.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_duration">duration</code></td>
<td>
<p>maximum number of simulation loops in the simulation. <code>positive integer</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_repetitions">repetitions</code></td>
<td>
<p>number of repetitions of the simulation. <code>positive integer</code></p>
</td></tr>
<tr><td><code id="prosper-models_+3A_crop_list">crop_list</code></td>
<td>
<p>crop rotation. <code>character vector</code>, elements must fit to the names in the <code>data.frame</code> weed.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_undersowing">undersowing</code></td>
<td>
<p>Numerical vector with two values between 0 and 1. See details.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_sens_seeds">sens_seeds</code></td>
<td>
<p>sensitive seeds added every year.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_dc">dc</code></td>
<td>
<p>crop sowing density, seeds per unit area. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_kc">kc</code></td>
<td>
<p>dimensionless crop competition parameter. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_kw">kw</code></td>
<td>
<p>dimensionless weed competition parameter. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_ssmax">SSmax</code></td>
<td>
<p>maximum of weed seed production per unit area. <code>positive integer</code>.</p>
</td></tr>
<tr><td><code id="prosper-models_+3A_pen_co">pen_co</code></td>
<td>
<p>penalty values for different weed cohorts. <code>numeric vector</code>, each element in [0,1].</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>prosper.ECHCG()</code> simulates originally the population dynamic of <em>Echinochloa crus-galli</em> using the data <code>param.ECHCG</code>. Different cohorts of weed seedlings are the focus of this model. The focus of this model is the effect of weeds that escape the selection pressure of herbicide treatment. These weeds keep the unselected genetic. Can they buffer the selection process? <em>E. crus-galli</em> is able to germinate over a long period after maize planting with decreasing reproductive success (Bagavathiannan, 2013). In the model all germinating individuals are represented by two cohorts; an early, major cohort with with high seed production, and a small late emerging with lower reproduction. Only the first cohort is controlled by a herbicide, which is a typical situation in Germany (Rossberg, 2016). The second cohort escapes the herbicide treatment unaffected. However, the second cohort can be suppressed, for example by an undersown crop. Three scenarios with different degrees of suppression, 0%, 30% and 100%, were simulated (Redwitz, 2016).
The parameter <code>undersowing</code> describes the probability of surviving a second, not selective pressure on weed seedlings, which germinate after the selective herbicide was applied.
<code>prosper.ECHCG</code> provides the setting for a simulation of the population dynamic of <em>Echinochloa crus-galli</em>.
</p>
<p><code>prosper.GALAP()</code> simulates originally the population dynamic of <em>Galium aparine</em> using the data <code>param.GALAP</code>. Whether sowing of susceptible weed seeds can restore an 'acceptable' resistance level of a population in the early stages of resistance development, is an extraordinary research question. The patchy occurrence of <em>Galium aparine</em> and its large seeds result in highly variable population dynamic parameters. Modeling has to take into account this variability. We used a simple population dynamics model structure (Redwitz et al., 2015). A seedbank in spring provides seeds out of which one cohort is germinating. The weeds are selected by herbicides, produce seeds, which are affected by seed predation and return to the seedbank in autumn. Data of a long term field experiment were used for parametrization (Daedlow, 2015).
</p>
<p><code>prosper.LOLRI()</code> performs a simulation of <em>Lolium rigidum</em> similar to PERTH (Renton et al. 2011) when it is used with <code>param.LOLRI</code>.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>prosper.ECHCG</code>: Population dynamic model of <em>Echinochloa crus-galli</em>
</p>
</li>
<li> <p><code>prosper.GALAP</code>: Population dynamic model of <em>Galium aparine</em>
</p>
</li>
<li> <p><code>prosper.LOLRI</code>: Population dynamic model of <em>Lolium multiflorum</em>
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Christoph von Redwitz, <a href="mailto:christoph.redwitz@uni-rostock.de">christoph.redwitz@uni-rostock.de</a>
</p>


<h3>References</h3>

<p>Redwitz, C von, Pannwitt H, Gerowitt B (2016): About the interplay of sensitive and resistant biotypes in weed populations - simulation exercises for Echinochloa crus-galli in maize crops. Proceedings - 28th German Conference on Weed Biology and Weed Control, Julius-Kuehn-Archiv, 93-99, 452.
</p>
<p>Redwitz, C von, Daedlow D, Gerowitt B (2015): Simulation exercises on long-term management of widespread herbicide resistance in a field weed population. Proceedings 17th Symposium of the European Weed Research Society, Montpellier, France, 108.
</p>
<p>Renton, M., Diggle, A., Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, Elsevier BV, 283, 14-27.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+weed-parameters">weed-parameters</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mod_echcg &lt;- prosper.ECHCG(param.weed = param.ECHCG, area=100, af=c(0.001), 
                             undersowing=0.2,dom=0.5,duration=7,repetitions=1)


#The model call for Redwitz et al. (2015)
undersowing_prob &lt;- c(1, 0.3, 0) #no undersowing, strong competition, complete dominance
years &lt;- 20
reps  &lt;- 4
####------------------------
simu_collect &lt;- list()
for(simu in 1:3){
simu_collect[[simu]] &lt;- prosper.ECHCG(area          = 100,
                                      param.weed    = param.ECHCG,
                                      thresh        = 20,
                                      duration      = years,
                                      af            = 0.001,     
                                      dom           = 1,      
                                      undersowing   = undersowing_prob[simu],  
                                      repetitions   = reps
                                             )
}

## End(Not run)

mod_galap &lt;- prosper.GALAP(param.weed=param.GALAP, repetitions=2, duration=10) 

mod_lolri    &lt;- prosper.LOLRI(param.weed=param.LOLRI, area=100, 
                                       duration=15, repetitions=3)
</code></pre>

<hr>
<h2 id='PROSPER-pkg'>PROSPER-pkg</h2><span id='topic+PROSPER-pkg'></span>

<h3>Description</h3>

<p>PROSPER simulates the development of annual plant populations with regard to population dynamics and genetics, especially herbicide resistance. It combines genetics on the individual level with a stochastic development on the population level. The genetic part is modeled after Renton et al. (2011), the stochastic part follows Daedlow (2015). All parameters delivered with PROSPER are based on an area of 1 squaremeter.
</p>


<h3>Source</h3>

<p>Renton, M.; Diggle, A.; Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, 283, 14-27.
</p>
<p>Daedlow, D. (2015): About the contribution of seed predation on weed demography. (Doctoral dissertation). University of Rostock, Faculty of Agriculture and Environmental Sciences. GBV Gemeinsamer Verbundkatalog (Accession No. 839752644).
</p>

<hr>
<h2 id='quanti'>Step by step: the model develops driven by field data</h2><span id='topic+quanti'></span>

<h3>Description</h3>

<p>This is a working horse of PROSPER. It quantifies the number or the proportion of individuals entering the next development stage using predefined formulas (<code>formul</code>). Typically these formulas are the results of experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quanti(
  origin = NA,
  step_name,
  crop,
  proportion = TRUE,
  equal_dis = TRUE,
  res_max = NA,
  res_min = 0,
  formul = NA,
  area,
  addit_variables = NA,
  log_values = TRUE,
  back_log = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quanti_+3A_origin">origin</code></td>
<td>
<p>numbers or log-numbers of individuals at the start for every genotype. <code>numeric</code>. Alternatively the columnaes of dfgenotype can be used.</p>
</td></tr>
<tr><td><code id="quanti_+3A_step_name">step_name</code></td>
<td>
<p>step name in the database containing the model parameters. <code>character</code>.</p>
</td></tr>
<tr><td><code id="quanti_+3A_crop">crop</code></td>
<td>
<p>current crop in the crop sequence, must fit to one of the crop names in the data frame with weed-parameters. <code>character</code>.</p>
</td></tr>
<tr><td><code id="quanti_+3A_proportion">proportion</code></td>
<td>
<p><code>logical</code>, <code>TRUE</code> when the output should be a proportion, otherwise it will be an absolute number.</p>
</td></tr>
<tr><td><code id="quanti_+3A_equal_dis">equal_dis</code></td>
<td>
<p><code>logical</code>, <code>TRUE</code> when the distribution of the weeds is spatially uniform across the <code>area</code>.</p>
</td></tr>
<tr><td><code id="quanti_+3A_res_max">res_max</code></td>
<td>
<p>output maximum, ignored when <code>origin=NA</code>. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="quanti_+3A_res_min">res_min</code></td>
<td>
<p>output minimum. <code>numeric</code>.</p>
</td></tr>
<tr><td><code id="quanti_+3A_formul">formul</code></td>
<td>
<p><code>character</code>. See details.</p>
</td></tr>
<tr><td><code id="quanti_+3A_area">area</code></td>
<td>
<p>number of area units. <code>positive numeric</code>.</p>
</td></tr>
<tr><td><code id="quanti_+3A_addit_variables">addit_variables</code></td>
<td>
<p>variables used in <code>formul</code> that are not predefined in <code>param.weed</code>. See details. <code>character vector</code></p>
</td></tr>
<tr><td><code id="quanti_+3A_log_values">log_values</code></td>
<td>
<p>logical, <code>TRUE</code> when <code>origin</code> in log-scale.</p>
</td></tr>
<tr><td><code id="quanti_+3A_back_log">back_log</code></td>
<td>
<p>logical, <code>TRUE</code> when the output is in log-scale. Only used when <code>proportion==FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Within PROSPER simulation models are build up with discrete simulation steps. These steps are conducted by functions like <code>pop_step</code> or <code>gen_reprod</code>. These functions affect the complete population and need a count of individuals or a proportion of the population that are affected of the specific simulation step. These numbers are calculated with <code>quanti</code>. The calculation is based on the data provided in <code><a href="#topic+weed-parameters">weed-parameters</a></code>. There a model is given for every parameter in the table. The parameter <code>formul</code> allows to use a different model if necessary. If no model is given at all, the simulation step is assumed to consist only of one value. The parameters of <code>param.weed</code> are normal distributed and the SE is used to draw them for the current calculation using <code>rnorm()</code>. These values are used to evaluate the model for simulation step. The resulting count or proportion can be used to perform the simulation step for the population.
</p>


<h3>Value</h3>

<p>Either a (log-)number of individuals or a proportion resp. a rate.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pop_step">pop_step</a></code> <code><a href="#topic+pop_germc">pop_germc</a></code> <code><a href="#topic+weed-parameters">weed-parameters</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#loads the example data for Echinochloa crus-galli
data(param.ECHCG)
param.weed &lt;- param.ECHCG
#how many seeds (natural, not log-scale) prodused by 100 plants in a corn stand on 100 area units?
quanti(origin=100, step_name="seed_prod_first", crop="corn",  proportion=FALSE,
                 area=100, log_values=FALSE, back_log=FALSE)
rm(param.ECHCG)
</code></pre>

<hr>
<h2 id='sel_herb'>Surviving the herbicide</h2><span id='topic+sel_herb'></span>

<h3>Description</h3>

<p><code>sel_herb</code> calculates the surviving number of each genotype. <code>sel_herb</code> selects for resistant individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel_herb(start, result, thresh, sdrate, rate, put, max_vec_length = 1e+07)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sel_herb_+3A_start">start</code></td>
<td>
<p>column names of parental cohorts. <code>character vector</code>.</p>
</td></tr>
<tr><td><code id="sel_herb_+3A_result">result</code></td>
<td>
<p>name of the results column. <code>character</code>.</p>
</td></tr>
<tr><td><code id="sel_herb_+3A_thresh">thresh</code></td>
<td>
<p>threshold herbicide rate to kill weeds without resistance. <code>numeric</code>, <code class="reqn">0 \le \code{thresh} \le 1</code>.</p>
</td></tr>
<tr><td><code id="sel_herb_+3A_sdrate">sdrate</code></td>
<td>
<p>variance of the herbicide rate reaching the weed. <code>positive numeric</code>, 1 = 1 unit standard deviation.</p>
</td></tr>
<tr><td><code id="sel_herb_+3A_rate">rate</code></td>
<td>
<p>percentage (%) of the registered herbicide dose. <code>positive numeric</code>, can exceed 100 %.</p>
</td></tr>
<tr><td><code id="sel_herb_+3A_put">put</code></td>
<td>
<p>probability of a weed to be untouched by the herbicide. <code>numeric</code>, <code class="reqn">0 \le \code{put} \le 1</code>.</p>
</td></tr>
<tr><td><code id="sel_herb_+3A_max_vec_length">max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For every genotype <code><a href="#topic+intern_herbicide">intern_herbicide</a></code> is called. If no genetics are included, the value from <code>start</code> is returned in <code>result</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sel_resist">sel_resist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>struc_preparation2(Rmx=10, af=c(0.02,0.01), epis=0, dom=1)
gen_freq( af=c(0.01,0.8), n_seeds=10000)
sel_herb(start="initialSB", result="winter", 
                       thresh=20, sdrate=0.4, rate=100, put=0.04)
</code></pre>

<hr>
<h2 id='sel_resist'>Calculating resistance values</h2><span id='topic+sel_resist'></span>

<h3>Description</h3>

<p>Calculates the phenotypic resistance value (Renton et al. 2011) for each genotype in <code>dfgenotype</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel_resist(Rmx, epis, dom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sel_resist_+3A_rmx">Rmx</code></td>
<td>
<p>maximum resistance value, if all gene loci under consideration are homozygous resistant. <code class="reqn">\code{numeric}, \ge 1</code>.</p>
</td></tr>
<tr><td><code id="sel_resist_+3A_epis">epis</code></td>
<td>
<p>epistasis value, describing the interaction between resistance alleles. <code>numeric</code>. <code>epis</code> = 0: no interaction i.e. additive effects of resistance alleles, <code>epis</code> &lt; 0: effect of resistance alleles is smaller than additive, <code>epis</code> &gt; 0: effect of resistance alleles is higher than additive.</p>
</td></tr>
<tr><td><code id="sel_resist_+3A_dom">dom</code></td>
<td>
<p>dominance of resistance alleles. <code>numeric vector</code> with length of <code>n_loci</code> (number of genes) and elements in [0,1].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used in <code>struc_preparation2</code> to calculate the resistance value using the following term:
</p>
<p style="text-align: center;"><code class="reqn">  1 + (Rmx - 1) * (sum(dom)/n\_loci)^{2^{epis}}). </code>
</p>

<p>Resistance values range from 1 to Rmx. The higher the resistance value is, the less the plant is susceptible to the herbicide.
<code>sel_resist</code> is only used by <code><a href="#topic+struc_preparation2">struc_preparation2</a></code>.
</p>


<h3>Value</h3>

<p><code>numeric vector</code> with resistance values
</p>


<h3>References</h3>

<p>Renton, M.; Diggle, A.; Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, 283, 14-27.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sel_herb">sel_herb</a></code> <code><a href="#topic+intern_herbicide">intern_herbicide</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#dfgenotype is usually generated by the function 'struc_preparation2'.
#Here, a simple example is done by hand.
var1 &lt;- c("00","01","02","10","11","12","20","21","21")
var2 &lt;- c(0,0,0,1,1,1,2,2,2)
var3 &lt;- c(0,1,2,0,1,2,0,1,2)
dfgenotype &lt;- data.frame(genotype=var1,l1=var2,l2=var3, stringsAsFactors = TRUE)
sel_resist(Rmx=10, epis=0, dom=1)
</code></pre>

<hr>
<h2 id='struc_addSimData'>Wrapper for <code>intern_SimData&lt;-</code>  generic</h2><span id='topic+struc_addSimData'></span><span id='topic+struc_addSimData+2Cprosper-method'></span>

<h3>Description</h3>

<p>This function ensures the correct form of data, before adding data of a new simulation cycle to a <code>prosper</code> object using <code>intern_SimData</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>struc_addSimData(object, value, add = TRUE)

## S4 method for signature 'prosper'
struc_addSimData(object, value, add = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="struc_addSimData_+3A_object">object</code></td>
<td>
<p>a <code>prosper</code> object</p>
</td></tr>
<tr><td><code id="struc_addSimData_+3A_value">value</code></td>
<td>
<p>codedata.frame or codedata.table</p>
</td></tr>
<tr><td><code id="struc_addSimData_+3A_add">add</code></td>
<td>
<p><code>logical</code>, when <code>TRUE</code> adding data to the existing data. Otherwise the value is completely new</p>
</td></tr>
</table>

<hr>
<h2 id='struc_endSim'>Check for some stop conditions.</h2><span id='topic+struc_endSim'></span>

<h3>Description</h3>

<p>Conditions that are necessary to continue the simulation are checked.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>struc_endSim(simcycle = year, break_col_names = "SB_autumn")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="struc_endSim_+3A_simcycle">simcycle</code></td>
<td>
<p>The count of simulation runs. The  value is <code>duration</code> <code class="reqn">\ge  0</code>.</p>
</td></tr>
<tr><td><code id="struc_endSim_+3A_break_col_names">break_col_names</code></td>
<td>
<p>The names of the columns in <code>dfgenotype</code>, which are the first in the next simulation cycle.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function combines two conditions to terminate the simulation. The first is the <code>duration</code> that is defined for the simulation. The second is the the number of individuals for the next simulation cycle: when the population is extinct the simulation ends.
</p>


<h3>Value</h3>

<p><code>logical</code> TRUE if the simulation has to stop.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+struc_saveSimData">struc_saveSimData</a></code>
</p>

<hr>
<h2 id='struc_preparation2'>Generating data input and output structures</h2><span id='topic+struc_preparation2'></span>

<h3>Description</h3>

<p><code>struc_preparation</code> creates the data input and output structures (data.frames and table) for the simulation run, 'dfgenotype', 'xprobab' and 'mf'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>struc_preparation2(Rmx = NA, af = NA, epis = NA, dom = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="struc_preparation2_+3A_rmx">Rmx</code></td>
<td>
<p>maximum resistance value, if all gene loci under consideration are homozygous resistant. <code class="reqn">\code{numeric}, \ge 1</code>.</p>
</td></tr>
<tr><td><code id="struc_preparation2_+3A_af">af</code></td>
<td>
<p>initial frequency of resistance alleles in the population. <code>numeric vector</code> with length of <code>ng</code> (number of genes) and elements in [0,1].</p>
</td></tr>
<tr><td><code id="struc_preparation2_+3A_epis">epis</code></td>
<td>
<p>epistasis value, describing the interaction between resistance alleles. <code>numeric</code>. <code>epis</code> = 0: no interaction i.e. additive effects of resistance alleles, <code>epis</code> &lt; 0: effect of resistance alleles is smaller than additive, <code>epis</code> &gt; 0: effect of resistance alleles is higher than additive.</p>
</td></tr>
<tr><td><code id="struc_preparation2_+3A_dom">dom</code></td>
<td>
<p>dominance of resistance alleles. <code>numeric vector</code> with length of <code>n_loci</code> (number of genes) and elements in [0,1].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prior to the simulation, a data.frame is generated to save results (<code>dfgenotype</code>). Additionally, a table with recombination probabilities (<code>xprobab</code>) is calculated. During the simulation run, probability values are not computed again but looked up in the table. PROSPER assumes diploid plants and maximum four resistance genes. To calculate the phenotypic resistance value for each genotype <code><a href="#topic+sel_resist">sel_resist</a></code> is called. If <code>n_loci &lt; 0</code> the structure is set up for no genetics at all in the simulation.
</p>


<h3>Value</h3>

<p>Returns a <code>list</code> of two <code>data.frame</code> and a <code>table</code>:
</p>

<ol>
<li><p> mf: all possible combinations of parental genotypes (see 'dfgenotype$genotype') are saved in one column 'mf' (male, female). The column 'mf' is a character vector. Each string of the vector has twice the length of the number of resistance loci under consideration.
</p>
</li>
<li><p> dfgenotype: the structure to save the results of one simulation cycle (year). After each cycle the data is reset to the new start values. The first column 'genotype' is a character vector. Each string of the vector has the length of the number of resistance loci under consideration. Each locus can have 0, 1 or 2 resistance alleles. The second column 'resist' saves resistance values that are calculated according to the equation in the section 'details'.
</p>
</li>
<li><p> xprobab: the probabilities of occurrence for all possible genotypes in the offspring (F-generation) with all possible parent genotypes (P-generation). Free recombination is assumed. Column names are the combinations of parental genes, row names are genotypes of the offspring.
</p>
</li></ol>



<h3>Warning</h3>

<p>The run of <code>struc_preparation</code> is time consuming. Duration strongly increases with the number of genes under consideration, <code>n_loci</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sel_resist">sel_resist</a></code> <code><a href="#topic+gen_freq">gen_freq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#generate the genotype and probability tables for a simulation with two resistance 
#loci with one dominant and one partial dominant resistant allele, no epistasis, and a 
#maximumx resistance value of 10. 
ls()
struc_preparation2(Rmx=10, af=c(0.01,0.5), epis=0, dom=c(1,0.3))
ls()
rm(dfgenotype, mf, xprobab)
</code></pre>

<hr>
<h2 id='struc_saveSimData'>Saving simulation data</h2><span id='topic+struc_saveSimData'></span>

<h3>Description</h3>

<p><code>struc_saveSimData</code> collects the data of one simulation cycle and returns an object to start a new simulation cycle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>struc_saveSimData(rep_counter, simcycle, start_names, end_names, simstruc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="struc_saveSimData_+3A_rep_counter">rep_counter</code></td>
<td>
<p>number of the current repetition. <code>integer</code>.</p>
</td></tr>
<tr><td><code id="struc_saveSimData_+3A_simcycle">simcycle</code></td>
<td>
<p>number of the current simulation cycle. <code>integer</code>.</p>
</td></tr>
<tr><td><code id="struc_saveSimData_+3A_start_names">start_names</code></td>
<td>
<p>names of the columns with the first population stage in the simulation. <code>character</code>.</p>
</td></tr>
<tr><td><code id="struc_saveSimData_+3A_end_names">end_names</code></td>
<td>
<p>names of the columns with the last population stage in the simulation. These are taken as the first stage in the next simulation cycle. <code>character</code>.</p>
</td></tr>
<tr><td><code id="struc_saveSimData_+3A_simstruc">simstruc</code></td>
<td>
<p>two numbers, the first defines the number of repetitions for the simulation, the second defines the number of simulation cycles in each repetition of the simulation. <code>numeric vector</code> with length 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the first simulation cycle the object <code>sim_result</code> of the class <code>prosper</code> is created and the first cycle results are saved. This object is the same for all cycles in all repetitions and gets all the results. The function can only be called once per simulation run.
</p>


<h3>Value</h3>

<p>A <code>prosper</code> object. SimData as a <code>data.table</code> with the repetition in the first and the simulation cycle in the second column.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+struc_endSim">struc_endSim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>struc_preparation2(Rmx=10, af=c(2,1), epis=0, dom=c(1,0.3))
dfgenotype$"SB_autumn" &lt;- c(4,23,0,123,53,98,45,3245,234)
dfgenotype$"SB_autumn_end" &lt;- c(0,2,0,123,434,5234,5678,123,2)
#creation of an example object with data of the first year
struc_saveSimData(rep_counter=1, simcycle=1, simstruc=c(5, 10), 
                                 start_names="SB_autumn", end_names="SB_autumn_end")
#creating some fictional population stages
dfgenotype$"SB_autumn" &lt;- c(1,1,1,1,0,0,0,0,4)
dfgenotype$"SB_autumn_end" &lt;- c(67,67,67,67,67,67,67,67,67)
#appending rows with the new results to the first results. necessary columns are inserted.
struc_saveSimData( rep_counter=1, simcycle=2,
                     start_names="SB_autumn", end_names="SB_autumn_end",
                      simstruc=c(repetitions, duration)) 
sim_result                               
rm(sim_result, dfgenotype, mf, xprobab)                               
</code></pre>

<hr>
<h2 id='summary'>Summary for prosper objects</h2><span id='topic+summary'></span><span id='topic+summary+2Cprosper-method'></span>

<h3>Description</h3>

<p>This function gives a first overview of a prosper simulation result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'prosper'
summary(object, geneticSumCol = "SB_autumn_end", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_+3A_object">object</code></td>
<td>
<p><code>prosper</code>, the result of a simulation with PROSPER.</p>
</td></tr>
<tr><td><code id="summary_+3A_geneticsumcol">geneticSumCol</code></td>
<td>
<p>column name of the prosper object the genetic summary is built of. <code>character</code>.</p>
</td></tr>
<tr><td><code id="summary_+3A_...">...</code></td>
<td>
<p>not in use.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the population dynamic part, means and standard deviations per simulation cycle for every numeric variable is calculated. Variability is introduced by repetitions. Additionally, a simple overview on factorial variables is given. For further calculations the numbers of individuals for each simulation cycle and repetition are provided.
</p>
<p>For the genetic part,  means and standard deviations of proportions of individuals with only sensitive alleles (allSalleles), only resistance alleles (allRalleles), mixed alleles (RSalleles), and the proportions of resistance alleles in the population (Ralleles) are calculated for each simulation cycle. The repetitions are responsible for the variability.
</p>


<h3>Value</h3>

<p>A named list of data.tables.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot">plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(param.LOLRI)
mod_lolri    &lt;- prosper.LOLRI(param.weed=param.LOLRI, area=30, af = c(0.005, 0.01),
                                       duration=3, repetitions=2)
summary(mod_lolri)
</code></pre>

<hr>
<h2 id='weed-parameters'>Parameterisations for population dynamic models - Examples</h2><span id='topic+weed-parameters'></span>

<h3>Description</h3>

<p>PROSPER entails full parameterized models. The parameterisation for these models are described here.
</p>


<h3>Format</h3>

<p>The provided data are given in tables with 10 columns.
</p>

<dl>
<dt>weed</dt><dd><p>the weed species.</p>
</dd> 
<dt>crop</dt><dd><p>the crop for which the parameters apply, <code>character</code>.</p>
</dd>
<dt>variable</dt><dd><p>the name of the variable, simulation step, <code>character</code>.</p>
</dd>
<dt>name</dt><dd><p>a naming of parameter, like a variable definition, <code>character</code>.</p>
</dd>
<dt>mean</dt><dd><p>the mean of the parameters defined with &quot;name&quot;, <code>numeric</code>.</p>
</dd>
<dt>se</dt><dd><p>the standard error of the parameters defined with &quot;name&quot;, <code>numeric</code>.</p>
</dd>
<dt>model</dt><dd><p>a (statistical) model to describe a specific simulation step. The model must be constructed of defined parameters with &quot;name&quot;, <code>character</code>.</p>
</dd>
<dt>range_low</dt><dd><p>a model might have a minimum level of use, then this value can be defined here, <code>numeric</code>.</p>
</dd>
<dt>range_up</dt><dd><p>a model might have a maximum level of use, then this value can be defined here, <code>numeric</code>.</p>
</dd>
<dt>source</dt><dd><p>if the value is taken from sources not owned by the author they can be named here.</p>
</dd>
</dl>



<h3>Details</h3>


<ol>
<li> <p><code>param.LOLRI</code> entails the data used for the introduction of PERTH by Renton et al. (2011) using the parameters of Pannell et al. (2004). With <code>prosper.LOLRI</code> one of the PERTH models can be reproduced.
</p>
</li>
<li> <p><code>param.ECHCG</code> contains parameters for population dynamic of <em>Echinochloa crus-galli</em> (Redwitz et al., 2016).
</p>
</li>
<li> <p><code>param.GALAP</code> contains parameters for population dynamic off <em>Galium aparine</em> taken from field experiments (Daedlow, 2015). 
</p>
</li></ol>



<h3>References</h3>


<ol>
<li><p> Bagavathiannan, MV, Norsworthy, JK (2013) Postdispersal loss of important arable weed seeds in the midsouthern United States. Weed Science, 61(4), 570-579.
</p>
</li>
<li><p> Bosnic, AC, Swanton, C (1997) Influence of Barnyardgrass (Echinochloa crus-galli) Time of Emergence and Density on Corn (Zea mays). Weed Science, 45(2), 276-282.
</p>
</li>
<li><p> Clay, SA, Kleinjan, J, Clay, DE, Forcella, Frank, Batchelor, W (2005) Growth and fecundity of several weed species in corn and soybean. Agronomy journal, 97(1), 294-302.
</p>
</li>
<li><p> Daedlow, D. (2015): About the contribution of seed predation on weed demography. (Doctoral dissertation). University of Rostock, Faculty of Agriculture and Environmental Sciences. GBV Gemeinsamer Verbundkatalog (Accession No. 839752644).
</p>
</li>
<li><p> Ogg, AG, Dawson, JH (1984) Time of emergence of eight weed species. Weed Science, 32(3), 327-335.
</p>
</li>
<li><p> Pannell, DJ, Stewart, V, Bennett, A, Monjardino, M, Schmidt, C, Powles, SB (2004) RIM:a bioeconomic model for integrated weedmanagement of Lolium rigidum in Western Australia. Agricultural Systems, 79, 305-325.
</p>
</li>
<li><p> Redwitz, C von, Pannwitt, H, Gerowitt, B (2016): About the interplay of sensitive and resistant biotypes in weed populations - simulation exercises for Echinochloa crus-galli in maize crops. Julius-Kuehn-Archiv 452, 93-99.
</p>
</li>
<li><p> Renton, M.; Diggle, A.; Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, 283, 14-27.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+prosper-models">prosper-models</a></code> <code><a href="#topic+quanti">quanti</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
