<!DOCTYPE html><html><head><title>Help for package protr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {protr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#protr-package'><p>protr: Generating Various Numerical Representation Schemes for Protein Sequences</p></a></li>
<li><a href='#AA2DACOR'><p>2D Autocorrelations Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AA3DMoRSE'><p>3D-MoRSE Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAACF'><p>Atom-Centred Fragments Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AABLOSUM100'><p>BLOSUM100 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AABLOSUM45'><p>BLOSUM45 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AABLOSUM50'><p>BLOSUM50 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AABLOSUM62'><p>BLOSUM62 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AABLOSUM80'><p>BLOSUM80 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AABurden'><p>Burden Eigenvalues Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAConn'><p>Connectivity Indices Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAConst'><p>Constitutional Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AACPSA'><p>CPSA Descriptors for 20 Amino Acids calculated by Discovery Studio</p></a></li>
<li><a href='#AADescAll'><p>All 2D Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAEdgeAdj'><p>Edge Adjacency Indices Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAEigIdx'><p>Eigenvalue-Based Indices Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAFGC'><p>Functional Group Counts Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAGeom'><p>Geometrical Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAGETAWAY'><p>GETAWAY Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAindex'><p>AAindex Data of 544 Physicochemical and Biological Properties</p>
for 20 Amino Acids</a></li>
<li><a href='#AAInfo'><p>Information Indices Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAMetaInfo'><p>Meta Information for the 20 Amino Acids</p></a></li>
<li><a href='#AAMOE2D'><p>2D Descriptors for 20 Amino Acids calculated by MOE 2011.10</p></a></li>
<li><a href='#AAMOE3D'><p>3D Descriptors for 20 Amino Acids calculated by MOE 2011.10</p></a></li>
<li><a href='#AAMolProp'><p>Molecular Properties Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAPAM120'><p>PAM120 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AAPAM250'><p>PAM250 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AAPAM30'><p>PAM30 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AAPAM40'><p>PAM40 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AAPAM70'><p>PAM70 Matrix for 20 Amino Acids</p></a></li>
<li><a href='#AARandic'><p>Randic Molecular Profiles Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AARDF'><p>RDF Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AATopo'><p>Topological Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AATopoChg'><p>Topological Charge Indices Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAWalk'><p>Walk and Path Counts Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#AAWHIM'><p>WHIM Descriptors for 20 Amino Acids calculated by Dragon</p></a></li>
<li><a href='#acc'><p>Auto Cross Covariance (ACC) for Generating Scales-Based Descriptors</p>
of the Same Length</a></li>
<li><a href='#crossSetSim'><p>Parallellized Protein Sequence Similarity Calculation Between Two Sets</p>
Based on Sequence Alignment (In-Memory Version)</a></li>
<li><a href='#extractAAC'><p>Amino Acid Composition Descriptor</p></a></li>
<li><a href='#extractAPAAC'><p>Amphiphilic Pseudo Amino Acid Composition (APseAAC) Descriptor</p></a></li>
<li><a href='#extractBLOSUM'><p>BLOSUM and PAM Matrix-Derived Descriptors</p></a></li>
<li><a href='#extractCTDC'><p>CTD Descriptors - Composition</p></a></li>
<li><a href='#extractCTDCClass'><p>CTD Descriptors - Composition (with customized amino acid</p>
classification support)</a></li>
<li><a href='#extractCTDD'><p>CTD Descriptors - Distribution</p></a></li>
<li><a href='#extractCTDDClass'><p>CTD Descriptors - Distribution (with customized amino acid</p>
classification support)</a></li>
<li><a href='#extractCTDT'><p>CTD Descriptors - Transition</p></a></li>
<li><a href='#extractCTDTClass'><p>CTD Descriptors - Transition (with customized amino acid</p>
classification support)</a></li>
<li><a href='#extractCTriad'><p>Conjoint Triad Descriptor</p></a></li>
<li><a href='#extractCTriadClass'><p>Conjoint Triad Descriptor (with customized amino acid</p>
classification support)</a></li>
<li><a href='#extractDC'><p>Dipeptide Composition Descriptor</p></a></li>
<li><a href='#extractDescScales'><p>Scales-Based Descriptors with 20+ classes of Molecular Descriptors</p></a></li>
<li><a href='#extractFAScales'><p>Scales-Based Descriptors derived by Factor Analysis</p></a></li>
<li><a href='#extractGeary'><p>Geary Autocorrelation Descriptor</p></a></li>
<li><a href='#extractMDSScales'><p>Scales-Based Descriptors derived by Multidimensional Scaling</p></a></li>
<li><a href='#extractMoran'><p>Moran Autocorrelation Descriptor</p></a></li>
<li><a href='#extractMoreauBroto'><p>Normalized Moreau-Broto Autocorrelation Descriptor</p></a></li>
<li><a href='#extractPAAC'><p>Pseudo Amino Acid Composition (PseAAC) Descriptor</p></a></li>
<li><a href='#extractProtFP'><p>Amino Acid Properties Based Scales Descriptors (Protein Fingerprint)</p></a></li>
<li><a href='#extractProtFPGap'><p>Amino Acid Properties Based Scales Descriptors (Protein Fingerprint)</p>
with Gap Support</a></li>
<li><a href='#extractPSSM'><p>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence</p></a></li>
<li><a href='#extractPSSMAcc'><p>Profile-based protein representation derived by PSSM</p>
(Position-Specific Scoring Matrix) and auto cross covariance</a></li>
<li><a href='#extractPSSMFeature'><p>Profile-based protein representation derived by PSSM</p>
(Position-Specific Scoring Matrix)</a></li>
<li><a href='#extractQSO'><p>Quasi-Sequence-Order (QSO) Descriptor</p></a></li>
<li><a href='#extractScales'><p>Scales-Based Descriptors derived by Principal Components Analysis</p></a></li>
<li><a href='#extractScalesGap'><p>Scales-Based Descriptors derived by Principal Components Analysis</p>
(with Gap Support)</a></li>
<li><a href='#extractSOCN'><p>Sequence-Order-Coupling Numbers</p></a></li>
<li><a href='#extractTC'><p>Tripeptide Composition Descriptor</p></a></li>
<li><a href='#getUniProt'><p>Retrieve Protein Sequences from UniProt by Protein ID</p></a></li>
<li><a href='#OptAA3d'><p>OptAA3d.sdf - 20 Amino Acids Optimized with MOE 2011.10 (Semiempirical AM1)</p></a></li>
<li><a href='#parGOSim'><p>Protein Similarity Calculation based on Gene Ontology</p>
(GO) Similarity</a></li>
<li><a href='#parSeqSim'><p>Parallellized Protein Sequence Similarity Calculation Based on</p>
Sequence Alignment (In-Memory Version)</a></li>
<li><a href='#parSeqSimDisk'><p>Parallellized Protein Sequence Similarity Calculation Based on</p>
Sequence Alignment (Disk-Based Version)</a></li>
<li><a href='#protcheck'><p>Protein sequence amino acid type sanity check</p></a></li>
<li><a href='#protseg'><p>Protein Sequence Segmentation/Partition</p></a></li>
<li><a href='#readFASTA'><p>Read Protein Sequences in FASTA Format</p></a></li>
<li><a href='#readPDB'><p>Read Protein Sequences in PDB Format</p></a></li>
<li><a href='#removeGaps'><p>Remove or replace gaps from protein sequences.</p></a></li>
<li><a href='#twoGOSim'><p>Protein Similarity Calculation based on Gene Ontology (GO) Similarity</p></a></li>
<li><a href='#twoSeqSim'><p>Protein Sequence Alignment for Two Protein Sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.7-0</td>
</tr>
<tr>
<td>Title:</td>
<td>Generating Various Numerical Representation Schemes for Protein
Sequences</td>
</tr>
<tr>
<td>Description:</td>
<td>Comprehensive toolkit for generating various numerical
    features of protein sequences described in Xiao et al. (2015)
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv042">doi:10.1093/bioinformatics/btv042</a>&gt;. For full functionality,
    the software 'ncbi-blast+' is needed, see
    <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html">https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html</a>
    for more information.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-3-Clause">BSD_3_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://nanx.me/protr/">https://nanx.me/protr/</a>, <a href="https://github.com/nanxstats/protr">https://github.com/nanxstats/protr</a>,
<a href="http://protr.org">http://protr.org</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/nanxstats/protr/issues">https://github.com/nanxstats/protr/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>ncbi-blast+ (see
&lt;https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html&gt;)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, Biostrings, GOSemSim, foreach, doParallel,
org.Hs.eg.db</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-30 06:19:21 UTC; nanx</td>
</tr>
<tr>
<td>Author:</td>
<td>Nan Xiao <a href="https://orcid.org/0000-0002-0250-5673"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Qing-Song Xu [aut],
  Dong-Sheng Cao [aut],
  Sebastian Mueller [ctb] (Alva Genomics)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nan Xiao &lt;me@nanx.me&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-30 07:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='protr-package'>protr: Generating Various Numerical Representation Schemes for Protein Sequences</h2><span id='topic+protr'></span><span id='topic+protr-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Comprehensive toolkit for generating various numerical features of protein sequences described in Xiao et al. (2015) <a href="https://doi.org/10.1093/bioinformatics/btv042">doi:10.1093/bioinformatics/btv042</a>. For full functionality, the software 'ncbi-blast+' is needed, see <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html">https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html</a> for more information.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Nan Xiao <a href="mailto:me@nanx.me">me@nanx.me</a> (<a href="https://orcid.org/0000-0002-0250-5673">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Qing-Song Xu <a href="mailto:qsxu@csu.edu.cn">qsxu@csu.edu.cn</a>
</p>
</li>
<li><p> Dong-Sheng Cao
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Sebastian Mueller (Alva Genomics) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://nanx.me/protr/">https://nanx.me/protr/</a>
</p>
</li>
<li> <p><a href="https://github.com/nanxstats/protr">https://github.com/nanxstats/protr</a>
</p>
</li>
<li> <p><a href="http://protr.org">http://protr.org</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/nanxstats/protr/issues">https://github.com/nanxstats/protr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='AA2DACOR'>2D Autocorrelations Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AA2DACOR'></span>

<h3>Description</h3>

<p>This dataset includes the 2D autocorrelations descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AA2DACOR)
</code></pre>

<hr>
<h2 id='AA3DMoRSE'>3D-MoRSE Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AA3DMoRSE'></span>

<h3>Description</h3>

<p>This dataset includes the 3D-MoRSE descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AA3DMoRSE)
</code></pre>

<hr>
<h2 id='AAACF'>Atom-Centred Fragments Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAACF'></span>

<h3>Description</h3>

<p>This dataset includes the atom-centred fragments descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAACF)
</code></pre>

<hr>
<h2 id='AABLOSUM100'>BLOSUM100 Matrix for 20 Amino Acids</h2><span id='topic+AABLOSUM100'></span>

<h3>Description</h3>

<p>BLOSUM100 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AABLOSUM100)
</code></pre>

<hr>
<h2 id='AABLOSUM45'>BLOSUM45 Matrix for 20 Amino Acids</h2><span id='topic+AABLOSUM45'></span>

<h3>Description</h3>

<p>BLOSUM45 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AABLOSUM45)
</code></pre>

<hr>
<h2 id='AABLOSUM50'>BLOSUM50 Matrix for 20 Amino Acids</h2><span id='topic+AABLOSUM50'></span>

<h3>Description</h3>

<p>BLOSUM50 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AABLOSUM50)
</code></pre>

<hr>
<h2 id='AABLOSUM62'>BLOSUM62 Matrix for 20 Amino Acids</h2><span id='topic+AABLOSUM62'></span>

<h3>Description</h3>

<p>BLOSUM62 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AABLOSUM62)
</code></pre>

<hr>
<h2 id='AABLOSUM80'>BLOSUM80 Matrix for 20 Amino Acids</h2><span id='topic+AABLOSUM80'></span>

<h3>Description</h3>

<p>BLOSUM80 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AABLOSUM80)
</code></pre>

<hr>
<h2 id='AABurden'>Burden Eigenvalues Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AABurden'></span>

<h3>Description</h3>

<p>This dataset includes the Burden eigenvalues descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AABurden)
</code></pre>

<hr>
<h2 id='AAConn'>Connectivity Indices Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAConn'></span>

<h3>Description</h3>

<p>This dataset includes the connectivity indices descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAConn)
</code></pre>

<hr>
<h2 id='AAConst'>Constitutional Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAConst'></span>

<h3>Description</h3>

<p>This dataset includes the constitutional descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAConst)
</code></pre>

<hr>
<h2 id='AACPSA'>CPSA Descriptors for 20 Amino Acids calculated by Discovery Studio</h2><span id='topic+AACPSA'></span>

<h3>Description</h3>

<p>This dataset includes the CPSA descriptors of the 20 amino acids
calculated by Discovery Studio (version 2.5) used for scales extraction
in this package.
</p>


<h3>Details</h3>

<p>All amino acid molecules had also been optimized with MOE 2011.10
(semiempirical AM1) before calculating these CPSA descriptors.
The SDF file containing the information of the optimized amino acid
molecules is included in this package. See <code><a href="#topic+OptAA3d">OptAA3d</a></code>
for more information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AACPSA)
</code></pre>

<hr>
<h2 id='AADescAll'>All 2D Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AADescAll'></span>

<h3>Description</h3>

<p>This dataset includes all the 2D descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AADescAll)
</code></pre>

<hr>
<h2 id='AAEdgeAdj'>Edge Adjacency Indices Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAEdgeAdj'></span>

<h3>Description</h3>

<p>This dataset includes the edge adjacency indices descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAEdgeAdj)
</code></pre>

<hr>
<h2 id='AAEigIdx'>Eigenvalue-Based Indices Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAEigIdx'></span>

<h3>Description</h3>

<p>This dataset includes the eigenvalue-based indices descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAEigIdx)
</code></pre>

<hr>
<h2 id='AAFGC'>Functional Group Counts Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAFGC'></span>

<h3>Description</h3>

<p>This dataset includes the functional group counts descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAFGC)
</code></pre>

<hr>
<h2 id='AAGeom'>Geometrical Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAGeom'></span>

<h3>Description</h3>

<p>This dataset includes the geometrical descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAGeom)
</code></pre>

<hr>
<h2 id='AAGETAWAY'>GETAWAY Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAGETAWAY'></span>

<h3>Description</h3>

<p>This dataset includes the GETAWAY descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAGETAWAY)
</code></pre>

<hr>
<h2 id='AAindex'>AAindex Data of 544 Physicochemical and Biological Properties
for 20 Amino Acids</h2><span id='topic+AAindex'></span>

<h3>Description</h3>

<p>The data was extracted from the AAindex1 database ver 9.1
(<a href="https://www.genome.jp/ftp/db/community/aaindex/old/ver9.1/aaindex1">https://www.genome.jp/ftp/db/community/aaindex/old/ver9.1/aaindex1</a>)
as of November, 2012 (data last modified on 2006-08-14).
</p>


<h3>Details</h3>

<p>With this dataset, users can investigate each property's accession number
and other details. See <a href="https://www.genome.jp/aaindex/">https://www.genome.jp/aaindex/</a>
for more information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAindex)
</code></pre>

<hr>
<h2 id='AAInfo'>Information Indices Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAInfo'></span>

<h3>Description</h3>

<p>This dataset includes the information indices descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAInfo)
</code></pre>

<hr>
<h2 id='AAMetaInfo'>Meta Information for the 20 Amino Acids</h2><span id='topic+AAMetaInfo'></span>

<h3>Description</h3>

<p>This dataset includes the meta information of
the 20 amino acids used for the 2D and 3D descriptor
calculation in this package. Each column represents:
</p>

<dl>
<dt><code>AAName</code></dt><dd><p>Amino acid name</p>
</dd>
<dt><code>Short</code></dt><dd><p>One-letter representation</p>
</dd>
<dt><code>Abbreviation</code></dt><dd><p>Three-letter representation</p>
</dd>
<dt><code>mol</code></dt><dd><p>SMILES representation</p>
</dd>
<dt><code>PUBCHEM_COMPOUND_CID</code></dt><dd><p>PubChem CID for the amino acid</p>
</dd>
<dt><code>PUBCHEM_LINK</code></dt><dd><p>PubChem link for the amino acid</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(AAMetaInfo)
</code></pre>

<hr>
<h2 id='AAMOE2D'>2D Descriptors for 20 Amino Acids calculated by MOE 2011.10</h2><span id='topic+AAMOE2D'></span>

<h3>Description</h3>

<p>This dataset includes the 2D descriptors of
the 20 amino acids calculated by MOE 2011.10
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAMOE2D)
</code></pre>

<hr>
<h2 id='AAMOE3D'>3D Descriptors for 20 Amino Acids calculated by MOE 2011.10</h2><span id='topic+AAMOE3D'></span>

<h3>Description</h3>

<p>This dataset includes the 3D descriptors of the 20 amino acids
calculated by MOE 2011.10 used for scales extraction in this package.
All amino acid molecules had also been optimized with MOE (semiempirical AM1)
before calculating these 3D descriptors.
The SDF file containing the information of the optimized amino acid molecules
is included in this package. See <code><a href="#topic+OptAA3d">OptAA3d</a></code> for more information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAMOE3D)
</code></pre>

<hr>
<h2 id='AAMolProp'>Molecular Properties Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAMolProp'></span>

<h3>Description</h3>

<p>This dataset includes the molecular properties descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAMolProp)
</code></pre>

<hr>
<h2 id='AAPAM120'>PAM120 Matrix for 20 Amino Acids</h2><span id='topic+AAPAM120'></span>

<h3>Description</h3>

<p>PAM120 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAPAM120)
</code></pre>

<hr>
<h2 id='AAPAM250'>PAM250 Matrix for 20 Amino Acids</h2><span id='topic+AAPAM250'></span>

<h3>Description</h3>

<p>PAM250 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAPAM250)
</code></pre>

<hr>
<h2 id='AAPAM30'>PAM30 Matrix for 20 Amino Acids</h2><span id='topic+AAPAM30'></span>

<h3>Description</h3>

<p>PAM30 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAPAM30)
</code></pre>

<hr>
<h2 id='AAPAM40'>PAM40 Matrix for 20 Amino Acids</h2><span id='topic+AAPAM40'></span>

<h3>Description</h3>

<p>PAM40 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAPAM40)
</code></pre>

<hr>
<h2 id='AAPAM70'>PAM70 Matrix for 20 Amino Acids</h2><span id='topic+AAPAM70'></span>

<h3>Description</h3>

<p>PAM70 Matrix for the 20 amino acids. The matrix was extracted from the
<code>Biostrings</code> package of Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAPAM70)
</code></pre>

<hr>
<h2 id='AARandic'>Randic Molecular Profiles Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AARandic'></span>

<h3>Description</h3>

<p>This dataset includes the Randic molecular profiles descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AARandic)
</code></pre>

<hr>
<h2 id='AARDF'>RDF Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AARDF'></span>

<h3>Description</h3>

<p>This dataset includes the RDF descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AARDF)
</code></pre>

<hr>
<h2 id='AATopo'>Topological Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AATopo'></span>

<h3>Description</h3>

<p>This dataset includes the topological descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AATopo)
</code></pre>

<hr>
<h2 id='AATopoChg'>Topological Charge Indices Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AATopoChg'></span>

<h3>Description</h3>

<p>This dataset includes the topological charge indices descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AATopoChg)
</code></pre>

<hr>
<h2 id='AAWalk'>Walk and Path Counts Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAWalk'></span>

<h3>Description</h3>

<p>This dataset includes the walk and path counts descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAWalk)
</code></pre>

<hr>
<h2 id='AAWHIM'>WHIM Descriptors for 20 Amino Acids calculated by Dragon</h2><span id='topic+AAWHIM'></span>

<h3>Description</h3>

<p>This dataset includes the WHIM descriptors of
the 20 amino acids calculated by Dragon (version 5.4)
used for scales extraction in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AAWHIM)
</code></pre>

<hr>
<h2 id='acc'>Auto Cross Covariance (ACC) for Generating Scales-Based Descriptors
of the Same Length</h2><span id='topic+acc'></span>

<h3>Description</h3>

<p>This function calculates the auto covariance and auto cross covariance
for generating scale-based descriptors of the same length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acc(mat, lag)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="acc_+3A_mat">mat</code></td>
<td>
<p>A <code>p * n</code> matrix. Each row represents one scale
(total <code>p</code> scales), each column represents one amino acid position
(total <code>n</code> amino acids).</p>
</td></tr>
<tr><td><code id="acc_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector, the element names are
constructed by: the scales index (crossed scales index) and lag index.
</p>


<h3>Note</h3>

<p>Please see the references for details about auto cross covariance.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Wold, S., Jonsson, J., Sjorstrom, M., Sandberg, M., &amp; Rannar, S. (1993).
DNA and peptide sequences and chemical processes multivariately modelled
by principal component analysis and partial least-squares projections
to latent structures.
<em>Analytica chimica acta</em>, 277(2), 239&ndash;253.
</p>
<p>Sjostrom, M., Rannar, S., &amp; Wieslander, A. (1995).
Polypeptide sequence property relationships in <em>Escherichia coli</em>
based on auto cross covariances.
<em>Chemometrics and intelligent laboratory systems</em>, 29(2), 295&ndash;305.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractScales">extractScales</a></code> for scales-based descriptors.
For more details, see <code><a href="#topic+extractDescScales">extractDescScales</a></code>
and <code><a href="#topic+extractProtFP">extractProtFP</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- 8 # p is the scales number
n &lt;- 200 # n is the amino acid number
lag &lt;- 7 # the lag paramter
mat &lt;- matrix(rnorm(p * n), nrow = p, ncol = n)
acc(mat, lag)
</code></pre>

<hr>
<h2 id='crossSetSim'>Parallellized Protein Sequence Similarity Calculation Between Two Sets
Based on Sequence Alignment (In-Memory Version)</h2><span id='topic+crossSetSim'></span>

<h3>Description</h3>

<p>Parallellized calculation of protein sequence similarity based on
sequence alignment between two sets of protein sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crossSetSim(
  protlist1,
  protlist2,
  type = "local",
  cores = 2,
  submat = "BLOSUM62",
  gap.opening = 10,
  gap.extension = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crossSetSim_+3A_protlist1">protlist1</code></td>
<td>
<p>A length <code>n</code> list containing <code>n</code> protein sequences,
each component of the list is a character string, storing one protein
sequence. Unknown sequences should be represented as <code>""</code>.</p>
</td></tr>
<tr><td><code id="crossSetSim_+3A_protlist2">protlist2</code></td>
<td>
<p>A length <code>n</code> list containing <code>m</code> protein sequences,
each component of the list is a character string, storing one protein
sequence. Unknown sequences should be represented as <code>""</code>.</p>
</td></tr>
<tr><td><code id="crossSetSim_+3A_type">type</code></td>
<td>
<p>Type of alignment, default is <code>"local"</code>,
can be <code>"global"</code> or <code>"local"</code>,
where <code>"global"</code> represents Needleman-Wunsch global alignment;
<code>"local"</code> represents Smith-Waterman local alignment.</p>
</td></tr>
<tr><td><code id="crossSetSim_+3A_cores">cores</code></td>
<td>
<p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users can use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p>
</td></tr>
<tr><td><code id="crossSetSim_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix, default is <code>"BLOSUM62"</code>,
can be one of <code>"BLOSUM45"</code>, <code>"BLOSUM50"</code>, <code>"BLOSUM62"</code>,
<code>"BLOSUM80"</code>, <code>"BLOSUM100"</code>, <code>"PAM30"</code>,
<code>"PAM40"</code>, <code>"PAM70"</code>, <code>"PAM120"</code>, or <code>"PAM250"</code>.</p>
</td></tr>
<tr><td><code id="crossSetSim_+3A_gap.opening">gap.opening</code></td>
<td>
<p>The cost required to open a gap of any length
in the alignment. Defaults to 10.</p>
</td></tr>
<tr><td><code id="crossSetSim_+3A_gap.extension">gap.extension</code></td>
<td>
<p>The cost to extend the length of an existing
gap by 1. Defaults to 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>n</code> x <code>m</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Sebastian Mueller &lt;<a href="https://alva-genomics.com">https://alva-genomics.com</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Be careful when testing this since it involves parallelisation
# and might produce unpredictable results in some environments

library("Biostrings")
library("foreach")
library("doParallel")

s1 &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
s2 &lt;- readFASTA(system.file("protseq/P08218.fasta", package = "protr"))[[1]]
s3 &lt;- readFASTA(system.file("protseq/P10323.fasta", package = "protr"))[[1]]
s4 &lt;- readFASTA(system.file("protseq/P20160.fasta", package = "protr"))[[1]]
s5 &lt;- readFASTA(system.file("protseq/Q9NZP8.fasta", package = "protr"))[[1]]

plist1 &lt;- list(s1 = s1, s2 = s2, s4 = s4)
plist2 &lt;- list(s3 = s3, s4_again = s4, s5 = s5, s1_again = s1)
psimmat &lt;- crossSetSim(plist1, plist2)
colnames(psimmat) &lt;- names(plist1)
rownames(psimmat) &lt;- names(plist2)
print(psimmat)
#                 s1         s2         s4
# s3       0.10236985 0.18858241 0.05819984
# s4_again 0.04921696 0.12124217 1.00000000
# s5       0.03943488 0.06391103 0.05714638
# s1_again 1.00000000 0.11825938 0.04921696

## End(Not run)
</code></pre>

<hr>
<h2 id='extractAAC'>Amino Acid Composition Descriptor</h2><span id='topic+extractAAC'></span>

<h3>Description</h3>

<p>This function calculates the Amino Acid Composition descriptor (dim: 20).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractAAC(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractAAC_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 20 named vector
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>M. Bhasin, G. P. S. Raghava.
Classification of Nuclear Receptors Based on
Amino Acid Composition and Dipeptide Composition.
<em>Journal of Biological Chemistry</em>, 2004, 279, 23262.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractDC">extractDC</a></code> and <code><a href="#topic+extractTC">extractTC</a></code>
for Dipeptide Composition and Tripeptide Composition descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractAAC(x)
</code></pre>

<hr>
<h2 id='extractAPAAC'>Amphiphilic Pseudo Amino Acid Composition (APseAAC) Descriptor</h2><span id='topic+extractAPAAC'></span>

<h3>Description</h3>

<p>This function calculates the Amphiphilic Pseudo Amino Acid
Composition (APseAAC, or APAAC) descriptor (dim: <code>20 + (n * lambda)</code>,
<code>n</code> is the number of properties selected, default is 80).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractAPAAC(
  x,
  props = c("Hydrophobicity", "Hydrophilicity"),
  lambda = 30,
  w = 0.05,
  customprops = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractAPAAC_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractAPAAC_+3A_props">props</code></td>
<td>
<p>A character vector, specifying the properties used.
2 properties are used by default, as listed below:
</p>

<dl>
<dt><code>'Hydrophobicity'</code></dt><dd><p>Hydrophobicity value of the 20 amino acids</p>
</dd>
<dt><code>'Hydrophilicity'</code></dt><dd><p>Hydrophilicity value of the 20 amino acids</p>
</dd></dl>
</td></tr>
<tr><td><code id="extractAPAAC_+3A_lambda">lambda</code></td>
<td>
<p>The lambda parameter for the APAAC descriptors, default is 30.</p>
</td></tr>
<tr><td><code id="extractAPAAC_+3A_w">w</code></td>
<td>
<p>The weighting factor, default is 0.05.</p>
</td></tr>
<tr><td><code id="extractAPAAC_+3A_customprops">customprops</code></td>
<td>
<p>A <code>n x 21</code> named data frame contains <code>n</code>
customized property. Each row contains one property.
The column order for different amino acid types is
<code>'AccNo'</code>, <code>'A'</code>, <code>'R'</code>, <code>'N'</code>,
<code>'D'</code>, <code>'C'</code>, <code>'E'</code>, <code>'Q'</code>,
<code>'G'</code>, <code>'H'</code>, <code>'I'</code>, <code>'L'</code>,
<code>'K'</code>, <code>'M'</code>, <code>'F'</code>, <code>'P'</code>,
<code>'S'</code>, <code>'T'</code>, <code>'W'</code>, <code>'Y'</code>,
<code>'V'</code>, and the columns should also be
<em>exactly</em> named like this.
The <code>AccNo</code> column contains the properties' names.
Then users should explicitly specify these properties
with these names in the argument <code>props</code>.
See the examples below for a demonstration.
The default value for <code>customprops</code> is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>20 + n * lambda</code> named vector,
<code>n</code> is the number of properties selected.
</p>


<h3>Note</h3>

<p>Note the default <code>20 * 2</code> <code>prop</code> values have already been
independently given in the function. Users can also specify
other (up to 544) properties with the Accession Number in
the <code><a href="#topic+AAindex">AAindex</a></code> data, with or without the default
three properties, which means users should explicitly specify
the properties to use. For this descriptor type, users need to
intelligently evaluate the underlying details of the descriptors
provided, instead of using this function with their data blindly.
It would be wise to use some negative and positive control comparisons
where relevant to help guide interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Kuo-Chen Chou. Prediction of Protein Cellular Attributes
Using Pseudo-Amino Acid Composition.
<em>PROTEINS: Structure, Function, and Genetics</em>, 2001, 43: 246-255.
</p>
<p>Kuo-Chen Chou. Using Amphiphilic Pseudo Amino Acid Composition
to Predict Enzyme Subfamily Classes. <em>Bioinformatics</em>, 2005, 21, 10-19.
</p>
<p>JACS, 1962, 84: 4240-4246. (C. Tanford). (The hydrophobicity data)
</p>
<p>PNAS, 1981, 78:3824-3828 (T.P.Hopp &amp; K.R.Woods). (The hydrophilicity data)
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractPAAC">extractPAAC</a></code> for the
pseudo amino acid composition (PseAAC) descriptor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractAPAAC(x)

myprops &lt;- data.frame(
  AccNo = c("MyProp1", "MyProp2", "MyProp3"),
  A = c(0.62, -0.5, 15), R = c(-2.53, 3, 101),
  N = c(-0.78, 0.2, 58), D = c(-0.9, 3, 59),
  C = c(0.29, -1, 47), E = c(-0.74, 3, 73),
  Q = c(-0.85, 0.2, 72), G = c(0.48, 0, 1),
  H = c(-0.4, -0.5, 82), I = c(1.38, -1.8, 57),
  L = c(1.06, -1.8, 57), K = c(-1.5, 3, 73),
  M = c(0.64, -1.3, 75), F = c(1.19, -2.5, 91),
  P = c(0.12, 0, 42), S = c(-0.18, 0.3, 31),
  T = c(-0.05, -0.4, 45), W = c(0.81, -3.4, 130),
  Y = c(0.26, -2.3, 107), V = c(1.08, -1.5, 43)
)

# use 2 default properties, 4 properties from the
# AAindex database, and 3 cutomized properties
extractAPAAC(
  x,
  customprops = myprops,
  props = c(
    "Hydrophobicity", "Hydrophilicity",
    "CIDH920105", "BHAR880101",
    "CHAM820101", "CHAM820102",
    "MyProp1", "MyProp2", "MyProp3"
  )
)
</code></pre>

<hr>
<h2 id='extractBLOSUM'>BLOSUM and PAM Matrix-Derived Descriptors</h2><span id='topic+extractBLOSUM'></span>

<h3>Description</h3>

<p>This function calculates BLOSUM matrix-derived descriptors.
For users' convenience, <code>protr</code> provides the BLOSUM45, BLOSUM50,
BLOSUM62, BLOSUM80, BLOSUM100, PAM30, PAM40, PAM70, PAM120, and PAM250
matrices for the 20 amino acids to select from.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractBLOSUM(x, submat = "AABLOSUM62", k, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractBLOSUM_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractBLOSUM_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix for the 20 amino acids. Should be one of
<code>AABLOSUM45</code>, <code>AABLOSUM50</code>, <code>AABLOSUM62</code>,
<code>AABLOSUM80</code>, <code>AABLOSUM100</code>, <code>AAPAM30</code>,
<code>AAPAM40</code>, <code>AAPAM70</code>, <code>AAPAM120</code>, or <code>AAPAM250</code>.
Default is <code>"AABLOSUM62"</code>.</p>
</td></tr>
<tr><td><code id="extractBLOSUM_+3A_k">k</code></td>
<td>
<p>Integer. The number of selected scales (i.e. the first
<code>k</code> scales) derived by the substitution matrix.
This can be selected according to the printed relative importance values.</p>
</td></tr>
<tr><td><code id="extractBLOSUM_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractBLOSUM_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the substitution matrix
(<code>submat</code>) before doing eigen decomposition? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractBLOSUM_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether we print the relative importance of
each scales (diagnal value of the eigen decomposition result matrix B)
or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector, <code>p</code> is the number
of scales selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Georgiev, A. G. (2009).
Interpretable numerical descriptors of amino acid space.
Journal of Computational Biology, 16(5), 703&ndash;723.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
blosum &lt;- extractBLOSUM(x, submat = "AABLOSUM62", k = 5, lag = 7, scale = TRUE, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractCTDC'>CTD Descriptors - Composition</h2><span id='topic+extractCTDC'></span>

<h3>Description</h3>

<p>This function calculates the Composition descriptor of the
CTD descriptors (dim: 21).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTDC(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTDC_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 21 named vector
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractCTDT">extractCTDT</a></code> and <code><a href="#topic+extractCTDD">extractCTDD</a></code>
for Transition and Distribution of the CTD descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractCTDC(x)
</code></pre>

<hr>
<h2 id='extractCTDCClass'>CTD Descriptors - Composition (with customized amino acid
classification support)</h2><span id='topic+extractCTDCClass'></span>

<h3>Description</h3>

<p>This function calculates the Composition descriptor of the
CTD descriptors, with customized amino acid classification support.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTDCClass(x, aagroup1, aagroup2, aagroup3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTDCClass_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractCTDCClass_+3A_aagroup1">aagroup1</code></td>
<td>
<p>A named list which contains the first group of customized
amino acid classification. See example below.</p>
</td></tr>
<tr><td><code id="extractCTDCClass_+3A_aagroup2">aagroup2</code></td>
<td>
<p>A named list which contains the second group of customized
amino acid classification. See example below.</p>
</td></tr>
<tr><td><code id="extractCTDCClass_+3A_aagroup3">aagroup3</code></td>
<td>
<p>A named list which contains the third group of customized
amino acid classification. See example below.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>k * 3</code> named vector, <code>k</code> is the number of
amino acid properties used.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractCTDTClass">extractCTDTClass</a></code> and
<code><a href="#topic+extractCTDDClass">extractCTDDClass</a></code> for Transition and Distribution
of the CTD descriptors with customized amino acid classification support.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]

# using five customized amino acid property classification
group1 &lt;- list(
  "hydrophobicity" = c("R", "K", "E", "D", "Q", "N"),
  "normwaalsvolume" = c("G", "A", "S", "T", "P", "D", "C"),
  "polarizability" = c("G", "A", "S", "D", "T"),
  "secondarystruct" = c("E", "A", "L", "M", "Q", "K", "R", "H"),
  "solventaccess" = c("A", "L", "F", "C", "G", "I", "V", "W")
)

group2 &lt;- list(
  "hydrophobicity" = c("G", "A", "S", "T", "P", "H", "Y"),
  "normwaalsvolume" = c("N", "V", "E", "Q", "I", "L"),
  "polarizability" = c("C", "P", "N", "V", "E", "Q", "I", "L"),
  "secondarystruct" = c("V", "I", "Y", "C", "W", "F", "T"),
  "solventaccess" = c("R", "K", "Q", "E", "N", "D")
)

group3 &lt;- list(
  "hydrophobicity" = c("C", "L", "V", "I", "M", "F", "W"),
  "normwaalsvolume" = c("M", "H", "K", "F", "R", "Y", "W"),
  "polarizability" = c("K", "M", "H", "F", "R", "Y", "W"),
  "secondarystruct" = c("G", "N", "P", "S", "D"),
  "solventaccess" = c("M", "S", "P", "T", "H", "Y")
)

extractCTDCClass(x, aagroup1 = group1, aagroup2 = group2, aagroup3 = group3)
</code></pre>

<hr>
<h2 id='extractCTDD'>CTD Descriptors - Distribution</h2><span id='topic+extractCTDD'></span>

<h3>Description</h3>

<p>This function calculates the Distribution descriptor of the
CTD descriptors (dim: 105).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTDD(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTDD_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 105 named vector
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractCTDC">extractCTDC</a></code> and <code><a href="#topic+extractCTDT">extractCTDT</a></code>
for Composition and Transition of the CTD descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractCTDD(x)
</code></pre>

<hr>
<h2 id='extractCTDDClass'>CTD Descriptors - Distribution (with customized amino acid
classification support)</h2><span id='topic+extractCTDDClass'></span>

<h3>Description</h3>

<p>This function calculates the Distribution descriptor of the
CTD descriptors, with customized amino acid classification support.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTDDClass(x, aagroup1, aagroup2, aagroup3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTDDClass_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractCTDDClass_+3A_aagroup1">aagroup1</code></td>
<td>
<p>A named list which contains the first group of customized
amino acid classification. See example below.</p>
</td></tr>
<tr><td><code id="extractCTDDClass_+3A_aagroup2">aagroup2</code></td>
<td>
<p>A named list which contains the second group of customized
amino acid classification. See example below.</p>
</td></tr>
<tr><td><code id="extractCTDDClass_+3A_aagroup3">aagroup3</code></td>
<td>
<p>A named list which contains the third group of customized
amino acid classification. See example below.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>k * 15</code> named vector, <code>k</code> is the number of
amino acid properties used.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractCTDCClass">extractCTDCClass</a></code> and
<code><a href="#topic+extractCTDTClass">extractCTDTClass</a></code> for Composition and Transition of
the CTD descriptors with customized amino acid classification support.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]

# using five customized amino acid property classification
group1 &lt;- list(
  "hydrophobicity" = c("R", "K", "E", "D", "Q", "N"),
  "normwaalsvolume" = c("G", "A", "S", "T", "P", "D", "C"),
  "polarizability" = c("G", "A", "S", "D", "T"),
  "secondarystruct" = c("E", "A", "L", "M", "Q", "K", "R", "H"),
  "solventaccess" = c("A", "L", "F", "C", "G", "I", "V", "W")
)

group2 &lt;- list(
  "hydrophobicity" = c("G", "A", "S", "T", "P", "H", "Y"),
  "normwaalsvolume" = c("N", "V", "E", "Q", "I", "L"),
  "polarizability" = c("C", "P", "N", "V", "E", "Q", "I", "L"),
  "secondarystruct" = c("V", "I", "Y", "C", "W", "F", "T"),
  "solventaccess" = c("R", "K", "Q", "E", "N", "D")
)

group3 &lt;- list(
  "hydrophobicity" = c("C", "L", "V", "I", "M", "F", "W"),
  "normwaalsvolume" = c("M", "H", "K", "F", "R", "Y", "W"),
  "polarizability" = c("K", "M", "H", "F", "R", "Y", "W"),
  "secondarystruct" = c("G", "N", "P", "S", "D"),
  "solventaccess" = c("M", "S", "P", "T", "H", "Y")
)

extractCTDDClass(x, aagroup1 = group1, aagroup2 = group2, aagroup3 = group3)
</code></pre>

<hr>
<h2 id='extractCTDT'>CTD Descriptors - Transition</h2><span id='topic+extractCTDT'></span>

<h3>Description</h3>

<p>This function calculates the Transition descriptor of the
CTD descriptors (dim: 21).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTDT(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTDT_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 21 named vector
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractCTDC">extractCTDC</a></code> and <code><a href="#topic+extractCTDD">extractCTDD</a></code>
for Composition and Distribution of the CTD descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractCTDT(x)
</code></pre>

<hr>
<h2 id='extractCTDTClass'>CTD Descriptors - Transition (with customized amino acid
classification support)</h2><span id='topic+extractCTDTClass'></span>

<h3>Description</h3>

<p>This function calculates the Transition descriptor of the
CTD descriptors, with customized amino acid classification support.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTDTClass(x, aagroup1, aagroup2, aagroup3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTDTClass_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractCTDTClass_+3A_aagroup1">aagroup1</code></td>
<td>
<p>A named list which contains the first group of customized
amino acid classification. See example below.</p>
</td></tr>
<tr><td><code id="extractCTDTClass_+3A_aagroup2">aagroup2</code></td>
<td>
<p>A named list which contains the second group of customized
amino acid classification. See example below.</p>
</td></tr>
<tr><td><code id="extractCTDTClass_+3A_aagroup3">aagroup3</code></td>
<td>
<p>A named list which contains the third group of customized
amino acid classification. See example below.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>k * 3</code> named vector, <code>k</code> is the number of
amino acid properties used.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractCTDCClass">extractCTDCClass</a></code> and
<code><a href="#topic+extractCTDDClass">extractCTDDClass</a></code> for Composition and Distribution of
the CTD descriptors with customized amino acid classification support.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]

# using five customized amino acid property classification
group1 &lt;- list(
  "hydrophobicity" = c("R", "K", "E", "D", "Q", "N"),
  "normwaalsvolume" = c("G", "A", "S", "T", "P", "D", "C"),
  "polarizability" = c("G", "A", "S", "D", "T"),
  "secondarystruct" = c("E", "A", "L", "M", "Q", "K", "R", "H"),
  "solventaccess" = c("A", "L", "F", "C", "G", "I", "V", "W")
)

group2 &lt;- list(
  "hydrophobicity" = c("G", "A", "S", "T", "P", "H", "Y"),
  "normwaalsvolume" = c("N", "V", "E", "Q", "I", "L"),
  "polarizability" = c("C", "P", "N", "V", "E", "Q", "I", "L"),
  "secondarystruct" = c("V", "I", "Y", "C", "W", "F", "T"),
  "solventaccess" = c("R", "K", "Q", "E", "N", "D")
)

group3 &lt;- list(
  "hydrophobicity" = c("C", "L", "V", "I", "M", "F", "W"),
  "normwaalsvolume" = c("M", "H", "K", "F", "R", "Y", "W"),
  "polarizability" = c("K", "M", "H", "F", "R", "Y", "W"),
  "secondarystruct" = c("G", "N", "P", "S", "D"),
  "solventaccess" = c("M", "S", "P", "T", "H", "Y")
)

extractCTDTClass(x, aagroup1 = group1, aagroup2 = group2, aagroup3 = group3)
</code></pre>

<hr>
<h2 id='extractCTriad'>Conjoint Triad Descriptor</h2><span id='topic+extractCTriad'></span>

<h3>Description</h3>

<p>This function calculates the Conjoint Triad descriptor (dim: 343).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTriad(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTriad_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 343 named vector
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>J.W. Shen, J. Zhang, X.M. Luo, W.L. Zhu,
K.Q. Yu, K.X. Chen, Y.X. Li, H.L. Jiang.
Predicting Protein-protein Interactions Based Only on Sequences Information.
<em>Proceedings of the National Academy of Sciences</em>. 007, 104, 4337&ndash;4341.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractCTriad(x)
</code></pre>

<hr>
<h2 id='extractCTriadClass'>Conjoint Triad Descriptor (with customized amino acid
classification support)</h2><span id='topic+extractCTriadClass'></span>

<h3>Description</h3>

<p>This function calculates the Conjoint Triad descriptor, with customized
amino acid classification support.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCTriadClass(x, aaclass)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractCTriadClass_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractCTriadClass_+3A_aaclass">aaclass</code></td>
<td>
<p>A list containing the customized amino acid
classification. See example below.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>k^3</code> named vector, where <code>k</code> is the number
of customized classes of the amino acids.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>J.W. Shen, J. Zhang, X.M. Luo, W.L. Zhu,
K.Q. Yu, K.X. Chen, Y.X. Li, H.L. Jiang.
Predicting Protein-protein Interactions Based Only on Sequences Information.
<em>Proceedings of the National Academy of Sciences</em>. 007, 104, 4337&ndash;4341.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]

# use customized amino acid classification (normalized van der Waals volume)
newclass &lt;- list(
  c("G", "A", "S", "T", "P", "D", "C"),
  c("N", "V", "E", "Q", "I", "L"),
  c("M", "H", "K", "F", "R", "Y", "W")
)

extractCTriadClass(x, aaclass = newclass)
</code></pre>

<hr>
<h2 id='extractDC'>Dipeptide Composition Descriptor</h2><span id='topic+extractDC'></span>

<h3>Description</h3>

<p>This function calculates the Dipeptide Composition descriptor (dim: 400).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractDC(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractDC_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 400 named vector
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>M. Bhasin, G. P. S. Raghava.
Classification of Nuclear Receptors Based on
Amino Acid Composition and Dipeptide Composition.
<em>Journal of Biological Chemistry</em>, 2004, 279, 23262.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractAAC">extractAAC</a></code> and <code><a href="#topic+extractTC">extractTC</a></code>
for Amino Acid Composition and Tripeptide Composition descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractDC(x)
</code></pre>

<hr>
<h2 id='extractDescScales'>Scales-Based Descriptors with 20+ classes of Molecular Descriptors</h2><span id='topic+extractDescScales'></span>

<h3>Description</h3>

<p>This function calculates the scales-based descriptors with
molecular descriptors sets calculated by Dragon, Discovery Studio and MOE.
Users can specify which molecular descriptors to select from one of these
deseriptor sets by specify the numerical or character index of the
molecular descriptors in the descriptor set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractDescScales(
  x,
  propmat,
  index = NULL,
  pc,
  lag,
  scale = TRUE,
  silent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractDescScales_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractDescScales_+3A_propmat">propmat</code></td>
<td>
<p>The matrix containing the descriptor set for the amino acids,
which can be chosen from
<code>AAMOE2D</code>, <code>AAMOE3D</code>, <code>AACPSA</code>, <code>AADescAll</code>,
<code>AA2DACOR</code>, <code>AA3DMoRSE</code>, <code>AAACF</code>, <code>AABurden</code>,
<code>AAConn</code>, <code>AAConst</code>, <code>AAEdgeAdj</code>, <code>AAEigIdx</code>,
<code>AAFGC</code>, <code>AAGeom</code>, <code>AAGETAWAY</code>, <code>AAInfo</code>,
<code>AAMolProp</code>, <code>AARandic</code>, <code>AARDF</code>, <code>AATopo</code>,
<code>AATopoChg</code>, <code>AAWalk</code>, and <code>AAWHIM</code>.</p>
</td></tr>
<tr><td><code id="extractDescScales_+3A_index">index</code></td>
<td>
<p>Integer vector or character vector. Specify which
molecular descriptors to select from one of these deseriptor
sets by specify the numerical or character index of the molecular
descriptors in the descriptor set.
Default is <code>NULL</code>, which means selecting all the molecular
descriptors in this descriptor set.</p>
</td></tr>
<tr><td><code id="extractDescScales_+3A_pc">pc</code></td>
<td>
<p>Integer. The maximum dimension of the space which the data
are to be represented in. Must be no greater than the number of
amino acid properties provided.</p>
</td></tr>
<tr><td><code id="extractDescScales_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractDescScales_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing MDS? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractDescScales_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
descscales &lt;- extractDescScales(
  x,
  propmat = "AATopo", index = c(37:41, 43:47),
  pc = 5, lag = 7, silent = FALSE
)
</code></pre>

<hr>
<h2 id='extractFAScales'>Scales-Based Descriptors derived by Factor Analysis</h2><span id='topic+extractFAScales'></span>

<h3>Description</h3>

<p>This function calculates scales-based descriptors
derived by Factor Analysis (FA).
Users can provide customized amino acid property matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractFAScales(
  x,
  propmat,
  factors,
  scores = "regression",
  lag,
  scale = TRUE,
  silent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractFAScales_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractFAScales_+3A_propmat">propmat</code></td>
<td>
<p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents one property.
Note that the one-letter row names must be provided for we need them to seek
the properties for each AA type.</p>
</td></tr>
<tr><td><code id="extractFAScales_+3A_factors">factors</code></td>
<td>
<p>Integer. The number of factors to be fitted.
Must be no greater than the number of AA properties provided.</p>
</td></tr>
<tr><td><code id="extractFAScales_+3A_scores">scores</code></td>
<td>
<p>Type of scores to produce. The default is <code>"regression"</code>,
which gives Thompson's scores, <code>"Bartlett"</code> given Bartlett's weighted
least-squares scores.</p>
</td></tr>
<tr><td><code id="extractFAScales_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids number
in the protein sequence.</p>
</td></tr>
<tr><td><code id="extractFAScales_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing Factor Analysis? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractFAScales_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether we print the SS loadings,
proportion of variance and the cumulative proportion of
the selected factors or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (factors) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Atchley, W. R., Zhao, J., Fernandes, A. D., &amp; Druke, T. (2005).
Solving the protein sequence metric problem.
Proceedings of the National Academy of Sciences of the United States of America,
102(18), 6395-6400.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
data(AATopo)
tprops &lt;- AATopo[, c(37:41, 43:47)] # select a set of topological descriptors
fa &lt;- extractFAScales(x, propmat = tprops, factors = 5, lag = 7, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractGeary'>Geary Autocorrelation Descriptor</h2><span id='topic+extractGeary'></span>

<h3>Description</h3>

<p>This function calculates the Geary
autocorrelation descriptor (dim: <code>length(props) * nlag</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractGeary(
  x,
  props = c("CIDH920105", "BHAR880101", "CHAM820101", "CHAM820102", "CHOC760101",
    "BIGC670101", "CHAM810101", "DAYM780201"),
  nlag = 30L,
  customprops = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractGeary_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractGeary_+3A_props">props</code></td>
<td>
<p>A character vector, specifying the
Accession Number of the target properties.
8 properties are used by default, as listed below:
</p>

<dl>
<dt>AccNo. CIDH920105</dt><dd><p>Normalized average hydrophobicity scales (Cid et al., 1992)</p>
</dd>
<dt>AccNo. BHAR880101</dt><dd><p>Average flexibility indices (Bhaskaran-Ponnuswamy, 1988)</p>
</dd>
<dt>AccNo. CHAM820101</dt><dd><p>Polarizability parameter (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHAM820102</dt><dd><p>Free energy of solution in water, kcal/mole (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHOC760101</dt><dd><p>Residue accessible surface area in tripeptide (Chothia, 1976)</p>
</dd>
<dt>AccNo. BIGC670101</dt><dd><p>Residue volume (Bigelow, 1967)</p>
</dd>
<dt>AccNo. CHAM810101</dt><dd><p>Steric parameter (Charton, 1981)</p>
</dd>
<dt>AccNo. DAYM780201</dt><dd><p>Relative mutability (Dayhoff et al., 1978b)</p>
</dd></dl>
</td></tr>
<tr><td><code id="extractGeary_+3A_nlag">nlag</code></td>
<td>
<p>Maximum value of the lag parameter. Default is <code>30</code>.</p>
</td></tr>
<tr><td><code id="extractGeary_+3A_customprops">customprops</code></td>
<td>
<p>A <code>n x 21</code> named data frame contains <code>n</code>
customized property. Each row contains one property.
The column order for different amino acid types is
<code>'AccNo'</code>, <code>'A'</code>, <code>'R'</code>, <code>'N'</code>,
<code>'D'</code>, <code>'C'</code>, <code>'E'</code>, <code>'Q'</code>,
<code>'G'</code>, <code>'H'</code>, <code>'I'</code>, <code>'L'</code>,
<code>'K'</code>, <code>'M'</code>, <code>'F'</code>, <code>'P'</code>,
<code>'S'</code>, <code>'T'</code>, <code>'W'</code>, <code>'Y'</code>,
<code>'V'</code>, and the columns should also be <em>exactly</em> named like this.
The <code>AccNo</code> column contains the properties' names.
Then users should explicitly specify these properties
with these names in the argument <code>props</code>.
See the examples below for a demonstration.
The default value for <code>customprops</code> is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>length(props) * nlag</code> named vector.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>AAindex: Amino acid index database.
<a href="https://www.genome.jp/dbget/aaindex.html">https://www.genome.jp/dbget/aaindex.html</a>
</p>
<p>Feng, Z.P. and Zhang, C.T. (2000)
Prediction of membrane protein types based on the hydrophobic
index of amino acids.
<em>Journal of Protein Chemistry</em>, 19, 269-275.
</p>
<p>Horne, D.S. (1988)
Prediction of protein helix content from
an autocorrelation analysis of sequence hydrophobicities.
<em>Biopolymers</em>, 27, 451-477.
</p>
<p>Sokal, R.R. and Thomson, B.A. (2006)
Population structure inferred by local spatial autocorrelation:
an usage from an Amerindian tribal population.
<em>American Journal of Physical Anthropology</em>, 129, 121-131.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractMoreauBroto">extractMoreauBroto</a></code> and <code><a href="#topic+extractMoran">extractMoran</a></code>
for Moreau-Broto autocorrelation descriptors and
Moran autocorrelation descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractGeary(x)

myprops &lt;- data.frame(
  AccNo = c("MyProp1", "MyProp2", "MyProp3"),
  A = c(0.62, -0.5, 15), R = c(-2.53, 3, 101),
  N = c(-0.78, 0.2, 58), D = c(-0.9, 3, 59),
  C = c(0.29, -1, 47), E = c(-0.74, 3, 73),
  Q = c(-0.85, 0.2, 72), G = c(0.48, 0, 1),
  H = c(-0.4, -0.5, 82), I = c(1.38, -1.8, 57),
  L = c(1.06, -1.8, 57), K = c(-1.5, 3, 73),
  M = c(0.64, -1.3, 75), F = c(1.19, -2.5, 91),
  P = c(0.12, 0, 42), S = c(-0.18, 0.3, 31),
  T = c(-0.05, -0.4, 45), W = c(0.81, -3.4, 130),
  Y = c(0.26, -2.3, 107), V = c(1.08, -1.5, 43)
)

# Use 4 properties in the AAindex database, and 3 cutomized properties
extractGeary(
  x,
  customprops = myprops,
  props = c(
    "CIDH920105", "BHAR880101",
    "CHAM820101", "CHAM820102",
    "MyProp1", "MyProp2", "MyProp3"
  )
)
</code></pre>

<hr>
<h2 id='extractMDSScales'>Scales-Based Descriptors derived by Multidimensional Scaling</h2><span id='topic+extractMDSScales'></span>

<h3>Description</h3>

<p>This function calculates scales-based descriptors
derived by Multidimensional Scaling (MDS).
Users can provide customized amino acid property matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractMDSScales(x, propmat, k, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractMDSScales_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractMDSScales_+3A_propmat">propmat</code></td>
<td>
<p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents one property.
Note that the one-letter row names must be provided for we need them to seek
the properties for each AA type.</p>
</td></tr>
<tr><td><code id="extractMDSScales_+3A_k">k</code></td>
<td>
<p>Integer. The maximum dimension of the space which the data
are to be represented in. Must be no greater than the number of AA
properties provided.</p>
</td></tr>
<tr><td><code id="extractMDSScales_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractMDSScales_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing MDS? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractMDSScales_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether to print the <code>k</code> eigenvalues
computed during the scaling process or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (dimensionality) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Venkatarajan, M. S., &amp; Braun, W. (2001).
New quantitative descriptors of amino acids based on multidimensional scaling
of a large number of physical-chemical properties.
Molecular modeling annual, 7(12), 445&ndash;453.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractScales">extractScales</a></code> for scales-based
descriptors derived by Principal Components Analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
data(AATopo)
tprops &lt;- AATopo[, c(37:41, 43:47)] # select a set of topological descriptors
mds &lt;- extractMDSScales(x, propmat = tprops, k = 5, lag = 7, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractMoran'>Moran Autocorrelation Descriptor</h2><span id='topic+extractMoran'></span>

<h3>Description</h3>

<p>This function calculates the Moran
autocorrelation descriptor (dim: <code>length(props) * nlag</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractMoran(
  x,
  props = c("CIDH920105", "BHAR880101", "CHAM820101", "CHAM820102", "CHOC760101",
    "BIGC670101", "CHAM810101", "DAYM780201"),
  nlag = 30L,
  customprops = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractMoran_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractMoran_+3A_props">props</code></td>
<td>
<p>A character vector, specifying the
Accession Number of the target properties.
8 properties are used by default, as listed below:
</p>

<dl>
<dt>AccNo. CIDH920105</dt><dd><p>Normalized average hydrophobicity scales (Cid et al., 1992)</p>
</dd>
<dt>AccNo. BHAR880101</dt><dd><p>Average flexibility indices (Bhaskaran-Ponnuswamy, 1988)</p>
</dd>
<dt>AccNo. CHAM820101</dt><dd><p>Polarizability parameter (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHAM820102</dt><dd><p>Free energy of solution in water, kcal/mole (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHOC760101</dt><dd><p>Residue accessible surface area in tripeptide (Chothia, 1976)</p>
</dd>
<dt>AccNo. BIGC670101</dt><dd><p>Residue volume (Bigelow, 1967)</p>
</dd>
<dt>AccNo. CHAM810101</dt><dd><p>Steric parameter (Charton, 1981)</p>
</dd>
<dt>AccNo. DAYM780201</dt><dd><p>Relative mutability (Dayhoff et al., 1978b)</p>
</dd></dl>
</td></tr>
<tr><td><code id="extractMoran_+3A_nlag">nlag</code></td>
<td>
<p>Maximum value of the lag parameter. Default is <code>30</code>.</p>
</td></tr>
<tr><td><code id="extractMoran_+3A_customprops">customprops</code></td>
<td>
<p>A <code>n x 21</code> named data frame contains <code>n</code>
customized property. Each row contains one property.
The column order for different amino acid types is
<code>'AccNo'</code>, <code>'A'</code>, <code>'R'</code>, <code>'N'</code>,
<code>'D'</code>, <code>'C'</code>, <code>'E'</code>, <code>'Q'</code>,
<code>'G'</code>, <code>'H'</code>, <code>'I'</code>, <code>'L'</code>,
<code>'K'</code>, <code>'M'</code>, <code>'F'</code>, <code>'P'</code>,
<code>'S'</code>, <code>'T'</code>, <code>'W'</code>, <code>'Y'</code>,
<code>'V'</code>, and the columns should also be <em>exactly</em> named like this.
The <code>AccNo</code> column contains the properties' names.
Then users should explicitly specify these properties
with these names in the argument <code>props</code>.
See the examples below for a demonstration.
The default value for <code>customprops</code> is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>length(props) * nlag</code> named vector.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>AAindex: Amino acid index database.
<a href="https://www.genome.jp/dbget/aaindex.html">https://www.genome.jp/dbget/aaindex.html</a>
</p>
<p>Feng, Z.P. and Zhang, C.T. (2000)
Prediction of membrane protein types based on the hydrophobic
index of amino acids.
<em>Journal of Protein Chemistry</em>, 19, 269-275.
</p>
<p>Horne, D.S. (1988)
Prediction of protein helix content from
an autocorrelation analysis of sequence hydrophobicities.
<em>Biopolymers</em>, 27, 451-477.
</p>
<p>Sokal, R.R. and Thomson, B.A. (2006)
Population structure inferred by local spatial autocorrelation:
an usage from an Amerindian tribal population.
<em>American Journal of Physical Anthropology</em>, 129, 121-131.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractMoreauBroto">extractMoreauBroto</a></code> and <code><a href="#topic+extractGeary">extractGeary</a></code>
for Moreau-Broto autocorrelation descriptors and
Geary autocorrelation descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractMoran(x)

myprops &lt;- data.frame(
  AccNo = c("MyProp1", "MyProp2", "MyProp3"),
  A = c(0.62, -0.5, 15), R = c(-2.53, 3, 101),
  N = c(-0.78, 0.2, 58), D = c(-0.9, 3, 59),
  C = c(0.29, -1, 47), E = c(-0.74, 3, 73),
  Q = c(-0.85, 0.2, 72), G = c(0.48, 0, 1),
  H = c(-0.4, -0.5, 82), I = c(1.38, -1.8, 57),
  L = c(1.06, -1.8, 57), K = c(-1.5, 3, 73),
  M = c(0.64, -1.3, 75), F = c(1.19, -2.5, 91),
  P = c(0.12, 0, 42), S = c(-0.18, 0.3, 31),
  T = c(-0.05, -0.4, 45), W = c(0.81, -3.4, 130),
  Y = c(0.26, -2.3, 107), V = c(1.08, -1.5, 43)
)

# Use 4 properties in the AAindex database, and 3 cutomized properties
extractMoran(
  x,
  customprops = myprops,
  props = c(
    "CIDH920105", "BHAR880101",
    "CHAM820101", "CHAM820102",
    "MyProp1", "MyProp2", "MyProp3"
  )
)
</code></pre>

<hr>
<h2 id='extractMoreauBroto'>Normalized Moreau-Broto Autocorrelation Descriptor</h2><span id='topic+extractMoreauBroto'></span>

<h3>Description</h3>

<p>This function calculates the normalized Moreau-Broto
autocorrelation descriptor (dim: <code>length(props) * nlag</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractMoreauBroto(
  x,
  props = c("CIDH920105", "BHAR880101", "CHAM820101", "CHAM820102", "CHOC760101",
    "BIGC670101", "CHAM810101", "DAYM780201"),
  nlag = 30L,
  customprops = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractMoreauBroto_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractMoreauBroto_+3A_props">props</code></td>
<td>
<p>A character vector, specifying the
Accession Number of the target properties.
8 properties are used by default, as listed below:
</p>

<dl>
<dt>AccNo. CIDH920105</dt><dd><p>Normalized average hydrophobicity scales (Cid et al., 1992)</p>
</dd>
<dt>AccNo. BHAR880101</dt><dd><p>Average flexibility indices (Bhaskaran-Ponnuswamy, 1988)</p>
</dd>
<dt>AccNo. CHAM820101</dt><dd><p>Polarizability parameter (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHAM820102</dt><dd><p>Free energy of solution in water, kcal/mole (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHOC760101</dt><dd><p>Residue accessible surface area in tripeptide (Chothia, 1976)</p>
</dd>
<dt>AccNo. BIGC670101</dt><dd><p>Residue volume (Bigelow, 1967)</p>
</dd>
<dt>AccNo. CHAM810101</dt><dd><p>Steric parameter (Charton, 1981)</p>
</dd>
<dt>AccNo. DAYM780201</dt><dd><p>Relative mutability (Dayhoff et al., 1978b)</p>
</dd></dl>
</td></tr>
<tr><td><code id="extractMoreauBroto_+3A_nlag">nlag</code></td>
<td>
<p>Maximum value of the lag parameter. Default is <code>30</code>.</p>
</td></tr>
<tr><td><code id="extractMoreauBroto_+3A_customprops">customprops</code></td>
<td>
<p>A <code>n x 21</code> named data frame contains <code>n</code>
customized property. Each row contains one property.
The column order for different amino acid types is
<code>'AccNo'</code>, <code>'A'</code>, <code>'R'</code>, <code>'N'</code>,
<code>'D'</code>, <code>'C'</code>, <code>'E'</code>, <code>'Q'</code>,
<code>'G'</code>, <code>'H'</code>, <code>'I'</code>, <code>'L'</code>,
<code>'K'</code>, <code>'M'</code>, <code>'F'</code>, <code>'P'</code>,
<code>'S'</code>, <code>'T'</code>, <code>'W'</code>, <code>'Y'</code>,
<code>'V'</code>, and the columns should also be <em>exactly</em> named like this.
The <code>AccNo</code> column contains the properties' names.
Then users should explicitly specify these properties
with these names in the argument <code>props</code>.
See the examples below for a demonstration.
The default value for <code>customprops</code> is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>length(props) * nlag</code> named vector.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>AAindex: Amino acid index database.
<a href="https://www.genome.jp/dbget/aaindex.html">https://www.genome.jp/dbget/aaindex.html</a>
</p>
<p>Feng, Z.P. and Zhang, C.T. (2000)
Prediction of membrane protein types based on the hydrophobic
index of amino acids.
<em>Journal of Protein Chemistry</em>, 19, 269-275.
</p>
<p>Horne, D.S. (1988)
Prediction of protein helix content from
an autocorrelation analysis of sequence hydrophobicities.
<em>Biopolymers</em>, 27, 451-477.
</p>
<p>Sokal, R.R. and Thomson, B.A. (2006)
Population structure inferred by local spatial autocorrelation:
an usage from an Amerindian tribal population.
<em>American Journal of Physical Anthropology</em>, 129, 121-131.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractMoran">extractMoran</a></code> and <code><a href="#topic+extractGeary">extractGeary</a></code>
for Moran autocorrelation descriptors and Geary autocorrelation descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractMoreauBroto(x)

myprops &lt;- data.frame(
  AccNo = c("MyProp1", "MyProp2", "MyProp3"),
  A = c(0.62, -0.5, 15), R = c(-2.53, 3, 101),
  N = c(-0.78, 0.2, 58), D = c(-0.9, 3, 59),
  C = c(0.29, -1, 47), E = c(-0.74, 3, 73),
  Q = c(-0.85, 0.2, 72), G = c(0.48, 0, 1),
  H = c(-0.4, -0.5, 82), I = c(1.38, -1.8, 57),
  L = c(1.06, -1.8, 57), K = c(-1.5, 3, 73),
  M = c(0.64, -1.3, 75), F = c(1.19, -2.5, 91),
  P = c(0.12, 0, 42), S = c(-0.18, 0.3, 31),
  T = c(-0.05, -0.4, 45), W = c(0.81, -3.4, 130),
  Y = c(0.26, -2.3, 107), V = c(1.08, -1.5, 43)
)

# Use 4 properties in the AAindex database, and 3 cutomized properties
extractMoreauBroto(
  x,
  customprops = myprops,
  props = c(
    "CIDH920105", "BHAR880101",
    "CHAM820101", "CHAM820102",
    "MyProp1", "MyProp2", "MyProp3"
  )
)
</code></pre>

<hr>
<h2 id='extractPAAC'>Pseudo Amino Acid Composition (PseAAC) Descriptor</h2><span id='topic+extractPAAC'></span>

<h3>Description</h3>

<p>This function calculates the Pseudo Amino Acid Composition (PseAAC)
descriptor (dim: <code>20 + lambda</code>, default is 50).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractPAAC(
  x,
  props = c("Hydrophobicity", "Hydrophilicity", "SideChainMass"),
  lambda = 30,
  w = 0.05,
  customprops = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractPAAC_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractPAAC_+3A_props">props</code></td>
<td>
<p>A character vector, specifying the properties used.
3 properties are used by default, as listed below:
</p>

<dl>
<dt><code>'Hydrophobicity'</code></dt><dd><p>Hydrophobicity value of the 20 amino acids</p>
</dd>
<dt><code>'Hydrophilicity'</code></dt><dd><p>Hydrophilicity value of the 20 amino acids</p>
</dd>
<dt><code>'SideChainMass'</code></dt><dd><p>Side-chain mass of the 20 amino acids</p>
</dd></dl>
</td></tr>
<tr><td><code id="extractPAAC_+3A_lambda">lambda</code></td>
<td>
<p>The lambda parameter for the PseAAC descriptors, default is 30.</p>
</td></tr>
<tr><td><code id="extractPAAC_+3A_w">w</code></td>
<td>
<p>The weighting factor, default is 0.05.</p>
</td></tr>
<tr><td><code id="extractPAAC_+3A_customprops">customprops</code></td>
<td>
<p>A <code>n x 21</code> named data frame contains <code>n</code>
customized property. Each row contains one property.
The column order for different amino acid types is
<code>'AccNo'</code>, <code>'A'</code>, <code>'R'</code>, <code>'N'</code>,
<code>'D'</code>, <code>'C'</code>, <code>'E'</code>, <code>'Q'</code>,
<code>'G'</code>, <code>'H'</code>, <code>'I'</code>, <code>'L'</code>,
<code>'K'</code>, <code>'M'</code>, <code>'F'</code>, <code>'P'</code>,
<code>'S'</code>, <code>'T'</code>, <code>'W'</code>, <code>'Y'</code>,
<code>'V'</code>, and the columns should also be <em>exactly</em> named like this.
The <code>AccNo</code> column contains the properties' names.
Then users should explicitly specify these properties
with these names in the argument <code>props</code>.
See the examples below for a demonstration.
The default value for <code>customprops</code> is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>20 + lambda</code> named vector
</p>


<h3>Note</h3>

<p>Note the default <code>20 * 3</code> <code>prop</code> values have already been
independently given in the function. Users can also specify
other (up to 544) properties with the Accession Number in
the <code><a href="#topic+AAindex">AAindex</a></code> data, with or without the default
three properties, which means users should explicitly specify
the properties to use. For this descriptor type, users need to
intelligently evaluate the underlying details of the descriptors
provided, instead of using this function with their data blindly.
It would be wise to use some negative and positive control comparisons
where relevant to help guide interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Kuo-Chen Chou. Prediction of Protein Cellular Attributes
Using Pseudo-Amino Acid Composition.
<em>PROTEINS: Structure, Function, and Genetics</em>, 2001, 43: 246-255.
</p>
<p>Kuo-Chen Chou. Using Amphiphilic Pseudo Amino Acid Composition
to Predict Enzyme Subfamily Classes. <em>Bioinformatics</em>, 2005, 21, 10-19.
</p>
<p>JACS, 1962, 84: 4240-4246. (C. Tanford). (The hydrophobicity data)
</p>
<p>PNAS, 1981, 78:3824-3828 (T.P.Hopp &amp; K.R.Woods). (The hydrophilicity data)
</p>
<p>CRC Handbook of Chemistry and Physics, 66th ed.,
CRC Press, Boca Raton, Florida (1985). (The side-chain mass data)
</p>
<p>R.M.C. Dawson, D.C. Elliott, W.H. Elliott, K.M. Jones,
Data for Biochemical Research 3rd ed., Clarendon Press Oxford (1986).
(The side-chain mass data)
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractAPAAC">extractAPAAC</a></code> for amphiphilic pseudo
amino acid composition descriptor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractPAAC(x)

myprops &lt;- data.frame(
  AccNo = c("MyProp1", "MyProp2", "MyProp3"),
  A = c(0.62, -0.5, 15), R = c(-2.53, 3, 101),
  N = c(-0.78, 0.2, 58), D = c(-0.9, 3, 59),
  C = c(0.29, -1, 47), E = c(-0.74, 3, 73),
  Q = c(-0.85, 0.2, 72), G = c(0.48, 0, 1),
  H = c(-0.4, -0.5, 82), I = c(1.38, -1.8, 57),
  L = c(1.06, -1.8, 57), K = c(-1.5, 3, 73),
  M = c(0.64, -1.3, 75), F = c(1.19, -2.5, 91),
  P = c(0.12, 0, 42), S = c(-0.18, 0.3, 31),
  T = c(-0.05, -0.4, 45), W = c(0.81, -3.4, 130),
  Y = c(0.26, -2.3, 107), V = c(1.08, -1.5, 43)
)

# use 3 default properties, 4 properties from the
# AAindex database, and 3 cutomized properties
extractPAAC(
  x,
  customprops = myprops,
  props = c(
    "Hydrophobicity", "Hydrophilicity", "SideChainMass",
    "CIDH920105", "BHAR880101",
    "CHAM820101", "CHAM820102",
    "MyProp1", "MyProp2", "MyProp3"
  )
)
</code></pre>

<hr>
<h2 id='extractProtFP'>Amino Acid Properties Based Scales Descriptors (Protein Fingerprint)</h2><span id='topic+extractProtFP'></span>

<h3>Description</h3>

<p>This function calculates amino acid properties based scales descriptors
(protein fingerprint). Users can specify which AAindex properties to
select from the AAindex database by specify the numerical or character
index of the properties in the AAindex database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractProtFP(x, index = NULL, pc, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractProtFP_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractProtFP_+3A_index">index</code></td>
<td>
<p>Integer vector or character vector. Specify which AAindex
properties to select from the AAindex database by specify the
numerical or character index of the properties in the AAindex database.
Default is <code>NULL</code>, means selecting all the AA properties
in the AAindex database.</p>
</td></tr>
<tr><td><code id="extractProtFP_+3A_pc">pc</code></td>
<td>
<p>Integer. Use the first pc principal components as the scales.
Must be no greater than the number of AA properties provided.</p>
</td></tr>
<tr><td><code id="extractProtFP_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractProtFP_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
before PCA? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractProtFP_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (principal components) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
fp &lt;- extractProtFP(x, index = c(160:165, 258:296), pc = 5, lag = 7, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractProtFPGap'>Amino Acid Properties Based Scales Descriptors (Protein Fingerprint)
with Gap Support</h2><span id='topic+extractProtFPGap'></span>

<h3>Description</h3>

<p>This function calculates amino acid properties based scales descriptors
(protein fingerprint) with gap support. Users can specify which AAindex
properties to select from the AAindex database by specify the numerical or
character index of the properties in the AAindex database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractProtFPGap(x, index = NULL, pc, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractProtFPGap_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.
Use '<code>-</code>' to represent gaps in the sequence.</p>
</td></tr>
<tr><td><code id="extractProtFPGap_+3A_index">index</code></td>
<td>
<p>Integer vector or character vector. Specify which AAindex
properties to select from the AAindex database by specify the
numerical or character index of the properties in the AAindex database.
Default is <code>NULL</code>, means selecting all the AA properties
in the AAindex database.</p>
</td></tr>
<tr><td><code id="extractProtFPGap_+3A_pc">pc</code></td>
<td>
<p>Integer. Use the first pc principal components as the scales.
Must be no greater than the number of AA properties provided.</p>
</td></tr>
<tr><td><code id="extractProtFPGap_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractProtFPGap_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
before PCA? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractProtFPGap_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (principal components) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># amino acid sequence with gaps
x &lt;- readFASTA(system.file("protseq/align.fasta", package = "protr"))$`IXI_235`
fp &lt;- extractProtFPGap(x, index = c(160:165, 258:296), pc = 5, lag = 7, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractPSSM'>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence</h2><span id='topic+extractPSSM'></span>

<h3>Description</h3>

<p>This function calculates the PSSM (Position-Specific Scoring Matrix) derived
by PSI-Blast for given protein sequence or peptides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractPSSM(
  seq,
  start.pos = 1L,
  end.pos = nchar(seq),
  psiblast.path = NULL,
  makeblastdb.path = NULL,
  database.path = NULL,
  iter = 5,
  silent = TRUE,
  evalue = 10L,
  word.size = NULL,
  gapopen = NULL,
  gapextend = NULL,
  matrix = "BLOSUM62",
  threshold = NULL,
  seg = "no",
  soft.masking = FALSE,
  culling.limit = NULL,
  best.hit.overhang = NULL,
  best.hit.score.edge = NULL,
  xdrop.ungap = NULL,
  xdrop.gap = NULL,
  xdrop.gap.final = NULL,
  window.size = NULL,
  gap.trigger = 22L,
  num.threads = 1L,
  pseudocount = 0L,
  inclusion.ethresh = 0.002
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractPSSM_+3A_seq">seq</code></td>
<td>
<p>Character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_start.pos">start.pos</code></td>
<td>
<p>Optional integer denoting the start position of the
fragment window. Default is <code>1</code>,
i.e. the first amino acid of the given sequence.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_end.pos">end.pos</code></td>
<td>
<p>Optional integer denoting the end position of the
fragment window. Default is <code>nchar(seq)</code>,
i.e. the last amino acid of the given sequence.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_psiblast.path">psiblast.path</code></td>
<td>
<p>Character string indicating the path of the
<code>psiblast</code> program.
If NCBI Blast+ was previously installed in the operation system,
the path will be automatically detected.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_makeblastdb.path">makeblastdb.path</code></td>
<td>
<p>Character string indicating the path of the
<code>makeblastdb</code> program.
If NCBI Blast+ was previously installed in the system,
the path will be automatically detected.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_database.path">database.path</code></td>
<td>
<p>Character string indicating the path of
a reference database (a FASTA file).</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_iter">iter</code></td>
<td>
<p>Number of iterations to perform for PSI-Blast.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether the PSI-Blast running output
should be shown or not (May not work on some Windows versions and
PSI-Blast versions), default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_evalue">evalue</code></td>
<td>
<p>Expectation value (E) threshold for saving hits.
Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_word.size">word.size</code></td>
<td>
<p>Word size for wordfinder algorithm. An integer &gt;= 2.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_gapopen">gapopen</code></td>
<td>
<p>Integer. Cost to open a gap.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_gapextend">gapextend</code></td>
<td>
<p>Integer. Cost to extend a gap.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_matrix">matrix</code></td>
<td>
<p>Character string. The scoring matrix name
(default is <code>"BLOSUM62"</code>).</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_threshold">threshold</code></td>
<td>
<p>Minimum word score such that the word is added to
the BLAST lookup table. A real value &gt;= 0.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_seg">seg</code></td>
<td>
<p>Character string. Filter query sequence with SEG (<code>"yes"</code>,
<code>"window locut hicut"</code>, or <code>"no"</code> to disable).
Default is <code>"no"</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_soft.masking">soft.masking</code></td>
<td>
<p>Logical. Apply filtering locations as soft masks?
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_culling.limit">culling.limit</code></td>
<td>
<p>An integer &gt;= 0. If the query range of a hit is
enveloped by that of at least this many higher-scoring hits,
delete the hit. Incompatible with <code>best.hit.overhang</code> and
<code>best_hit_score_edge</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_best.hit.overhang">best.hit.overhang</code></td>
<td>
<p>Best Hit algorithm overhang value
(A real value &gt;= 0 and =&lt; 0.5, recommended value: 0.1).
Incompatible with <code>culling_limit</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_best.hit.score.edge">best.hit.score.edge</code></td>
<td>
<p>Best Hit algorithm score edge value
(A real value &gt;=0 and =&lt; 0.5, recommended value: 0.1).
Incompatible with <code>culling_limit</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_xdrop.ungap">xdrop.ungap</code></td>
<td>
<p>X-dropoff value (in bits) for ungapped extensions.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_xdrop.gap">xdrop.gap</code></td>
<td>
<p>X-dropoff value (in bits) for preliminary gapped extensions.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_xdrop.gap.final">xdrop.gap.final</code></td>
<td>
<p>X-dropoff value (in bits) for final gapped alignment.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_window.size">window.size</code></td>
<td>
<p>An integer &gt;= 0. Multiple hits window size,
To specify 1-hit algorithm, use <code>0</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_gap.trigger">gap.trigger</code></td>
<td>
<p>Number of bits to trigger gapping. Default is <code>22</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_num.threads">num.threads</code></td>
<td>
<p>Integer. Number of threads (CPUs) to use in the
BLAST search. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_pseudocount">pseudocount</code></td>
<td>
<p>Integer. Pseudo-count value used when constructing PSSM.
Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="extractPSSM_+3A_inclusion.ethresh">inclusion.ethresh</code></td>
<td>
<p>E-value inclusion threshold for pairwise alignments.
Default is <code>0.002</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For given protein sequences or peptides, PSSM represents the
log-likelihood of the substitution of the 20 types of amino acids at that
position in the sequence. Note that the output value is not normalized.
</p>


<h3>Value</h3>

<p>The original PSSM, a numeric matrix which has
<code>end.pos - start.pos + 1</code> columns and <code>20</code> named rows.
</p>


<h3>Note</h3>

<p>The function requires the <code>makeblastdb</code> and <code>psiblast</code> programs
to be properly installed in the operation system or
their paths provided.
</p>
<p>The two command-line programs are included in the NCBI-BLAST+
software package. To install NCBI Blast+ for your operating system, see
<a href="https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html">https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html</a>
for detailed instructions.
</p>
<p>Ubuntu/Debian users can directly use the command
<code>sudo apt-get install ncbi-blast+</code> to install NCBI Blast+.
For OS X users, download <code>ncbi-blast- ... .dmg</code> then install.
For Windows users, download <code>ncbi-blast- ... .exe</code> then install.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Altschul, Stephen F., et al.
&quot;Gapped BLAST and PSI-BLAST: a new generation of protein database search programs.&quot;
<em>Nucleic acids research</em> 25.17 (1997): 3389&ndash;3402.
</p>
<p>Ye, Xugang, Guoli Wang, and Stephen F. Altschul.
&quot;An assessment of substitution scores for protein profile-profile comparison.&quot;
<em>Bioinformatics</em> 27.24 (2011): 3356&ndash;3363.
</p>
<p>Rangwala, Huzefa, and George Karypis.
&quot;Profile-based direct kernels for remote homology detection and fold recognition.&quot;
<em>Bioinformatics</em> 21.23 (2005): 4239&ndash;4247.
</p>


<h3>See Also</h3>

<p><a href="#topic+extractPSSMFeature">extractPSSMFeature</a> <a href="#topic+extractPSSMAcc">extractPSSMAcc</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (Sys.which("makeblastdb") == "" | Sys.which("psiblast") == "") {
  cat("Cannot find makeblastdb or psiblast. Please install NCBI Blast+ first")
} else {
  x &lt;- readFASTA(system.file(
    "protseq/P00750.fasta",
    package = "protr"
  ))[[1]]
  dbpath &lt;- tempfile("tempdb", fileext = ".fasta")
  invisible(file.copy(from = system.file(
    "protseq/Plasminogen.fasta",
    package = "protr"
  ), to = dbpath))

  pssmmat &lt;- extractPSSM(seq = x, database.path = dbpath)

  dim(pssmmat) # 20 x 562 (P00750: length 562, 20 Amino Acids)
}
</code></pre>

<hr>
<h2 id='extractPSSMAcc'>Profile-based protein representation derived by PSSM
(Position-Specific Scoring Matrix) and auto cross covariance</h2><span id='topic+extractPSSMAcc'></span>

<h3>Description</h3>

<p>This function calculates the feature vector based on the PSSM
by running PSI-Blast and auto cross covariance tranformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractPSSMAcc(pssmmat, lag)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractPSSMAcc_+3A_pssmmat">pssmmat</code></td>
<td>
<p>The PSSM computed by <code><a href="#topic+extractPSSM">extractPSSM</a></code>.</p>
</td></tr>
<tr><td><code id="extractPSSMAcc_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the number of amino acids
in the sequence (i.e. the number of columns in the PSSM matrix).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * 20^2</code> named numeric vector,
the element names are derived by the amino acid name abbreviation
(crossed amino acid name abbreviation) and lag index.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Wold, S., Jonsson, J., Sjorstrom, M., Sandberg,
M., &amp; Rannar, S. (1993).
DNA and peptide sequences and chemical processes multivariately modelled
by principal component analysis and partial least-squares projections
to latent structures.
<em>Analytica chimica acta</em>, 277(2), 239&ndash;253.
</p>


<h3>See Also</h3>

<p><a href="#topic+extractPSSM">extractPSSM</a> <a href="#topic+extractPSSMFeature">extractPSSMFeature</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (Sys.which("makeblastdb") == "" | Sys.which("psiblast") == "") {
  cat("Cannot find makeblastdb or psiblast. Please install NCBI Blast+")
} else {
  x &lt;- readFASTA(system.file(
    "protseq/P00750.fasta",
    package = "protr"
  ))[[1]]
  dbpath &lt;- tempfile("tempdb", fileext = ".fasta")
  invisible(file.copy(from = system.file(
    "protseq/Plasminogen.fasta",
    package = "protr"
  ), to = dbpath))

  pssmmat &lt;- extractPSSM(seq = x, database.path = dbpath)
  pssmacc &lt;- extractPSSMAcc(pssmmat, lag = 3)
  tail(pssmacc)
}
</code></pre>

<hr>
<h2 id='extractPSSMFeature'>Profile-based protein representation derived by PSSM
(Position-Specific Scoring Matrix)</h2><span id='topic+extractPSSMFeature'></span>

<h3>Description</h3>

<p>This function calculates the profile-based protein representation
derived by PSSM. The feature vector is based on the PSSM computed by
<code><a href="#topic+extractPSSM">extractPSSM</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractPSSMFeature(pssmmat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractPSSMFeature_+3A_pssmmat">pssmmat</code></td>
<td>
<p>The PSSM computed by <code><a href="#topic+extractPSSM">extractPSSM</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given sequence, the PSSM feature represents the log-likelihood
of the substitution of the 20 types of amino acids at that position
in the sequence.
</p>
<p>Each PSSM feature value in the vector represents the degree of
conservation of a given amino acid type. The value is normalized
to interval (0, 1) by the transformation 1/(1+e^(-x)).
</p>


<h3>Value</h3>

<p>A numeric vector which has <code>20 x N</code> named elements,
where <code>N</code> is the size of the window (number of rows of the PSSM).
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Ye, Xugang, Guoli Wang, and Stephen F. Altschul.
&quot;An assessment of substitution scores for protein profile-profile comparison.&quot;
<em>Bioinformatics</em> 27.24 (2011): 3356&ndash;3363.
</p>
<p>Rangwala, Huzefa, and George Karypis.
&quot;Profile-based direct kernels for remote homology detection and fold recognition.&quot;
<em>Bioinformatics</em> 21.23 (2005): 4239&ndash;4247.
</p>


<h3>See Also</h3>

<p><a href="#topic+extractPSSM">extractPSSM</a> <a href="#topic+extractPSSMAcc">extractPSSMAcc</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (Sys.which("makeblastdb") == "" | Sys.which("psiblast") == "") {
  cat("Cannot find makeblastdb or psiblast. Please install NCBI Blast+")
} else {
  x &lt;- readFASTA(system.file(
    "protseq/P00750.fasta",
    package = "protr"
  ))[[1]]
  dbpath &lt;- tempfile("tempdb", fileext = ".fasta")
  invisible(file.copy(from = system.file(
    "protseq/Plasminogen.fasta",
    package = "protr"
  ), to = dbpath))

  pssmmat &lt;- extractPSSM(seq = x, database.path = dbpath)
  pssmfeature &lt;- extractPSSMFeature(pssmmat)
  head(pssmfeature)
}
</code></pre>

<hr>
<h2 id='extractQSO'>Quasi-Sequence-Order (QSO) Descriptor</h2><span id='topic+extractQSO'></span>

<h3>Description</h3>

<p>This function calculates the Quasi-Sequence-Order (QSO) descriptor
(dim: <code>20 + 20 + (2 * nlag)</code>, default is 100).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractQSO(x, nlag = 30, w = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractQSO_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractQSO_+3A_nlag">nlag</code></td>
<td>
<p>The maximum lag, defualt is 30.</p>
</td></tr>
<tr><td><code id="extractQSO_+3A_w">w</code></td>
<td>
<p>The weighting factor, default is 0.1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>20 + 20 + (2 * nlag)</code> named vector
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Kuo-Chen Chou. Prediction of Protein Subcellar Locations by
Incorporating Quasi-Sequence-Order Effect.
<em>Biochemical and Biophysical Research Communications</em>,
2000, 278, 477-483.
</p>
<p>Kuo-Chen Chou and Yu-Dong Cai. Prediction of Protein Sucellular Locations by
GO-FunD-PseAA Predictor.
<em>Biochemical and Biophysical Research Communications</em>,
2004, 320, 1236-1239.
</p>
<p>Gisbert Schneider and Paul Wrede. The Rational Design of
Amino Acid Sequences by Artifical Neural Networks and Simulated
Molecular Evolution: Do Novo Design of an Idealized Leader Cleavge Site.
<em>Biophys Journal</em>, 1994, 66, 335-344.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractSOCN">extractSOCN</a></code> for sequence-order-coupling numbers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractQSO(x)
</code></pre>

<hr>
<h2 id='extractScales'>Scales-Based Descriptors derived by Principal Components Analysis</h2><span id='topic+extractScales'></span>

<h3>Description</h3>

<p>This function calculates scales-based descriptors
derived by Principal Components Analysis (PCA).
Users can provide customized amino acid property matrices.
This function implements the core computation procedure needed for
the scales-based descriptors derived by AA-Properties (AAindex)
and scales-based descriptors derived by 20+ classes of 2D and 3D
molecular descriptors (Topological, WHIM, VHSE, etc.) in the protr package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractScales(x, propmat, pc, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractScales_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractScales_+3A_propmat">propmat</code></td>
<td>
<p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents
one property. Note that the one-letter row names must be provided
for we need them to seek the properties for each AA type.</p>
</td></tr>
<tr><td><code id="extractScales_+3A_pc">pc</code></td>
<td>
<p>Integer. Use the first pc principal components as the scales.
Must be no greater than the number of AA properties provided.</p>
</td></tr>
<tr><td><code id="extractScales_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractScales_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before PCA? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractScales_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (principal components) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractDescScales">extractDescScales</a></code> scales descriptors based on
20+ classes of molecular descriptors, and <code><a href="#topic+extractProtFP">extractProtFP</a></code>
for amino acid property based scales descriptors (protein fingerprint).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
data(AAindex)
AAidxmat &lt;- t(na.omit(as.matrix(AAindex[, 7:26])))
scales &lt;- extractScales(x, propmat = AAidxmat, pc = 5, lag = 7, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractScalesGap'>Scales-Based Descriptors derived by Principal Components Analysis
(with Gap Support)</h2><span id='topic+extractScalesGap'></span>

<h3>Description</h3>

<p>This function calculates scales-based descriptors
derived by Principal Components Analysis (PCA), with gap support.
Users can provide customized amino acid property matrices.
This function implements the core computation procedure needed for
the scales-based descriptors derived by AA-Properties (AAindex)
and scales-based descriptors derived by 20+ classes of 2D and 3D
molecular descriptors (Topological, WHIM, VHSE, etc.) in the protr package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractScalesGap(x, propmat, pc, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractScalesGap_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.
Use '<code>-</code>' to represent gaps in the sequence.</p>
</td></tr>
<tr><td><code id="extractScalesGap_+3A_propmat">propmat</code></td>
<td>
<p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents
one property. Note that the one-letter row names must be provided
for we need them to seek the properties for each AA type.</p>
</td></tr>
<tr><td><code id="extractScalesGap_+3A_pc">pc</code></td>
<td>
<p>Integer. Use the first pc principal components as the scales.
Must be no greater than the number of AA properties provided.</p>
</td></tr>
<tr><td><code id="extractScalesGap_+3A_lag">lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td></tr>
<tr><td><code id="extractScalesGap_+3A_scale">scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before PCA? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="extractScalesGap_+3A_silent">silent</code></td>
<td>
<p>Logical. Whether to print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (principal components) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractProtFPGap">extractProtFPGap</a></code> for amino acid property based
scales descriptors (protein fingerprint) with gap support.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># amino acid sequence with gaps
x &lt;- readFASTA(system.file("protseq/align.fasta", package = "protr"))$`IXI_235`
data(AAindex)
AAidxmat &lt;- t(na.omit(as.matrix(AAindex[, 7:26])))
scales &lt;- extractScalesGap(x, propmat = AAidxmat, pc = 5, lag = 7, silent = FALSE)
</code></pre>

<hr>
<h2 id='extractSOCN'>Sequence-Order-Coupling Numbers</h2><span id='topic+extractSOCN'></span>

<h3>Description</h3>

<p>This function calculates the Sequence-Order-Coupling Numbers
(dim: <code>nlag * 2</code>, default is 60).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractSOCN(x, nlag = 30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractSOCN_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="extractSOCN_+3A_nlag">nlag</code></td>
<td>
<p>The maximum lag, defualt is 30.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length <code>nlag * 2</code> named vector
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Kuo-Chen Chou. Prediction of Protein Subcellar Locations by
Incorporating Quasi-Sequence-Order Effect.
<em>Biochemical and Biophysical Research Communications</em>,
2000, 278, 477-483.
</p>
<p>Kuo-Chen Chou and Yu-Dong Cai. Prediction of Protein Sucellular Locations by
GO-FunD-PseAA Predictor.
<em>Biochemical and Biophysical Research Communications</em>,
2004, 320, 1236-1239.
</p>
<p>Gisbert Schneider and Paul Wrede. The Rational Design of
Amino Acid Sequences by Artifical Neural Networks and Simulated
Molecular Evolution: Do Novo Design of an Idealized Leader Cleavge Site.
<em>Biophys Journal</em>, 1994, 66, 335-344.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractQSO">extractQSO</a></code> for quasi-sequence-order descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractSOCN(x)
</code></pre>

<hr>
<h2 id='extractTC'>Tripeptide Composition Descriptor</h2><span id='topic+extractTC'></span>

<h3>Description</h3>

<p>This function calculates the Tripeptide Composition descriptor (dim: 8,000).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractTC(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractTC_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A length 8,000 named vector
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>M. Bhasin, G. P. S. Raghava.
Classification of Nuclear Receptors Based on
Amino Acid Composition and Dipeptide Composition.
<em>Journal of Biological Chemistry</em>, 2004, 279, 23262.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extractAAC">extractAAC</a></code> and <code><a href="#topic+extractDC">extractDC</a></code>
for Amino Acid Composition and Dipeptide Composition descriptors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractTC(x)
</code></pre>

<hr>
<h2 id='getUniProt'>Retrieve Protein Sequences from UniProt by Protein ID</h2><span id='topic+getUniProt'></span>

<h3>Description</h3>

<p>This function retrieves protein sequences from uniprot.org by protein ID(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getUniProt(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUniProt_+3A_id">id</code></td>
<td>
<p>A character vector, as the protein ID(s).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, each component contains one protein sequence.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+readFASTA">readFASTA</a></code> for reading FASTA format files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Network latency may slow down this example
# Only test this when your connection is fast enough
ids &lt;- c("P00750", "P00751", "P00752")
getUniProt(ids)

## End(Not run)
</code></pre>

<hr>
<h2 id='OptAA3d'>OptAA3d.sdf - 20 Amino Acids Optimized with MOE 2011.10 (Semiempirical AM1)</h2><span id='topic+OptAA3d'></span>

<h3>Description</h3>

<p>OptAA3d.sdf - 20 Amino Acids Optimized with MOE 2011.10 (Semiempirical AM1)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># this operation requires the rcdk package
# require(rcdk)
# optaa3d = load.molecules(system.file("sysdata/OptAA3d.sdf", package = "protr"))
# view.molecule.2d(optaa3d[[1]])  # view the first AA
</code></pre>

<hr>
<h2 id='parGOSim'>Protein Similarity Calculation based on Gene Ontology
(GO) Similarity</h2><span id='topic+parGOSim'></span>

<h3>Description</h3>

<p>This function calculates protein similarity based on
Gene Ontology (GO) similarity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parGOSim(
  golist,
  type = c("go", "gene"),
  ont = c("MF", "BP", "CC"),
  organism = "human",
  measure = "Resnik",
  combine = "BMA"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parGOSim_+3A_golist">golist</code></td>
<td>
<p>A list, each component contains
a character vector of GO terms or one Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_type">type</code></td>
<td>
<p>Input type for <code>golist</code>, <code>"go"</code> for GO Terms,
<code>"gene"</code> for gene ID.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_ont">ont</code></td>
<td>
<p>Default is <code>"MF"</code>, can be one of <code>"MF"</code>,
<code>"BP"</code>, or <code>"CC"</code> subontologies.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_organism">organism</code></td>
<td>
<p>Organism name. Default is <code>"human"</code>, can be one of
<code>"anopheles"</code>, <code>"arabidopsis"</code>, <code>"bovine"</code>, <code>"canine"</code>,
<code>"chicken"</code>, <code>"chimp"</code>, <code>"coelicolor"</code>, <code>"ecolik12"</code>,
<code>"ecsakai"</code>, <code>"fly"</code>, <code>"human"</code>, <code>"malaria"</code>,
<code>"mouse"</code>, <code>"pig"</code>, <code>"rat"</code>, <code>"rhesus"</code>,
<code>"worm"</code>, <code>"xenopus"</code>, <code>"yeast"</code> or <code>"zebrafish"</code>.
Before specifying the organism, please install the corresponding genome wide
<a href="https://bioconductor.org/packages/release/BiocViews.html#___OrgDb">annotation data package</a>
for the selected organism.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_measure">measure</code></td>
<td>
<p>Default is <code>"Resnik"</code>, can be one of
<code>"Resnik"</code>, <code>"Lin"</code>, <code>"Rel"</code>, <code>"Jiang"</code>
or <code>"Wang"</code>.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_combine">combine</code></td>
<td>
<p>Default is <code>"BMA"</code>, can be one of <code>"max"</code>,
<code>"average"</code>, <code>"rcmax"</code> or <code>"BMA"</code>
for combining semantic similarity scores of multiple GO terms
associated with proteins.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>n</code> x <code>n</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+twoGOSim">twoGOSim</a></code> for calculating the
GO semantic similarity between two groups of GO terms or two Entrez gene IDs.
See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for paralleled protein similarity
calculation based on Smith-Waterman local alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Be careful when testing this since it involves GO similarity computation
# and might produce unpredictable results in some environments

library("GOSemSim")
library("org.Hs.eg.db")

# By GO Terms
# AP4B1
go1 &lt;- c(
  "GO:0005215", "GO:0005488", "GO:0005515",
  "GO:0005625", "GO:0005802", "GO:0005905"
)
# BCAS2
go2 &lt;- c(
  "GO:0005515", "GO:0005634", "GO:0005681",
  "GO:0008380", "GO:0031202"
)
# PDE4DIP
go3 &lt;- c(
  "GO:0003735", "GO:0005622", "GO:0005840",
  "GO:0006412"
)
golist &lt;- list(go1, go2, go3)
parGOSim(golist, type = "go", ont = "CC", measure = "Wang")

# By Entrez gene id
genelist &lt;- list(c("150", "151", "152", "1814", "1815", "1816"))
parGOSim(genelist, type = "gene", ont = "BP", measure = "Wang")

## End(Not run)
</code></pre>

<hr>
<h2 id='parSeqSim'>Parallellized Protein Sequence Similarity Calculation Based on
Sequence Alignment (In-Memory Version)</h2><span id='topic+parSeqSim'></span>

<h3>Description</h3>

<p>Parallellized calculation of protein sequence similarity based on
sequence alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parSeqSim(
  protlist,
  cores = 2,
  batches = 1,
  verbose = FALSE,
  type = "local",
  submat = "BLOSUM62",
  gap.opening = 10,
  gap.extension = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parSeqSim_+3A_protlist">protlist</code></td>
<td>
<p>A length <code>n</code> list containing <code>n</code> protein sequences,
each component of the list is a character string, storing one protein
sequence. Unknown sequences should be represented as <code>""</code>.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_cores">cores</code></td>
<td>
<p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users can use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_batches">batches</code></td>
<td>
<p>Integer. How many batches should we split the pairwise
similarity computations into. This is useful when you have a large
number of protein sequences, enough number of CPU cores, but not
enough RAM to compute and hold all the pairwise similarities
in a single batch. Defaults to 1.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_verbose">verbose</code></td>
<td>
<p>Print the computation progress?</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_type">type</code></td>
<td>
<p>Type of alignment, default is <code>"local"</code>,
can be <code>"global"</code> or <code>"local"</code>,
where <code>"global"</code> represents Needleman-Wunsch global alignment;
<code>"local"</code> represents Smith-Waterman local alignment.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix, default is <code>"BLOSUM62"</code>,
can be one of <code>"BLOSUM45"</code>, <code>"BLOSUM50"</code>, <code>"BLOSUM62"</code>,
<code>"BLOSUM80"</code>, <code>"BLOSUM100"</code>, <code>"PAM30"</code>,
<code>"PAM40"</code>, <code>"PAM70"</code>, <code>"PAM120"</code>, or <code>"PAM250"</code>.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_gap.opening">gap.opening</code></td>
<td>
<p>The cost required to open a gap of any length
in the alignment. Defaults to 10.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_gap.extension">gap.extension</code></td>
<td>
<p>The cost to extend the length of an existing
gap by 1. Defaults to 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>n</code> x <code>n</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+parSeqSimDisk">parSeqSimDisk</a></code> for the disk-based version.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Be careful when testing this since it involves parallelisation
# and might produce unpredictable results in some environments

library("Biostrings")
library("foreach")
library("doParallel")

s1 &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
s2 &lt;- readFASTA(system.file("protseq/P08218.fasta", package = "protr"))[[1]]
s3 &lt;- readFASTA(system.file("protseq/P10323.fasta", package = "protr"))[[1]]
s4 &lt;- readFASTA(system.file("protseq/P20160.fasta", package = "protr"))[[1]]
s5 &lt;- readFASTA(system.file("protseq/Q9NZP8.fasta", package = "protr"))[[1]]
plist &lt;- list(s1, s2, s3, s4, s5)
(psimmat &lt;- parSeqSim(plist, cores = 2, type = "local", submat = "BLOSUM62"))

## End(Not run)
</code></pre>

<hr>
<h2 id='parSeqSimDisk'>Parallellized Protein Sequence Similarity Calculation Based on
Sequence Alignment (Disk-Based Version)</h2><span id='topic+parSeqSimDisk'></span>

<h3>Description</h3>

<p>Parallellized calculation of protein sequence similarity based on
sequence alignment.
This version caches the partial results in each batch to the
hard drive and merges the results together in the end, which
reduces the memory usage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parSeqSimDisk(
  protlist,
  cores = 2,
  batches = 1,
  path = tempdir(),
  verbose = FALSE,
  type = "local",
  submat = "BLOSUM62",
  gap.opening = 10,
  gap.extension = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parSeqSimDisk_+3A_protlist">protlist</code></td>
<td>
<p>A length <code>n</code> list containing <code>n</code> protein sequences,
each component of the list is a character string, storing one protein
sequence. Unknown sequences should be represented as <code>""</code>.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_cores">cores</code></td>
<td>
<p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users can use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_batches">batches</code></td>
<td>
<p>Integer. How many batches should we split the pairwise
similarity computations into. This is useful when you have a large
number of protein sequences, enough number of CPU cores, but not
enough RAM to compute and hold all the pairwise similarities
in a single batch. Defaults to 1.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_path">path</code></td>
<td>
<p>Directory for caching the results in each batch.
Defaults to the temporary directory.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_verbose">verbose</code></td>
<td>
<p>Print the computation progress?</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_type">type</code></td>
<td>
<p>Type of alignment, default is <code>"local"</code>,
can be <code>"global"</code> or <code>"local"</code>,
where <code>"global"</code> represents Needleman-Wunsch global alignment;
<code>"local"</code> represents Smith-Waterman local alignment.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix, default is <code>"BLOSUM62"</code>,
can be one of <code>"BLOSUM45"</code>, <code>"BLOSUM50"</code>, <code>"BLOSUM62"</code>,
<code>"BLOSUM80"</code>, <code>"BLOSUM100"</code>, <code>"PAM30"</code>,
<code>"PAM40"</code>, <code>"PAM70"</code>, <code>"PAM120"</code>, or <code>"PAM250"</code>.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_gap.opening">gap.opening</code></td>
<td>
<p>The cost required to open a gap of any length
in the alignment. Defaults to 10.</p>
</td></tr>
<tr><td><code id="parSeqSimDisk_+3A_gap.extension">gap.extension</code></td>
<td>
<p>The cost to extend the length of an existing
gap by 1. Defaults to 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>n</code> x <code>n</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for the in-memory version.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Be careful when testing this since it involves parallelisation
# and might produce unpredictable results in some environments

library("Biostrings")
library("foreach")
library("doParallel")

s1 &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
s2 &lt;- readFASTA(system.file("protseq/P08218.fasta", package = "protr"))[[1]]
s3 &lt;- readFASTA(system.file("protseq/P10323.fasta", package = "protr"))[[1]]
s4 &lt;- readFASTA(system.file("protseq/P20160.fasta", package = "protr"))[[1]]
s5 &lt;- readFASTA(system.file("protseq/Q9NZP8.fasta", package = "protr"))[[1]]
set.seed(1010)
plist &lt;- as.list(c(s1, s2, s3, s4, s5)[sample(1:5, 100, replace = TRUE)])
psimmat &lt;- parSeqSimDisk(
  plist,
  cores = 2, batches = 10, verbose = TRUE,
  type = "local", submat = "BLOSUM62"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='protcheck'>Protein sequence amino acid type sanity check</h2><span id='topic+protcheck'></span>

<h3>Description</h3>

<p>This function checks if the protein sequence's amino acid types
are in the 20 default types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protcheck(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protcheck_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical. <code>TRUE</code> if all of the amino acid types
of the sequence are within the 20 default types.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
protcheck(x) # TRUE
protcheck(paste(x, "Z", sep = "")) # FALSE
</code></pre>

<hr>
<h2 id='protseg'>Protein Sequence Segmentation/Partition</h2><span id='topic+protseg'></span>

<h3>Description</h3>

<p>This function extracts the segmentations from the protein sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protseg(
  x,
  aa = c("A", "R", "N", "D", "C", "E", "Q", "G", "H", "I", "L", "K", "M", "F", "P", "S",
    "T", "W", "Y", "V"),
  k = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protseg_+3A_x">x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td></tr>
<tr><td><code id="protseg_+3A_aa">aa</code></td>
<td>
<p>A character, the amino acid type. One of
<code>'A'</code>, <code>'R'</code>, <code>'N'</code>, <code>'D'</code>,
<code>'C'</code>, <code>'E'</code>, <code>'Q'</code>, <code>'G'</code>,
<code>'H'</code>, <code>'I'</code>, <code>'L'</code>, <code>'K'</code>,
<code>'M'</code>, <code>'F'</code>, <code>'P'</code>, <code>'S'</code>,
<code>'T'</code>, <code>'W'</code>, <code>'Y'</code>, <code>'V'</code>.</p>
</td></tr>
<tr><td><code id="protseg_+3A_k">k</code></td>
<td>
<p>A positive integer, specifys the window size
(half of the window), default is 7.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list, each component contains one of the
segmentations (a character string), names of the list components
are the positions of the specified amino acid in the sequence.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
protseg(x, aa = "R", k = 5)
</code></pre>

<hr>
<h2 id='readFASTA'>Read Protein Sequences in FASTA Format</h2><span id='topic+readFASTA'></span>

<h3>Description</h3>

<p>This function reads protein sequences in FASTA format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readFASTA(
  file = system.file("protseq/P00750.fasta", package = "protr"),
  legacy.mode = TRUE,
  seqonly = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readFASTA_+3A_file">file</code></td>
<td>
<p>Path to the file containing the protein sequences
in FASTA format. If it does not contain an absolute or
relative path, the file name is relative to the current
working directory, <code><a href="base.html#topic+getwd">getwd</a></code>.
The default here is to read the <code>P00750.fasta</code> file which
is present in the <code>protseq</code> directory of the protr package.</p>
</td></tr>
<tr><td><code id="readFASTA_+3A_legacy.mode">legacy.mode</code></td>
<td>
<p>If set to <code>TRUE</code>, lines starting with a
semicolon (<code>;</code>) are ignored. Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="readFASTA_+3A_seqonly">seqonly</code></td>
<td>
<p>If set to <code>TRUE</code>, only sequences as returned
without attempt to modify them or to get their names and
annotations (execution time is divided approximately
by a factor 3). Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of the protein sequences.
</p>
<p>The three returned argument are just different forms of the same output.
If one is interested in a Mahalanobis metric over the original data space,
the first argument is all she/he needs. If a transformation into another
space (where one can use the Euclidean metric) is preferred, the second
returned argument is sufficient. Using A and B is equivalent in the
following sense.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Pearson, W.R. and Lipman, D.J. (1988)
Improved tools for biological sequence comparison.
<em>Proceedings of the National Academy of Sciences
of the United States of America</em>, <b>85</b>: 2444&ndash;2448.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+getUniProt">getUniProt</a></code> for retrieving
protein sequences from uniprot.org.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P00750 &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))
</code></pre>

<hr>
<h2 id='readPDB'>Read Protein Sequences in PDB Format</h2><span id='topic+readPDB'></span>

<h3>Description</h3>

<p>This function reads protein sequences in PDB (Protein Data Bank) format,
and return the amino acid sequences represented by single-letter code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readPDB(file = system.file("protseq/4HHB.pdb", package = "protr"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readPDB_+3A_file">file</code></td>
<td>
<p>Path to the file containing the protein sequences in PDB format.
If it does not contain an absolute or
relative path, the file name is relative to the current
working directory, <code><a href="base.html#topic+getwd">getwd</a></code>.
The default here is to read the <code>4HHB.PDB</code> file which
is present in the <code>protseq</code> directory of the protr package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of the protein sequence.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Protein Data Bank Contents Guide: Atomic Coordinate Entry Format Description,
Version 3.30. Accessed 2013-06-26.
<a href="https://files.wwpdb.org/pub/pdb/doc/format_descriptions/Format_v33_Letter.pdf">https://files.wwpdb.org/pub/pdb/doc/format_descriptions/Format_v33_Letter.pdf</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+readFASTA">readFASTA</a></code> for reading
protein sequences in FASTA format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Seq4HHB &lt;- readPDB(system.file("protseq/4HHB.pdb", package = "protr"))
</code></pre>

<hr>
<h2 id='removeGaps'>Remove or replace gaps from protein sequences.</h2><span id='topic+removeGaps'></span>

<h3>Description</h3>

<p>Remove/replace gaps or any irregular characters from protein sequences,
to make them suitable for feature extraction or sequence alignment
based similarity computation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeGaps(x, pattern = "-", replacement = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeGaps_+3A_x">x</code></td>
<td>
<p>character vector, containing the input protein sequence(s).</p>
</td></tr>
<tr><td><code id="removeGaps_+3A_pattern">pattern</code></td>
<td>
<p>character string contains the gap (or other irregular)
character to be removed or replaced. Default is <code>"-"</code>.
For advanced usage, see <code><a href="base.html#topic+gsub">gsub</a></code>.</p>
</td></tr>
<tr><td><code id="removeGaps_+3A_replacement">replacement</code></td>
<td>
<p>a replacement for matched characters.
Default is <code>""</code> (remove the matched character).</p>
</td></tr>
<tr><td><code id="removeGaps_+3A_...">...</code></td>
<td>
<p>addtional parameters for <code><a href="base.html#topic+gsub">gsub</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of protein sequence(s) with gaps or
irregular characters removed/replaced.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># amino acid sequences that contain gaps ("-")
aaseq &lt;- list(
  "MHGDTPTLHEYMLDLQPETTDLYCYEQLSDSSE-EEDEIDGPAGQAEPDRAHYNIVTFCCKCDSTLRLCVQS",
  "MHGDTPTLHEYMLDLQPETTDLYCYEQLNDSSE-EEDEIDGPAGQAEPDRAHYNIVTFCCKCDSTLRLCVQS"
)
## Not run: 
#' # gaps create issues for alignment
parSeqSim(aaseq)

# remove the gaps
nogapseq &lt;- removeGaps(aaseq)
parSeqSim(nogapseq)

## End(Not run)
</code></pre>

<hr>
<h2 id='twoGOSim'>Protein Similarity Calculation based on Gene Ontology (GO) Similarity</h2><span id='topic+twoGOSim'></span>

<h3>Description</h3>

<p>This function calculates the Gene Ontology (GO) similarity
between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoGOSim(
  id1,
  id2,
  type = c("go", "gene"),
  ont = c("MF", "BP", "CC"),
  organism = "human",
  measure = "Resnik",
  combine = "BMA"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoGOSim_+3A_id1">id1</code></td>
<td>
<p>Character vector. When length &gt; 1: each element is a GO term;
when length = 1: the Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_id2">id2</code></td>
<td>
<p>Character vector. When length &gt; 1: each element is a GO term;
when length = 1: the Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_type">type</code></td>
<td>
<p>Input type of id1 and id2, <code>'go'</code> for GO Terms,
<code>"gene"</code> for gene ID.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_ont">ont</code></td>
<td>
<p>Default is <code>"MF"</code>, can be one of <code>"MF"</code>,
<code>"BP"</code>, or <code>"CC"</code> subontologies.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_organism">organism</code></td>
<td>
<p>Organism name. Default is <code>"human"</code>, can be one of
<code>"anopheles"</code>, <code>"arabidopsis"</code>, <code>"bovine"</code>, <code>"canine"</code>,
<code>"chicken"</code>, <code>"chimp"</code>, <code>"coelicolor"</code>, <code>"ecolik12"</code>,
<code>"ecsakai"</code>, <code>"fly"</code>, <code>"human"</code>, <code>"malaria"</code>,
<code>"mouse"</code>, <code>"pig"</code>, <code>"rat"</code>, <code>"rhesus"</code>,
<code>"worm"</code>, <code>"xenopus"</code>, <code>"yeast"</code> or <code>"zebrafish"</code>.
Before specifying the organism, please install the corresponding genome wide
<a href="https://bioconductor.org/packages/release/BiocViews.html#___OrgDb">annotation data package</a>
for the selected organism.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_measure">measure</code></td>
<td>
<p>Default is <code>"Resnik"</code>, can be one of
<code>"Resnik"</code>, <code>"Lin"</code>, <code>"Rel"</code>, <code>"Jiang"</code>
or <code>"Wang"</code>.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_combine">combine</code></td>
<td>
<p>Default is <code>"BMA"</code>, can be one of <code>"max"</code>,
<code>"average"</code>, <code>"rcmax"</code> or <code>"BMA"</code>
for combining semantic similarity scores of multiple GO terms
associated with proteins.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Similarity value.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+parGOSim">parGOSim</a></code> for protein similarity calculation
based on Gene Ontology (GO) semantic similarity.
See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for paralleled protein similarity
calculation based on Smith-Waterman local alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Be careful when testing this since it involves GO similarity computation
# and might produce unpredictable results in some environments

library("GOSemSim")
library("org.Hs.eg.db")

# By GO terms
go1 &lt;- c("GO:0004022", "GO:0004024", "GO:0004023")
go2 &lt;- c("GO:0009055", "GO:0020037")
twoGOSim(go1, go2, type = "go", ont = "MF", measure = "Wang")

# By Entrez gene id
gene1 &lt;- "1956" # EGFR
gene2 &lt;- "2261" # FGFR3
twoGOSim(gene1, gene2, type = "gene", ont = "BP", measure = "Lin")

## End(Not run)
</code></pre>

<hr>
<h2 id='twoSeqSim'>Protein Sequence Alignment for Two Protein Sequences</h2><span id='topic+twoSeqSim'></span>

<h3>Description</h3>

<p>Sequence alignment between two protein sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoSeqSim(
  seq1,
  seq2,
  type = "local",
  submat = "BLOSUM62",
  gap.opening = 10,
  gap.extension = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoSeqSim_+3A_seq1">seq1</code></td>
<td>
<p>Character string, containing one protein sequence.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_seq2">seq2</code></td>
<td>
<p>Character string, containing another protein sequence.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_type">type</code></td>
<td>
<p>Type of alignment, default is <code>"local"</code>,
could be <code>"global"</code> or <code>"local"</code>,
where <code>"global"</code> represents Needleman-Wunsch global alignment;
<code>"local"</code> represents Smith-Waterman local alignment.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix, default is <code>"BLOSUM62"</code>,
can be one of <code>"BLOSUM45"</code>, <code>"BLOSUM50"</code>, <code>"BLOSUM62"</code>,
<code>"BLOSUM80"</code>, <code>"BLOSUM100"</code>, <code>"PAM30"</code>, <code>"PAM40"</code>,
<code>"PAM70"</code>, <code>"PAM120"</code>, or <code>"PAM250"</code>.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_gap.opening">gap.opening</code></td>
<td>
<p>The cost required to open a gap of any length
in the alignment. Defaults to 10.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_gap.extension">gap.extension</code></td>
<td>
<p>The cost to extend the length of an existing
gap by 1. Defaults to 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>Biostrings</code> object containing the alignment scores
and other alignment information.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for paralleled pairwise
protein similarity calculation based on sequence alignment.
See <code><a href="#topic+twoGOSim">twoGOSim</a></code> for calculating the GO semantic
similarity between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Be careful when testing this since it involves sequence alignment
# and might produce unpredictable results in some environments
library("Biostrings")
s1 &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
s2 &lt;- readFASTA(system.file("protseq/P10323.fasta", package = "protr"))[[1]]
seqalign &lt;- twoSeqSim(s1, s2)
summary(seqalign)
score(seqalign)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
