<!DOCTYPE html><html lang="en"><head><title>Help for package PRSPGx</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PRSPGx}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PRS_Dis_CT'><p>Construct disease PRS unadjusted or using clumping and thresholding</p></a></li>
<li><a href='#PRS_Dis_LDpred2'><p>Construct disease PRS using LDpred2</p></a></li>
<li><a href='#PRS_PGx_Bayes'><p>Construct PGx PRS using Bayesian regression</p></a></li>
<li><a href='#PRS_PGx_CT'><p>Construct PGx PRS unadjusted or using clumping and thresholding</p></a></li>
<li><a href='#PRS_PGx_Lasso'><p>Construct PGx PRS using penalized regression</p></a></li>
<li><a href='#PRSPGx.example'><p>Simulated example data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Construct PGx PRS</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Song Zhai &lt;zsviolet1993@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Construct pharmacogenomics (PGx) polygenic risk score (PRS) with PRS-PGx-Unadj (unadjusted), PRS-PGx-CT (clumping and thresholding), PRS-PGx-L, -GL, -SGL (penalized regression), PRS-PGx-Bayes (Bayesian regression). Package is based on &rdquo;Pharmacogenomics Polyenic Risk Score for Drug Response Prediction Using PRS-PGx Methods&rdquo; by Zhai, S., Zhang, H., Mehrotra, D.V., and Shen, J., 2021 (submitted).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gglasso (&ge; 1.5.0), SGL (&ge; 1.3.0), glmnet (&ge; 4.0.2), bigsnpr
(&ge; 1.5.2), Matrix (&ge; 1.2.18), GIGrvg (&ge; 0.5.0), MCMCpack (&ge;
1.4.6), bdsmatrix (&ge; 1.3.4), bigsparser (&ge; 0.4.0), lmtest (&ge;
0.9.37), mvtnorm (&ge; 1.1.0), propagate (&ge; 1.0.6), bigparallelr
(&ge; 0.2.3), methods (&ge; 3.6.3), bigstatsr (&ge; 1.2.3), Rfast (&ge;
1.9.9), matrixcalc (&ge; 1.0-3)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-07-19 15:16:57 UTC; zhaiso</td>
</tr>
<tr>
<td>Author:</td>
<td>Song Zhai [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-07-20 15:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='PRS_Dis_CT'>Construct disease PRS unadjusted or using clumping and thresholding</h2><span id='topic+PRS_Dis_CT'></span>

<h3>Description</h3>

<p>Shrink prognostic effect sizes by p-value cutoff (PRS-Dis-CT turns out to be PRS-Dis-Unadj when setting p-value cutoff = 1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_Dis_CT(
  DIS_GWAS,
  G_reference,
  pcutoff = 1e-05,
  clumping = TRUE,
  p1 = 1e-04,
  d1 = 250000,
  r1 = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PRS_Dis_CT_+3A_dis_gwas">DIS_GWAS</code></td>
<td>
<p>a numeric matrix containing disease GWAS summary statistics, including SNP ID, position, <code class="reqn">\beta</code>, SE(<code class="reqn">\beta</code>), p-value, N, and MAF</p>
</td></tr>
<tr><td><code id="PRS_Dis_CT_+3A_g_reference">G_reference</code></td>
<td>
<p>a numeric matrix containing the individual-level genotype information from the reference panel (e.g., 1KG)</p>
</td></tr>
<tr><td><code id="PRS_Dis_CT_+3A_pcutoff">pcutoff</code></td>
<td>
<p>a numeric value indicating the p-value cutoff</p>
</td></tr>
<tr><td><code id="PRS_Dis_CT_+3A_clumping">clumping</code></td>
<td>
<p>a logical flag indicating should clumping be performed</p>
</td></tr>
<tr><td><code id="PRS_Dis_CT_+3A_p1">p1</code></td>
<td>
<p>a numeric value indicating p-value threshold to decide flag SNPs in clumping</p>
</td></tr>
<tr><td><code id="PRS_Dis_CT_+3A_d1">d1</code></td>
<td>
<p>a numeric value indicating window size in clumping</p>
</td></tr>
<tr><td><code id="PRS_Dis_CT_+3A_r1">r1</code></td>
<td>
<p>a numeric value indicating correlation in clumping</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PRS-Dis-CT automatically sets predictive effect sizes equivalent to the prognostic effect sizes; and only need disease GWAS summary statistics
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Euesden, J., Lewis, C.M. &amp; O'Reilly, P.F. PRSice: Polygenic Risk Score software. Bioinformatics 564, 1466-1468 (2015).
</p>
<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PRSPGx.example); attach(PRSPGx.example)
coef_est &lt;- PRS_Dis_CT(DIS_GWAS, G_reference, pcutoff = 0.01, clumping = TRUE)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)

</code></pre>

<hr>
<h2 id='PRS_Dis_LDpred2'>Construct disease PRS using LDpred2</h2><span id='topic+PRS_Dis_LDpred2'></span>

<h3>Description</h3>

<p>Using snp_ldpred2_grid function from bigsnpr function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_Dis_LDpred2(DIS_GWAS, G_reference, pcausal, h2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PRS_Dis_LDpred2_+3A_dis_gwas">DIS_GWAS</code></td>
<td>
<p>a numeric matrix containing disease GWAS summary statistics, including SNP ID, position, <code class="reqn">\beta</code>, SE(<code class="reqn">\beta</code>), p-value, N, and MAF</p>
</td></tr>
<tr><td><code id="PRS_Dis_LDpred2_+3A_g_reference">G_reference</code></td>
<td>
<p>a numeric matrix containing the individual-level genotype information from the reference panel (e.g., 1KG)</p>
</td></tr>
<tr><td><code id="PRS_Dis_LDpred2_+3A_pcausal">pcausal</code></td>
<td>
<p>a numeric value indicating the hyper-parameter as the proportion of causal variants</p>
</td></tr>
<tr><td><code id="PRS_Dis_LDpred2_+3A_h2">h2</code></td>
<td>
<p>a numeric value indicating the estimated heritability</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PRS-Dis-LDpred2 automatically sets predictive effect sizes equivalent to the prognostic effect sizes; and only need disease GWAS summary statistics and external reference genotype
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Prive, F., Arbel, J. &amp; Vilhjalmsson, B.J. LDpred2: better, faster, stronger. Bioinformatics 36, 5424-5431 (2020).
</p>
<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PRSPGx.example); attach(PRSPGx.example)
coef_est &lt;- PRS_Dis_LDpred2(DIS_GWAS, G_reference, pcausal = 0.1, h2 = 0.4)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)


</code></pre>

<hr>
<h2 id='PRS_PGx_Bayes'>Construct PGx PRS using Bayesian regression</h2><span id='topic+PRS_PGx_Bayes'></span>

<h3>Description</h3>

<p>Flexibly shrink prognostic and predictive effect sizes simutaneously with glocal-local shrinkage parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_PGx_Bayes(
  PGx_GWAS,
  G_reference,
  n.itr = 1000,
  n.burnin = 500,
  n.gap = 10,
  paras,
  standardize = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PRS_PGx_Bayes_+3A_pgx_gwas">PGx_GWAS</code></td>
<td>
<p>a numeric list containing PGx GWAS summary statistics (with SNP ID, position, <code class="reqn">\beta</code>, <code class="reqn">\alpha</code>, 2-df p-value, MAF and N), SD(Y), and mean(T)</p>
</td></tr>
<tr><td><code id="PRS_PGx_Bayes_+3A_g_reference">G_reference</code></td>
<td>
<p>a numeric matrix containing the individual-level genotype information from the reference panel (e.g., 1KG)</p>
</td></tr>
<tr><td><code id="PRS_PGx_Bayes_+3A_n.itr">n.itr</code></td>
<td>
<p>a numeric value indicating the total number of MCMC iteration</p>
</td></tr>
<tr><td><code id="PRS_PGx_Bayes_+3A_n.burnin">n.burnin</code></td>
<td>
<p>a numeric value indicating the number of burn in</p>
</td></tr>
<tr><td><code id="PRS_PGx_Bayes_+3A_n.gap">n.gap</code></td>
<td>
<p>a numeric value indicating the MCMC gap</p>
</td></tr>
<tr><td><code id="PRS_PGx_Bayes_+3A_paras">paras</code></td>
<td>
<p>a numeric vector containg hyper-parameters (<code class="reqn">v</code>, <code class="reqn">\phi</code>)</p>
</td></tr>
<tr><td><code id="PRS_PGx_Bayes_+3A_standardize">standardize</code></td>
<td>
<p>a logical flag indicating should phenotype and genotype be standardized</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PRS-PGx-Bayes only needs PGx summary statistics and external reference genotype
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Ge, T., Chen, CY., Ni, Y. et al. Polygenic prediction via Bayesian regression and continuous shrinkage priors. Nat. Commun. 10, 1776 (2019).
</p>
<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PRSPGx.example); attach(PRSPGx.example)
paras = c(3, 5)
coef_est &lt;- PRS_PGx_Bayes(PGx_GWAS, G_reference, paras = paras, n.itr = 10, n.burnin = 5, n.gap = 1)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)


</code></pre>

<hr>
<h2 id='PRS_PGx_CT'>Construct PGx PRS unadjusted or using clumping and thresholding</h2><span id='topic+PRS_PGx_CT'></span>

<h3>Description</h3>

<p>Shrink prognostic and predictive effect sizes simutaneously by 2-df (main and interaction) p-value cutoff (PRS-PGx-CT turns out to be PRS-PGx-Unadj when setting p-value cutoff = 1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_PGx_CT(
  PGx_GWAS,
  G_reference,
  pcutoff = 1e-04,
  clumping = TRUE,
  p1 = 1e-04,
  d1 = 250000,
  r1 = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PRS_PGx_CT_+3A_pgx_gwas">PGx_GWAS</code></td>
<td>
<p>a numeric matrix containing PGx GWAS summary statistics, including SNP ID, MAF, position, <code class="reqn">\beta</code>, <code class="reqn">\alpha</code>, 2-df p-value, and N</p>
</td></tr>
<tr><td><code id="PRS_PGx_CT_+3A_g_reference">G_reference</code></td>
<td>
<p>a numeric matrix containing the individual-level genotype information from the reference panel (e.g., 1KG)</p>
</td></tr>
<tr><td><code id="PRS_PGx_CT_+3A_pcutoff">pcutoff</code></td>
<td>
<p>a numeric value indicating the p-value cutoff</p>
</td></tr>
<tr><td><code id="PRS_PGx_CT_+3A_clumping">clumping</code></td>
<td>
<p>a logical flag indicating should clumping be performed</p>
</td></tr>
<tr><td><code id="PRS_PGx_CT_+3A_p1">p1</code></td>
<td>
<p>a numeric value indicating p-value threshold to decide flag SNPs in clumping</p>
</td></tr>
<tr><td><code id="PRS_PGx_CT_+3A_d1">d1</code></td>
<td>
<p>a numeric value indicating window size in clumping</p>
</td></tr>
<tr><td><code id="PRS_PGx_CT_+3A_r1">r1</code></td>
<td>
<p>a numeric value indicating correlation in clumping</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PRS-PGx-CT only needs PGx summary statistics
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes, the third sublist contains 2-df p-values
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PRSPGx.example); attach(PRSPGx.example)
coef_est &lt;- PRS_PGx_CT(PGx_GWAS, G_reference, pcutoff = 0.01, clumping = TRUE)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)

</code></pre>

<hr>
<h2 id='PRS_PGx_Lasso'>Construct PGx PRS using penalized regression</h2><span id='topic+PRS_PGx_Lasso'></span>

<h3>Description</h3>

<p>Shrink prognostic and predictive effect sizes simultaneously via the penalized term. With different assumptions on the relationship between the two effects, can be PRS-PGx-L (Lasso), PRS-PGx-GL (Group Lasso), and PRS-PGx-SGL (Sparse Group Lasso)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_PGx_Lasso(Y, Tr, G, intercept = TRUE, lambda, method, alpha = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PRS_PGx_Lasso_+3A_y">Y</code></td>
<td>
<p>a numeric vector containing the quantitative trait</p>
</td></tr>
<tr><td><code id="PRS_PGx_Lasso_+3A_tr">Tr</code></td>
<td>
<p>a numeric vector containing the treatment assignment</p>
</td></tr>
<tr><td><code id="PRS_PGx_Lasso_+3A_g">G</code></td>
<td>
<p>a numeric matrix containing genotype information</p>
</td></tr>
<tr><td><code id="PRS_PGx_Lasso_+3A_intercept">intercept</code></td>
<td>
<p>a logical flag indicating should intercept be fitted (default=TRUE) or set to be FALSE</p>
</td></tr>
<tr><td><code id="PRS_PGx_Lasso_+3A_lambda">lambda</code></td>
<td>
<p>a numeric value indicating the penalty</p>
</td></tr>
<tr><td><code id="PRS_PGx_Lasso_+3A_method">method</code></td>
<td>
<p>a logical flag for different penalized regression methods: 1 = PRS-PGx-L, 2 = PRS-PGx-GL, 3 = PRS-PGx-SGL</p>
</td></tr>
<tr><td><code id="PRS_PGx_Lasso_+3A_alpha">alpha</code></td>
<td>
<p>a numeric value indicating the mixing parameter (only used when method = 3). alpha = 1 is the lasso penalty. alpha = 0 is the group lasso penalty</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PRS-PGx-Lasso requires individudal-level data
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Yang, Y. &amp; Zou, H. A fast unified algorithm for solving group-lasso penalize learning problems. Statistics and Computing 25, 1129-1141 (2015).
</p>
<p>Simon, N., Friedman, J., Hastie, T. &amp; Tibshirani, R. Fit a GLM (or cox model) with a combination of lasso and group lasso regularization. R package version, 1 (2015).
</p>
<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PRSPGx.example); attach(PRSPGx.example)
coef_est &lt;- PRS_PGx_Lasso(Y, Tr, G, lambda = 1, method = 1)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)


</code></pre>

<hr>
<h2 id='PRSPGx.example'>Simulated example data</h2><span id='topic+PRSPGx.example'></span>

<h3>Description</h3>

<p>Simulated example data required by PRS-DIS and PRS-PGx functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PRSPGx.example)
</code></pre>


<h3>Format</h3>

<p>A list with 8 sublists:
</p>

<dl>
<dt>PGx_GWAS</dt><dd><p>PGx GWAS including SNP ID, MAF, position, <code class="reqn">\beta</code>, <code class="reqn">\alpha</code>, 2-df p-value, and N; SD(Y), and mean(T)</p>
</dd>
<dt>DIS_GWAS</dt><dd><p>disease GWAS including SNP ID, MAF, position, <code class="reqn">\beta</code>, SE(<code class="reqn">\beta</code>), p-value, and N</p>
</dd>
<dt>G_reference</dt><dd><p>simulated individual-level genotype from the reference panel matched with the simulated sample PGx genotype</p>
</dd>
<dt>Y</dt><dd><p>simulated phenotype (continuous)</p>
</dd>
<dt>T</dt><dd><p>simulated treatment assignment, 1 = treatment, 0 = placebo</p>
</dd>
<dt>G</dt><dd><p>simulated sample PGx genotype with 100 SNPs and 4000 subjects</p>
</dd>
<dt>beta</dt><dd><p>simulated prognostic effect sizes (i.e., the underlying true prognostic effect sizes)</p>
</dd>
<dt>alpha</dt><dd><p>simulated predictive effect sizes (i.e., the underlying true predictive effect sizes)</p>
</dd></dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
