<!DOCTYPE html><html lang="en-US"><head><title>Help for package PSSMCOOL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PSSMCOOL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PSSMCOOL-package'><p>PSSMCOOL:Extracting Various Feature vectors from PSSM Matrix</p></a></li>
<li><a href='#aac_pssm'><p>AAC-PSSM feature vectors</p></a></li>
<li><a href='#aadp_pssm'><p>DPC-PSSM,AAC-PSSM and AADP-PSSM feature vectors</p></a></li>
<li><a href='#AATP_TPC'><p>AATP_TPC feature vector</p></a></li>
<li><a href='#AB_PSSM'><p>AB-PSSM feature vector</p></a></li>
<li><a href='#Averag_Block'><p>Averag Block feature vector</p></a></li>
<li><a href='#consunsus_sequence'><p>consunsus_sequence</p></a></li>
<li><a href='#CS_PSe_PSSM'><p>CSP-SegPseP-SegACP feature vector</p></a></li>
<li><a href='#DFMCA_PSSM'><p>DMACA-PSSM feature</p></a></li>
<li><a href='#Discrete_Cosine_Transform'><p>Discrete Cosin Transform Feature</p></a></li>
<li><a href='#disulfid'><p>Disulfide connectivity feature</p></a></li>
<li><a href='#DP_PSSM'><p>DP_PSSM feature vector</p></a></li>
<li><a href='#dpc_pssm'><p>DPC_PSSM feature vector</p></a></li>
<li><a href='#dwt_PSSM'><p>discrete wavelet transform feature vector</p></a></li>
<li><a href='#EDP_EEDP_MEDP'><p>EDP_EEDP_MEDP feature vector</p></a></li>
<li><a href='#FPSSM'><p>D-FPSSM and SF-PSSM feature vectors</p></a></li>
<li><a href='#FPSSM2'><p>Mixture of Two FPSSM Features</p></a></li>
<li><a href='#grey_pssm_pseAAC'><p>grey pssm feature vector</p></a></li>
<li><a href='#k_mers'><p>3-mer and 2-mer in dataframe</p></a></li>
<li><a href='#k_separated_bigrams_pssm'><p>k_separated_bigrams_pssm feature vector</p></a></li>
<li><a href='#kiderafactor'><p>kiderafactor feature</p></a></li>
<li><a href='#LPC_PSSM'><p>Linear predictive coding feature</p></a></li>
<li><a href='#MBMGACPSSM'><p>MBMGACPSSM feature</p></a></li>
<li><a href='#pse_pssm'><p>pseudo position-specific scoring matrix feature</p></a></li>
<li><a href='#pssm_ac'><p>auto covariance transformation feature vector</p></a></li>
<li><a href='#pssm_cc'><p>Cross covarianse feature vector</p></a></li>
<li><a href='#pssm_composition'><p>PSSM-COMPOSITION feature</p></a></li>
<li><a href='#PSSM_SD'><p>PSSM-SD feature</p></a></li>
<li><a href='#pssm_seg'><p>PSSM-Seg feature vector</p></a></li>
<li><a href='#pssm400'><p>PSSM400 feature</p></a></li>
<li><a href='#PSSMBLOCK'><p>PSSM BLOCK feature vector</p></a></li>
<li><a href='#RPM_PSSM'><p>RPM-PSSM feature vector</p></a></li>
<li><a href='#rpssm'><p>RPSSM feature</p></a></li>
<li><a href='#scsh2'><p>SCSH Feature vector</p></a></li>
<li><a href='#single_Average'><p>single Average feature</p></a></li>
<li><a href='#smoothed_PSSM'><p>smoothed PSSM feature</p></a></li>
<li><a href='#SOMA_PSSM'><p>SOMA PSSM Feature</p></a></li>
<li><a href='#SVD_PSSM'><p>Singular Value Decomposition (SVD)</p></a></li>
<li><a href='#three_mer'><p>3-Mer and 2-Mer</p></a></li>
<li><a href='#trigrame_pssm'><p>trigrame feature vector</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Features Extracted from Position Specific Scoring Matrix (PSSM)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.4</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, gtools, infotheo, phonTools, dtt</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alireza mohammadi &lt;alireza691111@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>Returns almost all features that has been extracted from Position Specific 
             Scoring Matrix (PSSM) so far, which is a matrix of L rows (L is protein length) 
             and 20 columns produced by 'PSI-BLAST' which is a program to produce
             PSSM Matrix from multiple sequence alignment of proteins
             see <a href="https://www.ncbi.nlm.nih.gov/books/NBK2590/">https://www.ncbi.nlm.nih.gov/books/NBK2590/</a> for mor details. some 
             of these features are described in Zahiri, J., et al.(2013)
             &lt;<a href="https://doi.org/10.1016%2Fj.ygeno.2013.05.006">doi:10.1016/j.ygeno.2013.05.006</a>&gt;,
             Saini, H., et al.(2016)
             &lt;<a href="https://doi.org/10.17706%2Fjsw.11.8.756-767">doi:10.17706/jsw.11.8.756-767</a>&gt;,
             Ding, S., et al.(2014)
             &lt;<a href="https://doi.org/10.1016%2Fj.biochi.2013.09.013">doi:10.1016/j.biochi.2013.09.013</a>&gt;,
             Cheng, C.W., et al.(2008)
             &lt;<a href="https://doi.org/10.1186%2F1471-2105-9-S12-S6">doi:10.1186/1471-2105-9-S12-S6</a>&gt;,
             Juan, E.Y., et al.(2009)
             &lt;<a href="https://doi.org/10.1109%2FCISIS.2009.194">doi:10.1109/CISIS.2009.194</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/BioCool-Lab/PSSMCOOL">https://github.com/BioCool-Lab/PSSMCOOL</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/BioCool-Lab/PSSMCOOL/issues">https://github.com/BioCool-Lab/PSSMCOOL/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, spelling, waveslim, knitr, rmarkdown</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-04 10:42:14 UTC; LapTop</td>
</tr>
<tr>
<td>Author:</td>
<td>Javad zahiri [aut],
  Alireza mohammadi [aut, cre],
  Saber mohammadi [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-04 13:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='PSSMCOOL-package'>PSSMCOOL:Extracting Various Feature vectors from PSSM Matrix</h2><span id='topic+PSSMCOOL-package'></span>

<h3>Description</h3>

<p>This package is a Comprehensive toolkit for generating various numerical features from PSSM Matrix correspond to each protein sequence and covers all features introduced in POSSUM Package and some other new features. see <a href="https://rpubs.com/alireza_69/640294">PSSMCOOL_Rpubs</a> for more information.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Alireza Mohammadi <a href="mailto:alireza691111@gmail.com">alireza691111@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Alireza Mohammadi <a href="mailto:alireza691111@gmail.com">alireza691111@gmail.com</a>
</p>
</li>
<li><p> Javad Zahiri <a href="mailto:zahiri@modares.ac.ir">zahiri@modares.ac.ir</a>
</p>
</li>
<li><p> Saber Mohammadi <a href="mailto:sabermohammdi.1994@gmail.com">sabermohammdi.1994@gmail.com</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><strong>All functions in this package:</strong>
</p>

<ul>
<li> <p><code><a href="#topic+AATP_TPC">AATP_TPC</a></code>
</p>
</li>
<li> <p><code><a href="#topic+AB_PSSM">AB_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+Averag_Block">Averag_Block</a></code>
</p>
</li>
<li> <p><code><a href="#topic+consunsus_sequence">consunsus_sequence</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CS_PSe_PSSM">CS_PSe_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+DFMCA_PSSM">DFMCA_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+Discrete_Cosine_Transform">Discrete_Cosine_Transform</a></code>
</p>
</li>
<li> <p><code><a href="#topic+disulfid">disulfid</a></code>
</p>
</li>
<li> <p><code><a href="#topic+DP_PSSM">DP_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+dpc_pssm">dpc_pssm</a></code>
</p>
</li>
<li> <p><code><a href="#topic+dwt_PSSM">dwt_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+EDP_EEDP_MEDP">EDP_EEDP_MEDP</a></code>
</p>
</li>
<li> <p><code><a href="#topic+FPSSM">FPSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+FPSSM2">FPSSM2</a></code>
</p>
</li>
<li> <p><code><a href="#topic+grey_pssm_pseAAC">grey_pssm_pseAAC</a></code>
</p>
</li>
<li> <p><code><a href="#topic+k_mers">k_mers</a></code>
</p>
</li>
<li> <p><code><a href="#topic+k_separated_bigrams_pssm">k_separated_bigrams_pssm</a></code>
</p>
</li>
<li> <p><code><a href="#topic+kiderafactor">kiderafactor</a></code>
</p>
</li>
<li> <p><code><a href="#topic+LPC_PSSM">LPC_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+MBMGACPSSM">MBMGACPSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+pse_pssm">pse_pssm</a></code>
</p>
</li>
<li> <p><code><a href="#topic+pssm400">pssm400</a></code>
</p>
</li>
<li> <p><code><a href="#topic+pssm_ac">pssm_ac</a></code>
</p>
</li>
<li> <p><code><a href="#topic+pssm_cc">pssm_cc</a></code>
</p>
</li>
<li> <p><code><a href="#topic+PSSM_SD">PSSM_SD</a></code>
</p>
</li>
<li> <p><code><a href="#topic+pssm_seg">pssm_seg</a></code>
</p>
</li>
<li> <p><code><a href="#topic+PSSMBLOCK">PSSMBLOCK</a></code>
</p>
</li>
<li> <p><code><a href="#topic+rpssm">rpssm</a></code>
</p>
</li>
<li> <p><code><a href="#topic+RPM_PSSM">RPM_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+scsh2">scsh2</a></code>
</p>
</li>
<li> <p><code><a href="#topic+single_Average">single_Average</a></code>
</p>
</li>
<li> <p><code><a href="#topic+smoothed_PSSM">smoothed_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+SOMA_PSSM">SOMA_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+SVD_PSSM">SVD_PSSM</a></code>
</p>
</li>
<li> <p><code><a href="#topic+trigrame_pssm">trigrame_pssm</a></code>
</p>
</li></ul>

<p><strong>Useful links:</strong>
</p>

<ul>
<li> <p><a href="https://rpubs.com/alireza_69/640294">https://rpubs.com/alireza_69/640294</a>
</p>
</li>
<li> <p><a href="https://github.com/BioCool-Lab/PSSMCOOL/">https://github.com/BioCool-Lab/PSSMCOOL/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/BioCool-Lab/PSSMCOOL/issues">https://github.com/BioCool-Lab/PSSMCOOL/issues</a>
</p>
</li></ul>


<hr>
<h2 id='aac_pssm'>AAC-PSSM feature vectors</h2><span id='topic+aac_pssm'></span>

<h3>Description</h3>

<p>AAC-PSSM Feature stands for Amino Acid composition which is actually mean of PSSM Matrix
columns which its length is 20. combination of this feature vector and DPC-PSSM feature vector would be
AADP-PSSM feature vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aac_pssm(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aac_pssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 20
</p>


<h3>References</h3>

<p>Liu, T., Zheng, X. and Wang, J. (2010) Prediction of protein structural class for
low-similarity sequences using support vector machine and PSI-BLAST profile, Biochimie, 92, 1330-1334.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-aac_pssm(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='aadp_pssm'>DPC-PSSM,AAC-PSSM and AADP-PSSM feature vectors</h2><span id='topic+aadp_pssm'></span>

<h3>Description</h3>

<p>This feature is combination of amino asid composition and dipeptide composition feature
vectors. DPC feature stands for dipeptide composition, to get this feature vector for both different columns
the elements in two consecutive rows corresponds to this columns, would be multiplied and this scenario is done
for all L-1 consecutive rows (L is protein length) and finally summation of these numbers divides by L-1. since
the result depends on two different columns, eventually DPC Feature vector would be of length 400. AAC-PSSM is actually
mean of PSSM Matrix columns which its length is 20. eventually AADP-PSSM is combination of these vectors
and with length 420.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aadp_pssm(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aadp_pssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 420
</p>


<h3>References</h3>

<p>Liu, T., Zheng, X. and Wang, J. (2010) Prediction of protein structural class for
low-similarity sequences using support vector machine and PSI-BLAST profile, Biochimie, 92, 1330-1334.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-aadp_pssm(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='AATP_TPC'>AATP_TPC feature vector</h2><span id='topic+AATP_TPC'></span>

<h3>Description</h3>

<p>For getting this features which have been used to protein structural class prediction,
at first mean of every column in PSSM Matrix is computed to achieve a 20-dimensional vector
called AAC.then by combining it with other vector of length 400 called TPC, which is similar to <code><a href="#topic+dpc_pssm">dpc_pssm</a></code>
AATP feature vector of length 420 is obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AATP_TPC(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AATP_TPC_+3A_pssm_name">pssm_name</code></td>
<td>
<p>is name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of two feature vectors with 400 and 420 dimensions
</p>


<h3>References</h3>

<p>Zhang, S., Ye, F. and Yuan, X. (2012) Using principal component analysis and support vector machine to predict protein
structural class for low-similarity sequences via PSSM, Journal of Biomolecular Structure &amp; Dynamics, 29, 634-642.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dpc_pssm">dpc_pssm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-AATP_TPC(paste0(system.file("extdata",package="PSSMCOOL"),"/C7GQS7.txt.pssm"))
</code></pre>

<hr>
<h2 id='AB_PSSM'>AB-PSSM feature vector</h2><span id='topic+AB_PSSM'></span>

<h3>Description</h3>

<p>to get This feature at first, each protein sequence is divided into
20 equal parts, each of which is called a block, and in each block the row vectors of the PSSM matrix related
to that block are added together and The resulting final vector is divided by the length of that block, which is
5
Finally, by placing these 20 vectors side by side, feature vector of length 400 is obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AB_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AB_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>AB-PSSM feature vector of length 400
</p>


<h3>References</h3>

<p>Jeong, J.C., Lin, X. and Chen, X.W. (2011) On position-specific scoring matrix for protein function prediction
, IEEE/ACM transactions on computational biology and bioinformatics / IEEE, ACM, 8, 308-315.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;- AB_PSSM(system.file("extdata","C7GRQ3.txt.pssm",package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='Averag_Block'>Averag Block feature vector</h2><span id='topic+Averag_Block'></span>

<h3>Description</h3>

<p>In this feature at first PSSM Matrix is divided to 20 Blocks. Then for each
Block mean of columns is computed to get 20-dimensional vector, eventually by appending these vectors to each other final feature
vector is obtained which would be of length 400. this feature vector is similar to <code><a href="#topic+PSSMBLOCK">PSSMBLOCK</a></code> for N=20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Averag_Block(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Averag_Block_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400
</p>


<h3>References</h3>

<p>L. Nanni, A. Lumini, and S. J. T. S. W. J. Brahnam, &quot;An empirical study of different approaches for protein classification,&quot;
vol. 2014, 2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>v&lt;-Averag_Block(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='consunsus_sequence'>consunsus_sequence</h2><span id='topic+consunsus_sequence'></span>

<h3>Description</h3>

<p>This feature vector is constructed from PSSM Matrix as:
<code class="reqn">\alpha(i)=argmax(P_{i,j})</code> where i varies between 1 and L and j between 1 and 20, L indicates protein length
and &quot;arg&quot; represents the argument of the maximum the ith base of the consensus sequence (CS) is then set to be the
<code class="reqn">\alpha(i)</code>th amino acid in the amino acid alphabet and a consensus sequence is constructed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consunsus_sequence(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="consunsus_sequence_+3A_pssm_name">pssm_name</code></td>
<td>
<p>is the name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>consunsus sequence wich extracted from PSSM
</p>


<h3>References</h3>

<p>Y. Liang, S. Liu, S. J. C. Zhang, and m. m. i. medicine, &quot;Prediction of protein structural
classes for low-similarity sequences based on consensus sequence and segmented PSSM,&quot; vol. 2015, 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-consunsus_sequence(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='CS_PSe_PSSM'>CSP-SegPseP-SegACP feature vector</h2><span id='topic+CS_PSe_PSSM'></span>

<h3>Description</h3>

<p>This feature vector is constructed  by fusing consensus sequence (CS), segmented PsePSSM, and
segmented auto-covariance transformation (ACT) based on PSSM. by consensus sequence a 40-dimensional feature
vector is obtained, in segmented PsePSSM group, by dividing PSSM Matrix to 2 and 3 segments a 380-dimensional
feature vector is obtained and in ACT group, similar to the previous group at first PSSM Matrix is divided
to 2 and 3 segments then a feature vector of length 280 is obtained.eventually by fusing these features a
700-dimensional feature vector is obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CS_PSe_PSSM(pssm_name, vec_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CS_PSe_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="CS_PSe_PSSM_+3A_vec_name">vec_name</code></td>
<td>
<p>a character that user imports to specify kind of feature vector which it can be varied between
four values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If vec_name equals to &quot;segmented_psepssm&quot; then a feature vector of length 380 is obtained.
if vec_name equals to &quot;segmented_acpssm&quot; then a feature vector of length 280 is obtained, and if vec_name
equals to &quot;cspssm&quot; the obtained feature vector would be of length 40 eventually if vec_name equals to
&quot;total&quot; then feature vector would be of length 700.
</p>


<h3>Value</h3>

<p>feature vector that its length depends on the vec_name which user imports. vec_name can be one
of &quot;cspssm&quot;, &quot;segmented_psepssm&quot;, &quot;segmented_acpssm&quot;, &quot;total&quot;.
</p>


<h3>References</h3>

<p>Y. Liang, S. Liu, S. J. C. Zhang, and m. m. i. medicine, &quot;Prediction of protein structural
classes for low-similarity sequences based on consensus sequence and segmented PSSM,&quot; vol. 2015, 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-CS_PSe_PSSM(system.file("extdata", "C7GSI6.txt.pssm", package="PSSMCOOL"),"total")
</code></pre>

<hr>
<h2 id='DFMCA_PSSM'>DMACA-PSSM feature</h2><span id='topic+DFMCA_PSSM'></span>

<h3>Description</h3>

<p>In this feature each column of PSSM Matrix, can be regarded as a time series. Each PSSM
contains 20 columns Hence, each PSSM can be considered as 20 time series.The detrended moving-average
cross-correlation analysis (DMCA) is developed to measure the level of cross-correlation between two
non-stationary time series by fusing the detrended cross-correlation analysis (DCCA) and the detrended
moving average(DMA).this function utilizes this algorithm for each column and each pair of columns to produce
a feature vector of length 290.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DFMCA_PSSM(pssm_name, n = 7)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DFMCA_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="DFMCA_PSSM_+3A_n">n</code></td>
<td>
<p>A parameter called the window size that must be smaller than the length of the sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 210
</p>


<h3>Note</h3>

<p>parameter n must be equal or greater than 3 and equal or less then L which L is length of protein
</p>


<h3>References</h3>

<p>Y. Liang, S. Zhang, S. J. S. Ding, and Q. i. E. Research, &quot;Accurate prediction of Gram-negative bacterial secreted protein
types by fusing multiple statistical features from PSI-BLAST profile,&quot; vol. 29, no. 6, pp. 469-481, 2018.
</p>
<p>Y. Liang and S. J. A. b. Zhang, &quot;Prediction of apoptosis protein’s subcellular localization by fusing two different
descriptors based on evolutionary information,&quot; vol. 66, no. 1, pp. 61-78, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-DFMCA_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"),7)
</code></pre>

<hr>
<h2 id='Discrete_Cosine_Transform'>Discrete Cosin Transform Feature</h2><span id='topic+Discrete_Cosine_Transform'></span>

<h3>Description</h3>

<p>To construct this feature vector, Two-Dimensional DCT algorithm has been used by applying
<code><a href="dtt.html#topic+dct">dct</a></code> function from dtt package which DCT stands for Discrete Cosin Transform.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Discrete_Cosine_Transform(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Discrete_Cosine_Transform_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400
</p>


<h3>References</h3>

<p>Wang, L., et al., Advancing the prediction accuracy of protein-protein interactions by utilizing
evolutionary information from position-specific scoring matrix and ensemble classifier. 2017. 418: p. 105-110.
</p>
<p>Y. Wang, Y. Ding, F. Guo, L. Wei, and J. J. P. o. Tang, &quot;Improved detection of DNA-binding proteins
via compression technology on PSSM information,&quot; vol. 12, no. 9, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-Discrete_Cosine_Transform(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='disulfid'>Disulfide connectivity feature</h2><span id='topic+disulfid'></span>

<h3>Description</h3>

<p>This feature is used to predict the disulfide bond within a protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>disulfid(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="disulfid_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the purpose of predicting disulfide bond in protein at first, the total
number of cysteine amino acids in the protein sequence is counted and their position in the protein sequence is
identified. Then, using a sliding window with length of 13, moved on the PSSM matrix from top to bottom so that
the middle of the window is on the amino acid cysteine, then the rows below the matrix obtained from the PSSM
matrix with dimension of 13 x 20 are placed next to each other to get a feature vector with a length of
260 = 20 * 13 per cysteine, and if the position of the first and last cysteine in the protein sequence is such
that the middle of sliding window is not on cysteine residue when moving on PSSM Matrix, then the required
number of zero rows from top and bottom is added to the PSSM matrix to achieve this goal.Thus, for every
cysteine amino-acid presented in protein sequence, a feature vector with a length of 260 is formed.Then all
the pairwise combinations of these cysteines is wrote in the first column of a table, and in front of each of these
pairwise combinations, the corresponding feature vectors are glued together to get a feature vector of length 520
for each of these compounds.Finally, the table obtained in this way will have the number of rows equal to the
number of all pairwise combinations of these cysteines and the number of columns will be equal to 521
(the first column includes the name of these pair combinations). And it is easy to divide this table into
training and testing data and predict the desired disulfide bonds between cysteines.
</p>


<h3>Value</h3>

<p>a table with number of all cysteine pairs in rows and 521 columns correspond to feature vector length.
</p>


<h3>References</h3>

<p>D.-J. Yu et al., &quot;Disulfide connectivity prediction based on modelled protein 3D structural information and
random forest regression,&quot; vol. 12, no. 3, pp. 611-621, 2014.
</p>
<p>N. J. Mapes Jr, C. Rodriguez, P. Chowriappa, S. J. C. Dua, and s. b. journal, &quot;Residue adjacency matrix based
feature engineering for predicting cysteine reactivity in proteins,&quot; vol. 17, pp. 90-100, 2019.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-disulfid(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='DP_PSSM'>DP_PSSM feature vector</h2><span id='topic+DP_PSSM'></span>

<h3>Description</h3>

<p>This feature results from the connection of two vectors. The vector is the first feature of a
vector with a length of 40, which calculates the average of positive and negative values for each column
separately and puts them together. in the second feature vector, correspond to each column the difference
between the numbers in the rows that have distance of k is calculated, and then the square average for the
differences that are positive is calculated, and the same action  for the differences that are negative is
performed. since k varies between 1 and <code class="reqn">\alpha</code>, and because the value of <code class="reqn">\alpha</code> in this function is equal to 2,
the length of the second feature vector will be 80, which by merging with the first feature vector, the total
feature vector of length 120 will be obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DP_PSSM(pssm_name, a = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DP_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM matrix file</p>
</td></tr>
<tr><td><code id="DP_PSSM_+3A_a">a</code></td>
<td>
<p>fixed parameter that user chooses which usually equals to 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 240
</p>


<h3>References</h3>

<p>Juan, E.Y., et al. (2009) Predicting Protein Subcellular Localizations for Gram-Negative Bacteria using DP-PSSM
and Support Vector Machines. Complex, Intelligent and Software Intensive Systems, 2009. CISIS'09. International
Conference on. IEEE, pp. 836-841.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-DP_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='dpc_pssm'>DPC_PSSM feature vector</h2><span id='topic+dpc_pssm'></span>

<h3>Description</h3>

<p>This Feature stands for dipeptide composition, to get this feature vector for both different
columns the elements in two consecutive rows corresponds to this columns, would be multiplied and this scenario
is done  for all L-1 consecutive rows (L is protein length) and finally summation of these numbers divides by
L-1. since the result depends on two different columns, eventually a Feature vector of length 400
will be obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dpc_pssm(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dpc_pssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400
</p>


<h3>References</h3>

<p>Liu, T., Zheng, X. and Wang, J. (2010) Prediction of protein structural class for
low-similarity sequences using support vector machine and PSI-BLAST profile, Biochimie, 92, 1330-1334.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-dpc_pssm(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))

</code></pre>

<hr>
<h2 id='dwt_PSSM'>discrete wavelet transform feature vector</h2><span id='topic+dwt_PSSM'></span>

<h3>Description</h3>

<p>In construction of this feature vector, the <code><a href="waveslim.html#topic+dwt.nondyadic">dwt.nondyadic</a></code> function is used from
&quot;waveslim&quot;, package
to calculate the discrete wavelet transform for each column of the PSSM matrix, which considers it as a
discrete signal. At last, 4 levels DWT is used to analysis of these discrete signals of PSSM (each column) and
extracted the PSSM-DWT feature from PSSM of protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dwt_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dwt_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of pssm Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 80
</p>


<h3>References</h3>

<p>Y. Wang, Y. Ding, F. Guo, L. Wei, and J. J. P. o. Tang, &quot;Improved detection of DNA-binding proteins
via compression technology on PSSM information,&quot; vol. 12, no. 9, 2017.
</p>
<p>Y. Wang, Y. Ding, J. Tang, Y. Dai, F. J. I. A. t. o. c. b. Guo, and bioinformatics, &quot;CrystalM: a
multi-view fusion approach for protein crystallization prediction,&quot; 2019.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-dwt_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='EDP_EEDP_MEDP'>EDP_EEDP_MEDP feature vector</h2><span id='topic+EDP_EEDP_MEDP'></span>

<h3>Description</h3>

<p>these are three feature vectors (EDP, EEDP, MEDP) which are used for prediction of protein
structural class for low-similarity sequences.at first ED-PSSM Matrix with 20*20 dimensions
is constructed from PSSM Matrix then by using this Matrix, EDP and EEDP vectors are
obtained eventually MEDP feature vector is obtained by fusing these vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EDP_EEDP_MEDP(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EDP_EEDP_MEDP_+3A_pssm_name">pssm_name</code></td>
<td>
<p>is name of PSSM Matrix  file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of three feature vectors (EDP, EEDP, MEDP)
</p>


<h3>References</h3>

<p>Zhang, L., Zhao, X. and Kong, L. (2014) Predict protein structural class for low-similarity sequences by evolutionary difference
information into the general form of Chou's pseudo amino acid composition, Journal of Theoretical Biology, 355, 105-110.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-EDP_EEDP_MEDP(paste0(system.file("extdata",package="PSSMCOOL"),"/C7GS61.txt.pssm"))
</code></pre>

<hr>
<h2 id='FPSSM'>D-FPSSM and SF-PSSM feature vectors</h2><span id='topic+FPSSM'></span>

<h3>Description</h3>

<p>This function produces list of two feature vectors named D-FPSSM and S-FPSSM
which then used by FPSSM2 function to construct feature vector of length 100 for each pair of
proteins which then used for protein-protein interaction prediction in each dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FPSSM(pssm_name, hk = 20)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FPSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="FPSSM_+3A_hk">hk</code></td>
<td>
<p>a parameter that indicates which amino acid alphabet must be used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>two feature vectors of different length which is used in later steps.
</p>


<h3>References</h3>

<p>Zahiri, J., et al. (2013) PPIevo: protein-protein interaction prediction from PSSM based
evolutionary information, Genomics, 102, 237-242.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-FPSSM(system.file("extdata","C7GQS7.txt.pssm",package="PSSMCOOL"),20)

</code></pre>

<hr>
<h2 id='FPSSM2'>Mixture of Two FPSSM Features</h2><span id='topic+FPSSM2'></span>

<h3>Description</h3>

<p>This function takes two PSSM files as argument and uses FPSSM function for making feature
vector of length 100 correspond to this pair of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FPSSM2(pssm_name1, pssm_name2, hk)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FPSSM2_+3A_pssm_name1">pssm_name1</code></td>
<td>
<p>The name of first PSSM Matrix file</p>
</td></tr>
<tr><td><code id="FPSSM2_+3A_pssm_name2">pssm_name2</code></td>
<td>
<p>The name of second PSSM Matrix file</p>
</td></tr>
<tr><td><code id="FPSSM2_+3A_hk">hk</code></td>
<td>
<p>a parameter that indicates which amino acid alphabet must be used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Feature vector of length 100
</p>


<h3>References</h3>

<p>Zahiri, J., et al. (2013) PPIevo: protein-protein interaction prediction from PSSM based
evolutionary information, Genomics, 102, 237-242.
</p>


<h3>See Also</h3>

<p><code><a href="entropy.html#topic+entropy">entropy</a></code>
</p>
<p><code><a href="infotheo.html#topic+mutinformation">mutinformation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> s1&lt;-system.file("extdata","C7GQS7.txt.pssm",package="PSSMCOOL")
 s2&lt;-system.file("extdata","C7GRQ3.txt.pssm",package="PSSMCOOL")
 s&lt;-FPSSM2(s1,s2,8)
</code></pre>

<hr>
<h2 id='grey_pssm_pseAAC'>grey pssm feature vector</h2><span id='topic+grey_pssm_pseAAC'></span>

<h3>Description</h3>

<p>This function produces a feature vector of length 100 which the first 20 components of this
vector is the normalized occurrence frequency of the native amino acids in the  protein. the next 20
components are mean of 20 PSSM columns and grey system model approach as elaborated in (Min et al. 2013)
is used to define the next 60 components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grey_pssm_pseAAC(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grey_pssm_pseAAC_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 100
</p>


<h3>References</h3>

<p>J.-L. Min, X. Xiao, and K.-C. J. B. r. i. Chou, &quot;iEzy-Drug: A web server for identifying the interaction
between enzymes and drugs in cellular networking,&quot; vol. 2013, 2013.
</p>
<p>X. Xiao, M. Hui, and Z. J. T. J. o. m. b. Liu, &quot;IAFP-Ense: an ensemble classifier for identifying
antifreeze protein by incorporating grey model and PSSM into PseAAC,&quot; vol. 249, no. 6, pp. 845-854, 2016.
</p>
<p>M. Kabir et al., &quot;Improving prediction of extracellular matrix proteins using evolutionary information via
a grey system model and asymmetric under-sampling technique,&quot; vol. 174, pp. 22-32, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-grey_pssm_pseAAC(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='k_mers'>3-mer and 2-mer in dataframe</h2><span id='topic+k_mers'></span>

<h3>Description</h3>

<p>This function produces all possible 2-mers or 3-mers by counting paths of length 2 or 3 in
a dataframe which is thought as a graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>k_mers(s, h)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="k_mers_+3A_s">s</code></td>
<td>
<p>a dataframe with 2 columns</p>
</td></tr>
<tr><td><code id="k_mers_+3A_h">h</code></td>
<td>
<p>is length of k-mer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>all k-mers by counting paths of length h in dataframe which is considered as a graph
</p>


<h3>Examples</h3>

<pre><code class='language-R'>s1&lt;-LETTERS[1:4]
s2&lt;-LETTERS[3:6]
s&lt;-data.frame(s1,s2)
dc&lt;-k_mers(s,3)
</code></pre>

<hr>
<h2 id='k_separated_bigrams_pssm'>k_separated_bigrams_pssm feature vector</h2><span id='topic+k_separated_bigrams_pssm'></span>

<h3>Description</h3>

<p>This feature is almost identical to the <code><a href="#topic+dpc_pssm">dpc_pssm</a></code> feature, and in fact the DPC feature is part of
this feature (for k=1) and for two different columns, considers rows that differ by the size of the unit k.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>k_separated_bigrams_pssm(pssm_name, k = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="k_separated_bigrams_pssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>is name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="k_separated_bigrams_pssm_+3A_k">k</code></td>
<td>
<p>a parameter that specifies separated length between amino acids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a feature vector of length 400
</p>


<h3>References</h3>

<p>Saini, H., et al.(2016) Protein Fold Recognition Using Genetic Algorithm Optimized Voting Scheme and Profile
Bigram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> X&lt;-k_separated_bigrams_pssm(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"),1)
</code></pre>

<hr>
<h2 id='kiderafactor'>kiderafactor feature</h2><span id='topic+kiderafactor'></span>

<h3>Description</h3>

<p>For product of this feature vector similar to smoothed_PSSM feature, firstly PSSM Matrix is
smoothed by appending zero vectors to its head and tail and utilizing sliding window of size odd, then this
smoothed PSSM Matrix is condensed by the Kidera factors to produce feature vector for each residue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kiderafactor(pssm_name, v = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kiderafactor_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="kiderafactor_+3A_v">v</code></td>
<td>
<p>vector of amino acids positions which we want to produce feature vector for them.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of feature vectors
</p>


<h3>References</h3>

<p>C. Fang, T. Noguchi, H. J. I. j. o. d. m. Yamana, and bioinformatics, &quot;Condensing position-specific scoring
matrixs by the Kidera factors for ligand-binding site prediction,&quot; vol. 12, no. 1, pp. 70-84, 2015.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+smoothed_PSSM">smoothed_PSSM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-kiderafactor(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"),c(2,3,8,9))
</code></pre>

<hr>
<h2 id='LPC_PSSM'>Linear predictive coding feature</h2><span id='topic+LPC_PSSM'></span>

<h3>Description</h3>

<p>This function uses Linear predictive coding algorithm for each column of PSSM Matrix
. so in this script <code><a href="phonTools.html#topic+lpc">lpc</a></code> function is used which produces a 14-dimensional
vector for each column, since PSSM has 20 column eventually it will be obtained a 20*14=280 dimensional
feature vector for each PSSM Matrix by this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LPC_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LPC_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 280
</p>


<h3>References</h3>

<p>L. Li et al., &quot;PSSP-RFE: accurate prediction of protein structural class by recursive feature extraction
from PSI-BLAST profile, physical-chemical property and functional annotations,&quot; vol. 9, no. 3, 2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-LPC_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='MBMGACPSSM'>MBMGACPSSM feature</h2><span id='topic+MBMGACPSSM'></span>

<h3>Description</h3>

<p>In this function three different autocorrelation descriptors based on PSSM are adopted,
which include: normalized Moreau-Broto autocorrelation, Moran autocorrelation and Geary autocorrelation
descriptors.Autocorrelation descriptor is a powerful statistical tool and defined based on the distribution
of amino acid properties along the sequence, which measures the correlation between two residues separated
by a distance of d in terms of their evolution scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MBMGACPSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MBMGACPSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 560
</p>


<h3>References</h3>

<p>Y. Liang, S. Liu, S. J. M. C. i. M. Zhang, and i. C. Chemistry, &quot;Prediction of protein structural class
based on different autocorrelation descriptors of position-specific scoring matrix,&quot;
vol. 73, no. 3, pp. 765-784, 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-MBMGACPSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='pse_pssm'>pseudo position-specific scoring matrix feature</h2><span id='topic+pse_pssm'></span>

<h3>Description</h3>

<p>This feature vector is combination of <code class="reqn">F_{PSSM}</code> feature vector and vector of
correlation factors correspond to 20 columns in PSSM Matrix. <code class="reqn">F_{PSSM}</code> actually is mean of PSSM Matrix
columns of length 20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pse_pssm(pssm_name, g = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pse_pssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>is the name of PSSM matrix file</p>
</td></tr>
<tr><td><code id="pse_pssm_+3A_g">g</code></td>
<td>
<p>a parameter Which its size corresponds to the database used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 20+20*g
</p>


<h3>References</h3>

<p>D.-J. Yu et al., &quot;Learning protein multi-view features in complex space,&quot; vol. 44, no. 5, pp. 1365-1379,
2013.
</p>
<p>Chou, K.C. and Shen, H.B. (2007) MemType-2L: a web server for predicting membrane proteins and their
types by incorporating evolution information through Pse-PSSM, Biochemical and Biophysical Research
Communications, 360, 339-345.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-pse_pssm(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='pssm_ac'>auto covariance transformation feature vector</h2><span id='topic+pssm_ac'></span>

<h3>Description</h3>

<p>The AC variable measures the correlation of the same property between two residues
separated by a distance of lg along the sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pssm_ac(pssm_name, lg = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pssm_ac_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of the PSSM Matrix file</p>
</td></tr>
<tr><td><code id="pssm_ac_+3A_lg">lg</code></td>
<td>
<p>a parameter which indicates distance between two residues</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector which its length depends on parameter lg. by default lg is 10 hence feature vector
would be of length 200.
</p>


<h3>Note</h3>

<p>in use of this function The lg parameter must be less than the length of the smallest sequence
in the database.
</p>


<h3>References</h3>

<p>Dong, Q., Zhou, S. and Guan, J. (2009) A new taxonomy-based protein fold recognition approach
based on autocross-covariance transformation, Bioinformatics, 25, 2655-2662.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-pssm_ac(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='pssm_cc'>Cross covarianse feature vector</h2><span id='topic+pssm_cc'></span>

<h3>Description</h3>

<p>The PSSM-CC variable measures the correlation of two different properties between two residues
separated by a distance of lg along the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pssm_cc(pssm_name, g = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pssm_cc_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="pssm_cc_+3A_g">g</code></td>
<td>
<p>shortest protein length in dataset minus one</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 3800
</p>


<h3>References</h3>

<p>Dong, Q., Zhou, S. and Guan, J. (2009) A new taxonomy-based protein fold recognition approach
based on autocross-covariance transformation, Bioinformatics, 25, 2655-2662.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-pssm_cc(system.file("extdata","C7GQS7.txt.pssm",package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='pssm_composition'>PSSM-COMPOSITION feature</h2><span id='topic+pssm_composition'></span>

<h3>Description</h3>

<p>This feature, which stands for auto covariance transformation, for jth column calculates the average
of this column, and then subtracts the resulting number from the elements on the i and (i + g)th rows of this column,
and finally multiplies them. by changing the variable i from 1 to L-g, it calculates the sum of these, since the
variable j changes between 1 and 20, and the variable g between 1 and 20 eventually a feature vector of length 200
will be obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pssm_composition(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pssm_composition_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix files</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400
</p>


<h3>References</h3>

<p>L. Zou, C. Nan, and F. J. B. Hu, &quot;Accurate prediction of bacterial type IV secreted effectors using amino acid
composition and PSSM profiles,&quot; vol. 29, no. 24, pp. 3135-3142, 2013.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-pssm_composition(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='PSSM_SD'>PSSM-SD feature</h2><span id='topic+PSSM_SD'></span>

<h3>Description</h3>

<p>In this feature, by considering a specific column, at first sum of all components in this column
is denoted by &quot;L&quot;, then starting from the first row in this column, the components are added together to
reaching a value less than or equal to 25
is calculated and stored . In the next step, the same work is done starting from the first row to reaching a
value less than or equal to 50
is done started from the last row, To reaching 25
By appending these saved numbers together for each column, a vector of length 4 is obtained. If this is done
for all the columns and the obtained vectors are connected to each other, for each protein, a feature vector
of length 80 is obtained which its name is PSSM-SD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSSM_SD(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PSSM_SD_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 80
</p>


<h3>References</h3>

<p>A. Dehzangi, K. Paliwal, J. Lyons, A. Sharma, A. J. I. A. T. o. C. B. Sattar, and Bioinformatics,
&quot;A segmentation-based method to extract structural and evolutionary features for protein fold
recognition,&quot; vol. 11, no. 3, pp. 510-519, 2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-PSSM_SD(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='pssm_seg'>PSSM-Seg feature vector</h2><span id='topic+pssm_seg'></span>

<h3>Description</h3>

<p>This feature vector uses PSSM-SD to produce Segmented Auto Covariance Features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pssm_seg(pssm_name, m = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pssm_seg_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="pssm_seg_+3A_m">m</code></td>
<td>
<p>a parameter between 1 and 11</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 100
</p>


<h3>References</h3>

<p>A. Dehzangi, K. Paliwal, J. Lyons, A. Sharma, A. J. I. A. T. o. C. B. Sattar, and Bioinformatics,
&quot;A segmentation-based method to extract structural and evolutionary features for protein fold
recognition,&quot; vol. 11, no. 3, pp. 510-519, 2014.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PSSM_SD">PSSM_SD</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>q&lt;-pssm_seg(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"),3)
</code></pre>

<hr>
<h2 id='pssm400'>PSSM400 feature</h2><span id='topic+pssm400'></span>

<h3>Description</h3>

<p>This function firstly normalizes PSSM Matrix by formula:
<code class="reqn">P-min(P)/max(P)-min(P)</code> then for any standard amino acid specifies its
position in protein sequence whereby a sub-matrix from PSSM corresponding
to these positions will be extracted, then for this sub-matrix computes
<code><a href="base.html#topic+colSums">colSums</a></code>
of its columns to create a vector of length 20, eventually a feature vector of
length 400 will be obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pssm400(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pssm400_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400
</p>


<h3>Note</h3>

<p>if a specific amino acid did not exist in protein then <code><a href="base.html#topic+colSums">colSums</a></code> of whole
PSSM is computed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-pssm400(system.file("extdata","C7GQS7.txt.pssm",package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='PSSMBLOCK'>PSSM BLOCK feature vector</h2><span id='topic+PSSMBLOCK'></span>

<h3>Description</h3>

<p>In this feature at first PSSM Matrix is divided to Blocks based on Number N which user imports. Then for each
Block mean of columns is computed to get 20-dimensional vector, eventually by appending these vectors to each other final feature
vector is obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSSMBLOCK(pssm_name, N = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PSSMBLOCK_+3A_pssm_name">pssm_name</code></td>
<td>
<p>neme of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="PSSMBLOCK_+3A_n">N</code></td>
<td>
<p>number of blocks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector that it's length depends on parameter N
</p>


<h3>References</h3>

<p>J.-Y. An, L. Zhang, Y. Zhou, Y.-J. Zhao, and D.-F. J. J. o. c. Wang, &quot;Computational methods using weighed-extreme learning
machine to predict protein self-interactions with protein evolutionary information,&quot; vol. 9, no. 1, p. 47, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-PSSMBLOCK(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"),5)
</code></pre>

<hr>
<h2 id='RPM_PSSM'>RPM-PSSM feature vector</h2><span id='topic+RPM_PSSM'></span>

<h3>Description</h3>

<p>In this feature The idea is similar to the probe concept used in microarray technologies, where probes are used to identify
genes. For the convenience, we call it residue probing method. In our application, each probe is an amino acid, which
corresponds to a particular column in the PSSM profiles. For each probe, we average the PSSM scores of all the amino acids
in the associated column with a PSSM value greater than zero in the sequence, which leads to a 1  20 feature vector. Once
again, for the 20 probes, the final feature for each protein sequence is a 1  400 vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RPM_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RPM_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>RPM-PSSM feature vector of length 400
</p>


<h3>References</h3>

<p>Jeong, J.C., Lin, X. and Chen, X.W. (2011) On position-specific scoring matrix for protein function prediction
, IEEE/ACM transactions on computational biology and bioinformatics / IEEE, ACM, 8, 308-315.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;- RPM_PSSM(system.file("extdata","C7GRQ3.txt.pssm",package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='rpssm'>RPSSM feature</h2><span id='topic+rpssm'></span>

<h3>Description</h3>

<p>To obtain this feature, first the columns of the PSSM matrix are merged to obtain an L*10 matrix. Then,
with a relationship similar to the auto covariance transformation feature, this feature with a length of 110 is
obtained from this matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rpssm(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rpssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 110
</p>


<h3>References</h3>

<p>Ding, S., et al. (2014) A protein structural classes prediction method based on predicted secondary structure and
PSI-BLAST profile, Biochimie, 97, 60-65.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-rpssm(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='scsh2'>SCSH Feature vector</h2><span id='topic+scsh2'></span>

<h3>Description</h3>

<p>This function gets a PSSM Matrix as input and extracts corresponding protein and consunsus
sequence from it. By placing these two vectors next to each other, a dataframe is created.
In each row of this dataframe, each component is connected to the next row components by an arrow. so a
directed graph is produced. next by use of previous functions a feature vector of length 400 or 8000 is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scsh2(pssm_name, k = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scsh2_+3A_pssm_name">pssm_name</code></td>
<td>
<p>is name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="scsh2_+3A_k">k</code></td>
<td>
<p>a parameter indicates length of k-mer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400 or 8000
</p>


<h3>References</h3>

<p>Zahiri, J., et al., LocFuse: human protein–protein interaction prediction via classifier fusion using protein
localization information. Genomics, 2014. 104(6): p. 496-503.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;- scsh2(system.file("extdata","C7GRQ3.txt.pssm",package="PSSMCOOL"),2)
</code></pre>

<hr>
<h2 id='single_Average'>single Average feature</h2><span id='topic+single_Average'></span>

<h3>Description</h3>

<p>This descriptor is a variant of the Average Block descriptor and is designed to group together rows
related to the same amino acid, thus considering domains of a sequence with similar conservation rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single_Average(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="single_Average_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 400
</p>


<h3>References</h3>

<p>L. Nanni, A. Lumini, and S. J. T. S. W. J. Brahnam, &quot;An empirical study of different approaches for protein classification,&quot;
vol. 2014, 2014.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Averag_Block">Averag_Block</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-single_Average(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='smoothed_PSSM'>smoothed PSSM feature</h2><span id='topic+smoothed_PSSM'></span>

<h3>Description</h3>

<p>In this function at first a Matrix called smoothed-PSSM is constructed from PSSM Matrix
by applying &quot;ws&quot; parameter which called sliding window size and taken from user and usually is equals to 7. Then
using other window size parameter &quot;w&quot; which usually equals to 11 at each position smoothed feature vector is
constructed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smoothed_PSSM(pssm_name, ws = 7, w = 50, v = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smoothed_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
<tr><td><code id="smoothed_PSSM_+3A_ws">ws</code></td>
<td>
<p>window size for smoothing PSSM Matrix</p>
</td></tr>
<tr><td><code id="smoothed_PSSM_+3A_w">w</code></td>
<td>
<p>window size for extracting feature vector</p>
</td></tr>
<tr><td><code id="smoothed_PSSM_+3A_v">v</code></td>
<td>
<p>vector of desired positions to extract their features</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the construction of a smoothed PSSM, each row vector of a residue <code class="reqn">\alpha_i</code> is represented and
smoothed by the summation of ws surrounding row vectors <code class="reqn">(V_{smoothed_i}=V_{i-(ws-1)/2}+...+V_i+...+
V_{i+(ws+1)/2})</code> For the N-terminal and C-terminal of a protein, (w-1)/2 ZERO vectors, are appended to the
head or tail of a smoothed PSSM profile. Using the smoothed PSSM encoding scheme the feature vector of a residue
<code class="reqn">\alpha_i</code> is represented by <code class="reqn">(V_{smoothed_i-(ws-1)/2},...,V_{smoothed_i},...,V_{smoothed_i+(ws+1)/2})</code>
The feature values in each vector are normalized to a range between -1 and 1.
</p>


<h3>Value</h3>

<p>a matrix of feature vectors
</p>


<h3>References</h3>

<p>Cheng, C.W., et al. (2008) Predicting RNA-binding sites of proteins using support vector machines and
evolutionary information, BMC Bioinformatics, 9 Suppl 12, S6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+kiderafactor">kiderafactor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-smoothed_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"),7,11,c(2,3,8,9))
</code></pre>

<hr>
<h2 id='SOMA_PSSM'>SOMA PSSM Feature</h2><span id='topic+SOMA_PSSM'></span>

<h3>Description</h3>

<p>In this function each column can be viewed as a stochastic time series, and each PSSM contains 20
columns, in other words, each PSSM contains 20 stochastic time series and Second-order moving average (SOMA)
algorithm is applied to these columns to extract SOMA PSSM feature vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SOMA_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SOMA_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 160
</p>


<h3>References</h3>

<p>Y. Liang, S. J. J. o. M. G. Zhang, and Modelling, &quot;Predict protein structural class by incorporating two different modes of evolutionary
information into Chou's general pseudo amino acid composition,&quot; vol. 78, pp. 110-117, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-SOMA_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='SVD_PSSM'>Singular Value Decomposition (SVD)</h2><span id='topic+SVD_PSSM'></span>

<h3>Description</h3>

<p>Singular value decomposition is a general purpose matrix factorization approach
that has many useful applications in signal processing and statistics. In this function SVD is
applied to a matrix representation of a protein with the aim of reducing its dimensionality
Given an input matrix Mat with dimensions N*M SVD is used to calculate its factorization
of the form: <code class="reqn">Mat=U\Sigma V,</code> where <code class="reqn">\Sigma</code> is a diagonal matrix whose diagonal
entries are known as the singular values of Mat. The resulting descriptor is the ordered
set of singular values: <code class="reqn">SVD\in\mathcal{R}^L,</code> where L=min(M,N).
and here <code><a href="base.html#topic+svd">svd</a></code> function is used for this purpose.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SVD_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SVD_PSSM_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of length 20
</p>


<h3>References</h3>

<p>L. Nanni, A. Lumini, and S. J. T. S. W. J. Brahnam, &quot;An empirical study of different approaches for protein classification,&quot;
vol. 2014, 2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-SVD_PSSM(system.file("extdata", "C7GQS7.txt.pssm", package="PSSMCOOL"))
</code></pre>

<hr>
<h2 id='three_mer'>3-Mer and 2-Mer</h2><span id='topic+three_mer'></span>

<h3>Description</h3>

<p>This function produces all possible k-mers from 20 amino acids for use in other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>three_mer(k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="three_mer_+3A_k">k</code></td>
<td>
<p>is length of k-mer which user imports</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix which its first row includes all k-mers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ax&lt;-three_mer(3)
</code></pre>

<hr>
<h2 id='trigrame_pssm'>trigrame feature vector</h2><span id='topic+trigrame_pssm'></span>

<h3>Description</h3>

<p>This feature vector is 8000-dimentional feature vector wich is computed from  tri-gram probability matrix
T obtained from PSSM Matrix.to achieve this purpose elements in three successive rows and arbitrary columns  are multiplied
together then these results are added together by changing variable i from 1 to L-1, which i is counter of row and L
indicates protein length. since there are 20 columns thus final feature vector would be of length 8000.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trigrame_pssm(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trigrame_pssm_+3A_pssm_name">pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>feature vector of lenght 8000
</p>


<h3>References</h3>

<p>Paliwal, K.K., et al. (2014) A tri-gram based feature extraction technique using linear probabilities of position
specific scoring matrix for protein fold recognition, IEEE transactions on nanobioscience, 13, 44-50
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-trigrame_pssm(paste0(system.file("extdata",package="PSSMCOOL"),"/C7GSI6.txt.pssm"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
