<!DOCTYPE html><html lang="en"><head><title>Help for package psSubpathway</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {psSubpathway}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compute_rank_score'><p>psSubpathway internal functions</p></a></li>
<li><a href='#DCSA'><p>Dynamic Changed Subpathway Analysis (DCSA)</p></a></li>
<li><a href='#envData'><p>The variables in the environment include subpathway list data, information of subpathway,an expression profile and a example result</p></a></li>
<li><a href='#FastSEAscore'><p>psSubpathway internal functions</p></a></li>
<li><a href='#getgenedensity'><p>psSubpathway internal functions</p></a></li>
<li><a href='#isPackageLoaded'><p>psSubpathway internal functions</p></a></li>
<li><a href='#ks_test_m'><p>psSubpathway internal functions</p></a></li>
<li><a href='#plotheatmap'><p>Plot a heatmap</p></a></li>
<li><a href='#plotSpwACmap'><p>Plot subpathway activity change map</p></a></li>
<li><a href='#plotSpwNetmap'><p>Polt a subpathway network map</p></a></li>
<li><a href='#plotSpwPSheatmap'><p>Plot subpathway phenotypic significant heat map</p></a></li>
<li><a href='#plotSubSEScurve'><p>Plot subtype set sample enrichment score curve graph</p></a></li>
<li><a href='#ReadClsFile'><p>psSubpathway internal functions</p></a></li>
<li><a href='#rndWalk'><p>psSubpathway internal functions</p></a></li>
<li><a href='#SEAscore'><p>psSubpathway internal functions</p></a></li>
<li><a href='#ssgsea'><p>psSubpathway internal functions</p></a></li>
<li><a href='#SubSEA'><p>Subtype Set Enrichment Analysis (SubSEA)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Flexible Identification of Phenotype-Specific Subpathways</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-3-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Xudong Han,
  Junwei Han,
  Qing Fei</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A network-based systems biology tool for flexible identification of phenotype-specific subpathways in the cancer gene expression data 
  with multiple categories (such as multiple subtype or developmental stages of cancer). Subtype Set Enrichment Analysis (SubSEA) and Dynamic Changed 
  Subpathway Analysis (DCSA) are developed to flexible identify subtype specific and dynamic changed subpathways respectively. The operation modes 
  include extraction of subpathways from biological pathways, inference of subpathway activities in the context of gene expression data, identification 
  of subtype specific subpathways with SubSEA, identification of dynamic changed subpathways associated with the cancer developmental stage with DCSA, 
  and visualization of the activities of resulting subpathways by using box plots and heat maps. Its capabilities render the tool could find the specific
  abnormal subpathways in the cancer dataset with multi-phenotype samples.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>GSVA, igraph, mpmi, pheatmap</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-18 08:05:23 UTC; 12859</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-18 08:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='compute_rank_score'>psSubpathway internal functions</h2><span id='topic+compute_rank_score'></span>

<h3>Description</h3>

<p>Compute rank score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>compute_rank_score
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='DCSA'>Dynamic Changed Subpathway Analysis (DCSA)</h2><span id='topic+DCSA'></span>

<h3>Description</h3>

<p>This function will perform the Dynamic Changed Subpathway Analysis (DCSA) method to estimate the dynamic
changed subpathways associated with the sample phenotypes (like the developmental stage of cancer).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DCSA(
  expr,
  input.cls = "",
  subpathwaylist = "Symbol",
  kcdf = "Gaussian",
  method = "gsva",
  min.sz = 5,
  max.sz = 1000,
  nperm = 100,
  fdr.th = 1,
  mx.diff = TRUE,
  parallel.sz = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DCSA_+3A_expr">expr</code></td>
<td>
<p>Matrix of gene expression values (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="DCSA_+3A_input.cls">input.cls</code></td>
<td>
<p>Input sample phenotype class vector file in CLS format.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_subpathwaylist">subpathwaylist</code></td>
<td>
<p>Character string denoting the gene label of the subpathway list is 'Entrezid' (default) or 'Symbol'.
Users can also enter their own subpathway list data. This list should be consistent with the gene label in the input gene expression profile.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_kcdf">kcdf</code></td>
<td>
<p>Character string denoting the kernel to use during the non-parametric estimation of the cumulative
distribution function of expression levels across samples when method=&quot;gsva&quot;. By default, 'kcdf=&quot;Gaussian&quot;' which
is suitable when input expression values are continuous, such as microarray fluorescent units in logarithmic scale,
RNA-seq log-CPMs, log-RPKMs or log-TPMs. When input expression values are integer counts, such as those derived from
RNA-seq experiments, then this argument should be set to 'kcdf=&quot;Poisson&quot;'.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_method">method</code></td>
<td>
<p>Method to employ in the estimation of subpathway enrichment scores per sample. By default, this is set to
'gsva' (HÃ¤nzelmann et al, 2013) and other options are 'ssgsea' (Barbie et al, 2009).</p>
</td></tr>
<tr><td><code id="DCSA_+3A_min.sz">min.sz</code></td>
<td>
<p>Minimum size of the resulting subpathway.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_max.sz">max.sz</code></td>
<td>
<p>Maximum size of the resulting subpathway.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations (default: 100).In practice, the users can set their own values as needed, and more than 1000 times may be fine in general.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_fdr.th">fdr.th</code></td>
<td>
<p>Cutoff value for FDR. The only subpathway with lower FDR.th are listed (default: 1).</p>
</td></tr>
<tr><td><code id="DCSA_+3A_mx.diff">mx.diff</code></td>
<td>
<p>Offers two approaches to calculate the sample enrichment score (SES) from the KS random walk statistic.
'mx.diff=FALSE': SES is calculated as the maximum distance of the random walk from 0. 'mx.diff=TRUE' (default): SES is
calculated as the magnitude difference between the largest positive and negative random walk deviations.</p>
</td></tr>
<tr><td><code id="DCSA_+3A_parallel.sz">parallel.sz</code></td>
<td>
<p>Number of processors to use when doing the calculations in parallel. If this argument is left with its
default value (parallel.sz=0) then it will use all available core processors unless we set this argument with a smaller number.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DCSA
</p>
<p>This function calculates the subpathway activity profile based on the gene expression profile and subpathway list by 'gsva' or 'ssgssea'.
Next, we used the information-theoretic measure of statistical dependence,
mutual information (MI), to estimate the dynamically changed subpathways associated with the sample phenotypes.
Finally we used the perturbation analysis of the gene label rearrangement to estimating the statistical significance.
</p>


<h3>Value</h3>

<p>A list containing the results of DCSA and subpathway activity profile.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load depend package.
require(GSVA)
require(parallel)
require(mpmi)
# get ACC disease stage gene expression profiling.
# ACCgenematrix&lt;-get("DCgenematrix")
# get path of the sample disease stage phenotype files.
# Stagelabels&lt;-system.file("extdata", "DClabels.cls", package = "psSubpathway")
# perform the DCSA method.
# DCSA(ACCgenematrix,input.cls=Stagelabels,nperm=50,fdr.th=0.01,parallel.sz=2)
# get the result of the SubSEA function
# DCSAresult&lt;-get("DCspwresult")
# str(DCSAresult)
# head(DCSAresult$DCSA)

# Simulated gene matrix.
genematrix &lt;- matrix(rnorm(500*40), nrow=500, dimnames=list(1:500, 1:40))
# Construct subpathway list data.
subpathwaylist &lt;- as.list(sample(2:100, size=20, replace=TRUE))
subpathwaylist &lt;- lapply(subpathwaylist, function(n) sample(1:500, size=n, replace=FALSE))
names(subpathwaylist)&lt;-c(paste(rep("spw",20),c(1:20)))
# Construct sample labels data.
stagelabel&lt;-list(phen=c("stage1","stage2","stage3","stage4"),
                   class.labes=c(rep("stage1",10),rep("stage2",10),
                   rep("stage3",10),rep("stage4",10)))
DCSAcs&lt;-DCSA(genematrix,stagelabel,subpathwaylist,nperm=0,parallel.sz=1)


</code></pre>

<hr>
<h2 id='envData'>The variables in the environment include subpathway list data, information of subpathway,an expression profile and a example result</h2><span id='topic+envData'></span><span id='topic+DCgenematrix'></span><span id='topic+DCspwresult'></span><span id='topic+spwtitle'></span><span id='topic+spwentrezidlist'></span><span id='topic+spwsymbollist'></span><span id='topic+spwnetworkdata'></span><span id='topic+Subgenematrix'></span><span id='topic+Subspwresult'></span><span id='topic+DClabes.cls'></span><span id='topic+Sublabels.cls'></span>

<h3>Description</h3>

<p>We used the k-clique algorithm to divide the human pathway of the KEGG database into subpathways and eliminated the smaller module that had a overlap above 80
These subpathway data will be divided into two parts: subpathway information(spwnetworkdata,spwtitle) and subpathway list data(spwentrezidlist: the gene Entrezid contained in each subpathway and spwsymbollist:the gene symbol contained in each subpathway).
Subgenematrix is the gene expression profile of the breast cancer we selected from GDCTCGA and Subspwresult is the result of applying Subgenematrix to Subtype Set Enrichment Analysis (SubSEA).In order to reduce the memory, we delete the gene in the gene expression profile that is not in the gene of the subpathway list.
We also selected the gene expression profile of the adrenocortical cancer (ACC) from GDCTCGA and performed Dynamic Changed Subpathway Analysis (DCSA). The data are DCgenematrix and DCspwresult
In order to reduce the memory, we delete the gene in the gene expression profile that is not in the gene of the subpathway list.
respectively. DClabes.cls and Sublabels.cls are the label vectors of the phenotypes of the samples of the two gene expression profiles.
</p>


<h3>Format</h3>

<p>An environment variable</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Xudong Han <a href="mailto:HanXD1217@163.com">HanXD1217@163.com</a></p>

<hr>
<h2 id='FastSEAscore'>psSubpathway internal functions</h2><span id='topic+FastSEAscore'></span>

<h3>Description</h3>

<p>Fast calculate phenotypic set enrichment score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>FastSEAscore
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='getgenedensity'>psSubpathway internal functions</h2><span id='topic+getgenedensity'></span>

<h3>Description</h3>

<p>Calculating the Kernel estimation of gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>getgenedensity
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='isPackageLoaded'>psSubpathway internal functions</h2><span id='topic+isPackageLoaded'></span>

<h3>Description</h3>

<p>Determine if the package is loaded, if no package is loaded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>isPackageLoaded
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='ks_test_m'>psSubpathway internal functions</h2><span id='topic+ks_test_m'></span>

<h3>Description</h3>

<p>Calculating subpathway Variation score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>ks_test_m
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='plotheatmap'>Plot a heatmap</h2><span id='topic+plotheatmap'></span>

<h3>Description</h3>

<p>Plot a heatmap of subpathway activity profile based on the parameters set by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotheatmap(
  inputdata,
  plotSubSEA = TRUE,
  fdr.th = 1,
  SES = "positive",
  phenotype = "all"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotheatmap_+3A_inputdata">inputdata</code></td>
<td>
<p>A list of result data generated by function 'SubSEA' or 'DCSA'.</p>
</td></tr>
<tr><td><code id="plotheatmap_+3A_plotsubsea">plotSubSEA</code></td>
<td>
<p>Determine the inputdata is the result data of function 'SubSEA' (default:plotSubSEA=TRUE) or
'DCSA' (plotSubSEA=FLASE).</p>
</td></tr>
<tr><td><code id="plotheatmap_+3A_fdr.th">fdr.th</code></td>
<td>
<p>Cutoff value for FDR. The only subpathway with lower FDR is plotted. (default: 1).</p>
</td></tr>
<tr><td><code id="plotheatmap_+3A_ses">SES</code></td>
<td>
<p>Parameter 'SES' is useful only when 'plotSubSEA' is TRUE. When 'plotSubSEA=TRUE',if 'SES' is positive,
the subpathway with high-expression will be plotted. when it is negative, plot low-expression subpathways.</p>
</td></tr>
<tr><td><code id="plotheatmap_+3A_phenotype">phenotype</code></td>
<td>
<p>Parameter 'phenotype' is useful only when 'plotSubSEA' is TRUE. 'phenotype' decides which phenotypic
significant subpathways to screen (which phenotypic result is applied to parameter 'fdr.th' and 'SES'.) and plot a
heat map of these subpathways.By default,'phenotype=&quot;all&quot;' which will screen the subpathways of all phenotypes and plot
a heat map. When the user wants to plot a subpathway heat map of the specified phenotype, this parameter should be set
to the name of the phenotype.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotheatmap
</p>
<p>The subpathways are screened according to the conditions set by the user and a heat map of the activity of these subpathways is drawn.
</p>


<h3>Value</h3>

<p>a heatmap
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load depend package.
library(pheatmap)
# get the Subspwresult which is the result of SubSEA function.
Subspwresult&lt;-get("Subspwresult")
# get the DCspwresult which is the result of DCSA function.
DCspwresult&lt;-get("DCspwresult")
# plot significant up-regulation subpathway heat map specific for each breast cancer subtype.
plotheatmap(Subspwresult,plotSubSEA=TRUE,fdr.th=0.01,SES="positive",phenotype="all")
# plot significant down-regulation subpathway heat map specific for each breast cancer subtype.
plotheatmap(Subspwresult,plotSubSEA=TRUE,fdr.th=0.01,SES="negative",phenotype="all")
# plot basal subtype specific significant subpathway heat map.
plotheatmap(Subspwresult,plotSubSEA=TRUE,fdr.th=0.01,SES="all",phenotype="Basal")
# plot adrenocortical cancer disease stages specific significant subpathway heat map.
plotheatmap(DCspwresult,plotSubSEA=FALSE,fdr.th=0.01)
</code></pre>

<hr>
<h2 id='plotSpwACmap'>Plot subpathway activity change map</h2><span id='topic+plotSpwACmap'></span>

<h3>Description</h3>

<p>Plot a box diagram and heat map of subpathway activity in each phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpwACmap(inputdata, spwid = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSpwACmap_+3A_inputdata">inputdata</code></td>
<td>
<p>A list of result data generated by function 'SubSEA' or 'DCSA'.</p>
</td></tr>
<tr><td><code id="plotSpwACmap_+3A_spwid">spwid</code></td>
<td>
<p>The subpathway id which the user wants to plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotSpwACmap
</p>
<p>Plot a box diagram of subpathway activity in each Phenotype and a heat map of the distribution of the phenotypic samples in the activity of the subpathways.
The subpathway activity change map includes subpathway active change box plot and subpathway active change. Each row in the heat map is all samples of a phenotype.
These samples are distributed in the subpathway high activity value area label is red, and the distribution in the low expression value area label is blue.
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the Subspwresult which is the result of SubSEA method.
Subspwresult&lt;-get("Subspwresult")
# plot the subpathway 00120_9 in the SubSEA function result.
plotSpwACmap(Subspwresult,spwid="00120_9")
# get the DCspwresult which is the result of DCSA method.
DCspwresult&lt;-get("DCspwresult")
# plot the subpathway 00982_2 in the DCSA function result.
plotSpwACmap(DCspwresult,spwid="00982_2")
</code></pre>

<hr>
<h2 id='plotSpwNetmap'>Polt a subpathway network map</h2><span id='topic+plotSpwNetmap'></span>

<h3>Description</h3>

<p>Visualize a subpathway network map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpwNetmap(
  spwid,
  layout = NULL,
  margin = 0,
  vertex.label.cex = 0.6,
  vertex.label.font = 1,
  vertex.size = 8,
  vertex.size2 = 6,
  edge.arrow.size = 0.2,
  edge.arrow.width = 3,
  edge.label.cex = 0.6,
  vertex.label.color = "black",
  vertex.color = "#BFFFBF",
  vertex.frame.color = "dimgray",
  edge.color = "dimgray",
  edge.label.color = "dimgray",
  sub = NULL,
  main = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSpwNetmap_+3A_spwid">spwid</code></td>
<td>
<p>The subpathway id which the user wants to plot.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_layout">layout</code></td>
<td>
<p>A matrix of x-y coordinates with two dims. Determine the placement
of the nodes for drawing a graph.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_margin">margin</code></td>
<td>
<p>A numeric. The value is usually between -0.5 and 0.5, which is able to zoom in or out a subpathway graph. The
default is 0.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.label.cex">vertex.label.cex</code></td>
<td>
<p>A numeric vector of node label size.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.label.font">vertex.label.font</code></td>
<td>
<p>A numeric vector of label font.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.size">vertex.size</code></td>
<td>
<p>A numeric vector of Node size. See <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.size2">vertex.size2</code></td>
<td>
<p>A numeric vector of Node size.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_edge.arrow.size">edge.arrow.size</code></td>
<td>
<p>Edge arrow size.The default is 0.2.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_edge.arrow.width">edge.arrow.width</code></td>
<td>
<p>Edge arrow width. The default is 3.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_edge.label.cex">edge.label.cex</code></td>
<td>
<p>Edge label size.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.label.color">vertex.label.color</code></td>
<td>
<p>A vector of node label colors. The default is black.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.color">vertex.color</code></td>
<td>
<p>A vector of node colors. The default is the KEGG node color.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_vertex.frame.color">vertex.frame.color</code></td>
<td>
<p>A vector of node frame color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_edge.color">edge.color</code></td>
<td>
<p>A vector of edge color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_edge.label.color">edge.label.color</code></td>
<td>
<p>A vector of edge label color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_sub">sub</code></td>
<td>
<p>A character string of subtitle.</p>
</td></tr>
<tr><td><code id="plotSpwNetmap_+3A_main">main</code></td>
<td>
<p>A character string of main title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotSpwNetmap
</p>
<p>The function plotSpwNetmap is able to display a subpathway graph.
The argument layout is used to determine the placement of the nodes
for drawing a graph.The layouts provided in igraph include 'layout_as_star', 'layout_as_tree',
'layout_in_circle', 'layout_nicely','layout_on_grid', 'layout_on_sphere', 'layout_randomly', 'layout_with_dh', 'layout_with_fr',
'layout_with_gem', 'layout_with_graphopt', 'layout_with_kk', 'layout_with_lgl', 'layout_with_mds'.
The 'layout_as_tree' generates a tree-like layout, so it is mainly for
trees. The 'layout_randomly' places the nodes randomly. The 'layout_in_circle' places
the nodes on a unit circle. Detailed information on the parameters can be found in <code><a href="igraph.html#topic+layout_">layout_</a></code>
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load depend package.
library(igraph)
# plot network graph of the subpathway 00982_2
plotSpwNetmap(spwid="00982_2",layout=layout_nicely)
</code></pre>

<hr>
<h2 id='plotSpwPSheatmap'>Plot subpathway phenotypic significant heat map</h2><span id='topic+plotSpwPSheatmap'></span>

<h3>Description</h3>

<p>Visualize subpathway activity significant heat map between phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpwPSheatmap(inputdata, spwid = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSpwPSheatmap_+3A_inputdata">inputdata</code></td>
<td>
<p>A list of result data generated by function 'SubSEA' or 'DCSA'.</p>
</td></tr>
<tr><td><code id="plotSpwPSheatmap_+3A_spwid">spwid</code></td>
<td>
<p>The subpathway id which the user wants to plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotSpwPSheatmap
</p>
<p>A heat map of the significance P-value of the activity of the subpathway between the phenotypes.
The rows and columns of the heat map are sample phenotype labels. The values shown in the heat map are the T-test
P values between the activity of the subpathway corresponding to the two phenotypes. The lower the number in the cells
in the heat map, the greater the difference in the activity of the subpathways between the two phenotypes.
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the Subspwresult which is the result of SubSEA method.
Subspwresult&lt;-get("Subspwresult")
# plot significant heat map between the activity of the subpathway in each subtype of breast cancer.
plotSpwPSheatmap(Subspwresult,spwid="00120_9")
# get the DCspwresult which is the result of DCSA method.
DCspwresult&lt;-get("DCspwresult")
# plot significant heat map between the activity of the subpathway in each stage of breast cancer.
plotSpwPSheatmap(DCspwresult,spwid="00982_2")
</code></pre>

<hr>
<h2 id='plotSubSEScurve'>Plot subtype set sample enrichment score curve graph</h2><span id='topic+plotSubSEScurve'></span>

<h3>Description</h3>

<p>Draw a sample enrichment score curve graph of a single or all subtypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSubSEScurve(inputdata, spwid = "", phenotype = "all")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSubSEScurve_+3A_inputdata">inputdata</code></td>
<td>
<p>A list of result data generated by function 'SubSEA' or 'DCSA'.</p>
</td></tr>
<tr><td><code id="plotSubSEScurve_+3A_spwid">spwid</code></td>
<td>
<p>The subpathway id which the user wants to plot.</p>
</td></tr>
<tr><td><code id="plotSubSEScurve_+3A_phenotype">phenotype</code></td>
<td>
<p>The 'phenotype&ldquo; specifies which phenotypic phenotype set enrichment curve is drawn for subpathway.
When 'phenotype=&quot;all&quot;' (default), the phenotype set enrichment score curve graph of subpathway under all phenotypes is drawn.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotSubSEScurve
</p>
<p>Plot a phenotype set enrichment score curve graph of a subpathway under all phenotypes or specified phenotype, including the
location of the maximum enrichment score (ES) and the leading-edge subset. This function can only be used for the results of
the 'SubSEA' function.
</p>


<h3>Value</h3>

<p>a plot
</p>
<p>an enrichment score curve graph
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the results of the SubSEA function for breast cancer subtypes.
Subspwresult&lt;-get("Subspwresult")
# plot enrichment score curve of the subpathway 00120_9 in all breast cancer subtypes.
plotSubSEScurve(Subspwresult,spwid="00120_9",phenotype="all")
# plot enrichment score curve of the subpathway 00120_9 in the basal breast cancer subtypes.
plotSubSEScurve(Subspwresult,spwid="00120_9",phenotype="Basal")
</code></pre>

<hr>
<h2 id='ReadClsFile'>psSubpathway internal functions</h2><span id='topic+ReadClsFile'></span>

<h3>Description</h3>

<p>These are functions read sample label file (.cls format).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>ReadClsFile
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='rndWalk'>psSubpathway internal functions</h2><span id='topic+rndWalk'></span>

<h3>Description</h3>

<p>Calculating random walks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>rndWalk
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='SEAscore'>psSubpathway internal functions</h2><span id='topic+SEAscore'></span>

<h3>Description</h3>

<p>Get subtype set enrichment score and sample locations, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSubSEScurve
</code></pre>


<h3>Details</h3>

<p>SEAscore
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='ssgsea'>psSubpathway internal functions</h2><span id='topic+ssgsea'></span>

<h3>Description</h3>

<p>Single sample GSEA (ssGSEA) calculates a gene set enrichment score
per sample as the normalized difference in empirical cumulative distribution
functions of gene expression ranks inside and outside the gene set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA

DCSA
</code></pre>


<h3>Details</h3>

<p>ssgsea
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>

<hr>
<h2 id='SubSEA'>Subtype Set Enrichment Analysis (SubSEA)</h2><span id='topic+SubSEA'></span>

<h3>Description</h3>

<p>The SubSEA (Subtype Set Enrichment Analysis) method to mine the specific subpathways of each sample Subtype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubSEA(
  expr,
  input.cls = "",
  subpathwaylist = "Symbol",
  kcdf = "Gaussian",
  method = "gsva",
  min.sz = 1,
  max.sz = Inf,
  nperm = 100,
  fdr.th = 1,
  mx.diff = TRUE,
  parallel.sz = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SubSEA_+3A_expr">expr</code></td>
<td>
<p>Matrix of gene expression values (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_input.cls">input.cls</code></td>
<td>
<p>Input sample class vector (phenotype) file in CLS format.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_subpathwaylist">subpathwaylist</code></td>
<td>
<p>Character string denoting the gene label of the subpahtway list is 'Entrezid' (default) or 'Symbol'.
Users can also enter their own subpathway list data. This list should be consistent with the gene label in the input gene expression profile.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_kcdf">kcdf</code></td>
<td>
<p>Character string denoting the kernel to use during the non-parametric estimation of the cumulative
distribution function of expression levels across samples when method=&quot;gsva&quot;. By default, 'kcdf=&quot;Gaussian&quot;' which
is suitable when input expression values are continuous, such as microarray fluorescent units in logarithmic scale,
RNA-seq log-CPMs, log-RPKMs or log-TPMs. When input expression values are integer counts, such as those derived from
RNA-seq experiments, then this argument should be set to 'kcdf=&quot;Poisson&quot;'.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_method">method</code></td>
<td>
<p>Method to employ in the estimation of subpathway enrichment scores per sample. By default,this is set to
'gsva' (HÃ¤nzelmann et al, 2013) and other options are 'ssgsea' (Barbie et al, 2009).</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_min.sz">min.sz</code></td>
<td>
<p>Minimum size of the resulting subpathway.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_max.sz">max.sz</code></td>
<td>
<p>Maximum size of the resulting subpathway.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations (default: 100). In practice, the users can set their own values as needed, and more than 1000 times may be fine in general.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_fdr.th">fdr.th</code></td>
<td>
<p>Cutoff value for FDR. The only subpathway with lower fdr.th are listed (default: 1).</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_mx.diff">mx.diff</code></td>
<td>
<p>Offers two approaches to calculate the sample enrichment score (SES) from the KS random walk statistic.
'mx.diff=FALSE': SES is calculated as the maximum distance of the random walk from 0. 'mx.diff=TRUE' (default): SES is
calculated as the magnitude difference between the largest positive and negative random walk deviations.</p>
</td></tr>
<tr><td><code id="SubSEA_+3A_parallel.sz">parallel.sz</code></td>
<td>
<p>Number of processors to use when doing the calculations in parallel. If this argument is left with its
default value (parallel.sz=0) then it will use all available core processors unless we set this argument with a smaller number.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>SubSEA
</p>
<p>This function calculates the subpathway activity profile based on the gene expression
profile and subpathway list by 'gsva' or 'ssgssea'. Then we calculate the sample enrichment score (SES) of each subpathway by Subtype Set
Enrichment Analysis (SubSEA).We permute the gene labels and recompute the SES for the permuted data, which generates
a null distribution for the SES.The P-value and the FDR value are calculated according to the perturbation analysis.
</p>


<h3>Value</h3>

<p>A list containing the results of the SubSEA and the subpathway activity profile.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Qingfei Kong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load depend package.
require(GSVA)
require(parallel)
# get breast cancer disease subtype gene expression profile.
Bregenematrix&lt;-get("Subgenematrix")
# get path of the sample disease subtype files.
Subtypelabels&lt;- system.file("extdata", "Sublabels.cls", package = "psSubpathway")
# SubSEA(Bregenematrix,input.cls=Subtypelabels,nperm=50,fdr.th=0.01,parallel.sz=2)
# get the result of the SubSEA function
SubSEAresult&lt;-get("Subspwresult")
str(SubSEAresult)
head(SubSEAresult$Basal)

# Simulated gene matrix
genematrix &lt;- matrix(rnorm(500*40), nrow=500, dimnames=list(1:500, 1:40))
# Construct subpathway list data.
subpathwaylist &lt;- as.list(sample(2:100, size=20, replace=TRUE))
subpathwaylist &lt;- lapply(subpathwaylist, function(n) sample(1:500, size=n, replace=FALSE))
names(subpathwaylist)&lt;-c(paste(rep("spw",20),c(1:20)))
# Construct sample labels data.
subtypelabel&lt;-list(phen=c("subtype1","subtype2","subtype3","subtype4"),
                   class.labes=c(rep("subtype1",10),rep("subtype2",10),
                                 rep("subtype3",10),rep("subtype4",10)))
SubSEAcs&lt;-SubSEA(genematrix,subtypelabel,subpathwaylist,nperm=0,parallel.sz=1)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
