<!DOCTYPE html><html><head><title>Help for package ptest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ptest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alpha'><p>Microarray time series experiment for yeast cell cycle from alpha experiment</p></a></li>
<li><a href='#B1'><p>Benchmark set B1</p></a></li>
<li><a href='#B2'><p>Benchmark set B2</p></a></li>
<li><a href='#B3'><p>Benchmark set B3</p></a></li>
<li><a href='#Cc'><p>Microarray time series experiment for Caulobacter crescentus bacterial cell cycle</p></a></li>
<li><a href='#cdc15'><p>Microarray time series experiment for yeast cell cycle from cdc15 experiment</p></a></li>
<li><a href='#cdc28'><p>Microarray time series experiment for yeast cell cycle from cdc28 experiment</p></a></li>
<li><a href='#FisherGTest'><p>Fisher's g test</p></a></li>
<li><a href='#fitHReg'><p>Fits Three Parameter Harmonic Regression</p></a></li>
<li><a href='#GetFitExtendedG'><p>Compute the extended Fisher's g test statistic</p></a></li>
<li><a href='#GetFitFisherG'><p>Compute the Fisher's g test statistic</p></a></li>
<li><a href='#GetFitHReg'><p>Compute loglikelihood ratio test statistic</p></a></li>
<li><a href='#GetFitHRegL1'><p>Compute loglikelihood ratio test statistic under the laplace noises</p></a></li>
<li><a href='#GetFitRobustG'><p>Compute the Robust g test statistic</p></a></li>
<li><a href='#pFisherg'><p>The cumulative distribution function for Fisher'g</p></a></li>
<li><a href='#pgram'><p>Periodogram computation</p></a></li>
<li><a href='#predictRSR'><p>Quantile computation for the RSR model</p></a></li>
<li><a href='#ptestg'><p>Test short time series for periodicity based on periodograms</p></a></li>
<li><a href='#ptestReg'><p>Test short time series for periodicity with maximum likelihood ratio tests</p></a></li>
<li><a href='#pvalueRSR'><p>P-value computation for the RSR model</p></a></li>
<li><a href='#rankSpectrum'><p>The robust rank-based spectral</p></a></li>
<li><a href='#simAR1'><p>Simulate AR(1) series</p></a></li>
<li><a href='#simHReg'><p>Simulate harmonic regression models</p></a></li>
<li><a href='#simLaplace'><p>Simulate series with the Laplace distribution</p></a></li>
<li><a href='#tableEg'><p>The result for the RSR method with respect to</p>
the extended Fisher's G test</a></li>
<li><a href='#tablegEven'><p>The result for the RSR method with respect to</p>
the Fisher'g test (Even series length)</a></li>
<li><a href='#tablegOdd'><p>The result for the RSR method with respect to</p>
the Fisher'g test (Odd series length)</a></li>
<li><a href='#tableRegL1Even'><p>The result for the RSR method with respect to</p>
the laplace likelihood ratio test (Even series length)</a></li>
<li><a href='#tableRegL1Odd'><p>The result for the RSR method with respect to</p>
the laplace likelihood ratio test (Odd series length)</a></li>
<li><a href='#tableRegLs'><p>The result for the RSR method with respect to</p>
the normal likelihood ratio test</a></li>
<li><a href='#tableRgAn'><p>The result for the RSR method with respect to</p>
the robust g test</a></li>
<li><a href='#tableRgEn'><p>The result for the RSR method with respect to</p>
the robust g test with frequency set &quot;En&quot;</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Periodicity Tests in Short Time Series</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-11-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuanhao Lai and A.I. McLeod</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>A.I. McLeod &lt;aimcleod@uwo.ca&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0),</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements p-value computations using an approximation to the cumulative distribution function for a variety of tests for periodicity. These tests include harmonic regression tests with normal and double exponential errors as well as modifications of Fisher's g test. An accompanying vignette illustrates the application of these tests.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>quantreg (&ge; 5.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, boot, lattice, rmarkdown, GeneCycle,</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-11-12 17:42:20 UTC; IanMcLeod</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-11-12 21:41:37</td>
</tr>
</table>
<hr>
<h2 id='alpha'>Microarray time series experiment for yeast cell cycle from alpha experiment</h2><span id='topic+alpha'></span>

<h3>Description</h3>

<p>6,178 yeast genes expression measures (log-ratios) with 
series length 18 from the alpha factor experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alpha)
</code></pre>


<h3>Format</h3>

<p>Matrix with 6178 rows and 18 columns.
Some missing data.
Rows and columns are labelled.
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:6178] &quot;YAL001C&quot; &quot;YAL002W&quot; &quot;YAL003W&quot; &quot;YAL004W&quot; ...
..$ : chr [1:18] &quot;alpha0&quot; &quot;alpha7&quot; &quot;alpha14&quot; &quot;alpha21&quot; ...</p>


<h3>Source</h3>

<p>The data is extracted from the ExpressionSet of 
the R package <code>yeastCC</code>.
</p>


<h3>References</h3>

<p>Spellman, P. T., Sherlock, G., Zhang, M. Q., Iyer, V. R.,
Anders, K., Eisen, M. B., ... &amp; Futcher, B. (1998). 
Comprehensive identification of cell cycle-regulated genes of 
the yeast Saccharomyces cerevisiae by microarray hybridization. 
Molecular biology of the cell, 9(12), 3273-3297.
</p>
<p>Dudoit S (2016). yeastCC: Spellman et al. (1998) and 
Pramila/Breeden (2006) yeast cell cycle microarray data. 
R package version 1.12.0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alpha)
qqnorm(colMeans(alpha, na.rm=TRUE))
qqnorm(rowMeans(alpha, na.rm=TRUE))
</code></pre>

<hr>
<h2 id='B1'>Benchmark set B1</h2><span id='topic+B1'></span>

<h3>Description</h3>

<p>List for yeast genes which are <b>most</b> likely to be periodic 
(the benchmark set 1 in de Lichtenberg et al. (2005)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(B1)
</code></pre>


<h3>Format</h3>

<p>A vector containg 113 genes' names.</p>


<h3>Details</h3>

<p>A total of 113 genes previously identified as periodically
expressed in small-scale experiments. The set encompasses the
104 genes used by Spellman et al. (1998) and nine genes added
by Johansson et al. (2003).
</p>


<h3>Source</h3>

<p>The raw data can be downloaded from 
<a href="http://www.cbs.dtu.dk/cellcycle/yeast_benchmark/benchmark.php">http://www.cbs.dtu.dk/cellcycle/yeast_benchmark/benchmark.php</a>.
</p>


<h3>References</h3>

<p>De Lichtenberg, U., Jensen, L. J., Fausboll, A., 
Jensen, T. S., Bork, P.,&amp; Brunak, S. (2005). 
Comparison of computational methods for the identification 
of cell cycle-regulated genes. Bioinformatics, 21(7), 1164-1171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alpha)
data(B1)
alphaB1 &lt;- alpha[rownames(alpha) \%in\% B1, ]

</code></pre>

<hr>
<h2 id='B2'>Benchmark set B2</h2><span id='topic+B2'></span>

<h3>Description</h3>

<p>List for yeast genes which are <b>most</b> likely to be periodic 
(the benchmark set 2 in de Lichtenberg et al. (2005)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(B2)
</code></pre>


<h3>Format</h3>

<p>A vector containg 352 genes' names.</p>


<h3>Details</h3>

<p>Genes whose promoters were bound (P-value below 0.01) by at
least one of nine known cell cycle transcription factors in both
of the Chromatin IP studies by Simon et al. (2001) and Lee et al.
(2002). To obtain a benchmark set that is independent of B1, we
removed all genes included in B1 (50). The resulting benchmark
set, B2, consists of 352 genes of which many should be expected
to be cell cycle regulated, since their promoters are associated
with known stage-specific cell cycle transcription factors.
</p>


<h3>Source</h3>

<p>The raw data can be downloaded from 
<a href="http://www.cbs.dtu.dk/cellcycle/yeast_benchmark/benchmark.php">http://www.cbs.dtu.dk/cellcycle/yeast_benchmark/benchmark.php</a>.
</p>


<h3>References</h3>

<p>De Lichtenberg, U., Jensen, L. J., Fausboll, A., 
Jensen, T. S., Bork, P.,&amp; Brunak, S. (2005). 
Comparison of computational methods for the identification 
of cell cycle-regulated genes. Bioinformatics, 21(7), 1164-1171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alpha)
data(B2)
alphaB2 &lt;- alpha[rownames(alpha)  \%in\% B2,]

</code></pre>

<hr>
<h2 id='B3'>Benchmark set B3</h2><span id='topic+B3'></span>

<h3>Description</h3>

<p>List for yeast genes which are <b>less</b> likely to be periodic
(the benchmark set 3 in de Lichtenberg et al. (2005)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(B3)
</code></pre>


<h3>Format</h3>

<p>A vector containg 518 genes' names.</p>


<h3>Details</h3>

<p>Genes annotated in MIPS (Mewes et al., 2002) as 'cell cycle
and DNA processing'. From these, we removed genes annotated
specifically as 'meiosis' and genes included in B1 (67), leaving
518 genes. As a large number of genes involved in the cell cycle
are not subject to transcriptional regulation (not periodic), and
because B1 was explicitly removed, a relatively small fraction
of these genes should be expected to be periodically expressed.
</p>


<h3>Source</h3>

<p>The raw data can be downloaded from 
<a href="http://www.cbs.dtu.dk/cellcycle/yeast_benchmark/benchmark.php">http://www.cbs.dtu.dk/cellcycle/yeast_benchmark/benchmark.php</a>.
</p>


<h3>References</h3>

<p>De Lichtenberg, U., Jensen, L. J., Fausboll, A., 
Jensen, T. S., Bork, P.,&amp; Brunak, S. (2005). 
Comparison of computational methods for the identification 
of cell cycle-regulated genes. Bioinformatics, 21(7), 1164-1171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alpha)
data(B3)
alphaB3 &lt;- alpha[rownames(alpha)  \%in\% B3,]

</code></pre>

<hr>
<h2 id='Cc'>Microarray time series experiment for Caulobacter crescentus bacterial cell cycle</h2><span id='topic+Cc'></span>

<h3>Description</h3>

<p>In this microarray experiment there are 3062 genes measured every 1 hour.
There are 19 is missing gene labels and these have been given 
labels ORFna1,...,ORFna19.
There 310 with duplicate labels.
Of these duplicate labels, 295 are duplicated twice, 12 are duplicated 3 times
and 3 are duplicated 4 times.
Duplicate labels are renamed ORF... to ORF...a and ORF...b etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Cc)
</code></pre>


<h3>Format</h3>

<p>Matrix with 3062 rows and 11 columns.
Some missing data.
Rows and columns are labelled.
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:3062] &quot;ORF06244a&quot; &quot;ORF03152a&quot; &quot;ORF03156a&quot; &quot;ORF03161a&quot; ...
..$ : chr [1:11] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...</p>


<h3>Details</h3>

<p>Gene expression from synchronized cultures of the bacterium 
Caulobacter crescentus (Laub et al., 2000).
(Laub et al., 2000) identified 553 genes whose
messenger RNA levels varied as a function of the cell cycle but their
statistical analysis was not very sophisticated and they probably 
identified too many genes.
Wichert et al. (2004) found that 44 genes were found which displayed
periodicity based on the Fisher's g-test using a FDR with q=0.05.
</p>


<h3>References</h3>

<p>Laub, M.T., McAdams,H.H., Feldblyum,T., 
Fraser,C.M. and Shapiro,L. (2000) Global analysis of the genetic 
network controlling a bacterial cell cycle Science,
290, 2144-2148.
</p>
<p>Wichert,S., Fokianos K. and Strimmer K. (2004) 
Identifying periodically expressed
transcrips in microarray time series data. Bioinformatics, 18, 5-20.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cc)
qqnorm(colMeans(Cc, na.rm=TRUE))
qqnorm(rowMeans(Cc, na.rm=TRUE))
</code></pre>

<hr>
<h2 id='cdc15'>Microarray time series experiment for yeast cell cycle from cdc15 experiment</h2><span id='topic+cdc15'></span>

<h3>Description</h3>

<p>6,178 yeast genes expression measures (log-ratios) with 
series length 24 from the cdc15 experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cdc15)
</code></pre>


<h3>Format</h3>

<p>Matrix with 6178 rows and 24 columns.
Some missing data.
Rows and columns are labelled.
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:6178] &quot;YAL001C&quot; &quot;YAL002W&quot; &quot;YAL003W&quot; &quot;YAL004W&quot; ...
..$ : chr [1:24] &quot;cdc15.10&quot; &quot;cdc15.30&quot; &quot;cdc15.50&quot; &quot;cdc15.70&quot; ...</p>


<h3>Source</h3>

<p>The data is extracted from the ExpressionSet of 
the R package <code>yeastCC</code>.
</p>


<h3>References</h3>

<p>Spellman, P. T., Sherlock, G., Zhang, M. Q., Iyer, 
V. R., Anders, K., Eisen, M. B., ... &amp; Futcher, B. (1998). 
Comprehensive identification of cell cycle-regulated genes of 
the yeast Saccharomyces cerevisiae by microarray hybridization. 
Molecular biology of the cell, 9(12), 3273-3297.
</p>
<p>Dudoit S (2016). yeastCC: Spellman et al. (1998) and 
Pramila/Breeden (2006) yeast cell cycle microarray data. 
R package version 1.12.0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cdc15)
qqnorm(colMeans(cdc15, na.rm=TRUE))
qqnorm(rowMeans(cdc15, na.rm=TRUE))
</code></pre>

<hr>
<h2 id='cdc28'>Microarray time series experiment for yeast cell cycle from cdc28 experiment</h2><span id='topic+cdc28'></span>

<h3>Description</h3>

<p>6,178 yeast genes expression measures (log-ratios) with series 
length 17 from the cdc28 experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cdc28)
</code></pre>


<h3>Format</h3>

<p>Matrix with 6178 rows and 17 columns.
Some missing data.
Rows and columns are labelled.
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:6178] &quot;YAL001C&quot; &quot;YAL002W&quot; &quot;YAL003W&quot; &quot;YAL004W&quot; ...
..$ : chr [1:17] &quot;cdc28.0&quot; &quot;cdc28.10&quot; &quot;cdc28.20&quot; &quot;cdc28.30&quot; ...</p>


<h3>Source</h3>

<p>The data is extracted from the ExpressionSet of 
the R package <code>yeastCC</code>.
</p>


<h3>References</h3>

<p>Spellman, P. T., Sherlock, G., Zhang, M. Q., Iyer, V. R.,
Anders, K., Eisen, M. B., ... &amp; Futcher, B. (1998). 
Comprehensive identification of cell cycle-regulated genes of 
the yeast Saccharomyces cerevisiae by microarray hybridization. 
Molecular biology of the cell, 9(12), 3273-3297.
</p>
<p>Dudoit S (2016). yeastCC: Spellman et al. (1998) and 
Pramila/Breeden (2006) yeast cell cycle microarray data. 
R package version 1.12.0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cdc28)
qqnorm(colMeans(cdc28, na.rm=TRUE))
qqnorm(rowMeans(cdc28, na.rm=TRUE))
</code></pre>

<hr>
<h2 id='FisherGTest'>Fisher's g test</h2><span id='topic+FisherGTest'></span>

<h3>Description</h3>

<p>Calculates p-value using Fisher's g-test for a single frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FisherGTest(z)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FisherGTest_+3A_z">z</code></td>
<td>
<p>Input series, no missing values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Program is interfaced to C for efficient computation.
</p>


<h3>Value</h3>

<p>Vector of length 3 containing the g test statistic, the p-value, 
and the frequency for the maximum periodogram.
</p>


<h3>Author(s)</h3>

<p>A.I. McLeod
</p>


<h3>References</h3>

<p>Fisher, R.A. (1929). Tests of significance in harmonic analysis. 
Proc. Roy. Soc. A, 125, 54-59.
</p>

<hr>
<h2 id='fitHReg'>Fits Three Parameter Harmonic Regression</h2><span id='topic+fitHReg'></span>

<h3>Description</h3>

<p>Estimates A, B and f in the harmonic regression,
y(t)=mu+A*cos(2*pi*f*t)+B*sin(2*pi*f*t)+e(t). 
The default algorithm is enumerative but an exact non-linear LS
option is also provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitHReg(y, t = 1:length(y), algorithm = c("enumerative", "exact"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitHReg_+3A_y">y</code></td>
<td>
<p>series.</p>
</td></tr>
<tr><td><code id="fitHReg_+3A_t">t</code></td>
<td>
<p>Time points.</p>
</td></tr>
<tr><td><code id="fitHReg_+3A_algorithm">algorithm</code></td>
<td>
<p>method for the optimization</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Program is interfaced to C for efficient computation.
</p>


<h3>Value</h3>

<p>Object of class &quot;HReg&quot; produced.
</p>


<h3>Author(s)</h3>

<p>A.I. McLeod and Yuanhao Lai
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(193)
z &lt;- simHReg(10,f=2.5/10,1,1)
ans &lt;- fitHReg(z)
ans$freq #optimal frequency = 0.2376238 
#
#ORF06806 in Cc dataset. 
z&lt;-c(0.42, 0.89, 1.44, 1.98, 2.21, 2.04, 0.82, 0.62, 0.56, 0.8, 1.33)
ans2 &lt;- fitHReg(z, algorithm="exact")
sum(resid(ans2)^2) #0.2037463
ans1 &lt;- fitHReg(z)
sum(resid(ans1)^2) #0.242072
#compare with nls()
t &lt;- 1:length(z)
ans &lt;- nls(z ~ mu+alpha*cos(2*pi*lambda*t+phi), 
              start=list(mu=1, alpha=1, lambda=0.1, phi=0.0))
coefficients(ans)
sum(resid(ans)^2) #0.2037

                                                        
</code></pre>

<hr>
<h2 id='GetFitExtendedG'>Compute the extended Fisher's g test statistic</h2><span id='topic+GetFitExtendedG'></span>

<h3>Description</h3>

<p>The extended Fisher's g te  statistic is computed for testing for periodicity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFitExtendedG(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetFitExtendedG_+3A_y">y</code></td>
<td>
<p>Vector containing the series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extend the Fisher's g test by enlarging the searching region 
of the frequency from the fourier frequencies to be 
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
</p>


<h3>Value</h3>

<p>Vector of length 2 containing the modified Fisher's G test statistic 
and the frequency for the maximum periodogram.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>


<h3>References</h3>

<p>Fisher, R.A. (1929). Tests of significance in harmonic analysis. 
Proc. Roy. Soc. A, 125, 54-59.
</p>

<hr>
<h2 id='GetFitFisherG'>Compute the Fisher's g test statistic</h2><span id='topic+GetFitFisherG'></span>

<h3>Description</h3>

<p>The Fisher's g test statistic is computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFitFisherG(z)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetFitFisherG_+3A_z">z</code></td>
<td>
<p>Vector containing the series.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of length 2 containing the modified Fisher's G test statistic 
and the frequency for the maximum periodogram.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>


<h3>References</h3>

<p>Fisher, R.A. (1929). Tests of significance in harmonic analysis. 
Proc. Roy. Soc. A, 125, 54-59.
</p>

<hr>
<h2 id='GetFitHReg'>Compute loglikelihood ratio test statistic</h2><span id='topic+GetFitHReg'></span>

<h3>Description</h3>

<p>The -2 loglikelihood ratio test statistic, -2LLR, is computed 
for testing for periodicity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFitHReg(y, t = 1:length(y))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetFitHReg_+3A_t">t</code></td>
<td>
<p>Vector of corresponding time points.</p>
</td></tr>
<tr><td><code id="GetFitHReg_+3A_z">z</code></td>
<td>
<p>Vector containing the series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To compute the likelihood ratio statistic, 
a harmonic regression model is fitted by 
the maximum likelihood estimations (MLE)
according to the selected model. In particular, the frequency f is found by 
the grid search among 
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
The MLE is equivalent to 
the least square estimation.
</p>


<h3>Value</h3>

<p>Vector of length 2 containing -2 loglikelihood ratio 
test statistic and the estimated frequency.
</p>


<h3>Author(s)</h3>

<p>A.I. McLeod
</p>


<h3>References</h3>

<p>Islam, M.S. (2008).  Peridocity, 
Change Detection and Prediction in Microarrays. 
Ph.D. Thesis, The University of Western Ontario.
</p>

<hr>
<h2 id='GetFitHRegL1'>Compute loglikelihood ratio test statistic under the laplace noises</h2><span id='topic+GetFitHRegL1'></span>

<h3>Description</h3>

<p>The -2 loglikelihood ratio test statistic, -2LLR, is computed 
for testing for periodicity under the laplace noises.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFitHRegL1(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetFitHRegL1_+3A_y">y</code></td>
<td>
<p>Vector containing the series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To compute the likelihood ratio statistic, 
a harmonic regression model is fitted by 
the maximum likelihood estimations (MLE)
according to the selected model. In particular, the frequency f is found by 
the grid search among 
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
The MLE is equivalent to 
the least absolute estimation. 
The computation is completed by the Exterior Point Methods 
with theimported function <code><a href="quantreg.html#topic+rq.fit.br">rq.fit.br</a></code> 
from the package <code>quantreg</code>.
</p>


<h3>Value</h3>

<p>Vector of length 2 containing 
the -2 loglikelihood ratio test statistic under the laplace noises 
and the estimated frequency.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>


<h3>References</h3>

<p>Li, T. H. (2010). A nonlinear method for robust spectral analysis. 
Signal Processing, IEEE Transactions on, 58(5), 2466-2474.
</p>

<hr>
<h2 id='GetFitRobustG'>Compute the Robust g test statistic</h2><span id='topic+GetFitRobustG'></span>

<h3>Description</h3>

<p>The Robust g test statistic is computed for testing for periodicity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFitRobustG(y, freqSet = c("An", "En"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetFitRobustG_+3A_y">y</code></td>
<td>
<p>Vector containing the series.</p>
</td></tr>
<tr><td><code id="GetFitRobustG_+3A_freqset">freqSet</code></td>
<td>
<p>The set to search frequencies.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>An</code> refers to searching frequencies in the range 
twice as dense as the region of the fourier frequencies, i.e.,
<code class="reqn">An = {0,1/n,\ldots,floor((2 n - 1)/2)/n}</code>.
<code>En</code> refers to searching frequencies in 
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
</p>


<h3>Value</h3>

<p>Vector of length 2 containing the Robust g test statistic 
and the frequency for the maximum periodogram.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>


<h3>References</h3>

<p>Ahdesmaki, M., Lahdesmaki, H., Pearson, R., Huttunen, H., 
and Yli-Harja O.(2005). 
<em>BMC Bioinformatics</em> <b>6</b>:117. 
<a href="http://www.biomedcentral.com/1471-2105/6/117">http://www.biomedcentral.com/1471-2105/6/117</a>
</p>

<hr>
<h2 id='pFisherg'>The cumulative distribution function for Fisher'g</h2><span id='topic+pFisherg'></span>

<h3>Description</h3>

<p>Compute the the cumulative distribution function for Fisher'g
from the selected method. See details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pFisherg(g, n, method = c("exact", "RSR"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pFisherg_+3A_g">g</code></td>
<td>
<p>Fisher'g statistic.</p>
</td></tr>
<tr><td><code id="pFisherg_+3A_n">n</code></td>
<td>
<p>length of the series.</p>
</td></tr>
<tr><td><code id="pFisherg_+3A_method">method</code></td>
<td>
<p>method to compute the distribution function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>method provides the following choices:
</p>
<p><code>exact</code>: Use the exact distribution.
</p>
<p><code>RSR</code>: Distribution fitted by the response surface regression method.
</p>


<h3>Value</h3>

<p>the distribution function.
</p>


<h3>Author(s)</h3>

<p>Yuanhao
</p>


<h3>References</h3>

<p>Fisher, R.A. (1929). Tests of significance in harmonic analysis. 
Proc. Roy. Soc. A, 125, 54-59. 
</p>
<p>MacKinnon, James (2001) : Computing numerical distribution functions in 
econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ptestg">ptestg</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Given the Fisher's g statistic, find the cumulative probability
pFisherg(g=0.3,n=10, method = "exact")
pFisherg(g=0.3,n=10, method = "RSR")

</code></pre>

<hr>
<h2 id='pgram'>Periodogram computation</h2><span id='topic+pgram'></span>

<h3>Description</h3>

<p>The periodogram is computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgram(z, fr = "default", method = c("periodogram", "regression"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pgram_+3A_z">z</code></td>
<td>
<p>time series vector of length n, say.</p>
</td></tr>
<tr><td><code id="pgram_+3A_fr">fr</code></td>
<td>
<p>use &quot;default&quot; for usual Fourier frequencies, 1/n, ..., floor(n/2)/n. 
Set fr = N, to evaluate the periodogram at the Fourier frequencies 
corresponding to a time series of length N. Finally set fr to any
desired set of frequencies. Note frequencies are in cycles per unit
time sometimes called temoral frequency to distinguish from angular
frequency. Both are widely used in time series.</p>
</td></tr>
<tr><td><code id="pgram_+3A_method">method</code></td>
<td>
<p>either periodogram or regression</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses FFT.
So if the length of z is a highly composite number, the computation
is very efficient. Otherwise the usual DFT is used.
</p>


<h3>Value</h3>

<p>Periodogram evaluated at the Fourier frequencies or R-square.
</p>


<h3>Author(s)</h3>

<p>A.I. McLeod and Yuanhao Lai
</p>


<h3>Examples</h3>

<pre><code class='language-R'>z&lt;-sunspot.year
n&lt;-length(z) 
I&lt;-pgram(z)
f&lt;-I[,1]
I &lt;- I[,2]
plot(f, I, xlab="f", ylab="f", type="l") 
title(main="Periodogram for Annual Sunpots, 1700-1988") 
#
z&lt;-c(0.42, 0.89, 1.44, 1.98, 2.21, 2.04, 0.82, 0.62, 0.56, 0.8, 1.33)
fr &lt;- (1:50)/101
pgram(z)
pgram(z, fr=101)
pgram(z, fr=fr)
pgram(z, method="regression")
pgram(z, method="regression", fr=101)
pgram(z, method="regression", fr=fr)
  
                                                        
</code></pre>

<hr>
<h2 id='predictRSR'>Quantile computation for the RSR model</h2><span id='topic+predictRSR'></span>

<h3>Description</h3>

<p>Compute the estimated quantiles of the required statistics from a given RSR model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictRSR(n, tableRSR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predictRSR_+3A_n">n</code></td>
<td>
<p>Length of series.</p>
</td></tr>
<tr><td><code id="predictRSR_+3A_tablersr">tableRSR</code></td>
<td>
<p>The RSR fitted models for the required statistic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of all the estimated quantiles at pc for 
specidifed sample sizes n from a given RSR model.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>

<hr>
<h2 id='ptestg'>Test short time series for periodicity based on periodograms</h2><span id='topic+ptestg'></span>

<h3>Description</h3>

<p>This function is used to test the existence of the periodicity for 
a short time series (length&lt;=100). 
Several methods based on periodograms are provided with 
the response surface method implemented for efficiently obtaining 
accurate p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptestg(z, method = c("Fisher", "robust", "extended", "extendedRobust",
  "FisherRSR"), multiple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptestg_+3A_z">z</code></td>
<td>
<p>A series or a matrix containg series as columns</p>
</td></tr>
<tr><td><code id="ptestg_+3A_method">method</code></td>
<td>
<p>The statistical test to be used. See details for more information.</p>
</td></tr>
<tr><td><code id="ptestg_+3A_multiple">multiple</code></td>
<td>
<p>Indicating whether z contains multiple series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The null hypothesis is set as no peridicities, H0: f=0. 
Discriptions of different test statistics (methods) are as follow:
</p>
<p><code>Fisher</code>: The Fisher's g test statistic. The p-value is computed 
directly from the exact distribution.
</p>
<p><code>robust</code>: The robust g test proprosed in Ahdesmaki et al. (2005), 
where the p-value is computed by the response surface regression method.
</p>
<p><code>extended</code>: The extended Fisher's g test statistic, which
extend the Fisher's g test by enlarging the searching region 
of the frequency from the fourier frequencies to be 
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
The p-value is computed by the response surface regression method.
</p>
<p><code>extendedRobust</code>: Extend the frequency searching region of the robust
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
The p-value is computed by the response surface regression method.
</p>
<p><code>FisherRSR</code>: Only for experimental purposes, the Fisher;s g test with 
p-value computed form the response surface regression method.
</p>


<h3>Value</h3>

<p>Object of class &quot;Htest&quot; produced.
</p>
<p>An object of class &quot;Htest&quot; is a list containing the following components:
</p>
<table>
<tr><td><code>obsStat</code></td>
<td>
<p>Vector containing the observed test statistics.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>Vector containing the p-values of the selected tests.</p>
</td></tr>
<tr><td><code>freq</code></td>
<td>
<p>Vector containing the estimated frequencies.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yuanhao Lai and A.I. McLeod
</p>


<h3>References</h3>

<p>Fisher, R.A. (1929). Tests of significance in harmonic analysis. 
Proc. Roy. Soc. A, 125, 54-59.
</p>
<p>Ahdesmaki, M., Lahdesmaki, H., Pearson, R., Huttunen, H., and Yli-Harja O.(2005). 
<em>BMC Bioinformatics</em> <b>6</b>:117. 
<a href="http://www.biomedcentral.com/1471-2105/6/117">http://www.biomedcentral.com/1471-2105/6/117</a>.
</p>
<p>MacKinnon, James (2001) : 
Computing numerical distribution functions 
in econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ptestReg">ptestReg</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate the harmonic regression model with standard Gaussian error terms
set.seed(193)
## Non-Fourier frequency
z &lt;- simHReg(n = 14, f=2/10, A = 2, B = 1, model="Gaussian",sig=1) 
ptestg(z,method="Fisher")
ptestg(z,method="robust")
ptestg(z,method="extended")
ptestg(z,method="extendedRobust")
ptestg(z,method="FisherRSR")

# Performe tests on the alpha factor experiment
data(alpha)
## Eliminate genes with missing observations
alpha.nonNA &lt;- alpha[complete.cases(alpha),]
## Using the multiple option to do the test for all the genes
## Transpose the data set so that each column stands for a gene
alpha.nonNA &lt;- t(alpha.nonNA)
result &lt;- ptestg(alpha.nonNA, method = "extended",multiple=TRUE) 
str(result)              

                                                                        
# The movtivating example: gene ORF06806 in Cc
data(Cc)
x &lt;- Cc[which(rownames(Cc)=="ORF06806"),]
plot(1:length(x),x,type="b", main="ORF06806",
     xlab="time",ylab="Gene expression")
ptestg(x,method="Fisher") #Fail to detect the periodicity
ptestg(x,method="robust") 
ptestg(x,method="extended") 

</code></pre>

<hr>
<h2 id='ptestReg'>Test short time series for periodicity with maximum likelihood ratio tests</h2><span id='topic+ptestReg'></span>

<h3>Description</h3>

<p>This function is used to test the existence of the periodicity for 
a short time series (length&lt;=100). Likelihood ratio tests under 
the Gaussian or the Laplace assumptions are provided with 
the response surface method implemented for efficiently obtaining 
accurate p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptestReg(z, method = c("LS", "L1"), multiple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptestReg_+3A_z">z</code></td>
<td>
<p>A series or a matrix containg series as columns</p>
</td></tr>
<tr><td><code id="ptestReg_+3A_method">method</code></td>
<td>
<p>The statistical test to be used. See details for more information.</p>
</td></tr>
<tr><td><code id="ptestReg_+3A_multiple">multiple</code></td>
<td>
<p>Indicating whether z contains multiple series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The null hypothesis is set as no peridicities, H0: f=0. 
Discriptions of different test statistics (methods) are as follow:
</p>
<p><code>LS</code>: The -2 loglikelihood ratio test statistic based on 
the likelihood ratio test with normal noises, 
where the p-values are efficiently computed by 
the response surface method.
</p>
<p><code>L1</code>: The -2 loglikelihood ratio test statistic based on 
the likelihood ratio test with Laplace noises, 
where the p-values are efficiently computed by 
the response surface method.
</p>


<h3>Value</h3>

<p>Object of class &quot;Htest&quot; produced.
</p>
<p>An object of class &quot;Htest&quot; is a list containing the following components:
</p>
<table>
<tr><td><code>obsStat</code></td>
<td>
<p>Vector containing the observed test statistics.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>Vector containing the p-values of the selected tests.</p>
</td></tr>
<tr><td><code>freq</code></td>
<td>
<p>Vector containing the estimated frequencies.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yuanhao Lai and A.I. McLeod
</p>


<h3>References</h3>

<p>Islam, M.S. (2008). Peridocity, Change Detection and Prediction in Microarrays. 
Ph.D. Thesis, The University of Western Ontario. 
</p>
<p>Li, T. H. (2010). A nonlinear method for robust spectral analysis. 
Signal Processing, IEEE Transactions on, 58(5), 2466-2474.
</p>
<p>MacKinnon, James (2001) : 
Computing numerical distribution functions 
in econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitHReg">fitHReg</a>, <a href="#topic+ptestg">ptestg</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate the harmonic regression model with standard Gaussian error terms
set.seed(193)
# Non-Fourier frequency
z &lt;- simHReg(n = 14, f=2/10, A = 2, B = 1, model="Gaussian",sig=1) 
ptestReg(z,method = "LS") #Normal likelihood ratio test
ptestReg(z,method = "L1") #Laplace likelihood ratio test  
fitHReg(z, algorithm="exact") #the nls fitted result 
    
                                           
# Performe tests on the alpha factor experiment
data(alpha)
## Eliminate genes with missing observations
alpha.nonNA &lt;- alpha[complete.cases(alpha),]
## Using the multiple option to do the test for all the genes
## Transpose the data set so that each column stands for a gene
alpha.nonNA &lt;- t(alpha.nonNA)
result &lt;- ptestReg(alpha.nonNA, method = "LS",multiple=TRUE) 
str(result)       


# The movtivating example: gene ORF06806 in Cc
data(Cc)
x &lt;- Cc[which(rownames(Cc)=="ORF06806"),]
plot(1:length(x),x,type="b", main="ORF06806",
     xlab="time",ylab="Gene expression")
ptestg(x,method="Fisher") #Fail to detect the periodicity
ptestReg(x,method="LS") #The periodicity is significantly not zero
ptestReg(x,method="L1") #The periodicity is significantly not zero

</code></pre>

<hr>
<h2 id='pvalueRSR'>P-value computation for the RSR model</h2><span id='topic+pvalueRSR'></span>

<h3>Description</h3>

<p>Compute the estimated p-value of the observed statistics from a given RSR model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalueRSR(n, Statistic, tableRSR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalueRSR_+3A_n">n</code></td>
<td>
<p>Length of series.</p>
</td></tr>
<tr><td><code id="pvalueRSR_+3A_statistic">Statistic</code></td>
<td>
<p>A list of observed statistics.</p>
</td></tr>
<tr><td><code id="pvalueRSR_+3A_tablersr">tableRSR</code></td>
<td>
<p>The RSR fitted models for the required statistic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of p-values.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>

<hr>
<h2 id='rankSpectrum'>The robust rank-based spectral</h2><span id='topic+rankSpectrum'></span>

<h3>Description</h3>

<p>This function implements the robust, rank-based spectral.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankSpectrum(x, freqSet = c("An", "En"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rankSpectrum_+3A_x">x</code></td>
<td>
<p>Series.</p>
</td></tr>
<tr><td><code id="rankSpectrum_+3A_freqset">freqSet</code></td>
<td>
<p>The set to search frequencies.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value of the robust rank-based spectral.
</p>

<hr>
<h2 id='simAR1'>Simulate AR(1) series</h2><span id='topic+simAR1'></span>

<h3>Description</h3>

<p>An AR(1) series with mean zero and variance 1 and
with autocorrelation paramater phi is simulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simAR1(n, phi = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simAR1_+3A_n">n</code></td>
<td>
<p>Length of series.</p>
</td></tr>
<tr><td><code id="simAR1_+3A_phi">phi</code></td>
<td>
<p>Autocorrelation parameter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model equation is:
z[t] = phi*z[t-1]+a[t],
where z[1] is N(0,1) and a[t] are NID(0, siga),
<code class="reqn">siga=\sqrt(1/(1-phi^2))</code>.
</p>


<h3>Value</h3>

<p>Series of length n.
</p>


<h3>Author(s)</h3>

<p>A.I. McLeod
</p>


<h3>References</h3>

<p>McLeod, A.I., Yu, Hao and Krougly, Z. (2007),  
Algorithms for Linear Time 
Series Analysis: With R Package, Journal of Statistical Software  23, 5 1-26.
</p>

<hr>
<h2 id='simHReg'>Simulate harmonic regression models</h2><span id='topic+simHReg'></span>

<h3>Description</h3>

<p>Simulates a harmonic regression.  Possible types of models are 
normal, t(5), Laplace, cubic and AR1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simHReg(n, f, A, B, model = c("Gaussian", "t5", "Laplace", "cubic", "AR1"),
  phi = 0, sig = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simHReg_+3A_n">n</code></td>
<td>
<p>Length of series.</p>
</td></tr>
<tr><td><code id="simHReg_+3A_f">f</code></td>
<td>
<p>Frequency.</p>
</td></tr>
<tr><td><code id="simHReg_+3A_a">A</code></td>
<td>
<p>Cosine amplitude.</p>
</td></tr>
<tr><td><code id="simHReg_+3A_b">B</code></td>
<td>
<p>Sine amplitude.</p>
</td></tr>
<tr><td><code id="simHReg_+3A_model">model</code></td>
<td>
<p>The model used for generating the error term. See details.</p>
</td></tr>
<tr><td><code id="simHReg_+3A_phi">phi</code></td>
<td>
<p>Only used if AR1 error distribution is selected.</p>
</td></tr>
<tr><td><code id="simHReg_+3A_sig">sig</code></td>
<td>
<p>The standard error of the series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generate a harmonic series y with length n, 
where <code class="reqn">y_t = A*cos(2*pi*f*t)+B*sin(2*pi*f*t)+sig*e_t,\ t=1,...,n,</code>
and e comes from one of the following specified distributions 
with mean 0 and standard error 1:
</p>
<p><code>Gaussian</code>: A standard normal distribution (i.i.d.).
</p>
<p><code>t5</code>: A t distribution with 5 degrees of freedom 
(i.i.d., standardized to mean 0 and variance 1).
</p>
<p><code>Laplace</code>: A Laplace (double exponential) distribution
(i.i.d., standardized to mean 0 and variance 1).
</p>
<p><code>cubic</code>: A standard normal distribution for e, 
but <code class="reqn">y=y^3</code> this time.
</p>
<p><code>AR1</code>: An AR(1) series with autocorrelation paramater phi
(standardized to mean 0 and variance 1).
</p>


<h3>Value</h3>

<p>Vector of length n, simulated harmonic series.
</p>


<h3>Author(s)</h3>

<p>A.I. McLeod and Yuanhao Lai
</p>


<h3>References</h3>

<p>McLeod, A.I., Yu, Hao and Krougly, Z. (2007),  
Algorithms for Linear Time 
Series Analysis: With R Package, Journal of Statistical Software  23, 5 1-26.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitHReg">fitHReg</a></code>, <code><a href="#topic+ptestReg">ptestReg</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Simulate the harmonic regression model with standard Gaussian error terms
z &lt;- simHReg(10, f=2/10, 1, 2, model="Gaussian",sig=1) #Fourier Frequency
plot(1:10,z,type="b")

#Simulate the AR(1) errors
z &lt;- simHReg(10, f=0/10, 0,0, model="AR1",phi=0.2,sig=1) 
acf(z)

</code></pre>

<hr>
<h2 id='simLaplace'>Simulate series with the Laplace distribution</h2><span id='topic+simLaplace'></span>

<h3>Description</h3>

<p>A Laplace (double exponential) series with mean zero and variance 1 is simulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simLaplace(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simLaplace_+3A_n">n</code></td>
<td>
<p>Length of series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model equation is: 
</p>
<p style="text-align: center;"><code class="reqn">e=(-1)^(u1\ge0.5) * ln(u2) / sqrt(2)</code>
</p>

<p>where u1 and u2 are independent uniform (0,1) random variables.
</p>


<h3>Value</h3>

<p>vector of length n, simulated harmonic series.
</p>


<h3>Author(s)</h3>

<p>Yuanhao Lai
</p>

<hr>
<h2 id='tableEg'>The result for the RSR method with respect to 
the extended Fisher's G test</h2><span id='topic+tableEg'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the extended Fisher's G test.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^6 with 500 replications and the
interpolated series sizes are c(8:19,seq(20,50,2),seq(55,100,5)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableEg
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions in 
econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tablegEven'>The result for the RSR method with respect to 
the Fisher'g test (Even series length)</h2><span id='topic+tablegEven'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the Fisher's g test 
when series length is even.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^6 with 1000 replications and the
interpolated series sizes are c(seq(8,50,2),seq(54,98,4),100).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tablegEven
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions in 
econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tablegOdd'>The result for the RSR method with respect to 
the Fisher'g test (Odd series length)</h2><span id='topic+tablegOdd'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the Fisher's g test 
when series length is odd.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^6 with 1000 replications and the
interpolated series sizes are c(seq(9,51,2),seq(55,99,4)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tablegOdd
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions in 
econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tableRegL1Even'>The result for the RSR method with respect to 
the laplace likelihood ratio test (Even series length)</h2><span id='topic+tableRegL1Even'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the laplace likelihood ratio test.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^5 with 100 replications and the
interpolated series sizes are c(seq(8,50,2),seq(54,98,4),100).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableRegL1Even
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions 
in econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tableRegL1Odd'>The result for the RSR method with respect to 
the laplace likelihood ratio test (Odd series length)</h2><span id='topic+tableRegL1Odd'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the laplace likelihood ratio test.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^5 with 100 replications and the
interpolated series sizes are c(seq(9,51,2),seq(55,99,4)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableRegL1Odd
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions 
in econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tableRegLs'>The result for the RSR method with respect to 
the normal likelihood ratio test</h2><span id='topic+tableRegLs'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the normal likelihood ratio test.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^6 with 1000 replications and the
interpolated series sizes are c(8:19,seq(20,50,2),seq(55,100,5)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableRegLs
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions 
in econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tableRgAn'>The result for the RSR method with respect to 
the robust g test</h2><span id='topic+tableRgAn'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the robust g test.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^5 with 200 replications and the
interpolated series sizes are c(8:19,seq(20,50,2),seq(55,100,5)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableRgAn
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions in 
econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

<hr>
<h2 id='tableRgEn'>The result for the RSR method with respect to 
the robust g test with frequency set &quot;En&quot;</h2><span id='topic+tableRgEn'></span>

<h3>Description</h3>

<p>This data set is used to build up the response surface regressions 
in order to obtain the p-values for the robust g test with 
frequency set &quot;En&quot;,
<code class="reqn">En = {j/101 | j=1,\dots,50 and j/101 \ge 1/n}</code>.
The simulation set-up of the response surface regression method is 
i.i.d. statistics of length 10^5 with 200 replications and the
interpolated series sizes are c(8:19,seq(20,50,2),seq(55,100,5)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableRgEn
</code></pre>


<h3>Format</h3>

<p>A list object with the following components:
</p>
<p><code>model_matrix</code>: A matrix with the estimated coefficents 
for the Response Surface Regressions (one row per quantile). 
</p>
<p><code>nc</code>: a vectpr of the lengths of the simulated series.
</p>
<p><code>pc</code>: A vector containing all probabilities at which  
the quantiles are estimated.
</p>
<p><code>r</code>: the rate in the basis function g(n)=(1/n)^r.
</p>
<p><code>q</code>: the number of parameters (except the intercept) used in 
the response surface regression, the intercept term is removed if negative.</p>


<h3>References</h3>

<p>MacKinnon, James (2001) : 
Computing numerical distribution functions in 
econometrics, Queen's Economics Department Working Paper, No. 1037.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
