<!DOCTYPE html><html lang="en"><head><title>Help for package ptm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ptm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.get.exepath'><p>Find Full Paths to Executables</p></a></li>
<li><a href='#.get.url'><p>Get Web Resource</p></a></li>
<li><a href='#aa.at'><p>Residue Found at the Requested Position</p></a></li>
<li><a href='#aa.comp'><p>Amino Acid Composition</p></a></li>
<li><a href='#get.seq'><p>Import a Protein Sequence from a Database</p></a></li>
<li><a href='#gracefully_fail'><p>Check that Internet Resource Work Properly and Fail Gracefully When Not</p></a></li>
<li><a href='#hmeto'><p>Human MetO sites oxidized by hydrogen peroxide treatment.</p></a></li>
<li><a href='#is.at'><p>Check Residue a Fixed Position</p></a></li>
<li><a href='#meto.list'><p>List Proteins Found in MetOSite Matching a Keyword</p></a></li>
<li><a href='#meto.scan'><p>Scans a Protein in Search of  MetO Sites</p></a></li>
<li><a href='#meto.search'><p>Search for Specific MetO Sites</p></a></li>
<li><a href='#pairwise.dist'><p>Compute Euclidean Distances</p></a></li>
<li><a href='#saro.dist'><p>Compute Distances to the Closest Aromatic Residues</p></a></li>
<li><a href='#saro.geometry'><p>Compute Geometric Parameters of S-Aromatic Motifs</p></a></li>
<li><a href='#saro.motif'><p>Search for S-Aromatic Motifs</p></a></li>
<li><a href='#xprod'><p>Compute Cross Product</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analyses of Protein Post-Translational Modifications</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains utilities for the analysis of post-translational modifications (PTMs) in proteins, 
    with particular emphasis on the sulfoxidation of methionine residues. Features include the ability to download, 
    filter and analyze data from the sulfoxidation database 'MetOSite'. Utilities to search and characterize S-aromatic motifs in proteins are also provided. 
    In addition, functions to analyze sequence environments around modifiable residues in proteins can be found. 
    For instance, 'ptm' allows to search for amino acids either overrepresented or avoided around the modifiable 
    residues from the proteins of interest. Functions tailored to test statistical hypothesis related to
    these differential sequence environments are also implemented. 
    Further and detailed information regarding the methods in this package can be found in (Aledo (2020) <a href="https://metositeptm.com">https://metositeptm.com</a>).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bitbucket.org/jcaledo/ptm">https://bitbucket.org/jcaledo/ptm</a>,
<a href="https://github.com/jcaledo/Rptm">https://github.com/jcaledo/Rptm</a>, <a href="https://metositeptm.com">https://metositeptm.com</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>bio3d (&ge; 2.3-4), curl, graphics, httr, jsonlite, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, markdown, testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-20 11:37:04 UTC; JCA</td>
</tr>
<tr>
<td>Author:</td>
<td>Juan Carlos Aledo [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juan Carlos Aledo &lt;caledo@uma.es&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-21 10:40:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='.get.exepath'>Find Full Paths to Executables</h2><span id='topic+.get.exepath'></span>

<h3>Description</h3>

<p>Finds the path to an executable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get.exepath(prg)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".get.exepath_+3A_prg">prg</code></td>
<td>
<p>name of the executable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the absolute path.
</p>

<hr>
<h2 id='.get.url'>Get Web Resource</h2><span id='topic+.get.url'></span>

<h3>Description</h3>

<p>Gets a web resource.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get.url(url, n_tries = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".get.url_+3A_url">url</code></td>
<td>
<p>url to be reached.</p>
</td></tr>
<tr><td><code id=".get.url_+3A_n_tries">n_tries</code></td>
<td>
<p>number of tries.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the response or an error message.
</p>

<hr>
<h2 id='aa.at'>Residue Found at the Requested Position</h2><span id='topic+aa.at'></span>

<h3>Description</h3>

<p>Returns the residue found at the requested position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.at(at, target, uniprot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aa.at_+3A_at">at</code></td>
<td>
<p>the position in the primary structure of the protein.</p>
</td></tr>
<tr><td><code id="aa.at_+3A_target">target</code></td>
<td>
<p>a character string specifying the UniProt ID of the protein of interest or, alternatively, the sequence of that protein.</p>
</td></tr>
<tr><td><code id="aa.at_+3A_uniprot">uniprot</code></td>
<td>
<p>logical, if TRUE the argument 'target' should be an ID.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please, note that when uniprot is set to FALSE, target can be the string returned by a suitable function, such as get.seq or other.
</p>


<h3>Value</h3>

<p>Returns a single character representing the residue found at the indicated position in the indicated protein.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>See Also</h3>

<p>is.at(), aa.comp()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: aa.at(28, 'P01009')
</code></pre>

<hr>
<h2 id='aa.comp'>Amino Acid Composition</h2><span id='topic+aa.comp'></span>

<h3>Description</h3>

<p>Returns a table with the amino acid composition of the target protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.comp(target, uniprot = TRUE, reference = 'human', init = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aa.comp_+3A_target">target</code></td>
<td>
<p>a character string specifying the UniProt ID of the protein of interest or, alternatively, the sequence of that protein.</p>
</td></tr>
<tr><td><code id="aa.comp_+3A_uniprot">uniprot</code></td>
<td>
<p>logical, if TRUE the argument 'target' should be an ID.</p>
</td></tr>
<tr><td><code id="aa.comp_+3A_reference">reference</code></td>
<td>
<p>amino acid frequencies (in percent) of the proteinogenic amino acids to be used as reference. It should be either 'human', 'up' (composition of proteins in UniProt in 2019). Alternatively, the user can pass as argument any vector with 20 values to be used as reference.</p>
</td></tr>
<tr><td><code id="aa.comp_+3A_init">init</code></td>
<td>
<p>logical, whether remove or not the first residue (initiation methionine) from the sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list where the first element is a dataframe with the observed and expected frequencies for each amino acid, the second element is the result of the Chi-squared test. In addition, a plot to reflect potential deviations from the reference standard composition is shown.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>See Also</h3>

<p>is.at(), renum.pdb(), renum.meto(), renum(), aa.at()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aa.comp('MPSSVSWGILLLAGLCCLVPVSLAEDPQGDAAQK', uniprot = FALSE)
</code></pre>

<hr>
<h2 id='get.seq'>Import a Protein Sequence from a Database</h2><span id='topic+get.seq'></span>

<h3>Description</h3>

<p>Imports a protein sequence from a selected database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.seq(id, db = 'uniprot', as.string = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.seq_+3A_id">id</code></td>
<td>
<p>the identifier of the protein of interest.</p>
</td></tr>
<tr><td><code id="get.seq_+3A_db">db</code></td>
<td>
<p>a character string specifying the desired database; it must be one of 'uniprot' or 'metosite'.</p>
</td></tr>
<tr><td><code id="get.seq_+3A_as.string">as.string</code></td>
<td>
<p>logical, if TRUE the imported sequence will be returned as a character string.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MetOSite uses the same type of protein ID than UniProt.
</p>


<h3>Value</h3>

<p>Returns a protein  sequence either as a character vector or a as a character string.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get.seq('P01009')
</code></pre>

<hr>
<h2 id='gracefully_fail'>Check that Internet Resource Work Properly and Fail Gracefully When Not</h2><span id='topic+gracefully_fail'></span>

<h3>Description</h3>

<p>Checks that internet resource works properly and fail gracefully when not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gracefully_fail(call, timeout = 10,  ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gracefully_fail_+3A_call">call</code></td>
<td>
<p>url of the resource.</p>
</td></tr>
<tr><td><code id="gracefully_fail_+3A_timeout">timeout</code></td>
<td>
<p>set maximum request time in seconds.</p>
</td></tr>
<tr><td><code id="gracefully_fail_+3A_...">...</code></td>
<td>
<p>further named parameters, such as query, headers, etc.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To be used as an ancillary function.
</p>


<h3>Value</h3>

<p>The response object or NULL when the server does not respond properly.
</p>


<h3>Author(s)</h3>

<p>thefactmachine
</p>


<h3>References</h3>

<p>https://gist.github.com/thefactmachine/18279b7796c0836d9188
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gracefully_fail("http://httpbin.org/delay/2")
</code></pre>

<hr>
<h2 id='hmeto'>Human MetO sites oxidized by hydrogen peroxide treatment.</h2><span id='topic+hmeto'></span>

<h3>Description</h3>

<p>A dataset containing data regarding human MetO sites oxidized by H2O2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmeto
</code></pre>


<h3>Format</h3>

<p>A data frame with 4472 rows and 15 variables:
</p>

<dl>
<dt>prot_id</dt><dd><p>UniProt ID of the oxidized protein</p>
</dd>
<dt>prot_name</dt><dd><p>the protein's name</p>
</dd>
<dt>met_pos</dt><dd><p>the position of the MetO site in the primary structure</p>
</dd>
<dt>met_vivo_vitro</dt><dd><p>conditions under which the oxidation experiment was carried out</p>
</dd>
<dt>MetOsites</dt><dd><p>array with all the sites oxidized in that protein</p>
</dd>
<dt>site_id</dt><dd><p>primary key identifying the site</p>
</dd>
<dt>positive</dt><dd><p>sequence environment of the MetO site</p>
</dd>
<dt>control</dt><dd><p>sequence environment of a non oxidized Met from the same protein</p>
</dd>
<dt>IDP</dt><dd><p>Intrinsically Disordered Proteins, 0: the protein is not found in DisProt; 1: the protein contains disordered regions; 2: the protein may contain disordered regions but the experimental evidences are ambiguous</p>
</dd>
<dt>IDR</dt><dd><p>Intrinsically Disordered Region, TRUE: the MetO site belong to the IDR, FALSE: the MetO site doesn't belong to the IDR</p>
</dd>
<dt>abundance</dt><dd><p>protein abundance, in ppm</p>
</dd>
<dt>N</dt><dd><p>protein length, in number of residues</p>
</dd>
<dt>met</dt><dd><p>number of methionine residues</p>
</dd>
<dt>fmet</dt><dd><p>relative frequency of Met in that protein</p>
</dd>
<dt>prot_vivo_vitro</dt><dd><p>whether the protein has been described to be oxidized in vivo, in vitro or under both conditions</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://metosite.uma.es/">https://metosite.uma.es/</a>
</p>

<hr>
<h2 id='is.at'>Check Residue a Fixed Position</h2><span id='topic+is.at'></span>

<h3>Description</h3>

<p>Checks if a given amino acid is at a given position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.at(at, target, aa = 'M', uniprot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.at_+3A_at">at</code></td>
<td>
<p>the position in the primary structure of the protein.</p>
</td></tr>
<tr><td><code id="is.at_+3A_target">target</code></td>
<td>
<p>a character string specifying the UniProt ID of the protein of interest or, alternatively, the sequence of that protein.</p>
</td></tr>
<tr><td><code id="is.at_+3A_aa">aa</code></td>
<td>
<p>the amino acid of interest.</p>
</td></tr>
<tr><td><code id="is.at_+3A_uniprot">uniprot</code></td>
<td>
<p>logical, if TRUE the argument 'target' should be an ID.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please, note that when uniprot is set to FALSE, target can be the string returned by a suitable function, such as get.seq or other.
</p>


<h3>Value</h3>

<p>Returns a boolean. Either the residue is present at that position or not.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>See Also</h3>

<p>aa.at(), aa.comp()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: is.at(28, 'P01009', 'Q')
</code></pre>

<hr>
<h2 id='meto.list'>List Proteins Found in MetOSite Matching a Keyword</h2><span id='topic+meto.list'></span>

<h3>Description</h3>

<p>Lists proteins found in MetOSite with names matching the keyword.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meto.list(keyword)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meto.list_+3A_keyword">keyword</code></td>
<td>
<p>a character string corresponding to the keyword</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a dataframe with the uniprot id, the protein name and the species, for those proteins present into MetOSite whose name contains the keyword.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>References</h3>

<p>Valverde et al. 2019. Bioinformatics 35:4849-4850 (PMID: 31197322)
</p>


<h3>See Also</h3>

<p>meto.search(), meto.scan()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meto.list('inhibitor')
</code></pre>

<hr>
<h2 id='meto.scan'>Scans a Protein in Search of  MetO Sites</h2><span id='topic+meto.scan'></span>

<h3>Description</h3>

<p>Scans a given protein in search of MetO sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meto.scan(up_id, report = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meto.scan_+3A_up_id">up_id</code></td>
<td>
<p>a character string corresponding to the UniProt ID.</p>
</td></tr>
<tr><td><code id="meto.scan_+3A_report">report</code></td>
<td>
<p>it should be a natural number between 1 and 3.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the 'report' parameter has been set to 1, this function returns a brief report providing the position,  the function category and literature references concerning the residues detected as MetO, if any. If we wish to obtain a more detailed report, the option should be: report = 2. Finally, If we want a detailed and printable report (saved in the current directory), we should set report = 3
</p>


<h3>Value</h3>

<p>This function returns a report regarding the MetO sites found, if any, in the protein of interest.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>References</h3>

<p>Valverde et al. 2019. Bioinformatics 35:4849-4850 (PMID: 31197322)
</p>


<h3>See Also</h3>

<p>meto.search(), meto.list()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meto.scan('P01009')
</code></pre>

<hr>
<h2 id='meto.search'>Search for Specific MetO Sites</h2><span id='topic+meto.search'></span>

<h3>Description</h3>

<p>Searches for specific MetO sites filtering MetOSite according to the selected criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meto.search(highthroughput.group = TRUE,
                   bodyguard.group = TRUE,
                   regulatory.group = TRUE,
                   gain.activity = 2, loss.activity = 2, gain.ppi = 2,
                   loss.ppi = 2, change.stability = 2, change.location = 2,
                   organism = -1, oxidant = -1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meto.search_+3A_highthroughput.group">highthroughput.group</code></td>
<td>
<p>logical, when FALSE the sites described in a high-throughput study (unknown effect) are filtered out.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_bodyguard.group">bodyguard.group</code></td>
<td>
<p>logical, when FALSE the sites postulated to function as ROS sink (because when oxidized no apparent effect can be detected) are filtered out.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_regulatory.group">regulatory.group</code></td>
<td>
<p>logical, when FALSE the sites whose oxidation affect the properties of the protein (and therefore may be involved in regulation) are filtered out.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_gain.activity">gain.activity</code></td>
<td>
<p>introduce 1 or 0 to indicate whether the oxidation of the selected sites implies a gain of activity or not, respectively. If we do not wish to use this property to filter, introduce 2.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_loss.activity">loss.activity</code></td>
<td>
<p>introduce 1 or 0 to indicate whether or not the oxidation of the selected sites implies a loss of activity or not, respectively. If we do not wish to use this property to filter, introduce 2.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_gain.ppi">gain.ppi</code></td>
<td>
<p>introduce 1 or 0 to indicate whether the oxidation of the selected sites implies a gain of protein-protein interaction or not, respectively. If we do not wish to use this property to filter, introduce 2.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_loss.ppi">loss.ppi</code></td>
<td>
<p>introduce 1 or 0 to indicate whether or not the oxidation of the selected sites implies a loss of protein-protein interaction or not, respectively. If we do not wish to use this property to filter, introduce 2.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_change.stability">change.stability</code></td>
<td>
<p>introduce 1 or 0 to indicate whether the oxidation of the selected sites leads to a change in the protein stability or not, respectively. If we do not wish to use this property to filter, introduce 2.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_change.location">change.location</code></td>
<td>
<p>introduce 1 or 0 to indicate whether or not the oxidation of the selected sites implies a change of localization or not, respectively. If we do not wish to use this property to filter, introduce 2.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_organism">organism</code></td>
<td>
<p>a character string indicating the scientific name of the species of interest, or -1 if we do not wish to filter by species.</p>
</td></tr>
<tr><td><code id="meto.search_+3A_oxidant">oxidant</code></td>
<td>
<p>a character string indicating the oxidant, or -1 if we do not wish to filter by oxidants.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that all the arguments of this function are optional. We only pass an argument to the function when we want to use that parameter to filter. Thus, meto.search() will return all the MetO sites found in the database MetOSite.
</p>


<h3>Value</h3>

<p>This function returns a dataframe with a line per MetO site.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>References</h3>

<p>Valverde et al. 2019. Bioinformatics 35:4849-4850 (PMID: 31197322)
</p>


<h3>See Also</h3>

<p>meto.scan(), meto.list()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meto.search(organism = 'Homo sapiens', oxidant = 'HClO')
</code></pre>

<hr>
<h2 id='pairwise.dist'>Compute Euclidean Distances</h2><span id='topic+pairwise.dist'></span>

<h3>Description</h3>

<p>Computes the pairwise distance matrix between two sets of points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwise.dist(a, b, squared = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairwise.dist_+3A_a">a</code>, <code id="pairwise.dist_+3A_b">b</code></td>
<td>
<p>matrices (NxD) and (MxD), respectively, where each row represents a D-dimensional point.</p>
</td></tr>
<tr><td><code id="pairwise.dist_+3A_squared">squared</code></td>
<td>
<p>return containing squared Euclidean distance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Euclidean distance matrix (NxM). An attribute &quot;squared&quot; set to the
value of param <code>squared</code> is provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pairwise.dist(matrix(1:9, ncol = 3), matrix(9:1, ncol = 3))
</code></pre>

<hr>
<h2 id='saro.dist'>Compute Distances to the Closest Aromatic Residues</h2><span id='topic+saro.dist'></span>

<h3>Description</h3>

<p>Computes distances to the closest aromatic residues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saro.dist(pdb, threshold = 7, rawdata = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="saro.dist_+3A_pdb">pdb</code></td>
<td>
<p>either the path to the PDB file of interest or the 4-letters identifier.</p>
</td></tr>
<tr><td><code id="saro.dist_+3A_threshold">threshold</code></td>
<td>
<p>distance in ångströms, between the S atom and the aromatic ring centroid, used as threshold.</p>
</td></tr>
<tr><td><code id="saro.dist_+3A_rawdata">rawdata</code></td>
<td>
<p>logical to indicate whether we also want the raw distance matrix between delta S and aromatic ring centroids.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each methionyl residue this function computes the distances to the closest aromatic ring from Y, F and W. When that distance is equal or lower to the threshold, it will be computed as a S-aromatic motif.
</p>


<h3>Value</h3>

<p>The function returns a dataframe with as many rows as methionyl residues are found in the protein. The distances in ångströms to the closest tyrosine, phenylalanine and triptophan are given in the columns, as well as the number of S-aromatic motifs detected with each of these amino acids. Also a raw distance matrix can be provided.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>References</h3>

<p>Reid, Lindley &amp; Thornton, FEBS Lett. 1985, 190:209-213.
</p>


<h3>See Also</h3>

<p>saro.motif(), saro.geometry()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: saro.dist('1CLL')
</code></pre>

<hr>
<h2 id='saro.geometry'>Compute Geometric Parameters of S-Aromatic Motifs</h2><span id='topic+saro.geometry'></span>

<h3>Description</h3>

<p>Computes distances and angles of S-aromatic motifs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saro.geometry(pdb, rA, chainA = 'A', rB, chainB = 'A')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="saro.geometry_+3A_pdb">pdb</code></td>
<td>
<p>either the path to the PDB file of interest or the 4-letters identifier.</p>
</td></tr>
<tr><td><code id="saro.geometry_+3A_ra">rA</code></td>
<td>
<p>numeric position of one of the two residues involved in the motif.</p>
</td></tr>
<tr><td><code id="saro.geometry_+3A_chaina">chainA</code></td>
<td>
<p>a character indicating the chain to which belong the first residue.</p>
</td></tr>
<tr><td><code id="saro.geometry_+3A_rb">rB</code></td>
<td>
<p>numeric position of the second residue involved in the motif.</p>
</td></tr>
<tr><td><code id="saro.geometry_+3A_chainb">chainB</code></td>
<td>
<p>a character indicating the chain to which belong the second residue.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance between the delta sulfur atom and the centroid of the aromatic ring is computed, as well as the angle between this vector and the one perpendicular to the plane containing the aromatic ring. Based on the distance (d) and the angle (theta) the user decide whether the two residues are considered to be S-bonded or not (usually when d &lt; 7 and theta &lt; 60º).
</p>


<h3>Value</h3>

<p>The function returns a dataframe providing the coordinates of the sulfur atom and the centroid (centroids when the aromatic residue is tryptophan), as well as the distance (ångströms) and the angle (degrees) mentioned above.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>References</h3>

<p>Reid, Lindley &amp; Thornton, FEBS Lett. 1985, 190, 209-213.
</p>


<h3>See Also</h3>

<p>saro.motif(), saro.dist()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: saro.geometry('1CLL', rA = 141, rB = 145)
</code></pre>

<hr>
<h2 id='saro.motif'>Search for S-Aromatic Motifs</h2><span id='topic+saro.motif'></span>

<h3>Description</h3>

<p>Searches for S-aromatic motifs in proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saro.motif(pdb, threshold = 7, onlySaro = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="saro.motif_+3A_pdb">pdb</code></td>
<td>
<p>either the path to the PDB file of interest or the 4-letters identifier.</p>
</td></tr>
<tr><td><code id="saro.motif_+3A_threshold">threshold</code></td>
<td>
<p>distance in ångströms, between the S atom and the aromatic ring centroid, used as threshold.</p>
</td></tr>
<tr><td><code id="saro.motif_+3A_onlysaro">onlySaro</code></td>
<td>
<p>logical, if FALSE the output includes information about Met residues that are not involved in S-aromatic motifs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each methionyl residue taking place in a S-aromatic motif, this function computes the aromatic residues involved, the distance between the delta sulfur and the aromatic ring's centroid, as well as the angle between the sulfur-aromatic vector and the normal vector of the plane containing the aromatic ring.
</p>


<h3>Value</h3>

<p>The function returns a dataframe reporting the S-aromatic motifs found for the protein of interest.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>References</h3>

<p>Reid, Lindley &amp; Thornton, FEBS Lett. 1985, 190, 209-213.
</p>


<h3>See Also</h3>

<p>saro.dist(), saro.geometry()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: saro.motif('1CLL')
</code></pre>

<hr>
<h2 id='xprod'>Compute Cross Product</h2><span id='topic+xprod'></span>

<h3>Description</h3>

<p>Computes the  cross product of two vectors in three-dimensional euclidean space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xprod(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="xprod_+3A_...">...</code></td>
<td>
<p>vectors involved in the cross product.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each methionyl residue taking place in a S-aromatic motif, this function computes the aromatic residue involved, the distance between the delta sulfur and the aromatic ring's centroid, as well as the angle between the sulfur-aromatic vector and the normal vector of the plane containing the aromatic ring.
</p>


<h3>Value</h3>

<p>This function returns a vector that is orthogonal to the plane containing the two vector used as arguments.
</p>


<h3>Author(s)</h3>

<p>Juan Carlos Aledo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xprod(c(1,1,1), c(1,2,1))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
