<!DOCTYPE html><html><head><title>Help for package pubmed.mineR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pubmed.mineR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Abstracts-class'><p>Class <code>"Abstracts"</code> Abstract Class</p></a></li>
<li><a href='#additional_info'>
<p>To extract sentences with nultiple keywords from Abstracts</p></a></li>
<li><a href='#alias_fn'>
<p>To extract sentences containing Alias of the Human Genes from Pubmed abstracts.</p></a></li>
<li><a href='#altnamesfun'>
<p>To Get Alternative names of Genes</p></a></li>
<li><a href='#BWI'>
<p>To obtain the Buzz Word Index of terms from the Abstracts.</p></a></li>
<li><a href='#cleanabs'><p>To clean the result of searchabsL</p></a></li>
<li><a href='#cleanabs-methods'><p>Methods for Function <code>cleanabs</code></p></a></li>
<li><a href='#cluster_words'><p>To Find the highest frequency of words within clusters</p></a></li>
<li><a href='#co_occurrence_advance'><p>Extracts multiple sentence with co-occurrence of two sets of terms)</p></a></li>
<li><a href='#co_occurrence_fn'><p>Extracts sentences with co-occurrence of two sets of terms</p></a></li>
<li><a href='#combineabs'><p>To combine the abstracts</p></a></li>
<li><a href='#combineabs-methods'><p><code>Abstracts</code> Method to Combine Abstracts</p></a></li>
<li><a href='#common_words_new'>
<p>R Data containing words which frequently in text</p></a></li>
<li><a href='#contextSearch'><p>For Context Search</p></a></li>
<li><a href='#contextSearch-methods'><p>Method for Context Search</p></a></li>
<li><a href='#cos_sim_calc'><p>To calculate the cosine similarity between terms.</p></a></li>
<li><a href='#cos_sim_calc_boot'><p>Cosine Similarity Calculation by Boot Strapping</p></a></li>
<li><a href='#currentabs_fn'>
<p>To Retrive the Abstracts for year.</p></a></li>
<li><a href='#Find_conclusion'>
<p>To find the conclusion from the abstract(s).</p></a></li>
<li><a href='#find_intro_conc_html'><p>To find the introduction and conclusion from the abstracts.</p></a></li>
<li><a href='#gene_atomization'><p>To Extract Genes from the Abstracts</p></a></li>
<li><a href='#genes_BWI'>
<p>Function to obtain the Buzz Word Index of Genes from the abstracts.</p></a></li>
<li><a href='#GeneToEntrez'><p>Data containing Entrez Ids</p></a></li>
<li><a href='#Genewise'><p>To Search the number of abstracts for Genes</p></a></li>
<li><a href='#Genewise-methods'><p>method to find the abstracts for the given gene.</p></a></li>
<li><a href='#get_DOIs'>
<p>function for extracting Digital Object Identifier (DOIs) of papers</p></a></li>
<li><a href='#get_gene_sentences'>
<p>To extract the sentences for genes from the corpus.</p></a></li>
<li><a href='#get_MedlinePlus'>
<p>To Get MedLinePlus Summary</p></a></li>
<li><a href='#get_NMids'>
<p>To extract NM ids from NCBI.</p></a></li>
<li><a href='#get_original_term'><p>To get the original terms from the corpus. deprecated</p></a></li>
<li><a href='#get_original_term2'>
<p>To get the original terms from the corpus.</p></a></li>
<li><a href='#get_PMCIDS'>
<p>To extract the PMC Ids of the abstracts.</p></a></li>
<li><a href='#get_PMCtable'>
<p>To fetch the given PMC article tables. Deprecated</p></a></li>
<li><a href='#get_Sequences'>
<p>To extract the Gene sequence from the NCBI.</p></a></li>
<li><a href='#getabs'><p>To get Abstracts for a given term.</p></a></li>
<li><a href='#getabs-methods'><p><code>getabs</code> To Get abstracts for a term</p></a></li>
<li><a href='#getabsT'><p>To get Abstracts for a given term.</p></a></li>
<li><a href='#getabsT-methods'><p>To Get Abstracts</p></a></li>
<li><a href='#Give_Sentences'>
<p>To extract sentences from the Abstracts</p></a></li>
<li><a href='#Give_Sentences_PMC'>
<p>To fetch the sentence from the PMC full text article</p></a></li>
<li><a href='#head_abbrev'>
<p>To extract the abbreviated term.</p></a></li>
<li><a href='#HGNC-class'><p>HGNC Class for package.</p></a></li>
<li><a href='#HGNC2UniprotID'><p> R Data containing HGNC2UniprotID data mapping.</p></a></li>
<li><a href='#HGNCdata'><p> R Data containing HGNC data.</p></a></li>
<li><a href='#input_for_find_intro_conc_html'><p>fetch the abstracts using E-utilities.</p></a></li>
<li><a href='#local_uniprotfun'>
<p>To Get Information from Uniprot.</p></a></li>
<li><a href='#names_fn'>
<p>To extract the sentences in asbtracts containing gene names from HGNC.</p></a></li>
<li><a href='#new_xmlreadabs'><p>To read the abstracts from the PubMed saved in XML format.</p></a></li>
<li><a href='#official_fn'>
<p>To extract the sentences containing official gene symbol from abstracts.</p></a></li>
<li><a href='#pmids_to_abstracts'>
<p>To Find and match the PMIDs to the abstracts.</p></a></li>
<li><a href='#previousabs_fn'>
<p>To Retrieve the Abstracts from the large corpus for given years.</p></a></li>
<li><a href='#prevsymbol_fn'>
<p>To extract the sentences containing Previous symbols of HGNC genes.</p></a></li>
<li><a href='#printabs'><p>To prind the total number of abstracts in an S4 object of class Abstracts , its start and end</p></a></li>
<li><a href='#pubtator_function'>
<p>function for text annotation using PubTator</p></a></li>
<li><a href='#pubtator_function_JSON'>
<p>function for text annotation using PubTator</p></a></li>
<li><a href='#pubtator_result_list_to_table'>
<p>Function to Convert Pubtator result from list into Table</p></a></li>
<li><a href='#readabs'><p>To read Abstracts</p></a></li>
<li><a href='#readabsnew'><p>To read Abstracts</p></a></li>
<li><a href='#ready'><p>To Initiate the Classes.</p></a></li>
<li><a href='#removeabs'><p>To remove abstracts for the query term.</p></a></li>
<li><a href='#removeabs-methods'><p><code>removeabs</code> To remove abstracts of a term from the data.</p></a></li>
<li><a href='#searchabsL'><p>To Search the abstracts of term(s) in a combination mode.</p></a></li>
<li><a href='#searchabsL-methods'><p>Searching Abstracts</p></a></li>
<li><a href='#searchabsT'><p>To Search Abstracts</p></a></li>
<li><a href='#searchabsT-methods'><p><code>searchabsT</code> Searching abstracts</p></a></li>
<li><a href='#sendabs'><p>To send abstracts</p></a></li>
<li><a href='#sendabs-methods'><p>To send the Data into a File</p></a></li>
<li><a href='#SentenceToken'><p>To Tokenize the sentences</p></a></li>
<li><a href='#space_quasher'><p>Removes extra spaces between words.</p></a></li>
<li><a href='#subabs'><p>To find sub-abstracts</p></a></li>
<li><a href='#subabs-methods'><p>Getting subabstracts</p></a></li>
<li><a href='#subsetabs'>
<p>To make subsets of large corpus.</p></a></li>
<li><a href='#subsetabs-methods'><p>To make subset of Abstracts.</p></a></li>
<li><a href='#tdm_for_lsa'><p>create Term Document Matrix for lsa analysis</p></a></li>
<li><a href='#uniprotfun'><p>To get information about gene from the UniProt.Deprecated.</p></a></li>
<li><a href='#whichcluster'><p>To fetch the cluster for words</p></a></li>
<li><a href='#word_associations'><p>Extracts the words associated (to the left and to the right) with a given word</p></a></li>
<li><a href='#word_atomizations'><p>Atomization of words</p></a></li>
<li><a href='#wordscluster'><p>To cluster the words</p></a></li>
<li><a href='#wordsclusterview'><p>To view the words in cluster</p></a></li>
<li><a href='#xmlgene_atomizations'><p>Gene atomization of xml abstracts.Deprecated.</p></a></li>
<li><a href='#xmlgene_atomizations_new'><p>Gene atomization of xml abstracts.</p></a></li>
<li><a href='#xmlreadabs'><p>To read the abstracts from the PubMed saved in XML format.</p></a></li>
<li><a href='#xmlword_atomizations'><p>Word atomizations of abstracts from xml format.</p></a></li>
<li><a href='#Yearwise'><p>To Search abstracts Year wise</p></a></li>
<li><a href='#Yearwise-methods'><p><code>Yearwise</code> Year wise extraction of Abstracts</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Text Mining of PubMed Abstracts</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.20</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Jyoti Rani, S.Ramachandran, Ab Rauf Shah</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>S. Ramachandran &lt;ramuigib@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Text mining of PubMed Abstracts (text and XML) from <a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), methods</td>
</tr>
<tr>
<td>Imports:</td>
<td>RCurl, XML, boot, R2HTML, RJSONIO</td>
</tr>
<tr>
<td>Collate:</td>
<td>'Abstracts-class.R' 'HGNC-class.R' 'Yearwise.R' 'Genewise.R'
'combineabs.R' 'gene_atomization.R' 'Find_conclusion.R'
'getabs.R' 'getabsT.R' 'gethgnc.R' 'ready.R' 'readabs.R'
'removeabs.R' 'searchabsL.R' 'searchabsT.R' 'sendabs.R'
'subabs.R' 'cleanabs.R' 'word_atomizations.R' 'SentenceToken.R'
'contextSearch.R' 'uniprotfun.R' 'local_uniprotfun.R'
'tdm_for_lsa.R' 'printabs.R' 'pubtator_function.R'
'cos_sim_calc.R' 'cos_sim_calc_boot.R' 'wordscluster.R'
'whichcluster.R' 'wordsclusterview.R' 'find_intro_conc_html.R'
'cluster_words.R' 'get_original_term.R' 'get_original_term2.R'
'input_for_find_intro_conc_html.R' 'xmlreadabs.R'
'xmlword_atomizations.R' 'xmlgene_atomizations.R'
'pubtator_result_list_to_table.R' 'genes_BWI.R' 'BWI.R'
'currentabs_fn.R' 'previousabs_fn.R' 'altnamesfun.R'
'subsetabs.R' 'prevsymbol_fn.R' 'alias_fn.R' 'get_NMids.R'
'get_PMCIDS.R' 'get_PMCtable.R' 'get_Sequences.R'
'Give_Sentences_PMC.R' 'head_abbrev.R' 'names_fn.R'
'official_fn.R' 'pmids_to_abstracts.R' 'get_gene_sentences.R'
'Give_Sentences.R' 'get_MedlinePlus.R' 'co_occurrence_fn.R'
'space_quasher.R' 'readabsnew.R' 'word_associations.R'
'get_DOIs.R' 'additional_info.R' 'new_xmlreadabs.R'
'pubtator_function_JSON.R' 'xmlgene_atomizations_new.R'
'co_occurrence_advance.R'</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-09 09:52:07 UTC; sramachandran</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-08 11:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Abstracts-class'>Class <code>"Abstracts"</code> Abstract Class</h2><span id='topic+Abstracts-class'></span>

<h3>Description</h3>

<p>S4 Class with three slots Journal, Abstract, PMID to store abstracts from PubMed</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("Abstracts", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Journal</code>:</dt><dd><p>Object of class <code>"character"</code>to store Journals of the abstracts from PubMed</p>
</dd>
<dt><code>Abstract</code>:</dt><dd><p>Object of class <code>"character"</code>to store Abstracts from the PubMed</p>
</dd>
<dt><code>PMID</code>:</dt><dd><p>Object of class <code>"numeric"</code>to store PMIDs of abstracts from PubMed</p>
</dd>
</dl>



<h3>Methods</h3>

<p>No methods defined with class &quot;Abstracts&quot; in the signature.
</p>


<h3>Author(s)</h3>

<p> S.Ramachandran, Ab Rauf Shah </p>


<h3>See Also</h3>

<p><code><a href="#topic+searchabsL">searchabsL</a> <a href="#topic+getabs">getabs</a> <a href="#topic+contextSearch">contextSearch</a> <a href="#topic+Genewise">Genewise</a> 
<a href="#topic+Yearwise">Yearwise</a> <a href="#topic+combineabs">combineabs</a> <a href="#topic+subabs">subabs</a> <a href="#topic+subsetabs">subsetabs</a> <a href="#topic+readabs">readabs</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("Abstracts")
</code></pre>

<hr>
<h2 id='additional_info'>
To extract sentences with nultiple keywords from Abstracts
</h2><span id='topic+additional_info'></span>

<h3>Description</h3>

<p><code>additional_info</code> will help to extract the sentences containing multiple query term(s) from a large corpus of multiple abstracts.</p>


<h3>Usage</h3>

<pre><code class='language-R'>additional_info(abs, pmid, keywords)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="additional_info_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
<tr><td><code id="additional_info_+3A_pmid">pmid</code></td>
<td>

<p>Vector of PMIDs from abstracts
</p>
</td></tr>
<tr><td><code id="additional_info_+3A_keywords">keywords</code></td>
<td>
<p>Character Vector of Terms 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a matrix object containing PMID, keywords and sentences 
</p>


<h3>Author(s)</h3>

<p>Surabhi Seth
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Give_Sentences">Give_Sentences</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: additional_info(abs = Abstract, pmid = "26564970"", keywords = "text-mining" )
</code></pre>

<hr>
<h2 id='alias_fn'>
To extract sentences containing Alias of the Human Genes from Pubmed abstracts.
</h2><span id='topic+alias_fn'></span>

<h3>Description</h3>

<p><code>alias_fn</code> This function returns the sentences containing alias of gene and the user given terms from the Abstracts using HGNC gee data table.
In this sense this function is a 2 Dimensional search.</p>


<h3>Usage</h3>

<pre><code class='language-R'>alias_fn(genes, data, abs, filename, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alias_fn_+3A_genes">genes</code></td>
<td>

<p><code>genes</code> a table containing genes (official symbol,first column) with its frequency of occurrence (second column)could be an output of gene_atomization function and subsequently subsetting the table using for example the code genes_table = subset(t2diababs_genes, select = c(&quot;Gene_symbol&quot;,&quot;Freq&quot;)). Alternatively, a custome gene table can be supplied with two columns, the first one being the column for Gene symbols and the second one being the Frequency of occurrence. If Frequency of occurrence is not available then a dummy value of 1 can be set.
</p>
</td></tr>
<tr><td><code id="alias_fn_+3A_data">data</code></td>
<td>

<p><code>data</code> is HGNC data
table with all 49 features (columns) available from the web site https://www.genenames.org/</p>
</td></tr>
<tr><td><code id="alias_fn_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
<tr><td><code id="alias_fn_+3A_filename">filename</code></td>
<td>

<p><code>filename</code> specifies the name of output file. Please note that the term alias will be suffixed to the given filename.
</p>
</td></tr>
<tr><td><code id="alias_fn_+3A_terms">terms</code></td>
<td>

<p><code>terms</code> query term(s) to be search in the abstracts, could be a vector of terms.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An output file containing sentences with aliases of genes.For convenience both the official symbol and the corresponding alias are written in the output. The PMID of the corresponding Abstract containing the extracted sentence also appears just before the sentence. Note that multiple sentences from different abstracts are clubbed together under one gene alias that appears in those sentences.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prevsymbol_fn">prevsymbol_fn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: alias_fn(genes,data,myabs,"nephro_",c("diabetic nephropathy","kidney disease"))

## genes output of gene_atomization()
</code></pre>

<hr>
<h2 id='altnamesfun'>
To Get Alternative names of Genes
</h2><span id='topic+altnamesfun'></span>

<h3>Description</h3>

<p>This function is used to retrieve the Alternative names of genes from UniProt using HGNC gene symbol. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>altnamesfun(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="altnamesfun_+3A_m">m</code></td>
<td>

<p>is a character vector of HGNC official gene symbols.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list of alternative names of given Gene symbols.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>References</h3>

<p>UniProt Consortium. &quot;The universal protein resource (UniProt).&quot; Nucleic acids research 36.suppl 1 (2008): D190-D195.
http://www.uniprot.org/
</p>


<h3>See Also</h3>

<p><code><a href="#topic+uniprotfun">uniprotfun</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = altnamesfun(c("ADIPOQ","BDNF"))
## here "ADIPOQ" is the HGNC gene symbol for which alternative name(s) is required.
</code></pre>

<hr>
<h2 id='BWI'>
To obtain the Buzz Word Index of terms from the Abstracts.
</h2><span id='topic+BWI'></span>

<h3>Description</h3>

<p>This function is used to obtain the Buzz word index value for the terms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BWI(current, previous, n, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BWI_+3A_current">current</code></td>
<td>

<p><code>current</code> an S4 object containing the Abtracts for the current year we require the BWI an output from currentabs_fn()
</p>
</td></tr>
<tr><td><code id="BWI_+3A_previous">previous</code></td>
<td>

<p><code>previous</code> an S4 object containing the Abstracts for years previous to current year of study an output from previousabs_fn().
</p>
</td></tr>
<tr><td><code id="BWI_+3A_n">n</code></td>
<td>

<p><code>n</code> is a character term for which Buzz Word Index is to be calculated.
</p>
</td></tr>
<tr><td><code id="BWI_+3A_n">N</code></td>
<td>

<p><code>N</code> is a character value specifying the theme from the large corpus.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list containing BWI value for the given word.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>References</h3>

<p>Jensen, Lars Juhl, Jasmin Saric, and Peer Bork. &quot;Literature mining for the biologist: from information retrieval to biological discovery.&quot; Nature reviews genetics 7.2 (2006): 119-129.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+genes_BWI">genes_BWI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: result = BWI(mycurrentabs, mypreviousabs, "insulin", "inflammation")
## BWI for the term "insulin" and the theme is inflammation. 
## Note that in the previous, years are starting one before the current year 2015; 
## current is an S4 object containing the output from currentabs_fn()
## previous is an S4 object containing the output from previousabs_fn().
## 'n' and 'N' are query terms whose BWI is sought and the theme respectively
</code></pre>

<hr>
<h2 id='cleanabs'>To clean the result of searchabsL</h2><span id='topic+cleanabs'></span>

<h3>Description</h3>

<p>It will remove the 'NONE' abstracts from the result of searchabsL.</p>


<h3>Usage</h3>

<pre><code class='language-R'>cleanabs(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cleanabs_+3A_object">object</code></td>
<td>
<p>an S4 object of class Abstracts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an S4 object of class Abstracts.</p>


<h3>Author(s)</h3>

<p>Jyoti Rani</p>


<h3>See Also</h3>

<p><code><a href="#topic+searchabsL">searchabsL</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test1 = searchabsL(abs, include=c("term1", "term2"));
test2 = cleanabs(test1)
## End(Not run)
## here 'abs' is an S4 object of class Abstracts
## 'term1', 'term2' are the searchterms
## test1 is an S4 object containing abstracts for given terms
## and test2 is an S4 object of class Abstracts containing clean abstracts of searchabsL</code></pre>

<hr>
<h2 id='cleanabs-methods'>Methods for Function <code>cleanabs</code></h2><span id='topic+cleanabs-methods'></span><span id='topic+cleanabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p>To clean 'NONE' part of searchabsL output.</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>From an S4 object of class 'Abstracts' the cleanabs function is able to clean the output of searchabsL by removing the 'NONE' part of resultant abstracts.
</p>
</dd>
</dl>

<hr>
<h2 id='cluster_words'>To Find the highest frequency of words within clusters</h2><span id='topic+cluster_words'></span>

<h3>Description</h3>

<p>Function for finding the word (term) of highest frequency within clusters.</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_words(wordscluster, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_words_+3A_wordscluster">wordscluster</code></td>
<td>
<p>an R object containing the output of wordscluster()</p>
</td></tr>
<tr><td><code id="cluster_words_+3A_n">n</code></td>
<td>
<p>a numeric vector containing cluster numbers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing cluster and its highest frequency word</p>


<h3>Author(s)</h3>

<p>S. Ramachandran</p>


<h3>See Also</h3>

<p><code><a href="#topic+wordscluster">wordscluster</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = cluster_words(wordscluster, 5)
## wordscluster is an R object of wordscluster
## 5 is number of cluster
## End(Not run)
</code></pre>

<hr>
<h2 id='co_occurrence_advance'>Extracts multiple sentence with co-occurrence of two sets of terms)
</h2><span id='topic+co_occurrence_advance'></span>

<h3>Description</h3>

<p>Extracts single or multiple sentences with co-occurrence of given terms</p>


<h3>Usage</h3>

<pre><code class='language-R'>co_occurrence_advance(abstract, term1, term2, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="co_occurrence_advance_+3A_abstract">abstract</code></td>
<td>

<p>an S4 object of class Abstracts
</p>
</td></tr>
<tr><td><code id="co_occurrence_advance_+3A_term1">term1</code></td>
<td>

<p>a character vector of terms
</p>
</td></tr>
<tr><td><code id="co_occurrence_advance_+3A_term2">term2</code></td>
<td>

<p>a character vector of terms
</p>
</td></tr>
<tr><td><code id="co_occurrence_advance_+3A_n">n</code></td>
<td>

<p>A numeric value, which can be 0,1,2. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sentences with co-occurrence of two terms will be extracted along with the 
corresponding PMIDs. 
The output will be a data frame. In regard to the argument n, when the value 
is 0 then
the co-occurrence is sought in the same sentence. When the 
value is 1, then the co-occurrence is sought in two consecutive
sentences, namely, first term in the first sentence and second term
in the next sentence. When the value is 2, then the co-occurrence
is sought in two sentences separated by a sentence without either term1
or term2. 
</p>


<h3>Value</h3>

<p>It will return a data frame object containing PMID,sentences and the terms 
pairs. 
</p>


<h3>Author(s)</h3>

<p>Shashwat Badoni
Surabhi Seth
</p>


<h3>See Also</h3>

<p><code><a href="#topic+co_occurrence_fn">co_occurrence_fn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: co_occurrence_advance(myabs,"resistance", c("genes","genetic"), 2</code></pre>

<hr>
<h2 id='co_occurrence_fn'>Extracts sentences with co-occurrence of two sets of terms</h2><span id='topic+co_occurrence_fn'></span>

<h3>Description</h3>

<p><code>co_occurrence_fn</code> will automatically extract sentences with co-occurrence of two sets of terms.</p>


<h3>Usage</h3>

<pre><code class='language-R'>co_occurrence_fn(terms1, abs, filename, terms2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="co_occurrence_fn_+3A_terms1">terms1</code></td>
<td>
<p>a character vector of terms.</p>
</td></tr>
<tr><td><code id="co_occurrence_fn_+3A_abs">abs</code></td>
<td>
<p>an S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="co_occurrence_fn_+3A_filename">filename</code></td>
<td>
<p>a single character,  filename</p>
</td></tr>
<tr><td><code id="co_occurrence_fn_+3A_terms2">terms2</code></td>
<td>
<p>a character vector of terms.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sentences with co-occurrence of two terms will be extracted along with the corresponding PMIDs. The data will be written in a text file with the user given filename and the word co_occurrence will be suffixed to it.</p>


<h3>Value</h3>

<p>A text file.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: co_occurrence_fn("resistance",myabs,"resistance_genetic",c("genes","genetic")
##
</code></pre>

<hr>
<h2 id='combineabs'>To combine the abstracts</h2><span id='topic+combineabs'></span>

<h3>Description</h3>

<p><code>combineabs</code> will automatically combine two abtracts of two objects.</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineabs(object1, object2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combineabs_+3A_object1">object1</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="combineabs_+3A_object2">object2</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two objects of class 'Abstracts' are combined to return non-redundant combined abstracts. It can be used sequentially to combine many objects of class 'Abstracts'.
It will also write the number of combined abstracts into a text file named &quot;data_out.txt&quot;</p>


<h3>Value</h3>

<p> An R object containing the combined abstracts, and a text file named &quot;data_out.txt&quot; containing the number of abstracts combined together </p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: res1 = combineabs(x,y)
## here 'x', 'y' are the S4 objects of class 'Abstracts'.</code></pre>

<hr>
<h2 id='combineabs-methods'><code>Abstracts</code> Method to Combine Abstracts</h2><span id='topic+combineabs-methods'></span><span id='topic+combineabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>combineabs</code> method to combine the abstracts. object1 and object2 are from <code>Abstracts</code> class.</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object1 = "Abstracts")</code></dt><dd><p>An S4 object of class &quot;Abstracts&quot;</p>
</dd>
<dt><code>signature(object2 = "Abstracts")</code></dt><dd><p>An S4 object of class &quot;Abstracts&quot;</p>
</dd>
</dl>

<hr>
<h2 id='common_words_new'>
R Data containing words which frequently in text
</h2><span id='topic+common_words_new'></span>

<h3>Description</h3>

<p>This dataset is used to remove common words from the abstracts. This step is used for size reduction for further data mining.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(common_words_new)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;common_words_new&quot;
</p>


<h3>Details</h3>

<p>The dataset containing common words used to remove them from the text for size reduction.
</p>


<h3>References</h3>

<p>https://en.wikipedia.org/wiki/Most_common_words_in_English
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(common_words_new)
</code></pre>

<hr>
<h2 id='contextSearch'>For Context Search</h2><span id='topic+contextSearch'></span>

<h3>Description</h3>

<p><code>contextSearch</code> is a method to extract the sentences containing a given query term</p>


<h3>Usage</h3>

<pre><code class='language-R'>contextSearch(object, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contextSearch_+3A_object">object</code></td>
<td>
<p>An S4 object of Class Abstracts containing text abstracts</p>
</td></tr>
<tr><td><code id="contextSearch_+3A_y">y</code></td>
<td>
<p>a character vector of term(s)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It takes object of class Abstracts and query term(s) as arguments and returns a text and latex file of the sentences containing query term.
The latex file can be further converted into PDF by using the system command in R i.e. system(&quot;pdflatex filename.tex&quot;). pdflatex is a shell command in Linux to convert the latex file into PDF. In the pdf file the terms are written in bold face type to enable ease of reading</p>


<h3>Value</h3>

<p>contextSearch() will write two files one is a text file named &quot;companion.txt&quot;, and other is a Latex file. If the single term is given in query then file name comes with the term name. If multiple terms are used then the file name will be &quot;combined.tex&quot; </p>


<h3>Author(s)</h3>

<p>Dr.S.Ramachandran, Jyoti Rani</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: contextSearch(x, "diabetes")
## here 'x' is S4 object of class 'Abstracts', and query term is 'diabetes'.
</code></pre>

<hr>
<h2 id='contextSearch-methods'>Method for Context Search</h2><span id='topic+contextSearch-methods'></span><span id='topic+contextSearch+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>contextSearch</code> will search the sentence for the given term(s).</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>The object from where it will search should be an S4 object of class Abstracts
</p>
</dd>
</dl>

<hr>
<h2 id='cos_sim_calc'>To calculate the cosine similarity between terms.</h2><span id='topic+cos_sim_calc'></span>

<h3>Description</h3>

<p><code>cos_sim_calc</code> calculates the cosine measure of similarity between pairs of terms from a corpus.</p>


<h3>Usage</h3>

<pre><code class='language-R'>cos_sim_calc(nummatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cos_sim_calc_+3A_nummatrix">nummatrix</code></td>
<td>
<p>A numerical matrix for e.g. a Term Document matrix (output from tdm_for_lsa)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The term document matrix is taken as input and cosine meausures of similarity between all pairs of terms are calculated.</p>


<h3>Value</h3>

<p>A tab delimited text file containing the similarity values between all pairs of terms.</p>


<h3>Note</h3>

<p>This file can be input to cytoscape directly.</p>


<h3>Author(s)</h3>

<p>S. Ramachandran</p>


<h3>References</h3>

<p>https://en.wikipedia.org/wiki/Cosine_similarity
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tdm_for_lsa">tdm_for_lsa</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: x = cos_sim_calc(nummatrix)
## here nummatrix is the 'Term Document Matrix' generated from tdm_for_lsa()</code></pre>

<hr>
<h2 id='cos_sim_calc_boot'>Cosine Similarity Calculation by Boot Strapping</h2><span id='topic+cos_sim_calc_boot'></span>

<h3>Description</h3>

<p><code>cos_sim_calc_boot</code> allows boot strap analysis. This function should be used as argument for 'statistic' in the boot function of 'boot' package.</p>


<h3>Usage</h3>

<pre><code class='language-R'>cos_sim_calc_boot(data, indices)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cos_sim_calc_boot_+3A_data">data</code></td>
<td>
<p>Term Document Matrix generated from <code>tdm_for_lsa</code> function of this package. In this matrix, rows are terms and columns are abstracts.</p>
</td></tr>
<tr><td><code id="cos_sim_calc_boot_+3A_indices">indices</code></td>
<td>
<p>index of matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>while calling this function we need to transpose the input tdm and can also set the number of replicates. boot package is required to call this function.</p>


<h3>Value</h3>

<p>It will return a matrix containing the cosine similarity of pairs of terms in the abstracts. This object is in same format as returned by the 'boot' function of 'boot' package.</p>


<h3>Author(s)</h3>

<p>Dr.S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tdm_for_lsa">tdm_for_lsa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test_boot = boot(data = t(nummatrix), statistic = cos_sim_calc_boot, R = 2)
## here 'nummatrix' is a Term Document Matrix, boot inbuilt function of boot package, 
## R is number of replicates here it is 2. User can extend this number.
</code></pre>

<hr>
<h2 id='currentabs_fn'>
To Retrive the Abstracts for year.
</h2><span id='topic+currentabs_fn'></span>

<h3>Description</h3>

<p>This function is used to extract the abstracts for year we want to study. Its output is used as input in other functions like BWI() and genes_BWI()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>currentabs_fn(yr_to_include, theme, parentabs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="currentabs_fn_+3A_yr_to_include">yr_to_include</code></td>
<td>

<p><code>yr_to_include</code> is the year for which we want to extract the Abstracts.
</p>
</td></tr>
<tr><td><code id="currentabs_fn_+3A_theme">theme</code></td>
<td>

<p><code>theme</code> is a character value specifying the themes for the Abstracts.
</p>
</td></tr>
<tr><td><code id="currentabs_fn_+3A_parentabs">parentabs</code></td>
<td>

<p><code>parentabs</code> an S4 object containing the Abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns an S4 object containing the abstracts of the given year.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+previousabs_fn">previousabs_fn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = currentabs_fn("2015", "atherosclerosis", diabetesabs)
## here "2015" is the year for which, we wish to extract the abstracts on theme"Atherosclerosis" 
## from the large corpus of diabetes i.e. diabetesabs.
</code></pre>

<hr>
<h2 id='Find_conclusion'>
To find the conclusion from the abstract(s).
</h2><span id='topic+Find_conclusion'></span>

<h3>Description</h3>

<p>This function is designed for the user convinience, so that user can get the conclusion from the abstract(s) with out reading the whole abstract(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Find_conclusion(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Find_conclusion_+3A_y">y</code></td>
<td>

<p>An S4 object of class 'Abstract'.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing conclusions of given abstract(s)
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: res1 = Find_conclusion(y)
##  here 'y' is an S4 object of class Abstract.
</code></pre>

<hr>
<h2 id='find_intro_conc_html'>To find the introduction and conclusion from the abstracts.</h2><span id='topic+find_intro_conc_html'></span>

<h3>Description</h3>

<p>it helps to fetch the introduction and conclusion part from the abstracts.</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_intro_conc_html(y, themes, all)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_intro_conc_html_+3A_y">y</code></td>
<td>
<p>and S4 object of class Abstracts </p>
</td></tr>
<tr><td><code id="find_intro_conc_html_+3A_themes">themes</code></td>
<td>
<p>a character vector containing terms to be search in the abstracts</p>
</td></tr>
<tr><td><code id="find_intro_conc_html_+3A_all">all</code></td>
<td>
<p>is logical, if true, will include title and author otherwise only abstracts will be considered.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>find_intro_conc_html</code>provides an HTML file containing space separated introduction and conclusion part from the abstracts of given query term as well as gives a link directly to PubMed for the resulting PMID.</p>


<h3>Value</h3>

<p>an HTML file.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>See Also</h3>

<p><code><a href="#topic+input_for_find_intro_conc_html">input_for_find_intro_conc_html</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = find_intro_conc_html(abs, "diet", all=FALSE)
## here 'abs' is an S4 object of class Abstracts
## and 'diet' is a term to be search from the abstracts
## this function works for small size of corpus, say about 30-40 abstracts</code></pre>

<hr>
<h2 id='gene_atomization'>To Extract Genes from the Abstracts</h2><span id='topic+gene_atomization'></span>

<h3>Description</h3>

<p><code>gene_atomization</code> will automatically fetch the genes (HGNC approved Symbol) from the text and report their frequencies. presently only HGNC approved symbols are used.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_atomization(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_atomization_+3A_m">m</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
</table>


<h3>Details</h3>

<p> The function writes a text file with file name &quot;data_table.txt&quot;. The function gene_atomization() is used to obtain the name of genes along with their frequencies of occurence. </p>


<h3>Value</h3>

<p> A tab delimited table containing gene name and their frequencies of occurrence.</p>


<h3>Author(s)</h3>

<p> S.Ramachandran, Jyoti Rani</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: gene_atomization(myabs)
## here myabs is an S4 object of class 'Abstracts'containing the abstracts
## uses older version of HGNC data (https://www.genenames.org/) by default. 
## users may also use other functions such as official_fn and related 
## family of functions for deeper data mining.
</code></pre>

<hr>
<h2 id='genes_BWI'>
Function to obtain the Buzz Word Index of Genes from the abstracts.
</h2><span id='topic+genes_BWI'></span>

<h3>Description</h3>

<p>This function provides the Buzz word index for each gene. The theme is the context in which the gene is studied for e.g. atherosclerosis. Using this function user can identify abstracts with emphasis on a given gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genes_BWI(currentabs, previousabs, theme, genes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genes_BWI_+3A_currentabs">currentabs</code></td>
<td>

<p><code>currentabs</code> an S4 object containing the Abtracts for the year we want to study. Output from currentabs_fn()
</p>
</td></tr>
<tr><td><code id="genes_BWI_+3A_previousabs">previousabs</code></td>
<td>

<p><code>previousabs</code> an S4 object containing the Abstracts for years previous than our year of study. Output from previousabs_fn().
</p>
</td></tr>
<tr><td><code id="genes_BWI_+3A_theme">theme</code></td>
<td>

<p><code>theme</code> a character value to categorize our search. For e.g. 'Atherosclerosis' from 'diabetes' Abstracts.
</p>
</td></tr>
<tr><td><code id="genes_BWI_+3A_genes">genes</code></td>
<td>

<p><code>genes</code> list of genes.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a dataframe containig Genes with their corresponding BWI values.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BWI">BWI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = genes_BWI(currentabs, previousabs, theme, genes)
## currentabs is an S4 object contaning the Abtracts for the year we want to study.
## previousabs is an S4 object contaning the Abtracts for the years previous 
## than our query year for e.g. before 2015
## theme is a character value specifying the search.
## genes is a character vector of gene symbols.
</code></pre>

<hr>
<h2 id='GeneToEntrez'>Data containing Entrez Ids</h2><span id='topic+GeneToEntrez'></span>

<h3>Description</h3>

<p>This dataset is used in <code>DAVID_info</code> function of the package, and it contains the Entrez Ids for the respective genes and these Entrez Ids will be used to get information about human genes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GeneToEntrez)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;GeneToEntrez&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GeneToEntrez)
</code></pre>

<hr>
<h2 id='Genewise'>To Search the number of abstracts for Genes</h2><span id='topic+Genewise'></span>

<h3>Description</h3>

<p><code>Genewise</code> reports the number of abstracts for given gene(s) name(s)</p>


<h3>Usage</h3>

<pre><code class='language-R'>Genewise(object, gene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Genewise_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="Genewise_+3A_gene">gene</code></td>
<td>
<p>a character input of gene name(HGNC approved symbol)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will report the number of abstracts containing the query gene term(s) [HGNC approved symbols], and the result is saved in a text file &quot;dataout.txt&quot;. Genewise() will report numbers of abstracts only. The abstracts themselves for corresponding gene names can be obtained using searchabsL() and searchabsT.</p>


<h3>Value</h3>

<p>Genewise will return an R object containing the abstracts for given gene, and a text file named &quot;dataout.txt&quot; containing the number of abstracts</p>


<h3>Author(s)</h3>

<p>S. Ramachandran, Jyoti Rani</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Genewise(x, "TLR4")
## here 'x' contains the S4 object of Abstracts.
</code></pre>

<hr>
<h2 id='Genewise-methods'>method to find the abstracts for the given gene.</h2><span id='topic+Genewise-methods'></span><span id='topic+Genewise+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>Genewise</code> The method Genewise will automatically report the numbers of abstracts for a given gene. It will write the result in the text file named &quot;dataout.txt&quot;</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd><p>This method will search in an S4 object, containiing abstracts. It will write a text file named &quot;dataout.txt&quot;, containing the number of abstracts for the query gene terms</p>
</dd> 
</dl>


<hr>
<h2 id='get_DOIs'>
function for extracting Digital Object Identifier (DOIs) of papers
</h2><span id='topic+get_DOIs'></span>

<h3>Description</h3>

<p><code>get_DOIs</code> is used to extract DOIs of papers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_DOIs(abs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_DOIs_+3A_abs">abs</code></td>
<td>

<p>An S4 object of class Abstracts
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>get_DOIs</code> allow users to get DOIs for individual papers.
</p>


<h3>Value</h3>

<p>It returns a list object containing DOIs. This is useful for further extraction of papers 
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = get_DOIs(vitiligoabs)
## 
</code></pre>

<hr>
<h2 id='get_gene_sentences'>
To extract the sentences for genes from the corpus.
</h2><span id='topic+get_gene_sentences'></span>

<h3>Description</h3>

<p><code>get_gene_sentences</code> is used to extract the exact sentence in which query gene is discussed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_sentences(genes, abs, filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_sentences_+3A_genes">genes</code></td>
<td>

<p><code>genes</code> a character vector containing the gene symbols.
</p>
</td></tr>
<tr><td><code id="get_gene_sentences_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts
</p>
</td></tr>
<tr><td><code id="get_gene_sentences_+3A_filename">filename</code></td>
<td>

<p><code>filename</code> specifies the output file name.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an output file containing the sentences for given gene. 
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_gene_sentences("RBP4", abstracts, "RBP4_sentence.txt")
</code></pre>

<hr>
<h2 id='get_MedlinePlus'>
To Get MedLinePlus Summary
</h2><span id='topic+get_MedlinePlus'></span>

<h3>Description</h3>

<p>This function is to get the summary from MedLinePlus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_MedlinePlus(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_MedlinePlus_+3A_x">x</code></td>
<td>

<p><code>x</code> is a character input of terms: for examples 'malaria', 'pneumonia', 'chronic diseases'
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a HTML file with name result_Medline_plus.html to be opened with any browser
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>References</h3>

<p>www.medlineplus.gov, Conuel T. Finding answers in a beauty shop. NIH MedlinePlus: the magazine [Internet]. 2012 Fall [cited 2013 Feb 9]; 7(3):24-26. Available from: https://medlineplus.gov/magazine/issues/fall12/articles/fall12pg24-26.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_MedlinePlus("malaria")</code></pre>

<hr>
<h2 id='get_NMids'>
To extract NM ids from NCBI.
</h2><span id='topic+get_NMids'></span>

<h3>Description</h3>

<p><code>get_NMids</code> is to fetch the NM ids from the NCBI for corresponding gene/s to further fetch the sequence of that gene/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_NMids(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_NMids_+3A_x">x</code></td>
<td>

<p><code>x</code> an R object containing Locus IDs for genes from NCBI2R package.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list object containing corresponding NM id from NCBI.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>References</h3>

<p>http://www.ncbi.nlm.nih.gov/gene
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_Sequences">get_Sequences</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: getNMids("5950")
## 5950 is Locus id of RBP4 gene.
</code></pre>

<hr>
<h2 id='get_original_term'>To get the original terms from the corpus. deprecated</h2><span id='topic+get_original_term'></span>

<h3>Description</h3>

<p><code>get_original_term</code> is used to get the exact term as it is present in corpus. This function is not recommended anymore.</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_original_term(m, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_original_term_+3A_m">m</code></td>
<td>

<p>an S4 object of class Abstracts containing the corpus.
</p>
</td></tr>
<tr><td><code id="get_original_term_+3A_n">n</code></td>
<td>

<p>a list object output from the function cluster_words
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list object contatining the terms.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wordscluster">wordscluster</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = get_original_term(abs, words)
## here abs is an S4 object of class Abstracts
## words is the output object of cluster_words()
</code></pre>

<hr>
<h2 id='get_original_term2'>
To get the original terms from the corpus.
</h2><span id='topic+get_original_term2'></span>

<h3>Description</h3>

<p><code>get_original_term2</code> is used to get the exact term as it is present in corpus. It takes one term at a time. For multiple terms we can use lapply.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_original_term2(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_original_term2_+3A_x">x</code></td>
<td>

<p><code>x</code> is a character value specifying the query term.
</p>
</td></tr>
<tr><td><code id="get_original_term2_+3A_y">y</code></td>
<td>

<p><code>y</code> is an S4 object containing abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list object containing accurate term.
</p>


<h3>Author(s)</h3>

<p>Jyoti Rani, S.Ramachandran.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_original_term">get_original_term</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = get_original_term("hba1c", diababs)
## here it will return accurate formation of hba1c i.e. HbA1c from diababs.
</code></pre>

<hr>
<h2 id='get_PMCIDS'>
To extract the PMC Ids of the abstracts.
</h2><span id='topic+get_PMCIDS'></span>

<h3>Description</h3>

<p><code>get_PMCIDs</code> is used to fetch the PMC Ids of the abstracts from the corpus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_PMCIDS(abs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_PMCIDS_+3A_abs">abs</code></td>
<td>

<p><code>abs</code>an S4 object of class Abtsracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list containing PMC Ids.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_PMCIDS(abstracts)
</code></pre>

<hr>
<h2 id='get_PMCtable'>
To fetch the given PMC article tables. Deprecated
</h2><span id='topic+get_PMCtable'></span>

<h3>Description</h3>

<p><code>get_PMCtable</code> is used to extract the full texr article by giving query PMC Id. Deprecated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_PMCtable(url)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_PMCtable_+3A_url">url</code></td>
<td>

<p><code>url</code> is url of query PMC Id.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a full text artcle.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>References</h3>

<p>http://www.ncbi.nlm.nih.gov/pmc/
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_PMCIDS">get_PMCIDS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_PMCtable("http://www.ncbi.nlm.nih.gov/pmc/?term=4039032")
</code></pre>

<hr>
<h2 id='get_Sequences'>
To extract the Gene sequence from the NCBI.
</h2><span id='topic+get_Sequences'></span>

<h3>Description</h3>

<p><code>get_Sequences</code> is used to fetch the sequences of genes using NM ids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Sequences(x, filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_Sequences_+3A_x">x</code></td>
<td>
<p> NM Id of the sequence.
</p>
</td></tr>
<tr><td><code id="get_Sequences_+3A_filename">filename</code></td>
<td>

<p><code>filename</code> specifies the name of output file.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a text file containing sequence.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_NMids">get_NMids</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_Sequences("NM_012238.4", "SIRT1")
</code></pre>

<hr>
<h2 id='getabs'>To get Abstracts for a given term.</h2><span id='topic+getabs'></span>

<h3>Description</h3>

<p><code>getabs</code> will automatically fetch the abstracts containing the query term. A base function of the package pubmed.mineR.</p>


<h3>Usage</h3>

<pre><code class='language-R'>getabs(object, x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getabs_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="getabs_+3A_x">x</code></td>
<td>
<p>A character string for the term</p>
</td></tr>
<tr><td><code id="getabs_+3A_y">y</code></td>
<td>
<p>logical, if TRUE, search will be case sensitive</p>
</td></tr>
</table>


<h3>Details</h3>

<p>getabs() is used to find and exctract the abstracts for any given term, from the large a large corpus of abstracts. It uses regexpr based search strategy.</p>


<h3>Value</h3>

<p>An S4 object of class 'Abstracts', containing the result abstracts for the given term.</p>


<h3>Author(s)</h3>

<p>Dr.S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: getabs(x, "term")
## x is an S4 obeject of class abstracts containing the abstracts.
</code></pre>

<hr>
<h2 id='getabs-methods'><code>getabs</code> To Get abstracts for a term</h2><span id='topic+getabs-methods'></span><span id='topic+getabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>getabs</code> will search for the abstracts of a given term. It is case sensitive.</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd><p>This method takes three arguments, first 'object' containing data to be search, 'x', the term to be search, 'y' is logical if set &quot;YES&quot; will consider the case of text.</p>
</dd>
</dl>


<hr>
<h2 id='getabsT'>To get Abstracts for a given term.</h2><span id='topic+getabsT'></span>

<h3>Description</h3>

<p><code>getabsT</code> will automatically fetch the abstracts containing the query term.</p>


<h3>Usage</h3>

<pre><code class='language-R'>getabsT(object, x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getabsT_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="getabsT_+3A_x">x</code></td>
<td>
<p>A character string for the term</p>
</td></tr>
<tr><td><code id="getabsT_+3A_y">y</code></td>
<td>
<p>is logical, if set TRUE, search will be case sensitive.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>getabsT() is similar to getabs(), but it performs more specific search.</p>


<h3>Value</h3>

<p>An object of class 'Abstracts', containing the resulted abstracts for term.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: getabsT(diabdata, "term")
</code></pre>

<hr>
<h2 id='getabsT-methods'>To Get Abstracts</h2><span id='topic+getabsT-methods'></span><span id='topic+getabsT+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>getabsT</code> will automatically return the abstracts of a term from the data.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>getabsT will search for the abstracts of a term in the data, and will automatically write the number of abstracts into a text file named &quot;dataout.txt&quot;.
</p>
</dd>
</dl>

<hr>
<h2 id='Give_Sentences'>
To extract sentences from the Abstracts
</h2><span id='topic+Give_Sentences'></span>

<h3>Description</h3>

<p><code>Give_Sentences</code> will help to extract the sentence containing query term/s from the large corpus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Give_Sentences(m, abs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Give_Sentences_+3A_m">m</code></td>
<td>

<p><code>m</code> a character term.
</p>
</td></tr>
<tr><td><code id="Give_Sentences_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a list object containing sentences 
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Give_Sentences_PMC">Give_Sentences_PMC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Give_Sentences("diabetes", Abstracts)
</code></pre>

<hr>
<h2 id='Give_Sentences_PMC'>
To fetch the sentence from the PMC full text article
</h2><span id='topic+Give_Sentences_PMC'></span>

<h3>Description</h3>

<p><code>Give_Sentences_PMC</code> is used to extract the sentences from the full text article of given PMC id/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Give_Sentences_PMC(PMCID, term)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Give_Sentences_PMC_+3A_pmcid">PMCID</code></td>
<td>

<p><code>PMCID</code> represents the PMC Id from where we want to extract the sentence.
</p>
</td></tr>
<tr><td><code id="Give_Sentences_PMC_+3A_term">term</code></td>
<td>

<p><code>term</code> represents the term contained in a sentence.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a list object containing the sentences for query term from the given article.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Give_Sentences_PMC(PMC4039032, "atherosclerosis")
</code></pre>

<hr>
<h2 id='head_abbrev'>
To extract the abbreviated term.
</h2><span id='topic+head_abbrev'></span>

<h3>Description</h3>

<p><code>head_abbrev</code> is used to find expansion for which abbreviation is used. 
It will help to find the falsely matching abbreviations from the abstracts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>head_abbrev(limits, term, pmid, abs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="head_abbrev_+3A_limits">limits</code></td>
<td>

<p><code>limits</code> specifies the limit up to which expansion should be displayed. Default is 50
</p>
</td></tr>
<tr><td><code id="head_abbrev_+3A_term">term</code></td>
<td>

<p><code>term</code> is the query term (abbreviation)
</p>
</td></tr>
<tr><td><code id="head_abbrev_+3A_pmid">pmid</code></td>
<td>

<p><code>pmid</code> describes the PMID 
</p>
</td></tr>
<tr><td><code id="head_abbrev_+3A_abs">abs</code></td>
<td>

<p><code>abs</code>an S4 object of class Abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a list.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: head_abbrev(50, "AR", "16893912", myabs)
</code></pre>

<hr>
<h2 id='HGNC-class'>HGNC Class for package.</h2><span id='topic+HGNC-class'></span>

<h3>Description</h3>

<p><code>"HGNC"</code> </p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("HGNC", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>HGNCID</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>ApprovedSymbol</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>ApprovedName</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>Status</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>PreviousSymbols</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>Aliases</code>:</dt><dd><p>Object of class <code>"character"</code> </p>
</dd>
<dt><code>Chromosome</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>AccessionNumbers</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
<dt><code>RefSeqIDs</code>:</dt><dd><p>Object of class <code>"character"</code></p>
</dd>
</dl>



<h3>Author(s)</h3>

<p> Dr.S.Ramachandran, Ab Rauf Shah</p>


<h3>See Also</h3>

<p><code><a href="#topic+Abstracts-class">Abstracts</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("HGNC")
</code></pre>

<hr>
<h2 id='HGNC2UniprotID'> R Data containing HGNC2UniprotID data mapping. </h2><span id='topic+HGNC2UniprotID'></span>

<h3>Description</h3>

<p> This dataset contains HGNC2UniprotID from Uniprot and is used in uniprotfn() function of this package, to get the information of a gene from the Uniprot. </p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HGNC2UniprotID)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;HGNC2UniprotID&quot;
</p>


<h3>Details</h3>

<p> The dataset contains HGNC2UniprotID </p>


<h3>References</h3>

<p>UniProt Consortium. &quot;The universal protein resource (UniProt).&quot; Nucleic acids research 36.suppl 1 (2008): D190-D195.
http://www.uniprot.org/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(HGNC2UniprotID)
</code></pre>

<hr>
<h2 id='HGNCdata'> R Data containing HGNC data. </h2><span id='topic+HGNCdata'></span>

<h3>Description</h3>

<p> This dataset contains data from Human Gene Nomenclature Committe i.e HGNC ID, HGNC approved symbol, approved name, gene synonyms, chromosome no., accession numbers and RefSeq ids.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HGNCdata)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;HGNCdata&quot;
</p>


<h3>Details</h3>

<p> The dataset contains HGNCdata </p>


<h3>References</h3>

<p>Povey, Sue, et al. &quot;The HUGO gene nomenclature committee (HGNC).&quot; Human genetics 109.6 (2001): 678-680.
http://www.genenames.org/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(HGNCdata)
</code></pre>

<hr>
<h2 id='input_for_find_intro_conc_html'>fetch the abstracts using E-utilities.</h2><span id='topic+input_for_find_intro_conc_html'></span>

<h3>Description</h3>

<p>it helps in searching and fetching the abstracts from E-utilities using PMIDs.</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_for_find_intro_conc_html(y, all)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="input_for_find_intro_conc_html_+3A_y">y</code></td>
<td>
<p>an S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="input_for_find_intro_conc_html_+3A_all">all</code></td>
<td>
<p>is logical if true, will include title and author also.</p>
</td></tr></table>


<h3>Details</h3>

<p>it takes an S4 object as input and uses its PMIDs to fetch the abstracts from E-utilities. The output will be used as input for find_intro_conc_html as it contains neat data i.e. abstracts only.</p>


<h3>Value</h3>

<p>a list containing abstracts and PMID</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>References</h3>

<p>literature/http:/eutils.ncbi.nlm.nih.gov/</p>


<h3>See Also</h3>

<p><code><a href="#topic+find_intro_conc_html">find_intro_conc_html</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test=input_for_find_intro_conc_html(abs)
## here 'abs' is an S4 object of class Abstracts.</code></pre>

<hr>
<h2 id='local_uniprotfun'>
To Get Information from Uniprot.
</h2><span id='topic+local_uniprotfun'></span>

<h3>Description</h3>

<p>It is an auxiliary function for altnamesfun.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>local_uniprotfun(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="local_uniprotfun_+3A_y">y</code></td>
<td>

<p><code>y</code> a character value containing HGNC Gene symbol
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It writes an output file named &quot;x.txt&quot; which will be used as input in altnamesfun().
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani
</p>


<h3>See Also</h3>

<p><code><a href="#topic+uniprotfun">uniprotfun</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: local_uniprotfun("TLR4")
## here it will generate an output file named "x.txt" containing 
## result for TLR4.
</code></pre>

<hr>
<h2 id='names_fn'>
To extract the sentences in asbtracts containing gene names from HGNC. 
</h2><span id='topic+names_fn'></span>

<h3>Description</h3>

<p><code>names_fn</code> matches the gene symbols to gene names and extract from HGNC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>names_fn(genes, data, abs, filename, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="names_fn_+3A_genes">genes</code></td>
<td>

<p><code>genes</code> is output of gene_atomization or a table containing HGNC gene symbols in first column with its frequency in second column.
</p>
</td></tr>
<tr><td><code id="names_fn_+3A_data">data</code></td>
<td>

<p><code>data</code> is HGNC data
table with all 49 features (columns) available from the web site https://www.genenames.org/
</p>
</td></tr>
<tr><td><code id="names_fn_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
<tr><td><code id="names_fn_+3A_filename">filename</code></td>
<td>

<p><code>filename</code> specifies the name of output file.
</p>
</td></tr>
<tr><td><code id="names_fn_+3A_terms">terms</code></td>
<td>

<p><code>terms</code> second query term to be searched in the same sentence (co-occurrence) of abstracts. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns an output file containing genes with their corresponding gene names and sentences with co-occurrences if any.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
names_fn(genes, data, diabetes_abs, "names", c("diabetic nephropathy", "DN"))

## End(Not run)
## genes output of gene_atomization()
</code></pre>

<hr>
<h2 id='new_xmlreadabs'>To read the abstracts from the PubMed saved in XML format.</h2><span id='topic+new_xmlreadabs'></span>

<h3>Description</h3>

<p><code>new_xmlreadabs</code> is modified form of xmlreadabs as it reads the abstracts downloaded or saved in XML format from PubMed. This function should be used for recent XML format from PubMed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>new_xmlreadabs(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="new_xmlreadabs_+3A_file">file</code></td>
<td>

<p>an XML file saved from PubMed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an S4 object of class Abstracts containing journals, abstracts and PMID.
</p>


<h3>Note</h3>

<p>This function is useful with recent format of XML files from PubMed. The older xmlreadabs will not work with recent format.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+readabsnew">readabsnew</a></code>
<code><a href="#topic+readabs">readabs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: xmlabs = new_xmlreadabs("easyPubMed_00001.txt")
## here "easyPubMed_00001.txt" is an xml file from PubMed using package easyPubMed</code></pre>

<hr>
<h2 id='official_fn'>
To extract the sentences containing official gene symbol from abstracts.
</h2><span id='topic+official_fn'></span>

<h3>Description</h3>

<p><code>official_fn</code> is used to fetch the sentences containing official gene symbol from HGNC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>official_fn(genes, abs, filename, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="official_fn_+3A_genes">genes</code></td>
<td>

<p><code>genes</code> is output of gene_atomization, or a table containing HGNC gene symbols in first column with its frequency in second column.
</p>
</td></tr>
<tr><td><code id="official_fn_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
<tr><td><code id="official_fn_+3A_filename">filename</code></td>
<td>

<p><code>filename</code> specifies the name of output file.
</p>
</td></tr>
<tr><td><code id="official_fn_+3A_terms">terms</code></td>
<td>

<p><code>terms</code> second query term to be searched in the same sentence (co-occurrence) of abstracts. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a text file containing corresponding official gene symbol.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
official_fn(genes, diabetes_abs, "genes", c("diabetic nephropathy", "DN"))

## End(Not run)
## genes output of gene_atomization()
</code></pre>

<hr>
<h2 id='pmids_to_abstracts'>
To Find and match the PMIDs to the abstracts.
</h2><span id='topic+pmids_to_abstracts'></span>

<h3>Description</h3>

<p><code>pmids_to_abstracts</code> is used to extract the abstract/s of query PMID/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pmids_to_abstracts(x, abs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pmids_to_abstracts_+3A_x">x</code></td>
<td>

<p><code>x</code> a numeric vector containing PMIDs
</p>
</td></tr>
<tr><td><code id="pmids_to_abstracts_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return an S4 object of class abstracts containing abstracts for query PMIDs.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: pmids_to_abstracts(26878666,abs)
</code></pre>

<hr>
<h2 id='previousabs_fn'>
To Retrieve the Abstracts from the large corpus for given years.
</h2><span id='topic+previousabs_fn'></span>

<h3>Description</h3>

<p>This function is used to extract the abstracts from the large corpus excluding the years and under a given theme. Its output is used in other functions like BWI and genes_BWI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>previousabs_fn(yrs_to_exclude, theme, parentabs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="previousabs_fn_+3A_yrs_to_exclude">yrs_to_exclude</code></td>
<td>

<p><code>yrs_to_exclude</code> is abstracts for the list of years we want to exclude from the corpus
</p>
</td></tr>
<tr><td><code id="previousabs_fn_+3A_theme">theme</code></td>
<td>

<p><code>theme</code> is a character value specifying the themes for the Abstracts.
</p>
</td></tr>
<tr><td><code id="previousabs_fn_+3A_parentabs">parentabs</code></td>
<td>

<p><code>parentabs</code> an S4 object containing the Abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns an S4 object containing the abstracts of the given year.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+currentabs_fn">currentabs_fn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = previousabs_fn(as.character(2015:2010), "atherosclerosis", diabetesabs
## here we will get the abstracts before 2010 for 'atherosclerosis'
## from the large corpus diabetesabs.
</code></pre>

<hr>
<h2 id='prevsymbol_fn'>
To extract the sentences containing Previous symbols of HGNC genes.
</h2><span id='topic+prevsymbol_fn'></span>

<h3>Description</h3>

<p><code>prevsymbol_fn</code> will return the sentences containing previous symbols of the genes from the abstracts using HGNC data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prevsymbol_fn(genes, data, abs, filename, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prevsymbol_fn_+3A_genes">genes</code></td>
<td>

<p><code>genes</code>  is output of gene_atomization, or a table containing HGNC gene symbols in first column with its frequency in second column.
</p>
</td></tr>
<tr><td><code id="prevsymbol_fn_+3A_data">data</code></td>
<td>

<p><code>data</code> is HGNC data
table with all 49 features (columns) available from the web site https://www.genenames.org/
</p>
</td></tr>
<tr><td><code id="prevsymbol_fn_+3A_abs">abs</code></td>
<td>

<p><code>abs</code> an S4 object of class Abstracts.
</p>
</td></tr>
<tr><td><code id="prevsymbol_fn_+3A_filename">filename</code></td>
<td>

<p><code>filename</code> specify the name of output file
</p>
</td></tr>
<tr><td><code id="prevsymbol_fn_+3A_terms">terms</code></td>
<td>

<p><code>terms</code> second query term to be searched in the same sentence (co-occurrence) of abstracts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a text file containing gene symbol with corresponding previous symbols.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran
</p>


<h3>See Also</h3>

<p><code><a href="#topic+names_fn">names_fn</a></code>, <code><a href="#topic+official_fn">official_fn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
prevsymbol_fn(genes, data, diabetes_abs, "prevsym", c("diabetic nephropathy", "DN"))

## End(Not run)
</code></pre>

<hr>
<h2 id='printabs'>To prind the total number of abstracts in an S4 object of class Abstracts , its start and end</h2><span id='topic+printabs'></span>

<h3>Description</h3>

<p>It gives overview of the abstracts in an S4 object of class Abstracts.</p>


<h3>Usage</h3>

<pre><code class='language-R'>printabs(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printabs_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>prints the total number of abstracts in an S4 object with additional information.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: printabs(myabs)
## here myabs is an S4 object of class Abstracts.
</code></pre>

<hr>
<h2 id='pubtator_function'>
function for text annotation using PubTator
</h2><span id='topic+pubtator_function'></span>

<h3>Description</h3>

<p><code>pubtator_function</code> is used to extract specific information from an abstract like Gene, chemical, and diseases etc. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pubtator_function(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pubtator_function_+3A_x">x</code></td>
<td>

<p>numeric value PMID.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>pubtator_function</code> allow users to get information about 'Gene', 'Chemical' and 'Disease' for given PMID. It uses online tool PubTator on R plateform. It also removes redundancy from the output. It takes one PMID at once, for multiple PMIDs user can use lapply() function.
</p>


<h3>Value</h3>

<p>It returns a list object containing Gene, Chemical, Disease and PMID. The corresponding concept id numbers are joined by a '&gt;' character. This is useful for further data mining 
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani
</p>


<h3>References</h3>

<p>Wei CH et. al., PubTator: a Web-based text mining tool for assisting Biocuration, Nucleic acids research, 2013, 41 (W1): W518-W522. doi: 10.1093/nar/gkt44
</p>
<p>Wei CH et. al., Accelerating literature curation with text-mining tools: a case study of using PubTator to curate genes in PubMed abstracts, Database (Oxford), bas041, 2012
</p>
<p>Wei CH et. al., PubTator: A PubMed-like interactive curation system for document triage and literature curation, in Proceedings of BioCreative 2012 workshop, Washington DC, USA, 145-150, 2012
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = pubtator_function(17922911)
## here pubtator_function() will extract the information from this given pmid.
</code></pre>

<hr>
<h2 id='pubtator_function_JSON'>
function for text annotation using PubTator
</h2><span id='topic+pubtator_function_JSON'></span>

<h3>Description</h3>

<p><code>pubtator_function</code> is used to extract specific information from an 
abstract like Gene, chemical, and diseases etc. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pubtator_function_JSON(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pubtator_function_JSON_+3A_x">x</code></td>
<td>

<p>numeric value PMID.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>pubtator_function_JSON</code> allow users to get information about 
'Gene', 'Chemical' and 'Disease' for given PMID. It uses online tool 
PubTator on R plateform. It also removes redundancy from the output. 
It takes one PMID at once, for multiple PMIDs user can use 
lapply() function.
</p>


<h3>Value</h3>

<p>It returns a list object containing Gene, Chemical, Disease and 
PMID. The corresponding concept id numbers 
are joined by a '&gt;' character. This is useful for further 
data mining 
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani
</p>


<h3>References</h3>

<p>Wei CH et. al., PubTator: a Web-based text mining tool for 
assisting Biocuration, Nucleic acids research, 2013, 41 (W1): W518-W522. 
doi: 10.1093/nar/gkt44
</p>
<p>Wei CH et. al., Accelerating literature curation with text-mining tools: 
a case study of using PubTator to curate genes in PubMed abstracts, 
Database (Oxford), bas041, 2012
</p>
<p>Wei CH et. al., PubTator: A PubMed-like interactive curation system for 
document triage and literature curation, in Proceedings of BioCreative 
2012 workshop, Washington DC, USA, 145-150, 2012
</p>


<h3>See Also</h3>

<p>pubtator_function()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = pubtator_function_JSON(17922911)
## here pubtator_function_JSON() will extract the information from 
## this given pmid.</code></pre>

<hr>
<h2 id='pubtator_result_list_to_table'>
Function to Convert Pubtator result from list into Table
</h2><span id='topic+pubtator_result_list_to_table'></span>

<h3>Description</h3>

<p>This function is used to collect the outputs of pubtator_function() after using lapply over multiple PMIDs. This function enables to convert it into table for easy reading and further analysis. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pubtator_result_list_to_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pubtator_result_list_to_table_+3A_x">x</code></td>
<td>

<p>here x is list output of pubtator_function().
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns table for pubtator_function output.
</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pubtator_function">pubtator_function</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = pubtator_result_list_to_table(x)
##here x is the output of pubtator_function
</code></pre>

<hr>
<h2 id='readabs'>To read Abstracts</h2><span id='topic+readabs'></span>

<h3>Description</h3>

<p><code>readabs</code> will automatically read the abstracts from the pubmed file.</p>


<h3>Usage</h3>

<pre><code class='language-R'>readabs(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readabs_+3A_x">x</code></td>
<td>
<p>Text file of PubMed abstracts. (Abstracts downloaded from PubMed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The saved file from a general pubmed search as text file is read via readabs().</p>


<h3>Value</h3>

<p>An S4 object of class &quot;Abstracts&quot;, and a text file with tab delimited headers Journal, Abstract, PMID written with file name &quot;newabs.txt&quot;.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: readabs("pubmed_result.txt")
##here pubmed_result.txt is the text file of abstracts saved from PubMed.
</code></pre>

<hr>
<h2 id='readabsnew'>To read Abstracts</h2><span id='topic+readabsnew'></span>

<h3>Description</h3>

<p><code>readabsnew</code> will automatically read the abstracts from the pubmed text file.</p>


<h3>Usage</h3>

<pre><code class='language-R'>readabsnew(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readabsnew_+3A_x">x</code></td>
<td>
<p>Text file of PubMed abstracts. (Abstracts downloaded from PubMed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The saved file from a general pubmed search as text file is read via readabsnew().</p>


<h3>Value</h3>

<p>An S4 object of class &quot;Abstracts&quot; and a text file with tab delimited headers Journal, Abstract, PMID written with file name &quot;newabs.txt&quot;.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: readabsnew("pubmed_result.txt")
##here pubmed_result.txt is the text file of abstracts saved from PubMed.
</code></pre>

<hr>
<h2 id='ready'>To Initiate the Classes.</h2><span id='topic+ready'></span>

<h3>Description</h3>

<p><code>ready</code> will initiate the classes neccessary for other functions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ready()
</code></pre>


<h3>Details</h3>

<p>This function is neccessary to initiate the classes which are needed for the implementation of other functions. </p>


<h3>Value</h3>

<p> classes</p>


<h3>Author(s)</h3>

<p>S. Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: ready()
</code></pre>

<hr>
<h2 id='removeabs'>To remove abstracts for the query term.</h2><span id='topic+removeabs'></span>

<h3>Description</h3>

<p><code>removeabs</code>will remove the abstracts from a corpus for a given term.</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeabs(object, x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeabs_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="removeabs_+3A_x">x</code></td>
<td>
<p>A character value</p>
</td></tr>
<tr><td><code id="removeabs_+3A_y">y</code></td>
<td>
<p>is logical, if set 'TRUE' search will be case specific</p>
</td></tr>
</table>


<h3>Details</h3>

<p>removeabs() finds the abstracts for the given term and remove them from the large set of abstracts.A text file of file name &quot;dataout.txt&quot; will be written containing the number of abstracts removed.</p>


<h3>Value</h3>

<p>An S4 object of class Abstracts and a text file named &quot;dataout.txt&quot;</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: removeabs(myabs, "atherosclerosis", TRUE)
</code></pre>

<hr>
<h2 id='removeabs-methods'><code>removeabs</code> To remove abstracts of a term from the data.</h2><span id='topic+removeabs-methods'></span><span id='topic+removeabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>removeabs</code> This function will search for the abstracts containing the given term to remove them from the data.</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd><p>This method depicts its function, it will remove the abstracts from the data, and the number of abstracts removed will be written the text file named &quot;dataout.txt&quot;</p>
</dd>
</dl>


<hr>
<h2 id='searchabsL'>To Search the abstracts of term(s) in a combination mode.</h2><span id='topic+searchabsL'></span>

<h3>Description</h3>

<p><code>searchabsL</code> will search for abstracts for the given term(s). Multiple combinations are allowed.</p>


<h3>Usage</h3>

<pre><code class='language-R'>searchabsL(object, yr, include, restrict, exclude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="searchabsL_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="searchabsL_+3A_yr">yr</code></td>
<td>
<p>character vector specifies the year of search</p>
</td></tr>
<tr><td><code id="searchabsL_+3A_include">include</code></td>
<td>
<p>character vector specifies the terms contained in the abstracts.</p>
</td></tr>
<tr><td><code id="searchabsL_+3A_restrict">restrict</code></td>
<td>
<p>character vector specifies the term contained in the abstracts for which search should be restricted.</p>
</td></tr>
<tr><td><code id="searchabsL_+3A_exclude">exclude</code></td>
<td>
<p>character vector specifies the terms contained in the abstracts for excluding these abstracts from the search results.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the arguments except for the object all other arguments have &quot;NONE&quot; as default. To export or write the result of searchabsL() we use sendabs() function.</p>


<h3>Value</h3>

<p>An object of class Abstracts satisfying the term combinations, In addition a text file named &quot;out.txt&quot; reporting the number of abstracts for given query term combinations.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>See Also</h3>

<p><code><a href="#topic+searchabsT">searchabsT</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: searchabsL(myabs, include="term")
searchabsL(myabs, yr="2013")
searchabsL(myabs, restrict="term")
searchabsL(myabs, exclude="term")
searchabsL(myabs, include="term", exclude="term2")
## End(Not run)
## Here myabs is the object of class Abstracts containing data,
## "term" is the query term to be search.
</code></pre>

<hr>
<h2 id='searchabsL-methods'>Searching Abstracts</h2><span id='topic+searchabsL-methods'></span><span id='topic+searchabsL+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>searchabsL</code> will automatically search the abstracts from the data for the given terms or their combination of several terms.</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>searchabsL will search the abstracts for the given term or combinations of several terms. In this method the argument &quot;include&quot; uses the boolean operator 'OR' and is liberal whereas the 'restrict' and 'exclude' use the boolean operator 'AND' to specify additional filters. If the restriction to individual terms are desired then they can be individually searched and then the multiple abstracts can be combined using combineasb() function.
</p>
</dd>
</dl>

<hr>
<h2 id='searchabsT'>To Search Abstracts</h2><span id='topic+searchabsT'></span>

<h3>Description</h3>

<p><code>searchabsT</code>It is similar to searchabsL() but performs more specific search. It performs case sensitive search.</p>


<h3>Usage</h3>

<pre><code class='language-R'>searchabsT(object, yr, include, restrict, exclude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="searchabsT_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="searchabsT_+3A_yr">yr</code></td>
<td>
<p>character vector specifies the year(s) of search.</p>
</td></tr>
<tr><td><code id="searchabsT_+3A_include">include</code></td>
<td>
<p>character vector specifies the term(s) for which abstracts to be searched.</p>
</td></tr>
<tr><td><code id="searchabsT_+3A_restrict">restrict</code></td>
<td>
<p>character vector specifies the term(s) contained in the abstracts for which search should be restricted.</p>
</td></tr>
<tr><td><code id="searchabsT_+3A_exclude">exclude</code></td>
<td>
<p>character vector specifies the term(s) contained in the abstracts for excluding these abstracts from our search results.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the arguments except the object all arguments have &quot;NONE&quot; as default. Use sendabs() function to write the results in a tab delimited text file.</p>


<h3>Value</h3>

<p>An object of class Abstracts meeting the term and the term combinations. A text file reporting the number of abstracts for the query terms and their combinations is als written with the filename &quot;out.txt&quot;.</p>


<h3>Author(s)</h3>

<p>Dr.S.Ramachandran</p>


<h3>See Also</h3>

<p><code><a href="#topic+searchabsL">searchabsL</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: searchabsT(myabs,yr="2013")
searchabsT(myabs,include="term")
searchabsT(myabs,restrict="term")
searchabsT(myabs,exclude="term")
searchabsT(myabs,yr="2013", include="term")
## End(Not run)
## Here myabs is an S4 object of class Abstracts containing the abstracts to search,
## "term" is the query term to be search. 
</code></pre>

<hr>
<h2 id='searchabsT-methods'> <code>searchabsT</code> Searching abstracts</h2><span id='topic+searchabsT-methods'></span><span id='topic+searchabsT+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>searchabsT</code> will perform a specific search for the given term.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>It is similar to the searchabsL method, but it is more specific than searchabsL, it is case sensitive, however searchabsL is not. 
</p>
</dd>
</dl>

<hr>
<h2 id='sendabs'>To send abstracts</h2><span id='topic+sendabs'></span>

<h3>Description</h3>

<p><code>sendabs</code> will send the abstracts into a tab delimited text file with the fields Journal, Abstract, and PMID.</p>


<h3>Usage</h3>

<pre><code class='language-R'>sendabs(object, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sendabs_+3A_object">object</code></td>
<td>
<p>An S4 object of class 'Abstracts'</p>
</td></tr>
<tr><td><code id="sendabs_+3A_x">x</code></td>
<td>
<p>&quot;filename.txt&quot; to write the abstracts</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A general writing function for object of class 'Abstracts'</p>


<h3>Value</h3>

<p>A tab delimited text file with headers Journal, Abstract, PMID.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: sendabs(myabs,"myabs.txt")
## here myabs is the S4 object of class 'Abstracts' and 
## 'abs.txt' is the file where abstracts will be written.
</code></pre>

<hr>
<h2 id='sendabs-methods'>To send the Data into a File</h2><span id='topic+sendabs-methods'></span><span id='topic+sendabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>sendabs</code> will write the data of an object of class 'Abstracts' into a tab delimited text file with header Journal, Abstract, and  
PMID</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>sendabs will send the data into a text file. It writes a tab delimited text file for PubMed abstracts containing Journal, Abstract, and PMID.
</p>
</dd>
</dl>

<hr>
<h2 id='SentenceToken'>To Tokenize the sentences</h2><span id='topic+SentenceToken'></span>

<h3>Description</h3>

<p><code>SentenceToken</code> will tokenize abstracts into individual sentences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>SentenceToken(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SentenceToken_+3A_x">x</code></td>
<td>
<p>is a character string; could be an output from paste</p>
</td></tr>
</table>


<h3>Details</h3>

<p> This function is necessary for extracting sentences from abstracts,  used by contextSearch function. The tokenization principle follows the overall strategy as described in <code>contextSearch</code></p>


<h3>Value</h3>

<p> A character vector of sentences </p>


<h3>Author(s)</h3>

<p> S.Ramachandran </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: SentenceToken(x)
</code></pre>

<hr>
<h2 id='space_quasher'>Removes extra spaces between words.</h2><span id='topic+space_quasher'></span>

<h3>Description</h3>

<p><code>space_quasher</code> will automatically remove extra spaces between words. Therefore only one space between any pair of words will be left</p>


<h3>Usage</h3>

<pre><code class='language-R'>space_quasher(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="space_quasher_+3A_x">x</code></td>
<td>
<p>x is a text with single or multiple sentences given within double quotes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The extra spaces between words in sentences is quashed to one via space_quasher().</p>


<h3>Value</h3>

<p>Sentences(s) in which extra spaces between any pair of words are quashed to one.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: space_quasher("I am       a ghostbuster. I have the tools    required to  hunt ghosts")
##here pubmed_result.txt is the text file of abstracts saved from PubMed.
</code></pre>

<hr>
<h2 id='subabs'>To find sub-abstracts</h2><span id='topic+subabs'></span>

<h3>Description</h3>

<p><code>subabs</code> will automatically extract the sub-abstracts from  large set of abstracts.</p>


<h3>Usage</h3>

<pre><code class='language-R'>subabs(object, start, end)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subabs_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
<tr><td><code id="subabs_+3A_start">start</code></td>
<td>
<p>integer, specifies starting limit of the range to perform search</p>
</td></tr>
<tr><td><code id="subabs_+3A_end">end</code></td>
<td>
<p>integer, specifies end limit of the range to perform search</p>
</td></tr>
</table>


<h3>Details</h3>

<p>From a large number of asbtracts wish to extract a subset of abstracts into a separate object.</p>


<h3>Value</h3>

<p>An R object of class 'Abstracts' containing the extracted abstracts meeting a given range.</p>


<h3>Author(s)</h3>

<p>Jyoti Rani, S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: subabs(myabs,1,5)
## Here 'myabs is an S4 object of class 'Abstracts', 
## 1 and 5 are the start and end respectively.
</code></pre>

<hr>
<h2 id='subabs-methods'>Getting subabstracts</h2><span id='topic+subabs-methods'></span><span id='topic+subabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>subabs</code> subabs will extract the sub abstracts corresponding to a given range, from the whole data.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>From an S4 object of class 'Abstracts' the subabs function is able to extract the abstracts corresponding to a given range.
</p>
</dd>
</dl>

<hr>
<h2 id='subsetabs'>
To make subsets of large corpus.
</h2><span id='topic+subsetabs'></span>

<h3>Description</h3>

<p>It is used to divide the large corpus into a given range. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetabs(object, indices)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetabs_+3A_object">object</code></td>
<td>

<p><code>object</code> is an S4 object containing Abstracts.
</p>
</td></tr>
<tr><td><code id="subsetabs_+3A_indices">indices</code></td>
<td>

<p><code>indices</code> a numeric range (e.g. 1:10, c(1,5,7,9,10)). 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns an S4 obejct of extracted Abstracts.
</p>


<h3>Author(s)</h3>

<p>S. Ramachandran.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = subsetabs(diabetesabs, 1:50)
## here we want to extract the Abstacts ranges from 1 to 50 
## from the large corpus of diabetes. 
</code></pre>

<hr>
<h2 id='subsetabs-methods'>To make subset of Abstracts.</h2><span id='topic+subsetabs-methods'></span><span id='topic+subsetabs+2CAbstracts-method'></span>

<h3>Description</h3>

<p><code>subsetabs</code> is used to subset of Abstracts from the large corpus. Its output is used in other functions like currentabs_fn and previousabs_fn
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd>
<p>subsetabs will divide the large corpus into subset. 
</p>
</dd>
</dl>

<hr>
<h2 id='tdm_for_lsa'>create Term Document Matrix for lsa analysis</h2><span id='topic+tdm_for_lsa'></span>

<h3>Description</h3>

<p>lsa package take &quot;Term Document Matrix&quot; as input, so it is needed to create a 'tdm' for Abstracts and <code>tdm_for_lsa</code> do the same as it find out the frequency of given term in each abstract and each abstract is considered as separate document. It prepares term document matrix of terms in the 'abstracts' corpus</p>


<h3>Usage</h3>

<pre><code class='language-R'>tdm_for_lsa(object, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tdm_for_lsa_+3A_object">object</code></td>
<td>
<p>An S4 object of class 'Abstracts'</p>
</td></tr>
<tr><td><code id="tdm_for_lsa_+3A_y">y</code></td>
<td>
<p>a character vector specifying the terms </p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Term Document Matrix (Numerical matrix) containing the raw frequencies of given terms in each abstract.
</p>


<h3>Author(s)</h3>

<p>Jyoti Rani
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: y = c("insulin", "inflammation", "obesity")
tdm_for_lsa(myabs,y)
## End(Not run)
</code></pre>

<hr>
<h2 id='uniprotfun'>To get information about gene from the UniProt.Deprecated.</h2><span id='topic+uniprotfun'></span>

<h3>Description</h3>

<p><code>uniprotfun</code> will access the UniProt data for a given gene as per HGNC approved gene symbols. Deprecated.</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniprotfun(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="uniprotfun_+3A_y">y</code></td>
<td>
<p>a HGNC approved gene symbol as character</p>
</td></tr>
</table>


<h3>Details</h3>

<p> This function retrieves data from the UniProt. At present uniprotfun() works with only HGNC approved gene symbols. </p>


<h3>Value</h3>

<p> A text file written with filename as the 'query' name suffixed with .txt</p>


<h3>Author(s)</h3>

<p> S.Ramachandran </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: uniprotfun("SIRT1")
</code></pre>

<hr>
<h2 id='whichcluster'>To fetch the cluster for words</h2><span id='topic+whichcluster'></span>

<h3>Description</h3>

<p><code>whichcluster</code> is used to get the cluster in which a given word (term) occurs.</p>


<h3>Usage</h3>

<pre><code class='language-R'>whichcluster(clusterobject, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="whichcluster_+3A_clusterobject">clusterobject</code></td>
<td>
<p>an R object containing the clusters of words output by <code>wordscluster</code> function.</p>
</td></tr>
<tr><td><code id="whichcluster_+3A_y">y</code></td>
<td>
<p>a character string of query terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing the number of cluster under which given term occurs.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>See Also</h3>

<p><code><a href="#topic+wordscluster">wordscluster</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test&lt;-whichcluster(x, "diabetes")
## here x is an R object output form wordscluster function.
## and "diabetes" is the term for which cluster number is to be searched. 
## End(Not run)
</code></pre>

<hr>
<h2 id='word_associations'>Extracts the words associated (to the left and to the right) with a given word 
</h2><span id='topic+word_associations'></span>

<h3>Description</h3>

<p><code>word_associations</code>will automatically extract associated words for a given word, namely the words immediately to teh left and to the right. The given word is usually in the middle except for those cases, where the given word occurrs either at the start or the end of the sentence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>word_associations(term, abs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="word_associations_+3A_term">term</code></td>
<td>
<p>is a single word
</p>
</td></tr>
<tr><td><code id="word_associations_+3A_abs">abs</code></td>
<td>
<p>an S4 object of class Abstracts
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Certain words are qualified by authors in various ways. For example, physical therapy, gene therapy etc. This functions is useful in extracting these qualified words in the form of available associated words. Useful for preparing terms to be given in co_occurrence_fn (). There could be other uses also.
</p>


<h3>Value</h3>

<table>
<tr><td><code>comp1</code></td>
<td>
<p>A list of all the word pairs in a given set of abstracts.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>S. Ramachandran
</p>


<h3>References</h3>

<p>Rani J, Shah AB, Ramachandran S. pubmed.mineR: an R package with text-mining algorithms to analyse PubMed abstracts. J Biosci. 2015 Oct;40(4):671-82. PubMed PMID: 26564970.
</p>


<h3>See Also</h3>

<p><code>Give_Sentences</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: word_associations("therapy",myabs
##
</code></pre>

<hr>
<h2 id='word_atomizations'>Atomization of words</h2><span id='topic+word_atomizations'></span>

<h3>Description</h3>

<p><code>word_atomizations</code> will automatically break the whole text into words nd rank them according to their frequency of occurence.</p>


<h3>Usage</h3>

<pre><code class='language-R'>word_atomizations(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="word_atomizations_+3A_m">m</code></td>
<td>
<p>An S4 object of class Abstracts</p>
</td></tr>
</table>


<h3>Details</h3>

<p>word_atomizations() will break down the whole text into words after removing the extra white space, punctuation marks and very common english words.</p>


<h3>Value</h3>

<p> A text file containing words with their frequencies</p>


<h3>Author(s)</h3>

<p>S. Ramachandran, Jyoti Sharma</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: word_atomizations(myabs)
## here myabs is the object containing abstracts.
</code></pre>

<hr>
<h2 id='wordscluster'>To cluster the words</h2><span id='topic+wordscluster'></span>

<h3>Description</h3>

<p><code>wordscluster</code> is used to cluster the words, using the levenshtein distance concept, which are coming together in combination with either 'prefixes' or 'suffixes' or other compound words. The first word, usually of lowest length, could be  'stemmed' word in many cases drastically so, is considered as representative for that cluster.</p>


<h3>Usage</h3>

<pre><code class='language-R'>wordscluster(lower, upper)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wordscluster_+3A_lower">lower</code></td>
<td>
<p>lower limit for characters in word. Default = 5.</p>
</td></tr>
<tr><td><code id="wordscluster_+3A_upper">upper</code></td>
<td>
<p>upper limit of characters in word. Default = 30</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is usefull for dampening the 'explotion' of words output from word_atomizations. This step enables easy examination of the terms.</p>


<h3>Value</h3>

<p>a list object of words clustered together and a text filenamed &quot;resulttable.txt&quot; with the columns cluster number, cluster size and representatives of clusters.</p>


<h3>Note</h3>

<p>The function may run faster when the lower limits are reduced but 'risks' producing plenty of 'decoy' situations. Their frequencies are very rare. Decoy situations: Some 'words' with part identity to other smaller words will runaway with smaller words. This event creates an unfavorable situation whereby the generated 'clusters' of words become difficult to interpret. This situation can be minimized by increasing the lower limit of word length, however at the cost of lowering computational speed. An example is: the word hypercholesterolemia runsaway with the smaller word 'lester' which could be another name.In this instance increasing the lower limit will be more usefull. Words longer than 30 characters are usually names of chemical comnpunds in IUPAC system of nomenclature.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>See Also</h3>

<p><code><a href="#topic+whichcluster">whichcluster</a> <a href="#topic+word_atomizations">word_atomizations</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
test=wordscluster(5, 10)
## here it will start making cluster of words of length with minimum of 5 characters 
## and maximum of 10 characters.  

## End(Not run)
</code></pre>

<hr>
<h2 id='wordsclusterview'>To view the words in cluster</h2><span id='topic+wordsclusterview'></span>

<h3>Description</h3>

<p><code>wordsclusterview</code> is used to view the words comes in cluster formed by <code>wordscluster</code> function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>wordsclusterview(words_cluster, all)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wordsclusterview_+3A_words_cluster">words_cluster</code></td>
<td>
<p>an R object containing output of wordscluster</p>
</td></tr>
<tr><td><code id="wordsclusterview_+3A_all">all</code></td>
<td>
<p> is logical and default is FALSE, if set to TRUE includes those with one member word.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first 5 words and 5 words near the median nd 5 words at the tail end are shown for clusters with more than 15 members. In case of cluster size less than 15, all the words are written in output.</p>


<h3>Value</h3>

<p>It returns a text file named word_cluster_view.txt</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Rani</p>


<h3>See Also</h3>

<p><code><a href="#topic+wordscluster">wordscluster</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test= wordsclusterview(cluster)
# here cluster is output from wordscluster
## End(Not run)</code></pre>

<hr>
<h2 id='xmlgene_atomizations'>Gene atomization of xml abstracts.Deprecated.</h2><span id='topic+xmlgene_atomizations'></span>

<h3>Description</h3>

<p><code>xmlgene_atomizations</code> is used to fetch the list of genes from the xml abstracts.Deprecated.</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmlgene_atomizations(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmlgene_atomizations_+3A_m">m</code></td>
<td>
<p>an S4 object of class Abstracts, output from xmlreadabs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing genes from the text with their frquency of occurence.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Sharma
</p>


<h3>See Also</h3>

<p><code><a href="#topic+xmlreadabs">xmlreadabs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = xmlgene_atomizations(xmlabs)
## xmlabs is an S4 object of class Abstracts i.e. output of xmlreadabs</code></pre>

<hr>
<h2 id='xmlgene_atomizations_new'>Gene atomization of xml abstracts.</h2><span id='topic+xmlgene_atomizations_new'></span>

<h3>Description</h3>

<p><code>xmlgene_atomizations_new</code> is used to fetch the list of genes
from the xml abstracts</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmlgene_atomizations_new(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmlgene_atomizations_new_+3A_m">m</code></td>
<td>
<p>an S4 object of class Abstracts, output from xmlreadabs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing genes from the text with their frquency of 
occurrence.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran, Jyoti Sharma
</p>


<h3>See Also</h3>

<p><code><a href="#topic+new_xmlreadabs">new_xmlreadabs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = xmlgene_atomizations(xmlabs)
## xmlabs is an S4 object of class Abstracts i.e. output of xmlreadabs</code></pre>

<hr>
<h2 id='xmlreadabs'>To read the abstracts from the PubMed saved in XML format.</h2><span id='topic+xmlreadabs'></span>

<h3>Description</h3>

<p><code>xmlreadabs</code> is modified form of readabs as it reads the abstracts downloaded/saved in XML format from PubMed. This is helpful to give clean and better result after preprocessing i.e. <code>word_atomizations</code>, <code>wordscluster</code> etc.</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmlreadabs(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmlreadabs_+3A_file">file</code></td>
<td>
<p>an XML file saved from PubMed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an S4 object of class Abstracts containing journals, abstracts and PMID.</p>


<h3>Author(s)</h3>

<p>S.Ramachandran</p>


<h3>See Also</h3>

<p><code><a href="#topic+readabs">readabs</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: xmlabs = xmlreadabs("pubmed_result.xml")
## here "pubmed_result.xml" is an xml format file downloaded from PubMed.</code></pre>

<hr>
<h2 id='xmlword_atomizations'>Word atomizations of abstracts from xml format.</h2><span id='topic+xmlword_atomizations'></span>

<h3>Description</h3>

<p><code>xmlword_atomizations</code> is used to process the abstracts from PubMed in XML format.</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmlword_atomizations(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmlword_atomizations_+3A_m">m</code></td>
<td>
<p>an S4 object of class Abstracts resulted from xmlreadabs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing words from the text with their frequencies.</p>


<h3>Note</h3>

<p><code>xmlword_atomizations</code> cannot work on output of readabs.</p>


<h3>Author(s)</h3>

<p>S. Ramachandran</p>


<h3>See Also</h3>

<p><code><a href="#topic+xmlreadabs">xmlreadabs</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: test = xmlword_atomizations(xmlabs)
## here xmlabs is an S4 object i.e. output of xmlreadabs</code></pre>

<hr>
<h2 id='Yearwise'>To Search abstracts Year wise</h2><span id='topic+Yearwise'></span>

<h3>Description</h3>

<p><code>Yearwise</code> reports the no. of abstracts in a year.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Yearwise(object, year)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Yearwise_+3A_object">object</code></td>
<td>
<p>An S4 object of class Abstracts.</p>
</td></tr>
<tr><td><code id="Yearwise_+3A_year">year</code></td>
<td>
<p>a character vector specifies the year.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Yearwise() is useful to find the no. of abstracts for the given year.</p>


<h3>Value</h3>

<p>A text file containing the no. of abstracts for given Year(s)</p>


<h3>Author(s)</h3>

<p>Dr.S.Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Yearwise(myabs, "2011") or
Yearwise(myabs, c("2011", "2013", "2009")
## End(Not run)
## Here myabs is the object containing  PubMed abstracts.
</code></pre>

<hr>
<h2 id='Yearwise-methods'> <code>Yearwise</code> Year wise extraction of Abstracts </h2><span id='topic+Yearwise-methods'></span><span id='topic+Yearwise+2CAbstracts-method'></span>

<h3>Description</h3>

 <p><code>Yearwise</code> will report the abstracts for given year(s). </p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Abstracts")</code></dt><dd><p> This method &quot;Yearwise&quot; is written to fetch the abstracts yearly. </p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
