<!DOCTYPE html><html><head><title>Help for package pulseTD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pulseTD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#correctionParams'><p>Correct the parameters of the pulse model</p></a></li>
<li><a href='#estimateExpression'><p>Estimated Expression</p></a></li>
<li><a href='#estimateParams'><p>Estimated pulse model parameters!</p></a></li>
<li><a href='#getParams'><p>Get pulseModel params</p></a></li>
<li><a href='#getParams,pulseTDmodel-method'><p>getParams-method</p></a></li>
<li><a href='#getRates'><p>Get rate values</p></a></li>
<li><a href='#getRates,pulseTDmodel-method'><p>getRates-method</p></a></li>
<li><a href='#integrateA'><p>integral function</p></a></li>
<li><a href='#integrateAC'><p>integral function</p></a></li>
<li><a href='#plotRates'><p>draw Method</p></a></li>
<li><a href='#plotRates,pulseTDmodel-method'><p>plotRate-Method</p></a></li>
<li><a href='#predictExpression'><p>predict Expression</p></a></li>
<li><a href='#predictExpression,pulseTDmodel-method'><p>predictExpression-method</p></a></li>
<li><a href='#preExp'><p>A gene list that stores predicted values of gene expression and a 0.95 confidence interval.</p></a></li>
<li><a href='#pulseModel'><p>pulse function</p></a></li>
<li><a href='#pulseRates'><p>A pulseTDModel Rdata file containing 100 gene expression data,</p>
transcriptional dynamic data already related parameters.</a></li>
<li><a href='#pulseTDmodel-class'><p>pulseTD-model</p></a></li>
<li><a href='#rpkmres'><p>a list containing the expression values of exons and introns</p></a></li>
<li><a href='#rpkmSim'><p>A simulated gene expressing Rdata file that contains expression data for total RNA, precursor RNA, and 4sU-labeled RNA.</p></a></li>
<li><a href='#splitA6Params'><p>esitmate transcription parameters</p></a></li>
<li><a href='#splitB4Params'><p>esitmate degradation parameters</p></a></li>
<li><a href='#splitC5Params'><p>esitmate processing parameters</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of Transcriptional Dynamics using Pulse Models
via 4su-Seq Data and RNA-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-5-5</td>
</tr>
<tr>
<td>Author:</td>
<td>XinWang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>XinWang &lt;xindd_2014@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool for analyzing the transcription, processing and degradation rates of genes by 4sU-seq (the Metabolic Label 4-thiouridine) data and RNA-seq (RNA sequencing) data. It can not only recognize the transcriptional dynamic rates at the measurement time points, but also obtain continuous changes in transcriptional dynamics. More importantly, it is able to predict the trend of mRNA (Mature RNA) transcription and expression changes in the future.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>AnnotationDbi, SummarizedExperiment, Rsamtools, Biobase,
S4Vectors, methods, parallel, GenomicFeatures, ggplot2, grid,
GenomicAlignments</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, TxDb.Hsapiens.UCSC.hg19.knownGene</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/bioWzz/pulseTD_0.1.0">https://github.com/bioWzz/pulseTD_0.1.0</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-17 01:32:57 UTC; Administrator</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-17 09:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='correctionParams'>Correct the parameters of the pulse model</h2><span id='topic+correctionParams'></span>

<h3>Description</h3>

<p>It is used to correct pulse model parameters.
During the estimation of parameters in <code><a href="#topic+estimateParams">estimateParams</a></code>,
some NA will occur, which may be caused by random initial values.
Therefore, some NA values can be estimated from the new ones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correctionParams(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correctionParams_+3A_object">object</code></td>
<td>
<p>a 'pulseTDmodel' that has been calculated with <code><a href="#topic+estimateParams">estimateParams</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'pulseTDmodel' that has been modified
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('pulseRates', package='pulseTD')
pulseRates_correct = correctionParams(pulseRates)
pulseRates_correct@fitfailure
</code></pre>

<hr>
<h2 id='estimateExpression'>Estimated Expression</h2><span id='topic+estimateExpression'></span>

<h3>Description</h3>

<p>This function is used to calculate the expression value of the gene.
It includs total exon expression values, total intron expression values, labeled exon expression values, and labeled intron expression values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateExpression(txdb, filelist, by = "gene", mode = "Union",
  singleEnd = TRUE, ignore.strand = TRUE, fragments = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimateExpression_+3A_txdb">txdb</code></td>
<td>
<p>Genomic annotation file</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_filelist">filelist</code></td>
<td>
<p>A vector of sam file paths containing 4sU labeled sam files and unlabeled files</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_by">by</code></td>
<td>
<p>In terms of genes or transcripts, the default is the gene</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_mode">mode</code></td>
<td>
<p>mode can be one of the pre-defined count methods such as &quot;Union&quot;, &quot;IntersectionStrict&quot;, or &quot;IntersectionNotEmpty&quot; or it a user supplied count function.</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_singleend">singleEnd</code></td>
<td>
<p>(Default TRUE) A logical indicating if reads are single or paired-end.</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_ignore.strand">ignore.strand</code></td>
<td>
<p>A logical indicating if strand should be considered when matching</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_fragments">fragments</code></td>
<td>
<p>(Default FALSE) A logical; applied to paired-end data only. fragments control which function is used to read the data which subsequently affects which records are included in counting.</p>
</td></tr>
<tr><td><code id="estimateExpression_+3A_...">...</code></td>
<td>
<p>The parameters of the <code>summarizeOverlaps</code> method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of expression
</p>


<h3>See Also</h3>

<p>Other parameters <code>summarizeOverlaps</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
test_path &lt;- file.path(system.file(package="pulseTD"),'extdata/test1.sorted.bam')
test_path2 &lt;- file.path(system.file(package="pulseTD"),'extdata/test2.sorted.bam')
rpkmres &lt;- estimateExpression(txdb,c(test_path,test_path2), by='gene')
data('rpkmres', package='pulseTD')
head(rpkmres$total_exp)
head(rpkmres$pre_exp)
</code></pre>

<hr>
<h2 id='estimateParams'>Estimated pulse model parameters!</h2><span id='topic+estimateParams'></span>

<h3>Description</h3>

<p>It is used to estimate pulse model parameters.
Transcription rates, processing rates, and degradation correspond to different pulse model parameters, respectively.
The transcription rate has 6 parameters, the processing rate has 5 parameters, and the degradation rate has 4 parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateParams(labexon, totexon, totintr, TimeGrid, tL,
  clusterNumber = NULL, loopnumber = 50, message = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimateParams_+3A_labexon">labexon</code></td>
<td>
<p>A matrix containing expression levels of 4su exons.For the calculation of the expression value, see <code><a href="#topic+estimateExpression">estimateExpression</a></code></p>
</td></tr>
<tr><td><code id="estimateParams_+3A_totexon">totexon</code></td>
<td>
<p>A matrix containing expression levels of total exons.For the calculation of the expression value, see <code><a href="#topic+estimateExpression">estimateExpression</a></code></p>
</td></tr>
<tr><td><code id="estimateParams_+3A_totintr">totintr</code></td>
<td>
<p>A matrix containing expression levels of total introns.For the calculation of the expression value, see <code><a href="#topic+estimateExpression">estimateExpression</a></code></p>
</td></tr>
<tr><td><code id="estimateParams_+3A_timegrid">TimeGrid</code></td>
<td>
<p>A vector of points in time at which experimental data is collected, not allowed to be repeated</p>
</td></tr>
<tr><td><code id="estimateParams_+3A_tl">tL</code></td>
<td>
<p>4sU labeled time during the experiment</p>
</td></tr>
<tr><td><code id="estimateParams_+3A_clusternumber">clusterNumber</code></td>
<td>
<p>Given the number of cluster cores, the default is the maximum number of available cores</p>
</td></tr>
<tr><td><code id="estimateParams_+3A_loopnumber">loopnumber</code></td>
<td>
<p>The number of iterations of the gradient descent when solving the parameter. The default is 50.</p>
</td></tr>
<tr><td><code id="estimateParams_+3A_message">message</code></td>
<td>
<p>Whether to print the log, the default is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'pulseTDmodel' containing the expression values of the filtered genes, a list of solved parameters, and some basic parameter information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('rpkmSim', package='pulseTD')
rpkm_TL &lt;- rpkmSim$labexon[1:2,]
rpkm_PT &lt;- rpkmSim$totintr[1:2,]
rpkm_TT &lt;- rpkmSim$totexon[1:2,]
TimeGrid &lt;- c(0, 15, 30, 45, 60, 75, 90, 105, 120, 135, 150, 165, 180)
tL &lt;- 10

pulseRates&lt;- estimateParams(rpkm_TL,rpkm_TT,rpkm_PT,TimeGrid,tL,clusterNumber=1,loopnumber=10)


</code></pre>

<hr>
<h2 id='getParams'>Get pulseModel params</h2><span id='topic+getParams'></span>

<h3>Description</h3>

<p>It is used to obtain the parameters of the pulse function J Comput Biol (2009 Feb) &lt;doi:10.1089/cmb.2008.13TT&gt;, including transcription parameters, degradation parameters and processing parameters.
You can use these parameters to analyze transcriptional characteristics, including steady-state analysis.
At the same time, you can use <code><a href="#topic+pulseModel">pulseModel</a></code> to view the curve corresponding to the parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getParams(object, stage, genename = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getParams_+3A_object">object</code></td>
<td>
<p>a 'pulseTDmodel' that has been calculated with <code><a href="#topic+estimateParams">estimateParams</a></code></p>
</td></tr>
<tr><td><code id="getParams_+3A_stage">stage</code></td>
<td>
<p>A character is one of three stages of transcriptional dynamics, transcription, processing and degradation.</p>
</td></tr>
<tr><td><code id="getParams_+3A_genename">genename</code></td>
<td>
<p>a vector, default is NULL
</p>

<ul>
<li><p> If it is NULL: Calculate the transcriptional dynamic rate of all genes
</p>
</li>
<li><p> If it is a gene vector: only calculate the transcriptional dynamic rate of a given gene vector
</p>
</li>
<li><p> If it is a numerical vector: only calculate the transcriptional dynamic rate of the gene corresponding to a given value.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix or vector containing six parameters: h0,h1,h2,t1 t2,beta
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('pulseRates', package='pulseTD')
TimeGrid &lt;- c(0, 15, 30, 45, 60, 75, 90, 105, 120, 135, 150, 165, 180)
pulseRates_correct &lt;- correctionParams(pulseRates)
transcription_params = getParams(pulseRates,'transcription')
degradation_params = getParams(pulseRates, 'degradation')
processing_params = getParams(pulseRates, 'processing')
head(transcription_params)
head(degradation_params)
head(processing_params)
transcription_params = getParams(pulseRates, 'transcription', genename=c(1,2,3))
head(transcription_params)
###
transcription_pulse = pulseModel(as.matrix(transcription_params[1,]), TimeGrid)
degradation_pulse = pulseModel(as.matrix(degradation_params[1,]), TimeGrid)
processing_pulse = pulseModel(as.matrix(processing_params[1,]), TimeGrid)

</code></pre>

<hr>
<h2 id='getParams+2CpulseTDmodel-method'>getParams-method</h2><span id='topic+getParams+2CpulseTDmodel-method'></span>

<h3>Description</h3>

<p>It is used to obtain transcription, processing and degradation rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'pulseTDmodel'
getParams(object, stage, genename = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getParams+2B2CpulseTDmodel-method_+3A_object">object</code></td>
<td>
<p>An object of class 'pulseTDmodel'</p>
</td></tr>
<tr><td><code id="getParams+2B2CpulseTDmodel-method_+3A_stage">stage</code></td>
<td>
<p>A character such as transcription, processing or degradation for rates.</p>
</td></tr>
<tr><td><code id="getParams+2B2CpulseTDmodel-method_+3A_genename">genename</code></td>
<td>
<p>A vector of gene names, default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix containing the rates
</p>

<hr>
<h2 id='getRates'>Get rate values</h2><span id='topic+getRates'></span>

<h3>Description</h3>

<p>It is used to calculate the transcriptional dynamic rate of a gene.
You can get the discrete or continuous rate values of the measurement time points.
At the same time, it has a predictive function that provides rate values for any future time node or any range of time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRates(object, stage, timevector = NULL, genename = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRates_+3A_object">object</code></td>
<td>
<p>a 'pulseTDmodel' that has been calculated with <code><a href="#topic+estimateParams">estimateParams</a></code></p>
</td></tr>
<tr><td><code id="getRates_+3A_stage">stage</code></td>
<td>
<p>A character is one of three stages of transcriptional dynamics, transcription, processing and degradation.</p>
</td></tr>
<tr><td><code id="getRates_+3A_timevector">timevector</code></td>
<td>
<p>A vector of time, which can be any time, defaults to NULL
</p>

<ul>
<li><p> If it is NULL: only calculate the transcriptional dynamics of the measurement time node, the time is obtained by t_time in pulseTDmodel
</p>
</li>
<li><p> If the time vector is positive: calculate the rate value for a given time, or the predicted rate if the time exceeds the measurement time node
</p>
</li>
<li><p> If the time vector is negative: Calculate the rate value at a given time, which represents the calculation of the transcriptional dynamic rate before the prediction of the experimental measurement.
</p>
</li></ul>
</td></tr>
<tr><td><code id="getRates_+3A_genename">genename</code></td>
<td>
<p>a vector, default is NULL
</p>

<ul>
<li><p> If it is NULL: Calculate the transcriptional dynamic rate of all genes
</p>
</li>
<li><p> If it is a gene vector: only calculate the transcriptional dynamic rate of a given gene vector
</p>
</li>
<li><p> If it is a numerical vector: only calculate the transcriptional dynamic rate of the gene corresponding to a given value.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix, each row representing a gene, each column with a table to calculate the time node, and the value representing the rate of transcriptional dynamics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(file.path(system.file(package="pulseTD"),"data","pulseRates.RData"))
pulseRates_correct &lt;- correctionParams(pulseRates)
transcription = getRates(pulseRates_correct,'transcription')
degradation = getRates(pulseRates_correct, 'degradation')
processing = getRates(pulseRates_correct, 'processing')
head(transcription)
head(degradation)
head(processing)
trans = getRates(pulseRates_correct, 'transcription', timevector=c(0,1,2,3))
head(transcription)
trans=getRates(pulseRates_correct, 'transcription', timevector=c(2,3),genename=c(2,3))
head(transcription)
</code></pre>

<hr>
<h2 id='getRates+2CpulseTDmodel-method'>getRates-method</h2><span id='topic+getRates+2CpulseTDmodel-method'></span>

<h3>Description</h3>

<p>It is used to obtain transcription, processing and degradation rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'pulseTDmodel'
getRates(object, stage, timevector = NULL,
  genename = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRates+2B2CpulseTDmodel-method_+3A_object">object</code></td>
<td>
<p>An object of class 'pulseTDmodel'</p>
</td></tr>
<tr><td><code id="getRates+2B2CpulseTDmodel-method_+3A_stage">stage</code></td>
<td>
<p>A character such as transcription, processing or degradation for rates.</p>
</td></tr>
<tr><td><code id="getRates+2B2CpulseTDmodel-method_+3A_timevector">timevector</code></td>
<td>
<p>A vector of times, default is NULL.</p>
</td></tr>
<tr><td><code id="getRates+2B2CpulseTDmodel-method_+3A_genename">genename</code></td>
<td>
<p>A vector of gene names, default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix containing the rates
</p>

<hr>
<h2 id='integrateA'>integral function</h2><span id='topic+integrateA'></span>

<h3>Description</h3>

<p>Calculate the exon expression value of the label time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrateA(xita, timegrid, tl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrateA_+3A_xita">xita</code></td>
<td>
<p>Six parameters of the pulse function :h0, h1, h2, t1, t2, beta</p>
</td></tr>
<tr><td><code id="integrateA_+3A_timegrid">timegrid</code></td>
<td>
<p>a vector or point of time</p>
</td></tr>
<tr><td><code id="integrateA_+3A_tl">tl</code></td>
<td>
<p>label time</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector or point of exon expression value
</p>

<hr>
<h2 id='integrateAC'>integral function</h2><span id='topic+integrateAC'></span>

<h3>Description</h3>

<p>Calculate the expression values at different time points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrateAC(xita, timegrid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrateAC_+3A_xita">xita</code></td>
<td>
<p>Six parameters of the pulse function :h0, h1, h2, t1, t2, beta</p>
</td></tr>
<tr><td><code id="integrateAC_+3A_timegrid">timegrid</code></td>
<td>
<p>a vector or point of time</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector or point of exon expression value
</p>

<hr>
<h2 id='plotRates'>draw Method</h2><span id='topic+plotRates'></span>

<h3>Description</h3>

<p>It is used to draw a rate image that contains six images.
The expression values of the real genes and the fitted expression values are plotted separately, and the rate of the change is varied with time.
Can also plot predicted expression values and rate values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRates(object, genename, predict = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRates_+3A_object">object</code></td>
<td>
<p>a pulseTDmodel that has been calculated with <code><a href="#topic+estimateParams">estimateParams</a></code></p>
</td></tr>
<tr><td><code id="plotRates_+3A_genename">genename</code></td>
<td>
<p>The name of the gene to be drawn</p>
</td></tr>
<tr><td><code id="plotRates_+3A_predict">predict</code></td>
<td>
<p>The time interval you want to draw,default is FALSE, or a vector of three values,(start, end, step)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>There is no return value, six pictures will be drawn separately.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('pulseRates', package='pulseTD')
plotRates(pulseRates, 'NM_001001181')
plotRates(pulseRates, 'NM_001001181', predict=c(0,180,20))
</code></pre>

<hr>
<h2 id='plotRates+2CpulseTDmodel-method'>plotRate-Method</h2><span id='topic+plotRates+2CpulseTDmodel-method'></span>

<h3>Description</h3>

<p>Draw the image of the fitting result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'pulseTDmodel'
plotRates(object, genename, predict = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRates+2B2CpulseTDmodel-method_+3A_object">object</code></td>
<td>
<p>An object of class 'pulseTDmodel'</p>
</td></tr>
<tr><td><code id="plotRates+2B2CpulseTDmodel-method_+3A_genename">genename</code></td>
<td>
<p>The name of the gene you want to draw.</p>
</td></tr>
<tr><td><code id="plotRates+2B2CpulseTDmodel-method_+3A_predict">predict</code></td>
<td>
<p>The time interval you want to draw.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>

<hr>
<h2 id='predictExpression'>predict Expression</h2><span id='topic+predictExpression'></span>

<h3>Description</h3>

<p>It is used to predict the expression of all gene at a given time, including the expression of pre-mRNA (Precursor RNA) and the expression of total mRNA (Mature RNA).
End time and time interval can be arbitrarily defined
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictExpression(object, tg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predictExpression_+3A_object">object</code></td>
<td>
<p>a 'pulseTDmodel' that has been calculated with <code><a href="#topic+estimateParams">estimateParams</a></code></p>
</td></tr>
<tr><td><code id="predictExpression_+3A_tg">tg</code></td>
<td>
<p>A vector of points in time at which experimental data is collected, not allowed to be repeated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing predicted values for each gene and a 0.95 confidence interval
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('pulseRates', package='pulseTD')
pulseRates_correct = correctionParams(pulseRates)
TimeGrid = seq(0,180,15)
preExp = predictExpression(pulseRates_correct, tg=TimeGrid)
data('preExp', packages='pulseTD')
df = data.frame(preExp[['NM_001002011']])
head(df)
</code></pre>

<hr>
<h2 id='predictExpression+2CpulseTDmodel-method'>predictExpression-method</h2><span id='topic+predictExpression+2CpulseTDmodel-method'></span>

<h3>Description</h3>

<p>It is used to predict the expression of all gene at a given time,
including the expression of pre-mRNA (Precursor RNA) and the expression of total mRNA (Mature RNA).
End time and time interval can be arbitrarily defined
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'pulseTDmodel'
predictExpression(object, tg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predictExpression+2B2CpulseTDmodel-method_+3A_object">object</code></td>
<td>
<p>a 'pulseTDmodel' that has been calculated with <code><a href="#topic+estimateParams">estimateParams</a></code></p>
</td></tr>
<tr><td><code id="predictExpression+2B2CpulseTDmodel-method_+3A_tg">tg</code></td>
<td>
<p>A vector of points in time at which experimental data is collected, not allowed to be repeated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing predicted values for each gene and a 0.95 confidence interval
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('pulseRates', package='pulseTD')
pulseRates_correct = correctionParams(pulseRates)
TimeGrid = seq(0,180,15)
preExp = predictExpression(pulseRates_correct, tg=TimeGrid)
data('preExp', packages='pulseTD')
df = data.frame(preExp[['NM_001002011']])
head(df)

</code></pre>

<hr>
<h2 id='preExp'>A gene list that stores predicted values of gene expression and a 0.95 confidence interval.</h2><span id='topic+preExp'></span>

<h3>Description</h3>

<p>A gene list that stores predicted values of gene expression and a 0.95 confidence interval.
</p>


<h3>Format</h3>

<p>A list</p>

<hr>
<h2 id='pulseModel'>pulse function</h2><span id='topic+pulseModel'></span>

<h3>Description</h3>

<p>Pulse function is a pulse function J Comput Biol (2009 Feb) &lt;doi:10.1089/cmb.2008.13TT&gt;.
This is a multiplication of two sigmoid functions, the parameter vector Theta = (h0, h1, h2, t1 t2, beta),
where h0, h1, h2, represent the initial state rate value, peak value and the steady state rate value is reached again,
t1 and t2 are the maximum times of the first and second rise or fall changes, respectively,
and beta is the slope of the two changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pulseModel(xita, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pulseModel_+3A_xita">xita</code></td>
<td>
<p>Six parameters of the pulse function :h0, h1, h2, t1, t2, beta. Details <code><a href="#topic+getParams">getParams</a></code></p>
</td></tr>
<tr><td><code id="pulseModel_+3A_x">x</code></td>
<td>
<p>a vector or point of time</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector or point of pulse function value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(file.path(system.file(package="pulseTD"),"data","pulseRates.RData"))
TimeGrid &lt;- c(0, 15, 30, 45, 60, 75, 90, 105, 120, 135, 150, 165, 180)
pulseRates_correct &lt;- correctionParams(pulseRates)
transcription_params = getParams(pulseRates_correct,'transcription')
degradation_params = getParams(pulseRates_correct, 'degradation')
processing_params = getParams(pulseRates_correct, 'processing')
###
transcription_pulse = pulseModel(as.matrix(transcription_params[1,]), TimeGrid)
degradation_pulse = pulseModel(as.matrix(degradation_params[1,]), TimeGrid)
processing_pulse = pulseModel(as.matrix(processing_params[1,]), TimeGrid)
</code></pre>

<hr>
<h2 id='pulseRates'>A pulseTDModel Rdata file containing 100 gene expression data,
transcriptional dynamic data already related parameters.</h2><span id='topic+pulseRates'></span>

<h3>Description</h3>

<p>A pulseTDModel Rdata file containing 100 gene expression data,
transcriptional dynamic data already related parameters.
</p>


<h3>Format</h3>

<p>A pulseTD model</p>

<hr>
<h2 id='pulseTDmodel-class'>pulseTD-model</h2><span id='topic+pulseTDmodel-class'></span><span id='topic+pulseTDmodel'></span>

<h3>Description</h3>

<p>'pulseTD-model' is S4 class for storing calculation results.
It includes pulse model parameters corresponding to transcription rate, processing rate and degradation rate, and also stores some basic parameters of the calculation process, such as gene name, sequencing time, labeling time and so on.
</p>


<h3>Slots</h3>


<dl>
<dt><code>genenames</code></dt><dd><p>All gene names entered by the user.</p>
</dd>
<dt><code>t_time</code></dt><dd><p>Storage time series.</p>
</dd>
<dt><code>tL</code></dt><dd><p>4sU labeled duration</p>
</dd>
<dt><code>fitfailure</code></dt><dd><p>Gene vector for parameter optimization failure</p>
</dd>
<dt><code>filterExpression.foursu_exons</code></dt><dd><p>'AnnotatedDataFrame' that stores 4sU-labeled exons expression values</p>
</dd>
<dt><code>filterExpression.total_exons</code></dt><dd><p>'AnnotatedDataFrame' that stores total exons expression values</p>
</dd>
<dt><code>filterExpression.total_introns</code></dt><dd><p>'AnnotatedDataFrame' that stores total introns expression values</p>
</dd>
<dt><code>ratesPar.degradation</code></dt><dd><p>'AnnotatedDataFrame' that stores degradation pulse parameters</p>
</dd>
<dt><code>ratesPar.processing</code></dt><dd><p>'AnnotatedDataFrame' that stores processing pulse parameters</p>
</dd>
<dt><code>ratesPar.transcription</code></dt><dd><p>'AnnotatedDataFrame' that stores transcription pulse parameters</p>
</dd>
<dt><code>ratesPar.score</code></dt><dd><p>'AnnotatedDataFrame' that stores the fit scores</p>
</dd>
</dl>

<hr>
<h2 id='rpkmres'>a list containing the expression values of exons and introns</h2><span id='topic+rpkmres'></span>

<h3>Description</h3>

<p>a list containing the expression values of exons and introns
</p>


<h3>Format</h3>

<p>A list</p>

<hr>
<h2 id='rpkmSim'>A simulated gene expressing Rdata file that contains expression data for total RNA, precursor RNA, and 4sU-labeled RNA.</h2><span id='topic+rpkmSim'></span>

<h3>Description</h3>

<p>A simulated gene expressing Rdata file that contains expression data for total RNA, precursor RNA, and 4sU-labeled RNA.
</p>


<h3>Format</h3>

<p>A list</p>

<hr>
<h2 id='splitA6Params'>esitmate transcription parameters</h2><span id='topic+splitA6Params'></span>

<h3>Description</h3>

<p>It is used to solve the transcription parameters, including the error function and the optimization function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitA6Params(TL, tL, TimeGrid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitA6Params_+3A_tl">TL</code></td>
<td>
<p>Exon expression value of 4sU label RNA</p>
</td></tr>
<tr><td><code id="splitA6Params_+3A_tl">tL</code></td>
<td>
<p>label time</p>
</td></tr>
<tr><td><code id="splitA6Params_+3A_timegrid">TimeGrid</code></td>
<td>
<p>a vector  of times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of transcription parameters,contains pulse parameters and fitting error values
</p>

<hr>
<h2 id='splitB4Params'>esitmate degradation parameters</h2><span id='topic+splitB4Params'></span>

<h3>Description</h3>

<p>It is used to solve the degradation parameters, including the error function and the optimization function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitB4Params(TT, pre_a, TimeGrid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitB4Params_+3A_tt">TT</code></td>
<td>
<p>exon expression value of total RNA</p>
</td></tr>
<tr><td><code id="splitB4Params_+3A_pre_a">pre_a</code></td>
<td>
<p>Parameter vector of transcription pulse function</p>
</td></tr>
<tr><td><code id="splitB4Params_+3A_timegrid">TimeGrid</code></td>
<td>
<p>a vector  of times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of degradation parameters,contains pulse parameters and fitting error values
</p>

<hr>
<h2 id='splitC5Params'>esitmate processing parameters</h2><span id='topic+splitC5Params'></span>

<h3>Description</h3>

<p>It is used to solve the processing parameters, including the error function and the optimization function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitC5Params(PT, pre_a, TimeGrid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitC5Params_+3A_pt">PT</code></td>
<td>
<p>Intron expression value of total RNA</p>
</td></tr>
<tr><td><code id="splitC5Params_+3A_pre_a">pre_a</code></td>
<td>
<p>Parameter vector of transcription pulse function</p>
</td></tr>
<tr><td><code id="splitC5Params_+3A_timegrid">TimeGrid</code></td>
<td>
<p>a vector of times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of processing parameters,contains pulse parameters and fitting error values
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
