<!DOCTYPE html><html><head><title>Help for package pureseqtmr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pureseqtmr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#are_tmhs'><p>Are the sequences transmembrance helices?</p></a></li>
<li><a href='#are_valid_protein_sequences'><p>Determine if these are all valid protein sequences</p></a></li>
<li><a href='#calc_distance_to_tmh_center_from_topology'><p>Calculate the the distance for each amino acid to the center</p>
of the TMH</a></li>
<li><a href='#calc_distance_to_tmh_center_from_topology_str'><p>Calculate the the distance for each amino acid to the center</p>
of the TMH</a></li>
<li><a href='#calc_distance_to_tmh_center_from_topology_str_cpp_stl'><p>Use Rcpp to calculate the distance to a TMH center</p></a></li>
<li><a href='#check_protein_sequence'><p>Check one protein sequence</p></a></li>
<li><a href='#check_protein_sequences'><p>Check one or more protein sequences</p></a></li>
<li><a href='#check_pureseqtm_installation'><p>Checks the installation of PureseqTM.</p>
Throws a helpful error message if incomplete,
else does nothing</a></li>
<li><a href='#check_topology'><p>Check if the topology is valid.</p></a></li>
<li><a href='#check_topology_str'><p>Check if the topology string is valid.</p>
Will stop if not.</a></li>
<li><a href='#convert_tmhmm_to_pureseqtm_topology'><p>Convert a TMHMM topology to a PureseqTM topology</p></a></li>
<li><a href='#count_n_tmhs'><p>Count the number of TMHs in a topology</p></a></li>
<li><a href='#create_pureseqtm_files'><p>Create the five PureseqTM output files, by running PureseqTM.</p></a></li>
<li><a href='#create_pureseqtm_proteome_file'><p>Create the output file of a PureseqTM proteome run</p></a></li>
<li><a href='#default_params_doc'><p>This function does nothing. It is intended to inherit is parameters'</p>
documentation.</a></li>
<li><a href='#get_default_pureseqtm_folder'><p>Get the path to the folder where this package installs</p>
PureseqTM by default</a></li>
<li><a href='#get_example_filename'><p>Get the full path to a PureseqTM example file.</p></a></li>
<li><a href='#get_example_filenames'><p>Get the full path to all PureseqTM example files</p></a></li>
<li><a href='#get_pureseqtm_url'><p>Get the URL of the PureseqTM source code</p></a></li>
<li><a href='#get_pureseqtm_version'><p>Get the PureseqTM version</p></a></li>
<li><a href='#install_pureseqtm'><p>Install PureseqTM to a local folder</p></a></li>
<li><a href='#is_on_appveyor'><p>Determines if the environment is AppVeyor</p></a></li>
<li><a href='#is_on_ci'><p>Determines if the environment is a continuous integration service</p></a></li>
<li><a href='#is_on_github_actions'><p>Determines if the environment is GitHub Actions</p></a></li>
<li><a href='#is_on_travis'><p>Determines if the environment is Travis CI</p></a></li>
<li><a href='#is_protein_name_line'><p>Is the line of text the name of a protein, as used within</p>
a FASTA filename?</a></li>
<li><a href='#is_pureseqtm_installed'><p>Measure if PureseqTM is installed locally</p></a></li>
<li><a href='#is_tmh'><p>Determine if the protein sequence contains at least one</p>
transmembrane helix.</a></li>
<li><a href='#is_topology_line'><p>Is the line of text the topology, as used within</p>
a FASTA filename?</a></li>
<li><a href='#is_valid_protein_sequence'><p>Determine if this a valid protein sequence</p></a></li>
<li><a href='#load_fasta_file_as_tibble'><p>Parse a FASTA file to a table</p>
with a <code>name</code> and <code>sequence</code> column</a></li>
<li><a href='#load_fasta_file_as_tibble_cpp'><p>Parse a FASTA file to a table</p>
with a <code>name</code> and <code>sequence</code> column</a></li>
<li><a href='#load_fasta_file_as_tibble_cpp_raw'><p>Use Rcpp to load a FASTA file</p></a></li>
<li><a href='#load_fasta_file_as_tibble_r'><p>Parse a FASTA file to a table</p>
with a <code>name</code> and <code>sequence</code> column</a></li>
<li><a href='#load_topology_file_as_tibble'><p>Parse a topology (<code>.topo</code>) file to a table</p>
with a <code>name</code> and <code>topology</code> column</a></li>
<li><a href='#mock_predict_topologies_from_sequences'><p>Do a mock prediction directy on a protein sequence,</p>
as can be useful in testing
Use predict_topologies_from_sequences for doing a real prediction.</a></li>
<li><a href='#mock_predict_topology'><p>Do a mock prediction of the topology of proteins</p></a></li>
<li><a href='#parse_pureseqtm_proteome_text'><p>Parse the output of a call to <code>PureseqTM_proteome.sh</code></p></a></li>
<li><a href='#plot_topology'><p>Plot the topology</p></a></li>
<li><a href='#predict_topologies_from_sequences'><p>Run PureseqTM directy on a protein sequence</p></a></li>
<li><a href='#predict_topology'><p>Predict the topology of proteins from file</p></a></li>
<li><a href='#predict_topology_from_sequence'><p>Run PureseqTM directy on a protein sequence</p></a></li>
<li><a href='#pureseqtmr'><p>pureseqtmr: estimate the topoplogy of membrane proteins</p></a></li>
<li><a href='#pureseqtmr_report'><p>Create a pureseqtmr report, to be used when reporting bugs</p></a></li>
<li><a href='#run_pureseqtm_proteome'><p>Run PureseqTM on a proteome</p></a></li>
<li><a href='#save_tibble_as_fasta_file'><p>Save the first two columns of a tibble as a FASTA file</p></a></li>
<li><a href='#tally_tmhs'><p>Count the number of transmembrane helices in a topology</p></a></li>
<li><a href='#uninstall_pureseqtm'><p>Uninstall PureseqTM</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Predict Transmembrane Protein Topology</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Proteins reside in either the cell plasma or in the
    cell membrane. A membrane protein goes through the 
    membrane at least once. Given the amino acid sequence of a
    membrane protein, the tool
    'PureseqTM' (<a href="https://github.com/PureseqTM/pureseqTM_package">https://github.com/PureseqTM/pureseqTM_package</a>,
    as described in "Efficient And Accurate Prediction Of Transmembrane 
    Topology From Amino acid sequence only.", Wang, Qing, et al (2019), 
    &lt;<a href="https://doi.org/10.1101%2F627307">doi:10.1101/627307</a>&gt;),
    can predict the topology of a membrane protein. This package
    allows one to use 'PureseqTM' from R.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, devtools, dplyr, ggplot2, Peptides, plyr,
rappdirs, readr, stringr, tibble, Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, markdown, rmarkdown, profvis</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/richelbilderbeek/pureseqtmr/">https://github.com/richelbilderbeek/pureseqtmr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/richelbilderbeek/pureseqtmr/">https://github.com/richelbilderbeek/pureseqtmr/</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>PureseqTM
(https://github.com/PureseqTM/pureseqTM_package)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-06 13:18:26 UTC; richel</td>
</tr>
<tr>
<td>Author:</td>
<td>Richèl J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richèl J.C. Bilderbeek &lt;richel@richelbilderbeek.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-06 13:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='are_tmhs'>Are the sequences transmembrance helices?</h2><span id='topic+are_tmhs'></span>

<h3>Description</h3>

<p>Are the sequences transmembrance helices?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_tmhs(protein_sequences, folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_tmhs_+3A_protein_sequences">protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
<tr><td><code id="are_tmhs_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of booleans of the same length as the
number of sequences. The ith element is <a href="base.html#topic+TRUE">TRUE</a> if
the ith protein sequence is a transmembrane helix
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  sequences &lt;- c(
    "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
    "VVIILTIRGNILVIMAVSLE"
  )
  are_tmhs(sequences)
}
</code></pre>

<hr>
<h2 id='are_valid_protein_sequences'>Determine if these are all valid protein sequences</h2><span id='topic+are_valid_protein_sequences'></span>

<h3>Description</h3>

<p>Determine if these are all valid protein sequences,
as can be used in topology prediction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_valid_protein_sequences(protein_sequences, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_valid_protein_sequences_+3A_protein_sequences">protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
<tr><td><code id="are_valid_protein_sequences_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the protein sequence is valid
</p>

<hr>
<h2 id='calc_distance_to_tmh_center_from_topology'>Calculate the the distance for each amino acid to the center
of the TMH</h2><span id='topic+calc_distance_to_tmh_center_from_topology'></span>

<h3>Description</h3>

<p>Calculate the the distance for each amino acid to the center
of the TMH
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_distance_to_tmh_center_from_topology(topology)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_distance_to_tmh_center_from_topology_+3A_topology">topology</code></td>
<td>
<p>the topology as a <a href="tibble.html#topic+tibble">tibble</a>
with the columns 'name' and 'topology',
where the 'name' column hold all the proteins' names,
and 'topology' contains the respective topologies as strings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with the columns 'name'
and 'position' and 'distance_to_tmh_center'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='calc_distance_to_tmh_center_from_topology_str'>Calculate the the distance for each amino acid to the center
of the TMH</h2><span id='topic+calc_distance_to_tmh_center_from_topology_str'></span>

<h3>Description</h3>

<p>Calculate the the distance for each amino acid to the center
of the TMH
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_distance_to_tmh_center_from_topology_str(topology_str)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_distance_to_tmh_center_from_topology_str_+3A_topology_str">topology_str</code></td>
<td>
<p>the topology as a string,
for example <code>000000111100000</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with the columns
'position' and 'distance_to_tmh_center'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='calc_distance_to_tmh_center_from_topology_str_cpp_stl'>Use Rcpp to calculate the distance to a TMH center</h2><span id='topic+calc_distance_to_tmh_center_from_topology_str_cpp_stl'></span>

<h3>Description</h3>

<p>Use Rcpp to calculate the distance to a TMH center
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_distance_to_tmh_center_from_topology_str_cpp_stl(topology_str)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_distance_to_tmh_center_from_topology_str_cpp_stl_+3A_topology_str">topology_str</code></td>
<td>
<p>a topology as a string</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with distances
</p>

<hr>
<h2 id='check_protein_sequence'>Check one protein sequence</h2><span id='topic+check_protein_sequence'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the protein sequence is invalid,
with a helpful error message.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_protein_sequence(protein_sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_protein_sequence_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, with
the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code>.
Use <a href="#topic+check_protein_sequence">check_protein_sequence</a> to check if
a protein sequence is valid.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A protein sequence is invalid if:
</p>

<ul>
<li><p> it has zero, two or more sequences
</p>
</li>
<li><p> the sequence contains zero, 1 or 2 amino acids
</p>
</li>
<li><p> the sequence contains characters that are not in the
amino acid uppercase alphabet,
that is <code>ACDEFGHIKLMNPQRSTVWY</code>
</p>
</li></ul>



<h3>Value</h3>

<p>nothing.
Will <a href="base.html#topic+stop">stop</a> if the protein sequence is invalid,
with a helpful error message.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_protein_sequence("FAMILYVW")
</code></pre>

<hr>
<h2 id='check_protein_sequences'>Check one or more protein sequences</h2><span id='topic+check_protein_sequences'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the protein sequence is invalid,
with a helpful error message.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_protein_sequences(protein_sequences)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_protein_sequences_+3A_protein_sequences">protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>A protein sequence is invalid if:
</p>

<ul>
<li><p> it has zero, two or more sequences
</p>
</li>
<li><p> the sequence contains zero, 1 or 2 amino acids
</p>
</li>
<li><p> the sequence contains characters that are not in the
amino acid uppercase alphabet,
that is <code>ACDEFGHIKLMNPQRSTVWY</code>
</p>
</li></ul>



<h3>Value</h3>

<p>nothing.
Will <a href="base.html#topic+stop">stop</a> at the first invalid protein sequence,
with a helpful error message.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_protein_sequences(c("FAMILYVW", "FAMILYVW"))
</code></pre>

<hr>
<h2 id='check_pureseqtm_installation'>Checks the installation of PureseqTM.
Throws a helpful error message if incomplete,
else does nothing</h2><span id='topic+check_pureseqtm_installation'></span>

<h3>Description</h3>

<p>Checks the installation of PureseqTM.
Throws a helpful error message if incomplete,
else does nothing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_pureseqtm_installation(folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_pureseqtm_installation_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will <a href="base.html#topic+stop">stop</a> with a helpful error message if
PureseqTM is not installed.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  check_pureseqtm_installation()
}
</code></pre>

<hr>
<h2 id='check_topology'>Check if the topology is valid.</h2><span id='topic+check_topology'></span>

<h3>Description</h3>

<p>Check if the argument is of the same type as a predicted topology,
as can be created with <a href="#topic+predict_topology">predict_topology</a>.
Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_topology(topology)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_topology_+3A_topology">topology</code></td>
<td>
<p>the topology as a <a href="tibble.html#topic+tibble">tibble</a>
with the columns 'name' and 'topology',
where the 'name' column hold all the proteins' names,
and 'topology' contains the respective topologies as strings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will <a href="base.html#topic+stop">stop</a> with a helpful error message if
the topology is invalid.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  fasta_filename &lt;- get_example_filename("1bhaA.fasta")
  topology &lt;- predict_topology(fasta_filename)
  check_topology(topology)
}
</code></pre>

<hr>
<h2 id='check_topology_str'>Check if the topology string is valid.
Will <a href="base.html#topic+stop">stop</a> if not.</h2><span id='topic+check_topology_str'></span>

<h3>Description</h3>

<p>Check if the topology string is valid.
Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_topology_str(topology_str)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_topology_str_+3A_topology_str">topology_str</code></td>
<td>
<p>the topology as a string,
for example <code>000000111100000</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will <a href="base.html#topic+stop">stop</a> with a helpful error message if
the topology is invalid.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_topology_str("0000000000000000000000000011111111111111111100000")
</code></pre>

<hr>
<h2 id='convert_tmhmm_to_pureseqtm_topology'>Convert a TMHMM topology to a PureseqTM topology</h2><span id='topic+convert_tmhmm_to_pureseqtm_topology'></span>

<h3>Description</h3>

<p>Convert a TMHMM topology to a PureseqTM topology
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_tmhmm_to_pureseqtm_topology(tmhmm_topology)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_tmhmm_to_pureseqtm_topology_+3A_tmhmm_topology">tmhmm_topology</code></td>
<td>
<p>topology as used by <code>TMHMM</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with column names
<code>name</code> and <code>topology</code>, as can be checked by <a href="#topic+check_topology">check_topology</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmhmm_topo_filename &lt;- system.file(
  "extdata", "UP000005640_9606_no_u.tmhmm", package = "pureseqtmr"
)
tmhmm_topology &lt;- load_topology_file_as_tibble(tmhmm_topo_filename)
convert_tmhmm_to_pureseqtm_topology(tmhmm_topology)
</code></pre>

<hr>
<h2 id='count_n_tmhs'>Count the number of TMHs in a topology</h2><span id='topic+count_n_tmhs'></span>

<h3>Description</h3>

<p>Count the number of TMHs in a topology
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_n_tmhs(topology_strs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_n_tmhs_+3A_topology_strs">topology_strs</code></td>
<td>
<p>the topologies as zero, one oor more strings,
for example <code>c("0", "1")</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>count_n_tmhs("00000000000000000000000000")
count_n_tmhs("00000000001111100000000000")
count_n_tmhs(c("0", "1"))
</code></pre>

<hr>
<h2 id='create_pureseqtm_files'>Create the five PureseqTM output files, by running PureseqTM.</h2><span id='topic+create_pureseqtm_files'></span>

<h3>Description</h3>

<p>Create the five PureseqTM output files, by running PureseqTM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_pureseqtm_files(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  temp_folder_name = tempfile(pattern = "pureseqt_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_pureseqtm_files_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
<tr><td><code id="create_pureseqtm_files_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="create_pureseqtm_files_+3A_temp_folder_name">temp_folder_name</code></td>
<td>
<p>path of a temporary folder.
The folder does not need to exist.
Files that are out in this folder are not automatically
deleted, which is not a problem, as the default
path given by <a href="base.html#topic+tempdir">tempdir</a> is automatically cleaned
by the operating system</p>
</td></tr>
</table>


<h3>Value</h3>

<p>full path to the files created
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  fasta_filename &lt;- get_example_filename("1bhaA.fasta")
  create_pureseqtm_files(fasta_filename)
}
</code></pre>

<hr>
<h2 id='create_pureseqtm_proteome_file'>Create the output file of a PureseqTM proteome run</h2><span id='topic+create_pureseqtm_proteome_file'></span>

<h3>Description</h3>

<p>Create the output file of a PureseqTM proteome run
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_pureseqtm_proteome_file(
  fasta_filename,
  topology_filename = tempfile(fileext = ".top"),
  folder_name = get_default_pureseqtm_folder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_pureseqtm_proteome_file_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
<tr><td><code id="create_pureseqtm_proteome_file_+3A_topology_filename">topology_filename</code></td>
<td>
<p>name of the file to save a protein's
topology to</p>
</td></tr>
<tr><td><code id="create_pureseqtm_proteome_file_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the filename
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (is_pureseqtm_installed()) {
  fasta_filename &lt;- get_example_filename("1bhaA.fasta")
  create_pureseqtm_proteome_file(fasta_filename)
}
</code></pre>

<hr>
<h2 id='default_params_doc'>This function does nothing. It is intended to inherit is parameters'
documentation.</h2><span id='topic+default_params_doc'></span>

<h3>Description</h3>

<p>This function does nothing. It is intended to inherit is parameters'
documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_params_doc(
  download_url,
  fasta_filename,
  fasta_file_text,
  folder_name,
  protein_sequence,
  protein_sequences,
  pureseqtm_filename,
  pureseqtm_proteome_text,
  pureseqtm_result,
  pureseqtm_url,
  temp_fasta_filename,
  temp_folder_name,
  tmhmm_topology,
  topology,
  topology_filename,
  topology_str,
  topology_strs,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_params_doc_+3A_download_url">download_url</code></td>
<td>
<p>the URL to download PureseqTM from</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_file_text">fasta_file_text</code></td>
<td>
<p>text of a FASTA file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, with
the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code>.
Use <a href="#topic+check_protein_sequence">check_protein_sequence</a> to check if
a protein sequence is valid.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_protein_sequences">protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_pureseqtm_filename">pureseqtm_filename</code></td>
<td>
<p>filename to write the PureseqTM results to</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_pureseqtm_proteome_text">pureseqtm_proteome_text</code></td>
<td>
<p>the output of a call
to <code>PureseqTM_proteome.sh</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_pureseqtm_result">pureseqtm_result</code></td>
<td>
<p>the result of a PureseqTM run</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_pureseqtm_url">pureseqtm_url</code></td>
<td>
<p>URL of the PureseqTM git repository</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_temp_fasta_filename">temp_fasta_filename</code></td>
<td>
<p>temporary FASTA filename,
which will deleted after usage</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_temp_folder_name">temp_folder_name</code></td>
<td>
<p>path of a temporary folder.
The folder does not need to exist.
Files that are out in this folder are not automatically
deleted, which is not a problem, as the default
path given by <a href="base.html#topic+tempdir">tempdir</a> is automatically cleaned
by the operating system</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tmhmm_topology">tmhmm_topology</code></td>
<td>
<p>topology as used by <code>TMHMM</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_topology">topology</code></td>
<td>
<p>the topology as a <a href="tibble.html#topic+tibble">tibble</a>
with the columns 'name' and 'topology',
where the 'name' column hold all the proteins' names,
and 'topology' contains the respective topologies as strings.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_topology_filename">topology_filename</code></td>
<td>
<p>name of the file to save a protein's
topology to</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_topology_str">topology_str</code></td>
<td>
<p>the topology as a string,
for example <code>000000111100000</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_topology_strs">topology_strs</code></td>
<td>
<p>the topologies as zero, one oor more strings,
for example <code>c("0", "1")</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_default_pureseqtm_folder'>Get the path to the folder where this package installs
PureseqTM by default</h2><span id='topic+get_default_pureseqtm_folder'></span>

<h3>Description</h3>

<p>Get the path to the folder where this package installs
PureseqTM by default
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_pureseqtm_folder()
</code></pre>


<h3>Value</h3>

<p>the path to the folder where this package installs
PureseqTM by default
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_pureseqtm_folder()
</code></pre>

<hr>
<h2 id='get_example_filename'>Get the full path to a PureseqTM example file.</h2><span id='topic+get_example_filename'></span>

<h3>Description</h3>

<p>Get the full path to a PureseqTM example file.
If the filename specified is not a PureseqTM example file,
this function will <a href="base.html#topic+stop">stop</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_filename(filename, folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_example_filename_+3A_filename">filename</code></td>
<td>
<p>name of the example file, without the path</p>
</td></tr>
<tr><td><code id="get_example_filename_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the full path to a PureseqTM example file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+get_example_filenames">get_example_filenames</a> to get all PureseqTM
example filenames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  get_example_filename("1bhaA.fasta")
}
</code></pre>

<hr>
<h2 id='get_example_filenames'>Get the full path to all PureseqTM example files</h2><span id='topic+get_example_filenames'></span>

<h3>Description</h3>

<p>Get the full path to all PureseqTM example files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_filenames(folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_example_filenames_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with all PureseqTM example files
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+get_example_filename">get_example_filename</a> to get the full
path to a PureseqTM example file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  get_example_filenames()
}
</code></pre>

<hr>
<h2 id='get_pureseqtm_url'>Get the URL of the PureseqTM source code</h2><span id='topic+get_pureseqtm_url'></span>

<h3>Description</h3>

<p>Get the URL of the PureseqTM source code
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pureseqtm_url()
</code></pre>


<h3>Value</h3>

<p>a URL as a character vector of one element
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_pureseqtm_url()
</code></pre>

<hr>
<h2 id='get_pureseqtm_version'>Get the PureseqTM version</h2><span id='topic+get_pureseqtm_version'></span>

<h3>Description</h3>

<p>Get the PureseqTM version
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pureseqtm_version(folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pureseqtm_version_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a version number as a character vector of one element,
for example <code>v0.10</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  get_pureseqtm_version()
}
</code></pre>

<hr>
<h2 id='install_pureseqtm'>Install PureseqTM to a local folder</h2><span id='topic+install_pureseqtm'></span>

<h3>Description</h3>

<p>Install PureseqTM to a local folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_pureseqtm(
  folder_name = get_default_pureseqtm_folder(),
  pureseqtm_url = get_pureseqtm_url()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_pureseqtm_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="install_pureseqtm_+3A_pureseqtm_url">pureseqtm_url</code></td>
<td>
<p>URL of the PureseqTM git repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  install_pureseqtm()

## End(Not run)
</code></pre>

<hr>
<h2 id='is_on_appveyor'>Determines if the environment is AppVeyor</h2><span id='topic+is_on_appveyor'></span>

<h3>Description</h3>

<p>Determines if the environment is AppVeyor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_appveyor()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on AppVeyor, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_appveyor()) {
    message("Running on AppVeyor")
  }
</code></pre>

<hr>
<h2 id='is_on_ci'>Determines if the environment is a continuous integration service</h2><span id='topic+is_on_ci'></span>

<h3>Description</h3>

<p>Determines if the environment is a continuous integration service
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on AppVeyor or Travis CI, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_ci()) {
    message("Running on a continuous integration service")
  }
</code></pre>

<hr>
<h2 id='is_on_github_actions'>Determines if the environment is GitHub Actions</h2><span id='topic+is_on_github_actions'></span>

<h3>Description</h3>

<p>Determines if the environment is GitHub Actions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_github_actions()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on GitHub Actions, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_github_actions()) {
    message("Running on GitHub Actions")
  }
</code></pre>

<hr>
<h2 id='is_on_travis'>Determines if the environment is Travis CI</h2><span id='topic+is_on_travis'></span>

<h3>Description</h3>

<p>Determines if the environment is Travis CI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_travis()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on Travis CI, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_ci()) {
    message("Running on Travis CI")
  }
</code></pre>

<hr>
<h2 id='is_protein_name_line'>Is the line of text the name of a protein, as used within
a FASTA filename?</h2><span id='topic+is_protein_name_line'></span>

<h3>Description</h3>

<p>Is the line of text the name of a protein, as used within
a FASTA filename?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_protein_name_line(line)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_protein_name_line_+3A_line">line</code></td>
<td>
<p>line of text from a FASTA filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if the line can be the name of a protein in a FASTA file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_protein_name_line("&gt;5H2A_CRIGR")
</code></pre>

<hr>
<h2 id='is_pureseqtm_installed'>Measure if PureseqTM is installed locally</h2><span id='topic+is_pureseqtm_installed'></span>

<h3>Description</h3>

<p>Measure if PureseqTM is installed locally
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_pureseqtm_installed(folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_pureseqtm_installed_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> is PureseqTM is installed locally,
<a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_pureseqtm_installed()
</code></pre>

<hr>
<h2 id='is_tmh'>Determine if the protein sequence contains at least one
transmembrane helix.</h2><span id='topic+is_tmh'></span>

<h3>Description</h3>

<p>Determine if the protein sequence contains at least one
transmembrane helix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_tmh(protein_sequence, folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_tmh_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, with
the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code>.
Use <a href="#topic+check_protein_sequence">check_protein_sequence</a> to check if
a protein sequence is valid.</p>
</td></tr>
<tr><td><code id="is_tmh_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if the protein sequence contains at least one
transmembrane helix
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  # This sequence is a TMH
  is_tmh("QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM")

  # This sequence is not a TMH
  is_tmh("VVIILTIRGNILVIMAVSLE")
}
</code></pre>

<hr>
<h2 id='is_topology_line'>Is the line of text the topology, as used within
a FASTA filename?</h2><span id='topic+is_topology_line'></span>

<h3>Description</h3>

<p>Is the line of text the topology, as used within
a FASTA filename? In this context, a topology is a string of zeroes
and ones, in which a one denotes that that amino acid is within
the membrane.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_topology_line(line)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_topology_line_+3A_line">line</code></td>
<td>
<p>line of text from a FASTA filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if the line can be the text of a topology in a
FASTA file.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This is a valid topology
is_topology_line("000010101011")

# This is an invalid topology
is_topology_line("invalid")
</code></pre>

<hr>
<h2 id='is_valid_protein_sequence'>Determine if this a valid protein sequence</h2><span id='topic+is_valid_protein_sequence'></span>

<h3>Description</h3>

<p>Determine if this is a valid protein sequence,
as can be used in topology prediction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_valid_protein_sequence(protein_sequence, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_valid_protein_sequence_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, with
the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code>.
Use <a href="#topic+check_protein_sequence">check_protein_sequence</a> to check if
a protein sequence is valid.</p>
</td></tr>
<tr><td><code id="is_valid_protein_sequence_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the protein sequence is valid
</p>

<hr>
<h2 id='load_fasta_file_as_tibble'>Parse a FASTA file to a table
with a <code>name</code> and <code>sequence</code> column</h2><span id='topic+load_fasta_file_as_tibble'></span>

<h3>Description</h3>

<p>Parse a FASTA file to a table
with a <code>name</code> and <code>sequence</code> column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_fasta_file_as_tibble(fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_fasta_file_as_tibble_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a>
with a <code>name</code> and <code>sequence</code> column
</p>


<h3>See Also</h3>

<p>use <a href="#topic+load_fasta_file_as_tibble_cpp">load_fasta_file_as_tibble_cpp</a>
to directly call the C++ function that does the actual
work. Use <a href="#topic+load_fasta_file_as_tibble_r">load_fasta_file_as_tibble_r</a> to call
the (approx ten thousand times slower) R function
</p>

<hr>
<h2 id='load_fasta_file_as_tibble_cpp'>Parse a FASTA file to a table
with a <code>name</code> and <code>sequence</code> column</h2><span id='topic+load_fasta_file_as_tibble_cpp'></span>

<h3>Description</h3>

<p>Parse a FASTA file to a table
with a <code>name</code> and <code>sequence</code> column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_fasta_file_as_tibble_cpp(fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_fasta_file_as_tibble_cpp_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a>
with a <code>name</code> and <code>sequence</code> column
</p>

<hr>
<h2 id='load_fasta_file_as_tibble_cpp_raw'>Use Rcpp to load a FASTA file</h2><span id='topic+load_fasta_file_as_tibble_cpp_raw'></span>

<h3>Description</h3>

<p>Use Rcpp to load a FASTA file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_fasta_file_as_tibble_cpp_raw(fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_fasta_file_as_tibble_cpp_raw_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>FASTA filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two character vectors,
named 'name' and 'sequence'
</p>

<hr>
<h2 id='load_fasta_file_as_tibble_r'>Parse a FASTA file to a table
with a <code>name</code> and <code>sequence</code> column</h2><span id='topic+load_fasta_file_as_tibble_r'></span>

<h3>Description</h3>

<p>Parse a FASTA file to a table
with a <code>name</code> and <code>sequence</code> column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_fasta_file_as_tibble_r(fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_fasta_file_as_tibble_r_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a>
with a <code>name</code> and <code>sequence</code> column
</p>

<hr>
<h2 id='load_topology_file_as_tibble'>Parse a topology (<code>.topo</code>) file to a table
with a <code>name</code> and <code>topology</code> column</h2><span id='topic+load_topology_file_as_tibble'></span>

<h3>Description</h3>

<p>Parse a topology (<code>.topo</code>) file to a table
with a <code>name</code> and <code>topology</code> column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_topology_file_as_tibble(topology_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_topology_file_as_tibble_+3A_topology_filename">topology_filename</code></td>
<td>
<p>name of the file to save a protein's
topology to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a>
with a <code>name</code> and <code>topology</code> column,
as can be checked by <a href="#topic+check_topology">check_topology</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>topology_filename &lt;- system.file(
 "extdata", "100507436.topo", package = "pureseqtmr"
)
load_topology_file_as_tibble(topology_filename)
</code></pre>

<hr>
<h2 id='mock_predict_topologies_from_sequences'>Do a mock prediction directy on a protein sequence,
as can be useful in testing
Use <a href="#topic+predict_topologies_from_sequences">predict_topologies_from_sequences</a> for doing a real prediction.</h2><span id='topic+mock_predict_topologies_from_sequences'></span>

<h3>Description</h3>

<p>Do a mock prediction directy on a protein sequence,
as can be useful in testing
Use <a href="#topic+predict_topologies_from_sequences">predict_topologies_from_sequences</a> for doing a real prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mock_predict_topologies_from_sequences(protein_sequences)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mock_predict_topologies_from_sequences_+3A_protein_sequences">protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a topology as a string of zeroes and ones, where a one denotes
that the corresponding amino acid is located within the membrane.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>protein_sequence &lt;- paste0(
  "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
  "SLAIADMLLGFLVMPVSMLTILYGYRWP"
)
mock_predict_topologies_from_sequences(protein_sequence)
</code></pre>

<hr>
<h2 id='mock_predict_topology'>Do a mock prediction of the topology of proteins</h2><span id='topic+mock_predict_topology'></span>

<h3>Description</h3>

<p>Uses <a href="#topic+predict_topology">predict_topology</a> for doing a real prediction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mock_predict_topology(fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mock_predict_topology_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with the columns 'name'
and 'topology', where the 'name' column hold all the proteins' names,
and 'topology' contains all respective topologies.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fasta_filename &lt;- tempfile()
save_tibble_as_fasta_file(
  t = tibble::tibble(
    name = c("A", "B"),
    sequence = c("FAMILYVW", "VWFAMILY")
  ),
  fasta_filename = fasta_filename
)
mock_predict_topology(fasta_filename)
</code></pre>

<hr>
<h2 id='parse_pureseqtm_proteome_text'>Parse the output of a call to <code>PureseqTM_proteome.sh</code></h2><span id='topic+parse_pureseqtm_proteome_text'></span>

<h3>Description</h3>

<p>Parse the output of a call to <code>PureseqTM_proteome.sh</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_pureseqtm_proteome_text(pureseqtm_proteome_text)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_pureseqtm_proteome_text_+3A_pureseqtm_proteome_text">pureseqtm_proteome_text</code></td>
<td>
<p>the output of a call
to <code>PureseqTM_proteome.sh</code></p>
</td></tr>
</table>

<hr>
<h2 id='plot_topology'>Plot the topology</h2><span id='topic+plot_topology'></span>

<h3>Description</h3>

<p>Plot the topology
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_topology(topology)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_topology_+3A_topology">topology</code></td>
<td>
<p>the topology as a <a href="tibble.html#topic+tibble">tibble</a>
with the columns 'name' and 'topology',
where the 'name' column hold all the proteins' names,
and 'topology' contains the respective topologies as strings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="ggplot2.html#topic+ggplot">ggplot</a> that displays the topology of
one or more proteins
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed() &amp;&amp; is_on_ci()) {
  fasta_filename &lt;- get_example_filename("test_proteome.fasta")
  topology &lt;- predict_topology(fasta_filename)
  plot_topology(topology)
}
</code></pre>

<hr>
<h2 id='predict_topologies_from_sequences'>Run PureseqTM directy on a protein sequence</h2><span id='topic+predict_topologies_from_sequences'></span>

<h3>Description</h3>

<p>Run PureseqTM directy on a protein sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_topologies_from_sequences(
  protein_sequences,
  folder_name = get_default_pureseqtm_folder(),
  temp_fasta_filename = tempfile(fileext = ".fasta")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_topologies_from_sequences_+3A_protein_sequences">protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
<tr><td><code id="predict_topologies_from_sequences_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="predict_topologies_from_sequences_+3A_temp_fasta_filename">temp_fasta_filename</code></td>
<td>
<p>temporary FASTA filename,
which will deleted after usage</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a topology as a string of zeroes and ones, where a one denotes
that the corresponding amino acid is located within the membrane.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+mock_predict_topologies_from_sequences">mock_predict_topologies_from_sequences</a>
to mock the prediction of protein sequences, as can be useful
in testing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  protein_sequence &lt;- paste0(
    "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
    "SLAIADMLLGFLVMPVSMLTILYGYRWP"
  )
  predict_topology_from_sequence(protein_sequence)
}
</code></pre>

<hr>
<h2 id='predict_topology'>Predict the topology of proteins from file</h2><span id='topic+predict_topology'></span>

<h3>Description</h3>

<p>Predict the topology of zero, one or more proteins,
of which the names and sequences are stored in the FASTA format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_topology(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  topology_filename = tempfile(fileext = ".top")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_topology_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
<tr><td><code id="predict_topology_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="predict_topology_+3A_topology_filename">topology_filename</code></td>
<td>
<p>name of the file to save a protein's
topology to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with the columns 'name'
and 'topology', where the 'name' column hold all the proteins' names,
and 'topology' contains all respective topologies.
</p>


<h3>Note</h3>

<p>unlike PureseqTM, the topologies predicted are returned in
the same order as the original sequences.
A bugreport is posted at the PureseqTM GitHub repository
at <a href="https://github.com/PureseqTM/PureseqTM_Package/issues/11">https://github.com/PureseqTM/PureseqTM_Package/issues/11</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+mock_predict_topology">mock_predict_topology</a> to do a mock prediction,
as can be useful in testing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  fasta_filename &lt;- get_example_filename("1bhaA.fasta")
  predict_topology(fasta_filename)
}
</code></pre>

<hr>
<h2 id='predict_topology_from_sequence'>Run PureseqTM directy on a protein sequence</h2><span id='topic+predict_topology_from_sequence'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the protein sequence is shorter than three
amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_topology_from_sequence(
  protein_sequence,
  folder_name = get_default_pureseqtm_folder(),
  temp_fasta_filename = tempfile(fileext = ".fasta")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_topology_from_sequence_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, with
the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td></tr>
<tr><td><code id="predict_topology_from_sequence_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="predict_topology_from_sequence_+3A_temp_fasta_filename">temp_fasta_filename</code></td>
<td>
<p>temporary FASTA filename,
which will deleted after usage</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a topology as a string of zeroes and ones, where a one denotes
that the corresponding amino acid is located within the membrane.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  protein_sequence &lt;- paste0(
    "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
    "SLAIADMLLGFLVMPVSMLTILYGYRWP"
  )
  predict_topology_from_sequence(protein_sequence)
}
</code></pre>

<hr>
<h2 id='pureseqtmr'>pureseqtmr: estimate the topoplogy of membrane proteins</h2><span id='topic+pureseqtmr'></span>

<h3>Description</h3>

<p>Proteins reside in either the cell plasma of in the
cell membrane. A membrane protein goes through the
membrane at least once. There are multiple ways to span this
hydrophobic layer. One common structure is the transmembrane (alpha)
helix (TMH). Given the amino acid sequence of a membrane protein, this
package predicts which parts of the protein are TMHs
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  # Obtain an example filename
  fasta_filename &lt;- get_example_filename("1bhaA.fasta")

  # Get the topology as a tibble
  topology &lt;- predict_topology(fasta_filename)

  # show the topology
  plot_topology(topology)
}
</code></pre>

<hr>
<h2 id='pureseqtmr_report'>Create a <a href="#topic+pureseqtmr">pureseqtmr</a> report, to be used when reporting bugs</h2><span id='topic+pureseqtmr_report'></span>

<h3>Description</h3>

<p>Create a <a href="#topic+pureseqtmr">pureseqtmr</a> report, to be used when reporting bugs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pureseqtmr_report(folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pureseqtmr_report_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pureseqtmr_report()
</code></pre>

<hr>
<h2 id='run_pureseqtm_proteome'>Run PureseqTM on a proteome</h2><span id='topic+run_pureseqtm_proteome'></span>

<h3>Description</h3>

<p>Run PureseqTM on a proteome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_pureseqtm_proteome(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  topology_filename = tempfile(fileext = ".top")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_pureseqtm_proteome_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
<tr><td><code id="run_pureseqtm_proteome_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
<a href="#topic+get_default_pureseqtm_folder">get_default_pureseqtm_folder</a></p>
</td></tr>
<tr><td><code id="run_pureseqtm_proteome_+3A_topology_filename">topology_filename</code></td>
<td>
<p>name of the file to save a protein's
topology to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the topology of the proteome, using the same output as PureseqTM.
Use <a href="#topic+predict_topology">predict_topology</a> to get the topology as a <a href="tibble.html#topic+tibble">tibble</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>


<ul>
<li>
<p>Use <a href="#topic+predict_topology">predict_topology</a> to predict the topology
of a proteome

</p>
</li>
<li>
<p>Use <a href="#topic+create_pureseqtm_files">create_pureseqtm_files</a> to only create the PureseqTM
output files

</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  fasta_filename &lt;- get_example_filename("1bhaA.fasta")
  run_pureseqtm_proteome(fasta_filename)
}
</code></pre>

<hr>
<h2 id='save_tibble_as_fasta_file'>Save the first two columns of a tibble as a FASTA file</h2><span id='topic+save_tibble_as_fasta_file'></span>

<h3>Description</h3>

<p>Save the first two columns of a tibble as a FASTA file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_tibble_as_fasta_file(t, fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save_tibble_as_fasta_file_+3A_t">t</code></td>
<td>
<p>a <a href="tibble.html#topic+tibble">tibble</a></p>
</td></tr>
<tr><td><code id="save_tibble_as_fasta_file_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>path to a FASTA file</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tally_tmhs'>Count the number of transmembrane helices in a topology</h2><span id='topic+tally_tmhs'></span>

<h3>Description</h3>

<p>Count the number of transmembrane helices in a topology
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tally_tmhs(topology)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tally_tmhs_+3A_topology">topology</code></td>
<td>
<p>the topology as a <a href="tibble.html#topic+tibble">tibble</a>
with the columns 'name' and 'topology',
where the 'name' column hold all the proteins' names,
and 'topology' contains the respective topologies as strings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with the number of
TMHs per protein
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pureseqtm_installed()) {
  tally_tmhs(
    predict_topology(
      get_example_filename("1bhaA.fasta")
    )
  )
}
</code></pre>

<hr>
<h2 id='uninstall_pureseqtm'>Uninstall PureseqTM</h2><span id='topic+uninstall_pureseqtm'></span>

<h3>Description</h3>

<p>Uninstall PureseqTM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uninstall_pureseqtm(folder_name = get_default_pureseqtm_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="uninstall_pureseqtm_+3A_folder_name">folder_name</code></td>
<td>
<p>name of the folder
where the PureseqTM files are installed.
The name of the PureseqTM binary file will be at
<code>[folder_name]/PureseqTM_Package</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
