<!DOCTYPE html><html lang="en"><head><title>Help for package pvclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pvclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#lung'><p>DNA Microarray Data of Lung Tumors</p></a></li>
<li><a href='#msfit'><p>Curve Fitting for Multiscale Bootstrap Resampling</p></a></li>
<li><a href='#msplot'><p>Drawing the Results of Curve Fitting for Pvclust Object</p></a></li>
<li><a href='#plot.pvclust'><p>Draws Dendrogram with P-values for Pvclust Object</p></a></li>
<li><a href='#print.pvclust'><p>Print Function for Pvclust Object</p></a></li>
<li><a href='#pvclust'><p>Calculating P-values for Hierchical Clustering</p></a></li>
<li><a href='#pvpick'><p>Find Clusters with High/Low P-values</p></a></li>
<li><a href='#seplot'><p>Diagnostic Plot for Standard Error of p-value</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>2.2-0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-11-19</td>
</tr>
<tr>
<td>Title:</td>
<td>Hierarchical Clustering with P-Values via Multiscale Bootstrap
Resampling</td>
</tr>
<tr>
<td>Author:</td>
<td>Ryota Suzuki &lt;suzuki@ef-prime.com&gt;,
        Yoshikazu Terada &lt;terada@sigmath.es.osaka-u.ac.jp&gt;, 
        Hidetoshi Shimodaira &lt;shimo@i.kyoto-u.ac.jp&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ryota Suzuki &lt;suzuki@ef-prime.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>MASS, parallel</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of multiscale bootstrap resampling for
             assessing the uncertainty in hierarchical cluster analysis.
             It provides SI (selective inference) p-value, AU (approximately unbiased)
             p-value and BP (bootstrap probability) value for each cluster in a dendrogram.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://stat.sys.i.kyoto-u.ac.jp/prog/pvclust/">http://stat.sys.i.kyoto-u.ac.jp/prog/pvclust/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-19 10:31:59 UTC; suzuki</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-19 12:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='lung'>DNA Microarray Data of Lung Tumors</h2><span id='topic+lung'></span>

<h3>Description</h3>

<p>DNA Microarray data of 73 lung tissues including 67 lung tumors.
There are 916 observations of genes for each lung tissue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lung)</code></pre>


<h3>Format</h3>

<p>data frame of size <code class="reqn">916 \times 73</code>.</p>


<h3>Details</h3>

<p>This dataset has been modified from original data. Each one
observation of duplicate genes has been removed. See <code>source</code>
section in this help for original data source.
</p>


<h3>Source</h3>

<p><a href="http://genome-www.stanford.edu/lung_cancer/adeno/">http://genome-www.stanford.edu/lung_cancer/adeno/</a></p>


<h3>References</h3>

<p>Garber, M. E. et al. (2001)
&quot;Diversity of gene expression in adenocarcinoma of the lung&quot;,
<em>Proceedings of the National Academy of Sciences</em>,
98, 13784-13789.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Reading the data
data(lung)

## Multiscale Bootstrap Resampling
lung.pv &lt;- pvclust(lung, nboot=100)

## CAUTION: nboot=100 may be too small for actual use.
##          We suggest nboot=1000 or larger.
##          plot/print functions will be useful for diagnostics.

## Plot the result
plot(lung.pv, cex=0.8, cex.pv=0.7)

ask.bak &lt;- par()$ask
par(ask=TRUE)

pvrect(lung.pv, alpha=0.9)
msplot(lung.pv, edges=c(51,62,68,71))

par(ask=ask.bak)

## Print a cluster with high p-value
lung.pp &lt;- pvpick(lung.pv, alpha=0.9)
lung.pp$clusters[[2]]

## Print its edge number
lung.pp$edges[2]


## We recommend parallel computing for large dataset as this one
## Not run: 
library(snow)
cl &lt;- makeCluster(10, type="MPI")
lung.pv &lt;- parPvclust(cl, lung, nboot=1000)

## End(Not run)
</code></pre>

<hr>
<h2 id='msfit'>Curve Fitting for Multiscale Bootstrap Resampling</h2><span id='topic+msfit'></span><span id='topic+plot.msfit'></span><span id='topic+lines.msfit'></span><span id='topic+summary.msfit'></span>

<h3>Description</h3>

<p><code>msfit</code> performs curve fitting for multiscale
bootstrap resampling. It generates an object of class
<code>msfit</code>. Several generic methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msfit(bp, r, nboot)

## S3 method for class 'msfit'
plot(x, curve=TRUE, main=NULL, sub=NULL, xlab=NULL, ylab=NULL, ...)

## S3 method for class 'msfit'
lines(x, col=2, lty=1, ...)

## S3 method for class 'msfit'
summary(object, digits=3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msfit_+3A_bp">bp</code></td>
<td>
<p>numeric vector of bootstrap probability values.</p>
</td></tr>
<tr><td><code id="msfit_+3A_r">r</code></td>
<td>
<p>numeric vector of relative sample size of bootstrap samples
defined as <code class="reqn">r=n'/n</code> for original sample size <code class="reqn">n</code> and
bootstrap sample size <code class="reqn">n'</code>.</p>
</td></tr>
<tr><td><code id="msfit_+3A_nboot">nboot</code></td>
<td>
<p>numeric value (vector) of the number of bootstrap
replications.</p>
</td></tr>
<tr><td><code id="msfit_+3A_x">x</code></td>
<td>
<p>object of class <code>msfit</code>.</p>
</td></tr>
<tr><td><code id="msfit_+3A_curve">curve</code></td>
<td>
<p>logical. If <code>TRUE</code>, the fitted curve is drawn.</p>
</td></tr>
<tr><td><code id="msfit_+3A_main">main</code>, <code id="msfit_+3A_sub">sub</code>, <code id="msfit_+3A_xlab">xlab</code>, <code id="msfit_+3A_ylab">ylab</code>, <code id="msfit_+3A_col">col</code>, <code id="msfit_+3A_lty">lty</code></td>
<td>
<p>generic graphic parameters.</p>
</td></tr>
<tr><td><code id="msfit_+3A_object">object</code></td>
<td>
<p>object of class <code>msfit</code>.</p>
</td></tr>
<tr><td><code id="msfit_+3A_digits">digits</code></td>
<td>
<p>integer indicating the precision to be used in rounding.</p>
</td></tr>
<tr><td><code id="msfit_+3A_...">...</code></td>
<td>
<p>other parameters to be used in the functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>function <code>msfit</code> performs the curve fitting for multiscale
bootstrap resampling. In package <code>pvclust</code> this function is only
called from the function <code>pvclust</code> (or <code>parPvclust</code>), and
may never be called from users. However one can access a list of
<code>msfit</code> objects by <code>x$msfit</code>, where <code>x</code> is an object of
class <code>pvclust</code>.
</p>


<h3>Value</h3>

<p><code>msfit</code> returns an object of class <code>msfit</code>. It contains
the following objects:
</p>
<table role = "presentation">
<tr><td><code>p</code></td>
<td>
<p>numeric vector of <code class="reqn">p</code>-values. <code>au</code> is AU
(Approximately Unbiased) <code class="reqn">p</code>-value computed by multiscale
bootstrap resampling, which is more accurate than BP value
(explained below) as unbiased <code class="reqn">p</code>-value. <code>bp</code> is BP
(Bootstrap Probability) value, which is simple but tends to be
unbiased when the absolute value of <code>c</code> (a value in <code>coef</code>
vector, explained below) is large.</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>numeric vector of estimated standard errors of <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>numeric vector related to geometric aspects of
hypotheses. <code>v</code> is signed distance and <code>c</code> is curvature of
the boundary.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>numeric value of the degree of freedom in curve fitting.</p>
</td></tr>
<tr><td><code>rss</code></td>
<td>
<p>residual sum of squares.</p>
</td></tr>
<tr><td><code>pchi</code></td>
<td>
<p><code class="reqn">p</code>-value of chi-square test based on asymptotic theory.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>


<h3>References</h3>

<p>Shimodaira, H. (2004)
&quot;Approximately unbiased tests of regions using multistep-multiscale
bootstrap resampling&quot;,
<em>Annals of Statistics</em>, 32, 2616-2641.
</p>
<p>Shimodaira, H. (2002)
&quot;An approximately unbiased test of phylogenetic tree selection&quot;,
<em>Systematic Biology</em>, 51, 492-508.
</p>

<hr>
<h2 id='msplot'>Drawing the Results of Curve Fitting for Pvclust Object</h2><span id='topic+msplot'></span>

<h3>Description</h3>

<p>draws the results of curve fitting for pvclust object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msplot(x, edges=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msplot_+3A_x">x</code></td>
<td>
<p>object of class <code>pvclust</code>.</p>
</td></tr>
<tr><td><code id="msplot_+3A_edges">edges</code></td>
<td>
<p>numeric vector of edge numbers to be plotted.</p>
</td></tr>
<tr><td><code id="msplot_+3A_...">...</code></td>
<td>
<p>other parameters to be used in the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.msfit">plot.msfit</a></code></p>

<hr>
<h2 id='plot.pvclust'>Draws Dendrogram with P-values for Pvclust Object</h2><span id='topic+plot.pvclust'></span><span id='topic+text.pvclust'></span>

<h3>Description</h3>

<p>plot dendrogram for a <code>pvclust</code> object and add <code class="reqn">p</code>-values for
clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pvclust'
plot(x, print.pv=TRUE, print.num=TRUE, float=0.01,
  col.pv=c(si=4, au=2, bp=3, edge=8), cex.pv=0.8, font.pv=NULL,
  col=NULL, cex=NULL, font=NULL, lty=NULL, lwd=NULL, main=NULL,
  sub=NULL, xlab=NULL, ...)

## S3 method for class 'pvclust'
text(x, col=c(au=2, bp=3, edge=8), print.num=TRUE,
  float=0.01, cex=NULL, font=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.pvclust_+3A_x">x</code></td>
<td>
<p>object of class <code>pvclust</code>, which is generated by
function <code>pvclust</code>. See <code><a href="#topic+pvclust">pvclust</a></code> for details.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_print.pv">print.pv</code></td>
<td>
<p>logical flag to specify whether print <code class="reqn">p</code>-values
around the edges (clusters), or character vector of length 0 to 3
which specifies the names of <code class="reqn">p</code>-values to print
(<code>c("si", "au", "bp")</code> for example).</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_print.num">print.num</code></td>
<td>
<p>logical flag to specify whether print edge numbers
below clusters.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_float">float</code></td>
<td>
<p>numeric value to adjust the height of <code class="reqn">p</code>-values from
edges.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_col.pv">col.pv</code></td>
<td>
<p>named numeric vector to specify the colors for
<code class="reqn">p</code>-values and edge numbers. For back compatibility
it can also be unnamed numeric vector of length 3, which
corresponds to the color of AU, BP values and edge numbers.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_cex.pv">cex.pv</code></td>
<td>
<p>numeric value which specifies the size of characters for
<code class="reqn">p</code>-values and edge numbers. See <code>cex</code> argument for
<code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_font.pv">font.pv</code></td>
<td>
<p>numeric value which specifies the font of characters
for <code class="reqn">p</code>-values and edge numbers. See <code>font</code> argument for
<code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_col">col</code>, <code id="plot.pvclust_+3A_cex">cex</code>, <code id="plot.pvclust_+3A_font">font</code></td>
<td>
<p>in <code>text</code> function, they correspond to
<code>col.pv</code>, <code>cex.pv</code> and <code>font.pv</code> in <code>plot</code>
function, respectively. In <code>plot</code> function they are used as
generic graphic parameters.</p>
</td></tr>
<tr><td><code id="plot.pvclust_+3A_lty">lty</code>, <code id="plot.pvclust_+3A_lwd">lwd</code>, <code id="plot.pvclust_+3A_main">main</code>, <code id="plot.pvclust_+3A_sub">sub</code>, <code id="plot.pvclust_+3A_xlab">xlab</code>, <code id="plot.pvclust_+3A_...">...</code></td>
<td>

<p>generic graphic parameters. See <code><a href="graphics.html#topic+par">par</a></code> for
details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots a dendrogram with <code class="reqn">p</code>-values for given object
of class <code>pvclust</code>.
SI <code class="reqn">p</code>-value (printed in blue color in default) is the approximately unbiased
<code class="reqn">p</code>-value for selective inference, and
AU <code class="reqn">p</code>-value (printed in red color in default) is also the approximately unbiased
<code class="reqn">p</code>-value but for non-selective inference. They ared calculated by multiscale bootstrap
resampling. BP value (printed in green color in default) is &quot;bootstrap
probability&quot; value, which is less accurate than AU value as
<code class="reqn">p</code>-value. One can consider that clusters (edges) with high SI or AU
values (e.g. 95%) are strongly supported by data.
SI value is newly introduced in Terada and Shimodaira (2017) for selective inference,
which is more appropriate for testing clusters identified by looking at the tree.
AU value has been used since Shimodaira (2002), which is not designed for selective inference.
AU is valid when you know the clusters before looking at the data.
See also documatation (Multiscale Bootstrap using Scaleboot Package, verison 0.4-0 or higher) in <code>scaleboot</code> package.
</p>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>


<h3>References</h3>

<p>Terada, Y. and Shimodaira, H. (2007)
&quot;Selective inference for the problem of regions via multiscale bootstrap&quot;,
<em>arXiv:1711.00949</em>.
</p>
<p>Shimodaira, H. (2004)
&quot;Approximately unbiased tests of regions using multistep-multiscale
bootstrap resampling&quot;,
<em>Annals of Statistics</em>, 32, 2616-2641.
</p>
<p>Shimodaira, H. (2002)
&quot;An approximately unbiased test of phylogenetic tree selection&quot;,
<em>Systematic Biology</em>, 51, 492-508.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+text.pvclust">text.pvclust</a></code></p>

<hr>
<h2 id='print.pvclust'>Print Function for Pvclust Object</h2><span id='topic+print.pvclust'></span>

<h3>Description</h3>

<p>print clustering method and distance measure used in
hierarchical clustering, <code class="reqn">p</code>-values and related statistics for
a <code>pvclust</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pvclust'
print(x, which=NULL, digits=3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.pvclust_+3A_x">x</code></td>
<td>
<p>object of class <code>pvclust</code>.</p>
</td></tr>
<tr><td><code id="print.pvclust_+3A_which">which</code></td>
<td>
<p>numeric vector which specifies the numbers of edges
(clusters) of which the values are printed. If <code>NULL</code> is given,
it prints the values of all edges. The default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="print.pvclust_+3A_digits">digits</code></td>
<td>
<p>integer indicating the precision to be used in
rounding.</p>
</td></tr>
<tr><td><code id="print.pvclust_+3A_...">...</code></td>
<td>
<p>other parameters used in the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>this function prints <code class="reqn">p</code>-values and some related
statistics.
</p>
<table role = "presentation">
<tr><td><code>au</code></td>
<td>
<p>AU (Approximately Unbiased) <code class="reqn">p</code>-value, which is more
accurate than BP value as unbiased <code class="reqn">p</code>-value. It is computed by
multiscale bootstrap resampling.</p>
</td></tr>
<tr><td><code>bp</code></td>
<td>
<p>BP (Bootstrap Probability) value, which is a simple
statistic computed by bootstrap resampling. This value tends to be
biased as <code class="reqn">p</code>-value when the absolute value of <code>c</code>
(explained below) is large.</p>
</td></tr>
<tr><td><code>se.au</code>, <code>se.bp</code></td>
<td>
<p>estimated standard errors for <code>au</code> and
<code>bp</code>, respectively.</p>
</td></tr>
<tr><td><code>v</code>, <code>c</code></td>
<td>
<p>values related to geometric aspects of
hypotheses. <code>v</code> is signed distance and <code>c</code> is curvature of
the boundary.</p>
</td></tr>
<tr><td><code>pchi</code></td>
<td>
<p><code class="reqn">p</code>-values of chi-square test based on asymptotic theory.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>

<hr>
<h2 id='pvclust'>Calculating P-values for Hierchical Clustering</h2><span id='topic+pvclust'></span><span id='topic+parPvclust'></span>

<h3>Description</h3>

<p>calculates <code class="reqn">p</code>-values for hierarchical clustering via
multiscale bootstrap resampling. Hierarchical clustering is done for
given data and <code class="reqn">p</code>-values are computed for each of the clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvclust(data, method.hclust="average",
        method.dist="correlation", use.cor="pairwise.complete.obs",
        nboot=1000, parallel=FALSE, r=seq(.5,1.4,by=.1),
        store=FALSE, weight=FALSE, iseed=NULL, quiet=FALSE)

parPvclust(cl=NULL, data, method.hclust="average",
           method.dist="correlation", use.cor="pairwise.complete.obs",
           nboot=1000, r=seq(.5,1.4,by=.1), store=FALSE, weight=FALSE,
           init.rand=NULL, iseed=NULL, quiet=FALSE)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvclust_+3A_data">data</code></td>
<td>
<p>numeric data matrix or data frame.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_method.hclust">method.hclust</code></td>
<td>

<p>the agglomerative method used in hierarchical clustering. This
should be (an abbreviation of) one of <code>"average"</code>, <code>"ward.D"</code>,
<code>"ward.D2"</code>, <code>"single"</code>, <code>"complete"</code>, <code>"mcquitty"</code>,
<code>"median"</code> or <code>"centroid"</code>. The default is
<code>"average"</code>. See <code>method</code> argument in
<code><a href="stats.html#topic+hclust">hclust</a></code>.
</p>
</td></tr>
<tr><td><code id="pvclust_+3A_method.dist">method.dist</code></td>
<td>
<p>the distance measure to be used. This should be
a character string, or a function which returns a <code>dist</code> object.
A character string should be (an abbreviation of)
one of <code>"correlation"</code>, <code>"uncentered"</code>,
<code>"abscor"</code> or those which are allowed for <code>method</code>
argument in <code>dist</code> function. The default is
<code>"correlation"</code>. See <em>details</em> section in this help and
<code>method</code> argument in <code><a href="stats.html#topic+dist">dist</a></code>.
</p>
</td></tr>
<tr><td><code id="pvclust_+3A_use.cor">use.cor</code></td>
<td>
<p>character string which specifies the method for
computing correlation with data including missing values. This
should be (an abbreviation of) one of <code>"all.obs"</code>,
<code>"complete.obs"</code> or <code>"pairwise.complete.obs"</code>. See
the <code>use</code> argument in <code><a href="stats.html#topic+cor">cor</a></code> function.
</p>
</td></tr>
<tr><td><code id="pvclust_+3A_nboot">nboot</code></td>
<td>
<p>the number of bootstrap replications. The default is
<code>1000</code>.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_parallel">parallel</code></td>
<td>
<p>switch for parallel computation.
If <code>FALSE</code> the computation is done in non-parallel mode.
If <code>TRUE</code> or a positive integer is supplied,
parallel computation is done with automatically generated PSOCK cluster.
Use <code>TRUE</code> for default cluster size (<code>parallel::detectCores() - 1</code>),
or specify the size by an integer.
If a <code>cluster</code> object is supplied the cluster is used for parallel computation.
Note that <code>NULL</code> is currently not allowed for using the default cluster.
</p>
</td></tr>
<tr><td><code id="pvclust_+3A_r">r</code></td>
<td>
<p>numeric vector which specifies the relative sample sizes of
bootstrap replications. For original sample size <code class="reqn">n</code> and
bootstrap sample size <code class="reqn">n'</code>, this is defined as <code class="reqn">r=n'/n</code>.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_store">store</code></td>
<td>
<p>locical. If <code>store=TRUE</code>, all bootstrap replications
are stored in the output object. The default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_cl">cl</code></td>
<td>
<p>a cluster object created by package <span class="pkg">parallel</span> or <span class="pkg">snow</span>.
If NULL, use the registered default cluster.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_weight">weight</code></td>
<td>
<p>logical. If <code>weight=TRUE</code>, resampling is made by
weight vector instead of index vector. Useful for large <code>r</code>
value (<code>r&gt;10</code>).  Currently, available only for distance
<code>"correlation"</code> and <code>"abscor"</code>.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_init.rand">init.rand</code></td>
<td>
<p>logical. If <code>init.rand=TRUE</code>, random number generators are initialized.
Use <code>iseed</code> argument to achieve reproducible results. <strong>This argument is duplicated and will be unavailable in the future.</strong></p>
</td></tr>
<tr><td><code id="pvclust_+3A_iseed">iseed</code></td>
<td>
<p>An integer. If non-<code>NULL</code> value is supplied random number generators are initialized.
It is passed to <code>set.seed</code> or <code>clusterSetRNGStream</code>.</p>
</td></tr>
<tr><td><code id="pvclust_+3A_quiet">quiet</code></td>
<td>
<p>logical. If <code>TRUE</code> it does not report the progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>pvclust</code> conducts multiscale bootstrap resampling to calculate
<code class="reqn">p</code>-values for each cluster in the result of hierarchical
clustering. <code>parPvclust</code> is the parallel version of this
procedure which depends on package <span class="pkg">parallel</span> for parallel computation.
</p>
<p>For data expressed as <code class="reqn">(n \times p)</code> matrix or data frame, we
assume that the data is <code class="reqn">n</code> observations of <code class="reqn">p</code> objects, which
are to be clustered. The <code class="reqn">i</code>'th row vector corresponds to the
<code class="reqn">i</code>'th observation of these objects and the <code class="reqn">j</code>'th column
vector corresponds to a sample of <code class="reqn">j</code>'th object with size <code class="reqn">n</code>.
</p>
<p>There are several methods to measure the dissimilarities between
objects. For data matrix <code class="reqn">X=\{x_{ij}\}</code>,
<code>"correlation"</code>
method takes
</p>
<p style="text-align: center;"><code class="reqn">
    1 - \frac{
      \sum_{i=1}^n (x_{ij} - \bar{x}_j) (x_{ik} - \bar{x}_k)
    }
    {
      \sqrt{\sum_{i=1}^n (x_{ij} - \bar{x}_j)^2}
      \sqrt{\sum_{i=1}^n (x_{ik} - \bar{x}_k)^2}
    }
    </code>
</p>

<p>for dissimilarity between <code class="reqn">j</code>'th and <code class="reqn">k</code>'th object, where
<code class="reqn">\bar{x}_j = \frac{1}{n} \sum_{i=1}^n x_{ij} \mbox{and}
  \bar{x}_k = \frac{1}{n} \sum_{i=1}^n x_{ik}</code>.
</p>
<p><code>"uncentered"</code> takes uncentered sample correlation
</p>
<p style="text-align: center;"><code class="reqn">
    1 - \frac{
      \sum_{i=1}^n x_{ij} x_{ik}
    }
    {
      \sqrt{\sum_{i=1}^n x_{ij}^2}
      \sqrt{\sum_{i=1}^n x_{ik}^2}
    }
    </code>
</p>

<p>and <code>"abscor"</code> takes the absolute value of sample correlation
</p>
<p style="text-align: center;"><code class="reqn">
    1 - \ \Biggl| \frac{
      \sum_{i=1}^n (x_{ij} - \bar{x}_j) (x_{ik} - \bar{x}_k)
    }
    {
      \sqrt{\sum_{i=1}^n (x_{ij} - \bar{x}_j)^2}
      \sqrt{\sum_{i=1}^n (x_{ik} - \bar{x}_k)^2}
    } \Biggl|.
  </code>
</p>



<h3>Value</h3>

<table role = "presentation">
<tr><td><code>hclust</code></td>
<td>
<p>hierarchical clustering for original data generated by
function <code>hclust</code>. See <code><a href="stats.html#topic+hclust">hclust</a></code> for details.</p>
</td></tr>
<tr><td><code>edges</code></td>
<td>
<p>data frame object which contains <code class="reqn">p</code>-values and
supporting informations such as standard errors.</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>data frame object which contains primitive information
about the result of multiscale bootstrap resampling.</p>
</td></tr>
<tr><td><code>msfit</code></td>
<td>
<p>list whose elements are results of curve fitting for
multiscale bootstrap resampling, of class <code>msfit</code>. See
<code><a href="#topic+msfit">msfit</a></code> for details.</p>
</td></tr>
<tr><td><code>nboot</code></td>
<td>
<p>numeric vector of number of bootstrap replications.</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>numeric vector of the relative sample size for bootstrap
replications.</p>
</td></tr>
<tr><td><code>store</code></td>
<td>
<p>list contains bootstrap replications if <code>store=TRUE</code>
was given for function <code>pvclust</code> or <code>parPvclust</code>.</p>
</td></tr>
<tr><td><code>version</code></td>
<td>
<p><code><a href="base.html#topic+package_version">package_version</a></code> of pvclust used
to generate this object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>


<h3>References</h3>

<p>Suzuki, R. and Shimodaira, H. (2006)
&quot;Pvclust: an R package for assessing the uncertainty in hierarchical clustering&quot;,
<em>Bioinformatics</em>, 22 (12): 1540-1542.
</p>
<p>Shimodaira, H. (2004)
&quot;Approximately unbiased tests of regions using multistep-multiscale
bootstrap resampling&quot;,
<em>Annals of Statistics</em>, 32, 2616-2641.
</p>
<p>Shimodaira, H. (2002)
&quot;An approximately unbiased test of phylogenetic tree selection&quot;,
<em>Systematic Biology</em>, 51, 492-508.
</p>
<p>Suzuki, R. and Shimodaira, H. (2004)
&quot;An application of multiscale bootstrap resampling to hierarchical
clustering of microarray data: How accurate are these clusters?&quot;,
<em>The Fifteenth International Conference on Genome Informatics 2004</em>,
P034.
</p>
<p><a href="http://www.sigmath.es.osaka-u.ac.jp/shimo-lab/prog/pvclust/">http://www.sigmath.es.osaka-u.ac.jp/shimo-lab/prog/pvclust/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lines.pvclust">lines.pvclust</a></code>, <code><a href="#topic+print.pvclust">print.pvclust</a></code>,
<code><a href="#topic+msfit">msfit</a></code>, <code><a href="#topic+plot.pvclust">plot.pvclust</a></code>,
<code><a href="#topic+text.pvclust">text.pvclust</a></code>, <code><a href="#topic+pvrect">pvrect</a></code> and
<code><a href="#topic+pvpick">pvpick</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>### example using Boston data in package MASS
data(Boston, package = "MASS")

## multiscale bootstrap resampling (non-parallel)
boston.pv &lt;- pvclust(Boston, nboot=100, parallel=FALSE)

## CAUTION: nboot=100 may be too small for actual use.
##          We suggest nboot=1000 or larger.
##          plot/print functions will be useful for diagnostics.

## plot dendrogram with p-values
plot(boston.pv)

ask.bak &lt;- par()$ask
par(ask=TRUE)

## highlight clusters with high au p-values
pvrect(boston.pv)

## print the result of multiscale bootstrap resampling
print(boston.pv, digits=3)

## plot diagnostic for curve fitting
msplot(boston.pv, edges=c(2,4,6,7))

par(ask=ask.bak)

## print clusters with high p-values
boston.pp &lt;- pvpick(boston.pv)
boston.pp

### Using a custom distance measure

## Define a distance function which returns an object of class "dist".
## The function must have only one argument "x" (data matrix or data.frame).
cosine &lt;- function(x) {
    x &lt;- as.matrix(x)
    y &lt;- t(x) %*% x
    res &lt;- 1 - y / (sqrt(diag(y)) %*% t(sqrt(diag(y))))
    res &lt;- as.dist(res)
    attr(res, "method") &lt;- "cosine"
    return(res)
}

result &lt;- pvclust(Boston, method.dist=cosine, nboot=100)
plot(result)

## Not run: 
### parallel computation
result.par &lt;- pvclust(Boston, nboot=1000, parallel=TRUE)
plot(result.par)

## End(Not run)

</code></pre>

<hr>
<h2 id='pvpick'>Find Clusters with High/Low P-values</h2><span id='topic+pvpick'></span><span id='topic+pvrect'></span><span id='topic+lines.pvclust'></span>

<h3>Description</h3>

<p>find clusters with relatively high/low
<code class="reqn">p</code>-values. <code>pvrect</code> and <code>lines</code> (S3 method for class
<code>pvclust</code>) highlight such clusters in existing plot, and
<code>pvpick</code> returns a list of such clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvpick(x, alpha=0.95, pv="au", type="geq", max.only=TRUE)

pvrect(x, alpha=0.95, pv="au", type="geq", max.only=TRUE, border=NULL, ...)

## S3 method for class 'pvclust'
lines(x, alpha=0.95, pv="au", type="geq", col=2, lwd=2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvpick_+3A_x">x</code></td>
<td>
<p>object of class <code>pvclust</code>.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_alpha">alpha</code></td>
<td>
<p>threshold value for <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_pv">pv</code></td>
<td>
<p>character string which specifies the <code class="reqn">p</code>-value to be
used. It should be one of <code>"si"</code>, <code>"au"</code> and <code>"bp"</code>,
corresponding to SI, AU <code class="reqn">p</code>-value and BP value, respectively. See
<code>plot.pvclust</code> for details.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_type">type</code></td>
<td>
<p>one of <code>"geq"</code>, <code>"leq"</code>, <code>"gt"</code> or
<code>"lt"</code>. If <code>"geq"</code> is specified, clusters with
<code class="reqn">p</code>-value <em>greater than or equals</em> the threshold given by
<code>"alpha"</code> are returned or displayed. Likewise <code>"leq"</code>
stands for <em>lower than or equals</em>, <code>"gt"</code> for
<em>greater than</em> and <code>"lt"</code> for <em>lower than</em> the
threshold value. The default is <code>"geq"</code>.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_max.only">max.only</code></td>
<td>
<p>logical. If some of clusters with high/low
<code class="reqn">p</code>-values have inclusion relation, only the largest cluster is
returned (or displayed) when <code>max.only=TRUE</code>.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_border">border</code></td>
<td>
<p>numeric value which specifies the color of borders of
rectangles.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_col">col</code></td>
<td>
<p>numeric value which specifies the color of lines.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_lwd">lwd</code></td>
<td>
<p>numeric value which specifies the width of lines.</p>
</td></tr>
<tr><td><code id="pvpick_+3A_...">...</code></td>
<td>
<p>other graphic parameters to be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>pvpick</code> returns a list which contains the following values.
</p>
<table role = "presentation">
<tr><td><code>clusters</code></td>
<td>
<p>a list of character string vectors. Each vector
corresponds to the names of objects in each cluster.</p>
</td></tr>
<tr><td><code>edges</code></td>
<td>
<p>numeric vector of edge numbers. The <code class="reqn">i</code>'th element
(number) corresponds to the <code class="reqn">i</code>'th name vector in
<code>clusters</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>

<hr>
<h2 id='seplot'>Diagnostic Plot for Standard Error of p-value</h2><span id='topic+seplot'></span>

<h3>Description</h3>

<p>draws diagnostic plot for standard error of <code class="reqn">p</code>-value
for pvclust object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seplot(object, type=c("au", "si", "bp"), identify=FALSE,
  main=NULL, xlab=NULL, ylab=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seplot_+3A_object">object</code></td>
<td>
<p>object of class <code>pvclust</code>.</p>
</td></tr>
<tr><td><code id="seplot_+3A_type">type</code></td>
<td>
<p>the type of <code class="reqn">p</code>-value to be plotted, one of <code>"si"</code>, <code>"au"</code>
or <code>"bp"</code>.</p>
</td></tr>
<tr><td><code id="seplot_+3A_identify">identify</code></td>
<td>
<p>logical. If <code>TRUE</code>, edge numbers can be identified
interactively. See <code><a href="graphics.html#topic+identify">identify</a></code> for basic usage.</p>
</td></tr>
<tr><td><code id="seplot_+3A_main">main</code>, <code id="seplot_+3A_xlab">xlab</code>, <code id="seplot_+3A_ylab">ylab</code></td>
<td>
<p>generic graphic parameters. See
<code><a href="graphics.html#topic+par">par</a></code> for details.</p>
</td></tr>
<tr><td><code id="seplot_+3A_...">...</code></td>
<td>
<p>other graphical parameters to be passed to generic
<code>plot</code> or <code>identify</code> function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ryota Suzuki <a href="mailto:suzuki@ef-prime.com">suzuki@ef-prime.com</a></p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
