<!DOCTYPE html><html lang="en"><head><title>Help for package QBMS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QBMS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#brapi_get_call'><p>Internal Function for Core BrAPI GET Calls</p></a></li>
<li><a href='#brapi_headers'><p>Generate Common HTTP Headers for API Requests</p></a></li>
<li><a href='#brapi_map'><p>BrAPI Mapping Table</p></a></li>
<li><a href='#brapi_post_search_allelematrix'><p>Internal Function Used for Core BrAPI POST Calls (Allele Matrix Search)</p></a></li>
<li><a href='#brapi_post_search_call'><p>Internal Function Used for Core BrAPI POST Calls</p></a></li>
<li><a href='#build_pedigree_table'><p>Building Pedigree Table Recursively</p></a></li>
<li><a href='#calc_biovars'><p>Calculate the Bioclimatic Variables</p></a></li>
<li><a href='#debug_qbms'><p>Debug Internal QBMS Status Object</p></a></li>
<li><a href='#get_async_page'><p>Asynchronously Fetch a Single API Page</p></a></li>
<li><a href='#get_async_pages'><p>Asynchronously Fetch Multiple API Pages</p></a></li>
<li><a href='#get_brapi_url'><p>Get the BrAPI Endpoint URL for a given QBMS function</p></a></li>
<li><a href='#get_germplasm_attributes'><p>Retrieve Attributes for a Specified Germplasm</p></a></li>
<li><a href='#get_germplasm_data'><p>Retrieve Observations Data for a Specified Germplasm.</p></a></li>
<li><a href='#get_germplasm_id'><p>Get Germplasm ID for a Specified Germplasm Name</p></a></li>
<li><a href='#get_germplasm_list'><p>Get the Germplasm List of the Current Active Study</p></a></li>
<li><a href='#get_hwsd2'><p>Get HWSD v2 Soil Data for a Given Location(s)</p></a></li>
<li><a href='#get_login_details'><p>Login Pop-Up Window</p></a></li>
<li><a href='#get_parents'><p>Get Direct Parents</p></a></li>
<li><a href='#get_pedigree_table'><p>Get the Pedigree Table</p></a></li>
<li><a href='#get_program_studies'><p>Get the List of Trials, Studies, and Locations Information for the Current Selected Program</p></a></li>
<li><a href='#get_program_trials'><p>Retrieve the List of Trials for the Active Breeding Program</p></a></li>
<li><a href='#get_qbms_connection'><p>Get the QBMS Connection</p></a></li>
<li><a href='#get_study_data'><p>Get the Observations Data of the Current Active Study</p></a></li>
<li><a href='#get_study_info'><p>Get the Details/Metadata of the Current Active Study</p></a></li>
<li><a href='#get_terraclimate'><p>Get TerraClimate Data for a Given Coordinate(s)</p></a></li>
<li><a href='#get_trial_data'><p>Get the Observations Data of the Current Active Trial</p></a></li>
<li><a href='#get_trial_obs_ontology'><p>Get the Traits Ontology/Metadata of the Current Active Trial</p></a></li>
<li><a href='#gigwa_get_allelematrix'><p>Get Markers Matrix in the Selected GIGWA Run</p></a></li>
<li><a href='#gigwa_get_markers'><p>Get Markers Map in the Selected GIGWA Run</p></a></li>
<li><a href='#gigwa_get_metadata'><p>Get the Metadata of the Current Active GIGWA Run</p></a></li>
<li><a href='#gigwa_get_samples'><p>Get the Samples List of the Current Active GIGWA Project</p></a></li>
<li><a href='#gigwa_get_sequences'><p>Get the Sequences of the Current Active GIGWA Project</p></a></li>
<li><a href='#gigwa_get_variants'><p>Get Available Variants in the Selected GIGWA Run</p></a></li>
<li><a href='#gigwa_list_dbs'><p>List GIGWA Databases</p></a></li>
<li><a href='#gigwa_list_projects'><p>Get the List of All Projects in the Selected GIGWA Database</p></a></li>
<li><a href='#gigwa_list_runs'><p>Get the List of the Run Names Available in the Selected GIGWA Project</p></a></li>
<li><a href='#gigwa_set_db'><p>Set the Current Active GIGWA Database by Name</p></a></li>
<li><a href='#gigwa_set_project'><p>Set the Current Active GIGWA Project</p></a></li>
<li><a href='#gigwa_set_run'><p>Set the Current Active GIGWA Run</p></a></li>
<li><a href='#ini_hwsd2'><p>Download and Setup HWSD v2.0 Data Files to Extract their Data Offline</p></a></li>
<li><a href='#ini_terraclimate'><p>Download TerraClimate netCDF Data Files to Extract their Data Offline</p></a></li>
<li><a href='#list_crops'><p>Retrieve Supported Crops from the Server</p></a></li>
<li><a href='#list_locations'><p>Get the List of Locations Information of the Current Selected Crop</p></a></li>
<li><a href='#list_programs'><p>Retrieve Breeding Programs for the Active Crop</p></a></li>
<li><a href='#list_studies'><p>Get the List of Studies in the Current Active Trial</p></a></li>
<li><a href='#list_trials'><p>List Trials in the Current Active Breeding Program</p></a></li>
<li><a href='#login_bms'><p>Login to the Server</p></a></li>
<li><a href='#login_breedbase'><p>Login to the BreedBase Server</p></a></li>
<li><a href='#login_gigwa'><p>Login to the GIGWA Server</p></a></li>
<li><a href='#login_oauth2'><p>Login using OAuth 2.0 Authentication</p></a></li>
<li><a href='#rbindlistx'><p>Make One Data.Table from a List of Many</p></a></li>
<li><a href='#rbindx'><p>Combine Data Frames by Row, Filling in Missing Columns</p></a></li>
<li><a href='#scan_brapi_endpoints'><p>Scan BrAPI Endpoints</p></a></li>
<li><a href='#set_crop'><p>Set the Current Active Crop</p></a></li>
<li><a href='#set_program'><p>Set the Current Active Breeding Program</p></a></li>
<li><a href='#set_qbms_config'><p>Configure BMS Server Settings</p></a></li>
<li><a href='#set_qbms_connection'><p>Set the QBMS Connection</p></a></li>
<li><a href='#set_study'><p>Set the Current Active Study</p></a></li>
<li><a href='#set_token'><p>Set Access Token Response</p></a></li>
<li><a href='#set_trial'><p>Set the Current Active Trial</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Query the Breeding Management System(s)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Khaled Al-Shamaa [aut, cre],
  Mariano Omar Crimi [ctb],
  Zakaria Kehel [ctb],
  Johan Aparicio [ctb],
  ICARDA [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Khaled Al-Shamaa &lt;k.el-shamaa@cgiar.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This R package assists breeders in linking data systems with their analytic pipelines, 
    a crucial step in digitizing breeding processes. It supports querying and retrieving 
    phenotypic and genotypic data from systems like 'EBS' <a href="https://ebs.excellenceinbreeding.org/">https://ebs.excellenceinbreeding.org/</a>, 
    'BMS' <a href="https://bmspro.io">https://bmspro.io</a>, 'BreedBase' <a href="https://breedbase.org">https://breedbase.org</a>, and 
    'GIGWA' <a href="https://github.com/SouthGreenPlatform/Gigwa2">https://github.com/SouthGreenPlatform/Gigwa2</a> (using 'BrAPI' <a href="https://brapi.org">https://brapi.org</a> calls). 
    Extra helper functions support environmental data sources, including 
    'TerraClimate' <a href="https://www.climatologylab.org/terraclimate.html">https://www.climatologylab.org/terraclimate.html</a> and 'FAO' 
    'HWSDv2' <a href="https://gaez.fao.org/pages/hwsd">https://gaez.fao.org/pages/hwsd</a> soil database. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://icarda-git.github.io/QBMS/">https://icarda-git.github.io/QBMS/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/icarda-git/QBMS/issues">https://github.com/icarda-git/QBMS/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr2, jsonlite, tcltk, utils, RNetCDF, stats, terra, RSQLite,
DBI, future, future.apply</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, remotes</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-18 12:02:46 UTC; Kel-shamaa</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-18 12:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='brapi_get_call'>Internal Function for Core BrAPI GET Calls</h2><span id='topic+brapi_get_call'></span>

<h3>Description</h3>

<p>Fetches data from an API endpoint, handles pagination by retrieving all pages, and consolidates the results into a single data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_get_call(call_url, nested = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="brapi_get_call_+3A_call_url">call_url</code></td>
<td>
<p>Character string specifying the base URL of the API endpoint to request.</p>
</td></tr>
<tr><td><code id="brapi_get_call_+3A_nested">nested</code></td>
<td>
<p>Logical value indicating whether to flatten nested lists in the JSON responses. Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs the following steps:
</p>

<ol>
<li><p> Fetches the first page synchronously to determine the total number of pages.
</p>
</li>
<li><p> If multiple pages exist, it asynchronously fetches the remaining pages using <code>get_async_pages()</code>.
</p>
</li>
<li><p> Consolidates the data from all pages into a single data frame.
</p>
</li>
<li><p> Updates global state variables with pagination information.
</p>
</li></ol>

<p>It relies on global variables from <code>qbms_globals</code> to manage state and configuration.
</p>


<h3>Value</h3>

<p>A list containing the consolidated data and associated metadata from the API response.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='brapi_headers'>Generate Common HTTP Headers for API Requests</h2><span id='topic+brapi_headers'></span>

<h3>Description</h3>

<p>This function constructs a list of standard HTTP headers required for making API 
requests, ensuring proper authentication and encoding. These headers are typically 
included with each API call to provide necessary information such as authorization 
tokens and content acceptance types. The function is designed to work with 
authenticated APIs, including BrAPI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_headers()
</code></pre>


<h3>Value</h3>

<p>A named list of common HTTP headers, including the authorization token (Bearer), 
content encoding, and accepted content types.
</p>
<p>A named list containing key HTTP headers, including:
</p>

<ul>
<li> <p><strong>Authorization:</strong> Bearer token used for authenticated API access.
</p>
</li>
<li> <p><strong>Accept-Encoding:</strong> Specifies supported compression types such as gzip and deflate.
</p>
</li>
<li> <p><strong>Accept:</strong> Specifies that the client accepts responses in JSON format.
</p>
</li></ul>



<h3>Note</h3>

<p>Ensure that the global state contains a valid authorization token before making API 
requests. This function retrieves the token from 'qbms_globals$state$token', which 
should be set after a successful login or authentication process.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='brapi_map'>BrAPI Mapping Table</h2><span id='topic+brapi_map'></span>

<h3>Description</h3>

<p>This internal table maps QBMS function names to corresponding BrAPI calls for different versions 
of the BrAPI standard. It is used internally by the QBMS system to translate function calls into 
BrAPI-compliant API requests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_map
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 31 rows and 3 columns.
</p>


<h3>Details</h3>

<p>This table supports both BrAPI v1 and v2 endpoints. It is regularly updated to reflect changes 
in the BrAPI standard and the inclusion of new API calls.
</p>

<hr>
<h2 id='brapi_post_search_allelematrix'>Internal Function Used for Core BrAPI POST Calls (Allele Matrix Search)</h2><span id='topic+brapi_post_search_allelematrix'></span>

<h3>Description</h3>

<p>This function is used internally to execute POST calls for retrieving the allele matrix
via BrAPI. It handles the post request, waits for the results asynchronously if required,
and processes the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_post_search_allelematrix(call_url, call_body, nested = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="brapi_post_search_allelematrix_+3A_call_url">call_url</code></td>
<td>
<p>BrAPI URL for the POST request.</p>
</td></tr>
<tr><td><code id="brapi_post_search_allelematrix_+3A_call_body">call_body</code></td>
<td>
<p>The request body to send with the POST request.</p>
</td></tr>
<tr><td><code id="brapi_post_search_allelematrix_+3A_nested">nested</code></td>
<td>
<p>Logical indicating whether to flatten the nested structure. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results obtained from the BrAPI POST call.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='brapi_post_search_call'>Internal Function Used for Core BrAPI POST Calls</h2><span id='topic+brapi_post_search_call'></span>

<h3>Description</h3>

<p>This function is used internally to execute POST calls to BrAPI endpoints and
retrieve the results while handling pagination and long-running tasks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_post_search_call(call_url, call_body, nested = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="brapi_post_search_call_+3A_call_url">call_url</code></td>
<td>
<p>BrAPI URL for the POST request.</p>
</td></tr>
<tr><td><code id="brapi_post_search_call_+3A_call_body">call_body</code></td>
<td>
<p>The request body to send with the POST request.</p>
</td></tr>
<tr><td><code id="brapi_post_search_call_+3A_nested">nested</code></td>
<td>
<p>Logical indicating whether to flatten the nested structure. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results obtained from the BrAPI POST call.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='build_pedigree_table'>Building Pedigree Table Recursively</h2><span id='topic+build_pedigree_table'></span>

<h3>Description</h3>

<p>Recursively builds a pedigree table by extracting and tracking parents for each genotype/germplasm in the provided list. 
The function handles backcross cases and updates the pedigree data frame with parent information for multiple generations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_pedigree_table(
  geno_list = NULL,
  pedigree_list = NULL,
  pedigree_df = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="build_pedigree_table_+3A_geno_list">geno_list</code></td>
<td>
<p>A character vector of genotype/germplasm names.</p>
</td></tr>
<tr><td><code id="build_pedigree_table_+3A_pedigree_list">pedigree_list</code></td>
<td>
<p>A character vector of associated pedigree strings, corresponding to the genotypes in <code>geno_list</code>.</p>
</td></tr>
<tr><td><code id="build_pedigree_table_+3A_pedigree_df">pedigree_df</code></td>
<td>
<p>A data frame of pedigrees from a previous iteration, used to accumulate pedigree data. If NULL, a new data frame is created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with three columns: 
- 'Variety': The identifier for the individual genotype.
- 'Female': The identifier for the female parent.
- 'Male': The identifier for the male parent.
The pedigree is built recursively, with individuals listed before any appearance as a parent.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='calc_biovars'>Calculate the Bioclimatic Variables</h2><span id='topic+calc_biovars'></span>

<h3>Description</h3>

<p>This function calculates the 19 standard bioclimatic variables derived from 
monthly temperature and precipitation data. Bioclimatic variables are often 
used in ecological modeling and species distribution modeling to capture biologically 
meaningful patterns in climate data, including annual trends, seasonality, and 
extreme environmental factors. 
</p>
<p>The bioclimatic variables represent metrics such as the annual mean temperature, 
temperature seasonality, and precipitation patterns (e.g., wettest or driest quarter).
These metrics help to model species distributions and analyze ecological dynamics.
</p>
<p>The bioclimatic variables are coded as follows:
</p>

<ul>
<li> <p><strong>BIO1</strong> = Annual Mean Temperature
</p>
</li>
<li> <p><strong>BIO2</strong> = Mean Diurnal Range (Mean of monthly (max temp - min temp))
</p>
</li>
<li> <p><strong>BIO3</strong> = Isothermality (BIO2/BIO7) (* 100)
</p>
</li>
<li> <p><strong>BIO4</strong> = Temperature Seasonality (standard deviation * 100)
</p>
</li>
<li> <p><strong>BIO5</strong> = Max Temperature of Warmest Month
</p>
</li>
<li> <p><strong>BIO6</strong> = Min Temperature of Coldest Month
</p>
</li>
<li> <p><strong>BIO7</strong> = Temperature Annual Range (BIO5 - BIO6)
</p>
</li>
<li> <p><strong>BIO8</strong> = Mean Temperature of Wettest Quarter
</p>
</li>
<li> <p><strong>BIO9</strong> = Mean Temperature of Driest Quarter
</p>
</li>
<li> <p><strong>BIO10</strong> = Mean Temperature of Warmest Quarter
</p>
</li>
<li> <p><strong>BIO11</strong> = Mean Temperature of Coldest Quarter
</p>
</li>
<li> <p><strong>BIO12</strong> = Annual Precipitation
</p>
</li>
<li> <p><strong>BIO13</strong> = Precipitation of Wettest Month
</p>
</li>
<li> <p><strong>BIO14</strong> = Precipitation of Driest Month
</p>
</li>
<li> <p><strong>BIO15</strong> = Precipitation Seasonality (Coefficient of Variation)
</p>
</li>
<li> <p><strong>BIO16</strong> = Precipitation of Wettest Quarter
</p>
</li>
<li> <p><strong>BIO17</strong> = Precipitation of Driest Quarter
</p>
</li>
<li> <p><strong>BIO18</strong> = Precipitation of Warmest Quarter
</p>
</li>
<li> <p><strong>BIO19</strong> = Precipitation of Coldest Quarter
</p>
</li></ul>

<p>These variables are computed using temperature and precipitation data in a standard format, 
and are critical for understanding species habitats and the effects of climate on ecosystems.
</p>
<p>This work is derived from the <a href="https://github.com/rspatial/dismo/blob/master/R/biovars.R">dismo R package</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_biovars(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_biovars_+3A_data">data</code></td>
<td>
<p>A data frame containing monthly climate data. The data frame must include:
</p>

<ul>
<li> <p><strong>year:</strong> The year for each set of monthly data.
</p>
</li>
<li> <p><strong>ppt:</strong> Monthly precipitation values (in mm).
</p>
</li>
<li> <p><strong>tmin:</strong> Monthly minimum temperature values (in degrees Celsius).
</p>
</li>
<li> <p><strong>tmax:</strong> Monthly maximum temperature values (in degrees Celsius).
</p>
</li></ul>

<p>The data should contain 12 rows (one for each month from January to December) per year, 
with the columns sorted in the order of year, ppt, tmin, and tmax.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with 19 columns representing the bioclimatic variables (BIO1 to BIO19) and 
an additional column for the year. The output data frame provides one row per year, with 
each column corresponding to one of the bioclimatic variables described above.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>
<p>Robert Hijmans, Museum of Vertebrate Zoology, UC Berkeley
</p>


<h3>References</h3>

<p>Nix, 1986. A biogeographic analysis of Australian elapid snakes. In: R. Longmore (ed.). 
Atlas of elapid snakes of Australia. Australian Flora and Fauna Series 7. 
Australian Government Publishing Service, Canberra.
</p>

<hr>
<h2 id='debug_qbms'>Debug Internal QBMS Status Object</h2><span id='topic+debug_qbms'></span>

<h3>Description</h3>

<p>Retrieves the internal QBMS status object for debugging purposes. This object contains
the current configuration and state of the QBMS session, including connection settings
and active tokens.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>debug_qbms()
</code></pre>


<h3>Value</h3>

<p>An environment object that holds the current QBMS configuration and state.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  obj &lt;- debug_qbms()
  obj$config
  obj$state
}

</code></pre>

<hr>
<h2 id='get_async_page'>Asynchronously Fetch a Single API Page</h2><span id='topic+get_async_page'></span>

<h3>Description</h3>

<p>Sends an asynchronous HTTP GET request to fetch data from a single API page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_async_page(full_url, nested)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_async_page_+3A_full_url">full_url</code></td>
<td>
<p>Character string specifying the full URL of the API endpoint to request.</p>
</td></tr>
<tr><td><code id="get_async_page_+3A_nested">nested</code></td>
<td>
<p>Logical value indicating whether to flatten nested lists in the JSON response.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses <code>future::future()</code> to perform the HTTP GET request asynchronously.
It retrieves the content from the specified URL, checks for HTTP errors, and parses the JSON response.
</p>


<h3>Value</h3>

<p>A future representing the asynchronous operation, which will resolve to a list containing the parsed JSON response.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_async_pages'>Asynchronously Fetch Multiple API Pages</h2><span id='topic+get_async_pages'></span>

<h3>Description</h3>

<p>Sends asynchronous HTTP GET requests to fetch data from multiple API pages concurrently.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_async_pages(pages, nested)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_async_pages_+3A_pages">pages</code></td>
<td>
<p>Character vector of full URLs specifying the API endpoints to request.</p>
</td></tr>
<tr><td><code id="get_async_pages_+3A_nested">nested</code></td>
<td>
<p>Logical value indicating whether to flatten nested lists in the JSON responses.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses <code>future.apply::future_lapply()</code> to perform concurrent HTTP GET requests for multiple pages.
It retrieves and parses the JSON responses from each URL provided.
</p>


<h3>Value</h3>

<p>A list of parsed JSON responses from each page.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_brapi_url'>Get the BrAPI Endpoint URL for a given QBMS function</h2><span id='topic+get_brapi_url'></span>

<h3>Description</h3>

<p>Constructs the BrAPI endpoint URL for a given QBMS function based on the configured server,
crop, and BrAPI version. The function name is mapped to the corresponding BrAPI call using
internal mapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_brapi_url(func_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_brapi_url_+3A_func_name">func_name</code></td>
<td>
<p>(string) The name of the QBMS function for which the BrAPI endpoint URL is required.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string representing the BrAPI endpoint URL.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_germplasm_attributes'>Retrieve Attributes for a Specified Germplasm</h2><span id='topic+get_germplasm_attributes'></span>

<h3>Description</h3>

<p>Retrieves a detailed list of attributes for a given germplasm, such as its origin, donors, and taxonomic information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_attributes(germplasm_name = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_germplasm_attributes_+3A_germplasm_name">germplasm_name</code></td>
<td>
<p>The name of the germplasm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the attributes associated with the specified germplasm.
</p>


<h3>Author(s)</h3>

<p>Johan Steven Aparicio, <a href="mailto:j.aparicio@cgiar.org">j.aparicio@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+get_germplasm_data">get_germplasm_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  germplasm_attributes &lt;- get_germplasm_attributes("Jabal")
}

</code></pre>

<hr>
<h2 id='get_germplasm_data'>Retrieve Observations Data for a Specified Germplasm.</h2><span id='topic+get_germplasm_data'></span>

<h3>Description</h3>

<p>Retrieves all available observations data for the given germplasm in the current active crop.
This data is aggregated across all trials in the crop database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_data(germplasm_name = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_germplasm_data_+3A_germplasm_name">germplasm_name</code></td>
<td>
<p>The name of the germplasm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing all available observations data for the specified germplasm.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+get_germplasm_attributes">get_germplasm_attributes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  germplasm_observations &lt;- get_germplasm_data("Jabal")
  head(germplasm_observations)
}

</code></pre>

<hr>
<h2 id='get_germplasm_id'>Get Germplasm ID for a Specified Germplasm Name</h2><span id='topic+get_germplasm_id'></span>

<h3>Description</h3>

<p>Retrieves the unique germplasm ID associated with the specified germplasm name for the current active crop.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_id(germplasm_name = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_germplasm_id_+3A_germplasm_name">germplasm_name</code></td>
<td>
<p>The name of the germplasm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string representing the germplasm's unique ID (germplasmDbId).
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+get_germplasm_data">get_germplasm_data</a></code>, <code><a href="#topic+get_germplasm_attributes">get_germplasm_attributes</a></code>
</p>

<hr>
<h2 id='get_germplasm_list'>Get the Germplasm List of the Current Active Study</h2><span id='topic+get_germplasm_list'></span>

<h3>Description</h3>

<p>Retrieves the list of germplasm (genetic material) used in the currently active study, 
which must be set using the <code><a href="#topic+set_study">set_study</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_list()
</code></pre>


<h3>Value</h3>

<p>A data frame containing the germplasm list for the active study.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, 
<code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+set_study">set_study</a></code> for related operations on crops and studies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  set_trial("IDYT39")
  set_study("IDYT39 Environment Number 9")
  germplasm &lt;- get_germplasm_list()
  head(germplasm)
}

</code></pre>

<hr>
<h2 id='get_hwsd2'>Get HWSD v2 Soil Data for a Given Location(s)</h2><span id='topic+get_hwsd2'></span>

<h3>Description</h3>

<p>Queries the HWSD v2 database to retrieve soil information for specific locations based on their coordinates.
For each location, the function extracts the Soil Mapping Unit (SMU) code and retrieves soil attributes based 
on the specified sequence (soil dominance) and depth layer. The function returns the input data frame augmented 
with soil data from the HWSDv2 dataset.
</p>
<p>The HWSD2_SMU table contains general information for each of the soil units 
occurring in any given SMU code (dominant soil unit and up to 11 associated soils).
</p>
<p>The SEQUENCE column refers to the sequence in which soil units within the SMU 
are presented (in order of percentage share). The dominant soil has sequence 1. 
The sequence can range between 1 and 12.
</p>
<p>The SHARE column refers to the share of the soil unit within the mapping unit in 
percentage. Shares of soil units within a mapping unit always sum up to 100 percent.
</p>
<p>The HWSD2_LAYERS table provides soil attributes per depth layer for each of the 
seven depth layers (D1 to D7) separately (represented in the LAYER column in the 
HWSD2_LAYERS table). The depth of the top and bottom of each layer is defined in
the TOPDEP and BOTDEP columns, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hwsd2(df, con, x = "longitude", y = "latitude", sequence = 1, layer = "D1")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_hwsd2_+3A_df">df</code></td>
<td>
<p>A data frame containing location information, including longitude and latitude in decimal degrees.</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_con">con</code></td>
<td>
<p>The HWSDv2 object returned by the <code>ini_hwsd2()</code> function, containing the raster and SQLite connection.</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_x">x</code></td>
<td>
<p>The column name in the data frame representing longitude (default is 'Longitude').</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_y">y</code></td>
<td>
<p>The column name in the data frame representing latitude (default is 'Latitude').</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_sequence">sequence</code></td>
<td>
<p>Integer indicating the soil unit's dominance order within the SMU (default is 1 for the dominant soil). 
Valid values range from 1 to 12.</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_layer">layer</code></td>
<td>
<p>String indicating the depth layer for which soil attributes should be retrieved (default is 'D1', with layers ranging from 'D1' to 'D7').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the original location data augmented by soil attributes for the specified sequence and layer. 
The data frame includes additional columns such as 'smu_id', 'SEQUENCE', 'LAYER', and other soil attributes.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ini_hwsd2">ini_hwsd2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  Location  &lt;- c('Tel-Hadya', 'Terbol', 'Marchouch')
  Latitude  &lt;- c(36.016, 33.808, 33.616)
  Longitude &lt;- c(36.943, 35.991, -6.716)
  sites     &lt;- data.frame(Location, Latitude, Longitude)
  
  hwsd2 &lt;- ini_hwsd2(data_path = 'C:/Users/Kel-shamaa/Downloads/HWSD v2/')
  sites &lt;- get_hwsd2(df = sites, con = hwsd2, x = 'Longitude', y = 'Latitude', 
                     sequence = 1, layer = 'D1')
}

</code></pre>

<hr>
<h2 id='get_login_details'>Login Pop-Up Window</h2><span id='topic+get_login_details'></span>

<h3>Description</h3>

<p>Opens a GUI pop-up window using Tcl/Tk to prompt the user for their username and password.
The window title and prompt message adapt based on the type of server being used (e.g., BMS, GIGWA).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_login_details()
</code></pre>


<h3>Value</h3>

<p>A vector containing the inserted username and password, with names 'usr' and 'pwd' respectively.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_parents'>Get Direct Parents</h2><span id='topic+get_parents'></span>

<h3>Description</h3>

<p>Utility function to split a given pedigree string and retrieve the pedigrees of the direct parents (female and male).
The function handles different formats of cross representations, such as single slashes (/), double slashes (//), 
or numbered crosses (e.g., /3/). It extracts the highest cross order when available and returns the sub-pedigree 
for the immediate parents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_parents(pedigree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_parents_+3A_pedigree">pedigree</code></td>
<td>
<p>A string providing the parentage through which a cultivar was obtained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of two items representing the direct female and male parents. If parent information is unavailable or unknown, 
'NA' is returned for the respective parent.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_pedigree_table'>Get the Pedigree Table</h2><span id='topic+get_pedigree_table'></span>

<h3>Description</h3>

<p>Retrieves a comprehensive pedigree table for the given dataset, which contains genotype names and pedigree strings. 
The function recursively traces parentage across generations and builds a pedigree table where each row corresponds 
to an individual, with columns for the female and male parents. It also handles cases of similar genotype names 
by standardizing them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigree_table(
  data,
  geno_column = "germplasmName",
  pedigree_column = "pedigree"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pedigree_table_+3A_data">data</code></td>
<td>
<p>A data frame containing genotype/germplasm data, including names and pedigree strings.</p>
</td></tr>
<tr><td><code id="get_pedigree_table_+3A_geno_column">geno_column</code></td>
<td>
<p>The name of the column that identifies the genotype/germplasm names.</p>
</td></tr>
<tr><td><code id="get_pedigree_table_+3A_pedigree_column">pedigree_column</code></td>
<td>
<p>The name of the column that contains the pedigree strings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with three columns:
- 'Variety': The identifier for the individual genotype.
- 'Female': The identifier for the female parent.
- 'Male': The identifier for the male parent.
The pedigree table is sorted such that individuals appear before any row where they are listed as a parent.
For founders (i.e., individuals with no parent information), 'NA' is used for the parental columns.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  set_trial("IDYT39")
  set_study("IDYT39 Environment Number 9")
  germplasm &lt;- get_germplasm_list()
  pedigree_table &lt;- get_pedigree_table(germplasm, "germplasmName", "pedigree")

  #############################
  # nadiv package way
  # library(nadiv)

  # Get additive relationship matrix in sparse matrix format
  # A &lt;- nadiv::makeA(pedigree_table)

  # Get A inverse matrix using base R function
  # AINV &lt;- solve(as.matrix(A))

  #############################
  # ASReml-R package way
  # library(asreml)

  # Represent A inverse matrix in an efficient way using i, j index and Ainverse value
  # Actual genotype names of any given index are in the attr(ainv, "rowNames")
  # ainv &lt;- asreml::ainverse(pedigree_table)

  #############################
  # Dummy data set for testing
  test &lt;- data.frame(genotype = c("X", "Y"),
                     pedigree = c("A//B/D/2/C", "B/C/3/A//B/C/2/D"))

  pedigree_table &lt;- get_pedigree_table(test, "genotype", "pedigree")
}

</code></pre>

<hr>
<h2 id='get_program_studies'>Get the List of Trials, Studies, and Locations Information for the Current Selected Program</h2><span id='topic+get_program_studies'></span>

<h3>Description</h3>

<p>Retrieves comprehensive information about the trials, studies, and environments/locations
within the current active breeding program, as configured in the internal state object using 
the <code><a href="#topic+set_program">set_program</a></code> function. This includes test and check entry counts for each study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_program_studies()
</code></pre>


<h3>Value</h3>

<p>A data frame containing detailed information for each study within the program's trials, 
including trial names, study names, location information, and entry counts.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  program_studies &lt;- get_program_studies()
  head(program_studies)
}

</code></pre>

<hr>
<h2 id='get_program_trials'>Retrieve the List of Trials for the Active Breeding Program</h2><span id='topic+get_program_trials'></span>

<h3>Description</h3>

<p>This internal function retrieves the list of trials for the currently active breeding program 
and crop combination. The crop and program must be set using <code><a href="#topic+set_crop">set_crop</a></code> and 
<code><a href="#topic+set_program">set_program</a></code> prior to calling this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_program_trials()
</code></pre>


<h3>Value</h3>

<p>A data frame containing information on trials for the active breeding program.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+list_trials">list_trials</a></code>
</p>

<hr>
<h2 id='get_qbms_connection'>Get the QBMS Connection</h2><span id='topic+get_qbms_connection'></span>

<h3>Description</h3>

<p>Retrieves the current QBMS connection object, which contains the server's configuration 
and state, including any active sessions and tokens. This can be used to save and restore 
connections between sessions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_qbms_connection()
</code></pre>


<h3>Value</h3>

<p>A list containing the current QBMS configuration and state.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_connection">set_qbms_connection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  
  df1  &lt;- get_germplasm_data("Jabal")
  con1 &lt;- get_qbms_connection()
  
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", engine = "gigwa", no_auth = TRUE)
  
  gigwa_set_db("DIVRICE_NB")
  gigwa_set_project("refNB")
  gigwa_set_run("03052022")
  
  df2  &lt;- gigwa_get_metadata()
  con2 &lt;- get_qbms_connection()
  
  set_qbms_connection(con1)
  df3 &lt;- get_germplasm_attributes("Jabal")
}

</code></pre>

<hr>
<h2 id='get_study_data'>Get the Observations Data of the Current Active Study</h2><span id='topic+get_study_data'></span>

<h3>Description</h3>

<p>Retrieves the observations data (e.g., measurements, variables) for the active study, 
which must be set using the <code><a href="#topic+set_study">set_study</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_data()
</code></pre>


<h3>Value</h3>

<p>A data frame containing the observation data for the active study, or <code>NULL</code> 
if no data is available.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, 
<code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+set_study">set_study</a></code> for related study operations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  set_trial("IDYT39")
  set_study("IDYT39 Environment Number 9")
  data &lt;- get_study_data()
  head(data)
}

</code></pre>

<hr>
<h2 id='get_study_info'>Get the Details/Metadata of the Current Active Study</h2><span id='topic+get_study_info'></span>

<h3>Description</h3>

<p>Retrieves detailed metadata for the currently active study. The study must be 
set using the <code><a href="#topic+set_study">set_study</a></code> function, and its details will be fetched 
from the BrAPI server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_info()
</code></pre>


<h3>Value</h3>

<p>A data frame containing the metadata of the active study. Returns <code>NULL</code> 
if no study metadata is available.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, 
<code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+set_study">set_study</a></code> for related crop and study management.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  set_trial("IDYT39")
  set_study("IDYT39 Environment Number 9")
  info &lt;- get_study_info()
}

</code></pre>

<hr>
<h2 id='get_terraclimate'>Get TerraClimate Data for a Given Coordinate(s)</h2><span id='topic+get_terraclimate'></span>

<h3>Description</h3>

<p>This function allows you to extract climate variables from the 
<a href="https://www.climatologylab.org/terraclimate.html">TerraClimate</a> dataset for specific geographic coordinates. 
TerraClimate is a global dataset of monthly climate data covering the years 1958-present. The function retrieves 
<a href="https://www.climatologylab.org/terraclimate-variables.html">climate variables</a> directly from the hosting server provided by the 
<a href="https://hpc.uidaho.edu/">University of Idaho</a>, avoiding the need to download large raster files in netCDF format.
Additionally, the function calculates <a href="https://www.worldclim.org/data/bioclim.html">bioclimatic variables</a> 
using the <code><a href="#topic+calc_biovars">calc_biovars</a></code> function, derived from the <a href="https://github.com/rspatial/dismo/blob/master/R/biovars.R">dismo R package</a>.
</p>
<p>The TerraClimate dataset is compared with <a href="https://www.worldclim.org/data/worldclim21.html">WorldClim</a> in several aspects:
</p>

<ul>
<li><p> TerraClimate: 1958-present vs. WorldClim: 1970-2000
</p>
</li>
<li><p> 14 climate variables vs. 7 climate variables in WorldClim
</p>
</li>
<li><p> Spatial resolution: ~4 km (TerraClimate) vs. ~1 km (WorldClim)
</p>
</li>
<li><p> Need to calculate bioclimatic variables (TerraClimate) vs. pre-calculated (WorldClim)
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>get_terraclimate(
  lat,
  lon,
  from = "1958-01-01",
  to = "2022-12-31",
  clim_vars = NULL,
  month_mask = NULL,
  offline = FALSE,
  data_path = "./data/"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_terraclimate_+3A_lat">lat</code></td>
<td>
<p>Vector of Latitude(s) in decimal degree format. Each entry corresponds to a location of interest.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_lon">lon</code></td>
<td>
<p>Vector of Longitude(s) in decimal degree format. Each entry corresponds to a location of interest.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_from">from</code></td>
<td>
<p>Start date as a string in 'YYYY-MM-DD' format. Defines the beginning of the time range for data extraction.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_to">to</code></td>
<td>
<p>End date as a string in 'YYYY-MM-DD' format. Defines the end of the time range for data extraction.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_clim_vars">clim_vars</code></td>
<td>
<p>List of climate variables to extract. Valid options include: <em>aet, def, pet, ppt, q, soil, 
srad, swe, tmax, tmin, vap, ws, vpd, and PDSI</em>. Default is <code>NULL</code>, which retrieves all variables.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_month_mask">month_mask</code></td>
<td>
<p>A vector specifying the months of interest, e.g., for specific seasons (e.g., planting season: 
<code>c(10:12, 1:5)</code>). Default is <code>NULL</code>, which retrieves data for all months.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_offline">offline</code></td>
<td>
<p>Logical value indicating whether to extract TerraClimate data from pre-downloaded netCDF files. 
Default is <code>FALSE</code>, meaning data is fetched from the remote server.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_data_path">data_path</code></td>
<td>
<p>String specifying the directory path where downloaded netCDF files are stored when working offline. 
Default is './data/'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two data frames for each coordinate pair (latitude and longitude):
</p>

<ul>
<li> <p><strong>climate:</strong> A data frame containing the requested climate variables for each month and location.
</p>
</li>
<li> <p><strong>biovars:</strong> A data frame with calculated bioclimatic variables, based on the extracted climate data.
</p>
</li></ul>

<p>Each data frame is in a format ready for further analysis in R.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>References</h3>

<p>Abatzoglou, J., Dobrowski, S., Parks, S. <em>et al.</em> TerraClimate, a high-resolution 
global dataset of monthly climate and climatic water balance from 1958-2015. 
<em>Sci Data</em> <strong>5</strong>, 170191 (2018). <a href="https://doi.org/10.1038/sdata.2017.191">doi:10.1038/sdata.2017.191</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ini_terraclimate">ini_terraclimate</a></code>, <code><a href="#topic+calc_biovars">calc_biovars</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # data &lt;- get_terraclimate(36.016, 36.943, '1979-09-01', '2012-06-30', 
  #                          c('ppt', 'tmin', 'tmax'), c(10:12,1:5))
  data &lt;- get_terraclimate(36.016, 36.943, '1979-09-01', '2012-06-30')

  View(data$climate[[1]])  # View the climate data
  View(data$biovars[[1]])  # View the bioclimatic variables
}

</code></pre>

<hr>
<h2 id='get_trial_data'>Get the Observations Data of the Current Active Trial</h2><span id='topic+get_trial_data'></span>

<h3>Description</h3>

<p>Retrieves the combined observations data (including all studies) for the current 
active trial, as configured in the internal state object using the <code><a href="#topic+set_trial">set_trial</a></code> function. 
This function iterates over all studies within the active trial and aggregates 
their observation data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trial_data()
</code></pre>


<h3>Value</h3>

<p>A data frame containing the combined observations data from all studies in the active trial.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  set_trial("IDYT39")
  MET &lt;- get_trial_data()
  head(MET)
}

</code></pre>

<hr>
<h2 id='get_trial_obs_ontology'>Get the Traits Ontology/Metadata of the Current Active Trial</h2><span id='topic+get_trial_obs_ontology'></span>

<h3>Description</h3>

<p>Retrieves the traits ontology or metadata for the current active trial, which 
includes detailed information about the observation variables used in the trial.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trial_obs_ontology()
</code></pre>


<h3>Value</h3>

<p>A data frame containing the traits ontology or metadata, filtered by the observation 
variables used in the current trial.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>, 
<code><a href="#topic+get_study_data">get_study_data</a></code> for retrieving study observations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  set_trial("IDYT39")
  ontology &lt;- get_trial_obs_ontology()
}

</code></pre>

<hr>
<h2 id='gigwa_get_allelematrix'>Get Markers Matrix in the Selected GIGWA Run</h2><span id='topic+gigwa_get_allelematrix'></span>

<h3>Description</h3>

<p>Retrieve a two-dimensional matrix of genotype data from the selected GIGWA run. This matrix is returned
based on filters for regions, samples, or variants. The data can be simplified to use numeric coding for
genotypes, or returned in its raw VCF-like format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_allelematrix(
  samples = NULL,
  start = 0,
  end = "",
  chrom = NULL,
  snps = NULL,
  snps_pageSize = 10000,
  samples_pageSize = 100,
  simplify = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gigwa_get_allelematrix_+3A_samples">samples</code></td>
<td>
<p>A list of sample names to include (optional). If NULL, all samples will be included.</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_start">start</code></td>
<td>
<p>Start position of the query region (zero-based, inclusive).</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_end">end</code></td>
<td>
<p>End position of the query region (zero-based, exclusive).</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_chrom">chrom</code></td>
<td>
<p>Reference sequence name (e.g., chromosome or contig).</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_snps">snps</code></td>
<td>
<p>A list of variant names to filter (optional).</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_snps_pagesize">snps_pageSize</code></td>
<td>
<p>Number of variants to fetch per page (default is 10,000).</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_samples_pagesize">samples_pageSize</code></td>
<td>
<p>Number of samples to fetch per page (default is 100).</p>
</td></tr>
<tr><td><code id="gigwa_get_allelematrix_+3A_simplify">simplify</code></td>
<td>
<p>Whether to simplify the returned data using numeric coding (default is TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with rows representing SNP markers and columns representing samples. 
Values are numeric codings (0: reference allele, 1: heterozygous, 2: alternative allele).
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  gigwa_set_run("run1")
  samples &lt;- gigwa_get_samples()
  chroms &lt;- gigwa_get_sequences()
  geno_data &lt;- gigwa_get_allelematrix(samples = samples[1:5],
                                      start   = 0,
                                      end     = 1234567,
                                      chrom   = chroms[1:3])
}
</code></pre>

<hr>
<h2 id='gigwa_get_markers'>Get Markers Map in the Selected GIGWA Run</h2><span id='topic+gigwa_get_markers'></span>

<h3>Description</h3>

<p>Retrieve a filtered list of SNP variants from the selected run. This function allows users to query variants
based on chromosomal regions and return results in simplified format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_markers(start = NULL, end = NULL, chrom = NULL, simplify = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gigwa_get_markers_+3A_start">start</code></td>
<td>
<p>Start position of the query region (zero-based, inclusive).</p>
</td></tr>
<tr><td><code id="gigwa_get_markers_+3A_end">end</code></td>
<td>
<p>End position of the query region (zero-based, exclusive).</p>
</td></tr>
<tr><td><code id="gigwa_get_markers_+3A_chrom">chrom</code></td>
<td>
<p>Reference sequence name (e.g., chromosome).</p>
</td></tr>
<tr><td><code id="gigwa_get_markers_+3A_simplify">simplify</code></td>
<td>
<p>Logical, if TRUE (default) returns data in a simplified HapMap-like format with columns for rs#, alleles, chromosome, and position.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of SNP markers, optionally simplified to include rs#, alleles, chromosome, and position.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  gigwa_set_run("run1")
  chroms &lt;- gigwa_get_sequences()
  geno_map &lt;- gigwa_get_markers(start = 0, end = 12345678, chrom = chroms[7])
}
</code></pre>

<hr>
<h2 id='gigwa_get_metadata'>Get the Metadata of the Current Active GIGWA Run</h2><span id='topic+gigwa_get_metadata'></span>

<h3>Description</h3>

<p>Retrieve metadata associated with the samples in the current active run, set using the 'gigwa_set_run()' function.
The metadata provides additional information about the samples in the selected run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_metadata()
</code></pre>


<h3>Value</h3>

<p>A data frame containing metadata attributes for each sample in the active run.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_run">gigwa_set_run</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("3kG_10M")
  gigwa_set_project("3003_ind")
  gigwa_set_run("1")
  metadata &lt;- gigwa_get_metadata()
}
</code></pre>

<hr>
<h2 id='gigwa_get_samples'>Get the Samples List of the Current Active GIGWA Project</h2><span id='topic+gigwa_get_samples'></span>

<h3>Description</h3>

<p>Retrieve the list of samples associated with the currently active GIGWA project,
set using 'gigwa_set_project()'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_samples()
</code></pre>


<h3>Value</h3>

<p>A vector of sample names in the selected project.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_project">gigwa_set_project</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  samples &lt;- gigwa_get_samples()
}
</code></pre>

<hr>
<h2 id='gigwa_get_sequences'>Get the Sequences of the Current Active GIGWA Project</h2><span id='topic+gigwa_get_sequences'></span>

<h3>Description</h3>

<p>Retrieve the list of sequences (e.g., chromosomes) associated with the currently
active project in GIGWA, which has been set using the 'gigwa_set_project()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_sequences()
</code></pre>


<h3>Value</h3>

<p>A vector of sequence names (e.g., chromosome names) for the selected project.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_project">gigwa_set_project</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  chroms &lt;- gigwa_get_sequences()
}
</code></pre>

<hr>
<h2 id='gigwa_get_variants'>Get Available Variants in the Selected GIGWA Run</h2><span id='topic+gigwa_get_variants'></span>

<h3>Description</h3>

<p>Retrieve variant data (e.g., SNP markers) for the selected GIGWA run based on filtering criteria, 
including minor allele frequency, missing data threshold, and sample subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_variants(
  max_missing = 1,
  min_maf = 0.5,
  samples = NULL,
  start = NULL,
  end = NULL,
  referenceName = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gigwa_get_variants_+3A_max_missing">max_missing</code></td>
<td>
<p>The maximum allowable missing data ratio, between 0 and 1 (default is 1, meaning up to 100% missing data).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_min_maf">min_maf</code></td>
<td>
<p>Minimum Minor Allele Frequency (MAF) between 0 and 0.5 (default is 0).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_samples">samples</code></td>
<td>
<p>A list of sample names to include in the query (optional). If NULL, all samples will be included.</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_start">start</code></td>
<td>
<p>Start position of the query region (zero-based, inclusive).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_end">end</code></td>
<td>
<p>End position of the query region (zero-based, exclusive).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_referencename">referenceName</code></td>
<td>
<p>The reference sequence name (e.g., chromosome).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame where the first 4 columns describe the SNP (rs# variant name, alleles, chrom, pos), 
and subsequent columns contain numerical genotyping information (0 for reference allele, 1 for heterozygous, and 2 for minor allele).
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  gigwa_set_run("run1")
  marker_matrix &lt;- gigwa_get_variants(max_missing = 0.2, 
                                      min_maf = 0.35, 
                                      samples = c("ind1", "ind3", "ind7"))
}

</code></pre>

<hr>
<h2 id='gigwa_list_dbs'>List GIGWA Databases</h2><span id='topic+gigwa_list_dbs'></span>

<h3>Description</h3>

<p>Retrieve the list of available databases from the connected GIGWA server.
An active connection is required. If not connected, the function will throw an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_list_dbs()
</code></pre>


<h3>Value</h3>

<p>A list of databases available on the connected GIGWA server.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_list_dbs()
}
</code></pre>

<hr>
<h2 id='gigwa_list_projects'>Get the List of All Projects in the Selected GIGWA Database</h2><span id='topic+gigwa_list_projects'></span>

<h3>Description</h3>

<p>Retrieve the list of projects available in the currently active GIGWA database, set
using 'gigwa_set_db()'. If no database is selected, the function will throw an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_list_projects()
</code></pre>


<h3>Value</h3>

<p>A list of project names in the selected database.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_db">gigwa_set_db</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_list_projects()
}
</code></pre>

<hr>
<h2 id='gigwa_list_runs'>Get the List of the Run Names Available in the Selected GIGWA Project</h2><span id='topic+gigwa_list_runs'></span>

<h3>Description</h3>

<p>Retrieve the list of available runs in the currently active GIGWA project, set using
'gigwa_set_project()'. If no project is selected, an error will be raised.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_list_runs()
</code></pre>


<h3>Value</h3>

<p>A list of run names associated with the selected project.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_project">gigwa_set_project</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  gigwa_list_runs()
}
</code></pre>

<hr>
<h2 id='gigwa_set_db'>Set the Current Active GIGWA Database by Name</h2><span id='topic+gigwa_set_db'></span>

<h3>Description</h3>

<p>Select a GIGWA database as the active database for subsequent operations. This updates
the internal configuration object and resets any previously selected projects or runs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_set_db(db_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gigwa_set_db_+3A_db_name">db_name</code></td>
<td>
<p>The name of the database to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Updates the internal configuration with the selected database.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_list_dbs">gigwa_list_dbs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
}
</code></pre>

<hr>
<h2 id='gigwa_set_project'>Set the Current Active GIGWA Project</h2><span id='topic+gigwa_set_project'></span>

<h3>Description</h3>

<p>Select a project from the active GIGWA database and set it as the current active project
in the internal state. This selection is used for retrieving related data, such as runs or samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_set_project(project_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gigwa_set_project_+3A_project_name">project_name</code></td>
<td>
<p>The name of the project to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Updates the internal state with the selected project.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_db">gigwa_set_db</a></code>, <code><a href="#topic+gigwa_list_projects">gigwa_list_projects</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
}
</code></pre>

<hr>
<h2 id='gigwa_set_run'>Set the Current Active GIGWA Run</h2><span id='topic+gigwa_set_run'></span>

<h3>Description</h3>

<p>Select a run from the active GIGWA project and set it as the current active run in the
internal state, enabling further data retrieval operations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_set_run(run_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gigwa_set_run_+3A_run_name">run_name</code></td>
<td>
<p>The name of the run to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Updates the internal state with the selected run.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_project">gigwa_set_project</a></code>, <code><a href="#topic+gigwa_list_runs">gigwa_list_runs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)
  gigwa_set_db("Sorghum-JGI_v1")
  gigwa_set_project("Nelson_et_al_2011")
  gigwa_set_run("run1")
}
</code></pre>

<hr>
<h2 id='ini_hwsd2'>Download and Setup HWSD v2.0 Data Files to Extract their Data Offline</h2><span id='topic+ini_hwsd2'></span>

<h3>Description</h3>

<p>Downloads and sets up the HWSD v2.0 data files required to extract soil data offline. The function
retrieves the HWSD raster soil unit map and the SQLite database containing soil attributes. If the files 
already exist in the specified directory, they are used directly. The function returns an object with 
the raster and SQLite connection for further queries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ini_hwsd2(data_path = "./data/", timeout = 300)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ini_hwsd2_+3A_data_path">data_path</code></td>
<td>
<p>String specifying the directory path where HWSD v2.0 data files are stored or should be downloaded (default is './data/').</p>
</td></tr>
<tr><td><code id="ini_hwsd2_+3A_timeout">timeout</code></td>
<td>
<p>Timeout in seconds for downloading each HWSD v2.0 data file (default is 300).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object ('con') containing two items: 
- 'raster': HWSDv2 raster object for spatial queries.
- 'sqlite': Connection to the HWSDv2 SQLite database.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_hwsd2">get_hwsd2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  hwsd2 &lt;- ini_hwsd2(data_path = 'C:/Users/Kel-shamaa/Downloads/HWSD v2/')
}
 
</code></pre>

<hr>
<h2 id='ini_terraclimate'>Download TerraClimate netCDF Data Files to Extract their Data Offline</h2><span id='topic+ini_terraclimate'></span>

<h3>Description</h3>

<p>This function facilitates the download of TerraClimate netCDF files for a specified 
time period and climate variables. TerraClimate data provides monthly climate data 
for global terrestrial surfaces, and this function allows you to store the data locally 
for offline extraction and analysis without the need to download the entire dataset.
</p>
<p>Users can specify a range of climate variables such as precipitation, temperature, 
evapotranspiration, soil moisture, and more. The downloaded files are saved in 
netCDF format, making them accessible for subsequent offline analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ini_terraclimate(
  from = "2019-09-01",
  to = "2022-06-30",
  clim_vars = c("ppt", "tmin", "tmax"),
  data_path = "./data/",
  timeout = 300
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ini_terraclimate_+3A_from">from</code></td>
<td>
<p>Start date as a string in the 'YYYY-MM-DD' format. This defines the beginning 
of the time period for which you want to download the climate data.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_to">to</code></td>
<td>
<p>End date as a string in the 'YYYY-MM-DD' format. This defines the end of the time 
period for which you want to download the climate data.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_clim_vars">clim_vars</code></td>
<td>
<p>A list of climate variables to download. Valid options include:
<em>aet</em> (Actual Evapotranspiration), <em>def</em> (Climate Water Deficit), <em>pet</em> (Potential 
Evapotranspiration), <em>ppt</em> (Precipitation), <em>q</em> (Runoff), <em>soil</em> (Soil Moisture), 
<em>srad</em> (Solar Radiation), <em>swe</em> (Snow Water Equivalent), <em>tmax</em> (Maximum Temperature), 
<em>tmin</em> (Minimum Temperature), <em>vap</em> (Vapor Pressure), <em>ws</em> (Wind Speed), 
<em>vpd</em> (Vapor Pressure Deficit), and <em>PDSI</em> (Palmer Drought Severity Index).
If NULL (default), all available variables will be downloaded.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_data_path">data_path</code></td>
<td>
<p>A string containing the directory path where the downloaded netCDF files 
will be stored. The default path is './data/'.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_timeout">timeout</code></td>
<td>
<p>Timeout in seconds for downloading each netCDF raster file. The default value is 300 seconds.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No explicit return value. The downloaded netCDF files are saved to the specified directory for 
offline use with the <code><a href="#topic+get_terraclimate">get_terraclimate</a></code> function to extract data for a given 
coordinate or set of coordinates.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_terraclimate">get_terraclimate</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Initialize TerraClimate data download for specific climate variables between two dates
  ini_terraclimate('2018-09-01', '2019-06-30', c('ppt', 'tmin', 'tmax'))
  
  # Coordinates for the location(s) of interest
  x &lt;- c(-6.716, 35.917, 76.884)
  y &lt;- c(33.616, 33.833, 23.111)
  
  # Extract TerraClimate data for the specified coordinates (online mode)
  a &lt;- get_terraclimate(y, x, '2018-09-01', '2019-06-30', c('ppt', 'tmin', 'tmax'))
  
  # View the extracted climate data and bioclimatic variables
  View(a$climate[[1]])
  View(a$biovars[[1]])
  
  # Extract TerraClimate data for the specified coordinates (offline mode)
  b &lt;- get_terraclimate(y, x, '2018-09-01', '2019-06-30', c('ppt', 'tmin', 'tmax'), offline = TRUE)
  
  # View the offline-extracted data
  View(b$climate[[1]])
  View(b$biovars[[1]])
}

</code></pre>

<hr>
<h2 id='list_crops'>Retrieve Supported Crops from the Server</h2><span id='topic+list_crops'></span>

<h3>Description</h3>

<p>Retrieves the list of crops supported by the connected server. If the crop 
list is cached in the internal state, it returns the cached data; otherwise, it 
sends a BrAPI GET request to fetch the crop list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_crops()
</code></pre>


<h3>Value</h3>

<p>A character vector containing the names of supported crops.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code> to configure and set the current active crop.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  list_crops()  # Retrieve list of supported crops
}

</code></pre>

<hr>
<h2 id='list_locations'>Get the List of Locations Information of the Current Selected Crop</h2><span id='topic+list_locations'></span>

<h3>Description</h3>

<p>Retrieves a list of locations associated with the current active crop, as 
configured in the internal state object using the <code><a href="#topic+set_crop">set_crop</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_locations()
</code></pre>


<h3>Value</h3>

<p>A data frame containing information about locations relevant to the current crop.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code> for related crop operations.
</p>

<hr>
<h2 id='list_programs'>Retrieve Breeding Programs for the Active Crop</h2><span id='topic+list_programs'></span>

<h3>Description</h3>

<p>Retrieves the list of breeding programs available for the currently selected crop. 
The crop must be set using the <code><a href="#topic+set_crop">set_crop</a></code> function prior to calling this.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_programs()
</code></pre>


<h3>Value</h3>

<p>A data frame containing the names of breeding programs available for the active crop.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+list_crops">list_crops</a></code> for managing server connection and crop selection.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set "wheat" as the active crop
  list_programs()  # Retrieve breeding programs for the active crop
}

</code></pre>

<hr>
<h2 id='list_studies'>Get the List of Studies in the Current Active Trial</h2><span id='topic+list_studies'></span>

<h3>Description</h3>

<p>Retrieves a list of studies (and associated locations) for the currently active trial, as configured 
using the <code><a href="#topic+set_trial">set_trial</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_studies()
</code></pre>


<h3>Value</h3>

<p>A data frame containing study names and associated location names. If no studies are available, an error is thrown.
</p>


<h3>Note</h3>

<p>This function must be called after a trial has been set using <code><a href="#topic+set_trial">set_trial</a></code>.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code> for related operations on crops, programs, and trials.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set crop
  set_program("Wheat International Nurseries")  # Set breeding program
  set_trial("IDYT39")  # Set trial
  list_studies()  # List studies
}

</code></pre>

<hr>
<h2 id='list_trials'>List Trials in the Current Active Breeding Program</h2><span id='topic+list_trials'></span>

<h3>Description</h3>

<p>Retrieves the list of trials for the current active breeding program. Optionally, 
filters trials by their starting year if specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_trials(year = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="list_trials_+3A_year">year</code></td>
<td>
<p>Numeric. An optional parameter to filter trials by their starting year. 
If not provided, all trials for the active program are returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the names of trials for the active breeding program. If no 
trials match the query, a warning is issued, and NA is returned.
</p>


<h3>Note</h3>

<p>The year filter is only supported for BMS databases.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code> for related operations involving crop and program selection.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set crop
  set_program("Wheat International Nurseries")  # Set breeding program
  list_trials()  # List trials
  list_trials(2022)  # List trials starting in 2022
}

</code></pre>

<hr>
<h2 id='login_bms'>Login to the Server</h2><span id='topic+login_bms'></span>

<h3>Description</h3>

<p>Connects to the BMS or related server using a username and password. If these are not provided,
a pop-up window will prompt the user to enter their credentials. The function handles authentication 
and stores the resulting access token internally for subsequent requests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_bms(username = NULL, password = NULL, encoding = "json")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="login_bms_+3A_username">username</code></td>
<td>
<p>The username (optional, default is NULL). If not provided, the pop-up window is triggered.</p>
</td></tr>
<tr><td><code id="login_bms_+3A_password">password</code></td>
<td>
<p>The password (optional, default is NULL). If not provided, the pop-up window is triggered.</p>
</td></tr>
<tr><td><code id="login_bms_+3A_encoding">encoding</code></td>
<td>
<p>Specifies how the request body should be encoded: 'form' (application/x-www-form-urlencoded), 
'multipart' (multipart/form-data), or 'json' (application/json). Default is &quot;json&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The access token is stored internally for future use.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()
}

</code></pre>

<hr>
<h2 id='login_breedbase'>Login to the BreedBase Server</h2><span id='topic+login_breedbase'></span>

<h3>Description</h3>

<p>Logs in to the BreedBase server using a username and password. If credentials are not provided,
a pop-up window will prompt the user. The function is a wrapper around the 'login_bms()' function,
with encoding set to 'form'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_breedbase(username = NULL, password = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="login_breedbase_+3A_username">username</code></td>
<td>
<p>The username (optional, default is NULL).</p>
</td></tr>
<tr><td><code id="login_breedbase_+3A_password">password</code></td>
<td>
<p>The password (optional, default is NULL).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The access token is stored internally for future use.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='login_gigwa'>Login to the GIGWA Server</h2><span id='topic+login_gigwa'></span>

<h3>Description</h3>

<p>Connect to the GIGWA server. If the 'username' or 'password' parameters are missing,
a login window will be triggered to capture these details.
</p>
<p>All connection settings (server URL, port, API path, and protocol) are read from the
'qbms_config' list. The function will request an authentication token from the server
and update the 'qbms_state' list with the token.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_gigwa(username = NULL, password = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="login_gigwa_+3A_username">username</code></td>
<td>
<p>The GIGWA username (optional, default is NULL).</p>
</td></tr>
<tr><td><code id="login_gigwa_+3A_password">password</code></td>
<td>
<p>The GIGWA password (optional, default is NULL).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The authentication token will be stored internally.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("http://localhost:59395/gigwa/index.jsp", time_out = 300, engine = "gigwa")

  # Login using your GIGWA account (interactive mode)
  login_gigwa()
  
  # You can pass GIGWA username and password as parameters (batch mode)
  # login_gigwa("gigwadmin", "nimda")
}
</code></pre>

<hr>
<h2 id='login_oauth2'>Login using OAuth 2.0 Authentication</h2><span id='topic+login_oauth2'></span>

<h3>Description</h3>

<p>Performs OAuth 2.0 authentication by sending the user to the authorization URL and exchanging the 
authorization code for an access token. This function supports caching of tokens for subsequent requests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_oauth2(
  authorize_url,
  access_url,
  client_id,
  client_secret = NULL,
  redirect_uri = "http://localhost:1410"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="login_oauth2_+3A_authorize_url">authorize_url</code></td>
<td>
<p>The URL where the client is redirected for user authorization.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_access_url">access_url</code></td>
<td>
<p>The URL used to exchange an authorization code for an access token.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_client_id">client_id</code></td>
<td>
<p>The client ID (consumer key) provided by the authorization server.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_client_secret">client_secret</code></td>
<td>
<p>The client secret provided by the authorization server (optional).</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_redirect_uri">redirect_uri</code></td>
<td>
<p>The URL where the user will be redirected after authorization (default is http://localhost:1410).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Updates the internal state with the access token and additional details.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='rbindlistx'>Make One Data.Table from a List of Many</h2><span id='topic+rbindlistx'></span>

<h3>Description</h3>

<p>Performs the equivalent of do.call(&quot;rbind&quot;, x) on data.frames, but much faster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbindlistx(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rbindlistx_+3A_x">x</code></td>
<td>
<p>A list containing data.table, data.frame, or list objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An unkeyed data.table containing a concatenation of all the items passed in.
</p>

<hr>
<h2 id='rbindx'>Combine Data Frames by Row, Filling in Missing Columns</h2><span id='topic+rbindx'></span>

<h3>Description</h3>

<p>Combines a list of data frames by row, filling in missing columns with NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbindx(..., dfs = list(...))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rbindx_+3A_...">...</code></td>
<td>
<p>The first argument data frame.</p>
</td></tr>
<tr><td><code id="rbindx_+3A_dfs">dfs</code></td>
<td>
<p>Input data frames to row bind together.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single data frame.
</p>

<hr>
<h2 id='scan_brapi_endpoints'>Scan BrAPI Endpoints</h2><span id='topic+scan_brapi_endpoints'></span>

<h3>Description</h3>

<p>Scans the available BrAPI endpoints on the configured source server and checks their accessibility. 
This function allows users to verify which BrAPI endpoints are available based on the provided IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scan_brapi_endpoints(programDbId = 0, trialDbId = 0, studyDbId = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scan_brapi_endpoints_+3A_programdbid">programDbId</code></td>
<td>
<p>(numeric) The programDbId to scan specific program-related endpoints (default is 0).</p>
</td></tr>
<tr><td><code id="scan_brapi_endpoints_+3A_trialdbid">trialDbId</code></td>
<td>
<p>(numeric) The trialDbId to scan specific trial-related endpoints (default is 0).</p>
</td></tr>
<tr><td><code id="scan_brapi_endpoints_+3A_studydbid">studyDbId</code></td>
<td>
<p>(numeric) The studyDbId to scan specific study-related endpoints (default is 0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame listing the QBMS function, BrAPI endpoint URL, and availability status for each endpoint.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='set_crop'>Set the Current Active Crop</h2><span id='topic+set_crop'></span>

<h3>Description</h3>

<p>Updates the internal configuration to set the selected crop as the active one. This 
must be called before performing crop-specific operations such as retrieving breeding programs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_crop(crop_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_crop_+3A_crop_name">crop_name</code></td>
<td>
<p>A string specifying the name of the crop to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The function updates the global state with the selected crop.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+list_crops">list_crops</a></code> to validate and retrieve the list of supported crops.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set "wheat" as the active crop
}

</code></pre>

<hr>
<h2 id='set_program'>Set the Current Active Breeding Program</h2><span id='topic+set_program'></span>

<h3>Description</h3>

<p>Updates the internal state to set the selected breeding program as active using the 
associated programDbId. This allows subsequent operations to be carried out 
within the context of this program.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_program(program_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_program_+3A_program_name">program_name</code></td>
<td>
<p>A string specifying the name of the breeding program to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The internal state is updated with the selected program.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+list_programs">list_programs</a></code> for related operations in the crop and program selection process.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set crop
  set_program("Wheat International Nurseries")  # Set breeding program
}

</code></pre>

<hr>
<h2 id='set_qbms_config'>Configure BMS Server Settings</h2><span id='topic+set_qbms_config'></span>

<h3>Description</h3>

<p>Configures the BMS server connection settings, including URL, API path, page size, and timeout.
This function allows you to set up the connection for different server backends like BMS, Gigwa, 
EBS, and Breedbase, and choose the appropriate BrAPI version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_qbms_config(
  url = "http://localhost",
  path = NULL,
  page_size = 1000,
  time_out = 120,
  no_auth = FALSE,
  engine = "bms",
  brapi_ver = "v1",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_qbms_config_+3A_url">url</code></td>
<td>
<p>The URL of the BMS login page or API base (default is &quot;http://localhost&quot;).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_path">path</code></td>
<td>
<p>The API path to use (default is NULL, which sets a path based on the engine).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_page_size">page_size</code></td>
<td>
<p>The number of records per page when making API calls (default is 1000).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_time_out">time_out</code></td>
<td>
<p>The maximum number of seconds to wait for a response (default is 120).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_no_auth">no_auth</code></td>
<td>
<p>Logical, whether the server requires authentication (default is FALSE).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_engine">engine</code></td>
<td>
<p>The backend system (default is &quot;bms&quot;). Options include &quot;bms&quot;, &quot;gigwa&quot;, &quot;breedbase&quot;, &quot;ebs&quot;.</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_brapi_ver">brapi_ver</code></td>
<td>
<p>The version of BrAPI to use, either &quot;v1&quot; or &quot;v2&quot; (default is &quot;v1&quot;).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_verbose">verbose</code></td>
<td>
<p>Logical, indicating whether to display progress information when making API calls (default is TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set_qbms_config("https://bms.icarda.org/ibpworkbench")

</code></pre>

<hr>
<h2 id='set_qbms_connection'>Set the QBMS Connection</h2><span id='topic+set_qbms_connection'></span>

<h3>Description</h3>

<p>Sets the QBMS connection object in the current environment, allowing users to restore
a saved connection, including configuration settings and session tokens.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_qbms_connection(env)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_qbms_connection_+3A_env">env</code></td>
<td>
<p>A list containing the saved connection configuration and state.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_qbms_connection">get_qbms_connection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  
  login_bms()
  set_crop("wheat")
  set_program("Wheat International Nurseries")
  
  df1  &lt;- get_germplasm_data("Jabal")
  con1 &lt;- get_qbms_connection()
  
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", engine = "gigwa", no_auth = TRUE)
  
  gigwa_set_db("DIVRICE_NB")
  gigwa_set_project("refNB")
  gigwa_set_run("03052022")
  
  df2  &lt;- gigwa_get_metadata()
  con2 &lt;- get_qbms_connection()
  
  set_qbms_connection(con1)
  df3 &lt;- get_germplasm_attributes("Jabal")
}

</code></pre>

<hr>
<h2 id='set_study'>Set the Current Active Study</h2><span id='topic+set_study'></span>

<h3>Description</h3>

<p>Updates the internal state to set the selected study as the current active study using the 
associated studyDbId. This allows operations to be performed within the context of the selected study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_study(study_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_study_+3A_study_name">study_name</code></td>
<td>
<p>A string specifying the name of the study to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The internal state is updated with the selected study.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+list_studies">list_studies</a></code> for related operations on crops, programs, trials, and studies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set crop
  set_program("Wheat International Nurseries")  # Set breeding program
  set_trial("IDYT39")  # Set trial
  set_study("IDYT39 Environment Number 9")  # Set study
}
 
</code></pre>

<hr>
<h2 id='set_token'>Set Access Token Response</h2><span id='topic+set_token'></span>

<h3>Description</h3>

<p>Stores the access token and associated details (such as username and expiration time) in the internal state.
The token is typically retrieved from the server during login and used for subsequent API requests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_token(token, user = "", expires_in = 3600)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_token_+3A_token">token</code></td>
<td>
<p>The access token string issued by the authorization server.</p>
</td></tr>
<tr><td><code id="set_token_+3A_user">user</code></td>
<td>
<p>The username associated with the token (optional).</p>
</td></tr>
<tr><td><code id="set_token_+3A_expires_in">expires_in</code></td>
<td>
<p>The lifetime of the access token in seconds (optional, default is 3600 seconds).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Updates the internal state with the token info.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='set_trial'>Set the Current Active Trial</h2><span id='topic+set_trial'></span>

<h3>Description</h3>

<p>Updates the internal state to set the selected trial as the current active trial using the 
associated trialDbId. This enables operations to be carried out within the context of the selected trial.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_trial(trial_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_trial_+3A_trial_name">trial_name</code></td>
<td>
<p>A string specifying the name of the trial to set as active.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. The internal state is updated with the selected trial.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+list_trials">list_trials</a></code> for operations involving crops, programs, and trials.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  login_bms()  # Log in to the server
  set_crop("wheat")  # Set crop
  set_program("Wheat International Nurseries")  # Set breeding program
  set_trial("IDYT39")  # Set trial
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
