<!DOCTYPE html><html><head><title>Help for package QBMS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QBMS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#brapi_get_call'><p>Internal Function Used for Core BrAPI GET Calls</p></a></li>
<li><a href='#brapi_headers'><p>Common HTTP Headers to Send</p></a></li>
<li><a href='#build_pedigree_table'><p>Building Pedigree Table Recursively</p></a></li>
<li><a href='#calc_biovars'><p>Calculate the Bioclimatic Variables</p></a></li>
<li><a href='#debug_qbms'><p>Debug Internal QBMS Status Object</p></a></li>
<li><a href='#get_async_page'><p>Async Version of HTTP GET Request</p></a></li>
<li><a href='#get_async_pages'><p>Run All Supplied Pages</p></a></li>
<li><a href='#get_germplasm_attributes'><p>Retrieve Attributes for a Specified Germplasm</p></a></li>
<li><a href='#get_germplasm_data'><p>Retrieve Observations Data for a Specified Germplasm.</p></a></li>
<li><a href='#get_germplasm_id'><p>Get Germplasm ID</p></a></li>
<li><a href='#get_germplasm_list'><p>Get the Germplasm List of the Current Active Study</p></a></li>
<li><a href='#get_hwsd2'><p>Get HWSD v2 Soil Data for a Given Location(s)</p></a></li>
<li><a href='#get_login_details'><p>Login Pop-Up Window</p></a></li>
<li><a href='#get_parents'><p>Get Direct Parents</p></a></li>
<li><a href='#get_pedigree_table'><p>Get the Pedigree Table</p></a></li>
<li><a href='#get_program_studies'><p>Get the List of Trials/Studies/Locations Information of the Current Selected Program</p></a></li>
<li><a href='#get_program_trials'><p>Internal Function Used to Retrieve the Rough List of Trials</p></a></li>
<li><a href='#get_qbms_connection'><p>Get the QBMS Connection</p></a></li>
<li><a href='#get_study_data'><p>Get the Observations Data of the Current Active Study</p></a></li>
<li><a href='#get_study_info'><p>Get the Details/Metadata of the Current Active Study</p></a></li>
<li><a href='#get_terraclimate'><p>Get TerraClimate Data for a Given Coordinate(s)</p></a></li>
<li><a href='#get_trial_data'><p>Get the Observations Data of the Current Active Trial</p></a></li>
<li><a href='#get_trial_obs_ontology'><p>Get the Traits Ontology/Metadata of the Current Active Trial</p></a></li>
<li><a href='#gigwa_get_metadata'><p>Get the Metadata of the Current Active GIGWA Run</p></a></li>
<li><a href='#gigwa_get_samples'><p>Get the Samples List of the Current Active GIGWA Run</p></a></li>
<li><a href='#gigwa_get_variants'><p>Get Available Variants in the Selected GIGWA Run</p></a></li>
<li><a href='#gigwa_list_dbs'><p>List GIGWA Databases</p></a></li>
<li><a href='#gigwa_list_projects'><p>Get the List of All Projects in the Selected GIGWA Database</p></a></li>
<li><a href='#gigwa_list_runs'><p>Get the List of the Run Names Available in the Selected GIGWA Project</p></a></li>
<li><a href='#gigwa_set_db'><p>Set the Current Active GIGWA Database by Name</p></a></li>
<li><a href='#gigwa_set_project'><p>Set the Current Active GIGWA Project</p></a></li>
<li><a href='#gigwa_set_run'><p>Set the Current Active GIGWA Run</p></a></li>
<li><a href='#ini_hwsd2'><p>Download and Setup HWSD v2.0 Data Files to Extract their Data Offline</p></a></li>
<li><a href='#ini_terraclimate'><p>Download TerraClimate netCDF Data Files to Extract their Data Offline</p></a></li>
<li><a href='#list_crops'><p>Get the List of Supported Crops</p></a></li>
<li><a href='#list_locations'><p>Get the List of Locations Information of the Current Selected Crop</p></a></li>
<li><a href='#list_programs'><p>Get the List of Breeding Programs Names</p></a></li>
<li><a href='#list_studies'><p>Get the List of Studies in the Current Active Trial</p></a></li>
<li><a href='#list_trials'><p>Get the List of Trials in the Current Active Breeding Program</p></a></li>
<li><a href='#login_bms'><p>Login to the Server</p></a></li>
<li><a href='#login_breedbase'><p>Login to the BreedBase Server</p></a></li>
<li><a href='#login_gigwa'><p>Login to the GIGWA Server</p></a></li>
<li><a href='#login_oauth2'><p>Login Using OAuth 2.0 Authentication</p></a></li>
<li><a href='#rbindlistx'><p>Make One Data.Table from a List of Many</p></a></li>
<li><a href='#rbindx'><p>Combine Data Frames by Row, Filling in Missing Columns</p></a></li>
<li><a href='#scan_brapi_endpoints'><p>Scan BrAPI Endpoints</p></a></li>
<li><a href='#set_crop'><p>Set the Current Active Crop</p></a></li>
<li><a href='#set_program'><p>Set the Current Active Breeding Program</p></a></li>
<li><a href='#set_qbms_config'><p>Configure BMS Server Settings</p></a></li>
<li><a href='#set_qbms_connection'><p>Set the QBMS Connection</p></a></li>
<li><a href='#set_study'><p>Set the Current Active Study</p></a></li>
<li><a href='#set_token'><p>Set Access Token Response</p></a></li>
<li><a href='#set_trial'><p>Set the Current Active Trial</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Query the Breeding Management System(s)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Khaled Al-Shamaa [aut, cre],
  Mariano Omar Crimi [ctb],
  Zakaria Kehel [ctb],
  Johan Aparicio [ctb],
  ICARDA [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Khaled Al-Shamaa &lt;k.el-shamaa@cgiar.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This R package assists breeders in linking data systems with their analytic pipelines, 
    a crucial step in digitizing breeding processes. It supports querying and retrieving 
    phenotypic and genotypic data from systems like 'EBS' <a href="https://ebs.excellenceinbreeding.org/">https://ebs.excellenceinbreeding.org/</a>, 
    'BMS' <a href="https://bmspro.io">https://bmspro.io</a>, 'BreedBase' <a href="https://breedbase.org">https://breedbase.org</a>, and 
    'GIGWA' <a href="https://github.com/SouthGreenPlatform/Gigwa2">https://github.com/SouthGreenPlatform/Gigwa2</a> (using 'BrAPI' <a href="https://brapi.org">https://brapi.org</a> calls). 
    Extra helper functions support environmental data sources, including 
    'TerraClimate' <a href="https://www.climatologylab.org/terraclimate.html">https://www.climatologylab.org/terraclimate.html</a> and 'FAO' 
    'HWSDv2' <a href="https://gaez.fao.org/pages/hwsd">https://gaez.fao.org/pages/hwsd</a> soil database. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://icarda-git.github.io/QBMS/">https://icarda-git.github.io/QBMS/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/icarda-git/QBMS/issues">https://github.com/icarda-git/QBMS/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr, jsonlite, tcltk, utils, RNetCDF, stats, terra, RSQLite,
DBI</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, remotes, async</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-07 08:56:21 UTC; Kel-shamaa</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-07 09:40:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='brapi_get_call'>Internal Function Used for Core BrAPI GET Calls</h2><span id='topic+brapi_get_call'></span>

<h3>Description</h3>

<p>This function is created for *internal use only* to call BrAPI in the GET method and
retrieve the raw response data and send back the results. This function takes
care of pagination, authentication, encoding, compression, decoding JSON response, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_get_call(call_url, nested = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brapi_get_call_+3A_call_url">call_url</code></td>
<td>
<p>BrAPI URL to call in GET method.</p>
</td></tr>
<tr><td><code id="brapi_get_call_+3A_nested">nested</code></td>
<td>
<p>Logical indicating whether retrieved JSON data will be flattened. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Result object returned by the JSON API response.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='brapi_headers'>Common HTTP Headers to Send</h2><span id='topic+brapi_headers'></span>

<h3>Description</h3>

<p>Builds the list of common HTTP headers to send with each API call.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brapi_headers()
</code></pre>


<h3>Value</h3>

<p>A list of common HTTP headers to send.
</p>

<hr>
<h2 id='build_pedigree_table'>Building Pedigree Table Recursively</h2><span id='topic+build_pedigree_table'></span>

<h3>Description</h3>

<p>Internal helper function to build the pedigree table recursively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_pedigree_table(
  geno_list = NULL,
  pedigree_list = NULL,
  pedigree_df = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_pedigree_table_+3A_geno_list">geno_list</code></td>
<td>
<p>A list of genotypes/germplasm names.</p>
</td></tr>
<tr><td><code id="build_pedigree_table_+3A_pedigree_list">pedigree_list</code></td>
<td>
<p>A list of associated pedigree strings.</p>
</td></tr>
<tr><td><code id="build_pedigree_table_+3A_pedigree_df">pedigree_df</code></td>
<td>
<p>Pedigree data.frame as per the previous call/iteration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with three columns corresponding to the identifiers for the 
individual, female parent, and male parent, respectively.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='calc_biovars'>Calculate the Bioclimatic Variables</h2><span id='topic+calc_biovars'></span>

<h3>Description</h3>

<p>Bioclimatic variables are derived from the monthly temperature and rainfall 
values in order to generate more biologically meaningful variables. These are 
often used in species distribution modeling and related ecological modeling 
techniques. The bioclimatic variables represent annual trends (e.g., mean 
annual temperature, annual precipitation) seasonality (e.g., annual range in 
temperature and precipitation) and extreme or limiting environmental factors 
(e.g., temperature of the coldest and warmest month, and precipitation of the 
wet and dry quarters). A quarter is a period of three months (1/4 of the year).
</p>
<p>They are coded as follows:
</p>

<ul>
<li><p> BIO1 = Annual Mean Temperature
</p>
</li>
<li><p> BIO2 = Mean Diurnal Range (Mean of monthly (max temp - min temp))
</p>
</li>
<li><p> BIO3 = Isothermality (BIO2/BIO7) (* 100)
</p>
</li>
<li><p> BIO4 = Temperature Seasonality (standard deviation *100)
</p>
</li>
<li><p> BIO5 = Max Temperature of Warmest Month
</p>
</li>
<li><p> BIO6 = Min Temperature of Coldest Month
</p>
</li>
<li><p> BIO7 = Temperature Annual Range (BIO5-BIO6)
</p>
</li>
<li><p> BIO8 = Mean Temperature of Wettest Quarter
</p>
</li>
<li><p> BIO9 = Mean Temperature of Driest Quarter
</p>
</li>
<li><p> BIO10 = Mean Temperature of Warmest Quarter
</p>
</li>
<li><p> BIO11 = Mean Temperature of Coldest Quarter
</p>
</li>
<li><p> BIO12 = Annual Precipitation
</p>
</li>
<li><p> BIO13 = Precipitation of Wettest Month
</p>
</li>
<li><p> BIO14 = Precipitation of Driest Month
</p>
</li>
<li><p> BIO15 = Precipitation Seasonality (Coefficient of Variation)
</p>
</li>
<li><p> BIO16 = Precipitation of Wettest Quarter
</p>
</li>
<li><p> BIO17 = Precipitation of Driest Quarter
</p>
</li>
<li><p> BIO18 = Precipitation of Warmest Quarter
</p>
</li>
<li><p> BIO19 = Precipitation of Coldest Quarter
</p>
</li></ul>

<p>This work is derivative from the <a href="https://github.com/rspatial/dismo/blob/master/R/biovars.R">dismo R package</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_biovars(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_biovars_+3A_data">data</code></td>
<td>
<p>Data.frame has 4 mandatory columns (year, ppt, tmin, and tmax), 
and 12 rows (months) for each year sorted from Jan to Dec.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame has 19 columns for &quot;bioclim&quot; variables (bio1-bio19) and 
last column for year (you will get one row per year).
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>
<p>Robert Hijmans, Museum of Vertebrate Zoology, UC Berkeley
</p>


<h3>References</h3>

<p>Nix, 1986. A biogeographic analysis of Australian elapid snakes. 
In: R. Longmore (ed.). Atlas of elapid snakes of Australia. 
Australian Flora and Fauna Series 7. Australian Government Publishing 
Service, Canberra.
</p>

<hr>
<h2 id='debug_qbms'>Debug Internal QBMS Status Object</h2><span id='topic+debug_qbms'></span>

<h3>Description</h3>

<p>Returns the internal QBMS status object for debugging purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>debug_qbms()
</code></pre>


<h3>Value</h3>

<p>An environment object containing package configuration and status.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>obj &lt;- debug_qbms()
obj$config
obj$state

</code></pre>

<hr>
<h2 id='get_async_page'>Async Version of HTTP GET Request</h2><span id='topic+get_async_page'></span>

<h3>Description</h3>

<p>A small helper function to create an 'async' version of the original HTTP GET request.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_async_page(full_url, nested)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_async_page_+3A_full_url">full_url</code></td>
<td>
<p>URL to retrieve.</p>
</td></tr>
<tr><td><code id="get_async_page_+3A_nested">nested</code></td>
<td>
<p>Logical indicating whether to flatten nested data frames. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Async version of the HTTP GET request.
</p>

<hr>
<h2 id='get_async_pages'>Run All Supplied Pages</h2><span id='topic+get_async_pages'></span>

<h3>Description</h3>

<p>A small helper function to create a deferred value that is resolved when all 
listed pages are resolved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_async_pages(pages, nested)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_async_pages_+3A_pages">pages</code></td>
<td>
<p>List of URLs to retrieve.</p>
</td></tr>
<tr><td><code id="get_async_pages_+3A_nested">nested</code></td>
<td>
<p>Logical indicating whether to flatten nested data frames. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Async deferred object.
</p>

<hr>
<h2 id='get_germplasm_attributes'>Retrieve Attributes for a Specified Germplasm</h2><span id='topic+get_germplasm_attributes'></span>

<h3>Description</h3>

<p>Retrieve Attributes for a Specified Germplasm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_attributes(germplasm_name = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_germplasm_attributes_+3A_germplasm_name">germplasm_name</code></td>
<td>
<p>The name of the germplasm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the attributes of the specified germplasm.
</p>


<h3>Author(s)</h3>

<p>Johan Steven Aparicio, <a href="mailto:j.aparicio@cgiar.org">j.aparicio@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+get_germplasm_data">get_germplasm_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Log in using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Retrieve attribute data of a specified germplasm in a crop
  germplasm_attributes &lt;- get_germplasm_attributes("Jabal")
}
</code></pre>

<hr>
<h2 id='get_germplasm_data'>Retrieve Observations Data for a Specified Germplasm.</h2><span id='topic+get_germplasm_data'></span>

<h3>Description</h3>

<p>Retrieves all available observations data for a given germplasm in the current 
crop database, regardless of the nested structure of programs/trials.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_data(germplasm_name = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_germplasm_data_+3A_germplasm_name">germplasm_name</code></td>
<td>
<p>The name of the germplasm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the aggregated observations data for the germplasm from all trials.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+get_germplasm_attributes">get_germplasm_attributes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")
  
  # Log in using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()
  
  # Select a crop
  set_crop("wheat")
  
  # Select a breeding program by name
  set_program("Wheat International Nurseries")
  
  # Retrieve observations data for a specified germplasm aggregated from all trials
  germplasm_observations &lt;- get_germplasm_data("Jabal")
}
</code></pre>

<hr>
<h2 id='get_germplasm_id'>Get Germplasm ID</h2><span id='topic+get_germplasm_id'></span>

<h3>Description</h3>

<p>Retrieves the germplasm ID for the given germplasm name in the current crop.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_id(germplasm_name = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_germplasm_id_+3A_germplasm_name">germplasm_name</code></td>
<td>
<p>The name of the germplasm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string of the germplasm ID.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+get_germplasm_data">get_germplasm_data</a></code>, <code><a href="#topic+get_germplasm_attributes">get_germplasm_attributes</a></code>
</p>

<hr>
<h2 id='get_germplasm_list'>Get the Germplasm List of the Current Active Study</h2><span id='topic+get_germplasm_list'></span>

<h3>Description</h3>

<p>Retrieves the germplasm list of the current active study as configured in the 
internal state object using 'set_study()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_germplasm_list()
</code></pre>


<h3>Value</h3>

<p>A data frame of the study germplasm list.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+set_study">set_study</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Select a specific environment/location dataset
  set_study("IDYT39 Environment Number 9")

  # Retrieve the germplasm list of the selected environment/location
  germplasm &lt;- get_germplasm_list()
}

</code></pre>

<hr>
<h2 id='get_hwsd2'>Get HWSD v2 Soil Data for a Given Location(s)</h2><span id='topic+get_hwsd2'></span>

<h3>Description</h3>

<p>The HWSD2_SMU table contains general information for each of the soil units 
occurring in any given SMU code (dominant soil unit and up to 11 associated soils).
</p>
<p>The SEQUENCE column refers to the sequence in which soil units within the SMU 
are presented (in order of percentage share). The dominant soil has sequence 1. 
The sequence can range between 1 and 12.
</p>
<p>The SHARE column refers to the share of the soil unit within the mapping unit in 
percentage. Shares of soil units within a mapping unit always sum up to 100 percent.
</p>
<p>The HWSD2_LAYERS table provides soil attributes per depth layer for each of the 
seven depth layers (D1 to D7) separately (represented in the LAYER column in the 
HWSD2_LAYERS table). The depth of the top and bottom of each layer is defined in
the TOPDEP and BOTDEP columns, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hwsd2(df, con, x = "longitude", y = "latitude", sequence = 1, layer = "D1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_hwsd2_+3A_df">df</code></td>
<td>
<p>data.frame that list locations info including the coordinates in decimal degree format.</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_con">con</code></td>
<td>
<p>the HWSDv2 object returns from the ini_hwsd2() function.</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_x">x</code></td>
<td>
<p>longitude column name in the df data.frame (default is 'Longitude').</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_y">y</code></td>
<td>
<p>latitude column name in the df data.frame (default is 'Latitude').</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_sequence">sequence</code></td>
<td>
<p>the sequence in which soil units are presented (in order of percentage share). 
The dominant soil has sequence 1. The sequence can range between 1 and 12.</p>
</td></tr>
<tr><td><code id="get_hwsd2_+3A_layer">layer</code></td>
<td>
<p>the depth layer range from 'D1' to 'D7'. The depth of the top and bottom of 
each layer is defined in the TOPDEP and BOTDEP columns, respectively.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ini_hwsd2">ini_hwsd2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # create a simple data.frame for a list of locations and their coordinates
  Location  &lt;- c('Tel-Hadya', 'Terbol', 'Marchouch')
  Latitude  &lt;- c(36.016, 33.808, 33.616)
  Longitude &lt;- c(36.943, 35.991, -6.716)
  
  sites &lt;- data.frame(Location, Latitude, Longitude)
  
  # initiate, download, and setup the HWSD v2 in a given local directory
  hwsd2 &lt;- ini_hwsd2(data_path = 'C:/Users/Kel-shamaa/Downloads/HWSD v2/')
  
  # query soil attributes for given sites using the HWSD v2 connection object
  #
  # sequence parameter, range between 1 and 12 (max), 1 is the dominant soil.
  # returned df has SHARE column refers to share%
  #
  # layer parameter refers to depth layer (D1 to D7).
  # returned df has TOPDEP/BOTDEP columns represent top/bottom layer depth in cm.
  sites &lt;- get_hwsd2(df = sites, 
                     con = hwsd2, 
                     x = 'Longitude', 
                     y = 'Latitude', 
                     sequence = 1, 
                     layer = 'D1')
}
</code></pre>

<hr>
<h2 id='get_login_details'>Login Pop-Up Window</h2><span id='topic+get_login_details'></span>

<h3>Description</h3>

<p>Builds a GUI pop-up window using Tcl/Tk to insert the username and password.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_login_details()
</code></pre>


<h3>Value</h3>

<p>A vector of inserted username and password.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_parents'>Get Direct Parents</h2><span id='topic+get_parents'></span>

<h3>Description</h3>

<p>Internal helper function to split the given pedigree string that provides the parentage
through which a cultivar was obtained and retrieve the pedigrees of the direct parents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_parents(pedigree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_parents_+3A_pedigree">pedigree</code></td>
<td>
<p>A string providing the parentage through which a cultivar was obtained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of two items, representing the direct female and male parents.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='get_pedigree_table'>Get the Pedigree Table</h2><span id='topic+get_pedigree_table'></span>

<h3>Description</h3>

<p>Retrieve the pedigree table starting from the current germplasm list and associated
pedigree string that provides the parentage through which a cultivar was obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigree_table(
  data,
  geno_column = "germplasmName",
  pedigree_column = "pedigree"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pedigree_table_+3A_data">data</code></td>
<td>
<p>Germplasm dataset as a data.frame.</p>
</td></tr>
<tr><td><code id="get_pedigree_table_+3A_geno_column">geno_column</code></td>
<td>
<p>Name of the column that identifies the genotype/germplasm names.</p>
</td></tr>
<tr><td><code id="get_pedigree_table_+3A_pedigree_column">pedigree_column</code></td>
<td>
<p>Name of the column that identifies the pedigree strings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with three columns corresponding to the identifiers for the individual,
female parent, and male parent, respectively. The row giving the pedigree of an
individual appears before any row where that individual appears as a parent.
Founders use NA in the parental columns.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Select a specific environment/location dataset
  set_study("IDYT39 Environment Number 9")

  # Retrieve the germplasm list of the selected environment/location
  germplasm &lt;- get_germplasm_list()

  pedigree_table &lt;- get_pedigree_table(germplasm, "germplasmName", "pedigree")

  #############################
  # nadiv package way
  # library(nadiv)

  # Get additive relationship matrix in sparse matrix format
  # A &lt;- nadiv::makeA(pedigree_table)

  # Get A inverse matrix using base R function
  # AINV &lt;- solve(as.matrix(A))

  #############################
  # ASReml-R package way
  # library(asreml)

  # Represent A inverse matrix in an efficient way using i, j index and Ainverse value
  # Actual genotype names of any given index are in the attr(ainv, "rowNames")
  # ainv &lt;- asreml::ainverse(pedigree_table)

  #############################
  # Dummy data set for testing
  test &lt;- data.frame(genotype = c("X", "Y"),
                     pedigree = c("A//B/D/2/C", "B/C/3/A//B/C/2/D"))

  pedigree_table &lt;- get_pedigree_table(test, "genotype", "pedigree")
}
</code></pre>

<hr>
<h2 id='get_program_studies'>Get the List of Trials/Studies/Locations Information of the Current Selected Program</h2><span id='topic+get_program_studies'></span>

<h3>Description</h3>

<p>Retrieves all environments/locations information of the trials studies in the 
current active program as configured in the internal state object using the 
'set_program()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_program_studies()
</code></pre>


<h3>Value</h3>

<p>A data frame of locations information for each study in the program trials.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # config your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # login using your account (interactive mode)
  # you can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # select a breeding program by name
  set_program("Wheat International Nurseries")

  # retrieve all environments/locations information in the selected program studies/trials
  program_studies &lt;- get_program_studies()
}

</code></pre>

<hr>
<h2 id='get_program_trials'>Internal Function Used to Retrieve the Rough List of Trials</h2><span id='topic+get_program_trials'></span>

<h3>Description</h3>

<p>This function is created for *internal use only* to retrieve the raw list of trials
from the pre-selected (i.e., currently active) crop and breeding program combination
as already configured in the internal state object using 'set_crop()' and 'set_program()'
functions, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_program_trials()
</code></pre>


<h3>Value</h3>

<p>A list of trials information.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+list_trials">list_trials</a></code>
</p>

<hr>
<h2 id='get_qbms_connection'>Get the QBMS Connection</h2><span id='topic+get_qbms_connection'></span>

<h3>Description</h3>

<p>Retrieves the QBMS connection object from the current environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_qbms_connection()
</code></pre>


<h3>Value</h3>

<p>A list containing the current connection configuration and status.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_connection">set_qbms_connection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")
     
  # Get germplasm data
  df1 &lt;- get_germplasm_data("Jabal")
  
  # Save the current connection (phenotypic server)
  con1 &lt;- get_qbms_connection()
  
  # Configure QBMS to connect to the genotypic server
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", engine = "gigwa", no_auth = TRUE)
  
  # Set the db, project, and run
  gigwa_set_db("DIVRICE_NB")
  gigwa_set_project("refNB")
  gigwa_set_run("03052022")
  
  # Get associated metadata
  df2 &lt;- gigwa_get_metadata()
  
  # Save the current connection (before switch)
  con2 &lt;- get_qbms_connection()
  
  # Load the saved phenotypic server connection
  set_qbms_connection(con1)
  
  # Continue retrieving germplasm attributes from the phenotypic server
  df3 &lt;- get_germplasm_attributes("Jabal")
}

</code></pre>

<hr>
<h2 id='get_study_data'>Get the Observations Data of the Current Active Study</h2><span id='topic+get_study_data'></span>

<h3>Description</h3>

<p>Retrieves the observations data of the current active study as configured in 
the internal state object using 'set_study()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_data()
</code></pre>


<h3>Value</h3>

<p>A data frame of the study observations data.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+set_study">set_study</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Select a specific environment/location dataset
  set_study("IDYT39 Environment Number 9")

  # Retrieve the data of the selected environment/location
  data &lt;- get_study_data()
}

</code></pre>

<hr>
<h2 id='get_study_info'>Get the Details/Metadata of the Current Active Study</h2><span id='topic+get_study_info'></span>

<h3>Description</h3>

<p>Retrieves the details/metadata of the current active study as configured in 
the internal state object using 'set_study()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_info()
</code></pre>


<h3>Value</h3>

<p>A data frame of the study details/metadata.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+set_study">set_study</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Select a specific environment/location dataset
  set_study("IDYT39 Environment Number 9")

  # Retrieve the general information of the selected environment/location
  info &lt;- get_study_info()
}

</code></pre>

<hr>
<h2 id='get_terraclimate'>Get TerraClimate Data for a Given Coordinate(s)</h2><span id='topic+get_terraclimate'></span>

<h3>Description</h3>

<p><a href="https://www.climatologylab.org/terraclimate.html">TerraClimate</a> is a monthly climate dataset 
for global terrestrial surfaces from 1958-2021. This function enables you to extract 
<a href="https://www.climatologylab.org/terraclimate-variables.html">climate variables</a> from the 
<a href="http://thredds.northwestknowledge.net:8080/thredds/terraclimate_aggregated.html">hosting server</a> 
provided by the <a href="https://hpc.uidaho.edu/">Idaho University</a> for a given coordinate(s) without 
a need to download the whole raster files in the netCDF format (~100MB per variable for each year) 
and provide them in a standard data frame format ready to use in your code. It also calculates the 
<a href="https://www.worldclim.org/data/bioclim.html">bioclimatic variables</a> using the <code><a href="#topic+calc_biovars">calc_biovars</a></code> 
function derivative from the <a href="https://github.com/rspatial/dismo/blob/master/R/biovars.R">dismo R package</a>.
</p>
<p>TerraClimate vs. <a href="https://www.worldclim.org/data/worldclim21.html">WorldClim</a>
</p>

<ul>
<li><p> 1958-2021 vs. 1970-2000
</p>
</li>
<li><p> 14 vs. 7 climate variables
</p>
</li>
<li><p> ~4 km vs. ~1 km spatial resolution
</p>
</li>
<li><p> need to calculate vs. pre-calculated 19 bioclimatic variables
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>get_terraclimate(
  lat,
  lon,
  from = "1958-01-01",
  to = "2022-12-31",
  clim_vars = NULL,
  month_mask = NULL,
  offline = FALSE,
  data_path = "./data/"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_terraclimate_+3A_lat">lat</code></td>
<td>
<p>Vector of Latitude(s) in decimal degree format.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_lon">lon</code></td>
<td>
<p>Vector of Longitude(s) in decimal degree format.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_from">from</code></td>
<td>
<p>Start date as a string in the 'YYYY-MM-DD' format.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_to">to</code></td>
<td>
<p>End date as a string in the 'YYYY-MM-DD' format.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_clim_vars">clim_vars</code></td>
<td>
<p>List of all climate variables to be imported. Valid list includes: <em>aet, def, pet,
ppt, q, soil, srad, swe, tmax, tmin, vap, ws, vpd, and PDSI</em>. Default is NULL for all.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_month_mask">month_mask</code></td>
<td>
<p>A list of all months of interest (e.g., planting season: <code>c(10:12,1:5)</code>). 
Default is NULL for all.</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_offline">offline</code></td>
<td>
<p>Extract TerraClimate data from downloaded netCDF files (default is FALSE)</p>
</td></tr>
<tr><td><code id="get_terraclimate_+3A_data_path">data_path</code></td>
<td>
<p>String contains the directory path where downloaded netCDF files exists (default is './data/')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two data.frame(s) for each pair of coordinates:
</p>

<ul>
<li> <p><strong>climate:</strong> includes the climate variables (<a href="https://www.climatologylab.org/terraclimate-variables.html">reference</a>).
</p>
</li>
<li> <p><strong>biovars:</strong> includes the calculated bioclimatic variables (<a href="https://www.worldclim.org/data/bioclim.html">reference</a>).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>References</h3>

<p>Abatzoglou, J., Dobrowski, S., Parks, S. <em>et al.</em> TerraClimate, a high-resolution 
global dataset of monthly climate and climatic water balance from 1958-2015. 
<em>Sci Data</em> <strong>5</strong>, 170191 (2018). 
<a href="https://doi.org/10.1038/sdata.2017.191">doi:10.1038/sdata.2017.191</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ini_terraclimate">ini_terraclimate</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # data &lt;- get_terraclimate(36.016, 36.943, 
  #                          '1979-09-01', '2012-06-30', 
  #                          c('ppt', 'tmin', 'tmax'), c(10:12,1:5))
  data &lt;- get_terraclimate(36.016, 36.943, '1979-09-01', '2012-06-30')

  View(data$climate[[1]])

  View(data$biovars[[1]])
}
</code></pre>

<hr>
<h2 id='get_trial_data'>Get the Observations Data of the Current Active Trial</h2><span id='topic+get_trial_data'></span>

<h3>Description</h3>

<p>Retrieves the observations data of the current active trial (i.e., including 
all studies within) as configured in the internal state object using 
'set_trial()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trial_data()
</code></pre>


<h3>Value</h3>

<p>A data frame of the trial observations data.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Retrieve multi-environment trial data
  MET &lt;- get_trial_data()
}

</code></pre>

<hr>
<h2 id='get_trial_obs_ontology'>Get the Traits Ontology/Metadata of the Current Active Trial</h2><span id='topic+get_trial_obs_ontology'></span>

<h3>Description</h3>

<p>Retrieves the traits ontology/metadata of the current active trial as 
configured in the internal state object using the 'set_trial()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trial_obs_ontology()
</code></pre>


<h3>Value</h3>

<p>A data frame of the traits ontology/metadata.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Get observation variable ontology
  ontology &lt;- get_trial_obs_ontology()
}

</code></pre>

<hr>
<h2 id='gigwa_get_metadata'>Get the Metadata of the Current Active GIGWA Run</h2><span id='topic+gigwa_get_metadata'></span>

<h3>Description</h3>

<p>This function retrieves the metadata of the current active run
as configured in the internal state object using the 'gigwa_set_run()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_metadata()
</code></pre>


<h3>Value</h3>

<p>A data.frame of all metadata associated with the samples in the selected run.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_run">gigwa_set_run</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("3kG_10M")

  # Select a project by name
  gigwa_set_project("3003_ind")
  
  # Select a specific run by name
  gigwa_set_run("1")
  
  # Get a list of all samples in the selected run
  metadata &lt;- gigwa_get_metadata()
}
</code></pre>

<hr>
<h2 id='gigwa_get_samples'>Get the Samples List of the Current Active GIGWA Run</h2><span id='topic+gigwa_get_samples'></span>

<h3>Description</h3>

<p>This function retrieves the samples list of the current active run
as configured in the internal state object using the 'gigwa_set_run()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_samples()
</code></pre>


<h3>Value</h3>

<p>A vector of all samples in the selected run.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_run">gigwa_set_run</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")

  # Select a project by name
  gigwa_set_project("Nelson_et_al_2011")
  
  # Select a specific run by name
  gigwa_set_run("run1")
  
  # Get a list of all samples in the selected run
  samples &lt;- gigwa_get_samples()
}
</code></pre>

<hr>
<h2 id='gigwa_get_variants'>Get Available Variants in the Selected GIGWA Run</h2><span id='topic+gigwa_get_variants'></span>

<h3>Description</h3>

<p>Query the variants (e.g., SNPs markers) in the selected GIGWA run that match a given criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_get_variants(
  max_missing = 1,
  min_maf = 0,
  samples = NULL,
  start = NULL,
  end = NULL,
  referenceName = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gigwa_get_variants_+3A_max_missing">max_missing</code></td>
<td>
<p>Maximum missing ratio (by sample) between 0 and 1 (default is 1 for 100%).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_min_maf">min_maf</code></td>
<td>
<p>Minimum Minor Allele Frequency (MAF) between 0 and 1 (default is 0 for 0%).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_samples">samples</code></td>
<td>
<p>A list of samples subset (default is NULL, which will retrieve for all samples).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_start">start</code></td>
<td>
<p>Start position of region (zero-based, inclusive) (e.g., 19750802).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_end">end</code></td>
<td>
<p>End position of region (zero-based, exclusive)    (e.g., 19850125).</p>
</td></tr>
<tr><td><code id="gigwa_get_variants_+3A_referencename">referenceName</code></td>
<td>
<p>Reference sequence name    (e.g., '6H' in the Barley LI-AM).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that has the first 4 columns describing attributes of the SNP 
(rs#: variant name, alleles: reference allele / alternative allele, chrom: chromosome name, 
and pos: position in bp), while the following columns describe the SNP value for a 
single sample line using numerical coding 0, 1, and 2 for reference, heterozygous, and 
alternative/minor alleles.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")

  # Select a project by name
  gigwa_set_project("Nelson_et_al_2011")
  
  # Select a specific run by name
  gigwa_set_run("run1")
  
  marker_matrix &lt;- gigwa_get_variants(max_missing = 0.2, 
                                      min_maf = 0.35, 
                                      samples = c("ind1", "ind3", "ind7"))
}
</code></pre>

<hr>
<h2 id='gigwa_list_dbs'>List GIGWA Databases</h2><span id='topic+gigwa_list_dbs'></span>

<h3>Description</h3>

<p>Get the list of existing databases in the current GIGWA server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_list_dbs()
</code></pre>


<h3>Value</h3>

<p>A list of existing databases.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # List existing databases in the GIGWA server
  gigwa_list_dbs()
}
</code></pre>

<hr>
<h2 id='gigwa_list_projects'>Get the List of All Projects in the Selected GIGWA Database</h2><span id='topic+gigwa_list_projects'></span>

<h3>Description</h3>

<p>Retrieves the projects list from the currently active database as configured 
in the internal configuration object using the 'gigwa_set_db()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_list_projects()
</code></pre>


<h3>Value</h3>

<p>A list of project names.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_db">gigwa_set_db</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")

  # List existing projects
  gigwa_list_projects()
}
</code></pre>

<hr>
<h2 id='gigwa_list_runs'>Get the List of the Run Names Available in the Selected GIGWA Project</h2><span id='topic+gigwa_list_runs'></span>

<h3>Description</h3>

<p>This function retrieves the runs list from the currently active project as configured 
in the internal configuration object using the 'gigwa_set_project()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_list_runs()
</code></pre>


<h3>Value</h3>

<p>A list of run names.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_project">gigwa_set_project</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")

  # Select a project by name
  gigwa_set_project("Nelson_et_al_2011")
  
  # List all runs in the selected project
  gigwa_list_runs()
}
</code></pre>

<hr>
<h2 id='gigwa_set_db'>Set the Current Active GIGWA Database by Name</h2><span id='topic+gigwa_set_db'></span>

<h3>Description</h3>

<p>This function updates the current active database in the internal
configuration object (including the brapi connection object).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_set_db(db_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gigwa_set_db_+3A_db_name">db_name</code></td>
<td>
<p>The name of the database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_list_dbs">gigwa_list_dbs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")
}
</code></pre>

<hr>
<h2 id='gigwa_set_project'>Set the Current Active GIGWA Project</h2><span id='topic+gigwa_set_project'></span>

<h3>Description</h3>

<p>This function updates the current active project in the internal state object 
using the programDbId retrieved from GIGWA which is associated with the given 
'project_name' parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_set_project(project_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gigwa_set_project_+3A_project_name">project_name</code></td>
<td>
<p>The name of the project.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_db">gigwa_set_db</a></code>, <code><a href="#topic+gigwa_list_projects">gigwa_list_projects</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")

  # Select a project by name
  gigwa_set_project("Nelson_et_al_2011")
}
</code></pre>

<hr>
<h2 id='gigwa_set_run'>Set the Current Active GIGWA Run</h2><span id='topic+gigwa_set_run'></span>

<h3>Description</h3>

<p>This function updates the current active run in the internal state object using the 
'studyDbIds' retrieved from GIGWA, which are associated with the given 'run_name' parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gigwa_set_run(run_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gigwa_set_run_+3A_run_name">run_name</code></td>
<td>
<p>The name of the run.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set_qbms_config">set_qbms_config</a></code>, <code><a href="#topic+gigwa_set_project">gigwa_set_project</a></code>, <code><a href="#topic+gigwa_list_runs">gigwa_list_runs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", 
                  time_out = 300, engine = "gigwa", no_auth = TRUE)

  # Select a database by name
  gigwa_set_db("Sorghum-JGI_v1")

  # Select a project by name
  gigwa_set_project("Nelson_et_al_2011")
  
  # Select a specific run by name
  gigwa_set_run("run1")
}
</code></pre>

<hr>
<h2 id='ini_hwsd2'>Download and Setup HWSD v2.0 Data Files to Extract their Data Offline</h2><span id='topic+ini_hwsd2'></span>

<h3>Description</h3>

<p>Download and Setup HWSD v2.0 Data Files to Extract their Data Offline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ini_hwsd2(data_path = "./data/", timeout = 300)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ini_hwsd2_+3A_data_path">data_path</code></td>
<td>
<p>String containing the directory path where downloaded HWSD v2.0 data files exist (default is './data/').</p>
</td></tr>
<tr><td><code id="ini_hwsd2_+3A_timeout">timeout</code></td>
<td>
<p>Timeout in seconds to download each HWSD v2.0 data file (default is 300).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>HWSDv2 object that has a list of two items: the HWSD2 raster and the HWSD2 SQLite connection.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_hwsd2">get_hwsd2</a></code>
</p>

<hr>
<h2 id='ini_terraclimate'>Download TerraClimate netCDF Data Files to Extract their Data Offline</h2><span id='topic+ini_terraclimate'></span>

<h3>Description</h3>

<p>Download TerraClimate netCDF Data Files to Extract their Data Offline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ini_terraclimate(
  from = "2019-09-01",
  to = "2022-06-30",
  clim_vars = c("ppt", "tmin", "tmax"),
  data_path = "./data/",
  timeout = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ini_terraclimate_+3A_from">from</code></td>
<td>
<p>Start date as a string in the 'YYYY-MM-DD' format.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_to">to</code></td>
<td>
<p>End date as a string in the 'YYYY-MM-DD' format.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_clim_vars">clim_vars</code></td>
<td>
<p>List of all climate variables to be imported. Valid list includes: <em>aet, def, pet,
ppt, q, soil, srad, swe, tmax, tmin, vap, ws, vpd, and PDSI</em>. Default is NULL for all.</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_data_path">data_path</code></td>
<td>
<p>String containing the directory path where downloaded netCDF files exist (default is './data/')</p>
</td></tr>
<tr><td><code id="ini_terraclimate_+3A_timeout">timeout</code></td>
<td>
<p>Timeout in seconds to download each netCDF raster file (default is 300).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_terraclimate">get_terraclimate</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  ini_terraclimate('2018-09-01', '2019-06-30', c('ppt', 'tmin', 'tmax'))
  
  x &lt;- c(-6.716, 35.917, 76.884)
  y &lt;- c(33.616, 33.833, 23.111)
  
  a &lt;- get_terraclimate(y, x, '2018-09-01', '2019-06-30', c('ppt', 'tmin', 'tmax'))
  
  a$climate[[1]]
  a$biovars[[1]]
  
  b &lt;- get_terraclimate(y, x, '2018-09-01', '2019-06-30', c('ppt', 'tmin', 'tmax'), offline = TRUE)
  
  b$climate[[1]]
  b$biovars[[1]]
}
</code></pre>

<hr>
<h2 id='list_crops'>Get the List of Supported Crops</h2><span id='topic+list_crops'></span>

<h3>Description</h3>

<p>Retrieves the list of supported crops.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_crops()
</code></pre>


<h3>Value</h3>

<p>A list of supported crops.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your BMS connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your BMS account (interactive mode)
  # You can pass the BMS username and password as parameters (batch mode)
  login_bms()

  # List supported crops in the BMS server
  list_crops()
}

</code></pre>

<hr>
<h2 id='list_locations'>Get the List of Locations Information of the Current Selected Crop</h2><span id='topic+list_locations'></span>

<h3>Description</h3>

<p>Retrieves the locations information of the current active crop as configured 
in the internal state object using the 'set_crop()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_locations()
</code></pre>


<h3>Value</h3>

<p>A data frame of the locations information.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>
</p>

<hr>
<h2 id='list_programs'>Get the List of Breeding Programs Names</h2><span id='topic+list_programs'></span>

<h3>Description</h3>

<p>Retrieves the breeding programs list from the current active crop as configured 
in the internal configuration object using 'set_crop()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_programs()
</code></pre>


<h3>Value</h3>

<p>A list of breeding programs names.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # List existing breeding programs
  list_programs()
}

</code></pre>

<hr>
<h2 id='list_studies'>Get the List of Studies in the Current Active Trial</h2><span id='topic+list_studies'></span>

<h3>Description</h3>

<p>Retrieves the studies list from the current active trial as configured in the 
internal state object using 'set_trial()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_studies()
</code></pre>


<h3>Value</h3>

<p>A list of study and location names.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # List all environments/locations information in the selected study/trial
  list_studies()
}

</code></pre>

<hr>
<h2 id='list_trials'>Get the List of Trials in the Current Active Breeding Program</h2><span id='topic+list_trials'></span>

<h3>Description</h3>

<p>Retrieves the trials list from the current active breeding program as configured 
in the internal state object using 'set_program()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_trials(year = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list_trials_+3A_year">year</code></td>
<td>
<p>The starting year to filter the list of trials (optional, default is NULL).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of trials names.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # List all studies/trials in the selected program
  list_trials()

  # Filter listed studies/trials by year
  list_trials(2022)
}

</code></pre>

<hr>
<h2 id='login_bms'>Login to the Server</h2><span id='topic+login_bms'></span>

<h3>Description</h3>

<p>Connects to the server. If the username or password parameters are missing,
then a login window will pop up to insert the username and password.
</p>
<p>All other connection parameters (i.e., server IP or domain, connection port,
API path, and connection protocol e.g., http://) will be retrieved from the
qbms_config list.
</p>
<p>This function will update both the qbms_config list (brapi connection
object in the con key) and the qbms_state list (token value in the token key).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_bms(username = NULL, password = NULL, encoding = "json")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="login_bms_+3A_username">username</code></td>
<td>
<p>The username (optional, default is NULL).</p>
</td></tr>
<tr><td><code id="login_bms_+3A_password">password</code></td>
<td>
<p>The password (optional, default is NULL).</p>
</td></tr>
<tr><td><code id="login_bms_+3A_encoding">encoding</code></td>
<td>
<p>How should the named list body be encoded? Can be one of form 
(application/x-www-form-urlencoded), multipart (multipart/form-data), 
or json (application/json).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your BMS connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your BMS account (interactive mode)
  # You can pass the BMS username and password as parameters (batch mode)
  login_bms()
}

</code></pre>

<hr>
<h2 id='login_breedbase'>Login to the BreedBase Server</h2><span id='topic+login_breedbase'></span>

<h3>Description</h3>

<p>Logs in to the BreedBase server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_breedbase(username = NULL, password = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="login_breedbase_+3A_username">username</code></td>
<td>
<p>The username (optional, default is NULL).</p>
</td></tr>
<tr><td><code id="login_breedbase_+3A_password">password</code></td>
<td>
<p>The password (optional, default is NULL).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='login_gigwa'>Login to the GIGWA Server</h2><span id='topic+login_gigwa'></span>

<h3>Description</h3>

<p>Connect to the GIGWA server. If the username or password parameters are missing,
a login window will pop up to insert the username and password.
</p>
<p>All other connection parameters (i.e., server IP or domain, connection port,
API path, and connection protocol e.g., http://) will be retrieved from the
'qbms_config' list.
</p>
<p>This function will update both the qbms_config list (brapi connection
object in the con key) and qbms_state list (token value in the token key).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_gigwa(username = NULL, password = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="login_gigwa_+3A_username">username</code></td>
<td>
<p>The GIGWA username (optional, default is NULL).</p>
</td></tr>
<tr><td><code id="login_gigwa_+3A_password">password</code></td>
<td>
<p>The GIGWA password (optional, default is NULL).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Configure your GIGWA connection
  set_qbms_config("http://localhost:59395/gigwa/index.jsp", time_out = 300, engine = "gigwa")

  # Login using your GIGWA account (interactive mode)
  login_gigwa()
  
  # You can pass GIGWA username and password as parameters (batch mode)
  # login_gigwa("gigwadmin", "nimda")
}
</code></pre>

<hr>
<h2 id='login_oauth2'>Login Using OAuth 2.0 Authentication</h2><span id='topic+login_oauth2'></span>

<h3>Description</h3>

<p>If the request for an access token is valid, the authorization server needs 
to generate an access token and return these to the client, typically along 
with some additional properties about the authorization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>login_oauth2(
  authorize_url,
  access_url,
  client_id,
  client_secret = NULL,
  redirect_uri = "http://localhost:1410",
  oauth2_cache = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="login_oauth2_+3A_authorize_url">authorize_url</code></td>
<td>
<p>URL to send the client for authorization.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_access_url">access_url</code></td>
<td>
<p>URL used to exchange unauthenticated for authenticated token.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_client_id">client_id</code></td>
<td>
<p>Consumer key, also sometimes called the client ID.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_client_secret">client_secret</code></td>
<td>
<p>Consumer secret, also sometimes called the client secret.</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_redirect_uri">redirect_uri</code></td>
<td>
<p>The URL that the user will be redirected to after authorization is complete (default is http://localhost:1410).</p>
</td></tr>
<tr><td><code id="login_oauth2_+3A_oauth2_cache">oauth2_cache</code></td>
<td>
<p>A logical value or a string. TRUE means to cache using the default cache file .httr-oauth, FALSE means don't cache, 
and NA means to guess using some sensible heuristics. A string means use the specified path as the cache file.
Default is FALSE (i.e., don't cache).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='rbindlistx'>Make One Data.Table from a List of Many</h2><span id='topic+rbindlistx'></span>

<h3>Description</h3>

<p>Performs the equivalent of do.call(&quot;rbind&quot;, x) on data.frames, but much faster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbindlistx(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbindlistx_+3A_x">x</code></td>
<td>
<p>A list containing data.table, data.frame, or list objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An unkeyed data.table containing a concatenation of all the items passed in.
</p>

<hr>
<h2 id='rbindx'>Combine Data Frames by Row, Filling in Missing Columns</h2><span id='topic+rbindx'></span>

<h3>Description</h3>

<p>Combines a list of data frames by row, filling in missing columns with NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbindx(..., dfs = list(...))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbindx_+3A_...">...</code></td>
<td>
<p>The first argument data frame.</p>
</td></tr>
<tr><td><code id="rbindx_+3A_dfs">dfs</code></td>
<td>
<p>Input data frames to row bind together.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single data frame.
</p>

<hr>
<h2 id='scan_brapi_endpoints'>Scan BrAPI Endpoints</h2><span id='topic+scan_brapi_endpoints'></span>

<h3>Description</h3>

<p>Scan available BrAPI endpoints on the configured source server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scan_brapi_endpoints(programDbId = 0, trialDbId = 0, studyDbId = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scan_brapi_endpoints_+3A_programdbid">programDbId</code></td>
<td>
<p>(numeric) ProgramDbId used for BrAPI endpoints scanning. Default is 0.</p>
</td></tr>
<tr><td><code id="scan_brapi_endpoints_+3A_trialdbid">trialDbId</code></td>
<td>
<p>(numeric) TrialDbId used for BrAPI endpoints scanning. Default is 0.</p>
</td></tr>
<tr><td><code id="scan_brapi_endpoints_+3A_studydbid">studyDbId</code></td>
<td>
<p>(numeric) StudyDbId used for BrAPI endpoints scanning. Default is 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame listing the QBMS function, BrAPI endpoint URL, and status.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='set_crop'>Set the Current Active Crop</h2><span id='topic+set_crop'></span>

<h3>Description</h3>

<p>Updates the current active crop in the internal configuration object (including 
the BrAPI connection object).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_crop(crop_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_crop_+3A_crop_name">crop_name</code></td>
<td>
<p>The name of the crop.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+list_crops">list_crops</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")
}

</code></pre>

<hr>
<h2 id='set_program'>Set the Current Active Breeding Program</h2><span id='topic+set_program'></span>

<h3>Description</h3>

<p>Updates the current active breeding program in the internal state object 
using the programDbId which is associated with the given program_name parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_program(program_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_program_+3A_program_name">program_name</code></td>
<td>
<p>The name of the breeding program.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+list_programs">list_programs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")
}

</code></pre>

<hr>
<h2 id='set_qbms_config'>Configure BMS Server Settings</h2><span id='topic+set_qbms_config'></span>

<h3>Description</h3>

<p>Sets the connection configuration of the BMS server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_qbms_config(
  url = "http://localhost/ibpworkbench/controller/auth/login",
  path = NULL,
  page_size = 1000,
  time_out = 120,
  no_auth = FALSE,
  engine = "bms",
  brapi_ver = "v1",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_qbms_config_+3A_url">url</code></td>
<td>
<p>URL of the BMS login page. Default is &quot;http://localhost/ibpworkbench/&quot;.</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_path">path</code></td>
<td>
<p>API path. Default is NULL.</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_page_size">page_size</code></td>
<td>
<p>Page size. Default is 1000.</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_time_out">time_out</code></td>
<td>
<p>Number of seconds to wait for a response until giving up. Default is 10.</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_no_auth">no_auth</code></td>
<td>
<p>TRUE if the server doesn't require authentication/login. Default is FALSE.</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_engine">engine</code></td>
<td>
<p>Backend database (bms default, breedbase, gigwa, ebs).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_brapi_ver">brapi_ver</code></td>
<td>
<p>BrAPI version (v1 or v2).</p>
</td></tr>
<tr><td><code id="set_qbms_config_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating if progress bar will display on the console when retrieving data from API. TRUE by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set_qbms_config("https://bmsdev-brapi.ibp.services/ibpworkbench")

</code></pre>

<hr>
<h2 id='set_qbms_connection'>Set the QBMS Connection</h2><span id='topic+set_qbms_connection'></span>

<h3>Description</h3>

<p>Sets the QBMS connection object to the current environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_qbms_connection(env)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_qbms_connection_+3A_env">env</code></td>
<td>
<p>A list containing the connection configuration and status to load.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_qbms_connection">get_qbms_connection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")
     
  # Get germplasm data
  df1 &lt;- get_germplasm_data("Jabal")
  
  # Save the current connection (phenotypic server)
  con1 &lt;- get_qbms_connection()
  
  # Configure QBMS to connect to the genotypic server
  set_qbms_config("https://gigwa.southgreen.fr/gigwa/", engine = "gigwa", no_auth = TRUE)
  
  # Set the db, project, and run
  gigwa_set_db("DIVRICE_NB")
  gigwa_set_project("refNB")
  gigwa_set_run("03052022")
  
  # Get associated metadata
  df2 &lt;- gigwa_get_metadata()
  
  # Save the current connection (before switch)
  con2 &lt;- get_qbms_connection()
  
  # Load the saved phenotypic server connection
  set_qbms_connection(con1)
  
  # Continue retrieving germplasm attributes from the phenotypic server
  df3 &lt;- get_germplasm_attributes("Jabal")
}

</code></pre>

<hr>
<h2 id='set_study'>Set the Current Active Study</h2><span id='topic+set_study'></span>

<h3>Description</h3>

<p>Updates the current active study in the internal state object using the 
studyDbId, which is associated with the given study_name 
parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_study(study_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_study_+3A_study_name">study_name</code></td>
<td>
<p>The name of the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+set_trial">set_trial</a></code>, <code><a href="#topic+list_studies">list_studies</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")

  # Select a specific environment/location dataset
  set_study("IDYT39 Environment Number 9")
}

</code></pre>

<hr>
<h2 id='set_token'>Set Access Token Response</h2><span id='topic+set_token'></span>

<h3>Description</h3>

<p>If the request for an access token is valid, the authorization server needs 
to generate an access token and return these to the client, typically along 
with some additional properties about the authorization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_token(token, user = "", expires_in = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_token_+3A_token">token</code></td>
<td>
<p>The access token string as issued by the authorization server.</p>
</td></tr>
<tr><td><code id="set_token_+3A_user">user</code></td>
<td>
<p>The username (optional).</p>
</td></tr>
<tr><td><code id="set_token_+3A_expires_in">expires_in</code></td>
<td>
<p>The lifetime in seconds of the access token (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>

<hr>
<h2 id='set_trial'>Set the Current Active Trial</h2><span id='topic+set_trial'></span>

<h3>Description</h3>

<p>Updates the current active trial in the internal state object using the 
trialDbId which is associated with the given trial_name parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_trial(trial_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_trial_+3A_trial_name">trial_name</code></td>
<td>
<p>The name of the trial.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Khaled Al-Shamaa, <a href="mailto:k.el-shamaa@cgiar.org">k.el-shamaa@cgiar.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+login_bms">login_bms</a></code>, <code><a href="#topic+set_crop">set_crop</a></code>, <code><a href="#topic+set_program">set_program</a></code>, <code><a href="#topic+list_trials">list_trials</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()) {
  # Configure your server connection
  set_qbms_config("https://bms.icarda.org/ibpworkbench")

  # Login using your account (interactive mode)
  # You can pass your username and password as parameters (batch mode)
  login_bms()

  # Select a crop by name
  set_crop("wheat")

  # Select a breeding program by name
  set_program("Wheat International Nurseries")

  # Select a specific study/trial by name
  set_trial("IDYT39")
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
