<!DOCTYPE html><html><head><title>Help for package QCEWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QCEWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#EWAS_plots'><p>Manhattan and Quantile-Quantile plots for EWAS results files</p></a></li>
<li><a href='#EWAS_QC'><p>Automated Quality Control of EWAS results files</p></a></li>
<li><a href='#EWAS_series'>
<p>Quality Control and Comparison of multiple EWAS results files</p></a></li>
<li><a href='#P_correlation'><p>Testing P-value distribution</p></a></li>
<li><a href='#P_lambda'><p>Calculation of the Lambda value</p></a></li>
<li><a href='#QCEWAS-package'><p>Quality Control of Epigenome-Wide Association Study Results</p></a></li>
<li><a href='#translate_header'><p>Translate column names into standard names</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fast and Easy Quality Control of EWAS Results Files</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2-3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-02-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter J. van der Most,
		Leanne K. Kupers,
		Ilja Nolte</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peter J. van der Most &lt;p.j.van.der.most@umcg.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), methods</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for (automated and manual) quality control of 
  the results of Epigenome-Wide Association Studies.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-03 15:08:08 UTC; Gebruilker</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-07 12:22:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='EWAS_plots'>Manhattan and Quantile-Quantile plots for EWAS results files</h2><span id='topic+EWAS_plots'></span>

<h3>Description</h3>

<p>This function is used by <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> to generate
quantile-quantile (QQ) and Manhattan plots. It can also be
called by users. Note that it does not generate the histogram
or volcano plot - this is done by <code><a href="#topic+EWAS_QC">EWAS_QC</a></code>
itself.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  EWAS_plots(dataset,
             plot_QQ = TRUE,
             plot_Man = TRUE,
             plot_cutoff_p = 0.05,
             plot_QQ_bands = FALSE,
             high_quality_plots = FALSE,
             save_name = "dataset",
             header_translations)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EWAS_plots_+3A_dataset">dataset</code></td>
<td>

<p>either a vector of p-values, or a data frame containing the
columns <code>CHR</code> (chromosome number), <code>MAPINFO</code>
(base-pair position), and <code>P_VAL</code> (p-value). If
different columnnames are used, the
<code>header_translations</code> argument can be used to translate
these. <code>CHR</code> and <code>MAPINFO</code> are only required for
generating a Manhattan plot. Note that, unlike
<code>EWAS_QC</code>, this function does not accept filenames,
only data frames or vectors.
</p>
</td></tr>
<tr><td><code id="EWAS_plots_+3A_plot_qq">plot_QQ</code>, <code id="EWAS_plots_+3A_plot_man">plot_Man</code></td>
<td>

<p>logicals determining whether a QQ and Manhattan plot are
made.
</p>
</td></tr>
<tr><td><code id="EWAS_plots_+3A_plot_cutoff_p">plot_cutoff_p</code></td>
<td>

<p>numeric: the threshold of p-values to be shown in the QQ and
Manhattan plots. Higher (less significant) p-values are
excluded from the plot. The default setting is 0.05, which
excludes 95% of data-points. It's <em>NOT</em> recommended to
increase the value above 0.05, as this may dramatically
increase running time and memory usage.
</p>
</td></tr>
<tr><td><code id="EWAS_plots_+3A_plot_qq_bands">plot_QQ_bands</code></td>
<td>

<p>logical, if <code>TRUE</code>, probability bands are added to the
QQ plot.
</p>
</td></tr>
<tr><td><code id="EWAS_plots_+3A_high_quality_plots">high_quality_plots</code></td>
<td>

<p>logical. Setting this to TRUE will save the graphs as 
high-resolution tiff images.
</p>
</td></tr>
<tr><td><code id="EWAS_plots_+3A_save_name">save_name</code></td>
<td>

<p>character string, the name used for the plot files (do not
add an extension: <code>EWAS_plots</code> will do this
automatically).
</p>
</td></tr>
<tr><td><code id="EWAS_plots_+3A_header_translations">header_translations</code></td>
<td>

<p>a table that translates the column names of <code>dataset</code>
to the standard names. See <code><a href="#topic+translate_header">translate_header</a></code>
for details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>EWAS_plots</code> is a fairly straightforward function. It
accepts a data table or a vector of p-values, and generates
QQ and (when chromosome and position data are included)
Manhattan plots from these.
</p>


<h3>Value</h3>

<p><code>EWAS_plots</code>' most important output are the two graphs.
However, it also returns a single, invisible, numeric value,
representing the lambda calculated over the p-values.
</p>

<hr>
<h2 id='EWAS_QC'>Automated Quality Control of EWAS results files</h2><span id='topic+EWAS_QC'></span>

<h3>Description</h3>

<p>The main function of the <a href="#topic+QCEWAS">QCEWAS</a> package.
<code>EWAS_QC</code> accepts a single EWAS results file and runs a
thorough quality check (QC),
optionally applies various filters and generates QQ, Volcano
and Manhattan plots. The function <code><a href="#topic+EWAS_series">EWAS_series</a></code>
can be used to process multiple results files sequentially.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  EWAS_QC(data,
          map,
          outputname,
          header_translations,
          threshold_outliers = c(NA, NA),
          markers_to_exclude,
          exclude_outliers = FALSE,
          exclude_X = FALSE, exclude_Y = FALSE,
          save_final_dataset = TRUE, gzip_final_dataset = TRUE,
          header_final_dataset = "standard",
          high_quality_plots = FALSE,
          return_beta = FALSE, N_return_beta = 500000L,
          ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EWAS_QC_+3A_data">data</code></td>
<td>

<p>a data frame with EWAS results, or the name of a file
containing the same. The table must include the columns
<code>PROBEID</code>, <code>BETA</code>, <code>SE</code>, and <code>P_VAL</code>.
Other columns may be included but will be ignored. If the
column names differ from the above, the argument
<code>header_translations</code> can be used to translate them.
If a filename is entered in this argument, it
will be imported via the <code><a href="utils.html#topic+read.table">read.table</a></code> function.
<code><a href="utils.html#topic+read.table">read.table</a></code> can handle a variety of formats,
including files compressed in the .gz format. <code>EWAS_QC</code>
will pass any named, unknown arguments to
<code><a href="utils.html#topic+read.table">read.table</a></code>, so you can specify the column
separator and NA string with the usual
<code><a href="utils.html#topic+read.table">read.table</a></code> arguments. (Note that this only
applied to importing the EWAS results, and not the map or
translation files.)
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_map">map</code></td>
<td>

<p>a data frame with chromosome and position values of the
probes, or the name of a file containing the same. This
argument is optional: if no map is specified,
<code>EWAS_QC</code> will skip the Manhattan plot and chromosome
filters. <code>map</code>	must include the columns <code>TARGETID</code>,
<code>CHR</code> (chromosome), and <code>MAPINFO</code> (position), using
those exact names. Other columns may be included but will be
ignored. If a filename is entered in this argument, it
will be imported via the <code><a href="utils.html#topic+read.table">read.table</a></code> function.
<code><a href="utils.html#topic+read.table">read.table</a></code> can handle a variety of formats,
including files compressed in the .gz format.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_outputname">outputname</code></td>
<td>

<p>a character string specifying the intended filename for the
output. This includes not only the cleaned results file and
the log, but also any graphs created. Do not include an
extension; <code>EWAS_QC</code> adds these automatically.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_header_translations">header_translations</code></td>
<td>

<p>a translation table for the column names of the input file,
or the name of a file containing the same. This argument is
optional: if not specified, <code>EWAS_QC</code> assumes the
default column names are used. See
<code><a href="#topic+translate_header">translate_header</a></code> for information on the
format.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_threshold_outliers">threshold_outliers</code></td>
<td>

<p>a numeric string of length two. This defines which effect
sizes will be treated as outliers. The first value specifies
the lower limit (i.e. markers with effect sizes below this
value are considered outliers), the second the upper limit.
The check for low or high outliers is skipped if the
respective value is set to <code>NA</code>. To skip the check
entirely, set this argument to <code>c(NA, NA)</code>.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_markers_to_exclude">markers_to_exclude</code></td>
<td>

<p>Either a vector or data frame containing a list of CpG IDs
that need to be excluded before starting the QC (in case of
a data frame only the first column will be processed), or
the name of a file containing the same. This argument is
optional: if not specified, no exclusions are made. Note
that when a single value (a vector of length 1) is
passed to this argument, <code>EWAS_QC</code> will treat it as a
filename even when no such file can be found. If you want
to remove a single CpG, either pass it to this argument
via a file, or add a dummy value to the vector to give it
length 2 (e.g. <code>c("cg02198983", "dummy")</code> ).
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_exclude_outliers">exclude_outliers</code></td>
<td>

<p>a logical value determining how outliers are treated. If
<code>TRUE</code>, they are excluded from the final dataset. If
<code>FALSE</code>, they are merely counted.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_exclude_x">exclude_X</code>, <code id="EWAS_QC_+3A_exclude_y">exclude_Y</code></td>
<td>

<p>logical values determining whether markers at the X and Y
chromosome respectively are excluded from the final dataset.
This requires providing a map to <code>EWAS_QC</code> via the
<code>map</code> argument.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_save_final_dataset">save_final_dataset</code></td>
<td>

<p>logical determining whether the cleaned dataset will be
saved.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_gzip_final_dataset">gzip_final_dataset</code></td>
<td>

<p>logical determining whether the saved dataset will be
compressed in the .gz format.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_header_final_dataset">header_final_dataset</code></td>
<td>

<p>either a character vector or a table determining the header
names used in the final dataset, or the name of a file
containing the same. If <code>"original"</code>, the
final dataset will use the same column names as the original
input file. If <code>"standard"</code>, it will use the default
<code>EWAS_QC</code> column names. If a table, it will be passed
to <code><a href="#topic+translate_header">translate_header</a></code> to convert the column
names. If a table, the default column names (<code>PROBEID</code>,
<code>BETA</code>, <code>SE</code>, and <code>P_VAL</code>) must be in the
second column, and the desired column names in the first.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_high_quality_plots">high_quality_plots</code></td>
<td>

<p>logical. Setting this to TRUE will save the graphs as 
high-resolution tiff images.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_return_beta">return_beta</code>, <code id="EWAS_QC_+3A_n_return_beta">N_return_beta</code></td>
<td>

<p>arguments used by <code><a href="#topic+EWAS_series">EWAS_series</a></code>. These are not
important for users and can be ignored. For the sake of
completeness: <code>return_beta</code> is a logical value; if
<code>TRUE</code>, the function return value includes a vector of
effect sizes. <code>N_return_beta</code> defines the length of the
vector.
</p>
</td></tr>
<tr><td><code id="EWAS_QC_+3A_...">...</code></td>
<td>

<p>arguments passed to <code><a href="utils.html#topic+read.table">read.table</a></code> for importing
the EWAS results file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>QCEWAS</code> includes a Quick-Start guide in the doc
folder of the library. This guide will explain how to
run a QC and how to interpret the results.
The start-up message when loading
<code>QCEWAS</code> will indicate where it can be found on your
computer. In brief, the QC consists of the following 5 stages:
</p>

<ul>
<li><p> Checking data integrity:
</p>
<p>The values inside the EWAS results are tested for validity.
If impossible p-values, effect-sizes, etc. are encountered,
<code>EWAS_QC</code> generates a warning in the R console and sets
them to <code>NA</code>.
</p>
</li>
<li><p> Filter for outliers and sex-chromosomes (optional)
</p>
<p>Counts the number of outlying markers, as well as chromosome
X and Y markers, and deletes them if specified. The markers
named in <code>markers_to_exclude</code> are removed here as well.
</p>
</li>
<li><p> Generating QC plots
</p>
<p>A histogram of beta and standard error distribution is
plotted.
</p>
<p>The p-values are checked by correlating and plotting them
against p-values calculated from the effect size and
standard error.
</p>
<p>A QQ plot is generated to test for over/undersignificance.
</p>
<p>A Manhattan plot is generated to see where the signals (if
any) are located.
</p>
<p>A Volcano plot is generated to check the distribution of
effect sizes vs. p values.
</p>
</li>
<li><p> Creating a QC log
</p>
<p>The log contains notes about any problems encountered during
the QC, as well as several tables describing the data.
</p>
</li>
<li><p> Saving the cleaned dataset (optional)
</p>
</li></ul>



<h3>Value</h3>

<p>The main output of <code>EWAS_QC</code> are the cleaned results
file, log file and QC graphs. However, the function also
returns a list with 9 elements:
</p>
<table>
<tr><td><code>data_input</code></td>
<td>
<p>the file name of the input file, if loaded
from a file. If not, this will be an empty character string.</p>
</td></tr>
<tr><td><code>file</code></td>
<td>
<p>the filename of the cleaned results file.</p>
</td></tr>
<tr><td><code>QC_success</code></td>
<td>
<p>logical, indicates whether <code>EWAS_QC</code> 
was able to run a full QC on the file. Note that a <code>TRUE</code>
value does not mean that no problems where encountered,
merely that the full QC was executed.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the lambda value of reported p-values in the
cleaned dataset.</p>
</td></tr>
<tr><td><code>p_cor</code></td>
<td>
<p>the correlation between reported and expected
(based on effect size and standard error) p values.</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>a named integer vector reporting how many markers
were in the original dataset, how many had missing values,
how many were on chromosomes X and Y, how many were outliers,
how many were removed and how many are in the final, cleaned
dataset. Has no relation to the <code>N</code> argument of
<code><a href="#topic+EWAS_series">EWAS_series</a></code>.</p>
</td></tr>
<tr><td><code>SE_median</code></td>
<td>
<p>a numeric value: the median of the standard
errors in the cleaned dataset.</p>
</td></tr>
<tr><td><code>mean_methylation</code></td>
<td>
<p>a <code>NULL</code>: this functionality
has not been implemented yet.</p>
</td></tr>
<tr><td><code>effect_size</code></td>
<td>
<p>if <code>return_beta</code> is <code>TRUE</code>, this
is a numeric vector of length <code>N_return_beta</code>,
containing a random selection of effect sizes from the
filtered dataset. If <code>FALSE</code>, this will be <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The function will return a warning if it encounters p-values
&lt; 1e-300, as this is close to the smallest number that R can
process correctly. Various functions in the <code>QCEWAS</code>
package will set these values to 1e-300 to ensure proper
handling.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+EWAS_series">EWAS_series</a></code> for running a QC over multiple
files.
</p>
<p>See <code><a href="#topic+EWAS_plots">EWAS_plots</a></code> and <code><a href="#topic+P_correlation">P_correlation</a></code>
for carrying out specific steps of the QC.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># For use in this example, the 2 sample files in the
# extdata folder of the QCEWAS library will be copied
# to your current R working directory. Running the QC
# generates 7 new files in your working directory:
# a cleaned, post-QC dataset, a log file, and 5 graphs.
# Consult the Quick-Start guide for more information on
# how to interpret these.
## Not run: 
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "sample_map.txt.gz"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "sample1.txt.gz"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)

QC_results &lt;- EWAS_QC(data = "sample1.txt.gz",
                      map = "sample_map.txt.gz",
                      outputname = "sample_output",
                      threshold_outliers = c(-20, 20),
                      exclude_outliers = FALSE,
                      exclude_X = TRUE, exclude_Y = FALSE,
                      save_final_dataset = TRUE, gzip_final_dataset = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='EWAS_series'>
Quality Control and Comparison of multiple EWAS results files
</h2><span id='topic+EWAS_series'></span>

<h3>Description</h3>

<p>This function runs a QC (via the function <code><a href="#topic+EWAS_QC">EWAS_QC</a></code>)
over multiple files and generates additional graphs to
comparing the results of these files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EWAS_series(EWAS_files,
            output_files,
            map,
            N,
            header_translations,
            save_final_dataset = TRUE,
            gzip_final_dataset = TRUE,
            high_quality_plots = FALSE,
            N_plot_beta = 500000L,
            ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EWAS_series_+3A_ewas_files">EWAS_files</code></td>
<td>

<p>a character vector containing the filenames of the EWAS
results to be QC'ed.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_output_files">output_files</code></td>
<td>

<p>a character vector containing the filenames of the output
files. Do not add an extension; <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> does
so automatically.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_map">map</code></td>
<td>

<p>a data frame with chromosome and position values of the CpGs
in <code>data</code>, or the name of a file containing the same.
See <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> for details. This argument is
optional: if not specified, <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> will not
generate a Manhattan plot and no filter for X and Y markers
can be performed.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_n">N</code></td>
<td>

<p>a data frame containing the filenames (as listed in the
<code>EWAS_files</code> argument) and sample sizes of the
datasets, or the name of a file containing the same. The
data frame must contain the columns <code>file</code> and
<code>N</code>, with those exact names. All files listed in
the <code>EWAS_files</code> argument must be included in the 
<code>file</code> column. This argument is optional: if not
specified, <code>EWAS_series</code> will not generate a precision
plot.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_header_translations">header_translations</code></td>
<td>

<p>a translation table for the column names of the EWAS files,
or the name of a file containing the same. See
<code><a href="#topic+translate_header">translate_header</a></code> for details.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_save_final_dataset">save_final_dataset</code>, <code id="EWAS_series_+3A_gzip_final_dataset">gzip_final_dataset</code>, <code id="EWAS_series_+3A_high_quality_plots">high_quality_plots</code></td>
<td>

<p>logical values. See <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> for details.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_n_plot_beta">N_plot_beta</code></td>
<td>

<p>integer specifying how many beta values per file should
be used in the effect-size comparison plot. Set this to
a value larger than the number of markers in the datasets
to include all markers.
</p>
</td></tr>
<tr><td><code id="EWAS_series_+3A_...">...</code></td>
<td>

<p>arguments passed to <code><a href="#topic+EWAS_QC">EWAS_QC</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>QCEWAS</code> includes a Quick-Start guide in the doc
folder of the library. This guide will explain how to
run a QC and how to interpret the results.
The start-up message when loading
<code>QCEWAS</code> will indicate where it can be found on your
computer. In brief,
<code>EWAS_series</code> works by calling <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> for
every filename given in <code>EWAS_files</code>. After all files
have been processed, it will generate two additional graphs: a
precision plot (provided <code>N</code> was specified) and a
beta-distribution plot. The former shows the distribution of
precision (1 / median standard error) against the
square root of the sample size of the results file. Normally,
one expects to see a roughly positive correlation (i.e. the
cohorts ought to cluster around the linear diagonal from the
lower left to the upper right). The presence of outliers means
that the outlying cohort(s) have a far higher/lower
uncertainty in their estimates that can be expected from their
sample size. This could indicate a different method, a
different measure (check the effect-size distribution plot) or
possibly over- or undersignificance of their estimates (check
the QQ plot and lambda value).
</p>
<p>The effect-size distribution plot allows comparison of the
effect-size scale of different files. One  expects the
distribution to become somewhat narrower as sample size
increases. However, large differences in scale suggest that
the files used different units for their measurements.
</p>
<p>As of version 1.2-0, the effect-size distribution plot shows
a random (rather than proportional) selection of effect-sizes
from the cohort. As a consequence, rerunning QC over a
dataset may result in a slightly different distribution plot
in each run. This is only a cosmetic issue (as the default
sample size is sufficiently large to include the majority of a
normally-sized EWAS dataset) and can be averted entirely by
changing the <code>N_plot_beta</code> argument to a value exceeding
the number of markers in the dataset(s).
</p>
<p>Both plots use numbers rather than names to identify files.
The full filenames and corresponding numbers are listed in the
EWAS_QC_legend.txt file that is generated after <code>EWAS_series</code>
completes.
</p>


<h3>Value</h3>

<p>The main output of <code>EWAS_series</code> are the cleaned results
files, logs and graphs. The function also returns an invisible
data frame (also saved as <code>EWAS_QC_legend.txt</code>), listing
the input file names, file numbers,
whether they passed a complete QC (note that this merely
indicates that the QC was completed, not that there were no
problems), the standard error and, if specified, the sample
size.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EWAS_QC">EWAS_QC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># For use in this example, the 4 sample files in the
# extdata folder of the QCEWAS library will be copied
# to your current R working directory. Running the QC
# generates several files in your working directory:
# consult the Quick-Start Guide for more information
# on how to interpret these.
## Not run: 
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "sample_map.txt.gz"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "sample1.txt.gz"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "sample2.txt.gz"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "translation_table.txt"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)

sample_list &lt;- c("sample1.txt.gz", "sample2.txt.gz")
sample_N &lt;- data.frame(file = sample_list,
                       N = c(77, 79),
                       stringsAsFactors = FALSE)
                       


QC_results &lt;- EWAS_series(EWAS_files = sample_list,
                          output_files = c("sample_output1", "sample_output2"),
                          map = "sample_map.txt.gz",
                          N = sample_N,
                          header_translations = "translation_table.txt",
                          save_final_dataset = FALSE,
                          threshold_outliers = c(-20, 20),
                          exclude_outliers = FALSE,
                          exclude_X = TRUE, exclude_Y = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='P_correlation'>Testing P-value distribution</h2><span id='topic+P_correlation'></span>

<h3>Description</h3>

<p>A sub-function of <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> that
tests if the reported p-values match the p-value that can be
derived from the effect-size and standard error values.
Aberrations between these indicate that the p-values have been
adjusted, or that there is some other problem with the data.
It also creates a plot of reported vs. expected p-values that
shows the correlation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  P_correlation(dataset,
                plot_correlation = TRUE,
                plot_if_threshold = FALSE,
                threshold_r = 0.99,
                high_quality_plots = FALSE,
                save_name = "dataset",
                header_translations, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="P_correlation_+3A_dataset">dataset</code></td>
<td>

<p>a data frame with the columns <code>BETA</code> (effect size),
<code>SE</code> (standard error), and <code>P_VAL</code> (p value). If
the column names differ from the above, the argument
header_translations can be used to translate them.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_plot_correlation">plot_correlation</code></td>
<td>

<p>logical, determines whether a graph is made of reported vs.
expected p values.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_plot_if_threshold">plot_if_threshold</code></td>
<td>

<p>logical. If <code>TRUE</code>, the plot is only generated if the
p-value correlation is below the specified threshold.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_threshold_r">threshold_r</code></td>
<td>

<p>numeric. If the p-value correlation is below this, a warning
is generated.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_high_quality_plots">high_quality_plots</code></td>
<td>

<p>logical. Setting this to TRUE will save the graph as a 
high-resolution tiff image.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_save_name">save_name</code></td>
<td>

<p>character string used for the output file. Do not add an
extension; <code>P_correlation</code> will do so automatically.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_header_translations">header_translations</code></td>
<td>

<p>a translation table for the header of <code>dataset</code>. See
<code><a href="#topic+translate_header">translate_header</a></code> for details.
</p>
</td></tr>
<tr><td><code id="P_correlation_+3A_...">...</code></td>
<td>

<p>arguments passed to the generic <code><a href="graphics.html#topic+plot">plot</a></code> function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>P_correlation</code> is primarly a subfunction of
<code><a href="#topic+EWAS_QC">EWAS_QC</a></code>, but it can be used separately.
</p>


<h3>Value</h3>

<p><code>P_correlation</code> returns a single numeric value,
representing the correlation between reported and expected
p-values.
</p>

<hr>
<h2 id='P_lambda'>Calculation of the Lambda value</h2><span id='topic+P_lambda'></span>

<h3>Description</h3>

<p>The Lambda value represents the inflation of p-values
compared to a normal distribution of p.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>P_lambda(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="P_lambda_+3A_p">p</code></td>
<td>
<p>a numeric vector of p-values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function removes any missing values from <code>p</code>, and then
returns:
</p>
<p><code>median(qchisq(p, df=1, lower.tail=FALSE)) / qchisq(0.5, 1)</code>
</p>
<p>The lambda value represents the inflation of the p-values
compared to a normal distribution. In a
genome-wide study, one would expect the results for the
vast majority of CpG
sites to accord with the null hypothesis, i.e. the p-values
are random, and have a normal
distribution. Only sites that are significantly associated
with the phenotype of interest should lie outside of the
normal distribution.
</p>
<p>Ideally the lambda value should be 1. Lambda represents the
<em>overall</em> difference with the expected distribution
- so the presence of a few significant results (i.e. p-values
that do not follow the normal distribution) does not bias it.
</p>
<p>However, if lambda is 2 or higher, it means that a
substantial portion of your dataset is more significant than
expected for a genome-wide study (i.e.
oversignificance). This could mean your dataset has been
filtered for low-significance markers. If this is not
the case, you should consider doing a genomic control
correction on the p-values, to correct the oversignificance.
</p>
<p>Similary,
values of 0.8 or lower indicate that your results are less
significant than would be expected from a random distribution
of p-values.
</p>


<h3>Value</h3>

<p>A single numeric value, the lambda value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  pvector &lt;- ppoints(10000)
  P_lambda(pvector)
  # The lambda of a random distribution of p-values equals 1
  
  pvector[pvector &gt; 0.9 &amp; pvector &lt; 0.91] &lt;- NA
  P_lambda(pvector)
  # If low-significance results are removed (i.e. there are more
  # significant results than expected) lambda increases
</code></pre>

<hr>
<h2 id='QCEWAS-package'>Quality Control of Epigenome-Wide Association Study Results</h2><span id='topic+QCEWAS-package'></span><span id='topic+QCEWAS'></span>

<h3>Description</h3>

<p>Functions for automated and manual quality control of
Epigenome-Wide Association Study results.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
    Package: </td><td style="text-align: left;"> QCEWAS</td>
</tr>
<tr>
 <td style="text-align: left;">
    Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
    Version: </td><td style="text-align: left;"> 1.2-3</td>
</tr>
<tr>
 <td style="text-align: left;">
    Date: </td><td style="text-align: left;"> 2023-02-03</td>
</tr>
<tr>
 <td style="text-align: left;">
    License: </td><td style="text-align: left;"> GPL (&gt;= 3)</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>When combining the results of multiple Epigenome-Wide
Association Studies (EWAS) into a meta-analysis, it is
recommended to run a quality check (QC) over the individual
files first. This is not only to check if the individual
results are valid, reliable, and of high quality, but also to
ensure that they are comparable between cohorts. The
<code>QCEWAS</code> package was developed to automate this process.
</p>
<p>The core function of <code>QCEWAS</code> is <code><a href="#topic+EWAS_QC">EWAS_QC</a></code>.
This function accepts the filename of a single EWAS results
file and performs a thorough quality check. For more
information, see the <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> help page. The
function <code><a href="#topic+EWAS_series">EWAS_series</a></code> is a wrapper function that
accepts multiple file names, and then runs <code><a href="#topic+EWAS_QC">EWAS_QC</a></code>
for all of them. It also generates a few additional plots to
compare the results of the files, allowing the user to spot
differences in effect-size distribution or standard errors.
</p>
<p>The functions <code><a href="#topic+EWAS_plots">EWAS_plots</a></code> and
<code><a href="#topic+P_correlation">P_correlation</a></code> are subroutines of
<code><a href="#topic+EWAS_QC">EWAS_QC</a></code> that can also be called by the user to
perform specific QC tasks.
</p>
<p>QCEWAS also includes a Quick-Start Guide. A link to the
guide is provided when the package is loaded into R.
</p>


<h3>Author(s)</h3>

<p>P.J. van der Most, Leanne K. Kupers and Ilja M. Nolte
</p>
<p>Maintainer: P.J. van der Most &lt;p.j.van.der.most@umcg.nl&gt;
</p>

<hr>
<h2 id='translate_header'>Translate column names into standard names</h2><span id='topic+translate_header'></span>

<h3>Description</h3>

<p>This function is used to translate non-standard column names
into the standard ones used by <code><a href="#topic+EWAS_QC">EWAS_QC</a></code> and other
functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  translate_header(header,
                   standard = c("PROBEID","BETA","SE","P_VAL"),
                   alternative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translate_header_+3A_header">header</code></td>
<td>
<p>character vector; the header to be translated.</p>
</td></tr>
<tr><td><code id="translate_header_+3A_standard">standard</code></td>
<td>
<p>character vector; the names <code>header</code>
should be translated into.</p>
</td></tr>
<tr><td><code id="translate_header_+3A_alternative">alternative</code></td>
<td>
<p>translation table; see below for more information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes the entries in <code>standard</code> one by one, and
checks them against the translation table for alternatives.
It will report any missing standard headers, as well as
duplicate ones.
</p>


<h3>Value</h3>

<p><code>translate_header</code> returns an object of class 'list' with 6 components:
</p>
<table>
<tr><td><code>header_h</code></td>
<td>
<p>character vector; the translated header.
Unknown columns are included under their old names.</p>
</td></tr>
<tr><td><code>missing_h</code></td>
<td>
<p>character vector; the standard column names
that were not found. If none, this returns <code>NULL</code>.</p>
</td></tr>
<tr><td><code>unknown_h</code></td>
<td>
<p>character vector; column names that could not
be converted to a standard name. Note that these columns
are also included in <code>header_h</code>. If none, this returns
<code>NULL</code>.</p>
</td></tr>
<tr><td><code>header_N</code>, <code>missing_N</code>, <code>unknown_N</code></td>
<td>
<p>integer; the lengths of
the above three vectors</p>
</td></tr>
</table>


<h3>Translation Table</h3>

<p>The translation table must meet the following requirements:
</p>

<ul>
<li><p> 2 columns, with the default column names (i.e. the ones
in the standard argument) in the first column, and the
alternatives in the second.
</p>
</li>
<li><p> Multiple alternatives are allowed for a single standard
name, but every alternative name must be in a separate row.
</p>
</li>
<li><p> The alternatives must be capitalized.
</p>
</li>
<li><p> No duplicate alternatives are allowed.
</p>
</li>
<li><p> A header line is not required, and will be ignored if present.
</p>
</li></ul>



<h3>Note</h3>

<p>The function will automatically capitalize the elements of
the <code>header</code> argument (so the alternatives in the translation
table must also be capitalized). Also, elements that are not
in <code>standard</code> will not be translated, even if they are present
in the translation table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># For use in this example, the 2 sample files in the
# extdata folder of the QCEWAS library will be copied
# to your current R working directory
## Not run: 
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "sample2.txt.gz"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "QCEWAS"),
                           "translation_table.txt"),
          to = getwd(), overwrite = FALSE, recursive = FALSE)

sample_ewas &lt;- read.table("sample2.txt.gz", header = TRUE,
                          stringsAsFactors = FALSE, nrow = 10)
colnames(sample_ewas)


translation_table &lt;- read.table("translation_table.txt", header = TRUE,
                                stringsAsFactors = FALSE)
sample_translation &lt;- translate_header(header = colnames(sample_ewas),
                                       alternative = translation_table)
sample_translation                                         

colnames(sample_ewas) &lt;- sample_translation$header_h

colnames(sample_ewas)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
