<!DOCTYPE html><html><head><title>Help for package QCGWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QCGWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#QCGWAS-package'><p>Quality Control of Genome Wide Association Study Results</p></a></li>
<li><a href='#calc_kurtosis'><p>Skewness and Kurtosis</p></a></li>
<li><a href='#check_P'><p>Checking GWAS p-values</p></a></li>
<li><a href='#convert_impstatus'><p>Convert imputation-status values to the QCGWAS standard</p></a></li>
<li><a href='#create_hapmap_reference'><p>Create an allele-reference file from HapMap data</p></a></li>
<li><a href='#filter_GWAS'><p>Automated filtering and reformatting of GWAS results files</p></a></li>
<li><a href='#gwa_sample'>
<p>Sample dataset for the QCGWAS package</p></a></li>
<li><a href='#header_translations'>
<p>Translation table for GWAS dataset headers</p></a></li>
<li><a href='#HQ_filter'><p>Select high-quality data in GWAS datasets</p></a></li>
<li><a href='#identify_column'><p>Identify non-standard column names</p></a></li>
<li><a href='#intensity_plot'><p>Generates an intensity plot from x, y datasets</p></a></li>
<li><a href='#load_GWAS'><p>Easy loading of GWAS results files</p></a></li>
<li><a href='#match_alleles'><p>Check and correct alleles in GWAS result files</p></a></li>
<li><a href='#plot_distribution'><p>GWAS effect-Size distribution plot</p></a></li>
<li><a href='#plot_precision'><p>GWAS Precision Plot</p></a></li>
<li><a href='#plot_regional'><p>Regional Association Plot</p></a></li>
<li><a href='#plot_skewness'><p>GWAS Skewness vs. Kurtosis Plot</p></a></li>
<li><a href='#QC_GWAS'><p>Automated Quality Control of GWAS Results files</p></a></li>
<li><a href='#QC_histogram'><p>Histogram(s) of expected and observed data distribution</p></a></li>
<li><a href='#QC_plots'><p>QQ and Manhattan plots</p></a></li>
<li><a href='#QQ_plot'><p>QQ plot(s) of expected vs. reported p-values</p></a></li>
<li><a href='#save_log'><p>Log entries for QC_GWAS</p></a></li>
<li><a href='#switch_strand'><p>Convert alleles to the opposing DNA strand</p></a></li>
<li><a href='#translate_header'><p>Translate column names into standard names</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quality Control of Genome Wide Association Study Results</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-9</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter J. van der Most and Ilja M. Nolte</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peter J. van der Most &lt;p.j.van.der.most@umcg.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for (automated and manual) quality control of the results of Genome Wide Association Studies.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-30 14:37:59 UTC; Gebruilker</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-30 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='QCGWAS-package'>Quality Control of Genome Wide Association Study Results</h2><span id='topic+QCGWAS-package'></span><span id='topic+QCGWAS'></span>

<h3>Description</h3>

<p>Functions for automated and manual quality control of Genome
Wide Association Study results.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> QCGWAS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0-9</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-05-30</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 3)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The core of the package is the function <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.
This function carries out an automated quality-control (QC) of
a Genome Wide Association Study (GWAS) results file, reporting
on the data-distribution, checking the SNPs for missing and
invalid data, comparing the alleles and allele-frequency to a
reference, and creating QQ and Manhattan plots.
</p>
<p>Although the number of arguments in <code>QC_GWAS</code> may seem
overwhelming, only three of them are required to run a basic
QC. The name of the file to be QC'ed should be passed to the
<code>filename</code> argument; the directory of said file to the
<code>dir_data</code> argument; and a header-translation table to
the <code>header_translations</code> argument. The results
will be saved in a number of files and graphs in the data
directory. For a quick introduction to QCGWAS, read the quick
reference guide that can be found in &quot;R\library\QCGWAS\doc&quot;.
</p>


<h3>Author(s)</h3>

<p>Peter J. van der Most and Ilja M. Nolte
</p>
<p>Maintainer: Peter J. van der Most &lt;p.j.van.der.most@umcg.nl&gt;
</p>


<h3>References</h3>

<p>Van der Most, P.J., Vaez, A., Prins, B.P., Loretto Munoz, M.,
Snieder, H., Alizadeh, B.Z. &amp; Nolte, I.M. (2014).
QCGWAS: A flexible R package for automated quality control of
Genome-Wide Association results. Bioinformatics 30(8):1185-1186.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+QC_GWAS">QC_GWAS</a></code>
</p>

<hr>
<h2 id='calc_kurtosis'>Skewness and Kurtosis</h2><span id='topic+calc_kurtosis'></span><span id='topic+calc_skewness'></span>

<h3>Description</h3>

<p>Functions for calculating skewness and kurtosis</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_kurtosis(input,
              FRQ_val = NULL, HWE_val = NULL,
              cal_val = NULL, imp_val = NULL, ...)
calc_skewness(input,
              FRQ_val = NULL, HWE_val = NULL,
              cal_val = NULL, imp_val = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_kurtosis_+3A_input">input</code></td>
<td>
<p>either a vector of effect sizes or a data frame
using the standard column names.</p>
</td></tr>
<tr><td><code id="calc_kurtosis_+3A_frq_val">FRQ_val</code>, <code id="calc_kurtosis_+3A_hwe_val">HWE_val</code>, <code id="calc_kurtosis_+3A_cal_val">cal_val</code>, <code id="calc_kurtosis_+3A_imp_val">imp_val</code>, <code id="calc_kurtosis_+3A_...">...</code></td>
<td>
<p>arguments
passed to <code><a href="#topic+HQ_filter">HQ_filter</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Kurtosis is a measure of how well a distribution matches a
Gaussian distribution. A Gaussian distribution has a kurtosis
of <code>0</code>. Negative kurtosis indicates a flatter
distribution curve, while positive kurtosis indicates a
sharper, thinner curve.
</p>
<p>Skewness is a measure of distribution asymmetry. A symmetrical
distribution has skewness <code>0</code>. A positive skewness
indicates a long tail towards higher values, while a negative
skewness indicates a long tail towards lower values.
</p>
<p>Kurtosis is calculated as:
</p>
<p><code>sum( (ES - mean(ES))^4) / ((length(ES)-1) * sd(ES)^4 )</code>
</p>
<p>Skewness is calculated as:
</p>
<p><code>sum( (ES - mean(ES))^3) / ((length(ES)-1) * sd(ES)^3 )</code>
</p>


<h3>Value</h3>

<p>Respectively the kurtosis and skewness of the input
effect-size distribution.
</p>


<h3>Note</h3>

<p>Both functions accept vectors as <code>input</code>. If <code>input</code>
is a data frame, the column names must match the standard
names used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> (<code>"EFFECT"</code> for
effect sizes, <code>"EFF_ALL_FREQ"</code> for allele frequency, etc.)
</p>


<h3>See Also</h3>

<p>For plotting skewness and kurtosis:
<code><a href="#topic+plot_skewness">plot_skewness</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data("gwa_sample")  
  
  calc_kurtosis(gwa_sample$EFFECT)
  calc_kurtosis(gwa_sample)
  calc_kurtosis(gwa_sample$EFF_ALL_FREQ)
  calc_kurtosis(gwa_sample,
                FRQ_val = 0.05, cal_val = 0.95,
                filter_NA = FALSE)
</code></pre>

<hr>
<h2 id='check_P'>Checking GWAS p-values</h2><span id='topic+check_P'></span>

<h3>Description</h3>

<p>A simple test to check if the reported p-values in a GWAS
results file match the other statistics. This function
calculates an expected p-value (from the effect size and
standard error) and then correlates it with the actual,
reported p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_P(dataset, HQ_subset,
        plot_correlation = FALSE, plot_if_threshold = FALSE,
        threshold_r = 0.99,
        save_name = "dataset", save_dir = getwd(),
        header_translations,
        use_log = FALSE, dataN = nrow(dataset), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_P_+3A_dataset">dataset</code></td>
<td>
<p>table with at least three
columns: p-value, effect size and standard error.</p>
</td></tr>
<tr><td><code id="check_P_+3A_hq_subset">HQ_subset</code></td>
<td>
<p>an <em>optional</em> logical or numeric vector
indicating the rows in <code>dataset</code> that contain high
quality SNPs.</p>
</td></tr>
<tr><td><code id="check_P_+3A_plot_correlation">plot_correlation</code></td>
<td>
<p>logical; should a scatterplot of
the reported vs. calculated p-values be made? If <code>TRUE</code>,
the plot is saved as a .png file.</p>
</td></tr>
<tr><td><code id="check_P_+3A_plot_if_threshold">plot_if_threshold</code></td>
<td>
<p>logical; if <code>TRUE</code>, the
scatterplot is only saved when the correlation between
reported and calculated p-values is lower than
<code>threshold_r</code>.</p>
</td></tr>
<tr><td><code id="check_P_+3A_threshold_r">threshold_r</code></td>
<td>
<p>numeric; the correlation threshold for the
scatterplot.</p>
</td></tr>
<tr><td><code id="check_P_+3A_save_name">save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the scatterplot.</p>
</td></tr>
<tr><td><code id="check_P_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
output files are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="check_P_+3A_header_translations">header_translations</code></td>
<td>
<p>translation table for column names
See <code><a href="#topic+translate_header">translate_header</a></code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column names used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="check_P_+3A_use_log">use_log</code>, <code id="check_P_+3A_datan">dataN</code></td>
<td>
<p>arguments used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>;
redundant when <code>check_P</code> is used separately.</p>
</td></tr>
<tr><td><code id="check_P_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>check_P</code> calculates the expected p-value by taking the
chi-square (1 degree of freedom) of the effect size divided by
the standard error squared.
</p>
<p>In a typical GWAS dataset, the expected and observed p-values
should correlate perfectly. If this isn't the case, the
problem either lies in a misidentified column, or the wrong
values were used when generating the dataset.
</p>


<h3>Value</h3>

<p>The correlation between expected and reported p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data("gwa_sample")

  selected_SNPs &lt;- HQ_filter(data = gwa_sample,
                             FRQ_val = 0.05,
                             cal_val = 0.95,
                             filter_NA = FALSE)
  # To calculate a correlation between predicted and actual p-values:
  check_P(gwa_sample, HQ_subset = selected_SNPs,
          plot_correlation = FALSE)
  
  # To plot the correlation:
  ## Not run: 
    check_P(gwa_sample, HQ_subset = selected_SNPs,
            plot_correlation = TRUE, plot_if_threshold = FALSE,
            save_name = "sample")
  
## End(Not run)
</code></pre>

<hr>
<h2 id='convert_impstatus'>Convert imputation-status values to the QCGWAS standard</h2><span id='topic+convert_impstatus'></span>

<h3>Description</h3>

<p>Converts imputation-status data to the standard values used
by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_impstatus(impstatus,
                  T_strings = c("1", "TRUE", "T"),
                  F_strings = c("0", "FALSE", "F"),
                  NA_strings = c(NA, "NA", ".", "-"),
                  use_log = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_impstatus_+3A_impstatus">impstatus</code></td>
<td>
<p>vector of imputation-status data.</p>
</td></tr>
<tr><td><code id="convert_impstatus_+3A_t_strings">T_strings</code></td>
<td>
<p>character-string(s) indicating imputed data.</p>
</td></tr>
<tr><td><code id="convert_impstatus_+3A_f_strings">F_strings</code></td>
<td>
<p>character-string(s) indicating genotyped
data.</p>
</td></tr>
<tr><td><code id="convert_impstatus_+3A_na_strings">NA_strings</code></td>
<td>
<p>character-string(s) indicating missing
data.</p>
</td></tr>
<tr><td><code id="convert_impstatus_+3A_use_log">use_log</code>, <code id="convert_impstatus_+3A_...">...</code></td>
<td>
<p>arguments used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>;
redundant when <code>convert_impstatus</code> is used separately.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to convert the imputation-status column
into the standard format, where <code>0</code> is genotyped and
<code>1</code> is imputed. Untranslated values (i.e. strings that do
not appear in any of the string arguments) will trigger a
warning message and are set to <code>NA</code>.
</p>


<h3>Value</h3>

<p>Numeric vector with values <code>0</code> for genotyped, <code>1</code>
for imputed and <code>NA</code> for unknown data.
</p>


<h3>Note</h3>

<p>The implementation of this function has changed. Previously,
only character data was translated using the string
arguments. Since version 1.0-4, the string arguments are used
for all data types, so the user can determine the conversion
of logical and numeric values as well.
</p>
<p>However, this does mean that values <code>1</code> and <code>0</code> are
no longer converted automatically. They must be specified in
the string arguments, or else the function will report them as
untranslated and converts them to <code>NA</code>. The same applies
to <code>TRUE</code>, <code>FALSE</code> and <code>NA</code>. Note the
difference between the character string <code>"NA"</code> and the
value <code>NA</code>.
</p>
<p>Finally, if the imputation-status column contains character
strings, the main QC function <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> requires
that all values are translated. If not, <code>QC_GWAS</code> will
abort the QC.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  status1 &lt;- c(0,0,0,1,1,1,2,NA)
  status2 &lt;- c(TRUE, FALSE, TRUE, FALSE, NA)
  status3 &lt;- c("imputed", "genotyped", "NA", NA)
  status4 &lt;- c(status1, status2, status3)
  
  ( outcome1 &lt;- convert_impstatus(status1,
                                  T_strings = 1,
                                  F_strings = 0,
                                  NA_strings = NA) )
  # status 1 contains an untranslated value "2", which is
  # converted to NA. To avoid the warning message:
  
  ( outcome1 &lt;- convert_impstatus(status1,
                                  T_strings = 1,
                                  F_strings = 0,
                                  NA_strings = c(NA, 2)) )
  
  
  ( outcome2 &lt;- convert_impstatus(status2,
                                  T_strings = TRUE,
                                  F_strings = FALSE,
                                  NA_strings = NA) )
  
  ( outcome3 &lt;- convert_impstatus(status3,
                                  T_strings = "imputed",
                                  F_strings = "genotyped",
                                  NA_strings = c("NA", NA)) )
  # Note that NA_strings includes both the character-string "NA"
  # and the value NA. Otherwise, one of the two would go 
  # "untranslated" and trigger a warning message.
  
  
  # And to check them all together
  ( outcome4 &lt;- convert_impstatus(status4,
                          T_strings = c(1, TRUE, "imputed"),
                          F_strings = c(0, FALSE, "genotyped"),
                          NA_strings = c("NA", NA, 2)) )
</code></pre>

<hr>
<h2 id='create_hapmap_reference'>Create an allele-reference file from HapMap data</h2><span id='topic+create_hapmap_reference'></span>

<h3>Description</h3>

<p>This function creates the standard allele reference file, as
used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> and <code><a href="#topic+match_alleles">match_alleles</a></code>,
from data publicly available at the website of the
international HapMap project (see 'References').
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_hapmap_reference(dir = getwd(),
   download_hapmap = FALSE, download_subset,
   hapmap_files = list.files(path = dir, pattern = "freqs_chr"),
   filename = "allele_reference_HapMap",
   save_txt = TRUE, save_rdata = !save_txt,
   return_reference = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_hapmap_reference_+3A_dir">dir</code></td>
<td>
<p>character string; the directory of
the input and output files. Note that R uses <em>forward</em>
slash (/) where Windows uses the backslash (\).</p>
</td></tr>
<tr><td><code id="create_hapmap_reference_+3A_download_hapmap">download_hapmap</code></td>
<td>
<p>logical; if <code>TRUE</code>, the required
allele-frequency files are downloaded from the HapMap website
into <code>dir</code>, and then turned into a reference. If
<code>FALSE</code>, the files specified in <code>hapmap_files</code> are
used.</p>
</td></tr>
<tr><td><code id="create_hapmap_reference_+3A_download_subset">download_subset</code></td>
<td>
<p>character-string; indicates the
population to download for creating the reference. Options
are: ASW, CEU, CHB, CHD, GIH, JPT, LWK, MEX, MKK, TSI, YRI.</p>
</td></tr>
<tr><td><code id="create_hapmap_reference_+3A_hapmap_files">hapmap_files</code></td>
<td>
<p>character vector of the filenames of
HapMap frequency-files to be included in the reference. The
default option includes all files with the string
&quot;freqs_chr&quot; in the filename. (This argument is only
used when <code>download_hapmap</code> is <code>FALSE</code>.)</p>
</td></tr>
<tr><td><code id="create_hapmap_reference_+3A_filename">filename</code></td>
<td>
<p>character string; the name of the output file,
<em>without</em> file-extension.</p>
</td></tr>
<tr><td><code id="create_hapmap_reference_+3A_save_txt">save_txt</code>, <code id="create_hapmap_reference_+3A_save_rdata">save_rdata</code></td>
<td>
<p>logical; should the reference be
saved as a tab-delimitated text file and/or an RData file?
If saved as RData, the object name <code>allele_ref_std</code> is
used for the reference table.</p>
</td></tr>
<tr><td><code id="create_hapmap_reference_+3A_return_reference">return_reference</code></td>
<td>
<p>logical; should the function return
the reference as it output value?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function removes SNPs with invalid alleles and with allele
frequencies that do not add up to <code>1</code>. It also removes
all instances of duplicate SNPids. If such entries are
encountered, a warning is printed in the R console and the
entries are saved in a .txt file in the output directory.
</p>
<p>Like the <code>QC_GWAS</code>, <code>create_hapmap_reference</code> codes
the X chromosome as <code>23</code>, Y as <code>24</code>, XY (not
available on HapMap website) as <code>25</code> and M as <code>26</code>.
</p>
<p>Both the .RData export and  the function return store the
alleles as factors rather than character strings.
</p>


<h3>Value</h3>

<p>If <code>return_reference</code> is <code>TRUE</code>, the function
returns the generated reference table. If <code>FALSE</code>, it
returns an invisible <code>NULL</code>.
</p>


<h3>References</h3>

<p>The required data is available at the Website of the
International HapMap project, under bulk data downloads &gt;
bulk data &gt; frequencies
</p>
<p>http://hapmap.ncbi.nlm.nih.gov
</p>
<p>The HapMap files downloaded by this function are subject to
the HapMap terms and policies. See:
http://hapmap.ncbi.nlm.nih.gov/datareleasepolicy.html
</p>


<h3>See Also</h3>

<p><code><a href="#topic+match_alleles">match_alleles</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # This command will download the CEU HapMap dataset and use
  # it to generate an allele-reference. Create a folder
  # "new_hapmap" to store the data and make sure there is
  # sufficient disk space and a reasonably fast internet
  # connection.

  ## Not run: 
    new_hapmap &lt;- create_hapmap_reference(dir = "C:/new_hapmap",
                                download_hapmap = TRUE, download_subset = "CEU",
                                filename = "new_hapmap", save_txt = TRUE,
                                return_reference = TRUE)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='filter_GWAS'>Automated filtering and reformatting of GWAS results files</h2><span id='topic+filter_GWAS'></span>

<h3>Description</h3>

<p>This function was created as a convenient way to automate the
removal of low-quality and non-autosomal SNPs. It
includes the same formatting options as <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_GWAS(ini_file,
            GWAS_files, output_names,
            gzip_output = TRUE,
            dir_GWAS = getwd(), dir_output = dir_GWAS,
            FRQ_HQ = NULL, HWE_HQ = NULL,
            cal_HQ = NULL, imp_HQ = NULL,
            FRQ_NA = TRUE, HWE_NA = TRUE,
            cal_NA = TRUE, imp_NA = TRUE,
            ignore_impstatus = FALSE,
            remove_X = FALSE, remove_Y = FALSE,
            remove_XY = FALSE, remove_M = FALSE,
            header_translations,
            check_impstatus = FALSE,
            imputed_T = c("1", "TRUE", "yes", "YES", "y", "Y"),
            imputed_F = c("0", "FALSE", "no", "NO", "n", "N"),
            imputed_NA = NULL,
            column_separators = c("\t", " ", "", ",", ";"),
            header = TRUE, nrows = -1, nrows_test = 1000,
            comment.char = "", na.strings = c("NA", "."),
            out_header = "original", out_quote = FALSE,
            out_sep = "\t", out_eol = "\n", out_na = "NA",
            out_dec = ".", out_qmethod = "escape",
            out_rownames = FALSE, out_colnames = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_GWAS_+3A_ini_file">ini_file</code></td>
<td>

<p>(the filename of) a table listing the files to be
processed and the filters to be applied. See 'Details'.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_gwas_files">GWAS_files</code></td>
<td>
<p>character vector: when no <code>ini_file</code> is
provided, this identifies the files to be processed. See
'Details'.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_output_names">output_names</code></td>
<td>
<p>character vector: the filenames for the
output files. The default option is to use the input
filenames. Note that, unlike with other <code>QCGWAS</code>
functions, the file extensions should be included (However,
the function will automatically add <code>".gz"</code> when the
files are compressed.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_gzip_output">gzip_output</code></td>
<td>
<p>logical; should the output files be
compressed?</p>
</td></tr>    
<tr><td><code id="filter_GWAS_+3A_dir_gwas">dir_GWAS</code>, <code id="filter_GWAS_+3A_dir_output">dir_output</code></td>
<td>
<p>character-strings specifying the
directory address of the folders for the input files and the
output, respectively. Note that R uses <em>forward</em> slash
(/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_frq_hq">FRQ_HQ</code>, <code id="filter_GWAS_+3A_hwe_hq">HWE_HQ</code>, <code id="filter_GWAS_+3A_cal_hq">cal_HQ</code>, <code id="filter_GWAS_+3A_imp_hq">imp_HQ</code></td>
<td>

<p>Numeric vectors. When no <code>ini_file</code> is provided, these
arguments specify the filter threshold-values for allele
frequency, HWE p-value, callrate and imputation quality,
respectively. Passed to <code><a href="#topic+HQ_filter">HQ_filter</a></code>.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_frq_na">FRQ_NA</code>, <code id="filter_GWAS_+3A_hwe_na">HWE_NA</code>, <code id="filter_GWAS_+3A_cal_na">cal_NA</code>, <code id="filter_GWAS_+3A_imp_na">imp_NA</code></td>
<td>

<p>Logical vectors. When no <code>ini_file</code> is provided, these
arguments specify whether missing values (of allele frequency,
HWE p-value, callrates and imputation quality, respectively)
are excluded (<code>TRUE</code>) or ignored (<code>FALSE</code>).
Passed to <code><a href="#topic+HQ_filter">HQ_filter</a></code>.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_ignore_impstatus">ignore_impstatus</code></td>
<td>

<p>Logical vector. When no <code>ini_file</code> is provided, this
argument specifies whether imputation status is taken into
account when applying the filters. If <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_remove_x">remove_X</code>, <code id="filter_GWAS_+3A_remove_y">remove_Y</code>, <code id="filter_GWAS_+3A_remove_xy">remove_XY</code>, <code id="filter_GWAS_+3A_remove_m">remove_M</code></td>
<td>

<p>logical; respectively whether X-chromosome, Y-chromosome,
pseudo-autosomal and mitochondrial SNPs are removed. Note:
these arguments accept only a single <code>TRUE</code> or
<code>FALSE</code> value.
Unlike the above settings, it's not possible to specify
them independently for every dataset.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_header_translations">header_translations</code></td>
<td>
<p>translation table for column names.
See <code><a href="#topic+translate_header">translate_header</a></code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_check_impstatus">check_impstatus</code></td>
<td>
<p>logical; should
<code><a href="#topic+convert_impstatus">convert_impstatus</a></code> be called to convert the
imputation-status column into standard values?</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_imputed_t">imputed_T</code>, <code id="filter_GWAS_+3A_imputed_f">imputed_F</code>, <code id="filter_GWAS_+3A_imputed_na">imputed_NA</code></td>
<td>
<p>arguments passed to
<code><a href="#topic+convert_impstatus">convert_impstatus</a></code>.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_column_separators">column_separators</code></td>
<td>
<p>character string or vector; specifies
the values used as column delimitator in the GWAS file(s). The
argument is passed to <code><a href="#topic+load_test">load_test</a></code>; see the
description of that function for more information.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_nrows_test">nrows_test</code></td>
<td>
<p>integer; the number of rows used for
&quot;trial-loading&quot;. Before loading the entire dataset, the
function <code><a href="#topic+load_test">load_test</a></code> is called to determine the
dataset's file-format by reading the top <code>x</code> lines, where
<code>x</code> is <code>nrows_test</code>. Setting <code>nrows_test</code> to
a low number
(e.g. <code>150</code>) means quick testing, but runs the risk of
missing problems in lower rows. To test the entire dataset,
set it to <code>-1</code>.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_header">header</code>, <code id="filter_GWAS_+3A_nrows">nrows</code>, <code id="filter_GWAS_+3A_comment.char">comment.char</code>, <code id="filter_GWAS_+3A_na.strings">na.strings</code>, <code id="filter_GWAS_+3A_...">...</code></td>
<td>

<p>arguments passed to <code><a href="utils.html#topic+read.table">read.table</a></code> when importing
the dataset.</p>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_out_header">out_header</code></td>
<td>
<p>Translation table for the column names of
the <em>output</em> file. This argument is the opposite of
<code>header_translations</code>: it translates the standard
column-names of <code>QC_GWAS</code> to user-defined ones.
<code>output_header</code> can be one of three things:
</p>

<ul>
<li><p> A user specified table similar to the one used by
<code><a href="#topic+translate_header">translate_header</a></code>. However, as this
translates standard names into non-standard ones, the
standard names should be in the right column, and the
desired ones in the left. There is also no requirement
for the names in the <em>left</em> column to be uppercase.
</p>
</li>
<li><p> The name of a file in <code>dir_GWAS</code> containing
such a table.
</p>
</li>
<li><p> Character string specifying a standard form. See
<code><a href="#topic+QC_GWAS">QC_GWAS</a></code>, section 'Output header' for the
options.
</p>
</li></ul>
</td></tr>
<tr><td><code id="filter_GWAS_+3A_out_quote">out_quote</code>, <code id="filter_GWAS_+3A_out_sep">out_sep</code>, <code id="filter_GWAS_+3A_out_eol">out_eol</code>, <code id="filter_GWAS_+3A_out_na">out_na</code>, <code id="filter_GWAS_+3A_out_dec">out_dec</code>, <code id="filter_GWAS_+3A_out_qmethod">out_qmethod</code>, <code id="filter_GWAS_+3A_out_rownames">out_rownames</code>, <code id="filter_GWAS_+3A_out_colnames">out_colnames</code></td>
<td>

<p>arguments passed to
<code><a href="utils.html#topic+write.table">write.table</a></code> when saving the final dataset.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The easiest way to use <code>filter_GWAS</code> is by passing an ini
file to the <code>ini_file</code> argument.
The ini file can be generated by running <code><a href="#topic+QC_series">QC_series</a></code>
with the <code>save_filtersettings</code> argument set to <code>TRUE</code>.
The output will include a file 'Check_filtersettings.txt',
describing the (high-quality) filter settings used for each
file (taking into account whether there was enough data, i.e.
whether the <code>use_threshold</code> was met, to apply the filters).
</p>
<p>The <code>ini_file</code> argument accepts both a table
or the name of a file in <code>dir_GWAS</code> or the
current R working directory.
</p>
<p>If no <code>ini_file</code> is specified, the function will use the
<code>GWAS_files</code>, x_HQ, x_NA and <code>ignore_impstatus</code>
arguments to construct such a table.
<code>GWAS_files</code> can either be a character vector or a single
value. If a single string, all filenames containing the string
will be processed. The other arguments can also be a vector or
a single value; if the latter, they will be recycled to create
a vector of the correct length.  
</p>
<p>If neither <code>ini_file</code> nor <code>GWAS_files</code> are specified,
the function will look for a file
<code>Check_filtersettings.txt</code>
in <code>dir_GWAS</code> and the current R working directory.
</p>
<p>Note that <code>ini_file</code> overrules the other filter settings,
i.e. one cannot adjust <code>ini_file</code> through the other
arguments.
</p>


<h3>Value</h3>

<p>An invisible logical vector, indicating which files were
successfully filtered.</p>


<h3>Note</h3>

<p>R is not the optimal platform for filtering GWAS files. This
function was added at the request of a user, but an UNIX script
is likely to be faster.</p>

<hr>
<h2 id='gwa_sample'>
Sample dataset for the QCGWAS package
</h2><span id='topic+gwa_sample'></span>

<h3>Description</h3>

<p>A fake GWAS results dataset for use in the examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gwa_sample)</code></pre>


<h3>Format</h3>

<p>A data frame with 10000 observations on the following 15 variables.
</p>

<dl>
<dt><code>MARKER</code></dt><dd><p>a character vector; the SNP IDs.</p>
</dd>
<dt><code>STRAND</code></dt><dd><p>a character vector; the DNA strand of
the listed alleles.</p>
</dd>
<dt><code>CHR</code></dt><dd><p>a character vector; the chromosome of the
listed SNP.</p>
</dd>
<dt><code>POSITION</code></dt><dd><p>a numeric vector; the basepair
position of the listed SNP.</p>
</dd>
<dt><code>EFFECT_ALL</code></dt><dd><p>a character vector; the effect
allele.</p>
</dd>
<dt><code>OTHER_ALL</code></dt><dd><p>a character vector; the non-effect
allele.</p>
</dd>
<dt><code>N_TOTAL</code></dt><dd><p>a numeric vector; the sample size.</p>
</dd>
<dt><code>EFF_ALL_FREQ</code></dt><dd><p>a numeric vector; frequency of
the effect-allele in the dataset.</p>
</dd>
<dt><code>HWE_PVAL</code></dt><dd><p>a numeric vector; Hardy-Weinberg
(HWE) p-value of the listed alleles.</p>
</dd>
<dt><code>CALLRATE</code></dt><dd><p>a numeric vector; SNP callrate.</p>
</dd>
<dt><code>EFFECT</code></dt><dd><p>a numeric vector; effect size of the
listed effect allele.</p>
</dd>
<dt><code>STDERR</code></dt><dd><p>a numeric vector; standard error of the
effect.</p>
</dd>
<dt><code>PVALUE</code></dt><dd><p>a numeric vector; p-value of the
effect.</p>
</dd>
<dt><code>IMPUTED</code></dt><dd><p>a numeric vector; imputation status -
i.e. whether the SNP is genotyped (<code>0</code>) or imputed
(<code>1</code>).</p>
</dd>
<dt><code>IMP_QUALITY</code></dt><dd><p>a numeric vector; imputation
quality.</p>
</dd>
</dl>


<hr>
<h2 id='header_translations'>
Translation table for GWAS dataset headers
</h2><span id='topic+header_translations'></span>

<h3>Description</h3>

<p>This is a sample translation table, as used by
<code><a href="#topic+translate_header">translate_header</a></code> to translate dataset column
names to the <code>QCGWAS</code> standard. An editable .txt version
can be found in &quot;R\library\QCGWAS\doc&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(header_translations)</code></pre>


<h3>Format</h3>

<p>A data frame with 104 observations on the following 2 variables.
</p>

<dl>
<dt><code>STANDARD</code></dt><dd><p>a character vector; standard column
names as used by QCGWAS.</p>
</dd>
<dt><code>ALTERNATIVE</code></dt><dd><p>a character vector; alternative
column names, as used in the datasets.</p>
</dd>
</dl>


<hr>
<h2 id='HQ_filter'>Select high-quality data in GWAS datasets</h2><span id='topic+HQ_filter'></span>

<h3>Description</h3>

<p>This function accepts a <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> dataset and
returns a vector of logical values indicating which entries
meet the quality criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HQ_filter(data,
          ignore_impstatus = FALSE,
          FRQ_val = NULL, HWE_val = NULL,
          cal_val = NULL, imp_val = NULL,
          filter_NA = TRUE,
          FRQ_NA = filter_NA, HWE_NA = filter_NA,
          cal_NA = filter_NA, imp_NA = filter_NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HQ_filter_+3A_data">data</code></td>
<td>
<p>table to be filtered. <code>HQ_filter</code>
assumes the dataset uses the standard <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>
column names.</p>
</td></tr>
<tr><td><code id="HQ_filter_+3A_ignore_impstatus">ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td></tr>
<tr><td><code id="HQ_filter_+3A_frq_val">FRQ_val</code>, <code id="HQ_filter_+3A_hwe_val">HWE_val</code>, <code id="HQ_filter_+3A_cal_val">cal_val</code>, <code id="HQ_filter_+3A_imp_val">imp_val</code></td>
<td>

<p>numeric; the <em>minimal</em> required value for allele frequency,
HWE p-value, callrate and imputation quality respectively.
Note that the allele-frequency filter is two-sided: for a
filter-value of <code>x</code>, it will exclude entries with
<code>freq &lt; x</code> and <code>freq &gt; 1 - x</code>.</p>
</td></tr>
<tr><td><code id="HQ_filter_+3A_filter_na">filter_NA</code></td>
<td>

<p>logical; if <code>TRUE</code>, then missing filter variables
will be excluded; if <code>FALSE</code>, they will be ignored.
<code>filter_NA</code> is the default setting for all variables.
Variable-specific settings can be specified with the
following arguments.</p>
</td></tr>
<tr><td><code id="HQ_filter_+3A_frq_na">FRQ_NA</code>, <code id="HQ_filter_+3A_hwe_na">HWE_NA</code>, <code id="HQ_filter_+3A_cal_na">cal_NA</code>, <code id="HQ_filter_+3A_imp_na">imp_NA</code></td>
<td>
<p>logical; variable-specific 
settings for <code>filter_NA</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A SNP is considered high-quality if it meets all quality
criteria. The thresholds are inclusive; i.e. SNPs that have a
value equal or higher than the threshold will be considered
high-quality.
</p>
<p>To filter missing values only, set the filter argument to
<code>NA</code>, and the corresponding NA-filter to <code>TRUE</code>.
</p>
<p>To disable filtering entirely, set to <code>NULL</code>. This
disables the filtering of missing values as well.
</p>
<p>When imputation status is missing or invalid (and
<code>ignore_impstatus</code> is <code>FALSE</code>), only the
allele-frequency filter will be applied.
</p>


<h3>Value</h3>

<p>A vector of logical values, indicating which values in
<code>data</code> meet (<code>TRUE</code>) or fail (<code>FALSE</code>) the
quality criteria.
</p>


<h3>Note</h3>

<p>The table entered in the <code>data</code> argument must use the
standard column names of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>. Functions
using <code>HQ_filter</code> usually allow the user to specify a
translation table. If not, <code><a href="#topic+translate_header">translate_header</a></code> can
be used to translate the header manually.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data("gwa_sample")

  selected_SNPs &lt;- HQ_filter(data = gwa_sample,
                             FRQ_val = 0.01,
                             cal_val = 0.95,
                             filter_NA = FALSE)
  summary(gwa_sample[selected_SNPs, ])
  
  selected_SNPs &lt;- HQ_filter(data = gwa_sample,
                             FRQ_val = 0.01,
                             cal_val = 0.95,
                             filter_NA = FALSE,
                             ignore_impstatus = TRUE)
  summary(gwa_sample[selected_SNPs, ])
</code></pre>

<hr>
<h2 id='identify_column'>Identify non-standard column names</h2><span id='topic+identify_column'></span>

<h3>Description</h3>

<p>This function is a subroutine of <code><a href="#topic+translate_header">translate_header</a></code>
and several other functions. It is used to translate non-standard
column names into standard ones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>identify_column(std_name, alt_names, header)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="identify_column_+3A_std_name">std_name</code></td>
<td>
<p>character string; the standard name for
the data-column.</p>
</td></tr>
<tr><td><code id="identify_column_+3A_alt_names">alt_names</code></td>
<td>
<p>translation table; with in the left column
the standard name(s) and in the right column possible
alternatives. See <code><a href="#topic+translate_header">translate_header</a></code> for more
details.</p>
</td></tr>
<tr><td><code id="identify_column_+3A_header">header</code></td>
<td>
<p>the column names of the dataset.
The names should be entirely in uppercase.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The purpose of <code>identify_column</code> is essentially to look
up in the translation table (<code>alt_names</code>) which of the
names in <code>header</code> can be translated into <code>std_name</code>.
</p>


<h3>Value</h3>

<p>An integer vector of the entry(s) in <code>header</code> (i.e. the
column-numbers) that can be translated into <code>std_name</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+translate_header">translate_header</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  sample_data &lt;-
    data.frame(SNP = paste("rs", 1:10, sep = ""),
               chrom = 2,
               effect = 1:10/10,
               misc = NA)
  sample_header &lt;- toupper(names(sample_data))


  alt_headers &lt;-
    data.frame(
      standard = c("MARKER", "MARKER", "CHR", "CHR"),
      alternative = c("MARKER", "SNP", "CHR", "CHROM"),
      stringsAsFactors = FALSE)
  
  identify_column(std_name = "EFFECT", alt_names = alt_headers,
                  header = sample_header)
  identify_column(std_name = "MARKER", alt_names = alt_headers,
                  header = sample_header)
  identify_column(std_name = "CHR", alt_names = alt_headers,
                  header = sample_header)
  identify_column(std_name = "MISC", alt_names = alt_headers,
                  header = sample_header)
</code></pre>

<hr>
<h2 id='intensity_plot'>Generates an intensity plot from x, y datasets</h2><span id='topic+intensity_plot'></span>

<h3>Description</h3>

<p>This function is used by <code><a href="#topic+match_alleles">match_alleles</a></code> to
generate an intensity plot. This function is currently only
partially implemented, so we recommend that users do not
bother with it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intensity_plot(x, y, strata, nbin = 20,
               xmax = max(x), xmin = min(x),
               ymax = max(y), ymin = min(y),
               strata_colours = c("black", "red", "turquoise3"),
               verbose = TRUE, xlab = "x", ylab = "y", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intensity_plot_+3A_x">x</code>, <code id="intensity_plot_+3A_y">y</code></td>
<td>
<p>numerical vectors; the x and y coordiantes of the
datapoints.</p>
</td></tr>
<tr><td><code id="intensity_plot_+3A_strata">strata</code></td>
<td>
<p>logical vector; indicates whether the datapoint
belongs to strata 1 or 2. If missing, all datapoints are
assumed to belong to strata 1.</p>
</td></tr>
<tr><td><code id="intensity_plot_+3A_nbin">nbin</code></td>
<td>
<p>integer: the number of bins (categories) on the x
and y axis.</p>
</td></tr>
<tr><td><code id="intensity_plot_+3A_xmax">xmax</code>, <code id="intensity_plot_+3A_xmin">xmin</code>, <code id="intensity_plot_+3A_ymax">ymax</code>, <code id="intensity_plot_+3A_ymin">ymin</code></td>
<td>
<p>numeric; the range of x and y
values shown in the plot.</p>
</td></tr>
<tr><td><code id="intensity_plot_+3A_strata_colours">strata_colours</code></td>
<td>
<p>character vector of length <code>3</code>,
indicating the colours to be used for entries of strata 1, 2
or mixed, respectively.</p>
</td></tr>
<tr><td><code id="intensity_plot_+3A_verbose">verbose</code></td>
<td>
<p>logical; determines whether a warning is printed
in the console when datapoints are removed.</p>
</td></tr>
<tr><td><code id="intensity_plot_+3A_xlab">xlab</code>, <code id="intensity_plot_+3A_ylab">ylab</code>, <code id="intensity_plot_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is intended as an alternative to the standard
scatter plot for the allele-frequency correlation graph.
</p>


<h3>Value</h3>

<p>An invisible object of type <code>list</code> with the following two
components:
</p>
<table>
<tr><td><code>NA_removed</code></td>
<td>
<p>The number of entries removed due to missing
values</p>
</td></tr>
<tr><td><code>outliers_removed</code></td>
<td>
<p>The number of entries removed because
they exceeded the thresholds specified by the min/max
arguments.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+match_alleles">match_alleles</a></code></p>

<hr>
<h2 id='load_GWAS'>Easy loading of GWAS results files</h2><span id='topic+load_GWAS'></span><span id='topic+load_test'></span>

<h3>Description</h3>

<p><code>load_GWAS</code> is wrapper-function of <code><a href="utils.html#topic+read.table">read.table</a></code>
that makes loading large GWAS results files less of a hassle. It
automatically unpacks .zip and .gz files and uses
<code>load_test</code> to determine which column separator the file
uses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_GWAS(filename, dir = getwd(),
          column_separators = c("\t", " ", "", ",", ";"),
          test_nrows = 1000,
          header = TRUE, nrows = -1,
          comment.char = "", na.strings = c("NA", "."),
          stringsAsFactors = FALSE, ...)

load_test(filename, dir = getwd(),
          column_separators = c("\t", " ", "", ",", ";"),
          test_nrows = 1000, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_GWAS_+3A_filename">filename</code></td>
<td>
<p>character string; the complete filename of the
file to be loaded. Note that compressed files (.gz or .zip
files) can only be unpacked if the filename of the archive
contains the extension of the archived file. For example, if
the archived file is named <code>"data1.csv"</code>, the archive
should be <code>"data1.csv.zip"</code>.</p>
</td></tr>
<tr><td><code id="load_GWAS_+3A_dir">dir</code></td>
<td>
<p>character string; the directory containing the file.
Note that R uses <em>forward</em> slash (/) where Windows uses
backslash (\).</p>
</td></tr>
<tr><td><code id="load_GWAS_+3A_column_separators">column_separators</code></td>
<td>
<p>character string or vector of
the column-separators to be tried by <code>load_test</code>.
White-space can be specified by <code>""</code>, but it is
recommended you try tab (<code>"\t"</code>) and space
(<code>" "</code>) first.</p>
</td></tr>
<tr><td><code id="load_GWAS_+3A_test_nrows">test_nrows</code></td>
<td>
<p>integer; the number of lines that
<code>load_test</code> checks in the trail-load. A smaller number
means faster loading, but also makes it more likely that
errors slip through. To check the entire dataset, set to
<code>-1</code>.</p>
</td></tr>
<tr><td><code id="load_GWAS_+3A_header">header</code>, <code id="load_GWAS_+3A_nrows">nrows</code>, <code id="load_GWAS_+3A_comment.char">comment.char</code>, <code id="load_GWAS_+3A_na.strings">na.strings</code>, <code id="load_GWAS_+3A_stringsasfactors">stringsAsFactors</code>, <code id="load_GWAS_+3A_...">...</code></td>
<td>

<p>Arguments passed to <code><a href="utils.html#topic+read.table">read.table</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>load_test</code> determines the correct column separator
simply by trying them individually until it finds one that
works (that is: one that results in a dataset with an equal
number of cells in every row AND at least five or more
columns). If none work, it reports the error-message generated
by the last column separator tried.
</p>
<p>The column separators are tried in the order specified by the
<code>column_separators</code> argument.
</p>
<p>By default, <code>load_test</code> only checks the first 1000 lines
(adjustable by the <code>test_nrows</code> argument); if the problem
lies further down in the dataset, it will not catch it. In such
a case, <code>load_GWAS</code> and <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> will crash
when attempting to load the dataset.
</p>
<p>A common problem is employing white-space (<code>""</code>) as
column separator for a file that uses empty fields to indicate
missing values. The separators surrounding an empty field are
adjacent, so R parses them as a single column separator. In
this particular example, specifying a single space
(<code>" "</code>) or tab (<code>"\t"</code>) as column separator solves
the problem (this is why the default setting of
<code>column_separators</code> puts these values before white-space).
</p>


<h3>Value</h3>

<p><code>load_GWAS</code> returns the table imported from the specified file.
</p>
<p><code>load_test</code> returns a list with 4 components:
</p>
<table>
<tr><td><code>success</code></td>
<td>
<p>logical; whether <code>load_test</code> was able to
load a dataset with five or more columns.</p>
</td></tr>
<tr><td><code>error</code></td>
<td>
<p>character string; if unable to load the file, this
returns the error-message of the last column separator to be
tried.</p>
</td></tr>
<tr><td><code>file_type</code></td>
<td>
<p>character string; the last three characters
of <code>filename</code>.</p>
</td></tr>
<tr><td><code>sep</code></td>
<td>
<p>the first column-separator that succeeded in loading
a dataset with five or more columns.</p>
</td></tr>
</table>


<h3>Note</h3>

<p><code>load_GWAS</code> uses the same default loading-settings as
<code>QC_GWAS</code>. <code>load_test</code>, on the other hand, has no
default values for <code>header</code>, <code>comment.char</code>,
<code>na.strings</code> and <code>stringsAsFactors</code>, and uses the
<code><a href="utils.html#topic+read.table">read.table</a></code> defaults instead.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## As the function requires a GWAS file to work,
  ## the following code should be adjusted before execution.
  ## Because this is a demonstration, the nrows argument is used
  ## to read only the first 100 rows.
  
  ## Not run: 
   data_GWAS &lt;-
      load_GWAS("GWA_results1.txt.zip",
                dir = "C:/GWAS_results",
                nrows = 100)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='match_alleles'>Check and correct alleles in GWAS result files</h2><span id='topic+match_alleles'></span>

<h3>Description</h3>

<p>This function checks the reported alleles and allele frequencies
in GWAS results data by comparing them to a reference table.
It will also uniformize the dataset by switching all SNPs to
the positive strand and flipping the alleles so that the
effect allele matches the reference minor allele.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_alleles(dataset, ref_set, HQ_subset,
              dataname = "dataset", ref_name = "reference",
       unmatched_data = !all(dataset$MARKER %in% ref_set$SNP),
              check_strand = FALSE,
              save_mismatches = TRUE, delete_mismatches = FALSE,
              delete_diffEAF = FALSE, threshold_diffEAF = 0.15,
              check_FRQ = TRUE, check_ambiguous = FALSE,
              plot_FRQ = FALSE, plot_intensity = FALSE, 
              plot_if_threshold = FALSE,
              threshold_r = 0.95,
              return_SNPs = FALSE, return_ref_values = FALSE,
              header_translations, header_reference,
              save_name = dataname, save_dir = getwd(),
              use_log = FALSE, log_SNPall = nrow(dataset))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_alleles_+3A_dataset">dataset</code></td>
<td>

<p>table containing the allele data. <code>dataset</code>
should always contain columns for the SNPID, the effect
allele and the other allele. Strand and allele frequency
may be required, depending upon the settings, while
effect size is optional. <code>match_alleles</code> accepts
non-standard column names, provided a translation table is
specified in <code>header_translations</code>. The order of columns
does not matter; nor does the presence of other columns.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_ref_set">ref_set</code></td>
<td>

<p>table containing the reference data. <code>ref_set</code>
should always contain columns for the SNPID, the minor
allele and the major allele. Minor allele frequency is only
required if <code>check_FRQ</code> or <code>check_ambiguous</code> are
<code>TRUE</code>. The standard column-names are <code>"SNP"</code>,
<code>"MINOR"</code>, <code>"MAJOR"</code> and <code>"MAF"</code>. Non-standard
column names are accepted if a translation table is
specified in <code>header_reference</code>. All SNPs <em>must</em>
be aligned to the positive strand.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_hq_subset">HQ_subset</code></td>
<td>
<p>an <em>optional</em> logical or numeric vector
indicating the rows in <code>dataset</code> that contain high
quality SNPs.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_dataname">dataname</code>, <code id="match_alleles_+3A_ref_name">ref_name</code></td>
<td>
<p>character strings; the names of the
dataset and reference, respectively. Used as identifiers
in the output files.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_unmatched_data">unmatched_data</code></td>
<td>
<p>logical; are there SNPs in the dataset
that do not appear in the reference? This argument is
currently redundant: the function will determine it
automatically.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_check_strand">check_strand</code></td>
<td>
<p>logical; should the function check for
negative-strand SNPs? If <code>FALSE</code>, all SNPs are assumed
to be on the positive strand.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_save_mismatches">save_mismatches</code></td>
<td>
<p>logical; should mismatching entries be
exported to a .txt file before they are corrected?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_delete_mismatches">delete_mismatches</code></td>
<td>
<p>logical; should mismatching SNPs
(that could not be corrected by strand-switching) have
their effect allele set to missing?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_delete_diffeaf">delete_diffEAF</code></td>
<td>
<p>logical; should SNPs that exceed the
<code>threshold_diffEAF</code> have their effect allele set to
missing?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_threshold_diffeaf">threshold_diffEAF</code></td>
<td>
<p>numeric; the max. allowed difference
between reported and reference allele frequency.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_check_frq">check_FRQ</code></td>
<td>
<p>logical; should the function correlate
the reported allele-frequency with that of the reference?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_check_ambiguous">check_ambiguous</code></td>
<td>
<p>logical; should the function do separate
frequency correlations and create separate plots for SNPs
with a strand-independent allele-pair (i.e. an A/T or C/G
configuration)?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_plot_frq">plot_FRQ</code></td>
<td>
<p>logical; should a scatterplot of reported vs.
reference allele-frequency be made?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_plot_intensity">plot_intensity</code></td>
<td>
<p>logical; if <code>TRUE</code>, instead of a
scatterplot an intensity plot is generated. This option is
currently only partially implemented. Leave to <code>FALSE</code>
for now.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_plot_if_threshold">plot_if_threshold</code></td>
<td>
<p>logical; if <code>TRUE</code>, the
scatterplot is only made when frequency correlation is below
the threshold specified by <code>threshold_r</code>.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_threshold_r">threshold_r</code></td>
<td>
<p>numeric; the correlation threshold value.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_return_snps">return_SNPs</code></td>
<td>
<p>logical; should the return value include
the relevant columns of <code>dataset</code>?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_return_ref_values">return_ref_values</code></td>
<td>
<p>logical; should the return-value
include the matching entries in <code>ref_set</code>?</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_header_translations">header_translations</code>, <code id="match_alleles_+3A_header_reference">header_reference</code></td>
<td>

<p>translation tables for converting the column names of
<code>dataset</code> and <code>ref_set</code> to standard names,
respectively. See <code><a href="#topic+translate_header">translate_header</a></code> for more
information.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_save_name">save_name</code></td>
<td>

<p>character string; the filename, <em>without</em>
extension, for the various output files.</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
output files are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_use_log">use_log</code>, <code id="match_alleles_+3A_log_snpall">log_SNPall</code></td>
<td>

<p>arguments used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>;
redundant when <code>match_alleles</code> is used separately.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>match_alleles</code> is one of the more complicated functions
of <code>QCGWAS</code>. However, what it does is quite simple:
</p>

<ul>
<li><p> Check for incorrect allele pairs
</p>
</li>
<li><p> Uniformize the output so that all SNPs are on the
positive strand, and identical SNPs will have the same
effect allele and other allele in all datasets
</p>
</li>
<li><p> Check the allele frequency
</p>
</li></ul>

<p>The complexity stems from the fact that these three tasks have
to be carried out together and often overlap.
So the actual function schematic looks like this:
</p>

<ul>
<li><p> Switch negative-strand SNPs (i.e. SNPs with <code>"-"</code>
in the strand-column) to the positive strand. This step can be
disabled by setting <code>check-strand</code> to <code>FALSE</code>.
</p>
</li>
<li><p> Correct (if possible) mismatching alleles. A mismatch
is when the reported allele-pair does not match that in
the reference. <code>match_alleles</code> will attempt to fix
the mismatch by &quot;strand-switching&quot; the alleles. The
assumption is that dataset merely reported the wrong
strand, so converting them to the opposing strand should
solve the mismatch. If it does not, the SNPs are truly
mismatches. If <code>delete_mismatches</code> is <code>TRUE</code>,
the effect alleles are set to <code>NA</code>; if <code>FALSE</code>,
they are restored to their original configuration and
excluded from the allele-frequency test. If
<code>save_mismatches</code> is <code>TRUE</code>, the entries are
exported in a .txt before being changed. Note that
<code>save_mismatches</code> only exports true mismatches (i.e.
not those that were fixed after strand-switching).
</p>
</li>
<li><p> Align the allele-pairs with the reference. In order to
have the same effect allele with the same SNP in every
dataset, SNPs are &quot;flipped&quot; so that the effect allele
matches the reference minor allele. Flipped alleles will
also have their allele frequency and effect size inverted.
</p>
</li>
<li><p> Check for undetected strand-mismatch by counting the
number of ambiguous and (if <code>check_FRQ</code> is <code>TRUE</code>)
suspect SNPs. Ambiguous SNPs are SNPs with an allele-pair
that is identical on both strands (i.e. A/T or C/G).
Suspect SNPs are ambiguous SNPs whose allele-frequency
differs strongly from that of the reference.
</p>
</li>
<li><p> Check allele-frequencies by correlating and/or
plotting them against the reference. If
<code>check_ambiguous</code> is <code>TRUE</code>, additional
scatterplots will be made for the subsets of ambiguous and
non-ambiguous SNPs. If <code>delete_diffEAF</code> is <code>TRUE</code>,
SNPs whose allele-frequency differs from the reference by
more than <code>threshold_diffEAF</code> have their effect alleles
set to <code>NA</code> as well. This entire step can be disabled
by setting <code>check_FRQ</code> to <code>FALSE</code>.
</p>
</li></ul>
  


<h3>Value</h3>

<p>An object of class 'list' with the following components:
</p>
<table>
<tr><td><code>FRQ_cor</code>, <code>FRQ_cor_ambiguous</code>, <code>FRQ_cor_nonambi</code></td>
<td>

<p>Allele-frequency correlations for all, ambiguous, and
non-ambiguous SNPs respectively.</p>
</td></tr>
<tr><td><code>n_SNPs</code></td>
<td>
<p>Total number of SNPs in <code>dataset</code></p>
</td></tr>
<tr><td><code>n_missing</code>, <code>n_missing_data</code>, <code>n_missing_ref</code></td>
<td>

<p>Number of SNPs with missing allele-data in either
<code>dataset</code> or <code>reference</code>, dataset only, and
reference only, respectively.</p>
</td></tr>
<tr><td><code>n_negative_strand</code>, <code>n_negative_switch</code>, <code>n_negative_mismatch</code></td>
<td>

<p>Number of negative-strand SNPs, the subset of negative-strand
SNPs that were strand-switched twice because they did not match
the reference, and the subset of double-switched SNPs that
were still mismatching after the second strand-switch.</p>
</td></tr>
<tr><td><code>n_strandswitch</code>, <code>n_mismatch</code></td>
<td>

<p>Number of SNPs that was strand-switched because they did not
match the reference, and the subset of those that still did
not match after the strand-switch.</p>
</td></tr>
<tr><td><code>n_flipped</code></td>
<td>
<p>Number of SNPs whose alleles were flipped to
align them with the reference.</p>
</td></tr>
<tr><td><code>n_ambiguous</code>, <code>n_suspect</code></td>
<td>
<p>Number of ambiguous SNPs, and
the subset of those that had a large allele-frequency aberration.</p>
</td></tr>
<tr><td><code>n_diffEAF</code></td>
<td>
<p>Number of SNPs whose allele-frequency differs from
the reference by more than <code>threshold_diffEAF</code>.</p>
</td></tr>
<tr><td><code>MARKER</code></td>
<td>
<p>When <code>return_SNPs</code> and/or <code>return_ref_values</code>
is <code>TRUE</code>, this returns the column of <code>dataset</code>
containing the SNP IDs. If not, this returns <code>NULL</code>.</p>
</td></tr>
<tr><td><code>EFFECT_ALL</code>, <code>OTHER_ALL</code>, <code>STRAND</code>, <code>EFFECT</code>, <code>EFF_ALL_FREQ</code></td>
<td>

<p>If <code>return_SNPs</code> is <code>TRUE</code>, these elements return
the corrected columns of data-set. If <code>FALSE</code>, these
return <code>NULL</code>. Note: <code>match_alleles</code> only returns
those columns that were checked; if <code>check_FRQ</code> is
<code>FALSE</code>, <code>EFF_ALL_FREQ</code> return <code>NULL</code>. The
same goes for <code>check_strand</code> and <code>STAND</code>.
<code>EFFECT</code> is only returned if present in <code>dataset</code>.
</p>
</td></tr>
<tr><td><code>ref_MINOR</code>, <code>ref_MAJOR</code>, <code>ref_MAF</code></td>
<td>

<p>If <code>return_ref_values</code> is <code>TRUE</code>, these elements
return the reference minor and major alleles and allele
frequency column for the SNPs in <code>MARKER</code>.If
<code>FALSE</code>, these return <code>NULL</code>. <code>ref_MAF</code> is
only returned when <code>check_FRQ</code> is <code>TRUE</code>.
</p>
</td></tr>
</table>


<h3>Interpreting the output</h3>

<p>The output of <code>match_alleles</code> may seem a bit overwhelming
at first, so here is a short explanation of what it means and
what you should pay attention to.
</p>
<p>The columns included in the return value when <code>return_SNPs</code>
is <code>TRUE</code> are the  post-matching dataset. This is only
relevant if you want to  continue working with the corrected
dataset. Similarly, the output of <code>return_ref_values</code> is
only used for comparing the post-matching dataset to the
reference.
</p>
<p><code>n_missing</code>, <code>n_missing_data</code> and
<code>n_missing_ref</code> report the prevalence of missing allele
data, but are otherwise irrelevant.
</p>
<p>The majority of return values serve to check whether
strand-switching was performed correctly. <code>n_strandswitch</code>
indicates how many SNPs were converted to the other strand
because of a mismatch with the reference. In our experience,
many cohorts do not include stand data, or simply set all SNPs
to <code>"+"</code>, so the presence of strand-switched SNPs
isn't an indicator of problems by itself. However, if the
strand-switching did not fix the mismatch, there may a problem.
The subset of strand-switched SNPs that could not be fixed is
reported as <code>n_mismatch</code>, and indicates incorrect allele
data or, possibly, triallelic SNPs.
</p>
<p>Depending on the argument <code>save_mismatches</code>, mismatching
entries are exported
as a .txt file, together with the reference data. This allows
the user to see which SNPs are affected.
</p>
<p>Another sign of trouble is when negative-strand SNPs
(<code>n_negative_strand</code>) are present (i.e. the cohort included
real strand data, rather just setting it to <code>"+"</code>), but
strand-switching still occurred. Negative-strand SNPs
are converted to the positive strand before their alleles are
compared to the reference, so they should not appear here.
If they do, it means that either the strand-column data is
incorrect, or it is an ordinary mismatch (see above).
</p>
<p>Negative-strand SNPs that are &quot;strand-switched&quot; will revert
to their original allele configuration (but the strand column
now reports them as being on positive strand). The output of
<code>QC_GWAS</code> calls them double strand-switches, but here
they are reported as <code>n_negative_switch</code>. The subset of
those that could not be fixed is <code>n_negative_mismatch</code>.
</p>
<p>Just to be clear: the relevant output is still
<code>n_strandswitch</code> and <code>n_negative_strand</code>, not 
<code>n_negative_switch</code> and <code>n_negative_mismatch</code>.
Whether it was the negative-strand SNPs that were switched or
not is not important: the important thing is that there were
negative-strand SNPs (i.e. the cohort included real strand
data rather setting everything to <code>"+"</code>); yet
strand-switches were still necessary and cannot be attributed
to mismatch (i.e. the strand data is incorrect). 
</p>
<p><code>n_flipped</code> counts how many SNPs had their alleles
reversed to match the effect allele with the reference
minor-allele. This is merely recorded for &quot;administrative&quot;
purposes, and shouldn't concern the user.
</p>
<p><code>n_ambiguous</code> and <code>n_suspect</code> are another test of
the strand information. Ambiguous SNPs are SNPs that have
the same allele pair on the positive and negative strands (i.e.
A/T or C/G). Matching them with the allele-reference therefor
won't detected incorrect strand-information. In a normal-sized
GWAS results file, about 15% of SNPs will be ambiguous.
</p>
<p>Suspect SNPs are the subset of ambiguous SNPs whose allele
frequency is significantly different from that in the reference
( &lt; 0.35. vs &gt; 0.65 or visa versa). In our experience,
a GWAS results file with 2.5M SNPs will have only a few dozen
suspect SNPs. However, if it's a sizable proportion of all
ambiguous SNPs, it indicates that the ambiguous SNPs are
listed for the wrong strand. This will also have resulted
in the wrong SNPs being flipped in the previous step, so
it should be visible in the allele-frequency correlation test
as well.
</p>
<p><code>n_diffEAF</code> counts SNPs with significantly different
allele-frequencies. A large number here indicates either
that the allele-frequencies are incorrect or listed for the
wrong allele (see below), or that the population used in the
dataset does not match that of the reference.
</p>
<p>The <code>FRQ_cor</code> value is the correlation between the
reported and reference allele-frequencies. If allele frequency
is correct, the correlation should be near <code>1</code>. If it's
close to <code>-1</code>, the listed frequency is that of the other
(i.e. non-effect) allele.
</p>
<p>the <code>FRQ_cor_ambiguous</code> and <code>FRQ_cor_nonambi</code> values
are the same test for the subsets of ambiguous and
non-ambiguous SNPs. If ambiguous SNPs are listed on the wrong
strand, then they will have been flipped incorrectly, so
allele-frequency correlation should also move towards <code>-1</code>.
</p>


<h3>Note</h3>

<p>The function does not delete SNPs, regardless of the
<code>delete_mismatches</code> <code>delete_diffEAF</code> arguments.
Setting these to <code>TRUE</code> means that any such
SNPs are marked by having their effect allele set to <code>NA</code>.
The actual deletion takes place inside <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_hapmap_reference">create_hapmap_reference</a></code> for creating an allele
reference from publicly-available HapMap data
</p>
<p><code><a href="#topic+switch_strand">switch_strand</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># In order to keep the QCGWAS package small, no allele reference
# is included. Use the create_hapmap_reference function (see
# above) to create one.

## Not run: 
  data("gwa_sample")
  hapmap_ref &lt;- read.table("C:/new_hapmap/new_hapmap.txt",
                  header = TRUE, stringsAsFactors = FALSE)

  match_alleles(gwa_sample, hapmap_ref,
    dataname = "sample data", ref_name = "HapMap",
    save_name = "test_allele1", save_dir = "C:/new_hapmap",
    check_strand = TRUE, plot_FRQ = TRUE)

 HQ_SNPs &lt;- HQ_filter(data = gwa_sample, filter_NA = TRUE,
                          filter_FRQ = 0.01, filter_cal = 0.95)
  match_alleles(gwa_sample, hapmap_ref,
    HQ_subset = HQ_SNPs,
    dataname = "sample data", ref_name = "HapMap",
    save_name = "test_allele2", save_dir = "C:/new_hapmap",
    check_strand = TRUE, plot_FRQ = TRUE)

  match_output &lt;-
    match_alleles(gwa_sample, hapmap_ref,
      HQ_subset = HQ_SNPs,
      delete_mismatches = TRUE, return_SNPs = TRUE,
      delete_diffEAF = TRUE, threshold_diffEAF = 0.15,
      dataname = "sample data", ref_name = "HapMap",
      save_name = "test_allele3", save_dir = "C:/new_hapmap",
      check_strand = TRUE, plot_FRQ = TRUE)
  
  if(sum(match_output$n_negative_strand,
         match_output$n_strandswitch, match_output$n_mismatch,
         match_output$n_flipped, match_output$n_diffEAF) &gt; 0){
    gwa_sample$EFFECT_ALL   &lt;- match_output$EFFECT_ALL
    gwa_sample$OTHER_ALL    &lt;- match_output$OTHER_ALL
    gwa_sample$STRAND       &lt;- match_output$STRAND
    gwa_sample$EFFECT       &lt;- match_output$EFFECT
    gwa_sample$EFF_ALL_FREQ &lt;- match_output$EFF_ALL_FREQ
  }
  
## End(Not run)
</code></pre>

<hr>
<h2 id='plot_distribution'>GWAS effect-Size distribution plot</h2><span id='topic+plot_distribution'></span>

<h3>Description</h3>

<p>This function generates the effect-size distribution boxplot
created by <code><a href="#topic+QC_series">QC_series</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_distribution(data_table,
                  names = 1:ncol(data_table),
                  include = TRUE,
                  plot_order = 1:ncol(data_table),
                  quantile_lines = FALSE,
                  save_name = "Graph_distribution",
                  save_dir = getwd(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_distribution_+3A_data_table">data_table</code></td>
<td>
<p>table with a column of effect
sizes for every dataset.</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_names">names</code></td>
<td>
<p>vector; the names for the datasets, for use in the
graph. Note: it's best to keep these very short, as long
labels won't be plotted. The default is
the column <em>numbers</em> of <code>data_table</code>.</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_include">include</code></td>
<td>
<p>logical vector indicating which columns of
<code>data_table</code> are included in the plot. The default
setting is to include all.</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_plot_order">plot_order</code></td>
<td>
<p>numeric vector determining the left-to-right
order of plotting the datasets (columns). <code><a href="#topic+QC_series">QC_series</a></code>
uses the sample size for this.</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_quantile_lines">quantile_lines</code></td>
<td>
<p>logical; should lines representing the
median and quartile values be included?</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_save_name">save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the graph file.</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
graph is saved. Note that R uses <em>forward</em>
slash (/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="graphics.html#topic+boxplot">boxplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When running a QC over multiple files, <code><a href="#topic+QC_series">QC_series</a></code>
collects the values of the <code>effectsize_HQ</code> output of
<code><a href="#topic+QC_GWAS">QC_GWAS</a></code> in a table, which is then passed to this
function. If there are significant
differences in the distribution of effect sizes, it usually
indicates that the datasets did not use the same model or unit.
</p>


<h3>Value</h3>

<p>An invisible <code>NULL</code>.</p>


<h3>Note</h3>

<p>There is a known bug with this function when called by
<code><a href="#topic+QC_series">QC_series</a></code>. As input for <code>names</code>,
<code>QC_series</code> pastes together a shortened filename and a
<code>"N = x"</code> string giving the dataset's sample size.
</p>
<p>The filenames are truncated to the first unique element; e.g.
files <code>"cohortX_male_HB.txt"</code> and
<code>"cohortX_female_HB.txt"</code> become
<code>"cohortX_male; N = 608"</code> and
<code>"cohortX_female; N = 643"</code>, respectively. However, if
the unique element is longer than approx. 15 characters,
the label is too long to be plotted. The only solution is to
change the filenames prior to passing the files to
<code>QC_series</code>.
</p>


<h3>See Also</h3>

<p>For comparing reported to expected effect-size distribution:
<code><a href="#topic+QC_histogram">QC_histogram</a></code>.
</p>
<p>For other plots comparing GWAS results files:
<code><a href="#topic+plot_precision">plot_precision</a></code> and <code><a href="#topic+plot_skewness">plot_skewness</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  data("gwa_sample")

  chunk1 &lt;- gwa_sample$EFFECT[1:1000]
  chunk2 &lt;- gwa_sample$EFFECT[1001:2000]
  chunk3 &lt;- gwa_sample$EFFECT[2001:3000]

  plot_distribution(
    data_table = data.frame(chunk1, chunk2, chunk3),
    names = c("chunk 1", "chunk 2", "chunk 3"),
    quantile_lines = TRUE,
    save_name = "sample_distribution")
    
## End(Not run)
</code></pre>

<hr>
<h2 id='plot_precision'>GWAS Precision Plot</h2><span id='topic+plot_precision'></span>

<h3>Description</h3>

<p>This function generates the precision plot created by
<code><a href="#topic+QC_series">QC_series</a></code>. Precision is defined as:
</p>
<p><code>1 / median standard-error</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_precision(SE, N,
               labels = NULL,
               save_name = "Graph_precision",
               save_dir = getwd(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_precision_+3A_se">SE</code>, <code id="plot_precision_+3A_n">N</code></td>
<td>
<p>numeric vectors containing the median standard-error
and sample size of the datasets, respectively.</p>
</td></tr>
<tr><td><code id="plot_precision_+3A_labels">labels</code></td>
<td>
<p>vector containing names or other identifiers for
the datapoints, to be plotted in the graph. Note: it's best
to keep these very short. To disable labeling, set to
<code>NULL</code> (default).</p>
</td></tr>
<tr><td><code id="plot_precision_+3A_save_name">save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the graph file.</p>
</td></tr>
<tr><td><code id="plot_precision_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
graph is saved. Note that R uses <em>forward</em>
slash (/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="plot_precision_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When running a QC over multiple files, <code><a href="#topic+QC_series">QC_series</a></code>
collects the values of the <code>SE_HQ_median</code> and <code>sample_size_HQ</code>
outputs of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> in a table, which is then
passed to this function to convert it to a plot.
</p>
<p>The plot is to provide a visual estimate whether the standard
errors are within the expected range. As sample size increases,
the median standard error is expected to decrease, so the plot
should show a linear relation.
</p>


<h3>Value</h3>

<p>An invisible <code>NULL</code>.</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_distribution">plot_distribution</a></code> and <code><a href="#topic+plot_skewness">plot_skewness</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  value_SE &lt;- c(0.078, 0.189, 0.077, 0.040, 0.021, 0.072)
  value_N &lt;- c(870, 830, 970, 690, 2200, 870)
  value_labels &lt;- paste("cohort", 1:6)
  
  plot_precision(SE = value_SE, N = value_N,
                 labels = value_labels,
                 save_name = "sample_precision")
                 
## End(Not run)
</code></pre>

<hr>
<h2 id='plot_regional'>Regional Association Plot</h2><span id='topic+plot_regional'></span>

<h3>Description</h3>

<p>A regional association plot is essentially a zoomed-in
Manhattan plot, allowing the researcher to look at associations
in a small, pre-defined area of the genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_regional(dataset,
              chr, start_pos, end_pos,
              plot_cutoff_p = 1,
              name_cutoff_p,
              data_name = NULL,
              save_name = "regional_association_plot",
              save_dir = getwd(),
              header_translations,
              main = "Regional association plot", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_regional_+3A_dataset">dataset</code></td>
<td>
<p>data frame containing the SNPs chromosome number,
position, p-value and (if a <code>name_cutoff_p</code> is specified)
their SNP IDs.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_chr">chr</code></td>
<td>
<p>character or numeric value indicating the chromosome
of interest.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_start_pos">start_pos</code>, <code id="plot_regional_+3A_end_pos">end_pos</code></td>
<td>
<p>Numeric; the position values of the
beginning and end of the region of interest.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_plot_cutoff_p">plot_cutoff_p</code></td>
<td>
<p>numeric - the threshold of p-values to be
shown in the QQ &amp; Manhattan plots. Higher (less
significant) p-values are not included in the plot. Note
that, unlike in <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> or <code><a href="#topic+QC_plots">QC_plots</a></code>,
the default value is <code>1</code> (i.e. includes everything),
rather than <code>0.05</code>. Setting it to <code>0.05</code> excludes
95% of data-points, which significantly reduces running time
and memory usage in larger datasets.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_name_cutoff_p">name_cutoff_p</code></td>
<td>
<p>numeric; SNPs with p-values lower than or
equal to this value will have their names plotted in the
graph. If set to <code>NULL</code> (default), no names will be
plotted.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_data_name">data_name</code></td>
<td>
<p>character string; the subtitle for the plot.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_save_name">save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the graph file.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
graph is saved. Note that R uses <em>forward</em>
slash (/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_header_translations">header_translations</code></td>
<td>
<p>translation table for column names.
See <code><a href="#topic+translate_header">translate_header</a></code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="plot_regional_+3A_main">main</code>, <code id="plot_regional_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, <code>plot_regional</code> expects <code>dataset</code> to use the
standard column-names used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>. A
translation table can be specified in <code>header_translations</code>
to allow non-standard names. See <code><a href="#topic+translate_header">translate_header</a></code>
for more information.
</p>


<h3>Value</h3>

<p>An invisible <code>NULL</code>.</p>


<h3>See Also</h3>

<p>For creating a Manhattan plot: <code><a href="#topic+QC_plots">QC_plots</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  data("gwa_sample")

  plot_regional(dataset = gwa_sample,
          chr = 2, start_pos = 55000000, end_pos = 75000000,
          data_name = "QC GWAS sample data",
          save_name = "sample_regional_association")
          
## End(Not run)
</code></pre>

<hr>
<h2 id='plot_skewness'>GWAS Skewness vs. Kurtosis Plot</h2><span id='topic+plot_skewness'></span>

<h3>Description</h3>

<p>This function generates the skewness vs. kurtosis plot created
by <code><a href="#topic+QC_series">QC_series</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_skewness(skewness,
              kurtosis,
              labels = paste("Study", 1:length(skewness)),
              plot_labels = "outliers",
              save_name = "Graph_skewness_kurtosis",
              save_dir = getwd(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_skewness_+3A_skewness">skewness</code>, <code id="plot_skewness_+3A_kurtosis">kurtosis</code></td>
<td>

<p>Vectors containing the skewness and kurtosis values of the
datasets</p>
</td></tr>
<tr><td><code id="plot_skewness_+3A_labels">labels</code></td>
<td>
<p>vector containing names or other identifiers for
the datapoints, to be plotted in the graph. Note: it's best
to keep these very short.</p>
</td></tr>
<tr><td><code id="plot_skewness_+3A_plot_labels">plot_labels</code></td>
<td>
<p>character string or logical determining
whether the values in <code>labels</code> are plotted next to the
data points. The possible settings are: <code>"none"</code> (or 
<code>FALSE</code>); <code>"all"</code> (or <code>TRUE</code>); and
<code>"outliers"</code> for outliers only.</p>
</td></tr>
<tr><td><code id="plot_skewness_+3A_save_name">save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the graph file.</p>
</td></tr>
<tr><td><code id="plot_skewness_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
graph is saved. Note that R uses <em>forward</em>
slash (/) where Windows uses backslash (\).</p>
</td></tr>
<tr><td><code id="plot_skewness_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When running a QC over multiple files, <code><a href="#topic+QC_series">QC_series</a></code>
collects the values of the <code>skewness_HQ</code> and <code>kurtosis_HQ</code>
output of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> in a table, which is then
passed to this function to convert it into a plot. Note that
this values are calculated over high-quality SNPs only.
</p>
<p>Kurtosis is a measure of how well a distribution matches a
Gaussian distribution. A Gaussian distribution has a kurtosis
of <code>0</code>. Negative kurtosis indicates a flatter distribution
curve, while positive kurtosis indicates a sharper, thinner
curve.
</p>
<p>Skewness is a measure of distribution asymmetry. A symmetrical
distribution has skewness <code>0</code>. A positive skewness
indicates a long tail towards higher values, while a negative
skewness indicates a long tail towards lower values.
</p>
<p>Ideally, one expects both the skewness and kurtosis of effect
sizes to be close to <code>0</code>. In practice, these statistics
can be hugely variable. <code>QC_series</code> uses only high-quality
effect sizes to calculate these values in order to reduce some
of the more extreme values. Still, it is recommended that you
compare the values to those of other GWAS with the same
phenotype, rather than relying on on the label outliers
command to identify problems.
</p>


<h3>Value</h3>

<p>An invisible <code>NULL</code>.</p>


<h3>See Also</h3>

<p>For calculating skewness and kurtosis: <code><a href="#topic+calc_kurtosis">calc_kurtosis</a></code>.
</p>
<p>For other plots comparing GWAS results files:
<code><a href="#topic+plot_precision">plot_precision</a></code> and <code><a href="#topic+plot_distribution">plot_distribution</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  value_S &lt;- c(0.05, -0.27, 0.10, 0.11)
  value_K &lt;- c( 6.7,  10.0, 10.1,  6.6)
  value_labels &lt;- paste("cohort", 1:4)
  
  ## Not run: 
  plot_skewness(skewness = value_S,
                kurtosis = value_K,
                labels = value_labels,
                plot_labels = "outliers",
                save_name = "sample_skewness_kurtosis")
## End(Not run)
</code></pre>

<hr>
<h2 id='QC_GWAS'>Automated Quality Control of GWAS Results files</h2><span id='topic+QC_GWAS'></span><span id='topic+QC_series'></span>

<h3>Description</h3>

<p><code>QC_GWAS</code> runs a full quality control (QC) over a single
GWAS results file. It removes missing and invalid data, checks
the alleles and allele frequency with a reference, tests the
reported p-value against both calculated and expected values,
creates QQ and Manhattan plots and reports the distribution of
the quality-parameters within the dataset, as well as various
QC statistics.
</p>
<p><code>QC_series</code> does the same thing for multiple GWAS results
files. It's mainly a wrapper that passes individual files to
<code>QC_GWAS</code>, but it has a few extra features, such as
making a checklist of important QC stats
and creating several graphs to compare the QC'ed files.
</p>
<p>Although the number of arguments in <code>QC_GWAS</code> may seem
overwhelming, only three of them are required to run a basic
QC. The name of the file to be QC'ed should be passed to the
<code>filename</code> argument; the directory of said file to the
<code>dir_data</code> argument; and a header-translation table to
the <code>header_translations</code> argument.
For a quick introduction to QCGWAS, read the quick
reference guide that can be found in &quot;R\library\QCGWAS\doc&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_GWAS(filename,
        filename_output = paste0("QC_", filename),
        dir_data = getwd(),
        dir_output = paste(dir_data, "QCGWASed", sep = "/"),
        dir_references = dir_data,
        header_translations,
        column_separators = c("\t", " ", "", ",", ";"),
        nrows = -1, nrows_test = 1000,
        header = TRUE, comment.char = "",
        na.strings = c("NA", "nan", "NaN", "."),
        imputed_T = c("1", "TRUE", "T"),
        imputed_F = c("0", "FALSE", "F"),
        imputed_NA = c(NA, "-"),
        save_final_dataset = TRUE,
        gzip_final_dataset = TRUE, order_columns = FALSE,
        spreadsheet_friendly_log = FALSE,
        out_header = "standard",
        out_quote = FALSE, out_sep = "\t", out_eol = "\n",
        out_na = "NA", out_dec = ".", out_qmethod = "escape",
        out_rownames = FALSE, out_colnames = TRUE,
        return_HQ_effectsizes = FALSE,
        remove_X = FALSE, remove_Y = FALSE,
        remove_XY = remove_Y, remove_M = FALSE,
        calculate_missing_p = FALSE,
        make_plots = TRUE, only_plot_if_threshold = TRUE,
        threshold_allele_freq_correlation = 0.95,
        threshold_p_correlation = 0.99,
        plot_intensity = FALSE,
        plot_histograms = make_plots, plot_QQ = make_plots,
        plot_QQ_bands = TRUE, plot_Manhattan = make_plots,
        plot_cutoff_p = 0.05,
        allele_ref_std, allele_name_std,
        allele_ref_alt, allele_name_alt,
        update_alt = FALSE, update_savename,
        update_as_rdata = FALSE, backup_alt = FALSE,
        remove_mismatches = TRUE,
        remove_mismatches_std = remove_mismatches,
        remove_mismatches_alt = remove_mismatches,
        threshold_diffEAF = 0.15, remove_diffEAF = FALSE,
        remove_diffEAF_std = remove_diffEAF,
        remove_diffEAF_alt = remove_diffEAF,
        check_ambiguous_alleles = FALSE,
        use_threshold = 0.1,
        useFRQ_threshold = use_threshold,
        useHWE_threshold = use_threshold,
        useCal_threshold = use_threshold,
        useImp_threshold = use_threshold,
        useMan_threshold = use_threshold,
        HQfilter_FRQ = 0.01, HQfilter_HWE = 10^-6,
        HQfilter_cal = 0.95, HQfilter_imp = 0.3,
        QQfilter_FRQ = c(NA, 0.01, 0.05),
        QQfilter_HWE = c(NA, 10^-6, 10^-4),
        QQfilter_cal = c(NA, 0.95, 0.99),
        QQfilter_imp = c(NA, 0.3, 0.5, 0.8),
        NAfilter = TRUE,
        NAfilter_FRQ = NAfilter, NAfilter_HWE = NAfilter,
        NAfilter_cal = NAfilter, NAfilter_imp = NAfilter,
        ignore_impstatus = FALSE,
        minimal_impQ_value = -0.5, maximal_impQ_value = 1.5,
        logI = 1L, logN = 1L, ...)
QC_series(data_files, datafile_tag, output_filenames,
          dir_data = getwd(),
          dir_output = paste(dir_data, "QCGWASed", sep = "/"),
          dir_references = dir_data,
          header_translations, out_header = "standard",
          allele_ref_std, allele_name_std,
          allele_ref_alt, allele_name_alt,
          update_alt = FALSE, update_savename,
          update_as_rdata = FALSE, backup_alt = FALSE,
          plot_effectsizes = TRUE, lim_effectsizes = NULL,
          plot_SE = TRUE, label_SE = TRUE,
          plot_SK = TRUE, label_SK = "outliers",
          save_filtersettings = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_GWAS_+3A_filename">filename</code>, <code id="QC_GWAS_+3A_data_files">data_files</code>, <code id="QC_GWAS_+3A_datafile_tag">datafile_tag</code></td>
<td>

<p><code>filename</code> and <code>data_files</code> are, respectively, the
name and names of the GWAS results file(s) to be QC'ed. If
no <code>data_files</code> are specified, <code>QC_series</code> will
process all filed in <code>dir_data</code> containing the string
passed to <code>datafiles_tag</code> in their filename. See
below for more information on the input requirements.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_filename_output">filename_output</code>, <code id="QC_GWAS_+3A_output_filenames">output_filenames</code></td>
<td>

<p>respectively the filename or names of the output of the QC.
This should not include an extension, since the QC will
automatically add one. The default is to use the input
filename with <code>"QC_"</code> prefixed.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_dir_data">dir_data</code>, <code id="QC_GWAS_+3A_dir_output">dir_output</code>, <code id="QC_GWAS_+3A_dir_references">dir_references</code></td>
<td>
<p>character strings
specifying the directory dress of the folders for, respectively,
the input file(s), the output file(s) and the auxillary files
(header-translation tables and allele references). Note that
R uses <em>forward</em> slash (/) where Windows uses backslash (\).
If <code>dir_output</code> does not exist, it will be created.
If no <code>dir_output</code> is specified, a folder named
<code>"QCGWASed"</code> will be created in <code>dir_data</code>.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_header_translations">header_translations</code></td>
<td>
<p>Translation table for the column 
names of the <em>input</em> file. Alternatively, the name of a file
in <code>dir_references</code> containing such a table. See
<code><a href="#topic+translate_header">translate_header</a></code> for details.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_column_separators">column_separators</code></td>
<td>
<p>character string or vector; specifies
the values used as column delimitator in the GWAS file. The
argument is passed to <code><a href="#topic+load_test">load_test</a></code>; see the
description of that function for more information.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_nrows_test">nrows_test</code></td>
<td>
<p>integer; the number of rows used for
&quot;trial-loading&quot;. Before loading the entire dataset, the
function <code><a href="#topic+load_test">load_test</a></code> is called to determine the
dataset's file-format by reading the top <code>x</code> lines, where
<code>x</code> is <code>nrows_test</code>. Setting <code>nrows_test</code> to a low number
(e.g. <code>150</code>) means quick testing, but runs the risk of
missing problems in lower rows. To test the entire dataset,
set it to <code>-1</code>.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_nrows">nrows</code>, <code id="QC_GWAS_+3A_header">header</code>, <code id="QC_GWAS_+3A_comment.char">comment.char</code></td>
<td>
<p>arguments passed to
<code><a href="utils.html#topic+read.table">read.table</a></code> when importing the dataset.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_na.strings">na.strings</code></td>
<td>
<p>character vector describing the values that
represent missing data in the dataset. Passed to
<code><a href="utils.html#topic+read.table">read.table</a></code>.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_imputed_t">imputed_T</code>, <code id="QC_GWAS_+3A_imputed_f">imputed_F</code>, <code id="QC_GWAS_+3A_imputed_na">imputed_NA</code></td>
<td>
<p>character vectors;
passed to <code><a href="#topic+convert_impstatus">convert_impstatus</a></code> (as <code>T_strings</code>,
<code>F_strings</code> and <code>NA_strings</code>, respectively) to
translate the imputation-status column. Note that the
current version of <code>QC_GWAS</code> <em>always</em> translates
the imputation status. Even when the dataset already has
the correct format, the user still needs to specify <code>1</code>,
<code>0</code> and <code>NA</code> for these arguments, respectively.
Also note that R distinguishes between the value <code>NA</code>
and the character string <code>"NA"</code>.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_save_final_dataset">save_final_dataset</code></td>
<td>
<p>logical; should the post-QC dataset
be saved?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_gzip_final_dataset">gzip_final_dataset</code></td>
<td>
<p>logical; should the post-QC dataset
be compressed?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_order_columns">order_columns</code></td>
<td>
<p>logical; should the post-QC dataset use
the default column order?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_spreadsheet_friendly_log">spreadsheet_friendly_log</code></td>
<td>
<p>logical; if <code>TRUE</code>, the
final log file will be tab-separated, for easy viewing in a
spreadsheet program. If <code>FALSE</code> (default), it will be
formatted for pretty viewing in a text-processing program.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_out_header">out_header</code></td>
<td>
<p>Translation table for the column names of
the <em>output</em> file. This argument is the opposite of
<code>header_translations</code>: it translates the standard
column-names of <code>QC_GWAS</code> to user-defined ones.
<code>output_header</code> can be one of three things:
</p>

<ul>
<li><p> A user specified table similar to the one used by
<code><a href="#topic+translate_header">translate_header</a></code>. However, as this
translates standard names into non-standard ones, the
standard names should be in the right column, and the
desired ones in the left. There is also no requirement
for the names in the <em>left</em> column to be uppercase.
</p>
</li>
<li><p> The name of a file in <code>dir_references</code> containing
such a column.
</p>
</li>
<li><p> Character string specifying a standard form. See
the section 'Output header' below for the options.
</p>
</li></ul>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_out_quote">out_quote</code>, <code id="QC_GWAS_+3A_out_sep">out_sep</code>, <code id="QC_GWAS_+3A_out_eol">out_eol</code>, <code id="QC_GWAS_+3A_out_na">out_na</code>, <code id="QC_GWAS_+3A_out_dec">out_dec</code>, <code id="QC_GWAS_+3A_out_qmethod">out_qmethod</code>, <code id="QC_GWAS_+3A_out_rownames">out_rownames</code>, <code id="QC_GWAS_+3A_out_colnames">out_colnames</code></td>
<td>
<p>arguments passed to
<code><a href="utils.html#topic+write.table">write.table</a></code> when saving the final dataset.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_return_hq_effectsizes">return_HQ_effectsizes</code></td>
<td>
<p>logical; return a vector of (max.
1000) high-quality effect-sizes? (In <code>QC_series</code>, this
is set by the <code>plot_effectsizes</code> argument.)</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_remove_x">remove_X</code>, <code id="QC_GWAS_+3A_remove_y">remove_Y</code>, <code id="QC_GWAS_+3A_remove_xy">remove_XY</code>, <code id="QC_GWAS_+3A_remove_m">remove_M</code></td>
<td>
<p>logical;
respectively whether X-chromosome, Y-chromosome,
pseudo-autosomal and mitochondrial SNPs are removed from the
dataset.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_calculate_missing_p">calculate_missing_p</code></td>
<td>
<p>logical; should the QC calculate
missing/invalid p-values in the dataset?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_make_plots">make_plots</code></td>
<td>
<p>logical; should the QC generate and save
QQ plots, a Manhattan plot, histograms of data distribution
and scatter plots of correlation in <code>dir_reference</code>?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_only_plot_if_threshold">only_plot_if_threshold</code></td>
<td>
<p>logical; should the scatterplots
only be made if the correlation is below a threshold value?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_threshold_allele_freq_correlation">threshold_allele_freq_correlation</code>, <code id="QC_GWAS_+3A_threshold_p_correlation">threshold_p_correlation</code></td>
<td>

<p>numeric; thresholds for reporting and plotting
the correlation between respectively
the allele frequency of the dataset and the reference, and
the calculated and reported p-values.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_plot_intensity">plot_intensity</code></td>
<td>
<p>logical; if <code>TRUE</code>, instead of a
scatterplot of allele correlations, an intensity plot is
generated. This option is currently only partially
implemented. Leave to <code>FALSE</code> for now.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_plot_histograms">plot_histograms</code></td>
<td>
<p>logical; should histograms of the effect
sizes, standard errors, allele frequencies, HWE p-values,
callrates and imputation quality be made?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_plot_qq">plot_QQ</code>, <code id="QC_GWAS_+3A_plot_manhattan">plot_Manhattan</code></td>
<td>
<p>logical; should QQ and Manhattan
plots be made?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_plot_qq_bands">plot_QQ_bands</code></td>
<td>
<p>logical; include probability bands in the
QQ plot?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_plot_cutoff_p">plot_cutoff_p</code></td>
<td>
<p>numeric; significance threshold for
inclusion in the QQ and Manhattan plots. The default value
(<code>0.05</code>) excludes 95% of SNPs, significantly reducing
running-time and memory usage. For this reason it is not
recommend to set a higher value when QC'ing a normal-sized
GWAS dataset.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_allele_ref_std">allele_ref_std</code>, <code id="QC_GWAS_+3A_allele_ref_alt">allele_ref_alt</code></td>
<td>
<p>the standard and alternative
allele-reference tables. Alternatively, the name of a file
in <code>dir_references</code> containing said table. Files in
<code>.RData</code> format are accepted, but the table's object
name must match the argument name. See <code><a href="#topic+match_alleles">match_alleles</a></code>
for more information on the input requirements.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_allele_name_std">allele_name_std</code>, <code id="QC_GWAS_+3A_allele_name_alt">allele_name_alt</code></td>
<td>
<p>character strings;
these name the standard and alternative allele reference in
the output. If no values are given, the function will use the
reference's filename (if specified) or a default name.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_update_alt">update_alt</code></td>
<td>
<p>logical; if the function encounters SNPs not
included in either the standard or alternative reference,
should these be added to the alternative reference file? If
no alternative reference was specified, this creates one.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_update_savename">update_savename</code></td>
<td>
<p>character string; the filename for
saving the updated alternative reference, <em>without</em>
extension. If <code>allele_ref_alt</code> is a filename, it is
not necessary to specify this argument.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_update_as_rdata">update_as_rdata</code></td>
<td>
<p>logical; should the updated alternative
allele reference be saved as <code>.RData</code> (<code>TRUE</code>) or
a tab-delimitated .txt file (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_backup_alt">backup_alt</code></td>
<td>
<p>logical; if the alternative allele reference is updated,
should a back-up be made of the original reference file?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_remove_mismatches">remove_mismatches</code>, <code id="QC_GWAS_+3A_remove_mismatches_std">remove_mismatches_std</code>, <code id="QC_GWAS_+3A_remove_mismatches_alt">remove_mismatches_alt</code></td>
<td>

<p>logical; should SNPs with mismatching alleles be removed
from the dataset? <code>remove_mismatches</code> serves as the
default value; the other two arguments determine this
setting for the standard and alternative references,
respectively.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_threshold_diffeaf">threshold_diffEAF</code></td>
<td>

<p>Numeric; the threshold for the difference between reported
and reference allele-frequency. SNPs for which the
difference exceeds the threshold are counted and
(optionally) removed.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_remove_diffeaf">remove_diffEAF</code>, <code id="QC_GWAS_+3A_remove_diffeaf_std">remove_diffEAF_std</code>, <code id="QC_GWAS_+3A_remove_diffeaf_alt">remove_diffEAF_alt</code></td>
<td>

<p>Logical; should SNPs that exceed the <code>threshold_diffEAF</code>
be removed from the dataset? <code>remove_diffEAF</code> serves as
the default value; the other two arguments determine this
setting for the standard and alternative references,
respectively.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_check_ambiguous_alleles">check_ambiguous_alleles</code></td>
<td>
<p>logical; check for SNPs with
strand-independent allele-configurations (i.e. A/T and C/G
SNPs)?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_use_threshold">use_threshold</code></td>
<td>

<p>numeric; threshold value. The relative or absolute number of
usable values required for a variable to be used in the QC.
These arguments prevent the QC from applying filters to variables
with no data. If a variable has less non-missing, non-invalid
values than specified in the threshold, it will be ignored;
i.e. no filter will be applied to it and no plots will be made.
Values <code>&gt; 1</code> specify the absolute threshold, while
values of <code>1</code> or lower specify the fraction of SNPs
remaining in the dataset. This argument is the
default threshold for all variables; variable-specific
thresholds can be set with the following arguments.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_usefrq_threshold">useFRQ_threshold</code>, <code id="QC_GWAS_+3A_usehwe_threshold">useHWE_threshold</code>, <code id="QC_GWAS_+3A_usecal_threshold">useCal_threshold</code>, <code id="QC_GWAS_+3A_useimp_threshold">useImp_threshold</code>, <code id="QC_GWAS_+3A_useman_threshold">useMan_threshold</code></td>
<td>

<p>numeric; variable-specific thresholds for allele frequency,
HWE p-value, callrate, imputation quality and Manhattan plot
(i.e. chromosome &amp; position values) respectively.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_hqfilter_frq">HQfilter_FRQ</code>, <code id="QC_GWAS_+3A_hqfilter_hwe">HQfilter_HWE</code>, <code id="QC_GWAS_+3A_hqfilter_cal">HQfilter_cal</code>, <code id="QC_GWAS_+3A_hqfilter_imp">HQfilter_imp</code></td>
<td>

<p>numeric; threshold values for the high-quality SNP filter.
SNPs that do not meet or exceed all four thresholds will be
excluded from several QC tests. The filters are for allele
frequency, HWE p-value, callrate &amp; imputation quality,
respectively, and are processed by <code><a href="#topic+HQ_filter">HQ_filter</a></code>.
See 'Filter arguments' for more information. Note: the high-quality
filter does not remove SNPs; it merely excludes them from
several QC tests.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_qqfilter_frq">QQfilter_FRQ</code>, <code id="QC_GWAS_+3A_qqfilter_hwe">QQfilter_HWE</code>, <code id="QC_GWAS_+3A_qqfilter_cal">QQfilter_cal</code>, <code id="QC_GWAS_+3A_qqfilter_imp">QQfilter_imp</code></td>
<td>

<p>numeric vector; threshold values for the QQ plot filters.
SNPs that do not meet or exceed the value will be excluded
from the QQ plot. Up to five values can be specified per
filter. The filters are for allele-frequency, HWE p-value,
callrate &amp; imputation quality respectively, and are
processed by <code><a href="#topic+QC_plots">QC_plots</a></code>. See 'Filter arguments' for more
information.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_nafilter">NAfilter</code>, <code id="QC_GWAS_+3A_nafilter_frq">NAfilter_FRQ</code>, <code id="QC_GWAS_+3A_nafilter_hwe">NAfilter_HWE</code>, <code id="QC_GWAS_+3A_nafilter_cal">NAfilter_cal</code>, <code id="QC_GWAS_+3A_nafilter_imp">NAfilter_imp</code></td>
<td>

<p>logical; should the high-quality and QQ filters exclude
(<code>TRUE</code>) or ignore (<code>FALSE</code>) missing values?
<code>NAfilter</code> is the default setting; the others allow
allow variable specific settings.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_ignore_impstatus">ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_minimal_impq_value">minimal_impQ_value</code>, <code id="QC_GWAS_+3A_maximal_impq_value">maximal_impQ_value</code></td>
<td>

<p>numeric; the minimal and maximal possible (i.e. non-invalid)
imputation quality values.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_logi">logI</code>, <code id="QC_GWAS_+3A_logn">logN</code></td>
<td>
<p>progress indicators used by <code>QC_series</code>:
irrelevant for users.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_plot_effectsizes">plot_effectsizes</code>, <code id="QC_GWAS_+3A_plot_se">plot_SE</code>, <code id="QC_GWAS_+3A_plot_sk">plot_SK</code></td>
<td>
<p>logical; additional
plot options for <code>QC_series</code>. The arguments toggle the
creation of plots comparing the effect-size distribution,
precision and skewness vs. kurtosis of all successfully QC'ed
datasets, respectively. See <code><a href="#topic+plot_distribution">plot_distribution</a></code>,
<code><a href="#topic+plot_precision">plot_precision</a></code> and <code><a href="#topic+plot_skewness">plot_skewness</a></code>
for more information.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_lim_effectsizes">lim_effectsizes</code></td>
<td>
<p>specifies the y-axis range of the
effect-size distribution plot.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_label_se">label_SE</code></td>
<td>
<p>logical; should the datapoints in the precision
plot be labeled?</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_label_sk">label_SK</code></td>
<td>
<p>character string; determines whether the
datapoints in the skewness vs. kurtosis plot are labeled.
Options are <code>"none"</code>, <code>"all"</code>, or <code>"outliers"</code>
(outliers only).</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_save_filtersettings">save_filtersettings</code></td>
<td>
<p>logical; saves the filtersettings
used by the high-quality filter to a file
'Check_filtersettings.txt' in the output directory. If
a file of that name already exists, the settings are added
to the end (i.e. it updates rather than overwrites existing
files). The file can be used as ini file by
<code><a href="#topic+filter_GWAS">filter_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="QC_GWAS_+3A_...">...</code></td>
<td>
<p>in <code>QC_series</code>: arguments passed to
<code>QC_GWAS</code>; in <code>QC_GWAS</code>: arguments passed to
<code><a href="utils.html#topic+read.table">read.table</a></code> when importing the dataset.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The full quality-control carried out by <code>QC_GWAS</code> consists
of 5 phases. The function takes a single dataset (or, rather,
the location and filename of a single dataset) and runs it
through the following phases:
</p>

<ul>
<li><p> 1: Importing the dataset
</p>
</li>
<li><p> 2: Checking data integrity
</p>
</li>
<li><p> 3: Checking alleles
</p>
</li>
<li><p> 4: Generating QC statistics and graphs
</p>
</li>
<li><p> 5: Saving the output
</p>
</li></ul>

<p><em>Phase 1: importing the dataset</em>
</p>
<p>GWAS results files come in a variety of formats, so
<code>QC_GWAS</code> is flexible about loading data. It
uses an autoloader function (<code><a href="#topic+load_GWAS">load_GWAS</a></code>) to
unpack <code>.zip</code> or <code>.gz</code> files and determine the
column-separator used in the file. See the section
'Requirement for the input dataset' for more information.
</p>
<p>Next, the function attempts to translate the dataset's column
names (the header) to standard names, so that it knows what
type of data a column contains. This is done by comparing
the column names to a translation table (specified in the
<code>header_translations</code> argument). See
<code><a href="#topic+translate_header">translate_header</a></code> for more information.
</p>
<p>Note that only the SNP ID, alleles, effect-size and standard
error columns are required. The absence of other standard
columns (chromosome, position, strand, allele frequency, HWE
p-value, callrate, imputation quality, imputation status and
used for imputation) will not cause the QC to abort.
Instead, a warning is printed on screen and in the log file,
and a dummy column filled with <code>NA</code> values is added to
the dataset.
</p>
<p>It is therefor important to check the log file: if a standard
column is present but not identified (because it is missing or
misspelled in the translation table) the QC will continue,
but is unable to check/use the data inside. The unidentified
column will be reported in the <code>columns_unidentified</code>
value of the <code>QC_GWAS</code> return or in the
&quot;QC_checklist.txt&quot; file generated by <code>QC_series</code>.
</p>
<p><em>Phase 2: checking data integrity</em>
</p>
<p>The purpose of phase 2 is to ensure that the dataset <em>can</em>
be QC'ed: that that all SNPs have the required data and that
all columns contain only valid (or missing) values.
</p>
<p>The first step is to remove SNPs that won't be used: monomorphic
SNPs and (if specified by the arguments) allosomal, 
pseudo-autosomal and mitochondrial SNPs. The function considers
SNPs monomorphic if they have a missing or invalid other
(non-effect) allele, identical alleles or an allele-frequency
of <code>1</code> or <code>0</code>.
</p>
<p>The second step is to check the imputation status column with
the function <code><a href="#topic+convert_impstatus">convert_impstatus</a></code>. See the section
'Requirement for the input dataset' for more information.  
Imputation status is one of the most important variables in
the dataset: if unknown, the HWE p-value, callrate and imputation
quality won't be used (unless <code>ignore_impstatus</code> is
<code>TRUE</code>), as the function cannot determine which
SNPs are imputed and which are not. For this reason, if
<code>convert_impstatus</code> is unable to translate any character 
string in the column, the QC will abort.
</p>
<p>The third step carries out three tests for all other standard
variables:
</p>

<ul>
<li><p> Does the column contain the correct type of data
(integer, numeric or character)?
</p>
</li>
<li><p> How many values are missing (<code>NA</code>)?
</p>
</li>
<li><p> How many values are invalid (= impossible)?
</p>
</li></ul>

<p>The exact nature of the three tests differs per variable: see
the documentation file in &quot;R\library\QCGWAS\doc&quot; for more
detail.
</p>
<p>The presence of the wrong data-type will cause the QC to abort.
Wrong data-type indicates either a problem in the file
itself, or with the way it was imported (in which case it is
most likely due to a mistranslated header).
</p>
<p>The final step is the removal of the invalid values and of
unusable SNPs. The variables MARKER, EFFECT_ALLELE, OTHER_ALLELE,
EFFECT and STDERR are considered crucial. SNPs with missing or
invalid values in any of these variables are removed the dataset.
Missing values in the other variables are ignored, while
invalid values are set to <code>NA</code>.
</p>
<p><em>Phase 3: checking alleles</em>
</p>
<p>This phase has three functions:
</p>

<ul>
<li><p> To check if the correct alleles are reported for each SNP
</p>
</li>
<li><p> To check if the allele-frequency is reported for the
correct (effect) allele
</p>
</li>
<li><p> To ensure that SNPs are aligned to the positive strand
and use the same effect-allele in all post-QC datasets
</p>
</li></ul>

<p>This is achieved by comparing the data to a reference, using
the function <code><a href="#topic+match_alleles">match_alleles</a></code>. First, all SNPs are
switched to the positive strand (the alleles are converted to
their opposing base and the strand-value is set to <code>"+"</code>).
If there are SNPs whose allele pair doesn't match the
reference, <code>match_alleles</code> assumes the information in
the strand column is absent or incorrect, and will also
switch those SNPs to the other (presumably positive) strand.
This step is referred to as strand-switching in QC output, and
is independent from the negative-strand SNP conversion. It is
therefor possible that a SNP is switched twice: once because
the strand-column indicates it is on the negative strand, and
twice because of a mismatch. This is referred to as double
strand-switch in the output, and indicates either the wrong
value in the strand column, or a mismatch with the reference.
In the latter case, it will most likely be picked up in the
next step.
</p>
<p>If the strand-switch does not fix the mismatch, the SNPs
are counted in the QC output as mismatches. Depending on the
<code>remove_mismatches</code> arguments, the SNPs will either be
removed from the dataset, or left in but excluded from the
further tests of the allele-matching.
</p>
<p>Next, <code>match_alleles</code> &quot;flips&quot; SNPs so that their effect
allele matches the reference minor allele. This ensures
that a SNP will have the same effect allele in all post-QC
datasets.
</p>
<p><code>match_alleles</code> also counts the number of SNPs with a
strand-independent allele configuration (A/T or C/G; these are
designated as &quot;ambiguous SNPs&quot;), and the subset of those with
an allele-frequency that is substantially different from the
reference (&quot;suspect SNPs&quot;). If a substantial proportion of
ambiguous SNPs is suspect, it indicates that the strand
information is incorrect. In our experience, a regular, 2.5M
SNP dataset usually consists of 15% ambiguous SNPs, of which
a few dozen will be suspect.
</p>
<p>The function also counts the number of SNPs whose allele
frequency differs from the reference by more than a set amount
(<code>threshold_difEAF</code>). If the relevant <code>remove_diffEAF</code>
argument is <code>TRUE</code>, these SNP will be excluded from the
dataset after the allele-matching is finished.
</p>
<p>The final step is to correlate the reported allele-frequency
against the reference. If allele-frequency is reported for
the correct (effect) allele, the correlation should be close
to <code>1</code>. If the outcome is close to <code>-1</code>, the reported
frequency is that of the other allele. Depending on the plot settings,
a scatter plot of reported vs. reference frequency is made
for all SNPs, and for the subsets of ambiguous and non-ambiguous
SNPs.
</p>
<p><em>The standard and alternative allele reference</em>
</p>
<p>The above steps describe what happens when the dataset is
compared to a single reference. However, we found that many
GWAS datasets contain SNPs not present in our standard HapMap
reference, so we added a second, flexible reference that can be
updated with any unknown SNPs the QC encounters.
</p>
<p>SNPs that are not found in either reference are converted
to the positive strand, and &quot;flipped&quot; if their allele frequency
is &gt; 0.50. If <code>update_alt</code> is <code>TRUE</code>, these SNPs are
then added to the alternative
reference and saved under the name <code>update_savename</code>.
There are a few caveats to this system: see the section
'Updating the alternative reference' for details.
</p>
<p><em>Phase 4: generating QC statistics and graphs</em>
</p>
<p>At this stage, no further changes will be made to the dataset
(except, optionally, to recalculate missing p-values).
The function will now start to calculate the QC statistics and
generate the important graphs. These are:
</p>

<ul>
<li><p> Create histograms of variable distribution (optional)
</p>
</li>
<li><p> Check p-values by correlating them to a p calculated
from the effect-size and standard-error (via the
<code><a href="#topic+check_P">check_P</a></code> function).
</p>
</li>
<li><p> Recalculate missing/invalid p-values (optional)
</p>
</li>
<li><p> Calculate QC statistics
</p>
</li>
<li><p> Create QQ and Manhattan plots (optional, see
<code><a href="#topic+QC_plots">QC_plots</a></code> function for more information).
</p>
</li></ul>

<p><em>Phase 5: saving the output</em>
</p>
<p>A series of tables is added to the bottom of
the log file, reporting the QC statistics and the data
distribution. If <code>save_final_dataset</code> is <code>TRUE</code>, the
post-QC data will be exported as a .txt file. The column names
and format of that file can be specified by the out arguments.
</p>


<h3>Value</h3>

<p>The most important output of the QC is the log file. See the
section 'QC output files' for more details. This section only
describes the function return within R.
</p>
<p><code>QC_series</code> returns a single, invisible, logical value,
indicating whether the alternative allele-reference has
been updated.
</p>
<p><code>QC_GWAS</code> returns an object of class 'list'. If the QC
was not successful, this list contains only five of the following
components (<code>QC_successful</code>, <code>filename_input</code>,
<code>filename_output</code>, <code>all_ref_changed</code>, <code>effectsize_return</code>).
If it was, it will contain all of them:
</p>
<table>
<tr><td><code>QC_successful</code></td>
<td>
<p>logical; indicates whether the QC was
completed. If <code>FALSE</code>, the function was either unable
to load the dataset, encountered an unexpected datatype, or
removed all SNPs during the QC. The log file and screen
output will indicate what triggered the abort.</p>
</td></tr>
<tr><td><code>filename_input</code>, <code>filename_output</code></td>
<td>
<p>the filenames of the
dataset pre- and post-QC respectively.</p>
</td></tr>
<tr><td><code>sample_size</code>, <code>sample_size_HQ</code></td>
<td>
<p>the highest reported sample
size for all SNPs and high-quality SNPs only, respectively.</p>
</td></tr>
<tr><td><code>lambda</code>, <code>lambda_geno</code>, <code>lambda_imp</code></td>
<td>
<p>the lambda values of
all, genotyped and imputed SNPs, respectively.</p>
</td></tr>
<tr><td><code>SNP_N_input</code></td>
<td>
<p>the number of SNPs in the original dataset.</p>
</td></tr>
<tr><td><code>SNP_N_input_monomorphic</code></td>
<td>
<p>the number of SNPs removed
because they are monomorphic.</p>
</td></tr>
<tr><td><code>SNP_N_input_monomorphic_identic_alleles</code></td>
<td>
<p>the subset of
above that had identical alleles, but allele-frequencies
that were not <code>0</code> or <code>1</code>.</p>
</td></tr>
<tr><td><code>SNP_N_input_chr</code></td>
<td>
<p>the number of SNPs removed because they
were X-chromosomal, Y-chromosomal, pseudo-autosomal or
mitochondrial (depends on the remove-arguments). If all
remove arguments were set to <code>FALSE</code>, this returns <code>NA</code>.</p>
</td></tr>
<tr><td><code>SNP_N_preQC</code></td>
<td>
<p>the number of SNPs that entered phase 2b
(i.e. after removal of the monomorphic and excluded-chromosome
SNPs).</p>
</td></tr>
<tr><td><code>SNP_N_preQC_unusable</code></td>
<td>
<p>the number of SNPs removed in phase
2d, due to missing or invalid crucial variables.</p>
</td></tr>
<tr><td><code>SNP_N_preQC_invalid</code></td>
<td>
<p>the number of SNPs with invalid,
non-crucial values in phase 2d.</p>
</td></tr>
<tr><td><code>SNP_N_preQC_min</code></td>
<td>
<p>the number of negative-strand SNPs in
phase 2d.</p>
</td></tr>
<tr><td><code>SNP_N_midQC</code></td>
<td>
<p>the number of SNPs in the dataset during
allele-matching (phase 3).</p>
</td></tr>
<tr><td><code>SNP_N_midQC_min</code></td>
<td>
<p>the number of negative strand SNPs in
phase 3.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_min_std</code>, <code>SNP_N_midQC_min_alt</code>, <code>SNP_N_midQC_min_new</code></td>
<td>

<p>the number of negative strand SNPs matched against,
respectively, the standard allele reference, the alternative
allele reference or neither.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_strandswitch_std</code>, <code>SNP_N_midQC_strandswitch_alt</code></td>
<td>

<p>the number of SNPs that were strand-switched because of
a mismatch with the standard and alternative allele reference,
respectively.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_strandswitch_std_min</code>, <code>SNP_N_midQC_strandswitch_alt_min</code></td>
<td>

<p>the subset of previous that were negative-strand SNPs. NOTE:
at this point in the QC, negative-strand SNPs have already
been converted to the positive strand, i.e. they should
<em>not</em> appear in this category. If they do, there is a
problem with the reported strand, or with the reference table.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_mismatch</code></td>
<td>
<p>the number of SNPs that were still
mismatching after the strand-switch.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_mismatch_std</code>, <code>SNP_N_midQC_mismatch_alt</code></td>
<td>

<p>subset of previous that were matched with the standard and
alternative allele reference, respectively.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_mismatch_std_min</code>, <code>SNP_N_midQC_mismatch_alt_min</code></td>
<td>

<p>subset of previous that are negative-strand SNPs.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_flip_std</code>, <code>SNP_N_midQC_flip_alt</code>, <code>SNP_N_midQC_flip_new</code></td>
<td>

<p>Number of SNPs that were flipped (had their alleles reversed)
when matched against, respectively, the standard allele reference,
the alternative allele reference or neither.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_ambiguous</code></td>
<td>
<p>the number of ambiguous SNPs</p>
</td></tr>
<tr><td><code>SNP_N_midQC_ambiguous_std</code>, <code>SNP_N_midQC_ambiguous_alt</code>, <code>SNP_N_midQC_ambiguous_new</code></td>
<td>
<p>subset of ambiguous SNPs matched
against, respectively, the standard allele reference,
the alternative allele reference or neither.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_suspect</code></td>
<td>
<p>the subset of ambiguous SNPs whose
allele frequencies differ strongly from those in the reference.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_suspect_std</code>, <code>SNP_N_midQC_suspect_alt</code></td>
<td>

<p>the subsets of previous matched against the standard and
alternative allele reference, respectively.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_diffEAF</code></td>
<td>
<p>the number of SNPs whose allele
frequency differs strongly from the reference.</p>
</td></tr>
<tr><td><code>SNP_N_midQC_diffEAF_std</code>, <code>SNP_N_midQC_diffEAF_alt</code></td>
<td>

<p>subset of previous that were matched with the standard and
alternative allele reference, respectively.</p>
</td></tr>
<tr><td><code>SNP_N_postQC</code></td>
<td>
<p>the number of SNPs in the final dataset.</p>
</td></tr>
<tr><td><code>SNP_N_postQC_geno</code>, <code>SNP_N_postQC_imp</code></td>
<td>
<p>the number of
genotyped and imputed SNPs in the final dataset.</p>
</td></tr>
<tr><td><code>SNP_N_postQC_invalid</code></td>
<td>
<p>the number of SNPs with invalid
values remaining in the final dataset. Note: any invalid
values have already been changed to <code>NA</code> at this point.
This merely counts how many of those SNPs are still in the
dataset.</p>
</td></tr>
<tr><td><code>SNP_N_postQC_min</code></td>
<td>
<p>the number of negative-strand SNPs
in the final dataset. Note: all SNPs have been switched to
the positive strand at this point. This merely counts how
many of those SNPs are still in the dataset.</p>
</td></tr>
<tr><td><code>SNP_N_postQC_HQ</code></td>
<td>
<p>the number of high-quality SNPs in the
final dataset.</p>
</td></tr>
<tr><td><code>fixed_HWE</code>, <code>fixed_callrate</code>, <code>fixed_sampleN</code>, <code>fixed_impQ</code></td>
<td>

<p>logical or character string; are HWE p-values, callrates,
sample-size and imputation quality values identical for all
relevant SNPs? If <code>TRUE</code>, it indicates that the parameters
have not been calculated and are dummy values. If a parameter
fails the threshold test, this returns <code>"insuf. data"</code>
(<code>"no data"</code> for sample size).</p>
</td></tr>
<tr><td><code>effect_25</code>, <code>effect_median</code>, <code>effect_75</code></td>
<td>
<p>the quartile values
of the effect-size distribution.</p>
</td></tr>
<tr><td><code>effect_mean</code></td>
<td>
<p>the mean of the effect-size distribution.</p>
</td></tr>
<tr><td><code>SE_median</code>, <code>SE_median_HQ</code></td>
<td>
<p>the median standard error of
all SNPs and high-quality SNPs only, respectively.</p>
</td></tr>
<tr><td><code>skewness</code>, <code>kurtosis</code></td>
<td>
<p>the skewness and kurtosis value of
the dataset.</p>
</td></tr>
<tr><td><code>skewness_HQ</code>, <code>kurtosis_HQ</code></td>
<td>
<p>the skewness and kurtosis value
for high-quality SNPs only.</p>
</td></tr>
<tr><td><code>all_ref_std_name</code>, <code>all_ref_alt_name</code></td>
<td>
<p>the names used for
the standard and alternative allele-references in the output.</p>
</td></tr>
<tr><td><code>all_MAF_std_r</code>, <code>all_MAF_alt_r</code></td>
<td>
<p>allele-frequency
correlation with the standard and alternative allele references.</p>
</td></tr>
<tr><td><code>all_ambiguous_MAF_std_r</code>, <code>all_ambiguous_MAF_alt_r</code></td>
<td>

<p>allele-frequency correlations for the subset of ambiguous
SNPs in the standard and alternative allele references,
respectively.</p>
</td></tr>
<tr><td><code>all_non_ambig_MAF_std_r</code>, <code>all_non_ambig_MAF_alt_r</code></td>
<td>

<p>allele-frequency correlations for the subset of non-ambiguous
SNPs in the standard and alternative allele references,
respectively.</p>
</td></tr>
<tr><td><code>all_ref_changed</code></td>
<td>
<p>logical; has an updated alternative
allele reference been saved?</p>
</td></tr>
<tr><td><code>effectsize_return</code></td>
<td>
<p>logical; is a vector of high-quality
effect-sizes returned in <code>effectsizes_HQ</code>?</p>
</td></tr>
<tr><td><code>effectsizes_HQ</code></td>
<td>
<p>if <code>effectsize_return</code> is <code>TRUE</code>,
a vector of 1000 high-quality effect-sizes; if not, <code>NULL</code>.
If a dataset contains less than 1000 high-quality SNPs, the
vector is padded with NA's to bring it to 1000 values.</p>
</td></tr>
<tr><td><code>pvalue_r</code></td>
<td>
<p>the correlation between reported and calculated
p-values.</p>
</td></tr>
<tr><td><code>visschers_stat</code>, <code>visschers_stat_HQ</code></td>
<td>
<p>the Visscher's statistic
for all SNPs and high-quality SNPs only, respectively.</p>
</td></tr>
<tr><td><code>columns_std_missing</code></td>
<td>
<p>the names of any missing, standard
columns: if no columns are missing, this returns <code>0</code>.</p>
</td></tr>
<tr><td><code>columns_std_empty</code></td>
<td>
<p>the names of any empty, standard
columns: if no columns are empty, this returns <code>0</code>.</p>
</td></tr>
<tr><td><code>columns_unidentified</code></td>
<td>
<p>the names of any unidentified
columns in the input dataset. If none, this returns <code>0</code>.</p>
</td></tr>
<tr><td><code>outcome_useFRQ</code>, <code>outcome_useHWE</code>, <code>outcome_useCal</code>, <code>outcome_useImp</code>, <code>outcome_useMan</code></td>
<td>
<p>logical; indicates whether
the variable passed the threshold test.</p>
</td></tr>
<tr><td><code>...</code></td>
<td>
<p>the remaining 'setting' components return
the the actual filter settings used in the QC (i.e.
taking into account whether the variables passed the
threshold test).</p>
</td></tr>
</table>


<h3>QC output files</h3>

<p>The results of the QC are reported in a variety of .txt and
.png files saved in <code>dir_output</code>. The files use the same
output name as the dataset, with an extension to indicate
their contents (i.e. '_log.txt', '_graph_QQ.png'). The .txt
files are tab-delimited and are best viewed in a spreadsheet
program. The most important one is the log file. This file
summarizes the results of the QC and the data inside the file.
</p>
<p><em>The log file</em>
</p>
<p>The top of the file is table of log entries, reporting any
(potential) problems encountered during the QC. Some of these
are just routine updates; the removal of SNPs with missing
data, for example. However, do check the other entries.
These report important but non-fatal problems, relating to
crucial missing data or invalid data. In such a case, and
provided the QC did not abort, the affected SNPs will have
been exported to a .txt file before being excluded, so the
user can inspect them without having to reload the entire
dataset. The .txt's are:
</p>
<p><code>[filename_output]_SNPs_invalid_allele2.txt</code>
</p>
<p><code>[filename_output]_SNPs_duplicates.txt</code>
</p>
<p><code>[filename_output]_SNPs_removed.txt</code>
</p>
<p><code>[filename_output]_SNPs_improbable_values.txt</code>
</p>
<p>(Note: the names of the files are slightly confusing: the
&quot;SNPs_removed&quot; file contains all SNPs removed in phase 2d.
This does not include monomorphic SNPs, or SNPs from excluded
chromosomes, as these are removed in phase 2a. Also, the
&quot;SNPs_improbable_values&quot; file does not include SNPs with
invalid values for crucial
variables, as these are already in the &quot;SNPs_removed&quot; file.)
</p>
<p>Another important but non-fatal problem is missing columns.
<code>QC_GWAS</code> uses a translation table to determine the
contents of a column. If the translation table is incomplete,
or contains a typo, the function will be unable to translate
(and therefor use) a column. If this involves, say, callrate,
it merely means the function cannot apply the callrate filters,
but the absence of p-values or imputation status will disable
many features of the QC. If you know that a data column is
present, yet the log reports it missing, check the translation
table. The <code>QC_series</code> checklist output and the
<code>columns_unidentified</code> value of the <code>QC_GWAS</code> return
report the names of any unidentified columns in the dataset.
</p>
<p>If the QC aborts, the log file should give some indication why
this occurred. However, if it was successful, there will be
several other tables in the log file.
</p>
<p>The second table reports the number of SNPs in the dataset at
various stages of the QC; as well as how many (and for what
reason) SNPs were removed.
</p>
<p>The third table gives an overview of the data itself. It
reports how many values were missing and invalid per variable
in both the pre- and post-QC datasets, as well as the quartile
and mean values of the post-QC data. A few notes on the nomenclature:
invalid values will have been removed (for crucial variables) or set
to missing (for non-crucial variables) in stage 2d. The post-QC
'invalid' column merely records how many of these SNPs remain in
the dataset. 'Unusable' values are the missing and invalid values
combined (shown as percentage of the total data). Finally,
pre-QC refers to the dataset during stage 2b-c, but
after monomorphic SNPs and SNPs from excluded chromosomes (stage
2a) have been removed.
</p>
<p>The fourth table reports on the allele-matching in phase 3.
The concepts are explained in 'details' and the
<code><a href="#topic+match_alleles">match_alleles</a></code> function; here we just mention
what the user needs to pay attention to. Strand-switching
counts SNPs whose strand was switched because it mismatched
with the reference. As many cohorts do not add strand data
(or set every SNP to <code>"+"</code>), the presence of such SNPs is
not a red flag by itself. However, if there are mismatching
SNPs (the subset of strand-switched SNPs that could not be
fixed), there is a problem with the allele data (or, possibly,
a triallelic SNP). Check the
<code>[filename_output]_SNPs_mismatches-[ref-name].txt</code> file
to see the affected SNPs.
</p>
<p>Another red flag is if there are strand-switched SNPs in a
dataset that <em>also</em> contains negative-strand SNPs (i.e.
the cohort included real strand data, rather just setting it
to <code>"+"</code> for all SNPs). Negative-strand SNPs are converted
to the positive strand beforehand, so they should not appear in
this step (if they do, they are counted in the &quot;double
strand-switch&quot; entry, but that is of minor importance). The real
problem is that if a cohort includes negative-strand SNPs (i.e.
real strand data), and there are still strand-switches, the strand
data must be incorrect. Whether the strand-switches and the
negative-strand SNPs overlap is unimportant.
</p>
<p>The third possible problem is when a large proportion of
ambiguous SNPs is suspect: it indicates that they are on the
wrong strand.
</p>
<p>Finally, a large number of SNPs with a deviating allele
frequency indicates either that the frequency is reported
for the wrong allele (see below) or that the dataset population
does not match that of the reference.
</p>
<p>The fifth table reports the QC outcome statistics. The p-value
and allele-frequency correlations should be close to <code>1</code>.
An allele-frequency correlation of <code>-1</code> means that the
frequency was reported for the wrong (non-effect) allele. As
for the p-value correlaction: in a
typical GWAS dataset, the expected and observed p-values
should correlate perfectly. If this isn't the case, it means
either that a column was misidentified when loading the dataset
or that the wrong values were used when generating the data.
</p>
<p>The sixth table reports how many SNPs were removed by the
various QQ plot filters.
</p>
<p>The seventh table reports the chromosomes and alleles present
in the final dataset.
</p>
<p>The eighth table counts invalid values in the pre-
and post-QC files for several variables. 'Extreme p' is a
value that is only used when <code>calculate_missing_p</code> is
<code>TRUE</code>. Any newly-calculated p-values that are
<code>&lt; 1E-300</code> will be set to <code>1E-300</code>, in order to
exclude any values of <code>0</code> (<code>1E-300</code> is close to the
smallest numeric value that R can handle safely).
</p>
<p>The final four tables contain the settings of the QC.
</p>
<p><em>The QC_series output</em>
</p>
<p><code>QC_series</code> saves a checklist, showing the most
important QC stats of the various files side by side, and
(depending on the plot-arguments) several graphs comparing
effect-size distribution, precision and skewness vs. kurtosis
of the QC'ed files.
</p>


<h3>Output header</h3>

<p>The standard-format values used by <code>out_header</code> are:
</p>

<ul>
<li> <p><code>"standard"</code> retains the default column names
used by <code>QC_GWAS</code>.
</p>
</li>
<li> <p><code>"original"</code> restores the column names used in
the input file.
</p>
</li>
<li> <p><code>"old"</code> uses the default column names of pre-v1.0b
versions of <code>QCGWAS</code>.
</p>
</li>
<li> <p><code>"GWAMA"</code>, <code>"PLINK"</code>, <code>"META"</code> and
<code>"GenABEL"</code> set the
column names to those use by the respective programs. Note
that META's alleleB corresponds to EFFECT_ALL in <code>QC_GWAS</code>.
</p>
</li></ul>



<h3>Requirement for the input dataset</h3>

<p><code>QC_GWAS</code> will automatically
unpack <code>.gz</code> and <code>.zip</code> files, provided the filename
includes the extension of the packed file. For example, if the
data file is named <code>"data1.csv"</code>, the zip file should be
<code>"data1.csv.zip"</code>. If it is named <code>"data1.zip"</code>,
<code>QC_GWAS</code> won't be able to &quot;call&quot; the file inside.
</p>
<p><code>QC_GWAS</code> is flexible when it comes to file-format. By
default, it can open datasets with a variety of column separators
and NA values (the user can specify these via the
<code>column_separators</code> and <code>na.strings</code> arguments).
Read the documentation of the auto-loader function
<code><a href="#topic+load_GWAS">load_GWAS</a></code> for more information.
</p>
<p>Chromosome values can be coded as numeric or character: 
values of <code>"X"</code>, <code>"Y"</code>, <code>"XY"</code> and
<code>"M"</code> will automatically be converted to <code>23</code>,
<code>24</code>, <code>25</code> and <code>26</code>, respectively.
</p>
<p>By default, imputation status is coded as integers <code>0</code>
(genotyped) and <code>1</code> (imputed). As of version 1.0-4,
imputation status will always be translated using the
<code>imputed_T</code>, <code>imputed_F</code> and <code>imputed_NA</code>
arguments. This means that the user must specify values for
these arguments, even when the dataset already uses the
standard format.
Because of the importance of imputation status, if the function
is unable to translate character values, the QC will abort.
</p>
<p>The minimal and maximal valid imputation quality values are
determined by the <code>minimal_impQ_value</code> and
<code>maximal_impQ_value</code> arguments.
</p>
<p>Standard errors and p-values of <code>0</code> are considered invalid and
removed in phase 2d, while values of <code>-1</code> will be set to
<code>NA</code>. Effect sizes of <code>-1</code> are accepted, unless the
standard error and/or the p value are also <code>-1</code>, in which
case it is also set to <code>NA</code>.  
</p>


<h3>Filter arguments</h3>

<p><code>QC_GWAS</code> has three sets arguments relating to filters:
the arguments for the HQ (high-quality), QQ (plot) and NA
(missing values) filters. The HQ and QQ arguments work
mostly in the same way, but there are a few key differences.
</p>
<p>The high-quality arguments accept single, numeric values that
determine the minimal values of allele-frequency (FRQ), HWE
p-value (HWE), callrate (cal) and imputation quality (imp) for
a SNP to be of high-quality. The high-quality filter is used
for the effect-size boxplot and the Manhattan plot, as well
as several QC stats.
</p>
<p>The QQ arguments accept a vector of max. 5 numeric values that
are applied sequentially as filters in the QQ plots.
</p>
<p>Both of these use the NA filter argument(s) to determine
whether to exclude or ignore missing values.
</p>
<p>Neither filter is used to <em>remove</em> SNPs; they merely
exclude them from several QC tests. Both HQ and QQ filter
criteria are only applied if the variable passed the threshold
test, i.e. if there are sufficient non-missing, non-invalid
values for the filter to be applied (see the <code>use_threshold</code>
argument for details). It's pointless to filter an empty column.
</p>
<p>If <code>ignore_impstatus</code> is <code>FALSE</code> (default), the 
imputation-quality criterion is only applied to
imputed SNPs, and the HWE p-value and callrate criteria only
to genotyped SNPs. If <code>TRUE</code>, the filters are applied to
all SNPs, regardless of the imputation status.
</p>
<p>The allele-frequency filters are two-sided:
when set to value x, SNPs with frequency &lt; x or &gt; 1 - x are
excluded.
</p>
<p>To filter missing values only, set the filter to <code>NA</code> and
the corresponding NA filter argument to <code>TRUE</code>. To disable
entirely, set to <code>NULL</code> (this means the NA-filter setting is
ignored as well).
</p>
<p>The differences between the HQ and QQ filters are:
</p>
<p>The HQ filter arguments accept a single value, the QQ filters
can accept up to 5.
</p>
<p>The HQ filter is a single filter: a SNP needs to meet all
relevant criteria to be considered high-quality. The QQ filter
values are applied separately.
</p>
<p>The QQ filter has an additional feature: if passed a value
<code>x &gt; 1</code>, it will calculate a filter value of
<code>x / sample size</code>. This is to allow size-based filtering
of allele frequencies. Note that this filter uses the sample
size listed for that specific SNP. If the sample size is
missing, the relevant NA-filter setting is used to determine
whether it should be excluded.
</p>


<h3>Updating the alternative reference</h3>

<p>There are two drawbacks to the way the function updates the
alternative reference file. One is a technical issue, but the
other can affect the QC of subsequent files.
</p>
<p>Firstly, the argument <code>update_alt</code> has a slightly misleading
name: the alternative-reference <em>file</em> is updated, but
the reference inside the R memory is not. If the user wants to
do further QCs with the updated reference, (s)he will have to
manually reload the updated file into R.
</p>
<p>This is caused by the way R handles data-alterations that occur
inside a function. Any changes made to data last only for the
duration of the function. Once the function terminates, the memory
reverts to its original state. In other words: the allele reference
is updated inside <code>QC_GWAS</code>, but goes back to the pre-QC
state once the QC is finished. <code>QC_series</code> deals with this by
automatically reloading the reference file whenever it is
updated, but, again, once the function terminates it will
revert to its original state.
</p>
<p>The second drawback is that the content of the alternative
reference is arbitrary, depending on which file
an unknown SNP is encountered in first. For example, suppose
that SNP rs31 has alleles A and G, an allele frequency that
varies around 0.5, and does not appear in the standard reference.
When it is added to the alternative reference, the allele
listed as the minor one depends entirely on the allele
frequency in the first file it is encountered in.
</p>
<p>More seriously, if the information in the first file is
incorrect, the SNP may be strand-switched or excluded
in subsequent files because it does not match the (incorrect)
reference. This is another reason why it is important to check
the log files: if there is a problem with a datafile's
strand, alleles or allele-frequency, and the alternative
reference was updated, the incorrect data may have been added
to the reference. If so, one should go back to a previous
reference file. The argument <code>backup_alt</code> is useful for
this, though note that <code>QC_series</code> only does this the
first time the reference is updated.
</p>
<p>Also, if one wants to QC a large number of files for a meta-GWAS,
one should use the same alternative allele reference file (and
let <code>QC_GWAS</code> update it) for every file, otherwise it is
possible  that rs13 may have a different effect alleles in some
post-QC files.
</p>


<h3>See Also</h3>

<p>For the plots created by <code>QC_series</code>:
<code><a href="#topic+plot_distribution">plot_distribution</a></code>,
<code><a href="#topic+plot_precision">plot_precision</a></code> and <code><a href="#topic+plot_skewness">plot_skewness</a></code>.
</p>
<p>For loading and preparing a GWAS dataset:
<code><a href="#topic+load_GWAS">load_GWAS</a></code>, <code><a href="#topic+translate_header">translate_header</a></code>,
<code><a href="#topic+convert_impstatus">convert_impstatus</a></code>.
</p>
<p>For carrying out separate steps of <code>QC_GWAS</code>:
<code><a href="#topic+match_alleles">match_alleles</a></code>, <code><a href="#topic+check_P">check_P</a></code>,
<code><a href="#topic+QC_plots">QC_plots</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## For instructions on how to run QC_GWAS and QC_series
## check the quick start guide in /R/library/QCGWAS/doc
</code></pre>

<hr>
<h2 id='QC_histogram'>Histogram(s) of expected and observed data distribution</h2><span id='topic+QC_histogram'></span><span id='topic+histogram_series'></span>

<h3>Description</h3>

<p><code>QC_histogram</code> creates two histograms: one showing the
observed data distribution of a numeric variable, and one
showing the expected distribution.
It includes the option to filter the data with the
high-quality filter. <code>histogram_series</code> generates a
series of such histograms for multiple filter settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_histogram(dataset, data_col = 1,
             save_name = "dataset", save_dir = getwd(),
             export_outliers = FALSE,
             filter_FRQ = NULL, filter_cal = NULL,
             filter_HWE = NULL, filter_imp = NULL,
             filter_NA = TRUE,
             filter_NA_FRQ = filter_NA, filter_NA_cal = filter_NA,
             filter_NA_HWE = filter_NA, filter_NA_imp = filter_NA,
             breaks = "Sturges",
             graph_name = colnames(dataset)[data_col],
             header_translations, check_impstatus = FALSE,
             ignore_impstatus = FALSE,
             T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
             F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
             NA_strings = c(NA, "NA", ".", "-"), ...)
histogram_series(dataset, data_col = 1,
   save_name = paste0("dataset_F", 1:nrow(plot_table)),
   save_dir = getwd(), export_outliers = FALSE,
   filter_FRQ = NULL, filter_cal = NULL,
   filter_HWE = NULL, filter_imp = NULL,
   filter_NA = TRUE,
   filter_NA_FRQ = filter_NA, filter_NA_cal = filter_NA,
   filter_NA_HWE = filter_NA, filter_NA_imp = filter_NA,
   breaks = "Sturges",
   header_translations, ignore_impstatus = FALSE,
   check_impstatus = FALSE,
   T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
   F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
   NA_strings = c(NA, "NA", ".", "-"),
   ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_histogram_+3A_dataset">dataset</code></td>
<td>
<p>vector or table containing the variable of interest.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_data_col">data_col</code></td>
<td>
<p>name or number of the column of <code>dataset</code>
containing the variable of interest.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_save_name">save_name</code></td>
<td>
<p>for <code>QC_histogram</code>, a character string;
for <code>histogram_series</code>, a vector of character strings;
specifying the filename(s) of the graph, <em>without</em>
extension.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
output files are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses the backslash (\).</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_export_outliers">export_outliers</code></td>
<td>
<p>logical or numeric value; should outlying
entries (which are excluded from the plot) be exported to an
output file? If numeric, the number specifies the max.
number of entries that is exported.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_filter_frq">filter_FRQ</code>, <code id="QC_histogram_+3A_filter_cal">filter_cal</code>, <code id="QC_histogram_+3A_filter_hwe">filter_HWE</code>, <code id="QC_histogram_+3A_filter_imp">filter_imp</code></td>
<td>

<p>Filter threshold-values for allele-frequency, callrate,
HWE p-value and imputation quality, respectively. Passed to
<code><a href="#topic+HQ_filter">HQ_filter</a></code>. <code>QC_histogram</code> takes only
single values, but  <code>histogram_series</code> accepts vectors
as well (see 'details').</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_filter_na">filter_NA</code></td>
<td>
<p>logical; if <code>TRUE</code>, then missing filter
variables will be excluded; if <code>FALSE</code>, they will be
ignored. <code>QC_histogram</code> takes only single values, but 
<code>histogram_series</code> accepts vectors as well (see
'Details'). <code>filter_NA</code> is the default setting for all
variables; variable-specific settings can be specified with
the following arguments.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_filter_na_frq">filter_NA_FRQ</code>, <code id="QC_histogram_+3A_filter_na_cal">filter_NA_cal</code>, <code id="QC_histogram_+3A_filter_na_hwe">filter_NA_HWE</code>, <code id="QC_histogram_+3A_filter_na_imp">filter_NA_imp</code></td>
<td>

<p>logical; variable-specific settings for <code>filter_NA</code>.
These arguments are passed to <code><a href="#topic+HQ_filter">HQ_filter</a></code>.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_breaks">breaks</code></td>
<td>
<p>argument passed to <code><a href="graphics.html#topic+hist">hist</a></code>; determines
the cell-borders in the histogram.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_graph_name">graph_name</code></td>
<td>
<p>character string; used in the title of the plot.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_header_translations">header_translations</code></td>
<td>
<p>translation table for column names.
See <code><a href="#topic+translate_header">translate_header</a></code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_check_impstatus">check_impstatus</code></td>
<td>
<p>logical; should
<code><a href="#topic+convert_impstatus">convert_impstatus</a></code> be called to convert the
imputation-status column into standard values?</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_ignore_impstatus">ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_t_strings">T_strings</code>, <code id="QC_histogram_+3A_f_strings">F_strings</code>, <code id="QC_histogram_+3A_na_strings">NA_strings</code></td>
<td>
<p>arguments passed to
<code><a href="#topic+convert_impstatus">convert_impstatus</a></code>.</p>
</td></tr>
<tr><td><code id="QC_histogram_+3A_...">...</code></td>
<td>
<p>in <code>histogram_series</code>: arguments passed to
<code>QC_histogram</code>; in <code>QC_histogram</code>, arguments passed
to <code><a href="graphics.html#topic+hist">hist</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>histogram_series</code> accepts multiple filter-values, and
passes these one by one to <code>QC_histogram</code> to generate a
series of histograms. For example, specifying:
</p>
<p><code>filter_FRQ = c(0.05, 0.10), filter_cal = c(0.90, 0.95)</code>
</p>
<p>will generate two histograms. The first excludes SNPs with
allele frequency &lt; 0.05 or callrate &lt; 0.90; the second allele 
frequency &lt; 0.10 or callrate &lt; 0.95. The same principle
applies to the <code>NA_filter</code> settings. If the vectors
submitted to the filter arguments are of unequal length, the
shorter vector will be recycled until it equals the length of
the longer (if possible). To filter missing values only, set
the filter to <code>NA</code> and the corresponding NA-filter
argument to <code>TRUE</code>. Setting the filter argument to
<code>NULL</code> will disable the filter entirely, regardless of
the NA filter setting.
</p>


<h3>Value</h3>

<p>Both functions return an invisible value <code>NULL</code>.</p>


<h3>See Also</h3>

<p>For creating QQ plots: <code><a href="#topic+QQ_plot">QQ_plot</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  data("gwa_sample")

  QC_histogram(dataset = gwa_sample, data_col = "EFFECT",
             save_name = "sample_histogram",
             filter_FRQ = 0.01, filter_cal = 0.95,
             filter_NA = FALSE,
             graph_name = "Effect size histogram")

  histogram_series(dataset = gwa_sample, data_col = "EFFECT",
             save_name = "sample_histogram",
             filter_FRQ = c(NA, 0.01, 0.01),
             filter_cal = c(NA, 0.95, 0.95),
             filter_NA = c(FALSE, FALSE, TRUE))

## End(Not run)</code></pre>

<hr>
<h2 id='QC_plots'>QQ and Manhattan plots</h2><span id='topic+QC_plots'></span>

<h3>Description</h3>

<p>This function creates the most important graphs of the QC: the
QQ plots and the Manhattan plot. It also calculates lambda,
and determines the effect of the filters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_plots(dataset,
         plot_QQ = TRUE, plot_Man = TRUE,
         FRQfilter_values = NULL, FRQfilter_NA = filter_NA,
         HWEfilter_values = NULL, HWEfilter_NA = filter_NA,
         calfilter_values = NULL, calfilter_NA = filter_NA,
         impfilter_values = NULL, impfilter_NA = filter_NA,
         impfilter_min = min(dataset$IMP_QUALITY, na.rm = TRUE),
         manfilter_FRQ = NULL, manfilter_HWE = NULL,
         manfilter_cal = NULL, manfilter_imp = NULL,
         filter_NA = TRUE,
         plot_cutoff_p = 0.05, plot_names = FALSE,
         QQ_colors = c("red", "blue", "orange", "green3", "yellow"),
         plot_QQ_bands = FALSE,
         save_name = "dataset", save_dir = getwd(),
         header_translations, use_log = FALSE,
         check_impstatus = FALSE, ignore_impstatus = FALSE,
         T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
         F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
         NA_strings = c(NA, "NA", ".", "-"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_plots_+3A_dataset">dataset</code></td>
<td>
<p>vector of p-values or a data frame containing
the p-value column and (depending on the settings) columns
for chromosome number, position, the quality parameters,
sample size and imputation status.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_plot_qq">plot_QQ</code>, <code id="QC_plots_+3A_plot_man">plot_Man</code></td>
<td>
<p>logical; should QQ and Manhattan plots
be saved?</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_frqfilter_values">FRQfilter_values</code>, <code id="QC_plots_+3A_hwefilter_values">HWEfilter_values</code>, <code id="QC_plots_+3A_calfilter_values">calfilter_values</code>, <code id="QC_plots_+3A_impfilter_values">impfilter_values</code></td>
<td>

<p>numeric vectors; the threshold values for the QQ plot filters.
The filters are for allele-frequency, HWE p-values, callrate
and imputation-quality parameters, respectively. A maximum of
five values can be specified per parameter.
</p>

<ul>
<li><p> Set to <code>NULL</code> to disable the QQ filter for that
parameter.
</p>
</li>
<li><p> To filter missing values only, set the filter value
to <code>NA</code> and the corresponding <code>filter_NA</code>
argument to <code>TRUE</code>.
</p>
</li>
<li><p> The allele-frequency filter is two-sided: for a
filter-value of <code>x</code>, it will exclude entries
with <code>freq &lt; x</code> and <code>freq &gt; 1 - x</code>.
</p>
</li>
<li><p> Values &gt;= 1 will be divided by the SNP's sample size.
This allows sample-size dependent filtering of allele
frequencies. Note that this uses the sample size reported
in the sample-size column for that specific SNP. SNPs
without sample size will be excluded if the corresponding
<code>filter_NA</code> argument is <code>TRUE</code> and ignored if
it is <code>FALSE</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="QC_plots_+3A_frqfilter_na">FRQfilter_NA</code>, <code id="QC_plots_+3A_hwefilter_na">HWEfilter_NA</code>, <code id="QC_plots_+3A_calfilter_na">calfilter_NA</code>, <code id="QC_plots_+3A_impfilter_na">impfilter_NA</code>, <code id="QC_plots_+3A_filter_na">filter_NA</code></td>
<td>

<p>logical; should the filters exclude (<code>TRUE</code>) or
ignore (<code>FALSE</code>) missing values? <code>filter_NA</code>
is the default setting, the others allow
variable-specific settings.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_impfilter_min">impfilter_min</code></td>
<td>
<p>numeric; the lowest possible value for
imputation-quality. This argument is currently redundant, as
it is calculated automatically.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_manfilter_frq">manfilter_FRQ</code>, <code id="QC_plots_+3A_manfilter_hwe">manfilter_HWE</code>, <code id="QC_plots_+3A_manfilter_cal">manfilter_cal</code>, <code id="QC_plots_+3A_manfilter_imp">manfilter_imp</code></td>
<td>

<p>single, numeric values; the filter-settings for allele-frequency,
HWE p-values, callrate and imputation quality respectively,
for the Manhattan plot. The arguments are passed to
<code><a href="#topic+HQ_filter">HQ_filter</a></code>.
To filter missing values only, set to <code>NA</code> and the
corresponding <code>filter_NA</code> argument to <code>TRUE</code>. To
disable filtering entirely, set to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_plot_cutoff_p">plot_cutoff_p</code></td>
<td>
<p>numeric; the threshold of p-values to be
shown in the QQ &amp; Manhattan plots. Higher (less
significant) p-values are excluded from the plot. The default
setting is <code>0.05</code>, which excludes 95% of data-points.
It's not recommended to increase the value above <code>0.05</code>,
as this may dramatically increase running time and memory usage.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_plot_names">plot_names</code></td>
<td>
<p>argument currently redundant.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_qq_colors">QQ_colors</code></td>
<td>
<p>vector of R color-values; the color of the
QQ filter-plots. The unfiltered data is black by default.
This argument sets the colors of the least (first value) to
most (last value) stringent filters. (For this setting,
filter values <code>&gt;= 1</code> (i.e. sample-size based filtering)
are considered less stringent than values <code>&lt; 1</code>.)</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_plot_qq_bands">plot_QQ_bands</code></td>
<td>
<p>logical; should probability bands be
added to the QQ plot?</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_save_name">save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the graphs.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
graphs are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses the backslash (\).</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_header_translations">header_translations</code></td>
<td>
<p>translation table for column names.
See <code><a href="#topic+translate_header">translate_header</a></code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_use_log">use_log</code></td>
<td>
<p>argument used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>;
redundant when <code>QC_plots</code> is used separately.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_check_impstatus">check_impstatus</code></td>
<td>
<p>logical; should the imputation-status
column be passed to <code><a href="#topic+convert_impstatus">convert_impstatus</a></code>?</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_ignore_impstatus">ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td></tr>
<tr><td><code id="QC_plots_+3A_t_strings">T_strings</code>, <code id="QC_plots_+3A_f_strings">F_strings</code>, <code id="QC_plots_+3A_na_strings">NA_strings</code></td>
<td>
<p>arguments passed
to <code><a href="#topic+convert_impstatus">convert_impstatus</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>QC_plots</code> grew out of phase 4 of
<code><a href="#topic+QC_GWAS">QC_GWAS</a></code>. It carries out three functions, hence
the vague name: it calculates  lambda, it applies the
QQ filters, and it creates the QQ and Manhattan plots (a
separate function is available for creating
regional-association plots: see below). The function schematic
is as follows:
</p>

<ul>
<li><p> Preparing the dataset: this step involves translating
the dataset header to the standard column-names (by
<code><a href="#topic+identify_column">identify_column</a></code>) and converting imputation
status (by <code><a href="#topic+convert_impstatus">convert_impstatus</a></code>). Both steps
are optional, and are disabled by default. If the function
cannot identify the imputation status column, it will
generate a warning message and disable the
imputation-status dependent filters.
</p>
</li>
<li><p> Calculating the QC stats: here it generates the
filters an calculates how many SNPs are removed. Lambda
is also calculated at this point.
</p>
</li>
<li><p> Creating a QQ graph of every variable for which
filters have been specified. Every graph contains an
unfiltered plot, plus plots for every effective filter.
(&quot;Effective&quot; means &quot;excludes more SNPs than the
previous, less-stringent filter&quot;.)
</p>
</li>
<li><p> Creating the Manhattan plot. The default Manhattan
plot covers chromosomes 1 to 23 (X). Fields for XY, Y and
M are added when such SNPs are present.
</p>
</li></ul>



<h3>Value</h3>

<p>An object of class 'list' with the following components:
</p>
<table>
<tr><td><code>lambda</code></td>
<td>
<p>vector of the lambda values of all SNPs,
genotyped SNPs and imputed SNPs, respectively.</p>
</td></tr>
<tr><td><code>ignore_impstatus</code></td>
<td>
<p>logical value indicating whether
imputation status was used when applying the filters.</p>
</td></tr>
<tr><td><code>FRQfilter_names</code>, <code>HWEfilter_names</code>, <code>calfilter_names</code>, <code>impfilter_names</code></td>
<td>

<p>character vectors naming the specified QQ filters.</p>
</td></tr>
<tr><td><code>FRQfilter_N</code>, <code>HWEfilter_N</code>, <code>calfilter_N</code>, <code>impfilter_N</code></td>
<td>

<p>numeric vectors; the number of SNPs removed by the specified
filters. Note that the filters are sorted before being
applied, so the order may not match that of the input.
Check the <code>filter_names</code> output to see the order
that was used inside <code>QC_plots</code>.</p>
</td></tr>
<tr><td><code>Manfilter_N</code></td>
<td>
<p>numeric; the number of SNPs removed by the
Manhattan filter. This does not include those SNPs removed
because they lacked p or chromosome/position-values, or failed
the p-cutoff threshold.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>By default, <code>QC_plots</code> expects <code>dataset</code> to use the
standard column-names used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>. A
translation table can be specified in <code>header_translations</code>
to allow non-standard names. See <code><a href="#topic+translate_header">translate_header</a></code>
for more information.
</p>
<p>The function accepts both integer and character chromosome
values. Character values of <code>"X"</code>, <code>"Y"</code>, <code>"XY"</code>
and <code>"M"</code> are automatically converted to integers. By
default, the Manhattan plot shows all autosomal
chromosomes and chromosome X. Fields for Y, XY and M are
added only when such SNPs are present.
</p>
<p>There must be more than 10 p-values at or below the
<code>plot_cutoff_p</code> threshold for the QQ and Manhattan
plots to be created.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_regional">plot_regional</a></code> for creating a regional association
plot.
</p>
<p><code><a href="#topic+check_P">check_P</a></code> for comparing the reported p-values to
the p expected from the effect size and standard error.
</p>
<p><code><a href="#topic+QQ_plot">QQ_plot</a></code> for generating simpler QQ plots.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  data("gwa_sample")

  QC_plots(dataset = gwa_sample,
    plot_QQ = TRUE, plot_QQ_bands = TRUE, plot_Man = TRUE,
    FRQfilter_values = c(NA, 0.01, 0.05, 3),
    calfilter_values = c(NA, 0.95, 0.99),
    manfilter_FRQ = 0.05, manfilter_cal = 0.95,
    filter_NA = TRUE, save_name = "sample_plots")
    
## End(Not run)
</code></pre>

<hr>
<h2 id='QQ_plot'>QQ plot(s) of expected vs. reported p-values</h2><span id='topic+QQ_plot'></span><span id='topic+QQ_series'></span>

<h3>Description</h3>

<p><code>QQ_plot</code> generates a simple QQ plot of the expected and
reported p-value distribution. It includes the option to
filter the data with the high-quality filter. <code>QQ_series</code>
generates a series of such QQ plots for multiple filter
settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QQ_plot(dataset, save_name = "dataset", save_dir = getwd(),
        filter_FRQ = NULL, filter_cal = NULL,
        filter_HWE = NULL, filter_imp = NULL,
        filter_NA = TRUE,
        filter_NA_FRQ = filter_NA, filter_NA_cal = filter_NA,
        filter_NA_HWE = filter_NA, filter_NA_imp = filter_NA,
        p_cutoff = 0.05, plot_QQ_bands = FALSE,
        header_translations,
        check_impstatus = FALSE, ignore_impstatus = FALSE,
        T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
        F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
        NA_strings = c(NA, "NA", ".", "-"), ...)
QQ_series(dataset, save_name = "dataset", save_dir = getwd(),
          filter_FRQ = NULL, filter_cal = NULL,
          filter_HWE = NULL, filter_imp = NULL,
          filter_NA = TRUE,
          filter_NA_FRQ = filter_NA, filter_NA_cal = filter_NA,
          filter_NA_HWE = filter_NA, filter_NA_imp = filter_NA,
          p_cutoff = 0.05, plot_QQ_bands = FALSE,
          header_translations,
          check_impstatus = FALSE, ignore_impstatus = FALSE,
          T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
          F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
          NA_strings = c(NA, "NA", ".", "-"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QQ_plot_+3A_dataset">dataset</code></td>
<td>
<p>a data frame containing
the p-value column and (depending on the settings) columns
for chromosome number, position, the quality parameters,
sample size and imputation status.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_save_name">save_name</code></td>
<td>
<p>for <code>QQ_plot</code>, a character string;
for <code>QQ_series</code>, a vector of character strings;
specifying the filename(s) of the graph, <em>without</em>
extension.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_save_dir">save_dir</code></td>
<td>
<p>character string; the directory where the
output files are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses the backslash (\).</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_filter_frq">filter_FRQ</code>, <code id="QQ_plot_+3A_filter_cal">filter_cal</code>, <code id="QQ_plot_+3A_filter_hwe">filter_HWE</code>, <code id="QQ_plot_+3A_filter_imp">filter_imp</code></td>
<td>

<p>Filter threshold-values for allele-frequency, callrate,
HWE p-value and imputation quality, respectively. Passed to
<code><a href="#topic+HQ_filter">HQ_filter</a></code>. <code>QQ_plot</code> takes only
single values, but  <code>QQ_series</code> accepts vectors
as well (see 'details').</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_filter_na">filter_NA</code></td>
<td>
<p>logical; if <code>TRUE</code>, then missing filter
variables will be excluded; if <code>FALSE</code>, they will be
ignored. <code>QQ_plot</code> takes only single values, but 
<code>QQ_series</code> accepts vectors as well (see
'Details'). <code>filter_NA</code> is the default setting for all
variables; variable-specific settings can be specified with
the following arguments.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_filter_na_frq">filter_NA_FRQ</code>, <code id="QQ_plot_+3A_filter_na_cal">filter_NA_cal</code>, <code id="QQ_plot_+3A_filter_na_hwe">filter_NA_HWE</code>, <code id="QQ_plot_+3A_filter_na_imp">filter_NA_imp</code></td>
<td>

<p>logical; variable-specific settings for <code>filter_NA</code>.
These arguments are passed to <code><a href="#topic+HQ_filter">HQ_filter</a></code>.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>numeric; the threshold of p-values to be
shown in the QQ plot(s). Higher (less significant) p-values
are excluded from the plot. The default setting is <code>0.05</code>,
which excludes 95% of data-points. It's not recommended to
increase the value above <code>0.05</code>, as this may
dramatically increase running time and memory usage.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_plot_qq_bands">plot_QQ_bands</code></td>
<td>
<p>logical; should probability bands be
added to the QQ plot?</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_header_translations">header_translations</code></td>
<td>
<p>translation table for column names.
See <code><a href="#topic+translate_header">translate_header</a></code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_check_impstatus">check_impstatus</code></td>
<td>
<p>logical; should the imputation-status
column be passed to <code><a href="#topic+convert_impstatus">convert_impstatus</a></code>?</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_ignore_impstatus">ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_t_strings">T_strings</code>, <code id="QQ_plot_+3A_f_strings">F_strings</code>, <code id="QQ_plot_+3A_na_strings">NA_strings</code></td>
<td>
<p>arguments passed
to <code><a href="#topic+convert_impstatus">convert_impstatus</a></code>.</p>
</td></tr>
<tr><td><code id="QQ_plot_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>QQ_series</code> accepts multiple filter-values, and
passes these one by one to <code>QQ_plot</code> to generate a
series of plots. For example, specifying:
</p>
<p><code>filter_FRQ = c(0.05, 0.10), filter_cal = c(0.90, 0.95)</code>
</p>
<p>will generate two plots. The first excludes SNPs with
allele frequency &lt; 0.05 or callrate &lt; 0.90; the second allele 
frequency &lt; 0.10 or callrate &lt; 0.95. The same principle
applies to the <code>NA_filter</code> settings. If the vectors
submitted to the filter arguments are of unequal length, the
shorter vector will be recycled until it equals the length of
the longer (if possible). To filter missing values only, set
the filter to <code>NA</code> and the corresponding NA-filter
argument to <code>TRUE</code>. Setting the filter argument to
<code>NULL</code> will disable the filter entirely, regardless of
the NA-filter setting.
</p>


<h3>Value</h3>

<p>Both functions return an invisible value <code>NULL</code>.</p>


<h3>See Also</h3>

<p><code><a href="#topic+QC_plots">QC_plots</a></code> for generating more complex QQ plots
as well as Manhattan plots.
</p>
<p><code><a href="#topic+QC_histogram">QC_histogram</a></code> for creating histograms.
</p>
<p><code><a href="#topic+check_P">check_P</a></code> for comparing the reported p-values to
the p expected from the effect size and standard error.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
    data("gwa_sample")
  
    QQ_plot(dataset = gwa_sample,
            save_name = "sample_QQ",
            filter_FRQ = 0.01, filter_cal = 0.95,
            filter_NA = FALSE)
  
    QQ_series(dataset = gwa_sample,
              save_name = "sample_QQ",
              filter_FRQ = c(NA, 0.01, 0.01),
              filter_cal = c(NA, 0.95, 0.95),
              filter_NA = c(FALSE, FALSE, TRUE))
  
## End(Not run)
</code></pre>

<hr>
<h2 id='save_log'>Log entries for QC_GWAS</h2><span id='topic+save_log'></span>

<h3>Description</h3>

<p>A subroutine for reporting problems encountered while running <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>. It's not designed for use outside of this context, so we recommend that users do not bother with it. It is described here for the sake of completeness only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_log(phaseL, checkL, typeL, SNPL = allSNPs,
         allSNPs = 1L, actionL, noteL = "", fileL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save_log_+3A_phasel">phaseL</code>, <code id="save_log_+3A_checkl">checkL</code>, <code id="save_log_+3A_typel">typeL</code></td>
<td>
<p>character-strings indicating (very briefly) what occurred to trigger a log entry.</p>
</td></tr>
<tr><td><code id="save_log_+3A_snpl">SNPL</code></td>
<td>
<p>integer value indicating the number of SNPs affected.</p>
</td></tr>
<tr><td><code id="save_log_+3A_allsnps">allSNPs</code></td>
<td>
<p>integer value indicating the total number of SNPs in the dataset.</p>
</td></tr>
<tr><td><code id="save_log_+3A_actionl">actionL</code></td>
<td>
<p>character-string describing very briefly the response to situation.</p>
</td></tr>
<tr><td><code id="save_log_+3A_notel">noteL</code></td>
<td>
<p>character-string for longer explanations of what happened.</p>
</td></tr>
<tr><td><code id="save_log_+3A_filel">fileL</code></td>
<td>
<p>character-string of the directory path and the file name,
<em>without</em> file extension, of the log file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>save_log</code> is used to add entries to the log file the moment they are triggered (as opposed to waiting until the QC concludes and then saving the entire log). In case of a fatal crash, the log file should therefor give some indication of where in the QC it occurred.
</p>
<p><code>save_log</code> does not create the log file (this is done by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code>): it merely appends entries to the bottom of the specified file, regardless of what that file is. Hence it is not recommended to use the function outside of <code>QC_GWAS</code>.
</p>


<h3>Value</h3>

<p><code>save_log</code> returns an invisible <code>NULL</code>. The relevant output is the entry added to the log file.</p>

<hr>
<h2 id='switch_strand'>Convert alleles to the opposing DNA strand</h2><span id='topic+switch_strand'></span>

<h3>Description</h3>

<p>This function is a subroutine of <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> and
<code><a href="#topic+match_alleles">match_alleles</a></code>. It converts allele-pairs to
the configuration on the opposing DNA strand.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>switch_strand(input, strand_col = FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="switch_strand_+3A_input">input</code></td>
<td>
<p>table with the alleles in column 1 and 2, and
(optionally) the strand-information (coded as <code>"+"</code> or
<code>"-"</code>) in column 3. Note: the alleles must be uppercase
characters.</p>
</td></tr>
<tr><td><code id="switch_strand_+3A_strand_col">strand_col</code></td>
<td>
<p>logical; if strand-information is present,
this switches the sign in column 3 as well.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table with two or three columns, depending on <code>strand_col</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+match_alleles">match_alleles</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>  data("gwa_sample")

  switched_data &lt;- gwa_sample[ , c("MARKER", "EFFECT_ALL",
                                   "OTHER_ALL", "STRAND")]
  switched_data[1:10, ]

  switched_data[ , 2:4] &lt;- switch_strand(input =
            switched_data[ , 2:4], strand_col = TRUE)
  switched_data[1:10, ]
</code></pre>

<hr>
<h2 id='translate_header'>Translate column names into standard names</h2><span id='topic+translate_header'></span>

<h3>Description</h3>

<p>This function is used to translate non-standard column names
into the standard ones used by <code><a href="#topic+QC_GWAS">QC_GWAS</a></code> and other
functions. It can also translate the standard names into
user-specified names (via the <code>out_header</code> argument of
<code>QC_GWAS</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translate_header(header,
    standard = c("MARKER", "CHR", "POSITION", "EFFECT_ALL",
                 "OTHER_ALL", "STRAND", "EFFECT", "STDERR",
                 "PVALUE", "EFF_ALL_FREQ", "HWE_PVAL",
                 "CALLRATE", "N_TOTAL", "IMPUTED",
                 "USED_FOR_IMP", "IMP_QUALITY"),
    alternative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translate_header_+3A_header">header</code></td>
<td>
<p>character vector; the header to be translated.</p>
</td></tr>
<tr><td><code id="translate_header_+3A_standard">standard</code></td>
<td>
<p>character vector; the names <code>header</code>
should be translated into.</p>
</td></tr>
<tr><td><code id="translate_header_+3A_alternative">alternative</code></td>
<td>
<p>translation table; see 'Details' for more information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In a nutshell: the <code>header</code> argument contains the names you
have; the <code>standard</code> argument contains the names you want; and
<code>alternative</code> is the conversion table.
</p>
<p>The table in <code>alternative</code> should have two columns. The
left column contains the standard names; the right column
possible alternatives. Only one alternative name should be
listed per row. <code>translate_header</code> automatically changes
the contents of <code>header</code> to uppercase, so <code>standard</code>
and the right column of <code>alternative</code> should be uppercase
as well.
</p>
<p>A sample translation table is provided in the package data
folder. It can be loaded via <code>data("header_translations")</code>.
An editable .txt version can be found in the
&quot;R\library\QCGWAS\doc&quot; folder.
</p>


<h3>Value</h3>

<p><code>translate_header</code> returns an object of class 'list' with 6 components:
</p>
<table>
<tr><td><code>header_h</code></td>
<td>
<p>character vector; the translated header.
Unknown columns are included under their old names.</p>
</td></tr>
<tr><td><code>missing_h</code></td>
<td>
<p>character vector; the standard column names
that were not found. If none, this returns <code>NULL</code>.</p>
</td></tr>
<tr><td><code>unknown_h</code></td>
<td>
<p>character vector; column names that could not
be converted to a standard name. Note that these columns
are also included in <code>header_h</code>. If none, this returns
<code>NULL</code>.</p>
</td></tr>
<tr><td><code>header_N</code>, <code>missing_N</code>, <code>unknown_N</code></td>
<td>
<p>integer; the lengths of
the above three vectors</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+header_translations">header_translations</a></code> for a sample translation
table.
</p>
<p><code><a href="#topic+identify_column">identify_column</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  sample_data &lt;-
    data.frame(SNP = paste("rs", 1:10, sep = ""),
               chrom = 2,
               effect = 1:10/10,
               misc = NA,
               stringsAsFactors = FALSE)
  # Creates a table with four columns:
  #   SNP, chrom, effect and misc.

  ( alt_headers &lt;-
      data.frame(
        standard = c("MARKER", "MARKER", "CHR", "CHR"),
        alternative = c("MARKER", "SNP", "CHR", "CHROM"),
        stringsAsFactors = FALSE) )
  # Creates the translation table
  #  with the standard names in column 1 and the alternatives
  #  in column 2.
  
  ( header_info &lt;- 
      translate_header(header = names(sample_data),
        standard = c("MARKER", "CHR", "EFFECT"),
        alternative = alt_headers)  )
        
  # Despite not being in the translation table, EFFECT is
  #  changed to uppercase because it is present in standard.
  #  misc is neither in standard or the translation table, so
  #  it is marked as unknown and left unchanged.

  names(sample_data) &lt;- header_info$header_h
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
