<!DOCTYPE html><html lang="en"><head><title>Help for package qqman</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qqman}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#qqman-package'><p>qqman: Q-Q and Manhattan Plots for GWAS Data</p></a></li>
<li><a href='#gwasResults'><p>Simulated GWAS results</p></a></li>
<li><a href='#manhattan'><p>Creates a manhattan plot</p></a></li>
<li><a href='#qq'><p>Creates a Q-Q plot</p></a></li>
<li><a href='#snpsOfInterest'><p>snpsOfInterest</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Q-Q and Manhattan Plots for GWAS Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.9</td>
</tr>
<tr>
<td>Author:</td>
<td>Stephen Turner &lt;vustephen@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stephen Turner &lt;vustephen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Create Q-Q and manhattan plots for GWAS data from PLINK results.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>calibrate</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/stephenturner/qqman">https://github.com/stephenturner/qqman</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/stephenturner/qqman/issues">https://github.com/stephenturner/qqman/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-23 11:58:41 UTC; turner</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-23 15:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='qqman-package'>qqman: Q-Q and Manhattan Plots for GWAS Data</h2><span id='topic+qqman'></span><span id='topic+qqman-package'></span>

<h3>Description</h3>

<p>Create Q-Q and manhattan plots for GWAS data from PLINK results.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/stephenturner/qqman">https://github.com/stephenturner/qqman</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/stephenturner/qqman/issues">https://github.com/stephenturner/qqman/issues</a>
</p>
</li></ul>


<hr>
<h2 id='gwasResults'>Simulated GWAS results</h2><span id='topic+gwasResults'></span>

<h3>Description</h3>

<p>Simulated GWAS results as obtained from <code>plink --assoc</code>.
</p>

<hr>
<h2 id='manhattan'>Creates a manhattan plot</h2><span id='topic+manhattan'></span>

<h3>Description</h3>

<p>Creates a manhattan plot from PLINK assoc output (or any data frame with 
chromosome, position, and p-value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan(
  x,
  chr = "CHR",
  bp = "BP",
  p = "P",
  snp = "SNP",
  col = c("gray10", "gray60"),
  chrlabs = NULL,
  suggestiveline = -log10(1e-05),
  genomewideline = -log10(5e-08),
  highlight = NULL,
  logp = TRUE,
  annotatePval = NULL,
  annotateTop = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="manhattan_+3A_x">x</code></td>
<td>
<p>A data.frame with columns &quot;BP,&quot; &quot;CHR,&quot; &quot;P,&quot; and optionally, &quot;SNP.&quot;</p>
</td></tr>
<tr><td><code id="manhattan_+3A_chr">chr</code></td>
<td>
<p>A string denoting the column name for the chromosome. Defaults to 
PLINK's &quot;CHR.&quot; Said column must be numeric. If you have X, Y, or MT 
chromosomes, be sure to renumber these 23, 24, 25, etc.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_bp">bp</code></td>
<td>
<p>A string denoting the column name for the chromosomal position. 
Defaults to PLINK's &quot;BP.&quot; Said column must be numeric.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_p">p</code></td>
<td>
<p>A string denoting the column name for the p-value. Defaults to 
PLINK's &quot;P.&quot; Said column must be numeric.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_snp">snp</code></td>
<td>
<p>A string denoting the column name for the SNP name (rs number). 
Defaults to PLINK's &quot;SNP.&quot; Said column should be a character.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_col">col</code></td>
<td>
<p>A character vector indicating which colors to alternate.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_chrlabs">chrlabs</code></td>
<td>
<p>A character vector equal to the number of chromosomes
specifying the chromosome labels (e.g., <code>c(1:22, "X", "Y", "MT")</code>).</p>
</td></tr>
<tr><td><code id="manhattan_+3A_suggestiveline">suggestiveline</code></td>
<td>
<p>Where to draw a &quot;suggestive&quot; line. Default 
-log10(1e-5). Set to FALSE to disable.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_genomewideline">genomewideline</code></td>
<td>
<p>Where to draw a &quot;genome-wide sigificant&quot; line. Default 
-log10(5e-8). Set to FALSE to disable.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of SNPs in your dataset to highlight. 
These SNPs should all be in your dataset.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_logp">logp</code></td>
<td>
<p>If TRUE, the -log10 of the p-value is plotted. It isn't very 
useful to plot raw p-values, but plotting the raw value could be useful for
other genome-wide plots, for example, peak heights, bayes factors, test 
statistics, other &quot;scores,&quot; etc.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_annotatepval">annotatePval</code></td>
<td>
<p>If set, SNPs below this p-value will be annotated on the plot. If logp is FALSE, SNPs above the specified value will be annotated.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_annotatetop">annotateTop</code></td>
<td>
<p>If TRUE, only annotates the top hit on each chromosome that is below the annotatePval threshold (or above if logp is FALSE).</p>
</td></tr>
<tr><td><code id="manhattan_+3A_...">...</code></td>
<td>
<p>Arguments passed on to other plot/points functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A manhattan plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>manhattan(gwasResults)
  
</code></pre>

<hr>
<h2 id='qq'>Creates a Q-Q plot</h2><span id='topic+qq'></span>

<h3>Description</h3>

<p>Creates a quantile-quantile plot from p-values from a GWAS study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qq(pvector, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qq_+3A_pvector">pvector</code></td>
<td>
<p>A numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="qq_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>plot()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Q-Q plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qq(gwasResults$P)

</code></pre>

<hr>
<h2 id='snpsOfInterest'>snpsOfInterest</h2><span id='topic+snpsOfInterest'></span>

<h3>Description</h3>

<p>Example SNPs of interest from simulated <code>gwasResults</code> data.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
