<!DOCTYPE html><html lang="en"><head><title>Help for package QRank</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QRank}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#QRank-package'>
<p>A Novel Quantile Regression Tool for eQTL Discovery</p></a></li>
<li><a href='#heter.QRank'>
<p>Heterogeneity index for &quot;QRank&quot;</p></a></li>
<li><a href='#print.QRank'>
<p>Print a QRank object</p></a></li>
<li><a href='#print.QRank.heter'>
<p>Print a QRank.heter object</p></a></li>
<li><a href='#QRank'>
<p>A new Quantile Rank-score (QRank) based test for the eQTL identification.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Novel Quantile Regression Approach for eQTL Discovery</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-12-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaoyu Song</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaoyu Song &lt;xs2148@cumc.columbia.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A Quantile Rank-score based test for the identification of expression quantitative trait loci.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>quantreg</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-01-11 17:13:44 UTC; Destiny</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-01-11 18:47:57</td>
</tr>
</table>
<hr>
<h2 id='QRank-package'>
A Novel Quantile Regression Tool for eQTL Discovery
</h2><span id='topic+QRank-package'></span>

<h3>Description</h3>

<p>A Quantile Rank-score (QRank) based test for the identification of expression quantitative trait loci (eQTLs). 
</p>


<h3>Details</h3>

<p>We use a Quantile Rank-score (QRank) based test to identify the expression quantitative trait loci (eQTLs) that are associated with the conditional quantile function of gene expressions. 
</p>


<h3>Author(s)</h3>

<p>Xiaoyu Song
</p>
<p>Maintainer: Xiaoyu Song &lt;xs2148@cumc.columbia.edu&gt;
</p>


<h3>References</h3>

<p>Xiaoyu Song, Gen Li, Zhenwei Zhou, Xianling Wang, Iuliana Ionita-Laza and Ying Wei (2016). QRank: A Novel Quantile Regression Tool for eQTL Discovery. <em>Under revision for Bioinformatics.</em> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)                                                                        #
n=300                                                                                #
x=rbinom(n, 2, 0.2)                                                                  #
y=rnorm(n, mean=0, sd=1)                                                             #
z=cbind(rbinom(n, 1, 0.3), rnorm(n, mean=2, sd=2))                                   # 
taus=c( 0.25, 0.5, 0.75)                                                             # 

# - run the proposed QRank approach                                                  #
QRank(gene=y, snp=x, cov=z, tau=taus)                                                #

# - output                                                                           #
#Composite.pvalue:                                                                   #
#[1] 0.2241873                                                                       #

#Quantile.specific.pvalue:                                                           #
#    0.25       0.5      0.75                                                        #
#0.5452044 0.1821452 0.5938421                                                       #
</code></pre>

<hr>
<h2 id='heter.QRank'>
Heterogeneity index for &quot;QRank&quot;
</h2><span id='topic+heter.QRank'></span>

<h3>Description</h3>

<p>Calculate the heterogeneity index of quantile regression coefficients at multiple quantile levels. It measures the variation of the quantile coefficients across quantile levels.
</p>
<p style="text-align: center;"><code class="reqn">heterogeneity=log(sd(\beta)/abs(mean(\beta)))</code>
</p>
 
<p>where <code class="reqn"> \beta </code> is the vector of quantile regression coefficients at multiple quantile levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heter.QRank(object, newtaus=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heter.QRank_+3A_object">object</code></td>
<td>

<p>Object returned from &quot;QRank&quot;
</p>
</td></tr>
<tr><td><code id="heter.QRank_+3A_newtaus">newtaus</code></td>
<td>

<p>a vector of quantile levels based on which heterogeneity index are calculated. Default is NULL, in which case the quantile levels inherited from &quot;QRank&quot; will be used. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>heterogeneity index</code></td>
<td>
<p>one hetergeneity index</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+QRank">QRank</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># continuted from "QRank"

taus=c( 0.25, 0.5, 0.75)
q = QRank(gene=y, snp=x, cov=z, tau=taus)  
heter.QRank(q) # default uses taus inherited from "QRank"
# - output
#Heterogeneity index:
#[1] 2.474184
heter.QRank(q,newtaus = 1:9/10) # calculate based on new taus values
# - output
#Heterogeneity index:
#[1] 2.69242

</code></pre>

<hr>
<h2 id='print.QRank'>
Print a QRank object
</h2><span id='topic+print.QRank'></span>

<h3>Description</h3>

<p>Print the object of QRank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QRank'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.QRank_+3A_x">x</code></td>
<td>

<p>Object returned from QRank.
</p>
</td></tr>
<tr><td><code id="print.QRank_+3A_...">...</code></td>
<td>

<p>Optional arguments
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+QRank">QRank</a></code>
</p>

<hr>
<h2 id='print.QRank.heter'>
Print a QRank.heter object
</h2><span id='topic+print.QRank.heter'></span>

<h3>Description</h3>

<p>Print the object of heter.QRank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QRank.heter'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.QRank.heter_+3A_x">x</code></td>
<td>

<p>Object returned from heter.QRank.
</p>
</td></tr>
<tr><td><code id="print.QRank.heter_+3A_...">...</code></td>
<td>

<p>Optional arguments
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+heter.QRank">heter.QRank</a></code>
</p>

<hr>
<h2 id='QRank'>
A new Quantile Rank-score (QRank) based test for the eQTL identification. 
</h2><span id='topic+QRank'></span>

<h3>Description</h3>

<p>A function to obtain the p-value on the association between a gene expression and a genetic variant based on quantile rank-score test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QRank(gene, snp, cov = NULL, tau)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QRank_+3A_gene">gene</code></td>
<td>

<p>a gene expression level from a selected gene. No parametric assumption is needed for underlying distribution.
</p>
</td></tr>
<tr><td><code id="QRank_+3A_snp">snp</code></td>
<td>

<p>a selected SNP. 
</p>
</td></tr>
<tr><td><code id="QRank_+3A_cov">cov</code></td>
<td>

<p>a vector or matrix of covariates. Default is NULL. 
</p>
</td></tr>
<tr><td><code id="QRank_+3A_tau">tau</code></td>
<td>

<p>the quantile levels to be estimated. Tau can be a single value or a vector of quantile levels. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function conducts Quantile Rank-score (QRank) based test for the continuous traits. It can be used to identify expression quantitative trait loci (eQTLs) that are associated with the conditional quantile functions of gene expression.  
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>composite.pvalue</code></td>
<td>
<p>a single p-value for across all quantile levels under consideration, testing H0:
No genetic association at the selected quantile levels.</p>
</td></tr>
<tr><td><code>quantile.specific.pvalue</code></td>
<td>
<p>p-values of each quantile level, testing <code class="reqn">H_0</code>: The genetic variant and gene expression are not associated at this quantile level.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xiaoyu Song
</p>


<h3>References</h3>

<p>Xiaoyu Song, Gen Li, Zhenwei Zhou, Xianling Wang, Iuliana Ionita-Laza and Ying Wei (2016). QRank: A Novel Quantile Regression Tool for eQTL Discovery. <em>Under revision for Bioinformatics.</em> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)                                                                        #
n=300                                                                                #
x=rbinom(n, 2, 0.2)                                                                  #
y=rnorm(n, mean=0, sd=1)                                                             #
z=cbind(rbinom(n, 1, 0.3), rnorm(n, mean=2, sd=2))                                   # 
taus=c( 0.25, 0.5, 0.75)                                                             # 

# - run the proposed QRank approach                                                  #
QRank(gene=y, snp=x, cov=z, tau=taus)                                                #

# - output                                                                           #
#Composite.pvalue:                                                                   #
#[1] 0.2241873                                                                       #

#Quantile.specific.pvalue:                                                           #
#    0.25       0.5      0.75                                                        #
#0.5452044 0.1821452 0.5938421                                                       #
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
