<!DOCTYPE html><html><head><title>Help for package QTL.gCIMapping.GUI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QTL.gCIMapping.GUI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#QTL.gCIMapping.GUI-package'><p>QTL Genome-Wide Composite Interval Mapping with Graphical User Interface</p></a></li>
<li><a href='#gen'>
<p>genotype example data</p></a></li>
<li><a href='#genf2'>
<p>genotype example data</p></a></li>
<li><a href='#map'>
<p>map example data</p></a></li>
<li><a href='#mapf2'>
<p>map example data</p></a></li>
<li><a href='#markerinsert'><p>To insert marker in genotype.</p></a></li>
<li><a href='#phe'>
<p>phenotype example data</p></a></li>
<li><a href='#phef2'>
<p>phenotype example data</p></a></li>
<li><a href='#WangF'><p>To perform QTL mapping with wang method</p></a></li>
<li><a href='#WangS'><p>The second step of wang method</p></a></li>
<li><a href='#WenF'><p>To perform QTL mapping with Wen method</p></a></li>
<li><a href='#WenS'><p>The second step of Wen method</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>QTL Genome-Wide Composite Interval Mapping with Graphical User
Interface</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, and Zhang Yuan-Ming</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Conduct multiple quantitative trait loci (QTL) mapping under the framework of random-QTL-effect linear mixed model. First, each position on the genome is detected in order to obtain a negative logarithm P-value curve against genome position. Then, all the peaks on each effect (additive or dominant) curve are viewed as potential QTL, all the effects of the potential QTL are included in a multi-QTL model, their effects are estimated by empirical Bayes in doubled haploid population or by adaptive lasso in F2 population, and true QTL are identified by likelihood radio test. See Wen et al. (2018) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbby058">doi:10.1093/bib/bby058</a>&gt;.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),shiny,MASS,qtl</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge;
0.12.17),methods,openxlsx,stringr,data.table,glmnet,doParallel,foreach,QTL.gCIMapping</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-12 03:08:51 UTC; 亚雯</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-12 04:40:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='QTL.gCIMapping.GUI-package'>QTL Genome-Wide Composite Interval Mapping with Graphical User Interface
</h2><span id='topic+QTL.gCIMapping.GUI-package'></span><span id='topic+QTL.gCIMapping.GUI'></span>

<h3>Description</h3>

<p>Conduct multiple quantitative trait loci (QTL) mapping under the framework of random-QTL-effect mixed linear model. First, each position on the genome is detected in order to construct a negative logarithm P-value curve against genome position. Then, all the peaks on each effect (additive or dominant) curve are viewed as potential QTL, all the effects of the potential QTL are included in a multi-QTL model, their effects are estimated by empirical Bayes in doubled haploid or by adaptive lasso in F2, and true QTL are identified by likelihood radio test.</p>


<h3>Usage</h3>

<pre><code class='language-R'>QTL.gCIMapping.GUI()
</code></pre>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> QTL.gCIMapping.GUI</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-10-8</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> shiny,MASS,qtl</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> methods,openxlsx,stringr,Rcpp</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL version 2 or newer</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>References</h3>

<p>An efficient multi-locus mixed model framework for the detection of small and linked QTLs in F2.Wen Yang-Jun, Zhang Ya-Wen, Zhang Jin, Feng Jian-Ying, Jim M. Dunwell, Zhang Yuan-Ming*
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: QTL.gCIMapping.GUI()
</code></pre>

<hr>
<h2 id='gen'>
genotype example data 
</h2><span id='topic+gen'></span><span id='topic+mar'></span><span id='topic+individual'></span>

<h3>Description</h3>

<p>GCIM format of DH genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(gen)
</code></pre>


<h3>Details</h3>

<p>Dataset input of file for WangF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='genf2'>
genotype example data 
</h2><span id='topic+genf2'></span><span id='topic+f2mar'></span><span id='topic+f2individual'></span>

<h3>Description</h3>

<p>GCIM format of F2 genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(genf2)
</code></pre>


<h3>Details</h3>

<p>Dataset input of file for WenF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='map'>
map example data 
</h2><span id='topic+map'></span><span id='topic+marker'></span><span id='topic+chr'></span><span id='topic+pos'></span>

<h3>Description</h3>

<p>GCIM format of DH map dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(map)
</code></pre>


<h3>Details</h3>

<p>Dataset input of file for WangF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='mapf2'>
map example data 
</h2><span id='topic+mapf2'></span><span id='topic+f2marker'></span><span id='topic+f2chr'></span><span id='topic+f2pos'></span>

<h3>Description</h3>

<p>GCIM format of F2 map dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(mapf2)
</code></pre>


<h3>Details</h3>

<p>Dataset input of file for WenF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='markerinsert'>To insert marker in genotype.</h2><span id='topic+markerinsert'></span>

<h3>Description</h3>

<p>a method that can insert marker in genotype.</p>


<h3>Usage</h3>

<pre><code class='language-R'>markerinsert(mp,geno,map,cl,gg1,gg2,gg0,flagRIL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="markerinsert_+3A_mp">mp</code></td>
<td>
<p>linkage map matrix after insert.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_geno">geno</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_map">map</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_cl">cl</code></td>
<td>
<p>walk speed.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_gg1">gg1</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_gg2">gg2</code></td>
<td>
<p>code for type 1.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_gg0">gg0</code></td>
<td>
<p>code for missing.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_flagril">flagRIL</code></td>
<td>
<p>RIL population or not.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mp&lt;-matrix(c(197.9196,198.7536,199.5876,200.4216,201.2453,
202.0691,202.8928,203.7521,204.6113,205.4706,206.3298,207.1891,
1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,3,3,3,3,3,3,
1,1,1,2,2,2,3,3,3,3,3,3,1,2,3,4,5,6,7,8,9,10,11,12),12,5)
map&lt;-matrix(c(1,1,1,1,197.9196,200.4216,202.8928,207.1891),4,2)
geno&lt;-matrix(c(1,99,99,99),1,4)
mark_insert&lt;-QTL.gCIMapping::markerinsert(mp,geno,map,cl=1,gg1=1,gg2=-1,
gg0=99,flagRIL=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='phe'>
phenotype example data 
</h2><span id='topic+phe'></span><span id='topic+posi'></span><span id='topic+trait1'></span><span id='topic+trait2'></span><span id='topic+trait3'></span>

<h3>Description</h3>

<p>GCIM format of DH phenotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(phe)
</code></pre>


<h3>Details</h3>

<p>Dataset input of file for WangF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='phef2'>
phenotype example data 
</h2><span id='topic+phef2'></span><span id='topic+f2posi'></span><span id='topic+f2trait1'></span><span id='topic+f2trait2'></span><span id='topic+f2trait3'></span>

<h3>Description</h3>

<p>GCIM format of F2 phenotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(phef2)
</code></pre>


<h3>Details</h3>

<p>Dataset input of file for WenF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='WangF'>To perform QTL mapping with wang method</h2><span id='topic+WangF'></span>

<h3>Description</h3>

<p>Genome-wide Composite Interval Mapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WangF(pheRaw,genRaw,mapRaw1,yygg1,flagRIL,cov_en,Population,WalkSpeed,CriLOD)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WangF_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_mapraw1">mapRaw1</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_yygg1">yygg1</code></td>
<td>
<p>the transformed covariate matrix .</p>
</td></tr>
<tr><td><code id="WangF_+3A_flagril">flagRIL</code></td>
<td>
<p>if RIL or not.</p>
</td></tr>
<tr><td><code id="WangF_+3A_cov_en">cov_en</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_population">Population</code></td>
<td>
<p>population flag.</p>
</td></tr>
<tr><td><code id="WangF_+3A_walkspeed">WalkSpeed</code></td>
<td>
<p>Walk speed for Genome-wide Scanning.(WalkSpeed=1).</p>
</td></tr>
<tr><td><code id="WangF_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL (CriLOD=2.5).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(gen)
data(phe)
data(map)
wf&lt;-WangF(pheRaw=phe,genRaw=gen,mapRaw1=map,yygg1=NULL,
flagRIL=0,cov_en=NULL,Population="DH",WalkSpeed=1,CriLOD=2.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='WangS'>The second step of wang method</h2><span id='topic+WangS'></span>

<h3>Description</h3>

<p>Genome-wide Composite Interval Mapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WangS(flag,CriLOD,NUM,pheRaw,chrRaw_name,yygg,mx,phe,chr_name,gen,
mapname,CLO)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WangS_+3A_flag">flag</code></td>
<td>
<p>fix or random model.</p>
</td></tr>
<tr><td><code id="WangS_+3A_crilod">CriLOD</code></td>
<td>
<p>LOD score.</p>
</td></tr>
<tr><td><code id="WangS_+3A_num">NUM</code></td>
<td>
<p>The number of trait.</p>
</td></tr>
<tr><td><code id="WangS_+3A_pheraw">pheRaw</code></td>
<td>
<p>Raw phenotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_chrraw_name">chrRaw_name</code></td>
<td>
<p>raw chromosome name.</p>
</td></tr>
<tr><td><code id="WangS_+3A_yygg">yygg</code></td>
<td>
<p>covariate matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_mx">mx</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_chr_name">chr_name</code></td>
<td>
<p>chromosome name.</p>
</td></tr>
<tr><td><code id="WangS_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_mapname">mapname</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_clo">CLO</code></td>
<td>
<p>Number of CPUs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(gen)
data(phe)
data(map)
W1re&lt;-WangF(pheRaw=phe,genRaw=gen,mapRaw1=map,yygg1=NULL,
flagRIL=0,cov_en=NULL,Population="DH",WalkSpeed=1,CriLOD=2.5)
###
ws&lt;-WangS(flag=1,CriLOD=2.5,NUM=1,pheRaw=phe,
chrRaw_name=W1re$chrRaw_name,yygg=W1re$yygg,mx=W1re$mx,
phe=W1re$phe,chr_name=W1re$chr_name,gen=W1re$gen,
mapname=W1re$mapname,CLO=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='WenF'>To perform QTL mapping with Wen method</h2><span id='topic+WenF'></span>

<h3>Description</h3>

<p>An efficient multi-locus mixed model framework for the detection of small and linked QTLs in F2</p>


<h3>Usage</h3>

<pre><code class='language-R'>WenF(pheRaw,genRaw,mapRaw1,yygg1,cov_en,WalkSpeed,CriLOD,dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WenF_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_mapraw1">mapRaw1</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_yygg1">yygg1</code></td>
<td>
<p>the transformed covariate matrix .</p>
</td></tr>
<tr><td><code id="WenF_+3A_cov_en">cov_en</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_walkspeed">WalkSpeed</code></td>
<td>
<p>Walk speed for Genome-wide Scanning.(WalkSpeed=1).</p>
</td></tr>
<tr><td><code id="WenF_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL (CriLOD=2.5).</p>
</td></tr>
<tr><td><code id="WenF_+3A_dir">dir</code></td>
<td>
<p>file path in your computer.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(genf2)
data(phef2)
data(mapf2)
wf&lt;-WenF(pheRaw=phef2,genRaw=genf2,mapRaw1=mapf2,
yygg1=NULL,cov_en=NULL,WalkSpeed=1,CriLOD=2.5,dir=tempdir())

## End(Not run)
</code></pre>

<hr>
<h2 id='WenS'>The second step of Wen method</h2><span id='topic+WenS'></span>

<h3>Description</h3>

<p>An efficient multi-locus mixed model framework for the detection of small and linked QTLs in F2</p>


<h3>Usage</h3>

<pre><code class='language-R'>WenS(flag,CriLOD,NUM,pheRaw,Likelihood,setseed,flagrqtl,yygg,mx,phe,
chr_name,v.map,gen.raw,a.gen.orig,d.gen.orig,n,names.insert2,X.ad.tran.data,X.ad.t4,dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WenS_+3A_flag">flag</code></td>
<td>
<p>random or fix model.</p>
</td></tr>
<tr><td><code id="WenS_+3A_crilod">CriLOD</code></td>
<td>
<p>LOD score.</p>
</td></tr>
<tr><td><code id="WenS_+3A_num">NUM</code></td>
<td>
<p>the number of trait.</p>
</td></tr>
<tr><td><code id="WenS_+3A_pheraw">pheRaw</code></td>
<td>
<p>raw phenotype matrix .</p>
</td></tr>
<tr><td><code id="WenS_+3A_likelihood">Likelihood</code></td>
<td>
<p>likelihood function.</p>
</td></tr>
<tr><td><code id="WenS_+3A_setseed">setseed</code></td>
<td>
<p>random seed set in which,
the cross validation is needed.</p>
</td></tr>
<tr><td><code id="WenS_+3A_flagrqtl">flagrqtl</code></td>
<td>
<p>do CIM or not.</p>
</td></tr>
<tr><td><code id="WenS_+3A_yygg">yygg</code></td>
<td>
<p>covariate matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_mx">mx</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_chr_name">chr_name</code></td>
<td>
<p>chromosome name.</p>
</td></tr>
<tr><td><code id="WenS_+3A_v.map">v.map</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_gen.raw">gen.raw</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_a.gen.orig">a.gen.orig</code></td>
<td>
<p>additive genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_d.gen.orig">d.gen.orig</code></td>
<td>
<p>dominant genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_n">n</code></td>
<td>
<p>number of individual.</p>
</td></tr>
<tr><td><code id="WenS_+3A_names.insert2">names.insert2</code></td>
<td>
<p>linkage map after insert.</p>
</td></tr>
<tr><td><code id="WenS_+3A_x.ad.tran.data">X.ad.tran.data</code></td>
<td>
<p>genotype matrix after insert.</p>
</td></tr>
<tr><td><code id="WenS_+3A_x.ad.t4">X.ad.t4</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_dir">dir</code></td>
<td>
<p>file storage path.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(genf2)
data(phef2)
data(mapf2)
WEN1re&lt;-WenF(pheRaw=phef2,genRaw=genf2,mapRaw1=mapf2,
yygg1=NULL,cov_en=NULL,WalkSpeed=1,CriLOD=2.5,dir=tempdir())
###
ws&lt;-WenS(flag=1,CriLOD=2.5,NUM=1,pheRaw=phef2,
Likelihood="REML",setseed=11001,flagrqtl=FALSE,
yygg=WEN1re$yygg,mx=WEN1re$mx,phe=WEN1re$phe,
chr_name=WEN1re$chr_name,v.map=WEN1re$v.map,
gen.raw=WEN1re$gen.raw,a.gen.orig=WEN1re$a.gen.orig,
d.gen.orig=WEN1re$d.gen.orig,n=WEN1re$n,
names.insert2=WEN1re$names.insert2,
X.ad.tran.data=WEN1re$X.ad.tran.data,
X.ad.t4=WEN1re$X.ad.t4,dir=tempdir())

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
