<!DOCTYPE html><html lang="en"><head><title>Help for package QTL.gCIMapping</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QTL.gCIMapping}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DHdata'>
<p>DH example data</p></a></li>
<li><a href='#Dodata'><p>Process raw data</p></a></li>
<li><a href='#F2data'>
<p>F2 example data from 2 environments</p></a></li>
<li><a href='#markerinsert'><p>To insert marker in genotype.</p></a></li>
<li><a href='#QTL.gCIMapping'><p>QTL Genome-Wide Composite Interval Mapping</p></a></li>
<li><a href='#Readdata'><p>Read raw data</p></a></li>
<li><a href='#WangF'><p>To perform QTL mapping with wang method</p></a></li>
<li><a href='#WangS'><p>The second step of wang method</p></a></li>
<li><a href='#WenF'><p>To perform QTL mapping with Wen method</p></a></li>
<li><a href='#WenS'><p>The second step of Wen method</p></a></li>
<li><a href='#ZhouF'><p>To perform QTL mapping with Wen method</p></a></li>
<li><a href='#ZhouMethod'><p>The second step of Zhou method for multiple environments</p></a></li>
<li><a href='#ZhouMethod_single_env'><p>The second step of Zhou method for single environment</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>QTL Genome-Wide Composite Interval Mapping</td>
</tr>
<tr>
<td>Version:</td>
<td>3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-02-24</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhou Ya-Hui, Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, and Zhang Yuan-Ming</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Conduct multiple quantitative trait loci (QTL) and QTL-by-environment interaction (QEI) mapping via ordinary or compressed variance component mixed models with random- or fixed QTL/QEI effects. First, each position on the genome is detected in order to obtain a negative logarithm P-value curve against genome position. Then, all the peaks on each effect (additive or dominant) curve or on each locus curve are viewed as potential main-effect QTLs and QEIs, all their effects are included in a multi-locus model, their effects are estimated by both least angle regression and empirical Bayes (or adaptive lasso) in backcross and F2 populations, and true QTLs and QEIs are identified by likelihood radio test. See Zhou et al. (2022) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbab596">doi:10.1093/bib/bbab596</a>&gt; and Wen et al. (2018) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbby058">doi:10.1093/bib/bby058</a>&gt;.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge;
0.12.17),methods,openxlsx,readxl,lars,stringr,data.table,glmnet,doParallel,foreach,MASS,qtl</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-24 12:55:13 UTC; win10</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-24 18:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='DHdata'>
DH example data 
</h2><span id='topic+DHdata'></span><span id='topic+marker'></span><span id='topic+chr'></span><span id='topic+pos'></span>

<h3>Description</h3>

<p>GCIM format of DH dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(DHdata)
</code></pre>


<h3>Details</h3>

<p>Input file for WangF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='Dodata'>Process raw data</h2><span id='topic+Dodata'></span>

<h3>Description</h3>

<p>Process raw data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dodata(
  fileFormat = NULL,
  Population = NULL,
  method = NULL,
  Model = NULL,
  readraw = NULL,
  MultiEnv = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Dodata_+3A_fileformat">fileFormat</code></td>
<td>
<p>Format of dataset.</p>
</td></tr>
<tr><td><code id="Dodata_+3A_population">Population</code></td>
<td>
<p>Population type.</p>
</td></tr>
<tr><td><code id="Dodata_+3A_method">method</code></td>
<td>
<p>Method &quot;GCIM&quot; or method &quot;GCIM-QEI&quot;</p>
</td></tr>
<tr><td><code id="Dodata_+3A_model">Model</code></td>
<td>
<p>Random or fixed model.</p>
</td></tr>
<tr><td><code id="Dodata_+3A_readraw">readraw</code></td>
<td>
<p>Raw data.</p>
</td></tr>
<tr><td><code id="Dodata_+3A_multienv">MultiEnv</code></td>
<td>
<p>Whether to perform multi-environment analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
readraw&lt;-Readdata(file=F2data,fileFormat="GCIM",
method="GCIM-QEI",filecov=NULL,
MCIMmap=NULL,MultiEnv=TRUE)
doda&lt;-Dodata(fileFormat="GCIM",Population="F2",
method="GCIM-QEI",Model="Random",
readraw,MultiEnv=TRUE)
</code></pre>

<hr>
<h2 id='F2data'>
F2 example data from 2 environments 
</h2><span id='topic+F2data'></span><span id='topic+F001'></span><span id='topic+F002'></span><span id='topic+F003'></span>

<h3>Description</h3>

<p>GCIM format of F2 dataset whith GCIM-QEI method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(F2data)
</code></pre>


<h3>Details</h3>

<p>Input file for ZhouF function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='markerinsert'>To insert marker in genotype.</h2><span id='topic+markerinsert'></span>

<h3>Description</h3>

<p>a method that can insert marker in genotype.</p>


<h3>Usage</h3>

<pre><code class='language-R'>markerinsert(mp,geno,map,cl,gg1,gg2,gg0,flagRIL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="markerinsert_+3A_mp">mp</code></td>
<td>
<p>linkage map matrix after insert.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_geno">geno</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_map">map</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_cl">cl</code></td>
<td>
<p>walk speed.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_gg1">gg1</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_gg2">gg2</code></td>
<td>
<p>code for type 1.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_gg0">gg0</code></td>
<td>
<p>code for missing.</p>
</td></tr>
<tr><td><code id="markerinsert_+3A_flagril">flagRIL</code></td>
<td>
<p>RIL population or not.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, and Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mp=matrix(c(197.9196,198.7536,199.5876,200.4216,201.2453,
202.0691,202.8928,203.7521,204.6113,205.4706,206.3298,207.1891,
1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,3,3,3,3,3,3,
1,1,1,2,2,2,3,3,3,3,3,3,1,2,3,4,5,6,7,8,9,10,11,12),12,5)
map=matrix(c(1,1,1,1,197.9196,200.4216,202.8928,207.1891),4,2)
geno=matrix(c(1,99,99,99),1,4)
QTL.gCIMapping::markerinsert(mp,geno,map,cl=1,gg1=1,gg2=-1,
gg0=99,flagRIL=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='QTL.gCIMapping'>QTL Genome-Wide Composite Interval Mapping</h2><span id='topic+QTL.gCIMapping'></span>

<h3>Description</h3>

<p>QTL Genome-Wide Composite Interval Mapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QTL.gCIMapping(
  file = NULL,
  fileFormat = "GCIM",
  filecov = NULL,
  MCIMmap = NULL,
  Population = NULL,
  method = "GCIM-QEI",
  MultiEnv = FALSE,
  Model = "Random",
  WalkSpeed = 1,
  CriLOD = 3,
  CriDis = 5,
  Likelihood = "REML",
  SetSeed = 11001,
  flagrqtl = FALSE,
  DrawPlot = TRUE,
  PlotFormat = "jpeg",
  Resolution = "Low",
  Trait = NULL,
  dir = NULL,
  CLO = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QTL.gCIMapping_+3A_file">file</code></td>
<td>
<p>File path and name in your computer.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_fileformat">fileFormat</code></td>
<td>
<p>Format for input file: GCIM, ICIM, Cart, or MCIM.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_filecov">filecov</code></td>
<td>
<p>Covariate file of QTLIciMapping or QTLNetwork.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_mcimmap">MCIMmap</code></td>
<td>
<p>Map file of QTLNetwork.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_population">Population</code></td>
<td>
<p>Population type: F2, BC1, BC2, DH, RIL.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_method">method</code></td>
<td>
<p>Method &quot;GCIM&quot; or method &quot;GCIM-QEI&quot;.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_multienv">MultiEnv</code></td>
<td>
<p>Whether to perform multi-environment analysis.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_model">Model</code></td>
<td>
<p>Random or fixed model.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_walkspeed">WalkSpeed</code></td>
<td>
<p>Walk speed for Genome-wide Scanning.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_cridis">CriDis</code></td>
<td>
<p>The distance of optimization.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_likelihood">Likelihood</code></td>
<td>
<p>This parameter is only for F2 population,including REML (restricted maximum likelihood ) and ML(maximum likelihood).</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_setseed">SetSeed</code></td>
<td>
<p>In which the cross validation experiment is needed. Generally speaking, the random seed in the cross-validation experiment was set as 11001. If some known genes are not identified by the seed, users may try to use some new random seeds. At this case, one better result may be obtained.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_flagrqtl">flagrqtl</code></td>
<td>
<p>This parameter is only for F2 population,   flagrqtl=&quot;FALSE&quot; in the first running. If the other software detects only one QTL in a neighborhood but this software finds two linked QTLs (one with additive effect and another with dominant effect) in the region, let flagrqtl=&quot;TRUE&quot;</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_drawplot">DrawPlot</code></td>
<td>
<p>This parameter is for all the populations, including FALSE and TRUE, DrawPlot=FALSE indicates no figure output, DrawPlot=TRUE indicates the output of the figure against genome position.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_plotformat">PlotFormat</code></td>
<td>
<p>This parameter is for all the figure files, including *.jpeg, *.png, *.tiff and *.pdf.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_resolution">Resolution</code></td>
<td>
<p>This parameter is for all the figure files, including Low and High.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_trait">Trait</code></td>
<td>
<p>Trait=1:3 indicates the analysis from the first trait to the third trait.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_dir">dir</code></td>
<td>
<p>This parameter is for the save path.</p>
</td></tr>
<tr><td><code id="QTL.gCIMapping_+3A_clo">CLO</code></td>
<td>
<p>Number of CPUs.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
QTL.gCIMapping(file=F2data,Population="F2",
MultiEnv=TRUE,Model="Random",CriLOD=3,
Trait=1,dir=tempdir(),CLO=2)
</code></pre>

<hr>
<h2 id='Readdata'>Read raw data</h2><span id='topic+Readdata'></span>

<h3>Description</h3>

<p>Read raw data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Readdata(
  file = NULL,
  fileFormat = NULL,
  method = NULL,
  filecov = NULL,
  MCIMmap = NULL,
  MultiEnv = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Readdata_+3A_file">file</code></td>
<td>
<p>Dataset input</p>
</td></tr>
<tr><td><code id="Readdata_+3A_fileformat">fileFormat</code></td>
<td>
<p>Format of dataset.</p>
</td></tr>
<tr><td><code id="Readdata_+3A_method">method</code></td>
<td>
<p>Method &quot;GCIM&quot; or method &quot;GCIM-QEI&quot;</p>
</td></tr>
<tr><td><code id="Readdata_+3A_filecov">filecov</code></td>
<td>
<p>Covariate file of QTLIciMapping or QTLNetwork.</p>
</td></tr>
<tr><td><code id="Readdata_+3A_mcimmap">MCIMmap</code></td>
<td>
<p>Map file of QTLNetwork.</p>
</td></tr>
<tr><td><code id="Readdata_+3A_multienv">MultiEnv</code></td>
<td>
<p>Whether to perform multi-environment analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
Readdata(file=F2data,fileFormat="GCIM",
method="GCIM-QEI",filecov=NULL,
MCIMmap=NULL,MultiEnv=TRUE)
</code></pre>

<hr>
<h2 id='WangF'>To perform QTL mapping with wang method</h2><span id='topic+WangF'></span>

<h3>Description</h3>

<p>To perform QTL mapping with wang method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WangF(
  pheRaw = NULL,
  genRaw = NULL,
  mapRaw1 = NULL,
  yygg1 = NULL,
  flagRIL = NULL,
  cov_en = NULL,
  Population = NULL,
  WalkSpeed = NULL,
  CriLOD = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WangF_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_mapraw1">mapRaw1</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_yygg1">yygg1</code></td>
<td>
<p>the transformed covariate matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_flagril">flagRIL</code></td>
<td>
<p>if RIL or not.</p>
</td></tr>
<tr><td><code id="WangF_+3A_cov_en">cov_en</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="WangF_+3A_population">Population</code></td>
<td>
<p>population flag.</p>
</td></tr>
<tr><td><code id="WangF_+3A_walkspeed">WalkSpeed</code></td>
<td>
<p>Walk speed for Genome-wide Scanning.</p>
</td></tr>
<tr><td><code id="WangF_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DHdata)
readraw&lt;-Readdata(file=DHdata,fileFormat="GCIM",
method="GCIM",filecov=NULL,MCIMmap=NULL,MultiEnv=FALSE)
DoResult&lt;-Dodata(fileFormat="GCIM",Population="DH",
method="GCIM",Model="Random",readraw,MultiEnv=FALSE)
ws&lt;-WangF(pheRaw=DoResult$pheRaw,genRaw=DoResult$genRaw,
mapRaw1=DoResult$mapRaw1,yygg1=DoResult$yygg1,
flagRIL=DoResult$flagRIL,cov_en=DoResult$cov_en,
Population="DH",WalkSpeed=1,CriLOD=2.5)
</code></pre>

<hr>
<h2 id='WangS'>The second step of wang method</h2><span id='topic+WangS'></span>

<h3>Description</h3>

<p>The second step of wang method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WangS(
  flag = NULL,
  CriLOD = NULL,
  NUM = NULL,
  pheRaw = NULL,
  chrRaw_name = NULL,
  yygg = NULL,
  mx = NULL,
  phe = NULL,
  chr_name = NULL,
  gen = NULL,
  mapname = NULL,
  CLO = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WangS_+3A_flag">flag</code></td>
<td>
<p>fix or random model.</p>
</td></tr>
<tr><td><code id="WangS_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
<tr><td><code id="WangS_+3A_num">NUM</code></td>
<td>
<p>The number of trait.</p>
</td></tr>
<tr><td><code id="WangS_+3A_pheraw">pheRaw</code></td>
<td>
<p>Raw phenotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_chrraw_name">chrRaw_name</code></td>
<td>
<p>raw chromosome name.</p>
</td></tr>
<tr><td><code id="WangS_+3A_yygg">yygg</code></td>
<td>
<p>covariate matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_mx">mx</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_chr_name">chr_name</code></td>
<td>
<p>chromosome name.</p>
</td></tr>
<tr><td><code id="WangS_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_mapname">mapname</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WangS_+3A_clo">CLO</code></td>
<td>
<p>Number of CPUs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DHdata)
readraw&lt;-Readdata(file=DHdata,fileFormat="GCIM",
method="GCIM",filecov=NULL,MCIMmap=NULL,MultiEnv=FALSE)
DoResult&lt;-Dodata(fileFormat="GCIM",Population="DH",
method="GCIM",Model="Random",readraw,MultiEnv=FALSE)
W1re&lt;-WangF(pheRaw=DoResult$pheRaw,genRaw=DoResult$genRaw,
mapRaw1=DoResult$mapRaw1,yygg1=DoResult$yygg1,
flagRIL=DoResult$flagRIL,cov_en=DoResult$cov_en,
Population="DH",WalkSpeed=1,CriLOD=2.5)
ws&lt;-WangS(flag=DoResult$flag,CriLOD=2.5,NUM=1,
pheRaw=DoResult$pheRaw,chrRaw_name=W1re$chrRaw_name,
yygg=W1re$yygg,mx=W1re$mx,phe=W1re$phe,
chr_name=W1re$chr_name,gen=W1re$gen,
mapname=W1re$mapname,CLO=1)
</code></pre>

<hr>
<h2 id='WenF'>To perform QTL mapping with Wen method</h2><span id='topic+WenF'></span>

<h3>Description</h3>

<p>To perform QTL mapping with Wen method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WenF(
  pheRaw = NULL,
  genRaw = NULL,
  mapRaw1 = NULL,
  yygg1 = NULL,
  cov_en = NULL,
  WalkSpeed = NULL,
  CriLOD = NULL,
  dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WenF_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_mapraw1">mapRaw1</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_yygg1">yygg1</code></td>
<td>
<p>the transformed covariate matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_cov_en">cov_en</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="WenF_+3A_walkspeed">WalkSpeed</code></td>
<td>
<p>Walk speed for Genome-wide Scanning.</p>
</td></tr>
<tr><td><code id="WenF_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
<tr><td><code id="WenF_+3A_dir">dir</code></td>
<td>
<p>file path in your computer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
readraw&lt;-Readdata(file=F2data,fileFormat="GCIM",
method="GCIM",filecov=NULL,MCIMmap=NULL,
MultiEnv=FALSE)
DoResult&lt;-Dodata(fileFormat="GCIM",Population="F2",
method="GCIM",Model="Random",readraw,MultiEnv=FALSE)
wf&lt;-WenF(pheRaw=DoResult$pheRaw,
genRaw=DoResult$genRaw,mapRaw1=DoResult$mapRaw1,
yygg1=DoResult$yygg1,cov_en=DoResult$cov_en,
WalkSpeed=1,CriLOD=2.5,dir=tempdir())
</code></pre>

<hr>
<h2 id='WenS'>The second step of Wen method</h2><span id='topic+WenS'></span>

<h3>Description</h3>

<p>The second step of Wen method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WenS(
  flag = NULL,
  CriLOD = NULL,
  NUM = NULL,
  pheRaw = NULL,
  Likelihood = NULL,
  SetSeed = NULL,
  flagrqtl = NULL,
  yygg = NULL,
  mx = NULL,
  phe = NULL,
  chr_name = NULL,
  v.map = NULL,
  gen.raw = NULL,
  a.gen.orig = NULL,
  d.gen.orig = NULL,
  n = NULL,
  names.insert2 = NULL,
  X.ad.tran.data = NULL,
  X.ad.t4 = NULL,
  dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WenS_+3A_flag">flag</code></td>
<td>
<p>random or fix model.</p>
</td></tr>
<tr><td><code id="WenS_+3A_crilod">CriLOD</code></td>
<td>
<p>LOD score.</p>
</td></tr>
<tr><td><code id="WenS_+3A_num">NUM</code></td>
<td>
<p>the number of trait.</p>
</td></tr>
<tr><td><code id="WenS_+3A_pheraw">pheRaw</code></td>
<td>
<p>raw phenotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_likelihood">Likelihood</code></td>
<td>
<p>likelihood function.</p>
</td></tr>
<tr><td><code id="WenS_+3A_setseed">SetSeed</code></td>
<td>
<p>random seed set in which,the cross validation is needed.</p>
</td></tr>
<tr><td><code id="WenS_+3A_flagrqtl">flagrqtl</code></td>
<td>
<p>do CIM or not.</p>
</td></tr>
<tr><td><code id="WenS_+3A_yygg">yygg</code></td>
<td>
<p>covariate matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_mx">mx</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_chr_name">chr_name</code></td>
<td>
<p>chromosome name.</p>
</td></tr>
<tr><td><code id="WenS_+3A_v.map">v.map</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_gen.raw">gen.raw</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_a.gen.orig">a.gen.orig</code></td>
<td>
<p>additive genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_d.gen.orig">d.gen.orig</code></td>
<td>
<p>dominant genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_n">n</code></td>
<td>
<p>number of individual.</p>
</td></tr>
<tr><td><code id="WenS_+3A_names.insert2">names.insert2</code></td>
<td>
<p>linkage map after insert.</p>
</td></tr>
<tr><td><code id="WenS_+3A_x.ad.tran.data">X.ad.tran.data</code></td>
<td>
<p>genotype matrix after insert.</p>
</td></tr>
<tr><td><code id="WenS_+3A_x.ad.t4">X.ad.t4</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="WenS_+3A_dir">dir</code></td>
<td>
<p>file storage path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
readraw&lt;-Readdata(file=F2data,fileFormat="GCIM",
method="GCIM",filecov=NULL,MCIMmap=NULL,MultiEnv=FALSE)
DoResult&lt;-Dodata(fileFormat="GCIM",Population="F2",
method="GCIM",Model="Random",readraw,MultiEnv=FALSE)
WEN1re&lt;-WenF(pheRaw=DoResult$pheRaw,
genRaw=DoResult$genRaw,mapRaw1=DoResult$mapRaw1,
yygg1=DoResult$yygg1,cov_en=DoResult$cov_en,
WalkSpeed=1,CriLOD=2.5,dir=tempdir())
ws&lt;-WenS(flag=DoResult$flag,CriLOD=2.5,NUM=1,
pheRaw=DoResult$pheRaw,Likelihood="REML",
SetSeed=11001,flagrqtl=FALSE,
yygg=WEN1re$yygg,mx=WEN1re$mx,phe=WEN1re$phe,
chr_name=WEN1re$chr_name,v.map=WEN1re$v.map,
gen.raw=WEN1re$gen.raw,
a.gen.orig=WEN1re$a.gen.orig,
d.gen.orig=WEN1re$d.gen.orig,n=WEN1re$n,
names.insert2=WEN1re$names.insert2,
X.ad.tran.data=WEN1re$X.ad.tran.data,
X.ad.t4=WEN1re$X.ad.t4,dir=tempdir())
</code></pre>

<hr>
<h2 id='ZhouF'>To perform QTL mapping with Wen method</h2><span id='topic+ZhouF'></span>

<h3>Description</h3>

<p>To perform QTL mapping with Wen method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZhouF(
  pheRaw = NULL,
  genRaw = NULL,
  mapRaw1 = NULL,
  WalkSpeed = NULL,
  CriLOD = NULL,
  dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZhouF_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouF_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouF_+3A_mapraw1">mapRaw1</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="ZhouF_+3A_walkspeed">WalkSpeed</code></td>
<td>
<p>Walk speed for Genome-wide Scanning.</p>
</td></tr>
<tr><td><code id="ZhouF_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
<tr><td><code id="ZhouF_+3A_dir">dir</code></td>
<td>
<p>file path in your computer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
readraw&lt;-Readdata(file=F2data,fileFormat="GCIM",
method="GCIM-QEI",filecov=NULL,
MCIMmap=NULL,MultiEnv=TRUE)
DoResult&lt;-Dodata(fileFormat="GCIM",
Population="F2",method="GCIM-QEI",
Model="Random",readraw,MultiEnv=TRUE)
ZhouMatrices&lt;-ZhouF(pheRaw=DoResult$pheRaw,
genRaw=DoResult$genRaw,
mapRaw1=DoResult$mapRaw1,
WalkSpeed=1,CriLOD=3,
dir=tempdir())
</code></pre>

<hr>
<h2 id='ZhouMethod'>The second step of Zhou method for multiple environments</h2><span id='topic+ZhouMethod'></span>

<h3>Description</h3>

<p>The second step of Zhou method for multiple environments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZhouMethod(
  Model = NULL,
  pheRaw = NULL,
  genRaw = NULL,
  mapRaw = NULL,
  CriLOD = NULL,
  NUM = NULL,
  EnvNum = NULL,
  yygg = NULL,
  genoname = NULL,
  Ax0 = NULL,
  Hx0 = NULL,
  Bx0 = NULL,
  Ax = NULL,
  Hx = NULL,
  Bx = NULL,
  dir = NULL,
  CriDis = NULL,
  CLO = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZhouMethod_+3A_model">Model</code></td>
<td>
<p>Random or fixed model.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_mapraw">mapRaw</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_num">NUM</code></td>
<td>
<p>The serial number of the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_envnum">EnvNum</code></td>
<td>
<p>The number of environments for each trait is a vector.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_yygg">yygg</code></td>
<td>
<p>covariate matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_genoname">genoname</code></td>
<td>
<p>linkage map matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_ax0">Ax0</code></td>
<td>
<p>AA genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_hx0">Hx0</code></td>
<td>
<p>Aa genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_bx0">Bx0</code></td>
<td>
<p>aa genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_ax">Ax</code></td>
<td>
<p>AA genotype matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_hx">Hx</code></td>
<td>
<p>Aa genotype matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_bx">Bx</code></td>
<td>
<p>aa genotype matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_dir">dir</code></td>
<td>
<p>file storage path.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_cridis">CriDis</code></td>
<td>
<p>The distance of optimization.</p>
</td></tr>
<tr><td><code id="ZhouMethod_+3A_clo">CLO</code></td>
<td>
<p>Number of CPUs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
readraw&lt;-Readdata(file=F2data,fileFormat="GCIM",
method="GCIM-QEI",filecov=NULL,
MCIMmap=NULL,MultiEnv=TRUE)
DoResult&lt;-Dodata(fileFormat="GCIM",
Population="F2",method="GCIM-QEI",
Model="Random",readraw,MultiEnv=TRUE)
ZhouMatrices&lt;-ZhouF(pheRaw=DoResult$pheRaw,
genRaw=DoResult$genRaw,mapRaw1=DoResult$mapRaw1,
WalkSpeed=1,CriLOD=3,dir=tempdir())
OutputZhou&lt;-ZhouMethod(Model="Random",
pheRaw=DoResult$pheRaw,genRaw=DoResult$genRaw,
mapRaw=ZhouMatrices$mapRaw,CriLOD=3,NUM=1,
EnvNum=DoResult$EnvNum,yygg=DoResult$yygg1,
genoname=ZhouMatrices$genoname,
Ax0=ZhouMatrices$Ax0,Hx0=ZhouMatrices$Hx0,
Bx0=ZhouMatrices$Bx0,Ax=ZhouMatrices$Ax,
Hx=ZhouMatrices$Hx,Bx=ZhouMatrices$Bx,
dir=tempdir(),CriDis=5,CLO=2)
</code></pre>

<hr>
<h2 id='ZhouMethod_single_env'>The second step of Zhou method for single environment</h2><span id='topic+ZhouMethod_single_env'></span>

<h3>Description</h3>

<p>The second step of Zhou method for single environment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZhouMethod_single_env(
  Model = NULL,
  pheRaw = NULL,
  genRaw = NULL,
  mapRaw = NULL,
  CriLOD = NULL,
  NUM = NULL,
  yygg = NULL,
  genoname = NULL,
  Ax0 = NULL,
  Hx0 = NULL,
  Bx0 = NULL,
  Ax = NULL,
  Hx = NULL,
  Bx = NULL,
  dir = NULL,
  CriDis = NULL,
  CLO = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZhouMethod_single_env_+3A_model">Model</code></td>
<td>
<p>Random or fixed model.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_pheraw">pheRaw</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_genraw">genRaw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_mapraw">mapRaw</code></td>
<td>
<p>linkage map matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD scores for significant QTL.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_num">NUM</code></td>
<td>
<p>The serial number of the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_yygg">yygg</code></td>
<td>
<p>covariate matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_genoname">genoname</code></td>
<td>
<p>linkage map matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_ax0">Ax0</code></td>
<td>
<p>AA genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_hx0">Hx0</code></td>
<td>
<p>Aa genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_bx0">Bx0</code></td>
<td>
<p>aa genotype matrix.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_ax">Ax</code></td>
<td>
<p>AA genotype matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_hx">Hx</code></td>
<td>
<p>Aa genotype matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_bx">Bx</code></td>
<td>
<p>aa genotype matrix with pseudo markers inserted.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_dir">dir</code></td>
<td>
<p>file storage path.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_cridis">CriDis</code></td>
<td>
<p>The distance of optimization.</p>
</td></tr>
<tr><td><code id="ZhouMethod_single_env_+3A_clo">CLO</code></td>
<td>
<p>Number of CPUs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F2data)
readraw&lt;-Readdata(file=F2data,fileFormat="GCIM",
method="GCIM-QEI",filecov=NULL,
MCIMmap=NULL,MultiEnv=FALSE)
DoResult&lt;-Dodata(fileFormat="GCIM",Population="F2",
method="GCIM-QEI",Model="Random",
readraw,MultiEnv=FALSE)
ZhouMatrices&lt;-ZhouF(pheRaw=DoResult$pheRaw,
genRaw=DoResult$genRaw,mapRaw1=DoResult$mapRaw1,
WalkSpeed=1,CriLOD=3,dir=tempdir())
OutputZhou&lt;-ZhouMethod_single_env(Model="Random",
pheRaw=DoResult$pheRaw,genRaw=DoResult$genRaw,
mapRaw=ZhouMatrices$mapRaw,CriLOD=3,NUM=1,
yygg=DoResult$yygg1,genoname=ZhouMatrices$genoname,
Ax0=ZhouMatrices$Ax0,Hx0=ZhouMatrices$Hx0,
Bx0=ZhouMatrices$Bx0,Ax=ZhouMatrices$Ax,
Hx=ZhouMatrices$Hx,Bx=ZhouMatrices$Bx,
dir=tempdir(),CriDis=5,CLO=2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
