<!DOCTYPE html><html lang="en"><head><title>Help for package qtl2ggplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtl2ggplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#color_patterns_get'><p>Set up col, pattern and group for plotting.</p></a></li>
<li><a href='#color_patterns_pheno'><p>Set up col, pattern, shape and group for plotting.</p></a></li>
<li><a href='#color_patterns_set'><p>Set up colors for patterns or points</p></a></li>
<li><a href='#ggplot_coef'><p>Plot QTL effects along chromosome</p></a></li>
<li><a href='#ggplot_genes'><p>Plot gene locations for a genomic interval</p></a></li>
<li><a href='#ggplot_genes_internal'><p>GGPlot internal routine for ggplot_genes</p></a></li>
<li><a href='#ggplot_listof_scan1coef'><p>Plot of object of class listof_scan1coeff</p></a></li>
<li><a href='#ggplot_onegeno'><p>Plot one individual's genome-wide genotypes</p></a></li>
<li><a href='#ggplot_peaks'><p>Plot QTL peak locations</p></a></li>
<li><a href='#ggplot_pxg'><p>Plot phenotype vs genotype</p></a></li>
<li><a href='#ggplot_scan1'><p>Plot a genome scan</p></a></li>
<li><a href='#ggplot_snpasso'><p>Plot SNP associations</p></a></li>
<li><a href='#listof_scan1coef'><p>List of scan1coef objects</p></a></li>
<li><a href='#sdp_to_pattern'><p>Convert sdp to pattern</p></a></li>
<li><a href='#summary_scan1'><p>Summary of scan1 object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Visualization for QTL Experiments</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions to plot QTL (quantitative trait loci) analysis
    results and related diagnostics.
    Part of 'qtl2', an upgrade of the 'qtl'
    package to better handle high-dimensional data and complex cross
    designs.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.7), assertthat, dplyr, ggplot2, purrr, stringr,
tidyr, rlang, graphics, RColorBrewer, grid, qtl2, ggrepel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>devtools, testthat, roxygen2, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/byandell/qtl2ggplot">https://github.com/byandell/qtl2ggplot</a>, <a href="https://kbroman.org/qtl2/">https://kbroman.org/qtl2/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/byandell/qtl2ggplot/issues">https://github.com/byandell/qtl2ggplot/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-15 15:45:51 UTC; brianyandell</td>
</tr>
<tr>
<td>Author:</td>
<td>Brian S Yandell [aut, cre],
  Karl W Broman [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Brian S Yandell &lt;brian.yandell@wisc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-15 16:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='color_patterns_get'>Set up col, pattern and group for plotting.</h2><span id='topic+color_patterns_get'></span>

<h3>Description</h3>

<p>Set up col, pattern and group for plotting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_patterns_get(scan1ggdata, col, palette = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="color_patterns_get_+3A_scan1ggdata">scan1ggdata</code></td>
<td>
<p>data frame to be used for plotting</p>
</td></tr>
<tr><td><code id="color_patterns_get_+3A_col">col</code></td>
<td>
<p>Color for <code>color</code> column in <code>scan1ggdata</code></p>
</td></tr>
<tr><td><code id="color_patterns_get_+3A_palette">palette</code></td>
<td>
<p>for colors (default <code>NULL</code> uses <code>"Dark2"</code> from <code>RColorBrewer</code> package)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of <code>colors</code> and <code>shapes</code>.
</p>

<hr>
<h2 id='color_patterns_pheno'>Set up col, pattern, shape and group for plotting.</h2><span id='topic+color_patterns_pheno'></span>

<h3>Description</h3>

<p>Set up col, pattern, shape and group for plotting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_patterns_pheno(
  scan1ggdata,
  lod,
  pattern,
  col,
  shape,
  patterns,
  facet = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="color_patterns_pheno_+3A_scan1ggdata">scan1ggdata</code></td>
<td>
<p>data frame to be used for plotting</p>
</td></tr>
<tr><td><code id="color_patterns_pheno_+3A_lod">lod</code></td>
<td>
<p>matrix of LOD scores by position and pheno</p>
</td></tr>
<tr><td><code id="color_patterns_pheno_+3A_pattern">pattern</code></td>
<td>
<p>allele pattern of form <code>AB:CDEFGH</code></p>
</td></tr>
<tr><td><code id="color_patterns_pheno_+3A_col">col</code></td>
<td>
<p>Color for <code>color</code> column in <code>scan1ggdata</code></p>
</td></tr>
<tr><td><code id="color_patterns_pheno_+3A_shape">shape</code></td>
<td>
<p>Shape for <code>shape</code> column in <code>scan1ggdata</code></p>
</td></tr>
<tr><td><code id="color_patterns_pheno_+3A_patterns">patterns</code></td>
<td>
<p>Connect SDP patterns: one of <code>c("none","all","hilit")</code></p>
</td></tr>
<tr><td><code id="color_patterns_pheno_+3A_facet">facet</code></td>
<td>
<p>use <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code> if not <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame <code>scan1ggdata</code> with additional objects.
</p>

<hr>
<h2 id='color_patterns_set'>Set up colors for patterns or points</h2><span id='topic+color_patterns_set'></span>

<h3>Description</h3>

<p>Set up colors for patterns or points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_patterns_set(
  scan1output,
  snpinfo,
  patterns,
  col,
  pattern,
  show_all_snps,
  col_hilit,
  drop_hilit,
  maxlod
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="color_patterns_set_+3A_scan1output">scan1output</code></td>
<td>
<p>output of linear mixed model for <code>phename</code> (see <code><a href="qtl2.html#topic+scan1">scan1</a></code>)</p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_snpinfo">snpinfo</code></td>
<td>
<p>Data frame with snp information</p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_patterns">patterns</code></td>
<td>
<p>Connect SDP patterns: one of <code>c("none","all","hilit")</code>.</p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_col">col</code></td>
<td>
<p>Color of other points, or colors for patterns</p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_pattern">pattern</code></td>
<td>
<p>allele pattern as of form <code>AB:CDEFGH</code></p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_show_all_snps">show_all_snps</code></td>
<td>
<p>show all SNPs if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_col_hilit">col_hilit</code></td>
<td>
<p>Color of highlighted points</p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_drop_hilit">drop_hilit</code></td>
<td>
<p>SNPs with LOD score within this amount of the maximum SNP association will be highlighted.</p>
</td></tr>
<tr><td><code id="color_patterns_set_+3A_maxlod">maxlod</code></td>
<td>
<p>Maximum LOD for drop of <code>drop_hilit</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of <code>col</code> and <code>pattern</code>.
</p>

<hr>
<h2 id='ggplot_coef'>Plot QTL effects along chromosome</h2><span id='topic+ggplot_coef'></span><span id='topic+ggplot_coefCC'></span><span id='topic+autoplot.scan1coef'></span>

<h3>Description</h3>

<p>Plot estimated QTL effects along a chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_coef(
  object,
  map,
  columns = NULL,
  col = NULL,
  scan1_output = NULL,
  gap = 25,
  ylim = NULL,
  bgcolor = "gray90",
  altbgcolor = "gray85",
  ylab = "QTL effects",
  xlim = NULL,
  ...
)

ggplot_coefCC(object, map, colors = qtl2::CCcolors, ...)

## S3 method for class 'scan1coef'
autoplot(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_coef_+3A_object">object</code></td>
<td>
<p>Estimated QTL effects (&quot;coefficients&quot;) as obtained from
<code><a href="qtl2.html#topic+scan1coef">scan1coef</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_map">map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code><a href="qtl2.html#topic+insert_pseudomarkers">insert_pseudomarkers</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_columns">columns</code></td>
<td>
<p>Vector of columns to plot</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_col">col</code></td>
<td>
<p>Vector of colors, same length as <code>columns</code>. If
NULL, some default choices are made.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_scan1_output">scan1_output</code></td>
<td>
<p>If provided, we make a two-panel plot with
coefficients on top and LOD scores below. Should have just one LOD
score column; if multiple, only the first is used.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_gap">gap</code></td>
<td>
<p>Gap between chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_ylim">ylim</code></td>
<td>
<p>y-axis limits. If <code>NULL</code>, we use the range of the plotted
coefficients.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_bgcolor">bgcolor</code></td>
<td>
<p>Background color for the plot.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_altbgcolor">altbgcolor</code></td>
<td>
<p>Background color for alternate chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_ylab">ylab</code></td>
<td>
<p>y-axis label</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_xlim">xlim</code></td>
<td>
<p>x-axis limits. If <code>NULL</code>, we use the range of the plotted
coefficients.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters.</p>
</td></tr>
<tr><td><code id="ggplot_coef_+3A_colors">colors</code></td>
<td>
<p>Colors to use for plotting.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ggplot_coefCC()</code> is the same as <code>ggplot_coef()</code>, but forcing
<code>columns=1:8</code> and using the Collaborative Cross colors,
<code><a href="qtl2.html#topic+CCcolors">CCcolors</a></code>.
</p>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggplot_scan1">ggplot_scan1</a></code>, <code><a href="#topic+ggplot_snpasso">ggplot_snpasso</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data
iron &lt;- qtl2::read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- qtl2::insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- qtl2::calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno[,1]
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)

# calculate coefficients for chromosome 7
coef &lt;- qtl2::scan1coef(probs[,7], pheno, addcovar=covar)

# plot QTL effects
ggplot2::autoplot(coef, map[7], columns=1:3)

</code></pre>

<hr>
<h2 id='ggplot_genes'>Plot gene locations for a genomic interval</h2><span id='topic+ggplot_genes'></span><span id='topic+autoplot.genes'></span>

<h3>Description</h3>

<p>Plot gene locations for a genomic interval, as rectangles with gene
symbol (and arrow indicating strand/direction) below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_genes(
  object,
  xlim = NULL,
  minrow = 4,
  padding = 0.2,
  colors = c("black", "red3", "green4", "blue3", "orange"),
  ...
)

## S3 method for class 'genes'
autoplot(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_genes_+3A_object">object</code></td>
<td>
<p>Object of class <code>object</code></p>
</td></tr>
<tr><td><code id="ggplot_genes_+3A_xlim">xlim</code></td>
<td>
<p>x-axis limits (in Mbp)</p>
</td></tr>
<tr><td><code id="ggplot_genes_+3A_minrow">minrow</code></td>
<td>
<p>Minimum number of rows of object</p>
</td></tr>
<tr><td><code id="ggplot_genes_+3A_padding">padding</code></td>
<td>
<p>Proportion to pad with white space around the object</p>
</td></tr>
<tr><td><code id="ggplot_genes_+3A_colors">colors</code></td>
<td>
<p>Vectors of colors, used sequentially and then re-used.</p>
</td></tr>
<tr><td><code id="ggplot_genes_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filename &lt;- file.path("https://raw.githubusercontent.com/rqtl",
                      "qtl2data/master/DOex", 
                      "c2_genes.rds")
tmpfile &lt;- tempfile()
download.file(filename, tmpfile, quiet=TRUE)
gene_tbl &lt;- readRDS(tmpfile)
unlink(tmpfile)

ggplot_genes(gene_tbl)

</code></pre>

<hr>
<h2 id='ggplot_genes_internal'>GGPlot internal routine for ggplot_genes</h2><span id='topic+ggplot_genes_internal'></span>

<h3>Description</h3>

<p>Plot genes at positions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_genes_internal(
  start,
  end,
  strand,
  rect_top,
  rect_bottom,
  colors,
  space,
  y,
  dir_symbol,
  name,
  xlim,
  xlab = "Position (Mbp)",
  ylab = "",
  bgcolor = "gray92",
  xat = NULL,
  legend.position = "none",
  vlines = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_genes_internal_+3A_start">start</code>, <code id="ggplot_genes_internal_+3A_end">end</code>, <code id="ggplot_genes_internal_+3A_strand">strand</code>, <code id="ggplot_genes_internal_+3A_rect_top">rect_top</code>, <code id="ggplot_genes_internal_+3A_rect_bottom">rect_bottom</code>, <code id="ggplot_genes_internal_+3A_colors">colors</code>, <code id="ggplot_genes_internal_+3A_space">space</code>, <code id="ggplot_genes_internal_+3A_y">y</code>, <code id="ggplot_genes_internal_+3A_dir_symbol">dir_symbol</code>, <code id="ggplot_genes_internal_+3A_name">name</code>, <code id="ggplot_genes_internal_+3A_xlim">xlim</code></td>
<td>
<p>usual parameters</p>
</td></tr>
<tr><td><code id="ggplot_genes_internal_+3A_legend.position">legend.position</code>, <code id="ggplot_genes_internal_+3A_vlines">vlines</code>, <code id="ggplot_genes_internal_+3A_xlab">xlab</code>, <code id="ggplot_genes_internal_+3A_ylab">ylab</code>, <code id="ggplot_genes_internal_+3A_bgcolor">bgcolor</code>, <code id="ggplot_genes_internal_+3A_xat">xat</code></td>
<td>
<p>hidden parameters</p>
</td></tr>
<tr><td><code id="ggplot_genes_internal_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.
</p>

<hr>
<h2 id='ggplot_listof_scan1coef'>Plot of object of class listof_scan1coeff</h2><span id='topic+ggplot_listof_scan1coef'></span><span id='topic+autoplot.listof_scan1coef'></span>

<h3>Description</h3>

<p>Plot object of class <code>listof_scan1coeff</code>, which is a list of objects of class <code>scan1coef</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_listof_scan1coef(
  object,
  map,
  columns = NULL,
  col = NULL,
  scan1_output = NULL,
  facet = "pattern",
  ...
)

## S3 method for class 'listof_scan1coef'
autoplot(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_listof_scan1coef_+3A_object">object</code></td>
<td>
<p>object of class <code>listof_scan1coeff</code></p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_map">map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code><a href="qtl2.html#topic+insert_pseudomarkers">insert_pseudomarkers</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_columns">columns</code></td>
<td>
<p>Vector of columns to plot</p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_col">col</code></td>
<td>
<p>Vector of colors, same length as <code>columns</code>. If
NULL, some default choices are made.</p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_scan1_output">scan1_output</code></td>
<td>
<p>If provided, we make a two-panel plot with
coefficients on top and LOD scores below. Should have just one LOD
score column; if multiple, only the first is used.</p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_facet">facet</code></td>
<td>
<p>Plot facets if multiple phenotypes and group provided (default = <code>"pattern"</code>).</p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_...">...</code></td>
<td>
<p>arguments for <code><a href="#topic+ggplot_coef">ggplot_coef</a></code></p>
</td></tr>
<tr><td><code id="ggplot_listof_scan1coef_+3A_pattern">pattern</code></td>
<td>
<p>Use phenotype names as pattern.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='ggplot_onegeno'>Plot one individual's genome-wide genotypes</h2><span id='topic+ggplot_onegeno'></span>

<h3>Description</h3>

<p>Plot one individual's genome-wide genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_onegeno(
  geno,
  map,
  ind = 1,
  chr = NULL,
  col = NULL,
  shift = FALSE,
  chrwidth = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_onegeno_+3A_geno">geno</code></td>
<td>
<p>Imputed phase-known genotypes, as a list of matrices
(as produced by <code><a href="qtl2.html#topic+maxmarg">maxmarg</a></code>) or a list of
three-dimensional arrays (as produced by <code><a href="qtl2.html#topic+guess_phase">guess_phase</a></code>).</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_map">map</code></td>
<td>
<p>Marker map (a list of vectors of marker positions).</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_ind">ind</code></td>
<td>
<p>Individual to plot, either a numeric index or an ID (can be a vector).</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_chr">chr</code></td>
<td>
<p>Selected chromosomes to plot; a vector of character strings.</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_col">col</code></td>
<td>
<p>Vector of colors for the different genotypes.</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_shift">shift</code></td>
<td>
<p>If TRUE, shift the chromosomes so they all start at 0.</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_chrwidth">chrwidth</code></td>
<td>
<p>Total width of rectangles for each chromosome, as a
fraction of the distance between them.</p>
</td></tr>
<tr><td><code id="ggplot_onegeno_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load qtl2 package for data and genoprob calculation
library(qtl2)

# read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# inferred genotypes
geno &lt;- maxmarg(probs)

# plot the inferred genotypes for the first individual
ggplot_onegeno(geno, map, shift = TRUE)

# plot the inferred genotypes for the first four individuals
ggplot_onegeno(geno, map, ind=1:4)
</code></pre>

<hr>
<h2 id='ggplot_peaks'>Plot QTL peak locations</h2><span id='topic+ggplot_peaks'></span>

<h3>Description</h3>

<p>Plot QTL peak locations (possibly with intervals) for multiple traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_peaks(
  peaks,
  map,
  chr = NULL,
  tick_height = 0.3,
  gap = 25,
  bgcolor = "gray90",
  altbgcolor = "gray85",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_peaks_+3A_peaks">peaks</code></td>
<td>
<p>Data frame such as that produced by
<code><a href="qtl2.html#topic+find_peaks">find_peaks</a></code>) containing columns
<code>chr</code>, <code>pos</code>, <code>lodindex</code>, and <code>lodcolumn</code>.
May also contain columns <code>ci_lo</code> and <code>ci_hi</code>, in
which case intervals will be plotted.</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_map">map</code></td>
<td>
<p>Marker map, used to get chromosome lengths (and start
and end positions).</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_chr">chr</code></td>
<td>
<p>Selected chromosomes to plot; a vector of character
strings.</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_tick_height">tick_height</code></td>
<td>
<p>Height of tick marks at the peaks (a number between 0 and 1).</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_gap">gap</code></td>
<td>
<p>Gap between chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_bgcolor">bgcolor</code></td>
<td>
<p>Background color for the plot.</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_altbgcolor">altbgcolor</code></td>
<td>
<p>Background color for alternate chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_peaks_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>See Also</h3>

<p><code><a href="qtl2.html#topic+find_peaks">find_peaks</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load qtl2 package for data and genoprob calculation
library(qtl2)

# read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# find peaks above lod=3.5 (and calculate 1.5-LOD support intervals)
peaks &lt;- find_peaks(out, map, threshold=3.5, drop=1.5)

# color peaks above 6 red; only show chromosomes with peaks
plot_peaks(peaks, map)
peaks$col &lt;- (peaks$lod &gt; 6)

ggplot_peaks(peaks, map[names(map) %in% peaks$chr], col = c("blue","red"),
           legend.title = "LOD &gt; 6")
</code></pre>

<hr>
<h2 id='ggplot_pxg'>Plot phenotype vs genotype</h2><span id='topic+ggplot_pxg'></span><span id='topic+mean_pxg'></span>

<h3>Description</h3>

<p>Plot phenotype vs genotype for a single putative QTL and a single phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_pxg(
  geno,
  pheno,
  sort = TRUE,
  SEmult = NULL,
  pooledSD = TRUE,
  jitter = 0.2,
  bgcolor = "gray90",
  seg_width = 0.4,
  seg_lwd = 2,
  seg_col = "black",
  hlines = NULL,
  hlines_col = "white",
  hlines_lty = 1,
  hlines_lwd = 1,
  vlines_col = "gray80",
  vlines_lty = 1,
  vlines_lwd = 3,
  force_labels = TRUE,
  alternate_labels = FALSE,
  omit_points = FALSE,
  ...
)

mean_pxg(geno, pheno, dataframe = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_pxg_+3A_geno">geno</code></td>
<td>
<p>Vector of genotypes, as produced by
<code><a href="qtl2.html#topic+maxmarg">maxmarg</a></code> with specific <code>chr</code> and
<code>pos</code>.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_pheno">pheno</code></td>
<td>
<p>Vector of phenotypes.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_sort">sort</code></td>
<td>
<p>If TRUE, sort genotypes from largest to smallest.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_semult">SEmult</code></td>
<td>
<p>If specified, interval estimates of the within-group
averages will be displayed, as <code>mean +/- SE * SEmult</code>.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_pooledsd">pooledSD</code></td>
<td>
<p>If TRUE and <code>SEmult</code> is specified, calculated
a pooled within-group SD. Otherwise, get separate estimates of
the within-group SD for each group.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_jitter">jitter</code></td>
<td>
<p>Amount to jitter the points horizontally, if a vector
of length &gt; 0, it is taken to be the actual jitter amounts
(with values between -0.5 and 0.5).</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_bgcolor">bgcolor</code></td>
<td>
<p>Background color for the plot.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_seg_width">seg_width</code></td>
<td>
<p>Width of segments at the estimated within-group averages</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_seg_lwd">seg_lwd</code></td>
<td>
<p>Line width used to plot estimated within-group averages</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_seg_col">seg_col</code></td>
<td>
<p>Line color used to plot estimated within-group averages</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_hlines">hlines</code></td>
<td>
<p>Locations of horizontal grid lines.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_hlines_col">hlines_col</code></td>
<td>
<p>Color of horizontal grid lines</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_hlines_lty">hlines_lty</code></td>
<td>
<p>Line type of horizontal grid lines</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_hlines_lwd">hlines_lwd</code></td>
<td>
<p>Line width of horizontal grid lines</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_vlines_col">vlines_col</code></td>
<td>
<p>Color of vertical grid lines</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_vlines_lty">vlines_lty</code></td>
<td>
<p>Line type of vertical grid lines</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_vlines_lwd">vlines_lwd</code></td>
<td>
<p>Line width of vertical grid lines</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_force_labels">force_labels</code></td>
<td>
<p>If TRUE, force all genotype labels to be shown.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_alternate_labels">alternate_labels</code></td>
<td>
<p>If TRUE, place genotype labels in two rows</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_omit_points">omit_points</code></td>
<td>
<p>If TRUE, omit the points, just plotting the averages (and, potentially, the +/- SE intervals).</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters, passed to <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_pxg_+3A_dataframe">dataframe</code></td>
<td>
<p>Supplied data frame, or constructed from <code>geno</code> and <code>pheno</code> if <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="qtl2.html#topic+plot_coef">plot_coef</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load qtl2 package for data and genoprob calculation
library(qtl2)

# read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# inferred genotype at a 28.6 cM on chr 16
geno &lt;- maxmarg(probs, map, chr=16, pos=28.6, return_char=TRUE)

# plot phenotype vs genotype
ggplot_pxg(geno, log10(iron$pheno[,1]), ylab=expression(log[10](Liver)))

# include +/- 2 SE intervals
ggplot_pxg(geno, log10(iron$pheno[,1]), ylab=expression(log[10](Liver)),
         SEmult=2)

# plot just the means
ggplot_pxg(geno, log10(iron$pheno[,1]), ylab=expression(log[10](Liver)),
         omit_points=TRUE)

# plot just the means +/- 2 SEs
ggplot_pxg(geno, log10(iron$pheno[,1]), ylab=expression(log[10](Liver)),
         omit_points=TRUE, SEmult=2)
</code></pre>

<hr>
<h2 id='ggplot_scan1'>Plot a genome scan</h2><span id='topic+ggplot_scan1'></span><span id='topic+autoplot.scan1'></span><span id='topic+ggplot_scan1_internal'></span>

<h3>Description</h3>

<p>Plot LOD curves for a genome scan
</p>
<p>Plot LOD curves for a genome scan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_scan1(
  object,
  map,
  lodcolumn = 1,
  chr = NULL,
  gap = 25,
  bgcolor = "gray90",
  altbgcolor = "gray85",
  ...
)

## S3 method for class 'scan1'
autoplot(object, ...)

ggplot_scan1_internal(
  map,
  lod,
  gap = 25,
  col = NULL,
  shape = NULL,
  pattern = NULL,
  facet = NULL,
  patterns = c("none", "all", "hilit"),
  chrName = "Chr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_scan1_+3A_object">object</code></td>
<td>
<p>Output of <code><a href="qtl2.html#topic+scan1">scan1</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_map">map</code></td>
<td>
<p>Map of pseudomarker locations.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>LOD score column to plot (a numeric index, or a
character string for a column name). One or more value(s) allowed.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_chr">chr</code></td>
<td>
<p>Selected chromosomes to plot; a vector of character
strings.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_gap">gap</code></td>
<td>
<p>Gap between chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_bgcolor">bgcolor</code></td>
<td>
<p>Background color for the plot.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_altbgcolor">altbgcolor</code></td>
<td>
<p>Background color for alternate chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_lod">lod</code></td>
<td>
<p>Matrix of lod (or other) values.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_col">col</code></td>
<td>
<p>Colors for points or lines, with labels.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_shape">shape</code></td>
<td>
<p>Shapes for points.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_pattern">pattern</code></td>
<td>
<p>Use to group values for plotting (default = <code>NULL</code>); typically provided by <code><a href="qtl2.html#topic+plot_snpasso">plot_snpasso</a></code> internal routine.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_facet">facet</code></td>
<td>
<p>Plot facets if multiple phenotypes and group provided (default = <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_patterns">patterns</code></td>
<td>
<p>Connect SDP patterns: one of <code>c("none","all","hilit")</code>.</p>
</td></tr>
<tr><td><code id="ggplot_scan1_+3A_chrname">chrName</code></td>
<td>
<p>Add prefix chromosome name (default <code>"Chr"</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggplot_coef">ggplot_coef</a></code>, <code><a href="#topic+ggplot_snpasso">ggplot_snpasso</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load qtl2 package for data and genoprob calculation
library(qtl2)

# read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# plot the results for selected chromosomes
chr &lt;- c(2,7,8,9,15,16)
ggplot_scan1(out, map, lodcolumn=1:2, chr=chr, col=c("darkslateblue","violetred"),
     legend.position=c(0.1,0.9))

# plot just one chromosome
ggplot_scan1(out, map, chr=8, lodcolumn=1:2, col=c("darkblue","violetred"))

# can also use autoplot from ggplot2
# lodcolumn can also be a column name
library(ggplot2)
autoplot(out, map, chr=8, lodcolumn=c("liver","spleen"), col=c("darkblue","violetred"))
</code></pre>

<hr>
<h2 id='ggplot_snpasso'>Plot SNP associations</h2><span id='topic+ggplot_snpasso'></span>

<h3>Description</h3>

<p>Plot SNP associations, with possible expansion from distinct snps to all snps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_snpasso(
  scan1output,
  snpinfo,
  genes = NULL,
  lodcolumn = 1,
  show_all_snps = TRUE,
  drop_hilit = NA,
  col_hilit = "violetred",
  col = "darkslateblue",
  ylim = NULL,
  gap = 25,
  minlod = 0,
  bgcolor = "gray90",
  altbgcolor = "gray85",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_snpasso_+3A_scan1output">scan1output</code></td>
<td>
<p>Output of <code><a href="qtl2.html#topic+scan1">scan1</a></code>.  Should
contain an attribute, <code>"snpinfo"</code>, as when
<code><a href="qtl2.html#topic+scan1">scan1</a></code> are run with SNP probabilities
produced by <code><a href="qtl2.html#topic+genoprob_to_snpprob">genoprob_to_snpprob</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_snpinfo">snpinfo</code></td>
<td>
<p>Data frame with SNP information with the following
columns (the last three are generally derived from with
<code><a href="qtl2.html#topic+index_snps">index_snps</a></code>):
</p>

<ul>
<li> <p><code>chr</code> - Character string or factor with chromosome
</p>
</li>
<li> <p><code>pos</code> - Position (in same units as in the <code>"map"</code>
attribute in <code>genoprobs</code>.
</p>
</li>
<li> <p><code>sdp</code> - Strain distribution pattern: an integer, between
1 and <code class="reqn">2^n - 2</code> where <code class="reqn">n</code> is the number of strains, whose
binary encoding indicates the founder genotypes
</p>
</li>
<li> <p><code>snp</code> - Character string with SNP identifier (if
missing, the rownames are used).
</p>
</li>
<li> <p><code>index</code> - Indices that indicate equivalent
groups of SNPs.
</p>
</li>
<li> <p><code>intervals</code> - Indexes that indicate which marker
intervals the SNPs reside.
</p>
</li>
<li> <p><code>on_map</code> - Indicate whether SNP coincides with a marker
in the <code>genoprobs</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_genes">genes</code></td>
<td>
<p>Optional data frame containing gene information for
the region, with columns 'start' and 'stop' in Mbp, 'strand'
(as '&quot;-&quot;', '&quot;+&quot;', or 'NA'), and 'Name'. If included, a
two-panel plot is produced, with SNP associations above and
gene locations below.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>LOD score column to plot (a numeric index, or a
character string for a column name). One or more value(s) allowed.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_show_all_snps">show_all_snps</code></td>
<td>
<p>If TRUE, expand to show all SNPs.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_drop_hilit">drop_hilit</code></td>
<td>
<p>SNPs with LOD score within this amount of the maximum SNP association will be highlighted.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_col_hilit">col_hilit</code></td>
<td>
<p>Color of highlighted points</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_col">col</code></td>
<td>
<p>Color of other points</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_ylim">ylim</code></td>
<td>
<p>y-axis limits</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_gap">gap</code></td>
<td>
<p>Gap between chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_minlod">minlod</code></td>
<td>
<p>Minimum LOD to display. (Mostly for GWAS, in which
case using 'minlod=1' will greatly increase the plotting speed,
since the vast majority of points would be omittted.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_bgcolor">bgcolor</code></td>
<td>
<p>Background color for the plot.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_altbgcolor">altbgcolor</code></td>
<td>
<p>Background color for alternate chromosomes.</p>
</td></tr>
<tr><td><code id="ggplot_snpasso_+3A_...">...</code></td>
<td>
<p>Additional graphics parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.
</p>


<h3>Hidden graphics parameters</h3>

<p>A number of graphics parameters can be passed via '...'. For
example, 'bgcolor' to control the background color and 'altbgcolor'
to control the background color on alternate chromosomes.
'cex' for character expansion for the points (default 0.5),
'pch' for the plotting character for the points (default 16),
and 'ylim' for y-axis limits.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggplot_scan1">ggplot_scan1</a></code>, <code><a href="#topic+ggplot_coef">ggplot_coef</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")

# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to SNP probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)

# Scan SNPs.
scan_snppr &lt;- qtl2::scan1(snppr, DOex$pheno)

# plot results
ggplot_snpasso(scan_snppr, snpinfo, show_all_snps=FALSE, patterns="all", drop_hilit=1.5)


# can also just type autoplot() if ggplot2 attached
library(ggplot2)

# plot just subset of distinct SNPs
autoplot(scan_snppr, snpinfo, show_all_snps=FALSE, drop_hilit=1.5)

# highlight SDP patterns in SNPs; connect with lines.
autoplot(scan_snppr, snpinfo, patterns="all",drop_hilit=4)

# query function for finding genes in region
gene_dbfile &lt;- system.file("extdata", "mouse_genes_small.sqlite", package="qtl2")
query_genes &lt;- qtl2::create_gene_query_func(gene_dbfile)
genes &lt;- query_genes(2, 97, 98)

# plot SNP association results with gene locations
autoplot(scan_snppr, snpinfo, patterns="hilit", drop_hilit=1.5, genes=genes)


</code></pre>

<hr>
<h2 id='listof_scan1coef'>List of scan1coef objects</h2><span id='topic+listof_scan1coef'></span><span id='topic+summary_listof_scan1coef'></span><span id='topic+summary.listof_scan1coef'></span><span id='topic+summary_scan1coef'></span><span id='topic+summary.scan1coef'></span><span id='topic+subset_listof_scan1coef'></span><span id='topic+subset.listof_scan1coef'></span><span id='topic++5B.listof_scan1coef'></span>

<h3>Description</h3>

<p>Create a list of scan1coef objects using <code><a href="qtl2.html#topic+scan1coef">scan1coef</a></code>.
</p>
<p>Summary of object of class <code><a href="#topic+listof_scan1coef">listof_scan1coef</a></code>, which is a list of objects of class <code>scan1coef</code>.
</p>
<p>Summary of object of class <code><a href="#topic+listof_scan1coef">listof_scan1coef</a></code>, which is a list of objects of class <code>scan1coef</code>.
</p>
<p>Subset of object of class <code><a href="#topic+listof_scan1coef">listof_scan1coef</a></code>, which is a list of objects of class <code>scan1coef</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listof_scan1coef(
  probs,
  phe,
  K = NULL,
  covar = NULL,
  blups = FALSE,
  center = FALSE,
  ...
)

summary_listof_scan1coef(
  object,
  scan1_object,
  map,
  coef_names = dimnames(object[[1]])[[2]],
  center = TRUE,
  ...
)

## S3 method for class 'listof_scan1coef'
summary(object, ...)

summary_scan1coef(object, scan1_object, map, ...)

## S3 method for class 'scan1coef'
summary(object, ...)

subset_listof_scan1coef(x, elements, ...)

## S3 method for class 'listof_scan1coef'
subset(x, ...)

## S3 method for class 'listof_scan1coef'
x[...]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="listof_scan1coef_+3A_probs">probs</code></td>
<td>
<p>genotype probabilities object for one chromosome from <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code></p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_phe">phe</code></td>
<td>
<p>data frame of phenotypes</p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_k">K</code></td>
<td>
<p>list of length 1 with kinship matrix</p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_covar">covar</code></td>
<td>
<p>matrix of covariates</p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_blups">blups</code></td>
<td>
<p>Create BLUPs if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_center">center</code></td>
<td>
<p>center coefficients if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_object">object</code></td>
<td>
<p>object of class <code>listof_scan1coef</code></p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_scan1_object">scan1_object</code></td>
<td>
<p>object from <code>scan1</code></p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_map">map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code><a href="qtl2.html#topic+insert_pseudomarkers">insert_pseudomarkers</a></code>.</p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_coef_names">coef_names</code></td>
<td>
<p>names of effect coefficients (default is all coefficient names)</p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_x">x</code></td>
<td>
<p>object of class <code>listof_scan1coef</code></p>
</td></tr>
<tr><td><code id="listof_scan1coef_+3A_elements">elements</code></td>
<td>
<p>indexes or names of list elements in x</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>listof_scan1coef</code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data
iron &lt;- qtl2::read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- qtl2::insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- qtl2::calc_genoprob(iron, map, error_prob=0.002)

# Ensure that covariates have names attribute
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)

# Calculate scan1coef on all phenotypes,
# returning a list of \code{\link{scan1coef}} objects
out &lt;- listof_scan1coef(probs[,7], iron$pheno, addcovar = covar, center = TRUE)

# Plot coefficients for all phenotypes
ggplot2::autoplot(out, map[7], columns = 1:3)

# Summary of coefficients at scan peak
scan_pr &lt;- qtl2::scan1(probs[,7], iron$pheno)
summary(out, scan_pr, map[7])

</code></pre>

<hr>
<h2 id='sdp_to_pattern'>Convert sdp to pattern</h2><span id='topic+sdp_to_pattern'></span>

<h3>Description</h3>

<p>Convert strain distribution pattern (sdp) to letter pattern.
Taken from package 'qtl2pattern' for internal use here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sdp_to_pattern(sdp, haplos, symmetric = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sdp_to_pattern_+3A_sdp">sdp</code></td>
<td>
<p>vector of sdp values</p>
</td></tr>
<tr><td><code id="sdp_to_pattern_+3A_haplos">haplos</code></td>
<td>
<p>letter codes for haplotypes (required)</p>
</td></tr>
<tr><td><code id="sdp_to_pattern_+3A_symmetric">symmetric</code></td>
<td>
<p>make patterns symmetric if <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of letter patterns
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='summary_scan1'>Summary of scan1 object</h2><span id='topic+summary_scan1'></span><span id='topic+summary.scan1'></span>

<h3>Description</h3>

<p>Summary of scan1 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_scan1(
  object,
  map,
  snpinfo = NULL,
  lodcolumn = seq_len(ncol(object)),
  chr = names(map),
  sum_type = c("common", "best"),
  drop = 1.5,
  show_all_snps = TRUE,
  ...
)

## S3 method for class 'scan1'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary_scan1_+3A_object">object</code></td>
<td>
<p>object from <code><a href="qtl2.html#topic+scan1">scan1</a></code></p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_map">map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code><a href="qtl2.html#topic+insert_pseudomarkers">insert_pseudomarkers</a></code>.</p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_snpinfo">snpinfo</code></td>
<td>
<p>Data frame with SNP information with the following
columns (the last three are generally derived from with
<code><a href="qtl2.html#topic+index_snps">index_snps</a></code>):
</p>

<ul>
<li> <p><code>chr</code> - Character string or factor with chromosome
</p>
</li>
<li> <p><code>pos</code> - Position (in same units as in the <code>"map"</code>
attribute in <code>genoprobs</code>.
</p>
</li>
<li> <p><code>sdp</code> - Strain distribution pattern: an integer, between
1 and <code class="reqn">2^n - 2</code> where <code class="reqn">n</code> is the number of strains, whose
binary encoding indicates the founder genotypes
</p>
</li>
<li> <p><code>snp</code> - Character string with SNP identifier (if
missing, the rownames are used).
</p>
</li>
<li> <p><code>index</code> - Indices that indicate equivalent
groups of SNPs.
</p>
</li>
<li> <p><code>intervals</code> - Indexes that indicate which marker
intervals the SNPs reside.
</p>
</li>
<li> <p><code>on_map</code> - Indicate whether SNP coincides with a marker
in the <code>genoprobs</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="summary_scan1_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>one or more lod columns</p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_chr">chr</code></td>
<td>
<p>one or more chromosome IDs</p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_sum_type">sum_type</code></td>
<td>
<p>type of summary</p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_drop">drop</code></td>
<td>
<p>LOD drop from maximum</p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_show_all_snps">show_all_snps</code></td>
<td>
<p>show all SNPs if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="summary_scan1_+3A_...">...</code></td>
<td>
<p>other arguments not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl summary
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data
iron &lt;- qtl2::read_cross2(system.file("extdata", "iron.zip", package="qtl2"))
# insert pseudomarkers into map
map &lt;- qtl2::insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- qtl2::calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- qtl2::get_x_covar(iron)

# perform genome scan
out &lt;- qtl2::scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# summary
summary(out, map)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
