<!DOCTYPE html><html lang="en"><head><title>Help for package qtl2fst</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtl2fst}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_genoprob_fst'><p>Calculate conditional genotype probabilities and write to fst database</p></a></li>
<li><a href='#cbind.fst_genoprob'><p>Join genotype probabilities for different chromosomes</p></a></li>
<li><a href='#fst_extract'><p>Extract genotype probabilities from fst database</p></a></li>
<li><a href='#fst_files'><p>List files used in fst_genoprob object</p></a></li>
<li><a href='#fst_genoprob'><p>Store genotype probabilities in fst database</p></a></li>
<li><a href='#fst_path'><p>Path used in fst_genoprob object</p></a></li>
<li><a href='#fst_restore'><p>Restore fst_genoprob object to original dimensions.</p></a></li>
<li><a href='#genoprob_to_alleleprob_fst'><p>Convert genotype probabilities to allele probabilities and write to fst database</p></a></li>
<li><a href='#rbind.fst_genoprob'><p>Join genotype probabilities for different individuals</p></a></li>
<li><a href='#replace_path'><p>Replace the path used in fst_genoprob object</p></a></li>
<li><a href='#subset_fst_genoprob'><p>Subsetting genotype probabilities</p></a></li>
<li><a href='#summary.fst_genoprob'><p>Summary of an fst_genoprob object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.30</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-11-23</td>
</tr>
<tr>
<td>Title:</td>
<td>Database Storage of Genotype Probabilities for QTL Mapping</td>
</tr>
<tr>
<td>Description:</td>
<td>Uses the 'fst' package to store genotype probabilities on disk for the 'qtl2' package. These genotype probabilities are a central data object for mapping quantitative trait loci (QTL), but they can be quite large. The facilities in this package enable the genotype probabilities to be stored on disk, leading to reduced memory usage with only a modest increase in computation time.</td>
</tr>
<tr>
<td>Author:</td>
<td>Karl W Broman <a href="https://orcid.org/0000-0002-4914-6671"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Brian S Yandell <a href="https://orcid.org/0000-0002-8774-9377"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Petr Simecek <a href="https://orcid.org/0000-0002-2922-7183"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karl W Broman &lt;broman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>fst, qtl2 (&ge; 0.24)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/rqtl/qtl2fst">https://github.com/rqtl/qtl2fst</a></td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-23 19:55:42 UTC; kbroman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-23 21:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_genoprob_fst'>Calculate conditional genotype probabilities and write to fst database</h2><span id='topic+calc_genoprob_fst'></span>

<h3>Description</h3>

<p>Uses a hidden Markov model to calculate the probabilities of the
true underlying genotypes given the observed multipoint marker
data, with possible allowance for genotyping errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_genoprob_fst(
  cross,
  fbase,
  fdir = ".",
  map = NULL,
  error_prob = 0.0001,
  map_function = c("haldane", "kosambi", "c-f", "morgan"),
  lowmem = FALSE,
  quiet = TRUE,
  cores = 1,
  compress = 0,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_genoprob_fst_+3A_cross">cross</code></td>
<td>
<p>Object of class <code>"cross2"</code>. For details, see the
<a href="https://kbroman.org/qtl2/assets/vignettes/developer_guide.html">R/qtl2 developer guide</a>.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_fbase">fbase</code></td>
<td>
<p>Base of filename for fst database.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_fdir">fdir</code></td>
<td>
<p>Directory for fst database.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_map">map</code></td>
<td>
<p>Genetic map of markers. May include pseudomarker
locations (that is, locations that are not within the marker
genotype data). If NULL, the genetic map in <code>cross</code> is used.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_error_prob">error_prob</code></td>
<td>
<p>Assumed genotyping error probability</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_map_function">map_function</code></td>
<td>
<p>Character string indicating the map function to
use to convert genetic distances to recombination fractions.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_lowmem">lowmem</code></td>
<td>
<p>If <code>FALSE</code>, split individuals into groups with
common sex and crossinfo and then precalculate the transition
matrices for a chromosome; potentially a lot faster but using more
memory.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_quiet">quiet</code></td>
<td>
<p>If <code>FALSE</code>, print progress messages.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_cores">cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code><a href="parallel.html#topic+makeCluster">parallel::makeCluster()</a></code>.</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_compress">compress</code></td>
<td>
<p>Amount of compression to use (value in the range 0-100; lower values mean larger file sizes)</p>
</td></tr>
<tr><td><code id="calc_genoprob_fst_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE (the default), refuse to overwrite any files that already exist.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is like calling <code>qtl2::calc_genoprob()</code> and then
<code>fst_genoprob()</code>, but in a way that hopefully saves memory by
doing it one chromosome at a time.
</p>


<h3>Value</h3>

<p>A list containing the attributes of <code>genoprob</code>
and the address for the created fst database.
Components are:
</p>

<ul>
<li> <p><code>dim</code> - List of all dimensions of 3-D arrays.
</p>
</li>
<li> <p><code>dimnames</code> - List of all dimension names of 3-D arrays.
</p>
</li>
<li> <p><code>is_x_chr</code> - Vector of all is_x_chr attributes.
</p>
</li>
<li> <p><code>chr</code> - Vector of (subset of) chromosome names for this object.
</p>
</li>
<li> <p><code>ind</code> - Vector of (subset of) individual names for this object.
</p>
</li>
<li> <p><code>mar</code> - Vector of (subset of) marker names for this object.
</p>
</li>
<li> <p><code>fst</code> - Path and base of file names for the fst database.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="qtl2.html#topic+calc_genoprob">qtl2::calc_genoprob()</a></code>, <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
gmap_w_pmar &lt;- insert_pseudomarkers(grav2$gmap, step=1)
fst_dir &lt;- file.path(tempdir(), "grav2_genoprob")
dir.create(fst_dir)
probs_fst &lt;- calc_genoprob_fst(grav2, "grav2", fst_dir, gmap_w_pmar, error_prob=0.002)

# clean up: remove all the files we created
unlink(fst_files(probs_fst))
</code></pre>

<hr>
<h2 id='cbind.fst_genoprob'>Join genotype probabilities for different chromosomes</h2><span id='topic+cbind.fst_genoprob'></span>

<h3>Description</h3>

<p>Join multiple genotype probability objects, as produced by
<code><a href="#topic+fst_genoprob">fst_genoprob()</a></code> for different individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fst_genoprob'
cbind(..., fbase = NULL, fdir = NULL, overwrite = FALSE, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cbind.fst_genoprob_+3A_...">...</code></td>
<td>
<p>Genotype probability objects as produced by
<code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>. Must have the same set of individuals.</p>
</td></tr>
<tr><td><code id="cbind.fst_genoprob_+3A_fbase">fbase</code></td>
<td>
<p>Base of fileame for fst database.
Needed if objects have different fst databases.</p>
</td></tr>
<tr><td><code id="cbind.fst_genoprob_+3A_fdir">fdir</code></td>
<td>
<p>Directory for fst database.</p>
</td></tr>
<tr><td><code id="cbind.fst_genoprob_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE (the default), refuse to overwrite existing <code>.fst</code> files.</p>
</td></tr>
<tr><td><code id="cbind.fst_genoprob_+3A_quiet">quiet</code></td>
<td>
<p>If TRUE, don't show any messages. Passed to <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single genotype probability object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rbind.fst_genoprob">rbind.fst_genoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
map &lt;- insert_pseudomarkers(grav2$gmap, step=1)
probsA &lt;- calc_genoprob(grav2[1:5,1:2], map, error_prob=0.002)
probsB &lt;- calc_genoprob(grav2[1:5,3:4], map, error_prob=0.002)
dir &lt;- tempdir()
fprobsA &lt;- fst_genoprob(probsA, "exampleAc", dir, overwrite=TRUE)
fprobsB &lt;- fst_genoprob(probsB, "exampleBc", dir, overwrite=TRUE)

# use cbind to combine probabilities for same individuals but different chromosomes
fprobs &lt;- cbind(fprobsA, fprobsB, fbase = "exampleABc", overwrite=TRUE)

# clean up: remove all the files we created
unlink(fst_files(fprobsA))
unlink(fst_files(fprobsB))
unlink(fst_files(fprobs))

</code></pre>

<hr>
<h2 id='fst_extract'>Extract genotype probabilities from fst database</h2><span id='topic+fst_extract'></span><span id='topic+fst2calc_genoprob'></span>

<h3>Description</h3>

<p>Extract genotype probabilities from fst database as an ordinary calc_genoprob object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst_extract(object)

fst2calc_genoprob(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fst_extract_+3A_object">object</code></td>
<td>
<p>Object of class <code>"fst_genoprob"</code>, linking to an fst database of genotype probabilities.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genotype probabilities are extracted from the fst database. Each chromosome is extracted in turn.
</p>


<h3>Value</h3>

<p>An object of class <code>"calc_genoprob"</code> (a list of 3-dimensional arrays).
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>fst2calc_genoprob</code>: Deprecated version (to be deleted)
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
map &lt;- insert_pseudomarkers(grav2$gmap, step=1)
probs &lt;- calc_genoprob(grav2, map, error_prob=0.002)
dir &lt;- tempdir()
fprobs &lt;- fst_genoprob(probs, "grav2", dir, overwrite=TRUE)
nprobs &lt;- fst_extract(fprobs)

# clean up: remove all the files we created
unlink(fst_files(fprobs))
</code></pre>

<hr>
<h2 id='fst_files'>List files used in fst_genoprob object</h2><span id='topic+fst_files'></span>

<h3>Description</h3>

<p>List all of the files used in an fst_genoprob object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst_files(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fst_files_+3A_object">object</code></td>
<td>
<p>An object of class <code>"fst_genoprob"</code> as created by <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of character strings with the full paths for all of the files used for the input <code>object</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fst_path">fst_path()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
probs &lt;- calc_genoprob(grav2, error_prob=0.002)
dir &lt;- tempdir()
fprobs &lt;- fst_genoprob(probs, "grav2", dir, overwrite=TRUE)

fst_path(fprobs)
fst_files(fprobs)

# clean up: remove all the files we created
unlink(fst_files(fprobs))
</code></pre>

<hr>
<h2 id='fst_genoprob'>Store genotype probabilities in fst database</h2><span id='topic+fst_genoprob'></span>

<h3>Description</h3>

<p>Save an R/qtl2 genotype probabilities object to a set of fst files for fast access with reduced memory usage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst_genoprob(
  genoprob,
  fbase,
  fdir = ".",
  compress = 0,
  verbose = TRUE,
  overwrite = FALSE,
  quiet = !verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fst_genoprob_+3A_genoprob">genoprob</code></td>
<td>
<p>Object of class <code>"calc_genoprob"</code>. For details, see the
<a href="https://kbroman.org/qtl2/assets/vignettes/developer_guide.html">R/qtl2 developer guide</a>
and <code><a href="qtl2.html#topic+calc_genoprob">qtl2::calc_genoprob()</a></code>.</p>
</td></tr>
<tr><td><code id="fst_genoprob_+3A_fbase">fbase</code></td>
<td>
<p>Base of filename for fst database.</p>
</td></tr>
<tr><td><code id="fst_genoprob_+3A_fdir">fdir</code></td>
<td>
<p>Directory for fst database.</p>
</td></tr>
<tr><td><code id="fst_genoprob_+3A_compress">compress</code></td>
<td>
<p>Amount of compression to use (value in the range 0-100; lower values mean larger file sizes)</p>
</td></tr>
<tr><td><code id="fst_genoprob_+3A_verbose">verbose</code></td>
<td>
<p>Opposite of <code>quiet</code>; deprecated argument (to be removed).</p>
</td></tr>
<tr><td><code id="fst_genoprob_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE (the default), refuse to overwrite any files that already exist.</p>
</td></tr>
<tr><td><code id="fst_genoprob_+3A_quiet">quiet</code></td>
<td>
<p>If FALSE (the default), show messages about fst database creation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genotype probabilities are stored in separate databases for each chromosome
as tables of (indivduals*genotypes) x (positions) in directory <code>fst</code>.
The <code>dim</code>, <code>dimnames</code> and <code>is_x_chr</code> elements of the object
have information about the entire fst database.
If a <code>fst_genoprob</code> object is a subset of another such object,
the <code>chr</code>, <code>ind</code>, and <code>mar</code> contain information about what is in the subset.
However, the <code>fst</code> databases are not altered in a subset, and can be restored by
<code><a href="#topic+fst_restore">fst_restore()</a></code>. The actual elements of an <code>"fst_genoprob"</code>
object are only accessible to the user after a call to <code><a href="base.html#topic+unclass">unclass()</a></code>; instead
the usual access to elements of the object invoke <code><a href="#topic+subset.fst_genoprob">subset.fst_genoprob()</a></code>.
</p>


<h3>Value</h3>

<p>A list containing the attributes of <code>genoprob</code>
and the address for the created fst database.
Components are:
</p>

<ul>
<li> <p><code>dim</code> - List of all dimensions of 3-D arrays.
</p>
</li>
<li> <p><code>dimnames</code> - List of all dimension names of 3-D arrays.
</p>
</li>
<li> <p><code>is_x_chr</code> - Vector of all is_x_chr attributes.
</p>
</li>
<li> <p><code>chr</code> - Vector of (subset of) chromosome names for this object.
</p>
</li>
<li> <p><code>ind</code> - Vector of (subset of) individual names for this object.
</p>
</li>
<li> <p><code>mar</code> - Vector of (subset of) marker names for this object.
</p>
</li>
<li> <p><code>fst</code> - Path and base of file names for the fst database.
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>fst_genoprob()</code>: Deprecated version (to be deleted)
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+fst_path">fst_path()</a></code>, <code><a href="#topic+fst_extract">fst_extract()</a></code>, <code><a href="#topic+fst_files">fst_files()</a></code>, <code><a href="#topic+replace_path">replace_path()</a></code>, <code><a href="#topic+fst_restore">fst_restore()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
map &lt;- insert_pseudomarkers(grav2$gmap, step=1)
probs &lt;- calc_genoprob(grav2, map, error_prob=0.002)
dir &lt;- tempdir()
fprobs &lt;- fst_genoprob(probs, "grav2", dir, overwrite=TRUE)

# clean up: remove all the files we created
unlink(fst_files(fprobs))
</code></pre>

<hr>
<h2 id='fst_path'>Path used in fst_genoprob object</h2><span id='topic+fst_path'></span>

<h3>Description</h3>

<p>Get the path used in an fst_genoprob object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst_path(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fst_path_+3A_object">object</code></td>
<td>
<p>An object of class <code>"fst_genoprob"</code> as created by <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character string with path (and initial file stem) for files used in the input <code>object</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fst_files">fst_files()</a></code>, <code><a href="#topic+replace_path">replace_path()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
probs &lt;- calc_genoprob(grav2, error_prob=0.002)
dir &lt;- tempdir()
fprobs &lt;- fst_genoprob(probs, "grav2", dir, overwrite=TRUE)

fst_path(fprobs)
fst_files(fprobs)

# clean up: remove all the files we created
unlink(fst_files(fprobs))
</code></pre>

<hr>
<h2 id='fst_restore'>Restore fst_genoprob object to original dimensions.</h2><span id='topic+fst_restore'></span><span id='topic+fst_genoprob_restore'></span>

<h3>Description</h3>

<p>Any <code>"fst_genoprob"</code> object has embedded its original data and dimensions.
This resets elements <code>ind</code>, <code>chr</code> and <code>mar</code> to the full set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst_restore(object)

fst_genoprob_restore(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fst_restore_+3A_object">object</code></td>
<td>
<p>Object of class <code>"fst_genoprob"</code> as produced by <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Object is unclassed and elements <code>ind</code>, <code>chr</code> and <code>mar</code> are changed before
reseting attributes as <code>"fst_genoprob"</code> object.
See <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code> for details on the object.
</p>


<h3>Value</h3>

<p>Input <code>object</code> with dimensions restored.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>fst_genoprob_restore()</code>: Deprecated version (to be removed).
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>, <code><a href="#topic+fst_extract">fst_extract()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
map &lt;- insert_pseudomarkers(grav2$gmap, step=1)
probs &lt;- calc_genoprob(grav2, map, error_prob=0.002)
dir &lt;- tempdir()
fprobs &lt;- fst_genoprob(probs, "grav2", dir, overwrite=TRUE)

# subset probabilities
fprobs2 &lt;- subset(fprobs, chr=1:2)

# use object to get the full probabilities back
fprobs5 &lt;- fst_restore(fprobs2)

# clean up: remove all the files we created
unlink(fst_files(fprobs))

</code></pre>

<hr>
<h2 id='genoprob_to_alleleprob_fst'>Convert genotype probabilities to allele probabilities and write to fst database</h2><span id='topic+genoprob_to_alleleprob_fst'></span>

<h3>Description</h3>

<p>Reduce genotype probabilities (as calculated by
<code><a href="qtl2.html#topic+calc_genoprob">qtl2::calc_genoprob()</a></code>) to allele probabilities, writing them to an fst database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genoprob_to_alleleprob_fst(
  probs,
  fbase,
  fdir = ".",
  quiet = TRUE,
  cores = 1,
  compress = 0,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_probs">probs</code></td>
<td>
<p>Genotype probabilities, as calculated from
<code><a href="qtl2.html#topic+calc_genoprob">qtl2::calc_genoprob()</a></code>.</p>
</td></tr>
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_fbase">fbase</code></td>
<td>
<p>Base of filename for fst database.</p>
</td></tr>
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_fdir">fdir</code></td>
<td>
<p>Directory for fst database.</p>
</td></tr>
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_quiet">quiet</code></td>
<td>
<p>IF <code>FALSE</code>, print progress messages.</p>
</td></tr>
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_cores">cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code><a href="parallel.html#topic+makeCluster">parallel::makeCluster()</a></code>.</p>
</td></tr>
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_compress">compress</code></td>
<td>
<p>Amount of compression to use (value in the range 0-100; lower values mean larger file sizes)</p>
</td></tr>
<tr><td><code id="genoprob_to_alleleprob_fst_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE (the default), refuse to overwrite any files that already exist.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is like calling <code>qtl2::genoprob_to_alleleprob()</code> and then
<code>fst_genoprob()</code>, but in a way that hopefully saves memory by
doing it one chromosome at a time.
</p>


<h3>Value</h3>

<p>Link to fst database for the <code>probs</code> input with probabilities
collapsed to alleles rather than genotypes.
</p>


<h3>See Also</h3>

<p><code><a href="qtl2.html#topic+genoprob_to_alleleprob">qtl2::genoprob_to_alleleprob()</a></code>, <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))
gmap_w_pmar &lt;- insert_pseudomarkers(iron$gmap, step=1)

# genotype probabilities
fst_dir &lt;- file.path(tempdir(), "iron_genoprob")
dir.create(fst_dir)
probs_fst &lt;- calc_genoprob_fst(iron, "iron", fst_dir, gmap_w_pmar, error_prob=0.002)

# allele probabilities
fst_dir_apr &lt;- file.path(tempdir(), "iron_alleleprob")
dir.create(fst_dir_apr)
aprobs_fst &lt;- genoprob_to_alleleprob_fst(probs_fst, "iron", fst_dir_apr)

# clean up: remove all the files we created
unlink(fst_files(probs_fst))
unlink(fst_files(aprobs_fst))
</code></pre>

<hr>
<h2 id='rbind.fst_genoprob'>Join genotype probabilities for different individuals</h2><span id='topic+rbind.fst_genoprob'></span>

<h3>Description</h3>

<p>Join multiple genotype probability objects, as produced by
<code><a href="#topic+fst_genoprob">fst_genoprob()</a></code> for different individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fst_genoprob'
rbind(..., fbase = NULL, fdir = NULL, overwrite = FALSE, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rbind.fst_genoprob_+3A_...">...</code></td>
<td>
<p>Genotype probability objects as produced by
<code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>. Must have the same set of markers and
genotypes.</p>
</td></tr>
<tr><td><code id="rbind.fst_genoprob_+3A_fbase">fbase</code></td>
<td>
<p>Base of fileame for fst database.
Needed if objects have different fst databases.</p>
</td></tr>
<tr><td><code id="rbind.fst_genoprob_+3A_fdir">fdir</code></td>
<td>
<p>Directory for fst database.</p>
</td></tr>
<tr><td><code id="rbind.fst_genoprob_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE (the default), refuse to overwrite existing <code>.fst</code> files</p>
</td></tr>
<tr><td><code id="rbind.fst_genoprob_+3A_quiet">quiet</code></td>
<td>
<p>If TRUE, don't show any messages. Passed to <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single genotype probability object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cbind.fst_genoprob">cbind.fst_genoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
map &lt;- insert_pseudomarkers(grav2$gmap, step=1)
probsA &lt;- calc_genoprob(grav2[1:5,], map, error_prob=0.002)
probsB &lt;- calc_genoprob(grav2[6:12,], map, error_prob=0.002)
dir &lt;- tempdir()
fprobsA &lt;- fst_genoprob(probsA, "exampleAr", dir, overwrite=TRUE)
fprobsB &lt;- fst_genoprob(probsB, "exampleBr", dir, overwrite=TRUE)

# use rbind to combine probabilities for same chromosomes but different individuals
fprobs &lt;- rbind(fprobsA, fprobsB, fbase = "exampleABr")

# clean up: remove all the files we created
unlink(fst_files(fprobsA))
unlink(fst_files(fprobsB))
unlink(fst_files(fprobs))

</code></pre>

<hr>
<h2 id='replace_path'>Replace the path used in fst_genoprob object</h2><span id='topic+replace_path'></span>

<h3>Description</h3>

<p>Replace the path used in an fst_genoprob object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replace_path(object, path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replace_path_+3A_object">object</code></td>
<td>
<p>An object of class <code>"fst_genoprob"</code> as created by <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
<tr><td><code id="replace_path_+3A_path">path</code></td>
<td>
<p>New path (directory + file stem as a single character string) to be used in the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input <code>object</code> with the path replaced.
If any of the expected files don't exist with the new path, warnings are issued.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fst_path">fst_path()</a></code>, <code><a href="#topic+fst_files">fst_files()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
probs &lt;- calc_genoprob(grav2, error_prob=0.002)
dir &lt;- tempdir()
fprobs &lt;- fst_genoprob(probs, "grav2", dir, overwrite=TRUE)

# move the probabilities into a different directory
new_dir &lt;- file.path(tempdir(), "subdir")
if(!dir.exists(new_dir)) dir.create(new_dir)
for(file in fst_files(fprobs)) {
   file.rename(file, file.path(new_dir, basename(file)))
}

# revise the path in fprobs
new_path &lt;- sub(dir, new_dir, fst_path(fprobs), fixed=TRUE)
fprobs &lt;- replace_path(fprobs, new_path)
</code></pre>

<hr>
<h2 id='subset_fst_genoprob'>Subsetting genotype probabilities</h2><span id='topic+subset_fst_genoprob'></span><span id='topic+subset.fst_genoprob'></span>

<h3>Description</h3>

<p>Pull out a specified set of individuals and/or chromosomes from
the results of <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_fst_genoprob(x, ind = NULL, chr = NULL, mar = NULL, ...)

## S3 method for class 'fst_genoprob'
subset(x, ind = NULL, chr = NULL, mar = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset_fst_genoprob_+3A_x">x</code></td>
<td>
<p>Genotype probabilities as output from <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
<tr><td><code id="subset_fst_genoprob_+3A_ind">ind</code></td>
<td>
<p>A vector of individuals: numeric indices, logical
values, or character string IDs</p>
</td></tr>
<tr><td><code id="subset_fst_genoprob_+3A_chr">chr</code></td>
<td>
<p>A vector of chromosomes: logical values, or character
string IDs. Numbers are interpreted as character string IDs.</p>
</td></tr>
<tr><td><code id="subset_fst_genoprob_+3A_mar">mar</code></td>
<td>
<p>A vector of marker names as character string IDs.</p>
</td></tr>
<tr><td><code id="subset_fst_genoprob_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input genotype probabilities, with the selected
individuals and/or chromsomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))

pr &lt;- calc_genoprob(grav2)
dir &lt;- tempdir()
fpr &lt;- fst_genoprob(pr, "grav2", dir)

# keep just individuals 1:5, chromosome 2
prsub &lt;- fpr[1:5,2]
# keep just chromosome 2
prsub2 &lt;- fpr[,2]

# clean up: remove all the files we created
unlink(fst_files(fpr))
</code></pre>

<hr>
<h2 id='summary.fst_genoprob'>Summary of an fst_genoprob object</h2><span id='topic+summary.fst_genoprob'></span>

<h3>Description</h3>

<p>Summarize an fst_genoprob object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fst_genoprob'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.fst_genoprob_+3A_object">object</code></td>
<td>
<p>An object of class <code>"fst_genoprob"</code>, as output by <code><a href="#topic+fst_genoprob">fst_genoprob()</a></code>.</p>
</td></tr>
<tr><td><code id="summary.fst_genoprob_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl2)
grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))

pr &lt;- calc_genoprob(grav2)
dir &lt;- tempdir()
fpr &lt;- fst_genoprob(pr, "grav2", dir)

# summary of fst_genoprob object
summary(fpr)

# clean up: remove all the files we created
unlink(fst_files(fpr))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
