<!DOCTYPE html><html lang="en"><head><title>Help for package qtl2pattern</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtl2pattern}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#allele1'><p>Allele plot for SNPs, alleles and allele pairs</p></a></li>
<li><a href='#create_probs_query_func'><p>Create a function to query genotype probabilities</p></a></li>
<li><a href='#gene_exon'><p>Get exons for set of genes</p></a></li>
<li><a href='#genoprob_to_patternprob'><p>Collapse genoprob according to pattern</p></a></li>
<li><a href='#get_feature_snp'><p>Match features with SNPs</p></a></li>
<li><a href='#get_gene_snp'><p>Match genes with SNPs</p></a></li>
<li><a href='#get.gene.locations'><p>Helper function to set gene locations on plot.</p></a></li>
<li><a href='#ggplot_merge_feature'><p>Plot of merge_feature object</p></a></li>
<li><a href='#ggplot_scan1pattern'><p>Plot scan pattern usign ggplot2</p></a></li>
<li><a href='#pattern_diplos'><p>Extract pattern of diplotypes</p></a></li>
<li><a href='#pattern_label'><p>Turn genotype probabilities into labels</p></a></li>
<li><a href='#read_fast'><p>Read fast database with possible rownames</p></a></li>
<li><a href='#read_probs'><p>Read genotype probability object from file</p></a></li>
<li><a href='#sdp_to_pattern'><p>Convert sdp to pattern</p></a></li>
<li><a href='#snpinfo_to_map'><p>Convert SNP info to map</p></a></li>
<li><a href='#snpprob_collapse'><p>Collapse genoprob according to pattern</p></a></li>
<li><a href='#summary.feature_snp'><p>Summary of features with SNP information</p></a></li>
<li><a href='#summary.feature_tbl'><p>Summary of features</p></a></li>
<li><a href='#summary.gene_snp'><p>Summary of genes overlapping SNPs</p></a></li>
<li><a href='#top_snps_pattern'><p>Top SNPs organized by allele pattern</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pattern Support for 'qtl2' Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-03-11</td>
</tr>
<tr>
<td>Description:</td>
<td>Routines in 'qtl2' to study allele patterns in quantitative trait loci (QTL)
    mapping over a chromosome.
    Useful in crosses with more than two alleles to identify how sets of alleles,
    genetically different strands at the same locus, have different response levels.
    Plots show profiles over a chromosome.
    Can handle multiple traits together.
    See <a href="https://github.com/byandell/qtl2pattern">https://github.com/byandell/qtl2pattern</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, tidyr, stringr, ggplot2, assertthat, qtl2, qtl2fst,
fst, rlang, stats, graphics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, qtl2ggplot</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/byandell/qtl2pattern">https://github.com/byandell/qtl2pattern</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-11 22:10:05 UTC; brianyandell</td>
</tr>
<tr>
<td>Author:</td>
<td>Brian S Yandell [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Brian S Yandell &lt;brian.yandell@wisc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-11 22:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='allele1'>Allele plot for SNPs, alleles and allele pairs</h2><span id='topic+allele1'></span><span id='topic+ggplot_allele1'></span><span id='topic+autoplot.allele1'></span>

<h3>Description</h3>

<p>Create table of alleles for various model fits.
</p>
<p>Plot alleles for haplotype, diplotype and top patterns and genome position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allele1(
  probD,
  phe_df = NULL,
  cov_mx = NULL,
  map = NULL,
  K_chr = NULL,
  patterns = NULL,
  alt = NULL,
  blups = FALSE,
  ...
)

ggplot_allele1(
  object,
  scan1_object = NULL,
  map = NULL,
  pos = NULL,
  trim = TRUE,
  legend.position = "none",
  ...
)

## S3 method for class 'allele1'
autoplot(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="allele1_+3A_probd">probD</code></td>
<td>
<p>object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code></p>
</td></tr>
<tr><td><code id="allele1_+3A_phe_df">phe_df</code></td>
<td>
<p>data frame with one phenotype</p>
</td></tr>
<tr><td><code id="allele1_+3A_cov_mx">cov_mx</code></td>
<td>
<p>covariate matrix</p>
</td></tr>
<tr><td><code id="allele1_+3A_map">map</code></td>
<td>
<p>Genome map (required if <code>scan1_object</code> present).</p>
</td></tr>
<tr><td><code id="allele1_+3A_k_chr">K_chr</code></td>
<td>
<p>kinship matrix</p>
</td></tr>
<tr><td><code id="allele1_+3A_patterns">patterns</code></td>
<td>
<p>data frame of pattern information</p>
</td></tr>
<tr><td><code id="allele1_+3A_alt">alt</code></td>
<td>
<p>Haplotype allele letter(s) for alternative to reference.</p>
</td></tr>
<tr><td><code id="allele1_+3A_blups">blups</code></td>
<td>
<p>Create BLUPs if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="allele1_+3A_...">...</code></td>
<td>
<p>Other parameters ignored.</p>
</td></tr>
<tr><td><code id="allele1_+3A_object">object</code></td>
<td>
<p>Object of class <code><a href="#topic+allele1">allele1</a></code>.</p>
</td></tr>
<tr><td><code id="allele1_+3A_scan1_object">scan1_object</code></td>
<td>
<p>Optional object of class <code><a href="qtl2.html#topic+scan1">scan1</a></code> to find peak.</p>
</td></tr>
<tr><td><code id="allele1_+3A_pos">pos</code></td>
<td>
<p>Genome position in Mbp (supercedes <code>scan1_object</code>).</p>
</td></tr>
<tr><td><code id="allele1_+3A_trim">trim</code></td>
<td>
<p>If <code>TRUE</code>, trim extreme alleles.</p>
</td></tr>
<tr><td><code id="allele1_+3A_legend.position">legend.position</code></td>
<td>
<p>Legend position (default is <code>"none"</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table with allele effects across sources.
</p>
<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
</p>

<hr>
<h2 id='create_probs_query_func'>Create a function to query genotype probabilities</h2><span id='topic+create_probs_query_func'></span>

<h3>Description</h3>

<p>Create a function that will connect to a database of genotype probability
information and return a list with 'probs' object and a 'map' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_probs_query_func(dbfile, method_val = "fst", probdir_val = "genoprob")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_probs_query_func_+3A_dbfile">dbfile</code></td>
<td>
<p>Name of database file</p>
</td></tr>
<tr><td><code id="create_probs_query_func_+3A_method_val">method_val</code></td>
<td>
<p>either <code>"fst"</code> or <code>"calc"</code> for type of genotype probabilities</p>
</td></tr>
<tr><td><code id="create_probs_query_func_+3A_probdir_val">probdir_val</code></td>
<td>
<p>name of probability directory (default <code>"genoprob"</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function assumes that <code>probdir_val</code> has a file with the
physical map with positions in Mbp and other files with genotype probabilities.
See <code><a href="#topic+read_probs">read_probs</a></code> for details on how probabilities are read.
See <code><a href="qtl2.html#topic+create_variant_query_func">create_variant_query_func</a></code> for original idea.
</p>


<h3>Value</h3>

<p>Function with six arguments, 'chr', 'start',
'end', 'allele', 'method' and 'probdir'. It returns a list with 'probs' and 'map' objects
spanning the region specified by the first three arguments.
The 'probs' element should be either a 'calc_genoprob' or 'fst_genoprob' object
(see <code><a href="qtl2fst.html#topic+fst_genoprob">fst_genoprob</a></code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

create_qv &lt;- function(dirpath) {
  # Download SNP info for DOex from web via RDS.
  # snpinfo is referenced internally in the created function.
  
  tmpfile &lt;- tempfile()
  download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
  snpinfo &lt;- readRDS(tmpfile)
  unlink(tmpfile)
  snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)
  
  function(chr, start, end) {
    if(chr != "2") return(NULL)
    if(start &lt; 96.5) start &lt;- 96.5
    if(end &gt; 98.5) end &lt;- 98.5
    if(start &gt;= end) return(NULL)
    dplyr::filter(snpinfo, .data$pos &gt;= start, .data$pos &lt;= end)
  }
}

query_variants &lt;- create_qv(dirpath)


create_qg &lt;- function(dirpath) {
  # Download Gene info for DOex from web via RDS
  # gene_tbl is referenced internally in the created function.
  
  tmpfile &lt;- tempfile()
  download.file(file.path(dirpath, "c2_genes.rds"), tmpfile, quiet=TRUE)
  gene_tbl &lt;- readRDS(tmpfile)
  unlink(tmpfile)
  
  function(chr, start, end) {
    if(chr != "2") return(NULL)
    if(start &lt; 96.5) start &lt;- 96.5
    if(end &gt; 98.5) end &lt;- 98.5
    if(start &gt;= end) return(NULL)
    dplyr::filter(gene_tbl, .data$end &gt;= start, .data$start &lt;= end)
  }
}

query_genes &lt;- create_qg(dirpath)


# Examples for probs require either FST or RDS storage of data.

</code></pre>

<hr>
<h2 id='gene_exon'>Get exons for set of genes</h2><span id='topic+gene_exon'></span><span id='topic+summary.gene_exon'></span><span id='topic+subset.gene_exon'></span><span id='topic+ggplot_gene_exon'></span><span id='topic+autoplot.gene_exon'></span>

<h3>Description</h3>

<p>Match up exon start,stop,strand with genes. Use <code>query_genes</code> to find features; see <code><a href="qtl2.html#topic+create_gene_query_func">create_gene_query_func</a></code>.
</p>
<p>Returns table of gene and its exons.
</p>
<p>Uses <code><a href="#topic+gene_exon">gene_exon</a></code> to plot genes, exons, mRNA with SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_exon(
  top_snps_tbl,
  feature_tbl = query_genes(chr_id, range_Mbp[1], range_Mbp[2])
)

## S3 method for class 'gene_exon'
summary(object, gene_name = NULL, top_snps_tbl = NULL, extra = 0.005, ...)

## S3 method for class 'gene_exon'
subset(x, gene_val, ...)

ggplot_gene_exon(
  object,
  top_snps_tbl = NULL,
  plot_now = TRUE,
  genes = unique(object$gene),
  ...
)

## S3 method for class 'gene_exon'
autoplot(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_exon_+3A_top_snps_tbl">top_snps_tbl</code></td>
<td>
<p>table from <code><a href="qtl2.html#topic+top_snps">top_snps</a></code></p>
</td></tr>
<tr><td><code id="gene_exon_+3A_feature_tbl">feature_tbl</code></td>
<td>
<p>table of features from <code>query_genes</code>; see <code><a href="qtl2.html#topic+create_gene_query_func">create_gene_query_func</a></code></p>
</td></tr>
<tr><td><code id="gene_exon_+3A_object">object</code></td>
<td>
<p>Object of class <code>gene_exon</code>.</p>
</td></tr>
<tr><td><code id="gene_exon_+3A_gene_name">gene_name</code></td>
<td>
<p>name of gene as character string</p>
</td></tr>
<tr><td><code id="gene_exon_+3A_extra">extra</code></td>
<td>
<p>extra region beyond gene for SNPs (in Mbp)</p>
</td></tr>
<tr><td><code id="gene_exon_+3A_...">...</code></td>
<td>
<p>arguments passed along to <code><a href="#topic+gene_exon">gene_exon</a></code></p>
</td></tr>
<tr><td><code id="gene_exon_+3A_x">x</code></td>
<td>
<p>Object of class <code>gene_exon</code>.</p>
</td></tr>
<tr><td><code id="gene_exon_+3A_gene_val">gene_val</code></td>
<td>
<p>Name of gene from object <code>x</code>.</p>
</td></tr>
<tr><td><code id="gene_exon_+3A_plot_now">plot_now</code></td>
<td>
<p>plot now if TRUE</p>
</td></tr>
<tr><td><code id="gene_exon_+3A_genes">genes</code></td>
<td>
<p>Names of genes in <code>object</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl of exon and gene features
</p>
<p>tbl of summary
</p>
<p>list of ggplots (see <code><a href="#topic+gene_exon">gene_exon</a></code>)
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to SNP probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)

# Scan SNPs.
scan_snppr &lt;- qtl2::scan1(snppr, DOex$pheno)

# Collect top SNPs
top_snps_tbl &lt;- top_snps_pattern(scan_snppr, snpinfo)

# Download Gene info for DOex from web via RDS
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_genes.rds"), tmpfile, quiet=TRUE)
gene_tbl &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Get Gene exon information.
out &lt;- gene_exon(top_snps_tbl, gene_tbl)
summary(out, gene = out$gene[1])


</code></pre>

<hr>
<h2 id='genoprob_to_patternprob'>Collapse genoprob according to pattern</h2><span id='topic+genoprob_to_patternprob'></span>

<h3>Description</h3>

<p>Collapse genoprob according to pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genoprob_to_patternprob(probs1, sdp, alleles = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genoprob_to_patternprob_+3A_probs1">probs1</code></td>
<td>
<p>object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code></p>
</td></tr>
<tr><td><code id="genoprob_to_patternprob_+3A_sdp">sdp</code></td>
<td>
<p>SNP distribution pattern</p>
</td></tr>
<tr><td><code id="genoprob_to_patternprob_+3A_alleles">alleles</code></td>
<td>
<p>use allele string if <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Convert genotype probabilities to pattern probabilities for pattern 1.
pattern_pr &lt;- genoprob_to_patternprob(pr, 7, TRUE)

str(pr)
str(pattern_pr)


</code></pre>

<hr>
<h2 id='get_feature_snp'>Match features with SNPs</h2><span id='topic+get_feature_snp'></span>

<h3>Description</h3>

<p>Find features that overlap with SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_feature_snp(snp_tbl, feature_tbl, extend = 0.005)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_feature_snp_+3A_snp_tbl">snp_tbl</code></td>
<td>
<p>tbl of SNPs from <code>assoc.map</code></p>
</td></tr>
<tr><td><code id="get_feature_snp_+3A_feature_tbl">feature_tbl</code></td>
<td>
<p>tbl of feature information from <code><a href="qtl2.html#topic+create_gene_query_func">create_gene_query_func</a></code></p>
</td></tr>
<tr><td><code id="get_feature_snp_+3A_extend">extend</code></td>
<td>
<p>extend region for SNPs in Mbp (default 0.005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl of features covering SNPs
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='get_gene_snp'>Match genes with SNPs</h2><span id='topic+get_gene_snp'></span>

<h3>Description</h3>

<p>Internal routine to find features that overlap with SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_snp(
  snp_tbl,
  feature_tbl,
  feature_snp = get_feature_snp(snp_tbl, feature_tbl, 0)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gene_snp_+3A_snp_tbl">snp_tbl</code></td>
<td>
<p>tbl of SNPs from <code>query_variants</code>; see package <code><a href="qtl2.html#topic+create_variant_query_func">create_variant_query_func</a></code></p>
</td></tr>
<tr><td><code id="get_gene_snp_+3A_feature_tbl">feature_tbl</code></td>
<td>
<p>tbl of feature information from <code>query_genes</code>; see package <code><a href="qtl2.html#topic+create_gene_query_func">create_gene_query_func</a></code></p>
</td></tr>
<tr><td><code id="get_gene_snp_+3A_feature_snp">feature_snp</code></td>
<td>
<p>tbl of feature information from <code><a href="#topic+get_feature_snp">get_feature_snp</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl of genes covering SNPs
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='get.gene.locations'>Helper function to set gene locations on plot.</h2><span id='topic+get.gene.locations'></span>

<h3>Description</h3>

<p>Figure out gene locations to make room for gene names.
Written original by Dan Gatti 2013-02-13
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.gene.locations(
  locs,
  xlim,
  text_size = 3,
  str_rect = c("iW", "i"),
  n_rows = 10,
  plot_width = 6,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.gene.locations_+3A_locs">locs</code></td>
<td>
<p>tbl of gene information</p>
</td></tr>
<tr><td><code id="get.gene.locations_+3A_xlim">xlim</code></td>
<td>
<p>X axis limits</p>
</td></tr>
<tr><td><code id="get.gene.locations_+3A_text_size">text_size</code></td>
<td>
<p>size of text (default 3)</p>
</td></tr>
<tr><td><code id="get.gene.locations_+3A_str_rect">str_rect</code></td>
<td>
<p>character spacing on left and right of rectangles (default c(&quot;iW&quot;,&quot;i&quot;))</p>
</td></tr>
<tr><td><code id="get.gene.locations_+3A_n_rows">n_rows</code></td>
<td>
<p>desired number of rows (default 10)</p>
</td></tr>
<tr><td><code id="get.gene.locations_+3A_plot_width">plot_width</code></td>
<td>
<p>width of default plot window (in inches)</p>
</td></tr>
<tr><td><code id="get.gene.locations_+3A_...">...</code></td>
<td>
<p>additional parameters (not used)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list object used by <code><a href="#topic+ggplot_feature_tbl">ggplot_feature_tbl</a></code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
Daniel Gatti, <a href="mailto:Dan.Gatti@jax.org">Dan.Gatti@jax.org</a>
</p>


<h3>References</h3>

<p><a href="https://github.com/dmgatti/DOQTL/blob/master/R/gene.plot.R">https://github.com/dmgatti/DOQTL/blob/master/R/gene.plot.R</a>
</p>

<hr>
<h2 id='ggplot_merge_feature'>Plot of merge_feature object</h2><span id='topic+ggplot_merge_feature'></span><span id='topic+autoplot.merge_feature'></span><span id='topic+merge_feature'></span><span id='topic+summary.merge_feature'></span>

<h3>Description</h3>

<p>Merge all SNPs in small region with LOD peaks across multiple phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_merge_feature(object, pheno, plot_by = c("pattern", "consequence"), ...)

## S3 method for class 'merge_feature'
autoplot(object, ...)

merge_feature(
  top_snps_tbl,
  snpinfo,
  out_lmm_snps,
  drop = 1.5,
  dropchar = 0,
  exons = gene_exon(top_snps_tbl)
)

## S3 method for class 'merge_feature'
summary(object, sum_type = c("SNP type", "pattern"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_merge_feature_+3A_object">object</code></td>
<td>
<p>of class <code>merge_feature</code></p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_pheno">pheno</code></td>
<td>
<p>name of phenotype to be plotted</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_plot_by">plot_by</code></td>
<td>
<p>element to plot by (one of <code>c("pattern","consequence")</code>)</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_...">...</code></td>
<td>
<p>other arguments not used</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_top_snps_tbl">top_snps_tbl</code></td>
<td>
<p>tbl from <code><a href="#topic+top_snps_pattern">top_snps_pattern</a></code> or <code><a href="qtl2.html#topic+top_snps">top_snps</a></code></p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_snpinfo">snpinfo</code></td>
<td>
<p>SNP information table</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_out_lmm_snps">out_lmm_snps</code></td>
<td>
<p>tbl from <code><a href="qtl2.html#topic+scan1">scan1</a></code> on SNPs</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_drop">drop</code></td>
<td>
<p>include LOD scores within <code>drop</code> of max for each phenotype</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_dropchar">dropchar</code></td>
<td>
<p>number of characters to drop on phenames</p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_exons">exons</code></td>
<td>
<p>table from <code><a href="#topic+gene_exon">gene_exon</a></code></p>
</td></tr>
<tr><td><code id="ggplot_merge_feature_+3A_sum_type">sum_type</code></td>
<td>
<p>one of <code>c("SNP type","pattern")</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>
<p>tbl with added information on genes and exons
</p>
<p>table summary
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to SNP probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)

# Scan SNPs.
scan_snppr &lt;- qtl2::scan1(snppr, DOex$pheno)

# Collect top SNPs
top_snps_tbl &lt;- top_snps_pattern(scan_snppr, snpinfo)
summary(top_snps_tbl)

# Download Gene info for DOex from web via RDS
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_genes.rds"), tmpfile, quiet=TRUE)
gene_tbl &lt;- readRDS(tmpfile)
unlink(tmpfile)

out &lt;- merge_feature(top_snps_tbl, snpinfo, scan_snppr, exons = gene_tbl)
summary(out, "pattern")


</code></pre>

<hr>
<h2 id='ggplot_scan1pattern'>Plot scan pattern usign ggplot2</h2><span id='topic+ggplot_scan1pattern'></span><span id='topic+autoplot.scan1pattern'></span><span id='topic+scan1pattern'></span><span id='topic+summary.scan1pattern'></span>

<h3>Description</h3>

<p>Plot scan pattern usign ggplot2
</p>
<p>Genome scan by pattern set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_scan1pattern(
  object,
  map,
  plot_type = c("lod", "coef", "coef_and_lod"),
  patterns = object$patterns$founders,
  columns = 1:3,
  min_lod = 3,
  lodcolumn = seq_along(patterns),
  facet = "pheno",
  ...
)

## S3 method for class 'scan1pattern'
autoplot(object, ...)

scan1pattern(
  probs1,
  phe,
  K = NULL,
  covar = NULL,
  map,
  patterns,
  condense_patterns = TRUE,
  blups = FALSE,
  do_scans = TRUE
)

## S3 method for class 'scan1pattern'
summary(object, map, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_scan1pattern_+3A_object">object</code></td>
<td>
<p>object of class <code><a href="#topic+scan1pattern">scan1pattern</a></code></p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_map">map</code></td>
<td>
<p>genome map</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_plot_type">plot_type</code></td>
<td>
<p>type of plot from <code>c("lod","coef")</code></p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_patterns">patterns</code></td>
<td>
<p>data frame of pattern information</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_columns">columns</code></td>
<td>
<p>columns used for coef plot</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_min_lod">min_lod</code></td>
<td>
<p>minimum LOD peak for contrast to be retained</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>columns used for scan1 plot (default all <code>patterns</code>)</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_facet">facet</code></td>
<td>
<p>Plot facets if multiple phenotypes and patterns provided (default = <code>"pheno"</code>).</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_...">...</code></td>
<td>
<p>additional parameters passed on to other methods</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_probs1">probs1</code></td>
<td>
<p>object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code></p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_phe">phe</code></td>
<td>
<p>data frame with one phenotype</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_k">K</code></td>
<td>
<p>kinship matrix</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_covar">covar</code></td>
<td>
<p>covariate matrix</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_condense_patterns">condense_patterns</code></td>
<td>
<p>remove snp_action from contrasts if TRUE</p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_blups">blups</code></td>
<td>
<p>Create BLUPs if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="ggplot_scan1pattern_+3A_do_scans">do_scans</code></td>
<td>
<p>Do scans if <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
</p>
<p>List containing:
</p>

<ul>
<li><p>patterns Data frame of summary for top patterns (column <code>founders</code> has pattern)
</p>
</li>
<li><p>dip_set Diplotype sets for contrasts
</p>
</li>
<li><p>group Group for each founder pattern
</p>
</li>
<li><p>scan Object of class <code><a href="qtl2.html#topic+scan1">scan1</a></code>.
</p>
</li>
<li><p>coef Object of class <code>listof_scan1coef</code>. See package 'qtl2ggplot'.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to SNP probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)

# Scan SNPs
scan_snppr &lt;- qtl2::scan1(snppr, DOex$pheno)
top_snps_tbl &lt;- top_snps_pattern(scan_snppr, snpinfo)

# Summarize to find top patterns
patterns &lt;- dplyr::arrange(summary(top_snps_tbl), dplyr::desc(max_lod))

# Scan using patterns.
scan_pat &lt;- scan1pattern(pr, DOex$pheno, map = DOex$pmap, patterns = patterns)

# Summary of scan1pattern.
summary(scan_pat, DOex$pmap)


</code></pre>

<hr>
<h2 id='pattern_diplos'>Extract pattern of diplotypes</h2><span id='topic+pattern_diplos'></span><span id='topic+pattern_haplos'></span>

<h3>Description</h3>

<p>Extract pattern of diplotypes
</p>
<p>Extract pattern of haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pattern_diplos(sdp, haplos, diplos, cont = NULL)

pattern_haplos(sdp, haplos)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pattern_diplos_+3A_sdp">sdp</code></td>
<td>
<p>vector of sdp from <code><a href="#topic+top_snps_pattern">top_snps_pattern</a></code></p>
</td></tr>
<tr><td><code id="pattern_diplos_+3A_haplos">haplos</code></td>
<td>
<p>vector of haplotype names</p>
</td></tr>
<tr><td><code id="pattern_diplos_+3A_diplos">diplos</code></td>
<td>
<p>vector of diplotype names</p>
</td></tr>
<tr><td><code id="pattern_diplos_+3A_cont">cont</code></td>
<td>
<p>vector of types of contrasts (<code>NULL</code> or from <code>c("add","dom","b6r","b6d")</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of diplotype patterns
</p>
<p>matrix of haplotype patterns
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='pattern_label'>Turn genotype probabilities into labels</h2><span id='topic+pattern_label'></span><span id='topic+pattern_sdp'></span>

<h3>Description</h3>

<p>Turn genotype probabilities into labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pattern_label(genos, allele = TRUE)

pattern_sdp(label, sdp = NULL, geno_names = sort(unique(label)))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pattern_label_+3A_genos">genos</code></td>
<td>
<p>matrix of genotype probabilities at locus</p>
</td></tr>
<tr><td><code id="pattern_label_+3A_allele">allele</code></td>
<td>
<p>Driver has alleles if <code>TRUE</code>, otherwise allele pairs.</p>
</td></tr>
<tr><td><code id="pattern_label_+3A_label">label</code></td>
<td>
<p>character string from <code><a href="#topic+pattern_label">pattern_label</a></code></p>
</td></tr>
<tr><td><code id="pattern_label_+3A_sdp">sdp</code></td>
<td>
<p>SNP distribution pattern for plot colors</p>
</td></tr>
<tr><td><code id="pattern_label_+3A_geno_names">geno_names</code></td>
<td>
<p>unique genotype names (alleles or allele pairs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector of genotype names.
</p>

<hr>
<h2 id='read_fast'>Read fast database with possible rownames</h2><span id='topic+read_fast'></span>

<h3>Description</h3>

<p>Read fast database with format <code>fst</code>. Use first column
of database (must be named 'ind') as rownames if desired. R/qtl2 routines assume data frames have
rownames to use to align individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fast(datapath, columns = NULL, rownames = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_fast_+3A_datapath">datapath</code></td>
<td>
<p>character string path to database</p>
</td></tr>
<tr><td><code id="read_fast_+3A_columns">columns</code></td>
<td>
<p>names or indexes for columns to be extracted</p>
</td></tr>
<tr><td><code id="read_fast_+3A_rownames">rownames</code></td>
<td>
<p>use first column of rownames if <code>TRUE</code> (can supply column number)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>extracted data frame with appropriate rows and columns.
</p>


<h3>See Also</h3>

<p><code><a href="fst.html#topic+read_fst">read_fst</a></code>
</p>

<hr>
<h2 id='read_probs'>Read genotype probability object from file</h2><span id='topic+read_probs'></span>

<h3>Description</h3>

<p>Read object from file stored according to method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_probs(
  chr = NULL,
  start_val = NULL,
  end_val = NULL,
  datapath,
  allele = TRUE,
  method,
  probdir = "genoprob"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_probs_+3A_chr">chr</code></td>
<td>
<p>vector of chromosome identifiers</p>
</td></tr>
<tr><td><code id="read_probs_+3A_start_val">start_val</code>, <code id="read_probs_+3A_end_val">end_val</code></td>
<td>
<p>start and end values in Mbp</p>
</td></tr>
<tr><td><code id="read_probs_+3A_datapath">datapath</code></td>
<td>
<p>name of folder with Derived Data</p>
</td></tr>
<tr><td><code id="read_probs_+3A_allele">allele</code></td>
<td>
<p>read haplotype allele probabilities (if <code>TRUE</code>) or diplotype allele-pair probabilities (if <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="read_probs_+3A_method">method</code></td>
<td>
<p>method of genoprob storage</p>
</td></tr>
<tr><td><code id="read_probs_+3A_probdir">probdir</code></td>
<td>
<p>genotype probability directory (default <code>"genoprob"</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with <code>probs</code> = large object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code> and <code>map</code> = physical map for selected <code>chr</code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='sdp_to_pattern'>Convert sdp to pattern</h2><span id='topic+sdp_to_pattern'></span><span id='topic+sdp_to_logical'></span>

<h3>Description</h3>

<p>Convert strain distribution pattern (sdp) to letter pattern.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sdp_to_pattern(sdp, haplos, symmetric = TRUE)

sdp_to_logical(sdp, haplos, symmetric = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sdp_to_pattern_+3A_sdp">sdp</code></td>
<td>
<p>vector of sdp values</p>
</td></tr>
<tr><td><code id="sdp_to_pattern_+3A_haplos">haplos</code></td>
<td>
<p>letter codes for haplotypes (required)</p>
</td></tr>
<tr><td><code id="sdp_to_pattern_+3A_symmetric">symmetric</code></td>
<td>
<p>make patterns symmetric if <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of letter patterns
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"


# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Extract strain distribution pattern.
sdp &lt;- snpinfo$sdp
# Find out how many alleles.
nallele &lt;- ceiling(log2(max(sdp)))
out &lt;- sdp_to_pattern(sdp, LETTERS[seq_len(nallele)])
# Show most frequent patterns. 
head(rev(sort(c(table(out)))))


</code></pre>

<hr>
<h2 id='snpinfo_to_map'>Convert SNP info to map</h2><span id='topic+snpinfo_to_map'></span>

<h3>Description</h3>

<p>Convert SNP info to map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpinfo_to_map(snpinfo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpinfo_to_map_+3A_snpinfo">snpinfo</code></td>
<td>
<p>Data frame with SNP information with the following
columns (the last three are generally derived from with
<code><a href="qtl2.html#topic+index_snps">index_snps</a></code>):
</p>

<ul>
<li> <p><code>chr</code> - Character string or factor with chromosome
</p>
</li>
<li> <p><code>pos</code> - Position (in same units as in the <code>"map"</code>
attribute in <code>genoprobs</code>.
</p>
</li>
<li> <p><code>sdp</code> - Strain distribution pattern: an integer, between
1 and <code class="reqn">2^n - 2</code> where <code class="reqn">n</code> is the number of strains, whose
binary encoding indicates the founder genotypes
</p>
</li>
<li> <p><code>snp</code> - Character string with SNP identifier (if
missing, the rownames are used).
</p>
</li>
<li> <p><code>index</code> - Indices that indicate equivalent
groups of SNPs.
</p>
</li>
<li> <p><code>intervals</code> - Indexes that indicate which marker
intervals the SNPs reside.
</p>
</li>
<li> <p><code>on_map</code> - Indicate whether SNP coincides with a marker
in the <code>genoprobs</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>map as list of vectors of marker positions.
</p>

<hr>
<h2 id='snpprob_collapse'>Collapse genoprob according to pattern</h2><span id='topic+snpprob_collapse'></span>

<h3>Description</h3>

<p>Collapse genoprob according to pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpprob_collapse(
  snpprobs,
  action = c("additive", "add+dom", "non-add", "recessive", "dominant", "basic")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpprob_collapse_+3A_snpprobs">snpprobs</code></td>
<td>
<p>object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code></p>
</td></tr>
<tr><td><code id="snpprob_collapse_+3A_action">action</code></td>
<td>
<p>SNP gene action type</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="qtl2.html#topic+calc_genoprob">calc_genoprob</a></code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to snp probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)


dim(snppr[[1]])
dim(snpprob_collapse(snppr, "additive")[[1]])


</code></pre>

<hr>
<h2 id='summary.feature_snp'>Summary of features with SNP information</h2><span id='topic+summary.feature_snp'></span>

<h3>Description</h3>

<p>Summary of features with SNP information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'feature_snp'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.feature_snp_+3A_object">object</code></td>
<td>
<p>tbl of feature information from <code><a href="#topic+get_feature_snp">get_feature_snp</a></code></p>
</td></tr>
<tr><td><code id="summary.feature_snp_+3A_...">...</code></td>
<td>
<p>additional parameters ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl of feature summaries by type
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='summary.feature_tbl'>Summary of features</h2><span id='topic+summary.feature_tbl'></span><span id='topic+subset.feature_tbl'></span><span id='topic+ggplot_feature_tbl'></span><span id='topic+autoplot.feature_tbl'></span>

<h3>Description</h3>

<p>Show count min and max of features by type
</p>
<p>Plot genes as rectangles followed by names.
Stagger genes for easy reading.
Written original by Dan Gatti 2013-02-13
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'feature_tbl'
summary(object, major = TRUE, ...)

## S3 method for class 'feature_tbl'
subset(x, start_val = 0, stop_val = max(x$stop), ...)

ggplot_feature_tbl(
  object,
  rect_col = "grey70",
  strand_col = c(`-` = "#1b9e77", `+` = "#d95f02"),
  type_col = c(gene = "black", pseudogene = "#1b9e77", other = "#d95f02"),
  text_size = 3,
  xlim = NULL,
  snp_pos = top_snps_tbl$pos,
  snp_lod = top_snps_tbl$lod,
  top_snps_tbl = NULL,
  snp_col = "grey70",
  extend = 0.005,
  ...
)

## S3 method for class 'feature_tbl'
autoplot(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.feature_tbl_+3A_object">object</code></td>
<td>
<p>tbl of gene information from <code>query_variants</code>; see <code><a href="qtl2.html#topic+create_variant_query_func">create_variant_query_func</a></code></p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_major">major</code></td>
<td>
<p>if <code>TRUE</code> (default), only summarize genes and exons</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_...">...</code></td>
<td>
<p>additional arguments (not used)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_x">x</code></td>
<td>
<p>tbl of feature information from <code><a href="#topic+get_feature_snp">get_feature_snp</a></code></p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_start_val">start_val</code>, <code id="summary.feature_tbl_+3A_stop_val">stop_val</code></td>
<td>
<p>start and stop positions for subset</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_rect_col">rect_col</code></td>
<td>
<p>fill color of rectangle (default &quot;grey70&quot;)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_strand_col">strand_col</code></td>
<td>
<p>edge color of rectangle by strand from <code>object</code> (default -=&quot;blue&quot;, +=&quot;red&quot;; none if NULL)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_type_col">type_col</code></td>
<td>
<p>color of type from <code>object</code> (default &quot;black&quot; for gene, &quot;blue&quot; for pseudogene; none if NULL)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_text_size">text_size</code></td>
<td>
<p>size of text (default 3)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_xlim">xlim</code></td>
<td>
<p>horizontal axis limits (default is range of features)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_snp_pos">snp_pos</code></td>
<td>
<p>position of SNPs in bp if used (default NULL)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_snp_lod">snp_lod</code></td>
<td>
<p>LOD of SNPs (for color plotting)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_top_snps_tbl">top_snps_tbl</code></td>
<td>
<p>table from <code><a href="qtl2.html#topic+top_snps">top_snps</a></code></p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_snp_col">snp_col</code></td>
<td>
<p>color of SNP vertical lines (default &quot;grey70&quot;)</p>
</td></tr>
<tr><td><code id="summary.feature_tbl_+3A_extend">extend</code></td>
<td>
<p>extend region for SNPs in bp (default 0.005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl of feature summaries by type
</p>
<p>tbl of feature summaries by type
</p>
<p>data frame of gene information (invisible)
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>
<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
Daniel Gatti, <a href="mailto:Dan.Gatti@jax.org">Dan.Gatti@jax.org</a>
</p>


<h3>References</h3>

<p><a href="https://github.com/dmgatti/DOQTL/blob/master/R/gene.plot.R">https://github.com/dmgatti/DOQTL/blob/master/R/gene.plot.R</a>
</p>

<hr>
<h2 id='summary.gene_snp'>Summary of genes overlapping SNPs</h2><span id='topic+summary.gene_snp'></span>

<h3>Description</h3>

<p>Summary of genes overlapping SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gene_snp'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.gene_snp_+3A_object">object</code></td>
<td>
<p>tbl of feature information from <code><a href="#topic+get_feature_snp">get_feature_snp</a></code></p>
</td></tr>
<tr><td><code id="summary.gene_snp_+3A_...">...</code></td>
<td>
<p>additional parameters ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl of feature summaries by type
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>

<hr>
<h2 id='top_snps_pattern'>Top SNPs organized by allele pattern</h2><span id='topic+top_snps_pattern'></span><span id='topic+summary.top_snps_pattern'></span><span id='topic+subset.top_snps_pattern'></span>

<h3>Description</h3>

<p>Separate fine mapping scans by allele pattern.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top_snps_pattern(
  scan1_output,
  snpinfo,
  drop = 1.5,
  show_all_snps = TRUE,
  haplos
)

## S3 method for class 'top_snps_pattern'
summary(object, sum_type = c("range", "best", "peak"), ...)

## S3 method for class 'top_snps_pattern'
subset(x, start_val = 0, end_val = max(x$pos), pheno = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="top_snps_pattern_+3A_scan1_output">scan1_output</code></td>
<td>
<p>output of linear mixed model for <code>phename</code> (see <code><a href="qtl2.html#topic+scan1">scan1</a></code>)</p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_snpinfo">snpinfo</code></td>
<td>
<p>Data frame with SNP information with the following
columns (the last three are generally derived from with
<code><a href="qtl2.html#topic+index_snps">index_snps</a></code>):
</p>

<ul>
<li> <p><code>chr</code> - Character string or factor with chromosome
</p>
</li>
<li> <p><code>pos</code> - Position (in same units as in the <code>"map"</code>
attribute in <code>genoprobs</code>.
</p>
</li>
<li> <p><code>sdp</code> - Strain distribution pattern: an integer, between
1 and <code class="reqn">2^n - 2</code> where <code class="reqn">n</code> is the number of strains, whose
binary encoding indicates the founder genotypes
</p>
</li>
<li> <p><code>snp_id</code> - Character string with SNP identifier (if
missing, the rownames are used).
</p>
</li>
<li> <p><code>index</code> - Indices that indicate equivalent
groups of SNPs.
</p>
</li>
<li> <p><code>intervals</code> - Indexes that indicate which marker
intervals the SNPs reside.
</p>
</li>
<li> <p><code>on_map</code> - Indicate whether SNP coincides with a marker
in the <code>genoprobs</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_drop">drop</code></td>
<td>
<p>include all SNPs within <code>drop</code> of max LOD (default 1.5)</p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_show_all_snps">show_all_snps</code></td>
<td>
<p>show all SNPs if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_haplos">haplos</code></td>
<td>
<p>optional argument identify codes for haplotypes</p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_object">object</code></td>
<td>
<p>object of class <code>top_snps_tbl</code></p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_sum_type">sum_type</code></td>
<td>
<p>type of summary (one of &quot;range&quot;,&quot;best&quot;)</p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_...">...</code></td>
<td>
<p>additional parameters ignored</p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_x">x</code></td>
<td>
<p>tbl of feature information from <code><a href="#topic+get_feature_snp">get_feature_snp</a></code></p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_start_val">start_val</code>, <code id="top_snps_pattern_+3A_end_val">end_val</code></td>
<td>
<p>start and end positions for subset</p>
</td></tr>
<tr><td><code id="top_snps_pattern_+3A_pheno">pheno</code></td>
<td>
<p>phenotype name(s) for subset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>table of top_snps at maximum lod for <code>pattern</code>
</p>
<p>table summary
</p>
<p>subset of <code>x</code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to SNP probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)

# Scan SNPs.
scan_snppr &lt;- qtl2::scan1(snppr, DOex$pheno)

# Collect top SNPs
top_snps_tbl &lt;- top_snps_pattern(scan_snppr, snpinfo)
summary(top_snps_tbl)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
