<!DOCTYPE html><html><head><title>Help for package qtlbook</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtlbook}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ch3a'><p>Data with a phenotype error</p></a></li>
<li><a href='#ch3b'><p>Data with bad markers</p></a></li>
<li><a href='#ch3c'><p>Data with misplaced markers</p></a></li>
<li><a href='#gutlength'><p>Gut length intercross data</p></a></li>
<li><a href='#iron'><p>Iron levels intercross data</p></a></li>
<li><a href='#myocard'><p>Myocarditis intercross data</p></a></li>
<li><a href='#nf1'><p>Neurofibromatosis type 1 backcross data</p></a></li>
<li><a href='#ovar'><p>Interspecific backcross in Drosophila</p></a></li>
<li><a href='#trout'><p>Rainbow trout doubled haploid data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.18-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-06-28</td>
</tr>
<tr>
<td>Title:</td>
<td>Datasets for the R/qtl Book</td>
</tr>
<tr>
<td>Author:</td>
<td>Karl W Broman <a href="https://orcid.org/0000-0002-4914-6671"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karl W Broman &lt;broman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Datasets for the book, A Guide to QTL Mapping with R/qtl.
    Broman and Sen (2009) &lt;<a href="https://doi.org/10.1007%2F978-0-387-92125-9">doi:10.1007/978-0-387-92125-9</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>qtl</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://rqtl.org/book">http://rqtl.org/book</a>, <a href="https://github.com/kbroman/qtlbook">https://github.com/kbroman/qtlbook</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kbroman/qtlbook/issues">https://github.com/kbroman/qtlbook/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-28 13:57:51 UTC; kbroman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-28 14:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ch3a'>Data with a phenotype error</h2><span id='topic+ch3a'></span>

<h3>Description</h3>

<p>Anonymous data with a phenotype error and a pair of individuals with very
similar phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ch3a
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>A backcross with 234 individuals, each with five phenotypes and typed at 166
markers.
</p>


<h3>Source</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Sen, S. (2009) <em>A Guide to QTL Mapping
with R/qtl.</em> Springer, New York.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ch3b">ch3b</a></code>, <code><a href="#topic+ch3c">ch3c</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ch3a)

# phenotype problem
pairs(ch3a$pheno)
ch3a$pheno[ch3a$pheno[,4]==0,] # individual 159

# individuals with similar genotypes
library(qtl)
cg &lt;- comparegeno(ch3a)
hist(cg, breaks=200)
max(cg[cg &lt; 1])
which(cg == max(cg[cg &lt; 1]), arr.ind=TRUE)

</code></pre>

<hr>
<h2 id='ch3b'>Data with bad markers</h2><span id='topic+ch3b'></span>

<h3>Description</h3>

<p>Anonymous data with markers showing severe segregation distortion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ch3b
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>An intercross with 144 individuals, each with one phenotype and typed at 145
markers.
</p>


<h3>Source</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Sen, S. (2009) <em>A Guide to QTL Mapping
with R/qtl.</em> Springer, New York.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ch3a">ch3a</a></code>, <code><a href="#topic+ch3c">ch3c</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ch3b)

library(qtl)
thetab &lt;- geno.table(ch3b)
plot(-log10(thetab[,ncol(thetab)]), ylab=expression(-log[10](P)))
thetab[thetab[,ncol(thetab)] &lt; 1e-6,]

</code></pre>

<hr>
<h2 id='ch3c'>Data with misplaced markers</h2><span id='topic+ch3c'></span>

<h3>Description</h3>

<p>Anonymous data with markers out of place.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ch3c
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>An intercross with 100 individuals, each with one real phenotype and typed
at 108 markers.
</p>


<h3>Source</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Sen, S. (2009) <em>A Guide to QTL Mapping
with R/qtl.</em> Springer, New York.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ch3a">ch3a</a></code>, <code><a href="#topic+ch3b">ch3b</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ch3c)

library(qtl)
ch3c &lt;- est.rf(ch3c)
plotRF(ch3c, chr=c(1,7,12,13,15))

</code></pre>

<hr>
<h2 id='gutlength'>Gut length intercross data</h2><span id='topic+gutlength'></span>

<h3>Description</h3>

<p>Data from a mouse intercross experiment on gut length, including both sexes.
All individuals carry the <code class="reqn">Sox10^{Dom}</code> mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gutlength
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>A mouse intercross between C3HeBFeJ (C3) and C57BL/6J (B6), with one F1
parent carrying the <code class="reqn">Sox10^{Dom}</code> mutation.  There are 1068
mice from reciprocal intercrosses.  Over 2000 mice were generated, but only
those individuals heterozygous at <code class="reqn">Sox10^{Dom}</code> were genotyped
and included in the data set. <code class="reqn">Sox10</code> is on chromosome 15, and so that
chromosome exhibits an unusual segregation pattern.  Some mice received the
mutation from their mother and some from their father.
</p>
<p>The primary phenotype here is gut length (in cm).  The phenotype
<code>cross</code> indicates the cross used to generate an animal.
</p>


<h3>Source</h3>

<p>E. Michelle Southard-Smith, Division of Genetic Medicine, Department
of Medicine, Vanderbilt University School of Medicine,
<a href="mailto:michelle.southard-smith@vanderbilt.edu">michelle.southard-smith@vanderbilt.edu</a>
</p>


<h3>References</h3>

<p>Owens, S. E., Broman, K. W., Wiltshire, T., Elmore, J. B.,
Bradley, K. M., Smith, J. R. and Southard-Smith, E. M. (2005) Genome-wide
linkage identifies novel modifier loci of aganglionosis in the
<code class="reqn">Sox10^{Dom}</code> model of Hirschsprung disease.  <em>Hum. Mol.
Genet.</em> <strong>14</strong>, 1549&ndash;1558.
</p>
<p>Broman, K. W., Sen, 'S, Owens, S. E., Manichaikul, A., Southard-Smith, E.
M. and Churchill G. A.  (2006) The X chromosome in quantitative trait locus
mapping.  <em>Genetics</em> <strong>174</strong>, 2151&ndash;2158.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+iron">iron</a></code>, <code><a href="#topic+myocard">myocard</a></code>, <code><a href="#topic+nf1">nf1</a></code>,
<code><a href="#topic+ovar">ovar</a></code>, <code><a href="#topic+trout">trout</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gutlength)
library(qtl)
plot(gutlength)

</code></pre>

<hr>
<h2 id='iron'>Iron levels intercross data</h2><span id='topic+iron'></span>

<h3>Description</h3>

<p>Data from a mouse intercross experiment (using the C57BL/6J/Ola and SWR/Ola
strains) on basal iron levels in the liver and spleen.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iron
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>An intercross with 284 individuals (including both sexes and both cross
directions), each with measures of iron (in <code class="reqn">\mu g/g</code>) in the
liver and spleen.
</p>


<h3>Source</h3>

<p>Andrew G. Smith, MRC Toxicology Unit, <a href="mailto:ags5@le.ac.uk">ags5@le.ac.uk</a>
</p>


<h3>References</h3>

<p>Grant, G. G., Robinson, S. W., Edwards, R. E., Clothier, B.,
Davies, R., Judah, D. J., Broman, K. W. and Smith, A. G. (2006) Multiple
polymorphic loci determine basal hepatic and splenic iron status in mice.
<em>Hepatology</em> <strong>44</strong>, 174&ndash;185.
</p>


<h3>See Also</h3>

<p><code>link{gutlength}</code>, <code>link{myocard}</code>, <code>link{nf1}</code>,
<code>link{ovar}</code>, <code>link{trout}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iron)
library(qtl)
plot(iron)

</code></pre>

<hr>
<h2 id='myocard'>Myocarditis intercross data</h2><span id='topic+myocard'></span>

<h3>Description</h3>

<p>Data from a mouse intercross experiment on myocarditis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>myocard
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>An intercross between the H-2s congenic mice A.SW and B10.S, with 296
individuals (including both sexes). The mice were injected with purified
murine cardiac myosin, and the area of infiltrated myocardium in heart
sections was measured.  The phenotype is the percent myocarditis.
</p>


<h3>Source</h3>

<p>Noel R. Rose, Department of Pathology, Johns Hopkins University,
<a href="mailto:nrrose@biostat.wisc.edu">nrrose@biostat.wisc.edu</a>
</p>


<h3>References</h3>

<p>Guler, M. L, Ligons, D. L., Wang, Y., Bianco, M., Broman, K. W.
and Rose, N. R. (2005) Two autoimmune diabetes loci influencing T cell
apoptosis control susceptibility to experimental autoimmune myocarditis.
<em>J. Immunol.</em> <strong>174</strong>: 2167&ndash;2173.
</p>


<h3>See Also</h3>

<p><code>link{gutlength}</code>, <code>link{iron}</code>, <code>link{nf1}</code>
<code>link{ovar}</code>, <code>link{trout}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(myocard)
library(qtl)
plot(myocard)

</code></pre>

<hr>
<h2 id='nf1'>Neurofibromatosis type 1 backcross data</h2><span id='topic+nf1'></span>

<h3>Description</h3>

<p>Data from a backcross experiment on neurofibromatosis type I.  All
individuals carry the <em>NPcis</em> mutation, received either from their
mother or from their father.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nf1
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>Backcrosses (C57BL/6J x A/J) x C57BL/6J and C57BL/6J x (A/J x C57BL/6J) with
a total of 254 individuals.  Individuals received the <em>NPcis</em> mutation
from either their mother or their father (indicated by the phenotype
<code>from.mom</code>.  The major phenotype, <code>affected</code> indicates whether the
mice were affected (1) or unaffected (0) with neurofibromatosis type 1.
</p>


<h3>References</h3>

<p>Reilly, K. M., Broman, K. W., Bronson, R. T., Tsang, S., Loisel,
D. A., Christy, E. S., Sun, Z., Diehl, J., Munroe, D. J. and Tuskan, R. G.
(2006) An imprinted locus epistatically influences <em>Nstr1</em> and
<em>Nstr2</em> to control resistance to nerve sheath tumors in a
neurofibromatosis type 1 mouse model.  <em>Cancer Research</em> <strong>66</strong>,
62&ndash;68.
#' @source Karlyne Reilly, Mouse Cancer Genetics Program, National Cancer
Institute at Frederick, <a href="mailto:kreilly@ncifcrf.gov">kreilly@ncifcrf.gov</a>
</p>


<h3>See Also</h3>

<p><code>link{gutlength}</code> <code>link{iron}</code>, <code>link{myocard}</code>,
<code>link{ovar}</code>, <code>link{trout}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nf1)
library(qtl)
plot(nf1)

</code></pre>

<hr>
<h2 id='ovar'>Interspecific backcross in Drosophila</h2><span id='topic+ovar'></span>

<h3>Description</h3>

<p>Data on ovariole number in a backcross between <em>D. simulans</em> and
<em>D. sechellia</em>; the majority of individuals were selected to be
recombinant in the region of a putative QTL on chromosome 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ovar
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>The data come from an interspecific Drosophila backcross.  <em>D.
simulans</em> was crossed to <em>D. sechellia</em>, and the <code class="reqn">F_1</code> hybrid
was crossed back to <em>D. simulans</em>.
</p>
<p>The phenotype of interest was ovariole number in females (a measure of
fitness).  Phenotypes <code>on1</code> and <code>on2</code> are the ovariole counts in
the left and right gonads.  The phenotype <code>onm</code> is the average of the
two counts; for many individuals, the ovariole count for one of two gonads
was missing, and so <code>onm</code> is missing.
</p>
<p>In an initial cross of 402 individuals, 383 had complete phenotype data.
Initial genotyping focused on 94 individuals with extreme phenotype.
</p>
<p>To increase the resolution of a major QTL identified on chromosome 3, a
second cross of approximately 7000 flies was performed, though only 1050
individuals showing a recombination event between two morphological markers,
<em>st</em> (bright red eyes) and <em>e</em> (dark brown body), were phenotyped
and genotyped; 1038 had complete phenotype data.  The aim was to oversample
recombinants of the region of the QTL.
</p>
<p>There are genotype data for 24 markers on 3 chromosomes.  (The fourth
chromosome had one marker, but showed no effect and is not included in these
data.)
</p>
<p>The phenotype <code>cross</code> indicates whether an individual came from the
first or second cross.
</p>
<p>Alleles <code>"I"</code> and <code>"E"</code> refer to <em>D. simulans</em> and <em>D.
sechellia</em>, respectively.
</p>


<h3>Source</h3>

<p>Virginie Orgogozo, Department of Ecology and Evolutionary Biology,
Princeton University, <a href="mailto:virginie.orgogozo@normalesup.org">virginie.orgogozo@normalesup.org</a>
</p>


<h3>References</h3>

<p>Orgogozo, V., Broman, K. W. and Stern, D. L.  (2006)
High-resolution QTL mapping reveals sign epistasis controlling ovariole
number between two Drosophila species. <em>Genetics</em> <strong>173</strong>, 197&ndash;205.
</p>


<h3>See Also</h3>

<p><code>link{gutlength}</code>, <code>link{iron}</code>, <code>link{myocard}</code>,
<code>link{nf1}</code>, <code>link{trout}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ovar)
library(qtl)
plot(ovar)

</code></pre>

<hr>
<h2 id='trout'>Rainbow trout doubled haploid data</h2><span id='topic+trout'></span>

<h3>Description</h3>

<p>Data from doubled haploid individuals derived from a cross between Oregon
State University (OSU) and Clearwater (CW) River rainbow trout clonal lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trout
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.</p>


<h3>Details</h3>

<p>Doubled haploid individuals were produced from a cross between Oregon State
University (OSU) and Clearwater (CW) river rainbow trout (<em>Oncorhynchus
mykiss</em>) clonal lines.  Eggs from one of eight outbred females, two from
Troutlodge (TL) and six from Spokane (SP), were irradiated to destroy
maternal nuclear DNA and fertilized with sperm from a single <code class="reqn">F_1</code>
male.  The first embryonic cleavage was blocked by heat shock to restore
diploidy.  There are a total of 554 individuals, with between 8 and 168
individuals from each of the eight females.
</p>
<p>The primary phenotype is time to hatch (<code>tth</code>).  An additional
&quot;phenotype&quot;, <code>female</code>, indicates maternal cytoplasmic environment (the
source of the egg).
</p>
<p>There are genotype data on 171 markers on 28 linkage groups.  The linkage
groups are named as in Nichols et al. (2002), though a pair of markers are
assigned to linkage group &quot;un&quot;, as they don't connect to any of the linkage
groups in Nichols et al. (2002).
</p>
<p>Note that the data have cross type <code>"dh"</code> (for doubled haploids); in
R/qtl they are treated just like a backcross, except that genotypes are
referred to as the homozygotes.
</p>


<h3>Source</h3>

<p>Krista M. Nichols, Department of Biological Sciences, Purdue
University, <a href="mailto:kmnichol@purdue.edu">kmnichol@purdue.edu</a>
</p>


<h3>References</h3>

<p>Nichols, K. M., Broman, K. W., Sundin, K., Young, J. M.,
Wheeler, P. A. and Thorgaard, G. H.  (2007) Quantitative trait loci by
maternal cytoplasmic environment interaction for development rate in
<em>Oncorhynchus mykiss</em>. <em>Genetics</em> <strong>175</strong>, 335&ndash;347.
</p>
<p>Nichols, K. M., Young, W. P., Danzmann, R. G., Robison, B. D., Rexroad, C.,
Noakes, M., Phillips, R. B., Bentzen, P., Spies, I., Knudsen, K., Allendorf,
F. W., Cunningham, B. M., Brunelli, J., Zhang, H., Ristow, S., Drew, R.,
Brown, K. H., Wheeler, P. A. and Thrgaard, G. H. (2002) A consolidated
linkage map for rainbow trout (<em>Oncorhynchus mykiss</em>).  <em>Animal
Genetics</em> <strong>34</strong>, 102&ndash;115.
</p>


<h3>See Also</h3>

<p><code>link{gutlength}</code>, <code>link{iron}</code>, <code>link{myocard}</code>,
<code>link{nf1}</code>, <code>link{ovar}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(trout)
library(qtl)
plot(trout)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
