<!DOCTYPE html><html lang="en"><head><title>Help for package qtlcharts</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtlcharts}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#qtlcharts-package'><p>R/qtlcharts: Interactive charts for QTL data</p></a></li>
<li><a href='#cbindQTLeffects'><p>Combine multiple runs of estQTLeffects</p></a></li>
<li><a href='#estQTLeffects'><p>Calculate QTL effects at each position across the genome</p></a></li>
<li><a href='#geneExpr'><p>Anonymized gene expression data</p></a></li>
<li><a href='#grav'><p>Arabidopsis QTL data on gravitropism</p></a></li>
<li><a href='#iboxplot'><p>Modern boxplot linked to underlying histrograms</p></a></li>
<li><a href='#iboxplot_output'><p>Shiny bindings for R/qtlcharts widgets</p></a></li>
<li><a href='#idotplot'><p>Interactive phenotype x genotype plot</p></a></li>
<li><a href='#iheatmap'><p>Interactive heat map</p></a></li>
<li><a href='#ipleiotropy'><p>Tool to explore pleiotropy</p></a></li>
<li><a href='#iplot'><p>Interactive scatterplot</p></a></li>
<li><a href='#iplotCorr'><p>Image of correlation matrix with linked scatterplot</p></a></li>
<li><a href='#iplotCurves'><p>Plot of a bunch of curves, linked to points in scatterplots</p></a></li>
<li><a href='#iplotMap'><p>Interactive genetic map plot</p></a></li>
<li><a href='#iplotMScanone'><p>Interactive LOD curve</p></a></li>
<li><a href='#iplotPXG'><p>Interactive phenotype x genotype plot</p></a></li>
<li><a href='#iplotRF'><p>Interactive plot of recombination fractions</p></a></li>
<li><a href='#iplotScanone'><p>Interactive LOD curve</p></a></li>
<li><a href='#iplotScantwo'><p>Interactive plot of 2d genome scan</p></a></li>
<li><a href='#itriplot'><p>Interactive plot of trinomial probabilities</p></a></li>
<li><a href='#qtlchartsversion'><p>print the installed version of R/qtlcharts</p></a></li>
<li><a href='#scat2scat'><p>Scatterplot driving another scatterplot</p></a></li>
<li><a href='#setScreenSize'><p>Set default maximum screen size</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.16</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-01-07</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactive Graphics for QTL Experiments</td>
</tr>
<tr>
<td>Author:</td>
<td>Karl W Broman <a href="https://orcid.org/0000-0002-4914-6671"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Michael Bostock [ctb, cph] (d3.js library in htmlwidgets/lib,
    https://d3js.org),
  jQuery Foundation [cph] (jQuery library in htmlwidgets/lib,
    https://jquery.com),
  jQuery contributors [ctb] (jQuery library in htmlwidgets/lib; see
    https://github.com/jquery/jquery/blob/master/AUTHORS.txt),
  jQuery UI contributors [ctb] (jQuery UI library in htmlwidgets/lib; see
    https://jqueryui.com/about/)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karl W Broman &lt;broman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Web-based interactive charts (using D3.js) for the analysis of
    experimental crosses to identify genetic loci (quantitative trait
    loci, QTL) contributing to variation in quantitative traits.
    Broman (2015) &lt;<a href="https://doi.org/10.1534%2Fgenetics.114.172742">doi:10.1534/genetics.114.172742</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15)</td>
</tr>
<tr>
<td>Imports:</td>
<td>qtl (&ge; 1.30-4), htmlwidgets, graphics, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>htmltools, jsonlite, knitr, devtools, roxygen2, rmarkdown,
testthat</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://kbroman.org/qtlcharts/">https://kbroman.org/qtlcharts/</a>,
<a href="https://github.com/kbroman/qtlcharts">https://github.com/kbroman/qtlcharts</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kbroman/qtlcharts/issues">https://github.com/kbroman/qtlcharts/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-07 21:58:04 UTC; kbroman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-07 22:52:41 UTC</td>
</tr>
</table>
<hr>
<h2 id='qtlcharts-package'>R/qtlcharts: Interactive charts for QTL data</h2><span id='topic+qtlcharts-package'></span><span id='topic+qtlcharts'></span>

<h3>Description</h3>

<p>A QTL is a *quantitative trait locus*: a genetic locus that
contributes to variation in a quantitative trait. The goal of
[R/qtlcharts](https://kbroman.org/qtlcharts/) is to provide
interactive data visualizations for QTL analyses, and to make these
visualizations available from R. It is a companion to the
[R/qtl](https://rqtl.org) package.
</p>


<h3>Details</h3>

<p>Vignettes online at the [R/qtlcharts website](https://kbroman.org/qtlcharts/):
</p>
<p>- [User guide](https://kbroman.org/qtlcharts/assets/vignettes/userGuide.html)
- [Developer guide](https://kbroman.org/qtlcharts/assets/vignettes/develGuide.html)
- [Use with R Markdown](https://kbroman.org/qtlcharts/assets/vignettes/Rmarkdown.html)
\[[Rmd source](https://github.com/kbroman/qtlcharts/blob/gh-pages/assets/vignettes/Rmarkdown.Rmd)\]
- [List of chart customization options](https://kbroman.org/qtlcharts/assets/vignettes/chartOpts.html)
</p>

<hr>
<h2 id='cbindQTLeffects'>Combine multiple runs of estQTLeffects</h2><span id='topic+cbindQTLeffects'></span>

<h3>Description</h3>

<p>Combine multiple runs of estQTLeffects by applying cbind to each
component
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cbindQTLeffects(..., labels)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cbindQTLeffects_+3A_...">...</code></td>
<td>
<p>Results of [estQTLeffects()]</p>
</td></tr>
<tr><td><code id="cbindQTLeffects_+3A_labels">labels</code></td>
<td>
<p>Vector of labels to use in the combination.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of matrices; each component corresponds to a position
in the genome and is a matrix with phenotypes x effects
</p>


<h3>See Also</h3>

<p>[estQTLeffects()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(fake.f2)
fake.f2 &lt;- calc.genoprob(fake.f2)
sex &lt;- fake.f2$pheno$sex
eff.fem &lt;- estQTLeffects(fake.f2[,sex==0], pheno.col=1)
eff.mal &lt;- estQTLeffects(fake.f2[,sex==1], pheno.col=1)
eff &lt;- cbindQTLeffects(eff.fem, eff.mal, labels=c("female", "male"))

</code></pre>

<hr>
<h2 id='estQTLeffects'>Calculate QTL effects at each position across the genome</h2><span id='topic+estQTLeffects'></span>

<h3>Description</h3>

<p>Calculates the effects of QTL at each position across the genome
using Haley-Knott regression, much like [qtl::effectscan()],
but considering multiple phenotypes and not plotting the results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estQTLeffects(cross, pheno.col = 1, what = c("means", "effects"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estQTLeffects_+3A_cross">cross</code></td>
<td>
<p>(Optional) Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="estQTLeffects_+3A_pheno.col">pheno.col</code></td>
<td>
<p>Phenotype columns in cross object.</p>
</td></tr>
<tr><td><code id="estQTLeffects_+3A_what">what</code></td>
<td>
<p>Indicates whether to calculate phenotype averages for
each genotype group or to turn these into additive and dominance
effects.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One should first run [qtl::calc.genoprob()];
if not, it is run with the default arguments.
</p>
<p>The estimated effects will be poorly estimated in the case of
selective genotyping, as Haley-Knott regression performs poorly in
this case.
</p>


<h3>Value</h3>

<p>list of matrices; each component corresponds to a position
in the genome and is a matrix with phenotypes x effects
</p>


<h3>See Also</h3>

<p>[iplotMScanone()], [qtl::effectscan()]
[cbindQTLeffects()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(grav)
library(qtl)
grav &lt;- reduce2grid(calc.genoprob(grav, step=1))
out &lt;- estQTLeffects(grav, phe=seq(1, nphe(grav), by=5))

</code></pre>

<hr>
<h2 id='geneExpr'>Anonymized gene expression data</h2><span id='topic+geneExpr'></span>

<h3>Description</h3>

<p>An anonymized set of gene expression values, for 100
genes all influenced by a common locus, plus a vector of genotypes
for the 491 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(geneExpr)
</code></pre>


<h3>Format</h3>

<p>A list containing a matrix 'expr' with the gene
expression data plus a vector 'genotype' with the genotypes.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, 2013-05-16
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geneExpr)

# heat map of correlation matrix, linked to scatterplots
iplotCorr(geneExpr$expr, geneExpr$genotype, reorder=TRUE)
</code></pre>

<hr>
<h2 id='grav'>Arabidopsis QTL data on gravitropism</h2><span id='topic+grav'></span>

<h3>Description</h3>

<p>Data from a QTL experiment on gravitropism in
Arabidopsis, with data on 162 recombinant inbred lines (Ler x
Cvi). The outcome is the root tip angle (in degrees) at two-minute
increments over eight hours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grav)
</code></pre>


<h3>Format</h3>

<p>An object of class '&quot;cross&quot;'; see [qtl::read.cross()].
</p>


<h3>Source</h3>

<p>Mouse Phenome Database, &lt;https://phenome.jax.org/projects/Moore1b&gt;
</p>


<h3>References</h3>

<p>Moore CR, Johnson LS, Kwak I-Y, Livny M, Broman KW,
Spalding EP (2013) High-throughput computer vision introduces the
time axis to a quantitative trait map of a plant growth
response. Genetics 195:1077-1086
([PubMed](https://pubmed.ncbi.nlm.nih.gov/23979570/))
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(grav)
times &lt;- attr(grav, "time")
phe &lt;- grav$pheno

iplotCurves(phe, times, phe[,c(61,121)], phe[,c(121,181)],
            chartOpts=list(curves_xlab="Time (hours)", curves_ylab="Root tip angle (degrees)",
                           scat1_xlab="Angle at 2 hrs", scat1_ylab="Angle at 4 hrs",
                           scat2_xlab="Angle at 4 hrs", scat2_ylab="Angle at 6 hrs"))
</code></pre>

<hr>
<h2 id='iboxplot'>Modern boxplot linked to underlying histrograms</h2><span id='topic+iboxplot'></span>

<h3>Description</h3>

<p>Creates an interactive graph for a large set of box plots (rendered
as lines connecting the quantiles), linked to underlying histograms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iboxplot(
  dat,
  qu = c(0.001, 0.01, 0.1, 0.25),
  orderByMedian = TRUE,
  breaks = 251,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iboxplot_+3A_dat">dat</code></td>
<td>
<p>Data matrix (individuals x variables)</p>
</td></tr>
<tr><td><code id="iboxplot_+3A_qu">qu</code></td>
<td>
<p>Quantiles to plot (All with 0 &lt; qu &lt; 0.5)</p>
</td></tr>
<tr><td><code id="iboxplot_+3A_orderbymedian">orderByMedian</code></td>
<td>
<p>If TRUE, reorder individuals by their median</p>
</td></tr>
<tr><td><code id="iboxplot_+3A_breaks">breaks</code></td>
<td>
<p>Number of bins in the histograms, or a vector of
locations of the breakpoints between bins (as in [graphics::hist()])</p>
</td></tr>
<tr><td><code id="iboxplot_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="iboxplot_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotCorr()], [scat2scat()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n.ind &lt;- 500
n.gene &lt;- 10000
expr &lt;- matrix(rnorm(n.ind * n.gene, (1:n.ind)/n.ind*3), ncol=n.gene)
dimnames(expr) &lt;- list(paste0("ind", 1:n.ind),
                       paste0("gene", 1:n.gene))

iboxplot(expr, chartOpts=list(xlab="Mice", ylab="Gene expression"))

</code></pre>

<hr>
<h2 id='iboxplot_output'>Shiny bindings for R/qtlcharts widgets</h2><span id='topic+iboxplot_output'></span><span id='topic+iboxplot_render'></span><span id='topic+idotplot_output'></span><span id='topic+idotplot_render'></span><span id='topic+iheatmap_output'></span><span id='topic+iheatmap_render'></span><span id='topic+ipleiotropy_output'></span><span id='topic+ipleiotropy_render'></span><span id='topic+iplot_output'></span><span id='topic+iplot_render'></span><span id='topic+iplotCorr_output'></span><span id='topic+iplotCorr_render'></span><span id='topic+iplotCurves_output'></span><span id='topic+iplotCurves_render'></span><span id='topic+iplotMScanone_output'></span><span id='topic+iplotMScanone_render'></span><span id='topic+iplotMap_output'></span><span id='topic+iplotMap_render'></span><span id='topic+iplotRF_output'></span><span id='topic+iplotRF_render'></span><span id='topic+iplotScanone_output'></span><span id='topic+iplotScanone_render'></span><span id='topic+iplotScantwo_output'></span><span id='topic+iplotScantwo_render'></span><span id='topic+itriplot_output'></span><span id='topic+itriplot_render'></span><span id='topic+qtlcharts-shiny'></span><span id='topic+scat2scat_output'></span><span id='topic+scat2scat_render'></span>

<h3>Description</h3>

<p>Output and render functions for using R/qtlcharts widgets within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iboxplot_output(outputId, width = "100%", height = "900")

iboxplot_render(expr, env = parent.frame(), quoted = FALSE)

idotplot_output(outputId, width = "100%", height = "530")

idotplot_render(expr, env = parent.frame(), quoted = FALSE)

iheatmap_output(outputId, width = "100%", height = "1000")

iheatmap_render(expr, env = parent.frame(), quoted = FALSE)

ipleiotropy_output(outputId, width = "100%", height = "580")

ipleiotropy_render(expr, env = parent.frame(), quoted = FALSE)

iplot_output(outputId, width = "100%", height = "580")

iplot_render(expr, env = parent.frame(), quoted = FALSE)

iplotCorr_output(outputId, width = "100%", height = "1000")

iplotCorr_render(expr, env = parent.frame(), quoted = FALSE)

iplotCurves_output(outputId, width = "100%", height = "1000")

iplotCurves_render(expr, env = parent.frame(), quoted = FALSE)

iplotMScanone_output(outputId, width = "100%", height = "580")

iplotMScanone_render(expr, env = parent.frame(), quoted = FALSE)

iplotMap_output(outputId, width = "100%", height = "680")

iplotMap_render(expr, env = parent.frame(), quoted = FALSE)

iplotRF_output(outputId, width = "100%", height = "1000")

iplotRF_render(expr, env = parent.frame(), quoted = FALSE)

iplotScanone_output(outputId, width = "100%", height = "580")

iplotScanone_render(expr, env = parent.frame(), quoted = FALSE)

iplotScantwo_output(outputId, width = "100%", height = "1000")

iplotScantwo_render(expr, env = parent.frame(), quoted = FALSE)

itriplot_output(outputId, width = "100%", height = "530")

itriplot_render(expr, env = parent.frame(), quoted = FALSE)

scat2scat_output(outputId, width = "100%", height = "530")

scat2scat_render(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iboxplot_output_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="iboxplot_output_+3A_width">width</code>, <code id="iboxplot_output_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like '&quot;100%&quot;',
'&quot;400px&quot;', '&quot;auto&quot;') or a number, which will be coerced to a
string and have '&quot;px&quot;' appended.</p>
</td></tr>
<tr><td><code id="iboxplot_output_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a networkD3 graph</p>
</td></tr>
<tr><td><code id="iboxplot_output_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate 'expr'.</p>
</td></tr>
<tr><td><code id="iboxplot_output_+3A_quoted">quoted</code></td>
<td>
<p>Is 'expr' a quoted expression (with 'quote()')? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>

<hr>
<h2 id='idotplot'>Interactive phenotype x genotype plot</h2><span id='topic+idotplot'></span>

<h3>Description</h3>

<p>Creates an interactive graph of phenotypes vs genotypes at a marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idotplot(x, y, indID = NULL, group = NULL, chartOpts = NULL, digits = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="idotplot_+3A_x">x</code></td>
<td>
<p>Vector of groups of individuals (e.g., a genotype)</p>
</td></tr>
<tr><td><code id="idotplot_+3A_y">y</code></td>
<td>
<p>Numeric vector (e.g., a phenotype)</p>
</td></tr>
<tr><td><code id="idotplot_+3A_indid">indID</code></td>
<td>
<p>Optional vector of character strings, shown with tool tips</p>
</td></tr>
<tr><td><code id="idotplot_+3A_group">group</code></td>
<td>
<p>Optional vector of categories for coloring points</p>
</td></tr>
<tr><td><code id="idotplot_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="idotplot_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplot()], [iplotPXG()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 100
g &lt;- sample(LETTERS[1:3], n, replace=TRUE)
y &lt;- rnorm(n, match(g, LETTERS[1:3])*10, 5)

idotplot(g, y)

</code></pre>

<hr>
<h2 id='iheatmap'>Interactive heat map</h2><span id='topic+iheatmap'></span>

<h3>Description</h3>

<p>Creates an interactive heatmap, with each cell linked to
plots of horizontal and vertical slices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iheatmap(z, x = NULL, y = NULL, chartOpts = NULL, digits = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iheatmap_+3A_z">z</code></td>
<td>
<p>Numeric matrix of dim 'length(x)' x 'length(y)'</p>
</td></tr>
<tr><td><code id="iheatmap_+3A_x">x</code></td>
<td>
<p>Vector of numeric values for the x-axis</p>
</td></tr>
<tr><td><code id="iheatmap_+3A_y">y</code></td>
<td>
<p>Vector of numeric values for the y-axis</p>
</td></tr>
<tr><td><code id="iheatmap_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="iheatmap_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the z-axis limits are from '-max(abs(z))'
to 'max(abs(z))', and negative cells are colored blue to white
which positive cells are colored white to red.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotCorr()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 101
x &lt;- y &lt;- seq(-2, 2, len=n)
z &lt;- matrix(ncol=n, nrow=n)
for(i in seq(along=x))
    for(j in seq(along=y))
        z[i,j] &lt;- x[i]*y[j]*exp(-x[i]^2 - y[j]^2)

iheatmap(z, x, y)

</code></pre>

<hr>
<h2 id='ipleiotropy'>Tool to explore pleiotropy</h2><span id='topic+ipleiotropy'></span>

<h3>Description</h3>

<p>Creates an interactive graph of a scatterplot of two phenotypes,
plus optionally the LOD curves for the two traits along one
chromosome, with a slider for selecting the locations of two QTL
which are then indicated on the LOD curves and the corresponding
genotypes used to color the points in the scatterplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ipleiotropy(
  cross,
  scanoneOutput = NULL,
  pheno.col = 1:2,
  lodcolumn = 1:2,
  chr = NULL,
  interval = NULL,
  fillgenoArgs = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ipleiotropy_+3A_cross">cross</code></td>
<td>
<p>(Optional) Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_scanoneoutput">scanoneOutput</code></td>
<td>
<p>(Optional) object of class '&quot;scanone&quot;',
as output from [qtl::scanone()].</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_pheno.col">pheno.col</code></td>
<td>
<p>Vector indicating two phenotype column in cross
object; either numeric or character strings (the latter being
the phenotype column names).</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>Vector of two numeric values indicating LOD score
columns to plot.</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_chr">chr</code></td>
<td>
<p>A single chromosome ID, as a character string.</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_interval">interval</code></td>
<td>
<p>A numeric vector of length 2, defining an interval
that indicates what portion of the chromosome should be
included.</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_fillgenoargs">fillgenoArgs</code></td>
<td>
<p>List of named arguments to pass to
[qtl::fill.geno()], if needed.</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="ipleiotropy_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>[qtl::fill.geno()]
is used to impute missing genotypes. In this case, arguments to
[qtl::fill.geno()] are passed as a list, for example
'fillgenoArgs=list(method=&quot;argmax&quot;, error.prob=0.002,
map.function=&quot;c-f&quot;)'.
</p>
<p>Individual IDs (viewable when hovering over a point in the
scatterplot of the two phenotypes) are taken from the input
'cross' object, using the [qtl::getid()] function in
R/qtl.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotScanone()], [iplotMScanone()],
[iplotPXG()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(fake.bc)
fake.bc &lt;- calc.genoprob(fake.bc[5,], step=1) # select chr 5
out &lt;- scanone(fake.bc, method="hk", pheno.col=1:2)

ipleiotropy(fake.bc, out)


# omit the LOD curves
ipleiotropy(fake.bc)

</code></pre>

<hr>
<h2 id='iplot'>Interactive scatterplot</h2><span id='topic+iplot'></span>

<h3>Description</h3>

<p>Creates an interactive scatterplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplot(x, y = NULL, group = NULL, indID = NULL, chartOpts = NULL, digits = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplot_+3A_x">x</code></td>
<td>
<p>Numeric vector of x values</p>
</td></tr>
<tr><td><code id="iplot_+3A_y">y</code></td>
<td>
<p>Numeric vector of y values</p>
</td></tr>
<tr><td><code id="iplot_+3A_group">group</code></td>
<td>
<p>Optional vector of categories for coloring the points</p>
</td></tr>
<tr><td><code id="iplot_+3A_indid">indID</code></td>
<td>
<p>Optional vector of character strings, shown with tool tips</p>
</td></tr>
<tr><td><code id="iplot_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="iplot_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotCorr()], [iplotCurves()], [itriplot()],
[idotplot()], [iplotPXG()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- rnorm(100)
grp &lt;- sample(1:3, 100, replace=TRUE)
y &lt;- x*grp + rnorm(100)

iplot(x, y, grp)

</code></pre>

<hr>
<h2 id='iplotCorr'>Image of correlation matrix with linked scatterplot</h2><span id='topic+iplotCorr'></span>

<h3>Description</h3>

<p>Creates an interactive graph with an image of a
correlation matrix linked to underlying scatterplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotCorr(
  mat,
  group = NULL,
  rows = NULL,
  cols = NULL,
  reorder = FALSE,
  corr = NULL,
  scatterplots = TRUE,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotCorr_+3A_mat">mat</code></td>
<td>
<p>Data matrix (individuals x variables)</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_group">group</code></td>
<td>
<p>Optional vector of groups of individuals (e.g., a genotype)</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_rows">rows</code></td>
<td>
<p>Selected rows of the correlation matrix to include in
the image. Ignored if 'corr' is provided.</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_cols">cols</code></td>
<td>
<p>Selected columns of the correlation matrix to include
in the image. Ignored if 'corr' is provided.</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_reorder">reorder</code></td>
<td>
<p>If TRUE, reorder the variables by
clustering. Ignored if 'corr' is provided as a subset of the
overall correlation matrix</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_corr">corr</code></td>
<td>
<p>Correlation matrix (optional).</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_scatterplots">scatterplots</code></td>
<td>
<p>If &lsquo;FALSE', don&rsquo;t have the heat map be
linked to scatterplots.</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="iplotCorr_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>'corr' may be provided as a subset of the overall
correlation matrix for the columns of 'mat'. In this case, the
'reorder', 'rows' and 'cols' arguments are ignored. The row and
column names of 'corr' must match the names of some subset of
columns of 'mat'.
</p>
<p>Individual IDs are taken from 'rownames(mat)'; they must match
'names(group)'.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iheatmap()], [scat2scat()], [iplotCurves()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geneExpr)

iplotCorr(geneExpr$expr, geneExpr$genotype, reorder=TRUE,
          chartOpts=list(cortitle="Correlation matrix",
                         scattitle="Scatterplot"))

# use Spearman's correlation
corr &lt;- cor(geneExpr$expr, method="spearman", use="pairwise.complete.obs")
# order by hierarchical clustering
o &lt;- hclust(as.dist(1-corr))$order

iplotCorr(geneExpr$expr[,o], geneExpr$genotype, corr=corr[o,o],
          chartOpts=list(cortitle="Spearman correlation",
                         scattitle="Scatterplot"))
</code></pre>

<hr>
<h2 id='iplotCurves'>Plot of a bunch of curves, linked to points in scatterplots</h2><span id='topic+iplotCurves'></span>

<h3>Description</h3>

<p>Creates an interactive graph with a panel having a number of curves
(say, a phenotype measured over time) linked to one or two (or no) scatter plots
(say, of the first vs middle and middle vs last times).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotCurves(
  curveMatrix,
  times = NULL,
  scatter1 = NULL,
  scatter2 = NULL,
  group = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotCurves_+3A_curvematrix">curveMatrix</code></td>
<td>
<p>Matrix (dim n_ind x n_times) with outcomes</p>
</td></tr>
<tr><td><code id="iplotCurves_+3A_times">times</code></td>
<td>
<p>Vector (length n_times) with time points for the
columns of curveMatrix</p>
</td></tr>
<tr><td><code id="iplotCurves_+3A_scatter1">scatter1</code></td>
<td>
<p>Matrix (dim n_ind x 2) with data for the first
scatterplot</p>
</td></tr>
<tr><td><code id="iplotCurves_+3A_scatter2">scatter2</code></td>
<td>
<p>Matrix (dim n_ind x 2) with data for the second
scatterplot</p>
</td></tr>
<tr><td><code id="iplotCurves_+3A_group">group</code></td>
<td>
<p>Optional vector of groups of individuals (e.g., a genotype)</p>
</td></tr>
<tr><td><code id="iplotCurves_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="iplotCurves_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotCorr()], [iplot()], [scat2scat()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># random growth curves, based on some data
times &lt;- 1:16
n &lt;- 100
start &lt;- rnorm(n, 5.2, 0.8)
slope1to5 &lt;- rnorm(n, 2.6, 0.5)
slope5to16 &lt;- rnorm(n, 0.24 + 0.09*slope1to5, 0.195)
y &lt;- matrix(ncol=16, nrow=n)
y[,1] &lt;- start
for(j in 2:5)
    y[,j] &lt;- y[,j-1] + slope1to5
for(j in 6:16)
    y[,j] &lt;- y[,j-1] + slope5to16
y &lt;- y + rnorm(prod(dim(y)), 0, 0.35)


iplotCurves(y, times, y[,c(1,5)], y[,c(5,16)],
            chartOpts=list(curves_xlab="Time", curves_ylab="Size",
                           scat1_xlab="Size at T=1", scat1_ylab="Size at T=5",
                           scat2_xlab="Size at T=5", scat2_ylab="Size at T=16"))

</code></pre>

<hr>
<h2 id='iplotMap'>Interactive genetic map plot</h2><span id='topic+iplotMap'></span>

<h3>Description</h3>

<p>Creates an interactive graph of a genetic marker map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotMap(
  map,
  chr = NULL,
  shift = FALSE,
  horizontal = FALSE,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotMap_+3A_map">map</code></td>
<td>
<p>Object of class '&quot;map&quot;', a list with each component
being a vector of marker positions. You can also provide an object of
class '&quot;cross&quot;', in which case the map is extracted with
[qtl::pull.map()].</p>
</td></tr>
<tr><td><code id="iplotMap_+3A_chr">chr</code></td>
<td>
<p>(Optional) Vector indicating the chromosomes to plot.</p>
</td></tr>
<tr><td><code id="iplotMap_+3A_shift">shift</code></td>
<td>
<p>If TRUE, shift each chromsome so that the initial marker
is at position 0.</p>
</td></tr>
<tr><td><code id="iplotMap_+3A_horizontal">horizontal</code></td>
<td>
<p>If TRUE, have chromosomes arranged horizontally</p>
</td></tr>
<tr><td><code id="iplotMap_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="iplotMap_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotScanone()], [iplotPXG()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(hyper)
map &lt;- pull.map(hyper)

iplotMap(map, shift=TRUE)

</code></pre>

<hr>
<h2 id='iplotMScanone'>Interactive LOD curve</h2><span id='topic+iplotMScanone'></span>

<h3>Description</h3>

<p>Creates an interactive graph of a set of single-QTL genome scans, as
calculated by [qtl::scanone()]. If 'cross' or
'effects' are provided, LOD curves will be linked to a panel
with estimated QTL effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotMScanone(
  scanoneOutput,
  cross = NULL,
  lodcolumn = NULL,
  pheno.col = NULL,
  times = NULL,
  effects = NULL,
  chr = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotMScanone_+3A_scanoneoutput">scanoneOutput</code></td>
<td>
<p>Object of class '&quot;scanone&quot;', as output
from [qtl::scanone()].</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_cross">cross</code></td>
<td>
<p>(Optional) Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>Numeric value indicating LOD score column to plot.</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_pheno.col">pheno.col</code></td>
<td>
<p>(Optional) Phenotype column in cross object.</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_times">times</code></td>
<td>
<p>(Optional) Vector (length equal to the number of LOD
score columns) with quantitative values to which the different LOD
score columns correspond (times of measurements, or something like
age or dose).  These need to be ordered and equally-spaced. If
omitted, the names of the columns in 'scanoneOutput' are used
and treated as qualitative.</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_effects">effects</code></td>
<td>
<p>(Optional) Estimated QTL effects, as obtained with
[estQTLeffects()].</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_chr">chr</code></td>
<td>
<p>(Optional) Optional vector indicating the chromosomes
for which LOD scores should be calculated. This should be a vector
of character strings referring to chromosomes by name; numeric
values are converted to strings. Refer to chromosomes with a
preceding - to have all chromosomes but those considered. A logical
(TRUE/FALSE) vector may also be used.</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="iplotMScanone_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If 'cross' is provided, Haley-Knott regression is
used to estimate QTL effects at each pseudomarker.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotScanone()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(grav)
library(qtl)
grav &lt;- calc.genoprob(grav, step=1)
grav &lt;- reduce2grid(grav)

# we're going to subset the phenotypes
phecol &lt;- seq(1, nphe(grav), by=5)

# the times were saved as an attributed
times &lt;- attr(grav, "time")[phecol]

# genome scan
out &lt;- scanone(grav, phe=phecol, method="hk")


# plot with qualitative labels on y-axis
iplotMScanone(out)


# plot with quantitative y-axis
iplotMScanone(out, times=times)

# estimate QTL effect for each time point at each genomic position
eff &lt;- estQTLeffects(grav, phe=seq(1, nphe(grav), by=5), what="effects")


# plot with QTL effects included (and with quantitative y-axis)
iplotMScanone(out, effects=eff, times=times,
              chartOpts=list(eff_ylab="QTL effect", eff_xlab="Time (hrs)"))

</code></pre>

<hr>
<h2 id='iplotPXG'>Interactive phenotype x genotype plot</h2><span id='topic+iplotPXG'></span>

<h3>Description</h3>

<p>Creates an interactive graph of phenotypes vs genotypes at a marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotPXG(
  cross,
  marker,
  pheno.col = 1,
  chartOpts = NULL,
  fillgenoArgs = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotPXG_+3A_cross">cross</code></td>
<td>
<p>Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="iplotPXG_+3A_marker">marker</code></td>
<td>
<p>Character string with marker name.</p>
</td></tr>
<tr><td><code id="iplotPXG_+3A_pheno.col">pheno.col</code></td>
<td>
<p>Phenotype column in cross object.</p>
</td></tr>
<tr><td><code id="iplotPXG_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="iplotPXG_+3A_fillgenoargs">fillgenoArgs</code></td>
<td>
<p>List of named arguments to pass to
[qtl::fill.geno()], if needed.</p>
</td></tr>
<tr><td><code id="iplotPXG_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function [qtl::fill.geno()] is used to
impute missing genotypes, with arguments passed as a list, for
example 'fillgenoArgs=list(method=&quot;argmax&quot;, error.prob=0.002,
map.function=&quot;c-f&quot;)'.
</p>
<p>Individual IDs (viewable when hovering over a point) are taken from
the input 'cross' object, using the [qtl::getid()]
function in R/qtl.
</p>
<p>By default, points are colored blue and pink according to whether
the marker genotype is observed or inferred, respectively.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[idotplot()], [iplot()], [iplotScanone()],
[iplotMap()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(hyper)
marker &lt;- sample(markernames(hyper), 1)

iplotPXG(hyper, marker)

# different colors
iplotPXG(hyper, marker, chartOpts=list(pointcolor=c("black", "gray")))

</code></pre>

<hr>
<h2 id='iplotRF'>Interactive plot of recombination fractions</h2><span id='topic+iplotRF'></span>

<h3>Description</h3>

<p>Creates an interactive graph of estimated recombination fractions and LOD scores for all pairs of markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotRF(cross, chr = NULL, chartOpts = NULL, digits = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotRF_+3A_cross">cross</code></td>
<td>
<p>Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="iplotRF_+3A_chr">chr</code></td>
<td>
<p>Optional vector indicating chromosomes to include. This
should be a vector of character strings referring to chromosomes by
name; numeric values are converted to strings.  Refer to
chromosomes with a preceding '-' to have all chromosomes but
those considered.  A logical (TRUE/FALSE) vector may also be used.</p>
</td></tr>
<tr><td><code id="iplotRF_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="iplotRF_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The usual 'height' and 'width' options in
'chartOpts' are ignored in this plot. Instead, you may provide
'pixelPerCell' (number of pixels per cell in the heat map),
'chrGap' (gap in pixels between chromosomes in the heat map),
'cellHeight' (height in pixels of each cell in the
cross-tabulation), 'cellWidth' (width in pixels of each cell
in the cross-tabulation), and 'hbot' (height in pixels of the
lower panels showing cross-sections of the heat map)
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[qtl::est.rf()], [qtl::plotRF()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(fake.f2)

fake.f2 &lt;- est.rf(fake.f2)

iplotRF(fake.f2)

</code></pre>

<hr>
<h2 id='iplotScanone'>Interactive LOD curve</h2><span id='topic+iplotScanone'></span>

<h3>Description</h3>

<p>Creates an interactive graph of a single-QTL genome scan, as
calculated by [qtl::scanone()]. If 'cross' is
provided, the LOD curves are linked to a phenotype x genotype plot
for a marker: Click on a marker on the LOD curve and see the
corresponding phenotype x genotype plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotScanone(
  scanoneOutput,
  cross = NULL,
  lodcolumn = 1,
  pheno.col = 1,
  chr = NULL,
  pxgtype = c("ci", "raw"),
  fillgenoArgs = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotScanone_+3A_scanoneoutput">scanoneOutput</code></td>
<td>
<p>Object of class '&quot;scanone&quot;', as output
from [qtl::scanone()].</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_cross">cross</code></td>
<td>
<p>(Optional) Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>Numeric value indicating LOD score column to plot.</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_pheno.col">pheno.col</code></td>
<td>
<p>(Optional) Phenotype column in cross object.</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_chr">chr</code></td>
<td>
<p>(Optional) Vector indicating the chromosomes
for which LOD scores should be calculated. This should be a vector
of character strings referring to chromosomes by name; numeric
values are converted to strings. Refer to chromosomes with a
preceding - to have all chromosomes but those considered. A logical
(TRUE/FALSE) vector may also be used.</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_pxgtype">pxgtype</code></td>
<td>
<p>If phenotype x genotype plot is to be shown, should
it be with means <code class="reqn">\pm</code> 2 SE ('&quot;ci&quot;'), or raw
phenotypes ('&quot;raw&quot;')?</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_fillgenoargs">fillgenoArgs</code></td>
<td>
<p>List of named arguments to pass to
[qtl::fill.geno()], if needed.</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td></tr>
<tr><td><code id="iplotScanone_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If 'cross' is provided, [qtl::fill.geno()]
is used to impute missing genotypes. In this case, arguments to
[qtl::fill.geno()] are passed as a list, for example
'fillgenoArgs=list(method=&quot;argmax&quot;, error.prob=0.002,
map.function=&quot;c-f&quot;)'.
</p>
<p>With 'pxgtype=&quot;raw&quot;', individual IDs (viewable when hovering
over a point in the phenotype-by-genotype plot) are taken from the
input 'cross' object, using the [qtl::getid()]
function in R/qtl.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotMScanone()], [iplotPXG()], [iplotMap()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(hyper)
hyper &lt;- calc.genoprob(hyper, step=1)
out &lt;- scanone(hyper)

# iplotScanone with no effects
iplotScanone(out, chr=c(1, 4, 6, 7, 15))


# iplotScanone with CIs
iplotScanone(out, hyper, chr=c(1, 4, 6, 7, 15))


# iplotScanone with raw phe x gen
iplotScanone(out, hyper, chr=c(1, 4, 6, 7, 15),
             pxgtype='raw')

</code></pre>

<hr>
<h2 id='iplotScantwo'>Interactive plot of 2d genome scan</h2><span id='topic+iplotScantwo'></span>

<h3>Description</h3>

<p>Creates an interactive plot of the results of
[qtl::scantwo()], for a two-dimensional, two-QTL genome
scan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplotScantwo(
  scantwoOutput,
  cross = NULL,
  lodcolumn = 1,
  pheno.col = 1,
  chr = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplotScantwo_+3A_scantwooutput">scantwoOutput</code></td>
<td>
<p>Output of [qtl::scantwo()]</p>
</td></tr>
<tr><td><code id="iplotScantwo_+3A_cross">cross</code></td>
<td>
<p>(Optional) Object of class '&quot;cross&quot;', see
[qtl::read.cross()].</p>
</td></tr>
<tr><td><code id="iplotScantwo_+3A_lodcolumn">lodcolumn</code></td>
<td>
<p>Numeric value indicating LOD score column to plot.</p>
</td></tr>
<tr><td><code id="iplotScantwo_+3A_pheno.col">pheno.col</code></td>
<td>
<p>(Optional) Phenotype column in cross object.</p>
</td></tr>
<tr><td><code id="iplotScantwo_+3A_chr">chr</code></td>
<td>
<p>(Optional) Optional vector indicating the chromosomes
for which LOD scores should be calculated. This should be a vector
of character strings referring to chromosomes by name; numeric
values are converted to strings. Refer to chromosomes with a
preceding - to have all chromosomes but those considered. A logical
(TRUE/FALSE) vector may also be used.</p>
</td></tr>
<tr><td><code id="iplotScantwo_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="iplotScantwo_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The estimated QTL effects, and the genotypes in the
phenotype x genotype plot, in the right-hand panels, are derived
following a single imputation to fill in missing data, and so are a
bit crude.
</p>
<p>Note that the usual 'height' and 'width' options in
'chartOpts' are ignored here. Instead, you may provide
'pixelPerCell' (number of pixels per cell in the heat map),
'chrGap' (gaps between chr in heat map), 'wright' (width
in pixels of right panels), and 'hbot' (height in pixels of
each of the lower panels)
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotScanone()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(qtl)
data(fake.f2)

fake.f2 &lt;- calc.genoprob(fake.f2, step=5)
out &lt;- scantwo(fake.f2, method="hk", verbose=FALSE)

iplotScantwo(out, fake.f2)

</code></pre>

<hr>
<h2 id='itriplot'>Interactive plot of trinomial probabilities</h2><span id='topic+itriplot'></span>

<h3>Description</h3>

<p>Creates an interactive graph of trinomial probabilities,
represented as points within an equilateral triangle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itriplot(p, indID = NULL, group = NULL, chartOpts = NULL, digits = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itriplot_+3A_p">p</code></td>
<td>
<p>Matrix of trinomial probabilities (n x 3); each row should sum to 1.</p>
</td></tr>
<tr><td><code id="itriplot_+3A_indid">indID</code></td>
<td>
<p>Optional vector of character strings, shown with tool tips</p>
</td></tr>
<tr><td><code id="itriplot_+3A_group">group</code></td>
<td>
<p>Optional vector of categories for coloring the points</p>
</td></tr>
<tr><td><code id="itriplot_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="itriplot_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplot()], [iplotPXG()], [idotplot()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 100
p &lt;- matrix(runif(3*n), ncol=3)
p &lt;- p / colSums(p)
g &lt;- sample(1:3, n, replace=TRUE)

itriplot(p, group=g)

</code></pre>

<hr>
<h2 id='qtlchartsversion'>print the installed version of R/qtlcharts</h2><span id='topic+qtlchartsversion'></span>

<h3>Description</h3>

<p>print the installed version of R/qtlcharts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtlchartsversion()
</code></pre>


<h3>Value</h3>

<p>Character string with version number
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qtlchartsversion()

</code></pre>

<hr>
<h2 id='scat2scat'>Scatterplot driving another scatterplot</h2><span id='topic+scat2scat'></span>

<h3>Description</h3>

<p>A pair of linked scatterplots, where each point the first
scatterplot corresponds to a scatter of points in the second
scatterplot. The first scatterplot corresponds to a pair of summary
measures for a larger dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scat2scat(scat1data, scat2data, group = NULL, chartOpts = NULL, digits = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scat2scat_+3A_scat1data">scat1data</code></td>
<td>
<p>Matrix with two columns; rownames are used as
identifiers. Can have an optional third column with categories for
coloring points in the first scatterplot (to be used if
'group' is not provided).</p>
</td></tr>
<tr><td><code id="scat2scat_+3A_scat2data">scat2data</code></td>
<td>
<p>List of matrices each with at least two columns,
to be shown in the second scatterplot. The components of the list
correspond to the rows in 'scat1dat'. An option third column
can contain categories. Row names identify individual points.</p>
</td></tr>
<tr><td><code id="scat2scat_+3A_group">group</code></td>
<td>
<p>Categories for coloring points in the first
scatterplot; length should be the number of rows in
'scat1data'.</p>
</td></tr>
<tr><td><code id="scat2scat_+3A_chartopts">chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td></tr>
<tr><td><code id="scat2scat_+3A_digits">digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotCorr()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some data
p &lt;- 500
n &lt;- 300
SD &lt;- runif(p, 1, 5)
r &lt;- runif(p, -1, 1)
scat2 &lt;- vector("list", p)
for(i in 1:p)
   scat2[[i]] &lt;- matrix(rnorm(2*n), ncol=2) %*% chol(SD[i]^2*matrix(c(1, r[i], r[i], 1), ncol=2))
scat1 &lt;- cbind(SD=SD, r=r)
# plot it
scat2scat(scat1, scat2)

</code></pre>

<hr>
<h2 id='setScreenSize'>Set default maximum screen size</h2><span id='topic+setScreenSize'></span>

<h3>Description</h3>

<p>Set the default screen size as a global option.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setScreenSize(size = c("normal", "small", "large"), height, width)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setScreenSize_+3A_size">size</code></td>
<td>
<p>Character vector representing screen size (normal,
small, large). Ignored if height and width are provided.</p>
</td></tr>
<tr><td><code id="setScreenSize_+3A_height">height</code></td>
<td>
<p>(Optional) Height in pixels</p>
</td></tr>
<tr><td><code id="setScreenSize_+3A_width">width</code></td>
<td>
<p>(Optional) Width in pixels</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Used to set a global option, 'qtlchartsScreenSize', that
contains the maximum height and maximum width for a chart in the
browser.
</p>
<p>'&quot;small&quot;', '&quot;normal&quot;', and '&quot;large&quot;' correspond to 600x900, 700x1000, and
1200x1600, for height x width, respectively.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>setScreenSize("large")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
