<!DOCTYPE html><html lang="en"><head><title>Help for package qtlDesign</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtlDesign}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Confidence+20interval+20expected+20widths'><p>Calculating expected QTL confidence interval widths</p></a></li>
<li><a href='#Information'><p>Information under null hypothesis of equal means</p></a></li>
<li><a href='#Information-cost+20functions'><p>Functions to calculate information-cost ratios</p></a></li>
<li><a href='#K1'><p>Calculate scores for minimum moment abberations.</p></a></li>
<li><a href='#mma'><p>Selective phenotyping with similarity measure 2</p></a></li>
<li><a href='#mma.level'><p>MMA utility</p></a></li>
<li><a href='#Optimal+20marker+20spacing'><p>Optimal marker spacing</p></a></li>
<li><a href='#Optimal+20selection+20fraction'><p>Optimal selection fraction</p></a></li>
<li><a href='#Package+20version'><p>Version of qtlDesign package</p></a></li>
<li><a href='#Power+20calculations'><p>Power, sample size, and detectable effect size calculations</p></a></li>
<li><a href='#qtlDesign-internal'><p>qtlDesign-internal</p></a></li>
<li><a href='#Thresholds+20and+20tail+20probabilities'><p>Calculating thresholds and tail probabilities for genome scans</p></a></li>
<li><a href='#Utility'><p>Utility functions</p></a></li>
<li><a href='#Variance+20and+20effect+20size'><p>Effect size, proportion variance explained, and error variance</p>
calculations</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Design of QTL (Quantitative Trait Locus) Experiments</td>
</tr>
<tr>
<td>Version:</td>
<td>0.953</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-04-12</td>
</tr>
<tr>
<td>Description:</td>
<td>Design of QTL (quantitative trait locus) experiments
  involves choosing which strains to cross, the type of cross,
  genotyping strategies, phenotyping strategies, and the number of
  progeny to raise and phenotype. This package provides tools to help
  make such choices.
  Sen and others (2007) &lt;<a href="https://doi.org/10.1007%2Fs00335-006-0090-y">doi:10.1007/s00335-006-0090-y</a>&gt;.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Saunak Sen &lt;sen@uthsc.edu&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Saunak Sen [aut, cre],
  Jaya Satagopan [ctb],
  Karl Broman [ctb],
  Gary Churchill [ctb],
  Brian Yandell [ctb]</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://www.senresearch.org">http://www.senresearch.org</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>qtl</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-13 14:45:27 UTC; sen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-15 15:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='Confidence+20interval+20expected+20widths'>Calculating expected QTL confidence interval widths</h2><span id='topic+ci.length'></span>

<h3>Description</h3>

<p>Provides expected confidence interval widths for QTL location when we
have  dense markers.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ci.length(cross,n,effect,p=0.95,sigma2=1,env.var,gen.var,bio.reps=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_cross">cross</code></td>
<td>
<p>String indicating cross type which is &quot;bc&quot;, for
backcross, &quot;f2&quot; for intercross, and &quot;ri&quot; for recombinant inbred lines.</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_n">n</code></td>
<td>
<p>Sample size</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_p">p</code></td>
<td>
<p>Confidence level for desired confidence interval</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_effect">effect</code></td>
<td>
<p>The QTL effect we want to detect.  For
<code>powercalc</code> and <code>samplesize</code> this is a numeric (vector).
For <code>detectable</code> it specifies the relative magnitude of the
additive and dominance components for the intercross.
The specification of <code>effect</code> depends on the cross.  For
backcross, it is the difference in means the heterozygote and
homozygote.  For RI lines it is half the difference in means of the
homozygotes, for intercross, it is a two component vector of the form
<code>c(a,d)</code>, where <code>a</code> is the additive effect (half the
difference between the homozygotes), and <code>d</code> is the dominance
effect (difference between the heterozygote and the average of the
homozygotes). The genotype means will be <code>-a-d/2</code>, <code>d/2</code>,
and <code>a-d/2.</code>  For <code>detectable</code>, optionally for the
intercross, one can use a string to specify the QTL effect type.
The  strings &quot;add&quot; or &quot;dom&quot; are used to denote an additive or
dominant model respectively for the phenotype.  It may be 
it can be a numerical vector of the form <code>c(a,d)</code> indicating
the relative magnitudes of the additive and dominance components (as
defined above).  The default is &quot;add&quot;.</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_sigma2">sigma2</code></td>
<td>
<p>Error variance; if this argument is absent,
<code>env.var</code> and <code>gen.var</code> must be specified.</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_env.var">env.var</code></td>
<td>
<p>Environmental (within genotype) variance</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_gen.var">gen.var</code></td>
<td>
<p>Genetic (between genotype) variance due to all loci
segregating between the parental lines.</p>
</td></tr>
<tr><td><code id="Confidence+2B20interval+2B20expected+2B20widths_+3A_bio.reps">bio.reps</code></td>
<td>
<p>Number of biological replicates per unique genotype.
This is usually 1 for backcross and intercross, but may be larger
for RI lines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With dense markers, the log likelihood follows a compound
process.  Approximate expected confidence intervals can be calculated
by pretending the log likelihood decays linearly with a drift rate
that depends on the effect size and cross type.
</p>


<h3>Value</h3>

<p>Returns the expected confidence interval width (scalar) in cM
assuming dense markers.</p>


<h3>Author(s)</h3>

<p>Saunak Sen</p>


<h3>References</h3>

<p>Dupuis J and Siegmund D (1999) Statistical methods for
mapping quantitative trait loci from a dense set of markers. Genetics
151:373-386.
</p>
<p>Darvasi A (1998) Experimental strategies for the genetic
dissection of complex traits in animal models.  Nature Genetics
18:19-24.
</p>
<p>Kong A and Wright FA (1994) Asymptotic theory for gene mapping.
Proceedings of the National Academy of Sciences of the USA
91:9705-9709.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+powercalc">powercalc</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>ci.length(cross="bc",n=400,effect=5,p=0.95,sigma2=1)
</code></pre>

<hr>
<h2 id='Information'>Information under null hypothesis of equal means</h2><span id='topic+info'></span><span id='topic+info.bc'></span><span id='topic+info.f2'></span><span id='topic+deflate'></span><span id='topic+deflate.bc'></span><span id='topic+deflate.f2'></span><span id='topic+nullinfo'></span>

<h3>Description</h3>

<p>Functions to calculate the information under the null hypothesis of
no effect.  Functions for discount factors for incomplete genotyping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>info(sel.frac,theta=0,cross)
info.bc(sel.frac,theta=0)
info.f2(sel.frac,theta=0)
deflate(theta,cross)
deflate.bc(theta)
deflate.f2(theta)
nullinfo(sel.frac)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Information_+3A_cross">cross</code></td>
<td>
<p>Cross type, either &quot;bc&quot; for backcross, or &quot;f2&quot; for intercross.</p>
</td></tr>
<tr><td><code id="Information_+3A_sel.frac">sel.frac</code></td>
<td>
<p>Selection fraction; proportion of extremes genotyped</p>
</td></tr>
<tr><td><code id="Information_+3A_theta">theta</code></td>
<td>
<p>Recombination fraction between flanking markers</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>nullinfo</code> function calculates the information
content per observation for any contrast between genotype means when
densely genotyping an <code>sel.frac</code> fraction of
the extreme phenotypic individuals.  The information content is
calculated under the null hypothesis of no difference between the
genotype means.  For small differences in genotype means, the
information content will be approximately equal to the null, but in
general, the information estimate under the null is the lower bound.
</p>
<p>The <code>info</code> function calculates  the information per observation
for  backcross, and F2 intercross  under the null hypothesis of equal
gentoype means.  The   information is calculated for a point in the
middle of an interval  spanned by markers separated by a recombination
fraction <code>theta</code>.  The function <code>deflate</code> calculates a
deflation factor for the information attenuation in the middle of a
marker interval relative to a completely typed location.
</p>


<h3>Value</h3>

<p>Information per individual for information functions, and the
discount factor for the discount functions.
</p>


<h3>Note</h3>

<p>Information is calculated under the equal means assumption.  This
approximation is very good in practice, and is slightly conservative.
If the difference between the means is large, these functions will
underestimate the information.  For power calculations, that is okay.</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary Churchill</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>Examples</h3>

<pre><code class='language-R'>nullinfo(0.5)
info(0.5,cross="bc")
info(0.5,cross="f2")
info(0.5,0.1,cross="bc")
info(0.5,0.1,cross="f2")
deflate(0.1,"bc")
deflate(0.1,"f2")
</code></pre>

<hr>
<h2 id='Information-cost+20functions'>Functions to calculate information-cost ratios</h2><span id='topic+info2cost'></span><span id='topic+info2cost.bc'></span><span id='topic+info2cost.f2'></span>

<h3>Description</h3>

<p>Functions to calculate information cost-ratios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>info2cost(sel.frac,cost,d,G=NULL,cross)
info2cost.bc(sel.frac,cost,d,G=NULL)
info2cost.f2(sel.frac,cost,d,G=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Information-cost+2B20functions_+3A_sel.frac">sel.frac</code></td>
<td>
<p>Selection fraction; proportion of individuals genotyped</p>
</td></tr>
<tr><td><code id="Information-cost+2B20functions_+3A_cost">cost</code></td>
<td>
<p>Genotyping cost in units of raising an individual.  When
<code>d=0</code> (dense genotyping), it is the cost of genotyping an
individual.  When <code>d!=0</code>, it is the cost of a single marker
genotype in an individual.</p>
</td></tr>
<tr><td><code id="Information-cost+2B20functions_+3A_d">d</code></td>
<td>
<p>Marker spacing in centiMorgans</p>
</td></tr>
<tr><td><code id="Information-cost+2B20functions_+3A_g">G</code></td>
<td>
<p>Genome size in Morgans</p>
</td></tr>
<tr><td><code id="Information-cost+2B20functions_+3A_cross">cross</code></td>
<td>
<p>Cross type, &quot;bc&quot;or &quot;f2&quot;</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>The information calculations are done under the null hypothesis
of no QTL effect.</p>


<h3>Value</h3>

<p>For <code>d!=0</code> it calculates the ratio of information in the
middle of a marker interval of length <code>d</code> cM to the cost of
genotyping the cross.  For <code>d=0</code>, it calculates the ratio of
information at any locus to the cost of genotyping the cross.</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary Churchill</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>See Also</h3>

<p><code><a href="#topic+info">info</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>info2cost(0.5,1,cross="bc")
info2cost(0.5,1,10,1450,cross="bc")
</code></pre>

<hr>
<h2 id='K1'>Calculate scores for minimum moment abberations.</h2><span id='topic+K1'></span><span id='topic+K12'></span><span id='topic+Kstat'></span><span id='topic+eff1'></span>

<h3>Description</h3>

<p>Calculate the MMA K1, K12, and the standardized dissimilarity score (eff1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Kstat(genomat, type = 1)
K1(genomat)
K12(genomat)
eff1(n, nmark, s1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="K1_+3A_genomat">genomat</code></td>
<td>
<p>Genotype matrix.</p>
</td></tr>
<tr><td><code id="K1_+3A_n">n</code></td>
<td>
<p>Desired sample size.</p>
</td></tr>
<tr><td><code id="K1_+3A_type">type</code></td>
<td>
<p>Type of dissimilarity measure desired (first or second
moment).</p>
</td></tr>
<tr><td><code id="K1_+3A_nmark">nmark</code></td>
<td>
<p>Number of markers.</p>
</td></tr>  
<tr><td><code id="K1_+3A_s1">s1</code></td>
<td>
<p>Dissimilarity score from <code>K1</code> or <code>K12</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Score or standardized score based on selected marker list.
<code>K1</code> and <code>K12</code> call <code>Kstat</code> with <code>type</code> = 1 and 2,
respectively. Kstat computes the minimum moment abberation
score. <code>eff1</code> computes the standardized genetic dissimilarity.
</p>


<h3>Author(s)</h3>

<p>Brian S. Yandell (<a href="mailto:byandell@wisc.edu">mailto:byandell@wisc.edu</a>)</p>


<h3>References</h3>

<p>Jin C, Lan H, Attie AD, Churchill GA, Bulutuglo D, Yandell BS (2004) Selective phenotyping for increased efficiency in genetic mapping studies. Genetics 168: 2285-2293.</p>


<h3>See Also</h3>

<p><code><a href="#topic+mma">mma</a></code>, <code><a href="qtl.html#topic+read.cross">read.cross</a></code></p>

<hr>
<h2 id='mma'>Selective phenotyping with similarity measure 2</h2><span id='topic+mma'></span>

<h3>Description</h3>

<p>Selective phenotyping with similarity measure 2 to select the most
dissimilar subset of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mma(genof, p, sequent = FALSE, exact = FALSE, dismat = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mma_+3A_genof">genof</code></td>
<td>
<p>Genotype matrix.</p>
</td></tr>
<tr><td><code id="mma_+3A_p">p</code></td>
<td>
<p>Sample size to select.</p>
</td></tr>
<tr><td><code id="mma_+3A_sequent">sequent</code></td>
<td>
<p>Perform sequential optimization if TRUE (see below).</p>
</td></tr>
<tr><td><code id="mma_+3A_exact">exact</code></td>
<td>
<p>Count allele differences if <code>FALSE</code>; binary 0 = same
number of alleles, 1 = different if <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="mma_+3A_dismat">dismat</code></td>
<td>
<p>Return dissimilarity matrix if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sequentially minimize 1st moment and then 2nd moment, swapping one
subject at a time.
<code>op</code> finds all the samples with same 1st moment similarity with mma
results. <code>op2</code> finds all the samples with the same 1st moment
similarity with every list from op result. A combination of <code>op</code>
and <code>op2</code> comes very close to exhaustive search in
practice. <code>moment2</code> find the best list with minimum 2nd moments
from the output of <code>op2</code>. Note that some warnings occurs
accompanying our return statement. The results are not affected though.
</p>
<p>This function combines several functions in Jin's original code.
<code>mma(genof,p,sequent=TRUE</code> is identical to the depricated
<code>mmasequent(genof,p</code>.
<code>mma(genof,p,exact=TRUE</code> is identical to the depricated
<code>mmaM1(genof,p</code> (actually, <code>mma</code> uses dissimilarity while
<code>mmaM1</code> used similarity = 1 - dissimilarity).
</p>


<h3>Value</h3>

<p>A list containing <code>cList</code>, <code>dismat</code> if that option is
<code>TRUE</code> and further optimized lists (<code>op</code>, <code>op2</code>,
<code>moment2</code>) if <code>sequent</code> is <code>TRUE</code>. 
vector as the first item. The list of items includes:
</p>
<table role = "presentation">
<tr><td><code>cList</code></td>
<td>
<p>vector of selected subjects by function mma</p>
</td></tr>
<tr><td><code>op</code></td>
<td>
<p>list containing vector of selection and update flag from
function op</p>
</td></tr>
<tr><td><code>op2</code></td>
<td>
<p>matrix of selection by function op2</p>
</td></tr>
<tr><td><code>moment2</code></td>
<td>
<p>vector of second moment calculations</p>
</td></tr>
<tr><td><code>dismat</code></td>
<td>
<p>dissimilarity matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Brian S. Yandell (<a href="mailto:byandell@wisc.edu">mailto:byandell@wisc.edu</a>)</p>


<h3>References</h3>

<p>Jin C, Lan H, Attie AD, Churchill GA, Bulutuglo D, Yandell BS (2004) Selective phenotyping for increased efficiency in genetic mapping studies. Genetics 168: 2285-2293.</p>


<h3>See Also</h3>

<p><code><a href="#topic+K1">K1</a></code>, <code><a href="qtl.html#topic+read.cross">read.cross</a></code></p>

<hr>
<h2 id='mma.level'>MMA utility</h2><span id='topic+mma.level'></span>

<h3>Description</h3>

<p>This routine is for internal use. It sets 3 levels to 0,1,2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mma.level(mat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mma.level_+3A_mat">mat</code></td>
<td>
<p>input matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts matrix to levels between 0 and 2.
</p>


<h3>Value</h3>

<p>Matrix of genotype levels between 0 and 2.
</p>


<h3>Author(s)</h3>

<p>Brian S. Yandell (<a href="mailto:byandell@wisc.edu">mailto:byandell@wisc.edu</a>)</p>


<h3>References</h3>

<p>Jin C, Lan H, Attie AD, Churchill GA, Bulutuglo D, Yandell BS (2004) Selective phenotyping for increased efficiency in genetic mapping studies. Genetics 168: 2285-2293.</p>


<h3>See Also</h3>

<p><code><a href="#topic+mma">mma</a></code>, <code><a href="qtl.html#topic+read.cross">read.cross</a></code></p>

<hr>
<h2 id='Optimal+20marker+20spacing'>Optimal marker spacing</h2><span id='topic+optspacing'></span><span id='topic+optspacing.bc'></span><span id='topic+optspacing.f2'></span>

<h3>Description</h3>

<p>Functions to find optimal marker spacing given cost.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optspacing(cost,G=NULL,sel.frac,cross)
optspacing.bc(cost,G=NULL,sel.frac)
optspacing.f2(cost,G=NULL,sel.frac)
optspacing(cost,G=NULL,sel.frac=NULL,cross)
optspacing.bc(cost,G=NULL,sel.frac=NULL)
optspacing.f2(cost,G=NULL,sel.frac=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Optimal+2B20marker+2B20spacing_+3A_cost">cost</code></td>
<td>
<p>Cost of genotyping in units of raising an individual</p>
</td></tr>
<tr><td><code id="Optimal+2B20marker+2B20spacing_+3A_sel.frac">sel.frac</code></td>
<td>
<p>Selection fraction; proportion of individuals genotyped</p>
</td></tr>
<tr><td><code id="Optimal+2B20marker+2B20spacing_+3A_g">G</code></td>
<td>
<p>Genome size in centiMorgans</p>
</td></tr>
<tr><td><code id="Optimal+2B20marker+2B20spacing_+3A_cross">cross</code></td>
<td>
<p>Cross type, &quot;bc&quot; or &quot;f2&quot;</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The function <code>optim</code> is used to search for the optima.
</p>


<h3>Value</h3>

<p>In the first form, with the selection fraction specified, the
spacing in centiMorgans that maximizes the information to cost ratio
in the middle of the marker interval.  In the second form, with the
selection fraction unspecified, it returns the value of
(<code>spacing</code>,<code>sel.frac</code>) which maximizes the information
to cost ratio in the middle of the marker interval.</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary Churchill</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+optim">optim</a></code>, <code><a href="stats.html#topic+optimize">optimize</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>optspacing(cost=0.1,G=1440,sel.frac=0.5,cross="bc")
optspacing(cost=30/3000,G=1440,sel.frac=NULL,cross="f2")
</code></pre>

<hr>
<h2 id='Optimal+20selection+20fraction'>Optimal selection fraction</h2><span id='topic+optselection'></span><span id='topic+optselection.bc'></span><span id='topic+optselection.f2'></span>

<h3>Description</h3>

<p>Functions to find optimal selection fractions given cost.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optselection(cost,d=0,G=NULL,cross)
optselection.bc(cost,d=0,G=NULL)
optselection.f2(cost,d=0,G=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Optimal+2B20selection+2B20fraction_+3A_cost">cost</code></td>
<td>
<p>Cost per genotype in units of raising individual</p>
</td></tr>
<tr><td><code id="Optimal+2B20selection+2B20fraction_+3A_d">d</code></td>
<td>
<p>Marker spacing in Morgans</p>
</td></tr>
<tr><td><code id="Optimal+2B20selection+2B20fraction_+3A_g">G</code></td>
<td>
<p>Genome size in Morgans</p>
</td></tr>
<tr><td><code id="Optimal+2B20selection+2B20fraction_+3A_cross">cross</code></td>
<td>
<p>Cross type, &quot;bc&quot; or &quot;f2&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>optimize</code> is used to search for the optima.</p>


<h3>Value</h3>

<p>The optimal selection  fraction.</p>


<h3>Author(s)</h3>

<p>Saunak Sen</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+optimize">optimize</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>optselection(1,cross="bc")
optselection(0.001,10,1450,cross="bc")
optselection(0.001,10,1450,cross="f2")
</code></pre>

<hr>
<h2 id='Package+20version'>Version of qtlDesign package</h2><span id='topic+version.qtlDesign'></span>

<h3>Description</h3>

<p>Returns the version number for the qtlDesign package.</p>


<h3>Usage</h3>

<pre><code class='language-R'>version.qtlDesign()
</code></pre>


<h3>Value</h3>

<p>The version number.
</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary
Churchill</p>

<hr>
<h2 id='Power+20calculations'>Power, sample size, and detectable effect size calculations </h2><span id='topic+powercalc'></span><span id='topic+detectable'></span><span id='topic+samplesize'></span><span id='topic+powercalc.bc'></span><span id='topic+detectable.bc'></span><span id='topic+samplesize.bc'></span><span id='topic+powercalc.f2'></span><span id='topic+detectable.f2'></span><span id='topic+samplesize.f2'></span><span id='topic+powercalc.ri'></span><span id='topic+detectable.ri'></span><span id='topic+samplesize.ri'></span>

<h3>Description</h3>

<p>Power, sample size, and minimum detectable effect size calculations are
performed for backcross, F2 intercross, and recombinant inbred (RI)
lines.</p>


<h3>Usage</h3>

<pre><code class='language-R'>powercalc(cross,n,effect,sigma2,env.var,gen.var,thresh=3,sel.frac=1,
          theta=0,bio.reps=1)
detectable(cross,n,effect=NULL,sigma2,env.var,gen.var,power=0.8,thresh=3,
           sel.frac=1,theta=0,bio.reps=1)
samplesize(cross,effect,sigma2,env.var,gen.var,power=0.8,thresh=3,
           sel.frac=1,theta=0,bio.reps=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Power+2B20calculations_+3A_cross">cross</code></td>
<td>
<p>String indicating cross type which is &quot;bc&quot;, for
backcross, &quot;f2&quot; for intercross, and &quot;ri&quot; for recombinant inbred lines.</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_n">n</code></td>
<td>
<p>Sample size</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_sigma2">sigma2</code></td>
<td>
<p>Error variance; if this argument is absent,
<code>env.var</code> and <code>gen.var</code> must be specified.</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_env.var">env.var</code></td>
<td>
<p>Environmental (within genotype) variance</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_gen.var">gen.var</code></td>
<td>
<p>Genetic (between genotype) variance due to all loci
segregating between the parental lines.</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_effect">effect</code></td>
<td>
<p>The QTL effect we want to detect.  For
<code>powercalc</code> and <code>samplesize</code> this is a numeric (vector).
For <code>detectable</code> it specifies the relative magnitude of the
additive and dominance components for the intercross.
The specification of <code>effect</code> depends on the cross.  For
backcross, it is the difference in means the heterozygote and
homozygote.  For RI lines it is half the difference in means of the
homozygotes, for intercross, it is a two component vector of the form
<code>c(a,d)</code>, where <code>a</code> is the additive effect (half the
difference between the homozygotes), and <code>d</code> is the dominance
effect (difference between the heterozygote and the average of the
homozygotes). The genotype means will be <code>-a-d/2</code>, <code>d/2</code>,
and <code>a-d/2.</code>  For <code>detectable</code>, optionally for the
intercross, one can use a string to specify the QTL effect type.
The  strings &quot;add&quot; or &quot;dom&quot; are used to denote an additive or
dominant model respectively for the phenotype.  It may be 
it can be a numerical vector of the form <code>c(a,d)</code> indicating
the relative magnitudes of the additive and dominance components (as
defined above).  The default is &quot;add&quot;.</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_power">power</code></td>
<td>
<p>Proportion indicating power desired</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_thresh">thresh</code></td>
<td>
<p>LOD threshold for declaring significance</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_sel.frac">sel.frac</code></td>
<td>
<p>Selection fraction</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_theta">theta</code></td>
<td>
<p>Recombination fraction corresponding to a marker interval</p>
</td></tr>
<tr><td><code id="Power+2B20calculations_+3A_bio.reps">bio.reps</code></td>
<td>
<p>Number of biological replicates per unique genotype.
This is usually 1 for backcross and intercross, but may be larger
for RI lines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These calculations are done assuming that the asymptotic chi-square
regimes apply.  A warning message is printed if the effective sample size
is less than 30 and either <code>sel.frac</code> is less than 1 or <code>theta</code>
is greater than 0.  First we calculate the effective sample size using the
width of the marker interval and the selection fraction.  The QTL is
assumed to be in the middle of the marker interval.  Then we use the fact
that the non-centrality parameter of the likelihood ration test is
<code class="reqn">m*\delta^2</code>, where <code class="reqn">m</code> is the effctive sample size and
<code class="reqn">\delta</code> is the QTL effect measured as the deviation of the genotype
means from the overall mean.  The chi-squared approximation is used to
calculate the power.  The minimum detectable effect size is obtained by
solving the power equation numerically using <code>uniroot</code>.  The theory
behind the information calculations is described by Sen et. al. (2005).
</p>
<p>A key input is the error variance, <code>sigma2</code> which is generally
unknown.  The user can enter the error variance directly, or estimate it
using <code>env.var</code> and <code>gen.var</code>.  The function <code>error.var</code>
is used to the error variance using estimates of the environmental variance
and genetic variance.  Another key input is the effect segregating in
a cross, which can be calculated using <code>gmeans2model</code>.  </p>


<h3>Value</h3>

<p>For <code>powercalc</code> the power is returned, along with the
proportion of variance explained.  For <code>detectable</code> the effect size
detectable is returned, along with the proportion of variance explained.
For backcross and RI lines this is the effect of an allelic
substitution.  For F2 intercross the additive and dominance components
are returned.  For <code>samplesize</code> the sample size (rounded up to the
nearest integer) is returned along with the proportion of variance
explained.</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary Churchill</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+uniroot">uniroot</a></code>. <code><a href="#topic+error.var">error.var</a></code>,
<code><a href="#topic+gmeans2effect">gmeans2effect</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>powercalc("bc",100,5,sigma2=1,sel.frac=1,theta=0)
powercalc(cross="ri",n=30,effect=5,env.var=64,gen.var=25,bio.rep=6)
detectable("bc",100,sigma2=1)
detectable(cross="ri",n=30,env.var=64,gen.var=25,bio.rep=8)
samplesize(cross="f2",effect=c(5,0),env.var=64,gen.var=25)
</code></pre>

<hr>
<h2 id='qtlDesign-internal'>qtlDesign-internal</h2><span id='topic+qtlDesign-internal'></span><span id='topic+nu'></span><span id='topic+tau'></span>

<h3>Description</h3>

<p>Internal qtlDesign functions
</p>


<h3>Details</h3>

<p>These are not to be called by the user.
</p>


<h3>Value</h3>

<p>Scalar value of the nu and tau functions in Siegmund (1985).
</p>


<h3>Author(s)</h3>

<p>Saunak Sen</p>


<h3>References</h3>

<p>Siegmund, D., 1985 Sequential Analysis: Tests and Confidence
Intervals.  Springer-Verlag, New York.</p>

<hr>
<h2 id='Thresholds+20and+20tail+20probabilities'>Calculating thresholds and tail probabilities for genome scans</h2><span id='topic+tailprob'></span><span id='topic+thresh'></span>

<h3>Description</h3>

<p>Provides genome-wide thresholds and tail probabilities for the maxima of
genome scans using Poisson approximations.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tailprob(t,G,cross,type="1",d=0.01,cov.dim=0)
thresh(G,cross,type="1",p=c(0.10,0.05,0.01),d=0.01,cov.dim=0,
       interval=c(1,40))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_g">G</code></td>
<td>
<p>Genome size in centiMorgans.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_t">t</code></td>
<td>
<p>LOD value for which tail probability is desired.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_p">p</code></td>
<td>
<p>Vector giving the genome-wide Type I error for which
thresholds are desired.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_cross">cross</code></td>
<td>
<p>String indicating cross type which is &quot;bc&quot;, for 
backcross, &quot;f2&quot; for intercross.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_type">type</code></td>
<td>
<p>Type of LOD score for which threshold is desired.  Right
now the only option is &quot;1&quot;, but more options will be added in the
future.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_d">d</code></td>
<td>
<p>Marker spacing in centiMorgans.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_cov.dim">cov.dim</code></td>
<td>
<p>Dimension of interacting covariate.  Set to 0 right now.</p>
</td></tr>
<tr><td><code id="Thresholds+2B20and+2B20tail+2B20probabilities_+3A_interval">interval</code></td>
<td>
<p>Interval over which to search for LOD threshold.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The tail probabilities are calculated using the method of
Dupuis and Siegmund (1999).  The thresholds are calculated by solving
the tail probability equation using <code>uniroot</code>.  At this time only
one-dimensional thresholds are calculated, but this function will be
extended in the future.
</p>


<h3>Value</h3>

<p>The function <code>tailprob</code> returns the probability that the
genome-wide maximum LOD score exceeds a particular value.  The
function <code>thresh</code> returns genome-wide LOD thresholds
corresponding to a particular Type I error rate.
</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary
Churchill</p>


<h3>References</h3>

<p>Dupuis J and Siegmund D (1999) Statistical methods for
mapping quantitative trait loci froma dense set of markers. Genetics
151:373-386.</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+uniroot">uniroot</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>tailprob(t=3,G=1440,cross="f2",d=10)
thresh(G=1440,cross="bc",d=10)
</code></pre>

<hr>
<h2 id='Utility'>Utility functions</h2><span id='topic+recomb'></span><span id='topic+genetic.dist'></span>

<h3>Description</h3>

<p>Utility functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recomb(d)
genetic.dist(theta)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Utility_+3A_d">d</code></td>
<td>
<p>Genetic distance in Morgans</p>
</td></tr>
<tr><td><code id="Utility_+3A_theta">theta</code></td>
<td>
<p>Recombination fraction</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>recomb</code> returns the recombination fraction
corresponding to a genetic distance in Morgans.  <code>genetic.dist</code>
returns the genetic distance in Morgans for a recombination fraction.
</p>


<h3>Note</h3>

<p>We assume Haldane mapping function for the genetic distance.</p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary Churchill</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>Examples</h3>

<pre><code class='language-R'>recomb(0.1)
genetic.dist(0.1)
</code></pre>

<hr>
<h2 id='Variance+20and+20effect+20size'>Effect size, proportion variance explained, and error variance
calculations</h2><span id='topic+error.var'></span><span id='topic+gmeans2effect'></span><span id='topic+prop.var'></span>

<h3>Description</h3>

<p>The function <code>error.var</code> estimates the error variance using
estimates of the environmental variance and genetic variance.  The effect
segregating at a locus, can be calculated using <code>gmeans2effect</code>
These are key inputs for power calculations.  The function
<code>prop.var</code> calculates the proportion of variance explained by a
locus given the effect size and error variance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error.var(cross,env.var=1,gen.var=0,bio.reps=1)
gmeans2effect(cross,means)
prop.var(cross,effect,sigma2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_cross">cross</code></td>
<td>
<p>String indicating cross type which is &quot;bc&quot;, for
backcross, &quot;f2&quot; for intercross, and &quot;ri&quot; for recombinant inbred lines.</p>
</td></tr>
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_env.var">env.var</code></td>
<td>
<p>Environmental (within genotype) variance</p>
</td></tr>
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_gen.var">gen.var</code></td>
<td>
<p>Genetic (between genotype) variance due to all loci
segregating between the parental lines.</p>
</td></tr>
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_bio.reps">bio.reps</code></td>
<td>
<p>Number of biological replicates per unique genotype.
This is usually 1 for backcross and intercross, but may be larger
for RI lines.</p>
</td></tr>
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_means">means</code></td>
<td>
<p>Vector of genotype means in the form <code>c(a,h,b)</code>,
where <code>a</code> is the mean of the &quot;AA&quot; homozygotes, <code>h</code> is
the mean of the &quot;AB&quot; heterozygotes, and <code>b</code> is the mean of the
&quot;BB&quot; homozygotes.</p>
</td></tr>
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_effect">effect</code></td>
<td>
<p>The QTL effect which depends on the cross.  For
backcross, it is the difference in means the heterozygote and
homozygote.  For RI lines it is half the difference in means of the
homozygotes, for intercross, it is a two component vector of the form
<code>c(a,d)</code>, where <code>a</code> is the additive effect (half the
difference between the homozygotes), and <code>d</code> is the dominance
effect (difference between the heterozygote and the average of the
homozygotes). The genotype means will be <code>-a-d/2</code>, <code>d/2</code>,
and <code>a-d/2.</code> </p>
</td></tr>
<tr><td><code id="Variance+2B20and+2B20effect+2B20size_+3A_sigma2">sigma2</code></td>
<td>
<p>Error variance.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>error.var</code> estimates the error variance
segregating in a cross using estimates of the environmental (within
genotype) variance, and the genetic (between genotype variance).  The
environmental variance is assumed to be invariant between cross types.
The genetic variance segregating in RI lines is assumed to be double
that in F2 intercross, and four times that of the backcross.  This
assumption  holds if all loci are additive.  The error variance at a
locus of interest is aproximately </p>
<p style="text-align: center;"><code class="reqn">\sigma_G^2/c + \sigma_E^2/m,</code>
</p>

<p>where <code class="reqn">\sigma_G^2</code> is the genetic variance
(<code>gen.var</code>), <code class="reqn">c</code> is a
constant depending on the cross type (1, for RI lines, 1/2 for F2
intercross, and 1/4 for backross), <code class="reqn">\sigma_E^2</code> is the
environmental
variance (<code>env.var</code>), and <code class="reqn">m</code> is the number of biological
replicates per unique genotype (<code>bio.reps</code>).
</p>
<p>The function <code>gmeans2effect</code> calculates the QTL effects from
genotype means depending on the cross.
</p>
<p>The function <code>prop.var</code> calculates the proportion of variance
attributable to a locus given the effects size(s) and the error
variance.  The definition of effect size is in the output of
<code>gmeans2effect</code> (see below).
</p>


<h3>Value</h3>

<p>For <code>error.var</code> the value is the estimated error variance
based on the assumptions mentioned above.  For <code>gmeans2effect</code>
the value depends on the type of cross.  For RI lines it is simply the
additive effect of the QTL which is half the difference between the
homozygote means.  For intercross, it is a vector giving the additive and
dominance components.  The additive component is half the difference
between the homozygote means, and the dominance component is the
difference between the heterozygotes and the average of the
homozygotes.  For the backcross, it is a vector of length 2,
<code>c(a-h,h-b)</code>, which is the effect of an allelic substitution of
an &quot;A&quot; allele in the backcrosses to each parental strain. </p>


<h3>Author(s)</h3>

<p>Saunak Sen, Jaya Satagopan, Karl Broman, and Gary Churchill</p>


<h3>References</h3>

<p>Sen S, Satagopan JM, Churchill GA (2005) 	
Quantitative trait locus study design from an information perspective.
Genetics, 170:447-64.</p>


<h3>See Also</h3>

<p><code><a href="#topic+powercalc">powercalc</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>error.var(cross="bc",env.var=1,gen.var=1,bio.reps=1)
error.var(cross="f2",env.var=1,gen.var=1,bio.reps=1)
error.var(cross="ri",env.var=1,gen.var=1,bio.reps=1)
error.var(cross="ri",env.var=1,gen.var=1,bio.reps=10)
gmeans2effect(cross="f2",means=c(0,1,2))
gmeans2effect(cross="f2",means=c(0,1,1))
gmeans2effect(cross="bc",means=c(0,1,1))
gmeans2effect(cross="ri",means=c(0,1,1))
prop.var(cross="bc",effect=5,sigma2=1)
prop.var(cross="f2",effect=c(5,0),sigma2=1)
prop.var(cross="ri",effect=5,sigma2=1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
