<!DOCTYPE html><html><head><title>Help for package qtlmt</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {qtlmt}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dataSets'>
<p>Data sets</p></a></li>
<li><a href='#mAdd1'>
<p>Add or drop all possible terms</p></a></li>
<li><a href='#misFct'>
<p>Miscellaneous functions</p></a></li>
<li><a href='#mlogLik'>
<p>Extract log-likelihood</p></a></li>
<li><a href='#mStep'>
<p>Model selection in multivariate multiple regression</p></a></li>
<li><a href='#mtcmim'>
<p>MTCMIM</p></a></li>
<li><a href='#mtcmimStep'>
<p>MTCMIM model selection</p></a></li>
<li><a href='#sureEps'>
<p>Epistasis in a SURE model</p></a></li>
<li><a href='#sureEst'>
<p>SURE model parameter estimation</p></a></li>
<li><a href='#sureStep'>
<p>SURE model selection</p></a></li>
<li><a href='#varGroup'>
<p>Group variables</p></a></li>
<li><a href='#varSelect'>
<p>Variable selection</p></a></li>
<li><a href='#varStep'>
<p>Variable selection</p></a></li>
<li><a href='#varT2'>
<p>Hotelling's T^2 test statistic</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.1-6</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-07-20</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for Mapping Multiple Complex Traits</td>
</tr>
<tr>
<td>Author:</td>
<td>Riyan Cheng [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Riyan Cheng &lt;riyancheng@hotmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides tools for joint analysis of multiple traits in a backcross (BC) or recombinant inbred lines (RIL) population. It can be used to select an optimal subset of traits for multiple-trait mapping, analyze multiple traits via the SURE model, which can associate different QTL with different traits, and perform multiple-trait composite multiple-interval mapping.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>no</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-07-20 20:54:40 UTC; rcheng</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-07-25 20:31:45 UTC</td>
</tr>
</table>
<hr>
<h2 id='dataSets'>
Data sets
</h2><span id='topic+dists'></span><span id='topic+eps'></span><span id='topic+eqtl'></span><span id='topic+gmap'></span><span id='topic+maps'></span><span id='topic+markers'></span><span id='topic+gdat'></span><span id='topic+mdat'></span><span id='topic+mpos'></span><span id='topic+qtl'></span><span id='topic+traits'></span><span id='topic+v'></span><span id='topic+xid'></span>

<h3>Description</h3>

<p>A collection of data sets for demonstration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(etrait)
</code></pre>

<hr>
<h2 id='mAdd1'>
Add or drop all possible terms
</h2><span id='topic+mAdd1'></span><span id='topic+mDrop1'></span>

<h3>Description</h3>

<p>Add or drop all possible terms to or from a multivariate multiple regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mAdd1(object, scope, test = c("none","Chisq","F"), k = 0, ...)

mDrop1(object, scope, test = c("none","Chisq","F"), k = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mAdd1_+3A_object">object</code></td>
<td>

<p>initial model.
</p>
</td></tr>
<tr><td><code id="mAdd1_+3A_scope">scope</code></td>
<td>

<p>a formula or a vector of variable names, giving lower/upper bound of the model.
</p>
</td></tr>
<tr><td><code id="mAdd1_+3A_test">test</code></td>
<td>

<p>if not &quot;none&quot;, the test statistic and p-value will be included in the output.
</p>
</td></tr>
<tr><td><code id="mAdd1_+3A_k">k</code></td>
<td>

<p>penalty on a parameter in AIC.
</p>
</td></tr>
<tr><td><code id="mAdd1_+3A_...">...</code></td>
<td>

<p>additional arguments to <code><a href="stats.html#topic+update">update</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object summarizing the differences in fit between the models.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+add1">add1</a></code> and <code><a href="#topic+mStep">mStep</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
mdf&lt;- data.frame(traits,markers)
## Not run: 
mlm&lt;- lm(cbind(T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,T14,T15,T16) ~
   m1 + m2 + m3 + m4 + m5, data=mdf)

up&lt;- formula(paste("~", paste("m",1:15,collapse=" + ",sep="")))
oa&lt;- mAdd1(mlm, scope=up, test="F", k=5, data=mdf)

lw&lt;- formula(paste("~ ", paste("m",1:3,collapse=" + ",sep="")))
od&lt;- mDrop1(mlm, scope=lw, test="F", k=5, data=mdf)

## End(Not run)
</code></pre>

<hr>
<h2 id='misFct'>
Miscellaneous functions
</h2><span id='topic+gv2mpos'></span><span id='topic+xid1ch'></span><span id='topic+xeff'></span><span id='topic+qeff'></span>

<h3>Description</h3>

<p>Functions that may be useful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'># create 'mpos' and 'dists'
gv2mpos(gmap,v)

# extract 'xid' on chromosome 'k'
xid1ch(mpos,v,k)

# extract covariate effects of a mtcmim object 'object'
xeff(a,xid)

# extract QTL effect from a mtcmim object 'object'
qeff(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="misFct_+3A_gmap">gmap</code></td>
<td>

<p>a genetic map, which is a data frame (chr=chromosome id, dist=genetic distance (cM) on the chromosome,...). The chromosome id should be an integer.
</p>
</td></tr>
<tr><td><code id="misFct_+3A_v">v</code></td>
<td>

<p>a list; v[[j]] indicates which x's in the model for y[,j].
</p>
</td></tr>
<tr><td><code id="misFct_+3A_mpos">mpos</code></td>
<td>

<p>a data frame (id=marker index, ch=chromosome id, m=marker index on the chromosome, dist=genetic position in cM on the chromosome). Chromosome id should be an integer.
</p>
</td></tr>
<tr><td><code id="misFct_+3A_k">k</code></td>
<td>

<p>which chromosome?
</p>
</td></tr>
<tr><td><code id="misFct_+3A_a">a</code></td>
<td>

<p>covariate effects of an <code><a href="#topic+mtcmim">mtcmim</a></code> object.
</p>
</td></tr>
<tr><td><code id="misFct_+3A_xid">xid</code></td>
<td>

<p>a list of length p, xid[[j]] specifies columns of x as covariates for y[,j] .
</p>
</td></tr>
<tr><td><code id="misFct_+3A_object">object</code></td>
<td>

<p>an <code><a href="#topic+mtcmim">mtcmim</a></code> object.
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+mtcmim">mtcmim</a></code> and <code><a href="#topic+sureEst">sureEst</a></code>.
</p>

<hr>
<h2 id='mlogLik'>
Extract log-likelihood
</h2><span id='topic+mlogLik'></span>

<h3>Description</h3>

<p>Extract log-likelihood from the object of a multivariate multiple regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlogLik(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mlogLik_+3A_object">object</code></td>
<td>

<p>object of a fitted multivariate multiple regression model.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log-likelihood.
</p>

<hr>
<h2 id='mStep'>
Model selection in multivariate multiple regression
</h2><span id='topic+mStep'></span>

<h3>Description</h3>

<p>Select a multivariate multiple regression model via model selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mStep(object, scope, direction=c("both","backward","forward"),
   trace=FALSE, keep=TRUE, steps=1000, k=2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mStep_+3A_object">object</code></td>
<td>

<p>initial model in model search.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_scope">scope</code></td>
<td>

<p>a single formula, which provides &lsquo;upper&rsquo;, or a list containing components &lsquo;upper&rsquo; and &lsquo;lower&rsquo;, both formulae; defines the lower and upper bound. See <code><a href="stats.html#topic+step">step</a></code>.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_direction">direction</code></td>
<td>

<p>forward selection, backward elimination or stepwise.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_trace">trace</code></td>
<td>

<p>whether to track the process for monitoring purpose.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_keep">keep</code></td>
<td>

<p>whether to return the change of terms and related statistics.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_steps">steps</code></td>
<td>

<p>maximum number of search steps.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_k">k</code></td>
<td>

<p>penalty on a parameter. The selection criterion is the known &quot;AIC&quot; if <code>k = 2</code> and is &quot;BIC&quot; if <code>k = log(n)</code> where &quot;n&quot; is the sample size.
</p>
</td></tr>
<tr><td><code id="mStep_+3A_...">...</code></td>
<td>

<p>additional arguments to <code><a href="stats.html#topic+update">update</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components of a <code><a href="stats.html#topic+lm">lm</a></code> object plus &lsquo;keep&rsquo; if required.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mAdd1">mAdd1</a></code> and <code><a href="#topic+mDrop1">mDrop1</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
mdf&lt;- data.frame(traits,markers)
## Not run: 
mlm&lt;- lm(cbind(T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,T14,T15,T16) ~
   m1 + m2 + m3 + m4 + m5, data=mdf)

lw&lt;- formula(paste("~ ", paste("m",1:3,collapse=" + ",sep="")))
up&lt;- formula(paste("~", paste("m",1:15,collapse=" + ",sep="")))

ob&lt;- mStep(mlm, scope=list(lower=lw), k=99, direction="backward", data=mdf)
of&lt;- mStep(mlm, scope=list(upper=up), k=5, direction="forward", data=mdf)
o1&lt;- mStep(mlm, scope=list(upper=up), k=5, direction="both", data=mdf)
o2&lt;- mStep(o1, scope=list(upper=up), k=2, direction="forward", data=mdf)

## End(Not run)
</code></pre>

<hr>
<h2 id='mtcmim'>
MTCMIM
</h2><span id='topic+mtcmim'></span>

<h3>Description</h3>

<p>Multiple-trait composite multiple-interval mapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mtcmim(y, mpos, mdat, x, xid, dists, a, b, sigma, qtl=NULL,
   eps=NULL, win=Inf, range=0, pp=1, len=2, init=1,
   iter=10000,tol=1e-12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mtcmim_+3A_y">y</code></td>
<td>

<p>a n by p matrix, whose columns are dependent variables.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_mpos">mpos</code></td>
<td>

<p>a data frame (id=marker index, ch=chromosome id, m=marker index on the chromosome, dist=genetic position in cM on the chromosome). Chromosome id should be an integer.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_mdat">mdat</code></td>
<td>

<p>a matrix of n rows; marker genotypes (1 or 0). Columns should correspond to markers in the order.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_x">x</code></td>
<td>

<p>covariates; n by m numerical matrix.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_xid">xid</code></td>
<td>

<p>a list of length p, xid[[j]] specifies columns of x as covariates for y[,j] .
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_dists">dists</code></td>
<td>

<p>a data frame (ch=chromosome id, mid=marker index, d=genetic position in cM on the chromosome); specifies initial QTL locations.  Chromosome id should be an integer.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_a">a</code></td>
<td>

<p>initial covariate effects including intercepts (if given).
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_b">b</code></td>
<td>

<p>initial QTL effects (if given).
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_sigma">sigma</code></td>
<td>

<p>initial residual variance-covariance (if given).
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_qtl">qtl</code></td>
<td>

<p>a list of length p (if given); qtl[[j]] specifies qtls for y[,j], which are defined by rows of <code>dists</code>.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_eps">eps</code></td>
<td>

<p>a data frame (y=which trait,q1=QTL one,q2=QTL two) (if not <code>NULL</code>); specifies epistatic terms.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_win">win</code></td>
<td>

<p>window width of search around existing QTL. Ignored if <code>range=0</code>.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_range">range</code></td>
<td>

<p>search range: genome-wide (0), the same chromosomes (-1).
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_pp">pp</code></td>
<td>

<p>mapping population: BC-1, RIL-selfing-2, RIL-brother-sister-mating-3.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_len">len</code></td>
<td>

<p>step length in search.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_init">init</code></td>
<td>

<p>whether <code>a</code>, <code>b</code> and <code>sigma</code> are used as initial values.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_iter">iter</code></td>
<td>

<p>maximum number of iterations in a numerical process to estimate model parameters.
</p>
</td></tr>
<tr><td><code id="mtcmim_+3A_tol">tol</code></td>
<td>

<p>convergence tolerance.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given the covariates, the number of QTL and epistasis that are specified for each trait, this function searches for the optimal genomic locations of the QTL, and estimates the model parameters.
</p>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>loglik</code></td>
<td>
<p>log-likelihood of the final model</p>
</td></tr>
<tr><td><code>a</code></td>
<td>
<p>covariate effects</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>QTL effects</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>residual variance-covariance</p>
</td></tr>
<tr><td><code>qtl</code></td>
<td>
<p>QTL for each trait</p>
</td></tr>
<tr><td><code>eps</code></td>
<td>
<p>epistatic terms</p>
</td></tr>
<tr><td><code>dists</code></td>
<td>
<p>QTL locations</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
qtl&lt;- vector("list",16); qtl[[1]]&lt;- c(1,2)
eps&lt;- data.frame(y=1,q1=1,q2=2)
dists&lt;- dists[c(4,11),]
x&lt;- mdat - 3/2
## Not run: 
o&lt;- mtcmim(traits, mpos, mdat, x, xid, dists, qtl=qtl, eps=eps,
   win=5, range=-1, pp=2, len=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='mtcmimStep'>
MTCMIM model selection
</h2><span id='topic+mtcmimAdd1'></span><span id='topic+mtcmimDrop1'></span><span id='topic+mtcmimStep'></span><span id='topic+mtcmimStep2'></span>

<h3>Description</h3>

<p>Model selection for multiple-trait composite multiple-interval mapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mtcmimAdd1(object, y, x, xid, mpos, mdat, pp=1, len=1, type=1,
   iter=10000, tol=1e-12, ext=FALSE)

mtcmimDrop1(object, y, x, xid, mpos, mdat, pp=1, len=1, type=1,
   iter=10000, tol=1e-12, ext=FALSE)

mtcmimStep(object, y, x, xid, mpos, mdat, cv=0,
   direction=c("both","backward","forward"), pp=1, len=1,
   type=1, iter=10000, tol=1e-12, ext=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mtcmimStep_+3A_object">object</code></td>
<td>

<p>an object of class <code><a href="#topic+mtcmim">mtcmim</a></code>.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_y">y</code></td>
<td>

<p>a n by p matrix, whose columns are dependent variables.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_x">x</code></td>
<td>

<p>covariates; n by m numerical matrix.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_xid">xid</code></td>
<td>

<p>a list of length p; xid[[j]] specifies columns of x as covariates for y[,j] .
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_mpos">mpos</code></td>
<td>

<p>a data frame (id=marker index, ch=chromosome id, m=marker index on the chromosome, dist=genetic position in cM on the chromosome).  Chromosome id should be an integer.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_mdat">mdat</code></td>
<td>

<p>a matrix of n rows; marker genotypes (1 or 0). columns should correspond to markers in the order.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_pp">pp</code></td>
<td>

<p>mapping population: BC-1, RIL-selfing-2, RIL-brother-sister-mating-3.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_len">len</code></td>
<td>

<p>step length in search.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_type">type</code></td>
<td>

<p>1 if traits can have the different sets of covariates and QTL, 2 if all have the same set of covariates and QTL.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_ext">ext</code></td>
<td>

<p>whether to perform an extensive search for an &quot;optimal&quot; model with the same number of QTL per phenotype.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_cv">cv</code></td>
<td>

<p>critical value used in the likelihood ratio test to determine adding/dropping a QTL.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_direction">direction</code></td>
<td>

<p>forward selection, backward elimination or both directions.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_iter">iter</code></td>
<td>

<p>maximum number of iterations in a numerical process to estimate model parameters.
</p>
</td></tr>
<tr><td><code id="mtcmimStep_+3A_tol">tol</code></td>
<td>

<p>convergence tolerance.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>loglik</code></td>
<td>
<p>log-likelihood of the final model</p>
</td></tr>
<tr><td><code>a</code></td>
<td>
<p>covariate effects</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>QTL effects</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>residual variance-covariance</p>
</td></tr>
<tr><td><code>qtl</code></td>
<td>
<p>QTL for each trait</p>
</td></tr>
<tr><td><code>dists</code></td>
<td>
<p>QTL locations</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Currently, not able to include epistatic effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mtcmim">mtcmim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
y&lt;- traits[,1:5]
qtl&lt;- vector("list",5); qtl[[1]]&lt;- c(1,2)
dists&lt;- dists[c(4,11),]
x&lt;- mdat - 3/2
## Not run: 
o&lt;- mtcmim(y, mpos, mdat, dists=dists, qtl=qtl, eps=NULL,
   win=5, range=-1, pp=2, len=1)
of&lt;- mtcmimAdd1(o, y=y, mpos=mpos, mdat=mdat, pp=2, len=3)
os&lt;- mtcmimStep(of, y=y, mpos=mpos, mdat=mdat, cv=25,
   direction="both", pp=2, len=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='sureEps'>
Epistasis in a SURE model
</h2><span id='topic+sureEps'></span>

<h3>Description</h3>

<p>Look for epistasis between existing QTL in a model via model selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sureEps(y, x, v, k, direction=c("both","backward","forward"),
   iter=10000, max.terms=200, steps=1000, tol=1e-12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sureEps_+3A_y">y</code></td>
<td>

<p>a n by p matrix, whose columns are dependent variables.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_x">x</code></td>
<td>

<p>a n by m matrix, whose columns are predictor variables
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_v">v</code></td>
<td>

<p>a list; v[[j]] indicates which x's in the model for y[,j], between which possible epistasis is looked for.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_k">k</code></td>
<td>

<p>penalty, 0 if missing or &lt;0.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_direction">direction</code></td>
<td>

<p>forward selection, backward elimination or stepwise.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_iter">iter</code></td>
<td>

<p>maximum number of iterations in a numerical process to estimate model parameters.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_max.terms">max.terms</code></td>
<td>

<p>maximum number of terms in the final model.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_steps">steps</code></td>
<td>

<p>maximum number of search steps.
</p>
</td></tr>
<tr><td><code id="sureEps_+3A_tol">tol</code></td>
<td>

<p>convergence tolerance.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>associated traits, epistatic markers and p-values
</p>


<h3>See Also</h3>

<p><code><a href="#topic+surStep">surStep</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
x&lt;- as.matrix(mdat-1/2)
y&lt;- as.matrix(traits)[,1:3]
v&lt;- list()
upper&lt;- list()
for(k in 1:ncol(y)){
   v[[k]]&lt;- numeric(0)
   upper[[k]]&lt;- 1:ncol(x)
}
## Not run: 
o&lt;- surStep(y, x, v=v, upper=upper, k=19, direction="both",
   iter=250, max.terms=250, steps=2000, tol=1e-12)
neps&lt;- 6 # suppose there are 6 possible epistatic effects
oeps&lt;- sureEps(y, x, o$v, k=qchisq(1-0.05/neps,1), direction="backward",
  iter=250, max.terms=200, steps=1000, tol=1e-12)

## End(Not run)
</code></pre>

<hr>
<h2 id='sureEst'>
SURE model parameter estimation
</h2><span id='topic+sureEst'></span>

<h3>Description</h3>

<p>Estimate parameters in a SURE model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sureEst(y, x, v, sigma, iter=10000, tol=1e-12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sureEst_+3A_y">y</code></td>
<td>

<p>a n by p matrix, whose columns are dependent variables.
</p>
</td></tr>
<tr><td><code id="sureEst_+3A_x">x</code></td>
<td>

<p>a n by m matrix, whose columns are predictor variables to select from.
</p>
</td></tr>
<tr><td><code id="sureEst_+3A_v">v</code></td>
<td>

<p>a list; v[[j]] indicates which x's in the model for y[,j].
</p>
</td></tr>
<tr><td><code id="sureEst_+3A_sigma">sigma</code></td>
<td>

<p>initial residual variance-covarance matrix (if given).
</p>
</td></tr>
<tr><td><code id="sureEst_+3A_iter">iter</code></td>
<td>

<p>maximum number of iterations in a numerical process to estimate model parameters.
</p>
</td></tr>
<tr><td><code id="sureEst_+3A_tol">tol</code></td>
<td>

<p>convergence tolerance.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>loglik</code></td>
<td>
<p>log-likelihood of the model</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>estimates of model coefficients</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>estimated residual variance-covariance</p>
</td></tr>
<tr><td><code>fitted.values</code></td>
<td>
<p>fitted mean values</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
x&lt;- as.matrix(mdat-1/2)
y&lt;- as.matrix(traits)[,1:3]
v&lt;- list(c(1,25,50),numeric(0),3)
## Not run: 
o&lt;- sureEst(y, x, v=v, iter=250, tol=1e-12)

## End(Not run)
</code></pre>

<hr>
<h2 id='sureStep'>
SURE model selection
</h2><span id='topic+sureAdd1'></span><span id='topic+sureDrop1'></span><span id='topic+sureStep'></span><span id='topic+surStep'></span>

<h3>Description</h3>

<p>Select a SURE model via model selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sureAdd1(object, y, x, range=NULL, iter=10000, tol=1e-12, ext=FALSE)

sureDrop1(object, y, x, range=NULL, iter=10000, tol=1e-12, ext=FALSE)

sureStep(object, y, x, cv, direction=c("both","backward","forward"),
   range=NULL, iter=10000, steps=1000, tol=1e-12, ext=FALSE)

surStep(y, x, v, lower, upper, k, direction=c("both","backward",
   "forward"), iter=10000, max.terms=200, steps=1000, tol=1e-12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sureStep_+3A_object">object</code></td>
<td>

<p>initial model in model search; can be an object of <code><a href="#topic+sureEst">sureEst</a></code>.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_y">y</code></td>
<td>

<p>a n by p matrix, whose columns are dependent variables.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_x">x</code></td>
<td>

<p>a n by m matrix, whose columns are predictor variables to select from.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_range">range</code></td>
<td>

<p>a list; range[[j]] indicates which x's (all if <code>NULL</code>) correspond to which y[,j].
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_cv">cv</code></td>
<td>

<p>critical value used in the likelihood ratio test to determine adding/dropping a term.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_v">v</code></td>
<td>

<p>a list; v[[j]] indicates which x's to start with for y[,j].
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_lower">lower</code></td>
<td>

<p>a list in the format of <code>v</code>; lower scope of the model.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_upper">upper</code></td>
<td>

<p>a list in the format of <code>v</code>; upper scope of the model.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_k">k</code></td>
<td>

<p>penalty, 0 if missing or &lt;0.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_direction">direction</code></td>
<td>

<p>forward selection, backward elimination or stepwise.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_iter">iter</code></td>
<td>

<p>maximum number of iterations in a numerical process to estimate model parameters.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_max.terms">max.terms</code></td>
<td>

<p>maximum number of terms in the final model.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_steps">steps</code></td>
<td>

<p>maximum number of search steps.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_tol">tol</code></td>
<td>

<p>convergence tolerance.
</p>
</td></tr>
<tr><td><code id="sureStep_+3A_ext">ext</code></td>
<td>

<p>whether to perform an extensive search for an &quot;optimal&quot; model with the same number of QTL per phenotype.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>loglik</code></td>
<td>
<p>log-likelihood of the model</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>estimates of model coefficients</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>estimates of residual variance-covariance</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>variables selected in the model</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
x&lt;- as.matrix(mdat-1/2)
y&lt;- as.matrix(traits)[,1:3]
v&lt;- list()
upper&lt;- list()
for(k in 1:ncol(y)){
   v[[k]]&lt;- numeric(0)
   upper[[k]]&lt;- 1:ncol(x)
}
## Not run: 
o1&lt;- surStep(y, x, v=v, upper=upper, k=19)
o2&lt;- sureStep(o1, y, x, cv=50, ext=FALSE)

# search for optimal model of o1
o3&lt;- sureStep(o1, y, x, cv=Inf, direction="forward", ext=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='varGroup'>
Group variables
</h2><span id='topic+varGroup'></span>

<h3>Description</h3>

<p>Group variables via variable selection such that the grouped variables are optimal for multivariate analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varGroup(x, z, zscope=NULL, k=qf(0.95,1,nrow(x)-2), kf=k/2,
   method=c("pool","best"), direction=c("both","forward","backward"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varGroup_+3A_x">x</code></td>
<td>

<p>a data matrix/frame. Columns are variables to select from.
</p>
</td></tr>
<tr><td><code id="varGroup_+3A_z">z</code></td>
<td>

<p>a data matrix/frame. Columns are variables with elements 1 or not 1 (any others).
</p>
</td></tr>
<tr><td><code id="varGroup_+3A_zscope">zscope</code></td>
<td>

<p>which variables in <code>z</code> to be used for grouping; all if &quot;NULL&quot;.
</p>
</td></tr>
<tr><td><code id="varGroup_+3A_k">k</code></td>
<td>

<p>entry/stay value in backward stepwise.
</p>
</td></tr>
<tr><td><code id="varGroup_+3A_kf">kf</code></td>
<td>

<p>entry/stay value in forward stepwise.
</p>
</td></tr>
<tr><td><code id="varGroup_+3A_method">method</code></td>
<td>

<p>grouping method at each step: pool all the groups selected from each <code>zcope</code> variable, or choose the largest group (see examples).
</p>
</td></tr>
<tr><td><code id="varGroup_+3A_direction">direction</code></td>
<td>

<p>forward selection, backward elimination or both stepwise.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>indicators of grouped variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
varGroup(traits, mdat, zscope=c(42,55), method="pool")
varGroup(traits, mdat, zscope=c(42,55), method="best")
</code></pre>

<hr>
<h2 id='varSelect'>
Variable selection
</h2><span id='topic+varSelect'></span>

<h3>Description</h3>

<p>Select a specific number of variables via variable selection that are optimal for multivariate analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varSelect(x, group, scope, nv, direction=c("backward","forward"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varSelect_+3A_x">x</code></td>
<td>

<p>a data matrix/frame. Columns are variables to select from.
</p>
</td></tr>
<tr><td><code id="varSelect_+3A_group">group</code></td>
<td>

<p>a grouping indicator of observations.
</p>
</td></tr>
<tr><td><code id="varSelect_+3A_scope">scope</code></td>
<td>

<p>which variables (i.e., columns of <code>x</code>) to select from.
</p>
</td></tr>
<tr><td><code id="varSelect_+3A_nv">nv</code></td>
<td>

<p>how many variables to be selected.
</p>
</td></tr>
<tr><td><code id="varSelect_+3A_direction">direction</code></td>
<td>

<p>forward selection or backward elimination.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>variables selected in the model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+varGroup">varGroup</a></code> and <code><a href="#topic+varStep">varStep</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
g55&lt;- varGroup(traits, mdat, zscope=55, method="best")
idx&lt;- sample(1:nrow(traits), replace=FALSE)
varSelect(traits[idx,], group=mdat[,55], scope=1:ncol(traits[idx,]),
   nv=length(g55[[1]]))
</code></pre>

<hr>
<h2 id='varStep'>
Variable selection
</h2><span id='topic+varAdd1'></span><span id='topic+varDrop1'></span><span id='topic+varStep'></span>

<h3>Description</h3>

<p>Add a variable, drop a variable, or select a subset of variables via variable selection that are optimal for multivariate analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varAdd1(x, group, vin=NULL, scope=1:ncol(x), k=0)

varDrop1(x, group, vin=1:ncol(x), k=0)

varStep(x, group, scope, k, kf=k/2, direction=c("both",
   "forward","backward"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varStep_+3A_x">x</code></td>
<td>

<p>a data matrix/frame. Columns are variables to select from.
</p>
</td></tr>
<tr><td><code id="varStep_+3A_group">group</code></td>
<td>

<p>a grouping indicator of observations.
</p>
</td></tr>
<tr><td><code id="varStep_+3A_vin">vin</code></td>
<td>

<p>which variables (i.e., columns of <code>x</code>) already in model. It defines the initial model.
</p>
</td></tr>
<tr><td><code id="varStep_+3A_scope">scope</code></td>
<td>

<p>which variables (i.e., columns of <code>x</code>) to select from.
</p>
</td></tr>
<tr><td><code id="varStep_+3A_k">k</code></td>
<td>

<p>entry/stay value in backward stepwise.
</p>
</td></tr>
<tr><td><code id="varStep_+3A_kf">kf</code></td>
<td>

<p>entry/stay value in forward stepwise.
</p>
</td></tr>
<tr><td><code id="varStep_+3A_direction">direction</code></td>
<td>

<p>forward selection, backward elimination or both stepwise.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>which variable to add (add1), which variable to drop (drop1), or a subset of variables in the final model (step).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+varSelect">varSelect</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
varAdd1(traits, group=mdat[,42], vin=10, scope=1:ncol(traits))
varStep(traits, group=mdat[,42], k=12, scope=1:ncol(traits),
   direction="back")
</code></pre>

<hr>
<h2 id='varT2'>
Hotelling's T^2 test statistic
</h2><span id='topic+varT2'></span>

<h3>Description</h3>

<p>Calculate the Hotelling's T^2 test statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varT2(x, group, equalVar=T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varT2_+3A_x">x</code></td>
<td>

<p>a data matrix/frame. Columns are variables to select from.
</p>
</td></tr>
<tr><td><code id="varT2_+3A_group">group</code></td>
<td>

<p>a grouping indicator of observations.
</p>
</td></tr>
<tr><td><code id="varT2_+3A_equalvar">equalVar</code></td>
<td>

<p>whether assume the same variance-covariance in two groups.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Hotelling's T^2 test statistic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(etrait)
varT2(traits, mdat[,42])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
