<!DOCTYPE html><html><head><title>Help for package QuantNorm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QuantNorm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#brain'><p>Brain RNA-Seq data for both human and mouse.</p></a></li>
<li><a href='#connection.matrix'><p>Construct connection matrix for network analysis</p></a></li>
<li><a href='#ENCODE'><p>Normalized ENCODE raw counts data for both human and mouse.</p></a></li>
<li><a href='#QuantNorm'><p>Adjust the distance matrix by quantile normalization for data with batch effect</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Mitigating the Adverse Impact of Batch Effects in Sample Pattern
Detection</td>
</tr>
<tr>
<td>Description:</td>
<td>Modifies the distance matrix obtained from data with batch effects, so as to improve the performance of sample pattern detection, such as clustering, dimension reduction, and construction of networks between subjects. The method has been published in Bioinformatics (Fei et al, 2018, &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty117">doi:10.1093/bioinformatics/bty117</a>&gt;). Also available on 'GitHub' <a href="https://github.com/tengfei-emory/QuantNorm">https://github.com/tengfei-emory/QuantNorm</a>.</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>GGally, ggplot2, network, pheatmap, rgl, sna</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-02-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Teng Fei, Tianwei Yu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Teng Fei &lt;tfei@emory.edu&gt;</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/tengfei-emory/QuantNorm/issues">https://github.com/tengfei-emory/QuantNorm/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-01 20:07:57 UTC; TFEI</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-01 20:23:25 UTC</td>
</tr>
</table>
<hr>
<h2 id='brain'>Brain RNA-Seq data for both human and mouse.</h2><span id='topic+brain'></span>

<h3>Description</h3>

<p>Brain RNA-Seq data for both human and mouse.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(brain)
</code></pre>


<h3>Format</h3>

<p>Large matrix with 15041 rows and 62 columns</p>


<h3>Source</h3>

<p>Zhang, Ye, et al. &quot;Purification and characterization of progenitor and mature human astrocytes reveals transcriptional and functional differences with mouse.&quot; Neuron 89.1 (2016): 37-53.
</p>

<hr>
<h2 id='connection.matrix'>Construct connection matrix for network analysis</h2><span id='topic+connection.matrix'></span>

<h3>Description</h3>

<p>For data with known labels, this function constructs a connection matrix between unique labels, such as unique cell types. The returned matrix can be used for subject-wise network construction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>connection.matrix(mat, label, threshold = 0.15, closest = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="connection.matrix_+3A_mat">mat</code></td>
<td>
<p>n*n dissimilarity (1-correlation) matrix (e.g. obtained by QuantNorm).</p>
</td></tr>
<tr><td><code id="connection.matrix_+3A_label">label</code></td>
<td>
<p>n-dimension vector for the labels of the n subjects. Replicates share the same label.</p>
</td></tr>
<tr><td><code id="connection.matrix_+3A_threshold">threshold</code></td>
<td>
<p>A number between 0 to 1. Two groups will be regarded as connected if average 1-correlation &lt; threshold.</p>
</td></tr>
<tr><td><code id="connection.matrix_+3A_closest">closest</code></td>
<td>
<p>True or False. Whether connect the closest group or not if the closest group cannot satisfy the threshold condition.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the connection matrix between unique labels.
</p>


<h3>Author(s)</h3>

<p>Teng Fei. Email: tfei@emory.edu
</p>


<h3>References</h3>

<p>Fei et al (2018), Mitigating the adverse impact of batch effects in sample pattern detection, Bioinformatics, https://doi.org/10.1093/bioinformatics/bty117.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(network); library(ggplot2); library(sna); library(GGally) #drawing network graph

data("ENCODE")

#Assigning the batches based on species
batches &lt;- c(rep(1,13),rep(2,13))

#QuantNorm correction
corrected.distance.matrix &lt;- QuantNorm(ENCODE,batches,method='row/column', cor_method='pearson',
                                       logdat=FALSE,standardize = TRUE, tol=1e-4)

#Constructing connection matrix

mat &lt;- connection.matrix(mat=corrected.distance.matrix,label=colnames(corrected.distance.matrix))

#Creating network object and plot
ENCODE.net=network(mat, directed=FALSE)
ENCODE.net %v% "Species" &lt;- c(rep('Human',13),rep('Mouse',13))
p0 &lt;- ggnet2(ENCODE.net,label=TRUE,color = 'Species', palette = "Set2",
             size = 3, vjust = -0.6,mode = "kamadakawai",label.size = 3,
             color.legend = 'Species')+theme(legend.position = 'bottom')
plot(p0)
</code></pre>

<hr>
<h2 id='ENCODE'>Normalized ENCODE raw counts data for both human and mouse.</h2><span id='topic+ENCODE'></span>

<h3>Description</h3>

<p>Normalized ENCODE raw counts data for both human and mouse.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ENCODE)
</code></pre>


<h3>Format</h3>

<p>Large matrix with 10309 rows and 26 columns</p>


<h3>Source</h3>

<p>Reproduced according to Gilad, Yoav, and Orna Mizrahi-Man. &quot;A reanalysis of mouse ENCODE comparative gene expression data.&quot; F1000Research 4 (2015).
</p>

<hr>
<h2 id='QuantNorm'>Adjust the distance matrix by quantile normalization for data with batch effect</h2><span id='topic+QuantNorm'></span>

<h3>Description</h3>

<p>This function applies quantile normalization on the distance matrix (dissimilarity matrix) and return the corrected distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QuantNorm(dat, batch, method = "row/column", cor_method = "spearman",
  tol = 0.01, max = 50, logdat = TRUE, standardize = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QuantNorm_+3A_dat">dat</code></td>
<td>
<p>The original p*n batch effect data with n subjects and p RNA-seq measurements or the n by n distance matrix.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_batch">batch</code></td>
<td>
<p>The vector of length n indicating which batch the subjects belong to.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_method">method</code></td>
<td>
<p>Method for the quantile normalization. There are two options: &quot;row/column&quot; and &quot;vectorize&quot;.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_cor_method">cor_method</code></td>
<td>
<p>Method to calculate the correlation matrix, can be 'spearman'(default), 'pearson' or 'kendall'.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_tol">tol</code></td>
<td>
<p>The tolerance for the iterative method &quot;row/column&quot;, which is the Euclidean distance of the vectorized two dissimilarity matrices before and after each iteration.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_max">max</code></td>
<td>
<p>Maximum number of the iteration if the tolerance is not reached.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_logdat">logdat</code></td>
<td>
<p>Whether conducting log transformation to data or not.</p>
</td></tr>
<tr><td><code id="QuantNorm_+3A_standardize">standardize</code></td>
<td>
<p>Whether conducting standardization [(dat - mean)/sqrt(var)] to data or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the corrected 1-correlation matrix between subjects.
</p>


<h3>Author(s)</h3>

<p>Teng Fei. Email: tfei@emory.edu
</p>


<h3>References</h3>

<p>Fei et al (2018), Mitigating the adverse impact of batch effects in sample pattern detection, Bioinformatics, &lt;https://doi.org/10.1093/bioinformatics/bty117&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(pheatmap) #drawing heatmap

data("ENCODE") #load the ENCODE data

#Before correction, the subjects are clustered by species
pheatmap(cor(ENCODE))

#Assigning the batches based on species
batches &lt;- c(rep(1,13),rep(2,13))

#QuantNorm correction
corrected.distance.matrix &lt;- QuantNorm(ENCODE,batches,method='row/column', cor_method='pearson',
                                       logdat=FALSE, standardize = TRUE, tol=1e-4)
pheatmap(1-corrected.distance.matrix)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
