<!DOCTYPE html><html lang="en-US"><head><title>Help for package QuESTr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {QuESTr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Pinus'><p>Transcriptomes of Pinus roots under a Temperature Gradient</p></a></li>
<li><a href='#q.clean'><p>Clean data by eliminating genes with many missing values</p></a></li>
<li><a href='#q.opt'><p>Estimate the optimal number of genes to construct QuEST model</p></a></li>
<li><a href='#q.pca'><p>Visualize gene expression similarity using principal coordinate analysis</p></a></li>
<li><a href='#q.rank'><p>Visualize R-squared value distribution in gene-environment interaction</p></a></li>
<li><a href='#q.sort'><p>Sort and truncate genes according to the strength of gene-environment interaction</p></a></li>
<li><a href='#quest'><p>Construct and apply the QuEST model with your own data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Constructing Quantitative Environment Sensor using
Transcriptomes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Takahiko Koizumi, Kenta Suzuki, Yasunori Ichihashi</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Takahiko Koizumi &lt;takahiko.koizumi@riken.jp&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A method for prediction of environmental conditions based on 
    transcriptome data linked with the environmental gradients. This package 
    provides functions to overview gene-environment relationships, to construct 
    the prediction model, and to predict environmental conditions where the 
    transcriptomes were generated. This package can quest for candidate genes 
    for the model construction even in non-model organisms' transcriptomes 
    without any genetic information. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/takakoizumi/QuESTr">https://github.com/takakoizumi/QuESTr</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, kernlab</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-15 12:24:38 UTC; taka</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-15 20:00:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='Pinus'>Transcriptomes of Pinus roots under a Temperature Gradient</h2><span id='topic+Pinus'></span>

<h3>Description</h3>

<p>This dataset gives the TPM values of 200 selected genes obtained from 60 Pinus root
samples (30 samples each for training and test data) under a temperature gradient,
generated by RNA-seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pinus</code></pre>


<h3>Details</h3>

<p>A gene expression data matrix of 30 root samples of P. thunbergii under five
temperature conditions (8, 13, 18, 23, 28 Â°C) with six biological replicates is in
the first element of the list.
</p>
<p>A gene expression data matrix of another 30 root samples of P. thunbergii under
the same condition is in the second one.
</p>
<p>Temperature conditions where 30 root samples in each data matrix were generated are
in the third one.
</p>
<p>Gene expressions are normalized in the TPM value.
</p>


<h3>Source</h3>

<p>original (not published)</p>


<h3>References</h3>

<p>original (not published)</p>

<hr>
<h2 id='q.clean'>Clean data by eliminating genes with many missing values</h2><span id='topic+q.clean'></span>

<h3>Description</h3>

<p>Clean data by eliminating genes with many missing values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>q.clean(x, missing = 0.1, lowest = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="q.clean_+3A_x">x</code></td>
<td>
<p>A data matrix (raw: samples, col: genes).</p>
</td></tr>
<tr><td><code id="q.clean_+3A_missing">missing</code></td>
<td>
<p>A ratio of missing values in each column allowed to be remained in the data.</p>
</td></tr>
<tr><td><code id="q.clean_+3A_lowest">lowest</code></td>
<td>
<p>The lowest value recognized in the data (e.g., TPM, FPKM, or raw read counts).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data matrix (raw: samples, col: qualified genes)
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train.raw &lt;- Pinus$train
ncol(train.raw)

train &lt;- q.clean(train.raw)
ncol(train)
</code></pre>

<hr>
<h2 id='q.opt'>Estimate the optimal number of genes to construct QuEST model</h2><span id='topic+q.opt'></span>

<h3>Description</h3>

<p>Estimate the optimal number of genes to construct QuEST model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>q.opt(x, y, range = 5:50, method = "linear", rep = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="q.opt_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: genes).</p>
</td></tr>
<tr><td><code id="q.opt_+3A_y">y</code></td>
<td>
<p>A vector of an environment in which the samples were collected.</p>
</td></tr>
<tr><td><code id="q.opt_+3A_range">range</code></td>
<td>
<p>A sequence of numbers of genes to be tested for MAE calculation (default: 5:50).</p>
</td></tr>
<tr><td><code id="q.opt_+3A_method">method</code></td>
<td>
<p>A string to specify the method of regression for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; regression model can be specified.</p>
</td></tr>
<tr><td><code id="q.opt_+3A_rep">rep</code></td>
<td>
<p>The number of replications for each case set by range (default: 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sample-MAE curve
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- q.clean(Pinus$train)
target &lt;- Pinus$target
q.opt(train[1:10, ], target[1:10], range = 5:15)
</code></pre>

<hr>
<h2 id='q.pca'>Visualize gene expression similarity using principal coordinate analysis</h2><span id='topic+q.pca'></span>

<h3>Description</h3>

<p>Visualize gene expression similarity using principal coordinate analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>q.pca(x, y, method = "linear", lower.thr = 0, n.gene = ncol(x), size = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="q.pca_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: genes).</p>
</td></tr>
<tr><td><code id="q.pca_+3A_y">y</code></td>
<td>
<p>A vector of an environment in which the samples were collected.</p>
</td></tr>
<tr><td><code id="q.pca_+3A_method">method</code></td>
<td>
<p>A string to specify the method of regression for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; regression model can be specified.</p>
</td></tr>
<tr><td><code id="q.pca_+3A_lower.thr">lower.thr</code></td>
<td>
<p>The lower threshold of R-squared value to be indicated in a PCA plot (default: 0).</p>
</td></tr>
<tr><td><code id="q.pca_+3A_n.gene">n.gene</code></td>
<td>
<p>The number of candidate genes for QuEST model to be indicated in a PCA plot (default: ncol(x)).</p>
</td></tr>
<tr><td><code id="q.pca_+3A_size">size</code></td>
<td>
<p>The size of symbols in a PCA plot (default: 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A PCA plot
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- q.clean(Pinus$train)
target &lt;- Pinus$target
q.pca(train, target)
</code></pre>

<hr>
<h2 id='q.rank'>Visualize R-squared value distribution in gene-environment interaction</h2><span id='topic+q.rank'></span>

<h3>Description</h3>

<p>Visualize R-squared value distribution in gene-environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>q.rank(
  x,
  y,
  method = "linear",
  lower.thr = 0,
  n.gene = ncol(x),
  upper.xlim = ncol(x)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="q.rank_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: genes).</p>
</td></tr>
<tr><td><code id="q.rank_+3A_y">y</code></td>
<td>
<p>A vector of an environment in which the samples were collected.</p>
</td></tr>
<tr><td><code id="q.rank_+3A_method">method</code></td>
<td>
<p>A string to specify the method of regression for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; regression model can be specified.</p>
</td></tr>
<tr><td><code id="q.rank_+3A_lower.thr">lower.thr</code></td>
<td>
<p>The lower threshold of R-squared value to be included in QuEST model (default: 0).</p>
</td></tr>
<tr><td><code id="q.rank_+3A_n.gene">n.gene</code></td>
<td>
<p>The number of genes to be included in QuEST model (default: ncol(x)).</p>
</td></tr>
<tr><td><code id="q.rank_+3A_upper.xlim">upper.xlim</code></td>
<td>
<p>The upper limitation of x axis (i.e., the number of genes) in the resulted figure (default: ncol(x)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A rank order plot
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- q.clean(Pinus$train)
target &lt;- Pinus$target
train &lt;- q.sort(train, target)
q.rank(train, target)
</code></pre>

<hr>
<h2 id='q.sort'>Sort and truncate genes according to the strength of gene-environment interaction</h2><span id='topic+q.sort'></span>

<h3>Description</h3>

<p>Sort and truncate genes according to the strength of gene-environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>q.sort(x, y, method = "linear", n.gene = ncol(x), trunc = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="q.sort_+3A_x">x</code></td>
<td>
<p>A data matrix (raw: samples, col: genes).</p>
</td></tr>
<tr><td><code id="q.sort_+3A_y">y</code></td>
<td>
<p>A vector of an environment in which the samples were collected.</p>
</td></tr>
<tr><td><code id="q.sort_+3A_method">method</code></td>
<td>
<p>A string to specify the method of regression for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; regression model can be specified.</p>
</td></tr>
<tr><td><code id="q.sort_+3A_n.gene">n.gene</code></td>
<td>
<p>The number of genes to be included in QuEST model (default: ncol(x)).</p>
</td></tr>
<tr><td><code id="q.sort_+3A_trunc">trunc</code></td>
<td>
<p>a threshold to be truncated (default: 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data matrix (raw: samples, col: sorted genes)
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- q.clean(Pinus$train)
target &lt;- Pinus$target
cor(target, train[, 1])

train &lt;- q.sort(train, target, trunc = 0.5)
cor(target, train[, 1])
</code></pre>

<hr>
<h2 id='quest'>Construct and apply the QuEST model with your own data</h2><span id='topic+quest'></span>

<h3>Description</h3>

<p>Construct and apply the QuEST model with your own data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quest(x, y, newx = x, method = "linear", lower.thr = 0, n.gene = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="quest_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: genes).</p>
</td></tr>
<tr><td><code id="quest_+3A_y">y</code></td>
<td>
<p>A vector of an environment in which the samples were collected.</p>
</td></tr>
<tr><td><code id="quest_+3A_newx">newx</code></td>
<td>
<p>A data matrix (row: samples, col: genes).</p>
</td></tr>
<tr><td><code id="quest_+3A_method">method</code></td>
<td>
<p>A string to specify the method of regression for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; regression model can be specified.</p>
</td></tr>
<tr><td><code id="quest_+3A_lower.thr">lower.thr</code></td>
<td>
<p>The lower threshold of R-squared value to be used in QuEST model (default: 0).</p>
</td></tr>
<tr><td><code id="quest_+3A_n.gene">n.gene</code></td>
<td>
<p>The number of candidate genes to be used in QuEST model (default: 30).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the environment in which the samples of newx were collected
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- q.clean(Pinus$train)
test &lt;- Pinus$test
test &lt;- test[, colnames(train)]
target &lt;- Pinus$target
cor(target, quest(train, target, newx = test, method = "cubic"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
