<!DOCTYPE html><html lang="en"><head><title>Help for package quiddich</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {quiddich}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#quiddich-package'><p>QUick IDentification of DIagnostic CHaracters</p></a></li>
<li><a href='#changesAA'><p>Identification of diagnostic molecular characters in a nucleotide alignment</p>
that also cause diagnostic characters in the corresponding amino acid
alignment</a></li>
<li><a href='#diagCharAA'><p>Identification of diagnostic molecular characters in amino acid alignments</p>
for the delineation of taxa</a></li>
<li><a href='#diagCharNA'><p>Identification of diagnostic molecular characters in nucleotide alignments</p>
for the delineation of taxa</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>QUick IDentification of DIagnostic CHaracters</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-03-21</td>
</tr>
<tr>
<td>Author:</td>
<td>A. Luise Kuehn</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>A. Luise Kuehn &lt;luise.kuehn@uni-greifswald.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides tools for an automated identification of diagnostic molecular characters, i.e. such columns in a given nucleotide or amino acid alignment that allow to distinguish taxa from each other. These characters can then be used to complement the formal descriptions of the taxa, which are often based on morphological and anatomical features. Especially for morphologically cryptic species, this will be helpful. QUIDDICH distinguishes between four different types of diagnostic characters. For more information, see "Kuehn, A.L., Haase, M. 2019. QUIDDICH: QUick IDentification of DIagnostic CHaracters."</td>
</tr>
<tr>
<td>Depends:</td>
<td>ape(&ge; 5.2), R(&ge; 0.12.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spider, adegenet</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-03-21 10:19:35 UTC; Luise</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-03-25 09:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='quiddich-package'>QUick IDentification of DIagnostic CHaracters</h2><span id='topic+quiddich-package'></span>

<h3>Description</h3>

<p>The package <code>QUIDDICH</code> provides tools for an automated identification of diagnostic molecular characters, i.e. such columns in a given nucleotide or amino acid alignment that allow to distinguish taxa from each other. These characters can be used to complement formal descriptions of taxa, which are often based on morphological and anatomical features. Especially for morphologically cryptic species, this is helpful. <code>QUIDDICH</code> distinguishes between four different types of diagnostic characters. For more information, see &quot;Kuehn, A.L., Haase, M. 2019. QUIDDICH: QUick IDentification of DIagnostic CHaracters.&quot;
</p>


<h3>Author(s)</h3>

<p>A. Luise Kuehn &lt;luise.kuehn@uni-greifswald.de&gt;
</p>

<hr>
<h2 id='changesAA'>Identification of diagnostic molecular characters in a nucleotide alignment
that also cause diagnostic characters in the corresponding amino acid
alignment</h2><span id='topic+changesAA'></span>

<h3>Description</h3>

<p>This function is a tool for an automated identification of such diagnostic
molecular characters in a nucleotide alignment that also cause diagnostic
characters in the corresponding amino acid alignment. For each taxon given
in <code>taxOfInt</code>, it identifies the mentioned characters and returns their
positions and types in the nucleotide alignment and their positions and
types in the corresponding amino acid alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>changesAA(DNAbin, taxVector = dimnames(DNAbin)[[1]], codonstart = 1,
  taxOfInt = "all", types = c("type1", "type2", "type3"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="changesAA_+3A_dnabin">DNAbin</code></td>
<td>
<p>An object (the nucleotide alignment) of class 'DNAbin'.</p>
</td></tr>
<tr><td><code id="changesAA_+3A_taxvector">taxVector</code></td>
<td>
<p>The taxon vector. Default assumes that each row in the
alignment belongs to a different taxon.</p>
</td></tr>
<tr><td><code id="changesAA_+3A_codonstart">codonstart</code></td>
<td>
<p>An integer defining where to start the translation of the
nucleotide alignment into the corresp. amino acid alignment. Default is 1.</p>
</td></tr>
<tr><td><code id="changesAA_+3A_taxofint">taxOfInt</code></td>
<td>
<p>A vector containing the taxa for which the above mentioned
characters shall be extracted. Default is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="changesAA_+3A_types">types</code></td>
<td>
<p>A vector containing the types of diagnostic characters
(in the nucleotide alignment) that shall be considered. The types
can be &quot;all&quot; or any combination of &quot;type1&quot;, &quot;type2&quot;, &quot;type3&quot; and &quot;type4&quot;.
Default is &quot;type1&quot;, &quot;type 2&quot; and &quot;type3&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>changesAA</code> returns a list, where each entry belongs to one
taxon of interest. Each taxon of interest has a matrix assigned to it, in
which the first and second column contain the positions and types of the
identified diagnostic characters in the nucleotide alignment and the third
and fourth column contain the positions and types of the diagnostic
characters in the corresponding amino acid alignment.
</p>
<p><code>type1</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of the remaining
taxa, and that it is fixed for one state in the taxon of interest.
<code>type2</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of the remaining
taxa, and that it is not fixed for one state in the taxon of interest.
<code>type3</code> means that the character is suitable to distinguish some (but
not all) individuals of the taxon of interest from all individuals of the
remaining taxa.
<code>type4</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of at least one
(but not all) other taxon while being fixed in both the taxon of interest
and the compared taxa.
</p>
<p><code>changesAA</code> returns for each taxon of interest the following elements:
</p>
<table role = "presentation">
<tr><td><code>posDiNu</code></td>
<td>
<p>The positions of the identified diagnostic characters in the
nucleotide alignment.</p>
</td></tr>
<tr><td><code>typeDiNu</code></td>
<td>
<p>The types of the identified diagnostic characters in the
nucleotide alignment.</p>
</td></tr>
<tr><td><code>posDiAA</code></td>
<td>
<p>The positions of the corresponding diagnostic characters in
the amino acid alignment.</p>
</td></tr>
<tr><td><code>typeDiAA</code></td>
<td>
<p>The types of the corresponding diagnostic characters in the
amino acid alignment.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>A. Luise Kuehn &lt;luise.kuehn@uni-greifswald.de&gt;
</p>


<h3>References</h3>

<p>Kuehn, A.L., Haase, M. (2019) QUIDDICH: QUick IDentification of
DIagnostic CHaracters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#using a dataset from spider
#install.packages("spider")
library(spider)
data("anoteropsis")
anoTax &lt;- sapply(strsplit(dimnames(anoteropsis)[[1]], split="_"),
 function(x) paste(x[1], x[2], sep="_"))
changesAA(anoteropsis, anoTax, codonstart=2, taxOfInt="all")
changesAA(anoteropsis, anoTax, codonstart=2, taxOfInt="Artoria_flavimanus",
 types="type4")

</code></pre>

<hr>
<h2 id='diagCharAA'>Identification of diagnostic molecular characters in amino acid alignments
for the delineation of taxa</h2><span id='topic+diagCharAA'></span>

<h3>Description</h3>

<p>This function is a tool for an automated identification of diagnostic
molecular characters that allow to distinguish taxa within an amino acid
alignment. For each taxon given in <code>taxOfInt</code>, it identifies the
diagnostic characters and returns their alignment positions, their types,
the states that are characteristic for the taxon of interest and
(in case of type 4 characters) the taxon that it was compared with.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagCharAA(AAbin, taxVector = dimnames(AAbin)[[1]], taxOfInt = "all",
  types = c("type1", "type2", "type3"), gapValid = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diagCharAA_+3A_aabin">AAbin</code></td>
<td>
<p>An object (the amino acid alignment) of class 'AAbin'.</p>
</td></tr>
<tr><td><code id="diagCharAA_+3A_taxvector">taxVector</code></td>
<td>
<p>The taxon vector. Default assumes that each row in the
alignment belongs to a different taxon.</p>
</td></tr>
<tr><td><code id="diagCharAA_+3A_taxofint">taxOfInt</code></td>
<td>
<p>A vector containing the taxa for which diagnostic
molecular characters shall be extracted. Default is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="diagCharAA_+3A_types">types</code></td>
<td>
<p>A vector containing the types of diagnostic molecular
characters that shall be extracted. The types can be &quot;all&quot; or any
combination of &quot;type1&quot;, &quot;type2&quot;, &quot;type3&quot; and &quot;type4&quot;. Default is &quot;type1&quot;,
&quot;type 2&quot; and &quot;type3&quot;.</p>
</td></tr>
<tr><td><code id="diagCharAA_+3A_gapvalid">gapValid</code></td>
<td>
<p>Boolean variable denoting if a gap can be a characteristic
state for taxon i (and taxon l in case of type 4). Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>diagCharAA</code> returns a list, where each entry belongs to one
taxon of interest. Each taxon of interest has a set of diagnostic
molecular characters (position, type, characteristic states for taxon of
interest, compared taxa) assigned to it.
<code>type1</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of the remaining
taxa, and that it is fixed for one state in the taxon of interest.
<code>type2</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of the remaining
taxa, and that it is not fixed for one state in the taxon of interest.
<code>type3</code> means that the character is suitable to distinguish some (but
not all) individuals of the taxon of interest from all individuals of the
remaining taxa.
<code>type4</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of at least one
(but not all) other taxon while being fixed in both the taxon of interest
and the compared taxa.
</p>
<p><code>diagCharAA</code> returns for each taxon of interest the following elements:
</p>
<table role = "presentation">
<tr><td><code>position</code></td>
<td>
<p>The positions of its diagnostic molecular characters.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>The types of the diagnostic molecular characters.</p>
</td></tr>
<tr><td><code>states</code></td>
<td>
<p>The states that are characteristic for the taxon i of interest,
i.e. states that are distinct from &quot;X&quot; and unique to the taxon of interest
(in case of type 1, 2 or 3), or fixed in the taxon of interest (type 4), resp.</p>
</td></tr>
<tr><td><code>compared taxa</code></td>
<td>
<p>Only relevant for type 4 characters. It contains the
name x if the character is found to be a type 4 character of the taxon of
interest when being compared to taxon x.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>A. Luise Kuehn &lt;luise.kuehn@uni-greifswald.de&gt;
</p>


<h3>References</h3>

<p>Kuehn, A.L., Haase, M. 2019. QUIDDICH: QUick IDentification of
DIagnostic CHaracters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#using a dataset from spider
#install.packages("spider")
#install.packages("ape")
library(spider)
library(ape)
data("anoteropsis")
anoTax &lt;- sapply(strsplit(dimnames(anoteropsis)[[1]], split="_"),
 function(x) paste(x[1], x[2], sep="_"))
anoteropsis_AA &lt;- trans(anoteropsis,code=1,codonstart=2)
diagCharAA(anoteropsis_AA, anoTax, taxOfInt="all")
diagCharAA(anoteropsis_AA, anoTax, taxOfInt="Artoria_flavimanus", types=c("type1","type2"))
</code></pre>

<hr>
<h2 id='diagCharNA'>Identification of diagnostic molecular characters in nucleotide alignments
for the delineation of taxa</h2><span id='topic+diagCharNA'></span>

<h3>Description</h3>

<p>This function is a tool for an automated identification of diagnostic
molecular characters that allow to distinguish taxa within a nucleotide
alignment. For each taxon given in <code>taxOfInt</code>, it identifies the
diagnostic characters and returns their alignment positions, their types,
the states that are characteristic for the taxon of interest and
(in case of type 4 characters) the taxon that it was compared with.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagCharNA(DNAbin, taxVector = dimnames(DNAbin)[[1]], taxOfInt = "all",
  types = c("type1", "type2", "type3"), gapValid = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diagCharNA_+3A_dnabin">DNAbin</code></td>
<td>
<p>An object (the nucleotide alignment) of class 'DNAbin'.</p>
</td></tr>
<tr><td><code id="diagCharNA_+3A_taxvector">taxVector</code></td>
<td>
<p>The taxon vector. Default assumes that each row in the
alignment belongs to a different taxon.</p>
</td></tr>
<tr><td><code id="diagCharNA_+3A_taxofint">taxOfInt</code></td>
<td>
<p>A vector containing the taxa for which diagnostic
molecular characters shall be extracted. Default is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="diagCharNA_+3A_types">types</code></td>
<td>
<p>A vector containing the types of diagnostic molecular
characters that shall be extracted. The types can be &quot;all&quot; or any
combination of &quot;type1&quot;, &quot;type2&quot;, &quot;type3&quot; and &quot;type4&quot;. Default is &quot;type1&quot;,
&quot;type 2&quot; and &quot;type3&quot;.</p>
</td></tr>
<tr><td><code id="diagCharNA_+3A_gapvalid">gapValid</code></td>
<td>
<p>Boolean variable denoting if a gap can be a characteristic
state for taxon i (and taxon l in case of type 4). Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>diagCharNA</code> returns a list, where each entry belongs to one
taxon of interest. Each taxon of interest has a set of diagnostic
molecular characters (position, type, characteristic states for taxon of
interest, compared taxa) assigned to it.
<code>type1</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of the remaining
taxa, and that it is fixed for one state in the taxon of interest.
<code>type2</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of the remaining
taxa, and that it is not fixed for one state in the taxon of interest.
<code>type3</code> means that the character is suitable to distinguish some (but
not all) individuals of the taxon of interest from all individuals of the
remaining taxa.
<code>type4</code> means that the character is suitable to distinguish each
individual of the taxon of interest from all individuals of at least one
(but not all) other taxon while being fixed in both the taxon of interest
and the compared taxa.
</p>
<p><code>diagCharNA</code> returns for each taxon of interest the following elements:
</p>
<table role = "presentation">
<tr><td><code>position</code></td>
<td>
<p>The positions of its diagnostic molecular characters.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>The types of the diagnostic molecular characters.</p>
</td></tr>
<tr><td><code>states</code></td>
<td>
<p>The states that are characteristic for the taxon i of interest,
i.e. states that are distinct from &quot;n&quot; and unique to the taxon of interest
(in case of type 1, 2 or 3), or fixed in the taxon of interest (type 4).</p>
</td></tr>
<tr><td><code>compared taxa</code></td>
<td>
<p>Only relevant for type 4 characters. It
contains the name x if the character is found to be a type 4 character of
the taxon of interest when being compared to taxon x.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>A. Luise Kuehn &lt;luise.kuehn@uni-greifswald.de&gt;
</p>


<h3>References</h3>

<p>Kuehn, A.L., Haase, M. 2019. QUIDDICH: QUick IDentification of
DIagnostic CHaracters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#using a dataset from spider
#install.packages("spider")
library(spider)
data("anoteropsis")
anoTax &lt;- sapply(strsplit(dimnames(anoteropsis)[[1]], split="_"),
	function(x) paste(x[1], x[2], sep="_"))
diagCharNA(anoteropsis, anoTax, taxOfInt="all")
diagCharNA(anoteropsis, anoTax, taxOfInt="all", types=c("type1","type2"))
#
#with loading of a fasta file
#install.packages("adegenet")
library(adegenet)
alignment &lt;- fasta2DNAbin(paste0(find.package("quiddich"), "/extData/example.fasta"))
taxonVector &lt;- as.vector(sapply(dimnames(alignment)[[1]], function(x) substr(x,1,4)))
diagCharNA(alignment, taxonVector)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
