<!DOCTYPE html><html lang="en"><head><title>Help for package quincunx</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {quincunx}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ancestry_categories'><p>Ancestry categories and classes</p></a></li>
<li><a href='#bind'><p>Bind PGS Catalog objects</p></a></li>
<li><a href='#clear_cache'><p>Clear quincunx cache of memoised functions</p></a></li>
<li><a href='#cohorts'><p>Constructor for the S4 cohorts object.</p></a></li>
<li><a href='#cohorts-class'><p>An S4 class to represent a set of cohorts</p></a></li>
<li><a href='#collect_pgs_ids'><p>Collect development and evaluation pgs_ids</p></a></li>
<li><a href='#collect_samples'><p>Collect samples_variants and samples_training</p></a></li>
<li><a href='#contains_question_mark'><p>Does a string contain a question mark?</p></a></li>
<li><a href='#count'><p>Extract the count field from a JSON response</p></a></li>
<li><a href='#filter_by_id'><p>Filter PGS Catalog objects by identifier.</p></a></li>
<li><a href='#first_non_na'><p>Returns the position of the first non-NA value</p></a></li>
<li><a href='#get_ancestry_categories'><p>Get ancestry categories and classes</p></a></li>
<li><a href='#get_ancestry_symbol_mappings'><p>Get PGS Catalog Ancestry Symbol Mappings</p></a></li>
<li><a href='#get_cohorts'><p>Get PGS Catalog Cohorts</p></a></li>
<li><a href='#get_performance_metrics'><p>Get PGS Catalog Performance Metrics</p></a></li>
<li><a href='#get_publications'><p>Get PGS Catalog Publications</p></a></li>
<li><a href='#get_releases'><p>Get PGS Catalog Releases</p></a></li>
<li><a href='#get_sample_sets'><p>Get PGS Catalog Sample Sets</p></a></li>
<li><a href='#get_scores'><p>Get PGS Catalog Scores</p></a></li>
<li><a href='#get_trait_categories'><p>Get PGS Catalog Trait Categories</p></a></li>
<li><a href='#get_traits'><p>Get PGS Catalog Traits</p></a></li>
<li><a href='#is_paginated'><p>Is the response paginated?</p></a></li>
<li><a href='#is_pgs_id'><p>Is a string a PGS Catalog identifier?</p></a></li>
<li><a href='#is_pubmed_id'><p>Is a string a PubMed ID?</p></a></li>
<li><a href='#list_to_s4'><p>Convert a named list to an S4 object</p></a></li>
<li><a href='#msg_400'><p>Pretty printing of 404 error</p></a></li>
<li><a href='#msg_404'><p>Pretty printing of 404 error</p></a></li>
<li><a href='#msg_empty'><p>Pretty printing of empty response</p></a></li>
<li><a href='#n'><p>Number of PGS Catalog entities</p></a></li>
<li><a href='#n_pages'><p>Number of pages</p></a></li>
<li><a href='#no_msg'><p>Pretty printing of no error message</p></a></li>
<li><a href='#nr_to_na'><p>Convert NR (Not Recorded) to NA (Not Available)</p></a></li>
<li><a href='#offsets'><p>Generate offset values</p></a></li>
<li><a href='#open_in_dbsnp'><p>Browse dbSNP from SNP identifiers.</p></a></li>
<li><a href='#open_in_pgs_catalog'><p>Browse PGS Catalog entities from the PGS Catalog Web Graphical User Interface</p></a></li>
<li><a href='#open_in_pubmed'><p>Browse PubMed from PubMed identifiers.</p></a></li>
<li><a href='#performance_metrics'><p>Constructor for the S4 performance_metrics object.</p></a></li>
<li><a href='#performance_metrics-class'><p>An S4 class to represent a set of PGS Catalog Performance Metrics</p></a></li>
<li><a href='#pgp_to_pgs'><p>Map PGP identifiers to PGS identifiers</p></a></li>
<li><a href='#pgp_to_ppm'><p>Map PGP identifiers to PPM identifiers</p></a></li>
<li><a href='#pgp_to_pss'><p>Map PGP identifiers to PSS identifiers</p></a></li>
<li><a href='#pgs_server'><p>PGS REST API server</p></a></li>
<li><a href='#pgs_to_pgp'><p>Map PGS identifiers to PGP identifiers</p></a></li>
<li><a href='#pgs_to_ppm'><p>Map PGS identifiers to PPM identifiers</p></a></li>
<li><a href='#pgs_to_pss'><p>Map PGS identifiers to PSS identifiers</p></a></li>
<li><a href='#pgs_to_study'><p>Map PGS identifiers to GWAS study identifiers</p></a></li>
<li><a href='#ppm_to_pgp'><p>Map PPM identifiers to PGP identifiers</p></a></li>
<li><a href='#ppm_to_pgs'><p>Map PPM identifiers to PGS identifiers</p></a></li>
<li><a href='#ppm_to_pss'><p>Map PPM identifiers to PSS identifiers</p></a></li>
<li><a href='#pss_to_pgp'><p>Map PSS identifiers to PGP identifiers</p></a></li>
<li><a href='#pss_to_pgs'><p>Map PSS identifiers to PGS identifiers</p></a></li>
<li><a href='#pss_to_ppm'><p>Map PSS identifiers to PPM identifiers</p></a></li>
<li><a href='#publications'><p>Constructor for the S4 publications object.</p></a></li>
<li><a href='#publications-class'><p>An S4 class to represent a set of PGS Catalog Publications</p></a></li>
<li><a href='#read_scoring_file'><p>Read a polygenic scoring file</p></a></li>
<li><a href='#releases'><p>Constructor for the S4 releases object.</p></a></li>
<li><a href='#releases-class'><p>An S4 class to represent a set of PGS Catalog Releases</p></a></li>
<li><a href='#remap_id'><p>Remap identifiers</p></a></li>
<li><a href='#request'><p>Request a resource from the PGS REST API</p></a></li>
<li><a href='#request_all'><p>Request a paginated resource from the PGS REST API</p></a></li>
<li><a href='#request_warning'><p>Pretty printing of http error</p></a></li>
<li><a href='#s4_to_list'><p>Convert an S4 object into a list</p></a></li>
<li><a href='#sample_sets'><p>Constructor for the S4 sample_sets object.</p></a></li>
<li><a href='#sample_sets-class'><p>An S4 class to represent a set of PGS Catalog Sample Sets</p></a></li>
<li><a href='#scores'><p>Constructor for the S4 scores object.</p></a></li>
<li><a href='#scores-class'><p>An S4 class to represent a set of PGS Catalog Polygenic Scores</p></a></li>
<li><a href='#setop'><p>Set operations on PGS Catalog objects</p></a></li>
<li><a href='#stages'><p>Study stages</p></a></li>
<li><a href='#study_to_pgs'><p>Map GWAS studies identifiers to PGS identifiers</p></a></li>
<li><a href='#subset-cohorts'><p>Subset a cohorts object</p></a></li>
<li><a href='#subset-performance_metrics'><p>Subset a performance_metrics object</p></a></li>
<li><a href='#subset-publications'><p>Subset a publications object</p></a></li>
<li><a href='#subset-releases'><p>Subset a releases object</p></a></li>
<li><a href='#subset-sample_sets'><p>Subset a sample_sets object</p></a></li>
<li><a href='#subset-scores'><p>Subset a scores object</p></a></li>
<li><a href='#subset-trait_categories'><p>Subset a trait_categories object</p></a></li>
<li><a href='#subset-traits'><p>Subset a traits object</p></a></li>
<li><a href='#sure'><p>Are you sure?</p></a></li>
<li><a href='#trait_categories'><p>Constructor for the S4 trait_categories object.</p></a></li>
<li><a href='#trait_categories-class'><p>An S4 class to represent a set of PGS Catalog Trait Categories</p></a></li>
<li><a href='#traits'><p>Constructor for the S4 traits object.</p></a></li>
<li><a href='#traits-class'><p>An S4 class to represent a set of PGS Catalog Traits</p></a></li>
<li><a href='#user_agent_id'><p>User agent identification</p></a></li>
<li><a href='#write_xlsx'><p>Export a PGS Catalog object to xlsx</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>REST API Client for the 'PGS' Catalog</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.9</td>
</tr>
<tr>
<td>Description:</td>
<td>Programmatic access to the 'PGS' Catalog.
    This package provides easy access to 'PGS' Catalog data by
    accessing the REST API <a href="https://www.pgscatalog.org/rest/">https://www.pgscatalog.org/rest/</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, ggplot2</td>
</tr>
<tr>
<td>Imports:</td>
<td>stringr, vroom, purrr, glue, dplyr, tidyjson, tibble,
lubridate, rlang, tidyr, httr, utils, rvest, progress, methods,
writexl, concatenate, memoise, readr</td>
</tr>
<tr>
<td>Collate:</td>
<td>'ancestry_categories.R' 'class-cohorts.R'
'class-performance_metrics.R' 'class-publications.R'
'class-releases.R' 'class-sample_sets.R' 'class-scores.R'
'class-trait_categories.R' 'class-traits.R' 'clear_cache.R'
'contains_question_mark.R' 'count.R' 'delay.R'
'drop_metadata_cols.R' 'first_non_na.R' 'generics.R' 'get.R'
'get_ancestry_categories.R' 'get_cohorts.R' 'get_column.R'
'get_performance_metrics.R' 'get_publications.R'
'get_releases.R' 'get_sample_sets.R' 'get_scores.R'
'get_trait_categories.R' 'get_traits.R' 'id_mapping.R'
'is_json_empty.R' 'is_paginated.R' 'is_pgs_id.R'
'is_pubmed_id.R' 'messages.R' 'n_pages.R' 'nr_to_na.R'
'offsets.R' 'open_in_dbsnp.R' 'open_in_pgs_catalog.R'
'open_in_pubmed.R' 'parse-ancestry_categories.R'
'parse-cohorts.R' 'parse-performance_metrics.R'
'parse-publications.R' 'parse-releases.R' 'parse-sample_sets.R'
'parse-scores.R' 'parse-trait_categories.R' 'parse-traits.R'
'parse_estimate.R' 'read_file_column_names.R'
'read_pgs_scoring_file.R' 'relocate_metadata_cols.R'
'remap_id.R' 'request.R' 's4-utils.R' 'stages.R' 'sure.R'
'unwrap_cohort.R' 'unwrap_demographics.R' 'unwrap_efotrait.R'
'unwrap_interval.R' 'unwrap_publication.R' 'unwrap_sample.R'
'warnings.R' 'write_xlsx.R'</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ramiromagno/quincunx">https://github.com/ramiromagno/quincunx</a>,
<a href="https://rmagno.eu/quincunx/">https://rmagno.eu/quincunx/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ramiromagno/quincunx/issues">https://github.com/ramiromagno/quincunx/issues</a></td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>patterninstitute/chic</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-20 15:38:56 UTC; rmagno</td>
</tr>
<tr>
<td>Author:</td>
<td>Ramiro Magno <a href="https://orcid.org/0000-0001-5226-3441"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Isabel Duarte <a href="https://orcid.org/0000-0003-0060-2936"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ana-Teresa Maia <a href="https://orcid.org/0000-0002-0454-9207"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  CINTESIS [cph, fnd],
  Pattern Institute [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ramiro Magno &lt;rmagno@pattern.institute&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-20 15:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ancestry_categories'>Ancestry categories and classes</h2><span id='topic+ancestry_categories'></span>

<h3>Description</h3>

<p>A dataset containing the ancestry categories defined in NHGRI-EBI GWAS
Catalog framework (Table 1, <a href="https://doi.org/10.1186/s13059-018-1396-2">doi:10.1186/s13059-018-1396-2</a>). Ancestry
categories are assigned to samples with distinct and well-defined patterns of
genetic variation. You will find these categories in the variable
<code>ancestry_category</code> of the following objects: <a href="#topic+scores-class">scores</a>,
<a href="#topic+performance_metrics-class">performance_metrics</a> and <a href="#topic+sample_sets-class">sample_sets</a>. Ancestry
categories (<code>ancestry_category</code>) are further clustered into ancestry
classes (<code>ancestry_class</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancestry_categories
</code></pre>


<h3>Format</h3>

<p>A data frame with 19 ancestry categories (rows) and 6 columns:
</p>

<dl>
<dt>ancestry_category</dt><dd><p>Ancestry category.</p>
</dd>
<dt>ancestry_class</dt><dd><p>To reduce the complexity associated with the many
ancestry categories, some have been merged into higher-level groupings
(<code>ancestry_class</code>). These groupings represent the current breadth of
data in the PGS Catalog and are likely to change as more data is added.</p>
</dd>
<dt>ancestry_class_symbol</dt><dd><p>3-letter code for the <code>ancestry_class</code>
e.g. <code>"EUR"</code> or <code>"MAE"</code>.</p>
</dd>
<dt>ancestry_class_colour</dt><dd><p>Hexadecimal colour code associated with
ancestry groupings (<code>ancestry_class</code>). This can be useful when
visually communicating about ancestries.</p>
</dd>
<dt>definition</dt><dd><p>Description of the ancestry category.</p>
</dd>
<dt>examples</dt><dd><p>Examples of detailed descriptions of sample ancestries
included in the category.</p>
</dd>
</dl>



<h3>Source</h3>


<dl>
<dt>Table 1 of Moralles et al. (2018):</dt><dd><p><a href="https://doi.org/10.1186/s13059-018-1396-2">doi:10.1186/s13059-018-1396-2</a></p>
</dd>
<dt>PGS Catalog Ancestry Documentation:</dt><dd><p><a href="http://www.pgscatalog.org/docs/ancestry/">http://www.pgscatalog.org/docs/ancestry/</a></p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>ancestry_categories

</code></pre>

<hr>
<h2 id='bind'>Bind PGS Catalog objects</h2><span id='topic+bind'></span>

<h3>Description</h3>

<p>Binds together PGS Catalog objects of the same class. Note that
<code>bind()</code> preserves duplicates whereas
<code><a href="#topic+setop">union</a></code> does not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bind_+3A_x">x</code></td>
<td>
<p>An object of either class <a href="#topic+scores-class">scores</a>, <a href="#topic+publications-class">publications</a>,
<a href="#topic+traits-class">traits</a>, <a href="#topic+performance_metrics-class">performance_metrics</a>, <a href="#topic+sample_sets-class">sample_sets</a>,
<a href="#topic+cohorts-class">cohorts</a> or <a href="#topic+trait_categories-class">trait_categories</a>.</p>
</td></tr>
<tr><td><code id="bind_+3A_...">...</code></td>
<td>
<p>Objects of the same class as <code>x</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get some `scores` objects:
my_scores_1 &lt;- get_scores(c('PGS000012', 'PGS000013'))
my_scores_2 &lt;- get_scores(c('PGS000013', 'PGS000014'))

# NB: with `bind()`, PGS000013 is repeated (as opposed to `union()`)
bind(my_scores_1, my_scores_2)@scores

</code></pre>

<hr>
<h2 id='clear_cache'>Clear quincunx cache of memoised functions</h2><span id='topic+clear_cache'></span>

<h3>Description</h3>

<p>quincunx uses memoised functions for the REST API calls. Use this function to
reset the cache.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clear_cache()
</code></pre>


<h3>Value</h3>

<p>Returns a logical value, indicating whether the resetting of the
cache was successful (<code>TRUE</code>) or not  <code>FALSE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>clear_cache()

</code></pre>

<hr>
<h2 id='cohorts'>Constructor for the S4 cohorts object.</h2><span id='topic+cohorts'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+cohorts-class">cohorts</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohorts(cohorts = s4cohorts_cohorts_tbl(), pgs_ids = s4cohorts_pgs_ids_tbl())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cohorts_+3A_cohorts">cohorts</code></td>
<td>
<p>A <code>s4cohorts_cohorts_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="cohorts_+3A_pgs_ids">pgs_ids</code></td>
<td>
<p>A <code>s4cohorts_pgs_ids_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+cohorts-class">cohorts</a>.
</p>

<hr>
<h2 id='cohorts-class'>An S4 class to represent a set of cohorts</h2><span id='topic+cohorts-class'></span>

<h3>Description</h3>

<p>The cohorts object consists of two tables (slots) that combined form a
relational database of a subset of cohorts. Each cohort is an observation
(row) in the <code>cohorts</code> table (first table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>cohorts</code></dt><dd><p>A table of cohorts. Each cohort (row) is identified by its
<code>cohort_symbol</code>. Columns:
</p>

<dl>
<dt>cohort_symbol</dt><dd><p>Cohort symbol. Example: <code>"CECILE"</code>.</p>
</dd>
<dt>cohort_name</dt><dd><p>Cohort full name. Example: <code>"CECILE Breast Cancer
Study"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_ids</code></dt><dd><p>A table of cohorts and their associated polygenic scores
identifiers. Columns:
</p>

<dl>
<dt>cohort_symbol</dt><dd><p>Cohort symbol. Example: <code>"CECILE"</code>.</p>
</dd>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: either <code>"gwas/dev"</code> or <code>"eval"</code>.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='collect_pgs_ids'>Collect development and evaluation pgs_ids</h2><span id='topic+collect_pgs_ids'></span>

<h3>Description</h3>

<p>Collect development and evaluation pgs_ids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collect_pgs_ids(tbl_json)
</code></pre>

<hr>
<h2 id='collect_samples'>Collect samples_variants and samples_training</h2><span id='topic+collect_samples'></span>

<h3>Description</h3>

<p>Collect samples_variants and samples_training and reference them by a unique
id (sample_id).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collect_samples(tbl_json)
</code></pre>

<hr>
<h2 id='contains_question_mark'>Does a string contain a question mark?</h2><span id='topic+contains_question_mark'></span>

<h3>Description</h3>

<p>Find which strings contain a question mark. This function uses the following
regular expression: <code>[\?]</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contains_question_mark(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contains_question_mark_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="contains_question_mark_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='count'>Extract the count field from a JSON response</h2><span id='topic+count'></span>

<h3>Description</h3>

<p>This function takes a string with a JSON response and returns the value of
the count field. If it fails to match the pattern then it returns
<code>NA_integer_</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count(json_string)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="count_+3A_json_string">json_string</code></td>
<td>
<p>a string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer value.
</p>

<hr>
<h2 id='filter_by_id'>Filter PGS Catalog objects by identifier.</h2><span id='topic+filter_by_id'></span>

<h3>Description</h3>

<p>Use <code>filter_by_id</code> to filter PGS Catalog objects by their respective
identifier (<code>id</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_by_id(x, id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_by_id_+3A_x">x</code></td>
<td>
<p>An object of class either<a href="#topic+scores-class">scores</a>,
<a href="#topic+publications-class">publications</a>, <a href="#topic+traits-class">traits</a>,
<a href="#topic+performance_metrics-class">performance_metrics</a>, or <a href="#topic+sample_sets-class">sample_sets</a>.</p>
</td></tr>
<tr><td><code id="filter_by_id_+3A_id">id</code></td>
<td>
<p>Identifier.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class either <a href="#topic+scores-class">scores</a>,
<a href="#topic+publications-class">publications</a>, <a href="#topic+traits-class">traits</a>,
<a href="#topic+performance_metrics-class">performance_metrics</a>, or <a href="#topic+sample_sets-class">sample_sets</a>.
</p>

<hr>
<h2 id='first_non_na'>Returns the position of the first non-NA value</h2><span id='topic+first_non_na'></span>

<h3>Description</h3>

<p>Returns the position of the first non-NA value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>first_non_na(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="first_non_na_+3A_x">x</code></td>
<td>
<p>An atomic vector.</p>
</td></tr>
</table>

<hr>
<h2 id='get_ancestry_categories'>Get ancestry categories and classes</h2><span id='topic+get_ancestry_categories'></span>

<h3>Description</h3>

<p>Retrieves ancestry categories and classes. This function simply returns the
object <code><a href="#topic+ancestry_categories">ancestry_categories</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ancestry_categories()
</code></pre>


<h3>Value</h3>

<p>A tibble with ancestry categories, classes and associated
information. See  <code><a href="#topic+ancestry_categories">ancestry_categories</a></code> for details about each
column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_ancestry_categories()

</code></pre>

<hr>
<h2 id='get_ancestry_symbol_mappings'>Get PGS Catalog Ancestry Symbol Mappings</h2><span id='topic+get_ancestry_symbol_mappings'></span>

<h3>Description</h3>

<p>Retrieves the mappings between the ancestry class symbols and ancestry class
via the PGS Catalog REST API. Note: this function is not exported and should
only be used for debugging reasons. Use in alternative
<code><a href="#topic+get_ancestry_categories">get_ancestry_categories</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ancestry_symbol_mappings(
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_ancestry_symbol_mappings_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_ancestry_symbol_mappings_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_ancestry_symbol_mappings_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar indicating download
progress from the REST API server.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a tibble of mappings between the ancestry symbols and their
name, e.g. EUR and European, respectively.
</p>

<hr>
<h2 id='get_cohorts'>Get PGS Catalog Cohorts</h2><span id='topic+get_cohorts'></span>

<h3>Description</h3>

<p>Retrieves cohorts via the PGS Catalog REST API. Please note that all
<code>cohort_symbol</code> is vectorised, thus allowing for batch mode search.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cohorts(
  cohort_symbol = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_cohorts_+3A_cohort_symbol">cohort_symbol</code></td>
<td>
<p>A cohort symbol or <code>NULL</code> if all cohorts are
intended.</p>
</td></tr>
<tr><td><code id="get_cohorts_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_cohorts_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_cohorts_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar indicating download
progress from the REST API server.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+cohorts-class">cohorts</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about specific cohorts by their symbols (acronyms)
get_cohorts(cohort_symbol = c('23andMe', 'IPOBCS'))

# Get info on all cohorts (may take a few minutes to download)
## Not run: 
get_cohorts()

## End(Not run)

</code></pre>

<hr>
<h2 id='get_performance_metrics'>Get PGS Catalog Performance Metrics</h2><span id='topic+get_performance_metrics'></span>

<h3>Description</h3>

<p>Retrieves performance metrics via the PGS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all performance metrics that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
associations that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_performance_metrics(
  ppm_id = NULL,
  pgs_id = NULL,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_performance_metrics_+3A_ppm_id">ppm_id</code></td>
<td>
<p>A character vector of PGS Catalog performance metrics accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_performance_metrics_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A <code>character</code> vector of PGS Catalog score accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_performance_metrics_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how performance metrics retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same performance metrics found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_performance_metrics_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all performance metrics are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_performance_metrics_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_performance_metrics_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_performance_metrics_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search.
</p>


<h3>Value</h3>

<p>A <a href="#topic+performance_metrics-class">performance_metrics</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get performance metrics catalogued with identifier 'PPM000001'
get_performance_metrics(ppm_id = 'PPM000001')

# Get performance metrics associated with polygenic score id 'PGS000001'
get_performance_metrics(pgs_id = 'PGS000001')

# To retrieve all catalogued performed metrics in PGS Catalog you simply
# leave the parameters `ppm_id` and `pgs_id` as `NULL`.
get_performance_metrics()

## End(Not run)

</code></pre>

<hr>
<h2 id='get_publications'>Get PGS Catalog Publications</h2><span id='topic+get_publications'></span>

<h3>Description</h3>

<p>Retrieves PGS publications via the PGS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all publications that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
associations that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_publications(
  pgp_id = NULL,
  pgs_id = NULL,
  pubmed_id = NULL,
  author = NULL,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_publications_+3A_pgp_id">pgp_id</code></td>
<td>
<p>A character vector of PGS Catalog publication accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A <code>character</code> vector of PGS Catalog score accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An <code>integer</code> vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_author">author</code></td>
<td>
<p>A character vector of author names, any author in the list of
authors in a publication, .e.g. <code>'Mavaddat'</code>.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how publications retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same publications found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all publications are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_publications_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search. For more details see the help vignette:
<code>vignette("getting-pgs-publications", package = "quincunx")</code>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+publications-class">publications</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get PGS publications by their identifier
get_publications(pgp_id = c('PGP000001', 'PGP000002'))

# By polygenic score identifier
get_publications(pgs_id = 'PGS000003')

# By PubMed identifier
get_publications(pubmed_id = '30554720')

# By author's last name
get_publications(author = 'Natarajan')

## End(Not run)
</code></pre>

<hr>
<h2 id='get_releases'>Get PGS Catalog Releases</h2><span id='topic+get_releases'></span>

<h3>Description</h3>

<p>This function retrieves PGS Catalog release information. Note that the
columns <code>pgs_id</code>, <code>ppm_id</code> and <code>pgp_id</code> contain in each
element a vector. These columns can be unnested using
<code><a href="tidyr.html#topic+unnest_longer">unnest_longer</a></code> (see Examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_releases(
  date = "latest",
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_releases_+3A_date">date</code></td>
<td>
<p>One or more dates formatted as <code>"YYYY-MM-DD"</code> for respective
releases, <code>"latest"</code> for the latest release, or <code>"all"</code> for all
releases.</p>
</td></tr>
<tr><td><code id="get_releases_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print information about the underlying requests to
the REST API server.</p>
</td></tr>
<tr><td><code id="get_releases_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings about the underlying requests to
the REST API server.</p>
</td></tr>
<tr><td><code id="get_releases_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar indicating download
progress from the REST API server.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame where each row is a release. Columns are:
</p>

<dl>
<dt>date</dt><dd><p>Release date.</p>
</dd>
<dt>n_pgs</dt><dd><p>Number of released Polygenic Score (PGS) identifiers
(<code>pgs_id</code>).</p>
</dd>
<dt>n_ppm</dt><dd><p>Number of released Performance Metric (PPM) identifiers
(<code>ppm_id</code>).</p>
</dd>
<dt>n_pgp</dt><dd><p>Number of released PGS Catalog Publication (PGP) identifiers
(<code>pgp_id</code>).</p>
</dd>
<dt>pgs_id</dt><dd><p>Released Polygenic Score (PGS) identifiers.</p>
</dd>
<dt>ppm_id</dt><dd><p>Released Performance Metric (PPM) identifiers.</p>
</dd>
<dt>pgp_id</dt><dd><p>Released PGS Catalog Publication (PGP) identifiers.</p>
</dd>
<dt>notes</dt><dd><p>News about the release.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get the latest release
get_releases()
get_releases(date = 'latest')

# Get all releases
get_releases(date = 'all')

# Get a specific release by date
get_releases(date = '2020-08-19')

## End(Not run)

</code></pre>

<hr>
<h2 id='get_sample_sets'>Get PGS Catalog Sample Sets</h2><span id='topic+get_sample_sets'></span>

<h3>Description</h3>

<p>Retrieves sample sets via the PGS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all sample sets that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
associations that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sample_sets(
  pss_id = NULL,
  pgs_id = NULL,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sample_sets_+3A_pss_id">pss_id</code></td>
<td>
<p>A character vector of PGS Catalog sample sets accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_sample_sets_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A <code>character</code> vector of PGS Catalog score accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_sample_sets_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how performance metrics retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same sample sets found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_sample_sets_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all sample sets are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_sample_sets_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_sample_sets_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_sample_sets_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar indicating download
progress from the REST API server.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search.
</p>


<h3>Value</h3>

<p>A <a href="#topic+sample_sets-class">sample_sets</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Search by PGS identifier
get_sample_sets(pgs_id = 'PGS000013')

# Search by the PSS identifier
get_sample_sets(pss_id = 'PSS000068')

## End(Not run)
</code></pre>

<hr>
<h2 id='get_scores'>Get PGS Catalog Scores</h2><span id='topic+get_scores'></span>

<h3>Description</h3>

<p>Retrieves polygenic scores via the PGS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all scores that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
associations that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_scores(
  pgs_id = NULL,
  efo_id = NULL,
  pubmed_id = NULL,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_scores_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A <code>character</code> vector of PGS Catalog score accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An <code>integer</code> vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how scores retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same scores found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all scores are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_scores_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search.
</p>


<h3>Value</h3>

<p>A <a href="#topic+scores-class">scores</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# By `pgs_id`
get_scores(pgs_id = 'PGS000088')

# By `efo_id`
get_scores(efo_id = 'EFO_0007992')

# By `pubmed_id`
get_scores(pubmed_id = '25748612')

## End(Not run)
</code></pre>

<hr>
<h2 id='get_trait_categories'>Get PGS Catalog Trait Categories</h2><span id='topic+get_trait_categories'></span>

<h3>Description</h3>

<p>Retrieves all trait categories via the PGS Catalog REST API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trait_categories(verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_trait_categories_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_trait_categories_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_trait_categories_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar indicating download
progress from the REST API server.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+trait_categories-class">trait_categories</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
get_trait_categories(progress_bar = FALSE)

</code></pre>

<hr>
<h2 id='get_traits'>Get PGS Catalog Traits</h2><span id='topic+get_traits'></span>

<h3>Description</h3>

<p>Retrieves traits via the PGS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all traits that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
traits that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits(
  efo_id = NULL,
  trait_term = NULL,
  exact_term = TRUE,
  include_children = FALSE,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_traits_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_trait_term">trait_term</code></td>
<td>
<p>A character vector of terms to be matched against trait
identifiers (<code>efo_id</code>), trait descriptions, synonyms thereof,
externally mapped terms, or even trait categories.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_exact_term">exact_term</code></td>
<td>
<p>A logical value, indicating whether to match the
<code>trait_term</code> exactly (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="get_traits_+3A_include_children">include_children</code></td>
<td>
<p>A logical value, indicating whether to include child
traits or not.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how performance metrics retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same sample sets found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all traits are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar indicating download
progress from the REST API server.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search.
</p>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get a trait by its EFO identifier
get_traits(efo_id = 'EFO_0004631')

# Get a trait by matching a term in EFO identifier (`efo_id`), label,
# description synonyms, categories, or external mapped terms
get_traits(trait_term = 'stroke', exact_term = FALSE)

# Get a trait matching its name (`trait`) exactly (default)
get_traits(trait_term = 'stroke', exact_term = TRUE)

# Get traits, excluding its children traits (default)
get_traits(trait_term = 'breast cancer')

# Get traits, including its children traits (check column `is_child` for
# child traits)
get_traits(trait_term = 'breast cancer', include_children = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='is_paginated'>Is the response paginated?</h2><span id='topic+is_paginated'></span>

<h3>Description</h3>

<p>Checks if the response is paginated by checking if a count element exists in
the response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_paginated(json_string)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_paginated_+3A_json_string">json_string</code></td>
<td>
<p>a string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value.
</p>

<hr>
<h2 id='is_pgs_id'>Is a string a PGS Catalog identifier?</h2><span id='topic+is_pgs_id'></span>

<h3>Description</h3>

<p>Find which strings are valid PGS Catalog identifiers (returns <code>TRUE</code>).
Association IDs are tested against the following regular expression:
<code>^PGS\\d{6}$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_pgs_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_pgs_id_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_pgs_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='is_pubmed_id'>Is a string a PubMed ID?</h2><span id='topic+is_pubmed_id'></span>

<h3>Description</h3>

<p>Find which strings are valid PubMed IDs (returns <code>TRUE</code>). PubMed IDs are
tested against the following regular expression: <code>^\\d+$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_pubmed_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_pubmed_id_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_pubmed_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='list_to_s4'>Convert a named list to an S4 object</h2><span id='topic+list_to_s4'></span>

<h3>Description</h3>

<p>Convert a named list to an S4 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_to_s4(list, class)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="list_to_s4_+3A_list">list</code></td>
<td>
<p>list</p>
</td></tr>
<tr><td><code id="list_to_s4_+3A_class">class</code></td>
<td>
<p>character vector indicating the S4 class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S4 object of class <code>class</code>.
</p>

<hr>
<h2 id='msg_400'>Pretty printing of 404 error</h2><span id='topic+msg_400'></span>

<h3>Description</h3>

<p>Pretty printing of 404 error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msg_400(response)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msg_400_+3A_response">response</code></td>
<td>
<p>A <code><a href="httr.html#topic+response">response</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect: printing.
</p>

<hr>
<h2 id='msg_404'>Pretty printing of 404 error</h2><span id='topic+msg_404'></span>

<h3>Description</h3>

<p>Pretty printing of 404 error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msg_404(response)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msg_404_+3A_response">response</code></td>
<td>
<p>A <code><a href="httr.html#topic+response">response</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect: printing.
</p>

<hr>
<h2 id='msg_empty'>Pretty printing of empty response</h2><span id='topic+msg_empty'></span>

<h3>Description</h3>

<p>Pretty printing of empty response
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msg_empty(response)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msg_empty_+3A_response">response</code></td>
<td>
<p>A <code><a href="httr.html#topic+response">response</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect: printing.
</p>

<hr>
<h2 id='n'>Number of PGS Catalog entities</h2><span id='topic+n'></span><span id='topic+n+2Cscores-method'></span><span id='topic+n+2Cpublications-method'></span><span id='topic+n+2Ctraits-method'></span><span id='topic+n+2Cperformance_metrics-method'></span><span id='topic+n+2Csample_sets-method'></span><span id='topic+n+2Ccohorts-method'></span><span id='topic+n+2Ctrait_categories-method'></span><span id='topic+n+2Creleases-method'></span>

<h3>Description</h3>

<p>This function returns the number of entities in a PGS Catalog object. To
avoid ambiguity with <code>dplyr::n()</code> use <code>quincunx::n()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n(x, unique = FALSE)

## S4 method for signature 'scores'
n(x, unique = FALSE)

## S4 method for signature 'publications'
n(x, unique = FALSE)

## S4 method for signature 'traits'
n(x, unique = FALSE)

## S4 method for signature 'performance_metrics'
n(x, unique = FALSE)

## S4 method for signature 'sample_sets'
n(x, unique = FALSE)

## S4 method for signature 'cohorts'
n(x, unique = FALSE)

## S4 method for signature 'trait_categories'
n(x, unique = FALSE)

## S4 method for signature 'releases'
n(x, unique = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="n_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+scores-class">scores</a>, <a href="#topic+publications-class">publications</a>,
<a href="#topic+traits-class">traits</a>, <a href="#topic+performance_metrics-class">performance_metrics</a>,
<a href="#topic+sample_sets-class">sample_sets</a>, <a href="#topic+cohorts-class">cohorts</a>,
<a href="#topic+trait_categories-class">trait_categories</a> or <a href="#topic+releases-class">releases</a> object.</p>
</td></tr>
<tr><td><code id="n_+3A_unique">unique</code></td>
<td>
<p>Whether to count only unique entries (<code>TRUE</code>) or not
(<code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer scalar.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Return the number of polygenic scores in a scores object:
my_scores &lt;- get_scores(pgs_id = c('PGS000007', 'PGS000007', 'PGS000042'))
n(my_scores)

# If you want to count unique scores only, then use the `unique` parameter:
n(my_scores, unique = TRUE)

# Total number of curated publications in the PGS Catalog:
all_pub &lt;- get_publications(interactive = FALSE, progress_bar = FALSE)
n(all_pub)

# Total number of curated traits in the PGS Catalog:
all_traits &lt;- get_traits(interactive = FALSE, progress_bar = FALSE)
n(all_traits)

</code></pre>

<hr>
<h2 id='n_pages'>Number of pages</h2><span id='topic+n_pages'></span>

<h3>Description</h3>

<p>Determine the number of pages to be requested from the total number of
results (<code>count</code>) and the number of results per page (<code>limit</code>). The
wording used here &mdash; <code>count</code> and <code>limit</code> &mdash; is borrowed from the
PGS Catalog REST API documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n_pages(count, limit = 50L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="n_pages_+3A_count">count</code></td>
<td>
<p>total number of results.</p>
</td></tr>
<tr><td><code id="n_pages_+3A_limit">limit</code></td>
<td>
<p>number of results per page.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of pages, an integer value.
</p>

<hr>
<h2 id='no_msg'>Pretty printing of no error message</h2><span id='topic+no_msg'></span>

<h3>Description</h3>

<p>Pretty printing of no error message
</p>


<h3>Usage</h3>

<pre><code class='language-R'>no_msg(response)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="no_msg_+3A_response">response</code></td>
<td>
<p>A <code><a href="httr.html#topic+response">response</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect: printing.
</p>

<hr>
<h2 id='nr_to_na'>Convert NR (Not Recorded) to NA (Not Available)</h2><span id='topic+nr_to_na'></span>

<h3>Description</h3>

<p>This function converts the <code>'NR'</code> string to <code>NA</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nr_to_na(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nr_to_na_+3A_x">x</code></td>
<td>
<p>a character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector whose <code>'NR'</code> have been replaced with
<code>NA</code>.
</p>

<hr>
<h2 id='offsets'>Generate offset values</h2><span id='topic+offsets'></span>

<h3>Description</h3>

<p>Generate offset values to be passed to the PGS Catalog REST API endpoints.
The offset parameter, together with the limit parameter, allow to access to a
desired range of results. The offset parameter specifies the starting index
of the range of results desired. It is a zero based index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>offsets(count, limit)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="offsets_+3A_count">count</code></td>
<td>
<p>total number of results.</p>
</td></tr>
<tr><td><code id="offsets_+3A_limit">limit</code></td>
<td>
<p>number of results per page.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Offset values, an integer vector.
</p>

<hr>
<h2 id='open_in_dbsnp'>Browse dbSNP from SNP identifiers.</h2><span id='topic+open_in_dbsnp'></span>

<h3>Description</h3>

<p>This function launches the web browser at dbSNP and opens a tab for each SNP
identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_dbsnp(variant_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="open_in_dbsnp_+3A_variant_id">variant_id</code></td>
<td>
<p>A variant identifier, a character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful. Note however that this
function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
open_in_dbsnp('rs56261590')

</code></pre>

<hr>
<h2 id='open_in_pgs_catalog'>Browse PGS Catalog entities from the PGS Catalog Web Graphical User Interface</h2><span id='topic+open_in_pgs_catalog'></span>

<h3>Description</h3>

<p>This function launches the web browser and opens a tab for each identifier on
the PGS Catalog web graphical user interface: <a href="https://www.pgscatalog.org/">https://www.pgscatalog.org/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_pgs_catalog(
  identifier = NULL,
  pgs_catalog_entity = c("pgs", "pgp", "pss", "efo")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="open_in_pgs_catalog_+3A_identifier">identifier</code></td>
<td>
<p>A vector of identifiers. The identifiers can be: PGS, PGP,
PSS or EFO identifiers.</p>
</td></tr>
<tr><td><code id="open_in_pgs_catalog_+3A_pgs_catalog_entity">pgs_catalog_entity</code></td>
<td>
<p>Either <code>'pgs'</code> (default), <code>'pgp'</code>,
<code>'pss'</code>, <code>'efo'</code>. This argument indicates the type of the
identifiers passed in <code>identifier</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful, or <code>FALSE</code> otherwise. But
note that this function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Open in PGS scores Catalog Web Graphical User Interface
open_in_pgs_catalog(c('PGS000001', 'PGS000002'))

# Open PGS Catalog Publications
open_in_pgs_catalog(c('PGP000001', 'PGP000002'),
  pgs_catalog_entity = 'pgp')

# Open Sample Sets (PSS)
open_in_pgs_catalog(c('PSS000001', 'PSS000002'),
  pgs_catalog_entity = 'pss')

# Open EFO traits (EFO)
open_in_pgs_catalog(c('EFO_0001645', 'MONDO_0007254'),
  pgs_catalog_entity = 'efo')

</code></pre>

<hr>
<h2 id='open_in_pubmed'>Browse PubMed from PubMed identifiers.</h2><span id='topic+open_in_pubmed'></span>

<h3>Description</h3>

<p>This function launches the web browser and opens a tab for each PubMed
citation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_pubmed(pubmed_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="open_in_pubmed_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>A PubMed identifier, either a character or an integer
vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful. Note however that this
function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
open_in_pubmed(c('26301688', '30595370'))

</code></pre>

<hr>
<h2 id='performance_metrics'>Constructor for the S4 performance_metrics object.</h2><span id='topic+performance_metrics'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+performance_metrics-class">performance_metrics</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>performance_metrics(
  performance_metrics = s4ppm_performance_metrics_tbl(),
  publications = s4ppm_publications_tbl(),
  sample_sets = s4ppm_sample_sets_tbl(),
  samples = s4ppm_samples_tbl(),
  demographics = s4ppm_demographics_tbl(),
  cohorts = s4ppm_pgs_cohorts_tbl(),
  pgs_effect_sizes = s4ppm_pgs_effect_sizes_tbl(),
  pgs_classification_metrics = s4ppm_pgs_classification_metrics_tbl(),
  pgs_other_metrics = s4ppm_pgs_other_metrics_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="performance_metrics_+3A_performance_metrics">performance_metrics</code></td>
<td>
<p>A <code>s4ppm_performance_metrics_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_publications">publications</code></td>
<td>
<p>A <code>s4ppm_publications_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_sample_sets">sample_sets</code></td>
<td>
<p>A <code>s4ppm_sample_sets_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_samples">samples</code></td>
<td>
<p>A <code>s4ppm_samples_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_demographics">demographics</code></td>
<td>
<p>A <code>s4ppm_demographics_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_cohorts">cohorts</code></td>
<td>
<p>A <code>s4ppm_pgs_cohorts_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_pgs_effect_sizes">pgs_effect_sizes</code></td>
<td>
<p>A <code>s4ppm_pgs_effect_sizes_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_pgs_classification_metrics">pgs_classification_metrics</code></td>
<td>
<p>A <code>s4ppm_pgs_classification_metrics_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="performance_metrics_+3A_pgs_other_metrics">pgs_other_metrics</code></td>
<td>
<p>A <code>s4ppm_pgs_other_metrics_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+performance_metrics-class">performance_metrics</a>.
</p>

<hr>
<h2 id='performance_metrics-class'>An S4 class to represent a set of PGS Catalog Performance Metrics</h2><span id='topic+performance_metrics-class'></span>

<h3>Description</h3>

<p>The performance_metrics object consists of nine tables (slots) that combined
form a relational database of a subset of performance metrics. Each
performance metric is an observation (row) in the <code>scores</code> table (first
table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>performance_metrics</code></dt><dd><p>A table of PGS Performance Metrics (PPM). Each PPM (row) is
uniquely identified by the <code>ppm_id</code> column. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>reported_trait</dt><dd><p>The author-reported trait that the PGS has been
developed to predict. Example: <code>"Breast Cancer"</code>.</p>
</dd>
<dt>covariates</dt><dd><p>Comma-separated list of covariates used in the prediction
model to evaluate the PGS.</p>
</dd>
<dt>comments</dt><dd><p>Any other information relevant to the understanding of the
performance metrics.</p>
</dd>
</dl>
</dd>
<dt><code>publications</code></dt><dd><p>A table of publications. Each publication (row) is
uniquely identified by the column <code>pgp_id</code>. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>pgp_id</dt><dd><p>PGS Publication identifier. Example: <code>"PGP000001"</code>.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier. Example: <code>"25855707"</code>.</p>
</dd>
<dt>publication_date</dt><dd><p>Publication date. Example: <code>"2020-09-28"</code>. Note
that the class of <code>publication_date</code> is <code><a href="base.html#topic+Date">Date</a></code>.</p>
</dd>
<dt>publication</dt><dd><p>Abbreviated name of the journal. Example: <code>"Am J Hum
Genet"</code>.</p>
</dd>
<dt>title</dt><dd><p>Publication title.</p>
</dd>
<dt>author_fullname</dt><dd><p>First author of the publication. Example:
<code>'Mavaddat N'</code>.</p>
</dd>
<dt>doi</dt><dd><p>Digital Object Identifier (DOI). This variable is also curated to
allow unpublished work (e.g. preprints) to be added to the catalog. Example:
<code>"10.1093/jnci/djv036"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>sample_sets</code></dt><dd><p>A table of sample sets. Each sample set (row) is uniquely
identified by the column <code>pss_id</code>. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>samples</code></dt><dd><p>A table of samples. Each sample (row) is uniquely identified by
the combination of values from the columns: <code>ppm_id</code>, <code>pss_id</code>,
and <code>sample_id</code>. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: should be always Evaluation (<code>"eval"</code>).</p>
</dd>
<dt>sample_size</dt><dd><p>Number of individuals included in the sample.</p>
</dd>
<dt>sample_cases</dt><dd><p>Number of cases.</p>
</dd>
<dt>sample_controls</dt><dd><p>Number of controls.</p>
</dd>
<dt>sample_percent_male</dt><dd><p>Percentage of male participants.</p>
</dd>
<dt>phenotype_description</dt><dd><p>Detailed phenotype description.</p>
</dd>
<dt>ancestry_category</dt><dd><p>Author reported ancestry is mapped to the best matching
ancestry category from the NHGRI-EBI GWAS Catalog framework (see
<code><a href="#topic+ancestry_categories">ancestry_categories</a></code>) for possible values.</p>
</dd>
<dt>ancestry</dt><dd><p>A more detailed description of sample ancestry
that usually matches the most specific description described by the authors
(e.g. French, Chinese).</p>
</dd>
<dt>country</dt><dd><p>Author reported countries of recruitment (if available).</p>
</dd>
<dt>ancestry_additional_description</dt><dd><p>Any additional description not
captured in the other columns (e.g. founder or genetically isolated
populations, or further description of admixed samples).</p>
</dd>
<dt>study_id</dt><dd><p>Associated GWAS Catalog study accession identifier, e.g.,
<code>"GCST002735"</code>.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier.</p>
</dd>
<dt>cohorts_additional_description</dt><dd><p>Any additional description about the
samples (e.g. sub-cohort information).</p>
</dd>
</dl>
</dd>
<dt><code>demographics</code></dt><dd><p>A table of sample demographics' variables. Each
demographics' variable (row) is uniquely identified by the combination of
values from the columns: <code>ppm_id</code>, <code>pss_id</code>, <code>sample_id</code>,
and <code>variable</code>. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate identifier to
identify each sample.</p>
</dd>
<dt>variable</dt><dd><p>Demographics variable. Following columns report about the
indicated variable.</p>
</dd>
<dt>estimate_type</dt><dd><p>Type of statistical estimate for variable.</p>
</dd>
<dt>estimate</dt><dd><p>The variable's statistical value.</p>
</dd>
<dt>unit</dt><dd><p>Unit of the variable.</p>
</dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p>
</dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p>
</dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p>
</dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p>
</dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p>
</dd>
</dl>
</dd>
<dt><code>cohorts</code></dt><dd><p>A table of cohorts. Each cohort (row) is uniquely identified by
the combination of values from the columns: <code>ppm_id</code>, <code>sample_id</code>
and <code>cohort_symbol</code>. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p>
</dd>
<dt>cohort_symbol</dt><dd><p>Cohort symbol.</p>
</dd>
<dt>cohort_name</dt><dd><p>Cohort full name.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_effect_sizes</code></dt><dd><p>A table of effect sizes per standard deviation change
in PGS. Examples include regression coefficients (betas) for continuous
traits, odds ratios (OR) and/or hazard ratios (HR) for dichotomous traits
depending on the availability of time-to-event data. Each effect size is
uniquely identified by the combination of values from the columns:
<code>ppm_id</code> and <code>effect_size_id</code>. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>effect_size_id</dt><dd><p>Effect size identifier. This is a surrogate identifier to
identify each effect size.</p>
</dd>
<dt>estimate_type_long</dt><dd><p>Long notation of the effect size (e.g. Odds Ratio).</p>
</dd>
<dt>estimate_type</dt><dd><p>Short notation of the effect size (e.g. OR).</p>
</dd>
<dt>estimate</dt><dd><p>The estimate's value.</p>
</dd>
<dt>unit</dt><dd><p>Unit of the estimate.</p>
</dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p>
</dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p>
</dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p>
</dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p>
</dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_classification_metrics</code></dt><dd><p>A table of classification metrics. Examples
include the Area under the Receiver Operating Characteristic (AUROC) or
Harrell's C-index (Concordance statistic). Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>classification_metrics_id</dt><dd><p>Classification metric identifier. This is a
surrogate identifier to identify each classification metric.</p>
</dd>
<dt>estimate_type_long</dt><dd><p>Long notation of the classification metric (e.g.
Concordance Statistic).</p>
</dd>
<dt>estimate_type</dt><dd><p>Short notation classification metric (e.g. C-index).</p>
</dd>
<dt>estimate</dt><dd><p>The estimate's value.</p>
</dd>
<dt>unit</dt><dd><p>Unit of the estimate.</p>
</dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p>
</dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p>
</dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p>
</dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p>
</dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_other_metrics</code></dt><dd><p>A table of other metrics that are neither effect
sizes nor classification metrics. Examples include: R² (proportion of the
variance explained), or reclassification metrics. Columns:
</p>

<dl>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
<dt>other_metrics_id</dt><dd><p>Other metric identifier. This is a
surrogate identifier to identify each metric.</p>
</dd>
<dt>estimate_type_long</dt><dd><p>Long notation of the metric. Example: &quot;Proportion of the variance explained&quot;.</p>
</dd>
<dt>estimate_type</dt><dd><p>Short notation metric. Example: &quot;R²&quot;.</p>
</dd>
<dt>estimate</dt><dd><p>The estimate's value.</p>
</dd>
<dt>unit</dt><dd><p>Unit of the estimate.</p>
</dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p>
</dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p>
</dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p>
</dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p>
</dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='pgp_to_pgs'>Map PGP identifiers to PGS identifiers</h2><span id='topic+pgp_to_pgs'></span>

<h3>Description</h3>

<p>Map PGP identifiers to PGS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgp_to_pgs(
  pgp_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgp_to_pgs_+3A_pgp_id">pgp_id</code></td>
<td>
<p>A character vector of PGS Catalog Publication identifiers,
e.g., &quot;PGP000001&quot;. If <code>NULL</code> then returns results for all PGP
identifiers in the Catalog.</p>
</td></tr>
<tr><td><code id="pgp_to_pgs_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgp_to_pgs_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgp_to_pgs_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgp_id</code> and <code>pgs_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgp_to_pgs('PGP000001')
pgp_to_pgs(c('PGP000017', 'PGP000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pgp_to_ppm'>Map PGP identifiers to PPM identifiers</h2><span id='topic+pgp_to_ppm'></span>

<h3>Description</h3>

<p>Map PGP identifiers to PPM identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgp_to_ppm(
  pgp_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgp_to_ppm_+3A_pgp_id">pgp_id</code></td>
<td>
<p>A character vector of PGS Catalog Publication identifiers,
e.g., &quot;PGP000001&quot;. If <code>NULL</code> then returns results for all PGP
identifiers in the Catalog.</p>
</td></tr>
<tr><td><code id="pgp_to_ppm_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgp_to_ppm_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgp_to_ppm_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgp_id</code> and <code>ppm_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgp_to_ppm('PGP000001')
pgp_to_ppm(c('PGP000017', 'PGP000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pgp_to_pss'>Map PGP identifiers to PSS identifiers</h2><span id='topic+pgp_to_pss'></span>

<h3>Description</h3>

<p>Map PGP identifiers to PSS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgp_to_pss(
  pgp_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgp_to_pss_+3A_pgp_id">pgp_id</code></td>
<td>
<p>A character vector of PGS Catalog Publication identifiers,
e.g., &quot;PGP000001&quot;. If <code>NULL</code> then returns results for all PGP
identifiers in the Catalog.</p>
</td></tr>
<tr><td><code id="pgp_to_pss_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgp_to_pss_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgp_to_pss_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgp_id</code> and <code>pss_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgp_to_pss('PGP000001')
pgp_to_pss(c('PGP000017', 'PGP000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pgs_server'>PGS REST API server</h2><span id='topic+pgs_server'></span>

<h3>Description</h3>

<p>PGS REST API server
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgs_server()
</code></pre>


<h3>Value</h3>

<p>A string containing PGS REST API server URL.
</p>

<hr>
<h2 id='pgs_to_pgp'>Map PGS identifiers to PGP identifiers</h2><span id='topic+pgs_to_pgp'></span>

<h3>Description</h3>

<p>Map PGS identifiers to PGP identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgs_to_pgp(
  pgs_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgs_to_pgp_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A character vector of PGS identifiers,
e.g., &quot;PGS000001&quot;. If <code>NULL</code> then returns results for all PGS
identifiers in the Catalog.</p>
</td></tr>
<tr><td><code id="pgs_to_pgp_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgs_to_pgp_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgs_to_pgp_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgs_id</code> and <code>pgp_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgs_to_pgp('PGS000001')
pgs_to_pgp(c('PGS000017', 'PGS000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pgs_to_ppm'>Map PGS identifiers to PPM identifiers</h2><span id='topic+pgs_to_ppm'></span>

<h3>Description</h3>

<p>Map PGS identifiers to PPM identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgs_to_ppm(pgs_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgs_to_ppm_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A character vector of PGS identifiers,
e.g., &quot;PGS000001&quot;.</p>
</td></tr>
<tr><td><code id="pgs_to_ppm_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgs_to_ppm_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgs_to_ppm_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgs_id</code> and <code>ppm_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgs_to_ppm('PGS000001')
pgs_to_ppm(c('PGS000017', 'PGS000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pgs_to_pss'>Map PGS identifiers to PSS identifiers</h2><span id='topic+pgs_to_pss'></span>

<h3>Description</h3>

<p>Map PGS identifiers to PSS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgs_to_pss(
  pgs_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgs_to_pss_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A character vector of PGS identifiers,
e.g., &quot;PGS000001&quot;. If <code>NULL</code> then returns results for all PGS
identifiers in the Catalog.</p>
</td></tr>
<tr><td><code id="pgs_to_pss_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgs_to_pss_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgs_to_pss_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgs_id</code> and <code>pss_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgs_to_pss('PGS000001')
pgs_to_pss(c('PGS000017', 'PGS000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pgs_to_study'>Map PGS identifiers to GWAS study identifiers</h2><span id='topic+pgs_to_study'></span>

<h3>Description</h3>

<p>Map PGS identifiers to GWAS study identifiers. Retrieves GWAS study
identifiers associated with samples used in the discovery stage of queried
PGS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pgs_to_study(
  pgs_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pgs_to_study_+3A_pgs_id">pgs_id</code></td>
<td>
<p>A character vector of PGS Catalog score accession identifiers.,
e.g., &quot;PGS000001&quot;. If <code>NULL</code> then returns results for all PGS
identifiers in the Catalog.</p>
</td></tr>
<tr><td><code id="pgs_to_study_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pgs_to_study_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pgs_to_study_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pgs_id</code> and <code>study_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pgs_to_study('PGS000001')
# Unmappable pgs ids will be missing, e.g., PGS000023
pgs_to_study(c('PGS000013', 'PGS000023'))

## End(Not run)
</code></pre>

<hr>
<h2 id='ppm_to_pgp'>Map PPM identifiers to PGP identifiers</h2><span id='topic+ppm_to_pgp'></span>

<h3>Description</h3>

<p>Map PPM identifiers to PGP identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppm_to_pgp(ppm_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppm_to_pgp_+3A_ppm_id">ppm_id</code></td>
<td>
<p>A character vector of PPM identifiers,
e.g., &quot;PPM000001&quot;.</p>
</td></tr>
<tr><td><code id="ppm_to_pgp_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="ppm_to_pgp_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="ppm_to_pgp_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>ppm_id</code> and <code>pgp_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ppm_to_pgp('PPM000001')
ppm_to_pgp(c('PPM000017', 'PPM000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='ppm_to_pgs'>Map PPM identifiers to PGS identifiers</h2><span id='topic+ppm_to_pgs'></span>

<h3>Description</h3>

<p>Map PPM identifiers to PGS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppm_to_pgs(ppm_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppm_to_pgs_+3A_ppm_id">ppm_id</code></td>
<td>
<p>A character vector of PPM identifiers,
e.g., &quot;PPPM000001&quot;.</p>
</td></tr>
<tr><td><code id="ppm_to_pgs_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="ppm_to_pgs_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="ppm_to_pgs_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>ppm_id</code> and <code>pgs_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ppm_to_pgs('PPM000001')
ppm_to_pgs(c('PPM000017', 'PPM000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='ppm_to_pss'>Map PPM identifiers to PSS identifiers</h2><span id='topic+ppm_to_pss'></span>

<h3>Description</h3>

<p>Map PPM identifiers to PSS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppm_to_pss(ppm_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppm_to_pss_+3A_ppm_id">ppm_id</code></td>
<td>
<p>A character vector of PPM identifiers,
e.g., &quot;PPM000001&quot;.</p>
</td></tr>
<tr><td><code id="ppm_to_pss_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="ppm_to_pss_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="ppm_to_pss_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>ppm_id</code> and <code>pss_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ppm_to_pss('PPM000001')
ppm_to_pss(c('PPM000017', 'PPM000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pss_to_pgp'>Map PSS identifiers to PGP identifiers</h2><span id='topic+pss_to_pgp'></span>

<h3>Description</h3>

<p>Map PSS identifiers to PGP identifiers. This is a slow function because it
starts by downloading first all Performance Metrics, as this is the linkage
between PSS and PGP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pss_to_pgp(pss_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pss_to_pgp_+3A_pss_id">pss_id</code></td>
<td>
<p>A character vector of PSS identifiers,
e.g., &quot;PSS000001&quot;.</p>
</td></tr>
<tr><td><code id="pss_to_pgp_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pss_to_pgp_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pss_to_pgp_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pss_id</code> and <code>pgp_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pss_to_pgp('PSS000001')
pss_to_pgp(c('PSS000017', 'PSS000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pss_to_pgs'>Map PSS identifiers to PGS identifiers</h2><span id='topic+pss_to_pgs'></span>

<h3>Description</h3>

<p>Map PSS identifiers to PGS identifiers. This is a slow function because it
starts by downloading first all Performance Metrics, as this is the linkage
between PSS and PGS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pss_to_pgs(pss_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pss_to_pgs_+3A_pss_id">pss_id</code></td>
<td>
<p>A character vector of PSS identifiers,
e.g., &quot;PSS000001&quot;.</p>
</td></tr>
<tr><td><code id="pss_to_pgs_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pss_to_pgs_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pss_to_pgs_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pss_id</code> and <code>pgs_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pss_to_pgs('PSS000001')
pss_to_pgs(c('PSS000017', 'PSS000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='pss_to_ppm'>Map PSS identifiers to PPM identifiers</h2><span id='topic+pss_to_ppm'></span>

<h3>Description</h3>

<p>Map PSS identifiers to PPM identifiers. This is a slow function because it
starts by downloading first all Performance Metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pss_to_ppm(pss_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pss_to_ppm_+3A_pss_id">pss_id</code></td>
<td>
<p>A character vector of PSS identifiers,
e.g., &quot;PSS000001&quot;.</p>
</td></tr>
<tr><td><code id="pss_to_ppm_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="pss_to_ppm_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="pss_to_ppm_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>pss_id</code> and <code>ppm_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pss_to_ppm('PSS000001')
pss_to_ppm(c('PSS000017', 'PSS000042'))

## End(Not run)
</code></pre>

<hr>
<h2 id='publications'>Constructor for the S4 publications object.</h2><span id='topic+publications'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+publications-class">publications</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>publications(publications = publications_tbl(), pgs_ids = pgs_ids_tbl())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="publications_+3A_publications">publications</code></td>
<td>
<p>A <code>publications_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="publications_+3A_pgs_ids">pgs_ids</code></td>
<td>
<p>A <code>pgs_ids_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+publications-class">publications</a>.
</p>

<hr>
<h2 id='publications-class'>An S4 class to represent a set of PGS Catalog Publications</h2><span id='topic+publications-class'></span>

<h3>Description</h3>

<p>The publications object consists of two tables (slots), each a table that
combined form a relational database of a subset of PGS Catalog Publications.
Each publication is an observation (row) in the <code>publications</code> table
(first table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>publications</code></dt><dd><p>A table of publications. Each publication (row) is
uniquely identified by the <code>pgp_id</code> column. Columns:
</p>

<dl>
<dt>pgp_id</dt><dd><p>PGS Publication identifier. Example: <code>"PGP000001"</code>.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier. Example: <code>"25855707"</code>.</p>
</dd>
<dt>publication_date</dt><dd><p>Publication date. Example: <code>"2020-09-28"</code>. Note
that the class of <code>publication_date</code> is <code><a href="base.html#topic+Date">Date</a></code>.</p>
</dd>
<dt>publication</dt><dd><p>Abbreviated name of the journal. Example: <code>"Am J Hum
Genet"</code>.</p>
</dd>
<dt>title</dt><dd><p>Publication title.</p>
</dd>
<dt>author_fullname</dt><dd><p>First author of the publication. Example:
<code>'Mavaddat N'</code>.</p>
</dd>
<dt>doi</dt><dd><p>Digital Object Identifier (DOI). This variable is also curated to
allow unpublished work (e.g. preprints) to be added to the catalog. Example:
<code>"10.1093/jnci/djv036"</code>.</p>
</dd>
<dt>authors</dt><dd><p>Concatenated list of all the publication authors.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_ids</code></dt><dd><p>A table of publication and associated PGS identifiers. Columns:
</p>

<dl>
<dt>pgp_id</dt><dd><p>PGS Publication identifier. Example: <code>"PGP000001"</code>.</p>
</dd>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>stage</dt><dd><p>PGS stage: either &quot;gwas/dev&quot; or &quot;eval&quot;.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='read_scoring_file'>Read a polygenic scoring file</h2><span id='topic+read_scoring_file'></span>

<h3>Description</h3>

<p>This function imports a PGS scoring file. For more information about the
scoring file schema check <code>vignette("pgs-scoring-file", package =
"quincunx")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_scoring_file(
  source,
  harmonized = FALSE,
  assembly = c("GRCh38", "GRCh37"),
  protocol = "http",
  metadata_only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_scoring_file_+3A_source">source</code></td>
<td>
<p>PGS scoring file. This can be specified in three forms: (i) a
PGS identifier, e.g. <code>"PGS000001"</code>, (ii) a path to a local file, e.g.
<code>"~/PGS000001.txt"</code> or <code>"~/PGS000001.txt.gz"</code> or (iii) a direct
URL to the PGS Catalog FTP server, e.g.
<code>"http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz"</code>.</p>
</td></tr>
<tr><td><code id="read_scoring_file_+3A_harmonized">harmonized</code></td>
<td>
<p>Whether to read an alternative, harmonized version of the
PGS scoring file. This version contains harmonized variant information.
This information is provided in extra columns whose names are prefixed with
<code>"hm_"</code>.</p>
</td></tr>
<tr><td><code id="read_scoring_file_+3A_assembly">assembly</code></td>
<td>
<p>If <code>harmonized</code> is <code>TRUE</code>, <code>assembly</code> indicates which the
genome assembly to choose for the harmonized variant data. <code>assembly</code> must
be either <code>"GRCh38"</code> (default) or <code>"GRCh37"</code>.</p>
</td></tr>
<tr><td><code id="read_scoring_file_+3A_protocol">protocol</code></td>
<td>
<p>Network protocol for communication with the PGS Catalog FTP
server: either <code>"http"</code> or <code>"ftp"</code>.</p>
</td></tr>
<tr><td><code id="read_scoring_file_+3A_metadata_only">metadata_only</code></td>
<td>
<p>Whether to read only the comment block (header) from the
scoring file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is a named list. The names are copied from the
arguments passed in <code>source</code>. Each element of the list contains
another list of two elements: <code>"metadata"</code> and <code>"data"</code>. The
&quot;metadata&quot; element contains data parsed from the header of the PGS scoring
file. The &quot;data&quot; element contains a data frame with as many rows as
variants that constitute the PGS score. The columns can vary. There are
mandatory and optional columns. The mandatory columns are those that
identify the variant, effect allele (<code>effect_allele</code>), and its
respective weight (<code>effect_weight</code>) in the score. The columns that
identify the variant can either be the <code>rsID</code> or the combination of
<code>chr_name</code> and <code>chr_position</code>. The &quot;data&quot; element will be
<code>NULL</code> is argument <code>metadata_only</code> is <code>TRUE</code>. For more
information about the scoring file schema check
<code>vignette("pgs-scoring-file", package = "quincunx")</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Read a PGS scoring file by PGS ID
# (internally, it translates the PGS ID
#  to the corresponding FTP URL)
try(read_scoring_file("PGS000655"))

# Equivalent to `read_scoring_file("PGS000655")`
url &lt;- paste0(
  "http://ftp.ebi.ac.uk/",
  "pub/databases/spot/pgs/scores/",
  "PGS000655/ScoringFiles/",
  "PGS000655.txt.gz"
)
read_scoring_file(url)


# Reading from a local file
try(read_scoring_file("~/PGS000655.txt.gz"))

## End(Not run)
</code></pre>

<hr>
<h2 id='releases'>Constructor for the S4 releases object.</h2><span id='topic+releases'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+releases-class">releases</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>releases(
  releases = s4releases_releases_tbl(),
  pgs_ids = s4releases_pgs_ids_tbl(),
  ppm_ids = s4releases_ppm_ids_tbl(),
  pgp_ids = s4releases_pgp_ids_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="releases_+3A_releases">releases</code></td>
<td>
<p>A <code>s4releases_releases_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="releases_+3A_pgs_ids">pgs_ids</code></td>
<td>
<p>A <code>s4releases_pgs_ids_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="releases_+3A_ppm_ids">ppm_ids</code></td>
<td>
<p>A <code>s4releases_ppm_ids_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="releases_+3A_pgp_ids">pgp_ids</code></td>
<td>
<p>A <code>s4releases_pgp_ids_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+releases-class">releases</a>.
</p>

<hr>
<h2 id='releases-class'>An S4 class to represent a set of PGS Catalog Releases</h2><span id='topic+releases-class'></span>

<h3>Description</h3>

<p>The releases object consists of four tables (slots) that combined form a
relational database of a subset of PGS Catalog releases. Each release is an
observation (row) in the <code>releases</code> table (first table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>releases</code></dt><dd><p>A table of PGS Catalog releases. Each release (row) is
uniquely identified by the release date (<code>date</code>). Columns:
</p>

<dl>
<dt>date</dt><dd><p>Release date.</p>
</dd>
<dt>n_pgs</dt><dd><p>Number of newly released Polygenic Scores.</p>
</dd>
<dt>n_ppm</dt><dd><p>Number of newly released PGS Performance Metrics.</p>
</dd>
<dt>n_pgp</dt><dd><p>Number of newly released PGS Publications.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_ids</code></dt><dd><p>A table of released Polygenic Scores (PGS) identifiers.
Columns:
</p>

<dl>
<dt>date</dt><dd><p>Release date.</p>
</dd>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier. Example: <code>"PGS000001"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>ppm_ids</code></dt><dd><p>A table of the released PGS Performance Metrics identifiers.
Columns:
</p>

<dl>
<dt>date</dt><dd><p>Release date.</p>
</dd>
<dt>ppm_id</dt><dd><p>A PGS Performance Metrics identifier. Example: <code>"PPM000001"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>pgp_ids</code></dt><dd><p>A table of the released PGS Publication identifiers. Columns:
</p>

<dl>
<dt>date</dt><dd><p>Release date.</p>
</dd>
<dt>pgp_id</dt><dd><p>PGS Publication identifier. Example: <code>"PGP000001"</code>.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='remap_id'>Remap identifiers</h2><span id='topic+remap_id'></span>

<h3>Description</h3>

<p>Remap identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remap_id(lst_tbls, old, new)
</code></pre>

<hr>
<h2 id='request'>Request a resource from the PGS REST API</h2><span id='topic+request'></span>

<h3>Description</h3>

<p>Performs a <code><a href="httr.html#topic+GET">GET</a></code> request on the endpoint as specified by
<code>resource_url</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>request(
  resource_url,
  base_url = pgs_server(),
  user_agent = user_agent_id(),
  verbose = FALSE,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="request_+3A_resource_url">resource_url</code></td>
<td>
<p>Endpoint URL. The endpoint is internally appended to the
<code>base_url</code>. It should start with a forward slash (<code>'/'</code>).</p>
</td></tr>
<tr><td><code id="request_+3A_base_url">base_url</code></td>
<td>
<p>The PGS REST API base URL.</p>
</td></tr>
<tr><td><code id="request_+3A_user_agent">user_agent</code></td>
<td>
<p>User agent.</p>
</td></tr>
<tr><td><code id="request_+3A_verbose">verbose</code></td>
<td>
<p>Whether to be verbose.</p>
</td></tr>
<tr><td><code id="request_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of four elements:
</p>

<dl>
<dt>resource</dt><dd><p>The URL endpoint.</p>
</dd>
<dt>code</dt><dd><p><a href="https://en.wikipedia.org/wiki/List_of_HTTP_status_codes">HTTP
status code</a>.</p>
</dd>
<dt>message</dt><dd><p>A string describing the status of the response obtained:
<code>'OK'</code> if successful or a description of the error.</p>
</dd>
<dt>json</dt><dd><p>JSON response as string.</p>
</dd>
</dl>


<hr>
<h2 id='request_all'>Request a paginated resource from the PGS REST API</h2><span id='topic+request_all'></span>

<h3>Description</h3>

<p>Performs a GET request on the specified <code>resource_url</code> and all its
pages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>request_all(
  resource_url = "/",
  base_url = pgs_server(),
  limit = 20L,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="request_all_+3A_resource_url">resource_url</code></td>
<td>
<p>Endpoint URL. The endpoint is internally appended to the
<code>base_url</code>. It should start with a forward slash (<code>/</code>).</p>
</td></tr>
<tr><td><code id="request_all_+3A_base_url">base_url</code></td>
<td>
<p>The PGS REST API base URL (one should not need to change its
default value).</p>
</td></tr>
<tr><td><code id="request_all_+3A_limit">limit</code></td>
<td>
<p>number of results per page.</p>
</td></tr>
<tr><td><code id="request_all_+3A_verbose">verbose</code></td>
<td>
<p>whether to print information about each API request.</p>
</td></tr>
<tr><td><code id="request_all_+3A_warnings">warnings</code></td>
<td>
<p>whether to print warnings related to API requests.</p>
</td></tr>
<tr><td><code id="request_all_+3A_progress_bar">progress_bar</code></td>
<td>
<p>whether to show a progress bar as the paginated resources
are retrieved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list four named elements:
</p>

<dl>
<dt>resource</dt><dd><p>The URL endpoint.</p>
</dd>
<dt>code</dt><dd><p><a href="https://en.wikipedia.org/wiki/List_of_HTTP_status_codes">HTTP
status code</a>.</p>
</dd>
<dt>message</dt><dd><p>A string describing the status of the response obtained. It is
&quot;OK&quot; if everything went OK or some other string describing the problem.</p>
</dd>
<dt>json</dt><dd><p>A list of JSON responses (each response is a string).</p>
</dd>
</dl>


<hr>
<h2 id='request_warning'>Pretty printing of http error</h2><span id='topic+request_warning'></span>

<h3>Description</h3>

<p>Pretty printing of http error
</p>


<h3>Usage</h3>

<pre><code class='language-R'>request_warning(response, warnings = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="request_warning_+3A_response">response</code></td>
<td>
<p>A <code><a href="httr.html#topic+response">response</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect: printing.
</p>

<hr>
<h2 id='s4_to_list'>Convert an S4 object into a list</h2><span id='topic+s4_to_list'></span>

<h3>Description</h3>

<p>Convert an S4 object into a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s4_to_list(s4_obj)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s4_to_list_+3A_s4_obj">s4_obj</code></td>
<td>
<p>an S4 object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list version of the S4 object.
</p>

<hr>
<h2 id='sample_sets'>Constructor for the S4 sample_sets object.</h2><span id='topic+sample_sets'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+sample_sets-class">sample_sets</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_sets(
  sample_sets = s4pss_sample_sets_tbl(),
  samples = s4pss_samples_tbl(),
  demographics = s4pss_demographics_tbl(),
  cohorts = s4pss_pgs_cohorts_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_sets_+3A_sample_sets">sample_sets</code></td>
<td>
<p>A <code>s4pss_sample_sets_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="sample_sets_+3A_samples">samples</code></td>
<td>
<p>A <code>s4pss_samples_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="sample_sets_+3A_demographics">demographics</code></td>
<td>
<p>A <code>s4pss_demographics_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="sample_sets_+3A_cohorts">cohorts</code></td>
<td>
<p>A <code>s4pss_pgs_cohorts_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+sample_sets-class">sample_sets</a>.
</p>

<hr>
<h2 id='sample_sets-class'>An S4 class to represent a set of PGS Catalog Sample Sets</h2><span id='topic+sample_sets-class'></span>

<h3>Description</h3>

<p>The sample_sets object consists of four tables (slots) that combined form a
relational database of a subset of PGS Catalog sample sets. Each sample set
is an observation (row) in the <code>sample_sets</code> table (first table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>sample_sets</code></dt><dd><p>A table of sample sets. Each sample set (row) is uniquely
identified by the column <code>pss_id</code>. Columns:
</p>

<dl>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>samples</code></dt><dd><p>A table of samples. Each sample (row) is uniquely identified by
the combination of values from the columns: <code>pss_id</code> and
<code>sample_id</code>. Columns:
</p>

<dl>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: should be always Evaluation (<code>"eval"</code>).</p>
</dd>
<dt>sample_size</dt><dd><p>Number of individuals included in the sample.</p>
</dd>
<dt>sample_cases</dt><dd><p>Number of cases.</p>
</dd>
<dt>sample_controls</dt><dd><p>Number of controls.</p>
</dd>
<dt>sample_percent_male</dt><dd><p>Percentage of male participants.</p>
</dd>
<dt>phenotype_description</dt><dd><p>Detailed phenotype description.</p>
</dd>
<dt>ancestry_category</dt><dd><p>Author reported ancestry is mapped to the best matching
ancestry category from the NHGRI-EBI GWAS Catalog framework (see
<code><a href="#topic+ancestry_categories">ancestry_categories</a></code>) for possible values.</p>
</dd>
<dt>ancestry</dt><dd><p>A more detailed description of sample ancestry
that usually matches the most specific description described by the authors
(e.g. French, Chinese).</p>
</dd>
<dt>country</dt><dd><p>Author reported countries of recruitment (if available).</p>
</dd>
<dt>ancestry_additional_description</dt><dd><p>Any additional description not
captured in the other columns (e.g. founder or genetically isolated
populations, or further description of admixed samples).</p>
</dd>
<dt>study_id</dt><dd><p>Associated GWAS Catalog study accession identifier, e.g.,
<code>"GCST002735"</code>.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier.</p>
</dd>
<dt>cohorts_additional_description</dt><dd><p>Any additional description about the
samples (e.g. sub-cohort information).</p>
</dd>
</dl>
</dd>
<dt><code>demographics</code></dt><dd><p>A table of sample demographics' variables. Each
demographics' variable (row) is uniquely identified by the combination of
values from the columns: <code>pss_id</code>, <code>sample_id</code>, and
<code>variable</code>. Columns:
</p>

<dl>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate identifier to
identify each sample.</p>
</dd>
<dt>variable</dt><dd><p>Demographics variable. Following columns report about the
indicated variable.</p>
</dd>
<dt>estimate_type</dt><dd><p>Type of statistical estimate for variable.</p>
</dd>
<dt>estimate</dt><dd><p>The variable's statistical value.</p>
</dd>
<dt>unit</dt><dd><p>Unit of the variable.</p>
</dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p>
</dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p>
</dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p>
</dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p>
</dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p>
</dd>
</dl>
</dd>
<dt><code>cohorts</code></dt><dd><p>A table of cohorts. Each cohort (row) is uniquely identified by
the combination of values from the columns: <code>pss_id</code>, <code>sample_id</code>
and <code>cohort_symbol</code>. Columns:
</p>

<dl>
<dt>pss_id</dt><dd><p>A PGS Sample Set identifier. Example: <code>"PSS000042"</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p>
</dd>
<dt>cohort_symbol</dt><dd><p>Cohort symbol.</p>
</dd>
<dt>cohort_name</dt><dd><p>Cohort full name.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='scores'>Constructor for the S4 scores object.</h2><span id='topic+scores'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+scores-class">scores</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scores(
  scores = s4scores_scores_tbl(),
  publications = s4scores_publications_tbl(),
  samples = s4scores_samples_tbl(),
  demographics = s4scores_demographics_tbl(),
  cohorts = s4scores_cohorts_tbl(),
  traits = s4scores_traits_tbl(),
  stages_tally = s4scores_stages_tally_tbl(),
  ancestry_frequencies = s4scores_ancestry_frequencies_tbl(),
  multi_ancestry_composition = s4scores_multi_ancestry_composition_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scores_+3A_scores">scores</code></td>
<td>
<p>A <code>s4scores_scores_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="scores_+3A_publications">publications</code></td>
<td>
<p>A <code>s4scores_publications_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="scores_+3A_samples">samples</code></td>
<td>
<p>A <code>s4scores_samples_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="scores_+3A_demographics">demographics</code></td>
<td>
<p>A <code>s4scores_demographics_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="scores_+3A_cohorts">cohorts</code></td>
<td>
<p>A <code>s4scores_cohorts_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="scores_+3A_traits">traits</code></td>
<td>
<p>A <code>s4scores_traits_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+scores-class">scores</a>.
</p>

<hr>
<h2 id='scores-class'>An S4 class to represent a set of PGS Catalog Polygenic Scores</h2><span id='topic+scores-class'></span>

<h3>Description</h3>

<p>The scores object consists of six tables (slots) that combined form a
relational database of a subset of PGS Catalog polygenic scores. Each score
is an observation (row) in the <code>scores</code> table (the first table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>scores</code></dt><dd><p>A table of polygenic scores. Each polygenic score (row) is
uniquely identified by the <code>pgs_id</code> column. Columns:
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier. Example: <code>"PGS000001"</code>.</p>
</dd>
<dt>pgs_name</dt><dd><p>This may be the name that the authors describe the PGS with
in the source publication, or a name that a curator of the PGS Catalog has
assigned to identify the score during the curation process (before a PGS
identifier has been given). Example: <code>PRS77_BC</code>.</p>
</dd>
<dt>scoring_file</dt><dd><p>URL to the scoring file on the PGS FTP server. Example:
<code>"http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz"</code>.</p>
</dd>
<dt>matches_publication</dt><dd><p>Indicate if the PGS data matches the published
polygenic score (<code>TRUE</code>). If not (<code>FALSE</code>), the authors have
provided an alternative polygenic for the Catalog and some other data, such
as performance metrics, may differ from the publication.</p>
</dd>
<dt>reported_trait</dt><dd><p>The author-reported trait that the PGS has been
developed to predict. Example: <code>"Breast Cancer"</code>.</p>
</dd>
<dt>trait_additional_description</dt><dd><p>Any additional description not captured
in the other columns. Example: <code>"Femoral neck BMD (g/cm2)"</code>.</p>
</dd>
<dt>pgs_method_name</dt><dd><p>The name or description of the method or computational
algorithm used to develop the PGS.</p>
</dd>
<dt>pgs_method_params</dt><dd><p>A description of the relevant inputs and parameters
relevant to the PGS development method/process.</p>
</dd>
<dt>n_variants</dt><dd><p>Number of variants used to calculate the PGS.</p>
</dd>
<dt>n_variants_interactions</dt><dd><p>Number of higher-order variant interactions
included in the PGS.</p>
</dd>
<dt>assembly</dt><dd><p>The version of the genome assembly that the variants present
in the PGS are associated with. Example: <code>GRCh37</code>.</p>
</dd>
<dt>license</dt><dd><p>The PGS Catalog distributes its data according to EBI's
standard Terms of Use. Some PGS have specific terms, licenses, or
restrictions (e.g. non-commercial use) that we highlight in this field, if
known.</p>
</dd>
</dl>
</dd>
<dt><code>publications</code></dt><dd><p>A table of publications. Each publication (row) is
uniquely identified by the <code>pgp_id</code> column. Columns:
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>pgp_id</dt><dd><p>PGS Publication identifier. Example: <code>"PGP000001"</code>.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier. Example: <code>"25855707"</code>.</p>
</dd>
<dt>publication_date</dt><dd><p>Publication date. Example: <code>"2020-09-28"</code>. Note
that the class of <code>publication_date</code> is <code><a href="base.html#topic+Date">Date</a></code>.</p>
</dd>
<dt>publication</dt><dd><p>Abbreviated name of the journal. Example: <code>"Am J Hum
Genet"</code>.</p>
</dd>
<dt>title</dt><dd><p>Publication title.</p>
</dd>
<dt>author_fullname</dt><dd><p>First author of the publication. Example:
<code>'Mavaddat N'</code>.</p>
</dd>
<dt>doi</dt><dd><p>Digital Object Identifier (DOI). This variable is also curated to
allow unpublished work (e.g. preprints) to be added to the catalog. Example:
<code>"10.1093/jnci/djv036"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>samples</code></dt><dd><p>A table of samples. Each sample (row) is uniquely identified by
the combination of values from the columns: <code>pgs_id</code> and
<code>sample_id</code>. Columns:
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic score identifier. An identifier that starts with
<code>'PGS'</code> and is followed by six digits, e.g. <code>'PGS000001'</code>.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: either <code>"discovery"</code> or <code>"training"</code>.</p>
</dd>
<dt>sample_size</dt><dd><p>Number of individuals included in the sample.</p>
</dd>
<dt>sample_cases</dt><dd><p>Number of cases.</p>
</dd>
<dt>sample_controls</dt><dd><p>Number of controls.</p>
</dd>
<dt>sample_percent_male</dt><dd><p>Percentage of male participants.</p>
</dd>
<dt>phenotype_description</dt><dd><p>Detailed phenotype description.</p>
</dd>
<dt>ancestry_category</dt><dd><p>Author reported ancestry is mapped to the best matching
ancestry category from the NHGRI-EBI GWAS Catalog framework (see
<code><a href="#topic+ancestry_categories">ancestry_categories</a></code>) for possible values.</p>
</dd>
<dt>ancestry</dt><dd><p>A more detailed description of sample ancestry
that usually matches the most specific description described by the authors
(e.g. French, Chinese).</p>
</dd>
<dt>country</dt><dd><p>Author reported countries of recruitment (if available).</p>
</dd>
<dt>ancestry_additional_description</dt><dd><p>Any additional description not
captured in the other columns (e.g. founder or genetically isolated
populations, or further description of admixed samples).</p>
</dd>
<dt>study_id</dt><dd><p>Associated GWAS Catalog study accession identifier, e.g.,
<code>"GCST002735"</code>.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier.</p>
</dd>
<dt>cohorts_additional_description</dt><dd><p>Any additional description about the
samples (e.g. sub-cohort information).</p>
</dd>
</dl>
</dd>
<dt><code>demographics</code></dt><dd><p>A table of sample demographics' variables. Each
demographics' variable (row) is uniquely identified by the combination of
values from the columns: <code>pgs_id</code>, <code>sample_id</code> and
<code>variable</code>. Columns:
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate identifier to
identify each sample.</p>
</dd>
<dt>variable</dt><dd><p>Demographics variable. Following columns report about the
indicated variable.</p>
</dd>
<dt>estimate_type</dt><dd><p>Type of statistical estimate for variable.</p>
</dd>
<dt>estimate</dt><dd><p>The variable's statistical value.</p>
</dd>
<dt>unit</dt><dd><p>Unit of the variable.</p>
</dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p>
</dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p>
</dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p>
</dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p>
</dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p>
</dd>
</dl>
</dd>
<dt><code>cohorts</code></dt><dd><p>A table of cohorts. Each cohort (row) is uniquely identified by
the combination of values from the columns: <code>pgs_id</code>, <code>sample_id</code>
and <code>cohort_symbol</code>. Columns:
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p>
</dd>
<dt>cohort_symbol</dt><dd><p>Cohort symbol.</p>
</dd>
<dt>cohort_name</dt><dd><p>Cohort full name.</p>
</dd>
</dl>
</dd>
<dt><code>traits</code></dt><dd><p>A table of EFO traits. Each trait (row) is uniquely identified
by the combination of the columns <code>pgs_id</code> and <code>efo_id</code>. Columns:
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>trait</dt><dd><p>Trait name.</p>
</dd>
<dt>description</dt><dd><p>Detailed description of the trait from EFO.</p>
</dd>
<dt>url</dt><dd><p>External link to the EFO entry.</p>
</dd>
</dl>
</dd>
<dt><code>stages_tally</code></dt><dd><p>A table of sample sizes and number of samples sets at each stage.
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: either <code>"gwas"</code>, <code>"dev"</code> or <code>"eval"</code>.</p>
</dd>
<dt>sample_size</dt><dd><p>Sample size.</p>
</dd>
<dt>n_sample_sets</dt><dd><p>Number of sample sets (only meaningful for the evaluation stage <code>"eval"</code>)</p>
</dd>
</dl>
</dd>
<dt><code>ancestry_frequencies</code></dt><dd><p>This table describes the ancestry composition at each stage.
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: either <code>"gwas"</code>, <code>"dev"</code> or <code>"eval"</code>.</p>
</dd>
<dt>ancestry_class_symbol</dt><dd><p>Ancestry class symbol.</p>
</dd>
<dt>frequency</dt><dd><p>Ancestry fraction (percentage).</p>
</dd>
</dl>
</dd>
<dt><code>multi_ancestry_composition</code></dt><dd><p>A table of a breakdown of the ancestries included in multi-ancestries.
</p>

<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
<dt>stage</dt><dd><p>Sample stage: either <code>"gwas"</code>, <code>"dev"</code> or <code>"eval"</code>.</p>
</dd>
<dt>multi_ancestry_class_symbol</dt><dd><p>Multi-ancestry class symbol.</p>
</dd>
<dt>ancestry_class_symbol</dt><dd><p>Ancestry class symbol.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='setop'>Set operations on PGS Catalog objects</h2><span id='topic+setop'></span><span id='topic+union'></span><span id='topic+intersect'></span><span id='topic+setdiff'></span><span id='topic+setequal'></span>

<h3>Description</h3>

<p>Performs set union, intersection, and (asymmetric!) difference on two objects
of either class <a href="#topic+scores-class">scores</a>, <a href="#topic+publications-class">publications</a>, <a href="#topic+traits-class">traits</a>,
<a href="#topic+performance_metrics-class">performance_metrics</a>, <a href="#topic+sample_sets-class">sample_sets</a>, <a href="#topic+cohorts-class">cohorts</a> or
<a href="#topic+trait_categories-class">trait_categories</a>. Note that <code>union()</code> removes duplicated
entities, whereas <code><a href="#topic+bind">bind()</a></code> does not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>union(x, y, ...)

intersect(x, y, ...)

setdiff(x, y, ...)

setequal(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setop_+3A_x">x</code>, <code id="setop_+3A_y">y</code></td>
<td>
<p>Objects of either class <a href="#topic+scores-class">scores</a>, <a href="#topic+publications-class">publications</a>,
<a href="#topic+traits-class">traits</a>, <a href="#topic+performance_metrics-class">performance_metrics</a>, <a href="#topic+sample_sets-class">sample_sets</a>,
<a href="#topic+cohorts-class">cohorts</a> or <a href="#topic+trait_categories-class">trait_categories</a>.</p>
</td></tr>
<tr><td><code id="setop_+3A_...">...</code></td>
<td>
<p>other arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>In the case of <code>union()</code>, <code>intersect()</code>, or <code>setdiff()</code>: an object of
the same class as <code>x</code> and <code>y</code>. In the case of <code>setequal()</code>, a
logical scalar.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get some `scores` objects:
my_scores_1 &lt;- get_scores(c('PGS000012', 'PGS000013'))
my_scores_2 &lt;- get_scores(c('PGS000013', 'PGS000014'))

#
# union()
#
# NB: with `union()`, PGS000013 is not repeated.
union(my_scores_1, my_scores_2)@scores

#
# intersect()
#
intersect(my_scores_1, my_scores_2)@scores

#
# setdiff()
#
setdiff(my_scores_1, my_scores_2)@scores

#
# setequal()
#
setequal(my_scores_1, my_scores_2)
setequal(my_scores_1, my_scores_1)
setequal(my_scores_2, my_scores_2)

</code></pre>

<hr>
<h2 id='stages'>Study stages</h2><span id='topic+stages'></span>

<h3>Description</h3>

<p>A dataset containing the various study stages assigned to samples in the PGS Catalog.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stages
</code></pre>


<h3>Format</h3>

<p>A data frame with 5 stages (rows) and 4 columns:
</p>

<dl>
<dt>stage</dt><dd><p>Study stage.</p>
</dd>
<dt>symbol</dt><dd><p>One-letter symbol for the stage, or a comma separated combination thereof.</p>
</dd>
<dt>name</dt><dd><p>Stage name.</p>
</dd>
<dt>definition</dt><dd><p>Stage description.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.pgscatalog.org/docs/ancestry">https://www.pgscatalog.org/docs/ancestry</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stages

</code></pre>

<hr>
<h2 id='study_to_pgs'>Map GWAS studies identifiers to PGS identifiers</h2><span id='topic+study_to_pgs'></span>

<h3>Description</h3>

<p>Map GWAS studies identifiers to PGS identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>study_to_pgs(study_id, verbose = FALSE, warnings = TRUE, progress_bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="study_to_pgs_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of GWAS Catalog study accession
identifiers, e.g., &quot;GCST001937&quot;.</p>
</td></tr>
<tr><td><code id="study_to_pgs_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="study_to_pgs_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if
any.</p>
</td></tr>
<tr><td><code id="study_to_pgs_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the queries are
performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns: <code>study_id</code> and <code>pgs_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
study_to_pgs('GCST001937')
study_to_pgs(c('GCST000998', 'GCST000338'))

## End(Not run)
</code></pre>

<hr>
<h2 id='subset-cohorts'>Subset a cohorts object</h2><span id='topic+subset-cohorts'></span><span id='topic++5B+2Ccohorts+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ccohorts+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ccohorts+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+cohorts-class">cohorts</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'cohorts,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'cohorts,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'cohorts,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-cohorts_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+cohorts-class">cohorts</a> object.</p>
</td></tr>
<tr><td><code id="subset-cohorts_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-cohorts_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-cohorts_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-cohorts_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+cohorts-class">cohorts</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get a few cohorts by their symbol:
my_cohorts &lt;- get_cohorts(c('23andMe', 'BioImage', 'Rotterdam-SI', 'SGWAS'),
                progress_bar = FALSE)

#
# Subsetting by position
#
my_cohorts[c(1, 3)]

#
# Subsetting by cohort symbol (character)
#
my_cohorts[c('23andMe', 'SGWAS')]

</code></pre>

<hr>
<h2 id='subset-performance_metrics'>Subset a performance_metrics object</h2><span id='topic+subset-performance_metrics'></span><span id='topic++5B+2Cperformance_metrics+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cperformance_metrics+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cperformance_metrics+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+performance_metrics-class">performance_metrics</a> by identifier or by position
using the <code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'performance_metrics,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'performance_metrics,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'performance_metrics,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-performance_metrics_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+performance_metrics-class">performance_metrics</a> object.</p>
</td></tr>
<tr><td><code id="subset-performance_metrics_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-performance_metrics_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-performance_metrics_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-performance_metrics_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+performance_metrics-class">performance_metrics</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get a few performance metrics:
my_ppm &lt;- get_performance_metrics(sprintf('PPM%06d', 38:42))

#
# Subsetting by position
#
my_ppm[c(1, 4)]

#
# Subsetting by performance metrics identifier (character)
#
my_ppm['PPM000042']

</code></pre>

<hr>
<h2 id='subset-publications'>Subset a publications object</h2><span id='topic+subset-publications'></span><span id='topic++5B+2Cpublications+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cpublications+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cpublications+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+publications-class">publications</a> by identifier or by position using
the <code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'publications,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'publications,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'publications,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-publications_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+publications-class">publications</a> object.</p>
</td></tr>
<tr><td><code id="subset-publications_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-publications_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-publications_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-publications_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+publications-class">publications</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get all publications in the PGS Catalog:
all_pub &lt;- get_publications(interactive = FALSE, progress_bar = FALSE)

#
# Subsetting by position
#
all_pub[1:5]

#
# Subsetting by publication identifier (character)
#
all_pub['PGP000001']

</code></pre>

<hr>
<h2 id='subset-releases'>Subset a releases object</h2><span id='topic+subset-releases'></span><span id='topic++5B+2Creleases+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Creleases+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Creleases+2Ccharacter+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Creleases+2CDate+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+releases-class">releases</a> by identifier (release date) or by
position using the <code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'releases,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'releases,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'releases,character,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'releases,Date,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-releases_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+releases-class">releases</a> object.</p>
</td></tr>
<tr><td><code id="subset-releases_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-releases_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-releases_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-releases_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+releases-class">releases</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get details about all PGS Catalog data releases thus far:
all_releases &lt;- get_releases(date = 'all', progress_bar = FALSE)

#
# Subsetting by position
#
# Releases are, by default, sorted by date in descending order, thus the
# first PGS Catalog release is in the last position of the returned
# `all_releases` object. Here's how you can extract that first release (last
# position in `all_releases`):
all_releases[n(all_releases)]

#
# Subsetting by date (character)
#
date_of_interest &lt;- '2021-06-11'
class(date_of_interest)
all_releases[date_of_interest]

#
# Subsetting by date (Date object)
#
date_of_interest &lt;- as.Date('2021-06-11')
class(date_of_interest)
all_releases[date_of_interest]

</code></pre>

<hr>
<h2 id='subset-sample_sets'>Subset a sample_sets object</h2><span id='topic+subset-sample_sets'></span><span id='topic++5B+2Csample_sets+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Csample_sets+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Csample_sets+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+sample_sets-class">sample_sets</a> by identifier or by position using
the <code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'sample_sets,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'sample_sets,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'sample_sets,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-sample_sets_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+sample_sets-class">sample_sets</a> object.</p>
</td></tr>
<tr><td><code id="subset-sample_sets_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-sample_sets_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-sample_sets_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-sample_sets_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+sample_sets-class">sample_sets</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get a few sample sets:
my_pss &lt;- get_sample_sets(sprintf('PSS%06d', 42:48))

#
# Subsetting by position
#
my_pss[c(1, 3)]

#
# Subsetting by sample set identifier (character)
#
my_pss['PSS000042']

</code></pre>

<hr>
<h2 id='subset-scores'>Subset a scores object</h2><span id='topic+subset-scores'></span><span id='topic++5B+2Cscores+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cscores+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cscores+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+scores-class">scores</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'scores,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'scores,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'scores,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-scores_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+scores-class">scores</a> object.</p>
</td></tr>
<tr><td><code id="subset-scores_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-scores_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-scores_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-scores_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+scores-class">scores</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get a few polygenic scores:
my_scores &lt;- get_scores(sprintf('PGS%06d', 10:14), progress_bar = FALSE)

#
# Subsetting by position
#
my_scores[c(1, 3, 5)]@scores

#
# Subsetting by PGS identifier (character)
#
my_scores[c('PGS000011', 'PGS000014')]@scores

</code></pre>

<hr>
<h2 id='subset-trait_categories'>Subset a trait_categories object</h2><span id='topic+subset-trait_categories'></span><span id='topic++5B+2Ctrait_categories+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ctrait_categories+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ctrait_categories+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+trait_categories-class">trait_categories</a> by trait category (string) or
by position using the <code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'trait_categories,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'trait_categories,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'trait_categories,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-trait_categories_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+trait_categories-class">trait_categories</a> object.</p>
</td></tr>
<tr><td><code id="subset-trait_categories_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-trait_categories_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-trait_categories_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-trait_categories_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+trait_categories-class">trait_categories</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get details about all trait categories:
all_trait_categories &lt;- get_trait_categories(progress_bar = FALSE)

#
# Subsetting by position
#
all_trait_categories[1:5]

#
# Subsetting by trait category (character)
#
all_trait_categories['Liver enzyme measurement']

</code></pre>

<hr>
<h2 id='subset-traits'>Subset a traits object</h2><span id='topic+subset-traits'></span><span id='topic++5B+2Ctraits+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ctraits+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ctraits+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+traits-class">traits</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'traits,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'traits,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'traits,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset-traits_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+traits-class">traits</a> object.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get a few traits:
my_traits &lt;- get_traits(trait_term = 'stroke', exact_term = FALSE,
               progress_bar = FALSE)

#
# Subsetting by position
#
my_traits[1]

#
# Subsetting by EFO trait identifier (character)
#
my_traits['EFO_0000712']

</code></pre>

<hr>
<h2 id='sure'>Are you sure?</h2><span id='topic+sure'></span>

<h3>Description</h3>

<p>This function asks you interactively for permission to continue or not. You
can specify a custom message before the question and also different messages
for a both a positive and negative answer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sure(
  before_question = NULL,
  after_saying_no = NULL,
  after_saying_yes = NULL,
  default_answer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sure_+3A_before_question">before_question</code></td>
<td>
<p>String with message to be printed before question.</p>
</td></tr>
<tr><td><code id="sure_+3A_after_saying_no">after_saying_no</code></td>
<td>
<p>String with message to be printed after answering
<code>'no'</code>.</p>
</td></tr>
<tr><td><code id="sure_+3A_after_saying_yes">after_saying_yes</code></td>
<td>
<p>String with message to be printed after answering
<code>'yes'</code>.</p>
</td></tr>
<tr><td><code id="sure_+3A_default_answer">default_answer</code></td>
<td>
<p>String with answer to question, if run in
non-interactive mode.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you run this function in non-interactive mode, you should pass an
automatic answer to <code>default_answer</code>: <code>'yes'</code> or <code>'no'</code>.
</p>


<h3>Value</h3>

<p>A logical indicating if answer was <code>'yes'</code>/<code>'y'</code>
(<code>TRUE</code>) or otherwise (<code>FALSE</code>).
</p>

<hr>
<h2 id='trait_categories'>Constructor for the S4 trait_categories object.</h2><span id='topic+trait_categories'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+trait_categories-class">trait_categories</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trait_categories(
  trait_categories = s4trait_categories_trait_categories_tbl(),
  traits = s4trait_categories_traits_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trait_categories_+3A_trait_categories">trait_categories</code></td>
<td>
<p>A <code>s4trait_categories_trait_categories_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="trait_categories_+3A_traits">traits</code></td>
<td>
<p>A <code>s4trait_categories_traits_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+trait_categories-class">trait_categories</a>.
</p>

<hr>
<h2 id='trait_categories-class'>An S4 class to represent a set of PGS Catalog Trait Categories</h2><span id='topic+trait_categories-class'></span>

<h3>Description</h3>

<p>The trait_categories object consists of two tables (slots) that combined form
a relational database of a subset of PGS Catalog trait categories. Each score
is an observation (row) in the <code>trait_categories</code> table (first table).
</p>


<h3>Slots</h3>


<dl>
<dt><code>trait_categories</code></dt><dd><p>A table of trait categories. Columns:
</p>

<dl>
<dt>trait_category</dt><dd><p>Trait category name.</p>
</dd>
</dl>
</dd>
<dt><code>traits</code></dt><dd><p>A table of associated traits. Columns:
</p>

<dl>
<dt>trait_category</dt><dd><p>Trait category name.</p>
</dd>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>trait</dt><dd><p>Trait name.</p>
</dd>
<dt>description</dt><dd><p>Detailed description of the trait from EFO.</p>
</dd>
<dt>url</dt><dd><p>External link to the EFO entry.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='traits'>Constructor for the S4 traits object.</h2><span id='topic+traits'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+traits-class">traits</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traits(
  traits = s4traits_traits_tbl(),
  pgs_ids = s4traits_pgs_ids_tbl(),
  child_pgs_ids = s4traits_child_pgs_ids_tbl(),
  trait_categories = s4traits_trait_categories_tbl(),
  trait_synonyms = s4traits_trait_synonyms_tbl(),
  trait_mapped_terms = s4traits_trait_mapped_terms_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="traits_+3A_traits">traits</code></td>
<td>
<p>A <code>s4traits_traits_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="traits_+3A_pgs_ids">pgs_ids</code></td>
<td>
<p>A <code>s4traits_pgs_ids_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="traits_+3A_child_pgs_ids">child_pgs_ids</code></td>
<td>
<p>A <code>s4traits_child_pgs_ids_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="traits_+3A_trait_categories">trait_categories</code></td>
<td>
<p>A <code>s4traits_trait_categories_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="traits_+3A_trait_synonyms">trait_synonyms</code></td>
<td>
<p>A <code>s4traits_trait_synonyms_tbl</code> tibble.</p>
</td></tr>
<tr><td><code id="traits_+3A_trait_mapped_terms">trait_mapped_terms</code></td>
<td>
<p>A <code>s4traits_trait_mapped_terms_tbl</code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+traits-class">traits</a>.
</p>

<hr>
<h2 id='traits-class'>An S4 class to represent a set of PGS Catalog Traits</h2><span id='topic+traits-class'></span>

<h3>Description</h3>

<p>The traits object consists of six slots, each a table
(<code><a href="tibble.html#topic+tibble">tibble</a></code>), that combined form a relational database of a
subset of PGS Catalog traits. Each trait is an observation (row) in
the <code>traits</code> table &mdash; main table. All tables have the column
<code>efo_id</code> as primary key.
</p>


<h3>Slots</h3>


<dl>
<dt><code>traits</code></dt><dd><p>A table of traits. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>parent_efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier of
the parent trait.</p>
</dd>
<dt>is_child</dt><dd><p>Is this trait obtained because it is a child of other trait?</p>
</dd>
<dt>trait</dt><dd><p>Trait name.</p>
</dd>
<dt>description</dt><dd><p>Detailed description of the trait from EFO.</p>
</dd>
<dt>url</dt><dd><p>External link to the EFO entry.</p>
</dd>
</dl>
</dd>
<dt><code>pgs_ids</code></dt><dd><p>A table of associated polygenic score identifiers. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>parent_efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier of
the parent trait.</p>
</dd>
<dt>is_child</dt><dd><p>Is this trait obtained because it is a child of other trait?</p>
</dd>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p>
</dd>
</dl>
</dd>
<dt><code>child_pgs_ids</code></dt><dd><p>A table of polygenic score identifiers associated with
the child traits. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>parent_efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier of
the parent trait.</p>
</dd>
<dt>is_child</dt><dd><p>Is this trait obtained because it is a child of other trait?</p>
</dd>
<dt>child_pgs_id</dt><dd><p>Polygenic Score (PGS) identifiers associated with child traits.</p>
</dd>
</dl>
</dd>
<dt><code>trait_categories</code></dt><dd><p>A table of associated trait categories. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>parent_efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier of
the parent trait.</p>
</dd>
<dt>is_child</dt><dd><p>Is this trait obtained because it is a child of other trait?</p>
</dd>
<dt>trait_category</dt><dd><p>Trait category name.</p>
</dd>
</dl>
</dd>
<dt><code>trait_synonyms</code></dt><dd><p>A table of associated trait synonyms. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>parent_efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier of
the parent trait.</p>
</dd>
<dt>is_child</dt><dd><p>Is this trait obtained because it is a child of other trait?</p>
</dd>
<dt>trait_synonyms</dt><dd><p>Trait synonyms.</p>
</dd>
</dl>
</dd>
<dt><code>trait_mapped_terms</code></dt><dd><p>A table of associated external references,
identifiers or other terms. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>parent_efo_id</dt><dd><p>An <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier of
the parent trait.</p>
</dd>
<dt>is_child</dt><dd><p>Is this trait obtained because it is a child of other trait?</p>
</dd>
<dt>trait_mapped_terms</dt><dd><p>Trait mapped terms.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='user_agent_id'>User agent identification</h2><span id='topic+user_agent_id'></span>

<h3>Description</h3>

<p>Generates an S3 <code>request</code> object as defined by the package <code>httr</code>,
that is used to identify this package as the user agent in requests to the
PGS REST API. The user agent identification string is: <code>"quincunx: R
Client for the PGS REST API"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>user_agent_id()
</code></pre>


<h3>Value</h3>

<p>An S3 <code>request</code> object as defined by the package <code>httr</code>.
</p>

<hr>
<h2 id='write_xlsx'>Export a PGS Catalog object to xlsx</h2><span id='topic+write_xlsx'></span>

<h3>Description</h3>

<p>This function exports a PGS Catalog object to Microsoft Excel xlsx file. Each
table (slot) is saved in its own sheet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_xlsx(x, file = stop("`file` must be specified"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_xlsx_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+scores-class">scores</a>, <a href="#topic+publications-class">publications</a>,
<a href="#topic+traits-class">traits</a>, <a href="#topic+performance_metrics-class">performance_metrics</a>,
<a href="#topic+sample_sets-class">sample_sets</a>, <a href="#topic+cohorts-class">cohorts</a>,
<a href="#topic+trait_categories-class">trait_categories</a> or <a href="#topic+releases-class">releases</a> object.</p>
</td></tr>
<tr><td><code id="write_xlsx_+3A_file">file</code></td>
<td>
<p>A file name to write to.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for its side effect.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
