<!DOCTYPE html><html><head><title>Help for package r.jive</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {r.jive}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#r.jive-package'>
<p>Perform JIVE Decompositions for Multi-Source Data</p></a></li>
<li><a href='#BRCA_Data'><p>BRCA TCGA Dataset</p></a></li>
<li><a href='#jive'>
<p>JIVE Decomposition for Multi-source Data</p></a></li>
<li><a href='#jive.predict'>
<p>Predict JIVE scores for new data</p></a></li>
<li><a href='#pjsum'>
<p>Calculate Number of Free Parameters for BIC Calculation</p></a></li>
<li><a href='#plot.jive'>
<p>Create Plots for a JIVE Object</p></a></li>
<li><a href='#show.image'>
<p>Draw a Heatmap from a Matrix</p></a></li>
<li><a href='#showHeatmaps'>
<p>Heatmaps for JIVE Decompositions</p></a></li>
<li><a href='#showPCA'>
<p>Principal Component Plots for JIVE Decomposition</p></a></li>
<li><a href='#showVarExplained'>
<p>Display Variance Explained</p></a></li>
<li><a href='#SimData'><p>Simulated Dataset</p></a></li>
<li><a href='#SimResults'><p>JIVE Results for Simulated Dataset</p></a></li>
<li><a href='#summary.jive'>
<p>Summarize a JIVE Decomposition</p></a></li>
<li><a href='#svdmiss'>
<p>Missing Data SVD</p></a></li>
<li><a href='#svdwrapper'>
<p>Wrapper Function to Perform SVD</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Perform JIVE Decomposition for Multi-Source Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-11-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael J. O'Connell [aut, cre],
  Eric F. Lock [aut],
  Adam Kaplan [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael J. O'Connell &lt;oconnemj@miamioh.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs the Joint and Individual Variation Explained (JIVE) decomposition on a list of data sets when the data share a dimension, returning low-rank matrices that capture the joint and individual structure of the data [O'Connell, MJ and Lock, EF (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw324">doi:10.1093/bioinformatics/btw324</a>&gt;]. It provides two methods of rank selection when the rank is unknown, a permutation test and a Bayesian Information Criterion (BIC) selection algorithm. Also included in the package are three plotting functions for visualizing the variance attributed to each data source: a bar plot that shows the percentages of the variability attributable to joint and individual structure, a heatmap that shows the structure of the variability, and principal component plots. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>gplots, abind, graphics, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-12 02:50:38 UTC; Eric</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-17 08:10:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='r.jive-package'>
Perform JIVE Decompositions for Multi-Source Data
</h2><span id='topic+r.jive-package'></span><span id='topic+r.jive'></span>

<h3>Description</h3>

<p>Performs the Joint and Individual Variation Explained (JIVE) decompositions on a list of data sets when the data share a dimension, returning low-rank matrices that capture the joint and individual structure of the data. It provides two methods of rank selection when the rank is unknown, a permutation test and a Bayesian Information Criterion (BIC) selection algorithm. Also included in the package are three plotting functions for visualizing the variance attributed to each data source: a bar plot that shows the percentages of the variability attributable to joint and individual structure, a heatmap that shows the structure of the variability, and principal component plots.  
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> r.jive</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.4</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-11-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>
<p>Maintainer: Michael J. O'Connell &lt;oconnemj@miamioh.edu&gt;
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 32(18):2877-2879, 2016.</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(10)
##Load data that were simulated as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimData) 
# Using default method ("perm")
Results &lt;- jive(SimData)
summary(Results)

# Using BIC rank selection
BIC_result &lt;- jive(SimData, method="bic")
summary(BIC_result)

###Load the permutation results
data(SimResults) 
# Visualize results
showVarExplained(Results)
# showVarExplained is also called by the "jive" S3 class default plot method

#show heatmaps
showHeatmaps(Results)

#show PCA plots
showPCA(Results,1,c(1,1))

</code></pre>

<hr>
<h2 id='BRCA_Data'>BRCA TCGA Dataset</h2><span id='topic+Data'></span><span id='topic+clusts'></span>

<h3>Description</h3>

<p>These data were obtained from the data freeze for The Cancer Genome Atlas flagship BRCA publication (Cancer Genome Atlas Network, 2013), and processed as described in Lock and Dunson, 2013.  Gene expression, methylation, and miRNA data are provided for 348 BRCA tumor samples.       
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BRCA_data)</code></pre>


<h3>Format</h3>

<p>This dataset is a list of three entries for three different molecular sources:
</p>

<ul>
<li><p> Data[[1]] (Data$Expression): gene expression matrix for 654 genes (rows) and 348 samples (columns)
</p>
</li>
<li><p> Data[[2]] (Data$Methylation): DNA methylation matrix for 574 cg sites (rows) and 348 samples (columns)
</p>
</li>
<li><p> Data[[3]] (Data$miRNA): miRNA expression matrix for 423 cg sites (rows) and 348 samples (columns).
</p>
</li></ul>

<p>The 348 columns are shared by the data sources (here, they correspond to tumor samples)</p>


<h3>References</h3>

<p>Cancer Genome Atlas Network. 2012. &rdquo;Comprehensive Molecular Portraits of Human Breast Tumours.&rdquo; Nature 490 (7418): 61-70.
</p>
<p>Lock, E.F. and Dunson, D.B. 2013. &rdquo;Bayesian Consensus Clustering.&rdquo; Bioinformatics 29 (20): 2610-16.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>

<hr>
<h2 id='jive'>
JIVE Decomposition for Multi-source Data
</h2><span id='topic+jive'></span><span id='topic+jive.iter'></span><span id='topic+bic.jive'></span><span id='topic+jive.perm'></span>

<h3>Description</h3>

<p>Given a list of linked data sets, this algorithm will return low-rank matrices of joint and individual structure. The jive function is a wrapper that centers and scales the data, replaces the missing values using the SVDmiss function if necessary, then proceeds with a specified rank selection method. The jive.iter function performs the joint and individual variation explained (JIVE) decomposition, given ranks and the processed data set. The functions jive.perm and bic.jive perform rank selection using a permutation test and the Bayesian Information Criterion, respectively. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jive(data, rankJ = 1, rankA = rep(1, length(data)), method = "perm",
      dnames = names(data), conv = "default", maxiter = 1000, scale = TRUE, center = TRUE,
      orthIndiv = TRUE, est = TRUE, showProgress=TRUE)

jive.iter(data, rankJ = 1, rankA = rep(1, length(data)), conv = 1e-06,
           maxiter = 1000, orthIndiv = TRUE, showProgress=TRUE)

jive.perm(data, nperms = 100, alpha = 0.05, est = TRUE, conv = 1e-06, 
           maxiter = 1000, orthIndiv = TRUE, showProgress=TRUE)

bic.jive(data, n = unlist(lapply(data, ncol)) * unlist(lapply(data, nrow)),
          d = unlist(lapply(data, nrow)), conv = 1e-06, maxiter = 1000,
          orthIndiv = TRUE, showProgress=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jive_+3A_data">data</code></td>
<td>

<p>A list of two or more linked data matrices on which to perform the JIVE decomposition.  These matrices must have the same column dimension, which is assumed to be common.  
</p>
</td></tr>
<tr><td><code id="jive_+3A_rankj">rankJ</code></td>
<td>

<p>An integer giving the joint rank of the data, if known. If not given, this will be calculated using the chosen method.  If the method is &quot;given&quot; then the default is 1.
</p>
</td></tr>
<tr><td><code id="jive_+3A_ranka">rankA</code></td>
<td>

<p>A vector giving the indvidual ranks of the data, if known. If not given, this will be calculated using the chosen method.  If the method is &quot;given&quot; then the default is rep(1, length(data)). 
</p>
</td></tr>
<tr><td><code id="jive_+3A_method">method</code></td>
<td>

<p>A string with the method to use for rank selection. Possible options are &quot;given&quot;, &quot;perm&quot;, and &quot;bic&quot;. The default is &quot;perm&quot;. If ranks are known, you should use &quot;given&quot;.  
</p>
</td></tr>
<tr><td><code id="jive_+3A_dnames">dnames</code></td>
<td>

<p>A vector containing the names of the data sources. Default is names(data).
</p>
</td></tr>
<tr><td><code id="jive_+3A_conv">conv</code></td>
<td>

<p>A value indicating the convergence criterion. 
</p>
</td></tr>
<tr><td><code id="jive_+3A_maxiter">maxiter</code></td>
<td>

<p>The maximum number of iterations for each instance of the JIVE algorithm. 
</p>
</td></tr>
<tr><td><code id="jive_+3A_scale">scale</code></td>
<td>

<p>A boolean indicating whether or not the data should be scaled.  If TRUE, each data set is divided by its Frobenius norm prior to the JIVE algorithm. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="jive_+3A_center">center</code></td>
<td>

<p>A boolean indicating whether or not the data should be centered.  If TRUE, the rows of each data set are mean-centered. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="jive_+3A_orthindiv">orthIndiv</code></td>
<td>

<p>A boolean indicating whether or not the algorithm should enforce orthogonality between individual structures. The default is TRUE.
</p>
</td></tr>
<tr><td><code id="jive_+3A_est">est</code></td>
<td>

<p>A boolean indicating whether or not the data should first be compressed via singular value decomposition before running the JIVE algorithm;  this will yield identical results, but can improve computational efficiency dramatically for data with more rows than columns.  The default is TRUE.
</p>
</td></tr>
<tr><td><code id="jive_+3A_showprogress">showProgress</code></td>
<td>

<p>A boolean indicating whether or not to give output showing the progress of the algorithm. If TRUE, the algorithm will print out updates about the number of iterations the algorithm is taking and the progress of the rank selection method, if applicable.  If FALSE, the algorithms will give no printed output when run.  
</p>
</td></tr>
<tr><td><code id="jive_+3A_nperms">nperms</code></td>
<td>

<p>A value indicating the number of permutations for rank estimation. Default is 100.
</p>
</td></tr>
<tr><td><code id="jive_+3A_alpha">alpha</code></td>
<td>

<p>A value between 0 and 1 giving the quantile to use for rank estimation. Default is .05.
</p>
</td></tr>
<tr><td><code id="jive_+3A_n">n</code></td>
<td>

<p>A vector for the total number of entries in each data source, for use in the BIC calculation. The default is to calculate the total number of entries in each element of data. 
</p>
</td></tr>
<tr><td><code id="jive_+3A_d">d</code></td>
<td>

<p>A vector for the total number of variables (rows) in each data source, for use in the BIC calculation. The default is to calculate the number of rows in each element of data. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is recommended to make all calls to the JIVE functions using the jive() wrapper, as this function does all of the pre-processing of the data (centering, scaling, handling missingness, and reducing the data set to increase computational efficiency). The algorithm will print the number of iterations for each call of the JIVE iteration function. 
</p>


<h3>Value</h3>

<p>Returns an object of class jive.
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>a list containing the centered and scaled data sets, with missing values replaced, if applicable.</p>
</td></tr>
<tr><td><code>joint</code></td>
<td>
<p>a list containing matrices that capture the joint structure of the data.</p>
</td></tr>
<tr><td><code>individual</code></td>
<td>
<p>a list containing matrices that capture the individual structure of the data.</p>
</td></tr>
<tr><td><code>rankJ</code></td>
<td>
<p>a value giving joint rank of the data.</p>
</td></tr>
<tr><td><code>rankA</code></td>
<td>
<p>a vector giving the individual ranks of the data.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>a string denoting the rank selection method used.</p>
</td></tr>
<tr><td><code>bic.table</code></td>
<td>
<p>if bic rank selection used, a matrix that shows the BIC values for different ranks.</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>if permutation rank selection used, a boolean stating whether or not the rank selection converged within the maximum number of iterations.</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>A list of rour elements: $Center and $Scale are booleans stating whether the data were centered or scaled, respectively, $'Center Values' gives the value subtracted from each row and $'Scale Values' gives the multiplicative scale factor for each source.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics, 32(18):2877-2879, 2016.
</p>
<p>Jere, S., Dauwels, J., Asif, M. T., Vie, N. M., Cichocki, A., and Jaillet, P. (2014). Extracting commuting patterns in railway networks through matrix decompositions. In 13th IEEE International Conference on Control, Automation, Robotics and Vision (ICARCV), pages 541-546. IEEE.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.jive">summary.jive</a></code>, <code><a href="#topic+plot.jive">plot.jive</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(10)
##Load data that were simulated as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimData) 
# Using default method ("perm")
Results &lt;- jive(SimData)
summary(Results)

# Using BIC rank selection
#We set the maximum number of iterations allowed to 50 to speed up this example.  
#In practice we recommend a higher value, such as the default of 1000, 
#to ensure that all results converge.
BIC_result &lt;- jive(SimData,method="bic",maxiter=50)  
summary(BIC_result)
</code></pre>

<hr>
<h2 id='jive.predict'>
Predict JIVE scores for new data
</h2><span id='topic+jive.predict'></span>

<h3>Description</h3>

<p>Computes joint and individual variation explained (JIVE) scores for new data via iterative least squares, with fixed loadings given by a previous JIVE analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jive.predict(data.new, jive.output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jive.predict_+3A_data.new">data.new</code></td>
<td>

<p>A list of two or more linked data matrices on which to estimate JIVE scores.  These matrices must have the same column dimension N, which is assumed to be common. 
</p>
</td></tr>
<tr><td><code id="jive.predict_+3A_jive.output">jive.output</code></td>
<td>

<p>An object of class &quot;jive&quot;, with row dimensions matching those for data.new.  
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>joint.scores</code></td>
<td>
<p>r X N matrix of joint scores</p>
</td></tr>
<tr><td><code>individual.scores</code></td>
<td>
<p>List where entry [[i]] gives the r_i X N matrix of individual scores for source i</p>
</td></tr>
<tr><td><code>errors</code></td>
<td>
<p>Vector of the proportion of total variance explained over iterations during estimation</p>
</td></tr>
<tr><td><code>joint.load</code></td>
<td>
<p>d X r matrix of joint loadings</p>
</td></tr> 
<tr><td><code>indiv.load</code></td>
<td>
<p>List where entry [[i]] gives the d_i X N matrix of individual laodings for source i</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Adam Kaplan
</p>


<h3>References</h3>

<p>Kaplan, A. and Lock, E.F. (2017). Prediction with Dimension Reduction of Multiple Molecular Data Sources for Patient Survival. arXiv:1704.02069, 2017.   
</p>


<h3>See Also</h3>

<p><code><a href="#topic+jive">jive</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load data that were simulated as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimData) 
##load JIVE results (using default settings) for simulated data 
data(SimResults) 
#predict JIVE scores for data (treated as "new data" here)
pred.results &lt;- jive.predict(SimData,Results) 
##estimated joint structure is pred.results$joint.load %*% pred.results$joint.scores
##estimated individual structure for source i is 
##pred.results$indiv.load[[i]] %*% pred.results$indiv.scores[[i]]
</code></pre>

<hr>
<h2 id='pjsum'>
Calculate Number of Free Parameters for BIC Calculation
</h2><span id='topic+pjsum'></span>

<h3>Description</h3>

<p>Computes the number of free parameters from the individual structure of the data.  Used internally to calculate the BIC for the JIVE decomposition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pjsum(dim, rank)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pjsum_+3A_dim">dim</code></td>
<td>

<p>A vector containing the number of rows of each data source. 
</p>
</td></tr>
<tr><td><code id="pjsum_+3A_rank">rank</code></td>
<td>

<p>A vector containing the ranks of the individual structure. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the number of free parameters. 
</p>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pjsum(c(25,50), c(1,2))
</code></pre>

<hr>
<h2 id='plot.jive'>
Create Plots for a JIVE Object
</h2><span id='topic+plot.jive'></span>

<h3>Description</h3>

<p>Three types of plots are available. By default (or type=&quot;var&quot;), this creates a bar plot showing the percentage of variability attributable to joint structure, individual structure, and residual variance. With type=&quot;heat&quot;, it will create a series of heatmaps. With type=&quot;pca&quot;, it will give principal component plots.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'jive'
plot(x, type="var", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.jive_+3A_x">x</code></td>
<td>

<p>An object of class &quot;jive&quot; to be plotted.
</p>
</td></tr>
<tr><td><code id="plot.jive_+3A_type">type</code></td>
<td>

<p>A string indicating the type of plot. The default, &quot;var&quot;, generates a bar plot of the variance explained, &quot;heat&quot; generates a heatmap, and &quot;pca&quot; generates principal component plots. 
</p>
</td></tr>
<tr><td><code id="plot.jive_+3A_...">...</code></td>
<td>

<p>Additional arguments to pass to the specific plotting functions. See documentation for <code><a href="#topic+showVarExplained">showVarExplained</a></code>,<code><a href="#topic+showHeatmaps">showHeatmaps</a></code>, and <code><a href="#topic+showPCA">showPCA</a></code> for more details.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+showVarExplained">showVarExplained</a></code>,<code><a href="#topic+showHeatmaps">showHeatmaps</a></code>,<code><a href="#topic+showPCA">showPCA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load JIVE results (using default settings) for simulated data 
##as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimResults) 
# Visualize results
# Bar plot of variation explained
plot(Results)
# Heatmap
plot(Results,type="heat")
# Principal compontents plots
plot(Results,type="pca",1,c(1,1))
</code></pre>

<hr>
<h2 id='show.image'>
Draw a Heatmap from a Matrix
</h2><span id='topic+show.image'></span>

<h3>Description</h3>

<p>Given a matrix, this function draws a heatmap. This function is used internally by the showHeatmaps function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show.image(Image, ylab = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show.image_+3A_image">Image</code></td>
<td>

<p>A matrix for which to create the heatmap.
</p>
</td></tr>
<tr><td><code id="show.image_+3A_ylab">ylab</code></td>
<td>

<p>A string for the y-label of the plot. 
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+showHeatmaps">showHeatmaps</a></code>, <code><a href="#topic+jive">jive</a></code>, <code><a href="#topic+plot.jive">plot.jive</a></code>
</p>

<hr>
<h2 id='showHeatmaps'>
Heatmaps for JIVE Decompositions
</h2><span id='topic+showHeatmaps'></span>

<h3>Description</h3>

<p>This function draws heatmaps for the components of a JIVE decomposition. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showHeatmaps(result, order_by = 0, show_all = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showHeatmaps_+3A_result">result</code></td>
<td>

<p>An object of class &quot;jive&quot;.
</p>
</td></tr>
<tr><td><code id="showHeatmaps_+3A_order_by">order_by</code></td>
<td>

<p>Specifies how to order the rows and columns of the heatmap.  If order_by=-1, the matrices are not re-ordered.  If order_by=0, orderings are determined by joint structure. Otherwise, order_by gives the number of the individual structure dataset to determine the ordering. In all cases orderings are determined by complete-linkage hiearchichal clustering of Euclidean distances.
</p>
</td></tr>
<tr><td><code id="showHeatmaps_+3A_show_all">show_all</code></td>
<td>

<p>Specifies whether to show the full decomposition of the data, JIVE estimates, and noise.  If show_all=FALSE, only the matrix (or matrices) that determined the column ordering is shown.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The columns correspond to the shared dimension (for example, a common sample set), and the ordering of the columns is the same for all matrices shown. 
</p>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+jive">jive</a></code>, <code><a href="#topic+plot.jive">plot.jive</a></code>, <code><a href="#topic+show.image">show.image</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load JIVE results (using default settings) for simulated data 
##as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimResults) 
# Display the heatmaps (may need to fiddle with plot window dimensions for this to appear well)
showHeatmaps(Results)
# Order by first data set individual structure
showHeatmaps(Results, order_by=1)
# Show only joint structure
showHeatmaps(Results, show_all=FALSE)
</code></pre>

<hr>
<h2 id='showPCA'>
Principal Component Plots for JIVE Decomposition
</h2><span id='topic+showPCA'></span>

<h3>Description</h3>

<p>Display principal component plots of the joint and individual structure of a data set after a joint and individual variation explained (JIVE) decomposition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showPCA(result, n_joint = 0, n_indiv = rep(0, length(result$data)), 
         Colors = "black", pch=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showPCA_+3A_result">result</code></td>
<td>

<p>An object of class &quot;jive&quot;.
</p>
</td></tr>
<tr><td><code id="showPCA_+3A_n_joint">n_joint</code></td>
<td>

<p>The number of joint components to plot. 
</p>
</td></tr>
<tr><td><code id="showPCA_+3A_n_indiv">n_indiv</code></td>
<td>

<p>The vector of the number of individual components to plot for each data set.
</p>
</td></tr>
<tr><td><code id="showPCA_+3A_colors">Colors</code></td>
<td>

<p>The colors of the data points in the plot. Can be a vector specifying a different color for each sample.
</p>
</td></tr>
<tr><td><code id="showPCA_+3A_pch">pch</code></td>
<td>

<p>Character to use for plotting.  Can be a vector specifying a different character for each sample.  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This shows the patterns in the column space that maximize variability of joint or individual structure, analogous to principal components.  A multi-panel figure with aligned scatterplots for each pair of principal components, across both joint and individual structure, will be generated.  Plotted points correspond to shared columns (e.g., samples).  
</p>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+jive">jive</a></code>, <code><a href="#topic+plot.jive">plot.jive</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load JIVE results (using default settings) for simulated data 
##as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimResults) 
# Visualize results
# Plot the three components, 1 joint and 1 individual from each source
showPCA(Results,1,c(1,1))
###This displays three scatterplots: 
#the first joint principal component vs. the first principal component individual to source 1,
#the first joint component vs.  the first component individual to source 2, and 
#the first component individual to source 1 vs. the first component individual to source 2.
</code></pre>

<hr>
<h2 id='showVarExplained'>
Display Variance Explained
</h2><span id='topic+showVarExplained'></span>

<h3>Description</h3>

<p>Creates a bar plot displaying the variance explained from a joint and individual variation explained (JIVE) decomposition. Shows the percentage of variance attributed to each of joint structure, individual structure, and residual variance. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showVarExplained(result, col = c("grey20", "grey43", "grey65"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showVarExplained_+3A_result">result</code></td>
<td>

<p>An object of class &quot;jive&quot;.
</p>
</td></tr>
<tr><td><code id="showVarExplained_+3A_col">col</code></td>
<td>

<p>A vector for the colors of the bars in the plot.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.jive">plot.jive</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load JIVE results (using default settings) for simulated data 
##as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimResults) 
# Visualize results
showVarExplained(Results)
# showVarExplained is also called by the "jive" S3 class default plot method
plot(Results)
</code></pre>

<hr>
<h2 id='SimData'>Simulated Dataset</h2><span id='topic+SimData'></span>

<h3>Description</h3>

<p>These data were simulated as described in Section 2.4 of Lock et al., 2013. There are two simulated sources, with rank 1 joint structure and rank 1 structure individual to each source.      
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SimData)</code></pre>


<h3>Format</h3>

<p>This dataset is a list of two entries:
</p>

<ul>
<li><p> Data[[1]] (Data$Data1): 50 variables (rows) and 100 samples (columns)
</p>
</li>
<li><p> Data[[2]] (Data$Data2): 50 variables (rows) and 100 samples (columns)
</p>
</li></ul>

<p>The 100 columns are shared by the sources.</p>


<h3>References</h3>

<p>Lock, EF, KA Hoadley, JS Marron, and AB Nobel. 2013. &rdquo;Joint and Individual Variation Explained (JIVE) for Integrated Analysis of Multiple Data Types.&rdquo; The Annals of Applied Statistics 7 (1): 523-42.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>

<hr>
<h2 id='SimResults'>JIVE Results for Simulated Dataset</h2><span id='topic+Results'></span>

<h3>Description</h3>

<p>JIVE results for the simulated data SimData, which were simulated as described in Section 2.4 of Lock et al., 2013. There are two simulated sources, with rank 1 joint structure and rank 1 structure individual to each source.  These results are obtained by running JIVE with permutation testing to select the ranks, and other defaults.        
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SimResults)</code></pre>


<h3>Format</h3>

<p>Results: an object of class 'jive'.</p>


<h3>References</h3>

<p>Lock, EF, KA Hoadley, JS Marron, and AB Nobel. 2013. &rdquo;Joint and Individual Variation Explained (JIVE) for Integrated Analysis of Multiple Data Types.&rdquo; The Annals of Applied Statistics 7 (1): 523-42.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>

<hr>
<h2 id='summary.jive'>
Summarize a JIVE Decomposition
</h2><span id='topic+summary.jive'></span><span id='topic+print.jive'></span>

<h3>Description</h3>

<p>Provides a summary of JIVE output. Displays the method used for rank selection, the chosen ranks, and a table of the proportion of variance attributable to joint structure, individual structure, and residual variance. print.jive only displays the variance table. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'jive'
summary(object, ...)

## S3 method for class 'jive'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.jive_+3A_object">object</code></td>
<td>

<p>An object of class &quot;jive&quot;.
</p>
</td></tr>
<tr><td><code id="summary.jive_+3A_x">x</code></td>
<td>

<p>An object of class &quot;jive&quot;.
</p>
</td></tr>
<tr><td><code id="summary.jive_+3A_...">...</code></td>
<td>

<p>Additional arguments. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list.
</p>
<table>
<tr><td><code>Method</code></td>
<td>
<p>a string containing the method used for rank selection.</p>
</td></tr>
<tr><td><code>Ranks</code></td>
<td>
<p>the method used for rank selection.</p>
</td></tr>
<tr><td><code>Variance</code></td>
<td>
<p>the method used for rank selection.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>References</h3>

<p>Lock, E. F., Hoadley, K. A., Marron, J. S., &amp; Nobel, A. B. (2013). Joint and individual variation explained (JIVE) for integrated analysis of multiple data types. The Annals of Applied Statistics, 7(1), 523-542.
</p>
<p>O'Connell, M. J., &amp; Lock, E.F. (2016). R.JIVE for Exploration of Multi-Source Molecular Data. Bioinformatics advance access: 10.1093/bioinformatics/btw324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+jive">jive</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load JIVE results (using default settings) for simulated data 
##as in Section 2.4 of Lock et al., 2013,
##with rank 1 joint structure, and rank 1 individual structure for each dataset
data(SimResults) 

# Summary method
summary(Results)

# Print method
Results
</code></pre>

<hr>
<h2 id='svdmiss'>
Missing Data SVD
</h2><span id='topic+SVDmiss'></span>

<h3>Description</h3>

<p>This function and description borrowed from R package SpatioTemporal (no longer on CRAN), from authors Paul D. Sampson and Johan Lindstrom. It completes a data matrix using iterative svd as described in Fuentes et. al. (2006). The function iterates between computing the singular value decomposition (svd) for the matrix and replacing the missing values by linear regression of the columns onto the first ncomp svd components. As initial replacement for the missing values regression on the column averages are used. The function will fail if entire rows and/or columns are missing from the data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SVDmiss(X, niter = 25, ncomp = min(4, dim(X)[2]), conv.reldiff = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="svdmiss_+3A_x">X</code></td>
<td>

<p>Data matrix, with missing values marked by NA.
</p>
</td></tr>
<tr><td><code id="svdmiss_+3A_niter">niter</code></td>
<td>

<p>Maximum number of iterations to run before exiting, Inf will run until the conv.reldiff criteria is met.
</p>
</td></tr>
<tr><td><code id="svdmiss_+3A_ncomp">ncomp</code></td>
<td>

<p>Number of SVD components to use in the reconstruction (&gt;0).
</p>
</td></tr>
<tr><td><code id="svdmiss_+3A_conv.reldiff">conv.reldiff</code></td>
<td>

<p>Assume the iterative procedure has converged when the relative difference between two consecutive iterations is less than conv.reldiff.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components: </p>
<table>
<tr><td><code>Xfill</code></td>
<td>
<p>The
completed data matrix with missing values replaced by
fitting the data to the <code>ncomp</code> most important svd
components</p>
</td></tr> <tr><td><code>svd</code></td>
<td>
<p>The result of svd on the completed
data matrix, i.e.  <code>svd(Xfill)</code></p>
</td></tr> <tr><td><code>status</code></td>
<td>
<p>A
vector of status variables: <code>diff</code>, the absolute
difference between the two last iterations;
<code>rel.diff</code>, the relative difference; <code>n.iter</code>,
the number of iterations; and <code>max.iter</code>, the
requested maximum number of iterations.</p>
</td></tr>
</table>

<hr>
<h2 id='svdwrapper'>
Wrapper Function to Perform SVD
</h2><span id='topic+svdwrapper'></span>

<h3>Description</h3>

<p>Performs SVD on a data matrix using the base svd() function in R, with a workaround to avoid LAPACK errors.  If an SVD of the data matrix gives an error, an SVD of its transpose will be performed.  Used internally when computing the JIVE decomposition.  Credit to Art Owen: <a href="https://stat.ethz.ch/pipermail/r-help/2007-October/143508.html">https://stat.ethz.ch/pipermail/r-help/2007-October/143508.html</a>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>svdwrapper(x, nu, nv, verbose=F )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="svdwrapper_+3A_x">x</code></td>
<td>

<p>a numeric matrix whos SVD decomposition is to be computed.
</p>
</td></tr>
<tr><td><code id="svdwrapper_+3A_nu">nu</code></td>
<td>

<p>the number of left singular vectors to be computed.
</p>
</td></tr>
<tr><td><code id="svdwrapper_+3A_nv">nv</code></td>
<td>

<p>the number of right singular vectors to be computed.
</p>
</td></tr>
<tr><td><code id="svdwrapper_+3A_verbose">verbose</code></td>
<td>

<p>logical.  Print error message if needed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An svd object, as returned by svd(x,nu=nu,nv=nv).
</p>


<h3>Author(s)</h3>

<p>Michael J. O'Connell and Eric F. Lock
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x&lt;-matrix(rnorm(100),nrow=10,ncol=10)
SVD = svdwrapper(x,nu=1,nv=1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
