<!DOCTYPE html><html><head><title>Help for package Racmacs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Racmacs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#acmap'><p>Generate a new acmap object</p></a></li>
<li><a href='#acmapAttributes'><p>Get acmap attributes</p></a></li>
<li><a href='#addOptimization'><p>Add a new optimization to an acmap object</p></a></li>
<li><a href='#adjustedLogTiterTable'><p>Get the reactivity adjusted log titer table</p></a></li>
<li><a href='#adjustedTiterTable'><p>Get the reactivity adjusted titer table</p></a></li>
<li><a href='#agAttributes'><p>Getting and setting antigen attributes</p></a></li>
<li><a href='#agCohesion'><p>Check map cohesion</p></a></li>
<li><a href='#agGroups'><p>Getting and setting antigen groups</p></a></li>
<li><a href='#agHomologousSr'><p>Get homologous sera for each antigen</p></a></li>
<li><a href='#agLabIDs'><p>Getting and setting antigen lab id information</p></a></li>
<li><a href='#agReactivityAdjustments'><p>Get and set antigen reactivity adjustments</p></a></li>
<li><a href='#agSequences'><p>Getting and setting antigen sequence information</p></a></li>
<li><a href='#applyMapTransform'><p>Apply the current map transformation</p></a></li>
<li><a href='#applyPlotspec'><p>Apply a plotspec from another acmap</p></a></li>
<li><a href='#as.json'><p>Convert map to json format</p></a></li>
<li><a href='#blob'><p>Plot a blob object</p></a></li>
<li><a href='#blobsize'><p>Calculate size of a blob object</p></a></li>
<li><a href='#bootstrapBlobs'><p>Calculate bootstrap blob data for an antigenic map</p></a></li>
<li><a href='#bootstrapMap'><p>Perform a bootstrap on a map</p></a></li>
<li><a href='#checkHemisphering'><p>Check for hemisphering or trapped points</p></a></li>
<li><a href='#colBases'><p>Getting and setting column bases</p></a></li>
<li><a href='#deprecated_functions'><p>Deprecated functions</p></a></li>
<li><a href='#dilutionStepsize'><p>Get or set the dilution stepsize associated with a map</p></a></li>
<li><a href='#dimensionTestMap'><p>Perform dimension testing on a map object</p></a></li>
<li><a href='#edit_agNames'><p>Edit antigen names in an acmap</p></a></li>
<li><a href='#edit_srNames'><p>Edit sera names in an acmap</p></a></li>
<li><a href='#export_viewer'><p>Export the map viewer</p></a></li>
<li><a href='#getOptimization'><p>Get optimization details from an acmap object</p></a></li>
<li><a href='#ggplot.acmap'><p>Plot an antigenic map using ggplot</p></a></li>
<li><a href='#htmlAdjustedTiterTable'><p>Return an html formatted titer table with antigen reactivity adjustments applied</p></a></li>
<li><a href='#htmlMergeReport'><p>Return an html formatted merge report</p></a></li>
<li><a href='#htmlTiterTable'><p>Return an html formatted titer table</p></a></li>
<li><a href='#keepBestOptimization'><p>Keep only the lowest stress map optimization</p></a></li>
<li><a href='#keepOptimizations'><p>Keep specified optimization runs</p></a></li>
<li><a href='#keepSingleOptimization'><p>Keep only a single optimization run</p></a></li>
<li><a href='#layerNames'><p>Get and set map layer names</p></a></li>
<li><a href='#listOptimizations'><p>Get all optimization details from an acmap object</p></a></li>
<li><a href='#logtiterTable'><p>Get the log titers from an acmap</p></a></li>
<li><a href='#logtiterTableLayers'><p>Return a list of logtiter table layers</p></a></li>
<li><a href='#make.acmap'><p>Make an antigenic map from scratch</p></a></li>
<li><a href='#map-table-distances'><p>Plot map vs table distances</p></a></li>
<li><a href='#mapBootstrapCoords'><p>Get bootstrap coordinates associated with a map</p></a></li>
<li><a href='#mapComment'><p>Get or set an optimization run comment</p></a></li>
<li><a href='#mapDescription'><p>Getting and setting the map description</p></a></li>
<li><a href='#mapDimensions'><p>Get the current map dimensions</p></a></li>
<li><a href='#mapDistances'><p>Return calculated map distances for an acmap</p></a></li>
<li><a href='#mapGadget'><p>Open a shiny gadget to view the map</p></a></li>
<li><a href='#mapName'><p>Getting and setting the map name</p></a></li>
<li><a href='#mapRelaxed'><p>Check if a map has been fully relaxed</p></a></li>
<li><a href='#mapResiduals'><p>Get a table of residuals from an acmap</p></a></li>
<li><a href='#mapStress'><p>Calculate the current map stress</p></a></li>
<li><a href='#mapTransformation'><p>Reading map transformation data</p></a></li>
<li><a href='#matchStrains'><p>Find matching antigens or sera between 2 maps</p></a></li>
<li><a href='#mergeMaps'><p>Merging maps</p></a></li>
<li><a href='#mergeReport'><p>Return a merge report</p></a></li>
<li><a href='#moveTrappedPoints'><p>Move trapped points</p></a></li>
<li><a href='#optimizationProperties'><p>Get optimization properties</p></a></li>
<li><a href='#optimizeAgReactivity'><p>Optimize antigen reactivity adjustments</p></a></li>
<li><a href='#optimizeMap'><p>Optimize an acmap</p></a></li>
<li><a href='#orderPoints'><p>Order antigens and sera</p></a></li>
<li><a href='#plot.acmap'><p>Plot an antigenic map</p></a></li>
<li><a href='#pointStress'><p>Get individual point stress</p></a></li>
<li><a href='#procrustesData'><p>Return procrustes data on a map comparison</p></a></li>
<li><a href='#procrustesMap'><p>Return procrustes information</p></a></li>
<li><a href='#ptAnnotations'><p>Getting and setting point annotation information</p></a></li>
<li><a href='#ptBaseCoords'><p>Getting and setting base coordinates</p></a></li>
<li><a href='#ptBootstrapBlob'><p>Get antigen or serum bootstrap blob information</p></a></li>
<li><a href='#ptBootstrapCoords'><p>Get antigen or serum bootstrap coordinates information</p></a></li>
<li><a href='#ptClades'><p>Getting and setting point clade information</p></a></li>
<li><a href='#ptCoords'><p>Getting and setting point coordinates</p></a></li>
<li><a href='#ptDrawingOrder'><p>Get and set point drawing order in map</p></a></li>
<li><a href='#ptLeverage'><p>Calculate point leverage</p></a></li>
<li><a href='#ptOpacity'><p>Set point opacity in a map</p></a></li>
<li><a href='#ptStyles'><p>Getting and setting point plotting styles</p></a></li>
<li><a href='#ptTriangulationBlob'><p>Get antigen or serum triangulation blob information</p></a></li>
<li><a href='#Racmacs-package'><p>Racmacs: Antigenic Cartography Macros</p></a></li>
<li><a href='#RacMerge.options'><p>Set acmap merge options</p></a></li>
<li><a href='#RacOptimizer.options'><p>Set acmap optimization options</p></a></li>
<li><a href='#RacViewer'><p>Create a RacViewer widget</p></a></li>
<li><a href='#RacViewer-shiny'><p>Shiny bindings for RacViewer</p></a></li>
<li><a href='#RacViewer.options'><p>Set viewer options</p></a></li>
<li><a href='#randomizeCoords'><p>Randomize map coordinates</p></a></li>
<li><a href='#read.acmap'><p>Read in acmap data from a file</p></a></li>
<li><a href='#read.titerTable'><p>Read in a table of titer data</p></a></li>
<li><a href='#realignMap'><p>Realign map to match another</p></a></li>
<li><a href='#realignOptimizations'><p>Realigns optimizations in the map</p></a></li>
<li><a href='#recalculateStress'><p>Recalculate the stress associated with an acmap optimization</p></a></li>
<li><a href='#reflectMap'><p>Reflect a map</p></a></li>
<li><a href='#relaxMap'><p>Relax a map</p></a></li>
<li><a href='#relaxMapOneStep'><p>Relax a map one step in the optimiser</p></a></li>
<li><a href='#removeOptimizations'><p>Remove map optimizations</p></a></li>
<li><a href='#removePoints'><p>Remove antigens and sera</p></a></li>
<li><a href='#rotateMap'><p>Rotate a map</p></a></li>
<li><a href='#runGUI'><p>Open the Racmacs GUI</p></a></li>
<li><a href='#save.acmap'><p>Save acmap data to a file</p></a></li>
<li><a href='#save.coords'><p>Save acmap coordinate data to a file</p></a></li>
<li><a href='#save.titerTable'><p>Save titer data to a file</p></a></li>
<li><a href='#setLegend'><p>Set acmap legend</p></a></li>
<li><a href='#sortOptimizations'><p>Sort optimizations by stress</p></a></li>
<li><a href='#splitTiterLayers'><p>Split a map made up from titer layers into a list of separate maps each with a titer table</p>
corresponding to one of the layers</a></li>
<li><a href='#srAttributes'><p>Getting and setting sera attributes</p></a></li>
<li><a href='#srGroups'><p>Getting and setting sera groups</p></a></li>
<li><a href='#srHomologousAgs'><p>Get and set homologous antigens for sera</p></a></li>
<li><a href='#srSequences'><p>Getting and setting sera sequence information</p></a></li>
<li><a href='#standardizeStrainNames'><p>Standardize strain names</p></a></li>
<li><a href='#stressTable'><p>Get a stress table from an acmap</p></a></li>
<li><a href='#subsetCommonPoints'><p>Remove antigens and sera</p></a></li>
<li><a href='#subsetMap'><p>Subset an antigenic map</p></a></li>
<li><a href='#tableColbases'><p>Calculate column bases for a titer table</p></a></li>
<li><a href='#tableDistances'><p>Return calculated table distances for an acmap</p></a></li>
<li><a href='#titerTable'><p>Getting and setting map titers</p></a></li>
<li><a href='#titerTableFlat'><p>Getting and setting the flat titer table</p></a></li>
<li><a href='#titerTableLayers'><p>Getting and setting titer table layers</p></a></li>
<li><a href='#translateMap'><p>Translate a map</p></a></li>
<li><a href='#triangulationBlobs'><p>Calculate triangulation blobs data for an antigenic map</p></a></li>
<li><a href='#unstableMaps'><p>Notes on unstable maps</p></a></li>
<li><a href='#update_ferret_seraNames'><p>Update the ferret serum names to match antigens</p></a></li>
<li><a href='#view'><p>S3 method for viewing objects</p></a></li>
<li><a href='#view.acmap'><p>Viewing racmap objects</p></a></li>
<li><a href='#view.default'><p>Default method for viewing objects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Antigenic Cartography Macros</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-30</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolkit for making antigenic maps from immunological assay data,
    in order to quantify and visualize antigenic differences between different
    pathogen strains as described in
    Smith et al. (2004) &lt;<a href="https://doi.org/10.1126%2Fscience.1097211">doi:10.1126/science.1097211</a>&gt; and used in the World
    Health Organization influenza vaccine strain selection process. Additional
    functions allow for the diagnostic evaluation of antigenic maps and an
    interactive viewer is provided to explore antigenic relationships amongst
    several strains and incorporate the visualization of associated genetic
    information.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, jsonlite, ks, brotli, shiny, shinyFiles, shinyjs,
htmlwidgets, ggplot2 (&ge; 3.0.0), htmltools, rmarchingcubes,
shape, ellipsis, MASS, magrittr, igraph, dplyr, vctrs, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, r3js, knitr, rmarkdown, rstudioapi, plotly,
geometry, readxl, stringr, tibble, tidyr, base64enc, lifecycle,
MCMCpack, spelling</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppProgress, RcppEnsmallen, rapidjsonr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://acorg.github.io/Racmacs/">https://acorg.github.io/Racmacs/</a>,
<a href="https://github.com/acorg/Racmacs/">https://github.com/acorg/Racmacs/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/acorg/Racmacs/issues">https://github.com/acorg/Racmacs/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-30 11:11:49 UTC; cambridge</td>
</tr>
<tr>
<td>Author:</td>
<td>Sam Wilks [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sam Wilks &lt;sw463@cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-30 11:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='acmap'>Generate a new acmap object</h2><span id='topic+acmap'></span>

<h3>Description</h3>

<p>This function generates a new acmap object, the base object for storing map
data in the Racmacs package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acmap(
  ag_names = NULL,
  sr_names = NULL,
  titer_table = NULL,
  ag_coords = NULL,
  sr_coords = NULL,
  check_duplicates = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="acmap_+3A_ag_names">ag_names</code></td>
<td>
<p>Antigen names</p>
</td></tr>
<tr><td><code id="acmap_+3A_sr_names">sr_names</code></td>
<td>
<p>Sera names</p>
</td></tr>
<tr><td><code id="acmap_+3A_titer_table">titer_table</code></td>
<td>
<p>Table of titer data</p>
</td></tr>
<tr><td><code id="acmap_+3A_ag_coords">ag_coords</code></td>
<td>
<p>Antigenic coordinates for an optimization run record
(optional)</p>
</td></tr>
<tr><td><code id="acmap_+3A_sr_coords">sr_coords</code></td>
<td>
<p>Sera coordinates for an optimization run record (optional)</p>
</td></tr>
<tr><td><code id="acmap_+3A_check_duplicates">check_duplicates</code></td>
<td>
<p>Issue a warning if duplicate antigen or sera names
are found</p>
</td></tr>
<tr><td><code id="acmap_+3A_...">...</code></td>
<td>
<p>Further arguments passed to <code>addOptimization()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fundamental unit of the Racmacs package is the <code>acmap</code> object,
short for Antigenic Cartography MAP. This object contains all the
information about an antigenic map. You can read in a new acmap object from
a file with the function <code>read.acmap()</code> and create a new acmap object
within an R session using the <code>acmap()</code> function.
</p>


<h3>Value</h3>

<p>Returns the new acmap object
</p>


<h3>See Also</h3>

<p>See <code>optimizeMap()</code> for generating new optimizations estimating
antigen similarity from the acmap titer data.
</p>
<p>Other functions for working with map data: 
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='acmapAttributes'>Get acmap attributes</h2><span id='topic+acmapAttributes'></span><span id='topic+numAntigens'></span><span id='topic+numSera'></span><span id='topic+numSeraGroups'></span><span id='topic+numPoints'></span><span id='topic+numOptimizations'></span><span id='topic+numLayers'></span>

<h3>Description</h3>

<p>Functions to get various attributes about an acmap object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numAntigens(map)

numSera(map)

numSeraGroups(map)

numPoints(map)

numOptimizations(map)

numLayers(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="acmapAttributes_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A number relating to the attribute
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='addOptimization'>Add a new optimization to an acmap object</h2><span id='topic+addOptimization'></span>

<h3>Description</h3>

<p>Function to add a new optimization to an acmap object, with specified values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addOptimization(
  map,
  ag_coords = NULL,
  sr_coords = NULL,
  number_of_dimensions = NULL,
  minimum_column_basis = "none",
  fixed_column_bases = NULL,
  ag_reactivity_adjustments = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addOptimization_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="addOptimization_+3A_ag_coords">ag_coords</code></td>
<td>
<p>Antigen coordinates for the new optimization (0 if not
specified)</p>
</td></tr>
<tr><td><code id="addOptimization_+3A_sr_coords">sr_coords</code></td>
<td>
<p>Sera coordinates for the new optimization (0 if not
specified)</p>
</td></tr>
<tr><td><code id="addOptimization_+3A_number_of_dimensions">number_of_dimensions</code></td>
<td>
<p>The number of dimensions of the new optimization</p>
</td></tr>
<tr><td><code id="addOptimization_+3A_minimum_column_basis">minimum_column_basis</code></td>
<td>
<p>The minimum column basis to use for the new
optimization</p>
</td></tr>
<tr><td><code id="addOptimization_+3A_fixed_column_bases">fixed_column_bases</code></td>
<td>
<p>A vector of fixed column bases with NA for sera
where the minimum column basis should be applied</p>
</td></tr>
<tr><td><code id="addOptimization_+3A_ag_reactivity_adjustments">ag_reactivity_adjustments</code></td>
<td>
<p>A vector of antigen reactivity adjustments to
apply to each antigen. Corresponding antigen titers will be adjusted by these
amounts when calculating column bases and table distances.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the acmap data object with new optimization added (but not
selected).
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='adjustedLogTiterTable'>Get the reactivity adjusted log titer table</h2><span id='topic+adjustedLogTiterTable'></span>

<h3>Description</h3>

<p>Return the log titer table plus any antigen reactivity adjustments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjustedLogTiterTable(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjustedLogTiterTable_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="adjustedLogTiterTable_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number from which
to take any antigen reactivity adjustments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix of adjusted log titers.
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='adjustedTiterTable'>Get the reactivity adjusted titer table</h2><span id='topic+adjustedTiterTable'></span>

<h3>Description</h3>

<p>Return the titer table plus any antigen reactivity adjustments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjustedTiterTable(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjustedTiterTable_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="adjustedTiterTable_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number from which
to take any antigen reactivity adjustments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character matrix of titers.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+htmlAdjustedTiterTable">htmlAdjustedTiterTable()</a></code>
</p>
<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='agAttributes'>Getting and setting antigen attributes</h2><span id='topic+agAttributes'></span><span id='topic+agIDs'></span><span id='topic+agIDs+3C-'></span><span id='topic+agDates'></span><span id='topic+agDates+3C-'></span><span id='topic+agReference'></span><span id='topic+agReference+3C-'></span><span id='topic+agNames'></span><span id='topic+agNames+3C-'></span><span id='topic+agExtra'></span><span id='topic+agExtra+3C-'></span><span id='topic+agPassage'></span><span id='topic+agPassage+3C-'></span><span id='topic+agLineage'></span><span id='topic+agLineage+3C-'></span><span id='topic+agReassortant'></span><span id='topic+agReassortant+3C-'></span><span id='topic+agStrings'></span><span id='topic+agStrings+3C-'></span><span id='topic+agContinent'></span><span id='topic+agContinent+3C-'></span>

<h3>Description</h3>

<p>These functions get and set the antigen attributes for a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agIDs(map)
agIDs(map) &lt;- value
agDates(map)
agDates(map) &lt;- value
agReference(map)
agReference(map) &lt;- value
agNames(map)
agNames(map) &lt;- value
agExtra(map)
agExtra(map) &lt;- value
agPassage(map)
agPassage(map) &lt;- value
agLineage(map)
agLineage(map) &lt;- value
agReassortant(map)
agReassortant(map) &lt;- value
agStrings(map)
agStrings(map) &lt;- value
agContinent(map)
agContinent(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agAttributes_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="agAttributes_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p><code>srAttributes()</code>
</p>
<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='agCohesion'>Check map cohesion</h2><span id='topic+agCohesion'></span><span id='topic+srCohesion'></span><span id='topic+mapCohesion'></span>

<h3>Description</h3>

<p>Checks the vertex connectivity of points in a map (the minimum number of
points needed to remove from the map to eliminate all paths from one point to
another point). This is for checking for example if after merging maps you
only have a small number of points in common between separate groups of
points, leading to a situation where groups of points cannot be robustly
positioned relative to each other. If the vertex connectivity is smaller than
the number of map dimensions + 1 then this will certainly be occurring and
will lead to an unstable map solution. <code>mapCohesion()</code> returns the minimum
vertex connectivity found between any given points, while <code>agCohesion()</code> and
<code>srCohesion()</code> return the vertex connectivity between each pair of antigens
and sera as a table helping to diagnose which antigens and sera are forming
separate groups. Note that for these purposes only detectable titers count
as connections and non-detectable titers are ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agCohesion(map)

srCohesion(map)

mapCohesion(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agCohesion_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scalar real value.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='agGroups'>Getting and setting antigen groups</h2><span id='topic+agGroups'></span><span id='topic+agGroups+3C-'></span>

<h3>Description</h3>

<p>These functions get and set the antigen groupings for a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agGroups(map)

agGroups(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agGroups_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="agGroups_+3A_value">value</code></td>
<td>
<p>A character or factor vector of groupings to apply to the
antigens</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A factor vector of groupings.
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='agHomologousSr'>Get homologous sera for each antigen</h2><span id='topic+agHomologousSr'></span>

<h3>Description</h3>

<p>Gets the indices of homologous sera for each antigen in an antigenic map.
See also the function <code>srHomologousAgs()</code> for getting and setting the
homologous antigens reciprocally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agHomologousSr(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agHomologousSr_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, where each entry is a vector of indices for homologous
sera, or a length 0 vector where no homologous serum is present
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='agLabIDs'>Getting and setting antigen lab id information</h2><span id='topic+agLabIDs'></span><span id='topic+agLabIDs+3C-'></span>

<h3>Description</h3>

<p>Getting and setting antigen lab id information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agLabIDs(map)

agLabIDs(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agLabIDs_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="agLabIDs_+3A_value">value</code></td>
<td>
<p>A list of character vectors with lab ids information for each
point</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of antigen laboratory IDs
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='agReactivityAdjustments'>Get and set antigen reactivity adjustments</h2><span id='topic+agReactivityAdjustments'></span><span id='topic+agReactivityAdjustments+3C-'></span>

<h3>Description</h3>

<p>Get and set antigen reactivity adjustments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agReactivityAdjustments(map)

agReactivityAdjustments(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agReactivityAdjustments_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="agReactivityAdjustments_+3A_value">value</code></td>
<td>
<p>A vector of antigen reactivity adjustments to apply</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of antigen reactivity adjustments
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='agSequences'>Getting and setting antigen sequence information</h2><span id='topic+agSequences'></span><span id='topic+agSequences+3C-'></span><span id='topic+agNucleotideSequences'></span><span id='topic+agNucleotideSequences+3C-'></span>

<h3>Description</h3>

<p>Getting and setting antigen sequence information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agSequences(map, missing_value = ".")

agSequences(map) &lt;- value

agNucleotideSequences(map, missing_value = ".")

agNucleotideSequences(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agSequences_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="agSequences_+3A_missing_value">missing_value</code></td>
<td>
<p>Character to use to fill in portions of the sequence matrix
where sequence data is missing.</p>
</td></tr>
<tr><td><code id="agSequences_+3A_value">value</code></td>
<td>
<p>A character matrix of sequences with rows equal to the number of
antigens</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character matrix of sequences, where each row represents an antigen.
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='applyMapTransform'>Apply the current map transformation</h2><span id='topic+applyMapTransform'></span>

<h3>Description</h3>

<p>Applies the map transformation associated with a selected optimization run to
a set of coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>applyMapTransform(coords, map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyMapTransform_+3A_coords">coords</code></td>
<td>
<p>Coordinates to transform</p>
</td></tr>
<tr><td><code id="applyMapTransform_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="applyMapTransform_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with transformation applied
</p>


<h3>See Also</h3>

<p>Other functions relating to map transformation: 
<code><a href="#topic+reflectMap">reflectMap</a>()</code>,
<code><a href="#topic+rotateMap">rotateMap</a>()</code>,
<code><a href="#topic+translateMap">translateMap</a>()</code>
</p>

<hr>
<h2 id='applyPlotspec'>Apply a plotspec from another acmap</h2><span id='topic+applyPlotspec'></span>

<h3>Description</h3>

<p>Copy point style from matching antigens and sera in another acmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>applyPlotspec(map, source_map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyPlotspec_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="applyPlotspec_+3A_source_map">source_map</code></td>
<td>
<p>An acmap object from which to copy point styles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the acmap object with updated point styles (unmatched point
styles unchanged)
</p>


<h3>See Also</h3>

<p>Other map point style functions: 
<code><a href="#topic+ptDrawingOrder">ptDrawingOrder</a>()</code>,
<code><a href="#topic+ptOpacity">ptOpacity</a></code>,
<code><a href="#topic+ptStyles">ptStyles</a></code>
</p>

<hr>
<h2 id='as.json'>Convert map to json format</h2><span id='topic+as.json'></span>

<h3>Description</h3>

<p>Convert map to json format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.json(map, pretty = FALSE, round_titers = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.json_+3A_map">map</code></td>
<td>
<p>The map data object</p>
</td></tr>
<tr><td><code id="as.json_+3A_pretty">pretty</code></td>
<td>
<p>Should json be output prettily with new lines and indentation?</p>
</td></tr>
<tr><td><code id="as.json_+3A_round_titers">round_titers</code></td>
<td>
<p>Should titers be rounded to the nearest integer before outputting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns map data as .ace json format
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='blob'>Plot a blob object</h2><span id='topic+blob'></span>

<h3>Description</h3>

<p>Plot a blob object such as that return from <code>agBootstrapBlob()</code> using the
<code>polygon()</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blob(x, col, border, lwd, alpha = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blob_+3A_x">x</code></td>
<td>
<p>The blob object to plot</p>
</td></tr>
<tr><td><code id="blob_+3A_col">col</code></td>
<td>
<p>Color for the blob fill</p>
</td></tr>
<tr><td><code id="blob_+3A_border">border</code></td>
<td>
<p>Color for the blob outline</p>
</td></tr>
<tr><td><code id="blob_+3A_lwd">lwd</code></td>
<td>
<p>Line width for the blob outline</p>
</td></tr>
<tr><td><code id="blob_+3A_alpha">alpha</code></td>
<td>
<p>Blob opacity</p>
</td></tr>
<tr><td><code id="blob_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>polygon()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for the side effect of plotting the blobs.
</p>


<h3>See Also</h3>

<p>Other additional plotting functions: 
<code><a href="#topic+blobsize">blobsize</a>()</code>
</p>

<hr>
<h2 id='blobsize'>Calculate size of a blob object</h2><span id='topic+blobsize'></span>

<h3>Description</h3>

<p>Returns either the area (for 2D blobs) or volume (for 3D blobs)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blobsize(blob)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blobsize_+3A_blob">blob</code></td>
<td>
<p>The blob object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector
</p>


<h3>See Also</h3>

<p>Other additional plotting functions: 
<code><a href="#topic+blob">blob</a>()</code>
</p>

<hr>
<h2 id='bootstrapBlobs'>Calculate bootstrap blob data for an antigenic map</h2><span id='topic+bootstrapBlobs'></span>

<h3>Description</h3>

<p>This function takes a map for which the function <code>bootstrapMap()</code> has already
been applied and draws contour blobs for each point illustrating how point
position varies in each bootstrap repeat. The blobs are calculated using
kernal density estimates according to these point distribution and drawn
so as to encompass a given proportion of this variation according to the
parameter <code>conf.level</code>. A <code>conf.level</code> set at 0.95 for example will draw
blobs that are calculated to encompass 95% of the positional variation seen
in the bootstrap repeats. Note however that the accuracy of these estimates
will depend on the number of bootstrap repeats performed, for example whether
100 or 1000 repeats were performed in the initial calculations using
<code>bootstrapMap()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrapBlobs(
  map,
  conf.level = 0.68,
  smoothing = 6,
  gridspacing = 0.25,
  antigens = TRUE,
  sera = TRUE,
  method = "ks"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootstrapBlobs_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="bootstrapBlobs_+3A_conf.level">conf.level</code></td>
<td>
<p>The proportion of positional variation captured by each blob</p>
</td></tr>
<tr><td><code id="bootstrapBlobs_+3A_smoothing">smoothing</code></td>
<td>
<p>The amount of smoothing to perform when performing the
kernel density estimate, larger equates to more smoothing</p>
</td></tr>
<tr><td><code id="bootstrapBlobs_+3A_gridspacing">gridspacing</code></td>
<td>
<p>grid spacing to use when calculating blobs, smaller values
will produce more accurate blobs with smoother edges but will take longer
to calculate.</p>
</td></tr>
<tr><td><code id="bootstrapBlobs_+3A_antigens">antigens</code></td>
<td>
<p>Should blobs be calculated for antigens</p>
</td></tr>
<tr><td><code id="bootstrapBlobs_+3A_sera">sera</code></td>
<td>
<p>Should blobs be calculated for sera</p>
</td></tr>
<tr><td><code id="bootstrapBlobs_+3A_method">method</code></td>
<td>
<p>One of &quot;MASS&quot;, the default, or &quot;ks&quot;, specifying the algorithm to
use when calculating blobs in 2D. 3D will always use ks::kde.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an acmap object that will then show the corresponding bootstrap
blobs when viewed or plotted.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='bootstrapMap'>Perform a bootstrap on a map</h2><span id='topic+bootstrapMap'></span>

<h3>Description</h3>

<p>This function takes the map and original titer table, and performs a version
of <a href="https://en.wikipedia.org/wiki/Bootstrapping_(statistics)">bootstrapping</a>
defined by the method argument. For each bootstrap run
this process is performed and a record of the coordinates of points in the
lowest stress solution is kept. See details for a description of the bootstrapping
methods you can apply.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrapMap(
  map,
  method,
  bootstrap_repeats = 1000,
  bootstrap_ags = TRUE,
  bootstrap_sr = TRUE,
  reoptimize = TRUE,
  optimizations_per_repeat = 100,
  ag_noise_sd = 0.7,
  titer_noise_sd = 0.7,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootstrapMap_+3A_map">map</code></td>
<td>
<p>The map object</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_method">method</code></td>
<td>
<p>One of &quot;resample&quot;, &quot;bayesian&quot; or &quot;noisy&quot; (see details)</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_bootstrap_repeats">bootstrap_repeats</code></td>
<td>
<p>The number of bootstrap repeats to perform</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_bootstrap_ags">bootstrap_ags</code></td>
<td>
<p>For &quot;resample&quot; and &quot;bayesian&quot; methods, whether to apply bootstrapping across antigens</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_bootstrap_sr">bootstrap_sr</code></td>
<td>
<p>For &quot;resample&quot; and &quot;bayesian&quot; methods, whether to apply bootstrapping across sera</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_reoptimize">reoptimize</code></td>
<td>
<p>Should the whole map be reoptimized with each bootstrap run. If FALSE,
the map is simply relaxed from it's current optimization with each run.</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_optimizations_per_repeat">optimizations_per_repeat</code></td>
<td>
<p>When re-optimizing the map from scratch, the
number of optimization runs to perform</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_ag_noise_sd">ag_noise_sd</code></td>
<td>
<p>The standard deviation (on the log titer scale) of measurement noise
applied per antigen when using the &quot;noisy&quot; method</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_titer_noise_sd">titer_noise_sd</code></td>
<td>
<p>The standard deviation (on the log titer scale) of measurement noise
applied per titer when using the &quot;noisy&quot; method</p>
</td></tr>
<tr><td><code id="bootstrapMap_+3A_options">options</code></td>
<td>
<p>Map optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Details</h3>



<h4>Bootstrapping methods</h4>

<p><strong>&quot;resample&quot;</strong>:
The <a href="https://en.wikipedia.org/wiki/Bootstrapping_(statistics)#Case_resampling">resample bootstrap</a>
is the most standard bootstrap method, a random resample of the titer table data is
taken <em>with replacement</em>. Depending on your specification, resampling is applied across
either individual antigens, individual sera or both antigens and sera.
In essence this method tries to let you see how robust the map is to inclusion of
particular titer measurements or antigens or sera. Like most bootstrapping techniques it
will prove give more reliable results the more antigens and sera you have in your map. It
won't work very well for a map of 5 sera and antigens for example, in this case a &quot;noisy&quot;
bootstrap may be better.
</p>
<p><strong>&quot;bayesian&quot;</strong>:
The <a href="https://en.wikipedia.org/wiki/Bootstrapping_(statistics)#Bayesian_bootstrap">bayesian bootstrap</a>
is akin to the resampling bootstrap, but rather than explicitly resampling data, weights are
assigned to each part of the titer table data according to random draws from a dirichilet distribution.
Under this scheme, every data point will play at least some role in making the map, even if only
weighted slightly. Sometimes this is helpful, if you know for example that the points in your map
are highly dependent upon the presence of a few antigens / sera / titers to achieve reasonable
triangulation of point positions and you don't really want to risk removing them completely and
ending up with bootstrap runs that are under-constrained, you might want to consider this approach.
On the other hand this might be exactly what you don't want and you want to know uncertainty that
can be generated when certain subsets of the data are excluded completely, in that case you probably
want to stick with the &quot;resample&quot; method.
</p>
<p><strong>&quot;noisy&quot;</strong>:
The noisy bootstrap, sometimes termed a
<a href="https://en.wikipedia.org/wiki/Bootstrapping_(statistics)#Smooth_bootstrap">smooth bootstrap</a>
involved adding normally distributed noise to each observation. The distribution of this noise can
be parameterised through the <code>ag_noise_sd</code> and <code>titer_noise_sd</code> arguments. <code>titer_noise_sd</code> refers to the
standard deviation (on the log scale) of noise added to each individual titer measurement in the table,
while <code>antigen_noise_sd</code> refers to the standard deviation of noise applied to titers for each antigen.
The reason for this distinction is that we have noticed with repeat measurements of influenza data there
is often both a random noise per titer and a random noise per antigen, i.e. in one repeat titers may all
be around one 2-fold higher on average, in addition to unbiased additional titer noise. If you wish to only
simulate additional noise per titer and not a per antigen effect, simply set <code>antigen_noise_sd</code> to 0. Note
that in order to use this most effectively it is best to have an idea of the amount and type of measurement
noise you may expect in your data and set these parameters accordingly.
</p>



<h3>Value</h3>

<p>Returns the map object updated with bootstrap information
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='checkHemisphering'>Check for hemisphering or trapped points</h2><span id='topic+checkHemisphering'></span>

<h3>Description</h3>

<p>Check for hemisphering or trapped points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkHemisphering(
  map,
  optimization_number = 1,
  grid_spacing = 0.25,
  stress_lim = 0.1,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkHemisphering_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="checkHemisphering_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization number</p>
</td></tr>
<tr><td><code id="checkHemisphering_+3A_grid_spacing">grid_spacing</code></td>
<td>
<p>When doing a grid search of more optimal point positions
the grid spacing to use</p>
</td></tr>
<tr><td><code id="checkHemisphering_+3A_stress_lim">stress_lim</code></td>
<td>
<p>The stress difference to use when classifying a point as
&quot;hemisphering&quot; or not</p>
</td></tr>
<tr><td><code id="checkHemisphering_+3A_options">options</code></td>
<td>
<p>A named list of options to pass to <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with information on any points that were found
to be hemisphering or trapped.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='colBases'>Getting and setting column bases</h2><span id='topic+colBases'></span><span id='topic+minColBasis'></span><span id='topic+minColBasis+3C-'></span><span id='topic+fixedColBases'></span><span id='topic+fixedColBases+3C-'></span>

<h3>Description</h3>

<p>Functions to get and set column bases specified for an optimization run,
either through the minimum column basis or through a vector of specified
column bases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minColBasis(map, optimization_number = 1)
minColBasis(map, optimization_number = 1) &lt;- value
fixedColBases(map, optimization_number = 1)
fixedColBases(map, optimization_number = 1) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colBases_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="colBases_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization run from which to get / set the data</p>
</td></tr>
<tr><td><code id="colBases_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In general a map can have column bases that are specified either
through a minimum column basis or a vector of fixed column bases for each
sera. When you call <code>minColBasis()</code>, it will return the minimum column
basis if it has been set, or &quot;fixed&quot; if column bases have instead been
fixed directly. The <code>colBases()</code> function will return the column bases as
calculated for a given optimization run. Setting column bases through this
function with <code style="white-space: pre;">&#8288;colBases()&lt;-&#8288;</code> will fix the column bases to the supplied
vector of values.
</p>
<p>Note that although the output from <code>colBases()</code> might be the same in a case
where a minimum column basis was set or a case where column bases were set
explicitly, when a minimum column basis is set, the column bases will still
depend on the log titers recorded against a given sera, so changing the
titers may therefore change the actual column bases calculated. For fixed
column bases case, column bases will remain fixed at their values
independently of measured titers.
</p>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p>Other map optimization attribute functions: 
<code><a href="#topic+mapComment">mapComment</a>()</code>,
<code><a href="#topic+mapDimensions">mapDimensions</a>()</code>,
<code><a href="#topic+mapStress">mapStress</a>()</code>,
<code><a href="#topic+mapTransformation">mapTransformation</a>()</code>,
<code><a href="#topic+ptBaseCoords">ptBaseCoords</a>()</code>,
<code><a href="#topic+ptCoords">ptCoords</a>()</code>
</p>

<hr>
<h2 id='deprecated_functions'>Deprecated functions</h2><span id='topic+deprecated_functions'></span><span id='topic+stressBlobs'></span>

<h3>Description</h3>

<p>These functions still work but have been deprecated in favour of another function. Arguments will be passed onto the new function with a warning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stressBlobs(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deprecated_functions_+3A_...">...</code></td>
<td>
<p>Arguments to pass to the new function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Values from the new function
</p>

<hr>
<h2 id='dilutionStepsize'>Get or set the dilution stepsize associated with a map</h2><span id='topic+dilutionStepsize'></span><span id='topic+dilutionStepsize+3C-'></span>

<h3>Description</h3>

<p>This defaults to 1 but can be changed using this function with knock-on
effects for how &lt; values are treated when maps are optimized or relaxed and
the way stress is calculated, see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dilutionStepsize(map)

dilutionStepsize(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dilutionStepsize_+3A_map">map</code></td>
<td>
<p>The acmap object from which to get or set the dilution stepsize</p>
</td></tr>
<tr><td><code id="dilutionStepsize_+3A_value">value</code></td>
<td>
<p>The dilution stepsize value to set</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Antigenic cartography was originally developed for HI titers which
typically follow a 2-fold dilution series starting from 1/10, then 1/20,
1/40 etc. This represents a &quot;dilution stepsize&quot; of 1 when converted to the
log2 scale. When no inhibition was recorded at the highest dilution, the
value is typically recorded as &lt;10 but the optimization regime effectively
treats this as a &lt;=5, the rationale being that, had the dilution series been
continued to higher concentrations, the next lowest titer would have been a
5. Over time the method has also been applied to other neutralization
assays that sometimes have a continuous read out with a lower end, in these
cases a &lt;10 really means a &lt;10 since any other values like 9.8 or 7.62
would also be possible. To indicate these continuous cases, you can specify
the dilution stepsize as 0. Equally, if the dilution regime followed a
different pattern, you can also set that here.
</p>


<h3>Value</h3>

<p>A number giving the current dilution stepsize setting for a map.
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='dimensionTestMap'>Perform dimension testing on a map object</h2><span id='topic+dimensionTestMap'></span>

<h3>Description</h3>

<p>Take a map object and perform cross-validation, seeing how well titers are
predicted when they are excluded from the map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dimensionTestMap(
  map,
  dimensions_to_test = 1:5,
  test_proportion = 0.1,
  minimum_column_basis = "none",
  fixed_column_bases = rep(NA, numSera(map)),
  number_of_optimizations = 1000,
  replicates_per_dimension = 100,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimensionTestMap_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_dimensions_to_test">dimensions_to_test</code></td>
<td>
<p>A numeric vector of dimensions to be tested</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_test_proportion">test_proportion</code></td>
<td>
<p>The proportion of data to be used as the test set for
each test run</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_minimum_column_basis">minimum_column_basis</code></td>
<td>
<p>The minimum column basis to use</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_fixed_column_bases">fixed_column_bases</code></td>
<td>
<p>A vector of fixed column bases with NA for sera
where the minimum column basis should be applied</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_number_of_optimizations">number_of_optimizations</code></td>
<td>
<p>The number of optimizations to perform when
creating each map for the dimension test</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_replicates_per_dimension">replicates_per_dimension</code></td>
<td>
<p>The number of tests to perform per dimension
tested</p>
</td></tr>
<tr><td><code id="dimensionTestMap_+3A_options">options</code></td>
<td>
<p>Map optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each run, the ag-sr titers that were randomly excluded are
predicted according to their relative positions in the map trained without
them. An RMSE is then calculated by comparing predicted titers inferred
from the map on the log scale to the actual log titers. This is done
separately for detectable titers (e.g. 40) and non-detectable titers (e.g.
&lt;10). For non-detectable titers, if the predicted titer is the same or
lower than the log-titer threshold, the error is set to 0.
</p>


<h3>Value</h3>

<p>Returns a data frame with the following columns. &quot;dimensions&quot; : the
dimension tested, &quot;mean_rmse_detectable&quot; : mean prediction rmse for
detectable titers across all runs. &quot;var_rmse_detectable&quot; the variance of
the prediction rmse for detectable titers across all runs, useful for
estimating confidence intervals. &quot;mean_rmse_nondetectable&quot; and
&quot;var_rmse_nondetectable&quot; the equivalent for non-detectable titers
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='edit_agNames'>Edit antigen names in an acmap</h2><span id='topic+edit_agNames'></span>

<h3>Description</h3>

<p>Edit antigen names in an acmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edit_agNames(map, old_names, new_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edit_agNames_+3A_map">map</code></td>
<td>
<p>The map data object to be updated</p>
</td></tr>
<tr><td><code id="edit_agNames_+3A_old_names">old_names</code></td>
<td>
<p>Old names to be replaced</p>
</td></tr>
<tr><td><code id="edit_agNames_+3A_new_names">new_names</code></td>
<td>
<p>Replacement for old names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the acmap object with antigen names updated.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='edit_srNames'>Edit sera names in an acmap</h2><span id='topic+edit_srNames'></span>

<h3>Description</h3>

<p>Edit sera names in an acmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edit_srNames(map, old_names, new_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edit_srNames_+3A_map">map</code></td>
<td>
<p>The map data object to be updated</p>
</td></tr>
<tr><td><code id="edit_srNames_+3A_old_names">old_names</code></td>
<td>
<p>Old names to be replaced</p>
</td></tr>
<tr><td><code id="edit_srNames_+3A_new_names">new_names</code></td>
<td>
<p>Replacement for old names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the acmap object with sera names updated.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='export_viewer'>Export the map viewer</h2><span id='topic+export_viewer'></span>

<h3>Description</h3>

<p>Export a map in a standalone html viewer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_viewer(map, file, selfcontained = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_viewer_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="export_viewer_+3A_file">file</code></td>
<td>
<p>File to save HTML into</p>
</td></tr>
<tr><td><code id="export_viewer_+3A_selfcontained">selfcontained</code></td>
<td>
<p>Whether to save the HTML as a single self-contained file
(with external resources base64 encoded) or a file with external resources
placed in an adjacent directory.</p>
</td></tr>
<tr><td><code id="export_viewer_+3A_...">...</code></td>
<td>
<p>Further parameters to <code>view()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Called for the side effect of saving the viewer to an html file but
invisibly returns the map viewer htmlwidget.
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='getOptimization'>Get optimization details from an acmap object</h2><span id='topic+getOptimization'></span>

<h3>Description</h3>

<p>Gets the details associated with the currently selected or specified acmap
optimization as a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOptimization(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getOptimization_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="getOptimization_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization data to access</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with information about the optimization
</p>


<h3>See Also</h3>

<p>See <code>listOptimizations()</code> for getting information about all
optimizations.
</p>

<hr>
<h2 id='ggplot.acmap'>Plot an antigenic map using ggplot</h2><span id='topic+ggplot.acmap'></span>

<h3>Description</h3>

<p>Method for plotting an antigenic map as a ggplot object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'acmap'
ggplot(
  data = NULL,
  mapping = NULL,
  optimization_number = 1,
  xlim = NULL,
  ylim = NULL,
  plot_ags = TRUE,
  plot_sr = TRUE,
  plot_blobs = TRUE,
  plot_hemisphering = TRUE,
  show_procrustes = TRUE,
  show_error_lines = FALSE,
  plot_stress = FALSE,
  indicate_outliers = "arrowheads",
  grid.col = "grey90",
  grid.lwd = 0.5,
  grid.margin.col = "grey50",
  grid.margin.lwd = grid.lwd,
  fill.alpha = 0.8,
  outline.alpha = 0.8,
  padding = 1,
  arrow_angle = 25,
  arrow_length = 0.2,
  margins = rep(0.5, 4),
  ...,
  environment = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot.acmap_+3A_data">data</code></td>
<td>
<p>The acmap to plot</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for plot, not currently used</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number to plot</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_xlim">xlim</code></td>
<td>
<p>optional x axis limits</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_ylim">ylim</code></td>
<td>
<p>optional y axis limits</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_plot_ags">plot_ags</code></td>
<td>
<p>logical, should antigens be plotted</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_plot_sr">plot_sr</code></td>
<td>
<p>logical, should antigens be plotted</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_plot_blobs">plot_blobs</code></td>
<td>
<p>logical, should stress blobs be plotted if present</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_plot_hemisphering">plot_hemisphering</code></td>
<td>
<p>logical, should hemisphering points be indicated, if
tested for already with <code>checkHemisphering()</code> (and if present)</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_show_procrustes">show_procrustes</code></td>
<td>
<p>logical, should procrustes lines be shown, if present</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_show_error_lines">show_error_lines</code></td>
<td>
<p>logical, should error lines be drawn</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_plot_stress">plot_stress</code></td>
<td>
<p>logical, should map stress be plotted in lower left corner</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_indicate_outliers">indicate_outliers</code></td>
<td>
<p>how should points outside the plotting region be
indicated, either FALSE, for not shown, or &quot;arrowheads&quot; for small arrowheads
like in the viewer.</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_grid.col">grid.col</code></td>
<td>
<p>grid line color</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_grid.lwd">grid.lwd</code></td>
<td>
<p>grid line width</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_grid.margin.col">grid.margin.col</code></td>
<td>
<p>grid margin color</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_grid.margin.lwd">grid.margin.lwd</code></td>
<td>
<p>grid margin line width</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_fill.alpha">fill.alpha</code></td>
<td>
<p>alpha for point fill</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_outline.alpha">outline.alpha</code></td>
<td>
<p>alpha for point outline</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_padding">padding</code></td>
<td>
<p>padding at limits of the antigenic map, ignored if xlim or
ylim set explicitly</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_arrow_angle">arrow_angle</code></td>
<td>
<p>angle of arrow heads drawn for procrustes lines</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_arrow_length">arrow_length</code></td>
<td>
<p>length of arrow heads drawn for procrustes lines in cm</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_margins">margins</code></td>
<td>
<p>margins in inches for the plot</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_...">...</code></td>
<td>
<p>additional arguments, not used</p>
</td></tr>
<tr><td><code id="ggplot.acmap_+3A_environment">environment</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the ggplot plot
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='htmlAdjustedTiterTable'>Return an html formatted titer table with antigen reactivity adjustments applied</h2><span id='topic+htmlAdjustedTiterTable'></span>

<h3>Description</h3>

<p>Prints an html formatted titer table, visualising
with colors things like which titers are the maximum for each sera.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>htmlAdjustedTiterTable(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="htmlAdjustedTiterTable_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="htmlAdjustedTiterTable_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number from which to take the
antigen reactivity adjustments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list() with a Rac_html_merge_report and shiny.tag class that can
be converted into an HTML string via as.character() and saved to a file
with save_html().
</p>

<hr>
<h2 id='htmlMergeReport'>Return an html formatted merge report</h2><span id='topic+htmlMergeReport'></span>

<h3>Description</h3>

<p>Prints an html formatted table merge report of a set of merged maps, visualising
with colors how different titers have been merged together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>htmlMergeReport(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="htmlMergeReport_+3A_map">map</code></td>
<td>
<p>An acmap object that was the result of merging several maps</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list() with a Rac_html_merge_report and shiny.tag class that can
be converted into an HTML string via as.character() and saved to a file
with save_html().
</p>


<h3>See Also</h3>

<p>Other map merging functions: 
<code><a href="#topic+RacMerge.options">RacMerge.options</a>()</code>,
<code><a href="#topic+mergeMaps">mergeMaps</a>()</code>,
<code><a href="#topic+mergeReport">mergeReport</a>()</code>,
<code><a href="#topic+splitTiterLayers">splitTiterLayers</a>()</code>
</p>

<hr>
<h2 id='htmlTiterTable'>Return an html formatted titer table</h2><span id='topic+htmlTiterTable'></span>

<h3>Description</h3>

<p>Prints an html formatted titer table, visualising
with colors things like which titers are the maximum for each sera.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>htmlTiterTable(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="htmlTiterTable_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list() with a Rac_html_merge_report and shiny.tag class that can
be converted into an HTML string via as.character() and saved to a file
with save_html().
</p>


<h3>See Also</h3>

<p>htmlAdjustedTiterTable
</p>

<hr>
<h2 id='keepBestOptimization'>Keep only the lowest stress map optimization</h2><span id='topic+keepBestOptimization'></span>

<h3>Description</h3>

<p>Keep only the lowest stress map optimization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>keepBestOptimization(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="keepBestOptimization_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with only the lowest stress optimization kept
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='keepOptimizations'>Keep specified optimization runs</h2><span id='topic+keepOptimizations'></span>

<h3>Description</h3>

<p>Keep only data from specified optimization runs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>keepOptimizations(map, optimization_numbers)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="keepOptimizations_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="keepOptimizations_+3A_optimization_numbers">optimization_numbers</code></td>
<td>
<p>Optimizations to keep</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the updated acmap object
</p>


<h3>See Also</h3>

<p>Other functions to work with map optimizations: 
<code><a href="#topic+optimizationProperties">optimizationProperties</a></code>,
<code><a href="#topic+removeOptimizations">removeOptimizations</a>()</code>,
<code><a href="#topic+sortOptimizations">sortOptimizations</a>()</code>
</p>

<hr>
<h2 id='keepSingleOptimization'>Keep only a single optimization run</h2><span id='topic+keepSingleOptimization'></span>

<h3>Description</h3>

<p>Keep only a single optimization run
</p>


<h3>Usage</h3>

<pre><code class='language-R'>keepSingleOptimization(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="keepSingleOptimization_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="keepSingleOptimization_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization run to keep</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with only one optimization kept
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='layerNames'>Get and set map layer names</h2><span id='topic+layerNames'></span><span id='topic+layerNames+3C-'></span>

<h3>Description</h3>

<p>Get and set map layer names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layerNames(map)

layerNames(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="layerNames_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="layerNames_+3A_value">value</code></td>
<td>
<p>A vector of new layer names to apply to the map</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of layer names
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='listOptimizations'>Get all optimization details from an acmap object</h2><span id='topic+listOptimizations'></span>

<h3>Description</h3>

<p>Gets the details associated with the all the optimizations of an acmap object
as a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listOptimizations(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listOptimizations_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of lists with information about the optimizations
</p>


<h3>See Also</h3>

<p>See <code>getOptimization()</code> for getting information about a single
optimization.
</p>

<hr>
<h2 id='logtiterTable'>Get the log titers from an acmap</h2><span id='topic+logtiterTable'></span>

<h3>Description</h3>

<p>Converts titers to the log scale via via the transformation
$log2(x/10)$, lessthan values are reduced by 1 on the log scale and greater
than values are increased by 1, hence &lt;10 =&gt; -1 and &gt;1280 =&gt; 8
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logtiterTable(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logtiterTable_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of titers converted to the log scale
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='logtiterTableLayers'>Return a list of logtiter table layers</h2><span id='topic+logtiterTableLayers'></span>

<h3>Description</h3>

<p>Return a list of logtiter table layers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logtiterTableLayers(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logtiterTableLayers_+3A_map">map</code></td>
<td>
<p>An acmap data object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric matrices with logtiter values
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='make.acmap'>Make an antigenic map from scratch</h2><span id='topic+make.acmap'></span>

<h3>Description</h3>

<p>This is a wrapper function for first making a map with table data then,
running optimizations to make the map otherwise done with <code>acmap()</code>
followed by <code>optimizeMap()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.acmap(
  titer_table = NULL,
  ag_names = NULL,
  sr_names = NULL,
  number_of_dimensions = 2,
  number_of_optimizations = 100,
  minimum_column_basis = "none",
  fixed_column_bases = NULL,
  sort_optimizations = TRUE,
  check_convergence = TRUE,
  verbose = TRUE,
  options = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.acmap_+3A_titer_table">titer_table</code></td>
<td>
<p>A table of titer data</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_ag_names">ag_names</code></td>
<td>
<p>A vector of antigen names</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_sr_names">sr_names</code></td>
<td>
<p>A vector of sera names</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_number_of_dimensions">number_of_dimensions</code></td>
<td>
<p>The number of dimensions in the map</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_number_of_optimizations">number_of_optimizations</code></td>
<td>
<p>The number of optimization runs to perform</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_minimum_column_basis">minimum_column_basis</code></td>
<td>
<p>The minimum column basis for the map</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_fixed_column_bases">fixed_column_bases</code></td>
<td>
<p>A vector of fixed values to use as column bases
directly, rather than calculating them from the titer table.</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_sort_optimizations">sort_optimizations</code></td>
<td>
<p>Should optimizations be sorted by stress
afterwards?</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_check_convergence">check_convergence</code></td>
<td>
<p>Should a basic check for convergence of lowest stress
optimization runs onto a similar solution be performed.</p>
</td></tr>
<tr><td><code id="make.acmap_+3A_verbose">verbose</code></td>
<td>
<p>Should progress messages be reported, see also
<code>RacOptimizer.options()</code></p>
</td></tr>
<tr><td><code id="make.acmap_+3A_options">options</code></td>
<td>
<p>List of named optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
<tr><td><code id="make.acmap_+3A_...">...</code></td>
<td>
<p>Further arguments to pass to <code>acmap()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an acmap object that has optimization run results.
</p>


<h3>See Also</h3>

<p>Other map optimization functions: 
<code><a href="#topic+RacOptimizer.options">RacOptimizer.options</a>()</code>,
<code><a href="#topic+moveTrappedPoints">moveTrappedPoints</a>()</code>,
<code><a href="#topic+optimizeMap">optimizeMap</a>()</code>,
<code><a href="#topic+randomizeCoords">randomizeCoords</a>()</code>,
<code><a href="#topic+relaxMapOneStep">relaxMapOneStep</a>()</code>,
<code><a href="#topic+relaxMap">relaxMap</a>()</code>
</p>

<hr>
<h2 id='map-table-distances'>Plot map vs table distances</h2><span id='topic+map-table-distances'></span><span id='topic+plot_map_table_distance'></span><span id='topic+plotly_map_table_distance'></span>

<h3>Description</h3>

<p>Plot map vs table distances
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_map_table_distance(
  map,
  optimization_number = 1,
  xlim,
  ylim,
  line_of_equality = TRUE
)

plotly_map_table_distance(
  map,
  optimization_number = 1,
  xlim,
  ylim,
  line_of_equality = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map-table-distances_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="map-table-distances_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number from which to take map and
table distances</p>
</td></tr>
<tr><td><code id="map-table-distances_+3A_xlim">xlim</code></td>
<td>
<p>The x limits of the plot</p>
</td></tr>
<tr><td><code id="map-table-distances_+3A_ylim">ylim</code></td>
<td>
<p>The y limits of the plot</p>
</td></tr>
<tr><td><code id="map-table-distances_+3A_line_of_equality">line_of_equality</code></td>
<td>
<p>Should the line x=y be added</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the ggplot2 object
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='mapBootstrapCoords'>Get bootstrap coordinates associated with a map</h2><span id='topic+mapBootstrapCoords'></span><span id='topic+mapBootstrap_ptBaseCoords'></span><span id='topic+mapBootstrap_agCoords'></span><span id='topic+mapBootstrap_srCoords'></span>

<h3>Description</h3>

<p>This can be used to get information about the bootstrap run results
after <code>bootstrapMap()</code> has been run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapBootstrap_ptBaseCoords(map)

mapBootstrap_agCoords(map)

mapBootstrap_srCoords(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapBootstrapCoords_+3A_map">map</code></td>
<td>
<p>The map object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of coordinate matrices for the points in each of
the bootstrap runs
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='mapComment'>Get or set an optimization run comment</h2><span id='topic+mapComment'></span><span id='topic+mapComment+3C-'></span>

<h3>Description</h3>

<p>Get or set an optimization run comment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapComment(map, optimization_number = 1)
mapComment(map, optimization_number = 1) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapComment_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapComment_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization run from which to get / set the data</p>
</td></tr>
<tr><td><code id="mapComment_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Gets or sets map comments for the optimization run.
</p>


<h3>See Also</h3>

<p>Other map optimization attribute functions: 
<code><a href="#topic+colBases">colBases</a>()</code>,
<code><a href="#topic+mapDimensions">mapDimensions</a>()</code>,
<code><a href="#topic+mapStress">mapStress</a>()</code>,
<code><a href="#topic+mapTransformation">mapTransformation</a>()</code>,
<code><a href="#topic+ptBaseCoords">ptBaseCoords</a>()</code>,
<code><a href="#topic+ptCoords">ptCoords</a>()</code>
</p>

<hr>
<h2 id='mapDescription'>Getting and setting the map description</h2><span id='topic+mapDescription'></span><span id='topic+mapDescription+3C-'></span>

<h3>Description</h3>

<p>Getting and setting the map description
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapDescription(map)
mapDescription(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapDescription_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapDescription_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='mapDimensions'>Get the current map dimensions</h2><span id='topic+mapDimensions'></span>

<h3>Description</h3>

<p>Get the current map dimensions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapDimensions(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapDimensions_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapDimensions_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization run from which to get / set the data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the number of dimensions for the optimization run.
</p>


<h3>See Also</h3>

<p>Other map optimization attribute functions: 
<code><a href="#topic+colBases">colBases</a>()</code>,
<code><a href="#topic+mapComment">mapComment</a>()</code>,
<code><a href="#topic+mapStress">mapStress</a>()</code>,
<code><a href="#topic+mapTransformation">mapTransformation</a>()</code>,
<code><a href="#topic+ptBaseCoords">ptBaseCoords</a>()</code>,
<code><a href="#topic+ptCoords">ptCoords</a>()</code>
</p>

<hr>
<h2 id='mapDistances'>Return calculated map distances for an acmap</h2><span id='topic+mapDistances'></span>

<h3>Description</h3>

<p>Takes the acmap object and calculates euclidean distances between antigens
and sera for the currently selected or specified optimization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapDistances(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapDistances_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapDistances_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of map distances with antigens as rows and sera as
columns.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='mapGadget'>Open a shiny gadget to view the map</h2><span id='topic+mapGadget'></span>

<h3>Description</h3>

<p>This function is equivalent to running <code>runGUI()</code> and loading a map file, but this
takes the acmap object to open as an input argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapGadget(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapGadget_+3A_map">map</code></td>
<td>
<p>The acmap object to open in the GUI</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value returned, called for the side effect of starting the gadget.
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='mapName'>Getting and setting the map name</h2><span id='topic+mapName'></span><span id='topic+mapName+3C-'></span>

<h3>Description</h3>

<p>Getting and setting the map name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapName(map)
mapName(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapName_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapName_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='mapRelaxed'>Check if a map has been fully relaxed</h2><span id='topic+mapRelaxed'></span>

<h3>Description</h3>

<p>Checks if the map optimization run can be relaxed further.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapRelaxed(map, optimization_number = 1, options = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapRelaxed_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapRelaxed_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization number</p>
</td></tr>
<tr><td><code id="mapRelaxed_+3A_options">options</code></td>
<td>
<p>List of named optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns TRUE or FALSE
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='mapResiduals'>Get a table of residuals from an acmap</h2><span id='topic+mapResiduals'></span>

<h3>Description</h3>

<p>This is the difference between the table distance and the map distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapResiduals(map, exclude_nd = FALSE, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapResiduals_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="mapResiduals_+3A_exclude_nd">exclude_nd</code></td>
<td>
<p>Should values associated with non-detectable measurements
like &lt;10 be set to NA</p>
</td></tr>
<tr><td><code id="mapResiduals_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of residuals, showing the residual error between
map distance and table distance for each antigen-sera pair.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='mapStress'>Calculate the current map stress</h2><span id='topic+mapStress'></span>

<h3>Description</h3>

<p>Calculate the current map stress
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapStress(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapStress_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="mapStress_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number for which to calculate stress</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A number giving the map stress
</p>


<h3>See Also</h3>

<p>Other map optimization attribute functions: 
<code><a href="#topic+colBases">colBases</a>()</code>,
<code><a href="#topic+mapComment">mapComment</a>()</code>,
<code><a href="#topic+mapDimensions">mapDimensions</a>()</code>,
<code><a href="#topic+mapTransformation">mapTransformation</a>()</code>,
<code><a href="#topic+ptBaseCoords">ptBaseCoords</a>()</code>,
<code><a href="#topic+ptCoords">ptCoords</a>()</code>
</p>

<hr>
<h2 id='mapTransformation'>Reading map transformation data</h2><span id='topic+mapTransformation'></span><span id='topic+mapTransformation+3C-'></span><span id='topic+mapTranslation'></span><span id='topic+mapTranslation+3C-'></span>

<h3>Description</h3>

<p>These functions can be used to query and if necessary set the map
transformation and map translation attributes for a given optimization run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapTransformation(map, optimization_number = 1)
mapTransformation(map, optimization_number = 1) &lt;- value
mapTranslation(map, optimization_number = 1)
mapTranslation(map, optimization_number = 1) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapTransformation_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="mapTransformation_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization run from which to get / set the data</p>
</td></tr>
<tr><td><code id="mapTransformation_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p>Other map optimization attribute functions: 
<code><a href="#topic+colBases">colBases</a>()</code>,
<code><a href="#topic+mapComment">mapComment</a>()</code>,
<code><a href="#topic+mapDimensions">mapDimensions</a>()</code>,
<code><a href="#topic+mapStress">mapStress</a>()</code>,
<code><a href="#topic+ptBaseCoords">ptBaseCoords</a>()</code>,
<code><a href="#topic+ptCoords">ptCoords</a>()</code>
</p>

<hr>
<h2 id='matchStrains'>Find matching antigens or sera between 2 maps</h2><span id='topic+matchStrains'></span><span id='topic+match_mapAntigens'></span><span id='topic+match_mapSera'></span>

<h3>Description</h3>

<p>Find matching antigens or sera between 2 maps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_mapAntigens(map1, map2)

match_mapSera(map1, map2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matchStrains_+3A_map1">map1</code></td>
<td>
<p>The map to match names from.</p>
</td></tr>
<tr><td><code id="matchStrains_+3A_map2">map2</code></td>
<td>
<p>The map to match names to.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the indices of matching strains in map 2, or NA in the
position of strains not found.
</p>


<h3>See Also</h3>

<p>Other functions to compare maps: 
<code><a href="#topic+procrustesData">procrustesData</a>()</code>,
<code><a href="#topic+procrustesMap">procrustesMap</a>()</code>,
<code><a href="#topic+realignMap">realignMap</a>()</code>,
<code><a href="#topic+realignOptimizations">realignOptimizations</a>()</code>
</p>

<hr>
<h2 id='mergeMaps'>Merging maps</h2><span id='topic+mergeMaps'></span>

<h3>Description</h3>

<p>Functions to merge together two tables or maps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeMaps(
  ...,
  method = "table",
  number_of_dimensions,
  number_of_optimizations,
  minimum_column_basis = "none",
  optimizer_options = list(),
  merge_options = list(),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergeMaps_+3A_...">...</code></td>
<td>
<p>acmaps to merge provided as either a list, or a series of separate arguments</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_method">method</code></td>
<td>
<p>The merge method to use, see details.</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_number_of_dimensions">number_of_dimensions</code></td>
<td>
<p>For merging that generates new optimization runs,
the number of dimensions.</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_number_of_optimizations">number_of_optimizations</code></td>
<td>
<p>For merging that generates new optimization
runs, the number of optimization runs to do.</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_minimum_column_basis">minimum_column_basis</code></td>
<td>
<p>For merging that generates new optimization runs,
the minimum column basis to use.</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_optimizer_options">optimizer_options</code></td>
<td>
<p>For merging that generates new optimization runs, optimizer
settings (see <code>RacOptimizer.options()</code>).</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_merge_options">merge_options</code></td>
<td>
<p>Options to use when merging titers (see <code>RacMerge.options()</code>).</p>
</td></tr>
<tr><td><code id="mergeMaps_+3A_verbose">verbose</code></td>
<td>
<p>Should progress messages be output?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Maps can be merged in a number of ways depending upon the desired
result.
</p>


<h4>Method 'table'</h4>

<p> As you would expect, this merges the tables of
the two maps but does not attempt to create any new optimizations and any
existing optimizations are lost. </p>



<h4>Method 'reoptimized-merge'</h4>

<p> This merges the tables and then
does a specified number of fresh optimizations from random starting
coordinates, ignoring any pre-existing optimization runs. It's exactly the
same as doing a 'table' merge and running <code>optimizeMap()</code> on the merged
table. </p>



<h4>Method 'incremental-merge'</h4>

<p> This takes the currently selected
optimization in the first map and then merges in the additional maps in
turn. Each time any points not already found in the first map (or the last
map in the incremental merge chain) are randomised and everything is
relaxed, this is repeated the specified number of times and the process is
repeated. </p>



<h4>Method 'frozen-overlay'</h4>

<p> This fixes the positions of points in
each map and tries to best match them simply through re-orientation. Once
the best re-orientation is found, points that are in common between the
maps are moved to the average position. </p>



<h4>Method 'relaxed-overlay'</h4>

<p> This is the same as the
frozen-overlay but points in the resulting map are then allowed to relax. </p>



<h4>Method 'frozen-merge'</h4>

<p> In this version, positions of all
points in the first map are fixed and remain fixed, so the original map
does not change. The second map is then realigned to the first as closely
as possible and then all the new points appearing in the second map are
allowed to relax into their new positions. This is a way to merge in new
antigens and sera into a map without affecting the first one at all (and
was first implemented in lisp). </p>



<h3>Value</h3>

<p>Returns the merged map object
</p>


<h3>See Also</h3>

<p>Other map merging functions: 
<code><a href="#topic+RacMerge.options">RacMerge.options</a>()</code>,
<code><a href="#topic+htmlMergeReport">htmlMergeReport</a>()</code>,
<code><a href="#topic+mergeReport">mergeReport</a>()</code>,
<code><a href="#topic+splitTiterLayers">splitTiterLayers</a>()</code>
</p>

<hr>
<h2 id='mergeReport'>Return a merge report</h2><span id='topic+mergeReport'></span>

<h3>Description</h3>

<p>Prints a raw text merge report from merging two map tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeReport(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergeReport_+3A_map">map</code></td>
<td>
<p>An acmap object that was the result of merging several maps</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character matrix of information on merged titers.
</p>


<h3>See Also</h3>

<p>Other map merging functions: 
<code><a href="#topic+RacMerge.options">RacMerge.options</a>()</code>,
<code><a href="#topic+htmlMergeReport">htmlMergeReport</a>()</code>,
<code><a href="#topic+mergeMaps">mergeMaps</a>()</code>,
<code><a href="#topic+splitTiterLayers">splitTiterLayers</a>()</code>
</p>

<hr>
<h2 id='moveTrappedPoints'>Move trapped points</h2><span id='topic+moveTrappedPoints'></span>

<h3>Description</h3>

<p>Sometimes points in a map optimization run get trapped in local optima, this
function tries to combat this by doing a grid search for each point
individually moving points if a better optima is found. Note that this only
performs grid searches individually so won't find cases where a group of
points are trapped together in a local optima.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moveTrappedPoints(
  map,
  optimization_number = 1,
  grid_spacing = 0.25,
  max_iterations = 10,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moveTrappedPoints_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="moveTrappedPoints_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization number to apply it to</p>
</td></tr>
<tr><td><code id="moveTrappedPoints_+3A_grid_spacing">grid_spacing</code></td>
<td>
<p>Grid spacing in antigenic units of the search grid to use
when searching for more optimal positions</p>
</td></tr>
<tr><td><code id="moveTrappedPoints_+3A_max_iterations">max_iterations</code></td>
<td>
<p>The maximum number of iterations of searching for
trapped points then relaxing the map to be performed</p>
</td></tr>
<tr><td><code id="moveTrappedPoints_+3A_options">options</code></td>
<td>
<p>List of named optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The search is iterative, searching for and moving points that are found to be
trapped before relaxing the map and searching again, stopping either when no
more trapped points are found or <code>max_iterations</code> is reached.
</p>


<h3>Value</h3>

<p>Returns the acmap object with updated coordinates (if any trapped
points found)
</p>


<h3>See Also</h3>

<p>Other map optimization functions: 
<code><a href="#topic+RacOptimizer.options">RacOptimizer.options</a>()</code>,
<code><a href="#topic+make.acmap">make.acmap</a>()</code>,
<code><a href="#topic+optimizeMap">optimizeMap</a>()</code>,
<code><a href="#topic+randomizeCoords">randomizeCoords</a>()</code>,
<code><a href="#topic+relaxMapOneStep">relaxMapOneStep</a>()</code>,
<code><a href="#topic+relaxMap">relaxMap</a>()</code>
</p>

<hr>
<h2 id='optimizationProperties'>Get optimization properties</h2><span id='topic+optimizationProperties'></span><span id='topic+allMapStresses'></span><span id='topic+allMapDimensions'></span>

<h3>Description</h3>

<p>Utility functions to get a vector of all the map optimization
properties.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allMapStresses(map)

allMapDimensions(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimizationProperties_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of values
</p>


<h3>See Also</h3>

<p>Other functions to work with map optimizations: 
<code><a href="#topic+keepOptimizations">keepOptimizations</a>()</code>,
<code><a href="#topic+removeOptimizations">removeOptimizations</a>()</code>,
<code><a href="#topic+sortOptimizations">sortOptimizations</a>()</code>
</p>

<hr>
<h2 id='optimizeAgReactivity'>Optimize antigen reactivity adjustments</h2><span id='topic+optimizeAgReactivity'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimizeAgReactivity(
  map,
  optimization_number = 1,
  reactivity_stress_weighting = 1,
  fixed_ag_reactivities = rep(NA, numAntigens(map)),
  start_pars = rep(0, numAntigens(map)),
  reoptimize = FALSE,
  number_of_optimizations = 100,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimizeAgReactivity_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number for which to optimize
antigen reactivity adjustments</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_reactivity_stress_weighting">reactivity_stress_weighting</code></td>
<td>
<p>The weighting to apply when calculating
how much antigen reactivity changes should additionally contribute to
stress in the optimization regime (see details).</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_fixed_ag_reactivities">fixed_ag_reactivities</code></td>
<td>
<p>A vector of fixed antigen reactivities,
use NA values to distinguish the positions you would still like to
be optimized.</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_start_pars">start_pars</code></td>
<td>
<p>A vector of starting parameters to use for the optimizer,
you can still supply starting parameters for antigens listed in
<code>fixed_ag_reactivities</code> but they will be ignored.</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_reoptimize">reoptimize</code></td>
<td>
<p>Should the map be reoptimized from scratch (slower but
more likely to explore other optima) when testing each reactivity adjustment
or simply relaxed from it's current coordinates (default)</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_number_of_optimizations">number_of_optimizations</code></td>
<td>
<p>If reoptimizing from scratch, how many optimization
runs should be performed each time.</p>
</td></tr>
<tr><td><code id="optimizeAgReactivity_+3A_options">options</code></td>
<td>
<p>A named list of additional options to pass to
<code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The acmap object is returned with antigen reactivity adjustments
set to the value calculated in the optimizer. This can be queried with
<code>agReactivityAdjustments()</code>.
</p>

<hr>
<h2 id='optimizeMap'>Optimize an acmap</h2><span id='topic+optimizeMap'></span>

<h3>Description</h3>

<p>Take an acmap object with a table of titer data and perform optimization runs
to try and find the best arrangement of antigens and sera to represent their
antigenic similarity. Optimizations generated from each run with different
random starting conditions will be added to the acmap object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimizeMap(
  map,
  number_of_dimensions,
  number_of_optimizations,
  minimum_column_basis = "none",
  fixed_column_bases = NULL,
  titer_weights = NULL,
  sort_optimizations = TRUE,
  check_convergence = TRUE,
  verbose = TRUE,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimizeMap_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_number_of_dimensions">number_of_dimensions</code></td>
<td>
<p>The number of dimensions for the new map</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_number_of_optimizations">number_of_optimizations</code></td>
<td>
<p>The number of optimization runs to perform</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_minimum_column_basis">minimum_column_basis</code></td>
<td>
<p>The minimum column basis to use (see details)</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_fixed_column_bases">fixed_column_bases</code></td>
<td>
<p>A vector of fixed values to use as column bases
directly, rather than calculating them from the titer table.</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_titer_weights">titer_weights</code></td>
<td>
<p>An optional matrix of weights to assign each titer when optimizing</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_sort_optimizations">sort_optimizations</code></td>
<td>
<p>Should optimizations be sorted by stress
afterwards?</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_check_convergence">check_convergence</code></td>
<td>
<p>Should a basic check for convergence of lowest stress
optimization runs onto a similar solution be performed.</p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_verbose">verbose</code></td>
<td>
<p>Should progress messages be reported, see also
<code>RacOptimizer.options()</code></p>
</td></tr>
<tr><td><code id="optimizeMap_+3A_options">options</code></td>
<td>
<p>List of named optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the core function to run map optimizations. In essence, for
each optimization run, points are randomly distributed in n-dimensional
space, the L-BFGS gradient-based optimization algorithm is applied to move
points into an optimal position. Depending on the map, this may not be a
trivial optimization process and results will depend upon the starting
conditions so multiple optimization runs may be required. For a full
explanation see <code>vignette("intro-to-antigenic-cartography")</code>.
</p>


<h4>Minimum column basis and fixed column bases</h4>

<p>Fixed column bases is a vector of fixed column bases for each sera, where
NA is specified (the default) column bases will be calculated according to
the <code>minimum_column_basis</code> setting. Again for a full explanation of column
bases and what they mean see <code>vignette("intro-to-antigenic-cartography")</code>.
</p>



<h3>Value</h3>

<p>Returns the acmap object updated with new optimizations.
</p>


<h3>See Also</h3>

<p>See <code>relaxMap()</code> for optimizing a given optimization starting from
its current coordinates.
</p>
<p>Other map optimization functions: 
<code><a href="#topic+RacOptimizer.options">RacOptimizer.options</a>()</code>,
<code><a href="#topic+make.acmap">make.acmap</a>()</code>,
<code><a href="#topic+moveTrappedPoints">moveTrappedPoints</a>()</code>,
<code><a href="#topic+randomizeCoords">randomizeCoords</a>()</code>,
<code><a href="#topic+relaxMapOneStep">relaxMapOneStep</a>()</code>,
<code><a href="#topic+relaxMap">relaxMap</a>()</code>
</p>

<hr>
<h2 id='orderPoints'>Order antigens and sera</h2><span id='topic+orderPoints'></span><span id='topic+orderAntigens'></span><span id='topic+orderSera'></span>

<h3>Description</h3>

<p>Functions to change the order of antigens and sera in a map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orderAntigens(map, order)

orderSera(map, order)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orderPoints_+3A_map">map</code></td>
<td>
<p>The map data object</p>
</td></tr>
<tr><td><code id="orderPoints_+3A_order">order</code></td>
<td>
<p>The new order of points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with points reordered
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='plot.acmap'>Plot an antigenic map</h2><span id='topic+plot.acmap'></span>

<h3>Description</h3>

<p>Method for plotting an antigenic map in two dimensions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'acmap'
plot(
  x,
  optimization_number = 1,
  xlim = NULL,
  ylim = NULL,
  plot_ags = TRUE,
  plot_sr = TRUE,
  plot_labels = FALSE,
  plot_blobs = TRUE,
  point_opacity = "automatic",
  show_procrustes = TRUE,
  show_error_lines = FALSE,
  plot_stress = FALSE,
  indicate_outliers = "arrowheads",
  grid.col = "grey90",
  grid.margin.col = "grey50",
  outlier.arrow.col = grid.col,
  fill.alpha = 0.8,
  outline.alpha = 0.8,
  procrustes.lwd = 2,
  procrustes.col = "black",
  procrustes.arr.type = "triangle",
  procrustes.arr.length = 0.2,
  procrustes.arr.width = 0.15,
  label.offset = 0,
  padding = 1,
  cex = 1,
  margins = rep(0.5, 4),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.acmap_+3A_x">x</code></td>
<td>
<p>The acmap to plot</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number to plot</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_xlim">xlim</code></td>
<td>
<p>optional x axis limits</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_ylim">ylim</code></td>
<td>
<p>optional y axis limits</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_plot_ags">plot_ags</code></td>
<td>
<p>logical, should antigens be plotted</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_plot_sr">plot_sr</code></td>
<td>
<p>logical, should antigens be plotted</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_plot_labels">plot_labels</code></td>
<td>
<p>should point labels be plotted, can be true, false or
&quot;antigens&quot; or &quot;sera&quot;</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_plot_blobs">plot_blobs</code></td>
<td>
<p>logical, should stress blobs be plotted if present</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_point_opacity">point_opacity</code></td>
<td>
<p>Either &quot;automatic&quot; or &quot;fixed&quot;. &quot;fixed&quot; fixes point
opacity to match those in <code>ptFill()</code> and <code>ptOutline()</code> and will not be
altered in procrustes plots or by the fill.alpha and outline.alpha
parameters.</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_show_procrustes">show_procrustes</code></td>
<td>
<p>logical, should procrustes lines be shown, if present</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_show_error_lines">show_error_lines</code></td>
<td>
<p>logical, should error lines be drawn</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_plot_stress">plot_stress</code></td>
<td>
<p>logical, should map stress be plotted in lower left corner</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_indicate_outliers">indicate_outliers</code></td>
<td>
<p>how should points outside the plotting region be
indicated, either FALSE, for not shown, &quot;arrowheads&quot; for small arrowheads
like in the viewer, or &quot;arrows&quot; for arrows pointing from the edge of the
plot margin, default is &quot;arrowheads&quot;.</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_grid.col">grid.col</code></td>
<td>
<p>grid line color</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_grid.margin.col">grid.margin.col</code></td>
<td>
<p>grid margin color</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_outlier.arrow.col">outlier.arrow.col</code></td>
<td>
<p>outlier arrow color</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_fill.alpha">fill.alpha</code></td>
<td>
<p>alpha for point fill</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_outline.alpha">outline.alpha</code></td>
<td>
<p>alpha for point outline</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_procrustes.lwd">procrustes.lwd</code></td>
<td>
<p>procrustes arrow line width</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_procrustes.col">procrustes.col</code></td>
<td>
<p>procrustes arrow color</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_procrustes.arr.type">procrustes.arr.type</code></td>
<td>
<p>procrustes arrow type (see <code>shape::Arrows()</code>)</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_procrustes.arr.length">procrustes.arr.length</code></td>
<td>
<p>procrustes arrow length (see <code>shape::Arrows()</code>)</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_procrustes.arr.width">procrustes.arr.width</code></td>
<td>
<p>procrustes arrow width (see <code>shape::Arrows()</code>)</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_label.offset">label.offset</code></td>
<td>
<p>amount by which any point labels should be offset from
point coordinates in fractions of a character width</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_padding">padding</code></td>
<td>
<p>padding at limits of the antigenic map, ignored if xlim or
ylim set explicitly</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_cex">cex</code></td>
<td>
<p>point size expansion factor</p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_margins">margins</code></td>
<td>
<p>margins in inches for the plot, use <code>NULL</code> for default margins from <code>par("mar")</code></p>
</td></tr>
<tr><td><code id="plot.acmap_+3A_...">...</code></td>
<td>
<p>additional arguments, not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Called for the side effect of plotting the map but invisibly
returns the map object.
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='pointStress'>Get individual point stress</h2><span id='topic+pointStress'></span><span id='topic+agStress'></span><span id='topic+srStress'></span><span id='topic+srStressPerTiter'></span><span id='topic+agStressPerTiter'></span>

<h3>Description</h3>

<p>Functions to get stress associated with individual points in a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agStress(map, antigens = TRUE, optimization_number = 1)

srStress(map, sera = TRUE, optimization_number = 1)

srStressPerTiter(map, sera = TRUE, optimization_number = 1)

agStressPerTiter(map, antigens = TRUE, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pointStress_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="pointStress_+3A_antigens">antigens</code></td>
<td>
<p>Which antigens to check stress for, specified by index or
name (defaults to all antigens).</p>
</td></tr>
<tr><td><code id="pointStress_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number</p>
</td></tr>
<tr><td><code id="pointStress_+3A_sera">sera</code></td>
<td>
<p>Which sera to check stress for, specified by index or name
(defaults to all sera).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of point stresses
</p>


<h3>See Also</h3>

<p>See <code>mapStress()</code> for getting the total map stress directly.
</p>
<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='procrustesData'>Return procrustes data on a map comparison</h2><span id='topic+procrustesData'></span>

<h3>Description</h3>

<p>Returns information about how similar point positions are in two maps,
to get an idea of how similar antigenic positions are in for example
maps made from two different datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>procrustesData(
  map,
  comparison_map,
  optimization_number = 1,
  comparison_optimization_number = 1,
  antigens = TRUE,
  sera = TRUE,
  translation = TRUE,
  scaling = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="procrustesData_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_comparison_map">comparison_map</code></td>
<td>
<p>The acmap data object to procrustes against</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization to use in the procrustes
calculation (other optimization runs are discarded)</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_comparison_optimization_number">comparison_optimization_number</code></td>
<td>
<p>The optimization run int the comparison
map to compare against</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to include (specified by name or index or TRUE/FALSE
for all/none)</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_sera">sera</code></td>
<td>
<p>Sera to include (specified by name or index or TRUE/FALSE for
all/none)</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_translation">translation</code></td>
<td>
<p>Should translation be allowed</p>
</td></tr>
<tr><td><code id="procrustesData_+3A_scaling">scaling</code></td>
<td>
<p>Should scaling be allowed (generally not recommended unless
comparing maps made with different assays)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with information on antigenic distances between the
aligned maps, and the rmsd of the point differences split by antigen
points, serum points and total, or all points. The distances are a vector
matching the number of points in the main map, with NA in the position of
any points not found in the comparison map.
</p>


<h3>See Also</h3>

<p>Other functions to compare maps: 
<code><a href="#topic+matchStrains">matchStrains</a></code>,
<code><a href="#topic+procrustesMap">procrustesMap</a>()</code>,
<code><a href="#topic+realignMap">realignMap</a>()</code>,
<code><a href="#topic+realignOptimizations">realignOptimizations</a>()</code>
</p>

<hr>
<h2 id='procrustesMap'>Return procrustes information</h2><span id='topic+procrustesMap'></span>

<h3>Description</h3>

<p>Returns information from one map procrusted to another.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>procrustesMap(
  map,
  comparison_map,
  optimization_number = 1,
  comparison_optimization_number = 1,
  antigens = TRUE,
  sera = TRUE,
  translation = TRUE,
  scaling = FALSE,
  keep_optimizations = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="procrustesMap_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_comparison_map">comparison_map</code></td>
<td>
<p>The acmap data object to procrustes against</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization to use in the procrustes
calculation (other optimization runs are discarded)</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_comparison_optimization_number">comparison_optimization_number</code></td>
<td>
<p>The optimization run int the comparison
map to compare against</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to include (specified by name or index or TRUE/FALSE
for all/none)</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_sera">sera</code></td>
<td>
<p>Sera to include (specified by name or index or TRUE/FALSE for
all/none)</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_translation">translation</code></td>
<td>
<p>Should translation be allowed</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_scaling">scaling</code></td>
<td>
<p>Should scaling be allowed (generally not recommended unless
comparing maps made with different assays)</p>
</td></tr>
<tr><td><code id="procrustesMap_+3A_keep_optimizations">keep_optimizations</code></td>
<td>
<p>Should all optimization runs be kept or only the
one to which the procrustes was applied.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an acmap object with procrustes information added, which will
be shown when the map is plotted. To avoid ambiguity about which
optimization run the procrustes was applied to, only the optimization run
specified by <code>optimization_number</code> is kept in the map returned.
</p>


<h3>See Also</h3>

<p>Other functions to compare maps: 
<code><a href="#topic+matchStrains">matchStrains</a></code>,
<code><a href="#topic+procrustesData">procrustesData</a>()</code>,
<code><a href="#topic+realignMap">realignMap</a>()</code>,
<code><a href="#topic+realignOptimizations">realignOptimizations</a>()</code>
</p>

<hr>
<h2 id='ptAnnotations'>Getting and setting point annotation information</h2><span id='topic+ptAnnotations'></span><span id='topic+agAnnotations'></span><span id='topic+srAnnotations'></span><span id='topic+agAnnotations+3C-'></span><span id='topic+srAnnotations+3C-'></span>

<h3>Description</h3>

<p>Getting and setting point annotation information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agAnnotations(map)

srAnnotations(map)

agAnnotations(map) &lt;- value

srAnnotations(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptAnnotations_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="ptAnnotations_+3A_value">value</code></td>
<td>
<p>A list of character vectors with annotations information for each
point</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of point annotations.
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='ptBaseCoords'>Getting and setting base coordinates</h2><span id='topic+ptBaseCoords'></span><span id='topic+agBaseCoords'></span><span id='topic+agBaseCoords+3C-'></span><span id='topic+srBaseCoords'></span><span id='topic+srBaseCoords+3C-'></span>

<h3>Description</h3>

<p>These functions get and set the base coordinates for a given optimization
run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptBaseCoords(map, optimization_number = 1)
agBaseCoords(map, optimization_number = 1)
agBaseCoords(map, optimization_number = 1) &lt;- value
srBaseCoords(map, optimization_number = 1)
srBaseCoords(map, optimization_number = 1) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptBaseCoords_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="ptBaseCoords_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization run from which to get / set the data</p>
</td></tr>
<tr><td><code id="ptBaseCoords_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p><code>agCoords()</code>
<code>srCoords()</code>
</p>
<p>Other map optimization attribute functions: 
<code><a href="#topic+colBases">colBases</a>()</code>,
<code><a href="#topic+mapComment">mapComment</a>()</code>,
<code><a href="#topic+mapDimensions">mapDimensions</a>()</code>,
<code><a href="#topic+mapStress">mapStress</a>()</code>,
<code><a href="#topic+mapTransformation">mapTransformation</a>()</code>,
<code><a href="#topic+ptCoords">ptCoords</a>()</code>
</p>

<hr>
<h2 id='ptBootstrapBlob'>Get antigen or serum bootstrap blob information</h2><span id='topic+ptBootstrapBlob'></span><span id='topic+agBootstrapBlob'></span><span id='topic+srBootstrapBlob'></span><span id='topic+agBootstrapBlobs'></span><span id='topic+srBootstrapBlobs'></span><span id='topic+ptBootstrapBlobs'></span>

<h3>Description</h3>

<p>Get antigen or serum bootstrap blob information for plotting with the <code>blob()</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agBootstrapBlob(map, antigen, optimization_number = 1)

srBootstrapBlob(map, serum, optimization_number = 1)

agBootstrapBlobs(map, optimization_number = 1)

srBootstrapBlobs(map, optimization_number = 1)

ptBootstrapBlobs(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptBootstrapBlob_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="ptBootstrapBlob_+3A_antigen">antigen</code></td>
<td>
<p>The antigen to get the blob for</p>
</td></tr>
<tr><td><code id="ptBootstrapBlob_+3A_optimization_number">optimization_number</code></td>
<td>
<p>Optimization number from which to get blob information</p>
</td></tr>
<tr><td><code id="ptBootstrapBlob_+3A_serum">serum</code></td>
<td>
<p>The serum to get the blob for</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class &quot;blob&quot; that can be plotted using the <code>blob()</code> function.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='ptBootstrapCoords'>Get antigen or serum bootstrap coordinates information</h2><span id='topic+ptBootstrapCoords'></span><span id='topic+agBootstrapCoords'></span><span id='topic+srBootstrapCoords'></span>

<h3>Description</h3>

<p>Get antigen or serum bootstrap coordinates information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptBootstrapCoords(map, point)

agBootstrapCoords(map, antigen)

srBootstrapCoords(map, serum)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptBootstrapCoords_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="ptBootstrapCoords_+3A_point">point</code></td>
<td>
<p>The point from which to get the bootstrap coords (numbered
antigens then sera)</p>
</td></tr>
<tr><td><code id="ptBootstrapCoords_+3A_antigen">antigen</code></td>
<td>
<p>The antigen to get the bootstrap coords</p>
</td></tr>
<tr><td><code id="ptBootstrapCoords_+3A_serum">serum</code></td>
<td>
<p>The serum from which to get the bootstrap coords</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of coordinates for the point in each of the
bootstrap runs
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='ptClades'>Getting and setting point clade information</h2><span id='topic+ptClades'></span><span id='topic+agClades'></span><span id='topic+srClades'></span><span id='topic+agClades+3C-'></span><span id='topic+srClades+3C-'></span>

<h3>Description</h3>

<p>Getting and setting point clade information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agClades(map)

srClades(map)

agClades(map) &lt;- value

srClades(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptClades_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="ptClades_+3A_value">value</code></td>
<td>
<p>A list of character vectors with clade information for each
point</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of clade information.
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='ptCoords'>Getting and setting point coordinates</h2><span id='topic+ptCoords'></span><span id='topic+agCoords'></span><span id='topic+srCoords'></span><span id='topic+ptCoords+3C-'></span><span id='topic+agCoords+3C-'></span><span id='topic+srCoords+3C-'></span>

<h3>Description</h3>

<p>Getting and setting of antigen and serum coordinates in a map optimization
run (by default the currently selected one).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agCoords(map, optimization_number = 1)

srCoords(map, optimization_number = 1)

ptCoords(map, optimization_number = 1)

ptCoords(map, optimization_number = 1) &lt;- value

agCoords(map, optimization_number = 1) &lt;- value

srCoords(map, optimization_number = 1) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptCoords_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="ptCoords_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number from which to get / set
the coordinates</p>
</td></tr>
<tr><td><code id="ptCoords_+3A_value">value</code></td>
<td>
<p>A matrix of new coordinates to set</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions get and set point coordinates in a map. By default
these coordinates refer to the currently selected optimization run, unless
otherwise specified through the <code>optimization_number</code> argument.
</p>
<p>99\
want to use but you should note that the outputs are actually the map base
coordinates after the transformation and translation associated with the
optimization run has been applied (see <code>mapTransformation()</code> and
<code>mapTranslation()</code> for more details). When you set the antigen or serum
coordinates through these functions, the transformed coordinates are
&quot;baked&quot; in and the map transformation and translation are reset.
Consequently if you want to apply a transformation to all coordinates
generally, you are better off modifying the map translation and
transformation directly, as is done by functions like <code>rotateMap()</code> and
<code>translateMap()</code>.
</p>


<h3>Value</h3>

<p>Returns a matrix of point coordinates.
</p>


<h3>See Also</h3>

<p><code>agBaseCoords()</code>
<code>srBaseCoords()</code>
<code>mapTransformation()</code>
<code>mapTranslation()</code>
</p>
<p>Other map optimization attribute functions: 
<code><a href="#topic+colBases">colBases</a>()</code>,
<code><a href="#topic+mapComment">mapComment</a>()</code>,
<code><a href="#topic+mapDimensions">mapDimensions</a>()</code>,
<code><a href="#topic+mapStress">mapStress</a>()</code>,
<code><a href="#topic+mapTransformation">mapTransformation</a>()</code>,
<code><a href="#topic+ptBaseCoords">ptBaseCoords</a>()</code>
</p>

<hr>
<h2 id='ptDrawingOrder'>Get and set point drawing order in map</h2><span id='topic+ptDrawingOrder'></span><span id='topic+ptDrawingOrder+3C-'></span>

<h3>Description</h3>

<p>Point drawing order is a vector of indices defining the order in
which points should be draw when plotting or viewing a map. Points
are indexed in the same order as antigens then followed by
sera.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptDrawingOrder(map)

ptDrawingOrder(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptDrawingOrder_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="ptDrawingOrder_+3A_value">value</code></td>
<td>
<p>The point drawing order</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of point drawing order information
</p>


<h3>See Also</h3>

<p>Other map point style functions: 
<code><a href="#topic+applyPlotspec">applyPlotspec</a>()</code>,
<code><a href="#topic+ptOpacity">ptOpacity</a></code>,
<code><a href="#topic+ptStyles">ptStyles</a></code>
</p>

<hr>
<h2 id='ptLeverage'>Calculate point leverage</h2><span id='topic+ptLeverage'></span><span id='topic+agLeverage'></span><span id='topic+srLeverage'></span><span id='topic+titerLeverage'></span>

<h3>Description</h3>

<p>These functions attempt to estimate leverage of each antigen, sera or titer
by removing it from the data, relaxing the map, then calculating the rmsd of
the procrustes comparison between the original and newly relaxed map. Column
bases will be recalculated unless you have specified them as fixed with
<code>fixedColBases()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agLeverage(map, antigens = TRUE, sera = TRUE)

srLeverage(map, antigens = TRUE, sera = TRUE)

titerLeverage(map, antigens = TRUE, sera = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptLeverage_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="ptLeverage_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to include when calculating the rmsd of the
procrustes (specified by name or index or TRUE/FALSE for all/none)</p>
</td></tr>
<tr><td><code id="ptLeverage_+3A_sera">sera</code></td>
<td>
<p>Sera to include when calculating the rmsd of the procrustes
(specified by name or index or TRUE/FALSE for all/none)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector of the leverage calculated for each of the
points.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='ptOpacity'>Set point opacity in a map</h2><span id='topic+ptOpacity'></span><span id='topic+agOpacity+3C-'></span><span id='topic+srOpacity+3C-'></span>

<h3>Description</h3>

<p>These are helper functions to quickly set the opacity of points in a map,
they set both the fill and outline color opacity by modifying the fill
and outline colors to include an alpha channel for opacity. If you need
more control, for example different opacities for the fill and outline
colors, you alter the fill and outline opacities yourself, for example
with the <code>grDevices::adjustcolor()</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agOpacity(map) &lt;- value

srOpacity(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptOpacity_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="ptOpacity_+3A_value">value</code></td>
<td>
<p>A vector of opacities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of point opacities.
</p>


<h3>See Also</h3>

<p>Other map point style functions: 
<code><a href="#topic+applyPlotspec">applyPlotspec</a>()</code>,
<code><a href="#topic+ptDrawingOrder">ptDrawingOrder</a>()</code>,
<code><a href="#topic+ptStyles">ptStyles</a></code>
</p>

<hr>
<h2 id='ptStyles'>Getting and setting point plotting styles</h2><span id='topic+ptStyles'></span><span id='topic+agShown'></span><span id='topic+srShown'></span><span id='topic+agShown+3C-'></span><span id='topic+srShown+3C-'></span><span id='topic+agSize'></span><span id='topic+srSize'></span><span id='topic+agSize+3C-'></span><span id='topic+srSize+3C-'></span><span id='topic+agFill'></span><span id='topic+srFill'></span><span id='topic+agFill+3C-'></span><span id='topic+srFill+3C-'></span><span id='topic+agOutline'></span><span id='topic+srOutline'></span><span id='topic+agOutline+3C-'></span><span id='topic+srOutline+3C-'></span><span id='topic+agOutlineWidth'></span><span id='topic+srOutlineWidth'></span><span id='topic+agOutlineWidth+3C-'></span><span id='topic+srOutlineWidth+3C-'></span><span id='topic+agRotation'></span><span id='topic+srRotation'></span><span id='topic+agRotation+3C-'></span><span id='topic+srRotation+3C-'></span><span id='topic+agAspect'></span><span id='topic+srAspect'></span><span id='topic+agAspect+3C-'></span><span id='topic+srAspect+3C-'></span><span id='topic+agShape'></span><span id='topic+srShape'></span><span id='topic+agShape+3C-'></span><span id='topic+srShape+3C-'></span>

<h3>Description</h3>

<p>These functions get and set the styles to use for each point when plotting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agShown(map)
srShown(map)
agShown(map) &lt;- value
srShown(map) &lt;- value
agSize(map)
srSize(map)
agSize(map) &lt;- value
srSize(map) &lt;- value
agFill(map)
srFill(map)
agFill(map) &lt;- value
srFill(map) &lt;- value
agOutline(map)
srOutline(map)
agOutline(map) &lt;- value
srOutline(map) &lt;- value
agOutlineWidth(map)
srOutlineWidth(map)
agOutlineWidth(map) &lt;- value
srOutlineWidth(map) &lt;- value
agRotation(map)
srRotation(map)
agRotation(map) &lt;- value
srRotation(map) &lt;- value
agAspect(map)
srAspect(map)
agAspect(map) &lt;- value
srAspect(map) &lt;- value
agShape(map)
srShape(map)
agShape(map) &lt;- value
srShape(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptStyles_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="ptStyles_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p>Other map point style functions: 
<code><a href="#topic+applyPlotspec">applyPlotspec</a>()</code>,
<code><a href="#topic+ptDrawingOrder">ptDrawingOrder</a>()</code>,
<code><a href="#topic+ptOpacity">ptOpacity</a></code>
</p>

<hr>
<h2 id='ptTriangulationBlob'>Get antigen or serum triangulation blob information</h2><span id='topic+ptTriangulationBlob'></span><span id='topic+agTriangulationBlob'></span><span id='topic+srTriangulationBlob'></span><span id='topic+agTriangulationBlobs'></span><span id='topic+srTriangulationBlobs'></span><span id='topic+ptTriangulationBlobs'></span>

<h3>Description</h3>

<p>Get antigen or serum triangulation blob information for plotting with the <code>blob()</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agTriangulationBlob(map, antigen, optimization_number = 1)

srTriangulationBlob(map, serum, optimization_number = 1)

agTriangulationBlobs(map, optimization_number = 1)

srTriangulationBlobs(map, optimization_number = 1)

ptTriangulationBlobs(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptTriangulationBlob_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="ptTriangulationBlob_+3A_antigen">antigen</code></td>
<td>
<p>The antigen to get the blob for</p>
</td></tr>
<tr><td><code id="ptTriangulationBlob_+3A_optimization_number">optimization_number</code></td>
<td>
<p>Optimization number from which to get blob information</p>
</td></tr>
<tr><td><code id="ptTriangulationBlob_+3A_serum">serum</code></td>
<td>
<p>The serum to get the blob for</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class &quot;blob&quot; that can be plotted using the <code>blob()</code> function.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='Racmacs-package'>Racmacs: Antigenic Cartography Macros</h2><span id='topic+Racmacs'></span><span id='topic+Racmacs-package'></span>

<h3>Description</h3>



<h4>The acmap data object</h4>

<p>The fundamental unit of the Racmacs package is the <a href="#topic+acmap">acmap</a>
object, short for Antigenic Cartography MAP. This object contains all the
information about an antigenic map. You can read in a new acmap object from a
file with the function <a href="#topic+read.acmap">read.acmap</a> and create a new acmap object within an
R session using the <a href="#topic+acmap">acmap</a> function.
</p>
<p>Key information associated with each acmap object is summarized in the
sections below.
</p>



<h4>The titer table</h4>

<p>Each <a href="#topic+acmap">acmap</a> object is built upon a table of data of measurements the
reactivity of a set of different sera against a set of different antigens.
Typically these measurements are HI assay measurements, but it is also
possible to use other similar assays as input data.
</p>
<p>For the table of data it is always assumed that sera form the <em>columns</em> and
antigens form the <em>rows</em>, as below. You can get and set titer data with
<a href="#topic+titerTable">titerTable</a>.
</p>



<h4>Optimizations</h4>

<p>Another key component of the <a href="#topic+acmap">acmap</a> object is a list of optimizations.
While acmap objects only have one table of data, they can have many
optimizations or none at all.
</p>
<p>Each optimization has the following main attributes (see the vignette on
optimizing antigenic maps for more details on minimum column bases and
stress):
</p>

<ul>
<li> <p><strong>Antigen coordinates</strong>, the coordinates of each antigen in this
optimization.
</p>
</li>
<li> <p><strong>Sera coordinates</strong>, the coordinates of each serum in this optimization.
</p>
</li>
<li> <p><strong>Minimum column basis</strong>, the minimum column basis assumed when calculating
this optimization.
</p>
</li>
<li> <p><strong>Stress</strong>, the stress of this optimization.
</p>
</li>
<li> <p><strong>Dimensions</strong>, the number of dimensions of this optimization.
</p>
</li></ul>




<h4>Plotting styles</h4>

<p>The final type of information that is contained in the acmap object is
information on point styles when plotting. By altering these attributes you
can change the appearance of the antigen and serum points in any maps
plotted, the main ones include:
</p>

<ul>
<li><p> Size
</p>
</li>
<li><p> Shape
</p>
</li>
<li><p> Fill color
</p>
</li>
<li><p> Outline color
</p>
</li></ul>




<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Sam Wilks <a href="mailto:sw463@cam.ac.uk">sw463@cam.ac.uk</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://acorg.github.io/Racmacs/">https://acorg.github.io/Racmacs/</a>
</p>
</li>
<li> <p><a href="https://github.com/acorg/Racmacs/">https://github.com/acorg/Racmacs/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/acorg/Racmacs/issues">https://github.com/acorg/Racmacs/issues</a>
</p>
</li></ul>


<hr>
<h2 id='RacMerge.options'>Set acmap merge options</h2><span id='topic+RacMerge.options'></span>

<h3>Description</h3>

<p>This function facilitates setting options for the acmap titer merging process by
returning a list of option settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RacMerge.options(sd_limit = NULL, dilution_stepsize = 1, method = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RacMerge.options_+3A_sd_limit">sd_limit</code></td>
<td>
<p>When merging titers, titers that have a standard deviation of
this amount or greater on the log2 scale will be set to &quot;*&quot; and excluded.
Setting this to NA removes any limit. The default value will be NA, unless
the titer merge method is specified as &quot;lispmds&quot; in which case the default
is 1 and standard deviation is calculated by division by n, instead of n-1,
in order to maintain backwards compatibility with previous approaches.</p>
</td></tr>
<tr><td><code id="RacMerge.options_+3A_dilution_stepsize">dilution_stepsize</code></td>
<td>
<p>The dilution stepsize to assume when merging titers (see
<code>dilutionStepsize()</code>)</p>
</td></tr>
<tr><td><code id="RacMerge.options_+3A_method">method</code></td>
<td>
<p>The titer merging method to use, either a string of &quot;conservative&quot; or &quot;likelihood&quot;, or a user defined function. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When merging measured titers, the general approach is to take the geometric
mean and use that as the merged titer, however in particular when &lt; values
are present there are different options that can be employed. In older
versions of Racmacs, &lt; values were converted to maximum possible numeric
titer after accounting for the dilution_stepsize factor, then the geometric
mean was taken. This approach can be used by specifying the method as
&quot;likelihood&quot; since, this approach gives a very rough approximation of the
most likely mean numeric value. In contrast, the &quot;conservative&quot; method and
current default returns the highest &lt; value that satisfies all the values
that were measured. As an example merging &lt;10 and 20, (assuming
dilution_stepsize = 1) would return a value of 10 with the &quot;likelihood&quot;
method and &lt;40 with the &quot;conservative&quot; method.
</p>


<h3>Value</h3>

<p>Returns a named list of merging options
</p>


<h3>See Also</h3>

<p>Other map merging functions: 
<code><a href="#topic+htmlMergeReport">htmlMergeReport</a>()</code>,
<code><a href="#topic+mergeMaps">mergeMaps</a>()</code>,
<code><a href="#topic+mergeReport">mergeReport</a>()</code>,
<code><a href="#topic+splitTiterLayers">splitTiterLayers</a>()</code>
</p>

<hr>
<h2 id='RacOptimizer.options'>Set acmap optimization options</h2><span id='topic+RacOptimizer.options'></span>

<h3>Description</h3>

<p>This function facilitates setting options for the acmap optimizer process by
returning a list of option settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RacOptimizer.options(
  dim_annealing = FALSE,
  method = "L-BFGS",
  maxit = 1000,
  num_basis = 10,
  armijo_constant = 1e-04,
  wolfe = 0.9,
  min_gradient_norm = 1e-06,
  factr = 1e-15,
  max_line_search_trials = 50,
  min_step = 1e-20,
  max_step = 1e+20,
  num_cores = getOption("RacOptimizer.num_cores"),
  report_progress = NULL,
  ignore_disconnected = FALSE,
  progress_bar_length = options()$width
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RacOptimizer.options_+3A_dim_annealing">dim_annealing</code></td>
<td>
<p>Should dimensional annealing be performed</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_method">method</code></td>
<td>
<p>The optimization method to use</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_maxit">maxit</code></td>
<td>
<p>The maximum number of iterations to use in the optimizer</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_num_basis">num_basis</code></td>
<td>
<p>Number of memory points to be stored (default 10).</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_armijo_constant">armijo_constant</code></td>
<td>
<p>Controls the accuracy of the line search routine for determining the Armijo condition.</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_wolfe">wolfe</code></td>
<td>
<p>Parameter for detecting the Wolfe condition.</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_min_gradient_norm">min_gradient_norm</code></td>
<td>
<p>Minimum gradient norm required to continue the optimization.</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_factr">factr</code></td>
<td>
<p>Minimum relative function value decrease to continue the optimization.</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_max_line_search_trials">max_line_search_trials</code></td>
<td>
<p>The maximum number of trials for the line search (before giving up).</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_min_step">min_step</code></td>
<td>
<p>The minimum step of the line search.</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_max_step">max_step</code></td>
<td>
<p>The maximum step of the line search.</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_num_cores">num_cores</code></td>
<td>
<p>The number of cores to run in parallel when running optimizations</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_report_progress">report_progress</code></td>
<td>
<p>Should progress be reported</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_ignore_disconnected">ignore_disconnected</code></td>
<td>
<p>Should the check for disconnected points be skipped</p>
</td></tr>
<tr><td><code id="RacOptimizer.options_+3A_progress_bar_length">progress_bar_length</code></td>
<td>
<p>Progress bar length when progress is reported</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more details, for example on &quot;dimensional annealing&quot; see
<code>vignette("intro-to-antigenic-cartography")</code>. For details on optimizer
settings like <code>maxit</code> see the underlying optimizer documentation at
<a href="https://ensmallen.org/">ensmallen.org</a>.
</p>


<h3>Value</h3>

<p>Returns a named list of optimizer options
</p>


<h3>See Also</h3>

<p>Other map optimization functions: 
<code><a href="#topic+make.acmap">make.acmap</a>()</code>,
<code><a href="#topic+moveTrappedPoints">moveTrappedPoints</a>()</code>,
<code><a href="#topic+optimizeMap">optimizeMap</a>()</code>,
<code><a href="#topic+randomizeCoords">randomizeCoords</a>()</code>,
<code><a href="#topic+relaxMapOneStep">relaxMapOneStep</a>()</code>,
<code><a href="#topic+relaxMap">relaxMap</a>()</code>
</p>

<hr>
<h2 id='RacViewer'>Create a RacViewer widget</h2><span id='topic+RacViewer'></span>

<h3>Description</h3>

<p>This creates an html widget for viewing antigenic maps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RacViewer(
  map,
  show_procrustes = FALSE,
  show_group_legend = FALSE,
  options = list(),
  width = NULL,
  height = NULL,
  elementId = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RacViewer_+3A_map">map</code></td>
<td>
<p>The map data object</p>
</td></tr>
<tr><td><code id="RacViewer_+3A_show_procrustes">show_procrustes</code></td>
<td>
<p>should procrustes lines be shown</p>
</td></tr>
<tr><td><code id="RacViewer_+3A_show_group_legend">show_group_legend</code></td>
<td>
<p>Show an interactive legend detailing different
groups as set by <code>agGroups()</code> and <code>srGroups()</code></p>
</td></tr>
<tr><td><code id="RacViewer_+3A_options">options</code></td>
<td>
<p>A named list of viewer options supplied to
<code>racviewer.options()</code></p>
</td></tr>
<tr><td><code id="RacViewer_+3A_width">width</code></td>
<td>
<p>Width of the widget</p>
</td></tr>
<tr><td><code id="RacViewer_+3A_height">height</code></td>
<td>
<p>Height of the widget</p>
</td></tr>
<tr><td><code id="RacViewer_+3A_elementid">elementId</code></td>
<td>
<p>DOM element ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class htmlwidget that will intelligently print itself
into HTML in a variety of contexts including the R console, within R
Markdown documents, and within Shiny output bindings.
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='RacViewer-shiny'>Shiny bindings for RacViewer</h2><span id='topic+RacViewer-shiny'></span><span id='topic+RacViewerOutput'></span><span id='topic+renderRacViewer'></span>

<h3>Description</h3>

<p>Output and render functions for using RacViewer within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RacViewerOutput(outputId, width = "100%", height = "100%")

renderRacViewer(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RacViewer-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="RacViewer-shiny_+3A_width">width</code>, <code id="RacViewer-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="RacViewer-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a RacViewer</p>
</td></tr>
<tr><td><code id="RacViewer-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="RacViewer-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An output or render function that enables the use of the widget
within Shiny applications.
</p>


<h3>See Also</h3>

<p>Other shiny app functions: 
<code><a href="#topic+runGUI">runGUI</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>
</p>

<hr>
<h2 id='RacViewer.options'>Set viewer options</h2><span id='topic+RacViewer.options'></span>

<h3>Description</h3>

<p>This function facilitates setting racviewer options by returning a list of
option settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RacViewer.options(
  point.opacity = NA,
  viewer.controls = "hidden",
  grid.display = "static",
  grid.col = "#cfcfcf",
  background.col = "#ffffff",
  show.names = FALSE,
  show.errorlines = FALSE,
  show.connectionlines = FALSE,
  show.titers = FALSE,
  xlim = NULL,
  ylim = NULL,
  translation = c(0, 0, 0),
  rotation = c(0, 0, 0),
  zoom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RacViewer.options_+3A_point.opacity">point.opacity</code></td>
<td>
<p>Default opacity for unselected points, or &quot;inherit&quot; to take
opacity from the color values themselves.</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_viewer.controls">viewer.controls</code></td>
<td>
<p>Should viewer controls be shown or hidden by default?</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_grid.display">grid.display</code></td>
<td>
<p>For 3d maps, should the grid be fixed in the background
or enclose and rotate along with the map</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_grid.col">grid.col</code></td>
<td>
<p>Color to use for the grid shown behind the map</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_background.col">background.col</code></td>
<td>
<p>Color for the viewer background</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_show.names">show.names</code></td>
<td>
<p>Toggle name labels on, can be true or false or &quot;antigens&quot;
or &quot;sera&quot;</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_show.errorlines">show.errorlines</code></td>
<td>
<p>Toggle error lines on</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_show.connectionlines">show.connectionlines</code></td>
<td>
<p>Toggle connection lines on</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_show.titers">show.titers</code></td>
<td>
<p>Toggle titer labels on</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_xlim">xlim</code></td>
<td>
<p>x limits to zoom the plot to</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_ylim">ylim</code></td>
<td>
<p>y limits to zoom the plot to</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_translation">translation</code></td>
<td>
<p>Plot starting translation</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_rotation">rotation</code></td>
<td>
<p>Plot starting rotation as an XYZ Euler rotation</p>
</td></tr>
<tr><td><code id="RacViewer.options_+3A_zoom">zoom</code></td>
<td>
<p>Plot starting zoom factor</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named list of viewer options
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='randomizeCoords'>Randomize map coordinates</h2><span id='topic+randomizeCoords'></span>

<h3>Description</h3>

<p>Moves map coordinates back into random starting conditions, as performed
before each optimization run. The maximum table distance is calculated
then points are randomized in a box with side length equal to maximum
table distance multiplied by <code>table_dist_factor</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>randomizeCoords(map, optimization_number = 1, table_dist_factor = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="randomizeCoords_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="randomizeCoords_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization number to randomize</p>
</td></tr>
<tr><td><code id="randomizeCoords_+3A_table_dist_factor">table_dist_factor</code></td>
<td>
<p>The expansion factor for the box size in which
points are randomized.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an updated map object
</p>


<h3>See Also</h3>

<p>Other map optimization functions: 
<code><a href="#topic+RacOptimizer.options">RacOptimizer.options</a>()</code>,
<code><a href="#topic+make.acmap">make.acmap</a>()</code>,
<code><a href="#topic+moveTrappedPoints">moveTrappedPoints</a>()</code>,
<code><a href="#topic+optimizeMap">optimizeMap</a>()</code>,
<code><a href="#topic+relaxMapOneStep">relaxMapOneStep</a>()</code>,
<code><a href="#topic+relaxMap">relaxMap</a>()</code>
</p>

<hr>
<h2 id='read.acmap'>Read in acmap data from a file</h2><span id='topic+read.acmap'></span>

<h3>Description</h3>

<p>Reads an antigenic map file and converts it into an acmap data object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.acmap(
  filename,
  optimization_number = NULL,
  sort_optimizations = FALSE,
  align_optimizations = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.acmap_+3A_filename">filename</code></td>
<td>
<p>Path to the file.</p>
</td></tr>
<tr><td><code id="read.acmap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>Numeric vector of optimization runs to keep, the
default, NULL, keeps information on all optimization runs</p>
</td></tr>
<tr><td><code id="read.acmap_+3A_sort_optimizations">sort_optimizations</code></td>
<td>
<p>Should optimizations be sorted in order of stress
when the map data is read?</p>
</td></tr>
<tr><td><code id="read.acmap_+3A_align_optimizations">align_optimizations</code></td>
<td>
<p>Should optimizations be rotated and translated to
match the orientation of the first optimization as closely as possible?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the acmap data object.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='read.titerTable'>Read in a table of titer data</h2><span id='topic+read.titerTable'></span>

<h3>Description</h3>

<p>Reads in a table of titer data, converting it to a matrix of titers with
labelled column and row names. Missing titers should be represented by an
asterisk character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.titerTable(filepath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.titerTable_+3A_filepath">filepath</code></td>
<td>
<p>Path to the table of titer data</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently supported file formats are .csv and .xls and .txt
</p>


<h3>Value</h3>

<p>Returns a matrix of titers.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='realignMap'>Realign map to match another</h2><span id='topic+realignMap'></span>

<h3>Description</h3>

<p>Realigns the coordinates of a map to match a target map as closely as
possible, based on a
<a href="https://en.wikipedia.org/wiki/Procrustes_analysis">procrustes analysis</a>.
Note that all optimization runs will be separately aligned to match as
closely as possible the first optimization run of the target map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>realignMap(map, target_map, translation = TRUE, scaling = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="realignMap_+3A_map">map</code></td>
<td>
<p>The acmap to realign.</p>
</td></tr>
<tr><td><code id="realignMap_+3A_target_map">target_map</code></td>
<td>
<p>The acmap to realign to.</p>
</td></tr>
<tr><td><code id="realignMap_+3A_translation">translation</code></td>
<td>
<p>Should translation be allowed</p>
</td></tr>
<tr><td><code id="realignMap_+3A_scaling">scaling</code></td>
<td>
<p>Should scaling be allowed (generally not recommended unless
comparing maps made with different assays)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a map object aligned to the target map
</p>


<h3>See Also</h3>

<p>Other functions to compare maps: 
<code><a href="#topic+matchStrains">matchStrains</a></code>,
<code><a href="#topic+procrustesData">procrustesData</a>()</code>,
<code><a href="#topic+procrustesMap">procrustesMap</a>()</code>,
<code><a href="#topic+realignOptimizations">realignOptimizations</a>()</code>
</p>

<hr>
<h2 id='realignOptimizations'>Realigns optimizations in the map</h2><span id='topic+realignOptimizations'></span>

<h3>Description</h3>

<p>Realigns all map optimizations through rotation and translation to match
point positions as closely as possible to the first optimization run. This
is done by default when optimizing a map and makes comparing point positions
in each optimization run much easier to do by eye.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>realignOptimizations(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="realignOptimizations_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the map with realigned optimizations
</p>


<h3>See Also</h3>

<p>Other functions to compare maps: 
<code><a href="#topic+matchStrains">matchStrains</a></code>,
<code><a href="#topic+procrustesData">procrustesData</a>()</code>,
<code><a href="#topic+procrustesMap">procrustesMap</a>()</code>,
<code><a href="#topic+realignMap">realignMap</a>()</code>
</p>

<hr>
<h2 id='recalculateStress'>Recalculate the stress associated with an acmap optimization</h2><span id='topic+recalculateStress'></span>

<h3>Description</h3>

<p>Recalculates the stress associated with the currently selected or
user-specified optimization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recalculateStress(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recalculateStress_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="recalculateStress_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the recalculated map stress for a given optimization
</p>


<h3>See Also</h3>

<p>See <code>pointStress()</code> for getting the stress of individual points.
</p>
<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='reflectMap'>Reflect a map</h2><span id='topic+reflectMap'></span>

<h3>Description</h3>

<p>Reflects map coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reflectMap(map, axis = "x", optimization_number = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reflectMap_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="reflectMap_+3A_axis">axis</code></td>
<td>
<p>Axis of reflection</p>
</td></tr>
<tr><td><code id="reflectMap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number (or NULL to apply to all
optimizations)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with reflection applied
</p>


<h3>See Also</h3>

<p>Other functions relating to map transformation: 
<code><a href="#topic+applyMapTransform">applyMapTransform</a>()</code>,
<code><a href="#topic+rotateMap">rotateMap</a>()</code>,
<code><a href="#topic+translateMap">translateMap</a>()</code>
</p>

<hr>
<h2 id='relaxMap'>Relax a map</h2><span id='topic+relaxMap'></span>

<h3>Description</h3>

<p>Optimize antigen and serum positions starting from their current coordinates
in the selected or specified optimization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relaxMap(
  map,
  optimization_number = 1,
  fixed_antigens = FALSE,
  fixed_sera = FALSE,
  titer_weights = NULL,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relaxMap_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="relaxMap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number to relax</p>
</td></tr>
<tr><td><code id="relaxMap_+3A_fixed_antigens">fixed_antigens</code></td>
<td>
<p>Antigens to set fixed positions for when relaxing</p>
</td></tr>
<tr><td><code id="relaxMap_+3A_fixed_sera">fixed_sera</code></td>
<td>
<p>Sera to set fixed positions for when relaxing</p>
</td></tr>
<tr><td><code id="relaxMap_+3A_titer_weights">titer_weights</code></td>
<td>
<p>An optional matrix of weights to assign each titer when optimizing</p>
</td></tr>
<tr><td><code id="relaxMap_+3A_options">options</code></td>
<td>
<p>List of named optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an acmap object with the optimization relaxed.
</p>


<h3>See Also</h3>

<p>See <code>optimizeMap()</code> for performing new optimization runs from random
starting coordinates.
</p>
<p>Other map optimization functions: 
<code><a href="#topic+RacOptimizer.options">RacOptimizer.options</a>()</code>,
<code><a href="#topic+make.acmap">make.acmap</a>()</code>,
<code><a href="#topic+moveTrappedPoints">moveTrappedPoints</a>()</code>,
<code><a href="#topic+optimizeMap">optimizeMap</a>()</code>,
<code><a href="#topic+randomizeCoords">randomizeCoords</a>()</code>,
<code><a href="#topic+relaxMapOneStep">relaxMapOneStep</a>()</code>
</p>

<hr>
<h2 id='relaxMapOneStep'>Relax a map one step in the optimiser</h2><span id='topic+relaxMapOneStep'></span>

<h3>Description</h3>

<p>Relax a map one step in the optimiser
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relaxMapOneStep(
  map,
  optimization_number = 1,
  fixed_antigens = FALSE,
  fixed_sera = FALSE,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relaxMapOneStep_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="relaxMapOneStep_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The map optimization number</p>
</td></tr>
<tr><td><code id="relaxMapOneStep_+3A_fixed_antigens">fixed_antigens</code></td>
<td>
<p>Antigens to set fixed positions for when relaxing</p>
</td></tr>
<tr><td><code id="relaxMapOneStep_+3A_fixed_sera">fixed_sera</code></td>
<td>
<p>Sera to set fixed positions for when relaxing</p>
</td></tr>
<tr><td><code id="relaxMapOneStep_+3A_options">options</code></td>
<td>
<p>List of named optimizer options, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an updated map object
</p>


<h3>See Also</h3>

<p>Other map optimization functions: 
<code><a href="#topic+RacOptimizer.options">RacOptimizer.options</a>()</code>,
<code><a href="#topic+make.acmap">make.acmap</a>()</code>,
<code><a href="#topic+moveTrappedPoints">moveTrappedPoints</a>()</code>,
<code><a href="#topic+optimizeMap">optimizeMap</a>()</code>,
<code><a href="#topic+randomizeCoords">randomizeCoords</a>()</code>,
<code><a href="#topic+relaxMap">relaxMap</a>()</code>
</p>

<hr>
<h2 id='removeOptimizations'>Remove map optimizations</h2><span id='topic+removeOptimizations'></span>

<h3>Description</h3>

<p>Remove all optimization run data from a map object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeOptimizations(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeOptimizations_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with all optimizations removed
</p>


<h3>See Also</h3>

<p>Other functions to work with map optimizations: 
<code><a href="#topic+keepOptimizations">keepOptimizations</a>()</code>,
<code><a href="#topic+optimizationProperties">optimizationProperties</a></code>,
<code><a href="#topic+sortOptimizations">sortOptimizations</a>()</code>
</p>

<hr>
<h2 id='removePoints'>Remove antigens and sera</h2><span id='topic+removePoints'></span><span id='topic+removeAntigens'></span><span id='topic+removeSera'></span>

<h3>Description</h3>

<p>Functions to remove antigens and sera from a map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeAntigens(map, antigens)

removeSera(map, sera)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removePoints_+3A_map">map</code></td>
<td>
<p>The map data object</p>
</td></tr>
<tr><td><code id="removePoints_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to remove (specified by name or index)</p>
</td></tr>
<tr><td><code id="removePoints_+3A_sera">sera</code></td>
<td>
<p>Sera to remove (specified by name or index)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with points removed
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='rotateMap'>Rotate a map</h2><span id='topic+rotateMap'></span>

<h3>Description</h3>

<p>Apply a rotation to an antigenic map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotateMap(map, degrees, axis = NULL, optimization_number = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rotateMap_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="rotateMap_+3A_degrees">degrees</code></td>
<td>
<p>Degrees of rotation</p>
</td></tr>
<tr><td><code id="rotateMap_+3A_axis">axis</code></td>
<td>
<p>Axis of rotation (if 3D), specified as &quot;x&quot;, &quot;y&quot;, or &quot;z&quot;</p>
</td></tr>
<tr><td><code id="rotateMap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number (or NULL to apply to all
optimizations)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with rotation applied
</p>


<h3>See Also</h3>

<p>Other functions relating to map transformation: 
<code><a href="#topic+applyMapTransform">applyMapTransform</a>()</code>,
<code><a href="#topic+reflectMap">reflectMap</a>()</code>,
<code><a href="#topic+translateMap">translateMap</a>()</code>
</p>

<hr>
<h2 id='runGUI'>Open the Racmacs GUI</h2><span id='topic+runGUI'></span>

<h3>Description</h3>

<p>This function opens the Racmacs GUI in a new window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runGUI()
</code></pre>


<h3>Value</h3>

<p>Nothing returned, called only for the side effect of starting the viewer.
</p>


<h3>See Also</h3>

<p>Other shiny app functions: 
<code><a href="#topic+RacViewer-shiny">RacViewer-shiny</a></code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>
</p>

<hr>
<h2 id='save.acmap'>Save acmap data to a file</h2><span id='topic+save.acmap'></span>

<h3>Description</h3>

<p>Save acmap data to a file. The preferred extension is &quot;.ace&quot;, although
the format of the file will be a json file of map data compressed using
'xz' compression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save.acmap(
  map,
  filename,
  compress = FALSE,
  pretty = !compress,
  round_titers = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save.acmap_+3A_map">map</code></td>
<td>
<p>The acmap data object.</p>
</td></tr>
<tr><td><code id="save.acmap_+3A_filename">filename</code></td>
<td>
<p>Path to the file.</p>
</td></tr>
<tr><td><code id="save.acmap_+3A_compress">compress</code></td>
<td>
<p>Should the file be xz compressed</p>
</td></tr>
<tr><td><code id="save.acmap_+3A_pretty">pretty</code></td>
<td>
<p>Should json be output prettily with new lines and indentation</p>
</td></tr>
<tr><td><code id="save.acmap_+3A_round_titers">round_titers</code></td>
<td>
<p>Should titers be rounded when outputted (this is needed
for acmacs web and lispmds compatibility)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for the side effect of saving the map data
to the file.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='save.coords'>Save acmap coordinate data to a file</h2><span id='topic+save.coords'></span>

<h3>Description</h3>

<p>Saves acmap coordinate data of all or specified antigens and sera to a .csv
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save.coords(
  map,
  filename,
  optimization_number = 1,
  antigens = TRUE,
  sera = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save.coords_+3A_map">map</code></td>
<td>
<p>The acmap data object.</p>
</td></tr>
<tr><td><code id="save.coords_+3A_filename">filename</code></td>
<td>
<p>Path to the file.</p>
</td></tr>
<tr><td><code id="save.coords_+3A_optimization_number">optimization_number</code></td>
<td>
<p>Optimization number from which to take coordinates</p>
</td></tr>
<tr><td><code id="save.coords_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to include, either as a numeric vector of indices or
character vector of names.</p>
</td></tr>
<tr><td><code id="save.coords_+3A_sera">sera</code></td>
<td>
<p>Sera to include, either as a numeric vector of indices or
character vector of names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for the side effect of saving the
coordinate data.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='save.titerTable'>Save titer data to a file</h2><span id='topic+save.titerTable'></span>

<h3>Description</h3>

<p>Saves titer data of all or specified antigens and sera to a .csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save.titerTable(map, filename, antigens = TRUE, sera = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save.titerTable_+3A_map">map</code></td>
<td>
<p>The acmap data object.</p>
</td></tr>
<tr><td><code id="save.titerTable_+3A_filename">filename</code></td>
<td>
<p>Path to the file.</p>
</td></tr>
<tr><td><code id="save.titerTable_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to include, either as a numeric vector of indices or
character vector of names.</p>
</td></tr>
<tr><td><code id="save.titerTable_+3A_sera">sera</code></td>
<td>
<p>Sera to include, either as a numeric vector of indices or
character vector of names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for the side effect of saving the titer
data to the file.
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='setLegend'>Set acmap legend</h2><span id='topic+setLegend'></span>

<h3>Description</h3>

<p>This sets the acmap legend used when viewing a map for example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setLegend(map, legend, fill, style.bottom = "8px", style.right = "8px")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setLegend_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="setLegend_+3A_legend">legend</code></td>
<td>
<p>A character vector of legend labels</p>
</td></tr>
<tr><td><code id="setLegend_+3A_fill">fill</code></td>
<td>
<p>The fill color to be used with the boxes that appear alongside
the legend labels</p>
</td></tr>
<tr><td><code id="setLegend_+3A_style.bottom">style.bottom</code></td>
<td>
<p>&quot;bottom&quot; style of the div, specifying how far from the
bottom of the viewport the bottom of the legend is spaced.</p>
</td></tr>
<tr><td><code id="setLegend_+3A_style.right">style.right</code></td>
<td>
<p>&quot;right&quot; style of the div, specifying how far from the
right of the viewport the bottom of the legend is spaced.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the updated acmap object
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

<hr>
<h2 id='sortOptimizations'>Sort optimizations by stress</h2><span id='topic+sortOptimizations'></span>

<h3>Description</h3>

<p>Sorts all the optimization runs for a given map object by stress
(lowest to highest). Note that this is done by default when running
<code>optimizeMap()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortOptimizations(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sortOptimizations_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with optimizations sorted by stress.
</p>


<h3>See Also</h3>

<p>Other functions to work with map optimizations: 
<code><a href="#topic+keepOptimizations">keepOptimizations</a>()</code>,
<code><a href="#topic+optimizationProperties">optimizationProperties</a></code>,
<code><a href="#topic+removeOptimizations">removeOptimizations</a>()</code>
</p>

<hr>
<h2 id='splitTiterLayers'>Split a map made up from titer layers into a list of separate maps each with a titer table
corresponding to one of the layers</h2><span id='topic+splitTiterLayers'></span>

<h3>Description</h3>

<p>Split a map made up from titer layers into a list of separate maps each with a titer table
corresponding to one of the layers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitTiterLayers(map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitTiterLayers_+3A_map">map</code></td>
<td>
<p>An acmap object with titer table layers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of acmap objects
</p>


<h3>See Also</h3>

<p>Other map merging functions: 
<code><a href="#topic+RacMerge.options">RacMerge.options</a>()</code>,
<code><a href="#topic+htmlMergeReport">htmlMergeReport</a>()</code>,
<code><a href="#topic+mergeMaps">mergeMaps</a>()</code>,
<code><a href="#topic+mergeReport">mergeReport</a>()</code>
</p>

<hr>
<h2 id='srAttributes'>Getting and setting sera attributes</h2><span id='topic+srAttributes'></span><span id='topic+srIDs'></span><span id='topic+srIDs+3C-'></span><span id='topic+srDates'></span><span id='topic+srDates+3C-'></span><span id='topic+srReference'></span><span id='topic+srReference+3C-'></span><span id='topic+srNames'></span><span id='topic+srNames+3C-'></span><span id='topic+srExtra'></span><span id='topic+srExtra+3C-'></span><span id='topic+srPassage'></span><span id='topic+srPassage+3C-'></span><span id='topic+srLineage'></span><span id='topic+srLineage+3C-'></span><span id='topic+srReassortant'></span><span id='topic+srReassortant+3C-'></span><span id='topic+srStrings'></span><span id='topic+srStrings+3C-'></span><span id='topic+srSpecies'></span><span id='topic+srSpecies+3C-'></span>

<h3>Description</h3>

<p>These functions get and set the sera attributes for a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srIDs(map)
srIDs(map) &lt;- value
srDates(map)
srDates(map) &lt;- value
srReference(map)
srReference(map) &lt;- value
srNames(map)
srNames(map) &lt;- value
srExtra(map)
srExtra(map) &lt;- value
srPassage(map)
srPassage(map) &lt;- value
srLineage(map)
srLineage(map) &lt;- value
srReassortant(map)
srReassortant(map) &lt;- value
srStrings(map)
srStrings(map) &lt;- value
srSpecies(map)
srSpecies(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srAttributes_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="srAttributes_+3A_value">value</code></td>
<td>
<p>New value to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either the requested attribute when using a getter function or
</p>
<p>the updated acmap object when using the setter function.
</p>


<h3>See Also</h3>

<p><code>agAttributes()</code>
</p>
<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='srGroups'>Getting and setting sera groups</h2><span id='topic+srGroups'></span><span id='topic+srGroups+3C-'></span>

<h3>Description</h3>

<p>These functions get and set the sera groupings for a map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srGroups(map)

srGroups(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srGroups_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="srGroups_+3A_value">value</code></td>
<td>
<p>A character or factor vector of groupings to apply to the sera</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A factor vector of serum groups
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='srHomologousAgs'>Get and set homologous antigens for sera</h2><span id='topic+srHomologousAgs'></span><span id='topic+srHomologousAgs+3C-'></span>

<h3>Description</h3>

<p>Get and set indices of homologous antigens to sera in an antigenic map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srHomologousAgs(map)

srHomologousAgs(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srHomologousAgs_+3A_map">map</code></td>
<td>
<p>An acmap object</p>
</td></tr>
<tr><td><code id="srHomologousAgs_+3A_value">value</code></td>
<td>
<p>A list, where each entry is a vector of indices for homologous
antigens, or a length 0 vector where no homologous antigen is present</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, where each entry is a vector of indices for homologous
antigens, or a length 0 vector where no homologous antigen is present.
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srSequences">srSequences</a>()</code>
</p>

<hr>
<h2 id='srSequences'>Getting and setting sera sequence information</h2><span id='topic+srSequences'></span><span id='topic+srSequences+3C-'></span><span id='topic+srNucleotideSequences'></span><span id='topic+srNucleotideSequences+3C-'></span>

<h3>Description</h3>

<p>Getting and setting sera sequence information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srSequences(map, missing_value = ".")

srSequences(map) &lt;- value

srNucleotideSequences(map, missing_value = ".")

srNucleotideSequences(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srSequences_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="srSequences_+3A_missing_value">missing_value</code></td>
<td>
<p>Character to use to fill in portions of the sequence matrix
where sequence data is missing.</p>
</td></tr>
<tr><td><code id="srSequences_+3A_value">value</code></td>
<td>
<p>A character matrix of sequences with rows equal to the number of
sera</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character matrix of sequences with rows equal to the number of
sera.
</p>


<h3>See Also</h3>

<p>Other antigen and sera attribute functions: 
<code><a href="#topic+agAttributes">agAttributes</a></code>,
<code><a href="#topic+agGroups">agGroups</a>()</code>,
<code><a href="#topic+agHomologousSr">agHomologousSr</a>()</code>,
<code><a href="#topic+agLabIDs">agLabIDs</a>()</code>,
<code><a href="#topic+agSequences">agSequences</a>()</code>,
<code><a href="#topic+ptAnnotations">ptAnnotations</a></code>,
<code><a href="#topic+ptClades">ptClades</a></code>,
<code><a href="#topic+srAttributes">srAttributes</a></code>,
<code><a href="#topic+srGroups">srGroups</a>()</code>,
<code><a href="#topic+srHomologousAgs">srHomologousAgs</a>()</code>
</p>

<hr>
<h2 id='standardizeStrainNames'>Standardize strain names</h2><span id='topic+standardizeStrainNames'></span>

<h3>Description</h3>

<p>This is a utility function to help standardise antigen names into a more
consistent format, also attempting to break apart different components
of the name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardizeStrainNames(
  names,
  default_species = NA,
  default_virus_type = "A",
  default_virus_subtype = "HXNX"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="standardizeStrainNames_+3A_names">names</code></td>
<td>
<p>Strain names to be standardised</p>
</td></tr>
<tr><td><code id="standardizeStrainNames_+3A_default_species">default_species</code></td>
<td>
<p>Are the strains isolated from a particular species?</p>
</td></tr>
<tr><td><code id="standardizeStrainNames_+3A_default_virus_type">default_virus_type</code></td>
<td>
<p>Default virus type to be used (if no type found in
name)</p>
</td></tr>
<tr><td><code id="standardizeStrainNames_+3A_default_virus_subtype">default_virus_subtype</code></td>
<td>
<p>Default virus subtype to be used (if no subtype
found in name)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a tibble of standardised names and extracted information
</p>

<hr>
<h2 id='stressTable'>Get a stress table from an acmap</h2><span id='topic+stressTable'></span>

<h3>Description</h3>

<p>Get a stress table from an acmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stressTable(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stressTable_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="stressTable_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number for which to calculate
stresses</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of stresses, showing how much each antigen and sera
measurement contributes to stress in the selected or specified
optimization.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='subsetCommonPoints'>Remove antigens and sera</h2><span id='topic+subsetCommonPoints'></span><span id='topic+subsetCommonAgs'></span><span id='topic+subsetCommonSrGroups'></span>

<h3>Description</h3>

<p>Functions to subset a list of maps to include only antigens, antigen groups, sera
or serum groups that are in common between them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetCommonAgs(maps)

subsetCommonSrGroups(maps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetCommonPoints_+3A_maps">maps</code></td>
<td>
<p>A list of map data objects</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetMap">subsetMap</a>()</code>
</p>

<hr>
<h2 id='subsetMap'>Subset an antigenic map</h2><span id='topic+subsetMap'></span>

<h3>Description</h3>

<p>Subset an antigenic map to contain only specified antigens and sera
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetMap(map, antigens = TRUE, sera = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetMap_+3A_map">map</code></td>
<td>
<p>The antigenic map object</p>
</td></tr>
<tr><td><code id="subsetMap_+3A_antigens">antigens</code></td>
<td>
<p>Antigens to keep, defaults to all.</p>
</td></tr>
<tr><td><code id="subsetMap_+3A_sera">sera</code></td>
<td>
<p>Sera to keep, defaults to all.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a new antigenic map containing only match antigens and sera
</p>


<h3>See Also</h3>

<p>Other functions for working with map data: 
<code><a href="#topic+acmap">acmap</a>()</code>,
<code><a href="#topic+addOptimization">addOptimization</a>()</code>,
<code><a href="#topic+agReactivityAdjustments">agReactivityAdjustments</a>()</code>,
<code><a href="#topic+as.json">as.json</a>()</code>,
<code><a href="#topic+edit_agNames">edit_agNames</a>()</code>,
<code><a href="#topic+edit_srNames">edit_srNames</a>()</code>,
<code><a href="#topic+keepBestOptimization">keepBestOptimization</a>()</code>,
<code><a href="#topic+keepSingleOptimization">keepSingleOptimization</a>()</code>,
<code><a href="#topic+layerNames">layerNames</a>()</code>,
<code><a href="#topic+orderPoints">orderPoints</a></code>,
<code><a href="#topic+read.acmap">read.acmap</a>()</code>,
<code><a href="#topic+read.titerTable">read.titerTable</a>()</code>,
<code><a href="#topic+removePoints">removePoints</a></code>,
<code><a href="#topic+save.acmap">save.acmap</a>()</code>,
<code><a href="#topic+save.coords">save.coords</a>()</code>,
<code><a href="#topic+save.titerTable">save.titerTable</a>()</code>,
<code><a href="#topic+subsetCommonPoints">subsetCommonPoints</a></code>
</p>

<hr>
<h2 id='tableColbases'>Calculate column bases for a titer table</h2><span id='topic+tableColbases'></span>

<h3>Description</h3>

<p>For more information on column bases, what they mean and how they are
calculated see <code>vignette("intro-to-antigenic-cartography")</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableColbases(
  titer_table,
  minimum_column_basis = "none",
  fixed_column_bases = rep(NA, ncol(titer_table)),
  ag_reactivity_adjustments = rep(0, nrow(titer_table))
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tableColbases_+3A_titer_table">titer_table</code></td>
<td>
<p>The titer table</p>
</td></tr>
<tr><td><code id="tableColbases_+3A_minimum_column_basis">minimum_column_basis</code></td>
<td>
<p>The minimum column basis to assume</p>
</td></tr>
<tr><td><code id="tableColbases_+3A_fixed_column_bases">fixed_column_bases</code></td>
<td>
<p>Fixed column bases to apply</p>
</td></tr>
<tr><td><code id="tableColbases_+3A_ag_reactivity_adjustments">ag_reactivity_adjustments</code></td>
<td>
<p>Reactivity adjustments to apply on a per-antigen basis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector of the log-converted column bases for the
table
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>
</p>

<hr>
<h2 id='tableDistances'>Return calculated table distances for an acmap</h2><span id='topic+tableDistances'></span>

<h3>Description</h3>

<p>Takes the acmap object and, assuming the column bases associated with the
currently selected or specified optimization, returns the table distances
calculated from the titer data. For more information on column bases and
their role in antigenic cartography see
<code>vignette("intro-to-antigenic-cartography")</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableDistances(map, optimization_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tableDistances_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="tableDistances_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of numeric table distances
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>
<p>Other functions relating to map stress calculation: 
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>
</p>

<hr>
<h2 id='titerTable'>Getting and setting map titers</h2><span id='topic+titerTable'></span><span id='topic+titerTable+3C-'></span>

<h3>Description</h3>

<p>Functions to get and set the map titer table. Note that when setting the
titer table like this any titer table layer information is lost, this is
normally not a problem unless the map is a result of merging two titer tables
together previously and you then go on the merge the titers again.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>titerTable(map)

titerTable(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="titerTable_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="titerTable_+3A_value">value</code></td>
<td>
<p>A character matrix of titers to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character matrix of titers.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+adjustedTiterTable">adjustedTiterTable()</a></code>, <code><a href="#topic+htmlTiterTable">htmlTiterTable()</a></code>
</p>
<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>
</p>

<hr>
<h2 id='titerTableFlat'>Getting and setting the flat titer table</h2><span id='topic+titerTableFlat'></span><span id='topic+titerTableFlat+3C-'></span>

<h3>Description</h3>

<p>These are underlying functions to get and set the &quot;flat&quot; version of the titer
table only. When a map is merged, the titer tables are merged but a record of
the original titers associated with each map are kept as titer table layers
so that information on the original set of titers that made up the merge is
not lost. At the same time, the merged titer version of the titer table is
created and saved as the titer_table_flat attribute. When you access titers
through the <code>titerTable()</code> function, the flat version of the titer table is
retrieved (only really a relevant distinction for merged maps). When you set
titers through <code style="white-space: pre;">&#8288;titerTable&lt;-()&#8288;</code> titer table layers are lost. These functions
allow you to manipulate the flat version without affecting the titer table
layers information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>titerTableFlat(map)

titerTableFlat(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="titerTableFlat_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="titerTableFlat_+3A_value">value</code></td>
<td>
<p>A character matrix of titers to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character matrix of titers.
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableLayers">titerTableLayers</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='titerTableLayers'>Getting and setting titer table layers</h2><span id='topic+titerTableLayers'></span><span id='topic+titerTableLayers+3C-'></span>

<h3>Description</h3>

<p>Functions to get and set the underlying titer table layers of a map (see
details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>titerTableLayers(map)

titerTableLayers(map) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="titerTableLayers_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="titerTableLayers_+3A_value">value</code></td>
<td>
<p>A list of titer table character vectors to set</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When you merge maps with <code>mergeMaps()</code> repeated antigen - serum
titers are merged to create a new titer table but information on the
original titers is not lost. The original titer tables, aligned to their
new positions in the merged table, are kept as separate layers that can be
accessed with these functions. If you have merged a whole bunch of
different maps, these functions can be useful to check for example,
variation in titer seen between a single antigen and serum pair.
</p>


<h3>Value</h3>

<p>A list of character matrices of titers.
</p>


<h3>See Also</h3>

<p>Other map attribute functions: 
<code><a href="#topic+acmapAttributes">acmapAttributes</a></code>,
<code><a href="#topic+adjustedLogTiterTable">adjustedLogTiterTable</a>()</code>,
<code><a href="#topic+adjustedTiterTable">adjustedTiterTable</a>()</code>,
<code><a href="#topic+dilutionStepsize">dilutionStepsize</a>()</code>,
<code><a href="#topic+logtiterTableLayers">logtiterTableLayers</a>()</code>,
<code><a href="#topic+mapDescription">mapDescription</a>()</code>,
<code><a href="#topic+mapName">mapName</a>()</code>,
<code><a href="#topic+titerTableFlat">titerTableFlat</a>()</code>,
<code><a href="#topic+titerTable">titerTable</a>()</code>
</p>

<hr>
<h2 id='translateMap'>Translate a map</h2><span id='topic+translateMap'></span>

<h3>Description</h3>

<p>Translates map coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translateMap(map, translation, optimization_number = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translateMap_+3A_map">map</code></td>
<td>
<p>The acmap object</p>
</td></tr>
<tr><td><code id="translateMap_+3A_translation">translation</code></td>
<td>
<p>Translation to apply (as vector or n x 1 matrix)</p>
</td></tr>
<tr><td><code id="translateMap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number (or NULL to apply to all
optimizations)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An acmap object with transformation applied
</p>


<h3>See Also</h3>

<p>Other functions relating to map transformation: 
<code><a href="#topic+applyMapTransform">applyMapTransform</a>()</code>,
<code><a href="#topic+reflectMap">reflectMap</a>()</code>,
<code><a href="#topic+rotateMap">rotateMap</a>()</code>
</p>

<hr>
<h2 id='triangulationBlobs'>Calculate triangulation blobs data for an antigenic map</h2><span id='topic+triangulationBlobs'></span>

<h3>Description</h3>

<p>This function is to help give an idea of how well coordinated each point is
in a map, and to give some idea of uncertainty in it's position. It works
by moving each point in a grid search and seeing how the total map stress
changes, see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>triangulationBlobs(
  map,
  optimization_number = 1,
  stress_lim = 1,
  grid_spacing = 0.25,
  antigens = TRUE,
  sera = TRUE,
  .check_relaxation = TRUE,
  .options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="triangulationBlobs_+3A_map">map</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number to check</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_stress_lim">stress_lim</code></td>
<td>
<p>The blob stress limit</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_grid_spacing">grid_spacing</code></td>
<td>
<p>Grid spacing to use when searching map space and
inferring the blob</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_antigens">antigens</code></td>
<td>
<p>Should triangulation blobs be calculated for antigens</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_sera">sera</code></td>
<td>
<p>Should triangulation blobs be calculated for sera</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_.check_relaxation">.check_relaxation</code></td>
<td>
<p>Should a check be performed that the map is fully
relaxed (all points in a local optima) before the search is performed</p>
</td></tr>
<tr><td><code id="triangulationBlobs_+3A_.options">.options</code></td>
<td>
<p>List of named optimizer options to use when checking map
relaxation, see <code>RacOptimizer.options()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The region or regions of the plot where total map stress is not
increased above a certain threshold (<code>stress_lim</code>) are shown when the map
is plotted. This function is really to check whether point positions are
clearly very uncertain, for example the underlying titers may support an
antigen being a certain distance away from a group of other points but due
to the positions of the sera against which it was titrated the direction
would be unclear, and you might see a blob that forms an arc or &quot;banana&quot;
that represents this. Note that it is not really a confidence interval
since a point may be well coordinated in terms of the optimization but
it's position may still be defined by perhaps only one particular titer
which is itself uncertain. For something more akin to confidence intervals
you can use other diagnostic functions like <code>bootstrapMap()</code>.
</p>


<h3>Value</h3>

<p>Returns the acmap data object with triangulation blob information added,
which will be shown when the map is plotted
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+unstableMaps">unstableMaps</a></code>
</p>

<hr>
<h2 id='unstableMaps'>Notes on unstable maps</h2><span id='topic+unstableMaps'></span>

<h3>Description</h3>

<p>Tips for exploring maps that are difficult to find a consistent optimal solution for.
</p>


<h3>Details</h3>

<p>Maps may be difficult to optimize or unstable for a variety of reasons, a common
one with larger maps being simply that it is difficult to find a global optima
and so many different local optima are found each time.
</p>
<p>One approach that can sometimes
help is to consider running the optimizer with <code>options = list(dim_annealing = TRUE)</code>
(see see <code>vignette("intro-to-antigenic-cartography")</code> for an explanation of the
dimensional annealing approach). However be wary that in our experience, while applying
dimensional annealing can sometimes significantly speed up finding a better minima, it
can also sometimes be more prone to getting stuck in worse local optima.
</p>
<p>If there are many missing or non-detectable titers it is also
possible that points in map are too poorly connected to find a robust
solution, to check this see <code>mapCohesion()</code>.
</p>


<h3>See Also</h3>

<p>Other map diagnostic functions: 
<code><a href="#topic+agCohesion">agCohesion</a>()</code>,
<code><a href="#topic+bootstrapBlobs">bootstrapBlobs</a>()</code>,
<code><a href="#topic+bootstrapMap">bootstrapMap</a>()</code>,
<code><a href="#topic+checkHemisphering">checkHemisphering</a>()</code>,
<code><a href="#topic+dimensionTestMap">dimensionTestMap</a>()</code>,
<code><a href="#topic+logtiterTable">logtiterTable</a>()</code>,
<code><a href="#topic+map-table-distances">map-table-distances</a></code>,
<code><a href="#topic+mapBootstrapCoords">mapBootstrapCoords</a></code>,
<code><a href="#topic+mapDistances">mapDistances</a>()</code>,
<code><a href="#topic+mapRelaxed">mapRelaxed</a>()</code>,
<code><a href="#topic+mapResiduals">mapResiduals</a>()</code>,
<code><a href="#topic+pointStress">pointStress</a></code>,
<code><a href="#topic+ptBootstrapBlob">ptBootstrapBlob</a></code>,
<code><a href="#topic+ptBootstrapCoords">ptBootstrapCoords</a>()</code>,
<code><a href="#topic+ptLeverage">ptLeverage</a></code>,
<code><a href="#topic+ptTriangulationBlob">ptTriangulationBlob</a></code>,
<code><a href="#topic+recalculateStress">recalculateStress</a>()</code>,
<code><a href="#topic+stressTable">stressTable</a>()</code>,
<code><a href="#topic+tableColbases">tableColbases</a>()</code>,
<code><a href="#topic+tableDistances">tableDistances</a>()</code>,
<code><a href="#topic+triangulationBlobs">triangulationBlobs</a>()</code>
</p>

<hr>
<h2 id='update_ferret_seraNames'>Update the ferret serum names to match antigens</h2><span id='topic+update_ferret_seraNames'></span>

<h3>Description</h3>

<p>Update the ferret serum names to match antigens
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_ferret_seraNames(map, dictionary_file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_ferret_seraNames_+3A_map">map</code></td>
<td>
<p>The map data object</p>
</td></tr>
<tr><td><code id="update_ferret_seraNames_+3A_dictionary_file">dictionary_file</code></td>
<td>
<p>The path to the dictionary file you want to use
(should be .csv). If not supplied then the default dictionary that comes
with Racmacs will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the updated map data object
</p>

<hr>
<h2 id='view'>S3 method for viewing objects</h2><span id='topic+view'></span>

<h3>Description</h3>

<p>S3 method for viewing objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>view(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="view_+3A_x">x</code></td>
<td>
<p>The object to view</p>
</td></tr>
<tr><td><code id="view_+3A_...">...</code></td>
<td>
<p>Additional arguments, not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>When called on an acmap object, returns an htmlwidget object that
can be used to interactively view the map. Otherwise by default it
simply calls the print method of the respective object with no return
value.
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>
</p>

<hr>
<h2 id='view.acmap'>Viewing racmap objects</h2><span id='topic+view.acmap'></span>

<h3>Description</h3>

<p>View a racmap object in the interactive viewer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'acmap'
view(
  x,
  optimization_number = 1,
  ...,
  .jsCode = NULL,
  .jsData = NULL,
  select_ags = NULL,
  select_sr = NULL,
  show_procrustes = NULL,
  show_diagnostics = NULL,
  num_optimizations = 1,
  options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="view.acmap_+3A_x">x</code></td>
<td>
<p>The acmap data object</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_optimization_number">optimization_number</code></td>
<td>
<p>The optimization number to view</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code>RacViewer()</code></p>
</td></tr>
<tr><td><code id="view.acmap_+3A_.jscode">.jsCode</code></td>
<td>
<p>Additional javascript code to be run after map has been loaded
and rendered</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_.jsdata">.jsData</code></td>
<td>
<p>Any data to supply to the .jsCode function</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_select_ags">select_ags</code></td>
<td>
<p>A vector of antigen indices to select in the plot</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_select_sr">select_sr</code></td>
<td>
<p>A vector of serum indices to select in the plot</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_show_procrustes">show_procrustes</code></td>
<td>
<p>If the map contains procrustes information, should
procrustes lines be shown by default?</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_show_diagnostics">show_diagnostics</code></td>
<td>
<p>If the map contains diagnostics information like
stress blobs or hemisphering, should it be shown by default?</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_num_optimizations">num_optimizations</code></td>
<td>
<p>Number of optimization runs to send to the viewer
for inclusion in the &quot;optimizations&quot; pane.</p>
</td></tr>
<tr><td><code id="view.acmap_+3A_options">options</code></td>
<td>
<p>A named list of viewer options to pass to
<code>RacViewer.options()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an htmlwidget object
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.default">view.default</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>
<p>Other shiny app functions: 
<code><a href="#topic+RacViewer-shiny">RacViewer-shiny</a></code>,
<code><a href="#topic+runGUI">runGUI</a>()</code>
</p>

<hr>
<h2 id='view.default'>Default method for viewing objects</h2><span id='topic+view.default'></span>

<h3>Description</h3>

<p>Default method for viewing objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
view(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="view.default_+3A_x">x</code></td>
<td>
<p>The object to view</p>
</td></tr>
<tr><td><code id="view.default_+3A_...">...</code></td>
<td>
<p>Additional arguments, passed to print.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value returned, simply calls the print method on the object
</p>


<h3>See Also</h3>

<p>Other functions to view maps: 
<code><a href="#topic+RacViewer.options">RacViewer.options</a>()</code>,
<code><a href="#topic+RacViewer">RacViewer</a>()</code>,
<code><a href="#topic+export_viewer">export_viewer</a>()</code>,
<code><a href="#topic+ggplot.acmap">ggplot.acmap</a>()</code>,
<code><a href="#topic+mapGadget">mapGadget</a>()</code>,
<code><a href="#topic+plot.acmap">plot.acmap</a>()</code>,
<code><a href="#topic+setLegend">setLegend</a>()</code>,
<code><a href="#topic+view.acmap">view.acmap</a>()</code>,
<code><a href="#topic+view">view</a>()</code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
