<!DOCTYPE html><html><head><title>Help for package RADstackshelpR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RADstackshelpR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#optimize_bigM'><p>Optimize the M parameter during denovo stacks assembly</p></a></li>
<li><a href='#optimize_m'><p>Optimize the m parameter during denovo stacks assembly</p></a></li>
<li><a href='#optimize_n'><p>Optimize the n parameter during denovo stacks assembly</p></a></li>
<li><a href='#vis_depth'><p>Visualize the effect of varying the m parameter on the mean depth of each sample</p></a></li>
<li><a href='#vis_loci'><p>Visualize the effect of varying a stacks parameter on the number of polymorphic loci retained</p></a></li>
<li><a href='#vis_snps'><p>Visualize the effect of varying a stacks parameter on the number of SNPs retained</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Optimize the De Novo Stacks Pipeline via R</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Offers a handful of useful wrapper functions which streamline
    the reading, analyzing, and visualizing of variant call format (vcf) files in R.
    This package was designed to facilitate an explicit pipeline for optimizing
    Stacks (Rochette et al., 2019) (&lt;<a href="https://doi.org/10.1111%2Fmec.15253">doi:10.1111/mec.15253</a>&gt;)
    parameters during de novo (without a reference genome) assembly and variant
    calling of restriction-enzyme associated DNA sequence (RADseq) data. The
    pipeline implemented here is based on the 2017 paper "Lost in Parameter Space"
    (Paris et al., 2017) (&lt;<a href="https://doi.org/10.1111%2F2041-210X.12775">doi:10.1111/2041-210X.12775</a>&gt;) which
    establishes clear recommendations for optimizing the parameters
    'm', 'M', and 'n', during the process of assembling loci.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>vcfR, ggplot2, ggridges, gridExtra</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-18 14:12:58 UTC; devder</td>
</tr>
<tr>
<td>Author:</td>
<td>Devon DeRaad <a href="https://orcid.org/0000-0003-3105-985X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Devon DeRaad &lt;devonderaad@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-19 09:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='optimize_bigM'>Optimize the M parameter during denovo stacks assembly</h2><span id='topic+optimize_bigM'></span>

<h3>Description</h3>

<p>This function requires the path to stacks vcf file(s) as input.
There are slots for varying the M parameter from 1-8 (as recommended by Paris et al. 2017).
After running stacks with each of the M options, plug the output vcf files into this
function to calculate the effect of varying M on the number of SNPs/loci built. Plug the output of this function into
vis_loci() to visualize the optimal the M parameter for your dataset at the 'R80' cutoff (Paris et al. 2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimize_bigM(
  M1 = NULL,
  M2 = NULL,
  M3 = NULL,
  M4 = NULL,
  M5 = NULL,
  M6 = NULL,
  M7 = NULL,
  M8 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimize_bigM_+3A_m1">M1</code></td>
<td>
<p>Path to the input vcf file for a run when M=1</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m2">M2</code></td>
<td>
<p>Path to the input vcf file for a run when M=2</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m3">M3</code></td>
<td>
<p>Path to the input vcf file for a run when M=3</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m4">M4</code></td>
<td>
<p>Path to the input vcf file for a run when M=4</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m5">M5</code></td>
<td>
<p>Path to the input vcf file for a run when M=5</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m6">M6</code></td>
<td>
<p>Path to the input vcf file for a run when M=6</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m7">M7</code></td>
<td>
<p>Path to the input vcf file for a run when M=7</p>
</td></tr>
<tr><td><code id="optimize_bigM_+3A_m8">M8</code></td>
<td>
<p>Path to the input vcf file for a run when M=8</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing four summary dataframes,
'snp' showing the number of non-missing SNPs retained in each sample at each m value, 'loci' showing the
number of non-missing loci retained in each sample at each m value, 'snp.R80' showing the total number of SNPs
retained at an 80% completeness cutoff, and 'loci.R80' showing the total number of polymorphic loci
retained at an 80% completeness cutoff.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>optimize_bigM(M1=system.file("extdata","bigM1.vcf.gz",package="RADstackshelpR",mustWork=TRUE))
</code></pre>

<hr>
<h2 id='optimize_m'>Optimize the m parameter during denovo stacks assembly</h2><span id='topic+optimize_m'></span>

<h3>Description</h3>

<p>This function requires the path to stacks vcf file(s) as input.
There are slots for varying the m parameter from 3-7 (as recommended by Paris et al. 2017).
After running stacks with each of the m options, plug the output vcf files into this
function to calculate the effect of varying m on depth and number of SNPs/loci built. Plug the output of this function into
vis_loci() to visualize the optimal the m parameter for your dataset at the 'R80' cutoff (Paris et al. 2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimize_m(m3 = NULL, m4 = NULL, m5 = NULL, m6 = NULL, m7 = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimize_m_+3A_m3">m3</code></td>
<td>
<p>Path to the input vcf file for a run when m=3</p>
</td></tr>
<tr><td><code id="optimize_m_+3A_m4">m4</code></td>
<td>
<p>Path to the input vcf file for a run when m=4</p>
</td></tr>
<tr><td><code id="optimize_m_+3A_m5">m5</code></td>
<td>
<p>Path to the input vcf file for a run when m=5</p>
</td></tr>
<tr><td><code id="optimize_m_+3A_m6">m6</code></td>
<td>
<p>Path to the input vcf file for a run when m=6</p>
</td></tr>
<tr><td><code id="optimize_m_+3A_m7">m7</code></td>
<td>
<p>Path to the input vcf file for a run when m=7</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing five summary dataframes, 'depth' showing depth per sample for each m value,
'snp' showing the number of non-missing SNPs retained in each sample at each m value, 'loci' showing the
number of non-missing loci retained in each sample at each m value, 'snp.R80' showing the total number of SNPs
retained at an 80% completeness cutoff, and 'loci.R80' showing the total number of polymorphic loci
retained at an 80% completeness cutoff.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>optimize_m(m3=system.file("extdata","m3.vcf.gz",package="RADstackshelpR",mustWork=TRUE))
</code></pre>

<hr>
<h2 id='optimize_n'>Optimize the n parameter during denovo stacks assembly</h2><span id='topic+optimize_n'></span>

<h3>Description</h3>

<p>This function requires the path to stacks vcf file(s) as input.
There are slots for varying the n parameter across M-1, M, and M-1 (as recommended by Paris et al. 2017).
After running stacks with each of the n options, plug the output vcf files into this
function to visualize the effect of varying m on number of SNPs and loci built to
recognize which value optimizes the n parameter for your dataset at the 'R80' cutoff (Paris et al. 2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimize_n(nequalsMminus1 = NULL, nequalsM = NULL, nequalsMplus1 = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimize_n_+3A_nequalsmminus1">nequalsMminus1</code></td>
<td>
<p>Path to the input vcf file for a run when n=M-1</p>
</td></tr>
<tr><td><code id="optimize_n_+3A_nequalsm">nequalsM</code></td>
<td>
<p>Path to the input vcf file for a run when n=M</p>
</td></tr>
<tr><td><code id="optimize_n_+3A_nequalsmplus1">nequalsMplus1</code></td>
<td>
<p>Path to the input vcf file for a run when n=M+1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe showing the number of SNPs and loci retained across filtering levels for each n value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>optimize_n(nequalsM =
system.file("extdata","nequalsm.vcf.gz",package="RADstackshelpR",mustWork=TRUE))
</code></pre>

<hr>
<h2 id='vis_depth'>Visualize the effect of varying the m parameter on the mean depth of each sample</h2><span id='topic+vis_depth'></span>

<h3>Description</h3>

<p>This function takes the list of dataframes output by optimize_m() as input.
The function then uses ggplot2 to visualize the effect of m on depth.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_depth(output = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_depth_+3A_output">output</code></td>
<td>
<p>A list containing 5 dataframes generated by optimize_m()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the depth of each sample at each given m value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vis_depth(output =
readRDS(system.file("extdata","optimize.m.output.RDS",package="RADstackshelpR",mustWork=TRUE)))
</code></pre>

<hr>
<h2 id='vis_loci'>Visualize the effect of varying a stacks parameter on the number of polymorphic loci retained</h2><span id='topic+vis_loci'></span>

<h3>Description</h3>

<p>This function takes the list of dataframes output by optimize_m(), optimize_M(), or optimize_n() as input.
The function then uses ggplot2 to visualize the effect of the given stacks on the number of polymorphic loci retained, reporting which value is optimal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_loci(output = NULL, stacks_param = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_loci_+3A_output">output</code></td>
<td>
<p>A list containing 5 dataframes generated by optimize_m()</p>
</td></tr>
<tr><td><code id="vis_loci_+3A_stacks_param">stacks_param</code></td>
<td>
<p>A character string indicating the stacks parameter iterated over</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the number of polymorphic loci retained at each given parameter value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vis_loci(output =
readRDS(system.file("extdata","optimize.m.output.RDS",package="RADstackshelpR",mustWork=TRUE)),
         stacks_param = "m")
</code></pre>

<hr>
<h2 id='vis_snps'>Visualize the effect of varying a stacks parameter on the number of SNPs retained</h2><span id='topic+vis_snps'></span>

<h3>Description</h3>

<p>This function takes the list of dataframes output by optimize_m(), optimize_M(), or optimize_n() as input.
The function then uses ggplot2 to visualize the effect of the given stacks on the number of SNPs retained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_snps(output = NULL, stacks_param = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_snps_+3A_output">output</code></td>
<td>
<p>A list containing 5 dataframes generated by optimize_m()</p>
</td></tr>
<tr><td><code id="vis_snps_+3A_stacks_param">stacks_param</code></td>
<td>
<p>A character string indicating the stacks parameter iterated over</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the number of SNPs retained at each given parameter value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vis_snps(output=
readRDS(system.file("extdata","optimize.m.output.RDS",package="RADstackshelpR",mustWork=TRUE)),
         stacks_param = "m")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
