<!DOCTYPE html><html><head><title>Help for package rapidphylo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rapidphylo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Human_Allele_Frequencies'><p>Allele frequencies from 31,000 single nucleotide polymorphisms</p></a></li>
<li><a href='#RDM'><p>Estimating tree-topology from allele frequency data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Rapidly Estimates Phylogeny from Large Allele Frequency Data
Using Root Distances Method</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Arindam RoyChoudhury [aut, cre, cph],
  Jing Peng [aut],
  Ying Li [aut],
  Laura Kubatko [aut, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arindam RoyChoudhury &lt;arr2014@med.cornell.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Rapidly estimates tree-topology from large allele frequency
    data using Root Distances Method, under a Brownian Motion Model. See
    Peng et al. (2021) &lt;<a href="https://doi.org/10.1016%2Fj.ympev.2021.107142">doi:10.1016/j.ympev.2021.107142</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ArindamRoyChoudhury/rapidphylo">https://github.com/ArindamRoyChoudhury/rapidphylo</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ArindamRoyChoudhury/rapidphylo/issues">https://github.com/ArindamRoyChoudhury/rapidphylo/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, phangorn, stats</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-01 16:37:06 UTC; yil4013</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-01 17:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Human_Allele_Frequencies'>Allele frequencies from 31,000 single nucleotide polymorphisms</h2><span id='topic+Human_Allele_Frequencies'></span>

<h3>Description</h3>

<p>The dataset “Human_Allele_Frequencies” is a 5 × 31,000 matrix
that contains allele frequencies from 31,000 single nucleotide
polymorphisms in Chromosomes 1-10 in 5 human populations.
The last population “San” is intended to be used as an outgroup.
The allele frequencies have been compiled from ALFRED database
at Yale University. The analysis from this dataset has been
published in Peng et al 2021.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Human_Allele_Frequencies
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 5 rows and 31000 columns.
</p>

<hr>
<h2 id='RDM'>Estimating tree-topology from allele frequency data</h2><span id='topic+RDM'></span>

<h3>Description</h3>

<p><code>RDM()</code> estimates a tree-topology from allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RDM(
  mat_allele_freq,
  outgroup,
  use = c("complete.obs", "pairwise.complete.obs", "everything", "all.obs",
    "na.or.complete")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDM_+3A_mat_allele_freq">mat_allele_freq</code></td>
<td>
<p>A <code class="reqn">(P+1) \times L</code> matrix containing the allele frequencies, where there are <code class="reqn">P</code> taxa, plus one outgroup, and <code class="reqn">L</code> loci.</p>
</td></tr>
<tr><td><code id="RDM_+3A_outgroup">outgroup</code></td>
<td>
<p>A variable that can be either the population name or a numerical row number of the outgroup data.</p>
</td></tr>
<tr><td><code id="RDM_+3A_use">use</code></td>
<td>
<p>Specify which part of data is used to compute the covariance matrix. The options are &quot;<code>complete.obs</code>&quot;, &quot;<code>pairwise.complete.obs</code>&quot;, &quot;<code>everything</code>&quot;, &quot;<code>all.obs</code>&quot;, and &quot;<code>na.or.complete</code>&quot;. See <code>stats::cov</code> for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input matrix is the observed values of the frequencies at tips <code class="reqn">1, 2, ..., P, P+1</code>.
A logit transformation is performed on the allele frequency data, so that the observed values
are approximately normal. (The logit transformation of r refers to <code class="reqn">\log\frac{r}{1-r}</code>.) The transformed matrix is converted into a data frame for further analyses.
</p>


<h3>Value</h3>

<p>An estimated tree-topology in Newick format.
</p>


<h3>References</h3>

<p>Peng J, Rajeevan H, Kubatko L, and RoyChoudhury A (2021) <em>A fast likelihood approach for estimation of large phylogenies from continuous trait data</em>. Molecular Phylogenetics and Evolution 161 107142.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A dataset "Human_Allele_Frequencies" is loaded with the package;
# it has allele frequencies in 31,000 sites for
# 4 human populations and one outgroup human population.

# check data dimension
dim(Human_Allele_Frequencies)

# run RDM function
rd_tre &lt;- RDM(Human_Allele_Frequencies, outgroup = "San", use = "pairwise.complete.obs")

# result visualization
plot(rd_tre, use.edge.length = FALSE, cex = 0.5)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
