<!DOCTYPE html><html><head><title>Help for package rBahadur</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rBahadur}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#am_covariance_structure'><p>Compute Diagonal plus Low Rank equilibrium covariance structure</p></a></li>
<li><a href='#am_equilibrium_parameters'><p>Functions to compute equilibrium parameters under assortative mating</p></a></li>
<li><a href='#am_simulate'><p>Simulate genotype/phenotype data under equilibrium univariate AM.</p></a></li>
<li><a href='#rb_dplr'><p>Binary random variates with Diagonal Plus Low Rank (dplr) correlations</p></a></li>
<li><a href='#rb_unstr'><p>Binary random variates with unstructured correlations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Assortative Mating Simulation and Multivariate Bernoulli
Variates</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulation of phenotype / genotype data under assortative mating. 
    Includes functions for generating Bahadur order-2 multivariate Bernoulli 
    variables with general and diagonal-plus-low-rank correlation structures. 
    Further details are provided in: Border and Malik (2022) 
    &lt;<a href="https://doi.org/10.1101%2F2022.10.13.512132">doi:10.1101/2022.10.13.512132</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/rborder/rBahadur">https://github.com/rborder/rBahadur</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-25 19:11:21 UTC; rsb</td>
</tr>
<tr>
<td>Author:</td>
<td>Richard Border <a href="https://orcid.org/0000-0002-6293-2968"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Osman Malik <a href="https://orcid.org/0000-0003-4477-481X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richard Border &lt;border.richard@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-25 19:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='am_covariance_structure'>Compute Diagonal plus Low Rank equilibrium covariance structure</h2><span id='topic+am_covariance_structure'></span>

<h3>Description</h3>

<p>Compute Diagonal plus Low Rank equilibrium covariance structure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>am_covariance_structure(beta, AF, r)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="am_covariance_structure_+3A_beta">beta</code></td>
<td>
<p>vector of standardized diploid allele-substitution effects</p>
</td></tr>
<tr><td><code id="am_covariance_structure_+3A_af">AF</code></td>
<td>
<p>vector of allele frequencies</p>
</td></tr>
<tr><td><code id="am_covariance_structure_+3A_r">r</code></td>
<td>
<p>cross-mate phenotypic correlation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector 'U' such that $D + U U^T$ corresponds to the expected haploid
LD-matrix given the specified genetic architecture (encoded by 'beta' and 'AF')
and cross-mate phenotypic correlation 'r'. It is assumed that the total phenotypic
variance at generation zero is one.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
h2_0 = .5; m = 200; n = 1000; r =.5; min_MAF=.1
betas &lt;- rnorm(m,0,sqrt(h2_0/m))
afs &lt;- runif(m, min_MAF, 1-min_MAF)
output &lt;- am_covariance_structure(betas, afs, r)
</code></pre>

<hr>
<h2 id='am_equilibrium_parameters'>Functions to compute equilibrium parameters under assortative mating</h2><span id='topic+am_equilibrium_parameters'></span><span id='topic+h2_eq'></span><span id='topic+rg_eq'></span><span id='topic+vg_eq'></span>

<h3>Description</h3>

<p>Compute heritability ('h2_eq'), genetic variance ('vg_0'), and cross-mate genetic
correlation ('rg_eq') at equilibrium under univariate primary-phenotypic assortative mating.
These equations can be derived from Nagylaki's results (see below) under the assumption
that number of causal variants is large (i.e., taking the limit as the number of causal
variants approaches infinity).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h2_eq(r, h2_0)

rg_eq(r, h2_0)

vg_eq(r, vg_0, h2_0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="am_equilibrium_parameters_+3A_r">r</code></td>
<td>
<p>cross-mate phenotypic correlation</p>
</td></tr>
<tr><td><code id="am_equilibrium_parameters_+3A_h2_0">h2_0</code></td>
<td>
<p>generation zero (panmictic) heritability</p>
</td></tr>
<tr><td><code id="am_equilibrium_parameters_+3A_vg_0">vg_0</code></td>
<td>
<p>generation zero (panmictic) additive genetic variance component</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single numerical quantity representing the equilibrium heritability (<code>h2_eq</code>),
the equilibrium cross-mate genetic correlation (<code>rg_eq</code>), or the equilibrium genetic
variance (<code>vg_eq</code>).
</p>


<h3>References</h3>

<p>Nagylaki, T. Assortative mating for a quantitative character. J. Math. Biology
16, 57â€“74 (1982). https://doi.org/10.1007/BF00275161
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
vg_0= .6; h2_0 = .5; r =.5
h2_eq(r, h2_0)
rg_eq(r, h2_0)
vg_eq(r, vg_0, h2_0)
</code></pre>

<hr>
<h2 id='am_simulate'>Simulate genotype/phenotype data under equilibrium univariate AM.</h2><span id='topic+am_simulate'></span>

<h3>Description</h3>

<p>Simulate genotype/phenotype data under equilibrium univariate AM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>am_simulate(h2_0, r, m, n, afs = NULL, min_MAF = 0.1, haplotypes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="am_simulate_+3A_h2_0">h2_0</code></td>
<td>
<p>generation zero (panmictic) heritability</p>
</td></tr>
<tr><td><code id="am_simulate_+3A_r">r</code></td>
<td>
<p>cross-mate phenotypic correlation</p>
</td></tr>
<tr><td><code id="am_simulate_+3A_m">m</code></td>
<td>
<p>number of biallelic causal variants</p>
</td></tr>
<tr><td><code id="am_simulate_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="am_simulate_+3A_afs">afs</code></td>
<td>
<p>(optional). Allele frequencies to use. If not provided, <code>m</code> will be drawn
uniformly from the interval [<code>min_MAF</code>, 1-<code>min_MAF</code>]</p>
</td></tr>
<tr><td><code id="am_simulate_+3A_min_maf">min_MAF</code></td>
<td>
<p>(optional) minimum minor allele frequency for causal variants.
Ignored if if <code>afs</code> is not NULL. Defaults to 0.1</p>
</td></tr>
<tr><td><code id="am_simulate_+3A_haplotypes">haplotypes</code></td>
<td>
<p>logical. If TRUE, includes (phased) haploid genotypes in output.
Defaults to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list including the following objects:
</p>

<ul>
<li> <p><code>y</code>: phenotype vector
</p>
</li>
<li> <p><code>g</code>: heritable component of the phenotype vector
</p>
</li>
<li> <p><code>X</code>: matrix of diploid genotypes
</p>
</li>
<li> <p><code>AF</code>: vector of allele frequencies
</p>
</li>
<li> <p><code>beta_std</code>: standardized genetic effects
</p>
</li>
<li> <p><code>beta_raw</code>: unstandardized genetic effects
</p>
</li>
<li> <p><code>H</code>: matrix of haploid genotypes (returned only if <code>haplotypes</code>=TRUE)
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
h2_0 = .5; m = 200; n = 1000; r =.5

## simulate genotype/phenotype data
sim_dat &lt;- am_simulate(h2_0, r, m, n)
str(sim_dat)

## empirical h2 vs expected equilibrium h2
(emp_h2 &lt;- var(sim_dat$g)/var(sim_dat$y))
h2_eq(r, h2_0)
</code></pre>

<hr>
<h2 id='rb_dplr'>Binary random variates with Diagonal Plus Low Rank (dplr) correlations</h2><span id='topic+rb_dplr'></span>

<h3>Description</h3>

<p>Generate second Bahadur order multivariate Bernoulli random variates with
Diagonal Plus Low Rank (dplr) correlation structures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rb_dplr(n, mu, U)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rb_dplr_+3A_n">n</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code id="rb_dplr_+3A_mu">mu</code></td>
<td>
<p>vector of means</p>
</td></tr>
<tr><td><code id="rb_dplr_+3A_u">U</code></td>
<td>
<p>outer product component matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This generates multivariate Bernoulli (MVB) random vectors with mean
vector 'mu' and correlation matrix <code class="reqn">C = D + U U^T</code> where <code class="reqn">D</code> is a diagonal
matrix with values dictated by 'U'. 'mu' must take values in the open unit interval
and 'U' must induce a valid second Bahadur order probability distribution. That is,
there must exist an MVB probability distribution with first moments 'mu' and
standardized central second moments <code class="reqn">C</code> such that all higher order central
moments are zero.
</p>


<h3>Value</h3>

<p>An <code class="reqn">n</code>-by-<code class="reqn">m</code> matrix of binary random variates, where <code class="reqn">m</code> is
the length of 'mu'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
h2_0 = .5; m = 200; n = 1000; r =.5; min_MAF=.1

## draw standardized diploid allele substitution effects
beta &lt;- scale(rnorm(m))*sqrt(h2_0 / m)

## draw allele frequencies
AF &lt;- runif(m, min_MAF, 1 - min_MAF)

## compute unstandardized effects
beta_unscaled &lt;- beta/sqrt(2*AF*(1-AF))

## generate corresponding haploid quantities
beta_hap &lt;- rep(beta, each=2)
AF_hap &lt;- rep(AF, each=2)

## compute equilibrium outer product covariance component
U &lt;- am_covariance_structure(beta, AF, r)

## draw multivariate Bernoulli haplotypes
H &lt;- rb_dplr(n, AF_hap, U)

## convert to diploid genotypes
G &lt;- H[,seq(1,ncol(H),2)] + H[,seq(2,ncol(H),2)]

## empirical allele frequencies vs target frequencies
emp_afs &lt;- colMeans(G)/2
plot(AF, emp_afs)

## construct phenotype
heritable_y &lt;-  G%*%beta_unscaled
nonheritable_y &lt;-  rnorm(n, 0, sqrt(1-h2_0))
y &lt;- heritable_y + nonheritable_y

## empirical h2 vs expected equilibrium h2
(emp_h2 &lt;- var(heritable_y)/var(y))
h2_eq(r, h2_0)
</code></pre>

<hr>
<h2 id='rb_unstr'>Binary random variates with unstructured correlations</h2><span id='topic+rb_unstr'></span>

<h3>Description</h3>

<p>Generate Bahadur order-2 multivariate Bernoulli random variates with
unstructured correlations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rb_unstr(n, mu, C)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rb_unstr_+3A_n">n</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code id="rb_unstr_+3A_mu">mu</code></td>
<td>
<p>vector of means</p>
</td></tr>
<tr><td><code id="rb_unstr_+3A_c">C</code></td>
<td>
<p>correlation matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This generates multivariate Bernoulli (MVB) random vectors with mean vector
'mu' and correlation matrix 'C'. 'mu' must take values in the open unit interval and
'C' must induce a valid second Bahadur order probability distribution. That is,
there must exist an MVB probability distribution with first moments 'mu' and
standardized central second moments 'C' such that all higher order central moments
are zero.
</p>


<h3>Value</h3>

<p>An <code class="reqn">n</code>-by-<code class="reqn">m</code> matrix of binary random variates, where <code class="reqn">m</code> is the
length of 'mu'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
h2_0 = .5; m = 200; n = 500; r =.5; min_MAF=.1

## draw standardized diploid allele substitution effects
beta &lt;- scale(rnorm(m))*sqrt(h2_0 / m)

## draw allele frequencies
AF &lt;- runif(m, min_MAF, 1 - min_MAF)

## compute unstandardized effects
beta_unscaled &lt;- beta/sqrt(2*AF*(1-AF))

## generate corresponding haploid quantities
beta_hap &lt;- rep(beta, each=2)
AF_hap &lt;- rep(AF, each=2)

## compute equilibrium outer product covariance component
U &lt;- am_covariance_structure(beta, AF, r)

## construct Correlation matrix
S &lt;- diag(1/sqrt(AF_hap*(1-AF_hap)))
DPLR &lt;- U%o%U
diag(DPLR) &lt;- 1
C &lt;- cov2cor(S%*%DPLR%*%S)

## draw multivariate Bernoulli haplotypes
H &lt;- rb_unstr(n, AF_hap, C)

## convert to diploid genotypes
G &lt;- H[,seq(1,ncol(H),2)] + H[,seq(2,ncol(H),2)]

## empirical allele frequencies vs target frequencies
emp_afs &lt;- colMeans(G)/2
plot(AF, emp_afs)

## construct phenotype
heritable_y &lt;-  G%*%beta_unscaled
nonheritable_y &lt;-  rnorm(n, 0, sqrt(1-h2_0))
y &lt;- heritable_y + nonheritable_y

## empirical h2 vs expected equilibrium h2
(emp_h2 &lt;- var(heritable_y)/var(y))
h2_eq(r, h2_0)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
