<!DOCTYPE html><html lang="en"><head><title>Help for package rBiasCorrection</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rBiasCorrection}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aggregated_input'><p>aggregated_input helper function</p></a></li>
<li><a href='#better_model'><p>better_model helper function</p></a></li>
<li><a href='#biascorrection'><p>Correct PCR-Bias in Quantitative DNA Methylation Analyses.</p></a></li>
<li><a href='#calibration_plot'><p>calibration_plot helper function</p></a></li>
<li><a href='#clean_dt'><p>clean_dt helper function</p></a></li>
<li><a href='#clean_up'><p>clean_up helper function</p></a></li>
<li><a href='#create_exampleplot'><p>create_exampleplot helper function</p></a></li>
<li><a href='#createbarerrorplots'><p>createbarerrorplots helper function</p></a></li>
<li><a href='#example._plot_coef_c'><p>example._plot_coef_c</p></a></li>
<li><a href='#example._plot_coef_h'><p>example._plot_coef_h</p></a></li>
<li><a href='#example._plot.df_agg'><p>example._plot.df_agg</p></a></li>
<li><a href='#example.data_calibration'><p>example.data_calibration</p></a></li>
<li><a href='#example.data_experimental'><p>example.data_experimental</p></a></li>
<li><a href='#get_timestamp'><p>get_timestamp helper function</p></a></li>
<li><a href='#handle_text_input'><p>handle_text_input helper function</p></a></li>
<li><a href='#on_start'><p>on_start helper function</p></a></li>
<li><a href='#plotting_utility'><p>plotting_utility helper function</p></a></li>
<li><a href='#regression_utility'><p>regression_utility helper function</p></a></li>
<li><a href='#solving_equations'><p>solving_equations helper function</p></a></li>
<li><a href='#statistics_list'><p>statistics_list helper function</p></a></li>
<li><a href='#substitutions_create'><p>substitutions_create helper function</p></a></li>
<li><a href='#write_csv'><p>write_csv helper function</p></a></li>
<li><a href='#write_log'><p>write_log helper function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Correct Bias in DNA Methylation Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the algorithms (with minor modifications)
    to correct bias in quantitative DNA methylation analyses as described
    by Moskalev et al. (2011) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkr213">doi:10.1093/nar/gkr213</a>&gt;. Publication:
    Kapsner et al. (2021) &lt;<a href="https://doi.org/10.1002%2Fijc.33681">doi:10.1002/ijc.33681</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kapsner/rBiasCorrection">https://github.com/kapsner/rBiasCorrection</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kapsner/rBiasCorrection/issues">https://github.com/kapsner/rBiasCorrection/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, future, future.apply, ggplot2, magrittr, nls2,
polynom, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggpubr, knitr, lintr, microbenchmark, rmarkdown, testthat (&ge;
3.0.1)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-20 18:30:01 UTC</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-20 17:25:04 UTC; user</td>
</tr>
<tr>
<td>Author:</td>
<td>Lorenz A. Kapsner <a href="https://orcid.org/0000-0003-1866-860X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut, cph],
  Evgeny A. Moskalev [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lorenz A. Kapsner &lt;lorenz.kapsner@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
</table>
<hr>
<h2 id='aggregated_input'>aggregated_input helper function</h2><span id='topic+aggregated_input'></span>

<h3>Description</h3>

<p>Internal function to present aggregated input data on which
calculations are performed. This function does only have an effect, if
repeated measurements are used for calibration a/o experimental data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aggregated_input(datatable, description, vec_cal, type = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aggregated_input_+3A_datatable">datatable</code></td>
<td>
<p>A data.table object that contains either the experimental
data or the calibration data.</p>
</td></tr>
<tr><td><code id="aggregated_input_+3A_description">description</code></td>
<td>
<p>A character string, indicating if <code>datatable</code>
contains either <em>&quot;calibration&quot;</em> data or <em>&quot;experimental&quot;</em> data.</p>
</td></tr>
<tr><td><code id="aggregated_input_+3A_vec_cal">vec_cal</code></td>
<td>
<p>The vector containing the CpG columns (output of 'clean_dt'
with description = &quot;calibration&quot;).</p>
</td></tr>
<tr><td><code id="aggregated_input_+3A_type">type</code></td>
<td>
<p>A single integer. Type of data to be corrected: either &quot;1&quot;
(one locus in many samples, e.g. pyrosequencing data) or &quot;2&quot; (many loci
in one sample, e.g. next-generation sequencing data or microarray data).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table in the long format with aggregated means for each CpG
site of each sample and the corresponding standard deviation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>experimental &lt;- rBiasCorrection::example.data_experimental
calibration &lt;- rBiasCorrection::example.data_calibration

vec_cal &lt;- calibration$vec_cal

experimental_aggregated &lt;- aggregated_input(
  datatable = experimental$dat,
  description = "experimental",
  vec_cal = vec_cal,
  type = 1
)
dim(experimental_aggregated)
class(experimental_aggregated)

calibration_aggregated &lt;- aggregated_input(
  datatable = calibration$dat,
  description = "calibration",
  vec_cal = vec_cal
)
dim(calibration_aggregated)
class(calibration_aggregated)

</code></pre>

<hr>
<h2 id='better_model'>better_model helper function</h2><span id='topic+better_model'></span>

<h3>Description</h3>

<p>Internal function to select the better model between
hyperbolic regression and cubic regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>better_model(
  statstable_pre,
  statstable_post_hyperbolic = NULL,
  statstable_post_cubic = NULL,
  selection_method = "SSE"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="better_model_+3A_statstable_pre">statstable_pre</code></td>
<td>
<p>A data.table object, containing the output of
<code>statisticsList_()</code> of the calculated regression parameters (form
the provided calibration data).</p>
</td></tr>
<tr><td><code id="better_model_+3A_statstable_post_hyperbolic">statstable_post_hyperbolic</code></td>
<td>
<p>A data.table object, containing
the output of <code>statisticsList_()</code> of the calculated regression
parameters form the calibration data corrected with hyperbolic regression.</p>
</td></tr>
<tr><td><code id="better_model_+3A_statstable_post_cubic">statstable_post_cubic</code></td>
<td>
<p>A data.table object, containing the output
of <code>statisticsList_()</code> of the calculated regression parameters form
the calibration data corrected with cubic regression.</p>
</td></tr>
<tr><td><code id="better_model_+3A_selection_method">selection_method</code></td>
<td>
<p>A character string. The method used to select the
regression algorithm to correct the respective CpG site. This is by
default the sum of squared errors (&quot;SSE&quot;). The second option is
&quot;RelError&quot;, which selects the regression method based on the theoretical
relative error after correction. This metric is calculated by correcting
the calibration data with both the hyperbolic regression and the cubic
regression and using them again as input data to calculate the 'goodness
of fit'-metrics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a data.table with 4 columns, the last column
being named 'better_model', which indicates in a binary manner,
if the hyperbolic model (better_model = 0) or the cubic model
(better_model = 1) result in a 'better' 'SSE' or 'RelError' respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# define list object to save all data
rv &lt;- list()
rv$minmax &lt;- TRUE
rv$selection_method &lt;- "RelError"
rv$sample_locus_name &lt;- "Test"
rv$seed &lt;- 1234

# define logfilename
logfilename &lt;- paste0(tempdir(), "/log.txt")

# import experimental file
exp_type_1 &lt;- rBiasCorrection::example.data_experimental
rv$fileimport_experimental &lt;- exp_type_1$dat

# import calibration file
cal_type_1 &lt;- rBiasCorrection::example.data_calibration
rv$fileimport_calibration &lt;- cal_type_1$dat
rv$vec_cal &lt;- cal_type_1$vec_cal


# perform regression
regression_results &lt;- regression_utility(
  rv$fileimport_calibration,
  "Testlocus",
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)

# extract regression results
rv$result_list &lt;- regression_results$result_list

# get regression statistics
rv$reg_stats &lt;- statistics_list(
  rv$result_list,
  minmax = rv$minmax
)

# select the better model based on the sum of squared errrors ("SSE")
rv$choices_list &lt;- better_model(
  statstable_pre = rv$reg_stats,
  selection_method = "SSE"
)


</code></pre>

<hr>
<h2 id='biascorrection'>Correct PCR-Bias in Quantitative DNA Methylation Analyses.</h2><span id='topic+biascorrection'></span>

<h3>Description</h3>

<p>This function implements the algorithms described by
Moskalev et. al in their article 'Correction of PCR-bias in quantitative
DNA methylation studies by means of cubic polynomial regression',
published 2011 in Nucleic acids research, Oxford University Press
(<a href="https://doi.org/10.1093/nar/gkr213">doi:10.1093/nar/gkr213</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biascorrection(
  experimental,
  calibration,
  samplelocusname,
  minmax = FALSE,
  correct_method = "best",
  selection_method = "SSE",
  type = 1,
  csvdir = paste0(tempdir(), "/csvdir/"),
  plotdir = paste0(tempdir(), "/plotdir/"),
  logfilename = paste0(tempdir(), "/log.txt"),
  plot_height = 5,
  plot_width = 7.5,
  plot_textsize = 16,
  seed = 1234,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="biascorrection_+3A_experimental">experimental</code></td>
<td>
<p>A character string. Path to the file containing the raw
methylation values of the samples under investigation.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_calibration">calibration</code></td>
<td>
<p>A character string. In type 1 data (one locus in many
samples, e.g. pyrosequencing data):
Path to the file containing the raw methylation values of the calibration
samples.
In type 2 data (many loci in one sample, e.g. next-generationsequencing
data or microarray data):
Path to the folder that contains at least 4 calibration files (one file
per calibration step). Please refer to the FAQ for more detailed
information on the specific file requirements (<a href="https://raw.githubusercontent.com/kapsner/PCRBiasCorrection/master/FAQ.md">https://raw.githubusercontent.com/kapsner/PCRBiasCorrection/master/FAQ.md</a>).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_samplelocusname">samplelocusname</code></td>
<td>
<p>A character string. In type 1 data: locus name -
name of the gene locus under investigation. In type 2 data: sample name -
name of the sample under investigation.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_minmax">minmax</code></td>
<td>
<p>A logical, indicating which equations are used for
BiasCorrection (default: FALSE). If TRUE, equations are used that include
the respective minima and maxima of the provided data.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_correct_method">correct_method</code></td>
<td>
<p>A character string. Method used to correct the PCR-
bias of the samples under investigation. One of &quot;best&quot; (default),
&quot;hyperbolic&quot; or &quot;cubic&quot;. If the method is set to &quot;best&quot; (short: &quot;b&quot;),
the algorithm will automatically determine the best fitting type of
regression for each CpG site based on <em>selection_method</em> (by
default: sum of squared errors, SSE,
<a href="https://en.wikipedia.org/wiki/Residual_sum_of_squares">https://en.wikipedia.org/wiki/Residual_sum_of_squares</a>). If the
method is set to &quot;hyperbolic&quot; (short: &quot;h&quot;) or &quot;cubic&quot; (short: &quot;c&quot;), the
PCR-bias correction of all samples under investigation will be performed
with the hyperbolic or the cubic regression respectively.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_selection_method">selection_method</code></td>
<td>
<p>A character string. The method used to select the
regression algorithm to correct the respective CpG site. This is by
default the sum of squared errors (&quot;SSE&quot;). The second option is
&quot;RelError&quot;, which selects the regression method based on the theoretical
relative error after correction. This metric is calculated by correcting
the calibration data with both the hyperbolic regression and the cubic
regression and using them again as input data to calculate the 'goodness
of fit'-metrics.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_type">type</code></td>
<td>
<p>A single integer. Type of data to be corrected: either &quot;1&quot; (one
locus in many samples, e.g. pyrosequencing data) or &quot;2&quot; (many loci in one
sample, e.g. next-generation sequencing data or microarray data).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_csvdir">csvdir</code></td>
<td>
<p>A character string. Directory to store the resulting tables.
(default = paste0(tempdir(), &quot;/plotdir/&quot;)). CAUTION: This directory will
be newly created on every call of the function - any preexisting files will
be deleted without a warning.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_plotdir">plotdir</code></td>
<td>
<p>A character string. Directory to store the resulting plots
(default = paste0(tempdir(), &quot;/plotdir/&quot;)). CAUTION: This directory will
be newly created on every call of the function - any preexisting files will
be deleted without a warning.</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to a file to save the log
messages (default = paste0(tempdir(), &quot;/log.txt&quot;)).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_plot_height">plot_height</code></td>
<td>
<p>A integer value. The height (unit: inch) of the
resulting plots (default: 5).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_plot_width">plot_width</code></td>
<td>
<p>A integer value. The width (unit: inch) of the
resulting plots (default: 7.5).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_plot_textsize">plot_textsize</code></td>
<td>
<p>A integer value. The textsize of the
resulting plots (default: 16).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_seed">seed</code></td>
<td>
<p>A integer value. The seed used when solving the unknowns in the
hyperbolic regression equation and the cubic regression equation.
Important for reproducibility (default: 1234).</p>
</td></tr>
<tr><td><code id="biascorrection_+3A_parallel">parallel</code></td>
<td>
<p>A boolean. If TRUE (the default value), initializing
'future::plan(&quot;multicore&quot;)' (on unix systems) or
'future::plan(&quot;multisession&quot;)' (on non-unix systems) before running the
code.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is a wrapper around all of &lsquo;rBiasCorrection'&rsquo;s
included functions. When executing it, it performs the whole workflow of
bias correction and writes resulting csv-files and plots, as well as a
log file to the local file system (the respective directories can be
specified with the function arguments). The return-value is TRUE, if the
correction of PCR measurement biases succeeds. If the correction fails,
an error message is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data.table::fwrite(
  rBiasCorrection::example.data_experimental$dat,
  paste0(tempdir(), "/experimental_data.csv")
)
data.table::fwrite(
  rBiasCorrection::example.data_calibration$dat,
  paste0(tempdir(), "/calibration_data.csv")
)
experimental &lt;- paste0(tempdir(), "/experimental_data.csv")
calibration &lt;- paste0(tempdir(), "/calibration_data.csv")

results &lt;- biascorrection(
  experimental = experimental,
  calibration = calibration,
  samplelocusname = "BRAF",
  parallel = FALSE
)


</code></pre>

<hr>
<h2 id='calibration_plot'>calibration_plot helper function</h2><span id='topic+calibration_plot'></span>

<h3>Description</h3>

<p>Internal function to carry out the plotting of the
calibrations curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calibration_plot(plotlist, coef_hyper, coef_cubic, plot_textsize, minmax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calibration_plot_+3A_plotlist">plotlist</code></td>
<td>
<p>A ggplot object containing a regression plot without
regression curves (output of <code>regression_utility()</code>).</p>
</td></tr>
<tr><td><code id="calibration_plot_+3A_coef_hyper">coef_hyper</code></td>
<td>
<p>A list containing the regression parameters of
the hyperbolic regression equation.</p>
</td></tr>
<tr><td><code id="calibration_plot_+3A_coef_cubic">coef_cubic</code></td>
<td>
<p>A list containing the regression parameters of
the cubic regression equation.</p>
</td></tr>
<tr><td><code id="calibration_plot_+3A_plot_textsize">plot_textsize</code></td>
<td>
<p>A integer value. The textsize of the
resulting plots (default: 16).</p>
</td></tr>
<tr><td><code id="calibration_plot_+3A_minmax">minmax</code></td>
<td>
<p>A logical, indicating which equations are used for
BiasCorrection (default: FALSE). If TRUE, equations are used that include
the respective minima and maxima of the provided data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list containing calibration plots.
</p>

<hr>
<h2 id='clean_dt'>clean_dt helper function</h2><span id='topic+clean_dt'></span>

<h3>Description</h3>

<p>Internal function, that checks the formatting of imported
files and prepares them to be applicable for the following pcr-bias
correction steps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_dt(datatable, description, type, logfilename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clean_dt_+3A_datatable">datatable</code></td>
<td>
<p>A data.table object that contains either the experimental
data or the calibration data.</p>
</td></tr>
<tr><td><code id="clean_dt_+3A_description">description</code></td>
<td>
<p>A character string, indicating if <code>datatable</code>
contains either <em>&quot;calibration&quot;</em> data or <em>&quot;experimental&quot;</em> data.</p>
</td></tr>
<tr><td><code id="clean_dt_+3A_type">type</code></td>
<td>
<p>A single integer. Type of data to be corrected: either &quot;1&quot;
(one locus in many samples, e.g. pyrosequencing data) or &quot;2&quot; (many loci
in one sample, e.g. next-generation sequencing data or microarray data).</p>
</td></tr>
<tr><td><code id="clean_dt_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to the logfile to save the log
messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If a valid file is provided, the function returns a cleaned
data.table, suited for BiasCorrection.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logfilename &lt;- paste0(tempdir(), "/log.txt")
cleaned_experimental &lt;- clean_dt(
  datatable = rBiasCorrection::example.data_experimental$dat,
  description = "experimental",
  type = 1,
  logfilename = logfilename
)
dim(cleaned_experimental)
class(cleaned_experimental)

</code></pre>

<hr>
<h2 id='clean_up'>clean_up helper function</h2><span id='topic+clean_up'></span>

<h3>Description</h3>

<p>Internal function to clean up directories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_up(plotdir, csvdir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clean_up_+3A_plotdir">plotdir</code></td>
<td>
<p>A character string. Path to the folder,
where plots are saved.</p>
</td></tr>
<tr><td><code id="clean_up_+3A_csvdir">csvdir</code></td>
<td>
<p>A character string. Path to the folder,
where resulting tables are saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function silently cleans up the current session and removes
both, the 'plotdir'- and the 'csvdir'-folders. It furthermore resets
the 'future'-backend to plan = &quot;sequential&quot;.
</p>


<h3>See Also</h3>

<p><a href="future.html#topic+plan">plan</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plotdir &lt;- paste0(tempdir(), "/plots/")
csvdir &lt;- paste0(tempdir(), "/csv/")

clean_up(plotdir, csvdir)

</code></pre>

<hr>
<h2 id='create_exampleplot'>create_exampleplot helper function</h2><span id='topic+create_exampleplot'></span>

<h3>Description</h3>

<p>Internal function to create an example plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_exampleplot(
  data,
  coef_hyper,
  coef_cubic,
  plot_height,
  plot_width,
  plot_textsize,
  filename
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_exampleplot_+3A_data">data</code></td>
<td>
<p>A data.table containing the aggregated
calibration data.</p>
</td></tr>
<tr><td><code id="create_exampleplot_+3A_coef_hyper">coef_hyper</code></td>
<td>
<p>A list containing the regression parameters of
the hyperbolic regression equation.</p>
</td></tr>
<tr><td><code id="create_exampleplot_+3A_coef_cubic">coef_cubic</code></td>
<td>
<p>A list containing the regression parameters of
the cubic regression equation.</p>
</td></tr>
<tr><td><code id="create_exampleplot_+3A_plot_height">plot_height</code></td>
<td>
<p>A integer value. The height (unit: inch) of the
resulting plots (default: 5).</p>
</td></tr>
<tr><td><code id="create_exampleplot_+3A_plot_width">plot_width</code></td>
<td>
<p>A integer value. The width (unit: inch) of the
resulting plots (default: 7.5).</p>
</td></tr>
<tr><td><code id="create_exampleplot_+3A_plot_textsize">plot_textsize</code></td>
<td>
<p>A integer value. The textsize of the
resulting plots (default: 16).</p>
</td></tr>
<tr><td><code id="create_exampleplot_+3A_filename">filename</code></td>
<td>
<p>A character. The filename, where to
store the resulting example plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates an example plot and stores
on the local filesystem.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gdat &lt;- rBiasCorrection::example._plot.df_agg

coef_h &lt;- rBiasCorrection::example._plot_coef_h
coef_c &lt;- rBiasCorrection::example._plot_coef_c

create_exampleplot(
  data = gdat,
  coef_hyper = coef_h,
  coef_cubic = coef_c,
  plot_height = 5,
  plot_width = 7.5,
  plot_textsize = 1,
  filename = paste0(tempdir(), "/exampleplot.png")
)

</code></pre>

<hr>
<h2 id='createbarerrorplots'>createbarerrorplots helper function</h2><span id='topic+createbarerrorplots'></span>

<h3>Description</h3>

<p>Internal function to create relative-error bar plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createbarerrorplots(
  statstable_pre,
  statstable_post,
  rv,
  type,
  locus_id = NULL,
  plotdir,
  logfilename,
  mode = NULL,
  plot_height = 5,
  plot_width = 7.5,
  plot_textsize = 16
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createbarerrorplots_+3A_statstable_pre">statstable_pre</code></td>
<td>
<p>A data.table object, containing the output of
<code>statisticsList_()</code> of the calculated regression parameters (form
the provided calibration data).</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_statstable_post">statstable_post</code></td>
<td>
<p>A data.table object, containing the output of
<code>statisticsList_()</code> of the calculated regression parameters form
the corrected calibration data.</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_rv">rv</code></td>
<td>
<p>A list object. A list that contains additional objects needed
for the algorithms.</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_type">type</code></td>
<td>
<p>A single integer. Type of data to be corrected: either &quot;1&quot;
(one locus in many samples, e.g. pyrosequencing data) or &quot;2&quot; (many loci
in one sample, e.g. next-generation sequencing data or microarray data).</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_locus_id">locus_id</code></td>
<td>
<p>A character string. Default: NULL. ID of the respective
locus (only used in type 2 correction).</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_plotdir">plotdir</code></td>
<td>
<p>A character string. Path to the folder,
where plots are saved.</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to a file to save the log
messages (default = paste0(tempdir(), &quot;/log.txt&quot;)).</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_mode">mode</code></td>
<td>
<p>A character string. Default: NULL. Used to indicate &quot;corrected&quot;
calibration data.</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_plot_height">plot_height</code></td>
<td>
<p>A integer value. The height (unit: inch) of the
resulting plots (default: 5).</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_plot_width">plot_width</code></td>
<td>
<p>A integer value. The width (unit: inch) of the
resulting plots (default: 7.5).</p>
</td></tr>
<tr><td><code id="createbarerrorplots_+3A_plot_textsize">plot_textsize</code></td>
<td>
<p>A integer value. The textsize of the
resulting plots (default: 16).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function creates error bar-plots to visualize the relative
error before and after bias correction and writes these plots to the
local filesystem.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# define list object to save all data
rv &lt;- list()
rv$minmax &lt;- TRUE
rv$selection_method &lt;- "RelError"
rv$sample_locus_name &lt;- "Test"
rv$seed &lt;- 1234

# define plotdir
rv$plotdir &lt;- paste0(tempdir(), "/plots/")
dir.create(rv$plotdir)

# define logfilename
logfilename &lt;- paste0(tempdir(), "/log.txt")

# import experimental file
exp_type_1 &lt;- rBiasCorrection::example.data_experimental
rv$fileimport_experimental &lt;- exp_type_1$dat

# import calibration file
cal_type_1 &lt;- rBiasCorrection::example.data_calibration
rv$fileimport_calibration &lt;- cal_type_1$dat
rv$vec_cal &lt;- cal_type_1$vec_cal


# perform regression
regression_results &lt;- regression_utility(
  rv$fileimport_calibration,
  "Testlocus",
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)

# extract regression results
rv$result_list &lt;- regression_results$result_list

# get regression statistics
rv$reg_stats &lt;- statistics_list(
  rv$result_list,
  minmax = TRUE
)

# select the better model based on the sum of squared errrors ("SSE")
rv$choices_list &lt;- better_model(
  statstable_pre = rv$reg_stats,
  selection_method = "SSE"
)

# correct calibration data (to show corrected calibration curves)
solved_eq_h &lt;- solving_equations(datatable = rv$fileimport_calibration,
                                 regmethod = rv$choices_list,
                                 type = 1,
                                 rv = rv,
                                 mode = "corrected",
                                 logfilename = logfilename,
                                 minmax = rv$minmax)
rv$fileimport_cal_corrected_h &lt;- solved_eq_h$results
colnames(rv$fileimport_cal_corrected_h) &lt;- colnames(
  rv$fileimport_calibration
)

# calculate new calibration curves from corrected calibration data
regression_results &lt;- regression_utility(
  data = rv$fileimport_cal_corrected_h,
  samplelocusname = rv$sample_locus_name,
  rv = rv,
  mode = "corrected",
  logfilename = logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)
rv$result_list_hyperbolic &lt;- regression_results$result_list


# save regression statistics to reactive value
rv$reg_stats_corrected_h &lt;- statistics_list(
  resultlist = rv$result_list_hyperbolic,
  minmax = rv$minmax
)

createbarerrorplots(
  statstable_pre = rv$reg_stats,
  statstable_post = rv$reg_stats_corrected_h,
  rv = rv,
  type = 1,
  locus_id = NULL,
  plotdir = rv$plotdir,
  logfilename = logfilename,
  mode = "corrected_h",
  plot_height = 5,
  plot_width = 7.5,
  plot_textsize = 1
)


</code></pre>

<hr>
<h2 id='example._plot_coef_c'>example._plot_coef_c</h2><span id='topic+example._plot_coef_c'></span>

<h3>Description</h3>

<p>A list containing exemplary coefficients
of the cubic regression equation for CpG site 1
to create an example plot.
</p>

<hr>
<h2 id='example._plot_coef_h'>example._plot_coef_h</h2><span id='topic+example._plot_coef_h'></span>

<h3>Description</h3>

<p>A list containing exemplary coefficients
of the hyperbolic regression equation for CpG site 1
to create an example plot.
</p>

<hr>
<h2 id='example._plot.df_agg'>example._plot.df_agg</h2><span id='topic+example._plot.df_agg'></span>

<h3>Description</h3>

<p>A data.table containing the aggregated
calibration data for CpG site 1
to create an example plot.
</p>

<hr>
<h2 id='example.data_calibration'>example.data_calibration</h2><span id='topic+example.data_calibration'></span>

<h3>Description</h3>

<p>A list containing the calibration data ($dat) and the
colnames ($vec_cal) needed by the algorithms to perform the bias
correction.
</p>

<hr>
<h2 id='example.data_experimental'>example.data_experimental</h2><span id='topic+example.data_experimental'></span>

<h3>Description</h3>

<p>A list containing the experimental data ($dat) and the
colnames ($vec_cal) needed by the algorithms to perform the bias
correction.
</p>

<hr>
<h2 id='get_timestamp'>get_timestamp helper function</h2><span id='topic+get_timestamp'></span>

<h3>Description</h3>

<p>Internal function to get the current timestamp to
write it to filenames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_timestamp()
</code></pre>


<h3>Value</h3>

<p>This function takes no argument and returns a formatted timestamp
of the current system time, which can be integrated e.g. into a filename.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+Sys.time">Sys.time</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_timestamp()

</code></pre>

<hr>
<h2 id='handle_text_input'>handle_text_input helper function</h2><span id='topic+handle_text_input'></span>

<h3>Description</h3>

<p>Internal function to remove punctuation and unneeded stuff
from user inputs with regular expressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>handle_text_input(textinput)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="handle_text_input_+3A_textinput">textinput</code></td>
<td>
<p>A character string with the textinput to perform these
predefined regular expressions on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a cleaned up character string, limited to a
maximum of 15 chars.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>textinput &lt;- "This is a dirty! text."
handle_text_input(textinput)

</code></pre>

<hr>
<h2 id='on_start'>on_start helper function</h2><span id='topic+on_start'></span>

<h3>Description</h3>

<p>Internal function, that initializes plotdir,
csvdir and logfilename.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>on_start(plotdir, csvdir, logfilename, parallel)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="on_start_+3A_plotdir">plotdir</code></td>
<td>
<p>A character string. Path to the folder,
where plots are saved.</p>
</td></tr>
<tr><td><code id="on_start_+3A_csvdir">csvdir</code></td>
<td>
<p>A character string. Path to the folder,
where resulting tables are saved.</p>
</td></tr>
<tr><td><code id="on_start_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to the logfile to save the log
messages.</p>
</td></tr>
<tr><td><code id="on_start_+3A_parallel">parallel</code></td>
<td>
<p>A boolean. If TRUE (the default value), initializing
'future::plan(&quot;multicore&quot;)' (on unix systems) or
'future::plan(&quot;multisession&quot;)' (on non-unix systems) before running the
code.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function silently creates the directories'plotdir' and 'csvdir'
on the local filesystem and initializes the logfile, specified with
'logfilename'. Furthermore, if 'parallel = TRUE', the 'future'-backend is
initialized.
</p>


<h3>See Also</h3>

<p><a href="future.html#topic+plan">plan</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # runtime &gt; 5 sec.
plotdir &lt;- paste0(tempdir(), "/plots/")
csvdir &lt;- paste0(tempdir(), "/csv/")
logfilename &lt;- paste0(tempdir(), "/log.txt")
parallel &lt;- FALSE

on_start(plotdir, csvdir, logfilename, parallel)


</code></pre>

<hr>
<h2 id='plotting_utility'>plotting_utility helper function</h2><span id='topic+plotting_utility'></span>

<h3>Description</h3>

<p>Internal function to carry out the plotting of the
calibrations curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotting_utility(
  data,
  plotlist_reg,
  type,
  samplelocusname,
  locus_id = NULL,
  rv,
  mode = NULL,
  plotdir,
  logfilename,
  minmax,
  plot_height = 5,
  plot_width = 7.5,
  plot_textsize = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotting_utility_+3A_data">data</code></td>
<td>
<p>A data.table object that contains the calibration data.</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_plotlist_reg">plotlist_reg</code></td>
<td>
<p>A list object contating regression plots without
regression curves (output of <code>regression_utility()</code>).</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_type">type</code></td>
<td>
<p>A single integer. Type of data to be corrected: either &quot;1&quot;
(one locus in many samples, e.g. pyrosequencing data) or &quot;2&quot; (many loci
in one sample, e.g. next-generation sequencing data or microarray data).</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_samplelocusname">samplelocusname</code></td>
<td>
<p>A character string. In type 1 data: locus name -
name of the gene locus under investigation. In type 2 data: sample name -
name of the sample under investigation.</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_locus_id">locus_id</code></td>
<td>
<p>A character string. Default: NULL. ID of the respective
locus (only used in type 2 correction).</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_rv">rv</code></td>
<td>
<p>A list object. A list that contains additional objects needed
for the algorithms.</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_mode">mode</code></td>
<td>
<p>A character string. Default: NULL. Used to indicate &quot;corrected&quot;
calibration data.</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_plotdir">plotdir</code></td>
<td>
<p>A character string. Path to the folder,
where plots are saved.</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to a file to save the log
messages (default = paste0(tempdir(), &quot;/log.txt&quot;)).</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_minmax">minmax</code></td>
<td>
<p>A logical, indicating which equations are used for
BiasCorrection (default: FALSE). If TRUE, equations are used that include
the respective minima and maxima of the provided data.</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_plot_height">plot_height</code></td>
<td>
<p>A integer value. The height (unit: inch) of the
resulting plots (default: 5).</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_plot_width">plot_width</code></td>
<td>
<p>A integer value. The width (unit: inch) of the
resulting plots (default: 7.5).</p>
</td></tr>
<tr><td><code id="plotting_utility_+3A_plot_textsize">plot_textsize</code></td>
<td>
<p>A integer value. The textsize of the
resulting plots (default: 16).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function creates calibration plots and writes them to the
local filesystem.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# define list object to save all data
rv &lt;- list()
rv$minmax &lt;- TRUE
rv$selection_method &lt;- "RelError"
rv$sample_locus_name &lt;- "Test"
rv$seed &lt;- 1234

# define logfilename
logfilename &lt;- paste0(tempdir(), "/log.txt")

# define plotdir
rv$plotdir &lt;- paste0(tempdir(), "/plots/")
dir.create(rv$plotdir)

# import experimental file
exp_type_1 &lt;- rBiasCorrection::example.data_experimental
rv$fileimport_experimental &lt;- exp_type_1$dat

# import calibration file
cal_type_1 &lt;- rBiasCorrection::example.data_calibration
rv$fileimport_calibration &lt;- cal_type_1$dat
rv$vec_cal &lt;- cal_type_1$vec_cal


# perform regression
regression_results &lt;- regression_utility(
  rv$fileimport_calibration,
  "Testlocus",
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)

# extract the plotlist
plotlist_reg &lt;- regression_results$plot_list

plotting_utility(
  data = rv$fileimport_calibration,
  plotlist_reg = plotlist_reg,
  type = 1,
  samplelocusname = rv$sample_locus_name,
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  plotdir = rv$plotdir,
  logfilename = logfilename,
  minmax = rv$minmax,
  plot_height = 5,
  plot_width = 7.5,
  plot_textsize = 1
)


</code></pre>

<hr>
<h2 id='regression_utility'>regression_utility helper function</h2><span id='topic+regression_utility'></span>

<h3>Description</h3>

<p>Internal function to carry out the regression calculations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>regression_utility(
  data,
  samplelocusname,
  locus_id = NULL,
  rv,
  mode = NULL,
  logfilename,
  minmax,
  seed = 1234
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="regression_utility_+3A_data">data</code></td>
<td>
<p>A data.table object that contains the calibration data.</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_samplelocusname">samplelocusname</code></td>
<td>
<p>A character string. In type 1 data: locus name -
name of the gene locus under investigation. In type 2 data: sample name -
name of the sample under investigation.</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_locus_id">locus_id</code></td>
<td>
<p>A character string. Default: NULL. ID of the respective
locus (only used in type 2 correction).</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_rv">rv</code></td>
<td>
<p>A list object. A list that contains additional objects needed
for the algorithms.</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_mode">mode</code></td>
<td>
<p>A character string. Default: NULL. Used to indicate &quot;corrected&quot;
calibration data.</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to a file to save the log
messages (default = paste0(tempdir(), &quot;/log.txt&quot;)).</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_minmax">minmax</code></td>
<td>
<p>A logical, indicating which equations are used for
BiasCorrection (default: FALSE). If TRUE, equations are used that include
the respective minima and maxima of the provided data.</p>
</td></tr>
<tr><td><code id="regression_utility_+3A_seed">seed</code></td>
<td>
<p>A integer value. The seed used when solving the unknowns in the
hyperbolic regression equation and the cubic regression equation.
Important for reproducibility (default: 1234).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function performs the regression calculations and returns
the results in a list.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># define list object to save all data
rv &lt;- list()
rv$minmax &lt;- TRUE
rv$selection_method &lt;- "RelError"
rv$sample_locus_name &lt;- "Test"
rv$seed &lt;- 1234

# define logfilename
logfilename &lt;- paste0(tempdir(), "/log.txt")

# import experimental file
exp_type_1 &lt;- rBiasCorrection::example.data_experimental
rv$fileimport_experimental &lt;- exp_type_1$dat

# import calibration file
cal_type_1 &lt;- rBiasCorrection::example.data_calibration
rv$fileimport_calibration &lt;- cal_type_1$dat
rv$vec_cal &lt;- cal_type_1$vec_cal

# perform regression
regression_results &lt;- regression_utility(
  rv$fileimport_calibration,
  "Testlocus",
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)
length(regression_results)
class(regression_results)

</code></pre>

<hr>
<h2 id='solving_equations'>solving_equations helper function</h2><span id='topic+solving_equations'></span>

<h3>Description</h3>

<p>Internal function to solve the hyperbolic and cubic regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solving_equations(
  datatable,
  regmethod,
  type,
  rv,
  mode = NULL,
  logfilename,
  minmax
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solving_equations_+3A_datatable">datatable</code></td>
<td>
<p>A data.table object that contains either the experimental
data or the calibration data.</p>
</td></tr>
<tr><td><code id="solving_equations_+3A_regmethod">regmethod</code></td>
<td>
<p>A data.table object, with 2 columns, containing the names
of the samples to correct (columns 1) and a binary variable
<em>better_model</em> that indicates, if the data should be corrected with
the hyperbolic regression parameters (better_model = 0) or with the
cubic regression parameters (better_model = 1).</p>
</td></tr>
<tr><td><code id="solving_equations_+3A_type">type</code></td>
<td>
<p>A single integer. Type of data to be corrected: either &quot;1&quot;
(one locus in many samples, e.g. pyrosequencing data) or &quot;2&quot; (many loci
in one sample, e.g. next-generation sequencing data or microarray data).</p>
</td></tr>
<tr><td><code id="solving_equations_+3A_rv">rv</code></td>
<td>
<p>A list object. A list that contains additional objects needed
for the algorithms.</p>
</td></tr>
<tr><td><code id="solving_equations_+3A_mode">mode</code></td>
<td>
<p>A character string. Default: NULL. Used to indicate &quot;corrected&quot;
calibration data.</p>
</td></tr>
<tr><td><code id="solving_equations_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to the logfile to save the log
messages.</p>
</td></tr>
<tr><td><code id="solving_equations_+3A_minmax">minmax</code></td>
<td>
<p>A logical, indicating which equations are used for
BiasCorrection (default: FALSE). If TRUE, equations are used that include
the respective minima and maxima of the provided data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function solves the equations of the hyperbolic and the
cubic regression and returns the respectively interpolated values of the
provided 'datatable'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# define list object to save all data
rv &lt;- list()
rv$minmax &lt;- TRUE
rv$selection_method &lt;- "RelError"
rv$sample_locus_name &lt;- "Test"
rv$seed &lt;- 1234

# define logfilename
logfilename &lt;- paste0(tempdir(), "/log.txt")

# import experimental file
exp_type_1 &lt;- rBiasCorrection::example.data_experimental
rv$fileimport_experimental &lt;- exp_type_1$dat

# import calibration file
cal_type_1 &lt;- rBiasCorrection::example.data_calibration
rv$fileimport_calibration &lt;- cal_type_1$dat
rv$vec_cal &lt;- cal_type_1$vec_cal


# perform regression
regression_results &lt;- regression_utility(
  rv$fileimport_calibration,
  "Testlocus",
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)

# extract regression results
rv$result_list &lt;- regression_results$result_list

# get regression statistics
rv$reg_stats &lt;- statistics_list(
  rv$result_list,
  minmax = TRUE
)

# select the better model based on the sum of squared errrors ("SSE")
rv$choices_list &lt;- better_model(
  statstable_pre = rv$reg_stats,
  selection_method = "SSE"
)

# correct calibration data (to show corrected calibration curves)
solved_eq_h &lt;- solving_equations(datatable = rv$fileimport_calibration,
                                 regmethod = rv$choices_list,
                                 type = 1,
                                 rv = rv,
                                 mode = "corrected",
                                 logfilename = logfilename,
                                 minmax = rv$minmax)
rv$fileimport_cal_corrected_h &lt;- solved_eq_h$results
colnames(rv$fileimport_cal_corrected_h) &lt;- colnames(
  rv$fileimport_calibration
)


</code></pre>

<hr>
<h2 id='statistics_list'>statistics_list helper function</h2><span id='topic+statistics_list'></span>

<h3>Description</h3>

<p>Internal function that converts the results_list (output of
<code>regression_utility()</code>) into a data.table object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>statistics_list(resultlist, minmax = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="statistics_list_+3A_resultlist">resultlist</code></td>
<td>
<p>A list object. The results_list output of
<code>regression_utility()</code>.</p>
</td></tr>
<tr><td><code id="statistics_list_+3A_minmax">minmax</code></td>
<td>
<p>A logical, indicating which equations are used for
BiasCorrection (default: FALSE). If TRUE, equations are used that include
the respective minima and maxima of the provided data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function takes the 'resultslist' and converts it to a
statistics list, which is basically a 'data.table' with the results
of the hyperbolic and the cubic regression.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># define list object to save all data
rv &lt;- list()
rv$minmax &lt;- TRUE
rv$selection_method &lt;- "RelError"
rv$sample_locus_name &lt;- "Test"
rv$seed &lt;- 1234

# define logfilename
logfilename &lt;- paste0(tempdir(), "/log.txt")

# import experimental file
exp_type_1 &lt;- rBiasCorrection::example.data_experimental
rv$fileimport_experimental &lt;- exp_type_1$dat

# import calibration file
cal_type_1 &lt;- rBiasCorrection::example.data_calibration
rv$fileimport_calibration &lt;- cal_type_1$dat
rv$vec_cal &lt;- cal_type_1$vec_cal

# perform regression
regression_results &lt;- regression_utility(
  rv$fileimport_calibration,
  "Testlocus",
  locus_id = NULL,
  rv = rv,
  mode = NULL,
  logfilename,
  minmax = rv$minmax,
  seed = rv$seed
)

# extract regression results
rv$result_list &lt;- regression_results$result_list

# get regression statistics
rv$reg_stats &lt;- statistics_list(
  rv$result_list,
  minmax = rv$minmax
)

</code></pre>

<hr>
<h2 id='substitutions_create'>substitutions_create helper function</h2><span id='topic+substitutions_create'></span>

<h3>Description</h3>

<p>Internal function to initialize a data.table object
to store the substitutions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>substitutions_create()
</code></pre>


<h3>Value</h3>

<p>This function takes no argument and initializes an empty
'data.table' to hold the substituted values, if substitutions occur during
BiasCorrection.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>substitutions &lt;- substitutions_create()
class(substitutions)

</code></pre>

<hr>
<h2 id='write_csv'>write_csv helper function</h2><span id='topic+write_csv'></span>

<h3>Description</h3>

<p>Internal function to store the created tables in csv files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_csv(table, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_csv_+3A_table">table</code></td>
<td>
<p>A data.table object to store on the local file system</p>
</td></tr>
<tr><td><code id="write_csv_+3A_filename">filename</code></td>
<td>
<p>The file name (including the path) to store <code>table</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function silently writes a 'data.table' object to a CSV file.
</p>


<h3>See Also</h3>

<p><a href="data.table.html#topic+fwrite">fwrite</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table &lt;- data.table::data.table(
  a = stats::runif(1000),
  b = stats::runif(1000)
)

write_csv(table, paste0(tempdir(), "/example.csv"))

</code></pre>

<hr>
<h2 id='write_log'>write_log helper function</h2><span id='topic+write_log'></span>

<h3>Description</h3>

<p>Internal function to write log-messages to the file
specified in logfilename.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_log(message, logfilename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_log_+3A_message">message</code></td>
<td>
<p>A character string containing the log message.</p>
</td></tr>
<tr><td><code id="write_log_+3A_logfilename">logfilename</code></td>
<td>
<p>A character string. Path to the logfile to save the log
messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function prints the logging message to the console and writes it
to the local logfile, specified with 'logfilename'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>message &lt;- "This is a logmessage"
logfilename &lt;- paste0(tempdir(), "/log.txt")

write_log(message, logfilename)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
