<!DOCTYPE html><html lang="en"><head><title>Help for package rbioapi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rbioapi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rbioapi'><p>rbioapi: User-Friendly R Interface to Biologic Web Services' API</p></a></li>
<li><a href='#rba_connection_test'><p>Test if the Supported Services Are Responding</p></a></li>
<li><a href='#rba_enrichr'><p>A One-step Wrapper for Gene-list Enrichment Using Enrichr</p></a></li>
<li><a href='#rba_enrichr_add_background'><p>Upload Background Gene-List to Enrichr</p></a></li>
<li><a href='#rba_enrichr_add_list'><p>Upload Your Gene-List to Enrichr</p></a></li>
<li><a href='#rba_enrichr_enrich'><p>Get Enrichr Enrichment Results</p></a></li>
<li><a href='#rba_enrichr_gene_map'><p>Find Enrichr Terms That Contain a Given Gene</p></a></li>
<li><a href='#rba_enrichr_libs'><p>Retrieve a List of available libraries from Enrichr</p></a></li>
<li><a href='#rba_enrichr_view_list'><p>View an Uploaded Gene List</p></a></li>
<li><a href='#rba_jaspar_collections'><p>List collections available in JASPAR</p></a></li>
<li><a href='#rba_jaspar_collections_matrices'><p>List matrices available in a JASPAR collection</p></a></li>
<li><a href='#rba_jaspar_matrix'><p>Get a Position Frequency Matrices (PFM) with annotations</p></a></li>
<li><a href='#rba_jaspar_matrix_search'><p>Search matrix profiles available in JASPAR</p></a></li>
<li><a href='#rba_jaspar_matrix_versions'><p>List matrix profile versions associated with a base ID</p></a></li>
<li><a href='#rba_jaspar_releases'><p>Get information about JASPAR database releases</p></a></li>
<li><a href='#rba_jaspar_sites'><p>Get binding sites of a matrix profile</p></a></li>
<li><a href='#rba_jaspar_species'><p>List available species in JASPAR</p></a></li>
<li><a href='#rba_jaspar_species_matrices'><p>List matrices available in JASPAR of a species</p></a></li>
<li><a href='#rba_jaspar_taxons'><p>List available taxonomic groups in JASPAR</p></a></li>
<li><a href='#rba_jaspar_taxons_matrices'><p>List matrices available in JASPAR of a taxonomic group</p></a></li>
<li><a href='#rba_jaspar_tffm'><p>Get a TF flexible models (TFFMs) information</p></a></li>
<li><a href='#rba_jaspar_tffm_search'><p>Search TF flexible models (TFFMs) available in JASPAR</p></a></li>
<li><a href='#rba_mieaa_cats'><p>Get Supported Enrichment Categories for a Species and miRNA Type</p></a></li>
<li><a href='#rba_mieaa_convert_type'><p>Convert Between Mature and precursor miRNA Accession</p></a></li>
<li><a href='#rba_mieaa_convert_version'><p>Convert miRNA accession Between Different miRBase Versions</p></a></li>
<li><a href='#rba_mieaa_enrich'><p>A One-step Wrapper for miRNA Enrichment Using miEAA</p></a></li>
<li><a href='#rba_mieaa_enrich_results'><p>Retrieve Results of a finished Enrichment Analysis from miEAA</p></a></li>
<li><a href='#rba_mieaa_enrich_status'><p>Check Status of a Submitted Enrichment Analysis in miEAA</p></a></li>
<li><a href='#rba_mieaa_enrich_submit'><p>Submit miEAA miRNA Enrichment Analysis Request</p></a></li>
<li><a href='#rba_options'><p>Set rbioapi Global Options</p></a></li>
<li><a href='#rba_pages'><p>Get Multiple Pages of a Paginated Resource</p></a></li>
<li><a href='#rba_panther_enrich'><p>PANTHER Over-Representation or Enrichment Analysis</p></a></li>
<li><a href='#rba_panther_family'><p>Get PANTHER Families and Sub-Families</p></a></li>
<li><a href='#rba_panther_homolog'><p>Search PANTHER for Homologs of Gene(s)</p></a></li>
<li><a href='#rba_panther_info'><p>Get PANTHER database Information</p></a></li>
<li><a href='#rba_panther_mapping'><p>Map A Gene-set to PANTHER Database</p></a></li>
<li><a href='#rba_panther_ortholog'><p>Search PANTHER for Orthologs of Gene(s)</p></a></li>
<li><a href='#rba_panther_tree_grafter'><p>PANTHER Tree Grafter</p></a></li>
<li><a href='#rba_reactome_analysis'><p>Reactome Over-Representation or Expression Analysis</p></a></li>
<li><a href='#rba_reactome_analysis_download'><p>Download Different Reactome Analysis Results</p></a></li>
<li><a href='#rba_reactome_analysis_import'><p>Import Saved Analysis JSON to Reactome</p></a></li>
<li><a href='#rba_reactome_analysis_mapping'><p>Maps Molecule Identifiers</p></a></li>
<li><a href='#rba_reactome_analysis_pdf'><p>Generate PDF file with Reactome Analysis Results</p></a></li>
<li><a href='#rba_reactome_analysis_species'><p>Compare Human Pathways with with Other Species</p></a></li>
<li><a href='#rba_reactome_analysis_token'><p>Return the Results Associated with a Token</p></a></li>
<li><a href='#rba_reactome_complex_list'><p>Get Complexes That Include a Molecule</p></a></li>
<li><a href='#rba_reactome_complex_subunits'><p>Get a Complex's Subunits</p></a></li>
<li><a href='#rba_reactome_diseases'><p>Reactome Diseases</p></a></li>
<li><a href='#rba_reactome_entity_other_forms'><p>Get Other forms of a Reactome Entity</p></a></li>
<li><a href='#rba_reactome_event_ancestors'><p>Get Reactome Events Ancestors</p></a></li>
<li><a href='#rba_reactome_event_hierarchy'><p>Get Full Event Hierarchy of a Species</p></a></li>
<li><a href='#rba_reactome_exporter_diagram'><p>Get a Reactome Event Diagram</p></a></li>
<li><a href='#rba_reactome_exporter_event'><p>Exports A Reactome Event to SBGN or SBML</p></a></li>
<li><a href='#rba_reactome_exporter_overview'><p>Get a Reactome Pathway Overview</p></a></li>
<li><a href='#rba_reactome_exporter_reaction'><p>Get a Reactome Reaction Event</p></a></li>
<li><a href='#rba_reactome_interactors_psicquic'><p>The interface From Reactome to PSICQUIC</p></a></li>
<li><a href='#rba_reactome_interactors_static'><p>Get Static(IntAct) Interaction Information of a Protein</p></a></li>
<li><a href='#rba_reactome_mapping'><p>Map External ID to Reactome Pathways/Reactions</p></a></li>
<li><a href='#rba_reactome_orthology'><p>Get Orthologous (Computationally Inferred) Events</p></a></li>
<li><a href='#rba_reactome_participant_of'><p>Get Larger Reactome Structures Which Include an Entity</p></a></li>
<li><a href='#rba_reactome_participants'><p>Get Participants of a Reactome Event</p></a></li>
<li><a href='#rba_reactome_pathways_events'><p>Get Events Contained in an Upstream Events</p></a></li>
<li><a href='#rba_reactome_pathways_low'><p>Get lower level pathways Containing a 'Physical Entity' or Event</p></a></li>
<li><a href='#rba_reactome_pathways_top'><p>Get Top Level Pathways in a Species</p></a></li>
<li><a href='#rba_reactome_people_id'><p>A person by his identifiers</p></a></li>
<li><a href='#rba_reactome_people_name'><p>Get Persons Information by Name</p></a></li>
<li><a href='#rba_reactome_query'><p>Query and Retrieve any Reactome knowledge-base Object</p></a></li>
<li><a href='#rba_reactome_species'><p>Get Reactome Species</p></a></li>
<li><a href='#rba_reactome_version'><p>The version number of current database</p></a></li>
<li><a href='#rba_reactome_xref'><p>Map Cross References IDs to Reactome ReferenceEntity</p></a></li>
<li><a href='#rba_string_annotations'><p>Retrieving Functional Annotation</p></a></li>
<li><a href='#rba_string_enrichment'><p>Getting Functional Enrichment</p></a></li>
<li><a href='#rba_string_enrichment_image'><p>Get STRING Enrichment Plot</p></a></li>
<li><a href='#rba_string_enrichment_ppi'><p>Get Protein-Protein Interaction Enrichment</p></a></li>
<li><a href='#rba_string_homology_inter'><p>Get Similarity Scores Hits of Proteins in Different Species</p></a></li>
<li><a href='#rba_string_homology_intra'><p>Get Similarity Scores Hits of Proteins in a Species</p></a></li>
<li><a href='#rba_string_interaction_partners'><p>Get All STRING Interaction Partners</p></a></li>
<li><a href='#rba_string_interactions_network'><p>Get STRING Network Interactions</p></a></li>
<li><a href='#rba_string_map_ids'><p>Map a Set of Identifiers to STRING Identifiers</p></a></li>
<li><a href='#rba_string_network_image'><p>Get STRING Network Image</p></a></li>
<li><a href='#rba_string_version'><p>Get Current STRING Version</p></a></li>
<li><a href='#rba_uniprot_antigens'><p>Get Antigens by UniProt Accession</p></a></li>
<li><a href='#rba_uniprot_antigens_search'><p>Search Antigens in UniProt</p></a></li>
<li><a href='#rba_uniprot_coordinates'><p>Get Genomic Coordinates of a Protein</p></a></li>
<li><a href='#rba_uniprot_coordinates_location'><p>Search UniProt entries by taxonomy and genomic coordinates</p></a></li>
<li><a href='#rba_uniprot_coordinates_location_genome'><p>Get Genome coordinate by Gene Sequence position</p></a></li>
<li><a href='#rba_uniprot_coordinates_location_protein'><p>Get Genome coordinate by Protein Sequence position</p></a></li>
<li><a href='#rba_uniprot_coordinates_search'><p>Search Genomic Coordinates of UniProt entries</p></a></li>
<li><a href='#rba_uniprot_epitope'><p>Retrieve Epitopes by Accession</p></a></li>
<li><a href='#rba_uniprot_epitope_search'><p>Search UniProt Epitopes</p></a></li>
<li><a href='#rba_uniprot_features'><p>Get UniProt protein sequence features by accession</p></a></li>
<li><a href='#rba_uniprot_features_search'><p>Search UniProt protein sequence features</p></a></li>
<li><a href='#rba_uniprot_genecentric'><p>Get Gene-Centric proteins by UniProt Accession</p></a></li>
<li><a href='#rba_uniprot_genecentric_search'><p>Search Gene-Centric Proteins</p></a></li>
<li><a href='#rba_uniprot_mutagenesis'><p>Get Mutagenesis by UniProt Accession</p></a></li>
<li><a href='#rba_uniprot_mutagenesis_search'><p>Search Mutagenesis in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteins'><p>Get UniProt entry by accession</p></a></li>
<li><a href='#rba_uniprot_proteins_crossref'><p>Get UniProt Entry by UniProt Cross-Reference Database and ID</p></a></li>
<li><a href='#rba_uniprot_proteins_search'><p>Search UniProt entries</p></a></li>
<li><a href='#rba_uniprot_proteomes'><p>Get proteome by proteome/proteins UPID</p></a></li>
<li><a href='#rba_uniprot_proteomes_search'><p>Search Proteomes in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteomics'><p>Get Proteomics Peptides Mapped to UniProt Protein (Deprecated)</p></a></li>
<li><a href='#rba_uniprot_proteomics_hpp'><p>Get HPP Proteomics data in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteomics_hpp_search'><p>Search HPP Proteomics data in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteomics_non_ptm'><p>Get Proteomics data in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteomics_non_ptm_search'><p>Search Proteomics data in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteomics_ptm'><p>Get Post-Translational Modification of UniProt Protein</p></a></li>
<li><a href='#rba_uniprot_proteomics_ptm_search'><p>Search Post-Translational Modification Proteomics in UniProt</p></a></li>
<li><a href='#rba_uniprot_proteomics_search'><p>Search Proteomics Peptides in UniProt (Deprecated)</p></a></li>
<li><a href='#rba_uniprot_proteomics_species'><p>Get UniProt Proteomics Metadata</p></a></li>
<li><a href='#rba_uniprot_ptm'><p>Get Post-Translational Modification of UniProt Protein (Deprecated)</p></a></li>
<li><a href='#rba_uniprot_ptm_search'><p>Search Post-Translational Modification in UniProt (Deprecated)</p></a></li>
<li><a href='#rba_uniprot_rna_edit'><p>Retrieve Epitope by Accession</p></a></li>
<li><a href='#rba_uniprot_rna_edit_search'><p>Search RNA Editing in UniProt</p></a></li>
<li><a href='#rba_uniprot_taxonomy'><p>Get UniProt Taxonomy Nodes</p></a></li>
<li><a href='#rba_uniprot_taxonomy_lca'><p>Get Lowest Common Ancestor (LCA) of Two Taxonomy Nodes</p></a></li>
<li><a href='#rba_uniprot_taxonomy_lineage'><p>Get Taxonomic Lineage</p></a></li>
<li><a href='#rba_uniprot_taxonomy_name'><p>Search UniProt Taxonomic Names</p></a></li>
<li><a href='#rba_uniprot_taxonomy_path'><p>Traverse UniProt Taxonomic Tree Path</p></a></li>
<li><a href='#rba_uniprot_taxonomy_relationship'><p>Get Shortest Path Between Two Taxonomy Nodes</p></a></li>
<li><a href='#rba_uniprot_uniparc'><p>Get UniParc entry</p></a></li>
<li><a href='#rba_uniprot_uniparc_bestguess'><p>Get UniParc Longest Sequence for Entries</p></a></li>
<li><a href='#rba_uniprot_uniparc_search'><p>Search UniParc Entries</p></a></li>
<li><a href='#rba_uniprot_uniparc_sequence'><p>Get UniParc Entries by Sequence</p></a></li>
<li><a href='#rba_uniprot_variation'><p>Get natural variants in UniProt by NIH-NCBI SNP database identifier</p></a></li>
<li><a href='#rba_uniprot_variation_search'><p>Search UniProt Natural Variants</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>User-Friendly R Interface to Biologic Web Services' API</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Currently fully supports Enrichr, JASPAR, miEAA, PANTHER,
    Reactome, STRING, and UniProt! The goal of rbioapi is to provide a
    user-friendly and consistent interface to biological databases and
    services. In a way that insulates the user from the technicalities of
    using web services API and creates a unified and easy-to-use interface
    to biological and medical web services. This is an ongoing project; New
    databases and services will be added periodically. Feel free to
    suggest any databases or services you often use.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://rbioapi.moosa-r.com">https://rbioapi.moosa-r.com</a>, <a href="https://github.com/moosa-r/rbioapi">https://github.com/moosa-r/rbioapi</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/moosa-r/rbioapi/issues">https://github.com/moosa-r/rbioapi/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>httr, jsonlite, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DT, knitr, png, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-07 21:50:26 UTC; moosa</td>
</tr>
<tr>
<td>Author:</td>
<td>Moosa Rezwani <a href="https://orcid.org/0000-0001-6325-4444"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Moosa Rezwani &lt;moosa.rezwani@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-07 23:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='rbioapi'>rbioapi: User-Friendly R Interface to Biologic Web Services' API</h2><span id='topic+rbioapi-package'></span><span id='topic+rbioapi'></span>

<h3>Description</h3>

<p>Currently fully supports Enrichr, JASPAR, miEAA, PANTHER, Reactome, STRING, and UniProt! The goal of rbioapi is to provide a user-friendly and consistent interface to biological databases and services. In a way that insulates the user from the technicalities of using web services API and creates a unified and easy-to-use interface to biological and medical web services. This is an ongoing project; New databases and services will be added periodically. Feel free to suggest any databases or services you often use.
</p>


<h3>Supported Database/Services</h3>


<ul>
<li><p> Enrichr
</p>
</li>
<li><p> JASPAR
</p>
</li>
<li><p> MiEAA
</p>
</li>
<li><p> PANTHER
</p>
</li>
<li><p> Reactome
</p>
</li>
<li><p> STRING
</p>
</li>
<li><p> UniProt
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Moosa Rezwani <a href="mailto:moosa.rezwani@gmail.com">moosa.rezwani@gmail.com</a> (<a href="https://orcid.org/0000-0001-6325-4444">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://rbioapi.moosa-r.com">https://rbioapi.moosa-r.com</a>
</p>
</li>
<li> <p><a href="https://github.com/moosa-r/rbioapi">https://github.com/moosa-r/rbioapi</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/moosa-r/rbioapi/issues">https://github.com/moosa-r/rbioapi/issues</a>
</p>
</li></ul>


<hr>
<h2 id='rba_connection_test'>Test if the Supported Services Are Responding</h2><span id='topic+rba_connection_test'></span>

<h3>Description</h3>

<p>Run this function to test the internet connectivity of your device and the
current status of the supported Services.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_connection_test(print_output = TRUE, diagnostics = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_connection_test_+3A_print_output">print_output</code></td>
<td>
<p>(Logical) (default = TRUE) Send the tests' output
to the console?</p>
</td></tr>
<tr><td><code id="rba_connection_test_+3A_diagnostics">diagnostics</code></td>
<td>
<p>(Logical) (default = FALSE) Show diagnostics and
detailed messages with internal information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function attempts to send a simple query to the supported services.
If the service successfully responded, you will be informed with a success
message; If not, the content of the error will be reported to you.
<br /> Please run this function if you encounter any errors while using
rbioapi. Also, if you need to contact support, kindly call this function
with 'diagnostic = TRUE' and include the output messages in your support
request.
</p>


<h3>Value</h3>

<p>Connection test for the supported servers will be displayed
in console and the results will be invisibly returned as a list.
</p>


<h3>See Also</h3>

<p>Other &quot;Helper functions&quot;: 
<code><a href="#topic+rba_options">rba_options</a>()</code>,
<code><a href="#topic+rba_pages">rba_pages</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_connection_test()


</code></pre>

<hr>
<h2 id='rba_enrichr'>A One-step Wrapper for Gene-list Enrichment Using Enrichr</h2><span id='topic+rba_enrichr'></span>

<h3>Description</h3>

<p>This function provides a convenient one-step wrapper for performing
enrichment analysis on a given gene list using Enrichr. It simplifies the
process by internally calling the necessary functions in the correct order.
See the details section for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr(
  gene_list,
  description = NULL,
  gene_set_library = "all",
  regex_library_name = TRUE,
  organism = "human",
  background_genes = NULL,
  progress_bar = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_+3A_gene_list">gene_list</code></td>
<td>
<p>A character vector with Entrez gene symbols of test genes.</p>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_description">description</code></td>
<td>
<p>(optional) A description to be associated with your
uploaded gene-set to Enrichr servers.</p>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_gene_set_library">gene_set_library</code></td>
<td>
<p>One of the:
</p>

<ol>
<li><p> &quot;all&quot; to select all of the available Enrichr gene-set libraries.
</p>
</li>
<li><p> A gene-set library name. You can retrieve the available options
for a given species using <code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a></code>.
</p>
</li>
<li><p> If regex_library_name = TRUE, A partially-matching name a regex
pattern that correspond to one or more of Enrichr library names.
</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_regex_library_name">regex_library_name</code></td>
<td>
<p>logical: if TRUE (default) the supplied
gene_set_library will be considered as a regex pattern.
If FALSE, gene_set_library will be considered as an exact match.</p>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_organism">organism</code></td>
<td>
<p>(default = &quot;human&quot;) Which model organism version of Enrichr
to use? Available options are: &quot;human&quot;, (H. sapiens &amp; M. musculus),
&quot;fly&quot; (D. melanogaster), &quot;yeast&quot; (S. cerevisiae), &quot;worm&quot; (C. elegans)
and &quot;fish&quot; (D. rerio).</p>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_background_genes">background_genes</code></td>
<td>
<p>A character vector of Entrez gene symbols of the
background genes.</p>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_progress_bar">progress_bar</code></td>
<td>
<p>logical: (default = TRUE) if multiple Enrichr libraries
are selected, should a progress bar be displayed?</p>
</td></tr>
<tr><td><code id="rba_enrichr_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will call other rba_enrichr_*** functions with the following
order:
</p>

<ol>
<li><p> (If necessary) Call <code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a></code> to obtain a list
of available libraries in Enrichr for the given organism.
</p>
</li>
<li><p> Call <code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a></code> to upload your gene-list
and obtain a 'user list ID'.
</p>
</li>
<li><p> (If necessary) Call <code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a></code> to
upload your background gene-list and obtain a 'background list ID'.
</p>
</li>
<li><p> Call <code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a></code> to perform enrichment analysis
on the gene-list against one or multiple Enrichr libraries
</p>
</li></ol>



<h3>Value</h3>

<p>A list containing data frames of the enrichment results of your
supplied gene-list against the selected Enrichr libraries.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://maayanlab.cloud/Enrichr/datasetStatistics&quot;
<br /> &quot;POST https://maayanlab.cloud/Enrichr/addList&quot;
<br /> &quot;POST https://maayanlab.cloud/speedrichr/api/addList&quot;
<br /> &quot;POST https://maayanlab.cloud/speedrichr/api/addbackground&quot;
<br /> &quot;GET https://maayanlab.cloud/Enrichr/enrich&quot;
<br /> &quot;POST https://maayanlab.cloud/speedrichr/api/backgroundenrich&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a>()</code>,
<code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a>()</code>,
<code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a>()</code>,
<code><a href="#topic+rba_enrichr_gene_map">rba_enrichr_gene_map</a>()</code>,
<code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a>()</code>,
<code><a href="#topic+rba_enrichr_view_list">rba_enrichr_view_list</a>()</code>
</p>
<p>Other &quot;Enrichment/Over-representation&quot;: 
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_enrichr(gene_list = c("TP53", "TNF", "EGFR"))

## End(Not run)

rba_enrichr(gene_list = c("TP53", "TNF", "EGFR"),
    gene_set_library = "GO_Molecular_Function_2017",
    regex_library_name = FALSE)


rba_enrichr(gene_list = c("TP53", "TNF", "EGFR"),
    gene_set_library = "go",
    regex_library_name = TRUE)


</code></pre>

<hr>
<h2 id='rba_enrichr_add_background'>Upload Background Gene-List to Enrichr</h2><span id='topic+rba_enrichr_add_background'></span>

<h3>Description</h3>

<p>In addition to the main gene list, you can also submit a background gene
list to Enrichr. This gene list can be used later to compute the
statistics of the enrichment analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr_add_background(background_genes, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_add_background_+3A_background_genes">background_genes</code></td>
<td>
<p>A character vector of Entrez gene symbols of the
background genes.</p>
</td></tr>
<tr><td><code id="rba_enrichr_add_background_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that <code><a href="#topic+rba_enrichr">rba_enrichr</a></code> provides a one-step and
more convenient way to automatically handle this and other required
function calls needed to perform gene set enrichment analysis with Enrichr.
</p>


<h3>Value</h3>

<p>A list with the unique IDs for your uploaded background gene list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://maayanlab.cloud/speedrichr/api/addbackground&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_enrichr">rba_enrichr</a></code>
</p>
<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a>()</code>,
<code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a>()</code>,
<code><a href="#topic+rba_enrichr_gene_map">rba_enrichr_gene_map</a>()</code>,
<code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a>()</code>,
<code><a href="#topic+rba_enrichr_view_list">rba_enrichr_view_list</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my_background_genes &lt;- c(
"NSUN3", "POLRMT", "NLRX1", "SFXN5", "ZC3H12C", "SLC25A39", "ARSG",
"DEFB29", "PCMTD2", "ACAA1A", "LRRC1", "2810432D09RIK", "SEPHS2",
"SAC3D1", "TMLHE", "LOC623451", "TSR2", "PLEKHA7", "GYS2", "ARHGEF12",
"HIBCH", "LYRM2", "ZBTB44", "ENTPD5", "RAB11FIP2", "LIPT1",
"INTU", "ANXA13", "KLF12", "SAT2", "GAL3ST2", "VAMP8", "FKBPL",
"AQP11", "TRAP1", "PMPCB", "TM7SF3", "RBM39", "BRI3", "KDR", "ZFP748",
"NAP1L1", "DHRS1", "LRRC56", "WDR20A", "STXBP2", "KLF1", "UFC1",
"CCDC16", "9230114K14RIK", "RWDD3", "2610528K11RIK")

rba_enrichr_add_background(background_genes = my_background_genes)


</code></pre>

<hr>
<h2 id='rba_enrichr_add_list'>Upload Your Gene-List to Enrichr</h2><span id='topic+rba_enrichr_add_list'></span>

<h3>Description</h3>

<p>This function uploads your gene list to Enrichr and retrieves a unique
'user list ID' required for performing enrichment analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr_add_list(
  gene_list,
  description = NULL,
  organism = "human",
  speedrichr = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_add_list_+3A_gene_list">gene_list</code></td>
<td>
<p>A character vector with Entrez gene symbols of test genes.</p>
</td></tr>
<tr><td><code id="rba_enrichr_add_list_+3A_description">description</code></td>
<td>
<p>(optional) A description to be associated with your
uploaded gene-set to Enrichr servers.</p>
</td></tr>
<tr><td><code id="rba_enrichr_add_list_+3A_organism">organism</code></td>
<td>
<p>(default = &quot;human&quot;) Which model organism version of Enrichr
to use? Available options are: &quot;human&quot;, (H. sapiens &amp; M. musculus),
&quot;fly&quot; (D. melanogaster), &quot;yeast&quot; (S. cerevisiae), &quot;worm&quot; (C. elegans)
and &quot;fish&quot; (D. rerio).</p>
</td></tr>
<tr><td><code id="rba_enrichr_add_list_+3A_speedrichr">speedrichr</code></td>
<td>
<p>logical (default = FALSE) Set to TRUE if you will use this
gene list with a background list; otherwise, set to FALSE. Only available
for human libraries. Refer to the details section for more information.</p>
</td></tr>
<tr><td><code id="rba_enrichr_add_list_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Enrichr uses separate APIs for analysis with or without a background gene
list. Set 'speedrichr = TRUE' if this gene list will be used with a
background gene list; otherwise, set it to FALSE. Gene lists submitted with
'speedrichr = TRUE' can only be analyzed with a background set, and those
submitted with 'speedrichr = FALSE' can only be analyzed without one.
Currently, background-based enrichment is supported only for human
libraries.
</p>
<p>Please note that <code><a href="#topic+rba_enrichr">rba_enrichr</a></code> provides a one-step and
more convenient way to automatically handle this and other required
function calls needed to perform gene set enrichment analysis with Enrichr.
</p>


<h3>Value</h3>

<p>A list with the unique IDs for your uploaded gene list. 'userListId'
is the ID required for later steps.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://maayanlab.cloud/Enrichr/addList&quot;
<br /> &quot;POST https://maayanlab.cloud/speedrichr/api/addList&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_enrichr">rba_enrichr</a></code>
</p>
<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a>()</code>,
<code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a>()</code>,
<code><a href="#topic+rba_enrichr_gene_map">rba_enrichr_gene_map</a>()</code>,
<code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a>()</code>,
<code><a href="#topic+rba_enrichr_view_list">rba_enrichr_view_list</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_enrichr_add_list(gene_list = c("TP53", "TNF", "EGFR"),
     description = "tumoral genes",
     speedrichr = FALSE)


rba_enrichr_add_list(gene_list = c("RAG1", "RAG2", "DNTT", "LIG4", "ARTEMIS"),
     description = "TCR rearrangment",
     speedrichr = TRUE)


</code></pre>

<hr>
<h2 id='rba_enrichr_enrich'>Get Enrichr Enrichment Results</h2><span id='topic+rba_enrichr_enrich'></span>

<h3>Description</h3>

<p>This function retrieves enrichment analysis results for your supplied
'user_list_id' against one or multiple Enrichr libraries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr_enrich(
  user_list_id,
  gene_set_library = "all",
  regex_library_name = TRUE,
  organism = "human",
  background_id = NULL,
  progress_bar = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_enrich_+3A_user_list_id">user_list_id</code></td>
<td>
<p>An ID returned after uploading a gene list
using <code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a></code>, with the 'speedrichr' set to
TRUE or FALSE depending on whether you intend to analyze this gene list
with or without a background gene list, respectively.</p>
</td></tr>
<tr><td><code id="rba_enrichr_enrich_+3A_gene_set_library">gene_set_library</code></td>
<td>
<p>One of the:
</p>

<ol>
<li><p> &quot;all&quot; to select all of the available Enrichr gene-set libraries.
</p>
</li>
<li><p> A gene-set library name. You can retrieve the available options
for a given species using <code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a></code>.
</p>
</li>
<li><p> If regex_library_name = TRUE, A partially-matching name a regex
pattern that correspond to one or more of Enrichr library names.
</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_enrichr_enrich_+3A_regex_library_name">regex_library_name</code></td>
<td>
<p>logical: if TRUE (default) the supplied
gene_set_library will be considered as a regex pattern.
If FALSE, gene_set_library will be considered as an exact match.</p>
</td></tr>
<tr><td><code id="rba_enrichr_enrich_+3A_organism">organism</code></td>
<td>
<p>(default = &quot;human&quot;) Which model organism version of Enrichr
to use? Available options are: &quot;human&quot;, (H. sapiens &amp; M. musculus),
&quot;fly&quot; (D. melanogaster), &quot;yeast&quot; (S. cerevisiae), &quot;worm&quot; (C. elegans)
and &quot;fish&quot; (D. rerio). If 'background_id' is provided, the only available
option is &quot;human&quot;.</p>
</td></tr>
<tr><td><code id="rba_enrichr_enrich_+3A_background_id">background_id</code></td>
<td>
<p>An ID returned after uploading a background gene
list using <code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a></code></p>
</td></tr>
<tr><td><code id="rba_enrichr_enrich_+3A_progress_bar">progress_bar</code></td>
<td>
<p>logical: (default = TRUE) if multiple Enrichr libraries
are selected, should a progress bar be displayed?</p>
</td></tr>
<tr><td><code id="rba_enrichr_enrich_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If 'background_id' is supplied, this function will interact with the
speedrichr API. In this case, 'user_list_id' must have been obtained
from a <code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a></code> call with the 'speedrichr'
parameter set to 'TRUE'. Additionally, this feature is only available
for &quot;human&quot; organism.
</p>
<p>Please note that <code><a href="#topic+rba_enrichr">rba_enrichr</a></code> provides a one-step and
more convenient way to automatically handle this and other required
function calls needed to perform gene set enrichment analysis with Enrichr.
</p>


<h3>Value</h3>

<p>A list containing data frames of the enrichment results of your
supplied gene-list against the selected Enrichr libraries.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://maayanlab.cloud/Enrichr/enrich&quot;
<br /> &quot;POST https://maayanlab.cloud/speedrichr/api/backgroundenrich&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_enrichr">rba_enrichr</a></code>
</p>
<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a>()</code>,
<code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a>()</code>,
<code><a href="#topic+rba_enrichr_gene_map">rba_enrichr_gene_map</a>()</code>,
<code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a>()</code>,
<code><a href="#topic+rba_enrichr_view_list">rba_enrichr_view_list</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_enrichr_enrich(user_list_id = 11111)

## End(Not run)
## Not run: 
rba_enrichr_enrich(user_list_id = 11111,
    gene_set_library = "GO_Molecular_Function_2017",
    regex_library_name = FALSE)

## End(Not run)
## Not run: 
rba_enrichr_enrich(user_list_id = 11111,
    gene_set_library = "go",
    regex_library_name = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_enrichr_gene_map'>Find Enrichr Terms That Contain a Given Gene</h2><span id='topic+rba_enrichr_gene_map'></span>

<h3>Description</h3>

<p>This function will search the gene and retrieve a list of Enrichr
Terms that contains that gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr_gene_map(gene, categorize = FALSE, organism = "human", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_gene_map_+3A_gene">gene</code></td>
<td>
<p>character: An Entrez gene symbol.</p>
</td></tr>
<tr><td><code id="rba_enrichr_gene_map_+3A_categorize">categorize</code></td>
<td>
<p>logical: Should the category information be included?</p>
</td></tr>
<tr><td><code id="rba_enrichr_gene_map_+3A_organism">organism</code></td>
<td>
<p>(default = &quot;human&quot;) Which model organism version of Enrichr
to use? Available options are: &quot;human&quot;, (H. sapiens &amp; M. musculus),
&quot;fly&quot; (D. melanogaster), &quot;yeast&quot; (S. cerevisiae), &quot;worm&quot; (C. elegans)
and &quot;fish&quot; (D. rerio).</p>
</td></tr>
<tr><td><code id="rba_enrichr_gene_map_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A nested list containing the search results of your supplied gene.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://maayanlab.cloud/Enrichr/genemap&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a>()</code>,
<code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a>()</code>,
<code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a>()</code>,
<code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a>()</code>,
<code><a href="#topic+rba_enrichr_view_list">rba_enrichr_view_list</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_enrichr_gene_map(gene = "p53")


rba_enrichr_gene_map(gene = "p53", categorize = TRUE)


</code></pre>

<hr>
<h2 id='rba_enrichr_libs'>Retrieve a List of available libraries from Enrichr</h2><span id='topic+rba_enrichr_libs'></span>

<h3>Description</h3>

<p>This function retrieves a list of libraries available in Enrichr along with
their associated statistics. Each library represents a collection of gene sets
that can be used for enrichment analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr_libs(organism = "human", store_in_options = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_libs_+3A_organism">organism</code></td>
<td>
<p>(default = &quot;human&quot;) Which model organism version of Enrichr
to use? Available options are: &quot;human&quot;, (H. sapiens &amp; M. musculus),
&quot;fly&quot; (D. melanogaster), &quot;yeast&quot; (S. cerevisiae), &quot;worm&quot; (C. elegans)
and &quot;fish&quot; (D. rerio).</p>
</td></tr>
<tr><td><code id="rba_enrichr_libs_+3A_store_in_options">store_in_options</code></td>
<td>
<p>logical: (default = TRUE) Should a list of available
Enrichr libraries be saved as a global option?</p>
</td></tr>
<tr><td><code id="rba_enrichr_libs_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, this function will save the library names as a global option
(&quot;rba_enrichr_libs&quot;) for other Enrichr functions that internally require
the names of Enrichr libraries. You should call this function once per
R session with the argument 'store_in_options = TRUE' before
using <code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a></code> or <code><a href="#topic+rba_enrichr">rba_enrichr</a></code>.
However, if you do not explicitly call it, rbioapi will automatically
execute this function in the background the when it is needed.
</p>
<p>Please note that <code><a href="#topic+rba_enrichr">rba_enrichr</a></code> provides a one-step and
more convenient way to automatically handle this and other required
function calls needed to perform gene set enrichment analysis with Enrichr.
</p>


<h3>Value</h3>

<p>A data frame with the names of available library in Enrichr and their
statistics.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://maayanlab.cloud/Enrichr/datasetStatistics&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_enrichr">rba_enrichr</a></code>
</p>
<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a>()</code>,
<code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a>()</code>,
<code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a>()</code>,
<code><a href="#topic+rba_enrichr_gene_map">rba_enrichr_gene_map</a>()</code>,
<code><a href="#topic+rba_enrichr_view_list">rba_enrichr_view_list</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_enrichr_libs()


</code></pre>

<hr>
<h2 id='rba_enrichr_view_list'>View an Uploaded Gene List</h2><span id='topic+rba_enrichr_view_list'></span>

<h3>Description</h3>

<p>Retrieve the list of uploaded genes with a given 'user list ID'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_enrichr_view_list(
  user_list_id,
  organism = "human",
  speedrichr = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_enrichr_view_list_+3A_user_list_id">user_list_id</code></td>
<td>
<p>a user list ID returned after uploading a gene
list using <code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a></code></p>
</td></tr>
<tr><td><code id="rba_enrichr_view_list_+3A_organism">organism</code></td>
<td>
<p>(default = &quot;human&quot;) Which model organism version of Enrichr
to use? Available options are: &quot;human&quot;, (H. sapiens &amp; M. musculus),
&quot;fly&quot; (D. melanogaster), &quot;yeast&quot; (S. cerevisiae), &quot;worm&quot; (C. elegans)
and &quot;fish&quot; (D. rerio).</p>
</td></tr>
<tr><td><code id="rba_enrichr_view_list_+3A_speedrichr">speedrichr</code></td>
<td>
<p>logical (default = FALSE) Did you upload your gene list
to speedrichr API? (i.e. did you intend to use this gene list along with
a background gene list?)</p>
</td></tr>
<tr><td><code id="rba_enrichr_view_list_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the genes and description associated to the
supplied user_list_id.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://maayanlab.cloud/Enrichr/view&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Chen, E.Y., Tan, C.M., Kou, Y. et al. Enrichr: interactive and
collaborative HTML5 gene list enrichment analysis tool. Bioinformatics
14, 128 (2013). https://doi.org/10.1186/1471-2105-14-128
</p>
</li>
<li><p> Maxim V. Kuleshov, Matthew R. Jones, Andrew D. Rouillard, Nicolas
F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, Sherry L. Jenkins,
Kathleen M. Jagodnik, Alexander Lachmann, Michael G. McDermott,
Caroline D. Monteiro, Gregory W. Gundersen, Avi Ma’ayan, Enrichr: a
comprehensive gene set enrichment analysis web server 2016 update,
Nucleic Acids Research, Volume 44, Issue W1, 8 July 2016, Pages W90–W97,
https://doi.org/10.1093/nar/gkw377
</p>
</li>
<li><p> Xie, Z., Bailey, A., Kuleshov, M. V., Clarke, D. J. B.,
Evangelista, J. E., Jenkins, S. L., Lachmann, A., Wojciechowicz, M. L.,
Kropiwnicki, E., Jagodnik, K. M., Jeon, M., &amp; Ma’ayan, A. (2021). Gene
set knowledge discovery with Enrichr. Current Protocols, 1, e90.
doi: 10.1002/cpz1.90
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#api">Enrichr API
Documentation</a>
</p>
</li>
<li> <p><a href="https://maayanlab.cloud/Enrichr/help#terms">Citations note
on Enrichr website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Enrichr&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_enrichr_add_background">rba_enrichr_add_background</a>()</code>,
<code><a href="#topic+rba_enrichr_add_list">rba_enrichr_add_list</a>()</code>,
<code><a href="#topic+rba_enrichr_enrich">rba_enrichr_enrich</a>()</code>,
<code><a href="#topic+rba_enrichr_gene_map">rba_enrichr_gene_map</a>()</code>,
<code><a href="#topic+rba_enrichr_libs">rba_enrichr_libs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_enrichr_view_list(user_list_id = 11111)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_jaspar_collections'>List collections available in JASPAR</h2><span id='topic+rba_jaspar_collections'></span>

<h3>Description</h3>

<p>JASPAR organizes matrix profiles into collections.
Using this function, you can retrieve a list of available collections
in a JASPAR release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_collections(release = 2024, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_collections_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with collections' names and URLs.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/collections/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_collections(release = 2024)


</code></pre>

<hr>
<h2 id='rba_jaspar_collections_matrices'>List matrices available in a JASPAR collection</h2><span id='topic+rba_jaspar_collections_matrices'></span>

<h3>Description</h3>

<p>Using this function you can list all matrix profiles
that are available in a collection from a JASPAR release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_collections_matrices(
  collection,
  release = 2024,
  only_last_version = FALSE,
  search = NULL,
  order = NULL,
  page_size = 1000,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_collections_matrices_+3A_collection">collection</code></td>
<td>
<p>JASPAR Collection's name. See
<a href="https://jaspar.elixir.no/docs/">JASPAR Collections</a> for
information. The accepted values are: &quot;CORE&quot;, &quot;CNE&quot;, &quot;PHYLOFACTS&quot;,
&quot;SPLICE&quot;, &quot;POLII&quot;, &quot;FAM&quot;, &quot;PBM&quot;, &quot;PBM_HOMEO&quot;, &quot;PBM_HLH&quot;, and
&quot;UNVALIDATED&quot;.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_only_last_version">only_last_version</code></td>
<td>
<p>Logical: (default = FALSE) If TRUE, only the
latest version of a matrix profile will be returned.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_search">search</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can alsoa use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_page_size">page_size</code></td>
<td>
<p>Numeric: (default = 1000) This resource returns paginated
results. What is the maximum numbers of results that you want to retrieve
per a page? Accepted values are between 1 and 1000.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_page">page</code></td>
<td>
<p>Numeric: Which page of the results to retrieve? The accepted
values depend on the page size and number of results.</p>
</td></tr>
<tr><td><code id="rba_jaspar_collections_matrices_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The results are paginated. You can control the page's size number
with the function's arguments. Also, you can use <code><a href="#topic+rba_pages">rba_pages</a></code>
to automatically iterate over multiple pages.
</p>


<h3>Value</h3>

<p>A list that contains a data frame with information of matrix
profiles available in the collection.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/collections/{collection}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_collections_matrices(collection = "CORE",
  release = 2024,
  page_size = 100,
  page = 2)


</code></pre>

<hr>
<h2 id='rba_jaspar_matrix'>Get a Position Frequency Matrices (PFM) with annotations</h2><span id='topic+rba_jaspar_matrix'></span>

<h3>Description</h3>

<p>Using this function you can retrieve a Position Frequency Matrices (PFM)
associated with a matrix profile Identifier along with its details and
annotations. If a base ID (i.e. without version suffix) was supplied,
the latest version will be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_matrix(matrix_id, file_format = NULL, save_to = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_matrix_+3A_matrix_id">matrix_id</code></td>
<td>
<p>Character: A matrix profile
Identifier. It has &quot;base_id.version&quot; naming schema.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_+3A_file_format">file_format</code></td>
<td>
<p>Character: Instead of returning a R object, you
can directly download the profile matrix in file with this format.
Supported formats are: &quot;yaml&quot;, &quot;jaspar&quot;, &quot;transfac&quot;, &quot;meme&quot; and &quot;pfm&quot;</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: (only if file_format was supplied) Save the file to an
automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file or directory path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains the PFM along with its details and
annotations. If file_format was supplied, an un-parsed character string
with the file's content.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/matrix/{matrix_id}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_matrix("MA0600.2")

## Not run: 
rba_jaspar_matrix(matrix_id = "MA0600.2",
                  file_format = "meme",
                  save_to = "my_matrix.meme")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_jaspar_matrix_search'>Search matrix profiles available in JASPAR</h2><span id='topic+rba_jaspar_matrix_search'></span>

<h3>Description</h3>

<p>You can use this function to list the JASPAR matrix
profiles that match your search query, or run the function without any
arguments to return a list of every matrix profile available in the
latest release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_matrix_search(
  term = NULL,
  tf_name = NULL,
  tf_class = NULL,
  tf_family = NULL,
  tax_group = NULL,
  tax_id = NULL,
  data_type = NULL,
  collection = NULL,
  release = 2024,
  only_last_version = FALSE,
  order = NULL,
  page_size = 1000,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_matrix_search_+3A_term">term</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_tf_name">tf_name</code></td>
<td>
<p>Character: Transcription factor names (Case-sensitive).</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_tf_class">tf_class</code></td>
<td>
<p>Character: Transcription factor class</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_tf_family">tf_family</code></td>
<td>
<p>Character: Transcription factor family</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_tax_group">tax_group</code></td>
<td>
<p>Character: Taxonomic group. Use
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a></code> to get a list of supported Taxonomic
groups.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_tax_id">tax_id</code></td>
<td>
<p>Numeric: NCBI taxonomic Identifier of species. Use
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a></code> to get a list of supported Species.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_data_type">data_type</code></td>
<td>
<p>Character: Type of the data (i.e The Methodology used
for matrix construction). For example: &quot;ChIP-seq&quot;, &quot;PBM&quot;</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_collection">collection</code></td>
<td>
<p>Character: JASPAR matrix profile collection name. USE
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a></code> to get a list of collection names.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_only_last_version">only_last_version</code></td>
<td>
<p>Logical: (default = FALSE) If TRUE, only the
latest version of a matrix profile will be returned.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can also use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_page_size">page_size</code></td>
<td>
<p>Numeric: (default = 1000) This resource returns paginated
results. What is the maximum numbers of results that you want to retrieve
per a page? Accepted values are between 1 and 1000.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_page">page</code></td>
<td>
<p>Numeric: Which page of the results to retrieve? The accepted
values depend on the page size and number of results.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br /> The results are paginated. You can control the page's size number
with the function's arguments. Also, you can use <code><a href="#topic+rba_pages">rba_pages</a></code>
to automatically iterate over multiple pages.
</p>


<h3>Value</h3>

<p>A list that contains a data frame of matrix profiles' information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/api/v1/matrix/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_matrix_search(term = "FOX")
rba_jaspar_matrix_search(tf_name = "FOXP3")
rba_jaspar_matrix_search(tf_name = "FOXP3", only_last_version = TRUE)
rba_jaspar_matrix_search(tf_class = "Zipper-Type")
rba_jaspar_matrix_search(tax_group = "insects")
rba_jaspar_matrix_search(page_size = 100)


</code></pre>

<hr>
<h2 id='rba_jaspar_matrix_versions'>List matrix profile versions associated with a base ID</h2><span id='topic+rba_jaspar_matrix_versions'></span>

<h3>Description</h3>

<p>Since JASPAR release 2010, the matrix profiles
are versioned; So, a matrix profile Identifier has &quot;base_id.version&quot;
naming schema. Using this function you can retrieve a list of matrix
profiles associated with a base (stable) ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_matrix_versions(base_id, order = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_matrix_versions_+3A_base_id">base_id</code></td>
<td>
<p>Character: A base (stable) Identifier. A matrix profile
identifier has &quot;base_id.version&quot; naming schema</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_versions_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can also use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_matrix_versions_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of matrix profiles' versions information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/matrix/{base_id}/versions/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_matrix_versions("MA0600")


</code></pre>

<hr>
<h2 id='rba_jaspar_releases'>Get information about JASPAR database releases</h2><span id='topic+rba_jaspar_releases'></span>

<h3>Description</h3>

<p>If a release number was supplied, this function will return the details
of that release. Otherwise, if the function was called without &quot;release&quot;
argument, a list of all JASPAR database releases will be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_releases(release_number = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_releases_+3A_release_number">release_number</code></td>
<td>
<p>Numeric: Which JASPAR database release number
information's to retrieve? If left NULL (the default), a list of all
JASPAR database releases will be returned. Available options are 1 to 8.</p>
</td></tr>
<tr><td><code id="rba_jaspar_releases_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains all JASPAR database releases' information or
details of a particular release.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/releases/&quot;
<br /> &quot;GET https://jaspar.elixir.no/api/v1/releases/{release_number}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_releases()
rba_jaspar_releases(7)


</code></pre>

<hr>
<h2 id='rba_jaspar_sites'>Get binding sites of a matrix profile</h2><span id='topic+rba_jaspar_sites'></span>

<h3>Description</h3>

<p>Use this function to retrieve a list of transcription factor binding sites
associated with a matrix profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_sites(matrix_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_sites_+3A_matrix_id">matrix_id</code></td>
<td>
<p>Character: A matrix profile
Identifier. It has &quot;base_id.version&quot; naming schema.</p>
</td></tr>
<tr><td><code id="rba_jaspar_sites_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains a data frame with binding sites information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/sites/{matrix_id}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_sites("MA0600.1")


</code></pre>

<hr>
<h2 id='rba_jaspar_species'>List available species in JASPAR</h2><span id='topic+rba_jaspar_species'></span>

<h3>Description</h3>

<p>JASPAR organizes matrix profiles from multiple species
in six taxonomic groups. Use this function to retrieve a list of
available species in a JASPAR database release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_species(release = 2024, search = NULL, order = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_species_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_+3A_search">search</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can also use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with information of available species.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/species/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_species(release = 2024)


</code></pre>

<hr>
<h2 id='rba_jaspar_species_matrices'>List matrices available in JASPAR of a species</h2><span id='topic+rba_jaspar_species_matrices'></span>

<h3>Description</h3>

<p>JASPAR curates matrix profiles from multiple species
in six taxonomic groups. Using this function you can list all
matrix profiles that are available in a JASPAR  release from a species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_species_matrices(
  tax_id,
  release = 2024,
  only_last_version = FALSE,
  search = NULL,
  order = NULL,
  page_size = 1000,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_species_matrices_+3A_tax_id">tax_id</code></td>
<td>
<p>Numeric: NCBI taxonomic Identifier of species. Use
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a></code> to get a list of supported Species.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_only_last_version">only_last_version</code></td>
<td>
<p>Logical: (default = FALSE) If TRUE, only the
latest version of a matrix profile will be returned.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_search">search</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can also use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_page_size">page_size</code></td>
<td>
<p>Numeric: (default = 1000) This resource returns paginated
results. What is the maximum numbers of results that you want to retrieve
per a page? Accepted values are between 1 and 1000.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_page">page</code></td>
<td>
<p>Numeric: Which page of the results to retrieve? The accepted
values depend on the page size and number of results.</p>
</td></tr>
<tr><td><code id="rba_jaspar_species_matrices_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The results are paginated. You can control the page's size number
with the function's arguments. Also, you can use <code><a href="#topic+rba_pages">rba_pages</a></code>
to automatically iterate over multiple pages.
</p>


<h3>Value</h3>

<p>A list that contains a data frame with information of matrix
profiles available for the species.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/species/{tax_id}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_species_matrices(tax_id = 9606, page_size = 100)


</code></pre>

<hr>
<h2 id='rba_jaspar_taxons'>List available taxonomic groups in JASPAR</h2><span id='topic+rba_jaspar_taxons'></span>

<h3>Description</h3>

<p>JASPAR organizes matrix profiles from multiple species
in six taxonomic groups. Use this function to retrieve a list of
available taxonomic groups in a JASPAR database release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_taxons(release = 2024, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_taxons_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with information of available species.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/taxon/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_taxons(release = 2024)


</code></pre>

<hr>
<h2 id='rba_jaspar_taxons_matrices'>List matrices available in JASPAR of a taxonomic group</h2><span id='topic+rba_jaspar_taxons_matrices'></span>

<h3>Description</h3>

<p>JASPAR organizes matrix profiles from multiple species
in six taxonomic groups. Using this function you can list all
matrix profiles that are available in a JASPAR release from a
taxonomic group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_taxons_matrices(
  tax_group,
  release = 2024,
  only_last_version = FALSE,
  search = NULL,
  order = NULL,
  page_size = 1000,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_tax_group">tax_group</code></td>
<td>
<p>Character: Taxonomic group. Use
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a></code> to get a list of supported Taxonomic
groups.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_only_last_version">only_last_version</code></td>
<td>
<p>Logical: (default = FALSE) If TRUE, only the
latest version of a matrix profile will be returned.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_search">search</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can also use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_page_size">page_size</code></td>
<td>
<p>Numeric: (default = 1000) This resource returns paginated
results. What is the maximum numbers of results that you want to retrieve
per a page? Accepted values are between 1 and 1000.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_page">page</code></td>
<td>
<p>Numeric: Which page of the results to retrieve? The accepted
values depend on the page size and number of results.</p>
</td></tr>
<tr><td><code id="rba_jaspar_taxons_matrices_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The results are paginated. You can control the page's size number
with the function's arguments. Also, you can use <code><a href="#topic+rba_pages">rba_pages</a></code>
to automatically iterate over multiple pages.
</p>


<h3>Value</h3>

<p>A list that contains a data frame with information of matrix
profiles available for the taxonomic group.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/taxon/{tax_group}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_taxons_matrices(tax_group = "plants", page_size = 100)


</code></pre>

<hr>
<h2 id='rba_jaspar_tffm'>Get a TF flexible models (TFFMs) information</h2><span id='topic+rba_jaspar_tffm'></span>

<h3>Description</h3>

<p>Using this function you can retrieve details and annotations of
Transcription Factor flexible models (TFFMs) associated with a TFFM
ID. If a base ID (i.e. without version suffix) was supplied, the latest
version will be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_tffm(tffm_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_tffm_+3A_tffm_id">tffm_id</code></td>
<td>
<p>Character: A TF flexible model (TFFM) Identifier.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains the TFFM's information and annotations.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/fttm/{tffm_id}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm_search">rba_jaspar_tffm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_tffm("TFFM0056.3")


</code></pre>

<hr>
<h2 id='rba_jaspar_tffm_search'>Search TF flexible models (TFFMs) available in JASPAR</h2><span id='topic+rba_jaspar_tffm_search'></span>

<h3>Description</h3>

<p>You can use this function to list the JASPAR TF flexible models (TFFMs)
that match your search query, or run the function without any
arguments to return a list of every matrix profile available in the
latest release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_jaspar_tffm_search(
  term = NULL,
  release = 2024,
  tax_group = NULL,
  search = NULL,
  order = NULL,
  page_size = 1000,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_jaspar_tffm_search_+3A_term">term</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_release">release</code></td>
<td>
<p>Numeric: (default = 2024) Which JASPAR database release
to use? Available options are: 2024, 2022, 2020, 2018, 2016, and 2014.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_tax_group">tax_group</code></td>
<td>
<p>Character: Taxonomic group. Use
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a></code> to get a list of supported Taxonomic
groups.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_search">search</code></td>
<td>
<p>Character: A search term.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_order">order</code></td>
<td>
<p>Character: A character string or a vector of character strings
of field names that will be used to order the results.
<br /> Providing multiple field names is supported. You can also use prefix
&quot;-&quot; before a field name to indicate reverse ordering.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_page_size">page_size</code></td>
<td>
<p>Numeric: (default = 1000) This resource returns paginated
results. What is the maximum numbers of results that you want to retrieve
per a page? Accepted values are between 1 and 1000.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_page">page</code></td>
<td>
<p>Numeric: Which page of the results to retrieve? The accepted
values depend on the page size and number of results.</p>
</td></tr>
<tr><td><code id="rba_jaspar_tffm_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br /> The results are paginated. You can control the page's size number
with the function's arguments. Also, you can use <code><a href="#topic+rba_pages">rba_pages</a></code>
to automatically iterate over multiple pages.
</p>


<h3>Value</h3>

<p>A list that contains a data frame with information of query hits'
TFFMs.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://jaspar.elixir.no/api/v1/api/v1/tffm/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A JASPAR 2024:
20th anniversary of the open-access database of transcription factor
binding profiles Nucleic Acids Res. in_press; doi: 10.1093/nar/gkad1059
</p>
</li>
<li><p> Khan, A. and Mathelier, A. JASPAR RESTful API: accessing JASPAR data
from any programming language. Bioinformatics, 2017,
doi: 10.1093/bioinformatics/btx804
</p>
</li>
<li>
<p><a href="https://jaspar.elixir.no/api/v1/docs/">JASPAR API Documentation</a>
</p>
</li>
<li> <p><a href="https://jaspar.elixir.no/faq/">Citations note
on JASPAR website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;JASPAR&quot;: 
<code><a href="#topic+rba_jaspar_collections">rba_jaspar_collections</a>()</code>,
<code><a href="#topic+rba_jaspar_collections_matrices">rba_jaspar_collections_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix">rba_jaspar_matrix</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_search">rba_jaspar_matrix_search</a>()</code>,
<code><a href="#topic+rba_jaspar_matrix_versions">rba_jaspar_matrix_versions</a>()</code>,
<code><a href="#topic+rba_jaspar_releases">rba_jaspar_releases</a>()</code>,
<code><a href="#topic+rba_jaspar_sites">rba_jaspar_sites</a>()</code>,
<code><a href="#topic+rba_jaspar_species">rba_jaspar_species</a>()</code>,
<code><a href="#topic+rba_jaspar_species_matrices">rba_jaspar_species_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons">rba_jaspar_taxons</a>()</code>,
<code><a href="#topic+rba_jaspar_taxons_matrices">rba_jaspar_taxons_matrices</a>()</code>,
<code><a href="#topic+rba_jaspar_tffm">rba_jaspar_tffm</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_jaspar_tffm_search(term = "FOX")
rba_jaspar_tffm_search(tax_group = "insects")
rba_jaspar_tffm_search(page_size = 100)


</code></pre>

<hr>
<h2 id='rba_mieaa_cats'>Get Supported Enrichment Categories for a Species and miRNA Type</h2><span id='topic+rba_mieaa_cats'></span>

<h3>Description</h3>

<p>For each Combination of species and miRNA type, Only a pre-defined
categories groups are supported. Use this function to retrieve a list
of supported categories for a given combination of Species and miRNA type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_cats(mirna_type, species, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_cats_+3A_mirna_type">mirna_type</code></td>
<td>
<p>Type of your miRNA accession. either &quot;mature&quot; or
&quot;precursor&quot;.</p>
</td></tr>
<tr><td><code id="rba_mieaa_cats_+3A_species">species</code></td>
<td>
<p>Fully or partially matching Scientific name, abbreviation
or NCBI taxon ID of one of the following species: </p>

<ol>
<li><p> &quot;Homo sapiens&quot;, &quot;hsa&quot; or 9606
</p>
</li>
<li><p> &quot;Mus musculus&quot;, &quot;mmu&quot; or 10090
</p>
</li>
<li><p> &quot;Rattus norvegicus&quot;, &quot;rno&quot; or 10116
</p>
</li>
<li><p> &quot;Arabidopsis thaliana&quot;, &quot;ath&quot; or 3702
</p>
</li>
<li><p> &quot;Bos taurus&quot;, &quot;bta&quot; or 9913
</p>
</li>
<li><p> &quot;Caenorhabditis elegans&quot;, &quot;cel&quot; or 6239
</p>
</li>
<li><p> &quot;Drosophila melanogaster&quot;, &quot;dme&quot; or 7227
</p>
</li>
<li><p> &quot;Danio rerio&quot;, &quot;dre&quot; or 7955
</p>
</li>
<li><p> &quot;Gallus gallus&quot;, &quot;gga&quot; or 9031
</p>
</li>
<li><p> &quot;Sus scrofa&quot;, &quot;ssc&quot; or  9823</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_mieaa_cats_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named character vector with the supported categories for
your supplied input combination.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/v1/enrichment_categories/{species}/{mirna_type}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_convert_type">rba_mieaa_convert_type</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_mieaa_cats("mature", "Homo sapiens")


</code></pre>

<hr>
<h2 id='rba_mieaa_convert_type'>Convert Between Mature and precursor miRNA Accession</h2><span id='topic+rba_mieaa_convert_type'></span>

<h3>Description</h3>

<p>miRBase miRNA accession could refer to either mature or precursor miRNAs.
(see: <a href="https://rnajournal.cshlp.org/content/9/3/277">A uniform system
for microRNA annotation</a>). Use this function to mature miRNA accession to
corresponding miRNA accessions or vice versa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_convert_type(
  mirna,
  input_type,
  only_unique = FALSE,
  simple_output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_convert_type_+3A_mirna">mirna</code></td>
<td>
<p>A vector of miRNA accessions to be converted.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_type_+3A_input_type">input_type</code></td>
<td>
<p>Type of your supplied miRNA accession. either &quot;mature&quot;
or &quot;precursor&quot;.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_type_+3A_only_unique">only_unique</code></td>
<td>
<p>(logical) miRBase precursor and mature miRNA accessions
are not uniquely mapped. (i.e. you may get more than one results for
a given accession). set this to TRUE to only retrieve the unique mappings.
(default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_type_+3A_simple_output">simple_output</code></td>
<td>
<p>(logical) If FALSE (default), the result will be a
two-columned data frame with your input and output accessions. Otherwise,
if TRUE, only the output miRNA accessions will be returned.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_type_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the arguments, a data frame or a character vectors
containing the miRNA accessions in your output version.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/v1/mirna_precursor_converter/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
Sys.sleep(1) # to prevent 429 error during R CMD check
rba_mieaa_convert_type(mirna = c("hsa-miR-20b-5p", "hsa-miR-144-5p"),
    input_type = "mature")


</code></pre>

<hr>
<h2 id='rba_mieaa_convert_version'>Convert miRNA accession Between Different miRBase Versions</h2><span id='topic+rba_mieaa_convert_version'></span>

<h3>Description</h3>

<p>miEAA works with miRBASE v22 accession. Using This function you can convert
a set of mature or precursor miRNA accession between two given miRBase
versions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_convert_version(
  mirna,
  mirna_type,
  input_version,
  output_version,
  simple_output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_convert_version_+3A_mirna">mirna</code></td>
<td>
<p>A vector of miRNA accessions to be converted.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_version_+3A_mirna_type">mirna_type</code></td>
<td>
<p>Type of your supplied miRNA accession. either &quot;mature&quot;
or &quot;precursor&quot;.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_version_+3A_input_version">input_version</code></td>
<td>
<p>(numeric) miRBase version of your supplied miRNA
accessions.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_version_+3A_output_version">output_version</code></td>
<td>
<p>(numeric) To what version should your miRNA accessions
be converted?</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_version_+3A_simple_output">simple_output</code></td>
<td>
<p>(logical) If FALSE (default), the result will be a
two-columned data frame with your input and output accessions. Otherwise,
if TRUE, only the output miRNA accessions will be returned.</p>
</td></tr>
<tr><td><code id="rba_mieaa_convert_version_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the arguments, a data frame or a character vectors
containing the miRNA accessions in your output version.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/v1/mirbase_converter/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_type">rba_mieaa_convert_type</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
Sys.sleep(1) # to prevent 429 error during R CMD check
rba_mieaa_convert_version(mirna = c("hsa-miR-20b-5p", "hsa-miR-144-5p"),
    mirna_type = "mature", input_version = 22, output_version =  16)


</code></pre>

<hr>
<h2 id='rba_mieaa_enrich'>A One-step Wrapper for miRNA Enrichment Using miEAA</h2><span id='topic+rba_mieaa_enrich'></span>

<h3>Description</h3>

<p>This function is a wrapper for the multiple function calls necessary to
perform enrichment analysis on a given miRNA list using miEAA. see details
section for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_enrich(
  test_set,
  mirna_type,
  test_type,
  species,
  categories = NULL,
  p_adj_method = "fdr",
  independent_p_adj = TRUE,
  sig_level = 0.05,
  min_hits = 2,
  ref_set = NULL,
  sort_by = "p_adjusted",
  sort_asc = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_enrich_+3A_test_set">test_set</code></td>
<td>
<p>a character vector with your mature or precursor miRBase
miRNA accessions. Note that </p>

<ol>
<li><p> Only miRBase v22 miRNA accession are accepted. You can use
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a></code> to convert your accessions to
miRBase v22.
</p>
</li>
<li><p> Your list should be entirely consisted of either mature or
precursor miRNA accession. A mixture of both is not accepted.
</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_mirna_type">mirna_type</code></td>
<td>
<p>Type of your supplied miRNA accession. either &quot;mature&quot;
or &quot;precursor&quot;.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_test_type">test_type</code></td>
<td>
<p>The analysis to perform. can be either &quot;ORA&quot; for 'Over
Representation Analysis' or &quot;GSEA&quot; for miRNA (Gene)
'Set Enrichment Analysis'. Note that in GSEA, your list should be sorted
beforehand based on some criterion.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_species">species</code></td>
<td>
<p>Fully or partially matching Scientific name, abbreviation
or NCBI taxon ID of one of the following species: </p>

<ol>
<li><p> &quot;Homo sapiens&quot;, &quot;hsa&quot; or 9606
</p>
</li>
<li><p> &quot;Mus musculus&quot;, &quot;mmu&quot; or 10090
</p>
</li>
<li><p> &quot;Rattus norvegicus&quot;, &quot;rno&quot; or 10116
</p>
</li>
<li><p> &quot;Arabidopsis thaliana&quot;, &quot;ath&quot; or 3702
</p>
</li>
<li><p> &quot;Bos taurus&quot;, &quot;bta&quot; or 9913
</p>
</li>
<li><p> &quot;Caenorhabditis elegans&quot;, &quot;cel&quot; or 6239
</p>
</li>
<li><p> &quot;Drosophila melanogaster&quot;, &quot;dme&quot; or 7227
</p>
</li>
<li><p> &quot;Danio rerio&quot;, &quot;dre&quot; or 7955
</p>
</li>
<li><p> &quot;Gallus gallus&quot;, &quot;gga&quot; or 9031
</p>
</li>
<li><p> &quot;Sus scrofa&quot;, &quot;ssc&quot; or  9823
</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_categories">categories</code></td>
<td>
<p>one or multiple Category names to be used for miRNA set
enrichment analysis. Note that </p>

<ul>
<li><p> Available categories varies based on your chosen specie and if
your supplied miRNA type is mature or precursor. Use
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a></code> to retrieve a list of available category
names for a given specie and miRNA type.
</p>
</li>
<li><p> If you supply NULL, the analysis will be performed on all of the
available categories.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_p_adj_method">p_adj_method</code></td>
<td>
<p>P-value adjustment method to be used. Should be one of:
&quot;none&quot;, &quot;fdr&quot; (default), &quot;bonferroni&quot;, &quot;BY&quot;, &quot;hochberg&quot;, &quot;holm&quot; or &quot;hommel&quot;</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_independent_p_adj">independent_p_adj</code></td>
<td>
<p>(logical) The scope and level of p-value adjustment;
if TRUE (default), the categories will be considered independent from
each other and the p-value will be adjusted separately for each category.
if FALSE, the p-value will be adjusted collectively over all categories.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_sig_level">sig_level</code></td>
<td>
<p>(numeric) The significance threshold of adjusted P-value.
values equal to or greater than this threshold will be dropped from the
results.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_min_hits">min_hits</code></td>
<td>
<p>(numeric) How many miRNA should a sub-category have from
your supplied test-list to be included in the results? (default is 2)</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_ref_set">ref_set</code></td>
<td>
<p>(Optional) Only applicable when test_type is &quot;ORA&quot;.
This character vector will be used as your reference (background or
universe) set for p-value calculations.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_sort_by">sort_by</code></td>
<td>
<p>A column name to the result's table based on that. one of:
&quot;category&quot;, &quot;subcategory&quot;, &quot;enrichment&quot;, &quot;p_value&quot;, &quot;p_adjusted&quot; (default),
&quot;q_value&quot; or &quot;observed&quot; .</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_sort_asc">sort_asc</code></td>
<td>
<p>(logical) If TRUE, the results will be sorted in ascending
order. If FALSE, the results will be sorted in descending order.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will call other rba_mieaa_*** functions with the following
order:
</p>

<ol>
<li><p> Call <code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a></code> to Submit an enrichment
analysis request to miEAA servers, using your supplied miRNA lists and
other arguments.
</p>
</li>
<li><p> Once your job was successfully submitted, it will call
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a></code> every 5 seconds, to check the
status of your running server-side job and whether your analysis job is
finished and the results are available.
</p>
</li>
<li><p> Call <code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a></code> to retrieve the results
of your enrichment analysis.
</p>
</li></ol>

<p>See each function's manual for more details.
</p>


<h3>Value</h3>

<p>A data frame with your enrichment analysis results.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_type">rba_mieaa_convert_type</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a>()</code>
</p>
<p>Other &quot;Enrichment/Over-representation&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_mieaa_enrich(test_set = c("hsa-miR-20b-5p", "hsa-miR-144-5p",
 "hsa-miR-17-5p", "hsa-miR-20a-5p"),
     mirna_type = "mature",
     test_type = "ORA",
     species = 9606,
     categories = "miRPathDB_GO_Biological_process_mature")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_mieaa_enrich_results'>Retrieve Results of a finished Enrichment Analysis from miEAA</h2><span id='topic+rba_mieaa_enrich_results'></span>

<h3>Description</h3>

<p>After your submitted enrichment analysis request has finished (check
using <code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a></code>), you can retrieve the results
using this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_enrich_results(job_id, sort_by = "p_adjusted", sort_asc = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_enrich_results_+3A_job_id">job_id</code></td>
<td>
<p>The job-id (a character string) of a submitted enrichment
analysis.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_results_+3A_sort_by">sort_by</code></td>
<td>
<p>A column name to the result's table based on that. one of:
&quot;category&quot;, &quot;subcategory&quot;, &quot;enrichment&quot;, &quot;p_value&quot;, &quot;p_adjusted&quot; (default),
&quot;q_value&quot; or &quot;observed&quot; .</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_results_+3A_sort_asc">sort_asc</code></td>
<td>
<p>(logical) If TRUE, the results will be sorted in ascending
order. If FALSE, the results will be sorted in descending order.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_results_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that using <code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a></code> is a more convenient way to
automatically perform this and other required function calls to
perform enrichment analysis on your input miRNA-set using miEAA.
</p>


<h3>Value</h3>

<p>A data frame with your enrichment analysis results.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/v1/results/{job_id}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_type">rba_mieaa_convert_type</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_mieaa_enrich_results("f52d1aef-6d3d-4d51-9020-82e68fe99012")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_mieaa_enrich_status'>Check Status of a Submitted Enrichment Analysis in miEAA</h2><span id='topic+rba_mieaa_enrich_status'></span>

<h3>Description</h3>

<p>After you have submitted your enrichment analysis (using
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a></code>) and retrieved a job-id,
you can use this function to check the status of your job. Status value
equal to 100 means that your requested analysis has finished and you may
retrieve the results using <code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_enrich_status(job_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_enrich_status_+3A_job_id">job_id</code></td>
<td>
<p>The job-id (a character string) of a submitted enrichment
analysis.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_status_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that using <code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a></code> is a more convenient way to
automatically perform this and other required function calls to
perform enrichment analysis on your input miRNA-set using miEAA.
</p>


<h3>Value</h3>

<p>A list containing the status value for a analysis that corresponds
to your supplied job-id.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/v1/job_status/{job_id}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_type">rba_mieaa_convert_type</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_submit">rba_mieaa_enrich_submit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Sys.sleep(1) # to prevent 429 error during R CMD check
rba_mieaa_enrich_status("f52d1aef-6d3d-4d51-9020-82e68fe99012")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_mieaa_enrich_submit'>Submit miEAA miRNA Enrichment Analysis Request</h2><span id='topic+rba_mieaa_enrich_submit'></span>

<h3>Description</h3>

<p>Using This function you can submit a request in miEAA servers to perform
Over-representation or GSEA Analysis for a given set of miRNA identifiers.
see &quot;arguments&quot; section for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_mieaa_enrich_submit(
  test_set,
  mirna_type,
  test_type,
  species = "hsa",
  categories = NULL,
  p_adj_method = "fdr",
  independent_p_adj = TRUE,
  sig_level = 0.05,
  min_hits = 2,
  ref_set = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_mieaa_enrich_submit_+3A_test_set">test_set</code></td>
<td>
<p>a character vector with your mature or precursor miRBase
miRNA accessions. Note that </p>

<ol>
<li><p> Only miRBase v22 miRNA accession are accepted. You can use
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a></code> to convert your accessions to
miRBase v22.
</p>
</li>
<li><p> Your list should be entirely consisted of either mature or
precursor miRNA accession. A mixture of both is not accepted.
</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_mirna_type">mirna_type</code></td>
<td>
<p>Type of your supplied miRNA accession. either &quot;mature&quot;
or &quot;precursor&quot;.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_test_type">test_type</code></td>
<td>
<p>The analysis to perform. can be either &quot;ORA&quot; for 'Over
Representation Analysis' or &quot;GSEA&quot; for miRNA (Gene)
'Set Enrichment Analysis'. Note that in GSEA, your list should be sorted
beforehand based on some criterion.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_species">species</code></td>
<td>
<p>Fully or partially matching Scientific name, abbreviation
or NCBI taxon ID of one of the following species: </p>

<ol>
<li><p> &quot;Homo sapiens&quot;, &quot;hsa&quot; or 9606
</p>
</li>
<li><p> &quot;Mus musculus&quot;, &quot;mmu&quot; or 10090
</p>
</li>
<li><p> &quot;Rattus norvegicus&quot;, &quot;rno&quot; or 10116
</p>
</li>
<li><p> &quot;Arabidopsis thaliana&quot;, &quot;ath&quot; or 3702
</p>
</li>
<li><p> &quot;Bos taurus&quot;, &quot;bta&quot; or 9913
</p>
</li>
<li><p> &quot;Caenorhabditis elegans&quot;, &quot;cel&quot; or 6239
</p>
</li>
<li><p> &quot;Drosophila melanogaster&quot;, &quot;dme&quot; or 7227
</p>
</li>
<li><p> &quot;Danio rerio&quot;, &quot;dre&quot; or 7955
</p>
</li>
<li><p> &quot;Gallus gallus&quot;, &quot;gga&quot; or 9031
</p>
</li>
<li><p> &quot;Sus scrofa&quot;, &quot;ssc&quot; or  9823
</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_categories">categories</code></td>
<td>
<p>one or multiple Category names to be used for miRNA set
enrichment analysis. Note that </p>

<ul>
<li><p> Available categories varies based on your chosen specie and if
your supplied miRNA type is mature or precursor. Use
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a></code> to retrieve a list of available category
names for a given specie and miRNA type.
</p>
</li>
<li><p> If you supply NULL, the analysis will be performed on all of the
available categories.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_p_adj_method">p_adj_method</code></td>
<td>
<p>P-value adjustment method to be used. Should be one of:
&quot;none&quot;, &quot;fdr&quot; (default), &quot;bonferroni&quot;, &quot;BY&quot;, &quot;hochberg&quot;, &quot;holm&quot; or &quot;hommel&quot;</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_independent_p_adj">independent_p_adj</code></td>
<td>
<p>(logical) The scope and level of p-value adjustment;
if TRUE (default), the categories will be considered independent from
each other and the p-value will be adjusted separately for each category.
if FALSE, the p-value will be adjusted collectively over all categories.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_sig_level">sig_level</code></td>
<td>
<p>(numeric) The significance threshold of adjusted P-value.
values equal to or greater than this threshold will be dropped from the
results.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_min_hits">min_hits</code></td>
<td>
<p>(numeric) How many miRNA should a sub-category have from
your supplied test-list to be included in the results? (default is 2)</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_ref_set">ref_set</code></td>
<td>
<p>(Optional) Only applicable when test_type is &quot;ORA&quot;.
This character vector will be used as your reference (background or
universe) set for p-value calculations.</p>
</td></tr>
<tr><td><code id="rba_mieaa_enrich_submit_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that using <code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a></code> is a more convenient way to
automatically perform this and other required function calls to
perform enrichment analysis on your input miRNA-set using miEAA.
</p>


<h3>Value</h3>

<p>A list that contains your submitted job's ID and a URL to
manually check for your job status.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://ccb-compute2.cs.uni-saarland.de/mieaa2/api/v1/enrichment_analysis/{species}/{type}/{test}/&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Fabian Kern, Tobias Fehlmann, Jeffrey Solomon, Louisa Schwed,
Nadja Grammes, Christina Backes, Kendall Van Keuren-Jensen,
David Wesley Craig,Eckart Meese, Andreas Keller, miEAA 2.0:
integrating multi-species microRNA enrichment analysis and workflow
management systems, Nucleic Acids Research, Volume 48, Issue W1,
02 July 2020, Pages W521–W528, https://doi.org/10.1093/nar/gkaa309
</p>
</li>
<li>
<p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/tutorial/api/">miEAA
browsable API tutorial</a>
</p>
</li>
<li> <p><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2/">Citations note
on miEAA website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;miEAA&quot;: 
<code><a href="#topic+rba_mieaa_cats">rba_mieaa_cats</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_type">rba_mieaa_convert_type</a>()</code>,
<code><a href="#topic+rba_mieaa_convert_version">rba_mieaa_convert_version</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_results">rba_mieaa_enrich_results</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich_status">rba_mieaa_enrich_status</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
Sys.sleep(1) # to prevent 429 error during R CMD check
rba_mieaa_enrich_submit(test_set = c("hsa-miR-20b-5p", "hsa-miR-144-5p"),
    mirna_type = "mature",
    test_type = "GSEA",
    species = 9606,
    categories = NULL)


</code></pre>

<hr>
<h2 id='rba_options'>Set rbioapi Global Options</h2><span id='topic+rba_options'></span>

<h3>Description</h3>

<p>A safe way to change rbioapi's global options and behavior. see &quot;arguments&quot;
section for available options.
<br /> Note that you are not limited to changing the options globally, you can
include the option names and values in the '...' argument of any rbioapi
function to alter the option(s) only in that function call;
e.g. example_function(x, diagnostics = TRUE, timeout = 300).
<br /> Alternatively, you can call this function with no arguments, i.e.
rba_options(), to retrieve a data frame of available rbioapi options and
their current values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_options(
  diagnostics = NULL,
  dir_name = NULL,
  retry_max = NULL,
  retry_wait = NULL,
  progress = NULL,
  save_file = NULL,
  skip_error = NULL,
  timeout = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_options_+3A_diagnostics">diagnostics</code></td>
<td>
<p>(Logical) (default = FALSE) Show diagnostics and
detailed messages with internal information.</p>
</td></tr>
<tr><td><code id="rba_options_+3A_dir_name">dir_name</code></td>
<td>
<p>(character) (default = &quot;rbioapi&quot;) If the package needs to
generate a file path to save the server's response, a directory with this
name will be created in your working directory to save your files.</p>
</td></tr>
<tr><td><code id="rba_options_+3A_retry_max">retry_max</code></td>
<td>
<p>(Numeric) (default = 0) How many times should rbioapi
retry in case of 5xx server responses, errors related to the server
or no internet connectivity?</p>
</td></tr>
<tr><td><code id="rba_options_+3A_retry_wait">retry_wait</code></td>
<td>
<p>(Numeric) (default = 10) Time in seconds to wait before
next retry in case of internet connection or server problems.</p>
</td></tr>
<tr><td><code id="rba_options_+3A_progress">progress</code></td>
<td>
<p>(Logical) (default = FALSE) Should a progress bar be
displayed?</p>
</td></tr>
<tr><td><code id="rba_options_+3A_save_file">save_file</code></td>
<td>
<p>(Logical or character) (default = FALSE) Either:</p>

<ul>
<li><p> TRUE: In this case, the raw server's response file will be
automatically saved to a proper file path. use &quot;dir_name&quot; argument to
change the file's parent directory.
</p>
</li>
<li><p> FALSE: (default) Do not automatically save server's response file.
</p>
</li>
<li><p> Character: (Only when changing the option via &quot;...&quot; in
a functions call) A valid file path to save the server's response
file to the function that you are calling.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_options_+3A_skip_error">skip_error</code></td>
<td>
<p>(Logical) (default = FALSE if R is in the interactive mode,
TRUE otherwise) If TRUE, the code execution  will not be stopped in case
of errors (anything but HTTP status 200 from the server); Instead the
error message will be returned as the function's output. However, if FALSE,
in case of any error, the code execution will be halted and an error
message will be issued.</p>
</td></tr>
<tr><td><code id="rba_options_+3A_timeout">timeout</code></td>
<td>
<p>(Numeric) (default = 90) The maximum time in seconds that
you are willing to wait for a server response before giving up and
stopping the function execution.</p>
</td></tr>
<tr><td><code id="rba_options_+3A_verbose">verbose</code></td>
<td>
<p>(Logical) (Default = TRUE) Generate short informative
messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because this function validates your supplied changes, please
<strong><em>only change rbioapi options using this function</em></strong> and avoid
directly editing them.
</p>


<h3>Value</h3>

<p>If called without any argument, a Data frame with available options
and their information; If Called with an argument, will Return
NULL but Alters that option globally.
</p>


<h3>See Also</h3>

<p>Other &quot;Helper functions&quot;: 
<code><a href="#topic+rba_connection_test">rba_connection_test</a>()</code>,
<code><a href="#topic+rba_pages">rba_pages</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rba_options()
## Not run: 
rba_options(verbose = FALSE)

## End(Not run)
## Not run: 
rba_options(save_file = TRUE)

## End(Not run)
## Not run: 
rba_options(diagnostics = TRUE, progress = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_pages'>Get Multiple Pages of a Paginated Resource</h2><span id='topic+rba_pages'></span>

<h3>Description</h3>

<p>Some resources return paginated results, meaning that you have to make
separate calls for each page. Using this function, you can iterate over
up to 100 pages. Just supply your rbioapi function and change to page
argument to &quot;pages:start_page:end_page&quot;, for example &quot;pages:1:5&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_pages(input_call, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_pages_+3A_input_call">input_call</code></td>
<td>
<p>A quoted call. supply a regular rbioapi function call,
but with two differences:</p>

<ol>
<li><p>: Wrap a quote() around it. meaning: quote(rba_example())
</p>
</li>
<li><p>: Set the argument that corresponds to the page number to
&quot;pages:start_page:end_page&quot;, for example &quot;pages:1:5&quot;.</p>
</li></ol>

<p>See the &quot;examples&quot; section to learn more.</p>
</td></tr>
<tr><td><code id="rba_pages_+3A_...">...</code></td>
<td>
<p>Experimental internal options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To prevent flooding the server, there will be a 1 second delay between
calls, also the user cannot iterate on more than 100 pages. The function
will also override skip_error option and will always set it to TRUE.
This means that in case of server response error (e.g. requesting pages
that do not exist) an error message be returned to you instead of
halting function's execution.
</p>


<h3>Value</h3>

<p>A named list where each element corresponds to a request's page.
</p>


<h3>See Also</h3>

<p>Other &quot;Helper functions&quot;: 
<code><a href="#topic+rba_connection_test">rba_connection_test</a>()</code>,
<code><a href="#topic+rba_options">rba_options</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_pages(input_call = quote(rba_uniprot_taxonomy(ids = 189831,
    hierarchy = "siblings",
    page_size = 50,
    page_number = "pages:1:5")))


rba_pages(input_call = quote(rba_uniprot_taxonomy_name(name = "adenovirus",
    field = "scientific",
    search_type = "contain",
    page_size = 200,
    page_number = "pages:1:5",
    verbose = FALSE)))


rba_pages(input_call = quote(rba_panther_info(what = "families",
    families_page = "pages:9:11")))


</code></pre>

<hr>
<h2 id='rba_panther_enrich'>PANTHER Over-Representation or Enrichment Analysis</h2><span id='topic+rba_panther_enrich'></span>

<h3>Description</h3>

<p>Use PANTHER services to perform over-representation enrichment analysis.
You can either provide a character vector of gene IDs for
over-representation analysis, or a data frame of gene IDs and expression
analysis.<br /> Please refer to the details section for more information on
the statistical analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_enrich(
  genes,
  organism,
  annot_dataset,
  test_type = NULL,
  correction = "FDR",
  cutoff = NULL,
  ref_genes = NULL,
  ref_organism = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_enrich_+3A_genes">genes</code></td>
<td>
<p>Either a character vector or a data frame. Depending on
this parameter, the analysis type is determined.
</p>

<dl>
<dt>Character vector:</dt><dd><p>If a character vector is supplied,
over-representation analysis will be performed using either Fisher's exact
test (default), or binomial.</p>
</dd>
<dt>Data frame:</dt><dd><p>If a data.frame is supplied, statistical enrichment
test is performed using Mann-Whitney U (Wilcoxon Rank-Sum) test. The
data frame should have two columns: the first column is a character vector
with gene identifiers and the second column is a numerical vector with
expression values.</p>
</dd>
</dl>

<p>In both cases, maximum of 10000 genes can be supplied.
The gene identifiers can be any of: Ensemble gene ID, Ensembl protein ID,
Ensembl transcript ID, Entrez gene ID, gene symbol, NCBI GI, HGNC ID,
International protein index ID, NCBI UniGene ID, UniProt accession
or UniProt ID.</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_organism">organism</code></td>
<td>
<p>(numeric) NCBI taxon ID. run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code>
with argument 'what = &quot;organisms&quot;' to get a list of PANTHER's
supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_annot_dataset">annot_dataset</code></td>
<td>
<p>A PANTHER dataset ID to test your input against it.
run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument 'what = &quot;datasets&quot;' to
get a list of PANTHER's supported datasets. Note that you should enter
the &quot;id&quot; of the dataset, not its label (e.g. entering &quot;biological_process&quot;
is incorrect, you should rather enter &quot;GO:0008150&quot;).</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_test_type">test_type</code></td>
<td>
<p>statistical test type to calculate the p values.
</p>

<ul>
<li><p> If performing over-representation analysis (i.e. 'genes' parameter
is a character vector), valid values are &quot;FISHER&quot; (default) or &quot;BINOMIAL&quot;.
</p>
</li>
<li><p> If performing statistical enrichment analysis (i.e. 'genes'
parameter is a data.frame), the only valid value is &quot;Mann-Whitney&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_correction">correction</code></td>
<td>
<p>p value correction method. either &quot;FDR&quot; (default),
&quot;BONFERRONI&quot; or &quot;NONE&quot;.</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_cutoff">cutoff</code></td>
<td>
<p>(Numeric) (Optional) a threshold to filter the results.
if correction is &quot;FDR&quot;, the threshold will be applied to fdr column's
values; if otherwise, the threshold will be applied to p value column.</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_ref_genes">ref_genes</code></td>
<td>
<p>(Optional, only valid if genes is a character vector)
A character vector of genes that will be used as the test's background
(reference/universe) gene set. If no value supplied, all of the genes in
specified organism will be used. The maximum length and supported IDs are
the same as 'genes' argument.</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_ref_organism">ref_organism</code></td>
<td>
<p>(Optional, only valid if genes is a character vector)
if 'ref_genes' is used, you can specify the organisms which correspond to
your supplied IDs in 'ref_genes' argument. see 'organism' argument for
supported values.</p>
</td></tr>
<tr><td><code id="rba_panther_enrich_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Over-representation Test</strong>: It assesses whether specific gene sets are
represented in your input gene list differently from what is expected by
chance. It uses Fisher's exact test or Binomial test to calculate p-values.
Fisher's exact test determines the probability of observing the gene
counts in a category based on a hypergeometric distribution; the binomial
test compares the observed proportion of genes in a category to the
expected proportion based on the reference list. A significant p-value
indicates over-representation or under-representation of a gene set.
</p>
<p><strong>Statistical Enrichment Test</strong>: The statistical enrichment test uses the
Mann-Whitney U (Wilcoxon Rank-Sum) test to assess if the expression values
associated with genes in a specific category differ significantly from the
overall distribution in the input list. This non-parametric test first
ranks the numerical values and computes whether the expression values
were randomly drawn from the overall distribution of values. A small
p-value indicates that the numerical values for the genes in the category
are significantly different from the background distribution, thus
non-random patterns.
</p>
<p>Please note that starting from rbioapi version 0.8.2, you can supply a
gene expression data frame to perform statistical enrichment analysis.
In earlier versions, only a character vector of gene IDs was possible,
thus only over-representation analysis.
</p>


<h3>Value</h3>

<p>A list with the parameters and results. If the analysis was
successful, the results data frame are returned in the &quot;results&quot; element
within the list. Otherwise, an error message will be returned under the
&quot;search$error&quot; element in the returned list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://www.pantherdb.org/services/oai/pantherdb/enrich/overrep&quot;
<br /> &quot;POST https://www.pantherdb.org/services/oai/pantherdb/enrich/statenrich&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_family">rba_panther_family</a>()</code>,
<code><a href="#topic+rba_panther_homolog">rba_panther_homolog</a>()</code>,
<code><a href="#topic+rba_panther_info">rba_panther_info</a>()</code>,
<code><a href="#topic+rba_panther_mapping">rba_panther_mapping</a>()</code>,
<code><a href="#topic+rba_panther_ortholog">rba_panther_ortholog</a>()</code>,
<code><a href="#topic+rba_panther_tree_grafter">rba_panther_tree_grafter</a>()</code>
</p>
<p>Other &quot;Enrichment/Over-representation&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_enrich(
  genes = c("p53", "BRCA1", "cdk2", "Q99835", "CDC42",
    "CDK1", "KIF23", "PLK1", "RAC2", "RACGAP1"),
  organism = 9606, annot_dataset = "GO:0008150",
  cutoff = 0.01
  )



expression_df &lt;- data.frame(
  genes = c("p53", "BRCA1", "cdk2", "Q99835", "CDC42",
    "CDK1", "KIF23", "PLK1", "RAC2", "RACGAP1"),
  expr = runif(10, 0, 100)
  )

rba_panther_enrich(
  genes = expression_df,
  organism = 9606,
  annot_dataset = "GO:0008150"
  )


</code></pre>

<hr>
<h2 id='rba_panther_family'>Get PANTHER Families and Sub-Families</h2><span id='topic+rba_panther_family'></span>

<h3>Description</h3>

<p>Using this function, you can retrieve Orthologs, MSA or Tree topology
information of a given PANTHER family.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_family(id, what, target_organisms = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_family_+3A_id">id</code></td>
<td>
<p>Panther family id.</p>
</td></tr>
<tr><td><code id="rba_panther_family_+3A_what">what</code></td>
<td>
<p>What to retrieve? One of: </p>

<ul>
<li><p> &quot;ortholog&quot;: Orthologs ('LDO' for least diverged and 'O' for more
diverged).
</p>
</li>
<li><p> &quot;msa&quot;: Multiple Sequence Alignment Information,
</p>
</li>
<li><p> &quot;tree&quot;: Tree topology and nodes attributes.
</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_panther_family_+3A_target_organisms">target_organisms</code></td>
<td>
<p>(numeric) NCBI taxon ID(s) to filter the results.
run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument 'what = &quot;organisms&quot;' to
get a list of PANTHER's supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_family_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For trees a list and otherwise a data frame with the requested
family's information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.pantherdb.org/services/oai/pantherdb/familyortholog&quot;
<br /> &quot;GET https://www.pantherdb.org/services/oai/pantherdb/familymsa&quot;
<br /> &quot;GET https://www.pantherdb.org/services/oai/pantherdb/treeinfo&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_panther_homolog">rba_panther_homolog</a>()</code>,
<code><a href="#topic+rba_panther_info">rba_panther_info</a>()</code>,
<code><a href="#topic+rba_panther_mapping">rba_panther_mapping</a>()</code>,
<code><a href="#topic+rba_panther_ortholog">rba_panther_ortholog</a>()</code>,
<code><a href="#topic+rba_panther_tree_grafter">rba_panther_tree_grafter</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_family("PTHR10000", what = "ortholog")


</code></pre>

<hr>
<h2 id='rba_panther_homolog'>Search PANTHER for Homologs of Gene(s)</h2><span id='topic+rba_panther_homolog'></span>

<h3>Description</h3>

<p>Using this function you can search and retrieve homolog of given gene(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_homolog(genes, organism, type = "P", target_organisms = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_homolog_+3A_genes">genes</code></td>
<td>
<p>Character vector of genes identifiers with maximum length of
10 or only one if seq_pos is supplied. Can be any of: Ensemble gene ID,
Ensemble protein ID, Ensemble transcript ID, Entrez gene ID, gene symbol,
NCBI GI, HGNC ID, International protein index ID, NCBI UniGene ID,
UniProt accession and/or UniProt ID.</p>
</td></tr>
<tr><td><code id="rba_panther_homolog_+3A_organism">organism</code></td>
<td>
<p>(numeric) NCBI taxon ID of the organism of your supplied
genes. run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument
'what = &quot;organisms&quot;' to get a list of PANTHER's supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_homolog_+3A_type">type</code></td>
<td>
<p>Homolog types to return. either &quot;P&quot; (default) for paralogs,
&quot;X&quot; for horizontal gene transfer and &quot;LDX&quot; for diverged horizontal gene
transfer.</p>
</td></tr>
<tr><td><code id="rba_panther_homolog_+3A_target_organisms">target_organisms</code></td>
<td>
<p>(numeric) NCBI taxon ID(s) to filter the results.
run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument 'what = &quot;organisms&quot;' to
get a list of PANTHER's supported organisms.
For Paralog, target organism and organism should be the same; Otherwise,
the target organism should be different from the input organism.</p>
</td></tr>
<tr><td><code id="rba_panther_homolog_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with homologs information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.pantherdb.org/services/oai/pantherdb/ortholog/homologOther&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_panther_family">rba_panther_family</a>()</code>,
<code><a href="#topic+rba_panther_info">rba_panther_info</a>()</code>,
<code><a href="#topic+rba_panther_mapping">rba_panther_mapping</a>()</code>,
<code><a href="#topic+rba_panther_ortholog">rba_panther_ortholog</a>()</code>,
<code><a href="#topic+rba_panther_tree_grafter">rba_panther_tree_grafter</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_homolog("OR4F5", organism = 9606, type = "P")


</code></pre>

<hr>
<h2 id='rba_panther_info'>Get PANTHER database Information</h2><span id='topic+rba_panther_info'></span>

<h3>Description</h3>

<p>Using this function you can retrieve a list of available organisms,
annotation datasets, families, and pathways which are supported in
PANTHER.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_info(what, organism_chr_loc = FALSE, families_page = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_info_+3A_what">what</code></td>
<td>
<p>what information to retrieve? should be one of: </p>

<ul>
<li><p> &quot;organisms&quot;: Retrieve supported organisms in PANTHER.
</p>
</li>
<li><p> &quot;datasets&quot;: Retrieve available annotation datasets.
</p>
</li>
<li><p> &quot;families&quot; Retrieve available family IDs.
</p>
</li>
<li><p> &quot;species_tree&quot; Retrieve the PANThER's species tree.
</p>
</li>
<li><p> &quot;pathways&quot; Retrieve available pathway IDs.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_panther_info_+3A_organism_chr_loc">organism_chr_loc</code></td>
<td>
<p>(Logical) (only when 'what = &quot;organisms&quot;')
If TRUE, only organisms with chromosome location will be returned.
If FALSE (default) every organisms will be returned.</p>
</td></tr>
<tr><td><code id="rba_panther_info_+3A_families_page">families_page</code></td>
<td>
<p>(Numeric) (only when 'what = &quot;families&quot;')
Family information is very long, so results are paginated. Use this
argument to define the page to retrieve.</p>
</td></tr>
<tr><td><code id="rba_panther_info_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For families and species tree, a list and otherwise a data frame
with pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.pantherdb.org/services/oai/pantherdb/supportedgenomes&quot;
<br /> &quot;GET https://www.pantherdb.org/services/oai/pantherdb/supportedannotdatasets&quot;
<br /> &quot;GET https://www.pantherdb.org/services/oai/pantherdb/supportedpantherfamilies&quot;
<br /> &quot;GET https://www.pantherdb.org/services/oai/pantherdb/supportedpantherpathways&quot;
<br /> &quot;GET https://www.pantherdb.org/services/oai/pantherdb/speciestree&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_panther_family">rba_panther_family</a>()</code>,
<code><a href="#topic+rba_panther_homolog">rba_panther_homolog</a>()</code>,
<code><a href="#topic+rba_panther_mapping">rba_panther_mapping</a>()</code>,
<code><a href="#topic+rba_panther_ortholog">rba_panther_ortholog</a>()</code>,
<code><a href="#topic+rba_panther_tree_grafter">rba_panther_tree_grafter</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_info(what = "organisms")


rba_panther_info(what = "families", families_page = 4)


</code></pre>

<hr>
<h2 id='rba_panther_mapping'>Map A Gene-set to PANTHER Database</h2><span id='topic+rba_panther_mapping'></span>

<h3>Description</h3>

<p>Using this function, you can search your genes in PANTHER database and
retrieve attributes and annotations associated to your genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_mapping(genes, organism, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_mapping_+3A_genes">genes</code></td>
<td>
<p>Character vector of genes identifiers with maximum length of
1000. Can be any of: Ensemble gene ID, Ensemble protein ID, Ensemble
transcript ID, Entrez gene ID, gene symbol, NCBI GI, HGNC ID,
International protein index ID, NCBI UniGene ID, UniProt accession
and/or UniProt ID.</p>
</td></tr>
<tr><td><code id="rba_panther_mapping_+3A_organism">organism</code></td>
<td>
<p>(numeric) NCBI taxon ID. run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code>
with argument 'what = &quot;organisms&quot;' to get a list of PANTHER's
supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_mapping_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing your unmapped inputs and mapped genes with
pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.pantherdb.org/services/oai/pantherdb/geneinfo&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_panther_family">rba_panther_family</a>()</code>,
<code><a href="#topic+rba_panther_homolog">rba_panther_homolog</a>()</code>,
<code><a href="#topic+rba_panther_info">rba_panther_info</a>()</code>,
<code><a href="#topic+rba_panther_ortholog">rba_panther_ortholog</a>()</code>,
<code><a href="#topic+rba_panther_tree_grafter">rba_panther_tree_grafter</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_mapping(genes = c("Cd40", 7124, "ENSG00000203747", "P33681"),
    organism = 9606)


</code></pre>

<hr>
<h2 id='rba_panther_ortholog'>Search PANTHER for Orthologs of Gene(s)</h2><span id='topic+rba_panther_ortholog'></span>

<h3>Description</h3>

<p>Using this function you can search and retrieve orthologs of given gene(s),
and optionally return the corresponding position in the target organisms'
protein sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_ortholog(
  genes,
  organism,
  type = "all",
  target_organisms = NULL,
  seq_pos = NULL,
  include_msa = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_ortholog_+3A_genes">genes</code></td>
<td>
<p>Character vector of genes identifiers with maximum length of
10 or only one if seq_pos is supplied. Can be any of: Ensemble gene ID,
Ensemble protein ID, Ensemble transcript ID, Entrez gene ID, gene symbol,
NCBI GI, HGNC ID, International protein index ID, NCBI UniGene ID,
UniProt accession and/or UniProt ID.</p>
</td></tr>
<tr><td><code id="rba_panther_ortholog_+3A_organism">organism</code></td>
<td>
<p>(numeric) NCBI taxon ID of the organism of your supplied
genes. run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument
'what = &quot;organisms&quot;' to get a list of PANTHER's supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_ortholog_+3A_type">type</code></td>
<td>
<p>Ortholog types to return. either &quot;all&quot; (default) or &quot;LDO&quot; to
only return least diverged orthologs.</p>
</td></tr>
<tr><td><code id="rba_panther_ortholog_+3A_target_organisms">target_organisms</code></td>
<td>
<p>(numeric) NCBI taxon ID(s) to filter the results.
run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument 'what = &quot;organisms&quot;' to
get a list of PANTHER's supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_ortholog_+3A_seq_pos">seq_pos</code></td>
<td>
<p>(Numeric) A position in the protein's sequence of the
supplied gene. should be in the range of the protein's length.</p>
</td></tr>
<tr><td><code id="rba_panther_ortholog_+3A_include_msa">include_msa</code></td>
<td>
<p>(Logical) Only if a sequence position is supplied,
should MSA (Multiple Sequence Alignment) information be included in the
results?</p>
</td></tr>
<tr><td><code id="rba_panther_ortholog_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with Orthologs information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://www.pantherdb.org/services/oai/pantherdb/ortholog/matchortho&quot;
<br /> &quot;POST https://www.pantherdb.org/services/oai/pantherdb/ortholog/homologpos&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_panther_family">rba_panther_family</a>()</code>,
<code><a href="#topic+rba_panther_homolog">rba_panther_homolog</a>()</code>,
<code><a href="#topic+rba_panther_info">rba_panther_info</a>()</code>,
<code><a href="#topic+rba_panther_mapping">rba_panther_mapping</a>()</code>,
<code><a href="#topic+rba_panther_tree_grafter">rba_panther_tree_grafter</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_ortholog("CD40", organism = 9606, type = "LDO")


</code></pre>

<hr>
<h2 id='rba_panther_tree_grafter'>PANTHER Tree Grafter</h2><span id='topic+rba_panther_tree_grafter'></span>

<h3>Description</h3>

<p>Use this function to retrieve a PANTHER family's tree topology information
with a node corresponding to your sequence grafted in the best location
in that tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_panther_tree_grafter(protein_seq, target_organisms = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_panther_tree_grafter_+3A_protein_seq">protein_seq</code></td>
<td>
<p>A character string with the protein's sequence. Maximum
allowed sequence length is 50kb.</p>
</td></tr>
<tr><td><code id="rba_panther_tree_grafter_+3A_target_organisms">target_organisms</code></td>
<td>
<p>(numeric) NCBI taxon ID(s) to filter the results.
run <code><a href="#topic+rba_panther_info">rba_panther_info</a></code> with argument 'what = &quot;organisms&quot;' to
get a list of PANTHER's supported organisms.</p>
</td></tr>
<tr><td><code id="rba_panther_tree_grafter_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information, see:
Haiming Tang, Robert D Finn, Paul D Thomas, TreeGrafter: phylogenetic
tree-based annotation of proteins with Gene Ontology terms and other
annotations, Bioinformatics, Volume 35, Issue 3, February 2019, Pages
518–520, <a href="https://doi.org/10.1093/bioinformatics/bty625">doi:10.1093/bioinformatics/bty625</a>
</p>


<h3>Value</h3>

<p>A list containing PANTHER tree topology information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.pantherdb.org/services/oai/pantherdb/graftsequence&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Huaiyu Mi, Dustin Ebert, Anushya Muruganujan, Caitlin Mills,
Laurent-Philippe Albou, Tremayne Mushayamaha, Paul D Thomas, PANTHER
version 16: a revised family classification, tree-based classification
tool, enhancer regions and extensive API, Nucleic Acids Research,
Volume 49, Issue D1, 8 January 2021, Pages D394–D403,
https://doi.org/10.1093/nar/gkaa1106
</p>
</li>
<li> <p><a href="https://www.pantherdb.org/services/details.jsp">PANTHER
Services Details</a>
</p>
</li>
<li>
<p><a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">Citations
note on PANTHER website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;PANTHER&quot;: 
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_panther_family">rba_panther_family</a>()</code>,
<code><a href="#topic+rba_panther_homolog">rba_panther_homolog</a>()</code>,
<code><a href="#topic+rba_panther_info">rba_panther_info</a>()</code>,
<code><a href="#topic+rba_panther_mapping">rba_panther_mapping</a>()</code>,
<code><a href="#topic+rba_panther_ortholog">rba_panther_ortholog</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_panther_tree_grafter("MKVLWAALLVTFLAGCQAKVEQAVETE")


</code></pre>

<hr>
<h2 id='rba_reactome_analysis'>Reactome Over-Representation or Expression Analysis</h2><span id='topic+rba_reactome_analysis'></span>

<h3>Description</h3>

<p>Using this function, you can perform Reactome Analysis In a convenient way.
The Analysis Type will be chosen depending on your supplied
input:</p>

<ol>
<li><p> If you supply a vector or a single-columned table,
&quot;Over-Representation&quot; analysis will be performed.
</p>
</li>
<li><p> If you supply a multi-column table, with the first column being
molecules identifiers and the rest being numeral expression values,
&quot;Expression&quot; analysis will be performed.</p>
</li></ol>

<p>See the details section for the accepted input types and format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis(
  input,
  input_format = NULL,
  projection = TRUE,
  interactors = FALSE,
  species = NULL,
  sort_by = "ENTITIES_PVALUE",
  order = "ASC",
  resource = "TOTAL",
  p_value = 1,
  include_disease = TRUE,
  min = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_+3A_input">input</code></td>
<td>
<p>A vector, data frame, matrix or a local file path or URL
that points to your data. See &quot;Details section&quot; for more information of
how to organize and supply your input.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_input_format">input_format</code></td>
<td>
<p>(Optional) This function will automatically identify
your supplied input's format. But in case of unexpected issues or if you
want to be explicit, set this argument to one of:</p>

<ul>
<li><p> &quot;table&quot;: If you supplied a data frame or matrix as input.
</p>
</li>
<li><p> &quot;vector&quot;: If you supplied a simple vector (numeric or character) as
input.
</p>
</li>
<li><p> &quot;file&quot;: If you supplied a local file path pointing to a
correctly-formatted text file.
</p>
</li>
<li><p> &quot;url&quot;: If you supplied a URL pointing to a correctly-formatted
text file.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_projection">projection</code></td>
<td>
<p>Logical (default = TRUE) Should non-human identifiers
be projected to their human equivalents? (using Reactome orthology data)</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_interactors">interactors</code></td>
<td>
<p>Logical (default = FALSE) Should IntAct interaction data
be used to increase the analysis background?</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human
is 9606), species name (e.g. &quot;Homo sapiens&quot;) or Reactome DbId (e.g
Homo sapiens is 48887). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>. Note that you cannot supply the species
parameter when projection parameter is TRUE.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_sort_by">sort_by</code></td>
<td>
<p>Sort the result based on what column? available choices
are: &quot;NAME&quot;, &quot;TOTAL_ENTITIES&quot;, &quot;TOTAL_INTERACTORS&quot;, &quot;TOTAL_REACTIONS&quot;,
&quot;FOUND_ENTITIES&quot;, &quot;FOUND_INTERACTORS&quot;, &quot;FOUND_REACTIONS&quot;, &quot;ENTITIES_RATIO&quot;,
&quot;ENTITIES_PVALUE&quot;, &quot;ENTITIES_FDR&quot; or &quot;REACTIONS_RATIO&quot;</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_order">order</code></td>
<td>
<p>Sort Order. Can be either &quot;ASC&quot; (default) or &quot;DESC&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_resource">resource</code></td>
<td>
<p>Filter results based on the resource. Default is &quot;TOTAL&quot;,
available choices are: &quot;TOTAL&quot;, &quot;UNIPROT&quot;, &quot;ENSEMBL&quot;, &quot;CHEBI&quot;, &quot;IUPHAR&quot;,
&quot;MIRBASE&quot;, &quot;NCBI_PROTEIN&quot;, &quot;EMBL&quot;, &quot;COMPOUND&quot;, &quot;ENTITIES_FDR&quot; or
&quot;PUBCHEM_COMPOUND&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_p_value">p_value</code></td>
<td>
<p>Set a P value threshold. Only results with P value equal to
or less than your supplied threshold will be returned. (default = 1,
Meaning no P value filtering)</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_include_disease">include_disease</code></td>
<td>
<p>Logical (default = TRUE) Should the disease pathways
be included in the results?</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_min">min</code></td>
<td>
<p>(numeric) Minimum number of entities that a pathways should have
to be included in the results.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_max">max</code></td>
<td>
<p>(numeric) Maximum number of entities that a pathways should have
to be included in the results.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can supply your table or vector input in numerous formats:</p>

<ol>
<li><p> A R object which can be data frame, matrix or a simple vector.
</p>
</li>
<li><p> A path to a local text file in your device that contains the molecules
data. (The file should be formatted correctly, see below.)
</p>
</li>
<li><p> A URL pointing to a text file on the web that contains the molecules
data. (The file should be formatted correctly, see below.</p>
</li></ol>

<p>If you supply a text file (as a local file path or URL), it should be
in TSV (Tab-Separated Values) format; Column names should start with &quot;#&quot;
character. Note that if you are providing the file for
&quot;Over-Representation&quot; analysis (i.e. Single columned-data) this header
line is optional and will be used as your 'Sample Name', otherwise it is
required.
<br /> Also, form the &quot;summary&quot; element in the function's output, you can see
how Reactome Interpreted your input and subsequently the type of analysis
that has been performed.
<br /> There is no strict criteria about the type of your molecules Identifiers,
Reactome will Map the IDs to it's internal database entities.
Nevertheless, You can check if all your identifiers has been found in
&quot;identifiersNotFound&quot; element in the function's output.
<br /> After Any Analysis, Reactome will associate a token to your analysis.
It can be later used to in function that requires the token (e.g to retrieve
the analysis results, download pdf).
<br /> Note that Reactome will store your token for only 7 days. You can
download your full results with
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a></code>, and re-import it anytime to
reactome (using <code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a></code>) to generate
a new token.
</p>


<h3>Value</h3>

<p>List containing the results and information of your analysis. Note
that you can use the token returned in the &quot;summary&quot; sub-list of the
results (i.e. results$summary$token) to retrieve your results later or
in other Reactome analysis functions.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://reactome.org/AnalysisService/identifiers/form&quot;
<br /> &quot;POST https://reactome.org/AnalysisService/identifiers/url&quot;
<br /> &quot;POST https://reactome.org/AnalysisService/identifiers/form/projection&quot;
<br /> &quot;POST https://reactome.org/AnalysisService/identifiers/url/projection&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_mapping">rba_reactome_analysis_mapping</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_token">rba_reactome_analysis_token</a>()</code>
</p>
<p>Other &quot;Enrichment/Over-representation&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_analysis(input = c("p53", "BRCA1", "cdk2", "Q99835", "CDC42"))

## End(Not run)
## Not run: 
rba_reactome_analysis(input = "c:/rbioapi/genes.txt")

## End(Not run)
## Not run: 
rba_reactome_analysis(input = "https://qazwsx.com/genes.txt")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_analysis_download'>Download Different Reactome Analysis Results</h2><span id='topic+rba_reactome_analysis_download'></span>

<h3>Description</h3>

<p>Based on the &quot;request&quot; argument, you can download different analysis
results data associated with a given token.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis_download(
  token,
  request,
  save_to = NULL,
  resource = "TOTAL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_download_+3A_token">token</code></td>
<td>
<p>A token associated to your previous Reactome analysis.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_download_+3A_request">request</code></td>
<td>
<p>What to download? Should be one of:</p>

<ul>
<li><p> &quot;found_ids&quot;: Download a CSV file containing the found user-supplied
identifiers in the analysis associated with your supplied token and
resource.
</p>
</li>
<li><p> &quot;not_found_ids&quot;&quot; Download a CSV file containing the user-supplied
Identifiers which has not been found in the analysis associated with your
supplied token.
</p>
</li>
<li><p> &quot;pathways&quot;: Download a CSV file containing Pathway analysis results
of the analysis associated with your supplied token and resource.
</p>
</li>
<li><p> &quot;results&quot;: Download a JSON file containing the complete analysis
results associated with your supplied token.
</p>
</li>
<li><p> &quot;results_gz&quot; Same as &quot;results&quot;, but the output will be compress
(gzipped).</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_analysis_download_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: Save the file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_analysis_download_+3A_resource">resource</code></td>
<td>
<p>(Only when request is &quot;found_ids&quot; or &quot;pathways&quot;)
Filter results based on the resource. Default is &quot;TOTAL&quot;,
available choices are:&quot;TOTAL&quot;, &quot;UNIPROT&quot;, &quot;ENSEMBL&quot;, &quot;CHEBI&quot;, &quot;IUPHAR&quot;,
&quot;MIRBASE&quot;, &quot;NCBI_PROTEIN&quot;, &quot;EMBL&quot;, &quot;COMPOUND&quot;, &quot;ENTITIES_FDR&quot; or
&quot;PUBCHEM_COMPOUND&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_download_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Token is associated to each Reactome analysis results and kept by Reactome
for at least 7 days. You can locate it in
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>'s output, under a sub-list named
&quot;summary&quot; (i.e. results$summary$token).
<br /> Use <code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a></code> to save a full report
in PDF format.
</p>


<h3>Value</h3>

<p>NULL, a CSV,JSON or Gzipped JSON file will be saved to disk
based on your input.
</p>


<h3>Corresponding API Resources</h3>

<p>GET https://reactome.org/AnalysisService/download/{token}/entities/
found/{resource}/{filename}.csv&quot;
<br /> GET https://reactome.org/AnalysisService/download/{token}/entities/
notfound/{filename}.csv&quot;
<br /> GET https://reactome.org/AnalysisService/download/{token}/pathways/
{resource}/{filename}.csv&quot;
<br /> GET https://reactome.org/AnalysisService/download/{token}/result.json&quot;
<br /> GET https://reactome.org/AnalysisService/download/{token}/result.json.gz&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a></code>
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>
</p>
<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_mapping">rba_reactome_analysis_mapping</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_token">rba_reactome_analysis_token</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_analysis_download(token = "MjAyMDEwMTYwMTI3MTNfMjY1MjM",
    request = "pathways", save_to = "found_ids.csv")

## End(Not run)
## Not run: 
rba_reactome_analysis_download(token = "MjAyMDEwMTYwMTI3MTNfMjY1MjM",
    request = "found_ids", save_to = "found_ids.csv")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_analysis_import'>Import Saved Analysis JSON to Reactome</h2><span id='topic+rba_reactome_analysis_import'></span>

<h3>Description</h3>

<p>If you have a JSON file of analysis results (only obtained via
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a></code> with the result argument
set to &quot;results&quot;, or &quot;results_gz&quot;), you can import the results back to
Reactome and retrieve a token.
<br /> This is useful when you want to use other Reactome services which require
a token but you do not have a token or your token has been
expired (i.e. more than 7 days passed from your analysis).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis_import(input, input_format = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_import_+3A_input">input</code></td>
<td>
<p>A local file path or URL that points to your -optionally
gzipped- JSON file.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_import_+3A_input_format">input_format</code></td>
<td>
<p>(Optional) This function will automatically identify
your supplied input's format. But in case of unexpected issues or if you
want to be explicit, set this argument to one of:</p>

<ul>
<li><p> &quot;file&quot;: If you supplied a local file path pointing to the JSON file.
</p>
</li>
<li><p> &quot;url&quot;: If you supplied a URL pointing to the JSON file.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_analysis_import_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the new token and other information of your
imported results.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/AnalysisService/import/&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/import/form&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/import/url&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_mapping">rba_reactome_analysis_mapping</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_token">rba_reactome_analysis_token</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_analysis_import("c:/rbioapi/res.json")

## End(Not run)
## Not run: 
rba_reactome_analysis_import("https://qaz.com/res.json.gz")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_analysis_mapping'>Maps Molecule Identifiers</h2><span id='topic+rba_reactome_analysis_mapping'></span>

<h3>Description</h3>

<p>Use this function to map molecule identifiers of different species to
Reactome Identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis_mapping(
  input,
  input_format = NULL,
  projection = TRUE,
  interactors = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_mapping_+3A_input">input</code></td>
<td>
<p>A vector, local file path or URL that points to your
identifiers list.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_mapping_+3A_input_format">input_format</code></td>
<td>
<p>(Optional) This function will automatically identify
your supplied input's format. But in case of unexpected issues or if you
want to be explicit, set this argument to one of:</p>

<ul>
<li><p> &quot;vector&quot;: If you supplied a simple vector (numeric or character) as
input.
</p>
</li>
<li><p> &quot;file&quot;: If you supplied a local file path pointing to a
correctly-formatted text file.
</p>
</li>
<li><p> &quot;url&quot;: If you supplied a URL pointing to a correctly-formatted
text file.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_analysis_mapping_+3A_projection">projection</code></td>
<td>
<p>Logical (default = TRUE) Should non-human identifiers
be projected to their human equivalents? (using Reactome orthology data)</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_mapping_+3A_interactors">interactors</code></td>
<td>
<p>Logical (default = FALSE) Should IntAct interaction data
be included?</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_mapping_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing your identifiers and the IDS and resources they
are mapped to.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/AnalysisService/mapping&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/mapping/form&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/mapping/form/projection&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/mapping&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/mapping/url&quot;
<br /> &quot;GET https://reactome.org/AnalysisService/mapping/url/projection&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_token">rba_reactome_analysis_token</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_analysis_mapping(c("Q8SQ34", "cd40"))

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_analysis_pdf'>Generate PDF file with Reactome Analysis Results</h2><span id='topic+rba_reactome_analysis_pdf'></span>

<h3>Description</h3>

<p>Use this function to save a detailed report of your previous analysis (That
you have done with <code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>). You need
to supply a 'token' associated to your previous analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis_pdf(
  token,
  species,
  save_to = NULL,
  number = 25,
  resource = "TOTAL",
  diagram_profile = "Modern",
  analysis_profile = "Standard",
  fireworks_profile = "Barium Lithium",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_pdf_+3A_token">token</code></td>
<td>
<p>A token associated to your previous Reactome analysis.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human Taxonomy
ID is 9606.) or species name (e.g. &quot;Homo sapiens&quot;). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: Save the file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_number">number</code></td>
<td>
<p>Numeric: Maximum number of the reported pathways. Cannot not
be greater than 50.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_resource">resource</code></td>
<td>
<p>Filter results based on the resource. Default is &quot;TOTAL&quot;,
available choices are: &quot;TOTAL&quot;, &quot;UNIPROT&quot;, &quot;ENSEMBL&quot;, &quot;CHEBI&quot;, &quot;IUPHAR&quot;,
&quot;MIRBASE&quot;, &quot;NCBI_PROTEIN&quot;, &quot;EMBL&quot;, &quot;COMPOUND&quot;, &quot;ENTITIES_FDR&quot; or
&quot;PUBCHEM_COMPOUND&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_diagram_profile">diagram_profile</code></td>
<td>
<p>Color profile of diagrams, should be either
&quot;Modern&quot; (default) or &quot;Standard&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_analysis_profile">analysis_profile</code></td>
<td>
<p>Color profile of analysis, should be one of:
&quot;Standard&quot; (default), &quot;Strosobar&quot; or &quot;Copper Plus&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_fireworks_profile">fireworks_profile</code></td>
<td>
<p>Color profile of overview diagram, should be one of:
&quot;Copper&quot;, &quot;Copper Plus&quot;, &quot;Barium Lithium&quot; or &quot;calcium salts&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_pdf_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Token is associated to each Reactome analysis results and kept by Reactome
for at least 7 days. You can locate it in
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>'s output, under a sub-list named
&quot;summary&quot; (i.e. results$summary$token).
<br /> Note that Reactome will store your token for only 7 days. You can
download your full results with
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a></code>, and re-import it anytime to
reactome (using <code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a></code>) to generate
a new token.
Use <code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a></code> to save your results
in other formats.
</p>


<h3>Value</h3>

<p>NULL, a PDF file will be saved to disk.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/AnalysisService/report/{token}/{species}/
{filename}.pdf&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a></code>
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>
</p>
<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_mapping">rba_reactome_analysis_mapping</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_token">rba_reactome_analysis_token</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_analysis_pdf(token = "MjAyMDEwMTYwMTI3MTNfMjY1MjM%3D",
    species = 9606, save_to = "my_analysis.pdf")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_analysis_species'>Compare Human Pathways with with Other Species</h2><span id='topic+rba_reactome_analysis_species'></span>

<h3>Description</h3>

<p>Use This function to Compare human's manually-curated pathways
and computationally inferred pathways (orthologous) in other species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis_species(
  species_dbid,
  sort_by = "ENTITIES_PVALUE",
  order = "ASC",
  resource = "TOTAL",
  p_value = 1,
  min = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_species_+3A_species_dbid">species_dbid</code></td>
<td>
<p>Numeric: Reactome DbId (e.g  Mus musculus is 48892) of
the species you want to compare with Homo sapiens. See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_sort_by">sort_by</code></td>
<td>
<p>Sort the result based on what column? available choices
are: &quot;NAME&quot;, &quot;TOTAL_ENTITIES&quot;, &quot;TOTAL_INTERACTORS&quot;, &quot;TOTAL_REACTIONS&quot;,
&quot;FOUND_ENTITIES&quot;, &quot;FOUND_INTERACTORS&quot;, &quot;FOUND_REACTIONS&quot;, &quot;ENTITIES_RATIO&quot;,
&quot;ENTITIES_PVALUE&quot;, &quot;ENTITIES_FDR&quot; or &quot;REACTIONS_RATIO&quot;</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_order">order</code></td>
<td>
<p>Sort Order. Can be either &quot;ASC&quot; (default) or &quot;DESC&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_resource">resource</code></td>
<td>
<p>Filter results based on the resource. Default is &quot;TOTAL&quot;,
available choices are: &quot;TOTAL&quot;, &quot;UNIPROT&quot;, &quot;ENSEMBL&quot;, &quot;CHEBI&quot;, &quot;IUPHAR&quot;,
&quot;MIRBASE&quot;, &quot;NCBI_PROTEIN&quot;, &quot;EMBL&quot;, &quot;COMPOUND&quot;, &quot;ENTITIES_FDR&quot; or
&quot;PUBCHEM_COMPOUND&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_p_value">p_value</code></td>
<td>
<p>Set a P value threshold. Only results with P value equal to
or less than your supplied threshold will be returned. (default = 1,
Meaning no P value filtering)</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_min">min</code></td>
<td>
<p>(numeric) Minimum number of entities that a pathways should have
to be included in the results.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_max">max</code></td>
<td>
<p>(numeric) Maximum number of entities that a pathways should have
to be included in the results.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_species_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reactome incorporate manually curated human reactions and PANTHER's
protein homology data to Computationally infer events in other eukaryotic
species.
<br /> In version 73 (11 June 2020), using an orthology-based approach,
Homo sapiens events was projected to 18,654 orthologous pathways (with
81,835 orthologous proteins) in 15 non-human species.
See <a href="https://reactome.org/documentation/inferred-events/">
Reactome Computationally Inferred Events</a> for more information.
</p>


<h3>Value</h3>

<p>List with the results of the comparison.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/AnalysisService/species/homoSapiens/{species}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_orthology">rba_reactome_orthology</a></code>
</p>
<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_mapping">rba_reactome_analysis_mapping</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_token">rba_reactome_analysis_token</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_analysis_species(species_dbid = 48892)


</code></pre>

<hr>
<h2 id='rba_reactome_analysis_token'>Return the Results Associated with a Token</h2><span id='topic+rba_reactome_analysis_token'></span>

<h3>Description</h3>

<p>Use a token generated After a Reactome analysis
(via <code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>) to Retrieve the analysis results.
The output format is identical to the returned object of
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_analysis_token(
  token,
  species,
  sort_by = "ENTITIES_PVALUE",
  order = "ASC",
  resource = "TOTAL",
  p_value = NULL,
  include_disease = TRUE,
  min = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_analysis_token_+3A_token">token</code></td>
<td>
<p>A token associated to your previous Reactome analysis.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human
is 9606), species name (e.g. &quot;Homo sapiens&quot;) or Reactome DbId (e.g
Homo sapiens is 48887). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_sort_by">sort_by</code></td>
<td>
<p>Sort the result based on what column? available choices
are: &quot;NAME&quot;, &quot;TOTAL_ENTITIES&quot;, &quot;TOTAL_INTERACTORS&quot;, &quot;TOTAL_REACTIONS&quot;,
&quot;FOUND_ENTITIES&quot;, &quot;FOUND_INTERACTORS&quot;, &quot;FOUND_REACTIONS&quot;, &quot;ENTITIES_RATIO&quot;,
&quot;ENTITIES_PVALUE&quot;, &quot;ENTITIES_FDR&quot; or &quot;REACTIONS_RATIO&quot;</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_order">order</code></td>
<td>
<p>Sort Order. Can be either &quot;ASC&quot; (default) or &quot;DESC&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_resource">resource</code></td>
<td>
<p>Filter results based on the resource. Default is &quot;TOTAL&quot;,
available choices are: &quot;TOTAL&quot;, &quot;UNIPROT&quot;, &quot;ENSEMBL&quot;, &quot;CHEBI&quot;, &quot;IUPHAR&quot;,
&quot;MIRBASE&quot;, &quot;NCBI_PROTEIN&quot;, &quot;EMBL&quot;, &quot;COMPOUND&quot;, &quot;ENTITIES_FDR&quot; or
&quot;PUBCHEM_COMPOUND&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_p_value">p_value</code></td>
<td>
<p>Set a P value threshold. Only results with P value equal to
or less than your supplied threshold will be returned. (default = 1,
Meaning no P value filtering)</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_include_disease">include_disease</code></td>
<td>
<p>Logical (default = TRUE) Should the disease pathways
be included in the results?</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_min">min</code></td>
<td>
<p>(numeric) Minimum number of entities that a pathways should have
to be included in the results.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_max">max</code></td>
<td>
<p>(numeric) Maximum number of entities that a pathways should have
to be included in the results.</p>
</td></tr>
<tr><td><code id="rba_reactome_analysis_token_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After Any Analysis, Reactome will associate a token to your analysis. It
can be later used to in function that requires the token (e.g to retrieve
the analysis results, download pdf).
<br /> Note that Reactome will store your token for only 7 days. You can
download your full results with
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a></code>, and re-import it anytime to
reactome (using <code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a></code>) to generate
a new token.
</p>


<h3>Value</h3>

<p>List containing the results and information of your analysis.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/AnalysisService/token/{token}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/AnalysisService/">Reactome Analysis
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>
</p>
<p>Other &quot;Reactome Analysis Service&quot;: 
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_download">rba_reactome_analysis_download</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_import">rba_reactome_analysis_import</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_mapping">rba_reactome_analysis_mapping</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_pdf">rba_reactome_analysis_pdf</a>()</code>,
<code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_analysis_token(token = "MjAyMDEwMTYwMTI3MTNfMjY1MjM",
    species = 9606)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_complex_list'>Get Complexes That Include a Molecule</h2><span id='topic+rba_reactome_complex_list'></span>

<h3>Description</h3>

<p>This function will retrieve a list of complexes that include your supplied
molecule as a component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_complex_list(id, resource, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_complex_list_+3A_id">id</code></td>
<td>
<p>Molecule's external Identifier</p>
</td></tr>
<tr><td><code id="rba_reactome_complex_list_+3A_resource">resource</code></td>
<td>
<p>What is the resource of your supplied ID? see:
<a href="https://reactome.org/content/schema/objects/ReferenceDatabase/">Reactome External
Identifiers</a></p>
</td></tr>
<tr><td><code id="rba_reactome_complex_list_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame where each row is a complex containing your supplied
molecule and columns are pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/complexes/{resource}/
{identifier}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Physical Entity Queries&quot;: 
<code><a href="#topic+rba_reactome_complex_subunits">rba_reactome_complex_subunits</a>()</code>,
<code><a href="#topic+rba_reactome_entity_other_forms">rba_reactome_entity_other_forms</a>()</code>,
<code><a href="#topic+rba_reactome_participant_of">rba_reactome_participant_of</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_complex_list(id = "3845", resource = "NCBI Gene")


rba_reactome_complex_list(id = "P00533", resource = "UniProt")


</code></pre>

<hr>
<h2 id='rba_reactome_complex_subunits'>Get a Complex's Subunits</h2><span id='topic+rba_reactome_complex_subunits'></span>

<h3>Description</h3>

<p>This function will return a list of subunits which are participants of
your supplied complex.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_complex_subunits(complex_id, exclude_structures = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_complex_subunits_+3A_complex_id">complex_id</code></td>
<td>
<p>Reactome stable Identifier of the complex.</p>
</td></tr>
<tr><td><code id="rba_reactome_complex_subunits_+3A_exclude_structures">exclude_structures</code></td>
<td>
<p>(logical) Should the contained complexes and
entity sets be excluded from the results? (default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_complex_subunits_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Subunits will be returned recursively; Which means that if a subunit was
itself a complex, subunit of that complex will be also returned in the
results.
</p>


<h3>Value</h3>

<p>Data frame which each row is a subunit of your supplied complex
and the columns are pertinent information of that subunit.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/complex/{id}/subunits&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Physical Entity Queries&quot;: 
<code><a href="#topic+rba_reactome_complex_list">rba_reactome_complex_list</a>()</code>,
<code><a href="#topic+rba_reactome_entity_other_forms">rba_reactome_entity_other_forms</a>()</code>,
<code><a href="#topic+rba_reactome_participant_of">rba_reactome_participant_of</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_complex_subunits(complex_id = "R-HSA-5674003",
    exclude_structures = FALSE)


</code></pre>

<hr>
<h2 id='rba_reactome_diseases'>Reactome Diseases</h2><span id='topic+rba_reactome_diseases'></span>

<h3>Description</h3>

<p>This function Retrieve a list of all diseases or disease DOIDs annotated in
Reactome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_diseases(doid = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_diseases_+3A_doid">doid</code></td>
<td>
<p>(logical) Return disease DOIDs instead of diseases?
(default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_diseases_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing all the disease annotation available at
Reactome. If doid was set to TRUE, DOID info will be returned instead.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/diseases&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/diseases/doid&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_diseases()


rba_reactome_diseases(doid = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_entity_other_forms'>Get Other forms of a Reactome Entity</h2><span id='topic+rba_reactome_entity_other_forms'></span>

<h3>Description</h3>

<p>This function retrieve a list containing all other forms of your supplied
Physical Entity ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_entity_other_forms(entity_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_entity_other_forms_+3A_entity_id">entity_id</code></td>
<td>
<p>Reactome's entity ID.</p>
</td></tr>
<tr><td><code id="rba_reactome_entity_other_forms_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>According to Reactome API documentation, &quot;These other forms are Physical
Entities that share the same Reference Entity identifier, e.g. PTEN
H93R R-HSA-2318524 and PTEN C124R R-HSA-2317439 are two forms of PTEN.&quot;
</p>


<h3>Value</h3>

<p>Data frame where each row is other forms of your supplied Entity ID
and columns are pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/entity/{id}/otherForms&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Physical Entity Queries&quot;: 
<code><a href="#topic+rba_reactome_complex_list">rba_reactome_complex_list</a>()</code>,
<code><a href="#topic+rba_reactome_complex_subunits">rba_reactome_complex_subunits</a>()</code>,
<code><a href="#topic+rba_reactome_participant_of">rba_reactome_participant_of</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_entity_other_forms("R-HSA-199420")


</code></pre>

<hr>
<h2 id='rba_reactome_event_ancestors'>Get Reactome Events Ancestors</h2><span id='topic+rba_reactome_event_ancestors'></span>

<h3>Description</h3>

<p>Along with Reactome's events hierarchy, This function will retrieve all the
events beginning from your supplied event up to the &quot;Top level Pathway&quot;.
see &quot;Details section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_event_ancestors(event_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_event_ancestors_+3A_event_id">event_id</code></td>
<td>
<p>Reactome event's identifier.</p>
</td></tr>
<tr><td><code id="rba_reactome_event_ancestors_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By Reactome's definition, Events are the building blocks of biological
processes and could be of two main classes: &quot;Pathway&quot; or
&quot;Reaction-like events&quot;. The events are organized in a hierarchical
structure; and each event could be child or parent to another event; The
hierarchy will always begin with a &quot;Top level pathway&quot; event. Also note
that a given event could be part of more that one hierarchies.
</p>


<h3>Value</h3>

<p>List which every element is a Data frame listing your supplied
event along with it's ancestor events. Because any given event can be
part of more than one pathway hierarchy, the list may contain multiple
data frames.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/event/{id}/ancestors&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Queries Related to Events&quot;: 
<code><a href="#topic+rba_reactome_event_hierarchy">rba_reactome_event_hierarchy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_event_ancestors("R-HSA-5673001")


</code></pre>

<hr>
<h2 id='rba_reactome_event_hierarchy'>Get Full Event Hierarchy of a Species</h2><span id='topic+rba_reactome_event_hierarchy'></span>

<h3>Description</h3>

<p>This function will retrieve the full Events hierarchy of your supplied
species. Directly under each species, each child element is a &quot;top Level
Pathway&quot;. You can traverse the events tree down by following the &quot;children&quot;
element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_event_hierarchy(species, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_event_hierarchy_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human Taxonomy
ID is 9606.) or species name (e.g. &quot;Homo sapiens&quot;). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_event_hierarchy_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By Reactome's definition, Events are the building blocks of biological
processes and could be of two main classes: &quot;Pathway&quot; or
&quot;Reaction-like events&quot;. The events are organized in a hierarchical
structure; and each event could be child or parent to another event; The
hierarchy will always begin with a &quot;Top level pathway&quot; event. Also note
that a given event could be part of more that one hierarchies.
</p>


<h3>Value</h3>

<p>List which is a representation of the species's events hierarchy
described in the &quot;Details section&quot;.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/eventsHierarchy/{species}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Queries Related to Events&quot;: 
<code><a href="#topic+rba_reactome_event_ancestors">rba_reactome_event_ancestors</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#very large response!
rba_reactome_event_hierarchy("Homo sapiens")

## End(Not run)
## Not run: 
#very large response!
rba_reactome_event_hierarchy(9606)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_exporter_diagram'>Get a Reactome Event Diagram</h2><span id='topic+rba_reactome_exporter_diagram'></span>

<h3>Description</h3>

<p>This function could be called in two scenarios: </p>

<ol>
<li><p> With create_document = FALSE: To retrieve an image of that event's
Diagram.
</p>
</li>
<li><p> With create_document = TRUE: To retrieve a PDF document with the
event's diagram image and additional information.</p>
</li></ol>

<p>see &quot;Details section&quot; for more information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_exporter_diagram(
  event_id,
  save_to = NULL,
  create_document = FALSE,
  resource = "TOTAL",
  diagram_profile = "Modern",
  analysis_profile = "Standard",
  token = NULL,
  exp_column = NULL,
  document_level = 1,
  output_format = "png",
  image_quality = 5,
  flag_element = NULL,
  flg_interactors = TRUE,
  sel = NULL,
  title = TRUE,
  margin = 15,
  ehld = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_exporter_diagram_+3A_event_id">event_id</code></td>
<td>
<p>Reactome event's identifier.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: Save the file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_create_document">create_document</code></td>
<td>
<p>logical: Create PDF document instead of image? (
default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_resource">resource</code></td>
<td>
<p>The analysis resource for which the results will be
overlaid on top of the given pathways overview,</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_diagram_profile">diagram_profile</code></td>
<td>
<p>Color profile of diagrams, should be either &quot;Modern&quot;
(default) or &quot;Standard&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_analysis_profile">analysis_profile</code></td>
<td>
<p>Color profile of analysis, should be one of:
&quot;Standard&quot; (default), &quot;Strosobar&quot; or &quot;Copper Plus&quot;</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_token">token</code></td>
<td>
<p>The analysis Token for which the results will be overlaid on
top of the given pathways overview. see:
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_exp_column">exp_column</code></td>
<td>
<p>numeric: (only if token is supplied) Specify the
expression column for the overlay.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_document_level">document_level</code></td>
<td>
<p>numeric: (Only if &quot;create_document&quot; is TRUE) if 0
(default) the event's children will not be included in the PDF document.
Set this to 1 to include event's children.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_output_format">output_format</code></td>
<td>
<p>(Only if &quot;create_document&quot; is FALSE) Image format
of the saved diagram. Can be one of: png (default), jpeg, svg or gif.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_image_quality">image_quality</code></td>
<td>
<p>Numeric: (Only if &quot;create_document&quot; is FALSE), a number
ranging from 1 to 10. 1 is the lowest quality and 10 is the highest
(default = 5).</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_flag_element">flag_element</code></td>
<td>
<p>(Only if &quot;create_document&quot; is FALSE) gene name, protein
ID, chemical ID or Reactome ID of a diagram's element to be flagged.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_flg_interactors">flg_interactors</code></td>
<td>
<p>Logical: (Only if &quot;create_document&quot; is FALSE) Should
the interactor be considered when flagging a diagram element? (default
= TRUE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_sel">sel</code></td>
<td>
<p>(Only if &quot;create_document&quot; is FALSE) CSV line for highlighting
element(s) selection in the diagram.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_title">title</code></td>
<td>
<p>Logical: (Only if &quot;create_document&quot; is FALSE) Should the pathway
name be displayed below the image? (default = TRUE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_margin">margin</code></td>
<td>
<p>Numeric: (Only if &quot;create_document&quot; is FALSE) A number ranging
from 0 to 20 to set as the image's margin. (default = 15)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_ehld">ehld</code></td>
<td>
<p>logical: (Only if &quot;create_document&quot; is FALSE) Should
&quot;Enhanced High Level Diagram&quot; be considered?</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_diagram_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the function is called with create_document = FALSE:
<br /> The result will be an image with the format supplied in &quot;output_format&quot;
argument. If the supplied event ID refers to a pathway, the image's
content will be the that pathways diagram. If the supplied event ID refers
to a sub-pathway or reaction event, the parent pathway's diagram will be
exported, with that reaction or sub-pathway's events highlighted.
<br /> Note that to export an image of reaction-like event separately, you
should use <code><a href="#topic+rba_reactome_exporter_reaction">rba_reactome_exporter_reaction</a></code>.
<br /> If the function is called with create_document = TRUE:
<br /> A PDF document will contain an image of the event's diagram and the
following information of that events: Summation, Literature references,
Edit history type, location, compartments and diseases.
note that if you call the function with &quot;document level = 1&quot;, information
of your supplied event's children will also be included.
</p>


<h3>Value</h3>

<p>NULL, Based to the inputs, an image or PDF file will be saved to
disk.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/exporter/diagram/{identifier}
.{ext}&quot;
<br /> &quot;GET https://reactome.org/ContentService/exporter/document/event/
{identifier}.pdf&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_exporter_reaction">rba_reactome_exporter_reaction</a></code>
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>
</p>
<p>Other &quot;Reactome Content Service - Format Exporter&quot;: 
<code><a href="#topic+rba_reactome_exporter_event">rba_reactome_exporter_event</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_overview">rba_reactome_exporter_overview</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_reaction">rba_reactome_exporter_reaction</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_exporter_diagram(event_id = "R-HSA-177929",
  create_document = FALSE)

## End(Not run)
## Not run: 
rba_reactome_exporter_diagram(event_id = "R-HSA-6787403",
    create_document = FALSE)

## End(Not run)
## Not run: 
rba_reactome_exporter_diagram(event_id = "R-HSA-177929",
    create_document = TRUE)

## End(Not run)
## Not run: 
rba_reactome_exporter_diagram(event_id = "R-HSA-177929",
    output_format = "svg",
    save_to = "reactome_event_diagram.svg")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_exporter_event'>Exports A Reactome Event to SBGN or SBML</h2><span id='topic+rba_reactome_exporter_event'></span>

<h3>Description</h3>

<p>This function will export a supplied Reactome Event (Pathway or Reaction)
to a SBGN (Systems Biology Graphical Notation) or SBML (Systems Biology
Markup Language)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_exporter_event(event_id, output_format, save_to = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_exporter_event_+3A_event_id">event_id</code></td>
<td>
<p>Reactome event's database IDs (DbId) or Stable IDs (StId).</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_event_+3A_output_format">output_format</code></td>
<td>
<p>Either &quot;sbgn&quot; or &quot;sbml&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_event_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: Save the file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_exporter_event_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL, According to the inputs, a SBGN or SBML file will be saved to
disk.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/exporter/event/
{identifier}.sbgn&quot;
<br /> &quot;GET https://reactome.org/ContentService/exporter/event/
{identifier}.sbml&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Format Exporter&quot;: 
<code><a href="#topic+rba_reactome_exporter_diagram">rba_reactome_exporter_diagram</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_overview">rba_reactome_exporter_overview</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_reaction">rba_reactome_exporter_reaction</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_exporter_event(event_id = "R-HSA-177929",
    output_format = "sbgn",
    save_to = "R-HSA-177929.sbgn")

## End(Not run)
## Not run: 
rba_reactome_exporter_event(event_id = "R-HSA-177929",
    output_format = "sbgn")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_exporter_overview'>Get a Reactome Pathway Overview</h2><span id='topic+rba_reactome_exporter_overview'></span>

<h3>Description</h3>

<p>This function will Save a Pathway Overview of the supplied specie
as an image file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_exporter_overview(
  species,
  output_format = "png",
  save_to = NULL,
  image_quality = 5,
  flag_element = NULL,
  flg_interactors = TRUE,
  sel = NULL,
  title = TRUE,
  margin = 15,
  diagram_profile = "Copper",
  token = NULL,
  resource = "TOTAL",
  exp_column = NULL,
  coverage = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_exporter_overview_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human Taxonomy
ID is 9606.) or species name (e.g. &quot;Homo sapiens&quot;). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_output_format">output_format</code></td>
<td>
<p>Images format, Can be one of: png (default), jpeg,
svg or gif.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: Save the file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_image_quality">image_quality</code></td>
<td>
<p>Numeric: A number  ranging from 1 to 10. 1 is the
lowest quality and 10 is the highest (default = 5).</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_flag_element">flag_element</code></td>
<td>
<p>Gene name, protein ID, chemical ID or Reactome ID of a
diagram's element to be flagged.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_flg_interactors">flg_interactors</code></td>
<td>
<p>Logical: Should the interactor be considered when
flagging a diagram element? (default = TRUE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_sel">sel</code></td>
<td>
<p>CSV line for highlighting element(s) selection in the diagram.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_title">title</code></td>
<td>
<p>Logical: Should the pathway name be displayed below the image?
(default = TRUE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_margin">margin</code></td>
<td>
<p>Numeric: A number ranging from 0 to 20 to set as the image's
margin. (default = 15)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_diagram_profile">diagram_profile</code></td>
<td>
<p>Color profile of diagrams, should be one of &quot;Copper&quot;
(default), &quot;Copper Plus&quot;, &quot;Barium Lithium&quot; or &quot;calcium salts&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_token">token</code></td>
<td>
<p>The analysis Token for which the results will be overlaid on
top of the given pathways overview. see:
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_resource">resource</code></td>
<td>
<p>The analysis resource for which the results will be
overlaid on top of the given pathways overview.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_exp_column">exp_column</code></td>
<td>
<p>numeric: (only if token is supplied) Specify the
expression column for the overlay.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_coverage">coverage</code></td>
<td>
<p>Logical: Should the analysis coverage values be overlaid?
(default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_overview_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL, Based to the inputs, an image file will be saved to disk.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/exporter/fireworks/{species}.{ext}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>
</p>
<p>Other &quot;Reactome Content Service - Format Exporter&quot;: 
<code><a href="#topic+rba_reactome_exporter_diagram">rba_reactome_exporter_diagram</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_event">rba_reactome_exporter_event</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_reaction">rba_reactome_exporter_reaction</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_exporter_overview(species = 9606,
    output_format = "svg",
    save_to = "human_pathways.svg")

## End(Not run)
## Not run: 
rba_reactome_exporter_overview(species = 9606,
    token = 123456789)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_exporter_reaction'>Get a Reactome Reaction Event</h2><span id='topic+rba_reactome_exporter_reaction'></span>

<h3>Description</h3>

<p>This function will Save a Reactome event of class &quot;ReactionLikeEvent&quot; as
an image file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_exporter_reaction(
  event_id,
  save_to = NULL,
  output_format = "png",
  resource = "TOTAL",
  diagram_profile = "Modern",
  analysis_profile = "Standard",
  token = NULL,
  exp_column = NULL,
  image_quality = 5,
  flag_element = NULL,
  flg_interactors = TRUE,
  sel = NULL,
  title = TRUE,
  margin = 15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_exporter_reaction_+3A_event_id">event_id</code></td>
<td>
<p>Reactome
<a href="https://reactome.org/content/schema/ReactionLikeEvent/">Reaction-like
event</a>'s identifier.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_save_to">save_to</code></td>
<td>
<p>NULL or Character:</p>

<ul>
<li><p> NULL: Save the file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the file to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_output_format">output_format</code></td>
<td>
<p>Images format, Can be one of: png (default), jpeg,
svg or gif.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_resource">resource</code></td>
<td>
<p>The analysis resource for which the results will be
overlaid on top of the given pathways overview.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_diagram_profile">diagram_profile</code></td>
<td>
<p>Color profile of diagrams, should be one of &quot;Copper&quot;
(default), &quot;Copper Plus&quot;, &quot;Barium Lithium&quot; or &quot;calcium salts&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_analysis_profile">analysis_profile</code></td>
<td>
<p>Color profile of analysis, should be one of:
&quot;Standard&quot; (default), &quot;Strosobar&quot; or &quot;Copper Plus&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_token">token</code></td>
<td>
<p>The analysis Token for which the results will be overlaid on
top of the given pathways overview. see:
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_exp_column">exp_column</code></td>
<td>
<p>numeric: (only if token is supplied) Specify the
expression column for the overlay.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_image_quality">image_quality</code></td>
<td>
<p>Numeric: A number  ranging from 1 to 10. 1 is the
lowest quality and 10 is the highest (default = 5).</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_flag_element">flag_element</code></td>
<td>
<p>Gene name, protein ID, chemical ID or Reactome ID of a
diagram's element to be flagged.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_flg_interactors">flg_interactors</code></td>
<td>
<p>Logical: Should the interactor be considered when
flagging a diagram element? (default = TRUE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_sel">sel</code></td>
<td>
<p>CSV line for highlighting element(s) selection in the diagram.</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_title">title</code></td>
<td>
<p>Logical: Should the pathway name be displayed below the image?
(default = TRUE)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_margin">margin</code></td>
<td>
<p>Numeric: A number ranging from 0 to 20 to set as the image's
margin. (default = 15)</p>
</td></tr>
<tr><td><code id="rba_reactome_exporter_reaction_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function will save Reaction-like event separately and out
of it's parent pathway context. To overlay a Reaction on it's parent
pathway, use <code><a href="#topic+rba_reactome_exporter_diagram">rba_reactome_exporter_diagram</a></code>.
</p>


<h3>Value</h3>

<p>NULL, Based to the inputs, an image file will be saved to disk.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/exporter/reaction/
{identifier}.{ext}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_exporter_diagram">rba_reactome_exporter_diagram</a></code>
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a></code>
</p>
<p>Other &quot;Reactome Content Service - Format Exporter&quot;: 
<code><a href="#topic+rba_reactome_exporter_diagram">rba_reactome_exporter_diagram</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_event">rba_reactome_exporter_event</a>()</code>,
<code><a href="#topic+rba_reactome_exporter_overview">rba_reactome_exporter_overview</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_reactome_exporter_diagram(event_id = "R-HSA-6787403",
    create_document = FALSE)

## End(Not run)
## Not run: 
rba_reactome_exporter_diagram(event_id = "R-HSA-6787403",
     output_format = "svg",
     save_to = "reactome_reacion_image.svg")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_reactome_interactors_psicquic'>The interface From Reactome to PSICQUIC</h2><span id='topic+rba_reactome_interactors_psicquic'></span>

<h3>Description</h3>

<p>You can call this function in two scenarios:
1- To retrieve information of all available PSICQUIC resources, call the
function without providing any argument; i.e
rba_reactome_interactors_psicquic().
2-To retrieve a list of interactors of specific protein(s), fill out the
function's arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_interactors_psicquic(
  proteins = NULL,
  resource = NULL,
  details = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_interactors_psicquic_+3A_proteins">proteins</code></td>
<td>
<p>Proteins to retrieve PSICQUIC interactors.</p>
</td></tr>
<tr><td><code id="rba_reactome_interactors_psicquic_+3A_resource">resource</code></td>
<td>
<p>The PSICQUIC resource for your supplied proteins. Call
rba_reactome_interactors_psicquic() without argument to get the available
options.</p>
</td></tr>
<tr><td><code id="rba_reactome_interactors_psicquic_+3A_details">details</code></td>
<td>
<p>Logical: If TRUE (default) a detailed list of interactors
will be returned. If FALSE, only a summary of available interactors will
be returned.</p>
</td></tr>
<tr><td><code id="rba_reactome_interactors_psicquic_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending your input, a list containing the detailed or summary of
PSICQUIC interactions or a data frame of all registered PSICQUIC
resources.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://reactome.org/ContentService/interactors/psicquic/molecules/
{resource}/details&quot;
<br /> &quot;POST https://reactome.org/ContentService/interactors/psicquic/molecules/
{resource}/summary&quot;
<br /> &quot;GET https://reactome.org/ContentService/interactors/psicquic/resources&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Molecule Interactors&quot;: 
<code><a href="#topic+rba_reactome_interactors_static">rba_reactome_interactors_static</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_interactors_psicquic()


rba_reactome_interactors_psicquic(proteins = c("TP53", "MYC"),
    resource = "BioGrid",
    details = FALSE)


rba_reactome_interactors_psicquic(proteins = c("TP53", "MYC"),
    resource = "BioGrid",
    details = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_interactors_static'>Get Static(IntAct) Interaction Information of a Protein</h2><span id='topic+rba_reactome_interactors_static'></span>

<h3>Description</h3>

<p>Reactome maintain a locally host a version of IntAct(Static) interactions
database. Using this function, you can retrieve IntAct information of
a protein(s) in two scenarios: </p>

<ol>
<li><p> If endpoint = &quot;details&quot; or &quot;summary&quot;: Retrieve a detailed/summary
information of your supplied protein accession(s) from IntAct database.
</p>
</li>
<li><p> If endpoint = &quot;pathway&quot;, Retrieve a list of Reactome pathways which
include your supplied protein accession. Pathways with the class
&quot;TopLevelPathway&quot; will be excluded.</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_interactors_static(
  proteins,
  endpoint = "details",
  only_diagrammed = FALSE,
  species = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_interactors_static_+3A_proteins">proteins</code></td>
<td>
<p>Uniprot proteins accession(s). If endpoint = &quot;pathway&quot;,
only a single protein accession can be supplied.</p>
</td></tr>
<tr><td><code id="rba_reactome_interactors_static_+3A_endpoint">endpoint</code></td>
<td>
<p>Can be one of: </p>

<ol>
<li><p> &quot;details&quot;: To return a detailed information of your supplied
protein(s) accession.
</p>
</li>
<li><p> &quot;summary&quot;: To return a summary of your supplied protein(s) accession
</p>
</li>
<li><p> &quot;pathway&quot;: To return a list of pathways containing the interacting
molecules (excluding TopLevelPathway class).</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_reactome_interactors_static_+3A_only_diagrammed">only_diagrammed</code></td>
<td>
<p>Logical: (only when &quot;endpoint = &quot;pathway&quot;)
If TRUE, pathways without diagram will be excluded. (default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_interactors_static_+3A_species">species</code></td>
<td>
<p>Only when &quot;endpoint = &quot;pathway&quot;, The scientific name
of the species to search for the pathways. See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_interactors_static_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List which it's content varies based on the  supplied &quot;endpoint&quot;
argument.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://reactome.org/ContentService/interactors/static/
molecules/details&quot;
<br /> &quot;POST https://reactome.org/ContentService/interactors/static/
molecules/summary&quot;
<br /> &quot;GET https://reactome.org/ContentService/interactors/static/
molecules/pathways&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Molecule Interactors&quot;: 
<code><a href="#topic+rba_reactome_interactors_psicquic">rba_reactome_interactors_psicquic</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_interactors_static(proteins = "Q9BXM7-1",
    endpoint = "pathways", species = "Homo sapiens")


rba_reactome_interactors_static(proteins = c("Q9BXM7-1", "Q13501"),
    endpoint = "details")


rba_reactome_interactors_static(proteins = c("Q9BXM7-1", "Q13501"),
    endpoint = "summary")


</code></pre>

<hr>
<h2 id='rba_reactome_mapping'>Map External ID to Reactome Pathways/Reactions</h2><span id='topic+rba_reactome_mapping'></span>

<h3>Description</h3>

<p>By providing an external identifier from a given resource, you can retrieve
a list of pathways/reactions that include your supplied ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_mapping(id, resource, map_to, species = "Homo sapiens", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_mapping_+3A_id">id</code></td>
<td>
<p>Molecule's external Identifier</p>
</td></tr>
<tr><td><code id="rba_reactome_mapping_+3A_resource">resource</code></td>
<td>
<p>What is the resource of your supplied ID? see:
<a href="https://reactome.org/content/schema/objects/ReferenceDatabase/">Reactome External
Identifiers</a></p>
</td></tr>
<tr><td><code id="rba_reactome_mapping_+3A_map_to">map_to</code></td>
<td>
<p>Either &quot;pathways&quot; or &quot;reactions&quot;.</p>
</td></tr>
<tr><td><code id="rba_reactome_mapping_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human
is 9606), species name (e.g. &quot;Homo sapiens&quot;) or Reactome DbId (e.g
Homo sapiens is 48887). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_mapping_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame where each row is a pathway/reaction and columns are
pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/mapping/{resource}/
{identifier}/pathways&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/mapping/{resource}/
{identifier}/reactions&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_mapping(id = "PTEN", resource =  "UniProt",
    map_to = "reactions", species = 9606)


</code></pre>

<hr>
<h2 id='rba_reactome_orthology'>Get Orthologous (Computationally Inferred) Events</h2><span id='topic+rba_reactome_orthology'></span>

<h3>Description</h3>

<p>Reactome incorporate manually curated human reactions and PANTHER's
protein homology data to Computationally infer events in other eukaryotic
species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_orthology(event_ids, species_dbid, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_orthology_+3A_event_ids">event_ids</code></td>
<td>
<p>Human Reactome event ID(s) to retrieve their orthologous
events.</p>
</td></tr>
<tr><td><code id="rba_reactome_orthology_+3A_species_dbid">species_dbid</code></td>
<td>
<p>Reactome database ID (DbId) of the target species. (e.g
Mus musculus is 48892). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_orthology_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In version 73 (11 June 2020), using an orthology-based approach,
Homo sapiens events was projected to 18,654 orthologous pathways (with
81,835 orthologous proteins) in 15 non-human species.
<br /> See <a href="https://reactome.org/documentation/inferred-events/">
Reactome Computationally Inferred Events</a> for more information.
</p>


<h3>Value</h3>

<p>List containing found Orthologous event(s) in your supplied species
and their pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://reactome.org/ContentService/data/orthologies/ids/
species/{speciesId}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_analysis_species">rba_reactome_analysis_species</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_orthology(event_ids = c("R-HSA-6799198", " R-HSA-72764"),
    species_dbid = 49633)


</code></pre>

<hr>
<h2 id='rba_reactome_participant_of'>Get Larger Reactome Structures Which Include an Entity</h2><span id='topic+rba_reactome_participant_of'></span>

<h3>Description</h3>

<p>This function will retrieve a list of complexes and sets that Your
supplied entity ID participates in (e.g. as a complex component,
reaction output).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_participant_of(entity_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_participant_of_+3A_entity_id">entity_id</code></td>
<td>
<p>Reactome's entity ID.</p>
</td></tr>
<tr><td><code id="rba_reactome_participant_of_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of Reactome database Entities which Your supplied ID is a
participant in them.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/entity/{id}/componentOf&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_participants">rba_reactome_participants</a></code>
</p>
<p>Other &quot;Reactome Content Service - Physical Entity Queries&quot;: 
<code><a href="#topic+rba_reactome_complex_list">rba_reactome_complex_list</a>()</code>,
<code><a href="#topic+rba_reactome_complex_subunits">rba_reactome_complex_subunits</a>()</code>,
<code><a href="#topic+rba_reactome_entity_other_forms">rba_reactome_entity_other_forms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_participant_of(entity_id = "R-HSA-199420")


</code></pre>

<hr>
<h2 id='rba_reactome_participants'>Get Participants of a Reactome Event</h2><span id='topic+rba_reactome_participants'></span>

<h3>Description</h3>

<p>Participating molecules in a Reactome comprises set of 'Physical Entity' and
'Reference Entities' class objects. Use this function to retrieve all, only
'Physical Entity' or only 'Reference Entities' participants of given event.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_participants(
  event_id,
  only_physical_entities = FALSE,
  only_reference_entities = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_participants_+3A_event_id">event_id</code></td>
<td>
<p>Reactome event's database ID (DbId) or Stable ID (StId).</p>
</td></tr>
<tr><td><code id="rba_reactome_participants_+3A_only_physical_entities">only_physical_entities</code></td>
<td>
<p>Logical: If TRUe, only participating
'Physical Entities' will be returned.</p>
</td></tr>
<tr><td><code id="rba_reactome_participants_+3A_only_reference_entities">only_reference_entities</code></td>
<td>
<p>Logical: If TRUe, only participating
'Reference Entities' will be returned.</p>
</td></tr>
<tr><td><code id="rba_reactome_participants_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A 'Physical Entity' Instance could include an individual molecule,
a multi-molecular complex or a set of molecule forming a group based on
some characteristics. a single molecule can have different 'Physical Entity'
instances based on it's associated attributes. For example, IgK Ig kappa
chain, has two 'Physical Entity' instances; one, with ID
<a href="https://reactome.org/content/schema/instance/browser/R-HSA-197041">
&quot;R-HSA-197041&quot;</a> refers to the secreted antibody protein to the
extra-cellular region; And the second one is with ID
<a href="https://reactome.org/content/schema/instance/browser/R-HSA-2038819">
&quot;R-HSA-2038819&quot;</a> and refers to the plasma-membrane-integrated form of
the antibody protein.
<br /> To make it possible to link multiple 'Physical Entity' instances of a
molecule, Reactome uses a data class named &quot;'Reference Entities'&quot;
which correspond to the invariant attribute of a molecule. for example,
both of the above-mentioned 'Physical Entities' see a 'Reference Entities'
named <a href="https://reactome.org/content/schema/instance/browser/57819">
&quot;UniProt:P01834 IGKC</a>.
<br /> See <a href="https://reactome.org/documentation/data-model/">Reactome
Data Model</a> for more information about the data model and Physical
Entities.
</p>


<h3>Value</h3>

<p>List with the participant of your supplied Event ID. A Data frame
if only physical or 'Reference Entities' was requested.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/participants/{id}&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/participants/{id}/
participatingPhysicalEntities&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/participants/{id}/
referenceEntities&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_reactome_participant_of">rba_reactome_participant_of</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_participants("R-HSA-5682012")


rba_reactome_participants("R-HSA-5682012", only_physical_entities = TRUE)


rba_reactome_participants("R-HSA-5682012", only_reference_entities = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_pathways_events'>Get Events Contained in an Upstream Events</h2><span id='topic+rba_reactome_pathways_events'></span>

<h3>Description</h3>

<p>A Reactome Event could be comprised of other events (meaning, a pathway that
include other pathways itself). Use this function to recursively return
all the events which reside downstream of your supplied event ID (or
an attribute of that events).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_pathways_events(event_id, attribute_name = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_pathways_events_+3A_event_id">event_id</code></td>
<td>
<p>Reactome event's database ID (DbId) or Stable ID (StId).</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_events_+3A_attribute_name">attribute_name</code></td>
<td>
<p>An attribute of the events to be returned instead of
the whole events. see <a href="https://reactome.org/content/schema/Event">
Reactome Data Schema: Event</a> for available options.</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_events_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By Reactome's definition, Events are the building blocks of biological
processes and could be of two main classes: &quot;Pathway&quot; or
&quot;Reaction-like events&quot;. The events are organized in a hierarchical
structure; and each event could be child or parent to another event; The
hierarchy will always begin with a &quot;Top level pathway&quot; event. Also note
that a given event could be part of more that one hierarchies.
</p>


<h3>Value</h3>

<p>Data frame where each row is a contained event and columns are
event's attributes. If an &quot;attribute_name&quot; argument was supplied, a
character vector will be returned.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/pathway/{id}/
containedEvents&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/pathway/{id}/
containedEvents/{attributeName}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Pathway Related Queries&quot;: 
<code><a href="#topic+rba_reactome_pathways_low">rba_reactome_pathways_low</a>()</code>,
<code><a href="#topic+rba_reactome_pathways_top">rba_reactome_pathways_top</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_pathways_events(event_id = "R-HSA-5673001")


rba_reactome_pathways_events(event_id = "R-HSA-5673001",
    attribute_name = "displayName")


</code></pre>

<hr>
<h2 id='rba_reactome_pathways_low'>Get lower level pathways Containing a 'Physical Entity' or Event</h2><span id='topic+rba_reactome_pathways_low'></span>

<h3>Description</h3>

<p>Use this function to search the event hierarchy and retrieve a list of
all lower level pathways (non TopLevelPathway class) that contain
a given 'Physical Entity' or Event. See &quot;Arguments section&quot; on how to
modify your search.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_pathways_low(
  entity_id,
  with_diagram = FALSE,
  all_forms = FALSE,
  species = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_pathways_low_+3A_entity_id">entity_id</code></td>
<td>
<p>The entity that should exist in the pathways.</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_low_+3A_with_diagram">with_diagram</code></td>
<td>
<p>Logical: only include pathways with diagram?</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_low_+3A_all_forms">all_forms</code></td>
<td>
<p>Logical: should other variants of your supplied entity_id
be considered? (e.g. same molecule but in different compartment,
secretory form etc.) see <code><a href="#topic+rba_reactome_participants">rba_reactome_participants</a></code>'s
&quot;Details section&quot; to learn more about how Reactome classifies molecules.</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_low_+3A_species">species</code></td>
<td>
<p>(optional) Numeric or Character: confine your search to a
specific species by providing it's NCBI Taxonomy identifier
(Human Taxonomy ID is 9606) or species name (e.g. &quot;Homo sapiens&quot;).
See <code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_low_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame where each row is a pathway that contains your supplied
entity and columns are pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/pathways/low/entity/{id}&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/pathways/low/diagram/
entity/{id}&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/pathways/low/diagram/
entity/{id}/allForms&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Pathway Related Queries&quot;: 
<code><a href="#topic+rba_reactome_pathways_events">rba_reactome_pathways_events</a>()</code>,
<code><a href="#topic+rba_reactome_pathways_top">rba_reactome_pathways_top</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_pathways_low(entity_id = "R-HSA-199420")


rba_reactome_pathways_low(entity_id = "R-HSA-199420", with_diagram = TRUE)


rba_reactome_pathways_low(entity_id = "R-HSA-199420", with_diagram = TRUE,
    all_forms = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_pathways_top'>Get Top Level Pathways in a Species</h2><span id='topic+rba_reactome_pathways_top'></span>

<h3>Description</h3>

<p>This function will Return a list of all pathways with the class
&quot;TopLevelPathway&quot; which are annotated in your supplied species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_pathways_top(species, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_pathways_top_+3A_species">species</code></td>
<td>
<p>Numeric or Character: NCBI Taxonomy identifier (Human Taxonomy
ID is 9606.) or species name (e.g. &quot;Homo sapiens&quot;). See
<code><a href="#topic+rba_reactome_species">rba_reactome_species</a></code> or
<a href="https://reactome.org/content/schema/objects/Species/">Reactome
Data Schema: Entries: Species</a>.</p>
</td></tr>
<tr><td><code id="rba_reactome_pathways_top_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reactome's Events hierarchy for any specie will begin with pathways with
class &quot;TopLevelPathway&quot; (e.g. &quot;Immune System&quot;, &quot;Metabolism of proteins&quot;).
further down in the event's hierarchy tree, each TopLevelPathway has
has other events itself (e.g. &quot;Adaptive immune system&quot;, &quot;Innate immune
system&quot;). Based on the chosen pathway, the hierarchy tree would typically
goes further down.
</p>


<h3>Value</h3>

<p>Data frame where each row is a Top Level Pathway and columns are
pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/pathways/top/{species}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Pathway Related Queries&quot;: 
<code><a href="#topic+rba_reactome_pathways_events">rba_reactome_pathways_events</a>()</code>,
<code><a href="#topic+rba_reactome_pathways_low">rba_reactome_pathways_low</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_pathways_top(species = 9606)


rba_reactome_pathways_top(species = "Saccharomyces cerevisiae")


</code></pre>

<hr>
<h2 id='rba_reactome_people_id'>A person by his identifiers</h2><span id='topic+rba_reactome_people_id'></span>

<h3>Description</h3>

<p>A person by his identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_people_id(
  person_id,
  authored_pathways = FALSE,
  publications = FALSE,
  attribute_name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_people_id_+3A_person_id">person_id</code></td>
<td>
<p>Reactome database ID (DbId) or ORCHID ID</p>
</td></tr>
<tr><td><code id="rba_reactome_people_id_+3A_authored_pathways">authored_pathways</code></td>
<td>
<p>Logical: Only return Pathway list authored by the
person? (default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_people_id_+3A_publications">publications</code></td>
<td>
<p>Logical: Only return publications list authored by the
person? (Defalt = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_people_id_+3A_attribute_name">attribute_name</code></td>
<td>
<p>(optional) A Reactome person attribute to return only.
see <a href="https://reactome.org/content/schema/Person/">Reactome Data
Schema: person</a> for available options.</p>
</td></tr>
<tr><td><code id="rba_reactome_people_id_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the requested informations of your supplied
person.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Person Queries&quot;: 
<code><a href="#topic+rba_reactome_people_name">rba_reactome_people_name</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_people_id("391309")


rba_reactome_people_id(person_id = "391309", authored_pathways = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_people_name'>Get Persons Information by Name</h2><span id='topic+rba_reactome_people_name'></span>

<h3>Description</h3>

<p>Using this function you can query people by partially matching or exact
name and retrieve a list of matching people in Reactome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_people_name(person_name, exact_match = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_people_name_+3A_person_name">person_name</code></td>
<td>
<p>first and last name of the person</p>
</td></tr>
<tr><td><code id="rba_reactome_people_name_+3A_exact_match">exact_match</code></td>
<td>
<p>Logical: should the supplied name be considered as
an exact match? (default = FALSE)</p>
</td></tr>
<tr><td><code id="rba_reactome_people_name_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List where each element is a search hit contains the person's
information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/people/name/{name}&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/people/name/{name}/exact&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Reactome Content Service - Person Queries&quot;: 
<code><a href="#topic+rba_reactome_people_id">rba_reactome_people_id</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_people_name("Jupe")


rba_reactome_people_name("Steve Jupe", exact_match = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_query'>Query and Retrieve any Reactome knowledge-base Object</h2><span id='topic+rba_reactome_query'></span>

<h3>Description</h3>

<p>Using this Comprehensive function, You can Retrieve any object from
<a href="https://reactome.org/content/schema/DatabaseObject/">Reactome
knowledge-base</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_query(
  ids,
  enhanced = FALSE,
  map = FALSE,
  attribute_name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_query_+3A_ids">ids</code></td>
<td>
<p>A single or Multiple database IDs (DbId), Stable IDs (StId) or
a mixture of both.</p>
</td></tr>
<tr><td><code id="rba_reactome_query_+3A_enhanced">enhanced</code></td>
<td>
<p>Logical: (Default = FALSE) If 'TRUE' more information on
the supplied entry will be returned. (You can set this argument to 'TRUE'
Only when you supply a single ID).</p>
</td></tr>
<tr><td><code id="rba_reactome_query_+3A_map">map</code></td>
<td>
<p>(Default = FALSE) Should the supplied IDs be mapped? This
argument will only be considered when you supply multiple IDs.
(e.g. when you supply previous version of stable identifiers.)</p>
</td></tr>
<tr><td><code id="rba_reactome_query_+3A_attribute_name">attribute_name</code></td>
<td>
<p>(Optional) Only Return an Attribute of the supplied
Database Object. (You can use this argument Only when you
supply a single ID)</p>
</td></tr>
<tr><td><code id="rba_reactome_query_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing your query outputs.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://reactome.org/ContentService/data/query/ids&quot;
<br /> &quot;POST https://reactome.org/ContentService/data/query/ids/map&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/query/{id}&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/query/enhanced/{id}&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/query/{id}/{attributeName}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_query(ids = c("8953958", "11982506", "R-ALL-9649879"))


rba_reactome_query(ids = "R-HSA-9656256", enhanced = TRUE)


rba_reactome_query(ids = "8863054", attribute_name = "displayName")


</code></pre>

<hr>
<h2 id='rba_reactome_species'>Get Reactome Species</h2><span id='topic+rba_reactome_species'></span>

<h3>Description</h3>

<p>Use this function to retrieve a table of Available species in Reactome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_species(only_main = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_species_+3A_only_main">only_main</code></td>
<td>
<p>Logical: If set to TRUE, will only return species which
have either manually-curated or computationally inferred pathways.</p>
</td></tr>
<tr><td><code id="rba_reactome_species_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame where each row is a species and columns are pertinent
information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/species/all&quot;
<br /> &quot;GET https://reactome.org/ContentService/data/species/main&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_species()


rba_reactome_species(only_main = TRUE)


</code></pre>

<hr>
<h2 id='rba_reactome_version'>The version number of current database</h2><span id='topic+rba_reactome_version'></span>

<h3>Description</h3>

<p>Returns the current version of Reactome database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_version(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_version_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character string containing the version of Reactome database.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/data/database/version&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_version()


</code></pre>

<hr>
<h2 id='rba_reactome_xref'>Map Cross References IDs to Reactome ReferenceEntity</h2><span id='topic+rba_reactome_xref'></span>

<h3>Description</h3>

<p>Use this function To retrieve a list of Reactome ReferenceEntity associated
to your supplied Cross Reference (i.e. External) ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_reactome_xref(xref_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_reactome_xref_+3A_xref_id">xref_id</code></td>
<td>
<p>molecule's cross-reference (external) identifier.</p>
</td></tr>
<tr><td><code id="rba_reactome_xref_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reactome cross-references external database's identifiers to it's database
Entries named ReferenceEntity, which resembles the invariant aspect of
a molecule. Thus there is a one-to-many relationship between Reactome's
ReferenceEntity object and the molecule's ID in external databases,
which in Reactome's terms is called Cross Reference.
<br /> See <code><a href="#topic+rba_reactome_participants">rba_reactome_participants</a></code>'s &quot;Details section&quot;
to learn more about how Reactome classifies molecules.
</p>


<h3>Value</h3>

<p>List containing the ReferenceEntity corresponding to your
supplied cross-reference (external) ID.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://reactome.org/ContentService/references/mapping/{identifier}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Marc Gillespie, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen
Rothfels, Andrea Senff-Ribeiro, Johannes Griss, Cristoffer Sevilla, Lisa
Matthews, Chuqiao Gong, Chuan Deng, Thawfeek Varusai, Eliot Ragueneau,
Yusra Haider, Bruce May, Veronica Shamovsky, Joel Weiser, Timothy Brunson,
Nasim Sanati, Liam Beckman, Xiang Shao, Antonio Fabregat, Konstantinos
Sidiropoulos, Julieth Murillo, Guilherme Viteri, Justin Cook, Solomon
Shorser, Gary Bader, Emek Demir, Chris Sander, Robin Haw, Guanming Wu,
Lincoln Stein, Henning Hermjakob, Peter D’Eustachio, The reactome
pathway knowledgebase 2022, Nucleic Acids Research, 2021;, kab1028,
https://doi.org/10.1093/nar/gkab1028
</p>
</li>
<li><p> Griss J, Viteri G, Sidiropoulos K, Nguyen V, Fabregat A,
Hermjakob H. ReactomeGSA - Efficient Multi-Omics Comparative Pathway
Analysis. Mol Cell Proteomics. 2020 Sep 9. doi: 10.1074/mcp. PubMed
PMID: 32907876.
</p>
</li>
<li> <p><a href="https://reactome.org/ContentService/">Reactome Content
Services API Documentation</a>
</p>
</li>
<li> <p><a href="https://reactome.org/cite/">Citations note on Reactome website</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
rba_reactome_xref("CD40")


rba_reactome_xref("ENSP00000361350")


</code></pre>

<hr>
<h2 id='rba_string_annotations'>Retrieving Functional Annotation</h2><span id='topic+rba_string_annotations'></span>

<h3>Description</h3>

<p>STRING cross-reference the proteins with several databases (see &quot;Details&quot;
section). By providing your input set o proteins (and optionally
background or universe protein set), you can use this function to
retrieve full set of terms (annotations) pertinent to your input proteins in
each database, among with information for each term.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_annotations(
  ids,
  species = NULL,
  allow_pubmed = FALSE,
  split_df = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_annotations_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_annotations_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_annotations_+3A_allow_pubmed">allow_pubmed</code></td>
<td>
<p>logical: (default = FALSE) PubMed usually  assigns a
large number of reference publications to each protein. In order to reduce
the output size, PubMed's results will be excluded from the results,
unless stated otherwise by setting this argument to TRUE.</p>
</td></tr>
<tr><td><code id="rba_string_annotations_+3A_split_df">split_df</code></td>
<td>
<p>(logical, default = TRUE), If TRUE, instead of one
data frame, results from different categories will be split into
multiple data frames based on their 'category'.</p>
</td></tr>
<tr><td><code id="rba_string_annotations_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>STRING currently maps to and retrieve enrichment results based on Gene
Ontology (GO), KEGG pathways, UniProt Keywords, PubMed publications, Pfam
domains, InterPro domains, and SMART domains.
<br /> Note that this function will return a full list of the terms containing
your supplied proteins. To perform enrichment and only retrieve a enriched
subset of the terms, use <code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a></code>.
</p>


<h3>Value</h3>

<p>A data frame which every row is an assigned terms and the columns
are the terms category, description, number of genes, and other pertinent
information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output_format}/functional_annotation?
identifiers={your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>,
  <a href="#topic+rba_string_enrichment">rba_string_enrichment</a>,
  <a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>
  </code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_annotations(ids = "TP53", species = 9606)


</code></pre>

<hr>
<h2 id='rba_string_enrichment'>Getting Functional Enrichment</h2><span id='topic+rba_string_enrichment'></span>

<h3>Description</h3>

<p>STRING cross-reference the proteins with several databases (see &quot;Details&quot;
section). By providing your input set o proteins (and optionally
background or universe protein set), you can use this function to
perform enrichment test and retrieve a list of enriched terms in each
database, among with pertinent information for each term.
Use <code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a></code> to retrieve the analysis
results as a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_enrichment(
  ids,
  species = NULL,
  background = NULL,
  split_df = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_enrichment_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.
Note that if only one id is supplied, STRING expands the network by 10
proteins.</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_+3A_background">background</code></td>
<td>
<p>character vector: A set of STRING protein IDs
to be used as the statistical background (or universe) when computing
P-value for the terms. Only STRING IDs are acceptable. (See
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> to map your IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_+3A_split_df">split_df</code></td>
<td>
<p>(logical, default = TRUE), If TRUE, instead of one
data frame, results from different categories will be split into
multiple data frames based on their 'category'.</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>STRING currently maps to and retrieve enrichment results based on Gene
Ontology (GO), KEGG pathways, UniProt Keywords, PubMed publications, Pfam
domains, InterPro domains, and SMART domains.
<br /> Note that this function will only return the enriched terms pertinent
to your proteins that have a p-value lesser than 0.1. To retrieve a full
list of the terms -unfiltered by enrichment p-values-, use
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a></code>.
</p>


<h3>Value</h3>

<p>A list of data frames which every row is an enriched terms with p-value
smaller than 0.1 and the columns are the terms category, description,
number of genes, p-value, fdr and other pertinent information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output_format}/enrichment?identifiers=
{your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>,
  <a href="#topic+rba_string_annotations">rba_string_annotations</a>,
  <a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>
  </code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>
<p>Other &quot;Enrichment/Over-representation&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_enrichment(ids = c("TP53", "TNF", "EGFR"), species = 9606)


</code></pre>

<hr>
<h2 id='rba_string_enrichment_image'>Get STRING Enrichment Plot</h2><span id='topic+rba_string_enrichment_image'></span>

<h3>Description</h3>

<p>In addition to performing enrichment analysis, STRING allows you to also
visualize the analysis results. Use <code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a></code>
to retrieve the analysis results as a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_enrichment_image(
  ids,
  species,
  category = "Process",
  image_format = "image",
  save_image = TRUE,
  group_by_similarity = NULL,
  color_palette = "mint_blue",
  number_of_term_shown = 10,
  x_axis = "signal",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_enrichment_image_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.
Note that if only one id is supplied, STRING expands the network by 10
proteins.</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_category">category</code></td>
<td>
<p>The terms set to use to perform enrichment analysis. valid
values are (See details for more info):
&quot;Process&quot; (default), &quot;Function&quot;, &quot;Component&quot;, &quot;Keyword&quot;, &quot;KEGG&quot;, &quot;RCTM&quot;,
&quot;HPO&quot;, &quot;MPO&quot;, &quot;DPO&quot;, &quot;WPO&quot;, &quot;ZPO&quot;, &quot;FYPO&quot;, &quot;Pfam&quot;, &quot;SMART&quot;, &quot;InterPro&quot;,
&quot;PMID&quot;, &quot;NetworkNeighborAL&quot;, &quot;COMPARTMENTS&quot;, &quot;TISSUES&quot;, &quot;DISEASES&quot;, or
&quot;WikiPathways&quot;</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_image_format">image_format</code></td>
<td>
<p>one of:</p>

<ul>
<li><p> &quot;image&quot;: PNG image with normal resolution.
</p>
</li>
<li><p> &quot;highres_image&quot;: High-resolution PNG image.
</p>
</li>
<li><p> &quot;svg&quot;: Scalable Vector Graphics image.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_save_image">save_image</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li><p> TRUE: Save the image to an automatically-generated path.
</p>
</li>
<li><p> FALSE: Do not save the image, just return it as an R object.
</p>
</li>
<li><p> Character string: A valid file path to save the image to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_group_by_similarity">group_by_similarity</code></td>
<td>
<p>Jackard index treshold to visually group the
related terms. Valid values are between 0.1 to 1 with increment of 0.1.
Default value is NULL (i.e. no grouping).</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_color_palette">color_palette</code></td>
<td>
<p>Color pallet to code FDR values. Valid values are:
&quot;mint_blue&quot; (default), &quot;lime_emerald&quot;, &quot;green_blue&quot;, &quot;peach_purple&quot;,
&quot;straw_navy&quot;, or &quot;yellow_pink&quot;</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_number_of_term_shown">number_of_term_shown</code></td>
<td>
<p>(default: 10) Maximum number of results to
include in the plot.</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_x_axis">x_axis</code></td>
<td>
<p>The variable to show on the x axis and rank the results based
on it. Valid values are: &quot;signal&quot; (default), &quot;strength&quot;, &quot;FDR&quot;, or
&quot;gene_count&quot;</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_image_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Available values for category are as follow. Default value is &quot;Process&quot;.
</p>

<ul>
<li><p> Process: Biological Process (Gene Ontology)
</p>
</li>
<li><p> Function: Molecular Function (Gene Ontology)
</p>
</li>
<li><p> Component: Cellular Component (Gene Ontology)
</p>
</li>
<li><p> Keyword: Annotated Keywords (UniProt)
</p>
</li>
<li><p> KEGG: KEGG Pathways
</p>
</li>
<li><p> RCTM: Reactome Pathways
</p>
</li>
<li><p> HPO: Human Phenotype (Monarch)
</p>
</li>
<li><p> MPO: The Mammalian Phenotype Ontology (Monarch)
</p>
</li>
<li><p> DPO: Drosophila Phenotype (Monarch)
</p>
</li>
<li><p> WPO: C. elegans Phenotype Ontology (Monarch)
</p>
</li>
<li><p> ZPO: Zebrafish Phenotype Ontology (Monarch)
</p>
</li>
<li><p> FYPO: Fission Yeast Phenotype Ontology (Monarch)
</p>
</li>
<li><p> Pfam: Protein Domains (Pfam)
</p>
</li>
<li><p> SMART: Protein Domains (SMART)
</p>
</li>
<li><p> InterPro: Protein Domains and Features (InterPro)
</p>
</li>
<li><p> PMID: Reference Publications (PubMed)
</p>
</li>
<li><p> NetworkNeighborAL: Local Network Cluster (STRING)
</p>
</li>
<li><p> COMPARTMENTS: Subcellular Localization (COMPARTMENTS)
</p>
</li>
<li><p> TISSUES: Tissue Expression (TISSUES)
</p>
</li>
<li><p> DISEASES: Disease-gene Associations (DISEASES)
</p>
</li>
<li><p> WikiPathways: WikiPathways</p>
</li></ul>



<h3>Value</h3>

<p>A plot summarizing the enrichment results, which can be PNG or
SVG depending on the inputs.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output_format}/enrichmentfigure&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>,
  <a href="#topic+rba_string_enrichment">rba_string_enrichment</a>,
  <a href="#topic+rba_string_annotations">rba_string_annotations</a>
  </code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>
<p>Other &quot;Enrichment/Over-representation&quot;: 
<code><a href="#topic+rba_enrichr">rba_enrichr</a>()</code>,
<code><a href="#topic+rba_mieaa_enrich">rba_mieaa_enrich</a>()</code>,
<code><a href="#topic+rba_panther_enrich">rba_panther_enrich</a>()</code>,
<code><a href="#topic+rba_reactome_analysis">rba_reactome_analysis</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  rba_string_enrichment_image(
  ids = c("TP53", "TNF", "EGFR"),
  species = 9606,
  category = "KEGG"
  )

## End(Not run)
## Not run: 
  rba_string_enrichment_image(
  ids = c("TP53", "TNF", "EGFR"),
  species = 9606,
  x_axis = "strength",
  number_of_term_shown = 20
  )

## End(Not run)
## Not run: 
  rba_string_enrichment_image(
  ids = c("TP53", "TNF", "EGFR"),
  species = 9606,
  color_palette = "straw_navy"
  )

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_string_enrichment_ppi'>Get Protein-Protein Interaction Enrichment</h2><span id='topic+rba_string_enrichment_ppi'></span>

<h3>Description</h3>

<p>Even when there is no annotation for your input proteins, STRING can Compare
your Given proteins interactions pattern with the background proteome-wide
interaction distribution to determine if your given set of proteins are
functionally related.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_enrichment_ppi(
  ids,
  species = NULL,
  required_score = NULL,
  background = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_enrichment_ppi_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_ppi_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_ppi_+3A_required_score">required_score</code></td>
<td>
<p>Numeric: A minimum of interaction score for an
interaction to be included in the image. if not supplied, the threshold
will be applied by STRING Based in the network. (low Confidence = 150,
Medium Confidence = 400, High Confidence = 700, Highest confidence = 900)</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_ppi_+3A_background">background</code></td>
<td>
<p>character vector: A set of STRING protein IDs
to be used as the background proteome. Only STRING IDs are acceptable.
(See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> to map your IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_enrichment_ppi_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with protein-protein interaction enrichment results.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output_format}/ppi_enrichment?identifiers=
{your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_enrichment_ppi(ids = c("p53", "BRCA1", "cdk2", "Q99835",
       "CDC42", "CDK1", "KIF23", "PLK1", "RAC2", "RACGAP1"),
    species = 9606)


</code></pre>

<hr>
<h2 id='rba_string_homology_inter'>Get Similarity Scores Hits of Proteins in Different Species</h2><span id='topic+rba_string_homology_inter'></span>

<h3>Description</h3>

<p>Using this function, you can retrieve highest Smith-Waterman bit scores
among your input proteins and proteins in every other STRING species
(e.g. the closest homologous protein of your input protein in other
species).
Bit Scores serve as similarity scores between protein sequence;
And, according to STRING documentations, as a proxy for protein homology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_homology_inter(ids, species = NULL, species_b = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_homology_inter_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_homology_inter_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier of your input proteins;
Human Taxonomy ID is 9606. (Recommended, but optional if your input is
less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_homology_inter_+3A_species_b">species_b</code></td>
<td>
<p>(optional) Numeric: one or more NCBI Taxonomy identifiers
of species to limit the closets homologous proteins search.</p>
</td></tr>
<tr><td><code id="rba_string_homology_inter_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function will return the highest similarity score hits of
your given protein(s) and their closets homologous proteins in other
species. to retrieve similarity scores of different proteins within the
same species see <code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a></code>.
<br /> Similarity matrix is imported -by STRING- from:
<a href="https://cube.univie.ac.at/research/software-databases/simap/">Similarity Matrix of
Proteins (SIMAP)</a>
</p>


<h3>Value</h3>

<p>A data frame with Your input proteins and it's closest homologous
proteins among all other (or a defined) STRING species.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output-format}/homology_best?
identifiers={your_identifiers}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>, <a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a></code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_homology_inter(ids = "p53",
    species = 9606,
    species_b = 7070)


rba_string_homology_inter(ids = "ENSP00000269305", species = 9606)


</code></pre>

<hr>
<h2 id='rba_string_homology_intra'>Get Similarity Scores Hits of Proteins in a Species</h2><span id='topic+rba_string_homology_intra'></span>

<h3>Description</h3>

<p>Using this function, you can retrieve the Smith-Waterman bit scores among
proteins of the same species.
Bit Scores serve as similarity scores between protein sequence;
And, according to STRING documentations, as a proxy for protein homology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_homology_intra(ids, species = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_homology_intra_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_homology_intra_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_homology_intra_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function will retrieve similarity scores of different
proteins &quot;within the same species&quot;. To Get a similarity scores of a given
protein and it's closets homologous proteins in other species, see
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a></code>.
<br /> Similarity matrix is imported -by STRING- from:
<a href="https://cube.univie.ac.at/research/software-databases/simap/">Similarity Matrix of
Proteins (SIMAP)</a>
</p>


<h3>Value</h3>

<p>A data frame with bit scores between your supplied proteins and
their self-hit. To Reduce the transferred data, STRING returns only one
half of the similarity matrix; This will not pose a problem because
similarity matrix is symmetrical.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output-format}/homology?identifiers=
{your_identifiers}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>, <a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a></code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_homology_intra(ids = c("CDK1", "CDK2"), species = 9606)


</code></pre>

<hr>
<h2 id='rba_string_interaction_partners'>Get All STRING Interaction Partners</h2><span id='topic+rba_string_interaction_partners'></span>

<h3>Description</h3>

<p>This function will retrieve all the STRING interactions which include your
proteins as one party of the interaction. (e.g. interaction between your
proteins and every other STRING proteins.)
<br /> Given the size of STRING database, this function could return a very
long results. See &quot;Arguments&quot; section for information on how to filter
the interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_interaction_partners(
  ids,
  species = NULL,
  required_score = NULL,
  network_type = "functional",
  limit = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_interaction_partners_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_interaction_partners_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_interaction_partners_+3A_required_score">required_score</code></td>
<td>
<p>Numeric: A minimum of interaction score for an
interaction to be included in the image. if not supplied, the threshold
will be applied by STRING Based in the network. (low Confidence = 150,
Medium Confidence = 400, High Confidence = 700, Highest confidence = 900)</p>
</td></tr>
<tr><td><code id="rba_string_interaction_partners_+3A_network_type">network_type</code></td>
<td>
<p>should be one of:</p>

<ul>
<li><p> &quot;functional&quot;: (default) The edge's indicate both physical and
functional associations.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_interaction_partners_+3A_limit">limit</code></td>
<td>
<p>Limit the number returned interaction partners per each of
your input proteins. (e.g. Number of the most confident interaction partner
to return per each input protein.)</p>
</td></tr>
<tr><td><code id="rba_string_interaction_partners_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function will retrieve the interactions between your input
proteins and every other STRING proteins. To retrieve the interaction
among your input protein-set, see
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a></code>.
</p>


<h3>Value</h3>

<p>A data frame which each row is a network interaction and the
columns contains interactor information and interaction scores.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output-format}/
interaction_partners?identifiers={your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>, <a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a></code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_interaction_partners(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    network_type = "functional")


    rba_string_interaction_partners(ids = "9606.ENSP00000269305",
    species = 9606,
    required_score = 700)


</code></pre>

<hr>
<h2 id='rba_string_interactions_network'>Get STRING Network Interactions</h2><span id='topic+rba_string_interactions_network'></span>

<h3>Description</h3>

<p>This function will retrieve Sting interaction pairs among your input protein
ids, with the combined score and separate score for each STRING score
channels. You can further expand your network to a defined size by
providing &quot;add_node&quot; parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_interactions_network(
  ids,
  species = NULL,
  required_score = NULL,
  add_nodes = NULL,
  network_type = "functional",
  use_query_labels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_interactions_network_+3A_ids">ids</code></td>
<td>
<p>Your protein IDs. It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_interactions_network_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_interactions_network_+3A_required_score">required_score</code></td>
<td>
<p>Numeric: A minimum of interaction score for an
interaction to be included in the image. if not supplied, the threshold
will be applied by STRING Based in the network. (low Confidence = 150,
Medium Confidence = 400, High Confidence = 700, Highest confidence = 900)</p>
</td></tr>
<tr><td><code id="rba_string_interactions_network_+3A_add_nodes">add_nodes</code></td>
<td>
<p>Numeric: Number of neighboring proteins to be added to the
network. If none supplied by the user, this argument value will depend
on the number of supplied &quot;ids&quot; argument:</p>

<ol>
<li><p> Single id: add_node will be set to 10 to retrieve the interaction
neighborhood  of you input protein.
</p>
</li>
<li><p> Multiple ids: add_node will be set to 0, thus the output will be the
interactions between your input proteins.</p>
</li></ol>
</td></tr>
<tr><td><code id="rba_string_interactions_network_+3A_network_type">network_type</code></td>
<td>
<p>should be one of:</p>

<ul>
<li><p> &quot;functional&quot;: (default) The edge's indicate both physical and
functional associations.
</p>
</li>
<li><p> &quot;physical&quot;: The edges indicate that two proteins have a physical
interaction or are parts of a complex.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_interactions_network_+3A_use_query_labels">use_query_labels</code></td>
<td>
<p>Logical: (Default = FALSE) Use the names supplied
with the 'ids' argument as the nodes labels instead of STRING's default
ones.</p>
</td></tr>
<tr><td><code id="rba_string_interactions_network_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function will return interactions between your set of
supplied proteins, or at most, expand the interaction network by the
given parameters. TO retrieve a list of all possible interacting proteins
with your given input, see
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a></code>.
</p>


<h3>Value</h3>

<p>A data frame which each row is a network interaction and the
columns contains interactor information and interaction scores: </p>

<ul>
<li><p> stringId_A: STRING identifier (protein A)
</p>
</li>
<li><p> stringId_B:STRING identifier (protein B)
</p>
</li>
<li><p> preferredName_A: common protein name (protein A)
</p>
</li>
<li><p> preferredName_B: common protein name (protein B)
</p>
</li>
<li><p> ncbiTaxonId: NCBI taxon identifier
</p>
</li>
<li><p> score: combined score
</p>
</li>
<li><p> nscore: gene neighborhood score
</p>
</li>
<li><p> fscore: gene fusion score
</p>
</li>
<li><p> pscore: phylogenetic profile score
</p>
</li>
<li><p> ascore: co-expression score
</p>
</li>
<li><p> escore: experimental score
</p>
</li>
<li><p> dscore: database score
</p>
</li>
<li><p> tscore: textmining score</p>
</li></ul>



<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output-format}/network?identifiers=
{your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>, <a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a></code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_interactions_network(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    network_type = "functional")


rba_string_interactions_network(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    species = 9606,
    add_nodes = 10)


</code></pre>

<hr>
<h2 id='rba_string_map_ids'>Map a Set of Identifiers to STRING Identifiers</h2><span id='topic+rba_string_map_ids'></span>

<h3>Description</h3>

<p>This function Calls STRING's API to Convert a set of identifiers
to STRING Identifiers. Although You can call STRING services with a variety
of common identifiers, It is recommended by STRING's documentations that
you first map Your Protein/genes IDs to STRING IDs and then proceed
with other STRING's functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_map_ids(ids, species = NULL, echo_query = FALSE, limit = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_map_ids_+3A_ids">ids</code></td>
<td>
<p>Your Common gene/protein Identifier(s) to be mapped.</p>
</td></tr>
<tr><td><code id="rba_string_map_ids_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_map_ids_+3A_echo_query">echo_query</code></td>
<td>
<p>(default = FALSE) Include your input IDs as a column of the
results.</p>
</td></tr>
<tr><td><code id="rba_string_map_ids_+3A_limit">limit</code></td>
<td>
<p>(Numeric, Optional) A limit on the number of matches per input
ID. The output are sorted to have the best matches first.</p>
</td></tr>
<tr><td><code id="rba_string_map_ids_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the mapped STRING IDs and other pertinent
information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output-format}/get_string_ids?identifiers=
{your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_map_ids(ids = c("TP53", "TNF", "EGFR"), species = 9606)


</code></pre>

<hr>
<h2 id='rba_string_network_image'>Get STRING Network Image</h2><span id='topic+rba_string_network_image'></span>

<h3>Description</h3>

<p>Depending on that you supplied a single protein ID or more than one protein
ID, this function will produce a static image of the interaction networks
among your input proteins or/and with other proteins. See the
&quot;Arguments&quot; section to learn more about how you can modify the network
image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_network_image(
  ids,
  image_format = "image",
  save_image = TRUE,
  species = NULL,
  add_color_nodes = NULL,
  add_white_nodes = NULL,
  required_score = NULL,
  network_flavor = "evidence",
  network_type = "functional",
  hide_node_labels = FALSE,
  use_query_labels = FALSE,
  hide_disconnected_nodes = FALSE,
  hide_structure_pics = FALSE,
  flat_nodes = FALSE,
  node_labels_center = FALSE,
  node_labels_font_size = 12,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_network_image_+3A_ids">ids</code></td>
<td>
<p>Your protein ID(s). It is strongly recommended to supply
STRING IDs. See <code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code> for more information.</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_image_format">image_format</code></td>
<td>
<p>one of:</p>

<ul>
<li><p> &quot;image&quot;: PNG image with normal resolution.
</p>
</li>
<li><p> &quot;highres_image&quot;: High-resolution PNG image.
</p>
</li>
<li><p> &quot;svg&quot;: Scalable Vector Graphics image.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_save_image">save_image</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li><p> TRUE: Save the image to an automatically-generated path.
</p>
</li>
<li><p> FALSE: Do not save the image, just return it as an R object.
</p>
</li>
<li><p> Character string: A valid file path to save the image to.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_species">species</code></td>
<td>
<p>Numeric: NCBI Taxonomy identifier; Human Taxonomy ID is 9606.
(Recommended, but optional if your input is less than 100 IDs.)</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_add_color_nodes">add_color_nodes</code></td>
<td>
<p>Numeric: The number of colored nodes (queried proteins
and first shell of interactors) to be added.</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_add_white_nodes">add_white_nodes</code></td>
<td>
<p>Numeric: The number of white nodes (second shell of
interactors) to be added after colored nodes.</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_required_score">required_score</code></td>
<td>
<p>Numeric (Between 0 to 1000): A minimum of interaction
score for an interaction to be included in the image. if not supplied, the
threshold will be applied by STRING Based in the network. (low Confidence
= 150, Medium Confidence = 400, High Confidence = 700, Highest
confidence = 900)</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_network_flavor">network_flavor</code></td>
<td>
<p>The style of network edges, should be one of:</p>

<ul>
<li><p> &quot;evidence&quot;: (default) Line's color is based on the type of evidences that
support the interaction.
</p>
</li>
<li><p> &quot;confidence&quot;: Line's thickness is an indicator of the
interaction's confidence score.
</p>
</li>
<li><p> &quot;action&quot;: Line's Shape is an indicator of the interaction's predicted
mode of actions.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_network_type">network_type</code></td>
<td>
<p>should be one of:</p>

<ul>
<li><p> &quot;functional&quot;: (default) The edge's indicate both physical and
functional associations.
</p>
</li>
<li><p> &quot;physical&quot;: The edges indicate that two proteins have a physical
interaction or are parts of a complex.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_hide_node_labels">hide_node_labels</code></td>
<td>
<p>Logical: (Default = FALSE) Hide proteins names from
the image</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_use_query_labels">use_query_labels</code></td>
<td>
<p>Logical: (Default = FALSE) Use the names supplied
with the 'ids' argument as the nodes labels instead of STRING's default
ones.</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_hide_disconnected_nodes">hide_disconnected_nodes</code></td>
<td>
<p>Logical: (Default = FALSE) Hide proteins that
are not connected to any other proteins from the image</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_hide_structure_pics">hide_structure_pics</code></td>
<td>
<p>Logical: (Default = FALSE) Hide protein's
structure picture from inside the bubbles</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_flat_nodes">flat_nodes</code></td>
<td>
<p>Logical: (Default = FALSE) Make the nodes design flat
instead of the default 3D design</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_node_labels_center">node_labels_center</code></td>
<td>
<p>Logical: (Default = FALSE) Position the protein
names labels center aligned on the nodes</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_node_labels_font_size">node_labels_font_size</code></td>
<td>
<p>Numeric (Between 5 to 50, Default = 12) Font
size of the protein nodes labels</p>
</td></tr>
<tr><td><code id="rba_string_network_image_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A network images which can be PNG or SVG depending on the inputs.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://string-db.org/api/{output-format}/network?identifiers=
{your_identifiers}&amp;{optional_parameters}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a></code>
</p>
<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_version">rba_string_version</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rba_string_network_image(ids = c("9606.ENSP00000269305",
    "9606.ENSP00000398698",
    "9606.ENSP00000275493"),
    network_type = "functional",
    save_image = FALSE)

## End(Not run)
## Not run: 
rba_string_network_image(ids = c("TP53", "TNF", "EGFR"),
    species = 9606,
    save_image = TRUE)

## End(Not run)
## Not run: 
rba_string_network_image(ids = "9606.ENSP00000269305",
    image_format = "highres_image",
    save_image = file.path(getwd(), "TP53_network.png"))

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_string_version'>Get Current STRING Version</h2><span id='topic+rba_string_version'></span>

<h3>Description</h3>

<p>Get STRING version and stable Address that this package currently uses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_string_version(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_string_version_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that STRING releases new version at approximately 2 years cycle.
Nevertheless, to insure reproducibility, STRING dedicates a stable address
for each release. Thus you can always reproduce research and results
obtained via a certain STRING version. If the version that rbioapi returns
is outdated, Kindly contact me.
</p>


<h3>Value</h3>

<p>A list with STRING version and stable address.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://string-db.org/api/{output_format}/version&quot;
</p>


<h3>References</h3>


<ul>
<li><p> Damian Szklarczyk, Rebecca Kirsch, Mikaela Koutrouli, Katerina
Nastou, Farrokh Mehryary, Radja Hachilif, Annika L Gable, Tao Fang,
Nadezhda T Doncheva, Sampo Pyysalo, Peer Bork, Lars J Jensen, Christian
von Mering, The STRING database in 2023: protein–protein association
networks and functional enrichment analyses for any sequenced genome of
interest, Nucleic Acids Research, Volume 51, Issue D1, 6 January 2023,
Pages D638–D646, https://doi.org/10.1093/nar/gkac1000
</p>
</li>
<li> <p><a href="https://string-db.org/help/api/">STRING API Documentation</a>
</p>
</li>
<li>
<p><a href="https://string-db.org/cgi/about?footer_active_subpage=references">
Citations note on STRING website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;STRING&quot;: 
<code><a href="#topic+rba_string_annotations">rba_string_annotations</a>()</code>,
<code><a href="#topic+rba_string_enrichment">rba_string_enrichment</a>()</code>,
<code><a href="#topic+rba_string_enrichment_image">rba_string_enrichment_image</a>()</code>,
<code><a href="#topic+rba_string_enrichment_ppi">rba_string_enrichment_ppi</a>()</code>,
<code><a href="#topic+rba_string_homology_inter">rba_string_homology_inter</a>()</code>,
<code><a href="#topic+rba_string_homology_intra">rba_string_homology_intra</a>()</code>,
<code><a href="#topic+rba_string_interaction_partners">rba_string_interaction_partners</a>()</code>,
<code><a href="#topic+rba_string_interactions_network">rba_string_interactions_network</a>()</code>,
<code><a href="#topic+rba_string_map_ids">rba_string_map_ids</a>()</code>,
<code><a href="#topic+rba_string_network_image">rba_string_network_image</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_string_version()


</code></pre>

<hr>
<h2 id='rba_uniprot_antigens'>Get Antigens by UniProt Accession</h2><span id='topic+rba_uniprot_antigens'></span>

<h3>Description</h3>

<p>UniProt maps Antigenic features from different sources to the proteins'
sequences. Using this function, you can retrieve all the Antigenic
features that has been map to a given UniProt protein's sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_antigens(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_antigens_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s).</p>
</td></tr>
<tr><td><code id="rba_uniprot_antigens_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the Antigenic features of your supplied
UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/antigen/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Antigen&quot;: 
<code><a href="#topic+rba_uniprot_antigens_search">rba_uniprot_antigens_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_antigens("P04626")


</code></pre>

<hr>
<h2 id='rba_uniprot_antigens_search'>Search Antigens in UniProt</h2><span id='topic+rba_uniprot_antigens_search'></span>

<h3>Description</h3>

<p>UniProt maps Antigenic (Antibody-binding) features from different sources
to the proteins' sequences. Using this function, you can search for
Antigenic sequences that has been map to UniProt proteins. You may also
refine your search with modifiers such as score etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_antigens_search(
  accession = NULL,
  antigen_sequence = NULL,
  antigen_id = NULL,
  ensembl_id = NULL,
  match_score = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_antigens_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_antigens_search_+3A_antigen_sequence">antigen_sequence</code></td>
<td>
<p>Protein sequence in the antigenic site.</p>
</td></tr>
<tr><td><code id="rba_uniprot_antigens_search_+3A_antigen_id">antigen_id</code></td>
<td>
<p>Human Protein Atlas (HPA) antigen ID. You can supply up to
20 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_antigens_search_+3A_ensembl_id">ensembl_id</code></td>
<td>
<p>Ensembl Stable Transcript ID. You can supply up to
20 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_antigens_search_+3A_match_score">match_score</code></td>
<td>
<p>(Numeric) Minimum alignment score for the antigen
sequence and the target protein sequence.</p>
</td></tr>
<tr><td><code id="rba_uniprot_antigens_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein (search
hit) and Antigenic features are organized under the &quot;features&quot; sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/antigen&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Antigen&quot;: 
<code><a href="#topic+rba_uniprot_antigens">rba_uniprot_antigens</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_antigens_search(antigen_id = "HPA001060")


</code></pre>

<hr>
<h2 id='rba_uniprot_coordinates'>Get Genomic Coordinates of a Protein</h2><span id='topic+rba_uniprot_coordinates'></span>

<h3>Description</h3>

<p>Using this function you can retrieve genomic Coordinates of a Protein by
either providing the protein's UniProt accession or it's ID in a
cross-reference database (Ensembl, CCDC, HGNC or RefSeq). You should
supply either 'accession' alone or 'db_type' and 'db_id' together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_coordinates(accession = NULL, db_type = NULL, db_id = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_coordinates_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_+3A_db_type">db_type</code></td>
<td>
<p>cross-reference database name, Should be one of:
&quot;Ensembl&quot;, &quot;CCDC&quot;, &quot;HGNC&quot; or &quot;RefSeq&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_+3A_db_id">db_id</code></td>
<td>
<p>Protein's ID in the cross-reference database</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about how UniProt imports and calculates genomic
coordinates data, see:
<br /> McGarvey, P. B., Nightingale, A., Luo, J., Huang, H., Martin, M. J.,
Wu, C., &amp; UniProt Consortium (2019). UniProt genomic mapping for
deciphering functional effects of missense variants. Human mutation,
40(6), 694–705. https://doi.org/10.1002/humu.23738
</p>


<h3>Value</h3>

<p>A list with genome coordinates of your supplied protein.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/coordinates/{accession}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/coordinates/{dbtype}:{dbid}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Coordinates&quot;: 
<code><a href="#topic+rba_uniprot_coordinates_location">rba_uniprot_coordinates_location</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_genome">rba_uniprot_coordinates_location_genome</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_protein">rba_uniprot_coordinates_location_protein</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_search">rba_uniprot_coordinates_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_coordinates(accession = "P25942")


rba_uniprot_coordinates(db_type = "HGNC", db_id = "CD40")


</code></pre>

<hr>
<h2 id='rba_uniprot_coordinates_location'>Search UniProt entries by taxonomy and genomic coordinates</h2><span id='topic+rba_uniprot_coordinates_location'></span>

<h3>Description</h3>

<p>For more information about how UniProt imports and calculates genomic
coordinates data, see:
<br /> McGarvey, P. B., Nightingale, A., Luo, J., Huang, H., Martin, M. J.,
Wu, C., &amp; UniProt Consortium (2019). UniProt genomic mapping for
deciphering functional effects of missense variants. Human mutation,
40(6), 694–705. https://doi.org/10.1002/humu.23738
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_coordinates_location(
  taxid,
  locations,
  in_range = TRUE,
  feature = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_coordinates_location_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_+3A_locations">locations</code></td>
<td>
<p>genomic location formatted as: chromosome:start-end.
(e.g. &quot;Y:17100001-19600000&quot;). If you omit chromosome, it will be interpreted
as any chromosome (e.g. &quot;1-10000&quot;).</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_+3A_in_range">in_range</code></td>
<td>
<p>Only return proteins that are in range.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_+3A_feature">feature</code></td>
<td>
<p>(logical) Get features?</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing UniProt proteins which match the supplied genomic
location and taxonomy ID.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/coordinates
/{taxonomy}/{locations}/feature&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/coordinates
/{taxonomy}/{locations}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Coordinates&quot;: 
<code><a href="#topic+rba_uniprot_coordinates">rba_uniprot_coordinates</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_genome">rba_uniprot_coordinates_location_genome</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_protein">rba_uniprot_coordinates_location_protein</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_search">rba_uniprot_coordinates_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_coordinates_location(taxid = 9606,
    locations = "Y:17100001-19600000", in_range = TRUE)


rba_uniprot_coordinates_location(taxid = 9606,
    locations = "20:39000001", in_range = FALSE)


</code></pre>

<hr>
<h2 id='rba_uniprot_coordinates_location_genome'>Get Genome coordinate by Gene Sequence position</h2><span id='topic+rba_uniprot_coordinates_location_genome'></span>

<h3>Description</h3>

<p>Using this function you can retrieve genome coordinates of a given UniProt
protein by providing Genome location position or range. You can either
supply 'g_position' alone or supply 'g_start' and 'g_end' together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_coordinates_location_genome(
  taxid,
  chromosome,
  g_position = NULL,
  g_start = NULL,
  g_end = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_coordinates_location_genome_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_genome_+3A_chromosome">chromosome</code></td>
<td>
<p>(Character or Numeric): Chromosome name, e.g. 1, 20, X.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_genome_+3A_g_position">g_position</code></td>
<td>
<p>(numeric) Genome location position</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_genome_+3A_g_start">g_start</code></td>
<td>
<p>(numeric) Genome location position start</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_genome_+3A_g_end">g_end</code></td>
<td>
<p>(numeric) Genome location position end</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_genome_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about how UniProt imports and calculates genomic
coordinates data, see:
<br /> McGarvey, P. B., Nightingale, A., Luo, J., Huang, H., Martin, M. J.,
Wu, C., &amp; UniProt Consortium (2019). UniProt genomic mapping for
deciphering functional effects of missense variants. Human mutation,
40(6), 694–705. https://doi.org/10.1002/humu.23738
</p>


<h3>Value</h3>

<p>Genome coordinates of your supplied proteins.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/coordinates/glocation
/{accession}:{pPosition}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/coordinates/glocation
/{accession}:{pStart}-{pEnd}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Coordinates&quot;: 
<code><a href="#topic+rba_uniprot_coordinates">rba_uniprot_coordinates</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location">rba_uniprot_coordinates_location</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_protein">rba_uniprot_coordinates_location_protein</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_search">rba_uniprot_coordinates_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 rba_uniprot_coordinates_location_genome(
 taxid = 9606, chromosome = 11, g_position = 36573305)


</code></pre>

<hr>
<h2 id='rba_uniprot_coordinates_location_protein'>Get Genome coordinate by Protein Sequence position</h2><span id='topic+rba_uniprot_coordinates_location_protein'></span>

<h3>Description</h3>

<p>Using this function you can retrieve genome coordinates of a given UniProt
protein by providing protein position or position range. You can either
supply 'p_position' alone or supply 'p_start' and 'p_end' together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_coordinates_location_protein(
  accession,
  p_position = NULL,
  p_start = NULL,
  p_end = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_coordinates_location_protein_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_protein_+3A_p_position">p_position</code></td>
<td>
<p>(numeric) Protein sequence position</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_protein_+3A_p_start">p_start</code></td>
<td>
<p>(numeric) Protein sequence position start</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_protein_+3A_p_end">p_end</code></td>
<td>
<p>(numeric) Protein sequence position end</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_location_protein_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about how UniProt imports and calculates genomic
coordinates data, see:
<br /> McGarvey, P. B., Nightingale, A., Luo, J., Huang, H., Martin, M. J.,
Wu, C., &amp; UniProt Consortium (2019). UniProt genomic mapping for
deciphering functional effects of missense variants. Human mutation,
40(6), 694–705. https://doi.org/10.1002/humu.23738
</p>


<h3>Value</h3>

<p>Genome coordinates of your supplied proteins.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/coordinates/location
/{accession}:{pPosition}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/coordinates/location
/{accession}:{pStart}-{pEnd}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Coordinates&quot;: 
<code><a href="#topic+rba_uniprot_coordinates">rba_uniprot_coordinates</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location">rba_uniprot_coordinates_location</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_genome">rba_uniprot_coordinates_location_genome</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_search">rba_uniprot_coordinates_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_coordinates_location_protein(accession = "P25942", p_position = 1)


rba_uniprot_coordinates_location_protein(accession = "P25942",
    p_start = 1, p_end = 277)


</code></pre>

<hr>
<h2 id='rba_uniprot_coordinates_search'>Search Genomic Coordinates of UniProt entries</h2><span id='topic+rba_uniprot_coordinates_search'></span>

<h3>Description</h3>

<p>Use this function to search genomic coordinates of UniProt entries.
You may also refine your search with modifiers such as chromosome, taxon
id etc. See &quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_coordinates_search(
  accession = NULL,
  chromosome = NULL,
  ensembl_id = NULL,
  gene = NULL,
  protein = NULL,
  taxid = NULL,
  location = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_coordinates_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_chromosome">chromosome</code></td>
<td>
<p>chromosome name, such as &quot;X&quot;, &quot;Y&quot;, 1, 20, etc. You can
supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_ensembl_id">ensembl_id</code></td>
<td>
<p>Ensembl Stable gene ID, transcript ID or translation ID.
You can supply up to 20 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_gene">gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_protein">protein</code></td>
<td>
<p><a href="https://www.uniprot.org/help/protein_names">UniProt
protein name</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_location">location</code></td>
<td>
<p>Genome location range such as &quot;58205437-58219305&quot;</p>
</td></tr>
<tr><td><code id="rba_uniprot_coordinates_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br /> For more information about how UniProt imports and calculates genomic
coordinates data, see:
<br /> McGarvey, P. B., Nightingale, A., Luo, J., Huang, H., Martin, M. J.,
Wu, C., &amp; UniProt Consortium (2019). UniProt genomic mapping for
deciphering functional effects of missense variants. Human mutation,
40(6), 694–705. https://doi.org/10.1002/humu.23738
</p>


<h3>Value</h3>

<p>List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing that
protein's coordinates information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/coordinates&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Coordinates&quot;: 
<code><a href="#topic+rba_uniprot_coordinates">rba_uniprot_coordinates</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location">rba_uniprot_coordinates_location</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_genome">rba_uniprot_coordinates_location_genome</a>()</code>,
<code><a href="#topic+rba_uniprot_coordinates_location_protein">rba_uniprot_coordinates_location_protein</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_coordinates_search(taxid = 9606, chromosome = "y")


</code></pre>

<hr>
<h2 id='rba_uniprot_epitope'>Retrieve Epitopes by Accession</h2><span id='topic+rba_uniprot_epitope'></span>

<h3>Description</h3>

<p>Use this function to retrieve epitope annotations linked to a UniProt entry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_epitope(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_epitope_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_epitope_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the UniProt epitope features details for the given
accession.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/epitope/{accession}&quot;
</p>


<h3>See Also</h3>

<p>Other &quot;UniProt - Epitopes&quot;: 
<code><a href="#topic+rba_uniprot_epitope_search">rba_uniprot_epitope_search</a>()</code>,
<code><a href="#topic+rba_uniprot_rna_edit">rba_uniprot_rna_edit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_epitope(accession = "P36222")


</code></pre>

<hr>
<h2 id='rba_uniprot_epitope_search'>Search UniProt Epitopes</h2><span id='topic+rba_uniprot_epitope_search'></span>

<h3>Description</h3>

<p>Use this function to search epitope data associated to UniProt entities,
using various criteria such as UniProt accession, epitope sequence,
IEDB ID, and match score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_epitope_search(
  accession = NULL,
  epitope_sequence = NULL,
  iedb_id = NULL,
  match_score = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_epitope_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_epitope_search_+3A_epitope_sequence">epitope_sequence</code></td>
<td>
<p>(Character) Epitope's proteins sequence</p>
</td></tr>
<tr><td><code id="rba_uniprot_epitope_search_+3A_iedb_id">iedb_id</code></td>
<td>
<p>(Numeric) <a href="https://www.iedb.org/">EIDB</a> epitope
Identifier(s). You can supply up to 20 accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_epitope_search_+3A_match_score">match_score</code></td>
<td>
<p>Integer: Minimum alignment score for the antigen sequence
and the target protein sequence.</p>
</td></tr>
<tr><td><code id="rba_uniprot_epitope_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that entity.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/epitope&quot;
</p>


<h3>See Also</h3>

<p>Other &quot;UniProt - Epitopes&quot;: 
<code><a href="#topic+rba_uniprot_epitope">rba_uniprot_epitope</a>()</code>,
<code><a href="#topic+rba_uniprot_rna_edit">rba_uniprot_rna_edit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_epitope_search(accession = c("Q84ZX5", "P36222"))


  rba_uniprot_epitope_search(epitope_sequence = "DKKCIEWEKAQHGA")


  rba_uniprot_epitope_search(iedb_id = 20354)


</code></pre>

<hr>
<h2 id='rba_uniprot_features'>Get UniProt protein sequence features by accession</h2><span id='topic+rba_uniprot_features'></span>

<h3>Description</h3>

<p>Use this function to retrieve
<a href="https://www.uniprot.org/help/sequence_annotation">sequence annotations
(features)</a> of a protein by it's UniProt accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_features(
  accession,
  types = NULL,
  categories = NULL,
  location = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_features_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_+3A_types">types</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> types. accepted values
are: &quot;INIT_MET&quot;, &quot;SIGNAL&quot;, &quot;PROPEP&quot;, &quot;TRANSIT&quot;, &quot;CHAIN&quot;, &quot;PEPTIDE&quot;,
&quot;TOPO_DOM&quot;, &quot;TRANSMEM&quot;, &quot;DOMAIN&quot;, &quot;REPEAT&quot;, &quot;CA_BIND&quot;, &quot;ZN_FING&quot;,
&quot;DNA_BIND&quot;, &quot;NP_BIND&quot;, &quot;REGION&quot;, &quot;COILED&quot;, &quot;MOTIF&quot;, &quot;COMPBIAS&quot;,
&quot;ACT_SITE&quot;, &quot;METAL&quot;, &quot;BINDING&quot;, &quot;SITE&quot;, &quot;NON_STD&quot;, &quot;MOD_RES&quot;, &quot;LIPID&quot;,
&quot;CARBOHYD&quot;, &quot;DISULFID&quot;, &quot;CROSSLNK&quot;, &quot;VAR_SEQ&quot;, &quot;VARIANT&quot;, &quot;MUTAGEN&quot;,
&quot;UNSURE&quot;, &quot;CONFLICT&quot;, &quot;NON_CONS&quot;, &quot;NON_TER&quot;, &quot;HELIX&quot;, &quot;TURN&quot;, &quot;STRAND&quot;
and/or &quot;INTRAMEM&quot;. You can supply up to 20 types.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_+3A_categories">categories</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> categories (subsection). accepted values
are: &quot;MOLECULE_PROCESSING&quot;, &quot;TOPOLOGY&quot;, &quot;SEQUENCE_INFORMATION&quot;,
&quot;STRUCTURAL&quot;, &quot;DOMAINS_AND_SITES&quot;, &quot;PTM&quot;, &quot;VARIANTS&quot; and/or &quot;MUTAGENESIS&quot;.
You can supply up to 8 categories.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_+3A_location">location</code></td>
<td>
<p>(character) Filter the features by the amino acid position in the sequence(s).
Provide the range as a character string with the format &quot;begin-end&quot;, e.g. &quot;35-70&quot;</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list in which you can find all of your given protein's sequence
annotations in a sub-list named &quot;features&quot;.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/features/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Features&quot;: 
<code><a href="#topic+rba_uniprot_features_search">rba_uniprot_features_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_features("Q99616")


rba_uniprot_features(accession = "Q99616", types = "DISULFID")


</code></pre>

<hr>
<h2 id='rba_uniprot_features_search'>Search UniProt protein sequence features</h2><span id='topic+rba_uniprot_features_search'></span>

<h3>Description</h3>

<p>UniProt maintains <a href="https://www.uniprot.org/help/sequence_annotation">
sequence annotations (features)</a> that describe regions
in the protein sequence. Using this function, you can search and
retrieve UniProt proteins' sequence annotations (features). you may also
refine your search query with variety of modifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_features_search(
  accession = NULL,
  gene = NULL,
  exact_gene = NULL,
  protein = NULL,
  reviewed = NULL,
  organism = NULL,
  taxid = NULL,
  categories = NULL,
  types = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_features_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_gene">gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names. e.g. if you supply
&quot;CD40&quot;, &quot;CD40 ligand&quot; will also be included.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_exact_gene">exact_gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt
exact gene name(s)</a>. You can supply up to 20 exact gene names. e.g.
if you supply &quot;CD40&quot;, &quot;CD40 ligand&quot; will not be included in the results.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_protein">protein</code></td>
<td>
<p><a href="https://www.uniprot.org/help/protein_names">UniProt
protein name</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_reviewed">reviewed</code></td>
<td>
<p>Logical: If TRUE, only return
&quot;UniProtKB/Swiss-Prot&quot; (reviewed) entries; If FALSE, only return TrEMBL
(un-reviewed) entries.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_organism">organism</code></td>
<td>
<p><a href="https://www.uniprot.org/taxonomy/">Organism name</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_categories">categories</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> categories (subsection). accepted values
are: &quot;MOLECULE_PROCESSING&quot;, &quot;TOPOLOGY&quot;, &quot;SEQUENCE_INFORMATION&quot;,
&quot;STRUCTURAL&quot;, &quot;DOMAINS_AND_SITES&quot;, &quot;PTM&quot;, &quot;VARIANTS&quot; and/or &quot;MUTAGENESIS&quot;.
You can supply up to 8 categories.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_types">types</code></td>
<td>
<p><a href="https://www.uniprot.org/help/sequence_annotation">
Sequence annotation (Features)</a> types. accepted values
are: &quot;INIT_MET&quot;, &quot;SIGNAL&quot;, &quot;PROPEP&quot;, &quot;TRANSIT&quot;, &quot;CHAIN&quot;, &quot;PEPTIDE&quot;,
&quot;TOPO_DOM&quot;, &quot;TRANSMEM&quot;, &quot;DOMAIN&quot;, &quot;REPEAT&quot;, &quot;CA_BIND&quot;, &quot;ZN_FING&quot;,
&quot;DNA_BIND&quot;, &quot;NP_BIND&quot;, &quot;REGION&quot;, &quot;COILED&quot;, &quot;MOTIF&quot;, &quot;COMPBIAS&quot;,
&quot;ACT_SITE&quot;, &quot;METAL&quot;, &quot;BINDING&quot;, &quot;SITE&quot;, &quot;NON_STD&quot;, &quot;MOD_RES&quot;, &quot;LIPID&quot;,
&quot;CARBOHYD&quot;, &quot;DISULFID&quot;, &quot;CROSSLNK&quot;, &quot;VAR_SEQ&quot;, &quot;VARIANT&quot;, &quot;MUTAGEN&quot;,
&quot;UNSURE&quot;, &quot;CONFLICT&quot;, &quot;NON_CONS&quot;, &quot;NON_TER&quot;, &quot;HELIX&quot;, &quot;TURN&quot;, &quot;STRAND&quot;
and/or &quot;INTRAMEM&quot;. You can supply up to 20 types.</p>
</td></tr>
<tr><td><code id="rba_uniprot_features_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br /> UniProt Entries are grouped in two sections:</p>

<ol>
<li><p> Reviewed(Swiss-Prot): Manually annotated records with information
extracted from literature and curator-evaluated computational analysis.
</p>
</li>
<li><p> Unreviewed (TrEMBL): Computationally analyzed records that await
full manual annotation.</p>
</li></ol>



<h3>Value</h3>

<p>List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that entity.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/features&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Features&quot;: 
<code><a href="#topic+rba_uniprot_features">rba_uniprot_features</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_features_search(accession = "Q99616")


rba_uniprot_features_search(gene = "cd40")


rba_uniprot_features_search(gene = "cd40 ligand")


rba_uniprot_features_search(gene = "cd40",  reviewed = TRUE)


rba_uniprot_features_search(accession = "Q99616",
    categories = c("MOLECULE_PROCESSING", "TOPOLOGY"))


rba_uniprot_features_search(accession = "Q99616", types = "DISULFID")


</code></pre>

<hr>
<h2 id='rba_uniprot_genecentric'>Get Gene-Centric proteins by UniProt Accession</h2><span id='topic+rba_uniprot_genecentric'></span>

<h3>Description</h3>

<p>Using this function you can retrieve gene-centrics data. For more
information, see <a href="https://www.uniprot.org/help/proteome">What are
proteomes?</a> and
<a href="https://www.uniprot.org/help/gene_centric_isoform_mapping">Automatic
gene-centric isoform mapping for eukaryotic reference proteome entries.</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_genecentric(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_genecentric_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_genecentric_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing information of Gene-Centric proteins.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/genecentric/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomes&quot;: 
<code><a href="#topic+rba_uniprot_genecentric_search">rba_uniprot_genecentric_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomes">rba_uniprot_proteomes</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomes_search">rba_uniprot_proteomes_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_genecentric("P29965")


</code></pre>

<hr>
<h2 id='rba_uniprot_genecentric_search'>Search Gene-Centric Proteins</h2><span id='topic+rba_uniprot_genecentric_search'></span>

<h3>Description</h3>

<p>Using this function you can search UniProt for available gene-centrics from
proteomes. For more information,
see <a href="https://www.uniprot.org/help/proteome">What are proteomes?</a> and
<a href="https://www.uniprot.org/help/gene_centric_isoform_mapping">Automatic
gene-centric isoform mapping for eukaryotic reference proteome entries.</a>
You may also refine your search with modifiers upid, accession and gene.
See &quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_genecentric_search(upid = NULL, accession = NULL, gene = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_genecentric_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_genecentric_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_genecentric_search_+3A_gene">gene</code></td>
<td>
<p>unique gene identifier(s) found in MOD,
<a href="https://www.ensembl.org/info/genome/genebuild/gene_names.html">Ensembl</a>,
Ensembl Genomes, <a href="https://www.uniprot.org/help/gene_name">OLN</a>,
<a href="https://www.uniprot.org/help/gene_name">ORF</a> or
<a href="https://www.uniprot.org/help/gene_name">UniProt Gene Name</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_genecentric_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>a list containing gene-centric proteins search hits.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/genecentric&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomes&quot;: 
<code><a href="#topic+rba_uniprot_genecentric">rba_uniprot_genecentric</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomes">rba_uniprot_proteomes</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomes_search">rba_uniprot_proteomes_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_genecentric_search(accession = "P59594")


rba_uniprot_genecentric_search(gene = "Spike")


rba_uniprot_genecentric_search(upid = "UP000000354")


</code></pre>

<hr>
<h2 id='rba_uniprot_mutagenesis'>Get Mutagenesis by UniProt Accession</h2><span id='topic+rba_uniprot_mutagenesis'></span>

<h3>Description</h3>

<p>UniProt describes the effects of mutations in proteins' amino acid
sequence on the biological properties of the protein, cell or the
organism. Using this function, you can get the
<a href="https://www.uniprot.org/help/mutagen">
Mutagenesis description</a> that has been mapped to a given UniProt protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_mutagenesis(accession, location = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_mutagenesis_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s).</p>
</td></tr>
<tr><td><code id="rba_uniprot_mutagenesis_+3A_location">location</code></td>
<td>
<p>A valid amino acid range (e.g. 10-25) within the sequence
range of the given proein.</p>
</td></tr>
<tr><td><code id="rba_uniprot_mutagenesis_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the mutagenesis description of your supplied
UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/mutagenesis/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Mutagenesis&quot;: 
<code><a href="#topic+rba_uniprot_mutagenesis_search">rba_uniprot_mutagenesis_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_mutagenesis(accession = "P0DTC2", location = "300-400")


</code></pre>

<hr>
<h2 id='rba_uniprot_mutagenesis_search'>Search Mutagenesis in UniProt</h2><span id='topic+rba_uniprot_mutagenesis_search'></span>

<h3>Description</h3>

<p>UniProt describes the effects of mutations in proteins' amino acid
sequence on the biological properties of the protein, cell or the
organism. Using this function, you can search for
<a href="https://www.uniprot.org/help/mutagen">
mutagenesis description</a> in UniProt proteins.
You may also refine your search. See &quot;Arguments section&quot; for more
information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_mutagenesis_search(
  accession = NULL,
  taxid = NULL,
  db_id = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_mutagenesis_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_mutagenesis_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_mutagenesis_search_+3A_db_id">db_id</code></td>
<td>
<p>The ID in a Cross-reference (external) database.
You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_mutagenesis_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein (search
hit) and mutagenesis description are organized under the
&quot;features&quot; sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/mutagenesis&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Mutagenesis&quot;: 
<code><a href="#topic+rba_uniprot_mutagenesis">rba_uniprot_mutagenesis</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#search all mutations in COVID19 proteins
rba_uniprot_mutagenesis_search(taxid = 2697049)


</code></pre>

<hr>
<h2 id='rba_uniprot_proteins'>Get UniProt entry by accession</h2><span id='topic+rba_uniprot_proteins'></span>

<h3>Description</h3>

<p>Use this function to retrieve a UniProt Entry by it's UniProt accession.
You can also use &quot;isoform&quot; or &quot;interaction&quot; arguments to retrieve
isoforms or interactor proteins of that entry. Note that in one function
call you can only set none or only one of &quot;isoform&quot; or &quot;interaction&quot; as
TRUE, not both of them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteins(accession, interaction = FALSE, isoforms = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteins_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_+3A_interaction">interaction</code></td>
<td>
<p>Logical: (default = FALSE) Only retrieve
<a href="https://www.uniprot.org/help/interaction_section">interaction</a>
information of your supplied UniProt entity?</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_+3A_isoforms">isoforms</code></td>
<td>
<p>Logical: (default = FALSE) Only retrieve
<a href="https://www.uniprot.org/help/alternative_products">isoforms</a> of your
supplied UniProt entity?</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains UniProt protein informations with your
supplied accession.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/proteins/{accession}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/proteins/interaction/{accession}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/proteins/{accession}/isoforms&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteins&quot;: 
<code><a href="#topic+rba_uniprot_proteins_crossref">rba_uniprot_proteins_crossref</a>()</code>,
<code><a href="#topic+rba_uniprot_proteins_search">rba_uniprot_proteins_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_proteins(accession = "P01730")


rba_uniprot_proteins(accession = "P01730", interaction = TRUE)


rba_uniprot_proteins(accession = "Q29983", isoforms = TRUE)


</code></pre>

<hr>
<h2 id='rba_uniprot_proteins_crossref'>Get UniProt Entry by UniProt Cross-Reference Database and ID</h2><span id='topic+rba_uniprot_proteins_crossref'></span>

<h3>Description</h3>

<p><a href="https://www.uniprot.org/database/">UniProt Cross-Reference</a> links
protein Entities with cross-reference (external) databases. Using this
function, you can retrieve a UniProt entity using external database name
and protein ID in that database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteins_crossref(
  db_id,
  db_name,
  reviewed = NULL,
  isoform = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteins_crossref_+3A_db_id">db_id</code></td>
<td>
<p>The protein ID in the cross-reference (external) database.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_crossref_+3A_db_name">db_name</code></td>
<td>
<p><a href="https://www.uniprot.org/database/">cross-reference</a>
(external database) name.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_crossref_+3A_reviewed">reviewed</code></td>
<td>
<p>Logical: (Optional) If TRUE, only returns
&quot;UniProtKB/Swiss-Prot&quot; (reviewed) entries; If FALSE, only returns TrEMBL
(un-reviewed) entries.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_crossref_+3A_isoform">isoform</code></td>
<td>
<p>Numeric: (Optional) you have two options:</p>

<ul>
<li><p> 0: Exclude isoforms.
</p>
</li>
<li><p> 1: Return isoforms only.</p>
</li></ul>

<p>see: <a href="https://www.uniprot.org/help/alternative_products">Alternative
products</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_crossref_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List which each element is a UniProt entity that correspond to
your supplied cross-reference database name and ID.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteins/{dbtype}:{dbid}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteins&quot;: 
<code><a href="#topic+rba_uniprot_proteins">rba_uniprot_proteins</a>()</code>,
<code><a href="#topic+rba_uniprot_proteins_search">rba_uniprot_proteins_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_proteins_crossref("cd40", "hgnc")


rba_uniprot_proteins_crossref("cd40", "hgnc", reviewed = TRUE)


rba_uniprot_proteins_crossref("mica", "hgnc", isoform = 0)


</code></pre>

<hr>
<h2 id='rba_uniprot_proteins_search'>Search UniProt entries</h2><span id='topic+rba_uniprot_proteins_search'></span>

<h3>Description</h3>

<p>Using this function, you can search and retrieve UniProt Knowledge-base
(UniProtKB) protein entries using variety of options. You may also
refine your search with modifiers such as sequence length, review status
etc. See &quot;Arguments&quot; section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteins_search(
  accession = NULL,
  reviewed = NULL,
  isoform = NULL,
  go_term = NULL,
  keyword = NULL,
  ec = NULL,
  gene = NULL,
  exact_gene = NULL,
  protein = NULL,
  organism = NULL,
  taxid = NULL,
  pubmed = NULL,
  seq_length = NULL,
  md5 = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteins_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_reviewed">reviewed</code></td>
<td>
<p>Logical: If TRUE, only return
&quot;UniProtKB/Swiss-Prot&quot; (reviewed) entries; If FALSE, only return TrEMBL
(un-reviewed) entries.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_isoform">isoform</code></td>
<td>
<p>Numeric: you have three options:</p>

<ul>
<li><p> 0: Exclude isoforms.
</p>
</li>
<li><p> 1: Return isoforms only.
</p>
</li>
<li><p> 2: Return both.</p>
</li></ul>

<p>see: <a href="https://www.uniprot.org/help/alternative_products">Alternative
products</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_go_term">go_term</code></td>
<td>
<p>Limit the search to entries associated with your supplied GO
(<a href="https://www.uniprot.org/help/gene_ontology">Gene Ontology</a>) term.
You can supply Either GO ID or a character string -partially or fully-
matching the term. e.g. &quot;GO:0001776&quot; or &quot;leukocyte homeostasis&quot;. if You
supply &quot;leukocyte&quot;, any term containing that word will be included,
e.g &quot;leukocyte chemotaxis&quot;, &quot;leukocyte activation&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_keyword">keyword</code></td>
<td>
<p>Limit the search to entries that contain your supplied
keyword. see: <a href="https://www.uniprot.org/keywords/">UniProt Keywords</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_ec">ec</code></td>
<td>
<p><a href="https://enzyme.expasy.org/">EC (Enzyme Commission) number(s)</a>.
You can supply up to 20 EC numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_gene">gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names. e.g. if you supply
&quot;CD40&quot;, &quot;CD40 ligand&quot; will also be included.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_exact_gene">exact_gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt
exact gene name(s)</a>. You can supply up to 20 exact gene names. e.g.
if you supply &quot;CD40&quot;, &quot;CD40 ligand&quot; will not be included in the results.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_protein">protein</code></td>
<td>
<p><a href="https://www.uniprot.org/help/protein_names">UniProt
protein name</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_organism">organism</code></td>
<td>
<p><a href="https://www.uniprot.org/taxonomy/">Organism name</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_pubmed">pubmed</code></td>
<td>
<p>Entries which <a href="https://www.uniprot.org/citations/">cite
to</a> the article with your supplied PubMed ID.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_seq_length">seq_length</code></td>
<td>
<p>An exact sequence length (e.g. 150) or a range of sequence
lengths (e.g. &quot;130-158&quot;).</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_md5">md5</code></td>
<td>
<p>Sequence md5 value.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteins_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.s
<br /> UniProt Entries are grouped in two sections:</p>

<ol>
<li><p> Reviewed(Swiss-Prot): Manually annotated records with information
extracted from literature and curator-evaluated computational analysis.
</p>
</li>
<li><p> Unreviewed (TrEMBL): Computationally analyzed records that await
full manual annotation.</p>
</li></ol>



<h3>Value</h3>

<p>A List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that entity.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteins&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteins&quot;: 
<code><a href="#topic+rba_uniprot_proteins">rba_uniprot_proteins</a>()</code>,
<code><a href="#topic+rba_uniprot_proteins_crossref">rba_uniprot_proteins_crossref</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_proteins_search(accession = "Q99616")


rba_uniprot_proteins_search(gene = "cd40")


rba_uniprot_proteins_search(gene = "cd40 ligand")


rba_uniprot_proteins_search(gene = "cd40",  reviewed = TRUE)


rba_uniprot_proteins_search(gene = "cd40",  reviewed = TRUE, isoform = 1)


rba_uniprot_proteins_search(keyword = "Inhibition of host chemokines by virus")


rba_uniprot_proteins_search(keyword = "chemokines")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomes'>Get proteome by proteome/proteins UPID</h2><span id='topic+rba_uniprot_proteomes'></span>

<h3>Description</h3>

<p>UniProt collects and annotates proteomes(Protein sets expressed in an
organism). Using this function you can search UniProt for available
proteomes. see <a href="https://www.uniprot.org/help/proteome">What are
proteomes?</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomes(upid, get_proteins = FALSE, reviewed = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomes_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_+3A_get_proteins">get_proteins</code></td>
<td>
<p>logical: set FALSE (default) to only return information
of the proteome with supplied UPID, set TRUE to also return the proteins
of the supplied proteome UPID.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_+3A_reviewed">reviewed</code></td>
<td>
<p>Logical:  Only considered when get_proteins is TRUE.
If TRUE, only return &quot;UniProtKB/Swiss-Prot&quot; (reviewed) proteins;
If FALSE, only return TrEMBL (un-reviewed) entries. leave it as NULL if you
do not want to filter proteins based on their review status.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing information of the proteome with your supplied
UPID that can contain the proteomes protein entries based on the value of
get_proteins argument.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/proteomes/proteins/{upid}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/proteomes/{upid}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomes&quot;: 
<code><a href="#topic+rba_uniprot_genecentric">rba_uniprot_genecentric</a>()</code>,
<code><a href="#topic+rba_uniprot_genecentric_search">rba_uniprot_genecentric_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomes_search">rba_uniprot_proteomes_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_proteomes(upid = "UP000000354")


rba_uniprot_proteomes(upid = "UP000000354", get_proteins = TRUE)


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomes_search'>Search Proteomes in UniProt</h2><span id='topic+rba_uniprot_proteomes_search'></span>

<h3>Description</h3>

<p>UniProt collects and annotates proteomes (Protein sets expressed in an
organism). Using this function you can search UniProt for available
proteomes. see <a href="https://www.uniprot.org/help/proteome">What are
proteomes?</a> for more information. You may also
refine your search with modifiers such as keyword, taxon id etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomes_search(
  name = NULL,
  upid = NULL,
  taxid = NULL,
  keyword = NULL,
  xref = NULL,
  genome_acc = NULL,
  is_ref_proteome = NULL,
  is_redundant = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomes_search_+3A_name">name</code></td>
<td>
<p>a keyword in proteome's name</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_keyword">keyword</code></td>
<td>
<p>Limit the search to entries that contain your supplied
keyword. see: <a href="https://www.uniprot.org/keywords/">UniProt Keywords</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_xref">xref</code></td>
<td>
<p>Proteome cross-references such as Genome assembly ID or
Biosample ID. You can supply up to 20 cross-reference IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_genome_acc">genome_acc</code></td>
<td>
<p>Genome accession associated with the proteome's components.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_is_ref_proteome">is_ref_proteome</code></td>
<td>
<p>(logical) If TRUE, only return reference proteomes; If
FALSE, only returns non-reference proteomes; If NULL (default), the results
will not be filtered by this criteria see
<a href="https://www.uniprot.org/help/reference_proteome">'What are reference
proteomes?'</a> for more information.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_is_redundant">is_redundant</code></td>
<td>
<p>(logical) If TRUE, only return redundant proteomes; If
FALSE, only returns non-redundant proteomes; If NULL (default), the results
will not be filtered by redundancy. see
<a href="https://www.uniprot.org/help/proteome_redundancy">'Reducing proteome
redundancy'</a> for more information.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomes_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A list where each element is a list that corresponds to a single
proteome (search hit) and contains informations pertinent to that proteome.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/proteomes&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomes&quot;: 
<code><a href="#topic+rba_uniprot_genecentric">rba_uniprot_genecentric</a>()</code>,
<code><a href="#topic+rba_uniprot_genecentric_search">rba_uniprot_genecentric_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomes">rba_uniprot_proteomes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_proteomes_search(name = "SARS-CoV")


rba_uniprot_proteomes_search(name = "SARS-CoV", is_ref_proteome = TRUE)


rba_uniprot_proteomes_search(name = "SARS-CoV", is_ref_proteome = TRUE)


rba_uniprot_proteomes_search(genome_acc = "AY274119")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics'>Get Proteomics Peptides Mapped to UniProt Protein (Deprecated)</h2><span id='topic+rba_uniprot_proteomics'></span>

<h3>Description</h3>

<p>This function is Deprecated. Please use
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a></code> instead.<br />
UniProt maps proteomics peptides from different sources to the proteins'
sequences. Using this function, you can retrieve all the proteomics
peptides features that has been map to a given UniProt protein's sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the proteomics peptides features of your supplied
UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Deprecated functions&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_search">rba_uniprot_proteomics_search</a>()</code>,
<code><a href="#topic+rba_uniprot_ptm">rba_uniprot_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_ptm_search">rba_uniprot_ptm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Deprecated
rba_uniprot_proteomics(accession = "P25942")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_hpp'>Get HPP Proteomics data in UniProt</h2><span id='topic+rba_uniprot_proteomics_hpp'></span>

<h3>Description</h3>

<p>UniProt maps post-translational modification proteomics data from different
sources to the proteins' sequences. Using this function, you can retrieve
all the HPP (Human Proteome Project) proteomics features that has been map
to a given UniProt protein's sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_hpp(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_hpp_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A list containing the proteomics data features of
your supplied UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics/nonPtm/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp_search">rba_uniprot_proteomics_hpp_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_species">rba_uniprot_proteomics_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_proteomics_hpp(accession = "P04234")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_hpp_search'>Search HPP Proteomics data in UniProt</h2><span id='topic+rba_uniprot_proteomics_hpp_search'></span>

<h3>Description</h3>

<p>Using this function, you can search for HPP (Human Proteome Project)
proteomics features that has been map to UniProt proteins. You may also
refine your search with modifiers such as data_source, peptide etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_hpp_search(
  accession = NULL,
  taxid = NULL,
  data_source = NULL,
  upid = NULL,
  peptide = NULL,
  unique = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You
can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_data_source">data_source</code></td>
<td>
<p>Proteomics data source. In addition to manual curation,
UniProt also import PTM annotations from the following databases:
</p>

<ul>
<li> <p><a href="https://hupo.org/human-proteome-project/">&quot;HPP&quot;</a>
</p>
</li></ul>

<p>Please use 'rba_uniprot_proteomics_species()' for more information on
the available data sources for a given species.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_peptide">peptide</code></td>
<td>
<p>Peptide sequence(s). You can supply up to 20 sequences.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_unique">unique</code></td>
<td>
<p>Logical: Should the results be filtered based on the Peptide's
uniqueness (the fact that a peptide maps to only 1 protein). If TRUE, Only
unique peptides will be returned, if FALSE only un-unique peptides will be
returned; If NULL (default) the results will not be filtered based on this.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_hpp_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s arguments
manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill every
argument; You may use whatever combinations of arguments you see fit for your
query.
</p>
<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein and
proteomics data are organized under the &quot;features&quot;
sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET
https://www.ebi.ac.uk/proteins/api/proteomics/ptm&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas Bursteinas,
Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd Turner, Maria
Martin, The Proteins API: accessing key integrated protein and genome
information, Nucleic Acids Research, Volume 45, Issue W1, 3 July 2017,
Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp">rba_uniprot_proteomics_hpp</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_species">rba_uniprot_proteomics_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_proteomics_hpp_search(peptide = "NDQVYQPLRDRDDAQYSHLGGNWAR")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_non_ptm'>Get Proteomics data in UniProt</h2><span id='topic+rba_uniprot_proteomics_non_ptm'></span>

<h3>Description</h3>

<p>UniProt maps proteomics data from different sources to the proteins'
sequences. Using this function, you can retrieve all the
non-post-translational modification proteomics features that has been map
to a given UniProt protein's sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_non_ptm(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_non_ptm_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A list containing the proteomics data features of
your supplied UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics/nonPtm/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp">rba_uniprot_proteomics_hpp</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_hpp_search">rba_uniprot_proteomics_hpp_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_species">rba_uniprot_proteomics_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_proteomics_non_ptm(accession = "P04234")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_non_ptm_search'>Search Proteomics data in UniProt</h2><span id='topic+rba_uniprot_proteomics_non_ptm_search'></span>

<h3>Description</h3>

<p>Using this function, you can search for non-Post-Translational Modification
proteomics features that has been map to UniProt proteins. You may also
refine your search with modifiers such as data_source, peptide etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_non_ptm_search(
  accession = NULL,
  taxid = NULL,
  data_source = NULL,
  upid = NULL,
  peptide = NULL,
  unique = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You
can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_data_source">data_source</code></td>
<td>
<p>Proteomics data source. In addition to manual curation,
UniProt also import PTM annotations from the following databases:
</p>

<ul>
<li><p> &quot;MaxQB&quot;
</p>
</li>
<li> <p><a href="https://peptideatlas.org/">&quot;PeptideAtlas&quot;</a>
</p>
</li>
<li><p> &quot;EPD&quot;
</p>
</li>
<li> <p><a href="https://www.proteomicsdb.org/">&quot;ProteomicsDB&quot;</a>
</p>
</li></ul>

<p>Please use 'rba_uniprot_proteomics_species()' for more information on
the available data sources for a given species.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_peptide">peptide</code></td>
<td>
<p>Peptide sequence(s). You can supply up to 20 sequences.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_unique">unique</code></td>
<td>
<p>Logical: Should the results be filtered based on the Peptide's
uniqueness (the fact that a peptide maps to only 1 protein). If TRUE, Only
unique peptides will be returned, if FALSE only un-unique peptides will be
returned; If NULL (default) the results will not be filtered based on this.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_non_ptm_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s arguments
manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill every
argument; You may use whatever combinations of arguments you see fit for your
query.
</p>
<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein and
proteomics data are organized under the &quot;features&quot;
sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET
https://www.ebi.ac.uk/proteins/api/proteomics/nonPtm&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas Bursteinas,
Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd Turner, Maria
Martin, The Proteins API: accessing key integrated protein and genome
information, Nucleic Acids Research, Volume 45, Issue W1, 3 July 2017,
Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp">rba_uniprot_proteomics_hpp</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_hpp_search">rba_uniprot_proteomics_hpp_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_species">rba_uniprot_proteomics_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_proteomics_non_ptm_search(peptide = "NDQVYQPLRDRDDAQYSHLGGNWAR")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_ptm'>Get Post-Translational Modification of UniProt Protein</h2><span id='topic+rba_uniprot_proteomics_ptm'></span>

<h3>Description</h3>

<p>UniProt maps post-translational modification proteomics data from different
sources to the proteins' sequences. Using this function, you can retrieve
all the post-translational modification features that has been map to a
given UniProt protein's sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_ptm(accession, confidence_score = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_ptm_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_+3A_confidence_score">confidence_score</code></td>
<td>
<p>(Character) Valid values: &quot;Bronze&quot;, &quot;Silver&quot;, or
&quot;gold&quot;.<br /> UniProt classifies modified residues
into three categories based on its false localization rate (FLR)
across multiple dataset. See
<a href="https://www.uniprot.org/help/mod_res_large_scale">Large scale modified residue</a>
for more information.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>see also:
<a href="https://www.uniprot.org/help/ptm_processing_section">PTM /
Processing section in UniProtKB</a>
</p>
<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A list containing the post-translational modification features of
your supplied UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics/ptm/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp">rba_uniprot_proteomics_hpp</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_hpp_search">rba_uniprot_proteomics_hpp_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_species">rba_uniprot_proteomics_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_proteomics_ptm(accession = "P04234")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_ptm_search'>Search Post-Translational Modification Proteomics in UniProt</h2><span id='topic+rba_uniprot_proteomics_ptm_search'></span>

<h3>Description</h3>

<p>Using this function, you can search for Post-Translational Modification
proteomics features that has been map to UniProt proteins. You may also
refine your search with modifiers such as data_source, peptide etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_ptm_search(
  accession = NULL,
  ptm = NULL,
  taxid = NULL,
  data_source = NULL,
  upid = NULL,
  peptide = NULL,
  unique = NULL,
  confidence_score = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_ptm">ptm</code></td>
<td>
<p>Post-translational modification name.
<a href="https://www.uniprot.org/keywords/KW-9991">Valid values</a> are:
&quot;Acetylation &quot;, &quot;ADP-ribosylation &quot;, &quot;Amidation &quot;, &quot;Autocatalytic cleavage
&quot;, &quot;Bromination &quot;, &quot;Citrullination &quot;, &quot;Cleavage on pair of basic residues
&quot;, &quot;Covalent protein-DNA linkage &quot;, &quot;Covalent protein-RNA linkage &quot;, &quot;CTQ
&quot;, &quot;D-amino acid &quot;, &quot;Disulfide bond &quot;, &quot;Formylation &quot;,
&quot;Gamma-carboxyglutamic acid &quot;, &quot;Glutathionylation &quot;, &quot;Glycoprotein &quot;,
&quot;Lipoprotein &quot;, &quot;Hydroxylation &quot;, &quot;Hypusine &quot;, &quot;Iodination &quot;, &quot;Isopeptide
bond &quot;, &quot;LTQ &quot;, &quot;Methylation &quot;, &quot;Nitration &quot;, &quot;Organic radical &quot;,
&quot;Oxidation &quot;, &quot;Peptidoglycan-anchor &quot;, &quot;Phosphopantetheine &quot;,
&quot;Phosphoprotein &quot;, &quot;Pyrrolidone carboxylic acid &quot;, &quot;Quinone &quot;,
&quot;S-nitrosylation &quot;, &quot;Sulfation &quot;, &quot;Thioester bond &quot;, &quot;Thioether bond &quot;,
&quot;TPQ &quot;, &quot;TTQ &quot;, &quot;Ubl conjugation &quot;, or &quot;Zymogen&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You
can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_data_source">data_source</code></td>
<td>
<p>Proteomics data source. In addition to manual curation,
UniProt also import PTM annotations from the following databases:
</p>

<ul>
<li> <p><a href="https://www.ebi.ac.uk/pride/">&quot;PRIDE&quot;</a>
</p>
</li>
<li> <p><a href="https://www.proteomexchange.org/ptmexchange">&quot;PTMeXchange&quot;</a>
</p>
</li></ul>

<p>Please use 'rba_uniprot_proteomics_species()' for more information on
the available data sources for a given species.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_peptide">peptide</code></td>
<td>
<p>Peptide sequence(s). You can supply up to 20 sequences.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_unique">unique</code></td>
<td>
<p>Logical: Should the results be filtered based on the Peptide's
uniqueness (the fact that a peptide maps to only 1 protein). If TRUE, Only
unique peptides will be returned, if FALSE only un-unique peptides will be
returned; If NULL (default) the results will not be filtered based on this.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_confidence_score">confidence_score</code></td>
<td>
<p>(Character) Valid values: &quot;Bronze&quot;, &quot;Silver&quot;, or
&quot;gold&quot;.<br /> UniProt classifies modified residues
into three categories based on its false localization rate (FLR)
across multiple dataset. See
<a href="https://www.uniprot.org/help/mod_res_large_scale">Large scale modified residue</a>
for more information.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_ptm_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s arguments
manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill every
argument; You may use whatever combinations of arguments you see fit for your
query.
</p>
<p>see also:
<a href="https://www.uniprot.org/help/post-translational_modification">PTM /
Processing section in UniProtKB</a>
</p>
<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein and
post-translational modification are organized under the &quot;features&quot;
sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET
https://www.ebi.ac.uk/proteins/api//proteomics/ptm&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas Bursteinas,
Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd Turner, Maria
Martin, The Proteins API: accessing key integrated protein and genome
information, Nucleic Acids Research, Volume 45, Issue W1, 3 July 2017,
Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp">rba_uniprot_proteomics_hpp</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_hpp_search">rba_uniprot_proteomics_hpp_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_species">rba_uniprot_proteomics_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_proteomics_ptm_search(peptide = "NDQVYQPLRDRDDAQYSHLGGNWAR")


</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_search'>Search Proteomics Peptides in UniProt (Deprecated)</h2><span id='topic+rba_uniprot_proteomics_search'></span>

<h3>Description</h3>

<p>This function is Deprecated. Please use
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a></code> instead.<br />
UniProt maps proteomics peptides from different sources to the proteins'
sequences. Using this function, you can search for  proteomics
peptides that has been map to UniProt proteins. You may also refine your
search with modifiers such as data_source, peptide etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_search(
  accession = NULL,
  data_source = NULL,
  taxid = NULL,
  upid = NULL,
  peptide = NULL,
  unique = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_search_+3A_data_source">data_source</code></td>
<td>
<p>Proteomics data source. You can choose up to two of:
</p>

<ul>
<li> <p><a href="https://www.uniprot.org/database/DB-0186">&quot;MaxQB&quot;</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/database/DB-0071">&quot;PeptideAtlas&quot;</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/database/DB-0205">&quot;EPD&quot;</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/database/DB-0229">&quot;ProteomicsDB&quot;</a>
</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_search_+3A_peptide">peptide</code></td>
<td>
<p>Peptide sequence(s). You can supply up to 20 sequences.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_search_+3A_unique">unique</code></td>
<td>
<p>Logical: Should the results be filtered based on the
Peptide's uniqueness (the fact that a peptide maps to only 1 protein). If
TRUE, Only unique peptides will be returned, if FALSE only un-unique
peptides will be returned; If NULL (default) the results will not be
filtered based on this.</p>
</td></tr>
<tr><td><code id="rba_uniprot_proteomics_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br /> see also: <a href="https://www.uniprot.org/help/proteomics">Mass
spectrometry-based proteomics data in UniProtKB</a>
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein and
proteomics peptides are organized under the &quot;features&quot; sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Deprecated functions&quot;: 
<code><a href="#topic+rba_uniprot_proteomics">rba_uniprot_proteomics</a>()</code>,
<code><a href="#topic+rba_uniprot_ptm">rba_uniprot_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_ptm_search">rba_uniprot_ptm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Deprecated
rba_uniprot_proteomics_search(peptide = "MEDYTKIEK")

## End(Not run)
## Not run: 
#Deprecated
rba_uniprot_proteomics_search(peptide = "MEDYTKIEK")

## End(Not run)
## Not run: 
#Deprecated
### this will generate a very large response!
  rba_uniprot_proteomics_search(taxid = 9606,
  data_source = "PeptideAtlas",
  progress = TRUE, timeout = 999999, unique = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_uniprot_proteomics_species'>Get UniProt Proteomics Metadata</h2><span id='topic+rba_uniprot_proteomics_species'></span>

<h3>Description</h3>

<p>Retrieve information on the available Species proteomics data sources in
UniProt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_proteomics_species(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_proteomics_species_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>UniProt categorizes proteomics data sources into three main data categories:
PTM (Post-Translational Modification), non-PTM, and HPP (Human Proteome
Project); each with corresponding API endpoints, and thus, rbioapi
functions.
</p>


<h3>Value</h3>

<p>A data frame with the available species as rows, and the columns
indicating the proteomics data sources, separated by three main data
categories: PTM (Post-Translational Modification), non-PTM, and
HPP (Human Proteome Project)
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics/species&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Proteomics&quot;: 
<code><a href="#topic+rba_uniprot_proteomics_hpp">rba_uniprot_proteomics_hpp</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_hpp_search">rba_uniprot_proteomics_hpp_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm">rba_uniprot_proteomics_non_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_non_ptm_search">rba_uniprot_proteomics_non_ptm_search</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_proteomics_species()


</code></pre>

<hr>
<h2 id='rba_uniprot_ptm'>Get Post-Translational Modification of UniProt Protein (Deprecated)</h2><span id='topic+rba_uniprot_ptm'></span>

<h3>Description</h3>

<p>This function is Deprecated. Please use
<code><a href="#topic+rba_uniprot_proteomics_ptm">rba_uniprot_proteomics_ptm</a></code> instead.<br />
UniProt maps post-translational modification features from different sources
to the proteins'  sequences. Using this function, you can retrieve all
the post-translational modification features that has been map to a given
UniProt protein's sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_ptm(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_ptm_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>see also:
<a href="https://www.uniprot.org/help/ptm_processing_section">PTM /
Processing section in UniProtKB</a>
</p>


<h3>Value</h3>

<p>A list containing the post-translational modification features of
your supplied UniProt protein's sequence.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics-ptm/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Deprecated functions&quot;: 
<code><a href="#topic+rba_uniprot_proteomics">rba_uniprot_proteomics</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_search">rba_uniprot_proteomics_search</a>()</code>,
<code><a href="#topic+rba_uniprot_ptm_search">rba_uniprot_ptm_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Deprecated
rba_uniprot_ptm(accession = "P04234")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_uniprot_ptm_search'>Search Post-Translational Modification in UniProt (Deprecated)</h2><span id='topic+rba_uniprot_ptm_search'></span>

<h3>Description</h3>

<p>This function is Deprecated. Please use
<code><a href="#topic+rba_uniprot_proteomics_ptm_search">rba_uniprot_proteomics_ptm_search</a></code> instead.<br />
UniProt maps proteomics peptides from different sources to the proteins'
sequences. Using this function, you can search for  proteomics
peptides that has been map to UniProt proteins. You may also refine your
search with modifiers such as data_source, peptide etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_ptm_search(
  accession = NULL,
  ptm = NULL,
  data_source = NULL,
  taxid = NULL,
  upid = NULL,
  peptide = NULL,
  unique = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_ptm_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_ptm">ptm</code></td>
<td>
<p>Post-translational modification name</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_data_source">data_source</code></td>
<td>
<p>Proteomics data source. You can choose up to two of:
</p>

<ul>
<li> <p><a href="https://www.uniprot.org/database/DB-0186">&quot;MaxQB&quot;</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/database/DB-0071">&quot;PeptideAtlas&quot;</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/database/DB-0205">&quot;EPD&quot;</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/database/DB-0229">&quot;ProteomicsDB&quot;</a>
</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_peptide">peptide</code></td>
<td>
<p>Peptide sequence(s). You can supply up to 20 sequences.</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_unique">unique</code></td>
<td>
<p>Logical: Should the results be filtered based on the
Peptide's uniqueness (the fact that a peptide maps to only 1 protein). If
TRUE, Only unique peptides will be returned, if FALSE only un-unique
peptides will be returned; If NULL (default) the results will not be
filtered based on this.</p>
</td></tr>
<tr><td><code id="rba_uniprot_ptm_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
<br /> see also:
<a href="https://www.uniprot.org/help/ptm_processing_section">PTM /
Processing section in UniProtKB</a>
</p>


<h3>Value</h3>

<p>A list Where each element correspond to a UniProt protein and
post-translational modification are organized under the &quot;features&quot;
sub-list.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/proteomics-ptm&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium, UniProt: the universal protein
knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1,
8 January 2021, Pages D480–D489, https://doi.org/10.1093/nar/gkaa1100
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;Deprecated functions&quot;: 
<code><a href="#topic+rba_uniprot_proteomics">rba_uniprot_proteomics</a>()</code>,
<code><a href="#topic+rba_uniprot_proteomics_search">rba_uniprot_proteomics_search</a>()</code>,
<code><a href="#topic+rba_uniprot_ptm">rba_uniprot_ptm</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Deprecated
rba_uniprot_ptm_search(peptide = "NDQVYQPLRDRDDAQYSHLGGNWAR")

## End(Not run)

</code></pre>

<hr>
<h2 id='rba_uniprot_rna_edit'>Retrieve Epitope by Accession</h2><span id='topic+rba_uniprot_rna_edit'></span>

<h3>Description</h3>

<p>Use this function to retrieve
<a href="https://www.uniprot.org/help/rna_editing">RNA-editing
events</a> (conversion, insertion, deletion of nucleotides) annotations
linked to a UniProt entry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_rna_edit(accession, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_rna_edit_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_rna_edit_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the UniProt RNA-editing features details for the
given accession.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/rna-edit/{accession}&quot;
</p>


<h3>See Also</h3>

<p>Other &quot;UniProt - Epitopes&quot;: 
<code><a href="#topic+rba_uniprot_epitope">rba_uniprot_epitope</a>()</code>,
<code><a href="#topic+rba_uniprot_epitope_search">rba_uniprot_epitope_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_rna_edit(accession = "Q16851")


</code></pre>

<hr>
<h2 id='rba_uniprot_rna_edit_search'>Search RNA Editing in UniProt</h2><span id='topic+rba_uniprot_rna_edit_search'></span>

<h3>Description</h3>

<p>UniProt Curates <a href="https://www.uniprot.org/help/rna_editing">RNA-editing
events</a> (conversion, insertion, deletion of nucleotides). Use this
function to search RNA editing records in UniProt using various
criteria such as accession, taxon ID, or variant location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_rna_edit_search(
  accession = NULL,
  taxid = NULL,
  variantlocation = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_rna_edit_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_rna_edit_search_+3A_taxid">taxid</code></td>
<td>
<p>(Numeric) NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_rna_edit_search_+3A_variantlocation">variantlocation</code></td>
<td>
<p>Character: RNA editing variant location(s).
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_rna_edit_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that entity.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/rna-editing&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  rba_uniprot_rna_edit_search(accession = c("Q16851", "Q16849"))


</code></pre>

<hr>
<h2 id='rba_uniprot_taxonomy'>Get UniProt Taxonomy Nodes</h2><span id='topic+rba_uniprot_taxonomy'></span>

<h3>Description</h3>

<p>Using this function, you can retrieve taxonomic nodes information by
providing their
<a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI taxonomic
identifiers</a>. also, you can explicitly retrieve other nodes in relation
to your supplied node's hierarchy in
<a href="https://www.uniprot.org/help/taxonomy">UniProt Taxonomy database</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_taxonomy(
  ids,
  hierarchy = NULL,
  node_only = TRUE,
  page_size = 200,
  page_number = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_taxonomy_+3A_ids">ids</code></td>
<td>
<p>(numeric) a single or a numeric vector of
<a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI taxonomic
identifier(s)</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_+3A_hierarchy">hierarchy</code></td>
<td>
<p>Retrieve taxonomic nodes that have specific hierarchical
relation to your supplied taxonomic node. should be one of: &quot;children&quot;,
&quot;parent&quot; or &quot;siblings&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_+3A_node_only">node_only</code></td>
<td>
<p>Retrieve only the node(s) information and exclude URL links
to parents, siblings and children nodes.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_+3A_page_size">page_size</code></td>
<td>
<p>(numeric) Only when hierarchy is supplied. hierarchy
information may be very long, thus UniProt API will paginate the results,
you may use this argument to control the pagination. maximum value is 200.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_+3A_page_number">page_number</code></td>
<td>
<p>(numeric) Only when hierarchy is supplied. hierarchy
information may be very long, thus UniProt API will paginate the results,
you may use this argument to control the pagination.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing your supplied nodes or their related nodes
taxonomic information.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/ids/{ids}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/ids/id/{id}/node&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/node&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/children&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/children/node&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/parent&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/parent/node&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/siblings&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/id/{id}/siblings/node&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Taxonomy&quot;: 
<code><a href="#topic+rba_uniprot_taxonomy_lca">rba_uniprot_taxonomy_lca</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lineage">rba_uniprot_taxonomy_lineage</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_name">rba_uniprot_taxonomy_name</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_path">rba_uniprot_taxonomy_path</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_relationship">rba_uniprot_taxonomy_relationship</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_taxonomy(ids = c(9606, 10090))


rba_uniprot_taxonomy(ids = 9989, hierarchy = "children")


</code></pre>

<hr>
<h2 id='rba_uniprot_taxonomy_lca'>Get Lowest Common Ancestor (LCA) of Two Taxonomy Nodes</h2><span id='topic+rba_uniprot_taxonomy_lca'></span>

<h3>Description</h3>

<p>Use this function to retrieve lowest common ancestor (LCA) of two
taxonomy nodes in
<a href="https://www.uniprot.org/help/taxonomy">UniProt Taxonomy database</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_taxonomy_lca(ids, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_taxonomy_lca_+3A_ids">ids</code></td>
<td>
<p>(numeric) Numeric vector of
<a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI taxonomic
identifiers</a>, with minimum length of two.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_lca_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with UniProt taxonomy information of your supplied taxonomy
elements.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/ancestor/{ids}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Taxonomy&quot;: 
<code><a href="#topic+rba_uniprot_taxonomy">rba_uniprot_taxonomy</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lineage">rba_uniprot_taxonomy_lineage</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_name">rba_uniprot_taxonomy_name</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_path">rba_uniprot_taxonomy_path</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_relationship">rba_uniprot_taxonomy_relationship</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_taxonomy_lca(c(9606,10090,9823,7712))


</code></pre>

<hr>
<h2 id='rba_uniprot_taxonomy_lineage'>Get Taxonomic Lineage</h2><span id='topic+rba_uniprot_taxonomy_lineage'></span>

<h3>Description</h3>

<p>Use this function to retrieve the taxonomic lineage of your supplied
taxonomy node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_taxonomy_lineage(id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_taxonomy_lineage_+3A_id">id</code></td>
<td>
<p>(numeric) a
<a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI taxonomic
identifier</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_lineage_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with a data frame containing All the nodes that preceded your
supplied node in the taxonomic tree. with your node as the first row
and the root node in the last row.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/lineage/{id}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Taxonomy&quot;: 
<code><a href="#topic+rba_uniprot_taxonomy">rba_uniprot_taxonomy</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lca">rba_uniprot_taxonomy_lca</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_name">rba_uniprot_taxonomy_name</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_path">rba_uniprot_taxonomy_path</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_relationship">rba_uniprot_taxonomy_relationship</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_taxonomy_lineage(id = 9989)


</code></pre>

<hr>
<h2 id='rba_uniprot_taxonomy_name'>Search UniProt Taxonomic Names</h2><span id='topic+rba_uniprot_taxonomy_name'></span>

<h3>Description</h3>

<p>Using this function, you can search and retrieve taxonomic nodes using
their names from
<a href="https://www.uniprot.org/help/taxonomy">UniProt Taxonomy database</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_taxonomy_name(
  name,
  field = "scientific",
  search_type = "equal_to",
  node_only = TRUE,
  page_size = 200,
  page_number = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_name">name</code></td>
<td>
<p>a name to to be used as search query.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_field">field</code></td>
<td>
<p>Specify the field that your supplied name should be searched.
It should be one of : &quot;scientific&quot; (default), &quot;common&quot; or &quot;mnemonic&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_search_type">search_type</code></td>
<td>
<p>The logical relationship between your supplied search
query and the taxonomic name field. It should be one of &quot;equal_to&quot;
(default), &quot;start_with&quot;, &quot;end_with&quot; or &quot;contain&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_node_only">node_only</code></td>
<td>
<p>(logical) Retrieve only the node(s) information and exclude URL links
to parents, siblings and children nodes. default = TRUE</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_page_size">page_size</code></td>
<td>
<p>(numeric) Your search results may be very long, thus
UniProt API will paginate the results, you may use this argument to control
the pagination. maximum value is 200.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_page_number">page_number</code></td>
<td>
<p>(numeric) Your search results may be very long, thus
UniProt API will paginate the results, you may use this argument to control
the pagination. maximum value is 200.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_name_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing taxonomic nodes that match your supplied inputs.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/name/{name}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/name/{name}/node&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Taxonomy&quot;: 
<code><a href="#topic+rba_uniprot_taxonomy">rba_uniprot_taxonomy</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lca">rba_uniprot_taxonomy_lca</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lineage">rba_uniprot_taxonomy_lineage</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_path">rba_uniprot_taxonomy_path</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_relationship">rba_uniprot_taxonomy_relationship</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_taxonomy_name(name = "homo", field = "scientific",
    search_type = "start_with")


rba_uniprot_taxonomy_name(name = "adenovirus", field = "scientific",
    search_type = "contain", page_size = 200, page_number = 2)


</code></pre>

<hr>
<h2 id='rba_uniprot_taxonomy_path'>Traverse UniProt Taxonomic Tree Path</h2><span id='topic+rba_uniprot_taxonomy_path'></span>

<h3>Description</h3>

<p>Using this function you can retrieve nodes that are located in the top or
the bottom of your supplied node in
<a href="https://www.uniprot.org/help/taxonomy">UniProt Taxonomy database tree</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_taxonomy_path(id, direction, depth = 5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_taxonomy_path_+3A_id">id</code></td>
<td>
<p>(numeric) a
<a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI taxonomic
identifier</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_path_+3A_direction">direction</code></td>
<td>
<p>direction of the taxonomic path, either &quot;TOP&quot; or &quot;BOTTOM&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_path_+3A_depth">depth</code></td>
<td>
<p>(numeric) How many levels should be traversed on
the taxonomic tree? (from 1 to 5, default = 5)</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_path_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a nested list containing the node which are in the path specified by
your supplied argument in the UniProt taxonomic tree.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/path&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Taxonomy&quot;: 
<code><a href="#topic+rba_uniprot_taxonomy">rba_uniprot_taxonomy</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lca">rba_uniprot_taxonomy_lca</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lineage">rba_uniprot_taxonomy_lineage</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_name">rba_uniprot_taxonomy_name</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_relationship">rba_uniprot_taxonomy_relationship</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_taxonomy_path(id = 9606, direction = "TOP", depth = 3)


rba_uniprot_taxonomy_path(id = 207598, direction = "BOTTOM", depth = 3)


</code></pre>

<hr>
<h2 id='rba_uniprot_taxonomy_relationship'>Get Shortest Path Between Two Taxonomy Nodes</h2><span id='topic+rba_uniprot_taxonomy_relationship'></span>

<h3>Description</h3>

<p>Use this function to retrieve the shortest path between two nodes
in the taxonomy tree of
<a href="https://www.uniprot.org/help/taxonomy">UniProt Taxonomy database</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_taxonomy_relationship(from, to, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_taxonomy_relationship_+3A_from">from</code></td>
<td>
<p><a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI
taxonomic identifier</a> of your initial node.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_relationship_+3A_to">to</code></td>
<td>
<p><a href="https://www.uniprot.org/help/taxonomic_identifier">NCBI
taxonomic identifier</a> of your final node.</p>
</td></tr>
<tr><td><code id="rba_uniprot_taxonomy_relationship_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a nested list containing the node which are in the shortest path
between your supplied nodes.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/relationship&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Taxonomy&quot;: 
<code><a href="#topic+rba_uniprot_taxonomy">rba_uniprot_taxonomy</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lca">rba_uniprot_taxonomy_lca</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_lineage">rba_uniprot_taxonomy_lineage</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_name">rba_uniprot_taxonomy_name</a>()</code>,
<code><a href="#topic+rba_uniprot_taxonomy_path">rba_uniprot_taxonomy_path</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_taxonomy_relationship(from = 9606, to = 10090)


</code></pre>

<hr>
<h2 id='rba_uniprot_uniparc'>Get UniParc entry</h2><span id='topic+rba_uniprot_uniparc'></span>

<h3>Description</h3>

<p>Use this function to retrieve UniParc entries. You can use either -and only
one of- UniProt accession, Cross-reference database id, UniParc ID or
UniProt Proteome UPID. You can also filter the returned content of
the returned UniParc entry. see &quot;Argument&quot; section for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_uniparc(
  upi = NULL,
  accession = NULL,
  db_id = NULL,
  upid = NULL,
  rf_dd_type = NULL,
  rf_db_id = NULL,
  rf_active = NULL,
  rf_tax_id = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_uniparc_+3A_upi">upi</code></td>
<td>
<p>unique UniParc Identifier.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_db_id">db_id</code></td>
<td>
<p>Protein ID in the cross-reference (external) database.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_rf_dd_type">rf_dd_type</code></td>
<td>
<p>Filter the content of the UniParc entry by
<a href="https://www.uniprot.org/database/">cross-reference</a> names. You can
supply multiple values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_rf_db_id">rf_db_id</code></td>
<td>
<p>Filter the content of the UniParc entry by protein
identifiers in any cross-reference database. You can supply multiple
values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_rf_active">rf_active</code></td>
<td>
<p>(logical ) Filter the content of UniParc entry based on
active status on source database:</p>

<ul>
<li><p> NULL: (default) don't filter contents based on active status.
</p>
</li>
<li><p> TRUE: only return contents which are still active.
</p>
</li>
<li><p> FALSE: Only return contents which are not active.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_rf_tax_id">rf_tax_id</code></td>
<td>
<p>(Numeric) Filter the content of the UniParc entry by
NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You can
supply multiple values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list which correspond to a UniParc entry.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/uniparc/accession/{accession} &quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/uniparc/dbreference/{dbid}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/uniparc/proteome/{upid}&quot;
<br /> &quot;GET https://ebi.ac.uk/proteins/api/uniparc/upi/{upi}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - UniParc&quot;: 
<code><a href="#topic+rba_uniprot_uniparc_bestguess">rba_uniprot_uniparc_bestguess</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_search">rba_uniprot_uniparc_search</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_sequence">rba_uniprot_uniparc_sequence</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_uniparc(upi = "UPI00000000C9")


rba_uniprot_uniparc(upi = "UPI00000000C9")


rba_uniprot_uniparc(upi = "UPI00000000C9", rf_active = FALSE)


</code></pre>

<hr>
<h2 id='rba_uniprot_uniparc_bestguess'>Get UniParc Longest Sequence for Entries</h2><span id='topic+rba_uniprot_uniparc_bestguess'></span>

<h3>Description</h3>

<p>This function returns the UniParc Entry with a cross-reference to the
longest active UniProtKB sequence (preferably from Swiss-Prot and if not
then TrEMBL). If it finds more than one longest active UniProtKB sequence
it returns 400 (Bad Request) error response with the list of cross
references found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_uniparc_bestguess(
  upi = NULL,
  accession = NULL,
  db_id = NULL,
  gene = NULL,
  taxid = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_uniparc_bestguess_+3A_upi">upi</code></td>
<td>
<p>unique UniParc Identifier.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_bestguess_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_bestguess_+3A_db_id">db_id</code></td>
<td>
<p>Protein ID in the cross-reference (external) database.
You can supply up to 100 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_bestguess_+3A_gene">gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_bestguess_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_bestguess_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list where each element correspond to a UniParc entry.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/uniparc/bestguess&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - UniParc&quot;: 
<code><a href="#topic+rba_uniprot_uniparc">rba_uniprot_uniparc</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_search">rba_uniprot_uniparc_search</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_sequence">rba_uniprot_uniparc_sequence</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_uniparc_bestguess("UPI00000000C9")


</code></pre>

<hr>
<h2 id='rba_uniprot_uniparc_search'>Search UniParc Entries</h2><span id='topic+rba_uniprot_uniparc_search'></span>

<h3>Description</h3>

<p>Use this function to search
<a href="https://www.uniprot.org/help/uniparc">UniProt Archive (UniParc)</a>
entries.You may also refine your search with modifiers such as sequence
length, taxon id etc. See &quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_uniparc_search(
  upi = NULL,
  accession = NULL,
  db_type = NULL,
  db_id = NULL,
  gene = NULL,
  protein = NULL,
  taxid = NULL,
  organism = NULL,
  sequence_checksum = NULL,
  ipr = NULL,
  signature_db = NULL,
  signature_id = NULL,
  upid = NULL,
  seq_length = NULL,
  rf_dd_type = NULL,
  rf_db_id = NULL,
  rf_active = NULL,
  rf_tax_id = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_uniparc_search_+3A_upi">upi</code></td>
<td>
<p>unique UniParc Identifier(s). You can supply up to 100 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_db_type">db_type</code></td>
<td>
<p><a href="https://www.uniprot.org/database/">cross-reference</a>
(external database) name.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_db_id">db_id</code></td>
<td>
<p>Protein ID in the cross-reference (external) database.
You can supply up to 100 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_gene">gene</code></td>
<td>
<p><a href="https://www.uniprot.org/help/gene_name">UniProt gene
name(s)</a>. You can supply up to 20 gene names.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_protein">protein</code></td>
<td>
<p><a href="https://www.uniprot.org/help/protein_names">UniProt
protein name</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_organism">organism</code></td>
<td>
<p><a href="https://www.uniprot.org/taxonomy/">Organism name</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_sequence_checksum">sequence_checksum</code></td>
<td>
<p>Sequence CRC64 checksum.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_ipr">ipr</code></td>
<td>
<p><a href="https://www.ebi.ac.uk/interpro/about/interpro/">InterPro
identifier(s)</a>. You can supply up to 20 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_signature_db">signature_db</code></td>
<td>
<p>InterPro's
<a href="https://interpro-documentation.readthedocs.io/en/latest/databases.html">signature
database</a>. You can supply up to 13 of the following values:
<br /> &quot;CATH&quot;, &quot;CDD&quot;, &quot;HAMAP&quot;, &quot;MobiDB Lite&quot;, &quot;Panther&quot;, &quot;Pfam&quot;, &quot;PIRSF&quot;,
&quot;PRINTS&quot;, &quot;Prosite&quot;, &quot;SFLD&quot;, &quot;SMART&quot;, &quot;SUPERFAMILY&quot; and/or &quot;TIGRfams&quot;</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_signature_id">signature_id</code></td>
<td>
<p>Signature ID in the InterPro's
<a href="https://interpro-documentation.readthedocs.io/en/latest/databases.html">signature
database</a>. You can supply up to 20 IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_upid">upid</code></td>
<td>
<p><a href="https://www.uniprot.org/help/proteome_id">UniProt Proteome
identifier (UPID)</a>. You can supply up to 100 UPIDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_seq_length">seq_length</code></td>
<td>
<p>An exact sequence length (e.g. 150) or a range of sequence
lengths (e.g. &quot;130-158&quot;).</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_rf_dd_type">rf_dd_type</code></td>
<td>
<p>Filter the content of the each UniParc entry by
<a href="https://www.uniprot.org/database/">cross-reference</a> names. You can
supply multiple values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_rf_db_id">rf_db_id</code></td>
<td>
<p>Filter the content of the each UniParc entry by protein
identifiers in any cross-reference database. You can supply multiple
values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_rf_active">rf_active</code></td>
<td>
<p>(logical ) Filter the content of each UniParc entry based on
active status on source database:</p>

<ul>
<li><p> NULL: (default) don't filter contents based on active status.
</p>
</li>
<li><p> TRUE: only return contents which are still active.
</p>
</li>
<li><p> FALSE: Only return contents which are not active.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_rf_tax_id">rf_tax_id</code></td>
<td>
<p>(Numeric) Filter the content of each UniParc entry by
NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You can
supply multiple values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>A List where each element corresponds to one UniParc entry returned
by your search query. The element itself is a sub-list containing sequence
information and reference entries.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://ebi.ac.uk/proteins/api/uniparc&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - UniParc&quot;: 
<code><a href="#topic+rba_uniprot_uniparc">rba_uniprot_uniparc</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_bestguess">rba_uniprot_uniparc_bestguess</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_sequence">rba_uniprot_uniparc_sequence</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_uniparc_search(upi = "UPI00000000C9")


rba_uniprot_uniparc_search(accession = "P30914")


rba_uniprot_uniparc_search(accession = "P30914", rf_active = TRUE)


rba_uniprot_uniparc_search(taxid = "694009", protein = "Nucleoprotein")


</code></pre>

<hr>
<h2 id='rba_uniprot_uniparc_sequence'>Get UniParc Entries by Sequence</h2><span id='topic+rba_uniprot_uniparc_sequence'></span>

<h3>Description</h3>

<p>Retrieve UniParc Entry by providing an exact sequence. Note that partial
matches will not be accepted. You can also filter the returned content of
the returned UniParc entry. see &quot;Argument&quot; section for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_uniparc_sequence(
  sequence,
  rf_dd_type = NULL,
  rf_db_id = NULL,
  rf_active = NULL,
  rf_tax_id = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_uniparc_sequence_+3A_sequence">sequence</code></td>
<td>
<p>Exact UniParc protein sequence. Partial matches will not be
accepted.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_sequence_+3A_rf_dd_type">rf_dd_type</code></td>
<td>
<p>Filter the content of the UniParc entry by
<a href="https://www.uniprot.org/database/">cross-reference</a> names. You can
supply multiple values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_sequence_+3A_rf_db_id">rf_db_id</code></td>
<td>
<p>Filter the content of the UniParc entry by protein
identifiers in any cross-reference database. You can supply multiple
values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_sequence_+3A_rf_active">rf_active</code></td>
<td>
<p>(logical ) Filter the content of UniParc entry based on
active status on source database:</p>

<ul>
<li><p> NULL: (default) don't filter contents based on active status.
</p>
</li>
<li><p> TRUE: only return contents which are still active.
</p>
</li>
<li><p> FALSE: Only return contents which are not active.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_sequence_+3A_rf_tax_id">rf_tax_id</code></td>
<td>
<p>(Numeric) Filter the content of the UniParc entry by
NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>. You can
supply multiple values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_uniparc_sequence_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list which correspond to a UniParc entry.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;POST https://ebi.ac.uk/proteins/api/uniparc/sequence&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - UniParc&quot;: 
<code><a href="#topic+rba_uniprot_uniparc">rba_uniprot_uniparc</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_bestguess">rba_uniprot_uniparc_bestguess</a>()</code>,
<code><a href="#topic+rba_uniprot_uniparc_search">rba_uniprot_uniparc_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_uniparc_sequence("GMRSCPRGCSQRGRCENGRCVCNPGYTGEDC")


</code></pre>

<hr>
<h2 id='rba_uniprot_variation'>Get natural variants in UniProt by NIH-NCBI SNP database identifier</h2><span id='topic+rba_uniprot_variation'></span>

<h3>Description</h3>

<p>Retrieve natural variant annotations of a sequence using UniProt protein
accession, dbSNP or HGVS expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_variation(
  id,
  id_type,
  source_type = NULL,
  consequence_type = NULL,
  wild_type = NULL,
  alternative_sequence = NULL,
  location = NULL,
  save_peff = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_variation_+3A_id">id</code></td>
<td>
<p>An ID which can be either a
<a href="https://www.uniprot.org/help/accession_numbers">UniProt primary or
secondary accession</a>, NIH-NCBI dbSNP ID or HGVS expression.
<a href="https://www.ncbi.nlm.nih.gov/snp/">NIH-NCBI dbSNP id</a> or
<a href="https://varnomen.hgvs.org/">HGVS Expression</a>.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_id_type">id_type</code></td>
<td>
<p>The type of supplied ID argument, one of:
<a href="https://www.uniprot.org/help/accession_numbers">&quot;uniprot&quot;</a>,
<a href="https://www.ncbi.nlm.nih.gov/snp/">&quot;dbsnp&quot;</a> or
<a href="https://varnomen.hgvs.org/">&quot;hgvs&quot;</a></p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_source_type">source_type</code></td>
<td>
<p>Variation's source type. You can choose up to two of:
&quot;UniProt&quot;, &quot;large scale study&quot; and/or &quot;mixed&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_consequence_type">consequence_type</code></td>
<td>
<p>Variation's consequence type. You can choose up to
two of: &quot;missense&quot;, &quot;stop gained&quot; or &quot;stop lost&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_wild_type">wild_type</code></td>
<td>
<p>Wild type amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and &quot;*&quot; for stop
codon. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_alternative_sequence">alternative_sequence</code></td>
<td>
<p>Alternative amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and &quot;*&quot; for stop codon
and &quot;-&quot; for deletion. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_location">location</code></td>
<td>
<p>A valid amino acid range (e.g. 10-25) within the sequence
range where the variation occurs. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_save_peff">save_peff</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li><p> FALSE: (default) Do not save PEFF file, just return as a list object.
</p>
</li>
<li><p> TRUE: Save as PEFF file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the PEFF file.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_variation_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list where each element is a list that corresponds to a UniProt
protein entry.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/variation/dbsnp/{dbid}&quot;
<br /> &quot;GET https://www.ebi.ac.uk/proteins/api/variation/hgvs/{hgvs}&quot;
<br /> &quot;GET https://www.ebi.ac.uk/proteins/api/variation/{accession}&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Variation&quot;: 
<code><a href="#topic+rba_uniprot_variation_search">rba_uniprot_variation_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_variation(id = "rs121434451", id_type = "dbsnp")


rba_uniprot_variation(id = "NC_000008.11:g.22119227C&gt;T", id_type = "hgvs")


rba_uniprot_variation(id = "O43593", id_type = "uniprot")


</code></pre>

<hr>
<h2 id='rba_uniprot_variation_search'>Search UniProt Natural Variants</h2><span id='topic+rba_uniprot_variation_search'></span>

<h3>Description</h3>

<p>Using this function, you can search and retrieve
<a href="https://www.uniprot.org/help/variant">Natural variant(s)</a> that
has been annotated in the protein's sequences. You may also refine your
search with modifiers such as source type, disease etc. See
&quot;Arguments section&quot; for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rba_uniprot_variation_search(
  accession = NULL,
  source_type = NULL,
  consequence_type = NULL,
  wild_type = NULL,
  alternative_sequence = NULL,
  location = NULL,
  disease = NULL,
  omim = NULL,
  evidence = NULL,
  taxid = NULL,
  db_type = NULL,
  db_id = NULL,
  save_peff = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rba_uniprot_variation_search_+3A_accession">accession</code></td>
<td>
<p><a href="https://www.uniprot.org/help/accession_numbers">
UniProtKB primary or secondary accession</a>(s). You can supply up to 100
accession numbers.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_source_type">source_type</code></td>
<td>
<p>Variation's source type. You can choose up to two of:
&quot;UniProt&quot;, &quot;large scale study&quot; and/or &quot;mixed&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_consequence_type">consequence_type</code></td>
<td>
<p>Variation's consequence type. You can choose up to
two of: &quot;missense&quot;, &quot;stop gained&quot; or &quot;stop lost&quot;.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_wild_type">wild_type</code></td>
<td>
<p>Wild type amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and &quot;*&quot; for stop
codon. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_alternative_sequence">alternative_sequence</code></td>
<td>
<p>Alternative amino acid. Accepted values are IUPAC
single-letter amino acid (e.g. D for    Aspartic acid) and &quot;*&quot; for stop codon
and &quot;-&quot; for deletion. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_location">location</code></td>
<td>
<p>A valid amino acid range (e.g. 10-25) within the sequence
range where the variation occurs. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_disease">disease</code></td>
<td>
<p><a href="https://www.uniprot.org/diseases/">Human disease</a>
that are associated with a sequence variation. Accepted values are
disease name (e.g. Alzheimer disease 18), partial disease name
(Alzheimer) and/or disease acronym (e.g. AD). You can supply up to
20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_omim">omim</code></td>
<td>
<p><a href="https://www.ncbi.nlm.nih.gov/omim">OMIM</a> ID that is
associated with a variation. You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_evidence">evidence</code></td>
<td>
<p>Pubmed ID of the variation's
<a href="https://www.uniprot.org/citations/">citation</a> You can supply up
to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_taxid">taxid</code></td>
<td>
<p>NIH-NCBI <a href="https://www.uniprot.org/taxonomy/">Taxon ID</a>.
You can supply up to 20 taxon IDs.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_db_type">db_type</code></td>
<td>
<p>cross-reference database of the variation. You can supply
up to two of the following:</p>

<ul>
<li><p> &quot;dbSNP&quot;: <a href="https://www.ncbi.nlm.nih.gov/snp/">NIH-NCBI dbSNP
database</a>.
</p>
</li>
<li><p> &quot;cosmic curate&quot;: <a href="https://cancer.sanger.ac.uk/cosmic/">COSMIC
(the Catalogue of Somatic Mutations in Cancer)</a>
</p>
</li>
<li><p> &quot;ClinVar&quot;: <a href="https://www.ncbi.nlm.nih.gov/clinvar/">NIH-NCBI
ClinVar</a>
</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_db_id">db_id</code></td>
<td>
<p>The variation ID in a Cross-reference (external) database.
You can supply up to 20 values.</p>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_save_peff">save_peff</code></td>
<td>
<p>Logical or Character:</p>

<ul>
<li><p> FALSE: (default) Do not save PEFF file, just return as a list object.
</p>
</li>
<li><p> TRUE: Save as PEFF file to an automatically-generated path.
</p>
</li>
<li><p> Character string: A valid file path to save the PEFF file.</p>
</li></ul>
</td></tr>
<tr><td><code id="rba_uniprot_variation_search_+3A_...">...</code></td>
<td>
<p>rbioapi option(s). See <code><a href="#topic+rba_options">rba_options</a></code>'s
arguments manual for more information on available options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a search function. Thus, you are not required to fill
every argument; You may use whatever combinations of arguments you see
fit for your query.
</p>


<h3>Value</h3>

<p>List where each element corresponds to one UniProt entity returned
by your search query. The element itself is a sub-list containing all
information that UniProt has about that Variation.
</p>


<h3>Corresponding API Resources</h3>

<p>&quot;GET https://www.ebi.ac.uk/proteins/api/variation&quot;
</p>


<h3>References</h3>


<ul>
<li><p> The UniProt Consortium , UniProt: the Universal Protein
Knowledgebase in 2025, Nucleic Acids Research, 2024;, gkae1010,
https://doi.org/10.1093/nar/gkae1010
</p>
</li>
<li><p> Andrew Nightingale, Ricardo Antunes, Emanuele Alpi, Borisas
Bursteinas, Leonardo Gonzales, Wudong Liu, Jie Luo, Guoying Qi, Edd
Turner, Maria Martin, The Proteins API: accessing key integrated protein
and genome information, Nucleic Acids Research, Volume 45, Issue W1,
3 July 2017, Pages W539–W544, https://doi.org/10.1093/nar/gkx237
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/proteins/api/doc/">Proteins API
Documentation</a>
</p>
</li>
<li> <p><a href="https://www.uniprot.org/help/publications">Citations note
on UniProt website</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other &quot;UniProt - Variation&quot;: 
<code><a href="#topic+rba_uniprot_variation">rba_uniprot_variation</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rba_uniprot_variation_search(accession = "P05067")


rba_uniprot_variation_search(disease = "alzheimer disease, 18")


rba_uniprot_variation_search(disease = "alzheimer",
    wild_type = "A", alternative_sequence = "T")


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
