<!DOCTYPE html><html lang="en"><head><title>Help for package rbiom</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rbiom}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rbiom-package'><p>rbiom: Read/Write, Transform, and Summarize BIOM Data</p></a></li>
<li><a href='#adiv_boxplot'><p>Visualize alpha diversity with boxplots.</p></a></li>
<li><a href='#adiv_corrplot'><p>Visualize alpha diversity with scatterplots and trendlines.</p></a></li>
<li><a href='#adiv_matrix'><p>Create a matrix of samples x alpha diversity metrics.</p></a></li>
<li><a href='#adiv_stats'><p>Test alpha diversity for associations with metadata.</p></a></li>
<li><a href='#adiv_table'><p>Calculate the alpha diversity of each sample.</p></a></li>
<li><a href='#as_rbiom'><p>Convert a variety of data types to an rbiom object.</p></a></li>
<li><a href='#as.list.rbiom'><p>Convert an rbiom object to a base R list.</p></a></li>
<li><a href='#as.matrix.rbiom'><p>Convert an rbiom object to a simple count matrix.</p></a></li>
<li><a href='#babies'><p>Longitudinal Stool Samples from Infants (n = 2,684)</p></a></li>
<li><a href='#bdiv_boxplot'><p>Visualize BIOM data with boxplots.</p></a></li>
<li><a href='#bdiv_clusters'><p>Cluster samples by beta diversity k-means.</p></a></li>
<li><a href='#bdiv_corrplot'><p>Visualize beta diversity with scatterplots and trendlines.</p></a></li>
<li><a href='#bdiv_heatmap'><p>Display beta diversities in an all vs all grid.</p></a></li>
<li><a href='#bdiv_ord_plot'><p>Ordinate samples and taxa on a 2D plane based on beta diversity distances.</p></a></li>
<li><a href='#bdiv_ord_table'><p>Calculate PCoA and other ordinations, including taxa biplots and statistics.</p></a></li>
<li><a href='#bdiv_stats'><p>Test beta diversity for associations with metadata.</p></a></li>
<li><a href='#bdiv_table'><p>Distance / dissimilarity between samples.</p></a></li>
<li><a href='#bdply'><p>Apply a function to each subset of an rbiom object.</p></a></li>
<li><a href='#biom_merge'><p>Combine several rbiom objects into one.</p></a></li>
<li><a href='#convert_to'><p>Convert biom data to an external package class.</p></a></li>
<li><a href='#distmat_ord_table'><p>Run ordinations on a distance matrix.</p></a></li>
<li><a href='#distmat_stats'><p>Run statistics on a distance matrix vs a categorical or numeric variable.</p></a></li>
<li><a href='#documentation_biom.rbiom'><p>documentation_biom.rbiom</p></a></li>
<li><a href='#documentation_clusters'><p>documentation_clusters</p></a></li>
<li><a href='#documentation_cmp'><p>documentation_cmp</p></a></li>
<li><a href='#documentation_default'><p>documentation_default</p></a></li>
<li><a href='#documentation_dist_test'><p>documentation_dist_test</p></a></li>
<li><a href='#documentation_heatmap'><p>documentation_heatmap</p></a></li>
<li><a href='#documentation_plot_return'><p>documentation_plot_return</p></a></li>
<li><a href='#documentation_rank.2'><p>documentation_rank.2</p></a></li>
<li><a href='#documentation_rank.NULL'><p>documentation_rank.NULL</p></a></li>
<li><a href='#documentation_return.biom'><p>documentation_return.biom</p></a></li>
<li><a href='#documentation_taxa.4'><p>documentation_taxa.4</p></a></li>
<li><a href='#export'><p>Export data to QIIME 2 or mothur.</p></a></li>
<li><a href='#gems'><p>Global Enteric Multicenter Study (n = 1,006)</p></a></li>
<li><a href='#glimpse.rbiom'><p>Get a glimpse of your metadata.</p></a></li>
<li><a href='#hmp50'><p>Human Microbiome Project - demo dataset (n = 50)</p></a></li>
<li><a href='#modify_metadata'><p>Create, modify, and delete metadata fields.</p></a></li>
<li><a href='#plot_heatmap'><p>Create a heatmap with tracks and dendrograms from any matrix.</p></a></li>
<li><a href='#pull.rbiom'><p>Map sample names to metadata field values.</p></a></li>
<li><a href='#rare_corrplot'><p>Visualize rarefaction curves with scatterplots and trendlines.</p></a></li>
<li><a href='#rare_multiplot'><p>Combines rare_corrplot and rare_stacked into a single figure.</p></a></li>
<li><a href='#rare_stacked'><p>Visualize the number of observations per sample.</p></a></li>
<li><a href='#rarefy'><p>Rarefy OTU counts.</p></a></li>
<li><a href='#rarefy_cols'><p>Transform a counts matrix.</p></a></li>
<li><a href='#rbiom_objects'><p>Working with rbiom Objects.</p></a></li>
<li><a href='#rbiom-deprecated'><p>Deprecated functions in package <span class="pkg">rbiom</span>.</p></a></li>
<li><a href='#read_biom'><p>Parse counts, metadata, taxonomy, and phylogeny from a BIOM file.</p></a></li>
<li><a href='#read_fasta'><p>Parse a fasta file into a named character vector.</p></a></li>
<li><a href='#read_tree'><p>Read a newick formatted phylogenetic tree.</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#sample_sums'><p>Summarize the taxa observations in each sample.</p></a></li>
<li><a href='#slice_metadata'><p>Subset to a specific number of samples.</p></a></li>
<li><a href='#speed'><p>Speed Ups.</p></a></li>
<li><a href='#stats_boxplot'><p>Visualize categorical metadata effects on numeric values.</p></a></li>
<li><a href='#stats_corrplot'><p>Visualize regression with scatterplots and trendlines.</p></a></li>
<li><a href='#stats_table'><p>Run non-parametric statistics on a data.frame.</p></a></li>
<li><a href='#subset'><p>Subset an rbiom object by sample names, OTU names, metadata, or taxonomy.</p></a></li>
<li><a href='#taxa_boxplot'><p>Visualize BIOM data with boxplots.</p></a></li>
<li><a href='#taxa_clusters'><p>Cluster samples by taxa abundances k-means.</p></a></li>
<li><a href='#taxa_corrplot'><p>Visualize taxa abundance with scatterplots and trendlines.</p></a></li>
<li><a href='#taxa_heatmap'><p>Display taxa abundances as a heatmap.</p></a></li>
<li><a href='#taxa_map'><p>Map OTUs names to taxa names at a given rank.</p></a></li>
<li><a href='#taxa_stacked'><p>Display taxa abundances as a stacked bar graph.</p></a></li>
<li><a href='#taxa_stats'><p>Test taxa abundances for associations with metadata.</p></a></li>
<li><a href='#taxa_sums'><p>Get summary taxa abundances.</p></a></li>
<li><a href='#taxa_table'><p>Taxa abundances per sample.</p></a></li>
<li><a href='#tree_subset'><p>Create a subtree by specifying tips to keep.</p></a></li>
<li><a href='#with'><p>Evaluate expressions on metadata.</p></a></li>
<li><a href='#write_biom'><p>Save an rbiom object to a file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Read/Write, Analyze, and Visualize 'BIOM' Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-03-17</td>
</tr>
<tr>
<td>Description:</td>
<td>
    A toolkit for working with Biological Observation Matrix ('BIOM') files.
    Read/write all 'BIOM' formats. Compute rarefaction, alpha diversity, and 
    beta diversity (including 'UniFrac'). Summarize counts by taxonomic level. 
    Subset based on metadata. Generate visualizations and statistical analyses. 
    CPU intensive operations are coded in C for speed.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://cmmr.github.io/rbiom/">https://cmmr.github.io/rbiom/</a>, <a href="https://github.com/cmmr/rbiom">https://github.com/cmmr/rbiom</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cmmr/rbiom/issues">https://github.com/cmmr/rbiom/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>rmarkdown, phyloseq, npregfast, withr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, mgcv, stats, utils, ape, dplyr, emmeans, fillpattern,
ggbeeswarm, ggnewscale, ggplot2, ggrepel, ggtext, jsonlite,
magrittr, parallelly, patchwork, pillar, plyr, readr, readxl,
slam, vegan</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cli, crayon, ggdensity, glue, labeling, lifecycle, openxlsx,
optparse, pkgconfig, prettycode, R6, rlang, scales, testthat,
tibble, tsne, uwot</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-18 17:13:51 UTC; Daniel</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel P. Smith <a href="https://orcid.org/0000-0002-2479-2044"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Alkek Center for Metagenomics and Microbiome Research [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel P. Smith &lt;dansmith01@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-18 17:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='rbiom-package'>rbiom: Read/Write, Transform, and Summarize BIOM Data</h2><span id='topic+rbiom'></span><span id='topic+rbiom-package'></span>

<h3>Description</h3>

<p>A toolkit for working with Biological Observation Matrix (BIOM) files.
Features include reading/writing all BIOM formats, rarefaction, alpha
diversity, beta diversity (including UniFrac), summarizing counts by
taxonomic level, and sample subsetting. Standalone functions for
reading, writing, and subsetting phylogenetic trees are also provided.
All CPU intensive operations are encoded in C with multi-thread support.
</p>


<h3>Multithreading</h3>

<p>Many rbiom functions support multithreading:
</p>
<p>The default behavior of these function is to run on as many cores as are
available in the local compute environment. If you wish to limit the number
of simultaneous threads, set <code>RcppParallel</code>'s <code>numThreads</code> option.
For instance:
</p>
<pre>
    RcppParallel::setThreadOptions(numThreads = 4)
</pre>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Daniel P. Smith <a href="mailto:dansmith01@gmail.com">dansmith01@gmail.com</a> (<a href="https://orcid.org/0000-0002-2479-2044">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Alkek Center for Metagenomics and Microbiome Research [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://cmmr.github.io/rbiom/">https://cmmr.github.io/rbiom/</a>
</p>
</li>
<li> <p><a href="https://github.com/cmmr/rbiom">https://github.com/cmmr/rbiom</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/cmmr/rbiom/issues">https://github.com/cmmr/rbiom/issues</a>
</p>
</li></ul>


<hr>
<h2 id='adiv_boxplot'>Visualize alpha diversity with boxplots.</h2><span id='topic+adiv_boxplot'></span>

<h3>Description</h3>

<p>Visualize alpha diversity with boxplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adiv_boxplot(
  biom,
  x = NULL,
  adiv = "Shannon",
  layers = "x",
  stat.by = x,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  patterns = FALSE,
  flip = FALSE,
  stripe = NULL,
  ci = "ci",
  level = 0.95,
  p.adj = "fdr",
  outliers = NULL,
  xlab.angle = "auto",
  p.label = 0.05,
  transform = "none",
  caption = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adiv_boxplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_x">x</code></td>
<td>
<p>A categorical metadata column name to use for the x-axis. Or
<code>NULL</code>, which groups all samples into a single category.</p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("bar", "box" ("x"), "violin", "dot", "strip", "crossbar", "errorbar", "linerange", "pointrange")</code>.
Single letter abbreviations are also accepted. For instance,
<code>c("box", "dot")</code> is equivalent to <code>c("x", "d")</code> and <code>"xd"</code>.
Default: <code>"x"</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_patterns">patterns</code></td>
<td>
<p>Patterns for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, pattern
names (<code>"brick"</code>, <code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, etc), or a named
vector mapping groups to specific pattern names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_flip">flip</code></td>
<td>
<p>Transpose the axes, so that taxa are present as rows instead
of columns. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_stripe">stripe</code></td>
<td>
<p>Shade every other x position. Default: <em>same as flip</em></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_ci">ci</code></td>
<td>
<p>How to calculate min/max of the <b>crossbar</b>,
<b>errorbar</b>, <b>linerange</b>, and <b>pointrange</b> layers.
Options are: <code>"ci"</code> (confidence interval), <code>"range"</code>,
<code>"sd"</code> (standard deviation), <code>"se"</code> (standard error), and
<code>"mad"</code> (median absolute deviation).
The center mark of <b>crossbar</b> and <b>pointrange</b> represents
the mean, except for <code>"mad"</code> in which case it represents the median.
Default: <code>"ci"</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_outliers">outliers</code></td>
<td>
<p>Show boxplot outliers? <code>TRUE</code> to always show.
<code>FALSE</code> to always hide. <code>NULL</code> to only hide them when
overlaying a dot or strip chart.  Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_p.label">p.label</code></td>
<td>
<p>Minimum adjusted p-value to display on the plot with a
bracket.
</p>

<dl>
<dt><code>p.label = 0.05</code> - </dt><dd><p> Show p-values that are &lt;= 0.05. </p>
</dd>
<dt><code>p.label = 0</code> - </dt><dd><p> Don't show any p-values on the plot. </p>
</dd>
<dt><code>p.label = 1</code> - </dt><dd><p> Show all p-values on the plot. </p>
</dd>
</dl>

<p>If a numeric vector with more than one value is
provided, they will be used as breaks for asterisk notation.
Default: <code>0.05</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="adiv_boxplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>d.size = 2</code> ensures only the points on the
<b>dot</b> layer have their size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Patterns are added using the fillpattern R package. Options are <code>"brick"</code>,
<code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, <code>"herringbone"</code>, <code>"hexagon"</code>, <code>"octagon"</code>,
<code>"rain"</code>, <code>"saw"</code>, <code>"shingle"</code>, <code>"rshingle"</code>, <code>"stripe"</code>, and <code>"wave"</code>,
optionally abbreviated and/or suffixed with modifiers. For example,
<code>"hex10_sm"</code> for the hexagon pattern rotated 10 degrees and shrunk by 2x.
See <code><a href="fillpattern.html#topic+fill_pattern">fillpattern::fill_pattern()</a></code> for complete documentation of options.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other alpha_diversity: 
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>,
<code><a href="#topic+adiv_table">adiv_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    adiv_boxplot(biom, x="Body Site", stat.by="Body Site")
    
    adiv_boxplot(biom, x="Sex", stat.by="Body Site", adiv=c("otu", "shan"), layers = "bld")
    
    adiv_boxplot(biom, x="body", stat.by="sex", adiv=".all", flip=TRUE, layers="p")
    
    
    # Each plot object includes additional information.
    fig &lt;- adiv_boxplot(biom, x="Body Site")
    
    ## Computed Data Points -------------------
    fig$data
    
    ## Statistics Table -----------------------
    fig$stats
    
    ## ggplot2 Command ------------------------
    fig$code
    
</code></pre>

<hr>
<h2 id='adiv_corrplot'>Visualize alpha diversity with scatterplots and trendlines.</h2><span id='topic+adiv_corrplot'></span>

<h3>Description</h3>

<p>Visualize alpha diversity with scatterplots and trendlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adiv_corrplot(
  biom,
  x,
  adiv = "Shannon",
  layers = "tc",
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  p.adj = "fdr",
  transform = "none",
  alt = "!=",
  mu = 0,
  caption = TRUE,
  check = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adiv_corrplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_x">x</code></td>
<td>
<p>Dataset field with the x-axis values. Equivalent to the <code>regr</code>
argument in <code><a href="#topic+stats_table">stats_table()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("trend", "confidence", "point", "name", "residual")</code>. Single
letter abbreviations are also accepted. For instance,
<code>c("trend", "point")</code> is equivalent to <code>c("t", "p")</code> and <code>"tp"</code>.
Default: <code>"tc"</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'none'</code>, <code>'emmeans'</code>, or
<code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="adiv_corrplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>p.size = 2</code> ensures only the points have their
size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other alpha_diversity: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>,
<code><a href="#topic+adiv_table">adiv_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    p &lt;- adiv_corrplot(babies, "age", stat.by = "deliv", fit = "gam")
    
    p
    
    p$stats
    
    p$code
</code></pre>

<hr>
<h2 id='adiv_matrix'>Create a matrix of samples x alpha diversity metrics.</h2><span id='topic+adiv_matrix'></span>

<h3>Description</h3>

<p>Create a matrix of samples x alpha diversity metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adiv_matrix(biom, adiv = ".all", transform = "none", cpus = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adiv_matrix_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="adiv_matrix_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>".all"</code></p>
</td></tr>
<tr><td><code id="adiv_matrix_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="adiv_matrix_+3A_cpus">cpus</code></td>
<td>
<p>The number of CPUs to use. Set to <code>NULL</code> to use all available,
or to <code>1</code> to disable parallel processing. Default: <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix with samples as rows. The first column is
<b>Depth</b>. Remaining columns are the alpha diversity metric names
given by <code>adiv</code>: one or more of <b>OTUs</b>, <b>Shannon</b>,
<b>Chao1</b>, <b>Simpson</b>, and <b>InvSimpson</b>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- slice_head(hmp50, n = 5)
    
    adiv_matrix(biom)
</code></pre>

<hr>
<h2 id='adiv_stats'>Test alpha diversity for associations with metadata.</h2><span id='topic+adiv_stats'></span>

<h3>Description</h3>

<p>A convenience wrapper for <code><a href="#topic+adiv_table">adiv_table()</a></code> + <code><a href="#topic+stats_table">stats_table()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adiv_stats(
  biom,
  regr = NULL,
  stat.by = NULL,
  adiv = "Shannon",
  split.by = NULL,
  transform = "none",
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  alt = "!=",
  mu = 0,
  p.adj = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adiv_stats_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_regr">regr</code></td>
<td>
<p>Dataset field with the x-axis (independent; predictive)
values. Must be numeric. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_split.by">split.by</code></td>
<td>
<p>Dataset field(s) that the data should be split by prior to
any calculations. Must be categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'wilcox'</code>, <code>'kruskal'</code>,
<code>'emmeans'</code>, or <code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="adiv_stats_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble data.frame with fields from the table below. This tibble
object provides the <code style="white-space: pre;">&#8288;$code&#8288;</code> operator to print the R code used to generate
the statistics.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Field</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean </td><td style="text-align: left;"> Estimated marginal mean. See <code><a href="emmeans.html#topic+emmeans">emmeans::emmeans()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean.diff </td><td style="text-align: left;"> Difference in means. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope </td><td style="text-align: left;"> Trendline slope. See <code><a href="emmeans.html#topic+emtrends">emmeans::emtrends()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope.diff </td><td style="text-align: left;"> Difference in slopes. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .h1 </td><td style="text-align: left;"> Alternate hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .p.val </td><td style="text-align: left;"> Probability that null hypothesis is correct. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.p </td><td style="text-align: left;"> <code>.p.val</code> after adjusting for multiple comparisons. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .effect.size </td><td style="text-align: left;"> Effect size. See <code><a href="emmeans.html#topic+eff_size">emmeans::eff_size()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .lower </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .upper </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .se </td><td style="text-align: left;"> Standard error. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .n </td><td style="text-align: left;"> Number of samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .df </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .stat </td><td style="text-align: left;"> Wilcoxon or Kruskal-Wallis rank sum statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .t.ratio </td><td style="text-align: left;"> <code>.mean</code> / <code>.se</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .r.sqr </td><td style="text-align: left;"> Percent of variation explained by the model. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.r </td><td style="text-align: left;"> <code>.r.sqr</code>, taking degrees of freedom into account. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .aic </td><td style="text-align: left;"> Akaike Information Criterion (predictive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .bic </td><td style="text-align: left;"> Bayesian Information Criterion (descriptive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .loglik </td><td style="text-align: left;"> Log-likelihood goodness-of-fit score. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .fit.p </td><td style="text-align: left;"> P-value for observing this fit by chance. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>See Also</h3>

<p>Other alpha_diversity: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+adiv_table">adiv_table</a>()</code>
</p>
<p>Other stats_tables: 
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>,
<code><a href="#topic+stats_table">stats_table</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    biom &lt;- rarefy(hmp50)
      
    adiv_stats(biom, stat.by = "Sex")[,1:6]
      
    adiv_stats(biom, stat.by = "Sex", split.by = "Body Site")[,1:6]
    
    adiv_stats(biom, stat.by = "Body Site", test = "kruskal")
</code></pre>

<hr>
<h2 id='adiv_table'>Calculate the alpha diversity of each sample.</h2><span id='topic+adiv_table'></span>

<h3>Description</h3>

<p>Calculate the alpha diversity of each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adiv_table(
  biom,
  adiv = "Shannon",
  md = ".all",
  transform = "none",
  cpus = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adiv_table_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="adiv_table_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="adiv_table_+3A_md">md</code></td>
<td>
<p>Dataset field(s) to include in the output data frame, or <code>'.all'</code>
to include all metadata fields. Default: <code>'.all'</code></p>
</td></tr>
<tr><td><code id="adiv_table_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="adiv_table_+3A_cpus">cpus</code></td>
<td>
<p>The number of CPUs to use. Set to <code>NULL</code> to use all available,
or to <code>1</code> to disable parallel processing. Default: <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of alpha diversity values.
Each combination of sample/depth/<code>adiv</code> has its own row.
Column names are <b>.sample</b>, <b>.depth</b>, <b>.adiv</b>,
and <b>.diversity</b>, followed by any metadata fields requested by
<code>md</code>.
</p>


<h3>See Also</h3>

<p>Other alpha_diversity: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    # Subset to 10 samples.
    biom &lt;- slice(hmp50, 1:10)
    adiv_table(biom)
    
    biom &lt;- rarefy(biom)
    adiv_table(biom, md = NULL)
</code></pre>

<hr>
<h2 id='as_rbiom'>Convert a variety of data types to an rbiom object.</h2><span id='topic+as_rbiom'></span>

<h3>Description</h3>

<p>Construct an rbiom object. The returned object is an R6 reference class.
Use <code>b &lt;- a$clone()</code> to create copies, not <code>b &lt;- a</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_rbiom(biom, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as_rbiom_+3A_biom">biom</code></td>
<td>
<p>Object which can be coerced to an rbiom-class object.
For example:
</p>

<dl>
<dt><em>file</em> - </dt><dd><p> Filepath or URL to a biom file. </p>
</dd>
<dt><em>matrix</em> - </dt><dd><p> An abundance matrix with OTUs in rows and samples in columns. </p>
</dd>
<dt><code>phyloseq</code>-class object - </dt><dd><p> From the phyloseq Bioconductor R package. </p>
</dd>
<dt><em>list</em> - </dt><dd><p> With <code>counts</code> and optionally <code>metadata</code>, <code>taxonomy</code>, <code>tree</code>, etc (see details). </p>
</dd>
</dl>
</td></tr>
<tr><td><code id="as_rbiom_+3A_...">...</code></td>
<td>
<p>Properties to overwrite in biom: <code>metadata</code>, <code>taxonomy</code>,
<code>tree</code>, etc (see details). Setting <code>underscores</code> here will pass it to
<code>read_tree()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # create a simple matrix ------------------------
    mtx &lt;- matrix(
      data     = floor(runif(24) * 1000), 
      nrow     = 6, 
      dimnames = list(paste0("OTU", 1:6), paste0("Sample", 1:4)) )
    mtx
    
    # and some sample metadata ----------------------
    df &lt;- data.frame(
      .sample   = paste0("Sample", 1:4),
      treatment = c("A", "B", "A", "B"),
      days      = c(12, 3, 7, 8) )
    
    # convert data set to rbiom ---------------------
    biom &lt;- as_rbiom(mtx, metadata = df, id = "My BIOM")
    biom

</code></pre>

<hr>
<h2 id='as.list.rbiom'>Convert an rbiom object to a base R list.</h2><span id='topic+as.list.rbiom'></span>

<h3>Description</h3>

<p>Convert an rbiom object to a base R list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
as.list(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.list.rbiom_+3A_x">x</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="as.list.rbiom_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with names
<code>c('counts', 'metadata', 'taxonomy', 'tree', 'sequences', 'id', 'comment', 'date', 'generated_by')</code>.
</p>


<h3>See Also</h3>

<p>Other conversion: 
<code><a href="#topic+as.matrix.rbiom">as.matrix.rbiom</a>()</code>
</p>

<hr>
<h2 id='as.matrix.rbiom'>Convert an rbiom object to a simple count matrix.</h2><span id='topic+as.matrix.rbiom'></span>

<h3>Description</h3>

<p>Identical to running <code>as.matrix(biom$counts)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
as.matrix(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.matrix.rbiom_+3A_x">x</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="as.matrix.rbiom_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A base R matrix with OTUs as rows and samples as columns.
</p>


<h3>See Also</h3>

<p>Other conversion: 
<code><a href="#topic+as.list.rbiom">as.list.rbiom</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    as.matrix(hmp50)[1:5,1:5]

</code></pre>

<hr>
<h2 id='babies'>Longitudinal Stool Samples from Infants (n = 2,684)</h2><span id='topic+babies'></span>

<h3>Description</h3>

<p>Longitudinal Stool Samples from Infants (n = 2,684)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>babies
</code></pre>


<h3>Format</h3>

<p>An rbiom object with 2,684 samples.
Includes metadata and taxonomy.
</p>

<dl>
<dt>Subject ID - </dt><dd><p>ID1, ID2, ..., ID12</p>
</dd>
<dt>Sex - </dt><dd><p>Male or Female</p>
</dd>
<dt>Age (days) - </dt><dd><p>1 - 266</p>
</dd>
<dt>Child's diet - </dt><dd><p>&quot;Breast milk&quot;, &quot;Breast milk and formula&quot;, or &quot;Formula&quot;</p>
</dd>
<dt>Sample collection - </dt><dd><p>&quot;Frozen upon collection&quot; or &quot;Stored in alcohol&quot;</p>
</dd>
<dt>Antibiotic exposure - </dt><dd><p>Yes or No</p>
</dd>
<dt>Antifungal exposure - </dt><dd><p>Yes or No</p>
</dd>
<dt>Delivery mode - </dt><dd><p>Cesarean or Vaginal</p>
</dd>
<dt>Solid food introduced (Age) - </dt><dd><p>116 - 247</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.nature.com/articles/s41467-018-04641-7">https://www.nature.com/articles/s41467-018-04641-7</a> and <a href="https://doi.org/10.1038/s41467-017-01973-8">doi:10.1038/s41467-017-01973-8</a>
</p>

<hr>
<h2 id='bdiv_boxplot'>Visualize BIOM data with boxplots.</h2><span id='topic+bdiv_boxplot'></span>

<h3>Description</h3>

<p>Visualize BIOM data with boxplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_boxplot(
  biom,
  x = NULL,
  bdiv = "Bray-Curtis",
  layers = "x",
  weighted = TRUE,
  tree = NULL,
  within = NULL,
  between = NULL,
  stat.by = x,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  patterns = FALSE,
  flip = FALSE,
  stripe = NULL,
  ci = "ci",
  level = 0.95,
  p.adj = "fdr",
  outliers = NULL,
  xlab.angle = "auto",
  p.label = 0.05,
  transform = "none",
  caption = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_boxplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_x">x</code></td>
<td>
<p>A categorical metadata column name to use for the x-axis. Or
<code>NULL</code>, which groups all samples into a single category.</p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("bar", "box" ("x"), "violin", "dot", "strip", "crossbar", "errorbar", "linerange", "pointrange")</code>.
Single letter abbreviations are also accepted. For instance,
<code>c("box", "dot")</code> is equivalent to <code>c("x", "d")</code> and <code>"xd"</code>.
Default: <code>"x"</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_within">within</code>, <code id="bdiv_boxplot_+3A_between">between</code></td>
<td>
<p>Dataset field(s) for intra- or inter- sample
comparisons. Alternatively, dataset field names given elsewhere can
be prefixed with <code>'=='</code> or <code>'!='</code> to assign them to <code>within</code> or
<code>between</code>, respectively. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_patterns">patterns</code></td>
<td>
<p>Patterns for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, pattern
names (<code>"brick"</code>, <code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, etc), or a named
vector mapping groups to specific pattern names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_flip">flip</code></td>
<td>
<p>Transpose the axes, so that taxa are present as rows instead
of columns. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_stripe">stripe</code></td>
<td>
<p>Shade every other x position. Default: <em>same as flip</em></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_ci">ci</code></td>
<td>
<p>How to calculate min/max of the <b>crossbar</b>,
<b>errorbar</b>, <b>linerange</b>, and <b>pointrange</b> layers.
Options are: <code>"ci"</code> (confidence interval), <code>"range"</code>,
<code>"sd"</code> (standard deviation), <code>"se"</code> (standard error), and
<code>"mad"</code> (median absolute deviation).
The center mark of <b>crossbar</b> and <b>pointrange</b> represents
the mean, except for <code>"mad"</code> in which case it represents the median.
Default: <code>"ci"</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_outliers">outliers</code></td>
<td>
<p>Show boxplot outliers? <code>TRUE</code> to always show.
<code>FALSE</code> to always hide. <code>NULL</code> to only hide them when
overlaying a dot or strip chart.  Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_p.label">p.label</code></td>
<td>
<p>Minimum adjusted p-value to display on the plot with a
bracket.
</p>

<dl>
<dt><code>p.label = 0.05</code> - </dt><dd><p> Show p-values that are &lt;= 0.05. </p>
</dd>
<dt><code>p.label = 0</code> - </dt><dd><p> Don't show any p-values on the plot. </p>
</dd>
<dt><code>p.label = 1</code> - </dt><dd><p> Show all p-values on the plot. </p>
</dd>
</dl>

<p>If a numeric vector with more than one value is
provided, they will be used as breaks for asterisk notation.
Default: <code>0.05</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_boxplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>d.size = 2</code> ensures only the points on the
<b>dot</b> layer have their size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Patterns are added using the fillpattern R package. Options are <code>"brick"</code>,
<code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, <code>"herringbone"</code>, <code>"hexagon"</code>, <code>"octagon"</code>,
<code>"rain"</code>, <code>"saw"</code>, <code>"shingle"</code>, <code>"rshingle"</code>, <code>"stripe"</code>, and <code>"wave"</code>,
optionally abbreviated and/or suffixed with modifiers. For example,
<code>"hex10_sm"</code> for the hexagon pattern rotated 10 degrees and shrunk by 2x.
See <code><a href="fillpattern.html#topic+fill_pattern">fillpattern::fill_pattern()</a></code> for complete documentation of options.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    bdiv_boxplot(biom, x="==Body Site", bdiv="UniFrac", stat.by="Body Site")
</code></pre>

<hr>
<h2 id='bdiv_clusters'>Cluster samples by beta diversity k-means.</h2><span id='topic+bdiv_clusters'></span>

<h3>Description</h3>

<p>Cluster samples by beta diversity k-means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_clusters(
  biom,
  bdiv = "Bray-Curtis",
  weighted = TRUE,
  normalized = TRUE,
  tree = NULL,
  k = 5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_clusters_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_clusters_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_clusters_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_clusters_+3A_normalized">normalized</code></td>
<td>
<p>Only changes the &quot;Weighted UniFrac&quot; calculation.
Divides result by the total branch weights. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_clusters_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_clusters_+3A_k">k</code></td>
<td>
<p>Number of clusters. Default: <code>5L</code></p>
</td></tr>
<tr><td><code id="bdiv_clusters_+3A_...">...</code></td>
<td>
<p>Passed on to <code>stats::kmeans()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric factor assigning samples to clusters.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other clustering: 
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    biom$metadata$bray_cluster &lt;- bdiv_clusters(biom)
    
    pull(biom, 'bray_cluster')[1:10]
    
    bdiv_ord_plot(biom, stat.by = "bray_cluster")
</code></pre>

<hr>
<h2 id='bdiv_corrplot'>Visualize beta diversity with scatterplots and trendlines.</h2><span id='topic+bdiv_corrplot'></span>

<h3>Description</h3>

<p>Visualize beta diversity with scatterplots and trendlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_corrplot(
  biom,
  x,
  bdiv = "Bray-Curtis",
  layers = "tc",
  weighted = TRUE,
  tree = NULL,
  within = NULL,
  between = NULL,
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  p.adj = "fdr",
  transform = "none",
  ties = "random",
  seed = 0,
  alt = "!=",
  mu = 0,
  caption = TRUE,
  check = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_corrplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_x">x</code></td>
<td>
<p>Dataset field with the x-axis values. Equivalent to the <code>regr</code>
argument in <code><a href="#topic+stats_table">stats_table()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("trend", "confidence", "point", "name", "residual")</code>. Single
letter abbreviations are also accepted. For instance,
<code>c("trend", "point")</code> is equivalent to <code>c("t", "p")</code> and <code>"tp"</code>.
Default: <code>"tc"</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_within">within</code>, <code id="bdiv_corrplot_+3A_between">between</code></td>
<td>
<p>Dataset field(s) for intra- or inter- sample
comparisons. Alternatively, dataset field names given elsewhere can
be prefixed with <code>'=='</code> or <code>'!='</code> to assign them to <code>within</code> or
<code>between</code>, respectively. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'none'</code>, <code>'emmeans'</code>, or
<code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_ties">ties</code></td>
<td>
<p>When <code>transform="rank"</code>, how to rank identical values.
Options are: <code>c("average", "first", "last", "random", "max", "min")</code>.
See <code>rank()</code> for details. Default: <code>"random"</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bdiv_corrplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>p.size = 2</code> ensures only the points have their
size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    bdiv_corrplot(biom, "Age", stat.by = "Sex", layers = "tcp")
</code></pre>

<hr>
<h2 id='bdiv_heatmap'>Display beta diversities in an all vs all grid.</h2><span id='topic+bdiv_heatmap'></span>

<h3>Description</h3>

<p>Display beta diversities in an all vs all grid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_heatmap(
  biom,
  bdiv = "Bray-Curtis",
  weighted = TRUE,
  tree = NULL,
  tracks = NULL,
  grid = "devon",
  label = TRUE,
  label_size = NULL,
  rescale = "none",
  clust = "complete",
  trees = TRUE,
  asp = 1,
  tree_height = 10,
  track_height = 10,
  legend = "right",
  title = TRUE,
  xlab.angle = "auto",
  underscores = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_heatmap_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_tracks">tracks</code></td>
<td>
<p>A character vector of metadata fields to display as tracks
at the top of the plot. Or, a list as expected by the <code>tracks</code>
argument of <code><a href="#topic+plot_heatmap">plot_heatmap()</a></code>. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_grid">grid</code></td>
<td>
<p>Color palette name, or a list with entries for <code>label</code>,
<code>colors</code>, <code>range</code>, <code>bins</code>, <code>na.color</code>, and/or
<code>guide</code>. See the Track Definitions section for details.
Default: <code>"devon"</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_label">label</code></td>
<td>
<p>Label the matrix rows and columns. You can supply a list
or logical vector of length two to control row labels and column
labels separately, for example
<code>label = c(rows = TRUE, cols = FALSE)</code>, or simply
<code>label = c(TRUE, FALSE)</code>. Other valid options are <code>"rows"</code>,
<code>"cols"</code>, <code>"both"</code>, <code>"bottom"</code>, <code>"right"</code>,
and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_label_size">label_size</code></td>
<td>
<p>The font size to use for the row and column labels. You
can supply a numeric vector of length two to control row label sizes
and column label sizes separately, for example
<code>c(rows = 20, cols = 8)</code>, or simply <code>c(20, 8)</code>.
Default: <code>NULL</code>, which computes:
<code>pmax(8, pmin(20, 100 / dim(mtx)))</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_rescale">rescale</code></td>
<td>
<p>Rescale rows or columns to all have a common min/max.
Options: <code>"none"</code>, <code>"rows"</code>, or <code>"cols"</code>.
Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_clust">clust</code></td>
<td>
<p>Clustering algorithm for reordering the rows and columns by
similarity. You can supply a list or character vector of length two to
control the row and column clustering separately, for example
<code>clust = c(rows = "complete", cols = NA)</code>, or simply
<code>clust = c("complete", NA)</code>. Options are:
</p>

<dl>
<dt><code>FALSE</code> or <code>NA</code> - </dt><dd><p> Disable reordering. </p>
</dd>
<dt>An <code>hclust</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+hclust">stats::hclust()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"ward.D"</code>,
<code>"ward.D2"</code>, <code>"single"</code>, <code>"complete"</code>,
<code>"average"</code>, <code>"mcquitty"</code>, <code>"median"</code>, or
<code>"centroid"</code>. </p>
</dd>
</dl>

<p>Default: <code>"complete"</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_trees">trees</code></td>
<td>
<p>Draw a dendrogram for rows (left) and columns (top). You can
supply a list or logical vector of length two to control the row tree
and column tree separately, for example
<code>trees = c(rows = TRUE, cols = FALSE)</code>,
or simply <code>trees = c(TRUE, FALSE)</code>.
Other valid options are <code>"rows"</code>, <code>"cols"</code>, <code>"both"</code>,
<code>"left"</code>, <code>"top"</code>, and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio (height/width) for entire grid.
Default: <code>1</code> (square)</p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_tree_height">tree_height</code>, <code id="bdiv_heatmap_+3A_track_height">track_height</code></td>
<td>
<p>The height of the dendrogram or annotation
tracks as a percentage of the overall grid size. Use a numeric vector
of length two to assign <code>c(top, left)</code> independently.
Default: <code>10</code> (10% of the grid's height)</p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_legend">legend</code></td>
<td>
<p>Where to place the legend. Options are: <code>"right"</code> or
<code>"bottom"</code>. Default: <code>"right"</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_title">title</code></td>
<td>
<p>Plot title. Set to <code>TRUE</code> for a default title, <code>NULL</code> for
no title, or any character string. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bdiv_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to ggplot2::theme().
For example, <code>labs.subtitle = "Plot subtitle"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>


<h3>Annotation Tracks</h3>

<p>Metadata can be displayed as colored tracks above the heatmap. Common use
cases are provided below, with more thorough documentation available at
https://cmmr.github.io/rbiom .
</p>
<pre>## Categorical ----------------------------
tracks = "Body Site"
tracks = list('Body Site' = "bright")
tracks = list('Body Site' = c('Stool' = "blue", 'Saliva' = "green"))

## Numeric --------------------------------
tracks = "Age"
tracks = list('Age' = "reds")

## Multiple Tracks ------------------------
tracks = c("Body Site", "Age")
tracks = list('Body Site' = "bright", 'Age' = "reds")
tracks = list(
  'Body Site' = c('Stool' = "blue", 'Saliva' = "green"),
  'Age'       = list('colors' = "reds") )
</pre>
<p>The following entries in the track definitions are understood:
</p>

<dl>
<dt><code>colors</code> - </dt><dd><p> A pre-defined palette name or custom set of colors to map to. </p>
</dd>
<dt><code>range</code> - </dt><dd><p> The c(min,max) to use for scale values. </p>
</dd>
<dt><code>label</code> - </dt><dd><p> Label for this track. Defaults to the name of this list element. </p>
</dd>
<dt><code>side</code> - </dt><dd><p> Options are <code>"top"</code> (default) or <code>"left"</code>. </p>
</dd>
<dt><code>na.color</code> - </dt><dd><p> The color to use for <code>NA</code> values. </p>
</dd>
<dt><code>bins</code> - </dt><dd><p> Bin a gradient into this many bins/steps. </p>
</dd>
<dt><code>guide</code> - </dt><dd><p> A list of arguments for guide_colorbar() or guide_legend(). </p>
</dd>
</dl>

<p>All built-in color palettes are colorblind-friendly.
</p>
<p>Categorical palette names: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Numeric palette names: <code>"reds"</code>, <code>"oranges"</code>, <code>"greens"</code>,
<code>"purples"</code>, <code>"grays"</code>, <code>"acton"</code>, <code>"bamako"</code>,
<code>"batlow"</code>, <code>"bilbao"</code>, <code>"buda"</code>, <code>"davos"</code>,
<code>"devon"</code>, <code>"grayC"</code>, <code>"hawaii"</code>, <code>"imola"</code>,
<code>"lajolla"</code>, <code>"lapaz"</code>, <code>"nuuk"</code>, <code>"oslo"</code>,
<code>"tokyo"</code>, <code>"turku"</code>, <code>"bam"</code>, <code>"berlin"</code>,
<code>"broc"</code>, <code>"cork"</code>, <code>"lisbon"</code>, <code>"roma"</code>,
<code>"tofino"</code>, <code>"vanimo"</code>, and <code>"vik"</code>.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # Keep and rarefy the 10 most deeply sequenced samples.
    hmp10 &lt;- rarefy(hmp50, n = 10)
    
    bdiv_heatmap(hmp10, tracks=c("Body Site", "Age"))
    
    bdiv_heatmap(hmp10, bdiv="uni", weighted=c(TRUE,FALSE), tracks="sex")
</code></pre>

<hr>
<h2 id='bdiv_ord_plot'>Ordinate samples and taxa on a 2D plane based on beta diversity distances.</h2><span id='topic+bdiv_ord_plot'></span>

<h3>Description</h3>

<p>Ordinate samples and taxa on a 2D plane based on beta diversity distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_ord_plot(
  biom,
  bdiv = "Bray-Curtis",
  ord = "PCoA",
  weighted = TRUE,
  layers = "petm",
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  tree = NULL,
  test = "adonis2",
  seed = 0,
  permutations = 999,
  rank = -1,
  taxa = 4,
  p.top = Inf,
  p.adj = "fdr",
  unc = "singly",
  caption = TRUE,
  underscores = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_ord_plot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_ord">ord</code></td>
<td>
<p>Method for reducing dimensionality. Options are:
</p>

<dl>
<dt><code>"PCoA"</code> - </dt><dd><p> Principal coordinate analysis; <code><a href="ape.html#topic+pcoa">ape::pcoa()</a></code>. </p>
</dd>
<dt><code>"UMAP"</code> - </dt><dd><p> Uniform manifold approximation and projection; <code><a href="uwot.html#topic+umap">uwot::umap()</a></code>. </p>
</dd>
<dt><code>"NMDS"</code> - </dt><dd><p> Nonmetric multidimensional scaling; <code><a href="vegan.html#topic+metaMDS">vegan::metaMDS()</a></code>. </p>
</dd>
<dt><code>"tSNE"</code> - </dt><dd><p> t-distributed stochastic neighbor embedding; <code><a href="tsne.html#topic+tsne">tsne::tsne()</a></code>. </p>
</dd>
</dl>

<p>Multiple/abbreviated values allowed. Default: <code>"PCoA"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("point", "spider", "ellipse", "name", "mean", "taxon", "arrow")</code>.
The first four are sample-centric; the last three are taxa-centric.
Single letter abbreviations are also accepted. For instance,
<code>c("point", "ellipse")</code> is equivalent to <code>c("p", "e")</code> and <code>"pe"</code>.
Default: <code>"pe"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_stat.by">stat.by</code></td>
<td>
<p>The categorical or numeric metadata field over which
statistics should be calculated. Required.</p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_test">test</code></td>
<td>
<p>Permutational test for accessing significance. Options are:
</p>

<dl>
<dt><code>"adonis2"</code> - </dt><dd><p> Permutational MANOVA; <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code>. </p>
</dd>
<dt><code>"mrpp"</code> - </dt><dd><p> Multiple response permutation procedure; <code><a href="vegan.html#topic+mrpp">vegan::mrpp()</a></code>. </p>
</dd>
<dt><code>"none"</code> - </dt><dd><p> Don't run any statistics. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"adonis2"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_permutations">permutations</code></td>
<td>
<p>Number of random permutations to use.
Default: <code>999</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_p.top">p.top</code></td>
<td>
<p>Only display taxa with the most significant differences in
abundance. If <code>p.top</code> is &gt;= 1, then the <code>p.top</code> most
significant taxa are displayed. If <code>p.top</code> is less than one, all
taxa with an adjusted p-value &lt;= <code>p.top</code> are displayed.
Recommended to be used in combination with the <code>taxa</code> parameter
to set a lower bound on the mean abundance of considered taxa.
Default: <code>Inf</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_plot_+3A_...">...</code></td>
<td>
<p>Parameters for layer geoms (e.g. <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>).
Prefixing parameter names with a layer name ensures that a particular
parameter is passed to, and only to, that layer. For instance,
<code>point.size = 2</code> or <code>p.size = 2</code> ensures only the points
have their size set to <code>2</code>. Points can also be controlled with
the <code>pt.</code> prefix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot.
The computed sample coordinates and ggplot command
are available as <code>$data</code> and <code>$code</code> respectively.
If <code>stat.by</code> is given, then <code>$stats</code> and
<code>$stats$code</code> are set.
If <code>rank</code> is given, then <code>$data$taxa_coords</code>,
<code>$taxa_stats</code>, and <code>$taxa_stats$code</code> are set.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other ordination: 
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+distmat_ord_table">distmat_ord_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    bdiv_ord_plot(biom, layers="pemt", stat.by="Body Site", rank="g")
    
</code></pre>

<hr>
<h2 id='bdiv_ord_table'>Calculate PCoA and other ordinations, including taxa biplots and statistics.</h2><span id='topic+bdiv_ord_table'></span>

<h3>Description</h3>

<p>The biplot parameters (<code>taxa</code>, <code>unc</code>, <code>p.top</code>, and
<code>p.adj</code>) only only have an effect when <code>rank</code> is not <code>NULL</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_ord_table(
  biom,
  bdiv = "Bray-Curtis",
  ord = "PCoA",
  weighted = TRUE,
  md = NULL,
  k = 2,
  stat.by = NULL,
  split.by = NULL,
  tree = NULL,
  test = "adonis2",
  seed = 0,
  permutations = 999,
  rank = NULL,
  taxa = 6,
  p.top = Inf,
  p.adj = "fdr",
  unc = "singly",
  underscores = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_ord_table_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_ord">ord</code></td>
<td>
<p>Method for reducing dimensionality. Options are:
</p>

<dl>
<dt><code>"PCoA"</code> - </dt><dd><p> Principal coordinate analysis; <code><a href="ape.html#topic+pcoa">ape::pcoa()</a></code>. </p>
</dd>
<dt><code>"UMAP"</code> - </dt><dd><p> Uniform manifold approximation and projection; <code><a href="uwot.html#topic+umap">uwot::umap()</a></code>. </p>
</dd>
<dt><code>"NMDS"</code> - </dt><dd><p> Nonmetric multidimensional scaling; <code><a href="vegan.html#topic+metaMDS">vegan::metaMDS()</a></code>. </p>
</dd>
<dt><code>"tSNE"</code> - </dt><dd><p> t-distributed stochastic neighbor embedding; <code><a href="tsne.html#topic+tsne">tsne::tsne()</a></code>. </p>
</dd>
</dl>

<p>Multiple/abbreviated values allowed. Default: <code>"PCoA"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_md">md</code></td>
<td>
<p>Dataset field(s) to include in the output data frame, or <code>'.all'</code>
to include all metadata fields. Default: <code>'.all'</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_k">k</code></td>
<td>
<p>Number of ordination dimensions to return. Either <code>2L</code> or
<code>3L</code>. Default: <code>2L</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_stat.by">stat.by</code></td>
<td>
<p>The categorical or numeric metadata field over which
statistics should be calculated. Required.</p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_split.by">split.by</code></td>
<td>
<p>Dataset field(s) that the data should be split by prior to
any calculations. Must be categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_test">test</code></td>
<td>
<p>Permutational test for accessing significance. Options are:
</p>

<dl>
<dt><code>"adonis2"</code> - </dt><dd><p> Permutational MANOVA; <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code>. </p>
</dd>
<dt><code>"mrpp"</code> - </dt><dd><p> Multiple response permutation procedure; <code><a href="vegan.html#topic+mrpp">vegan::mrpp()</a></code>. </p>
</dd>
<dt><code>"none"</code> - </dt><dd><p> Don't run any statistics. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"adonis2"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_permutations">permutations</code></td>
<td>
<p>Number of random permutations to use.
Default: <code>999</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to compute biplot coordinates and
statistics for, or <code>NULL</code> to disable. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_p.top">p.top</code></td>
<td>
<p>Only display taxa with the most significant differences in
abundance. If <code>p.top</code> is &gt;= 1, then the <code>p.top</code> most
significant taxa are displayed. If <code>p.top</code> is less than one, all
taxa with an adjusted p-value &lt;= <code>p.top</code> are displayed.
Recommended to be used in combination with the <code>taxa</code> parameter
to set a lower bound on the mean abundance of considered taxa.
Default: <code>Inf</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bdiv_ord_table_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to <code><a href="uwot.html#topic+umap">uwot::umap()</a></code>,
<code><a href="ape.html#topic+pcoa">ape::pcoa()</a></code>, <code><a href="vegan.html#topic+metaMDS">vegan::metaMDS()</a></code>, or <code><a href="tsne.html#topic+tsne">tsne::tsne()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with columns <code>.sample</code>, <code>.weighted</code>,
<code>.bdiv</code>, <code>.ord</code>, <code>.x</code>, <code>.y</code>, and (optionally)
<code>.z</code>. Any columns given by <code>md</code>, <code>split.by</code>, and
<code>stat.by</code> are included as well.
If <code>stat.by</code> is given, then <code>$stats</code> and
<code>$stats$code)</code> are set.
If <code>rank</code> is given, then <code>$taxa_coords</code>,
<code>$taxa_stats</code>, and <code>$taxa_stats$code</code> are set.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other ordination: 
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+distmat_ord_table">distmat_ord_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    ord &lt;- bdiv_ord_table(hmp50, "bray", "pcoa", stat.by="Body Site", rank="g")
    head(ord)
    
    ord$stats
    
    ord$taxa_stats
    
    
</code></pre>

<hr>
<h2 id='bdiv_stats'>Test beta diversity for associations with metadata.</h2><span id='topic+bdiv_stats'></span>

<h3>Description</h3>

<p>A convenience wrapper for <code><a href="#topic+bdiv_table">bdiv_table()</a></code> + <code><a href="#topic+stats_table">stats_table()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_stats(
  biom,
  regr = NULL,
  stat.by = NULL,
  bdiv = "Bray-Curtis",
  weighted = TRUE,
  tree = NULL,
  within = NULL,
  between = NULL,
  split.by = NULL,
  transform = "none",
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  alt = "!=",
  mu = 0,
  p.adj = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_stats_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_regr">regr</code></td>
<td>
<p>Dataset field with the x-axis (independent; predictive)
values. Must be numeric. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_within">within</code>, <code id="bdiv_stats_+3A_between">between</code></td>
<td>
<p>Dataset field(s) for intra- or inter- sample
comparisons. Alternatively, dataset field names given elsewhere can
be prefixed with <code>'=='</code> or <code>'!='</code> to assign them to <code>within</code> or
<code>between</code>, respectively. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_split.by">split.by</code></td>
<td>
<p>Dataset field(s) that the data should be split by prior to
any calculations. Must be categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'wilcox'</code>, <code>'kruskal'</code>,
<code>'emmeans'</code>, or <code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="bdiv_stats_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble data.frame with fields from the table below. This tibble
object provides the <code style="white-space: pre;">&#8288;$code&#8288;</code> operator to print the R code used to generate
the statistics.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Field</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean </td><td style="text-align: left;"> Estimated marginal mean. See <code><a href="emmeans.html#topic+emmeans">emmeans::emmeans()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean.diff </td><td style="text-align: left;"> Difference in means. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope </td><td style="text-align: left;"> Trendline slope. See <code><a href="emmeans.html#topic+emtrends">emmeans::emtrends()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope.diff </td><td style="text-align: left;"> Difference in slopes. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .h1 </td><td style="text-align: left;"> Alternate hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .p.val </td><td style="text-align: left;"> Probability that null hypothesis is correct. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.p </td><td style="text-align: left;"> <code>.p.val</code> after adjusting for multiple comparisons. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .effect.size </td><td style="text-align: left;"> Effect size. See <code><a href="emmeans.html#topic+eff_size">emmeans::eff_size()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .lower </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .upper </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .se </td><td style="text-align: left;"> Standard error. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .n </td><td style="text-align: left;"> Number of samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .df </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .stat </td><td style="text-align: left;"> Wilcoxon or Kruskal-Wallis rank sum statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .t.ratio </td><td style="text-align: left;"> <code>.mean</code> / <code>.se</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .r.sqr </td><td style="text-align: left;"> Percent of variation explained by the model. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.r </td><td style="text-align: left;"> <code>.r.sqr</code>, taking degrees of freedom into account. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .aic </td><td style="text-align: left;"> Akaike Information Criterion (predictive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .bic </td><td style="text-align: left;"> Bayesian Information Criterion (descriptive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .loglik </td><td style="text-align: left;"> Log-likelihood goodness-of-fit score. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .fit.p </td><td style="text-align: left;"> P-value for observing this fit by chance. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>
<p>Other stats_tables: 
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>,
<code><a href="#topic+stats_table">stats_table</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
      
    bdiv_stats(biom, stat.by = "Sex", bdiv = c("bray", "unifrac"))[,1:7]
    
    biom &lt;- subset(biom, `Body Site` %in% c('Saliva', 'Stool', 'Buccal mucosa'))
    bdiv_stats(biom, stat.by = "Body Site", split.by = "==Sex")[,1:6]
</code></pre>

<hr>
<h2 id='bdiv_table'>Distance / dissimilarity between samples.</h2><span id='topic+bdiv_table'></span><span id='topic+bdiv_matrix'></span><span id='topic+bdiv_distmat'></span>

<h3>Description</h3>

<p>Distance / dissimilarity between samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdiv_table(
  biom,
  bdiv = "Bray-Curtis",
  weighted = TRUE,
  normalized = TRUE,
  tree = NULL,
  md = ".all",
  within = NULL,
  between = NULL,
  delta = ".all",
  transform = "none",
  ties = "random",
  seed = 0,
  cpus = NULL
)

bdiv_matrix(
  biom,
  bdiv = "Bray-Curtis",
  weighted = TRUE,
  normalized = TRUE,
  tree = NULL,
  within = NULL,
  between = NULL,
  transform = "none",
  ties = "random",
  seed = 0,
  cpus = NULL,
  underscores = FALSE
)

bdiv_distmat(
  biom,
  bdiv = "Bray-Curtis",
  weighted = TRUE,
  normalized = TRUE,
  tree = NULL,
  within = NULL,
  between = NULL,
  transform = "none",
  cpus = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdiv_table_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_normalized">normalized</code></td>
<td>
<p>Only changes the &quot;Weighted UniFrac&quot; calculation.
Divides result by the total branch weights. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_md">md</code></td>
<td>
<p>Dataset field(s) to include in the output data frame, or <code>'.all'</code>
to include all metadata fields. Default: <code>'.all'</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_within">within</code>, <code id="bdiv_table_+3A_between">between</code></td>
<td>
<p>Dataset field(s) for intra- or inter- sample
comparisons. Alternatively, dataset field names given elsewhere can
be prefixed with <code>'=='</code> or <code>'!='</code> to assign them to <code>within</code> or
<code>between</code>, respectively. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_delta">delta</code></td>
<td>
<p>For numeric metadata, report the absolute difference in values
for the two samples, for instance <code>2</code> instead of <code>"10 vs 12"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_ties">ties</code></td>
<td>
<p>When <code>transform="rank"</code>, how to rank identical values.
Options are: <code>c("average", "first", "last", "random", "max", "min")</code>.
See <code>rank()</code> for details. Default: <code>"random"</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_cpus">cpus</code></td>
<td>
<p>The number of CPUs to use. Set to <code>NULL</code> to use all available,
or to <code>1</code> to disable parallel processing. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="bdiv_table_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>


<dl>
<dt><code>bdiv_matrix()</code> - </dt><dd><p> An R matrix of samples x samples. </p>
</dd>
<dt><code>bdiv_distmat()</code> - </dt><dd><p> A dist-class distance matrix. </p>
</dd>
<dt><code>bdiv_table()</code> - </dt><dd>
<p>A tibble data.frame with columns names .sample1, .sample2, .weighted,
.bdiv, .distance, and any fields requested by <code>md</code>. Numeric metadata
fields will be returned as <code>abs(x - y)</code>; categorical metadata fields as
<code>"x"</code>, <code>"y"</code>, or <code>"x vs y"</code>. </p>
</dd>
</dl>



<h3>Metadata Comparisons</h3>

<p>Prefix metadata fields with <code>==</code> or <code>!=</code> to limit comparisons to within or
between groups, respectively. For example, <code>stat.by = '==Sex'</code> will
run calculations only for intra-group comparisons, returning &quot;Male&quot; and
&quot;Female&quot;, but NOT &quot;Female vs Male&quot;. Similarly, setting
<code>stat.by = '!=Body Site'</code> will only show the inter-group comparisons, such
as &quot;Saliva vs Stool&quot;, &quot;Anterior nares vs Buccal mucosa&quot;, and so on.
</p>
<p>The same effect can be achieved by using the <code>within</code> and <code>between</code>
parameters. <code>stat.by = '==Sex'</code> is equivalent to
<code style="white-space: pre;">&#8288;stat.by = 'Sex', within = 'Sex'&#8288;</code>.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # Subset to four samples
    biom &lt;- hmp50$clone()
    biom$counts &lt;- biom$counts[,c("HMP18", "HMP19", "HMP20", "HMP21")]
    
    # Return in long format with metadata
    bdiv_table(biom, 'unifrac', md = ".all")
    
    # Only look at distances among the stool samples
    bdiv_table(biom, 'unifrac', md = c("==Body Site", "Sex"))
    
    # Or between males and females
    bdiv_table(biom, 'unifrac', md = c("Body Site", "!=Sex"))
    
    # All-vs-all matrix
    bdiv_matrix(biom, 'unifrac')
    
    # All-vs-all distance matrix
    dm &lt;- bdiv_distmat(biom, 'unifrac')
    dm
    plot(hclust(dm))

</code></pre>

<hr>
<h2 id='bdply'>Apply a function to each subset of an rbiom object.</h2><span id='topic+bdply'></span><span id='topic+blply'></span>

<h3>Description</h3>

<p><code>blply()</code> and <code>bdply()</code> let you divide your biom dataset into smaller
pieces, run a function on those smaller rbiom objects, and return the
results as a data.frame or list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdply(biom, vars, FUN, ..., iters = list(), prefix = FALSE)

blply(biom, vars, FUN, ..., iters = list(), prefix = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bdply_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="bdply_+3A_vars">vars</code></td>
<td>
<p>A character vector of metadata fields. Each unique combination
of values in these columns will be used to create a subsetted
rbiom object to pass to <code>FUN.</code> If <code>NULL</code>,
<code>biom</code> will be passed to <code>FUN</code> unaltered. Unambiguous
abbreviations of metadata fields are also accepted.</p>
</td></tr>
<tr><td><code id="bdply_+3A_fun">FUN</code></td>
<td>
<p>The function to execute on each subset of <code>biom</code>.
For <code>bdply()</code>, the returned value will be coerced to a data.frame.
For <code>blply()</code>, any returned value is unmodified.</p>
</td></tr>
<tr><td><code id="bdply_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to <code>FUN</code>.</p>
</td></tr>
<tr><td><code id="bdply_+3A_iters">iters</code></td>
<td>
<p>A named list of values to pass to <code>FUN</code>. Unlike
<code>...</code>, these will be iterated over in all combinations.
Default: <code>list()</code></p>
</td></tr>
<tr><td><code id="bdply_+3A_prefix">prefix</code></td>
<td>
<p>When <code>TRUE</code>, prefixes the names in in <code>iters</code> with
a '.' in the final data.frame or 'split_labels' attribute.
Default: <code>FALSE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can also specify additional variables for your function to iterate over
in unique combinations.
</p>
<p>Calls <code><a href="plyr.html#topic+ddply">plyr::ddply()</a></code> or <code><a href="plyr.html#topic+dlply">plyr::dlply()</a></code> internally.
</p>


<h3>Value</h3>

<p>For <code>bdply()</code>, a tibble data.frame comprising the accumulated
outputs of <code>FUN</code>, along with the columns specified by
<code>vars</code> and <code>iters</code>. For <code>blply()</code>, a named list that has details
about <code>vars</code> and <code>iters</code> in <code>attr(,'split_labels')</code>.
</p>


<h3>See Also</h3>

<p>Other metadata: 
<code><a href="#topic+glimpse.rbiom">glimpse.rbiom</a>()</code>
</p>
<p>Other biom: 
<code><a href="#topic+biom_merge">biom_merge</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    bdply(hmp50, "Sex", `$`, 'n_samples')
    
    blply(hmp50, "Sex", `$`, 'n_samples') %&gt;% unlist()
    
    bdply(hmp50, c("Body Site", "Sex"), function (b) {
      adm &lt;- adiv_matrix(b)[,c("Shannon", "Simpson")]
      apply(adm, 2L, mean)
    })
    
    iters &lt;- list(w = c(TRUE, FALSE), d = c("bray", "euclid"))
    bdply(hmp50, "Sex", iters = iters, function (b, w, d) {
      r &lt;- range(bdiv_distmat(biom = b, bdiv = d, weighted = w))
      round(data.frame(min = r[[1]], max = r[[2]]))
    })

</code></pre>

<hr>
<h2 id='biom_merge'>Combine several rbiom objects into one.</h2><span id='topic+biom_merge'></span>

<h3>Description</h3>

<p>WARNING: It is generally ill-advised to merge BIOM datasets, as OTUs
mappings are dependent on upstream clustering and are not equivalent
between BIOM files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biom_merge(
  ...,
  metadata = NA,
  taxonomy = NA,
  tree = NULL,
  sequences = NA,
  id = NA,
  comment = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="biom_merge_+3A_...">...</code></td>
<td>
<p>Any number of rbiom objects (e.g. from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>), lists of
rbiom objects, or valid arguments to the <code>biom</code> parameter of
<code><a href="#topic+as_rbiom">as_rbiom()</a></code> (for instance file names).</p>
</td></tr>
<tr><td><code id="biom_merge_+3A_metadata">metadata</code>, <code id="biom_merge_+3A_taxonomy">taxonomy</code>, <code id="biom_merge_+3A_tree">tree</code>, <code id="biom_merge_+3A_sequences">sequences</code>, <code id="biom_merge_+3A_id">id</code>, <code id="biom_merge_+3A_comment">comment</code></td>
<td>
<p>Replace the corresponding
data in the merged rbiom object with these values. Set to <code>NULL</code> to
not inherit a particular component. The default, <code>NA</code>, will attempt
to create the component based on <code>...</code> values. The merged
phylogenetic tree cannot be inferred.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>See Also</h3>

<p>Other biom: 
<code><a href="#topic+bdply">bdply</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    b1 &lt;- as_rbiom(hmp50$counts[,1:4])
    b2 &lt;- as_rbiom(hmp50$counts[,5:8])
    
    biom &lt;- biom_merge(b1, b2)
    print(biom)
    
    biom$tree     &lt;- hmp50$tree
    biom$metadata &lt;- hmp50$metadata
    print(biom)
</code></pre>

<hr>
<h2 id='convert_to'>Convert biom data to an external package class.</h2><span id='topic+convert_to'></span><span id='topic+convert_to_SE'></span><span id='topic+convert_to_TSE'></span><span id='topic+convert_to_phyloseq'></span>

<h3>Description</h3>

<p>Requires the relevant Bioconductor R package to be installed:
</p>

<dl>
<dt><code>convert_to_phyloseq</code> - </dt><dd> <p><a href="https://bioconductor.org/packages/phyloseq/">phyloseq</a> </p>
</dd>
<dt><code>convert_to_SE</code> - </dt><dd> <p><a href="https://bioconductor.org/packages/SummarizedExperiment/">SummarizedExperiment</a> </p>
</dd>
<dt><code>convert_to_TSE</code> - </dt><dd> <p><a href="https://bioconductor.org/packages/TreeSummarizedExperiment/">TreeSummarizedExperiment</a> </p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>convert_to_SE(biom, ...)

convert_to_TSE(biom, ...)

convert_to_phyloseq(biom, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_to_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="convert_to_+3A_...">...</code></td>
<td>
<p>Not Used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A SummarizedExperiment object includes counts, metadata, and taxonomy.
</p>
<p>phyloseq and TreeSummarizedExperiment additionally includes the tree and sequences.
</p>


<h3>Value</h3>

<p>A phyloseq, SummarizedExperiment, or TreeSummarizedExperiment object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
    library(rbiom)

    print(hmp50)

    # Requires 'phyloseq', a Bioconductor R package
    if (nzchar(system.file(package = "phyloseq"))) {
      physeq &lt;- convert_to_phyloseq(hmp50)
      print(physeq)
    }

    # Requires 'SummarizedExperiment', a Bioconductor R package
    if (nzchar(system.file(package = "SummarizedExperiment"))) {
      se &lt;- convert_to_SE(hmp50)
      print(se)
    }

    # Requires 'TreeSummarizedExperiment', a Bioconductor R package
    if (nzchar(system.file(package = "TreeSummarizedExperiment"))) {
      tse &lt;- convert_to_TSE(hmp50)
      print(tse)
    }

## End(Not run)
</code></pre>

<hr>
<h2 id='distmat_ord_table'>Run ordinations on a distance matrix.</h2><span id='topic+distmat_ord_table'></span>

<h3>Description</h3>

<p>Run ordinations on a distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distmat_ord_table(dm, ord = "PCoA", k = 2L, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="distmat_ord_table_+3A_dm">dm</code></td>
<td>
<p>A <code>dist</code>-class distance matrix, as returned from
<code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code> or <code><a href="stats.html#topic+dist">stats::dist()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="distmat_ord_table_+3A_ord">ord</code></td>
<td>
<p>Method for reducing dimensionality. Options are:
</p>

<dl>
<dt><code>"PCoA"</code> - </dt><dd><p> Principal coordinate analysis; <code><a href="ape.html#topic+pcoa">ape::pcoa()</a></code>. </p>
</dd>
<dt><code>"UMAP"</code> - </dt><dd><p> Uniform manifold approximation and projection; <code><a href="uwot.html#topic+umap">uwot::umap()</a></code>. </p>
</dd>
<dt><code>"NMDS"</code> - </dt><dd><p> Nonmetric multidimensional scaling; <code><a href="vegan.html#topic+metaMDS">vegan::metaMDS()</a></code>. </p>
</dd>
<dt><code>"tSNE"</code> - </dt><dd><p> t-distributed stochastic neighbor embedding; <code><a href="tsne.html#topic+tsne">tsne::tsne()</a></code>. </p>
</dd>
</dl>

<p>Multiple/abbreviated values allowed. Default: <code>"PCoA"</code></p>
</td></tr>
<tr><td><code id="distmat_ord_table_+3A_k">k</code></td>
<td>
<p>Number of ordination dimensions to return. Either <code>2L</code> or
<code>3L</code>. Default: <code>2L</code></p>
</td></tr>
<tr><td><code id="distmat_ord_table_+3A_...">...</code></td>
<td>
<p>Additional arguments for <code>ord</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with columns <code>.sample</code>, <code>.ord</code>, <code>.x</code>,
<code>.y</code>, and (optionally) <code>.z</code>.
</p>


<h3>See Also</h3>

<p>Other ordination: 
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    dm  &lt;- bdiv_distmat(hmp50, "bray")
    ord &lt;- distmat_ord_table(dm, "PCoA")
    head(ord)
    
</code></pre>

<hr>
<h2 id='distmat_stats'>Run statistics on a distance matrix vs a categorical or numeric variable.</h2><span id='topic+distmat_stats'></span>

<h3>Description</h3>

<p>Run statistics on a distance matrix vs a categorical or numeric variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distmat_stats(dm, groups, test = "adonis2", seed = 0, permutations = 999)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="distmat_stats_+3A_dm">dm</code></td>
<td>
<p>A <code>dist</code>-class distance matrix, as returned from
<code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code> or <code><a href="stats.html#topic+dist">stats::dist()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="distmat_stats_+3A_groups">groups</code></td>
<td>
<p>A named vector of grouping values. The names should
correspond to <code>attr(dm, 'Labels')</code>. Values can be either
categorical or numeric. Required.</p>
</td></tr>
<tr><td><code id="distmat_stats_+3A_test">test</code></td>
<td>
<p>Permutational test for accessing significance. Options are:
</p>

<dl>
<dt><code>"adonis2"</code> - </dt><dd><p> Permutational MANOVA; <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code>. </p>
</dd>
<dt><code>"mrpp"</code> - </dt><dd><p> Multiple response permutation procedure; <code><a href="vegan.html#topic+mrpp">vegan::mrpp()</a></code>. </p>
</dd>
<dt><code>"none"</code> - </dt><dd><p> Don't run any statistics. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"adonis2"</code></p>
</td></tr>
<tr><td><code id="distmat_stats_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="distmat_stats_+3A_permutations">permutations</code></td>
<td>
<p>Number of random permutations to use.
Default: <code>999</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with summary statistics from <code><a href="vegan.html#topic+permustats">vegan::permustats()</a></code>.
The columns are:
</p>

<dl>
<dt><em>.n</em> - </dt><dd><p> The size of the distance matrix. </p>
</dd>
<dt><em>.stat</em> - </dt><dd>
<p>The observed statistic. For mrpp, this is the overall
weighted mean of group mean distances. </p>
</dd>
<dt><em>.z</em> - </dt><dd>
<p>The difference of observed statistic and mean of permutations
divided by the standard deviation of permutations (also known
as z-values). Evaluated from permuted values without observed
statistic. </p>
</dd>
<dt><em>.p.val</em> - </dt><dd><p> Probability calculated by <code>test</code>. </p>
</dd>
</dl>

<p>R commands for reproducing the results are in <code>$code</code>.
</p>


<h3>See Also</h3>

<p>Other beta_diversity: 
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+bdiv_ord_table">bdiv_ord_table</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+bdiv_table">bdiv_table</a>()</code>
</p>
<p>Other stats_tables: 
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+stats_table">stats_table</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    hmp10        &lt;- hmp50$clone()
    hmp10$counts &lt;- hmp10$counts[,1:10]
    
    dm &lt;- bdiv_distmat(hmp10, 'unifrac')
    
    distmat_stats(dm, groups = pull(hmp10, 'Body Site'))
    
    distmat_stats(dm, groups = pull(hmp10, 'Age'))
    
    # See the R code used to calculate these statistics:
    stats &lt;- distmat_stats(dm, groups = pull(hmp10, 'Age'))
    stats$code

</code></pre>

<hr>
<h2 id='documentation_biom.rbiom'>documentation_biom.rbiom</h2><span id='topic+documentation_biom.rbiom'></span>

<h3>Description</h3>

<p>documentation_biom.rbiom
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_biom.rbiom_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="documentation_biom.rbiom_+3A_.data">.data</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="documentation_biom.rbiom_+3A_x">x</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="documentation_biom.rbiom_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="documentation_biom.rbiom_+3A_data">data</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='documentation_clusters'>documentation_clusters</h2><span id='topic+documentation_clusters'></span>

<h3>Description</h3>

<p>documentation_clusters
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_clusters_+3A_k">k</code></td>
<td>
<p>Number of clusters. Default: <code>5L</code></p>
</td></tr>
<tr><td><code id="documentation_clusters_+3A_rank">rank</code></td>
<td>
<p>Which taxa rank to use. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>.otu</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric factor assigning samples to clusters.
</p>

<hr>
<h2 id='documentation_cmp'>documentation_cmp</h2><span id='topic+documentation_cmp'></span>

<h3>Description</h3>

<p>documentation_cmp
</p>


<h3>Metadata Comparisons</h3>

<p>Prefix metadata fields with <code>==</code> or <code>!=</code> to limit comparisons to within or
between groups, respectively. For example, <code>stat.by = '==Sex'</code> will
run calculations only for intra-group comparisons, returning &quot;Male&quot; and
&quot;Female&quot;, but NOT &quot;Female vs Male&quot;. Similarly, setting
<code>stat.by = '!=Body Site'</code> will only show the inter-group comparisons, such
as &quot;Saliva vs Stool&quot;, &quot;Anterior nares vs Buccal mucosa&quot;, and so on.
</p>
<p>The same effect can be achieved by using the <code>within</code> and <code>between</code>
parameters. <code>stat.by = '==Sex'</code> is equivalent to
<code style="white-space: pre;">&#8288;stat.by = 'Sex', within = 'Sex'&#8288;</code>.
</p>

<hr>
<h2 id='documentation_default'>documentation_default</h2><span id='topic+documentation_default'></span>

<h3>Description</h3>

<p>documentation_default
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_default_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_mtx">mtx</code></td>
<td>
<p>A matrix-like object.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Only required when
computing UniFrac distances. Default: <code>biom$tree</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_md">md</code></td>
<td>
<p>Dataset field(s) to include in the output data frame, or <code>'.all'</code>
to include all metadata fields. Default: <code>'.all'</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_bdiv">bdiv</code></td>
<td>
<p>Beta diversity distance algorithm(s) to use. Options are:
<code>"Bray-Curtis"</code>, <code>"Manhattan"</code>, <code>"Euclidean"</code>,
<code>"Jaccard"</code>, and <code>"UniFrac"</code>. For <code>"UniFrac"</code>, a
phylogenetic tree must be present in <code>biom</code> or explicitly
provided via <code style="white-space: pre;">&#8288;tree=&#8288;</code>. Multiple/abbreviated values allowed.
Default: <code>"Bray-Curtis"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_ord">ord</code></td>
<td>
<p>Method for reducing dimensionality. Options are:
</p>

<dl>
<dt><code>"PCoA"</code> - </dt><dd><p> Principal coordinate analysis; <code><a href="ape.html#topic+pcoa">ape::pcoa()</a></code>. </p>
</dd>
<dt><code>"UMAP"</code> - </dt><dd><p> Uniform manifold approximation and projection; <code><a href="uwot.html#topic+umap">uwot::umap()</a></code>. </p>
</dd>
<dt><code>"NMDS"</code> - </dt><dd><p> Nonmetric multidimensional scaling; <code><a href="vegan.html#topic+metaMDS">vegan::metaMDS()</a></code>. </p>
</dd>
<dt><code>"tSNE"</code> - </dt><dd><p> t-distributed stochastic neighbor embedding; <code><a href="tsne.html#topic+tsne">tsne::tsne()</a></code>. </p>
</dd>
</dl>

<p>Multiple/abbreviated values allowed. Default: <code>"PCoA"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When
<code>weighted=FALSE</code>, only presence/absence is considered.
Multiple values allowed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_normalized">normalized</code></td>
<td>
<p>Only changes the &quot;Weighted UniFrac&quot; calculation.
Divides result by the total branch weights. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_delta">delta</code></td>
<td>
<p>For numeric metadata, report the absolute difference in values
for the two samples, for instance <code>2</code> instead of <code>"10 vs 12"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_sparse">sparse</code></td>
<td>
<p>If <code>TRUE</code>, returns a
<code><a href="slam.html#topic+matrix">slam::simple_triplet_matrix()</a></code>, otherwise returns a
normal R matrix object. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_p.top">p.top</code></td>
<td>
<p>Only display taxa with the most significant differences in
abundance. If <code>p.top</code> is &gt;= 1, then the <code>p.top</code> most
significant taxa are displayed. If <code>p.top</code> is less than one, all
taxa with an adjusted p-value &lt;= <code>p.top</code> are displayed.
Recommended to be used in combination with the <code>taxa</code> parameter
to set a lower bound on the mean abundance of considered taxa.
Default: <code>Inf</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_y.transform">y.transform</code></td>
<td>
<p>The transformation to apply to the y-axis. Visualizing
differences of both high- and low-abundance taxa is best done with
a non-linear axis. Options are:
</p>

<dl>
<dt><code>"sqrt"</code> - </dt><dd><p> square-root transformation </p>
</dd>
<dt><code>"log1p"</code> - </dt><dd><p> log(y + 1) transformation </p>
</dd>
<dt><code>"none"</code> - </dt><dd><p> no transformation </p>
</dd>
</dl>

<p>These methods allow visualization of both high- and low-abundance
taxa simultaneously, without complaint about 'zero' count
observations. Default: <code>"sqrt"</code>
Use <code>xaxis.transform</code> or <code>yaxis.transform</code> to pass custom values
directly to ggplot2's <code style="white-space: pre;">&#8288;scale_*&#8288;</code> functions.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_flip">flip</code></td>
<td>
<p>Transpose the axes, so that taxa are present as rows instead
of columns. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_stripe">stripe</code></td>
<td>
<p>Shade every other x position. Default: <em>same as flip</em></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_ci">ci</code></td>
<td>
<p>How to calculate min/max of the <b>crossbar</b>,
<b>errorbar</b>, <b>linerange</b>, and <b>pointrange</b> layers.
Options are: <code>"ci"</code> (confidence interval), <code>"range"</code>,
<code>"sd"</code> (standard deviation), <code>"se"</code> (standard error), and
<code>"mad"</code> (median absolute deviation).
The center mark of <b>crossbar</b> and <b>pointrange</b> represents
the mean, except for <code>"mad"</code> in which case it represents the median.
Default: <code>"ci"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_p.label">p.label</code></td>
<td>
<p>Minimum adjusted p-value to display on the plot with a
bracket.
</p>

<dl>
<dt><code>p.label = 0.05</code> - </dt><dd><p> Show p-values that are &lt;= 0.05. </p>
</dd>
<dt><code>p.label = 0</code> - </dt><dd><p> Don't show any p-values on the plot. </p>
</dd>
<dt><code>p.label = 1</code> - </dt><dd><p> Show all p-values on the plot. </p>
</dd>
</dl>

<p>If a numeric vector with more than one value is
provided, they will be used as breaks for asterisk notation.
Default: <code>0.05</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_outliers">outliers</code></td>
<td>
<p>Show boxplot outliers? <code>TRUE</code> to always show.
<code>FALSE</code> to always hide. <code>NULL</code> to only hide them when
overlaying a dot or strip chart.  Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_k">k</code></td>
<td>
<p>Number of ordination dimensions to return. Either <code>2L</code> or
<code>3L</code>. Default: <code>2L</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_split.by">split.by</code></td>
<td>
<p>Dataset field(s) that the data should be split by prior to
any calculations. Must be categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_dm">dm</code></td>
<td>
<p>A <code>dist</code>-class distance matrix, as returned from
<code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code> or <code><a href="stats.html#topic+dist">stats::dist()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_groups">groups</code></td>
<td>
<p>A named vector of grouping values. The names should
correspond to <code>attr(dm, 'Labels')</code>. Values can be either
categorical or numeric. Required.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_df">df</code></td>
<td>
<p>The dataset (data.frame or tibble object). &quot;Dataset fields&quot;
mentioned below should match column names in <code>df</code>. Required.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_regr">regr</code></td>
<td>
<p>Dataset field with the x-axis (independent; predictive)
values. Must be numeric. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_resp">resp</code></td>
<td>
<p>Dataset field with the y-axis (dependent; response) values,
such as taxa abundance or alpha diversity.
Default: <code>attr(df, 'response')</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_color.by">color.by</code></td>
<td>
<p>Dataset field with the group to color by. Must be
categorical. Default: <code>stat.by</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_shape.by">shape.by</code></td>
<td>
<p>Dataset field with the group for shapes. Must be
categorical. Default: <code>stat.by</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_patterns">patterns</code></td>
<td>
<p>Patterns for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, pattern
names (<code>"brick"</code>, <code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, etc), or a named
vector mapping groups to specific pattern names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'wilcox'</code>, <code>'kruskal'</code>,
<code>'emmeans'</code>, or <code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="documentation_default_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_within">within</code>, <code id="documentation_default_+3A_between">between</code></td>
<td>
<p>Dataset field(s) for intra- or inter- sample
comparisons. Alternatively, dataset field names given elsewhere can
be prefixed with <code>'=='</code> or <code>'!='</code> to assign them to <code>within</code> or
<code>between</code>, respectively. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_cpus">cpus</code></td>
<td>
<p>The number of CPUs to use. Set to <code>NULL</code> to use all available,
or to <code>1</code> to disable parallel processing. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_permutations">permutations</code></td>
<td>
<p>Number of random permutations to use.
Default: <code>999</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_depths">depths</code></td>
<td>
<p>Rarefaction depths to show in the plot, or <code>NULL</code> to
auto-select. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_rline">rline</code></td>
<td>
<p>Where to draw a horizontal line on the plot, intended to show
a particular rarefaction depth. Set to <code>TRUE</code> to show an
auto-selected rarefaction depth or <code>FALSE</code> to not show a line.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_clone">clone</code></td>
<td>
<p>Create a copy of <code>biom</code> before modifying. If <code>FALSE</code>, <code>biom</code>
is modified in place as a side-effect. See <a href="#topic+speed">speed ups</a> for
use cases. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_labels">labels</code></td>
<td>
<p>Show sample names under each bar. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="documentation_default_+3A_ties">ties</code></td>
<td>
<p>When <code>transform="rank"</code>, how to rank identical values.
Options are: <code>c("average", "first", "last", "random", "max", "min")</code>.
See <code>rank()</code> for details. Default: <code>"random"</code></p>
</td></tr>
</table>

<hr>
<h2 id='documentation_dist_test'>documentation_dist_test</h2><span id='topic+documentation_dist_test'></span>

<h3>Description</h3>

<p>documentation_dist_test
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_dist_test_+3A_stat.by">stat.by</code></td>
<td>
<p>The categorical or numeric metadata field over which
statistics should be calculated. Required.</p>
</td></tr>
<tr><td><code id="documentation_dist_test_+3A_test">test</code></td>
<td>
<p>Permutational test for accessing significance. Options are:
</p>

<dl>
<dt><code>"adonis2"</code> - </dt><dd><p> Permutational MANOVA; <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code>. </p>
</dd>
<dt><code>"mrpp"</code> - </dt><dd><p> Multiple response permutation procedure; <code><a href="vegan.html#topic+mrpp">vegan::mrpp()</a></code>. </p>
</dd>
<dt><code>"none"</code> - </dt><dd><p> Don't run any statistics. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"adonis2"</code></p>
</td></tr>
</table>

<hr>
<h2 id='documentation_heatmap'>documentation_heatmap</h2><span id='topic+documentation_heatmap'></span>

<h3>Description</h3>

<p>documentation_heatmap
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_heatmap_+3A_grid">grid</code></td>
<td>
<p>Color palette name, or a list with entries for <code>label</code>,
<code>colors</code>, <code>range</code>, <code>bins</code>, <code>na.color</code>, and/or
<code>guide</code>. See the Track Definitions section for details.
Default: <code>list(label = "Grid Value", colors = "imola")</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_label">label</code></td>
<td>
<p>Label the matrix rows and columns. You can supply a list
or logical vector of length two to control row labels and column
labels separately, for example
<code>label = c(rows = TRUE, cols = FALSE)</code>, or simply
<code>label = c(TRUE, FALSE)</code>. Other valid options are <code>"rows"</code>,
<code>"cols"</code>, <code>"both"</code>, <code>"bottom"</code>, <code>"right"</code>,
and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_label_size">label_size</code></td>
<td>
<p>The font size to use for the row and column labels. You
can supply a numeric vector of length two to control row label sizes
and column label sizes separately, for example
<code>c(rows = 20, cols = 8)</code>, or simply <code>c(20, 8)</code>.
Default: <code>NULL</code>, which computes:
<code>pmax(8, pmin(20, 100 / dim(mtx)))</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_rescale">rescale</code></td>
<td>
<p>Rescale rows or columns to all have a common min/max.
Options: <code>"none"</code>, <code>"rows"</code>, or <code>"cols"</code>.
Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_trees">trees</code></td>
<td>
<p>Draw a dendrogram for rows (left) and columns (top). You can
supply a list or logical vector of length two to control the row tree
and column tree separately, for example
<code>trees = c(rows = TRUE, cols = FALSE)</code>,
or simply <code>trees = c(TRUE, FALSE)</code>.
Other valid options are <code>"rows"</code>, <code>"cols"</code>, <code>"both"</code>,
<code>"left"</code>, <code>"top"</code>, and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_clust">clust</code></td>
<td>
<p>Clustering algorithm for reordering the rows and columns by
similarity. You can supply a list or character vector of length two to
control the row and column clustering separately, for example
<code>clust = c(rows = "complete", cols = NA)</code>, or simply
<code>clust = c("complete", NA)</code>. Options are:
</p>

<dl>
<dt><code>FALSE</code> or <code>NA</code> - </dt><dd><p> Disable reordering. </p>
</dd>
<dt>An <code>hclust</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+hclust">stats::hclust()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"ward.D"</code>,
<code>"ward.D2"</code>, <code>"single"</code>, <code>"complete"</code>,
<code>"average"</code>, <code>"mcquitty"</code>, <code>"median"</code>, or
<code>"centroid"</code>. </p>
</dd>
</dl>

<p>Default: <code>"complete"</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_dist">dist</code></td>
<td>
<p>Distance algorithm to use when reordering the rows and columns
by similarity. You can supply a list or character vector of length
two to control the row and column clustering separately, for example
<code>dist = c(rows = "euclidean", cols = "maximum")</code>, or simply
<code>dist = c("euclidean", "maximum")</code>. Options are:
</p>

<dl>
<dt>A <code>dist</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+dist">stats::dist()</a></code> or <code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"euclidean"</code>,
<code>"maximum"</code>, <code>"manhattan"</code>, <code>"canberra"</code>,
<code>"binary"</code>, or <code>"minkowski"</code>. </p>
</dd>
</dl>

<p>Default: <code>"euclidean"</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_tree_height">tree_height</code>, <code id="documentation_heatmap_+3A_track_height">track_height</code></td>
<td>
<p>The height of the dendrogram or annotation
tracks as a percentage of the overall grid size. Use a numeric vector
of length two to assign <code>c(top, left)</code> independently.
Default: <code>10</code> (10% of the grid's height)</p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio (height/width) for entire grid.
Default: <code>1</code> (square)</p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_legend">legend</code></td>
<td>
<p>Where to place the legend. Options are: <code>"right"</code> or
<code>"bottom"</code>. Default: <code>"right"</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_title">title</code></td>
<td>
<p>Plot title. Set to <code>TRUE</code> for a default title, <code>NULL</code> for
no title, or any character string. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="documentation_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to ggplot2::theme().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>

<hr>
<h2 id='documentation_plot_return'>documentation_plot_return</h2><span id='topic+documentation_plot_return'></span>

<h3>Description</h3>

<p>documentation_plot_return
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>

<hr>
<h2 id='documentation_rank.2'>documentation_rank.2</h2><span id='topic+documentation_rank.2'></span>

<h3>Description</h3>

<p>documentation_rank.2
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_rank.2_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to compute biplot coordinates and
statistics for, or <code>NULL</code> to disable. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>2</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='documentation_rank.NULL'>documentation_rank.NULL</h2><span id='topic+documentation_rank.NULL'></span>

<h3>Description</h3>

<p>documentation_rank.NULL
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_rank.NULL_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to compute biplot coordinates and
statistics for, or <code>NULL</code> to disable. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>NULL</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='documentation_return.biom'>documentation_return.biom</h2><span id='topic+documentation_return.biom'></span>

<h3>Description</h3>

<p>documentation_return.biom
</p>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>

<hr>
<h2 id='documentation_taxa.4'>documentation_taxa.4</h2><span id='topic+documentation_taxa.4'></span>

<h3>Description</h3>

<p>documentation_taxa.4
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="documentation_taxa.4_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>4</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='export'>Export data to QIIME 2 or mothur.</h2><span id='topic+export'></span><span id='topic+write_mothur'></span><span id='topic+write_qiime2'></span>

<h3>Description</h3>

<p>Populates a directory with the following files, formatted according to
QIIME 2 or mothur's specifications.
</p>

<ul>
<li> <p><code>biom_counts.tsv</code>
</p>
</li>
<li> <p><code>biom_metadata.tsv</code>
</p>
</li>
<li> <p><code>biom_taxonomy.tsv</code>
</p>
</li>
<li> <p><code>biom_tree.nwk</code>
</p>
</li>
<li> <p><code>biom_seqs.fna</code>
</p>
</li></ul>

<p><code>biom_counts.tsv</code> will always be created. The others are dependent on
whether the content is present in the <code>biom</code> argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_mothur(biom, dir = tempfile(), prefix = "biom_")

write_qiime2(biom, dir = tempfile(), prefix = "biom_")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="export_+3A_dir">dir</code></td>
<td>
<p>Where to save the files. If the directory doesn't exist, it will
be created. Default: <code>tempfile()</code></p>
</td></tr>
<tr><td><code id="export_+3A_prefix">prefix</code></td>
<td>
<p>A string to prepend to each file name. Default: <code>'biom_'</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The normalized directory path that was written to (invisibly).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    tdir &lt;- tempfile()
    
    write_qiime2(hmp50, tdir, 'qiime2_')
    write_mothur(hmp50, tdir, 'mothur_')
    
    list.files(tdir)
    
    readLines(file.path(tdir, 'qiime2_metadata.tsv'), n = 4)
    
    readLines(file.path(tdir, 'mothur_taxonomy.tsv'), n = 3)
    
    unlink(tdir, recursive = TRUE)

</code></pre>

<hr>
<h2 id='gems'>Global Enteric Multicenter Study (n = 1,006)</h2><span id='topic+gems'></span>

<h3>Description</h3>

<p>Global Enteric Multicenter Study (n = 1,006)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gems
</code></pre>


<h3>Format</h3>

<p>An rbiom object with 1,006 samples.
Includes metadata and taxonomy.
</p>

<dl>
<dt>diarrhea - </dt><dd><p>Case or Control</p>
</dd>
<dt>age - </dt><dd><p>0 - 4.8 (years old)</p>
</dd>
<dt>country - </dt><dd><p>Bangladesh, Gambia, Kenya, or Mali</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://doi.org/10.1186/gb-2014-15-6-r76">doi:10.1186/gb-2014-15-6-r76</a> and <a href="https://doi.org/10.1093/nar/gkx1027">doi:10.1093/nar/gkx1027</a>
</p>

<hr>
<h2 id='glimpse.rbiom'>Get a glimpse of your metadata.</h2><span id='topic+glimpse.rbiom'></span>

<h3>Description</h3>

<p>Get a glimpse of your metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
glimpse(x, width = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glimpse.rbiom_+3A_x">x</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="glimpse.rbiom_+3A_width">width</code></td>
<td>
<p>Width of output. See <code><a href="pillar.html#topic+glimpse">pillar::glimpse()</a></code> documentation.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="glimpse.rbiom_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The original <code>biom</code>, invisibly.
</p>


<h3>See Also</h3>

<p>Other metadata: 
<code><a href="#topic+bdply">bdply</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    glimpse(hmp50)
    
</code></pre>

<hr>
<h2 id='hmp50'>Human Microbiome Project - demo dataset (n = 50)</h2><span id='topic+hmp50'></span>

<h3>Description</h3>

<p>Human Microbiome Project - demo dataset (n = 50)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmp50
</code></pre>


<h3>Format</h3>

<p>An rbiom object with 50 samples.
Includes metadata, taxonomy, phylogeny, and sequences.
</p>

<dl>
<dt>Sex - </dt><dd><p>Male or Female</p>
</dd>
<dt>Body Site - </dt><dd><p>Anterior nares, Buccal mucosa, Mid vagina, Saliva, or Stool</p>
</dd>
<dt>Age - </dt><dd><p>21 - 40</p>
</dd>
<dt>BMI - </dt><dd><p>19 - 32</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://portal.hmpdacc.org">https://portal.hmpdacc.org</a>
</p>

<hr>
<h2 id='modify_metadata'>Create, modify, and delete metadata fields.</h2><span id='topic+modify_metadata'></span><span id='topic+mutate.rbiom'></span><span id='topic+rename.rbiom'></span>

<h3>Description</h3>

<p>mutate() creates new fields in <code style="white-space: pre;">&#8288;$metadata&#8288;</code> that are functions of existing
metadata fields. It can also modify (if the name is the same as an existing
field) and delete fields (by setting their value to NULL).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
mutate(.data, ..., clone = TRUE)

## S3 method for class 'rbiom'
rename(.data, ..., clone = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="modify_metadata_+3A_.data">.data</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="modify_metadata_+3A_...">...</code></td>
<td>
<p>Passed on to <code><a href="dplyr.html#topic+mutate">dplyr::mutate()</a></code> or <code><a href="dplyr.html#topic+rename">dplyr::rename()</a></code>.</p>
</td></tr>
<tr><td><code id="modify_metadata_+3A_clone">clone</code></td>
<td>
<p>Create a copy of <code>biom</code> before modifying. If <code>FALSE</code>, <code>biom</code>
is modified in place as a side-effect. See <a href="#topic+speed">speed ups</a> for
use cases. Default: <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>See Also</h3>

<p>Other transformations: 
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+slice_metadata">slice_metadata</a></code>,
<code><a href="#topic+subset">subset</a>()</code>,
<code><a href="#topic+with">with</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    biom &lt;- slice_max(hmp50, BMI, n = 6)
    biom$metadata
    
    # Add a new field to the metadata
    biom &lt;- mutate(biom, Obsese = BMI &gt;= 30)
    biom$metadata
    
    # Rename a metadata field
    biom &lt;- rename(biom, 'Age (years)' = "Age")
    biom$metadata

</code></pre>

<hr>
<h2 id='plot_heatmap'>Create a heatmap with tracks and dendrograms from any matrix.</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>Create a heatmap with tracks and dendrograms from any matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  mtx,
  grid = list(label = "Grid Value", colors = "imola"),
  tracks = NULL,
  label = TRUE,
  label_size = NULL,
  rescale = "none",
  trees = TRUE,
  clust = "complete",
  dist = "euclidean",
  asp = 1,
  tree_height = 10,
  track_height = 10,
  legend = "right",
  title = NULL,
  xlab.angle = "auto",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap_+3A_mtx">mtx</code></td>
<td>
<p>A numeric <code>matrix</code> with named rows and columns.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_grid">grid</code></td>
<td>
<p>Color palette name, or a list with entries for <code>label</code>,
<code>colors</code>, <code>range</code>, <code>bins</code>, <code>na.color</code>, and/or
<code>guide</code>. See the Track Definitions section for details.
Default: <code>list(label = "Grid Value", colors = "imola")</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_tracks">tracks</code></td>
<td>
<p>List of track definitions. See details below.
Default: <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_label">label</code></td>
<td>
<p>Label the matrix rows and columns. You can supply a list
or logical vector of length two to control row labels and column
labels separately, for example
<code>label = c(rows = TRUE, cols = FALSE)</code>, or simply
<code>label = c(TRUE, FALSE)</code>. Other valid options are <code>"rows"</code>,
<code>"cols"</code>, <code>"both"</code>, <code>"bottom"</code>, <code>"right"</code>,
and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_label_size">label_size</code></td>
<td>
<p>The font size to use for the row and column labels. You
can supply a numeric vector of length two to control row label sizes
and column label sizes separately, for example
<code>c(rows = 20, cols = 8)</code>, or simply <code>c(20, 8)</code>.
Default: <code>NULL</code>, which computes:
<code>pmax(8, pmin(20, 100 / dim(mtx)))</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_rescale">rescale</code></td>
<td>
<p>Rescale rows or columns to all have a common min/max.
Options: <code>"none"</code>, <code>"rows"</code>, or <code>"cols"</code>.
Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_trees">trees</code></td>
<td>
<p>Draw a dendrogram for rows (left) and columns (top). You can
supply a list or logical vector of length two to control the row tree
and column tree separately, for example
<code>trees = c(rows = TRUE, cols = FALSE)</code>,
or simply <code>trees = c(TRUE, FALSE)</code>.
Other valid options are <code>"rows"</code>, <code>"cols"</code>, <code>"both"</code>,
<code>"left"</code>, <code>"top"</code>, and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_clust">clust</code></td>
<td>
<p>Clustering algorithm for reordering the rows and columns by
similarity. You can supply a list or character vector of length two to
control the row and column clustering separately, for example
<code>clust = c(rows = "complete", cols = NA)</code>, or simply
<code>clust = c("complete", NA)</code>. Options are:
</p>

<dl>
<dt><code>FALSE</code> or <code>NA</code> - </dt><dd><p> Disable reordering. </p>
</dd>
<dt>An <code>hclust</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+hclust">stats::hclust()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"ward.D"</code>,
<code>"ward.D2"</code>, <code>"single"</code>, <code>"complete"</code>,
<code>"average"</code>, <code>"mcquitty"</code>, <code>"median"</code>, or
<code>"centroid"</code>. </p>
</dd>
</dl>

<p>Default: <code>"complete"</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_dist">dist</code></td>
<td>
<p>Distance algorithm to use when reordering the rows and columns
by similarity. You can supply a list or character vector of length
two to control the row and column clustering separately, for example
<code>dist = c(rows = "euclidean", cols = "maximum")</code>, or simply
<code>dist = c("euclidean", "maximum")</code>. Options are:
</p>

<dl>
<dt>A <code>dist</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+dist">stats::dist()</a></code> or <code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"euclidean"</code>,
<code>"maximum"</code>, <code>"manhattan"</code>, <code>"canberra"</code>,
<code>"binary"</code>, or <code>"minkowski"</code>. </p>
</dd>
</dl>

<p>Default: <code>"euclidean"</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio (height/width) for entire grid.
Default: <code>1</code> (square)</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_tree_height">tree_height</code>, <code id="plot_heatmap_+3A_track_height">track_height</code></td>
<td>
<p>The height of the dendrogram or annotation
tracks as a percentage of the overall grid size. Use a numeric vector
of length two to assign <code>c(top, left)</code> independently.
Default: <code>10</code> (10% of the grid's height)</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_legend">legend</code></td>
<td>
<p>Where to place the legend. Options are: <code>"right"</code> or
<code>"bottom"</code>. Default: <code>"right"</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_title">title</code></td>
<td>
<p>Plot title. Default: <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to ggplot2::theme().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>


<h3>Track Definitions</h3>

<p>One or more colored tracks can be placed on the left and/or top of the
heatmap grid to visualize associated metadata values.
</p>
<pre>## Categorical ----------------------------
cat_vals &lt;- sample(c("Male", "Female"), 10, replace = TRUE)
tracks   &lt;- list('Sex' = cat_vals)
tracks   &lt;- list('Sex' = list(values = cat_vals, colors = "bright"))
tracks   &lt;- list('Sex' = list(
  values = cat_vals, 
  colors = c('Male' = "blue", 'Female' = "red")) )

## Numeric --------------------------------
num_vals &lt;- sample(25:40, 10, replace = TRUE)
tracks   &lt;- list('Age' = num_vals)
tracks   &lt;- list('Age' = list(values = num_vals, colors = "greens"))
tracks   &lt;- list('Age' = list(values = num_vals, range = c(0,50)))
tracks   &lt;- list('Age' = list(
  label  = "Age (Years)",
  values = num_vals, 
  colors = c("azure", "darkblue", "darkorchid") ))

## Multiple Tracks ------------------------
tracks &lt;- list('Sex' = cat_vals, 'Age' = num_vals)
tracks &lt;- list(
  list(label = "Sex", values = cat_vals, colors = "bright"),
  list(label = "Age", values = num_vals, colors = "greens") )
  
  
mtx           &lt;- matrix(sample(1:50), ncol = 10)
dimnames(mtx) &lt;- list(letters[1:5], LETTERS[1:10])
plot_heatmap(mtx = mtx, tracks = tracks)
</pre>
<p>The following entries in the track definitions are understood:
</p>

<dl>
<dt><code>values</code> - </dt><dd><p> The metadata values. When unnamed, order must match matrix. </p>
</dd>
<dt><code>range</code> - </dt><dd><p> The c(min,max) to use for scale values. </p>
</dd>
<dt><code>label</code> - </dt><dd><p> Label for this track. Defaults to the name of this list element. </p>
</dd>
<dt><code>side</code> - </dt><dd><p> Options are <code>"top"</code> (default) or <code>"left"</code>. </p>
</dd>
<dt><code>colors</code> - </dt><dd><p> A pre-defined palette name or custom set of colors to map to. </p>
</dd>
<dt><code>na.color</code> - </dt><dd><p> The color to use for <code>NA</code> values. </p>
</dd>
<dt><code>bins</code> - </dt><dd><p> Bin a gradient into this many bins/steps. </p>
</dd>
<dt><code>guide</code> - </dt><dd><p> A list of arguments for guide_colorbar() or guide_legend(). </p>
</dd>
</dl>

<p>All built-in color palettes are colorblind-friendly. See
<a href="https://cmmr.github.io/rbiom/articles/aes.html#discrete-palettes">Mapping Metadata to Aesthetics</a>
for images of the palettes.
</p>
<p>Categorical palette names: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Numeric palette names: <code>"reds"</code>, <code>"oranges"</code>, <code>"greens"</code>,
<code>"purples"</code>, <code>"grays"</code>, <code>"acton"</code>, <code>"bamako"</code>,
<code>"batlow"</code>, <code>"bilbao"</code>, <code>"buda"</code>, <code>"davos"</code>,
<code>"devon"</code>, <code>"grayC"</code>, <code>"hawaii"</code>, <code>"imola"</code>,
<code>"lajolla"</code>, <code>"lapaz"</code>, <code>"nuuk"</code>, <code>"oslo"</code>,
<code>"tokyo"</code>, <code>"turku"</code>, <code>"bam"</code>, <code>"berlin"</code>,
<code>"broc"</code>, <code>"cork"</code>, <code>"lisbon"</code>, <code>"roma"</code>,
<code>"tofino"</code>, <code>"vanimo"</code>, and <code>"vik"</code>.
</p>


<h3>See Also</h3>

<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    set.seed(123)
    mtx &lt;- matrix(runif(5*8), nrow = 5, dimnames = list(LETTERS[1:5], letters[1:8]))
    
    plot_heatmap(mtx)
    plot_heatmap(mtx, grid="oranges")
    plot_heatmap(mtx, grid=list(colors = "oranges", label = "Some %", bins = 5))
    
    tracks &lt;- list(
      'Number' = sample(1:ncol(mtx)),
      'Person' = list(
        values = factor(sample(c("Alice", "Bob"), ncol(mtx), TRUE)),
        colors = c('Alice' = "purple", 'Bob' = "darkcyan") ),
      'State' = list(
        side   = "left",
        values = sample(c("TX", "OR", "WA"), nrow(mtx), TRUE),
        colors = "bright" )
    )
    
    plot_heatmap(mtx, tracks=tracks)
    
</code></pre>

<hr>
<h2 id='pull.rbiom'>Map sample names to metadata field values.</h2><span id='topic+pull.rbiom'></span>

<h3>Description</h3>

<p>Map sample names to metadata field values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
pull(.data, var = -1, name = ".sample", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pull.rbiom_+3A_.data">.data</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="pull.rbiom_+3A_var">var</code></td>
<td>
<p>The metadata field name specified as:
</p>

<ul>
<li><p>The metadata field name to retrieve. Can be abbreviated.
</p>
</li>
<li><p>A positive integer, giving the position counting from the left.
</p>
</li>
<li><p>A negative integer, giving the position counting from the right.
</p>
</li></ul>

<p>Default: <code>-1</code></p>
</td></tr>
<tr><td><code id="pull.rbiom_+3A_name">name</code></td>
<td>
<p>The column to be used as names for a named vector.
Specified in a similar manner as var. Default: <code>".sample"</code></p>
</td></tr>
<tr><td><code id="pull.rbiom_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of metadata values, named with sample names.
</p>


<h3>See Also</h3>

<p><code>taxa_map()</code>
</p>
<p>Other samples: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    pull(hmp50, 'Age') %&gt;% head()
    
    pull(hmp50, 'bod') %&gt;% head(4)
    
</code></pre>

<hr>
<h2 id='rare_corrplot'>Visualize rarefaction curves with scatterplots and trendlines.</h2><span id='topic+rare_corrplot'></span>

<h3>Description</h3>

<p>Visualize rarefaction curves with scatterplots and trendlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rare_corrplot(
  biom,
  adiv = "Shannon",
  layers = "tc",
  rline = TRUE,
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  test = "none",
  fit = "log",
  at = NULL,
  level = 0.95,
  p.adj = "fdr",
  transform = "none",
  alt = "!=",
  mu = 0,
  caption = TRUE,
  check = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rare_corrplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("trend", "confidence", "point", "name", "residual")</code>. Single
letter abbreviations are also accepted. For instance,
<code>c("trend", "point")</code> is equivalent to <code>c("t", "p")</code> and <code>"tp"</code>.
Default: <code>"tc"</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_rline">rline</code></td>
<td>
<p>Where to draw a horizontal line on the plot, intended to show
a particular rarefaction depth. Set to <code>TRUE</code> to show an
auto-selected rarefaction depth or <code>FALSE</code> to not show a line.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'none'</code>, <code>'emmeans'</code>, or
<code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline.
Options are <code>'lm'</code>, <code>'log'</code>, and <code>'gam'</code>.
Default: <code>'log'</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="rare_corrplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>p.size = 2</code> ensures only the points have their
size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other rarefaction: 
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+sample_sums">sample_sums</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- subset(hmp50, `Body Site` %in% c('Saliva', 'Stool'))
    rare_corrplot(biom, stat.by = "body", adiv = c("sh", "o"), facet.by = "Sex")
    
</code></pre>

<hr>
<h2 id='rare_multiplot'>Combines rare_corrplot and rare_stacked into a single figure.</h2><span id='topic+rare_multiplot'></span>

<h3>Description</h3>

<p>Combines rare_corrplot and rare_stacked into a single figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rare_multiplot(
  biom,
  adiv = "Shannon",
  layers = "tc",
  rline = TRUE,
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  test = "none",
  fit = "log",
  at = NULL,
  level = 0.95,
  p.adj = "fdr",
  transform = "none",
  alt = "!=",
  mu = 0,
  caption = TRUE,
  check = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rare_multiplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_adiv">adiv</code></td>
<td>
<p>Alpha diversity metric(s) to use. Options are: <code>"OTUs"</code>,
<code>"Shannon"</code>, <code>"Chao1"</code>, <code>"Simpson"</code>, and/or
<code>"InvSimpson"</code>. Set <code>adiv=".all"</code> to use all metrics.
Multiple/abbreviated values allowed.
Default: <code>"Shannon"</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("trend", "confidence", "point", "name", "residual")</code>. Single
letter abbreviations are also accepted. For instance,
<code>c("trend", "point")</code> is equivalent to <code>c("t", "p")</code> and <code>"tp"</code>.
Default: <code>"tc"</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_rline">rline</code></td>
<td>
<p>Where to draw a horizontal line on the plot, intended to show
a particular rarefaction depth. Set to <code>TRUE</code> to show an
auto-selected rarefaction depth or <code>FALSE</code> to not show a line.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'none'</code>, <code>'emmeans'</code>, or
<code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline.
Options are <code>'lm'</code>, <code>'log'</code>, and <code>'gam'</code>.
Default: <code>'log'</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="rare_multiplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>p.size = 2</code> ensures only the points have their
size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other rarefaction: 
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+sample_sums">sample_sums</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    rare_multiplot(hmp50, stat.by = "Body Site")
    
</code></pre>

<hr>
<h2 id='rare_stacked'>Visualize the number of observations per sample.</h2><span id='topic+rare_stacked'></span>

<h3>Description</h3>

<p>Visualize the number of observations per sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rare_stacked(
  biom,
  rline = TRUE,
  counts = TRUE,
  labels = TRUE,
  y.transform = "log10",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rare_stacked_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="rare_stacked_+3A_rline">rline</code></td>
<td>
<p>Where to draw a horizontal line on the plot, intended to show
a particular rarefaction depth. Set to <code>TRUE</code> to show an
auto-selected rarefaction depth, <code>FALSE</code> to not show a line, or
an integer for a custom position.
Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="rare_stacked_+3A_counts">counts</code></td>
<td>
<p>Display the number of samples and reads remaining after
rarefying to <code>rline</code> reads per sample. Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="rare_stacked_+3A_labels">labels</code></td>
<td>
<p>Show sample names under each bar. Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="rare_stacked_+3A_y.transform">y.transform</code></td>
<td>
<p>Y-axis transformation. Options are <code>"log10"</code> or
<code>"none"</code>.  Default: <code>"log10"</code>.
Use <code>xaxis.transform</code> or <code>yaxis.transform</code> to pass custom values
directly to ggplot2's <code style="white-space: pre;">&#8288;scale_*&#8288;</code> functions.</p>
</td></tr>
<tr><td><code id="rare_stacked_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with <code>r.</code> to ensure it gets
passed to (and only to) <a href="ggplot2.html#topic+geom_hline">geom_hline</a>. For instance,
<code>r.color = "black"</code> ensures only the horizontal rarefaction line
has its color set to <code>"black"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>


<h3>See Also</h3>

<p>Other rarefaction: 
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+sample_sums">sample_sums</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    rare_stacked(hmp50)
    
    rare_stacked(hmp50, rline = 500, r.linewidth = 2, r.linetype = "twodash")
    
    fig &lt;- rare_stacked(hmp50, counts = FALSE)
    fig$code
    
</code></pre>

<hr>
<h2 id='rarefy'>Rarefy OTU counts.</h2><span id='topic+rarefy'></span>

<h3>Description</h3>

<p>Sub-sample OTU observations such that all samples have an equal number.
If called on data with non-integer abundances, values will be re-scaled to
integers between 1 and <code>depth</code> such that they sum to <code>depth</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rarefy(biom, depth = 0.1, n = NULL, seed = 0, clone = TRUE, cpus = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rarefy_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="rarefy_+3A_depth">depth</code></td>
<td>
<p>How many observations to keep per sample. When
<code style="white-space: pre;">&#8288;0 &lt; depth &lt; 1&#8288;</code>, it is taken as the minimum percentage of the
dataset's observations to keep. Ignored when <code>n</code> is specified.
Default: <code>0.1</code></p>
</td></tr>
<tr><td><code id="rarefy_+3A_n">n</code></td>
<td>
<p>The number of samples to keep. When <code style="white-space: pre;">&#8288;0 &lt; n &lt; 1&#8288;</code>, it is taken as
the percentage of samples to keep. If negative, that number or
percentage of samples is dropped. If <code>0</code>, all samples are kept. If
<code>NULL</code>, <code>depth</code> is used instead.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rarefy_+3A_seed">seed</code></td>
<td>
<p>An integer seed for randomizing which observations to keep or
drop. If you need to create different random rarefactions of the same
data, set the seed to a different number each time.</p>
</td></tr>
<tr><td><code id="rarefy_+3A_clone">clone</code></td>
<td>
<p>Create a copy of <code>biom</code> before modifying. If <code>FALSE</code>, <code>biom</code>
is modified in place as a side-effect. See <a href="#topic+speed">speed ups</a> for
use cases. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="rarefy_+3A_cpus">cpus</code></td>
<td>
<p>The number of CPUs to use. Set to <code>NULL</code> to use all available,
or to <code>1</code> to disable parallel processing. Default: <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>See Also</h3>

<p>Other rarefaction: 
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+sample_sums">sample_sums</a>()</code>
</p>
<p>Other transformations: 
<code><a href="#topic+modify_metadata">modify_metadata</a></code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+slice_metadata">slice_metadata</a></code>,
<code><a href="#topic+subset">subset</a>()</code>,
<code><a href="#topic+with">with</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    sample_sums(hmp50) %&gt;% head()
    
    biom &lt;- rarefy(hmp50)
    sample_sums(biom) %&gt;% head()

</code></pre>

<hr>
<h2 id='rarefy_cols'>Transform a counts matrix.</h2><span id='topic+rarefy_cols'></span><span id='topic+rescale_cols'></span><span id='topic+rescale_rows'></span>

<h3>Description</h3>

<p>Rarefaction subset counts so that all samples have the same number of
observations. Rescaling rows or cols scales the matrix values so that row
sums or column sums equal 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rarefy_cols(mtx, depth = 0.1, n = NULL, seed = 0L, cpus = NULL)

rescale_cols(mtx)

rescale_rows(mtx)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rarefy_cols_+3A_mtx">mtx</code></td>
<td>
<p>A matrix-like object.</p>
</td></tr>
<tr><td><code id="rarefy_cols_+3A_depth">depth</code></td>
<td>
<p>How many observations to keep per sample. When
<code style="white-space: pre;">&#8288;0 &lt; depth &lt; 1&#8288;</code>, it is taken as the minimum percentage of the
dataset's observations to keep. Ignored when <code>n</code> is specified.
Default: <code>0.1</code></p>
</td></tr>
<tr><td><code id="rarefy_cols_+3A_n">n</code></td>
<td>
<p>The number of samples to keep. When <code style="white-space: pre;">&#8288;0 &lt; n &lt; 1&#8288;</code>, it is taken as
the percentage of samples to keep. If negative, that number or
percentage of samples is dropped. If <code>0</code>, all samples are kept. If
<code>NULL</code>, <code>depth</code> is used instead.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="rarefy_cols_+3A_seed">seed</code></td>
<td>
<p>A positive integer to use for seeding the random number
generator. If you need to create different random rarefactions of the
same matrix, set this seed value to a different number each time.</p>
</td></tr>
<tr><td><code id="rarefy_cols_+3A_cpus">cpus</code></td>
<td>
<p>The number of CPUs to use. Set to <code>NULL</code> to use all available,
or to <code>1</code> to disable parallel processing. Default: <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The rarefied or rescaled matrix.
</p>


<h3>See Also</h3>

<p>Other rarefaction: 
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+sample_sums">sample_sums</a>()</code>
</p>
<p>Other transformations: 
<code><a href="#topic+modify_metadata">modify_metadata</a></code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+slice_metadata">slice_metadata</a></code>,
<code><a href="#topic+subset">subset</a>()</code>,
<code><a href="#topic+with">with</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # rarefy_cols --------------------------------------
    biom &lt;- hmp50$clone()
    sample_sums(biom) %&gt;% head(10)

    biom$counts %&lt;&gt;% rarefy_cols(depth=1000)
    sample_sums(biom) %&gt;% head(10)
    
    
    # rescaling ----------------------------------------
    mtx &lt;- matrix(sample(1:20), nrow=4)
    mtx
    
    rowSums(mtx)
    rowSums(rescale_rows(mtx))
    
    colSums(mtx)
    colSums(rescale_cols(mtx))

</code></pre>

<hr>
<h2 id='rbiom_objects'>Working with rbiom Objects.</h2><span id='topic+rbiom_objects'></span>

<h3>Description</h3>

<p>Rbiom objects make it easy to access and manipulate your BIOM data, ensuring
all the disparate components remain in sync. These objects behave largely
like lists, in that you can access and assign to them using the <code>$</code>
operator. The sections below list all the fields which can be read and/or
written, and the helper functions for common tasks like rarefying and
subsetting. To create an rbiom object, see <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
</p>
<p>Use <code style="white-space: pre;">&#8288;$clone()&#8288;</code> to create a copy of an rbiom object. This is necessary
because rbiom objects are <strong>passed by reference</strong>. The usual <code style="white-space: pre;">&#8288;&lt;-&#8288;</code> assignment
operator will simply create a second reference to the same object - it will
not create a second object. See <a href="#topic+speed">speed ups</a> for more details.
</p>


<h3>Readable Fields</h3>

<p>Reading from fields will not change the rbiom object.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Accessor</strong> </td><td style="text-align: left;"> <strong>Content</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$counts&#8288;</code> </td><td style="text-align: left;"> Abundance of each OTU in each sample. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$metadata&#8288;</code> </td><td style="text-align: left;"> Sample mappings to metadata (treatment, patient, etc). </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$taxonomy&#8288;</code> </td><td style="text-align: left;"> OTU mappings to taxonomic ranks (genus, phylum, etc). </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$otus&#8288;</code>, <code style="white-space: pre;">&#8288;$n_otus&#8288;</code> </td><td style="text-align: left;"> OTU names. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$samples&#8288;</code>, <code style="white-space: pre;">&#8288;$n_samples&#8288;</code> </td><td style="text-align: left;"> Sample names. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$fields&#8288;</code>, <code style="white-space: pre;">&#8288;$n_fields&#8288;</code> </td><td style="text-align: left;"> Metadata field names. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$ranks&#8288;</code>, <code style="white-space: pre;">&#8288;$n_ranks&#8288;</code> </td><td style="text-align: left;"> Taxonomic rank names. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$tree&#8288;</code>, <code style="white-space: pre;">&#8288;$sequences&#8288;</code> </td><td style="text-align: left;"> Phylogenetic tree / sequences for the OTUs, or <code>NULL</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$id&#8288;</code>, <code style="white-space: pre;">&#8288;$comment&#8288;</code> </td><td style="text-align: left;"> Arbitrary strings for describing the dataset. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$depth&#8288;</code> </td><td style="text-align: left;"> Rarefaction depth, or <code>NULL</code> if unrarefied. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$date&#8288;</code> </td><td style="text-align: left;"> Date from BIOM file. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Writable Fields</h3>

<p>Assigning new values to these components will trigger
validation checks and inter-component synchronization.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Component</strong> </td><td style="text-align: left;"> <strong>What can be assigned.</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$counts&#8288;</code> </td><td style="text-align: left;"> Matrix of abundances; OTUs (rows) by samples (columns). </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$metadata&#8288;</code> </td><td style="text-align: left;"> Data.frame with <code>'.sample'</code> column, or a file name. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$taxonomy&#8288;</code> </td><td style="text-align: left;"> Data.frame with <code>'.otu'</code> as the first column. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$otus&#8288;</code> </td><td style="text-align: left;"> Character vector with new names for the OTUs. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$samples&#8288;</code> </td><td style="text-align: left;"> Character vector with new names for the samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$tree&#8288;</code> </td><td style="text-align: left;"> Phylo object with the phylogenetic tree for the OTUs. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$sequences&#8288;</code> </td><td style="text-align: left;"> Named character vector of OTU reference sequences. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$id&#8288;</code>, <code style="white-space: pre;">&#8288;$comment&#8288;</code> </td><td style="text-align: left;"> String with dataset's title or comment. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code style="white-space: pre;">&#8288;$date&#8288;</code> </td><td style="text-align: left;"> Date-like object, or <code>"%Y-%m-%dT%H:%M:%SZ"</code> string. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Transformations</h3>

<p>All functions return an rbiom object.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Function</strong> </td><td style="text-align: left;"> <strong>Transformation</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code>&lt;rbiom&gt;$clone()</code> </td><td style="text-align: left;"> Safely duplicate an rbiom object. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+subset">&lt;rbiom&gt;[</a></code> </td><td style="text-align: left;"> Subset to a specific set of sample names. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+subset">subset()</a></code> </td><td style="text-align: left;"> Subset samples according to metadata properties. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+slice">slice()</a></code> </td><td style="text-align: left;"> Subset to a specific number of samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+mutate">mutate()</a></code> </td><td style="text-align: left;"> Create, modify, and delete metadata fields. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code><a href="#topic+rarefy">rarefy()</a></code> </td><td style="text-align: left;"> Sub-sample OTU counts to an even sampling depth. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # Duplicate the HMP50 example dataset.
    biom &lt;- hmp50$clone()
    
    
    # Display an overall summary of the rbiom object.
    biom
    
    
    # Markdown syntax for comments is recommended.
    biom$comment %&gt;% cli::cli_text()
    
    
    # Demonstrate a few accessors.
    biom$n_samples
    biom$fields
    biom$metadata
    
    
    # Edit the metadata table.
    biom$metadata$rand &lt;- sample(1:50)
    biom %&lt;&gt;% mutate(Obese = BMI &gt;= 30, Sex = NULL)
    biom %&lt;&gt;% rename('Years Old' = "Age")
    biom$metadata
    
    
    # Subset the rbiom object
    biom %&lt;&gt;% subset(`Body Site` == "Saliva" &amp; !Obese)
    biom$metadata
    
    
    # Rarefy to an even sampling depth
    sample_sums(biom)
    
    biom %&lt;&gt;% rarefy()
    sample_sums(biom)

</code></pre>

<hr>
<h2 id='rbiom-deprecated'>Deprecated functions in package <span class="pkg">rbiom</span>.</h2><span id='topic+rbiom-deprecated'></span><span id='topic+alpha.div-deprecated'></span><span id='topic+alpha.div'></span><span id='topic+beta.div-deprecated'></span><span id='topic+beta.div'></span><span id='topic+counts-deprecated'></span><span id='topic+counts'></span><span id='topic+info-deprecated'></span><span id='topic+info'></span><span id='topic+metadata-deprecated'></span><span id='topic+metadata'></span><span id='topic+nsamples-deprecated'></span><span id='topic+nsamples'></span><span id='topic+ntaxa-deprecated'></span><span id='topic+ntaxa'></span><span id='topic+phylogeny-deprecated'></span><span id='topic+phylogeny'></span><span id='topic+read.biom-deprecated'></span><span id='topic+read.biom'></span><span id='topic+read.fasta-deprecated'></span><span id='topic+read.fasta'></span><span id='topic+read.tree-deprecated'></span><span id='topic+read.tree'></span><span id='topic+sample.names-deprecated'></span><span id='topic+sample.names'></span><span id='topic+select-deprecated'></span><span id='topic+select.rbiom'></span><span id='topic+sequences-deprecated'></span><span id='topic+sequences'></span><span id='topic+subtree-deprecated'></span><span id='topic+subtree'></span><span id='topic+taxa.names-deprecated'></span><span id='topic+taxa.names'></span><span id='topic+taxa.ranks-deprecated'></span><span id='topic+taxa.ranks'></span><span id='topic+taxa.rollup-deprecated'></span><span id='topic+taxa.rollup'></span><span id='topic+taxonomy-deprecated'></span><span id='topic+taxonomy'></span><span id='topic+tips-deprecated'></span><span id='topic+tips'></span><span id='topic+unifrac-deprecated'></span><span id='topic+unifrac'></span><span id='topic+write.biom-deprecated'></span><span id='topic+write.biom'></span><span id='topic+write.fasta-deprecated'></span><span id='topic+write.fasta'></span><span id='topic+write.tree-deprecated'></span><span id='topic+write.tree'></span><span id='topic+write.xlsx-deprecated'></span><span id='topic+write.xlsx'></span><span id='topic+as.percent-deprecated'></span><span id='topic+as.percent'></span><span id='topic+comments-deprecated'></span><span id='topic+comments'></span><span id='topic+depth-deprecated'></span><span id='topic+depth'></span><span id='topic+depths_barplot-deprecated'></span><span id='topic+depths_barplot'></span><span id='topic+has.phylogeny-deprecated'></span><span id='topic+has.phylogeny'></span><span id='topic+has.sequences-deprecated'></span><span id='topic+has.sequences'></span><span id='topic+id-deprecated'></span><span id='topic+id'></span><span id='topic+is.rarefied-deprecated'></span><span id='topic+is.rarefied'></span><span id='topic+repair-deprecated'></span><span id='topic+repair'></span><span id='topic+sample_subset-deprecated'></span><span id='topic+sample_subset'></span><span id='topic+sample.sums-deprecated'></span><span id='topic+sample.sums'></span><span id='topic+taxa_max-deprecated'></span><span id='topic+taxa_max'></span><span id='topic+taxa.means-deprecated'></span><span id='topic+taxa.means'></span><span id='topic+taxa.sums-deprecated'></span><span id='topic+taxa.sums'></span><span id='topic+top.taxa-deprecated'></span><span id='topic+top.taxa'></span><span id='topic+top_taxa-deprecated'></span><span id='topic+top_taxa'></span><span id='topic+comments-set-deprecated'></span><span id='topic+comments+3C-'></span><span id='topic+counts-set-deprecated'></span><span id='topic+counts+3C-'></span><span id='topic+id-set-deprecated'></span><span id='topic+id+3C-'></span><span id='topic+metadata-set-deprecated'></span><span id='topic+metadata+3C-'></span><span id='topic+phylogeny-set-deprecated'></span><span id='topic+phylogeny+3C-'></span><span id='topic+sample.names-set-deprecated'></span><span id='topic+sample.names+3C-'></span><span id='topic+sequences-set-deprecated'></span><span id='topic+sequences+3C-'></span><span id='topic+taxa.names-set-deprecated'></span><span id='topic+taxa.names+3C-'></span><span id='topic+taxa.ranks-set-deprecated'></span><span id='topic+taxa.ranks+3C-'></span><span id='topic+taxonomy-set-deprecated'></span><span id='topic+taxonomy+3C-'></span>

<h3>Description</h3>

<p>The functions listed below are deprecated and will be defunct in
the near future. When possible, alternative functions with similar
functionality are also mentioned. Help pages for deprecated functions are
available at <code>help("&lt;function&gt;-deprecated")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha.div(biom, rarefy = FALSE)

beta.div(
  biom,
  method = "Bray-Curtis",
  weighted = TRUE,
  tree = NULL,
  long = FALSE,
  md = FALSE
)

counts(biom)

info(biom)

metadata(biom, field = NULL, cleanup = FALSE)

nsamples(biom)

ntaxa(biom)

phylogeny(biom)

read.biom(src, tree = "auto", prune = FALSE)

read.fasta(file, ids = NULL)

read.tree(src)

sample.names(biom)

## S3 method for class 'rbiom'
select(
  .data,
  samples = NULL,
  nTop = NULL,
  nRandom = NULL,
  seed = 0,
  biom = NULL,
  ...
)

sequences(biom)

subtree(tree, tips)

taxa.names(biom)

taxa.ranks(biom)

taxa.rollup(
  biom,
  rank = "OTU",
  map = NULL,
  lineage = FALSE,
  sparse = FALSE,
  taxa = NULL,
  long = FALSE,
  md = FALSE
)

taxonomy(biom, ranks = NULL, unc = "asis")

tips(x)

unifrac(biom, weighted = TRUE, tree = NULL)

write.biom(biom, file, format = "json")

write.fasta(seqs, outfile = NULL)

write.tree(tree, file = NULL)

write.xlsx(biom, outfile, depth = 0.1, seed = 0)

as.percent(biom)

comments(biom)

depth(biom)

depths_barplot(
  biom,
  rline = TRUE,
  counts = TRUE,
  labels = TRUE,
  transform = "log10",
  ...
)

has.phylogeny(biom)

has.sequences(biom)

id(biom)

is.rarefied(biom)

repair(biom)

sample_subset(x, ...)

sample.sums(biom, long = FALSE, md = FALSE)

taxa_max(biom, rank = -1, lineage = FALSE, unc = "singly")

taxa.means(biom, rank = NULL)

taxa.sums(biom, rank = NULL)

top.taxa(biom, rank = "OTU", n = Inf)

top_taxa(biom, rank = "OTU", n = Inf)

comments(x) &lt;- value

counts(x) &lt;- value

id(x) &lt;- value

metadata(x) &lt;- value

phylogeny(x) &lt;- value

sample.names(x) &lt;- value

sequences(x) &lt;- value

taxa.names(x) &lt;- value

taxa.ranks(x) &lt;- value

taxonomy(x) &lt;- value
</code></pre>


<h3><code>alpha.div</code></h3>

<p>Use <code><a href="#topic+adiv_matrix">adiv_matrix()</a></code> or <code><a href="#topic+adiv_table">adiv_table()</a></code> instead.
</p>


<h3><code>beta.div</code></h3>

<p>Use <code><a href="#topic+bdiv_table">bdiv_table()</a></code> or <code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code> instead.
</p>


<h3><code>counts</code></h3>

<p>Use <code style="white-space: pre;">&#8288;$counts&#8288;</code> instead.
</p>


<h3><code>info</code></h3>

<p>Use <code>biom$id</code>, <code>biom$comment</code>, etc instead.
</p>


<h3><code>metadata</code></h3>

<p>Use <code>biom$metadata</code> or <code>pull(biom, field)</code> instead.
</p>


<h3><code>nsamples</code></h3>

<p>Use <code>biom$n_samples</code> instead.
</p>


<h3><code>ntaxa</code></h3>

<p>Use <code>biom$n_otus</code> instead.
</p>


<h3><code>phylogeny</code></h3>

<p>Use <code>biom$tree</code> instead.
</p>


<h3><code>read.biom</code></h3>

<p>Use <code><a href="#topic+as_rbiom">as_rbiom()</a></code> instead.
</p>


<h3><code>read.fasta</code></h3>

<p>Use <code><a href="#topic+read_fasta">read_fasta()</a></code> instead.
</p>


<h3><code>read.tree</code></h3>

<p>Use <code><a href="#topic+read_tree">read_tree()</a></code> instead.
</p>


<h3><code>sample.names</code></h3>

<p>Use <code>biom$samples</code> instead.
</p>


<h3><code>select</code></h3>

<p>Use <code><a href="#topic+slice">slice()</a></code> instead.
</p>


<h3><code>sequences</code></h3>

<p>Use <code>biom$sequences</code> instead.
</p>


<h3><code>subtree</code></h3>

<p>Use <code><a href="#topic+tree_subset">tree_subset()</a></code> instead.
</p>


<h3><code>taxa.names</code></h3>

<p>Use <code>biom$otus</code> instead.
</p>


<h3><code>taxa.ranks</code></h3>

<p>Use <code>biom$ranks</code> instead.
</p>


<h3><code>taxa.rollup</code></h3>

<p>Use <code><a href="#topic+taxa_table">taxa_table()</a></code> <code><a href="#topic+taxa_matrix">taxa_matrix()</a></code> instead.
</p>


<h3><code>taxonomy</code></h3>

<p>Use <code style="white-space: pre;">&#8288;$taxonomy&#8288;</code> instead.
</p>


<h3><code>tips</code></h3>

<p>Use <code>tree$tip.label</code> instead.
</p>


<h3><code>unifrac</code></h3>

<p>Use <code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code> or <code><a href="#topic+bdiv_table">bdiv_table()</a></code> instead.
For <code>weighted=TRUE</code>, returns non-normalized values.
</p>


<h3><code>write.biom</code></h3>

<p>Use <code><a href="#topic+write_biom">write_biom()</a></code> instead.
</p>


<h3><code>write.fasta</code></h3>

<p>Use <code><a href="#topic+write_fasta">write_fasta()</a></code> instead.
</p>


<h3><code>write.tree</code></h3>

<p>Use <code><a href="#topic+write_tree">write_tree()</a></code> instead.
</p>


<h3><code>write.xlsx</code></h3>

<p>Use <code><a href="#topic+write_xlsx">write_xlsx()</a></code> instead.
</p>


<h3><code>as.percent</code></h3>

<p>Use <code>biom$counts %&lt;&gt;% rescale_cols()</code> instead.
</p>


<h3><code>comments</code></h3>

<p>Use <code>biom$comment</code> instead.
</p>


<h3><code>depth</code></h3>

<p>Use <code><a href="#topic+sample_sums">sample_sums()</a></code> instead.
</p>


<h3><code>depths_barplot</code></h3>

<p>Use <code><a href="#topic+rare_stacked">rare_stacked()</a></code> instead.
</p>


<h3><code>has.phylogeny</code></h3>

<p>Use <code>!is.null(biom$tree)</code> instead.
</p>


<h3><code>has.sequences</code></h3>

<p>Use <code>!is.null(biom$sequences)</code> instead.
</p>


<h3><code>id</code></h3>

<p>Use <code>biom$id</code> instead.
</p>


<h3><code>is.rarefied</code></h3>

<p>Use <code>!is.null(biom$depth)</code> instead.
</p>


<h3><code>repair</code></h3>

<p>Use <code>as_rbiom(as.list(biom))</code> instead.
</p>


<h3><code>sample_subset</code></h3>

<p>Use <code>biom$metadata %&lt;&gt;% base::subset()</code> instead.
</p>


<h3><code>sample.sums</code></h3>

<p>Use <code><a href="#topic+sample_sums">sample_sums()</a></code> or <code><a href="#topic+adiv_table">adiv_table()</a></code> instead.
</p>


<h3><code>taxa_max</code></h3>

<p>Use <code>taxa_apply(biom, max, sort = 'desc')</code> instead.
</p>


<h3><code>taxa.means</code></h3>

<p>Use <code><a href="#topic+taxa_means">taxa_means()</a></code> instead.
</p>


<h3><code>taxa.sums</code></h3>

<p>Use <code><a href="#topic+taxa_sums">taxa_sums()</a></code> instead.
</p>


<h3><code>top.taxa</code></h3>

<p>Use <code><a href="#topic+taxa_sums">taxa_sums()</a></code> instead.
</p>


<h3><code>top_taxa</code></h3>

<p>Use <code><a href="#topic+taxa_sums">taxa_sums()</a></code> instead.
</p>


<h3><code>comments-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$comment &lt;-&#8288;</code> instead.
</p>


<h3><code>counts-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$counts &lt;-&#8288;</code> instead.
</p>


<h3><code>id-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$id &lt;-&#8288;</code> instead.
</p>


<h3><code>metadata-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$metadata &lt;-&#8288;</code> instead.
</p>


<h3><code>phylogeny-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$tree &lt;-&#8288;</code> instead.
</p>


<h3><code>sample.names-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$samples &lt;-&#8288;</code> instead.
</p>


<h3><code>sequences-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$sequences &lt;-&#8288;</code> instead.
</p>


<h3><code>taxa.names-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$otus &lt;-&#8288;</code> instead.
</p>


<h3><code>taxa.ranks-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$ranks &lt;-&#8288;</code> instead.
</p>


<h3><code>taxonomy-set</code></h3>

<p>Use <code style="white-space: pre;">&#8288;biom$taxonomy &lt;-&#8288;</code> instead.
</p>

<hr>
<h2 id='read_biom'>Parse counts, metadata, taxonomy, and phylogeny from a BIOM file.</h2><span id='topic+read_biom'></span>

<h3>Description</h3>

<p>Parse counts, metadata, taxonomy, and phylogeny from a BIOM file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_biom(src, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_biom_+3A_src">src</code></td>
<td>
<p>Input data as either a file path, URL, or JSON string.
BIOM files can be formatted according to
version 1.0 (JSON) or 2.1 (HDF5)
<a href="http://biom-format.org/documentation/">specifications</a>, or as
classical tabular format. URLs must begin with <code>http://</code>,
<code>https://</code>, <code>ftp://</code>, or <code>ftps://</code>. JSON files must
have <code>{</code> as their first character. Compressed (gzip or bzip2)
BIOM files are also supported. NOTE: to read HDF5 formatted BIOM
files, the BioConductor R package <code>rhdf5</code> must be installed.</p>
</td></tr>
<tr><td><code id="read_biom_+3A_...">...</code></td>
<td>
<p>Properties to set in the new rbiom object, for example,
<code>metadata</code>, <code>id</code>, <code>comment</code>, or <code>tree</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>See Also</h3>

<p><code>as_rbiom()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)

    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read_biom(infile)

    print(biom)

    # Taxa Abundances
    biom$counts[1:4,1:10] %&gt;% as.matrix()
    
    biom$taxonomy %&gt;% head()

    # Metadata
    biom$metadata %&gt;% head()
    
    table(biom$metadata$Sex, biom$metadata$`Body Site`)
    
    sprintf("Mean age: %.1f", mean(biom$metadata$Age))

    # Phylogenetic tree
    biom$tree %&gt;%
      tree_subset(1:10) %&gt;%
      plot()

</code></pre>

<hr>
<h2 id='read_fasta'>Parse a fasta file into a named character vector.</h2><span id='topic+read_fasta'></span>

<h3>Description</h3>

<p>Parse a fasta file into a named character vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fasta(file, ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_fasta_+3A_file">file</code></td>
<td>
<p>A file/URL with fasta-formatted sequences. Can optionally be
compressed with gzip, bzip2, xz, or lzma.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_ids">ids</code></td>
<td>
<p>Character vector of IDs to retrieve. The default, <code>NULL</code>,
will retrieve everything.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named character vector in which names are the fasta headers and
values are the sequences.
</p>

<hr>
<h2 id='read_tree'>Read a newick formatted phylogenetic tree.</h2><span id='topic+read_tree'></span>

<h3>Description</h3>

<p>A phylogenetic tree is required for computing UniFrac distance matrices.
You can load a tree from a file or by providing the tree string directly.
This tree must be in Newick format, also known as parenthetic format and
New Hampshire format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_tree(src, underscores = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_tree_+3A_src">src</code></td>
<td>
<p>Input data as either a file path, URL, or Newick string.
Compressed (gzip or bzip2) files are also supported.</p>
</td></tr>
<tr><td><code id="read_tree_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phylo</code> class object representing the tree.
</p>


<h3>See Also</h3>

<p>Other phylogeny: 
<code><a href="#topic+tree_subset">tree_subset</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "newick.tre", package = "rbiom")
    tree &lt;- read_tree(infile)
    print(tree)
    
    tree &lt;- read_tree("
        (A:0.99,((B:0.87,C:0.89):0.51,(((D:0.16,(E:0.83,F:0.96)
        :0.94):0.69,(G:0.92,(H:0.62,I:0.85):0.54):0.23):0.74,J:0.1
        2):0.43):0.67);")
    plot(tree)

</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+.'></span><span id='topic+availableCores'></span><span id='topic+ddply'></span><span id='topic+ldply'></span><span id='topic+llply'></span><span id='topic++25+3E+25'></span><span id='topic++25+3C+3E+25'></span><span id='topic+glimpse'></span><span id='topic+na.omit'></span><span id='topic+left_join'></span><span id='topic+mutate'></span><span id='topic+pull'></span><span id='topic+relocate'></span><span id='topic+rename'></span><span id='topic+select'></span><span id='topic+slice'></span><span id='topic+slice_head'></span><span id='topic+slice_tail'></span><span id='topic+slice_min'></span><span id='topic+slice_max'></span><span id='topic+slice_sample'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>dplyr</dt><dd><p><code><a href="dplyr.html#topic+mutate-joins">left_join</a></code>, <code><a href="dplyr.html#topic+mutate">mutate</a></code>, <code><a href="dplyr.html#topic+pull">pull</a></code>, <code><a href="dplyr.html#topic+relocate">relocate</a></code>, <code><a href="dplyr.html#topic+rename">rename</a></code>, <code><a href="dplyr.html#topic+select">select</a></code>, <code><a href="dplyr.html#topic+slice">slice</a></code>, <code><a href="dplyr.html#topic+slice">slice_head</a></code>, <code><a href="dplyr.html#topic+slice">slice_max</a></code>, <code><a href="dplyr.html#topic+slice">slice_min</a></code>, <code><a href="dplyr.html#topic+slice">slice_sample</a></code>, <code><a href="dplyr.html#topic+slice">slice_tail</a></code></p>
</dd>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+compound">%&lt;&gt;%</a></code>, <code><a href="magrittr.html#topic+pipe">%&gt;%</a></code></p>
</dd>
<dt>parallelly</dt><dd><p><code><a href="parallelly.html#topic+availableCores">availableCores</a></code></p>
</dd>
<dt>pillar</dt><dd><p><code><a href="pillar.html#topic+glimpse">glimpse</a></code></p>
</dd>
<dt>plyr</dt><dd><p><code><a href="plyr.html#topic+quoted">.</a></code>, <code><a href="plyr.html#topic+ddply">ddply</a></code>, <code><a href="plyr.html#topic+ldply">ldply</a></code>, <code><a href="plyr.html#topic+llply">llply</a></code></p>
</dd>
<dt>stats</dt><dd><p><code><a href="stats.html#topic+na.fail">na.omit</a></code></p>
</dd>
</dl>

<hr>
<h2 id='sample_sums'>Summarize the taxa observations in each sample.</h2><span id='topic+sample_sums'></span><span id='topic+sample_apply'></span>

<h3>Description</h3>

<p>Summarize the taxa observations in each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_sums(biom, rank = -1, sort = NULL, unc = "singly")

sample_apply(biom, FUN, rank = -1, sort = NULL, unc = "singly", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_sums_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="sample_sums_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="sample_sums_+3A_sort">sort</code></td>
<td>
<p>Sort the result. Options: <code>NULL</code> - don't sort; <code>"asc"</code> - in
ascending order (smallest to largest); <code>"desc"</code> - in descending order
(largest to smallest). Ignored when the result is not a simple
numeric vector. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="sample_sums_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="sample_sums_+3A_fun">FUN</code></td>
<td>
<p>The function to apply to each column of <code>taxa_matrix()</code>.</p>
</td></tr>
<tr><td><code id="sample_sums_+3A_...">...</code></td>
<td>
<p>Optional arguments to <code>FUN</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For <code>sample_sums</code>, A named numeric vector of the number of
observations in each sample. For <code>sample_apply</code>, a named vector or
list with the results of <code>FUN</code>. The names are the taxa IDs.
</p>


<h3>See Also</h3>

<p>Other samples: 
<code><a href="#topic+pull.rbiom">pull.rbiom</a>()</code>
</p>
<p>Other rarefaction: 
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>
</p>
<p>Other taxa_abundance: 
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    library(ggplot2)
    
    sample_sums(hmp50, sort = 'asc') %&gt;% head()
    
    # Unique OTUs and "cultured" classes per sample
    nnz &lt;- function (x) sum(x &gt; 0) # number of non-zeroes
    sample_apply(hmp50, nnz, 'otu') %&gt;% head()
    sample_apply(hmp50, nnz, 'class', unc = 'drop') %&gt;% head()
    
    # Number of reads in each sample's most abundant family
    sample_apply(hmp50, base::max, 'f', sort = 'desc') %&gt;% head()
    
    ggplot() + geom_histogram(aes(x=sample_sums(hmp50)), bins = 20)
</code></pre>

<hr>
<h2 id='slice_metadata'>Subset to a specific number of samples.</h2><span id='topic+slice_metadata'></span><span id='topic+slice.rbiom'></span><span id='topic+slice_head.rbiom'></span><span id='topic+slice_tail.rbiom'></span><span id='topic+slice_min.rbiom'></span><span id='topic+slice_max.rbiom'></span><span id='topic+slice_sample.rbiom'></span>

<h3>Description</h3>

<p>Subset to a specific number of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
slice(.data, ..., .by = NULL, .preserve = FALSE, clone = TRUE)

## S3 method for class 'rbiom'
slice_head(.data, n, prop, by = NULL, clone = TRUE, ...)

## S3 method for class 'rbiom'
slice_tail(.data, n, prop, by = NULL, clone = TRUE, ...)

## S3 method for class 'rbiom'
slice_min(
  .data,
  order_by,
  n,
  prop,
  by = NULL,
  with_ties = TRUE,
  na_rm = FALSE,
  clone = TRUE,
  ...
)

## S3 method for class 'rbiom'
slice_max(
  .data,
  order_by,
  n,
  prop,
  by = NULL,
  with_ties = TRUE,
  na_rm = FALSE,
  clone = TRUE,
  ...
)

## S3 method for class 'rbiom'
slice_sample(
  .data,
  n,
  prop,
  by = NULL,
  weight_by = NULL,
  replace = FALSE,
  clone = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slice_metadata_+3A_.data">.data</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_...">...</code></td>
<td>
<p>For <code>slice()</code>, integer row indexes. For other <code style="white-space: pre;">&#8288;slice_*()&#8288;</code>
functions, not used. See <code><a href="dplyr.html#topic+slice">dplyr::slice()</a></code>.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_.by">.by</code>, <code id="slice_metadata_+3A_by">by</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>&lt;<code><a href="dplyr.html#topic+dplyr_tidy_select">tidy-select</a></code>&gt; Optionally, a selection of columns to
group by for just this operation, functioning as an alternative to <code><a href="dplyr.html#topic+group_by">group_by()</a></code>. For
details and examples, see <a href="dplyr.html#topic+dplyr_by">?dplyr_by</a>.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_.preserve">.preserve</code></td>
<td>
<p>Relevant when the <code>.data</code> input is grouped.
If <code>.preserve = FALSE</code> (the default), the grouping structure
is recalculated based on the resulting data, otherwise the grouping is kept as is.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_clone">clone</code></td>
<td>
<p>Create a copy of <code>biom</code> before modifying. If <code>FALSE</code>, <code>biom</code>
is modified in place as a side-effect. See <a href="#topic+speed">speed ups</a> for
use cases. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_n">n</code>, <code id="slice_metadata_+3A_prop">prop</code></td>
<td>
<p>Provide either <code>n</code>, the number of rows, or <code>prop</code>, the
proportion of rows to select. If neither are supplied, <code>n = 1</code> will be
used. If <code>n</code> is greater than the number of rows in the group
(or <code>prop &gt; 1</code>), the result will be silently truncated to the group size.
<code>prop</code> will be rounded towards zero to generate an integer number of
rows.
</p>
<p>A negative value of <code>n</code> or <code>prop</code> will be subtracted from the group
size. For example, <code>n = -2</code> with a group of 5 rows will select 5 - 2 = 3
rows; <code>prop = -0.25</code> with 8 rows will select 8 * (1 - 0.25) = 6 rows.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_order_by">order_by</code></td>
<td>
<p>&lt;<code><a href="rlang.html#topic+args_data_masking">data-masking</a></code>&gt; Variable or
function of variables to order by. To order by multiple variables, wrap
them in a data frame or tibble.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_with_ties">with_ties</code></td>
<td>
<p>Should ties be kept together? The default, <code>TRUE</code>,
may return more rows than you request. Use <code>FALSE</code> to ignore ties,
and return the first <code>n</code> rows.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_na_rm">na_rm</code></td>
<td>
<p>Should missing values in <code>order_by</code> be removed from the result?
If <code>FALSE</code>, <code>NA</code> values are sorted to the end (like in <code><a href="dplyr.html#topic+arrange">arrange()</a></code>), so
they will only be included if there are insufficient non-missing values to
reach <code>n</code>/<code>prop</code>.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_weight_by">weight_by</code></td>
<td>
<p>&lt;<code><a href="rlang.html#topic+args_data_masking">data-masking</a></code>&gt; Sampling
weights. This must evaluate to a vector of non-negative numbers the same
length as the input. Weights are automatically standardised to sum to 1.</p>
</td></tr>
<tr><td><code id="slice_metadata_+3A_replace">replace</code></td>
<td>
<p>Should sampling be performed with (<code>TRUE</code>) or without
(<code>FALSE</code>, the default) replacement.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>See Also</h3>

<p>Other transformations: 
<code><a href="#topic+modify_metadata">modify_metadata</a></code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+subset">subset</a>()</code>,
<code><a href="#topic+with">with</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # The last 3 samples in the metadata table.
    biom &lt;- slice_tail(hmp50, n = 3)
    biom$metadata
    
    # The 3 oldest subjects sampled.
    biom &lt;- slice_max(hmp50, Age, n = 3)
    biom$metadata
    
    # Pick 3 samples at random.
    biom &lt;- slice_sample(hmp50, n = 3)
    biom$metadata

</code></pre>

<hr>
<h2 id='speed'>Speed Ups.</h2><span id='topic+speed'></span>

<h3>Description</h3>

<p>When working with very large datasets, you can make use of these tips and
tricks to speed up operations on rbiom objects.
</p>


<h3>Skip Cloning</h3>

<p>Functions that modify rbiom objects, like <code><a href="#topic+subset">subset()</a></code> and <code><a href="#topic+rarefy">rarefy()</a></code>, will
automatically clone the object before modifying it. This is to make these
functions behave as most R users would expect - but at a performance trade
off.
</p>
<p>Rather than:
</p>
<div class="sourceCode r"><pre>biom &lt;- subset(biom, ...)
biom &lt;- rarefy(biom)
</pre></div>
<p>Modify <code>biom</code> in place like this:
</p>
<div class="sourceCode r"><pre>subset(biom, clone = FALSE, ...)
rarefy(biom, clone = FALSE)

# Or:
biom$metadata %&lt;&gt;% subset(...)
biom$counts %&lt;&gt;% rarefy_cols()
</pre></div>


<h3>Drop Components</h3>



<h4>Sequences</h4>

<p>Reference sequences for OTUs will be imported along with the rest of your
dataset and stored in <code style="white-space: pre;">&#8288;$sequences&#8288;</code>. However, rbiom doesn't currently use
these sequences for anything (except writing them back out with
<code><a href="#topic+write_biom">write_biom()</a></code> or <code><a href="#topic+write_fasta">write_fasta()</a></code>).
</p>
<p>You can delete them from your rbiom object with:
</p>
<div class="sourceCode r"><pre>biom$sequences &lt;- NULL
</pre></div>



<h4>Tree</h4>

<p>The phylogenetic reference tree for OTUs is only used for calculating
UniFrac distances. If you aren't using UniFrac, the tree can be dropped
from the rbiom object with:
</p>
<div class="sourceCode r"><pre>biom$tree &lt;- NULL
</pre></div>
<p>Alternatively, you can store the tree separately from the rbiom object and
provide it to just the functions that use it. For example:
</p>
<div class="sourceCode r"><pre>tree &lt;- biom$tree
biom$tree &lt;- NULL
dm &lt;- bdiv_distmat(biom, 'unifrac', tree = tree)
</pre></div>



<h3>Increase Caching</h3>

<p>Caching is enabled by default - up to 20 MB per R session.
</p>
<p>For large datasets, increasing the cache size can help. The size is
specified in bytes by an R option or environment variable.
</p>
<div class="sourceCode r"><pre>options(rbiom.cache_size=200 * 1024 ^ 2) # 200 MB
Sys.setenv(RBIOM_CACHE_SIZE=1024 ^ 3)    # 1 GB
</pre></div>
<p>You can also specify a cache directory where results can be preserved from
one R session to the next.
</p>
<div class="sourceCode r"><pre>options(rbiom.cache_dir=tools::R_user_dir("rbiom", "cache"))
Sys.setenv(RBIOM_CACHE_DIR="~/rbiom_cache")
</pre></div>
<p>Other quick notes about caching:
</p>

<ul>
<li><p> Setting the cache directory to <code>"FALSE"</code> will disable caching.
</p>
</li>
<li><p> R options will override environment variables.
</p>
</li>
<li><p> The key hash algorithm can be set with <code>options(rbiom.cache_hash=rlang::hash)</code>.
</p>
</li></ul>



<h3>Summary Layers</h3>

<p>The figure-generating functions allow you to display every data point.
However, when you have thousands of data points, rendering every single one
can be slow. Instead, set the <code>layers</code> parameter to use other options.
</p>
<div class="sourceCode r"><pre>adiv_boxplot(biom, layers = "bl")  # bar, linerange
adiv_corrplot(biom, layers = "tc") # trend, confidence
bdiv_ord_plot(biom, layers = "e")  # ellipse
</pre></div>

<hr>
<h2 id='stats_boxplot'>Visualize categorical metadata effects on numeric values.</h2><span id='topic+stats_boxplot'></span>

<h3>Description</h3>

<p>Visualize categorical metadata effects on numeric values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stats_boxplot(
  df,
  x = NULL,
  y = attr(df, "response"),
  layers = "x",
  stat.by = x,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  patterns = FALSE,
  test = "auto",
  flip = FALSE,
  stripe = NULL,
  ci = "ci",
  level = 0.95,
  p.adj = "fdr",
  p.top = Inf,
  outliers = NULL,
  xlab.angle = "auto",
  p.label = 0.05,
  caption = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stats_boxplot_+3A_df">df</code></td>
<td>
<p>The dataset (data.frame or tibble object). &quot;Dataset fields&quot;
mentioned below should match column names in <code>df</code>. Required.</p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_x">x</code></td>
<td>
<p>A categorical metadata column name to use for the x-axis. Or
<code>NULL</code>, which groups all samples into a single category.</p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_y">y</code></td>
<td>
<p>A numeric metadata column name to use for the y-axis.
Default: <code>attr(df, 'response')</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("bar", "box" ("x"), "violin", "dot", "strip", "crossbar", "errorbar", "linerange", "pointrange")</code>.
Single letter abbreviations are also accepted. For instance,
<code>c("box", "dot")</code> is equivalent to <code>c("x", "d")</code> and <code>"xd"</code>.
Default: <code>"x"</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_patterns">patterns</code></td>
<td>
<p>Patterns for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, pattern
names (<code>"brick"</code>, <code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, etc), or a named
vector mapping groups to specific pattern names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'auto'</code> or <code>'none'</code>. <code>'auto'</code>
will choose Wilcox or Kruskal-Wallis depending on the number of
groups.</p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_flip">flip</code></td>
<td>
<p>Transpose the axes, so that taxa are present as rows instead
of columns. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_stripe">stripe</code></td>
<td>
<p>Shade every other x position. Default: <em>same as flip</em></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_ci">ci</code></td>
<td>
<p>How to calculate min/max of the <b>crossbar</b>,
<b>errorbar</b>, <b>linerange</b>, and <b>pointrange</b> layers.
Options are: <code>"ci"</code> (confidence interval), <code>"range"</code>,
<code>"sd"</code> (standard deviation), <code>"se"</code> (standard error), and
<code>"mad"</code> (median absolute deviation).
The center mark of <b>crossbar</b> and <b>pointrange</b> represents
the mean, except for <code>"mad"</code> in which case it represents the median.
Default: <code>"ci"</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_p.top">p.top</code></td>
<td>
<p>Only display taxa with the most significant differences in
abundance. If <code>p.top</code> is &gt;= 1, then the <code>p.top</code> most
significant taxa are displayed. If <code>p.top</code> is less than one, all
taxa with an adjusted p-value &lt;= <code>p.top</code> are displayed.
Recommended to be used in combination with the <code>taxa</code> parameter
to set a lower bound on the mean abundance of considered taxa.
Default: <code>Inf</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_outliers">outliers</code></td>
<td>
<p>Show boxplot outliers? <code>TRUE</code> to always show.
<code>FALSE</code> to always hide. <code>NULL</code> to only hide them when
overlaying a dot or strip chart.  Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_p.label">p.label</code></td>
<td>
<p>Minimum adjusted p-value to display on the plot with a
bracket.
</p>

<dl>
<dt><code>p.label = 0.05</code> - </dt><dd><p> Show p-values that are &lt;= 0.05. </p>
</dd>
<dt><code>p.label = 0</code> - </dt><dd><p> Don't show any p-values on the plot. </p>
</dd>
<dt><code>p.label = 1</code> - </dt><dd><p> Show all p-values on the plot. </p>
</dd>
</dl>

<p>If a numeric vector with more than one value is
provided, they will be used as breaks for asterisk notation.
Default: <code>0.05</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="stats_boxplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>d.size = 2</code> ensures only the points on the
<b>dot</b> layer have their size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Patterns are added using the fillpattern R package. Options are <code>"brick"</code>,
<code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, <code>"herringbone"</code>, <code>"hexagon"</code>, <code>"octagon"</code>,
<code>"rain"</code>, <code>"saw"</code>, <code>"shingle"</code>, <code>"rshingle"</code>, <code>"stripe"</code>, and <code>"wave"</code>,
optionally abbreviated and/or suffixed with modifiers. For example,
<code>"hex10_sm"</code> for the hexagon pattern rotated 10 degrees and shrunk by 2x.
See <code><a href="fillpattern.html#topic+fill_pattern">fillpattern::fill_pattern()</a></code> for complete documentation of options.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    df &lt;- adiv_table(rarefy(hmp50))
    stats_boxplot(df, x = "Body Site")
    stats_boxplot(df, x = "Sex", stat.by = "Body Site", layers = "be")
</code></pre>

<hr>
<h2 id='stats_corrplot'>Visualize regression with scatterplots and trendlines.</h2><span id='topic+stats_corrplot'></span>

<h3>Description</h3>

<p>Visualize regression with scatterplots and trendlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stats_corrplot(
  df,
  x,
  y = attr(df, "response"),
  layers = "tc",
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  p.adj = "fdr",
  p.top = Inf,
  alt = "!=",
  mu = 0,
  caption = TRUE,
  check = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stats_corrplot_+3A_df">df</code></td>
<td>
<p>The dataset (data.frame or tibble object). &quot;Dataset fields&quot;
mentioned below should match column names in <code>df</code>. Required.</p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_x">x</code></td>
<td>
<p>Dataset field with the x-axis values. Equivalent to the <code>regr</code>
argument in <code><a href="#topic+stats_table">stats_table()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_y">y</code></td>
<td>
<p>A numeric metadata column name to use for the y-axis.
Default: <code>attr(df, 'response')</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("trend", "confidence", "point", "name", "residual")</code>. Single
letter abbreviations are also accepted. For instance,
<code>c("trend", "point")</code> is equivalent to <code>c("t", "p")</code> and <code>"tp"</code>.
Default: <code>"tc"</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'none'</code>, <code>'emmeans'</code>, or
<code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_p.top">p.top</code></td>
<td>
<p>Only display taxa with the most significant differences in
abundance. If <code>p.top</code> is &gt;= 1, then the <code>p.top</code> most
significant taxa are displayed. If <code>p.top</code> is less than one, all
taxa with an adjusted p-value &lt;= <code>p.top</code> are displayed.
Recommended to be used in combination with the <code>taxa</code> parameter
to set a lower bound on the mean abundance of considered taxa.
Default: <code>Inf</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="stats_corrplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>p.size = 2</code> ensures only the points have their
size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- subset(hmp50, `Body Site` %in% c('Saliva', 'Stool'))
    df   &lt;- adiv_table(rarefy(biom))
    stats_corrplot(df, "age", stat.by = "body")
    stats_corrplot(
      df       = df, 
      x        = "Age", 
      stat.by  = "Body Site", 
      facet.by = "Sex", 
      layers   = "trend" )
</code></pre>

<hr>
<h2 id='stats_table'>Run non-parametric statistics on a data.frame.</h2><span id='topic+stats_table'></span>

<h3>Description</h3>

<p>A simple interface to lower-level statistics functions, including
<code><a href="stats.html#topic+wilcox.test">stats::wilcox.test()</a></code>, <code><a href="stats.html#topic+kruskal.test">stats::kruskal.test()</a></code>, <code><a href="emmeans.html#topic+emmeans">emmeans::emmeans()</a></code>,
and <code><a href="emmeans.html#topic+emtrends">emmeans::emtrends()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stats_table(
  df,
  regr = NULL,
  resp = attr(df, "response"),
  stat.by = NULL,
  split.by = NULL,
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  alt = "!=",
  mu = 0,
  p.adj = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stats_table_+3A_df">df</code></td>
<td>
<p>The dataset (data.frame or tibble object). &quot;Dataset fields&quot;
mentioned below should match column names in <code>df</code>. Required.</p>
</td></tr>
<tr><td><code id="stats_table_+3A_regr">regr</code></td>
<td>
<p>Dataset field with the x-axis (independent; predictive)
values. Must be numeric. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_resp">resp</code></td>
<td>
<p>Dataset field with the y-axis (dependent; response) values,
such as taxa abundance or alpha diversity.
Default: <code>attr(df, 'response')</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_split.by">split.by</code></td>
<td>
<p>Dataset field(s) that the data should be split by prior to
any calculations. Must be categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'wilcox'</code>, <code>'kruskal'</code>,
<code>'emmeans'</code>, or <code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="stats_table_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="stats_table_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble data.frame with fields from the table below. This tibble
object provides the <code style="white-space: pre;">&#8288;$code&#8288;</code> operator to print the R code used to generate
the statistics.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Field</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean </td><td style="text-align: left;"> Estimated marginal mean. See <code><a href="emmeans.html#topic+emmeans">emmeans::emmeans()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean.diff </td><td style="text-align: left;"> Difference in means. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope </td><td style="text-align: left;"> Trendline slope. See <code><a href="emmeans.html#topic+emtrends">emmeans::emtrends()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope.diff </td><td style="text-align: left;"> Difference in slopes. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .h1 </td><td style="text-align: left;"> Alternate hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .p.val </td><td style="text-align: left;"> Probability that null hypothesis is correct. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.p </td><td style="text-align: left;"> <code>.p.val</code> after adjusting for multiple comparisons. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .effect.size </td><td style="text-align: left;"> Effect size. See <code><a href="emmeans.html#topic+eff_size">emmeans::eff_size()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .lower </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .upper </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .se </td><td style="text-align: left;"> Standard error. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .n </td><td style="text-align: left;"> Number of samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .df </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .stat </td><td style="text-align: left;"> Wilcoxon or Kruskal-Wallis rank sum statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .t.ratio </td><td style="text-align: left;"> <code>.mean</code> / <code>.se</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .r.sqr </td><td style="text-align: left;"> Percent of variation explained by the model. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.r </td><td style="text-align: left;"> <code>.r.sqr</code>, taking degrees of freedom into account. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .aic </td><td style="text-align: left;"> Akaike Information Criterion (predictive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .bic </td><td style="text-align: left;"> Bayesian Information Criterion (descriptive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .loglik </td><td style="text-align: left;"> Log-likelihood goodness-of-fit score. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .fit.p </td><td style="text-align: left;"> P-value for observing this fit by chance. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>See Also</h3>

<p>Other stats_tables: 
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    df &lt;- taxa_table(biom, rank = "Family")
    stats_table(df, stat.by = "Body Site")[,1:6]
    
    df &lt;- adiv_table(biom)
    stats_table(df, stat.by = "Sex", split.by = "Body Site")[,1:7]
</code></pre>

<hr>
<h2 id='subset'>Subset an rbiom object by sample names, OTU names, metadata, or taxonomy.</h2><span id='topic+subset'></span><span id='topic+subset.rbiom'></span><span id='topic++5B.rbiom'></span><span id='topic+na.omit.rbiom'></span><span id='topic+subset_taxa'></span>

<h3>Description</h3>

<p>Dropping samples or OTUs will lead to observations being removed from the
OTU matrix (<code>biom$counts</code>). OTUs and samples with zero observations are
automatically removed from the rbiom object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
subset(x, subset, clone = TRUE, ...)

## S3 method for class 'rbiom'
x[i, j, ..., clone = TRUE, drop = FALSE]

## S3 method for class 'rbiom'
na.omit(object, fields = ".all", clone = TRUE, ...)

subset_taxa(x, subset, clone = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset_+3A_x">x</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="subset_+3A_subset">subset</code></td>
<td>
<p>Logical expression for rows to keep. See <code><a href="base.html#topic+subset">base::subset()</a></code>.</p>
</td></tr>
<tr><td><code id="subset_+3A_clone">clone</code></td>
<td>
<p>Create a copy of <code>biom</code> before modifying. If <code>FALSE</code>, <code>biom</code>
is modified in place as a side-effect. See <a href="#topic+speed">speed ups</a> for
use cases. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="subset_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset_+3A_i">i</code>, <code id="subset_+3A_j">j</code></td>
<td>
<p>The sample or OTU names to keep. Or a logical/integer vector
indicating which sample names from <code>biom$samples</code> or <code>biom$otus</code> to
keep. Subsetting with <code style="white-space: pre;">&#8288;[i]&#8288;</code> takes <code>i</code> as samples, whereas <code style="white-space: pre;">&#8288;[i,j]&#8288;</code>
takes <code>i</code> as otus and <code>j</code> as samples (corresponding to <code style="white-space: pre;">&#8288;[rows, cols]&#8288;</code>
in the underlying <code>biom$counts</code> matrix).</p>
</td></tr>
<tr><td><code id="subset_+3A_drop">drop</code></td>
<td>
<p>Not used</p>
</td></tr>
<tr><td><code id="subset_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="subset_+3A_fields">fields</code></td>
<td>
<p>Which metadata field(s) to check for <code>NA</code>s, or <code>".all"</code> to
check all metadata fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rbiom_objects">rbiom object</a>.
</p>


<h3>See Also</h3>

<p>Other transformations: 
<code><a href="#topic+modify_metadata">modify_metadata</a></code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+slice_metadata">slice_metadata</a></code>,
<code><a href="#topic+with">with</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    library(dplyr)
    
    # Subset to specific samples
    biom &lt;- hmp50[c('HMP20', 'HMP42', 'HMP12')]
    biom$metadata
    
    # Subset to specific OTUs
    biom &lt;- hmp50[c('LtbAci52', 'UncO2012'),] # &lt;- Trailing ,
    biom$taxonomy
    
    # Subset to specific samples and OTUs
    biom &lt;- hmp50[c('LtbAci52', 'UncO2012'), c('HMP20', 'HMP42', 'HMP12')]
    as.matrix(biom)
    
    # Subset samples according to metadata
    biom &lt;- subset(hmp50, `Body Site` %in% c('Saliva') &amp; Age &lt; 25)
    biom$metadata
    
    # Subset OTUs according to taxonomy
    biom &lt;- subset_taxa(hmp50, Phylum == 'Cyanobacteria')
    biom$taxonomy
    
    # Remove samples with NA metadata values
    biom &lt;- mutate(hmp50, BS2 = na_if(`Body Site`, 'Saliva'))
    biom$metadata
    biom &lt;- na.omit(biom)
    biom$metadata

</code></pre>

<hr>
<h2 id='taxa_boxplot'>Visualize BIOM data with boxplots.</h2><span id='topic+taxa_boxplot'></span>

<h3>Description</h3>

<p>Visualize BIOM data with boxplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_boxplot(
  biom,
  x = NULL,
  rank = -1,
  layers = "x",
  taxa = 6,
  unc = "singly",
  other = FALSE,
  p.top = Inf,
  stat.by = x,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  patterns = FALSE,
  flip = FALSE,
  stripe = NULL,
  ci = "ci",
  level = 0.95,
  p.adj = "fdr",
  outliers = NULL,
  xlab.angle = "auto",
  p.label = 0.05,
  transform = "none",
  y.transform = "sqrt",
  caption = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_boxplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_x">x</code></td>
<td>
<p>A categorical metadata column name to use for the x-axis. Or
<code>NULL</code>, which puts taxa along the x-axis. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("bar", "box" ("x"), "violin", "dot", "strip", "crossbar", "errorbar", "linerange", "pointrange")</code>.
Single letter abbreviations are also accepted. For instance,
<code>c("box", "dot")</code> is equivalent to <code>c("x", "d")</code> and <code>"xd"</code>.
Default: <code>"x"</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_p.top">p.top</code></td>
<td>
<p>Only display taxa with the most significant differences in
abundance. If <code>p.top</code> is &gt;= 1, then the <code>p.top</code> most
significant taxa are displayed. If <code>p.top</code> is less than one, all
taxa with an adjusted p-value &lt;= <code>p.top</code> are displayed.
Recommended to be used in combination with the <code>taxa</code> parameter
to set a lower bound on the mean abundance of considered taxa.
Default: <code>Inf</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_patterns">patterns</code></td>
<td>
<p>Patterns for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, pattern
names (<code>"brick"</code>, <code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, etc), or a named
vector mapping groups to specific pattern names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_flip">flip</code></td>
<td>
<p>Transpose the axes, so that taxa are present as rows instead
of columns. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_stripe">stripe</code></td>
<td>
<p>Shade every other x position. Default: <em>same as flip</em></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_ci">ci</code></td>
<td>
<p>How to calculate min/max of the <b>crossbar</b>,
<b>errorbar</b>, <b>linerange</b>, and <b>pointrange</b> layers.
Options are: <code>"ci"</code> (confidence interval), <code>"range"</code>,
<code>"sd"</code> (standard deviation), <code>"se"</code> (standard error), and
<code>"mad"</code> (median absolute deviation).
The center mark of <b>crossbar</b> and <b>pointrange</b> represents
the mean, except for <code>"mad"</code> in which case it represents the median.
Default: <code>"ci"</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_outliers">outliers</code></td>
<td>
<p>Show boxplot outliers? <code>TRUE</code> to always show.
<code>FALSE</code> to always hide. <code>NULL</code> to only hide them when
overlaying a dot or strip chart.  Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_p.label">p.label</code></td>
<td>
<p>Minimum adjusted p-value to display on the plot with a
bracket.
</p>

<dl>
<dt><code>p.label = 0.05</code> - </dt><dd><p> Show p-values that are &lt;= 0.05. </p>
</dd>
<dt><code>p.label = 0</code> - </dt><dd><p> Don't show any p-values on the plot. </p>
</dd>
<dt><code>p.label = 1</code> - </dt><dd><p> Show all p-values on the plot. </p>
</dd>
</dl>

<p>If a numeric vector with more than one value is
provided, they will be used as breaks for asterisk notation.
Default: <code>0.05</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_y.transform">y.transform</code></td>
<td>
<p>The transformation to apply to the y-axis. Visualizing
differences of both high- and low-abundance taxa is best done with
a non-linear axis. Options are:
</p>

<dl>
<dt><code>"sqrt"</code> - </dt><dd><p> square-root transformation </p>
</dd>
<dt><code>"log1p"</code> - </dt><dd><p> log(y + 1) transformation </p>
</dd>
<dt><code>"none"</code> - </dt><dd><p> no transformation </p>
</dd>
</dl>

<p>These methods allow visualization of both high- and low-abundance
taxa simultaneously, without complaint about 'zero' count
observations. Default: <code>"sqrt"</code>
Use <code>xaxis.transform</code> or <code>yaxis.transform</code> to pass custom values
directly to ggplot2's <code style="white-space: pre;">&#8288;scale_*&#8288;</code> functions.</p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_boxplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>d.size = 2</code> ensures only the points on the
<b>dot</b> layer have their size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Patterns are added using the fillpattern R package. Options are <code>"brick"</code>,
<code>"chevron"</code>, <code>"fish"</code>, <code>"grid"</code>, <code>"herringbone"</code>, <code>"hexagon"</code>, <code>"octagon"</code>,
<code>"rain"</code>, <code>"saw"</code>, <code>"shingle"</code>, <code>"rshingle"</code>, <code>"stripe"</code>, and <code>"wave"</code>,
optionally abbreviated and/or suffixed with modifiers. For example,
<code>"hex10_sm"</code> for the hexagon pattern rotated 10 degrees and shrunk by 2x.
See <code><a href="fillpattern.html#topic+fill_pattern">fillpattern::fill_pattern()</a></code> for complete documentation of options.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    taxa_boxplot(biom, stat.by = "Body Site", stripe = TRUE)
    taxa_boxplot(biom, layers = "bed", rank = c("Phylum", "Genus"), flip = TRUE)
    taxa_boxplot(
      biom    = subset(biom, `Body Site` %in% c('Saliva', 'Stool')), 
      taxa    = 3, 
      layers  = "ps", 
      stat.by = "Body Site",
      colors  = c('Saliva' = "blue", 'Stool' = "red") )
    
</code></pre>

<hr>
<h2 id='taxa_clusters'>Cluster samples by taxa abundances k-means.</h2><span id='topic+taxa_clusters'></span>

<h3>Description</h3>

<p>Cluster samples by taxa abundances k-means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_clusters(biom, rank = ".otu", k = 5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_clusters_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_clusters_+3A_rank">rank</code></td>
<td>
<p>Which taxa rank to use. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>.otu</code>.</p>
</td></tr>
<tr><td><code id="taxa_clusters_+3A_k">k</code></td>
<td>
<p>Number of clusters. Default: <code>5L</code></p>
</td></tr>
<tr><td><code id="taxa_clusters_+3A_...">...</code></td>
<td>
<p>Passed on to <code>stats::kmeans()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric factor assigning samples to clusters.
</p>


<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>
<p>Other clustering: 
<code><a href="#topic+bdiv_clusters">bdiv_clusters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    biom$metadata$otu_cluster &lt;- taxa_clusters(biom)
    
    pull(biom, 'otu_cluster')[1:10]
    
    bdiv_ord_plot(biom, layers = "p", stat.by = "otu_cluster")
</code></pre>

<hr>
<h2 id='taxa_corrplot'>Visualize taxa abundance with scatterplots and trendlines.</h2><span id='topic+taxa_corrplot'></span>

<h3>Description</h3>

<p>Visualize taxa abundance with scatterplots and trendlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_corrplot(
  biom,
  x,
  rank = -1,
  layers = "tc",
  taxa = 6,
  lineage = FALSE,
  unc = "singly",
  other = FALSE,
  stat.by = NULL,
  facet.by = NULL,
  colors = TRUE,
  shapes = TRUE,
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  p.adj = "fdr",
  transform = "none",
  ties = "random",
  seed = 0,
  alt = "!=",
  mu = 0,
  caption = TRUE,
  check = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_corrplot_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_x">x</code></td>
<td>
<p>Dataset field with the x-axis values. Equivalent to the <code>regr</code>
argument in <code><a href="#topic+stats_table">stats_table()</a></code>. Required.</p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_layers">layers</code></td>
<td>
<p>One or more of
<code>c("trend", "confidence", "point", "name", "residual")</code>. Single
letter abbreviations are also accepted. For instance,
<code>c("trend", "point")</code> is equivalent to <code>c("t", "p")</code> and <code>"tp"</code>.
Default: <code>"tc"</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_colors">colors</code></td>
<td>
<p>How to color the groups. Options are:
</p>

<dl>
<dt><code>TRUE</code> - </dt><dd><p> Automatically select colorblind-friendly colors. </p>
</dd>
<dt><code>FALSE</code> or <code>NULL</code> - </dt><dd><p> Don't use colors. </p>
</dd>
<dt>a palette name - </dt><dd><p> Auto-select colors from this set. E.g. <code>"okabe"</code> </p>
</dd>
<dt>character vector - </dt><dd><p> Custom colors to use. E.g. <code>c("red", "#00FF00")</code> </p>
</dd>
<dt>named character vector - </dt><dd><p> Explicit mapping. E.g. <code>c(Male = "blue", Female = "red")</code> </p>
</dd>
</dl>

<p>See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_shapes">shapes</code></td>
<td>
<p>Shapes for each group.
Options are similar to <code>colors</code>'s: <code>TRUE</code>, <code>FALSE</code>, <code>NULL</code>, shape
names (typically integers 0 - 17), or a named vector mapping
groups to specific shape names.
See &quot;Aesthetics&quot; section below for additional information.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'none'</code>, <code>'emmeans'</code>, or
<code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_ties">ties</code></td>
<td>
<p>When <code>transform="rank"</code>, how to rank identical values.
Options are: <code>c("average", "first", "last", "random", "max", "min")</code>.
See <code>rank()</code> for details. Default: <code>"random"</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_caption">caption</code></td>
<td>
<p>Add methodology caption beneath the plot.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_check">check</code></td>
<td>
<p>Generate additional plots to aid in assessing data normality.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_corrplot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass along to ggplot2 functions.
Prefix a parameter name with a layer name to pass it to only that
layer. For instance, <code>p.size = 2</code> ensures only the points have their
size set to <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points, ggplot2 command,
stats table, and stats table commands are available as <code style="white-space: pre;">&#8288;$data&#8288;</code>,
<code style="white-space: pre;">&#8288;$code&#8288;</code>, <code style="white-space: pre;">&#8288;$stats&#8288;</code>, and <code style="white-space: pre;">&#8288;$stats$code&#8288;</code>, respectively.
</p>


<h3>Aesthetics</h3>

<p>All built-in color palettes are colorblind-friendly. The available
categorical palette names are: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Shapes can be given as per base R - numbers 0 through 17 for various shapes,
or the decimal value of an ascii character, e.g. a-z = 65:90; A-Z = 97:122 to use
letters instead of shapes on the plot. Character strings may used as well.
</p>


<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    biom &lt;- rarefy(subset(hmp50, `Body Site` %in% c('Buccal mucosa', 'Saliva')))
    taxa_corrplot(biom, x = "BMI", stat.by = "Body Site", taxa = 'Streptococcus')
</code></pre>

<hr>
<h2 id='taxa_heatmap'>Display taxa abundances as a heatmap.</h2><span id='topic+taxa_heatmap'></span>

<h3>Description</h3>

<p>Display taxa abundances as a heatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_heatmap(
  biom,
  rank = -1,
  taxa = 6,
  tracks = NULL,
  grid = "bilbao",
  other = FALSE,
  unc = "singly",
  lineage = FALSE,
  label = TRUE,
  label_size = NULL,
  rescale = "none",
  trees = TRUE,
  clust = "complete",
  dist = "euclidean",
  asp = 1,
  tree_height = 10,
  track_height = 10,
  legend = "right",
  title = TRUE,
  xlab.angle = "auto",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_heatmap_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_tracks">tracks</code></td>
<td>
<p>A character vector of metadata fields to display as tracks
at the top of the plot. Or, a list as expected by the <code>tracks</code>
argument of <code><a href="#topic+plot_heatmap">plot_heatmap()</a></code>. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_grid">grid</code></td>
<td>
<p>Color palette name, or a list as expected <code><a href="#topic+plot_heatmap">plot_heatmap()</a></code>.
Default: <code>"bilbao"</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_label">label</code></td>
<td>
<p>Label the matrix rows and columns. You can supply a list
or logical vector of length two to control row labels and column
labels separately, for example
<code>label = c(rows = TRUE, cols = FALSE)</code>, or simply
<code>label = c(TRUE, FALSE)</code>. Other valid options are <code>"rows"</code>,
<code>"cols"</code>, <code>"both"</code>, <code>"bottom"</code>, <code>"right"</code>,
and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_label_size">label_size</code></td>
<td>
<p>The font size to use for the row and column labels. You
can supply a numeric vector of length two to control row label sizes
and column label sizes separately, for example
<code>c(rows = 20, cols = 8)</code>, or simply <code>c(20, 8)</code>.
Default: <code>NULL</code>, which computes:
<code>pmax(8, pmin(20, 100 / dim(mtx)))</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_rescale">rescale</code></td>
<td>
<p>Rescale rows or columns to all have a common min/max.
Options: <code>"none"</code>, <code>"rows"</code>, or <code>"cols"</code>.
Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_trees">trees</code></td>
<td>
<p>Draw a dendrogram for rows (left) and columns (top). You can
supply a list or logical vector of length two to control the row tree
and column tree separately, for example
<code>trees = c(rows = TRUE, cols = FALSE)</code>,
or simply <code>trees = c(TRUE, FALSE)</code>.
Other valid options are <code>"rows"</code>, <code>"cols"</code>, <code>"both"</code>,
<code>"left"</code>, <code>"top"</code>, and <code>"none"</code>.
Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_clust">clust</code></td>
<td>
<p>Clustering algorithm for reordering the rows and columns by
similarity. You can supply a list or character vector of length two to
control the row and column clustering separately, for example
<code>clust = c(rows = "complete", cols = NA)</code>, or simply
<code>clust = c("complete", NA)</code>. Options are:
</p>

<dl>
<dt><code>FALSE</code> or <code>NA</code> - </dt><dd><p> Disable reordering. </p>
</dd>
<dt>An <code>hclust</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+hclust">stats::hclust()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"ward.D"</code>,
<code>"ward.D2"</code>, <code>"single"</code>, <code>"complete"</code>,
<code>"average"</code>, <code>"mcquitty"</code>, <code>"median"</code>, or
<code>"centroid"</code>. </p>
</dd>
</dl>

<p>Default: <code>"complete"</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_dist">dist</code></td>
<td>
<p>Distance algorithm to use when reordering the rows and columns
by similarity. You can supply a list or character vector of length
two to control the row and column clustering separately, for example
<code>dist = c(rows = "euclidean", cols = "maximum")</code>, or simply
<code>dist = c("euclidean", "maximum")</code>. Options are:
</p>

<dl>
<dt>A <code>dist</code> class object</dt><dd><p> E.g. from <code><a href="stats.html#topic+dist">stats::dist()</a></code> or <code><a href="#topic+bdiv_distmat">bdiv_distmat()</a></code>. </p>
</dd>
<dt>A method name - </dt><dd> <p><code>"euclidean"</code>,
<code>"maximum"</code>, <code>"manhattan"</code>, <code>"canberra"</code>,
<code>"binary"</code>, or <code>"minkowski"</code>. </p>
</dd>
</dl>

<p>Default: <code>"euclidean"</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio (height/width) for entire grid.
Default: <code>1</code> (square)</p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_tree_height">tree_height</code>, <code id="taxa_heatmap_+3A_track_height">track_height</code></td>
<td>
<p>The height of the dendrogram or annotation
tracks as a percentage of the overall grid size. Use a numeric vector
of length two to assign <code>c(top, left)</code> independently.
Default: <code>10</code> (10% of the grid's height)</p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_legend">legend</code></td>
<td>
<p>Where to place the legend. Options are: <code>"right"</code> or
<code>"bottom"</code>. Default: <code>"right"</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_title">title</code></td>
<td>
<p>Plot title. Set to <code>TRUE</code> for a default title, <code>NULL</code> for
no title, or any character string. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="taxa_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to ggplot2::theme().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>


<h3>Annotation Tracks</h3>

<p>Metadata can be displayed as colored tracks above the heatmap. Common use
cases are provided below, with more thorough documentation available at
https://cmmr.github.io/rbiom .
</p>
<pre>## Categorical ----------------------------
tracks = "Body Site"
tracks = list('Body Site' = "bright")
tracks = list('Body Site' = c('Stool' = "blue", 'Saliva' = "green"))

## Numeric --------------------------------
tracks = "Age"
tracks = list('Age' = "reds")

## Multiple Tracks ------------------------
tracks = c("Body Site", "Age")
tracks = list('Body Site' = "bright", 'Age' = "reds")
tracks = list(
  'Body Site' = c('Stool' = "blue", 'Saliva' = "green"),
  'Age'       = list('colors' = "reds") )
</pre>
<p>The following entries in the track definitions are understood:
</p>

<dl>
<dt><code>colors</code> - </dt><dd><p> A pre-defined palette name or custom set of colors to map to. </p>
</dd>
<dt><code>range</code> - </dt><dd><p> The c(min,max) to use for scale values. </p>
</dd>
<dt><code>label</code> - </dt><dd><p> Label for this track. Defaults to the name of this list element. </p>
</dd>
<dt><code>side</code> - </dt><dd><p> Options are <code>"top"</code> (default) or <code>"left"</code>. </p>
</dd>
<dt><code>na.color</code> - </dt><dd><p> The color to use for <code>NA</code> values. </p>
</dd>
<dt><code>bins</code> - </dt><dd><p> Bin a gradient into this many bins/steps. </p>
</dd>
<dt><code>guide</code> - </dt><dd><p> A list of arguments for guide_colorbar() or guide_legend(). </p>
</dd>
</dl>

<p>All built-in color palettes are colorblind-friendly.
</p>
<p>Categorical palette names: <code>"okabe"</code>, <code>"carto"</code>, <code>"r4"</code>,
<code>"polychrome"</code>, <code>"tol"</code>, <code>"bright"</code>, <code>"light"</code>,
<code>"muted"</code>, <code>"vibrant"</code>, <code>"tableau"</code>, <code>"classic"</code>,
<code>"alphabet"</code>, <code>"tableau20"</code>, <code>"kelly"</code>, and <code>"fishy"</code>.
</p>
<p>Numeric palette names: <code>"reds"</code>, <code>"oranges"</code>, <code>"greens"</code>,
<code>"purples"</code>, <code>"grays"</code>, <code>"acton"</code>, <code>"bamako"</code>,
<code>"batlow"</code>, <code>"bilbao"</code>, <code>"buda"</code>, <code>"davos"</code>,
<code>"devon"</code>, <code>"grayC"</code>, <code>"hawaii"</code>, <code>"imola"</code>,
<code>"lajolla"</code>, <code>"lapaz"</code>, <code>"nuuk"</code>, <code>"oslo"</code>,
<code>"tokyo"</code>, <code>"turku"</code>, <code>"bam"</code>, <code>"berlin"</code>,
<code>"broc"</code>, <code>"cork"</code>, <code>"lisbon"</code>, <code>"roma"</code>,
<code>"tofino"</code>, <code>"vanimo"</code>, and <code>"vik"</code>.
</p>


<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    # Keep and rarefy the 10 most deeply sequenced samples.
    hmp10 &lt;- rarefy(hmp50, n = 10)
    
    taxa_heatmap(hmp10, rank = "Phylum", tracks = "Body Site")
    
    taxa_heatmap(hmp10, rank = "Genus", tracks = c("sex", "bo"))
    
    taxa_heatmap(hmp10, rank = "Phylum", tracks = list(
      'Sex'       = list(colors = c(m = "#0000FF", f = "violetred")), 
      'Body Site' = list(colors = "muted", label = "Source") ))
    
</code></pre>

<hr>
<h2 id='taxa_map'>Map OTUs names to taxa names at a given rank.</h2><span id='topic+taxa_map'></span>

<h3>Description</h3>

<p>Map OTUs names to taxa names at a given rank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_map(
  biom,
  rank = NULL,
  taxa = Inf,
  unc = "singly",
  lineage = FALSE,
  other = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_map_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_map_+3A_rank">rank</code></td>
<td>
<p>When <code>NULL</code>, the entire biom$taxonomy data.frame is returned,
transformed as per <code>unc</code>. Alternatively, a single taxonomic rank
(rank name or integer position in <code>biom$ranks</code>) which returns a named
character vector for mapping OTUs to taxa names.</p>
</td></tr>
<tr><td><code id="taxa_map_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_map_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_map_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_map_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble data.frame when <code>rank=NULL</code>, or a character vector named
with the OTU names.
</p>


<h3>See Also</h3>

<p><code>pull.rbiom()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    library(dplyr, warn.conflicts = FALSE)
    
    # In $taxonomy, .otu is the first column (like a row identifier)  -----
    hmp50$taxonomy %&gt;% head(4)
    
    # In taxa_map, .otu is the last column (most precise rank)  -----------
    taxa_map(hmp50) %&gt;% head(4)
    
    # Generate an OTU to Genus mapping  -----------------------------------
    taxa_map(hmp50, "Genus") %&gt;% head(4)
    
    # Sometimes taxonomic names are incomplete ----------------------------
    otus &lt;- c('GemAsacc', 'GcbBacte', 'Unc58411')
    taxa_map(hmp50, unc = "asis") %&gt;% filter(.otu %in% otus) %&gt;% select(Phylum:.otu)
    
    # rbiom can replace them with unique placeholders ---------------------
    taxa_map(hmp50, unc = "singly") %&gt;% filter(.otu %in% otus) %&gt;% select(Class:.otu)
    
    # Or collapse them into groups ----------------------------------------
    taxa_map(hmp50, unc = "grouped") %&gt;% filter(.otu %in% otus) %&gt;% select(Class:Genus)

</code></pre>

<hr>
<h2 id='taxa_stacked'>Display taxa abundances as a stacked bar graph.</h2><span id='topic+taxa_stacked'></span>

<h3>Description</h3>

<p>Display taxa abundances as a stacked bar graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_stacked(
  biom,
  rank = -1,
  taxa = 6,
  colors = TRUE,
  patterns = FALSE,
  label.by = NULL,
  order.by = NULL,
  facet.by = NULL,
  dist = "euclidean",
  clust = "complete",
  other = TRUE,
  unc = "singly",
  lineage = FALSE,
  xlab.angle = 90,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_stacked_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_colors">colors</code>, <code id="taxa_stacked_+3A_patterns">patterns</code></td>
<td>
<p>A character vector of colors or patterns to use in
the graph. A named character vector can be used to map taxon names to
specific colors or patterns. Set to <code>TRUE</code> to auto-select colors
or patterns, or to <code>FALSE</code> to disable per-taxa colors or patterns.
Default: <code>colors=TRUE, patterns=FALSE</code>.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_label.by">label.by</code>, <code id="taxa_stacked_+3A_order.by">order.by</code></td>
<td>
<p>What metadata column to use for labeling and/or
sorting the samples across the x-axis. Set <code>label.by='.sample'</code>
to display sample names. When <code>order.by=NULL</code>, samples are
arranged based on <code>dist</code> and <code>clust</code>, below.
Default: <code>label.by=NULL, order.by=NULL</code>.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_facet.by">facet.by</code></td>
<td>
<p>Dataset field(s) to use for faceting. Must be categorical.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_dist">dist</code>, <code id="taxa_stacked_+3A_clust">clust</code></td>
<td>
<p>Distance (<code><a href="stats.html#topic+dist">stats::dist()</a></code>) and clustering
(<code><a href="stats.html#topic+hclust">stats::hclust()</a></code>) methods to use for automatically arranging
samples along the x-axis to put samples with similar composition near
one another. Default: <code>dist="euclidean", clust="complete"</code>.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_xlab.angle">xlab.angle</code></td>
<td>
<p>Angle of the labels at the bottom of the plot.
Options are <code>"auto"</code>, <code>'0'</code>, <code>'30'</code>, and <code>'90'</code>.
Default: <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="taxa_stacked_+3A_...">...</code></td>
<td>
<p>Parameters for underlying functions. Prefixing parameter names
with a layer name ensures that a particular parameter is passed to,
and only to, that layer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2</code> plot. The computed data points and ggplot
command are available as <code style="white-space: pre;">&#8288;$data&#8288;</code> and <code style="white-space: pre;">&#8288;$code&#8288;</code>,
respectively.
</p>


<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>
<p>Other visualization: 
<code><a href="#topic+adiv_boxplot">adiv_boxplot</a>()</code>,
<code><a href="#topic+adiv_corrplot">adiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_boxplot">bdiv_boxplot</a>()</code>,
<code><a href="#topic+bdiv_corrplot">bdiv_corrplot</a>()</code>,
<code><a href="#topic+bdiv_heatmap">bdiv_heatmap</a>()</code>,
<code><a href="#topic+bdiv_ord_plot">bdiv_ord_plot</a>()</code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a>()</code>,
<code><a href="#topic+rare_corrplot">rare_corrplot</a>()</code>,
<code><a href="#topic+rare_multiplot">rare_multiplot</a>()</code>,
<code><a href="#topic+rare_stacked">rare_stacked</a>()</code>,
<code><a href="#topic+stats_boxplot">stats_boxplot</a>()</code>,
<code><a href="#topic+stats_corrplot">stats_corrplot</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    taxa_stacked(biom, rank="Phylum")
    
    taxa_stacked(biom, rank = "genus", facet.by = "body site")
    
</code></pre>

<hr>
<h2 id='taxa_stats'>Test taxa abundances for associations with metadata.</h2><span id='topic+taxa_stats'></span>

<h3>Description</h3>

<p>A convenience wrapper for <code><a href="#topic+taxa_table">taxa_table()</a></code> + <code><a href="#topic+stats_table">stats_table()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_stats(
  biom,
  regr = NULL,
  stat.by = NULL,
  rank = -1,
  taxa = 6,
  lineage = FALSE,
  unc = "singly",
  other = FALSE,
  split.by = NULL,
  transform = "none",
  test = "emmeans",
  fit = "gam",
  at = NULL,
  level = 0.95,
  alt = "!=",
  mu = 0,
  p.adj = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_stats_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_regr">regr</code></td>
<td>
<p>Dataset field with the x-axis (independent; predictive)
values. Must be numeric. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_stat.by">stat.by</code></td>
<td>
<p>Dataset field with the statistical groups. Must be
categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_split.by">split.by</code></td>
<td>
<p>Dataset field(s) that the data should be split by prior to
any calculations. Must be categorical. Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_test">test</code></td>
<td>
<p>Method for computing p-values: <code>'wilcox'</code>, <code>'kruskal'</code>,
<code>'emmeans'</code>, or <code>'emtrends'</code>. Default: <code>'emmeans'</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_fit">fit</code></td>
<td>
<p>How to fit the trendline. <code>'lm'</code>, <code>'log'</code>, or <code>'gam'</code>.
Default: <code>'gam'</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_at">at</code></td>
<td>
<p>Position(s) along the x-axis where the means or slopes should be
evaluated. Default: <code>NULL</code>, which samples 100 evenly spaced positions
and selects the position where the p-value is most significant.</p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_level">level</code></td>
<td>
<p>The confidence level for calculating a confidence interval.
Default: <code>0.95</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_alt">alt</code></td>
<td>
<p>Alternative hypothesis direction. Options are <code>'!='</code>
(two-sided; not equal to <code>mu</code>), <code>'&lt;'</code> (less than <code>mu</code>), or <code>'&gt;'</code>
(greater than <code>mu</code>). Default: <code>'!='</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_mu">mu</code></td>
<td>
<p>Reference value to test against. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="taxa_stats_+3A_p.adj">p.adj</code></td>
<td>
<p>Method to use for multiple comparisons adjustment of
p-values. Run <code>p.adjust.methods</code> for a list of available
options. Default: <code>"fdr"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble data.frame with fields from the table below. This tibble
object provides the <code style="white-space: pre;">&#8288;$code&#8288;</code> operator to print the R code used to generate
the statistics.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Field</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean </td><td style="text-align: left;"> Estimated marginal mean. See <code><a href="emmeans.html#topic+emmeans">emmeans::emmeans()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .mean.diff </td><td style="text-align: left;"> Difference in means. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope </td><td style="text-align: left;"> Trendline slope. See <code><a href="emmeans.html#topic+emtrends">emmeans::emtrends()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .slope.diff </td><td style="text-align: left;"> Difference in slopes. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .h1 </td><td style="text-align: left;"> Alternate hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .p.val </td><td style="text-align: left;"> Probability that null hypothesis is correct. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.p </td><td style="text-align: left;"> <code>.p.val</code> after adjusting for multiple comparisons. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .effect.size </td><td style="text-align: left;"> Effect size. See <code><a href="emmeans.html#topic+eff_size">emmeans::eff_size()</a></code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .lower </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .upper </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .se </td><td style="text-align: left;"> Standard error. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .n </td><td style="text-align: left;"> Number of samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .df </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .stat </td><td style="text-align: left;"> Wilcoxon or Kruskal-Wallis rank sum statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .t.ratio </td><td style="text-align: left;"> <code>.mean</code> / <code>.se</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   .r.sqr </td><td style="text-align: left;"> Percent of variation explained by the model. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .adj.r </td><td style="text-align: left;"> <code>.r.sqr</code>, taking degrees of freedom into account. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .aic </td><td style="text-align: left;"> Akaike Information Criterion (predictive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .bic </td><td style="text-align: left;"> Bayesian Information Criterion (descriptive models). </td>
</tr>
<tr>
 <td style="text-align: left;">
   .loglik </td><td style="text-align: left;"> Log-likelihood goodness-of-fit score. </td>
</tr>
<tr>
 <td style="text-align: left;">
   .fit.p </td><td style="text-align: left;"> P-value for observing this fit by chance. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>
<p>Other stats_tables: 
<code><a href="#topic+adiv_stats">adiv_stats</a>()</code>,
<code><a href="#topic+bdiv_stats">bdiv_stats</a>()</code>,
<code><a href="#topic+distmat_stats">distmat_stats</a>()</code>,
<code><a href="#topic+stats_table">stats_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    biom &lt;- rarefy(hmp50)
    
    taxa_stats(biom, stat.by = "Body Site", rank = "Family")[,1:6]
</code></pre>

<hr>
<h2 id='taxa_sums'>Get summary taxa abundances.</h2><span id='topic+taxa_sums'></span><span id='topic+taxa_means'></span><span id='topic+taxa_apply'></span>

<h3>Description</h3>

<p>Get summary taxa abundances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_sums(
  biom,
  rank = -1,
  sort = NULL,
  lineage = FALSE,
  unc = "singly",
  transform = "none"
)

taxa_means(
  biom,
  rank = -1,
  sort = NULL,
  lineage = FALSE,
  unc = "singly",
  transform = "none"
)

taxa_apply(
  biom,
  FUN,
  rank = -1,
  sort = NULL,
  lineage = FALSE,
  unc = "singly",
  transform = "none",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_sums_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_sort">sort</code></td>
<td>
<p>Sort the result. Options: <code>NULL</code>, <code>"asc"</code>, or <code>"desc"</code>, where
<code>NULL</code> will not sort the result. <code>"asc"</code> will sort in ascending order
(smallest to largest), and <code>"desc"</code> in descending order (largest to
smallest). Ignored when the result is not a simple numeric vector.
Default: <code>NULL</code></p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_fun">FUN</code></td>
<td>
<p>The function to apply to each row of the <code>taxa_matrix()</code>.</p>
</td></tr>
<tr><td><code id="taxa_sums_+3A_...">...</code></td>
<td>
<p>Optional arguments to <code>FUN</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For <code>taxa_sums</code> and <code>taxa_means</code>, a named numeric vector.
For <code>taxa_apply</code>, a named vector or list with the results of <code>FUN</code>.
The names are the taxa IDs.
</p>


<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_table">taxa_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    taxa_sums(hmp50) %&gt;% head(4)
    
    taxa_means(hmp50, 'Family') %&gt;% head(5)
    
    taxa_apply(hmp50, max) %&gt;% head(5)
    
    taxa_apply(hmp50, fivenum) %&gt;% head(5)
</code></pre>

<hr>
<h2 id='taxa_table'>Taxa abundances per sample.</h2><span id='topic+taxa_table'></span><span id='topic+taxa_matrix'></span>

<h3>Description</h3>


<dl>
<dt><code>taxa_matrix()</code> - </dt><dd><p> Accepts a single <code>rank</code> and returns a matrix. </p>
</dd>
<dt><code>taxa_table()</code> - </dt><dd><p> Can accept more than one <code>rank</code> and returns a tibble data.frame.  </p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>taxa_table(
  biom,
  rank = -1,
  taxa = 6,
  lineage = FALSE,
  md = ".all",
  unc = "singly",
  other = FALSE,
  transform = "none",
  ties = "random",
  seed = 0
)

taxa_matrix(
  biom,
  rank = -1,
  taxa = NULL,
  lineage = FALSE,
  sparse = FALSE,
  unc = "singly",
  other = FALSE,
  transform = "none",
  ties = "random",
  seed = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxa_table_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="taxa_table_+3A_rank">rank</code></td>
<td>
<p>What rank(s) of taxa to display. E.g. <code>"Phylum"</code>,
<code>"Genus"</code>, <code>".otu"</code>, etc. An integer vector can also be
given, where <code>1</code> is the highest rank, <code>2</code> is the second
highest, <code>-1</code> is the lowest rank, <code>-2</code> is the second
lowest, and <code>0</code> is the OTU &quot;rank&quot;. Run <code>biom$ranks</code> to
see all options for a given rbiom object. Default: <code>-1</code>.</p>
</td></tr>
<tr><td><code id="taxa_table_+3A_taxa">taxa</code></td>
<td>
<p>Which taxa to display. An integer value will show the top n
most abundant taxa. A value 0 &lt;= n &lt; 1 will show any taxa with that
mean abundance or greater (e.g. <code>0.1</code> implies &gt;= 10%). A
character vector of taxa names will show only those named taxa.
Default: <code>6</code>.</p>
</td></tr>
<tr><td><code id="taxa_table_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce
<code style="white-space: pre;">&#8288;Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales&#8288;</code>.
Otherwise the taxa name will simply be <code>Coriobacteriales</code>. You
want to set this to TRUE when <code>unc = "asis"</code> and you have taxa
names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_md">md</code></td>
<td>
<p>Dataset field(s) to include in the output data frame, or <code>'.all'</code>
to include all metadata fields. Default: <code>'.all'</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_other">other</code></td>
<td>
<p>Sum all non-itemized taxa into an &quot;Other&quot; taxa. When
<code>FALSE</code>, only returns taxa matched by the <code>taxa</code>
argument. Specifying <code>TRUE</code> adds &quot;Other&quot; to the returned set.
A string can also be given to imply <code>TRUE</code>, but with that
value as the name to use instead of &quot;Other&quot;.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_transform">transform</code></td>
<td>
<p>Transformation to apply. Options are:
<code>c("none", "rank", "log", "log1p", "sqrt", "percent")</code>. <code>"rank"</code> is
useful for correcting for non-normally distributions before applying
regression statistics. Default: <code>"none"</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_ties">ties</code></td>
<td>
<p>When <code>transform="rank"</code>, how to rank identical values.
Options are: <code>c("average", "first", "last", "random", "max", "min")</code>.
See <code>rank()</code> for details. Default: <code>"random"</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_seed">seed</code></td>
<td>
<p>Random seed for permutations. Must be a non-negative integer.
Default: <code>0</code></p>
</td></tr>
<tr><td><code id="taxa_table_+3A_sparse">sparse</code></td>
<td>
<p>If <code>TRUE</code>, returns a
<code><a href="slam.html#topic+matrix">slam::simple_triplet_matrix()</a></code>, otherwise returns a
normal R matrix object. Default: <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>


<dl>
<dt><code>taxa_matrix()</code> - </dt><dd>
<p>A numeric matrix with taxa as rows, and samples as columns. </p>
</dd>
<dt><code>taxa_table()</code> - </dt><dd>
<p>A tibble data frame with column names .sample, .taxa, .abundance, and any requested by <code>md</code>. </p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other taxa_abundance: 
<code><a href="#topic+sample_sums">sample_sums</a>()</code>,
<code><a href="#topic+taxa_boxplot">taxa_boxplot</a>()</code>,
<code><a href="#topic+taxa_clusters">taxa_clusters</a>()</code>,
<code><a href="#topic+taxa_corrplot">taxa_corrplot</a>()</code>,
<code><a href="#topic+taxa_heatmap">taxa_heatmap</a>()</code>,
<code><a href="#topic+taxa_stacked">taxa_stacked</a>()</code>,
<code><a href="#topic+taxa_stats">taxa_stats</a>()</code>,
<code><a href="#topic+taxa_sums">taxa_sums</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    hmp50$ranks
    
    taxa_matrix(hmp50, 'Phylum')[1:4,1:6]
    
    taxa_table(hmp50, 'Phylum')
</code></pre>

<hr>
<h2 id='tree_subset'>Create a subtree by specifying tips to keep.</h2><span id='topic+tree_subset'></span>

<h3>Description</h3>

<p>Create a subtree by specifying tips to keep.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_subset(tree, tips, underscores = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_subset_+3A_tree">tree</code></td>
<td>
<p>A phylo object, as returned from <code><a href="#topic+read_tree">read_tree()</a></code>.</p>
</td></tr>
<tr><td><code id="tree_subset_+3A_tips">tips</code></td>
<td>
<p>A character, numeric, or logical vector of tips to keep.</p>
</td></tr>
<tr><td><code id="tree_subset_+3A_underscores">underscores</code></td>
<td>
<p>When parsing the tree, should underscores be kept as
is? By default they will be converted to spaces (unless the entire ID
is quoted). Default <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phylo</code> object for the subtree.
</p>


<h3>See Also</h3>

<p>Other phylogeny: 
<code><a href="#topic+read_tree">read_tree</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "newick.tre", package = "rbiom")
    tree &lt;- read_tree(infile)
    tree
    
    subtree &lt;- tree_subset(tree, tips = head(tree$tip.label))
    subtree

</code></pre>

<hr>
<h2 id='with'>Evaluate expressions on metadata.</h2><span id='topic+with'></span><span id='topic+with.rbiom'></span><span id='topic+within.rbiom'></span>

<h3>Description</h3>

<p><code>with()</code> will return the result of your expression. <code>within()</code> will return
an rbiom object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rbiom'
with(data, expr, ...)

## S3 method for class 'rbiom'
within(data, expr, clone = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="with_+3A_data">data</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.</p>
</td></tr>
<tr><td><code id="with_+3A_expr">expr</code></td>
<td>
<p>Passed on to <code><a href="base.html#topic+with">base::with()</a></code> or <code><a href="base.html#topic+with">base::within()</a></code>.</p>
</td></tr>
<tr><td><code id="with_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="with_+3A_clone">clone</code></td>
<td>
<p>Create a copy of <code>biom</code> before modifying. If <code>FALSE</code>, <code>biom</code>
is modified in place as a side-effect. See <a href="#topic+speed">speed ups</a> for
use cases. Default: <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>See description.
</p>


<h3>See Also</h3>

<p>Other transformations: 
<code><a href="#topic+modify_metadata">modify_metadata</a></code>,
<code><a href="#topic+rarefy">rarefy</a>()</code>,
<code><a href="#topic+rarefy_cols">rarefy_cols</a>()</code>,
<code><a href="#topic+slice_metadata">slice_metadata</a></code>,
<code><a href="#topic+subset">subset</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom) 
    
    with(hmp50, table(`Body Site`, Sex))
    
    biom &lt;- within(hmp50, {
      age_bin = cut(Age, 5)
      bmi_bin = cut(BMI, 5)
    })
    biom$metadata

</code></pre>

<hr>
<h2 id='write_biom'>Save an rbiom object to a file.</h2><span id='topic+write_biom'></span><span id='topic+write_metadata'></span><span id='topic+write_counts'></span><span id='topic+write_taxonomy'></span><span id='topic+write_fasta'></span><span id='topic+write_tree'></span><span id='topic+write_xlsx'></span>

<h3>Description</h3>

<p>Automatically creates directories and adds compression based on file name.
</p>

<dl>
<dt><code>write_biom()</code> - </dt><dd><p> According to <a href="http://biom-format.org/documentation/">BIOM format</a> specification. </p>
</dd>
<dt><code>write_xlsx()</code> - </dt><dd><p> Raw data and summary tables in Excel file format. See details. </p>
</dd>
<dt><code>write_fasta()</code> - </dt><dd><p> Sequences only in fasta format. <code>biom</code> may also be a named character vector. </p>
</dd>
<dt><code>write_tree()</code> - </dt><dd><p> Phylogenetic tree only in newick format. <code>biom</code> may also be a phylo object. </p>
</dd>
<dt><code>write_counts()</code>, <code>write_metadata()</code>, <code>write_taxonomy()</code> - </dt><dd><p> Tab-separated values. </p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>write_biom(biom, file, format = "json")

write_metadata(biom, file, quote = FALSE, sep = "\t", ...)

write_counts(biom, file, quote = FALSE, sep = "\t", ...)

write_taxonomy(biom, file, quote = FALSE, sep = "\t", ...)

write_fasta(biom, file = NULL)

write_tree(biom, file = NULL)

write_xlsx(biom, file, depth = 0.1, n = NULL, seed = 0, unc = "singly")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_biom_+3A_biom">biom</code></td>
<td>
<p>An <a href="#topic+rbiom_objects">rbiom object</a>, such as from <code><a href="#topic+as_rbiom">as_rbiom()</a></code>.
Any value accepted by <code><a href="#topic+as_rbiom">as_rbiom()</a></code> can also be given here.</p>
</td></tr>
<tr><td><code id="write_biom_+3A_file">file</code></td>
<td>
<p>Path to the output file. File names ending in <code>.gz</code> or
<code>.bz2</code> will be compressed accordingly.
Setting <code>file=NULL</code> for <code>write_fasta()</code>, <code>write_tree()</code>, and
<code>write_biom(format='json')</code>, and returns a string of the output which
would have been written. For <code>write_biom(format='tab')</code>, <code>file=NULL</code>
returns the tibble that would have been written.</p>
</td></tr>
<tr><td><code id="write_biom_+3A_format">format</code></td>
<td>
<p>Options are <code>"tab"</code>, <code>"json"</code>, and <code>"hdf5"</code>,
corresponding to classic tabular format, BIOM format version 1.0 and
biom version 2.1, respectively. NOTE: to write HDF5 formatted BIOM
files, the BioConductor R package <code>rhdf5</code> must be installed.
Default: <code>"json"</code></p>
</td></tr>
<tr><td><code id="write_biom_+3A_quote">quote</code>, <code id="write_biom_+3A_sep">sep</code>, <code id="write_biom_+3A_...">...</code></td>
<td>
<p>Parameters passed on to <code><a href="utils.html#topic+write.table">write.table()</a></code>.
Default: <code style="white-space: pre;">&#8288;quote=FALSE, sep="\t"&#8288;</code></p>
</td></tr>
<tr><td><code id="write_biom_+3A_depth">depth</code>, <code id="write_biom_+3A_n">n</code></td>
<td>
<p>Passed on to <code><a href="#topic+rarefy_cols">rarefy_cols()</a></code>. For <code>write_xlsx()</code> only,
<code>depth=0</code> disables rarefaction. Default: <code style="white-space: pre;">&#8288;depth=0.1, n=NULL&#8288;</code></p>
</td></tr>
<tr><td><code id="write_biom_+3A_seed">seed</code></td>
<td>
<p>Random seed to use in rarefying. See <code><a href="#topic+rarefy_cols">rarefy_cols()</a></code> function
for details. Must be a non-negative integer. Default: <code>0</code></p>
</td></tr>
<tr><td><code id="write_biom_+3A_unc">unc</code></td>
<td>
<p>How to handle unclassified, uncultured, and similarly ambiguous
taxa names. Options are:
</p>

<dl>
<dt><code>"singly"</code> - </dt><dd><p> Replaces them with the OTU name. </p>
</dd>
<dt><code>"grouped"</code> - </dt><dd><p> Replaces them with a higher rank's name. </p>
</dd>
<dt><code>"drop"</code> - </dt><dd><p> Excludes them from the result. </p>
</dd>
<dt><code>"asis"</code> - </dt><dd><p> To not check/modify any taxa names. </p>
</dd>
</dl>

<p>Abbreviations are allowed. Default: <code>"singly"</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>write_xlsx()</code>, <code>attributes(biom)</code> are saved as additional worksheets if
the attribute is a data frame, matrix, or dist -class object. An attribute
named 'Reads Per Step' is treated specially and merged with the usual 'Reads
Per Sample' tab.
</p>


<h3>Value</h3>

<p>The normalized filepath that was written to (invisibly), unless
<code>file=NULL</code> (see <code>file</code> argument above).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    write_tree(hmp50) %&gt;% substr(1, 50)
    
    if (FALSE) {
    
      hmp10        &lt;- hmp50$clone()
      hmp10$counts &lt;- hmp10$counts[,1:10] %&gt;% rarefy_cols()
      
      attr(hmp10, "Weighted UniFrac")   &lt;- bdiv_distmat(hmp10, 'unifrac')
      attr(hmp10, "Unweighted Jaccard") &lt;- bdiv_distmat(hmp10, 'jaccard', weighted=FALSE)
      
      outfile &lt;- write_xlsx(hmp10, tempfile(fileext = ".xlsx"))
    }

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
