<!DOCTYPE html><html><head><title>Help for package rbiom</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rbiom}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#alpha.div'><p>Estimate the diversity of each sample.</p></a></li>
<li><a href='#beta.div'><p>Make a distance matrix of samples vs samples.</p></a></li>
<li><a href='#counts'><p>Get the abundance counts.</p></a></li>
<li><a href='#info'><p>Get biom's misc information.</p></a></li>
<li><a href='#metadata'><p>Get the sample metadata.</p></a></li>
<li><a href='#nsamples'><p>Number of samples in a BIOM.</p></a></li>
<li><a href='#ntaxa'><p>Number of taxa in a BIOM.</p></a></li>
<li><a href='#phylogeny'><p>Get the phylogenetic tree.</p></a></li>
<li><a href='#print'><p>Summarize the contents of a BIOM object</p></a></li>
<li><a href='#rarefy'><p>Subset counts so that all samples have the same number of observations.</p></a></li>
<li><a href='#rbiom'><p>rbiom: Read/Write, Transform, and Summarize BIOM Data</p></a></li>
<li><a href='#read.biom'><p>Extracts counts, metadata, taxonomy, and phylogeny from a biom file.</p></a></li>
<li><a href='#read.fasta'><p>Parse a fasta file into a named character vector.</p></a></li>
<li><a href='#read.tree'><p>Read a newick formatted phylogenetic tree.</p></a></li>
<li><a href='#sample.names'><p>Get the sample names.</p></a></li>
<li><a href='#select'><p>Reduce samples to a specific list</p></a></li>
<li><a href='#sequences'><p>DNA sequence associated with each taxonomic identifier.</p></a></li>
<li><a href='#subset'><p>Subset samples using the BIOM object's metadata</p></a></li>
<li><a href='#subtree'><p>Create a subtree by specifying tips to keep.</p></a></li>
<li><a href='#taxa.names'><p>Get the taxa names.</p></a></li>
<li><a href='#taxa.ranks'><p>Get the taxa ranks.</p></a></li>
<li><a href='#taxa.rollup'><p>Generate a matrix of samples by taxa, at the specified taxonomic rank.</p></a></li>
<li><a href='#taxonomy'><p>Get the taxonomy table.</p></a></li>
<li><a href='#tips'><p>Names of a phylogenetic tree's tips/leafs.</p></a></li>
<li><a href='#unifrac'><p>Compute Weighted and Unweighted UniFrac distance matrices.</p></a></li>
<li><a href='#write.biom'><p>Write counts, metadata, taxonomy, and phylogeny to a biom file.</p></a></li>
<li><a href='#write.fasta'><p>Write sequences from a BIOM object to a file in fasta format.</p></a></li>
<li><a href='#write.tree'><p>Write a newick formatted phylogenetic tree.</p></a></li>
<li><a href='#write.xlsx'><p>Write data and summary information to a Microsoft Excel-compatible workbook.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Read/Write, Transform, and Summarize 'BIOM' Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel P. Smith &lt;dansmith@orst.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
    A toolkit for working with Biological Observation Matrix ('BIOM') files.
    Features include reading/writing all 'BIOM' formats, rarefaction, alpha
    diversity, beta diversity (including 'UniFrac'), summarizing counts by 
    taxonomic level, and sample subsetting. Standalone functions for 
    reading, writing, and subsetting phylogenetic trees are also provided. 
    All CPU intensive operations are encoded in C with multi-thread support.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://cmmr.github.io/rbiom/index.html">https://cmmr.github.io/rbiom/index.html</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cmmr/rbiom/issues">https://github.com/cmmr/rbiom/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppParallel</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, methods, plyr, Rcpp, RcppParallel, rjson, rlang,
R.utils, slam, openxlsx</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ape, reshape2, rhdf5, testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-05 02:14:43 UTC; Daniel</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel P. Smith <a href="https://orcid.org/0000-0002-2479-2044"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-05 05:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>Pipe operator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='alpha.div'>Estimate the diversity of each sample.</h2><span id='topic+alpha.div'></span>

<h3>Description</h3>

<p>Estimate the diversity of each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha.div(biom, rarefy = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha.div_+3A_biom">biom</code></td>
<td>
<p>A <code>matrix</code>, <code>simple_triplet_matrix</code>, or <code>BIOM</code>
object, as returned from <a href="#topic+read.biom">read.biom</a>. For matrices, the rows and
columns are assumed to be the taxa and samples, respectively.</p>
</td></tr>
<tr><td><code id="alpha.div_+3A_rarefy">rarefy</code></td>
<td>
<p>Control how/whether rarefactions are done prior to alpha 
diversity computations. Options are: 
</p>

<dl>
<dt><code>FALSE</code></dt><dd><p> Use each sample's current set of observations without
applying any rarefaction. (Default) </p>
</dd>
<dt><code>TRUE</code></dt><dd><p> Automatically select and apply a single rarefaction. </p>
</dd>
<dt><code>"multi"</code></dt><dd><p> Automatically select and apply multiple rarefactions. </p>
</dd>
<dt><em>integer vector</em></dt><dd><p> Rarefy at the specified depth(s). </p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of four diversity values for each sample in
<code>biom</code>. The column names are <b>Sample</b>, <b>Depth</b> and the 
diversity metrics: <b>OTUs</b>, <b>Shannon</b>, <b>Chao1</b>, <b>Simpson</b>, 
and <b>InvSimpson</b>. The row names are the sample names, except when
multiple rarefactions are done.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    ad &lt;- alpha.div(biom)
    head(ad)
    
    biom &lt;- subset(biom, `Body Site` == "Saliva" &amp; Age &lt; 26)
    ad   &lt;- alpha.div(biom, "multi")
    boxplot(Shannon ~ Depth, data=ad, xlab="Reads", ylab="Diversity")
    
</code></pre>

<hr>
<h2 id='beta.div'>Make a distance matrix of samples vs samples.</h2><span id='topic+beta.div'></span>

<h3>Description</h3>

<p>Make a distance matrix of samples vs samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta.div(biom, method, weighted = TRUE, tree = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beta.div_+3A_biom">biom</code></td>
<td>
<p>A <code>matrix</code>, <code>simple_triplet_matrix</code>, or <code>BIOM</code> 
object, as returned from <a href="#topic+read.biom">read.biom</a>. For matrices, the rows and 
columns are assumed to be the taxa and samples, respectively.</p>
</td></tr>
<tr><td><code id="beta.div_+3A_method">method</code></td>
<td>
<p>The distance algorithm to use. Options are:
<b>&ldquo;manhattan&rdquo;</b>, <b>&ldquo;euclidean&rdquo;</b>, 
<b>&ldquo;bray-curtis&rdquo;</b>, <b>&ldquo;jaccard&rdquo;</b>, and
<b>&ldquo;unifrac&rdquo;</b>. Non-ambiguous abbrevations of the method 
names are also accepted. A phylogentic tree must be present in 
<code>biom</code> or explicitly provided via <code>tree=</code> to use the UniFrac methods.</p>
</td></tr>
<tr><td><code id="beta.div_+3A_weighted">weighted</code></td>
<td>
<p>Take relative abundances into account. When 
<code>weighted=FALSE</code>, only presence/absence is considered.</p>
</td></tr>
<tr><td><code id="beta.div_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object representing the phylogenetic
relationships of the taxa in <code>biom</code>. Will be taken from the tree
embedded in the <code>biom</code> object if not explicitly specified. Only
required for computing UniFrac distance matrices.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A distance matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    biom &lt;- select(biom, 1:10)
    
    dm &lt;- beta.div(biom, 'unifrac')
    
    as.matrix(dm)[1:4,1:4]
    plot(hclust(dm))

</code></pre>

<hr>
<h2 id='counts'>Get the abundance counts.</h2><span id='topic+counts'></span>

<h3>Description</h3>

<p>Get the abundance counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>counts(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="counts_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix of the sample abundance counts in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    counts(biom)[1:4,1:5]

</code></pre>

<hr>
<h2 id='info'>Get biom's misc information.</h2><span id='topic+info'></span>

<h3>Description</h3>

<p>Get biom's misc information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>info(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="info_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the metadata in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    info(biom)

</code></pre>

<hr>
<h2 id='metadata'>Get the sample metadata.</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>Get the sample metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metadata_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the metadata in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    metadata(biom)[1:4,1:3]

</code></pre>

<hr>
<h2 id='nsamples'>Number of samples in a BIOM.</h2><span id='topic+nsamples'></span>

<h3>Description</h3>

<p>Number of samples in a BIOM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nsamples(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nsamples_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of samples present.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    nsamples(biom)

</code></pre>

<hr>
<h2 id='ntaxa'>Number of taxa in a BIOM.</h2><span id='topic+ntaxa'></span>

<h3>Description</h3>

<p>Number of taxa in a BIOM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ntaxa(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ntaxa_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of taxa present.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    ntaxa(biom)

</code></pre>

<hr>
<h2 id='phylogeny'>Get the phylogenetic tree.</h2><span id='topic+phylogeny'></span>

<h3>Description</h3>

<p>Get the phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylogeny(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylogeny_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phylo</code> class object of the tree in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    summary(phylogeny(biom))

</code></pre>

<hr>
<h2 id='print'>Summarize the contents of a BIOM object</h2><span id='topic+print'></span><span id='topic+print.BIOM'></span>

<h3>Description</h3>

<p>Summarize the contents of a BIOM object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BIOM'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_+3A_x">x</code></td>
<td>
<p>A BIOM object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
<tr><td><code id="print_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL (invisibly)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    print(biom)

</code></pre>

<hr>
<h2 id='rarefy'>Subset counts so that all samples have the same number of observations.</h2><span id='topic+rarefy'></span>

<h3>Description</h3>

<p>Subset counts so that all samples have the same number of observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rarefy(biom, depth = NULL, seed = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rarefy_+3A_biom">biom</code></td>
<td>
<p>A <code>matrix</code>, <code>simple_triplet_matrix</code>, or <code>BIOM</code> 
object, as returned from <a href="#topic+read.biom">read.biom</a>. For matrices, the rows and 
columns are assumed to be the taxa and samples, respectively.</p>
</td></tr>
<tr><td><code id="rarefy_+3A_depth">depth</code></td>
<td>
<p>The number of observations to keep, per sample. If set to
<code>NULL</code>, a depth will be automatically selected. Samples that have
fewer than this number of observations will be dropped. If called on
data with non-integer abundances, values will be re-scaled to integers 
between 1 and <code>depth</code> such that they sum to <code>depth</code>.</p>
</td></tr>
<tr><td><code id="rarefy_+3A_seed">seed</code></td>
<td>
<p>An integer to use for seeding the random number generator. If
you need to create different random rarefactions of the same <code>BIOM</code>
object, set this seed value to a different number each time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>matrix</code>, <code>simple_triplet_matrix</code>, or <code>BIOM</code> 
object, depending on the input object type. The type of object provided
is the same type that is returned. The retained observations are randomly
selected, based on a seed value derived from the <code>BIOM</code> object. 
Therefore, rarefying the same biom to the same depth will always produce
the same resultant rarification.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)

    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    range(slam::col_sums(biom$counts))

    biom &lt;- rarefy(biom, depth=1000)
    range(slam::col_sums(biom$counts))

</code></pre>

<hr>
<h2 id='rbiom'>rbiom: Read/Write, Transform, and Summarize BIOM Data</h2><span id='topic+rbiom'></span>

<h3>Description</h3>

<p>A toolkit for working with Biological Observation Matrix (BIOM) files.
Features include reading/writing all BIOM formats, rarefaction, alpha
diversity, beta diversity (including UniFrac), summarizing counts by 
taxonomic level, and sample subsetting. Standalone functions for 
reading, writing, and subsetting phylogenetic trees are also provided. 
All CPU intensive operations are encoded in C with multi-thread support.
</p>


<h3>Multithreading</h3>

<p>Many <code>rbiom</code> functions support multithreading:
</p>
<p>The default behavior of these function is to run on as many cores as are
available in the local compute environment. If you wish to limit the number
of simultaneous threads, set <code>RcppParallel</code>'s <code>numThreads</code> option.
For instance:
</p>
<pre>
    RcppParallel::setThreadOptions(numThreads = 4)
</pre>

<hr>
<h2 id='read.biom'>Extracts counts, metadata, taxonomy, and phylogeny from a biom file.</h2><span id='topic+read.biom'></span>

<h3>Description</h3>

<p>Extracts counts, metadata, taxonomy, and phylogeny from a biom file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.biom(src, tree = "auto", prune = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.biom_+3A_src">src</code></td>
<td>
<p>Input data as either a file path, URL, or JSON string.
<code>read.biom</code> can read BIOM files formatted according to both the
version 1.0 (JSON) and 2.1 (HDF5)
<a href="http://biom-format.org/documentation/">specifications</a> as well as
classical tabular format. URLs must begin with <kbd>http://</kbd>,
<kbd>https://</kbd>, <kbd>ftp://</kbd>, or <kbd>ftps://</kbd>. JSON files must have
<code>{</code> as their first non-whitespace character. Compressed (gzip or
bzip2) BIOM files are also supported. NOTE: to read HDF5 formatted BIOM
files, the BioConductor R package <code>rhdf5</code> must be installed.</p>
</td></tr>
<tr><td><code id="read.biom_+3A_tree">tree</code></td>
<td>
<p>The default value of <code>auto</code> will read the tree from the 
BIOM file specified in <code>src</code>, if present. The value <code>TRUE</code> 
will do the same, but will generate an error message if a tree is not 
present. Setting <code>tree=FALSE</code> will return a <code>BIOM</code> object 
without any tree data. You may also provide a file path, URL, or Newick
string to load that tree data into the final <code>BIOM</code> object.</p>
</td></tr>
<tr><td><code id="read.biom_+3A_prune">prune</code></td>
<td>
<p>Should samples and taxa with zero observations be discarded?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>BIOM</code> class object containing the parsed data. This object
can be treated as a list with the following named elements:
</p>

<dl>
<dt>counts</dt><dd><p>A numeric <code>slam</code> sparse matrix of observation
counts. Taxa (OTUs) as rows and samples as columns.</p>
</dd>
<dt>metadata</dt><dd><p>A data frame containing any embedded metadata.
Row names are sample IDs.</p>
</dd>
<dt>taxonomy</dt><dd><p>Character matrix of taxonomic names, if given.
Row names are taxa (OTU) IDs. Column rows are named Kingdom,
Phylum, Class, Order, Family, Genus, Species, and Strain, or
TaxLvl.1, TaxLvl.2, ... , TaxLvl.N when more than 8 levels of
taxonomy are encoded in the biom file.</p>
</dd>
<dt>phylogeny</dt><dd><p>An object of class <code>phylo</code> defining the
phylogenetic relationships between the taxa. Although the
official specification for BIOM only includes phylogenetic trees
in BIOM version 2.1, if a BIOM version 1.0 file includes a
<code>phylogeny</code> entry with newick data, then it will be loaded
here as well. The <span class="pkg">ape</span> package has additional functions
for working with <code>phylo</code> objects.</p>
</dd>
<dt>sequences</dt><dd><p>A named character vector, where the names are
taxonomic identifiers and the values are the sequences they
represent. These values are not part of the official BIOM
specification, but will be read and written when defined.</p>
</dd>
<dt>info</dt><dd><p>A list of other attributes defined in the BIOM file,
such as <code>id</code>, <code>type</code>, <code>format</code>, <code>format_url</code>,
<code>generated_by</code>, <code>date</code>, <code>matrix_type</code>,
<code>matrix_element_type</code>, <code>Comment</code>, and <code>shape</code></p>
</dd>
</dl>

<p><code>metadata</code>, <code>taxonomy</code>, and <code>phylogeny</code> are optional
components of the BIOM file specification and therefore will be empty
in the returned object when they are not provided by the BIOM file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)

    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)

    summary(biom)

    # Taxa Abundances
    as.matrix(biom$counts[1:4,1:4])

    top5 &lt;- names(head(rev(sort(slam::row_sums(biom$counts))), 5))
    biom$taxonomy[top5,c('Family', 'Genus')]
    as.matrix(biom$counts[top5, 1:6])

    # Metadata
    table(biom$metadata$Sex, biom$metadata$`Body Site`)
    sprintf("Mean age: %.1f", mean(biom$metadata$Age))

    # Phylogenetic tree
    tree &lt;- biom$phylogeny
    top5.tree &lt;- rbiom::subtree(tree, top5)
    ape::plot.phylo(top5.tree)

</code></pre>

<hr>
<h2 id='read.fasta'>Parse a fasta file into a named character vector.</h2><span id='topic+read.fasta'></span>

<h3>Description</h3>

<p>Parse a fasta file into a named character vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fasta(file, ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fasta_+3A_file">file</code></td>
<td>
<p>A file with fasta-formatted sequences. Can optionally be compressed with gzip, bzip2, xz, or lzma.</p>
</td></tr>
<tr><td><code id="read.fasta_+3A_ids">ids</code></td>
<td>
<p>Character vector of IDs to retrieve. The default, <code>NULL</code>, will retrieve everything.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named character vector in which names are the fasta headers and values are the sequences.
</p>

<hr>
<h2 id='read.tree'>Read a newick formatted phylogenetic tree.</h2><span id='topic+read.tree'></span>

<h3>Description</h3>

<p>A phylogenetic tree is required for computing UniFrac distance matrices.
You can load a tree either from a file or by providing the tree string
directly. This tree must be in Newick format, also known as parenthetic
format and New Hampshire format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.tree(src)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.tree_+3A_src">src</code></td>
<td>
<p>Input data as either a file path, URL, or Newick string. URLs 
must begin with <kbd>http://</kbd>, <kbd>https://</kbd>, <kbd>ftp://</kbd>, or 
<kbd>ftps://</kbd>. Newick strings must have <code>(</code> as their first 
non-whitespace character. Compressed (gzip or bzip2) Newick files 
are also supported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phylo</code> class object representing the tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "newick.tre", package = "rbiom")
    tree &lt;- read.tree(infile)
    
    tree &lt;- read.tree("
        (t9:0.99,((t5:0.87,t2:0.89):0.51,(((t10:0.16,(t7:0.83,t4:0.96)
        :0.94):0.69,(t6:0.92,(t3:0.62,t1:0.85):0.54):0.23):0.74,t8:0.1
        2):0.43):0.67);")

</code></pre>

<hr>
<h2 id='sample.names'>Get the sample names.</h2><span id='topic+sample.names'></span>

<h3>Description</h3>

<p>Get the sample names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample.names(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample.names_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the sample IDs / names in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    sample.names(biom)

</code></pre>

<hr>
<h2 id='select'>Reduce samples to a specific list</h2><span id='topic+select'></span>

<h3>Description</h3>

<p>Reduce samples to a specific list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select(biom, samples = NULL, nTop = NULL, nRandom = NULL, seed = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_+3A_biom">biom</code></td>
<td>
<p>A BIOM object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
<tr><td><code id="select_+3A_samples">samples</code></td>
<td>
<p>Sample names, indices, or a logical vector identifying
the samples to keep. The latter two should be based on the order of
sample names given by <code>colnames(biom$counts)</code>.</p>
</td></tr>
<tr><td><code id="select_+3A_ntop">nTop</code></td>
<td>
<p>Selects this number of samples, taking the sample with the most
observations first, then the sample with the second-most observations,
etc. If <code>nTop</code> is higher than the number of samples in the dataset, 
the entire dataset will be returned. See note.</p>
</td></tr>
<tr><td><code id="select_+3A_nrandom">nRandom</code></td>
<td>
<p>Randomly selects this number of samples. If higher than the
number of samples in the dataset, the entire dataset will be returned.
See note.</p>
</td></tr>
<tr><td><code id="select_+3A_seed">seed</code></td>
<td>
<p>Random seed, used when selecting <code>nRandom</code> samples.
</p>
<p>Note: Generally, you will specify only one of the filters: <code>samples</code>,
<code>nTop</code>, or <code>nRandom</code>. However, specifying multiple filters is
allowed; they will be applied in the order listed above.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>BIOM</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subset">subset</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    ex1 &lt;- select(biom, c('HMP14', 'HMP22', 'HMP03'))
    ex2 &lt;- select(biom, c(32, 11, 28, 16, 46, 5))
    ex3 &lt;- select(biom, 1:50 %% 6 == 0)
    ex4 &lt;- select(biom, nRandom = 10)
    ex5 &lt;- select(biom, nTop = 5)
    ex6 &lt;- select(biom, samples = 10:40, nTop = 20, nRandom = 10)

</code></pre>

<hr>
<h2 id='sequences'>DNA sequence associated with each taxonomic identifier.</h2><span id='topic+sequences'></span>

<h3>Description</h3>

<p>DNA sequence associated with each taxonomic identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sequences(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sequences_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named character vector of sequences in <code>biom</code>. If this data
is not present, then returns <code>NULL</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    sequences(biom)[1:4]
    
    # Write to a compressed fasta file in the temporary directory:
    seqs &lt;- sequences(biom)
    conn &lt;- bzfile(file.path(tempdir(), "Sequences.fa.bz2"), "w")
    cat(sprintf("&gt;%s\n%s", names(seqs), seqs), file=conn, sep="\n")
    close(conn)
    
    # You can also use the write.fasta function for this task:
    write.fasta(biom, file.path(tempdir(), "Sequences.fa.gz"))

</code></pre>

<hr>
<h2 id='subset'>Subset samples using the BIOM object's metadata</h2><span id='topic+subset'></span><span id='topic+subset.BIOM'></span>

<h3>Description</h3>

<p>Subset samples using the BIOM object's metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BIOM'
subset(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_+3A_x">x</code></td>
<td>
<p>A BIOM object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
<tr><td><code id="subset_+3A_...">...</code></td>
<td>
<p>Test to run on the metadata to identify samples to retain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>BIOM</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+select">select</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    ex1 &lt;- subset(biom, Age &gt; 30)
    ex2 &lt;- subset(biom, `Body Site` %in% c("Saliva", "Stool"))
    ex3 &lt;- subset(biom, Age &lt; 25 &amp; BMI &gt; 22)

</code></pre>

<hr>
<h2 id='subtree'>Create a subtree by specifying tips to keep.</h2><span id='topic+subtree'></span>

<h3>Description</h3>

<p>Create a subtree by specifying tips to keep.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subtree(tree, tips)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtree_+3A_tree">tree</code></td>
<td>
<p>A phylo object, as returned from <a href="#topic+read.tree">read.tree</a>.</p>
</td></tr>
<tr><td><code id="subtree_+3A_tips">tips</code></td>
<td>
<p>A character, numeric, or logical vector of tips to keep.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phylo</code> object for the subtree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "newick.tre", package = "rbiom")
    tree &lt;- read.tree(infile)
    
    leafs   &lt;- tips(tree)
    subtree &lt;- subtree(tree, head(leafs))

</code></pre>

<hr>
<h2 id='taxa.names'>Get the taxa names.</h2><span id='topic+taxa.names'></span>

<h3>Description</h3>

<p>Get the taxa names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.names(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.names_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the taxa IDs / names in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    taxa.names(biom) %&gt;% head()

</code></pre>

<hr>
<h2 id='taxa.ranks'>Get the taxa ranks.</h2><span id='topic+taxa.ranks'></span>

<h3>Description</h3>

<p>Get the taxa ranks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.ranks(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.ranks_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the taxa ranks in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxonomy">taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    taxa.ranks(biom)

</code></pre>

<hr>
<h2 id='taxa.rollup'>Generate a matrix of samples by taxa, at the specified taxonomic rank.</h2><span id='topic+taxa.rollup'></span>

<h3>Description</h3>

<p>Generate a matrix of samples by taxa, at the specified taxonomic rank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.rollup(biom, rank = "OTU", map = NULL, lineage = FALSE, sparse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.rollup_+3A_biom">biom</code></td>
<td>
<p>A <code>matrix</code>, <code>simple_triplet_matrix</code>, or <code>BIOM</code> 
object, as returned from <a href="#topic+read.biom">read.biom</a>. For matrices, the rows and 
columns are assumed to be the taxa and samples, respectively.</p>
</td></tr>
<tr><td><code id="taxa.rollup_+3A_rank">rank</code></td>
<td>
<p>The taxonomic rank. E.g. <b>&ldquo;OTU&rdquo;</b>, 
<b>&ldquo;Phylum&rdquo;</b>, etc. May also be given numerically: 0 for
OTU, 1 for the highest level (i.e. Kingdom), and extending to the number
of taxonomic ranks encoded in the original biom file. See example below
to fetch the names of all available ranks.</p>
</td></tr>
<tr><td><code id="taxa.rollup_+3A_map">map</code></td>
<td>
<p>A character matrix defining the value that each taxa IDs is
assigned for each taxonomic rank. If <code>map=NULL</code> and <code>biom</code> is a
<code>BIOM</code> class object, the map will be automatically loaded from 
<code>biom$taxonomy</code>. <code>map</code> must not be <code>null</code> when <code>biom</code>
is a <code>matrix</code> or <code>simple_triplet_matrix</code>. See the example below
for an example of <code>map</code>'s structure.</p>
</td></tr>
<tr><td><code id="taxa.rollup_+3A_lineage">lineage</code></td>
<td>
<p>Include all ranks in the name of the taxa. For instance,
setting to <code>TRUE</code> will produce 
<code>Bacteria; Actinobacteria; Coriobacteriia; Coriobacteriales</code>. 
Whereas setting to <code>FALSE</code> (the default) will return simply
<code>Coriobacteriales</code>. You want to set this to TRUE if you have
genus names (such as <em>Incertae_Sedis</em>) that map to multiple higher
level ranks.</p>
</td></tr>
<tr><td><code id="taxa.rollup_+3A_sparse">sparse</code></td>
<td>
<p>If true, returns a sparse matrix as described by 
<code>slam::simple_triplet_matrix</code>, otherwise returns a normal R
matrix object. Sparse matrices will likely be considerably more
memory efficient in this scenario.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix with samples as column names, and taxonomic
identifiers as row names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    colnames(biom$taxonomy)
    
    phyla &lt;- taxa.rollup(biom, 'Phylum')
    phyla[1:4,1:6]
    
    # Custom matrices should be formatted like so:
    counts &lt;- as.matrix(biom$counts)
    map    &lt;- biom$taxonomy
    
    counts[1:3,1:6]
    map[1:3,1:4]
    
    phyla &lt;- taxa.rollup(counts, 'Phylum', map=map)
    phyla[1:3,1:6]

</code></pre>

<hr>
<h2 id='taxonomy'>Get the taxonomy table.</h2><span id='topic+taxonomy'></span>

<h3>Description</h3>

<p>Get the taxonomy table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy(biom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonomy_+3A_biom">biom</code></td>
<td>
<p>A <code>BIOM</code> object, as returned from <a href="#topic+read.biom">read.biom</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character matrix of the named taxonomies in <code>biom</code>.
</p>


<h3>See Also</h3>

<p>Other accessor functions: 
<code><a href="#topic+counts">counts</a>()</code>,
<code><a href="#topic+info">info</a>()</code>,
<code><a href="#topic+metadata">metadata</a>()</code>,
<code><a href="#topic+nsamples">nsamples</a>()</code>,
<code><a href="#topic+ntaxa">ntaxa</a>()</code>,
<code><a href="#topic+phylogeny">phylogeny</a>()</code>,
<code><a href="#topic+sample.names">sample.names</a>()</code>,
<code><a href="#topic+sequences">sequences</a>()</code>,
<code><a href="#topic+taxa.names">taxa.names</a>()</code>,
<code><a href="#topic+taxa.ranks">taxa.ranks</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    
    taxonomy(biom)[1:4,]

</code></pre>

<hr>
<h2 id='tips'>Names of a phylogenetic tree's tips/leafs.</h2><span id='topic+tips'></span>

<h3>Description</h3>

<p>Names of a phylogenetic tree's tips/leafs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tips(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tips_+3A_x">x</code></td>
<td>
<p>A phylo object, as returned from <a href="#topic+read.tree">read.tree</a>..</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with the leaf names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "newick.tre", package = "rbiom")
    tree &lt;- read.tree(infile)
    
    leafs   &lt;- tips(tree)
    subtree &lt;- subtree(tree, head(leafs))

</code></pre>

<hr>
<h2 id='unifrac'>Compute Weighted and Unweighted UniFrac distance matrices.</h2><span id='topic+unifrac'></span>

<h3>Description</h3>

<p>This is the function called internally by <a href="#topic+beta.div">beta.div</a>, but is made
visible here so you can use it with matrices and trees without having to
first convert them to <code>BIOM</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unifrac(biom, weighted = TRUE, tree = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unifrac_+3A_biom">biom</code></td>
<td>
<p>A <code>matrix</code>, <code>simple_triplet_matrix</code>, or <code>BIOM</code> 
object, as returned from <a href="#topic+read.biom">read.biom</a>. For matrices, the rows and 
columns are assumed to be the taxa and samples, respectively.</p>
</td></tr>
<tr><td><code id="unifrac_+3A_weighted">weighted</code></td>
<td>
<p>Use weighted UniFrac, which takes abundance into account
rather than simply presence/absence.</p>
</td></tr>
<tr><td><code id="unifrac_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object providing a phylogenetic tree for the
taxa names in <code>biom</code>. If <code>tree=NULL</code>, then the tree will be
loaded from <code>biom</code>, if encoded there.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A distance matrix of class <code>dist</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "hmp50.bz2", package = "rbiom")
    biom &lt;- read.biom(infile)
    biom &lt;- select(biom, 1:10)
    
    dm &lt;- unifrac(biom)
    plot(hclust(dm), cex=.8)
    as.matrix(dm)[1:4,1:4]
    
    # Using a custom matrix and tree
    mtx &lt;- matrix(sample.int(12*20), ncol=20)
    dimnames(mtx) &lt;- list(LETTERS[1:12], letters[1:20])
    tree &lt;- ape::as.phylo(hclust(dist(mtx)))
    
    dm &lt;- unifrac(mtx, tree=tree)
    as.matrix(dm)[1:4,1:4]
    
</code></pre>

<hr>
<h2 id='write.biom'>Write counts, metadata, taxonomy, and phylogeny to a biom file.</h2><span id='topic+write.biom'></span>

<h3>Description</h3>

<p>Write counts, metadata, taxonomy, and phylogeny to a biom file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.biom(biom, file, format = "json")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.biom_+3A_biom">biom</code></td>
<td>
<p>The BIOM object to save to the file.</p>
</td></tr>
<tr><td><code id="write.biom_+3A_file">file</code></td>
<td>
<p>Path to the output file.</p>
</td></tr>
<tr><td><code id="write.biom_+3A_format">format</code></td>
<td>
<p>Options are <b>&ldquo;tab&rdquo;</b>, 
<b>&ldquo;json&rdquo;</b>, and <b>&ldquo;hdf5&rdquo;</b>, 
corresponding to classic tabular format, biom format version 1.0 and 
biom version 2.1, respectively. Abbreviations are also accepted. See
<a href="http://biom-format.org/documentation/">http://biom-format.org/documentation/</a> for details. NOTE: to write
HDF5 formatted BIOM files, the BioConductor R package <code>rhdf5</code> must
be installed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>On success, returns <code>NULL</code> invisibly.
</p>

<hr>
<h2 id='write.fasta'>Write sequences from a BIOM object to a file in fasta format.</h2><span id='topic+write.fasta'></span>

<h3>Description</h3>

<p>Write sequences from a BIOM object to a file in fasta format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.fasta(seqs, outfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.fasta_+3A_seqs">seqs</code></td>
<td>
<p>A named character vector where names are sequence names and
values are the sequences. Also accepts a <code>BIOM</code>
object which contains sequences.</p>
</td></tr>
<tr><td><code id="write.fasta_+3A_outfile">outfile</code></td>
<td>
<p>Path to the output fasta file. Files ending in <code>.gz</code> or
<code>.bz2</code> will be compressed accordingly.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>On success, returns <code>NULL</code> invisibly.
</p>

<hr>
<h2 id='write.tree'>Write a newick formatted phylogenetic tree.</h2><span id='topic+write.tree'></span>

<h3>Description</h3>

<p>Write a newick formatted phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.tree(tree = NULL, file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.tree_+3A_tree">tree</code></td>
<td>
<p>A <code>phylo</code> object, as returned from <a href="#topic+read.tree">read.tree</a>. Also 
accepts a <code>BIOM</code> object if it has a phylogentic tree.</p>
</td></tr>
<tr><td><code id="write.tree_+3A_file">file</code></td>
<td>
<p>Filename or connection to write the newick file to (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If file is NULL, the newick string as a character vector. Otherwise,
the return value from <code>writeChar</code>, typically invsible(NULL).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    library(rbiom)
    
    infile &lt;- system.file("extdata", "newick.tre", package = "rbiom")
    tree   &lt;- read.tree(infile)
    newick &lt;- write.tree(tree)

</code></pre>

<hr>
<h2 id='write.xlsx'>Write data and summary information to a Microsoft Excel-compatible workbook.</h2><span id='topic+write.xlsx'></span>

<h3>Description</h3>

<p>Write data and summary information to a Microsoft Excel-compatible workbook.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.xlsx(biom, outfile, depth = NULL, seed = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.xlsx_+3A_biom">biom</code></td>
<td>
<p>The BIOM object to save to the file.</p>
</td></tr>
<tr><td><code id="write.xlsx_+3A_outfile">outfile</code></td>
<td>
<p>Path to the output xlsx file.</p>
</td></tr>
<tr><td><code id="write.xlsx_+3A_depth">depth</code></td>
<td>
<p>Depth to rarefy to. See <code>rarefy</code> function for details.
Only use <code>depth</code> with <code>BIOM</code> files of type
'OTU table' and integer count values.</p>
</td></tr>
<tr><td><code id="write.xlsx_+3A_seed">seed</code></td>
<td>
<p>Random seed to use in rarefying. See <code>rarefy</code> function
for details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>On success, returns <code>NULL</code> invisibly.
</p>


<h3>Note</h3>

<p>Any data frame attributes on <code>biom</code> will be included as separate 
worksheets. An attribute named 'Reads Per Step' is treated specially and 
merged with the usual 'Reads Per Sample' tab - if provided, its row names 
should match those in <code>biom</code> exactly.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
