<!DOCTYPE html><html><head><title>Help for package RCircos</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RCircos}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RCircos-package'>
<p>Circos-like Plot of Genomic Data</p></a></li>
<li><a href='#RCircos.Adjust.Scatter.Values'>
<p>Adjust Scatter Plot Data to Fit the Data Track</p></a></li>
<li><a href='#RCircos.Area.Highlight'>
<p>Highlight PLot Area Cross One or More Tracks</p></a></li>
<li><a href='#RCircos.Area.Plot'>
<p>Paint Areas on One Data Track</p></a></li>
<li><a href='#RCircos.Chromosome.Ideogram.Plot'>
<p>Plot Chromosome Ideogram</p></a></li>
<li><a href='#RCircos.Clear.Track'>
<p>Erase One or More Data Tracks</p></a></li>
<li><a href='#RCircos.Clear.Zoom.Area'>
<p>Erase Zoom-in Plot Area</p></a></li>
<li><a href='#RCircos.Customized.Connection.Plot'>
<p>Draw Connectors between Chromosome Ideogram and Gene Names</p></a></li>
<li><a href='#RCircos.Customized.Shape.Plot'>
<p>Plot A Customized Shape</p></a></li>
<li><a href='#RCircos.Data.Point'>
<p>Convert A Genomic Position to An Index of Circos Plot Position</p></a></li>
<li><a href='#RCircos.defaultBasePerUnits'>
<p>The Default Base Pair Number of A Circos Unit</p></a></li>
<li><a href='#RCircos.defaultCharWidth'>
<p>Default Character With in Circos Unit</p></a></li>
<li><a href='#RCircos.defaultChromPadding'>
<p>The Default Value of GAP in Circos (Chromosome) Unit between two Chromosome Ideogram</p></a></li>
<li><a href='#RCircos.defaultCircosUnits'>
<p>Default Circos Units for RCircos</p></a></li>
<li><a href='#RCircos.defaultTextSize'>
<p>Default Character Size of RCircos Plot</p></a></li>
<li><a href='#RCircos.Env'>
<p>RCircos Environment</p></a></li>
<li><a href='#RCircos.Gene.Connector.Plot'>
<p>Draw Connectors between Chromosome Ideogram and Gene Names</p></a></li>
<li><a href='#RCircos.Gene.Label.Data'>
<p>Sample Data for Gene Labels</p></a></li>
<li><a href='#RCircos.Gene.Name.Plot'>
<p>Draw Gene Names on a Data Track</p></a></li>
<li><a href='#RCircos.Get.Arrow.Shape'>
<p>Get Default Coordinates for An Arrow Shape</p></a></li>
<li><a href='#RCircos.Get.Chromosome.Order'>
<p>Find the order of Chromosome Names</p></a></li>
<li><a href='#RCircos.Get.Core.Components'>
<p>Methods to Retrieve RCircos Core Components Used in Current RCircos Session</p></a></li>
<li><a href='#RCircos.Get.Data.Point.Height'>
<p>Calculate the Height of Data in A Track</p></a></li>
<li><a href='#RCircos.Get.Default.Parameters'>
<p>Retrieve Parameter Values Stored in RCircos Environment</p></a></li>
<li><a href='#RCircos.Get.Gene.Label.Locations'>
<p>Calculate Plot Positions for Gene Labels</p></a></li>
<li><a href='#RCircos.Get.Gene.Name.Plot.Parameters'>
<p>Calculate Parameters for Gene Name Plot</p></a></li>
<li><a href='#RCircos.Get.Heatmap.Color.Scale'>
<p>Generate Color Scales for Heatmap Plot</p></a></li>
<li><a href='#RCircos.Get.Heatmap.Color.Scale.Location'>
<p>Calculate X and Y Coordinates for Color Scale</p></a></li>
<li><a href='#RCircos.Get.Heatmap.Data.Colors'>
<p>Set Colors for One Track of Heatmap</p></a></li>
<li><a href='#RCircos.Get.Link.Colors'>
<p>Set up Colors for Link Lines or Ribbons</p></a></li>
<li><a href='#RCircos.Get.Paired.Points.Positions'>
<p>Calcualte Plot Positions for A Genomic Interval</p></a></li>
<li><a href='#RCircos.Get.Plot.Boundary'>
<p>Calculate Boundary of A Data Track</p></a></li>
<li><a href='#RCircos.Get.Plot.Colors'>
<p>Assign Plot Colors for Non-Link Data Plot</p></a></li>
<li><a href='#RCircos.Get.Plot.Layers'>
<p>Get the Layer Numbers for Tile Plot</p></a></li>
<li><a href='#RCircos.Get.Polygon.Height'>
<p>Calculate the Height of Polygons in A Data Track</p></a></li>
<li><a href='#RCircos.Get.Single.Point.Positions'>
<p>Calculate Plot Positions for Genomic Data</p></a></li>
<li><a href='#RCircos.Get.Start.End.Locations'>
<p>Calculate Start and End Positions for Plot Data</p></a></li>
<li><a href='#RCircos.Get.Track.Positions'>
<p>Calculate Plot Positions for A Data Track</p></a></li>
<li><a href='#RCircos.Get.Zoom.Data'>
<p>Retriev Subset of Plot Data for Zoom</p></a></li>
<li><a href='#RCircos.Get.Zoom.Range'>
<p>Retrieve Genomic Interval from Genomic Data for Zoom Plot</p></a></li>
<li><a href='#RCircos.Heatmap.Data'>
<p>Sample Data for RCircos Heatmap Plot</p></a></li>
<li><a href='#RCircos.Heatmap.Plot'>
<p>Draw Heatmap for One Data Track</p></a></li>
<li><a href='#RCircos.heatmapColors'>
<p>Heatmap Colors Supported by Current Version of RCircos</p></a></li>
<li><a href='#RCircos.Histogram.Data'>
<p>Sample Data for RCircos Histogram Plot</p></a></li>
<li><a href='#RCircos.Histogram.Plot'>
<p>Plot Histogram for One Data Track</p></a></li>
<li><a href='#RCircos.Initialize.Plot.Parameters'>
<p>Initialize RCircos Plot Parameters</p></a></li>
<li><a href='#RCircos.Line.Data'>
<p>Sample Data for RCircos Line Plot</p></a></li>
<li><a href='#RCircos.Line.Plot'>
<p>Draw Lines for One Data Track</p></a></li>
<li><a href='#RCircos.Link.Data'>
<p>Sample Data for RCircos Link Plot</p></a></li>
<li><a href='#RCircos.Link.Line'>
<p>Calculate x and y Coordinates for a Bezire Curve</p></a></li>
<li><a href='#RCircos.Link.Plot'>
<p>Draw Link Lines between Two or More Genomic Positions</p></a></li>
<li><a href='#RCircos.List.Plot.Parameters'>
<p>List RCircos Plot Parameters</p></a></li>
<li><a href='#RCircos.Mouse.Expr.Data'>
<p>Sample Data of Mouse Gene Expression</p></a></li>
<li><a href='#RCircos.Multiple.Species.Core.Components'>
<p>Setup RCircos Core Components with Multiple Species Genomes</p></a></li>
<li><a href='#RCircos.Multiple.Species.Dataset'>
<p>Combine Multiple Data Sets form Different Genomes</p></a></li>
<li><a href='#RCircos.paddingConst'>
<p>Padding Distance Constant between Two Chromosomes</p></a></li>
<li><a href='#RCircos.Parallel.Line.Plot'>
<p>Draw Lines between Two Genomic Positions on Same Chromosome</p></a></li>
<li><a href='#RCircos.Plot.Heatmap.Color.Scale'>
<p>Plot A Color Scale for Heatmap</p></a></li>
<li><a href='#RCircos.plotTypes'>
<p>Plot Types Supported by Current RCircos Version</p></a></li>
<li><a href='#RCircos.Point.Plot'>
<p>Point Plot for One Data Track</p></a></li>
<li><a href='#RCircos.Polygon.Data'>
<p>RCircos Polygon Demo Data</p></a></li>
<li><a href='#RCircos.Polygon.Plot'>
<p>Draw Polygons on One Data Track</p></a></li>
<li><a href='#RCircos.Pseudo.Ideogram.From.Labels'>
<p>Generate A Pseudo Chromosome Ideogram from List of Lables</p></a></li>
<li><a href='#RCircos.Pseudo.Ideogram.From.Table'>
<p>Generate Pseudo Chromosome Ideogram from Plot Data  Table</p></a></li>
<li><a href='#RCircos.Rat.Expr.Data'>
<p>Sample Data of Rat Gene Expression</p></a></li>
<li><a href='#RCircos.Reset.Core.Components'>
<p>Reset Methods for Modifying RCircos Core Components</p></a></li>
<li><a href='#RCircos.Ribbon.Data'>
<p>Sample Data for RCircos Ribbon Plot</p></a></li>
<li><a href='#RCircos.Ribbon.Plot'>
<p>Draw Ribbon between Two Genomic Regions</p></a></li>
<li><a href='#RCircos.Scatter.Data'>
<p>Sample Data for RCircos Scatter Plot</p></a></li>
<li><a href='#RCircos.Scatter.Plot'>
<p>Scater Plot for One Data Track</p></a></li>
<li><a href='#RCircos.Set.Base.Plot.Positions'>
<p>Setup Base Plot Positions</p></a></li>
<li><a href='#RCircos.Set.Core.Components'>
<p>Initialize Core Components for RCircos Plot</p></a></li>
<li><a href='#RCircos.Set.Cytoband.Data'>
<p>Set Chromosome Cytoband Data for Circos Plot</p></a></li>
<li><a href='#RCircos.Set.Plot.Area'>
<p>Setup Plot Area for RCircos Plot</p></a></li>
<li><a href='#RCircos.Set.Zoom.Plot.Positions'>
<p>Set Position for Zoom-in Plot</p></a></li>
<li><a href='#RCircos.Sort.Genomic.Data'>
<p>Sort Genomic or Ideogram Data</p></a></li>
<li><a href='#RCircos.Tile.Data'>
<p>Sample Data for RCircos Tile Plot</p></a></li>
<li><a href='#RCircos.Tile.Plot'>
<p>Plot Tiles for One Data Track</p></a></li>
<li><a href='#RCircos.Track.Outline'>
<p>Draw Outline for A Data Track</p></a></li>
<li><a href='#RCircos.Validate.Cyto.Info'>
<p>Validate Chromosome Ideogram Information</p></a></li>
<li><a href='#RCircos.Validate.Genomic.Data'>
<p>Validate Genomic Data</p></a></li>
<li><a href='#RCircos.Validate.Genomic.Info'>
<p>Validate Genomic Information</p></a></li>
<li><a href='#RCircos.Validate.Plot.Parameters'>
<p>Validate Plot Parameters</p></a></li>
<li><a href='#RCircos.Validate.Track.Positions'>
<p>Validate Track Positions</p></a></li>
<li><a href='#RCircos.Vertical.Line.Plot'>
<p>Draw Vertical Lines on One Data Track</p></a></li>
<li><a href='#RCircos.Workflow'>
<p>A Simple RCircos Plot Workflow</p></a></li>
<li><a href='#RCircos.Zoom.Area.Outline'>
<p>Draw Outline for Zoom Plot Area</p></a></li>
<li><a href='#RCircos.Zoom.Paired.Plot.Positions'>
<p>Calculate Zoomed Index of Circos Plot Positions for Genomic Intervals</p></a></li>
<li><a href='#RCircos.Zoom.Single.Plot.Positions'>
<p>Calculate Zoomed Index of Circos Plot Positions for Single Point Genomic Data</p></a></li>
<li><a href='#RCircos.ZoomIn.Chromosome'>
<p>Zoom in One or Partial Chromosome Ideogram</p></a></li>
<li><a href='#RCircos.ZoomIn.Plot'>
<p>Zoomed-in Plot of Small Genomic Region</p></a></li>
<li><a href='#RCircos.ZoomOut.Chromosome'>
<p>Zoom out Chromosome Ideogram</p></a></li>
<li><a href='#UCSC.Baylor.3.4.Rat.cytoBandIdeogram'>
<p>Cytoband Information for Rat Chromosome Ideogram</p></a></li>
<li><a href='#UCSC.HG19.Human.CytoBandIdeogram'>
<p>Cytoband Information for Human Chromosome Ideogram</p></a></li>
<li><a href='#UCSC.HG38.Human.CytoBandIdeogram'>
<p>Human Chromosome Ideogram Version 38</p></a></li>
<li><a href='#UCSC.Mouse.GRCm38.CytoBandIdeogram'>
<p>Cytoband Inforamtion for Mouse Chromosome Ideogram</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Circos 2D Track Plot</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-12-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Hongen Zhang </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hongen Zhang &lt;hzhang.henry@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A simple and flexible way to generate Circos 2D track plot images for genomic data visualization is implemented in this package. The types of plots include: heatmap, histogram, lines, scatterplot, tiles and plot items for further decorations include connector, link (lines and ribbons), and text (gene) label. All functions require only R graphics package that comes with R base installation.  </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/hzhanghenry/RCircos">https://github.com/hzhanghenry/RCircos</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-12-19 17:10:42 UTC; hzhang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-19 18:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='RCircos-package'>
Circos-like Plot of Genomic Data
</h2><span id='topic+RCircos-package'></span><span id='topic+RCircos'></span>

<h3>Description</h3>

<p>This is the R implement of basic Circos plots of genimic data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> RCircos</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-8-01</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2) </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>RCircos package provides a simple and flexible way to make Circos 2D track plots with R and could be easily integrated into other R data processing and graphic manipulation pipelines for presenting large-scale multi-sample genomic research data. It can also serve as a base tool to generate complex Circos images. Currently, RCircos supports genomic data plot for three species: human, mouse, and rat. Data plots include chromosome ideogram with cytoband, heatmap, histogram, lines, scatterplot, and tiles. Plot items for further decorations have connectors, links (lines and ribbons), and text (gene) labels.  Each plot is implemented with a specific function and input data for all functions are data frames which can be objects read from text files or generated with other R pipelines.
</p>


<h3>Author(s)</h3>

<p>Henry H Zhang
</p>
<p>Maintainer: Henry H Zhang &lt;hzhang.henry@gmail.com&gt;
</p>


<h3>References</h3>

<p>Hongen Zhang, Paul Meltzer, and Sean Davis. RCircos: an R package for Circos 2D track plots. BMC Bioinformatics, 2013, 14:244.
</p>

<hr>
<h2 id='RCircos.Adjust.Scatter.Values'>
Adjust Scatter Plot Data to Fit the Data Track
</h2><span id='topic+RCircos.Adjust.Scatter.Values'></span>

<h3>Description</h3>

<p>Adjust scatter plot values to fit the plot track. The data with lowest value will be plot on the centre of first subtrack and the one with highest value will be on the centre of top subtrack. After this adjustment, values could be added to inside position of plot track as final plot positions. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Adjust.Scatter.Values(scatter.values=NULL, min.value=NULL,  
    max.value=NULL, track.height=NULL, subtrack=NULL) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Adjust.Scatter.Values_+3A_scatter.values">scatter.values</code></td>
<td>

<p>Numeric vector, the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Adjust.Scatter.Values_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value of scatter data.
</p>
</td></tr>
<tr><td><code id="RCircos.Adjust.Scatter.Values_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value of scatter data.
</p>
</td></tr>
<tr><td><code id="RCircos.Adjust.Scatter.Values_+3A_track.height">track.height</code></td>
<td>

<p>Non-negative numeric, height of plot track.
</p>
</td></tr>
<tr><td><code id="RCircos.Adjust.Scatter.Values_+3A_subtrack">subtrack</code></td>
<td>

<p>Non-negative nummeric, the number of sub tracks.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Area.Highlight'>
Highlight PLot Area Cross One or More Tracks
</h2><span id='topic+RCircos.Area.Highlight'></span>

<h3>Description</h3>

<p>Highlight a plot area with transparent color cross one or more tracks. RCircos core components and graphic device must be initialized first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Area.Highlight(highlight.area=NULL, track.num=NULL,  
    side=c("in", "out"), hightlight.color=rgb(0.5, 0.5, 0, 0.5),  
    inside.pos=NULL, outside.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Area.Highlight_+3A_highlight.area">highlight.area</code></td>
<td>

<p>Vector with chromosome name, start and end position to be highlighted.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Highlight_+3A_track.num">track.num</code></td>
<td>

<p>Vector of non-negative integer, which track or track(s) to be highlighted.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Highlight_+3A_side">side</code></td>
<td>

<p>Character vector, location relative to the ideogram, either &quot;in&quot; or &quot;out&quot;.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Highlight_+3A_hightlight.color">hightlight.color</code></td>
<td>

<p>An RGB color definition, alpha value must be defined.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Highlight_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) 
of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Highlight_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) 
of the track.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
data(RCircos.Gene.Label.Data)

RCircos.Set.Core.Components(UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=NULL, tracks.inside=10, tracks.outside=0)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

highlight.area &lt;- c("chr1", 100000, 200000)
RCircos.Area.Highlight(highlight.area=c("chr1", 100000, 200000),  
    track.num=c(1:3), side="in")
RCircos.Area.Highlight(highlight.areac("chr10", 100000, 200000),  
    inside.pos=1.5, outside.pos=2)  

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Area.Plot'>
Paint Areas on One Data Track
</h2><span id='topic+RCircos.Area.Plot'></span>

<h3>Description</h3>

<p>Paint area  on one data track with different height and different locations inside of a track. Plot types includes &quot;mountain&quot;, &quot;curtain&quot;, and &quot;band&quot;. Plot data should have genomic positions(chromosome names, start and end positions) as well as height values. For band type plot, two columns of height values are required.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Area.Plot(area.data=NULL, data.col=c(4,5),
    track.num=NULL, side=c("in", "out"), 
    plot.type=c("mountain", "curtain", "band"), 
    min.value=NULL, max.value=NULL, area.color="gray",
    border.col="black", inside.pos=NULL, outside.pos=NULL, 
    genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Area.Plot_+3A_area.data">area.data</code></td>
<td>

<p>A data frame with two or three columns for genomic positions, one or more columns for heights of each data point, and an optional column for polygon colour.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input data set that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>  
<tr><td><code id="RCircos.Area.Plot_+3A_plot.type">plot.type</code></td>
<td>

<p>Character vector, either &quot;mountain&quot;, &quot;curtain&quot;, or &quot;band&quot;
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value in data column of polygon data.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value in data column of polygon data.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_area.color">area.color</code></td>
<td>

<p>Color names for fill of the area.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_border.col">border.col</code></td>
<td>

<p>Color name for border color, default null.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Area.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Polygon.Data);

RCircos.Set.Core.Components(  
    cyto.info=UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=NULL, tracks.inside=10, tracks.outside=5)  
RCircos.Set.Plot.Area();
RCircos.Chromosome.Ideogram.Plot()

load("RCircos/data/RCircos.Histogram.Data.RData")

area.data &lt;- RCircos.Histogram.Data;
adj.value &lt;- runif(nrow(area.data), 0, 0.4)
area.data["DataT"] &lt;- 0.5 + adj.value
area.data["DataB"] &lt;- 0.5 - adj.value

RCircos.Area.Plot(area.data, data.col=4, plot.type="mountain", 
    inside.pos=1.2, outside.pos=1.5, is.sorted=FALSE)

RCircos.Area.Plot(area.data, data.col=4, plot.type="curtain", 
    inside.pos=0.9, outside.pos=1.1, is.sorted=FALSE)

RCircos.Area.Plot(area.data, data.col=c(5,6), plot.type="band", 
    inside.pos=0.4, outside.pos=0.7, is.sorted=FALSE)
## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Chromosome.Ideogram.Plot'>
Plot Chromosome Ideogram
</h2><span id='topic+RCircos.Chromosome.Ideogram.Plot'></span><span id='topic+RCircos.Draw.Chromosome.Ideogram'></span><span id='topic+RCircos.Highligh.Chromosome.Ideogram'></span><span id='topic+RCircos.Ideogram.Tick.Plot'></span><span id='topic+RCircos.Label.Chromosome.Names'></span>

<h3>Description</h3>

<p>Draw chromosome ideogram, chromosome highlights and chromosome names. Graphic  device must be initialized first. The original function was split to three new  ones in order to plot chromosome highlight or names in different area. RCircos core components and graphic device must be initialized first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Chromosome.Ideogram.Plot(tick.interval=0)
RCircos.Draw.Chromosome.Ideogram(ideo.pos=NULL, ideo.width=NULL)
RCircos.Highligh.Chromosome.Ideogram(highlight.pos=NULL, highlight.width=NULL)
RCircos.Ideogram.Tick.Plot(tick.interval=50, track.for.ticks=3)
RCircos.Label.Chromosome.Names(chr.name.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_tick.interval">tick.interval</code></td>
<td>

<p>Non-negative integer, the distance between two chromosome ideogram ticks in million base pairs. Set to 0 (zero) will ignore the ideogram ticks.
</p>
</td></tr>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_ideo.pos">ideo.pos</code></td>
<td>

<p>Non-negative numeric, the distance of chromosome ideogram position from center of plot area.
</p>
</td></tr>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_ideo.width">ideo.width</code></td>
<td>

<p>Non-negative numeric, the width of chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_highlight.pos">highlight.pos</code></td>
<td>

<p>Non-negative numeric, the distance of chromosome highlight line position from center of plot area.
</p>
</td></tr>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_highlight.width">highlight.width</code></td>
<td>

<p>Non-negative integer, the width of chromosome highlight lines.
</p>
</td></tr>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_track.for.ticks">track.for.ticks</code></td>
<td>

<p>Non-negative integer, total number of tracks for ideogram tick height.
</p>
</td></tr>
<tr><td><code id="RCircos.Chromosome.Ideogram.Plot_+3A_chr.name.pos">chr.name.pos</code></td>
<td>

<p>Non-negative numeric, the distance of chromosome names from center of plot area.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>RCircos.Chromosome.Ideogram.Plot() is the default method to plot chromosome ideogram, chromosome highlights, and chromosome names.  
</p>
<p>RCircos.Draw.Chromosome.Ideogram() plots chromosome ideogram only.  
</p>
<p>RCircos.Highligh.Chromosome.Ideogram() highlights chromosomes with different color lines only.  
</p>
<p>RCircos.Ideogram.Tick.Plot() draws chromosome ideogram ticks starting from chromosome highlight positions.  
</p>
<p>RCircos.Label.Chromosome.Names(chr.name.pos=NULL) add chromosome names for chromosome ideogram at default or customized location.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG38.Human.CytoBandIdeogram)
cyto.info &lt;- UCSC.HG38.Human.CytoBandIdeogram
RCircos.Set.Core.Components(cyto.info, chr.exclude=NULL,  
    tracks.inside=10, tracks.outside=0 )
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Clear.Track'>
Erase One or More Data Tracks
</h2><span id='topic+RCircos.Clear.Track'></span>

<h3>Description</h3>

<p>Erase one or more data track or all tracks from current track to the center of plot area. RCircos core components and graphic device must be already initialized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Clear.Track(track.num=NULL, side=NULL, to.center=FALSE, 
    inside.pos=NULL, outside.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Clear.Track_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track to be erased.
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Track_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram. It could be NULL is both inside.pos and outside.pos are defined.
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Track_+3A_to.center">to.center</code></td>
<td>

<p>Logic, FALSE for erasing current track only and TRUE for erasing all tracks from current track to the center of plot plot area. 
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Track_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Track_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram)
data(RCircos.Link.Data)
data(RCircos.Scatter.Data)

RCircos.Set.Core.Components(UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=c("chrX", "chrY"),  
    tracks.inside=10, tracks.outside=0 )

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

RCircos.Scatter.Plot(RCircos.Scatter.Data, data.col=5,  
    track.num=1, side="in", by.fold=1)
RCircos.Link.Plot(RCircos.Link.Data, track.num=3)  

RCircos.Clear.Track(track.num=1, side="in", to.center=FALSE)
RCircos.Clear.Track(track.num=5, side="in", to.center=TRUE) 

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Clear.Zoom.Area'>
Erase Zoom-in Plot Area
</h2><span id='topic+RCircos.Clear.Zoom.Area'></span>

<h3>Description</h3>

<p>Erase zoom-in plot area. RCircos core components and graphic device must be already initialized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Clear.Zoom.Area(zoom.pos=NULL, track.num=NULL,  
        inside.pos=NULL, outside.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Clear.Zoom.Area_+3A_zoom.pos">zoom.pos</code></td>
<td>

<p>Non-negative integer vector, representing the index of RCircos base plot positions.
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Zoom.Area_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track to be erased.
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Zoom.Area_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Clear.Zoom.Area_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG38.Human.CytoBandIdeogram)
cyto.info &lt;- UCSC.HG38.Human.CytoBandIdeogram
RCircos.Set.Core.Components(cyto.info, chr.exclude=NULL,  
    tracks.inside=10, tracks.outside=0 )

RCircos.Set.Plot.Area()
RCircos.Clear.Zoom.Area(zoom.pos=c(1:10000), inside.pos=2, outside.pos=2.5) 

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Customized.Connection.Plot'>
Draw Connectors between Chromosome Ideogram and Gene Names
</h2><span id='topic+RCircos.Customized.Connection.Plot'></span>

<h3>Description</h3>

<p>Draw connection lines (simple lines plot) between two sets of data points. One example of usage is to label genes at modified plot position and connect the gene label to its genomic position.os core components and graphics device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Customized.Connection.Plot(gene.data, label.data, 
			gene.pos=NULL, label.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Customized.Connection.Plot_+3A_gene.data">gene.data</code></td>
<td>

<p>A data frame with chromosome names and actual genomic positions for the genes to be labeled.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Connection.Plot_+3A_label.data">label.data</code></td>
<td>

<p>A data frame with chromosome name and genomic positions that will be used to label gene names.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Connection.Plot_+3A_gene.pos">gene.pos</code></td>
<td>

<p>Float numeric, scale factor relative to the center of plot area (0) for start points of connection lines.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Connection.Plot_+3A_label.pos">label.pos</code></td>
<td>

<p>Float numeric, scale factor relative to the center of plot area (0) for end points of connection lines.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Customized.Shape.Plot'>
Plot A Customized Shape
</h2><span id='topic+RCircos.Customized.Shape.Plot'></span>

<h3>Description</h3>

<p>Plot one customized shape on a data track. RCircos core components and graphic device must be initialized first. The customized shape should be represented by coordinates of a polygon inside a circle with radius of 1 and default location is at 12 o'clock.  When plotting, the polygon center will be scaled and transformed for new size and location. For example, following code will plot an arrow:
</p>
<p>polygonX &lt;- c(0, -0.7, -0.2, -0.2, 0.2, 0.2, 0.7, 0)
polygonY &lt;- c(-1, 0.7,  0.4,  1,   1,   0.4, 0.7, -1)
plot(c(-2, 2), c(-2, 2)) 
polygon(polygonX, polygonY, col=&quot;red&quot;)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Customized.Shape.Plot(shape.data=NULL, track.num=NULL,  
    side=c("in", "out"), location=NULL, shape.color="red",
    inside.pos=NULL, outside.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_shape.data">shape.data</code></td>
<td>

<p>A two dimensional numeric matrix for coordinates of a polygon
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_location">location</code></td>
<td>

<p>Vector with chromosome name, start position, and end position where the shap to be drawn.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_shape.color">shape.color</code></td>
<td>

<p>Character vector, color for the shape.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Customized.Shape.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)  

data(UCSC.HG19.Human.CytoBandIdeogram)
RCircos.Set.Core.Components(UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=NULL, tracks.inside=10, tracks.outside=0)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

shape &lt;- cbind(c(0, -0.7, -0.2, -0.2, 0.2, 0.2, 0.7, 0),  
        c(-1, 0.7,  0.4,  1,   1,   0.4, 0.7, -1))
RCircos.Customized.Shape.Plot(shape, track.num=1, side="in")

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Data.Point'>
Convert A Genomic Position to An Index of Circos Plot Position
</h2><span id='topic+RCircos.Data.Point'></span>

<h3>Description</h3>

<p>Calculate the index of Circos plot positions (a set of x- and y-coordinates for a circular line) for a genomic position (chromosome name and start  position). This function is for internal use only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Data.Point(chromosome=NULL, start.posotion=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Data.Point_+3A_chromosome">chromosome</code></td>
<td>

<p>Character vector, a chromosome name with prefix of &quot;chr&quot;, e.g., &quot;chr1&quot;.
</p>
</td></tr>
<tr><td><code id="RCircos.Data.Point_+3A_start.posotion">start.posotion</code></td>
<td>

<p>Non-negative integer, start position of a genomic interval (e.g, a gene or a band) on the chromosome.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer representing the index of RCircos plot positions.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.defaultBasePerUnits'>
The Default Base Pair Number of A Circos Unit
</h2><span id='topic+RCircos.defaultBasePerUnits'></span>

<h3>Description</h3>

<p>The default total number of base pairs a Circos unit represents. This is a read only parameter and the actual value in a RCircos session could be modified with get and reset methods for plot parameters.
</p>


<h3>Format</h3>

<p>The format is:
num 30000
</p>

<hr>
<h2 id='RCircos.defaultCharWidth'>
Default Character With in Circos Unit
</h2><span id='topic+RCircos.defaultCharWidth'></span>

<h3>Description</h3>

<p>Default Circos units of a character on RCircos plot. This is the character size for best resolution and is a read only parameter.
</p>


<h3>Format</h3>

<p>The format is:
num 500
</p>

<hr>
<h2 id='RCircos.defaultChromPadding'>
The Default Value of GAP in Circos (Chromosome) Unit between two Chromosome Ideogram 
</h2><span id='topic+RCircos.defaultChromPadding'></span>

<h3>Description</h3>

<p>The default total unit of a gap bewtween the ends of two neighbor chromosome ideograms. This is a read only parameter.
</p>


<h3>Format</h3>

<p>The format is:
num 300
</p>

<hr>
<h2 id='RCircos.defaultCircosUnits'>
Default Circos Units for RCircos
</h2><span id='topic+RCircos.defaultCircosUnits'></span>

<h3>Description</h3>

<p>The default Circos units for RCircos. A Circos unit is a point which represents an inteval of genome length on the circumustance of a circle line. By default an unit represents a point which equal to the length of 30000 base pairs and length of human genome is used to calculate the total number of Circos units. This is a read only parameter and is for internal use to modify plot parameters for RCircso plot with non-human genomes.
</p>


<h3>Format</h3>

<p>The format is:
num 103190
</p>

<hr>
<h2 id='RCircos.defaultTextSize'>
Default Character Size of RCircos Plot
</h2><span id='topic+RCircos.defaultTextSize'></span>

<h3>Description</h3>

<p>The default character size for text plot of RCircos in best resolution. This is a read only parameter and the actual value in a RCircos session could be modified with get and reset methods for plot parameters.
</p>


<h3>Format</h3>

<p>The format is:
num 0.4
</p>

<hr>
<h2 id='RCircos.Env'>
RCircos Environment
</h2><span id='topic+RCircos.Env'></span>

<h3>Description</h3>

<p>RCircos.Env is the user environment to hold RCircos core components and other objects for RCircos plot. Each object in the environment can be viewed with get mothod.
</p>


<h3>Format</h3>

<p>The format is:
&lt;environment: 0xd67930&gt; 
</p>

<hr>
<h2 id='RCircos.Gene.Connector.Plot'>
Draw Connectors between Chromosome Ideogram and Gene Names
</h2><span id='topic+RCircos.Gene.Connector.Plot'></span>

<h3>Description</h3>

<p>Draw a set of connectors (three lines for each connector) between chromosome ideogram or a data track and gene names. RCircos core components and graphics device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Gene.Connector.Plot(genomic.data=NULL, track.num=NULL, 
        side="in", inside.pos=NULL, outside.pos=NULL, 
        genomic.columns=3, is.sorted=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_genomic.data">genomic.data</code></td>
<td>

<p>A data frame with the first four columns for chromosome names, start positions, end positions, and gene names.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the connectors are plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot; representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Connector.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Gene.Label.Data)
data(UCSC.HG19.Human.CytoBandIdeogram)

cyto.info &lt;- UCSC.HG19.Human.CytoBandIdeogram;
RCircos.Set.Core.Components(cyto.info, chr.exclude=NULL,  
       tracks.inside=10, tracks.outside=0)

gene.data &lt;- RCircos.Gene.Label.Data
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Gene.Connector.Plot(gene.data, track.num=1, side="in")
RCircos.Gene.Connector.Plot(gene.data, side="out",  
    inside.pos=2.5, outside.pos=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Gene.Label.Data'>
Sample Data for Gene Labels 
</h2><span id='topic+RCircos.Gene.Label.Data'></span>

<h3>Description</h3>

<p>RCircos.Gene.Label.Data contains genomic position information for 192 genes. Each row of the data contains genomic position and gene name for one gene in the order of chromosome name, start position, end position, and gene name.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Gene.Label.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 192 observations on the following 4 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Gene</code></dt><dd><p>a factor with levels <code>ABL1</code> <code>ACN9</code> <code>ACVR1B</code> <code>AKT1</code> <code>ALK</code> <code>APC</code> <code>ARHGEF2</code> <code>ARID1A</code> <code>ASXL1</code> <code>ATM</code> <code>ATRX</code> <code>BAG3</code> <code>BAI3</code> <code>BAP1</code> <code>BCAN</code> <code>BCL2</code> <code>BRAF</code> <code>BRCA1</code> <code>BRCA2</code> <code>CA12</code> <code>CA9</code> <code>CALU</code> <code>CARD11</code> <code>CBL</code> <code>CCND1</code> <code>CD34</code> <code>CDC73</code> <code>CDH1</code> <code>CDK4</code> <code>CDKN2A</code> <code>CDX2</code> <code>CEACAM7</code> <code>CEBPA</code> <code>CES3</code> <code>CRLF2</code> <code>CSF1R</code> <code>CSF3</code> <code>CTNNA1</code> <code>CTNNB1</code> <code>CUL2</code> <code>CYLD</code> <code>CYP1A1</code> <code>DAXX</code> <code>DCC</code> <code>DES</code> <code>DIRAS1</code> <code>DIRC2</code> <code>DKK3</code> <code>DLD</code> <code>DMBT1</code> <code>DNMT3A</code> <code>DPYD</code> <code>EGF</code> <code>EGFR</code> <code>EGR3</code> <code>EIF4G2</code> <code>EML4</code> <code>ENO1</code> <code>ENO2</code> <code>ERBB2</code> <code>ERBB3</code> <code>ERBB4</code> <code>ERG</code> <code>EZH2</code> <code>F10</code> <code>FAM123B</code> <code>FBXW7</code> <code>FGFR1</code> <code>FGFR2</code> <code>FGFR3</code> <code>FH</code> <code>FHIT</code> <code>FLCN</code> <code>FLT3</code> <code>FOXL2</code> <code>FZR1</code> <code>GATA1</code> <code>GATA2</code> <code>GATA3</code> <code>GFAP</code> <code>GNA11</code> <code>GNAQ</code> <code>GNAS</code> <code>GRP</code> <code>GSTM1</code> <code>HIVEP3</code> <code>HNF1A</code> <code>HRAS</code> <code>IDH1</code> <code>IDH2</code> <code>IFNA1</code> <code>IL2</code> <code>ITGB5</code> <code>JAK1</code> <code>JAK2</code> <code>JAK3</code> <code>KDR</code> <code>KIT</code> <code>KRAS</code> <code>KRT20</code> <code>MAGI1</code> <code>MAP2K4</code> <code>MEN1</code> <code>MET</code> <code>MGMT</code> <code>MIB1</code> <code>MKI67</code> <code>MLH1</code> <code>MPL</code> <code>MSH2</code> <code>MSH6</code> <code>MUC1</code> <code>MUC17</code> <code>MUC2</code> <code>MUTYH</code> <code>MVP</code> <code>MYC</code> <code>MYD88</code> <code>NAT1</code> <code>NAT2</code> <code>NES</code> <code>NF1</code> <code>NF2</code> <code>NOTCH1</code> <code>NOTCH2</code> <code>NPM1</code> <code>NRAS</code> <code>PBRM1</code> <code>PCNA</code> <code>PDGFRA</code> <code>PDZD4</code> <code>PGR</code> <code>PHOX2B</code> <code>PIK3CA</code> <code>PMS2</code> <code>PPP2R1A</code> <code>PRCC</code> <code>PRKAR1A</code> <code>PTCH1</code> <code>PTEN</code> <code>PTGS2</code> <code>PTPN11</code> <code>RB1</code> <code>REEP5</code> <code>RET</code> <code>RNF139</code> <code>RNF2</code> <code>RPS15</code> <code>RUNX1</code> <code>SDHA</code> <code>SDHAF1</code> <code>SDHAF2</code> <code>SDHB</code> <code>SDHC</code> <code>SDHD</code> <code>SELT</code> <code>SETD2</code> <code>SLC38A1</code> <code>SLC6A2</code> <code>SLTM</code> <code>SMAD4</code> <code>SMARCA4</code> <code>SMARCB1</code> <code>SMO</code> <code>SMOX</code> <code>SMUG1</code> <code>SOCS1</code> <code>SRC</code> <code>SST</code> <code>STC1</code> <code>STK11</code> <code>SUFU</code> <code>SYP</code> <code>TCF7L2</code> <code>TET2</code> <code>TFE3</code> <code>TFEB</code> <code>TMEM97</code> <code>TNFAIP3</code> <code>TNFSF13</code> <code>TP53</code> <code>TPD52L2</code> <code>TPM4</code> <code>TSC1</code> <code>TSC2</code> <code>TSHR</code> <code>TYK2</code> <code>VHL</code> <code>VIM</code> <code>WT1</code> <code>XRCC1</code> <code>ZNF135</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished data. 
</p>

<hr>
<h2 id='RCircos.Gene.Name.Plot'>
Draw Gene Names on a Data Track
</h2><span id='topic+RCircos.Gene.Name.Plot'></span>

<h3>Description</h3>

<p>Label gene names along chromosome ideogram or a data track. RCircos core components and graphic device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Gene.Name.Plot(gene.data=NULL, name.col=NULL,  
    track.num=NULL, side="in", inside.pos=NULL,  
    outside.pos=NULL, genomic.columns=3, is.sorted=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_gene.data">gene.data</code></td>
<td>

<p>A data frame. The first three columns should be chromosome name, start position, end position.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_name.col">name.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input data that contains gene names.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the gene names are plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Gene.Name.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
data(RCircos.Gene.Label.Data)

cyto.info &lt;- UCSC.HG19.Human.CytoBandIdeogram
RCircos.Set.Core.Components(cyto.info,  
    chr.exclude=c("chrX", "chrY"),  
    num.inside=10, num.outside=0)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

gene.data &lt;- RCircos.Gene.Label.Data;
RCircos.Gene.Name.Plot(gene.data, name.col=4,  
    track.num=2, side="in")
RCircos.Gene.Name.Plot(gene.data, name.col=4,  
    track.num=2, side="out",  
    inside.pos=2.5, outside.pos=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Get.Arrow.Shape'>
Get Default Coordinates for An Arrow Shape
</h2><span id='topic+RCircos.Get.Arrow.Shape'></span>

<h3>Description</h3>

<p>Get default coordinates for an arrow shape. The arrow is represented as a polygon inside of a circle with radius of 1. The cocordinates will be different based which side of chromosome ideogram the arrow will be plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Arrow.Shape(side="in")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Arrow.Shape_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two dimensional numeric matrix for x and y coordinates of a polygon.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(RCircos)
arrow &lt;- RCircos.Get.Arrow.Shape("in")
arrow &lt;- RCircos.Get.Arrow.Shape("out")
</code></pre>

<hr>
<h2 id='RCircos.Get.Chromosome.Order'>
Find the order of Chromosome Names
</h2><span id='topic+RCircos.Get.Chromosome.Order'></span>

<h3>Description</h3>

<p>Generate an ordered chromosome names from input. For human and other mammalian animals, numeric names (integers or Roman numbers) will go first followed by chromosome X, Y, and M. If all chromosome names are character only, they will be in alphabetical order. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Chromosome.Order(chromosomes=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Chromosome.Order_+3A_chromosomes">chromosomes</code></td>
<td>

<p>Character vector, names of chromosomes.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector, ordered chromosome name list.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(RCircos)
chromosomes &lt;- paste0("chr", c(1:10, 21,22, 11:20, "X", "Y"))  
chromosomes &lt;- RCircos.Get.Chromosome.Order(chromosomes)  
</code></pre>

<hr>
<h2 id='RCircos.Get.Core.Components'>
Methods to Retrieve RCircos Core Components Used in Current RCircos Session
</h2><span id='topic+RCircos.Get.Plot.Parameters'></span><span id='topic+RCircos.Get.Plot.Positions'></span><span id='topic+RCircos.Get.Plot.Ideogram'></span>

<h3>Description</h3>

<p>Retrieve base plot positions, plot ideogram, and plot parameters used in current RCircos session. RCircos core components must be initialized first. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Plot.Parameters()
RCircos.Get.Plot.Positions()
RCircos.Get.Plot.Ideogram()
</code></pre>


<h3>Value</h3>

<p>RCircos.Get.Plot.Parameters() returns a list that holding all parameters used in current RCircos session.
</p>
<p>RCircos.Get.Plot.Positions() returns a data frame with three columns for x, and y coordinates of a circular line with radius of 1 as well as the degrees used for text rotation at each point of the positions. 
</p>
<p>RCircos.Get.Plot.Ideogram() returns a data frame containing all information for chromosome ideogram plot. The first five column are orginal cytoband data and following are chromosome highlight colors, and plot colors, length, number of chromosome unit, and relative location on the circular layout of each band.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Data.Point.Height'>
Calculate the Height of Data in A Track
</h2><span id='topic+RCircos.Get.Data.Point.Height'></span>

<h3>Description</h3>

<p>Calculate data point height inside a plot track such as scatter location, top or bottom location of a bar, layer of a tile or parallel link line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Data.Point.Height(plot.values=NULL, min.value=NULL, 
    max.value=NULL, plot.type=NULL, track.height=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Data.Point.Height_+3A_plot.values">plot.values</code></td>
<td>

<p>Numeric, the data to be plotted on a data track.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Data.Point.Height_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, the minimum value of data range.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Data.Point.Height_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, the maximum value of data range.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Data.Point.Height_+3A_plot.type">plot.type</code></td>
<td>

<p>Character vector, plot type, valid values are &quot;bar&quot;, &quot;histogram&quot;, &quot;uniform&quot;, or &quot;points&quot;.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Data.Point.Height_+3A_track.height">track.height</code></td>
<td>

<p>Non-negative numeric, height of plot track.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector with values between 0 ~ 1
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data.values &lt;- runif(1000, -4, 11)
library(RCircos)
data.height &lt;- RCircos.Get.Data.Point.Height(data.values,  
    min.value=-4, max.value=14, plot.type="points",  
    track.height=NULL)
</code></pre>

<hr>
<h2 id='RCircos.Get.Default.Parameters'>
Retrieve Parameter Values Stored in RCircos Environment
</h2><span id='topic+RCircos.Get.Default.Circos.Units'></span><span id='topic+RCircos.Get.Default.Base.Per.Units'></span><span id='topic+RCircos.Get.Default.Chrom.Padding'></span><span id='topic+RCircos.Get.Padding.Constant'></span><span id='topic+RCircos.Get.Supported.HeatmapColors'></span><span id='topic+RCircos.Get.Supported.Plot.Types'></span><span id='topic+RCircos.Get.Default.Char.Width'></span><span id='topic+RCircos.Get.Default.Text.Size'></span>

<h3>Description</h3>

<p>These methods are used to view the defaul parameters stored in RCircos Environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Default.Circos.Units()
RCircos.Get.Default.Base.Per.Units()
RCircos.Get.Default.Chrom.Padding()
RCircos.Get.Padding.Constant()
RCircos.Get.Supported.HeatmapColors()
RCircos.Get.Supported.Plot.Types()
RCircos.Get.Default.Char.Width()
RCircos.Get.Default.Text.Size()
</code></pre>


<h3>Value</h3>

<p>RCircos.Get.Default.Circos.Units() returns the default number of units (points) for a circular line. This values is derived from  human genome length (3x10^9) and one unit represents length of 3000 base pairs.
</p>
<p>RCircos.Get.Default.Base.Per.Units() returns the default number of base pairs that a unit (point) will represent.
</p>
<p>RCircos.Get.Default.Chrom.Padding() returns the default lenght of gaps between two neighbot chromosomes in chromosome (Circos) unit.
</p>
<p>RCircos.Get.Padding.Constant() returns the default ratio of chromosome padding length and total default Circos (chromosome) units.
</p>
<p>RCircos.Get.Supported.HeatmapColors() returns the types of colors for heatmap plot.
</p>
<p>RCircos.Get.Supported.Plot.Types() returns the plot types supported by current RCircos version.
</p>
<p>RCircos.Get.Default.Char.Width() returns the default number of units a character will span on the plot.
</p>
<p>RCircos.Get.Default.Text.Size() returns the default size of text in RCircos.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Gene.Label.Locations'>
Calculate Plot Positions for Gene Labels
</h2><span id='topic+RCircos.Get.Gene.Label.Locations'></span>

<h3>Description</h3>

<p>This function calculates new plot positions from genomic positions for gene labeling. In case there are too many genes in a genomic interval, the lables may become overlapped so that correct numbers and new plot positions are often needed to make the labels readable. If all labels cannot be plotted for the given genomic interval extra labels will be ignored. This function is for internal use only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Gene.Label.Locations(genomic.data=NULL, 
	genomic.columns=3, is.sorted = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Gene.Label.Locations_+3A_genomic.data">genomic.data</code></td>
<td>

<p>A data frame with leading columns for chromosome name, start position, and/or end position.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Gene.Label.Locations_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Gene.Label.Locations_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>All or subset of input data frame with a new column for plot positions.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Gene.Name.Plot.Parameters'>
Calculate Parameters for Gene Name Plot
</h2><span id='topic+RCircos.Get.Gene.Name.Plot.Parameters'></span>

<h3>Description</h3>

<p>Calculate maximum number of gene names to be plotted for each chromosome based on plot RCircos plot parameters. RCircos core components must be initialized first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Gene.Name.Plot.Parameters()
</code></pre>


<h3>Value</h3>

<p>A data frame with names, maximum number of labels could be plotted, start position, end position, and label width for each chromosome.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Heatmap.Color.Scale'>
Generate Color Scales for Heatmap Plot
</h2><span id='topic+RCircos.Get.Heatmap.Color.Scale'></span>

<h3>Description</h3>

<p>Create color scales for heatmap plot. Currently there are six color scales could be generated: <br /><br />
BlueWhiteRed:   colors from blue (lowest) to white then red (highest) <br />
GreenWhiteRed:  colors from green (lowest) to white then red (highest) <br />
GreenYellowRed: colors from green (lowest) to yellow then red (highest) <br />
GreenBlackRed:  colors from green (lowest) to black then red (highest) <br />
YellowToRed:    colors from yellow (lowest) to red (highest) <br />
BlackOnly:      black only
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Heatmap.Color.Scale(heatmap.color=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Heatmap.Color.Scale_+3A_heatmap.color">heatmap.color</code></td>
<td>

<p>Character vector, one of &quot;BlueWhiteRed&quot;, &quot;GreenWhiteRed&quot;, &quot;GreenYellowRed&quot;, &quot;GreenBlackRed&quot;, &quot;YellowToRed&quot;, and &quot;BlackOnly&quot;.  
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector or two column matrix of rgb colors.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(RCircos)
colorScales &lt;- RCircos.Get.Heatmap.Color.Scale(heatmap.color="BlueWhiteRed")
</code></pre>

<hr>
<h2 id='RCircos.Get.Heatmap.Color.Scale.Location'>
Calculate X and Y Coordinates for Color Scale
</h2><span id='topic+RCircos.Get.Heatmap.Color.Scale.Location'></span>

<h3>Description</h3>

<p>Calculate X and Y coordinates for the topleft corner of heatmap color scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Heatmap.Color.Scale.Location(scale.location = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Heatmap.Color.Scale.Location_+3A_scale.location">scale.location</code></td>
<td>

<p>Integer of 1 ~ 12, represents the plot location (default is 1):<br />
1:  &quot;bottom&quot; <br />
2:  &quot;left&quot; <br />
3:  &quot;top&quot; <br />
4.: &quot;right&quot; <br />
5:  &quot;bottomleft&quot; <br />
6:  &quot;bottomright&quot; <br />
7:  &quot;leftright&quot; <br />
8:  &quot;leftbottom&quot; <br />
9:  &quot;topleft&quot; <br />
10: &quot;topright&quot; <br />
11: &quot;righttop&quot; <br />
12: &quot;rightbottom&quot; <br />
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric verctor of length 2 for x and y coordinates
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(RCircos)
location &lt;- RCircos.Get.Heatmap.Color.Scale.Location(1)
</code></pre>

<hr>
<h2 id='RCircos.Get.Heatmap.Data.Colors'>
Set Colors for One Track of Heatmap
</h2><span id='topic+RCircos.Get.Heatmap.Data.Colors'></span>

<h3>Description</h3>

<p>Assign colors to each value of a numeric vector for heatmap plot. The colour scale needs to be calculated from whole datasets when there are more than one data column so the minimum and maximum values for heatmap data must be provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Heatmap.Data.Colors(heatmap.value=NULL, min.value=NULL, max.value=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Heatmap.Data.Colors_+3A_heatmap.value">heatmap.value</code></td>
<td>

<p>A numeric vector.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Heatmap.Data.Colors_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, the minimu value for heatmap scale.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Heatmap.Data.Colors_+3A_max.value">max.value</code></td>
<td>

<p>Numeric,the maximum value for heatmap scale.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of R color names with length same as the length of data values.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Heatmap.Data)
hColors &lt;- RCircos.Get.Heatmap.Data.Colors(RCircos.Heatmap.Data[,5], min.value=-3, max.value=3)
## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Get.Link.Colors'>
Set up Colors for Link Lines or Ribbons
</h2><span id='topic+RCircos.Get.Link.Colors'></span>

<h3>Description</h3>

<p>Assign colors for link lines or Ribbons. The defult colors are rainbow with lenght of total link lines or ribbons. If by.chromosome is set to TRUE, red color will be used for links between positions of same chromosomes and blue color for links between different chromosomes. Users can also define their own colors by appending a column with color names to link data and give the column name as &quot;PlotColor&quot;. 
</p>
<p>This function is for internal use only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Link.Colors(link.data, genomic.columns=3, by.chromosome=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Link.Colors_+3A_link.data">link.data</code></td>
<td>

<p>A data frame with paired genomic positions in each row. The first three columns are for chromosome names, start and end position of one genomic position followed by three columns for chromosome names, start and end position of the second genomic position. 
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Link.Colors_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total columns that define the positions of a genomic interval (chromosome name, start and optional end positions).
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Link.Colors_+3A_by.chromosome">by.chromosome</code></td>
<td>

<p>Logic. If true, red color will be used for links between positions of same chromosomes and blue color for links between different chromosomes. Otherwise, user defined or random colors (rainbow) will be automatically assigned for each line.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color names with length same as the rows of input data.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Paired.Points.Positions'>
Calcualte Plot Positions for A Genomic Interval
</h2><span id='topic+RCircos.Get.Paired.Points.Positions'></span>

<h3>Description</h3>

<p>Calculate x and y coordinates for a genomic interval which needs two plot positions (start and end) such as link lines/ribbons, tiles, parallel lines. As a set of points for a circlar line is held in the RCircos session, we only need the index of the point for each genomic position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Paired.Points.Positions(genomic.data=NULL, genomic.columns=3,  
    plot.type=c("link", "ribbon", "pLink", "polygon", "tile"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Paired.Points.Positions_+3A_genomic.data">genomic.data</code></td>
<td>

<p>A data frame contains paired genomic positions (chromosome names, start and end positions). The data does not need to be sorted.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Paired.Points.Positions_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position (chromosome name, start and/or end position).
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Paired.Points.Positions_+3A_plot.type">plot.type</code></td>
<td>

<p>Chraracter vector, either &quot;link&quot;, &quot;ribbon&quot;, &quot;pLink&quot;, &quot;polygon&quot;, or &quot;tile&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame same as input but with two new columns for index of plot positions on circular line. 
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
data(RCircos.Link.Data)

RCircos.Set.Core.Components(UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=NULL, track.inside=10, tracks.outside=0)
linkData&lt;-RCircos.Get.Paired.Points.Positions(RCircos.Link.Data,
    genomic.columns=3, plot.type="link")

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Get.Plot.Boundary'>
Calculate Boundary of A Data Track
</h2><span id='topic+RCircos.Get.Plot.Boundary'></span>

<h3>Description</h3>

<p>Calculate plot track location based on track number, plot side, or user defined inside and outside position relative to plot area center. If all arguments are defined, user defined inside and outside position will be used. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Plot.Boundary(track.num=NULL, side=NULL, 
	inside.pos=NULL, outside.pos=NULL, erase.area=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Plot.Boundary_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Plot.Boundary_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram. It could be NULL if both inside.pos and outside.pos are defined.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Plot.Boundary_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Plot.Boundary_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Plot.Boundary_+3A_erase.area">erase.area</code></td>
<td>

<p>Logic, if the track is going to be erased.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of length 2, x and y coordinates of a data track.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Plot.Colors'>
Assign Plot Colors for Non-Link Data Plot
</h2><span id='topic+RCircos.Get.Plot.Colors'></span>

<h3>Description</h3>

<p>Assign plot colors for each data point of genomic data except of link plot (link lines or ribbons). Users can define their own colors by appending a column with color names to plot data and give the column name as &quot;PlotColor&quot;. Otherwise, default colors will be used.
</p>
<p>This function is for internal use only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Plot.Colors(plot.data, color)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Plot.Colors_+3A_plot.data">plot.data</code></td>
<td>

<p>A data frame with the first three columns as chromosome name, start and end position followed by data values for one or more samples.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Plot.Colors_+3A_color">color</code></td>
<td>

<p>Character vector of color names.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color names with length same as the rows of input data.
</p>


<h3>Author(s)</h3>

<p>Hongen Zhang
</p>

<hr>
<h2 id='RCircos.Get.Plot.Layers'>
Get the Layer Numbers for Tile Plot
</h2><span id='topic+RCircos.Get.Plot.Layers'></span>

<h3>Description</h3>

<p>Check out overlaps between different genomic positions on same chromosome and set up layer numbers for each line of the genomic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Plot.Layers(genomic.data=NULL, genomic.columns=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Plot.Layers_+3A_genomic.data">genomic.data</code></td>
<td>

<p>A data frame with genomic positions (chromosomes, start and end positions) and the positions should  be already validated and sorted by chromosome then start position.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Plot.Layers_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic positions.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A non-negative integer vector with length same as the total rows of input data.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(RCircos)
data(RCircos.Tile.Data)
layers &lt;- RCircos.Get.Plot.Layers(RCircos.Tile.Data, 3)
</code></pre>

<hr>
<h2 id='RCircos.Get.Polygon.Height'>
Calculate the Height of Polygons in A Data Track
</h2><span id='topic+RCircos.Get.Polygon.Height'></span>

<h3>Description</h3>

<p>Calculate polygon heights (bottom and top location) inside a plot track. Polygon can be draw from inside position to outside position,  from outside position to inside position, or from the middle to both direction of the track. This is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Polygon.Height(data.heights, min.value=NULL,  
    max.value=NULL, inside.pos=NULL, outside.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Polygon.Height_+3A_data.heights">data.heights</code></td>
<td>

<p>Numeric vector, the data to be plotted on a data track.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Polygon.Height_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, the minimum value of data range.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Polygon.Height_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, the maximum value of data range.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Polygon.Height_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, the close position of a plot track relative to the center of plot area.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Polygon.Height_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, the far position of a plot track relative  to the center of plot area.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Non-negative umeric matrix of column 2 representing the bottom and top positions of polygons.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Single.Point.Positions'>
Calculate Plot Positions for Genomic Data
</h2><span id='topic+RCircos.Get.Single.Point.Positions'></span>

<h3>Description</h3>

<p>Calculate x and y coordinates of one genomic position for each row of a genomic data set. As a set of points for a circlar line is held in the RCircos session, we only need the index of the point for each genomic position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Single.Point.Positions(genomic.data=NULL, genomic.columns=3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Single.Point.Positions_+3A_genomic.data">genomic.data</code></td>
<td>

<p>A data frame contains genomic positions (at least two or three columns for chromosome names, start and/or end positions). 
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Single.Point.Positions_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position (chromosome name, start and/or end position).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame same as input but with a new column for index of plot positions on a circular line. 
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Heatmap.Data)
dat &lt;- RCircos.Get.Single.Point.Positions(RCircos.Heatmap.Data)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Get.Start.End.Locations'>
Calculate Start and End Positions for Plot Data
</h2><span id='topic+RCircos.Get.Start.End.Locations'></span>

<h3>Description</h3>

<p>Generate start and end positions from single point for special plot types such as heatmap, histogram, or rectangle-like which need start and end positions along the track. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Start.End.Locations(plot.data, plot.width)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Start.End.Locations_+3A_plot.data">plot.data</code></td>
<td>

<p>Data frame returned from RCircos.Get.Single.Point.Positions() and the last column is the plot locations
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Start.End.Locations_+3A_plot.width">plot.width</code></td>
<td>

<p>Non-negative integer in number of base pairs, e.g., width or heatmap or histogram of each data points
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Get.Track.Positions'>
Calculate Plot Positions for A Data Track 
</h2><span id='topic+RCircos.Get.Track.Positions'></span>

<h3>Description</h3>

<p>Calculate inner and outer plot positions for a data track. RCircos core components must be initialized first. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Track.Positions(side=NULL, track.num=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Track.Positions_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Track.Positions_+3A_track.num">track.num</code></td>
<td>

<p>Integer, representing the ordinal number of the plot track
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of length 2. The outer and inner positions of a track.
</p>


<h3>Author(s)</h3>

<p>Hongen Zhang
</p>

<hr>
<h2 id='RCircos.Get.Zoom.Data'>
Retriev Subset of Plot Data for Zoom
</h2><span id='topic+RCircos.Get.Zoom.Data'></span>

<h3>Description</h3>

<p>Retrieve subset of plot data based on gene name or row header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Zoom.Data(plot.data=NULL, name.col= NULL,
 	genomic.columns=3, target.gene=NULL, neighbor.genes=5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Zoom.Data_+3A_plot.data">plot.data</code></td>
<td>

<p>A data frame containing genomic positions, gene names, and plot values. The data should be already sorted by chromosome names then start positions.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Zoom.Data_+3A_name.col">name.col</code></td>
<td>

<p>Non-negative integer, which column is for gene names.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Zoom.Data_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Zoom.Data_+3A_target.gene">target.gene</code></td>
<td>

<p>Character vector, target gene names.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Zoom.Data_+3A_neighbor.genes">neighbor.genes</code></td>
<td>

<p>Non-negative integer, number of neighbor genes on each side of taget gene to be included in subset.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame, subset of input data.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Heatmap.Data)
zoom.data &lt;- RCircos.Get.Zoom.Data(RCircos.Heatmap.Data, 
	name.col=4, target.gene="TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Get.Zoom.Range'>
Retrieve Genomic Interval from Genomic Data for Zoom Plot
</h2><span id='topic+RCircos.Get.Zoom.Range'></span>

<h3>Description</h3>

<p>Extract zoom in range (chromosome name, start position, and end position) from zoom data with a gene name or row header. If  using this function to get zoom-in range, there will be no need to call RCircos.Validate.Genomic.Info().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Get.Zoom.Range(zoom.data=NULL, genomic.columns=3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Get.Zoom.Range_+3A_zoom.data">zoom.data</code></td>
<td>

<p>A data frame with genomic positions, gene names, and plot values for genes to be zoomed-in only.
</p>
</td></tr>
<tr><td><code id="RCircos.Get.Zoom.Range_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with chromosome names, start end end positions.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Heatmap.Data)
zoom.data &lt;- RCircos.Get.Zoom.Data(RCircos.Heatmap.Data,  
    name.col=4, target.gene="TP53")
zoom.info &lt;- RCircos.Get.Zoom.Range(zoom.data, genomic.columns=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Heatmap.Data'>
Sample Data for RCircos Heatmap Plot
</h2><span id='topic+RCircos.Heatmap.Data'></span>

<h3>Description</h3>

<p>A dataset with the first three columns as chromosome names, start position, and end position followed by a column of gene names then columns of gene expression values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Heatmap.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 6660 observations on the following 10 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GeneName</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>X786.O</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A498</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A549.ATCC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ACHN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BT.549</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CAKI.1</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Modified from NCBI GEO dataset: GSE32474.
</p>

<hr>
<h2 id='RCircos.Heatmap.Plot'>
Draw Heatmap for One Data Track
</h2><span id='topic+RCircos.Heatmap.Plot'></span>

<h3>Description</h3>

<p>Draw one track of heatmap. RCircos core components and graphic device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Heatmap.Plot(heatmap.data=NULL, data.col=NULL, 
    track.num=NULL, side=c("in", "out"), 
    min.value=NULL, max.value=NULL, inside.pos=NULL, 
    outside.pos=NULL, genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_heatmap.data">heatmap.data</code></td>
<td>

<p>A data frame with the first three columns as chromosome names, start position and end position of each data point followed by data values for one or more samples.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input data set that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative Integer, representing the ordinal number of the plot track where the heatmap will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value for heatmap scale.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value for heatmap scale
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Heatmap.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
data(RCircos.Heatmap.Data)

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

RCircos.Heatmap.Plot(RCircos.Heatmap.Data, data.col=5, 
	track.num=4, side="in")
RCircos.Heatmap.Plot(RCircos.Heatmap.Data, data.col=5, 
	side="in", inside.pos=0.9, outside.pos=1.2)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.heatmapColors'>
Heatmap Colors Supported by Current Version of RCircos
</h2><span id='topic+RCircos.heatmapColors'></span>

<h3>Description</h3>

<p>Default heatmap colors (total of 6) supported by current version of RCircos. This is a read only data and can be viwed with the method of RCircos.Get.Supported.HeatmapColors().
</p>


<h3>Format</h3>

<p>The format is:
chr [1:6] &quot;BlueWhiteRed&quot; &quot;GreenWhiteRed&quot; &quot;GreenYellowRed&quot; 
&quot;GreenBlackRed&quot; &quot;YellowToRed&quot; &quot;BlackOnly&quot;
</p>

<hr>
<h2 id='RCircos.Histogram.Data'>
Sample Data for RCircos Histogram Plot 
</h2><span id='topic+RCircos.Histogram.Data'></span>

<h3>Description</h3>

<p>A dataset with the first three columns as chromosome names, start position, and end position followed by a column of histogram data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Histogram.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 324 observations on the following 4 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Data</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpulished data. 
</p>

<hr>
<h2 id='RCircos.Histogram.Plot'>
Plot Histogram for One Data Track
</h2><span id='topic+RCircos.Histogram.Plot'></span>

<h3>Description</h3>

<p>Draw one track of histogram. RCircos core components and graphic device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Histogram.Plot(hist.data=NULL, data.col=4,
    track.num=NULL, side=c("in", "out"), 
    min.value=NULL, max.value=NULL, inside.pos=NULL, 
    outside.pos=NULL, genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Histogram.Plot_+3A_hist.data">hist.data</code></td>
<td>

<p>A data frame with the first three columns as chromosome names, start position and end position of each data point followed by data values for one or more samples.
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input data that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the histogram will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value for histogram
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value for histogram
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Histogram.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Histogram.Data);

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=5) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

RCircos.Histogram.Plot(RCircos.Histogram.Data, 
	data.col=4, track.num=1, side="in")
RCircos.Histogram.Plot(RCircos.Histogram.Data, 
	data.col=4, side="in", inside.pos=1.7, 
	outside.pos=1.8)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Initialize.Plot.Parameters'>
Initialize RCircos Plot Parameters
</h2><span id='topic+RCircos.Initialize.Plot.Parameters'></span>

<h3>Description</h3>

<p>Setup default values for RCircos plot parameters. All parameters are held in one list. This function is for internal use only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Initialize.Plot.Parameters(tracks.inside, tracks.outside)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Initialize.Plot.Parameters_+3A_tracks.inside">tracks.inside</code></td>
<td>

<p>Non-negative integer, how many tracks will be put inside of chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Initialize.Plot.Parameters_+3A_tracks.outside">tracks.outside</code></td>
<td>

<p>Non-negative integer, how many tracks will be put outside of chromosome ideogram.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Line.Data'>
Sample Data for RCircos Line Plot 
</h2><span id='topic+RCircos.Line.Data'></span>

<h3>Description</h3>

<p>A data frame with the first three columns as chromosome name, start and end position followed by columns of DNA copy number variant data (num.mark and seg.mean).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Line.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 2037 observations on the following 5 variables.
</p>

<dl>
<dt><code>chromosome</code></dt><dd><p>a factor with levels <code>1</code> <code>10</code> <code>11</code> <code>12</code> <code>13</code> <code>14</code> <code>15</code> <code>16</code> <code>17</code> <code>18</code> <code>19</code> <code>2</code> <code>20</code> <code>21</code> <code>22</code> <code>3</code> <code>4</code> <code>5</code> <code>6</code> <code>7</code> <code>8</code> <code>9</code> <code>X</code> <code>Y</code></p>
</dd>
<dt><code>start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>stop</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>num.mark</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>seg.mean</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished data. 
</p>

<hr>
<h2 id='RCircos.Line.Plot'>
Draw Lines for One Data Track
</h2><span id='topic+RCircos.Line.Plot'></span>

<h3>Description</h3>

<p>Draw one track of continue lines which connect each neighbor data points. RCircos core components and graphics device must bae initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Line.Plot(line.data=NULL, data.col=4, track.num=NULL,  
    side=c("in", "out"), min.value=NULL, max.value=NULL,  
    inside.pos=NULL, outside.pos=NULL, genomic.columns=3,  
    is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Line.Plot_+3A_line.data">line.data</code></td>
<td>

<p>A data frame with the first three columns for chromosome names, start and end position of each data point followed by data values for one or more samples.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input dataset that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value of line data.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value of line data
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Line.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Line.Data);

RCircos.Set.Core.Components(  
    cyto.info=UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=c("chrX", "chrY"),  
    num.inside=10, num.outside=0) 

RCircos.Set.Plot.Area() 
RCircos.Chromosome.Ideogram.Plot()  

RCircos.Line.Plot(RCircos.Line.Data, data.col=5,  
    track.num=1, side="in")
RCircos.Line.Plot(RCircos.Line.Data, data.col=5,  
    side="in", inside.pos=1.5, outside.pos=1.75)  

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Link.Data'>
Sample Data for RCircos Link Plot 
</h2><span id='topic+RCircos.Link.Data'></span>

<h3>Description</h3>

<p>A data frame containing paired genomic position data for each row. The content of each row must be in the order of name, start and end position of chromosome A then name, start and end position of chromosome B. This data frame is directly used to draw link lines without extra processing.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Link.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 71 observations on the following 6 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr14</code> <code>chr15</code> <code>chr17</code> <code>chr19</code> <code>chr2</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Chromosome.1</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code></p>
</dd>
<dt><code>chromStart.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd.1</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished data. 
</p>

<hr>
<h2 id='RCircos.Link.Line'>
Calculate x and y Coordinates for a Bezire Curve
</h2><span id='topic+RCircos.Link.Line'></span>

<h3>Description</h3>

<p>Calculate a quandratic Bezire curve between two points with (0,0) as control points. This function is for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Link.Line(line.start=NULL, line.end=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Link.Line_+3A_line.start">line.start</code></td>
<td>

<p>The x and y coordinates of a point where the Bezier curve starts.
</p>
</td></tr>
<tr><td><code id="RCircos.Link.Line_+3A_line.end">line.end</code></td>
<td>

<p>The x and y coordinates of a point where the Bezier curve ends.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hongen Zhang
</p>

<hr>
<h2 id='RCircos.Link.Plot'>
Draw Link Lines between Two or More Genomic Positions
</h2><span id='topic+RCircos.Link.Plot'></span>

<h3>Description</h3>

<p>Draw lines (quandratic Bezier curves) between paired genomic positions. RCircos core components and graphic device must be initialize before drawing. Link lines are always in the center of plot area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Link.Plot(link.data=NULL, track.num=NULL,
    by.chromosome=FALSE, start.pos=NULL, genomic.columns=3,
    is.sorted=TRUE, lineWidth=rep(1, nrow(link.data)))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Link.Plot_+3A_link.data">link.data</code></td>
<td>

<p>A data frame with paired genomic positions in each row. The two sets of genomic positions must have same number of columns, e.g., chromosome names, start and end positions or only chromosome names and start positions.     
</p>
</td></tr>
<tr><td><code id="RCircos.Link.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the link lines will start.
</p>
</td></tr>
<tr><td><code id="RCircos.Link.Plot_+3A_by.chromosome">by.chromosome</code></td>
<td>

<p>Logic. If true, red color will be used for links between positions of same chromosomes and blue color for links between different chromosomes. Otherwise, user defined or random colors (rainbow) will be automatically assigned for each line.
</p>
</td></tr>
<tr><td><code id="RCircos.Link.Plot_+3A_start.pos">start.pos</code></td>
<td>

<p>Non-negative numeric, scale factor relative to chromosome ideogram position. Must be smaller than 1.
</p>
</td></tr>
<tr><td><code id="RCircos.Link.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Link.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>  
<tr><td><code id="RCircos.Link.Plot_+3A_linewidth">lineWidth</code></td>
<td>

<p>Non-negative integer vector, width for each link line.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Link.Data);

RCircos.Set.Core.Components(
	cyto.inf=UCSC.HG19.Human.CytoBandIdeogram,  
	chr.exclude=c("chrX", "chrY"), 
	num.inside=10, num.outside=0)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(RCircos.Link.Data, track.num=4,
	by.chromosome=TRUE)
RCircos.Link.Plot(RCircos.Link.Data, start.pos=0.75,
	by.chromosome=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.List.Plot.Parameters'>
List RCircos Plot Parameters
</h2><span id='topic+RCircos.List.Plot.Parameters'></span>

<h3>Description</h3>

<p>Print out onto screen all plot parameters used by current RCircos session. RCircos core components must be already initialized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.List.Plot.Parameters()
</code></pre>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) RCircos.List.Plot.Parameters()

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Mouse.Expr.Data'>
Sample Data of Mouse Gene Expression
</h2><span id='topic+RCircos.Mouse.Expr.Data'></span>

<h3>Description</h3>

<p>A data frame containing mouse genomic position information and gene expresio
n values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Mouse.Expr.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 16499 observations on the following 5 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>txStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>txEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Gene</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Expr.Mean</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Modified from GEO dataset: GSE42081.
</p>

<hr>
<h2 id='RCircos.Multiple.Species.Core.Components'>
Setup RCircos Core Components with Multiple Species Genomes
</h2><span id='topic+RCircos.Multiple.Species.Core.Components'></span>

<h3>Description</h3>

<p>Setup RCircos core components with chromosome ideograms of multiple species for visulization of similarity and difference of genomic data between different species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Multiple.Species.Core.Components(cyto.info.list, species.list, 
    chr.exclude= NULL, tracks.inside, tracks.outside)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Multiple.Species.Core.Components_+3A_cyto.info.list">cyto.info.list</code></td>
<td>

<p>A List contains data frames for chromosome ideogram data of different species. All data frames in the list must have same columns.
</p>
</td></tr>
<tr><td><code id="RCircos.Multiple.Species.Core.Components_+3A_species.list">species.list</code></td>
<td>

<p>Character vector represent multiple species. Each one will be used as prefix of chromosome names for relavant species. The order of species must match the order of the species in cyto.info.list.
</p>
</td></tr>
<tr><td><code id="RCircos.Multiple.Species.Core.Components_+3A_chr.exclude">chr.exclude</code></td>
<td>

<p>Character vector of chromosome names to be excluded from plot, e.g., chr.exclude &lt;- c(&quot;chrX&quot;, &quot;chrY&quot;);
</p>
</td></tr>
<tr><td><code id="RCircos.Multiple.Species.Core.Components_+3A_tracks.inside">tracks.inside</code></td>
<td>

<p>Integer, total number of data tracks inside of chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Multiple.Species.Core.Components_+3A_tracks.outside">tracks.outside</code></td>
<td>

<p>Integer, total number of data tracks outside of chromosome ideogram.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hongen Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
data(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)

cyto.list &lt;- list(UCSC.Mouse.GRCm38.CytoBandIdeogram,  
               UCSC.Baylor.3.4.Rat.cytoBandIdeogram);
species &lt;- c("M", "R")

RCircos.Multiple.Species.Core.Components(  
       cyto.list, species, chr.exclude=NULL,  
       tracks.inside=10, tracks.outside=0)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Multiple.Species.Dataset'>
Combine Multiple Data Sets form Different Genomes
</h2><span id='topic+RCircos.Multiple.Species.Dataset'></span>

<h3>Description</h3>

<p>Combine and modify the chromosome names in multiple species datasets to match the chromosomes in multiple species cytoband data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Multiple.Species.Dataset(data.list, species)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Multiple.Species.Dataset_+3A_data.list">data.list</code></td>
<td>

<p>List of genomic data from multiple species.
</p>
</td></tr>
<tr><td><code id="RCircos.Multiple.Species.Dataset_+3A_species">species</code></td>
<td>

<p>Character vector for prefix of chromosome names to identify different species. The order of species in species list must match the species in data.list.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame same as the input data list but with modified chromosome names.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Mouse.Expr.Data)
data(RCircos.Rat.Expr.Data)
dataSets &lt;- list(RCircos.Mouse.Expr.Data, RCircos.Rat.Expr.Data)
dataset &lt;- RCircos.Get.Multiple.Species.Dataset(dataSets,  
    species.list=c("M", "L"))

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.paddingConst'>
Padding Distance Constant between Two Chromosomes
</h2><span id='topic+RCircos.paddingConst'></span>

<h3>Description</h3>

<p>Default length of padding area between two chromosomes on chromosome ideogram track. This is a read only data and can be viewed with RCircos.Get.Padding.Constant() method.
</p>


<h3>Format</h3>

<p>The format is:
num 0.00291
</p>

<hr>
<h2 id='RCircos.Parallel.Line.Plot'>
Draw Lines between Two Genomic Positions on Same Chromosome
</h2><span id='topic+RCircos.Parallel.Line.Plot'></span>

<h3>Description</h3>

<p>Plot link lines inside a data track between two genomic positions on same chromosome. RCircos core components and graphic device must be initialized first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Parallel.Line.Plot(line.data=NULL, track.num=NULL, 
    side=c("in", "out"), line.width=1, inside.pos=NULL,
    outside.pos=NULL, genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_line.data">line.data</code></td>
<td>

<p>A data frame containing chromosome, start and end position for each line.
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, the number of track to plot
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, relative location to ideogram, either &quot;in&quot; or &quot;out&quot;.
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_line.width">line.width</code></td>
<td>

<p>Non-negative numeric, same as lwd in graphic package
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be 3.
</p>
</td></tr>
<tr><td><code id="RCircos.Parallel.Line.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Line.Data);

RCircos.Set.Core.Components(
    cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
    chr.exclude=c("chrX", "chrY"), 
    tracks.inside=10, tracks.outside=5) 
RCircos.Set.Plot.Area();
RCircos.Chromosome.Ideogram.Plot()

RCircos.Parallel.Line.Plot(line.data, track.num=5, side="in")
RCircos.Parallel.Line.Plot(line.data, line.width=2, 
    inside.pos=2, outside.pos=2.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Plot.Heatmap.Color.Scale'>
Plot A Color Scale for Heatmap
</h2><span id='topic+RCircos.Plot.Heatmap.Color.Scale'></span>

<h3>Description</h3>

<p>Plot a color scale showing the color ranges of heatmap. One or more tracks should be defined for plots in outside of chromosome ideogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Plot.Heatmap.Color.Scale(max.value=NULL, min.value=NULL,
        color.type="BlueWhiteRed", scale.location=1, 
        scale.width=0, scale.height=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Plot.Heatmap.Color.Scale_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value of heatmap scale
</p>
</td></tr>
<tr><td><code id="RCircos.Plot.Heatmap.Color.Scale_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value of heatmap scale
</p>
</td></tr>
<tr><td><code id="RCircos.Plot.Heatmap.Color.Scale_+3A_color.type">color.type</code></td>
<td>

<p>Character vector for color specification, either &quot;BlueWhiteRed&quot;, &quot;GreenWhiteRed&quot;, &quot;GreenYellowRed&quot;,&quot;GreenBlackRed&quot;, &quot;YellowToRed&quot;, or &quot;BlackOnly&quot;. Default is &quot;BlueWhiteRed&quot;.
</p>
</td></tr>
<tr><td><code id="RCircos.Plot.Heatmap.Color.Scale_+3A_scale.location">scale.location</code></td>
<td>

<p>Non negative integer representing color scale location, 1~4 for 
bottom, left, top, and right.
</p>
</td></tr>
<tr><td><code id="RCircos.Plot.Heatmap.Color.Scale_+3A_scale.width">scale.width</code></td>
<td>

<p>Non-negative numeric, width of color scale. if not defined, 1/2 of x axis will be used
</p>
</td></tr>
<tr><td><code id="RCircos.Plot.Heatmap.Color.Scale_+3A_scale.height">scale.height</code></td>
<td>

<p>Non-negative numeric, height of color scale. if not defined, 1/10 of scaleWidth axis will be used
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plot.new()
plot.window(c(-5, 5), c(-5, 5))
RCircos.Plot.Heatmap.Color.Scale(max.value=3, min.value=-3,
    scaleLocation=1, colorType="BlueWhiteRed", 
    scaleWidth=0, scaleHeight=0)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.plotTypes'>
Plot Types Supported by Current RCircos Version
</h2><span id='topic+RCircos.plotTypes'></span>

<h3>Description</h3>

<p>All supported plot types saved in RCircos Environment. The values are read only throug RCircos.Get.Supported.Plot.Types().
</p>


<h3>Format</h3>

<p>The format is:
char[1:15] &quot;link&quot;, &quot;ribbon&quot;, &quot;pLink&quot;, &quot;polygon&quot;, &quot;tile&quot;,  &quot;ideogram&quot;, &quot;heatmap&quot;, &quot;bar&quot;, &quot;histogram&quot;, &quot;cLine&quot;,&quot;vLine&quot;, &quot;points&quot;,  &quot;connector&quot;, &quot;ticks&quot;,  &quot;text&quot;
</p>

<hr>
<h2 id='RCircos.Point.Plot'>
Point Plot for One Data Track
</h2><span id='topic+RCircos.Point.Plot'></span>

<h3>Description</h3>

<p>Plot points on a track without track outlines and sub-track lines. The size and location od each point can be adjusted with the data values. RCircos core components and graphic device must be initialized first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Point.Plot(point.data=NULL, data.col=4, 
    track.num=NULL, side=c("in", "out"), 
    min.value=NULL, max.value=NULL, 
    point.type=19, with.height=TRUE, 
    with.size=FALSE, point.scale=1, 
    inside.pos=NULL, outside.pos=NULL, 
    genomic.columns=3,  is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Point.Plot_+3A_point.data">point.data</code></td>
<td>

<p>A data frame with leading columns of chromosome names, start position and/or end position of each data point followed by data values for one or more samples
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input dataset that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value of point data.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value of point data
</p>
</td></tr>  
<tr><td><code id="RCircos.Point.Plot_+3A_point.type">point.type</code></td>
<td>

<p>Non-negative integer for pch. Default is 19.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_with.height">with.height</code></td>
<td>

<p>Logic, if TURE, the location of each point will be adjusted according to the data value.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_with.size">with.size</code></td>
<td>

<p>Logic, if TURE, the size of each point will be adjusted according to the data value.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_point.scale">point.scale</code></td>
<td>

<p>Non-negative numeric, more scale for point size, must be greater than or equal to 1.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Point.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram,
	chr.exclude=c("chrX", "chrY"), 
	num.inside=10, num.outside=0)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

data(RCircos.Scatter.Data)
RCircos.Scatter.Plot(RCircos.Scatter.Data, 
	data.col=5, track.num=4, side="in", 
	with.height=TRUE, with.size=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Polygon.Data'>
RCircos Polygon Demo Data
</h2><span id='topic+RCircos.Polygon.Data'></span>

<h3>Description</h3>

<p>A data frame with four columns for chromosome names, start and end position for each genomic interval, and values for polygon heights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("RCircos.Polygon.Data")</code></pre>


<h3>Format</h3>

<p>A data frame with 77 observations on the following 4 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Data</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>


<hr>
<h2 id='RCircos.Polygon.Plot'>
Draw Polygons on One Data Track
</h2><span id='topic+RCircos.Polygon.Plot'></span>

<h3>Description</h3>

<p>Plot polygons with different height and different locations inside of a track. Polygon plot is an alternative bar plot that takes both positive and negative height values and genomic intervals with different lengths. Plot data should have genomic positions(chromosome names, start and end positions) as well as height values. Optional column for polygon colors may follow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Polygon.Plot(polygon.data=NULL, data.col=NULL,  
    track.num=NULL, side=c("in", "out"), border.col=NULL,  
    polygon.col="pink", min.value=NULL, max.value=NULL,  
    inside.pos=NULL, outside.pos=NULL, genomic.columns=3,  
    is.sorted = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Polygon.Plot_+3A_polygon.data">polygon.data</code></td>
<td>

<p>A data frame with three columns for genomic positions, one or columns for polygon heights, and an optional column for polygon colour.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input data set that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_border.col">border.col</code></td>
<td>

<p>Vector of color names for border color, default null.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_polygon.col">polygon.col</code></td>
<td>

<p>Color name for fill of polygon.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value in data column of polygon data.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value in data column of polygon data.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Polygon.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Polygon.Data);

RCircos.Set.Core.Components(  
    cyto.info=UCSC.HG19.Human.CytoBandIdeogram,  
    chr.exclude=c("chrX", "chrY"), 
    tracks.inside=10, tracks.outside=5)  
RCircos.Set.Plot.Area();
RCircos.Chromosome.Ideogram.Plot()

RCircos.Polygon.Plot(RCircos.Polygon.Data,  
    track.num=1, side="in")

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Pseudo.Ideogram.From.Labels'>
Generate A Pseudo Chromosome Ideogram from List of Lables
</h2><span id='topic+RCircos.Pseudo.Ideogram.From.Labels'></span>

<h3>Description</h3>

<p>Generate a pseudo ideogram from a character list for RCircos plot Each chromosome will have same length and no stain colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Pseudo.Ideogram.From.Labels(chromosomes=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Pseudo.Ideogram.From.Labels_+3A_chromosomes">chromosomes</code></td>
<td>

<p>A vector of labels (such as gene names) used for chromosome names and each element must be unique.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame as an ideogram data table.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
geneNames &lt;- paste0("Gene_", 1:20)
cyto.info &lt;- RCircos.Pseudo.Ideogram.From.Labels(geneNames)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Pseudo.Ideogram.From.Table'>
Generate Pseudo Chromosome Ideogram from Plot Data  Table
</h2><span id='topic+RCircos.Pseudo.Ideogram.From.Table'></span>

<h3>Description</h3>

<p>Generate a pseudo ideogram from plot data for Circos plot. Plot data must have at least two columns serving as chromosome names and band names. An optional gene length or locations can also be provided, e.g.,
</p>
<p>Groups  LocusName   LocusPosition
Group1  1           1.5
Group1  3           2.0
Group1  4           3.0
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Pseudo.Ideogram.From.Table(plot.data=NULL, location.col=NULL, band.col=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Pseudo.Ideogram.From.Table_+3A_plot.data">plot.data</code></td>
<td>

<p>A data frame with two or three columns, the first column is for chromosome names.
</p>
</td></tr>
<tr><td><code id="RCircos.Pseudo.Ideogram.From.Table_+3A_location.col">location.col</code></td>
<td>

<p>Non negative integer, column number in plot.data from which end positions will be derived for each chromosome or each band. This argument must be defined and could be either length of the locus or relative locus position.
</p>
</td></tr>
<tr><td><code id="RCircos.Pseudo.Ideogram.From.Table_+3A_band.col">band.col</code></td>
<td>

<p>Non negative integer, column number in plot.data which serves as band names, this argument is optional.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with ideogram data.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Gene.Label.Data);
plot.data &lt;- RCircos.Gene.Label.Data;
cyto.info &lt;- RCircos.Get.Pseudo.Ideogram(plot.data=plot.data, 
	location.col=2, band.col=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Rat.Expr.Data'>
Sample Data of Rat Gene Expression
</h2><span id='topic+RCircos.Rat.Expr.Data'></span>

<h3>Description</h3>

<p>A data frame containing rat genes,  genomic position and expression values of these genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Rat.Expr.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 11426 observations on the following 5 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code></p>
</dd>
<dt><code>txStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>txEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Gene</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Expr.Mean</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Modified from GEO dataset: GSE42081.
</p>

<hr>
<h2 id='RCircos.Reset.Core.Components'>
Reset Methods for Modifying RCircos Core Components
</h2><span id='topic+RCircos.Reset.Plot.Ideogram'></span><span id='topic+RCircos.Reset.Plot.Parameters'></span><span id='topic+RCircos.Reset.Plot.Positions'></span>

<h3>Description</h3>

<p>Reset RCircos core components: plot ideogram, plot position, and plot parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Reset.Plot.Ideogram(chrom.ideo)
RCircos.Reset.Plot.Positions(plot.positions)
RCircos.Reset.Plot.Parameters(new.params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Reset.Core.Components_+3A_chrom.ideo">chrom.ideo</code></td>
<td>

<p>Data frame, object of RCircos cytoband data returned from RCircos.Get.Plot.Ideogram(). 
</p>
</td></tr>
<tr><td><code id="RCircos.Reset.Core.Components_+3A_plot.positions">plot.positions</code></td>
<td>

<p>Data frame, object of RCircos plot positions returned from RCircos.Get.Plot.Positions().
</p>
</td></tr>
<tr><td><code id="RCircos.Reset.Core.Components_+3A_new.params">new.params</code></td>
<td>

<p>List, object of RCircos plot positions returned from RCircos.Get.Plot.Parameters()
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);

RCircos.Set.Core.Components(
    cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
    chr.exclude=c("chrX", "chrY"), 
    tracks.inside=10, tracks.outside=5) 

ideo &lt;- RCircos.Get.Plot.Ideogram()
rows &lt;- which(ideo$BandColor=="red"
ideo$BandColor[rows] &lt;- "green"
RCircos.Reset.Plot.Ideogram(ideo)

params &lt;- RCircos.Get.Plot.Parameters()
params$base.per.unit &lt;- 6000
RCircos.Reset.Plot.Paramters(params)

positions &lt;- RCircos.Get.Plot.Positions();
positions[1, 3] &lt;- 0;
RCircos.Reset.Plot.Positions(positions);

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Ribbon.Data'>
Sample Data for RCircos Ribbon Plot
</h2><span id='topic+RCircos.Ribbon.Data'></span>

<h3>Description</h3>

<p>A data frame containing paired genomic position data for each row. The contents of each row are in the order of name, start and end positions of chromosome A then name, start and end positions of chromosome B. This data frame is directly used to draw ribbons without extra processing.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Ribbon.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 4 observations on the following 6 variables.
</p>

<dl>
<dt><code>chromA</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr5</code> <code>chr8</code></p>
</dd>
<dt><code>chromStartA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEndA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromB</code></dt><dd><p>a factor with levels <code>chr10</code> <code>chr13</code> <code>chr17</code> <code>chr18</code></p>
</dd>
<dt><code>chromStartB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEndB</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished data. 
</p>

<hr>
<h2 id='RCircos.Ribbon.Plot'>
Draw Ribbon between Two Genomic Regions
</h2><span id='topic+RCircos.Ribbon.Plot'></span>

<h3>Description</h3>

<p>Draw various ribbons instead of lines with uniform thickness. The thickness of ribbons are relative to the size of the genomic regions to be linked. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Ribbon.Plot(ribbon.data=NULL, track.num=NULL, 
    by.chromosome=FALSE, twist=FALSE, start.pos=NULL, 
    genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_ribbon.data">ribbon.data</code></td>
<td>

<p>A data frame with paired genomic positions in each row. The first three columns of each row are for chromosome name, start and end positions of one genomic position followed by three columns for chromosome name, start and end positions of the second genomic position. 
</p>
</td></tr>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non negative-integer, representing the ordinal number of the plot track where the ribbons will start.
</p>
</td></tr>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_by.chromosome">by.chromosome</code></td>
<td>

<p>Logic, If true, red color will be used for ribbons between positions of same chromosomes and blue color for ribbons between different chromosomes. Otherwise, different colors will be automatically assigned for each ribbon.
</p>
</td></tr>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_twist">twist</code></td>
<td>

<p>Logic, TRUE for the ribbons with twisted shape and FALSE for no twist.
</p>
</td></tr>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_start.pos">start.pos</code></td>
<td>

<p>Non-negative numeric, scale factor relative to chromosome ideogram position. Must be smaller than 1.
</p>
</td></tr>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Ribbon.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Link.Data);
data(RCircos.Ribbon.Data);

cyto.info &lt;- UCSC.HG19.Human.CytoBandIdeogram
chr.exclude &lt;- c("chrX", "chrY")
RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

RCircos.Link.Plot(RCircos.Link.Data, track.num=4, twist=TRUE)
RCircos.Ribbon.Plot(RCircos.Ribbon.Data, 
	track.num=4, twist=TRUE, by.chromosome=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Scatter.Data'>
Sample Data for RCircos Scatter Plot 
</h2><span id='topic+RCircos.Scatter.Data'></span>

<h3>Description</h3>

<p>A data frame with the first three columns as chromosome name, start and end position followed by columns of DNA copy number variant data (num.mark and seg.mean). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Scatter.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 1757 observations on the following 5 variables.
</p>

<dl>
<dt><code>chromosome</code></dt><dd><p>a factor with levels <code>1</code> <code>10</code> <code>11</code> <code>12</code> <code>13</code> <code>14</code> <code>15</code> <code>16</code> <code>17</code> <code>18</code> <code>19</code> <code>2</code> <code>20</code> <code>21</code> <code>22</code> <code>3</code> <code>4</code> <code>5</code> <code>6</code> <code>7</code> <code>8</code> <code>9</code> <code>X</code> <code>Y</code></p>
</dd>
<dt><code>start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>stop</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>num.mark</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>seg.mean</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>


<hr>
<h2 id='RCircos.Scatter.Plot'>
Scater Plot for One Data Track
</h2><span id='topic+RCircos.Scatter.Plot'></span>

<h3>Description</h3>

<p>Draw one track of scatter plot. RCircos core components and graphics device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Scatter.Plot(scatter.data=NULL, data.col=4, 
    track.num=NULL, side=c("in", "out"), by.fold=0,
    min.value=NULL, max.value=NULL, inside.pos=NULL, 
    outside.pos=NULL, genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Scatter.Plot_+3A_scatter.data">scatter.data</code></td>
<td>

<p>A data frame with leading columns of chromosome names, start position and/or end position of each data point followed by data values for one or more samples
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the column in input dataset that contains the data to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_by.fold">by.fold</code></td>
<td>

<p>Zero or a positive float number, if it is greater than zero, any data point with a value greater than or equal to the by.fold will be plot as red color and data points with values less or equal to the negative by.fold will be plot as blue colr. Otherwise, black color is used.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value of scatter plot data.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value of scatter plot data
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Scatter.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

data(RCircos.Scatter.Data)
RCircos.Scatter.Plot(RCircos.Scatter.Data, data.col=5, 
	track.num=4, side="in", by.fold=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Set.Base.Plot.Positions'>
Setup Base Plot Positions
</h2><span id='topic+RCircos.Set.Base.Plot.Positions'></span>

<h3>Description</h3>

<p>Calculate x and y coordinates for a circular line that served as base plot positions to derive positions for all other plot items. Total number of coordinates are based on total chromosome units and padding. Degrees that text should be rotated at each point are also calculated. This function is for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Set.Base.Plot.Positions(total.points=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Set.Base.Plot.Positions_+3A_total.points">total.points</code></td>
<td>

<p>Non-negative integer, total number of points to form a circular line.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Set.Core.Components'>
Initialize Core Components for RCircos Plot
</h2><span id='topic+RCircos.Set.Core.Components'></span>

<h3>Description</h3>

<p>Setup core components for RCircos Plot including plot parameters, cytoband data, and base plot positions. This function must be called before chromosome ideogram and any other data plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Set.Core.Components(cyto.info, chr.exclude = NULL, tracks.inside, tracks.outside)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Set.Core.Components_+3A_cyto.info">cyto.info</code></td>
<td>

<p>A data frame with chromosome ideogram data and columns are ordered by chromosome names, start position, end position, band names, and stain intensity for each chromosome band.
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Core.Components_+3A_chr.exclude">chr.exclude</code></td>
<td>

<p>Character vector of chromosome names to be excluded from plot, e.g., chr.exclude &lt;- c(&quot;chrX&quot;, &quot;chrY&quot;);
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Core.Components_+3A_tracks.inside">tracks.inside</code></td>
<td>

<p>Non-negative integer, total number of data tracks inside of chromosome ideogram to be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Core.Components_+3A_tracks.outside">tracks.outside</code></td>
<td>

<p>Non-negative integer, total number of data tracks outside of chromosome ideogram to be plotted.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hongen Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) 

## End(Not run) 
</code></pre>

<hr>
<h2 id='RCircos.Set.Cytoband.Data'>
Set Chromosome Cytoband Data for Circos Plot
</h2><span id='topic+RCircos.Set.Cytoband.Data'></span>

<h3>Description</h3>

<p>Set chromosome cytoband data (one of three RCircos core components) for Circos plot. The cytoband data will be put into RCircos Environment. This function is for internal use and users may not directly call this function and use get and reset methods instead when need to modify ideogram data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Set.Cytoband.Data(cyto.band.info=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Set.Cytoband.Data_+3A_cyto.band.info">cyto.band.info</code></td>
<td>

<p>A data frame containing the chromosome ideogram data returned from RCircos.Validate.Cyto.Info() or a sorted ideogram table with correct chromosome names,start and end positions, band names, and staing status.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Set.Plot.Area'>
Setup Plot Area for RCircos Plot
</h2><span id='topic+RCircos.Set.Plot.Area'></span>

<h3>Description</h3>

<p>Open an new window for plotting. RCircos core components must be initialized before call this function. All four margins are set to 0.25 and window size are set to plot.radius in RCircos parameters. This function is optional and could be replaced by call par(), plot.new(), and plot.window() if users want control the plot window by themselves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Set.Plot.Area(margins = 0.25, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Set.Plot.Area_+3A_margins">margins</code></td>
<td>

<p>Non-negative numeric, value for margins around plot area.
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Plot.Area_+3A_...">...</code></td>
<td>

<p>Other arguments passed...
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=5) 
RCircos.Set.Plot.Area() 

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Set.Zoom.Plot.Positions'>
Set Position for Zoom-in Plot
</h2><span id='topic+RCircos.Set.Zoom.Plot.Positions'></span>

<h3>Description</h3>

<p>Set plot position for zooming-in plot by selecting index of default plot positions. The zoomed plot will be aligned with the original plot position such as gene position or a cytoband. The zoom area was defined as fraction of circle circumference, default is 1/8.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Set.Zoom.Plot.Positions(zoom.info=NULL, total.genes=11,
    area.length=0.25, fixed.width=FALSE, gene.width=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Set.Zoom.Plot.Positions_+3A_zoom.info">zoom.info</code></td>
<td>

<p>A vector with chromosome name, start and end position to be zoomed.
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Zoom.Plot.Positions_+3A_total.genes">total.genes</code></td>
<td>

<p>Non-negative integer from 1 ~ maximum number of genes to be plotted. The maximum number of genes is calculated based on
track height and will be adjusted inside the function.
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Zoom.Plot.Positions_+3A_area.length">area.length</code></td>
<td>

<p>Non-negative numeric, for better layout, it must be smaller than or equal to 1/4 (quarter of the circumstance).
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Zoom.Plot.Positions_+3A_fixed.width">fixed.width</code></td>
<td>

<p>Logic, if the plot witdth for each data point is fixed or not.
</p>
</td></tr>
<tr><td><code id="RCircos.Set.Zoom.Plot.Positions_+3A_gene.width">gene.width</code></td>
<td>

<p>Non-negative integer, number of units for gene width.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector for index of RCircos.Base.Position
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram);

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) 

zoom.info &lt;- c("chr1", 10000, 50000)
zoom.pos &lt;- RCircos.Set.Zoom.Plot.Positions(zoom.info)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Sort.Genomic.Data'>
Sort Genomic or Ideogram Data
</h2><span id='topic+RCircos.Sort.Genomic.Data'></span>

<h3>Description</h3>

<p>Sort genomics/ideogram data. The order of chromosome names should be numeric names (integers or Roman numbers) first then character names. If chromosome names are all characters alphabets order will be used. This function could be used before making RCircos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Sort.Genomic.Data(genomic.data=NULL, is.ideo=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Sort.Genomic.Data_+3A_genomic.data">genomic.data</code></td>
<td>

<p>A data frame with the first two or three columns for chromosome names, start and end positions. If it is ideogram data, next two columns must be band names, and Giemsa stain status.
</p>
</td></tr>
<tr><td><code id="RCircos.Sort.Genomic.Data_+3A_is.ideo">is.ideo</code></td>
<td>

<p>Logic, weither the genomic data is ideogram or not.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame same as input data but ordered by chromosome names then start positions
</p>


<h3>Author(s)</h3>

<p>Henry
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)

data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
cyto &lt;- UCSC.Mouse.GRCm38.CytoBandIdeogram
ideogram &lt;- RCircos.Sort.Genomic.Data(  
    genomic.data=UCSC.Mouse.GRCm38.CytoBandIdeogram,  
    is.ideo=TURE)

data(RCircos.Line.Data)
lineData &lt;- RCircos.Sort.Genomic.Data( 	genomic.data=RCircos.Line.Data, 
    is.ideo=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Tile.Data'>
Sample Data for RCircos Tile Plot 
</h2><span id='topic+RCircos.Tile.Data'></span>

<h3>Description</h3>

<p>A data frame contining genomic position data only (chromosome, start and end positions). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RCircos.Tile.Data)</code></pre>


<h3>Format</h3>

<p>A data frame with 152 observations on the following 3 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished Data. 
</p>

<hr>
<h2 id='RCircos.Tile.Plot'>
Plot Tiles for One Data Track
</h2><span id='topic+RCircos.Tile.Plot'></span>

<h3>Description</h3>

<p>Draw one track of Tiles. RCircos core components and graphics device must be initialized before drawing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Tile.Plot(tile.data=NULL, track.num=NULL, 
    side= c("in", "out"), inside.pos=NULL, outside.pos=NULL, 
    genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Tile.Plot_+3A_tile.data">tile.data</code></td>
<td>

<p>A data frame with the three columns as chromosome names, start position and end position of each data point. Other column(s) is optional.
</p>
</td></tr>
<tr><td><code id="RCircos.Tile.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Tile.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Tile.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Tile.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Tile.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative interger. Must be 3 for tile plot.
</p>
</td></tr>
<tr><td><code id="RCircos.Tile.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

data(RCircos.Tile.Data);
RCircos.Tile.Plot(tile.data=RCircos.Tile.Data, 
	track.num=4, side="in")

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Track.Outline'>
Draw Outline for A Data Track
</h2><span id='topic+RCircos.Track.Outline'></span>

<h3>Description</h3>

<p>Draw outline with subtracks for one data track. RCircos core components and graphic device must be initialized first. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Track.Outline(inside.pos=NULL, outside.pos=NULL,
    num.layers=1, chrom.list=NULL, track.colors=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Track.Outline_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Track.Outline_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>  <tr><td><code id="RCircos.Track.Outline_+3A_num.layers">num.layers</code></td>
<td>

<p>Non-negative integer, number of sub-tracks lines. 0 for no sub-track line.
</p>
</td></tr>
<tr><td><code id="RCircos.Track.Outline_+3A_chrom.list">chrom.list</code></td>
<td>

<p>List of chromosome names for which outline will be drawn.
</p>
</td></tr>
<tr><td><code id="RCircos.Track.Outline_+3A_track.colors">track.colors</code></td>
<td>

<p>Vector of color names for outline of each chromosome.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=5) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

locations &lt;- RCircos.Track.Positions(side="in", track.num=1);  
out.pos &lt;- locations[1];
in.pos  &lt;- locations[2];
subtracks &lt;- 5;
RCircos.Track.Outline(inside.pos=locations[2], 
	outside.pos=locations[1], num.layers=5,
	chrom.list=NULL, track.colors=NULL) 

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Validate.Cyto.Info'>
Validate Chromosome Ideogram Information
</h2><span id='topic+RCircos.Validate.Cyto.Info'></span>

<h3>Description</h3>

<p>Validate chromosome ideogram information for correct chromosome order including of correct order of chromosome names, chromosome start, and end positions. The prefix of &quot;chr&quot; in chromosome names is no longer required version 1.2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Validate.Cyto.Info(cyto.info=NULL, chr.exclude=NULL, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Validate.Cyto.Info_+3A_cyto.info">cyto.info</code></td>
<td>

<p>A data frame with chromosome ideogram data, e.g., an object returned by function of read.table() which read a file containing full information of cytoBandIdeo table from UCSC genome browser.
</p>
</td></tr>
<tr><td><code id="RCircos.Validate.Cyto.Info_+3A_chr.exclude">chr.exclude</code></td>
<td>

<p>Character vector, name(s) of chromosomes to be excluded from RCircos plot.
</p>
</td></tr>
<tr><td><code id="RCircos.Validate.Cyto.Info_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the cyto.info has been sorted.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with validated chromosome ideogram table.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG38.Human.CytoBandIdeogram)
cyto.info &lt;- RCircos.Validate.Cyto.Info(
    cyto.info=UCSC.HG38.Human.CytoBandIdeogram,
    chr.exclude=NULL, is.sorted=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Validate.Genomic.Data'>
Validate Genomic Data
</h2><span id='topic+RCircos.Validate.Genomic.Data'></span>

<h3>Description</h3>

<p>Validate input dataset for correct chromosome names, chromosome start, and chromosome end positions. Chromosome names will be converted to character vectors if they are factor variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Validate.Genomic.Data(genomic.data=NULL, 
    plot.type=c("plot", "link"), genomic.columns=3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Validate.Genomic.Data_+3A_genomic.data">genomic.data</code></td>
<td>

<p>Data frame with genomic position data. 
</p>
</td></tr>
<tr><td><code id="RCircos.Validate.Genomic.Data_+3A_plot.type">plot.type</code></td>
<td>

<p>Character vector, either &quot;plot&quot; or &quot;link&quot;.
</p>
</td></tr>
<tr><td><code id="RCircos.Validate.Genomic.Data_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position (chromosome name, start and/or end position).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None. Function will exit with an error prompt.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(RCircos.Heatmap.Data)
RCircos.Validate.Genomic.Data(RCircos.Heatmap.Data, 
    plot.type="plot", genomic.columns=3)

data(RCircos.Link.Data)
RCircos.Validate.Genomic.Data(RCircos.Link.Data, 
    plot.type="link", genomic.columns=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Validate.Genomic.Info'>
Validate Genomic Information
</h2><span id='topic+RCircos.Validate.Genomic.Info'></span>

<h3>Description</h3>

<p>Check out if the chromosome name, start and end positions in the range of ideogram data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Validate.Genomic.Info(genomic.info=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Validate.Genomic.Info_+3A_genomic.info">genomic.info</code></td>
<td>

<p>A vector with a chromosome name, start and end position on the chromosome
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG38.Human.CytoBandIdeogram)

RCircos.Set.Core.Components(UCSC.HG38.Human.CytoBandIdeogram, 
    chr.exclude=NULL, tracks.inside=10, tracks.outside=0 )
RCircos.Validate.Genomic.Info(c("chr1", 10000, 500000))

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Validate.Plot.Parameters'>
Validate Plot Parameters
</h2><span id='topic+RCircos.Validate.Plot.Parameters'></span>

<h3>Description</h3>

<p>Validate values of plot parameters before reset them in RCircos Environment. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Validate.Plot.Parameters(parameters=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Validate.Plot.Parameters_+3A_parameters">parameters</code></td>
<td>

<p>A list returned from RCircos.Get.Plot.Parameters().
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Validate.Track.Positions'>
Validate Track Positions
</h2><span id='topic+RCircos.Validate.Track.Positions'></span>

<h3>Description</h3>

<p>Validate if user defined plot locations (inside.pos and outside.pos) are inside of plot area and do not overlap with chromosome ideogram. RCircos core components must be initialized first. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Validate.Track.Positions(inside.pos=0, outside.pos=0, 
	erase.area=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Validate.Track.Positions_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Validate.Track.Positions_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Validate.Track.Positions_+3A_erase.area">erase.area</code></td>
<td>

<p>Logic, if the track is going to be erased.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector with length of 2 for the real plot position of a customized track.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Vertical.Line.Plot'>
Draw Vertical Lines on One Data Track
</h2><span id='topic+RCircos.Vertical.Line.Plot'></span>

<h3>Description</h3>

<p>Plot vertical lines on a data track for each genomic position  without track 
outlines and sub-track lines. RCircos core components and graphic devices must 
be initialized first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Vertical.Line.Plot(line.data=NULL, track.num=NULL, 
    side=c("in", "out"), line.width=1, inside.pos=NULL,
    outside.pos=NULL, genomic.columns=3, is.sorted=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_line.data">line.data</code></td>
<td>

<p>A data frame with leading columns as chromosome names, start position and/or end
position of each data point followed by data values for one or more samples.
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, representing the ordinal number of the plot track where 
the lines will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_side">side</code></td>
<td>

<p>Character vector, either &quot;in&quot; or &quot;out&quot;, representing the position related to 
chromosome ideogram.
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_line.width">line.width</code></td>
<td>

<p>Non-negative integer, width of vertical lines
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of 
the track .
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of 
the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_genomic.columns">genomic.columns</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position in each row. 
Must be either 3 or 2.
</p>
</td></tr>
<tr><td><code id="RCircos.Vertical.Line.Plot_+3A_is.sorted">is.sorted</code></td>
<td>

<p>Logic, whether the data is sorted by chromosome names and start positions.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram);
data(RCircos.Line.Data);

RCircos.Set.Core.Components(
    cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
    chr.exclude=c("chrX", "chrY"), 
    tracks.inside=10, tracks.outside=5) 
RCircos.Set.Plot.Area();
RCircos.Chromosome.Ideogram.Plot()

RCircos.Vertical.Line.Plot(RCircos.Line.Data, track.num=1, side="in")
RCircos.Vertical.Line.Plot(RCircos.Line.Data, side="in", 
    inside.pos=1.5, outside.pos=1.75)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Workflow'>
A Simple RCircos Plot Workflow
</h2><span id='topic+RCircos.Workflow'></span>

<h3>Description</h3>

<p>Print out on screen to list procedures for a basic RCircos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Workflow()
</code></pre>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
RCircos.Workflow()

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Zoom.Area.Outline'>
Draw Outline for Zoom Plot Area
</h2><span id='topic+RCircos.Zoom.Area.Outline'></span>

<h3>Description</h3>

<p>Draw outline and  subtrack lines for zoomed-in area of a track in outside of chromosome ideogram. RCircos core components must be initialized first. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Zoom.Area.Outline(zoom.pos=NULL, inside.pos=NULL,  
    outside.pos=NULL, num.layers=5, fill.col="white")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Zoom.Area.Outline_+3A_zoom.pos">zoom.pos</code></td>
<td>

<p>Non-negative numeric vector, the index of RCircos plot position.
</p>
</td></tr>
<tr><td><code id="RCircos.Zoom.Area.Outline_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Zoom.Area.Outline_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.Zoom.Area.Outline_+3A_num.layers">num.layers</code></td>
<td>

<p>Non-negative integer, number of sub-tracks in a data track.
</p>
</td></tr>
<tr><td><code id="RCircos.Zoom.Area.Outline_+3A_fill.col">fill.col</code></td>
<td>

<p>Character vector for color name to fill the polygon.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram)

RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=5) 
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

data(RCircos.Heatmap.Data)
zoom.data &lt;- RCircos.Get.Zoom.Data(RCircos.Heatmap.Data, 
			name.col=4, genomic.columns=3,
			target.gene="SP5", neighbor.genes=5)
zoom.range &lt;- RCircos.Get.Zoom.Range(zoom.data, 3)
zoom.pos &lt;- RCircos.Set.Zoom.Plot.Positions(zoom.range, 
        total.genes=11, area.length=0.25, gene.width=NULL)

RCircos.Zoom.Area.Outline(zoom.pos=NULL, inside.pos=2.5,
	outside.pos=3, num.layers=5, fill.col="white")

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Zoom.Paired.Plot.Positions'>
Calculate Zoomed Index of Circos Plot Positions for Genomic Intervals
</h2><span id='topic+RCircos.Zoom.Paired.Plot.Positions'></span>

<h3>Description</h3>

<p>Scale up the index of Circos plot positions of a small range for zoom-in plot with data that require two positions to plot e.g., tiles and parallele lines. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Zoom.Paired.Plot.Positions(zoom.data=NULL, zoom.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Zoom.Paired.Plot.Positions_+3A_zoom.data">zoom.data</code></td>
<td>

<p>Data frame, data to be plotted onto zoom area.
</p>
</td></tr>
<tr><td><code id="RCircos.Zoom.Paired.Plot.Positions_+3A_zoom.pos">zoom.pos</code></td>
<td>

<p>Non-negative integer, index of RCircos base plot positions.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two columns integer matrix representing the index of RCircos plot positions.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.Zoom.Single.Plot.Positions'>
Calculate Zoomed Index of Circos Plot Positions for Single Point Genomic Data 
</h2><span id='topic+RCircos.Zoom.Single.Plot.Positions'></span>

<h3>Description</h3>

<p>Scale up the index of Circos plot positions of a small range for zoom-in plot with data that require only one position to plot e.g., scatter plot. This function is mainly for internal use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Zoom.Single.Plot.Positions(zoom.data=NULL, zoom.pos=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.Zoom.Single.Plot.Positions_+3A_zoom.data">zoom.data</code></td>
<td>

<p>Data frame, data to be plotted onto zoom area.
</p>
</td></tr>
<tr><td><code id="RCircos.Zoom.Single.Plot.Positions_+3A_zoom.pos">zoom.pos</code></td>
<td>

<p>Non-negative integer, index of RCircos base plot positions.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer representing the index of RCircos plot positions.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>

<hr>
<h2 id='RCircos.ZoomIn.Chromosome'>
Zoom in One or Partial Chromosome Ideogram
</h2><span id='topic+RCircos.ZoomIn.Chromosome'></span>

<h3>Description</h3>

<p>Increase plot length for one or partial chromosome in order to plot more data points. This works on original chromosome ideogram data and should be done before set RCircos core components. Also, this function only work with one region each time. If there are more than one regions to be zoomed in, repeat this function for each region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.ZoomIn.Chromosome(ideogram=NULL, chromosome=NULL, from=NULL, 
    to=NULL, zoom.in=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.ZoomIn.Chromosome_+3A_ideogram">ideogram</code></td>
<td>

<p>Data frame, chromosome ideogram data with chromosome names, start position, end position, band name and stains for each band or each chromosome. 
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Chromosome_+3A_chromosome">chromosome</code></td>
<td>

<p>Character vector, name of chromosome to be zoomed in.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Chromosome_+3A_from">from</code></td>
<td>

<p>Non-negative integer, genomic coordinates of start position of chromosome region to be zoomed in.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Chromosome_+3A_to">to</code></td>
<td>

<p>Non-negative integer, genomic coordinates of end position of chromosome region to be zoomed in.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Chromosome_+3A_zoom.in">zoom.in</code></td>
<td>

<p>Non-negative number, fold to zoom in.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame, ideogram with new zoomed in region.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
library(RCircos) 
data(UCSC.HG19.Human.CytoBandIdeogram) 
zoomedIdeo &lt;- RCircos.ZoomIn.Chromosome( 
    ideogram=UCSC.HG19.Human.CytoBandIdeogram, 
    chromosome="chr17", from=10000, to=20000, 
    zoom.in=1000)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.ZoomIn.Plot'>
Zoomed-in Plot of Small Genomic Region
</h2><span id='topic+RCircos.Plot.Zoomed.Area'></span><span id='topic+RCircos.Plot.Zoomed.Gene.Connectors'></span><span id='topic+RCircos.Plot.Zoomed.Heatmap'></span><span id='topic+RCircos.Plot.Zoomed.Histogram'></span><span id='topic+RCircos.Plot.Zoomed.Ideogram.Ticks'></span><span id='topic+RCircos.Plot.Zoomed.Continue.Lines'></span><span id='topic+RCircos.Plot.Zoomed.Parallel.Lines'></span><span id='topic+RCircos.Plot.Zoomed.Vertical.Lines'></span><span id='topic+RCircos.Plot.Zoomed.Scatters'></span><span id='topic+RCircos.Plot.Zoomed.Tiles'></span><span id='topic+RCircos.Plot.Zoomed.Polygons'></span><span id='topic+RCircos.Label.Zoom.Region'></span><span id='topic+RCircos.Mark.Zoom.Area'></span>

<h3>Description</h3>

<p>Zoom-in a small area on outide of chromosome ideogram to show the details for: heatmap, histogram, gene connector, gene labels, continue lines, vertical lines, parallel lines, scatters(points), areas, and tiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.Plot.Zoomed.Area(zoom.data=NULL, 
    plot.type="mountain", data.col=NULL, track.num=NULL, 
    zoom.pos=NULL, min.value=NULL, max.value=NULL, 
    area.color="gray", border.col="black", inside.pos=NULL, 
    outside.pos=NULL, outline=TRUE)
 
RCircos.Plot.Zoomed.Gene.Connectors(zoom.data=NULL,  
    track.num=NULL, zoom.pos=NULL, line.width=1,  
    inside.pos=NULL, outside.pos=NULL)  

RCircos.Plot.Zoomed.Heatmap(zoom.data=NULL, data.col=NULL,  
    track.num=NULL, zoom.pos=NULL, min.value=NULL,  
    max.value=NULL, inside.pos=NULL, outside.pos=NULL)  

RCircos.Plot.Zoomed.Histogram(zoom.data=NULL, data.col=NULL,  
    track.num=NULL, zoom.pos=NULL, min.value=NULL,  
    max.value=NULL, inside.pos=NULL, outside.pos=NULL, 
    outline=TRUE)  

RCircos.Plot.Zoomed.Ideogram.Ticks(zoom.info=NULL,  
    track.num=NULL, zoom.pos=NULL, tick.interval=5,  
    inside.pos=NULL, outside.pos=NULL)  

RCircos.Plot.Zoomed.Continue.Lines(zoom.data=NULL,  
    data.col=NULL, track.num=NULL, zoom.pos=NULL,  
    line.width=1, min.value=NULL, max.value=NULL,  
    inside.pos=NULL, outside.pos=NULL, outline=TRUE)  

RCircos.Plot.Zoomed.Parallel.Lines(zoom.data=NULL, 
    track.num=NULL, zoom.pos=NULL, genomic.cols=3, 
    line.width=NULL,inside.pos=NULL, outside.pos=NULL, 
    outline=FALSE)

RCircos.Plot.Zoomed.Vertical.Lines(zoom.data=NULL,  
    track.num=NULL, zoom.pos=NULL, line.width=1,  
    inside.pos=NULL, outside.pos=NULL, outline=FALSE)  

RCircos.Plot.Zoomed.Scatters(zoom.data=NULL, data.col=NULL,  
    track.num=NULL, zoom.pos=NULL, min.value=NULL,  
    max.value=NULL, point.type=16, by.fold=0,  
    with.size=TRUE, with.height=FALSE, point.scale=1,  
    inside.pos=NULL, outside.pos=NULL, outline=TRUE)  

RCircos.Plot.Zoomed.Tiles(zoom.data=NULL, track.num=NULL,  
    zoom.pos=NULL, genomic.cols=3, layers=5, border.col=NULL,  
    inside.pos=NULL, outside.pos=NULL, outline=TRUE)  

RCircos.Plot.Zoomed.Polygons(zoom.data=NULL, data.col=4,   
    track.num=NULL, zoom.pos=NULL, genomic.cols=3,  
    min.value=NULL, max.value=NULL, border.col=NULL,  
    inside.pos=NULL, outside.pos=NULL, outline=TRUE) 

RCircos.Label.Zoom.Region(zoom.data=NULL, name.col=NULL,  
    track.num=NULL, zoom.pos=NULL, text.size=0.75,  
    inside.pos=NULL, outside.pos=NULL)  

RCircos.Mark.Zoom.Area(zoom.range=NULL, track.num=1,  
    zoom.pos=NULL, fill.color="yellow",  
    inside.pos=NULL, outside.pos=NULL)  
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_zoom.data">zoom.data</code></td>
<td>

<p>A data frame containing genomic positions, gene names, and plot values for zoom-in genes/rows.
</p>
</td></tr>  <tr><td><code id="RCircos.ZoomIn.Plot_+3A_zoom.range">zoom.range</code></td>
<td>

<p>Character vecter, chromosome name, start and end positions of a genomic interval to be zoomed in.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_data.col">data.col</code></td>
<td>

<p>Non-negative integer/vector, which column(s) is plot data values.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_track.num">track.num</code></td>
<td>

<p>Non-negative integer, which track will be plotted.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_zoom.pos">zoom.pos</code></td>
<td>

<p>Non-negative numeric vector, the index of RCircos plot position.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_min.value">min.value</code></td>
<td>

<p>Numeric, minimum value for heatmap scale.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_max.value">max.value</code></td>
<td>

<p>Numeric, maximum value for heatmap scale.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_inside.pos">inside.pos</code></td>
<td>

<p>Non-negative numeric, inside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_outside.pos">outside.pos</code></td>
<td>

<p>Non-negative numeric, outside position (relative to the centre of plot area) of the track.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_name.col">name.col</code></td>
<td>

<p>Non-negative integer, which column is for labels.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_genomic.cols">genomic.cols</code></td>
<td>

<p>Non-negative integer, total number of columns for genomic position information (chromosome, start and/or end position).
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_fill.color">fill.color</code></td>
<td>

<p>Character vector for name of a color to fill the polygon.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_border.col">border.col</code></td>
<td>

<p>Character vector for name of a color to draw the polygon border.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_area.color">area.color</code></td>
<td>

<p>Character vector for name of a color to fill the area.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_text.size">text.size</code></td>
<td>

<p>Non-negative numeric, character size for text plot.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_line.width">line.width</code></td>
<td>

<p>Non-negative numeric, width of lines.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_point.type">point.type</code></td>
<td>

<p>Non-negative integer, type of character for point plot (Default 19).
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_plot.type">plot.type</code></td>
<td>

<p>Character vector, either &quot;mountain&quot;, &quot;curtain&quot;, or &quot;ban&quot;, types of area plot.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_by.fold">by.fold</code></td>
<td>

<p>Zero or a positive float number, if it is greater than zero, any data point with a value greater than or equal to the by.fold will be plot as red color and data points with values less or equal to the negative by.fold will be plot as blue colr. Otherwise, black color is used.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_with.size">with.size</code></td>
<td>

<p>Logic. If TRUE, the point size will match to the data value in scatter/point plot.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_with.height">with.height</code></td>
<td>

<p>Logic. If TURE, point location will match to the data value in scatter/point plot.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_point.scale">point.scale</code></td>
<td>

<p>Non-negative numeric, more scale to add for scatter/point size in scatter/point plot.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_outline">outline</code></td>
<td>

<p>Logic. If TURE, outline will be drawn for zoomed area.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_zoom.info">zoom.info</code></td>
<td>

<p>Character vector contains chromosome name, start and end position.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_tick.interval">tick.interval</code></td>
<td>

<p>Non-negative integer, length of genomic interval in million base pairs.
</p>
</td></tr>
<tr><td><code id="RCircos.ZoomIn.Plot_+3A_layers">layers</code></td>
<td>

<p>Non-negative integer, number of subtrack in the data track.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>RCircos.Plot.Zoomed.Area() paints partial area inside the zoomed region (an irregular polygon ship). Plot types include &quot;mountain&quot;, &quot;curtain&quot; (reversed mountain), and &quot;band&quot;.
</p>
<p>RCircos.Plot.Zoomed.Gene.Connectors() plot zoomed-in connectors for small number of genes(default 11).
</p>
<p>RCircos.Plot.Zoomed.Heatmap() plot zoomed-in heatmap for small number of genes (default 11).
</p>
<p>RCircos.Plot.Zoomed.Histogram() plot zoomed-in histogram for small number of genes (default 11).
</p>
<p>RCircos.Plot.Zoomed.Ideogram.Ticks() plot chromosome ticks for a small genomic intervals.
</p>
<p>RCircos.Plot.Zoomed.Continue.Lines() plot zoomed-in continue lines between neighbors of small number genes (default 11).
</p>
<p>RCircos.Plot.Zoomed.Parallel.Lines() plot zoomed-in parallel lines for small number of genomic intevals (default 11).
</p>
<p>RCircos.Plot.Zoomed.Vertical.Lines() plot zoomed-in vertical lines from bottom to top of a track for small number of genomic intevals (default 11).
</p>
<p>RCircos.Plot.Zoomed.Scatters() plot zoomed-in scatters for small number of genes(default 11).
</p>
<p>RCircos.Plot.Zoomed.Tiles() plot zoomed-in tiles for small number of genes(default 11).
</p>
<p>RCircos.Plot.Zoomed.Polygons() plot zoomed-in polygons in a samll genomic region.
</p>
<p>RCircos.Label.Zoom.Region() plot gene names for small number of genes(default 11).
</p>
<p>RCircos.Mark.Zoom.Area() highligh the area between zoomed-in plot area and chromosome ideogram to mark the original genomic interval.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos);
data(UCSC.HG19.Human.CytoBandIdeogram)

cyto &lt;- UCSC.HG19.Human.CytoBandIdeogram
RCircos.Set.Core.Components(cyto, NULL, 10, 5)

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

data(RCircos.Heatmap.Data)
min.value &lt;- min(as.matrix(RCircos.Heatmap.Data[,5:10]))
max.value &lt;- max(as.matrix(RCircos.Heatmap.Data[,5:10]))

zoom.data &lt;- RCircos.Get.Zoom.Data(RCircos.Heatmap.Data,
        name.col=4, genomic.columns=3,
        target.gene="SP5", neighbor.genes=5)
zoom.range &lt;- RCircos.Get.Zoom.Range(zoom.data, 3)
zoom.pos &lt;- RCircos.Set.Zoom.Plot.Positions(zoom.range,
        total.genes=11, area.length=0.25, gene.width=NULL)

RCircos.Plot.Zoomed.Heatmap(zoom.data, data.col=5,
        track.num=2, zoom.pos=zoom.pos,
        min.value=min.value, max.value=max.value,
        inside.pos=NULL, outside.pos=NULL)

RCircos.Mark.Zoom.Area(zoom.pos, zoom.data, color="red",
        track.num=2, inside.pos=NULL, outside.pos=NULL)

RCircos.Label.Zoom.Region(zoom.data, name.col=4,
        track.num=3, zoom.pos=zoom.pos, text.size=0.75,
        inside.pos=NULL, outside.pos=NULL)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.ZoomOut.Chromosome'>
Zoom out Chromosome Ideogram
</h2><span id='topic+RCircos.ZoomOut.Chromosome'></span>

<h3>Description</h3>

<p>Zoom out chromosome ideogram to leave a small room between the first and the last chromosome (at 12 O'clock) so that track names could be added. This function works on the RCircos chromosome ideogram component so the core components must be set first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCircos.ZoomOut.Chromosome(zoom.out.ratio=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCircos.ZoomOut.Chromosome_+3A_zoom.out.ratio">zoom.out.ratio</code></td>
<td>

<p>A float number between 0 and 1, the ratio of target chromosome ideogram to be zoomed out.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None. New ideogram object is set to RCircos environment.
</p>


<h3>Author(s)</h3>

<p>Henry Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(RCircos)
data(UCSC.HG19.Human.CytoBandIdeogram)
RCircos.Set.Core.Components(
	cyto.info=UCSC.HG19.Human.CytoBandIdeogram, 
	chr.exclude=c("chrX", "chrY"), 
	tracks.inside=10, tracks.outside=0) RCircos.ZoomOut.Chromosome(zoom.out.ratio=0.95)

## End(Not run)
</code></pre>

<hr>
<h2 id='UCSC.Baylor.3.4.Rat.cytoBandIdeogram'>
Cytoband Information for Rat Chromosome Ideogram 
</h2><span id='topic+UCSC.Baylor.3.4.Rat.cytoBandIdeogram'></span>

<h3>Description</h3>

<p>A data frame containing chromosome name, start and end position, band name, and intensity of Giemsa stains for each cytoband.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)</code></pre>


<h3>Format</h3>

<p>A data frame with 246 observations on the following 5 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code></p>
</dd>
<dt><code>ChromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ChromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Band</code></dt><dd><p>a factor with levels <code>p11</code> <code>p12</code> <code>p13</code> <code>p14</code> <code>p15</code> <code>p16</code> <code>p21</code> <code>p22</code> <code>q11</code> <code>q12</code> <code>q12.1</code> <code>q12.2</code> <code>q12.3</code> <code>q12.4</code> <code>q12.5</code> <code>q13</code> <code>q14</code> <code>q15</code> <code>q16</code> <code>q21</code> <code>q22</code> <code>q23</code> <code>q24</code> <code>q25</code> <code>q26</code> <code>q27</code> <code>q31</code> <code>q32</code> <code>q32.1</code> <code>q32.2</code> <code>q32.3</code> <code>q33</code> <code>q34</code> <code>q35</code> <code>q36</code> <code>q37</code> <code>q38</code> <code>q41</code> <code>q42</code> <code>q43</code> <code>q44</code> <code>q45</code> <code>q51</code> <code>q52</code> <code>q53</code> <code>q54</code> <code>q55</code></p>
</dd>
<dt><code>Stain</code></dt><dd><p>a factor with levels <code>gneg</code> <code>gpos</code> <code>gvar</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>http://genome.ucsc.edu 
</p>

<hr>
<h2 id='UCSC.HG19.Human.CytoBandIdeogram'>
Cytoband Information for Human Chromosome Ideogram 
</h2><span id='topic+UCSC.HG19.Human.CytoBandIdeogram'></span>

<h3>Description</h3>

<p>A data frame containing chromosome name, start and end position, band name, and intensity of Giemsa stains for each cytoband. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(UCSC.HG19.Human.CytoBandIdeogram)</code></pre>


<h3>Format</h3>

<p>A data frame with 862 observations on the following 5 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>ChromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ChromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Band</code></dt><dd><p>a factor with levels <code>p11</code> <code>p11.1</code> <code>p11.11</code> <code>p11.12</code> <code>p11.2</code> <code>p11.21</code> <code>p11.22</code> <code>p11.23</code> <code>p11.3</code> <code>p11.31</code> <code>p11.32</code> <code>p11.4</code> <code>p12</code> <code>p12.1</code> <code>p12.2</code> <code>p12.3</code> <code>p12.31</code> <code>p12.32</code> <code>p12.33</code> <code>p13</code> <code>p13.1</code> <code>p13.11</code> <code>p13.12</code> <code>p13.13</code> <code>p13.2</code> <code>p13.3</code> <code>p13.31</code> <code>p13.32</code> <code>p13.33</code> <code>p14</code> <code>p14.1</code> <code>p14.2</code> <code>p14.3</code> <code>p15</code> <code>p15.1</code> <code>p15.2</code> <code>p15.3</code> <code>p15.31</code> <code>p15.32</code> <code>p15.33</code> <code>p15.4</code> <code>p15.5</code> <code>p16.1</code> <code>p16.2</code> <code>p16.3</code> <code>p21</code> <code>p21.1</code> <code>p21.2</code> <code>p21.3</code> <code>p21.31</code> <code>p21.32</code> <code>p21.33</code> <code>p22</code> <code>p22.1</code> <code>p22.11</code> <code>p22.12</code> <code>p22.13</code> <code>p22.2</code> <code>p22.3</code> <code>p22.31</code> <code>p22.32</code> <code>p22.33</code> <code>p23</code> <code>p23.1</code> <code>p23.2</code> <code>p23.3</code> <code>p24.1</code> <code>p24.2</code> <code>p24.3</code> <code>p25.1</code> <code>p25.2</code> <code>p25.3</code> <code>p26.1</code> <code>p26.2</code> <code>p26.3</code> <code>p31.1</code> <code>p31.2</code> <code>p31.3</code> <code>p32.1</code> <code>p32.2</code> <code>p32.3</code> <code>p33</code> <code>p34.1</code> <code>p34.2</code> <code>p34.3</code> <code>p35.1</code> <code>p35.2</code> <code>p35.3</code> <code>p36.11</code> <code>p36.12</code> <code>p36.13</code> <code>p36.21</code> <code>p36.22</code> <code>p36.23</code> <code>p36.31</code> <code>p36.32</code> <code>p36.33</code> <code>q11</code> <code>q11.1</code> <code>q11.2</code> <code>q11.21</code> <code>q11.22</code> <code>q11.221</code> <code>q11.222</code> <code>q11.223</code> <code>q11.23</code> <code>q12</code> <code>q12.1</code> <code>q12.11</code> <code>q12.12</code> <code>q12.13</code> <code>q12.2</code> <code>q12.3</code> <code>q13</code> <code>q13.1</code> <code>q13.11</code> <code>q13.12</code> <code>q13.13</code> <code>q13.2</code> <code>q13.3</code> <code>q13.31</code> <code>q13.32</code> <code>q13.33</code> <code>q13.4</code> <code>q13.41</code> <code>q13.42</code> <code>q13.43</code> <code>q13.5</code> <code>q14</code> <code>q14.1</code> <code>q14.11</code> <code>q14.12</code> <code>q14.13</code> <code>q14.2</code> <code>q14.3</code> <code>q15</code> <code>q15.1</code> <code>q15.2</code> <code>q15.3</code> <code>q16.1</code> <code>q16.2</code> <code>q16.3</code> <code>q21</code> <code>q21.1</code> <code>q21.11</code> <code>q21.12</code> <code>q21.13</code> <code>q21.2</code> <code>q21.21</code> <code>q21.22</code> <code>q21.23</code> <code>q21.3</code> <code>q21.31</code> <code>q21.32</code> <code>q21.33</code> <code>q22</code> <code>q22.1</code> <code>q22.11</code> <code>q22.12</code> <code>q22.13</code> <code>q22.2</code> <code>q22.3</code> <code>q22.31</code> <code>q22.32</code> <code>q22.33</code> <code>q23</code> <code>q23.1</code> <code>q23.2</code> <code>q23.3</code> <code>q23.31</code> <code>q23.32</code> <code>q23.33</code> <code>q24</code> <code>q24.1</code> <code>q24.11</code> <code>q24.12</code> <code>q24.13</code> <code>q24.2</code> <code>q24.21</code> <code>q24.22</code> <code>q24.23</code> <code>q24.3</code> <code>q24.31</code> <code>q24.32</code> <code>q24.33</code> <code>q25</code> <code>q25.1</code> <code>q25.2</code> <code>q25.3</code> <code>q25.31</code> <code>q25.32</code> <code>q25.33</code> <code>q26</code> <code>q26.1</code> <code>q26.11</code> <code>q26.12</code> <code>q26.13</code> <code>q26.2</code> <code>q26.3</code> <code>q26.31</code> <code>q26.32</code> <code>q26.33</code> <code>q27</code> <code>q27.1</code> <code>q27.2</code> <code>q27.3</code> <code>q28</code> <code>q28.1</code> <code>q28.2</code> <code>q28.3</code> <code>q29</code> <code>q31.1</code> <code>q31.2</code> <code>q31.21</code> <code>q31.22</code> <code>q31.23</code> <code>q31.3</code> <code>q31.31</code> <code>q31.32</code> <code>q31.33</code> <code>q32</code> <code>q32.1</code> <code>q32.11</code> <code>q32.12</code> <code>q32.13</code> <code>q32.2</code> <code>q32.3</code> <code>q32.31</code> <code>q32.32</code> <code>q32.33</code> <code>q33</code> <code>q33.1</code> <code>q33.2</code> <code>q33.3</code> <code>q34</code> <code>q34.1</code> <code>q34.11</code> <code>q34.12</code> <code>q34.13</code> <code>q34.2</code> <code>q34.3</code> <code>q35</code> <code>q35.1</code> <code>q35.2</code> <code>q35.3</code> <code>q36.1</code> <code>q36.2</code> <code>q36.3</code> <code>q37.1</code> <code>q37.2</code> <code>q37.3</code> <code>q41</code> <code>q42.11</code> <code>q42.12</code> <code>q42.13</code> <code>q42.2</code> <code>q42.3</code> <code>q43</code> <code>q44</code></p>
</dd>
<dt><code>Stain</code></dt><dd><p>a factor with levels <code>acen</code> <code>gneg</code> <code>gpos100</code> <code>gpos25</code> <code>gpos50</code> <code>gpos75</code> <code>gvar</code> <code>stalk</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>http://genome.ucsc.edu 
</p>

<hr>
<h2 id='UCSC.HG38.Human.CytoBandIdeogram'>
Human Chromosome Ideogram Version 38
</h2><span id='topic+UCSC.HG38.Human.CytoBandIdeogram'></span>

<h3>Description</h3>

<p>Human chromosome ideogram v38 in the data frame format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("UCSC.HG38.Human.CytoBandIdeogram")</code></pre>


<h3>Format</h3>

<p>A data frame with 862 observations on the following 5 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr20</code> <code>chr21</code> <code>chr22</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>chromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Name</code></dt><dd><p>a factor with levels <code>p11</code> <code>p11.1</code> <code>p11.11</code> <code>p11.12</code> <code>p11.2</code> <code>p11.21</code> <code>p11.22</code> <code>p11.23</code> <code>p11.3</code> <code>p11.31</code> <code>p11.32</code> <code>p11.4</code> <code>p12</code> <code>p12.1</code> <code>p12.2</code> <code>p12.3</code> <code>p12.31</code> <code>p12.32</code> <code>p12.33</code> <code>p13</code> <code>p13.1</code> <code>p13.11</code> <code>p13.12</code> <code>p13.13</code> <code>p13.2</code> <code>p13.3</code> <code>p13.31</code> <code>p13.32</code> <code>p13.33</code> <code>p14</code> <code>p14.1</code> <code>p14.2</code> <code>p14.3</code> <code>p15</code> <code>p15.1</code> <code>p15.2</code> <code>p15.3</code> <code>p15.31</code> <code>p15.32</code> <code>p15.33</code> <code>p15.4</code> <code>p15.5</code> <code>p16.1</code> <code>p16.2</code> <code>p16.3</code> <code>p21</code> <code>p21.1</code> <code>p21.2</code> <code>p21.3</code> <code>p21.31</code> <code>p21.32</code> <code>p21.33</code> <code>p22</code> <code>p22.1</code> <code>p22.11</code> <code>p22.12</code> <code>p22.13</code> <code>p22.2</code> <code>p22.3</code> <code>p22.31</code> <code>p22.32</code> <code>p22.33</code> <code>p23</code> <code>p23.1</code> <code>p23.2</code> <code>p23.3</code> <code>p24.1</code> <code>p24.2</code> <code>p24.3</code> <code>p25.1</code> <code>p25.2</code> <code>p25.3</code> <code>p26.1</code> <code>p26.2</code> <code>p26.3</code> <code>p31.1</code> <code>p31.2</code> <code>p31.3</code> <code>p32.1</code> <code>p32.2</code> <code>p32.3</code> <code>p33</code> <code>p34.1</code> <code>p34.2</code> <code>p34.3</code> <code>p35.1</code> <code>p35.2</code> <code>p35.3</code> <code>p36.11</code> <code>p36.12</code> <code>p36.13</code> <code>p36.21</code> <code>p36.22</code> <code>p36.23</code> <code>p36.31</code> <code>p36.32</code> <code>p36.33</code> <code>q11</code> <code>q11.1</code> <code>q11.2</code> <code>q11.21</code> <code>q11.22</code> <code>q11.221</code> <code>q11.222</code> <code>q11.223</code> <code>q11.23</code> <code>q12</code> <code>q12.1</code> <code>q12.11</code> <code>q12.12</code> <code>q12.13</code> <code>q12.2</code> <code>q12.3</code> <code>q13</code> <code>q13.1</code> <code>q13.11</code> <code>q13.12</code> <code>q13.13</code> <code>q13.2</code> <code>q13.3</code> <code>q13.31</code> <code>q13.32</code> <code>q13.33</code> <code>q13.4</code> <code>q13.41</code> <code>q13.42</code> <code>q13.43</code> <code>q13.5</code> <code>q14</code> <code>q14.1</code> <code>q14.11</code> <code>q14.12</code> <code>q14.13</code> <code>q14.2</code> <code>q14.3</code> <code>q15</code> <code>q15.1</code> <code>q15.2</code> <code>q15.3</code> <code>q16.1</code> <code>q16.2</code> <code>q16.3</code> <code>q21</code> <code>q21.1</code> <code>q21.11</code> <code>q21.12</code> <code>q21.13</code> <code>q21.2</code> <code>q21.21</code> <code>q21.22</code> <code>q21.23</code> <code>q21.3</code> <code>q21.31</code> <code>q21.32</code> <code>q21.33</code> <code>q22</code> <code>q22.1</code> <code>q22.11</code> <code>q22.12</code> <code>q22.13</code> <code>q22.2</code> <code>q22.3</code> <code>q22.31</code> <code>q22.32</code> <code>q22.33</code> <code>q23</code> <code>q23.1</code> <code>q23.2</code> <code>q23.3</code> <code>q23.31</code> <code>q23.32</code> <code>q23.33</code> <code>q24</code> <code>q24.1</code> <code>q24.11</code> <code>q24.12</code> <code>q24.13</code> <code>q24.2</code> <code>q24.21</code> <code>q24.22</code> <code>q24.23</code> <code>q24.3</code> <code>q24.31</code> <code>q24.32</code> <code>q24.33</code> <code>q25</code> <code>q25.1</code> <code>q25.2</code> <code>q25.3</code> <code>q25.31</code> <code>q25.32</code> <code>q25.33</code> <code>q26</code> <code>q26.1</code> <code>q26.11</code> <code>q26.12</code> <code>q26.13</code> <code>q26.2</code> <code>q26.3</code> <code>q26.31</code> <code>q26.32</code> <code>q26.33</code> <code>q27</code> <code>q27.1</code> <code>q27.2</code> <code>q27.3</code> <code>q28</code> <code>q28.1</code> <code>q28.2</code> <code>q28.3</code> <code>q29</code> <code>q31.1</code> <code>q31.2</code> <code>q31.21</code> <code>q31.22</code> <code>q31.23</code> <code>q31.3</code> <code>q31.31</code> <code>q31.32</code> <code>q31.33</code> <code>q32</code> <code>q32.1</code> <code>q32.11</code> <code>q32.12</code> <code>q32.13</code> <code>q32.2</code> <code>q32.3</code> <code>q32.31</code> <code>q32.32</code> <code>q32.33</code> <code>q33</code> <code>q33.1</code> <code>q33.2</code> <code>q33.3</code> <code>q34</code> <code>q34.1</code> <code>q34.11</code> <code>q34.12</code> <code>q34.13</code> <code>q34.2</code> <code>q34.3</code> <code>q35</code> <code>q35.1</code> <code>q35.2</code> <code>q35.3</code> <code>q36.1</code> <code>q36.2</code> <code>q36.3</code> <code>q37.1</code> <code>q37.2</code> <code>q37.3</code> <code>q41</code> <code>q42.11</code> <code>q42.12</code> <code>q42.13</code> <code>q42.2</code> <code>q42.3</code> <code>q43</code> <code>q44</code></p>
</dd>
<dt><code>Stain</code></dt><dd><p>a factor with levels <code>acen</code> <code>gneg</code> <code>gpos100</code> <code>gpos25</code> <code>gpos50</code> <code>gpos75</code> <code>gvar</code> <code>stalk</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>http://genome.ucsc.edu
</p>

<hr>
<h2 id='UCSC.Mouse.GRCm38.CytoBandIdeogram'>
Cytoband Inforamtion for Mouse Chromosome Ideogram 
</h2><span id='topic+UCSC.Mouse.GRCm38.CytoBandIdeogram'></span>

<h3>Description</h3>

<p>A data frame containing chromosome name, start and end position, band name, and intensity of Giemsa stains for each cytoband. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(UCSC.Mouse.GRCm38.CytoBandIdeogram)</code></pre>


<h3>Format</h3>

<p>A data frame with 403 observations on the following 5 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a factor with levels <code>chr1</code> <code>chr10</code> <code>chr11</code> <code>chr12</code> <code>chr13</code> <code>chr14</code> <code>chr15</code> <code>chr16</code> <code>chr17</code> <code>chr18</code> <code>chr19</code> <code>chr2</code> <code>chr3</code> <code>chr4</code> <code>chr5</code> <code>chr6</code> <code>chr7</code> <code>chr8</code> <code>chr9</code> <code>chrX</code> <code>chrY</code></p>
</dd>
<dt><code>ChromStart</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ChromEnd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Band</code></dt><dd><p>a factor with levels <code>qA</code> <code>qA1</code> <code>qA1.1</code> <code>qA1.2</code> <code>qA1.3</code> <code>qA2</code> <code>qA3</code> <code>qA3.1</code> <code>qA3.2</code> <code>qA3.3</code> <code>qA4</code> <code>qA5</code> <code>qA5.1</code> <code>qA5.2</code> <code>qA5.3</code> <code>qA6</code> <code>qA7.1</code> <code>qA7.2</code> <code>qA7.3</code> <code>qB</code> <code>qB1</code> <code>qB1.1</code> <code>qB1.2</code> <code>qB1.3</code> <code>qB2</code> <code>qB2.1</code> <code>qB2.2</code> <code>qB2.3</code> <code>qB3</code> <code>qB3.1</code> <code>qB3.2</code> <code>qB3.3</code> <code>qB4</code> <code>qB5</code> <code>qB5.1</code> <code>qB5.2</code> <code>qB5.3</code> <code>qC</code> <code>qC1</code> <code>qC1.1</code> <code>qC1.2</code> <code>qC1.3</code> <code>qC2</code> <code>qC3</code> <code>qC3.1</code> <code>qC3.2</code> <code>qC3.3</code> <code>qC4</code> <code>qC5</code> <code>qC6</code> <code>qC7</code> <code>qD</code> <code>qD1</code> <code>qD2</code> <code>qD2.1</code> <code>qD2.2</code> <code>qD2.3</code> <code>qD3</code> <code>qE</code> <code>qE1</code> <code>qE1.1</code> <code>qE1.2</code> <code>qE1.3</code> <code>qE2</code> <code>qE2.1</code> <code>qE2.2</code> <code>qE2.3</code> <code>qE3</code> <code>qE3.1</code> <code>qE3.2</code> <code>qE3.3</code> <code>qE4</code> <code>qE5</code> <code>qF</code> <code>qF1</code> <code>qF2</code> <code>qF2.1</code> <code>qF2.2</code> <code>qF2.3</code> <code>qF3</code> <code>qF4</code> <code>qF5</code> <code>qG1</code> <code>qG1.1</code> <code>qG1.2</code> <code>qG1.3</code> <code>qG2</code> <code>qG3</code> <code>qH1</code> <code>qH2</code> <code>qH2.1</code> <code>qH2.2</code> <code>qH2.3</code> <code>qH3</code> <code>qH4</code> <code>qH5</code> <code>qH6</code></p>
</dd>
<dt><code>Stain</code></dt><dd><p>a factor with levels <code>gneg</code> <code>gpos100</code> <code>gpos33</code> <code>gpos66</code> <code>gpos75</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>http://genome.ucsc.edu 
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
