<!DOCTYPE html><html lang="en"><head><title>Help for package RCNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RCNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#correct_gc_bias'><p>correct_gc_bias: Estimate and correct GC bias in coverage</p></a></li>
<li><a href='#create_RCNA_object'><p>RCNA_object constructor</p></a></li>
<li><a href='#estimate_feature_score'><p>estimate_feature_score: Estimate CNV score for each gene in the annotation file</p></a></li>
<li><a href='#estimate_nkr'><p>estimate_nkr: Estimate CNA &quot;normal&quot; karyotype ranges</p></a></li>
<li><a href='#example_obj'><p>Example RCNA_object</p></a></li>
<li><a href='#RCNA_analysis-class'><p>RCNA_analysis constructor</p></a></li>
<li><a href='#RCNA_object-class'><p>RCNA.object definition</p></a></li>
<li><a href='#run_RCNA'><p>run_RCNA: Perform RCNA copy number detection workflow</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Robust Copy Number Alteration Detection (RCNA)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Detects copy number alteration events in targeted exon sequencing 
    data for tumor samples without matched normal controls. The advantage of 
    this method is that it can be applied to smaller sequencing panels including 
    evaluations of exon, transcript, gene, or even user specified genetic
    regions of interest. Functions in the package include steps for GC-content
    correction, calculation of quantile based normal karyotype ranges,
    and calculation of feature score.  Cutoffs for "normal" quantile and score
    are user-adjustable.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>doParallel, methods, utils, modeest, data.table, foreach,
R.utils</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, usethis</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Collate:</td>
<td>'classes.r' 'estimate_feature_score.r' 'estimate_nkr.r'
'correct_gc_bias.r' 'RCNA.r' 'data.R'</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-02 13:13:50 UTC; mbradley</td>
</tr>
<tr>
<td>Author:</td>
<td>Matt Bradley [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matt Bradley &lt;mbradley@emmes.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-03 18:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='correct_gc_bias'>correct_gc_bias: Estimate and correct GC bias in coverage</h2><span id='topic+correct_gc_bias'></span><span id='topic+correct_gc_bias.default'></span><span id='topic+correct_gc_bias.RCNA_object'></span>

<h3>Description</h3>

<p>This generic function is used to run to calculate and correct GC-content-based coverage bias
</p>
<p>This function optionally estimates and then corrects the GC bias based on a GC-content factor file that is either generated or provided by the user using a sliding window approach. It creates a GC factor file and a corrected coverage file, both of which are placed in the output directory under '/gc'.
</p>
<p>This function optionally estimates and then corrects the GC bias based on a GC-content factor file that is either generated or provided by the user using a sliding window approach. It creates a GC factor file and a corrected coverage file, both of which are placed in the output directory under '/gc'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_gc_bias(obj, ...)

## Default S3 method:
correct_gc_bias(
  obj = NULL,
  df = NULL,
  sample.names = NULL,
  ano.file,
  out.dir = NULL,
  ncpus = 1,
  file.raw.coverage = NULL,
  file.corrected.coverage = NULL,
  file.gc.factor = NULL,
  win.size = 75,
  gc.step = 0.01,
  estimate_gc = TRUE,
  verbose = FALSE,
  ...
)

## S3 method for class 'RCNA_object'
correct_gc_bias(obj, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="correct_gc_bias_+3A_obj">obj</code></td>
<td>
<p>A RCNA_object type object - parameters will be pulled from the object instead, specifically from the 'gcParams' slot.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_...">...</code></td>
<td>
<p>Additional arguments (unused)</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_df">df</code></td>
<td>
<p>Path to the config file, or a 'data.frame' object containing the valid parameters. Valid column names are 'file.raw.coverage', 'file.gc.factor', 'file.corrected.coverage', and 'sample.names'. Additional columns will be ignored.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_sample.names">sample.names</code></td>
<td>
<p>Character vector of sample names. Alternatively can be specified in 'df'.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_ano.file">ano.file</code></td>
<td>
<p>Location of the annotation file. This file must be in CSV format and contain the following information (with column headers as specified): &quot;feature,chromosome,start,end&quot;.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_out.dir">out.dir</code></td>
<td>
<p>Output directory for results. A subdirectory for results will be created under this + '/nkr/'.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_ncpus">ncpus</code></td>
<td>
<p>Integer number of CPUs to use. Specifying more than one allows this function to be parallelized by feature.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_file.raw.coverage">file.raw.coverage</code></td>
<td>
<p>Character vector listing the raw input coverage files. Must be the same length as 'sample.names'. Alternatively can be specified in 'df'.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_file.corrected.coverage">file.corrected.coverage</code></td>
<td>
<p>Character vector listing the corrected input coverage files. If not specified new names will be generated based on the raw coverage files.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_file.gc.factor">file.gc.factor</code></td>
<td>
<p>Character vector listing the GC factor files used to correct coverage. If 'estimate_gc=FALSE' then this must be provided. Otherwise it is ignored.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_win.size">win.size</code></td>
<td>
<p>Size in base pairs of the sliding window used to estimate and correct the GC bias.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_gc.step">gc.step</code></td>
<td>
<p>Bin size for GC bias in the GC factor file. If the GC factor file is provided then the file must have corresponding bin sizes.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_estimate_gc">estimate_gc</code></td>
<td>
<p>Logical determining if GC content estimation should be performed. If set to 'FALSE' then a factor file must be provided via 'file.gc.factor' or in 'df'.</p>
</td></tr>
<tr><td><code id="correct_gc_bias_+3A_verbose">verbose</code></td>
<td>
<p>If set to TRUE will display more detail</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be run as a stand-alone or as part of <a href="#topic+run_RCNA">run_RCNA</a>
</p>
<p>The 'df' argument corresponds to the 'gcParams' matrix on <a href="#topic+RCNA_object-class">RCNA_object</a>. Valid column names are 'sample.names', 'file.raw.coverage', 'file.corrected.coverage', and 'file.gc.factor'. The 'file.gc.factor' column is not required if 'estimate_gc=TRUE'. Additional columns will be ignored.
</p>
<p>For more parameter information, see <a href="#topic+estimate_nkr.default">estimate_nkr.default</a>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>
<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>
<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>


<h3>See Also</h3>

<p><a href="#topic+RCNA_object-class">RCNA_object</a>, <a href="#topic+RCNA_analysis-class">RCNA_analysis</a>, <a href="#topic+run_RCNA">run_RCNA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run GC-bias estimation and correction on example object
# See \link{example_obj} for more information on example
example_obj@ano.file &lt;- system.file("examples" ,"annotations-example.csv",
 package = "RCNA")
raw.cov &lt;- system.file("examples", "coverage",
                       paste0(example_obj@sample.names, ".txt.gz"), package = "RCNA")
example_obj@gcParams$file.raw.coverage &lt;- raw.cov
example_obj
# Create output directory
dir.create(file.path("output", "gc"), recursive = TRUE)
# Estimate and correct GC bias, append results
correct_gc_analysisObj &lt;- correct_gc_bias(example_obj)
example_obj@commands &lt;- c(example_obj@commands, correct_gc_analysisObj)

</code></pre>

<hr>
<h2 id='create_RCNA_object'>RCNA_object constructor</h2><span id='topic+create_RCNA_object'></span>

<h3>Description</h3>

<p>An S4 class used to specify parameters for an analysis run
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_RCNA_object(
  sample.names,
  ano.file,
  ncpu = 1,
  out.dir = tempdir(),
  file.coverage = NULL,
  gcParams = NULL,
  win.size = 75,
  gc.step = 0.01,
  file.raw.coverage = NULL,
  file.corrected.coverage = NULL,
  file.gc.factor = NULL,
  estimate_gc = TRUE,
  nkrParams = NULL,
  file.nkr.coverage = NULL,
  nkr = 0.9,
  x.norm = NULL,
  norm.cov.matrix = NULL,
  scoreParams = NULL,
  file.score.coverage = NULL,
  score.cutoff = 0.5,
  low.score.cutoff = NULL,
  high.score.cutoff = NULL,
  commands = list(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_RCNA_object_+3A_sample.names">sample.names</code></td>
<td>
<p>Character vector containing names of subjects</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_ano.file">ano.file</code></td>
<td>
<p>Character single file path detailing a feature-wise annotation file</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_ncpu">ncpu</code></td>
<td>
<p>Numeric value specifying number of cores to use for analysis. Multiple cores will lead to parallel execution.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_out.dir">out.dir</code></td>
<td>
<p>Character vector containing the name of each subject's output directory</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_file.coverage">file.coverage</code></td>
<td>
<p>Character vector containing the path to the input coverage files for NKR and CNA score estimation.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_gcparams">gcParams</code></td>
<td>
<p>Data Frame storing all run parameters for the correct_gc_bias function. Can be specified by a file path to a CSV file, 'data.frame', or (if not specified) will be generated by other arguments.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_win.size">win.size</code></td>
<td>
<p>Numeric value detailing the size of the sliding window used to calculate and detect correct GC-content correction.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_gc.step">gc.step</code></td>
<td>
<p>Numeric value detailing the size of each GC-content bin. If providing pre-calculated GC factor file this must match the bins in that file.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_file.raw.coverage">file.raw.coverage</code></td>
<td>
<p>Character vector containing the filename of the raw coverage files for GC-content correction. Must be used in combination with 'estimate_gc' set to TRUE.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_file.corrected.coverage">file.corrected.coverage</code></td>
<td>
<p>Character vector containing the filename of the corrected coverage files.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_file.gc.factor">file.gc.factor</code></td>
<td>
<p>Character vector containing the filename of GC factor files. Used if and only if 'estimate_gc' is set to FALSE.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_estimate_gc">estimate_gc</code></td>
<td>
<p>Logical that determines if GC bias factor is calculated. If set to TRUE then GC factor files will be generated for each sample. If set to FALSE then GC factor files must be supplied via 'file.gc.factor'.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_nkrparams">nkrParams</code></td>
<td>
<p>Data Frame storing all run parameters for the estimate_nkr function. Can be specified by a file path to a CSV file, 'data.frame', or (if not specified) will be generated by other arguments.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_file.nkr.coverage">file.nkr.coverage</code></td>
<td>
<p>Character vector containing the filename of the input coverage file for NKR estimation. Defaults to 'file coverage' if not specified.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_nkr">nkr</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the coverage quantile that should be considered a &quot;normal&quot; karyotype range for each position. Lowering this value may increase sensitivity but also Type I error.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_x.norm">x.norm</code></td>
<td>
<p>Logical vector with length equal to the length of 'sample.names', denoting whether each subject has to be X-normalized. Subjects with an XX karyotype should be set to TRUE to avoid double-counting the coverage on the X chromosome. Set to FALSE if chrX coverage is already normalized.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_norm.cov.matrix">norm.cov.matrix</code></td>
<td>
<p>Character containing the directory or file name of the normalized coverage matrix. Generated by <code><a href="#topic+estimate_nkr">estimate_nkr</a></code> if file doesn't exist.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_scoreparams">scoreParams</code></td>
<td>
<p>Data Frame storing all run parameters for the estimate_feature_score function. Can be specified by a file path to a CSV file, 'data.frame', or (if not specified) will be generated by other arguments.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_file.score.coverage">file.score.coverage</code></td>
<td>
<p>Character vector containing the input coverage file for the scoring function. Defaults to 'file.coverage' if not specified.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_score.cutoff">score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the score filter on the results file. This parameter creates a symmetrical cutoff around 0, filtering all results whose absolute value is less than the specified value. Non-symmetrical cutoffs can be specified using 'low.score.cutoff' and 'high.score.cutoff'.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_low.score.cutoff">low.score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the lower score cutoff. Defaults to 'score.cutoff' if not specified.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_high.score.cutoff">high.score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the upper score cutoff. Defaults to 'score.cutoff' if not specified.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_commands">commands</code></td>
<td>
<p>RCNA_analysis object storing commands and parameters from previous function runs on this object. For more information, see <code><a href="#topic+RCNA_analysis-class">RCNA_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="create_RCNA_object_+3A_verbose">verbose</code></td>
<td>
<p>Show more messages and warnings. Useful for debugging.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+RCNA_object-class">RCNA_object</a> class object with the specified parameters.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RCNA_analysis-class">RCNA_analysis</a></code>, <code><a href="#topic+run_RCNA">run_RCNA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an example object - see \link{example_obj} for more information.
samples &lt;- c("ex-sample-1", "ex-sample-2", "ex-sample-3")
ex.obj &lt;- create_RCNA_object(sample.names = samples,
                    ano.file = system.file("examples" ,"annotations-example.csv", package = "RCNA"),
                    out.dir = "output",
                    file.raw.coverage = system.file("examples", "coverage",
                       paste0(samples, ".txt.gz"), package = "RCNA"),
                    norm.cov.matrix = file.path("output", "norm-cov-matrix.csv.gz"),
                    nkr = 0.9,
                    x.norm = "FALSE",
                    low.score.cutoff = -0.35,
                    high.score.cutoff = 0.35,
                    ncpu = 1)
class(ex.obj)

</code></pre>

<hr>
<h2 id='estimate_feature_score'>estimate_feature_score: Estimate CNV score for each gene in the annotation file</h2><span id='topic+estimate_feature_score'></span><span id='topic+estimate_feature_score.default'></span><span id='topic+estimate_feature_score.RCNA_object'></span>

<h3>Description</h3>

<p>This function estimates the the CNA score for each feature in the annotation file. It creates two flat file text tables with a row for each feature, which is placed in the output directory under '/score' - one with the score filter applied and one with all score results reported.
</p>
<p>This function estimates the the CNA score for each feature in the annotation file. It creates two flat file text tables with a row for each feature, which is placed in the output directory under '/score' - one with the score filter applied and one with all score results reported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_feature_score(obj, ...)

## Default S3 method:
estimate_feature_score(
  obj = NULL,
  df = NULL,
  sample.names = NULL,
  ano.file,
  out.dir = NULL,
  ncpus = 1,
  file.score.coverage = NULL,
  score.cutoff = 0.5,
  low.score.cutoff = NULL,
  high.score.cutoff = NULL,
  verbose = FALSE,
  ...
)

## S3 method for class 'RCNA_object'
estimate_feature_score(obj, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimate_feature_score_+3A_obj">obj</code></td>
<td>
<p>A RCNA_object type object - parameters will be pulled from the object instead, specifically from the 'scoreParams' slot.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_...">...</code></td>
<td>
<p>Additional arguments (unused)</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_df">df</code></td>
<td>
<p>Path to the config file, or a 'data.frame' object containing the valid parameters. Valid column names are 'file.score.coverage' and 'sample.names'. Additional columns will be ignored.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_sample.names">sample.names</code></td>
<td>
<p>Character vector of sample names. Alternatively can be specified in 'df'.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_ano.file">ano.file</code></td>
<td>
<p>Location of the annotation file. This file must be in CSV format and contain the following information (with column headers as specified): &quot;feature,chromosome,start,end&quot;.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_out.dir">out.dir</code></td>
<td>
<p>Output directory for results. A subdirectory for results will be created under this + '/nkr/'.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_ncpus">ncpus</code></td>
<td>
<p>Integer number of CPUs to use. Specifying more than one allows this function to be parallelized by feature.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_file.score.coverage">file.score.coverage</code></td>
<td>
<p>Character vector listing the input coverage files. Must be the same length as 'sample.names'. Alternatively can be specified in 'df'.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_score.cutoff">score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the score filter on the results file. This parameter creates a symmetrical cutoff around 0, filtering all results whose absolute value is less than the specified value. Non-symmetrical cutoffs can be specified using 'low.score.cutoff' and 'high.score.cutoff'.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_low.score.cutoff">low.score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the lower score cutoff. Defaults to 'score.cutoff' if not specified.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_high.score.cutoff">high.score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the upper score cutoff. Defaults to 'score.cutoff' if not specified.</p>
</td></tr>
<tr><td><code id="estimate_feature_score_+3A_verbose">verbose</code></td>
<td>
<p>If set to TRUE will display more detail</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be run as a stand-alone or as part of <a href="#topic+run_RCNA">run_RCNA</a>.
</p>
<p>The 'df' argument corresponds to the 'scoreParams' matrix on <a href="#topic+RCNA_object-class">RCNA_object</a>. Valid column names are 'sample.names' and 'file.score.coverage'. Additional columns will be ignored.
</p>
<p>For more parameter information, see <a href="#topic+estimate_feature_score.default">estimate_feature_score.default</a>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>
<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>
<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>


<h3>See Also</h3>

<p><a href="#topic+RCNA_object-class">RCNA_object</a>, <a href="#topic+RCNA_analysis-class">RCNA_analysis</a>, <a href="#topic+run_RCNA">run_RCNA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Estimate feature scores on example object
# See \link{example_obj} for more information on example
example_obj@ano.file &lt;- system.file("examples" ,"annotations-example.csv", package = "RCNA")
example_obj
# Create output directories
dir.create(file.path("output", "score"), recursive = TRUE)
# Copy example GC-corrected coverage files
cov.corrected &lt;- system.file("examples", "gc", package = "RCNA")
file.copy(from = cov.corrected, to = "output", recursive = TRUE)
# Copy example NKR results for "feature_a"
nkr.res &lt;- system.file("examples", "nkr", package = "RCNA")
file.copy(from = nkr.res, to = "output", recursive = TRUE)
# Run score estimation for "feature_a" and append results
estimate_feature_score_analysisObj &lt;- estimate_feature_score(example_obj)
example_obj@commands &lt;- c(example_obj@commands, estimate_feature_score_analysisObj)

</code></pre>

<hr>
<h2 id='estimate_nkr'>estimate_nkr: Estimate CNA &quot;normal&quot; karyotype ranges</h2><span id='topic+estimate_nkr'></span><span id='topic+estimate_nkr.default'></span><span id='topic+estimate_nkr.RCNA_object'></span>

<h3>Description</h3>

<p>This generic function is used to run normal karyotype range estimation.
</p>
<p>This function estimates the normal karyotype range for each feature in the annotation file. It creates an .RData object for each feature, which is placed in the output directory under '/nkr'. This intermediate output is used in <a href="#topic+estimate_feature_score">estimate_feature_score</a>.
</p>
<p>This function estimates the normal karyotype range for each feature in the annotation file. It creates an .RData object for each feature, which is placed in the output directory under '/nkr'. This intermediate output is used in <a href="#topic+estimate_feature_score">estimate_feature_score</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_nkr(obj, ...)

## Default S3 method:
estimate_nkr(
  obj = NULL,
  df = NULL,
  sample.names = NULL,
  ano.file,
  out.dir = NULL,
  ncpus = 1,
  file.ci.coverage = NULL,
  nkr = 0.9,
  x.norm = NULL,
  norm.cov.matrix = NULL,
  verbose = FALSE,
  ...
)

## S3 method for class 'RCNA_object'
estimate_nkr(obj, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimate_nkr_+3A_obj">obj</code></td>
<td>
<p>A RCNA_object type object - parameters will be pulled from the object instead, specifically from the 'nkrParams' slot.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_...">...</code></td>
<td>
<p>Additional arguments (unused)</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_df">df</code></td>
<td>
<p>Path to the config file, or a 'data.frame' object containing the valid parameters. Valid column names are 'file.nkr.coverage', 'x.norm', and 'sample.names'. Additional columns will be ignored.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_sample.names">sample.names</code></td>
<td>
<p>Character vector of sample names. Alternatively can be specified in 'df'.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_ano.file">ano.file</code></td>
<td>
<p>Location of the annotation file. This file must be in CSV format and contain the following information (with column headers as specified): &quot;feature,chromosome,start,end&quot;.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_out.dir">out.dir</code></td>
<td>
<p>Output directory for results. A subdirectory for results will be created under this + '/nkr/'.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_ncpus">ncpus</code></td>
<td>
<p>Integer number of CPUs to use. Specifying more than one allows this function to be parallelized by feature.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_file.ci.coverage">file.ci.coverage</code></td>
<td>
<p>Character vector listing the input coverage files. Must be the same length as 'sample.names'. Alternatively can be specified in 'df'.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_nkr">nkr</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the coverage quantile that should be considered a &quot;normal&quot; karyotype range for each position. Lowering this value may increase sensitivity but also Type I error.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_x.norm">x.norm</code></td>
<td>
<p>Whether or not to perform normalization for normal female/XX karyotype (default = FALSE). Can be specified for each sample separately via 'df' column labeled 'x.norm'.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_norm.cov.matrix">norm.cov.matrix</code></td>
<td>
<p>Character file path detailing the location of the normalized coverage matrix generated by this function. Re-using this file between runs can cut down on runtime significantly for large sample sizes. If the file doesn't exist yet it will be created at this location. If this file name ends in &quot;.gz&quot; then the output will be compressed using gzip.</p>
</td></tr>
<tr><td><code id="estimate_nkr_+3A_verbose">verbose</code></td>
<td>
<p>If set to TRUE will display more detail</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be run as a stand-alone or as part of <a href="#topic+run_RCNA">run_RCNA</a>
</p>
<p>The 'df' argument corresponds to the 'nkrParams' matrix on <a href="#topic+RCNA_object-class">RCNA_object</a>. Valid column names are 'sample.names', 'file.ci.coverage', and 'x.norm'. Additional columns will be ignored.
</p>
<p>For more parameter information, see <a href="#topic+estimate_nkr.default">estimate_nkr.default</a>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>
<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>
<p>A <a href="#topic+RCNA_analysis-class">RCNA_analysis</a> class object that describes the input parameters and output files generated by this step of the workflow.
</p>


<h3>See Also</h3>

<p><a href="#topic+RCNA_object-class">RCNA_object</a>, <a href="#topic+RCNA_analysis-class">RCNA_analysis</a>, <a href="#topic+run_RCNA">run_RCNA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run NKR estimation on example object
# See \link{example_obj} for more information on example
example_obj@ano.file &lt;- system.file("examples" ,"annotations-example.csv", package = "RCNA")
example_obj
# Create output directory
dir.create(file.path("output", "nkr"), recursive = TRUE)
# Copy example GC-corrected coverage files
cov.corrected &lt;- system.file("examples", "gc", package = "RCNA")
file.copy(from = cov.corrected, to = "output", recursive = TRUE)
# Run NKR estimation, append results
estimate_nkr_analysisObj &lt;- estimate_nkr(example_obj)
example_obj@commands &lt;- c(example_obj@commands, estimate_nkr_analysisObj)

</code></pre>

<hr>
<h2 id='example_obj'>Example RCNA_object</h2><span id='topic+example_obj'></span>

<h3>Description</h3>

<p>An example RCNA object used to run examples and demonstrate the structure of the custom S4 object provided in this package.
This example uses a dummy feature (&quot;feature_a&quot;) and three coverage files which were subset to the length of the dummy feature to be concise and quick to run.
An annotation file has been included in the 'inst/' directory along with the coverage files. This object is compiled in the '<code>create_RCNA_object</code>&lsquo; function&rsquo;s documentation.
In order to use this example, you should make the following replacements:
<code>
example_obj@ano.file &lt;- system.file("examples" ,"annotations-example.csv", package = "RCNA")
raw.cov &lt;- system.file("examples", "coverage", paste0(samples, ".txt.gz"), package = "RCNA")
example_obj@gcParams$file.raw.coverage &lt;- raw.cov
</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_obj
</code></pre>


<h3>Format</h3>

<p>An RCNA object created using create_RCNA_object(). See <a href="#topic+create_RCNA_object">create_RCNA_object</a> for more details on the slots of this object.
</p>


<h3>Details</h3>

<p>This will set the location of the example annotation file and the example raw coverage files to the flat files included with the package.
</p>

<hr>
<h2 id='RCNA_analysis-class'>RCNA_analysis constructor</h2><span id='topic+RCNA_analysis-class'></span><span id='topic+RCNA_analysis'></span>

<h3>Description</h3>

<p>An S4 class used to track parameters from a specific RCNA function execution.
</p>


<h3>Slots</h3>


<dl>
<dt><code>call</code></dt><dd><p>A character vector detailing the function call that ('correct_gc_bias', 'estimate_nkr', 'estimate_feature_score') was performed to produce this S4 object.</p>
</dd>
<dt><code>params</code></dt><dd><p>A list corresponding to the parameters that were submitted with the associated function call</p>
</dd>
<dt><code>res.files</code></dt><dd><p>A list containing the names of the flat files that were created by the documented function call.</p>
</dd>
</dl>

<hr>
<h2 id='RCNA_object-class'>RCNA.object definition</h2><span id='topic+RCNA_object-class'></span><span id='topic+RCNA_object'></span>

<h3>Description</h3>

<p>An S4 class used to specify parameters for an analysis run
</p>


<h3>Slots</h3>


<dl>
<dt><code>sample.names</code></dt><dd><p>Required. Character vector containing names of subjects</p>
</dd>
<dt><code>ano.file</code></dt><dd><p>Required. Character single file path detailing a feature-wise annotation file</p>
</dd>
<dt><code>out.dir</code></dt><dd><p>Required. Character vector containing the name of each subject?s output directory</p>
</dd>
<dt><code>gcParams</code></dt><dd><p>Data Frame storing all run parameters for the correct_gc_bias function</p>
</dd>
<dt><code>nkrParams</code></dt><dd><p>Data Frame storing all run parameters for the estimate_nkr function</p>
</dd>
<dt><code>scoreParams</code></dt><dd><p>Data Frame storing all run parameters for the estimate_feature_score function</p>
</dd>
<dt><code>commands</code></dt><dd><p>RCNA_analysis object storing commands and parameters from previous function runs on this object</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="#topic+run_RCNA">run_RCNA</a></code>
<code><a href="#topic+RCNA_analysis-class">RCNA_analysis</a></code>,
</p>

<hr>
<h2 id='run_RCNA'>run_RCNA: Perform RCNA copy number detection workflow</h2><span id='topic+run_RCNA'></span><span id='topic+run_RCNA.default'></span><span id='topic+run_RCNA.RCNA_object'></span>

<h3>Description</h3>

<p>'run_RCNA' will execute <a href="#topic+correct_gc_bias">correct_gc_bias</a>, <a href="#topic+estimate_nkr">estimate_nkr</a>, and <a href="#topic+estimate_feature_score">estimate_feature_score</a> in that specific order. For more information, see each of those functions' individual documentation.
</p>
<p>'run_RCNA' will execute <a href="#topic+correct_gc_bias">correct_gc_bias</a>, <a href="#topic+estimate_nkr">estimate_nkr</a>, and <a href="#topic+estimate_feature_score">estimate_feature_score</a> in that specific order. For more information, see each of those functions' individual documentation, or <a href="#topic+create_RCNA_object">create_RCNA_object</a>.
</p>
<p>'run_RCNA' will execute <a href="#topic+correct_gc_bias">correct_gc_bias</a>, <a href="#topic+estimate_nkr">estimate_nkr</a>, and <a href="#topic+estimate_feature_score">estimate_feature_score</a> in that specific order. For more information, see each of those functions' individual documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_RCNA(obj, ...)

## Default S3 method:
run_RCNA(
  obj = NULL,
  sample.names,
  ano.file,
  out.dir = tempdir(),
  gcParams = NULL,
  win.size = 75,
  gc.step = 0.01,
  file.raw.coverage = NULL,
  file.corrected.coverage = NULL,
  file.gc.factor = NULL,
  estimate_gc = TRUE,
  nkrParams,
  file.nkr.coverage = NULL,
  ncpu = 1,
  nkr = 0.9,
  x.norm = NULL,
  scoreParams,
  score.cutoff = 0.5,
  low.score.cutoff = NULL,
  high.score.cutoff = NULL,
  commands = c(),
  verbose = FALSE,
  ...
)

## S3 method for class 'RCNA_object'
run_RCNA(obj, estimate_gc = TRUE, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_RCNA_+3A_obj">obj</code></td>
<td>
<p>An 'RCNA_object' type created by <a href="#topic+create_RCNA_object">create_RCNA_object</a>.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_...">...</code></td>
<td>
<p>Additional arguments (unused).</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_sample.names">sample.names</code></td>
<td>
<p>Character vector containing names of subjects</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_ano.file">ano.file</code></td>
<td>
<p>Character single file path detailing a feature-wise annotation file</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_out.dir">out.dir</code></td>
<td>
<p>Character vector containing the name of each subject's output directory</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_gcparams">gcParams</code></td>
<td>
<p>Data Frame storing all run parameters for the correct_gc_bias function. Can be specified by a file path to a CSV file, 'data.frame', or (if not specified) will be generated by other arguments.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_win.size">win.size</code></td>
<td>
<p>Numeric value detailing the size of the sliding window used to calculate and detect correct GC-content correction.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_gc.step">gc.step</code></td>
<td>
<p>Numeric value detailing the size of each GC-content bin. If providing pre-calculated GC factor file this must match the bins in that file.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_file.raw.coverage">file.raw.coverage</code></td>
<td>
<p>Character vector containing the filename of the raw coverage files for GC-content correction. Must be used in combination with 'estimate_gc' set to TRUE.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_file.corrected.coverage">file.corrected.coverage</code></td>
<td>
<p>Character vector containing the filename of the corrected coverage files.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_file.gc.factor">file.gc.factor</code></td>
<td>
<p>Character vector containing the filename of GC factor files. Used if and only if 'estimate_gc' is set to FALSE.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_estimate_gc">estimate_gc</code></td>
<td>
<p>A logical which determines if GC estimation should be performed. For more information, see <a href="#topic+correct_gc_bias">correct_gc_bias</a>.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_nkrparams">nkrParams</code></td>
<td>
<p>Data Frame storing all run parameters for the estimate_nkr function. Can be specified by a file path to a CSV file, 'data.frame', or (if not specified) will be generated by other arguments.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_file.nkr.coverage">file.nkr.coverage</code></td>
<td>
<p>Character vector containing the filename of the input coverage file for NKR estimation. Defaults to 'file coverage' if not specified.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_ncpu">ncpu</code></td>
<td>
<p>Numeric value specifying number of cores to use for analysis. Multiple cores will lead to parallel execution.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_nkr">nkr</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the coverage quantile that should be considered a &quot;normal&quot; karyotype range for each position. Lowering this value may increase sensitivity but also Type I error.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_x.norm">x.norm</code></td>
<td>
<p>Logical vector with length equal to the length of 'sample.names', denoting whether each subject has to be X-normalized. Subjects with an XX karyotype should be set to TRUE to avoid double-counting the coverage on the X chromosome. Set to FALSE if chrX coverage is already normalized.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_scoreparams">scoreParams</code></td>
<td>
<p>Data Frame storing all run parameters for the estimate_feature_score function. Can be specified by a file path to a CSV file, 'data.frame', or (if not specified) will be generated by other arguments.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_score.cutoff">score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the score filter on the results file. This parameter creates a symmetrical cutoff around 0, filtering all results whose absolute value is less than the specified value. Non-symmetrical cutoffs can be specified using 'low.score.cutoff' and 'high.score.cutoff'.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_low.score.cutoff">low.score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the lower score cutoff. Defaults to 'score.cutoff' if not specified.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_high.score.cutoff">high.score.cutoff</code></td>
<td>
<p>Numeric between 0 and 1 which specifies the upper score cutoff. Defaults to 'score.cutoff' if not specified.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_commands">commands</code></td>
<td>
<p>RCNA_analysis object storing commands and parameters from previous function runs on this object. For more information, see <code><a href="#topic+RCNA_analysis-class">RCNA_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="run_RCNA_+3A_verbose">verbose</code></td>
<td>
<p>If set to TRUE will display more detailed error messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+RCNA_object-class">RCNA_object</a> class object that was used during the workflow, with <a href="#topic+RCNA_analysis">RCNA_analysis</a> objects in the 'commands' slot that describes the run parameters and results of each step in the workflow.
</p>
<p>A <a href="#topic+RCNA_object-class">RCNA_object</a> class object that was used during the workflow, with <a href="#topic+RCNA_analysis">RCNA_analysis</a> objects in the 'commands' slot that describes the run parameters and results of each step in the workflow. For more details on outputs, see <a href="#topic+estimate_nkr">estimate_nkr</a>, <a href="#topic+correct_gc_bias">correct_gc_bias</a>, and <a href="#topic+estimate_feature_score">estimate_feature_score</a>.
</p>
<p>A <a href="#topic+RCNA_object-class">RCNA_object</a> class object that was used during the workflow, with <a href="#topic+RCNA_analysis">RCNA_analysis</a> objects in the 'commands' slot that describes the run parameters and results of each step in the workflow. For more details on outputs, see <a href="#topic+estimate_nkr">estimate_nkr</a>, <a href="#topic+correct_gc_bias">correct_gc_bias</a>, and <a href="#topic+estimate_feature_score">estimate_feature_score</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+RCNA_object-class">RCNA_object</a>, <a href="#topic+RCNA_analysis-class">RCNA_analysis</a>, <a href="#topic+correct_gc_bias">correct_gc_bias</a>, <a href="#topic+run_RCNA">run_RCNA</a>, <a href="#topic+estimate_feature_score">estimate_feature_score</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run RCNA workflow on example object
# See ?example_obj for more information on example
example_obj@ano.file &lt;- system.file("examples" ,"annotations-example.csv", package = "RCNA")
raw.cov &lt;- system.file("examples", "coverage",
                       paste0(example_obj@sample.names, ".txt.gz"), package = "RCNA")
example_obj@gcParams$file.raw.coverage &lt;- raw.cov
example_obj
# Run RCNA workflow
result_obj &lt;- run_RCNA(example_obj)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
