<!DOCTYPE html><html><head><title>Help for package rcompanion</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rcompanion}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rcompanion-package'><p>Functions to Support Extension Education Program Evaluation</p></a></li>
<li><a href='#accuracy'><p>Minimum maximum accuracy, mean absolute percent error,</p>
median absolute error,
root mean square error, coefficient of variation,
and Efron's pseudo r-squared</a></li>
<li><a href='#Anderson'><p>Hypothetical data for Alexander Anderson</p></a></li>
<li><a href='#AndersonBias'><p>Hypothetical data for Alexander Anderson with gender bias</p></a></li>
<li><a href='#AndersonRainBarrel'><p>Hypothetical data for Alexander Anderson on rain barrel installation</p></a></li>
<li><a href='#AndersonRainGarden'><p>Hypothetical data for Alexander Anderson on rain garden installation</p></a></li>
<li><a href='#blom'><p>Normal scores transformation</p></a></li>
<li><a href='#BobBelcher'><p>Hypothetical data for ratings of instructors in unreplicated CBD</p></a></li>
<li><a href='#Breakfast'><p>Hypothetical data for students' breakfast habits and travel to school</p></a></li>
<li><a href='#BrendonSmall'><p>Hypothetical data for Brendon Small and company</p></a></li>
<li><a href='#BullyHill'><p>Hypothetical data for proportion of students passing a certification</p></a></li>
<li><a href='#Catbus'><p>Hypothetical data for Catbus and company</p></a></li>
<li><a href='#cateNelson'><p>Cate-Nelson models for bivariate data</p></a></li>
<li><a href='#cateNelsonFixedY'><p>Cate-Nelson models for bivariate data with a fixed critical Y value</p></a></li>
<li><a href='#cldList'><p>Compact letter display for lists of comparisons</p></a></li>
<li><a href='#cliffDelta'><p>Cliff's delta</p></a></li>
<li><a href='#cohenG'><p>Cohen's g and odds ratio for paired contingency tables</p></a></li>
<li><a href='#cohenH'><p>Cohen's h to compare proportions for 2 x 2 contingency tables</p></a></li>
<li><a href='#cohenW'><p>Cohen's w (omega)</p></a></li>
<li><a href='#compareGLM'><p>Compare fit statistics for glm models</p></a></li>
<li><a href='#compareLM'><p>Compare fit statistics for lm models</p></a></li>
<li><a href='#countRSquare'><p>Count pseudo r-squared for logistic and other binary outcome models</p></a></li>
<li><a href='#cramerV'><p>Cramer's V (phi)</p></a></li>
<li><a href='#cramerVFit'><p>Cramer's V for chi-square goodness-of-fit tests</p></a></li>
<li><a href='#efronRSquared'><p>Efron's pseudo r-squared</p></a></li>
<li><a href='#epsilonSquared'><p>Epsilon-squared</p></a></li>
<li><a href='#freemanTheta'><p>Freeman's theta</p></a></li>
<li><a href='#fullPTable'><p>Convert a lower triangle matrix to a full matrix</p></a></li>
<li><a href='#groupwiseCMH'><p>Post-hoc tests for Cochran-Mantel-Haenszel test</p></a></li>
<li><a href='#groupwiseGeometric'><p>Groupwise geometric means and confidence intervals</p></a></li>
<li><a href='#groupwiseHuber'><p>Groupwise Huber M-estimators and confidence intervals</p></a></li>
<li><a href='#groupwiseMean'><p>Groupwise means and confidence intervals</p></a></li>
<li><a href='#groupwiseMedian'><p>Groupwise medians and confidence intervals</p></a></li>
<li><a href='#groupwisePercentile'><p>Groupwise percentiles and confidence intervals</p></a></li>
<li><a href='#groupwiseSum'><p>Groupwise sums</p></a></li>
<li><a href='#HayleySmith'><p>Hypothetical data for responses about adopting lawn care practices</p></a></li>
<li><a href='#kendallW'><p>Kendall's W with bootstrapped confidence interval</p></a></li>
<li><a href='#mangiaficoD'><p>Mangiafico's d</p></a></li>
<li><a href='#Monarchs'><p>Hypothetical data for monarch butterflies in gardens</p></a></li>
<li><a href='#multiMangiaficoD'><p>Mangiafico's d</p></a></li>
<li><a href='#multiVDA'><p>Pairwise Vargha and Delaney's A and Cliff's delta</p></a></li>
<li><a href='#nagelkerke'><p>Pseudo r-squared measures for various models</p></a></li>
<li><a href='#nagelkerkeHermite'><p>[Defunct!] Pseudo r-squared measures for hermite models</p></a></li>
<li><a href='#nominalSymmetryTest'><p>Exact and McNemar symmetry tests for paired contingency tables</p></a></li>
<li><a href='#Nurseries'><p>Data for proportion of good practices followed by plant nuseries</p></a></li>
<li><a href='#oneSampleDominance'><p>Dominance statistic for one-sample data</p></a></li>
<li><a href='#ordinalEtaSquared'><p>Eta-squared for ordinal variables</p></a></li>
<li><a href='#pairedSampleDominance'><p>Dominance statistic for two-sample paired data</p></a></li>
<li><a href='#pairwiseDifferences'><p>[Defunct!] Pairwise differences for unreplicated CBD</p></a></li>
<li><a href='#pairwiseMcnemar'><p>Pairwise McNemar and related tests for Cochran Q test post-hoc</p></a></li>
<li><a href='#pairwiseMedianMatrix'><p>Pairwise Mood's median tests with matrix output</p></a></li>
<li><a href='#pairwiseMedianTest'><p>Pairwise Mood's median tests</p></a></li>
<li><a href='#pairwiseModelAnova'><p>Compare model objects with F test and likelihood ratio test</p></a></li>
<li><a href='#pairwiseNominalIndependence'><p>Pairwise tests of independence for nominal data</p></a></li>
<li><a href='#pairwiseNominalMatrix'><p>Pairwise tests of independence for nominal data with matrix output</p></a></li>
<li><a href='#pairwiseOrdinalIndependence'><p>Pairwise tests of independence for tables with one ordered nominal variable</p></a></li>
<li><a href='#pairwiseOrdinalMatrix'><p>[Defunct!] Pairwise two-sample ordinal regression with matrix output</p></a></li>
<li><a href='#pairwiseOrdinalPairedMatrix'><p>[Defunct!] Pairwise two-sample ordinal regression for paired data</p>
with matrix output</a></li>
<li><a href='#pairwiseOrdinalPairedTest'><p>[Defunct!] Pairwise two-sample ordinal regression for paired data</p></a></li>
<li><a href='#pairwiseOrdinalTest'><p>[Defunct!] Pairwise two-sample ordinal regression</p></a></li>
<li><a href='#pairwisePercentileTest'><p>Pairwise permutation tests for percentiles</p></a></li>
<li><a href='#pairwisePermutationMatrix'><p>Pairwise two-sample independence tests with matrix output</p></a></li>
<li><a href='#pairwisePermutationSymmetry'><p>Pairwise two-sample symmetry tests</p></a></li>
<li><a href='#pairwisePermutationSymmetryMatrix'><p>Pairwise two-sample symmetry tests with matrix output</p></a></li>
<li><a href='#pairwisePermutationTest'><p>Pairwise two-sample independence tests</p></a></li>
<li><a href='#pairwiseRobustMatrix'><p>[Defunct!] Pairwise two-sample robust tests with matrix output</p></a></li>
<li><a href='#pairwiseRobustTest'><p>[Defunct!] Pairwise two-sample robust tests</p></a></li>
<li><a href='#pairwiseSignMatrix'><p>[Defunct!] Pairwise sign tests with matrix output</p></a></li>
<li><a href='#pairwiseSignTest'><p>[Defunct!] Pairwise sign tests</p></a></li>
<li><a href='#Pennsylvania18'><p>Votes for the Democratic candidate in Pennsylvania 18 in 2016 and 2018</p></a></li>
<li><a href='#percentileTest'><p>Test of percentiles by permutation test</p></a></li>
<li><a href='#phi'><p>phi</p></a></li>
<li><a href='#plotDensityHistogram'><p>Histogram with a density curve</p></a></li>
<li><a href='#plotNormalDensity'><p>Density plot with a normal curve</p></a></li>
<li><a href='#plotNormalHistogram'><p>Histogram with a normal curve</p></a></li>
<li><a href='#plotPredy'><p>Plot a predicted line from a bivariate model</p></a></li>
<li><a href='#PMCMRTable'><p>Convert PMCMR Objects to a Data Frame</p></a></li>
<li><a href='#Pooh'><p>Hypothetical data for paired ratings of Pooh Bear</p></a></li>
<li><a href='#PoohPiglet'><p>Hypothetical data for ratings of Pooh, Piglet, and Tigger</p></a></li>
<li><a href='#quantileCI'><p>Quantiles and confidence intervals</p></a></li>
<li><a href='#Religion'><p>Hypothetical data for change in religion after a caucusing event</p></a></li>
<li><a href='#scheirerRayHare'><p>Scheirer Ray Hare test</p></a></li>
<li><a href='#spearmanRho'><p>Spearman's rho, Kendall's tau, Pearson's r</p></a></li>
<li><a href='#transformTukey'><p>Tukey's Ladder of Powers</p></a></li>
<li><a href='#vda'><p>Vargha and Delaney's A</p></a></li>
<li><a href='#wilcoxonOneSampleR'><p>r effect size for Wilcoxon one-sample signed-rank test</p></a></li>
<li><a href='#wilcoxonOneSampleRC'><p>Rank biserial correlation coefficient for one-sample Wilcoxon test</p></a></li>
<li><a href='#wilcoxonOR'><p>Agresti's Generalized Odds Ratio for Stochastic Dominance</p></a></li>
<li><a href='#wilcoxonPairedR'><p>r effect size for Wilcoxon two-sample paired signed-rank test</p></a></li>
<li><a href='#wilcoxonPairedRC'><p>Matched-pairs rank biserial correlation coefficient</p></a></li>
<li><a href='#wilcoxonPS'><p>Grissom and Kim's Probability of Superiority (PS)</p></a></li>
<li><a href='#wilcoxonR'><p>r effect size for Wilcoxon two-sample rank-sum test</p></a></li>
<li><a href='#wilcoxonRG'><p>Glass rank biserial correlation coefficient</p></a></li>
<li><a href='#wilcoxonZ'><p>Wilcoxon z statistic</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.4.35</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Functions to Support Extension Education Program Evaluation</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions and datasets to support Summary and Analysis of
             Extension Program Evaluation in R, and An R
             Companion for the Handbook of Biological Statistics. 
             Vignettes are available at <a href="https://rcompanion.org">https://rcompanion.org</a>.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://CRAN.R-project.org/package=rcompanion">https://CRAN.R-project.org/package=rcompanion</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats (&ge; 4.1.0), graphics (&ge; 4.1.0), boot (&ge; 1.3-28),
DescTools (&ge; 0.99.43), multcompView (&ge; 0.1-8), plyr (&ge;
1.8.6), coin (&ge; 1.4-2), lmtest (&ge; 0.9-38), nortest (&ge; 1.0-4)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Salvatore Mangiafico &lt;mangiafico@njaes.rutgers.edu&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-17 22:29:08 UTC; salva</td>
</tr>
<tr>
<td>Author:</td>
<td>Salvatore Mangiafico
    <a href="https://orcid.org/0000-0003-4668-6821"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-17 23:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='rcompanion-package'>Functions to Support Extension Education Program Evaluation</h2><span id='topic+rcompanion-package'></span>

<h3>Description</h3>

<p>Functions and datasets to support Summary and Analysis of
Extension Program Evaluation in R and An R
Companion for the Handbook of Biological Statistics.
</p>


<h3>Useful functions</h3>

<p>There are several functions that provide summary statistics for
grouped data. These function titles tend to start with <code>"groupwise"</code>.
They provide means, medians, geometric means, and Huber M-estimators
for groups, along with confidence intervals by traditional
methods and bootstrap.
</p>
<p>Functions to produce effect size statistics, 
some with bootstrapped confidence intervals, 
include those for Cramer's V, 
Cohen's g and odds ratio for paired tables, Cohen's h, 
Cohen's w, Vargha and Delaney's A, Cliff's delta,
r for one-sample, two-sample, and paired Wilcoxon and Mann-Whitney tests, 
epsilon-squared, and Freeman's theta. 
</p>
<p>The <code><a href="#topic+accuracy">accuracy</a></code> function
reports statistics for
models including minimum maximum accuracy, MAPE, RMSE, 
Efron's pseudo r-squared, and coefficient of variation.
</p>
<p>The functions <code><a href="#topic+nagelkerke">nagelkerke</a></code> and <code><a href="#topic+efronRSquared">efronRSquared</a></code>
provide pseudo R-squared values for a variety of model types, as well as
a likelihood ratio test for the model as a whole.
</p>
<p>There are also functions that are useful for comparing models.
<code><a href="#topic+compareLM">compareLM</a></code>,  <code><a href="#topic+compareGLM">compareGLM</a></code>, and 
<code><a href="#topic+pairwiseModelAnova">pairwiseModelAnova</a></code>.
These use goodness-of-fit measures like AIC, BIC, and BICc, or likelihood
ratio tests. 
</p>
<p>Functions for nominal data include post-hoc tests for 
Cochran-Mantel-Haenszel test (<code><a href="#topic+groupwiseCMH">groupwiseCMH</a></code>),
for McNemar-Bowker test (<code><a href="#topic+pairwiseMcnemar">pairwiseMcnemar</a></code>),
and for tests of association like Chi-square, Fisher exact, and G-test
(<code><a href="#topic+pairwiseNominalIndependence">pairwiseNominalIndependence</a></code>).
</p>
<p>There are a few useful plotting functions, including 
<code><a href="#topic+plotNormalHistogram">plotNormalHistogram</a></code> that plots a histogram of values and 
overlays
a normal curve, and <code><a href="#topic+plotPredy">plotPredy</a></code> which plots of line for predicted
values for a bivariate model.  Other plotting functions include producing
density plots.
</p>
<p>A function close to my heart is <code><a href="#topic+cateNelson">cateNelson</a></code>, which performs
Cate-Nelson analysis for bivariate data.
</p>


<h3>Vignettes and examples</h3>

<p>The functions in this package are used in
&quot;Extension Education Program Evaluation in R&quot; which is available at
<a href="https://rcompanion.org/handbook/">https://rcompanion.org/handbook/</a>
and &quot;An R Companion for the Handbook of Biological Statistics&quot; 
which is available at <a href="https://rcompanion.org/rcompanion/">https://rcompanion.org/rcompanion/</a>.
</p>
<p>The documentation for each function includes an example as well.
</p>


<h3>Version notes</h3>

<p>Version 2.0 is not entirely back-compatible
as several functions have been removed.
These include some of the pairwise methods that can be replaced with
better methods.  Also, some functions have been removed or modified
in order to import fewer packages.
</p>
<p>Removed packages are indicated with 'Defunct' in their titles.
</p>

<hr>
<h2 id='accuracy'>Minimum maximum accuracy, mean absolute percent error,
median absolute error, 
root mean square error, coefficient of variation,
and Efron's pseudo r-squared</h2><span id='topic+accuracy'></span>

<h3>Description</h3>

<p>Produces a table of fit statistics for multiple models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accuracy(fits, plotit = FALSE, digits = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accuracy_+3A_fits">fits</code></td>
<td>
<p>A series of model object names. 
Must be a list of model objects or a single model object.</p>
</td></tr>
<tr><td><code id="accuracy_+3A_plotit">plotit</code></td>
<td>
<p>If <code>TRUE</code>, produces plots of the predicted values
vs. the actual values for each model.</p>
</td></tr>
<tr><td><code id="accuracy_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="accuracy_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>plot</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces a table of fit statistics for multiple models: 
minimum maximum accuracy, mean absolute percentage error,
median absolute error,
root mean square error, normalized root mean square error, 
Efron's pseudo r-squared, and coefficient of variation.
</p>
<p>For minimum maximum accuracy, larger indicates
a better fit, 
and a perfect fit is equal to 1.
</p>
<p>For mean absolute error (MAE), smaller
indicates a better fit,
and a perfect fit is equal to 0.
It has the same units as the dependent variable.
Note that here, MAE is simply the mean of the absolute
values of the differences of predicted values and the
observed values 
(<code>MAE = mean(abs(predy - actual))</code>).
There are other definitions of MAE and similar-sounding
terms.
</p>
<p>Median absolute error (MedAE) is similar, except employing
the median rather than the mean.
</p>
<p>For mean absolute percent error (MAPE), smaller
indicates a better fit,
and a perfect fit is equal to 0. The result is reported
as a fraction.  That is, a result of 0.1 is equal to 10
</p>
<p>Root mean square error (RMSE) has the same units as the predicted
values.
</p>
<p>Normalized root mean square error (NRMSE) is RMSE divided by
the mean or the median of the values of the dependent variable.
</p>
<p>Efron's pseudo r-squared is calculated as 1 minus the residual sum 
of squares divided by the total sum of squares.  For linear models
(<code>lm</code> model objects), Efron's pseudo r-squared will be equal  
to r-squared.  For other models, it should not be interpreted
as r-squared, but can still be useful as a relative measure.
</p>
<p><code>CV.prcnt</code> is the coefficient of variation for the model.
Here it is expressed as a percent.  That is, a result of 10 = 10
</p>
<p>Model objects currently supported: lm, glm, nls, betareg, gls,
lme, lmer, lmerTest, glmmTMB, 
rq, loess, gam, glm.nb, glmRob, mblm, and rlm.
</p>


<h3>Value</h3>

<p>A list of two objects: The series of model calls, and a data 
frame of statistics for each model.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/G_14.html">https://rcompanion.org/handbook/G_14.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareLM">compareLM</a></code>, 
<code><a href="#topic+compareGLM">compareGLM</a></code>, 
<code><a href="#topic+nagelkerke">nagelkerke</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BrendonSmall)
BrendonSmall$Calories = as.numeric(BrendonSmall$Calories)
BrendonSmall$Calories2 = BrendonSmall$Calories ^ 2
model.1 = lm(Sodium ~ Calories, data = BrendonSmall)

accuracy(model.1, plotit=FALSE)

model.2 = lm(Sodium ~ Calories + Calories2, data = BrendonSmall)
model.3 = glm(Sodium ~ Calories, data = BrendonSmall, family="Gamma")
quadplat = function(x, a, b, clx) {
          ifelse(x  &lt; clx, a + b * x   + (-0.5*b/clx) * x   * x,
                           a + b * clx + (-0.5*b/clx) * clx * clx)}
model.4 = nls(Sodium ~ quadplat(Calories, a, b, clx),
              data = BrendonSmall,
              start = list(a=519, b=0.359, clx = 2300))
              
accuracy(list(model.1, model.2, model.3, model.4), plotit=FALSE)

### Perfect and poor model fits
X = c(1, 2,  3,  4,  5,  6, 7, 8, 9, 10, 11, 12)
Y = c(1, 2,  3,  4,  5,  6, 7, 8, 9, 10, 11, 12)
Z = c(1, 12, 13, 6, 10, 13, 4, 3, 5,  6, 10, 14)
perfect = lm(Y ~ X)
poor    = lm(Z ~ X)
accuracy(list(perfect, poor), plotit=FALSE)

</code></pre>

<hr>
<h2 id='Anderson'>Hypothetical data for Alexander Anderson</h2><span id='topic+Anderson'></span>

<h3>Description</h3>

<p>A matrix of counts for students passing or failing a pesticide 
training course across four counties. Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Anderson
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 4 rows and 2 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_04.html">https://rcompanion.org/handbook/H_04.html</a>
</p>

<hr>
<h2 id='AndersonBias'>Hypothetical data for Alexander Anderson with gender bias</h2><span id='topic+AndersonBias'></span>

<h3>Description</h3>

<p>A data frame of counts for students passing or failing a  
pesticicde training course across four counties, with gender of students. 
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AndersonBias
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 16 rows and 4 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_06.html">https://rcompanion.org/handbook/H_06.html</a>
</p>

<hr>
<h2 id='AndersonRainBarrel'>Hypothetical data for Alexander Anderson on rain barrel installation</h2><span id='topic+AndersonRainBarrel'></span>

<h3>Description</h3>

<p>A matrix of paired counts for students planning to install rain
barrels before and after a class. Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AndersonRainBarrel
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 2 rows and 2 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_05.html">https://rcompanion.org/handbook/H_05.html</a>
</p>

<hr>
<h2 id='AndersonRainGarden'>Hypothetical data for Alexander Anderson on rain garden installation</h2><span id='topic+AndersonRainGarden'></span>

<h3>Description</h3>

<p>A matrix of paired counts for students planning to install rain
gardens before and after a class. Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AndersonRainGarden
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 3 rows and 3 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_05.html">https://rcompanion.org/handbook/H_05.html</a>
</p>

<hr>
<h2 id='blom'>Normal scores transformation</h2><span id='topic+blom'></span>

<h3>Description</h3>

<p>Normal scores transformation
(Inverse normal transformation)
by Elfving, Blom, van der Waerden, Tukey, 
and rankit methods,
as well as z score transformation
(standardization)
and scaling to a range (normalization).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blom(
  x,
  method = "general",
  alpha = pi/8,
  complete = FALSE,
  na.last = "keep",
  na.rm = TRUE,
  adjustN = TRUE,
  min = 1,
  max = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blom_+3A_x">x</code></td>
<td>
<p>A vector of numeric values.</p>
</td></tr>
<tr><td><code id="blom_+3A_method">method</code></td>
<td>
<p>Any one <code>"general"</code> (the default), 
<code>"blom"</code>, <code>vdw</code>,
<code>"tukey"</code>, <code>"elfving"</code>,
<code>"rankit"</code>,
<code>zscore</code>, or <code>scale</code>.</p>
</td></tr>
<tr><td><code id="blom_+3A_alpha">alpha</code></td>
<td>
<p>A value used in the <code>"general"</code> method.
If alpha=pi/8 (the default), the <code>"general"</code> method reduces  
to the <code>"elfving"</code> method.
If alpha=3/8, the <code>"general"</code> method reduces 
to the <code>"blom"</code> method.
If alpha=1/2, the <code>"general"</code> method reduces to the 
<code>"rankit"</code> method.
If alpha=1/3, the <code>"general"</code> method reduces to the 
<code>"tukey"</code> method.
If alpha=0, the <code>"general"</code> method reduces to the 
<code>"vdw"</code> method.</p>
</td></tr>
<tr><td><code id="blom_+3A_complete">complete</code></td>
<td>
<p>If <code>TRUE</code>, <code>NA</code> values are removed
before transformation. The default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="blom_+3A_na.last">na.last</code></td>
<td>
<p>Passed to <code>rank</code> in the normal scores methods. 
See the documentation for the
<code>rank</code> function.
The default is <code>"keep"</code>.</p>
</td></tr>
<tr><td><code id="blom_+3A_na.rm">na.rm</code></td>
<td>
<p>Used in the <code>"zscore"</code> and <code>"scale"</code> methods.
Passed to <code>mean</code>, <code>min</code>, and <code>max</code>
functions in those methods.
The default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="blom_+3A_adjustn">adjustN</code></td>
<td>
<p>If <code>TRUE</code>, the default, the normal scores methods 
use only non-<code>NA</code> values to determine the sample size,
<code>N</code>.  This seems to work well under default conditions
where <code>NA</code> values are retained, even if there are
a high percentage of <code>NA</code> values.</p>
</td></tr>
<tr><td><code id="blom_+3A_min">min</code></td>
<td>
<p>For the <code>"scale"</code> method, the minimum value of the
transformed values.</p>
</td></tr>
<tr><td><code id="blom_+3A_max">max</code></td>
<td>
<p>For the <code>"scale"</code> method, the maximum value of the
transformed values.</p>
</td></tr>
<tr><td><code id="blom_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code>rank</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, <code>NA</code> values are retained in the output.
This behavior can be changed with the <code>na.rm</code> argument
for <code>"zscore"</code> and <code>"scale"</code> methods, or
with <code>na.last</code> for the normal scores methods.
Or <code>NA</code> values can be removed from the input with
<code>complete=TRUE</code>.
</p>
<p>For normal scores methods, if there are <code>NA</code> values
or tied values,
it is helpful to look up 
the documentation for <code>rank</code>.
</p>
<p>In general, for normal scores methods, either of the arguments
<code>method</code> or <code>alpha</code> can be used.
With the current algorithms, there is no need to use both.
</p>
<p>Normal scores transformation will return a
normal distribution with a mean of 0 and
a standard deviation of 1.
</p>
<p>The <code>"scale"</code> method coverts values to the range specified
in <code>max</code> and <code>min</code> without transforming the distribution
of values. By default, the <code>"scale"</code> method converts values
to a 1 to 10 range.
Using the <code>"scale"</code> method with
<code>min = 0</code> and <code>max = 1</code> is
sometimes called &quot;normalization&quot;.
</p>
<p>The <code>"zscore"</code> method converts values by the usual method
for z scores: <code>(x - mean(x)) / sd(x)</code>. The transformed
values with have a mean of 0 and a standard deviation of
1 but won't be coerced into a normal distribution.
Sometimes this method is called &quot;standardization&quot;.
</p>


<h3>Value</h3>

<p>A vector of numeric values.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>Conover, 1995, Practical Nonparametric Statistics, 3rd.
</p>
<p>Solomon &amp; Sawilowsky, 2009, 
Impact of rank-based normalizing transformations 
on the accuracy of test scores.
</p>
<p>Beasley and Erickson, 2009, Rank-based inverse normal 
transformations are increasingly used, but are they merited?
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(12345)
A = rlnorm(100)
## Not run: hist(A)
### Convert data to normal scores by Elfving method
B = blom(A)
## Not run: hist(B)
### Convert data to z scores 
C = blom(A, method="zscore")
## Not run: hist(C)
### Convert data to a scale of 1 to 10 
D = blom(A, method="scale")
## Not run: hist(D)

### Data from Sokal and Rohlf, 1995, 
### Biometry: The Principles and Practice of Statistics
### in Biological Research
Value = c(709,679,699,657,594,677,592,538,476,508,505,539)
Sex   = c(rep("Male",3), rep("Female",3), rep("Male",3), rep("Female",3))
Fat   = c(rep("Fresh", 6), rep("Rancid", 6))
ValueBlom = blom(Value)
Sokal = data.frame(ValueBlom, Sex, Fat)
model = lm(ValueBlom ~ Sex * Fat, data=Sokal)
anova(model)
## Not run: 
hist(residuals(model))
plot(predict(model), residuals(model))

## End(Not run)

</code></pre>

<hr>
<h2 id='BobBelcher'>Hypothetical data for ratings of instructors in unreplicated CBD</h2><span id='topic+BobBelcher'></span>

<h3>Description</h3>

<p>A data frame of Likert responses for five
instructors for each of 8 respondents. 
Arranged in unreplicated complete block design.
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BobBelcher
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 40 rows and 3 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/F_10.html">https://rcompanion.org/handbook/F_10.html</a>
</p>

<hr>
<h2 id='Breakfast'>Hypothetical data for students' breakfast habits and travel to school</h2><span id='topic+Breakfast'></span>

<h3>Description</h3>

<p>A two-dimensional contingency table, in which Breakfast 
is an ordered nominal variable,
and Travel is a non-ordered nominal variable.
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Breakfast
</code></pre>


<h3>Format</h3>

<p>An object of class <code>table</code> with 3 rows and 5 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_09.html">https://rcompanion.org/handbook/H_09.html</a>
</p>

<hr>
<h2 id='BrendonSmall'>Hypothetical data for Brendon Small and company</h2><span id='topic+BrendonSmall'></span>

<h3>Description</h3>

<p>A data frame of the intake of calories and sodium for students 
in five classes. Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BrendonSmall
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 45 rows and 6 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/I_10.html">https://rcompanion.org/handbook/I_10.html</a>
</p>

<hr>
<h2 id='BullyHill'>Hypothetical data for proportion of students passing a certification</h2><span id='topic+BullyHill'></span>

<h3>Description</h3>

<p>A data frame of counts of students passing and failing.
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BullyHill
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 12 rows and 5 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/J_02.html">https://rcompanion.org/handbook/J_02.html</a>
</p>

<hr>
<h2 id='Catbus'>Hypothetical data for Catbus and company</h2><span id='topic+Catbus'></span>

<h3>Description</h3>

<p>A data frame of the number of steps taken by students in three
classes. Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Catbus
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 26 rows and 5 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/C_03.html">https://rcompanion.org/handbook/C_03.html</a>
</p>

<hr>
<h2 id='cateNelson'>Cate-Nelson models for bivariate data</h2><span id='topic+cateNelson'></span>

<h3>Description</h3>

<p>Produces critical-x and critical-y values for bivariate data
according to a Cate-Nelson analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cateNelson(
  x,
  y,
  plotit = TRUE,
  hollow = TRUE,
  xlab = "X",
  ylab = "Y",
  trend = "positive",
  clx = 1,
  cly = 1,
  xthreshold = 0.1,
  ythreshold = 0.1,
  progress = TRUE,
  verbose = TRUE,
  listout = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cateNelson_+3A_x">x</code></td>
<td>
<p>A vector of values for the x variable.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_y">y</code></td>
<td>
<p>A vector of values for the y variable.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_plotit">plotit</code></td>
<td>
<p>If <code>TRUE</code>, produces plots of the output.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_hollow">hollow</code></td>
<td>
<p>If <code>TRUE</code>, uses hollow circles on the plot to indicate
data not fitting the model.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_xlab">xlab</code></td>
<td>
<p>The label for the x-axis.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_ylab">ylab</code></td>
<td>
<p>The label for the y-axis.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_trend">trend</code></td>
<td>
<p><code>"postive"</code> if the trend of y vs. x is generally
positive. <code>"negative"</code> if negative.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_clx">clx</code></td>
<td>
<p>Indicates which of the listed critical x values 
should be chosen as the critical x value for the final model.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_cly">cly</code></td>
<td>
<p>Indicates which of the listed critical y values 
should be chosen as the critical y value for the final model.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_xthreshold">xthreshold</code></td>
<td>
<p>Indicates the proportion of potential critical x values
to display in the output. A value of <code>1</code> would display
all of them.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_ythreshold">ythreshold</code></td>
<td>
<p>Indicates the proportion of potential critical y values
to display in the output. A value of <code>1</code> would display
all of them.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_progress">progress</code></td>
<td>
<p>If <code>TRUE</code>, prints an indicator of progress as for loops
progress.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_verbose">verbose</code></td>
<td>
<p>If <code>FALSE</code>, suppresses printed output of tables.</p>
</td></tr>
<tr><td><code id="cateNelson_+3A_listout">listout</code></td>
<td>
<p>If <code>TRUE</code>, outputs a list of data frames instead of a
a single data frame.  This allows a data frame of critical
values and associated statistics to be extracted, for example
if one would want to sort by Cramer's V.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cate-Nelson analysis divides bivariate data into two groups.
For data with a positive trend, one group has a
large <code>x</code> value associated with a large <code>y</code> value, and   
the other group has a small <code>x</code> value associated with a small   
<code>y</code> value. For a negative trend, a small <code>x</code> is
associated with a large <code>y</code>, and so on. 
</p>
<p>The analysis is useful for bivariate data which don't conform well
to linear, curvilinear, or plateau models.
</p>
<p>This function will fail if either of the largest two or smallest 
two x values are identical.
</p>


<h3>Value</h3>

<p>A data frame of statistics from the analysis: number of observations,
critical level for x, sum of squares, critical value for y, the
number of observations in each of the quadrants (I, II, III, IV),
the number of observations that conform with the model, 
the proportion of observations that conform with the model, 
the number of observations that do not conform to the model,
the proportion of observations that do not conform to the model,
a p-value for the Fisher exact test for the data divided into
the groups indicated by the model,
and Cramer's V for the data divided into
the groups indicated by the model.
</p>
<p>Output also includes printed lists of critical values,
explanation of the values in the data frame,
and plots: y vs. x; sum of squares vs. critical x value;
the number of observations that do not conform to the model vs.
critical y value; 
and y vs. x with the critical values shown as lines on the plot, 
and the quadrants labeled.
</p>


<h3>Note</h3>

<p>The method in this function follows <cite>Cate, R. B., &amp; Nelson, L.A. 
(1971). A simple statistical procedure for partitioning soil test 
correlation data into two classes. Soil Science Society of America 
Proceedings 35, 658-660.</cite>
</p>
<p>An earlier version of this function was published in <cite>Mangiafico, 
S.S. 2013. Cate-Nelson Analysis for Bivariate Data Using 
R-project. J.of Extension 51:5, 5TOT1.</cite>
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/rcompanion/h_02.html">https://rcompanion.org/rcompanion/h_02.html</a>
</p>
<p>Cate, R. B., &amp; Nelson, L.A. (1971). A simple statistical procedure
for partitioning soil test correlation data into two classes. 
Soil Science Society of America Proceedings 35, 658–660.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cateNelsonFixedY">cateNelsonFixedY</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nurseries)
cateNelson(x          = Nurseries$Size,
           y          = Nurseries$Proportion,
           plotit     = TRUE,
           hollow     = TRUE,
           xlab       = "Nursery size in hectares",
           ylab       = "Proportion of good practices adopted",
           trend      = "positive",
           clx        = 1,
           xthreshold = 0.10,
           ythreshold = 0.15)

</code></pre>

<hr>
<h2 id='cateNelsonFixedY'>Cate-Nelson models for bivariate data with a fixed critical Y value</h2><span id='topic+cateNelsonFixedY'></span>

<h3>Description</h3>

<p>Produces critical-x values for bivariate data
according to a Cate-Nelson analysis
for a given critical Y value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cateNelsonFixedY(
  x,
  y,
  cly = 0.95,
  plotit = TRUE,
  hollow = TRUE,
  xlab = "X",
  ylab = "Y",
  trend = "positive",
  clx = 1,
  outlength = 20,
  sortstat = "error"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cateNelsonFixedY_+3A_x">x</code></td>
<td>
<p>A vector of values for the x variable.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_y">y</code></td>
<td>
<p>A vector of values for the y variable.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_cly">cly</code></td>
<td>
<p>= Critical Y value.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_plotit">plotit</code></td>
<td>
<p>If <code>TRUE</code>, produces plots of the output.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_hollow">hollow</code></td>
<td>
<p>If <code>TRUE</code>, uses hollow circles on the plot to indicate
data not fitting the model.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_xlab">xlab</code></td>
<td>
<p>The label for the x-axis.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_ylab">ylab</code></td>
<td>
<p>The label for the y-axis.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_trend">trend</code></td>
<td>
<p><code>"postive"</code> if the trend of y vs. x is generally
positive. <code>"negative"</code> if negative.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_clx">clx</code></td>
<td>
<p>Indicates which of the listed critical x values 
should be chosen as the critical x value for the plot.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_outlength">outlength</code></td>
<td>
<p>Indicates the number of potential critical x values
to display in the output.</p>
</td></tr>
<tr><td><code id="cateNelsonFixedY_+3A_sortstat">sortstat</code></td>
<td>
<p>The statistic to sort by.  Any of <code>"error"</code> 
(the default),
<code>"phi"</code>, <code>"fisher"</code>, or <code>"pearson"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cate-Nelson analysis divides bivariate data into two groups.
For data with a positive trend, one group has a
large <code>x</code> value associated with a large <code>y</code> value, and   
the other group has a small <code>x</code> value associated with a small   
<code>y</code> value. For a negative trend, a small <code>x</code> is
associated with a large <code>y</code>, and so on. 
</p>
<p>The analysis is useful for bivariate data which don't conform well
to linear, curvilinear, or plateau models.
</p>


<h3>Value</h3>

<p>A data frame of statistics from the analysis:
critical level for x, critical value for y, the
number of observations in each of the quadrants (I, II, III, IV),
the number of observations that conform with the model, 
the number of observations that do not conform to the model,
the proportion of observations that conform with the model,
the proportion of observations that do not conform to the model,
a p-value for the Fisher exact test for the data divided into
the groups indicated by the model,
phi for the data divided into
the groups indicated by the model,
and Pearson's chi-square for the data divided into
the groups indicated by the model.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/rcompanion/h_02.html">https://rcompanion.org/rcompanion/h_02.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cateNelson">cateNelson</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nurseries)
cateNelsonFixedY(x          = Nurseries$Size,
                 y          = Nurseries$Proportion,
                 cly        = 0.70,
                 plotit     = TRUE,
                 hollow     = TRUE,
                 xlab       = "Nursery size in hectares",
                 ylab       = "Proportion of good practices adopted",
                 trend      = "positive",
                 clx        = 1,
                 outlength  = 15)

</code></pre>

<hr>
<h2 id='cldList'>Compact letter display for lists of comparisons</h2><span id='topic+cldList'></span>

<h3>Description</h3>

<p>Produces a compact letter display (cld) from pairwise 
comparisons that were summarized in a table of comparisons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cldList(
  formula = NULL,
  data = NULL,
  comparison = NULL,
  p.value = NULL,
  threshold = 0.05,
  print.comp = FALSE,
  remove.space = TRUE,
  remove.equal = TRUE,
  remove.zero = TRUE,
  swap.colon = TRUE,
  swap.vs = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cldList_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the variable holding p-values and
the variable holding the comparisons. 
e.g. P.adj ~ Comparison.</p>
</td></tr>
<tr><td><code id="cldList_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="cldList_+3A_comparison">comparison</code></td>
<td>
<p>A vector of text describing comparisons, 
with each element in a form similar to
&quot;Treat.A - Treat.B = 0&quot;.  Spaces and &quot;=&quot; and &quot;0&quot;
are removed by default</p>
</td></tr>
<tr><td><code id="cldList_+3A_p.value">p.value</code></td>
<td>
<p>A vector of p-values corresponding to the comparisons
in the <code>comparison</code> argument</p>
</td></tr>
<tr><td><code id="cldList_+3A_threshold">threshold</code></td>
<td>
<p>The alpha value.  That is, the p-value below which the
comparison will be considered significant</p>
</td></tr>
<tr><td><code id="cldList_+3A_print.comp">print.comp</code></td>
<td>
<p>If <code>TRUE</code>, prints out a data frame of the
modified text of the comparisons.  Useful for debugging</p>
</td></tr>
<tr><td><code id="cldList_+3A_remove.space">remove.space</code></td>
<td>
<p>If <code>TRUE</code>, removes spaces from the text of the
comparisons</p>
</td></tr>
<tr><td><code id="cldList_+3A_remove.equal">remove.equal</code></td>
<td>
<p>If <code>TRUE</code>, removes &quot;=&quot; from the text of the
comparisons</p>
</td></tr>
<tr><td><code id="cldList_+3A_remove.zero">remove.zero</code></td>
<td>
<p>If <code>TRUE</code>, removes &quot;0&quot; from the text of the
comparisons</p>
</td></tr>
<tr><td><code id="cldList_+3A_swap.colon">swap.colon</code></td>
<td>
<p>If <code>TRUE</code>, swaps &quot;:&quot; with &quot;-&quot; in the text of the
comparisons</p>
</td></tr>
<tr><td><code id="cldList_+3A_swap.vs">swap.vs</code></td>
<td>
<p>If <code>TRUE</code>, swaps &quot;vs&quot; with &quot;-&quot; in the text of the
comparisons</p>
</td></tr>
<tr><td><code id="cldList_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>multcompLetters</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>comparison</code> and <code>p.value</code>.
</p>
<p>This function relies upon the <code>multcompLetters</code>
function in the <code>multcompView</code> package. The text for the
comparisons
passed to <code>multcompLetters</code> should be in the form
&quot;Treat.A-Treat.B&quot;.  Currently by default <code>cldList</code> removes
spaces, equal signs, and zeros, by default, 
and so can use 
text in the form e.g.
&quot;Treat.A - Treat.B = 0&quot;.
It also changes &quot;:&quot; to &quot;-&quot;, and so can use
text in the form e.g.
&quot;Treat.A : Treat.B&quot;.
</p>


<h3>Value</h3>

<p>A data frame of group names, group separation letters,
and monospaced separtions letters
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.  
</p>
<p>It is often helpful to reorder the factor levels in the data
set so that the group with the largest e.g.
mean or median is first, and so on.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/G_06.html">https://rcompanion.org/handbook/G_06.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fullPTable">fullPTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BrendonSmall)

model = aov(Calories ~ Instructor, data=BrendonSmall)

TUK = TukeyHSD(model, "Instructor", ordered = TRUE)

### Convert the TukeyHSD output to a standard data frame

TUK = as.data.frame(TUK$Instructor)
names(TUK) = gsub(" ", ".", names(TUK))

HSD = data.frame(Comparison=row.names(TUK), 
                 diff=TUK$diff, lwr=TUK$lwr, lwr=TUK$lwr, p.adj=TUK$p.adj)

HSD

cldList(p.adj ~ Comparison, data = HSD,
        threshold = 0.05,
        remove.space=FALSE)

</code></pre>

<hr>
<h2 id='cliffDelta'>Cliff's delta</h2><span id='topic+cliffDelta'></span>

<h3>Description</h3>

<p>Calculates Cliff's delta
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cliffDelta(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  reportIncomplete = FALSE,
  brute = FALSE,
  verbose = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cliffDelta_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_brute">brute</code></td>
<td>
<p>If <code>FALSE</code>, the default, the statistic is based on the
U statistic from the <code>wilcox.test</code> function.
If <code>TRUE</code>, the function will compare values
in the two samples directly.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, reports the proportion of ties and
the proportions of (Ya &gt; Yb) and (Ya &lt; Yb).</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="cliffDelta_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcox.test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cliff's delta is an effect size statistic appropriate
in cases where a Wilcoxon-Mann-Whitney test might be used.
It ranges from -1 to 1, with 0 indicating stochastic equality,
and 1 indicating that the first group dominates the second.
It is linearly related to Vargha and Delaney's A.
</p>
<p>By default, the function calculates Cliff's delta from the &quot;W&quot; 
U statistic from the
<code>wilcox.test</code> function.
Specifically, <code>VDA = U/(n1*n2); CD = (VDA-0.5)*2</code>.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>. If there are more than two groups,
only the first two groups are used.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, Cliff's delta is positive.
When the data in the second group are greater than
in the first group, Cliff's delta is negative.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>When Cliff's delta is close to 1 or close to -1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, Cliff's delta.
Or a small data frame consisting of Cliff's delta,
and the lower and upper confidence limits.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_04.html">https://rcompanion.org/handbook/F_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vda">vda</a></code>, 
<code><a href="#topic+multiVDA">multiVDA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
cliffDelta(Steps ~ Gender, data=Catbus)

</code></pre>

<hr>
<h2 id='cohenG'>Cohen's g and odds ratio for paired contingency tables</h2><span id='topic+cohenG'></span>

<h3>Description</h3>

<p>Calculates Cohen's g and odds ratio
for paired contingency tables, such as those that 
might be analyzed with   
McNemar or McNemar-Bowker tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohenG(
  x,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cohenG_+3A_x">x</code></td>
<td>
<p>A two-way contingency table. It must be square. 
It can have two or
more levels for each dimension.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="cohenG_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a 2 x 2 table, where a and d are the concordant cells
and b and c are discordant cells:
Odds ratio is b/c;
P is b/(b+c);
and Cohen's g is P - 0.5.
</p>
<p>In the 2 x 2 case, the statistics are directional.
That is, when cell [1, 2] in the table is greater than
cell [2, 1], OR is greater than 1, P is greater than 0.5,
and g is positive.  
</p>
<p>In the opposite case, OR is less than 1,
P is less than 0.5, and g is negative.
</p>
<p>In the 2 x 2 case, when the effect is small, the
confidence interval for OR can pass through 1,
for g can pass through 0, and for P can pass through 0.5.
</p>
<p>For tables larger than 2 x 2, the statistics are not directional.
That is, OR is always &gt;= 1, P is always &gt;= 0.5, and
g is always positive.  
Because of this, if <code>type="perc"</code>, the confidence interval will
never cross the values for no effect 
(OR = 1, P = 0.5, or g = 0).
Because of this, the confidence interval range
in this case should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross the values for no effect.
</p>
<p>When the reported statistics are close to their extremes,
or with small counts, 
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A list containing: a data frame of results of the global statistics;
and a data frame of results of the pairwise statistics.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_05.html">https://rcompanion.org/handbook/H_05.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nominalSymmetryTest">nominalSymmetryTest</a></code>, 
<code><a href="#topic+cohenH">cohenH</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### 2 x 2 repeated matrix example
data(AndersonRainBarrel)
cohenG(AndersonRainBarrel)
                    
### 3 x 3 repeated matrix
data(AndersonRainGarden)
cohenG(AndersonRainGarden)

</code></pre>

<hr>
<h2 id='cohenH'>Cohen's h to compare proportions for 2 x 2 contingency tables</h2><span id='topic+cohenH'></span>

<h3>Description</h3>

<p>Calculates Cohen's h
for 2 x 2 contingency tables, such as those that 
might be analyzed with   
a chi-square test of association.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohenH(x, observation = "row", verbose = TRUE, digits = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cohenH_+3A_x">x</code></td>
<td>
<p>A 2 x 2 contingency table.</p>
</td></tr>
<tr><td><code id="cohenH_+3A_observation">observation</code></td>
<td>
<p>If <code>"row"</code>, the row constitutes an observation.
That is, the sum of each row is 100 percent.
If <code>"column"</code>, the column constitutes an observation.  
That is, the sum of each column is 100 percent.</p>
</td></tr>
<tr><td><code id="cohenH_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints the proportions for each observation.</p>
</td></tr>
<tr><td><code id="cohenH_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cohen's h is an effect size to compare two proportions.
For a 2 x 2 table: 
Cohen's h equals Phi2 - Phi1, where,
If observations are in rows, P1 = a/(a+b) and P2 = c/(c+d).
If observations are in columns, P1 = a/(a+c) and P2 = b/(b+d).          
Phi = 2 * asin(sqrt(P))
</p>


<h3>Value</h3>

<p>A single statistic.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_10.html">https://rcompanion.org/handbook/H_10.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cohenG">cohenG</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pennsylvania18)
Pennsylvania18
cohenH(Pennsylvania18, observation="row")

</code></pre>

<hr>
<h2 id='cohenW'>Cohen's w (omega)</h2><span id='topic+cohenW'></span>

<h3>Description</h3>

<p>Calculates Cohen's w for a table of nominal variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohenW(
  x,
  y = NULL,
  p = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 4,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cohenW_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations for one dimension
of a two-way table.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_y">y</code></td>
<td>
<p>If <code>x</code> is a vector, <code>y</code> is the vector of observations for
the second dimension of a two-way table.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_p">p</code></td>
<td>
<p>If <code>x</code> is a vector of observed counts, <code>p</code> can be given as
a vector of theoretical probabilties,
as in a chi-square goodness of fit test.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.
In the case of the goodness-of-fit
scenario, setting this to <code>TRUE</code>
will have no effect.</p>
</td></tr>
<tr><td><code id="cohenW_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>chisq.test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cohen's w is used as a measure of association
between two nominal variables, or as an effect size
for a chi-square test of association.  For a 2 x 2 table,
the absolute value of the phi statistic is the same as
Cohen's w.  
The value of Cohen's w is not bound by 1 on the upper end.
</p>
<p>Cohen's w is &quot;naturally nondirectional&quot;. That is,
the value will always be zero or positive.
Because of this, if <code>type="perc"</code>,
the confidence interval will
never cross zero.
The confidence interval range should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross zero.  
</p>
<p>When w is close to 0 or very large,
or with small counts, 
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, Cohen's w.
Or a small data frame consisting of Cohen's w,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_10.html">https://rcompanion.org/handbook/H_10.html</a>
</p>
<p>Cohen J. 1992. &quot;A Power Primer&quot;. Psychological Bulletin 12(1): 155-159.
</p>
<p>Cohen, J. 1988. Statistical Power Analysis for the Behavioral Sciences, 2nd Ed. Routledge.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cramerV">cramerV</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with table
data(Anderson)
fisher.test(Anderson)
cohenW(Anderson)

### Example for goodness-of-fit
### Bird foraging example, Handbook of Biological Statistics
observed = c(70,   79,   3,    4)
expected = c(0.54, 0.40, 0.05, 0.01)
chisq.test(observed, p = expected)
cohenW(observed, p = expected)

### Example with two vectors
Species = c(rep("Species1", 16), rep("Species2", 16))
Color   = c(rep(c("blue", "blue", "blue", "green"),4),
            rep(c("green", "green", "green", "blue"),4))
fisher.test(Species, Color)
cohenW(Species, Color)

</code></pre>

<hr>
<h2 id='compareGLM'>Compare fit statistics for glm models</h2><span id='topic+compareGLM'></span>

<h3>Description</h3>

<p>Produces a table of fit statistics for multiple glm models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareGLM(fits, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareGLM_+3A_fits">fits</code></td>
<td>
<p>A series of model object names, separated by commas.</p>
</td></tr>
<tr><td><code id="compareGLM_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>list</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces a table of fit statistics for multiple glm models: 
AIC, AICc, BIC, p-value, pseudo R-squared
(McFadden, Cox and Snell, Nagelkerke).
</p>
<p>Smaller values for AIC, AICc, and BIC indicate a better balance
of goodness-of-fit of the model and the complexity of the
model. The goal is to find a model that adequately explains the 
data without having too many terms.
</p>
<p>BIC tends to choose models with fewer parameters relative to AIC.
</p>
<p>For comparisons with AIC, etc., to be valid, both models must
have the same data, without transformations, use the same 
dependent variable, and be fit with the same method.
They do not need to be nested.
</p>
<p>The function will fail if a model formula is
longer than 500 characters.
</p>


<h3>Value</h3>

<p>A list of two objects: The series of model calls, and a data 
frame of statistics for each model.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/rcompanion/e_07.html">https://rcompanion.org/rcompanion/e_07.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareLM">compareLM</a></code>, 
<code><a href="#topic+pairwiseModelAnova">pairwiseModelAnova</a></code>, 
<code><a href="#topic+accuracy">accuracy</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Compare among logistic regresion models
data(AndersonBias)
model.0 = glm(Result ~ 1, weight = Count, data = AndersonBias,
             family = binomial(link="logit"))
model.1 = glm(Result ~ County, weight = Count, data = AndersonBias,
             family = binomial(link="logit"))
model.2 = glm(Result ~ County + Gender, weight = Count, data = AndersonBias,
             family = binomial(link="logit"))
model.3 = glm(Result ~ County + Gender + County:Gender, weight = Count, 
             data = AndersonBias, family = binomial(link="logit"))
compareGLM(model.0, model.1, model.2, model.3)              

</code></pre>

<hr>
<h2 id='compareLM'>Compare fit statistics for lm models</h2><span id='topic+compareLM'></span>

<h3>Description</h3>

<p>Produces a table of fit statistics for multiple lm models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareLM(fits, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareLM_+3A_fits">fits</code></td>
<td>
<p>A series of model object names, separated by commas.</p>
</td></tr>
<tr><td><code id="compareLM_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>list</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces a table of fit statistics for multiple lm models: 
AIC, AICc, BIC, p-value, R-squared, and adjusted R-squared.
</p>
<p>Smaller values for AIC, AICc, and BIC indicate a better balance
of goodness-of-fit of the model and the complexity of the
model. The goal is to find a model that adequately explains the 
data without having too many terms.
</p>
<p>BIC tends to choose models with fewer parameters relative to AIC.
</p>
<p>In the table, <code>Shapiro.W</code> and <code>Shapiro.p</code> are the
W statistic and p-value for the Shapiro-Wilks test on the residuals
of the model.
</p>
<p>For comparisons with AIC, etc., to be valid, both models must
have the same data, without transformations, use the same 
dependent variable, and be fit with the same method.
They do not need to be nested.
</p>
<p>The function will fail if a model formula is
longer than 500 characters.
</p>


<h3>Value</h3>

<p>A list of two objects: The series of model calls, and a data 
frame of statistics for each model.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/I_10.html">https://rcompanion.org/handbook/I_10.html</a>,
<a href="https://rcompanion.org/rcompanion/e_05.html">https://rcompanion.org/rcompanion/e_05.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareGLM">compareGLM</a></code>, 
<code><a href="#topic+pairwiseModelAnova">pairwiseModelAnova</a></code>,
<code><a href="#topic+accuracy">accuracy</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Compare among polynomial models
data(BrendonSmall)
BrendonSmall$Calories = as.numeric(BrendonSmall$Calories)

BrendonSmall$Calories2 = BrendonSmall$Calories * BrendonSmall$Calories
BrendonSmall$Calories3 = BrendonSmall$Calories * BrendonSmall$Calories * 
                         BrendonSmall$Calories
BrendonSmall$Calories4 = BrendonSmall$Calories * BrendonSmall$Calories * 
                         BrendonSmall$Calories * BrendonSmall$Calories
model.1 = lm(Sodium ~ Calories, data = BrendonSmall)
model.2 = lm(Sodium ~ Calories + Calories2, data = BrendonSmall)
model.3 = lm(Sodium ~ Calories + Calories2 + Calories3, data = BrendonSmall)
model.4 = lm(Sodium ~ Calories + Calories2 + Calories3 + Calories4,
             data = BrendonSmall)
compareLM(model.1, model.2, model.3, model.4)

</code></pre>

<hr>
<h2 id='countRSquare'>Count pseudo r-squared for logistic and other binary outcome models</h2><span id='topic+countRSquare'></span>

<h3>Description</h3>

<p>Produces the count pseudo 
r-squared measure for models with a binary outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countRSquare(
  fit,
  digits = 3,
  suppressWarnings = TRUE,
  plotit = FALSE,
  jitter = FALSE,
  pch = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countRSquare_+3A_fit">fit</code></td>
<td>
<p>The fitted model object for which to determine pseudo r-squared.
<code>glm</code> and <code>glmmTMB</code> are supported.
Others may work as well.</p>
</td></tr>
<tr><td><code id="countRSquare_+3A_digits">digits</code></td>
<td>
<p>The number of digits in the outputted values.</p>
</td></tr>
<tr><td><code id="countRSquare_+3A_suppresswarnings">suppressWarnings</code></td>
<td>
<p>If <code>TRUE</code>, suppresses warning messages.</p>
</td></tr>
<tr><td><code id="countRSquare_+3A_plotit">plotit</code></td>
<td>
<p>If <code>TRUE</code>, produces a simple plot of
actual vs. predicted values.</p>
</td></tr>
<tr><td><code id="countRSquare_+3A_jitter">jitter</code></td>
<td>
<p>If <code>TRUE</code>, jitters the &quot;actual&quot; values in the plot.</p>
</td></tr>
<tr><td><code id="countRSquare_+3A_pch">pch</code></td>
<td>
<p>Passed to <code>plot</code>.</p>
</td></tr>
<tr><td><code id="countRSquare_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The count pseudo r-squared is simply the number of correctly
predicted observations divided the total number of observations.
</p>
<p>This version is appropriate for models with a binary outcome.
</p>
<p>The adjusted value deducts the count of the most frequent
outcome from both the numerator and the denominator.
</p>
<p>It is recommended that the model is fit on data in long
format.  That is, that the <code>weight</code> option not be used in
the model.
</p>
<p>The function makes no provisions for <code>NA</code> values.
It is recommended that <code>NA</code> values be removed before
the determination of the model.
</p>


<h3>Value</h3>

<p>A list including a description of the submitted model,
a data frame with the pseudo r-squared results,
and a confusion matrix of the results.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://stats.oarc.ucla.edu/other/mult-pkg/faq/general/faq-what-are-pseudo-r-squareds/">https://stats.oarc.ucla.edu/other/mult-pkg/faq/general/faq-what-are-pseudo-r-squareds/</a>,
<a href="https://rcompanion.org/handbook/H_08.html">https://rcompanion.org/handbook/H_08.html</a>,
<a href="https://rcompanion.org/rcompanion/e_06.html">https://rcompanion.org/rcompanion/e_06.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nagelkerke">nagelkerke</a></code>, 
<code><a href="#topic+efronRSquared">efronRSquared</a></code>, 
<code><a href="#topic+accuracy">accuracy</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AndersonBias)

### Covert data to long format

Long = AndersonBias[rep(row.names(AndersonBias), AndersonBias$Count),
                    c("Result", "County", "Gender")]
rownames(Long) = seq(1:nrow(Long))
str(Long)

### Fit model and determine count r-square

model = glm(Result ~ County + Gender + County:Gender,
            data = Long,
            family = binomial())

countRSquare(model)

</code></pre>

<hr>
<h2 id='cramerV'>Cramer's V (phi)</h2><span id='topic+cramerV'></span>

<h3>Description</h3>

<p>Calculates Cramer's V for a table of nominal variables;
confidence intervals by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cramerV(
  x,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 4,
  bias.correct = FALSE,
  reportIncomplete = FALSE,
  verbose = FALSE,
  tolerance = 1e-16,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cramerV_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations for one dimension
of a two-way table.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_y">y</code></td>
<td>
<p>If <code>x</code> is a vector, <code>y</code> is the vector of observations for
the second dimension of a two-way table.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_bias.correct">bias.correct</code></td>
<td>
<p>If <code>TRUE</code>, a bias correction is applied.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints additional statistics.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_tolerance">tolerance</code></td>
<td>
<p>If the variance of the bootstrapped values are less than
<code>tolerance</code>, NA is returned for the confidence interval
values.</p>
</td></tr>
<tr><td><code id="cramerV_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>chisq.test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cramer's V is used as a measure of association
between two nominal variables, or as an effect size
for a chi-square test of association.  For a 2 x 2 table,
the absolute value of the phi statistic is the same as
Cramer's V.
</p>
<p>Because V is always positive, if <code>type="perc"</code>,
the confidence interval will
never cross zero. In this case, 
the confidence interval range should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross zero.  
</p>
<p>When V is close to 0 or very large,
or with small counts, 
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, Cramer's V.  
Or a small data frame consisting of Cramer's V,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_10.html">https://rcompanion.org/handbook/H_10.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phi">phi</a></code>, 
<code><a href="#topic+cohenW">cohenW</a></code>, 
<code><a href="#topic+cramerVFit">cramerVFit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with table
data(Anderson)
fisher.test(Anderson)
cramerV(Anderson)

### Example with two vectors
Species = c(rep("Species1", 16), rep("Species2", 16))
Color   = c(rep(c("blue", "blue", "blue", "green"),4),
            rep(c("green", "green", "green", "blue"),4))
fisher.test(Species, Color)
cramerV(Species, Color)

</code></pre>

<hr>
<h2 id='cramerVFit'>Cramer's V for chi-square goodness-of-fit tests</h2><span id='topic+cramerVFit'></span>

<h3>Description</h3>

<p>Calculates Cramer's V for a vector of counts and expected 
counts; confidence intervals by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cramerVFit(
  x,
  p = rep(1/length(x), length(x)),
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 4,
  reportIncomplete = FALSE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cramerVFit_+3A_x">x</code></td>
<td>
<p>A vector of observed counts.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_p">p</code></td>
<td>
<p>A vector of expected or default probabilities.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints additional statistics.</p>
</td></tr>
<tr><td><code id="cramerVFit_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>chisq.test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This modification of Cramer's V
could be used to indicate an effect size
in cases where a chi-square goodness-of-fit test might be used.
It indicates the degree of deviation of observed counts
from the expected probabilities.
</p>
<p>In the case of equally-distributed expected frequencies,
Cramer's V will be equal to 1 when all counts are in one category,
and it will be equal to 0 when the counts are equally distributed
across categories.
This does not hold if the expected frequencies are not
equally-distributed.
</p>
<p>Because V is always positive,
if <code>type="perc"</code>,
the confidence interval will
never cross zero, and should not
be used for statistical inference. 
However, if <code>type="norm"</code>, the confidence interval
may cross zero.
</p>
<p>When V is close to 0 or 1,
or with small counts, 
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>
<p>In addition, the function will not return a confidence
interval if there are zeros in any cell.
</p>


<h3>Value</h3>

<p>A single statistic, Cramer's V.  
Or a small data frame consisting of Cramer's V,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_03.html">https://rcompanion.org/handbook/H_03.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cramerV">cramerV</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Equal probabilities example
### From https://rcompanion.org/handbook/H_03.html
nail.color = c("Red", "None", "White", "Green", "Purple", "Blue")
observed   = c( 19,    3,      1,       1,       2,        2    )
expected   = c( 1/6,   1/6,    1/6,     1/6,     1/6,      1/6  )
chisq.test(x = observed, p = expected)
cramerVFit(x = observed, p = expected)

### Unequal probabilities example
### From https://rcompanion.org/handbook/H_03.html
race = c("White", "Black", "American Indian", "Asian", "Pacific Islander",
          "Two or more races")
observed = c(20, 9, 9, 1, 1, 1)
expected = c(0.775, 0.132, 0.012, 0.054, 0.002, 0.025)
chisq.test(x = observed, p = expected)
cramerVFit(x = observed, p = expected)

### Examples of perfect and zero fits
cramerVFit(c(100, 0, 0, 0, 0))
cramerVFit(c(10, 10, 10, 10, 10))

</code></pre>

<hr>
<h2 id='efronRSquared'>Efron's pseudo r-squared</h2><span id='topic+efronRSquared'></span>

<h3>Description</h3>

<p>Produces Efron's pseudo r-squared from certain models,
or vectors of
residuals, predicted values, and actual values.
Alternately produces minimum maximum accuracy, 
mean absolute percent error, 
root mean square error, or coefficient of variation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efronRSquared(
  model = NULL,
  actual = NULL,
  predicted = NULL,
  residual = NULL,
  statistic = "EfronRSquared",
  plotit = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="efronRSquared_+3A_model">model</code></td>
<td>
<p>A model of the class lm, glm, nls, betareg, gls,
lme, lmerMod, lmerModLmerTest, glmmTMB,
rq, loess, gam, negbin, glmRob, rlm,
or mblm.</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_actual">actual</code></td>
<td>
<p>A vector of actual y values</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_predicted">predicted</code></td>
<td>
<p>A vector of predicted values</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_residual">residual</code></td>
<td>
<p>A vector of residuals</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_statistic">statistic</code></td>
<td>
<p>The statistic to produce.
One of <code>"EfronRSquared"</code>,
<code>"MinMaxAccuracy"</code>,
<code>"MAE"</code>,
<code>"MAPE"</code>,
<code>"MSE"</code>,
<code>"RMSE"</code>,
<code>"NRMSE.Mean"</code>,
<code>"CV"</code>.</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_plotit">plotit</code></td>
<td>
<p>If <code>TRUE</code>, produces plots of the predicted values
vs. the actual values.</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="efronRSquared_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>plot</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Efron's pseudo r-squared is calculated as 1 minus the residual sum 
of squares divided by the total sum of squares.  For linear models
(<code>lm</code> model objects), Efron's pseudo r-squared will be equal  
to r-squared.
</p>
<p>This function produces the same statistics as does the
<code>accuracy</code> function.
While the <code>accuracy</code> function extracts values from a model
object, this function allows for the manual entry
of residual, predicted, or actual values.
</p>
<p>It is recommended that the user consults the <code>accuracy</code>
function
for further details on these statistics, such as if the reported
value is presented as a percentage or fraction.
</p>
<p>If <code>model</code>is not supplied,
two of the following need to passed to the function:
<code>actual</code>, <code>predicted</code>, <code>residual</code>.
</p>
<p>Note that, for some model objects, to extract residuals
and predicted values on the original scale,
a <code>type="response"</code>
option needs to be added to the call, e.g.
<code>residuals(model.object, type="response")</code>.
</p>


<h3>Value</h3>

<p>A single statistic
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_16.html">https://rcompanion.org/handbook/F_16.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+accuracy">accuracy</a></code>, 
<code><a href="#topic+nagelkerke">nagelkerke</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BrendonSmall)
BrendonSmall$Calories = as.numeric(BrendonSmall$Calories)
BrendonSmall$Calories2 = BrendonSmall$Calories ^ 2
model.1 = lm(Sodium ~ Calories + Calories2, data = BrendonSmall)

efronRSquared(model.1)

efronRSquared(model.1, statistic="MAPE")

efronRSquared(actual=BrendonSmall$Sodium, residual=model.1$residuals)
efronRSquared(residual=model.1$residuals, predicted=model.1$fitted.values)
efronRSquared(actual=BrendonSmall$Sodium, predicted=model.1$fitted.values)

summary(model.1)$r.squared

</code></pre>

<hr>
<h2 id='epsilonSquared'>Epsilon-squared</h2><span id='topic+epsilonSquared'></span>

<h3>Description</h3>

<p>Calculates epsilon-squared 
as an effect size statistic,
following a Kruskal-Wallis test, 
or for a table with one ordinal
variable and one nominal variable; 
confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epsilonSquared(
  x,
  g = NULL,
  group = "row",
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epsilonSquared_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations of an ordinal variable.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_g">g</code></td>
<td>
<p>If <code>x</code> is a vector, <code>g</code> is the vector of observations for
the grouping, nominal variable.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_group">group</code></td>
<td>
<p>If <code>x</code> is a table or matrix, <code>group</code> indicates whether
the <code>"row"</code> or the <code>"column"</code> variable is
the nominal, grouping variable.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="epsilonSquared_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>kruskal.test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Epsilon-squared is used as a measure of association
for the Kruskal-Wallis test or for a two-way
table with one ordinal and one nominal variable.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>Because epsilon-squared is always positive, 
if <code>type="perc"</code>, the confidence interval will
never cross zero, and should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross zero. 
</p>
<p>When epsilon-squared is close to 0 or very large,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, epsilon-squared.
Or a small data frame consisting of epsilon-squared,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>King, B.M., P.J. Rosopa, and E.W. Minium. 2018. 
Statistical Reasoning in the Behavioral Sciences, 7th ed. Wiley.
</p>
<p><a href="https://rcompanion.org/handbook/F_08.html">https://rcompanion.org/handbook/F_08.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+multiVDA">multiVDA</a></code>,
<code><a href="#topic+ordinalEtaSquared">ordinalEtaSquared</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Breakfast)
library(coin)
chisq_test(Breakfast, scores = list("Breakfast" = c(-2, -1, 0, 1, 2)))
epsilonSquared(Breakfast)

data(PoohPiglet)
kruskal.test(Likert ~ Speaker, data = PoohPiglet)
epsilonSquared(x = PoohPiglet$Likert, g = PoohPiglet$Speaker)

### Same data, as matrix of counts
data(PoohPiglet)
XT = xtabs( ~ Speaker + Likert , data = PoohPiglet)
epsilonSquared(XT)

</code></pre>

<hr>
<h2 id='freemanTheta'>Freeman's theta</h2><span id='topic+freemanTheta'></span>

<h3>Description</h3>

<p>Calculates Freeman's theta for a table with one ordinal
variable and one nominal variable; confidence intervals
by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freemanTheta(
  x,
  g = NULL,
  group = "row",
  verbose = FALSE,
  progress = FALSE,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freemanTheta_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations of an ordinal variable.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_g">g</code></td>
<td>
<p>If <code>x</code> is a vector, <code>g</code> is the vector of observations for
the grouping, nominal variable.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_group">group</code></td>
<td>
<p>If <code>x</code> is a table or matrix, <code>group</code> indicates whether
the <code>"row"</code> or the <code>"column"</code> variable is
the nominal, grouping variable.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints statistics for each
comparison.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_progress">progress</code></td>
<td>
<p>If <code>TRUE</code>, prints a message as each comparison is
conducted.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="freemanTheta_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Freeman's coefficent of differentiation (theta)
is used as a measure of association
for a two-way
table with one ordinal and one nominal variable.
See Freeman (1965).
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>Because theta is always positive, if <code>type="perc"</code>,
the confidence interval will
never cross zero, and should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross zero.
</p>
<p>When theta is close to 0 or very large,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, Freeman's theta.
Or a small data frame consisting of Freeman's theta,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>Freeman, L.C. 1965. Elementary Applied Statistics for Students
in Behavioral Science. Wiley.
</p>
<p><a href="https://rcompanion.org/handbook/H_11.html">https://rcompanion.org/handbook/H_11.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+epsilonSquared">epsilonSquared</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Breakfast)
library(coin)
chisq_test(Breakfast, scores = list("Breakfast" = c(-2, -1, 0, 1, 2)))
freemanTheta(Breakfast)

### Example from Freeman (1965), Table 10.6
Counts = c(1, 2, 5, 2, 0, 10, 5, 5, 0, 0, 0, 0, 2, 2, 1, 0, 0, 0, 2, 3)
Matrix = matrix(Counts, byrow=TRUE, ncol=5,
                dimnames = list(Marital.status = c("Single", "Married",
                                                   "Widowed", "Divorced"),
                                Social.adjustment = c("5","4","3","2","1")))
Matrix
freemanTheta(Matrix)

### Example after Kruskal Wallis test
data(PoohPiglet)
kruskal.test(Likert ~ Speaker, data = PoohPiglet)
freemanTheta(x = PoohPiglet$Likert, g = PoohPiglet$Speaker)

### Same data, as table of counts
data(PoohPiglet)
XT = xtabs( ~ Speaker + Likert , data = PoohPiglet)
freemanTheta(XT)

### Example from Freeman (1965), Table 10.7
Counts = c(52, 28, 40, 34, 7, 9, 16, 10, 8, 4, 10, 9, 12,6, 7, 5)
Matrix = matrix(Counts, byrow=TRUE, ncol=4,
                dimnames = list(Preferred.trait = c("Companionability",
                                                    "PhysicalAppearance",
                                                    "SocialGrace",
                                                    "Intelligence"),
                                Family.income = c("4", "3", "2", "1")))
Matrix
freemanTheta(Matrix, verbose=TRUE)

</code></pre>

<hr>
<h2 id='fullPTable'>Convert a lower triangle matrix to a full matrix</h2><span id='topic+fullPTable'></span>

<h3>Description</h3>

<p>Converts a lower triangle matrix to a full matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fullPTable(PT)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fullPTable_+3A_pt">PT</code></td>
<td>
<p>A lower triangle matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is useful to convert a lower triangle matrix
of p-values from a pairwise test to a full matrix.
A full matrix can be passed to <code>multcompLetters</code>
in the <code>multcompView</code> package to produce a compact
letter display.
</p>


<h3>Value</h3>

<p>A full matrix.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_08.html">https://rcompanion.org/handbook/F_08.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cldList">cldList</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with pairwise.wilcox.test
data(BrendonSmall)
BrendonSmall$Instructor = factor(BrendonSmall$Instructor,
                          levels = c('Brendon Small', 'Jason Penopolis',
                                     'Paula Small', 'Melissa Robbins', 
                                     'Coach McGuirk'))
P   = pairwise.wilcox.test(x = BrendonSmall$Score, g = BrendonSmall$Instructor)
PT  = P$p.value
PT
PT1 = fullPTable(PT)
PT1
library(multcompView)
multcompLetters(PT1)

</code></pre>

<hr>
<h2 id='groupwiseCMH'>Post-hoc tests for Cochran-Mantel-Haenszel test</h2><span id='topic+groupwiseCMH'></span>

<h3>Description</h3>

<p>Conducts groupwise tests of association on a three-way 
contingency table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwiseCMH(
  x,
  group = 3,
  fisher = TRUE,
  gtest = FALSE,
  chisq = FALSE,
  method = "fdr",
  correct = "none",
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwiseCMH_+3A_x">x</code></td>
<td>
<p>A three-way contingency table.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_group">group</code></td>
<td>
<p>The dimension of the table to use as the grouping variable.
Will be <code>1</code>, <code>2</code>, or <code>3</code>.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_fisher">fisher</code></td>
<td>
<p>If <code>TRUE</code>, conducts Fisher exact test.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_gtest">gtest</code></td>
<td>
<p>If <code>TRUE</code>, conducts G test of association.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_chisq">chisq</code></td>
<td>
<p>If <code>TRUE</code>, conducts Chi-square test of association.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_method">method</code></td>
<td>
<p>The method to use to adjust p-values.  See <code>?p.adjust</code>.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_correct">correct</code></td>
<td>
<p>The correction to apply to the G test. 
See <code>GTest</code>.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_digits">digits</code></td>
<td>
<p>The number of digits for numbers in the output.</p>
</td></tr>
<tr><td><code id="groupwiseCMH_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>chisq.test</code> or <code>GTest</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If more than one of <code>fisher</code>, <code>gtest</code>, or <code>chisq</code> is
set to <code>TRUE</code>, only one type of test of association
will be conducted.
</p>


<h3>Value</h3>

<p>A data frame of groups, test used, p-values, and adjusted p-values.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_06.html">https://rcompanion.org/handbook/H_06.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nominalSymmetryTest">nominalSymmetryTest</a></code>, 
<code><a href="#topic+pairwiseMcnemar">pairwiseMcnemar</a></code>,
<code><a href="#topic+pairwiseNominalIndependence">pairwiseNominalIndependence</a></code>, 
<code><a href="#topic+pairwiseNominalMatrix">pairwiseNominalMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Post-hoc for Cochran-Mantel-Haenszel test
data(AndersonBias)
Table = xtabs(Count ~ Gender + Result + County,
              data=AndersonBias)
ftable(Table)
mantelhaen.test(Table)
groupwiseCMH(Table,
             group   = 3,
             fisher  = TRUE,
             gtest   = FALSE,
             chisq   = FALSE,
             method  = "fdr",
             correct = "none",
             digits  = 3)
                      
</code></pre>

<hr>
<h2 id='groupwiseGeometric'>Groupwise geometric means and confidence intervals</h2><span id='topic+groupwiseGeometric'></span>

<h3>Description</h3>

<p>Calculates geometric means and confidence intervals for
groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwiseGeometric(
  formula = NULL,
  data = NULL,
  var = NULL,
  group = NULL,
  conf = 0.95,
  na.rm = TRUE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwiseGeometric_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_var">var</code></td>
<td>
<p>The measurement variable to use. The name is in double quotes.</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_group">group</code></td>
<td>
<p>The grouping variable to use. The name is in double quotes.
Multiple names are listed as a vector. (See example.)</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_conf">conf</code></td>
<td>
<p>The confidence interval to use.</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>TRUE</code>, removes NA values in the measurement variable.</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.</p>
</td></tr>
<tr><td><code id="groupwiseGeometric_+3A_...">...</code></td>
<td>
<p>Other arguments.  Not currently used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>data</code>, <code>var</code>, and <code>group</code>. (See examples).
</p>
<p>The function computes means, standard deviations, standard errors, 
and confidence intervals on log-transformed values. Confidence
intervals are calculated in the traditional
manner with the t-distribution on the transformed values,
and then back-transforms the confidence interval limits.  
These statistics assume that
the data are log-normally distributed. For data not meeting this
assumption, medians and confidence intervals by bootstrap may be more 
appropriate.
</p>


<h3>Value</h3>

<p>A data frame of geometric means, standard deviations, standard 
errors, and confidence intervals.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. The first variable on the
left side is used as the measurement variable.  The variables on the
right side are used for the grouping variables.
</p>
<p>Results for ungrouped (one-sample) data can be obtained by either
setting the right side of the formula to 1, e.g.  y ~ 1, or by
setting <code>group=NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/C_03.html">https://rcompanion.org/handbook/C_03.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwiseMean">groupwiseMean</a></code>, 
<code><a href="#topic+groupwiseMedian">groupwiseMedian</a></code>,
<code><a href="#topic+groupwiseHuber">groupwiseHuber</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with formula notation 
data(Catbus)
groupwiseGeometric(Steps ~ Gender + Teacher,
                   data   = Catbus)

### Example with variable notation                                              
data(Catbus)
groupwiseGeometric(data   = Catbus,
                   var    = "Steps",
                   group  = c("Gender", "Teacher"))
                      
</code></pre>

<hr>
<h2 id='groupwiseHuber'>Groupwise Huber M-estimators and confidence intervals</h2><span id='topic+groupwiseHuber'></span>

<h3>Description</h3>

<p>Calculates Huber M-estimator and confidence intervals for
groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwiseHuber(
  formula = NULL,
  data = NULL,
  var = NULL,
  group = NULL,
  conf.level = 0.95,
  ci.type = "wald",
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwiseHuber_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_var">var</code></td>
<td>
<p>The measurement variable to use. The name is in double quotes.</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_group">group</code></td>
<td>
<p>The grouping variable to use. The name is in double quotes.
Multiple names are listed as a vector. (See example.)</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_conf.level">conf.level</code></td>
<td>
<p>The confidence interval to use.</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_ci.type">ci.type</code></td>
<td>
<p>The type of confidence interval to use. Can be
<code>"wald"</code> or <code>"boot"</code>.
See <code>HuberM</code> for details.</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.</p>
</td></tr>
<tr><td><code id="groupwiseHuber_+3A_...">...</code></td>
<td>
<p>Other arguments passed to the <code>HuberM</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A wrapper for the <code>DescTools::HuberM</code> function
to allow easy output for multiple groups.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>data</code>, <code>var</code>, and <code>group</code>. (See examples).
</p>
<p>Results for ungrouped (one-sample) data can be obtained by either
setting the right side of the formula to 1, e.g.  y ~ 1, or by
setting <code>group=NULL</code>.
</p>


<h3>Value</h3>

<p>A data frame of requested statistics by group.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. The first variable on the
left side is used as the measurement variable.  The variables on the
right side are used for the grouping variables.          
</p>
<p>It is recommended to remove <code>NA</code> values before using this
function.  At the time of writing, <code>NA</code> values will cause the
function to fail if confidence intervals are requested.
</p>
<p>At the time of writing, the <code>ci.type="boot"</code> option
produces <code>NA</code> results. This is a result from the
<code>DescTools::HuberM</code> function.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/rcompanion/d_08a.html">https://rcompanion.org/rcompanion/d_08a.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwiseMean">groupwiseMean</a></code>, 
<code><a href="#topic+groupwiseMedian">groupwiseMedian</a></code>,
<code><a href="#topic+groupwiseGeometric">groupwiseGeometric</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with formula notation
data(Catbus)
groupwiseHuber(Steps ~ Teacher + Gender,
               data      = Catbus,
               ci.type   = "wald")
               
### Example with variable notation
data(Catbus)
groupwiseHuber(data      = Catbus,
               var       = "Steps",
               group     = c("Teacher", "Gender"),
               ci.type   = "wald")

### Example with NA value and without confidence intervals
data(Catbus)
Catbus1 = Catbus
Catbus1[1, 'Steps'] = NA
groupwiseHuber(Steps ~ Teacher + Gender,
               data      = Catbus1,
               conf.level   = NA)
                                      
</code></pre>

<hr>
<h2 id='groupwiseMean'>Groupwise means and confidence intervals</h2><span id='topic+groupwiseMean'></span>

<h3>Description</h3>

<p>Calculates means and confidence intervals for
groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwiseMean(
  formula = NULL,
  data = NULL,
  var = NULL,
  group = NULL,
  trim = 0,
  na.rm = FALSE,
  conf = 0.95,
  R = 5000,
  boot = FALSE,
  traditional = TRUE,
  normal = FALSE,
  basic = FALSE,
  percentile = FALSE,
  bca = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwiseMean_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_var">var</code></td>
<td>
<p>The measurement variable to use. The name is in double quotes.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_group">group</code></td>
<td>
<p>The grouping variable to use. The name is in double quotes.
Multiple names are listed as a vector. (See example.)</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_trim">trim</code></td>
<td>
<p>The proportion of observations trimmed from each end of the
values before the mean is calculated. (As in <code>mean()</code>)</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>TRUE</code>, <code>NA</code> values are removed during 
calculations. (As in <code>mean()</code>)</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_conf">conf</code></td>
<td>
<p>The confidence interval to use.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_r">R</code></td>
<td>
<p>The number of bootstrap replicates to use for bootstrapped
statistics.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_boot">boot</code></td>
<td>
<p>If <code>TRUE</code>, includes the mean of the bootstrapped means.
This can be used as an estimate of the mean for
the group.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_traditional">traditional</code></td>
<td>
<p>If <code>TRUE</code>, includes the traditional confidence
intervals for the group means, using the t-distribution.
If <code>trim</code> is not 0, 
the traditional confidence interval
will produce <code>NA</code>.
Likewise, if there are <code>NA</code> values that are not
removed, the traditional confidence interval
will produce <code>NA</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_normal">normal</code></td>
<td>
<p>If <code>TRUE</code>, includes the normal confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_basic">basic</code></td>
<td>
<p>If <code>TRUE</code>, includes the basic confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_percentile">percentile</code></td>
<td>
<p>If <code>TRUE</code>, includes the percentile confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_bca">bca</code></td>
<td>
<p>If <code>TRUE</code>, includes the BCa confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.</p>
</td></tr>
<tr><td><code id="groupwiseMean_+3A_...">...</code></td>
<td>
<p>Other arguments passed to the <code>boot</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>data</code>, <code>var</code>, and <code>group</code>. (See examples).
</p>
<p>Results for ungrouped (one-sample) data can be obtained by either
setting the right side of the formula to 1, e.g.  y ~ 1, or by
setting <code>group=NULL</code> when using <code>var</code>.
</p>


<h3>Value</h3>

<p>A data frame of requested statistics by group.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. The first variable on the
left side is used as the measurement variable.  The variables on the
right side are used for the grouping variables.
</p>
<p>In general, it is advisable to handle <code>NA</code> values before
using this function.
With some options, the function may not handle missing values well,
or in the manner desired by the user.
In particular, if <code>bca=TRUE</code> and there are <code>NA</code> values,
the function may fail.
</p>
<p>For a traditional method to calculate confidence intervals 
on trimmed means,
see Rand Wilcox, Introduction to Robust Estimation and
Hypothesis Testing.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/C_03.html">https://rcompanion.org/handbook/C_03.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwiseMedian">groupwiseMedian</a></code>, 
<code><a href="#topic+groupwiseHuber">groupwiseHuber</a></code>,
<code><a href="#topic+groupwiseGeometric">groupwiseGeometric</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with formula notation
data(Catbus)
groupwiseMean(Steps ~ Teacher + Gender,
              data        = Catbus,
              traditional = FALSE,
              percentile  = TRUE)

### Example with variable notation
data(Catbus)
groupwiseMean(data        = Catbus,
              var         = "Steps",
              group       = c("Teacher", "Gender"),
              traditional = FALSE,
              percentile  = TRUE)
                                     
</code></pre>

<hr>
<h2 id='groupwiseMedian'>Groupwise medians and confidence intervals</h2><span id='topic+groupwiseMedian'></span>

<h3>Description</h3>

<p>Calculates medians and confidence intervals for
groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwiseMedian(
  formula = NULL,
  data = NULL,
  var = NULL,
  group = NULL,
  conf = 0.95,
  R = 5000,
  boot = FALSE,
  pseudo = FALSE,
  basic = FALSE,
  normal = FALSE,
  percentile = FALSE,
  bca = TRUE,
  wilcox = FALSE,
  exact = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwiseMedian_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_var">var</code></td>
<td>
<p>The measurement variable to use. The name is in double quotes.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_group">group</code></td>
<td>
<p>The grouping variable to use. The name is in double quotes.
Multiple names are listed as a vector. (See example.)</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_conf">conf</code></td>
<td>
<p>The confidence interval to use.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_r">R</code></td>
<td>
<p>The number of bootstrap replicates to use for bootstrapped
statistics.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_boot">boot</code></td>
<td>
<p>If <code>TRUE</code>, includes the mean of the bootstrapped medians.
This can be used as an estimate of the median for
the group.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_pseudo">pseudo</code></td>
<td>
<p>If <code>TRUE</code>, includes the pseudo median from
<code>wilcox.test</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_basic">basic</code></td>
<td>
<p>If <code>TRUE</code>, includes the basic confidence
intervals for the group means by bootstrap.
See <code>boot::boot.ci</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_normal">normal</code></td>
<td>
<p>If <code>TRUE</code>, includes the normal confidence
intervals for the group means by bootstrap.
See <code>boot::boot.ci</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_percentile">percentile</code></td>
<td>
<p>If <code>TRUE</code>, includes the percentile confidence
intervals for the group means by bootstrap.
See <code>boot::boot.ci</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_bca">bca</code></td>
<td>
<p>If <code>TRUE</code>, includes the BCa confidence
intervals for the group means by bootstrap.
See <code>boot::boot.ci</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_wilcox">wilcox</code></td>
<td>
<p>If <code>TRUE</code>, includes the wilcox confidence
intervals from  <code>stats::wilcox.test</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_exact">exact</code></td>
<td>
<p>If <code>TRUE</code>, includes the &quot;exact&quot; confidence
intervals from  <code>DescTools::MedianCI</code>.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.</p>
</td></tr>
<tr><td><code id="groupwiseMedian_+3A_...">...</code></td>
<td>
<p>Other arguments passed to the <code>boot</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>data</code>, <code>var</code>, and <code>group</code>. (See examples).
</p>
<p>With some options, the function may not handle missing values well.
This seems to happen particularly with <code>bca = TRUE</code>.
</p>


<h3>Value</h3>

<p>A data frame of requested statistics by group.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. The first variable on the
left side is used as the measurement variable.  The variables on the
right side are used for the grouping variables.
</p>
<p>Results for ungrouped (one-sample) data can be obtained by either
setting the right side of the formula to 1, e.g.  y ~ 1, or by
setting <code>group=NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/E_04.html">https://rcompanion.org/handbook/E_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwiseMean">groupwiseMean</a></code>, 
<code><a href="#topic+groupwiseHuber">groupwiseHuber</a></code>,
<code><a href="#topic+groupwiseGeometric">groupwiseGeometric</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with formula notation
data(Catbus)
groupwiseMedian(Steps ~ Teacher + Gender,
                data        = Catbus,
                bca         = FALSE,
                percentile  = TRUE,
                R           = 1000)
                
### Example with variable notation
data(Catbus)
groupwiseMedian(data         = Catbus,
                var         = "Steps",
                group       = c("Teacher", "Gender"),
                bca         = FALSE,
                percentile  = TRUE,
                R           = 1000)
                      
</code></pre>

<hr>
<h2 id='groupwisePercentile'>Groupwise percentiles and confidence intervals</h2><span id='topic+groupwisePercentile'></span>

<h3>Description</h3>

<p>Calculates percentiles and confidence intervals for
groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwisePercentile(
  formula = NULL,
  data = NULL,
  var = NULL,
  group = NULL,
  conf = 0.95,
  tau = 0.5,
  type = 7,
  R = 5000,
  boot = FALSE,
  basic = FALSE,
  normal = FALSE,
  percentile = FALSE,
  bca = TRUE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwisePercentile_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_var">var</code></td>
<td>
<p>If no formula is given, the measurement variable to use. 
The name is in double quotes.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_group">group</code></td>
<td>
<p>The grouping variable to use. The name is in double quotes.
Multiple names are listed as a vector. (See example.)</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_conf">conf</code></td>
<td>
<p>The confidence interval to use.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_tau">tau</code></td>
<td>
<p>The percentile to use, expressed as a quantile,
e.g. 0.5 for median, 0.25 for 25th percentile.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_type">type</code></td>
<td>
<p>The <code>type</code> value passed to the <code>quantile</code> function.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_r">R</code></td>
<td>
<p>The number of bootstrap replicates to use for bootstrapped
statistics.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_boot">boot</code></td>
<td>
<p>If <code>TRUE</code>, includes the mean of the bootstrapped percentile.
This can be used as an estimate of the percentile for
the group.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_basic">basic</code></td>
<td>
<p>If <code>TRUE</code>, includes the basic confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_normal">normal</code></td>
<td>
<p>If <code>TRUE</code>, includes the normal confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_percentile">percentile</code></td>
<td>
<p>If <code>TRUE</code>, includes the percentile confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_bca">bca</code></td>
<td>
<p>If <code>TRUE</code>, includes the BCa confidence
intervals for the group means by bootstrap.
See <code><a href="boot.html#topic+boot.ci">boot.ci</a></code>.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.</p>
</td></tr>
<tr><td><code id="groupwisePercentile_+3A_...">...</code></td>
<td>
<p>Other arguments passed to the <code>boot</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>data</code>, <code>var</code>, and <code>group</code>. (See examples).
</p>
<p>With some options, the function may not handle missing values well.
This seems to happen particularly with <code>bca = TRUE</code>.
</p>


<h3>Value</h3>

<p>A data frame of requested statistics by group
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. The first variable on the
left side is used as the measurement variable.  The variables on the
right side are used for the grouping variables.
</p>
<p>Results for ungrouped (one-sample) data can be obtained by either
setting the right side of the formula to 1, e.g.  y ~ 1, or by
setting <code>group=NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_15.html">https://rcompanion.org/handbook/F_15.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwiseMean">groupwiseMean</a></code>, 
<code><a href="#topic+groupwiseHuber">groupwiseHuber</a></code>, 
<code><a href="#topic+groupwiseGeometric">groupwiseGeometric</a></code>, 
<code><a href="#topic+groupwiseMedian">groupwiseMedian</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with formula notation
data(Catbus)
groupwisePercentile(Steps ~ Teacher + Gender,
                    data        = Catbus,
                    tau         = 0.25,
                    bca         = FALSE,
                    percentile  = TRUE,
                    R           = 1000)
                
### Example with variable notation
data(Catbus)
groupwisePercentile(data         = Catbus,
                    var         = "Steps",
                    group       = c("Teacher", "Gender"),
                    tau         = 0.25,
                    bca         = FALSE,
                    percentile  = TRUE,
                    R           = 1000)
                      
</code></pre>

<hr>
<h2 id='groupwiseSum'>Groupwise sums</h2><span id='topic+groupwiseSum'></span>

<h3>Description</h3>

<p>Calculates sums for groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupwiseSum(
  formula = NULL,
  data = NULL,
  var = NULL,
  group = NULL,
  digits = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupwiseSum_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="groupwiseSum_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="groupwiseSum_+3A_var">var</code></td>
<td>
<p>The measurement variable to use. The name is in double quotes.</p>
</td></tr>
<tr><td><code id="groupwiseSum_+3A_group">group</code></td>
<td>
<p>The grouping variable to use. The name is in double quotes.
Multiple names are listed as a vector. (See example.)</p>
</td></tr>
<tr><td><code id="groupwiseSum_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.
The default is <code>NULL</code>, which results in no
rounding of values.</p>
</td></tr>
<tr><td><code id="groupwiseSum_+3A_...">...</code></td>
<td>
<p>Other arguments passed to the <code>sum</code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>data</code>, <code>var</code>, and <code>group</code>. (See examples).
</p>


<h3>Value</h3>

<p>A data frame of statistics by group.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. The first variable on the
left side is used as the measurement variable.  The variables on the
right side are used for the grouping variables.
</p>
<p>Beginning in version 2.0, there is no rounding of results by
default.  Rounding results can cause confusion if the
user is expecting exact sums.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwiseMean">groupwiseMean</a></code>, 
<code><a href="#topic+groupwiseMedian">groupwiseMedian</a></code>,
<code><a href="#topic+groupwiseHuber">groupwiseHuber</a></code>,
<code><a href="#topic+groupwiseGeometric">groupwiseGeometric</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with formula notation
data(AndersonBias)
groupwiseSum(Count ~ Result + Gender,
             data        = AndersonBias)
                
### Example with variable notation
data(AndersonBias)
groupwiseSum(data        = AndersonBias,
             var         = "Count",
             group       = c("Result", "Gender"))
                      
</code></pre>

<hr>
<h2 id='HayleySmith'>Hypothetical data for responses about adopting lawn care practices</h2><span id='topic+HayleySmith'></span>

<h3>Description</h3>

<p>A data frame in long form with yes/no responses for four
lawn care practices for each of 14 respondents.
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HayleySmith
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 56 rows and 3 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_05.html">https://rcompanion.org/handbook/H_05.html</a>
</p>

<hr>
<h2 id='kendallW'>Kendall's W with bootstrapped confidence interval</h2><span id='topic+kendallW'></span>

<h3>Description</h3>

<p>Calculates Kendall's W coefficient of concordance,
which can be used as an effect size statistic for
unreplicated complete block design
such as where Friedman's test might be used.
This function is a wrapper for the <code>KendallW</code>
function in the <code>DescTools</code> package,
with the addition of bootstrapped
confidence intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kendallW(
  x,
  correct = TRUE,
  na.rm = FALSE,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kendallW_+3A_x">x</code></td>
<td>
<p>A k x m matrix or table, with k treatments in rows 
and m raters or blocks in columns.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_correct">correct</code></td>
<td>
<p>Passed to <code>KendallW</code>.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_na.rm">na.rm</code></td>
<td>
<p>Passed to <code>KendallW</code>.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="kendallW_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>KendallW</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the <code>KendallW</code> function in the <code>DescTools</code> package
for details.
</p>
<p>When W is close to 0 or very large,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>
<p>Because W is always positive, if <code>type="perc"</code>,
the confidence interval will
never cross zero, and should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross zero. 
</p>
<p>When producing confidence intervals by bootstrap, 
this function treats each rater or block as an observation.
It is not clear to the author if this approach produces accurate
confidence intervals, but it appears to be reasonable.
</p>


<h3>Value</h3>

<p>A single statistic, W.  
Or a small data frame consisting of W,
and the lower and upper confidence limits.
</p>


<h3>Acknowledgments</h3>

<p>My thanks to Indrajeet Patil, author of <code>ggstatsplot</code>,
and <code>groupedstats</code> for help in the inspiring and 
coding of this function.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_10.html">https://rcompanion.org/handbook/F_10.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BobBelcher)
Table = xtabs(Likert ~ Instructor + Rater, data = BobBelcher)
kendallW(Table)

</code></pre>

<hr>
<h2 id='mangiaficoD'>Mangiafico's d</h2><span id='topic+mangiaficoD'></span>

<h3>Description</h3>

<p>Calculates Mangiafico's d, which is the difference in medians
divided by the pooled median absolute deviation,
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mangiaficoD(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  reportIncomplete = FALSE,
  verbose = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mangiaficoD_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, reports the median difference and MAD.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="mangiaficoD_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>mad()</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mangiafico's d is an appropriate effect size statistic where
Mood's median test, or another test comparing two medians,
might be used.  Note that the response variable is treated
as at least interval.
</p>
<p>For normal samples, the result will be somewhat similar to
Cohen's d.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>. If there are more than two groups,
only the first two groups are used.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, d is positive.
When the data in the second group are greater than
in the first group, d is negative.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>When d is close to 0 or close to 1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, d.
Or a small data frame consisting of d,
and the lower and upper confidence limits.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_05.html">https://rcompanion.org/handbook/F_05.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+multiMangiaficoD">multiMangiaficoD</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
mangiaficoD(Steps ~ Gender, data=Catbus, verbose=TRUE)

Nadja = c(5,5,6,6,6,7,7,11,11,11)
Nandor = c(0,1,2,3,4,5,6,7,8,9,10,11)
mangiaficoD(x = Nadja, y = Nandor, verbose=TRUE)

</code></pre>

<hr>
<h2 id='Monarchs'>Hypothetical data for monarch butterflies in gardens</h2><span id='topic+Monarchs'></span>

<h3>Description</h3>

<p>A data frame of the number of monarch butterflies in three
gardens. Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Monarchs
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 24 rows and 2 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/J_01.html">https://rcompanion.org/handbook/J_01.html</a>
</p>

<hr>
<h2 id='multiMangiaficoD'>Mangiafico's d</h2><span id='topic+multiMangiaficoD'></span>

<h3>Description</h3>

<p>Calculates Mangiafico's d,
which is the difference in medians
divided by the pooled median absolute deviation,
for several groups in a pairwise manner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiMangiaficoD(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiMangiaficoD_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="multiMangiaficoD_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="multiMangiaficoD_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable.</p>
</td></tr>
<tr><td><code id="multiMangiaficoD_+3A_g">g</code></td>
<td>
<p>If no formula is given, the grouping variable.</p>
</td></tr>
<tr><td><code id="multiMangiaficoD_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="multiMangiaficoD_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>mad()</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mangiafico's d is an appropriate effect size statistic where
Mood's median test, or another test comparing two medians,
might be used.  Note that the response variable is treated
as at least interval.
</p>
<p>When the data in the first group are greater than
in the second group, d is positive.
When the data in the second group are greater than
in the first group, d is negative.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>


<h3>Value</h3>

<p>A list containing a data frame of pairwise statistics,
and the comparison with the most extreme value
of the statistic.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>
<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_09.html">https://rcompanion.org/handbook/F_09.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mangiaficoD">mangiaficoD</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
multiMangiaficoD(Steps ~ Teacher, data=Catbus)

</code></pre>

<hr>
<h2 id='multiVDA'>Pairwise Vargha and Delaney's A and Cliff's delta</h2><span id='topic+multiVDA'></span>

<h3>Description</h3>

<p>Calculates Vargha and Delaney's A (VDA),
Cliff's delta (CD), 
and the Glass rank biserial coefficient, rg,
for several groups in a pairwise manner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiVDA(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  statistic = "VDA",
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiVDA_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="multiVDA_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="multiVDA_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable.</p>
</td></tr>
<tr><td><code id="multiVDA_+3A_g">g</code></td>
<td>
<p>If no formula is given, the grouping variable.</p>
</td></tr>
<tr><td><code id="multiVDA_+3A_statistic">statistic</code></td>
<td>
<p>One of <code>"VDA"</code>, <code>"CD"</code>, or <code>"rg"</code>.
This determines which statistic will be
evaluated to determine the comparison with the
most divergent groups.</p>
</td></tr>
<tr><td><code id="multiVDA_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="multiVDA_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcox.test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VDA and CD are effect size statistic appropriate
in cases where a Wilcoxon-Mann-Whitney test might be used.
Here, the pairwise approach would be used in cases where
a Kruskal-Wallis test might be used.
VDA ranges from 0 to 1, 
with 0.5 indicating stochastic equality,
and 1 indicating that the first group dominates the second.
CD ranges from -1 to 1, with 0 indicating stochastic equality,
and 1 indicating that the first group dominates the second.
rg ranges from -1 to 1, 
depending on sample size,
with 0 indicating no effect,
and a positive result indicating
that values in the first group are greater than in the second.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>In the function output,
<code>VDA.m</code> is the greater of VDA or 1-VDA.
<code>CD.m</code> is the absolute value of CD.
<code>rg.m</code> is the absolute value of rg.
</p>
<p>The function calculates VDA and Cliff's delta from the &quot;W&quot; 
U statistic from the
<code>wilcox.test</code> function.
Specifically, <code>VDA = U/(n1*n2); CD = (VDA-0.5)*2</code>.
</p>
<p>rg  is calculated as 2 times the difference of mean of ranks
for each group divided by the total sample size.
It appears that rg is equivalent to Cliff's delta.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>var</code>, and <code>group</code>.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>


<h3>Value</h3>

<p>A list containing a data frame of pairwise statistics,
and the comparison with the most extreme value
of the chosen statistic.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_08.html">https://rcompanion.org/handbook/F_08.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vda">vda</a></code>, 
<code><a href="#topic+cliffDelta">cliffDelta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PoohPiglet)
multiVDA(Likert ~ Speaker, data=PoohPiglet)

</code></pre>

<hr>
<h2 id='nagelkerke'>Pseudo r-squared measures for various models</h2><span id='topic+nagelkerke'></span>

<h3>Description</h3>

<p>Produces McFadden, Cox and Snell, and Nagelkerke pseudo 
r-squared measures, along with p-values, for models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nagelkerke(fit, null = NULL, restrictNobs = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nagelkerke_+3A_fit">fit</code></td>
<td>
<p>The fitted model object for which to determine pseudo r-squared.</p>
</td></tr>
<tr><td><code id="nagelkerke_+3A_null">null</code></td>
<td>
<p>The null model object against which to compare the fitted model 
object. The null model must be nested in the fitted model to be 
valid. Specifying the null
is optional for some model object types
and is required for others.</p>
</td></tr>
<tr><td><code id="nagelkerke_+3A_restrictnobs">restrictNobs</code></td>
<td>
<p>If <code>TRUE</code>, limits the observations for the null
model to those used in the fitted model.
Works with only some model object types.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pseudo R-squared values are not directly comparable to the  
R-squared for OLS models.  Nor can they be interpreted as the  
proportion of the variability in the dependent variable that is  
explained by model. Instead pseudo R-squared measures are relative
measures among similar models indicating how well the model
explains the data.
</p>
<p>Cox and Snell is also referred to as ML. Nagelkerke is also  
referred to as Cragg and Uhler.
</p>
<p>Model objects accepted are lm, glm, gls, lme, lmer, lmerTest, nls,
clm, clmm, vglm, glmer, glmmTMB, negbin, zeroinfl, betareg, and rq.            
</p>
<p>Model objects that require the null model to 
be defined are nls, lmer, glmer, and clmm. 
Other objects use the <code>update</code> function to
define the null model.
</p>
<p>Likelihoods are found using ML (<code>REML = FALSE</code>).
</p>
<p>The fitted model and the null model
should be properly nested.
That is, the terms of one need to be a subset of the the other,
and they should have the same set of observations.
One issue arises when there are <code>NA</code>
values in one variable but not another, and observations with 
<code>NA</code> are removed in the model fitting.  The result may be
fitted and null models with
different sets of observations.
Setting <code>restrictNobs</code> to <code>TRUE</code> 
ensures that only observations in
the fit model are used in the null model.
This appears to work for <code>lm</code> and some <code>glm</code> models,
but causes the function to fail for other model
object types.
</p>
<p>Some pseudo R-squared measures may not be appropriate
or useful for some model types.
</p>
<p>Calculations are based on log likelihood values for models.
Results may be different than those based on deviance.
</p>


<h3>Value</h3>

<p>A list of six objects describing the models used, the pseudo 
r-squared values, the likelihood ratio test for the model,
the number of observations for the models,
messages, and any warnings.
</p>


<h3>Acknowledgments</h3>

<p>My thanks to
Jan-Herman Kuiper of Keele University for suggesting
the <code>restrictNobs</code> fix.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/G_10.html">https://rcompanion.org/handbook/G_10.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+efronRSquared">efronRSquared</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Logistic regression example
data(AndersonBias)
model = glm(Result ~ County + Gender + County:Gender,
           weight = Count,
           data = AndersonBias,
           family = binomial(link="logit"))
nagelkerke(model)

### Quadratic plateau example 
### With nls, the  null needs to be defined
data(BrendonSmall)
quadplat = function(x, a, b, clx) {
          ifelse(x  &lt; clx, a + b * x   + (-0.5*b/clx) * x   * x,
                           a + b * clx + (-0.5*b/clx) * clx * clx)}
model = nls(Sodium ~ quadplat(Calories, a, b, clx),
            data = BrendonSmall,
            start = list(a   = 519,
                         b   = 0.359,
                         clx = 2304))
nullfunct = function(x, m){m}
null.model = nls(Sodium ~ nullfunct(Calories, m),
             data = BrendonSmall,
             start = list(m   = 1346))
nagelkerke(model, null=null.model)

</code></pre>

<hr>
<h2 id='nagelkerkeHermite'>[Defunct!] Pseudo r-squared measures for hermite models</h2><span id='topic+nagelkerkeHermite'></span>

<h3>Description</h3>

<p>Defunct. Produces McFadden, Cox and Snell, and Nagelkerke pseudo 
R-squared measures, along with p-value for the model, 
for hermite regression objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nagelkerkeHermite(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nagelkerkeHermite_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='nominalSymmetryTest'>Exact and McNemar symmetry tests for paired contingency tables</h2><span id='topic+nominalSymmetryTest'></span>

<h3>Description</h3>

<p>Conducts an omnibus symmetry test for a paired 
contingency table and then post-hoc pairwise tests.  
This is similar to  
McNemar and McNemar-Bowker tests in use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nominalSymmetryTest(x, method = "fdr", digits = 3, exact = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nominalSymmetryTest_+3A_x">x</code></td>
<td>
<p>A two-way contingency table. It must be square. 
It can have two or
more levels for each dimension.</p>
</td></tr>
<tr><td><code id="nominalSymmetryTest_+3A_method">method</code></td>
<td>
<p>The method to adjust multiple p-values. 
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="nominalSymmetryTest_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="nominalSymmetryTest_+3A_exact">exact</code></td>
<td>
<p>If <code>TRUE</code>, uses the <code>binom.test</code> function.
If <code>FALSE</code>, uses the <code>mcnemar.test</code> function.</p>
</td></tr>
<tr><td><code id="nominalSymmetryTest_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The omnibus McNemar test
may fail when there are zeros in critical
cells.
</p>
<p>Currently, the <code>exact=TRUE</code> with a table greater
than 2 x 2 will not produce an omnibus test result.
</p>


<h3>Value</h3>

<p>A list containing: a data frame of results of the global test;
a data frame of results of the pairwise results;
and a data frame mentioning the p-value adjustment method.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_05.html">https://rcompanion.org/handbook/H_05.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseMcnemar">pairwiseMcnemar</a></code>, 
<code><a href="#topic+groupwiseCMH">groupwiseCMH</a></code>,
<code><a href="#topic+pairwiseNominalIndependence">pairwiseNominalIndependence</a></code>, 
<code><a href="#topic+pairwiseNominalMatrix">pairwiseNominalMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### 2 x 2 repeated matrix example
data(AndersonRainBarrel)
nominalSymmetryTest(AndersonRainBarrel)
                    
### 3 x 3 repeated matrix example
data(AndersonRainGarden)
nominalSymmetryTest(AndersonRainGarden,
                    exact = FALSE)
                                                              
</code></pre>

<hr>
<h2 id='Nurseries'>Data for proportion of good practices followed by plant nuseries</h2><span id='topic+Nurseries'></span>

<h3>Description</h3>

<p>A data frame with two variables: size of plant nursery in
hectares, and proportion of good practices followed by
the nursery
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nurseries
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 38 rows and 2 columns.
</p>


<h3>Source</h3>

<p>Mangiafico, S.S., Newman, J.P., Mochizuki, M.J.,
and Zurawski, D. (2008). Adoption of sustainable practices
to protect and conserve water resources in container nurseries
with greenhouse facilities. Acta horticulturae 797, 367-372.
</p>

<hr>
<h2 id='oneSampleDominance'>Dominance statistic for one-sample data</h2><span id='topic+oneSampleDominance'></span>

<h3>Description</h3>

<p>Calculates a dominance 
effect size statistic
compared with a theoretical median
for one-sample data
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oneSampleDominance(
  x,
  mu = 0,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  na.rm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oneSampleDominance_+3A_x">x</code></td>
<td>
<p>A vector of numeric values.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_mu">mu</code></td>
<td>
<p>The median against which to compare the values.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>TRUE</code>, removes <code>NA</code> values from
the input vector <code>x</code>.</p>
</td></tr>
<tr><td><code id="oneSampleDominance_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The calculated <code>Dominance</code> statistic is simply 
the proportion of observations greater than <code>mu</code> minus the
the proportion of observations less than <code>mu</code>.
</p>
<p>It will range from -1 to 1, with 0 indicating that the median is
equal to <code>mu</code>, 
and 1 indicating that the observations are all greater in value
than <code>mu</code>,
and -1 indicating that the observations are all less in value
than <code>mu</code>.
</p>
<p>This statistic is appropriate for truly ordinal data,
and could be considered an effect size statistic for
a one-sample sign test. 
</p>
<p>Ordered category data need to re-coded as
numeric, e.g. as with <code>as.numeric(Ordinal.variable)</code>.
</p>
<p>When the statistic is close to 1 or close to -1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>
<p>VDA is the analogous statistic, converted to a probability,
ranging from 0 to 1, specifically,
<code>VDA = Dominance / 2 + 0.5</code>.
</p>


<h3>Value</h3>

<p>A small data frame consisting of descriptive statistics,
the dominance statistic,
and potentially the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_03.html">https://rcompanion.org/handbook/F_03.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairedSampleDominance">pairedSampleDominance</a></code>,
<code><a href="#topic+cliffDelta">cliffDelta</a></code>,
<code><a href="#topic+vda">vda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
library(DescTools)
SignTest(Catbus$Rating, mu=5.5)
oneSampleDominance(Catbus$Rating, mu=5.5)

</code></pre>

<hr>
<h2 id='ordinalEtaSquared'>Eta-squared for ordinal variables</h2><span id='topic+ordinalEtaSquared'></span>

<h3>Description</h3>

<p>Calculates eta-squared 
as an effect size statistic,
following a Kruskal-Wallis test, 
or for a table with one ordinal
variable and one nominal variable; 
confidence intervals by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ordinalEtaSquared(
  x,
  g = NULL,
  group = "row",
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ordinalEtaSquared_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations of an ordinal variable.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_g">g</code></td>
<td>
<p>If <code>x</code> is a vector, <code>g</code> is the vector of observations for
the grouping, nominal variable.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_group">group</code></td>
<td>
<p>If <code>x</code> is a table or matrix, <code>group</code> indicates whether
the <code>"row"</code> or the <code>"column"</code> variable is
the nominal, grouping variable.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="ordinalEtaSquared_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>kruskal.test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Eta-squared is used as a measure of association
for the Kruskal-Wallis test or for a two-way
table with one ordinal and one nominal variable.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>Because eta-squared is always positive, 
if <code>type="perc"</code>, the confidence interval will
never cross zero, and should not
be used for statistical inference.
However, if <code>type="norm"</code>, the confidence interval
may cross zero. 
</p>
<p>When eta-squared is close to 0 or very large,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, eta-squared.
Or a small data frame consisting of eta-squared,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>Cohen, B.H. 2013. Explaining Psychological Statistics, 4th ed. 
Wiley.
</p>
<p><a href="https://rcompanion.org/handbook/F_08.html">https://rcompanion.org/handbook/F_08.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+freemanTheta">freemanTheta</a></code>, 
<code><a href="#topic+epsilonSquared">epsilonSquared</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Breakfast)
library(coin)
chisq_test(Breakfast, scores = list("Breakfast" = c(-2, -1, 0, 1, 2)))
ordinalEtaSquared(Breakfast)

data(PoohPiglet)
kruskal.test(Likert ~ Speaker, data = PoohPiglet)
ordinalEtaSquared(x = PoohPiglet$Likert, g = PoohPiglet$Speaker)

### Same data, as matrix of counts
data(PoohPiglet)
XT = xtabs( ~ Speaker + Likert , data = PoohPiglet)
ordinalEtaSquared(XT)

</code></pre>

<hr>
<h2 id='pairedSampleDominance'>Dominance statistic for two-sample paired data</h2><span id='topic+pairedSampleDominance'></span>

<h3>Description</h3>

<p>Calculates a dominance
effect size statistic
for two-sample paired data
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairedSampleDominance(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  na.rm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairedSampleDominance_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>TRUE</code>, removes <code>NA</code> values from
the input vectors or data frame.</p>
</td></tr>
<tr><td><code id="pairedSampleDominance_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The calculated <code>Dominance</code> statistic is simply 
the proportion of observations in <code>x</code> greater the paired
observations in <code>y</code>,
minus
the proportion of observations in <code>x</code> less than the paired
observations in <code>y</code>
</p>
<p>It will range from -1 to 1, with
and 1 indicating that 
the all the observations in <code>x</code> are greater than 
the paired observations in <code>y</code>,
and -1 indicating that
the all the observations in <code>y</code> are greater than 
the paired observations in <code>x</code>.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>. If there are more than two groups,
only the first two groups are used.
</p>
<p>This statistic is appropriate for truly ordinal data,
and could be considered an effect size statistic for
a two-sample paired sign test.
</p>
<p>Ordered category data need to re-coded as
numeric, e.g. as with <code>as.numeric(Ordinal.variable)</code>.
</p>
<p>When the statistic is close to 1 or close to -1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>
<p>VDA is the analogous statistic, converted to a probability,
ranging from 0 to 1, specifically,
<code>VDA = Dominance / 2 + 0.5</code>
</p>


<h3>Value</h3>

<p>A small data frame consisting of descriptive statistics,
the dominance statistic,
and potentially the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_07.html">https://rcompanion.org/handbook/F_07.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+oneSampleDominance">oneSampleDominance</a></code>,
<code><a href="#topic+vda">vda</a></code>,
<code><a href="#topic+cliffDelta">cliffDelta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pooh)
Time.1 = Pooh$Likert[Pooh$Time == 1]
Time.2 = Pooh$Likert[Pooh$Time == 2]
library(DescTools)
SignTest(x = Time.1, y = Time.2)
pairedSampleDominance(x = Time.1, y = Time.2)
pairedSampleDominance(Likert ~ Time, data=Pooh)

</code></pre>

<hr>
<h2 id='pairwiseDifferences'>[Defunct!] Pairwise differences for unreplicated CBD</h2><span id='topic+pairwiseDifferences'></span>

<h3>Description</h3>

<p>Defunct. Calculates the differences in the response variable
for each pair of levels of a grouping variable 
in an unreplicated complete block design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseDifferences(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseDifferences_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseMcnemar'>Pairwise McNemar and related tests for Cochran Q test post-hoc</h2><span id='topic+pairwiseMcnemar'></span>

<h3>Description</h3>

<p>Conducts pairwise McNemar, exact, and permutation
tests as a post-hoc to Cochran Q test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseMcnemar(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  block = NULL,
  test = "exact",
  method = "fdr",
  digits = 3,
  correct = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseMcnemar_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group | block.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_x">x</code></td>
<td>
<p>The response variable.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_g">g</code></td>
<td>
<p>The grouping variable.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_block">block</code></td>
<td>
<p>The blocking variable.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_test">test</code></td>
<td>
<p>If <code>"exact"</code>, conducts an exact test of symmetry
analogous to a McNemar test.
If <code>"mcnemar"</code>, conducts a McNemar test of symmetry.
If <code>"permutation"</code>, conducts a permutation test 
analogous to a McNemar test.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_method">method</code></td>
<td>
<p>The method for adjusting multiple p-values.
See <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="pairwiseMcnemar_+3A_correct">correct</code></td>
<td>
<p>If <code>TRUE</code>, applies a continuity correction
for the McNemar test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The component tables for the pairwise tests
must be of size 2 x 2.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, <code>g</code>, and <code>block</code>.
</p>


<h3>Value</h3>

<p>A list containing: a data frame of results of the global test;
a data frame of results of the pairwise results;
and a data frame mentioning the p-value adjustment method.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
The second variable on the
right side is used for the blocking variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_07.html">https://rcompanion.org/handbook/H_07.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nominalSymmetryTest">nominalSymmetryTest</a></code>, 
<code><a href="#topic+groupwiseCMH">groupwiseCMH</a></code>,
<code><a href="#topic+pairwiseNominalIndependence">pairwiseNominalIndependence</a></code>, 
<code><a href="#topic+pairwiseNominalMatrix">pairwiseNominalMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Cochran Q post-hoc example
data(HayleySmith)
library(DescTools)
CochranQTest(Response ~ Practice | Student,
             data = HayleySmith)
HayleySmith$Practice = factor(HayleySmith$Practice,
                          levels = c("MowHeight", "SoilTest",
                                     "Clippings", "Irrigation"))
PT = pairwiseMcnemar(Response ~ Practice | Student,
                     data    = HayleySmith,
                     test    = "exact",
                     method  = "fdr",
                     digits  = 3)
PT
PT = PT$Pairwise
cldList(comparison = PT$Comparison,
        p.value    = PT$p.adjust,
        threshold  = 0.05)
                                                             
</code></pre>

<hr>
<h2 id='pairwiseMedianMatrix'>Pairwise Mood's median tests with matrix output</h2><span id='topic+pairwiseMedianMatrix'></span>

<h3>Description</h3>

<p>Conducts pairwise Mood's median tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseMedianMatrix(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  digits = 4,
  method = "fdr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseMedianMatrix_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairwiseMedianMatrix_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwiseMedianMatrix_+3A_x">x</code></td>
<td>
<p>The response variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwiseMedianMatrix_+3A_g">g</code></td>
<td>
<p>The grouping variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwiseMedianMatrix_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits to round output.</p>
</td></tr>
<tr><td><code id="pairwiseMedianMatrix_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwiseMedianMatrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>coin::median_test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>g</code>.
</p>
<p>Mood's median test compares medians among two or more groups.
See <a href="https://rcompanion.org/handbook/F_09.html">https://rcompanion.org/handbook/F_09.html</a> for
futher discussion of this test.
</p>
<p>The <code>pairwiseMedianMatrix</code> function
can be used as a post-hoc method following an omnibus Mood's
median test. It passes the data for pairwise groups to
<code>coin::median_test</code>.
</p>
<p>The matrix output can be converted to a compact letter display,
as in the example.
</p>


<h3>Value</h3>

<p>A list consisting of:
a matrix of p-values;
the p-value adjustment method;
a matrix of adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_09.html">https://rcompanion.org/handbook/F_09.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseMedianTest">pairwiseMedianTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PoohPiglet)
PoohPiglet$Speaker = factor(PoohPiglet$Speaker,
                          levels = c("Pooh", "Tigger", "Piglet"))
PT = pairwiseMedianMatrix(Likert ~ Speaker,
                          data   = PoohPiglet,
                          exact  = NULL,
                          method = "fdr")$Adjusted
PT                           
library(multcompView)
multcompLetters(PT,
                compare="&lt;",
                threshold=0.05,
                Letters=letters)                    

</code></pre>

<hr>
<h2 id='pairwiseMedianTest'>Pairwise Mood's median tests</h2><span id='topic+pairwiseMedianTest'></span>

<h3>Description</h3>

<p>Conducts pairwise Mood's median tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseMedianTest(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  digits = 4,
  method = "fdr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseMedianTest_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairwiseMedianTest_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwiseMedianTest_+3A_x">x</code></td>
<td>
<p>The response variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwiseMedianTest_+3A_g">g</code></td>
<td>
<p>The grouping variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwiseMedianTest_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits to round output.</p>
</td></tr>
<tr><td><code id="pairwiseMedianTest_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwiseMedianTest_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>coin::median_test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>g</code>.
</p>
<p>Mood's median test compares medians among two or more groups.
See <a href="https://rcompanion.org/handbook/F_09.html">https://rcompanion.org/handbook/F_09.html</a> for
further discussion of this test.
</p>
<p>The <code>pairwiseMedianTest</code> function
can be used as a post-hoc method following an omnibus Mood's
median test.  It passes the data for pairwise groups to
<code>coin::median_test</code>.
</p>
<p>The output can be converted to a compact letter display,
as in the example.
</p>


<h3>Value</h3>

<p>A dataframe of the groups being compared, the p-values,
and the adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_09.html">https://rcompanion.org/handbook/F_09.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseMedianMatrix">pairwiseMedianMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PoohPiglet)
PoohPiglet$Speaker = factor(PoohPiglet$Speaker,
                     levels = c("Pooh", "Tigger", "Piglet"))
PT = pairwiseMedianTest(Likert ~ Speaker,
                        data   = PoohPiglet,
                        exact  = NULL,
                        method = "fdr")
PT                         
cldList(comparison = PT$Comparison,
        p.value    = PT$p.adjust,
        threshold  = 0.05)                         

</code></pre>

<hr>
<h2 id='pairwiseModelAnova'>Compare model objects with F test and likelihood ratio test</h2><span id='topic+pairwiseModelAnova'></span>

<h3>Description</h3>

<p>Compares a series of models with pairwise F tests and
likelihood ratio tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseModelAnova(fits, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseModelAnova_+3A_fits">fits</code></td>
<td>
<p>A series of model object names, separated by commas.</p>
</td></tr>
<tr><td><code id="pairwiseModelAnova_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>list</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For comparisons to be valid, both models must
have the same data, without transformations, use the same 
dependent variable, and be fit with the same method.
</p>
<p>To be valid, models need to be nested.
</p>


<h3>Value</h3>

<p>A list of: The calls of the models compared; a data frame of 
comparisons and F tests; and a data frame of
comparisons and likelihood ratio tests.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareGLM">compareGLM</a></code>, <code><a href="#topic+compareLM">compareLM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Compare among polynomial models
data(BrendonSmall)
BrendonSmall$Calories = as.numeric(BrendonSmall$Calories)

BrendonSmall$Calories2 = BrendonSmall$Calories * BrendonSmall$Calories
BrendonSmall$Calories3 = BrendonSmall$Calories * BrendonSmall$Calories * 
                         BrendonSmall$Calories
BrendonSmall$Calories4 = BrendonSmall$Calories * BrendonSmall$Calories * 
                         BrendonSmall$Calories * BrendonSmall$Calories
model.1 = lm(Sodium ~ Calories, data = BrendonSmall)
model.2 = lm(Sodium ~ Calories + Calories2, data = BrendonSmall)
model.3 = lm(Sodium ~ Calories + Calories2 + Calories3, data = BrendonSmall)
model.4 = lm(Sodium ~ Calories + Calories2 + Calories3 + Calories4,
             data = BrendonSmall)
pairwiseModelAnova(model.1, model.2, model.3, model.4)

</code></pre>

<hr>
<h2 id='pairwiseNominalIndependence'>Pairwise tests of independence for nominal data</h2><span id='topic+pairwiseNominalIndependence'></span>

<h3>Description</h3>

<p>Conducts pairwise tests for a 2-dimensional matrix,
in which at at least one dimension has more than two
levels, as a post-hoc test. Conducts Fisher exact, Chi-square,
or G-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseNominalIndependence(
  x,
  compare = "row",
  fisher = TRUE,
  gtest = TRUE,
  chisq = TRUE,
  method = "fdr",
  correct = "none",
  stats = FALSE,
  cramer = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseNominalIndependence_+3A_x">x</code></td>
<td>
<p>A two-way contingency table. At least one dimension should have
more than two levels.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_compare">compare</code></td>
<td>
<p>If <code>"row"</code>, treats the rows as the grouping variable.
If <code>"column"</code>, treats the columns as the grouping variable.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_fisher">fisher</code></td>
<td>
<p>If <code>"TRUE"</code>, conducts fisher exact test.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_gtest">gtest</code></td>
<td>
<p>If <code>"TRUE"</code>, conducts G-test.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_chisq">chisq</code></td>
<td>
<p>If <code>"TRUE"</code>, conducts Chi-square test of association.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_method">method</code></td>
<td>
<p>The method to adjust multiple p-values. 
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_correct">correct</code></td>
<td>
<p>The correction method to pass to <code>DescTools::GTest</code>.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_stats">stats</code></td>
<td>
<p>If <code>"TRUE"</code>, includes the Chi-square value and degrees of 
freedom for Chi-square tests, and the G value.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_cramer">cramer</code></td>
<td>
<p>If <code>"TRUE"</code>, includes an effect size, Cramer's V in the
output.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="pairwiseNominalIndependence_+3A_...">...</code></td>
<td>
<p>Additional arguments, passed to <code>stats::fisher.test</code>, 
<code>DescTools::GTest</code>, or <code>stats::chisq.test</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of comparisons, p-values, and adjusted p-values.
</p>


<h3>Acknowledgments</h3>

<p>My thanks to
Carole Elliott of Kings Park &amp; Botanic Gardens
for suggesting the inclusion on the chi-square statistic
and degrees of freedom in the output.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_04.html">https://rcompanion.org/handbook/H_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseMcnemar">pairwiseMcnemar</a></code>, <code><a href="#topic+groupwiseCMH">groupwiseCMH</a></code>,
<code><a href="#topic+nominalSymmetryTest">nominalSymmetryTest</a></code>, 
<code><a href="#topic+pairwiseNominalMatrix">pairwiseNominalMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Independence test for a 4 x 2 matrix
data(Anderson)
fisher.test(Anderson)
Anderson = Anderson[(c("Heimlich", "Bloom", "Dougal", "Cobblestone")),]
PT = pairwiseNominalIndependence(Anderson,
                                 fisher = TRUE,
                                 gtest  = FALSE,
                                 chisq  = FALSE,
                                 cramer = TRUE)
PT                                
cldList(comparison = PT$Comparison,
        p.value    = PT$p.adj.Fisher,
        threshold  = 0.05)                             
                                                              
</code></pre>

<hr>
<h2 id='pairwiseNominalMatrix'>Pairwise tests of independence for nominal data with matrix output</h2><span id='topic+pairwiseNominalMatrix'></span>

<h3>Description</h3>

<p>Conducts pairwise tests for a 2-dimensional matrix,
in which at at least one dimension has more than two
levels, as a post-hoc test. Conducts Fisher exact, Chi-square,
or G-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseNominalMatrix(
  x,
  compare = "row",
  fisher = TRUE,
  gtest = FALSE,
  chisq = FALSE,
  method = "fdr",
  correct = "none",
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseNominalMatrix_+3A_x">x</code></td>
<td>
<p>A two-way contingency table. At least one dimension should have
more than two levels.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_compare">compare</code></td>
<td>
<p>If <code>"row"</code>, treats the rows as the grouping variable.
If <code>"column"</code>, treats the columns as the grouping variable.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_fisher">fisher</code></td>
<td>
<p>If <code>"TRUE"</code>, conducts fisher exact test.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_gtest">gtest</code></td>
<td>
<p>If <code>"TRUE"</code>, conducts G-test.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_chisq">chisq</code></td>
<td>
<p>If <code>"TRUE"</code>, conducts Chi-square test of association.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_method">method</code></td>
<td>
<p>The method to adjust multiple p-values. 
See <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_correct">correct</code></td>
<td>
<p>The correction method to pass to <code>DescTools::GTest</code>.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="pairwiseNominalMatrix_+3A_...">...</code></td>
<td>
<p>Additional arguments, passed to <code>stats::fisher.test</code>, 
<code>DescTools::GTest</code>, or <code>stats::chisq.test</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list consisting of: the test used,
a matrix of unadjusted p-values,
the p-value adjustment method used,
and a matrix of adjusted p-values.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_04.html">https://rcompanion.org/handbook/H_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseMcnemar">pairwiseMcnemar</a></code>, 
<code><a href="#topic+groupwiseCMH">groupwiseCMH</a></code>,
<code><a href="#topic+nominalSymmetryTest">nominalSymmetryTest</a></code>, 
<code><a href="#topic+pairwiseNominalIndependence">pairwiseNominalIndependence</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Independence test for a 4 x 2 matrix
data(Anderson)
fisher.test(Anderson)
Anderson = Anderson[(c("Heimlich", "Bloom", "Dougal", "Cobblestone")),]
PT = pairwiseNominalMatrix(Anderson,
                           fisher = TRUE,
                           gtest  = FALSE,
                           chisq  = FALSE)$Adjusted
PT
library(multcompView)
multcompLetters(PT)
                                                              
</code></pre>

<hr>
<h2 id='pairwiseOrdinalIndependence'>Pairwise tests of independence for tables with one ordered nominal variable</h2><span id='topic+pairwiseOrdinalIndependence'></span>

<h3>Description</h3>

<p>Conducts pairwise tests for a 2-dimensional table,
in which one variable is ordered nominal and one variable
is non-ordered nominal.  
The function relies on the <code>coin</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseOrdinalIndependence(
  x,
  compare = "row",
  scores = NULL,
  method = "fdr",
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseOrdinalIndependence_+3A_x">x</code></td>
<td>
<p>A two-way contingency table. One dimension is ordered and one
is non-ordered nominal.</p>
</td></tr>
<tr><td><code id="pairwiseOrdinalIndependence_+3A_compare">compare</code></td>
<td>
<p>If <code>"row"</code>, treats the rows as the grouping variable.
If <code>"column"</code>, treats the columns as the grouping 
variable.</p>
</td></tr>
<tr><td><code id="pairwiseOrdinalIndependence_+3A_scores">scores</code></td>
<td>
<p>Optional vector to specify the spacing of the ordered
variable.</p>
</td></tr>
<tr><td><code id="pairwiseOrdinalIndependence_+3A_method">method</code></td>
<td>
<p>The method to adjust multiple p-values. 
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwiseOrdinalIndependence_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="pairwiseOrdinalIndependence_+3A_...">...</code></td>
<td>
<p>Additional arguments, passed to <code>stats::chisq_test</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of comparisons, p-values, and adjusted p-values.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_09.html">https://rcompanion.org/handbook/H_09.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseNominalIndependence">pairwiseNominalIndependence</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Independence test for table with one ordered variable
data(Breakfast)
require(coin)
chisq_test(Breakfast,
           scores = list("Breakfast" = c(-2, -1, 0, 1, 2)))
PT = pairwiseOrdinalIndependence(Breakfast, compare = "row")
PT
cldList(comparison = PT$Comparison, 
        p.value    = PT$p.value, 
        threshold  = 0.05)
        
### Similar to Kruskal-Wallis test for Likert data
data(PoohPiglet)
XT = xtabs(~ Speaker + Likert, data = PoohPiglet)
XT
require(coin)
chisq_test(XT,
           scores = list("Likert" = c(1, 2, 3, 4, 5)))
PT=pairwiseOrdinalIndependence(XT, compare = "row")
PT
cldList(comparison = PT$Comparison, 
        p.value    = PT$p.value, 
        threshold  = 0.05)         
                                                              
</code></pre>

<hr>
<h2 id='pairwiseOrdinalMatrix'>[Defunct!] Pairwise two-sample ordinal regression with matrix output</h2><span id='topic+pairwiseOrdinalMatrix'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise two-sample ordinal regression 
across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseOrdinalMatrix(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseOrdinalMatrix_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseOrdinalPairedMatrix'>[Defunct!] Pairwise two-sample ordinal regression for paired data
with matrix output</h2><span id='topic+pairwiseOrdinalPairedMatrix'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise two-sample ordinal regression 
across groups for paired data
with matrix output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseOrdinalPairedMatrix(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseOrdinalPairedMatrix_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseOrdinalPairedTest'>[Defunct!] Pairwise two-sample ordinal regression for paired data</h2><span id='topic+pairwiseOrdinalPairedTest'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise two-sample ordinal regression 
across groups for paired data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseOrdinalPairedTest(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseOrdinalPairedTest_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseOrdinalTest'>[Defunct!] Pairwise two-sample ordinal regression</h2><span id='topic+pairwiseOrdinalTest'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise two-sample ordinal regression 
across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseOrdinalTest(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseOrdinalTest_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwisePercentileTest'>Pairwise permutation tests for percentiles</h2><span id='topic+pairwisePercentileTest'></span>

<h3>Description</h3>

<p>Conducts pairwise permutation tests across groups 
for percentiles, medians, and proportion below a
threshold value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwisePercentileTest(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  test = "median",
  tau = 0.5,
  type = 7,
  threshold = NA,
  comparison = "&lt;",
  r = 1000,
  digits = 4,
  progress = "TRUE",
  method = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwisePercentileTest_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_test">test</code></td>
<td>
<p>The statistic to compare between groups.  Can be
<code>"median"</code>, <code>"percentile"</code>, <code>"iqr"</code>,
<code>"proportion"</code>,
<code>"mean"</code>, or <code>"variance"</code>.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_tau">tau</code></td>
<td>
<p>If <code>"percentile"</code> is chosen as the <code>test</code>,
<code>tau</code> indicates the percentile to test.  Expressed
as a quantile.  That is, 0.5 indicates a test for medians.
0.75 indicates a test for 75th percentiles.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_type">type</code></td>
<td>
<p>The <code>type</code> value passed to the <code>quantile</code> function.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_threshold">threshold</code></td>
<td>
<p>If <code>"proportion"</code> is chosen as the <code>test</code>,
<code>threshold</code> indicates the value of the dependent variable
to use as the threshold.  For example, to test if there is a 
different in the proportion of observations below $10,000,
<code>threshold = 10000</code> would be used.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_comparison">comparison</code></td>
<td>
<p>If <code>"proportion"</code> is chosen as the <code>test</code>,
<code>comparison</code> indicates the inequality to use.  Options are
<code>"&lt;"</code>, <code>"&lt;="</code>, <code>"&gt;"</code>, <code>"&gt;="</code>, or , <code>"=="</code></p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_r">r</code></td>
<td>
<p>The number of replicates in the permutation test.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_progress">progress</code></td>
<td>
<p>If <code>TRUE</code>, prints a dot for every 1 percent of the 
progress while conducting the test.</p>
</td></tr>
<tr><td><code id="pairwisePercentileTest_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function conducts pairwise tests using the 
<code>percentileTest</code> function. The user can consult the
documentation for that function for additional details.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>.
</p>


<h3>Value</h3>

<p>A dataframe of the groups being compared, the p-values,
and the adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_15.html">https://rcompanion.org/handbook/F_15.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+percentileTest">percentileTest</a></code>, 
<code><a href="#topic+groupwisePercentile">groupwisePercentile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(BrendonSmall)
PT = pairwisePercentileTest(Sodium ~ Instructor, 
                            data = BrendonSmall, 
                            test = "percentile", 
                            tau  = 0.75)
PT
cldList(p.adjust ~ Comparison,
        data       = PT,
        threshold  = 0.05)
        
data(BrendonSmall)
PT = pairwisePercentileTest(Sodium ~ Instructor, 
                            data       = BrendonSmall, 
                            test       = "proportion", 
                            threshold  = 1300)
PT
cldList(p.adjust ~ Comparison,
        data       = PT,
        threshold  = 0.05)                         

## End(Not run)

</code></pre>

<hr>
<h2 id='pairwisePermutationMatrix'>Pairwise two-sample independence tests with matrix output</h2><span id='topic+pairwisePermutationMatrix'></span>

<h3>Description</h3>

<p>Conducts pairwise two-sample independence tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwisePermutationMatrix(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  method = "fdr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwisePermutationMatrix_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairwisePermutationMatrix_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwisePermutationMatrix_+3A_x">x</code></td>
<td>
<p>The response variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationMatrix_+3A_g">g</code></td>
<td>
<p>The grouping variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationMatrix_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwisePermutationMatrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>coin::independence_test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>g</code>.
</p>
<p>This function is a wrapper for <code>coin::independence_test</code>,
passing pairwise groups to the function. It's critical to read
and understand the documentation for this function to understand
its use and options.
</p>
<p>For some options for common tests, see Horthorn et al., 2008.
</p>


<h3>Value</h3>

<p>A list consisting of:
A matrix of p-values;
the p-value adjustment method;
a matrix of adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/K_02.html">https://rcompanion.org/handbook/K_02.html</a>
</p>
<p>Hothorn, T., K. Hornik, M.A. van de Wiel, and A. Zeileis. 2008. 
Implementing a Class of Permutation Tests: The coin Package. 
Journal of Statistical Software, 28(8), 1–23.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwisePermutationTest">pairwisePermutationTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Fisher-Pitman test

data(BrendonSmall)

library(coin)
                                 
independence_test(Sodium ~ Instructor, data = BrendonSmall, 
                  teststat = "quadratic") 
                                      
PT = pairwisePermutationMatrix(Sodium ~ Instructor,
                               data     = BrendonSmall,
                               teststat = "quadratic",
                               method   = "fdr")
PT

PA = PT$Adjusted
library(multcompView)
multcompLetters(PA,
                compare="&lt;",
                threshold=0.05,
                Letters=letters)   

</code></pre>

<hr>
<h2 id='pairwisePermutationSymmetry'>Pairwise two-sample symmetry tests</h2><span id='topic+pairwisePermutationSymmetry'></span>

<h3>Description</h3>

<p>Conducts pairwise two-sample symmetry tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwisePermutationSymmetry(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  b = NULL,
  method = "fdr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwisePermutationSymmetry_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group | block.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetry_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetry_+3A_x">x</code></td>
<td>
<p>The response variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetry_+3A_g">g</code></td>
<td>
<p>The grouping variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetry_+3A_b">b</code></td>
<td>
<p>The blocking variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetry_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetry_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>coin::symmetry_test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, <code>g</code>, and <code>b</code>.
</p>
<p>This function is a wrapper for <code>coin::symmetry_test</code>,
passing pairwise groups to the function. It's critical to read
and understand the documentation for this function to understand
its use and options.
</p>


<h3>Value</h3>

<p>A dataframe of the groups being compared, the p-values,
and the adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
The second variable on the
right side is used for the blocking variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/K_03.html">https://rcompanion.org/handbook/K_03.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwisePermutationSymmetryMatrix">pairwisePermutationSymmetryMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BobBelcher)

BobBelcher$Instructor = factor( BobBelcher$Instructor, 
                                levels = c("Linda Belcher", "Louise Belcher",
                                           "Tina Belcher", "Bob Belcher",
                                           "Gene Belcher"))
                                           
library(coin)

symmetry_test(Likert ~ Instructor | Rater, data= BobBelcher,
              ytrafo   = rank_trafo,
              teststat = "quadratic")

PT = pairwisePermutationSymmetry(Likert ~ Instructor | Rater,
                                 data     = BobBelcher,
                                 ytrafo   = rank_trafo,
                                 teststat = "quadratic",
                                 method   = "fdr")
PT

cldList(comparison = PT$Comparison,
        p.value    = PT$p.adjust,
       threshold  = 0.05)

</code></pre>

<hr>
<h2 id='pairwisePermutationSymmetryMatrix'>Pairwise two-sample symmetry tests with matrix output</h2><span id='topic+pairwisePermutationSymmetryMatrix'></span>

<h3>Description</h3>

<p>Conducts pairwise two-sample symmetry tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwisePermutationSymmetryMatrix(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  b = NULL,
  method = "fdr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_x">x</code></td>
<td>
<p>The response variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_g">g</code></td>
<td>
<p>The grouping variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_b">b</code></td>
<td>
<p>The blocking variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwisePermutationSymmetryMatrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>coin::symmetry_test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, <code>g</code>, and <code>b</code>.
</p>
<p>This function is a wrapper for <code>coin::symmetry_test</code>,
passing pairwise groups to the function. It's critical to read
and understand the documentation for this function to understand
its use and options.
</p>


<h3>Value</h3>

<p>A list consisting of:
A matrix of p-values;
the p-value adjustment method;
a matrix of adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
The second variable on the
right side is used for the blocking variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/K_03.html">https://rcompanion.org/handbook/K_03.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwisePermutationSymmetry">pairwisePermutationSymmetry</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BobBelcher)

BobBelcher$Instructor = factor( BobBelcher$Instructor, 
                                levels = c("Linda Belcher", "Louise Belcher",
                                           "Tina Belcher", "Bob Belcher",
                                           "Gene Belcher"))

library(coin)

symmetry_test(Likert ~ Instructor | Rater, data= BobBelcher,
              ytrafo   = rank_trafo,
              teststat = "quadratic")

PT = pairwisePermutationSymmetryMatrix(Likert ~ Instructor | Rater,
                                 data     = BobBelcher,
                                 ytrafo   = rank_trafo,
                                 teststat = "quadratic",
                                 method   = "fdr")
PT

PA = PT$Adjusted
library(multcompView)
multcompLetters(PA,
                compare="&lt;",
                threshold=0.05,
                Letters=letters)     

</code></pre>

<hr>
<h2 id='pairwisePermutationTest'>Pairwise two-sample independence tests</h2><span id='topic+pairwisePermutationTest'></span>

<h3>Description</h3>

<p>Conducts pairwise two-sample independence tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwisePermutationTest(
  formula = NULL,
  data = NULL,
  x = NULL,
  g = NULL,
  method = "fdr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwisePermutationTest_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the measurement variable and
the grouping variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="pairwisePermutationTest_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="pairwisePermutationTest_+3A_x">x</code></td>
<td>
<p>The response variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationTest_+3A_g">g</code></td>
<td>
<p>The grouping variable as a vector.</p>
</td></tr>
<tr><td><code id="pairwisePermutationTest_+3A_method">method</code></td>
<td>
<p>The p-value adjustment method to use for multiple tests.
See <code>stats::p.adjust</code>.</p>
</td></tr>
<tr><td><code id="pairwisePermutationTest_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code>coin::independence_test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>g</code>.
</p>
<p>This function is a wrapper for <code>coin::independence_test</code>,
passing pairwise groups to the function. It's critical to read
and understand the documentation for this function to understand
its use and options.
</p>
<p>For some options for common tests, see Horthorn et al., 2008.
</p>


<h3>Value</h3>

<p>A dataframe of the groups being compared, the p-values,
and the adjusted p-values.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/K_02.html">https://rcompanion.org/handbook/K_02.html</a>
</p>
<p>Hothorn, T., K. Hornik, M.A. van de Wiel, and A. Zeileis. 2008. 
Implementing a Class of Permutation Tests: The coin Package. 
Journal of Statistical Software, 28(8), 1–23.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwisePermutationMatrix">pairwisePermutationMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Fisher-Pitman test

data(BrendonSmall)

library(coin)
                     
independence_test(Sodium ~ Instructor, data = BrendonSmall, 
                 teststat="quadratic")
                                       
PT = pairwisePermutationTest(Sodium ~ Instructor,
                             data   = BrendonSmall,
                             teststat="quadratic",
                             method = "fdr")
PT

cldList(comparison = PT$Comparison,
        p.value    = PT$p.adjust,
        threshold  = 0.05)

</code></pre>

<hr>
<h2 id='pairwiseRobustMatrix'>[Defunct!] Pairwise two-sample robust tests with matrix output</h2><span id='topic+pairwiseRobustMatrix'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise two-sample robust tests across groups
with matrix output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseRobustMatrix(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseRobustMatrix_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseRobustTest'>[Defunct!] Pairwise two-sample robust tests</h2><span id='topic+pairwiseRobustTest'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise two-sample robust tests across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseRobustTest(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseRobustTest_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseSignMatrix'>[Defunct!] Pairwise sign tests with matrix output</h2><span id='topic+pairwiseSignMatrix'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise sign tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseSignMatrix(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseSignMatrix_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='pairwiseSignTest'>[Defunct!] Pairwise sign tests</h2><span id='topic+pairwiseSignTest'></span>

<h3>Description</h3>

<p>Defunct. Performs pairwise sign tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseSignTest(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseSignTest_+3A_...">...</code></td>
<td>
<p>Anything.</p>
</td></tr>
</table>

<hr>
<h2 id='Pennsylvania18'>Votes for the Democratic candidate in Pennsylvania 18 in 2016 and 2018</h2><span id='topic+Pennsylvania18'></span>

<h3>Description</h3>

<p>A two-by-two matrix with the proportion of votes for the
Democratic candidate in two races, in 2016 and 2018.
2016 is the Presidential election with Hilary Clinton as the
Democratic candidate.  2018 is a House of Representatives
election with Conor Lamb.  These data are for
Pennsylvania's 18th Congressional District.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pennsylvania18
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 2 rows and 2 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_10.html">https://rcompanion.org/handbook/H_10.html</a>
</p>

<hr>
<h2 id='percentileTest'>Test of percentiles by permutation test</h2><span id='topic+percentileTest'></span>

<h3>Description</h3>

<p>Conducts a permutation test to compare two groups for medians,
percentiles, or proportion below a threshold value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>percentileTest(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  test = "median",
  tau = 0.5,
  type = 7,
  threshold = NA,
  comparison = "&lt;",
  r = 1000,
  digits = 4,
  progress = "TRUE"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="percentileTest_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_test">test</code></td>
<td>
<p>The statistic to compare between groups.  Can be
<code>"median"</code>, <code>"percentile"</code>, <code>"iqr"</code>,
<code>"proportion"</code>,
<code>"mean"</code>, or <code>"variance"</code>.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_tau">tau</code></td>
<td>
<p>If <code>"percentile"</code> is chosen as the <code>test</code>,
<code>tau</code> indicates the percentile to test.  Expressed
as a quantile.  That is, 0.5 indicates a test for medians.
0.75 indicates a test for 75th percentiles.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_type">type</code></td>
<td>
<p>The <code>type</code> value passed to the <code>quantile</code> function.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_threshold">threshold</code></td>
<td>
<p>If <code>"proportion"</code> is chosen as the <code>test</code>,
<code>threshold</code> indicates the value of the dependent variable
to use as the threshold.  For example, to test if there is a 
different in the proportion of observations below $10,000,
<code>threshold = 10000</code> would be used.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_comparison">comparison</code></td>
<td>
<p>If <code>"proportion"</code> is chosen as the <code>test</code>,
<code>comparison</code> indicates the inequality to use.  Options are
<code>"&lt;"</code>, <code>"&lt;="</code>, <code>"&gt;"</code>, <code>"&gt;="</code>, or , <code>"=="</code></p>
</td></tr>
<tr><td><code id="percentileTest_+3A_r">r</code></td>
<td>
<p>The number of replicates in the permutation test.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="percentileTest_+3A_progress">progress</code></td>
<td>
<p>If <code>TRUE</code>, prints a dot for every 1 percent of 
progress while conducting the test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will test for a difference in medians, percentiles,
interquartile ranges, proportion of observations above or below
some threshold value, 
means, or variances between two groups
by permutation test.
</p>
<p>The permutation test simply permutes the observed values over the
two groups and counts how often the calculated statistic is
at least as extreme as the original observed statistic.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code> and <code>y</code>.
</p>
<p>The function removes cases with NA in any of the variables.
</p>
<p>If the independent variable has more than two groups,
only the first two levels of the factor variable will be used.
</p>
<p>The p-value returned is a two-sided test.
</p>


<h3>Value</h3>

<p>A list of three data frames with
the data used, a summary for each group,
and the p-value from the test.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the independent variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_15.html">https://rcompanion.org/handbook/F_15.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BrendonSmall)
percentileTest(Sodium ~ Instructor, 
               data=BrendonSmall, 
               test="median")

percentileTest(Sodium ~ Instructor, 
               data=BrendonSmall, 
               test="percentile", 
               tau = 0.75)

percentileTest(Sodium ~ Instructor, 
               data=BrendonSmall, 
               test="proportion", 
               threshold = 1300)
               
</code></pre>

<hr>
<h2 id='phi'>phi</h2><span id='topic+phi'></span>

<h3>Description</h3>

<p>Calculates phi for a 2 x 2 table of nominal variables;
confidence intervals by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phi(
  x,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  verbose = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phi_+3A_x">x</code></td>
<td>
<p>Either a 2 x 2 table or a 2 x 2 matrix.
Can also be a vector of observations for one dimension
of a 2 x 2 table.</p>
</td></tr>
<tr><td><code id="phi_+3A_y">y</code></td>
<td>
<p>If <code>x</code> is a vector, <code>y</code> is the vector of observations for
the second dimension of a 2 x2  table.</p>
</td></tr>
<tr><td><code id="phi_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="phi_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="phi_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="phi_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="phi_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="phi_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints the table of counts.</p>
</td></tr>
<tr><td><code id="phi_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="phi_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="phi_+3A_...">...</code></td>
<td>
<p>Additional arguments. (Ignored.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>phi is used as a measure of association
between two binomial variables, or as an effect size
for a chi-square test of association for a 2 x 2 table.
The absolute value of the phi statistic is the same as
Cramer's V for a 2 x 2 table.
</p>
<p>Unlike Cramer's V, phi can be positive or negative (or zero), and
ranges from -1 to 1.
</p>
<p>When phi is close to its extremes,
or with small counts, 
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, phi. 
Or a small data frame consisting of phi,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/H_10.html">https://rcompanion.org/handbook/H_10.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cramerV">cramerV</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with table
Matrix = matrix(c(13, 26, 26, 13), ncol=2)
phi(Matrix)

### Example with two vectors
Species = c(rep("Species1", 16), rep("Species2", 16))
Color   = c(rep(c("blue", "blue", "blue", "green"),4),
            rep(c("green", "green", "green", "blue"),4))
phi(Species, Color)

</code></pre>

<hr>
<h2 id='plotDensityHistogram'>Histogram with a density curve</h2><span id='topic+plotDensityHistogram'></span>

<h3>Description</h3>

<p>Produces a histogram for a vector of values and adds a density
curve of the distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDensityHistogram(
  x,
  prob = FALSE,
  col = "gray",
  main = "",
  linecol = "black",
  lwd = 2,
  adjust = 1,
  bw = "nrd0",
  kernel = "gaussian",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDensityHistogram_+3A_x">x</code></td>
<td>
<p>A vector of values.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_prob">prob</code></td>
<td>
<p>If <code>FALSE</code>, then counts are displayed in the histogram.
If <code>TRUE</code>, then the density is shown.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_col">col</code></td>
<td>
<p>The color of the histogram bars.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_main">main</code></td>
<td>
<p>The title displayed for the plot.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_linecol">linecol</code></td>
<td>
<p>The color of the line in the plot.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_lwd">lwd</code></td>
<td>
<p>The width of the line in the plot.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_adjust">adjust</code></td>
<td>
<p>Passed to <code><a href="stats.html#topic+density">density</a></code>. A lower value makes the density
plot smoother.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_bw">bw</code></td>
<td>
<p>Passed to <code><a href="stats.html#topic+density">density</a></code>.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_kernel">kernel</code></td>
<td>
<p>Passed to <code><a href="stats.html#topic+density">density</a></code>.</p>
</td></tr>
<tr><td><code id="plotDensityHistogram_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="graphics.html#topic+hist">hist</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function relies on the <code>hist</code> function. The density curve
relies on the <code>density</code> function.
</p>


<h3>Value</h3>

<p>Produces a plot. Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/C_04.html">https://rcompanion.org/handbook/C_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotNormalHistogram">plotNormalHistogram</a></code>, 
<code><a href="#topic+plotNormalDensity">plotNormalDensity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Plot of residuals from a model fit with lm
data(Catbus)
model = lm(Steps ~ Gender + Teacher,
           data = Catbus)
plotDensityHistogram(residuals(model))          

</code></pre>

<hr>
<h2 id='plotNormalDensity'>Density plot with a normal curve</h2><span id='topic+plotNormalDensity'></span>

<h3>Description</h3>

<p>Produces a density plot for a vector of values and adds 
a normal curve with the same mean and standard deviation. The 
plot can be used to quickly compare the distribution of data
to a normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNormalDensity(
  x,
  col1 = "white",
  col2 = "gray",
  col3 = "blue",
  border = NA,
  main = "",
  lwd = 2,
  length = 1000,
  adjust = 1,
  bw = "nrd0",
  kernel = "gaussian",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotNormalDensity_+3A_x">x</code></td>
<td>
<p>A vector of values.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_col1">col1</code></td>
<td>
<p>The color of the density plot. Usually not visible.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_col2">col2</code></td>
<td>
<p>The color of the density polygon.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_col3">col3</code></td>
<td>
<p>The color of the normal line.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_border">border</code></td>
<td>
<p>The color of the border around the density polygon.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_main">main</code></td>
<td>
<p>The title displayed for the plot.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_lwd">lwd</code></td>
<td>
<p>The width of the line in the plot.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_length">length</code></td>
<td>
<p>The number of points in the line in the plot.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_adjust">adjust</code></td>
<td>
<p>Passed to  <code><a href="stats.html#topic+density">density</a></code>. 
A lower value makes the density plot smoother.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_bw">bw</code></td>
<td>
<p>Passed to  <code><a href="stats.html#topic+density">density</a></code>.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_kernel">kernel</code></td>
<td>
<p>Passed to  <code><a href="stats.html#topic+density">density</a></code>.</p>
</td></tr>
<tr><td><code id="plotNormalDensity_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots a polygon based on the <code>density</code> function. 
The normal curve has the same mean and standard deviation as the 
values in the vector.
</p>


<h3>Value</h3>

<p>Produces a plot. Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/I_01.html">https://rcompanion.org/handbook/I_01.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotNormalHistogram">plotNormalHistogram</a></code>,
<code><a href="#topic+plotDensityHistogram">plotDensityHistogram</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Plot of residuals from a model fit with lm
data(Catbus)
model = lm(Steps ~ Gender + Teacher,
           data = Catbus)
 plotNormalDensity(residuals(model))          

</code></pre>

<hr>
<h2 id='plotNormalHistogram'>Histogram with a normal curve</h2><span id='topic+plotNormalHistogram'></span>

<h3>Description</h3>

<p>Produces a histogram for a vector of values and adds a normal
curve with the same mean and standard deviation. The plot
can be used to quickly compare the distribution of data
to a normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNormalHistogram(
  x,
  prob = FALSE,
  col = "gray",
  main = "",
  linecol = "blue",
  lwd = 2,
  length = 1000,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotNormalHistogram_+3A_x">x</code></td>
<td>
<p>A vector of values.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_prob">prob</code></td>
<td>
<p>If <code>FALSE</code>, then counts are displayed in the histogram.
If <code>TRUE</code>, then the density is shown.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_col">col</code></td>
<td>
<p>The color of the histogram bars.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_main">main</code></td>
<td>
<p>The title displayed for the plot.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_linecol">linecol</code></td>
<td>
<p>The color of the line in the plot.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_lwd">lwd</code></td>
<td>
<p>The width of the line in the plot.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_length">length</code></td>
<td>
<p>The number of points in the line in the plot.</p>
</td></tr>
<tr><td><code id="plotNormalHistogram_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="graphics.html#topic+hist">hist</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function relies on the <code>hist</code> function. The normal curve
has the same mean and standard deviation as the values in the
vector.
</p>


<h3>Value</h3>

<p>Produces a plot. Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/I_01.html">https://rcompanion.org/handbook/I_01.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotNormalDensity">plotNormalDensity</a></code>,
<code><a href="#topic+plotDensityHistogram">plotDensityHistogram</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Plot of residuals from a model fit with lm
data(Catbus)
model = lm(Steps ~ Gender + Teacher,
           data = Catbus)
 plotNormalHistogram(residuals(model))          

</code></pre>

<hr>
<h2 id='plotPredy'>Plot a predicted line from a bivariate model</h2><span id='topic+plotPredy'></span>

<h3>Description</h3>

<p>Plots the best fit line for a model with one y variable and
one x variable, or with one y variable and polynomial
x variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPredy(
  data,
  x,
  y,
  model,
  order = 1,
  x2 = NULL,
  x3 = NULL,
  x4 = NULL,
  x5 = NULL,
  pch = 16,
  xlab = "X",
  ylab = "Y",
  length = 1000,
  lty = 1,
  lwd = 2,
  col = "blue",
  type = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPredy_+3A_data">data</code></td>
<td>
<p>The name of the data frame.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_x">x</code></td>
<td>
<p>The name of the x variable.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_y">y</code></td>
<td>
<p>The name of the y variable.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_model">model</code></td>
<td>
<p>The name of the model object.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_order">order</code></td>
<td>
<p>If plotting a polynomial function, the order of the polynomial.
Otherwise can be left as <code>1</code>.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_x2">x2</code></td>
<td>
<p>If applicable, the name of the second order 
polynomial x variable.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_x3">x3</code></td>
<td>
<p>If applicable, the name of the third order 
polynomial x variable.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_x4">x4</code></td>
<td>
<p>If applicable, the name of the fourth order 
polynomial x variable.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_x5">x5</code></td>
<td>
<p>If applicable, the name of the fifth order 
polynomial x variable.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_pch">pch</code></td>
<td>
<p>The shape of the plotted data points.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_xlab">xlab</code></td>
<td>
<p>The label for the x-axis.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_ylab">ylab</code></td>
<td>
<p>The label for the y-axis.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_length">length</code></td>
<td>
<p>The number of points used to draw the line.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_lty">lty</code></td>
<td>
<p>The style of the plotted line.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_lwd">lwd</code></td>
<td>
<p>The width of the plotted line.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_col">col</code></td>
<td>
<p>The col of the plotted line.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_type">type</code></td>
<td>
<p>Passed to <code>predict</code>. Required for certain models.</p>
</td></tr>
<tr><td><code id="plotPredy_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>plot</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Any model for which <code>predict()</code> is defined can be used.
</p>


<h3>Value</h3>

<p>Produces a plot. Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/I_10.html">https://rcompanion.org/handbook/I_10.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Plot of linear model fit with lm
data(BrendonSmall)
model = lm(Weight ~ Calories, data = BrendonSmall) 
plotPredy(data  = BrendonSmall,
          y     = Weight,
          x     = Calories,
          model = model,
          xlab  = "Calories per day",
          ylab  = "Weight in kilograms")
           
### Plot of polynomial model fit with lm
data(BrendonSmall)
BrendonSmall$Calories2 = BrendonSmall$Calories * BrendonSmall$Calories
model = lm(Sodium ~ Calories + Calories2, data = BrendonSmall) 
plotPredy(data  = BrendonSmall,
          y     = Sodium,
          x     = Calories,
          x2    = Calories2,
          model = model,
          order = 2,
          xlab  = "Calories per day",
          ylab  = "Sodium intake per day")

### Plot of quadratic plateau model fit with nls
data(BrendonSmall)
quadplat = function(x, a, b, clx) {
          ifelse(x  &lt; clx, a + b * x   + (-0.5*b/clx) * x   * x,
                           a + b * clx + (-0.5*b/clx) * clx * clx)}
model = nls(Sodium ~ quadplat(Calories, a, b, clx),
            data = BrendonSmall,
            start = list(a   = 519,
                         b   = 0.359,
                         clx = 2304))
plotPredy(data  = BrendonSmall,
          y     = Sodium,
          x     = Calories,
          model = model,
          xlab  = "Calories per day",
          ylab  = "Sodium intake per day")

### Logistic regression example requires type option
data(BullyHill)
Trials = cbind(BullyHill$Pass, BullyHill$Fail)
model.log = glm(Trials ~ Grade, data = BullyHill,
                family = binomial(link="logit"))
plotPredy(data  = BullyHill,
          y     = Percent,
          x     = Grade,
          model = model.log,
          type  = "response",
          xlab  = "Grade",
          ylab  = "Proportion passing")

</code></pre>

<hr>
<h2 id='PMCMRTable'>Convert PMCMR Objects to a Data Frame</h2><span id='topic+PMCMRTable'></span>

<h3>Description</h3>

<p>Extracts a data frame of comparisons and p-values
from an PMCMR object from the PMCMRplus
package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PMCMRTable(PMCMR, reverse = TRUE, digits = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PMCMRTable_+3A_pmcmr">PMCMR</code></td>
<td>
<p>A PMCMR object</p>
</td></tr>
<tr><td><code id="PMCMRTable_+3A_reverse">reverse</code></td>
<td>
<p>If <code>TRUE</code>, reports the comparison as e.g. (B - A = 0).
This will more closely match the output of 
<code>PMCMRplus::summary.PMCMR</code> for all-pairs comparisons.
If <code>FALSE</code>, reports the comparison as e.g. (A - B = 0).
This will result in the output from <code>rcompanion::cldList</code>
matching the output of
<code>PMCMRplus::summaryGroup</code></p>
</td></tr>
<tr><td><code id="PMCMRTable_+3A_digits">digits</code></td>
<td>
<p>The significant digits in the output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Should produce meaningful output for all-pairs 
and many-to-one comparisons.
</p>


<h3>Value</h3>

<p>A data frame of comparisons and p-values
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_08.html">https://rcompanion.org/handbook/F_08.html</a>
</p>

<hr>
<h2 id='Pooh'>Hypothetical data for paired ratings of Pooh Bear</h2><span id='topic+Pooh'></span>

<h3>Description</h3>

<p>A data frame of Likert responses for instructor Pooh Bear
for each of 10 respondents, paired before and after. 
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pooh
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 20 rows and 4 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/F_06.html">https://rcompanion.org/handbook/F_06.html</a>
</p>

<hr>
<h2 id='PoohPiglet'>Hypothetical data for ratings of Pooh, Piglet, and Tigger</h2><span id='topic+PoohPiglet'></span>

<h3>Description</h3>

<p>A data frame of Likert responses for instructors Pooh Bear,
Piglet, and Tigger.
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PoohPiglet
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 30 rows and 2 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/F_08.html">https://rcompanion.org/handbook/F_08.html</a>
</p>

<hr>
<h2 id='quantileCI'>Quantiles and confidence intervals</h2><span id='topic+quantileCI'></span>

<h3>Description</h3>

<p>Calculates an estimate for a quantile 
and confidence intervals for
a vector of discrete or continuous
values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quantileCI(
  x,
  tau = 0.5,
  level = 0.95,
  method = "binomial",
  type = 3,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quantileCI_+3A_x">x</code></td>
<td>
<p>The vector of observations.
Can be an ordered factor as long as <code>type</code>
is 1 or 3.</p>
</td></tr>
<tr><td><code id="quantileCI_+3A_tau">tau</code></td>
<td>
<p>The quantile to use,
e.g. 0.5 for median, 0.25 for 25th percentile.</p>
</td></tr>
<tr><td><code id="quantileCI_+3A_level">level</code></td>
<td>
<p>The confidence interval to use, 
e.g. 0.95 for 95 percent confidence interval.</p>
</td></tr>
<tr><td><code id="quantileCI_+3A_method">method</code></td>
<td>
<p>If <code>"binomial"</code>, uses the binomial distribution 
the confidence limits.
If <code>"normal"</code>, uses the normal approximation to the
binomial distribution.</p>
</td></tr>
<tr><td><code id="quantileCI_+3A_type">type</code></td>
<td>
<p>The <code>type</code> value passed to the <code>quantile</code> function.</p>
</td></tr>
<tr><td><code id="quantileCI_+3A_digits">digits</code></td>
<td>
<p>The number of significant figures to use in output.</p>
</td></tr>
<tr><td><code id="quantileCI_+3A_...">...</code></td>
<td>
<p>Other arguments, ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Conover recommends the <code>"binomial"</code> method for sample
sizes less than or equal to 20.
With the current implementation, 
this method can be used also for
larger sample sizes.
</p>


<h3>Value</h3>

<p>A data frame of summary statistics, quantile estimate, 
and confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/E_04.html">https://rcompanion.org/handbook/E_04.html</a>
</p>
<p>Conover, W.J., Practical Nonparametric Statistics, 3rd.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+groupwisePercentile">groupwisePercentile</a></code>, 
<code><a href="#topic+groupwiseMedian">groupwiseMedian</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### From Conover, Practical Nonparametric Statistics, 3rd
Hours = c(46.9, 47.2, 49.1, 56.5, 56.8, 59.2, 59.9, 63.2,
          63.3, 63.4, 63.7, 64.1, 67.1, 67.7, 73.3, 78.5)
quantileCI(Hours)

### Example with ordered factor
set.seed(12345)
Pool = factor(c("smallest", "small", "medium", "large", "largest"),
             ordered=TRUE, 
             levels=c("smallest", "small", "medium", "large", "largest"))
Sample = sample(Pool, 24, replace=TRUE)
quantileCI(Sample)
                 
</code></pre>

<hr>
<h2 id='Religion'>Hypothetical data for change in religion after a caucusing event</h2><span id='topic+Religion'></span>

<h3>Description</h3>

<p>A matrix of paired counts for religion of people before 
and after an event.
Hypothetical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Religion
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 4 rows and 4 columns.
</p>


<h3>Source</h3>

<p><a href="https://rcompanion.org/handbook/H_05.html">https://rcompanion.org/handbook/H_05.html</a>
</p>

<hr>
<h2 id='scheirerRayHare'>Scheirer Ray Hare test</h2><span id='topic+scheirerRayHare'></span>

<h3>Description</h3>

<p>Conducts Scheirer Ray Hare test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scheirerRayHare(
  formula = NULL,
  data = NULL,
  y = NULL,
  x1 = NULL,
  x2 = NULL,
  type = 2,
  tie.correct = TRUE,
  ss = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scheirerRayHare_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
two independent variables. e.g. y ~ x1 + x2.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_y">y</code></td>
<td>
<p>If no formula is given, the response variable.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_x1">x1</code></td>
<td>
<p>If no formula is given, the first independent variable.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_x2">x2</code></td>
<td>
<p>If no formula is given, the second independent variable.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_type">type</code></td>
<td>
<p>The type of sum of squares to be used.
Acceptable options are <code>1</code>, <code>2</code>,
<code>"I"</code>, or <code>"II"</code>.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_tie.correct">tie.correct</code></td>
<td>
<p>If <code>"TRUE"</code>, applies a correction for ties in the 
response variable.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_ss">ss</code></td>
<td>
<p>If <code>"TRUE"</code>, includes the sums of squares in the output.</p>
</td></tr>
<tr><td><code id="scheirerRayHare_+3A_verbose">verbose</code></td>
<td>
<p>If <code>"TRUE"</code>, outputs statistics used in the analysis 
by direct print.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Scheirer Ray Hare test is a nonparametric test used for a 
two-way factorial experiment.  It is described by Sokal and
Rohlf (1995).
</p>
<p>It is sometimes recommended that the design should be balanced,
and that there should be at least five observations for each
cell in the interaction.
</p>
<p>One might consider using aligned ranks transformation anova
instead of the Scheirer Ray Hare test.
</p>
<p>Note that for unbalanced designs, 
by default, a type-II sum-of-squares
approach is used.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>y</code>, <code>x1</code>, and <code>x2</code>.
</p>
<p>The function removes cases with NA in any of the variables.
</p>


<h3>Value</h3>

<p>A data frame of results similar to an anova table. Output from the
<code>verbose</code> option is printed directly and not returned with
the data frame.
</p>


<h3>Acknowledgments</h3>

<p>Thanks to Guillaume Loignon for the suggestion to
include type-II sum-of-squares.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the first independent variable.
The second variable on the
right side is used for the second independent variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>Sokal, R.R. and F.J. Rohlf. 1995. Biometry. 3rd ed. W.H. Freeman, 
New York.
</p>
<p><a href="https://rcompanion.org/handbook/F_14.html">https://rcompanion.org/handbook/F_14.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example from Sokal and Rohlf, 1995.
Value = c(709,679,699,657,594,677,592,538,476,508,505,539)
Sex   = c(rep("Male",3), rep("Female",3), rep("Male",3), rep("Female",3))
Fat   = c(rep("Fresh", 6), rep("Rancid", 6))
Sokal = data.frame(Value, Sex, Fat)

scheirerRayHare(Value ~ Sex + Fat, data=Sokal)
                                                             
</code></pre>

<hr>
<h2 id='spearmanRho'>Spearman's rho, Kendall's tau, Pearson's r</h2><span id='topic+spearmanRho'></span>

<h3>Description</h3>

<p>Calculates Spearmans's rho, Kendall's tau, or Pearson's r
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spearmanRho(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  method = "spearman",
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spearmanRho_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the two paired variables,
e.g.  <code>~ x + y</code>. The variables should be
vectors of the same length.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_x">x</code></td>
<td>
<p>If no formula is given, the values for one variable.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_y">y</code></td>
<td>
<p>The values for the other variable.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_method">method</code></td>
<td>
<p>One of &quot;spearman&quot;, &quot;kendall&quot;, or &quot;pearson&quot;.
Passed to <code>cor</code>.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="spearmanRho_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>cor</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper for <code>stats::cor</code>
with the addition of confidence intervals.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the returned statistic is close to -1 or close to 1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, rho, tau, or r.
Or a small data frame consisting of rho, tau, or r, 
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/I_10.html">https://rcompanion.org/handbook/I_10.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
spearmanRho( ~ Steps + Rating, data=Catbus)

</code></pre>

<hr>
<h2 id='transformTukey'>Tukey's Ladder of Powers</h2><span id='topic+transformTukey'></span>

<h3>Description</h3>

<p>Conducts Tukey's Ladder of Powers on a vector of values to 
produce a more-normally distributed vector of values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transformTukey(
  x,
  start = -10,
  end = 10,
  int = 0.025,
  plotit = TRUE,
  verbose = FALSE,
  quiet = FALSE,
  statistic = 1,
  returnLambda = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transformTukey_+3A_x">x</code></td>
<td>
<p>A vector of values.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_start">start</code></td>
<td>
<p>The starting value of lambda to try.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_end">end</code></td>
<td>
<p>The ending value of lambda to try.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_int">int</code></td>
<td>
<p>The interval between lambda values to try.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_plotit">plotit</code></td>
<td>
<p>If <code>TRUE</code>, produces plots of Shapiro-Wilks W or 
Anderson-Darling A vs. lambda, a histogram of transformed
values, and a quantile-quantile plot of transformed values.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints extra output for Shapiro-Wilks
W or Anderson-Darling A vs. lambda.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_quiet">quiet</code></td>
<td>
<p>If <code>TRUE</code>, doesn't print any output to the screen.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_statistic">statistic</code></td>
<td>
<p>If <code>1</code>, uses Shapiro-Wilks test.
Will report <code>NA</code> if the sample size is greater than
5000.
If <code>2</code>, uses Anderson-Darling test.</p>
</td></tr>
<tr><td><code id="transformTukey_+3A_returnlambda">returnLambda</code></td>
<td>
<p>If <code>TRUE</code>, returns only the lambda value,
not the vector of transformed values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function simply loops through lamdba values from <code>start</code>
to <code>end</code> at an interval of <code>int</code>.
</p>
<p>The function then chooses the lambda which maximizes the 
Shapiro-Wilks W statistic or minimizes the Anderson-Darling A
statistic.
</p>
<p>It may be beneficial to add a constant to the input vector so that
all values are posititive.  For left-skewed data, a (Constant - X)
transformation may be helpful. Large values may need to be scaled.
</p>


<h3>Value</h3>

<p>The transformed vector of values. The chosen lambda value is printed
directly.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/I_12.html">https://rcompanion.org/handbook/I_12.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Log-normal distribution example
Conc = rlnorm(100)
Conc.trans = transformTukey(Conc)
                      
</code></pre>

<hr>
<h2 id='vda'>Vargha and Delaney's A</h2><span id='topic+vda'></span>

<h3>Description</h3>

<p>Calculates Vargha and Delaney's A (VDA)
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vda(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  reportIncomplete = FALSE,
  brute = FALSE,
  verbose = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vda_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="vda_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="vda_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="vda_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="vda_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="vda_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="vda_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="vda_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="vda_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="vda_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="vda_+3A_brute">brute</code></td>
<td>
<p>If <code>FALSE</code>, the default, the statistic is based on the
U statistic from the <code>wilcox.test</code> function.
If <code>TRUE</code>, the function will compare values
in the two samples directly.</p>
</td></tr>
<tr><td><code id="vda_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, reports the proportion of ties and
the proportions of (Ya &gt; Yb) and (Ya &lt; Yb).</p>
</td></tr>
<tr><td><code id="vda_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="vda_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcox.test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VDA is an effect size statistic appropriate
in cases where a Wilcoxon-Mann-Whitney test might be used.
It ranges from 0 to 1, with 0.5 indicating stochastic equality,
and 1 indicating that the first group dominates the second.
</p>
<p>By default, the function calculates VDA from the &quot;W&quot; U statistic
from the <code>wilcox.test</code> function.
Specifically, <code>VDA = U/(n1*n2)</code>.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>. If there are more than two groups,
only the first two groups are used.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, VDA is greater than 0.5.
When the data in the second group are greater than
in the first group, VDA is less than 0.5.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>When VDA is close to 0 or close to 1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, VDA.
Or a small data frame consisting of VDA,
and the lower and upper confidence limits.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_04.html">https://rcompanion.org/handbook/F_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cliffDelta">cliffDelta</a></code>, 
<code><a href="#topic+multiVDA">multiVDA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
vda(Steps ~ Gender, data=Catbus)

</code></pre>

<hr>
<h2 id='wilcoxonOneSampleR'>r effect size for Wilcoxon one-sample signed-rank test</h2><span id='topic+wilcoxonOneSampleR'></span>

<h3>Description</h3>

<p>Calculates r effect size
for a Wilcoxon one-sample signed-rank test; 
confidence intervals by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonOneSampleR(
  x,
  mu = NULL,
  adjustn = TRUE,
  coin = FALSE,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonOneSampleR_+3A_x">x</code></td>
<td>
<p>A vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_mu">mu</code></td>
<td>
<p>The value to compare <code>x</code> to, as in <code>wilcox.test</code></p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_adjustn">adjustn</code></td>
<td>
<p>If <code>TRUE</code>, reduces the sample size in the calculation
of <code>r</code> by the number of observations equal to
<code>mu</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_coin">coin</code></td>
<td>
<p>If <code>FALSE</code>, the default, the Z value
is extracted from a function similar to the
<code>wilcox.test</code> function in the stats package.
If <code>TRUE</code>, the Z value
is extracted from the <code>wilcox_test</code> function in the
coin package.  This method may be much slower, especially
if a confidence interval is produced.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleR_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcoxsign_test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>r  is calculated as Z divided by 
square root of the number of observations.
</p>
<p>The calculated statistic is equivalent to the statistic returned
by the <code>wilcoxPairedR</code> function with one group equal
to a vector of <code>mu</code>.
The author knows of no reference for this technique.
</p>
<p>This statistic typically reports a smaller effect size
(in absolute value) than does
the matched-pairs rank biserial correlation coefficient 
(<code>wilcoxonOneSampleRC</code>), and may not reach a value
of -1 or 1 if there are values tied with <code>mu</code>.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data are greater than <code>mu</code>, r is positive.
When the data are less than <code>mu</code>, r is negative.
</p>
<p>When r is close to extremes,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, r.
Or a small data frame consisting of r,
and the lower and upper confidence limits.
</p>


<h3>Acknowledgments</h3>

<p>My thanks to
Peter Stikker for the suggestion to adjust the sample size
for ties with <code>mu</code>.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_02.html">https://rcompanion.org/handbook/F_02.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X = c(1,2,3,3,3,3,4,4,4,4,4,5,5,5,5,5)
wilcox.test(X, mu=3, exact=FALSE)
wilcoxonOneSampleR(X, mu=3)

</code></pre>

<hr>
<h2 id='wilcoxonOneSampleRC'>Rank biserial correlation coefficient for one-sample Wilcoxon test</h2><span id='topic+wilcoxonOneSampleRC'></span>

<h3>Description</h3>

<p>Calculates rank biserial correlation coefficient
effect size
for one-sample Wilcoxon signed-rank test; confidence intervals
by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonOneSampleRC(
  x,
  mu = NULL,
  zero.method = "Wilcoxon",
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonOneSampleRC_+3A_x">x</code></td>
<td>
<p>A vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_mu">mu</code></td>
<td>
<p>The value to compare <code>x</code> to, as in <code>wilcox.test</code></p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_zero.method">zero.method</code></td>
<td>
<p>If <code>"Wilcoxon"</code>,
differences of zero are discarded and then ranks
are determined.
If <code>"Pratt"</code>,
ranks are determined, 
and then differences of zero are discarded.
If <code>"none"</code>,
differences of zero are not discarded.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints information on sample size and ranks.</p>
</td></tr>
<tr><td><code id="wilcoxonOneSampleRC_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcoxsign_test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When rc is close to extremes,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, rc.  
Or a small data frame consisting of rc,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_02.html">https://rcompanion.org/handbook/F_02.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wilcoxonPairedRC">wilcoxonPairedRC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example with one zero difference
A = c(11,12,13,14,15,16,17,18,19,20)
#' wilcoxonOneSampleRC(x = A, mu=15)
wilcoxonOneSampleRC(x = A, mu=15, verbose=TRUE, zero.method="Wilcoxon")
wilcoxonOneSampleRC(x = A, mu=15, verbose=TRUE, zero.method="Pratt")
wilcoxonOneSampleRC(x = A, mu=15, verbose=TRUE, zero.method="none")

</code></pre>

<hr>
<h2 id='wilcoxonOR'>Agresti's Generalized Odds Ratio for Stochastic Dominance</h2><span id='topic+wilcoxonOR'></span>

<h3>Description</h3>

<p>Calculates Agresti's Generalized Odds Ratio 
for Stochastic Dominance
(OR) with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonOR(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonOR_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, reports the proportion of ties and
the proportions of (Ya &gt; Yb) and (Ya &lt; Yb).</p>
</td></tr>
<tr><td><code id="wilcoxonOR_+3A_...">...</code></td>
<td>
<p>Additional arguments, not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>OR is an effect size statistic appropriate
in cases where a Wilcoxon-Mann-Whitney test might be used.
</p>
<p>OR is defined as P(Ya &gt; Yb) / P(Ya &lt; Yb).
</p>
<p>OR can range from 0 to infinity. An OR of 1 indicates stochastic 
equality between the two groups. An OR greater than 1 indicates
that the first group dominates the second group.
An OR less than 1 indicates
that the second group dominates the first.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>. If there are more than two groups,
only the first two groups are used.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>With a small sample size, or with an OR near its extremes,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, OR.
Or a small data frame consisting of OR,
and the lower and upper confidence limits.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>Grissom, R.J. and J.J. Kim. 2012. Effect Sizes for Research.
2nd ed. Routledge, New York.
</p>
<p><a href="https://rcompanion.org/handbook/F_04.html">https://rcompanion.org/handbook/F_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wilcoxonPS">wilcoxonPS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
wilcoxonOR(Steps ~ Gender, data=Catbus, verbose=TRUE)

</code></pre>

<hr>
<h2 id='wilcoxonPairedR'>r effect size for Wilcoxon two-sample paired signed-rank test</h2><span id='topic+wilcoxonPairedR'></span>

<h3>Description</h3>

<p>Calculates r effect size
for a Wilcoxon two-sample paired signed-rank test; 
confidence intervals by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonPairedR(
  x,
  g = NULL,
  adjustn = TRUE,
  coin = FALSE,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  cases = TRUE,
  digits = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonPairedR_+3A_x">x</code></td>
<td>
<p>A vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_g">g</code></td>
<td>
<p>The vector of observations for
the grouping, nominal variable.
Only the first two levels of the nominal variable are used.
The data must be ordered so that the first observation of the
of the first group is paired with the first observation of the
second group.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_adjustn">adjustn</code></td>
<td>
<p>If <code>TRUE</code>, reduces the sample size in the calculation
of <code>r</code> by the number of tied pairs.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_coin">coin</code></td>
<td>
<p>If <code>FALSE</code>, the default, the Z value
is extracted from a function similar to the
<code>wilcox.test</code> function in the stats package.
If <code>TRUE</code>, the Z value
is extracted from the <code>wilcox_test</code> function in the
coin package.  This method may be much slower, especially
if a confidence interval is produced.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_cases">cases</code></td>
<td>
<p>By default the <code>N</code> used in the formula for <code>r</code>
is the number of pairs.  If <code>cases=FALSE</code>,
the <code>N</code> used in the formula for <code>r</code>
is the total number of observations, as some sources suggest.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedR_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcoxsign_test</code> 
function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>r  is calculated as Z divided by 
square root of the number of observations in one group. This 
results in a statistic that ranges from -1 to 1.
This range doesn't hold if <code>cases=FALSE</code>.
</p>
<p>This statistic typically reports a smaller effect size
(in absolute value) than does
the matched-pairs rank biserial correlation coefficient 
(<code>wilcoxonPairedRC</code>), and may not reach a value
of -1 or 1 if there are ties in the paired differences.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, r is positive.
When the data in the second group are greater than
in the first group, r is negative.
Be cautious with this interpretation, as R will alphabetize
groups if <code>g</code> is not already a factor.
</p>
<p>When r is close to extremes,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, r.
Or a small data frame consisting of r,
and the lower and upper confidence limits.
</p>


<h3>Acknowledgments</h3>

<p>My thanks to
Peter Stikker for the suggestion to adjust the sample size for ties.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_06.html">https://rcompanion.org/handbook/F_06.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wilcoxonPairedRC">wilcoxonPairedRC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pooh)
Time1 = Pooh$Likert[Pooh$Time==1]
Time2 = Pooh$Likert[Pooh$Time==2]
wilcox.test(x = Time1, y = Time2, paired=TRUE, exact=FALSE)
wilcoxonPairedR(x = Pooh$Likert, g = Pooh$Time)

</code></pre>

<hr>
<h2 id='wilcoxonPairedRC'>Matched-pairs rank biserial correlation coefficient</h2><span id='topic+wilcoxonPairedRC'></span>

<h3>Description</h3>

<p>Calculates matched-pairs rank biserial correlation coefficient
effect size
for paired Wilcoxon signed-rank test; confidence intervals
by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonPairedRC(
  x,
  g = NULL,
  zero.method = "Wilcoxon",
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonPairedRC_+3A_x">x</code></td>
<td>
<p>A vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_g">g</code></td>
<td>
<p>The vector of observations for
the grouping, nominal variable.
Only the first two levels of the nominal variable are used.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_zero.method">zero.method</code></td>
<td>
<p>If <code>"Wilcoxon"</code>,
differences of zero are discarded and then ranks
are determined.
If <code>"Pratt"</code>,
ranks are determined, 
and then differences of zero are discarded.
If <code>"none"</code>,
differences of zero are not discarded.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints information on sample size and ranks.</p>
</td></tr>
<tr><td><code id="wilcoxonPairedRC_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>rank</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, rc is positive.
When the data in the second group are greater than
in the first group, rc is negative.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups if <code>g</code> is not already a factor.
</p>
<p>When rc is close to extremes,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, rc.  
Or a small data frame consisting of rc,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>King, B.M., P.J. Rosopa, and E.W. Minium. 2011. 
Statistical Reasoning in the Behavioral Sciences, 6th ed.
</p>
<p><a href="https://rcompanion.org/handbook/F_06.html">https://rcompanion.org/handbook/F_06.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wilcoxonPairedR">wilcoxonPairedR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pooh)
Time1 = Pooh$Likert[Pooh$Time==1]
Time2 = Pooh$Likert[Pooh$Time==2]
wilcox.test(x = Time1, y = Time2, paired=TRUE, exact=FALSE)
wilcoxonPairedRC(x = Pooh$Likert, g = Pooh$Time)

### Example from King, Rosopa, and Minium
Placebo = c(24,39,29,28,25,32,31,33,31,22)
Drug    = c(28,29,34,21,28,15,17,28,16,12)
Y = c(Placebo, Drug)
Group = factor(c(rep("Placebo", length(Placebo)),  
                 rep("Drug", length(Drug))), 
                 levels=c("Placebo", "Drug"))
wilcoxonPairedRC(x = Y, g = Group)

### Example with some zero differences
A = c(11,12,13,14,15,16,17,18,19,20)
B = c(12,14,16,18,20,22,12,10,19,20)
Y = c(A, B)
Group = factor(c(rep("A", length(A)),  
                 rep("B", length(B))))
wilcoxonPairedRC(x = Y, g = Group, verbose=TRUE, zero.method="Wilcoxon")
wilcoxonPairedRC(x = Y, g = Group, verbose=TRUE, zero.method="Pratt")
wilcoxonPairedRC(x = Y, g = Group, verbose=TRUE, zero.method="none")

</code></pre>

<hr>
<h2 id='wilcoxonPS'>Grissom and Kim's Probability of Superiority (PS)</h2><span id='topic+wilcoxonPS'></span>

<h3>Description</h3>

<p>Calculates Grissom and Kim's Probability of Superiority (PS)
with confidence intervals by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonPS(
  formula = NULL,
  data = NULL,
  x = NULL,
  y = NULL,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonPS_+3A_formula">formula</code></td>
<td>
<p>A formula indicating the response variable and
the independent variable. e.g. y ~ group.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_data">data</code></td>
<td>
<p>The data frame to use.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_x">x</code></td>
<td>
<p>If no formula is given, the response variable for one group.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_y">y</code></td>
<td>
<p>The response variable for the other group.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, reports the proportion of ties and
the proportions of (Ya &gt; Yb) and (Ya &lt; Yb).</p>
</td></tr>
<tr><td><code id="wilcoxonPS_+3A_...">...</code></td>
<td>
<p>Additional arguments, not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PS is an effect size statistic appropriate
in cases where a Wilcoxon-Mann-Whitney test might be used.
It ranges from 0 to 1, with 0.5 indicating stochastic equality,
and 1 indicating that the first group dominates the second.
</p>
<p>PS is defined as P(Ya &gt; Yb), with no provision made for tied values
across groups.
</p>
<p>If there are no tied values, PS will be equal to VDA. 
</p>
<p>The input should include either <code>formula</code> and <code>data</code>;
or <code>x</code>, and <code>y</code>. If there are more than two groups,
only the first two groups are used.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, PS is greater than 0.5.
When the data in the second group are greater than
in the first group, PS is less than 0.5.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups in the formula interface if the grouping variable
is not already a factor.
</p>
<p>When PS is close to 0 or close to 1,
or with small sample size,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, PS.
Or a small data frame consisting of PS,
and the lower and upper confidence limits.
</p>


<h3>Note</h3>

<p>The parsing of the formula is simplistic. 
The first variable on the
left side is used as the measurement variable.  
The first variable on the
right side is used for the grouping variable.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>Grissom, R.J. and J.J. Kim. 2012. Effect Sizes for Research. 
2nd ed. Routledge, New York.
</p>
<p><a href="https://rcompanion.org/handbook/F_04.html">https://rcompanion.org/handbook/F_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cliffDelta">cliffDelta</a></code>, 
<code><a href="#topic+vda">vda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Catbus)
wilcoxonPS(Steps ~ Gender, data=Catbus, verbose=TRUE)

</code></pre>

<hr>
<h2 id='wilcoxonR'>r effect size for Wilcoxon two-sample rank-sum test</h2><span id='topic+wilcoxonR'></span>

<h3>Description</h3>

<p>Calculates r effect size
for Mann-Whitney two-sample rank-sum test,
or a table with an ordinal variable and a
nominal variable with two levels; confidence intervals
by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonR(
  x,
  g = NULL,
  group = "row",
  coin = FALSE,
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonR_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_g">g</code></td>
<td>
<p>If <code>x</code> is a vector, <code>g</code> is the vector of observations for
the grouping, nominal variable.
Only the first two levels of the nominal variable are used.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_group">group</code></td>
<td>
<p>If <code>x</code> is a table or matrix, <code>group</code> indicates whether
the <code>"row"</code> or the <code>"column"</code> variable is
the nominal, grouping variable.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_coin">coin</code></td>
<td>
<p>If <code>FALSE</code>, the default, the Z value
is extracted from a function similar to the
<code>wilcox.test</code> function in the stats package.
If <code>TRUE</code>, the Z value
is extracted from the <code>wilcox_test</code> function in the
coin package.  This method may be much slower, especially
if a confidence interval is produced.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="wilcoxonR_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>wilcox_test</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>r  is calculated as Z divided by 
square root of the total observations.
</p>
<p>This statistic reports a smaller effect size than does
Glass rank biserial correlation coefficient 
(<code>wilcoxonRG</code>), and cannot reach 
-1 or 1.  This effect is exaserbated when sample sizes
are not equal.
</p>
<p>Currently, the function makes no provisions for <code>NA</code>
values in the data.  It is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, r is positive.
When the data in the second group are greater than
in the first group, r is negative.
Be cautious with this interpretation, as R will alphabetize
groups if <code>g</code> is not already a factor.
</p>
<p>When r is close to extremes,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, r.  
Or a small data frame consisting of r,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p><a href="https://rcompanion.org/handbook/F_04.html">https://rcompanion.org/handbook/F_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+freemanTheta">freemanTheta</a></code>, 
<code><a href="#topic+wilcoxonRG">wilcoxonRG</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Breakfast)
Table = Breakfast[1:2,]
library(coin)
chisq_test(Table, scores = list("Breakfast" = c(-2, -1, 0, 1, 2)))
wilcoxonR(Table)

data(Catbus)
wilcox.test(Steps ~ Gender, data = Catbus)
wilcoxonR(x = Catbus$Steps, g = Catbus$Gender)

</code></pre>

<hr>
<h2 id='wilcoxonRG'>Glass rank biserial correlation coefficient</h2><span id='topic+wilcoxonRG'></span>

<h3>Description</h3>

<p>Calculates Glass rank biserial correlation coefficient
effect size
for Mann-Whitney two-sample rank-sum test,
or a table with an ordinal variable and a
nominal variable with two levels; confidence intervals
by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonRG(
  x,
  g = NULL,
  group = "row",
  ci = FALSE,
  conf = 0.95,
  type = "perc",
  R = 1000,
  histogram = FALSE,
  digits = 3,
  reportIncomplete = FALSE,
  verbose = FALSE,
  na.last = NA,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonRG_+3A_x">x</code></td>
<td>
<p>Either a two-way table or a two-way matrix.
Can also be a vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_g">g</code></td>
<td>
<p>If <code>x</code> is a vector, <code>g</code> is the vector of observations for
the grouping, nominal variable.
Only the first two levels of the nominal variable are used.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_group">group</code></td>
<td>
<p>If <code>x</code> is a table or matrix, <code>group</code> indicates whether
the <code>"row"</code> or the <code>"column"</code> variable is
the nominal, grouping variable.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_ci">ci</code></td>
<td>
<p>If <code>TRUE</code>, returns confidence intervals by bootstrap.
May be slow.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_conf">conf</code></td>
<td>
<p>The level for the confidence interval.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_type">type</code></td>
<td>
<p>The type of confidence interval to use.
Can be any of &quot;<code>norm</code>&quot;, &quot;<code>basic</code>&quot;, 
&quot;<code>perc</code>&quot;, or &quot;<code>bca</code>&quot;.
Passed to <code>boot.ci</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_r">R</code></td>
<td>
<p>The number of replications to use for bootstrap.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_histogram">histogram</code></td>
<td>
<p>If <code>TRUE</code>, produces a histogram of bootstrapped values.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_reportincomplete">reportIncomplete</code></td>
<td>
<p>If <code>FALSE</code> (the default),
<code>NA</code> will be reported in cases where there
are instances of the calculation of the statistic
failing during the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, prints information on factor levels and ranks.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_na.last">na.last</code></td>
<td>
<p>Passed to <code>rank</code>. For example, can be set to
<code>TRUE</code> to assign <code>NA</code> values a minimum rank.</p>
</td></tr>
<tr><td><code id="wilcoxonRG_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>rank</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>rg  is calculated as 2 times the difference of mean of ranks
for each group divided by the total sample size.
It appears that rg is equivalent to Cliff's delta.
</p>
<p><code>NA</code> values can be handled by the <code>rank</code> function.
In this case, using <code>verbose=TRUE</code> is helpful
to understand how the <code>rg</code> statistic is calculated.
Otherwise, it is recommended that <code>NA</code>s be removed
beforehand.
</p>
<p>When the data in the first group are greater than
in the second group, rg is positive.
When the data in the second group are greater than
in the first group, rg is negative.
</p>
<p>Be cautious with this interpretation, as R will alphabetize
groups if <code>g</code> is not already a factor.
</p>
<p>When rg is close to extremes,
or with small counts in some cells,
the confidence intervals 
determined by this
method may not be reliable, or the procedure may fail.
</p>


<h3>Value</h3>

<p>A single statistic, rg.  
Or a small data frame consisting of rg,
and the lower and upper confidence limits.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>
</p>


<h3>References</h3>

<p>King, B.M., P.J. Rosopa, and E.W. Minium. 2011. 
Statistical Reasoning in the Behavioral Sciences, 6th ed.
</p>
<p><a href="https://rcompanion.org/handbook/F_04.html">https://rcompanion.org/handbook/F_04.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wilcoxonR">wilcoxonR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Breakfast)
Table = Breakfast[1:2,]
library(coin)
chisq_test(Table, scores = list("Breakfast" = c(-2, -1, 0, 1, 2)))
wilcoxonRG(Table)

data(Catbus)
wilcox.test(Steps ~ Gender, data = Catbus)
wilcoxonRG(x = Catbus$Steps, g = Catbus$Gender)

### Example from King, Rosopa, and Minium
Criticism = c(-3, -2, 0, 0, 2, 5, 7, 9)
Praise = c(0, 2, 3, 4, 10, 12, 14, 19, 21)
Y = c(Criticism, Praise)
Group = factor(c(rep("Criticism", length(Criticism)),  
                rep("Praise", length(Praise))))
wilcoxonRG(x = Y, g = Group, verbose=TRUE)

</code></pre>

<hr>
<h2 id='wilcoxonZ'>Wilcoxon z statistic</h2><span id='topic+wilcoxonZ'></span>

<h3>Description</h3>

<p>Calculates the z statistic for a Wilcoxon
two-sample, paired, or one-sample test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxonZ(
  x,
  y = NULL,
  mu = 0,
  paired = FALSE,
  exact = FALSE,
  correct = FALSE,
  digits = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxonZ_+3A_x">x</code></td>
<td>
<p>A vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonZ_+3A_y">y</code></td>
<td>
<p>For the two-sample and paired cases,
a second vector of observations.</p>
</td></tr>
<tr><td><code id="wilcoxonZ_+3A_mu">mu</code></td>
<td>
<p>For the one-sample case,
the value to compare <code>x</code> to, as in <code>wilcox.test</code></p>
</td></tr>
<tr><td><code id="wilcoxonZ_+3A_paired">paired</code></td>
<td>
<p>As used in <code>wilcox.test</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonZ_+3A_exact">exact</code></td>
<td>
<p>As used in <code>wilcox.test</code>, 
default here is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonZ_+3A_correct">correct</code></td>
<td>
<p>As used in <code>wilcox.test</code>, 
default here is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="wilcoxonZ_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits in the output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses code from <code>wilcox.test</code>,
and reports the <code>z</code> statistic,
which is calculated by the original function
but isn't returned.
</p>
<p>The returned value will be NA if the function attempts an
exact test.
</p>
<p>For the paired case, the observations in <code>x</code> and
and <code>y</code> should be ordered such that the
first observation in <code>x</code> is paired with the first observation
in <code>y</code>, and so on.
</p>


<h3>Value</h3>

<p>A single statistic, <code>z</code>.
</p>


<h3>Author(s)</h3>

<p>Salvatore Mangiafico, <a href="mailto:mangiafico@njaes.rutgers.edu">mangiafico@njaes.rutgers.edu</a>, 
R Core Team
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pooh)
wilcoxonZ(x = Pooh$Likert[Pooh$Time==1], y = Pooh$Likert[Pooh$Time==2],
          paired=TRUE, exact=FALSE, correct=FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
