<!DOCTYPE html><html><head><title>Help for package RCPA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RCPA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#downloadGEO'><p>Download GEO data</p></a></li>
<li><a href='#getCePaPathwayCatalogue'><p>Get KEGG pathway catalogue (network) for CePa.ORA and CePa.GSA methods</p></a></li>
<li><a href='#getCommonDEGenes'><p>Get common significant DE genes from multiple DE Analysis results</p></a></li>
<li><a href='#getCommonPathways'><p>Retrieve common significant pathways from multiple pathway analysis results</p></a></li>
<li><a href='#getEntrezAnnotation'><p>Get Entrez annotation</p></a></li>
<li><a href='#getGeneSets'><p>Get gene sets</p></a></li>
<li><a href='#getSPIAKEGGNetwork'><p>Get KEGG pathway network for SPIA method</p></a></li>
<li><a href='#getSupportedPlatforms'><p>Get supported platforms</p></a></li>
<li><a href='#loadData'><p>Load data from GitHub</p></a></li>
<li><a href='#plotBarChart'><p>Plot a bar chart of the pathway analysis results</p></a></li>
<li><a href='#plotDEGeneHeatmap'><p>Plot gene heatmap from a SummarizedExperiment object with DE analysis results</p></a></li>
<li><a href='#plotForest'><p>Plot pathway forest plot from pathway/geneset/meta analysis results</p></a></li>
<li><a href='#plotKEGGMap'><p>Plot KEGG map with DE genes</p></a></li>
<li><a href='#plotMA'><p>Plot MA plot from DE analysis results</p></a></li>
<li><a href='#plotPathwayHeatmap'><p>Plot pathways heatmap plot from pathway/geneset/meta analysis results</p></a></li>
<li><a href='#plotPathwayNetwork'><p>Plot a pathway network</p></a></li>
<li><a href='#plotVennDE'><p>Plot Venn diagram from multiple DE Analysis results</p></a></li>
<li><a href='#plotVennPathway'><p>Plot Venn diagram from multiple pathway analysis results</p></a></li>
<li><a href='#plotVolcanoDE'><p>Plot volcano plot from Pathway analysis results</p></a></li>
<li><a href='#plotVolcanoPathway'><p>Plot volcano plot from Pathway analysis results</p></a></li>
<li><a href='#processAffymetrix'><p>Process and normalize affymetrix-based dataset</p></a></li>
<li><a href='#processAgilent'><p>Process and normalize agilent-based dataset</p></a></li>
<li><a href='#runConsensusAnalysis'><p>Perform Consensus Analysis</p></a></li>
<li><a href='#runDEAnalysis'><p>Differential expression analysis</p></a></li>
<li><a href='#runDEMetaAnalysis'><p>Combine DE analysis results</p></a></li>
<li><a href='#runGeneSetAnalysis'><p>Gene Set Enrichment Analysis</p></a></li>
<li><a href='#runPathwayAnalysis'><p>Topology-based Pathway Analysis</p></a></li>
<li><a href='#runPathwayMetaAnalysis'><p>Perform Meta Analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Consensus Pathway Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a set of functions to perform pathway analysis and meta analysis from multiple gene expression datasets, as well as visualization of the results. The package is a wrapper of the following packages: Ritchie et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkv007">doi:10.1093/nar/gkv007</a>&gt;, Love et al. (2014) &lt;<a href="https://doi.org/10.1186%2Fs13059-014-0550-8">doi:10.1186/s13059-014-0550-8</a>&gt;, Robinson et al. (2010) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtp616">doi:10.1093/bioinformatics/btp616</a>&gt;, Korotkevich et al. (2016) &lt;<a href="https://arxiv.org/abs/10.1101/060012">arXiv:10.1101/060012</a>&gt;, Efron et al. (2015) <a href="https://CRAN.R-project.org/package=GSA">https://CRAN.R-project.org/package=GSA</a>, and Gu, Z. (2012) <a href="https://CRAN.R-project.org/package=CePa">https://CRAN.R-project.org/package=CePa</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), tidyr, dplyr, ggplot2, utils, stats</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Biobase, DESeq2, GEOquery, edgeR, limma, RCyjs, fgsea,
BrowserViz, SummarizedExperiment, AnnotationDbi, ROntoTools</td>
</tr>
<tr>
<td>Imports:</td>
<td>AnnotationDbi, SummarizedExperiment, BiocManager, Biobase,
DESeq2, GEOquery, edgeR, limma, stringr(&ge; 1.5.0), ggnewscale,
ggrepel, graph, httr, rlang, ggpattern, scales,
RobustRankAggreg, methods, jsonlite, IRdisplay</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown, testthat (&ge; 3.0.0), ROntoTools,
fgsea, GSA, CePa, meta, png, ggvenn, RCurl, XML, S4Vectors,
hgu133plus2.db, org.Hs.eg.db, oligo</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Config/testthat/stop-on-test-error:</td>
<td>false</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/tinnlab/RCPA/issues">https://github.com/tinnlab/RCPA/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-28 20:53:29 UTC; hanguyen</td>
</tr>
<tr>
<td>Author:</td>
<td>Ha Nguyen [aut, cre],
  Zeynab Maghsoudi [aut],
  Phi Hung Bya [aut],
  Tin Nguyen [fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ha Nguyen &lt;hvn0006@auburn.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-30 09:40:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='downloadGEO'>Download GEO data</h2><span id='topic+downloadGEO'></span>

<h3>Description</h3>

<p>This function download and process data from GEO for microarray and RNASeq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downloadGEO(GEOID, protocol = c("affymetrix", "agilent"), platform, destDir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="downloadGEO_+3A_geoid">GEOID</code></td>
<td>
<p>The ID of the GEO dataset.</p>
</td></tr>
<tr><td><code id="downloadGEO_+3A_protocol">protocol</code></td>
<td>
<p>The protocol of selected GEO dataset.</p>
</td></tr>
<tr><td><code id="downloadGEO_+3A_platform">platform</code></td>
<td>
<p>The platform of selected GEO dataset.</p>
</td></tr>
<tr><td><code id="downloadGEO_+3A_destdir">destDir</code></td>
<td>
<p>A path to save downloaded data. If the directory does not exist, it will be created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of file paths to the downloaded files. The first element is the metadata file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
# Affymetrix
downloadPath &lt;- file.path(tempdir(), "GSE59761")
fileList &lt;- RCPA::downloadGEO(GEOID = "GSE59761", protocol = "affymetrix",
                                platform ="GPL16311", destDir = downloadPath)

</code></pre>

<hr>
<h2 id='getCePaPathwayCatalogue'>Get KEGG pathway catalogue (network) for CePa.ORA and CePa.GSA methods</h2><span id='topic+getCePaPathwayCatalogue'></span>

<h3>Description</h3>

<p>Get KEGG pathway catalogue for CePa.ORA and CePa.GSA methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCePaPathwayCatalogue(org = "hsa", updateCache = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCePaPathwayCatalogue_+3A_org">org</code></td>
<td>
<p>The organism abbreviation. E.g, hsa, mmu, dme, etc.
To see the full list of supported organisms, visit https://www.genome.jp/kegg/catalog/org_list.html.</p>
</td></tr>
<tr><td><code id="getCePaPathwayCatalogue_+3A_updatecache">updateCache</code></td>
<td>
<p>A parameter to enable/disable cache update.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with three elements: network, names and sizes for CePa.ORA and CePa.GSA methods.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cepaNetwork &lt;- getCePaPathwayCatalogue("hsa")

</code></pre>

<hr>
<h2 id='getCommonDEGenes'>Get common significant DE genes from multiple DE Analysis results</h2><span id='topic+getCommonDEGenes'></span>

<h3>Description</h3>

<p>Get a list of common significant DE genes from multiple DE Analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCommonDEGenes(
  DEResults,
  pThreshold = 0.05,
  useFDR = TRUE,
  stat = "logFC",
  statThreshold = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCommonDEGenes_+3A_deresults">DEResults</code></td>
<td>
<p>A list of data frames with the results of DE analysis.</p>
</td></tr>
<tr><td><code id="getCommonDEGenes_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to determine if a gene is differentially expressed.</p>
</td></tr>
<tr><td><code id="getCommonDEGenes_+3A_usefdr">useFDR</code></td>
<td>
<p>Use the FDR adjusted p-value instead of the nominal p-value.</p>
</td></tr>
<tr><td><code id="getCommonDEGenes_+3A_stat">stat</code></td>
<td>
<p>The additional statistics column to use for filtering differentially expressed genes.</p>
</td></tr>
<tr><td><code id="getCommonDEGenes_+3A_statthreshold">statThreshold</code></td>
<td>
<p>The absolute value of the statistic threshold to use for filtering differentially expressed genes.
Default is 0, which means no filtering.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame wtih three columns: ID (Entrez IDs), Symbol and Description
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)

affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

DEResults &lt;- list(
    "Affymetrix - GSE5281" = rowData(affyDEExperiment),
    "Agilent - GSE61196"   = rowData(agilDEExperiment),
    "RNASeq - GSE153873"   = rowData(RNASeqDEExperiment)
)

commonDEGenes &lt;- RCPA::getCommonDEGenes(DEResults)

print(head(commonDEGenes))

</code></pre>

<hr>
<h2 id='getCommonPathways'>Retrieve common significant pathways from multiple pathway analysis results</h2><span id='topic+getCommonPathways'></span>

<h3>Description</h3>

<p>Query a list of common significant pathways from multiple pathway analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCommonPathways(PAResults, pThreshold = 0.05, useFDR = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCommonPathways_+3A_paresults">PAResults</code></td>
<td>
<p>A list of data frames with the results of pathway analysis.</p>
</td></tr>
<tr><td><code id="getCommonPathways_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to determine if a pathway is enriched or significant.</p>
</td></tr>
<tr><td><code id="getCommonPathways_+3A_usefdr">useFDR</code></td>
<td>
<p>Use the FDR adjusted p-value instead of the nominal p-value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame contains pathway ID and pathway names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)


affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")

PAResults &lt;- list(
    "Affymetrix - GSE5281" = affyFgseaResult,
    "Agilent - GSE61196" = agilFgseaResult,
    "RNASeq - GSE153873" = RNASeqFgseaResult
)

commonPathways &lt;- RCPA::getCommonPathways(PAResults)

print(head(commonPathways))



</code></pre>

<hr>
<h2 id='getEntrezAnnotation'>Get Entrez annotation</h2><span id='topic+getEntrezAnnotation'></span>

<h3>Description</h3>

<p>This function gets Entrez annotation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEntrezAnnotation(entrezIds)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEntrezAnnotation_+3A_entrezids">entrezIds</code></td>
<td>
<p>A vector of Entrez IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with Entrez annotation. The columns are ID (Entrez ID), Symbol, Description, OtherDesignations, OtherAliases and Chromosome.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
geneAnno &lt;- getEntrezAnnotation(c("77267466", "77267467"))

</code></pre>

<hr>
<h2 id='getGeneSets'>Get gene sets</h2><span id='topic+getGeneSets'></span>

<h3>Description</h3>

<p>This function retrieves gene sets for a given organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneSets(
  database = c("KEGG", "GO"),
  org = "hsa",
  taxid = 9606,
  namespace = c("biological_process", "molecular_function", "cellular_component"),
  minSize = 1,
  maxSize = 1000,
  useCache = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGeneSets_+3A_database">database</code></td>
<td>
<p>The database of the gene sets. E.g, KEGG, GO.</p>
</td></tr>
<tr><td><code id="getGeneSets_+3A_org">org</code></td>
<td>
<p>The organism abbreviation. E.g, hsa, mmu, dme, etc.
To see the full list of supported organisms, visit https://www.genome.jp/kegg/catalog/org_list.html.
This parameter is only used when database is KEGG.</p>
</td></tr>
<tr><td><code id="getGeneSets_+3A_taxid">taxid</code></td>
<td>
<p>The NCBI taxonomy ID of the organism.
This parameter is only used when database is GO.</p>
</td></tr>
<tr><td><code id="getGeneSets_+3A_namespace">namespace</code></td>
<td>
<p>The namespace of the GO terms. E.g, biological_process, molecular_function, cellular_component.</p>
</td></tr>
<tr><td><code id="getGeneSets_+3A_minsize">minSize</code></td>
<td>
<p>The minimum size of the gene sets.</p>
</td></tr>
<tr><td><code id="getGeneSets_+3A_maxsize">maxSize</code></td>
<td>
<p>The maximum size of the gene sets.</p>
</td></tr>
<tr><td><code id="getGeneSets_+3A_usecache">useCache</code></td>
<td>
<p>A boolean parameter specifying if using pre-saved downloaded geneset database. It is FALSE by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with three elements: database, genesets and names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)

KEGGgenesets &lt;- getGeneSets("KEGG", org = "hsa", 
                              minSize = 10, maxSize = 1000, useCache = TRUE)

GOterms &lt;- getGeneSets("GO", taxid = 9606, 
                        namespace = "biological_process", 
                        minSize = 10, maxSize = 1000, useCache = TRUE)


</code></pre>

<hr>
<h2 id='getSPIAKEGGNetwork'>Get KEGG pathway network for SPIA method</h2><span id='topic+getSPIAKEGGNetwork'></span>

<h3>Description</h3>

<p>Get KEGG pathway network for SPIA method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSPIAKEGGNetwork(org = "hsa", updateCache = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSPIAKEGGNetwork_+3A_org">org</code></td>
<td>
<p>The organism abbreviation. E.g, hsa, mmu, dme, etc.
To see the full list of supported organisms, visit https://www.genome.jp/kegg/catalog/org_list.html.</p>
</td></tr>
<tr><td><code id="getSPIAKEGGNetwork_+3A_updatecache">updateCache</code></td>
<td>
<p>A parameter to disable/enable cache update.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with three elements: network, names and sizes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
spiaNetwork &lt;- getSPIAKEGGNetwork("hsa")

</code></pre>

<hr>
<h2 id='getSupportedPlatforms'>Get supported platforms</h2><span id='topic+getSupportedPlatforms'></span>

<h3>Description</h3>

<p>Get supported platforms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSupportedPlatforms()
</code></pre>


<h3>Value</h3>

<p>A named list of supported platforms, where the names are the platform IDs and the values are the corresponding annotation packages from Bioconductor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
supportedPlatforms &lt;- getSupportedPlatforms()

</code></pre>

<hr>
<h2 id='loadData'>Load data from GitHub</h2><span id='topic+loadData'></span>

<h3>Description</h3>

<p>This function loads data from GitHub.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadData(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadData_+3A_name">name</code></td>
<td>
<p>The name of the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Load the data with the specified name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
RNASeqDataset &lt;- loadData("RNASeqDataset")


</code></pre>

<hr>
<h2 id='plotBarChart'>Plot a bar chart of the pathway analysis results</h2><span id='topic+plotBarChart'></span>

<h3>Description</h3>

<p>This function plots a bar chart of the pathway analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBarChart(
  results,
  limit = Inf,
  label = "name",
  by = c("normalizedScore", "score", "pFDR", "p.value"),
  maxNegLog10PValue = 5,
  pThreshold = 0.05,
  useFDR = TRUE,
  selectedPathways = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBarChart_+3A_results">results</code></td>
<td>
<p>A named list of data frame with pathway analysis results.
The columns are ID, name, p.value, pFDR, size, nDE, score and normalizedScore.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_limit">limit</code></td>
<td>
<p>The maximum number of pathways to plot.
The pathway will be sorted by the average absolute value of the -log10(p-value) or -log10(pFDR) if the 'by' parameter is 'p.value' or 'pFDR'.
Otherwise, the pathway will be sorted by the average value of the 'by' parameter.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_label">label</code></td>
<td>
<p>The column to use for the labels.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_by">by</code></td>
<td>
<p>The column to use for the bar heights.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_maxneglog10pvalue">maxNegLog10PValue</code></td>
<td>
<p>The maximum -log10(p-value) to plot.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to use for significance.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_usefdr">useFDR</code></td>
<td>
<p>If TRUE, use FDR adjusted p-values for the significance threshold. Otherwise, use raw p-values.
This parameter is used to mark the color of the bars and is independent of the 'by' parameter.</p>
</td></tr>
<tr><td><code id="plotBarChart_+3A_selectedpathways">selectedPathways</code></td>
<td>
<p>A vector of pathways ID, which is in the same format as ID column in the pathway analysis result, to be included in the plot.
If it is NULL, all pathways will be included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)

affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")
metaPAResult &lt;- loadData("metaPAResult")

PAResults &lt;- list(
    "Affymetrix - GSE5281" = affyFgseaResult,
    "Agilent - GSE61196" = agilFgseaResult,
    "RNASeq - GSE153873" = RNASeqFgseaResult,
    "Meta-analysis" = metaPAResult
)

selectedPathways &lt;- c("path:hsa05010", "path:hsa05012", "path:hsa05014", 
                       "path:hsa05016", "path:hsa05017", "path:hsa05020", 
                       "path:hsa05022", "path:hsa04724", "path:hsa04727", 
                       "path:hsa04725", "path:hsa04728", "path:hsa04726",
                      "path:hsa04720", "path:hsa04730", "path:hsa04723", 
                      "path:hsa04721", "path:hsa04722")
resultsToPlot &lt;- lapply(PAResults, 
                    function(df) df[df$ID %in% selectedPathways,])

plotObj &lt;- RCPA::plotBarChart(resultsToPlot) + 
            ggplot2::ggtitle("FGSEA Analysis Results")


</code></pre>

<hr>
<h2 id='plotDEGeneHeatmap'>Plot gene heatmap from a SummarizedExperiment object with DE analysis results</h2><span id='topic+plotDEGeneHeatmap'></span>

<h3>Description</h3>

<p>Plot gene heatmap from a SummarizedExperiment object with DE analysis results.
The heatmap contains p-values and log fold changes from the DE analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDEGeneHeatmap(
  DEResults,
  genes,
  useFDR = TRUE,
  labels = NULL,
  logFCLims = c(-5, 5),
  negLog10pValueLims = c(0, 5)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDEGeneHeatmap_+3A_deresults">DEResults</code></td>
<td>
<p>A named list of data frame of DE analysis results.</p>
</td></tr>
<tr><td><code id="plotDEGeneHeatmap_+3A_genes">genes</code></td>
<td>
<p>A vector of gene id (e.g. Entrez IDs) to plot.
The genes must be in the ID column of the data frame in DEResults.</p>
</td></tr>
<tr><td><code id="plotDEGeneHeatmap_+3A_usefdr">useFDR</code></td>
<td>
<p>If TRUE, use FDR adjusted p-values. Otherwise, use raw p-values.</p>
</td></tr>
<tr><td><code id="plotDEGeneHeatmap_+3A_labels">labels</code></td>
<td>
<p>A vector of labels for the genes. If not provided, the gene IDs will be used as labels.</p>
</td></tr>
<tr><td><code id="plotDEGeneHeatmap_+3A_logfclims">logFCLims</code></td>
<td>
<p>A vector of length 2 specifying the minimum and maximum log fold change to plot.</p>
</td></tr>
<tr><td><code id="plotDEGeneHeatmap_+3A_neglog10pvaluelims">negLog10pValueLims</code></td>
<td>
<p>A vector of length 2 specifying the minimum and maximum -log10(p-value) to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap of the genes from ggplot2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)

affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")
metaDEResult &lt;- loadData("metaDEResult")
genesets &lt;- loadData("genesets")

DEResults &lt;- list(
    "Affymetrix - GSE5281" = rowData(affyDEExperiment),
    "Agilent - GSE61196" = rowData(agilDEExperiment),
    "RNASeq - GSE153873" = rowData(RNASeqDEExperiment)
)

 metaDEResult &lt;- metaDEResult[order(metaDEResult$pFDR),]

 alzheimerGenes &lt;- genesets$genesets[["path:hsa05010"]]
 genesToPlot &lt;- head(metaDEResult[metaDEResult$ID %in% alzheimerGenes, ], 50)$ID

 genesAnnotation &lt;- RCPA::getEntrezAnnotation(genesToPlot)
 labels &lt;- genesAnnotation[genesToPlot, "Description"]

 genesOrderByFC &lt;- order(metaDEResult[match(genesToPlot, metaDEResult$ID), "logFC"])
 resultsToPlot &lt;- c(DEResults, list(metaDEResult))
 names(resultsToPlot) &lt;- c(names(DEResults), "Meta-analysis")

 plotObj &lt;- RCPA::plotDEGeneHeatmap(
     resultsToPlot,
     genesToPlot[genesOrderByFC],
     labels = labels[genesOrderByFC],
     negLog10pValueLims = c(0, 5), logFCLims = c(-1, 1)
 )



</code></pre>

<hr>
<h2 id='plotForest'>Plot pathway forest plot from pathway/geneset/meta analysis results</h2><span id='topic+plotForest'></span>

<h3>Description</h3>

<p>pathways heatmap plot from pathway/geneset/meta analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotForest(
  resultsList,
  yAxis = c("ID", "name"),
  statLims = c(-2.5, 2.5),
  useFDR = TRUE,
  selectedPathways = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotForest_+3A_resultslist">resultsList</code></td>
<td>
<p>A named list of dataframes from pathway analysis, geneset analysis, and/or meta analysis results.
The columns are ID, name, description, p.value, pFDR, size, nDE, score and normalizedScore.</p>
</td></tr>
<tr><td><code id="plotForest_+3A_yaxis">yAxis</code></td>
<td>
<p>The column to use for the y-axis.</p>
</td></tr>
<tr><td><code id="plotForest_+3A_statlims">statLims</code></td>
<td>
<p>A numeric vector of length 2 specifying the limits for score to use in the x-axis.</p>
</td></tr>
<tr><td><code id="plotForest_+3A_usefdr">useFDR</code></td>
<td>
<p>Logical to indicate whether to use FDR or p-value.</p>
</td></tr>
<tr><td><code id="plotForest_+3A_selectedpathways">selectedPathways</code></td>
<td>
<p>A vector of pathways ID, which is in the same format as ID column in the pathway analysis result, to be included in the plot.
If it is NULL, all pathways will be included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object for presenting the heatmap of the pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")
metaPAResult &lt;- loadData("metaPAResult")

PAResults &lt;- list(
    "Affymetrix - GSE5281" = affyFgseaResult,
    "Agilent - GSE61196" = agilFgseaResult,
    "RNASeq - GSE153873" = RNASeqFgseaResult,
    "Meta-analysis" = metaPAResult
)

selectedPathways &lt;- c("path:hsa05010", "path:hsa05012", "path:hsa05014", "path:hsa05016",
                      "path:hsa05017", "path:hsa05020", "path:hsa05022", "path:hsa04724",
                      "path:hsa04727", "path:hsa04725", "path:hsa04728", "path:hsa04726",
                      "path:hsa04720", "path:hsa04730", "path:hsa04723", "path:hsa04721",
                      "path:hsa04722")
resultsToPlot &lt;- lapply(PAResults, function(df) df[df$ID %in% selectedPathways,])

plotObj &lt;- RCPA::plotForest(resultsToPlot, yAxis = "name", statLims = c(-3.5, 3.5))


</code></pre>

<hr>
<h2 id='plotKEGGMap'>Plot KEGG map with DE genes</h2><span id='topic+plotKEGGMap'></span>

<h3>Description</h3>

<p>This function plots KEGG map with DE genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotKEGGMap(
  DEResults,
  KEGGPathwayID,
  statistic = "logFC",
  useFDR = TRUE,
  pThreshold = 0.05,
  statLimit = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotKEGGMap_+3A_deresults">DEResults</code></td>
<td>
<p>A named list of data frame of DE analysis results.
The columns of each data frame should be at least ID, logFC, p.value and pFDR.</p>
</td></tr>
<tr><td><code id="plotKEGGMap_+3A_keggpathwayid">KEGGPathwayID</code></td>
<td>
<p>The KEGG pathway ID.</p>
</td></tr>
<tr><td><code id="plotKEGGMap_+3A_statistic">statistic</code></td>
<td>
<p>The column name of the statistic used to plot the DE genes.
If statistic is p.value or pFDR, all genes are colored.
Otherwise, only DE genes are colored.</p>
</td></tr>
<tr><td><code id="plotKEGGMap_+3A_usefdr">useFDR</code></td>
<td>
<p>If TRUE, DE genes are selected based on pFDR, otherwise p.value.</p>
</td></tr>
<tr><td><code id="plotKEGGMap_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to select DE genes.
Only used when statistic is not p.value or pFDR.</p>
</td></tr>
<tr><td><code id="plotKEGGMap_+3A_statlimit">statLimit</code></td>
<td>
<p>The absolute value of the statistic to color the DE genes.
If statistic is p.value or pFDR, this parameter is the limit of -log10(p-value).
Otherwise, this parameter is the limit of the absolute value of the statistic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<ul>
<li><p> plot: A ggplot object of the KEGG map.
</p>
</li>
<li><p> width: The width of the KEGG map.
</p>
</li>
<li><p> height: The height of the KEGG map.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)

affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

DEResults &lt;- list(
    "Affymetrix - GSE5281" = rowData(affyDEExperiment),
    "Agilent - GSE61196" = rowData(agilDEExperiment),
    "RNASeq - GSE153873" = rowData(RNASeqDEExperiment)
)

plotObj &lt;- RCPA::plotKEGGMap(DEResults, "hsa05010", stat = "logFC", pThreshold = 1, statLimit = 1)


</code></pre>

<hr>
<h2 id='plotMA'>Plot MA plot from DE analysis results</h2><span id='topic+plotMA'></span>

<h3>Description</h3>

<p>Plot MA plot from DE analysis results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMA(
  DEResult,
  pThreshold = 0.05,
  useFDR = TRUE,
  logFCThreshold = 1,
  labels = NULL,
  fitMethod = "loess"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMA_+3A_deresult">DEResult</code></td>
<td>
<p>A data frame with DE analysis results.
The columns are ID, p.value, pFDR, logFC, and aveExpr.</p>
</td></tr>
<tr><td><code id="plotMA_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to color significant points.</p>
</td></tr>
<tr><td><code id="plotMA_+3A_usefdr">useFDR</code></td>
<td>
<p>Use FDR instead of p-value for significance.</p>
</td></tr>
<tr><td><code id="plotMA_+3A_logfcthreshold">logFCThreshold</code></td>
<td>
<p>The log2 fold change threshold to color significant points.</p>
</td></tr>
<tr><td><code id="plotMA_+3A_labels">labels</code></td>
<td>
<p>named vector of labels to use for points, e.g., c(&quot;gene1&quot; = &quot;Gene 1&quot;, &quot;gene2&quot; = &quot;Gene 2&quot;)</p>
</td></tr>
<tr><td><code id="plotMA_+3A_fitmethod">fitMethod</code></td>
<td>
<p>The method to use for fitting the loess line.
If NULL then no line is drawn.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)

RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

plotObj &lt;- RCPA::plotMA(rowData(RNASeqDEExperiment), logFCThreshold = 0.5) + 
ggtitle("RNASeq - GSE153873")


</code></pre>

<hr>
<h2 id='plotPathwayHeatmap'>Plot pathways heatmap plot from pathway/geneset/meta analysis results</h2><span id='topic+plotPathwayHeatmap'></span>

<h3>Description</h3>

<p>pathways heatmap plot from pathway/geneset/meta analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPathwayHeatmap(
  resultsList,
  yAxis = c("ID", "name"),
  negLog10pValueLims = c(0, 5),
  useFDR = TRUE,
  selectedPathways = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPathwayHeatmap_+3A_resultslist">resultsList</code></td>
<td>
<p>A named list of dataframes from pathway analysis, geneset analysis, and/or meta analysis results.
The columns are ID, name, description, p.value, pFDR, size, nDE, score and normalizedScore.</p>
</td></tr>
<tr><td><code id="plotPathwayHeatmap_+3A_yaxis">yAxis</code></td>
<td>
<p>The column to use for the y-axis.</p>
</td></tr>
<tr><td><code id="plotPathwayHeatmap_+3A_neglog10pvaluelims">negLog10pValueLims</code></td>
<td>
<p>A vector of length 2 specifying the minimum and maximum -log10(p-value) to plot.</p>
</td></tr>
<tr><td><code id="plotPathwayHeatmap_+3A_usefdr">useFDR</code></td>
<td>
<p>Logical to indicate whether to use FDR or p-value.</p>
</td></tr>
<tr><td><code id="plotPathwayHeatmap_+3A_selectedpathways">selectedPathways</code></td>
<td>
<p>A vector of pathways ID, which is in the same format as ID column in the pathway analysis result, to be included in the plot.
If it is NULL, all pathways will be included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object for presenting the heatmap of the pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")
metaPAResult &lt;- loadData("metaPAResult")

PAResults &lt;- list(
    "Affymetrix - GSE5281" = affyFgseaResult,
    "Agilent - GSE61196" = agilFgseaResult,
    "RNASeq - GSE153873" = RNASeqFgseaResult,
    "Meta-analysis" = metaPAResult
)

selectedPathways &lt;- c("path:hsa05010", "path:hsa05012", "path:hsa05014", 
                      "path:hsa05016", "path:hsa05017", "path:hsa05020", 
                      "path:hsa05022", "path:hsa04724", "path:hsa04727", 
                      "path:hsa04725", "path:hsa04728", "path:hsa04726", 
                      "path:hsa04720", "path:hsa04730", "path:hsa04723", 
                      "path:hsa04721", "path:hsa04722")
                      
resultsToPlot &lt;- lapply(PAResults, function(df) df[df$ID %in% selectedPathways,])

plotObj &lt;- RCPA::plotPathwayHeatmap(resultsToPlot, yAxis = "name")


</code></pre>

<hr>
<h2 id='plotPathwayNetwork'>Plot a pathway network</h2><span id='topic+plotPathwayNetwork'></span>

<h3>Description</h3>

<p>This function plots a pathway network.
This function needs an interactive environment with browser view support to work.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPathwayNetwork(
  PAResults,
  genesets,
  selectedPathways = NULL,
  statistic = "pFDR",
  mode = c("continuous", "discrete"),
  pThreshold = 0.05,
  useFDR = TRUE,
  edgeThreshold = 0.5,
  statLimit = 4,
  discreteColors = NULL,
  continuousScaleFunc = NULL,
  NAColor = "#ffffff",
  borderColor = "#333333",
  nodeSizeFnc = function(id) length(genesets[[id]])^0.75,
  borderWidthFnc = function(id) 1,
  edgeWidthFnc = function(from, to) 1,
  styleFile = system.file(package = "RCPA", "extdata", "pieStyle.js"),
  file = tempfile(fileext = ".html")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPathwayNetwork_+3A_paresults">PAResults</code></td>
<td>
<p>A named list of data frame of Pathway analysis results.
The columns of each data frame should be at least ID, name, p.value and pFDR.
An optional column &quot;color&quot; can be used to specify the color of the nodes.
If the column &quot;color&quot; is not specified, the color of the nodes will be determined by the mode and the statistic.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_genesets">genesets</code></td>
<td>
<p>A genesets object that is obtained from getGeneSets function.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_selectedpathways">selectedPathways</code></td>
<td>
<p>A vector of pathway IDs to be included in the plot.
If it is NULL, all pathways from genesets will be included.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_statistic">statistic</code></td>
<td>
<p>A character value of the statistic to use for the network.
The statistic should be one of the columns of the data frame in the results.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_mode">mode</code></td>
<td>
<p>A character value of the mode to use to color the nodes.
The mode should be one of &quot;discrete&quot;, &quot;continuous&quot;.
If the mode is &quot;discrete&quot;, the color of the nodes are determined by whether the p-value is significant or not.
If the mode is &quot;continuous&quot;, the color of the nodes are proportional to the statistic.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_pthreshold">pThreshold</code></td>
<td>
<p>A numeric value of p-value threshold.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_usefdr">useFDR</code></td>
<td>
<p>A logical value indicating whether to use FDR or not.
This parameter is independent of the pThreshold.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_edgethreshold">edgeThreshold</code></td>
<td>
<p>A numeric from 0 to 1 indicating the threshold to draw edges.
edgeThreshold of 0.1 means that edges are drawn if the number of genes in common is greater than 1% of the smaller gene set.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_statlimit">statLimit</code></td>
<td>
<p>A numeric value of the maximum absolute value of the statistic.
If statistic is p.value or pFDR, this parameter is the limit of -log10(p-value).</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_discretecolors">discreteColors</code></td>
<td>
<p>A character vector of colors to use for the discrete mode.
The length of the vector must be the same as the number of results.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_continuousscalefunc">continuousScaleFunc</code></td>
<td>
<p>A function that takes a numeric value from -1 to 1 and returns a color.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_nacolor">NAColor</code></td>
<td>
<p>A character value of the color to use for NA values.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_bordercolor">borderColor</code></td>
<td>
<p>A character value of the color to use for the border of the nodes.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_nodesizefnc">nodeSizeFnc</code></td>
<td>
<p>A function that takes a character value of the ID of the node and returns a numeric value of the size of the node.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_borderwidthfnc">borderWidthFnc</code></td>
<td>
<p>A function that takes a character value of the ID of the node and returns a numeric value of the width of the border of the node.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_edgewidthfnc">edgeWidthFnc</code></td>
<td>
<p>A function that takes a character value of the ID of the from node and a character value of the ID of the to node and returns a numeric value of the width of the edge.</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_stylefile">styleFile</code></td>
<td>
<p>A character value of the path to the style file.
If NULL, the default style file will be used, which is located at system.file(package=&quot;RCPA&quot;, &quot;extdata&quot;, &quot;pieStyle.js&quot;)</p>
</td></tr>
<tr><td><code id="plotPathwayNetwork_+3A_file">file</code></td>
<td>
<p>A character value of the path to the html file to be created.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will plot a pathway network using the results of pathway analysis.
The nodes of the network are the pathways and the edges are the pathways that have at least a certain number of genes in common defined by the edgeThreshold.
The size of the nodes are proportional to the number of genes in the pathway.
The color of the nodes are proportional to the statistic used if the mode is &quot;continuous&quot;.
If the mode is &quot;discrete&quot;, the color of the nodes are determined by whether the p-value is significant or not.
The width of the edges are proportional to the number of genes in common.
</p>


<h3>Value</h3>

<p>A character value of the html content of the plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)

affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")
metaPAResult &lt;- loadData("metaPAResult")
genesets &lt;- loadData("genesets")

PAResults &lt;- list(
    "Affymetrix - GSE5281" = affyFgseaResult,
    "Agilent - GSE61196" = agilFgseaResult,
    "RNASeq - GSE153873" = RNASeqFgseaResult,
    "Meta-analysis" = metaPAResult
)

genesetsToPlot &lt;- metaPAResult$ID[order(metaPAResult$pFDR)][1:30]

pltHtml &lt;- RCPA::plotPathwayNetwork(
    PAResults,
    genesets = genesets,
    selectedPathways = genesetsToPlot,
    edgeThreshold = 0.75,
    mode = "continuous",
    statistic = "normalizedScore"
)



</code></pre>

<hr>
<h2 id='plotVennDE'>Plot Venn diagram from multiple DE Analysis results</h2><span id='topic+plotVennDE'></span>

<h3>Description</h3>

<p>Plot a Venn diagram from multiple DE Analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVennDE(
  DEResults,
  pThreshold = 0.05,
  useFDR = TRUE,
  stat = "logFC",
  statThreshold = 0,
  topToList = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotVennDE_+3A_deresults">DEResults</code></td>
<td>
<p>A list of data frames with the results of DE analysis.</p>
</td></tr>
<tr><td><code id="plotVennDE_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to determine if a gene is differentially expressed.</p>
</td></tr>
<tr><td><code id="plotVennDE_+3A_usefdr">useFDR</code></td>
<td>
<p>Use the FDR adjusted p-value instead of the raw p-value.</p>
</td></tr>
<tr><td><code id="plotVennDE_+3A_stat">stat</code></td>
<td>
<p>The additional statistics column to use for filtering differentially expressed genes.</p>
</td></tr>
<tr><td><code id="plotVennDE_+3A_statthreshold">statThreshold</code></td>
<td>
<p>The absolute value of the statistic threshold to use for filtering differentially expressed genes.
Default is 0, which means no filtering.</p>
</td></tr>
<tr><td><code id="plotVennDE_+3A_toptolist">topToList</code></td>
<td>
<p>The number of common DE genes that are used to annotate the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)

affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

DEResults &lt;- list(
    "Affymetrix - GSE5281" = rowData(affyDEExperiment),
    "Agilent - GSE61196"   = rowData(agilDEExperiment),
    "RNASeq - GSE153873"   = rowData(RNASeqDEExperiment)
)

DEResultUps &lt;- lapply(DEResults, function(df) df[!is.na(df$logFC) &amp; df$logFC &gt; 0, ])

DEResultDowns &lt;- lapply(DEResults, function(df) df[!is.na(df$logFC) &amp; df$logFC &lt; 0, ])

if (require("ggvenn", quietly = TRUE)){
p1 &lt;- RCPA::plotVennDE(DEResults) + 
        ggplot2::ggtitle("All DE Genes")
p2 &lt;- RCPA::plotVennDE(DEResultUps) +
        ggplot2::ggtitle("Up-regulated DE Genes")
p3 &lt;- RCPA::plotVennDE(DEResultDowns) + 
        ggplot2::ggtitle("Down-regulated DE Genes")
}


</code></pre>

<hr>
<h2 id='plotVennPathway'>Plot Venn diagram from multiple pathway analysis results</h2><span id='topic+plotVennPathway'></span>

<h3>Description</h3>

<p>Plot a Venn diagram from multiple pathway analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVennPathway(PAResults, pThreshold = 0.05, useFDR = TRUE, topToList = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotVennPathway_+3A_paresults">PAResults</code></td>
<td>
<p>A list of data frames with the results of pathway analysis.</p>
</td></tr>
<tr><td><code id="plotVennPathway_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to determine if a pathway is enriched.</p>
</td></tr>
<tr><td><code id="plotVennPathway_+3A_usefdr">useFDR</code></td>
<td>
<p>Use the FDR adjusted p-value instead of the raw p-value.</p>
</td></tr>
<tr><td><code id="plotVennPathway_+3A_toptolist">topToList</code></td>
<td>
<p>The number of common signifcant pathways that are used to annotate the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)


affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")
metaPAResult &lt;- loadData("metaPAResult")

PAResults &lt;- list(
    "Affymetrix - GSE5281" = affyFgseaResult,
    "Agilent - GSE61196" = agilFgseaResult,
    "RNASeq - GSE153873" = RNASeqFgseaResult,
    "Meta-analysis" = metaPAResult
)

PAREsultUps &lt;- lapply(PAResults, function(df) df[df$normalizedScore &gt; 0,])

PAREsultDowns &lt;- lapply(PAResults, function(df) df[df$normalizedScore &lt; 0,])

if (require("ggvenn", quietly = TRUE)){
p1 &lt;- RCPA::plotVennPathway(PAResults, pThreshold = 0.05) +
    ggplot2::ggtitle("All Significant Pathways")
p2 &lt;- RCPA::plotVennPathway(PAREsultUps, pThreshold = 0.05) +
    ggplot2::ggtitle("Significantly Up-regulated Pathways")
p3 &lt;- RCPA::plotVennPathway(PAREsultDowns, pThreshold = 0.05) +
    ggplot2::ggtitle("Significantly Down-regulated Pathways")
}

</code></pre>

<hr>
<h2 id='plotVolcanoDE'>Plot volcano plot from Pathway analysis results</h2><span id='topic+plotVolcanoDE'></span>

<h3>Description</h3>

<p>Plot volcano plot from Pathway analysis results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVolcanoDE(DEResult, pThreshold = 0.05, useFDR = TRUE, logFCThreshold = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotVolcanoDE_+3A_deresult">DEResult</code></td>
<td>
<p>A data frame with Pathway analysis results.
The columns are ID, name, description, p.value, pFDR, size, nDE, score and normalizedScore.</p>
</td></tr>
<tr><td><code id="plotVolcanoDE_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to use for the horizontal line.</p>
</td></tr>
<tr><td><code id="plotVolcanoDE_+3A_usefdr">useFDR</code></td>
<td>
<p>Whether to use the pFDR column instead of the p.value column.</p>
</td></tr>
<tr><td><code id="plotVolcanoDE_+3A_logfcthreshold">logFCThreshold</code></td>
<td>
<p>The logFC threshold to use for the vertical line.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)
library(SummarizedExperiment)

affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

p1 &lt;- RCPA::plotVolcanoDE(rowData(affyDEExperiment), logFCThreshold = 0.5) +
    ggplot2::ggtitle("Affymetrix - GSE5281")
p2 &lt;- RCPA::plotVolcanoDE(rowData(agilDEExperiment), logFCThreshold = 0.5) +
    ggplot2::ggtitle("Agilent - GSE61196")
p3 &lt;- RCPA::plotVolcanoDE(rowData(RNASeqDEExperiment), logFCThreshold = 0.5) +
    ggplot2::ggtitle("RNASeq - GSE153873")


</code></pre>

<hr>
<h2 id='plotVolcanoPathway'>Plot volcano plot from Pathway analysis results</h2><span id='topic+plotVolcanoPathway'></span>

<h3>Description</h3>

<p>Plot volcano plot from Pathway analysis results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVolcanoPathway(
  PAResult,
  xAxis = c("normalizedScore", "score"),
  yAxis = c("-log10(pFDR)", "-log10(p.value)"),
  pThreshold = 0.05,
  label = "name",
  IDsToLabel = NULL,
  topToLabel = 10,
  sideToLabel = c("both", "left", "right")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotVolcanoPathway_+3A_paresult">PAResult</code></td>
<td>
<p>A data frame with Pathway analysis results.
The columns are ID, name, description, p.value, pFDR, pathwaySize, nDE, score and normalizedScore.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_xaxis">xAxis</code></td>
<td>
<p>The column to use for the x-axis.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_yaxis">yAxis</code></td>
<td>
<p>The column to use for the y-axis.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_pthreshold">pThreshold</code></td>
<td>
<p>The p-value threshold to use for the horizontal line.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_label">label</code></td>
<td>
<p>The column to use for the labels. Default is &quot;name&quot;.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_idstolabel">IDsToLabel</code></td>
<td>
<p>A vector of IDs to label.
When NULL, the top pathways are labeled. Default is NULL.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_toptolabel">topToLabel</code></td>
<td>
<p>The number of top pathways to label when IDsToLabels is NULL.</p>
</td></tr>
<tr><td><code id="plotVolcanoPathway_+3A_sidetolabel">sideToLabel</code></td>
<td>
<p>The side of the plot to label.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")
metaPAResult &lt;- loadData("metaPAResult")

p1 &lt;- RCPA::plotVolcanoPathway(affyFgseaResult, sideToLabel = "left")
p2 &lt;- RCPA::plotVolcanoPathway(agilFgseaResult, sideToLabel = "left")
p3 &lt;- RCPA::plotVolcanoPathway(RNASeqFgseaResult, sideToLabel = "left")
p4 &lt;- RCPA::plotVolcanoPathway(metaPAResult, sideToLabel = "left")



</code></pre>

<hr>
<h2 id='processAffymetrix'>Process and normalize affymetrix-based dataset</h2><span id='topic+processAffymetrix'></span>

<h3>Description</h3>

<p>This function process CEL files and normalize expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processAffymetrix(dir, samples = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processAffymetrix_+3A_dir">dir</code></td>
<td>
<p>The path to the directory containing CEL files.</p>
</td></tr>
<tr><td><code id="processAffymetrix_+3A_samples">samples</code></td>
<td>
<p>A vector of samples IDs. If NULL, the function will automatically detect the samples in the directory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Read and normalize expression data for affymetrix using RMA method
</p>


<h3>Value</h3>

<p>A matrix of normalized expression data. Rows are probes and columns are samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
geoId &lt;- "GSE59761"
downloadPath &lt;- file.path(tempdir(), geoId)
fileList &lt;- RCPA::downloadGEO(GEOID = geoId, protocol = "affymetrix",
                               platform ="GPL16311", destDir = downloadPath)
# process only 3 samples
expression &lt;- RCPA::processAffymetrix(downloadPath, 
                   samples = c("GSM1446171", "GSM1446172", "GSM1446173"))

</code></pre>

<hr>
<h2 id='processAgilent'>Process and normalize agilent-based dataset</h2><span id='topic+processAgilent'></span>

<h3>Description</h3>

<p>This function process TXT files and normalize expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processAgilent(dir, samples = NULL, greenOnly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processAgilent_+3A_dir">dir</code></td>
<td>
<p>The path to the directory containing TXT files.</p>
</td></tr>
<tr><td><code id="processAgilent_+3A_samples">samples</code></td>
<td>
<p>A vector of samples IDs. If NULL, the function will automatically detect the samples in the directory.</p>
</td></tr>
<tr><td><code id="processAgilent_+3A_greenonly">greenOnly</code></td>
<td>
<p>Logical, for use with source, should the green (Cy3) channel only be read, or are both red and green required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Read and normalize expression data for agilent using limma normexp, loess, and quantile methods
</p>


<h3>Value</h3>

<p>A matrix of normalized expression data. Rows are probes and columns are samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
geoId &lt;- "GSE28522"

downloadPath &lt;- file.path(tempdir(), geoId)
fileList &lt;- RCPA::downloadGEO(GEOID = geoId, protocol = "agilent",
                              platform ="GPL4133", destDir = downloadPath)

expression &lt;- RCPA::processAgilent(downloadPath, greenOnly = FALSE)


</code></pre>

<hr>
<h2 id='runConsensusAnalysis'>Perform Consensus Analysis</h2><span id='topic+runConsensusAnalysis'></span>

<h3>Description</h3>

<p>This function performs consensus analysis using two methods.
These methods are weighted.mean and RRA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runConsensusAnalysis(
  PAResults,
  method = c("weightedZMean", "RRA"),
  weightsList = NULL,
  useFDR = TRUE,
  rank.by = c("normalizedScore", "pFDR", "both"),
  backgroundSpace = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runConsensusAnalysis_+3A_paresults">PAResults</code></td>
<td>
<p>A list of at least length two from enrichment analysis results.</p>
</td></tr>
<tr><td><code id="runConsensusAnalysis_+3A_method">method</code></td>
<td>
<p>The consensus analsyis method. This can be either weighted.mean or RRA.</p>
</td></tr>
<tr><td><code id="runConsensusAnalysis_+3A_weightslist">weightsList</code></td>
<td>
<p>A vector of integer values.
Each element shows the corresponding input result weight.
When selected method is weighted.mean this parameter needs to be specified.
If it is null all the weights are considered as equal.</p>
</td></tr>
<tr><td><code id="runConsensusAnalysis_+3A_usefdr">useFDR</code></td>
<td>
<p>A logical parameter, indicating if adjusted p-values should be used.</p>
</td></tr>
<tr><td><code id="runConsensusAnalysis_+3A_rank.by">rank.by</code></td>
<td>
<p>An string parameter which specifies how the input results should be ranked.
This parameter is used when the selected method is RRA.</p>
</td></tr>
<tr><td><code id="runConsensusAnalysis_+3A_backgroundspace">backgroundSpace</code></td>
<td>
<p>A list of lists with the same length as PAResults.
Each list contains underlying space (set of pathways) for each input dataframe in PAResults.
This parameter is optional.
NULL means all input dataframes share a common space.
So the union pathways of all input dataframes is taken into account.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of consensus analysis result, which contains the following columns:
</p>

<ul>
<li><p>ID: The ID of pathway
</p>
</li>
<li><p>p.value: The p-value of pathway
</p>
</li>
<li><p>pFDR: The adjusted p-value using Benjamini-Hochberg method
</p>
</li>
<li><p>name: The name of pathway
</p>
</li>
<li><p>pathwaySize: The size of pathway
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)

affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")

consensusPAResult &lt;- RCPA::runConsensusAnalysis(
    list(affyFgseaResult, agilFgseaResult, RNASeqFgseaResult),
    method = "weightedZMean"
)

print(head(consensusPAResult))


</code></pre>

<hr>
<h2 id='runDEAnalysis'>Differential expression analysis</h2><span id='topic+runDEAnalysis'></span>

<h3>Description</h3>

<p>This function performs differential expression analysis using either limma, DESeq2 or edgeR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runDEAnalysis(
  summarizedExperiment,
  method = c("limma", "DESeq2", "edgeR"),
  design,
  contrast,
  annotation = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runDEAnalysis_+3A_summarizedexperiment">summarizedExperiment</code></td>
<td>
<p>SummarizedExperiment object</p>
</td></tr>
<tr><td><code id="runDEAnalysis_+3A_method">method</code></td>
<td>
<p>Method to use for differential expression analysis. Can be &quot;limma&quot;, &quot;DESeq2&quot; or &quot;edgeR&quot;.</p>
</td></tr>
<tr><td><code id="runDEAnalysis_+3A_design">design</code></td>
<td>
<p>A design model output by model.matrix.</p>
</td></tr>
<tr><td><code id="runDEAnalysis_+3A_contrast">contrast</code></td>
<td>
<p>A contrast matrix. See limma::makeContrasts.</p>
</td></tr>
<tr><td><code id="runDEAnalysis_+3A_annotation">annotation</code></td>
<td>
<p>A data frame mapping between probe IDs and entrez gene IDs.
If not provided, the function will try to get the mapping from the platform annotation in the SummarizedExperiment object.
If the annotation is not available, the function will return the probe IDs.
Regardless of the type of annotation, it must contains two columns: FROM and TO,
where FROM is the probe ID and TO is the entrez gene ID.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A SummarizedExperiment object with DE analysis results appended to the rowData slot with the following columns:
</p>

<ul>
<li><p>ID: gene ID. If annotation is provided, this will be the entrez gene ID. Otherwise, it will be the probe ID.
</p>
</li>
<li><p>logFC: log2 fold change
</p>
</li>
<li><p>p.value: p-value from the DE analysis using the specified method
</p>
</li>
<li><p>pFDR: p-value adjusted for multiple testing using Benjamini-Hochberg method
</p>
</li>
<li><p>statistic: statistic from the DE analysis using the specified method.
For limma, this is the t-statistic.
For DESeq2, this is the Wald statistic.
For edgeR, this is the log fold change.
</p>
</li>
<li><p>avgExpr: 
For limma, it is the average expression.
For DESeq2, it is the log base mean.
For edgeR, it is the log CPM.

</p>
</li>
<li><p>logFCSE: standard error of the log fold change.
</p>
</li>
<li><p>sampleSize: sample size used for DE analysis.
</p>
</li></ul>

<p>The assay slot will contain the input expression/count matrix,
and the rownames will be mapped to the gene IDs if annotation is found in the input SummarizedExperiment object
or in the annotation parameter.
Other slots will be the same as in the input SummarizedExperiment object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)

# GSE5281
affyDataset &lt;- loadData("affyDataset")
affyDesign &lt;- model.matrix(~0 + condition + region + condition:region, 
                             data = colData(affyDataset))
colnames(affyDesign) &lt;- make.names(colnames(affyDesign))
affyContrast &lt;- limma::makeContrasts(conditionalzheimer-conditionnormal, 
                                     levels=affyDesign)

if (require("hgu133plus2.db", quietly = TRUE)){
affyDEExperiment &lt;- RCPA::runDEAnalysis(affyDataset, method = "limma", 
                                        design = affyDesign,
                                        contrast = affyContrast, 
                                        annotation = "GPL570")
                                        
# check the DE analysis results

print(head(rowData(affyDEExperiment)))
}



# GSE61196
agilDataset &lt;- loadData("agilDataset")
agilDesign &lt;- model.matrix(~0 + condition, 
                            data = colData(agilDataset))
agilContrast &lt;- limma::makeContrasts(conditionalzheimer-conditionnormal, 
                                     levels=agilDesign)

# Create Probe mapping
GPL4133Anno &lt;- GEOquery::dataTable(GEOquery::getGEO("GPL4133"))@table
GPL4133GeneMapping &lt;- data.frame(FROM = GPL4133Anno$SPOT_ID, 
                                 TO = as.character(GPL4133Anno$GENE),
                                 stringsAsFactors = FALSE)
GPL4133GeneMapping &lt;- GPL4133GeneMapping[!is.na(GPL4133GeneMapping$TO), ]

agilDEExperiment &lt;- RCPA::runDEAnalysis(agilDataset, method = "limma", 
                                        design = agilDesign,
                                        contrast = agilContrast, 
                                        annotation = GPL4133GeneMapping)
print(head(rowData(agilDEExperiment)))

# GSE153873
RNASeqDataset &lt;- loadData("RNASeqDataset")
RNASeqDesign &lt;- model.matrix(~0 + condition, data = colData(RNASeqDataset))
RNASeqContrast &lt;- limma::makeContrasts(conditionalzheimer-conditionnormal, 
                                      levels=RNASeqDesign)


if (require("org.Hs.eg.db", quietly = TRUE)){
    GeneSymbolMapping &lt;- AnnotationDbi::select(org.Hs.eg.db, 
                                            keys = rownames(RNASeqDataset),
                                            columns = c("SYMBOL", "ENTREZID"), 
                                            keytype = "SYMBOL")
    colnames(GeneSymbolMapping) &lt;- c("FROM", "TO")

    RNASeqDEExperiment &lt;- RCPA::runDEAnalysis(RNASeqDataset,
                           method = "DESeq2",
                           design = RNASeqDesign,
                           contrast = RNASeqContrast,
                           annotation = GeneSymbolMapping)
    print(head(rowData(RNASeqDEExperiment)))
}

</code></pre>

<hr>
<h2 id='runDEMetaAnalysis'>Combine DE analysis results</h2><span id='topic+runDEMetaAnalysis'></span>

<h3>Description</h3>

<p>This function performs mata analysis on multiple DE analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runDEMetaAnalysis(
  DEResults,
  method = c("stouffer", "fisher", "addCLT", "geoMean", "minP", "REML")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runDEMetaAnalysis_+3A_deresults">DEResults</code></td>
<td>
<p>A list of dataframes containing DE analysis results.
Each dataframe must have ID, p.value, logFC and logFCSE columns.</p>
</td></tr>
<tr><td><code id="runDEMetaAnalysis_+3A_method">method</code></td>
<td>
<p>The method to combine p-values. It can be one of &quot;fisher&quot;, &quot;stouffer&quot;, &quot;geoMean&quot;, &quot;addCLT&quot;, &quot;minP&quot;, or &quot;REML&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing combined DE analysis results.
The dataframe has ID, p.value, pDFR, logFC, and logFCSE columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
library(SummarizedExperiment)
affyDEExperiment &lt;- loadData("affyDEExperiment")
agilDEExperiment &lt;- loadData("agilDEExperiment")
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")

metaDEResult &lt;- RCPA::runDEMetaAnalysis(list(
    rowData(affyDEExperiment)[1:1000,],
    rowData(agilDEExperiment)[1:1000,],
    rowData(RNASeqDEExperiment)[1:1000,]
), method = "stouffer")


</code></pre>

<hr>
<h2 id='runGeneSetAnalysis'>Gene Set Enrichment Analysis</h2><span id='topic+runGeneSetAnalysis'></span>

<h3>Description</h3>

<p>This function performs gene set enrichment analysis using either ORA, fgsea, GSA, ks, or wilcox approaches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runGeneSetAnalysis(
  summarizedExperiment,
  genesets,
  method = c("ora", "fgsea", "gsa", "ks", "wilcox"),
  ORAArgs = list(pThreshold = 0.05),
  FgseaArgs = list(sampleSize = 101, minSize = 1, maxSize = Inf, eps = 1e-50, scoreType =
    "std", nproc = 0, gseaParam = 1, BPPARAM = NULL, nPermSimple = 1000, absEps = NULL),
  GSAArgs = list(method = "maxmean", random.seed = NULL, knn.neighbors = 10, s0 = NULL,
    s0.perc = NULL, minsize = 15, maxsize = 500, restand = TRUE, restand.basis =
    "catalog", nperms = 200, xl.mode = "regular", xl.time = NULL, xl.prevfit = NULL)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runGeneSetAnalysis_+3A_summarizedexperiment">summarizedExperiment</code></td>
<td>
<p>The generated SummarizedExpriment object from DE analysis result.</p>
</td></tr>
<tr><td><code id="runGeneSetAnalysis_+3A_genesets">genesets</code></td>
<td>
<p>The gene sets definition, ex. KEGG genesets from getGeneSets function.</p>
</td></tr>
<tr><td><code id="runGeneSetAnalysis_+3A_method">method</code></td>
<td>
<p>The gene set enrichment analsyis method, including ORA, fgsea, GSA, ks, and wilcox.</p>
</td></tr>
<tr><td><code id="runGeneSetAnalysis_+3A_oraargs">ORAArgs</code></td>
<td>
<p>A list of other passed arguments to ORA. pThreshold is used as p.value cutoff to pick DE genes.</p>
</td></tr>
<tr><td><code id="runGeneSetAnalysis_+3A_fgseaargs">FgseaArgs</code></td>
<td>
<p>A list of other passed arguments to fgsea. See fgsea function.</p>
</td></tr>
<tr><td><code id="runGeneSetAnalysis_+3A_gsaargs">GSAArgs</code></td>
<td>
<p>A list of other passed arguments to GSA. See GSA function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of gene set enrichment analysis result, which contains the following columns:
</p>

<ul>
<li><p>ID: The ID of the gene set
</p>
</li>
<li><p>p.value: The p-value of the gene set
</p>
</li>
<li><p>pFDR: The adjusted p-value of the gene set using the Benjamini-Hochberg method
</p>
</li>
<li><p>score: The enrichment score of the gene set
</p>
</li>
<li><p>normalizedScore: The normalized enrichment score of the gene set
</p>
</li>
<li><p>sampleSize: The total number of samples in the study
</p>
</li>
<li><p>name: The name of the gene set
</p>
</li>
<li><p>pathwaySize: The size of the gene set
</p>
</li></ul>

<p>The returned data frame is sorted based on the pathways' nominal p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)

RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")
genesets &lt;- loadData("genesets")


oraResult &lt;- runGeneSetAnalysis(RNASeqDEExperiment, genesets,
                               method = "ora", 
                               ORAArgs = list(pThreshold = 0.05))
print(head(oraResult))
                                           
fgseaResult &lt;- runGeneSetAnalysis(RNASeqDEExperiment, genesets, 
                               method = "fgsea",
                               FgseaArgs = list(minSize = 10, maxSize = Inf))
print(head(fgseaResult))


</code></pre>

<hr>
<h2 id='runPathwayAnalysis'>Topology-based Pathway Analysis</h2><span id='topic+runPathwayAnalysis'></span>

<h3>Description</h3>

<p>This function performs patwhay analysis using SPIA, CePaORA, and CePaGSA methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runPathwayAnalysis(
  summarizedExperiment,
  network,
  method = c("spia", "cepaORA", "cepaGSA"),
  SPIAArgs = list(all = NULL, nB = 2000, verbose = TRUE, beta = NULL, combine = "fisher",
    pThreshold = 0.05),
  CePaORAArgs = list(bk = NULL, cen = c("equal.weight", "in.degree", "out.degree",
    "betweenness", "in.reach", "out.reach"), cen.name = c("equal.weight", "in.degree",
    "out.degree", "betweenness", "in.reach", "out.reach"), iter = 1000, pThreshold =
    0.05),
  CePaGSAArgs = list(cen = c("equal.weight", "in.degree", "out.degree", "betweenness",
    "in.reach", "out.reach"), cen.name = c("equal.weight", "in.degree", "out.degree",
    "betweenness", "in.reach", "out.reach"), nlevel = "tvalue_abs", plevel = "mean", iter
    = 1000)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runPathwayAnalysis_+3A_summarizedexperiment">summarizedExperiment</code></td>
<td>
<p>The generated SummarizedExpriment object from DE analysis result.</p>
</td></tr>
<tr><td><code id="runPathwayAnalysis_+3A_network">network</code></td>
<td>
<p>The pathways network object.</p>
</td></tr>
<tr><td><code id="runPathwayAnalysis_+3A_method">method</code></td>
<td>
<p>The pathway analsyis method, including SPIA, cepaORA, and cepaGSA.</p>
</td></tr>
<tr><td><code id="runPathwayAnalysis_+3A_spiaargs">SPIAArgs</code></td>
<td>
<p>A list of other passed arguments to spia. See spia function.</p>
</td></tr>
<tr><td><code id="runPathwayAnalysis_+3A_cepaoraargs">CePaORAArgs</code></td>
<td>
<p>A list of other passed arguments to CePaORA. See CePa function.</p>
</td></tr>
<tr><td><code id="runPathwayAnalysis_+3A_cepagsaargs">CePaGSAArgs</code></td>
<td>
<p>A list of other passed arguments to CePaGSA. See CePa function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of pathway analysis result, which contains the following columns
</p>

<ul>
<li><p>ID: The ID of the gene set
</p>
</li>
<li><p>p.value: The p-value of the gene set
</p>
</li>
<li><p>pFDR: The adjusted p-value of the gene set using the Benjamini-Hochberg method
</p>
</li>
<li><p>score: The enrichment score of the gene set
</p>
</li>
<li><p>normalizedScore: The normalized enrichment score of the gene set
</p>
</li>
<li><p>sampleSize: The total number of samples in the study
</p>
</li>
<li><p>name: The name of the gene set
</p>
</li>
<li><p>pathwaySize: The size of the gene set
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(RCPA)
RNASeqDEExperiment &lt;- loadData("RNASeqDEExperiment")
spiaNetwork &lt;- loadData("spiaNetwork")
cepaNetwork &lt;- loadData("cepaNetwork")

spiaResult &lt;- runPathwayAnalysis(RNASeqDEExperiment, spiaNetwork, method = "spia")
cepaORAResult &lt;- runPathwayAnalysis(RNASeqDEExperiment, cepaNetwork, method = "cepaORA")

</code></pre>

<hr>
<h2 id='runPathwayMetaAnalysis'>Perform Meta Analysis</h2><span id='topic+runPathwayMetaAnalysis'></span>

<h3>Description</h3>

<p>This function performs meta analysis on multiple pathway analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runPathwayMetaAnalysis(
  PAResults,
  method = c("stouffer", "fisher", "addCLT", "geoMean", "minP", "REML")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runPathwayMetaAnalysis_+3A_paresults">PAResults</code></td>
<td>
<p>A list of at least size two of data frames obtained from pathway analysis</p>
</td></tr>
<tr><td><code id="runPathwayMetaAnalysis_+3A_method">method</code></td>
<td>
<p>A method used to combine pathway analysis results, which can be &quot;stouffer&quot;, &quot;fisher&quot;, &quot;addCLT&quot;, &quot;geoMean&quot;, &quot;minP&quot;, or &quot;REML&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs meta-analysis on multiple pathway analysis results.
</p>


<h3>Value</h3>

<p>A dataframe of meta analysis results including the following columns:
</p>

<ul>
<li><p>ID: The ID of pathway
</p>
</li>
<li><p>name: The name of pathway
</p>
</li>
<li><p>p.value: The meta p-value of pathway
</p>
</li>
<li><p>pFDR: The adjusted meta p-value of pathway using Benjamini-Hochberg method
</p>
</li>
<li><p>score: The combined score of pathway
</p>
</li>
<li><p>normalizedScore: The combined normalized score of pathway
</p>
</li>
<li><p>pathwaySize: The size of pathway
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>

library(RCPA)
affyFgseaResult &lt;- loadData("affyFgseaResult")
agilFgseaResult &lt;- loadData("agilFgseaResult")
RNASeqFgseaResult &lt;- loadData("RNASeqFgseaResult")

metaPAResult &lt;- RCPA::runPathwayMetaAnalysis(
    list(affyFgseaResult, agilFgseaResult, RNASeqFgseaResult),
    method = "stouffer"
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
