<!DOCTYPE html><html><head><title>Help for package RCytoGPS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RCytoGPS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Chromosome-class'><p>The <code>Chromosome</code> Class</p></a></li>
<li><a href='#cytoband-data'><p>Example Cytoband Data</p></a></li>
<li><a href='#CytobandData-class'><p>The <code>CytobandData</code> Class</p></a></li>
<li><a href='#CytobandData-methods'><p>Idiogram Graphs</p></a></li>
<li><a href='#cytobandLocations'><p>Cytoband Locations</p></a></li>
<li><a href='#preclean'><p>Pre-clean Karyotypes</p></a></li>
<li><a href='#readLGF'><p>Extracting LGF karyotype data from JSON files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.2.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-27</td>
</tr>
<tr>
<td>Title:</td>
<td>Using Cytogenetics Data in R</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin R. Coombes, Dwayne Tally</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines classes and methods to process text-based
  cytogenetics using the CytoGPS web site, then import the results
  into R for further analysis and graphing.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, Mercator</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, graphics, grDevices, rjson</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-27 16:05:10 UTC; KRC</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-27 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Chromosome-class'>The <code>Chromosome</code> Class</h2><span id='topic+Chromosome-class'></span><span id='topic+Chromosome'></span><span id='topic+image+2CChromosome-method'></span>

<h3>Description</h3>

<p>A class to represent a single chromsome in order to plot an image of
the Giemsa-stained cytobands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Chromosome(I, res = 2500, maxbase = 250000000)
## S4 method for signature 'Chromosome'
image(x, horiz = FALSE, mai = NULL, showBandNames = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Chromosome-class_+3A_i">I</code></td>
<td>
<p>A human chromosome identifier; one of the values in <code>c(1:22,"X", "Y")</code>.</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_res">res</code></td>
<td>
<p>An integer resolution; the number of pixels used to
represent an entire chromosome.</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_maxbase">maxbase</code></td>
<td>
<p>An integer representing an upper bound on the length of
the longest chromosome, mneasured in base-pairs.</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_x">x</code></td>
<td>
<p>An object of the <code>Chromosome</code> class.</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_horiz">horiz</code></td>
<td>
<p>A logical value: should the image of the chromsome be
oriented horizontally.</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_mai">mai</code></td>
<td>
<p>margin inches, as in the usual graphical argument to <code>par</code>.</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_showbandnames">showBandNames</code></td>
<td>
<p>logical; shold the names of the bands be written on the plot?</p>
</td></tr>
<tr><td><code id="Chromosome-class_+3A_...">...</code></td>
<td>
<p>Additional arguments to the image method; ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Conventional karyotyping describes chromosomal abnormalities (in a
standardized text-based nomenclature) that are visible through a
microsope. Karyotyping relies on a technique known as Giemsa staining,
which creates a banding pattern along the chromosome of different
shades of gray. This class is used to plot images of individual
chromosomes, corectly reflecting the size and color of the bands..
</p>


<h3>Value</h3>

<p>The <code>Chromosome</code> constructor returns an object of the
<code>Chromosome</code> class. The <code>image</code> method invisibly returns its
first argument.
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code>:</dt><dd><p>A character value, typically of the form <code>"chr1"</code>.</p>
</dd>
<dt><code>label</code>:</dt><dd><p>A chartacter value, typiclly of the form
<code>"Chr 1"</code>.</p>
</dd>
<dt><code>grid</code>:</dt><dd><p>An integer vector (of length <code>res</code>)
repesenting the base position along the chromosome.</p>
</dd>
<dt><code>range</code>:</dt><dd><p>An integer vector of length 2 marking teh
starting and ending position of the banded part of the chromosome,
in bases.</p>
</dd>
<dt><code>stain</code>:</dt><dd><p>An integer vector (of length <code>res</code>), where
the values indicate the color of the Giemsa stain for that part of
the chromosome.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>image:</dt><dd><p><code>signature(object = "RCytoGPS")</code>
Creates an image of the chromosme, with bands colored according
to Giemsa staining.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Tally DG, Coombes KR. RCytoGPS: An R Package for Analyzing
and Visualizing Cytogenetic Data. In preparation.
</p>
<p>Abrams ZB, Tally DG,  Zhang L, Coombes CE, Payne PRO, Abruzzo LV,
Coombes KR. Pattern recognition in lymphoid malignancies using CytoGPS
and Mercator. Under review.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- Chromosome(6)
image(x)
image(x, showBandNames = TRUE)
image(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='cytoband-data'>Example Cytoband Data</h2><span id='topic+cytoband-data'></span><span id='topic+cytoData'></span>

<h3>Description</h3>

<p>This data set contains the genomic locations of cytobands along with
loss-gain-fusion percentages for three group of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cytoData)
</code></pre>


<h3>Format</h3>

<p>A data matrix (<code>cytobandLocations</code>) containing 868 rows and
14 columns. Each row contains one of the cytobands defined in the 2016
update to ISCN nomenclature. The rownames are the standard cytoband
names; for example, <code>1p36.33</code>. The first five columns are the
same as in the <code><a href="#topic+cytobandLocations">cytobandLocations</a></code> data set. The remaining
nine columns form three sets of three, recording the percentage of Loss,
Gain, and Fusion events in three sets of samples, labeled &quot;A&quot;, &quot;B&quot;,
and &quot;C&quot;.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>Source</h3>

<p>The cytoband locations were downloaded from the UCSC Genome Browser
and synchronized with the list of cytobands in ISCN 2016. The
percentages were computed using tools in this package from subsets of
the Mitelman Database of Chromosome Aberrations and Gene Fusions in
Cancer.
</p>


<h3>References</h3>

<p>J. McGowan-Jordan, A. Simons, M. Schmid (editors).
ISCN 2016: An International System for Human Cytogenomic Nomenclature.
Karger Publishing, Basel, 2016.
</p>
<p>Mitelman, F., B. Johansson, and F. Mertens, Catalog of chromosome
aberrations in cancer. Vol. 1. 1991: Wiley-Liss New York.
</p>

<hr>
<h2 id='CytobandData-class'>The <code>CytobandData</code> Class</h2><span id='topic+CytobandData-class'></span><span id='topic+CytobandData'></span><span id='topic+summary+2CCytobandData-method'></span>

<h3>Description</h3>

<p>A class to represent genome-wide data measured or summarized at
cytoband-level resoluton.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CytobandData(data, info, genome = NULL)
## S4 method for signature 'CytobandData'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CytobandData-class_+3A_data">data</code></td>
<td>
<p> A data frame that must contain at least one
numeric column and may contain the five cytoband location
columns. If the latter are missing, they must be supplied by the
<code>genome</code> argument.</p>
</td></tr>
<tr><td><code id="CytobandData-class_+3A_info">info</code></td>
<td>
<p> A data frame with <code>Label</code> and
<code>Description</code> columns. If missing, it will be created from the
column names in the <code>data</code> argument.</p>
</td></tr>
<tr><td><code id="CytobandData-class_+3A_genome">genome</code></td>
<td>
<p> A data frame containing cytoband
locations. Often, simply uses the <code><a href="#topic+cytobandLocations">cytobandLocations</a></code>
object describing build 38 of the human genome.</p>
</td></tr>
<tr><td><code id="CytobandData-class_+3A_object">object</code></td>
<td>
<p>An object of the <code>CytobandData</code> class.</p>
</td></tr>
<tr><td><code id="CytobandData-class_+3A_...">...</code></td>
<td>
<p>Additional arguments to the summary method; ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Conventional karyotyping describes chromosomal abnormalities (in a
standardized text-based nomenclature) that are visible through a
microsope. Karyotyping relies on a technique known as Giemsa staining,
which creates a banding pattern along the chromosome of different
shades of gray. The Loss-Gain-Fusion (LGF) model implemented at the
<a href="http://cytogps.org">http://cytogps.org</a> web site converts text-based karyotypes into
binary vectors, stored in JSON files. Our <code><a href="#topic+readLGF">readLGF</a></code>
function parses the JSON files to create R data structures, including
cytoband-resolution summaries of the frequencies of
abnormalities. These summaries can be used to create
<code>CytobandData</code> objects, which can be visualized using the
<code><a href="graphics.html#topic+barplot">barplot</a></code> and <code><a href="graphics.html#topic+image">image</a></code> methods.
</p>


<h3>Value</h3>

<p>The <code>CytobandData</code> conxtructor returns an object of the
<code>CytobandData</code> class. The <code>summary</code> method returns a table,
which is the usual return value when the summary method is applied to a
data frame (in this case, the <code>DATA</code> slot).
</p>


<h3>Slots</h3>


<dl>
<dt><code>DATA</code>:</dt><dd><p>A data frame containg five columns  (Chromosome,
loc.start, loc.end, Band, and Stain) defining chromosomal
locations of cytobands, along with one or more columns of
numerical data at cytoband resolution.</p>
</dd>
<dt><code>INFO</code>:</dt><dd><p>A data frame with two columns (Label and
Description) that describes the columns of the <code>DATA</code> slot.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>summary:</dt><dd><p><code>signature(object = "RCytoGPS")</code>
Returns a summary of the <code>DATA</code> slot.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Tally DG, Coombes KR. RCytoGPS: An R Package for Analyzing
and Visualizing Cytogenetic Data. In preparation.
</p>
<p>Abrams ZB, Tally DG,  Zhang L, Coombes CE, Payne PRO, Abruzzo LV,
Coombes KR. Pattern recognition in lymphoid malignancies using CytoGPS
and Mercator. Under review.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jsonDir &lt;- system.file("Examples/JSONfiles", package = "RCytoGPS")
temp &lt;- readLGF(folder = jsonDir)
cytoData &lt;- data.frame(temp[["CL"]],temp[["frequency"]])
bandData &lt;- CytobandData(cytoData)
class(bandData)
summary(bandData)
</code></pre>

<hr>
<h2 id='CytobandData-methods'>Idiogram Graphs</h2><span id='topic+image+2CCytobandData-method'></span><span id='topic+barplot+2CCytobandData-method'></span><span id='topic+Idiogram+20Graphs'></span>

<h3>Description</h3>

<p>The <code>image</code> and <code>barplot</code> methods of the
<code><a href="#topic+CytobandData-class">CytobandData-class</a></code> provide flexible displays of genome
wide data that has been summarized at cytoband resolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'CytobandData'
image(x, chr, what,
         pal = palette(), nrows = 2, labels = NULL,
         horiz = FALSE, axes = chr != "all", debug = FALSE, legend = FALSE,
         sigcolumn = NA, sigcut = 0.01, alpha = 63, clip = FALSE)
## S4 method for signature 'CytobandData'
barplot(height, what, col = "blue", altcol = "#FED4C4",
                   ylab = "Percent", hline = NULL,
                   xform = function(x) x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CytobandData-methods_+3A_x">x</code></td>
<td>
<p>An object of the <code><a href="#topic+CytobandData-class">CytobandData-class</a></code>, which
combines numerical data at cytoband resolution with information on
the chromosomal locations of the cytobands.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_chr">chr</code></td>
<td>
<p>The specific chromosome you want to view, typically in
<code>c(1:22, "X", "Y")</code>. if you want to see all the chromosomes
at once then you can set <code>chr = "all"</code>.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_what">what</code></td>
<td>
<p>A vector or list containing the names of the numerical
column(s) that you want to display from the data frame. The
<code>barplot</code> method only shows a single data column at a time. The
plots resulting from the <code>image</code> method change depending on
whether you supply a vector or a list, as well as on the length. For
more details, see the vignettes, espefcialy the image gallery.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_pal">pal</code></td>
<td>
<p>a charcter vector containing the colors you want to use
for different data shown in the plot.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_horiz">horiz</code></td>
<td>
<p>A logical value determining whether images present the
chromosome idiograms horizontally or vertically.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_nrows">nrows</code></td>
<td>
<p>Only used when <code>chr = "all"</code> to determine the number
of rows to use to display different chromosomes. Must be an integer
between 1 and 4.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_labels">labels</code></td>
<td>
<p>Only use when <code>what</code> is a character vector (not a
list) and <code>chr</code> is mot equal to <code>"all"</code>. Used to label
differnt columns of displayed data.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_axes">axes</code></td>
<td>
<p>Logical value; should axes be displayed?</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_legend">legend</code></td>
<td>
<p>Logicla value; should a legend be added to the plot.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_sigcolumn">sigcolumn</code></td>
<td>
<p>The three parameters <code>sigcolumn</code>,
<code>sigcut</code>, and <code>alpha</code> are used as a set. The first names a
numerical column in the data set used to define &quot;significance&quot;. The
second is a vector of cutoffs that mark levels of significance. The
latter is a number between 0 and 255 denoting the transparency level
assigned to the color for each significance level.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_sigcut">sigcut</code></td>
<td>
<p>See <code>sigcolumn</code>.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_alpha">alpha</code></td>
<td>
<p>See <code>sigcolumn</code>.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_clip">clip</code></td>
<td>
<p>A logical value; should te length of the chromosome fill
the device (if TRUE) or be plotted relative to the length of Chromosome
1 (if FALSE). Currently only used when plotting two values, one on
either side of a single chromosome.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_debug">debug</code></td>
<td>
<p>Logical value; should the method print out debugging
information. Probably best to ignore.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_height">height</code></td>
<td>
<p>An object of the <code><a href="#topic+CytobandData-class">CytobandData-class</a></code>, which
combines numerical data at cytoband resolution with information on
the chromosomal locations of the cytobands.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_col">col</code></td>
<td>
<p>a character vector containing the colors you want to use
for different data shown in the plot.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_altcol">altcol</code></td>
<td>
<p>Determines the color used, lternating with white, in the
x-axis plot of all chromosomes.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_ylab">ylab</code></td>
<td>
<p>Label for the y-axis in a barplot.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_hline">hline</code></td>
<td>
<p>Numeric vector of heights at which toadd a horizontal
line.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_xform">xform</code></td>
<td>
<p>Function to transform the data before plotting. Defauly
is the identity map, which does nothing.</p>
</td></tr>
<tr><td><code id="CytobandData-methods_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Both the <code>image</code> and <code>barplot</code> methods invisibly return
their first argument, and object fo the <code>CytobandData</code> class.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Tally DG, Coombes KR. RCytoGPS: An R Package for Analyzing
and Visualizing Cytogenetic Data. In preparation.
</p>
<p>Abrams ZB, Tally DG,  Zhang L, Coombes CE, Payne PRO, Abruzzo LV,
Coombes KR. Pattern recognition in lymphoid malignancies using CytoGPS
and Mercator. Under review.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jsonDir &lt;- system.file("Examples/JSONfiles", package = "RCytoGPS")
x &lt;- readLGF(folder = jsonDir)
cytoData &lt;- data.frame(x[["CL"]], x[["frequency"]])
bandData &lt;- CytobandData(cytoData)
datacolumns &lt;- names(x[["frequency"]])
barplot(bandData, what = datacolumns[1], col="forestgreen")

image(bandData, what = datacolumns[1:3], chr = 2)
image(bandData, what = datacolumns[1:3], chr = "all")
image(bandData, what = as.list(datacolumns[1:2]), chr = 2)
image(bandData, what = as.list(datacolumns[1:2]), chr = "all")
</code></pre>

<hr>
<h2 id='cytobandLocations'>Cytoband Locations</h2><span id='topic+cytobandLocations'></span><span id='topic+idiocolors'></span>

<h3>Description</h3>

<p>This data set contains the genomic locations of cytobands based on
both the latest build of the human genome (GRch38) and the latest
update to the International Standard for human Cytogenomic
Nomenclature (ISCN). Note that the CRch38 locations are unchanged from
build GRch37 (hg19). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cytobandLocations)
</code></pre>


<h3>Format</h3>

<p>A data matrix (<code>cytobandLocations</code>) containing 868 rows and
5 columns. Each row contains one of the cytobands defined in the 2016
update to ISCN nomenclature. The rownames are the standard cytoband
names; for example, <code>1p36.33</code>. The columns are
</p>

<dl>
<dt>Chromosome</dt><dd><p>The name of the human chromosome, stored as
<code>chr#</code>.</p>
</dd>
<dt><code>loc.start</code></dt><dd><p>The starting base position of the band.</p>
</dd>
<dt><code>loc.end</code></dt><dd><p>The ending base position of the band.</p>
</dd>
<dt><code>Band</code></dt><dd><p>The band name without the chromosome; for
example, <code>p36.33</code>.</p>
</dd>
<dt><code>Stain</code></dt><dd><p>A factor containing the name of the Giemsa-stain
color of the band in a karyotype image.</p>
</dd>
</dl>

<p>Also, a vector (<code>idiocolors</code>) of length eight that translates the
Giemsa stain names into appropriate colors.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>Source</h3>

<p>The starting point for these data were the cytoband locations
downloaded from the UCSC Genome Browser. We confirmed that the data
were unchanged between human genome builds GRCh36 (hg18), GRCh37
(hg19), and GRCh38. However, the list of cytobands from UCSC was not
consistent with the list of cytobands in ISCN 2016. We manually edited
the source file to be compliant. It also matches the list of cytobands
produced at <a href="http://cytogps.org">http://cytogps.org</a>.
</p>


<h3>References</h3>

<p>J. McGowan-Jordan, A. Simons, M. Schmid (editors).
ISCN 2016: An International System for Human Cytogenomic Nomenclature.
Karger Publishing, Basel, 2016.
</p>

<hr>
<h2 id='preclean'>Pre-clean Karyotypes</h2><span id='topic+preclean'></span>

<h3>Description</h3>

<p>A function to clean karyotype data, by deleting known comments that do not
adhere to the ISCN standard.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preclean(x, targetColumns, dirt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preclean_+3A_x">x</code></td>
<td>
<p>A data frame containing at least one column of karyotype data.</p>
</td></tr>
<tr><td><code id="preclean_+3A_targetcolumns">targetColumns</code></td>
<td>
<p>Either a numeric vector of column indices or a
character vector of column names.</p>
</td></tr>
<tr><td><code id="preclean_+3A_dirt">dirt</code></td>
<td>
<p>A character vector containing items to delete from all karyotypes,</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The core input data worked on by the <code>RCytoGPS</code> are karyotypes,
which are text strings written to conform to the ISCN standard. At
many institutions, the cytogeneticists have developed idiosyncratic
conventions that they use to add comments into the string. In most
cases, these karyotypes are simply stored as text strings in a local
database. In particular, they are not checked for synatx or grammar
errors. By contrast, the implementation of the CytoGPS algorithm at
the web site <a href="http://cytogps.org">http://cytogps.org</a> uses a formal approach with
lexer and parser. As a result, many karyotypes are rejected by the
system.
</p>
<p>The <code>preclean</code> function uses <code><a href="base.html#topic+gsub">gsub</a></code> to delete a list
of known (local) comments from all karyotypes, making it more likley
that they will be successfully processed by the lexer and parser. We
provide an example list derived from experience at our own
institution.
</p>
<p><em>Implementation Note:</em> The <code>preclean</code> function removes
strings in the order that they are contained in the <code>dirt</code>
vector. So, you have to be carefully not to delete parts of a long
phrase before trying to delete the whole phrase. For example, you
should not remove &quot;clonal&quot; before removing &quot;nonclonal&quot;. 
</p>


<h3>Value</h3>

<p>A data frame of te same size and with the same number of columns as
the input data frame.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Zhang L, Abruzzo LV, Heerema NA, Li S, Dillon T, Rodriguez R,
Coombes KR, Payne PRO. CytoGPS: a web-enabled karyotype analysis tool
for cytogenetics. Bioinformatics. 2019 Dec 15;35(24):5365-5366.
</p>
<p>Abrams ZB, Tally DG,  Zhang L, Coombes CE, Payne PRO, Abruzzo LV,
Coombes KR. Pattern recognition in lymphoid malignancies using CytoGPS
and Mercator. In Press.
</p>
<p>Abrams ZB, Tally DG, Coombes KR. RCytoGPS: An R Package for Analyzing
and Visualizing Cytogenetic Data. In preparation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cleanDir &lt;- system.file("PreClean", package = "RCytoGPS")
bad &lt;- read.delim(file.path(cleanDir, "badStrings.txt"), header=FALSE)
bad &lt;- as.vector(as.matrix(bad))
input &lt;- read.csv(file.path(cleanDir, "startKaryotypes.csv"))
myclean &lt;- preclean(input, 4:5, bad)
</code></pre>

<hr>
<h2 id='readLGF'>Extracting LGF karyotype data from JSON files </h2><span id='topic+readLGF'></span>

<h3>Description</h3>

<p>A function to read binary karyotype data, stored in LGF format in JSON
files produced by the CytoGPS web site, into R for further analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readLGF(files = NULL, folder = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readLGF_+3A_files">files</code></td>
<td>
<p>The name of the JSON file (or a character vector of
such file names) from which you want to extract and format
data. If <code>NULL</code>, then it will extract all JSON files within
the folder path provided.</p>
</td></tr>
<tr><td><code id="readLGF_+3A_folder">folder</code></td>
<td>
<p>The specified directory/folder from which the user
wants to extract JSON files. If <code>NULL</code>, then the function
will look in the current working directory.</p>
</td></tr>
<tr><td><code id="readLGF_+3A_verbose">verbose</code></td>
<td>
<p>A logical value; should the function keep you informed about what it is doing?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>CytoGPS is an algorithm that converts conventional karyotypes from the
standard text-based notation (the International Standard for Human
Cytogenetic/Cytogenomic Nomenclature; ISCN) into binary vectors with
three bits (loss, gain, or fusion) per cytoband, which we call the LGF
model. The web site <a href="http://cytogps.org">http://cytogps.org</a> provides an
implementation that allows users to upload text files containing one
karyotype per line. It produces its output as a file in JavaScript
Object Notation (JSON).
</p>
<p>The <code>readLGF</code> function reads and parses these JSON files and
converts them into an R data structure. The <code>raw</code> component of
this structure contains binary matrices that can serve as input to the
Mercator package (see <code><a href="Mercator.html#topic+Mercator-class">Mercator-class</a></code>) for
unsupervised analyses. The <code>frequency</code> component summarizes the
fraction of input karyotype-clones with each abnormality, and can be
visualized with other tools in the <code>RCytoGPS</code> package.
</p>


<h3>Value</h3>

<p>A list containing five elements:
</p>
<table>
<tr><td><code>source:</code></td>
<td>
<p>A character vector containing the names of the
JSON files from which data was read.</p>
</td></tr>
<tr><td><code>raw:</code></td>
<td>
<p>A list of lists, one per JSON source file. Each
internal list contains two elements, <code>Status</code> and
<code>LGF</code>. <code>Status</code> is a data frame with one row per
karyotype in the input file, describing the results of CytoGPS
parsing and mapping. Results can be &quot;<code>Success</code>&quot;,
&quot;<code>Nonfixable grammar error</code>&quot;,
&quot;<code>Validation error</code>&quot;,
&quot;<code>Fixable grammar error and success</code>&quot;,  or
&quot;<code>Fixable grammar error but containing validation error</code>&quot;.
<code>LGF</code> is a data frame where the columns are LGF-cytobands and the
rows are clones from the successfully processed input karyotypes;
each karyotype can have multiple clones. Entries are zero or one
indicating the absence or presence of an abnormality.</p>
</td></tr>
<tr><td><code>frequency:</code></td>
<td>
<p>A data frame, where the rows are cytobands
and the columns are Loss, Gain, and Fusion, with three columns per
input file. Entries are the fraction of karyotype clones with
that abnormality.</p>
</td></tr>
<tr><td><code>size:</code></td>
<td>
<p>An integer vector containing the total number of
clones per input file. These values can be used to turn fractions
back into counts.</p>
</td></tr>
<tr><td><code>CL:</code></td>
<td>
<p>A data frame with one row per cytoband detailing
the chromosomal location and (grayscale) color of the band
produced by Giemsa staining.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Dwayne G. Tally <a href="mailto:dtally110@hotmail.com">dtally110@hotmail.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Zhang L, Abruzzo LV, Heerema NA, Li S, Dillon T, Rodriguez R,
Coombes KR, Payne PRO. CytoGPS: a web-enabled karyotype analysis tool
for cytogenetics. Bioinformatics. 2019 Dec 15;35(24):5365-5366.
</p>
<p>Abrams ZB, Tally DG,  Zhang L, Coombes CE, Payne PRO, Abruzzo LV,
Coombes KR. Pattern recognition in lymphoid malignancies using CytoGPS
and Mercator. Under review.
</p>
<p>Abrams ZB, Tally DG, Coombes KR. RCytoGPS: An R Package for Analyzing
and Visualizing Cytogenetic Data. In preparation.
</p>


<h3>See Also</h3>

<p><code><a href="Mercator.html#topic+Mercator-class">Mercator-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jsonDir &lt;- system.file("Examples/JSONfiles", package = "RCytoGPS")
x &lt;- readLGF(folder = jsonDir)

jsonFile &lt;- dir(jsonDir, pattern = "*.json")[1]
y &lt;- readLGF(jsonFile, jsonDir)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
