<!DOCTYPE html><html lang="en"><head><title>Help for package rda</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rda}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#brain'><p>Brain Cancer Microarray Data</p></a></li>
<li><a href='#brain.x'><p>Brain Cancer Microarray Expression Data</p></a></li>
<li><a href='#brain.y'><p>Brain Cancer Microarray Data Class Labels</p></a></li>
<li><a href='#colon'><p>Colon Cancer Microarray Data</p></a></li>
<li><a href='#colon.x'><p>Colon Cancer Microarray Expression Data</p></a></li>
<li><a href='#colon.y'><p>Colon Cancer Microarray Data Class Labels</p></a></li>
<li><a href='#genelist.rda'><p>RDA Shrunken Gene List Function</p></a></li>
<li><a href='#plot.rdacv'><p>A function that plots the result from rda.cv</p></a></li>
<li><a href='#predict.rda'><p>RDA Prediction Function</p></a></li>
<li><a href='#rda'><p>Main RDA Function</p></a></li>
<li><a href='#rda.cv'><p>RDA Cross Validation Function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.2-1</td>
</tr>
<tr>
<td>VersionNote:</td>
<td>Archived on 2020-01-01 on CRAN</td>
</tr>
<tr>
<td>Title:</td>
<td>Shrunken Centroids Regularized Discriminant Analysis</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions implementing the shrunken centroids regularized discriminant analysis for
        classification purpose in high dimensional data. The method is described in 
        Guo at al. (2013) &lt;<a href="https://doi.org/10.1093%2Fbiostatistics%2Fkxj035">doi:10.1093/biostatistics/kxj035</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/valentint/rda">https://github.com/valentint/rda</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/valentint/rda/issues">https://github.com/valentint/rda/issues</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-02 16:20:02 UTC</td>
</tr>
<tr>
<td>Author:</td>
<td>Yaqian Guo [aut],
  Trevor Hastie [aut],
  Robert Tibshirani [aut],
  Valentin Todorov <a href="https://orcid.org/0000-0003-4215-0245"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Valentin Todorov &lt;valentin@todorov.at&gt;</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-01 21:21:44 UTC; valen</td>
</tr>
</table>
<hr>
<h2 id='brain'>Brain Cancer Microarray Data</h2><span id='topic+brain'></span>

<h3>Description</h3>

<p>The <code>brain</code> data contains two objects: <code>brain.x</code>, microarray
expression data for 42 brain cancer samples and <code>brain.y</code>, the class
labels for these samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(brain)</code></pre>


<h3>Format</h3>

<p>An expression data matrix (42x5597) <code>brain.x</code> and a class label
vector (42) <code>brain.y</code> for 42 samples.
</p>


<h3>Details</h3>

<p><code>brain.y</code> is the class labels of the 42 samples. <code>brain.x</code> is
the microarray expression data matrix with each row representing a sample.
</p>


<h3>Source</h3>

<p>Pomeroy, S. et al. (2002) <em>Prediction of Central Nervous System
Embryonal Tumor Outcome Based on Gene Expression</em>. Nature, Vol 415,
p436-442.
</p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>

<hr>
<h2 id='brain.x'>Brain Cancer Microarray Expression Data</h2><span id='topic+brain.x'></span>

<h3>Description</h3>

<p><code>brain.x</code> is the microarray expression data for 42 brain cancer
samples.
</p>

<hr>
<h2 id='brain.y'>Brain Cancer Microarray Data Class Labels</h2><span id='topic+brain.y'></span>

<h3>Description</h3>

<p><code>brain.y</code> is the class labels for the brain cancer samples.
</p>

<hr>
<h2 id='colon'>Colon Cancer Microarray Data</h2><span id='topic+colon'></span>

<h3>Description</h3>

<p>The <code>colon</code> data contains two objects: <code>colon.x</code>, microarray
expression data for 62 colon cancer samples and <code>colon.y</code>, the class
labels for these samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(colon)</code></pre>


<h3>Format</h3>

<p>An expression data matrix (62x2000) <code>colon.x</code> and a class label
vector (62) <code>colon.y</code> for 62 samples.
</p>


<h3>Details</h3>

<p><code>colon.y</code> is the class labels of the 62 samples. <code>colon.x</code> is
the microarray expression data matrix with each row representing a sample.
</p>


<h3>Source</h3>

<p>Alon, U. et al. (1999) <em>Broad Patterns of Gene Expression Revealed
by Clustering Analysis of Tumor and Normal Colon Tissues Probed by
Oligonucleotide Arrays</em>. PNAS, Vol 96, p6745-6750. The data set is
available at <a href="http://microarray.princeton.edu/oncology/">http://microarray.princeton.edu/oncology/</a>.
</p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>

<hr>
<h2 id='colon.x'>Colon Cancer Microarray Expression Data</h2><span id='topic+colon.x'></span>

<h3>Description</h3>

<p><code>colon.x</code> is the microarray expression data for 62 colon cancer
samples.
</p>

<hr>
<h2 id='colon.y'>Colon Cancer Microarray Data Class Labels</h2><span id='topic+colon.y'></span>

<h3>Description</h3>

<p><code>colon.y</code> is the class labels for the colon cancer samples.
</p>

<hr>
<h2 id='genelist.rda'>RDA Shrunken Gene List Function</h2><span id='topic+genelist.rda'></span>

<h3>Description</h3>

<p>A function that returns the shrunken gene (variable) names by RDA for a
particular (alpha, delta) combination.</p>


<h3>Usage</h3>

<pre><code class='language-R'>genelist.rda(x, y, alpha, delta, prior=table(y)/length(y),
             gnames=NULL, regularization="S")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genelist.rda_+3A_x">x</code></td>
<td>
<p>The training data set for which you want to obtain the
shrunken gene list. It must be a numerical matrix. The columns are sample
observations and the rows are variables.</p>
</td></tr>
<tr><td><code id="genelist.rda_+3A_y">y</code></td>
<td>
<p>The class labels for the columns of 'x'.</p>
</td></tr>
<tr><td><code id="genelist.rda_+3A_prior">prior</code></td>
<td>
<p>A numerical vector that gives the prior proportion of each
class. By default, it is set to be the sample frequencies unless users
want to specify a different one.</p>
</td></tr>
<tr><td><code id="genelist.rda_+3A_alpha">alpha</code></td>
<td>
<p>A single regularization value for alpha. Users must supply
this option.</p>
</td></tr>
<tr><td><code id="genelist.rda_+3A_delta">delta</code></td>
<td>
<p>A threshold value for delta. Users must supply this 
option.</p>
</td></tr>
<tr><td><code id="genelist.rda_+3A_gnames">gnames</code></td>
<td>
<p>A character vector that specifies the names of the
variables of the training data set 'x'. By default, it is set to be
<code>NULL</code> and the function uses either the row names of 'x' (if it
exists) or the row index <code>1:nrow(x)</code>. Users can provide their
customized gene name list. However, the length of the name vector must be
the same as the number of rows of 'x'.</p>
</td></tr>
<tr><td><code id="genelist.rda_+3A_regularization">regularization</code></td>
<td>
<p>The type of regularization. It is either 'S' or
'R'. The default value is 'S'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>genelist.rda</code> will return a vector of names for those shrunken
genes by RDA for a particular (alpha, delta).</p>


<h3>Value</h3>

<p>A character vector of the names of the shrunken genes. 
</p>


<h3>Author(s)</h3>

<p> Yaqian Guo, Trevor Hastie and Robert Tibshirani </p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(colon)
colon.x &lt;- t(colon.x)
genenames &lt;- genelist.rda(colon.x, colon.y, alpha=0.1, delta=0.3)
</code></pre>

<hr>
<h2 id='plot.rdacv'>A function that plots the result from rda.cv</h2><span id='topic+plot.rdacv'></span>

<h3>Description</h3>

<p>Plot the cross validation error matrix and the number of
shrunken gene matrix obtained from RDA cross-validation analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rdacv'
plot(x, type=c("both", "error", "gene"), nice=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.rdacv_+3A_x">x</code></td>
<td>
<p>The fit from <code>rda.cv</code>.</p>
</td></tr>
<tr><td><code id="plot.rdacv_+3A_type">type</code></td>
<td>
<p>A character string specifying what to plot. If 'both',
then heatmaps for both cv error and shrunken genes are plotted; if
'error', only the error map is produced; if 'gene', only the gene
map is produced. This option is useful if users want to generate a
specific plot. Default is 'both'.</p>
</td></tr>
<tr><td><code id="plot.rdacv_+3A_nice">nice</code></td>
<td>
<p>A logical flag. If 'TRUE', then 1-dim curves are
plotted instead of heatmaps. This is useful when the length of
alpha or delta is small. Heatmap in this case can be awful-looking.
For example, if alpha=0.5 is a single value, while delta=seq(10),
then both cv error and shrunken genes will be plotted as a 1-dim
function of delta or vice versa when the length of delta is small.</p>
</td></tr>
<tr><td><code id="plot.rdacv_+3A_...">...</code></td>
<td>
<p>Additional arguments for generic plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plot.rdacv</code> produces two heatmaps for the cross validation
error matrix and the number of shrunken genes matrix obtained from
<code>rda.cv</code>.
</p>


<h3>Value</h3>

<p>A list of returning values:
</p>
<table role = "presentation">
<tr><td><code>one.se.pos</code></td>
<td>
<p>A 2-column matrix of the positions of the one standard 
error boundary points on the CV error heatmap. The first column indicates
the alpha positions and the second column for the delta positions.</p>
</td></tr>
<tr><td><code>min.cv.pos</code></td>
<td>
<p>A 2-column matrix of the positions of the minimal CV
error points on the CV error heatmap. The first column indicates
the alpha positions and the second column for the delta positions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Yaqian Guo, Trevor Hastie and Robert Tibshirani  </p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## These examples will take too long and R CMD check will complain ...
 
    data(colon)
    fit &lt;- rda(t(colon.x), colon.y)
    fit.cv &lt;- rda.cv(fit, x=t(colon.x), y=colon.y)
    plot.rdacv(fit.cv)

</code></pre>

<hr>
<h2 id='predict.rda'>RDA Prediction Function</h2><span id='topic+predict.rda'></span>

<h3>Description</h3>

<p>A function that predicts the class labels for new samples using RDA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rda'
predict(object, x, y, xnew, prior, alpha, delta,
            type=c("class", "posterior", "nonzero"),
            trace=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.rda_+3A_object">object</code></td>
<td>
<p>An <code>rda</code> fit object obtained from the function
<code>rda</code>.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_x">x</code></td>
<td>
<p>The training data matrix as used in the 'fit' object.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_y">y</code></td>
<td>
<p>The class labels for the columns of 'x' as used in the
'fit' object.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_xnew">xnew</code></td>
<td>
<p>The new data matrix used to predict the class labels of the
new samples. Must be a numerical matrix with rows corresponding to
variables and columns corresponding to the samples. The number of rows
must be the same as 'x'.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_prior">prior</code></td>
<td>
<p>A numerical vector that gives the prior proportion of
each class. By default, it is set to the <code>fit</code> component from the
training step unless users want to specify a new one for prediction.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_alpha">alpha</code></td>
<td>
<p>A particular regularization value for alpha. Often, this
is the optimal alpha value obtained from the cross-validation step.
But it could be any other value that users set. A vector of values is also
acceptable. If missing, the function will use the default values from
the <code>fit</code> component.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_delta">delta</code></td>
<td>
<p>A particular threshold value for delta. Often, this is
the optimal delta value obtained from the cross-validation step. But
it could be any other value that users set. A vector of values is also
acceptable. If missing, the function will use the default values from 
the <code>fit</code> component.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_type">type</code></td>
<td>
<p>A character string specifying which type of prediction
is desired. If 'class', then the predicted class labels are returned;
if 'posterior', then the predicted posterior probabilities for each
sample belonging to a class are returned; if 'nonzero', then the
indicators of shrunken genes are returned. 'class' is the default value.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_trace">trace</code></td>
<td>
<p>A logical flag indicating whether the intermediate steps
should be printed.</p>
</td></tr>
<tr><td><code id="predict.rda_+3A_...">...</code></td>
<td>
<p>Additional arguments for generic predict.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>predict.rda</code> does various predictions on the new test samples
based on fit from the training samples.
</p>


<h3>Value</h3>

<p>If option &quot;type='class'&quot;, the function will return the predicted class
labels for the new test samples. The format is a 3-dim array. The first
index corresponds to the alpha value(s) while the second index corresponds
to the delta value(s). The last index is the predicted labels for the
new samples. A reduced-dimensional array is possible if the length of
alpha or delta is 1.
</p>
<p>If option &quot;type='posterior'&quot;, the function will return the predicted
posterior probabilities of the new test samples belonging to different
classes. The format is a 4-dim array. The first index corresponds to the
alpha value(s) while the second index corresponds to the delta value(s).
The third index represents the samples in 'xnew'. The last index is the
class labels. A reduced-dimensional array is possible if the length of
alpha or delta is 1.
</p>
<p>If option &quot;type='nonzero'&quot;, the function will return a 3-dim indicator
array of the shrunken genes by RDA with 3 indices corresponding to alpha,
delta and the indices of the genes respectively. A reduced-dimensional
array is possible if the length of alpha or delta is 1.
</p>


<h3>Author(s)</h3>

<p>Yaqian Guo, Trevor Hastie and Robert Tibshirani</p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>


<h3>See Also</h3>

<p>Also see <code><a href="#topic+rda">rda</a></code> and <code><a href="#topic+rda.cv">rda.cv</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(colon)
colon.x &lt;- t(colon.x)

## divide the data set into a training set and a test
## set using a ratio of 2:1.
tr.index &lt;- sample(1:62, 40)
fit &lt;- rda(colon.x[, tr.index], colon.y[tr.index])

## predict the class labels of the test set at alpha=0.1
## and delta=0.5
ynew &lt;- predict(fit, x=colon.x[, tr.index], y=colon.y[tr.index], 
                xnew=colon.x[, -tr.index], alpha=0.1, delta=0.5)

## calculate the prediction error
sum(ynew != colon.y[-tr.index])
</code></pre>

<hr>
<h2 id='rda'>Main RDA Function</h2><span id='topic+rda'></span>

<h3>Description</h3>

<p>The function that does RDA analysis on high dimensional data, e.g.,
microarray expression data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rda(x, y, xnew=NULL, ynew=NULL, prior=table(y)/length(y),
    alpha=seq(0, 0.99, len=10), delta=seq(0, 3, len=10), 
    regularization=c("S", "R"), genelist=FALSE, trace=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rda_+3A_x">x</code></td>
<td>
<p>The training data set. It must be a numerical matrix.
The columns are sample observations and the rows are variables. For
example, in the microarray settings, &quot;x&quot; is the gene expression
matrix with the columns corresponding to the arrays while the rows
corresponding to the genes.</p>
</td></tr>
<tr><td><code id="rda_+3A_y">y</code></td>
<td>
<p>The class labels of the training samples (columns) in 'x', which
must be consecutive integers starting from 1.</p>
</td></tr>
<tr><td><code id="rda_+3A_xnew">xnew</code></td>
<td>
<p>The test data matrix. It has the same structure as 'x'.
The columns are samples and the rows are variables.</p>
</td></tr>
<tr><td><code id="rda_+3A_ynew">ynew</code></td>
<td>
<p>The class labels of the test samples. Same requirement as for
'y'.</p>
</td></tr>
<tr><td><code id="rda_+3A_prior">prior</code></td>
<td>
<p>A numerical vector that gives the prior proportion of
each class. Its length is equal to the number of classes. If not 
supplied, it is set to the sample proportions by default.</p>
</td></tr>
<tr><td><code id="rda_+3A_alpha">alpha</code></td>
<td>
<p>A numerical vector of the regularization values for alpha.
A single value is allowed. If not supplied, the default one will be used.</p>
</td></tr>
<tr><td><code id="rda_+3A_delta">delta</code></td>
<td>
<p>A numerical vector of the threshold values for delta.
A single value is allowed. If not supplied, the default one will be used.</p>
</td></tr>
<tr><td><code id="rda_+3A_regularization">regularization</code></td>
<td>
<p>Define which regularization method to use. 'S'
stands for regularization on covariance; 'R' stands for regularization on
correlation. 'S' is the default option.</p>
</td></tr>
<tr><td><code id="rda_+3A_genelist">genelist</code></td>
<td>
<p>A logical flag. If 'TRUE', then the function will return
an array of indices indicating the genes remained for each (alpha, delta)
combination. By default, this is set to 'FALSE'.</p>
</td></tr>
<tr><td><code id="rda_+3A_trace">trace</code></td>
<td>
<p>A logical flag. If 'TRUE', then the intermediate
computation steps will be displayed. Caution: this would lead to a
very long output display. By default, this is set to 'FALSE'.</p>
</td></tr>  
</table>


<h3>Details</h3>

<p><code>rda</code> does RDA analysis on high dimensional data. This is the main
function of the package.</p>


<h3>Value</h3>

<p>The function will return an 'rda' object with the following list of
components: 
</p>
<table role = "presentation">
<tr><td><code>alpha</code></td>
<td>
<p>The vector of the regularization values for alpha used in
the function.</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>The vector of the threshold values for delta used in the
function.</p>
</td></tr>
<tr><td><code>prior</code></td>
<td>
<p>The vector of the prior proportion of each class used
in the function.</p>
</td></tr>
<tr><td><code>error</code></td>
<td>
<p>The training error matrix. The rows correspond to the
alpha values while the columns correspond to the delta values.</p>
</td></tr> 
<tr><td><code>yhat</code></td>
<td>
<p>A 3-dim array giving the predicted class labels of 'y'.
The first index corresponds to the alpha values while the second index
corresponds to the delta values. The third index is the predicted class
labels for the corresponding samples. However, when the length of alpha or
delta is 1, this could be a 2-dim matrix or even a 1-dim vector.</p>
</td></tr>
<tr><td><code>ngene</code></td>
<td>
<p>The matrix of the number of shrunken genes. The rows
correspond to the alpha values while the columns correspond to the
delta values.</p>
</td></tr>
<tr><td><code>centroids</code></td>
<td>
<p>The group centroids matrix. It has the same number
of rows as 'x' and the number of columns is the total number of classes.
Each column is the centroids vector of the samples within that class.</p>
</td></tr>
<tr><td><code>centroid.overall</code></td>
<td>
<p>A single vector giving the grand mean vector
of all the samples in the 'x' matrix.</p>
</td></tr>
<tr><td><code>yhat.new</code></td>
<td>
<p>A 3-dim array of the predicted class labels for the
columns of 'xnew' if 'xnew' is provided. The first index corresponds to
the alpha values while the second index corresponds to the delta values.
The third index is the predicted class labels for the corresponding
samples. However, when the length of alpha or delta is 1, this can be a
2-dim matrix or even a 1-dim vector.</p>
</td></tr>
<tr><td><code>posterior</code></td>
<td>
<p>A 4-dim array giving the posterior probabilities of
each column of 'xnew' belonging to a class if 'xnew' is provided. The
first index corresponds to the alpha values while the second index
corresponds to the delta values. The third index is the corresponding
columns in 'xnew'. The last index corresponds to different classes.
However, an array of reduced dimensions may be produced if any of these
four indices has length of 1.</p>
</td></tr>
<tr><td><code>testerror</code></td>
<td>
<p>The test error matrix if both <code>xnew</code> and
<code>ynew</code> are supplied. The rows correspond to the alpha values
while the columns correspond to the delta values.</p>
</td></tr>
<tr><td><code>gene.list</code></td>
<td>
<p>A 3-dim array giving the indicator whether a gene
is shrunken or not for a particular (alpha, delta) if &quot;genelist&quot; option
is 'TRUE'. '0' means that gene is shrunken while '1' otherwise. The first
two indices correspond to alpha and delta. A reduced-dimensional array is
possible if either alpha or delta is of length 1.</p>
</td></tr>  
<tr><td><code>reg</code></td>
<td>
<p>The type of regularization used in calculation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yaqian Guo, Trevor Hastie and Robert Tibshirani</p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rda.cv">rda.cv</a></code> and <code><a href="#topic+predict.rda">predict.rda</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(colon)
colon.x &lt;- t(colon.x)
fit &lt;- rda(colon.x, colon.y)
</code></pre>

<hr>
<h2 id='rda.cv'>RDA Cross Validation Function</h2><span id='topic+rda.cv'></span>

<h3>Description</h3>

<p>A function that does RDA cross-validation analysis on the training data
set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rda.cv(fit, x, y, prior, alpha, delta, nfold=min(table(y), 10),
       folds=balanced.folds(y), trace=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rda.cv_+3A_fit">fit</code></td>
<td>
<p>An <code>rda</code> fit object obtained from the <code>rda</code>
function.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_x">x</code></td>
<td>
<p>The training data set as used in the <code>rda</code> function.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_y">y</code></td>
<td>
<p>The class labels of the training samples (columns) in &quot;x&quot; as
used in <code>rda</code> function.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_prior">prior</code></td>
<td>
<p>A numerical vector that gives the prior proportion of each
class. Its length should be equal to the number of classes. By default,
the function uses the one coming along with the <code>fit</code> object
unless users want to specify some other prior vector.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_alpha">alpha</code></td>
<td>
<p>A numerical vector of the regularization values for alpha.
By default, the function uses the one coming along with the <code>fit</code>
object unless users want to do cross-validation based on some other
values of alpha.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_delta">delta</code></td>
<td>
<p>A numerical vector of the threshold values for delta.
By default, the function uses the one coming along with the <code>fit</code>
object unless users want to do cross-validation based on some other
values of delta.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_nfold">nfold</code></td>
<td>
<p>An integer number to specify the number of folds in
the cross-validation analysis. This option is overwritten when the
<code>folds</code> option is specified at the same time.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_folds">folds</code></td>
<td>
<p>A list that provides the folds used in the cross-validation
analysis. Each component of the list is an integer vector of the sample
indices. See examples below for more details.</p>
</td></tr>
<tr><td><code id="rda.cv_+3A_trace">trace</code></td>
<td>
<p>A logical flag indicating whether the intermediate steps 
should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>rda.cv</code> does the RDA-based cross-validation on the training data 
set.
</p>


<h3>Value</h3>

<p>The <code>rda.cv</code> function will return an object of class <code>rdacv</code>
with the following list of components:
</p>
<table role = "presentation">
<tr><td><code>alpha</code></td>
<td>
<p>The vector of the regularization values for alpha used in
the cross-validation.</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>The vector of the threshold values for delta used in the
cross-validation.</p>
</td></tr>
<tr><td><code>prior</code></td>
<td>
<p>The vector of the prior proportion of each class used in
the cross-validation.</p>
</td></tr>
<tr><td><code>nfold</code></td>
<td>
<p>The number of folds used in the cross-validation.</p>
</td></tr>
<tr><td><code>folds</code></td>
<td>
<p>The folds used in the cross-validation.</p>
</td></tr>
<tr><td><code>yhat.new</code></td>
<td>
<p>The 3-dim array of the predicted class labels of the
training samples for each combination (alpha, delta). The first index
corresponds to the alpha values while the second index corresponds to
the delta values. The third index is the predicted class labels for the
corresponding samples.</p>
</td></tr>
<tr><td><code>err</code></td>
<td>
<p>The training error matrix from cross-validation. The rows
correspond to the alpha values while the columns correspond to the delta
values. It is automatically generated by the function.</p>
</td></tr>
<tr><td><code>cv.err</code></td>
<td>
<p>The test error (or cross-validation error) matrix. The
rows correspond to the alpha values while the columns correspond to
the delta values.</p>
</td></tr>
<tr><td><code>ngene</code></td>
<td>
<p>The matrix of the number of shrunken genes. The rows
correspond to the alpha values while the columns correspond to
the delta values. Note: the number of shrunken genes is based on
the average result from cross-validation.</p>
</td></tr>
<tr><td><code>reg</code></td>
<td>
<p>The type of regularization used in cross-validation.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>The sample size of the training data set.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yaqian Guo, Trevor Hastie and Robert Tibshirani</p>


<h3>References</h3>

<p>Y. Guo, T. Hastie, R. Tibshirani, (2006).
Regularized linear discriminant analysis and its application in microarrays, 
<em>Biostatistics</em> <b>8</b> pp. 86&ndash;100.
<a href="https://doi.org/10.1093/biostatistics/kxj035">doi:10.1093/biostatistics/kxj035</a>.
</p>


<h3>See Also</h3>

<p>Also see <code><a href="#topic+rda">rda</a></code> and <code><a href="#topic+predict.rda">predict.rda</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## These examples will take too long and R CMD check will complain ...

    data(colon)
    colon.x &lt;- t(colon.x)
    fit &lt;- rda(colon.x, colon.y)
    fit.cv &lt;- rda.cv(fit, x=colon.x, y=colon.y)
    
    ## to use the customized folds in cross-validation,
    ## for example, 6-fold with 11, 11, 10, 10, 10, 10 samples 
    ## in the respective folds, you can do the follows:
    index &lt;- sample(1:62, 62)
    folds &lt;- list()
    folds[[1]] &lt;- index[1:11]
    folds[[2]] &lt;- index[12:22]
    folds[[3]] &lt;- index[23:32]
    folds[[4]] &lt;- index[33:42]
    folds[[5]] &lt;- index[43:52]
    folds[[6]] &lt;- index[53:62]
    fit.cv &lt;- rda.cv(fit, colon.x, colon.y, folds=folds)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
