<!DOCTYPE html><html><head><title>Help for package rdi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rdi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calcRDI'><p>Calculate repertoire distances</p></a></li>
<li><a href='#calcVDJcounts'><p>Calculate repertoire counts</p></a></li>
<li><a href='#convertRDI'><p>Convert RDI measures</p></a></li>
<li><a href='#plotRDIladder'><p>RDI ladder plotting function</p></a></li>
<li><a href='#rdi'><p>Calculate RDI dissimilarity matrix</p></a></li>
<li><a href='#rdiAxis'><p>RDI Axis annotation function</p></a></li>
<li><a href='#rdiLadder'><p>RDI ladder</p></a></li>
<li><a href='#rdiModel'><p>RDI Models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-05-01</td>
</tr>
<tr>
<td>Title:</td>
<td>Repertoire Dissimilarity Index</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods for calculation and visualization of the Repertoire
    Dissimilarity Index. Citation: Bolen and Rubelt, et al (2017) 
    &lt;<a href="https://doi.org/10.1186%2Fs12859-017-1556-5">doi:10.1186/s12859-017-1556-5</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://rdi.readthedocs.io">http://rdi.readthedocs.io</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://bitbucket.org/cbolen1/rdicore/issues">https://bitbucket.org/cbolen1/rdicore/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>beanplot, gplots, pdist, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, ggplot2</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-04 14:59:52 UTC; bolenc</td>
</tr>
<tr>
<td>Author:</td>
<td>Christopher Bolen [aut, cre],
  Florian Rubelt [aut],
  Jason Vander Heiden [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christopher Bolen &lt;cbolen1@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-07 11:14:21 UTC</td>
</tr>
</table>
<hr>
<h2 id='calcRDI'>Calculate repertoire distances</h2><span id='topic+calcRDI'></span>

<h3>Description</h3>

<p>Calculate repertoire distances from a matrix of vdjCounts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcRDI(vdjCounts, distMethod = c("euclidean", "cor"), subsample = TRUE,
  nIter = 100, constScale = TRUE, units = c("lfc", "pct"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcRDI_+3A_vdjcounts">vdjCounts</code></td>
<td>
<p>a matrix of repertoire counts, as created by calcVDJCounts</p>
</td></tr>
<tr><td><code id="calcRDI_+3A_distmethod">distMethod</code></td>
<td>
<p>one of c(&quot;euclidean&quot;,&quot;cor&quot;) determining how to calculate the 
distance from the matrix of vdj counts. See Details.</p>
</td></tr>
<tr><td><code id="calcRDI_+3A_subsample">subsample</code></td>
<td>
<p>logical; if true, all repertoires are subsampled to be equal 
size.</p>
</td></tr>
<tr><td><code id="calcRDI_+3A_niter">nIter</code></td>
<td>
<p>value defining how many times the subsampling should be 
repeated. Only used if subsample is TRUE.</p>
</td></tr>
<tr><td><code id="calcRDI_+3A_constscale">constScale</code></td>
<td>
<p>logical; if <code>TRUE</code>, vdjCounts will be scaled such that the sum of
each column will be equal to 500 counts (an arbitrary constant). 
Otherwise, the columns will be scaled to the average count of all 
the columns.</p>
</td></tr>
<tr><td><code id="calcRDI_+3A_units">units</code></td>
<td>
<p>One of &quot;lfc&quot; or &quot;pct&quot;. This determines the method used for 
transforming the repertoire counts. See Details.</p>
</td></tr>
<tr><td><code id="calcRDI_+3A_...">...</code></td>
<td>
<p>additional parameters; these are ignored by this function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two options for distance methods, &quot;euclidean&quot; and &quot;cor&quot;. Euclidean refers to
standard euclidean distance, and is the standard for the RDI measure described in 
(Bolen et al. Bioinformatics 2016). In contrast, cor refers to a correlation-based 
distance metric, where the distance is defined as <code>(1-correlation)</code> between each 
column of vdjCounts. 
</p>
<p>The <code>units</code> parameter is used to determine the transformation function for the
repertoire counts. If <code>units='lfc'</code> (default), then the arcsinh transformation 
is applied to the count matrix, resulting in a distance metric which 
will scale with the average log fold change of each gene. In contrast,
<code>units='pct'</code> will result in no transformation of the count matrix, and distances
will be proportional to the average percent change of each gene, instead. Note that
&quot;units&quot; is a bit of a misnomer, as the distance metric doesn't actually represent the
true log-fold or percent change in the repertoires. In order to actually estimate 
these parameters, refer to the <a href="#topic+rdiModel">rdiModel</a> and <a href="#topic+convertRDI">convertRDI</a>
functions.
</p>


<h3>Value</h3>

<p>A dissimilarity structure containing distances between repertoires, averaged
across each subsampe run.
In addition to the standard attributes in a dist object, 
three additional attributes are defined as follows:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <em>ngenes</em>  </td><td style="text-align: left;"> integers, the number of genes in each column of "genes" 
                                that were included in at least one repertoire.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <em>nseq</em>  </td><td style="text-align: left;"> integer, the number of sequences used after subsampling  
                              the repertoires. If <code>subsample=FALSE</code>, this is not
                              defined.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <em>units</em> </td><td style="text-align: left;"> string, either "lfc" or "pct", depending on the "units" in the 
                    original call
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
#create genes
genes = sample(letters, 10000, replace=TRUE)

#create sequence annotations
seqAnnot = data.frame(donor = sample(1:4, 10000, replace=TRUE),
                      cellType = sample(c("B","T"), 10000, replace=TRUE)
                     )
##generate repertoire counts
cts = calcVDJcounts(genes,seqAnnot) 

##calculate RDI 
d = calcRDI(cts)

##calculate RDI in percent space
d_pct = calcRDI(cts,units="pct")

##convert RDI to actual 'lfc' estimates and compare
dtrue = convertRDI(d)$pred
plot(d, dtrue)

</code></pre>

<hr>
<h2 id='calcVDJcounts'>Calculate repertoire counts</h2><span id='topic+calcVDJcounts'></span>

<h3>Description</h3>

<p>Create count matrices for a set of repertoires
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcVDJcounts(genes, seqAnnot, select = NULL, combine = NULL,
  vdjDrop = NULL, splitBy = NULL, simplifyNames = TRUE,
  splitCommas = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcVDJcounts_+3A_genes">genes</code></td>
<td>
<p>vector (or matrix) containing the gene calls for each sequence.
If genes is a matrix, counts will be calculated for each 
column of 'genes', and the resulting count matrices will be
concatenated. See Details.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_seqannot">seqAnnot</code></td>
<td>
<p>matrix containing repertoire annotations.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_select">select</code></td>
<td>
<p>a list containing definitions of which repertoires to use. 
See Details.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_combine">combine</code></td>
<td>
<p>a list defining repertoires that should be grouped together.
See Details.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_vdjdrop">vdjDrop</code></td>
<td>
<p>a list specifying specific genes to exclude from analysis.
See Details.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_splitby">splitBy</code></td>
<td>
<p>the columns in seqAnnot to use for splitting repertoires. 
Default is to use all columns in seqAnnot.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_simplifynames">simplifyNames</code></td>
<td>
<p>logical; if true, any columns of seqAnnot where all selected
(and collapsed) sequences share the same value will not be 
used to make the names of sequenceCodes.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_splitcommas">splitCommas</code></td>
<td>
<p>logical; if true, seqAnnot is assumed to contain.
comma-separated lists of sequence annotations, which will be 
split up before generating sequence codes. Note: setting this 
to TRUE will make processing much slower.</p>
</td></tr>
<tr><td><code id="calcVDJcounts_+3A_...">...</code></td>
<td>
<p>additional parameters; these are ignored by this function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most cases, <code>genes</code> will be a single vector or one-column matrix. However, 
there are some cases where a row of <code>seqAnnot</code> corresponds to two (or more) genes
(e.g. the V and J gene segments of a single immune sequence). Rather than make multiple
rows for each gene, the <code>calcVDJcounts</code> function provides the option to provide 
a multi-column matrix for <code>genes</code>. The counts for each column will be tallied 
separately, and are then concatenated.
</p>
<p>To ensure equal variance across all repertoires, the default RDI metric uses 
subsampling to ensure that all repertoires have the same number of sequences. The 
default RDI metric subsamples all repertoires to the size of the smallest repertoire, 
which may result in a loss of power for comparisons between larger repertoires.
In order to increase power for various tests, it is often useful to only calculate 
the repertoire counts for a subset of the repertoires in seqAnnot. This can be done by
using the <code>select</code> and <code>combine</code> parameters to specify which 
repertoires to include in the analysis.
</p>
<p>Both parameters are lists containing entries
with the same name as one of the columns of seqAnnot. For <code>select</code>, each entry is 
a vector defining which values to include (e.g., to include only Visit 1 and 3, you 
might specify <code>select=list(visit=c("V1","V3"))</code>, where the <code>'visit'</code> column 
in seqAnnot contains the values <code>"V1"</code>,<code>"V2"</code>, and <code>"V3"</code>). In this 
case, any rows of <code>genes</code> and <code>seqAnnot</code> that come from a repertoire not 
specified in <code>select</code> will be discarded. By default, if a <code>select</code> code is 
not specified for a column in <code>seqAnnot</code>, all values from that column will be 
included.
</p>
<p>The <code>combine</code> parameter works in a similar fashion, but instead of a vector
describing which parameters to include, you can specify a vector of regular 
expressions, and any values of the <code>seqAnnot</code> column that match the regular
expression will be combined into a single repertoire (e.g. to combine visits 1 and 3 
into a single repertoire, you might specify <code>combine=list(visit="V[13]")</code>). 
</p>
<p>The <code>vdjDrop</code> parameter is also useful for limiting sequences. Like 
<code>select</code> and <code>combine</code>, this is a named list, with entries corresponding to 
the columns of <code>genes</code>. Each entry of <code>vdjDrop</code> is a vector of gene segment
names to remove from the analysis. All sequences containing those genes are removed 
from the analysis before subsampling.
</p>
<p>Once unwanted rows have been removed, the columns of <code>seqAnnot</code> are concatenated 
to generate &quot;repertoire&quot; labels for each row. The repertoire labels are then used
to split the rows of <code>genes</code>, and gene prevalence is tallied within a repertoire.
By default, columns of <code>seqAnnot</code> that are constant after subsetting will not be
included in the label. However, this can be controlled by the <code>simplifyNames</code> 
parameter. If <code>simplifyNames</code> is FALSE, all columns of  <code>seqAnnot</code> are 
included when generating labels.
</p>


<h3>Value</h3>

<p>A matrix where each row represents a gene, and each column represents a 
repertoire.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#create genes
genes = sample(letters, 10000, replace=TRUE)

#create sequence annotations
seqAnnot = data.frame(donor = sample(1:4, 10000, replace=TRUE),
                      visit = sample(c("V1","V2","V3"), 10000, replace=TRUE),
                      cellType = sample(c("B","T"), 10000, replace=TRUE)
                     )
                     
##generate repertoire counts for all repertoires
cts = calcVDJcounts(genes,seqAnnot) 

##Only include visit 1
cts = calcVDJcounts(genes,seqAnnot, select=list(visit="V1"))


## Just T cell repertoires, combining visit 1 and 3 together, and dropping visit 2
cts = calcVDJcounts(genes,seqAnnot, 
                    select=list(cellType="T", visit=c("V1","V3")), 
                    combine=list(visit="V[13]")) 

</code></pre>

<hr>
<h2 id='convertRDI'>Convert RDI measures</h2><span id='topic+convertRDI'></span>

<h3>Description</h3>

<p>Method to convert RDI values to fold/percent change
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertRDI(d, models = NULL, calcSD = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convertRDI_+3A_d">d</code></td>
<td>
<p>Distance matrix (as produced by <a href="#topic+calcRDI">calcRDI</a>), or a vector of 
distances.</p>
</td></tr>
<tr><td><code id="convertRDI_+3A_models">models</code></td>
<td>
<p>Set of RDI models, as produced by <a href="#topic+rdiModel">rdiModel</a>. If <code>NULL</code>,
RDI models will be calculated based on the attributes in the distance
matrix.</p>
</td></tr>
<tr><td><code id="convertRDI_+3A_calcsd">calcSD</code></td>
<td>
<p>logical; if <code>TRUE</code>, standard deviations for each estimate will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The convertRDI function works by first generating a model for the RDI values at a given
repertoire size and feature count using the <a href="#topic+rdiModel">rdiModel</a> function (see that 
method's help file for more details). The RDI models predict the average 
log-fold/percent change across a range of RDI values, and allows us to convert RDI to
a more stable and interpretable metric.  
</p>
<p>In addition to the average log-fold or percent change value, <a href="#topic+rdiModel">rdiModel</a> 
also generates models for the standard deviation at each RDI value. This is useful for
understanding the confidence intervals around the fold change estimate.
</p>


<h3>Value</h3>

<p>A list containing either one or two features:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <em>pred</em>  </td><td style="text-align: left;"> The converted predictions; same length as <code>d</code>. </td>
</tr>
<tr>
 <td style="text-align: right;">
  <em>sd</em>  </td><td style="text-align: left;"> If <code>calcSD==T</code>, a set of standard deviation estimates for each
               prediction. 
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
#create genes
genes = sample(letters, 10000, replace=TRUE)
#create sequence annotations
seqAnnot = data.frame(donor = sample(1:4, 10000, replace=TRUE))
#calculate RDI
d = rdi(genes, seqAnnot)

##convert RDI to actual 'lfc' estimates and compare
dtrue = convertRDI(d)$pred
plot(d, dtrue)

##look at SD ranges around lfc estimates
dtrue = convertRDI(d, calcSD=TRUE)

##plot using ggplot2
library(ggplot2)
x = as.numeric(d)
y = as.numeric(dtrue$pred)
sd = as.numeric(dtrue$sd)
qplot(x,y)+geom_errorbar(aes(x=x, ymin=y-sd, ymax=y+sd))

</code></pre>

<hr>
<h2 id='plotRDIladder'>RDI ladder plotting function</h2><span id='topic+plotRDIladder'></span>

<h3>Description</h3>

<p>function for adding a pre-computed RDI ladder onto a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRDIladder(ladder, side = 4, toPlot = NULL, labelLadder = TRUE,
  add = TRUE, cex = 0.7, lineCol = NULL, fillCol = "#AAAAAA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRDIladder_+3A_ladder">ladder</code></td>
<td>
<p>the ladder object to add, as created by <a href="#topic+rdiLadder">rdiLadder</a></p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_side">side</code></td>
<td>
<p>integer; value between 1 and 4 indicating where the ladder will be 
added. 1 - bottom, 2 - left, 3 - top, 4 - right.</p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_toplot">toPlot</code></td>
<td>
<p>logical vector; which ladders should be plotted? By default, 
ladders that are significantly overlapped by their neighbor and
those that are majority outside the plotting region are removed.</p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_labelladder">labelLadder</code></td>
<td>
<p>logical; if <code>TRUE</code>, each curve will be annotated with the ladder name</p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_add">add</code></td>
<td>
<p>logical; if <code>TRUE</code>, the ladder will be added to the current plot</p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_cex">cex</code></td>
<td>
<p>character expansion for ladder labels.</p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_linecol">lineCol</code></td>
<td>
<p>the colors to be used for the ladder border. If the length of <code>col</code> 
is less than the length of ladder, <code>col</code> will be recycled.</p>
</td></tr>
<tr><td><code id="plotRDIladder_+3A_fillcol">fillCol</code></td>
<td>
<p>the colors to be used to fill the ladder. If the length of <code>col</code> 
is less than the length of ladder, <code>col</code> will be recycled.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used in conjunction with <a href="#topic+rdiLadder">rdiLadder</a> to add a useful 
annotation to any plot containing RDI values. 
</p>
<p>Because RDI values vary according to the number of genes and size of the repertoires,
they are not useful as numbers by themselves. Instead, it is useful to compare them 
with estimates of the true difference between the two repertoires. This function
adds a series of density curves along one side of a standard plotting region, each
one representing the most likely RDI values between two repertoires that vary by a set
amount. 
</p>
<p>By default, not all density curves from the <code>ladder</code> parameter are plotted. 
Instead, the function intelligently chooses which ladders to plot based on the amount
of overlap between neighboring ladders. If a ladder is significantly overlapped by the
ladder below it, then the ladder will not be plotted. In addition, if the mean of a 
ladder is outside the main plotting region, it will be dropped. In order to control this 
behavior, you can directly specify which ladders are plotted using the <code>toPlot</code> 
parameter.
</p>


<h3>Value</h3>

<p>Invisibly returns the location of the ladder (if side 1 or 3, the y location; 
otherwise, the x location).
</p>


<h3>See Also</h3>

<p><a href="#topic+rdiLadder">rdiLadder</a>, <a href="#topic+rdiModel">rdiModel</a>, <a href="#topic+rdiAxis">rdiAxis</a>
</p>

<hr>
<h2 id='rdi'>Calculate RDI dissimilarity matrix</h2><span id='topic+rdi'></span>

<h3>Description</h3>

<p>Wrapper function for calculating RDIs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdi(genes, seqAnnot, params = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rdi_+3A_genes">genes</code></td>
<td>
<p>vector (or matrix) containing the gene calls for each sequence.
If genes is a matrix, counts will be calculated for each 
column of 'genes', and the resulting count matrices will be
concatenated.</p>
</td></tr>
<tr><td><code id="rdi_+3A_seqannot">seqAnnot</code></td>
<td>
<p>matrix containing repertoire annotations. Must be same length
as 'genes'.</p>
</td></tr>
<tr><td><code id="rdi_+3A_params">params</code></td>
<td>
<p>list; contains parameters to pass to child functions.
Should contain <code>countParams</code> and <code>distParams</code> lists,
which contain parameters for <a href="#topic+calcVDJcounts">calcVDJcounts</a> and
<a href="#topic+calcRDI">calcRDI</a>, respectively. See Details.</p>
</td></tr>
<tr><td><code id="rdi_+3A_...">...</code></td>
<td>
<p>other parameters to pass to <a href="#topic+calcVDJcounts">calcVDJcounts</a> and 
<a href="#topic+calcRDI">calcRDI</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper for the two core functions of RDI, 
<a href="#topic+calcVDJcounts">calcVDJcounts</a> and <a href="#topic+calcRDI">calcRDI</a>. To control the function of 
both <code>calcVDJcounts</code> and <code>calcRDI</code>, additional parameters can be specified
either directly in the RDI function call, or parameters for the individual functions 
can be wrapped up into lists of parameters and passed into the <code>params</code> parameter.
<code>params</code> should be a list containing at least one of two parameter lists:
<code>countParams</code> and <code>distParams</code>, which  
will be passed to <code>calcVDJcounts</code> and <code>calcRDI</code>, respectively. An example
analysis is included below.
</p>


<h3>Value</h3>

<p>Dissimilarity structure, as calculated by dist. In addition to the standard
attributes returned by dist, two additional attributes are defined as follows:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <em>nseq</em>  </td><td style="text-align: left;"> integer, the number of sequences used after subsampling  
                              the repertoires</td>
</tr>
<tr>
 <td style="text-align: right;">
  <em>ngenes</em>  </td><td style="text-align: left;"> integers, the number of genes in each column of "genes" 
                                that were included in at least one repertoire.
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
#create genes
genes = sample(letters, 10000, replace=TRUE)

#create sequence annotations
seqAnnot = data.frame(donor = sample(1:4, 10000, replace=TRUE),
                      visit = sample(c("V1","V2","V3"), 10000, replace=TRUE),
                      cellType = sample(c("B","T"), 10000, replace=TRUE)
                     )
                     
#parameters
params = list(
  countParams = list(
    select = list(
      visit = c("V1","V3"),
      cellType = "B"
    ),
    combine = list(
      visit = "V[13]"
    ),
    simplifyNames = FALSE
  ),
  distParams = list(
    constScale=FALSE
  )
)

##calculate RDI
d = rdi(genes, seqAnnot, params)

##plot using hierarchical clustering
plot(hclust(d))

</code></pre>

<hr>
<h2 id='rdiAxis'>RDI Axis annotation function</h2><span id='topic+rdiAxis'></span>

<h3>Description</h3>

<p>This function takes a RDI model, as generated by <a href="#topic+rdiModel">rdiModel</a>, 
and adds an axis with annotations in the fold change space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdiAxis(model, side = 2, at = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rdiAxis_+3A_model">model</code></td>
<td>
<p>A model object, as generated by <a href="#topic+rdiModel">rdiModel</a>.</p>
</td></tr>
<tr><td><code id="rdiAxis_+3A_side">side</code></td>
<td>
<p>The side the axis will be added to. (1 - bottom; 2 - left; 3 - top;
4 - right). Default is 2.</p>
</td></tr>
<tr><td><code id="rdiAxis_+3A_at">at</code></td>
<td>
<p>The points at which the tick marks are drawn. By default, 
tickmarks are placed at 'round' fold/percent change values
using the &quot;<a href="base.html#topic+pretty">pretty</a>&quot; breakpoints function. This may
not be ideal if log-RDI values are being plotted.</p>
</td></tr>
<tr><td><code id="rdiAxis_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass to <a href="graphics.html#topic+axis">axis</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is designed to replace the default axes generated by a plot 
function. Instead of annotating the true RDI value, <code>rdiAxis</code> will estimate
the &quot;true difference&quot; values at various points within the plotting region, and will
annotate the axis with those estimates. 
</p>
<p>It is worth noting that although the RDI value can range below <code>rdiModel</code>'s 
estimate for &quot;identical&quot; repertoires, no negative true difference values will be 
annotated, as these values do not make sense.
</p>


<h3>See Also</h3>

<p><a href="#topic+rdiModel">rdiModel</a>, <a href="#topic+rdiLadder">rdiLadder</a>, <a href="#topic+plotRDIladder">plotRDIladder</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#create genes
genes = sample(letters, 10000, replace=TRUE)
#create sequence annotations
seqAnnot = data.frame(donor = sample(1:10, 10000, replace=TRUE))
#calculate RDI
d = rdi(genes, seqAnnot)

##create a "baseVect" with the same probability as our features
##since we sampled uniformly, the base vector has equal probability
baseVect = rep(1/length(letters),length(letters))

##generate an RDI model
m = rdiModel(attr(d, "nseq"), baseVects=baseVect)

##convert RDI to lfc
td = convertRDI(d,models=m)$pred

par(mar=c(4,4,1,4),las=1,mgp=c(3,0.5,0))
plot(td,d, ylab="RDI", xlab="LFC")

##now add "true difference" axis annotation to the right side of the plot
rdiAxis(m, side=4)

</code></pre>

<hr>
<h2 id='rdiLadder'>RDI ladder</h2><span id='topic+rdiLadder'></span>

<h3>Description</h3>

<p>Function for creating the RDI ladder for a specific number of sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdiLadder(n, ngenes = NULL, baseVects = NULL, diffPoints = NULL,
  units = c("lfc", "pct"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rdiLadder_+3A_n">n</code></td>
<td>
<p>the repertoire size; alternatively, an rdiModel object as created by
<a href="#topic+rdiModel">rdiModel</a>.</p>
</td></tr>
<tr><td><code id="rdiLadder_+3A_ngenes">ngenes</code></td>
<td>
<p>numeric vector indicating the number of genes in each chain.
If baseVect is not provided, this parameter will be used to 
generate a base prevalence vector for each of the genes.</p>
</td></tr>
<tr><td><code id="rdiLadder_+3A_basevects">baseVects</code></td>
<td>
<p>A vector or list of vectors representing the total prevalence of 
each gene (for each chain) in the dataset. See <a href="#topic+rdiModel">rdiModel</a> for details.</p>
</td></tr>
<tr><td><code id="rdiLadder_+3A_diffpoints">diffPoints</code></td>
<td>
<p>numeric vector; each value specifices either a log2 fold change or 
percent deviation value (depending on the 'units') at which 
the RDI ladder will be calculated.</p>
</td></tr>
<tr><td><code id="rdiLadder_+3A_units">units</code></td>
<td>
<p>String; either &quot;lfc&quot; or &quot;pct&quot;, depending on what transform was used
in the original RDI calculations. See Details.</p>
</td></tr>
<tr><td><code id="rdiLadder_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <a href="#topic+rdiModel">rdiModel</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because RDI values vary according to the number of genes and size of the repertoires,
they are not useful as numbers by themselves. Instead, it is useful to compare them 
with estimates of the true difference between the two repertoires. 
This function uses the models generated by <a href="#topic+rdiModel">rdiModel</a> to generate estimated
RDI values corresponding to a set of pre-defined true distance (log-fold change or 
percent) values. This function is primarily meant to be used in conjunction with
<a href="#topic+plotRDIladder">plotRDIladder</a> in order to add a useful reference point for RDI values. 
</p>
<p>The units used for the RDI model should always match the units used to generate your
RDI values. For more details on units, refer to the details of <a href="#topic+calcRDI">calcRDI</a>
</p>


<h3>Value</h3>

<p>A list of the same length as diffPoints, with each entry in the list containing
the mean RDI value and standard deviation corresponding to a given true difference value.
</p>


<h3>See Also</h3>

<p><a href="#topic+plotRDIladder">plotRDIladder</a>, <a href="#topic+rdiModel">rdiModel</a>, <a href="#topic+rdiAxis">rdiAxis</a>
</p>

<hr>
<h2 id='rdiModel'>RDI Models</h2><span id='topic+rdiModel'></span>

<h3>Description</h3>

<p>Generate models equating RDI values to true differences in underlying 
prevalence values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdiModel(n, ngenes = NULL, baseVects = NULL, nIter = 50, nSample = 20,
  units = c("lfc", "pct"), constScale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rdiModel_+3A_n">n</code></td>
<td>
<p>the repertoire size.</p>
</td></tr>
<tr><td><code id="rdiModel_+3A_ngenes">ngenes</code></td>
<td>
<p>numeric vector indicating the number of genes in each chain.
If <code>baseVects</code> is not provided, this parameter will be used to 
generate a base prevalence vector for each of the genes.</p>
</td></tr>
<tr><td><code id="rdiModel_+3A_basevects">baseVects</code></td>
<td>
<p>A vector or list of vectors representing the total prevalence of 
each gene (for each chain) in the dataset. 
Differential datasets will be created from alterations of
this vector. If not provided, a base vector will be randomly .
generated at each subsample step containing the number of genes
specified by ngenes.</p>
</td></tr>
<tr><td><code id="rdiModel_+3A_niter">nIter</code></td>
<td>
<p>The number of iterations (i.e. number of datasets to generate).</p>
</td></tr>
<tr><td><code id="rdiModel_+3A_nsample">nSample</code></td>
<td>
<p>The number of samples to generate for each subsample. Each sample 
will have a different true fold change, but the same starting vector</p>
</td></tr>
<tr><td><code id="rdiModel_+3A_units">units</code></td>
<td>
<p>String; either &quot;lfc&quot; or &quot;pct&quot;, depending on what transform was used
in the original RDI calculations. See Details.</p>
</td></tr>
<tr><td><code id="rdiModel_+3A_constscale">constScale</code></td>
<td>
<p>logical; if <code>TRUE</code>, vdjCounts will be scaled such that the sum of
each column will be equal to 500 (an arbitrary constant). Otherwise,
the columns will be scaled to the average count of all the columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method uses simulated sequencing datasets to estimate the RDI values for datasets
with a known true deviation. 
</p>
<p>Briefly, a baseline probability vector (either randomly generated or supplied by the
<code>baseVects</code> parameter) is randomly perturbed, and the difference between the 
baseline vector and the perturbed vector is calculated. Then, <code>nSample</code> sequencing
datasets of size n are randomly drawn from both the baseline vector and the perturbed 
vector, and the RDI distance between all datasets calculated. This process is repeated 
<code>nIter</code> times, resulting in a dataset of RDI values and matched true differences.
A set of spline models is then fit to the data: one from RDI to true difference, and 
another from true difference to RDI value, allowing for bi-directional conversions. 
</p>
<p>If a baseline probability vector is not provided, one will be generated from an 
empirical model of gene segment prevalence. However, for best performance, this is not 
recommended. Estimates of true fold change is very sensitive to the distribution of 
features in your count dataset, and it is important that your baseline vector match
your overall dataset as accurately as possible. The best baseline vector is almost 
always the average feature prevalence across all repertoires in a dataset, although
manually generated baseline vectors may also work well.
</p>
<p>The units used for the RDI model should always match the units used to generate your
RDI values. For more details on units, refer to the details of <a href="#topic+calcRDI">calcRDI</a>.
</p>


<h3>Value</h3>

<p>A list containing three objects:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <em>fit</em>      </td><td style="text-align: left;"> an object of class <code>"smooth.spline"</code>, based on a spline 
                  model with the true difference (lfc or pct) as the independent (x)
                  and RDI as the dependent (y). Used for converting from true difference
                  to RDI.  </td>
</tr>
<tr>
 <td style="text-align: right;">
  <em>rev.fit</em>  </td><td style="text-align: left;"> an object of class <code>"smooth.spline"</code>. 
                  The opposite of fit. Used for converting from RDI to true difference.
                   </td>
</tr>
<tr>
 <td style="text-align: right;">
  <em>units</em>    </td><td style="text-align: left;"> one of <code>c("lfc","pct")</code>, representing the units of the true
                  difference values.
</td>
</tr>

</table>



<h3>See Also</h3>

<p><a href="#topic+rdiAxis">rdiAxis</a>, <a href="#topic+rdiLadder">rdiLadder</a>, <a href="#topic+plotRDIladder">plotRDIladder</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#create genes
genes = sample(letters, 10000, replace=TRUE)
#create sequence annotations
seqAnnot = data.frame(donor = sample(1:4, 10000, replace=TRUE))
#calculate RDI
d = rdi(genes, seqAnnot)

##create a "baseVect" with the same probability as our features
##since we sampled uniformly, the base vector has equal probability
baseVect = rep(1/length(letters),length(letters))

##generate an RDI model
m = rdiModel(attr(d, "nseq"), baseVects=baseVect)

##plot the spline model
plot(m$fit, xlab="log fold change",ylab="RDI",type='l')

##convert RDI to log fold change
mean = predict(m$rev.fit, d)$y
mean[mean&lt;0] = 0

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
