<!DOCTYPE html><html lang="en"><head><title>Help for package Rdiagnosislist</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Rdiagnosislist}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Rdiagnosislist-package'><p>Rdiagnosislist: A package for manipulating SNOMED CT diagnosis lists</p></a></li>
<li><a href='#acronyms'><p>Extract acronyms stated in the description of SNOMED CT concepts</p></a></li>
<li><a href='#addComposeLookupToCDB'><p>Add composition lookups to CDB</p></a></li>
<li><a href='#addInactiveConcepts'><p>Add inactive concepts to a SNOMEDcodelist or SNOMEDconcept vector</p></a></li>
<li><a href='#addWordnet'><p>Use WordNet to assist concept database creation</p></a></li>
<li><a href='#as.data.frame.SNOMEDconcept'><p>Return the SNOMED CT concept IDs for a set of terms</p></a></li>
<li><a href='#attrConcept'><p>Retrieve all attributes of a set of SNOMED CT concepts</p></a></li>
<li><a href='#batchDecompose'><p>Creates a set of lookups for SNOMED composition</p></a></li>
<li><a href='#c.SNOMEDconcept'><p>Concatenate vectors of SNOMED CT concepts</p></a></li>
<li><a href='#compose'><p>Select more specific SNOMED CT concepts based on attributes</p></a></li>
<li><a href='#createCDB'><p>Creates an environment containing CDB files</p></a></li>
<li><a href='#createSNOMEDindices'><p>Create indices for tables in a SNOMED environment</p></a></li>
<li><a href='#createTransitive'><p>Create a transitive closure table for is-a relationships for</p>
faster ancestor / descendant lookups</a></li>
<li><a href='#decompose'><p>Decomposition of meaning of a finding or disorder SNOMED CT concept</p></a></li>
<li><a href='#description'><p>Obtain descriptions for a set of SNOMED CT terms</p></a></li>
<li><a href='#downloadOrphanet'><p>Download Orphanet to assist with adding synonyms for rare diseases</p></a></li>
<li><a href='#downloadWordnet'><p>Download WordNet to assist concept database creation</p></a></li>
<li><a href='#exclude_irrelevant_findings'><p>Sample inclusion, exclusion and blacklist sets for a MiADE CDB</p></a></li>
<li><a href='#expandSNOMED'><p>Expand or contract a SNOMEDcodelist</p></a></li>
<li><a href='#export'><p>Export a SNOMEDcodelist</p></a></li>
<li><a href='#exportMiADECDB'><p>Exports CDB files for MedCAT / MiADE</p></a></li>
<li><a href='#exportSNOMEDenvir'><p>Export a SNOMED environment to a folder</p></a></li>
<li><a href='#getMaps'><p>Obtain Read 2, CTV3, ICD-10 and OPCS4 maps for SNOMED CT concepts</p></a></li>
<li><a href='#getRefset'><p>Retrieves a Refset from the REFSET table</p></a></li>
<li><a href='#getSNOMED'><p>Retrieves SNOMED CT dictionary from the global environment</p></a></li>
<li><a href='#hasAttributes'><p>Whether SNOMED CT concepts have particular attributes</p></a></li>
<li><a href='#HISTORY'><p>Sample history substitution table from SNOMED CT dictionary</p></a></li>
<li><a href='#htmlCodelistHierarchy'><p>Export a SNOMEDcodelist hierarchy to HTML</p></a></li>
<li><a href='#inactiveIncluded'><p>Check if inactive terms are included in SNOMED CT dictionary</p></a></li>
<li><a href='#is.SNOMEDcodelist'><p>Check if an object is a SNOMEDcodelist</p></a></li>
<li><a href='#is.SNOMEDconcept'><p>Check if an object is a SNOMEDconcept</p></a></li>
<li><a href='#loadREADMAPS'><p>Load mappings from Read to SNOMED CT into an R data.table</p></a></li>
<li><a href='#loadSNOMED'><p>Load SNOMED CT files from a folder(s) into R data.table objects</p></a></li>
<li><a href='#MANUAL_SYNONYMS'><p>Sample manual synonym table to assist in creation of concept database</p></a></li>
<li><a href='#parents'><p>Ancestors and descendants of SNOMED CT concepts</p></a></li>
<li><a href='#print.SNOMEDcodelist'><p>Display a SNOMEDcodelist on screen</p></a></li>
<li><a href='#print.SNOMEDconcept'><p>Display a SNOMEDconcept object with descriptions</p></a></li>
<li><a href='#print.SNOMEDfindings'><p>Print method for output of 'decompose' function</p></a></li>
<li><a href='#QUERY'><p>Sample query table from SNOMED CT dictionary</p></a></li>
<li><a href='#READMAPS'><p>Sample mappings from Read to SNOMED CT</p></a></li>
<li><a href='#relatedConcepts'><p>Obtain related concepts for a set of SNOMED CT concepts</p></a></li>
<li><a href='#sampleSNOMED'><p>Sample SNOMED CT dictionary</p></a></li>
<li><a href='#semanticType'><p>Retrieves semantic types using the text 'tag' in the description</p></a></li>
<li><a href='#simplify'><p>Retrieves closest single ancestor within a given set of SNOMED CT</p>
concepts</a></li>
<li><a href='#SNOMED_CONCEPT'><p>Sample concept table from SNOMED CT dictionary</p></a></li>
<li><a href='#SNOMED_DESCRIPTION'><p>Sample description table from SNOMED CT dictionary</p></a></li>
<li><a href='#SNOMED_EXTENDEDMAP'><p>Sample extended map table from SNOMED CT dictionary</p></a></li>
<li><a href='#SNOMED_REFSET'><p>Sample refset table from SNOMED CT dictionary</p></a></li>
<li><a href='#SNOMED_RELATIONSHIP'><p>Sample relationship tables from SNOMED CT dictionary</p></a></li>
<li><a href='#SNOMED_SIMPLEMAP'><p>Sample SIMPLE map table from SNOMED CT dictionary</p></a></li>
<li><a href='#SNOMEDcodelist'><p>Convert a data.frame to a SNOMEDcodelist object</p></a></li>
<li><a href='#union.SNOMEDconcept'><p>Set operations for SNOMEDconcept vectors</p></a></li>
<li><a href='#unique.SNOMEDconcept'><p>Unique vector of SNOMED CT concepts</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Manipulate SNOMED CT Diagnosis Lists</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions and methods for manipulating 'SNOMED CT'
    concepts. The package contains functions for loading the 'SNOMED CT'
    release into a convenient R environment, selecting 'SNOMED CT'
    concepts using regular expressions, and navigating the 'SNOMED CT'
    ontology. It provides the 'SNOMEDconcept' S3 class for a vector of
    'SNOMED CT' concepts (stored as 64-bit integers) and the
    'SNOMEDcodelist' S3 class for a table of concepts IDs with
    descriptions. The package can be used to construct sets of
    'SNOMED CT' concepts for research (&lt;<a href="https://doi.org/10.1093%2Fjamia%2Focac158">doi:10.1093/jamia/ocac158</a>&gt;).
    For more information about 'SNOMED CT' visit
    <a href="https://www.snomed.org/">https://www.snomed.org/</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.12.8), bit64, methods</td>
</tr>
<tr>
<td>Depends:</td>
<td>readxl, R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Anoop D. Shah <a href="https://orcid.org/0000-0002-8907-5724"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anoop D. Shah &lt;anoop@doctors.org.uk&gt;</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-28 17:06:42 UTC; anoop</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-29 08:30:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='Rdiagnosislist-package'>Rdiagnosislist: A package for manipulating SNOMED CT diagnosis lists</h2><span id='topic+Rdiagnosislist'></span><span id='topic+Rdiagnosislist-package'></span>

<h3>Description</h3>

<p>The Rdiagnosislist package makes it easy to load a SNOMED dictionary
into R and use the hierarchies to search for concepts and navigate
relations between concepts.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Anoop D. Shah <a href="mailto:anoop@doctors.org.uk">anoop@doctors.org.uk</a> (<a href="https://orcid.org/0000-0002-8907-5724">ORCID</a>)
</p>

<hr>
<h2 id='acronyms'>Extract acronyms stated in the description of SNOMED CT concepts</h2><span id='topic+acronyms'></span>

<h3>Description</h3>

<p>Returns acronyms, if any, expressed within SNOMED CT descriptions
in the form 'ABCD - Another Bland Cardiovascular Disease'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acronyms(conceptIds, SNOMED = getSNOMED())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="acronyms_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector</p>
</td></tr>
<tr><td><code id="acronyms_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing SNOMED dictionary. Defaults
to an object named 'SNOMED' in the global environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table with the following columns: id, conceptId,
type = 'Acronym', term = acronym
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()
acronyms('Heart failure')

# Get all synonyms and acronyms
rva &lt;- SNOMEDconcept('Right ventricular abnormality')
rbind(description(rva, include_synonyms = TRUE), acronyms(rva))
</code></pre>

<hr>
<h2 id='addComposeLookupToCDB'>Add composition lookups to CDB</h2><span id='topic+addComposeLookupToCDB'></span>

<h3>Description</h3>

<p>Creates a composition lookup table for a set of SNOMED CT concepts
based on output of &lsquo;decompose&rsquo;, and adds it to the CDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addComposeLookupToCDB(
  decompositions,
  CDB,
  maxcol = 10,
  SNOMED = getSNOMED(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addComposeLookupToCDB_+3A_decompositions">decompositions</code></td>
<td>
<p>vector of filenames of decompose output (read
by fread) or data.frame containing outputs of decompose function</p>
</td></tr>
<tr><td><code id="addComposeLookupToCDB_+3A_cdb">CDB</code></td>
<td>
<p>concept database environment, containing a table called
FINDINGS</p>
</td></tr>
<tr><td><code id="addComposeLookupToCDB_+3A_maxcol">maxcol</code></td>
<td>
<p>maximum number of attributes columns. If NULL it is
determined from the data. It might be helpful to specify it so that
downstream databases and programs know exactly how many columns to
expect. We suggest setting it to 10 which should handle all 
possible SNOMED CT concept decompositions.</p>
</td></tr>
<tr><td><code id="addComposeLookupToCDB_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED CT dictionary</p>
</td></tr>
<tr><td><code id="addComposeLookupToCDB_+3A_...">...</code></td>
<td>
<p>other arguments to pass to fread</p>
</td></tr>
</table>


<h3>Value</h3>

<p>CDB environment with an additional data.table &lsquo;COMPOSELOOKUP&rsquo;
with columns rootId, attr_1 ... attr_X (up to
maxcol), with, due_to, without, origId (all with data type
integer64 and class 'SNOMEDconcept')
</p>


<h3>See Also</h3>

<p>decompose, compose, batchDecompose
</p>
<p>Other CDB functions: 
<code><a href="#topic+addWordnet">addWordnet</a>()</code>,
<code><a href="#topic+batchDecompose">batchDecompose</a>()</code>,
<code><a href="#topic+createCDB">createCDB</a>()</code>,
<code><a href="#topic+downloadOrphanet">downloadOrphanet</a>()</code>,
<code><a href="#topic+downloadWordnet">downloadWordnet</a>()</code>
</p>

<hr>
<h2 id='addInactiveConcepts'>Add inactive concepts to a SNOMEDcodelist or SNOMEDconcept vector</h2><span id='topic+addInactiveConcepts'></span>

<h3>Description</h3>

<p>Adds SNOMED concepts linked by the QUERY or HISTORY tables
that are mapped to or descendants of concepts in a SNOMEDcodelist
or a SNOMEDconcept vector. If a SNOMEDcodelist, it is automatically
converted to the 'simple' format (all items enumerated).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addInactiveConcepts(x, provenance = 0:3, SNOMED = getSNOMED())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addInactiveConcepts_+3A_x">x</code></td>
<td>
<p>SNOMEDcodelist or SNOMEDconcept object</p>
</td></tr>
<tr><td><code id="addInactiveConcepts_+3A_provenance">provenance</code></td>
<td>
<p>vector of provenance values to use</p>
</td></tr>
<tr><td><code id="addInactiveConcepts_+3A_snomed">SNOMED</code></td>
<td>
<p>SNOMED environment containing HISTORY and QUERY tables</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is recommended to use this function to convert a reference
into a codelist for running a query against an electronic health
record database which might contain historic SNOMED CT concepts.
</p>


<h3>Value</h3>

<p>SNOMEDcodelist or SNOMEDconcept with linked inactive concepts included
</p>


<h3>See Also</h3>

<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>

<hr>
<h2 id='addWordnet'>Use WordNet to assist concept database creation</h2><span id='topic+addWordnet'></span>

<h3>Description</h3>

<p>Adds terms from a WordNet thesaurus to a concept database, matching
on term. It is recommended to restrict the wordnet categories to
ensure that words with multiple meanings are not linked to the wrong
synonym. This function also corrects some known errors in WordNet to
avoid them being passed on to the CDB; currently this applies to 
'allergy = allergic reaction', 'cuneiform bone = triquetral'
and 'trauma' = 'injury', but more corrections can be done if needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addWordnet(
  CDB_TABLE,
  wn_categories,
  WN,
  CHECK_TABLE = NULL,
  errors_to_remove = list(c("allergy", "allergic reaction"), c("allergic",
    "allergic reaction"), c("trauma", "injury"), c("traumatic", "injury"), c("skinny",
    "skin"), c("cuneiform bone", "triquetral bone"), c("upset", "disorder"),
    c("disorderliness", "disorder")),
  noisy = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addWordnet_+3A_cdb_table">CDB_TABLE</code></td>
<td>
<p>data.frame or data.table with columns
conceptId (integer64) and term (character, with space before
and after) containing existing descriptions in the CDB</p>
</td></tr>
<tr><td><code id="addWordnet_+3A_wn_categories">wn_categories</code></td>
<td>
<p>WordNet categories to use</p>
</td></tr>
<tr><td><code id="addWordnet_+3A_wn">WN</code></td>
<td>
<p>WordNet data.table as returned by downloadWordnet</p>
</td></tr>
<tr><td><code id="addWordnet_+3A_check_table">CHECK_TABLE</code></td>
<td>
<p>other table in the same format as CDB_TABLE
to check for WordNet synonyms that link to another unrelated
concept, where this synonym will be excluded because of the risk
of errors</p>
</td></tr>
<tr><td><code id="addWordnet_+3A_errors_to_remove">errors_to_remove</code></td>
<td>
<p>list of character vectors of length two
containing synonym pairs to be removed. The first entry of the
pair will be removed from the WordNet file before it is used for
adding to CDB</p>
</td></tr>
<tr><td><code id="addWordnet_+3A_noisy">noisy</code></td>
<td>
<p>whether to output test comments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>CDB_TABLE with extra rows for Wordnet synonyms
</p>


<h3>References</h3>

<p><a href="https://wordnet.princeton.edu/">https://wordnet.princeton.edu/</a>
</p>


<h3>See Also</h3>

<p>downloadWordnet
</p>
<p>Other CDB functions: 
<code><a href="#topic+addComposeLookupToCDB">addComposeLookupToCDB</a>()</code>,
<code><a href="#topic+batchDecompose">batchDecompose</a>()</code>,
<code><a href="#topic+createCDB">createCDB</a>()</code>,
<code><a href="#topic+downloadOrphanet">downloadOrphanet</a>()</code>,
<code><a href="#topic+downloadWordnet">downloadWordnet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>WORDNET &lt;- data.table::data.table(cat = c('noun.body', 'noun.state'),
  wordnetId = bit64::as.integer64('1', '2'),
  synonyms = list(c('heart', 'pump', 'ticker'),
  c('infection', 'infectious')),
  parents = list('cardiovascular system',
  'pathologic process'), 
  adj = list('cardiac', 'infectious'))
# Add Wordnet synonyms to a concept database table
SNOMED &lt;- sampleSNOMED()
CDB_TABLE &lt;- description(c('Heart', 'Infection'),
  include_synonyms = TRUE)[type == 'Synonym',
  list(conceptId, term = paste0(' ', tolower(term), ' '))]
addWordnet(CDB_TABLE, 'noun.state', WORDNET)
</code></pre>

<hr>
<h2 id='as.data.frame.SNOMEDconcept'>Return the SNOMED CT concept IDs for a set of terms</h2><span id='topic+as.data.frame.SNOMEDconcept'></span><span id='topic+as.integer64.SNOMEDconcept'></span><span id='topic+SNOMEDconcept'></span><span id='topic+as.SNOMEDconcept'></span>

<h3>Description</h3>

<p>Carries out an exact or regular expression match to
return the concept ID for a set of search terms, or
converts a character, integer or integer64 vector to a
SNOMEDconcept object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDconcept'
as.data.frame(x, ...)

## S3 method for class 'SNOMEDconcept'
as.integer64(x, ...)

SNOMEDconcept(
  x,
  active_only = TRUE,
  exact_match = TRUE,
  unique = TRUE,
  SNOMED = getSNOMED()
)

as.SNOMEDconcept(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.data.frame.SNOMEDconcept_+3A_x">x</code></td>
<td>
<p>character vector of terms to match, or
character vector containing SNOMED CT concept IDs, or
64-bit integer vector containing SNOMED CT concept IDs</p>
</td></tr>
<tr><td><code id="as.data.frame.SNOMEDconcept_+3A_...">...</code></td>
<td>
<p>additional arguments to send to grepl if using
regular expression matching</p>
</td></tr>
<tr><td><code id="as.data.frame.SNOMEDconcept_+3A_active_only">active_only</code></td>
<td>
<p>whether or not to include inactive concepts</p>
</td></tr>
<tr><td><code id="as.data.frame.SNOMEDconcept_+3A_exact_match">exact_match</code></td>
<td>
<p>if TRUE, only an exact (case sensitive)
match is performed. If FALSE, a regular expression match
is performed.</p>
</td></tr>
<tr><td><code id="as.data.frame.SNOMEDconcept_+3A_unique">unique</code></td>
<td>
<p>whether to include no more than one instance of each
SNOMED CT concept</p>
</td></tr>
<tr><td><code id="as.data.frame.SNOMEDconcept_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing SNOMED dictionary. Defaults
to an object named 'SNOMED' in the global environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a SNOMEDconcept object (vector of 64-bit integers) containing
unique SNOMED CT concept IDs
</p>


<h3>See Also</h3>

<p>Other SNOMEDconcept functions: 
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMEDconcept('Heart failure', SNOMED = sampleSNOMED()) -&gt; hf
is.SNOMEDconcept(hf)
SNOMEDconcept('900000000000003001')
as.SNOMEDconcept('900000000000003001')
</code></pre>

<hr>
<h2 id='attrConcept'>Retrieve all attributes of a set of SNOMED CT concepts</h2><span id='topic+attrConcept'></span>

<h3>Description</h3>

<p>Returns the portion of the SNOMED CT relationship tables containing
relationships where the given concepts are either the source or the 
destination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>attrConcept(
  conceptIds,
  SNOMED = getSNOMED(),
  tables = c("RELATIONSHIP", "STATEDRELATIONSHIP"),
  active_only = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="attrConcept_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs</p>
</td></tr>
<tr><td><code id="attrConcept_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="attrConcept_+3A_tables">tables</code></td>
<td>
<p>character vector of relationship tables to use</p>
</td></tr>
<tr><td><code id="attrConcept_+3A_active_only">active_only</code></td>
<td>
<p>whether to return only active attributes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table with the following columns: 
sourceId (concept ID of source for relationship),
destinationId (concept ID of source for relationship),
typeId (concept ID of relationship type),
typeName (description of relationship type)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

attrConcept(as.SNOMEDconcept('Heart failure'))
</code></pre>

<hr>
<h2 id='batchDecompose'>Creates a set of lookups for SNOMED composition</h2><span id='topic+batchDecompose'></span>

<h3>Description</h3>

<p>Creates composition lookup table for a set of SNOMED CT concepts
and exports the results to a file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>batchDecompose(conceptIds, CDB, output_filename, SNOMED = getSNOMED(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="batchDecompose_+3A_conceptids">conceptIds</code></td>
<td>
<p>SNOMED CT concept IDs for creating decompositions</p>
</td></tr>
<tr><td><code id="batchDecompose_+3A_cdb">CDB</code></td>
<td>
<p>concept database environment, containing a table called
FINDINGS</p>
</td></tr>
<tr><td><code id="batchDecompose_+3A_output_filename">output_filename</code></td>
<td>
<p>filename of output file</p>
</td></tr>
<tr><td><code id="batchDecompose_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="batchDecompose_+3A_...">...</code></td>
<td>
<p>other arguments to pass through to decompose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if successful
</p>


<h3>See Also</h3>

<p>decompose, compose, addComposeLookupToCDB
</p>
<p>Other CDB functions: 
<code><a href="#topic+addComposeLookupToCDB">addComposeLookupToCDB</a>()</code>,
<code><a href="#topic+addWordnet">addWordnet</a>()</code>,
<code><a href="#topic+createCDB">createCDB</a>()</code>,
<code><a href="#topic+downloadOrphanet">downloadOrphanet</a>()</code>,
<code><a href="#topic+downloadWordnet">downloadWordnet</a>()</code>
</p>

<hr>
<h2 id='c.SNOMEDconcept'>Concatenate vectors of SNOMED CT concepts</h2><span id='topic+c.SNOMEDconcept'></span>

<h3>Description</h3>

<p>SNOMEDconcept is an S3 class for vectors of SNOMED concept IDs
as 64-bit integers. This function concatenates two or more
SNOMEDconcept vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDconcept'
c(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c.SNOMEDconcept_+3A_...">...</code></td>
<td>
<p>SNOMEDconcept vectors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>concatenation of vectors
</p>


<h3>See Also</h3>

<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hf &lt;- SNOMEDconcept('Heart failure', SNOMED = sampleSNOMED())
hf2 &lt;- c(hf, hf)
</code></pre>

<hr>
<h2 id='compose'>Select more specific SNOMED CT concepts based on attributes</h2><span id='topic+compose'></span>

<h3>Description</h3>

<p>Finds the most specific SNOMED CT concepts that matches the
combination of a root concept and attributes. Based on a
composeLookup table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose(
  conceptId,
  CDB,
  attributes_conceptIds = bit64::integer64(0),
  due_to_conceptIds = bit64::integer64(0),
  without_conceptIds = bit64::integer64(0),
  with_conceptIds = bit64::integer64(0),
  SNOMED = getSNOMED()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_+3A_conceptid">conceptId</code></td>
<td>
<p>SNOMED CT concept to refine</p>
</td></tr>
<tr><td><code id="compose_+3A_cdb">CDB</code></td>
<td>
<p>SNOMED CT concept database, as created by createCDB.
An environment containing the following data tables: FINDINGS,
QUAL, CAUSES, BODY, FINDINGS, OTHERSUB, OVERLAP, TRANSITIVE.
Must also contain a COMPOSELOOKUP table created by
addComposeLookupToCDB</p>
</td></tr>
<tr><td><code id="compose_+3A_attributes_conceptids">attributes_conceptIds</code></td>
<td>
<p>SNOMED concept Ids of attributes of concept
e.g. laterality, severity, acuteness</p>
</td></tr>
<tr><td><code id="compose_+3A_due_to_conceptids">due_to_conceptIds</code></td>
<td>
<p>SNOMED concept Ids of cause</p>
</td></tr>
<tr><td><code id="compose_+3A_without_conceptids">without_conceptIds</code></td>
<td>
<p>SNOMED concept Ids of conditions stated to be
absent</p>
</td></tr>
<tr><td><code id="compose_+3A_with_conceptids">with_conceptIds</code></td>
<td>
<p>SNOMED concept Ids of conditions also present</p>
</td></tr>
<tr><td><code id="compose_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing SNOMED CT tables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a refined SNOMED concept Id
</p>


<h3>See Also</h3>

<p>decompose, batchDecompose, addComposeLookupToCDB
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
SNOMED &lt;- sampleSNOMED()
miniCDB &lt;- createCDB(SNOMED)

D &lt;- decompose('Cor pulmonale', CDB = miniCDB)
print(D)

# --------------------------------------------------------
# 83291003 | Cor pulmonale (disorder)
# --------------------------------------------------------
# Root : 367363000 | Right ventricular failure (disorder)
# - Due to : 19829001 | Disorder of lung (disorder)

# Compile decompositions into a lookup table
miniCDB &lt;- addComposeLookupToCDB(D, CDB = miniCDB))

compose(as.SNOMEDconcept('Right heart failure'),
  due_to_conceptIds = as.SNOMEDconcept('Disorder of lung'),
  CDB = miniCDB, composeLookup = CL)
# [1] "83291003 | Cor pulmonale (disorder)"

## End(Not run)
</code></pre>

<hr>
<h2 id='createCDB'>Creates an environment containing CDB files</h2><span id='topic+createCDB'></span>

<h3>Description</h3>

<p>Extracts SNOMED CT concepts from appropriate places in the 
hierarchy to create a set of CDB files in an environment.
Uses WordNet and manual synonyms if available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createCDB(
  SNOMED = getSNOMED(),
  TRANSITIVE = NULL,
  WN = NULL,
  MANUAL_SYNONYMS = NULL,
  noisy = TRUE,
  stopwords = c("the", "of", "by", "with", "to", "into", "and", "or", "at", "as",
    "and/or", "in")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createCDB_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="createCDB_+3A_transitive">TRANSITIVE</code></td>
<td>
<p>transitive closure table, generated by
createTransitive. It is regenerated if not provided.</p>
</td></tr>
<tr><td><code id="createCDB_+3A_wn">WN</code></td>
<td>
<p>WordNet data.table as returned by downloadWordnet
containing WordNet data from appropriate
categories, in the format: cat (character), wordnetId (integer64),
synonyms (list), parents (list), adj (list)</p>
</td></tr>
<tr><td><code id="createCDB_+3A_manual_synonyms">MANUAL_SYNONYMS</code></td>
<td>
<p>data.table with columns term1 and term2,
containing additional exact synonyms or abbreviations</p>
</td></tr>
<tr><td><code id="createCDB_+3A_noisy">noisy</code></td>
<td>
<p>whether to output status messages</p>
</td></tr>
<tr><td><code id="createCDB_+3A_stopwords">stopwords</code></td>
<td>
<p>vector of stopwords</p>
</td></tr>
</table>


<h3>Value</h3>

<p>environment containing the following data tables: FINDINGS,
QUAL, CAUSES, BODY, OTHERCAUSE, OTHERSEARCH, OVERLAP, TRANSITIVE
</p>


<h3>See Also</h3>

<p>exportMiADECDB, MANUAL_SYNONYMS
</p>
<p>Other CDB functions: 
<code><a href="#topic+addComposeLookupToCDB">addComposeLookupToCDB</a>()</code>,
<code><a href="#topic+addWordnet">addWordnet</a>()</code>,
<code><a href="#topic+batchDecompose">batchDecompose</a>()</code>,
<code><a href="#topic+downloadOrphanet">downloadOrphanet</a>()</code>,
<code><a href="#topic+downloadWordnet">downloadWordnet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not run
# data(MANUAL_SYNONYMS)
# WN &lt;- downloadWordnet()
# MANUAL_SYNONYMS &lt;- rbind(MANUAL_SYNONYMS, downloadOrphanet())
# CDBNEW &lt;- createCDB(WN = WN, MANUAL_SYNONYMS = MANUAL_SYNONYMS)
</code></pre>

<hr>
<h2 id='createSNOMEDindices'>Create indices for tables in a SNOMED environment</h2><span id='topic+createSNOMEDindices'></span>

<h3>Description</h3>

<p>Creates relevant indices for fast searching of SNOMED CT tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createSNOMEDindices(SNOMED)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createSNOMEDindices_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing data.table objects: CONCEPT,
DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The environment with indices added to each table for
fast searching
</p>


<h3>See Also</h3>

<p>CONCEPT, DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP, 
REFSET, SIMPLEMAP, EXTENDEDMAP, loadSNOMED, sampleSNOMED
</p>

<hr>
<h2 id='createTransitive'>Create a transitive closure table for is-a relationships for
faster ancestor / descendant lookups</h2><span id='topic+createTransitive'></span>

<h3>Description</h3>

<p>Returns a data.table containing ancestor / descendant relationships
which can be used in ancestors and descendants functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createTransitive(
  conceptIds,
  SNOMED = getSNOMED(),
  tables = c("RELATIONSHIP", "STATEDRELATIONSHIP")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createTransitive_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs
for the subset of concepts to include in the transitive table.</p>
</td></tr>
<tr><td><code id="createTransitive_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="createTransitive_+3A_tables">tables</code></td>
<td>
<p>vector of names of relationship table(s) to use;
by default use both RELATIONSHIP and STATEDRELATIONSHIP</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>[ancestors()] and [descendants()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

TRANSITIVE &lt;- createTransitive('Heart failure')
</code></pre>

<hr>
<h2 id='decompose'>Decomposition of meaning of a finding or disorder SNOMED CT concept</h2><span id='topic+decompose'></span>

<h3>Description</h3>

<p>Decomposes a SNOMED CT term into separate components according
to the SNOMED CT information model and text parsing. Each term
may have a number of possible decompositions. Requires a CDB
environment created by createCDB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decompose(
  conceptIds,
  diagnosis_text = NULL,
  CDB,
  SNOMED = getSNOMED(),
  noisy = FALSE,
  omit_unmatched = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="decompose_+3A_conceptids">conceptIds</code></td>
<td>
<p>vector of SNOMED CT concepts to decompose</p>
</td></tr>
<tr><td><code id="decompose_+3A_diagnosis_text">diagnosis_text</code></td>
<td>
<p>vector of SNOMED CT terms (or in theory any
text that has the same meaning as the SNOMED CT concept). If NULL,
decompositions are created for all SNOMED CT synonyms of the
concepts.</p>
</td></tr>
<tr><td><code id="decompose_+3A_cdb">CDB</code></td>
<td>
<p>an environment containing CDB files, as created by
createCDB</p>
</td></tr>
<tr><td><code id="decompose_+3A_snomed">SNOMED</code></td>
<td>
<p>an environment containing the SNOMED CT dictionary</p>
</td></tr>
<tr><td><code id="decompose_+3A_noisy">noisy</code></td>
<td>
<p>whether to output messages (for debugging)</p>
</td></tr>
<tr><td><code id="decompose_+3A_omit_unmatched">omit_unmatched</code></td>
<td>
<p>whether to omit rows in which some attributes
could not be matched to SNOMED CT concepts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a SNOMEDfinding objects, which is a data.table with columns
rootId (integer64), with (integer64), due_to (integer64),
after (integer64), without (integer64), body_site (integer64),
severity (integer64), stage (integer64), laterality (integer64),
roottext (character), partId (integer64),
other_conceptId (character, a string of SNOMED CT concept IDs
separated by spaces), origId (integer64)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
miniCDB &lt;- createCDB(SNOMED = sampleSNOMED())
D &lt;- decompose(as.SNOMEDconcept('Cor pulmonale',
  SNOMED = sampleSNOMED()), CDB = miniCDB, SNOMED = sampleSNOMED())

## End(Not run)

# -------------------------------------------------------
# 83291003 | Cor pulmonale (disorder)
# -------------------------------------------------------
# Root : 367363000 | Right ventricular failure (disorder)
# - Due to : 19829001 | Disorder of lung (disorder)
</code></pre>

<hr>
<h2 id='description'>Obtain descriptions for a set of SNOMED CT terms</h2><span id='topic+description'></span>

<h3>Description</h3>

<p>Returns the descriptions matching a set of concept IDs from
a SNOMED dictionary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>description(
  conceptIds,
  include_synonyms = FALSE,
  active_only = TRUE,
  SNOMED = getSNOMED()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="description_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector</p>
</td></tr>
<tr><td><code id="description_+3A_include_synonyms">include_synonyms</code></td>
<td>
<p>whether to return all synonyms, or just the
Fully Specified Name, ensuring just one row per concept (default)</p>
</td></tr>
<tr><td><code id="description_+3A_active_only">active_only</code></td>
<td>
<p>whether to include only active descriptions</p>
</td></tr>
<tr><td><code id="description_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing SNOMED dictionary. Defaults
to an object named 'SNOMED' in the global environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table with the following columns: id, conceptId, type
(only if include_synonyms = TRUE), term,
active (only if active_only = FALSE)
</p>


<h3>See Also</h3>

<p>htmlCodelistHierarchy
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hf &lt;- SNOMEDconcept('Heart failure', SNOMED = sampleSNOMED())
description(hf, include_synonyms = FALSE, SNOMED = sampleSNOMED())
</code></pre>

<hr>
<h2 id='downloadOrphanet'>Download Orphanet to assist with adding synonyms for rare diseases</h2><span id='topic+downloadOrphanet'></span>

<h3>Description</h3>

<p>Downloads the Orphanet nomenclature pack and converts it into a
format to be appended to MANUAL_SYNONYMS and used in concept
database creation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downloadOrphanet(
  orphanet_url =
    "https://www.orphadata.com/data/nomenclature/packs/Orphanet_Nomenclature_Pack_EN.zip",
  masterfile_name = NULL,
  SNOMED = getSNOMED()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downloadOrphanet_+3A_orphanet_url">orphanet_url</code></td>
<td>
<p>URL or filepath to Orphanet zip file</p>
</td></tr>
<tr><td><code id="downloadOrphanet_+3A_masterfile_name">masterfile_name</code></td>
<td>
<p>name of xls file containing Orphanet synonyms.
If omitted, it is assumed to be the only file with 'MasterFile'
in its name, and the program will search for it in the Orphanet
zip file.</p>
</td></tr>
<tr><td><code id="downloadOrphanet_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing SNOMED CT dictionary</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table containing Orphanet synonyms in the format
</p>


<h3>References</h3>

<p><a href="https://www.orphadata.com/pack-nomenclature/">https://www.orphadata.com/pack-nomenclature/</a>
</p>


<h3>See Also</h3>

<p>downloadWordnet, MANUAL_SYNONYMS
</p>
<p>Other CDB functions: 
<code><a href="#topic+addComposeLookupToCDB">addComposeLookupToCDB</a>()</code>,
<code><a href="#topic+addWordnet">addWordnet</a>()</code>,
<code><a href="#topic+batchDecompose">batchDecompose</a>()</code>,
<code><a href="#topic+createCDB">createCDB</a>()</code>,
<code><a href="#topic+downloadWordnet">downloadWordnet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not run
# ORPHANET &lt;- downloadOrphanet()
</code></pre>

<hr>
<h2 id='downloadWordnet'>Download WordNet to assist concept database creation</h2><span id='topic+downloadWordnet'></span>

<h3>Description</h3>

<p>Downloads the WordNet thesaurus and converts it into a format to be
used by addWordnet to add extra synonyms to a concept database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downloadWordnet(
  wordnet_url = "https://wordnetcode.princeton.edu/wn3.1.dict.tar.gz",
  wn_categories = c("noun.body", "noun.state", "noun.process", "noun.animal",
    "noun.plant", "noun.phenomenon")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downloadWordnet_+3A_wordnet_url">wordnet_url</code></td>
<td>
<p>URL or filepath to WordNet tar.gz file</p>
</td></tr>
<tr><td><code id="downloadWordnet_+3A_wn_categories">wn_categories</code></td>
<td>
<p>WordNet categories from which to extract data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table containing WordNet data from appropriate
categories, in the format: cat (character), wordnetId (integer64),
synonyms (list), parents (list), adj (list)
</p>


<h3>References</h3>

<p><a href="https://wordnet.princeton.edu/">https://wordnet.princeton.edu/</a>
</p>


<h3>See Also</h3>

<p>addWordnet
</p>
<p>Other CDB functions: 
<code><a href="#topic+addComposeLookupToCDB">addComposeLookupToCDB</a>()</code>,
<code><a href="#topic+addWordnet">addWordnet</a>()</code>,
<code><a href="#topic+batchDecompose">batchDecompose</a>()</code>,
<code><a href="#topic+createCDB">createCDB</a>()</code>,
<code><a href="#topic+downloadOrphanet">downloadOrphanet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not run
# WORDNET &lt;- downloadWordnet()
</code></pre>

<hr>
<h2 id='exclude_irrelevant_findings'>Sample inclusion, exclusion and blacklist sets for a MiADE CDB</h2><span id='topic+exclude_irrelevant_findings'></span><span id='topic+blacklist_vague_findings'></span><span id='topic+blacklist_almost_all_except_diseases'></span>

<h3>Description</h3>

<p>Returns a set of SNOMED concepts (as a SNOMEDconcept vector)
which can be used to exclude
findings in the MedCAT named entity recognition step, or blacklist
(filter out) findings from the final output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exclude_irrelevant_findings(SNOMED = getSNOMED())

blacklist_vague_findings(SNOMED = getSNOMED())

blacklist_almost_all_except_diseases(SNOMED = getSNOMED())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exclude_irrelevant_findings_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
</table>


<h3>Details</h3>


<dl>
<dt>exclude_irrelevant_findings</dt><dd><p>social history 
(except housing problems and care needs),
administrative statuses (except registered disabled) and
for concept detection</p>
</dd>
<dt>blacklist_vague_findings</dt><dd><p>vague findings and disorders,
intended to be used in the blacklist</p>
</dd>
<dt>blacklist_almost_all_except_diseases</dt><dd><p>almost all findings
and vague disorders, intended to be used in the blacklist</p>
</dd>
</dl>



<h3>Value</h3>

<p>SNOMEDconcept vector containing findings to exclude
</p>


<h3>See Also</h3>

<p>exportMiADECDB, createCDB
</p>
<p>Other MiADE functions: 
<code><a href="#topic+exportMiADECDB">exportMiADECDB</a>()</code>
</p>
<p>Other MiADE functions: 
<code><a href="#topic+exportMiADECDB">exportMiADECDB</a>()</code>
</p>
<p>Other MiADE functions: 
<code><a href="#topic+exportMiADECDB">exportMiADECDB</a>()</code>
</p>

<hr>
<h2 id='expandSNOMED'>Expand or contract a SNOMEDcodelist</h2><span id='topic+expandSNOMED'></span><span id='topic+contractSNOMED'></span><span id='topic+showCodelistHierarchy'></span>

<h3>Description</h3>

<p>SNOMEDcodelist is an S3 class for sets of SNOMED concepts.
In the 'contracted' form, it may contain only parents and not
child terms (to create a more succinct list). The 'Expanded'
form contains all concepts. The output of 'showCodelistHierarchy'
includes all hierarchies contained within the codelist in a 
format suitable for display.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expandSNOMED(x, SNOMED = getSNOMED(), ...)

contractSNOMED(x, SNOMED = getSNOMED(), ...)

showCodelistHierarchy(
  x,
  SNOMED = getSNOMED(),
  max_excluded_descendants = 200,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expandSNOMED_+3A_x">x</code></td>
<td>
<p>SNOMEDcodelist to expand or contract. If x is not a
SNOMEDcodelist, it is coerced to one by as.SNOMEDcodelist</p>
</td></tr>
<tr><td><code id="expandSNOMED_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="expandSNOMED_+3A_...">...</code></td>
<td>
<p>other arguments to pass to as.SNOMEDcodelist</p>
</td></tr>
<tr><td><code id="expandSNOMED_+3A_max_excluded_descendants">max_excluded_descendants</code></td>
<td>
<p>(integer) whether to show excluded
descendants as long as they do not exceed this number (a
limit is suggested to avoid the program crashing if there
are too many descendants). If this number is exceeded, the
program will initially try to include children only, and if
there are still too many, it will ignore all descendants.
An 'included' column is added to the codelist
showing which terms are included. This can make it easy to see
if a codelist is consistent with the SNOMED CT ontology.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'SNOMEDcodelist' with attribute
Expanded = TRUE
</p>


<h3>See Also</h3>

<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>
<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>
<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

my_concepts &lt;- SNOMEDconcept('Heart failure')
my_codelist &lt;- SNOMEDcodelist(data.frame(conceptId = my_concepts,
  include_desc = TRUE))
expanded_codelist &lt;- expandSNOMED(my_codelist)
contractSNOMED(expanded_codelist)
</code></pre>

<hr>
<h2 id='export'>Export a SNOMEDcodelist</h2><span id='topic+export'></span><span id='topic+export.SNOMEDcodelist'></span>

<h3>Description</h3>

<p>Writes a SNOMEDcodelist to file. If the filename is NULL,
a filename is created from the 'codelist_name' attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export(x, ...)

## S3 method for class 'SNOMEDcodelist'
export(x, filename = NULL, metadata = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_+3A_x">x</code></td>
<td>
<p>SNOMEDcodelist object to export to file</p>
</td></tr>
<tr><td><code id="export_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="export_+3A_filename">filename</code></td>
<td>
<p>character vector of length 1 for the file
to write to. If NULL, a filename is generated from the
codelist filename.</p>
</td></tr>
<tr><td><code id="export_+3A_metadata">metadata</code></td>
<td>
<p>whether metadata is exported (default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisibly returns the exported codelist
</p>


<h3>See Also</h3>

<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>
<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>

<hr>
<h2 id='exportMiADECDB'>Exports CDB files for MedCAT / MiADE</h2><span id='topic+exportMiADECDB'></span>

<h3>Description</h3>

<p>Produces a set of files for the findings / problems algorithm of
MedCAT and MiADE. Uses the CDB environment created using createCDB
which can incorporate additional manual synonyms or synonyms from
WordNet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportMiADECDB(
  CDB,
  export_folderpath,
  lang_refset_files = NULL,
  exclude = NULL,
  include = NULL,
  exclude_historic = descendants("Disorder", SNOMED = getSNOMED()),
  blacklist = NULL,
  SNOMED = getSNOMED()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exportMiADECDB_+3A_cdb">CDB</code></td>
<td>
<p>concept database environment created by createCDB</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_export_folderpath">export_folderpath</code></td>
<td>
<p>folder path to export to</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_lang_refset_files">lang_refset_files</code></td>
<td>
<p>character vector of file paths to
SNOMED CT language refset files, in order to identify the
preferred term for each concept. If NULL, the Fully Specified Name
minus the semantic type suffix is used as the preferred term
(e.g. if the Fully Specified Name is &lsquo;Cancer (disorder)&rsquo;, the
default preferred term is &lsquo;Cancer&rsquo;.</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_exclude">exclude</code></td>
<td>
<p>a SNOMEDconcept or SNOMEDcodelist object specifying
concepts to exclude from the concept database. By
default, all concepts in the FINDINGS, CAUSES, BODY, LATERALITY,
MORPH, SEVERITY, STAGE and QUAL tables will be included.</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_include">include</code></td>
<td>
<p>a SNOMEDconcept or SNOMEDcodelist object specifying
additional concepts to include in the concept database. By
default, all findings are included for potential export, but
there may additional concepts of other semantic types
(e.g. situation concepts) that need to be included. Inclusion
takes place after exclusion, i.e. a concept in both the include
and exclude lists will be included.</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_exclude_historic">exclude_historic</code></td>
<td>
<p>a SNOMEDconcept or SNOMEDcodelist object
specifying concepts to be excluded from the &lsquo;historic&rsquo; lookup,
i.e. those that should not be converted into historic forms.
The default is to not do this conversion for disorders, only for
procedures.</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_blacklist">blacklist</code></td>
<td>
<p>a SNOMEDconcept or SNOMEDcodelist object specifying
concepts to filter out of the final output. By default, concepts
in the CDB of any semantic type other than &lsquo;finding&rsquo; or &lsquo;disorder&rsquo;
are excluded. The blacklist can be used to exclude a subset of
findings or disorders that are not useful for the particular
application.</p>
</td></tr>
<tr><td><code id="exportMiADECDB_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following files are exported:
</p>
<p>For MedCAT (named entity recognition and linking):
</p>

<dl>
<dt>problems_cdb.csv</dt><dd><p> - CSV file in MedCAT concept database
format
containing cui (SNOMED CT concept ID), name, name_status ('P'
for preferred term, 'N' for terms that must be disambiguated
(e.g. acronyms or short terms), 'A' for synonym),
ontologies = SNO (for SNOMED CT)</p>
</dd> 
</dl>

<p>For MiADE postprocessing:
</p>

<dl>
<dt>negated.csv</dt><dd><p> - CSV file with columns findingId (SNOMED CT
concept ID of the underlying finding / disorder) and
situationId (SNOMED CT concept ID of the pre-coordinated
situation concept for negation of the finding / disorder).
Sorted by findingId.</p>
</dd>
<dt>historic.csv</dt><dd><p> - CSV file with columns findingId (SNOMED CT
concept ID of the underlying finding / disorder) and
situationId (SNOMED CT concept ID of the pre-coordinated
situation concept for 'history of' the finding / disorder).
Sorted by findingId.</p>
</dd>
<dt>suspected.csv</dt><dd><p> - CSV file with columns findingId (SNOMED CT
concept ID of the underlying finding / disorder) and
situationId (SNOMED CT concept ID of the pre-coordinated
situation concept for 'suspected' finding / disorder).
Sorted by findingId.</p>
</dd>
<dt>overlap.csv</dt><dd><p> - CSV file with columns findingId (SNOMED CT
concept ID of the underlying finding / disorder) and
otherId (SNOMED CT concept ID of a concept with the same
description but a different semantic type, typically a
morphologic abnormality). Sorted by otherId.</p>
</dd>
<dt>problem_blacklist.csv</dt><dd><p> - CSV file without header with one
column containing SNOMED CT concept IDs for concepts that
may be identified by MedCAT as part of text analysis but 
should not be included in final MiADE output, Examples include
procedure codes which may be used to link to precoordinated
&lsquo;history of...&rsquo; concepts. This file can also be used to
force MiADE to ignore any specific SNOMED CT concepts in the
output. Sorted in ascending order.</p>
</dd>
</dl>

<p>For more information about MiADE, visit
<a href="https://www.ucl.ac.uk/health-informatics/research/miade/miade-software-and-availability">https://www.ucl.ac.uk/health-informatics/research/miade/miade-software-and-availability</a>
</p>
<p>For more information about MedCAT, visit 
<a href="https://github.com/CogStack/MedCAT">https://github.com/CogStack/MedCAT</a>
</p>


<h3>Value</h3>

<p>TRUE if successful
</p>


<h3>See Also</h3>

<p>createCDB, downloadWordnet, downloadOrphanet, MANUAL_SYNONYMS,
exclude_irrelevant_findings
</p>
<p>Other MiADE functions: 
<code><a href="#topic+exclude_irrelevant_findings">exclude_irrelevant_findings</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not run
# exportMiADECDB(CDB, export_folderpath = tempdir())
</code></pre>

<hr>
<h2 id='exportSNOMEDenvir'>Export a SNOMED environment to a folder</h2><span id='topic+exportSNOMEDenvir'></span>

<h3>Description</h3>

<p>Creates tab separated files which can be reloaded with relevant indices for fast searching of SNOMED CT tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportSNOMEDenvir(SNOMED, folder)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exportSNOMEDenvir_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing data.table objects: CONCEPT,
DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP, REFSET,
SIMPLEMAP, EXTENDEDMAP</p>
</td></tr>
<tr><td><code id="exportSNOMEDenvir_+3A_folder">folder</code></td>
<td>
<p>path to folder where files will be written</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>CONCEPT, DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP
</p>

<hr>
<h2 id='getMaps'>Obtain Read 2, CTV3, ICD-10 and OPCS4 maps for SNOMED CT concepts</h2><span id='topic+getMaps'></span>

<h3>Description</h3>

<p>Returns concepts mapped to SNOMED CT from either the
SIMPLEMAP table in the SNOMED dictionay (Clinical Terms Version 3,
CTV3 maps, one per concept),
the EXTENDEDMAP table (ICD-10 and OPCS4 maps) or a separate mapping
table with Read Clinical Terms Version 2 (Read 2) and 
CTV3 maps. A sample mapping table (READMAPS) is provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMaps(
  x,
  mappingtable = NULL,
  to = c("read2", "ctv3", "icd10", "opcs4", "ctv3simple"),
  SNOMED = getSNOMED(),
  single_row_per_concept = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMaps_+3A_x">x</code></td>
<td>
<p>SNOMEDcodelist or SNOMEDconcept object. If it is a
SNOMEDconcept object it is first converted to a SNOMEDcodelist.
If it is a SNOMEDcodelist it is first converted to 'simple' format.
Columns named 'read2_code' or 'read2_term' (if adding Read 2 maps) or
'ctv3_concept' or ctv3_termid' (if adding CTV3 maps) will be
overwritten.</p>
</td></tr>
<tr><td><code id="getMaps_+3A_mappingtable">mappingtable</code></td>
<td>
<p>data.table containing mapping in the format
described in 'Details'. The MAPS dataset in this package provides
a sample.
It must contain a unique field 'conceptId', and fields named
'read2_code' and 'read2_term' (for mapping to Read 2)
or 'ctv3_concept' and 'ctv3_termid' (for mapping to CTV3).</p>
</td></tr>
<tr><td><code id="getMaps_+3A_to">to</code></td>
<td>
<p>character vector stating which terminologies to map to.
Options are 'icd10', 'opcs4', 'ctv3simple' (use tables included
within the SNOMED dictionary), or 'read2' or 'ctv3' (require a
separate mapping table such as READMAPS).
Beware that including multiple destination terminologies may
result in a significant expansion
of the number of rows if single_row_per_concept is FALSE.</p>
</td></tr>
<tr><td><code id="getMaps_+3A_snomed">SNOMED</code></td>
<td>
<p>an environment containing the SNOMED CT dictionary.
If not supplied, it will be obtained using getSNOMED().</p>
</td></tr>
<tr><td><code id="getMaps_+3A_single_row_per_concept">single_row_per_concept</code></td>
<td>
<p>(logical) if TRUE (default), the function
returns a single row per concept with Read 2 and CTV3 maps returned
as lists (i.e. multiple entries within a single cell). This means
the output is a valid SNOMEDcodelist object. If FALSE, returns
multiple rows per concept (one for each map).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mapping table can be created from the NHS Digital 'Data
Migration' pack files which contain 'forward' maps of Read 2 and
CTV3 to SNOMED CT.
These are intended for converting individual
entries in electronic health records to SNOMED CT. The 
'forward' map files contain a SNOMED CT map for every Read 2 or
CTV3 code, but not all the SNOMED CT concepts are mapped.
Future SNOMED CT concepts will also not be mapped.
</p>
<p>These maps can be used for converting SNOMED CT codelists into
Read 2 or CTV3 format for running queries, such as to characterise
patient phenotypes or identify patient populations for research.
They cannot be used in the reverse direction (to map a Read 2/CTV3
codelist to SNOMED CT) because some of the SNOMED CT terms will
be missed out, and the list will be incomplete.
</p>
<p>The mapping table must be a data.table object with columns:
conceptId (integer64, unique),
read2_code (character list of 7-character Read 2 codes),
read2_term (character list of Read 2 terms),
ctv3_concept (character list of CTV3 concept codes),
ctv3_termid (character list of CTV3 term description codes)
</p>


<h3>Value</h3>

<p>a data.table containing the columns conceptId and either
'read2_code' and 'read2_term' (for mapping to Read 2),
'ctv3_concept' and 'ctv3_termid' (for mapping to CTV3 using the
mapping table), 'ctv3_simple' (mapping to CTV3 using SIMPLEMAP
within the SNOMED dictionary), 'icd10_code' or 'opcs4_code'
(mapped using EXTENDEDMAP within the SNOMED dictionary).
If single_row_per_concept is TRUE, the mapped rows are of type
'list' and the output is also a SNOMEDcodelist in 'simple' format,
otherwise the output may have multiple rows per
conceptId. Note that each Read 2, CTV3, ICD-10 or OPCS4 term may
be mapped to multiple SNOMED CT concepts.
</p>


<h3>See Also</h3>

<p>READMAPS, loadREADMAPS
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load sample SNOMED CT dictionary into the global environment
# so it is available to the functions in this example
SNOMED &lt;- sampleSNOMED()
# Use the sample READMAPS table in this package
data(READMAPS)

# Example: Mapping a single concept
getMaps(SNOMEDconcept('Heart failure'), mappingtable = READMAPS,
  to = 'read2')
# Example: Mapping a concept and its descendants
getMaps(descendants(SNOMEDconcept('Heart failure')),
  mappingtable = READMAPS, to = 'read2')
# Example: Mapping a codelist
getMaps(SNOMEDcodelist(SNOMEDconcept('Heart failure')),
  mappingtable = READMAPS, to = c('ctv3', 'ctv3simple', 'icd10'))
</code></pre>

<hr>
<h2 id='getRefset'>Retrieves a Refset from the REFSET table</h2><span id='topic+getRefset'></span>

<h3>Description</h3>

<p>Retrieves a Refset from the REFSET table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRefset(conceptIds, SNOMED = getSNOMED())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getRefset_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector of Refset SNOMED
concept IDs, or something that can be coerced to a SNOMEDconcept</p>
</td></tr>
<tr><td><code id="getRefset_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a SNOMEDconcept vector of conceptIds of members of the
selected refset(s)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

getRefset(c('Renal clinical finding simple reference set',
  'Care planning activities simple reference set'))
</code></pre>

<hr>
<h2 id='getSNOMED'>Retrieves SNOMED CT dictionary from the global environment</h2><span id='topic+getSNOMED'></span>

<h3>Description</h3>

<p>Returns an object named 'SNOMED' from the global
environment. Returns an error if no such object exists,
or if it is not an environment containing tables named
CONCEPT, RELATIONSHIP, STATEDRELATIONSHIP and DESCRIPTION.
There is no attempt to check that these tables are actually valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSNOMED(SNOMEDname = "SNOMED")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSNOMED_+3A_snomedname">SNOMEDname</code></td>
<td>
<p>name of the SNOMED environment to search for</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SNOMED environment from the global environment
</p>


<h3>See Also</h3>

<p>CONCEPT, DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP, 
REFSET, SIMPLEMAP, EXTENDEDMAP, loadSNOMED, sampleSNOMED
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()
SNOMED2 &lt;- getSNOMED()

# To display metadata for this SNOMED CT dictionary
SNOMED2$metadata
</code></pre>

<hr>
<h2 id='hasAttributes'>Whether SNOMED CT concepts have particular attributes</h2><span id='topic+hasAttributes'></span>

<h3>Description</h3>

<p>For each concept in the first list, whether it has the attribute
in the second list. Returns a vector of Booleans.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hasAttributes(
  sourceIds,
  destinationIds,
  typeIds = bit64::as.integer64("116680003"),
  SNOMED = getSNOMED(),
  tables = c("RELATIONSHIP", "STATEDRELATIONSHIP"),
  active_only = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hasAttributes_+3A_sourceids">sourceIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs
for children, recycled if necessary</p>
</td></tr>
<tr><td><code id="hasAttributes_+3A_destinationids">destinationIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept
IDs for parents, recycled if necessary</p>
</td></tr>
<tr><td><code id="hasAttributes_+3A_typeids">typeIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs
for renationship types, recycled if necessary.
Defaults to 116680003 = 'Is a' (child/parent)</p>
</td></tr>
<tr><td><code id="hasAttributes_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="hasAttributes_+3A_tables">tables</code></td>
<td>
<p>character vector of relationship tables to use</p>
</td></tr>
<tr><td><code id="hasAttributes_+3A_active_only">active_only</code></td>
<td>
<p>whether only active relationships
should be considered, default TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of Booleans stating whether the attribute exists
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data.table::setDTthreads(threads = 1) # for CRAN testing

SNOMED &lt;- sampleSNOMED()

hasAttributes(c('Heart failure', 'Acute heart failure'),
  c('Heart structure', 'Heart failure'),
  c('Finding site', 'Is a'))
</code></pre>

<hr>
<h2 id='HISTORY'>Sample history substitution table from SNOMED CT dictionary</h2><span id='topic+HISTORY'></span>

<h3>Description</h3>

<p>Sample of the SNOMED CT table showing current equivalents for inactive concepts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HISTORY)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>OLDCONCEPTID</dt><dd><p> integer64: concept ID of the inactive concepts</p>
</dd> 
<dt>OLDCONCEPTSTATUS</dt><dd><p> integer: status of the old concept</p>
</dd> 
<dt>NEWCONCEPTID</dt><dd><p> integer64: concept ID of the new concept</p>
</dd>
<dt>NEWCONCEPTSTATUS</dt><dd><p> integer: status of the new concept</p>
</dd> 
<dt>PATH</dt><dd><p> character: path</p>
</dd> 
<dt>ISAMBIGUOUS </dt><dd><p> integer: whether ambiguous</p>
</dd> 
<dt>ITERATIONS</dt><dd><p> integer: number of iterations</p>
</dd> 
<dt>OLDCONCEPTFSN</dt><dd><p> character: old concept Fully Specified Name</p>
</dd>
<dt>OLDCONCEPTFSN_TAGCOUNT</dt><dd><p> integer: number of tags for old concept</p>
</dd>
<dt>NEWCONCEPTFSN</dt><dd><p> integer: new concept Fully Specified Name</p>
</dd>
<dt>NEWCONCEPTFSN_STATUS</dt><dd><p> integer: new concept Fully Specified Name status</p>
</dd>
<dt>TLH_IDENTICALFLAG</dt><dd><p> integer: whether TLH identical</p>
</dd>
<dt>FSN_TAGLESSIDENTICALFLAG</dt><dd><p> integer: whether Fully Specified Names are identical ignoring the tags</p>
</dd>
<dt>FSN_TAGIDENTICALFLAG</dt><dd><p> integer: whether Fully Specified Names tags are identical</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other SNOMEDsample: 
<code><a href="#topic+QUERY">QUERY</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>
</p>
<p>Other sampleSNOMED: 
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a TEST environment and load the sample dictionaries
TEST &lt;- new.env()
data(CONCEPT, envir = TEST)
data(DESCRIPTION, envir = TEST)
data(RELATIONSHIP, envir = TEST)
data(STATEDRELATIONSHIP, envir = TEST)
data(HISTORY, envir = TEST)

# Show properties of the history table
str(TEST$HISTORY)
</code></pre>

<hr>
<h2 id='htmlCodelistHierarchy'>Export a SNOMEDcodelist hierarchy to HTML</h2><span id='topic+htmlCodelistHierarchy'></span>

<h3>Description</h3>

<p>Exports a codelist with hierarchy as HTML for easy viewing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>htmlCodelistHierarchy(
  x,
  file = NULL,
  title = NULL,
  description = NULL,
  extracols = NULL,
  SNOMED = getSNOMED(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="htmlCodelistHierarchy_+3A_x">x</code></td>
<td>
<p>a SNOMEDcodelist, codelistHierarchy (output of
showCodelistHierarchy), or an object which can be coerced to
a SNOMEDcodelist (such as a SNOMEDconcept vector).</p>
</td></tr>
<tr><td><code id="htmlCodelistHierarchy_+3A_file">file</code></td>
<td>
<p>filename to export to. If NULL, no file is written</p>
</td></tr>
<tr><td><code id="htmlCodelistHierarchy_+3A_title">title</code></td>
<td>
<p>title of HTML document</p>
</td></tr>
<tr><td><code id="htmlCodelistHierarchy_+3A_description">description</code></td>
<td>
<p>paragraph of description text (excluding
&lt;p&gt;&lt;/p&gt; tags)</p>
</td></tr>
<tr><td><code id="htmlCodelistHierarchy_+3A_extracols">extracols</code></td>
<td>
<p>character vector of additional columns of 
codelist_with_hierarchy to include in HTML output</p>
</td></tr>
<tr><td><code id="htmlCodelistHierarchy_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing the SNOMED dictionary to use</p>
</td></tr>
<tr><td><code id="htmlCodelistHierarchy_+3A_...">...</code></td>
<td>
<p>extra arguments to pass to as.SNOMEDcodelist</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector containing HTML output
</p>


<h3>See Also</h3>

<p>showCodelistHierarchy
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data.table::setDTthreads(threads = 1) # for CRAN testing
SNOMED &lt;- sampleSNOMED() # use sample SNOMED for testing

my_concepts &lt;- SNOMEDconcept('Acute heart failure')
my_codelist &lt;- SNOMEDcodelist(data.frame(conceptId = my_concepts,
  include_desc = TRUE))
my_codelist &lt;- getMaps(my_codelist, to = 'icd10')
htmlCodelistHierarchy(my_codelist, file = paste0(tempdir(),
  '/codelist.html'), extracols = 'icd10_code')
# The codelist.html file can now be viewed in a web browser

# Clean up temporary file
file.remove(paste0(tempdir(), '/codelist.html'))
</code></pre>

<hr>
<h2 id='inactiveIncluded'>Check if inactive terms are included in SNOMED CT dictionary</h2><span id='topic+inactiveIncluded'></span>

<h3>Description</h3>

<p>Checks the active_only flag in the metadata of a SNOMED
environment to determine whether inactive terms are
included
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inactiveIncluded(SNOMED = getSNOMED())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inactiveIncluded_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing SNOMED dictionary, defaults
to an object named 'SNOMED' in the global environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE or FALSE (logical vector of length one)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a TEST environment and load the sample dictionaries
TEST &lt;- sampleSNOMED()
inactiveIncluded(TEST)
assign('metadata', list(active_only = TRUE), envir = TEST)
inactiveIncluded(TEST)
</code></pre>

<hr>
<h2 id='is.SNOMEDcodelist'>Check if an object is a SNOMEDcodelist</h2><span id='topic+is.SNOMEDcodelist'></span>

<h3>Description</h3>

<p>SNOMEDcodelist is an S3 class for lists of SNOMED codes.
This function checks whether the object has the class
SNOMEDcodelist, and whether the specified attributes are
as per the arguments (if the arguments are left as NULL, as per
default, they are not checked).
The function does not check if the codelist contains valid data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.SNOMEDcodelist(
  x,
  format = NULL,
  codelist_name = NULL,
  version = NULL,
  author = NULL,
  date = NULL,
  SNOMED = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.SNOMEDcodelist_+3A_x">x</code></td>
<td>
<p>object to check</p>
</td></tr>
<tr><td><code id="is.SNOMEDcodelist_+3A_format">format</code></td>
<td>
<p>Whether the codelist is expressed as a simple
enumeration of concepts ('simple'), as a set of concept
hierarchies ('tree') or as a set of hierarchies showing all
concepts ('exptree'). Codelists can be converted between
the formats, but the result of conversion may depend on the 
SNOMED CT dictionary being used.</p>
</td></tr>
<tr><td><code id="is.SNOMEDcodelist_+3A_codelist_name">codelist_name</code></td>
<td>
<p>Name of the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="is.SNOMEDcodelist_+3A_version">version</code></td>
<td>
<p>Version of the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="is.SNOMEDcodelist_+3A_author">author</code></td>
<td>
<p>Author of the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="is.SNOMEDcodelist_+3A_date">date</code></td>
<td>
<p>Date assigned to the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="is.SNOMEDcodelist_+3A_snomed">SNOMED</code></td>
<td>
<p>Dummy argument to ensure that this function works with
as.SNOMEDcodelist</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical vector of length one: TRUE or FALSE
</p>


<h3>See Also</h3>

<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>

<hr>
<h2 id='is.SNOMEDconcept'>Check if an object is a SNOMEDconcept</h2><span id='topic+is.SNOMEDconcept'></span>

<h3>Description</h3>

<p>SNOMEDconcept is an S3 class for vectors of SNOMED concept IDs
as 64-bit integers. This function checks whether the object has
the class SNOMEDconcept and is a vector of 64-bit integers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.SNOMEDconcept(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.SNOMEDconcept_+3A_x">x</code></td>
<td>
<p>object to check</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical vector of length one: TRUE or FALSE
</p>


<h3>See Also</h3>

<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>

<hr>
<h2 id='loadREADMAPS'>Load mappings from Read to SNOMED CT into an R data.table</h2><span id='topic+loadREADMAPS'></span>

<h3>Description</h3>

<p>Creates a mapping table derived from NHS Digital
Data Migration distribution. These tables are available from
the Technology Reference Update Distribution:
<a href="https://isd.digital.nhs.uk/trud/user/guest/group/0/pack/9/subpack/9/releases">https://isd.digital.nhs.uk/trud/user/guest/group/0/pack/9/subpack/9/releases</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadREADMAPS(
  not_assured_rcsctmap_uk,
  not_assured_rctermsctmap_uk,
  assured_ctv3sctmap2_uk
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadREADMAPS_+3A_not_assured_rcsctmap_uk">not_assured_rcsctmap_uk</code></td>
<td>
<p>File containing Read 2 codes
mapped to SNOMED CT, in file:
'Not Clinically Assured/rcsctmap_uk_20200401000001.txt'</p>
</td></tr>
<tr><td><code id="loadREADMAPS_+3A_not_assured_rctermsctmap_uk">not_assured_rctermsctmap_uk</code></td>
<td>
<p>File containing Read 2 terms
mapped to SNOMED CT, in file:
'Not Clinically Assured/rctermsctmap_uk_20200401000001.txt'</p>
</td></tr>
<tr><td><code id="loadREADMAPS_+3A_assured_ctv3sctmap2_uk">assured_ctv3sctmap2_uk</code></td>
<td>
<p>File containing CTV3 concepts
and terms mapped to SNOMED CT, in file:
'Clinically Assured/ctv3sctmap2_uk_20200401000001.txt'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The final release was in April 2020. The mapping tables are
intended for converting entries in clinical records from
Read Version 2 (Read 2) to SNOMED CT, and Clinical Terms
Version 3 (CTV3) to SNOMED CT.
</p>
<p>These maps can be used for converting SNOMED CT codelists into
Read 2 or CTV3 format for running queries, such as to characterise
patient phenotypes or identify patient populations for research.
They cannot be used in the reverse direction (to map a Read 2/CTV3
codelist to SNOMED CT) because some of the SNOMED CT terms will
be missed out, and the list will be incomplete.
</p>
<p>This function uses the following three mapping files:
</p>

<dl>
<dt>not_assured_rcsctmap_uk</dt><dd><p> File containing Read 2 codes
mapped to SNOMED CT, in file:
'Not Clinically Assured/rcsctmap_uk_20200401000001.txt'</p>
</dd>
<dt>not_assured_rctermsctmap_uk</dt><dd><p> File containing Read 2 terms
mapped to SNOMED CT, in file:
'Not Clinically Assured/rctermsctmap_uk_20200401000001.txt'</p>
</dd>
<dt>assured_ctv3sctmap2_uk</dt><dd><p> File containing CTV3 concepts
and terms mapped to SNOMED CT, in file:
'Clinically Assured/ctv3sctmap2_uk_20200401000001.txt'</p>
</dd>
</dl>

<p>The output data.table has the following columns:
</p>

<dl>
<dt>conceptId</dt><dd><p> integer64: SNOMED CT conceptId (primary key)</p>
</dd> 
<dt>read2_code</dt><dd><p> list: character list of 7-character Read 2 codes</p>
</dd>
<dt>read2_term</dt><dd><p> list: character list of Read 2 terms</p>
</dd>
<dt>ctv3_concept</dt><dd><p> list: character list of CTV3 concept codes</p>
</dd>
<dt>ctv3_termid</dt><dd><p> list: character list of CTV3 term description codes</p>
</dd>
</dl>



<h3>Value</h3>

<p>A data.table with columns conceptId, read2_code,
ctv3_concept, ctv3_termid
</p>


<h3>See Also</h3>

<p>READMAPS, getMaps, loadSNOMED
</p>

<hr>
<h2 id='loadSNOMED'>Load SNOMED CT files from a folder(s) into R data.table objects</h2><span id='topic+loadSNOMED'></span>

<h3>Description</h3>

<p>Identifies relevant SNOMED CT files from the folder structure of
a SNOMED CT distribution. This includes the core 'Snapshot' tables
mapping tables from the 'Refset' folder and the history substitution
table and query table.
The relevant tables are loaded into an R environment, which can be
saved and then easily retrieved for future use. Files from
two folders (e.g. International and UK versions) can be loaded
together, and are automatically appended by the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadSNOMED(folders, active_only = TRUE, version = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadSNOMED_+3A_folders">folders</code></td>
<td>
<p>Vector of folder paths containing SNOMED CT files</p>
</td></tr>
<tr><td><code id="loadSNOMED_+3A_active_only">active_only</code></td>
<td>
<p>Whether to limit to current (active) SNOMED CT
concepts</p>
</td></tr>
<tr><td><code id="loadSNOMED_+3A_version">version</code></td>
<td>
<p>Version description. If NULL, it is derived from the
folder paths and expressed in the form: INT date &amp; UK date</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SNOMED CT files are available from the NHS Digital Technology
Reference Update Distribution:
<a href="https://isd.digital.nhs.uk/trud/user/guest/group/0/home">https://isd.digital.nhs.uk/trud/user/guest/group/0/home</a>
</p>


<h3>Value</h3>

<p>An environment containing data.table objects: CONCEPT,
DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP, REFSET,
SIMPLEMAP, EXTENDEDMAP, HISTORY (optional), QUERY (optional)
</p>


<h3>See Also</h3>

<p>loadREADMAPS, CONCEPT, DESCRIPTION, RELATIONSHIP,
STATEDRELATIONSHIP, REFSET, SIMPLEMAP, EXTENDEDMAP, QUERY, HISTORY
sampleSNOMED, getSNOMED, exportSNOMEDenvir
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a TEST environment and load the sample dictionaries
TEST &lt;- sampleSNOMED()

# Export to temporary directory
exportSNOMEDenvir(TEST, tempdir())

# Try to import using the loadSNOMED function
TEST2 &lt;- loadSNOMED(tempdir(), active_only = FALSE)

# Check that reimported SNOMED dictionary is the same as the original
all.equal(TEST$CONCEPT, TEST2$CONCEPT)
all.equal(TEST$DESCRIPTION, TEST2$DESCRIPTION)
all.equal(TEST$RELATIONSHIP, TEST2$RELATIONSHIP)
all.equal(TEST$STATEDRELATIONSHIP, TEST2$STATEDRELATIONSHIP)
all.equal(TEST$REFSET, TEST2$REFSET)
all.equal(TEST$SIMPLEMAP, TEST2$SIMPLEMAP)
all.equal(TEST$EXTENDEDMAP, TEST2$EXTENDEDMAP)
</code></pre>

<hr>
<h2 id='MANUAL_SYNONYMS'>Sample manual synonym table to assist in creation of concept database</h2><span id='topic+MANUAL_SYNONYMS'></span>

<h3>Description</h3>

<p>Sample manual synonym table to assist in creation of concept database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MANUAL_SYNONYMS)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>synonym</dt><dd><p> character: Synonym, in lower case if case is unimportant</p>
</dd> 
<dt>snomed</dt><dd><p> character: SNOMED CT term, in lower case if case is unimportant</p>
</dd>
<dt>bidirectional</dt><dd><p> boolean: whether synonym can only ever mean
snomed. Not to be used for abbreviations or acronyms.</p>
</dd>
</dl>



<h3>See Also</h3>

<p>createCDB
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MANUAL_SYNONYMS)
str(MANUAL_SYNONYMS)
</code></pre>

<hr>
<h2 id='parents'>Ancestors and descendants of SNOMED CT concepts</h2><span id='topic+parents'></span><span id='topic+ancestors'></span><span id='topic+children'></span><span id='topic+descendants'></span>

<h3>Description</h3>

<p>Returns concepts with 'Is a' or inverse 'Is a'
relationship with a set of target concepts. 
Ancestors include parents and all higher relations.
Descendants include children and all lower relations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parents(
  conceptIds,
  include_self = FALSE,
  SNOMED = getSNOMED(),
  TRANSITIVE = NULL,
  ...
)

ancestors(
  conceptIds,
  include_self = FALSE,
  SNOMED = getSNOMED(),
  TRANSITIVE = NULL,
  ...
)

children(
  conceptIds,
  include_self = FALSE,
  SNOMED = getSNOMED(),
  TRANSITIVE = NULL,
  ...
)

descendants(
  conceptIds,
  include_self = FALSE,
  SNOMED = getSNOMED(),
  TRANSITIVE = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parents_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs</p>
</td></tr>
<tr><td><code id="parents_+3A_include_self">include_self</code></td>
<td>
<p>whether to include the original concept(s) in the
output, default = FALSE</p>
</td></tr>
<tr><td><code id="parents_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="parents_+3A_transitive">TRANSITIVE</code></td>
<td>
<p>transitive closure table for ancestors and
descendants, containing is-a relationships. This table can be 
created by createTransitive to speed up the ancestor / descendant
functions. If a TRANSITIVE table is provided, the SNOMED environment
is not used and relatedConcepts is not called. TRANSITIVE should be
a data.table with columns ancestorId and descendantId.</p>
</td></tr>
<tr><td><code id="parents_+3A_...">...</code></td>
<td>
<p>other arguments to pass to relatedConcepts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a bit64 vector of SNOMED CT concepts
</p>


<h3>See Also</h3>

<p>[createTransitive()] for creation of TRANSITIVE table, and
[relatedConcepts()] for the underlying function to extract
SNOMED CT relationships.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

parents('Heart failure')
children('Heart failure')
ancestors('Heart failure')
descendants('Heart failure')
</code></pre>

<hr>
<h2 id='print.SNOMEDcodelist'>Display a SNOMEDcodelist on screen</h2><span id='topic+print.SNOMEDcodelist'></span>

<h3>Description</h3>

<p>Displays a SNOMEDcodelist on screen, including metadata.
Truncates term descriptions in order to fit within the line width.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDcodelist'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.SNOMEDcodelist_+3A_x">x</code></td>
<td>
<p>SNOMEDcodelist object to print to screen</p>
</td></tr>
<tr><td><code id="print.SNOMEDcodelist_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisibly returns the codelist
</p>


<h3>See Also</h3>

<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+SNOMEDcodelist">SNOMEDcodelist</a>()</code>,
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>
</p>

<hr>
<h2 id='print.SNOMEDconcept'>Display a SNOMEDconcept object with descriptions</h2><span id='topic+print.SNOMEDconcept'></span>

<h3>Description</h3>

<p>SNOMEDconcept is an S3 class for vectors of SNOMED concept IDs
as 64-bit integers. This function checks whether the object has
the class SNOMEDconcept and is a vector of 64-bit integers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDconcept'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.SNOMEDconcept_+3A_x">x</code></td>
<td>
<p>SNOMEDconcept object, or something that can be
coerced to one</p>
</td></tr>
<tr><td><code id="print.SNOMEDconcept_+3A_...">...</code></td>
<td>
<p>not required</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisibly returns a character vector of the SNOMED CT
concepts with descriptions separated by pipe (|)
</p>


<h3>See Also</h3>

<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>

<hr>
<h2 id='print.SNOMEDfindings'>Print method for output of 'decompose' function</h2><span id='topic+print.SNOMEDfindings'></span>

<h3>Description</h3>

<p>Print method for output of 'decompose' function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDfindings'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.SNOMEDfindings_+3A_x">x</code></td>
<td>
<p>SNOMEDfindings object to display</p>
</td></tr>
<tr><td><code id="print.SNOMEDfindings_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='QUERY'>Sample query table from SNOMED CT dictionary</h2><span id='topic+QUERY'></span>

<h3>Description</h3>

<p>Sample of the SNOMED CT table of ancestor / descendant relationships
for inactive concepts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(QUERY)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>supertypeId</dt><dd><p> integer64: concept ID of the ancestor (active) concept</p>
</dd> 
<dt>subtypeId</dt><dd><p> integer64: concept ID of the descendant (inactive) concept</p>
</dd> 
<dt>provenance</dt><dd><p> integer: provenance of relationship. Provenance = 0 means subsumption is always true. Provenance = 1 means subsumption is usually true (but there is a theoretical risk of false positives). Provenance = 2 means both ancestors and descendents are only approximately known. Provenance = 3 means original code had at least two distinct meanings and all are being returned</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other SNOMEDsample: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a TEST environment and load the sample dictionaries
TEST &lt;- new.env()
data(CONCEPT, envir = TEST)
data(DESCRIPTION, envir = TEST)
data(RELATIONSHIP, envir = TEST)
data(STATEDRELATIONSHIP, envir = TEST)
data(QUERY, envir = TEST)

# Show properties of the query table
str(TEST$QUERY)
</code></pre>

<hr>
<h2 id='READMAPS'>Sample mappings from Read to SNOMED CT</h2><span id='topic+READMAPS'></span>

<h3>Description</h3>

<p>A sample of a mapping table derived from NHS Digital maps. 
Contains concepts in Read Clinical Terms Version 2 and 
Clinical Terms Version 3 that map to a set of SNOMED CT
concepts, according to a supplied mapping file. 
The source data are available from the NHS Digital Technology
Reference data Update Distribution
<a href="https://isd.digital.nhs.uk/trud/user/guest/group/0/pack/9/subpack/9/releases">https://isd.digital.nhs.uk/trud/user/guest/group/0/pack/9/subpack/9/releases</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(READMAPS)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>conceptId</dt><dd><p> integer64: SNOMED CT conceptId (primary key)</p>
</dd> 
<dt>read2_code</dt><dd><p> list: character list of 7-character Read V2 codes</p>
</dd>
<dt>read2_term</dt><dd><p> list: character list of Read V2 terms</p>
</dd>
<dt>ctv3_concept</dt><dd><p> list: character list of CTV3 concept codes</p>
</dd>
<dt>ctv3_termid</dt><dd><p> list: character list of CTV3 term description codes</p>
</dd>
</dl>



<h3>See Also</h3>

<p>loadREADMAPS, getMaps
</p>
<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Show properties of the READMAPS table
data(READMAPS)
str(READMAPS)
</code></pre>

<hr>
<h2 id='relatedConcepts'>Obtain related concepts for a set of SNOMED CT concepts</h2><span id='topic+relatedConcepts'></span>

<h3>Description</h3>

<p>Returns concepts with a particular relation to a supplied set of
SNOMED CT concepts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relatedConcepts(
  conceptIds,
  typeId = bit64::as.integer64("116680003"),
  tables = c("RELATIONSHIP", "STATEDRELATIONSHIP"),
  reverse = FALSE,
  recursive = FALSE,
  active_only = TRUE,
  SNOMED = getSNOMED()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="relatedConcepts_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector</p>
</td></tr>
<tr><td><code id="relatedConcepts_+3A_typeid">typeId</code></td>
<td>
<p>concept ID of relationship type.
Defaults to 116680003 = Is a</p>
</td></tr>
<tr><td><code id="relatedConcepts_+3A_tables">tables</code></td>
<td>
<p>vector of names of relationship table(s) to use;
by default use both RELATIONSHIP and STATEDRELATIONSHIP</p>
</td></tr>
<tr><td><code id="relatedConcepts_+3A_reverse">reverse</code></td>
<td>
<p>whether to reverse the relationship</p>
</td></tr>
<tr><td><code id="relatedConcepts_+3A_recursive">recursive</code></td>
<td>
<p>whether to re-apply the function on the outputs</p>
</td></tr>
<tr><td><code id="relatedConcepts_+3A_active_only">active_only</code></td>
<td>
<p>whether to limit the output to active concepts only</p>
</td></tr>
<tr><td><code id="relatedConcepts_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table with the following columns: id, conceptId, type
(only if include_synonyms = TRUE), term,
active (only if active_only = FALSE)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load sample SNOMED CT dictionary
SNOMED &lt;- sampleSNOMED()

# Example: anatomical site of a finding
findingSite &lt;- function(x){
  relatedConcepts(as.SNOMEDconcept(x),
    typeId = as.SNOMEDconcept('Finding site'))
}

description(findingSite('Heart failure'))
# Heart structure (body structure)
</code></pre>

<hr>
<h2 id='sampleSNOMED'>Sample SNOMED CT dictionary</h2><span id='topic+sampleSNOMED'></span>

<h3>Description</h3>

<p>Returns an environment containing a selection of SNOMED CT
terms, their relationships and descriptions which are
provided with the package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleSNOMED()
</code></pre>


<h3>Value</h3>

<p>environment containing four data.table objects:
CONCEPT, DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP
and a list named 'metadata'
</p>


<h3>See Also</h3>

<p>CONCEPT, DESCRIPTION, RELATIONSHIP, STATEDRELATIONSHIP, 
REFSET, SIMPLEMAP, EXTENDEDMAP, HISTORY, QUERY, loadSNOMED, sampleSNOMED
</p>
<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>TEST &lt;- sampleSNOMED()
inactiveIncluded(TEST)
SNOMEDconcept('Heart failure', SNOMED = TEST)

# To display metadata for this SNOMED CT dictionary
sampleSNOMED()$metadata
</code></pre>

<hr>
<h2 id='semanticType'>Retrieves semantic types using the text 'tag' in the description</h2><span id='topic+semanticType'></span>

<h3>Description</h3>

<p>Uses the fully specified name in the DESCRIPTION table. If there are
multiple fully specified names, the name with the most recent
effectiveTime will be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>semanticType(conceptIds, SNOMED = getSNOMED())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="semanticType_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs</p>
</td></tr>
<tr><td><code id="semanticType_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of semantic tags corresponding to the conceptIDs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

semanticType(as.SNOMEDconcept(c('Heart failure', 'Is a')))
</code></pre>

<hr>
<h2 id='simplify'>Retrieves closest single ancestor within a given set of SNOMED CT
concepts</h2><span id='topic+simplify'></span>

<h3>Description</h3>

<p>Returns a vector of SNOMED CT concept IDs for an ancestor of each
concept that is within a second list. If multiple ancestors are
included in the second list, the concept is not simplified (i.e.
the original concept ID is returned).
This functionality can be used to translate concepts into simpler
forms for display, e.g. 'Heart failure' instead of 'Heart failure
with reduced ejection fraction'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simplify(
  conceptIds,
  ancestorIds,
  SNOMED = getSNOMED(),
  tables = c("RELATIONSHIP", "STATEDRELATIONSHIP")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simplify_+3A_conceptids">conceptIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs
for concepts for which an ancestor is sought</p>
</td></tr>
<tr><td><code id="simplify_+3A_ancestorids">ancestorIds</code></td>
<td>
<p>character or integer64 vector of SNOMED concept IDs
for possible ancestors</p>
</td></tr>
<tr><td><code id="simplify_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="simplify_+3A_tables">tables</code></td>
<td>
<p>character vector of relationship tables to use</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is intended for use with active SNOMED CT concepts only.
</p>


<h3>Value</h3>

<p>a data.table with the following columns:
originalId (integer64) = original conceptId,
ancestorId (integer64) = closest single ancestor, or original
concept ID if no ancestor is included among ancestorIds
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

original_terms &lt;- c('Systolic heart failure', 'Is a',
  'Heart failure with reduced ejection fraction',
  'Acute kidney injury due to circulatory failure (disorder)')
# Note in this example 'Is a' has no parents in ancestors,
# and acute kidney failure has two parents in ancestors
# so neither of the parents will be chosen.
# Also test out inclusion of duplicate concepts.

ancestors &lt;- simplify(c(as.SNOMEDconcept(original_terms),
  as.SNOMEDconcept(original_terms)[3:4]),
  as.SNOMEDconcept(c('Heart failure', 'Acute heart failure',
  'Cardiorenal syndrome (disorder)')))
print(cbind(original_terms, description(ancestors$ancestorId)$term))
</code></pre>

<hr>
<h2 id='SNOMED_CONCEPT'>Sample concept table from SNOMED CT dictionary</h2><span id='topic+SNOMED_CONCEPT'></span><span id='topic+CONCEPT'></span>

<h3>Description</h3>

<p>A sample of the SNOMED CT concept table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CONCEPT)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>id</dt><dd><p> integer64: SNOMED CT conceptId (primary key)</p>
</dd> 
<dt>moduleId</dt><dd><p> integer64: class of SNOMED CT concept (whether it is used for recording information or is a metadata concept)</p>
</dd>
<dt>definitionStatusId</dt><dd><p> integer64: 900000000000074008 = primitive concept, 900000000000073002 = defined by conditions</p>
</dd>
<dt>effectiveTime</dt><dd><p> IDate: when the concept became active</p>
</dd>
<dt>active</dt><dd><p> logical: whether this concept is currently active</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Show properties of the CONCEPT table
data('CONCEPT')
str(CONCEPT)
</code></pre>

<hr>
<h2 id='SNOMED_DESCRIPTION'>Sample description table from SNOMED CT dictionary</h2><span id='topic+SNOMED_DESCRIPTION'></span><span id='topic+DESCRIPTION'></span>

<h3>Description</h3>

<p>A sample of the SNOMED CT description table. Each concept may has
a fully specified name and may have any number of synonyms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DESCRIPTION)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>id</dt><dd><p> integer64: description ID</p>
</dd>
<dt>moduleId</dt><dd><p> integer64: class of SNOMED CT concept (whether it
is used for recording information or is a metadata concept)</p>
</dd>
<dt>conceptId</dt><dd><p> integer64: SNOMED CT concept ID</p>
</dd>
<dt>languageCode</dt><dd><p> character: 'en' = English</p>
</dd>
<dt>typeId</dt><dd><p> integer64: 900000000000013009 = Synonym,
900000000000003001 = Fully Specified Name</p>
</dd>
<dt>term</dt><dd><p> character: term description</p>
</dd>
<dt>caseSignificanceId</dt><dd><p> integer64:
900000000000020002 = Initial character case sensitive,
900000000000017005 = Whole term case sensitive,
900000000000448009 = Whole term case insensitive</p>
</dd>
<dt>effectiveTime</dt><dd><p> IDate: when the concept became active</p>
</dd>
<dt>active</dt><dd><p> logical: whether this concept is currently active</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Show properties of the DESCRIPTION table
data('DESCRIPTION')
str(DESCRIPTION)
</code></pre>

<hr>
<h2 id='SNOMED_EXTENDEDMAP'>Sample extended map table from SNOMED CT dictionary</h2><span id='topic+SNOMED_EXTENDEDMAP'></span><span id='topic+EXTENDEDMAP'></span>

<h3>Description</h3>

<p>A sample of the SNOMED CT extended map table, containing maps to ICD-10 and OPCS4.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(EXTENDEDMAP)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>moduleId</dt><dd><p> integer64: core metadata concept: 449080006 = SNOMED CT to ICD-10 rule-based mapping module, 999000031000000106 = SNOMED CT United Kingdom Edition reference set module</p>
</dd>
<dt>refsetId</dt><dd><p> integer64: foundation metadata concept: 447562003 = ICD-10 complex map reference set, 1126441000000105 = Office of Population Censuses and Surveys Classification of Interventions and Procedures Version 4.9 complex map reference set, 999002271000000101 = International Classification of Diseases, Tenth Revision, Fifth Edition, five character code United Kingdom complex map reference set</p>
</dd>
<dt>referencedComponentId</dt><dd><p> integer64: SNOMED CT conceptId of the concept mapped</p>
</dd>
<dt>mapGroup</dt><dd><p> integer: mapping group</p>
</dd>
<dt>mapPriority</dt><dd><p> integer: priority of alternative maps (1 = highest)</p>
</dd>
<dt>mapRule</dt><dd><p> character: advice on mapping rule</p>
</dd>
<dt>mapAdvice</dt><dd><p> character: mapping advice</p>
</dd>
<dt>mapTarget</dt><dd><p> character: target ICD-10 or OPCS4 code. The optional period between the third and fourth character has been removed for consistency.</p>
</dd>
<dt>mapCategoryId</dt><dd><p> integer64: foundation metadata concept describing the quality of the map</p>
</dd>
<dt>effectiveTime</dt><dd><p> IDate: when the concept became active</p>
</dd>
<dt>active</dt><dd><p> logical: whether this concept is currently active</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset and show its properties
data('EXTENDEDMAP')
str(EXTENDEDMAP)

# This EXTENDEDMAP table is part of the sample SNOMED CT dictionary
# Hence this should show the same properties as above
str(sampleSNOMED()$EXTENDEDMAP)
</code></pre>

<hr>
<h2 id='SNOMED_REFSET'>Sample refset table from SNOMED CT dictionary</h2><span id='topic+SNOMED_REFSET'></span><span id='topic+REFSET'></span>

<h3>Description</h3>

<p>A sample of the SNOMED CT refset table. This contains SNOMED CT
codelists that are used for partiular operational or clinical
purposes, and are curated by SNOMED CT. The id column of the
refset table is not included, in order to save space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(REFSET)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>moduleId</dt><dd><p> integer64: SNOMED CT core metadata concept, stating whether the refset is from the SNOMED CT core module or the UK extension.</p>
</dd>
<dt>refsetId</dt><dd><p> integer64: SNOMED CT conceptId of the refset. These concepts have semantic type 'foundation metadata concept'</p>
</dd> 
<dt>referencedComponentId</dt><dd><p> integer64: SNOMED CT conceptId of the member of the refset</p>
</dd> 
<dt>effectiveTime</dt><dd><p> IDate: when the concept became active</p>
</dd>
<dt>active</dt><dd><p> logical: whether this concept is currently active</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset and show its properties
data('REFSET')
str(REFSET)

# This REFSET table is part of the sample SNOMED CT dictionary
# Hence this should show the same properties as above
str(sampleSNOMED()$REFSET)
</code></pre>

<hr>
<h2 id='SNOMED_RELATIONSHIP'>Sample relationship tables from SNOMED CT dictionary</h2><span id='topic+SNOMED_RELATIONSHIP'></span><span id='topic+RELATIONSHIP'></span><span id='topic+STATEDRELATIONSHIP'></span><span id='topic+SNOMED_STATEDRELATIONSHIP'></span>

<h3>Description</h3>

<p>Samples of the SNOMED CT tables of stated relationships (RELATIONSHIP)
and inferred relationships (RELATIONSHIP).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RELATIONSHIP); data(STATEDRELATIONSHIP)

STATEDRELATIONSHIP
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>
<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 329 rows and 10 columns.
</p>


<h3>Details</h3>


<dl>
<dt>id</dt><dd><p> integer64: ID of the relationship record (primary key)</p>
</dd> 
<dt>active</dt><dd><p> logical: whether this concept is currently active</p>
</dd>
<dt>moduleId</dt><dd><p> integer64: class of SNOMED CT concept (whether it
is used for recording information or is a metadata concept)</p>
</dd>
<dt>sourceId</dt><dd><p> integer64: source SNOMED CT concept for the relationship</p>
</dd>
<dt>destinationId</dt><dd><p> integer64: destination SNOMED CT concept for the relationship</p>
</dd>
<dt>relationshipGroup</dt><dd><p> integer: group ID for relationships that are grouped</p>
</dd>
<dt>characteristicTypeId</dt><dd><p> integer64: 900000000000011006 = Inferred relationship</p>
</dd>
<dt>modifierId</dt><dd><p> integer64: 900000000000451002 = Existential restriction modifier</p>
</dd>
<dt>effectiveTime</dt><dd><p> IDate: when the concept became active</p>
</dd>
<dt>typeId</dt><dd><p> integer64: type of relationship, e.g.
116680003 = Is a, 42752001 = Due to, 246090004 = Associated finding,
363698007 = Finding site, 363702006 = Has focus</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other SNOMEDsample: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+QUERY">QUERY</a></code>
</p>
<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_SIMPLEMAP">SNOMED_SIMPLEMAP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a TEST environment and load the sample dictionaries
TEST &lt;- new.env()
data(CONCEPT, envir = TEST)
data(DESCRIPTION, envir = TEST)
data(RELATIONSHIP, envir = TEST)
data(STATEDRELATIONSHIP, envir = TEST) 

# Show properties of the relationship tables
str(TEST$RELATIONSHIP)
str(TEST$STATEDRELATIONSHIP)
</code></pre>

<hr>
<h2 id='SNOMED_SIMPLEMAP'>Sample SIMPLE map table from SNOMED CT dictionary</h2><span id='topic+SNOMED_SIMPLEMAP'></span><span id='topic+SIMPLEMAP'></span>

<h3>Description</h3>

<p>A sample of the SNOMED CT SIMPLE map table, containing maps to ICD-10 and OPCS4.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SIMPLEMAP)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.table"</code>
</p>


<h3>Details</h3>


<dl>
<dt>moduleId</dt><dd><p> integer64: core metadata concept: 900000000000207008 = SNOMED CT core module, 999000021000000109 = SNOMED CT United Kingdom clinical extension reference set module, 999000031000000106 = SNOMED CT United Kingdom Edition reference set module</p>
</dd>
<dt>refsetId</dt><dd><p> integer64: foundation metadata concept: 900000000000497000 = CTV3 simple map reference set, 446608001 = ICD-O simple map reference set, 1323081000000108 = Coronavirus disease 19 caused by severe acute respiratory syndrome coronavirus 2 test result communication to general practice concept simple map reference set, 1323091000000105 = Coronavirus disease 19 caused by severe acute respiratory syndrome coronavirus 2 test result communication to general practice description simple map reference set, 82551000000107 = National Health Service England National Genomic Test Directory whole genome sequencing test simple map reference set</p>
</dd>
<dt>referencedComponentId</dt><dd><p> integer64: SNOMED CT conceptId of the concept mapped</p>
</dd>
<dt>mapTarget</dt><dd><p> character: target ICD-O or CTV3 code</p>
</dd>
<dt>effectiveTime</dt><dd><p> IDate: when the concept became active</p>
</dd>
<dt>active</dt><dd><p> logical: whether this concept is currently active</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other sampleSNOMED: 
<code><a href="#topic+HISTORY">HISTORY</a></code>,
<code><a href="#topic+READMAPS">READMAPS</a></code>,
<code><a href="#topic+SNOMED_CONCEPT">SNOMED_CONCEPT</a></code>,
<code><a href="#topic+SNOMED_DESCRIPTION">SNOMED_DESCRIPTION</a></code>,
<code><a href="#topic+SNOMED_EXTENDEDMAP">SNOMED_EXTENDEDMAP</a></code>,
<code><a href="#topic+SNOMED_REFSET">SNOMED_REFSET</a></code>,
<code><a href="#topic+SNOMED_RELATIONSHIP">SNOMED_RELATIONSHIP</a></code>,
<code><a href="#topic+sampleSNOMED">sampleSNOMED</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset and show its properties
data('SIMPLEMAP')
str(SIMPLEMAP)

# This SIMPLEMAP table is part of the sample SNOMED CT dictionary
# Hence this should show the same properties as above
str(sampleSNOMED()$SIMPLEMAP)
</code></pre>

<hr>
<h2 id='SNOMEDcodelist'>Convert a data.frame to a SNOMEDcodelist object</h2><span id='topic+SNOMEDcodelist'></span><span id='topic+as.SNOMEDcodelist'></span>

<h3>Description</h3>

<p>SNOMEDcodelist is an S3 class for lists of SNOMED CT  concepts.
It consists of conceptId and include_desc columns. The 
option to include descendants allows the creation of more succinct
SNOMED codelists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNOMEDcodelist(
  x,
  include_desc = FALSE,
  format = c("simple", "tree", "exptree"),
  codelist_name = NULL,
  version = NULL,
  author = NULL,
  date = NULL,
  SNOMED = getSNOMED(),
  show_excluded_descendants = FALSE
)

as.SNOMEDcodelist(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SNOMEDcodelist_+3A_x">x</code></td>
<td>
<p>vector of SNOMED CT concept IDs, something which can
be coerced to a SNOMEDconcept object, or a data.frame with
a column 'conceptId' containing SNOMED CT concept concept IDs
in integer64 or text format and optional column 'include_desc'
(Boolean) stating whether descendants of the term should be
included.</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_include_desc">include_desc</code></td>
<td>
<p>Boolean vector stating whether descendants
are included, recycled if necessary. Default = FALSE.
Ignored if x contains a column 'include_desc'</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_format">format</code></td>
<td>
<p>Whether the codelist is expressed as a simple
enumeration of concepts ('simple'), as a set of concept
hierarchies ('tree'), or concept hierarchies showing all
descendant terms ('exptree'). Codelists can be converted
between the formats, but the result of conversion may depend on
the SNOMED CT dictionary being used.</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_codelist_name">codelist_name</code></td>
<td>
<p>Name of the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_version">version</code></td>
<td>
<p>Version of the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_author">author</code></td>
<td>
<p>Author of the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_date">date</code></td>
<td>
<p>Date attributed to the codelist (character vector of length 1)</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_snomed">SNOMED</code></td>
<td>
<p>environment containing a SNOMED dictionary</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_show_excluded_descendants">show_excluded_descendants</code></td>
<td>
<p>Whether to show excluded
descendants alongside the codes included in the codelist (for
a 'tree' or 'exptree' format codelist).</p>
</td></tr>
<tr><td><code id="SNOMEDcodelist_+3A_...">...</code></td>
<td>
<p>other arguments to pass to SNOMEDcodelist</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input is a data.frame or data.table with column names 'conceptId'
and optionally 'include_desc', which is FALSE by default, but if
TRUE then the codelist automatically includes all active descendants
of that concept.
</p>
<p>If the codelist is intended to contain inactive concepts, it can
only exist in the 'simple' format. Inactive concepts will be lost if
the codelist is converted between formats.
</p>
<p>as.SNOMEDcodelist converts its argument into a SNOMEDcodelist but
leaves it unchanged if it is already a SNOMEDcodelist.
</p>


<h3>Value</h3>

<p>An object of class 'SNOMEDcodelist'
</p>


<h3>See Also</h3>

<p>htmlCodelistHierarchy
</p>
<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>
<p>Other SNOMEDcodelist functions: 
<code><a href="#topic+addInactiveConcepts">addInactiveConcepts</a>()</code>,
<code><a href="#topic+expandSNOMED">expandSNOMED</a>()</code>,
<code><a href="#topic+export">export</a>()</code>,
<code><a href="#topic+is.SNOMEDcodelist">is.SNOMEDcodelist</a>()</code>,
<code><a href="#topic+print.SNOMEDcodelist">print.SNOMEDcodelist</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNOMED &lt;- sampleSNOMED()

my_concepts &lt;- SNOMEDconcept('Heart failure')
SNOMEDcodelist(my_concepts)
SNOMEDcodelist(data.frame(conceptId = my_concepts))
as.SNOMEDcodelist(data.frame(conceptId = my_concepts,
  include_desc = TRUE))
</code></pre>

<hr>
<h2 id='union.SNOMEDconcept'>Set operations for SNOMEDconcept vectors</h2><span id='topic+union.SNOMEDconcept'></span><span id='topic+union'></span><span id='topic+union.default'></span><span id='topic+intersect.SNOMEDconcept'></span><span id='topic+intersect'></span><span id='topic+intersect.default'></span><span id='topic+setdiff.SNOMEDconcept'></span><span id='topic+setdiff'></span><span id='topic+setdiff.default'></span>

<h3>Description</h3>

<p>The default set functions in the base package do not handle integer64
vectors correctly, so this package also provides new generic functions
for union, intersect and setdiff, which enable the appropriate
object-specific function to be called according to the class of the
vector. This means that SNOMEDconcept vectors will remain as 
SNOMEDconcept vectors when these functions are used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDconcept'
union(x, y)

union(x, y)

## Default S3 method:
union(x, y)

## S3 method for class 'SNOMEDconcept'
intersect(x, y)

intersect(x, y)

## Default S3 method:
intersect(x, y)

## S3 method for class 'SNOMEDconcept'
setdiff(x, y)

setdiff(x, y)

## Default S3 method:
setdiff(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="union.SNOMEDconcept_+3A_x">x</code></td>
<td>
<p>SNOMEDconcept vector</p>
</td></tr>
<tr><td><code id="union.SNOMEDconcept_+3A_y">y</code></td>
<td>
<p>SNOMEDconcept vector, or an object that can be coerced
to SNOMEDconcept by as.SNOMEDconcept</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an integer64 vector of SNOMEDconcept class
</p>


<h3>See Also</h3>

<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>
<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+unique.SNOMEDconcept">unique.SNOMEDconcept</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sys_acute &lt;- SNOMEDconcept(c('Systolic heart failure',
  'Acute heart failure'), SNOMED = sampleSNOMED())
acute_left_right &lt;- SNOMEDconcept(c('Acute heart failure',
  'Left heart failure', 'Right heart failure'),
  SNOMED = sampleSNOMED())
union(sys_acute, acute_left_right) 
intersect(sys_acute, acute_left_right)
setdiff(sys_acute, acute_left_right)
</code></pre>

<hr>
<h2 id='unique.SNOMEDconcept'>Unique vector of SNOMED CT concepts</h2><span id='topic+unique.SNOMEDconcept'></span>

<h3>Description</h3>

<p>SNOMEDconcept is an S3 class for vectors of SNOMED concept IDs
as 64-bit integers. This function returns a vector containing only
unique SNOMEDconcept values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SNOMEDconcept'
unique(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="unique.SNOMEDconcept_+3A_x">x</code></td>
<td>
<p>SNOMEDconcept vector</p>
</td></tr>
<tr><td><code id="unique.SNOMEDconcept_+3A_...">...</code></td>
<td>
<p>other variables to pass on to the underlying 'unique' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SNOMEDconcept vector with duplicates removed
</p>


<h3>See Also</h3>

<p>Other SNOMEDconcept functions: 
<code><a href="#topic+as.data.frame.SNOMEDconcept">as.data.frame.SNOMEDconcept</a>()</code>,
<code><a href="#topic+c.SNOMEDconcept">c.SNOMEDconcept</a>()</code>,
<code><a href="#topic+is.SNOMEDconcept">is.SNOMEDconcept</a>()</code>,
<code><a href="#topic+print.SNOMEDconcept">print.SNOMEDconcept</a>()</code>,
<code><a href="#topic+union.SNOMEDconcept">union.SNOMEDconcept</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hf &lt;- SNOMEDconcept('Heart failure', SNOMED = sampleSNOMED())
hf2 &lt;- c(hf, hf)
unique(hf2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
