<!DOCTYPE html><html><head><title>Help for package RDML</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RDML}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[.GetFData'><p>Extract data points from <code>RDML</code> object</p></a></li>
<li><a href='#adpsType'><p>adpsType R6 class.</p></a></li>
<li><a href='#annotationType'><p>annotationType R6 class.</p></a></li>
<li><a href='#as.character.idType'><p>Convert <code>idType</code> object to <code>character</code></p></a></li>
<li><a href='#as.character.reactIdType'><p>Convert <code>reactIdType</code> object to <code>character</code></p></a></li>
<li><a href='#AsDendrogram'><p><code>RDML$AsDendrogram()</code> wrapper</p></a></li>
<li><a href='#AsTable'><p><code>RDML$AsTable()</code> wrapper</p></a></li>
<li><a href='#baseTemperatureType'><p>baseTemperatureType R6 class.</p></a></li>
<li><a href='#cdnaSynthesisMethodType'><p>cdnaSynthesisMethodType R6 class.</p></a></li>
<li><a href='#commercialAssayType'><p>commercialAssayType R6 class.</p></a></li>
<li><a href='#cqDetectionMethodType'><p>cqDetectionMethodType R6 class.</p></a></li>
<li><a href='#dataCollectionSoftwareType'><p>dataCollectionSoftwareType R6 class.</p></a></li>
<li><a href='#dataType'><p>dataType R6 class.</p></a></li>
<li><a href='#documentationType'><p>documentationType R6 class.</p></a></li>
<li><a href='#dyeType'><p>dyeType R6 class.</p></a></li>
<li><a href='#enumType'><p>enumType R6 class.</p></a></li>
<li><a href='#experimenterType'><p>experimenterType R6 class.</p></a></li>
<li><a href='#experimentType'><p>experimentType R6 class.</p></a></li>
<li><a href='#GetFData'><p><code>RDML$GetFData()</code> wrapper</p></a></li>
<li><a href='#gradientType'><p>gradientType R6 class.</p></a></li>
<li><a href='#idReferencesType'><p>idReferencesType R6 class.</p></a></li>
<li><a href='#idType'><p>idType R6 class.</p></a></li>
<li><a href='#labelFormatType'><p>labelFormatType R6 class.</p></a></li>
<li><a href='#lidOpenType'><p>lidOpenType R6 class.</p></a></li>
<li><a href='#loopType'><p>loopType R6 class.</p></a></li>
<li><a href='#mdpsType'><p>mdpsType R6 class.</p></a></li>
<li><a href='#measureType'><p>measureType R6 class.</p></a></li>
<li><a href='#MergeRDMLs'><p>Merges <code>RDML</code> objects</p></a></li>
<li><a href='#new'><p>Creates new instance of <code>RDML</code> class object</p></a></li>
<li><a href='#nucleotideType'><p>nucleotideType R6 class.</p></a></li>
<li><a href='#oligoType'><p>oligoType R6 class.</p></a></li>
<li><a href='#pauseType'><p>pauseType R6 class.</p></a></li>
<li><a href='#pcrFormatType'><p>pcrFormatType R6 class.</p></a></li>
<li><a href='#primingMethodType'><p>primingMethodType R6 class.</p></a></li>
<li><a href='#quantityType'><p>quantityType R6 class.</p></a></li>
<li><a href='#quantityUnitType'><p>quantityUnitType R6 class.</p></a></li>
<li><a href='#RDML'><p>R6 class <code>RDML</code> &ndash; contains methods to read and overview fluorescence</p>
data from RDML v1.1 and v1.2 format files</a></li>
<li><a href='#RDML.AsDendrogram'><p>Represents structure of <code>RDML</code> file as dendrogram</p></a></li>
<li><a href='#RDML.AsTable'><p>Represents fields of <code>RDML</code> object as <code>data.frame</code></p></a></li>
<li><a href='#RDML.GetFData'><p>Gets fluorescence data vectors from <code>RDML</code> object</p></a></li>
<li><a href='#RDML.SetFData'><p>Sets fluorescence data vectors to <code>RDML</code> object</p></a></li>
<li><a href='#rdmlBaseType'><p>Base R6 class for RDML package.</p></a></li>
<li><a href='#rdmlEdit'><p>RDML Editor Graphical User Interface</p></a></li>
<li><a href='#rdmlIdType'><p>rdmlIdType R6 class.</p></a></li>
<li><a href='#reactIdType'><p>reactIdType R6 class.</p></a></li>
<li><a href='#reactType'><p>reactType R6 class.</p></a></li>
<li><a href='#runType'><p>runType R6 class.</p></a></li>
<li><a href='#sampleType'><p>sampleType R6 class.</p></a></li>
<li><a href='#sampleTypeType'><p>sampleTypeType R6 class.</p></a></li>
<li><a href='#sequencesType'><p>sequencesType R6 class.</p></a></li>
<li><a href='#SetFData'><p><code>RDML$SetFData()</code> wrapper</p></a></li>
<li><a href='#stepType'><p>stepType R6 class.</p></a></li>
<li><a href='#targetType'><p>targetType R6 class.</p></a></li>
<li><a href='#targetTypeType'><p>targetTypeType R6 class.</p></a></li>
<li><a href='#temperatureType'><p>temperatureType R6 class.</p></a></li>
<li><a href='#templateQuantityType'><p>templateQuantityType R6 class.</p></a></li>
<li><a href='#thermalCyclingConditionsType'><p>thermalCyclingConditionsType R6 class.</p></a></li>
<li><a href='#xRefType'><p>xRefType R6 class.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Importing Real-Time Thermo Cycler (qPCR) Data from RDML Format
Files</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-06-25</td>
</tr>
<tr>
<td>Description:</td>
<td>Imports real-time thermo cycler (qPCR) data from Real-time PCR
    Data Markup Language (RDML) and transforms to the appropriate formats of
    the 'qpcR' and 'chipPCR' packages. Contains a dendrogram visualization 
    for the structure of RDML object and GUI for RDML editing.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kablag/RDML">https://github.com/kablag/RDML</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>checkmate (&ge; 1.6.2), data.table, pipeR, readxl, rlist (&ge;
0.4), R6 (&ge; 2.0.1), stringr, tools (&ge; 3.2), xml2 (&ge; 1.0),
lubridate (&ge; 1.6.0)</td>
</tr>
<tr>
<td>Collate:</td>
<td>'RDML.types.R' 'RDML.R' 'RDML.AsDendrogram.R' 'RDML.AsTable.R'
'RDML.GetFData.R' 'RDML.Merge.R' 'RDML.SetFData.R'
'RDML.init.R' 'functional_wrappers.R' 'rdmlEdit.R'</td>
</tr>
<tr>
<td>Suggests:</td>
<td>chipPCR, magrittr, reshape2, qpcR, dplyr, ggplot2, knitr,
kfigr, MBmca, shiny, shinyjs, shinythemes, shinyMolBio, V8,
testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-25 09:51:34 UTC; kablag</td>
</tr>
<tr>
<td>Author:</td>
<td>Konstantin A. Blagodatskikh [cre, aut],
  Stefan Roediger [aut],
  Michal Burdukiewicz
    <a href="https://orcid.org/0000-0001-8926-582X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Andrej-Nikolai Spiess [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Konstantin A. Blagodatskikh &lt;k.blag@yandex.ru&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-25 11:40:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B.GetFData'>Extract data points from <code>RDML</code> object</h2><span id='topic++5B.GetFData'></span><span id='topic++5B.RDML'></span>

<h3>Description</h3>

<p>Extract data points from <code>RDML</code> object as.data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RDML'
x[i, j, dp.type = "adp"]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.GetFData_+3A_x">x</code></td>
<td>
<p><code>RDML</code> object.</p>
</td></tr>
<tr><td><code id="+2B5B.GetFData_+3A_i">i</code>, <code id="+2B5B.GetFData_+3A_j">j</code></td>
<td>
<p>indices.</p>
</td></tr>
<tr><td><code id="+2B5B.GetFData_+3A_dp.type">dp.type</code></td>
<td>
<p>Type of fluorescence data (i.e. 'adp' for qPCR or 'mdp' for
melting).</p>
</td></tr>
</table>

<hr>
<h2 id='adpsType'>adpsType R6 class.</h2><span id='topic+adpsType'></span>

<h3>Description</h3>

<p>adpsType R6 class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adpsType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Contains <code>matrix</code> of amplification data. Must have three columns: </p>

<dl>
<dt>cyc</dt><dd><p>PCR cycle at which data
point was collected (every cycle must have unique number).</p>
</dd>
<dt>tmp</dt><dd><p>temperature in degrees Celsius at the time of measurement (optional).</p>
</dd>
<dt>fluor</dt><dd><p>raw fluorescence intensity measured.</p>
</dd></dl>
<p> Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Initialization</h3>

<pre>adpsType$new(fpoints)</pre>


<h3>Fields</h3>


<dl>
<dt><code>fpoints</code></dt><dd><p><a href="checkmate.html#topic+assertMatrix">assertMatrix</a>. Matrix with amplification data
points.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>#cycles
cyc &lt;- c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17,
18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
34, 35, 36, 37, 38, 39, 40)
#fluorescence
fluo &lt;- c(2.0172, 2.0131, 2.0035, 2, 2.0024, 2.0056, 2.0105, 2.0179,
2.0272, 2.0488, 2.0922, 2.1925, 2.3937, 2.7499, 3.3072, 4.0966,
5.0637, 6.0621, 7.0239, 7.8457, 8.5449, 9.1282, 9.6022, 9.9995,
10.2657, 10.4989, 10.6813, 10.8209, 10.9158, 10.9668, 11.0053,
11.0318, 11.0446, 11.044, 11.0052, 10.9671, 10.9365, 10.9199,
10.897, 10.8316)
#temperature
temp &lt;- c(55, 55, 55, 55, 54, 54, 55, 55, 55, 55, 55, 55, 55, 55, 55,
55, 55, 55, 55, 55, 55, 55, 55, 56, 55, 55, 55, 55, 55, 55, 55,
55, 55, 55, 55, 55, 55, 55, 55, 55)

#combine all variables into a proper object
data &lt;- data.frame(cyc = cyc, tmp = temp, fluor = fluo)

#create adps object
adpsType$new(data)

#create adps object without temperature data
adpsType$new(data[, -2])
</code></pre>

<hr>
<h2 id='annotationType'>annotationType R6 class.</h2><span id='topic+annotationType'></span>

<h3>Description</h3>

<p>Annotate samples by setting a property and its value. For example,
sex could be a property with the possible values M or F. Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotationType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Fields</h3>

 <dl>
<dt>property</dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.
Property name</p>
</dd> <dt>value</dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Value</p>
</dd> </dl>



<h3>Examples</h3>

<pre><code class='language-R'>#set sex property
annotationType$new(property = "sex", value = "M")
</code></pre>

<hr>
<h2 id='as.character.idType'>Convert <code>idType</code> object to <code>character</code></h2><span id='topic+as.character.idType'></span>

<h3>Description</h3>

<p>Function to convert <code>idType</code> object to <code>character</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'idType'
as.character(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.character.idType_+3A_x">x</code></td>
<td>
<p><code>idType</code> object.</p>
</td></tr>
<tr><td><code id="as.character.idType_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed.</p>
</td></tr>
</table>

<hr>
<h2 id='as.character.reactIdType'>Convert <code>reactIdType</code> object to <code>character</code></h2><span id='topic+as.character.reactIdType'></span>

<h3>Description</h3>

<p>Function to convert <code>reactIdType</code> object to <code>character</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'reactIdType'
as.character(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.character.reactIdType_+3A_x">x</code></td>
<td>
<p><code>reactIdType</code> object.</p>
</td></tr>
<tr><td><code id="as.character.reactIdType_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed.</p>
</td></tr>
</table>

<hr>
<h2 id='AsDendrogram'><code>RDML$AsDendrogram()</code> wrapper</h2><span id='topic+AsDendrogram'></span>

<h3>Description</h3>

<p>Read more at <a href="#topic+RDML.AsDendrogram">RDML.AsDendrogram</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AsDendrogram(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AsDendrogram_+3A_obj">obj</code></td>
<td>
<p><code>RDML</code> object.</p>
</td></tr>
<tr><td><code id="AsDendrogram_+3A_...">...</code></td>
<td>
<p><code>AsDendrogram</code> params.</p>
</td></tr>
</table>

<hr>
<h2 id='AsTable'><code>RDML$AsTable()</code> wrapper</h2><span id='topic+AsTable'></span>

<h3>Description</h3>

<p>Read more at <a href="#topic+RDML.AsTable">RDML.AsTable</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AsTable(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AsTable_+3A_obj">obj</code></td>
<td>
<p><code>RDML</code> object.</p>
</td></tr>
<tr><td><code id="AsTable_+3A_...">...</code></td>
<td>
<p><code>AsTable</code> params.</p>
</td></tr>
</table>

<hr>
<h2 id='baseTemperatureType'>baseTemperatureType R6 class.</h2><span id='topic+baseTemperatureType'></span>

<h3>Description</h3>

<p>Parent class for inner usage. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseTemperatureType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>baseTemperatureType$new(duration,
  temperatureChange = NULL, durationChange = NULL, measure = NULL, ramp =
  NULL)</pre>


<h3>Fields</h3>


<dl>
<dt><code>duration</code></dt><dd><p><a href="checkmate.html#topic+checkCount">checkCount</a>. Duration of this
step in seconds.</p>
</dd>
<dt><code>temperatureChange</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. Change
of the temperature between two consecutive cycles: actual temperature
= temperature + (temperatureChange * cycle counter)</p>
</dd>
<dt><code>durationChange</code></dt><dd><p><a href="checkmate.html#topic+checkCount">checkCount</a>. Change of the
duration between two consecutive cycles: actual duration = duration +
(durationChange * cycle counter)</p>
</dd>
<dt><code>measure</code></dt><dd><p><a href="#topic+measureType">measureType</a>. Indicates to make a measurement
and store it as meltcurve or real-time data.</p>
</dd>
<dt><code>ramp</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. Allowed temperature
change between two consecutive cycles in degrees Celsius per second. If unstated,
the maximal change rate is assumed.</p>
</dd>
</dl>


<hr>
<h2 id='cdnaSynthesisMethodType'>cdnaSynthesisMethodType R6 class.</h2><span id='topic+cdnaSynthesisMethodType'></span>

<h3>Description</h3>

<p>Description of the cDNA synthesis method. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cdnaSynthesisMethodType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>cdnaSynthesisMethodType$new(enzyme = NULL,
  primingMethod = NULL, dnaseTreatment = NULL, thermalCyclingConditions =
  NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>enzyme</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Name of the enzyme used for
reverse transcription.</p>
</dd>
<dt><code>primingMethod</code></dt><dd><p><a href="#topic+primingMethodType">primingMethodType</a>.</p>
</dd>
<dt><code>dnaseTreatment</code></dt><dd><p><a href="checkmate.html#topic+checkFlag">checkFlag</a> if <code>TRUE</code>RNA was
DNAse treated prior cDNA synthesis.</p>
</dd>
<dt><code>thermalCyclingConditions</code></dt><dd><p><a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='commercialAssayType'>commercialAssayType R6 class.</h2><span id='topic+commercialAssayType'></span>

<h3>Description</h3>

<p>For some commercial assays, the primer sequences may be unknown. This element
allows to describe commercial assays. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>commercialAssayType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>commercialAssayType$new(company, orderNumber)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>company</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>orderNumber</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>


<hr>
<h2 id='cqDetectionMethodType'>cqDetectionMethodType R6 class.</h2><span id='topic+cqDetectionMethodType'></span>

<h3>Description</h3>

<p>The method used to determine the Cq value.
Can take values:
</p>

<dl>
<dt>&quot;automated threshold and baseline settings&quot;</dt><dd></dd>
<dt>&quot;manual threshold and baseline settings&quot;</dt><dd></dd>
<dt>&quot;second derivative maximum&quot;</dt><dd></dd>
<dt>&quot;other&quot;</dt><dd></dd>
</dl>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cqDetectionMethodType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>cqDetectionMethodType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>


<hr>
<h2 id='dataCollectionSoftwareType'>dataCollectionSoftwareType R6 class.</h2><span id='topic+dataCollectionSoftwareType'></span>

<h3>Description</h3>

<p>Software name and version used to collect and analyze the data. Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataCollectionSoftwareType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>dataCollectionSoftwareType$new(name, version)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>name</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>version</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>dataCollectionSoftwareType$new(name = "ExampleSoft",
                               version = "1.0")
</code></pre>

<hr>
<h2 id='dataType'>dataType R6 class.</h2><span id='topic+dataType'></span>

<h3>Description</h3>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>dataType$new(tar, cq = NULL, excl = NULL,
adp = NULL, mdp = NULL, endPt = NULL, bgFluor = NULL, bgFluorSlp = NULL,
quantFluor = NULL)</pre>


<h3>Fields</h3>


<dl>
<dt><code>tar</code></dt><dd><p><a href="#topic+idReferencesType">idReferencesType</a>.
TargetID - A reference to a target.</p>
</dd>
<dt><code>cq</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.
Calculated fractional PCR cycle used for downstream quantification.
Negative values express following condition: Not Available: -1.0 </p>
</dd>
<dt><code>excl</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Excluded. If <code>excl</code>
is present, this entry should not be evaluated. Do not set this element
to <code>FALSE</code> if the entry is valid. Instead, leave the entire <code>excl</code>
element out instead. It may contain a string with a reason for the exclusion.
Several reasons for exclusion should be
seperated by semicolons &quot;;&quot;.</p>
</dd>
<dt><code>adp</code></dt><dd><p><a href="#topic+adpsType">adpsType</a>.</p>
</dd>
<dt><code>mdp</code></dt><dd><p><a href="#topic+mdpsType">mdpsType</a>.</p>
</dd>
<dt><code>endPt</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a></p>
</dd></dl>
<p>. Value of the endpoint measurement.
</p>
<dl>
<dt><code>bgFluor</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. Background
fluorescence (the y-intercept of the baseline trend based on the estimated
background fluorescence). </p>
</dd>
<dt><code>bgFluorSlp</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.
Background fluorescence slope - The slope of the baseline trend based on
the estimated background fluorescence. The element should be absent to
indicate a slope of 0.0; If this element is present without the <code>bgFluor</code>
element it should be ignored. </p>
</dd>
<dt><code>quantFluor</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. Quantification flourescence -
The fluorescence value corresponding to the treshold line.</p>
</dd> </dl>



<h3>Methods</h3>


<dl>
<dt><code>AsDataFrame(dp.type = "adp")</code></dt><dd><p>Represents amplification
(</p>
<pre>dp.type = "adp"</pre><p>) or melting (<code>dp.type = "mdp"</code>) data
points as <code>data.frame</code></p>
</dd>
</dl>


<hr>
<h2 id='documentationType'>documentationType R6 class.</h2><span id='topic+documentationType'></span>

<h3>Description</h3>

<p>These elements should be used if the same description applies to many
samples, targets or experiments. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>documentationType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>documentationType$new(id, text = NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>. Identificator.</p>
</dd>
<dt><code>text</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Text.</p>
</dd>
</dl>


<hr>
<h2 id='dyeType'>dyeType R6 class.</h2><span id='topic+dyeType'></span>

<h3>Description</h3>

<p>Detailed information about the dye. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dyeType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>dyeType$new(id, description = NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>. Identificator.</p>
</dd>
<dt><code>description</code></dt><dd> <p><a href="checkmate.html#topic+checkString">checkString</a>. Description.
</p>
</dd></dl>


<hr>
<h2 id='enumType'>enumType R6 class.</h2><span id='topic+enumType'></span>

<h3>Description</h3>

<p>Generic class for creating objects thet can take limited list of values. <br />
Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enumType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>enumType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Value.</p>
</dd>
</dl>


<hr>
<h2 id='experimenterType'>experimenterType R6 class.</h2><span id='topic+experimenterType'></span>

<h3>Description</h3>

<p>Contact details of the experimenter. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>experimenterType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>experimenterType$new(id, firstName, lastName,
  email = NULL, labName = NULL, labAddress = NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>. Identificator.</p>
</dd>
<dt><code>firstName</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. First name.</p>
</dd>
<dt><code>lastName</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Last name.</p>
</dd>
<dt><code>email</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Email.</p>
</dd>
<dt><code>labName</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Lab name.</p>
</dd>
<dt><code>labAddress</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Lab address.</p>
</dd>
</dl>


<hr>
<h2 id='experimentType'>experimentType R6 class.</h2><span id='topic+experimentType'></span>

<h3>Description</h3>

<p>A qPCR experiment. It may contain several runs (<a href="#topic+runType">runType</a>). Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>experimentType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>experimentType$new(id, description = NULL,
  documentation = NULL, run = NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>.</p>
</dd>
<dt><code>description</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>documentation</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>run</code></dt><dd><p><code>list</code> of <a href="#topic+runType">runType</a>.</p>
</dd>
</dl>



<h3>Methods</h3>

<dl>
<dt><code>AsDataFrame(dp.type = "adp",
  long.table = FALSE)</code></dt><dd><p>Represents amplification (<code>dp.type = "adp"</code>) or
melting (<code>dp.type = "mdp"</code>) data points as <code>data.frame</code>.
<code>long.table = TRUE</code> means that fluorescence data for all runs and
reacts will be at one collumn.</p>
</dd></dl>


<hr>
<h2 id='GetFData'><code>RDML$GetFData()</code> wrapper</h2><span id='topic+GetFData'></span>

<h3>Description</h3>

<p>Read more at <a href="#topic+RDML.GetFData">RDML.GetFData</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFData(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetFData_+3A_obj">obj</code></td>
<td>
<p><code>RDML</code> object.</p>
</td></tr>
<tr><td><code id="GetFData_+3A_...">...</code></td>
<td>
<p><code>GetFData</code> params.</p>
</td></tr>
</table>

<hr>
<h2 id='gradientType'>gradientType R6 class.</h2><span id='topic+gradientType'></span>

<h3>Description</h3>

<p>Details of the temperature gradient across the PCR block. Inherits:
<a href="#topic+baseTemperatureType">baseTemperatureType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gradientType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>gradientType$new(highTemperature,
  lowTemperature, ...)</pre>


<h3>Fields</h3>


<dl>
<dt><code>highTemperature</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. The highest temperature of
the gradient in degrees Celsius.</p>
</dd>
<dt><code>lowTemperature</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. The lowest temperature of
the gradient in degrees Celsius.</p>
</dd>
<dt><code>...</code></dt><dd><p> Params of parent class <a href="#topic+baseTemperatureType">baseTemperatureType</a>. </p>
</dd></dl>


<hr>
<h2 id='idReferencesType'>idReferencesType R6 class.</h2><span id='topic+idReferencesType'></span>

<h3>Description</h3>

<p>Contains id of another RDML object. Inherits: <a href="#topic+idType">idType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idReferencesType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>idReferencesType$new(id)</pre>


<h3>Fields</h3>


<dl>
<dt><code>id</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Identificator.</p>
</dd>
</dl>


<hr>
<h2 id='idType'>idType R6 class.</h2><span id='topic+idType'></span>

<h3>Description</h3>

<p>Contains identificator for varius RDML types. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>idType$new(id)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Identificator.</p>
</dd>
</dl>


<hr>
<h2 id='labelFormatType'>labelFormatType R6 class.</h2><span id='topic+labelFormatType'></span>

<h3>Description</h3>

<p>Label used for <a href="#topic+pcrFormatType">pcrFormatType</a>.
Can take values:
</p>

<dl>
<dt>ABC</dt><dd></dd>
<dt>123</dt><dd></dd>
<dt>A1a1</dt><dd></dd>
</dl>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>labelFormatType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>labelFormatType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>


<hr>
<h2 id='lidOpenType'>lidOpenType R6 class.</h2><span id='topic+lidOpenType'></span>

<h3>Description</h3>

<p>This step waits for the user to open the lid and continues afterwards. It
allows to stop the program and to wait for the user to add for example
enzymes and continue the program afterwards. The temperature of the previous
step is maintained. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lidOpenType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>lidOpenType$new()</pre>

<hr>
<h2 id='loopType'>loopType R6 class.</h2><span id='topic+loopType'></span>

<h3>Description</h3>

<p>This step allows to form a loop or to exclude some steps. It allows to jump
to a certain &quot;goto&quot; step for &quot;repeat&quot; times. If the &quot;goto&quot; step is outside of
the loop range, it must have &quot;repeat&quot; value &quot;0&quot;. Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loopType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>loopType$new(goto, repeat.n)</pre>


<h3>Fields</h3>

 <dl>
<dt><code>goto</code></dt><dd><p><a href="checkmate.html#topic+assertCount">assertCount</a>.  The
step to go to to form the loop.</p>
</dd>
<dt><code>repeat.n</code></dt><dd><p><a href="checkmate.html#topic+assertCount">assertCount</a>. Determines how many times the loop is
repeated. The first run through the loop is counted as 0, the last loop is
&quot;repeat&quot; - 1.</p>
</dd></dl>


<hr>
<h2 id='mdpsType'>mdpsType R6 class.</h2><span id='topic+mdpsType'></span>

<h3>Description</h3>

<p>Contains <code>matrix</code> of melting data points (single data points measured
during amplification).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mdpsType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Columns: </p>

<dl>
<dt>tmp</dt><dd><p>(temperature in degrees Celsius at the time of measurement.
Every point must have unique value.</p>
</dd>
<dt>fluor</dt><dd><p>fluorescence intensity measured without any correction
(including baselining).</p>
</dd></dl>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Initialization</h3>

<pre>mdpsType$new(fpoints)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>fpoints</code></dt><dd><p><a href="checkmate.html#topic+assertMatrix">assertMatrix</a>. Matrix with amplification data points.</p>
</dd>
</dl>


<hr>
<h2 id='measureType'>measureType R6 class.</h2><span id='topic+measureType'></span>

<h3>Description</h3>

<p>Can take values:
</p>

<dl>
<dt>real time</dt><dd></dd>
<dt>meltcurve</dt><dd></dd>
</dl>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>measureType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>measureType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>


<hr>
<h2 id='MergeRDMLs'>Merges <code>RDML</code> objects</h2><span id='topic+MergeRDMLs'></span>

<h3>Description</h3>

<p>Merges list of <code>RDML</code> objects. The first object in the list becomes base object.
If <code>experiments</code> or <code>runs</code> have same name they will be combined. 
Reacts with same <code>id</code>, <code>experiment</code> and <code>run</code> overwrite each other!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MergeRDMLs(to.merge)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MergeRDMLs_+3A_to.merge">to.merge</code></td>
<td>
<p><code>RDML</code> objects that should be merged.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "lc96_bACTXY.rdml", sep ="")
lc96 &lt;- RDML$new(filename)
filename &lt;- paste(PATH, "/extdata/", "stepone_std.rdml", sep ="")
stepone &lt;- RDML$new(filename)
merged &lt;- MergeRDMLs(list(lc96,stepone))
merged$AsDendrogram()

## End(Not run)
</code></pre>

<hr>
<h2 id='new'>Creates new instance of <code>RDML</code> class object</h2><span id='topic+new'></span><span id='topic+RDML.new'></span>

<h3>Description</h3>

<p>This function has been designed to import data from RDML v1.1 and v1.2 format
files or from <code>xls</code> file generated by <em>Applied Biosystems 7500</em>. To
import from <code>xls</code> this file have to contain <code>Sample Setup</code> and
<code>Multicomponent Data</code> sheets!
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="new_+3A_filename">filename</code></td>
<td>
<p><code>string</code> &ndash; path to file</p>
</td></tr>
<tr><td><code id="new_+3A_show.progress">show.progress</code></td>
<td>
<p><code>logical</code> &ndash; show loading progress bar if
<code>TRUE</code></p>
</td></tr>
<tr><td><code id="new_+3A_conditions.sep">conditions.sep</code></td>
<td>
<p>separator for condition defined at sample name</p>
</td></tr>
<tr><td><code id="new_+3A_format">format</code></td>
<td>
<p><code>string</code> &ndash; input file format. Possible values
<code>auto</code>, <code>rdml</code>, <code>abi</code>, <code>excel</code>, <code>csv</code>. See
Details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>File format options: </p>
<dl>
<dt>auto</dt><dd><p>Tries to detect format by
extension. <code>.xlsx</code> &ndash; <code>excel</code>, <code>.xls</code> &ndash; <code>abi</code>,
<code>.csv</code> &ndash; <code>csv</code>, other &ndash; <code>rdml</code></p>
</dd><dt>abi</dt><dd><p>Reads <code>.xls</code>
files generated by <em>ABI 7500 v.2</em>. To create such files use File&gt;Export;
check 'Sample Setup' and 'Multicomponent Data'; select 'One File'</p>
</dd>
<dt>excel</dt><dd><p><code>.xls</code> or <code>.xslx</code> file with sheets 'description',
'adp', 'mdp'. See example file <code>table.xlsx</code></p>
</dd><dt>csv</dt><dd><p><code>.csv</code> file
with first column 'cyc' or 'tmp' and fluorescence data in other columns</p>
</dd>
<dt>rdml</dt><dd><p><code>.rdml</code> or <code>.lc96p</code> files</p>
</dd></dl>



<h3>Warning</h3>

<p>Although the format RDML claimed as data exchange format,
the specific implementation of the format at devices from real
manufacturers differ significantly. Currently this function is checked
against RDML data from devices: <em>Bio-Rad CFX96</em>, <em>Roche
LightCycler 96</em> and <em>Applied Biosystems StepOne</em>.
</p>


<h3>Author(s)</h3>

<p>Konstantin A. Blagodatskikh &lt;k.blag@yandex.ru&gt;, Stefan Roediger
&lt;stefan.roediger@b-tu.de&gt;, Michal Burdukiewicz
&lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Import from RDML file
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "lc96_bACTXY.rdml", sep ="")
lc96 &lt;- RDML$new(filename)

## Some kind of overview for lc96
lc96$AsTable(name.pattern = sample[[react$sample$id]]$description)
lc96$AsDendrogram()

## End(Not run)
</code></pre>

<hr>
<h2 id='nucleotideType'>nucleotideType R6 class.</h2><span id='topic+nucleotideType'></span>

<h3>Description</h3>

<p>Type of nucleic acid used as a template in the experiment. May have following values:
</p>

<dl>
<dt>DNA</dt><dd></dd>
<dt>genomic DNA</dt><dd></dd>
<dt>cDNA</dt><dd></dd>
<dt>RNA</dt><dd></dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>nucleotideType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Initialization</h3>

<pre>nucleotideType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Value.</p>
</dd>
</dl>


<hr>
<h2 id='oligoType'>oligoType R6 class.</h2><span id='topic+oligoType'></span>

<h3>Description</h3>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oligoType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>oligoType$new(threePrimeTag = NULL,
  fivePrimeTag = NULL, sequence)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>threePrimeTag</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Description of
three prime modification (if present). </p>
</dd>
<dt><code>fivePrimeTag</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Description of
five prime modification (if present).</p>
</dd>
<dt><code>sequence</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>


<hr>
<h2 id='pauseType'>pauseType R6 class.</h2><span id='topic+pauseType'></span>

<h3>Description</h3>

<p>This step allows to pause at a certain temperature. It is typically the last
step in an amplification protocol. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pauseType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>pauseType$new(temperature)</pre>


<h3>Fields</h3>

 <dl>
<dt><code>temperature</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.
The temperature in degrees Celsius maintained during the pause.</p>
</dd>
</dl>


<hr>
<h2 id='pcrFormatType'>pcrFormatType R6 class.</h2><span id='topic+pcrFormatType'></span>

<h3>Description</h3>

<p>The display format of the PCR, analogous to the the qPCR instrument run format.
Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcrFormatType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Rotor formats always have 1 column; rows correspond to the number of places
in the rotor. Values for common formats are: </p>

<table>
<tr>
 <td style="text-align: left;"> Format </td><td style="text-align: left;">
rows </td><td style="text-align: left;"> columns </td><td style="text-align: left;"> rowLabel </td><td style="text-align: left;"> columnLabel </td>
</tr>
<tr>
 <td style="text-align: left;"> single-well </td><td style="text-align: left;"> 1
</td><td style="text-align: left;"> 1       </td><td style="text-align: left;"> 123      </td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;"> 48-well plate </td><td style="text-align: left;"> 6    </td><td style="text-align: left;">
8       </td><td style="text-align: left;"> ABC      </td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;"> 96-well plate </td><td style="text-align: left;"> 8    </td><td style="text-align: left;"> 12
</td><td style="text-align: left;"> ABC      </td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;"> 384-well plate </td><td style="text-align: left;"> 16   </td><td style="text-align: left;"> 24      </td><td style="text-align: left;">
ABC      </td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;"> 1536-well plate </td><td style="text-align: left;"> 32   </td><td style="text-align: left;"> 48      </td><td style="text-align: left;"> ABC
</td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;"> 3072-well array </td><td style="text-align: left;"> 32   </td><td style="text-align: left;"> 96      </td><td style="text-align: left;"> A1a1
</td><td style="text-align: left;"> A1a1        </td>
</tr>
<tr>
 <td style="text-align: left;"> 5184-well chip </td><td style="text-align: left;"> 72   </td><td style="text-align: left;"> 72      </td><td style="text-align: left;"> ABC      </td><td style="text-align: left;">
123         </td>
</tr>
<tr>
 <td style="text-align: left;"> 32-well rotor </td><td style="text-align: left;"> 32   </td><td style="text-align: left;"> 1       </td><td style="text-align: left;"> 123      </td><td style="text-align: left;"> 123
</td>
</tr>
<tr>
 <td style="text-align: left;"> 72-well rotor </td><td style="text-align: left;"> 72   </td><td style="text-align: left;"> 1       </td><td style="text-align: left;"> 123      </td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;">
100-well rotor </td><td style="text-align: left;"> 100  </td><td style="text-align: left;"> 1       </td><td style="text-align: left;"> 123      </td><td style="text-align: left;"> 123         </td>
</tr>
<tr>
 <td style="text-align: left;"> free
format </td><td style="text-align: left;"> -1   </td><td style="text-align: left;"> 1       </td><td style="text-align: left;"> 123      </td><td style="text-align: left;"> 123 </td>
</tr>

</table>
<p> If rows field has value -1,
the function will not try to reconstruct a plate and just display all run
data in a single column. If the columns field has value 1 then the function will
not display a column label.
</p>


<h3>Initialization</h3>

<pre>pcrFormatType$new(rows, columns, rowLabel, columnLabel)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>rows</code></dt><dd><p><a href="checkmate.html#topic+checkCount">checkCount</a>.</p>
</dd>
<dt><code>columns</code></dt><dd><p><a href="checkmate.html#topic+checkCount">checkCount</a>.</p>
</dd>
<dt><code>rowLabel</code></dt><dd><p><a href="#topic+labelFormatType">labelFormatType</a>.</p>
</dd>
<dt><code>columnLabel</code></dt><dd><p><a href="#topic+labelFormatType">labelFormatType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='primingMethodType'>primingMethodType R6 class.</h2><span id='topic+primingMethodType'></span>

<h3>Description</h3>

<p>The primers used in the reverse transcription. Can take values:
</p>

<dl>
<dt>oligo-dt</dt><dd></dd>
<dt>random</dt><dd></dd>
<dt>target-specific</dt><dd></dd>
<dt>oligo-dt and random</dt><dd></dd>
<dt>other</dt><dd></dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>primingMethodType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Initialization</h3>

<pre>primingMethodType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Value.</p>
</dd>
</dl>


<hr>
<h2 id='quantityType'>quantityType R6 class.</h2><span id='topic+quantityType'></span>

<h3>Description</h3>

<p>A quantity is always defined by its value and its unit. Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quantityType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>quantityType$new(value, unit)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. Value.</p>
</dd>
<dt><code>unit</code></dt><dd><p><a href="#topic+quantityUnitType">quantityUnitType</a>. Unit.</p>
</dd>
</dl>


<hr>
<h2 id='quantityUnitType'>quantityUnitType R6 class.</h2><span id='topic+quantityUnitType'></span>

<h3>Description</h3>

<p>The unit the quantity. Can take values:
</p>

<dl>
<dt>cop</dt><dd><p>copies per microliter  </p>
</dd>
<dt>fold</dt><dd><p>fold change  </p>
</dd>
<dt>dil</dt><dd><p>dilution (10 would mean 1:10 dilution)  </p>
</dd>
<dt>nMol</dt><dd><p>nanomol per microliter </p>
</dd>
<dt>ng</dt><dd><p>nanogram per microliter </p>
</dd>
<dt>other</dt><dd><p>other unit (must be linear, no exponents or logarithms allowed) </p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>quantityUnitType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Initialization</h3>

<pre>quantityUnitType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Value.</p>
</dd>
</dl>


<hr>
<h2 id='RDML'>R6 class <code>RDML</code> &ndash; contains methods to read and overview fluorescence 
data from RDML v1.1 and v1.2 format files</h2><span id='topic+RDML'></span>

<h3>Description</h3>

<p>This class is a container for RDML format data (Lefever et al. 
2009). The data may be further transformed to the appropriate format of the 
<code>qpcR</code> (Ritz et al. 2008, Spiess et al. 2008) and <code>chipPCR</code> 
(Roediger et al. 2015) packages (see <a href="#topic+RDML.new">RDML.new</a> for import details). 
Real-time PCR Data Markup Language (RDML) is the recommended file format 
element in the Minimum Information for Publication of Quantitative Real-Time 
PCR Experiments (MIQE) guidelines (Bustin et al. 2009). The inner structure of 
imported data faithfully reflects the structure of RDML file v1.2. All data with 
the exception for fluorescence values can be represented as <code>data.frame</code> by 
method <code>AsTable</code>. Such possibility of data representation streamlines 
sample filtering (by targets, types, etc.) and serves as request for <code>GetFData</code>
method, which extracts fluorescence data for specified samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RDML
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Fields</h3>

<p>Type, structure of data and description of fields can be 
viewed at RDML v1.2 file description. Names of fields are first level of 
XML tree.
</p>


<h3>Methods</h3>


<dl>
<dt>new</dt><dd><p>creates a new instance of <code>RDML</code> class object (see <a href="#topic+RDML.new">RDML.new</a>)</p>
</dd> 
<dt>AsTable</dt><dd><p>represent RDML data as <code>data.frame</code> (see <a href="#topic+RDML.AsTable">RDML.AsTable</a>)</p>
</dd>
<dt>GetFData</dt><dd><p>gets fluorescence data (see <a href="#topic+RDML.GetFData">RDML.GetFData</a>)</p>
</dd>
<dt>SetFData</dt><dd><p>sets fluorescence data (see <a href="#topic+RDML.SetFData">RDML.SetFData</a>)</p>
</dd>
<dt>Merge</dt><dd><p>merges two <code>RDML</code> to one (see <a href="#topic+MergeRDMLs">MergeRDMLs</a>)</p>
</dd>
<dt>AsDendrogram</dt><dd><p>represents structure of <code>RDML</code> object as dendrogram(see 
<a href="#topic+RDML.AsDendrogram">RDML.AsDendrogram</a>)</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Konstantin A. Blagodatskikh &lt;k.blag@yandex.ru&gt;, Stefan Roediger 
&lt;stefan.roediger@b-tu.de&gt;, Michal Burdukiewicz 
&lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>References</h3>

<p>RDML format http://www.rdml.org/ <code>R6</code> package 
http://cran.r-project.org/web/packages/R6/index.html
</p>
<p><code>qpcR</code> package http://cran.r-project.org/web/packages/qpcR/index.html
</p>
<p><code>chipPCR</code> package: 
http://cran.r-project.org/web/packages/chipPCR/index.html
</p>
<p>Roediger S, Burdukiewicz M and Schierack P (2015). chipPCR: an R Package 
to Pre-Process Raw Data of Amplification Curves. <em>Bioinformatics</em> first 
published online April 24, 2015 doi:10.1093/bioinformatics/btv205
</p>
<p>Ritz, C., Spiess, A.-N., 2008. qpcR: an R package for sigmoidal model 
selection in quantitative real-time polymerase chain reaction analysis. 
<em>Bioinformatics</em> 24, 1549&ndash;1551. 
doi:10.1093/bioinformatics/btn227
</p>
<p>Spiess, A.-N., Feig, C., Ritz, C., 2008. Highly accurate sigmoidal fitting 
of real-time PCR data by introducing a parameter for asymmetry. <em>BMC 
Bioinformatics</em> 9, 221. doi:10.1186/1471-2105-9-221
</p>
<p>Bustin, S.A., Benes, V., Garson, J.A., Hellemans, J., Huggett, J., Kubista,
M., Mueller, R., Nolan, T., Pfaffl, M.W., Shipley, G.L., Vandesompele, J., 
Wittwer, C.T., 2009. The MIQE guidelines: minimum information for 
publication of quantitative real-time PCR experiments. <em>Clin. Chem.</em> 
55, 611&ndash;622.  doi:10.1373/clinchem.2008.112797
</p>
<p>Lefever, S., Hellemans, J., Pattyn, F., Przybylski, D.R., Taylor, C., 
Geurts, R., Untergasser, A., Vandesompele, J., RDML consortium, 2009. RDML:
structured language and reporting guidelines for real-time quantitative PCR
data.  <em>Nucleic Acids Res.</em> 37, 2065&ndash;2069. doi:10.1093/nar/gkp056
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## EXAMPLE 1:
## internal dataset lc96_bACTXY.rdml (in 'data' directory)
## generated by Roche LightCycler 96. Contains qPCR data
## with four targets and two types.
## Import with default settings.
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "lc96_bACTXY.rdml", sep ="")
lc96 &lt;- RDML$new(filename)

tab &lt;- lc96$AsTable(name.pattern = paste(sample[[react$sample$id]]$description,
                                         react$id$id), 
                    quantity = sample[[react$sample$id]]$quantity$value)
## Show dyes names
unique(tab$target.dyeId)
## Show types of the samples for dye 'FAM'
library(dplyr)
unique(filter(tab, target.dyeId == "FAM")$sample.type)

## Show template quantities for dye 'FAM' type 'std'#' 
## Not run: 
COPIES &lt;- filter(tab, target.dyeId == "FAM", sample.type == "std")$quantity
## Define calibration curves (type of the samples - 'std').
## No replicates.
library(qpcR)
CAL &lt;- modlist(lc96$GetFData(filter(tab,
                                    target.dyeId == "FAM", 
                                    sample.type == "std")),
               baseline="lin", basecyc=8:15)
## Define samples to predict (first two samples with the type - 'unkn').
PRED &lt;- modlist(lc96$GetFData(filter(tab, 
                                    target.dyeId == "FAM", 
                                    sample.type == "unkn")),
               baseline="lin", basecyc=8:15)
## Conduct quantification.
calib(refcurve = CAL, predcurve = PRED, thresh = "cpD2",
      dil = COPIES)

## End(Not run)
## Not run: 
## EXAMPLE 2:
## internal dataset lc96_bACTXY.rdml (in 'data' directory)
## generated by Roche LightCycler 96. Contains qPCR data
## with four targets and two types.
## Import with default settings.
library(chipPCR)                        
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "lc96_bACTXY.rdml", sep ="")
lc96 &lt;- RDML$new(filename)

tab &lt;- lc96$AsTable(name.pattern = paste(sample[[react$sample$id]]$description,
                                         react$id$id), 
                    quantity = sample[[react$sample$id]]$quantity$value)
## Show targets names
unique(tab$target)
## Fetch cycle dependent fluorescence for HEX chanel
tmp &lt;- lc96$GetFData(filter(tab, target == "bACT", sample.type == "std"))
## Fetch vector of dillutions 
dilution &lt;- filter(tab, target.dyeId == "FAM", sample.type == "std")$quantity

## Use plotCurves function from the chipPCR package to 
## get an overview of the amplification curves
tmp &lt;- as.data.frame(tmp)
plotCurves(tmp[,1], tmp[,-1])
par(mfrow = c(1,1))
## Use inder function from the chipPCR package to 
## calculate the Cq (second derivative maximum, SDM)
SDMout &lt;- sapply(2L:ncol(tmp), function(i) {
  SDM &lt;- summary(inder(tmp[, 1], tmp[, i]), print = FALSE)[2]
})

## Use the effcalc function from the chipPCR package and 
## plot the results for the calculation of the amplification
## efficiency analysis.
plot(effcalc(dilution, SDMout), CI = TRUE)

## End(Not run)
## Not run: 
## EXAMPLE 3:
## internal dataset BioRad_qPCR_melt.rdml (in 'data' directory)
## generated by Bio-Rad CFX96. Contains qPCR and melting data.
## Import with custom name pattern.
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "BioRad_qPCR_melt.rdml", sep ="")
cfx96 &lt;- RDML$new(filename)
## Use plotCurves function from the chipPCR package to 
## get an overview of the amplification curves
library(chipPCR)
## Extract all qPCR data 
tab &lt;- cfx96$AsTable()
cfx96.qPCR &lt;- as.data.frame(cfx96$GetFData(tab))
plotCurves(cfx96.qPCR[,1], cfx96.qPCR[,-1], type = "l")

## Extract all melting data 
cfx96.melt &lt;- cfx96$GetFData(tab, dp.type = "mdp")
## Show some generated names for samples.
colnames(cfx96.melt)[2L:5]
## Select columns that contain
## samples with dye 'EvaGreen' and have type 'pos'
## using filtering by names.
cols &lt;- cfx96$GetFData(filter(tab, grepl("pos_EvaGreen$", fdata.name)),
                       dp.type = "mdp")
## Conduct melting curve analysis.
library(qpcR)
invisible(meltcurve(cols, fluos = 2:ncol(cols),
          temps = rep(1, ncol(cols) - 1)))

## End(Not run)
</code></pre>

<hr>
<h2 id='RDML.AsDendrogram'>Represents structure of <code>RDML</code> file as dendrogram</h2><span id='topic+RDML.AsDendrogram'></span>

<h3>Description</h3>

<p>Plots and/or returns the structure of <code>RDML</code> file as 
<code><a href="stats.html#topic+dendrogram">dendrogram</a></code> (tree-like structure.)
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDML.AsDendrogram_+3A_plot.dendrogram">plot.dendrogram</code></td>
<td>
<p>plots dendrogram if <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dendrogram</code> object
</p>


<h3>Author(s)</h3>

<p>Konstantin A. Blagodatskikh &lt;k.blag@yandex.ru&gt;, Stefan Roediger 
&lt;stefan.roediger@b-tu.de&gt;, Michal Burdukiewicz 
&lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "BioRad_qPCR_melt.rdml", sep ="")
cfx96 &lt;- RDML$new(filename)
#plot dendrogram
cfx96$AsDendrogram()
#assign dendrogram to the object
dendr &lt;- cfx96$AsDendrogram(plot.dendrogram = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='RDML.AsTable'>Represents fields of <code>RDML</code> object as <code>data.frame</code></h2><span id='topic+RDML.AsTable'></span>

<h3>Description</h3>

<p>Formats particular fields of <code>RDML</code> object as <code>data.frame</code>s,
filters or passes them to <code><a href="#topic+RDML.GetFData">RDML.GetFData</a></code> and <code>RDML.SetFData</code>
functions.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDML.AsTable_+3A_.default">.default</code></td>
<td>
<p><code>list</code> of default columns</p>
</td></tr>
<tr><td><code id="RDML.AsTable_+3A_name.pattern">name.pattern</code></td>
<td>
<p>expression to form <code>fdata.name</code> (see Examples)</p>
</td></tr>
<tr><td><code id="RDML.AsTable_+3A_add.columns">add.columns</code></td>
<td>
<p><code>list</code> of additional columns</p>
</td></tr>
<tr><td><code id="RDML.AsTable_+3A_treat.null.as.na">treat.null.as.na</code></td>
<td>
<p>if value is <code>NULL</code> then convert it to <code>NA</code>. Helps to deal with incomplete records.</p>
</td></tr>
<tr><td><code id="RDML.AsTable_+3A_...">...</code></td>
<td>
<p>additional columns</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default input this function forms <code>data.frame</code> with following columns:
</p>
 <dl>
<dt>exp.id</dt><dd><p>experiment$id</p>
</dd> <dt>run.id</dt><dd><p>run$id</p>
</dd>
<dt>react.id</dt><dd><p>react$id</p>
</dd>
<dt>position</dt><dd><p>react$position</p>
</dd>
<dt>sample</dt><dd><p>react$sample</p>
</dd>
<dt>target</dt><dd><p>data$tar$id</p>
</dd> <dt>target.dyeId</dt><dd><p>target[[data$id]]$dyeId</p>
</dd>
<dt>sample.type</dt><dd><p>sample[[react$sample]]$type</p>
</dd> </dl>
<p> You can overload default
columns list by parameter <code>.default</code> but note that columns
</p>
<pre>exp.id, run.id, react.id, target</pre><p> are necessary for usage
<code>AsTable</code> output as
input for <code>GetFData</code> and <code>SetFData</code>. <br /> Additional columns can be
introduced by specifying them at input parameter <code>...</code> (see Examples).
All default and additional columns accession expressions must be named.
</p>
<p>Experiment, run, react and data to which belongs each fluorescence data vector
can be accessed by <code>experiment, run, react, data</code> (see Examples).
</p>
<p>Result table does not contain data from experiments with ids starting with '.'!
</p>


<h3>Author(s)</h3>

<p>Konstantin A. Blagodatskikh &lt;k.blag@yandex.ru&gt;, Stefan Roediger
&lt;stefan.roediger@b-tu.de&gt;, Michal Burdukiewicz
&lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## internal dataset stepone_std.rdml (in 'data' directory)
## generated by Applied Biosystems Step-One. Contains qPCR data.
library(chipPCR)
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "stepone_std.rdml", sep ="")
stepone &lt;- RDML$new(filename)
## Mark fluorescense data which Cq &gt; 30 and add quantities to
## AsTable output.
## Names for fluorescense data will contain sample name and react
## positions
tab &lt;- stepone$AsTable(
         name.pattern = paste(react$sample$id, react$position),
         add.columns = list(cq30 = if(data$cq &gt;= 30) "&gt;=30" else "&lt;30",
         quantity = sample[[react$sample$id]]$quantity$value)
         )
## Show cq30 and quantities
tab[, c("cq30", "quantity")]
## Get fluorescence values for 'std' type samples
## in format ready for ggplot function
library(dplyr)
fdata &lt;- stepone$GetFData(
           filter(tab, sample.type == "std"),
           long.table = TRUE)
## Plot fdata with colour by cq30 and shape by quantity
library(ggplot2)
ggplot(fdata, aes(x = cyc, y = fluor,
                  group = fdata.name,
                  colour = cq30,
                  shape = as.factor(quantity))) +
                  geom_line() + geom_point()

## End(Not run)
</code></pre>

<hr>
<h2 id='RDML.GetFData'>Gets fluorescence data vectors from <code>RDML</code> object</h2><span id='topic+RDML.GetFData'></span>

<h3>Description</h3>

<p>Gets fluorescence data vectors from <code>RDML</code> object for specified method
of experiment.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDML.GetFData_+3A_request">request</code></td>
<td>
<p>Output from AsTable method(<a href="#topic+RDML.AsTable">RDML.AsTable</a>)</p>
</td></tr>
<tr><td><code id="RDML.GetFData_+3A_dp.type">dp.type</code></td>
<td>
<p>Type of fluorescence data (i.e. 'adp' for qPCR or 'mdp' for
melting)</p>
</td></tr>
<tr><td><code id="RDML.GetFData_+3A_long.table">long.table</code></td>
<td>
<p>Output table is ready for ggplot (See <a href="#topic+RDML.AsTable">RDML.AsTable</a>
for example)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>matrix</code> which contains selected fluorescence data and 
additional information fromm request if <code>long.table = TRUE</code>.
</p>


<h3>Author(s)</h3>

<p>Konstantin A. Blagodatskikh &lt;k.blag@yandex.ru&gt;, Stefan Roediger 
&lt;stefan.roediger@b-tu.de&gt;, Michal Burdukiewicz 
&lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## internal dataset BioRad_qPCR_melt.rdml (in 'data' directory)
## generated by Bio-Rad CFX96. Contains qPCR and melting data.
## Import without splitting by targets/types and with
## custom name pattern.
PATH &lt;- path.package("RDML")
filename &lt;- paste(PATH, "/extdata/", "BioRad_qPCR_melt.rdml", sep ="")
cfx96 &lt;- RDML$new(filename)
## Select melting fluorescence data with sample.type 'unkn'.
library(dplyr)
tab &lt;- cfx96$AsTable()
fdata &lt;- cfx96$GetFData(filter(tab, sample.type == "unkn"),
                        dp.type = "adp")
## Show names for obtained fdata
colnames(fdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='RDML.SetFData'>Sets fluorescence data vectors to <code>RDML</code> object</h2><span id='topic+RDML.SetFData'></span>

<h3>Description</h3>

<p>Sets fluorescence data vectors to <code>RDML</code> object for specified method
of experiment.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDML.SetFData_+3A_data">data</code></td>
<td>
<p><code>matrix</code> containing in the first column data corresponding to 
all fluorescence values in the following columns. The name of the first column 
is the name of variable and names of other column are <code>fdata.names</code> (links 
to rows at <code>description</code>).</p>
</td></tr>
<tr><td><code id="RDML.SetFData_+3A_description">description</code></td>
<td>
<p>output from <code>AsTable</code> function that describes fluorescence data.</p>
</td></tr>
<tr><td><code id="RDML.SetFData_+3A_fdata.type">fdata.type</code></td>
<td>
<p>'adp' for qPCR, 'mdp' for melting data.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
PATH &lt;- path.package("RDML")
filename &lt;- paste0(PATH, "/extdata/", "stepone_std.rdml")
cfx96 &lt;- RDML$new(filename)
## Use plotCurves function from the chipPCR package to 
## get an overview of the amplification curves
library(chipPCR)
## Extract all qPCR data 
tab &lt;- cfx96$AsTable()
tab2 &lt;- tab
tab2$run.id &lt;- "cpp"
cfx96.qPCR &lt;- as.data.frame(cfx96$GetFData(tab))
cpp &lt;- cbind(cyc = cfx96.qPCR[, 1],
 apply(cfx96.qPCR[, -1], 2, 
   function(y) CPP(x = cfx96.qPCR[, 1], y = y)$y.norm))
cfx96$SetFData(cpp, tab2)
library(ggplot2)
library(gridExtra)
cfx96.gg &lt;- cfx96$GetFData(tab, long.table = TRUE)
cpp.gg &lt;- cfx96$GetFData(tab2,
                         long.table = TRUE)
plot1 &lt;- ggplot(cfx96.gg, aes(x = cyc, y = fluor,
                group=fdata.name)) +
                 geom_line() +
                 ggtitle("Raw data")
plot2 &lt;- ggplot(cpp.gg, aes(x = cyc, y = fluor,
                group=fdata.name)) +
                 geom_line() +
                 ggtitle("CPP processed data")
grid.arrange(plot1, plot2, nrow=2)

## End(Not run)
</code></pre>

<hr>
<h2 id='rdmlBaseType'>Base R6 class for RDML package.</h2><span id='topic+rdmlBaseType'></span>

<h3>Description</h3>

<p>Most classes from RDML package inherit this class. It is designed for internal
usage and should not be directly accessed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdmlBaseType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>rdmlBaseType$new()</pre>


<h3>Methods</h3>


<dl>
<dt><code>.asXMLnodes(node.name)</code></dt><dd><p>Represents
object as XML nodes. Should not be called directly. <code>node.name</code> &ndash;
name of the root node for the generated XML
tree.</p>
</dd>
<dt><code>print(...)</code></dt><dd><p>prints object</p>
</dd></dl>


<hr>
<h2 id='rdmlEdit'>RDML Editor Graphical User Interface</h2><span id='topic+rdmlEdit'></span>

<h3>Description</h3>

<p>Launches graphical user interface that can edit RDML metadata and show
qPCR or melting curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdmlEdit()
</code></pre>

<hr>
<h2 id='rdmlIdType'>rdmlIdType R6 class.</h2><span id='topic+rdmlIdType'></span>

<h3>Description</h3>

<p>This element can be used to assign a publisher and id to the RDML file.<br />
Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdmlIdType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>rdmlIdType$new(publisher, serialNumber,
  MD5Hash = NULL)</pre>


<h3>Fields</h3>


<dl>
<dt><code>publisher</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. RDML file publisher.</p>
</dd>
<dt><code>serialNumber</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Serial number.</p>
</dd>
<dt><code>MD5Hash</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. An MD5Hash calculated
over the complete file after removing all rdmlIDTypes and all whitespaces
between elements.</p>
</dd>
</dl>


<hr>
<h2 id='reactIdType'>reactIdType R6 class.</h2><span id='topic+reactIdType'></span>

<h3>Description</h3>

<p>Contains identificator for reactType. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reactIdType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>reactIdType$new(id)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="checkmate.html#topic+checkCount">checkCount</a>. Identificator.</p>
</dd>
</dl>


<hr>
<h2 id='reactType'>reactType R6 class.</h2><span id='topic+reactType'></span>

<h3>Description</h3>

<p>A reaction is an independent chemical reaction corresponding for example to a
well in a 96 well plate, a capillary in a rotor, a through-hole on an array,
etc. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reactType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>The ID of this reaction
</p>
<p>Schemas : </p>
 <ul>
<li><p> rotor : assign IDs according to the position of the
sample on the rotor (1 for the 1st sample, 2 for the 2nd, ...) </p>
</li>
<li><p> plate
(96/384/1536 well) : the IDs are assigned in a row-first/column-second
manner. For each row, the samples are numbered according to the increasing
column number. At the end of a row, the numbering starts at the first column
of the next row. An example for this type of plate can be found below :
</p>

<table>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> 1  </td><td style="text-align: left;"> 2  </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;"> A   </td><td style="text-align: left;"> 1  </td><td style="text-align: left;"> 2  </td><td style="text-align: left;">
3 </td><td style="text-align: left;">     </td>
</tr>
<tr>
 <td style="text-align: left;"> B   </td><td style="text-align: left;"> 13 </td><td style="text-align: left;"> 14 </td><td style="text-align: left;">   </td><td style="text-align: left;">     </td>
</tr>
<tr>
 <td style="text-align: left;"> ... </td><td style="text-align: left;">    </td><td style="text-align: left;"> </td><td style="text-align: left;">
</td><td style="text-align: left;">    </td>
</tr>

</table>
<p> or </p>

<table>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> 1  </td><td style="text-align: left;"> 2  </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;"> 1 </td><td style="text-align: left;"> 1
</td><td style="text-align: left;"> 2  </td><td style="text-align: left;"> 3 </td><td style="text-align: left;">     </td>
</tr>
<tr>
 <td style="text-align: left;"> 2   </td><td style="text-align: left;"> 13 </td><td style="text-align: left;"> 14 </td><td style="text-align: left;">   </td><td style="text-align: left;">     </td>
</tr>
<tr>
 <td style="text-align: left;"> ... </td><td style="text-align: left;">
</td><td style="text-align: left;">    </td><td style="text-align: left;">   </td><td style="text-align: left;">    </td>
</tr>

</table>

</li>
<li><p> multi-array plate (BioTrove) : the IDs are assigned in a
row-first/column-second manner, ignoring the organisation of sub-arrays. For
each row, the samples are numbered according to the increasing column number.
At the end of a row, the the next row. An example for this type of plate can
be found below : todo... </p>
</li></ul>



<h3>Initialization</h3>

<pre>reactType$new(id, sample, data = NULL, pcrFormat = pcrFormatType$new(8, 12, labelFormatType$new("123"), labelFormatType$new("ABC")))</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+reactIdType">reactIdType</a>. See 'Details'.</p>
</dd>
<dt><code>sample</code></dt><dd><p><a href="#topic+idReferencesType">idReferencesType</a>. SampleID - A reference to a
sample.</p>
</dd>
<dt><code>data</code></dt><dd><p><code>list</code> of <a href="#topic+dataType">dataType</a>.</p>
</dd>
<dt><code>position</code></dt><dd><p>Human readable form of the <code>react id</code> (i.e. '13' -&gt; 'B1')..</p>
</dd>
</dl>



<h3>Methods</h3>

<dl>
<dt><code>AsDataFrame(dp.type =
  "adp")</code></dt><dd><p>Represents amplification (<code>dp.type = "adp"</code>) or melting
(<code>dp.type = "mdp"</code>) data points of all targets as one
<code>data.frame</code></p>
</dd> <dt><code>.recalcPosition(pcrformat)</code></dt><dd><p>Converts <code>react
  id</code> to the human readable form (i.e. '13' -&gt; 'B1'). This converted value can be
accessed by <code>position</code> field. <code>pcrFormat</code> is
<code>pcrFormatType</code>. Currently, only 'ABC' and '123' are supported as
labels. For '123' '123' the <code>Position</code> will look like 'r01c01', for
'ABC' '123' it will be 'A01' and for '123' 'ABC' it will be 01A. 'ABC' 'ABC'
is not currently supported. Note that 'ABC' will result in loss of
information if the experiment contains more than 26 rows!</p>
</dd></dl>


<hr>
<h2 id='runType'>runType R6 class.</h2><span id='topic+runType'></span>

<h3>Description</h3>

<p>A run is a set of reactions performed in one &quot;run&quot;, for example one plate,
one rotor, one array, one chip. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>runType$new(id, description = NULL,
  documentation = NULL, experimenter = NULL, instrument = NULL,
  dataCollectionSoftware = NULL, backgroundDeterminationMethod = NULL,
  cqDetectionMethod = NULL, thermalCyclingConditions = NULL, pcrFormat,
  runDate = NULL, react = NULL)</pre>


<h3>Fields</h3>


<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>.</p>
</dd>
<dt><code>description</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>documentation</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>experimenter</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>instrument</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Description of the
instrument used to aquire the data.</p>
</dd>
<dt><code>dataCollectionSoftware</code></dt><dd><p><a href="#topic+dataCollectionSoftwareType">dataCollectionSoftwareType</a>.
Description of the software used to analyze/collect the data.</p>
</dd>
<dt><code>backgroundDeterminationMethod</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.
Description of method used to determine the background. </p>
</dd>
<dt><code>cqDetectionMethod</code></dt><dd><p><a href="#topic+cqDetectionMethodType">cqDetectionMethodType</a>. Description of method
used to calculate the quantification cycle. </p>
</dd>
<dt><code>thermalCyclingConditions</code></dt><dd><p><a href="#topic+idReferencesType">idReferencesType</a>. The program
used to aquire the data.</p>
</dd>
<dt><code>pcrFormat</code></dt><dd><p><a href="#topic+adpsType">adpsType</a>.</p>
</dd>
<dt><code>runDate</code></dt><dd><p><a href="#topic+adpsType">adpsType</a>. Time stamp of data acquisition.</p>
</dd>
<dt><code>react</code></dt><dd><p><code>list</code> of <a href="#topic+adpsType">adpsType</a>.</p>
</dd> </dl>



<h3>Methods</h3>


<dl>
<dt><code>AsDataFrame(dp.type = "adp")</code></dt><dd><p>Represents amplification
(<code>dp.type = "adp"</code>) or melting (<code>dp.type = "mdp"</code>) data
points as <code>data.frame</code></p>
</dd></dl>


<hr>
<h2 id='sampleType'>sampleType R6 class.</h2><span id='topic+sampleType'></span>

<h3>Description</h3>

<p>A sample is a template solution with defined concentation. Since dilutions
of the same material differ in concentration, they are considered different
samples. A technical replicate samples should contain the same name (reactions
are performed on the same material), and biological replicates should contain
different names (the template derived from the different biological replicates
is are divergent). Serial dilutions in a standard curve must have different
names (preferably stating their dillution).
Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>sampleType$new(id, description = NULL,
 documentation = NULL, xRef =  NULL, annotation = NULL, type =
 sampleTypeType$new("unkn"), interRunCalibrator = FALSE, quantity = NULL,
 calibratorSample = FALSE, cdnaSynthesisMethod = NULL, templateQuantity =
 NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>. Concentration of the template in nanogram
per microliter in the final reaction mix. </p>
</dd>
<dt><code>description</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>documentation</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>xRef</code></dt><dd><p><code>list</code> of <a href="#topic+xRefType">xRefType</a>.</p>
</dd>
<dt><code>annotation</code></dt><dd><p><code>list</code> of <a href="#topic+annotationType">annotationType</a>.</p>
</dd>
<dt><code>type</code></dt><dd><p><a href="#topic+sampleTypeType">sampleTypeType</a>.</p>
</dd>
<dt><code>interRunCalibrator</code></dt><dd><p><a href="checkmate.html#topic+checkFlag">checkFlag</a>. <code>TRUE</code>
if this sample is used as inter run calibrator. </p>
</dd>
<dt><code>quantity</code></dt><dd><p><a href="#topic+quantityType">quantityType</a>. Quantity - The reference
quantity of this sample. It should be only used if the sample is part of a
standard curve. The provided value will be used to quantify unknown samples
in absolute quantification assays. Only the use of positive integers (like 1,
10, 100, 1000) and fractions (e.g. 1, 0.1, 0.01, 0.001) is acceptable.
The use of exponents (1, 2, 3, 4 or -1, -2, -3, -4) if forbidden,
because it will not be interpreted as 10E1, 10E2, 10E3, 10E4 or 10E-1, 10E-2,
10E-3, 10E-4. </p>
</dd>
<dt><code>calibratorSample</code></dt><dd><p><a href="checkmate.html#topic+checkFlag">checkFlag</a>. <code>TRUE</code> if this
sample is used as calibrator sample. </p>
</dd>
<dt><code>cdnaSynthesisMethod</code></dt><dd><p><a href="#topic+cdnaSynthesisMethodType">cdnaSynthesisMethodType</a>.</p>
</dd>
<dt><code>templateQuantity</code></dt><dd><p><a href="#topic+templateQuantityType">templateQuantityType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='sampleTypeType'>sampleTypeType R6 class.</h2><span id='topic+sampleTypeType'></span>

<h3>Description</h3>

<p>Can take values:
</p>

<dl>
<dt>unkn</dt><dd><p>unknown sample</p>
</dd>
<dt>ntc</dt><dd><p>non template control</p>
</dd>
<dt>nac</dt><dd><p>no amplification control</p>
</dd>
<dt>std</dt><dd><p>standard sample</p>
</dd>
<dt>ntp</dt><dd><p>no target present</p>
</dd>
<dt>nrt</dt><dd><p>minusRT</p>
</dd>
<dt>pos</dt><dd><p>positive control</p>
</dd>
<dt>opt</dt><dd><p>optical calibrator sample</p>
</dd></dl>



<h3>Usage</h3>

<pre><code class='language-R'>sampleTypeType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Details</h3>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Initialization</h3>

<pre>sampleTypeType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Value.</p>
</dd>
</dl>


<hr>
<h2 id='sequencesType'>sequencesType R6 class.</h2><span id='topic+sequencesType'></span>

<h3>Description</h3>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sequencesType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>sequencesType$new(forwardPrimer = NULL,
reversePrimer = NULL, probe1 = NULL, probe2 = NULL, amplicon = NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>forwardPrimer</code></dt><dd><p><a href="#topic+oligoType">oligoType</a>.</p>
</dd>
<dt><code>reversePrimer</code></dt><dd><p><a href="#topic+oligoType">oligoType</a>.</p>
</dd>
<dt><code>probe1</code></dt><dd><p><a href="#topic+oligoType">oligoType</a>.</p>
</dd>
<dt><code>probe2</code></dt><dd><p><a href="#topic+oligoType">oligoType</a>.</p>
</dd>
<dt><code>amplicon</code></dt><dd><p><a href="#topic+oligoType">oligoType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='SetFData'><code>RDML$SetFData()</code> wrapper</h2><span id='topic+SetFData'></span>

<h3>Description</h3>

<p>Read more at <a href="#topic+RDML.SetFData">RDML.SetFData</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SetFData(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SetFData_+3A_obj">obj</code></td>
<td>
<p><code>RDML</code> object.</p>
</td></tr>
<tr><td><code id="SetFData_+3A_...">...</code></td>
<td>
<p><code>SetFData</code> params.</p>
</td></tr>
</table>

<hr>
<h2 id='stepType'>stepType R6 class.</h2><span id='topic+stepType'></span>

<h3>Description</h3>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stepType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>stepType$new(nr, description = NULL,
  temperature = NULL, gradient = NULL, loop = NULL, pause = NULL, lidOpen =
  NULL)</pre>


<h3>Fields</h3>

 <dl>
<dt><code>nr</code></dt><dd><p><a href="checkmate.html#topic+checkCount">checkCount</a>. The
incremental number of the step. First step should have value 1. The increment
between steps should be constant and equivalent to 1.</p>
</dd>
<dt><code>description</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>temperature</code></dt><dd><p><a href="#topic+temperatureType">temperatureType</a>.</p>
</dd>
<dt><code>gradient</code></dt><dd><p><a href="#topic+gradientType">gradientType</a>.</p>
</dd>
<dt><code>loop</code></dt><dd><p><a href="#topic+loopType">loopType</a>.</p>
</dd> <dt><code>pause</code></dt><dd><p><a href="#topic+pauseType">pauseType</a>.</p>
</dd>
<dt><code>lidOpen</code></dt><dd><p><a href="#topic+lidOpenType">lidOpenType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='targetType'>targetType R6 class.</h2><span id='topic+targetType'></span>

<h3>Description</h3>

<p>A target is a PCR reaction with defined set of primers. PCR reactions
for the same gene with distinct primer sequences are considered different
targets. Inherits:
<a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>targetType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>targetType$new(id, description = NULL,
  documentation = NULL, xRef = NULL, type, amplificationEfficiencyMethod =
  NULL, amplificationEfficiency = NULL, amplificationEfficiencySE = NULL,
  detectionLimit = NULL, dyeId, sequences = NULL, commercialAssay = NULL)</pre>


<h3>Fields</h3>

 <dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>.</p>
</dd>
<dt><code>description</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>documentation</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>xRef</code></dt><dd><p><code>list</code> of <a href="#topic+xRefType">xRefType</a>.</p>
</dd>
<dt><code>type</code></dt><dd><p><a href="#topic+targetTypeType">targetTypeType</a>.</p>
</dd>
<dt><code>amplificationEfficiencyMethod</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>amplificationEfficiency</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.</p>
</dd>
<dt><code>amplificationEfficiencySE</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.</p>
</dd>
<dt><code>detectionLimit</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.</p>
</dd>
<dt><code>dyeId</code></dt><dd><p><a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>sequences</code></dt><dd><p><a href="#topic+sequencesType">sequencesType</a>.</p>
</dd>
<dt><code>commercialAssay</code></dt><dd><p><a href="#topic+commercialAssayType">commercialAssayType</a>.</p>
</dd> </dl>


<hr>
<h2 id='targetTypeType'>targetTypeType R6 class.</h2><span id='topic+targetTypeType'></span>

<h3>Description</h3>

<p>Can take values:
</p>

<dl>
<dt>ref</dt><dd><p>reference target</p>
</dd>
<dt>toi</dt><dd><p>target of interest</p>
</dd>
</dl>

<p>Inherits: <a href="#topic+enumType">enumType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>targetTypeType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>targetTypeType$new(value)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>value</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
</dl>


<hr>
<h2 id='temperatureType'>temperatureType R6 class.</h2><span id='topic+temperatureType'></span>

<h3>Description</h3>

<p>This step keeps a constant temperature on the heat block. Inherits:
<a href="#topic+baseTemperatureType">baseTemperatureType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>temperatureType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>temperatureType$new(temperature, ...)</pre>


<h3>Fields</h3>

 <dl>
<dt><code>temperature</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>.
The temperature of the step in  degrees Celsius.</p>
</dd>
<dt><code>...</code></dt><dd><p> Params of parent class <a href="#topic+baseTemperatureType">baseTemperatureType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='templateQuantityType'>templateQuantityType R6 class.</h2><span id='topic+templateQuantityType'></span>

<h3>Description</h3>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>templateQuantityType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>templateQuantityType$new(conc, nucleotide)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>conc</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. Concentration of the template in nanogram</p>
</dd>
</dl>
<p>per microliter in the final reaction mix.
</p>
<dl>
<dt><code>nucleotide</code></dt><dd><p><a href="#topic+nucleotideType">nucleotideType</a>.</p>
</dd>
</dl>


<hr>
<h2 id='thermalCyclingConditionsType'>thermalCyclingConditionsType R6 class.</h2><span id='topic+thermalCyclingConditionsType'></span>

<h3>Description</h3>

<p>A cycling program for PCR or to amplify cDNA. Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>thermalCyclingConditionsType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>thermalCyclingConditionsType$new(id,
  description = NULL, documentation = NULL, lidTemperature = NULL,
  experimenter = NULL, step)</pre>


<h3>Fields</h3>

 <dl>
<dt><code>id</code></dt><dd><p><a href="#topic+idType">idType</a>.</p>
</dd>
<dt><code>description</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>.</p>
</dd>
<dt><code>documentation</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.</p>
</dd>
<dt><code>lidTemperature</code></dt><dd><p><a href="checkmate.html#topic+checkNumber">checkNumber</a>. The temperature in
degrees Celsius of the lid during cycling. </p>
</dd>
<dt><code>experimenter</code></dt><dd><p><code>list</code> of <a href="#topic+idReferencesType">idReferencesType</a>.
Reference to the person who made or uses this protocol. </p>
</dd>
<dt><code>step</code></dt><dd><p><code>list</code> of <a href="#topic+stepType">stepType</a>. The steps a protocol
runs through to amplify DNA.</p>
</dd>
</dl>


<hr>
<h2 id='xRefType'>xRefType R6 class.</h2><span id='topic+xRefType'></span>

<h3>Description</h3>

<p>Inherits: <a href="#topic+rdmlBaseType">rdmlBaseType</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xRefType
</code></pre>


<h3>Format</h3>

<p>An <code><a href="R6.html#topic+R6Class">R6Class</a></code> generator object.</p>


<h3>Initialization</h3>

<pre>xRefType$new(name = NULL, id = NULL)</pre>
<p>@section Fields: </p>

<dl>
<dt><code>name</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. Reference to an external
database, </p>
</dd></dl>
<p> for example &quot;GenBank&quot;.
</p>
<dl>
<dt><code>id</code></dt><dd><p><a href="checkmate.html#topic+checkString">checkString</a>. The ID of the entry within
the external database, for example &quot;AJ832138&quot;.</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
