<!DOCTYPE html><html><head><title>Help for package rDNAse</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rDNAse}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#diversity'><p>diversity function for make_kmer_vec</p></a></li>
<li><a href='#dnacheck'><p>Check if the DNA sequence are in the 4 default types</p></a></li>
<li><a href='#extrDAC'><p>The Dinucleotide-based Auto Covariance Descriptor</p></a></li>
<li><a href='#extrDACC'><p>The Dinucleotide-based Auto-cross Covariance Descriptor</p></a></li>
<li><a href='#extrDCC'><p>The Dinucleotide-based Cross Covariance Descriptor</p></a></li>
<li><a href='#extrPseDNC'><p>The Pseudo Dinucleotide Composition Descriptor</p></a></li>
<li><a href='#extrPseKNC'><p>The Pseudo K-tupler Composition Descriptor</p></a></li>
<li><a href='#extrTAC'><p>The Trinucleotide-based Auto Covariance Descriptor</p></a></li>
<li><a href='#extrTACC'><p>The Trinucleotide-based Auto-cross Covariance Descriptor</p></a></li>
<li><a href='#extrTCC'><p>The Trinucleotide-based Cross Covariance Descriptor</p></a></li>
<li><a href='#getGenbank'><p>Get DNA/RNA Sequences from Genbank by GI ID</p></a></li>
<li><a href='#id_x_s'><p>id_x_s function for make_kmer_vec</p></a></li>
<li><a href='#kmer'><p>The Basic Kmer Descriptor</p></a></li>
<li><a href='#make_idkmer_vec'><p>The Increment Of Diversity Descriptors</p></a></li>
<li><a href='#make_kmer_index'><p>Calculate The Basic Kmer Feature Vector</p></a></li>
<li><a href='#parGOSim'><p>DNA Sequence Similarity Calculation based on Gene Ontology (GO) Similarity</p></a></li>
<li><a href='#parSeqSim'><p>Parallellized DNA/RNA Sequence Similarity Calculation based on Sequence Alignment</p></a></li>
<li><a href='#readFASTA'><p>Read DNA/RNA Sequences in FASTA Format</p></a></li>
<li><a href='#revchars'><p>The Reverse chars</p></a></li>
<li><a href='#twoGOSim'><p>DNA Similarity Calculation based on Gene Ontology (GO) Similarity</p></a></li>
<li><a href='#twoSeqSim'><p>DNA/RNA Sequence Alignment for Two DNA/RNA Sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1-1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-6-24</td>
</tr>
<tr>
<td>Title:</td>
<td>Generating Various Numerical Representation Schemes of DNA
Sequences</td>
</tr>
<tr>
<td>Description:</td>
<td>Comprehensive toolkit for generating various numerical                                 representation schemes of DNA sequence. The descriptors and similarity
    scores included are extensively used in bioinformatics and chemogenomics.</td>
</tr>
<tr>
<td>Author:</td>
<td>Min-feng Zhu &lt;wind2zhu@163.com&gt;, Jie Dong &lt;biomed@csu.edu.cn&gt;,
        Dong-sheng Cao &lt;oriental-cds@163.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Min-feng Zhu &lt;wind2zhu@163.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/wind22zhu/rDNAse">https://github.com/wind22zhu/rDNAse</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/wind22zhu/rDNAse/issues">https://github.com/wind22zhu/rDNAse/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biostrings, GOSemSim, foreach, doParallel, RCurl,
org.Hs.eg.db</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-07-14 00:59:48 UTC; root</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-07-14 07:18:52</td>
</tr>
</table>
<hr>
<h2 id='diversity'>diversity function for make_kmer_vec</h2><span id='topic+diversity'></span>

<h3>Description</h3>

<p>diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diversity(vec)
</code></pre>


<h3>Value</h3>

<p>diversity
</p>

<hr>
<h2 id='dnacheck'>Check if the DNA sequence are in the 4 default types</h2><span id='topic+dnacheck'></span>

<h3>Description</h3>

<p>Check if the DNA sequence are in the 4 default types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dnacheck(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dnacheck_+3A_x">x</code></td>
<td>
<p>A character vector, as the input DNA sequence.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function checks if the DNA sequence types are in the 4.
</p>


<h3>Value</h3>

<p>Logical. <code>TRUE</code> if all of the DNA types of the sequence
are within the 4 default types.
</p>
<p>The result character vector
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
dnacheck(x) # TRUE
dnacheck(paste(x, 'Z', sep = '')) # FALSE
</code></pre>

<hr>
<h2 id='extrDAC'>The Dinucleotide-based Auto Covariance Descriptor</h2><span id='topic+extrDAC'></span>

<h3>Description</h3>

<p>The Dinucleotide-based Auto Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrDAC(x, index = c("Twist", "Tilt"), nlag = 2, normaliztion = FALSE,
  customprops = NULL, allprop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrDAC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrDAC_+3A_index">index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 38
different physicochemical indices (Table 1), which the users can choose.</p>
</td></tr>
<tr><td><code id="extrDAC_+3A_nlag">nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td></tr>
<tr><td><code id="extrDAC_+3A_normaliztion">normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrDAC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
<tr><td><code id="extrDAC_+3A_allprop">allprop</code></td>
<td>
<p>all the 38 physicochemical indices will be
employed to generate the feature vector. Its default value is False.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the dinucleotide-based auto covariance descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Dong Q, Zhou S, Guan J. A new taxonomy-based protein fold recognition approach based on
autocross-covariance transformation. <em>Bioinformatics</em>, 2009, 25(20): 2655-2662.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrDCC">extrDCC</a></code> and <code><a href="#topic+extrDACC">extrDACC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrDAC(x)
</code></pre>

<hr>
<h2 id='extrDACC'>The Dinucleotide-based Auto-cross Covariance Descriptor</h2><span id='topic+extrDACC'></span>

<h3>Description</h3>

<p>The Dinucleotide-based Auto-cross Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrDACC(x, index = c("Twist", "Tilt"), nlag = 2, normaliztion = FALSE,
  customprops = NULL, allprop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrDACC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrDACC_+3A_index">index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 38
different physicochemical indices (Table 1), which the users can choose.</p>
</td></tr>
<tr><td><code id="extrDACC_+3A_nlag">nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td></tr>
<tr><td><code id="extrDACC_+3A_normaliztion">normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrDACC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
<tr><td><code id="extrDACC_+3A_allprop">allprop</code></td>
<td>
<p>all the 38 physicochemical indices will be
employed to generate the feature vector. Its default value is False.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the dinucleotide-based auto-cross covariance descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Dong Q, Zhou S, Guan J. A new taxonomy-based protein fold recognition approach based on
autocross-covariance transformation. <em>Bioinformatics</em>, 2009, 25(20): 2655-2662.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrDAC">extrDAC</a></code> and <code><a href="#topic+extrDCC">extrDCC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrDACC(x)
</code></pre>

<hr>
<h2 id='extrDCC'>The Dinucleotide-based Cross Covariance Descriptor</h2><span id='topic+extrDCC'></span>

<h3>Description</h3>

<p>The Dinucleotide-based Cross Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrDCC(x, index = c("Twist", "Tilt"), nlag = 2, normaliztion = FALSE,
  customprops = NULL, allprop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrDCC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrDCC_+3A_index">index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 38
different physicochemical indices (Table 1), which the users can choose.</p>
</td></tr>
<tr><td><code id="extrDCC_+3A_nlag">nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td></tr>
<tr><td><code id="extrDCC_+3A_normaliztion">normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrDCC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
<tr><td><code id="extrDCC_+3A_allprop">allprop</code></td>
<td>
<p>all the 38 physicochemical indices will be
employed to generate the feature vector. Its default value is False.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the dinucleotide-based cross covariance descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Dong Q, Zhou S, Guan J. A new taxonomy-based protein fold recognition approach based on
autocross-covariance transformation. <em>Bioinformatics</em>, 2009, 25(20): 2655-2662.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrDAC">extrDAC</a></code> and <code><a href="#topic+extrDACC">extrDACC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrDCC(x)
</code></pre>

<hr>
<h2 id='extrPseDNC'>The Pseudo Dinucleotide Composition Descriptor</h2><span id='topic+extrPseDNC'></span>

<h3>Description</h3>

<p>The Pseudo Dinucleotide Composition Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrPseDNC(x, lambda = 3, w = 0.05, normalize = FALSE,
  customprops = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrPseDNC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrPseDNC_+3A_lambda">lambda</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length of the shortest
sequence in the dataset). It represents the highest counted rank (or tier) of the correlation along a
DNA sequence. Its default value is 3.</p>
</td></tr>
<tr><td><code id="extrPseDNC_+3A_w">w</code></td>
<td>
<p>the weight factor ranged from 0 to 1. Its default value is 0.05.</p>
</td></tr>
<tr><td><code id="extrPseDNC_+3A_normalize">normalize</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrPseDNC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the pseudo dinucleotide composition Descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Chen W, Feng P M, Lin H, et al. iRSpot-PseDNC: identify recombination spots with pseudo dinucleotide composition.
<em>Nucleic acids research</em>, 2013: gks1450.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrPseKNC">extrPseKNC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrPseDNC(x)
</code></pre>

<hr>
<h2 id='extrPseKNC'>The Pseudo K-tupler Composition Descriptor</h2><span id='topic+extrPseKNC'></span>

<h3>Description</h3>

<p>The Pseudo K-tupler Composition Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrPseKNC(x, lambda = 1, k = 3, normalize = FALSE, w = 0.5,
  customprops = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrPseKNC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrPseKNC_+3A_lambda">lambda</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length of the shortest
sequence in the dataset). It represents the highest counted rank (or tier) of the correlation along a
DNA sequence. Its default value is 3.</p>
</td></tr>
<tr><td><code id="extrPseKNC_+3A_k">k</code></td>
<td>
<p>an integer larger than 0 represents the k-tuple. Its default value is 3.</p>
</td></tr>
<tr><td><code id="extrPseKNC_+3A_normalize">normalize</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrPseKNC_+3A_w">w</code></td>
<td>
<p>the weight factor ranged from 0 to 1. Its default value is 0.05.</p>
</td></tr>
<tr><td><code id="extrPseKNC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the pseudo k-tupler composition Descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Guo S H, Deng E Z, Xu L Q, et al. iNuc-PseKNC: a sequence-based predictor for predicting nucleosome positioning
in genomes with pseudo k-tuple nucleotide composition. <em>Bioinformatics</em>, 2014: btu083.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrPseDNC">extrPseDNC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrPseKNC(x)
</code></pre>

<hr>
<h2 id='extrTAC'>The Trinucleotide-based Auto Covariance Descriptor</h2><span id='topic+extrTAC'></span>

<h3>Description</h3>

<p>The Trinucleotide-based Auto Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrTAC(x, index = c("Dnase I", "Nucleosome"), nlag = 2,
  normaliztion = FALSE, customprops = NULL, allprop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrTAC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrTAC_+3A_index">index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 12
different physicochemical indices (Table 2), which the users can choose.</p>
</td></tr>
<tr><td><code id="extrTAC_+3A_nlag">nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td></tr>
<tr><td><code id="extrTAC_+3A_normaliztion">normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrTAC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
<tr><td><code id="extrTAC_+3A_allprop">allprop</code></td>
<td>
<p>all the 12 physicochemical indices will be
employed to generate the feature vector. Its default value is False.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the trinucleotide-based auto covariance Descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrTCC">extrTCC</a></code> and <code><a href="#topic+extrTACC">extrTACC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrTAC(x)
</code></pre>

<hr>
<h2 id='extrTACC'>The Trinucleotide-based Auto-cross Covariance Descriptor</h2><span id='topic+extrTACC'></span>

<h3>Description</h3>

<p>The Trinucleotide-based Auto-cross Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrTACC(x, index = c("Dnase I", "Nucleosome"), nlag = 2,
  normaliztion = FALSE, customprops = NULL, allprop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrTACC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrTACC_+3A_index">index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 12
different physicochemical indices (Table 2), which the users can choose.</p>
</td></tr>
<tr><td><code id="extrTACC_+3A_nlag">nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td></tr>
<tr><td><code id="extrTACC_+3A_normaliztion">normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="extrTACC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
<tr><td><code id="extrTACC_+3A_allprop">allprop</code></td>
<td>
<p>all the 12 physicochemical indices will be
employed to generate the feature vector. Its default value is False.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the trinucleotide-based auto-cross covariance descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrTAC">extrTAC</a></code> and <code><a href="#topic+extrTCC">extrTCC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrTACC(x)
</code></pre>

<hr>
<h2 id='extrTCC'>The Trinucleotide-based Cross Covariance Descriptor</h2><span id='topic+extrTCC'></span>

<h3>Description</h3>

<p>The Trinucleotide-based Cross Covariance Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extrTCC(x, index = c("Dnase I", "Nucleosome"), nlag = 2,
  customprops = NULL, normaliztion = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extrTCC_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="extrTCC_+3A_index">index</code></td>
<td>
<p>the physicochemical indices, it should be a list and there are 12
different physicochemical indices (Table 2), which the users can choose.</p>
</td></tr>
<tr><td><code id="extrTCC_+3A_nlag">nlag</code></td>
<td>
<p>an integer larger than or equal to 0 and less than or equal to L-2 (L means the length
of the shortest DNA sequence in the dataset). It represents the distance between two dinucleotides.</p>
</td></tr>
<tr><td><code id="extrTCC_+3A_customprops">customprops</code></td>
<td>
<p>the users can use their own indices to generate the feature vector. It should be a dict,
the key is dinucleotide (string), and its corresponding value is a list type.</p>
</td></tr>
<tr><td><code id="extrTCC_+3A_normaliztion">normaliztion</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the trinucleotide-based cross covariance Descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+extrTAC">extrTAC</a></code> and <code><a href="#topic+extrTACC">extrTACC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
extrTCC(x)
</code></pre>

<hr>
<h2 id='getGenbank'>Get DNA/RNA Sequences from Genbank by GI ID</h2><span id='topic+getGenbank'></span>

<h3>Description</h3>

<p>Get DNA/RNA Sequences from Genbank by GI ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGenbank(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGenbank_+3A_id">id</code></td>
<td>
<p>A character vector, as the GI ID(s).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function get DNA/RNA sequences from Genbank by GI ID(s).
</p>


<h3>Value</h3>

<p>A list, each component contains one of the DNA/RNA sequences.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+readFASTA">readFASTA</a></code> for reading FASTA format files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Network latency may slow down this example
# Only test this when your connection is fast enough
require(RCurl)

ids = c(2, 11)
getGenbank(ids)
</code></pre>

<hr>
<h2 id='id_x_s'>id_x_s function for make_kmer_vec</h2><span id='topic+id_x_s'></span>

<h3>Description</h3>

<p>id_x_s
</p>


<h3>Usage</h3>

<pre><code class='language-R'>id_x_s(vec_x, vec_s, diversity_s)
</code></pre>


<h3>Value</h3>

<p>id
</p>

<hr>
<h2 id='kmer'>The Basic Kmer Descriptor</h2><span id='topic+kmer'></span>

<h3>Description</h3>

<p>The Basic Kmer Descriptor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmer(x, k = 2, upto = FALSE, normalize = FALSE, reverse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmer_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="kmer_+3A_k">k</code></td>
<td>
<p>the k value of kmer, it should be an integer larger than 0.</p>
</td></tr>
<tr><td><code id="kmer_+3A_upto">upto</code></td>
<td>
<p>generate all the kmers: 1mer, 2mer, ..., kmer. The output feature vector is
the combination of all these kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="kmer_+3A_normalize">normalize</code></td>
<td>
<p>with this option, the final feature vector will be normalized based
on the total occurrences of all kmers. Therefore, the elements in the feature vectors
represent the frequencies of kmers. The default value of this parameter is False.</p>
</td></tr>
<tr><td><code id="kmer_+3A_reverse">reverse</code></td>
<td>
<p>make reverse complements into a single feature, The default value of this parameter is False.
if reverse is True, this method returns the reverse compliment kmer feature vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the basic kmer descriptor
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the parameters normalize and upto are both True, and then the feature vector is
the combination of all these normalized kmers, e.g. the combination of normalized 1-kmer
and normalized 2-kmer when k=2, normalize=True, upto=True.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Noble W S, Kuehn S, Thurman R, et al. Predicting the in vivo signature of human gene regulatory sequences.
<em>Bioinformatics</em>, 2005, 21 Suppl 1, i338-343.
Lee D, Karchin R, Beer M A. Discriminative prediction of mammalian enhancers from
DNA sequence. <em>Genome research</em>. 2005, 21, 2167-2180.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+make_kmer_index">make_kmer_index</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
kmer(x)
</code></pre>

<hr>
<h2 id='make_idkmer_vec'>The Increment Of Diversity Descriptors</h2><span id='topic+IncDiv'></span><span id='topic+make_idkmer_vec'></span>

<h3>Description</h3>

<p>The Increment Of Diversity Descriptors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_idkmer_vec(k = 6, x, pos, neg, upto = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_idkmer_vec_+3A_k">k</code></td>
<td>
<p>the k value of kmer, it should be an integer larger than 0,the default value is 6.</p>
</td></tr>
<tr><td><code id="make_idkmer_vec_+3A_x">x</code></td>
<td>
<p>the input data, which should be a list or file type.</p>
</td></tr>
<tr><td><code id="make_idkmer_vec_+3A_pos">pos</code></td>
<td>
<p>the positive source data, which should be a or type.</p>
</td></tr>
<tr><td><code id="make_idkmer_vec_+3A_neg">neg</code></td>
<td>
<p>the negative source data, which should be or type.</p>
</td></tr>
<tr><td><code id="make_idkmer_vec_+3A_upto">upto</code></td>
<td>
<p>generate all the kmers: 1mer, 2mer, ..., kmer. The output feature vector is
the combination of all these kmers. The default value of this parameter is True</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the The Basic Kmer Descriptor
</p>


<h3>Value</h3>

<p>if upto is True, A length <code>k * 2</code> named vector, <code>k</code> is the k value of kmer;
if upto is False, A length 2 named vector
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Chen W, Luo L, Zhang L. The organization of nucleosomes around splice sites.
<em>Nucleic acids research</em>, 2010, 38(9): 2788-2798.
Liu G, Liu J, Cui X, et al. Sequence-dependent prediction of recombination hotspots in
Saccharomyces cerevisiae. <em>Journal of theoretical biology</em>, 2012, 293: 49-54.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+kmer">kmer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pos = readFASTA(system.file('dnaseq/pos.fasta', package = 'rDNAse'))
neg = readFASTA(system.file('dnaseq/neg.fasta', package = 'rDNAse'))
x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
make_idkmer_vec(k = 6, x, pos, neg)
</code></pre>

<hr>
<h2 id='make_kmer_index'>Calculate The Basic Kmer Feature Vector</h2><span id='topic+make_kmer_index'></span>

<h3>Description</h3>

<p>Calculate The Basic Kmer Feature Vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_kmer_index(k, alphabet = "ACGT")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_kmer_index_+3A_k">k</code></td>
<td>
<p>the k value of kmer, it should be an integer larger than 0.</p>
</td></tr>
<tr><td><code id="make_kmer_index_+3A_alphabet">alphabet</code></td>
<td>
<p>the</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculate the basic kmer feature vector.
</p>


<h3>Value</h3>

<p>The result character vector
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+kmer">kmer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>make_kmer_index(2, alphabet = "ACGT")
</code></pre>

<hr>
<h2 id='parGOSim'>DNA Sequence Similarity Calculation based on Gene Ontology (GO) Similarity</h2><span id='topic+parGOSim'></span>

<h3>Description</h3>

<p>DNA Sequence Similarity Calculation based on Gene Ontology (GO) Similarity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parGOSim(golist, type = c("go", "gene"), ont = "MF", organism = "human",
  measure = "Resnik", combine = "BMA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parGOSim_+3A_golist">golist</code></td>
<td>
<p>A character vector, each component contains
a character vector of GO terms or one Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_type">type</code></td>
<td>
<p>Input type of <code>golist</code>, <code>'go'</code> for GO Terms, <code>'gene'</code> for gene ID.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_ont">ont</code></td>
<td>
<p>Default is <code>'MF'</code>, could be one of <code>'MF'</code>, <code>'BP'</code>, or <code>'CC'</code> subontologies.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_organism">organism</code></td>
<td>
<p>Default is <code>'human'</code>, could be one of <code>'anopheles'</code>, <code>'arabidopsis'</code>, <code>'bovine'</code>, <code>'canine'</code>,
<code>'chicken'</code>, <code>'chimp'</code>, <code>'coelicolor'</code>, <code>'ecolik12'</code>,
<code>'ecsakai'</code>, <code>'fly'</code>, <code>'human'</code>, <code>'malaria'</code>,
<code>'mouse'</code>, <code>'pig'</code>, <code>'rat'</code>, <code>'rhesus'</code>,
<code>'worm'</code>, <code>'xenopus'</code>, <code>'yeast'</code> or <code>'zebrafish'</code>.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_measure">measure</code></td>
<td>
<p>Default is <code>'Resnik'</code>, could be one of <code>'Resnik'</code>, <code>'Lin'</code>, <code>'Rel'</code>, <code>'Jiang'</code> or <code>'Wang'</code>.</p>
</td></tr>
<tr><td><code id="parGOSim_+3A_combine">combine</code></td>
<td>
<p>Default is <code>'BMA'</code>, could be one of <code>'max'</code>, <code>'average'</code>, <code>'rcmax'</code> or <code>'BMA'</code>
for combining semantic similarity scores of multiple GO terms associated with DNA.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates
DNA sequence similarity based on Gene Ontology (GO) similarity.
</p>


<h3>Value</h3>

<p>A <code>n</code> x <code>n</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu  &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+twoGOSim">twoGOSim</a></code> for calculating the
GO semantic similarity between two groups of GO terms or two Entrez gene IDs.
See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for paralleled DNA similarity
calculation based on Smith-Waterman local alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Be careful when testing this since it involves GO similarity computation
# and might produce unpredictable results in some environments

require(GOSemSim)
require(org.Hs.eg.db)

# by GO Terms
go1 = c('GO:0005215', 'GO:0005488', 'GO:0005515', 'GO:0005625', 'GO:0005802', 'GO:0005905')  # AP4B1
go2 = c('GO:0005515', 'GO:0005634', 'GO:0005681', 'GO:0008380', 'GO:0031202')  # BCAS2
go3 = c('GO:0003735', 'GO:0005622', 'GO:0005840', 'GO:0006412')  # PDE4DIP
glist = list(go1, go2, go3)
gsimmat1 = parGOSim(glist, type = 'go', ont = 'CC')
print(gsimmat1)

# by Entrez gene id
genelist = list(c('150', '151', '152', '1814', '1815', '1816'))
gsimmat2 = parGOSim(genelist, type = 'gene')
print(gsimmat2)
## End(Not run)
</code></pre>

<hr>
<h2 id='parSeqSim'>Parallellized DNA/RNA Sequence Similarity Calculation based on Sequence Alignment</h2><span id='topic+parSeqSim'></span>

<h3>Description</h3>

<p>Parallellized DNA/RNA Sequence Similarity Calculation based on Sequence Alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parSeqSim(dnalist, cores = 2, type = "local", submat = "BLOSUM62")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parSeqSim_+3A_dnalist">dnalist</code></td>
<td>
<p>A length <code>n</code> list containing <code>n</code> DNA/RNA sequences,
each component of the list is a character string, storing one DNA/RNA sequence.
Unknown sequences should be represented as <code>''</code>.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_cores">cores</code></td>
<td>
<p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users could use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_type">type</code></td>
<td>
<p>Type of alignment, default is <code>'local'</code>,
could be <code>'global'</code> or <code>'local'</code>,
where <code>'global'</code> represents Needleman-Wunsch global alignment;
<code>'local'</code> represents Smith-Waterman local alignment.</p>
</td></tr>
<tr><td><code id="parSeqSim_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix, default is <code>'BLOSUM62'</code>, could be one of
<code>'BLOSUM45'</code>, <code>'BLOSUM50'</code>, <code>'BLOSUM62'</code>, <code>'BLOSUM80'</code>, <code>'BLOSUM100'</code>,
<code>'PAM30'</code>, <code>'PAM40'</code>, <code>'PAM70'</code>, <code>'PAM120'</code>, <code>'PAM250'</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implemented the parallellized version for calculating
DNA/RNA sequence similarity based on sequence alignment.
</p>


<h3>Value</h3>

<p>A <code>n</code> x <code>n</code> similarity matrix.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu  &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code>twoSeqSim</code> for DNA/RNA sequence alignment
for two DNA/RNA sequences. See <code><a href="#topic+parGOSim">parGOSim</a></code> for
DNA/RNA similarity calculation based on
Gene Ontology (GO) semantic similarity.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Be careful when testing this since it involves parallelisation
# and might produce unpredictable results in some environments

require(Biostrings)
require(foreach)
require(doParallel)

s1 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[1]]
s2 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[2]]
s3 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[3]]
s4 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[4]]
s5 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[5]]
plist = list(s1, s2, s3, s4, s5)
psimmat = parSeqSim(plist, cores = 2, type = 'local', submat = 'BLOSUM62')
print(psimmat)
</code></pre>

<hr>
<h2 id='readFASTA'>Read DNA/RNA Sequences in FASTA Format</h2><span id='topic+readFASTA'></span>

<h3>Description</h3>

<p>Read DNA/RNA Sequences in FASTA Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readFASTA(file, legacy.mode = TRUE, seqonly = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readFASTA_+3A_file">file</code></td>
<td>
<p>The name of the file which the sequences in fasta format are
to be read from. If it does not contain an absolute or
relative path, the file name is relative to the current
working directory, <code><a href="base.html#topic+getwd">getwd</a></code>.
The default here is to read the <code>example.fasta</code> file which
is present in the <code>protseq</code> directory of the protr package.</p>
</td></tr>
<tr><td><code id="readFASTA_+3A_legacy.mode">legacy.mode</code></td>
<td>
<p>If set to <code>TRUE</code>, lines starting with a semicolon ';'
are ignored. Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="readFASTA_+3A_seqonly">seqonly</code></td>
<td>
<p>If set to <code>TRUE</code>, only sequences as returned without
attempt to modify them or to get their names and
annotations (execution time is divided approximately
by a factor 3). Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reads DNA/RNA sequences in FASTA format.
</p>


<h3>Value</h3>

<p>The result character vector
</p>


<h3>Note</h3>

<p>Note
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>References</h3>

<p>Pearson, W.R. and Lipman, D.J. (1988)
Improved tools for biological sequence comparison.
<em>Proceedings of the National Academy of Sciences
of the United States of America</em>, <b>85</b>: 2444-2448
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNAse'))
</code></pre>

<hr>
<h2 id='revchars'>The Reverse chars</h2><span id='topic+revchars'></span>

<h3>Description</h3>

<p>The Reverse chars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>revchars(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="revchars_+3A_x">x</code></td>
<td>
<p>the input data, which should be a string.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates Reverse chars
</p>


<h3>Value</h3>

<p>A vector
</p>


<h3>Note</h3>

<p>if the user defined physicochemical indices have not been normalized, it should be normalized.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 'GACTGAACTGCACTTTGGTTTCATATTATTTGCTC'
revchars(x)
</code></pre>

<hr>
<h2 id='twoGOSim'>DNA Similarity Calculation based on Gene Ontology (GO) Similarity</h2><span id='topic+twoGOSim'></span>

<h3>Description</h3>

<p>DNA Similarity Calculation based on Gene Ontology (GO) Similarity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoGOSim(id1, id2, type = c("go", "gene"), ont = "MF", organism = "human",
  measure = "Resnik", combine = "BMA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoGOSim_+3A_id1">id1</code></td>
<td>
<p>A character vector. length &gt; 1: each element is a GO term;
length = 1: the Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_id2">id2</code></td>
<td>
<p>A character vector. length &gt; 1: each element is a GO term;
length = 1: the Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_type">type</code></td>
<td>
<p>Input type of id1 and id2, <code>'go'</code> for GO Terms, <code>'gene'</code> for gene ID.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_ont">ont</code></td>
<td>
<p>Default is <code>'MF'</code>, could be one of <code>'MF'</code>, <code>'BP'</code>, or <code>'CC'</code> subontologies.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_organism">organism</code></td>
<td>
<p>Default is <code>'human'</code>, could be one of <code>'anopheles'</code>, <code>'arabidopsis'</code>, <code>'bovine'</code>, <code>'canine'</code>,
<code>'chicken'</code>, <code>'chimp'</code>, <code>'coelicolor'</code>, <code>'ecolik12'</code>,
<code>'ecsakai'</code>, <code>'fly'</code>, <code>'human'</code>, <code>'malaria'</code>,
<code>'mouse'</code>, <code>'pig'</code>, <code>'rat'</code>, <code>'rhesus'</code>,
<code>'worm'</code>, <code>'xenopus'</code>, <code>'yeast'</code> or <code>'zebrafish'</code>.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_measure">measure</code></td>
<td>
<p>Default is <code>'Resnik'</code>, could be one of <code>'Resnik'</code>, <code>'Lin'</code>, <code>'Rel'</code>, <code>'Jiang'</code> or <code>'Wang'</code>.</p>
</td></tr>
<tr><td><code id="twoGOSim_+3A_combine">combine</code></td>
<td>
<p>Default is <code>'BMA'</code>, could be one of <code>'max'</code>, <code>'average'</code>, <code>'rcmax'</code> or <code>'BMA'</code>
for combining semantic similarity scores of multiple GO terms associated with DNA.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the Gene Ontology (GO) similarity
between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Value</h3>

<p>A n x n matrix.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu  &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+parGOSim">parGOSim</a></code> for
DNA similarity calculation based on
Gene Ontology (GO) semantic similarity.
See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for paralleled DNA similarity
calculation based on Smith-Waterman local alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Be careful when testing this since it involves GO similarity computation
# and might produce unpredictable results in some environments

require(GOSemSim)
require(org.Hs.eg.db)

# by GO terms
go1 = c("GO:0004022", "GO:0004024", "GO:0004023")
go2 = c("GO:0009055", "GO:0020037")
gsim1 = twoGOSim(go1, go2, type = 'go', ont = 'MF', measure = 'Wang')
print(gsim1)

# by Entrez gene id
gene1 = '241'
gene2 = '251'
gsim2 = twoGOSim(gene1, gene2, type = 'gene', ont = 'BP', measure = 'Lin')
print(gsim2)
## End(Not run)
</code></pre>

<hr>
<h2 id='twoSeqSim'>DNA/RNA Sequence Alignment for Two DNA/RNA Sequences</h2><span id='topic+twoSeqSim'></span>

<h3>Description</h3>

<p>DNA/RNA Sequence Alignment for Two DNA/RNA Sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoSeqSim(seq1, seq2, type = "local", submat = "BLOSUM62")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoSeqSim_+3A_seq1">seq1</code></td>
<td>
<p>A character string, containing one DNA/RNA sequence.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_seq2">seq2</code></td>
<td>
<p>A character string, containing another DNA/RNA sequence.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_type">type</code></td>
<td>
<p>Type of alignment, default is <code>'local'</code>,
could be <code>'global'</code> or <code>'local'</code>,
where <code>'global'</code> represents Needleman-Wunsch global alignment;
<code>'local'</code> represents Smith-Waterman local alignment.</p>
</td></tr>
<tr><td><code id="twoSeqSim_+3A_submat">submat</code></td>
<td>
<p>Substitution matrix, default is <code>'BLOSUM62'</code>, could be one of
<code>'BLOSUM45'</code>, <code>'BLOSUM50'</code>, <code>'BLOSUM62'</code>, <code>'BLOSUM80'</code>, <code>'BLOSUM100'</code>,
<code>'PAM30'</code>, <code>'PAM40'</code>, <code>'PAM70'</code>, <code>'PAM120'</code>, <code>'PAM250'</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the sequence alignment between two DNA/RNA sequences.
</p>


<h3>Value</h3>

<p>An Biostrings object containing the scores and other alignment information.
</p>


<h3>Author(s)</h3>

<p>Min-feng Zhu  &lt;<a href="mailto:wind2zhu@163.com">wind2zhu@163.com</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+parSeqSim">parSeqSim</a></code> for paralleled pairwise
DNA/RNA similarity calculation based on sequence alignment.
See <code><a href="#topic+twoGOSim">twoGOSim</a></code> for calculating the
GO semantic similarity between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Be careful when testing this since it involves sequence alignment
# and might produce unpredictable results in some environments

require(Biostrings)

s1 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[1]]
s2 = readFASTA(system.file('dnaseq/hs.fasta', package = 'rDNA'))[[2]]
seqalign = twoSeqSim(s1, s2)
summary(seqalign)
print(seqalign@score)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
