<!DOCTYPE html><html><head><title>Help for package ReAD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ReAD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#em_hmm'><p>EM algorithm in combination with a non-parametric algorithm for estimation of the rLIS statistic.</p></a></li>
<li><a href='#ReAD'><p>Replicability analysis across two genome-wide association studies accounting for the linkage disequilibrium structure.</p></a></li>
<li><a href='#SimuData'><p>Simulate two sequences of p-values by accounting for the local dependence structure via a hidden Markov model.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Powerful Replicability Analysis of Genome-Wide Association
Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A robust and powerful approach is developed for replicability analysis of two Genome-wide association studies (GWASs) accounting for the linkage disequilibrium (LD) among genetic variants. The LD structure in two GWASs is captured by a four-state hidden Markov model (HMM). The unknowns involved in the HMM are estimated by an efficient expectation-maximization (EM) algorithm in combination with a non-parametric estimation of functions. By incorporating information from adjacent locations via the HMM, this approach identifies the entire clusters of genotype-phenotype associated signals, improving the power of replicability analysis while effectively controlling the false discovery rate.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>Rcpp (&ge; 1.0.10), qvalue</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-28 10:12:42 UTC; P53</td>
</tr>
<tr>
<td>Author:</td>
<td>Yan Li [aut, cre, cph],
  Haochen lei [aut],
  Xiaoquan Wen [aut],
  Hongyuan Cao [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yan Li &lt;yanli_@jlu.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-30 11:00:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='em_hmm'>EM algorithm in combination with a non-parametric algorithm for estimation of the rLIS statistic.</h2><span id='topic+em_hmm'></span>

<h3>Description</h3>

<p>Estimate the rLIS values accounting for the linkage disequilibrium across two genome-wide association studies via the four-state hidden Markov model. Apply a step-up procedure to control the FDR of replicability null.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>em_hmm(pa_in, pb_in, pi0a_in, pi0b_in)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="em_hmm_+3A_pa_in">pa_in</code></td>
<td>
<p>A numeric vector of p-values from study 1.</p>
</td></tr>
<tr><td><code id="em_hmm_+3A_pb_in">pb_in</code></td>
<td>
<p>A numeric vector of p-values from study 2.</p>
</td></tr>
<tr><td><code id="em_hmm_+3A_pi0a_in">pi0a_in</code></td>
<td>
<p>An initial estimate of the null probability in study 1.</p>
</td></tr>
<tr><td><code id="em_hmm_+3A_pi0b_in">pi0b_in</code></td>
<td>
<p>An initial estimate of the null probability in study 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>rLIS</code></td>
<td>
<p>The estimated rLIS for replicability null.</p>
</td></tr>
<tr><td><code>fdr</code></td>
<td>
<p>The adjusted values based on rLIS for FDR control.</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>
<p>The log-likelihood value with converged estimates of the unknowns.</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>An estimate of the stationary probabilities of four states (0,0), (0,1), (1,0), (1,1).</p>
</td></tr>
<tr><td><code>A</code></td>
<td>
<p>An estimate of the 4-by-4 transition matrix.</p>
</td></tr>
<tr><td><code>f1</code></td>
<td>
<p>A non-parametric estimate for the non-null probability density function in study 1.</p>
</td></tr>
<tr><td><code>f2</code></td>
<td>
<p>A non-parametric estimate for the non-null probability density function in study 2.</p>
</td></tr>
</table>

<hr>
<h2 id='ReAD'>Replicability analysis across two genome-wide association studies accounting for the linkage disequilibrium structure.</h2><span id='topic+ReAD'></span>

<h3>Description</h3>

<p>Replicability analysis across two genome-wide association studies accounting for the linkage disequilibrium structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReAD(pa, pb)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReAD_+3A_pa">pa</code></td>
<td>
<p>A numeric vector of p-values from study 1.</p>
</td></tr>
<tr><td><code id="ReAD_+3A_pb">pb</code></td>
<td>
<p>A numeric vector of p-values from study 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list:
</p>
<table>
<tr><td><code>rLIS</code></td>
<td>
<p>The estimated rLIS for replicability null.</p>
</td></tr>
<tr><td><code>fdr</code></td>
<td>
<p>The adjusted values based on rLIS for FDR control.</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>
<p>The log-likelihood value with converged estimates of the unknowns.</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>An estimate of the stationary probabilities of four states (0,0), (0,1), (1,0), (1,1).</p>
</td></tr>
<tr><td><code>A</code></td>
<td>
<p>An estimate of the 4-by-4 transition matrix.</p>
</td></tr>
<tr><td><code>f1</code></td>
<td>
<p>A non-parametric estimate for the non-null probability density function in study 1.</p>
</td></tr>
<tr><td><code>f2</code></td>
<td>
<p>A non-parametric estimate for the non-null probability density function in study 2.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate p-values in two studies locally dependent via a four-state hidden Markov model
data &lt;- SimuData(J = 10000)
p1 = data$pa; p2 = data$pb; theta1 = data$theta1; theta2 = data$theta2
# Run ReAD to identify replicable signals
res.read = ReAD(p1, p2)
sig.idx = which(res.read$fdr &lt;= 0.05)

</code></pre>

<hr>
<h2 id='SimuData'>Simulate two sequences of p-values by accounting for the local dependence structure via a hidden Markov model.</h2><span id='topic+SimuData'></span>

<h3>Description</h3>

<p>Simulate two sequences of p-values by accounting for the local dependence structure via a hidden Markov model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SimuData(
  J = 10000,
  pi = c(0.25, 0.25, 0.25, 0.25),
  A = 0.6 * diag(4) + 0.1,
  muA = 2,
  muB = 2,
  sdA = 1,
  sdB = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SimuData_+3A_j">J</code></td>
<td>
<p>The number of features to be tested in two studies.</p>
</td></tr>
<tr><td><code id="SimuData_+3A_pi">pi</code></td>
<td>
<p>The stationary probabilities of four hidden joint states.</p>
</td></tr>
<tr><td><code id="SimuData_+3A_a">A</code></td>
<td>
<p>The 4-by-4 transition matrix.</p>
</td></tr>
<tr><td><code id="SimuData_+3A_mua">muA</code></td>
<td>
<p>Mean of the normal distribution generating the p-value in study 1.</p>
</td></tr>
<tr><td><code id="SimuData_+3A_mub">muB</code></td>
<td>
<p>Mean of the normal distribution generating the p-value in study 2.</p>
</td></tr>
<tr><td><code id="SimuData_+3A_sda">sdA</code></td>
<td>
<p>The standard deviation of the normal distribution generating the p-value in study 1.</p>
</td></tr>
<tr><td><code id="SimuData_+3A_sdb">sdB</code></td>
<td>
<p>The standard deviation of the normal distribution generating the p-value in study 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list:
</p>
<table>
<tr><td><code>pa</code></td>
<td>
<p>A numeric vector of p-values from study 1.</p>
</td></tr>
<tr><td><code>pb</code></td>
<td>
<p>A numeric vector of p-values from study 2.</p>
</td></tr>
<tr><td><code>theta1</code></td>
<td>
<p>The true states of features in study 1.</p>
</td></tr>
<tr><td><code>theta2</code></td>
<td>
<p>The true states of features in study 2.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
