<!DOCTYPE html><html><head><title>Help for package read.gb</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {read.gb}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#extract.gb'><p>Extracts and returns a specific item from .gb records</p></a></li>
<li><a href='#read.gb'><p>Opens files with .gb extensions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Open GenBank Files</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Opens complete record(s) with .gb extension from the NCBI/GenBank Nucleotide database and returns a list containing shaped record(s). These kind of files contains detailed records of DNA samples (locus, organism, type of sequence, source of the sequence...). An example of record can be found at <a href="https://www.ncbi.nlm.nih.gov/nuccore/HE799070">https://www.ncbi.nlm.nih.gov/nuccore/HE799070</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Collate:</td>
<td>'CDS.F.R' 'C_region.F.R' 'DNA.treatment.R' 'D_loop.F.R'
'D_segment.F.R' 'F_UTR.F.R' 'Feature.sep.R' 'J_segment.F.R'
'N_region.F.R' 'Reference.sep.R' 'Reorganize.report.R'
'STS.F.R' 'S_region.F.R' 'T_UTR.F.R' 'V_region.F.R'
'V_segment.F.R' 'assembly_gap.F.R' 'centromere.F.R' 'exon.F.R'
'extract.gb.R' 'gap.F.R' 'gene.F.R' 'iDNA.F.R' 'intron.F.R'
'mRNA.F.R' 'mat_peptide.F.R' 'misc_RNA.F.R' 'misc_binding.F.R'
'misc_difference.F.R' 'misc_feature.F.R' 'misc_recomb.F.R'
'misc_structure.F.R' 'mobile_element.F.R' 'modified_base.F.R'
'ncRNA.F.R' 'old_sequence.F.R' 'operon.F.R' 'oriT.F.R'
'polyA_site.F.R' 'precursor_RNA.F.R' 'prim_transcript.F.R'
'primer_bind.F.R' 'propeptide.F.R' 'protein_bind.F.R'
'rRNA.F.R' 'read.gb.R' 'regulatory.F.R' 'rep_origin.F.R'
'repeat_region.F.R' 'sig_peptide.F.R' 'source.F.R'
'stem_loop.F.R' 'tRNA.F.R' 'telomere.F.R' 'tmRNA.F.R'
'transit_peptide.F.R' 'unsure.F.R' 'variation.F.R'</td>
</tr>
<tr>
<td>Imports:</td>
<td>rentrez</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-24 14:12:21 UTC; robin</td>
</tr>
<tr>
<td>Author:</td>
<td>Robin Mercier [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Robin Mercier &lt;robin.largon.mercier@hotmail.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-06 15:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='extract.gb'>Extracts and returns a specific item from .gb records</h2><span id='topic+extract.gb'></span>

<h3>Description</h3>

<p>This function regroup every items found in a list if they correspond to the character string specified. Works for general parts (ACCESSION, SOURCE...) and parts of FEATURES (source, gene, CDS...)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract.gb(x, Item)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract.gb_+3A_x">x</code></td>
<td>
<p>The name of the list containing the records.</p>
</td></tr>
<tr><td><code id="extract.gb_+3A_item">Item</code></td>
<td>
<p>character. The part to extract. It have to match case.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing the data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
extract.gb(Example, "gene")

## End(Not run)

</code></pre>

<hr>
<h2 id='read.gb'>Opens files with .gb extensions</h2><span id='topic+read.gb'></span>

<h3>Description</h3>

<p>This function opens complete record(s) with .gb extension from the NCBI/GenBank Nucleotide database and returns a list containing shaped record(s). These kind of files contains detailed records of DNA samples (locus, organism, type of sequence, source of the sequence...). An example of record can be found at <a href="https://www.ncbi.nlm.nih.gov/nuccore/HE799070">https://www.ncbi.nlm.nih.gov/nuccore/HE799070</a>. Records with &gt; 200'000 bp may experience long processing times, especially if they have numerous FEATURES items. Also works for reports obtained with rentrez package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.gb(x, DNA = TRUE, Type = "full", Source = "File")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.gb_+3A_x">x</code></td>
<td>
<p>character. The name of the file which the data are to be read from, or the character string containing the data. It can contains several records</p>
</td></tr>
<tr><td><code id="read.gb_+3A_dna">DNA</code></td>
<td>
<p>logical. If TRUE, the DNA sequence in the ORIGIN part will be merged in one character string. If FALSE, the default layout will be kept. Default if TRUE</p>
</td></tr>
<tr><td><code id="read.gb_+3A_type">Type</code></td>
<td>
<p>character. Should the output contain FEATURES and REFERENCE parts ? Possible values are &quot;full&quot; for a full record, &quot;nofeat&quot; to ignore FEATURES part, &quot;noref&quot; to ignore REFERENCE part and &quot;nfnr&quot; to ignore both parts. Default is &quot;full&quot;</p>
</td></tr>
<tr><td><code id="read.gb_+3A_source">Source</code></td>
<td>
<p>character. If x is a character string, use &quot;Char&quot;. If x is a filename use &quot;File&quot;. Default is &quot;File&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a table containing the data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
read.gb(File = "sequence.gb", DNA = TRUE, Type = "full", Source = "File")

## End(Not run)

## Not run: 
require(rentrez)
data &lt;- rentrez::entrez_fetch(db = "Nucleotide", id = "508082122", rettype = "gb")
read.gb(File = data, DNA = TRUE, Type = "full", Source = "Char")

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
