<!DOCTYPE html><html><head><title>Help for package Rediscover</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Rediscover}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#A_example'><p>A_example data</p></a></li>
<li><a href='#A_Matrix'><p>A_Matrix data</p></a></li>
<li><a href='#AMP_COAD'><p>AMP_COAD data</p></a></li>
<li><a href='#B_example'><p>B_example data</p></a></li>
<li><a href='#B_Matrix'><p>B_Matrix data</p></a></li>
<li><a href='#discoversomaticInteractions'><p>discoversomaticInteractions</p></a></li>
<li><a href='#getMutex'><p>getMutex function</p></a></li>
<li><a href='#getMutexAB'><p>getMutexAB function</p></a></li>
<li><a href='#getMutexGroup'><p>getMutexGroup function</p></a></li>
<li><a href='#getPM'><p>getPM function</p></a></li>
<li><a href='#InternalFunctions'><p>Rediscover Internal Functions</p></a></li>
<li><a href='#PM_AMP_COAD'><p>PM_AMP_COAD data</p></a></li>
<li><a href='#PM_COAD'><p>PM_COAD data</p></a></li>
<li><a href='#PMatrix-class'><p>An S4 class to store the probabilities</p></a></li>
<li><a href='#TCGA_COAD'><p>TCGA_COAD</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identify Mutually Exclusive Mutations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Juan A. Ferrer-Bonsoms, Laura Jareno, and Angel Rubio</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juan A. Ferrer-Bonsoms &lt;jafhernandez@tecnun.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An optimized method for identifying mutually 
    exclusive genomic events.  Its main contribution is a
    statistical analysis based on the Poisson-Binomial 
    distribution that takes into account that some samples
    are more mutated than others. See [Canisius, Sander, John WM Martens,
    and  Lodewyk FA Wessels. (2016)  "A novel independence test for
    somatic alterations in cancer shows that 
    biology drives mutual exclusivity but chance explains
    most co-occurrence." Genome biology 17.1 : 1-17. &lt;<a href="https://doi.org/10.1186%2Fs13059-016-1114-x">doi:10.1186/s13059-016-1114-x</a>&gt;].
    The mutations matrices are sparse matrices. The method developed takes 
    advantage of the advantages of this type of matrix to save 
    time and computing resources.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), Matrix, PoissonBinomial, ShiftConvolvePoibin,
utils, matrixStats</td>
</tr>
<tr>
<td>Imports:</td>
<td>maftools, data.table, parallel, RColorBrewer, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, RUnit, BiocStyle, BiocGenerics, dplyr,
kableExtra, magick, stats, qvalue</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>biocViews:</td>
<td>mutex</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-14 10:35:54 UTC; jafhernandez</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-14 12:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='A_example'>A_example data</h2><span id='topic+A_example'></span>

<h3>Description</h3>

<p>A binary matrix of class <code>matrix</code> used as toy example in <code>getPM</code> and <code>getMutex</code> and <code>getMutexAB</code> and <code>getMutexGroup</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("A_example")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:1000, 1:500] 0 0 0 0 0 1 0 0 0 1 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(A_example)
</code></pre>

<hr>
<h2 id='A_Matrix'>A_Matrix data</h2><span id='topic+A_Matrix'></span>

<h3>Description</h3>

<p>A binary dgCMatrix matrix used as toy example in <code>getPM</code> and <code>getMutex</code> and <code>getMutexAB</code> and <code>getMutexGroup</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("A_Matrix")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>Formal class 'dgCMatrix' [package &quot;Matrix&quot;] with 6 slots
</p>
<p>..@ i       : int [1:249838] 5 9 10 11 13 14 18 20 23 24 ...
</p>
<p>..@ p       : int [1:501] 0 503 1010 1506 1995 2497 2981 3488 4002 4474 ...
</p>
<p>..@ Dim     : int [1:2] 1000 500
</p>
<p>..@ Dimnames:List of 2
</p>
<p>.. ..$ : NULL
</p>
<p>.. ..$ : NULL
</p>
<p>..@ x       : num [1:249838] 1 1 1 1 1 1 1 1 1 1 ...
</p>
<p>..@ factors : list()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(A_Matrix)
</code></pre>

<hr>
<h2 id='AMP_COAD'>AMP_COAD data</h2><span id='topic+AMP_COAD'></span>

<h3>Description</h3>

<p>A binary matrix, with information about amplifications in Colon Adenocarcinoma, created by applying <code>GDCquery</code> and used as real example in <code>getMutexAB</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("AMP_COAD")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>num [1:1000, 1:391] 0 0 0 0 0 0 0 0 0 0 ...
</p>
<p>- attr(*, &quot;dimnames&quot;)=List of 2
</p>
<p>..$ : chr [1:1000] &quot;ENSG00000212993.4&quot; &quot;ENSG00000279524.1&quot; &quot;ENSG00000136997.13&quot; &quot;ENSG00000101294.15&quot; ...
</p>
<p>..$ : chr [1:391] &quot;TCGA-CA-6718&quot; &quot;TCGA-D5-6931&quot; &quot;TCGA-AZ-6601&quot; &quot;TCGA-G4-6320&quot; ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AMP_COAD)
## maybe str(AMP_COAD) 
</code></pre>

<hr>
<h2 id='B_example'>B_example data</h2><span id='topic+B_example'></span>

<h3>Description</h3>

<p>A binary matrix of class <code>matrix</code> used as toy example in <code>getPM</code> and <code>getMutex</code> and <code>getMutexAB</code> and <code>getMutexGroup</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("B_example")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>int [1:1000, 1:500] 0 1 1 0 1 0 0 0 1 1 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(B_example)
</code></pre>

<hr>
<h2 id='B_Matrix'>B_Matrix data</h2><span id='topic+B_Matrix'></span>

<h3>Description</h3>

<p>A binary dgCMatrix matrix used as toy example in <code>getPM</code> and <code>getMutex</code> and <code>getMutexAB</code> and <code>getMutexGroup</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("B_Matrix")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>Formal class 'dgCMatrix' [package &quot;Matrix&quot;] with 6 slots
</p>
<p>..@ i       : int [1:249526] 1 2 4 8 9 11 13 15 18 20 ...
</p>
<p>..@ p       : int [1:501] 0 498 1014 1527 2048 2558 3036 3511 4035 4537 ...
</p>
<p>..@ Dim     : int [1:2] 1000 500
</p>
<p>..@ Dimnames:List of 2
</p>
<p>.. ..$ : NULL
</p>
<p>.. ..$ : NULL
</p>
<p>..@ x       : num [1:249526] 1 1 1 1 1 1 1 1 1 1 ...
</p>
<p>..@ factors : list()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(B_Matrix)
</code></pre>

<hr>
<h2 id='discoversomaticInteractions'>discoversomaticInteractions</h2><span id='topic+discoversomaticInteractions'></span>

<h3>Description</h3>

<p>Function adapted to maftools where given a .maf file, it graphs the somatic interactions between a group of genes, i.e., the combination of gene expression and mutation data to detect mutually exclusive or co-ocurring events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>discoversomaticInteractions(
  maf,
  top = 25,
  genes = NULL,
  pvalue = c(0.05, 0.01),
  getMutexMethod = "ShiftedBinomial",
  getMutexMixed = TRUE,
  returnAll = TRUE,
  geneOrder = NULL,
  fontSize = 0.8,
  showSigSymbols = TRUE,
  showCounts = FALSE,
  countStats = "all",
  countType = "all",
  countsFontSize = 0.8,
  countsFontColor = "black",
  colPal = "BrBG",
  showSum = TRUE,
  colNC = 9,
  nShiftSymbols = 5,
  sigSymbolsSize = 2,
  sigSymbolsFontSize = 0.9,
  pvSymbols = c(46, 42),
  limitColorBreaks = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="discoversomaticInteractions_+3A_maf">maf</code></td>
<td>
<p>maf object generated by read.maf</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_top">top</code></td>
<td>
<p>check for interactions among top 'n' number of genes. Defaults to top 25. genes</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_genes">genes</code></td>
<td>
<p>List of genes among which interactions should be tested. If not provided, test will be performed between top 25 genes.</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_pvalue">pvalue</code></td>
<td>
<p>Default c(0.05, 0.01) p-value threshold. You can provide two values for upper and lower threshold.</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_getmutexmethod">getMutexMethod</code></td>
<td>
<p>Method for the 'getMutex' function (by default &quot;ShiftedBinomial&quot;)</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_getmutexmixed">getMutexMixed</code></td>
<td>
<p>Mixed parameter for the 'getMutex' function (by default TRUE)</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_returnall">returnAll</code></td>
<td>
<p>If TRUE returns test statistics for all pair of tested genes. Default FALSE, returns for only genes below pvalue threshold.</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_geneorder">geneOrder</code></td>
<td>
<p>Plot the results in given order. Default NULL.</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_fontsize">fontSize</code></td>
<td>
<p>cex for gene names. Default 0.8</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_showsigsymbols">showSigSymbols</code></td>
<td>
<p>Default TRUE. Heighlight significant pairs</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_showcounts">showCounts</code></td>
<td>
<p>Default FALSE. Include number of events in the plot</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_countstats">countStats</code></td>
<td>
<p>Default 'all'. Can be 'all' or 'sig'</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_counttype">countType</code></td>
<td>
<p>Default 'all'. Can be 'all', 'cooccur', 'mutexcl'</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_countsfontsize">countsFontSize</code></td>
<td>
<p>Default 0.8</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_countsfontcolor">countsFontColor</code></td>
<td>
<p>Default 'black'</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_colpal">colPal</code></td>
<td>
<p>colPalBrewer palettes. Default 'BrBG'. See RColorBrewer::display.brewer.all() for details</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_showsum">showSum</code></td>
<td>
<p>show [sum] with gene names in plot, Default TRUE</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_colnc">colNC</code></td>
<td>
<p>Number of different colors in the palette, minimum 3, default 9</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_nshiftsymbols">nShiftSymbols</code></td>
<td>
<p>shift if positive shift SigSymbols by n to the left, default 5</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_sigsymbolssize">sigSymbolsSize</code></td>
<td>
<p>size of symbols in the matrix and in legend. Default 2</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_sigsymbolsfontsize">sigSymbolsFontSize</code></td>
<td>
<p>size of font in legends. Default 0.9</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_pvsymbols">pvSymbols</code></td>
<td>
<p>vector of pch numbers for symbols of p-value for upper and lower thresholds c(upper, lower). Default c(46, 42)</p>
</td></tr>
<tr><td><code id="discoversomaticInteractions_+3A_limitcolorbreaks">limitColorBreaks</code></td>
<td>
<p>limit color to extreme values. Default TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally, this function run the getMutex function. With the 'getMutexMethod' parameter user might select
the 'method' parameter of the getMutex function. For more details run '?getMutex'
</p>
<p>#' @return A list of data.tables and it will print a heatmap with the results.
</p>


<h3>References</h3>

<p>Mayakonda A, Lin DC, Assenov Y, Plass C, Koeffler HP. 2018. Maftools:
efficient and comprehensive analysis of somatic variants in cancer.
Genome Research. http://dx.doi.org/10.1101/gr.239244.118
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

  #An example of how to perform the function,
  #using data from TCGA, Colon Adenocarcinoma in this case. 
  
  #coad.maf &lt;- GDCquery_Maf("COAD", pipelines = "muse") %&gt;% read.maf
  coad.maf &lt;- read.maf(GDCquery_Maf("COAD", pipelines = "muse"))
  discoversomaticInteractions(maf = coad.maf, top = 35, pvalue = c(1e-2, 2e-3))
  
## End(Not run)




</code></pre>

<hr>
<h2 id='getMutex'>getMutex function</h2><span id='topic+getMutex'></span>

<h3>Description</h3>

<p>Given a binary matrix and its corresponding probability matrix pij, compute the Poisson Binomial
method to estimate mutual exclusive events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMutex(
  A = NULL,
  PM = getPM(A),
  lower.tail = TRUE,
  method = "ShiftedBinomial",
  mixed = TRUE,
  th = 0.05,
  verbose = FALSE,
  parallel = FALSE,
  no_cores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMutex_+3A_a">A</code></td>
<td>
<p>The binary matrix</p>
</td></tr>
<tr><td><code id="getMutex_+3A_pm">PM</code></td>
<td>
<p>The corresponding probability matrix of A. It can be computed using function getPM. By default equal to getPM(A)</p>
</td></tr>
<tr><td><code id="getMutex_+3A_lower.tail">lower.tail</code></td>
<td>
<p>True if mutually exclusive test. False for co-ocurrence. By default is TRUE.</p>
</td></tr>
<tr><td><code id="getMutex_+3A_method">method</code></td>
<td>
<p>one of the following: &quot;ShiftedBinomial&quot; (default),&quot;Exact&quot;, &quot;Binomial&quot;, and &quot;RefinedNormal&quot;.</p>
</td></tr>
<tr><td><code id="getMutex_+3A_mixed">mixed</code></td>
<td>
<p>option to compute lower p-values with an exact method. By default TRUE</p>
</td></tr>
<tr><td><code id="getMutex_+3A_th">th</code></td>
<td>
<p>upper threshold of p.value to apply the exact method.</p>
</td></tr>
<tr><td><code id="getMutex_+3A_verbose">verbose</code></td>
<td>
<p>The verbosity of the output</p>
</td></tr>
<tr><td><code id="getMutex_+3A_parallel">parallel</code></td>
<td>
<p>If the exact method is executed with a parallel process.</p>
</td></tr>
<tr><td><code id="getMutex_+3A_no_cores">no_cores</code></td>
<td>
<p>number of cores. If not stated number of cores of the CPU - 1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>we  implemented three different approximations of the Poison-Binomial distribution function:
</p>

<ul>
<li><p> &quot;ShiftedBinomial&quot; (by default) that correspond to a shifted Binomial with three parameters (Peköz, Shwartz, Christiansen, &amp; Berlowitz, 2010).
</p>
</li>
<li><p>&quot;Exact&quot; that use the exact formula using the 'PoissonBinomial' Rpackage based on the work from (Biscarri, Zhao, &amp; Brunner, 2018).
</p>
</li>
<li><p>&quot;Binomial&quot; with two parameters (Cam, 1960).
</p>
</li>
<li><p>&quot;RefinedNormal&quot; that is based on the work from  (Volkova, 1996).
</p>
</li></ul>

<p>If 'mixed' option is selected (by default is FALSE), the &quot;Exact&quot; method is computed for p-values lower than a threshold
('th' parameter, that by default is 0.05). When the exact method is computed, it is possible to parallelize the process by
selecting the option 'parallel' (by default FALSE) and setting the number of cores ('no_cores' parameter)
</p>


<h3>Value</h3>

<p>A symmetric matrix with the p-values of the corresponding test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  #This first example is a basic 
  #example of how to perform getMutex. 
  
  data("A_example")
  PMA &lt;- getPM(A_example)
  mismutex &lt;- getMutex(A=A_example,PM=PMA)
  
  
  #The next example, is the same as the first one but,
  # using a matrix of class Matrix. 
  
  data("A_Matrix")
  A_Matrix &lt;- A_Matrix[1:100,1:50]
  #small for the example
  PMA_Matrix &lt;- getPM(A_Matrix)
  mismutex &lt;- getMutex(A=A_Matrix,PM=PMA_Matrix)
  
  
  ## Not run: 
  #Finally, the last example, shows a real 
  #example of how to perform this function when using 
  #data from TCGA, Colon Adenocarcinoma in this case. 
  
  data("TCGA_COAD")
  data("PM_COAD")
  
  PM_COAD &lt;- getMutex(TCGA_COAD, PM_COAD)
  
## End(Not run)
  

</code></pre>

<hr>
<h2 id='getMutexAB'>getMutexAB function</h2><span id='topic+getMutexAB'></span>

<h3>Description</h3>

<p>Given two binary matrices and its corresponding probability matrices PAij and PBij, compute the Poisson Binomial
method to estimate mutual exclusive events between A and B
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMutexAB(
  A,
  PMA = getPM(A),
  B,
  PMB = getPM(B),
  lower.tail = TRUE,
  method = "ShiftedBinomial",
  mixed = TRUE,
  th = 0.05,
  verbose = FALSE,
  parallel = FALSE,
  no_cores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMutexAB_+3A_a">A</code></td>
<td>
<p>The binary matrix of events A</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_pma">PMA</code></td>
<td>
<p>The corresponding probability matrix of A. It can be computed using function getPM. By default equal to getPM(A)</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_b">B</code></td>
<td>
<p>The binary matrix of events B</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_pmb">PMB</code></td>
<td>
<p>The corresponding probability matrix of B. It can be computed using function getPM. By default equal to getPM(B)</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_lower.tail">lower.tail</code></td>
<td>
<p>True if mutually exclusive test. False for co-ocurrence. By default is TRUE.</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_method">method</code></td>
<td>
<p>one of the following: &quot;ShiftedBinomial&quot; (default),&quot;Exact&quot;, &quot;RefinedNormal&quot;, and &quot;Binomial&quot;.</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_mixed">mixed</code></td>
<td>
<p>option to compute lower p-values with an exact method. By default TRUE</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_th">th</code></td>
<td>
<p>upper threshold of p-value to apply the exact method.</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_verbose">verbose</code></td>
<td>
<p>The verbosity of the output</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_parallel">parallel</code></td>
<td>
<p>If the exact method is executed with a parallel process.</p>
</td></tr>
<tr><td><code id="getMutexAB_+3A_no_cores">no_cores</code></td>
<td>
<p>number of cores. If not stated number of cores of the CPU - 1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>we  implemented three different approximations of the Poison-Binomial distribution function:
</p>

<ul>
<li><p> &quot;ShiftedBinomial&quot; (by default) that correspond to a shifted Binomial with three parameters (Peköz, Shwartz, Christiansen, &amp; Berlowitz, 2010).
</p>
</li>
<li><p>&quot;Exact&quot; that use the exact formula using the 'PoissonBinomial' Rpackage based on the work from (Biscarri, Zhao, &amp; Brunner, 2018).
</p>
</li>
<li><p>&quot;Binomial&quot; with two parameters (Cam, 1960).
</p>
</li>
<li><p>&quot;RefinedNormal&quot; that is based on the work from  (Volkova, 1996).
</p>
</li></ul>

<p>If 'mixed' option is selected (by default is FALSE), the &quot;Exact&quot; method is computed for p-values lower than a threshold
('th' parameter, that by default is 0.05). When the exact method is computed, it is possible to parallelize the process by
selecting the option 'parallel' (by default FALSE) and setting the number of cores ('no_cores' parameter)
</p>


<h3>Value</h3>

<p>A  matrix with the p-values of the corresponding test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
     
  
  #The next example, is the same as the first
  # one but, using a matrix of class Matrix. 
  
  data("A_Matrix")
  data("B_Matrix")
  PMA &lt;- getPM(A_Matrix)
  PMB &lt;- getPM(B_Matrix)
  mismutex &lt;- getMutexAB(A=A_Matrix, PM=PMA, B=B_Matrix, PMB = PMB)
  
  #Finally, the last example, shows a 
  #real example of how to perform this function
  # when using data from TCGA, Colon Adenocarcinoma in this case. 
  
  ## Not run: 
  data("TCGA_COAD_AMP")
  data("AMP_COAD")
  data("PM_TCGA_COAD_AMP")
  data("PM_AMP_COAD")
  
  mismutex &lt;- getMutexAB(A=TCGA_COAD_AMP, 
                         PMA=PM_TCGA_COAD_AMP,
                         B=AMP_COAD,
                         PMB = PM_AMP_COAD)
   
## End(Not run)
 

</code></pre>

<hr>
<h2 id='getMutexGroup'>getMutexGroup function</h2><span id='topic+getMutexGroup'></span>

<h3>Description</h3>

<p>Given a binary matrix and its corresponding probability matrix pij, compute the Poisson Binomial
method to estimate mutual exclusive events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMutexGroup(A = NULL, PM = NULL, type = "Impurity", lower.tail = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMutexGroup_+3A_a">A</code></td>
<td>
<p>The binary matrix</p>
</td></tr>
<tr><td><code id="getMutexGroup_+3A_pm">PM</code></td>
<td>
<p>The corresponding probability matrix of A. It can be computed using function getPM. By default equal to getPM(A)</p>
</td></tr>
<tr><td><code id="getMutexGroup_+3A_type">type</code></td>
<td>
<p>one of Coverage, Exclusivity or Impurity. By default is Impurity</p>
</td></tr>
<tr><td><code id="getMutexGroup_+3A_lower.tail">lower.tail</code></td>
<td>
<p>True if mutually exclusive test. False for co-ocurrence. By default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A symmetric matrix with the p.value of the corresponding test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  #This first example is a basic 
  #example of how to perform getMutexGroup
  
  data("A_example")
  A2 &lt;- A_example[,1:30]
  A2[1,1:10] &lt;- 1
  A2[2,1:10] &lt;- 0
  A2[3,1:10] &lt;- 0
  A2[1,11:20] &lt;- 0
  A2[2,11:20] &lt;- 1
  A2[3,11:20] &lt;- 0
  A2[1,21:30] &lt;- 0
  A2[2,21:30] &lt;- 0
  A2[3,21:30] &lt;- 1
  PM2 &lt;- getPM(A2)
  A &lt;- A2[1:3,]
  PM &lt;- PM2[1:3,]
  
  getMutexGroup(A, PM, "Impurity")
  getMutexGroup(A, PM, "Coverage")
  getMutexGroup(A, PM, "Exclusivity")
  

</code></pre>

<hr>
<h2 id='getPM'>getPM function</h2><span id='topic+getPM'></span>

<h3>Description</h3>

<p>Given a binary matrix estimates the corresponding probability  matrix pij.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPM(A)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPM_+3A_a">A</code></td>
<td>
<p>The binary matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'PMatrix' object with the corresponding probability estimations.
This 'PMatrix' object stored the corresponding coefficients of the 
logistic regression computed. With this coefficients it is possible to build
the complete matrix of probabilities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

  #This first example is a basic example of how to perform getPM: 
  
  data("A_example")
  PMA &lt;- getPM(A_example)
  
  #The next example, is the same as the first one but, 
  #using a matrix of class Matrix: 
  
  data("A_Matrix")
  PMA_Matrix &lt;- getPM(A_Matrix)
  
  ## Not run: 
  #Finally, the last example, shows a real example 
  #of how to perform this function when when using
  #data from TCGA, Colon Adenocarcinoma in this case: 
  data("TCGA_COAD")
  PM_COAD &lt;- getPM(TCGA_COAD)
  
## End(Not run)


</code></pre>

<hr>
<h2 id='InternalFunctions'>Rediscover Internal Functions</h2><span id='topic+InternalFunctions'></span><span id='topic+speedglm.wfit2'></span><span id='topic+expand.grid_fast'></span><span id='topic+get_os'></span><span id='topic+control_2'></span><span id='topic+is.sparse_2'></span>

<h3>Description</h3>

<p>Internal functions used by EventPointer in the different
steps of the algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>speedglm.wfit2(
  y,
  X,
  intercept = TRUE,
  weights = NULL,
  row.chunk = NULL,
  family = gaussian(),
  start = NULL,
  etastart = NULL,
  mustart = NULL,
  offset = NULL,
  acc = 1e-08,
  maxit = 25,
  k = 2,
  sparselim = 0.9,
  camp = 0.01,
  eigendec = TRUE,
  tol.values = 1e-07,
  tol.vectors = 1e-07,
  tol.solve = .Machine$double.eps,
  sparse = NULL,
  method = c("eigen", "Cholesky", "qr"),
  trace = FALSE,
  ...
)

expand.grid_fast(a, b)

get_os()

control_2(
  B,
  symmetric = TRUE,
  tol.values = 1e-07,
  tol.vectors = 1e-07,
  out.B = TRUE,
  method = c("eigen", "Cholesky")
)

is.sparse_2(X, sparselim = 0.9, camp = 0.05)
</code></pre>


<h3>Value</h3>

<p>Internal outputs
</p>

<hr>
<h2 id='PM_AMP_COAD'>PM_AMP_COAD data</h2><span id='topic+PM_AMP_COAD'></span>

<h3>Description</h3>

<p>Probability matrix, with information of genes being amplified in samples in Colon Adenocarcinoma, created by AMP_COAD.rda applying <code>getPM</code> and used as real example and <code>getMutexAB</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("PM_AMP_COAD")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>num [1:1000, 1:391] 0.118 0.118 0.118 0.118 0.114 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PM_AMP_COAD)
</code></pre>

<hr>
<h2 id='PM_COAD'>PM_COAD data</h2><span id='topic+PM_COAD'></span>

<h3>Description</h3>

<p>Probability matrix, with information of genes being mutated in samples in Colon Adenocarcinoma, created by TCGA_COAD.rda applying <code>getPM</code> and used as real example in <code>getMutex</code> and <code>getMutexAB</code> and <code>getMutexGroup</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("PM_COAD")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>Formal class 'PMatrix' [package &quot;Rediscover&quot;] with 2 slots
</p>
<p>..@ rowExps: num [1:399] 13.1 1.02 7.43 3.26 0.4 ...
</p>
<p>..@ colExps: num [1:17616] 2.54 1.78 1.76 1.35 0.6 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PM_COAD)
</code></pre>

<hr>
<h2 id='PMatrix-class'>An S4 class to store the probabilities</h2><span id='topic+PMatrix-class'></span><span id='topic+PMatrix'></span><span id='topic++5B+2CPMatrix+2Cmatrix+2Cmissing+2CANY-method'></span><span id='topic++5B+2CPMatrix+2Cmissing+2Cnumeric+2CANY-method'></span><span id='topic++5B+2CPMatrix+2Cnumeric+2Cmissing+2CANY-method'></span><span id='topic++5B+2CPMatrix+2Cnumeric+2Cnumeric+2CANY-method'></span><span id='topic+dim+2CPMatrix-method'></span>

<h3>Description</h3>

<p>An S4 class to store the probabilities of gene i being mutated in sample j
</p>


<h3>Slots</h3>


<dl>
<dt><code>rowExps</code></dt><dd><p>Sample depending estimated coefficients obtained from the logistic regression</p>
</dd>
<dt><code>colExps</code></dt><dd><p>gene depending estimated coefficients obtained from the logistic regression</p>
</dd>
</dl>

<hr>
<h2 id='TCGA_COAD'>TCGA_COAD</h2><span id='topic+TCGA_COAD'></span>

<h3>Description</h3>

<p>A binary matrix, with information about genes mutations in Colon Adenocarcinoma, created by applying <code>maftools</code> to .maf file and used as real example in <code>getPM</code> and <code>getMutex</code> and <code>getMutexAB</code> and <code>getMutexGroup</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("TCGA_COAD")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>num [1:399, 1:17616] 1 1 1 1 1 1 1 1 1 1 ...
</p>
<p>- attr(*, &quot;dimnames&quot;)=List of 2
</p>
<p>..$ : chr [1:399] &quot;TCGA-CA-6718&quot; &quot;TCGA-D5-6931&quot; &quot;TCGA-AZ-6601&quot; &quot;TCGA-G4-6320&quot; ...
</p>
<p>..$ : chr [1:17616] &quot;APC&quot; &quot;TP53&quot; &quot;TTN&quot; &quot;KRAS&quot; ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TCGA_COAD)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
