<!DOCTYPE html><html><head><title>Help for package refdb</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {refdb}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check_fields'><p>Internal check for fields</p></a></li>
<li><a href='#fields_dbs'><p>Functions to set fields for various databases</p></a></li>
<li><a href='#filter_scores'><p>Scores for filtering operations</p></a></li>
<li><a href='#get_ncbi_taxonomy'><p>Get NCBI taxonomy</p></a></li>
<li><a href='#igraph_from_taxo'><p>Create a graph from a taxonomic table</p></a></li>
<li><a href='#make_ncbi_table'><p>Parse NCBI XML and make a table</p></a></li>
<li><a href='#ncbi_taxo_rank'><p>Taxonomic ranks of the NCBI Taxonomy database</p></a></li>
<li><a href='#process_geo_ncbi'><p>Process coordinate column returned by NCBI</p></a></li>
<li><a href='#refdb_check_seq_conflict'><p>Check for conflicts in sequences</p></a></li>
<li><a href='#refdb_check_seq_homogeneity'><p>Check for genetic homogeneity of taxa</p></a></li>
<li><a href='#refdb_check_tax_conflict'><p>Check for conflicts in taxonomy</p></a></li>
<li><a href='#refdb_check_tax_typo'><p>Check for typos in taxonomic names</p></a></li>
<li><a href='#refdb_clean_seq_crop_primers'><p>Crop genetic sequences with a set of primers</p></a></li>
<li><a href='#refdb_clean_seq_remove_gaps'><p>Remove gaps from genetic sequences</p></a></li>
<li><a href='#refdb_clean_seq_remove_sideN'><p>Remove repeated side N from genetic sequences</p></a></li>
<li><a href='#refdb_clean_tax_harmonize_nomenclature'><p>Harmonize taxonomic name nomenclature</p></a></li>
<li><a href='#refdb_clean_tax_NA'><p>Convert missing taxonomic names to NA</p></a></li>
<li><a href='#refdb_clean_tax_remove_blank'><p>Remove blank characters from taxonomic names</p></a></li>
<li><a href='#refdb_clean_tax_remove_extra'><p>Remove extra words from taxonomic names</p></a></li>
<li><a href='#refdb_clean_tax_remove_subsp'><p>Remove subspecific information from taxonomic names</p></a></li>
<li><a href='#refdb_clean_tax_remove_uncertainty'><p>Remove terms indicating uncertainty in taxonomic names</p></a></li>
<li><a href='#refdb_export_dada2'><p>Export reference database for DADA2</p></a></li>
<li><a href='#refdb_export_idtaxa'><p>Export reference database for DECIPHER (IDTAXA)</p></a></li>
<li><a href='#refdb_export_mothur'><p>Export reference database for Mothur</p></a></li>
<li><a href='#refdb_fill_tax_downstream'><p>Fill missing data in taxonomy</p></a></li>
<li><a href='#refdb_fill_tax_upstream'><p>Fill missing data in taxonomy</p></a></li>
<li><a href='#refdb_filter_ref_scope'><p>Filter records by taxonomic scope of studies</p></a></li>
<li><a href='#refdb_filter_seq_ambiguous'><p>Filter sequences based on their number of ambiguous character.</p></a></li>
<li><a href='#refdb_filter_seq_duplicates'><p>Filter duplicated sequences.</p></a></li>
<li><a href='#refdb_filter_seq_homopolymers'><p>Filter sequences based on their number of repeated character.</p></a></li>
<li><a href='#refdb_filter_seq_length'><p>Filter sequences based on their number of character.</p></a></li>
<li><a href='#refdb_filter_seq_primer'><p>Filter sequences based on the presence of primers.</p></a></li>
<li><a href='#refdb_filter_seq_stopcodon'><p>Filter sequences based on their number of of stop codons.</p></a></li>
<li><a href='#refdb_filter_tax_na'><p>Filter records NA taxa</p></a></li>
<li><a href='#refdb_filter_tax_precision'><p>Filter records based on their taxonomic precision.</p></a></li>
<li><a href='#refdb_get_fields'><p>Get fields of a reference database</p></a></li>
<li><a href='#refdb_import_BOLD'><p>Download and import BOLD records</p></a></li>
<li><a href='#refdb_import_NCBI'><p>Download and import NCBI Nucleotide records</p></a></li>
<li><a href='#refdb_merge'><p>Merge reference databases</p></a></li>
<li><a href='#refdb_plot_map'><p>Plot an interactive map</p></a></li>
<li><a href='#refdb_plot_seqlen_hist'><p>Plot an histogram of sequence lengths</p></a></li>
<li><a href='#refdb_plot_tax_barplot'><p>Barplots of the number of records for the most represented taxa</p></a></li>
<li><a href='#refdb_plot_tax_tree'><p>Reference database taxonomy tree</p></a></li>
<li><a href='#refdb_plot_tax_treemap'><p>Reference database treemap</p></a></li>
<li><a href='#refdb_report'><p>Compile a report with different checks</p></a></li>
<li><a href='#refdb_sample_tax'><p>Sample records within taxa</p></a></li>
<li><a href='#refdb_set_fields'><p>Associate columns to fields</p></a></li>
<li><a href='#refdb_set_ncbitax'><p>Replace the current taxonomy using the NCBI Taxonomy database</p></a></li>
<li><a href='#refdb_write_fields'><p>Write fields to a file</p></a></li>
<li><a href='#refdb-package'><p>refdb: A DNA Reference Library Manager</p></a></li>
<li><a href='#valid_taxo_rank'><p>Ranks considered as valid by refdb</p></a></li>
<li><a href='#xml_extract'><p>Extract XML elements</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A DNA Reference Library Manager</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francois Keck &lt;francois.keck@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Reference database manager offering a set of functions
    to import, organize, clean, filter, audit and export reference genetic
    data. Provide functions to download sequence data from
    Bold Systems (<a href="https://www.boldsystems.org/">https://www.boldsystems.org/</a>) and
    NCBI GenBank <a href="https://www.ncbi.nlm.nih.gov/genbank/">https://www.ncbi.nlm.nih.gov/genbank/</a>.
    Designed as an environment for semi-automatic and assisted
    construction of reference databases and to improve standardization
    and repeatability in barcoding and metabarcoding studies.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://fkeck.github.io/refdb/">https://fkeck.github.io/refdb/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fkeck/refdb/issues">https://github.com/fkeck/refdb/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>tibble, readr, dplyr, stringr, tidyr, rentrez, bold, taxize,
xml2, bioseq, ape, igraph, ggplot2, ggraph, yaml, rlang,
rmarkdown, leaflet</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), covr, DT, knitr, forcats</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-22 15:51:27 UTC; ecoadmin</td>
</tr>
<tr>
<td>Author:</td>
<td>Francois Keck <a href="https://orcid.org/0000-0002-3323-4167"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-22 17:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='check_fields'>Internal check for fields</h2><span id='topic+check_fields'></span>

<h3>Description</h3>

<p>Internal check for fields
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_fields(x, what = c("source", "id", "taxonomy", "sequence", "marker"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_fields_+3A_x">x</code></td>
<td>
<p>a reference database (tibble object).</p>
</td></tr>
<tr><td><code id="check_fields_+3A_what">what</code></td>
<td>
<p>a vector of fields to be checked.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible or error.
</p>

<hr>
<h2 id='fields_dbs'>Functions to set fields for various databases</h2><span id='topic+fields_dbs'></span><span id='topic+refdb_set_fields_NCBI'></span><span id='topic+refdb_set_fields_BOLD'></span><span id='topic+refdb_set_fields_PR2'></span><span id='topic+refdb_set_fields_diatbarcode'></span>

<h3>Description</h3>

<p>Functions to set fields for various databases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_set_fields_NCBI(x)

refdb_set_fields_BOLD(x)

refdb_set_fields_PR2(x)

refdb_set_fields_diatbarcode(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fields_dbs_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns <code>x</code> with updated attributes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
refdb_set_fields_BOLD(lib)

</code></pre>

<hr>
<h2 id='filter_scores'>Scores for filtering operations</h2><span id='topic+filter_scores'></span><span id='topic+.filter_seq_length'></span>

<h3>Description</h3>

<p>Scores for filtering operations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.filter_seq_length(x, gaps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_scores_+3A_x">x</code></td>
<td>
<p>a reference database</p>
</td></tr>
<tr><td><code id="filter_scores_+3A_gaps">gaps</code></td>
<td>
<p>should gaps be included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector
</p>

<hr>
<h2 id='get_ncbi_taxonomy'>Get NCBI taxonomy</h2><span id='topic+get_ncbi_taxonomy'></span>

<h3>Description</h3>

<p>Download and parse NCBI taxonomy records
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ncbi_taxonomy(id, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ncbi_taxonomy_+3A_id">id</code></td>
<td>
<p>A vector of id for records in the NCBI Taxonomy database.</p>
</td></tr>
<tr><td><code id="get_ncbi_taxonomy_+3A_verbose">verbose</code></td>
<td>
<p>print information in the console.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with each row corresponding to an id and each column
to a taxonomic level.
</p>

<hr>
<h2 id='igraph_from_taxo'>Create a graph from a taxonomic table</h2><span id='topic+igraph_from_taxo'></span>

<h3>Description</h3>

<p>Create a graph representation from a taxonomic classification
included in a reference
database. For this function to work, taxonomic fields must be set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>igraph_from_taxo(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="igraph_from_taxo_+3A_x">x</code></td>
<td>
<p>a reference database (tibble).</p>
</td></tr>
<tr><td><code id="igraph_from_taxo_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names to use a subset of columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <span class="pkg">igraph</span> object representing taxonomic relationships.
</p>

<hr>
<h2 id='make_ncbi_table'>Parse NCBI XML and make a table</h2><span id='topic+make_ncbi_table'></span>

<h3>Description</h3>

<p>Parse NCBI XML and make a table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_ncbi_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_ncbi_table_+3A_x">x</code></td>
<td>
<p>A XML nodeset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>

<hr>
<h2 id='ncbi_taxo_rank'>Taxonomic ranks of the NCBI Taxonomy database</h2><span id='topic+ncbi_taxo_rank'></span>

<h3>Description</h3>

<p>Taxonomic ranks of the NCBI Taxonomy database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_taxo_rank()
</code></pre>


<h3>Value</h3>

<p>a vector of ordered ranks
</p>

<hr>
<h2 id='process_geo_ncbi'>Process coordinate column returned by NCBI</h2><span id='topic+process_geo_ncbi'></span>

<h3>Description</h3>

<p>Process coordinate column returned by NCBI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_geo_ncbi(x, col = "lat_lon")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_geo_ncbi_+3A_x">x</code></td>
<td>
<p>NCBI dataframe.</p>
</td></tr>
<tr><td><code id="process_geo_ncbi_+3A_col">col</code></td>
<td>
<p>column name containing geographical coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NCBI dataframe.
</p>

<hr>
<h2 id='refdb_check_seq_conflict'>Check for conflicts in sequences</h2><span id='topic+refdb_check_seq_conflict'></span>

<h3>Description</h3>

<p>Check for conflicts in sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_check_seq_conflict(x, na_omit = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_check_seq_conflict_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_check_seq_conflict_+3A_na_omit">na_omit</code></td>
<td>
<p>if <code>FALSE</code> conflicts involving
NA taxonomic names are also reported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two-columns tibbles reporting duplicated
sequences with different taxonomy.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
lib &lt;- refdb_set_fields(lib,
                        taxonomy = c(family = "family_name",
                            genus = "genus_name",
                            species = "species_name"),
                        sequence = "DNA_seq",
                        marker = "marker")
refdb_check_seq_conflict(lib)

</code></pre>

<hr>
<h2 id='refdb_check_seq_homogeneity'>Check for genetic homogeneity of taxa</h2><span id='topic+refdb_check_seq_homogeneity'></span>

<h3>Description</h3>

<p>This function assesses the genetic similarity among
sequences within each taxa.
It takes user defined thresholds (one threshold per taxonomic level)
to warn about sequences which are singularly different
(based on median distance) from the others.
Sequences in the reference database must be aligned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_check_seq_homogeneity(x, levels, min_n_seq = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_check_seq_homogeneity_+3A_x">x</code></td>
<td>
<p>a reference database (sequences must be aligned).</p>
</td></tr>
<tr><td><code id="refdb_check_seq_homogeneity_+3A_levels">levels</code></td>
<td>
<p>a named vector of genetic similarity thresholds.
Names must correspond to taxonomic levels (taxonomic fields)
and values must be included in the interval [0, 1].
For example to assess homogeneity at 5 percents (within species) and
10 percents (within genus): <code>levels = c(species = 0.05, genus = 0.1)</code></p>
</td></tr>
<tr><td><code id="refdb_check_seq_homogeneity_+3A_min_n_seq">min_n_seq</code></td>
<td>
<p>the minimum number of sequences for a
taxon to be tested.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For every tested taxonomic levels, the algorithm
checks all sequences in every taxa
(for which the total number of sequence is &gt; <code>min_n_seq</code>)
In each taxon, the pairwise distance matrix among all the sequences
belonging to this taxon is computed. A sequence is tagged as suspicious
and returned by the function
if its median genetic distance from the other sequences is higher than
the threshold set by the user (<code>levels</code> argument).
</p>


<h3>Value</h3>

<p>A dataframe reporting suspicious sequences whose median distance
to other sequences of the same taxon is greater than
the specified threshold. The first column &quot;level_threshold_homogeneity&quot;
indicates the lowest taxonomic level for which the threshold has been
exceeded and the second column &quot;value_threshold_homogeneity&quot; gives
the computed median distance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "homogeneity.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_check_seq_homogeneity(lib, levels = c(species = 0.05, genus = 0.1))

</code></pre>

<hr>
<h2 id='refdb_check_tax_conflict'>Check for conflicts in taxonomy</h2><span id='topic+refdb_check_tax_conflict'></span>

<h3>Description</h3>

<p>Check for conflicts in taxonomy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_check_tax_conflict(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_check_tax_conflict_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two-columns tibbles reporting for each taxonomic level
the taxa with identical names but different upstream taxonomy.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
lib &lt;- refdb_set_fields(lib,
                        taxonomy = c(family = "family_name",
                            genus = "genus_name",
                            species = "species_name"),
                        sequence = "DNA_seq",
                        marker = "marker")
refdb_check_tax_conflict(lib)

</code></pre>

<hr>
<h2 id='refdb_check_tax_typo'>Check for typos in taxonomic names</h2><span id='topic+refdb_check_tax_typo'></span>

<h3>Description</h3>

<p>This function uses the generalized Levenshtein (edit) distance
to identify possible issue with taxonomic names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_check_tax_typo(x, tol = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_check_tax_typo_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_check_tax_typo_+3A_tol">tol</code></td>
<td>
<p>the edit distance below which two taxonomic names are reported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two-columns tibbles reporting for each taxonomic level
the pairs of taxonomic names sharing the same upstream taxonomy and for
which the generalized Levenshtein (edit) distance is below
the <code>tol</code> value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
lib &lt;- refdb_set_fields(lib,
                        taxonomy = c(family = "family_name",
                            genus = "genus_name",
                            species = "species_name"),
                        sequence = "DNA_seq",
                        marker = "marker")
refdb_check_tax_typo(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_seq_crop_primers'>Crop genetic sequences with a set of primers</h2><span id='topic+refdb_clean_seq_crop_primers'></span>

<h3>Description</h3>

<p>Crop genetic sequences with a set of primers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_seq_crop_primers(
  x,
  primer_forward,
  primer_reverse,
  max_error_in = 0.1,
  max_error_out = 0.1,
  include_primers = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_seq_crop_primers_+3A_x">x</code></td>
<td>
<p>a reference database with a defined sequence field.</p>
</td></tr>
<tr><td><code id="refdb_clean_seq_crop_primers_+3A_primer_forward">primer_forward</code></td>
<td>
<p>primer forward.</p>
</td></tr>
<tr><td><code id="refdb_clean_seq_crop_primers_+3A_primer_reverse">primer_reverse</code></td>
<td>
<p>primer reverse.</p>
</td></tr>
<tr><td><code id="refdb_clean_seq_crop_primers_+3A_max_error_in">max_error_in</code>, <code id="refdb_clean_seq_crop_primers_+3A_max_error_out">max_error_out</code></td>
<td>
<p>maximum error for a match
(frequency based on primer length).</p>
</td></tr>
<tr><td><code id="refdb_clean_seq_crop_primers_+3A_include_primers">include_primers</code></td>
<td>
<p>a logical indicating whether the detected primers are
included in the cropped sequences.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_seq_crop_primers(lib, "AGT", "TTTA")

</code></pre>

<hr>
<h2 id='refdb_clean_seq_remove_gaps'>Remove gaps from genetic sequences</h2><span id='topic+refdb_clean_seq_remove_gaps'></span>

<h3>Description</h3>

<p>Remove gaps from genetic sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_seq_remove_gaps(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_seq_remove_gaps_+3A_x">x</code></td>
<td>
<p>a reference database with a defined sequence field.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_seq_remove_gaps(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_seq_remove_sideN'>Remove repeated side N from genetic sequences</h2><span id='topic+refdb_clean_seq_remove_sideN'></span>

<h3>Description</h3>

<p>Remove repeated side N from genetic sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_seq_remove_sideN(x, side = "both")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_seq_remove_sideN_+3A_x">x</code></td>
<td>
<p>a reference database with a defined sequence field.</p>
</td></tr>
<tr><td><code id="refdb_clean_seq_remove_sideN_+3A_side">side</code></td>
<td>
<p>which side to clean.
Can be one of <code>"left"</code>, <code>"right"</code> or <code>"both"</code> (default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_seq_remove_sideN(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_tax_harmonize_nomenclature'>Harmonize taxonomic name nomenclature</h2><span id='topic+refdb_clean_tax_harmonize_nomenclature'></span>

<h3>Description</h3>

<p>Harmonize taxonomic name nomenclature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_tax_harmonize_nomenclature(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_tax_harmonize_nomenclature_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_harmonize_nomenclature_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> and <code>organism</code> fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_tax_harmonize_nomenclature(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_tax_NA'>Convert missing taxonomic names to NA</h2><span id='topic+refdb_clean_tax_NA'></span>

<h3>Description</h3>

<p>Convert missing taxonomic names to NA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_tax_NA(x, cols = NULL, hybrid = TRUE, uncertain = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_tax_NA_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_NA_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> and <code>organism</code> fields.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_NA_+3A_hybrid">hybrid</code></td>
<td>
<p>hybrids are converted to NA (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_NA_+3A_uncertain">uncertain</code></td>
<td>
<p>taxa with qualifiers of uncertainty (cf., aff., etc.)
are converted to NA (default <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_tax_NA(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_tax_remove_blank'>Remove blank characters from taxonomic names</h2><span id='topic+refdb_clean_tax_remove_blank'></span>

<h3>Description</h3>

<p>Remove blank characters from taxonomic names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_tax_remove_blank(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_tax_remove_blank_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_remove_blank_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> and <code>organism</code> fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_tax_remove_blank(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_tax_remove_extra'>Remove extra words from taxonomic names</h2><span id='topic+refdb_clean_tax_remove_extra'></span>

<h3>Description</h3>

<p>Remove extra words from taxonomic names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_tax_remove_extra(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_tax_remove_extra_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_remove_extra_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> and <code>organism</code> fields.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As the function can match words like &quot;g.&quot;, &quot;s.&quot; or &quot;x&quot;, which can
have a signification in some nomenclatures, it is recommended to
execute <a href="#topic+refdb_clean_tax_harmonize_nomenclature">refdb_clean_tax_harmonize_nomenclature</a> first.
</p>


<h3>Value</h3>

<p>A reference database.
</p>
<p>lib &lt;- read.csv(system.file(&quot;extdata&quot;, &quot;baetidae_bold.csv&quot;, package = &quot;refdb&quot;))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_tax_remove_extra(lib)
</p>

<hr>
<h2 id='refdb_clean_tax_remove_subsp'>Remove subspecific information from taxonomic names</h2><span id='topic+refdb_clean_tax_remove_subsp'></span>

<h3>Description</h3>

<p>Remove subspecific information from taxonomic names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_tax_remove_subsp(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_tax_remove_subsp_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_remove_subsp_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> and <code>organism</code> fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_tax_remove_subsp(lib)

</code></pre>

<hr>
<h2 id='refdb_clean_tax_remove_uncertainty'>Remove terms indicating uncertainty in taxonomic names</h2><span id='topic+refdb_clean_tax_remove_uncertainty'></span>

<h3>Description</h3>

<p>Remove terms indicating uncertainty in taxonomic names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_clean_tax_remove_uncertainty(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_clean_tax_remove_uncertainty_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_clean_tax_remove_uncertainty_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> and <code>organism</code> fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>Warning</h3>

<p>Marks of taxonomic uncertainty provided by specialists
are not without value. The consequences of their deletion
must be well understood by the user before using this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_clean_tax_remove_uncertainty(lib)

</code></pre>

<hr>
<h2 id='refdb_export_dada2'>Export reference database for DADA2</h2><span id='topic+refdb_export_dada2'></span>

<h3>Description</h3>

<p>Write reference database in formats which can be used with the
functions of the package <span class="pkg">dada2</span>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_export_dada2(x, file, mode = "taxonomy")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_export_dada2_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_export_dada2_+3A_file">file</code></td>
<td>
<p>a path to the file to be written.</p>
</td></tr>
<tr><td><code id="refdb_export_dada2_+3A_mode">mode</code></td>
<td>
<p>character string to determine the type of file to produce.
Use <code>"taxonomy"</code> to produce a file for function <code>assignTaxonomy</code>
or <code>"species"</code> to produce a file for function <code>assignSpecies</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_export_dada2(lib, tempfile())

</code></pre>

<hr>
<h2 id='refdb_export_idtaxa'>Export reference database for DECIPHER (IDTAXA)</h2><span id='topic+refdb_export_idtaxa'></span>

<h3>Description</h3>

<p>Write a reference database in file formats which can be used
to train the <code>IDTAXA</code> classifier implemented in <code>DECIPHER</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_export_idtaxa(x, file, taxid = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_export_idtaxa_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_export_idtaxa_+3A_file">file</code></td>
<td>
<p>a file path without extension. This will be used to create
a .fasta file and two .txt files.</p>
</td></tr>
<tr><td><code id="refdb_export_idtaxa_+3A_taxid">taxid</code></td>
<td>
<p>should the taxid file be generated
(can be very slow with large databases)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The functions generates three files.
</p>
<p>- A fasta files containing the sequences with their IDs.
This file must be imported as a <code>DNAStringSet</code>
to be used with <code>DECIPHER</code>, using eg:<br />
<code>Biostrings::readDNAStringSet("ex_seqs.fasta")</code>
</p>
<p>- A text files containing the sequence taxonomic assignment.
This file must be imported as a character vector
to be used with <code>DECIPHER</code>, using eg:<br />
<code>readr::read_lines("ex_taxo.txt")</code>
</p>
<p>- A text file (&quot;taxid&quot;) containing the taxonomic ranks
associated with each taxon. This is an asterisk delimited file
which must be imported as a dataframe (see LearnTaxa), using eg:<br />
<code>readr::read_delim("ex_ranks.txt",
                   col_names = c('Index', 'Name', 'Parent', 'Level', 'Rank'),
                   delim = "*", quote = "")</code>
</p>
<p>The taxid file can be very slow to write for large datasets.
Therefore it is not generated by default.
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_export_idtaxa(lib, tempfile())

</code></pre>

<hr>
<h2 id='refdb_export_mothur'>Export reference database for Mothur</h2><span id='topic+refdb_export_mothur'></span>

<h3>Description</h3>

<p>Write a reference database in formats which can be used
with <code>Mothur</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_export_mothur(x, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_export_mothur_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_export_mothur_+3A_file">file</code></td>
<td>
<p>a file path. This will be used to create
a .fasta file and a .txt file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_export_mothur(lib, tempfile())

</code></pre>

<hr>
<h2 id='refdb_fill_tax_downstream'>Fill missing data in taxonomy</h2><span id='topic+refdb_fill_tax_downstream'></span>

<h3>Description</h3>

<p>Replace NA values in taxonomic classification
using upstream ranks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_fill_tax_downstream(x, qualifier = "indet.")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_fill_tax_downstream_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_fill_tax_downstream_+3A_qualifier">qualifier</code></td>
<td>
<p>a string to add the new labels.
Default ensure that <code>refdb_clean_tax_NA</code> will correctly
identify the label as NA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>See Also</h3>

<p>refdb_fill_tax_upstream to replace NA values
using downstream data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_fill_tax_downstream(lib)

</code></pre>

<hr>
<h2 id='refdb_fill_tax_upstream'>Fill missing data in taxonomy</h2><span id='topic+refdb_fill_tax_upstream'></span>

<h3>Description</h3>

<p>Replace NA values in taxonomic classification
using downstream ranks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_fill_tax_upstream(x, qualifier = "undef.")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_fill_tax_upstream_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_fill_tax_upstream_+3A_qualifier">qualifier</code></td>
<td>
<p>a string to add the new labels.
Default ensure that <code>refdb_clean_tax_NA</code> will correctly
identify the label as NA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>


<h3>See Also</h3>

<p>refdb_fill_tax_downstream to replace terminal NA values
using upstream data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_fill_tax_upstream(lib)

</code></pre>

<hr>
<h2 id='refdb_filter_ref_scope'>Filter records by taxonomic scope of studies</h2><span id='topic+refdb_filter_ref_scope'></span>

<h3>Description</h3>

<p>Filter records by taxonomic scope of studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_ref_scope(x, max_tax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_ref_scope_+3A_x">x</code></td>
<td>
<p>a reference database (tibble).</p>
</td></tr>
<tr><td><code id="refdb_filter_ref_scope_+3A_max_tax">max_tax</code></td>
<td>
<p>the maximum (widest) taxonomic focus of the study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A reference field (one ore more columns) must be set to use
this function. If reference is not available (NA) for a record,
the record is not dropped.
</p>


<h3>Value</h3>

<p>a reference database (tibble).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
lib$refs &lt;- rep("REF_1", nrow(lib))
lib &lt;- refdb_set_fields(lib, reference = "refs")
refdb_filter_ref_scope(lib, max_tax = "family_name")

</code></pre>

<hr>
<h2 id='refdb_filter_seq_ambiguous'>Filter sequences based on their number of ambiguous character.</h2><span id='topic+refdb_filter_seq_ambiguous'></span>

<h3>Description</h3>

<p>Filter sequences based on their number of ambiguous character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_seq_ambiguous(x, max_ambig = 3L, char = "N")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_seq_ambiguous_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_ambiguous_+3A_max_ambig">max_ambig</code></td>
<td>
<p>maximum number of ambiguous character.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_ambiguous_+3A_char">char</code></td>
<td>
<p>characters interpreted as ambiguous (vector).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_seq_ambiguous(lib)

</code></pre>

<hr>
<h2 id='refdb_filter_seq_duplicates'>Filter duplicated sequences.</h2><span id='topic+refdb_filter_seq_duplicates'></span>

<h3>Description</h3>

<p>Exclude duplicated sequences.
This is based both on sequences and taxonomy.
NA values are assumed to be comparable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_seq_duplicates(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_seq_duplicates_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_seq_duplicates(lib)

</code></pre>

<hr>
<h2 id='refdb_filter_seq_homopolymers'>Filter sequences based on their number of repeated character.</h2><span id='topic+refdb_filter_seq_homopolymers'></span>

<h3>Description</h3>

<p>Filter sequences based on their number of repeated character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_seq_homopolymers(x, max_len = 16L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_seq_homopolymers_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_homopolymers_+3A_max_len">max_len</code></td>
<td>
<p>maximum number of repeated character (homopolymer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_seq_homopolymers(lib)

</code></pre>

<hr>
<h2 id='refdb_filter_seq_length'>Filter sequences based on their number of character.</h2><span id='topic+refdb_filter_seq_length'></span>

<h3>Description</h3>

<p>Filter sequences based on their number of character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_seq_length(x, min_len = NULL, max_len = NULL, gaps = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_seq_length_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_length_+3A_min_len">min_len</code>, <code id="refdb_filter_seq_length_+3A_max_len">max_len</code></td>
<td>
<p>minimum and maximum sequence lengths.
Use <code>NULL</code> (default) to ignore.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_length_+3A_gaps">gaps</code></td>
<td>
<p>if <code>TRUE</code> gaps are accounted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_seq_length(lib, 50L)

</code></pre>

<hr>
<h2 id='refdb_filter_seq_primer'>Filter sequences based on the presence of primers.</h2><span id='topic+refdb_filter_seq_primer'></span>

<h3>Description</h3>

<p>Filter sequences based on the presence of primers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_seq_primer(
  x,
  primer_forward = NULL,
  primer_reverse = NULL,
  max_error_forward = 0.1,
  max_error_reverse = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_seq_primer_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_primer_+3A_primer_forward">primer_forward</code></td>
<td>
<p>forward primer.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_primer_+3A_primer_reverse">primer_reverse</code></td>
<td>
<p>reverse primer.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_primer_+3A_max_error_forward">max_error_forward</code>, <code id="refdb_filter_seq_primer_+3A_max_error_reverse">max_error_reverse</code></td>
<td>
<p>maximum error for match
(frequency base on primer length).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_seq_primer(lib, "ACTA")

</code></pre>

<hr>
<h2 id='refdb_filter_seq_stopcodon'>Filter sequences based on their number of of stop codons.</h2><span id='topic+refdb_filter_seq_stopcodon'></span>

<h3>Description</h3>

<p>Filter sequences based on their number of of stop codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_seq_stopcodon(x, max_stop = 0, code, codon_frame = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_seq_stopcodon_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_stopcodon_+3A_max_stop">max_stop</code></td>
<td>
<p>maximum number of stop codons.</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_stopcodon_+3A_code">code</code></td>
<td>
<p>an integer indicating the genetic code to use for translation
(see <a href="bioseq.html#topic+genetic-codes">genetic-codes</a>).</p>
</td></tr>
<tr><td><code id="refdb_filter_seq_stopcodon_+3A_codon_frame">codon_frame</code></td>
<td>
<p>an integer giving the nucleotide position where
to start translation. If <code>NA</code> (the default), the three different frames
are tested and the frame producing the lowest number of stop codons will
be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_seq_stopcodon(lib, code = 5)

</code></pre>

<hr>
<h2 id='refdb_filter_tax_na'>Filter records NA taxa</h2><span id='topic+refdb_filter_tax_na'></span>

<h3>Description</h3>

<p>Remove records where taxa is NA if it is not
the only representant of the upper clade.
Note that the function maybe slow on large datasets.
//EXPERIMENTAL//
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_tax_na(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_tax_na_+3A_x">x</code></td>
<td>
<p>a reference database.
(column name of the reference database).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_tax_na(lib)

</code></pre>

<hr>
<h2 id='refdb_filter_tax_precision'>Filter records based on their taxonomic precision.</h2><span id='topic+refdb_filter_tax_precision'></span>

<h3>Description</h3>

<p>Filter records based on their taxonomic precision.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_filter_tax_precision(x, min_tax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_filter_tax_precision_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_filter_tax_precision_+3A_min_tax">min_tax</code></td>
<td>
<p>minimum taxonomic level
(column name of the reference database).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (filtered reference database).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_filter_tax_precision(lib, min_tax = "family_name")

</code></pre>

<hr>
<h2 id='refdb_get_fields'>Get fields of a reference database</h2><span id='topic+refdb_get_fields'></span>

<h3>Description</h3>

<p>Get fields of a reference database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_get_fields(x, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_get_fields_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_get_fields_+3A_silent">silent</code></td>
<td>
<p>if <code>TRUE</code> silently and invisibly
returns fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The list of fields is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
refdb_get_fields(lib)

</code></pre>

<hr>
<h2 id='refdb_import_BOLD'>Download and import BOLD records</h2><span id='topic+refdb_import_BOLD'></span>

<h3>Description</h3>

<p>This function allows to search and download data
from the the BOLD database.
Additionally it allows to convert the BOLD taxonomic classification
into the NCBI classification using the NCBI Taxonomy database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_import_BOLD(
  taxon = NULL,
  ids = NULL,
  bin = NULL,
  container = NULL,
  institutions = NULL,
  researchers = NULL,
  geo = NULL,
  ncbi_taxo = TRUE,
  full = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_import_BOLD_+3A_taxon">taxon</code>, <code id="refdb_import_BOLD_+3A_ids">ids</code>, <code id="refdb_import_BOLD_+3A_bin">bin</code>, <code id="refdb_import_BOLD_+3A_container">container</code>, <code id="refdb_import_BOLD_+3A_institutions">institutions</code>, <code id="refdb_import_BOLD_+3A_researchers">researchers</code>, <code id="refdb_import_BOLD_+3A_geo">geo</code></td>
<td>
<p>terms
used to search the BOLD database. See <a href="bold.html#topic+bold_seqspec">bold_seqspec</a> from
which they are inherited.</p>
</td></tr>
<tr><td><code id="refdb_import_BOLD_+3A_ncbi_taxo">ncbi_taxo</code></td>
<td>
<p>if <code>TRUE</code> (the default) the BOLD taxonomy is
replaced by the NCBI taxonomy using <code><a href="#topic+refdb_set_ncbitax">refdb_set_ncbitax</a></code>.</p>
</td></tr>
<tr><td><code id="refdb_import_BOLD_+3A_full">full</code></td>
<td>
<p>if <code>TRUE</code> the complete data are returned for every records.
If <code>FALSE</code> (the default) only a subset of the most
important columns is returned.</p>
</td></tr>
<tr><td><code id="refdb_import_BOLD_+3A_verbose">verbose</code></td>
<td>
<p>print information in the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses several functions of the <span class="pkg">bold</span>
package to interface with the BOLD API. The NCBI Taxonomy
database is queried using the <span class="pkg">rentrez</span> package.
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>Large requests</h3>

<p>As detailed in the manual of bold_seqspec, some large requests
(e.g. high ranked taxa) can lead to errors. In that case one
strategy can be to sequentially download data for lower rank
taxa. See <a href="https://docs.ropensci.org/bold/#large-data">https://docs.ropensci.org/bold/#large-data</a>.
</p>


<h3>See Also</h3>

<p>link[bold]bold_stats and <a href="bold.html#topic+bold_seqspec">bold_seqspec</a>
from <span class="pkg">bold</span> which are used internally.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
goera_bold &lt;- refdb_import_BOLD(taxon = "Goera pilosa", ncbi_taxo = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='refdb_import_NCBI'>Download and import NCBI Nucleotide records</h2><span id='topic+refdb_import_NCBI'></span>

<h3>Description</h3>

<p>This function allows to search and download data
from the the NCBI Nucleotide database.
Additionally it uses the NCBI Taxonomy database
to get the sequence taxonomic classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_import_NCBI(
  query,
  full = FALSE,
  max_seq_length = 10000,
  seq_bin = 200,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_import_NCBI_+3A_query">query</code></td>
<td>
<p>a character string with the query.</p>
</td></tr>
<tr><td><code id="refdb_import_NCBI_+3A_full">full</code></td>
<td>
<p>a logical. If FALSE (the default), only a subset of
the most important fields is included in the result.</p>
</td></tr>
<tr><td><code id="refdb_import_NCBI_+3A_max_seq_length">max_seq_length</code></td>
<td>
<p>a numeric giving the maximum length of sequences
to retrieve. Useful to exclude complete genomes.</p>
</td></tr>
<tr><td><code id="refdb_import_NCBI_+3A_seq_bin">seq_bin</code></td>
<td>
<p>number of sequences to download at once.</p>
</td></tr>
<tr><td><code id="refdb_import_NCBI_+3A_verbose">verbose</code></td>
<td>
<p>print information in the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses several functions of the <span class="pkg">rentrez</span>
package to interface with the NCBI's EUtils API.*
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>Errors</h3>

<p><code>Error in curl::curl_fetch_memory(url, handle = handle) :
transfer closed with outstanding read data remaining</code> <br />
This error seems to appear with long sequences.
You can try to decrease <code>max_seq_length</code> to exclude them.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
silo_ncbi &lt;- refdb_import_NCBI("Silo COI")

## End(Not run)

</code></pre>

<hr>
<h2 id='refdb_merge'>Merge reference databases</h2><span id='topic+refdb_merge'></span>

<h3>Description</h3>

<p>Merge several reference database by common fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_merge(..., keep = "fields_all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_merge_+3A_...">...</code></td>
<td>
<p>reference databases (tibbles).</p>
</td></tr>
<tr><td><code id="refdb_merge_+3A_keep">keep</code></td>
<td>
<p>determines which columns to keep.
Can be <code>"fields_all"</code> (default), <code>"fields_shared"</code>
or <code>"all"</code> (see Details).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Columns are merged only if they are associated to the same field.
</p>
<p>The <code>keep</code> argument determines which columns are returned as follow.
<code>"fields_all"</code> (the default) returns all the fields
existing in all the reference databases.
<code>"fields_shared"</code> returns only the fields shared by
all the reference databases.
<code>"all"</code> returns all the columns of all the databases.
Columns which are not associated to a field are not merged and are prefixed
with the name of the object they originated from.
</p>


<h3>Value</h3>

<p>a merged reference database (tibble).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib_1 &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib_1 &lt;- refdb_set_fields_BOLD(lib_1)
lib_2 &lt;- lib_1
refdb_merge(lib_1, lib_2)

</code></pre>

<hr>
<h2 id='refdb_plot_map'>Plot an interactive map</h2><span id='topic+refdb_plot_map'></span>

<h3>Description</h3>

<p>This functions generate an interactive maps showing
the location of the records of a reference database.
Note that only records with
latitude and longitude data will be displayed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_plot_map(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_plot_map_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An interactive map object from the <span class="pkg">leaflet</span> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
lib &lt;- refdb_set_fields(lib, latitude = "lat", longitude = "lon")
refdb_plot_map(lib)

</code></pre>

<hr>
<h2 id='refdb_plot_seqlen_hist'>Plot an histogram of sequence lengths</h2><span id='topic+refdb_plot_seqlen_hist'></span>

<h3>Description</h3>

<p>Plot an histogram of sequence lengths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_plot_seqlen_hist(x, remove_gaps = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_plot_seqlen_hist_+3A_x">x</code></td>
<td>
<p>a reference database</p>
</td></tr>
<tr><td><code id="refdb_plot_seqlen_hist_+3A_remove_gaps">remove_gaps</code></td>
<td>
<p>a logical (default <code>TRUE</code> to control whether
gaps (-) should be removed prior computing sequence lengths.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object. This means the plot
can be further customized using <span class="pkg">ggplot2</span> compatible functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_plot_seqlen_hist(lib)

</code></pre>

<hr>
<h2 id='refdb_plot_tax_barplot'>Barplots of the number of records for the most represented taxa</h2><span id='topic+refdb_plot_tax_barplot'></span>

<h3>Description</h3>

<p>Generate a multipanel plot where, for each taxonomic level,
a barplot represent the number of records available
in the reference database for the most represented taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_plot_tax_barplot(x, show_n = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_plot_tax_barplot_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_barplot_+3A_show_n">show_n</code></td>
<td>
<p>an integer value indicating the number of taxa to show in each panel.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
lib &lt;- refdb_set_fields(lib, latitude = "lat", longitude = "lon")
refdb_plot_tax_barplot(lib)

</code></pre>

<hr>
<h2 id='refdb_plot_tax_tree'>Reference database taxonomy tree</h2><span id='topic+refdb_plot_tax_tree'></span>

<h3>Description</h3>

<p>Represent the hierarchical structure of the taxonomic information
of a reference database as a tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_plot_tax_tree(
  x,
  leaf_col = NULL,
  color_col = NULL,
  freq_labels = 0,
  expand_plot = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_plot_tax_tree_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_tree_+3A_leaf_col">leaf_col</code></td>
<td>
<p>a column name referring to the taxonomic level
for the leaves of the tree. If not provided (<code>NULL</code>)
the function tries to find a relevant level.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_tree_+3A_color_col">color_col</code></td>
<td>
<p>a column name referring to the taxonomic level
for the color of the leaves (must be higher or equal to the level
of <code>leaf_col</code>). If not provided (<code>NULL</code>)
the function tries to find a relevant level.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_tree_+3A_freq_labels">freq_labels</code></td>
<td>
<p>a numeric value to adjust
the number of printed labels (minimum frequency).
Default is zero which means all non-NA labels are printed.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_tree_+3A_expand_plot">expand_plot</code></td>
<td>
<p>a value to expand the limits of the plot.
Useful if the labels are too long.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The underlying graph is computed using the non-exported function
<code>igraph_from_taxo</code>.
</p>


<h3>Value</h3>

<p>A <span class="pkg">ggplot2</span> (<span class="pkg">ggraph</span>) object. This means the plot
can be further customized using <span class="pkg">ggplot2</span> compatible functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_plot_tax_tree(lib)

</code></pre>

<hr>
<h2 id='refdb_plot_tax_treemap'>Reference database treemap</h2><span id='topic+refdb_plot_tax_treemap'></span>

<h3>Description</h3>

<p>Represent the hierarchical structure of the taxonomic information
of a reference database as a set of nested rectangles (treemap).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_plot_tax_treemap(x, cols = NULL, freq_labels = c(0.01, 0.003))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_plot_tax_treemap_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_treemap_+3A_cols">cols</code></td>
<td>
<p>a vector of column names referring to taxonomic levels
to include in the treemap. If not provided (<code>NULL</code>)
the function tries to find a relevant subset of columns.</p>
</td></tr>
<tr><td><code id="refdb_plot_tax_treemap_+3A_freq_labels">freq_labels</code></td>
<td>
<p>a numeric vector of length two to adjust
the number of printed labels (see Details).
</p>
<p>Only the columns provided in the <code>cols</code> argument are represented
in the treemap. Large labels are printed for the highest rank, while light
text labels are printed for the lowest rank
Intermediate ranks are drawn but their names are not shown.
</p>
<p>The number of labels printed are determined by <code>freq_labels</code>.
The first value gives the threshold
for the highest rank (large labels) and the second value gives
the threshold for the lowest rank (light text labels).
</p>
<p>The underlying graph is computed using the non-exported function
<code>igraph_from_taxo</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <span class="pkg">ggplot2</span> (<span class="pkg">ggraph</span>) object. This means the plot
can be further customized using <span class="pkg">ggplot2</span> compatible functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_plot_tax_treemap(lib)

</code></pre>

<hr>
<h2 id='refdb_report'>Compile a report with different checks</h2><span id='topic+refdb_report'></span>

<h3>Description</h3>

<p>This function produce an HTML report to investigate potential issues
in a reference database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_report(x, file = NULL, view = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_report_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_report_+3A_file">file</code></td>
<td>
<p>the file (path) to write the report. If <code>NULL</code>
the report is written in the user temp directory.</p>
</td></tr>
<tr><td><code id="refdb_report_+3A_view">view</code></td>
<td>
<p>A logical. If <code>TRUE</code> (default), the file is instantly
opened in the web browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function invisibly returns the file where the report was written.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
lib &lt;- refdb_set_fields(lib,
                        taxonomy = c(family = "family_name",
                            genus = "genus_name",
                            species = "species_name"),
                        sequence = "DNA_seq",
                        marker = "marker")
tmp &lt;- tempfile()

refdb_report(lib, tmp, view = FALSE)

</code></pre>

<hr>
<h2 id='refdb_sample_tax'>Sample records within taxa</h2><span id='topic+refdb_sample_tax'></span>

<h3>Description</h3>

<p>This function can be useful to keep a maximum of records per taxa.
This function require dplyr dev version to work because of slice_sample.
Will be exported once available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_sample_tax(x, n_max = 10, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_sample_tax_+3A_x">x</code></td>
<td>
<p>a reference database.</p>
</td></tr>
<tr><td><code id="refdb_sample_tax_+3A_n_max">n_max</code></td>
<td>
<p>maximum number of records to keep for each taxa.</p>
</td></tr>
<tr><td><code id="refdb_sample_tax_+3A_cols">cols</code></td>
<td>
<p>an optional vector of column names.
If <code>NULL</code> (default), the function is applied to the columns
associated with the <code>taxonomy</code> field.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reference database.
</p>

<hr>
<h2 id='refdb_set_fields'>Associate columns to fields</h2><span id='topic+refdb_set_fields'></span>

<h3>Description</h3>

<p>Associate columns to fields so they are recognized
and appropriately treated by refdb functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_set_fields(
  x,
  source = NA,
  id = NA,
  organism = NA,
  taxonomy = NA,
  sequence = NA,
  marker = NA,
  latitude = NA,
  longitude = NA,
  reference = NA,
  config_yaml = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_set_fields_+3A_x">x</code></td>
<td>
<p>a reference database (tibble).</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_source">source</code></td>
<td>
<p>name of the column which contains the data source.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_id">id</code></td>
<td>
<p>name of the column which contains the record IDs.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_organism">organism</code></td>
<td>
<p>name of the column which contains the names of the organisms.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_taxonomy">taxonomy</code></td>
<td>
<p>a vector of column names.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_sequence">sequence</code></td>
<td>
<p>name of the column which contains the sequences.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_marker">marker</code></td>
<td>
<p>name of the column which contains marker names.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_latitude">latitude</code></td>
<td>
<p>name of the column which contains latitudes (WGS 84)</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_longitude">longitude</code></td>
<td>
<p>name of the column which contains longitudes (WGS 84).</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_reference">reference</code></td>
<td>
<p>a vector of column names.</p>
</td></tr>
<tr><td><code id="refdb_set_fields_+3A_config_yaml">config_yaml</code></td>
<td>
<p>a file path to a YAML file</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Taxonomy reordering. NA to ignore, NULL to delete.
Fields set using config_yaml always overwrite those set by arguments
</p>


<h3>Value</h3>

<p>The function returns <code>x</code> with updated attributes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
lib &lt;- refdb_set_fields(lib,
                        taxonomy = c(family = "family_name",
                            genus = "genus_name",
                            species = "species_name"),
                        sequence = "DNA_seq",
                        marker = "marker")

</code></pre>

<hr>
<h2 id='refdb_set_ncbitax'>Replace the current taxonomy using the NCBI Taxonomy database</h2><span id='topic+refdb_set_ncbitax'></span>

<h3>Description</h3>

<p>Replace the current taxonomy using the NCBI Taxonomy database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_set_ncbitax(
  x,
  min_level = "species",
  force_species_name = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_set_ncbitax_+3A_x">x</code></td>
<td>
<p>a reference database (tibble) with one or several columns
giving the taxonomy of each record and explicitly
indicated in the field taxonomy. See <a href="#topic+refdb_set_fields">refdb_set_fields</a>.</p>
</td></tr>
<tr><td><code id="refdb_set_ncbitax_+3A_min_level">min_level</code></td>
<td>
<p>minimum taxonomic level at which taxonomy
should be replaced. Default is the finest level (<code>"species"</code>).</p>
</td></tr>
<tr><td><code id="refdb_set_ncbitax_+3A_force_species_name">force_species_name</code></td>
<td>
<p>if <code>TRUE</code>, species not found in NCBI Taxonomy
will keep their original names instead of NAs.</p>
</td></tr>
<tr><td><code id="refdb_set_ncbitax_+3A_verbose">verbose</code></td>
<td>
<p>print information in the console.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reference database with the NCBI taxonomy
for the genus level and higher ranks.
(the original taxonomy above the genus level is removed).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
lib &lt;- read.csv(system.file("extdata", "baetidae_bold.csv", package = "refdb"))
lib &lt;- refdb_set_fields_BOLD(lib)
refdb_set_ncbitax(lib)

## End(Not run)
</code></pre>

<hr>
<h2 id='refdb_write_fields'>Write fields to a file</h2><span id='topic+refdb_write_fields'></span>

<h3>Description</h3>

<p>This function can be used to save fields defined
using e.g. <code>refdb_set_fields</code> to a file.
Data are saved in YAML and can be read again using the
<code>config_yaml</code> argument of <code>refdb_set_fields</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refdb_write_fields(x, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refdb_write_fields_+3A_x">x</code></td>
<td>
<p>a reference database with some fields to be saved.</p>
</td></tr>
<tr><td><code id="refdb_write_fields_+3A_file">file</code></td>
<td>
<p>a path to the file to write.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for its side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lib &lt;- read.csv(system.file("extdata", "ephem.csv", package = "refdb"))
tmp &lt;- tempfile()
refdb_write_fields(lib, tmp)

</code></pre>

<hr>
<h2 id='refdb-package'>refdb: A DNA Reference Library Manager</h2><span id='topic+refdb'></span><span id='topic+refdb-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Reference database manager offering a set of functions to import, organize, clean, filter, audit and export reference genetic data. Provide functions to download sequence data from Bold Systems (<a href="https://www.boldsystems.org/">https://www.boldsystems.org/</a>) and NCBI GenBank <a href="https://www.ncbi.nlm.nih.gov/genbank/">https://www.ncbi.nlm.nih.gov/genbank/</a>. Designed as an environment for semi-automatic and assisted construction of reference databases and to improve standardization and repeatability in barcoding and metabarcoding studies.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Francois Keck <a href="mailto:francois.keck@gmail.com">francois.keck@gmail.com</a> (<a href="https://orcid.org/0000-0002-3323-4167">ORCID</a>) [copyright holder]
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://fkeck.github.io/refdb/">https://fkeck.github.io/refdb/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/fkeck/refdb/issues">https://github.com/fkeck/refdb/issues</a>
</p>
</li></ul>


<hr>
<h2 id='valid_taxo_rank'>Ranks considered as valid by refdb</h2><span id='topic+valid_taxo_rank'></span>

<h3>Description</h3>

<p>Ranks considered as valid by refdb
</p>


<h3>Usage</h3>

<pre><code class='language-R'>valid_taxo_rank()
</code></pre>


<h3>Value</h3>

<p>a vector of ordered ranks.
</p>


<h3>References</h3>

<p>This is a simplified version of the
list <code>rank_ref</code> available in <span class="pkg">taxize</span>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>valid_taxo_rank()

</code></pre>

<hr>
<h2 id='xml_extract'>Extract XML elements</h2><span id='topic+xml_extract'></span>

<h3>Description</h3>

<p>Combine xml_find_first and xml_text to extract elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xml_extract(x, xpath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xml_extract_+3A_x">x</code></td>
<td>
<p>A document, node, or node set.</p>
</td></tr>
<tr><td><code id="xml_extract_+3A_xpath">xpath</code></td>
<td>
<p>A string containing a xpath expression.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector, the same length as x.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
