<!DOCTYPE html><html><head><title>Help for package refseqR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {refseqR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#refseqR'><p>refseqR: Common computational operations working with GenBank</p></a></li>
<li><a href='#refseq_AAseq'><p>Extract the amino acid sequence into a Biostrings object</p></a></li>
<li><a href='#refseq_CDScoords'><p>Extract the coding sequences (CDS) coordinates from a XM accession</p></a></li>
<li><a href='#refseq_CDSseq'><p>Extract the CDS nucleotide sequence into a Biostrings object</p></a></li>
<li><a href='#refseq_description'><p>Get the sequence Description</p></a></li>
<li><a href='#refseq_fromGene'><p>Get the mRNA or protein accession</p></a></li>
<li><a href='#refseq_fromXM'><p>Get mRNA features</p></a></li>
<li><a href='#refseq_geneSymbol'><p>Get the gene symbol</p></a></li>
<li><a href='#refseq_mol_wt'><p>Extract the molecular weigth from an XP accession</p></a></li>
<li><a href='#refseq_XMfromXP'><p>Get the XM accession from XP accession</p></a></li>
<li><a href='#refseq_XPfromXM'><p>Get the XP accession from XM accession</p></a></li>
<li><a href='#refseq_XPlength'><p>Get the protein length from an XP accession</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Common Computational Operations Working with RefSeq Entries
(GenBank)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jose V. Die &lt;jose.die@uco.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fetches NCBI data (RefSeq <a href="https://www.ncbi.nlm.nih.gov/refseq/">https://www.ncbi.nlm.nih.gov/refseq/</a> database) and provides an environment to  
    extract information at the level of gene, mRNA or protein accessions. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jdieramon/refseqR">https://github.com/jdieramon/refseqR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jdieramon/refseqR/issues">https://github.com/jdieramon/refseqR/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>IRanges, rentrez, tibble, Biostrings</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-15 17:04:22 UTC; jdieramon</td>
</tr>
<tr>
<td>Author:</td>
<td>Jose V. Die <a href="https://orcid.org/0000-0002-7506-8590"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Llu√≠s Revilla Sancho
    <a href="https://orcid.org/0000-0001-9747-2570"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-22 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='refseqR'>refseqR: Common computational operations working with GenBank</h2><span id='topic+refseqR'></span><span id='topic+refseqR-package'></span>

<h3>Description</h3>

<p>refseqR is a framework of common computational operations working with RefSeq entries (GenBank)
</p>


<h3>Author(s)</h3>

<p>Jose V. Die <a href="mailto:jose.die@uco.es">jose.die@uco.es</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/jdieramon/refseqR">https://github.com/jdieramon/refseqR</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/jdieramon/refseqR/issues">https://github.com/jdieramon/refseqR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='refseq_AAseq'>Extract the amino acid sequence into a Biostrings object</h2><span id='topic+refseq_AAseq'></span>

<h3>Description</h3>

<p><code>refseq_AAseq()</code> Parses a single/multiple XP accessions (GenBank format) and extract
the amino acid sequences into a <code>AAStringSet</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_AAseq(xp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_AAseq_+3A_xp">xp</code></td>
<td>
<p>A character string containing a single/multiple XP ids.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of <code>AAStringSet</code> class.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xp = c("XP_004487758", "XP_004488550", "XP_004501961")
my_aa &lt;- refseq_AAseq(xp)
# Now, the `AAStringSet`can be easily used to make a fasta file :
# writeXStringSet(x= my_aa, filepath = "aa_result")


</code></pre>

<hr>
<h2 id='refseq_CDScoords'>Extract the coding sequences (CDS) coordinates from a XM accession</h2><span id='topic+refseq_CDScoords'></span>

<h3>Description</h3>

<p><code>refseq_CDScoords()</code> Parses an XM accession (Genbank format) and extract the CDS coordinates.
The CDS coordinates refer to the mRNA molecule.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_CDScoords(xm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_CDScoords_+3A_xm">xm</code></td>
<td>
<p>A character string of the single/multiple XM id.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>IRanges</code> object with the start and end position of the CDS of the
putative mRNAs.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_CDSseq">refseq_CDSseq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xm = c("XM_004487701")
refseq_CDScoords(xm)


xm = c("XM_004487701", "XM_004488493")
refseq_CDScoords(xm)

</code></pre>

<hr>
<h2 id='refseq_CDSseq'>Extract the CDS nucleotide sequence into a Biostrings object</h2><span id='topic+refseq_CDSseq'></span>

<h3>Description</h3>

<p><code>refseq_CDSseq()</code> Parses a single/multiple XM acessions (Genbank format) and extract
the CDS nucleotide sequences into a <code>DNAStringSet</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_CDSseq(xm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_CDSseq_+3A_xm">xm</code></td>
<td>
<p>A character string of the single/multiple XM id.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of <code>DNAStringSet</code> class.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_CDScoords">refseq_CDScoords</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xm &lt;-  c("XM_004487701", "XM_004488493", "XM_004501904")
my_cds &lt;- refseq_CDSseq(xm)
# Now, the `DNAStringSet` can easily used to make a fasta file :
# writeXStringSet(x= my_cds, filepath = "cds_result")

</code></pre>

<hr>
<h2 id='refseq_description'>Get the sequence Description</h2><span id='topic+refseq_description'></span>

<h3>Description</h3>

<p><code>refseq_description()</code> Returns the sequence description from a single XM, XP, or Gene id. accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_description(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_description_+3A_id">id</code></td>
<td>
<p>A character string of the XP, XM, or Gene id.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the sequence description corresponding to the specified sequence as <code>id</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_XMfromXP">refseq_XMfromXP</a></code> to obtain the XM ids that encode a set of XP ids.
</p>
<p><code><a href="#topic+refseq_XPfromXM">refseq_XPfromXM</a></code> to obtain the XP ids encoded by a set of XM ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Get the sequence descriptions from a set of XM accessions
 xm = c("XM_004487701")
 sapply(xm, function(x) refseq_description(x), USE.NAMES = FALSE)

 # Get the sequence descriptions from a set of XP accessions
 xp = c("XP_004487758")
 sapply(xp, function(x) refseq_description(x), USE.NAMES = FALSE)


#' # Get the sequence descriptions from a set of Gene accessions
locs &lt;- c("LOC101512347", "LOC101506901")
sapply(locs, function(x) refseq_description(x), USE.NAMES = FALSE)

</code></pre>

<hr>
<h2 id='refseq_fromGene'>Get the mRNA or protein accession</h2><span id='topic+refseq_fromGene'></span>

<h3>Description</h3>

<p><code>refseq_fromGene()</code> Returns the XP or XM accession from a single gene id. accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_fromGene(gene_symbol,sequence, retries)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_fromGene_+3A_gene_symbol">gene_symbol</code></td>
<td>
<p>A character string of the gene symbol.</p>
</td></tr>
<tr><td><code id="refseq_fromGene_+3A_sequence">sequence</code></td>
<td>
<p>A character string of the &quot;XM&quot; or &quot;XP&quot; to fetch data from mRNA or protein databases, respectively.</p>
</td></tr>
<tr><td><code id="refseq_fromGene_+3A_retries">retries</code></td>
<td>
<p>A numeric value to control the number of retry attempts to handle 502 errors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the XP or XM id accessions corresponding to the especified <code style="white-space: pre;">&#8288;gene symbol&#8288;</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_XMfromXP">refseq_XMfromXP</a></code> to obtain the XM ids that encode a set of XP ids.
</p>
<p><code><a href="#topic+refseq_XPfromXM">refseq_XPfromXM</a></code> to obtain the XP ids encoded by a set of XM ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get the XM accessions from a set of gene ids
locs &lt;- c("LOC101512347")
sapply(locs, function(x) refseq_fromGene (x, sequence = "XM", retries = 3), USE.NAMES = FALSE)

# Get the XP accessions from a set of gene ids
locs &lt;- c("LOC101512347")
sapply(locs, function(x) refseq_fromGene (x, sequence = "XP", retries = 3), USE.NAMES = FALSE)

</code></pre>

<hr>
<h2 id='refseq_fromXM'>Get mRNA features</h2><span id='topic+refseq_fromXM'></span>

<h3>Description</h3>

<p><code>refseq_fromXM()</code> Returns a number of features from a single/multiple mRNA accession(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_fromXM(xm , feat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_fromXM_+3A_xm">xm</code></td>
<td>
<p>A character string of the XM id.</p>
</td></tr>
<tr><td><code id="refseq_fromXM_+3A_feat">feat</code></td>
<td>
<p>A character string of the selected features. Allowed features: 'caption',
'moltype', 'sourcedb', 'updatedate', 'slen', 'organism', 'title'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tibble</code> of summarized results including columns:
</p>

<ul>
<li><p> caption, mRNA accession
</p>
</li>
<li><p> moltype, type of molecule
</p>
</li>
<li><p> sourcedb, database (GenBank)
</p>
</li>
<li><p> updatedate, date of updated record
</p>
</li>
<li><p> slen, molecule length (in bp)
</p>
</li>
<li><p> organism
</p>
</li>
<li><p> title, sequence description
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_fromGene">refseq_fromGene</a></code> to obtain the XP or XM accession from a single gene id. accession.
</p>
<p><code><a href="#topic+refseq_XPfromXM">refseq_XPfromXM</a></code> to obtain the XP ids encoded by a set of XM ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get several molecular features from a set of mRNA accessions
xm = c("XM_004487701", "XM_004488493", "XM_004501904")
feat = c("caption", "moltype", "sourcedb", "slen")
refseq_fromXM(xm ,feat)

</code></pre>

<hr>
<h2 id='refseq_geneSymbol'>Get the gene symbol</h2><span id='topic+refseq_geneSymbol'></span>

<h3>Description</h3>

<p><code>refseq_geneSymbol()</code> Returns the gene symbol from a single Gene id. accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_geneSymbol (id, db, retries)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_geneSymbol_+3A_id">id</code></td>
<td>
<p>A character string of the XP or XM id.</p>
</td></tr>
<tr><td><code id="refseq_geneSymbol_+3A_db">db</code></td>
<td>
<p>A character string of the &quot;nuccore&quot; or &quot;protein&quot; database.</p>
</td></tr>
<tr><td><code id="refseq_geneSymbol_+3A_retries">retries</code></td>
<td>
<p>A numeric value to control the number of retry attempts to handle internet errors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the gene symbol corresponding to the especified accession as <code>id</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_XMfromXP">refseq_XMfromXP</a></code> to obtain the XM ids that encode a set of XP ids.
</p>
<p><code><a href="#topic+refseq_XPfromXM">refseq_XPfromXM</a></code> to obtain the XP ids encoded by a set of XM ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the gene symbol from a set of XM accessions
xm = c("XM_004487701", "XM_004488493")
sapply(xm, function(x) refseq_geneSymbol (x, db = "nuccore", retries = 3), USE.NAMES = FALSE)

# Get the gene symbol from a set of XP accessions
xp = c("XP_004487758")
sapply(xp, function(x) refseq_geneSymbol (x, db = "protein", retries = 3), USE.NAMES = FALSE)

</code></pre>

<hr>
<h2 id='refseq_mol_wt'>Extract the molecular weigth from an XP accession</h2><span id='topic+refseq_mol_wt'></span>

<h3>Description</h3>

<p><code>refseq_mol_wt()</code> Parses an XP accession output (Genbank format) and extract the molecular weight
(in Daltons).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_mol_wt(xp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_mol_wt_+3A_xp">xp</code></td>
<td>
<p>A character string of the XP id.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First, get the character vector containing the fetched record. Then, this
function parses the fectched record and returns the molecular weight.
</p>


<h3>Value</h3>

<p>A numeric vector represeting the molecular weight of the protein(s) especified as <code>xp</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Get the molecular weight from a single XP accession
 xp &lt;- "XP_020244413"
 refseq_mol_wt(xp)

 # Get the molecular weight from from a set of XP accessions
 xp = c("XP_004487758", "XP_004488550")
 sapply(xp, function(x) refseq_mol_wt(x), USE.NAMES = TRUE)

</code></pre>

<hr>
<h2 id='refseq_XMfromXP'>Get the XM accession from XP accession</h2><span id='topic+refseq_XMfromXP'></span>

<h3>Description</h3>

<p><code>refseq_XMfromXP()</code> Returns the XM accession from a single XP accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_XMfromXP(xp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_XMfromXP_+3A_xp">xp</code></td>
<td>
<p>A character string of the XP id.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the XM ids that encode the XP especified as <code>xp</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_XPfromXM">refseq_XPfromXM</a></code> to obtain the XP ids encoded by a set of XM ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Get the XM id from a single XP accession
 xp &lt;- "XP_020244413"
 refseq_XMfromXP(xp)


 # Get the XM ids from a set of XP accessions
 xp = c("XP_004487758", "XP_004488550")
 sapply(xp, function(x) refseq_XMfromXP(x), USE.NAMES = FALSE)

</code></pre>

<hr>
<h2 id='refseq_XPfromXM'>Get the XP accession from XM accession</h2><span id='topic+refseq_XPfromXM'></span>

<h3>Description</h3>

<p><code>refseq_XPfromXM()</code> Returns the XP accession from a single XM accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_XPfromXM(xm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_XPfromXM_+3A_xm">xm</code></td>
<td>
<p>A character string of the XP id.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the XP id encoded by the XM especified as <code>xm</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_XMfromXP">refseq_XMfromXP</a></code> to obtain the XM ids that encode a set of XP ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Get the XP id from a single XM accession
 xm &lt;- "XM_004487701"
 refseq_XPfromXM(xm)

 
 # Get the XP ids from a set of XM accessions
 xm = c("XM_004487701", "XM_004488493")
 sapply(xm, function(x) refseq_XPfromXM(x), USE.NAMES = FALSE) 

</code></pre>

<hr>
<h2 id='refseq_XPlength'>Get the protein length from an XP accession</h2><span id='topic+refseq_XPlength'></span>

<h3>Description</h3>

<p><code>refseq_XPlength()</code> Returns the amino acid length from a single XP accession.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseq_XPlength(xp, retries)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refseq_XPlength_+3A_xp">xp</code></td>
<td>
<p>A character string of the XP id.</p>
</td></tr>
<tr><td><code id="refseq_XPlength_+3A_retries">retries</code></td>
<td>
<p>A numeric value to control the number of retry attempts to handle internet errors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the aa length of the protein especified as <code>xp</code>.
</p>


<h3>Author(s)</h3>

<p>Jose V. Die
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refseq_XPfromXM">refseq_XPfromXM</a></code> to obtain the XP ids encoded by a set of XM ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Get the XM ids from a set of XP accessions
 xp = c("XP_004487758", "XP_004488550")
 sapply(xp, function(x) refseq_XPlength(x, retries = 3), USE.NAMES = FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
