<!DOCTYPE html><html><head><title>Help for package REGENT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {REGENT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#REGENT-package'>
<p>Risk Estimation for Genetic and Environmental Traits</p>
</p></a></li>
<li><a href='#EnvironmentalA'><p>Example file for single level environmental factors</p></a></li>
<li><a href='#EnvironmentalB'><p>Example file for single multiple level environmental factors</p></a></li>
<li><a href='#GeneticA'><p>Example file for SNPs conferring multiplicative risks</p></a></li>
<li><a href='#GeneticB'><p>Example file for SNPs conferring additive risks</p></a></li>
<li><a href='#Inds'><p>Example of  individual file for REGENT.predict</p></a></li>
<li><a href='#REGENT.model'><p>REGENT.model</p></a></li>
<li><a href='#REGENT.predict'><p>REGENT.predict</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Risk Estimation for Genetic and Environmental Traits</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-08-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel J.M. Crouch, Graham H.M. Goddard &amp; Cathryn M. Lewis</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Crouch &lt;djmcrouch@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Produces population distribution of disease risk and statistical risk categories, and predicts risks for individuals with genotype information.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.14.0)</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-08-19 09:15:41 UTC; dancrouch</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-08-19 13:49:39</td>
</tr>
</table>
<hr>
<h2 id='REGENT-package'>
Risk Estimation for Genetic and Environmental Traits
</h2><span id='topic+REGENT'></span><span id='topic+REGENT-package'></span>

<h3>Description</h3>

<p>Provides risk estimation and categorisation for populations and individuals
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">

Package: </td><td style="text-align: left;"> REGENT</td>
</tr>
<tr>
 <td style="text-align: left;">

Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">

Version: </td><td style="text-align: left;"> 1.0.6</td>
</tr>
<tr>
 <td style="text-align: left;">

Date: </td><td style="text-align: left;"> 2015-08-18</td>
</tr>
<tr>
 <td style="text-align: left;">

License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">

LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Daniel Crouch, Graham Goddard &amp; Cathryn Lewis.
</p>
<p>Maintainer: Daniel Crouch - djmcrouch@gmail.com
</p>


<h3>References</h3>

<p>Crouch, Goddard &amp; Lewis (2011)
</p>
<p>Goddard &amp; Lewis, Risk categorization for complex disorders according to genotype relative risk and precision in parameter estimates (2010)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+REGENT.model">REGENT.model</a></code>,<code><a href="#topic+REGENT.predict">REGENT.predict</a></code>,<code><a href="#topic+GeneticA">GeneticA</a></code>,<code><a href="#topic+GeneticB">GeneticB</a></code>,<code><a href="#topic+EnvironmentalA">EnvironmentalA</a></code>,<code><a href="#topic+EnvironmentalB">EnvironmentalB</a></code>,<code><a href="#topic+Inds">Inds</a></code>
</p>

<hr>
<h2 id='EnvironmentalA'>Example file for single level environmental factors</h2><span id='topic+EnvironmentalA'></span>

<h3>Description</h3>

<p>Example data for Crohns disease in correct input format. Also as text file &quot;EnvironmentalA.txt&quot; in the data folder for this package. data from Calkins, BM (1989).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("REGENT")</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>References</h3>

<p>from Calkins, BM (1989) A meta-analysis of the role of smoking in inflammatory bowel disease
</p>

<hr>
<h2 id='EnvironmentalB'>Example file for single multiple level environmental factors</h2><span id='topic+EnvironmentalB'></span>

<h3>Description</h3>

<p>Example data for Crohns disease in correct input format. Also as text file &quot;EnvironmentalB.txt&quot; in the data folder for this package. *PLEASE NOTE that the multilevel smoking data in EnvironmentalB is entirely artificial.*
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("REGENT")</code></pre>


<h3>Format</h3>

<p>data frame</p>

<hr>
<h2 id='GeneticA'>Example file for SNPs conferring multiplicative risks</h2><span id='topic+GeneticA'></span>

<h3>Description</h3>

<p>Example data for Crohns disease in correct input format. Also as text file &quot;GeneticA.txt&quot; in the data folder for this package. Data is from Franke et al (2010), 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("REGENT")</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>References</h3>

<p>Franke et al. (2010), Genome-wide meta-analysis increases to 71 the number of confirmed Crohn's disease susceptibility loci  
</p>

<hr>
<h2 id='GeneticB'>Example file for SNPs conferring additive risks</h2><span id='topic+GeneticB'></span>

<h3>Description</h3>

<p>Example data for Crohns disease in correct input format. Also as text file &quot;GeneticB.txt&quot; in the data folder for this package. Data is from Franke et al (2010), 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("REGENT")</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>References</h3>

<p>Franke et al. (2010), Genome-wide meta-analysis increases to 71 the number of confirmed Crohn's disease susceptibility loci  
</p>

<hr>
<h2 id='Inds'>Example of  individual file for REGENT.predict </h2><span id='topic+Inds'></span>

<h3>Description</h3>

<p>Columns for risk factors, rows for individuals. Entries refer to genotypes (number of risk alleles as defined in the locus file, ie can also be protective alleles) or exposure levels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("REGENT")</code></pre>


<h3>Format</h3>

<p>data frame</p>

<hr>
<h2 id='REGENT.model'>REGENT.model</h2><span id='topic+REGENT.model'></span>

<h3>Description</h3>

<p>REGENT.model provides the population distribution of risk and proportion of the population in each risk category based on genetic(SNP)  and environmental exposures.</p>


<h3>Usage</h3>

<pre><code class='language-R'>
REGENT.model(AnalysisName,LocusFile=NULL,EnvFile=NULL
,prev=0.001,cv=0.05,alpha=0.05,sims=100000
,indsims=100000,SmallSampAdjust=0.5,BaseRange=0.01
,PlotMax=5,Block=100)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="REGENT.model_+3A_analysisname">AnalysisName</code></td>
<td>
<p>String, must be provided. Output files will be named according to this argument. Running multiple analyses with the same name will cause previous files to be overwritten.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_locusfile">LocusFile</code></td>
<td>
<p>File path string. Location of file containing table of SNP input data. Required columns should have headers SNP, MAF, Ncase, Ncontrol. Risks should either be provided in one column with header RR, or two columns with headers RR_het and RR_hom. Other columns may be present but will not be used in the analysis. Each SNP is a row. Additional columns may be provided but will be ignored.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_envfile">EnvFile</code></td>
<td>
<p> File path string. Location of file containing table of environmental risk data. Required columns should have headers Factor, Exposure, RR, SE. If multiple exposure levels exist, then the columns should be named Factor, RR1, Exposure1, SE1, RR2, Exposure2, SE2, etc. Each factor is a row. Additional columns may be provided but will be ignored</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_prev">prev</code></td>
<td>
<p>Prevalance of the disease or trait. Default 0.001.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_cv">cv</code></td>
<td>
<p>Coefficient of variation. Default 0.05.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_alpha">alpha</code></td>
<td>
<p>One minus the desired width of confidence intervals around multilocus risk estimates. Default 0.05 giving 95 percent confidence intervals.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_sims">sims</code></td>
<td>
<p>Number of simulations to perform for each single factor risk estimate, for obtaining confidence intervals. Default 100000.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_indsims">indsims</code></td>
<td>
<p>Number of individuals in the simulated population, for obtaining multilocus genotype frequencies. Default 100000</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_smallsampadjust">SmallSampAdjust</code></td>
<td>
<p>Adjustment for small sample sizes, when calculating the standard error of homozygous risk genotypes. Default 0.5</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_baserange">BaseRange</code></td>
<td>
<p>Proportion of population used to calculate the baseline risk (the risk closest to the average in the population). This is to avoid choosing rare, uncertain risk estimates by chance. Default 0.01.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_plotmax">PlotMax</code></td>
<td>
<p>Value at which to truncate the Y-axis of risk distribution plots. High risks are typically rare and of less interest when assessing the distribution in the population. Default 5.</p>
</td></tr>
<tr><td><code id="REGENT.model_+3A_block">Block</code></td>
<td>
<p>Number of multilocus genotypes held in memory during confidence interval calculation. Higher values should decrease computation time. We advise increasing this substantially (10000+) on high performance systems. Default 100.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>4 files are created by REGENT.model.A)All model details, inputs and log information are written to the main output file which is named after the argument provided to AnalysisName.B)Colour and C)greyscale plots of the risk distribution are also provided, and D)the raw data used to create these in a text file.
</p>
<p>See the example folder included in  this package for the correct input file format. 
</p>


<h3>Value</h3>

<p>A  list including elements
</p>
<table>
<tr><td><code>categories</code></td>
<td>
<p>Table giving upper and lower  boundaries for each risk category: Reduced, Average, Elevated and High.</p>
</td></tr>
<tr><td><code>baseline</code></td>
<td>
<p>Single value specifying the baseline risk before rebasing; required when passing the object to REGENT.predict</p>
</td></tr>
<tr><td><code>LocusFile</code></td>
<td>
<p>Table of genetic data used for analysis. NULL if argument LocusFile was set to NULL.</p>
</td></tr>
<tr><td><code>EnvFile</code></td>
<td>
<p>Table of environmental data used for analysis. NULL if argument EnvFile was set to NULL.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Graham Goddard, Daniel Crouch and Cathryn Lewis. Email: djmcrouch@gmail.com </p>


<h3>See Also</h3>

<p><code><a href="#topic+REGENT.predict">REGENT.predict</a></code>,<code><a href="#topic+GeneticA">GeneticA</a></code>,<code><a href="#topic+GeneticB">GeneticB</a></code>,<code><a href="#topic+EnvironmentalA">EnvironmentalA</a></code>,<code><a href="#topic+EnvironmentalB">EnvironmentalB</a></code>,<code><a href="#topic+Inds">Inds</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(REGENT)

#Load example data from package

data("REGENT")

write.table(GeneticA,file="GeneticA.txt")
write.table(GeneticB,file="GeneticB.txt")
write.table(EnvironmentalA,file="EnvironmentalA.txt")
write.table(EnvironmentalB,file="EnvironmentalB.txt")

x=REGENT.model(AnalysisName="Example",LocusFile="GeneticA.txt",EnvFile="EnvironmentalA.txt")

x

</code></pre>

<hr>
<h2 id='REGENT.predict'>REGENT.predict</h2><span id='topic+REGENT.predict'></span>

<h3>Description</h3>

<p>REGENT.predict takes genotype and exposure information for individuals and calculates their absolute and relative risk of disease, and categorises them as reduced, average, elevated or high risk based on the risk categorisation model determined by REGENT.model.</p>


<h3>Usage</h3>

<pre><code class='language-R'>
REGENT.predict(AnalysisName,model,ind,prev=0.001,cv=0.05,sims=100000,Block=100,alpha=0.05,
SmallSampAdjust=0.5)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="REGENT.predict_+3A_analysisname">AnalysisName</code></td>
<td>
<p>String, must be provided. The output file will be named according to this argument, with the suffix &quot;_Predictions.txt&quot;. Running multiple analyses with the same name will cause previous files to be overwritten.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_model">model</code></td>
<td>
<p>Must be provided. Either a file path string giving the location of a file created by REGENT.model (the main file containing model information), or a variable containing the object returned by REGENT.model.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_ind">ind</code></td>
<td>
<p>Must be provided. File path giving the location of individual file, which should have columns for each risk factor (with header of SNP names or Factor names as provided to REGENT.model) and a row for each individual. Genotypes are encoded 0, 1 or 2 describing the number of copies of the risk allele (as defined in the model). Environmental factors are encoded 0, 1, 2, 3 etc. depending on how many exposure levels were modelled. The row header contains individual names.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_prev">prev</code></td>
<td>
<p>Prevalance of the disease or trait. Default 0.001.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_cv">cv</code></td>
<td>
<p>Coefficient of variation. Default 0.05.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_sims">sims</code></td>
<td>
<p>Number of simulations to perform for each single factor risk estimate, for obtaining confidence intervals. Default 100000.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_block">Block</code></td>
<td>
<p>Number of multilocus genotypes held in memory during confidence interval calculation. Higher values should decrease computation time. We advise increasing this substantially (10000+) on high performance systems. Default 100.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_alpha">alpha</code></td>
<td>
<p>One minus the desired width of confidence intervals around multilocus risk estimates. Default 0.05 giving 95 percent confidence intervals.</p>
</td></tr>
<tr><td><code id="REGENT.predict_+3A_smallsampadjust">SmallSampAdjust</code></td>
<td>
<p>Adjustment for small sample sizes, when calculating the standard error of homozygous risk genotypes. Default 0.5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Email: djmcrouch@gmail.com 
</p>
<p>One file is created by REGENT.predict, with the contents of the returned object and the input parameters/data, plus analysis log. 
</p>
<p>See the example folder included in  this package for the correct input file format. 
</p>


<h3>Value</h3>

<p>Table with columns: Absolute risk, genotype relative risk, lower confidence interval, upper confidence interval, risk category, and borderline category status. 
</p>


<h3>Author(s)</h3>

<p>Graham Goddard, Daniel Crouch and Cathryn Lewis </p>


<h3>See Also</h3>

<p><code><a href="#topic+REGENT.model">REGENT.model</a></code>,<code><a href="#topic+GeneticA">GeneticA</a></code>,<code><a href="#topic+GeneticB">GeneticB</a></code>,<code><a href="#topic+EnvironmentalA">EnvironmentalA</a></code>,<code><a href="#topic+EnvironmentalB">EnvironmentalB</a></code>,<code><a href="#topic+Inds">Inds</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Load example data from package

library(REGENT)

data("REGENT")

write.table(GeneticA,file="GeneticA.txt")
write.table(GeneticB,file="GeneticB.txt")
write.table(EnvironmentalA,file="EnvironmentalA.txt")
write.table(EnvironmentalB,file="EnvironmentalB.txt")
write.table(Inds,file="Inds.txt")

#Create model

x=REGENT.model(AnalysisName="Example",LocusFile="GeneticB.txt",EnvFile="EnvironmentalA.txt")

#Option 1, read model from object

y=REGENT.predict(AnalysisName="Example",model=x,ind="Inds.txt")

#Option 2, read model from file

y=REGENT.predict(AnalysisName="Example",model="Example.txt",ind="Inds.txt")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
