<!DOCTYPE html><html><head><title>Help for package rehh.data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rehh.data}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rehh.data-package'>
<p>Data Only: Searching for Footprints of Selection using Haplotype Homozygosity Based Tests</p>
Description: Contains example data for the 'rehh' package.</a></li>
<li><a href='#wgscan.cgu'><p>Whole genome scan results for the CGU (Creole from Guadeloupe island)</p></a></li>
<li><a href='#wgscan.eut'><p>Whole genome scan results for a pool of European taurine cattle</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Maintainer:</td>
<td>Mathieu Gautier &lt;mathieu.gautier@supagro.inra.fr&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Mathieu Gautier, Alexander Klassmann and Renaud Vitalis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Only: Searching for Footprints of Selection using Haplotype
Homozygosity Based Tests</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains example data for the 'rehh' package. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-11-07 17:07:09 UTC; mathieu</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-11-08 11:10:29</td>
</tr>
</table>
<hr>
<h2 id='rehh.data-package'>
Data Only: Searching for Footprints of Selection using Haplotype Homozygosity Based Tests
Description: Contains example data for the 'rehh' package.
</h2><span id='topic+rehh.data-package'></span><span id='topic+rehh.data'></span>

<h3>Description</h3>

<p>Contains example data for the package rehh.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> rehh.data</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.0</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL(&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Index:
</p>
<pre>
wgscan.cgu              Whole genome scan results for the CGU
                        (Creole from Guadeloupe island)
wgscan.eut              Whole genome scan results for a pool of
                        European taurine cattle
</pre>


<h3>References</h3>

<p>Gautier M., Klassmann A., and Vitalis R. (2016). rehh: An R package to detect footprints of selection in genome-wide SNP data from haplotype structure.
<em>Molecular Ecology Resources</em>, submitted
</p>
<p>Gautier M. and Vitalis R. (2012). rehh: An R package to detect footprints of 
selection in genome-wide SNP data from haplotype structure. <em>Bioinformatics</em>, <b>28</b>(8), 1176&ndash;1177.
</p>
<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World
Creole cattle breed. <em>Molecular Ecology</em>, <b>20</b>, 3128&ndash;3143.
</p>

<hr>
<h2 id='wgscan.cgu'>Whole genome scan results for the CGU (Creole from Guadeloupe island)</h2><span id='topic+wgscan.cgu'></span>

<h3>Description</h3>

<p>A dataframe object with of 44,057 rows (SNPs) and 7 columns: i) chromosome name (CHR), ii) position of the SNP in bp (POSITION), iii) Ancestral allele frequency (freq_A), iv) iHH for the ancestral allele (iHH_A), v) iHH for the derived allele (iHH_D), vi) iES using the estimator by Tang et al. (2007) (iES_Tang_et_al_2007), vii) iES using the estimator by Sabeti et al. (2007) (iES_Sabeti_et_al_2007).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(wgscan.cgu)</code></pre>


<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World
Creole cattle breed. <em>Molecular Ecology</em>, <b>20</b>, 3128&ndash;3143.
</p>

<hr>
<h2 id='wgscan.eut'>Whole genome scan results for a pool of European taurine cattle</h2><span id='topic+wgscan.eut'></span>

<h3>Description</h3>

<p>A dataframe object with 44,057 rows (SNPs) and 7 columns: i) chromosome name (CHR), ii) position of the SNP in bp (POSITION), iii) Ancestral allele frequency (freq_A), iv) iHH for the ancestral allele (iHH_A), v) iHH for the derived allele (iHH_D), vi) iES using the estimator by Tang et al. (2007) (iES_Tang_et_al_2007), vii) iES using the estimator by Sabeti et al. (2007) (iES_Sabeti_et_al_2007).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(wgscan.eut)</code></pre>


<h3>References</h3>

<p>Gautier, M. and Naves, M. (2011). Footprints of selection in the ancestral admixture of a New World
Creole cattle breed. <em>Molecular Ecology</em>, <b>20</b>, 3128&ndash;3143.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
