<!DOCTYPE html><html lang="en"><head><title>Help for package Relatedness</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Relatedness}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Relatedness-package'>
<p>Maximum Likelihood Estimation of Relatedness using EM Algorithm</p></a></li>
<li><a href='#BoucleEMacc'>
<p>C code for the EM</p></a></li>
<li><a href='#Cross'>
<p>Crossing matrix</p></a></li>
<li><a href='#Delta'>
<p>List of Relatedness Coefficients</p></a></li>
<li><a href='#Frequencies'>
<p>Allele Frequencies</p></a></li>
<li><a href='#Genotype'>
<p>Genotype Matrix</p></a></li>
<li><a href='#RelCoef'><p>Relatedness Coefficients Estimation for individuals</p></a></li>
<li><a href='#RelCoefLine'><p>Relatedness Coefficients Estimation for Lines</p></a></li>
<li><a href='#RelComb'><p>Computation of Linear Combination of Relatedness Coefficients</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Maximum Likelihood Estimation of Relatedness using EM Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-11-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Fabien Laporte, Tristan Mary-Huard</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fabien Laporte &lt;fabien.laporte@inra.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Inference of relatedness coefficients from a bi-allelic genotype matrix using a Maximum Likelihood estimation, Laporte, F., Charcosset, A. and Mary-Huard, T. (2017) &lt;<a href="https://doi.org/10.1111%2Fbiom.12634">doi:10.1111/biom.12634</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-11-17 08:41:06 UTC; fabien</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-11-17 10:51:45 UTC</td>
</tr>
</table>
<hr>
<h2 id='Relatedness-package'>
Maximum Likelihood Estimation of Relatedness using EM Algorithm
</h2><span id='topic+Relatedness-package'></span><span id='topic+Relatedness'></span>

<h3>Description</h3>

<p>Inference of relatedness coefficients from a bi-allelic genotype matrix using a Maximum Likelihood estimation, Laporte, F., Charcosset, A. and Mary-Huard, T. (2017) &lt;doi:10.1111/biom.12634&gt;.
</p>


<h3>Details</h3>

<p>This package infers the relatedness distribution coefficients for all couple of individuals in a set from their genotype, provided in a bi-allelic genotype matrix. The main function is 'RelCoef' which infers those coefficients. The arguments of this function are a genotype matrix for individuals and an frequency matrix that displays the allelic frequency at each marker in each population. Alternatively, a parental genotype matrix and a crossing matrix can be used. Additional information about structure membership can also be provided via a ParentPop vector (for more details see the help of 'RelCoef'). The main matrix is writen with C language, make sure you can use this code.
</p>


<h3>Author(s)</h3>

<p>Fabien Laporte, Tristan Mary-Huard
</p>
<p>Maintainer: Fabien Laporte &lt;fabien.laporte@inra.fr&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require('Relatedness')
data(Genotype)
data(Frequencies)
data(Cross)
RelCoef(IndividualGenom=matrix(0,ncol=0,nrow=0),ParentalLineGenom=Genotype,
Freq=Frequencies,Crossing=Cross,ParentPop=rep(1,20),Phased=TRUE,NbCores=2)
</code></pre>

<hr>
<h2 id='BoucleEMacc'>
C code for the EM
</h2><span id='topic+BoucleEMacc'></span>

<h3>Description</h3>

<p>A C code used in the function 'RelCoeff'.
</p>

<hr>
<h2 id='Cross'>
Crossing matrix
</h2><span id='topic+Cross'></span>

<h3>Description</h3>

<p>The crossing matrix for the example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Cross")</code></pre>


<h3>Format</h3>

<p>The format is:
int [1:5, 1:2] 1 2 3 4 5 6 7 8 9 10
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cross)
head(Cross)
</code></pre>

<hr>
<h2 id='Delta'>
List of Relatedness Coefficients
</h2><span id='topic+Delta'></span>

<h3>Description</h3>

<p>A list of relatedness coefficients obtained with the example of <cite>RelCoef</cite>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Delta")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 15
$ Delta1 : num [1:4, 1:4] 0.1872 0 0 0 0.0199 ...
$ Delta2 : num [1:4, 1:4] 0.00 0.00 0.00 0.00 3.55e-05 ...
$ Delta3 : num [1:4, 1:4] 0 0 0 0 0.0472 ...
$ Delta4 : num [1:4, 1:4] 0 0 0 0 0.0322 ...
$ Delta5 : num [1:4, 1:4] 0 0 0 0 0.0871 ...
$ Delta6 : num [1:4, 1:4] 0 0 0 0 0.0395 ...
$ Delta7 : num [1:4, 1:4] 0 0 0 0 0.0429 ...
$ Delta8 : num [1:4, 1:4] 0 0 0 0 0.0386 ...
$ Delta9 : num [1:4, 1:4] 0.8128 0 0 0 0.0202 ...
$ Delta10: num [1:4, 1:4] 0 0 0 0 0.000731 ...
$ Delta11: num [1:4, 1:4] 0 0 0 0 0.028 ...
$ Delta12: num [1:4, 1:4] 0 0 0 0 0.0849 ...
$ Delta13: num [1:4, 1:4] 0 0 0 0 0.0174 ...
$ Delta14: num [1:4, 1:4] 0 0 0 0 0.0437 ...
$ Delta15: num [1:4, 1:4] 0 0 0 0 0.498 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Delta)
print(Delta$Delta7)
</code></pre>

<hr>
<h2 id='Frequencies'>
Allele Frequencies
</h2><span id='topic+Frequencies'></span>

<h3>Description</h3>

<p>The allele frequencies matrix for the example with 5000 markers and one population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Frequencies")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:5000, 1:2] 0.268 0.786 0.804 0.238 0.235 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Frequencies)
head(Frequencies)
</code></pre>

<hr>
<h2 id='Genotype'>
Genotype Matrix
</h2><span id='topic+Genotype'></span>

<h3>Description</h3>

<p>The Parental Line Genom matrix for the example with 10 parental lines genotyped with 5000 markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Genotype")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:5000, 1:10] 0 1 0 0 0 1 1 0 1 1 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Genotype)
head(Genotype)
</code></pre>

<hr>
<h2 id='RelCoef'>Relatedness Coefficients Estimation for individuals
</h2><span id='topic+RelCoef'></span>

<h3>Description</h3>

<p>This function performs Maximum Likelihood estimation for the relatedness coefficients between individuals based on a bi-allelic genotype matrix. Alternatively, a parental genotype matrix and a crossing matrix can be used. In that case information about structure can also be taken into account via a ParentPop vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RelCoef(IndividualGenom = matrix(0, nrow=0, ncol=0), 
        ParentalLineGenom = matrix(0, nrow=0, ncol=0), 
        Freq = matrix(0, nrow=0, ncol=0), 
        Crossing = matrix(0, nrow=0, ncol=0), ParentPop = rep(0,0),
        Combination = list(), Phased = FALSE, Details = FALSE,
        NbInit = 5, Prec = 10^(-4), NbCores = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RelCoef_+3A_individualgenom">IndividualGenom</code></td>
<td>
<p>Genotype matrix of individuals. Each individual is described by 2 columns. Each row corresponds to a marker. Entries of matrix IndividualGenom should be either 0 or 1. Either IndividualGenom or ParentalLineGenom has to be provided.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_parentallinegenom">ParentalLineGenom</code></td>
<td>
<p>Genotype matrix of parental lines. Each parental line is described by one column with rows corresponding to markers. Entries of ParentalLineGenome should be either 0 or 1.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_freq">Freq</code></td>
<td>
<p>Allelic frequencies for allele 1 at each markers and for all populations (one column per population, one line per marker).
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_crossing">Crossing</code></td>
<td>
<p>Required when argument ParentalLineGenom is provided. A 2-column matrix where each row corresponds to a crossing between 2 parents. Parents should be numbered according to their order of appearance in the ParentalLineGenom matrix.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_parentpop">ParentPop</code></td>
<td>
<p>Only available if ParentalLineGenom is displayed. A vector of numbers corresponding to population membership for the parental lines.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_combination">Combination</code></td>
<td>
<p>If provided, a list of vector with two components. The jth vector is composed with the number of the first hybrid and the number of the second hybrid of the jth couple to study.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_phased">Phased</code></td>
<td>
<p>A Boolean with value TRUE if observations are phased.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_details">Details</code></td>
<td>
<p>A Boolean variable. If TRUE, the relatedness mode graph is displayed.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_nbinit">NbInit</code></td>
<td>
<p>Number of initial values for the EM algorithm.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_prec">Prec</code></td>
<td>
<p>Convergence precision parameter for the EM algorithm.
</p>
</td></tr>
<tr><td><code id="RelCoef_+3A_nbcores">NbCores</code></td>
<td>
<p>Number of cores used by the algorithm (Default is the number of cores available minus one). Only available for linux and Mac.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Argument IndividualGenom should be used if the available data consist in genotypic information only. By default the data are assumed to be unphased and the function returns 9 relatedness coefficients. If data are phased, use argument Phased = TRUE to obtain the 15 relatedness coefficients. Note that in that case the ordering of the 2 columns per individual in IndividualGenome does matter. Alternatively, if the genotyped individuals are hybrids resulting from the crossing of parental lines (or combinations of parental gametes), it is possible to provide a ParentalLineGenom and a Crossing matrix directly. Additionally, the population membership of the parents can be provided via argument ParentPop. Whatever the arguments used to enter the genotypic data, the allelic frequencies of the markers have to be provided using argument Freq. Arguments NbInit and Prec are tuning parameters for the EM algorithm used for likelihood maximization. 
</p>


<h3>Value</h3>

<p>By default, relatedness coefficients are displayed for all couple of genotyped individuals (or hybrids). In that case the function returns a list of matrices, each corresponding to a specific relatedness coefficients (details about relatedness coefficients can be obtained by displaying the relatedness mode graph with argument Details). Element (i,j) of matrix k corresponds to the kth estimated relatedness coefficient for the couple of individuals i and j. Alternatively, if a list of couples is specified with argument Combination, the function returns a list of vectors (each vector corresponding to an relatedness coefficient). In that case element i of vector k corresponds to the kth relatedness coefficient of the ith couple specified in Combination.
</p>


<h3>Warning </h3>

<p>In absence of population structure, some relatedness coefficients are not identifiable.
Since an EM algorithm is run for each couple of individuals, the procedure can be time consuming for large panels.
</p>


<h3>Author(s)</h3>

<p>Fabien Laporte, 'UMR Genetique Quantitative et Evolution' INRA France.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require('Relatedness')
data(Genotype)
data(Frequencies)
data(Cross)
RelatednessCoefficient &lt;- RelCoef(IndividualGenom=matrix(0,ncol=0,nrow=0),
				  ParentalLineGenom=Genotype,
				  Freq=Frequencies,Crossing=Cross,
				  ParentPop=rep(1,8),Phased=TRUE,NbCores=2)
print(RelatednessCoefficient$Delta3)
</code></pre>

<hr>
<h2 id='RelCoefLine'>Relatedness Coefficients Estimation for Lines
</h2><span id='topic+RelCoefLine'></span>

<h3>Description</h3>

<p>This function performs Maximum Likelihood estimation for the relatedness coefficients between lines based on a bi-allelic genotype matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RelCoefLine(LineGenom = matrix(0,nrow=0,ncol=0), 
	    Freq = matrix(0,nrow=0,ncol=0), 
	    LinePop = rep(0,0), 
	    Combination = NULL, 
	    NbInit = 5, Prec = 10^(-4), NbCores = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RelCoefLine_+3A_linegenom">LineGenom</code></td>
<td>
<p>Genotype matrix of lines. Each line is described by 1 column. Each row corresponds to a marker. Entries of matrix Genotype should be either 0 or 1.
</p>
</td></tr>
<tr><td><code id="RelCoefLine_+3A_freq">Freq</code></td>
<td>
<p>Allelic frequencies for allele 1 at each markers and for all populations (one column per population, one line per marker).
</p>
</td></tr>
<tr><td><code id="RelCoefLine_+3A_linepop">LinePop</code></td>
<td>
<p>A vector of numbers corresponding to population membership for the parental lines.
</p>
</td></tr>
<tr><td><code id="RelCoefLine_+3A_combination">Combination</code></td>
<td>
<p>If provided, a list of vector with two components. The jth vector is composed with the number of the first hybrid and the number of the second hybrid of the jth couple to study.
</p>
</td></tr>
<tr><td><code id="RelCoefLine_+3A_nbinit">NbInit</code></td>
<td>
<p>Number of initial values for the EM algorithm.
</p>
</td></tr>
<tr><td><code id="RelCoefLine_+3A_prec">Prec</code></td>
<td>
<p>Convergence precision parameter for the EM algorithm.
</p>
</td></tr>
<tr><td><code id="RelCoefLine_+3A_nbcores">NbCores</code></td>
<td>
<p>Number of cores used by the algorithm (Default is the number of cores available minus one). Only available for linux and Mac.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default, relatedness coefficients are displayed for all couple of genotyped lines. In that case the function returns a matrix corresponding to the Simple Relatedness Coefficient, i.e. the probability that each couple of lines are related. Element (i,j) of the matrix corresponds to the estimated relatedness coefficient for the couple of lines i and j. Alternatively, if a list of couples is specified with argument Combination, the function returns a list of coefficients (each coefficient corresponding to an relatedness coefficient). In that case element i of the list corresponds to the estimated relatedness coefficient of the ith couple specified in Combination.
</p>


<h3>Author(s)</h3>

<p>Fabien Laporte, 'UMR Genetique Quantitative et Evolution' INRA France.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require('Relatedness')
data(Genotype)
data(Frequencies)
data(Cross)
SimpleRelatedness &lt;- RelCoefLine(LineGenom=Genotype,Freq=Frequencies,
LinePop=rep(1,8),NbCores=2)
print(SimpleRelatedness)
</code></pre>

<hr>
<h2 id='RelComb'>Computation of Linear Combination of Relatedness Coefficients
</h2><span id='topic+RelComb'></span>

<h3>Description</h3>

<p>Compute any relatedness synthetic criterion based on a linear combination of the relatedness coefficients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RelComb(Combination, Delta, 
		 Crossing = matrix(0, nrow = 0, ncol = 0), 
		 ParentPop = rep(0, 0), 
		 ShowIdentifiable = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RelComb_+3A_combination">Combination</code></td>
<td>
<p>A vector, with length identical to the length of <code>Delta</code> where the kth element corresponds to the coefficient for the kth relatedness coefficient, 
or alternatively a character in the following list: 
'simple relatedness', 
'double relatedness', 
'first inbreeding', 
'second inbreeding', 
'double inbreeding'.
</p>
</td></tr>
<tr><td><code id="RelComb_+3A_delta">Delta</code></td>
<td>
<p>A list of matrices, each corresponding to a specific relatedness coefficients. Element (i,j) of matrix k corresponds to the kth estimated relatedness coefficient for the couple of individuals i and j.
</p>
</td></tr>
<tr><td><code id="RelComb_+3A_crossing">Crossing</code></td>
<td>
<p>A 2-column matrix where each row corresponds to a crossing between 2 parents. Parents should be numbered.
</p>
</td></tr>
<tr><td><code id="RelComb_+3A_parentpop">ParentPop</code></td>
<td>
<p>A vector of numbers corresponding to population membership for the parental lines.
</p>
</td></tr>
<tr><td><code id="RelComb_+3A_showidentifiable">ShowIdentifiable</code></td>
<td>
<p>A boolean describing whether the combination should be displayed only for identifiable cases. Default value is TRUE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can be applied to a list of relatedness coefficients - as produced by the RelCoeff function - to compute any synthetic criterion based on a linear combination of the relatedness coefficients, for all couples. Additional information Crossing and ParentPop are required if they were used in the RelCoeff function to obtain the relatedness coefficients. The function automatically checks the identifiability of the combination to be evaluated. Several classical genetic criteria are implemented by default and can be computed using the Csuros argument. Alternatively, the user can provide a vector of coefficients to be applied through the Combination argument.
</p>


<h3>Value</h3>

<p>If identifiability is satisfied for all pairs of individuals, all criteria are computed and returned in a matrix. If identifiability is not guaranteed for all pairs, the function will return a matrix with NA entries for the potentially non-estimable pairs. This by-default behavior can be bypassed if required by setting ShowIdentifiable to FALSE.
</p>


<h3>Author(s)</h3>

<p>Fabien Laporte, 'UMR Genetique Quantitative et Evolution' INRA France.
</p>


<h3>References</h3>

<p>Csuros M (2014) Non-identifiability of identity coefficients at biallelic loci. <em>Theoretical Population Biology</em> 92: 22-24.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require('Relatedness')
data(Delta)
RelatednessComb &lt;- RelComb(Combination='simple relatedness', Delta, ShowIdentifiable = TRUE)
print(RelatednessComb)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
