<!DOCTYPE html><html><head><title>Help for package rentrez</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rentrez}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rentrez'><p>rentrez</p></a></li>
<li><a href='#entrez_citmatch'><p>Fetch pubmed ids matching specially formatted citation strings</p></a></li>
<li><a href='#entrez_db_links'><p>List available links for records from a given NCBI database</p></a></li>
<li><a href='#entrez_db_searchable'><p>List available search fields for a given database</p></a></li>
<li><a href='#entrez_db_summary'><p>Retrieve summary information about an NCBI database</p></a></li>
<li><a href='#entrez_dbs'><p>List databases available from the NCBI</p></a></li>
<li><a href='#entrez_fetch'><p>Download data from NCBI databases</p></a></li>
<li><a href='#entrez_global_query'><p>Find the number of records that match a given term across all NCBI Entrez databases</p></a></li>
<li><a href='#entrez_info'><p>Get information about EUtils databases</p></a></li>
<li><a href='#entrez_link'><p>Get links to datasets related to records from an NCBI database</p></a></li>
<li><a href='#entrez_post'><p>Post IDs to Eutils for later use</p></a></li>
<li><a href='#entrez_search'><p>Search the NCBI databases using EUtils</p></a></li>
<li><a href='#entrez_summary'><p>Get summaries of objects in NCBI datasets from a unique ID</p></a></li>
<li><a href='#extract_from_esummary'><p>Extract elements from a list of esummary records</p></a></li>
<li><a href='#linkout_urls'><p>Extract URLs from an elink object</p></a></li>
<li><a href='#parse_pubmed_xml'><p>Summarize an XML record from pubmed.</p></a></li>
<li><a href='#set_entrez_key'><p>Set the ENTREZ_KEY variable to be used by all rentrez functions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.2.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-11-06</td>
</tr>
<tr>
<td>Title:</td>
<td>'Entrez' in R</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>XML, httr (&ge; 0.5), jsonlite (&ge; 0.9)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/rentrez/">https://docs.ropensci.org/rentrez/</a> (website)
<a href="https://github.com/ropensci/rentrez/">https://github.com/ropensci/rentrez/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/rentrez/issues/">https://github.com/ropensci/rentrez/issues/</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Provides an R interface to the NCBI's 'EUtils' API, 
    allowing users to search databases like 'GenBank' 
    <a href="https://www.ncbi.nlm.nih.gov/genbank/">https://www.ncbi.nlm.nih.gov/genbank/</a> and 'PubMed' 
    <a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a>, process the 
    results of those searches and pull data into their R sessions.</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-10 20:43:06 UTC; david</td>
</tr>
<tr>
<td>Author:</td>
<td>David Winter <a href="https://orcid.org/0000-0002-6165-0029"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Scott Chamberlain <a href="https://orcid.org/0000-0003-1444-9135"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Han Guangchun <a href="https://orcid.org/0000-0001-9277-2507"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Winter &lt;david.winter@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-10 21:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='rentrez'>rentrez</h2><span id='topic+rentrez'></span><span id='topic+rentrez-package'></span>

<h3>Description</h3>

<p>rentrez provides functions to search for, discover and download data from
the NCBI's databases using their EUtils function.
</p>


<h3>Details</h3>

<p>Users are expected to know a little bit about the EUtils API, which is well
documented: <a href="https://www.ncbi.nlm.nih.gov/books/NBK25500/">https://www.ncbi.nlm.nih.gov/books/NBK25500/</a>
</p>
<p>The NCBI will ban IPs that don't use EUtils within their <a href="https://www.ncbi.nlm.nih.gov/corehtml/query/static/eutils_help.html">user guidelines</a>. In particular
/enumerated
/item  Don't send more than three request per second (rentrez enforces this limit)
/item  If you plan on sending a sequence of more than ~100 requests, do so outside of peak times for the US
/item  For large requests use the web history method (see examples for <code><a href="#topic+entrez_search">entrez_search</a></code> or use <code><a href="#topic+entrez_post">entrez_post</a></code> to upload IDs)

</p>

<hr>
<h2 id='entrez_citmatch'>Fetch pubmed ids matching specially formatted citation strings</h2><span id='topic+entrez_citmatch'></span>

<h3>Description</h3>

<p>Fetch pubmed ids matching specially formatted citation strings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_citmatch(bdata, db = "pubmed", retmode = "xml", config = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_citmatch_+3A_bdata">bdata</code></td>
<td>
<p>character, containing citation data. 
Each citation must be represented in a pipe-delimited format
journal_title|year|volume|first_page|author_name|your_key|
The final field &quot;your_key&quot; is arbitrary, and can used as you see
fit. Fields can be left empty, but be sure to keep 6 pipes.</p>
</td></tr>
<tr><td><code id="entrez_citmatch_+3A_db">db</code></td>
<td>
<p>character, the database to search. Defaults to pubmed,
the only database currently available</p>
</td></tr>
<tr><td><code id="entrez_citmatch_+3A_retmode">retmode</code></td>
<td>
<p>character, file format to retrieve. Defaults to xml, as 
per the API documentation, though note the API only returns plain text</p>
</td></tr>
<tr><td><code id="entrez_citmatch_+3A_config">config</code></td>
<td>
<p>vector configuration options passed to httr::GET</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing PMIDs
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available configs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ex_cites &lt;- c("proc natl acad sci u s a|1991|88|3248|mann bj|test1|",
              "science|1987|235|182|palmenberg ac|test2|")
entrez_citmatch(ex_cites)

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_db_links'>List available links for records from a given NCBI database</h2><span id='topic+entrez_db_links'></span>

<h3>Description</h3>

<p>For a given database, fetch a list of other databases that contain
cross-referenced records. The names of these records can be used as the
<code>db</code> argument in <code><a href="#topic+entrez_link">entrez_link</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_db_links(db, config = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_db_links_+3A_db">db</code></td>
<td>
<p>character, name of database to search</p>
</td></tr>
<tr><td><code id="entrez_db_links_+3A_config">config</code></td>
<td>
<p>config vector passed to <code>httr::GET</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An eInfoLink object (sub-classed from list) summarizing linked-databases.
Can be coerced to a data-frame with <code>as.data.frame</code>. Printing the object
the name of each element (which is the correct name for <code>entrez_link</code>,
and can be used to get (a little) more information about each linked database
(see example below).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+entrez_link">entrez_link</a></code>
</p>
<p>Other einfo: 
<code><a href="#topic+entrez_db_searchable">entrez_db_searchable</a>()</code>,
<code><a href="#topic+entrez_db_summary">entrez_db_summary</a>()</code>,
<code><a href="#topic+entrez_dbs">entrez_dbs</a>()</code>,
<code><a href="#topic+entrez_info">entrez_info</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxid &lt;- entrez_search(db="taxonomy", term="Osmeriformes")$ids
tax_links &lt;- entrez_db_links("taxonomy")
tax_links
entrez_link(dbfrom="taxonomy", db="pmc", id=taxid)

sra_links &lt;- entrez_db_links("sra")
as.data.frame(sra_links)

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_db_searchable'>List available search fields for a given database</h2><span id='topic+entrez_db_searchable'></span>

<h3>Description</h3>

<p>Fetch a list of search fields that can be used with a given database. Fields
can be used as part of the <code>term</code> argument to <code><a href="#topic+entrez_search">entrez_search</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_db_searchable(db, config = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_db_searchable_+3A_db">db</code></td>
<td>
<p>character, name of database to get search field from</p>
</td></tr>
<tr><td><code id="entrez_db_searchable_+3A_config">config</code></td>
<td>
<p>config vector passed to <code>httr::GET</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An eInfoSearch object (subclassed from list) summarizing linked-databases. 
Can be coerced to a data-frame with <code>as.data.frame</code>. Printing the object
shows only the names of each available search field.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+entrez_search">entrez_search</a></code>
</p>
<p>Other einfo: 
<code><a href="#topic+entrez_db_links">entrez_db_links</a>()</code>,
<code><a href="#topic+entrez_db_summary">entrez_db_summary</a>()</code>,
<code><a href="#topic+entrez_dbs">entrez_dbs</a>()</code>,
<code><a href="#topic+entrez_info">entrez_info</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pmc_fields &lt;- entrez_db_searchable("pmc")
pmc_fields[["AFFL"]]
entrez_search(db="pmc", term="Otago[AFFL]", retmax=0)
entrez_search(db="pmc", term="Auckland[AFFL]", retmax=0)

sra_fields &lt;- entrez_db_searchable("sra")
as.data.frame(sra_fields)

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_db_summary'>Retrieve summary information about an NCBI database</h2><span id='topic+entrez_db_summary'></span>

<h3>Description</h3>

<p>Retrieve summary information about an NCBI database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_db_summary(db, config = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_db_summary_+3A_db">db</code></td>
<td>
<p>character, name of database to summaries</p>
</td></tr>
<tr><td><code id="entrez_db_summary_+3A_config">config</code></td>
<td>
<p>config vector passed to <code>httr::GET</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector with the following data
</p>
<p>DbName Name of database
</p>
<p>Description Brief description of the database
</p>
<p>Count Number of records contained in the database
</p>
<p>MenuName Name in web-interface to EUtils
</p>
<p>DbBuild Unique ID for current build of database
</p>
<p>LastUpdate Date of most recent update to database
</p>


<h3>See Also</h3>

<p>Other einfo: 
<code><a href="#topic+entrez_db_links">entrez_db_links</a>()</code>,
<code><a href="#topic+entrez_db_searchable">entrez_db_searchable</a>()</code>,
<code><a href="#topic+entrez_dbs">entrez_dbs</a>()</code>,
<code><a href="#topic+entrez_info">entrez_info</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
entrez_db_summary("pubmed")

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_dbs'>List databases available from the NCBI</h2><span id='topic+entrez_dbs'></span>

<h3>Description</h3>

<p>Retrieves the names of  databases available through the EUtils API
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_dbs(config = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_dbs_+3A_config">config</code></td>
<td>
<p>config vector passed to <code>httr::GET</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector listing available dbs
</p>


<h3>See Also</h3>

<p>Other einfo: 
<code><a href="#topic+entrez_db_links">entrez_db_links</a>()</code>,
<code><a href="#topic+entrez_db_searchable">entrez_db_searchable</a>()</code>,
<code><a href="#topic+entrez_db_summary">entrez_db_summary</a>()</code>,
<code><a href="#topic+entrez_info">entrez_info</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
entrez_dbs()

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_fetch'>Download data from NCBI databases</h2><span id='topic+entrez_fetch'></span>

<h3>Description</h3>

<p>Pass unique identifiers to an NCBI database and receive data files in a
variety of formats.
A set of unique identifiers mustbe specified with either the <code>db</code>
argument (which directly specifies the IDs as a numeric or character vector)
or a <code>web_history</code> object as returned by 
<code><a href="#topic+entrez_link">entrez_link</a></code>, <code><a href="#topic+entrez_search">entrez_search</a></code> or 
<code><a href="#topic+entrez_post">entrez_post</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_fetch(
  db,
  id = NULL,
  web_history = NULL,
  rettype,
  retmode = "",
  parsed = FALSE,
  config = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_fetch_+3A_db">db</code></td>
<td>
<p>character, name of the database to use</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_id">id</code></td>
<td>
<p>vector (numeric or character), unique ID(s) for records in database
<code>db</code>. In the case of sequence databases these IDs can take form of an
NCBI accession followed by a version number (eg AF123456.1 or AF123456.2).</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_web_history">web_history</code></td>
<td>
<p>a web_history object</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_rettype">rettype</code></td>
<td>
<p>character, format in which to get data (eg, fasta, xml...)</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_retmode">retmode</code></td>
<td>
<p>character, mode in which to receive data, defaults to an empty
string (corresponding to the default mode for rettype).</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_parsed">parsed</code></td>
<td>
<p>boolean should entrez_fetch attempt to parse the resulting 
file. Only works with xml records (including those with rettypes other than
&quot;xml&quot;) at present</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_config">config</code></td>
<td>
<p>vector, httr configuration options passed to httr::GET</p>
</td></tr>
<tr><td><code id="entrez_fetch_+3A_...">...</code></td>
<td>
<p>character, additional terms to add to the request, see NCBI
documentation linked to in references for a complete list</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The format for returned records is set by that arguments <code>rettype</code> (for
a particular format) and <code>retmode</code> for a general format (JSON, XML text
etc). See  <a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/">Table 1</a> 
in the linked reference for the set of 
formats available for each database. In particular, note that sequence
databases (nuccore, protein and their relatives) use specific format names
(eg &quot;native&quot;, &quot;ipg&quot;) for different flavours of xml.
</p>
<p>For the most part, this function returns a character vector containing the 
fetched records. For XML records (including 'native', 'ipg', 'gbc' sequence
records), setting <code>parsed</code> to <code>TRUE</code> will return an
<code>XMLInternalDocument</code>,
</p>


<h3>Value</h3>

<p>character string containing the file created
</p>
<p>XMLInternalDocument a parsed XML document if parsed=TRUE and
rettype is a flavour of XML.
</p>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_EFetch_">https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_EFetch_</a>
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available '<code>httr</code>' configs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
katipo &lt;- "Latrodectus katipo[Organism]"
katipo_search &lt;- entrez_search(db="nuccore", term=katipo)
kaitpo_seqs &lt;- entrez_fetch(db="nuccore", id=katipo_search$ids, rettype="fasta")
#xml
kaitpo_seqs &lt;- entrez_fetch(db="nuccore", id=katipo_search$ids, rettype="native")

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_global_query'>Find the number of records that match a given term across all NCBI Entrez databases</h2><span id='topic+entrez_global_query'></span>

<h3>Description</h3>

<p>Find the number of records that match a given term across all NCBI Entrez databases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_global_query(term, config = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_global_query_+3A_term">term</code></td>
<td>
<p>the search term to use</p>
</td></tr>
<tr><td><code id="entrez_global_query_+3A_config">config</code></td>
<td>
<p>vector configuration options passed to httr::GET</p>
</td></tr>
<tr><td><code id="entrez_global_query_+3A_...">...</code></td>
<td>
<p>additional arguments to add to the query</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector with counts for each a database
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available configs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
NCBI_data_on_best_butterflies_ever &lt;- entrez_global_query(term="Heliconius")

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_info'>Get information about EUtils databases</h2><span id='topic+entrez_info'></span>

<h3>Description</h3>

<p>Gather information about EUtils generally, or a given Eutils database.
Note: The most common uses-cases for the einfo util are finding the list of
search fields available for a given database or the other NCBI databases to
which records in a given database might be linked. Both these use cases
are implemented in higher-level functions that return just this information
(<code>entrez_db_searchable</code> and <code>entrez_db_links</code> respectively).
Consequently most users will not have a reason to use this function (though
it is exported by <code>rentrez</code> for the sake of completeness.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_info(db = NULL, config = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_info_+3A_db">db</code></td>
<td>
<p>character database about which to retrieve information (optional)</p>
</td></tr>
<tr><td><code id="entrez_info_+3A_config">config</code></td>
<td>
<p>config vector passed on to <code>httr::GET</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>XMLInternalDocument with information describing either all the
databases available in Eutils (if db is not set) or one particular database
(set by 'db')
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available httr configurations
</p>
<p>Other einfo: 
<code><a href="#topic+entrez_db_links">entrez_db_links</a>()</code>,
<code><a href="#topic+entrez_db_searchable">entrez_db_searchable</a>()</code>,
<code><a href="#topic+entrez_db_summary">entrez_db_summary</a>()</code>,
<code><a href="#topic+entrez_dbs">entrez_dbs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
all_the_data &lt;- entrez_info()
XML::xpathSApply(all_the_data, "//DbName", xmlValue)
entrez_dbs()

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_link'>Get links to datasets related to records from an NCBI database</h2><span id='topic+entrez_link'></span>

<h3>Description</h3>

<p>Discover records related to a set of unique identifiers from
an NCBI database. The object returned by this function depends on the value
set for the <code>cmd</code> argument. Printing the returned object lists the names
, and provides a brief description, of the elements included in the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_link(
  dbfrom,
  web_history = NULL,
  id = NULL,
  db = NULL,
  cmd = "neighbor",
  by_id = FALSE,
  config = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_link_+3A_dbfrom">dbfrom</code></td>
<td>
<p>character Name of database from which the Id(s) originate</p>
</td></tr>
<tr><td><code id="entrez_link_+3A_web_history">web_history</code></td>
<td>
<p>a web_history object</p>
</td></tr>
<tr><td><code id="entrez_link_+3A_id">id</code></td>
<td>
<p>vector with unique ID(s) for records in database <code>db</code>.</p>
</td></tr>
<tr><td><code id="entrez_link_+3A_db">db</code></td>
<td>
<p>character Name of the database to search for links (or use &quot;all&quot; to 
search all databases available for <code>db</code>. <code>entrez_db_links</code> allows you
to discover databases that might have linked information (see examples).</p>
</td></tr>
<tr><td><code id="entrez_link_+3A_cmd">cmd</code></td>
<td>
<p>link function to use. Allowed values include
</p>

<ul>
<li><p> neighbor (default). Returns a set of IDs in <code>db</code> linked to the
input IDs in <code>dbfrom</code>.
</p>
</li>
<li><p> neighbor_score. As 'neighbor&rdquo;, but additionally returns similarity scores.
</p>
</li>
<li><p> neighbor_history. As &lsquo;neighbor&rsquo;, but returns web history objects.
</p>
</li>
<li><p> acheck. Returns a list of linked databases available from NCBI for a set of IDs.
</p>
</li>
<li><p> ncheck. Checks for the existence of links within a single database.
</p>
</li>
<li><p> lcheck. Checks for external (i.e. outside NCBI) links.
</p>
</li>
<li><p> llinks. Returns a list of external links for each ID, excluding links
provided by libraries.
</p>
</li>
<li><p> llinkslib. As 'llinks' but additionally includes links provided by
libraries.
</p>
</li>
<li><p> prlinks. As 'llinks' but returns only the primary external link for
each ID.
</p>
</li></ul>
</td></tr>
<tr><td><code id="entrez_link_+3A_by_id">by_id</code></td>
<td>
<p>logical If FALSE (default) return a single 
<code>elink</code> objects containing links for all of the provided <code>id</code>s. 
Alternatively, if TRUE return a list of <code>elink</code> objects, one for each 
ID in <code>id</code>.</p>
</td></tr>
<tr><td><code id="entrez_link_+3A_config">config</code></td>
<td>
<p>vector configuration options passed to httr::GET</p>
</td></tr>
<tr><td><code id="entrez_link_+3A_...">...</code></td>
<td>
<p>character Additional terms to add to the request, see NCBI
documentation linked to in references for a complete list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An elink object containing the data defined by the <code>cmd</code> argument
(if by_id=FALSE) or a list of such object (if by_id=TRUE).
</p>
<p>file XMLInternalDocument xml file resulting from search, parsed with
<code><a href="XML.html#topic+xmlTreeParse">xmlTreeParse</a></code>
</p>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_ELink_">https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_ELink_</a>
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available configs
</p>
<p><code>entrez_db_links</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 pubmed_search &lt;- entrez_search(db = "pubmed", term ="10.1016/j.ympev.2010.07.013[doi]")
 linked_dbs &lt;- entrez_db_links("pubmed")
 linked_dbs
 nucleotide_data &lt;- entrez_link(dbfrom = "pubmed", id = pubmed_search$ids, db ="nuccore")
 #Sources for the full text of the paper 
 res &lt;- entrez_link(dbfrom="pubmed", db="", cmd="llinks", id=pubmed_search$ids)
 linkout_urls(res)

## End(Not run)

</code></pre>

<hr>
<h2 id='entrez_post'>Post IDs to Eutils for later use</h2><span id='topic+entrez_post'></span>

<h3>Description</h3>

<p>Post IDs to Eutils for later use
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_post(db, id = NULL, web_history = NULL, config = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_post_+3A_db">db</code></td>
<td>
<p>character Name of the database from which the IDs were taken</p>
</td></tr>
<tr><td><code id="entrez_post_+3A_id">id</code></td>
<td>
<p>vector with unique ID(s) for records in database <code>db</code>.</p>
</td></tr>
<tr><td><code id="entrez_post_+3A_web_history">web_history</code></td>
<td>
<p>A web_history object. Can be used to add to additional
identifiers to an existing web environment on the NCBI</p>
</td></tr>
<tr><td><code id="entrez_post_+3A_config">config</code></td>
<td>
<p>vector of configuration options passed to httr::GET</p>
</td></tr>
<tr><td><code id="entrez_post_+3A_...">...</code></td>
<td>
<p>character Additional terms to add to the request, see NCBI
documentation linked to in references for a complete list</p>
</td></tr>
</table>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_EPost_">https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_EPost_</a>
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available httr configurations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:   
so_many_snails &lt;- entrez_search(db="nuccore", 
                      "Gastropoda[Organism] AND COI[Gene]", retmax=200)
upload &lt;- entrez_post(db="nuccore", id=so_many_snails$ids)
first &lt;- entrez_fetch(db="nuccore", rettype="fasta", web_history=upload,
                      retmax=10)
second &lt;- entrez_fetch(db="nuccore", file_format="fasta", web_history=upload,
                       retstart=10, retmax=10)

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_search'>Search the NCBI databases using EUtils</h2><span id='topic+entrez_search'></span>

<h3>Description</h3>

<p>Search a given NCBI database with a particular query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_search(
  db,
  term,
  config = NULL,
  retmode = "xml",
  use_history = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_search_+3A_db">db</code></td>
<td>
<p>character, name of the database to search for.</p>
</td></tr>
<tr><td><code id="entrez_search_+3A_term">term</code></td>
<td>
<p>character, the search term. The syntax used in making these
searches is described in the Details of this help message, the package
vignette and reference given below.</p>
</td></tr>
<tr><td><code id="entrez_search_+3A_config">config</code></td>
<td>
<p>vector configuration options passed to httr::GET</p>
</td></tr>
<tr><td><code id="entrez_search_+3A_retmode">retmode</code></td>
<td>
<p>character, one of json (default) or xml. This will make no
difference in most cases.</p>
</td></tr>
<tr><td><code id="entrez_search_+3A_use_history">use_history</code></td>
<td>
<p>logical. If TRUE return a web_history object for use in 
later calls to the NCBI</p>
</td></tr>
<tr><td><code id="entrez_search_+3A_...">...</code></td>
<td>
<p>character, additional terms to add to the request, see NCBI
documentation linked to in references for a complete list</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The NCBI uses a search term syntax where search terms can be associated with 
a specific search field with square brackets. So, for instance &ldquo;Homo[ORGN]&rdquo;
denotes a search for Homo in the &ldquo;Organism&rdquo; field. The names and
definitions of these fields can be identified using
<code><a href="#topic+entrez_db_searchable">entrez_db_searchable</a></code>.
</p>
<p>Searches can make use of several fields by combining them via the boolean
operators AND, OR and NOT. So, using the search term&ldquo;((Homo[ORGN] AND APP[GENE]) NOT
Review[PTYP])&rdquo; in PubMed would identify articles matching the gene APP in
humans, and exclude review articles. More examples of the use of these search
terms, and the more specific MeSH terms for precise searching, 
is given in the package vignette. <code>rentrez</code> handles special characters
and URL encoding (e.g. replacing spaces with plus signs) on the client side,
so there is no need to include these in search term
</p>
<p>The<code>rentrez</code> tutorial provides some tips on how to make the most of 
searches to the NCBI. In particular, the sections on uses of the &quot;Filter&quot;
field and MeSH terms may in formulating precise searches.
</p>


<h3>Value</h3>

<p>ids integer Unique IDS returned by the search
</p>
<p>count integer Total number of hits for the search
</p>
<p>retmax integer Maximum number of hits returned by the search
</p>
<p>web_history A web_history object for use in subsequent calls to NCBI
</p>
<p>QueryTranslation character, search term as the NCBI interpreted it
</p>
<p>file either and XMLInternalDocument xml file resulting from search, parsed with
<code><a href="XML.html#topic+xmlTreeParse">xmlTreeParse</a></code> or, if <code>retmode</code> was set to json a list
resulting from the returned JSON file being parsed with
<code><a href="jsonlite.html#topic+fromJSON">fromJSON</a></code>.
</p>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_ESearch_">https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_ESearch_</a>
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available httr configurations
</p>
<p><code><a href="#topic+entrez_db_searchable">entrez_db_searchable</a></code> to get a set of search fields that
can be used in <code>term</code> for any database
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   query &lt;- "Gastropoda[Organism] AND COI[Gene]"
   web_env_search &lt;- entrez_search(db="nuccore", query, use_history=TRUE)
   cookie &lt;- web_env_search$WebEnv
   qk &lt;- web_env_search$QueryKey 
   snail_coi &lt;- entrez_fetch(db = "nuccore", WebEnv = cookie, query_key = qk,
                             file_format = "fasta", retmax = 10)

## End(Not run)
## Not run: 

fly_id &lt;- entrez_search(db="taxonomy", term="Drosophila")
#Oh, right. There is a genus and a subgenus name Drosophila...
#how can we limit this search
(tax_fields &lt;- entrez_db_searchable("taxonomy"))
#"RANK" loots promising
tax_fields$RANK
entrez_search(db="taxonomy", term="Drosophila &amp; Genus[RANK]")

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_summary'>Get summaries of objects in NCBI datasets from a unique ID</h2><span id='topic+entrez_summary'></span>

<h3>Description</h3>

<p>The NCBI offer two distinct formats for summary documents.
Version 1.0 is a relatively limited summary of a database record based on a 
shared Document Type Definition. Version 1.0 summaries are only available as
XML and are not available for some newer databases
Version 2.0 summaries generally contain more information about a given
record, but each database has its own distinct format. 2.0 summaries are 
available for records in all databases and as JSON and XML files. 
As of version 0.4, rentrez fetches version 2.0 summaries by default and
uses JSON as the exchange format (as JSON object can be more easily converted
into native R types). Existing scripts which relied on the structure and
naming of the &quot;Version 1.0&quot; summary files can be updated by setting the new
<code>version</code> argument to &quot;1.0&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_summary(
  db,
  id = NULL,
  web_history = NULL,
  version = c("2.0", "1.0"),
  always_return_list = FALSE,
  retmode = NULL,
  config = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_summary_+3A_db">db</code></td>
<td>
<p>character Name of the database to search for</p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_id">id</code></td>
<td>
<p>vector with unique ID(s) for records in database <code>db</code>.
In the case of sequence databases these IDs can take form of an
NCBI accession followed by a version number (eg AF123456.1 or AF123456.2)</p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_web_history">web_history</code></td>
<td>
<p>A web_history object</p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_version">version</code></td>
<td>
<p>either 1.0 or 2.0 see above for description</p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_always_return_list">always_return_list</code></td>
<td>
<p>logical, return a list  of esummary objects even
when only one ID is provided (see description for a note about this option)</p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_retmode">retmode</code></td>
<td>
<p>either &quot;xml&quot; or &quot;json&quot;. By default, xml will be used for
version 1.0 records, json for version 2.0.</p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_config">config</code></td>
<td>
<p>vector configuration options passed to <code>httr::GET</code></p>
</td></tr>
<tr><td><code id="entrez_summary_+3A_...">...</code></td>
<td>
<p>character Additional terms to add to the request, see NCBI
documentation linked to in references for a complete list</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, entrez_summary returns a single record when only one ID is
passed and a list of such records when multiple IDs are passed. This can lead
to unexpected behaviour when the results of a variable number of IDs (perhaps the
result of <code>entrez_search</code>) are processed with an apply family function
or in a for-loop. If you use this function as part of a function or script that
generates a variably-sized vector of IDs setting <code>always_return_list</code> to 
<code>TRUE</code> will avoid these problems. The function
<code>extract_from_esummary</code> is  provided for the specific case of extracting
named elements from a list of esummary objects, and is designed to work on
single objects as well as lists.
</p>


<h3>Value</h3>

<p>A list of esummary records (if multiple IDs are passed and
always_return_list if FALSE) or a single record.
</p>
<p>file XMLInternalDocument xml file containing the entire record
returned by the NCBI.
</p>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_ESummary_">https://www.ncbi.nlm.nih.gov/books/NBK25499/#_chapter4_ESummary_</a>
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code> for available configs
</p>
<p><code><a href="#topic+extract_from_esummary">extract_from_esummary</a></code> which can be used to extract
elements from a list of esummary records
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 pop_ids = c("307082412", "307075396", "307075338", "307075274")
 pop_summ &lt;- entrez_summary(db="popset", id=pop_ids)
 extract_from_esummary(pop_summ, "title")
 
 # clinvar example
 res &lt;- entrez_search(db = "clinvar", term = "BRCA1", retmax=10)
 cv &lt;- entrez_summary(db="clinvar", id=res$ids)
 cv
 extract_from_esummary(cv, "title", simplify=FALSE)
 extract_from_esummary(cv, "trait_set")[1:2] 
 extract_from_esummary(cv, "gene_sort") 

## End(Not run)
</code></pre>

<hr>
<h2 id='extract_from_esummary'>Extract elements from a list of esummary records</h2><span id='topic+extract_from_esummary'></span>

<h3>Description</h3>

<p>Extract elements from a list of esummary records
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_from_esummary(esummaries, elements, simplify = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_from_esummary_+3A_esummaries">esummaries</code></td>
<td>
<p>Either an esummary or an esummary_list (as returned by
entrez_summary).</p>
</td></tr>
<tr><td><code id="extract_from_esummary_+3A_elements">elements</code></td>
<td>
<p>the names of the element to extract</p>
</td></tr>
<tr><td><code id="extract_from_esummary_+3A_simplify">simplify</code></td>
<td>
<p>logical, if possible return a vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List or vector containing requested elements
</p>


<h3>See Also</h3>

<p><code><a href="#topic+entrez_summary">entrez_summary</a></code> for examples of this function in action.
</p>

<hr>
<h2 id='linkout_urls'>Extract URLs from an elink object</h2><span id='topic+linkout_urls'></span>

<h3>Description</h3>

<p>Extract URLs from an elink object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linkout_urls(elink)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linkout_urls_+3A_elink">elink</code></td>
<td>
<p>elink object (returned by entrez_link) containing Urls</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of character vectors, one per ID each containing of URLs for that
ID.
</p>


<h3>See Also</h3>

<p>entrez_link
</p>

<hr>
<h2 id='parse_pubmed_xml'>Summarize an XML record from pubmed.</h2><span id='topic+parse_pubmed_xml'></span>

<h3>Description</h3>

<p>Note: this function assumes all records are of the type &quot;PubmedArticle&quot;
and will return an empty record for any other type (including books).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_pubmed_xml(record)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_pubmed_xml_+3A_record">record</code></td>
<td>
<p>Either and XMLInternalDocument or character the record to be
parsed ( expected to come from <code><a href="#topic+entrez_fetch">entrez_fetch</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a single pubmed_record object, or a list of several
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
hox_paper &lt;- entrez_search(db="pubmed", term="10.1038/nature08789[doi]")
hox_rel &lt;- entrez_link(db="pubmed", dbfrom="pubmed", id=hox_paper$ids)
recs &lt;- entrez_fetch(db="pubmed", 
                       id=hox_rel$links$pubmed_pubmed[1:3], 
                       rettype="xml")
parse_pubmed_xml(recs)

</code></pre>

<hr>
<h2 id='set_entrez_key'>Set the ENTREZ_KEY variable to be used by all rentrez functions</h2><span id='topic+set_entrez_key'></span>

<h3>Description</h3>

<p>The NCBI allows users to access more records (10 per second) if they 
register for and use an API key. This function allows users to set this key
for all calls to rentrez functions during a particular R session. See the
vignette section &quot;Using API keys&quot; for a detailed description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_entrez_key(key)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_entrez_key_+3A_key">key</code></td>
<td>
<p>character. Value to set ENTREZ_KEY to (i.e. your API key).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical of length one, TRUE is the value was set FALSE if not.
value is returned inside invisible(), i.e. it is not printed to screen 
when the function is called.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
