<!DOCTYPE html><html><head><title>Help for package ReporterScore</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ReporterScore}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ReporterScore-package'><p>ReporterScore: Generalized Reporter Score-Based Enrichment Analysis for Omics Data</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#cm_test_k'><p>Test the proper clusters k for c_means</p></a></li>
<li><a href='#Compound_htable'><p>Compound htable from 'KEGG'</p></a></li>
<li><a href='#CPDlist'><p>The CPDlist used for enrichment.</p></a></li>
<li><a href='#custom_modulelist'><p>Build a custom modulelist</p></a></li>
<li><a href='#custom_modulelist_from_org'><p>Custom modulelist from a specific organism</p></a></li>
<li><a href='#gene2ko'><p>Transfer gene symbol table to KO table</p></a></li>
<li><a href='#genedf'><p>human gene table</p></a></li>
<li><a href='#get_features'><p>get features in a modulelist</p></a></li>
<li><a href='#get_reporter_score'><p>Calculate reporter score</p></a></li>
<li><a href='#GOlist'><p>The GOlist used for enrichment.</p></a></li>
<li><a href='#hsa_kegg_pathway'><p>pathway information for &quot;hsa&quot;</p></a></li>
<li><a href='#KO_abundance'><p>The KOs abundance table and group table.</p></a></li>
<li><a href='#KO_enrich'><p>Perform enrichment analysis</p></a></li>
<li><a href='#KO_fisher'><p>Perform fisher's exact enrichment analysis</p></a></li>
<li><a href='#KO_gsa'><p>Perform gene set analysis</p></a></li>
<li><a href='#KO_gsea'><p>Perform gene set enrichment analysis</p></a></li>
<li><a href='#KO_gsva'><p>Perform Gene Set Variation Analysis</p></a></li>
<li><a href='#KO_htable'><p>KO htable from 'KEGG'</p></a></li>
<li><a href='#KO_padog'><p>Perform Pathway Analysis with Down-weighting of Overlapping Genes (PADOG)</p></a></li>
<li><a href='#KO_safe'><p>Perform Significance Analysis of Function and Expression</p></a></li>
<li><a href='#KO_sea'><p>Perform Simultaneous Enrichment Analysis</p></a></li>
<li><a href='#ko.test'><p>Differential analysis or Correlation analysis for KO-abundance table</p></a></li>
<li><a href='#KOlist'><p>The KOlist used for enrichment.</p></a></li>
<li><a href='#load_CARDinfo'><p>Load the CARDinfo (from CARD database)</p></a></li>
<li><a href='#load_GOlist'><p>Load the GOlist (from 'GO' database)</p></a></li>
<li><a href='#load_htable'><p>Load the specific table (from 'KEGG')</p></a></li>
<li><a href='#mmu_kegg_pathway'><p>pathway information for &quot;mmu&quot;</p></a></li>
<li><a href='#modify_description'><p>Modify the pathway description before plotting</p></a></li>
<li><a href='#Module_htable'><p>Module htable from 'KEGG'</p></a></li>
<li><a href='#Pathway_htable'><p>Pathway htable from 'KEGG'</p></a></li>
<li><a href='#plot_enrich_res'><p>Plot enrich_res</p></a></li>
<li><a href='#plot_features_box'><p>Plot features boxplot</p></a></li>
<li><a href='#plot_features_distribution'><p>plot the Z-score of features distribution</p></a></li>
<li><a href='#plot_features_heatmap'><p>Plot features heatmap</p></a></li>
<li><a href='#plot_features_in_pathway'><p>Plot features trend in one pathway or module</p></a></li>
<li><a href='#plot_features_network'><p>Plot features network</p></a></li>
<li><a href='#plot_htable'><p>Plot htable levels</p></a></li>
<li><a href='#plot_KEGG_map'><p>plot_KEGG_map</p></a></li>
<li><a href='#plot_report'><p>Plot the reporter_res</p></a></li>
<li><a href='#plot_report_circle_packing'><p>Plot the reporter_res as circle_packing</p></a></li>
<li><a href='#plot_significance'><p>Plot the significance of pathway</p></a></li>
<li><a href='#plot.cm_res'><p>Plot c_means result</p></a></li>
<li><a href='#print.reporter_score'><p>Print reporter_score</p></a></li>
<li><a href='#print.rs_by_cm'><p>Print rs_by_cm</p></a></li>
<li><a href='#pvalue2zs'><p>Transfer p-value of KOs to Z-score</p></a></li>
<li><a href='#reporter_score'><p>One step to get the reporter score of your KO abundance table.</p></a></li>
<li><a href='#reporter_score_res'><p>'reporter_score()' result from KO_abundance_test</p></a></li>
<li><a href='#RSA_by_cm'><p>Reporter score analysis after C-means clustering</p></a></li>
<li><a href='#up_level_KO'><p>Upgrade the KO level</p></a></li>
<li><a href='#update_CARDinfo'><p>update CARDinfo from (from 'CARD' database)</p></a></li>
<li><a href='#update_GOlist'><p>Update the GO2gene files (from 'GO' database)</p></a></li>
<li><a href='#update_KEGG'><p>Update files from 'KEGG'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generalized Reporter Score-Based Enrichment Analysis for Omics
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Inspired by the classic 'RSA', we developed the improved 'Generalized Reporter 
    Score-based Analysis (GRSA)' method, implemented in the R package 'ReporterScore', along 
    with comprehensive visualization methods and pathway databases. 'GRSA' is a threshold-free 
    method that works well with all types of biomedical features, such as genes, chemical compounds, 
    and microbial species. Importantly, the 'GRSA' supports multi-group and longitudinal experimental 
    designs, because of the included multi-group-compatible statistical methods. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, dplyr, stats, ggplot2 (&ge; 3.2.0), pcutils (&ge;
0.2.5), utils, scales, ggnewscale, ggrepel, reshape2, stringr,
foreach</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, plyr, e1071, factoextra, snow, doSNOW,
pheatmap, readr, R.utils, KEGGREST, clusterProfiler,
enrichplot, pathview, GSA, vegan, MetaNet, igraph, ggraph,
PADOG, safe, rSEA, GSVA</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Asa12138/ReporterScore/issues">https://github.com/Asa12138/ReporterScore/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Asa12138/ReporterScore">https://github.com/Asa12138/ReporterScore</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-26 14:10:05 UTC; asa</td>
</tr>
<tr>
<td>Author:</td>
<td>Chen Peng <a href="https://orcid.org/0000-0002-9449-7606"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chen Peng &lt;pengchen2001@zju.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-29 16:30:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='ReporterScore-package'>ReporterScore: Generalized Reporter Score-Based Enrichment Analysis for Omics Data</h2><span id='topic+ReporterScore'></span><span id='topic+ReporterScore-package'></span>

<h3>Description</h3>

<p>Inspired by the classic 'RSA', we developed the improved 'Generalized Reporter Score-based Analysis (GRSA)' method, implemented in the R package 'ReporterScore', along with comprehensive visualization methods and pathway databases. 'GRSA' is a threshold-free method that works well with all types of biomedical features, such as genes, chemical compounds, and microbial species. Importantly, the 'GRSA' supports multi-group and longitudinal experimental designs, because of the included multi-group-compatible statistical methods.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Chen Peng <a href="mailto:pengchen2001@zju.edu.cn">pengchen2001@zju.edu.cn</a> (<a href="https://orcid.org/0000-0002-9449-7606">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/Asa12138/ReporterScore">https://github.com/Asa12138/ReporterScore</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Asa12138/ReporterScore/issues">https://github.com/Asa12138/ReporterScore/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq_len(5) %&gt;% sum()

</code></pre>

<hr>
<h2 id='cm_test_k'>Test the proper clusters k for c_means</h2><span id='topic+cm_test_k'></span><span id='topic+c_means'></span>

<h3>Description</h3>

<p>Test the proper clusters k for c_means
</p>
<p>C-means cluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cm_test_k(otu_group, filter_var, fast = TRUE)

c_means(otu_group, k_num, filter_var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cm_test_k_+3A_otu_group">otu_group</code></td>
<td>
<p>standardize data</p>
</td></tr>
<tr><td><code id="cm_test_k_+3A_filter_var">filter_var</code></td>
<td>
<p>filter the highest var</p>
</td></tr>
<tr><td><code id="cm_test_k_+3A_fast">fast</code></td>
<td>
<p>whether do the gap_stat?</p>
</td></tr>
<tr><td><code id="cm_test_k_+3A_k_num">k_num</code></td>
<td>
<p>cluster number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>
<p>ggplot
</p>


<h3>See Also</h3>

<p>Other C_means: 
<code><a href="#topic+RSA_by_cm">RSA_by_cm</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("e1071") &amp;&amp; requireNamespace("factoextra")) {
  data(otutab, package = "pcutils")
  pcutils::hebing(otutab, metadata$Group) -&gt; otu_group
  cm_test_k(otu_group, filter_var = 0.7)
  cm_res &lt;- c_means(otu_group, k_num = 3, filter_var = 0.7)
  plot(cm_res, 0.8)
}

</code></pre>

<hr>
<h2 id='Compound_htable'>Compound htable from 'KEGG'</h2><span id='topic+Compound_htable'></span>

<h3>Description</h3>

<p>Compound htable from 'KEGG'
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='CPDlist'>The CPDlist used for enrichment.</h2><span id='topic+CPDlist'></span>

<h3>Description</h3>

<p>an list contains two data.frame named pathway and module.
</p>


<h3>Format</h3>

<p>four columns in each data.frame.
</p>

<dl>
<dt>id</dt><dd><p>&quot;map0010&quot; or &quot;M00001&quot;</p>
</dd>
<dt>K_num</dt><dd><p>contians how many Compounds in this pathway or module</p>
</dd>
<dt>KOs</dt><dd><p>Compounds name</p>
</dd>
<dt>Description</dt><dd><p>the description of this pathway or module</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='custom_modulelist'>Build a custom modulelist</h2><span id='topic+custom_modulelist'></span><span id='topic+transform_modulelist'></span>

<h3>Description</h3>

<p>Build a custom modulelist
</p>
<p>Transform a modulelist to a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_modulelist(pathway2ko, pathway2desc = NULL, verbose = TRUE)

transform_modulelist(mymodulelist, mode = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_modulelist_+3A_pathway2ko">pathway2ko</code></td>
<td>
<p>user input annotation of Pathway to KO mapping, a data.frame of 2 column with pathway and ko.</p>
</td></tr>
<tr><td><code id="custom_modulelist_+3A_pathway2desc">pathway2desc</code></td>
<td>
<p>user input of Pathway TO Description mapping, a data.frame of 2 column with pathway and description.</p>
</td></tr>
<tr><td><code id="custom_modulelist_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="custom_modulelist_+3A_mymodulelist">mymodulelist</code></td>
<td>
<p>mymodulelist</p>
</td></tr>
<tr><td><code id="custom_modulelist_+3A_mode">mode</code></td>
<td>
<p>1~2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a custom modulelist
</p>
<p>modulelist
</p>


<h3>See Also</h3>

<p>Other modulelist: 
<code><a href="#topic+custom_modulelist_from_org">custom_modulelist_from_org</a>()</code>,
<code><a href="#topic+get_features">get_features</a>()</code>
</p>
<p>Other modulelist: 
<code><a href="#topic+custom_modulelist_from_org">custom_modulelist_from_org</a>()</code>,
<code><a href="#topic+get_features">get_features</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mydat &lt;- data.frame(pathway = paste0("PATHWAY", rep(seq_len(2), each = 5)), ko = paste0("K", 1:10))
mymodulelist &lt;- custom_modulelist(mydat)
print(mymodulelist)
transform_modulelist(mymodulelist)
</code></pre>

<hr>
<h2 id='custom_modulelist_from_org'>Custom modulelist from a specific organism</h2><span id='topic+custom_modulelist_from_org'></span>

<h3>Description</h3>

<p>Custom modulelist from a specific organism
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_modulelist_from_org(
  org = "hsa",
  feature = "ko",
  gene = "symbol",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_modulelist_from_org_+3A_org">org</code></td>
<td>
<p>kegg organism, listed in https://www.genome.jp/kegg/catalog/org_list.html, default, &quot;hsa&quot;</p>
</td></tr>
<tr><td><code id="custom_modulelist_from_org_+3A_feature">feature</code></td>
<td>
<p>one of &quot;ko&quot;, &quot;gene&quot;, &quot;compound&quot;</p>
</td></tr>
<tr><td><code id="custom_modulelist_from_org_+3A_gene">gene</code></td>
<td>
<p>one of &quot;symbol&quot;,&quot;id&quot;</p>
</td></tr>
<tr><td><code id="custom_modulelist_from_org_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>modulelist
</p>


<h3>See Also</h3>

<p>Other modulelist: 
<code><a href="#topic+custom_modulelist">custom_modulelist</a>()</code>,
<code><a href="#topic+get_features">get_features</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hsa_pathway &lt;- custom_modulelist_from_org(org = "hsa", feature = "gene")
</code></pre>

<hr>
<h2 id='gene2ko'>Transfer gene symbol table to KO table</h2><span id='topic+gene2ko'></span>

<h3>Description</h3>

<p>You can use 'clusterProfiler::bitr()' to transfer your table from other gene_id to gene_symbol.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene2ko(genedf, org = "hsa")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene2ko_+3A_genedf">genedf</code></td>
<td>
<p>,rowname is gene symbol (e.g. PFKM), colnames is samples</p>
</td></tr>
<tr><td><code id="gene2ko_+3A_org">org</code></td>
<td>
<p>kegg organism, listed in 'https://www.genome.jp/kegg/catalog/org_list.html', default, 'hsa'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>kodf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genedf")
KOdf &lt;- gene2ko(genedf, org = "hsa")
</code></pre>

<hr>
<h2 id='genedf'>human gene table</h2><span id='topic+genedf'></span>

<h3>Description</h3>

<p>human gene table
</p>


<h3>See Also</h3>

<p>Other test_data: 
<code><a href="#topic+KO_abundance">KO_abundance</a></code>,
<code><a href="#topic+reporter_score_res">reporter_score_res</a></code>
</p>

<hr>
<h2 id='get_features'>get features in a modulelist</h2><span id='topic+get_features'></span><span id='topic+get_KOs'></span>

<h3>Description</h3>

<p>get features in a modulelist
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_features(map_id = "map00010", ko_stat = NULL, modulelist = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_features_+3A_map_id">map_id</code></td>
<td>
<p>map_id in modulelist</p>
</td></tr>
<tr><td><code id="get_features_+3A_ko_stat">ko_stat</code></td>
<td>
<p>NULL or ko_stat result from <code><a href="#topic+pvalue2zs">pvalue2zs</a></code></p>
</td></tr>
<tr><td><code id="get_features_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain 'id','K_num','KOs','Description' columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>KOids, or data.frame with these KOids.
</p>


<h3>See Also</h3>

<p>Other modulelist: 
<code><a href="#topic+custom_modulelist_from_org">custom_modulelist_from_org</a>()</code>,
<code><a href="#topic+custom_modulelist">custom_modulelist</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_features(map_id = "map00010")
</code></pre>

<hr>
<h2 id='get_reporter_score'>Calculate reporter score</h2><span id='topic+get_reporter_score'></span>

<h3>Description</h3>

<p>Calculate reporter score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_reporter_score(
  ko_stat,
  type = c("pathway", "module")[1],
  feature = "ko",
  threads = 1,
  modulelist = NULL,
  perm = 4999,
  verbose = TRUE,
  p.adjust.method2 = "BH",
  min_exist_KO = 3,
  max_exist_KO = 600
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_reporter_score_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat result from <code><a href="#topic+pvalue2zs">pvalue2zs</a></code></p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_type">type</code></td>
<td>
<p>'pathway' or 'module' for default KOlist for microbiome, 'CC', 'MF', 'BP', 'ALL' for default GOlist for homo sapiens. And org in listed in 'https://www.genome.jp/kegg/catalog/org_list.html' such as 'hsa' (if your kodf is come from a specific organism, you should specify type here).</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_feature">feature</code></td>
<td>
<p>one of 'ko', 'gene', 'compound'</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_threads">threads</code></td>
<td>
<p>default 1</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain 'id','K_num','KOs','Description' columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_perm">perm</code></td>
<td>
<p>permutation number, default: 4999.</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_p.adjust.method2">p.adjust.method2</code></td>
<td>
<p>p.adjust.method for the correction of ReporterScore, see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_min_exist_ko">min_exist_KO</code></td>
<td>
<p>min exist KO number in a pathway (default, 3, when a pathway contains KOs less than 3, there will be no RS)</p>
</td></tr>
<tr><td><code id="get_reporter_score_+3A_max_exist_ko">max_exist_KO</code></td>
<td>
<p>max exist KO number in a pathway (default, 600, when a pathway contains KOs more than 600, there will be no RS)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>reporter_res data.frame
</p>


<h3>See Also</h3>

<p>Other GRSA: 
<code><a href="#topic+ko.test">ko.test</a>()</code>,
<code><a href="#topic+pvalue2zs">pvalue2zs</a>()</code>,
<code><a href="#topic+reporter_score">reporter_score</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("KO_abundance_test")
ko_pvalue &lt;- ko.test(KO_abundance, "Group", metadata)
ko_stat &lt;- pvalue2zs(ko_pvalue, mode = "directed")
reporter_s1 &lt;- get_reporter_score(ko_stat, perm = 499)

</code></pre>

<hr>
<h2 id='GOlist'>The GOlist used for enrichment.</h2><span id='topic+GOlist'></span>

<h3>Description</h3>

<p>an list contains three data.frame named BP, CC, MF.
</p>


<h3>Format</h3>

<p>four columns in each data.frame.
</p>

<dl>
<dt>id</dt><dd><p>&quot;map0010&quot; or &quot;M00001&quot;</p>
</dd>
<dt>K_num</dt><dd><p>contians how many Genes in this GO term</p>
</dd>
<dt>KOs</dt><dd><p>Genes name</p>
</dd>
<dt>Description</dt><dd><p>the description of this GO term</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='hsa_kegg_pathway'>pathway information for &quot;hsa&quot;</h2><span id='topic+hsa_kegg_pathway'></span>

<h3>Description</h3>

<p>pathway information for &quot;hsa&quot;
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='KO_abundance'>The KOs abundance table and group table.</h2><span id='topic+KO_abundance'></span><span id='topic+metadata'></span>

<h3>Description</h3>

<p>The KOs abundance table and group table.
</p>
<p>The KOs abundance table and group table.
</p>


<h3>See Also</h3>

<p>Other test_data: 
<code><a href="#topic+genedf">genedf</a></code>,
<code><a href="#topic+reporter_score_res">reporter_score_res</a></code>
</p>

<hr>
<h2 id='KO_enrich'>Perform enrichment analysis</h2><span id='topic+KO_enrich'></span><span id='topic+as.enrich_res'></span>

<h3>Description</h3>

<p>This function performs KO enrichment analysis using the 'clusterProfiler' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_enrich(
  ko_stat,
  padj_threshold = 0.05,
  logFC_threshold = NULL,
  add_mini = NULL,
  p.adjust.method = "BH",
  type = c("pathway", "module")[1],
  feature = "ko",
  modulelist = NULL,
  verbose = TRUE
)

as.enrich_res(gsea_res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_enrich_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat dataframe from <code><a href="#topic+ko.test">ko.test</a></code>.</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>p.adjust threshold to determine whether a feature significant or not. p.adjust &lt; padj_threshold, default: 0.05</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_logfc_threshold">logFC_threshold</code></td>
<td>
<p>logFC threshold to determine whether a feature significant or not. abs(logFC)&gt;logFC_threshold, default: NULL</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_add_mini">add_mini</code></td>
<td>
<p>add_mini when calculate the logFC. e.g (10+0.1)/(0+0.1), default 0.05*min(avg_abundance)</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>The method used for p-value adjustment (default: &quot;BH&quot;).</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_type">type</code></td>
<td>
<p>&quot;pathway&quot; or &quot;module&quot; for default KOlist_file.</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_feature">feature</code></td>
<td>
<p>one of &quot;ko&quot;, &quot;gene&quot;, &quot;compound&quot;</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="KO_enrich_+3A_gsea_res">gsea_res</code></td>
<td>
<p>gsea_res from KO_gsea</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the enrichment results.
</p>
<p>enrich_res object
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## use `enricher` from the `clusterProfiler` package.
if (requireNamespace("clusterProfiler")) {
  data("reporter_score_res")
  enrich_res &lt;- KO_enrich(reporter_score_res)
  plot(enrich_res)
}
</code></pre>

<hr>
<h2 id='KO_fisher'>Perform fisher's exact enrichment analysis</h2><span id='topic+KO_fisher'></span>

<h3>Description</h3>

<p>Perform fisher's exact enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_fisher(
  ko_stat,
  padj_threshold = 0.05,
  logFC_threshold = NULL,
  add_mini = NULL,
  p.adjust.method = "BH",
  type = c("pathway", "module")[1],
  feature = "ko",
  modulelist = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_fisher_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat dataframe from <code><a href="#topic+ko.test">ko.test</a></code>.</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>p.adjust threshold to determine whether a feature significant or not. p.adjust &lt; padj_threshold, default: 0.05</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_logfc_threshold">logFC_threshold</code></td>
<td>
<p>logFC threshold to determine whether a feature significant or not. abs(logFC)&gt;logFC_threshold, default: NULL</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_add_mini">add_mini</code></td>
<td>
<p>add_mini when calculate the logFC. e.g (10+0.1)/(0+0.1), default 0.05*min(avg_abundance)</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>The method used for p-value adjustment (default: &quot;BH&quot;).</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_type">type</code></td>
<td>
<p>&quot;pathway&quot; or &quot;module&quot; for default KOlist_file.</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_feature">feature</code></td>
<td>
<p>one of &quot;ko&quot;, &quot;gene&quot;, &quot;compound&quot;</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="KO_fisher_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## use `fisher.test` from the `stats` package.
data("reporter_score_res")
fisher_res &lt;- KO_fisher(reporter_score_res)
</code></pre>

<hr>
<h2 id='KO_gsa'>Perform gene set analysis</h2><span id='topic+KO_gsa'></span>

<h3>Description</h3>

<p>Perform gene set analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_gsa(
  reporter_res,
  method = "Two class unpaired",
  p.adjust.method = "BH",
  verbose = TRUE,
  perm = 1000,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_gsa_+3A_reporter_res">reporter_res</code></td>
<td>
<p>reporter_res</p>
</td></tr>
<tr><td><code id="KO_gsa_+3A_method">method</code></td>
<td>
<p>Problem type: &quot;quantitative&quot; for a continuous parameter; &quot;Two class unpaired&quot; ; &quot;Survival&quot; for censored survival outcome; &quot;Multiclass&quot; : more than 2 groups, coded 1,2,3...; &quot;Two class paired&quot; for paired outcomes, coded -1,1 (first pair), -2,2 (second pair), etc</p>
</td></tr>
<tr><td><code id="KO_gsa_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>&quot;BH&quot;</p>
</td></tr>
<tr><td><code id="KO_gsa_+3A_verbose">verbose</code></td>
<td>
<p>TRUE</p>
</td></tr>
<tr><td><code id="KO_gsa_+3A_perm">perm</code></td>
<td>
<p>1000</p>
</td></tr>
<tr><td><code id="KO_gsa_+3A_...">...</code></td>
<td>
<p>additional parameters to <code><a href="GSA.html#topic+GSA">GSA</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>enrich_res object
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## use `GSA` from the `GSA` package.
if (requireNamespace("GSA")) {
  data("reporter_score_res")
  gsa_res &lt;- KO_gsa(reporter_score_res, p.adjust.method = "none", perm = 200)
  plot(gsa_res)
}

</code></pre>

<hr>
<h2 id='KO_gsea'>Perform gene set enrichment analysis</h2><span id='topic+KO_gsea'></span>

<h3>Description</h3>

<p>Perform gene set enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_gsea(
  ko_stat,
  weight = "logFC",
  add_mini = NULL,
  p.adjust.method = "BH",
  type = c("pathway", "module")[1],
  feature = "ko",
  modulelist = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_gsea_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat dataframe from <code><a href="#topic+ko.test">ko.test</a></code>.</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_weight">weight</code></td>
<td>
<p>the metric used for ranking, default: logFC</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_add_mini">add_mini</code></td>
<td>
<p>add_mini when calculate the logFC. e.g (10+0.1)/(0+0.1), default 0.05*min(avg_abundance)</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>The method used for p-value adjustment (default: &quot;BH&quot;).</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_type">type</code></td>
<td>
<p>&quot;pathway&quot; or &quot;module&quot; for default KOlist_file.</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_feature">feature</code></td>
<td>
<p>one of &quot;ko&quot;, &quot;gene&quot;, &quot;compound&quot;</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="KO_gsea_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DOSE object
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>message("The following example require some time to run:")

## use `GSEA` from the `clusterProfiler` package.
if (requireNamespace("clusterProfiler")) {
  data("reporter_score_res")
  gsea_res &lt;- KO_gsea(reporter_score_res, p.adjust.method = "none")
  enrichplot::gseaplot(gsea_res, geneSetID = data.frame(gsea_res)$ID[1])
  gsea_res_df &lt;- as.enrich_res(gsea_res)
  plot(gsea_res_df)
}

</code></pre>

<hr>
<h2 id='KO_gsva'>Perform Gene Set Variation Analysis</h2><span id='topic+KO_gsva'></span>

<h3>Description</h3>

<p>Perform Gene Set Variation Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_gsva(
  reporter_res,
  verbose = TRUE,
  method = "wilcox.test",
  p.adjust.method = "BH",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_gsva_+3A_reporter_res">reporter_res</code></td>
<td>
<p>reporter_res</p>
</td></tr>
<tr><td><code id="KO_gsva_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="KO_gsva_+3A_method">method</code></td>
<td>
<p>see <code><a href="#topic+ko.test">ko.test</a></code></p>
</td></tr>
<tr><td><code id="KO_gsva_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p.adjust.method</p>
</td></tr>
<tr><td><code id="KO_gsva_+3A_...">...</code></td>
<td>
<p>additional parameters to <code><a href="GSVA.html#topic+gsva">gsva</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>enrich_res
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## use `gsva` from the `GSVA` package.
if (requireNamespace("GSVA")) {
  data("reporter_score_res")
  gsva_res &lt;- KO_gsva(reporter_score_res, p.adjust.method = "none")
}

</code></pre>

<hr>
<h2 id='KO_htable'>KO htable from 'KEGG'</h2><span id='topic+KO_htable'></span>

<h3>Description</h3>

<p>KO htable from 'KEGG'
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='KO_padog'>Perform Pathway Analysis with Down-weighting of Overlapping Genes (PADOG)</h2><span id='topic+KO_padog'></span>

<h3>Description</h3>

<p>Perform Pathway Analysis with Down-weighting of Overlapping Genes (PADOG)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_padog(
  reporter_res,
  verbose = TRUE,
  perm = 1000,
  p.adjust.method = "BH",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_padog_+3A_reporter_res">reporter_res</code></td>
<td>
<p>The input reporter result.</p>
</td></tr>
<tr><td><code id="KO_padog_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, print verbose messages. Default is TRUE.</p>
</td></tr>
<tr><td><code id="KO_padog_+3A_perm">perm</code></td>
<td>
<p>The number of permutations. Default is 1000.</p>
</td></tr>
<tr><td><code id="KO_padog_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>Method for p-value adjustment. Default is &quot;BH&quot;.</p>
</td></tr>
<tr><td><code id="KO_padog_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <code><a href="PADOG.html#topic+padog">padog</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing PADOG results for KO enrichment.
</p>
<p>A data frame with columns &quot;ID,&quot; &quot;Description,&quot; &quot;K_num,&quot; &quot;Exist_K_num,&quot; &quot;p.value,&quot; and &quot;p.adjust.&quot;
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## use `PADOG` from the `PADOG` package.
if (requireNamespace("PADOG")) {
  data("reporter_score_res")
  padog_res &lt;- KO_padog(reporter_score_res,
    verbose = TRUE,
    perm = 200, p.adjust.method = "none"
  )
}

</code></pre>

<hr>
<h2 id='KO_safe'>Perform Significance Analysis of Function and Expression</h2><span id='topic+KO_safe'></span>

<h3>Description</h3>

<p>Perform Significance Analysis of Function and Expression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_safe(
  reporter_res,
  verbose = TRUE,
  perm = 1000,
  C.matrix = NULL,
  p.adjust.method = "BH",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_safe_+3A_reporter_res">reporter_res</code></td>
<td>
<p>The input reporter result.</p>
</td></tr>
<tr><td><code id="KO_safe_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, print verbose messages. Default is TRUE.</p>
</td></tr>
<tr><td><code id="KO_safe_+3A_perm">perm</code></td>
<td>
<p>The number of permutations. Default is 1000.</p>
</td></tr>
<tr><td><code id="KO_safe_+3A_c.matrix">C.matrix</code></td>
<td>
<p>The contrast matrix. Default is NULL, and it will be generated from the module list.</p>
</td></tr>
<tr><td><code id="KO_safe_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>Method for p-value adjustment. Default is &quot;BH&quot;.</p>
</td></tr>
<tr><td><code id="KO_safe_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <code><a href="safe.html#topic+safe">safe</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing SAFE results for KO enrichment.
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## use `safe` from the `safe` package.
if (requireNamespace("safe")) {
  data("reporter_score_res")
  safe_res &lt;- KO_safe(reporter_score_res,
    verbose = TRUE,
    perm = 200, p.adjust.method = "none"
  )
}

</code></pre>

<hr>
<h2 id='KO_sea'>Perform Simultaneous Enrichment Analysis</h2><span id='topic+KO_sea'></span>

<h3>Description</h3>

<p>Perform Simultaneous Enrichment Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KO_sea(reporter_res, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KO_sea_+3A_reporter_res">reporter_res</code></td>
<td>
<p>The input reporter result.</p>
</td></tr>
<tr><td><code id="KO_sea_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, print verbose messages. Default is TRUE.</p>
</td></tr>
<tr><td><code id="KO_sea_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <code><a href="rSEA.html#topic+SEA">SEA</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>enrich_res
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+plot_enrich_res">plot_enrich_res</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## use `SEA` from the `rSEA` package.
if (requireNamespace("rSEA")) {
  data("reporter_score_res")
  sea_res &lt;- KO_sea(reporter_score_res, verbose = TRUE)
}

</code></pre>

<hr>
<h2 id='ko.test'>Differential analysis or Correlation analysis for KO-abundance table</h2><span id='topic+ko.test'></span>

<h3>Description</h3>

<p>Differential analysis or Correlation analysis for KO-abundance table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ko.test(
  kodf,
  group,
  metadata = NULL,
  method = "wilcox.test",
  pattern = NULL,
  p.adjust.method1 = "none",
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ko.test_+3A_kodf">kodf</code></td>
<td>
<p>KO_abundance table, rowname is ko id (e.g. K00001),colnames is samples.</p>
</td></tr>
<tr><td><code id="ko.test_+3A_group">group</code></td>
<td>
<p>The comparison groups (at least two categories) in your data, one column name of metadata when metadata exist or a vector whose length equal to columns number of kodf. And you can use factor levels to change order.</p>
</td></tr>
<tr><td><code id="ko.test_+3A_metadata">metadata</code></td>
<td>
<p>sample information data.frame contains group</p>
</td></tr>
<tr><td><code id="ko.test_+3A_method">method</code></td>
<td>
<p>the type of test. Default is 'wilcox.test'. Allowed values include:
</p>

<ul>
<li> <p><code><a href="stats.html#topic+t.test">t.test</a></code> (parametric) and <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code> (non-parametric). Perform comparison between two groups of samples. If the grouping variable contains more than two levels, then a pairwise comparison is performed.
</p>
</li>
<li> <p><code><a href="stats.html#topic+anova">anova</a></code> (parametric) and <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> (non-parametric). Perform one-way ANOVA test comparing multiple groups.
</p>
</li>
<li><p> 'pearson', 'kendall', or 'spearman' (correlation), see <code><a href="stats.html#topic+cor">cor</a></code>.</p>
</li></ul>
</td></tr>
<tr><td><code id="ko.test_+3A_pattern">pattern</code></td>
<td>
<p>a named vector matching the group, e.g. c('G1'=1,'G2'=3,'G3'=2), use the correlation analysis with specific pattern to calculate p-value.</p>
</td></tr>
<tr><td><code id="ko.test_+3A_p.adjust.method1">p.adjust.method1</code></td>
<td>
<p>p.adjust.method for 'ko.test', see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="ko.test_+3A_threads">threads</code></td>
<td>
<p>default 1</p>
</td></tr>
<tr><td><code id="ko.test_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ko_pvalue data.frame
</p>


<h3>See Also</h3>

<p>Other GRSA: 
<code><a href="#topic+get_reporter_score">get_reporter_score</a>()</code>,
<code><a href="#topic+pvalue2zs">pvalue2zs</a>()</code>,
<code><a href="#topic+reporter_score">reporter_score</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("KO_abundance_test")
ko_pvalue &lt;- ko.test(KO_abundance, "Group", metadata)

</code></pre>

<hr>
<h2 id='KOlist'>The KOlist used for enrichment.</h2><span id='topic+KOlist'></span>

<h3>Description</h3>

<p>an list contains two data.frame named pathway and module.
</p>


<h3>Format</h3>

<p>four columns in each data.frame.
</p>

<dl>
<dt>id</dt><dd><p>&quot;map0010&quot; or &quot;M00001&quot;</p>
</dd>
<dt>K_num</dt><dd><p>contians how many KOs in this pathway or module</p>
</dd>
<dt>KOs</dt><dd><p>KOs name</p>
</dd>
<dt>Description</dt><dd><p>the description of this pathway or module</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='load_CARDinfo'>Load the CARDinfo (from CARD database)</h2><span id='topic+load_CARDinfo'></span>

<h3>Description</h3>

<p>Load the CARDinfo (from CARD database)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_CARDinfo(verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_CARDinfo_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>CARDinfo
</p>

<hr>
<h2 id='load_GOlist'>Load the GOlist (from 'GO' database)</h2><span id='topic+load_GOlist'></span><span id='topic+load_GOinfo'></span>

<h3>Description</h3>

<p>Load the GOlist (from 'GO' database)
</p>
<p>Load the GOinfo (from GO)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_GOlist(verbose = TRUE)

load_GOinfo(verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_GOlist_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>GOlist
</p>
<p>GOinfo
</p>

<hr>
<h2 id='load_htable'>Load the specific table (from 'KEGG')</h2><span id='topic+load_htable'></span><span id='topic+load_KOlist'></span><span id='topic+load_CPDlist'></span><span id='topic+load_KO_desc'></span><span id='topic+load_KO_htable'></span><span id='topic+load_Pathway_htable'></span><span id='topic+load_Module_htable'></span><span id='topic+load_Compound_htable'></span><span id='topic+load_org_pathway'></span>

<h3>Description</h3>

<p>Load the specific table (from 'KEGG')
</p>
<p>Load the KOlist (from 'KEGG')
</p>
<p>Load the CPDlist (from 'KEGG')
</p>
<p>Load the KO description (from 'KEGG')
</p>
<p>Load the KO_htable (from 'KEGG')
</p>
<p>Load the Pathway_htable (from 'KEGG')
</p>
<p>Load the Module_htable (from 'KEGG')
</p>
<p>Load the Compound_htable (from 'KEGG')
</p>
<p>Load the pathway information for an organism (from 'KEGG')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_htable(type, verbose = TRUE)

load_KOlist(verbose = TRUE)

load_CPDlist(verbose = TRUE)

load_KO_desc(verbose = TRUE)

load_KO_htable(verbose = TRUE)

load_Pathway_htable(verbose = TRUE)

load_Module_htable(verbose = TRUE)

load_Compound_htable(verbose = TRUE)

load_org_pathway(org = "hsa", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_htable_+3A_type">type</code></td>
<td>
<p>&quot;ko&quot;, &quot;module&quot;, &quot;pathway&quot;, &quot;compound&quot; ...</p>
</td></tr>
<tr><td><code id="load_htable_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="load_htable_+3A_org">org</code></td>
<td>
<p>kegg organism, listed in https://www.genome.jp/kegg/catalog/org_list.html, default, &quot;hsa&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>KO_htable
</p>
<p>KOlist
</p>
<p>CPDlist
</p>
<p>KO description
</p>
<p>KO_htable
</p>
<p>Pathway_htable
</p>
<p>Module_htable
</p>
<p>Compound_htable
</p>
<p>KOlist
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Pathway_htable &lt;- load_htable("pathway")
head(Pathway_htable)
</code></pre>

<hr>
<h2 id='mmu_kegg_pathway'>pathway information for &quot;mmu&quot;</h2><span id='topic+mmu_kegg_pathway'></span>

<h3>Description</h3>

<p>pathway information for &quot;mmu&quot;
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>
</p>

<hr>
<h2 id='modify_description'>Modify the pathway description before plotting</h2><span id='topic+modify_description'></span>

<h3>Description</h3>

<p>Modify the pathway description before plotting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modify_description(
  reporter_res,
  pattern = " - Homo sapiens (human)",
  replacement = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="modify_description_+3A_reporter_res">reporter_res</code></td>
<td>
<p>reporter_res</p>
</td></tr>
<tr><td><code id="modify_description_+3A_pattern">pattern</code></td>
<td>
<p>str, like &quot; - Homo sapiens (human)&quot;</p>
</td></tr>
<tr><td><code id="modify_description_+3A_replacement">replacement</code></td>
<td>
<p>str, like &quot;&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>reporter_res
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("reporter_score_res")
modify_description(reporter_score_res, pattern = " - Homo sapiens (human)")
</code></pre>

<hr>
<h2 id='Module_htable'>Module htable from 'KEGG'</h2><span id='topic+Module_htable'></span>

<h3>Description</h3>

<p>Module htable from 'KEGG'
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Pathway_htable">Pathway_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='Pathway_htable'>Pathway htable from 'KEGG'</h2><span id='topic+Pathway_htable'></span>

<h3>Description</h3>

<p>Pathway htable from 'KEGG'
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+CPDlist">CPDlist</a></code>,
<code><a href="#topic+Compound_htable">Compound_htable</a></code>,
<code><a href="#topic+GOlist">GOlist</a></code>,
<code><a href="#topic+KO_htable">KO_htable</a></code>,
<code><a href="#topic+KOlist">KOlist</a></code>,
<code><a href="#topic+Module_htable">Module_htable</a></code>,
<code><a href="#topic+hsa_kegg_pathway">hsa_kegg_pathway</a></code>,
<code><a href="#topic+mmu_kegg_pathway">mmu_kegg_pathway</a></code>
</p>

<hr>
<h2 id='plot_enrich_res'>Plot enrich_res</h2><span id='topic+plot_enrich_res'></span><span id='topic+plot.enrich_res'></span>

<h3>Description</h3>

<p>Plot enrich_res
</p>
<p>Plot enrich_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_enrich_res(
  enrich_res,
  mode = 1,
  padj_threshold = 0.05,
  show_ID = FALSE,
  Pathway_description = TRUE,
  facet_level = FALSE,
  facet_anno = NULL,
  str_width = 50,
  facet_str_width = 15,
  ...
)

## S3 method for class 'enrich_res'
plot(
  x,
  mode = 1,
  padj_threshold = 0.05,
  show_ID = FALSE,
  Pathway_description = TRUE,
  facet_level = FALSE,
  facet_anno = NULL,
  str_width = 50,
  facet_str_width = 15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_enrich_res_+3A_enrich_res">enrich_res</code></td>
<td>
<p>enrich_res object</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_mode">mode</code></td>
<td>
<p>plot style: 1~2</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>p.adjust threshold</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_show_id">show_ID</code></td>
<td>
<p>show pathway id</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_pathway_description">Pathway_description</code></td>
<td>
<p>show KO description rather than KO id.</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_facet_level">facet_level</code></td>
<td>
<p>facet plot if the type is &quot;pathway&quot; or &quot;module&quot;</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_facet_anno">facet_anno</code></td>
<td>
<p>annotation table for facet, two columns, first is level summary, second is pathway id.</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_str_width">str_width</code></td>
<td>
<p>default: 50</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_facet_str_width">facet_str_width</code></td>
<td>
<p>str width for facet label</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
<tr><td><code id="plot_enrich_res_+3A_x">x</code></td>
<td>
<p>enrich_res object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>
<p>ggplot
</p>


<h3>See Also</h3>

<p>Other common_enrich: 
<code><a href="#topic+KO_enrich">KO_enrich</a>()</code>,
<code><a href="#topic+KO_fisher">KO_fisher</a>()</code>,
<code><a href="#topic+KO_gsa">KO_gsa</a>()</code>,
<code><a href="#topic+KO_gsea">KO_gsea</a>()</code>,
<code><a href="#topic+KO_gsva">KO_gsva</a>()</code>,
<code><a href="#topic+KO_padog">KO_padog</a>()</code>,
<code><a href="#topic+KO_safe">KO_safe</a>()</code>,
<code><a href="#topic+KO_sea">KO_sea</a>()</code>
</p>

<hr>
<h2 id='plot_features_box'>Plot features boxplot</h2><span id='topic+plot_features_box'></span><span id='topic+plot_KOs_box'></span>

<h3>Description</h3>

<p>Plot features boxplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_features_box(
  kodf,
  group = NULL,
  metadata = NULL,
  map_id = "map00780",
  select_ko = NULL,
  only_sig = FALSE,
  box_param = NULL,
  modulelist = NULL,
  KO_description = FALSE,
  str_width = 50
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_features_box_+3A_kodf">kodf</code></td>
<td>
<p>KO_abundance table, rowname is ko id (e.g. K00001),colnames is samples. or result of 'get_reporter_score'</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_group">group</code></td>
<td>
<p>The compare group (two category) in your data, one column name of metadata when metadata exist or a vector whose length equal to columns number of kodf.</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_metadata">metadata</code></td>
<td>
<p>metadata</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_select_ko">select_ko</code></td>
<td>
<p>select which ko</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_only_sig">only_sig</code></td>
<td>
<p>only show the significant features</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_box_param">box_param</code></td>
<td>
<p>parameters pass to <code><a href="pcutils.html#topic+group_box">group_box</a></code></p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_ko_description">KO_description</code></td>
<td>
<p>show KO description rather than KO id.</p>
</td></tr>
<tr><td><code id="plot_features_box_+3A_str_width">str_width</code></td>
<td>
<p>str_width to wrap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("reporter_score_res")
plot_features_box(reporter_score_res,
  select_ko = c("K00059", "K00208", "K00647", "K00652", "K00833", "K01012"),
  box_param = list(p_value1 = FALSE, trend_line = TRUE)
)
plot_features_box(reporter_score_res,
  select_ko = "K00059", KO_description = TRUE,
  box_param = list(p_value1 = FALSE, trend_line = TRUE)
)

</code></pre>

<hr>
<h2 id='plot_features_distribution'>plot the Z-score of features distribution</h2><span id='topic+plot_features_distribution'></span><span id='topic+plot_KOs_distribution'></span>

<h3>Description</h3>

<p>plot the Z-score of features distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_features_distribution(
  reporter_res,
  map_id,
  text_size = 4,
  text_position = NULL,
  rug_length = 0.04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_features_distribution_+3A_reporter_res">reporter_res</code></td>
<td>
<p>result of 'reporter_score'</p>
</td></tr>
<tr><td><code id="plot_features_distribution_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
<tr><td><code id="plot_features_distribution_+3A_text_size">text_size</code></td>
<td>
<p>text_size=4</p>
</td></tr>
<tr><td><code id="plot_features_distribution_+3A_text_position">text_position</code></td>
<td>
<p>text_position, e.g. c(x=3,y=0.4)</p>
</td></tr>
<tr><td><code id="plot_features_distribution_+3A_rug_length">rug_length</code></td>
<td>
<p>rug_length=0.04</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("reporter_score_res")
plot_features_distribution(reporter_score_res, map_id = c("map05230", "map03010"))

</code></pre>

<hr>
<h2 id='plot_features_heatmap'>Plot features heatmap</h2><span id='topic+plot_features_heatmap'></span><span id='topic+plot_KOs_heatmap'></span>

<h3>Description</h3>

<p>Plot features heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_features_heatmap(
  kodf,
  group = NULL,
  metadata = NULL,
  map_id = "map00780",
  select_ko = NULL,
  only_sig = FALSE,
  columns = NULL,
  modulelist = NULL,
  KO_description = FALSE,
  str_width = 50,
  heatmap_param = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_features_heatmap_+3A_kodf">kodf</code></td>
<td>
<p>KO_abundance table, rowname is ko id (e.g. K00001),colnames is samples. or result of 'get_reporter_score'</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_group">group</code></td>
<td>
<p>The compare group (two category) in your data, one column name of metadata when metadata exist or a vector whose length equal to columns number of kodf.</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_metadata">metadata</code></td>
<td>
<p>metadata</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_select_ko">select_ko</code></td>
<td>
<p>select which ko</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_only_sig">only_sig</code></td>
<td>
<p>only show the significant KOs</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_columns">columns</code></td>
<td>
<p>change columns</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_ko_description">KO_description</code></td>
<td>
<p>show KO description rather than KO id.</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_str_width">str_width</code></td>
<td>
<p>str_width to wrap</p>
</td></tr>
<tr><td><code id="plot_features_heatmap_+3A_heatmap_param">heatmap_param</code></td>
<td>
<p>parameters pass to <code><a href="pheatmap.html#topic+pheatmap">pheatmap</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("pheatmap")) {
  data("reporter_score_res")
  plot_features_heatmap(reporter_score_res, map_id = "map00780")
}

</code></pre>

<hr>
<h2 id='plot_features_in_pathway'>Plot features trend in one pathway or module</h2><span id='topic+plot_features_in_pathway'></span><span id='topic+plot_KOs_in_pathway'></span>

<h3>Description</h3>

<p>Plot features trend in one pathway or module
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_features_in_pathway(
  ko_stat,
  map_id = "map00780",
  modulelist = NULL,
  select_ko = NULL,
  box_color = reporter_color,
  show_number = TRUE,
  scale = FALSE,
  feature_type = "KOs",
  line_color = c(Depleted = "seagreen", Enriched = "orange", None = "grey", Significant =
    "red2")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_features_in_pathway_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat result from <code><a href="#topic+pvalue2zs">pvalue2zs</a></code> or result of 'get_reporter_score'</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_select_ko">select_ko</code></td>
<td>
<p>select which ko</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_box_color">box_color</code></td>
<td>
<p>box and point color, default: c(&quot;#e31a1c&quot;,&quot;#1f78b4&quot;)</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_show_number">show_number</code></td>
<td>
<p>show the numbers.</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_scale">scale</code></td>
<td>
<p>scale the data by row.</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_feature_type">feature_type</code></td>
<td>
<p>show in the title ,default: KOs</p>
</td></tr>
<tr><td><code id="plot_features_in_pathway_+3A_line_color">line_color</code></td>
<td>
<p>line color, default: c(&quot;Depleted&quot;=&quot;seagreen&quot;,&quot;Enriched&quot;=&quot;orange&quot;,&quot;None&quot;=&quot;grey&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("reporter_score_res")
plot_features_in_pathway(ko_stat = reporter_score_res, map_id = "map00860")

</code></pre>

<hr>
<h2 id='plot_features_network'>Plot features network</h2><span id='topic+plot_features_network'></span><span id='topic+plot_KOs_network'></span>

<h3>Description</h3>

<p>Plot features network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_features_network(
  ko_stat,
  map_id = "map00780",
  near_pathway = FALSE,
  modulelist = NULL,
  kos_color = c(Depleted = "seagreen", Enriched = "orange", None = "grey", Significant =
    "red2", Pathway = "#80b1d3"),
  pathway_label = TRUE,
  kos_label = TRUE,
  mark_module = FALSE,
  mark_color = NULL,
  return_net = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_features_network_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat result from <code><a href="#topic+pvalue2zs">pvalue2zs</a></code> or result of 'get_reporter_score'</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_near_pathway">near_pathway</code></td>
<td>
<p>show the near_pathway if any features exist.</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_kos_color">kos_color</code></td>
<td>
<p>default, c(&quot;Depleted&quot;=&quot;seagreen&quot;,&quot;Enriched&quot;=&quot;orange&quot;,&quot;None&quot;=&quot;grey&quot;,&quot;Significant&quot;=&quot;red2&quot;)</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_pathway_label">pathway_label</code></td>
<td>
<p>show pathway_label?</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_kos_label">kos_label</code></td>
<td>
<p>show kos_label?</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_mark_module">mark_module</code></td>
<td>
<p>mark the modules?</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_mark_color">mark_color</code></td>
<td>
<p>mark colors, default, c(&quot;Depleted&quot;=&quot;seagreen&quot;,&quot;Enriched&quot;=&quot;orange&quot;,&quot;None&quot;=&quot;grey&quot;,&quot;Significant&quot;=&quot;red2&quot;)</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_return_net">return_net</code></td>
<td>
<p>return the network</p>
</td></tr>
<tr><td><code id="plot_features_network_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="MetaNet.html#topic+c_net_plot">c_net_plot</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>network plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("MetaNet")) {
  data("reporter_score_res")
  plot_features_network(reporter_score_res, map_id = "map05230")
  plot_features_network(reporter_score_res, map_id = "map00780", near_pathway = TRUE)
}
</code></pre>

<hr>
<h2 id='plot_htable'>Plot htable levels</h2><span id='topic+plot_htable'></span>

<h3>Description</h3>

<p>Plot htable levels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_htable(type = "ko", select = NULL, htable = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_htable_+3A_type">type</code></td>
<td>
<p>&quot;ko&quot;, &quot;module&quot;, &quot;pathway&quot;, &quot;compound&quot;</p>
</td></tr>
<tr><td><code id="plot_htable_+3A_select">select</code></td>
<td>
<p>select ids</p>
</td></tr>
<tr><td><code id="plot_htable_+3A_htable">htable</code></td>
<td>
<p>custom a htable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("KO_abundance_test")
plot_htable(select = rownames(KO_abundance))

</code></pre>

<hr>
<h2 id='plot_KEGG_map'>plot_KEGG_map</h2><span id='topic+plot_KEGG_map'></span>

<h3>Description</h3>

<p>plot_KEGG_map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_KEGG_map(
  ko_stat,
  map_id = "map00780",
  modulelist = NULL,
  type = "pathway",
  feature = "ko",
  color_var = "Z_score",
  save_dir,
  color = c("seagreen", "grey", "orange")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_KEGG_map_+3A_ko_stat">ko_stat</code></td>
<td>
<p>ko_stat result from <code><a href="#topic+pvalue2zs">pvalue2zs</a></code> or result of 'get_reporter_score'</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain &quot;id&quot;,&quot;K_num&quot;,&quot;KOs&quot;,&quot;Description&quot; columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_type">type</code></td>
<td>
<p>&quot;pathway&quot; or &quot;module&quot; for default KOlist for microbiome, &quot;CC&quot;, &quot;MF&quot;, &quot;BP&quot;, &quot;ALL&quot; for default GOlist for homo sapiens. And org in listed in 'https://www.genome.jp/kegg/catalog/org_list.html' such as &quot;hsa&quot; (if your kodf is come from a specific organism, you should specify type here).</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_feature">feature</code></td>
<td>
<p>one of &quot;ko&quot;, &quot;gene&quot;, &quot;compound&quot;</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_color_var">color_var</code></td>
<td>
<p>use which variable to color</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_save_dir">save_dir</code></td>
<td>
<p>where to save the png files</p>
</td></tr>
<tr><td><code id="plot_KEGG_map_+3A_color">color</code></td>
<td>
<p>color</p>
</td></tr>
</table>


<h3>Value</h3>

<p>png files
</p>


<h3>References</h3>

<p>https://zhuanlan.zhihu.com/p/357687076
</p>


<h3>Examples</h3>

<pre><code class='language-R'>message("The following example will download some files, run yourself:")

if (requireNamespace("pathview")) {
  output_dir &lt;- tempdir()
  data("reporter_score_res")
  plot_KEGG_map(reporter_score_res$ko_stat,
    map_id = "map00780", type = "pathway",
    feature = "ko", color_var = "Z_score", save_dir = output_dir
  )
}

</code></pre>

<hr>
<h2 id='plot_report'>Plot the reporter_res</h2><span id='topic+plot_report'></span><span id='topic+plot_report_bar'></span>

<h3>Description</h3>

<p>Plot the reporter_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_report(
  reporter_res,
  rs_threshold = 1.64,
  mode = 1,
  y_text_size = 13,
  str_width = 100,
  show_ID = FALSE,
  Pathway_description = TRUE,
  facet_level = FALSE,
  facet_anno = NULL,
  facet_str_width = 15
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_report_+3A_reporter_res">reporter_res</code></td>
<td>
<p>result of 'get_reporter_score' or 'reporter_score'</p>
</td></tr>
<tr><td><code id="plot_report_+3A_rs_threshold">rs_threshold</code></td>
<td>
<p>plot threshold vector, default:1.64</p>
</td></tr>
<tr><td><code id="plot_report_+3A_mode">mode</code></td>
<td>
<p>1~2 plot style.</p>
</td></tr>
<tr><td><code id="plot_report_+3A_y_text_size">y_text_size</code></td>
<td>
<p>y_text_size</p>
</td></tr>
<tr><td><code id="plot_report_+3A_str_width">str_width</code></td>
<td>
<p>str_width to wrap</p>
</td></tr>
<tr><td><code id="plot_report_+3A_show_id">show_ID</code></td>
<td>
<p>show pathway id</p>
</td></tr>
<tr><td><code id="plot_report_+3A_pathway_description">Pathway_description</code></td>
<td>
<p>show KO description rather than KO id.</p>
</td></tr>
<tr><td><code id="plot_report_+3A_facet_level">facet_level</code></td>
<td>
<p>facet plot if the type is &quot;pathway&quot; or &quot;module&quot;</p>
</td></tr>
<tr><td><code id="plot_report_+3A_facet_anno">facet_anno</code></td>
<td>
<p>annotation table for facet, two columns, first is level summary, second is pathway id.</p>
</td></tr>
<tr><td><code id="plot_report_+3A_facet_str_width">facet_str_width</code></td>
<td>
<p>str width for facet label</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("reporter_score_res")
plot_report(reporter_score_res, rs_threshold = c(2.5, -2.5), y_text_size = 10, str_width = 40)
</code></pre>

<hr>
<h2 id='plot_report_circle_packing'>Plot the reporter_res as circle_packing</h2><span id='topic+plot_report_circle_packing'></span>

<h3>Description</h3>

<p>Plot the reporter_res as circle_packing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_report_circle_packing(
  reporter_res,
  rs_threshold = 1.64,
  mode = 2,
  facet_anno = NULL,
  show_ID = FALSE,
  Pathway_description = TRUE,
  str_width = 10,
  show_level_name = "all",
  show_tip_label = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_report_circle_packing_+3A_reporter_res">reporter_res</code></td>
<td>
<p>result of 'get_reporter_score'</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_rs_threshold">rs_threshold</code></td>
<td>
<p>plot threshold vector, default:1.64</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_mode">mode</code></td>
<td>
<p>1~2 plot style.</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_facet_anno">facet_anno</code></td>
<td>
<p>annotation table for facet, more two columns, last is pathway name, last second is pathway id.</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_show_id">show_ID</code></td>
<td>
<p>show pathway id</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_pathway_description">Pathway_description</code></td>
<td>
<p>show KO description rather than KO id.</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_str_width">str_width</code></td>
<td>
<p>str_width to wrap</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_show_level_name">show_level_name</code></td>
<td>
<p>show the level name?</p>
</td></tr>
<tr><td><code id="plot_report_circle_packing_+3A_show_tip_label">show_tip_label</code></td>
<td>
<p>show the tip label?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("reporter_score_res")
if (requireNamespace("igraph") &amp;&amp; requireNamespace("ggraph")) {
  plot_report_circle_packing(reporter_score_res, rs_threshold = c(2, -2), str_width = 40)
}

</code></pre>

<hr>
<h2 id='plot_significance'>Plot the significance of pathway</h2><span id='topic+plot_significance'></span>

<h3>Description</h3>

<p>Plot the significance of pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_significance(reporter_res, map_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_significance_+3A_reporter_res">reporter_res</code></td>
<td>
<p>result of 'get_reporter_score' or 'reporter_score'</p>
</td></tr>
<tr><td><code id="plot_significance_+3A_map_id">map_id</code></td>
<td>
<p>the pathway or module id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("reporter_score_res")
plot_significance(reporter_score_res, map_id = c("map05230", "map03010"))

</code></pre>

<hr>
<h2 id='plot.cm_res'>Plot c_means result</h2><span id='topic+plot.cm_res'></span>

<h3>Description</h3>

<p>Plot c_means result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cm_res'
plot(
  x,
  filter_membership,
  mode = 1,
  show.clust.cent = TRUE,
  show_num = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.cm_res_+3A_x">x</code></td>
<td>
<p>a cm_res object</p>
</td></tr>
<tr><td><code id="plot.cm_res_+3A_filter_membership">filter_membership</code></td>
<td>
<p>filter membership</p>
</td></tr>
<tr><td><code id="plot.cm_res_+3A_mode">mode</code></td>
<td>
<p>1~2</p>
</td></tr>
<tr><td><code id="plot.cm_res_+3A_show.clust.cent">show.clust.cent</code></td>
<td>
<p>show cluster center?</p>
</td></tr>
<tr><td><code id="plot.cm_res_+3A_show_num">show_num</code></td>
<td>
<p>show number of each cluster?</p>
</td></tr>
<tr><td><code id="plot.cm_res_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='print.reporter_score'>Print reporter_score</h2><span id='topic+print.reporter_score'></span>

<h3>Description</h3>

<p>Print reporter_score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'reporter_score'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.reporter_score_+3A_x">x</code></td>
<td>
<p>reporter_score</p>
</td></tr>
<tr><td><code id="print.reporter_score_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.rs_by_cm'>Print rs_by_cm</h2><span id='topic+print.rs_by_cm'></span>

<h3>Description</h3>

<p>Print rs_by_cm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rs_by_cm'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.rs_by_cm_+3A_x">x</code></td>
<td>
<p>rs_by_cm</p>
</td></tr>
<tr><td><code id="print.rs_by_cm_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='pvalue2zs'>Transfer p-value of KOs to Z-score</h2><span id='topic+pvalue2zs'></span>

<h3>Description</h3>

<p>Transfer p-value of KOs to Z-score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalue2zs(
  ko_pvalue,
  mode = c("directed", "mixed")[1],
  p.adjust.method1 = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalue2zs_+3A_ko_pvalue">ko_pvalue</code></td>
<td>
<p>data.frame from <code><a href="#topic+ko.test">ko.test</a></code></p>
</td></tr>
<tr><td><code id="pvalue2zs_+3A_mode">mode</code></td>
<td>
<p>'mixed' or 'directed' (default, only for two groups differential analysis or multi-groups correlation analysis.), see details in <code><a href="#topic+pvalue2zs">pvalue2zs</a></code>.</p>
</td></tr>
<tr><td><code id="pvalue2zs_+3A_p.adjust.method1">p.adjust.method1</code></td>
<td>
<p>p.adjust.method for 'ko.test', see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>'<strong>mixed</strong>' mode is the original reporter-score method from Patil, K. R. et al. PNAS 2005.
In this mode, the reporter score is <strong>undirected</strong>, and the larger the reporter score, the more significant the enrichment, but it cannot indicate the up-and-down regulation information of the pathway! (Liu, L. et al. iMeta 2023.)
</p>
<p>steps:
</p>
<p>1. Use the Wilcoxon rank sum test to obtain the P value of the significance of each KO difference between the two groups (ie <code class="reqn">P_{koi}</code>, i represents a certain KO);
</p>
<p>2. Using an inverse normal distribution, convert the P value of each KO into a Z value (<code class="reqn">Z_{koi}</code>), the formula:
</p>
<p><code class="reqn">Z_{koi}=\theta ^{-1}(1-P_{koi})</code>
</p>
<p>3. 'Upgrade' KO to pathway: <code class="reqn">Z_{koi}</code>, calculate the Z value of the pathway, the formula:
</p>
<p><code class="reqn">Z_{pathway}=\frac{1}{\sqrt{k}}\sum Z_{koi}</code>
</p>
<p>where k means A total of k KOs were annotated to the corresponding pathway;
</p>
<p>4. Evaluate the degree of significance: permutation (permutation) 1000 times, get the random distribution of <code class="reqn">Z_{pathway}</code>, the formula:
</p>
<p><code class="reqn">Z_{adjustedpathway}=(Z_{pathway}-\mu _k)/\sigma _k</code>
</p>
<p><code class="reqn">\mu _k</code> is The mean of the random distribution, <code class="reqn">\sigma _k</code> is the standard deviation of the random distribution.
</p>
<p>Instead, '<strong>directed</strong>' mode is a derived version of 'mixed', referenced from <code>https://github.com/wangpeng407/ReporterScore</code>.
</p>
<p>This approach is based on the same assumption of many differential analysis methods: the expression of most genes has no significant change.
</p>
<p>steps:
</p>
<p>1. Use the Wilcoxon rank sum test to obtain the P value of the significance of each KO difference between the two groups (ie <code class="reqn">P_{koi}</code>, i represents a certain KO), and then divide the P value by 2, that is, the range of (0,1] becomes (0,0.5], <code class="reqn">P_{koi}=P_{koi}/2</code>;
</p>
<p>2. Using an inverse normal distribution, convert the P value of each KO into a Z value (<code class="reqn">Z_{koi}</code>), the formula:
</p>
<p><code class="reqn">Z_{koi}=\theta ^{-1}(1-P_{koi})</code>
</p>
<p>since the above P value is less than 0.5, all Z values will be greater than 0;
</p>
<p>3. Considering whether each KO is up-regulated or down-regulated, calculate <code class="reqn">diff\_KO</code>,
</p>
<p><code class="reqn">Z_{koi}=-Z_{koi}\ \ \ \ (diff\_KO&lt;0)</code>,
</p>
<p>so <code class="reqn">Z_{koi}</code> is greater than 0 Up-regulation, <code class="reqn">Z_{koi}</code> less than 0 is down-regulation;
</p>
<p>4. 'Upgrade' KO to pathway: <code class="reqn">Z_{koi}</code>, calculate the Z value of the pathway, the formula:
</p>
<p><code class="reqn">Z_{pathway}=\frac{1}{\sqrt{k}}\sum Z_{koi}</code>
</p>
<p>where k means A total of k KOs were annotated to the corresponding pathway;
</p>
<p>5. Evaluate the degree of significance: permutation (permutation) 1000 times, get the random distribution of <code class="reqn">Z_{pathway}</code>, the formula:
</p>
<p><code class="reqn">Z_{adjustedpathway}=(Z_{pathway}-\mu _k)/\sigma _k</code>
</p>
<p><code class="reqn">\mu _k</code> is The mean of the random distribution, <code class="reqn">\sigma _k</code> is the standard deviation of the random distribution.
</p>
<p>The finally obtained <code class="reqn">Z_{adjustedpathway}</code> is the Reporter score value enriched for each pathway.
In this mode, the Reporter score is directed, and a larger positive value represents a significant up-regulation enrichment, and a smaller negative values represent significant down-regulation enrichment.
</p>
<p>However, the disadvantage of this mode is that when a pathway contains about the same number of significantly up-regulates KOs and significantly down-regulates KOs, the final absolute value of Reporter score may approach 0, becoming a pathway that has not been significantly enriched.
</p>


<h3>Value</h3>

<p>ko_stat data.frame
</p>


<h3>References</h3>

<p>1. Patil, K. R. &amp; Nielsen, J. Uncovering transcriptional regulation of metabolism by using metabolic network topology. Proc Natl Acad Sci U S A 102, 2685–2689 (2005).
2. Liu, L., Zhu, R. &amp; Wu, D. Misuse of reporter score in microbial enrichment analysis. iMeta n/a, e95.
3. <code>https://github.com/wangpeng407/ReporterScore</code>
</p>


<h3>See Also</h3>

<p>Other GRSA: 
<code><a href="#topic+get_reporter_score">get_reporter_score</a>()</code>,
<code><a href="#topic+ko.test">ko.test</a>()</code>,
<code><a href="#topic+reporter_score">reporter_score</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("KO_abundance_test")
ko_pvalue &lt;- ko.test(KO_abundance, "Group", metadata)
ko_stat &lt;- pvalue2zs(ko_pvalue, mode = "directed")

</code></pre>

<hr>
<h2 id='reporter_score'>One step to get the reporter score of your KO abundance table.</h2><span id='topic+reporter_score'></span><span id='topic+GRSA'></span><span id='topic+RSA'></span>

<h3>Description</h3>

<p>One step to get the reporter score of your KO abundance table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reporter_score(
  kodf,
  group,
  metadata = NULL,
  method = "wilcox.test",
  pattern = NULL,
  p.adjust.method1 = "none",
  mode = c("directed", "mixed")[1],
  verbose = TRUE,
  feature = "ko",
  type = c("pathway", "module")[1],
  p.adjust.method2 = "BH",
  modulelist = NULL,
  threads = 1,
  perm = 4999,
  min_exist_KO = 3,
  max_exist_KO = 600
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reporter_score_+3A_kodf">kodf</code></td>
<td>
<p>KO_abundance table, rowname is ko id (e.g. K00001),colnames is samples.</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_group">group</code></td>
<td>
<p>The comparison groups (at least two categories) in your data, one column name of metadata when metadata exist or a vector whose length equal to columns number of kodf. And you can use factor levels to change order.</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_metadata">metadata</code></td>
<td>
<p>sample information data.frame contains group</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_method">method</code></td>
<td>
<p>the type of test. Default is 'wilcox.test'. Allowed values include:
</p>

<ul>
<li> <p><code><a href="stats.html#topic+t.test">t.test</a></code> (parametric) and <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code> (non-parametric). Perform comparison between two groups of samples. If the grouping variable contains more than two levels, then a pairwise comparison is performed.
</p>
</li>
<li> <p><code><a href="stats.html#topic+anova">anova</a></code> (parametric) and <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> (non-parametric). Perform one-way ANOVA test comparing multiple groups.
</p>
</li>
<li><p> 'pearson', 'kendall', or 'spearman' (correlation), see <code><a href="stats.html#topic+cor">cor</a></code>.</p>
</li></ul>
</td></tr>
<tr><td><code id="reporter_score_+3A_pattern">pattern</code></td>
<td>
<p>a named vector matching the group, e.g. c('G1'=1,'G2'=3,'G3'=2), use the correlation analysis with specific pattern to calculate p-value.</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_p.adjust.method1">p.adjust.method1</code></td>
<td>
<p>p.adjust.method for 'ko.test', see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="reporter_score_+3A_mode">mode</code></td>
<td>
<p>'mixed' or 'directed' (default, only for two groups differential analysis or multi-groups correlation analysis.), see details in <code><a href="#topic+pvalue2zs">pvalue2zs</a></code>.</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_feature">feature</code></td>
<td>
<p>one of 'ko', 'gene', 'compound'</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_type">type</code></td>
<td>
<p>'pathway' or 'module' for default KOlist for microbiome, 'CC', 'MF', 'BP', 'ALL' for default GOlist for homo sapiens. And org in listed in 'https://www.genome.jp/kegg/catalog/org_list.html' such as 'hsa' (if your kodf is come from a specific organism, you should specify type here).</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_p.adjust.method2">p.adjust.method2</code></td>
<td>
<p>p.adjust.method for the correction of ReporterScore, see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="reporter_score_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain 'id','K_num','KOs','Description' columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_threads">threads</code></td>
<td>
<p>default 1</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_perm">perm</code></td>
<td>
<p>permutation number, default: 4999.</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_min_exist_ko">min_exist_KO</code></td>
<td>
<p>min exist KO number in a pathway (default, 3, when a pathway contains KOs less than 3, there will be no RS)</p>
</td></tr>
<tr><td><code id="reporter_score_+3A_max_exist_ko">max_exist_KO</code></td>
<td>
<p>max exist KO number in a pathway (default, 600, when a pathway contains KOs more than 600, there will be no RS)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>reporter_score object:
</p>
<table>
<tr><td><code>kodf</code></td>
<td>
<p>your input KO_abundance table</p>
</td></tr>
<tr><td><code>ko_stat</code></td>
<td>
<p>ko statistics result contains p.value and z_score</p>
</td></tr>
<tr><td><code>reporter_s</code></td>
<td>
<p>the reporter score in each pathway</p>
</td></tr>
<tr><td><code>modulelist</code></td>
<td>
<p>default KOlist or customized modulelist dataframe</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>The comparison groups in your data</p>
</td></tr>
<tr><td><code>metadata</code></td>
<td>
<p>sample information dataframe contains group</p>
</td></tr>
</table>
<p>for the 'reporter_s' in result, whose columns represent:
</p>
<table>
<tr><td><code>ID</code></td>
<td>
<p>pathway id</p>
</td></tr>
<tr><td><code>Description</code></td>
<td>
<p>pathway description</p>
</td></tr>
<tr><td><code>K_num</code></td>
<td>
<p>total number of KOs/genes in the pathway</p>
</td></tr>
<tr><td><code>Exist_K_num</code></td>
<td>
<p>number of KOs/genes in your inputdata that exist in the pathway</p>
</td></tr>
<tr><td><code>Significant_K_num</code></td>
<td>
<p>number of kos/genes in your inputdata that are significant in the pathway</p>
</td></tr>
<tr><td><code>Z_score</code></td>
<td>
<p><code class="reqn">Z_{pathway}=\frac{1}{\sqrt{k}}\sum Z_{koi}</code></p>
</td></tr>
<tr><td><code>BG_Mean</code></td>
<td>
<p>Background mean, <code class="reqn">\mu _k</code></p>
</td></tr>
<tr><td><code>BG_Sd</code></td>
<td>
<p>Background standard deviation, <code class="reqn">\sigma _k</code></p>
</td></tr>
<tr><td><code>ReporterScore</code></td>
<td>
<p>ReporterScore of the pathway, <code class="reqn">ReporterScore=(Z_{pathway}-\mu _k)/\sigma _k</code></p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>p.value of the ReporterScore</p>
</td></tr>
<tr><td><code>p.adjust</code></td>
<td>
<p>adjusted p.value by p.adjust.method2</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other GRSA: 
<code><a href="#topic+get_reporter_score">get_reporter_score</a>()</code>,
<code><a href="#topic+ko.test">ko.test</a>()</code>,
<code><a href="#topic+pvalue2zs">pvalue2zs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>message("The following example require some time to run:")

data("KO_abundance_test")
reporter_score_res &lt;- reporter_score(KO_abundance, "Group", metadata,
  mode = "directed", perm = 499
)
head(reporter_score_res$reporter_s)
reporter_score_res2 &lt;- reporter_score(KO_abundance, "Group2", metadata,
  mode = "mixed",
  method = "kruskal.test", p.adjust.method1 = "none", perm = 499
)
reporter_score_res3 &lt;- reporter_score(KO_abundance, "Group2", metadata,
  mode = "directed",
  method = "pearson", pattern = c("G1" = 1, "G2" = 3, "G3" = 2), perm = 499
)

</code></pre>

<hr>
<h2 id='reporter_score_res'>'reporter_score()' result from KO_abundance_test</h2><span id='topic+reporter_score_res'></span><span id='topic+reporter_score_res2'></span>

<h3>Description</h3>

<p>'reporter_score()' result from KO_abundance_test
</p>
<p>'reporter_score()' result from KO_abundance_test
</p>


<h3>Format</h3>

<p>a list contain 7 elements.
</p>

<dl>
<dt>kodf</dt><dd><p>your input KO_abundance table</p>
</dd>
<dt>ko_stat</dt><dd><p>ko statistics result contains p.value and z_score</p>
</dd>
<dt>reporter_s</dt><dd><p>the reporter score in each pathway</p>
</dd>
<dt>modulelist</dt><dd><p>default KOlist or customized modulelist dataframe</p>
</dd>
<dt>group</dt><dd><p>The compare group (two category) in your data</p>
</dd>
<dt>metadata</dt><dd><p>sample information dataframe contains group</p>
</dd>
</dl>



<h3>See Also</h3>

<p>Other test_data: 
<code><a href="#topic+KO_abundance">KO_abundance</a></code>,
<code><a href="#topic+genedf">genedf</a></code>
</p>

<hr>
<h2 id='RSA_by_cm'>Reporter score analysis after C-means clustering</h2><span id='topic+RSA_by_cm'></span><span id='topic+GRSA_by_cm'></span><span id='topic+extract_cluster'></span><span id='topic+plot_c_means'></span>

<h3>Description</h3>

<p>Reporter score analysis after C-means clustering
</p>
<p>Extract one cluster from rs_by_cm object
</p>
<p>Plot c_means result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RSA_by_cm(
  kodf,
  group,
  metadata = NULL,
  k_num = NULL,
  filter_var = 0.7,
  verbose = TRUE,
  method = "pearson",
  ...
)

extract_cluster(rsa_cm_res, cluster = 1)

plot_c_means(
  rsa_cm_res,
  filter_membership,
  mode = 1,
  show.clust.cent = TRUE,
  show_num = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RSA_by_cm_+3A_kodf">kodf</code></td>
<td>
<p>KO_abundance table, rowname is ko id (e.g. K00001),colnames is samples.</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_group">group</code></td>
<td>
<p>The comparison groups (at least two categories) in your data, one column name of metadata when metadata exist or a vector whose length equal to columns number of kodf. And you can use factor levels to change order.</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_metadata">metadata</code></td>
<td>
<p>sample information data.frame contains group</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_k_num">k_num</code></td>
<td>
<p>if NULL, perform the cm_test_k, else an integer</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_filter_var">filter_var</code></td>
<td>
<p>see c_means</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_method">method</code></td>
<td>
<p>method from <code><a href="#topic+reporter_score">reporter_score</a></code></p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_rsa_cm_res">rsa_cm_res</code></td>
<td>
<p>a cm_res object</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_cluster">cluster</code></td>
<td>
<p>integer</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_filter_membership">filter_membership</code></td>
<td>
<p>filter membership 0~1.</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_mode">mode</code></td>
<td>
<p>1~2</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_show.clust.cent">show.clust.cent</code></td>
<td>
<p>show cluster center?</p>
</td></tr>
<tr><td><code id="RSA_by_cm_+3A_show_num">show_num</code></td>
<td>
<p>show number of each cluster?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rs_by_cm
</p>
<p>reporter_score object
</p>
<p>ggplot
</p>


<h3>See Also</h3>

<p>Other C_means: 
<code><a href="#topic+cm_test_k">cm_test_k</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>message("The following example require some time to run:")

if (requireNamespace("e1071") &amp;&amp; requireNamespace("factoextra")) {
  data("KO_abundance_test")
  rsa_cm_res &lt;- RSA_by_cm(KO_abundance, "Group2", metadata,
    k_num = 3,
    filter_var = 0.7, method = "pearson", perm = 199
  )
  extract_cluster(rsa_cm_res, cluster = 1)
}

</code></pre>

<hr>
<h2 id='up_level_KO'>Upgrade the KO level</h2><span id='topic+up_level_KO'></span>

<h3>Description</h3>

<p>Upgrade the KO level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>up_level_KO(
  KO_abundance,
  level = "pathway",
  show_name = FALSE,
  modulelist = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="up_level_KO_+3A_ko_abundance">KO_abundance</code></td>
<td>
<p>KO_abundance</p>
</td></tr>
<tr><td><code id="up_level_KO_+3A_level">level</code></td>
<td>
<p>one of 'pathway', 'module', 'level1', 'level2', 'level3', 'module1', 'module2', 'module3'.</p>
</td></tr>
<tr><td><code id="up_level_KO_+3A_show_name">show_name</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="up_level_KO_+3A_modulelist">modulelist</code></td>
<td>
<p>NULL or customized modulelist dataframe, must contain 'id','K_num','KOs','Description' columns. Take the 'KOlist' as example, use <code><a href="#topic+custom_modulelist">custom_modulelist</a></code>.</p>
</td></tr>
<tr><td><code id="up_level_KO_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("KO_abundance_test")
KO_level1 &lt;- up_level_KO(KO_abundance, level = "level1", show_name = TRUE)
</code></pre>

<hr>
<h2 id='update_CARDinfo'>update CARDinfo from (from 'CARD' database)</h2><span id='topic+update_CARDinfo'></span>

<h3>Description</h3>

<p>update CARDinfo from (from 'CARD' database)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_CARDinfo(download_dir = NULL, card_data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_CARDinfo_+3A_download_dir">download_dir</code></td>
<td>
<p>download_dir</p>
</td></tr>
<tr><td><code id="update_CARDinfo_+3A_card_data">card_data</code></td>
<td>
<p>card_data from https://card.mcmaster.ca/download/0/broadstreet-v3.2.8.tar.bz2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='update_GOlist'>Update the GO2gene files (from 'GO' database)</h2><span id='topic+update_GOlist'></span><span id='topic+update_GOinfo'></span>

<h3>Description</h3>

<p>Download links:
<code>http://geneontology.org/docs/download-ontology/</code>
<code>https://asa12138.github.io/FileList/GOlist.rda</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_GOlist(download_dir = NULL, GO_file = NULL)

update_GOinfo(download_dir = NULL, obo_file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_GOlist_+3A_download_dir">download_dir</code></td>
<td>
<p>download_dir</p>
</td></tr>
<tr><td><code id="update_GOlist_+3A_go_file">GO_file</code></td>
<td>
<p>GO_file</p>
</td></tr>
<tr><td><code id="update_GOlist_+3A_obo_file">obo_file</code></td>
<td>
<p>obo_file from http://current.geneontology.org/ontology/go.obo</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='update_KEGG'>Update files from 'KEGG'</h2><span id='topic+update_KEGG'></span><span id='topic+update_KO_file'></span><span id='topic+update_htable'></span><span id='topic+update_org_pathway'></span><span id='topic+download_org_pathway'></span><span id='topic+get_org_pathway'></span>

<h3>Description</h3>

<p>Download links:
</p>
<p><code>https://rest.kegg.jp/list/pathway</code>
<code>https://rest.kegg.jp/link/pathway/ko</code>
<code>https://rest.kegg.jp/link/pathway/compound</code>
<code>https://rest.kegg.jp/list/module</code>
<code>https://rest.kegg.jp/link/module/ko</code>
<code>https://rest.kegg.jp/link/module/compound</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_KEGG(download_dir)

update_KO_file(download_dir, RDSfile = NULL)

update_htable(type, keg_file = NULL, download = FALSE, download_dir = NULL)

update_org_pathway(
  org = "hsa",
  RDS_file = NULL,
  download = TRUE,
  download_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_KEGG_+3A_download_dir">download_dir</code></td>
<td>
<p>where to save the .keg file?</p>
</td></tr>
<tr><td><code id="update_KEGG_+3A_rdsfile">RDSfile</code></td>
<td>
<p>saved KO_files.RDS file</p>
</td></tr>
<tr><td><code id="update_KEGG_+3A_type">type</code></td>
<td>
<p>&quot;ko&quot;, &quot;module&quot;, &quot;pathway&quot;, &quot;compound&quot; ...</p>
</td></tr>
<tr><td><code id="update_KEGG_+3A_keg_file">keg_file</code></td>
<td>
<p>path of a .keg file, such as ko00001.keg from https://www.genome.jp/kegg-bin/download_htext?htext=ko00001&amp;format=htext.</p>
</td></tr>
<tr><td><code id="update_KEGG_+3A_download">download</code></td>
<td>
<p>save the .keg file?</p>
</td></tr>
<tr><td><code id="update_KEGG_+3A_org">org</code></td>
<td>
<p>kegg organism, listed in https://www.genome.jp/kegg/catalog/org_list.html, default, &quot;hsa&quot;</p>
</td></tr>
<tr><td><code id="update_KEGG_+3A_rds_file">RDS_file</code></td>
<td>
<p>path of a org.RDS file if you saved before.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
