<!DOCTYPE html><html lang="en"><head><title>Help for package REPTILE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {REPTILE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#REPTILE-package'>
<p>Regulatory Element Prediction</p></a></li>
<li><a href='#calculate_epimark_deviation'>
<p>Internal - calculating intensity deviation feature</p></a></li>
<li><a href='#get_option_parser_compute_score'><p>Internal - parsing options for REPTILE_compute_score.R</p></a></li>
<li><a href='#get_option_parser_evaluation'><p>Internal - parsing options for REPTILE_evaluate_prediction.R</p></a></li>
<li><a href='#get_option_parser_training'>
<p>Internal - parsing options for REPTILE_train.R</p></a></li>
<li><a href='#read_epigenomic_data'>
<p>Reading epigenomic data from epimark file</p></a></li>
<li><a href='#read_label'>
<p>Reading labels of regions from label file</p></a></li>
<li><a href='#reptile_eval_prediction'>
<p>Evaluating the prediction results</p></a></li>
<li><a href='#reptile_predict'>
<p>Predicting enhancer activity of given regions</p></a></li>
<li><a href='#reptile_predict_genome_wide'>
<p>Predicting enhancer activity</p></a></li>
<li><a href='#reptile_predict_one_mode'>
<p>Internal - predicting enhancer activity of DMRs or query regions</p></a></li>
<li><a href='#reptile_train'>
<p>Learn a REPTILE enhancer model</p></a></li>
<li><a href='#reptile_train_one_mode'>
<p>Internal - Learn single random forest classifier</p></a></li>
<li><a href='#rsd'><p>REPTILE sample data (rsd)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Regulatory DNA Element Prediction</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-6-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Yupeng He</td>
</tr>
<tr>
<td>Description:</td>
<td>Predicting regulatory DNA elements based on epigenomic signatures. This package is more of a set of building blocks than a direct solution. REPTILE regulatory prediction pipeline is built on this R package. See <a href="https://github.com/yupenghe/REPTILE">https://github.com/yupenghe/REPTILE</a> for more information.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yupeng He &lt;yupeng.he.bioinfo@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/yupenghe/REPTILE">https://github.com/yupenghe/REPTILE</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-2-Clause">BSD_2_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.2), foreach (&ge; 1.4.3), doParallel (&ge; 1.0.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>optparse (&ge; 1.3.2), randomForest (&ge; 4.6-12), flux(&ge; 0.3-0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-06-21 06:36:30 UTC; yupeng</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-06-21 09:23:58</td>
</tr>
</table>
<hr>
<h2 id='REPTILE-package'>
Regulatory Element Prediction
</h2><span id='topic+REPTILE-package'></span><span id='topic+REPTILE'></span>

<h3>Description</h3>

<p>Predicting DNA regulatory elements based on epigenomic
signatures. This package is more of a set of building blocks than a
direct solution. REPTILE regulatory prediction pipeline is built on
this R package. Please check the url below for details:
</p>
<p><a href="https://github.com/yupenghe/REPTILE">https://github.com/yupenghe/REPTILE</a>
</p>


<h3>Details</h3>

<p>Accurate enhancer identification is critical for understanding the
spatiotemporal transcriptional regulation during development as well as
the functional impact of disease-related non-coding genetic
variants. REPTILE is a algorithm to identify the precise location of
enhancers by integrating histone modification data and base-resolution
DNA methylation profiles.
</p>
<p>REPTILE was designed based on three observations: 1) regions that are
differentially methylated (or differentially methylated regions, DMRs)
across diverse cell and tissue types strongly overlap with
enhancers. 2) With base-resolution DNA methylation data, the
boundaries of DMRs can be accurately defined, circumventing the
difficulty of determining enhancer boundaries. 3) DMR size is often
smaller (~500bp) than known enhancers, known negative regions (regions
with no observable enhancer activity) and genomic windows used in
enhancer prediction (~2kb), all of which we termed as &quot;query
regions&quot;. Together with the association between transcription factor
binding and DNA methylation level, DMRs may serve as high-resolution
enhancer candidates and capture the local epigenomic patterns that
would otherwise be averaged/washed out in analysis focusing on the
query regions.
</p>
<p>Running REPTILE involves four major steps. First, to identify DMRs, we
compared the methylomes of target sample (where putative enhancers
will be generated) and several other samples with different
cell/tissue types (as reference). In the next step, input 
files for REPTILE are prepared, which store the information of query
regions, DMRs and the epigenomic data. Taking these inputs, REPTILE
represents each DMR or query region as a feature vector, where each
element corresponds to either intensity or intensity deviation of one
epigenetic mark. Intensity deviation is defined as the intensity in
target sample subtracted by the mean intensity in reference samples
(i.e. reference epigenome) and it captures the tissue-specificity of
each epigenetic mark. In the third step, based on the feature vectors
of known enhancers and  negative regions as well as the feature
vectors of the DMRs within them, we trained an enhancer model,
containing two random forest classifiers, which respectively predict
enhancer activities of query regions and DMRs. In the last step,
REPTILE uses the enhancer model to calculate enhancer confidence
scores for DMRs and query regions, based on which the final
predictions are made.
</p>
<p>The two key concepts on REPTILE are:
</p>

<ul>
<li><p> Query regions - known enhancers, known negative regions and
genomic windows used for enhancer prediction
</p>
</li>
<li><p> DMRs - differentially methylated regions
</p>
</li></ul>

<p>In REPTILE, DMRs are used as high-resolution candidates to capture the
fine epigenomic signatures in query regions.
</p>


<h3>Author(s)</h3>

<p>Yupeng He
</p>
<p>Maintainer: Yupeng He &lt;yupeng.he.bioinfo@gmail.com&gt;
</p>


<h3>References</h3>

<p>He, Yupeng et al., <em>REPTILE: Regulatory Element Prediction based on TIssue-specific Local
Epigenetic marks</em>, in preparation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("REPTILE")
data("rsd")

## Training (needs a few minutes and ~1.8 Gb memory)
reptile.model &lt;- reptile_train(rsd$training_data$region_epimark,
                               rsd$training_data$region_label,
                               rsd$training_data$DMR_epimark,
                               rsd$training_data$DMR_label,
                               ntree=50)

## Prediction
## - REPTILE
pred &lt;- reptile_predict(reptile.model,
                        rsd$test_data$region_epimark,
                        rsd$test_data$DMR_epimark)
## - Random guessing
pred_guess = runif(length(pred$D))
names(pred_guess) = names(pred$D)

## Evaluation
res_reptile &lt;- reptile_eval_prediction(pred$D,
                                       rsd$test_data$region_label)
res_guess &lt;- reptile_eval_prediction(pred_guess,
                                     rsd$test_data$region_label)
## - Print AUROC and AUPR
cat(paste0("REPTILE\n",
           "  AUROC = ",round(res_reptile$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_reptile$AUPR,digit=3))
    ,"\n")
cat(paste0("Random guessing\n",
           "  AUROC = ",round(res_guess$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_guess$AUPR,digit=3))
   ,"\n")
</code></pre>

<hr>
<h2 id='calculate_epimark_deviation'>
Internal - calculating intensity deviation feature
</h2><span id='topic+calculate_epimark_deviation'></span>

<h3>Description</h3>

<p>Internal function used to calculate the intensity deviation features.
It is based on the epigenomic signatures of a given region in target sample,
where prediction will be generated, and reference samples. Intensity
deviation is defined as the intensity in target sample subtracted
by the mean intensity in reference samples (i.e. reference
epigenome) and it captures the tissue-specificity of each epigenetic
mark.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_epimark_deviation(data_info, x, query_sample,
                            ref_sample = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_epimark_deviation_+3A_data_info">data_info</code></td>
<td>

<p>data.frame instance generated by reading data information file
specifying the samples and marks used in the analysis. The
data.frame includes at least two columns named &quot;sample&quot; and &quot;mark&quot;,
corresponding to the samples and marks included.
</p>
</td></tr>
<tr><td><code id="calculate_epimark_deviation_+3A_x">x</code></td>
<td>

<p>data.frame instance generated by reading epimark file. The first
four columns of the data.frame are &quot;chr&quot;, &quot;start&quot;, &quot;end&quot; and &quot;id&quot; of
each region in the epimark file. The rest columns contain values of
epigenetic marks in samples as specified in data_info and column
names are under MARK_SAMPLE format, such as &quot;H3K4me1_mESC&quot;.    
</p>
</td></tr>
<tr><td><code id="calculate_epimark_deviation_+3A_query_sample">query_sample</code></td>
<td>

<p>name of the target sample
</p>
</td></tr>
<tr><td><code id="calculate_epimark_deviation_+3A_ref_sample">ref_sample</code></td>
<td>

<p>a vector of names of the reference sample(s)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame instance containing intensity deviation values of each mark
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_epigenomic_data">read_epigenomic_data</a></code>
</p>

<hr>
<h2 id='get_option_parser_compute_score'>Internal - parsing options for REPTILE_compute_score.R
</h2><span id='topic+get_option_parser_compute_score'></span>

<h3>Description</h3>

<p>Internal function used to parsing options for
&quot;REPTILE_compute_score.R&quot; script in the REPTILE enhancer
prediction pipeline:
</p>
<p><a href="https://github.com/yupenghe/REPTILE">https://github.com/yupenghe/REPTILE</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_option_parser_compute_score()
</code></pre>


<h3>Value</h3>

<p>An instance of the OptionParser class.
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>

<hr>
<h2 id='get_option_parser_evaluation'>Internal - parsing options for REPTILE_evaluate_prediction.R
</h2><span id='topic+get_option_parser_evaluation'></span>

<h3>Description</h3>

<p>Internal function used to parsing options for
&quot;REPTILE_evaluate_prediction.R&quot; script in the REPTILE enhancer
prediction pipeline:
</p>
<p><a href="https://github.com/yupenghe/REPTILE">https://github.com/yupenghe/REPTILE</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_option_parser_evaluation()
</code></pre>


<h3>Value</h3>

<p>An instance of the OptionParser class.
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>

<hr>
<h2 id='get_option_parser_training'>
Internal - parsing options for REPTILE_train.R
</h2><span id='topic+get_option_parser_training'></span>

<h3>Description</h3>

<p>Internal function used to parsing options
for &quot;REPTILE_train.R&quot; script in the REPTILE enhancer
prediction pipeline:
</p>
<p><a href="https://github.com/yupenghe/REPTILE">https://github.com/yupenghe/REPTILE</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_option_parser_training()
</code></pre>


<h3>Value</h3>

<p>An instance of the OptionParser class.
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>

<hr>
<h2 id='read_epigenomic_data'>
Reading epigenomic data from epimark file
</h2><span id='topic+read_epigenomic_data'></span>

<h3>Description</h3>

<p>Function to read epimark file from disk and generate data.frame
instance. It is used to read epigenomic data from file on disk and
generate the input data.frame instance to fuel the model training,
prediction and other following steps.
Epimark file is a tab-separated file with a header. The first four
columns are &quot;chr&quot;, &quot;start&quot;, &quot;end&quot; and &quot;id&quot;, specifying the
chromosome, start, end and id of regions. Each of the remaining
columns contain values of one epigenetic mark in one sample
(condition, cell or tissue type, etc) and the column name follows
&quot;MARK_SAMPLE&quot; format, such as &quot;H3K4me1_mESC&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_epigenomic_data(data_info, epimark_file, query_sample,
                     ref_sample = NULL, incl_dev = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_epigenomic_data_+3A_data_info">data_info</code></td>
<td>

<p>data.frame generated by reading data information file specifying the
samples and marks used in the analysis. The data.frame includes    
at least two columns named &quot;sample&quot; and &quot;mark&quot;, corresponding to
the samples and marks included.
</p>
</td></tr>
<tr><td><code id="read_epigenomic_data_+3A_epimark_file">epimark_file</code></td>
<td>

<p>name of epimark file
</p>
</td></tr>
<tr><td><code id="read_epigenomic_data_+3A_query_sample">query_sample</code></td>
<td>

<p>name of the target sample
</p>
</td></tr>
<tr><td><code id="read_epigenomic_data_+3A_ref_sample">ref_sample</code></td>
<td>

<p>a vector of names of the reference sample(s)
</p>
</td></tr>
<tr><td><code id="read_epigenomic_data_+3A_incl_dev">incl_dev</code></td>
<td>

<p>logical value indicates whether to calculate the intensity deviation
feature. Intensity deviation is defined as the intensity in target
sample subtracted by the mean intensity in reference samples
(i.e. reference epigenome) and it captures the tissue-specificity of
each epigenetic mark.    
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame instance containing intensity and intensity deviation
values of each mark for each region
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_label">read_label</a></code>
</p>

<hr>
<h2 id='read_label'>
Reading labels of regions from label file
</h2><span id='topic+read_label'></span>

<h3>Description</h3>

<p>Function to read epimark file from disk and generate data.frame
instance. It is used to read epigenomic data from file on disk and
generate the input data.frame instance to fuel the model training,
prediction and other following steps. Label file is a tab-separated
file with a header. The first column contains the id of each
region. The second or more columns specify whether a certain region
is enhancer (1) or not (0) in a specific sample. Each of these columns
corresponds to one sample and the name of the column is the sample name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_label(label_file, query_sample)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_label_+3A_label_file">label_file</code></td>
<td>

<p>name of label file on disk
</p>
</td></tr>
<tr><td><code id="read_label_+3A_query_sample">query_sample</code></td>
<td>

<p>name(s) of sample(s), in which you would like to have label
information
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an data.frame instance containing label of each region in query
samples The possible values and their meanings of a label are:
</p>
<p><code>NA</code> - unknwon (will be ignored)
</p>
<p><code>0</code> - not enhancer
</p>
<p><code>1</code> - enhancer
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_epigenomic_data">read_epigenomic_data</a></code>
</p>

<hr>
<h2 id='reptile_eval_prediction'>
Evaluating the prediction results
</h2><span id='topic+reptile_eval_prediction'></span>

<h3>Description</h3>

<p>Function used to evaluate the predictions by comparing
enhancer scores from <code>reptile_predict</code> or
<code>reptile_predict_genome_wide</code> and the correct labels. Area under
the Receiver Operating Characteristic (ROC) curve (AUROC) and Area
under the Precision-Recall curve (AUPR) will be calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reptile_eval_prediction(predictions,annotations)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reptile_eval_prediction_+3A_predictions">predictions</code></td>
<td>

<p>vector of enhancer scores for regions. The name of each value
(score) corresponds to the id of the region.
</p>
</td></tr>
<tr><td><code id="reptile_eval_prediction_+3A_annotations">annotations</code></td>
<td>

<p>vector of labels for regions with the same length as predictions.
The name of each value (label) corresponds to the id of the
region. Only two values are allowed in <code>annotations</code>: 0
(negative) and 1 (positive). No <code>NA</code> is allowed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two numbers
</p>
<table role = "presentation">
<tr><td><code>AUROC</code></td>
<td>
<p>Area under the Receiver Operating Characteristic (ROC) curve</p>
</td></tr>
<tr><td><code>AUPR</code></td>
<td>
<p>Area under the Precision-Recall curve</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reptile_predict">reptile_predict</a></code>,
<code><a href="#topic+reptile_predict_genome_wide">reptile_predict_genome_wide</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("REPTILE")
data("rsd")

## Training
rsd_model &lt;- reptile_train(rsd$training_data$region_epimark,
                           rsd$training_data$region_label,
                           rsd$training_data$DMR_epimark,
                           rsd$training_data$DMR_label,
                           ntree=50)

## Prediction
## - REPTILE
pred &lt;- reptile_predict(rsd_model,
                        rsd$test_data$region_epimark,
                        rsd$test_data$DMR_epimark)
## - Random guessing
pred_guess = runif(length(pred$D))
names(pred_guess) = names(pred$D)

## Evaluation
res_reptile &lt;- reptile_eval_prediction(pred$D,
                                       rsd$test_data$region_label)
res_guess &lt;- reptile_eval_prediction(pred_guess,
                                     rsd$test_data$region_label)
## - Print AUROC and AUPR
cat(paste0("REPTILE\n",
           "  AUROC = ",round(res_reptile$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_reptile$AUPR,digit=3))
    ,"\n")
cat(paste0("Random guessing\n",
           "  AUROC = ",round(res_guess$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_guess$AUPR,digit=3))
   ,"\n")

</code></pre>

<hr>
<h2 id='reptile_predict'>
Predicting enhancer activity of given regions
</h2><span id='topic+reptile_predict'></span>

<h3>Description</h3>

<p>Predicting enhancer activities of query regions based on the enhancer
model from <code>reptile_train</code> in training step. This function
calculates the combined enhancer score for each query region (given
region) as the maximum among the score of whole query region and the
scores of DMRs within it. This function is for generating genome-wide
enhancer predictions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reptile_predict(reptile_model,
                epimark_region,
                epimark_DMR = NULL,
                family = "randomForest")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reptile_predict_+3A_reptile_model">reptile_model</code></td>
<td>

<p>Enhancer model from <code>reptile_train</code>. It is a list containing
two objects of class <code>randomForest</code> or <code>glm</code> when
<code>family</code> is set to be &quot;Logistic&quot;
</p>
</td></tr>
<tr><td><code id="reptile_predict_+3A_epimark_region">epimark_region</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each query
region
</p>
</td></tr>
<tr><td><code id="reptile_predict_+3A_epimark_dmr">epimark_DMR</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each DMR
</p>
</td></tr>
<tr><td><code id="reptile_predict_+3A_family">family</code></td>
<td>

<p>classifier family used in the enhancer model
</p>
<p>Default: RandomForest
</p>
<p>Classifiers available:
</p>
<p>- RandomForest: random forest
</p>
<p>- Logistic: logistic regression
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing three vectors
</p>
<table role = "presentation">
<tr><td><code>D</code></td>
<td>
<p>Combined enhancer score of each query region</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>Enhancer score of each query region</p>
</td></tr>
<tr><td><code>DMR</code></td>
<td>
<p>Enhancer score of each DMR</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reptile_predict_genome_wide">reptile_predict_genome_wide</a></code>
</p>
<p><code><a href="#topic+reptile_train">reptile_train</a></code>
</p>
<p><code><a href="#topic+read_epigenomic_data">read_epigenomic_data</a></code>
</p>
<p><code><a href="#topic+read_label">read_label</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("REPTILE")
data("rsd")

## Training
rsd_model &lt;- reptile_train(rsd$training_data$region_epimark,
                           rsd$training_data$region_label,
                           rsd$training_data$DMR_epimark,
                           rsd$training_data$DMR_label,
                           ntree=50)

## Prediction
## - REPTILE
pred &lt;- reptile_predict(rsd_model,
                        rsd$test_data$region_epimark,
                        rsd$test_data$DMR_epimark)
## - Random guessing
pred_guess = runif(length(pred$D))
names(pred_guess) = names(pred$D)

## Evaluation
res_reptile &lt;- reptile_eval_prediction(pred$D,
                                       rsd$test_data$region_label)
res_guess &lt;- reptile_eval_prediction(pred_guess,
                                     rsd$test_data$region_label)
## - Print AUROC and AUPR
cat(paste0("REPTILE\n",
           "  AUROC = ",round(res_reptile$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_reptile$AUPR,digit=3))
    ,"\n")
cat(paste0("Random guessing\n",
           "  AUROC = ",round(res_guess$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_guess$AUPR,digit=3))
   ,"\n")
</code></pre>

<hr>
<h2 id='reptile_predict_genome_wide'>
Predicting enhancer activity
</h2><span id='topic+reptile_predict_genome_wide'></span>

<h3>Description</h3>

<p>Predicting enhancer activities of query regions based on the enhancer
model from <code>reptile_train</code> in training step. This function
calculates the enhancer scores of DMRs and query regions. It does not
try to generate combined enhancer scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reptile_predict_genome_wide(reptile_model,
                            epimark_region,
                            epimark_DMR = NULL,
                            family = "randomForest")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reptile_predict_genome_wide_+3A_reptile_model">reptile_model</code></td>
<td>

<p>Enhancer model from <code>reptile_train</code>. It is a list containing
two objects of class <code>randomForest</code> or <code>glm</code> when
<code>family</code> is set to be &quot;Logistic&quot;
</p>
</td></tr>
<tr><td><code id="reptile_predict_genome_wide_+3A_epimark_region">epimark_region</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each query
region
</p>
</td></tr>
<tr><td><code id="reptile_predict_genome_wide_+3A_epimark_dmr">epimark_DMR</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each DMR
</p>
</td></tr>
<tr><td><code id="reptile_predict_genome_wide_+3A_family">family</code></td>
<td>

<p>classifier family used in the enhancer model
</p>
<p>Default: RandomForest
</p>
<p>Classifiers available:
</p>
<p>- RandomForest: random forest
</p>
<p>- Logistic: logistic regression
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two vectors
</p>
<table role = "presentation">
<tr><td><code>R</code></td>
<td>
<p>Enhancer score of each query region</p>
</td></tr>
<tr><td><code>DMR</code></td>
<td>
<p>Enhancer score of each DMR</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reptile_predict">reptile_predict</a></code>
</p>
<p><code><a href="#topic+reptile_train">reptile_train</a></code>
</p>
<p><code><a href="#topic+read_epigenomic_data">read_epigenomic_data</a></code>
</p>
<p><code><a href="#topic+read_label">read_label</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("REPTILE")
data("rsd")

## Training
rsd_model &lt;- reptile_train(rsd$training_data$region_epimark,
                           rsd$training_data$region_label,
                           rsd$training_data$DMR_epimark,
                           rsd$training_data$DMR_label,
                           ntree=50)

## Prediction
## - REPTILE
pred &lt;- reptile_predict(rsd_model,
                        rsd$test_data$region_epimark,
                        rsd$test_data$DMR_epimark)
## - Random guessing
pred_guess = runif(length(pred$D))
names(pred_guess) = names(pred$D)

## Evaluation
res_reptile &lt;- reptile_eval_prediction(pred$D,
                                       rsd$test_data$region_label)
res_guess &lt;- reptile_eval_prediction(pred_guess,
                                     rsd$test_data$region_label)
## - Print AUROC and AUPR
cat(paste0("REPTILE\n",
           "  AUROC = ",round(res_reptile$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_reptile$AUPR,digit=3))
    ,"\n")
cat(paste0("Random guessing\n",
           "  AUROC = ",round(res_guess$AUROC,digit=3),
           "\n",
           "  AUPR  = ",round(res_guess$AUPR,digit=3))
   ,"\n")
</code></pre>

<hr>
<h2 id='reptile_predict_one_mode'>
Internal - predicting enhancer activity of DMRs or query regions
</h2><span id='topic+reptile_predict_one_mode'></span>

<h3>Description</h3>

<p>Internal function used to predict the enhancer activity of either DMRs
or query regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reptile_predict_one_mode(reptile_classifier,
                         epimark,
                         family)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reptile_predict_one_mode_+3A_reptile_classifier">reptile_classifier</code></td>
<td>

<p>An object of class <code>randomForest</code> or <code>glm</code> when
<code>family</code> is set to be &quot;Logistic&quot;.
</p>
</td></tr>
<tr><td><code id="reptile_predict_one_mode_+3A_epimark">epimark</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each query
region    
</p>
</td></tr>
<tr><td><code id="reptile_predict_one_mode_+3A_family">family</code></td>
<td>

<p>classifier family used in the enhancer model
</p>
<p>Default: RandomForest
</p>
<p>Classifiers available:
</p>
<p>- RandomForest: random forest
</p>
<p>- Logistic: logistic regression
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vecotr of enhancer score of each query region or DMR
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reptile_predict">reptile_predict</a></code>,
<code><a href="#topic+reptile_predict_genome_wide">reptile_predict_genome_wide</a></code>
</p>

<hr>
<h2 id='reptile_train'>
Learn a REPTILE enhancer model
</h2><span id='topic+reptile_train'></span>

<h3>Description</h3>

<p>Learn a REPTILE enhancer model based on epigenomic signature of
known enhancers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reptile_train(epimark_region, label_region,
              epimark_DMR = NULL, label_DMR = NULL,
              family = "randomForest", ntree = 2000,
              nodesize = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reptile_train_+3A_epimark_region">epimark_region</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each query
region.
</p>
</td></tr>
<tr><td><code id="reptile_train_+3A_label_region">label_region</code></td>
<td>

<p>factor instance from read_label, containing the label of each
query region. The possible values and their meanings of a label
are: 0 (not enhancer), 1 (enhancer) and NA (unknwon and it will be
ignored).
</p>
</td></tr>
<tr><td><code id="reptile_train_+3A_epimark_dmr">epimark_DMR</code></td>
<td>

<p>data.frame instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each DMR.
If either this value or label_DMR is NULL, the output enhancer model
will not inlclude a classifier for predicting the enhancer
activities of DMRs.  
Default: NULL 
</p>
</td></tr>
<tr><td><code id="reptile_train_+3A_label_dmr">label_DMR</code></td>
<td>

<p>factor instance from read_label, containing the label of each
DMR. The possible values and their meanings of a label
are: 0 (not enhancer), 1 (enhancer) and NA (unknwon and it will be
ignored).
If either this value or label_DMR is NULL, the output
enhancer model will not inlclude a classifier for predicting the
enhancer activities of DMRs.
Default: NULL 
</p>
</td></tr>
<tr><td><code id="reptile_train_+3A_family">family</code></td>
<td>

<p>classifier family used in the enhancer model
</p>
<p>Default: RandomForest
</p>
<p>Classifiers available:
</p>
<p>- RandomForest: random forest
</p>
<p>- Logistic: logistic regression
</p>
</td></tr>
<tr><td><code id="reptile_train_+3A_ntree">ntree</code></td>
<td>

<p>Number of tree to be constructed in the random forest model. See
the function randomForest() in &quot;randomForest&quot; package for more
information. 
Default: 2000
</p>
</td></tr>
<tr><td><code id="reptile_train_+3A_nodesize">nodesize</code></td>
<td>

<p>Minimum size of terminal nodes. See the function randomForest()
in &quot;randomForest&quot; package for more information.
Default: 1
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two objects of class <code>randomForest</code>.
</p>
<table role = "presentation">
<tr><td><code>D</code></td>
<td>
<p>Classifier for DMRs. It is an <code>randomForest</code> object or
<code>glm</code> object when <code>family</code> is set to be &quot;Logistic&quot;.</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>Classifier for query regions. It is an <code>randomForest</code>
object or <code>glm</code> object when <code>family</code> is set to be
&quot;Logistic&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>References</h3>

<p>Breiman, L. (2001), <em>Random Forests</em>, Machine Learning 45(1),
5-32.
</p>
<p>A. Liaw and M. Wiener (2002), <em>Classification and Regression by
randomForest</em>, R News 2(3), 18&ndash;22.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_epigenomic_data">read_epigenomic_data</a></code>,
<code><a href="#topic+read_label">read_label</a></code>,
<code><a href="#topic+reptile_predict">reptile_predict</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("REPTILE")
data("rsd")

## Training
rsd_model &lt;- reptile_train(rsd$training_data$region_epimark,
                           rsd$training_data$region_label,
                           rsd$training_data$DMR_epimark,
                           rsd$training_data$DMR_label,
                           ntree=5)

print(rsd_model$D)
print(rsd_model$R)
</code></pre>

<hr>
<h2 id='reptile_train_one_mode'>
Internal - Learn single random forest classifier
</h2><span id='topic+reptile_train_one_mode'></span>

<h3>Description</h3>

<p>Internal function to learn a random forest classifier
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reptile_train_one_mode(epimark, label,
                       family, ntree, nodesize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reptile_train_one_mode_+3A_epimark">epimark</code></td>
<td>

<p><code>data.frame</code> instance from read_epigenomic_data, which containing
intensity and intensity deviation values of each mark for each DMR
or query region.    
</p>
</td></tr>
<tr><td><code id="reptile_train_one_mode_+3A_label">label</code></td>
<td>

<p><code>factor</code> instance from read_label, containing the label of each
query region. The possible values and their meanings of a label
are: <code>0</code> (not enhancer), <code>1</code> (enhancer) and <code>NA</code>
(unknwon and will be ignored).
</p>
</td></tr>
<tr><td><code id="reptile_train_one_mode_+3A_family">family</code></td>
<td>

<p>Classifier family used in the enhancer model
Default: RandomForest
Classifiers available:
- RandomForest: random forest
- Logistic: logistic regression
</p>
</td></tr>
<tr><td><code id="reptile_train_one_mode_+3A_ntree">ntree</code></td>
<td>

<p>Number of tree to be constructed in the random forest model. See the
function randomForest() in &quot;randomForest&quot; package for more
information.
Default: 2000    
</p>
</td></tr>
<tr><td><code id="reptile_train_one_mode_+3A_nodesize">nodesize</code></td>
<td>

<p>Minimum size of terminal nodes. See the function randomForest()
in &quot;randomForest&quot; package for more information.
Default: 1
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>randomForest</code> object or <code>glm</code> object when <code>family</code>
is set to be &quot;Logistic&quot;.
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a>
</p>


<h3>References</h3>

<p>Breiman, L. (2001), <em>Random Forests</em>, Machine Learning 45(1),
5-32.
</p>
<p>A. Liaw and M. Wiener (2002), <em>Classification and Regression by
randomForest</em>, R News 2(3), 18&ndash;22.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reptile_train">reptile_train</a></code>
</p>

<hr>
<h2 id='rsd'>REPTILE sample data (rsd)</h2><span id='topic+rsd'></span>

<h3>Description</h3>

<p>sample data for testing REPTILE training, prediction and evaluation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rsd)</code></pre>


<h3>Format</h3>

<p>A list containing two lists.
</p>
<p><code>training_data</code> is the data used for training REPTILE enhancer
model. This list has four elements: <code>region_epimark</code>,
<code>DMR_epimark</code>, <code>region_label</code> and <code>DMR_label</code>. The
former two store the epigenomic signatures of query regions and
DMRs. The latter two label which a certain query region or DMR is
enhancer (<code>1</code>) or negative instance (<code>0</code>)
</p>
<p><code>test_data</code> is for training REPTILE enhancer model and it
has four elements: <code>region_epimark</code>, <code>DMR_epimark</code> and
<code>region_label</code>. The former two store the
epigenomic signatures of query regions and DMRs. The
<code>region_label</code> indicates whether a certain query region or DMR is
enhancer (<code>1</code>) or negative instance (<code>0</code>)  
</p>


<h3>Author(s)</h3>

<p>Yupeng He <a href="mailto:yupeng.he.bioinfo@gmail.com">yupeng.he.bioinfo@gmail.com</a> </p>


<h3>Source</h3>

<p><code>training_data</code> was based on the EP300 binding sites (positives),
promoters (negatives) and randomly chosen genomic loci (negatives) in
mouse embryonic stem cells.
</p>
<p>The <code>test_data</code> data was constructed based on <em>in vivo</em>
validated mouse sequences from VISTA enhancer browser (Oct 24th,
2015). The labels indicate the activity in mouse heart tissues from
E11.5 embryo.
</p>
<p>See the papers included in References for details.
</p>


<h3>References</h3>

<p>He, Yupeng et al., <em>REPTILE: Regulatory Element Prediction based on TIssue-specific Local
Epigenetic marks</em>, in preparation
</p>
<p>Visel, Axel et al. (2007), <em>VISTA Enhancer Browser - a database of
tissue-specific human enhancers</em> Nucleic acids research 35. suppl 1
<a href="http://enhancer.lbl.gov/">http://enhancer.lbl.gov/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Visualizing rsd data
library("REPTILE")
data(rsd)

## Epigenomic signature of query region grouped by labels
ind_pos = rsd$training_data$region_label == 1
pos_region = rsd$training_data$region_epimark[ind_pos,]
neg_region = rsd$training_data$region_epimark[!ind_pos,]

## Epigenomic signature of DMRs grouped by labels
ind_pos = rsd$training_data$DMR_label == 1
pos_DMR = rsd$training_data$DMR_epimark[ind_pos,]
neg_DMR = rsd$training_data$DMR_epimark[!ind_pos,]

## Prepare the data format required for plotting
n = ncol(rsd$training_data$DMR_epimark) ## Number of features
feature_data_DMR = list()
feature_data_region = list()
for(i in 1:n){
    feature_data_DMR &lt;- append(feature_data_DMR,
                               list(neg_DMR[,i],pos_DMR[,i],
                                    NA,NA))
    feature_data_region &lt;- append(feature_data_region,
                                  list(neg_region[,i],pos_region[,i],
                                       NA,NA))
}

## Plot the feature distribution
par(mar=c(4,8,4,4))
## - query region
b &lt;- boxplot(feature_data_region,
             xlab = "feature value",
             notch=TRUE,outline=FALSE,yaxt='n',
             xlim = c(1,n*4-2),ylim=c(-7,7),
             horizontal=TRUE,
             col=c(rgb(65,105,225,max=255),rgb(250,128,114,max=255)),
             main = "Feature value distribution in query regions"
             )
text(par("usr")[1]-0.2, seq(1.5,n*4-2,by=4),
     labels=gsub("_","-",colnames(rsd$training_data$region_epimark)),
     xpd = TRUE,adj=1)
legend(-8,4*n+4,c("negative","enhancer"),ncol=2,
       fill = c(rgb(250,128,114,max=255),rgb(65,105,225,max=255)),
       xpd=TRUE,bty='n')

## - DMR
b &lt;- boxplot(feature_data_DMR,
             xlab = "feature value",
             notch=TRUE,outline=FALSE,yaxt='n',
             xlim = c(1,n*4-2),ylim=c(-7,7),
             horizontal=TRUE,
             col=c(rgb(65,105,225,max=255),rgb(250,128,114,max=255)),
             main = "Feature value distribution in DMRs"
             )
text(par("usr")[1]-0.2, seq(1.5,n*4-2,by=4),
     labels=gsub("_","-",colnames(rsd$training_data$DMR_epimark)),
     xpd = TRUE,adj=1)
legend(-8,4*n+4,c("negative","enhancer"),ncol=2,
       fill = c(rgb(250,128,114,max=255),rgb(65,105,225,max=255)),
       xpd=TRUE,bty='n')
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
