<!DOCTYPE html><html lang="en"><head><title>Help for package resevol</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {resevol}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#initialise_inds'><p>Initialise individuals</p></a></li>
<li><a href='#make_landscape'><p>Landscape initialisation</p></a></li>
<li><a href='#mg_n1'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_n2'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_n3'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_n4'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_n5'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_v1'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_v2'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_v3'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_v4'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mg_v5'><p>Sample mine_gmatrix outputs</p></a></li>
<li><a href='#mine_gmatrix'><p>Mine G-matrices</p></a></li>
<li><a href='#run_farm_sim'><p>Initialise individuals and simulate farming</p></a></li>
<li><a href='#stress_test'><p>Mine_gmatrix output stress</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulate Agricultural Production and Evolution of Pesticide
Resistance</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats(&ge; 4.0.0), utils (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>A. Bradley Duthie &lt;brad.duthie@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulates individual-based models of agricultural pest management
    and the evolution of pesticide resistance. Management occurs on a spatially
    explicit landscape that is divided into an arbitrary number of farms that
    can grow one of up to 10 crops and apply one of up to 10 pesticides. Pest
    genomes are modelled in a way that allows for any number of pest traits with
    an arbitrary covariance structure that is constructed using an evolutionary
    algorithm in the mine_gmatrix() function. Simulations are then run using the
    run_farm_sim() function. This package thereby allows for highly mechanistic
    social-ecological models of the evolution of pesticide resistance under
    different types of crop rotation and pesticide application regimes.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bradduthie.github.io/resevol/">https://bradduthie.github.io/resevol/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/bradduthie/resevol/issues">https://github.com/bradduthie/resevol/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, markdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-30 15:46:47 UTC; brad</td>
</tr>
<tr>
<td>Author:</td>
<td>A. Bradley Duthie <a href="https://orcid.org/0000-0001-8343-4995"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Rose McKeon [aut, ctr],
  Rosie Mangan [ctr],
  Luc Bussi&amp;eacute;re [ctr],
  Matthew Tinsley [ctr]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-30 16:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='initialise_inds'>Initialise individuals</h2><span id='topic+initialise_inds'></span>

<h3>Description</h3>

<p>Initialise new individuals into the IBM. This function is generally not needed
because it is run inside the run_farm_sim function to generate new individuals
for simulations. To initialise individuals with this function, it is necessary
to set the mine_output argument to output from the mine_gmatrix function. This
output includes all of the information necessary to build individuals with
genomes that produce traits that covary in a pre-specified way. The arguments
of this function include addition information for building the individual
array, which is a two-dimensional array in which each individual occupies a 
row, and each column specifies a character of the individual (including all
genome loci). See vignettes for a more detailed explanation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialise_inds(
  mine_output,
  N = 1000,
  xdim = 100,
  ydim = 100,
  repro = "sexual",
  neutral_loci = 10,
  max_age = 9,
  min_age_move = 0,
  max_age_move = 9,
  min_age_reproduce = 0,
  max_age_reproduce = 9,
  min_age_feed = 0,
  max_age_feed = 9,
  food_consume = 0.25,
  pesticide_consume = 0.1,
  rand_age = FALSE,
  move_distance = 1,
  food_needed_surv = 0.25,
  pesticide_tolerated_surv = 0.1,
  food_needed_repr = 0,
  pesticide_tolerated_repr = 0,
  reproduction_type = "lambda",
  mating_distance = 1,
  lambda_value = 1,
  movement_bouts = 1,
  selfing = TRUE,
  feed_while_moving = FALSE,
  pesticide_while_moving = FALSE,
  mortality_type = 0,
  age_food_threshold = NA,
  age_pesticide_threshold = NA,
  metabolism = 0,
  baseline_metabolism = 0,
  min_age_metabolism = 1,
  max_age_metabolism = 9,
  trait_means = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initialise_inds_+3A_mine_output">mine_output</code></td>
<td>
<p>The output from mine_gmatrix</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_n">N</code></td>
<td>
<p>Number of individuals to be initialised</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_xdim">xdim</code></td>
<td>
<p>Horizontal dimensions of the landscape</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_ydim">ydim</code></td>
<td>
<p>Vertical dimensions of the landscape</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_repro">repro</code></td>
<td>
<p>Type of reproduction allowed: &quot;asexual&quot;, &quot;sexual&quot;, and
&quot;biparental&quot;. Note that if repro != &quot;asexual&quot;, this causes a diploid genome.</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_neutral_loci">neutral_loci</code></td>
<td>
<p>The number of neutral loci individuals have (must be &gt; 0)</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_max_age">max_age</code></td>
<td>
<p>The maximum age of an individual</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_min_age_move">min_age_move</code></td>
<td>
<p>The minimum age at which an individual can move</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_max_age_move">max_age_move</code></td>
<td>
<p>The maximum age at which an individual can move</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_min_age_reproduce">min_age_reproduce</code></td>
<td>
<p>The minimum age which an individual can reproduce</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_max_age_reproduce">max_age_reproduce</code></td>
<td>
<p>The maximum age which an individual can reproduce</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_min_age_feed">min_age_feed</code></td>
<td>
<p>The minimum age at which an individual feeds</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_max_age_feed">max_age_feed</code></td>
<td>
<p>The maximum age at which an individual feeds</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_food_consume">food_consume</code></td>
<td>
<p>The amount of food consumed during feeding</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_pesticide_consume">pesticide_consume</code></td>
<td>
<p>Amount of pesticide consumed while on a cell</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_rand_age">rand_age</code></td>
<td>
<p>Initialise individuals with a random age (TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_move_distance">move_distance</code></td>
<td>
<p>Maximum cells moved in one bout of movement</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_food_needed_surv">food_needed_surv</code></td>
<td>
<p>Food needed to survive (if over min_age_feed)</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_pesticide_tolerated_surv">pesticide_tolerated_surv</code></td>
<td>
<p>Pesticide tolerated by individual</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_food_needed_repr">food_needed_repr</code></td>
<td>
<p>Food needed to reproduce 1 offspring</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_pesticide_tolerated_repr">pesticide_tolerated_repr</code></td>
<td>
<p>Pesticide tolerated to allow reproduction</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_reproduction_type">reproduction_type</code></td>
<td>
<p>Poisson reproduction (&quot;lambda&quot;) vs &quot;food_based&quot;</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_mating_distance">mating_distance</code></td>
<td>
<p>Distance in cells within which mate is available</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_lambda_value">lambda_value</code></td>
<td>
<p>individual value for poisson reproduction</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_movement_bouts">movement_bouts</code></td>
<td>
<p>Number of bouts of movement per time step</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_selfing">selfing</code></td>
<td>
<p>If sexual reproduction, is selfing allowed? (TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_feed_while_moving">feed_while_moving</code></td>
<td>
<p>Do individuals feed after each movement bout?</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_pesticide_while_moving">pesticide_while_moving</code></td>
<td>
<p>Individuals consume pesticide after move bout?</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_mortality_type">mortality_type</code></td>
<td>
<p>Type of mortality (currently only one option)</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_age_food_threshold">age_food_threshold</code></td>
<td>
<p>Age at which food threshold is enacted</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_age_pesticide_threshold">age_pesticide_threshold</code></td>
<td>
<p>Age at which pesticide threshold is enacted</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_metabolism">metabolism</code></td>
<td>
<p>The amount of consumed food lost each time step</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_baseline_metabolism">baseline_metabolism</code></td>
<td>
<p>A fixed baseline rate added to 'metabolism'+</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_min_age_metabolism">min_age_metabolism</code></td>
<td>
<p>The minimum age affected by metabolism</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_max_age_metabolism">max_age_metabolism</code></td>
<td>
<p>The maximum age affected by metabolism</p>
</td></tr>
<tr><td><code id="initialise_inds_+3A_trait_means">trait_means</code></td>
<td>
<p>The means of the custom pest traits</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-dimensional array of individuals for simulation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gmt       &lt;- matrix(data = 0, nrow = 2, ncol = 2);
diag(gmt) &lt;- 1;
mg        &lt;- mine_gmatrix(gmatrix = gmt, loci = 4, layers = 2, indivs = 100, 
                         npsize = 100, max_gen = 4, prnt_out = FALSE);
inds      &lt;- initialise_inds(mine_output = mg, N = 40, repro = "asexual");
</code></pre>

<hr>
<h2 id='make_landscape'>Landscape initialisation</h2><span id='topic+make_landscape'></span>

<h3>Description</h3>

<p>Initialise the landscape for a simulation. This should not normally need to 
be done explicitly with this function because it is run inside of the 
run_farm_sim function, but this gives the option to generate a landscape 
without actually running a simulation. All landscapes are produced as three
dimensional arrays with varying numbers of rows and columns that determine
landscape size, and a depth of 21 layers. The top layer defines which cells
belong to which farm, while the remaining layers define how much of a given
crop is on the landscape cell (2-11) or how much pesticide has been applied
to it (12-21). An arbitrary number of farms are placed in a blocked design 
on the landscape using a shortest split-line algorithm, which attempts to 
make farm size as even as possible. Specifying public land is possible, and
adds sections of land that are not farms, but this is not recommended.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_landscape(
  terrain = NA,
  rows = NA,
  cols = NA,
  depth = 21,
  farms = 4,
  public_land = 0,
  farm_var = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_landscape_+3A_terrain">terrain</code></td>
<td>
<p>a layer of terrain that can be added to specify farm position</p>
</td></tr>
<tr><td><code id="make_landscape_+3A_rows">rows</code></td>
<td>
<p>The dimension of the other side of the landscape (e.g., Longitude)</p>
</td></tr>
<tr><td><code id="make_landscape_+3A_cols">cols</code></td>
<td>
<p>The dimension of one side of the landscape (e.g., Latitude)</p>
</td></tr>
<tr><td><code id="make_landscape_+3A_depth">depth</code></td>
<td>
<p>The number of layers in the 3D landscape</p>
</td></tr>
<tr><td><code id="make_landscape_+3A_farms">farms</code></td>
<td>
<p>The number of farms on the landscape</p>
</td></tr>
<tr><td><code id="make_landscape_+3A_public_land">public_land</code></td>
<td>
<p>The proportion of landscape cells that are not farmland</p>
</td></tr>
<tr><td><code id="make_landscape_+3A_farm_var">farm_var</code></td>
<td>
<p>Does the land distribution vary among farms (&gt;=0, &lt;1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the_land A cols by rows landscape with randomly distributed cell types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>land &lt;- make_landscape(rows = 10, cols = 10, depth = 2, farms = 4)
</code></pre>

<hr>
<h2 id='mg_n1'>Sample mine_gmatrix outputs</h2><span id='topic+mg_n1'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure of the identity matrix (replicate 1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_n1
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_n2'>Sample mine_gmatrix outputs</h2><span id='topic+mg_n2'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure of the identity matrix (replicate 2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_n2
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_n3'>Sample mine_gmatrix outputs</h2><span id='topic+mg_n3'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure of the identity matrix (replicate 3)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_n3
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_n4'>Sample mine_gmatrix outputs</h2><span id='topic+mg_n4'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure  of the identity matrix (replicate 4)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_n4
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_n5'>Sample mine_gmatrix outputs</h2><span id='topic+mg_n5'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure of the identity matrix (replicate 5)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_n5
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_v1'>Sample mine_gmatrix outputs</h2><span id='topic+mg_v1'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure including negative and positive values 
(replicate 1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_v1
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_v2'>Sample mine_gmatrix outputs</h2><span id='topic+mg_v2'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure including negative and positive values 
(replicate 2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_v2
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_v3'>Sample mine_gmatrix outputs</h2><span id='topic+mg_v3'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure including negative and positive values 
(replicate 3)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_v3
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_v4'>Sample mine_gmatrix outputs</h2><span id='topic+mg_v4'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure including negative and positive values 
(replicate 4)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_v4
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mg_v5'>Sample mine_gmatrix outputs</h2><span id='topic+mg_v5'></span>

<h3>Description</h3>

<p>Output from mine_gmatrix that maps random normals from 12 loci to four traits
with a covariance structure including negative and positive values 
(replicate 5)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mg_v5
</code></pre>


<h3>Format</h3>

<p>A list of 8 elements used in individual-based model initialisation
</p>

<hr>
<h2 id='mine_gmatrix'>Mine G-matrices</h2><span id='topic+mine_gmatrix'></span>

<h3>Description</h3>

<p>Mine networks for establishing the link between genome and g-matrix. The 
output from this function is required to run individual-based simulations
in the rest of the package. The key input to this function, 'gmatrix', is a
(square) covariance matrix, with each row and column representing a trait for
the individual-based model. This function will run an evolutionary algorithm
to try to find a network that produces traits with the covariance structure
of gmatrix from a set of random standard normal values. The network from loci
values to trait values goes through a number of linked nodes to achieve this, 
and each generation tests the stress of the resulting network in terms of 
expected squared deviation of trait covariances from the input gmatrix. 
Simulations can take minutes to hours or longer, depending on parameters 
chosen and the number of traits. See vignettes for a more comprehensive
explanation for what this function is doing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mine_gmatrix(
  loci = 18,
  layers = 6,
  indivs = 1000,
  npsize = 2000,
  mu_pr = 0.05,
  mu_sd = 0.01,
  max_gen = 1000,
  pr_cross = 0.05,
  sampleK = 40,
  chooseK = 4,
  term_cri = -5.3,
  sd_ini = 0.1,
  use_cor = FALSE,
  prnt_out = TRUE,
  gmatrix
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mine_gmatrix_+3A_loci">loci</code></td>
<td>
<p>The number of loci for an individual. Simulations can allow for 
both haploid and diploid individuals. Allele values at each loci affect trait
values through a network of intermediary nodes.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_layers">layers</code></td>
<td>
<p>The number of hidden layers in the network linking loci to 
traits.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_indivs">indivs</code></td>
<td>
<p>The number of individuals initialised in each generation of the 
evolutionary algorithm to test among-individual trait correlations. 
Individuals are initialised with allele values drawn from a standard normal 
distribution.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_npsize">npsize</code></td>
<td>
<p>The size of the population of networks in each generation of the
evolutionary algorithm. Each network is a discrete individual in the 
population.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_mu_pr">mu_pr</code></td>
<td>
<p>The probability that a value in the network will mutate in a 
generation. Mutation events change the existing value by adding a new value 
drawn from a normal distribution with a mean of 0 and standard deviation of
mu_sd.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_mu_sd">mu_sd</code></td>
<td>
<p>The standard deviation of the random normal value mean centered 
at 0 that is added to the existing value of the network when a mutation event
occurs.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_max_gen">max_gen</code></td>
<td>
<p>The maximum number of generations that the evolutionary 
algorithm is allowed to run before terminating (regardless of how well the 
evolved covariance structure matches the pre-specified gmatrix).</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_pr_cross">pr_cross</code></td>
<td>
<p>The probability that a focal network in the population will 
initiate a crossover of a subset of its values with a randomly selected 
second network (note that any given network might therefore be included in 
more than one crossover event in a generation). The size of the subset is 
determined randomly.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_samplek">sampleK</code></td>
<td>
<p>During a round of selection, the number of random networks 
chosen to compete in a tournament. A single generation will include as many 
tournaments as necessary to create a new network population of size npsize.</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_choosek">chooseK</code></td>
<td>
<p>During a round of selection tournament, the number of networks 
within the sampleK random subset of the tournament that have the highest 
fitness will be selected to populate the next generation of networks</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_term_cri">term_cri</code></td>
<td>
<p>The criteria for terminating the evolutionary algorithm. The 
algorithm will terminate if a network is found in which the mean squared 
deviation of the covariance matrix elements from gmatrix is less than 
exp(term_crit).</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_sd_ini">sd_ini</code></td>
<td>
<p>The standard deviation of initialised network values at the 
start of the evolutionary algorithm. All network values are initialised by 
randomly sampling from a normal distribution with a mean of 0 and a 
standard deviation of sd_ini</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_use_cor">use_cor</code></td>
<td>
<p>Should the gmatrix be treated as a correlation matrix rather 
than a covariance matrix when calculating fitness?</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_prnt_out">prnt_out</code></td>
<td>
<p>Should the function print out progress showing the stress for 
each generation</p>
</td></tr>
<tr><td><code id="mine_gmatrix_+3A_gmatrix">gmatrix</code></td>
<td>
<p>The pre-specified trait covariance matrix. This will define
what the covariance will be between each trait when allele values are drawn 
from a standard normal distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of eight elements that includes the following: (1) A vector of
input parameters, (2) the pre-specified covariance matrix, (3) matrix defining
the effects of loci values on the first layer of the network, (4) a three
dimensional array link the first network layer to trait values, (5) a matrix
of the marginal effect of each locus on each trait, (6) the mined covariance
structure, (7) all network values to be inserted into individual genomes, and
(8) the log stress of the mined matrix against the pre-specified matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gmt       &lt;- matrix(data = 0, nrow = 4, ncol = 4);
diag(gmt) &lt;- 1;
mg        &lt;- mine_gmatrix(gmatrix = gmt, loci = 4, layers = 3, indivs = 100, 
                         npsize = 100, max_gen = 2, prnt_out = FALSE);
</code></pre>

<hr>
<h2 id='run_farm_sim'>Initialise individuals and simulate farming</h2><span id='topic+run_farm_sim'></span>

<h3>Description</h3>

<p>Initialises a new set of individuals and then simulates farming over time.
This is the main function that runs individual-based simulations of crop and
pesticide use and the evolution of pesticide resistance over time. To run 
this function, output from the mine_gmatrix function is required to specify
the covariance structure of individual traits and individual genomes. The
arguments to this function are used to initialise a landscape with the 
make_landscape function and initialise individuals with the initialise_inds
function. After initialisation, the simulation continues for up to a set 
number of time steps (unless extinction occurs), and individuals on the
landscape feed, encounter pesticide, move, reproduce, and die depending upon
the arguments specified in this function. After a specified number of time
steps, the crop or pesticide applied to a landscape cell can also change. The
end result is an evolving population of individuals that express traits 
that can potentially affect fitness (e.g., food consumption, pesticide
consumption, movement). Population level statistics are calculated by 
default and printed to a CSV, but individual level data (which includes all 
individual characteristics in a large table) need to be turned on because 
files can become extremely large (use print_inds with extreme caution and
print_last with care).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_farm_sim(
  mine_output,
  N = 1000,
  xdim = 100,
  ydim = 100,
  repro = "sexual",
  neutral_loci = 1000,
  max_age = 9,
  min_age_move = 0,
  max_age_move = 9,
  min_age_reproduce = 0,
  max_age_reproduce = 9,
  min_age_feed = 0,
  max_age_feed = 9,
  food_consume = 0.25,
  pesticide_consume = 0.1,
  rand_age = FALSE,
  move_distance = 1,
  food_needed_surv = 0.25,
  pesticide_tolerated_surv = 0.1,
  food_needed_repr = 0,
  pesticide_tolerated_repr = 0,
  reproduction_type = "lambda",
  mating_distance = 1,
  lambda_value = 1,
  movement_bouts = 1,
  selfing = TRUE,
  feed_while_moving = FALSE,
  pesticide_while_moving = FALSE,
  mortality_type = 0,
  age_food_threshold = 0,
  age_pesticide_threshold = 0,
  farms = 4,
  time_steps = 100,
  mutation_pr = 0,
  crossover_pr = 0,
  mutation_type = 0,
  net_mu_layers = 0,
  net_mu_dir = 0,
  mutation_direction = 0,
  crop_init = "random",
  crop_rotation_type = 2,
  crop_rotation_time = 1,
  pesticide_init = "random",
  pesticide_rotation_type = 2,
  pesticide_rotation_time = 1,
  crop_per_cell = 1,
  pesticide_per_cell = 1,
  crop_sd = 0,
  pesticide_sd = 0,
  crop_min = 0,
  crop_max = 1000,
  pesticide_min = 0,
  pesticide_max = 1000,
  crop_number = 2,
  pesticide_number = 1,
  print_inds = FALSE,
  print_gens = TRUE,
  print_last = FALSE,
  K_on_birth = 1e+06,
  pesticide_start = 0,
  immigration_rate = 0,
  get_f_coef = FALSE,
  get_stats = TRUE,
  metabolism = 0,
  baseline_metabolism = 0,
  min_age_metabolism = 1,
  max_age_metabolism = 9,
  terrain = NA,
  trait_means = NULL,
  land_edge = "torus",
  crop_growth = 0,
  crop_growth_type = "none"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_farm_sim_+3A_mine_output">mine_output</code></td>
<td>
<p>The output from mine_gmatrix, which will be used to
initialise the genomes and traits of pests.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_n">N</code></td>
<td>
<p>The number of individuals that are initialised in a simulation. 
Individuals are initialised in a random location on the landscape, and at 
least two individuals are needed.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_xdim">xdim</code></td>
<td>
<p>The number of cells in the horizontal dimension of the landscape. 
This value must be an integer greater than two.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_ydim">ydim</code></td>
<td>
<p>The number of cells in the vertical dimension of the landscape. 
This value must be an integer greater than two.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_repro">repro</code></td>
<td>
<p>The type of reproduction that individuals undergo in the 
simulation. There are three options: (1) &quot;asexual,&quot; in which individuals 
reproduce clonally and offspring have haploid genomes and traits identical to
their mother with the potential for mutation; (2) &quot;sexual,&quot; in which 
individuals are monoecious (both female and male) and offspring have diploid 
genomes with alleles inherited from both parents with mutation and 
recombination; (3) &quot;biparental,&quot; in which individuals are dioecious (only 
female or male) and offspring have diploid genomes with alleles inherited 
from both parents with mutation and recombination.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_neutral_loci">neutral_loci</code></td>
<td>
<p>The number of loci that are completely neutral (i.e., have
no effect on fitness). These loci can be used to monitor genetic drift or
calculate inbreeding coefficients.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_max_age">max_age</code></td>
<td>
<p>This is the maximum number of time steps that an individual can
survive. Individuals that are older than this age in a time step will always 
die.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_min_age_move">min_age_move</code></td>
<td>
<p>This is the minimum age at which an individual can move. 
Individuals below this age will always remain on their current cell.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_max_age_move">max_age_move</code></td>
<td>
<p>This is the maximum age at which an individual can move. 
Individuals above this age will always remain on their current cell.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_min_age_reproduce">min_age_reproduce</code></td>
<td>
<p>This is the minimum age at which an individual can be
reproductively active. No individuals below this age will engage in any 
reproductive activity, nor will they be recognised as potential mates by 
other individuals.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_max_age_reproduce">max_age_reproduce</code></td>
<td>
<p>This is the maximum age at which an individual can be
reproductively active. No individuals above this age will engage in any
reproductive activity, nor will they be recognised as potential mates by
other individuals.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_min_age_feed">min_age_feed</code></td>
<td>
<p>This is the minimum age at which an individual can eat. No
individuals below this age will be able to consume food on the landscape.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_max_age_feed">max_age_feed</code></td>
<td>
<p>This is the maximum age at which an individual can eat. No
individuals above this age will be able to consume food on the landscape.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_food_consume">food_consume</code></td>
<td>
<p>This defines how much food an individual will consume from
the cell on which it is feeding. Food consumption can take on any positive
real value, and an individual will consume up to this amount if possible (if 
not, they will consume however much food is left within their landscape 
cell).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_consume">pesticide_consume</code></td>
<td>
<p>This defines how much pesticide an individual will
consume from the cell on which it resides. Pesticide consumption can take on 
any positive real value, and an individual will consume up to this amount if
possible (if not, they will consume however much pesticide has been placed on
the landscape cell).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_rand_age">rand_age</code></td>
<td>
<p>This argument determines whether individuals in the simulation
will be initialised with a random age selected uniformly from zero to 
max_age. If FALSE, then all individuals will be initialised at age zero.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_move_distance">move_distance</code></td>
<td>
<p>This is the maximum number of cells that an individual 
can move, in any direction, on the landscape during one bout of movement.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_food_needed_surv">food_needed_surv</code></td>
<td>
<p>This is the amount of food that an individual needs to
consume to survive. If the individual has not consumed this amount of food 
before the age of age_food_threshold, then they will die in the time step.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_tolerated_surv">pesticide_tolerated_surv</code></td>
<td>
<p>This is the amount of pesticide that an 
individual can tolerate and still survive. If the individual has consumed 
more than this amount of pesticide on or after the age of 
age_pesticide_threshold, then they will die in the time step.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_food_needed_repr">food_needed_repr</code></td>
<td>
<p>This is the amount of food that an individual needs to
produce one offspring. The total number of offspring that an individual 
produces in a time step is the floor value of their food consumption divided 
by this value.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_tolerated_repr">pesticide_tolerated_repr</code></td>
<td>
<p>This is the amount of pesticide tolerated 
below which an individual can reproduce. Note that individuals above the 
threshold can still mate and sire offspring,</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_reproduction_type">reproduction_type</code></td>
<td>
<p>This determines how individuals reproduce; the two 
options are &quot;lambda&quot; and &quot;food_based.&quot; If &quot;lambda,&quot; then the number of 
offspring an individual produces is sampled from a Poisson distribution with 
a fixed rate parameter lambda_value (potentially adjusted by other factors in
the simulation). If &quot;food_based,&quot; then the number of offspring produced is 
based on the amount of food consumed by the individual.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_mating_distance">mating_distance</code></td>
<td>
<p>This is the distance in cells (any direction) away from
a focal individual from which they can successfully find and identify a mate 
(e.g., if 0, then only individuals on the same cell are potential mates).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_lambda_value">lambda_value</code></td>
<td>
<p>This is the rate parameter for the Poisson sampling of 
offspring number; it only applies when reproduction_type is set to &quot;lambda.&quot;</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_movement_bouts">movement_bouts</code></td>
<td>
<p>This is the number of times an individual can move in a 
single time step (i.e., the number of cells that it can potentially visit). 
Each time an individual visits a new cell, it can potentially feed or consume 
pesticide.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_selfing">selfing</code></td>
<td>
<p>This determines whether or not self-fertilisation is allowed 
when repro is set to &quot;sexual.&quot;</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_feed_while_moving">feed_while_moving</code></td>
<td>
<p>If TRUE, then individuals will feed in each movement 
bout when they arrive to a new landscape cell.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_while_moving">pesticide_while_moving</code></td>
<td>
<p>If TRUE, then individuals will consume pesticide
in each movement bout when they arrive to a new landscape cell.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_mortality_type">mortality_type</code></td>
<td>
<p>This determines how mortality is enacted in the 
simulation. Currently there is only one mortality type possible; mortality 
occurs if individuals exceed their maximum age, do not consume enough food, 
or consume too much pesticide.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_age_food_threshold">age_food_threshold</code></td>
<td>
<p>This is the age at which mortality associated with 
feeding is enacted, so an individual younger than this age will not die if 
they have not yet consumed sufficient food to satisfy food_needed_surv.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_age_pesticide_threshold">age_pesticide_threshold</code></td>
<td>
<p>This is the age at which mortality associated 
with pesticide consumption is enacted, so an individual younger than this age
will not die even if they have exceeded their pesticide threshold.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_farms">farms</code></td>
<td>
<p>This is the number of farms to be placed on the landscape. Farms 
are placed in blocks of roughly equal sizes using a shortest splitline 
algorithm. Farms operate independently in terms of what crops they grow and 
pesticides they apply.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_time_steps">time_steps</code></td>
<td>
<p>This is the number of time steps that a simulation will run.
Simulations will be terminated before this number if extinction occurs.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_mutation_pr">mutation_pr</code></td>
<td>
<p>This is the probability of mutation occurring at any locus 
of a newly produced offspring.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crossover_pr">crossover_pr</code></td>
<td>
<p>This is the probability of crossover between two 
homologous loci. This only applies for diploid genomes.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_mutation_type">mutation_type</code></td>
<td>
<p>This determines how mutation is modelled. If 0, then a 
completely new allele value is drawn from a normal distribution with a mean 
of mutation_direction and a standard deviation of 1 (or 1 / sqrt(2) for 
diploids, so that the expected standard devation of the sum of both allele 
values is 1). If 1, then a new value is drawn from a normal distribution with
mean mutation_direction and standard deviation of 1, and this new value is 
then added to the existing allele value.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_net_mu_layers">net_mu_layers</code></td>
<td>
<p>This is the proportion of the genome that can evolve. If 
0, then only loci values (green circles in Figure 1) can mutate. If 1, then 
loci and the first column of arrows (green circles to first column of blue 
squares in Figure 1) can mutate. If 2, then the first two columns of arrows 
in Figure 1 can mutate, and so forth. Fewer mutation layers will constrain 
the covariance among traits, while more mutation layers will allow the 
covariance structure to evolve more readily.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_net_mu_dir">net_mu_dir</code></td>
<td>
<p>The direction along the network in which net_mu_layers 
applies (not loci, green circles in Figure 1, can always mutate). If 1, then 
net_mu_layers applies in the direction from loci to traits. If 0, then the 
direction applies from traits to loci (i.e., net_mu_dir = 0 and 
net_mu_layers = 1 would mean that only the arrow values between the last 
hidden layer and traits in Figure 1 could mutate).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_mutation_direction">mutation_direction</code></td>
<td>
<p>This allows mutations to be biased in one direction.
A default value of 0 makes positive or negative allele values equally likely.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_init">crop_init</code></td>
<td>
<p>Initial crop type for each farm. This can be set in one of 
two ways. First, the default value &quot;random&quot; will randomly assign each farm to
an initial crop to produce. Second, a vector can be used to specify the crop
initialised on each farm. The vector must be the same length as the number of
farms, and the value of each element 'i' of the vector defines which crop is
initialised for each farm i. Hence, a crop_init vector must have as many
elements as there are farms, and vector elements must include natural numbers
from 1 to the total number of crops.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_rotation_type">crop_rotation_type</code></td>
<td>
<p>This determines how crop types are rotated across 
the landscape. This can be set in one of two ways. First, a natural number
can specify a rotation type: (1) crops will never rotate, (2) a new crop type
will be randomly chosen every crop_rotation_time time steps for each farm, 
or (3) farms will cycle through crop types in order, with a change from one
crop type to another every crop_rotation_time time step. Second, a square 
matrix can specify the probability of transition from a focal crop type 
(rows) to the next crop type (columns). Matrix rows must therefore sum to 1. 
For example, an identity matrix (1s in the diagonal and 0s in the 
off-diagonal) would specify crops that never rotate (i.e., crop i always 
rotates to itself).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_rotation_time">crop_rotation_time</code></td>
<td>
<p>This determines how many time steps a crop is left 
before being refreshed and potentially changed. Note that even if the crop 
type does not change, this value still has the effect of determining how 
often crops are replenished (if some have been eaten since the last time they
were replenished).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_init">pesticide_init</code></td>
<td>
<p>Initial pesticide type for each farm. This can be set in
one of two ways. First, the default value &quot;random&quot; will randomly assign each 
farm to an initial pesticide to apply. Second, a vector can be used to 
specify the pesticide initialised on each farm. The vector must be the same 
length as the number of farms, and the value of each element 'i' of the 
vector defines which pesticide is initialised for each farm i. Hence, a 
pesticide_init vector must have as many elements as there are farms, and 
vector elements must include natural numbers from 1 to the total number of 
pesticides.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_rotation_type">pesticide_rotation_type</code></td>
<td>
<p>This determines how pesticide types are rotated
across the landscape. This can be set in one of two ways. First, a natural 
number can specify a rotation type: (1) pesticides will never rotate, (2) a 
new pesticide type will be randomly chosen every pesticide_rotation_time time
steps for each farm, or (3) farms will cycle through pesticide types in 
order, with a change from one pesticide type to another every 
pesticide_rotation_time time step. Second, a square matrix can specify the 
probability of transition from a focal pesticide type (rows) to the next 
pesticide type (columns). Matrix rows must therefore sum to 1. For example, 
an identity matrix (1s in the diagonal and 0s in the off-diagonal) would 
specify pesticides that never rotate (i.e., pesticide i always rotates to 
itself).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_rotation_time">pesticide_rotation_time</code></td>
<td>
<p>This determines how many time steps a pesticide
is left before being replenished and potentially changed. Note that unlike 
crops, pesticide levels do not decrease on the landscape over time (e.g., 
with consumption).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_per_cell">crop_per_cell</code></td>
<td>
<p>This determines the expected amount of crop that is 
placed on a single landscape cell. The more crop on a cell, the more that can
be potentially consumed by individuals.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_per_cell">pesticide_per_cell</code></td>
<td>
<p>This determines how much pesticide is placed on a 
single landscape cell. The higher concentration of pesticide per cell, the 
more that individuals on the cell will imbibe and potentially be affected by.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_sd">crop_sd</code></td>
<td>
<p>This is the standard deviation of crop number placed on 
landscape cells. A default value of 0 assumes that all cells have the same 
amount of crop.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_sd">pesticide_sd</code></td>
<td>
<p>This is the standard deviation of pesticide applied to 
each landscape cell. A default value of 0 assumes that each cell has the same
concentration of pesticide applied.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_min">crop_min</code></td>
<td>
<p>This is the minimum amount of crop that is possible to have on
a single cell (i.e., crop values will never be initialised to be lower than 
this value).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_max">crop_max</code></td>
<td>
<p>This is the maximum amount of crop that is possible to have on
a single cell (i.e., crop values will never be initialised to be higher than 
this value).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_min">pesticide_min</code></td>
<td>
<p>This is the minimum concentration of pesticide that is 
possible to have on a single cell (i.e., pesticide values will never be 
initialised to be lower than this value).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_max">pesticide_max</code></td>
<td>
<p>This is the maximum concentration of pesticide that is 
possible to have on a single cell (i.e., pesticide values will never be 
initialised to be higher than this value).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_number">crop_number</code></td>
<td>
<p>This is the number of unique crops that can exist on the 
landscape during the course of a simulation. The maximum number of possible 
crops is 10.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_number">pesticide_number</code></td>
<td>
<p>This is the number of unique pesticides that can exist
on the landscape during the course of a simulation. The maximum number of 
possible pesticides is 10.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_print_inds">print_inds</code></td>
<td>
<p>If TRUE, a CSV file will print in the working directory with
every individual and all of their characteristics (i.e., locations, traits, 
genomes) in every time step. By default, this is set to FALSE and should only
be set to TRUE with extreme caution, as large populations persisting over 
long periods of time can produce extremely large CSV files.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_print_gens">print_gens</code></td>
<td>
<p>If TRUE, the time step and the population size will be 
printed to the R console as the simulation is running.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_print_last">print_last</code></td>
<td>
<p>If TRUE, a CSV file will print in the working directory with
every individual and all of their characteristics (i.e., locations, traits, 
genomes) in only the last time step. Note that for large populations, the 
file size generated can be very large (10s to 100s of GBs).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_k_on_birth">K_on_birth</code></td>
<td>
<p>This is a carrying capacity applied to new individuals 
across the entire landscape. If the total number of offspring in a time step 
exceeds this value, then offspring are removed at random until the total 
number of new offspring equals K_on_birth. In practice, this can help speed 
up simulations by avoiding the unnecessary production of individuals when 
most will perish.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_pesticide_start">pesticide_start</code></td>
<td>
<p>This is the time step at which pesticide begins to be 
applied. No pesticide will be applied prior to this start time, so 
individuals will not experience any effects of pesticide. This can be useful 
as a tool to burn in the population prior to introducing pesticide.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_immigration_rate">immigration_rate</code></td>
<td>
<p>This is the number of immigrant individuals arriving 
in the landscape in each time step. Immigrants are initialised in random 
locations with the same network structure (Figure 1) as individuals 
initialised at the start of the simulation, and with allele values randomly 
drawn from a standard normal distribution.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_get_f_coef">get_f_coef</code></td>
<td>
<p>This determines whether or not inbreeding coefficients will 
be calculated for sexual populations and printed off in CSV files. Because 
this can add some computation time, it is best to set to FALSE unless it is 
needed.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_get_stats">get_stats</code></td>
<td>
<p>If TRUE, a CSV file will print in the working directory with 
summary statistics for each time step. This is set to TRUE by default.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_metabolism">metabolism</code></td>
<td>
<p>This determines the rate at which food consumed in previous 
time steps is lost in subsequent time steps, which can be especially relevant
if food consumed determines survival or reproductive output. Values of 0 mean
that stored gains will always persist throughout an individuals lifetime, 
while very high values will model the gains of one time step being wiped out 
in subsequent time steps (if, e.g., the objective is to model individuals 
needing to consume food successfully in each time step to survive or 
reproduce, as opposed to having a feeding life history stage followed by a 
mating and reproduction stage).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_baseline_metabolism">baseline_metabolism</code></td>
<td>
<p>This fixes a baseline metabolic rate at which food 
consumed in previous time steps is lost in subsequent steps. This fixed value
is always added to metabolism for each individual. By default, this value is 
0.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_min_age_metabolism">min_age_metabolism</code></td>
<td>
<p>This determines the minimum age at which losses of 
food consumed in previous time steps enacted by metabolism and 
baseline_metabolism can occur.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_max_age_metabolism">max_age_metabolism</code></td>
<td>
<p>This determines the maximum age at which losses of 
food consumed in previous time steps enacted by metabolism and 
baseline_metabolism can occur.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_terrain">terrain</code></td>
<td>
<p>Insert a custom terrain of different farms, which takes the
form of a matrix that includes a sequence of natural numbers in all matrix
elements. For example, if there are 4 farms, then all matrix elements must
be 1, 2, 3, or 4. Beyond this requirement, there is no restriction on where 
different farms are placed; the do not even need to be contiguous on the 
landscape. Note that a custom terrain will override the arguments farms, 
xdim, and ydim. For example, if the matrix given to the terrain argument has
10 rows and 10 columns, then the simulation will automatically set xdim and
ydim equal to 10 without any warnings. Also note that these terrain values do
not necessarily need to be farms. Through the use of a custom landscape and 
pesticide rotation option, these cells could represent something like 
diversionary feeding sites or even buildings or rivers. See vignettes and 
other documentation for details.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_trait_means">trait_means</code></td>
<td>
<p>This provides the mean values of the evolving pest traits
in the initialised pest population, which by default are zero. To change 
mean trait values in the initialised population, the trait_means argument
requires a vector of the same length as the number of evolving pest traits
defined within the function (i.e., if there are two evolving traits in the
simulation, &quot;T1&quot; and &quot;T2&quot;, then trait_means should be a vector of length 2).
Note that mean trait values may change as the population evolves, so the
values in this vector define only the means of the initial population.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_land_edge">land_edge</code></td>
<td>
<p>This sets what happens at the edge of the landscape. Three 
options are possible, including &quot;torus&quot;, &quot;leaky&quot;, and &quot;reflect&quot;. A torus
results in no edge, such that pests that leave one side of the landscape
end up on the other side. A leaky edge causes pests to leave the landscape
and the simulation entirely, so they are effectively no longer recorded. A 
reflective edge causes pests to bounce at the edge back in the direction from
which they came. The default and recommended edge is a torus.</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_growth">crop_growth</code></td>
<td>
<p>This sets the amount by which landscape cell values
increase for each crop from one time step to the next. This might be used to
model the increase in biomass caused by plant growth on the landscape cell.
The way in which growth is enacted is set by crop_growth_type. Values for
crop_growth can either be a single number (in which case, the growth amount
applies to all crop types) or a vector of the same length as crop_types (in
which case, vector indices give the amount of growth for each crop number).</p>
</td></tr>
<tr><td><code id="run_farm_sim_+3A_crop_growth_type">crop_growth_type</code></td>
<td>
<p>This clarifies whether crop growth should be 
&quot;none&quot; (default), &quot;geometric&quot;, or &quot;linear&quot;. If geometric, then the value of a
landscape cell increases to cell_value * (1 + crop_growth). If linear, then
a fixed value crop_growth is added to the existing cell value. Note that for
linear growth, this fixed value is only added if the existing cell value is
greater than zero (else it is assumed that there is no crop to grow).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output in the R console is a list with two elements; the first 
element is a vector of parameter values used by the model, and the second 
element is the landscape in the simulation. The most relevant output will be
produced as CSV files within the working directory. When get_stats = TRUE, 
a file named 'population_data.csv' is produced in the working directory. When
print_last = TRUE, a complete array of all individuals and their 
characteristics is printed for the last time step in the working directory in
a file named 'last_time_step.csv' (for large simulations, this file can be 
&gt; 1GB in size). When print_inds = TRUE, a complete array of all individuals 
in all time steps is produced in the working directory in a file named
'individuals.csv' (use this option with extreme caution for all but the
smallest simulations).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gmt       &lt;- matrix(data = 0, nrow = 4, ncol = 4);
diag(gmt) &lt;- 1;
mg        &lt;- mine_gmatrix(gmatrix = gmt, loci = 4, layers = 3, indivs = 100, 
                         npsize = 100, max_gen = 2, prnt_out = FALSE);
sim       &lt;- run_farm_sim(mine_output = mg, N = 100, xdim = 40, ydim = 40, 
                         repro = "asexual", time_steps = 1, 
                         print_inds = FALSE, print_gens = FALSE,
                         print_last = FALSE, get_stats = FALSE);
</code></pre>

<hr>
<h2 id='stress_test'>Mine_gmatrix output stress</h2><span id='topic+stress_test'></span>

<h3>Description</h3>

<p>Run a diagnoistic test on the stress output of mine_gmatrix. The mine_gmatrix
function produces a network from loci to traits for a pre-specified trait
covariance structure. This covariance structure is estimated from network
values, but can vary due to error in the loci values (standard random normal
numbers). This function will test the stress of a network (mean deviation 
between the estimated covariance matrix and the pre-specified one) a given
number of times to produce a distribution of stress estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stress_test(mine_output, indivs = 1000, reps = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stress_test_+3A_mine_output">mine_output</code></td>
<td>
<p>The output from mine_gmatrix, which will be used to
initialise the genomes and traits of pests.</p>
</td></tr>
<tr><td><code id="stress_test_+3A_indivs">indivs</code></td>
<td>
<p>The number of individuals to use in the stress test. Higher
values produce a larger sample size for more accurate estimates of the true
covariance structure produced by the network, and therefore the actual stress
expected from it.</p>
</td></tr>
<tr><td><code id="stress_test_+3A_reps">reps</code></td>
<td>
<p>The number of times a new set of individuals will be initialised
for estimating the covariance between traits and calculating its stress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of stress values the same length as 'reps'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gmt       &lt;- matrix(data = 0, nrow = 4, ncol = 4);
diag(gmt) &lt;- 1;
mg        &lt;- mine_gmatrix(gmatrix = gmt, loci = 4, layers = 3, indivs = 100, 
                         npsize = 100, max_gen = 2, prnt_out = FALSE);
stresses  &lt;- stress_test(mine_output = mg);
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
