<!DOCTYPE html><html><head><title>Help for package reutils</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {reutils}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#reutils-package'><p>Talk to the NCBI EUtils</p></a></li>
<li><a href='#[,efetch,character,missing-method'><p>EFetch accessors</p></a></li>
<li><a href='#[,einfo,ANY,missing-method'><p>EInfo accessors</p></a></li>
<li><a href='#[,elink,ANY,missing-method'><p>ELink Accessors</p></a></li>
<li><a href='#[,esearch,numeric,missing-method'><p>ESearch Accessors</p></a></li>
<li><a href='#[,esummary,character,ANY-method'><p>ESummary accessors</p></a></li>
<li><a href='#content'><p>Extract the data content from an Entrez request</p></a></li>
<li><a href='#database'><p>database</p></a></li>
<li><a href='#ecitmatch'><p>ecitmatch - searching PubMed with citation data</p></a></li>
<li><a href='#efetch'><p>efetch - downloading full records</p></a></li>
<li><a href='#egquery'><p>egquery - performing a global Entrez search</p></a></li>
<li><a href='#einfo'><p>einfo - getting database statistics and search fields</p></a></li>
<li><a href='#elink'><p>elink - finding related data through Entrez links</p></a></li>
<li><a href='#entrez_linkset-class'><p>Class <code>"entrez_linkset"</code></p></a></li>
<li><a href='#entrez_uid-class'><p>Class <code>"entrez_uid"</code></p></a></li>
<li><a href='#epost'><p>epost - uploading UIDs to Entrez</p></a></li>
<li><a href='#esearch'><p>esearch - searching an Entrez database</p></a></li>
<li><a href='#espell'><p>espell - retrieving spelling suggestions</p></a></li>
<li><a href='#esummary'><p>esummary - downloading Document Summaries</p></a></li>
<li><a href='#eutil_error-class'><p>Class <code>"eutil_error"</code>:</p></a></li>
<li><a href='#eutil-class'><p>Class <code>"eutil"</code>: Reference classes that hold the response from EUtils</p>
requests.</a></li>
<li><a href='#getError'><p>getError</p></a></li>
<li><a href='#getUrl'><p>getUrl</p></a></li>
<li><a href='#linkset'><p>linkset</p></a></li>
<li><a href='#make_flattener'><p>Flatten nested lists</p></a></li>
<li><a href='#ncbi_retrieval_type'><p>Set the NCBI rettype</p></a></li>
<li><a href='#performQuery'><p>performQuery</p></a></li>
<li><a href='#querykey'><p>querykey</p></a></li>
<li><a href='#retmode'><p>retmode</p></a></li>
<li><a href='#rettype'><p>rettype</p></a></li>
<li><a href='#uid'><p>uid</p></a></li>
<li><a href='#webenv'><p>webenv</p></a></li>
<li><a href='#xattr'><p>Extract the attributes of XML nodes</p></a></li>
<li><a href='#xname'><p>Extract the tag name of XML nodes</p></a></li>
<li><a href='#xset'><p>Extract a node set from an XML document</p></a></li>
<li><a href='#xvalue'><p>Extract the content of XML leaf nodes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Talk to the NCBI EUtils</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-09-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Gerhard Schöfl &lt;gerhard.schofl@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gerhard Schöfl &lt;gerhard.schofl@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An interface to NCBI databases such as PubMed, GenBank, or GEO
    powered by the Entrez Programming Utilities (EUtils). The nine EUtils
    provide programmatic access to the NCBI Entrez query and database
    system for searching and retrieving biological data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, jsonlite, methods, RCurl, stats, XML, tibble (&ge;
1.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 0.8)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/gschofl/reutils">https://github.com/gschofl/reutils</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/gschofl/reutils/issues">https://github.com/gschofl/reutils/issues</a></td>
</tr>
<tr>
<td>Collate:</td>
<td>'utils.R' 'eutil-error.R' 'eutil.R' 'ecitmatch.R'
'parse-params.R' 'efetch.R' 'egquery.R' 'einfo.R' 'elink.R'
'epost.R' 'esearch.R' 'espell.R' 'esummary.R' 'parse-docsum.R'
'reutils-package.R' 'zzz.R'</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-03 10:32:25 UTC; gerhard</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-03 16:39:17</td>
</tr>
</table>
<hr>
<h2 id='reutils-package'>Talk to the NCBI EUtils</h2><span id='topic+reutils'></span><span id='topic+reutils-package'></span>

<h3>Description</h3>

<p>An interface to NCBI databases such as PubMed, GenBank, or GEO
powered by the Entrez Programming Utilities (EUtils). The nine EUtils
provide programmatic access to the NCBI Entrez query and database
system for searching and retrieving biological data.
</p>


<h3>Details</h3>

<p>With nine <a href="https://www.ncbi.nlm.nih.gov/books/NBK25501/">Entrez Progamming Utilities</a>,
NCBI provides a programmatical interface to the Entrez query and database
system for searching and retrieving requested data
</p>
<p>Each of these tools corresponds to an <code>R</code> function in the <code>reutils</code>
package described below.
</p>
<p>The output returned by the EUtils is typically in XML format. To gain access to
this output you have several options:
</p>

<ol>
<li><p> Use the <code>content(as = "xml")</code> method to extract the output as an
<code>XMLInternalDocument</code> object and process it further using the
facilities provided by the <code>XML</code> package.
</p>
</li>
<li><p> Use the <code>content(as = "parsed")</code> method to extract the output
into <code>data.frame</code>s. Note that this is currently only implemented for
<em>docsum</em>s returned by <code><a href="#topic+esummary">esummary</a></code>, <em>uilist</em>s returned
by <code><a href="#topic+esearch">esearch</a></code>, and the output returned by <code><a href="#topic+einfo">einfo</a></code>.
</p>
</li>
<li><p> Access specific nodes in the XML tree using XPath expressions with the
reference class methods <code>#xmlValue</code>, <code>#xmlAttr</code>, or <code>#xmlName</code>
built into <code><a href="#topic+eutil-class">eutil</a></code> objects.
</p>
</li></ol>

<p>The Entrez Programming Utilities can also generate output in other formats,
such as plain-text Fasta or GenBank files for sequence databases,
or the MedLine format for the literature database. The type of output is
generally controlled by setting the <code>retmode</code> and <code>rettype</code> arguments
when calling a EUtil.
Please check the relevant
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25497/#chapter2.Usage_Guidelines_and_Requiremen">usage guidelines</a>
when using these services. Note that Entrez server requests are subject to
frequency limits.
</p>


<h3>Main functions</h3>


<ul>
<li> <p><code><a href="#topic+esearch">esearch</a></code>: Search and retrieve primary UIDs for use
with <code>esummary</code>, <code>elink</code>, or <code>efetch</code>.
<code>esearch</code> additionally returns term translations and optionally
stores results for future use in the user's Web Environment.
</p>
</li>
<li> <p><code><a href="#topic+esummary">esummary</a></code>: Retrieve document summaries from
a list of primary UIDs (Provided as a character vector or as an
<code>esearch</code> object).
</p>
</li>
<li> <p><code><a href="#topic+egquery">egquery</a></code>: Provides Entrez database counts in XML
for a single search term using a Global Query.
</p>
</li>
<li> <p><code><a href="#topic+einfo">einfo</a></code>: Retrieve field names, term counts, last
update, and available updates for each database.
</p>
</li>
<li> <p><code><a href="#topic+efetch">efetch</a></code>: Retrieve data records in a specified
format corresponding to a list of primary UIDs or from the user's Web
Environment in the Entrez History server.
</p>
</li>
<li> <p><code><a href="#topic+elink">elink</a></code>: Returns a list of UIDs (and relevancy
scores) from a target database that are related to a list of UIDs in
the same database or in another Entrez database.
</p>
</li>
<li> <p><code><a href="#topic+epost">epost</a></code>: Uploads primary UIDs to the users's Web 
Environment on the Entrez history server for subsequent use with
<code>esummary</code>, <code>elink</code>, or <code>efetch</code>.
</p>
</li>
<li> <p><code><a href="#topic+espell">espell</a></code>: Provide spelling suggestions.
</p>
</li>
<li> <p><code><a href="#topic+ecitmatch">ecitmatch</a></code>: Retrieves PubMed IDs (PMIDs) that
correspond to a set of input citation strings
</p>
</li>
<li> <p><code><a href="#topic+content">content</a></code>: Extract the content of a request from the
<code><a href="#topic+eutil-class">eutil</a></code> object returned by any of the above functions.
</p>
</li></ul>



<h3>Package options</h3>

<p><span class="pkg">reutils</span> uses three <code><a href="base.html#topic+options">options</a></code> to configure behaviour:
</p>

<ul>
<li> <p><code>reutils.email</code>: NCBI requires that a user of their API provides an
email address with a call to Entrez. If you are going to perform a lot
of queries consider setting <code>reutils.email</code> to your email address in
your .Rprofile file.
</p>
</li>
<li> <p><code>reutils.show.headlines</code>: By default <code><a href="#topic+efetch-class">efetch</a></code>
objects containing text data show only the first 12 lines. This is quite handy
if you have downloaded a fairly large genome in Genbank file format. This
can be changed by setting the global option <code>reutils.show.headlines</code> to
another numeric value or <code>NULL</code>.
</p>
</li>
<li> <p><code>reutils.verbose.queries</code>: If you perform many queries interactively
you might want to get messages announcing the queries you run. You can do so by setting
the option <code>reutils.verbose.queries</code> to <code>TRUE</code>.
</p>
</li>
<li> <p><code>reutils.test.remote</code>: Unit tests that require online access to NCBI
services are disabled by default, as they cannot be garanteed to be
available/working under all circumstances. Set the option
codereutils.test.remote to <code>TRUE</code> to run the full suite of tests.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Gerhard Schöfl <a href="mailto:gerhard.schofl@gmail.com">gerhard.schofl@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# combine esearch and efetch
#
# Download PubMed records that are indexed in MeSH for both 'Chlamydia' and 
# 'genome' and were published in 2013.
query &lt;- "Chlamydia[mesh] and genome[mesh] and 2013[pdat]"

# Upload the PMIDs for this search to the History server
pmids &lt;- esearch(query, "pubmed", usehistory = TRUE)
pmids

## Not run: 
# Fetch the records
articles &lt;- efetch(pmids)

# Use XPath expressions with the #xmlValue() or #xmlAttr() methods to directly
# extract specific data from the XML records stored in the 'efetch' object.
titles &lt;- articles$xmlValue("//ArticleTitle")
abstracts &lt;- articles$xmlValue("//AbstractText")

#
# combine epost with esummary/efetch
#
# Download protein records corresponding to a list of GI numbers.
uid &lt;- c("194680922", "50978626", "28558982", "9507199", "6678417")

# post the GI numbers to the Entrez history server
p &lt;- epost(uid, "protein")

# retrieve docsums with esummary
docsum &lt;- content(esummary(p, version = "1.0"), "parsed")
docsum

# download FASTAs as 'text' with efetch
prot &lt;- efetch(p, retmode = "text", rettype = "fasta")
prot

# retrieve the content from the efetch object
fasta &lt;- content(prot)

## End(Not run)
</code></pre>

<hr>
<h2 id='+5B+2Cefetch+2Ccharacter+2Cmissing-method'>EFetch accessors</h2><span id='topic++5B+2Cefetch+2Ccharacter+2Cmissing-method'></span><span id='topic++5B+5B+2Cefetch+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Extract XML nodes from an <code><a href="#topic+efetch-class">efetch</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'efetch,character,missing'
x[i, j]

## S4 method for signature 'efetch,character'
x[[i]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2Cefetch+2B2Ccharacter+2B2Cmissing-method_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+efetch-class">efetch</a></code> object containing XML data.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cefetch+2B2Ccharacter+2B2Cmissing-method_+3A_i">i</code></td>
<td>
<p>An XPath expression specifying the XML nodes to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cefetch+2B2Ccharacter+2B2Cmissing-method_+3A_j">j</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An XML node set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- efetch("195055", "protein", "gp", "xml")
p['//GBFeature[GBFeature_key="mat_peptide"]//GBQualifier_value']

## End(Not run)
</code></pre>

<hr>
<h2 id='+5B+2Ceinfo+2CANY+2Cmissing-method'>EInfo accessors</h2><span id='topic++5B+2Ceinfo+2CANY+2Cmissing-method'></span><span id='topic++5B+5B+2Ceinfo+2CANY-method'></span>

<h3>Description</h3>

<p>Extract parts of a parsed <code><a href="#topic+einfo-class">einfo</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'einfo,ANY,missing'
x[i, j]

## S4 method for signature 'einfo,ANY'
x[[i]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2Ceinfo+2B2CANY+2B2Cmissing-method_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+einfo-class">einfo</a></code> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Ceinfo+2B2CANY+2B2Cmissing-method_+3A_i">i</code></td>
<td>
<p>Numeric or character indices specifying the elements to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Ceinfo+2B2CANY+2B2Cmissing-method_+3A_j">j</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+Extract">Extract</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- einfo("pubmed")
e[1:5]
e["Description"]
e[["Links"]]

e2 &lt;- einfo("pubmed", retmode = 'json')
e2[["header"]]
e2[["einforesult"]][["dbinfo"]][["description"]]

## End(Not run)
</code></pre>

<hr>
<h2 id='+5B+2Celink+2CANY+2Cmissing-method'>ELink Accessors</h2><span id='topic++5B+2Celink+2CANY+2Cmissing-method'></span><span id='topic++5B+2Celink+2Ccharacter+2CANY-method'></span>

<h3>Description</h3>

<p>Extract UIDs from an <code><a href="#topic+elink">elink</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'elink,ANY,missing'
x[i, j]

## S4 method for signature 'elink,character,ANY'
x[i]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2Celink+2B2CANY+2B2Cmissing-method_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+elink-class">elink</a></code> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Celink+2B2CANY+2B2Cmissing-method_+3A_i">i</code></td>
<td>
<p>Numeric or character indices.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Celink+2B2CANY+2B2Cmissing-method_+3A_j">j</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+entrez_linkset-class">entrez_linkset</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- elink(c("34577062", "24475906"), dbFrom = "nuccore")
e[1]

## End(Not run)
</code></pre>

<hr>
<h2 id='+5B+2Cesearch+2Cnumeric+2Cmissing-method'>ESearch Accessors</h2><span id='topic++5B+2Cesearch+2Cnumeric+2Cmissing-method'></span>

<h3>Description</h3>

<p>Extract UIDs from an <code><a href="#topic+esearch">esearch</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'esearch,numeric,missing'
x[i, j]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2Cesearch+2B2Cnumeric+2B2Cmissing-method_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+esearch-class">esearch</a></code> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cesearch+2B2Cnumeric+2B2Cmissing-method_+3A_i">i</code></td>
<td>
<p>Numeric indices.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cesearch+2B2Cnumeric+2B2Cmissing-method_+3A_j">j</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+entrez_uid-class">entrez_uid</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- esearch("Mus musculus", "protein", retmax = 20)
e[1:5]
## pass the subset directly on to esummary or efetch
content(esummary(e[1:5]), "parsed")

## End(Not run)
</code></pre>

<hr>
<h2 id='+5B+2Cesummary+2Ccharacter+2CANY-method'>ESummary accessors</h2><span id='topic++5B+2Cesummary+2Ccharacter+2CANY-method'></span><span id='topic++5B+5B+2Cesummary+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Extract XML nodes from an <code><a href="#topic+esummary-class">esummary</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'esummary,character,ANY'
x[i]

## S4 method for signature 'esummary,character'
x[[i]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2Cesummary+2B2Ccharacter+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+esummary-class">esummary</a></code> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cesummary+2B2Ccharacter+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>An XPath expression.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An XML node set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ds &lt;- esummary("470338", "protein")
ds["//Slen/node()"]

as.numeric(XML::xmlValue(ds[["//Slen"]]))

## End(Not run)
</code></pre>

<hr>
<h2 id='content'>Extract the data content from an Entrez request</h2><span id='topic+content'></span><span id='topic+content+2Cecitmatch-method'></span><span id='topic+content+2Cefetch-method'></span><span id='topic+content+2Cegquery-method'></span><span id='topic+content+2Ceinfo-method'></span><span id='topic+content+2Celink-method'></span><span id='topic+content+2Cepost-method'></span><span id='topic+content+2Cesearch-method'></span><span id='topic+content+2Cespell-method'></span><span id='topic+content+2Cesummary-method'></span><span id='topic+content+2Ceutil-method'></span>

<h3>Description</h3>

<p>There are five ways to access data returned by an Entrez request: as a character
string <code>(as = "text")</code>, as a <code><a href="base.html#topic+textConnection">textConnection</a></code>
<code>(as = "textConnection")</code>, as an <code><a href="XML.html#topic+XMLInternalDocument-class">XMLInternalDocument</a></code>
<code>(as = "xml")</code> or <code>json</code> object <code>(as = "json")</code>
(depending on the <code>retmode</code> with which the request was performed),
or parsed into a native R object, e.g. a <code>list</code> or a <code>data.frame</code>
<code>(as = "parsed")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>content(x, ...)

## S4 method for signature 'eutil'
content(x, ...)

## S4 method for signature 'ecitmatch'
content(x, as = "text")

## S4 method for signature 'efetch'
content(x, as = NULL)

## S4 method for signature 'egquery'
content(x, as = NULL)

## S4 method for signature 'einfo'
content(x, as = NULL)

## S4 method for signature 'elink'
content(x, as = NULL)

## S4 method for signature 'epost'
content(x, as = NULL)

## S4 method for signature 'esearch'
content(x, as = NULL)

## S4 method for signature 'espell'
content(x, as = NULL)

## S4 method for signature 'esummary'
content(x, as = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="content_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="content_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
<tr><td><code id="content_+3A_as">as</code></td>
<td>
<p>Type of output: <code>"text"</code>, <code>"xml"</code>, <code>"json"</code>,
<code>"textConnection"</code>, or <code>"parsed"</code>. <code>content</code> attempts to
figure out the most appropriate output type, based on the <code>retmode</code> of
the object.</p>
</td></tr>
</table>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: Access the data content from an <code>eutil</code> object.
</p>
</li>
<li> <p><code>ecitmatch</code>: Return PubMed IDs if <code>as = "parsed"</code>.
</p>
</li>
<li> <p><code>efetch</code>: Access the data content from an <code>efetch</code> request.
</p>
</li>
<li> <p><code>egquery</code>: Access the data content from an <code>egquery</code> request.
</p>
</li>
<li> <p><code>einfo</code>: Access the data content from an <code>einfo</code> request.
</p>
</li>
<li> <p><code>elink</code>: Access the data content from an <code>elink</code> request.
</p>
</li>
<li> <p><code>epost</code>: Access the data content from an <code>epost</code> request.
</p>
</li>
<li> <p><code>esearch</code>: Access the data content from an <code>esearch</code> request.
</p>
</li>
<li> <p><code>espell</code>: Access the data content from an <code>espell</code> request.
</p>
</li>
<li> <p><code>esummary</code>: Access the data content from an <code>esummary</code> request.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## einfo() defaults to retmode 'xml'
e &lt;- einfo()

## automatically return data as an 'XMLInternalDocument'.
if (e$no_errors()) {
  content(e)

  ## return the XML data as character string.
  cat(content(e, "text"))

  ## return DbNames parsed into a character vector.
  content(e, "parsed")
}

## return data as a JSON object
e2 &lt;- einfo(db = "gene", retmode = "json")
if (e2$no_errors()) {
  content(e2)
}

## return a textConnection to allow linewise reading of the data.
x &lt;- efetch("CP000828", "nuccore", rettype = "gbwithparts", retmode = "text")
con &lt;- content(x, as = "textConnection")
readLines(con, 2)
close(con)

## End(Not run)
</code></pre>

<hr>
<h2 id='database'>database</h2><span id='topic+database'></span><span id='topic+database+2Centrez_linkset-method'></span><span id='topic+database+2Centrez_uid-method'></span><span id='topic+database+2Ceutil-method'></span>

<h3>Description</h3>

<p>Retrieve the target database name from an <code><a href="#topic+eutil-class">eutil</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>database(x, ...)

## S4 method for signature 'eutil'
database(x, ...)

## S4 method for signature 'entrez_linkset'
database(x, ...)

## S4 method for signature 'entrez_uid'
database(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="database_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="database_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: Retrieve the target database name from an <code>eutil</code>
object.
</p>
</li>
<li> <p><code>entrez_linkset</code>: Retrieve the target database name from an <code>entrez_linkset</code>
object.
</p>
</li>
<li> <p><code>entrez_uid</code>: Retrieve the target database name from an <code>entrez_uid</code>
object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- esearch("Mus musculus", "taxonomy")
database(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='ecitmatch'>ecitmatch - searching PubMed with citation data</h2><span id='topic+ecitmatch'></span>

<h3>Description</h3>

<p><code>ecitmatch</code> serves as an API to the PubMed
<a href="https://www.ncbi.nlm.nih.gov/pubmed/batchcitmatch">batch citation matcher</a>.
It retrieves PubMed IDs (PMIDs) that correspond to a set of input citation strings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ecitmatch(bdata, db = "pubmed", retmode = "xml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ecitmatch_+3A_bdata">bdata</code></td>
<td>
<p>Citation strings. Each input citation must be represented
by a citation string in the following format:
<em>journal_title|year|volume|first_page|author_name|your_key|</em></p>
</td></tr>
<tr><td><code id="ecitmatch_+3A_db">db</code></td>
<td>
<p>Database to search. The only supported value is &lsquo;pubmed&rsquo;.</p>
</td></tr>
<tr><td><code id="ecitmatch_+3A_retmode">retmode</code></td>
<td>
<p>Retrieval mode. The only supported value is &lsquo;xml&rsquo;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code><a href="#topic+ecitmatch-class">ecitmatch</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>citstrings &lt;- c("proc natl acad sci u s a|1991|88|3248|mann bj|Art1|",
                "science|1987|235|182|palmenber ac|Art2|")
x &lt;- ecitmatch(citstrings)
x
if (x$no_errors()) {
  content(x, "parsed")
}

</code></pre>

<hr>
<h2 id='efetch'>efetch - downloading full records</h2><span id='topic+efetch'></span>

<h3>Description</h3>

<p><code>efetch</code> performs calls to the NCBI EFetch utility to retrieve data records
in the requested format for an NCBI Accession Number, one or more primary UIDs,
or for a set of UIDs stored in the user's web environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efetch(uid, db = NULL, rettype = NULL, retmode = NULL, outfile = NULL,
  retstart = NULL, retmax = NULL, querykey = NULL, webenv = NULL,
  strand = NULL, seqstart = NULL, seqstop = NULL, complexity = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="efetch_+3A_uid">uid</code></td>
<td>
<p>(Required) A list of UIDs provided either as a character vector, as an
<code>esearch</code> object, or by reference to a Web Environment
and a query key obtained directly from previous calls to <code><a href="#topic+esearch">esearch</a></code>
(if <code>usehistory = TRUE</code>), <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code>.
If UIDs are provided as a plain character vector, <code>db</code> must be
specified explicitly, and all of the UIDs must be from the database
specified by <code>db</code>.</p>
</td></tr>
<tr><td><code id="efetch_+3A_db">db</code></td>
<td>
<p>(Required if <code>uid</code> is a character vector of UIDs)
Database from which to retrieve records. See
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25497/table/chapter2.T._entrez_unique_identifiers_ui/?report=objectonly">here</a>
for the supported databases.</p>
</td></tr>
<tr><td><code id="efetch_+3A_rettype">rettype</code></td>
<td>
<p>A character string specifying the retrieval type, such as 'abstract' or
'medline' for PubMed, 'gp' or 'fasta' for Protein, or 'gb', or 'fasta' for Nuccore. See 
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/?report=objectonly">here</a>
for the available values for each database.</p>
</td></tr>
<tr><td><code id="efetch_+3A_retmode">retmode</code></td>
<td>
<p>A character string specifying the data mode of the records returned,
such as 'text' or 'xml'. See 
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/?report=objectonly">here</a>
for the available values for each database.</p>
</td></tr>
<tr><td><code id="efetch_+3A_outfile">outfile</code></td>
<td>
<p>A character string naming a file for writing the data to.
Required if more than 500 UIDs are retrieved at once. In this case UIDs
have to be provided by reference to a Web Environment and a query key
obtained directly from previous calls to <code><a href="#topic+esearch">esearch</a></code>
(if <code>usehistory = TRUE</code>), <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code>.</p>
</td></tr>
<tr><td><code id="efetch_+3A_retstart">retstart</code></td>
<td>
<p>Numeric index of the first record to be retrieved.</p>
</td></tr>
<tr><td><code id="efetch_+3A_retmax">retmax</code></td>
<td>
<p>Total number of records from the input set to be retrieved.</p>
</td></tr>
<tr><td><code id="efetch_+3A_querykey">querykey</code></td>
<td>
<p>An integer specifying which of the UID lists attached
to a user's Web Environment will be used as input to <code>efetch</code>.
(Usually obtained drectely from objects returned by a previous call to
<code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code>.)</p>
</td></tr>
<tr><td><code id="efetch_+3A_webenv">webenv</code></td>
<td>
<p>A character string specifying the Web Environment that
contains the UID list. (Usually obtained directely from objects returned
by a previous call to <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or
<code><a href="#topic+elink">elink</a></code>.)</p>
</td></tr>
<tr><td><code id="efetch_+3A_strand">strand</code></td>
<td>
<p>Strand of DNA to retrieve. (1: plus strand, 2: minus strand)</p>
</td></tr>
<tr><td><code id="efetch_+3A_seqstart">seqstart</code></td>
<td>
<p>First sequence base to retrieve.</p>
</td></tr>
<tr><td><code id="efetch_+3A_seqstop">seqstop</code></td>
<td>
<p>Last sequence base to retrieve.</p>
</td></tr>
<tr><td><code id="efetch_+3A_complexity">complexity</code></td>
<td>
<p>Data content to return. (0: entire data structure,
1: bioseq, 2: minimal bioseq-set, 3: minimal nuc-prot, 4: minimal pub-set)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the official online documentation for NCBI's
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch">EUtilities</a>
for additional information.
</p>
<p>See
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/?report=objectonly">here</a>
for the default values for <code>rettype</code> and <code>retmode</code>, as well as a list of the available
databases for the  EFetch utility.
</p>


<h3>Value</h3>

<p>An <code><a href="#topic+efetch-class">efetch</a></code> object.
</p>


<h3>Note</h3>

<p>If you are going to retrieve more than 500 UIDs at once, you will have to provide
the UIDs by reference to a Web Environment and a query key obtained from previous
calls to <code><a href="#topic+esearch">esearch</a></code> (if <code>usehistory = TRUE</code>), <code><a href="#topic+epost">epost</a></code>
or <code><a href="#topic+elink">elink</a></code> <strong>and</strong> you will have to specify an <code>outfile</code> to
write the data to, rather than collecting the data into an R object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+content">content</a></code>, <code><a href="#topic+getUrl">getUrl</a></code>, <code><a href="#topic+getError">getError</a></code>,
<code><a href="#topic+database">database</a></code>, <code><a href="#topic+retmode">retmode</a></code>, <code><a href="#topic+rettype">rettype</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## From Protein, retrieve a raw GenPept record and write it to a file.
p &lt;- efetch("195055", "protein", "gp")
p

write(content(p, "text"), file = "~/AAD15290.gp")

## Get accessions for a list of GenBank IDs (GIs)
acc &lt;- efetch(c("1621261", "89318838", "68536103", "20807972", "730439"),
              "protein", rettype = "acc")
acc
acc &lt;- strsplit(content(acc), "\n")[[1]]
acc

## Get GIs from a list of accession numbers
gi &lt;- efetch(c("CAB02640.1", "EAS10332.1", "YP_250808.1", "NP_623143.1", "P41007.1"),
             "protein", "uilist")
gi

## we can conveniently extract the UIDs using the eutil method #xmlValue(xpath)
gi$xmlValue("/IdList/Id")

## or we can extract the contents of the efetch query using the fuction content()
## and use the XML package to retrieve the UIDs
doc &lt;- content(gi)
XML::xpathSApply(doc, "/IdList/Id", XML::xmlValue)

## Get the scientific name for an organism starting with the NCBI taxon id.
tx &lt;- efetch("527031", "taxonomy")
tx
 
## Convenience accessor for XML nodes of interest using XPath
## Extract the TaxIds of the Lineage
tx["//LineageEx/Taxon/TaxId"]

## Use an XPath expession to extract the scientific name.
tx$xmlValue("/TaxaSet/Taxon/ScientificName")

## Iteratively retrieve a large number of records
# First store approx. 8400 UIDs on the History server.
uid &lt;- esearch(term = "hexokinase", db = 'protein', usehistory = TRUE)
# Fetch the records and write to file in batches of 500.
efetch(uid, rettype = "fasta", retmode = "text", outfile = "~/tmp/hexokinases.fna")


## End(Not run)
</code></pre>

<hr>
<h2 id='egquery'>egquery - performing a global Entrez search</h2><span id='topic+egquery'></span>

<h3>Description</h3>

<p><code>egquery</code> retrieves the number of records in all Entrez databases for
a single text query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>egquery(term)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="egquery_+3A_term">term</code></td>
<td>
<p>A valid Entrez text query.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the official online documentation for NCBIs
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EGQuery">EUtilities</a>
for additional information.
</p>


<h3>Value</h3>

<p>An <code><a href="#topic+egquery-class">egquery</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Determine the number of records for mouse in Entrez.
e &lt;- egquery("mouse[orgn]")
e
</code></pre>

<hr>
<h2 id='einfo'>einfo - getting database statistics and search fields</h2><span id='topic+einfo'></span>

<h3>Description</h3>

<p><code>einfo</code> queries the NCBI EInfo utility to retrieve the names of all
valid Entrez databases, or, if <code>db</code> is provided, to retrieve statistics
for a single database, including lists of indexing fields and available link
names. Version 2.0 data is requested by default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>einfo(db = NULL, version = "2.0", retmode = "xml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="einfo_+3A_db">db</code></td>
<td>
<p>A valid NCBI database name. If <code>NULL</code>, a list of all current
NCBI databases is returned.</p>
</td></tr>
<tr><td><code id="einfo_+3A_version">version</code></td>
<td>
<p>Specifies version 2.0 EInfo XML. Set to <code>NULL</code> for the
older version.</p>
</td></tr>
<tr><td><code id="einfo_+3A_retmode">retmode</code></td>
<td>
<p>'xml' (default) or 'json'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the official online documentation for NCBI's
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EInfo">EUtilities</a>
for additional information.
</p>


<h3>Value</h3>

<p>An <code><a href="#topic+einfo-class">einfo</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+content">content</a></code>, <code><a href="#topic+getUrl">getUrl</a></code>, <code><a href="#topic+getError">getError</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Fetch a list of all current Entrez database names
einfo()

## Fetch statistics for an Entrez database and parse
## the data into a data.frame
x &lt;- einfo("gene")
if (x$no_errors()) {
  content(x, "parsed")
}



## Fetch statistics for an Entrez database in JSON format
## and parse the data into a list
x &lt;- einfo("pubmed", retmode = "json")
if (x$no_errors()) {
  content(x, "parsed")
}

## End(Not run)
</code></pre>

<hr>
<h2 id='elink'>elink - finding related data through Entrez links</h2><span id='topic+elink'></span>

<h3>Description</h3>

<p><code>elink</code> generates a list of UIDs in a specified Entrez database that
are linked to a set of input UIDs in either the same or another
database. For instance, the ELink utility can find Entrez gene records
linked to records in Entrez Protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>elink(uid, dbFrom = NULL, dbTo = NULL, linkname = NULL,
  usehistory = FALSE, cmd = "neighbor", correspondence = FALSE,
  querykey = NULL, webenv = NULL, term = NULL, holding = NULL,
  datetype = NULL, reldate = NULL, mindate = NULL, maxdate = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="elink_+3A_uid">uid</code></td>
<td>
<p>(Required) A character vector of UIDs.</p>
</td></tr>
<tr><td><code id="elink_+3A_dbfrom">dbFrom</code></td>
<td>
<p>Initial database containing the UIDs in the input list.</p>
</td></tr>
<tr><td><code id="elink_+3A_dbto">dbTo</code></td>
<td>
<p>Destination database from which to retrieve linked UIDs. If
not provided links will be sought in the database containing the input UIDs.</p>
</td></tr>
<tr><td><code id="elink_+3A_linkname">linkname</code></td>
<td>
<p>Name of the Entrez link to retrieve. Every link in
Entrez is given a name of the form <em>dbFrom_dbTo_subset</em>.</p>
</td></tr>
<tr><td><code id="elink_+3A_usehistory">usehistory</code></td>
<td>
<p>If <code>TRUE</code> search results are stored directly in
the user's Web environment so that they can be used in subsequents 
calls to <code><a href="#topic+esummary">esummary</a></code> or <code><a href="#topic+efetch">efetch</a></code>.</p>
</td></tr>
<tr><td><code id="elink_+3A_cmd">cmd</code></td>
<td>
<p>ELink command mode (default: 'neighbor'). See Details.</p>
</td></tr>
<tr><td><code id="elink_+3A_correspondence">correspondence</code></td>
<td>
<p>if <code>TRUE</code> correspondence between query UIDs and
destination UIDs is preserved.</p>
</td></tr>
<tr><td><code id="elink_+3A_querykey">querykey</code></td>
<td>
<p>Query key.</p>
</td></tr>
<tr><td><code id="elink_+3A_webenv">webenv</code></td>
<td>
<p>Web Environment.</p>
</td></tr>
<tr><td><code id="elink_+3A_term">term</code></td>
<td>
<p>Search query to limit the output set of linked UIDs.</p>
</td></tr>
<tr><td><code id="elink_+3A_holding">holding</code></td>
<td>
<p>Name of LinkOut provider.</p>
</td></tr>
<tr><td><code id="elink_+3A_datetype">datetype</code></td>
<td>
<p>Type of date to limit the search. One of 'mdat'
(modification date), 'pdat' (publication date) or 'edat' (Entrez date).</p>
</td></tr>
<tr><td><code id="elink_+3A_reldate">reldate</code></td>
<td>
<p>umber of days back for which search items are
returned.</p>
</td></tr>
<tr><td><code id="elink_+3A_mindate">mindate</code></td>
<td>
<p>Minimum date of search range. Format YYYY/MM/DD.</p>
</td></tr>
<tr><td><code id="elink_+3A_maxdate">maxdate</code></td>
<td>
<p>Maximum date of search range. Format YYYY/MM/DD.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the official online documentation for NCBI's
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ELink">EUtilities</a>
for additional information.
</p>
<p>If <code>dbTo</code> and <code>dbFrom</code> are set to the same database, ELink will
return neighbors within that database.
</p>
<p>Elink commands (cmd) specify the function that elink will perform.
Available commands are:
</p>

<ul>
<li><p>&quot;<strong>neighbor</strong>&quot; (Default) ELink returns a set of UIDs in dbTo
linked to the input UIDs in dbFrom.
</p>
</li>
<li><p>&quot;<strong>neighbor_score</strong>&quot; ELink returns a set of UIDs within the
same database as the input UIDs along with similarity scores.
</p>
</li>
<li><p>&quot;<strong>neighbor_history</strong>&quot; ELink posts the output UIDs to the
Entrez History server and returns a query_key and WebEnv parameter.
Alternatively this is achieved by setting <code>usehistory=TRUE</code>
</p>
</li></ul>



<h3>Value</h3>

<p>An <code><a href="#topic+elink-class">elink</a></code> object.
</p>


<h3>See Also</h3>

<p>Combine calls to ELink with other EUtils:
<code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>.
</p>
<p>Accessor methods:
<code><a href="#topic+content">content</a></code>, <code><a href="#topic+getUrl">getUrl</a></code>, <code><a href="#topic+getError">getError</a></code>,
<code><a href="#topic+database">database</a></code>, <code><a href="#topic+uid">uid</a></code>, <code><a href="#topic+linkset">linkset</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Find one set of Gene IDs linked to nuccore GIs 34577062 and 24475906
e &lt;- elink(c("927442695", "312836839"), dbFrom = "nuccore", dbTo = "gene")
e

## Not run: 
## Find related articles to PMID 20210808
p &lt;- elink("20210808", dbFrom = "pubmed", dbTo = "pubmed")
p

## Extract linked UIDs from the "pubmed" to "pubmed_reviews" link
linkset(p, "pubmed_pubmed_reviews")

## or
p["pubmed_pubmed_reviews"]

## retrive the abstracts for the first five linked reviews
abstracts &lt;- efetch(p["pubmed_pubmed_reviews"][1:5], rettype = "abstract")

## End(Not run)
</code></pre>

<hr>
<h2 id='entrez_linkset-class'>Class <code>"entrez_linkset"</code></h2><span id='topic+entrez_linkset-class'></span>

<h3>Description</h3>

<p>A list containing a set of links as returned by a call to <code><a href="#topic+elink">elink</a></code>.
Each element of the list is a character vector of UIDs of class 
<code>"entrez_link"</code> with three attributes:
</p>

<dl>
<dt><code>score</code>:</dt><dd><p>Similarity scores between query UIDs and the linked UIDs</p>
</dd>
<dt><code>database</code>:</dt><dd><p>The destination database of the ELink query.</p>
</dd>
<dt><code>linkName</code>:</dt><dd><p>Name of the retrieved Entrez link of the form
<em>dbFrom_dbTo_subset</em></p>
</dd>
</dl>



<h3>Details</h3>

<p>An <code>"entrez_linkset"</code> has two global attributes:
</p>

<dl>
<dt><code>uid</code>:</dt><dd><p>The input UIDs.</p>
</dd>
<dt><code>database</code>:</dt><dd><p>The database containing the input UIDs.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>###
</code></pre>

<hr>
<h2 id='entrez_uid-class'>Class <code>"entrez_uid"</code></h2><span id='topic+entrez_uid-class'></span>

<h3>Description</h3>

<p>A container for UIDs returned by a call to <code><a href="#topic+esearch">esearch</a></code>.
It is essentially a character vector of UIDs supplemented with a number
of attributes:
</p>

<dl>
<dt><code>retmax</code>:</dt><dd><p>Total number of hits retrieved from the Entrez server.</p>
</dd>
<dt><code>retstart</code>:</dt><dd><p>Index of the first hit retrieved from the Entrez server.</p>
</dd>
<dt><code>count</code>:</dt><dd><p>Total number of hits for a search query.</p>
</dd>
<dt><code>query_translation</code>:</dt><dd><p>Details of how Entrez translated the query.</p>
</dd>
<dt><code>querykey</code>:</dt><dd><p>If <code>usehistory = TRUE</code>, the query key,
otherwise <code>NA</code>.</p>
</dd>
<dt><code>webenv</code>:</dt><dd><p>If <code>usehistory = TRUE</code>, the Web envronment string,
otherwise <code>NA</code>.</p>
</dd>
<dt><code>database</code>:</dt><dd><p>Name of the queried database.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>###
</code></pre>

<hr>
<h2 id='epost'>epost - uploading UIDs to Entrez</h2><span id='topic+epost'></span>

<h3>Description</h3>

<p><code>epost</code> uses the Entrez EPost utility to upload primary UIDs to the Entrez History server
or append a list of UIDs to an existing set of UIDs attached to a Web Environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epost(uid, db = NULL, webenv = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epost_+3A_uid">uid</code></td>
<td>
<p>(Required) List of UIDs provided as a character or as an <code>esearch</code>
object.</p>
</td></tr>
<tr><td><code id="epost_+3A_db">db</code></td>
<td>
<p>(Required if <code>uid</code> is a character vector) Database containing the
UIDs in the input list.</p>
</td></tr>
<tr><td><code id="epost_+3A_webenv">webenv</code></td>
<td>
<p>(Optional) Web Environment. If provided, this parameter
specifies the Web Environment that will receive the UIDs sent by
<code>epost</code>. <code>epost</code> will create a new query key associated with that
Web Environment. The <code>webenv</code> value is usually returned by a previous
call to <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code>.
If no <code>webenv</code> parameter is provided, the EPost utility will create a
new Web Environment and post the UIDs to query key 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>epost</code> returns an integer label called a query key and an encoded cookie string
called a Web environment. <code>epost</code> objects can then be used instead of a UID
list in subsequent calls to <code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>, or
<code><a href="#topic+elink">elink</a></code>.
</p>
<p>See the official online documentation for NCBI's
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EPost">EUtilities</a>
for additional information.
</p>


<h3>Value</h3>

<p>An <code><a href="#topic+epost-class">epost</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## post a list of protein GIs to the Entrez History server
gi &lt;- c("194680922", "50978626", "28558982", "9507199", "6678417")
p &lt;- epost(gi, "protein")
p
</code></pre>

<hr>
<h2 id='esearch'>esearch - searching an Entrez database</h2><span id='topic+esearch'></span>

<h3>Description</h3>

<p><code>esearch</code> performs searches using the the NCBI ESearch utility to retrieve
primary UIDs matching a text query. These UIDs can be used in subsequent calls
to <code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>, or <code><a href="#topic+elink">elink</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>esearch(term, db = "nuccore", rettype = "uilist", retmode = "xml",
  retstart = 0, retmax = 100, usehistory = FALSE, webenv = NULL,
  querykey = NULL, sort = NULL, field = NULL, datetype = NULL,
  reldate = NULL, mindate = NULL, maxdate = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="esearch_+3A_term">term</code></td>
<td>
<p>A valid Entrez text query.</p>
</td></tr>
<tr><td><code id="esearch_+3A_db">db</code></td>
<td>
<p>Database to search (default: nuccore).</p>
</td></tr>
<tr><td><code id="esearch_+3A_rettype">rettype</code></td>
<td>
<p>Retrieval type. (default: 'uilist', alternative: 'count')</p>
</td></tr>
<tr><td><code id="esearch_+3A_retmode">retmode</code></td>
<td>
<p>Retrieval mode. (default: 'xml', alternative: 'json')</p>
</td></tr>
<tr><td><code id="esearch_+3A_retstart">retstart</code></td>
<td>
<p>Numeric index of the first UID in the
retrieved set to be shown in the XML output (default: 0).</p>
</td></tr>
<tr><td><code id="esearch_+3A_retmax">retmax</code></td>
<td>
<p>Total number of UIDs to be retrieved (default: 100).</p>
</td></tr>
<tr><td><code id="esearch_+3A_usehistory">usehistory</code></td>
<td>
<p>If <code>TRUE</code>, search results are posted directly to
the Entrez History Server so that they can be used in subsequent 
calls to <code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>, or
<code><a href="#topic+elink">elink</a></code>. Also, <code>usehistory</code> must be set to <code>TRUE</code>
for <code>esearch</code> to interpret query key values included in <code>term</code>
or to accept a <code>webenv</code> as input.</p>
</td></tr>
<tr><td><code id="esearch_+3A_webenv">webenv</code></td>
<td>
<p>Web environment string returned by a previous call to
<code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code>.
When provided, <code>esearch</code> will append the results of the search to
the pre-existing Web environment. Providing <code>webenv</code> also allows
query keys to be used in <code>term</code> so that previous search sets can be
combined or limited.</p>
</td></tr>
<tr><td><code id="esearch_+3A_querykey">querykey</code></td>
<td>
<p>query key returned by a previous call to
<code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code>.
When provided, <code>esearch</code> will find the intersection of the set
specified by <code>querykey</code> and the set retrieved by the query in <code>term</code>
(i.e. joins the two with AND).</p>
</td></tr>
<tr><td><code id="esearch_+3A_sort">sort</code></td>
<td>
<p>Method used to sort UIDs in the ESearch output. The available
values vary by database. Example values are &lsquo;relevance&rsquo; and
&lsquo;name&rsquo; for Gene and &lsquo;first author&rsquo; and &lsquo;pub date&rsquo; for
PubMed.</p>
</td></tr>
<tr><td><code id="esearch_+3A_field">field</code></td>
<td>
<p>Optional. Search field used to limit the entire search
term.</p>
</td></tr>
<tr><td><code id="esearch_+3A_datetype">datetype</code></td>
<td>
<p>Optional. Type of date to limit the search. One of &quot;mdat&quot;
(modification date), &quot;pdat&quot; (publication date) or &quot;edat&quot; (Entrez date)</p>
</td></tr>
<tr><td><code id="esearch_+3A_reldate">reldate</code></td>
<td>
<p>Optional. Number of days back for which search items are
returned.</p>
</td></tr>
<tr><td><code id="esearch_+3A_mindate">mindate</code></td>
<td>
<p>Optional. Minimum date of search range. Format
YYYY/MM/DD, YYYY/MM, or YYYY.</p>
</td></tr>
<tr><td><code id="esearch_+3A_maxdate">maxdate</code></td>
<td>
<p>Optional. Maximum date of search range. Format
YYYY/MM/DD, YYYY/MM, or YYYY.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the official online documentation for NCBI's
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESearch">EUtilities</a>
for additional information on this EUtility.
</p>


<h3>Value</h3>

<p>An <code><a href="#topic+esearch-class">esearch</a></code> object.
</p>


<h3>See Also</h3>

<p>Combine calls to ESearch with other EUtils:
<code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>.
</p>
<p>Accessor methods:
<code><a href="#topic+content">content</a></code>, <code><a href="#topic+getUrl">getUrl</a></code>, <code><a href="#topic+getError">getError</a></code>,
<code><a href="#topic+database">database</a></code>, <code><a href="#topic+uid">uid</a></code>,
<code><a href="#topic+webenv">webenv</a></code>, <code><a href="#topic+querykey">querykey</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Search PubMed for articles with the term "Chlamydia psittaci" in the
## title that were published in 2013.
pmid &lt;- esearch("Chlamydia psittaci[titl] and 2013[pdat]", "pubmed")
pmid

## Not run: 
## Extract the query results either as an XML tree or parsed into
## a character vector
xml &lt;- content(pmid, "xml")
uids &lt;- uid(pmid)

## Alternatively post the UIDs to the History Server.
pmid &lt;- esearch("Chlamydia psittaci[titl] and 2013[pdat]", "pubmed",
                usehistory = TRUE)
pmid

## Associate new search results with the existing search results.
pmid2 &lt;- esearch("Chlamydia psittaci[titl] and 2012[pdat]", "pubmed",
                 usehistory = TRUE, webenv = webenv(pmid))
pmid2

## Sort results by author
pmid3 &lt;- esearch("Chlamydia psittaci[titl] and 2013[pdat]", "pubmed",
                 sort = "first author")
pmid3

## End(Not run)
</code></pre>

<hr>
<h2 id='espell'>espell - retrieving spelling suggestions</h2><span id='topic+espell'></span>

<h3>Description</h3>

<p>For a text query retrieve an XML containing the original query
and spelling suggestions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>espell(term, db = "pubmed")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="espell_+3A_term">term</code></td>
<td>
<p>An Entrez text query.</p>
</td></tr>
<tr><td><code id="espell_+3A_db">db</code></td>
<td>
<p>An Entrez database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code><a href="#topic+espell-class">espell</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>e &lt;- espell("Chlamidia")
e
</code></pre>

<hr>
<h2 id='esummary'>esummary - downloading Document Summaries</h2><span id='topic+esummary'></span>

<h3>Description</h3>

<p><code>esummary</code> performs calls to the NCBI ESummary utility to retrieve document
summaries (DocSums) for a list of primary UIDs or for a set of UIDs stored in the
user's web environment (using the Entrez History server).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>esummary(uid, db = NULL, retstart = 1, retmax = 10000, querykey = NULL,
  webenv = NULL, retmode = "xml", version = "2.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="esummary_+3A_uid">uid</code></td>
<td>
<p>(Required)
List of UIDs provided either as a character vector, as an
<code>esearch</code> or <code>elink</code> object, or by reference to a Web
Environment and a query key obtained directly from objects returned
by previous calls to <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or
<code><a href="#topic+elink">elink</a></code>.
If UIDs are provided as a plain character vector, <code>db</code> must be
specified explicitly, and all of the UIDs must be from the database
specified by <code>db</code>.</p>
</td></tr>
<tr><td><code id="esummary_+3A_db">db</code></td>
<td>
<p>(Required only when <code>id</code> is a character vector of UIDs)
Database from which to retrieve DocSums.</p>
</td></tr>
<tr><td><code id="esummary_+3A_retstart">retstart</code></td>
<td>
<p>Numeric index of the first DocSum to be retrieved
(default: 1).</p>
</td></tr>
<tr><td><code id="esummary_+3A_retmax">retmax</code></td>
<td>
<p>Total number of DocSums from the input set to be retrieved
(maximum: 10,000).</p>
</td></tr>
<tr><td><code id="esummary_+3A_querykey">querykey</code></td>
<td>
<p>An integer specifying which of the UID lists attached
to a user's Web Environment will be used as input to <code>efetch</code>.
(Usually obtained drectely from objects returned by previous
<code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or <code><a href="#topic+elink">elink</a></code> calls.)</p>
</td></tr>
<tr><td><code id="esummary_+3A_webenv">webenv</code></td>
<td>
<p>A character string specifying the Web Environment that
contains the UID list. (Usually obtained directely from objects returned
by previous <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+epost">epost</a></code> or
<code><a href="#topic+elink">elink</a></code> calls.)</p>
</td></tr>
<tr><td><code id="esummary_+3A_retmode">retmode</code></td>
<td>
<p>Retrieval mode. (default: 'xml', alternative: 'json')</p>
</td></tr>
<tr><td><code id="esummary_+3A_version">version</code></td>
<td>
<p>If &quot;2.0&quot; <code>esummary</code> will retrieve version 2.0
ESummary XML output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the official online documentation for NCBI's
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESummary">EUtilities</a>
for additional information.
</p>


<h3>Value</h3>

<p>An <code><a href="#topic+esummary-class">esummary</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+content">content</a></code>, <code><a href="#topic+getUrl">getUrl</a></code>, <code><a href="#topic+getError">getError</a></code>,
<code><a href="#topic+database">database</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Retrieve the Document Summary information for a set of
## UIDs frome the Nuccore datanase.
ds &lt;- esummary(c("1060721643", "1060721620", "1060721618"), "nuccore")
ds

## Not run: 
## parse the XML into a data frame
df &lt;- content(ds, "parsed")
df

## use XPath expressions to extract nodes of interest
ds['//TaxId/text()']

## End(Not run)
</code></pre>

<hr>
<h2 id='eutil_error-class'>Class <code>"eutil_error"</code>:</h2><span id='topic+eutil_error'></span><span id='topic+eutil_error-class'></span>

<h3>Description</h3>

<p>A container for handling errors when trying to parse XML files returned
by Entrez.
</p>


<h3>Fields</h3>


<dl>
<dt><code>Error</code></dt><dd><p>messages returned by the Entrez server when no search could be
performed like, e.g., &quot;Invalid db name specified&quot;.</p>
</dd>
<dt><code>errmsg</code></dt><dd><p>Error messages pertaining to the search like e.g.,
&quot;PhraseNotFound&quot;.</p>
</dd>
<dt><code>wrnmsg</code></dt><dd><p>Warnings like, e.g., &quot;No items found.&quot;</p>
</dd>
</dl>


<h3>Methods</h3>


<dl>
<dt><code>all_empty()</code></dt><dd><p>Are all error fields <code>NULL</code>?</p>
</dd>
<dt><code>check_errors(object, verbose = TRUE)</code></dt><dd><p>check if a <code>linkS4class{eutil}</code> object contains errors</p>
</dd>
</dl>


<h3>Extends</h3>

<p>All reference classes extend and inherit methods from
<code>"<a href="methods.html#topic+envRefClass-class">envRefClass</a>"</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getError">getError</a></code>, <code><a href="#topic+eutil-class">eutil</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("eutil_error")
</code></pre>

<hr>
<h2 id='eutil-class'>Class <code>"eutil"</code>: Reference classes that hold the response from EUtils
requests.</h2><span id='topic+ecitmatch-class'></span><span id='topic+efetch-class'></span><span id='topic+egquery-class'></span><span id='topic+einfo-class'></span><span id='topic+elink-class'></span><span id='topic+epost-class'></span><span id='topic+esearch-class'></span><span id='topic+espell-class'></span><span id='topic+esummary-class'></span><span id='topic+eutil'></span><span id='topic+eutil-class'></span>

<h3>Description</h3>

<p>The reference classes <code><a href="#topic+eutil-class">eutil</a></code>, <code><a href="#topic+einfo-class">einfo</a></code>,
<code><a href="#topic+esearch-class">esearch</a></code>, <code><a href="#topic+esummary-class">esummary</a></code>,
<code><a href="#topic+efetch-class">efetch</a></code>, <code><a href="#topic+elink-class">elink</a></code>, <code><a href="#topic+epost-class">epost</a></code>,
<code><a href="#topic+egquery-class">egquery</a></code>, <code><a href="#topic+espell-class">espell</a></code>, and
<code><a href="#topic+ecitmatch-class">ecitmatch</a></code> implement the request generator for interaction
with the NCBI services.
They should not be used direcly, but initialized through the respective
constructor functions <code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>,
<code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>,
<code><a href="#topic+epost">epost</a></code>, <code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, and
<code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Fields</h3>


<dl>
<dt><code>params</code></dt><dd><p>A named <code>list</code> of query parameters.</p>
</dd>
<dt><code>errors</code></dt><dd><p>A <code><a href="#topic+eutil_error-class">eutil_error</a></code> object.</p>
</dd>
<dt><code>content</code></dt><dd><p>Result of an Entrez request stored as a character vector.</p>
</dd>
</dl>


<h3>Methods</h3>


<dl>
<dt><code>get_content(as = "text", ...)</code></dt><dd><p>Return the results of an Entrez query as text, xml, json, a textConnection,
or parsed to a list or data.frame; should not be used directly, use
<code><a href="#topic+content">content</a></code> instead.</p>
</dd>
<dt><code>get_error()</code></dt><dd><p>Return <code><a href="#topic+eutil_error-class">eutil_error</a></code>s; should not be used directly,
use <code><a href="#topic+getError">getError</a></code> instead.</p>
</dd>
<dt><code>get_url()</code></dt><dd><p>Return the URL used for an Entrez query; should not be used directly,
use <code><a href="#topic+getUrl">getUrl</a></code> instead.</p>
</dd>
<dt><code>perform_query(method = "GET", ...)</code></dt><dd><p>Perform an Entrez query using either http GET or POST requests;
should not be used directly.</p>
</dd>
<dt><code>xmlAttr(xpath, name, as = "character", default = NA_character_)</code></dt><dd><p>Extract the value of XML attributes given a valid XPath expression
and an attribute name</p>
</dd>
<dt><code>xmlName(xpath, as = "character", default = NA_character_)</code></dt><dd><p>Extract the tag names of XML nodes given a valid XPath expression.</p>
</dd>
<dt><code>xmlSet(xpath, ...)</code></dt><dd><p>Extract a set of XML nodes given a valid XPath expression.</p>
</dd>
<dt><code>xmlValue(xpath, as = "character", default = NA_character_)</code></dt><dd><p>Extract the text value of XML leaf nodes given a valid XPath expression.</p>
</dd>
</dl>


<h3>Extends</h3>

<p>All reference classes extend and inherit methods from
<code>"<a href="methods.html#topic+envRefClass-class">envRefClass</a>"</code>. Furthermore, <code>"einfo"</code>,
<code>"esearch"</code>, <code>"esummary"</code>, <code>"efetch"</code>, <code>"elink"</code>,
<code>"epost"</code>, <code>"egquery"</code>, <code>"espell"</code>, and <code>"ecitmatch"</code>
all extend the <code>"eutil"</code> class.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+eutil-class">eutil</a></code>, <code><a href="#topic+einfo">einfo</a></code>,
<code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>, <code><a href="#topic+efetch">efetch</a></code>,
<code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>, <code><a href="#topic+egquery">egquery</a></code>,
<code><a href="#topic+espell">espell</a></code>, and <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("eutil")
</code></pre>

<hr>
<h2 id='getError'>getError</h2><span id='topic+getError'></span><span id='topic+getError+2Ceutil-method'></span>

<h3>Description</h3>

<p>Retrieve a http or XML parsing error from an <code><a href="#topic+eutil-class">eutil</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getError(x, ...)

## S4 method for signature 'eutil'
getError(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getError_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="getError_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code><a href="#topic+eutil_error-class">eutil_error</a></code> object.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: a http or XML parsing error from an <code>eutil</code> object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- efetch("Nonsensical_accession_nr", "protein", rettype = "fasta")
getError(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='getUrl'>getUrl</h2><span id='topic+getUrl'></span><span id='topic+getUrl+2Ceutil-method'></span>

<h3>Description</h3>

<p>Retrieve the URL used to perform an Entrez E-Utilities query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getUrl(x, ...)

## S4 method for signature 'eutil'
getUrl(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUrl_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="getUrl_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: etrieve the URL used to perform an Entrez E-Utilities
query from an <code>eutil</code> object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- efetch("AV333213.1", "protein", rettype = "fasta")
getUrl(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='linkset'>linkset</h2><span id='topic+linkset'></span><span id='topic+linkset+2Celink-method'></span><span id='topic+linkset+2Centrez_linkset-method'></span>

<h3>Description</h3>

<p>Retrieve a linkset from an <code><a href="#topic+elink-class">elink</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linkset(x, linkname = NULL, ...)

## S4 method for signature 'entrez_linkset'
linkset(x, linkname = NULL, ...)

## S4 method for signature 'elink'
linkset(x, linkname = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linkset_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+elink-class">elink</a></code> object.</p>
</td></tr>
<tr><td><code id="linkset_+3A_linkname">linkname</code></td>
<td>
<p>(optional) Name of the Entrez link to retrieve. Every link in
Entrez is given a name of the form <em>dbFrom_dbTo_subset</em>. If <code>NULL</code>,
all available links are retrieved from the object.</p>
</td></tr>
<tr><td><code id="linkset_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>entrez_linkset</code>: Retrieve a linkset from an <code>elink</code> object.
</p>
</li>
<li> <p><code>elink</code>: Retrieve a <code>linkset</code> from an <code>elink</code> object.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Find related articles to PMID 20210808 and xtract linked UIDs from the
## "pubmed" to "pubmed_reviews" link
x &lt;- elink("20210808", dbFrom = "pubmed", dbTo = "pubmed", cmd = "neighbor_score")
linkset(x, "pubmed_pubmed_reviews")

## End(Not run)
</code></pre>

<hr>
<h2 id='make_flattener'>Flatten nested lists</h2><span id='topic+make_flattener'></span>

<h3>Description</h3>

<p>Generate a function that accepts an arbitrarily deeply nested <code>list</code>.
Use <code>flatten.at</code> to set the level of nestedness at which the flatterner
will start to flatten.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_flattener(flatten.at = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_flattener_+3A_flatten.at">flatten.at</code></td>
<td>
<p>An <code>integer</code> specifying the layer after which to 
start the flattening. <code>1</code> means to start at the very top.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A function that can be used to flatten a nested list
</p>

<hr>
<h2 id='ncbi_retrieval_type'>Set the NCBI rettype</h2><span id='topic+ncbi_retrieval_type'></span>

<h3>Description</h3>

<p>Set the NCBI rettype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_retrieval_type(db, rettype = NULL, retmode = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncbi_retrieval_type_+3A_db">db</code></td>
<td>
<p>A valid NCBI database.</p>
</td></tr>
<tr><td><code id="ncbi_retrieval_type_+3A_rettype">rettype</code></td>
<td>
<p>Optional.</p>
</td></tr>
<tr><td><code id="ncbi_retrieval_type_+3A_retmode">retmode</code></td>
<td>
<p>Optional.</p>
</td></tr>
</table>

<hr>
<h2 id='performQuery'>performQuery</h2><span id='topic+performQuery'></span><span id='topic+performQuery+2Ceutil-method'></span>

<h3>Description</h3>

<p>performQuery
</p>


<h3>Usage</h3>

<pre><code class='language-R'>performQuery(x, method = "GET", ...)

## S4 method for signature 'eutil'
performQuery(x, method = "GET", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="performQuery_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="performQuery_+3A_method">method</code></td>
<td>
<p>One of &ldquo;GET&rdquo; or &ldquo;POST&rdquo;.</p>
</td></tr>
<tr><td><code id="performQuery_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: Perform an Entrez query using either http GET or
POST requests.
</p>
</li></ul>

<hr>
<h2 id='querykey'>querykey</h2><span id='topic+querykey'></span><span id='topic+querykey+2Centrez_uid-method'></span><span id='topic+querykey+2Cepost-method'></span><span id='topic+querykey+2Cesearch-method'></span>

<h3>Description</h3>

<p>An integer query key returned by an ESearch, EPost or ELink call if
the History server was used. Otherwise <code>NA</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>querykey(x, ...)

## S4 method for signature 'epost'
querykey(x, ...)

## S4 method for signature 'entrez_uid'
querykey(x, ...)

## S4 method for signature 'esearch'
querykey(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="querykey_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="querykey_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer or <code>NA</code>.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>epost</code>: Retrieve the querykey from an <code>epost</code> object.
</p>
</li>
<li> <p><code>entrez_uid</code>: Retrieve the querykey from an <code>entrez_uid</code> object.
</p>
</li>
<li> <p><code>esearch</code>: Retrieve the querykey from an <code>esearch</code> object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- esearch("Mus musculus", "taxonomy", usehistory = TRUE)
querykey(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='retmode'>retmode</h2><span id='topic+retmode'></span><span id='topic+retmode+2Ceutil-method'></span>

<h3>Description</h3>

<p>Get the &ldquo;retrieval mode&rdquo; of an <code><a href="#topic+eutil-class">eutil</a></code> object.
It is usually one of <code>xml</code>, <code>json</code>, <code>text</code>, or <code>asn.1</code>. 
It is set to <code>NULL</code> if &ldquo;retrieval mode&rdquo; is not supported by an
E-Utility.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retmode(x, ...)

## S4 method for signature 'eutil'
retmode(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retmode_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="retmode_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string or <code>NULL</code>.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: Access the &ldquo;retrieval mode&rdquo; of an <code>eutil</code>
object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- efetch("10090", "taxonomy")
retmode(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='rettype'>rettype</h2><span id='topic+rettype'></span><span id='topic+rettype+2Ceutil-method'></span>

<h3>Description</h3>

<p>Get the &ldquo;retrieval type&rdquo; of an <code><a href="#topic+eutil-class">eutil</a></code> object. See 
<a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/?report=objectonly">here</a>
for the available retrieval types for different NCBI databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rettype(x, ...)

## S4 method for signature 'eutil'
rettype(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rettype_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="rettype_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>eutil</code>: Access the &ldquo;retrieval type&rdquo; of an <code>eutil</code>
object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- esearch("Mus musculus", "taxonomy")
rettype(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='uid'>uid</h2><span id='topic+uid'></span><span id='topic+uid+2Celink-method'></span><span id='topic+uid+2Centrez_linkset-method'></span><span id='topic+uid+2Centrez_uid-method'></span><span id='topic+uid+2Cesearch-method'></span>

<h3>Description</h3>

<p>Retrieve the list of UIDs returned by a call to ESearch or ELink.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uid(x, ...)

## S4 method for signature 'entrez_linkset'
uid(x, ...)

## S4 method for signature 'elink'
uid(x, ...)

## S4 method for signature 'entrez_uid'
uid(x, ...)

## S4 method for signature 'esearch'
uid(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="uid_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="uid_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>entrez_linkset</code>: Retrieve the list of UIDs from an <code>entrez_linkset</code> object.
</p>
</li>
<li> <p><code>elink</code>: Retrieve the list of UIDs from an <code>elink</code> object.
</p>
</li>
<li> <p><code>entrez_uid</code>: Retrieve the list of UIDs from an <code>entrez_uid</code> object.
</p>
</li>
<li> <p><code>esearch</code>: Retrieve the list of UIDs from an <code>esearch</code> object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+elink">elink</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- esearch("Mus musculus", "taxonomy")
uid(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='webenv'>webenv</h2><span id='topic+webenv'></span><span id='topic+webenv+2Centrez_uid-method'></span><span id='topic+webenv+2Cepost-method'></span><span id='topic+webenv+2Cesearch-method'></span>

<h3>Description</h3>

<p>Retrieve the Web environment string returned from an ESearch, EPost or ELink call.
<code>NA</code> if the History server was not used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>webenv(x, ...)

## S4 method for signature 'epost'
webenv(x, ...)

## S4 method for signature 'entrez_uid'
webenv(x, ...)

## S4 method for signature 'esearch'
webenv(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="webenv_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+eutil-class">eutil</a></code> object.</p>
</td></tr>
<tr><td><code id="webenv_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string or <code>NA</code>.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>epost</code>: Retrieve the webenv string from an <code>epost</code> object.
</p>
</li>
<li> <p><code>entrez_uid</code>: Retrieve the webenv string from an <code>entrez_uid</code> object.
</p>
</li>
<li> <p><code>esearch</code>: Retrieve the webenv string from an <code>esearch</code> object.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+einfo">einfo</a></code>, <code><a href="#topic+esearch">esearch</a></code>, <code><a href="#topic+esummary">esummary</a></code>,
<code><a href="#topic+efetch">efetch</a></code>, <code><a href="#topic+elink">elink</a></code>, <code><a href="#topic+epost">epost</a></code>,
<code><a href="#topic+egquery">egquery</a></code>, <code><a href="#topic+espell">espell</a></code>, <code><a href="#topic+ecitmatch">ecitmatch</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
e &lt;- esearch("Mus musculus", "taxonomy", usehistory = TRUE)
webenv(e)

## End(Not run)
</code></pre>

<hr>
<h2 id='xattr'>Extract the attributes of XML nodes</h2><span id='topic+xattr'></span>

<h3>Description</h3>

<p>Extract the attributes of XML nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xattr(doc, path, name, as = "character", default = NA_character_, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xattr_+3A_doc">doc</code></td>
<td>
<p>An object of class <code>XMLInternalDocument</code>.</p>
</td></tr>
<tr><td><code id="xattr_+3A_path">path</code></td>
<td>
<p>An XPath expression.</p>
</td></tr>
<tr><td><code id="xattr_+3A_name">name</code></td>
<td>
<p>Name of the attribute to be extracted.</p>
</td></tr>
<tr><td><code id="xattr_+3A_as">as</code></td>
<td>
<p>Mode of return value (<code>character</code>, <code>integer</code>, <code>numeric</code>).</p>
</td></tr>
<tr><td><code id="xattr_+3A_default">default</code></td>
<td>
<p>Default return value.</p>
</td></tr>
<tr><td><code id="xattr_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="XML.html#topic+xpathApply">xpathApply</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='xname'>Extract the tag name of XML nodes</h2><span id='topic+xname'></span>

<h3>Description</h3>

<p>Extract the tag name of XML nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xname(doc, path, as = "character", default = NA_character_, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xname_+3A_doc">doc</code></td>
<td>
<p>An object of class <code>XMLInternalDocument</code>.</p>
</td></tr>
<tr><td><code id="xname_+3A_path">path</code></td>
<td>
<p>An XPath expression.</p>
</td></tr>
<tr><td><code id="xname_+3A_as">as</code></td>
<td>
<p>Mode of return value (<code>character</code>, <code>integer</code>, <code>numeric</code>).</p>
</td></tr>
<tr><td><code id="xname_+3A_default">default</code></td>
<td>
<p>Default return value.</p>
</td></tr>
<tr><td><code id="xname_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="XML.html#topic+xpathApply">xpathApply</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='xset'>Extract a node set from an XML document</h2><span id='topic+xset'></span>

<h3>Description</h3>

<p>Extract a node set from an XML document
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xset(doc, path, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xset_+3A_doc">doc</code></td>
<td>
<p>An object of class <code>XMLInternalDocument</code>.</p>
</td></tr>
<tr><td><code id="xset_+3A_path">path</code></td>
<td>
<p>An XPath expression.</p>
</td></tr>
<tr><td><code id="xset_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="XML.html#topic+xpathApply">xpathApply</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='xvalue'>Extract the content of XML leaf nodes</h2><span id='topic+xvalue'></span>

<h3>Description</h3>

<p>Extract the content of XML leaf nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xvalue(doc, path, as = "character", default = NA_character_, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xvalue_+3A_doc">doc</code></td>
<td>
<p>An object of class <code>XMLInternalDocument</code>.</p>
</td></tr>
<tr><td><code id="xvalue_+3A_path">path</code></td>
<td>
<p>An XPath expression.</p>
</td></tr>
<tr><td><code id="xvalue_+3A_as">as</code></td>
<td>
<p>Mode of return value (<code>character</code>, <code>integer</code>, <code>numeric</code>).</p>
</td></tr>
<tr><td><code id="xvalue_+3A_default">default</code></td>
<td>
<p>Default return value.</p>
</td></tr>
<tr><td><code id="xvalue_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="XML.html#topic+xpathApply">xpathApply</a></code>.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
