<!DOCTYPE html><html><head><title>Help for package revert</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {revert}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getReversions'><p>Detecting reversion mutations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Reversion Mutation Identifier for Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Hui Xiao [aut, cre],
  Adam Mills [aut],
  John Alexander [ctb],
  Stephen Pettitt [aut],
  Syed Haider [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hui Xiao &lt;hui.xiao@icr.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool for detecting reversions for a given pathogenic mutation from 
	next-generation DNA sequencing data. It analyses reads aligned to the locus of the 
	pathogenic mutation and reports reversion events where secondary mutations have 
	restored or undone the deleterious effect of the original pathogenic mutation, e.g., 
	secondary indels complement to a frameshift pathogenic mutation converting the 
	orignal frameshift mutation into inframe mutaions, deletions or SNVs that replaced 
	the original pathogenic mutation restoring the open reading frame, SNVs changing the 
	stop codon caused by the original nonsense SNV into an amino acid, etc.</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>R &gt;= 4.1.0, SAMtools &gt;= 1.11</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, futile.logger</td>
</tr>
<tr>
<td>Imports:</td>
<td>IRanges, GenomicRanges, Biostrings, Rsamtools, BSgenome,
BSgenome.Hsapiens.UCSC.hg38</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-23 15:35:55 UTC; hxiao</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-23 16:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='getReversions'>Detecting reversion mutations</h2><span id='topic+getReversions'></span>

<h3>Description</h3>

<p>Function for detecting reversions for a given pathogenic mutation from reads alignment of NGS genomic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReversions(
  bam.file,
  genome.version = "BSgenome.Hsapiens.UCSC.hg38",
  chromosome,
  pathog.mut.start,
  pathog.mut.type = "SNV",
  snv.reference.allele = NULL,
  snv.alternative.allele = NULL,
  deletion.sequence = NULL,
  deletion.length = NULL,
  insertion.sequence = NULL,
  flanking.window = 100,
  minus.strand = FALSE,
  check.wildtype.reads = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReversions_+3A_bam.file">bam.file</code></td>
<td>
<p>Character. 
The input bam file containing aligned reads.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_genome.version">genome.version</code></td>
<td>
<p>Character. 
Genome version of alignments in bam file and the position of pathogenic mutation.
The name of genome version is specified by the available BSgenome data packages: BSgenome::available.genomes().
Default is &quot;BSgenome.Hsapiens.UCSC.hg38&quot;.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_chromosome">chromosome</code></td>
<td>
<p>Character. 
Name of chromosome where pathogenic mutation is located, e.g., &quot;chr17&quot; or &quot;17&quot;, &quot;chrX&quot; or &quot;X&quot;.
The chromosome name should be concordant with the chromosome identifiers used in bam.file and genome.version</p>
</td></tr>
<tr><td><code id="getReversions_+3A_pathog.mut.start">pathog.mut.start</code></td>
<td>
<p>Integer. 
Genomic coordinate of the start position of pathogenic mutation.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_pathog.mut.type">pathog.mut.type</code></td>
<td>
<p>Character. 
Type of pathogenic mutation: &quot;SNV&quot;, &quot;DEL&quot; or &quot;INS&quot;.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_snv.reference.allele">snv.reference.allele</code></td>
<td>
<p>Character.
Reference allele of pathogenic mutation if it is a single nucleotide variant (SNV).
Default is NULL.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_snv.alternative.allele">snv.alternative.allele</code></td>
<td>
<p>Character.
Alternative allele of pathogenic mutation if it is a SNV.
Default is NULL.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_deletion.sequence">deletion.sequence</code></td>
<td>
<p>Character. 
Deleted nucleotides of pathogenic mutation if it is a deletion (DEL). 
Default is NULL.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_deletion.length">deletion.length</code></td>
<td>
<p>Integer. 
Number of deleted nucleotides of pathogenic mutation if it is a DEL. 
Parameters deletion.sequence and deletion.length can not both be NULL when pathogenic mutation is DEL.
Default is NULL.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_insertion.sequence">insertion.sequence</code></td>
<td>
<p>Character. 
Inserted nucleotides of pathogenic mutation if it is an insertion (INS). 
Default is NULL.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_flanking.window">flanking.window</code></td>
<td>
<p>Integer. 
Length of flanking regions (bp) to pathogenic mutation locus for reversion detection. 
Default is 100.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_minus.strand">minus.strand</code></td>
<td>
<p>Logical. 
TRUE if the gene in question is on the reverse strand. 
Default is FALSE.</p>
</td></tr>
<tr><td><code id="getReversions_+3A_check.wildtype.reads">check.wildtype.reads</code></td>
<td>
<p>Logical. 
TRUE if assume wildtype reads mapped to pathogenic mutation are restored to wildtype and alternative reversions will be detected from the wildtype reads.
Only used for pathogenic mutation targeted gene editing experiment.   
Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two tables summarizing the reversion detection:
1. The numbers of different types of reads analysed in the input bam file
2. Reversion mutation table including the following columns: 
rev_id: Unique ID for reversion event
rev_freq: Frequency of reversion event
rev_type: Type of reversion event, i.e., complement reversion to pathogenic mutation, replacement reversion of pathogenic mutation, or alternative reversion to pathogenic mutation 
rev_mut_number: Index of each mutation in a reversion event
mut_id: Unique ID for reversion mutation 
chr: Chromosome
mut_start_pos: Start position of reversion mutation
mut_type: Type of reversion mutation, i.e., SNV, DEL or INS
mut_seq: Sequence changes of mutation, i.e., inserted or deleted sequences for indels, or reference and alternative alleles for SNVs 
mut_length: Length of mut_seq, 0 for SNV
mut_hgvs: HGVS Genomic DNA ID of reversion mutation
pathog_mut_hgvs: Original pathogenic reversion mutation
dist_to_pathog_mut: Distance between original pathogenic mutation and reversion mutation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# To detect reversions for BRCA2 mutation "chr13:g.32338763-32338764delAT"
bam.file2 &lt;- system.file('extdata', 'toy_alignments_2.bam', package = 'revert')
reversions &lt;- getReversions(
    bam.file = bam.file2, 
    genome.version = "BSgenome.Hsapiens.UCSC.hg38",
    chromosome = "chr13", 
    pathog.mut.start = 32338763, 
    pathog.mut.type = "DEL", 
    deletion.sequence = "AT",
    deletion.length = 2,
    flanking.window = 100,
    minus.strand = FALSE )

}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
