<!DOCTYPE html><html><head><title>Help for package RevEcoR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RevEcoR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RevEcoR'><p>The RevEcoR package</p></a></li>
<li><a href='#anno.species'><p>Annotation profiles of seven well-studied oral species</p></a></li>
<li><a href='#calculateCooperationIndex'><p>Calculating the metabolic competition and complementarity index</p></a></li>
<li><a href='#compose'><p>Compose multiple functions</p></a></li>
<li><a href='#confidencescore-methods'><p>Conficence score</p></a></li>
<li><a href='#getGsMN-methods'><p>The genome scale metabolic network</p></a></li>
<li><a href='#getOrgMetabolicData'><p>Get organism metabolic data from KEGG database</p></a></li>
<li><a href='#getSeedSets'><p>Identify seed compounds of each organism</p></a></li>
<li><a href='#gut_microbiome'><p>Annotation profiles of 116 gut prevalent species</p></a></li>
<li><a href='#Interactions'><p>Calculating the species interactions</p></a></li>
<li><a href='#kegg_buc'><p>Metabolic profiles of KEGG organism Buchnera aphidicola APS (Acyrthosiphon</p>
pisum) (KEGG organism code: buc)</a></li>
<li><a href='#kegg_ptr'><p>Metabolic profiles of KEGG organism Pan troglodytes (chimpanzee) (KEGG</p>
organism code: ptr)</a></li>
<li><a href='#KosarajuSCC'><p>Caculating the strong connected components (SCC) of a network</p></a></li>
<li><a href='#len-methods'><p>the length of the seed set</p></a></li>
<li><a href='#nonseed-methods'><p>Non seed of the network</p></a></li>
<li><a href='#reconstructGsMN'><p>Reconstuction of the specific-organism genome-scale metabolic network</p></a></li>
<li><a href='#RefDbcache'><p>Reference data for global metabolic construction</p>
</p>
<p>The reference metabolic pathway data contains KOs, substrates and products, as well as</p>
a constructed reference global network, which used for metabolic network
reconstruction</a></li>
<li><a href='#seedset-class'><p><code>seedset-class</code></p></a></li>
<li><a href='#seedSize-methods'><p>Size of the each seed source component</p></a></li>
<li><a href='#show,seedset-method'><p>The show generic function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Reverse Ecology Analysis on Microbiome</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-3-28</td>
</tr>
<tr>
<td>Author:</td>
<td>Yang Cao, Fei Li</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yang Cao &lt;yiluheihei@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of the reverse ecology framework. Reverse ecology
    refers to the use of genomics to study ecology with no a priori assumptions
    about the organism(s) under consideration, linking organisms to their
    environment. It allows researchers to reconstruct the metabolic networks and
    study the ecology of poorly characterized microbial species from their
    genomic information, and has substantial potentials for microbial community
    ecological analysis.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.14)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, igraph, XML, stringr, magrittr, gtools, plyr, purrr,
methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-04-03 08:42:54 UTC; caoyang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-04-03 17:14:01</td>
</tr>
</table>
<hr>
<h2 id='RevEcoR'>The RevEcoR package</h2><span id='topic+RevEcoR'></span><span id='topic+RevEcoR-package'></span>

<h3>Description</h3>

<p>This package implementation the applications of reverse ecology. Reverse ecology refers to the use of genomics to study ecology with no a priori assumptions about the organism(s) under consideration, linking the organism and their environment. Prediction the cooperation among species and hosts.
</p>

<hr>
<h2 id='anno.species'>Annotation profiles of seven well-studied oral species</h2><span id='topic+anno.species'></span>

<h3>Description</h3>

<p>A dataset containing the the KEGG orthology annotation profiles of seven oral
species which was downloaded from the Integrated Microbial Genomes (IMG).
</p>


<h3>Format</h3>

<p>A list with seven elements and each elements represents the annotation profile of the species</p>


<h3>Details</h3>

<p>This datasets constains the KEGG orthology annotation information of
seven oral species whose interactions were carefully and well
characterized. The human oral microbiota is relatively  #' well described.
The name of these seven species is: <em>Aggregatibacter</em>, 
<em>actinomycetemcomitans D7S-1</em>, <em>Fusobacterium nucleatum
polymorphum ATCC 10953</em>, <em>Porphyromonas gingivalis ATCC 33277</em>,
<em>Streptococcus gordonii str. Challis substr. CH1</em>,
<em>Streptococcus oralis SK23, ATCC 35037</em>, <em>Veillonella atypica
ACS-134-V-Col7a</em>. For more annotation information on these species, see
<a href="img.jgi.doe.gov/">img.jgi.doe.gov/</a>.
</p>

<ul>
<li><p> Aa, Aggregatibacter actinomycetemcomitans D7S-1
</p>
</li>
<li><p> Ao, Actinomyces oris K20
</p>
</li>
<li><p> Fn, Fusobacterium nucleatum polymorphum ATCC 10953
</p>
</li>
<li><p> Pg, Porphyromonas gingivalis ATCC 33277
</p>
</li>
<li><p> Sg, Streptococcus gordonii str. Challis substr. CH1
</p>
</li>
<li><p> So, Streptococcus oralis SK23, ATCC 35037
</p>
</li>
<li><p> Va, Veillonella atypica ACS-134-V-Col7a  
</p>
</li></ul>



<h3>Source</h3>

<p><a href="img.jgi.doe.gov/">img.jgi.doe.gov/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anno.species)
</code></pre>

<hr>
<h2 id='calculateCooperationIndex'>Calculating the metabolic competition and complementarity index</h2><span id='topic+calculateCooperationIndex'></span>

<h3>Description</h3>

<p>Calculating the metabolic competition complementarity index among all metabolic 
networks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateCooperationIndex(g, ..., threshold = 0, p = FALSE, nperm = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateCooperationIndex_+3A_g">g</code></td>
<td>
<p>igraph that represents a metabolic network, see <code><a href="#topic+reconstructGsMN">reconstructGsMN</a></code></p>
</td></tr>
<tr><td><code id="calculateCooperationIndex_+3A_...">...</code></td>
<td>
<p>a list of metabolic networks or a network append to g</p>
</td></tr>
<tr><td><code id="calculateCooperationIndex_+3A_threshold">threshold</code></td>
<td>
<p>threshold, the cutoff of confidence score to be serve as a 
seed set, default is 0.2</p>
</td></tr>
<tr><td><code id="calculateCooperationIndex_+3A_p">p</code></td>
<td>
<p>a logical value which determins whether the calculated index is 
statistical or biological significant. default is FALSE</p>
</td></tr>
<tr><td><code id="calculateCooperationIndex_+3A_nperm">nperm</code></td>
<td>
<p>the number of permuations of metabolic network node labes, which
is used for p value calculation, default is 1000.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Metabolic competition index is defined as the fraction of compounds 
in a species seed set of metabolic network that are alse included in its 
partner; However, metabolic complementarity index is the fraction of 
compounds in one species seed set of metabolic network appearing in the 
metabolic network but not in the seed set of its partner; The biosynthetic 
support score represents the extent to which the metabolic requirements of a 
potential parasitic organism can be supported by the biosynthetic capacity of
a potential host. It is measured by calculating the fraction of the source 
components of a, in which at least one of the compounds can be found in the 
network of b. However, seed compounds are associated with a confidence score 
(1/size of SCC), so this fraction is calculated as a mormalized weighted sum. 
</p>
<p>The ith row and jth col elements of the returnd matrix represents the 
metabolic competition index or complementarity index of the ith network on the
jth metabolic network.
</p>


<h3>Value</h3>

<p>a cooperation index matrix whose nrow and ncol is equal to the number 
of species to be compared, for more see details.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+complementarityIndex">complementarityIndex</a></code>,
<code><a href="#topic+competitionIndex">competitionIndex</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## metabolic network reconstruction and seed set identity of sample data anno.species
net &lt;- lapply(anno.species,reconstructGsMN)
interactions &lt;- calculateCooperationIndex(net)

## End(Not run)
</code></pre>

<hr>
<h2 id='compose'>Compose multiple functions</h2><span id='topic++25.+25'></span><span id='topic+compose'></span>

<h3>Description</h3>

<p>In infix and prefix forms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose(...)

f %.% g
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compose_+3A_...">...</code></td>
<td>
<p>n functions to apply in order from right to left</p>
</td></tr>
<tr><td><code id="compose_+3A_f">f</code>, <code id="compose_+3A_g">g</code></td>
<td>
<p>two functions to compose for the infix form</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function was from hadley wickham's package pryr, for more details see <a href="https://github.com/hadley/pryr">https://github.com/hadley/pryr</a>
</p>


<h3>Author(s)</h3>

<p>Hadley wickham
</p>


<h3>Examples</h3>

<pre><code class='language-R'>not_null &lt;- `!` %.% is.null
not_null(4)
not_null(NULL)

add1 &lt;- function(x) x + 1
compose(add1,add1)(8)
</code></pre>

<hr>
<h2 id='confidencescore-methods'>Conficence score</h2><span id='topic+confidencescore'></span><span id='topic+confidencescore+2Cseedset-method'></span><span id='topic+confidencescore-methods'></span>

<h3>Description</h3>

<p>Caculate confidence score of seed set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>confidencescore(object)

## S4 method for signature 'seedset'
confidencescore(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="confidencescore-methods_+3A_object">object</code></td>
<td>
<p><code>seedset</code> class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
confidencescore(seed.set)

## End(Not run)
</code></pre>

<hr>
<h2 id='getGsMN-methods'>The genome scale metabolic network</h2><span id='topic+getGsMN'></span><span id='topic+getGsMN+2Cseedset-method'></span><span id='topic+getGsMN-methods'></span>

<h3>Description</h3>

<p>T he genome scale metabolic network (GsMN) whose seed set is caculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGsMN(object)

## S4 method for signature 'seedset'
getGsMN(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGsMN-methods_+3A_object">object</code></td>
<td>
<p><code>seedset</code> class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a igraph
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getGsMN(seed.set)

## End(Not run)
</code></pre>

<hr>
<h2 id='getOrgMetabolicData'>Get organism metabolic data from KEGG database</h2><span id='topic+getOrgMetabolicData'></span>

<h3>Description</h3>

<p>This function helps us  to obtain the specific-organism pathway map, prasing 
this maps to get metabolic data contains reaction, substrate and product.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOrgMetabolicData(org)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getOrgMetabolicData_+3A_org">org</code></td>
<td>
<p>characters, the KEGG organism code, e.g. &quot;buc&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>getOrgMetabolicData</code> helps us to download metabolic 
data of a given organism from KEGG database with REST-style KEGG API. 
Enzyme reactions take place in this organism (org) and its metabolites 
(substrates and products), that will be used for organism-specific genome
scale metabolic network reconstruction, can be obtained with this function.
</p>


<h3>Value</h3>

<p>a three length df, consists of  enzyme reaction names, substrates and
products
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getSeedSets">getSeedSets</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
metabolic.data &lt;- getOrgMetabolicData("buc")

## End(Not run)
</code></pre>

<hr>
<h2 id='getSeedSets'>Identify seed compounds of each organism</h2><span id='topic+getSeedSets'></span>

<h3>Description</h3>

<p>Detect a given metabolic network and idendity the seed compounds of each
organism
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSeedSets(g, threshold = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSeedSets_+3A_g">g</code></td>
<td>
<p>an igraph object which represents a given organism-specific
metaboliic network</p>
</td></tr>
<tr><td><code id="getSeedSets_+3A_threshold">threshold</code></td>
<td>
<p>numeric constant ranges from 0 to 1, default is 0.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the compound in the same source SCC all equally to be included
in the seed set, each of these compounds was assigned a confidence level,
C=1/(size of souce SCC), denoting the compounds probability of being a
seed. This threshold was used to determin whether a compound should be a
seed.
</p>


<h3>Value</h3>

<p>a two-length list which consists of network and the seed set
compounds of the given organism-specific metabolic network,
....
</p>


<h3>See Also</h3>

<p><code><a href="#topic+KosarajuSCC">KosarajuSCC</a></code>,<code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## get metabolic annotated data of a specific species
metabolic.data &lt;- getOrgMetabolicData("buc")
## metabolic network reconstruction
net &lt;- reconstructGsMN(metabolic.data)

## End(Not run)
</code></pre>

<hr>
<h2 id='gut_microbiome'>Annotation profiles of 116 gut prevalent species</h2><span id='topic+gut_microbiome'></span>

<h3>Description</h3>

<p>A dataset containing the the KEGG orthology annotation profiles of 116 gut 
prevlent species which was downloaded from the Integrated Microbial Genomes 
(IMG).
</p>


<h3>Format</h3>

<p>A list with 116 elements and each elements represents the annotation
profile of the species</p>


<h3>Details</h3>

<p>This dataset focused on a list of 116 prevalent gut species, whose 
genome sequence is available in IMG database and sequence coverage is more 
than 1
annotation profiles of this 116 species was collected from IMG database.
</p>
<p>With a in-house R script, we obtained genomic data for all organisms from
the Department of Integrated Microbial Genomes project (IMG). For each
species, the list of genes mapped to the Kyoto Encyclopedia of Genes and
Genomes orthologous groups (KEGG KOs) was downloaded. For more annotation 
information on these species, see <a href="img.jgi.doe.gov/">img.jgi.doe.gov/</a>.
</p>


<h3>Source</h3>

<p><a href="img.jgi.doe.gov/">img.jgi.doe.gov/</a>
</p>

<hr>
<h2 id='Interactions'>Calculating the species interactions</h2><span id='topic+Interactions'></span><span id='topic+competitionIndex'></span><span id='topic+complementarityIndex'></span>

<h3>Description</h3>

<p>Calculating the metabolic complementarity index and complementarity index of 
based on species metabolic network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>complementarityIndex(g1, g2, seed.set1, seed.set2, threshold = 0, p = FALSE,
  nperm = 1000)

competitionIndex(g1, g2, seed.set1, seed.set2, threshold = 0, p = FALSE,
  nperm = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Interactions_+3A_g1">g1</code></td>
<td>
<p>igraph object, a species-specific metabolic network.</p>
</td></tr>
<tr><td><code id="Interactions_+3A_g2">g2</code></td>
<td>
<p>igraph object, a species-specific metabolic network, the complementary
network of g1</p>
</td></tr>
<tr><td><code id="Interactions_+3A_seed.set1">seed.set1</code></td>
<td>
<p>seeds slot of a seed-set object, seeds of the metabolic 
network g1, more details see <code><a href="#topic+seedset-class">seedset-class</a></code>.</p>
</td></tr>
<tr><td><code id="Interactions_+3A_seed.set2">seed.set2</code></td>
<td>
<p>seeds slot of a seed-set object, seeds of the metabolic 
network g2, more details see <code><a href="#topic+seedset-class">seedset-class</a></code>.</p>
</td></tr>
<tr><td><code id="Interactions_+3A_threshold">threshold</code></td>
<td>
<p>the cutoff of confidence score to be serve as a seed set, 
default is 0.</p>
</td></tr>
<tr><td><code id="Interactions_+3A_p">p</code></td>
<td>
<p>a logical value which determins whether the calculated index is 
statistical or biological significant. default is FALSE.</p>
</td></tr>
<tr><td><code id="Interactions_+3A_nperm">nperm</code></td>
<td>
<p>the number of permuations of metabolic network node labes, which
is used for complementarity index's P value calculating, default is 1000.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Metabolic competition index is defined as the fraction of compounds 
in a species seed set of metabolic network that are also included in its partner; 
However, metabolic complementarity index is the fraction of compounds in one 
species seed set of metabolic network appearing in the metabolic network but 
not in the seed set of its partner. However, seed compounds are associated 
with a confidence score (1/size of SCC), so this fraction is calculated as a 
normalized weighted sum.
</p>
<p>Based on the metabolic network and seed sets of species, this functions help 
us to predict the species interactions of species1 on the presence of species2.
</p>


<h3>Value</h3>

<p>a two length list: complementarity index or competition index: range 
from 0 to 1, p value of complementarity index. Or a single value of 
complementarity or competition index while p is FALSE.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getSeedSets">getSeedSets</a></code>,
<code><a href="#topic+calculateCooperationIndex">calculateCooperationIndex</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## metabolic network reconstruction and seed set identity of sample data anno.species
net &lt;- lapply(anno.species,reconstructGsMN)
seed.sets &lt;- lapply(net, getSeedSets) 
seed.sets &lt;- lapply(seed.sets, function(x)x@seeds)

## calculate the complementarity index of the first species
complementarity.index &lt;- complementarityIndex(net[[1]],net[[2]], 
 seed.sets[[1]], seed.sets[[2]])
competition.index &lt;- competitionIndex(net[[1]],net[[2]], 
 seed.sets[[1]], seed.sets[[2]])

## End(Not run)
</code></pre>

<hr>
<h2 id='kegg_buc'>Metabolic profiles of KEGG organism Buchnera aphidicola APS (Acyrthosiphon 
pisum) (KEGG organism code: buc)</h2><span id='topic+kegg_buc'></span>

<h3>Description</h3>

<p>kegg organism buc metabolic information, which consists of enzymatic reactions
and metabolites.
</p>


<h3>Format</h3>

<p>A data frame with 418 observations on three variables.
</p>
<p>[,1] .attrs.name, character (reaction: R)
</p>
<p>[,2] substrate.name, list (substrates: cpd)
</p>
<p>[,3] product.name, list (products: cpd)</p>


<h3>Details</h3>

<p>buc metatolic information:
</p>

<ul>
<li><p> .attrs.name: Enzymatic reactions that organism involved
</p>
</li>
<li><p> substrate.name: Substrates of the corresponding reaction.
</p>
</li>
<li><p> product.name: Products of the corresponding reaction.
</p>
</li></ul>


<hr>
<h2 id='kegg_ptr'>Metabolic profiles of KEGG organism Pan troglodytes (chimpanzee) (KEGG
organism code: ptr)</h2><span id='topic+kegg_ptr'></span>

<h3>Description</h3>

<p>kegg organism ptr metabolic information, which consists of enzymatic reactions
and metabolites.
</p>


<h3>Format</h3>

<p>A data frame with 1858 observations on three variables.
</p>
<p>[,1] .attrs.name, character (reaction: R)
</p>
<p>[,2] substrate.name, list (substrates: cpd)
</p>
<p>[,3] product.name, list (products: cpd)</p>


<h3>Details</h3>

<p>ptr metatolic information:
</p>
 <ul>
<li><p> .attrs.name: Enzymatic reactions that organism involved </p>
</li>
<li>
<p>substrate.name: Substrates of the corresponding reaction. </p>
</li>
<li><p> product.name:
Products of the corresponding reaction. </p>
</li></ul>


<hr>
<h2 id='KosarajuSCC'>Caculating the strong connected components (SCC) of a network</h2><span id='topic+KosarajuSCC'></span>

<h3>Description</h3>

<p>This function utilizes Kosaraju's algorithm to caculate the
strong connetected components descomposition of a given network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KosarajuSCC(g)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KosarajuSCC_+3A_g">g</code></td>
<td>
<p>a igraph object to be caculated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list which length is equal to the number of SCCs, each element represents a Scc
</p>


<h3>References</h3>

<p><em>AV Aho, JE Hopcroft, JD Ullman: The design and analysis of computer algorithms, 1974</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getSeedSets">getSeedSets</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
metabolic.data &lt;- getOrgMetabolicData("buc")
## metabolic network reconstruction
net &lt;- reconstructGsMN(metabolic.data)
scc &lt;- KosarajuSCC(net)

## End(Not run) 
</code></pre>

<hr>
<h2 id='len-methods'>the length of the seed set</h2><span id='topic+len'></span><span id='topic+len+2Cseedset-method'></span><span id='topic+len-methods'></span>

<h3>Description</h3>

<p>Caculate the number of the seed source components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>len(object)

## S4 method for signature 'seedset'
len(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="len-methods_+3A_object">object</code></td>
<td>
<p><code>seed-set</code> class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interger
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
len(seed.set)

## End(Not run)
</code></pre>

<hr>
<h2 id='nonseed-methods'>Non seed of the network</h2><span id='topic+nonseed'></span><span id='topic+nonseed+2Cseedset-method'></span><span id='topic+nonseed-methods'></span>

<h3>Description</h3>

<p>Non seed of the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nonseed(object)

## S4 method for signature 'seedset'
nonseed(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nonseed-methods_+3A_object">object</code></td>
<td>
<p><code>seedset</code> class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
nonseed(seed.set)

## End(Not run)
</code></pre>

<hr>
<h2 id='reconstructGsMN'>Reconstuction of the specific-organism genome-scale metabolic network</h2><span id='topic+reconstructGsMN'></span>

<h3>Description</h3>

<p>Reconstruction of genome-scale metabolic network (GsMN) whose 
nodes represents compounds and whose edges represents reactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconstructGsMN(metabolic.data, RefData = RefDbcache, threshold = 10,
  is.gaint = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reconstructGsMN_+3A_metabolic.data">metabolic.data</code></td>
<td>
<p>df or a character vector. More details see function 
<code>getOrgMetabolicData</code> and <code>details</code></p>
</td></tr>
<tr><td><code id="reconstructGsMN_+3A_refdata">RefData</code></td>
<td>
<p>The reference metabolic data. It does not need reference data 
While organism metabolic data was collected from KEGG database, and RefData 
is set to NULL. Otherwise, RefDbCache, an internal dataset in this package,
was taken as the Reference metabolic data for Genome scale metabolic reconstruction.</p>
</td></tr>
<tr><td><code id="reconstructGsMN_+3A_threshold">threshold</code></td>
<td>
<p>numeric, Nodes belonging to components with fewer than the 
value of threshold nodes will be ignored. This is a good option for networks
that contain many small and trivial components. Default is 10.</p>
</td></tr>
<tr><td><code id="reconstructGsMN_+3A_is.gaint">is.gaint</code></td>
<td>
<p>logical, Ignore all nodes except those in the giant component:
selecting the only main largest component (connected set of nodes) of the
network. All smaller components will be ignored. This is a good option for
networks with a dominant component. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input of this function can be of two forms. If organims is 
collected in KEGG database, it can be obtained with 
<code>getOrgMetabolicData</code> which is a data frame. Otherwise, 
<code>metabolic.data</code> could be a  character vecotr which contains the KEGG 
Orthology annotated information on this organism, e.g. we can download this 
KO annotation profile in the <a href="https://img.jgi.doe.gov">https://img.jgi.doe.gov</a> website for 
species detected in a human microbime which not contained in KEGG organism 
database. Several functions, such as <code>link{read.table}</code> and 
<code><a href="utils.html#topic+read.delim">read.delim</a></code> could help us to read KO annotation profile.
</p>


<h3>Value</h3>

<p>igraph object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getOrgMetabolicData">getOrgMetabolicData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## not run (organism in KEGG)
## metabolic.data &lt;- getOrgMetabolicData("buc")
## g &lt;- reconstructGsMN(metabolic.data)

## species detected in a human microbiome
annodir &lt;- system.file("extdata","koanno.tab",package = "RevEcoR")
metabolic.data &lt;- read.delim2(file=annodir,stringsAsFactors=FALSE)
##load the reference metabolic data
data(RefDbcache)
g2 &lt;- reconstructGsMN(metabolic.data, RefData = RefDbcache)

</code></pre>

<hr>
<h2 id='RefDbcache'>Reference data for global metabolic construction 
The reference metabolic pathway data contains KOs, substrates and products, as well as 
a constructed reference global network, which used for metabolic network
reconstruction</h2><span id='topic+RefDbcache'></span>

<h3>Description</h3>

<p>Reference data for global metabolic construction 
</p>
<p>The reference metabolic pathway data contains KOs, substrates and products, as well as 
a constructed reference global network, which used for metabolic network
reconstruction
</p>


<h3>Format</h3>

<p>The format is: List of 7 KO, substrate, product, user, date, version, 
reference network</p>


<h3>Details</h3>

<p>Information this dataset is involved:
</p>

<ul>
<li><p> KO, all KEGG orthlogy enties in KEGG metabolic pathways.
</p>
</li>
<li><p> substrate, substrate of enzymatic reactions in all KEGG metabolic pathways.
</p>
</li>
<li><p> product, product of enzymatic reactions in all KEGG metabolic pathways.
</p>
</li>
<li><p> user who download this data.
</p>
</li>
<li><p> date, the date this data is downloaded.
</p>
</li>
<li><p> version, R version used to obtained it.
</p>
</li>
<li><p> network, the global network which is reconstructed based on all the metabolites.
</p>
</li></ul>



<h3>References</h3>

<p><a href="https://www.bioconductor.org/packages/release/bioc/html/mmnet.html">https://www.bioconductor.org/packages/release/bioc/html/mmnet.html</a>
</p>

<hr>
<h2 id='seedset-class'><code>seedset-class</code></h2><span id='topic+seedset-class'></span>

<h3>Description</h3>

<p>Object representing the seed sets of a given metabolic network
</p>


<h3>Slots</h3>


<dl>
<dt><code>GsMN,</code></dt><dd><p>a igraph network</p>
</dd>
<dt><code>seeds,</code></dt><dd><p>a character list represents seeds of a given metabolic 
network which is composed of  the KEGG compound index.</p>
</dd>
</dl>


<h3>method</h3>


<ul>
<li><p>getGsMN, <code>signature(object = "seedset")</code>:
get the genome scale metabolic network whose seed set is caculated
</p>
</li>
<li><p>len, <code>signature(object = "seedset")</code>:
return the number of source SCC
</p>
</li>
<li><p>seedSize, <code>signature(object = "seedset")</code>:
returns the sizes of each source SCCs
</p>
</li>
<li><p>nonseed, <code>signature(object = "seedset")</code>:
the non seeds of the GsMN
</p>
</li>
<li><p>show, <code>signature(object = "seedset")</code>:
show the short summary of a seedset class
</p>
</li>
<li><p>confidencescore, <code>signature(object = "seedset")</code>:
confidence score of the seed set
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getSeedSets">getSeedSets</a></code>,<code><a href="#topic+getGsMN">getGsMN</a></code>,<code><a href="#topic+len">len</a></code>,
<code><a href="#topic+nonseed">nonseed</a></code>,<code><a href="#topic+seedSize">seedSize</a></code>,<code><a href="#topic+confidencescore">confidencescore</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#' ## generate a metabolic network in igraph class and a seed set of this graph
annodir &lt;- system.file("extdata","koanno.tab",package = "RevEcoR")
metabolic.data &lt;- read.delim2(file=annodir,stringsAsFactors=FALSE)
g &lt;- reconstructGsMN(metabolic.data)
seeds &lt;- getSeedSets(g)@seeds
seed.set &lt;- new("seedset",GsMN = g, seeds = seeds)

## End(Not run)
</code></pre>

<hr>
<h2 id='seedSize-methods'>Size of the each seed source component</h2><span id='topic+seedSize'></span><span id='topic+seedSize+2Cseedset-method'></span><span id='topic+seedSize-methods'></span>

<h3>Description</h3>

<p>Caculate the size of each seed source component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seedSize(object)

## S4 method for signature 'seedset'
seedSize(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seedSize-methods_+3A_object">object</code></td>
<td>
<p><code>seedset</code> class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector represents size of each source seed componet of network
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seedSize(seed.set)

## End(Not run)
</code></pre>

<hr>
<h2 id='show+2Cseedset-method'>The show generic function</h2><span id='topic+show'></span><span id='topic+show+2Cseedset-method'></span><span id='topic+show-methods'></span>

<h3>Description</h3>

<p>Show a short summary of seedset object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'seedset'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2Cseedset-method_+3A_object">object</code></td>
<td>
<p><code>seed-set</code> class</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+seedset-class">seedset-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
show(seed.set)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
