<!DOCTYPE html><html><head><title>Help for package RevGadgets</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RevGadgets}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculateShiftBayesFactor'><p>Bayes Factors in support of a shift in diversification rates over a given</p>
time interval.</a></li>
<li><a href='#colFun'><p>Color Function</p></a></li>
<li><a href='#combineTraces'><p>Combine traces</p></a></li>
<li><a href='#densiTreeWithBranchData'><p>DensiTree-style plot with branch-specific data</p></a></li>
<li><a href='#dropTip'><p>dropTip</p></a></li>
<li><a href='#geom_stepribbon'><p>plot geom stepribbon for diversification rates</p></a></li>
<li><a href='#getMAP'><p>get MAP</p></a></li>
<li><a href='#matchNodes'><p>match Nodes</p></a></li>
<li><a href='#plotAncStatesMAP'><p>plot Ancestral States MAP</p></a></li>
<li><a href='#plotAncStatesPie'><p>plot Ancestral States Pie</p></a></li>
<li><a href='#plotDiversityOBDP'><p>Plot Diversity Distribution from OBDP Analysis</p></a></li>
<li><a href='#plotDivRates'><p>Plot Diversification Rates</p></a></li>
<li><a href='#plotFBDTree'><p>Plot FBD tree</p></a></li>
<li><a href='#plotHiSSE'><p>plotHiSSE</p></a></li>
<li><a href='#plotMassExtinctions'><p>Plot Mass Extinction Support</p></a></li>
<li><a href='#plotMuSSE'><p>plotMuSSE</p></a></li>
<li><a href='#plotPopSizes'><p>Plot Population Sizes</p></a></li>
<li><a href='#plotPostPredStats'><p>plot Posterior Predictive Statistics</p></a></li>
<li><a href='#plotTrace'><p>Plot trace</p></a></li>
<li><a href='#plotTree'><p>Plot tree</p></a></li>
<li><a href='#plotTreeFull'><p>Plot Full tree</p></a></li>
<li><a href='#posteriorSamplesToParametricPrior'><p>Priors from MCMC samples</p></a></li>
<li><a href='#processAncStates'><p>Process Ancestral States</p></a></li>
<li><a href='#processBranchData'><p>processBranchData</p></a></li>
<li><a href='#processDivRates'><p>Process Diversification Rates</p></a></li>
<li><a href='#processPopSizes'><p>Process Population Sizes</p></a></li>
<li><a href='#processPostPredStats'><p>process Posterior Predictive Statistics</p></a></li>
<li><a href='#processSSE'><p>Title</p></a></li>
<li><a href='#readOBDP'><p>Read OBDP Outputs</p></a></li>
<li><a href='#readTrace'><p>Read trace</p></a></li>
<li><a href='#readTrees'><p>Read trees</p></a></li>
<li><a href='#removeBurnin'><p>Remove Burnin</p></a></li>
<li><a href='#rerootPhylo'><p>Reroot Phylo</p></a></li>
<li><a href='#RevGadgets'><p>RevGadgets</p></a></li>
<li><a href='#setMRFGlobalScaleHyperpriorNShifts'><p>Sets a global scale parameter for a GMRF or HSMRF model given a prior</p>
mean number of effective shifts.</a></li>
<li><a href='#simulateMRF'><p>Simulates a single Markov random field trajectory.</p></a></li>
<li><a href='#summarizeTrace'><p>Summarize trace</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization and Post-Processing of 'RevBayes' Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Carrie Tribble &lt;ctribble09@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Processes and visualizes the output of complex phylogenetic analyses from the 'RevBayes' phylogenetic graphical modeling software.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/revbayes/RevGadgets">https://github.com/revbayes/RevGadgets</a>,
<a href="https://revbayes.github.io/tutorials/intro/revgadgets">https://revbayes.github.io/tutorials/intro/revgadgets</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/revbayes/RevGadgets/issues">https://github.com/revbayes/RevGadgets/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 5.4), phytools (&ge; 0.7-70), dplyr (&ge; 1.0.0), ggtree
(&ge; 3.6.1), tidytree (&ge; 0.3.4), treeio (&ge; 1.12.0), ggplot2
(&ge; 3.4.0), reshape (&ge; 0.8.8), methods (&ge; 4.1.0), tidyr (&ge;
1.1.0), tibble (&ge; 3.0.1), gginnards (&ge; 0.0.3), ggplotify (&ge;
0.0.5), ggpp, ggimage, png (&ge; 0.1-7), stats (&ge; 4.0.1), utils
(&ge; 4.0.1), grDevices (&ge; 4.0.1), deeptime (&ge; 0.1.0), scales
(&ge; 1.1.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, phangorn</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-29 20:17:15 UTC; carrietribble</td>
</tr>
<tr>
<td>Author:</td>
<td>Carrie Tribble <a href="https://orcid.org/0000-0001-7263-7885"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Michael R. May [aut],
  William A. Freyman [aut],
  Michael J. Landis [aut],
  Lim Jun Ying [aut],
  Joelle Barido-Sottani [aut],
  Andrew Magee [aut],
  Bjorn Tore Kopperud [aut],
  Sebastian Hohna [aut],
  Nagashima Kengo [ctb],
  Schliep Klaus [ctb],
  Ronja J. Billenstein [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-29 20:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculateShiftBayesFactor'>Bayes Factors in support of a shift in diversification rates over a given
time interval.</h2><span id='topic+calculateShiftBayesFactor'></span>

<h3>Description</h3>

<p>This function computes the Bayes Factor in favor of a rate-shift between
time t1 and t2 (t1 &lt; t2).
The default assumption (suitable to standard HSMRF and GMRF models) is
that the prior probability of a shift is 0.5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateShiftBayesFactor(
  rate_trace,
  time_trace,
  rate_name,
  time_name,
  t1,
  t2,
  prior_prob = 0.5,
  decrease = TRUE,
  return_2lnBF = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateShiftBayesFactor_+3A_rate_trace">rate_trace</code></td>
<td>
<p>(list; no default) The processed Rev output of the rate
of interest through time for computation (output of readTrace()).</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_time_trace">time_trace</code></td>
<td>
<p>(list; no default) The processed Rev output of the
change/interval times of the rate of interest through time for computation
(output of readTrace()).</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_rate_name">rate_name</code></td>
<td>
<p>(character; no default) The name of the parameter
(e.g. &quot;speciation&quot;) for which Bayes Factor is to be calculated.</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_time_name">time_name</code></td>
<td>
<p>(character; no default) The name of the interval times
(e.g. &quot;interval_times) for the rate change times.</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_t1">t1</code></td>
<td>
<p>(numeric; no default) Support will be assessed for a shift between
time t1 and time t2 (t1 &lt; t2).</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_t2">t2</code></td>
<td>
<p>(numeric; no default) Support will be assessed for a shift between
time t1 and time t2 (t1 &lt; t2).</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_prior_prob">prior_prob</code></td>
<td>
<p>(numeric; 0.5) The prior probability of a shift over this
interval (default of 0.5 applies to standard HSMRF- and GMRF-based models).</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_decrease">decrease</code></td>
<td>
<p>(logical; default TRUE) Should support be assessed for a
decrease in the parameter (if TRUE) or an increase (if FALSE) between
t1 and t2?</p>
</td></tr>
<tr><td><code id="calculateShiftBayesFactor_+3A_return_2lnbf">return_2lnBF</code></td>
<td>
<p>(logical; TRUE) Should the 2ln(BF) be returned
(if TRUE) or simply the BF (if FALSE)?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Bayes Factor.
</p>


<h3>References</h3>

<p>Kass and Raftery (1995) Bayes Factors.
<em>JASA</em>, <b>90 (430)</b>, 773-795.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#' # download the example datasets to working directory
url_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_times.log"
dest_path_times &lt;- "primates_EBD_speciation_times.log"
download.file(url_times, dest_path_times)

url_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_rates.log"
dest_path_rates &lt;- "primates_EBD_speciation_rates.log"
download.file(url_rates, dest_path_rates)

# to run on your own data, change this to the path to your data file
speciation_time_file &lt;- dest_path_times
speciation_rate_file &lt;- dest_path_rates

speciation_times &lt;- readTrace(speciation_time_file, burnin = 0.25)
speciation_rate &lt;- readTrace(speciation_rate_file, burnin = 0.25)

calculateShiftBayesFactor(speciation_rate,
                          speciation_times,
                          "speciation",
                          "interval_times",
                          0.0,40.0,
                          decrease=FALSE)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_times, dest_path_rates)


</code></pre>

<hr>
<h2 id='colFun'>Color Function</h2><span id='topic+colFun'></span>

<h3>Description</h3>

<p>Produce default RevGadgets colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colFun(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colFun_+3A_n">n</code></td>
<td>
<p>(integer; no default) Number of colors to return. Maximum of 12.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces a vector of colors from the default RevGadgets colors
of length given by n, maximum of 12 colors.
</p>


<h3>Value</h3>

<p>Character vector of color hex codes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my_colors &lt;- colFun(2)

</code></pre>

<hr>
<h2 id='combineTraces'>Combine traces</h2><span id='topic+combineTraces'></span>

<h3>Description</h3>

<p>Combine traces into one trace file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineTraces(traces, burnin = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combineTraces_+3A_traces">traces</code></td>
<td>
<p>(list of data frames; no default) Name of a list of data
frames, such as produced by readTrace().</p>
</td></tr>
<tr><td><code id="combineTraces_+3A_burnin">burnin</code></td>
<td>
<p>(single numeric value; default = 0.0) Fraction of generations
to discard (if value provided is between 0 and 1) or number of generations
to discard (if value provided is greater than 1) before combining the
samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Combines multiple traces from independent MCMC replicates
into one trace file.
</p>


<h3>Value</h3>

<p>combineTraces() returns a list of data frames of length 1,
corresponding to the combination of the provided samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


#' # download the example dataset to working directory
url_1 &lt;-
"https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_run_1.log"
dest_path_1 &lt;- "primates_cytb_GTR_run_1.log"
download.file(url_1, dest_path_1)

url_2 &lt;-
"https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_run_2.log"
dest_path_2 &lt;- "primates_cytb_GTR_run_2.log"
download.file(url_2, dest_path_2)

# to run on your own data, change this to the path to your data file
file_1 &lt;- dest_path_1
file_2 &lt;- dest_path_2

# read in the multiple trace files
multi_trace &lt;- readTrace(path = c(file_1, file_2), burnin = 0.0)

# combine samples after discarding 10% burnin
combined_trace &lt;- combineTraces(trace = multi_trace,
                                burnin = 0.1)

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_1, dest_path_2)



</code></pre>

<hr>
<h2 id='densiTreeWithBranchData'>DensiTree-style plot with branch-specific data</h2><span id='topic+densiTreeWithBranchData'></span>

<h3>Description</h3>

<p>This function plots a distribution of trees (e.g obtained from an
MCMC inference) with branch-specific rates or other data.
The plot is similar to those produced by DensiTree, i.e all the trees are
overlapped with each other. The data is expected to be given per node,
and will be associated with the branch above its corresponding node.
Its values are plotted as a color gradient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>densiTreeWithBranchData(
  tree_files = NULL,
  burnin = 0.1,
  trees = NULL,
  data = NULL,
  data_name = NULL,
  type = "cladogram",
  consensus = NULL,
  direction = "rightwards",
  scaleX = FALSE,
  width = 1,
  lty = 1,
  cex = 0.8,
  font = 3,
  tip.color = 1,
  adj = 0,
  srt = 0,
  keep_underscores = FALSE,
  label_offset = 0.01,
  scale_bar = TRUE,
  jitter = list(amount = 0, random = TRUE),
  color_gradient = c("red", "yellow", "green"),
  alpha = NULL,
  bias = 1,
  data_intervals = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="densiTreeWithBranchData_+3A_tree_files">tree_files</code></td>
<td>
<p>vector of tree files in NEXUS format with data attached
to the branches/nodes of the tree. All trees should have the same tip
labels (the order can change). Either <code>tree_files</code> or both
<code>trees</code> and <code>data</code> have to be specified.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_burnin">burnin</code></td>
<td>
<p>fraction of samples to discard from the tree files as burn-in.
Default 0.1.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_trees">trees</code></td>
<td>
<p>multiPhylo object or list of trees in phylo format.
All trees should have the same tip labels (the order can change). Either
<code>tree_files</code> or both <code>trees</code> and <code>data</code> have to be specified.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_data">data</code></td>
<td>
<p>data to be plotted on the tree - expected to be a list of
vectors in the same order as the trees, each vector in the order of the
tips and nodes of the corresponding tree</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_data_name">data_name</code></td>
<td>
<p>Only used when reading from <code>tree_files</code>.
Name of the data to be plotted, if multiple are present.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_type">type</code></td>
<td>
<p>character string specifying the type of phylogeny.
Options are &quot;cladogram&quot; (default) or &quot;phylogram&quot;.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_consensus">consensus</code></td>
<td>
<p>A tree or character vector which is used to define
the order of the tip labels. If NULL will be calculated from the trees.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_direction">direction</code></td>
<td>
<p>a character string specifying the direction of the tree.
Options are &quot;rightwards&quot; (default), &quot;leftwards&quot;, &quot;upwards&quot; and &quot;downwards&quot;.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_scalex">scaleX</code></td>
<td>
<p>whether to scale trees to have identical heights.
Default FALSE.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_width">width</code></td>
<td>
<p>width of the tree edges.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_lty">lty</code></td>
<td>
<p>line type of the tree edges.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_cex">cex</code></td>
<td>
<p>a numeric value giving the factor scaling of the tip labels.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_font">font</code></td>
<td>
<p>an integer specifying the type of font for the labels: 1
(plain text), 2 (bold), 3 (italic, the default), or 4 (bold italic).</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_tip.color">tip.color</code></td>
<td>
<p>color of the tip labels.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_adj">adj</code></td>
<td>
<p>a numeric specifying the justification of the text strings of
the tip labels: 0 (left-justification), 0.5 (centering), or 1
(right-justification).</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_srt">srt</code></td>
<td>
<p>a numeric giving how much the labels are rotated in degrees.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_keep_underscores">keep_underscores</code></td>
<td>
<p>whether the underscores in tip labels should be
written as spaces (the default) or left as they are (if TRUE).</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_label_offset">label_offset</code></td>
<td>
<p>a numeric giving the space between the nodes and the
tips of the phylogeny and their corresponding labels.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_scale_bar">scale_bar</code></td>
<td>
<p>whether to add a scale bar to the plot. Default TRUE.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_jitter">jitter</code></td>
<td>
<p>controls whether to shift trees. a list with two arguments:
the amount of jitter and random or equally spaced (see details below)</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_color_gradient">color_gradient</code></td>
<td>
<p>range of colors to be used for the data, in order of
increasing values. Defaults to red to yellow to green.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_alpha">alpha</code></td>
<td>
<p>transparency parameter for tree colors. If NULL will be set
based on the number of trees.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_bias">bias</code></td>
<td>
<p>bias applied to the color gradient. See
<code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code> for more details.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_data_intervals">data_intervals</code></td>
<td>
<p>value intervals used for the color gradient.
Can be given as a vector of interval boundaries or min and max values.If
NULL will be set based on the data.</p>
</td></tr>
<tr><td><code id="densiTreeWithBranchData_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If no consensus tree is provided, a consensus tree will be computed.
This should avoid too many unnecessary crossings of edges.
Trees should be rooted, other wise the output may not be visually pleasing.
The <code>jitter</code> parameter controls whether to shift trees so that
they are not exactly on top of each other.
If <code>amount = 0</code>, no jitter is applied. If <code>random = TRUE</code>,
the applied jitter is calculated as <code>runif(n, -amount, amount)</code>,
otherwise <code>seq(-amount, amount, length=n)</code>, where <code>n</code>
is the number of trees.
</p>


<h3>Value</h3>

<p>No return value, produces plot in base R
</p>


<h3>References</h3>

<p>This code is adapted from the <code><a href="phangorn.html#topic+densiTree">densiTree</a></code>
function by Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>.
densiTree is inspired from the
<a href="https://www.cs.auckland.ac.nz/~remco/DensiTree/">DensiTree</a>
program by Remco Bouckaert.
</p>
<p>Remco R. Bouckaert (2010) DensiTree: making sense of sets of phylogenetic
trees <em>Bioinformatics</em>, <b>26 (10)</b>, 1372-1373.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# generate random trees &amp; data
trees &lt;- lapply(1:5, function(x) ape::rcoal(5))
data &lt;- lapply(1:5, function(x) stats::runif(9, 1, 10))

# densiTree plot
densiTreeWithBranchData(trees = trees, data = data, width = 2)

# densiTree plot with different colors
densiTreeWithBranchData(trees = trees, data = data,
                        color_gradient = c("green", "blue"), width = 2)

</code></pre>

<hr>
<h2 id='dropTip'>dropTip</h2><span id='topic+dropTip'></span>

<h3>Description</h3>

<p>Drop one or multiple tips from your tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dropTip(tree, tips)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dropTip_+3A_tree">tree</code></td>
<td>
<p>(list of lists of treedata objects; no default) Name of a list of
lists of treedata objects, such as produced by readTrees().</p>
</td></tr>
<tr><td><code id="dropTip_+3A_tips">tips</code></td>
<td>
<p>(character or numeric, no default) The tips(s) to drop. Either a
single taxon name or node number or vector of such.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Modifies a tree object (in RevGadget's format) by dropping one or more tips
from the tree and from any associated data. Wrapper for treeio::drop.tip().
</p>


<h3>Value</h3>

<p>returns a list of list of treedata objects, with the modified tips.
</p>


<h3>See Also</h3>

<p>treeio: <a href="treeio.html#topic+drop.tip">drop.tip</a> and ape: <a href="ape.html#topic+drop.tip">drop.tip</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file &lt;- system.file("extdata",
                    "sub_models/primates_cytb_GTR_MAP.tre",
                    package="RevGadgets")
tree &lt;- readTrees(paths = file)
tree_dropped &lt;- dropTip(tree, "Otolemur_crassicaudatus")


</code></pre>

<hr>
<h2 id='geom_stepribbon'>plot geom stepribbon for diversification rates</h2><span id='topic+geom_stepribbon'></span><span id='topic+GeomStepribbon'></span>

<h3>Description</h3>

<p>Modified from <code>RmcdrPlugin.KMggplot2</code> step ribbon plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_stepribbon(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geom_stepribbon_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this
layer, either as a <code>ggproto</code> <code>Geom</code> subclass or as a string naming the
stat stripped of the <code>stat_</code> prefix (e.g. <code>"count"</code> rather than
<code>"stat_count"</code>)</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_stepribbon_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>geom_stepribbon</code> is an extension of the <code>geom_ribbon</code>, and
is optimized for Kaplan-Meier plots with pointwise confidence intervals
or a confidence band.
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+geom_ribbon">geom_ribbon</a></code> <code>geom_stepribbon</code>
inherits from <code>geom_ribbon</code>.
<code>geom_stepribbon</code> is modified from
<code>RcmdrPlugin.KMggplot2::geom_stepribbon</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
huron &lt;- data.frame(year = 1875:1972, level = as.vector(LakeHuron))

h &lt;- ggplot2::ggplot(huron, ggplot2::aes(year))

h + geom_stepribbon(ggplot2::aes(ymin = level - 1, ymax = level + 1),
                    fill = "grey70") +
    ggplot2::geom_step(ggplot2::aes(y = level))

# contrast ggplot2::geom_ribbon with geom_stepribbon:
h + ggplot2::geom_ribbon(ggplot2::aes(ymin = level - 1, ymax = level + 1),
                         fill = "grey70") +
    ggplot2::geom_line(ggplot2::aes(y = level))

</code></pre>

<hr>
<h2 id='getMAP'>get MAP</h2><span id='topic+getMAP'></span>

<h3>Description</h3>

<p>Calculates the Maximum a Posteriori estimate for the trace of a
quantitative variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMAP(var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMAP_+3A_var">var</code></td>
<td>
<p>(numeric vector; no default) Vector of the samples from the
trace of a quantitative variable</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses the SANN method of the optim() function to approximate the MAP estimate
</p>


<h3>Value</h3>

<p>the MAP estimate
</p>


<h3>See Also</h3>

<p><a href="stats.html#topic+optim">optim</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example dataset to working directory
url &lt;-
  "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR.log"
dest_path &lt;- "primates_cytb_GTR.log"
download.file(url, dest_path)

# to run on your own data, change this to the path to your data file
file &lt;- dest_path

trace &lt;- readTrace(paths = file)
MAP &lt;- getMAP(trace[[1]]$"pi[1]")

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path)


</code></pre>

<hr>
<h2 id='matchNodes'>match Nodes</h2><span id='topic+matchNodes'></span>

<h3>Description</h3>

<p>match Nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchNodes(phy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matchNodes_+3A_phy">phy</code></td>
<td>
<p>(tree in ape format; no default) Tree on which to match nodes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame that translates ape node numbers to RevBayes node
numbers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
treefile &lt;- system.file("extdata", "bds/primates.tre", package="RevGadgets")
tree &lt;- readTrees(treefile)
map &lt;- matchNodes(tree[[1]][[1]]@phylo)

</code></pre>

<hr>
<h2 id='plotAncStatesMAP'>plot Ancestral States MAP</h2><span id='topic+plotAncStatesMAP'></span>

<h3>Description</h3>

<p>Plots the MAP estimates of ancestral states. Can accommodate cladogenetic
reconstructions by plotting on shoulders. Defaults to varying the symbols by
color to indicate estimated ancestral state and varying the size of the
symbol to indicate the posterior probability of that estimate, but symbol
shape may also vary to accommodate black and white figures. For more details
on the aesthetics options, see parameter details below. For data with many
character states (such as chromosome counts), vary the size of the symbol
by estimated ancestral state, and vary the posterior probability of that
estimate by a color gradient. Text labels at nodes and tips are also
available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAncStatesMAP(
  t,
  cladogenetic = FALSE,
  tip_labels = TRUE,
  tip_labels_size = 2,
  tip_labels_offset = 1,
  tip_labels_italics = FALSE,
  tip_labels_formatted = FALSE,
  tip_labels_remove_underscore = TRUE,
  tip_labels_states = FALSE,
  tip_labels_states_size = 2,
  tip_labels_states_offset = 0.1,
  node_labels_as = NULL,
  node_labels_size = 2,
  node_labels_offset = 0.1,
  node_labels_centered = FALSE,
  node_size_as = "state_posterior",
  node_color_as = "state",
  node_shape_as = NULL,
  node_shape = 19,
  node_color = "default",
  node_size = c(2, 6),
  tip_states = TRUE,
  tip_states_size = node_size,
  tip_states_shape = node_shape,
  state_transparency = 0.75,
  tree_layout = "rectangular",
  timeline = FALSE,
  geo = timeline,
  geo_units = list("epochs", "periods"),
  time_bars = timeline,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAncStatesMAP_+3A_t">t</code></td>
<td>
<p>(treedata object; none) Output of processAncStates() function
containing tree and ancestral states.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_cladogenetic">cladogenetic</code></td>
<td>
<p>(logical; FALSE) Plot shoulder states of cladogenetic
analyses?</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels">tip_labels</code></td>
<td>
<p>(logical; TRUE) Label taxa labels at tips?</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_size">tip_labels_size</code></td>
<td>
<p>(numeric; 2) Size of tip labels.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_offset">tip_labels_offset</code></td>
<td>
<p>(numeric; 1) Horizontal offset of tip labels from
tree.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_italics">tip_labels_italics</code></td>
<td>
<p>(logical; FALSE) Italicize tip labels?</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_formatted">tip_labels_formatted</code></td>
<td>
<p>(logical; FALSE) Do the tip labels contain 
manually added formatting information? Will set parse = TRUE in geom_text()
and associated functions to interpret formatting. See ?plotmath for more.
Cannot be TRUE if tip_labels_italics = TRUE.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_remove_underscore">tip_labels_remove_underscore</code></td>
<td>
<p>(logical; TRUE) Remove underscores from
tip labels?</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_states">tip_labels_states</code></td>
<td>
<p>(logical; FALSE) Optional plotting of text at tips
in addition
to taxa labels.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_states_size">tip_labels_states_size</code></td>
<td>
<p>(numeric; 2) Size of state labels at tips.
Ignored if
tip_labels_states is FALSE.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_labels_states_offset">tip_labels_states_offset</code></td>
<td>
<p>(numeric; 0.1) Horizontal offset of tip
state labels.
Ignored if tip_labels_states = NULL.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_labels_as">node_labels_as</code></td>
<td>
<p>(character; NULL) Optional plotting of text at nodes.
Possible values are &quot;state&quot; for the ancestral states , &quot;state_posterior&quot;
for posterior probabilities of the estimated ancestral state,
&quot;node_posterior&quot; or the posterior probability of the node on the tree,
or NULL for not plotting any text at the nodes (default).</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_labels_size">node_labels_size</code></td>
<td>
<p>(numeric; 2) Size of node labels text. Ignored if
node_labels_as = NULL.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_labels_offset">node_labels_offset</code></td>
<td>
<p>(numeric; 0.1) Horizontal offset of node labels
from nodes. Ignored if node_labels_as = NULL.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_labels_centered">node_labels_centered</code></td>
<td>
<p>(logical; FALSE) Should node labels be centered
over the nodes? Defaults to FALSE: adjusting node labels to the right of
nodes and left of shoulders.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_size_as">node_size_as</code></td>
<td>
<p>(character; &quot;state_posterior&quot;) How to vary size of
node symbols. Options are &quot;state_posterior&quot; (default) for posterior
probabilities of the estimated ancestral state, &quot;node_posterior&quot; or the
posterior probability of the node on the tree, &quot;state&quot; for vary size by the
ancestral state itself in cases where there are many character states
(e.g. chromosome numbers; we do not recommend this option for characters
with few states), or NULL for fixed symbol size.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_color_as">node_color_as</code></td>
<td>
<p>(character; &quot;state&quot;) How to vary to color of node
symbols. Options are &quot;state&quot; (default) to vary by estimated ancestral states,
&quot;state_posterior&quot; for posterior probabilities of the estimated ancestral
state, &quot;node_posterior&quot; or the posterior probability of the node on the tree,
or NULL to set all as one color.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_shape_as">node_shape_as</code></td>
<td>
<p>(character; NULL) Option to vary node symbol by shape.
Options are NULL to keep shape constant or &quot;state&quot; to vary shape by
ancestral state.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_shape">node_shape</code></td>
<td>
<p>(integer; 19) Shape type for nodes. If node_shape_as =
&quot;state&quot;, provide a vector with length of the number of states. See ggplot2
documentation for details:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point</a></p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_color">node_color</code></td>
<td>
<p>(&quot;character&quot;; &quot;default&quot;) Colors for node symbols. Defaults
to default RevGadgets colors. If node_color_as = &quot;state', provide a vector of
length of the character states. If your color vector is labeled with state
labels, the legend will be displayed in the order of the labels. If
node_color_as = &quot;posterior&quot;, provide a vector of length 2 to generate a
color gradient.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_node_size">node_size</code></td>
<td>
<p>(numeric; c(2, 6)) Range of sizes, or fixed size, for node
symbols. If node_size_as = &quot;state_posterior&quot;, &quot;node_posterior&quot;, or &quot;state&quot;,
numeric vector of length two. If node_size_as = NULL, numeric vector of
length one. Size regulates the area of the symbol, following ggplot2 best
practices: <a href="https://ggplot2.tidyverse.org/reference/scale_size.html">https://ggplot2.tidyverse.org/reference/scale_size.html</a>)</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_states">tip_states</code></td>
<td>
<p>(logical; TRUE) Plot states of taxa at tips?</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_states_size">tip_states_size</code></td>
<td>
<p>(numeric; node_size) Size for tip symbols. Defaults
to the same size as node symbols.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tip_states_shape">tip_states_shape</code></td>
<td>
<p>(integer; node_shape) Shape for tip symbols.
Defaults to the same as node symbols.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_state_transparency">state_transparency</code></td>
<td>
<p>(integer; 0.75) Alpha (transparency) of state
symbols- varies from 0 to 1.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_tree_layout">tree_layout</code></td>
<td>
<p>(character; &quot;rectangular&quot;) Tree shape layout, passed to
ggtree(). Options are 'rectangular', 'slanted', 'ellipse', 'roundrect',
'fan', 'circular', 'inward_circular', 'radial', 'equal_angle', 'daylight'
or 'ape'. When cladogenetic = TRUE, only &quot;rectangular&quot; and 'circular' are
available.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_timeline">timeline</code></td>
<td>
<p>(logical; FALSE) Plot time tree with labeled x-axis with
timescale in MYA.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_geo">geo</code></td>
<td>
<p>(logical; timeline) Add a geological timeline? Defaults to the
same as timeline.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_geo_units">geo_units</code></td>
<td>
<p>(list; list(&quot;epochs&quot;, &quot;periods&quot;)) Which geological units to
include in the geo timescale. May be &quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, 
&quot;eras&quot;, or a list of two of those units.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_time_bars">time_bars</code></td>
<td>
<p>(logical; timeline) Add vertical gray bars to indicate
geological timeline units if geo == TRUE or regular time intervals (in MYA)
if geo == FALSE.</p>
</td></tr>
<tr><td><code id="plotAncStatesMAP_+3A_...">...</code></td>
<td>
<p>(various) Additional arguments passed to ggtree::ggtree().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Standard ancestral state reconstruction example with various aesthetics

# process file
file &lt;- system.file("extdata",
                    "comp_method_disc/ase_freeK.tree",
                    package="RevGadgets")
example &lt;- processAncStates(file,
                            state_labels = c("1" = "Awesome",
                                             "2" = "Beautiful",
                                             "3" = "Cool!"))

# have states vary by color and indicate state pp with size (default)
plotAncStatesMAP(t = example)

# have states vary by color and indicate state pp with size ,
# and add a timeline
plotAncStatesMAP(t = example, timeline = TRUE)

# have states vary by color and symbol, label nodes with pp of states
plotAncStatesMAP(t = example,  node_shape_as = "state",
                 node_size = 4, node_shape = c(15, 17,20),
                 node_size_as = NULL, node_labels_as = "state_posterior")

# black and white figure - state as symbols and state pp with text
plotAncStatesMAP(t = example, node_color_as = NULL,
                 node_shape_as = "state", node_shape =  c(15, 17,20),
                 node_size_as = NULL, node_size = 4,
                 node_labels_as = "state_posterior",
                 node_color = "grey", state_transparency = 1)

# default with circular tree
plotAncStatesMAP(t = example, tree_layout = "circular")


# Chromosome evolution example

# process file
file &lt;- system.file("extdata",
                    "chromo/ChromEvol_simple_final.tree",
                    package="RevGadgets")
chromo_example &lt;- processAncStates(file, labels_as_numbers = TRUE)

# plot
plotAncStatesMAP(t = chromo_example, node_color_as = "state_posterior",
                 node_size_as = "state", node_color = colFun(2),
                 tip_labels_offset = 0.005, node_labels_as = "state",
                 node_labels_offset = 0, tip_labels_states = TRUE,
                 tip_labels_states_offset = 0, tip_states = FALSE)

# DEC example (cladogenetic)

# process file
file &lt;- system.file("extdata", "dec/simple.ase.tre", package="RevGadgets")
labs &lt;- c("1" = "K", "2" = "O", "3" = "M", "4" = "H", "5" = "KO",
"6" = "KM", "7" = "OM", "8" = "KH", "9" = "OH", "10" = "MH", "11" = "KOM",
"12" = "KOH", "13" = "KMH", "14" = "OMH", "15" = "KOMH")
dec_example &lt;- processAncStates(file, state_labels = labs)

# plot
plotAncStatesMAP(t = dec_example,
                 cladogenetic = TRUE,
                 tip_labels_offset = 0.5)


</code></pre>

<hr>
<h2 id='plotAncStatesPie'>plot Ancestral States Pie</h2><span id='topic+plotAncStatesPie'></span>

<h3>Description</h3>

<p>Plot character states and posterior probabilities as pies on nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAncStatesPie(
  t,
  cladogenetic = FALSE,
  tip_labels = TRUE,
  tip_labels_size = 2,
  tip_labels_offset = 1,
  tip_labels_italics = FALSE,
  tip_labels_formatted = FALSE,
  tip_labels_remove_underscore = TRUE,
  tip_labels_states = FALSE,
  tip_labels_states_size = 2,
  tip_labels_states_offset = 0.1,
  node_labels_as = NULL,
  node_labels_size = 2,
  node_labels_offset = 0.1,
  pie_colors = "default",
  node_pie_size = 1,
  shoulder_pie_size = node_pie_size,
  tip_pies = TRUE,
  tip_pie_size = 0.5,
  node_pie_nudge_x = 0,
  node_pie_nudge_y = 0,
  tip_pie_nudge_x = node_pie_nudge_x,
  tip_pie_nudge_y = node_pie_nudge_y,
  shoulder_pie_nudge_x = node_pie_nudge_x,
  shoulder_pie_nudge_y = node_pie_nudge_y,
  state_transparency = 0.75,
  timeline = FALSE,
  geo = timeline,
  geo_units = list("epochs", "periods"),
  time_bars = timeline,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAncStatesPie_+3A_t">t</code></td>
<td>
<p>(treedata object; none) Output of processAncStates() function
containing tree and ancestral states.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_cladogenetic">cladogenetic</code></td>
<td>
<p>(logical; FALSE) Plot shoulder pies of cladogenetic
analyses?</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels">tip_labels</code></td>
<td>
<p>(logical; TRUE) Label taxa labels at tips?</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_size">tip_labels_size</code></td>
<td>
<p>(numeric; 2) Size of tip labels.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_offset">tip_labels_offset</code></td>
<td>
<p>(numeric; 1) Horizontal offset of tip labels from
tree.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_italics">tip_labels_italics</code></td>
<td>
<p>(logical; FALSE) Italicize tip labels?</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_formatted">tip_labels_formatted</code></td>
<td>
<p>(logical; FALSE) Do the tip labels contain 
manually added formatting information? Will set parse = TRUE in geom_text()
and associated functions to interpret formatting. See ?plotmath for more.
Cannot be TRUE if tip_labels_italics = TRUE.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_remove_underscore">tip_labels_remove_underscore</code></td>
<td>
<p>(logical; TRUE) Remove underscores from
tip labels?</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_states">tip_labels_states</code></td>
<td>
<p>(logical; FALSE) Optional plotting of text at tips
in addition to taxa labels. Will plot the MAP state label.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_states_size">tip_labels_states_size</code></td>
<td>
<p>(numeric; 2) Size of state labels at tips.
Ignored if tip_labels_states is FALSE.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_labels_states_offset">tip_labels_states_offset</code></td>
<td>
<p>(numeric; 0.1) Horizontal offset of tip
state labels. Ignored if tip_labels_states = NULL.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_node_labels_as">node_labels_as</code></td>
<td>
<p>(character; NULL) Optional plotting of text at nodes.
Possible values are &quot;state&quot; for the MAP ancestral states, &quot;node_posterior&quot;
for the posterior probability of the node on the tree, “state_posterior” 
for the posterior probability of the MAP, or NULL for not plotting any
text at the nodes (default).</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_node_labels_size">node_labels_size</code></td>
<td>
<p>(numeric; 2) Size of node labels text. Ignored if
node_labels_as = NULL.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_node_labels_offset">node_labels_offset</code></td>
<td>
<p>(numeric; 0.1) Horizontal offset of node labels
from nodes. Ignored if node_labels_as = NULL.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_pie_colors">pie_colors</code></td>
<td>
<p>(&quot;character&quot;; &quot;default&quot;) Colors for states in pies.
If &quot;default&quot;, plots the default RevGadgets colors. Provide a character
vector of hex codes or other R-readable colors the same length of the number
of character states. Names of the vector should correspond to state labels.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_node_pie_size">node_pie_size</code></td>
<td>
<p>(numeric; 1) Size (diameter) of the pies at nodes.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_shoulder_pie_size">shoulder_pie_size</code></td>
<td>
<p>(numeric; node_pie_size) Size (diameter) of the
pies at shoulders for cladogenetic plots.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_pies">tip_pies</code></td>
<td>
<p>(logical; TRUE) Plot pies tips?</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_pie_size">tip_pie_size</code></td>
<td>
<p>(numeric; 0.5) Size (diameter) of the pies at tips.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_node_pie_nudge_x">node_pie_nudge_x</code></td>
<td>
<p>(numeric; 0) If pies aren't centered, adjust by
nudging</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_node_pie_nudge_y">node_pie_nudge_y</code></td>
<td>
<p>(numeric; 0) If pies aren't centered, adjust by
nudging</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_pie_nudge_x">tip_pie_nudge_x</code></td>
<td>
<p>(numeric; node_pie_nudge_x) If pies aren't centered,
adjust by nudging</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_tip_pie_nudge_y">tip_pie_nudge_y</code></td>
<td>
<p>(numeric; node_pie_nudge_y) If pies aren't centered,
adjust by nudging</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_shoulder_pie_nudge_x">shoulder_pie_nudge_x</code></td>
<td>
<p>(numeric; node_pie_nudge_x) If pies aren't
centered, adjust by nudging</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_shoulder_pie_nudge_y">shoulder_pie_nudge_y</code></td>
<td>
<p>(numeric; node_pie_nudge_y) If pies aren't
centered, adjust by nudging</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_state_transparency">state_transparency</code></td>
<td>
<p>(integer; 0.75) Alpha (transparency) of state
symbols- varies from 0 to 1.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_timeline">timeline</code></td>
<td>
<p>(logical; FALSE) Plot tree with labeled x-axis with
timescale in MYA.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_geo">geo</code></td>
<td>
<p>(logical; timeline) Add a geological timeline? Defaults to the
same as timeline.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_geo_units">geo_units</code></td>
<td>
<p>(list; list(&quot;epochs&quot;, &quot;periods&quot;)) Which geological units to
include in the geo timescale. May be &quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, 
&quot;eras&quot;, or a list of two of those units.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_time_bars">time_bars</code></td>
<td>
<p>(logical; timeline) Add vertical gray bars to indicate
geological timeline units if geo == TRUE or regular time intervals (in MYA)
if geo == FALSE.</p>
</td></tr>
<tr><td><code id="plotAncStatesPie_+3A_...">...</code></td>
<td>
<p>(various) Additional arguments passed to ggtree::ggtree().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# Standard ancestral state reconstruction example

# process file and assign state labels
file &lt;- system.file("extdata",
                    "comp_method_disc/ase_freeK.tree",
                    package="RevGadgets")
example &lt;- processAncStates(file,
                            state_labels = c("1" = "Awesome",
                                             "2" = "Beautiful",
                                             "3" = "Cool!"))
# plot (this may take a while)
plotAncStatesPie(t = example)

# DEC Biogeographic range evolution example (with timeline)

# process file
file &lt;- system.file("extdata", "dec/simple.ase.tre", package="RevGadgets")

# labels that correspond to each region/ possible combination of regions
labs &lt;- c("1" = "K", "2" = "O", "3" = "M", "4" = "H", "5" = "KO",
          "6" = "KM", "7" = "OM", "8" = "KH", "9" = "OH", "10" = "MH",
          "11" = "KOM", "12" = "KOH", "13" = "KMH", "14" = "OMH",
          "15" = "KOMH")
dec_example  &lt;- processAncStates(file , state_labels = labs)
# Use the state_labels in returned tidytree object to define color palette
# These state_labels may be a subset of the labels you provided
# (not all possible regions may be sampled in the dataset)
colors &lt;- colorRampPalette(colFun(12))(length(dec_example@state_labels))
names(colors) &lt;- dec_example@state_labels

# plot
plotAncStatesPie(t = dec_example, pie_colors = colors, tip_labels_size = 3,
        cladogenetic = TRUE, tip_labels_offset = 0.25, timeline = TRUE,
        geo = FALSE) +
        ggplot2::theme(legend.position = c(0.1, 0.75))


</code></pre>

<hr>
<h2 id='plotDiversityOBDP'>Plot Diversity Distribution from OBDP Analysis</h2><span id='topic+plotDiversityOBDP'></span>

<h3>Description</h3>

<p>Plots the probability distribution of the number of lineages through time inferred with the 
Occurrence Birth Death Process
#'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDiversityOBDP(
  Kt_mean,
  xlab = "Time",
  ylab = "Number of lineages",
  xticks_n_breaks = 5,
  col_Hidden = "dodgerblue3",
  col_LTT = "gray25",
  col_Total = "forestgreen",
  col_Hidden_interval = "dodgerblue2",
  col_Total_interval = "darkolivegreen4",
  palette_Hidden = c("transparent", "dodgerblue2", "dodgerblue3", "dodgerblue4", "black"),
  palette_Total = c("transparent", "green4", "forestgreen", "black"),
  line_size = 0.7,
  interval_line_size = 0.5,
  show_Hidden = TRUE,
  show_LTT = TRUE,
  show_Total = TRUE,
  show_intervals = TRUE,
  show_densities = TRUE,
  show_expectations = TRUE,
  use_interpolate = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDiversityOBDP_+3A_kt_mean">Kt_mean</code></td>
<td>
<p>(data.frame; no default) The processed data.frame (output of readOBDP()).</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_xlab">xlab</code></td>
<td>
<p>(character; &quot;Time&quot;) The label of the x-axis.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_ylab">ylab</code></td>
<td>
<p>(character; &quot;Number of lineages&quot;) The label of the y-axis.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_xticks_n_breaks">xticks_n_breaks</code></td>
<td>
<p>(numeric; 5) An integer guiding the number of major breaks.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_col_hidden">col_Hidden</code></td>
<td>
<p>(character; &quot;dodgerblue3&quot;) The color of the hidden lineages plot line.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_col_ltt">col_LTT</code></td>
<td>
<p>(character; &quot;gray25&quot;) The color of the LTT plot line.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_col_total">col_Total</code></td>
<td>
<p>(character; &quot;forestgreen&quot;) The color of the total lineages plot line.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_col_hidden_interval">col_Hidden_interval</code></td>
<td>
<p>(character; &quot;dodgerblue2&quot;) The color of the credible interval lines around the hidden lineages plot.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_col_total_interval">col_Total_interval</code></td>
<td>
<p>(character; &quot;darkolivegreen4&quot;) The color of the credible interval lines around the total lineages plot.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_palette_hidden">palette_Hidden</code></td>
<td>
<p>(character; c(&quot;transparent&quot;, &quot;dodgerblue2&quot;, &quot;dodgerblue3&quot;, &quot;dodgerblue4&quot;, &quot;black&quot;)) The palette of the hidden lineages plot distribution.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_palette_total">palette_Total</code></td>
<td>
<p>(character; c(&quot;transparent&quot;, &quot;green4&quot;, &quot;forestgreen&quot;, &quot;black&quot;)) The palette of the total lineages plot distribution.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_line_size">line_size</code></td>
<td>
<p>(numeric; 0.7) The width of the lineage plot line.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_interval_line_size">interval_line_size</code></td>
<td>
<p>(numeric; 0.5) The width of the credible interval.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_show_hidden">show_Hidden</code></td>
<td>
<p>(boolean; TRUE) Whether to show the plot for hidden lineages.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_show_ltt">show_LTT</code></td>
<td>
<p>(boolean; TRUE) Whether to show the plot for observed lineages.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_show_total">show_Total</code></td>
<td>
<p>(boolean; TRUE) Whether to show the plot for total lineages.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_show_intervals">show_intervals</code></td>
<td>
<p>(boolean; TRUE) Whether to show the credible intervals.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_show_densities">show_densities</code></td>
<td>
<p>(boolean; TRUE) Whether to show the diversity densities.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_show_expectations">show_expectations</code></td>
<td>
<p>(boolean; TRUE) Whether to show the diversity expectations.</p>
</td></tr>
<tr><td><code id="plotDiversityOBDP_+3A_use_interpolate">use_interpolate</code></td>
<td>
<p>(boolean; TRUE) Whether to interpolate densities.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# first run readOBDP()
start_time_trace_file &lt;- 
     system.file("extdata", "obdp/start_time_trace.p", package="RevGadgets")
popSize_distribution_matrices_file &lt;- 
     system.file("extdata", "obdp/Kt_trace.p", package="RevGadgets")
trees_trace_file &lt;- 
     system.file("extdata", "obdp/mcmc_OBDP_trees.p", package="RevGadgets")
    
Kt_mean &lt;- readOBDP( start_time_trace_file=start_time_trace_file, 
                     popSize_distribution_matrices_file=popSize_distribution_matrices_file, 
                     trees_trace_file=trees_trace_file )

# then get the customized ggplot object with plotDiversityOBDP()
p &lt;- plotDiversityOBDP( Kt_mean,
                        xlab="Time (My)",
                        ylab="Number of lineages",
                        xticks_n_breaks=21,
                        col_Hidden="dodgerblue3",
                        col_LTT="gray25",
                        col_Total="forestgreen",
                        col_Hidden_interval="dodgerblue2",
                        col_Total_interval="darkolivegreen4",
                        palette_Hidden=c("transparent", "dodgerblue2", "dodgerblue3", 
                                         "dodgerblue4", "black"),
                        palette_Total=c("transparent", "green4", "forestgreen", "black"),
                        line_size=0.7,
                        interval_line_size=0.5,
                        show_Hidden=TRUE,
                        show_LTT=TRUE,
                        show_Total=TRUE,
                        show_intervals=TRUE,
                        show_densities=TRUE,
                        show_expectations=TRUE,
                        use_interpolate=TRUE )

# basic plot
p

# option: add a stratigraphic scale
library(deeptime)
library(ggplot2)
q &lt;- gggeo_scale(p, dat="periods", height=unit(1.3, "line"), abbrv=F, size=4.5, neg=T)
r &lt;- gggeo_scale(q, dat="epochs", height=unit(1.1, "line"), abbrv=F, size=3.5, neg=T, 
                    skip=c("Paleocene", "Pliocene", "Pleistocene", "Holocene"))
s &lt;- gggeo_scale(r, dat="stages", height=unit(1, "line"), abbrv=T, size=2.5, neg=T)
s

## End(Not run)

</code></pre>

<hr>
<h2 id='plotDivRates'>Plot Diversification Rates</h2><span id='topic+plotDivRates'></span>

<h3>Description</h3>

<p>Plots the output of an episodic diversification rate analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDivRates(rates, facet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDivRates_+3A_rates">rates</code></td>
<td>
<p>(list of dataframes; no default) A list of dataframes,
such as produced by processDivRates(), containing the data on rates
and interval times for each type of rate to be plotted (e.g.
speciation rate, etc.).</p>
</td></tr>
<tr><td><code id="plotDivRates_+3A_facet">facet</code></td>
<td>
<p>(logical; TRUE) plot rates in separate facets.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the output of episodic diversification rate analyses. Takes as
input the output of processDivRates() and plotting parameters.
For now, only variable names (under &quot;item&quot;) that contain the word &quot;rate&quot; are
included in the plot.
</p>
<p>The return object can be manipulated. For example, you can change the
axis labels, the color palette, whether the axes are to be linked, or the
overall plotting style/theme, just as with any ggplot object.
</p>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example datasets to working directory

url_ex_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_extinction_times.log"
dest_path_ex_times &lt;- "primates_EBD_extinction_times.log"
download.file(url_ex_times, dest_path_ex_times)

url_ex_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_extinction_rates.log"
dest_path_ex_rates &lt;- "primates_EBD_extinction_rates.log"
download.file(url_ex_rates, dest_path_ex_rates)

url_sp_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_times.log"
dest_path_sp_times &lt;- "primates_EBD_speciation_times.log"
download.file(url_sp_times, dest_path_sp_times)

url_sp_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_rates.log"
dest_path_sp_rates &lt;- "primates_EBD_speciation_rates.log"
download.file(url_sp_rates, dest_path_sp_rates)

# to run on your own data, change this to the path to your data file
speciation_time_file &lt;- dest_path_sp_times
speciation_rate_file &lt;- dest_path_sp_rates
extinction_time_file &lt;- dest_path_ex_times
extinction_rate_file &lt;- dest_path_ex_rates

rates &lt;- processDivRates(speciation_time_log = speciation_time_file,
                         speciation_rate_log = speciation_rate_file,
                         extinction_time_log = extinction_time_file,
                         extinction_rate_log = extinction_rate_file,
                         burnin = 0.25)

# then plot results:
p &lt;- plotDivRates(rates = rates);p

# change the x-axis
p &lt;- p + ggplot2::xlab("Thousands of years ago");p

# change the colors
p &lt;- p + ggplot2::scale_fill_manual(values = c("red",
                                               "green",
                                               "yellow",
                                               "purple")) +
  ggplot2::scale_color_manual(values = c("red",
                                         "green",
                                         "yellow",
                                         "purple"));p

# let's say we don't want to plot relative-extinction rate,
# and use the same y-axis for all three rates
rates &lt;- rates[!grepl("relative-extinction", rates$item),]
p2 &lt;- plotDivRates(rates)
p2 &lt;- p2 + ggplot2::facet_wrap(ggplot2::vars(item), scale = "fixed");p2

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_sp_times, dest_path_ex_times,
            dest_path_sp_rates, dest_path_ex_rates)



</code></pre>

<hr>
<h2 id='plotFBDTree'>Plot FBD tree</h2><span id='topic+plotFBDTree'></span>

<h3>Description</h3>

<p>Plots a single FBD tree, such as an MCC or MAP tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotFBDTree(
  tree,
  timeline = FALSE,
  geo = timeline,
  geo_units = list("epochs", "periods"),
  time_bars = timeline,
  node_age_bars = TRUE,
  tip_age_bars = TRUE,
  age_bars_color = "blue",
  age_bars_colored_by = NULL,
  age_bars_width = 1.5,
  node_labels = NULL,
  node_labels_color = "black",
  node_labels_size = 3,
  node_labels_offset = 0,
  tip_labels = TRUE,
  tip_labels_italics = FALSE,
  tip_labels_formatted = FALSE,
  tip_labels_remove_underscore = TRUE,
  tip_labels_color = "black",
  tip_labels_size = 3,
  tip_labels_offset = 0,
  node_pp = FALSE,
  node_pp_shape = 16,
  node_pp_color = "black",
  node_pp_size = "variable",
  branch_color = "black",
  color_branch_by = NULL,
  line_width = 1,
  label_sampled_ancs = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotFBDTree_+3A_tree">tree</code></td>
<td>
<p>(list of lists of treedata objects; no default) Name of a list
of lists of treedata objects, such as produced by readTrees(). This
object should only contain only one summary tree from one trace file. If it
contains multiple trees or multiple traces, only the first will be used.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_timeline">timeline</code></td>
<td>
<p>(logical; FALSE) Plot time tree with labeled x-axis with
timescale in MYA.
#'</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_geo">geo</code></td>
<td>
<p>(logical; timeline) Add a geological timeline? Defaults to the
same as timeline.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_geo_units">geo_units</code></td>
<td>
<p>(list; list(&quot;epochs&quot;, &quot;periods&quot;)) Which geological units to
include in the geo timescale. May be &quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, 
&quot;eras&quot;, or a list of two of those units.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_time_bars">time_bars</code></td>
<td>
<p>(logical; timeline) Add vertical gray bars to indicate
geological timeline units if geo == TRUE or regular time intervals (in MYA)
if geo == FALSE.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_age_bars">node_age_bars</code></td>
<td>
<p>(logical; TRUE) Plot time tree with node age bars?</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_age_bars">tip_age_bars</code></td>
<td>
<p>(logical; FALSE) Plot node age bars for the tips as
well? Useful for plotting serial sampled analyses or fossilized birth-death
analyses, or any cases where some tip ages are estimated.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_age_bars_color">age_bars_color</code></td>
<td>
<p>(character; &quot;blue&quot;) Color for node/tip age bars.
If age_bars_colored_by specifies a variable (not NULL), you must provide
two colors, low and high values for a gradient. Colors must be either R
valid color names or valid hex codes.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_age_bars_colored_by">age_bars_colored_by</code></td>
<td>
<p>(character; NULL) Specify column to color
node/tip age bars by, such as &quot;posterior&quot;. If null, all age bars plotted the
same color, specified by age_bars_color</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_age_bars_width">age_bars_width</code></td>
<td>
<p>(numeric; 1.5) Change line width for age bars</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_labels">node_labels</code></td>
<td>
<p>(character; NULL) Plot text labels at nodes, specified
by the name of the corresponding column in the tidytree object. If NULL,
no text is plotted.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_labels_color">node_labels_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color to plot node_labels,
either as a valid R color name or a valid hex code.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_labels_size">node_labels_size</code></td>
<td>
<p>(numeric; 3) Size of node labels</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_labels_offset">node_labels_offset</code></td>
<td>
<p>(numeric; 0) Horizontal offset of node labels
from nodes.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels">tip_labels</code></td>
<td>
<p>(logical; TRUE) Plot tip labels?</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels_italics">tip_labels_italics</code></td>
<td>
<p>(logical; FALSE) Plot tip labels in italics?</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels_formatted">tip_labels_formatted</code></td>
<td>
<p>(logical; FALSE) Do the tip labels contain 
manually added formatting information? Will set parse = TRUE in geom_text()
and associated functions to interpret formatting. See ?plotmath for more.
Cannot be TRUE if tip_labels_italics = TRUE.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels_remove_underscore">tip_labels_remove_underscore</code></td>
<td>
<p>(logical; FALSE) Should underscores be
replaced by spaces in tip labels?</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels_color">tip_labels_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color to plot tip labels,
either as a valid R color name or a valid hex code.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels_size">tip_labels_size</code></td>
<td>
<p>(numeric; 3) Size of tip labels</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_tip_labels_offset">tip_labels_offset</code></td>
<td>
<p>(numeric; 1) Horizontal offset of tip labels from
tree.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_pp">node_pp</code></td>
<td>
<p>(logical; FALSE) Plot posterior probabilities as symbols at
nodes? Specify symbol aesthetics with node_pp_shape, node_pp_color, and
node_pp_size.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_pp_shape">node_pp_shape</code></td>
<td>
<p>(integer; 1) Integer corresponding to point shape
(value between 0-25). See ggplot2 documentation for details:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point</a></p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_pp_color">node_pp_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color for node_pp symbols, either
as valid R color name(s) or hex code(s). Can be a single character string
specifying a single color, or a vector of length two specifying two colors
to form a gradient. In this case, posterior probabilities will be indicated
by color along the specified gradient.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_node_pp_size">node_pp_size</code></td>
<td>
<p>(numeric or character; 1) Size for node_pp symbols.
If numeric, the size will be fixed at the specified value. If a character,
it should specify &quot;variable&quot;, indicating that size should be scaled by the
posterior value. Size regulates the area of the shape,
following ggplot2 best practices:
<a href="https://ggplot2.tidyverse.org/reference/scale_size.html">https://ggplot2.tidyverse.org/reference/scale_size.html</a>)</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_branch_color">branch_color</code></td>
<td>
<p>(character; &quot;black&quot;) A single character string
specifying the color (R color name or hex code) for all branches OR a
vector of length 2 specifying two colors for a gradient, used to color the
branches according to the variable specified in color_branch_by. If only 1
color is provided and you specify color_branch_by, default colors will be
chosen (low = &quot;#005ac8&quot;, high = &quot;#fa7850&quot;).</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_color_branch_by">color_branch_by</code></td>
<td>
<p>(character; NULL ) Optional name of one quantitative
variable in the treedata object to color branches, such as a rate.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_line_width">line_width</code></td>
<td>
<p>(numeric; 1) Change line width for branches</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_label_sampled_ancs">label_sampled_ancs</code></td>
<td>
<p>(logical; FALSE) Label any sampled ancestors?
Will inherent tip labels aesthetics for size and color.</p>
</td></tr>
<tr><td><code id="plotFBDTree_+3A_...">...</code></td>
<td>
<p>(various) Additional arguments passed to ggtree::ggtree().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a single tree, such as an MCC or MAP tree, with
special features for plotting the output of fossilized
birth-death analyses.
</p>


<h3>Value</h3>

<p>returns a single plot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file &lt;- system.file("extdata", "fbd/bears.mcc.tre", package="RevGadgets")
tree &lt;- readTrees(paths = file)
plotFBDTree(tree = tree, timeline = TRUE, tip_labels_italics = FALSE,
            tip_labels_remove_underscore = TRUE,
            node_age_bars = TRUE, age_bars_colored_by = "posterior",
            age_bars_color = rev(colFun(2))) +
  ggplot2::theme(legend.position=c(.25, .85))

</code></pre>

<hr>
<h2 id='plotHiSSE'>plotHiSSE</h2><span id='topic+plotHiSSE'></span>

<h3>Description</h3>

<p>plotHiSSE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHiSSE(rates)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHiSSE_+3A_rates">rates</code></td>
<td>
<p>(data.frame; no default) a data frame containing columns
&quot;value&quot;, &quot;rate&quot;, &quot;hidden_state&quot;, &quot;observed_state&quot; (such as the output
of processSSE())</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# download the example dataset to working directory

url &lt;- "https://revbayes.github.io/tutorials/intro/data/primates_HiSSE_2.log"
dest_path &lt;- "primates_HiSSE_2.log"
download.file(url, dest_path)

# to run on your own data, change this to the path to your data file
hisse_file &lt;- dest_path

pdata &lt;- processSSE(hisse_file)
p &lt;- plotHiSSE(pdata);p

# change colors:
p + ggplot2::scale_fill_manual(values = c("red","green"))

# change x-axis label
p + ggplot2::xlab("Rate (events/Ma)")

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path)


</code></pre>

<hr>
<h2 id='plotMassExtinctions'>Plot Mass Extinction Support</h2><span id='topic+plotMassExtinctions'></span>

<h3>Description</h3>

<p>Plots the support (as 2ln Bayes factors) for mass extinctions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMassExtinctions(
  mass_extinction_trace,
  mass_extinction_times,
  mass_extinction_name,
  prior_prob,
  return_2lnBF = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMassExtinctions_+3A_mass_extinction_trace">mass_extinction_trace</code></td>
<td>
<p>(list; no default) The processed Rev output of
the mass extinction probabilities (output of readTrace()).</p>
</td></tr>
<tr><td><code id="plotMassExtinctions_+3A_mass_extinction_times">mass_extinction_times</code></td>
<td>
<p>(numeric; no default) Vector of the fixed grid
of times at which mass extinctions were allowed to occur.</p>
</td></tr>
<tr><td><code id="plotMassExtinctions_+3A_mass_extinction_name">mass_extinction_name</code></td>
<td>
<p>(character; no default) The name of the mass
extinction probability parameter (e.g. &quot;mass_extinction_probabilities&quot;)
for which support is to be calculated/plotted.</p>
</td></tr>
<tr><td><code id="plotMassExtinctions_+3A_prior_prob">prior_prob</code></td>
<td>
<p>(numeric; no default) The per-interval prior probability
of a mass extinction (one minus the p parameter in RevBayes'
dnReversibleJumpMixture()).</p>
</td></tr>
<tr><td><code id="plotMassExtinctions_+3A_return_2lnbf">return_2lnBF</code></td>
<td>
<p>(logical; TRUE) Should the 2ln(BF) be returned (if TRUE)
or simply the BF (if FALSE)?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Works only for analyses with a fixed grid where mass extinctions may occur.
</p>
<p>The return object can be manipulated. For example, you can change the axis
labels, the color palette, whether the axes are to be linked, or the overall
plotting style/theme, just as with any ggplot object.
</p>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>References</h3>

<p>Kass and Raftery (1995) Bayes Factors.
<em>JASA</em>, <b>90 (430)</b>, 773-795.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example dataset to working directory
url &lt;-
  "https://revbayes.github.io/tutorials/intro/data/crocs_mass_extinction_probabilities.log"
dest_path &lt;- "crocs_mass_extinction_probabilities.log"
download.file(url, dest_path)

# to run on your own data, change this to the path to your data file
mass_extinction_probability_file &lt;- dest_path

mass_extinction_probabilities &lt;-
              readTrace(mass_extinction_probability_file,burnin = 0.25)

# prior probability of mass extinction at any time
prior_n_expected &lt;- 0.1
n_intervals &lt;- 100
prior_prob &lt;- prior_n_expected/(n_intervals-1)

# times when mass extinctions were allowed
tree_age &lt;- 243.5
interval_times &lt;- tree_age * seq(1/n_intervals,(n_intervals-1) /
                                                   n_intervals,1/n_intervals)

# then plot results:
p &lt;- plotMassExtinctions(mass_extinction_trace=mass_extinction_probabilities,
                         mass_extinction_times=interval_times,
                         mass_extinction_name="mass_extinction_probabilities"
                         ,prior_prob);p

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path)


</code></pre>

<hr>
<h2 id='plotMuSSE'>plotMuSSE</h2><span id='topic+plotMuSSE'></span>

<h3>Description</h3>

<p>plotMuSSE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMuSSE(rates)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMuSSE_+3A_rates">rates</code></td>
<td>
<p>(data.frame; no default) a data frame containing columns
&quot;value&quot;, &quot;rate&quot;, &quot;hidden_state&quot;, &quot;observed_state&quot; (such as the output
of processSSE())</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example dataset to working directory

url &lt;-
  "https://revbayes.github.io/tutorials/intro/data/primates_BiSSE_activity_period.log"
dest_path &lt;- "primates_BiSSE_activity_period.log"
download.file(url, dest_path)

# to run on your own data, change this to the path to your data file
bisse_file &lt;- dest_path

pdata &lt;- processSSE(bisse_file)
p &lt;- plotMuSSE(pdata);p

# change colors:
p + ggplot2::scale_fill_manual(values = c("red","green"))

# change x-axis label
p + ggplot2::xlab("Rate (events/Ma)")

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path)

</code></pre>

<hr>
<h2 id='plotPopSizes'>Plot Population Sizes</h2><span id='topic+plotPopSizes'></span>

<h3>Description</h3>

<p>Plots the output of a coalescent demographic analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPopSizes(
  df,
  plot_CIs = TRUE,
  add = FALSE,
  existing_plot = NULL,
  col = "#00883a"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPopSizes_+3A_df">df</code></td>
<td>
<p>(data frame) such as produced by processPopSizes(), containing
the data on population sizes and corresponding grid points (points in time for population size evaluation)</p>
</td></tr>
<tr><td><code id="plotPopSizes_+3A_plot_cis">plot_CIs</code></td>
<td>
<p>(boolean; default: TRUE) specifies whether the credible intervals should be plotted.</p>
</td></tr>
<tr><td><code id="plotPopSizes_+3A_add">add</code></td>
<td>
<p>(boolean; default: FALSE) specifies whether the new plot should be added to an existing ggplot2 object. If TRUE,
the existing_plot has to be given.</p>
</td></tr>
<tr><td><code id="plotPopSizes_+3A_existing_plot">existing_plot</code></td>
<td>
<p>(ggplot2 object; default: NULL) a ggplot2 object to which the new plot should be added.</p>
</td></tr>
<tr><td><code id="plotPopSizes_+3A_col">col</code></td>
<td>
<p>(string; default: &quot;#00883a&quot;) color for the trajectories</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the output of coalescent demographic analyses. Takes as
input the output of processPopSizes() and plotting parameters.
</p>
<p>The return object can be manipulated. For example, you can change the
axis labels, the color palette, whether the axes are to be linked, or the
overall plotting style/theme, just as with any ggplot object.
</p>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- dplyr::tibble("time" = c(0.0, 1.0, 2.0, 3.0, 4.0),
                    "value" = c(1.0, 1.5, 2.0, 1.5, 1.5),
                    "upper" = c(3.5, 7.0, 6.5, 5.0, 5.0),
                    "lower" = c(0.5, 0.1, 0.5, 0.5, 0.8))

plotPopSizes(df)

</code></pre>

<hr>
<h2 id='plotPostPredStats'>plot Posterior Predictive Statistics</h2><span id='topic+plotPostPredStats'></span>

<h3>Description</h3>

<p>Plots the posterior predictive statistics data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPostPredStats(
  data,
  prob = c(0.9, 0.95),
  col = NULL,
  side = "both",
  type = "strict",
  PPES = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPostPredStats_+3A_data">data</code></td>
<td>
<p>(list of data frames; no default) A list of data frames
of the empirical and simulated values, such as the output of
processPostPredStats.R</p>
</td></tr>
<tr><td><code id="plotPostPredStats_+3A_prob">prob</code></td>
<td>
<p>(vector of numerics; default c(0.9, 0.95)) The
posterior-predictive intervals to shade.</p>
</td></tr>
<tr><td><code id="plotPostPredStats_+3A_col">col</code></td>
<td>
<p>(vector of colors; default NULL) The colors for each quantile.
Defaults to blue and red.</p>
</td></tr>
<tr><td><code id="plotPostPredStats_+3A_side">side</code></td>
<td>
<p>(character; default &quot;both&quot;) Whether the plotted/colored 
intervals are on &quot;both&quot; sides, the &quot;left&quot; side, or the &quot;right&quot; 
side of the distribution.</p>
</td></tr>
<tr><td><code id="plotPostPredStats_+3A_type">type</code></td>
<td>
<p>(character; default &quot;strict&quot;) Whether equal values are
considered as less extreme as the observed data (&quot;strict&quot;) or half of the
equal values are considered to be higher and half to be lower (&quot;midpoint&quot;)</p>
</td></tr>
<tr><td><code id="plotPostPredStats_+3A_ppes">PPES</code></td>
<td>
<p>(boolean; default FALSE) Whether we provide the posterior
predictive effect size (PPES).</p>
</td></tr>
<tr><td><code id="plotPostPredStats_+3A_...">...</code></td>
<td>
<p>Additional arguments are passed to stats::density().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces one ggplot object per metric. Intended
to plot the results of the RevBayes tutorial:
Assessing Phylogenetic Reliability Using RevBayes and P3
Model adequacy testing using posterior prediction (Data Version).
</p>
<p>Each plot shows the rejection region for the provided quantiles,
as well as a p-value for the observed statistic. If side=&quot;left&quot; (or &quot;right&quot;),
then the p-value is the fraction of simulated statistics that are less than
( or greater than) or equal to the observed statistic. If side=&quot;both&quot;, then
the p-value is calculated by first fitting a KDE to the samples, then
computing the fraction of simulated statistics with density lower than the
density of he observed statistic; in this sense, the &quot;both&quot; option computes
the size of HPD defined by the observed statistic.
</p>


<h3>Value</h3>

<p>A list of ggplot objects, where each plot contains a density
distribution of the predicted values and a dashed line of the empirical
value. The blue shaded region of the density plot corresponds to the 5%
two-sided quantile and the orange corresponds to the 2% two-sided quantile.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example datasets to working directory

url_emp &lt;-
   "https://revbayes.github.io/tutorials/intro/data/empirical_data_pps_example.csv"
dest_path_emp &lt;- "empirical_data_pps_example.csv"
download.file(url_emp, dest_path_emp)

url_sim &lt;-
   "https://revbayes.github.io/tutorials/intro/data/simulated_data_pps_example.csv"
dest_path_sim &lt;- "simulated_data_pps_example.csv"
download.file(url_sim, dest_path_sim)

# to run on your own data, change this to the path to your data file
file_sim &lt;- dest_path_sim
file_emp &lt;- dest_path_emp

t &lt;- processPostPredStats(path_sim = file_sim,
                         path_emp = file_emp)
plots &lt;- plotPostPredStats(data = t)
plots[[1]]

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_sim, dest_path_emp)



</code></pre>

<hr>
<h2 id='plotTrace'>Plot trace</h2><span id='topic+plotTrace'></span>

<h3>Description</h3>

<p>Plots the posterior distributions of variables from trace file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTrace(trace, color = "default", vars = NULL, match = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTrace_+3A_trace">trace</code></td>
<td>
<p>(list of data frames; no default) Name of a list of data frames,
such as produced by readTrace(). If the readTrace() output
contains multiple traces (such as from multiple runs), summarizeTrace() will
provide summaries for each trace individually, as well as the combined trace.</p>
</td></tr>
<tr><td><code id="plotTrace_+3A_color">color</code></td>
<td>
<p>(&quot;character&quot;; &quot;default&quot;) Colors for parameters. Defaults to
default RevGadgets colors. For non-default colors, provide a named vector of
length of the number of parameters.</p>
</td></tr>
<tr><td><code id="plotTrace_+3A_vars">vars</code></td>
<td>
<p>(character or character vector; NULL) The specific name(s) of
the variable(s) to be summarized.</p>
</td></tr>
<tr><td><code id="plotTrace_+3A_match">match</code></td>
<td>
<p>(character; NULL) A string to match to a group of parameters.
For example, match = &quot;er&quot; will plot the variables &quot;er[1]&quot;, &quot;er[2]&quot;, &quot;er[3]&quot;,
etc.. match will only work if your search string is followed by brackets in
one or more of the column names of the provided trace file. match = &quot;er&quot; will
only return the exchangeability parameters, but will not plot &quot;Posterior&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the posterior distributions of continuous variables from one or
multiple traces (as in, from multiple runs). Shaded regions under the curve
represent the 95% credible interval.  If multiple traces are provided,
plotTrace() will plot each run independently as well as plot the combined
output. Note that for variables ith very different distributions, overlaying
the plots may result in illegible figures. In these cases, we recommend
plotting each parameter separately.
</p>


<h3>Value</h3>

<p>plotTrace() returns a list of the length of provided trace object,
plus one combined trace. Each element of the list contains a ggplot object
with plots of the provided parameters. These plots may be modified in
typical ggplot fashion.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# example with quantitative parameters

# download the example dataset to working directory
url_gtr &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR.log"
dest_path_gtr &lt;- "primates_cytb_GTR.log"
download.file(url_gtr, dest_path_gtr)

# to run on your own data, change this to the path to your data file
file &lt;- dest_path_gtr

one_trace &lt;- readTrace(paths = file)
plots &lt;- plotTrace(trace = one_trace,
                    vars = c("pi[1]","pi[2]","pi[3]","pi[4]"))
plots[[1]]

# add custom colors
plots &lt;- plotTrace(trace = one_trace,
                   vars = c("pi[3]","pi[4]","pi[1]","pi[2]"),
                   color = c("pi[1]" = "green",
                             "pi[2]"= "red",
                             "pi[3]"= "blue",
                             "pi[4]"= "orange"))
plots[[1]]

# make the same plot, using match
plots &lt;- plotTrace(trace = one_trace, match = "pi")
plots[[1]]

#' # remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_gtr)

# plot some qualitative variables

# download the example dataset to working directory
url_rj &lt;- "https://revbayes.github.io/tutorials/intro/data/freeK_RJ.log"
dest_path_rj &lt;- "freeK_RJ.log"
download.file(url_rj, dest_path_rj)

file &lt;- dest_path_rj
trace &lt;- readTrace(path = file)

plots &lt;- plotTrace(trace = trace,
                   vars = c("prob_rate_12", "prob_rate_13",
                            "prob_rate_31", "prob_rate_32"))
plots[[1]]

# with custom colors
plots &lt;- plotTrace(trace = trace,
                   vars = c("prob_rate_12", "prob_rate_13",
                            "prob_rate_31", "prob_rate_32"),
                   color = c("prob_rate_12" = "green",
                             "prob_rate_13" = "red",
                             "prob_rate_31"= "blue",
                             "prob_rate_32" = "orange"))
plots[[1]]

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_rj)



</code></pre>

<hr>
<h2 id='plotTree'>Plot tree</h2><span id='topic+plotTree'></span>

<h3>Description</h3>

<p>Plots a single tree, such as an MCC or MAP tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTree(
  tree,
  timeline = FALSE,
  geo_units = list("epochs", "periods"),
  geo = timeline,
  time_bars = timeline,
  node_age_bars = FALSE,
  age_bars_color = "blue",
  age_bars_colored_by = NULL,
  age_bars_width = 1.5,
  node_labels = NULL,
  node_labels_color = "black",
  node_labels_size = 3,
  node_labels_offset = 0,
  tip_labels = TRUE,
  tip_labels_italics = FALSE,
  tip_labels_formatted = FALSE,
  tip_labels_remove_underscore = TRUE,
  tip_labels_color = "black",
  tip_labels_size = 3,
  tip_labels_offset = 0,
  node_pp = FALSE,
  node_pp_shape = 16,
  node_pp_color = "black",
  node_pp_size = "variable",
  branch_color = "black",
  color_branch_by = NULL,
  line_width = 1,
  tree_layout = "rectangular",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTree_+3A_tree">tree</code></td>
<td>
<p>(list of lists of treedata objects; no default) Name of a list of
lists of treedata objects, such as produced by readTrees(). This object
should only contain only one summary tree from one trace file. If it contains
multiple trees or multiple traces, only the first will be used.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_timeline">timeline</code></td>
<td>
<p>(logical; FALSE) Plot time tree with labeled x-axis with
timescale in MYA.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_geo_units">geo_units</code></td>
<td>
<p>(list; list(&quot;epochs&quot;, &quot;periods&quot;)) Which geological units to
include in the geo timescale. May be &quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, 
&quot;eras&quot;, or a list of two of those units.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_geo">geo</code></td>
<td>
<p>(logical; timeline) Add a geological timeline? Defaults to the
same as timeline.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_time_bars">time_bars</code></td>
<td>
<p>(logical; timeline) Add vertical gray bars to indicate
geological timeline units if geo == TRUE or regular time intervals (in MYA)
if geo == FALSE.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_age_bars">node_age_bars</code></td>
<td>
<p>(logical; FALSE) Plot time tree with node age bars?</p>
</td></tr>
<tr><td><code id="plotTree_+3A_age_bars_color">age_bars_color</code></td>
<td>
<p>(character; &quot;blue&quot;) Color for node age bars. If
age_bars_colored_by specifies a variable (not NULL), you must provide two
colors, low and high values for a gradient. Colors must be either
R valid color names or valid hex codes.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_age_bars_colored_by">age_bars_colored_by</code></td>
<td>
<p>(character; NULL) Specify column to color node age
bars by, such as &quot;posterior&quot;. If null, all node age bars plotted the same
color, specified by age_bars_color</p>
</td></tr>
<tr><td><code id="plotTree_+3A_age_bars_width">age_bars_width</code></td>
<td>
<p>(numeric; 1.5) Change line width for age bars</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_labels">node_labels</code></td>
<td>
<p>(character; NULL) Plot text labels at nodes, specified by
the name of the corresponding column in the tidytree object. If NULL, no
text is plotted.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_labels_color">node_labels_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color to plot node_labels,
either as a valid R color name or a valid hex code.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_labels_size">node_labels_size</code></td>
<td>
<p>(numeric; 3) Size of node labels</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_labels_offset">node_labels_offset</code></td>
<td>
<p>(numeric; 0) Horizontal offset of node labels from
nodes.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels">tip_labels</code></td>
<td>
<p>(logical; TRUE) Plot tip labels?</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels_italics">tip_labels_italics</code></td>
<td>
<p>(logical; FALSE) Plot tip labels in italics?</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels_formatted">tip_labels_formatted</code></td>
<td>
<p>(logical; FALSE) Do the tip labels contain 
manually added formatting information? Will set parse = TRUE in geom_text()
and associated functions to interpret formatting. See ?plotmath for more.
Cannot be TRUE if tip_labels_italics = TRUE.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels_remove_underscore">tip_labels_remove_underscore</code></td>
<td>
<p>(logical; TRUE) Remove underscores in tip
labels?</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels_color">tip_labels_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color to plot tip labels, either
as a valid R color name or a valid hex code.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels_size">tip_labels_size</code></td>
<td>
<p>(numeric; 3) Size of tip labels</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tip_labels_offset">tip_labels_offset</code></td>
<td>
<p>(numeric; 1) Horizontal offset of tip labels from
tree.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_pp">node_pp</code></td>
<td>
<p>(logical; FALSE) Plot posterior probabilities as symbols at
nodes? Specify symbol aesthetics with node_pp_shape, node_pp_color, and
node_pp_size.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_pp_shape">node_pp_shape</code></td>
<td>
<p>(integer; 1) Integer corresponding to point shape
(value between 0-25). See ggplot2 documentation for details:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point</a></p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_pp_color">node_pp_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color for node_pp symbols, either
as valid R color name(s) or hex code(s). Can be a single character string
specifying a single color, or a vector of length two specifying two colors
to form a gradient. In this case, posterior probabilities will be indicated
by color along the specified gradient.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_node_pp_size">node_pp_size</code></td>
<td>
<p>(numeric or character; 1) Size for node_pp symbols.
If numeric, the size will be fixed at the specified value. If a character,
it should specify &quot;variable&quot;, indicating that size should be scaled by the
posterior value. Size regulates the area of the shape, following ggplot2
best practices:
<a href="https://ggplot2.tidyverse.org/reference/scale_size.html">https://ggplot2.tidyverse.org/reference/scale_size.html</a>)</p>
</td></tr>
<tr><td><code id="plotTree_+3A_branch_color">branch_color</code></td>
<td>
<p>(character; &quot;black&quot;) A single character string
specifying the color (R color name or hex code) for all branches OR a
vector of length 2 specifying two colors for a gradient, used to color the
branches according to the variable specified in color_branch_by. If only 1
color is provided and you specify color_branch_by, default colors will be
chosen (low = &quot;#005ac8&quot;, high = &quot;#fa7850&quot;).</p>
</td></tr>
<tr><td><code id="plotTree_+3A_color_branch_by">color_branch_by</code></td>
<td>
<p>(character; NULL ) Optional name of one quantitative
variable in the treedata object to color branches, such as a rate.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_line_width">line_width</code></td>
<td>
<p>(numeric; 1) Change line width for branches</p>
</td></tr>
<tr><td><code id="plotTree_+3A_tree_layout">tree_layout</code></td>
<td>
<p>(character; &quot;rectangular&quot;) Tree shape layout, passed
to ggtree(). Options are 'rectangular', 'cladogram', 'slanted', 'ellipse',
'roundrect', 'fan', 'circular', 'inward_circular', 'radial', 'equal_angle',
'daylight', or 'ape'.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_...">...</code></td>
<td>
<p>(various) Additional arguments passed to ggtree::ggtree().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a single tree, such as an MCC or MAP tree, with
optionally labeled posterior probabilities at nodes, a
timescale plotted on the x - axis, and 95% CI for node ages.
</p>


<h3>Value</h3>

<p>returns a single plot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example of standard tree plot

file &lt;- system.file("extdata",
                    "sub_models/primates_cytb_GTR_MAP.tre",
                    package="RevGadgets")
tree &lt;- readTrees(paths = file)
# Reroot tree before plotting
tree_rooted &lt;- rerootPhylo(tree = tree, outgroup = "Galeopterus_variegatus")
# Plot
p &lt;- plotTree(tree = tree_rooted, node_labels = "posterior");p

# Plot unladderized tree
p &lt;- plotTree(tree = tree_rooted,
              node_labels = "posterior",
              ladderize = FALSE);p

# We can add a scale bar:
p + ggtree::geom_treescale(x = -0.35, y = -1)

# Example of coloring branches by rate
file &lt;- system.file("extdata",
                    "relaxed_ou/relaxed_OU_MAP.tre",
                    package="RevGadgets")
tree &lt;- readTrees(paths = file)
p &lt;- plotTree(tree = tree,
              node_age_bars = FALSE,
              node_pp = FALSE,
              tip_labels_remove_underscore = TRUE,
              tip_labels_italics = FALSE,
              color_branch_by = "branch_thetas",
              line_width = 1.7) +
       ggplot2::theme(legend.position=c(.1, .9));p

</code></pre>

<hr>
<h2 id='plotTreeFull'>Plot Full tree</h2><span id='topic+plotTreeFull'></span>

<h3>Description</h3>

<p>Plots a tree, such as an MCC or MAP tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTreeFull(
  tree,
  timeline,
  geo,
  geo_units,
  time_bars,
  node_age_bars,
  tip_age_bars,
  age_bars_color,
  age_bars_colored_by,
  age_bars_width,
  node_labels,
  node_labels_color,
  node_labels_size,
  node_labels_offset,
  tip_labels,
  tip_labels_italics,
  tip_labels_formatted,
  tip_labels_remove_underscore,
  tip_labels_color,
  tip_labels_size,
  tip_labels_offset,
  label_sampled_ancs,
  node_pp,
  node_pp_shape,
  node_pp_color,
  node_pp_size,
  branch_color,
  color_branch_by,
  line_width,
  tree_layout,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTreeFull_+3A_tree">tree</code></td>
<td>
<p>(list of lists of treedata objects; no default) Name of a list
of lists of treedata objects, such as produced by readTrees(). This object
should only contain only one summary tree from one trace file. If it
contains multiple trees or multiple traces, only the first will be used.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_timeline">timeline</code></td>
<td>
<p>(logical; FALSE) Plot time tree with labeled x-axis with
timescale in MYA.
#'</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_geo">geo</code></td>
<td>
<p>(logical; timeline) Add a geological timeline? Defaults to the
same as timeline.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_geo_units">geo_units</code></td>
<td>
<p>(list; list(&quot;epochs&quot;, &quot;periods&quot;)) Which geological units to
include in the geo timescale. May be &quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, 
&quot;eras&quot;, or a list of two of those units.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_time_bars">time_bars</code></td>
<td>
<p>(logical; timeline) Add vertical gray bars to indicate
geological timeline units if geo == TRUE or regular time intervals (in MYA)
if geo == FALSE.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_age_bars">node_age_bars</code></td>
<td>
<p>(logical; TRUE) Plot time tree with node age bars?</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_age_bars">tip_age_bars</code></td>
<td>
<p>(logical; FALSE) Plot node age bars for the tips as
well? Useful for plotting serial sampled analyses or fossilized birth-death
analyses, or any cases where some tip ages are estimated.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_age_bars_color">age_bars_color</code></td>
<td>
<p>(character; &quot;blue&quot;) Color for node/tip age bars.
If age_bars_colored_by specifies a variable (not NULL), you must provide
two colors, low and high values for a gradient. Colors must be either R
valid color names or valid hex codes.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_age_bars_colored_by">age_bars_colored_by</code></td>
<td>
<p>(character; NULL) Specify column to color
node/tip age bars by, such as &quot;posterior&quot;. If null, all age bars plotted the
same color, specified by age_bars_color</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_age_bars_width">age_bars_width</code></td>
<td>
<p>(numeric; 1) Change line width for age bars</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_labels">node_labels</code></td>
<td>
<p>(character; NULL) Plot text labels at nodes, specified by
the name of the corresponding column in the tidytree object. If NULL, no
text is plotted.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_labels_color">node_labels_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color to plot node_labels,
either as a valid R color name or a valid hex code.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_labels_size">node_labels_size</code></td>
<td>
<p>(numeric; 3) Size of node labels</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_labels_offset">node_labels_offset</code></td>
<td>
<p>(numeric; 0) Horizontal offset of node labels
from nodes.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels">tip_labels</code></td>
<td>
<p>(logical; TRUE) Plot tip labels?</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels_italics">tip_labels_italics</code></td>
<td>
<p>(logical; FALSE) Plot tip labels in italics?</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels_formatted">tip_labels_formatted</code></td>
<td>
<p>(logical; FALSE) Do the tip labels contain 
manually added formatting information? Will set parse = TRUE in geom_text()
and associated functions to interpret formatting. See ?plotmath for more.
Cannot be TRUE if tip_labels_italics = TRUE.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels_remove_underscore">tip_labels_remove_underscore</code></td>
<td>
<p>(logical; FALSE) Should underscores be
replaced by spaces in tip labels?</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels_color">tip_labels_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color to plot tip labels, either
as a valid
R color name or a valid hex code.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels_size">tip_labels_size</code></td>
<td>
<p>(numeric; 3) Size of tip labels</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tip_labels_offset">tip_labels_offset</code></td>
<td>
<p>(numeric; 1) Horizontal offset of tip labels from
tree.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_label_sampled_ancs">label_sampled_ancs</code></td>
<td>
<p>(logical; FALSE) Label any sampled ancestors? Will
inherent tip labels aesthetics for size and color.
</p>
<p># added in from plotTree</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_pp">node_pp</code></td>
<td>
<p>(logical; FALSE) Plot posterior probabilities as symbols at
nodes? Specify symbol aesthetics with node_pp_shape, node_pp_color, and
node_pp_size.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_pp_shape">node_pp_shape</code></td>
<td>
<p>(integer; 1) Integer corresponding to point shape
(value between 0-25). See ggplot2 documentation for details:
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point">https://ggplot2.tidyverse.org/articles/ggplot2-specs.html#point</a></p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_pp_color">node_pp_color</code></td>
<td>
<p>(character; &quot;black&quot;) Color for node_pp symbols,
either as valid R color name(s) or hex code(s). Can be a single character
string specifying a single color, or a vector of length two specifying two
colors to form a gradient. In this case, posterior probabilities will be
indicated by color along the specified gradient.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_node_pp_size">node_pp_size</code></td>
<td>
<p>(numeric or character; 1) Size for node_pp symbols.
If numeric, the size will be fixed at the specified value. If a character,
it should specify &quot;variable&quot;, indicating that size should be scaled by the
posterior value. Size regulates the area of the shape, following ggplot2
best practices:
<a href="https://ggplot2.tidyverse.org/reference/scale_size.html">https://ggplot2.tidyverse.org/reference/scale_size.html</a>)</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_branch_color">branch_color</code></td>
<td>
<p>(character; &quot;black&quot;) A single character string
specifying the color (R color name or hex code) for all branches OR a vector
of length 2 specifying two colors for a gradient, used to color the branches
according to the variable specified in color_branch_by. If only 1 color is
provided and you specify color_branch_by, default colors will be chosen
(low = &quot;#005ac8&quot;, high = &quot;#fa7850&quot;).</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_color_branch_by">color_branch_by</code></td>
<td>
<p>(character; NULL ) Optional name of one quantitative
variable in the treedata object to color branches, such as a rate.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_line_width">line_width</code></td>
<td>
<p>(numeric; 1) Change line width for branches</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_tree_layout">tree_layout</code></td>
<td>
<p>(character; &quot;rectangular&quot;) Tree shape layout, passed to
ggtree(). Options are 'rectangular', 'cladogram', 'slanted', 'ellipse',
'roundrect', 'fan', 'circular', 'inward_circular', 'radial', 'equal_angle',
'daylight' or 'ape'.</p>
</td></tr>
<tr><td><code id="plotTreeFull_+3A_...">...</code></td>
<td>
<p>(various) Additional arguments passed to ggtree::ggtree().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a single tree, such as an MCC or MAP tree, with
the full set of functionality to be called by plotTree()
and plotFBDTree()
</p>


<h3>Value</h3>

<p>returns a single plot object.
</p>


<h3>See Also</h3>

<p>called by <a href="#topic+plotTree">plotTree</a> and
<a href="#topic+plotFBDTree">plotFBDTree</a>
</p>

<hr>
<h2 id='posteriorSamplesToParametricPrior'>Priors from MCMC samples</h2><span id='topic+posteriorSamplesToParametricPrior'></span>

<h3>Description</h3>

<p>Turn posterior samples collected by MCMC into a parametric prior
distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posteriorSamplesToParametricPrior(
  samples,
  distribution,
  variance_inflation_factor = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="posteriorSamplesToParametricPrior_+3A_samples">samples</code></td>
<td>
<p>(numeric vector; no default) MCMC samples for a single
parameter.</p>
</td></tr>
<tr><td><code id="posteriorSamplesToParametricPrior_+3A_distribution">distribution</code></td>
<td>
<p>(character; no default) The distribution to fit. Options
are gamma for strictly positive
parameters and normal for unbounded parameters.</p>
</td></tr>
<tr><td><code id="posteriorSamplesToParametricPrior_+3A_variance_inflation_factor">variance_inflation_factor</code></td>
<td>
<p>(single numeric value; default = 2.0) Makes
the prior variance larger than the variance of the posterior</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distributions are fit by the method of moments.
The function allows inflating the prior variance relative to the posterior
being supplied.
</p>


<h3>Value</h3>

<p>Numeric vector of parameters with names (to avoid rate/scale and
var/sd confusion).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example datasets to working directory

url_ex_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_extinction_times.log"
dest_path_ex_times &lt;- "primates_EBD_extinction_times.log"
download.file(url_ex_times, dest_path_ex_times)

url_ex_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_extinction_rates.log"
dest_path_ex_rates &lt;- "primates_EBD_extinction_rates.log"
download.file(url_ex_rates, dest_path_ex_rates)

url_sp_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_times.log"
dest_path_sp_times &lt;- "primates_EBD_speciation_times.log"
download.file(url_sp_times, dest_path_sp_times)

url_sp_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_rates.log"
dest_path_sp_rates &lt;- "primates_EBD_speciation_rates.log"
download.file(url_sp_rates, dest_path_sp_rates)

# to run on your own data, change this to the path to your data file
speciation_time_file &lt;- dest_path_sp_times
speciation_rate_file &lt;- dest_path_sp_rates
extinction_time_file &lt;- dest_path_ex_times
extinction_rate_file &lt;- dest_path_ex_rates

primates &lt;- processDivRates(speciation_time_log = speciation_time_file,
                            speciation_rate_log = speciation_rate_file,
                            extinction_time_log = extinction_time_file,
                            extinction_rate_log = extinction_rate_file,
                            burnin = 0.25)

speciation_rates &lt;-
       dplyr::pull(primates[which(primates$item == "speciation rate"),],
                  "value")
speciation_1_gamma_prior &lt;-
          posteriorSamplesToParametricPrior(speciation_rates,"gamma")

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_sp_times, dest_path_ex_times,
            dest_path_sp_rates, dest_path_ex_rates)


</code></pre>

<hr>
<h2 id='processAncStates'>Process Ancestral States</h2><span id='topic+processAncStates'></span>

<h3>Description</h3>

<p>Process data for ancestral states plotting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processAncStates(
  path,
  state_labels = NULL,
  labels_as_numbers = FALSE,
  missing_to_NA = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processAncStates_+3A_path">path</code></td>
<td>
<p>(character string; no default) File path to annotated tree.</p>
</td></tr>
<tr><td><code id="processAncStates_+3A_state_labels">state_labels</code></td>
<td>
<p>(character vector; NULL) Vector of labels for ancestral
states named with the current state labels in annotated tree file
(as characters).</p>
</td></tr>
<tr><td><code id="processAncStates_+3A_labels_as_numbers">labels_as_numbers</code></td>
<td>
<p>(logical; FALSE) Should the state labels be treated
as integers (for example, as chromosome numbers)?</p>
</td></tr>
<tr><td><code id="processAncStates_+3A_missing_to_na">missing_to_NA</code></td>
<td>
<p>(logical; TRUE) Should missing data, coded as &quot;?&quot;, be
coded to NA? If TRUE, the state will not be plotted. If FALSE, it will be
considered an additional state  when plotting.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A treedata object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# standard ancestral state estimation example
file &lt;- system.file("extdata",
                    "comp_method_disc/ase_freeK.tree",
                    package="RevGadgets")
example &lt;- processAncStates(file,
                            state_labels = c("1" = "Awesome",
                                             "2" = "Beautiful",
                                             "3" = "Cool!"))

#chromosome evolution example
file &lt;- system.file("extdata",
                    "chromo/ChromEvol_simple_final.tree",
                    package="RevGadgets")
chromo_example &lt;- processAncStates(file, labels_as_numbers = TRUE)

</code></pre>

<hr>
<h2 id='processBranchData'>processBranchData</h2><span id='topic+processBranchData'></span>

<h3>Description</h3>

<p>processBranchData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processBranchData(
  tree,
  dat,
  burnin = 0.25,
  parnames = c("avg_lambda", "avg_mu", "num_shifts"),
  summary = "median",
  net_div = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processBranchData_+3A_tree">tree</code></td>
<td>
<p>(treedata object; no default) a phylogenetic tree in the
treedata format, or a list of lists of a single tree data object, such as the
output of readTrees().</p>
</td></tr>
<tr><td><code id="processBranchData_+3A_dat">dat</code></td>
<td>
<p>(data.frame or list; no default) a data frame, or a list
(of length 1) of a data frame, with branch specific data, such as the output
of readTrace().</p>
</td></tr>
<tr><td><code id="processBranchData_+3A_burnin">burnin</code></td>
<td>
<p>(numeric; 0.25) fraction of the markov-chain to discard</p>
</td></tr>
<tr><td><code id="processBranchData_+3A_parnames">parnames</code></td>
<td>
<p>(character vector; c(&quot;avg_lambda&quot;, &quot;avg_mu&quot;, &quot;num_shifts&quot;))
Names of parameters to process</p>
</td></tr>
<tr><td><code id="processBranchData_+3A_summary">summary</code></td>
<td>
<p>(character; &quot;median&quot;) function to summarize the continuous
parameter. Typically mean or median</p>
</td></tr>
<tr><td><code id="processBranchData_+3A_net_div">net_div</code></td>
<td>
<p>(logical; FALSE) Calculate net diversification?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a treedata file with attached branch-specific data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example dataset to working directory
url_rates &lt;-
  "https://revbayes.github.io/tutorials/intro/data/primates_BDS_rates.log"
dest_path_rates &lt;- "primates_BDS_rates.log"
download.file(url_rates, dest_path_rates)

url_tree &lt;-
  "https://revbayes.github.io/tutorials/divrate/data/primates_tree.nex"
dest_path_tree &lt;- "primates_tree.nex"
download.file(url_tree, dest_path_tree)

# to run on your own data, change this to the path to your data file
treefile &lt;- dest_path_tree
logfile &lt;- dest_path_rates

branch_data &lt;- readTrace(logfile)
tree &lt;- readTrees(paths = treefile)

annotated_tree &lt;- processBranchData(tree, branch_data, summary = "median")

# you can plot this output
p &lt;- plotTree(tree = annotated_tree,
              node_age_bars = FALSE,
              node_pp = FALSE,
              tip_labels = FALSE,
              color_branch_by = "avg_lambda",
              line_width = 0.8) +
     ggplot2::theme(legend.position=c(.1, .9));p
# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_tree, dest_path_rates)

</code></pre>

<hr>
<h2 id='processDivRates'>Process Diversification Rates</h2><span id='topic+processDivRates'></span>

<h3>Description</h3>

<p>Processing the output of a episodic diversification rate analysis with
mass-extinction events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processDivRates(
  speciation_time_log = "",
  speciation_rate_log = "",
  extinction_time_log = "",
  extinction_rate_log = "",
  fossilization_time_log = "",
  fossilization_rate_log = "",
  burnin = 0.25,
  probs = c(0.025, 0.975),
  summary = "median"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processDivRates_+3A_speciation_time_log">speciation_time_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to speciation times log file(s)</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_speciation_rate_log">speciation_rate_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to speciation rates log file(s)</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_extinction_time_log">extinction_time_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to extinction times log file(s)</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_extinction_rate_log">extinction_rate_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to extinction rates log file(s)</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_fossilization_time_log">fossilization_time_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to fossilization times log file(s)</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_fossilization_rate_log">fossilization_rate_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to fossilization rates log file(s)</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_burnin">burnin</code></td>
<td>
<p>(single numeric value; default = 0) Fraction of generations to
discard (if value provided is between 0 and 1) or number of generations (if
value provided is greater than 1). Passed to readTrace().</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_probs">probs</code></td>
<td>
<p>(numeric vector; c(0.025, 0.975)) a vector of length two
containing the upper and lower bounds for the confidence intervals.</p>
</td></tr>
<tr><td><code id="processDivRates_+3A_summary">summary</code></td>
<td>
<p>typically &quot;mean&quot; or &quot;median&quot;; the metric to summarize the
posterior distribution. Defaults to &quot;median&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For processing the output of an episodic diversification rate analysis.
processDivRates() assumes that the epochs are fixed rather than inferred.
Additionally, it assumes that times correspond to rates such that the first
rate parameter (i.e. speciation[1]) corresponds to the present. Conversely,
the first time parameter (i.e. interval_times[1]) corresponds to the first
time interval after the present, moving backwards in time. processDivRates()
relies on readTrace and produces a list object that can be read by
plotDivRates() to visualize the results. For now, only one log file per
parameter type is accepted (i.e. log files from multiple runs must be
combined before reading into the function).
</p>


<h3>Value</h3>

<p>List object with processed rate and time parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example datasets to working directory

url_ex_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_extinction_times.log"
dest_path_ex_times &lt;- "primates_EBD_extinction_times.log"
download.file(url_ex_times, dest_path_ex_times)

url_ex_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_extinction_rates.log"
dest_path_ex_rates &lt;- "primates_EBD_extinction_rates.log"
download.file(url_ex_rates, dest_path_ex_rates)

url_sp_times &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_times.log"
dest_path_sp_times &lt;- "primates_EBD_speciation_times.log"
download.file(url_sp_times, dest_path_sp_times)

url_sp_rates &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_EBD_speciation_rates.log"
dest_path_sp_rates &lt;- "primates_EBD_speciation_rates.log"
download.file(url_sp_rates, dest_path_sp_rates)

# to run on your own data, change this to the path to your data file
speciation_time_file &lt;- dest_path_sp_times
speciation_rate_file &lt;- dest_path_sp_rates
extinction_time_file &lt;- dest_path_ex_times
extinction_rate_file &lt;- dest_path_ex_rates

rates &lt;- processDivRates(speciation_time_log = speciation_time_file,
                         speciation_rate_log = speciation_rate_file,
                         extinction_time_log = extinction_time_file,
                         extinction_rate_log = extinction_rate_file,
                         burnin = 0.25)

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_sp_times, dest_path_ex_times,
            dest_path_sp_rates, dest_path_ex_rates)


</code></pre>

<hr>
<h2 id='processPopSizes'>Process Population Sizes</h2><span id='topic+processPopSizes'></span>

<h3>Description</h3>

<p>Processing the output of a coalescent demographic analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processPopSizes(
  population_size_log = "",
  interval_change_points_log = "",
  model = "constant",
  burnin = 0.25,
  probs = c(0.025, 0.975),
  summary = "median",
  num_grid_points = 100,
  spacing = "exponential",
  max_age = NULL,
  min_age = NULL,
  distribution = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processPopSizes_+3A_population_size_log">population_size_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to population sizes log file(s)</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_interval_change_points_log">interval_change_points_log</code></td>
<td>
<p>(vector of character strings or
single character string; &quot;&quot;) Path to interval change points log file(s).
If not given, a constant process with only one population size is assumed.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_model">model</code></td>
<td>
<p>(string, default: &quot;constant&quot;) The demographic model of the intervals.
Can be &quot;constant&quot; or &quot;linear&quot;.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_burnin">burnin</code></td>
<td>
<p>(single numeric value; default: 0.25) Fraction of generations to
discard (if value provided is between 0 and 1) or number of generations (if
value provided is greater than 1).</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_probs">probs</code></td>
<td>
<p>(numeric vector; c(0.025, 0.975)) a vector of length two
containing the upper and lower bounds for the confidence intervals.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_summary">summary</code></td>
<td>
<p>(string, default: &quot;median&quot;) the metric to summarize the
posterior distribution, typically &quot;mean&quot; or &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_num_grid_points">num_grid_points</code></td>
<td>
<p>(numeric; default: 100) defines the number of grid points through time for which to
evaluate the demographic functions.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_spacing">spacing</code></td>
<td>
<p>(string, default: &quot;exponential&quot;) The spacing of grid points. Can be &quot;exponential&quot; or &quot;equal&quot;.
Exponentially spaced grid points are dense towards the present and have larger distances towards the past.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_max_age">max_age</code></td>
<td>
<p>(numeric; default: NULL, i.e. not provided) defines the maximal age up to which the demographic functions should be evaluated.
If not provided, it will either be automatically set to 1e5 (in case of a constant process) or
to the maximal age provided with the interval_change_points_log.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_min_age">min_age</code></td>
<td>
<p>(numeric; default: NULL, i.e. not provided) defines the minimal age up to which the demographic functions should be evaluated.
If not provided, it will either be automatically set to 1e2 (in case of a constant process) or
to the minimal age provided with the interval_change_points_log. Can not be 0 in case of exponential spacing.</p>
</td></tr>
<tr><td><code id="processPopSizes_+3A_distribution">distribution</code></td>
<td>
<p>(boolean; default: FALSE) specifies whether the summary data frame will be returned 
(distribution = FALSE) or a matrix with distributions of population size for each point on the grid and 
with the times of the grid points as row names (distribution = TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For processing the output of a coalescent demographic analysis.
processPopSizes() assumes that the the first size parameter (i.e. population_size[1])
corresponds to the present. processPopSizes() partly
relies on readTrace and produces a list object that can be read by
plotPopSizes() to visualize the results. For now, only one log file per
parameter type is accepted (i.e. log files from multiple runs must be
combined before reading into the function).
</p>


<h3>Value</h3>

<p>List object with processed rate and, if applicable, time parameters (if distribution = FALSE). 
Matrix object with distributions of population size (if distribution = TRUE). If applicable, one row for each point on the grid, 
with the times of the grid points as row names.
</p>

<hr>
<h2 id='processPostPredStats'>process Posterior Predictive Statistics</h2><span id='topic+processPostPredStats'></span>

<h3>Description</h3>

<p>Reads in and processes posterior-predictive statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processPostPredStats(path_sim, path_emp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processPostPredStats_+3A_path_sim">path_sim</code></td>
<td>
<p>(character string; no default) Path to the .csv file
containing the simulated data results</p>
</td></tr>
<tr><td><code id="processPostPredStats_+3A_path_emp">path_emp</code></td>
<td>
<p>(character string; no default) Path to the .csv file
containing the empirical values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data frames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example datasets to working directory

url_emp &lt;-
   "https://revbayes.github.io/tutorials/intro/data/empirical_data_pps_example.csv"
dest_path_emp &lt;- "empirical_data_pps_example.csv"
download.file(url_emp, dest_path_emp)

url_sim &lt;-
   "https://revbayes.github.io/tutorials/intro/data/simulated_data_pps_example.csv"
dest_path_sim &lt;- "simulated_data_pps_example.csv"
download.file(url_sim, dest_path_sim)

# to run on your own data, change this to the path to your data file
file_sim &lt;- dest_path_sim
file_emp &lt;- dest_path_emp

t &lt;- processPostPredStats(path_sim = file_sim,
                         path_emp = file_emp)

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_sim, dest_path_emp)



</code></pre>

<hr>
<h2 id='processSSE'>Title</h2><span id='topic+processSSE'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processSSE(
  path,
  speciation = "speciation",
  extinction = "extinction",
  speciation_hidden = "speciation_hidden",
  rates = c(speciation, extinction, "net-diversification"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processSSE_+3A_path">path</code></td>
<td>
<p>(vector of character strings; no default) File path(s) to
trace file.</p>
</td></tr>
<tr><td><code id="processSSE_+3A_speciation">speciation</code></td>
<td>
<p>(single character string; &quot;speciation&quot;) RevBayes variable
name</p>
</td></tr>
<tr><td><code id="processSSE_+3A_extinction">extinction</code></td>
<td>
<p>(single character string; &quot;extinction&quot;) RevBayes variable
name</p>
</td></tr>
<tr><td><code id="processSSE_+3A_speciation_hidden">speciation_hidden</code></td>
<td>
<p>(single character string; &quot;speciation_hidden&quot;)
RevBayes variable name</p>
</td></tr>
<tr><td><code id="processSSE_+3A_rates">rates</code></td>
<td>
<p>(vector; c(speciation, extinction, &quot;net-diversification&quot;))
names of rates to be included in plot</p>
</td></tr>
<tr><td><code id="processSSE_+3A_...">...</code></td>
<td>
<p>additional arguments passed to readTrace()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example dataset to working directory

url &lt;-
  "https://revbayes.github.io/tutorials/intro/data/primates_BiSSE_activity_period.log"
dest_path &lt;- "primates_BiSSE_activity_period.log"
download.file(url, dest_path)

# to run on your own data, change this to the path to your data file
bisse_file &lt;- dest_path

pdata &lt;- processSSE(bisse_file)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path)


</code></pre>

<hr>
<h2 id='readOBDP'>Read OBDP Outputs</h2><span id='topic+readOBDP'></span>

<h3>Description</h3>

<p>Reads and formats the outputs of an analysis with the Occurrence Birth Death Process (MCMC parameter 
inference + diversity estimation)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readOBDP(
  start_time_trace_file,
  popSize_distribution_matrices_file,
  trees_trace_file
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readOBDP_+3A_start_time_trace_file">start_time_trace_file</code></td>
<td>
<p>(character; no default)  Trace of the starting times along the MCMC chain.</p>
</td></tr>
<tr><td><code id="readOBDP_+3A_popsize_distribution_matrices_file">popSize_distribution_matrices_file</code></td>
<td>
<p>(character; no default)  Kt matrices computed with 'fnInferAncestralPopSize' in RevBayes.</p>
</td></tr>
<tr><td><code id="readOBDP_+3A_trees_trace_file">trees_trace_file</code></td>
<td>
<p>(character; no default)  Trace of the trees.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# first run readOBDP()
start_time_trace_file &lt;- 
     system.file("extdata", "obdp/start_time_trace.p", package="RevGadgets")
popSize_distribution_matrices_file &lt;- 
     system.file("extdata", "obdp/Kt_trace.p", package="RevGadgets")
trees_trace_file &lt;- 
     system.file("extdata", "obdp/mcmc_OBDP_trees.p", package="RevGadgets")
    
Kt_mean &lt;- readOBDP( start_time_trace_file=start_time_trace_file, 
                     popSize_distribution_matrices_file=popSize_distribution_matrices_file, 
                     trees_trace_file=trees_trace_file )

# then get the customized ggplot object with plotDiversityOBDP()
p &lt;- plotDiversityOBDP( Kt_mean,
                        xlab="Time (My)",
                        ylab="Number of lineages",
                        xticks_n_breaks=21,
                        col_Hidden="dodgerblue3",
                        col_LTT="gray25",
                        col_Total="forestgreen",
                        col_Hidden_interval="dodgerblue2",
                        col_Total_interval="darkolivegreen4",
                        palette_Hidden=c("transparent", "dodgerblue2", "dodgerblue3", 
                                         "dodgerblue4", "black"),
                        palette_Total=c("transparent", "green4", "forestgreen", "black"),
                        line_size=0.7,
                        interval_line_size=0.5,
                        show_Hidden=TRUE,
                        show_LTT=TRUE,
                        show_Total=TRUE,
                        show_intervals=TRUE,
                        show_densities=TRUE,
                        show_expectations=TRUE,
                        use_interpolate=TRUE )

# basic plot
p

# option: add a stratigraphic scale
library(deeptime)
library(ggplot2)
q &lt;- gggeo_scale(p, dat="periods", height=unit(1.3, "line"), abbrv=F, size=4.5, neg=T)
r &lt;- gggeo_scale(q, dat="epochs", height=unit(1.1, "line"), abbrv=F, size=3.5, neg=T, 
                    skip=c("Paleocene", "Pliocene", "Pleistocene", "Holocene"))
s &lt;- gggeo_scale(r, dat="stages", height=unit(1, "line"), abbrv=T, size=2.5, neg=T)
s

## End(Not run)

</code></pre>

<hr>
<h2 id='readTrace'>Read trace</h2><span id='topic+readTrace'></span>

<h3>Description</h3>

<p>Reads in MCMC log files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readTrace(
  paths,
  format = "simple",
  delim = "\t",
  burnin = 0.1,
  check.names = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readTrace_+3A_paths">paths</code></td>
<td>
<p>(vector of character strings; no default) File path(s) to trace
file.</p>
</td></tr>
<tr><td><code id="readTrace_+3A_format">format</code></td>
<td>
<p>(single character string; default = simple) Indicates type of
MCMC trace, complex indicates cases where trace contains vectors of vectors/
matrices - mnStochasticVariable monitor will sometimes be of this type.</p>
</td></tr>
<tr><td><code id="readTrace_+3A_delim">delim</code></td>
<td>
<p>(single character string; default = &quot;\t&quot;) Delimiter of file.</p>
</td></tr>
<tr><td><code id="readTrace_+3A_burnin">burnin</code></td>
<td>
<p>(single numeric value; default = 0.1) Fraction of generations
to  discard (if value provided is between 0 and 1) or number of generations
(if value provided is greater than 1).</p>
</td></tr>
<tr><td><code id="readTrace_+3A_check.names">check.names</code></td>
<td>
<p>(logical; default = FALSE) Passed to utils::read.table();
indicates if utils::read.table() should check column names and replace
syntactically invalid characters.</p>
</td></tr>
<tr><td><code id="readTrace_+3A_...">...</code></td>
<td>
<p>(various) Additional arguments passed to utils::read.table().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads in one or multiple MCMC log files from the same analysis
and discards a user-specified burn-in, compatible with multiple monitor
types. If the trace contains vectors of vectors and the user does not specify
format = &quot;complex&quot;, readTrace() will read in those columns as factors
rather than as numeric vectors.
</p>


<h3>Value</h3>

<p>List of dataframes (of length 1 if only 1 log file provided).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read and process a single trace file


# download the example dataset to working directory
url_gtr &lt;-
   "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR.log"
dest_path_gtr &lt;- "primates_cytb_GTR.log"
download.file(url_gtr, dest_path_gtr)

# to run on your own data, change this to the path to your data file
file_single &lt;- dest_path_gtr

one_trace &lt;- readTrace(paths = file_single)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_gtr)

# read and process multiple trace files, such as from multiple runs of
# the same analysis

# download the example dataset to working directory
url_1 &lt;-
"https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_run_1.log"
dest_path_1 &lt;- "primates_cytb_GTR_run_1.log"
download.file(url_1, dest_path_1)

url_2 &lt;-
"https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_run_2.log"
dest_path_2 &lt;- "primates_cytb_GTR_run_2.log"
download.file(url_2, dest_path_2)

# to run on your own data, change this to the path to your data file
file_1 &lt;- dest_path_1
file_2 &lt;- dest_path_2

# read in the multiple trace files
multi_trace &lt;- readTrace(path = c(file_1, file_2), burnin = 0.0)

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_1, dest_path_2)


</code></pre>

<hr>
<h2 id='readTrees'>Read trees</h2><span id='topic+readTrees'></span>

<h3>Description</h3>

<p>Reads in a tree file containing one or multiple trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readTrees(paths, tree_name = "psi", burnin = 0, n_cores = 1L, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readTrees_+3A_paths">paths</code></td>
<td>
<p>(vector of character strings; no default) File path(s) to
tree(s).</p>
</td></tr>
<tr><td><code id="readTrees_+3A_tree_name">tree_name</code></td>
<td>
<p>(character string; default psi) Name of the tree variable.</p>
</td></tr>
<tr><td><code id="readTrees_+3A_burnin">burnin</code></td>
<td>
<p>(single numeric value; default = 0.1) Fraction of generations
to discard (if value provided is between 0 and 1) or number of generations
(if value provided is greater than 1).</p>
</td></tr>
<tr><td><code id="readTrees_+3A_n_cores">n_cores</code></td>
<td>
<p>(integer; default 1) Number of cores for parallelizing.</p>
</td></tr>
<tr><td><code id="readTrees_+3A_verbose">verbose</code></td>
<td>
<p>(logical; default true) Display a status bar?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads in a tree file in either nexus or newick format, and containing a
single tree or multiple trees (as in the results of a Bayesian analysis).
For reading in annotated tree files of continuous character evolution,
the parameter must be considered a node parameter rather than branch
parameter. Set isNodeParameter = TRUE in the extended newick monitor
(mnExtNewick)
</p>


<h3>Value</h3>

<p>A list (across runs) of lists (across samples) of treedata objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read in a single nexus file

# download the example dataset to working directory
url_nex &lt;-
 "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_MAP.tre"
dest_path_nex &lt;- "primates_cytb_GTR_MAP.tre"
download.file(url_nex, dest_path_nex)

# to run on your own data, change this to the path to your data file
file &lt;- dest_path_nex
tree_single_old &lt;- readTrees(paths = file)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_nex)

# read in a single newick string

# download the example dataset to working directory
url_new &lt;-
 "https://revbayes.github.io/tutorials/intro/data/primates.tre"
dest_path_new &lt;- "primates.tre"
download.file(url_new, dest_path_new)

# to run on your own data, change this to the path to your data file
file_new &lt;- dest_path_new
tree_new &lt;- readTrees(paths = file_new)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_new)


# read in a tree trace (may take a few seconds)

# download the example dataset to working directory
url_multi &lt;-
 "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR.trees"
dest_path_multi &lt;- "primates_cytb_GTR.trees"
download.file(url_multi, dest_path_multi)

# to run on your own data, change this to the path to your data file
file_multi &lt;- dest_path_multi
tree_multi &lt;- readTrees(paths = file_multi)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_multi)


</code></pre>

<hr>
<h2 id='removeBurnin'>Remove Burnin</h2><span id='topic+removeBurnin'></span>

<h3>Description</h3>

<p>Removes burnin from MCMC trace
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeBurnin(trace, burnin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeBurnin_+3A_trace">trace</code></td>
<td>
<p>(list of data frames; no default) Name of a list of data frames,
such as produced by readTrace().</p>
</td></tr>
<tr><td><code id="removeBurnin_+3A_burnin">burnin</code></td>
<td>
<p>(single numeric value; 0.1) Fraction of generations to
discard (if value provided is between 0 and 1) or number of generations (if
value provided is greater than 1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Removes burnin from an MCMC trace, such as the output of readTrace(). If
multiple traces are provided, this function will remove the burnin from each.
</p>


<h3>Value</h3>

<p>List of dataframes (of length 1 if only 1 log file provided).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# download the example dataset to working directory
url_gtr &lt;-
    "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR.log"
dest_path_gtr &lt;- "primates_cytb_GTR.log"
download.file(url_gtr, dest_path_gtr)

# to run on your own data, change this to the path to your data file
file_single &lt;- dest_path_gtr

one_trace &lt;- readTrace(paths = file_single)
one_trace_burnin &lt;- removeBurnin(trace = one_trace, burnin = 0.1)

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_gtr)


</code></pre>

<hr>
<h2 id='rerootPhylo'>Reroot Phylo</h2><span id='topic+rerootPhylo'></span>

<h3>Description</h3>

<p>Reroots a phylogeny given an outgroup taxon or clade
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rerootPhylo(tree, outgroup)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rerootPhylo_+3A_tree">tree</code></td>
<td>
<p>(list of lists of treedata objects; no default) Name of a list of
lists of treedata objects, such as produced by readTrees().</p>
</td></tr>
<tr><td><code id="rerootPhylo_+3A_outgroup">outgroup</code></td>
<td>
<p>(character, no default) Name of the outgroup(s). Either a
single taxon name or a character vector of length two to specify a clade;
in this case the root will be placed at the midpoint of the branch subtending
the two taxa's MRCA. Modified from phytools::reroot().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Modifies a tree object by rerooting using a specified
outgroup taxon or clade. Places the root at the midpoint
of the branch subtending the outgroup. If the input
contains multiple trees, all trees will be rerooted.
</p>


<h3>Value</h3>

<p>returns a list of list of treedata objects, with the trees rooted.
</p>


<h3>See Also</h3>

<p>phytools: <a href="phytools.html#topic+reroot">reroot</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file &lt;- system.file("extdata",
                    "sub_models/primates_cytb_GTR_MAP.tre",
                    package="RevGadgets")
tree &lt;- readTrees(paths = file)
# root with one taxon
tree_rooted &lt;- rerootPhylo(tree = tree, outgroup = "Galeopterus_variegatus")
# root with clade, specified by two taxa
tree_rooted &lt;- rerootPhylo(tree = tree,
                           outgroup = c("Varecia_variegata_variegata",
                                        "Propithecus_coquereli"))

</code></pre>

<hr>
<h2 id='RevGadgets'>RevGadgets</h2><span id='topic+RevGadgets'></span>

<h3>Description</h3>

<p>This package provides functions to process and plot the output of RevBayes
analyses.
</p>

<hr>
<h2 id='setMRFGlobalScaleHyperpriorNShifts'>Sets a global scale parameter for a GMRF or HSMRF model given a prior
mean number of effective shifts.</h2><span id='topic+setMRFGlobalScaleHyperpriorNShifts'></span>

<h3>Description</h3>

<p>This function finds the global scale parameter value that produces the
desired prior mean number of &quot;effective&quot; rate shifts. Given a specified
magnitude for an effective shift, shift_size, an effective shift occurs
when two adjacent values are more than shift_size-fold apart from each other.
That is, an effective shift is the event that rate[i+1]/rate[i] &gt; shift_size
or rate[i+1]/rate[i] &lt; 1/shift_size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setMRFGlobalScaleHyperpriorNShifts(
  n_episodes,
  model,
  prior_n_shifts = log(2),
  shift_size = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setMRFGlobalScaleHyperpriorNShifts_+3A_n_episodes">n_episodes</code></td>
<td>
<p>(numeric; no default) The number of episodes in the random
field (the parameter vector will be this long).</p>
</td></tr>
<tr><td><code id="setMRFGlobalScaleHyperpriorNShifts_+3A_model">model</code></td>
<td>
<p>(character; no default) What model should the global scale
parameter be set for? Options are &quot;GMRF&quot; and &quot;HSMRF&quot; for first-order models
(also allowable: &quot;GMRF1&quot; and &quot;HSMRF1&quot;) and &quot;GMRF2&quot; and HSMRF2&quot; for
second-order models.</p>
</td></tr>
<tr><td><code id="setMRFGlobalScaleHyperpriorNShifts_+3A_prior_n_shifts">prior_n_shifts</code></td>
<td>
<p>(numeric; log(2)) The desired prior mean number of
shifts.</p>
</td></tr>
<tr><td><code id="setMRFGlobalScaleHyperpriorNShifts_+3A_shift_size">shift_size</code></td>
<td>
<p>(numeric; 2) The magnitude of change that defines an
effective shift (measured as a fold-change).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Finding these values for a HSMRF model can take several seconds for large
values of n_episodes because of the required numerical integration.
</p>


<h3>Value</h3>

<p>The hyperprior.
</p>


<h3>References</h3>

<p>Magee et al. (2019) Locally adaptive Bayesian birth-death model
successfully detects slow and rapid rate shifts.
doi: https://doi.org/10.1101/853960
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get global scale for a HSMRF model with 100 episodes.
gs &lt;- setMRFGlobalScaleHyperpriorNShifts(100, "HSMRF")

# Plot a draw from this HSMRF distribution

trajectory &lt;- simulateMRF(n_episodes = 100,
                          model = "HSMRF",
                          global_scale_hyperprior = gs)

plot(1:100,
     rev(trajectory),
     type = "l",
     xlab = "time",
     ylab = "speciation rate")

</code></pre>

<hr>
<h2 id='simulateMRF'>Simulates a single Markov random field trajectory.</h2><span id='topic+simulateMRF'></span>

<h3>Description</h3>

<p>This function simulates a draw from a HSMRF or GMRF distribution given a
user-specified global scale parameter. The MRF can be taken to be on the
log-scale (such as for a birth rate) or the real-scale. The first value
must be specified
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateMRF(
  n_episodes,
  model,
  global_scale_hyperprior,
  initial_value = NULL,
  exponentiate = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateMRF_+3A_n_episodes">n_episodes</code></td>
<td>
<p>(numeric; no default) The number of episodes in the random
field (the parameter vector will be this long).</p>
</td></tr>
<tr><td><code id="simulateMRF_+3A_model">model</code></td>
<td>
<p>(character; no default) What model should the global scale
parameter be set for? Options are &quot;GMRF&quot; and &quot;HSMRF&quot;.</p>
</td></tr>
<tr><td><code id="simulateMRF_+3A_global_scale_hyperprior">global_scale_hyperprior</code></td>
<td>
<p>(numeric; no default) The hyperprior on the
global scale parameter.</p>
</td></tr>
<tr><td><code id="simulateMRF_+3A_initial_value">initial_value</code></td>
<td>
<p>(numeric; NULL) The first value in the MRF. If no value
is specified, the field is assumed to start at 0 (if exponentiate=FALSE) or
1 (if exponentiate=TRUE).</p>
</td></tr>
<tr><td><code id="simulateMRF_+3A_exponentiate">exponentiate</code></td>
<td>
<p>(logical; TRUE) If TRUE, the MRF model is taken to be on
the log-scale and the values are returned on the real-scale (note this means
that the specified initial value will be the log of the true initial value).
If FALSE, the model is taken to be on the real scale.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector drawn from the specified MRF model on the specified
(log- or real-) scale.
</p>


<h3>References</h3>

<p>Magee et al. (2020) Locally adaptive Bayesian birth-death model
successfully detects slow and rapid rate shifts.
<em>PLoS Computational Biology</em>, <b>16 (10)</b>: e1007999.
</p>
<p>Faulkner, James R., and Vladimir N. Minin. Locally adaptive smoothing with
Markov random fields and shrinkage priors.
<em>Bayesian analysis</em>, <b>13 (1)</b>, 225.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulate a 100-episode HSMRF model for a speciation-rate through time
trajectory &lt;- simulateMRF(n_episodes = 100,
                          model = "HSMRF",
                          global_scale_hyperprior = 0.0021)
plot(1:100,
     rev(trajectory),
     type = "l",
     xlab = "time",
     ylab = "speciation rate")

</code></pre>

<hr>
<h2 id='summarizeTrace'>Summarize trace</h2><span id='topic+summarizeTrace'></span>

<h3>Description</h3>

<p>Summarizes trace file(s) that have been read into memory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarizeTrace(trace, vars)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarizeTrace_+3A_trace">trace</code></td>
<td>
<p>(list of data frames; no default) Name of a list of data frames,
such as produced by readTrace(). If the readTrace() output
contains multiple traces (such as from multiple runs), summarizeTrace() will
provide summaries for each trace individually, as well as the combined trace.</p>
</td></tr>
<tr><td><code id="summarizeTrace_+3A_vars">vars</code></td>
<td>
<p>(character or character vector; no default) The name of the
variable(s) to be summarized.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summarizes a trace file for continuous or discrete characters by
computing the mean and 95% credible interval for quantitative
character and the 95% credible set for discrete characters.
</p>


<h3>Value</h3>

<p>summarizeTrace() returns a list of the length of provided variables.
For quantitative variables, it returns the mean and 95
For discrete variables, it returns the 95
associated probabilities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# continuous character only example, one run

# download the example dataset to working directory
url_gtr &lt;-
    "https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR.log"
dest_path_gtr &lt;- "primates_cytb_GTR.log"
download.file(url_gtr, dest_path_gtr)

# to run on your own data, change this to the path to your data file
file_single &lt;- dest_path_gtr

one_trace &lt;- readTrace(paths = file_single)
trace_sum &lt;- summarizeTrace(trace = one_trace,
                            vars = c("pi[1]","pi[2]","pi[3]","pi[4]"))
trace_sum[["pi[1]"]]

# remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_gtr)

# continuous character example, multiple runs

#' # download the example dataset to working directory
url_1 &lt;-
"https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_run_1.log"
dest_path_1 &lt;- "primates_cytb_GTR_run_1.log"
download.file(url_1, dest_path_1)

url_2 &lt;-
"https://revbayes.github.io/tutorials/intro/data/primates_cytb_GTR_run_2.log"
dest_path_2 &lt;- "primates_cytb_GTR_run_2.log"
download.file(url_2, dest_path_2)

# to run on your own data, change this to the path to your data file
file_1 &lt;- dest_path_1
file_2 &lt;- dest_path_2

# read in the multiple trace files
multi_trace &lt;- readTrace(path = c(file_1, file_2), burnin = 0.0)

trace_sum_multi &lt;- summarizeTrace(trace = multi_trace,
                                  vars = c("pi[1]","pi[2]","pi[3]","pi[4]"))
trace_sum_multi[["pi[1]"]]

# remove files
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_1, dest_path_2)

# discrete character example

# download the example dataset to working directory
url_rj &lt;- "https://revbayes.github.io/tutorials/intro/data/freeK_RJ.log"
dest_path_rj &lt;- "freeK_RJ.log"
download.file(url_rj, dest_path_rj)

file &lt;- dest_path_rj
trace &lt;- readTrace(path = file)

trace_sum_discrete &lt;- summarizeTrace(trace = trace,
                                     vars = c("prob_rate_12",
                                              "prob_rate_13",
                                              "prob_rate_31",
                                              "prob_rate_32"))
trace_sum_discrete[["prob_rate_12"]]

#' # remove file
# WARNING: only run for example dataset!
# otherwise you might delete your data!
file.remove(dest_path_rj)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
