<!DOCTYPE html><html><head><title>Help for package RFclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RFclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gbm'>
<p>Multi-omic profiling of glioblastoma samples</p></a></li>
<li><a href='#RFCluster'>
<p>A wrapper for Random Forest Consensus Clustering</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Random Forest Cluster Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Ankur Chakravarthy, PhD</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ankur Chakravarthy &lt;ankur.chakravarthy.10@ucl.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to perform random forest consensus clustering of different data types. The package is designed to accept a list of matrices from different assays, typically from high-throughput molecular profiling so that class discovery may be jointly performed. For references, please see Tao Shi &amp; Steve Horvath (2006) &lt;<a href="https://doi.org/10.1198%2F106186006X94072">doi:10.1198/106186006X94072</a>&gt; &amp; Monti et al (2003) &lt;<a href="https://doi.org/10.1023%2FA%3A1023949509487">doi:10.1023/A:1023949509487</a>&gt; .</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>ConsensusClusterPlus,randomForest</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-21 00:11:32 UTC; ankur</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-21 02:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gbm'>
Multi-omic profiling of glioblastoma samples
</h2><span id='topic+gbm'></span>

<h3>Description</h3>

<p>These data serve as an example dataset to execute RFCluster on.These were processed and originally included in the iCluster R package which is likely to be archived on CRAN. I
have reincluded this dataset here to permit the example to be run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gbm")</code></pre>


<h3>Format</h3>

<p>A list of matrices containing Copy Number, Methylation and Expression estimates for 55 GBMs for 1500-1800 genes. Data were originally derived by The Cancer Genome Atlas project.
</p>


<h3>Source</h3>

<p>https://doi.org/10.1093/bioinformatics/btp659
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gbm)
</code></pre>

<hr>
<h2 id='RFCluster'>
A wrapper for Random Forest Consensus Clustering
</h2><span id='topic+RFCluster'></span>

<h3>Description</h3>

<p>This takes a list of matrices of different data types , features in rows, samples in columns, and performs random forest clustering (one-dimensional). When multiple data types are available this is one way of modelling the data together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFCluster(Data, ClustAlg = "pam", MaxK, nTrees = 1000,
exportFigures = "pdf", ClustReps = 500, ProjectName = "RFCluster",
verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFCluster_+3A_data">Data</code></td>
<td>

<p>Named list, contains matrices with samples in columns, features in rows. The names of the list should represent the platform or the feature type, such as expression, or CN, or clin; as long as it is distinct.
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_clustalg">ClustAlg</code></td>
<td>

<p>Algorithm for consensus clustering
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_maxk">MaxK</code></td>
<td>

<p>Maximum number of clusters you are searching for
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_ntrees">nTrees</code></td>
<td>

<p>How many trees are we using in the random forest to generate a proximity matrix?
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_projectname">ProjectName</code></td>
<td>

<p>Name of the project, to annotate plots and other output
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_clustreps">ClustReps</code></td>
<td>

<p>Number of replicates for consensus clustering
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_verbose">verbose</code></td>
<td>

<p>Should output be verbose?
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_exportfigures">exportFigures</code></td>
<td>

<p>Format of the results file for figures et cetera to be exported to
</p>
</td></tr>
<tr><td><code id="RFCluster_+3A_...">...</code></td>
<td>

<p>Other optional arguments, passed onto ConsensusClusterPlus; see that package's documentation for a full set.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Standard output for ConsensusClusterPlus runs.
</p>


<h3>Author(s)</h3>

<p>Ankur Chakravarthy, PhD
</p>


<h3>References</h3>

<p>Monti, S., Tamayo, P., Mesirov, J. et al. Machine Learning (2003) 52: 91. https://doi.org/10.1023/A:1023949509487
</p>
<p>Tao Shi &amp; Steve Horvath (2006) Unsupervised Learning With Random Forest Predictors, Journal of Computational and Graphical Statistics, 15:1, 118-138, DOI: 10.1198/106186006X94072
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RFclust)

#Get GBM example data from the iCluster package, repackaged to maintain CRAN compatibility
data(gbm)

#Transpose so columns are samples and features are rows
gbm.t &lt;- lapply(gbm, t)

#Make sure the sample names are the same across the matrices for the different
#samples - the code breaks otherwise

colnames(gbm.t[[2]]) &lt;- colnames(gbm.t[[3]]) &lt;- colnames(gbm.t[[1]])

#Run function on that dataset - these methods are computationally intensive
#so automatic testing during build has been disabled (takes &gt; 5s).
#Users may test the software by running the code separately as the example is reproducible

Test.cluster &lt;- RFCluster(Data = gbm.t, ClustAlg = "pam", MaxK = 5,
nTrees = 10, ProjectName = "RFCluster_Test", ClustReps = 50 , writeTable = FALSE, plot = NULL)
unlink("RFCluster_Test",recursive = TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
