<!DOCTYPE html><html><head><title>Help for package RFLPtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RFLPtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BLASTdata'><p> Example data set for BLAST data</p></a></li>
<li><a href='#diffDist'><p> Distance Matrix Computation</p></a></li>
<li><a href='#FragMatch'>
<p>Compute matches for RFLP data via FragMatch.</p></a></li>
<li><a href='#germ'>
<p>Compute matches for RFLP data via GERM.</p></a></li>
<li><a href='#linCombDist'><p> Linear Combination of Distances</p></a></li>
<li><a href='#newDataGerm'><p> Example data set from GERM software</p></a></li>
<li><a href='#nrBands'>
<p>Function to compute number of bands.</p></a></li>
<li><a href='#read.blast'>
<p>Read BLAST data</p></a></li>
<li><a href='#read.rflp'>
<p>Read RFLP data</p></a></li>
<li><a href='#refDataGerm'><p> Example data set from GERM software</p></a></li>
<li><a href='#RFLPcombine'>
<p>Combine RFLP data sets</p></a></li>
<li><a href='#RFLPdata'><p> Example data set for RFLP data</p></a></li>
<li><a href='#RFLPdist'>
<p>Compute distances for RFLP data.</p></a></li>
<li><a href='#RFLPdist2'>
<p>Compute distances for RFLP data.</p></a></li>
<li><a href='#RFLPdist2ref'>
<p>Compute distance between RFLP data and RFLP reference data.</p></a></li>
<li><a href='#RFLPlod'>
<p>Remove bands below LOD</p></a></li>
<li><a href='#RFLPplot'>
<p>Function to plot RFLP data.</p></a></li>
<li><a href='#RFLPqc'>
<p>Quality control for RFLP data</p></a></li>
<li><a href='#RFLPref'><p> Example data set for RFLP reference</p></a></li>
<li><a href='#RFLPrefplot'>
<p>Function for a visual comparison of RFLP samples with reference samples.</p></a></li>
<li><a href='#RFLPtools-package'>
<p>Tools To Analyse RFLP-Data</p></a></li>
<li><a href='#sim2dist'>
<p>Convert similarity matrix to dist object.</p></a></li>
<li><a href='#simMatrix'>
<p>Similarity matrix for BLAST data.</p></a></li>
<li><a href='#simulateRFLPdata'>
<p>Simulate RFLP data.</p></a></li>
<li><a href='#write.hclust'>
<p>Cut a hierarchical cluster tree and write cluster identifiers to a</p>
text file.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools to Analyse RFLP Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-02-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Fabienne Flessa [aut],
  Alexandra Kehl <a href="https://orcid.org/0000-0002-1547-0713"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Mohammed Aslam Imtiaz [aut],
  Matthias Kohl <a href="https://orcid.org/0000-0001-9514-8910"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matthias Kohl &lt;Matthias.Kohl@stamats.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to analyse DNA fragment samples (i.e. derived from RFLP-analysis) and standalone BLAST report files (i.e. DNA sequence analysis).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 4.0.0), RColorBrewer</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, graphics, grDevices</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, lattice, MKomics</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-08 09:24:24 UTC; kohlm</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-08 09:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BLASTdata'> Example data set for BLAST data </h2><span id='topic+BLASTdata'></span>

<h3>Description</h3>

<p>This is an example data set for BLAST data generated with standalone BLAST from NCBI. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RFLPdata)</code></pre>


<h3>Format</h3>

<p>A data frame with 737 observations on the following four variables
</p>

<dl>
<dt><code>query.id</code></dt><dd><p>character: sequence identifier. </p>
</dd>
<dt><code>subject.id</code></dt><dd><p>character: subject identifier. </p>
</dd>
<dt><code>identity</code></dt><dd><p>numeric: identity between sequences (in percent). </p>
</dd>
<dt><code>alignment.length</code></dt><dd><p>integer: number of nucleotides. </p>
</dd>
<dt><code>mismatches</code></dt><dd><p>integer: number of mismatches. </p>
</dd>
<dt><code>gap.opens</code></dt><dd><p>integer: number of gaps. </p>
</dd>
<dt><code>q.start</code></dt><dd><p>integer: query sequence start. </p>
</dd>
<dt><code>q.end</code></dt><dd><p>integer: query sequence end. </p>
</dd>
<dt><code>s.start</code></dt><dd><p>integer: subject sequence start. </p>
</dd>
<dt><code>s.end</code></dt><dd><p>integer: subject sequence end. </p>
</dd>
<dt><code>evalue</code></dt><dd><p>numeric: evalue. </p>
</dd>
<dt><code>bit.score</code></dt><dd><p>numeric: score value. </p>
</dd>
</dl>



<h3>Details</h3>

<p>The data was generated with standalone BLAST from NCBI.
Pairwise similarities of DNA sequences are calculated among all sequences 
to analyse applying Standalone Blast with the parameters -m 8 -r 2 -G 5 -E 2.
</p>
<p>Alternatively data can be generated with &quot;local BLAST&quot; implemented in 
BioEdit v7.0.9 using the additional parameters -m 8 -r 2 -G 5 -E 2 and 
by selecting &quot;open output&quot; and &quot;tabular output&quot;.
</p>


<h3>Source</h3>

<p>The data set was generated by F. Flessa.
</p>


<h3>References</h3>

<p>Standalone Blast download: <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
<p>Blast News: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews</a>
</p>
<p>BioEdit: <a href="https://bioedit.software.informer.com/">https://bioedit.software.informer.com/</a>
</p>
<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BLASTdata)
str(BLASTdata)
</code></pre>

<hr>
<h2 id='diffDist'> Distance Matrix Computation </h2><span id='topic+diffDist'></span>

<h3>Description</h3>

<p>This function computes and returns the distance matrix computed by
using the specified distance measure to compute the distances between
the rows of a data matrix. Instead of the row values as in the case of 
<code><a href="stats.html#topic+dist">dist</a></code>, the successive differences of the row values
are used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diffDist(x, method = "euclidean", diag = FALSE, upper = FALSE, p = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diffDist_+3A_x">x</code></td>
<td>
<p>a numeric matrix, data frame or <code>"dist"</code> object.</p>
</td></tr>
<tr><td><code id="diffDist_+3A_method">method</code></td>
<td>
<p>the distance measure to be used. This must be one of
<code>"euclidean"</code>, <code>"maximum"</code>, <code>"manhattan"</code>,
<code>"canberra"</code>, <code>"binary"</code> or <code>"minkowski"</code>.
Any unambiguous substring can be given.</p>
</td></tr>
<tr><td><code id="diffDist_+3A_diag">diag</code></td>
<td>
<p>logical value indicating whether the diagonal of the
distance matrix should be printed by <code>print.dist</code>.</p>
</td></tr>
<tr><td><code id="diffDist_+3A_upper">upper</code></td>
<td>
<p>logical value indicating whether the upper triangle of the
distance matrix should be printed by <code>print.dist</code>.</p>
</td></tr>
<tr><td><code id="diffDist_+3A_p">p</code></td>
<td>
<p>The power of the Minkowski distance.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes and returns the distance matrix computed by
using the specified distance measure to compute the distances between
the rows of a data matrix. Instead of the row values as in the case of 
<code><a href="stats.html#topic+dist">dist</a></code>, the successive differences of the row values
are used.
</p>
<p>It's a simple wrapper function arround <code><a href="stats.html#topic+dist">dist</a></code>. For
more details about the distances we refer to <code><a href="stats.html#topic+dist">dist</a></code>.
</p>
<p>The function may be helpful, if there is a shift w.r.t.\ the measured
bands; e.g.\ <code>c(550, 500, 300, 250)</code> vs.\ <code>c(510, 460, 260, 210)</code>.
</p>


<h3>Value</h3>

<p><code>diffDist</code> returns an object of class <code>"dist"</code>; cf. <code><a href="stats.html#topic+dist">dist</a></code>.
</p>


<h3>Author(s)</h3>

<p> Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a></p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## assume a shift in the measured bands
M &lt;- rbind(c(550, 500, 300, 250), c(510, 460, 260, 210),
           c(550, 500, 300, 200))
dist(M)
diffDist(M)
</code></pre>

<hr>
<h2 id='FragMatch'>
Compute matches for RFLP data via FragMatch.
</h2><span id='topic+FragMatch'></span>

<h3>Description</h3>

<p>Compute matches for RFLP data using FragMatch - a program for the analysis of DNA 
fragment data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FragMatch(newData, refData, maxValue = 1000, errorBound = 25,
          weight = 1, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FragMatch_+3A_newdata">newData</code></td>
<td>
<p> data.frame with new RFLP data; see <code><a href="#topic+newDataGerm">newDataGerm</a></code>.</p>
</td></tr>
<tr><td><code id="FragMatch_+3A_refdata">refData</code></td>
<td>
<p> data.frame with reference RFLP data; see <code><a href="#topic+refDataGerm">refDataGerm</a></code>.</p>
</td></tr>
<tr><td><code id="FragMatch_+3A_maxvalue">maxValue</code></td>
<td>
<p> numeric: maximum value for which the error bound is applied. 
Can be a vector of length larger than 1.</p>
</td></tr>
<tr><td><code id="FragMatch_+3A_errorbound">errorBound</code></td>
<td>
<p> numeric: error bound corresponding to <code>maxValue</code>. 
Can be a vector of length larger than 1.</p>
</td></tr>
<tr><td><code id="FragMatch_+3A_weight">weight</code></td>
<td>
<p> numeric: weight for weighting partial matches; see details section.</p>
</td></tr>
<tr><td><code id="FragMatch_+3A_na.rm">na.rm</code></td>
<td>
<p> logical: indicating whether NA values should be stripped before the 
computation proceeds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A rather simple algorithm which consists of counting the number of matches where
it is considered a match if the value is inside a range of +/- <code>errorBound</code>.
</p>
<p>If there is more than one enzyme, one can use weights to give the partial perfect 
matches for a certain enzyme a higher (or also smaller) weight.
</p>


<h3>Value</h3>

<p>A character matrix with entries of the form <code>"a_b"</code> which means that there
were <code>a</code> out of <code>b</code> possible matches.
</p>


<h3>Author(s)</h3>

 
<p>Mohammed Aslam Imtiaz, Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>T. A. Saari,  S. K. Saari, C. D. Campbell, I. J Alexander, I. C. Anderson.
FragMatch - a program for the analysis of DNA fragment data.
Mycorrhiza 2007, 17:133-136
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+newDataGerm">newDataGerm</a></code>, <code><a href="#topic+refDataGerm">refDataGerm</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(refDataGerm)
  data(newDataGerm)
  
  res &lt;- FragMatch(newDataGerm, refDataGerm)
</code></pre>

<hr>
<h2 id='germ'>
Compute matches for RFLP data via GERM.
</h2><span id='topic+germ'></span>

<h3>Description</h3>

<p>Compute matches for RFLP data using the Good-Enough RFLP Matcher (GERM) program.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>germ(newData, refData, parameters = list("Max forward error" = 25,
                                         "Max backward error" = 25,
                                         "Max sum error" = 100,
                                         "Lower measurement limit" = 100), 
     method = "joint", na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="germ_+3A_newdata">newData</code></td>
<td>
<p> data.frame with new RFLP data; see <code><a href="#topic+newDataGerm">newDataGerm</a></code>.</p>
</td></tr>
<tr><td><code id="germ_+3A_refdata">refData</code></td>
<td>
<p> data.frame with reference RFLP data; see <code><a href="#topic+refDataGerm">refDataGerm</a></code>.</p>
</td></tr>
<tr><td><code id="germ_+3A_parameters">parameters</code></td>
<td>
<p> list of the four program parameters of GERM; see details section.</p>
</td></tr>
<tr><td><code id="germ_+3A_method">method</code></td>
<td>
<p> matching and ranking method used for computation; see details section.</p>
</td></tr>
<tr><td><code id="germ_+3A_na.rm">na.rm</code></td>
<td>
<p> logical: indicating whether NA values should be stripped before the computation proceeds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are four matching and ranking methods which are <code>"joint"</code>, <code>"forward"</code>, 
<code>"backward"</code>, and <code>"sum"</code>. For more details see Dickie et al. (2003).
</p>
<p>The parameters of the GERM software are:
<code>"Max forward error"</code>: Used if &quot;matching and ranking method&quot; is set to <code>"forward"</code> or <code>"joint"</code>.
<code>"Max backward error"</code>: Used if &quot;matching and ranking method&quot; is set to <code>"backward"</code> or <code>"joint"</code>.
<code>"Max sum error"</code>: Used for matching if &quot;matching and ranking method&quot; is set to <code>"sum"</code>.
<code>"Lower measurement limit"</code>: The lower bound of measurements (often 100 or 50, depending on ladder used).
</p>


<h3>Value</h3>

<p>A named list with the results.
</p>


<h3>Author(s)</h3>

 
<p>Mohammed Aslam Imtiaz, Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Ian A. Dickie, Peter G. Avis, David J. McLaughlin, Peter B. Reich. 
Good-Enough RFLP Matcher (GERM) program.
Mycorrhiza 2003, 13:171-172.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+newDataGerm">newDataGerm</a></code>, <code><a href="#topic+refDataGerm">refDataGerm</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(refDataGerm)
  data(newDataGerm)
  
  ## Example 1
  res1 &lt;- germ(newDataGerm[1:7,], refDataGerm)
  
  ## Example 2
  res2 &lt;- germ(newDataGerm[8:15,], refDataGerm)
  
  ## Example 3
  res3 &lt;- germ(newDataGerm[16:20,], refDataGerm)
  
  ## all three examples in one step
  res.all &lt;- germ(newDataGerm, refDataGerm)
</code></pre>

<hr>
<h2 id='linCombDist'> Linear Combination of Distances </h2><span id='topic+linCombDist'></span>

<h3>Description</h3>

<p>This function computes linear combinations of distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linCombDist(x, distfun1, w1, distfun2, w2, diag = FALSE, upper = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linCombDist_+3A_x">x</code></td>
<td>
<p>object which is passed to <code>distfun1</code> and <code>distfun2</code>.</p>
</td></tr>
<tr><td><code id="linCombDist_+3A_distfun1">distfun1</code></td>
<td>
<p>function used to compute an object of class <code>"dist"</code>.</p>
</td></tr>
<tr><td><code id="linCombDist_+3A_w1">w1</code></td>
<td>
<p>weight for result of <code>distfun1</code>.</p>
</td></tr>
<tr><td><code id="linCombDist_+3A_distfun2">distfun2</code></td>
<td>
<p>function used to compute an object of class <code>"dist"</code>.</p>
</td></tr>
<tr><td><code id="linCombDist_+3A_w2">w2</code></td>
<td>
<p>weight for result of <code>distfun2</code>.</p>
</td></tr>
<tr><td><code id="linCombDist_+3A_diag">diag</code></td>
<td>
<p> see <code><a href="stats.html#topic+dist">dist</a></code> </p>
</td></tr>
<tr><td><code id="linCombDist_+3A_upper">upper</code></td>
<td>
<p> see <code><a href="stats.html#topic+dist">dist</a></code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes and returns the distance matrix computed by
a linear combination of two distance matrices.
</p>


<h3>Value</h3>

<p><code>linCombDist</code> returns an object of class <code>"dist"</code>; cf. <code><a href="stats.html#topic+dist">dist</a></code>.
</p>


<h3>Author(s)</h3>

<p> Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a></p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## assume a shift in the measured bands
M &lt;- rbind(c(550, 500, 300, 250), c(510, 460, 260, 210),
           c(700, 650, 450, 400), c(550, 490, 310, 250))
dist(M)
diffDist(M)

## convex combination of dist and diffDist
linCombDist(M, distfun1 = dist, w1 = 0.5, distfun2 = diffDist, w2 = 0.5)

## linear combination
linCombDist(M, distfun1 = dist, w1 = 2, distfun2 = diffDist, w2 = 5)

## maximum distance
linCombDist(M, distfun1 = function(x) dist(x, method = "maximum"), w1 = 0.5, 
            distfun2 = function(x) diffDist(x, method = "maximum"), w2 = 0.5)
            
data(RFLPdata)
distfun &lt;- function(x) linCombDist(x, distfun1 = dist, w1 = 0.1, distfun2 = diffDist, w2 = 0.9)
par(mfrow = c(2, 2))
plot(hclust(RFLPdist(RFLPdata, nrBands = 3, distfun = distfun)), cex = 0.7, cex.lab = 0.7)
RFLPplot(RFLPdata, nrBands = 3, distfun = distfun, mar.bottom = 6, cex.axis = 0.8)
plot(hclust(RFLPdist(RFLPdata, nrBands = 3)), cex = 0.7, cex.lab = 0.7)
RFLPplot(RFLPdata, nrBands = 3, mar.bottom = 6, cex.axis = 0.8)
</code></pre>

<hr>
<h2 id='newDataGerm'> Example data set from GERM software </h2><span id='topic+newDataGerm'></span>

<h3>Description</h3>

<p>This is the reference data taken from the GERM software. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(newDataGerm)</code></pre>


<h3>Format</h3>

<p>A data frame with 20 observations on the following six variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Enzyme</code></dt><dd><p>character: enzyme used.</p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Genus</code></dt><dd><p>character: genus of sample. </p>
</dd>
<dt><code>Species</code></dt><dd><p>character: species of sample. </p>
</dd>
</dl>



<h3>Details</h3>

<p>See GERM software.
</p>


<h3>Source</h3>

<p>The data set was taken from the GERM software (table 'Example Unknowns').
</p>


<h3>References</h3>

<p>Ian A. Dickie, Peter G. Avis, David J. McLaughlin, Peter B. Reich. 
Good-Enough RFLP Matcher (GERM) program.
Mycorrhiza 2003, 13:171-172.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(newDataGerm)
str(newDataGerm)
</code></pre>

<hr>
<h2 id='nrBands'>
Function to compute number of bands.
</h2><span id='topic+nrBands'></span>

<h3>Description</h3>

<p>Computes groups based on the number of bands per sample in a RFLP data set.
Each group comprises RFLP-samples with equal number of bands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nrBands(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nrBands_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data; see <code><a href="#topic+RFLPdata">RFLPdata</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function computes groups based on the number of bands per sample 
in a RFLP data set. Each group comprises RFLP-samples with equal 
number of bands.
</p>


<h3>Value</h3>

<p> Number of bands per RFLP-samples. </p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="#topic+RFLPdist2">RFLPdist2</a></code>, <code><a href="stats.html#topic+dist">dist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
nrBands(RFLPdata)
</code></pre>

<hr>
<h2 id='read.blast'>
Read BLAST data
</h2><span id='topic+read.blast'></span>

<h3>Description</h3>

<p>Function to read BLAST data generated with standalone BLAST from NCBI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.blast(file, sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.blast_+3A_file">file</code></td>
<td>
<p> character: BLAST file to read in.</p>
</td></tr>
<tr><td><code id="read.blast_+3A_sep">sep</code></td>
<td>
<p> the field separator character. Values on each line of the file are 
separated by this character. Default <code>"\t"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function reads data which was generated with standalone BLAST from NCBI;
see ftp://ftp.ncbi.nih.gov/blast/executables/release/.
</p>
<p>Possible steps:<br />
1) Install NCBI BLAST<br />
2) Generate and import database(s)<br />
3) Apply BLAST with options <code>outfmt</code> and <code>out</code>; e.g.<br />
<code>blastn -query Testquery -db Testdatabase -outfmt 6 -out out.txt</code><br />
or<br />
<code>blastn -query Testquery -db Testdatabase -outfmt 10 -out out.csv</code><br />
One can also call BLAST from inside R by using function <code>system</code><br />
system(&quot;blastn -query Testquery -db Testdatabase -outfmt 6 -out out.txt&quot;)<br />
4) Read in the results<br />
<code>test.res &lt;- read.blast(file = "out.txt")</code><br />
or<br />
<code>test.res &lt;- read.blast(file = "out.csv", sep = ",")</code>
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with variables
</p>

<dl>
<dt><code>query.id</code></dt><dd><p>character: sequence identifier. </p>
</dd>
<dt><code>subject.id</code></dt><dd><p>character: subject identifier. </p>
</dd>
<dt><code>identity</code></dt><dd><p>numeric: identity between sequences (in percent). </p>
</dd>
<dt><code>alignment.length</code></dt><dd><p>integer: number of nucleotides. </p>
</dd>
<dt><code>mismatches</code></dt><dd><p>integer: number of mismatches. </p>
</dd>
<dt><code>gap.opens</code></dt><dd><p>integer: number of gaps. </p>
</dd>
<dt><code>q.start</code></dt><dd><p>integer: query sequence start. </p>
</dd>
<dt><code>q.end</code></dt><dd><p>integer: query sequence end. </p>
</dd>
<dt><code>s.start</code></dt><dd><p>integer: subject sequence start. </p>
</dd>
<dt><code>s.end</code></dt><dd><p>integer: subject sequence end. </p>
</dd>
<dt><code>evalue</code></dt><dd><p>numeric: evalue. </p>
</dd>
<dt><code>bit.score</code></dt><dd><p>numeric: score value. </p>
</dd>
</dl>



<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Standalone Blast download: <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
<p>Blast News: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews</a>
</p>
<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+BLASTdata">BLASTdata</a></code>, <code><a href="#topic+simMatrix">simMatrix</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>Dir &lt;- system.file("extdata", package = "RFLPtools") # input directory 
filename &lt;- file.path(Dir, "BLASTexample.txt")
BLAST1 &lt;- read.blast(file = filename)
str(BLAST1)
</code></pre>

<hr>
<h2 id='read.rflp'>
Read RFLP data
</h2><span id='topic+read.rflp'></span>

<h3>Description</h3>

<p>Function to read RFLP data (e.g. generated with software package Gene
Profiler 4.05 (Scanalytics Inc.)) for DNA fragment analysis and
genotyping, and exported to a text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.rflp(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.rflp_+3A_file">file</code></td>
<td>
<p> character: RFLP file to read in.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function reads data from a text file which was generated e.g. with the 
software package Gene Profiler 4.05 (Scanalytics Inc.) for DNA fragment
analysis and genotyping. The data file contains sample identifier (<code>Sample</code>), 
band number (<code>Band</code>), molecular weight (<code>MW</code>) and gel identifier (<code>Gel</code>) 
(see <code><a href="#topic+RFLPdata">RFLPdata</a></code>).
</p>
<p>If gel identifier <code>Gel</code> is missing it is extracted from the sample identifier 
<code>Sample</code>.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Gel</code></dt><dd><p>character: gel identifier. </p>
</dd>
</dl>



<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="#topic+RFLPdist">RFLPdist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>Dir &lt;- system.file("extdata", package = "RFLPtools") # input directory 
filename &lt;- file.path(Dir, "RFLPexample.txt")
RFLP1 &lt;- read.rflp(file = filename)
str(RFLP1)

filename &lt;- file.path(Dir, "AZ091016_report.txt")
RFLP2 &lt;- read.rflp(file = filename)
str(RFLP2)
</code></pre>

<hr>
<h2 id='refDataGerm'> Example data set from GERM software </h2><span id='topic+refDataGerm'></span>

<h3>Description</h3>

<p>This is the reference data taken from the GERM software. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(refDataGerm)</code></pre>


<h3>Format</h3>

<p>A data frame with 250 observations on the following six variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Enzyme</code></dt><dd><p>character: enzyme used.</p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Genus</code></dt><dd><p>character: genus of sample. </p>
</dd>
<dt><code>Species</code></dt><dd><p>character: species of sample. </p>
</dd>
</dl>



<h3>Details</h3>

<p>See GERM software.
</p>


<h3>Source</h3>

<p>The data set was taken from the GERM software (table 'Example Data').
</p>


<h3>References</h3>

<p>Ian A. Dickie, Peter G. Avis, David J. McLaughlin, Peter B. Reich. 
Good-Enough RFLP Matcher (GERM) program.
Mycorrhiza 2003, 13:171-172.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(refDataGerm)
str(refDataGerm)
</code></pre>

<hr>
<h2 id='RFLPcombine'>
Combine RFLP data sets
</h2><span id='topic+RFLPcombine'></span>

<h3>Description</h3>

<p>Function to combine an arbitrary number of RFLP data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPcombine(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPcombine_+3A_...">...</code></td>
<td>
<p> two or more data.frames with RFLP data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data sets are combined using <code><a href="base.html#topic+rbind">rbind</a></code>.
</p>
<p>If data sets with identical sample identifiers are given, the
identifiers are made unique using <code><a href="base.html#topic+make.unique">make.unique</a></code>.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Gel</code></dt><dd><p>character: gel identifier. </p>
</dd>
</dl>



<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
res &lt;- RFLPcombine(RFLPdata, RFLPdata, RFLPdata)
RFLPplot(res, nrBands = 4)
</code></pre>

<hr>
<h2 id='RFLPdata'> Example data set for RFLP data </h2><span id='topic+RFLPdata'></span>

<h3>Description</h3>

<p>This is an example data set for RFLP data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RFLPdata)</code></pre>


<h3>Format</h3>

<p>A data frame with 737 observations on the following four variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Gel</code></dt><dd><p>character: gel identifier. </p>
</dd>
</dl>



<h3>Details</h3>

<p>The molecular weight was determined using the software package Gene Profiler 4.05 
(Scanalytics Inc.) for DNA fragment analysis and genotyping, and exported to a 
text file.
</p>


<h3>Source</h3>

<p>The data set was generated by F. Flessa.
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
str(RFLPdata)
</code></pre>

<hr>
<h2 id='RFLPdist'>
Compute distances for RFLP data.
</h2><span id='topic+RFLPdist'></span>

<h3>Description</h3>

<p>Within each group containing RFLP-samples exhibiting a equal number of bands, 
the distance between the molecular weights is computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPdist(x, distfun = dist, nrBands, LOD = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPdist_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data; see <code><a href="#topic+RFLPdata">RFLPdata</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist_+3A_distfun">distfun</code></td>
<td>
<p> function computing the distance with default <code>dist</code>;
cf. <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist_+3A_nrbands">nrBands</code></td>
<td>
<p> if not missing, then only samples with the specified number
of bands are considered. </p>
</td></tr>
<tr><td><code id="RFLPdist_+3A_lod">LOD</code></td>
<td>
<p> threshold for low-bp bands. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each number of bands the given distance between the molecular weights is computed. 
The result is a named list of distances where the names correspond to the number of 
bands which occur in each group.
</p>
<p>If <code>nrBands</code> is specified only samples with this number of bands are considered.
</p>
<p>If <code>LOD &gt; 0</code> is specified, all values below <code>LOD</code> are removed before the 
distances are calculated.
</p>


<h3>Value</h3>

<p>A named list with the distances; see <code><a href="stats.html#topic+dist">dist</a></code>.
</p>
<p>In case <code>nrBands</code> is not missing, an object of S3 class <code>dist</code>.
</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>
<p>Poussier, Stephane; Trigalet-Demery, Danielle; Vandewalle, Peggy; Goffinet, Bruno; Luisetti, Jacques; Trigalet, Andre.
Genetic diversity of Ralstonia solanacearum as assessed by PCR-RFLP of the hrp gene region, AFLP and 16S rRNA sequence 
analysis, and identification of an African subdivision.
Microbiology 2000 146:1679-1692 
</p>
<p>Matsumoto, Masaru; Furuya, Naruto; Takanami, Yoichi; Matsuyama, Nobuaki.  
RFLP analysis of the PCR-amplified 28S rDNA in Rhizoctonia solani. 
Mycoscience 1996 37:351 - 356
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="stats.html#topic+dist">dist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Euclidean distance
data(RFLPdata)
res &lt;- RFLPdist(RFLPdata)
names(res) ## number of bands
res$"6"

RFLPdist(RFLPdata, nrBands = 6)

## Other distances
res1 &lt;- RFLPdist(RFLPdata, distfun = function(x) dist(x, method = "manhattan"))
res2 &lt;- RFLPdist(RFLPdata, distfun = function(x) dist(x, method = "maximum"))
res[[1]]
res1[[1]]
res2[[1]]

## cut dendrogram at height 50
clust4bd &lt;- hclust(res[[2]])
cgroups50 &lt;- cutree(clust4bd, h=50)
cgroups50

## or
library(MKomics)
res3 &lt;- RFLPdist(RFLPdata, distfun = corDist)
res3$"9"

## hierarchical clustering
par(mfrow = c(2,2))
plot(hclust(res[[1]]), main = "Euclidean distance")
plot(hclust(res1[[1]]), main = "Manhattan distance")
plot(hclust(res2[[1]]), main = "Maximum distance")
plot(hclust(res3[[1]]), main = "Pearson correlation distance")


## Similarity matrix
library(MKomics)
myCol &lt;- colorRampPalette(brewer.pal(8, "RdYlGn"))(128)
ord &lt;- order.dendrogram(as.dendrogram(hclust(res[[1]])))
temp &lt;- as.matrix(res[[1]])
simPlot(temp[ord,ord], col = rev(myCol), minVal = 0, 
        labels = colnames(temp), title = "(Dis-)Similarity Plot")


## or
library(lattice)
levelplot(temp[ord,ord], col.regions = rev(myCol),
          at = do.breaks(c(0, max(temp)), 128),
          xlab = "", ylab = "",
          ## Rotate label of x axis
          scales = list(x = list(rot = 90)),
          main = "(Dis-)Similarity Plot")

## multidimensional scaling
loc &lt;- cmdscale(res[[5]])
x &lt;- loc[,1]
y &lt;- -loc[,2]
plot(x, y, type="n", xlab="", ylab="", xlim = 1.05*range(x), main="Multidemsional scaling")
text(x, y, rownames(loc), cex=0.8)
</code></pre>

<hr>
<h2 id='RFLPdist2'>
Compute distances for RFLP data.
</h2><span id='topic+RFLPdist2'></span>

<h3>Description</h3>

<p>If gel image quality is low, faint bands may be disregarded and 
may lead to wrong conclusions. This function computes the distance 
between the molecular weights of RFLP samples, including samples 
containing one or more additional bands. Thus, failures during 
band detection could be identified. Visualisation of band patterns 
using this method can be done by <code><a href="#topic+RFLPplot">RFLPplot</a></code> using the 
argument <code>nrMissing</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPdist2(x, distfun = dist, nrBands, nrMissing, LOD = 0,
          diag = FALSE, upper = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPdist2_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data; see <code><a href="#topic+RFLPdata">RFLPdata</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist2_+3A_distfun">distfun</code></td>
<td>
<p> function computing the distance with default <code>dist</code>;
cf. <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist2_+3A_nrbands">nrBands</code></td>
<td>
<p> samples with number of bands equal to <code>nrBands</code> are to be considered.</p>
</td></tr>
<tr><td><code id="RFLPdist2_+3A_nrmissing">nrMissing</code></td>
<td>
<p> number of bands that might be missing.</p>
</td></tr>
<tr><td><code id="RFLPdist2_+3A_lod">LOD</code></td>
<td>
<p> threshold for low-bp bands. </p>
</td></tr>
<tr><td><code id="RFLPdist2_+3A_diag">diag</code></td>
<td>
<p> see <code><a href="stats.html#topic+dist">dist</a></code> </p>
</td></tr>
<tr><td><code id="RFLPdist2_+3A_upper">upper</code></td>
<td>
<p> see <code><a href="stats.html#topic+dist">dist</a></code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given number of bands the given distance between the molecular weights
is computed. It is assumed that a number of bands might be missing. Hence
all samples with number of bands in nrBands, nrBands+1, ..., nrBands+nrMissing
are compared. 
</p>
<p>If <code>LOD &gt; 0</code> is specified, it is assumed that missing bands can only occur for
molecular weights smaller than <code>LOD</code>. As a consequence only samples which 
have <code>nrBands</code> bands with molecular weight larger or equal to <code>LOD</code> are
selected.
</p>
<p>For computing the distance between the molecular weight of a sample S1 with x bands 
and a Sample S2 with x+y bands the distances between the molecular weight of sample S1
and the molecular weight of all possible subsets of S2 with x bands are computed.
The distance between S1 and S2 is then defined as the minimum of all these distances.
</p>
<p>If <code>LOD &gt; 0</code> is specified, only all combinations of values below <code>LOD</code> are 
considered.
</p>
<p>This option may be useful, if gel image quality is low, and the detection of bands 
is doubtful.
</p>


<h3>Value</h3>

<p>An object of class <code>"dist"</code> returned; cf. <code><a href="stats.html#topic+dist">dist</a></code>.
</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>
<p>Ian A. Dickie, Peter G. Avis, David J. McLaughlin, Peter B. Reich. 
Good-Enough RFLP Matcher (GERM) program.
Mycorrhiza 2003, 13:171-172.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="#topic+nrBands">nrBands</a></code>, <code><a href="#topic+RFLPdist">RFLPdist</a></code>, <code><a href="stats.html#topic+dist">dist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Euclidean distance
data(RFLPdata)
nrBands(RFLPdata)
res0 &lt;- RFLPdist(RFLPdata, nrBands = 4)
res1 &lt;- RFLPdist2(RFLPdata, nrBands = 4, nrMissing = 1)
res2 &lt;- RFLPdist2(RFLPdata, nrBands = 4, nrMissing = 2)
res3 &lt;- RFLPdist2(RFLPdata, nrBands = 4, nrMissing = 3)

## assume missing bands only below LOD
res1.lod &lt;- RFLPdist2(RFLPdata, nrBands = 4, nrMissing = 1, LOD = 60)

## hierarchical clustering
par(mfrow = c(2,2))
plot(hclust(res0), main = "0 bands missing")
plot(hclust(res1), main = "1 band missing")
plot(hclust(res2), main = "2 bands missing")
plot(hclust(res3), main = "3 bands missing")

## missing bands only below LOD
par(mfrow = c(1,2))
plot(hclust(res0), main = "0 bands missing")
plot(hclust(res1.lod), main = "1 band missing below LOD")

## Similarity matrix
library(MKomics)
myCol &lt;- colorRampPalette(brewer.pal(8, "RdYlGn"))(128)
ord &lt;- order.dendrogram(as.dendrogram(hclust(res1)))
temp &lt;- as.matrix(res1)
simPlot(temp[ord,ord], col = rev(myCol), minVal = 0, 
        labels = colnames(temp), title = "(Dis-)Similarity Plot")

## missing bands only below LOD
ord &lt;- order.dendrogram(as.dendrogram(hclust(res1.lod)))
temp &lt;- as.matrix(res1.lod)
simPlot(temp[ord,ord], col = rev(myCol), minVal = 0, 
        labels = colnames(temp), title = "(Dis-)Similarity Plot\n1 band missing below LOD")


## or
library(lattice)
levelplot(temp[ord,ord], col.regions = rev(myCol),
          at = do.breaks(c(0, max(temp)), 128),
          xlab = "", ylab = "",
          ## Rotate label of x axis
          scales = list(x = list(rot = 90)),
          main = "(Dis-)Similarity Plot")


## Other distances
res11 &lt;- RFLPdist2(RFLPdata, distfun = function(x) dist(x, method = "manhattan"),
                 nrBands = 4, nrMissing = 1)
res12 &lt;- RFLPdist2(RFLPdata, distfun = corDist, nrBands = 4, nrMissing = 1)
res13 &lt;- RFLPdist2(RFLPdata, distfun = corDist, nrBands = 4, nrMissing = 1, LOD = 60)
par(mfrow = c(2,2))
plot(hclust(res1), main = "Euclidean distance\n1 band missing")
plot(hclust(res11), main = "Manhattan distance\n1 band missing")
plot(hclust(res12), main = "Pearson correlation distance\n1 band missing")
plot(hclust(res13), main = "Pearson correlation distance\n1 band missing below LOD")
</code></pre>

<hr>
<h2 id='RFLPdist2ref'>
Compute distance between RFLP data and RFLP reference data.
</h2><span id='topic+RFLPdist2ref'></span>

<h3>Description</h3>

<p>Function to compute distance between RFLP data and RFLP reference data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPdist2ref(x, ref, distfun = dist, nrBands, LOD = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPdist2ref_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data; e.g. <code><a href="#topic+RFLPdata">RFLPdata</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist2ref_+3A_ref">ref</code></td>
<td>
<p> data.frame with RFLP reference data; e.g. <code><a href="#topic+RFLPref">RFLPref</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist2ref_+3A_distfun">distfun</code></td>
<td>
<p> function computing the distance with default <code>dist</code>;
cf. <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPdist2ref_+3A_nrbands">nrBands</code></td>
<td>
<p> only samples and reference samples with this number
of bands are considered. </p>
</td></tr>
<tr><td><code id="RFLPdist2ref_+3A_lod">LOD</code></td>
<td>
<p> threshold for low-bp bands. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each sample with <code>nrBands</code> bands the distance to each reference 
sample with <code>nrBands</code> bands is computed. The result is a matrix with 
the corresponding distances where rows represent the samples and columns
the reference samples.
</p>
<p>If <code>LOD &gt; 0</code> is specified, all values below <code>LOD</code> are removed before the 
distances are calculated. This applies to <code>x</code> and <code>ref</code>.
</p>


<h3>Value</h3>

<p>A matrix with distances.
</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="stats.html#topic+dist">dist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Euclidean distance
data(RFLPdata)
data(RFLPref)
nrBands(RFLPref)
RFLPdist2ref(RFLPdata, RFLPref, nrBands = 4)
RFLPdist2ref(RFLPdata, RFLPref, nrBands = 6)

Dir &lt;- system.file("extdata", package = "RFLPtools") # input directory 
filename &lt;- file.path(Dir, "AZ091016_report.txt")
RFLP1 &lt;- read.rflp(file = filename)
RFLP2 &lt;- RFLPqc(RFLP1)
nrBands(RFLP2)
RFLPdist2ref(RFLP1, RFLPref, nrBands = 4)
RFLPdist2ref(RFLP1, RFLPref, nrBands = 5)
</code></pre>

<hr>
<h2 id='RFLPlod'>
Remove bands below LOD
</h2><span id='topic+RFLPlod'></span>

<h3>Description</h3>

<p>Function to exclude bands below a given LOD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPlod(x, LOD)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPlod_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data.</p>
</td></tr>
<tr><td><code id="RFLPlod_+3A_lod">LOD</code></td>
<td>
<p> threshold for low-bp bands.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Low-bp bands may be regarded as unreliable. Function 
<code>RFLPlod</code> can be used to exclude such bands, which
are likely to be absent in some other samples, before
further analyses.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Gel</code></dt><dd><p>character: gel identifier. </p>
</dd>
</dl>



<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
## remove bands with MW smaller than 60
RFLPdata.lod &lt;- RFLPlod(RFLPdata, LOD = 60)
par(mfrow = c(1, 2))
RFLPplot(RFLPdata, nrBands = 4, ylim = c(40, 670))
RFLPplot(RFLPdata.lod, nrBands = 4, ylim = c(40, 670))
title(sub = "After applying RFLPlod")
</code></pre>

<hr>
<h2 id='RFLPplot'>
Function to plot RFLP data.
</h2><span id='topic+RFLPplot'></span>

<h3>Description</h3>

<p>Given RFLP data is plotted where the samples are sorted according
to the corresponding dendrogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPplot(x, nrBands, nrMissing, distfun = dist, 
         hclust.method = "complete", mar.bottom = 5, 
         cex.axis = 0.5, colBands, xlab = "", 
         ylab = "molecular weight", ylim, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPplot_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data; see <code><a href="#topic+RFLPdata">RFLPdata</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_nrbands">nrBands</code></td>
<td>
<p> if not missing, then only samples with the specified number
of bands are considered. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_nrmissing">nrMissing</code></td>
<td>
<p> if not missing, then it is assumed that some bands may be missing.
That is, all samples with number of bands in nrBands, nrBands+1, ..., nrBands+nrMissing
are considered.</p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_distfun">distfun</code></td>
<td>
<p> function computing the distance with default <code>dist</code>;
see <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_hclust.method">hclust.method</code></td>
<td>
<p> method used for hierarchical clustering;
see <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_mar.bottom">mar.bottom</code></td>
<td>
<p> bottom margin of the plot; see <code><a href="graphics.html#topic+par">par</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_cex.axis">cex.axis</code></td>
<td>
<p> size of the x-axis annotation. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_colbands">colBands</code></td>
<td>
<p> color for the bands. Has to be of length 1 or number of samples. 
If missing, <code>"Set1"</code> of <span class="pkg">RColorBrewer</span> is used; see
<code><a href="RColorBrewer.html#topic+ColorBrewer">ColorBrewer</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_xlab">xlab</code></td>
<td>
<p> passed to function <code><a href="base.html#topic+plot">plot</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_ylab">ylab</code></td>
<td>
<p> passed to function <code><a href="base.html#topic+plot">plot</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_ylim">ylim</code></td>
<td>
<p> passed to function <code><a href="base.html#topic+plot">plot</a></code>. If missing an appropriate
range of y-values is computed. </p>
</td></tr>
<tr><td><code id="RFLPplot_+3A_...">...</code></td>
<td>
<p> additional arguments passed to function <code><a href="base.html#topic+plot">plot</a></code>
except <code>xlim</code> which is defined inside of <code>RFLPplot</code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>RFLP data is plotted. The samples are sorted according to the corresponding
dendrogram which is computed via function <code>hclust</code>.
</p>
<p>The option to specify <code>nrMissing</code> may be useful, if gel image quality is low, 
and the detection of bands is doubtful.
</p>


<h3>Value</h3>

<p>invisible</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="stats.html#topic+dist">dist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
par(mfrow = c(1,2))
plot(hclust(RFLPdist(RFLPdata, nrBands = 3)), cex = 0.7)
RFLPplot(RFLPdata, nrBands = 3, mar.bottom = 6, cex.axis = 0.8)

par(mfrow = c(1,2))
plot(hclust(RFLPdist2(RFLPdata, nrBands = 9, nrMissing = 1)), cex = 0.7)
RFLPplot(RFLPdata, nrBands = 9, nrMissing = 1, mar.bottom = 6, cex.axis = 0.8)


distfun &lt;- function(x) dist(x, method = "maximum")
par(mfrow = c(1,2))
plot(hclust(RFLPdist(RFLPdata, nrBands = 3, distfun = distfun), 
            method = "average"), cex = 0.7, cex.lab = 0.7)
RFLPplot(RFLPdata, nrBands = 3, distfun = distfun, hclust.method = "average", 
         mar.bottom = 6, cex.axis = 0.8)
</code></pre>

<hr>
<h2 id='RFLPqc'>
Quality control for RFLP data
</h2><span id='topic+RFLPqc'></span>

<h3>Description</h3>

<p>Function to perform quality control for RFLP data based on a
comparison between the total length of the digested PCR
amplification product and the sum of the fragment lengths. If the
sum is smaller or larger than the PCR amplification product
(within a certain range to define), the samples can be excluded
from further analyses. This function is helpful for data sets
containig faint or uncertain bands. It is necessary to include 
the total length of the PCR amplification product for each sample 
as largest fragment in the data set, see <code><a href="#topic+RFLPdata">RFLPdata</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPqc(x, rm.band1 = TRUE, QC.lo = 0.8, QC.up = 1.07, QC.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPqc_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data.</p>
</td></tr>
<tr><td><code id="RFLPqc_+3A_rm.band1">rm.band1</code></td>
<td>
<p> logical: remove first band.</p>
</td></tr>
<tr><td><code id="RFLPqc_+3A_qc.lo">QC.lo</code></td>
<td>
<p> numeric: a real number in (0,1).</p>
</td></tr>
<tr><td><code id="RFLPqc_+3A_qc.up">QC.up</code></td>
<td>
<p> numeric: a real number larger than 1.</p>
</td></tr>
<tr><td><code id="RFLPqc_+3A_qc.rm">QC.rm</code></td>
<td>
<p> logical: remove samples with unsufficient quality.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In case the first band corresponds to the total length of the fragment one can perform
a quality control comparing the length of the first band with the sum of the lengths
of the remaining bands for each sample. If the sum is smaller than <code>QC.lo</code> times 
the length of the first band or larger than <code>QC.up</code> times the length of the first 
band, respectively, a text message is printed. 
</p>
<p>If <code>rm.band1 = TRUE</code> band 1 of all samples is removed and the remaining band 
numbers are reduced by 1. 
</p>
<p>If <code>QC.rm = TRUE</code> samples of insufficient quality are entirely removed from the 
given data and the resulting <code>data.frame</code> is returned.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Gel</code></dt><dd><p>character: gel identifier. </p>
</dd>
</dl>



<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="#topic+RFLPdist">RFLPdist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>Dir &lt;- system.file("extdata", package = "RFLPtools") # input directory 
filename &lt;- file.path(Dir, "AZ091016_report.txt")
RFLP1 &lt;- read.rflp(file = filename)
str(RFLP1)

RFLP2 &lt;- RFLPqc(RFLP1, rm.band1 = FALSE) # identical to RFLP1
identical(RFLP1, RFLP2)

RFLP3 &lt;- RFLPqc(RFLP1)
str(RFLP3)

RFLP4 &lt;- RFLPqc(RFLP1, rm.band1 = TRUE, QC.rm = TRUE)
str(RFLP4)
</code></pre>

<hr>
<h2 id='RFLPref'> Example data set for RFLP reference</h2><span id='topic+RFLPref'></span>

<h3>Description</h3>

<p>This is an example data set for RFLP reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RFLPref)</code></pre>


<h3>Format</h3>

<p>A data frame with 35 observations on the following five variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Taxonname</code></dt><dd><p>character: taxon name. </p>
</dd>
<dt><code>Accession</code></dt><dd><p>character: accession number. </p>
</dd>
</dl>



<h3>Details</h3>

<p>This example data set for RFLP reference consists of seven RFLP reference 
samples. Taxon names are assigned by sequence comparison with GenBank 
database (<a href="https://www.ncbi.nlm.nih.gov/BLAST/">https://www.ncbi.nlm.nih.gov/BLAST/</a>), and supplemented with 
imaginary accession numbers.
</p>


<h3>Source</h3>

<p>The data set was generated by F. Flessa.
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPref)
str(RFLPref)
</code></pre>

<hr>
<h2 id='RFLPrefplot'>
Function for a visual comparison of RFLP samples with reference samples.
</h2><span id='topic+RFLPrefplot'></span>

<h3>Description</h3>

<p>Given RFLP samples are plotted together with reference samples and
sorted by their distance to the reference sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFLPrefplot(x, ref, distfun = dist, nrBands, mar.bottom = 5, 
            cex.main = 1.2, cex.axis = 0.5, devNew = FALSE, 
            colBands, xlab = "", ylab = "molecular weight", 
            ylim, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFLPrefplot_+3A_x">x</code></td>
<td>
<p> data.frame with RFLP data; e.g. <code><a href="#topic+RFLPdata">RFLPdata</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_ref">ref</code></td>
<td>
<p> data.frame with RFLP reference data; e.g. <code><a href="#topic+RFLPref">RFLPref</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_distfun">distfun</code></td>
<td>
<p> function computing the distance with default <code>dist</code>;
see <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_nrbands">nrBands</code></td>
<td>
<p> if not missing, then only samples with the specified number
of bands are considered. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_mar.bottom">mar.bottom</code></td>
<td>
<p> bottom margin of the plot; see <code><a href="graphics.html#topic+par">par</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_cex.main">cex.main</code></td>
<td>
<p> size of the plot title. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_cex.axis">cex.axis</code></td>
<td>
<p> size of the x-axis annotation. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_devnew">devNew</code></td>
<td>
<p> logical. Open new graphics device for each plot. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_colbands">colBands</code></td>
<td>
<p> color for the bands. Has to be of length 1 or number of samples. 
If missing, <code>"Set1"</code> of <span class="pkg">RColorBrewer</span> is used; see
<code><a href="RColorBrewer.html#topic+ColorBrewer">ColorBrewer</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_xlab">xlab</code></td>
<td>
<p> passed to function <code><a href="base.html#topic+plot">plot</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_ylab">ylab</code></td>
<td>
<p> passed to function <code><a href="base.html#topic+plot">plot</a></code>. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_ylim">ylim</code></td>
<td>
<p> passed to function <code><a href="base.html#topic+plot">plot</a></code>. If missing an appropriate
range of y-values is computed. </p>
</td></tr>
<tr><td><code id="RFLPrefplot_+3A_...">...</code></td>
<td>
<p> additional arguments passed to function <code><a href="base.html#topic+plot">plot</a></code>
except <code>xlim</code> which is defined inside of <code>RFLPplot</code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given RFLP samples are plotted together with reference samples and
sorted by their distance to the reference sample.
</p>


<h3>Value</h3>

<p>invisible</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPplot">RFLPplot</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
data(RFLPref)
dev.new(width = 12)
RFLPrefplot(RFLPdata, RFLPref, nrBands = 4, cex.axis = 0.5)

dev.new()
RFLPrefplot(RFLPdata, RFLPref, nrBands = 6, cex.axis = 0.8)
RFLPrefplot(RFLPdata, RFLPref, nrBands = 9, cex.axis = 0.8)

RFLPrefplot(RFLPdata, RFLPref[RFLPref$Sample == "Ni_29_A3",], nrBands = 4, cex.axis = 0.7)

Dir &lt;- system.file("extdata", package = "RFLPtools") # input directory 
filename &lt;- file.path(Dir, "AZ091016_report.txt")
RFLP1 &lt;- read.rflp(file = filename)
RFLP2 &lt;- RFLPqc(RFLP1)

dev.new(width = 12)
RFLPrefplot(RFLP1, RFLPref, nrBands = 4, cex.axis = 0.8)

dev.new()
RFLPrefplot(RFLP1, RFLPref, nrBands = 5, cex.axis = 0.8)
</code></pre>

<hr>
<h2 id='RFLPtools-package'>
Tools To Analyse RFLP-Data
</h2><span id='topic+RFLPtools-package'></span><span id='topic+RFLPtools'></span>

<h3>Description</h3>

<p><span class="pkg">RFLPtools</span> provides functions to analyse DNA fragment samples 
(i.e. derived from RFLP-analysis) and standalone BLAST report 
files (i.e. DNA sequence analysis).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> RFLPtools</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0 </td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-02-07 </td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R(&gt;= 4.0.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> stats, utils, graphics, grDevices, RColorBrewer</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> knitr, rmarkdown, lattice, MKomics</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> LGPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<h3>Author(s)</h3>

<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Mohammed Aslam Imtiaz,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a><br />
</p>
<p><em>Maintainer:</em> Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a>
</p>


<h3>References</h3>

<p>Local Blast download: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download</a>
</p>
<p>Blast News: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews</a>
</p>
<p>Ian A. Dickie, Peter G. Avis, David J. McLaughlin, Peter B. Reich. 
Good-Enough RFLP Matcher (GERM) program.
Mycorrhiza 2003, 13:171-172.
</p>
<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>
<p>Matsumoto, Masaru; Furuya, Naruto; Takanami, Yoichi; Matsuyama, Nobuaki.  
RFLP analysis of the PCR-amplified 28S rDNA in Rhizoctonia solani. 
Mycoscience 1996 37:351-356.
</p>
<p>Persoh, D., Melcher, M., Flessa, F., Rambold, G.: 
First fungal community analyses of endophytic ascomycetes associated with Viscum album ssp. austriacum 
and itshost Pinus sylvestris. Fungal Biology 2010 Jul;114(7):585-96.
</p>
<p>Poussier, Stephane; Trigalet-Demery, Danielle; Vandewalle, Peggy; Goffinet, Bruno; Luisetti, Jacques; Trigalet, Andre.
Genetic diversity of Ralstonia solanacearum as assessed by PCR-RFLP of the hrp gene region, AFLP and 16S rRNA sequence 
analysis, and identification of an African subdivision.
Microbiology 2000 146:1679-1692.
</p>
<p>T. A. Saari,  S. K. Saari, C. D. Campbell, I. J Alexander, I. C. Anderson.
FragMatch - a program for the analysis of DNA fragment data.
Mycorrhiza 2007, 17:133-136
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
res &lt;- RFLPdist(RFLPdata)
plot(hclust(res[[1]]), main = "Euclidean distance")

par(mfrow = c(1,2))
plot(hclust(RFLPdist(RFLPdata, nrBands = 3)), cex = 0.7)
RFLPplot(RFLPdata, nrBands = 3, mar.bottom = 6, cex.axis = 0.8)

data(RFLPref)
RFLPrefplot(RFLPdata, RFLPref, nrBands = 6, cex.axis = 0.8)


library(MKomics)
data(BLASTdata)
res &lt;- simMatrix(BLASTdata, sequence.range = TRUE, Min = 500)
myCol &lt;- colorRampPalette(brewer.pal(8, "RdYlGn"))(128)
simPlot(res, col = myCol, minVal = 0, 
        labels = colnames(res), title = "(Dis-)Similarity Plot")
</code></pre>

<hr>
<h2 id='sim2dist'>
Convert similarity matrix to dist object.
</h2><span id='topic+sim2dist'></span>

<h3>Description</h3>

<p>Function to convert similarity matrix to object of S3 class <code>"dist"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim2dist(x, maxSim = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim2dist_+3A_x">x</code></td>
<td>
<p> symmetric matrix: similarity matrix. </p>
</td></tr>
<tr><td><code id="sim2dist_+3A_maxsim">maxSim</code></td>
<td>
<p> maximum similarity possible. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Similarity is converted to distance by <code>maxSim - x</code>. 
The resulting matrix is converted to an object of S3 class <code>"dist"</code>
by <code>as.dist</code>
</p>


<h3>Value</h3>

<p>Object of S3 class <code>"dist"</code> is returned; see <code><a href="stats.html#topic+dist">dist</a></code>.
</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+BLASTdata">BLASTdata</a></code>, <code><a href="#topic+simMatrix">simMatrix</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BLASTdata)

## without sequence range
## Not run: 
res &lt;- simMatrix(BLASTdata)

## End(Not run)

## with sequence range
range(BLASTdata$alignment.length)
res1 &lt;- simMatrix(BLASTdata, sequence.range = TRUE, Min = 100, Max = 450)
res2 &lt;- simMatrix(BLASTdata, sequence.range = TRUE, Min = 500)

## visualize similarity matrix
library(MKomics)
simPlot(res2, minVal = 0, 
        labels = colnames(res2), title = "(Dis-)Similarity Plot")


## or
library(lattice)
myCol &lt;- colorRampPalette(brewer.pal(8, "RdYlGn"))(128)
levelplot(res2, col.regions = myCol,
          at = do.breaks(c(0, max(res2)), 128),
          xlab = "", ylab = "",
          ## Rotate label of x axis
          scales = list(x = list(rot = 90)),
          main = "(Dis-)Similarity Plot")

## convert to distance
res.d &lt;- sim2dist(res2)

## hierarchical clustering
plot(hclust(res.d))
</code></pre>

<hr>
<h2 id='simMatrix'>
Similarity matrix for BLAST data.
</h2><span id='topic+simMatrix'></span>

<h3>Description</h3>

<p>Function to compute similarity matrix for all-vs-all BLAST results of rDNA sequences 
generated with standalone BLAST from NCBI or local BLAST implemented in BioEdit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simMatrix(x, sequence.range = FALSE, Min, Max)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simMatrix_+3A_x">x</code></td>
<td>
<p> data.frame with BLAST data; see <code><a href="#topic+BLASTdata">BLASTdata</a></code>.</p>
</td></tr>
<tr><td><code id="simMatrix_+3A_sequence.range">sequence.range</code></td>
<td>
<p> logical: use sequence range.</p>
</td></tr>
<tr><td><code id="simMatrix_+3A_min">Min</code></td>
<td>
<p> minimum sequence length. </p>
</td></tr>
<tr><td><code id="simMatrix_+3A_max">Max</code></td>
<td>
<p> maximum sequence length. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The given BLAST data is used to compute a similarity matrix using the following algorithm:
First, the length of each sequence (LS) comprised in the input data file is extracted. 
If there is more than one comparison for one sequence including different parts of the 
respective sequence, that one with maximum base length is chosen. 
Subsequently, the number of matching bases (mB) is calculated by multiplying 
two variables comprised in the BLAST output: the identity between sequences (%) and 
the number of nucleotides divided by 100. The, resulting value is rounded to integer. 
Furthermore, the similarity is calculated by dividing mB by LS. 
Finally, the similarity matrix including all sequences is built. 
If the similarity of a combination is not shown in the BLAST report file (because 
the similarity was lower than 70%), this comparison is included in the similarity 
matrix with the result zero.
</p>


<h3>Value</h3>

<p>Similarity matrix.
</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Standalone Blast download: <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
<p>Blast News: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastNews</a>
</p>
<p>BioEdit: <a href="https://bioedit.software.informer.com/">https://bioedit.software.informer.com/</a>
</p>
<p>Persoh, D., Melcher, M., Flessa, F., Rambold, G.: 
First fungal community analyses of endophytic ascomycetes associated with Viscum album ssp. austriacum 
and itshost Pinus sylvestris. Fungal Biology 2010 Jul;114(7):585-96.
</p>
<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+BLASTdata">BLASTdata</a></code>, <code><a href="#topic+sim2dist">sim2dist</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BLASTdata)

## without sequence range
## code takes some time
## Not run: 
res &lt;- simMatrix(BLASTdata)

## End(Not run)

## with sequence range
range(BLASTdata$alignment.length)
res1 &lt;- simMatrix(BLASTdata, sequence.range = TRUE, Min = 100, Max = 450)
res2 &lt;- simMatrix(BLASTdata, sequence.range = TRUE, Min = 500)
</code></pre>

<hr>
<h2 id='simulateRFLPdata'>
Simulate RFLP data.
</h2><span id='topic+simulateRFLPdata'></span>

<h3>Description</h3>

<p>Simulates RFLP data for comparions of algorithms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateRFLPdata(N = 10, nrBands = 3:12, bandCenters = seq(100, 800, by = 100),
                 delta = 50, refData = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateRFLPdata_+3A_n">N</code></td>
<td>
<p> integer: number samples which shall be simulated per number of bands.</p>
</td></tr>
<tr><td><code id="simulateRFLPdata_+3A_nrbands">nrBands</code></td>
<td>
<p> integer: vector of number of bands.</p>
</td></tr>
<tr><td><code id="simulateRFLPdata_+3A_bandcenters">bandCenters</code></td>
<td>
<p> numeric: vector of band centers.</p>
</td></tr>
<tr><td><code id="simulateRFLPdata_+3A_delta">delta</code></td>
<td>
<p> numeric: uniform distribution with <code>min = bandCenter - delta</code> and <code>max = bandCenter + delta</code>
is used.</p>
</td></tr>
<tr><td><code id="simulateRFLPdata_+3A_refdata">refData</code></td>
<td>
<p> logical: if TRUE, additonal columns <code>Taxonname</code> and <code>Accesion</code> are generated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can be used to simulate RFLP data. For every number of band specified in 
<code>nrBands</code> a total number of <code>N</code> samples are generated. 
</p>
<p>First the band centers are randomly selected (with replacement) from <code>bandCenter</code> 
which form the centers of intervals of length <code>2*delta</code>. From these intervals 
uniform random numbers are drawn leading to randomly generated RFLP data.
</p>


<h3>Value</h3>

<p>A data frame with <code>N*length(nrBands)</code> observations on the following four variables
</p>

<dl>
<dt><code>Sample</code></dt><dd><p>character: sample identifier. </p>
</dd>
<dt><code>Band</code></dt><dd><p>integer: band number. </p>
</dd>
<dt><code>MW</code></dt><dd><p>integer: molecular weight. </p>
</dd>
<dt><code>Enzyme</code></dt><dd><p>character: enzyme name. </p>
</dd>
</dl>

<p>is generated. If <code>refData = TRUE</code> then the following two additional variables
are added.
</p>

<dl>
<dt><code>Taxonname</code></dt><dd><p>character: taxon name. </p>
</dd>
<dt><code>Accession</code></dt><dd><p>character: accession number. </p>
</dd>
</dl>



<h3>Author(s)</h3>

 
<p>Mohammed Aslam Imtiaz, Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+RFLPdata">RFLPdata</a></code>, <code><a href="#topic+RFLPref">RFLPref</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>  simData &lt;- simulateRFLPdata()
</code></pre>

<hr>
<h2 id='write.hclust'>
Cut a hierarchical cluster tree and write cluster identifiers to a 
text file.
</h2><span id='topic+write.hclust'></span>

<h3>Description</h3>

<p>The tree obtained by a hierarchical cluster analysis is cut into
groups by using <code><a href="stats.html#topic+cutree">cutree</a></code> and the results are
exported to a text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.hclust(x, file, prefix, h = NULL, k = NULL, append = FALSE, dec = ",")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.hclust_+3A_x">x</code></td>
<td>
<p> object of class <code>hclust</code>: result of hierarchical cluster analysis
computed via function <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="write.hclust_+3A_file">file</code></td>
<td>
<p> either a character string naming a file or a connection open
for writing.  <code>""</code> indicates output to the console.</p>
</td></tr>
<tr><td><code id="write.hclust_+3A_prefix">prefix</code></td>
<td>
<p> character. Information about the cluster analysis.</p>
</td></tr>
<tr><td><code id="write.hclust_+3A_h">h</code></td>
<td>
<p>numeric scalar or vector with heights where the tree should
be cut.</p>
</td></tr>
<tr><td><code id="write.hclust_+3A_k">k</code></td>
<td>
<p> an integer scalar or vector with the desired number of groups.</p>
</td></tr>
<tr><td><code id="write.hclust_+3A_append">append</code></td>
<td>
<p> logical. Only relevant if <code>file</code> is a character string.  If
<code>TRUE</code>, the output is appended to the file.  If <code>FALSE</code>, any
existing file of the name is destroyed.</p>
</td></tr>
<tr><td><code id="write.hclust_+3A_dec">dec</code></td>
<td>
<p> the string to use for decimal points in numeric or complex
columns: must be a single character. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The results are written to file by a call to <code><a href="utils.html#topic+write.table">write.table</a></code>
where the columns in the resulting file are seperated by tabulators 
(i.e. <code>sep="\t"</code>) and no row names are exported (i.e. <code>row.names = FALSE</code>).
</p>


<h3>Author(s)</h3>

 
<p>Fabienne Flessa <a href="mailto:Fabienne.Flessa@uni-bayreuth.de">Fabienne.Flessa@uni-bayreuth.de</a>,<br />
Alexandra Kehl <a href="mailto:Alexandra.Kehl@uni-tuebingen.de">Alexandra.Kehl@uni-tuebingen.de</a>,<br />
Matthias Kohl <a href="mailto:Matthias.Kohl@stamats.de">Matthias.Kohl@stamats.de</a> 
</p>


<h3>References</h3>

<p>Flessa, F., Kehl, A., Kohl, M.
Analysing diversity and community structures using PCR-RFLP: a new software application.
Molecular Ecology Resources 2013 Jul; 13(4):726-33. 
</p>


<h3>See Also</h3>

 <p><code><a href="utils.html#topic+write.table">write.table</a></code>, <code><a href="stats.html#topic+cutree">cutree</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RFLPdata)
res &lt;- RFLPdist(RFLPdata, nrBands = 4)
cl &lt;- hclust(res)
## Not run: 
write.hclust(cl, file = "Test.txt", prefix = "Bd4", h = 50)

## End(Not run)

res &lt;- RFLPdist2(RFLPdata, nrBands = 4, nrMissing = 1)
cl &lt;- hclust(res)
## Not run: 
write.hclust(cl, file = "Test.txt", append = TRUE, prefix = "Bd4_Mis1", h = 60)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
