<!DOCTYPE html><html><head><title>Help for package Rgb</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Rgb}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Annotation'>
<p>Annotation track constructors</p></a></li>
<li><a href='#crossable-class'><p>Class <code>"crossable"</code></p></a></li>
<li><a href='#draw.bg'>
<p>Background for track plots</p></a></li>
<li><a href='#draw.boxes'>
<p>Boxes plot of a track</p></a></li>
<li><a href='#draw.density'>
<p>Two-dimensional density plot of a track</p></a></li>
<li><a href='#draw.hist'>
<p>Histogram plot of a track</p></a></li>
<li><a href='#draw.pileup'>
<p>Pileup plot of a BAM track</p></a></li>
<li><a href='#draw.points'>
<p>Scatter plot of a track</p></a></li>
<li><a href='#draw.seq'>
<p>Plot a sequence of nucleotides</p></a></li>
<li><a href='#draw.steps'>
<p>Stairway plot of a track</p></a></li>
<li><a href='#drawable-class'><p>Class <code>"drawable"</code></p></a></li>
<li><a href='#drawable.list'>
<p>Drawable list constructor</p></a></li>
<li><a href='#drawable.list-class'><p>Class <code>"drawable.list"</code></p></a></li>
<li><a href='#findDrawables'>
<p>Find drawable objects in memory</p></a></li>
<li><a href='#hsBands'><p>Homo sapiens GRCh37 chromosome banding</p></a></li>
<li><a href='#hsGenes'><p>Subset of Homo sapiens GRCh37 genes</p></a></li>
<li><a href='#RDT+20storage+20files'>
<p>Single refTable object storage</p></a></li>
<li><a href='#read.gtf'>
<p>Parses a GTF2 file</p></a></li>
<li><a href='#refTable-class'><p>Class <code>"refTable"</code></p></a></li>
<li><a href='#refTable-constructor'>
<p>refTable class constructor</p></a></li>
<li><a href='#segMerge'>
<p>Merges consecutive segments</p></a></li>
<li><a href='#segOverlap'>
<p>Merges overlapping segments</p></a></li>
<li><a href='#singlePlot'>
<p>Whole genome plot of a drawable list</p></a></li>
<li><a href='#sliceable-class'><p>Class <code>"sliceable"</code></p></a></li>
<li><a href='#subtrack'>
<p>Extract elements within a genomic window</p></a></li>
<li><a href='#tk.browse'>
<p>R Genome browser</p></a></li>
<li><a href='#tk.convert'>
<p>RDT file conversion</p></a></li>
<li><a href='#tk.tracks'>
<p>Track management for Genome browser</p></a></li>
<li><a href='#tk.utils'>
<p>Tk interface utilities</p></a></li>
<li><a href='#track-constructors'>
<p>Track constructors</p></a></li>
<li><a href='#track.bam-class'><p>Class <code>"track.bam"</code></p></a></li>
<li><a href='#track.bands-class'><p>Class <code>"track.bands"</code></p></a></li>
<li><a href='#track.CNV-class'><p>Class <code>"track.CNV"</code></p></a></li>
<li><a href='#track.exons-class'><p>Class <code>"track.exons"</code></p></a></li>
<li><a href='#track.fasta-class'><p>Class <code>"track.fasta"</code></p></a></li>
<li><a href='#track.fasta-constructors'>
<p>Track constructors for FASTA files</p></a></li>
<li><a href='#track.genes-class'><p>Class <code>"track.genes"</code></p></a></li>
<li><a href='#track.table-class'><p>Class <code>"track.table"</code></p></a></li>
<li><a href='#yline'>
<p>Compute Y line to avoid collision</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The R Genome Browser</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Sylvain Mareschal</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sylvain Mareschal &lt;mareschal@ovsa.fr&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bioinformatics.ovsa.fr/Rgb">https://bioinformatics.ovsa.fr/Rgb</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/maressyl/R.Rgb/issues">https://github.com/maressyl/R.Rgb/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Classes and methods to efficiently handle (slice, annotate, draw ...) genomic features (such as genes or transcripts), and an interactive interface to browse them. Performances and main features are highlighted in Mareschal et al (2014) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtu185">doi:10.1093/bioinformatics/btu185</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), methods</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tcltk, MASS, graphics, grDevices</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>tk.browse() requires Tcl-tk &gt;= 8.6 (shipped with R
&gt;= 3.4.0 on Windows, may have to be manually updated on other
systems)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-18 10:08:11 UTC; sylvain</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-18 10:32:33 UTC</td>
</tr>
</table>
<hr>
<h2 id='Annotation'>
Annotation track constructors
</h2><span id='topic+Annotation'></span><span id='topic+track.table.GTF'></span><span id='topic+track.exons.CCDS'></span><span id='topic+track.exons.GENCODE'></span><span id='topic+track.CNV.DGV'></span><span id='topic+track.genes.NCBI'></span><span id='topic+track.bands.UCSC'></span>

<h3>Description</h3>

<p>These functions constructs <code><a href="#topic+track.table-class">track.table</a></code> inheriting objects from free annotation files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  track.table.GTF(file, name = NA, attr = "split", features = "exon", quiet = FALSE, 
    .chromosomes, ...)
  track.exons.CCDS(file, name = "CCDS exons", ...)
  track.exons.GENCODE(file, name = "GENCODE exons",
    extra = c("gene_id", "gene_name", "exon_id"), ...)
  track.CNV.DGV(file, name = "DGV CNV", ...)
  track.genes.NCBI(file, name = "NCBI genes", selection, ...)
  track.bands.UCSC(file, name = "UCSC bands", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Annotation_+3A_file">file</code></td>
<td>

<p>Single character value, the path to the raw file to parse. See the 'Details' section below.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_name">name</code></td>
<td>

<p>Single character value, the <code>name</code> field for the <code><a href="#topic+track.table-class">track.table</a></code> object. For <code>track.table.GTF</code>, use <code>NA</code> to refer to the &quot;source&quot; column content (if it contains a unique value for all rows).
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_attr">attr</code></td>
<td>

<p>To be passed to <code><a href="#topic+read.gtf">read.gtf</a></code>.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_features">features</code></td>
<td>

<p>To be passed to <code><a href="#topic+read.gtf">read.gtf</a></code>.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_quiet">quiet</code></td>
<td>

<p>To be passed to <code><a href="#topic+read.gtf">read.gtf</a></code>.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_.chromosomes">.chromosomes</code></td>
<td>

<p>To be passed to the <code><a href="#topic+track.table">track.table</a></code> constructor.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_...">...</code></td>
<td>

<p>Further arguments are passed to the class constructor, as a result most of the handled arguments are <code><a href="#topic+track.table">track.table</a></code> arguments. Consider notably <code>.organism</code> and <code>.assembly</code> for track annotation.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_extra">extra</code></td>
<td>

<p>Character vector, names of optional columns to keep from the GENCODE GTF file.
</p>
</td></tr>
<tr><td><code id="Annotation_+3A_selection">selection</code></td>
<td>

<p>Character vector, filter to apply on the &quot;group_label&quot; column for NCBI genes. Raises an error with the possible values when missing.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>track.table.GTF</code> imports a &quot;Gene Feature Transfert&quot; file, as proposed by the UCSC Table Browser at <a href="http://www.genome.ucsc.edu/cgi-bin/hgTables">http://www.genome.ucsc.edu/cgi-bin/hgTables</a>) for a large amount of species. See the <code><a href="#topic+read.gtf">read.gtf</a></code> manual for further details.
</p>
<p><code>track.exons.CCDS</code> contains various transcripts from the &quot;Consensus Coding DNA Sequence&quot; project, currently only available for mouse and human (see the NCBI data repository at <a href="https://ftp.ncbi.nlm.nih.gov/pub/CCDS/">https://ftp.ncbi.nlm.nih.gov/pub/CCDS/</a>, and look for a file named &quot;CCDS_current.txt&quot;).
</p>
<p><code>track.exons.GENCODE</code> contains various transcripts from the GENCODE project, currently only available for mouse and human (see the dedicated website at <a href="https://www.gencodegenes.org/">https://www.gencodegenes.org/</a>). This function is intended to run on the GTF version of the data.
</p>
<p><code>track.CNV.DGV</code> parses constitutive copy number variations from the current version of the Database of Genomic Variants, downloadable from <a href="http://dgv.tcag.ca/dgv/app/downloads">http://dgv.tcag.ca/dgv/app/downloads</a> using &quot;DGV Variants&quot; links. The whole database is dedicated to the human specy only.
</p>
<p><code>track.genes.NCBI</code> parses the gene list from the MapView project of the NCBI, for one of many species available at <a href="https://ftp.ncbi.nih.gov/genomes/MapView/">https://ftp.ncbi.nih.gov/genomes/MapView/</a>. Select your specy of interest, then browse &quot;sequence&quot;, &quot;current&quot; and &quot;initial_release&quot; (if the directories are available, they are not for certain species). Download the file named &quot;seq_gene.md.gz&quot;. As many assemblies are included in the file, a first call to the function without &quot;selection&quot; is required, to list the available values. A second call with the appropriate assembly name will produce the desired track file.
</p>
<p><code>track.bands.UCSC</code> produces a track of cytogenetic banding, as made available by the UCSC for many species at <a href="http://hgdownload.cse.ucsc.edu/downloads.html">http://hgdownload.cse.ucsc.edu/downloads.html</a>. Select the specy and assembly version that suits your needs, and look for a file named &quot;cytoBand.txt.gz&quot; in the &quot;Annotation database&quot; section.
</p>


<h3>Value</h3>

<p>Return a <code><a href="#topic+track.table-class">track.table</a></code>-inheriting object (of class <code><a href="#topic+track.exons-class">track.exons</a></code>, <code><a href="#topic+track.CNV-class">track.CNV</a></code>, <code><a href="#topic+track.genes-class">track.genes</a></code> or <code><a href="#topic+track.bands-class">track.bands</a></code>).
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Example of <code>track.exons.CCDS</code> raw file (current human assembly) : <a href="https://ftp.ncbi.nlm.nih.gov/pub/CCDS/current_human/CCDS.current.txt">https://ftp.ncbi.nlm.nih.gov/pub/CCDS/current_human/CCDS.current.txt</a>
</p>
<p>Example of <code>track.exons.GENCODE</code> raw file (human assembly 'GRCh38') : <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.annotation.gtf.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.annotation.gtf.gz</a>
</p>
<p>Example of <code>track.CNV.DGV</code> raw file (human assembly 'hg19') : <a href="http://dgv.tcag.ca/dgv/docs/GRCh37_hg19_variants_2013-05-31.txt">http://dgv.tcag.ca/dgv/docs/GRCh37_hg19_variants_2013-05-31.txt</a>
</p>
<p>Example of <code>track.genes.NCBI</code> raw file (human assembly 'GRCh37') : <a href="https://ftp.ncbi.nih.gov/genomes/MapView/Homo_sapiens/sequence/BUILD.37.3/initial_release/seq_gene.md.gz">https://ftp.ncbi.nih.gov/genomes/MapView/Homo_sapiens/sequence/BUILD.37.3/initial_release/seq_gene.md.gz</a>
</p>
<p>Example of <code>track.bands.UCSC</code> raw file (human assembly 'hg19') : <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/cytoBand.txt.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/cytoBand.txt.gz</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.fasta-constructors">track.fasta-constructors</a></code>, <code><a href="#topic+Annotation">Annotation</a></code>
</p>
<p><code><a href="#topic+track.table-class">track.table-class</a></code>, <code><a href="#topic+track.exons-class">track.exons-class</a></code>, <code><a href="#topic+track.CNV-class">track.CNV-class</a></code>, <code><a href="#topic+track.genes-class">track.genes-class</a></code> or <code><a href="#topic+track.bands-class">track.bands-class</a></code>
</p>
<p><code><a href="#topic+tk.browse">tk.browse</a></code>, <code><a href="#topic+browsePlot">browsePlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # From the "How-to" vignette, section "Custom annotation tracks"
  file &lt;- system.file("extdata/Cosmic_ATM.gtf.gz", package="Rgb")
  tt &lt;- track.table.GTF(file)
</code></pre>

<hr>
<h2 id='crossable-class'>Class <code>"crossable"</code></h2><span id='topic+crossable-class'></span>

<h3>Description</h3>

<p>Reference classes extending this virtual class must have a <code>slice</code> method, as a generic <code>cross</code> method based on it is provided.
</p>
<p>Its only purpose is currently to add the <code>cross</code> method to <code>"<a href="#topic+track.table-class">track.table</a>"</code>, as <code>"<a href="#topic+sliceable-class">sliceable</a>"</code> does not guarantee that <code>slice</code> returns a data.frame as <code>crossable</code> needs one.
</p>


<h3>Extends</h3>

<p>Class <code><a href="#topic+sliceable-class">sliceable</a></code>, directly.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+sliceable-class">sliceable</a></code>, distance 2.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li></ul>



<h3>Methods</h3>


<dl>
<dt><code>cross(annotation, colname = , type = , fuzziness = , maxElements = , location = , precision = , quiet = )</code>:</dt><dd>
<p>Add a new column computed from overlaps with an other crossable object.<br />
- <strong>annotation</strong>    : other crossable object to compute overlap with.<br />
- <strong>colname</strong>       : single character value, the name of the new column to add to .self. If NULL or NA, the result will be returned rather than added to .self.<br />
- <strong>type</strong>          : single character value, either :<br />
'cover', to compute coverage of 'annotation' elements for each .self element<br />
'count', to count 'annotation' elements overlapping each .self element<br />
'cytoband', to get cytogenetic coordinates from a cytoband annotation track<br />
an 'annotation' column name, to list 'annotation' elements overlapping each .self element<br />
- <strong>fuzziness</strong>     : single integer value, to be added on each side of .self elements when computing overlaps.<br />
- <strong>maxElements</strong>   : single integer value, when more overlaps are found, lists are replaced by counts. Can be NA to disable this behavior.<br />
- <strong>location</strong>      : character vector, the 'chrom' / 'start' / 'end' .self columns to use for annotation.<br />
- <strong>precision</strong>     : single integer value from 1 to 4, amount of digits to consider in banding (type='cytoband').<br />
- <strong>quiet</strong>         : single logical value, wether to throw progression messages or not.
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> chromosomes (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> show (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> slice (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+drawable-class">drawable</a></code>, <code><a href="#topic+sliceable-class">sliceable</a></code>, <code><a href="#topic+track.table-class">track.table</a></code>, <code>cghRA.array</code>
</p>

<hr>
<h2 id='draw.bg'>
Background for track plots
</h2><span id='topic+draw.bg'></span>

<h3>Description</h3>

<p>This function draws the background for the other track ploting functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.bg(start, end, ylab = "", ylab.horiz = FALSE, ysub = as.character(NA),
    mar = c(0.2, 5, 0.2, 1), xaxt = "s", yaxt = "n", yaxs = "r", ylim = c(0, 1),
    cex.lab = 1, cex.axis = 1, mgp = c(3, 1, 0), tck = NA, tcl = -0.5,
    xaxp = as.numeric(NA), yaxp = as.numeric(NA), bty = "o", las = 0, xgrid = TRUE,
    new = FALSE, bg = NA, bg.inner = NA, fg = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.bg_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_ylab">ylab</code></td>
<td>

<p>The name of the Y axis. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_ylab.horiz">ylab.horiz</code></td>
<td>

<p>Single logical value, whether to print <code>ylab</code> horizontally or not.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_ysub">ysub</code></td>
<td>

<p>Similar to <code>ylab</code>, but written on a closer line when <code>yaxt</code> is &quot;n&quot;. Can be <code>NA</code> to disable it.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_mar">mar</code></td>
<td>

<p>A numerical vector of the form &quot;c(bottom, left, top, right)&quot; which gives the number of lines of margin to be specified on the four sides of the plot. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_xaxt">xaxt</code></td>
<td>

<p>Whether to plot an X axis (&quot;s&quot;) or not (&quot;n&quot;). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_yaxt">yaxt</code></td>
<td>

<p>Whether to plot an Y axis (&quot;s&quot;) or not (&quot;n&quot;). If no Y axis is drawn, <code>ysub</code> may be used to plot a sub-title to the Y axis. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_yaxs">yaxs</code></td>
<td>

<p>Y axis style, &quot;r&quot; enlarges the Y limits by 4 percents on each side for a cleaner look, &quot;i&quot; will not. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_ylim">ylim</code></td>
<td>

<p>The Y axis limits as a numerical vector of the form &quot;c(start, end)&quot; of the plot. Note that start &gt; end is allowed and leads to a &quot;reversed axis&quot;. Use &quot;NULL&quot; to guess the axis range from the data. See <code>plot.default</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_cex.lab">cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_cex.axis">cex.axis</code></td>
<td>

<p>The relative character size of x and y axis annotations (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_mgp">mgp</code></td>
<td>

<p>Length 3 vector defining the distance between the plot area and respectively the Y axis label, Y axis annotations and Y axis line (default: 3, 1, 0). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_tck">tck</code></td>
<td>

<p>The length of tick marks as a fraction of the smaller of the width or height of the plot (default: NA, meaning using <code>tcl</code> instead). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_tcl">tcl</code></td>
<td>

<p>The absolute length of a tick marks. Note that positive numbers put them inside the plot area (default: -0.5). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_xaxp">xaxp</code></td>
<td>

<p>Length 3 vector defining the ticks on the X axis : X of first tick, X of last tick and number of intervals between them (default: NA). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_yaxp">yaxp</code></td>
<td>

<p>Length 3 vector defining the ticks on the Y axis : Y of first tick, Y of last tick and number of intervals between them (default: NA). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_bty">bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of &quot;o&quot; (the default), &quot;l&quot;, &quot;7&quot;, &quot;c&quot;, &quot;u&quot;, or &quot;]&quot; the resulting box resembles the corresponding upper case letter. A value of &quot;n&quot; suppresses the box. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_las">las</code></td>
<td>

<p>The direction of both X and Y axis labels: 0 for labels parallel to the axes, 1 for horizontal labels, 2 for labels perpendicular to the axes and 3 for vertical labels. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_xgrid">xgrid</code></td>
<td>

<p>Single logical value, whether to draw a grid on X axis or not.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_new">new</code></td>
<td>

<p>Single logical value, whether to plot on top of previous track (<code>TRUE</code>) or in a new track (<code>FALSE</code>, default).
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_bg">bg</code></td>
<td>

<p>Single character value, defining the color of the background (margins included) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it, <code>NA</code> corresponds to the default <code>par</code>-based behavior.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_bg.inner">bg.inner</code></td>
<td>

<p>Single character value, defining the color of the background (margins excluded) as an english name or a hexadecimal code. <code>NA</code> corresponds to the default <code>par</code>-based behavior.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.bg_+3A_...">...</code></td>
<td>

<p>Not used, only ignores other arguments.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.boxes'>
Boxes plot of a track
</h2><span id='topic+draw.boxes'></span>

<h3>Description</h3>

<p>This function draws a slice of a track content, with a distinct box for each track element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.boxes(slice, start, end, maxElements = 50, maxDepth = 100, label = TRUE,
    labelStrand = FALSE, labelCex = 1, labelSrt = 0, labelAdj = "center",
    labelOverflow = TRUE, labelFamily = "sans", labelColor = "#000000",
	fillColor = "#BBBBBB", border = "#666666", cex.lab = 1, spacing = 0.2,
    bty = "o", groupBy = NA, groupPosition = NA, groupSize = NA, groupLwd = 1,
    fg = "#000000", normalize.y = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.boxes_+3A_slice">slice</code></td>
<td>

<p>A <code>data.frame</code> holding the data to plot, with elements in rows and data in columns.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_maxelements">maxElements</code></td>
<td>

<p>Single integer value, the maximum amount of boxes on the plot (if exhausted, only the amount of elements will be ploted).
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_maxdepth">maxDepth</code></td>
<td>

<p>Single integer value, the maximum amount of box heights allowed on the plot to avoid overlaps (if exhausted an error message will be ploted, turning <code>label</code> to <code>FALSE</code> might help).
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_label">label</code></td>
<td>

<p>Single logical value, whether to print labels on boxes or not.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labelstrand">labelStrand</code></td>
<td>

<p>Single logical value, whether to add the strand at the end of labels or not.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labelcex">labelCex</code></td>
<td>

<p>Single numeric value, character expansion factor for labels.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labelsrt">labelSrt</code></td>
<td>

<p>Single numeric value, string rotation angle for labels.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labeladj">labelAdj</code></td>
<td>

<p>'left', 'right' or 'center', the horizontal adjustement of the labels on the boxes.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labeloverflow">labelOverflow</code></td>
<td>

<p>Single logical value, whether to write labels on boxes too narrow to host them or not.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labelfamily">labelFamily</code></td>
<td>

<p>Single character value, the font family to use for labels ('serif', 'sans', 'mono' or 'Hershey'). 'serif' and 'sans' are not monospaced fonts, so label box sizes and collision handling might not work as expected with them.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_labelcolor">labelColor</code></td>
<td>

<p>The color to use for box labels (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>boxes</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;).
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_fillcolor">fillColor</code></td>
<td>

<p>The color to fill boxes with (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;).
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_border">border</code></td>
<td>

<p>The color to use for box borders (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;). Special values <code>NA</code> and <code>"fillColor"</code> can also be used to disable borders or use <code>fillColor</code> respectively.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_cex.lab">cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_spacing">spacing</code></td>
<td>

<p>Single numeric value, the vertical spacing between boxes, in proportion of the box height. Can alternatively be a single character value pointing a column to use to provide per-box spacing.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_bty">bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of &quot;o&quot; (the default), &quot;l&quot;, &quot;7&quot;, &quot;c&quot;, &quot;u&quot;, or &quot;]&quot; the resulting box resembles the corresponding upper case letter. A value of &quot;n&quot; suppresses the box. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_groupby">groupBy</code></td>
<td>

<p>Single character value, the name of a <code>slice</code> column to use for feature grouping. Grouped features are drawn on the same line, and joined by an horizontal segment. Use <code>NA</code> to disable feature grouping.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_groupposition">groupPosition</code></td>
<td>

<p>Single character value, the name of an integer <code>slice</code> column with the position of the feature in the group (starting at 1). If <code>groupBy</code> is provided with <code>groupPosition</code> and <code>groupSize</code>, groups containing out-of-range elements will be extended to the screen boundary. Use <code>NA</code> if feature grouping is disabled, or if awareness of out-of-range features is not critical.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_groupsize">groupSize</code></td>
<td>

<p>Single character value, the name of an integer <code>slice</code> column with the total amount of features in the group. See <code>groupPosition</code>.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_grouplwd">groupLwd</code></td>
<td>

<p>Single numeric value, the width of the line drawn to group elements.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_normalize.y">normalize.y</code></td>
<td>

<p>Single logical value, whether to normalize Y axis to 0:1 or have a Y axis growing with the number of stacked boxes. Default behavior is to fit all drawn boxes in 0:1 (default <code>ylim</code>), but setting <code>normalize.y</code> to <code>FALSE</code> allows to limit the display to a certain amount of boxes or to keep a consistant Y scaling between plots (for <code><a href="#topic+singlePlot">singlePlot</a></code> as an example), provided that <code>ylim</code> is set manually and accordingly.
</p>
</td></tr>
<tr><td><code id="draw.boxes_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+yline">yline</a></code>
</p>
<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.density'>
Two-dimensional density plot of a track
</h2><span id='topic+draw.density'></span>

<h3>Description</h3>

<p>This function is similar to <code><a href="#topic+draw.points">draw.points</a></code>, but draws a 2D density plot of the points instead of the points themselves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.density(slice, start, end, column = "value", cex.lab = 1, bty = "o",
    fg = "#000000", pal = grDevices::grey, border = NA, depth = 8, dpi = 7, bw.x = 0.005,
    bw.y = 0.2, precision = 1, skewing = 1.75, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.density_+3A_slice">slice</code></td>
<td>

<p>A <code>data.frame</code> holding the data to plot, with elements in rows and data in columns.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_column">column</code></td>
<td>

<p>Single character value, the name of the <code>slice</code> column to use for bar heights.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_cex.lab">cex.lab</code></td>
<td>

<p>See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_bty">bty</code></td>
<td>

<p>See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_pal">pal</code></td>
<td>

<p>A function returning a set of colors when provided a set of intensities between 0 and 1 (typically <code><a href="grDevices.html#topic+grey">grey</a></code>), for the various levels of density.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_border">border</code></td>
<td>

<p>The color to use for polygon borders (as a name, an integer or an hexadecimal character description). Special value <code>NA</code> can also be used to disable borders.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_depth">depth</code></td>
<td>

<p>Single integer value, the amount of different ranges to simplify the density into before plotting (corresponds to the <code>nlevels</code> argument of the <code><a href="graphics.html#topic+filled.contour">filled.contour</a></code> function).
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_dpi">dpi</code></td>
<td>

<p>Single numeric value, the Dots Per Inches resolution of the grid on which <code><a href="MASS.html#topic+kde2d">kde2d</a></code> will compute the density. The final grid size thus also depends on the figure dimensions, as provided in inches by <code>par("din")</code>.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_bw.x">bw.x</code></td>
<td>

<p>Single numeric value, the bandwidth to use for density estimation on the X (genomic) axis. Notice this value will be multiplied by the genomic width of the plotted window (in Mbp), to enforce a similar resolution at all zoom levels.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_bw.y">bw.y</code></td>
<td>

<p>Single numeric value, the bandiwdth to use for density estimation on the Y axis. It was typically chosen for a Y axis ranging from -1 to 1, wider axes could require wider bandwidths.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_precision">precision</code></td>
<td>

<p>Single numeric value, providing a simpler way to control the sharpness of the density plot than setting <code>depth</code>, <code>dpi</code>, <code>bw.x</code> and <code>bw.y</code> individually. The default <code>precision</code> of 1 has no effect on these 4 other arguments, values lower than 1 produce more averaged plots, values higher than 1 produce plots on which individual points are easier to distinguish from each others. 0 corresponds to an unreachable state of &quot;null precision&quot;, this argument should thus be defined as a strictly positive number.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_skewing">skewing</code></td>
<td>

<p>Single numeric value, defining how the color scale should be skewed toward small values. Higher <code>skewing</code> values makes zones of low density (typically isolated points) darker and decreases the overall contrast.
</p>
</td></tr>
<tr><td><code id="draw.density_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.hist'>
Histogram plot of a track
</h2><span id='topic+draw.hist'></span>

<h3>Description</h3>

<p>This function draws a slice of a track content, with a distinct vertical bar for each track element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.hist(slice, start, end, column = "value", fillColor = "#666666",
    border = "#666666", cex.lab = 1, origin = 0, bty = "o", fg = "#000000",
    ylim = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.hist_+3A_slice">slice</code></td>
<td>

<p>A <code>data.frame</code> holding the data to plot, with elements in rows and data in columns.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_column">column</code></td>
<td>

<p>Single character value, the name of the <code>slice</code> column to use for bar heights.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_fillcolor">fillColor</code></td>
<td>

<p>The color to fill vertical bars with (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;).
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_border">border</code></td>
<td>

<p>The color to use for box borders (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;). Special values <code>NA</code> and <code>"fillColor"</code> can also be used to disable borders or use <code>fillColor</code> respectively.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_cex.lab">cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_origin">origin</code></td>
<td>

<p>Single numeric value, the Y value of the horizontal side common to all boxes. Can also be the name of a <code>slice</code> numeric column to use as a segment-specific origin.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_bty">bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of &quot;o&quot; (the default), &quot;l&quot;, &quot;7&quot;, &quot;c&quot;, &quot;u&quot;, or &quot;]&quot; the resulting box resembles the corresponding upper case letter. A value of &quot;n&quot; suppresses the box. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_ylim">ylim</code></td>
<td>

<p>Numeric vector of length two, defining the Y axis boundaries. Any or both of them can be NA, meaning the missing boundary will be inferred from the data to plot.
</p>
</td></tr>
<tr><td><code id="draw.hist_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.pileup'>
Pileup plot of a BAM track
</h2><span id='topic+draw.pileup'></span>

<h3>Description</h3>

<p>This function draws a slice of a sequence pileup, highlighting polymorphisms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.pileup(slice, start, end, ylim = NA, bty = "o", label = TRUE, labelCex = 0.75,
    bases = c(A = "#44CC44", C = "#4444CC", G = "#FFCC00", T = "#CC4444"),
    maxRange = 500, cex.lab = 1, alphaOrder = 3, alphaMin = 0.1, fg = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.pileup_+3A_slice">slice</code></td>
<td>

<p>An integer matrix of read counts, with nucleotides in rows and positions in columns. Both dimensions must be named.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_ylim">ylim</code></td>
<td>

<p>See <code>plot.default</code>. <code>NA</code> will compute a dynamic ylim fiting the data.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_bty">bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of &quot;o&quot; (the default), &quot;l&quot;, &quot;7&quot;, &quot;c&quot;, &quot;u&quot;, or &quot;]&quot; the resulting box resembles the corresponding upper case letter. A value of &quot;n&quot; suppresses the box. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_label">label</code></td>
<td>

<p>Single logical value, whether to print nucleotide on bars or not.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_labelcex">labelCex</code></td>
<td>

<p>Single numeric value, character expansion factor for labels.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_bases">bases</code></td>
<td>

<p>Named character vector, defining the color to use for each nucleotide.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_maxrange">maxRange</code></td>
<td>

<p>Single integer value, nothing will be ploted if the plot window is wider by this value (in bases).
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_cex.lab">cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_alphaorder">alphaOrder</code></td>
<td>

<p>Single numeric value, the order of the formula controlling the transparency. Increase this value to increase sensitivity to rare variants.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_alphamin">alphaMin</code></td>
<td>

<p>Single numeric value, the minimal intensity in the formula controlling the transparency (between 0 and 1). Perfectly homozyguous positions will typically use this intensity of color.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.pileup_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.bam-class">track.bam-class</a></code>
</p>
<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.points'>
Scatter plot of a track
</h2><span id='topic+draw.points'></span>

<h3>Description</h3>

<p>This function draws a slice of a track content, with a distinct point for each track element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.points(slice, start, end, column = "value", pointColor = "#666666",
    cex.lab = 1, cex = 0.6, pch = "+", bty = "o", fg = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.points_+3A_slice">slice</code></td>
<td>

<p>A <code>data.frame</code> holding the data to plot, with elements in rows and data in columns.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_column">column</code></td>
<td>

<p>Single character value, the name of the <code>slice</code> column to use for bar heights.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_pointcolor">pointColor</code></td>
<td>

<p>The color to use for points (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;).
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_cex.lab">cex.lab</code></td>
<td>

<p>See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_cex">cex</code></td>
<td>

<p>See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_pch">pch</code></td>
<td>

<p>See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_bty">bty</code></td>
<td>

<p>See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.points_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.seq'>
Plot a sequence of nucleotides
</h2><span id='topic+draw.seq'></span>

<h3>Description</h3>

<p>This function draws a slice of a character vector, with labels and distinct colors for each nucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.seq(slice = NULL, start, end, bty = "o", labelCex = 0.75,
    bases = c(A = "#44CC44", C = "#4444CC", G = "#FFCC00", T = "#CC4444"),
    maxRange = 500, cex.lab = 1, fg = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.seq_+3A_slice">slice</code></td>
<td>

<p>Character vector, with a single letter per element.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_bty">bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of &quot;o&quot; (the default), &quot;l&quot;, &quot;7&quot;, &quot;c&quot;, &quot;u&quot;, or &quot;]&quot; the resulting box resembles the corresponding upper case letter. A value of &quot;n&quot; suppresses the box. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_labelcex">labelCex</code></td>
<td>

<p>Single numeric value, character expansion factor for labels.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_bases">bases</code></td>
<td>

<p>Named character vector, defining the color to use for each nucleotide (names have to be uppercase, <code>slice</code> is converted to uppercase for matching).
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_maxrange">maxRange</code></td>
<td>

<p>Single integer value, nothing will be ploted if the plot window is wider by this value (in bases).
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_cex.lab">cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.seq_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.fasta-class">track.fasta-class</a></code>
</p>
<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

<hr>
<h2 id='draw.steps'>
Stairway plot of a track
</h2><span id='topic+draw.steps'></span>

<h3>Description</h3>

<p>This function draws each element sliced from a track as a separate podium, defined by several start and end genomic coordinates. This representation may prove useful to represent results of Minimal Common Regions from algorithms such as SRA or GISTIC (see the 'cghRA' package).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  draw.steps(slice, start, end, startColumns = "start", endColumns = "end",
    maxDepth = 100, label = TRUE, labelStrand = FALSE, labelCex = 1, labelSrt = 0,
    labelAdj = "center", labelOverflow = TRUE, labelFamily = "sans",
    labelColor = "#000000", fillColor = "#BBBBBB", border = "#666666", cex.lab = 1,
    spacing = 0.1, bty = "o", fg = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.steps_+3A_slice">slice</code></td>
<td>

<p>A <code>data.frame</code> holding the data to plot, with elements in rows and data in columns.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_startcolumns">startColumns</code></td>
<td>

<p>Character vector naming the columns in <code>slice</code> to be used as step starts, from the widest to the narrowest.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_endcolumns">endColumns</code></td>
<td>

<p>Character vector naming the columns in <code>slice</code> to be used as step ends, from the widest to the narrowest.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_maxdepth">maxDepth</code></td>
<td>

<p>Single integer value, the maximum amount of box heights allowed on the plot to avoid overlaps (if exhausted an error message will be ploted, turning <code>label</code> to <code>FALSE</code> might help).
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_label">label</code></td>
<td>

<p>Single logical value, whether to print labels on boxes or not.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labelstrand">labelStrand</code></td>
<td>

<p>Single logical value, whether to add the strand at the end of labels or not.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labelcex">labelCex</code></td>
<td>

<p>Single numeric value, character expansion factor for labels.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labelsrt">labelSrt</code></td>
<td>

<p>Single numeric value, string rotation angle for labels.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labeladj">labelAdj</code></td>
<td>

<p>'left', 'right' or 'center', the horizontal adjustement of the labels on the boxes.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labeloverflow">labelOverflow</code></td>
<td>

<p>Single logical value, whether to write labels on boxes too narrow to host them or not.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labelfamily">labelFamily</code></td>
<td>

<p>Single character value, the font family to use for labels ('serif', 'sans', 'mono' or 'Hershey'). 'serif' and 'sans' are not monospaced fonts, so label box sizes and collision handling might not work as expected with them.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_labelcolor">labelColor</code></td>
<td>

<p>The color to use for box labels (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>boxes</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;).
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_fillcolor">fillColor</code></td>
<td>

<p>The color to fill boxes with (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;).
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_border">border</code></td>
<td>

<p>The color to use for box borders (as a name, an integer or an hexadecimal character description). It can alternatively be a function without argument, which returns a vector of as many colors as <code>slice</code> has rows. It can make direct variable call to any argument described on this page (including custom arguments passed via &quot;...&quot;). Special values <code>NA</code> and <code>"fillColor"</code> can also be used to disable borders or use <code>fillColor</code> respectively.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_cex.lab">cex.lab</code></td>
<td>

<p>The relative character size of x and y axis labels (default: 1). See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_spacing">spacing</code></td>
<td>

<p>Single numeric value, the vertical spacing between boxes, in proportion of the box height.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_bty">bty</code></td>
<td>

<p>A character string which determined the type of box which is drawn about plots. If bty is one of &quot;o&quot; (the default), &quot;l&quot;, &quot;7&quot;, &quot;c&quot;, &quot;u&quot;, or &quot;]&quot; the resulting box resembles the corresponding upper case letter. A value of &quot;n&quot; suppresses the box. See <code>par</code>.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_fg">fg</code></td>
<td>

<p>Single character value, defining the color of the foreground (axes, labels...) as an english name or a hexadecimal code. Similar to <code>par</code>'s argument but not relying on it.
</p>
</td></tr>
<tr><td><code id="draw.steps_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+draw.bg">draw.bg</a></code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+yline">yline</a></code>
</p>
<p><code><a href="#topic+draw.bg">draw.bg</a></code>, <code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.density">draw.density</a></code>, <code><a href="#topic+draw.hist">draw.hist</a></code>, <code><a href="#topic+draw.pileup">draw.pileup</a></code>, <code><a href="#topic+draw.points">draw.points</a></code>, <code><a href="#topic+draw.seq">draw.seq</a></code>
</p>

<hr>
<h2 id='drawable-class'>Class <code>"drawable"</code></h2><span id='topic+drawable-class'></span>

<h3>Description</h3>

<p>Reference classes extending this virtual class must have a <code>draw</code> method, so their objects can be managed by <code><a href="#topic+tk.browse">tk.browse</a></code> and <code><a href="#topic+browsePlot">browsePlot</a></code>.
</p>


<h3>Extends</h3>

<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>


<dl>
<dt><code>name</code>:</dt><dd><p>Custom name for the object, as a <code>character</code> vector of length 1.</p>
</dd>
<dt><code>parameters</code>:</dt><dd><p>A <code>list</code>, storing object-specific parameters to use as <code>draw</code> arguments.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code>callParams(chrom, start, end, ...)</code>:</dt><dd>
<p>Called with draw() arguments, it returns the final argument list handling default and overloaded parameters.<br />
- <strong>chrom, start, end, ...</strong>   : arguments passed to draw().
</p>
</dd>
<dt><code>check(warn = )</code>:</dt><dd>
<p>Raises an error if the object is not valid, else returns TRUE
</p>
</dd>
<dt><code>chromosomes()</code>:</dt><dd>
<p>[Virtual method]<br />
Returns the chromosome list as a vector. NULL is valid if non relevant, but should be avoided when possible.
</p>
</dd>
<dt><code>defaultParams(...)</code>:</dt><dd>
<p>Returns class-specific defaults for graphical parameters. Inheriting class should overload it to define their own defaults.<br />
- <strong>...</strong>   : may be used by inheriting methods, especially for inter-dependant parameters.
</p>
</dd>
<dt><code>draw(chrom, start = , end = , ...)</code>:</dt><dd>
<p>[Virtual method]<br />
Draws the object content corresponding to the defined genomic window, usually in a single plot area with coordinates in x and custom data in y.<br />
Overloading methods should use .self$callParams(chrom, start, end ...) to handle drawing parameters and NA coordinates in a consistent way.<br />
- <strong>chrom</strong>   : single integer, numeric or character value, the chromosomal location.<br />
- <strong>start</strong>   : single integer or numeric value, inferior boundary of the window. NA should refer to 0.<br />
- <strong>end</strong>     : single integer or numeric value, superior boundary of the window. NA should refer to .self$getChromEnd().<br />
- <strong>...</strong>     : additionnal drawing parameters (precede but do not overwrite parameters stored in the object).
</p>
</dd>
<dt><code>fix.param(parent = )</code>:</dt><dd>
<p>Edit drawing parameters using a Tcl-tk GUI<br />
- <strong>parent</strong>   : tcltk parent frame for inclusion, or NULL.
</p>
</dd>
<dt><code>getChromEnd(chrom)</code>:</dt><dd>
<p>[Virtual method]<br />
Returns as a single integer value the ending position of the object description of the given chromosome. NA (integer) is valid if non relevant, but should be avoided when possible.<br />
- <strong>chrom</strong>   : single integer, numeric or character value, the chromosomal location. NA is not required to be handled.
</p>
</dd>
<dt><code>getName()</code>:</dt><dd>
<p>'name' field accessor.
</p>
</dd>
<dt><code>getParam(name, ...)</code>:</dt><dd>
<p>Returns the parameter stored, or the default value if no custom value is stored for it.<br />
- <strong>name</strong>   : single character value, the name of the parameter to return.<br />
- <strong>...</strong>    : to be passed to defaultParams(), especially for inter-dependant parameters.
</p>
</dd>
<dt><code>initialize(name = , parameters = , ...)</code>:</dt><dd></dd>
<dt><code>setName(value)</code>:</dt><dd>
<p>'name' field mutator.
</p>
</dd>
<dt><code>setParam(name, value)</code>:</dt><dd>
<p>Updates a parameter stored in the object.<br />
- <strong>name</strong>    : single character value, the name of the parameter to set.<br />
- <strong>value</strong>   : the new value to assign to the parameter (any type). If missing the parameter is discarded, thus returning to dynamic default value.
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> show (<a href="methods.html#topic+envRefClass-class">envRefClass</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sliceable-class">sliceable</a></code>, <code><a href="#topic+crossable-class">crossable</a></code>
</p>

<hr>
<h2 id='drawable.list'>
Drawable list constructor
</h2><span id='topic+drawable.list'></span>

<h3>Description</h3>

<p>Produces a <code><a href="#topic+drawable.list-class">drawable.list</a></code> object, for <code><a href="#topic+tk.browse">tk.browse</a></code> and <code><a href="#topic+browsePlot">browsePlot</a></code> input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  drawable.list(files = character(0), objects = NULL, hidden = FALSE, warn = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drawable.list_+3A_files">files</code></td>
<td>

<p>Character vector, path and names of the files holding the drawables object (with .rdt or .rds extensions, see <code><a href="#topic+saveRDT">saveRDT</a></code> and <code><a href="base.html#topic+saveRDS">saveRDS</a></code> for further details). Can contain <code>NA</code> values, but providing a path allows updates to be saved.
</p>
</td></tr>
<tr><td><code id="drawable.list_+3A_objects">objects</code></td>
<td>

<p>List of <code><a href="#topic+drawable-class">drawable</a></code>-inheriting objects to include in the list. If <code>NULL</code>, objects will be extracted from <code>files</code>, else a list with as many elements as <code>files</code> is required.
</p>
</td></tr>
<tr><td><code id="drawable.list_+3A_hidden">hidden</code></td>
<td>

<p>Logical vector, whether to show tracks in <code><a href="#topic+tk.browse">tk.browse</a></code> and <code><a href="#topic+browsePlot">browsePlot</a></code> or not (this choice is not definitive). Recycled if necessary.
</p>
</td></tr>
<tr><td><code id="drawable.list_+3A_warn">warn</code></td>
<td>

<p>Single logical value, to be passed to the appropriate <code>check</code> method.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+drawable.list-class">drawable.list</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+drawable.list-class">drawable.list-class</a></code>, <code><a href="#topic+drawable-class">drawable-class</a></code>, <code><a href="#topic+tk.browse">tk.browse</a></code>, <code><a href="#topic+browsePlot">browsePlot</a></code>
</p>

<hr>
<h2 id='drawable.list-class'>Class <code>"drawable.list"</code></h2><span id='topic+drawable.list-class'></span>

<h3>Description</h3>

<p>The purpose of this class is to store and manage a collection of <code><a href="#topic+drawable-class">drawable</a></code> objects. These collections are to be used by <code><a href="#topic+tk.browse">tk.browse</a></code> and <code><a href="#topic+browsePlot">browsePlot</a></code> as input.
</p>
<p>Objects can be created by the <code><a href="#topic+drawable.list">drawable.list</a></code> constructor, and edited / created using the <code><a href="#topic+tk.tracks">tk.tracks</a></code> Tcl-tk interface.
</p>
<p>By default, the <code>drawable.list$add()</code> method is only able to handle objects from <code><a href="#topic+drawable-class">drawable</a></code>-inheriting classes saved in <a href="#topic+saveRDT">RDT</a> or <a href="base.html#topic+saveRDS">RDS</a> individual files. This can however be extended defining functions named <code>drawableFromFile.EXTENSION</code> and <code>drawableFromClass.CLASS</code>, in the global environment or a package. Such a function will take the same arguments as the <code>drawable.list$add()</code> method, and will only have to return a <code><a href="#topic+drawable-class">drawable</a></code>-inheriting object.
</p>


<h3>Extends</h3>

<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>


<dl>
<dt><code>classes</code>:</dt><dd><p>Read-only, returns a vector of <code>objects</code> classes.</p>
</dd>
<dt><code>count</code>:</dt><dd><p>Read-only, returns the length of <code>objects</code>, as a single integer.</p>
</dd>
<dt><code>files</code>:</dt><dd><p>Character vector, the paths where each drawable object is to be stored.</p>
</dd>
<dt><code>hidden</code>:</dt><dd><p>Logical vector, whether each object is to drawn or hidden in plots.</p>
</dd>
<dt><code>names</code>:</dt><dd><p>Read-only, returns a vector of <code>objects</code> 'name' fields.</p>
</dd>
<dt><code>objects</code>:</dt><dd><p>List of <code><a href="#topic+drawable-class">drawable</a></code>-inheriting objects.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code>add(file, track = , hidden = , ...)</code>:</dt><dd>
<p>Add a track to the list.<br />
- <strong>file</strong>     : single character value, the path to the file containing the 'drawable' object to add.<br />
- <strong>track</strong>    : a 'drawable' object to add. If NULL, will be extracted from 'file'.<br />
- <strong>hidden</strong>   : single logical value, whether the track is to be shown on plots or hidden. This value can be changed later.
- <strong>...</strong>      : further arguments to be passed to drawableFromFile.EXTENSION or drawableFromClass.CLASS, if relevant.
</p>
</dd>
<dt><code>check(warn = )</code>:</dt><dd>
<p>Raises an error if the object is not valid, else returns TRUE
</p>
</dd>
<dt><code>fix.files(parent = )</code>:</dt><dd>
<p>Edit drawable list using a Tcl-tk GUI<br />
- <strong>parent</strong>   : tcltk parent frame for inclusion, or NULL.
</p>
</dd>
<dt><code>fix.param(selection = , parent = )</code>:</dt><dd>
<p>Edit drawing parameters using a Tcl-tk GUI<br />
- <strong>selection</strong>   : single integer value, the position of the track selected in the list.<br />
- <strong>parent</strong>      : tcltk parent frame for inclusion, or NULL.
</p>
</dd>
<dt><code>get(index, what = )</code>:</dt><dd>
<p>Returns a single 'what' from the series<br />
- <strong>index</strong>   : single numeric value, the position of the track to get.<br />
- <strong>what</strong>    : single character value, the field to be exracted.
</p>
</dd>
<dt><code>getByClasses(classes, what = )</code>:</dt><dd>
<p>Returns a subset of 'what' from the series, querying by class inheritance<br />
- <strong>classes</strong>   : character vector, the class names of the objects to get (inheriting classes are picked too).<br />
- <strong>what</strong>      : single character value, the field to be exracted.
</p>
</dd>
<dt><code>getByNames(names, what = )</code>:</dt><dd>
<p>Returns a subset of 'what' from the series, querying by track name<br />
- <strong>names</strong>   : character vector, the names of the objects to get.<br />
- <strong>what</strong>    : single character value, the field to be exracted.
</p>
</dd>
<dt><code>getByPositions(positions, what = )</code>:</dt><dd>
<p>Returns a subset of 'what' from the series, querying by position<br />
- <strong>positions</strong>   : integer vector, the positions of the objects to get.<br />
- <strong>what</strong>        : single character value, the field to be exracted.
</p>
</dd>
<dt><code>getChromEnd(chrom)</code>:</dt><dd>
<p>Returns as a single integer value the maximal ending position of the object descriptions of the given chromosome.<br />
- <strong>chrom</strong>   : single integer, numeric or character value, the chromosomal location.
</p>
</dd></dl>
<p>,
</p>
<dl>
<dt><code>initialize(files = , objects = , hidden = , ...)</code>:</dt><dd></dd>
<dt><code>moveDown(toMove)</code>:</dt><dd>
<p>Increases the position of a track, switching position with the next one<br />
- <strong>toMove</strong>   : single numeric value, the position of the track to move.
</p>
</dd>
<dt><code>moveUp(toMove)</code>:</dt><dd>
<p>Decreases the position of a track, switching position with the previous one<br />
- <strong>toMove</strong>   : single numeric value, the position of the track to move.
</p>
</dd>
<dt><code>remove(toRemove)</code>:</dt><dd>
<p>Remove one or many tracks from the list<br />
- <strong>toRemove</strong>   : numeric vector, the positions of the tracks to remove.
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> show (<a href="methods.html#topic+envRefClass-class">envRefClass</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+drawable.list">drawable.list</a></code>, <code><a href="#topic+drawable-class">drawable-class</a></code>, <code><a href="#topic+tk.browse">tk.browse</a></code>, <code><a href="#topic+browsePlot">browsePlot</a></code>
</p>

<hr>
<h2 id='findDrawables'>
Find drawable objects in memory
</h2><span id='topic+findDrawables'></span>

<h3>Description</h3>

<p>This function searches an environment for <code><a href="#topic+drawable-class">drawable-class</a></code> inheriting objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  findDrawables(varNames = NA, envir = globalenv())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findDrawables_+3A_varnames">varNames</code></td>
<td>

<p>Character vector, the R expression(s) of potential <code><a href="#topic+drawable-class">drawable-class</a></code> inheriting objects (or lists of such objects) to check. If <code>NA</code>, an object name list will be generated from <code>envir</code>. Non-<code>NA</code> values are mainly used for internal recursive calls on lists, users should not have to change it.
</p>
</td></tr>
<tr><td><code id="findDrawables_+3A_envir">envir</code></td>
<td>

<p>The <code>environment</code> to look into for <code><a href="#topic+drawable-class">drawable-class</a></code> inheriting objects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Objects are currently found if defined as individual variables, as parts of <code><a href="#topic+drawable.list-class">drawable.list</a></code> objects or into standard R <code>lists</code>. <code>lists</code> are explored recursively, so <code>lists</code> embedded into other <code>lists</code> are explored too, whatever their depths.
</p>


<h3>Value</h3>

<p>Returns a character vector containing the R expression(s) to be evaluated in <code>envir</code> to get the <code><a href="#topic+drawable-class">drawable-class</a></code> inheriting objects.
</p>
<p>This vector carries an &quot;envir&quot; attribute containing the value passed to this function via the <code>envir</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tk.tracks">tk.tracks</a></code>
</p>

<hr>
<h2 id='hsBands'>Homo sapiens GRCh37 chromosome banding</h2><span id='topic+hsBands'></span>

<h3>Description</h3>

<p>Whole list of GRCh37 human chromosome G-banding, from the UCSC repository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(hsBands)
</code></pre>


<h3>Format</h3>

<p><code>track.table</code> object with 862 rows and the following columns : &quot;name&quot;, &quot;chrom&quot;, &quot;strand&quot;, &quot;start&quot;, &quot;end&quot;, &quot;stain&quot;.
</p>


<h3>Source</h3>

<p>University of California, Santa Cruz (genome.ucsc.edu)
</p>

<hr>
<h2 id='hsGenes'>Subset of Homo sapiens GRCh37 genes</h2><span id='topic+hsGenes'></span>

<h3>Description</h3>

<p>8000 randomly choosen GRCh37 human genes, from the NCBI repository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(hsGenes)
</code></pre>


<h3>Format</h3>

<p><code>data.frame</code> with 8000 rows and the following columns : &quot;chrom&quot;, &quot;start&quot;, &quot;end&quot; and &quot;name&quot;.
</p>


<h3>Source</h3>

<p>National Center for Biotechnology Information (ftp.ncbi.nih.gov)
</p>

<hr>
<h2 id='RDT+20storage+20files'>
Single refTable object storage
</h2><span id='topic+saveRDT'></span><span id='topic+readRDT'></span>

<h3>Description</h3>

<p>Functions to write a single <code><a href="#topic+refTable-class">refTable</a></code> object to a file, and to restore it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  saveRDT(object, file, compress = "gzip", compression_level = 6)
  readRDT(file, version = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDT+2B20storage+2B20files_+3A_object">object</code></td>
<td>

<p>An object of class <code><a href="#topic+refTable-class">refTable</a></code> to store.
</p>
</td></tr>
<tr><td><code id="RDT+2B20storage+2B20files_+3A_file">file</code></td>
<td>

<p>A connection or the name of the file where the R object is saved to or read from. The '.rdt' file extension is recommended, but not mandatory.
</p>
</td></tr>
<tr><td><code id="RDT+2B20storage+2B20files_+3A_compress">compress</code></td>
<td>

<p>To be passed to <code><a href="base.html#topic+save">save</a></code>
</p>
</td></tr>
<tr><td><code id="RDT+2B20storage+2B20files_+3A_compression_level">compression_level</code></td>
<td>

<p>To be passed to <code><a href="base.html#topic+save">save</a></code>
</p>
</td></tr>
<tr><td><code id="RDT+2B20storage+2B20files_+3A_version">version</code></td>
<td>

<p>Single logical value, whether to return the stored object or the version of the package used to store it.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions mimic the <code><a href="base.html#topic+saveRDS">saveRDS</a></code> and <code><a href="base.html#topic+saveRDS">saveRDS</a></code> system, without storing the class definition in the file (which can lead to about 100 useless Ko of data and longer loading times). It is intented to manage all classes extending <code><a href="#topic+refTable-class">refTable</a></code>, but no guarantee is provided for classes with non-atomic slots (particularly environment-derived ones).
</p>


<h3>Value</h3>

<p><code>saveRDT</code> returns nothing, <code>readRDT</code> returns the object stored in the file or a single character value (depends on the <code>version</code> argument).
</p>


<h3>Note</h3>

<p>To avoid whole-environment copying, environments of <code>function</code> slots are discarded.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refTable-class">refTable-class</a></code>, <code><a href="base.html#topic+saveRDS">saveRDS</a></code>, <code><a href="base.html#topic+saveRDS">saveRDS</a></code>
</p>

<hr>
<h2 id='read.gtf'>
Parses a GTF2 file
</h2><span id='topic+read.gtf'></span>

<h3>Description</h3>

<p>This function parses a simple &quot;Gene Transfer Format&quot; (GTF2.2) into a data.frame, as distributed by the UCSC Table Browser.
</p>
<p>As this format is an extension of the &quot;Gene Feature Format&quot; (GFF3), some retro-compatibility can be expected but not guaranteed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read.gtf(file, attr = c("split", "intact", "skip"), features = NULL, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.gtf_+3A_file">file</code></td>
<td>

<p>Single character value, the path and name of the GTF2 file to parse (possibly gzipped).
</p>
</td></tr>
<tr><td><code id="read.gtf_+3A_attr">attr</code></td>
<td>

<p>Single character value, defining how to deal with attributes. &quot;skip&quot; discards the attributes data, &quot;intact&quot; does not process it and &quot;split&quot; adds a column for each attribute (identified by their names).
</p>
</td></tr>
<tr><td><code id="read.gtf_+3A_features">features</code></td>
<td>

<p>Character vector, if not <code>NULL</code> only rows with a &quot;feature&quot; column value from this list will be kept.
</p>
</td></tr>
<tr><td><code id="read.gtf_+3A_quiet">quiet</code></td>
<td>

<p>Single logical value, whether to send diagnostic messages or not.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with the standard GTF2 columns. The &quot;strand&quot; column is converted to <code>factor</code>, &quot;?&quot; are turned to <code>NA</code> and &quot;.&quot; are kept for features where stranding is not relevant (See the GFF3 specification).
</p>


<h3>Note</h3>

<p>Currently not implemented :
</p>

<ul>
<li><p>FASTA section and sequences (error raising)
</p>
</li>
<li><p>Special character escaping (error raising)
</p>
</li>
<li><p>Attribute quotes (kept)
</p>
</li>
<li><p>Sections (all data pooled)
</p>
</li>
<li><p>Meta data (ignored)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>GTF2.2 specification : <a href="http://mblab.wustl.edu/GTF22.html">http://mblab.wustl.edu/GTF22.html</a>
</p>
<p>GFF3 Sequence Ontology specification : <a href="http://www.sequenceontology.org/gff3.shtml">http://www.sequenceontology.org/gff3.shtml</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table.GTF">track.table.GTF</a></code>
</p>

<hr>
<h2 id='refTable-class'>Class <code>"refTable"</code></h2><span id='topic+refTable-class'></span>

<h3>Description</h3>

<p>This class is similar to the <code>data.frame</code> standard R class, following the object-oriented paradigm. The use of Reference Class system allows significative memory and time saving, making this class more suitable than <code>data.frame</code> to handle large tabular data.
</p>
<p>Objects can be created by two distincts means :
</p>

<ul>
<li><p>The <code><a href="#topic+refTable">refTable</a></code> constructor, similar to the <code>data.frame</code> constructor. It imports a single <code>data.frame</code> or a collection of vectors into the object, and check immediatly for validity.
</p>
</li>
<li><p>The <code>new</code> function (standard behavior for S4 and reference classes), which produces an empty object and do NOT check for validity. You can provide as arguments the values to use as the new object fields, if you know what you are doing.
</p>
</li></ul>



<h3>Extends</h3>

<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Implementation</h3>



<h4>Data storage</h4>

<p>refTable objects store data into their <code>values</code> field as vectors. As <code>values</code> is an environment, manipulating a refTable implies a pass-by-reference paradigm rather than the standard R pass-by-copy, i.e. data duplication (and so time and memory wasting) is widely reduced. As an example, updating a single cell in a <code>data.frame</code> leads to the duplication of the whole table in memory ('before' and 'after' versions), while in a refTable the duplication is limited to the involved column. 
</p>



<h4>Column names</h4>

<p>To facilitate column renaming, the vectors in <code>values</code> are not named according to the user-level column names, but according to references stored in the <code>colReferences</code> field (integers greater than 0 converted to characters). Rename a column only updates the <code>colNames</code> field and leave the <code>values</code> one alone, as the column reference does not change.
</p>



<h4>Data extraction</h4>

<p>Data extracted from refTable are usually returned as <code>data.frame</code>, for a more comfortable R usage. The extraction mechanism handles <code>data.frame</code> extraction mechanisms, and relies on the <code>indexes</code> method to handle the others.
</p>

<ul>
<li><p>Rows and columns may be selected by a numeric vector, as for R <code>data.frame</code> and vectors.
</p>
</li>
<li><p>They also may be selected by a logical vector, defining for each row / column if it is to be selected (<code>TRUE</code>) or not (<code>FALSE</code>). Such vectors are recycled if not long enough to cover all the rows / columns.
</p>
</li>
<li><p>A character vector defining the names of the rows / columns to select may also be used to extract data.
</p>
</li>
<li><p>The <code>NULL</code> value may be used to select all rows / columns.
</p>
</li>
<li><p>An unevaluated expression, as returned by <code>expression</code> or <code>parse</code> may be used to select rows in the table environment. See 'examples'.
</p>
</li></ul>




<h3>Fields</h3>


<dl>
<dt><code>colCount</code>:</dt><dd><p>Single <code>integer</code> value, the amount of rows in the table.</p>
</dd>
<dt><code>colIterator</code>:</dt><dd><p>Single <code>integer</code> value, last column reference used.</p>
</dd>
<dt><code>colNames</code>:</dt><dd><p><code>Character</code> vector, the names of all rows (may be empty).</p>
</dd>
<dt><code>colReferences</code>:</dt><dd><p><code>Character</code> vector, the column names in the <code>values</code> environment.</p>
</dd>
<dt><code>rowCount</code>:</dt><dd><p>Single <code>integer</code> value, the amount of rows in the table.</p>
</dd>
<dt><code>rowNamed</code>:</dt><dd><p>Single <code>logical</code> value, whether row names should be considered or not.</p>
</dd>
<dt><code>rowNames</code>:</dt><dd><p><code>Character</code> vector, the names of all rows (may be empty).</p>
</dd>
<dt><code>values</code>:</dt><dd><p>An <code>environment</code> storing the columns as vectors.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code>addColumn(content, name, after = )</code>:</dt><dd>
<p>Adds a column in the table<br />
- <strong>content</strong>   : values to fill in the new column, as a vector.<br />
- <strong>name</strong>      : name of the new column, as character.<br />
- <strong>after</strong>     : where to add the column, as the index (numeric) or name (character) of the column on its left
</p>
</dd>
<dt><code>addDataFrame(dataFrame)</code>:</dt><dd>
<p>Adds a data.frame content to the refTable<br />
- <strong>dataFrame</strong>   : the data to add.
</p>
</dd>
<dt><code>addEmptyRows(amount, newNames)</code>:</dt><dd>
<p>Add rows filled with NA at the bottom of the table.<br />
- <strong>amount</strong>     : single integer value, the amount of rows to add.<br />
- <strong>newNames</strong>   : character vector, the names of the new rows. Ignored if the table is not row named.
</p>
</dd>
<dt><code>addList(dataList, row.names)</code>:</dt><dd>
<p>Adds a list content to the refTable<br />
- <strong>dataList</strong>    : the data to add.<br />
- <strong>row.names</strong>   : character vector with the names of the enw rows.
</p>
</dd>
<dt><code>addVectors(..., row.names)</code>:</dt><dd>
<p>Adds vectors to the refTable<br />
- <strong>...</strong>   : named vectors to add.
</p>
</dd>
<dt><code>check(warn = )</code>:</dt><dd>
<p>Raises an error if the object is not valid, else returns TRUE
</p>
</dd>
<dt><code>coerce(j = , class, levels, ...)</code>:</dt><dd>
<p>Coerces a single column to a different class<br />
- <strong>j</strong>        : column index (numeric) or name (character).<br />
- <strong>class</strong>    : name of the class to coerce 'j' to.<br />
- <strong>levels</strong>   : if 'class' is factor, the levels to use.<br />
- <strong>...</strong>      : further arguments to be passed to the 'as' method (for atomics) or function (for other classes).
</p>
</dd>
<dt><code>colOrder(newOrder, na.last = , decreasing = )</code>:</dt><dd>
<p>Reorder the columns of the tables (duplication / subsetting are NOT handled)<br />
- <strong>newOrder</strong>   : new order to apply, as an integer vector of character vector of column names.
</p>
</dd>
<dt><code>delColumns(targets)</code>:</dt><dd>
<p>Deletes a column from the table<br />
- <strong>targets</strong>   : character vector, the name(s) of the column(s) to delete.
</p>
</dd>
<dt><code>erase()</code>:</dt><dd>
<p>Remove all the rows and columns in the table.
</p>
</dd>
<dt><code>extract(i = , j = , drop = , asObject = )</code>:</dt><dd>
<p>Extracts values into a data.frame or vector<br />
- <strong>i</strong>          : row selection, see indexes() for further details.<br />
- <strong>j</strong>          : column selection, see indexes() for further details.<br />
- <strong>drop</strong>       : if TRUE and querying a single column, will return a vector instead of a data.frame.<br />
- <strong>asObject</strong>   : if TRUE results will be served in the same class as the current object.
</p>
</dd>
<dt><code>fill(i = , j = , newValues)</code>:</dt><dd>
<p>Replaces values in a single column<br />
- <strong>i</strong>           : row indexes (numeric) or names (character). NULL or missing for all rows.<br />
- <strong>j</strong>           : column index (numeric) or name (character).<br />
- <strong>newValues</strong>   : vector of values to put in the object
</p>
</dd>
<dt><code>getColCount()</code>:</dt><dd>
<p>'colCount' field accessor.
</p>
</dd>
<dt><code>getColNames()</code>:</dt><dd>
<p>'colNames' field accessor.
</p>
</dd>
<dt><code>getLevels(j = )</code>:</dt><dd>
<p>Get levels of a factor column<br />
- <strong>j</strong>   : column index (numeric) or name (character).
</p>
</dd>
<dt><code>getRowCount()</code>:</dt><dd>
<p>'rowCount' field accessor.
</p>
</dd>
<dt><code>getRowNames()</code>:</dt><dd>
<p>'rowNames' field accessor.
</p>
</dd>
<dt><code>indexes(i, type = )</code>:</dt><dd>
<p>Checks row or column references and return numeric indexes<br />
- <strong>i</strong>   : reference to the rows or columns to select (NA not allowed), as :<br />
- missing or NULL (all rows or columns)<br />
- vector of numeric indexes to select<br />
- vector of character indexes to select (if the dimension is named)<br />
- vector of logical with TRUE on each value to select, FALSE otherwise<br />
- expression object (as returned by e(...)), to be evaluated in the 'values' environment
</p>
</dd>
<dt><code>initialize(rowCount = , rowNames = , rowNamed = , colCount = , colNames = , colReferences = , colIterator = , values = , ...)</code>:</dt><dd></dd>
<dt><code>metaFields()</code>:</dt><dd>
<p>Returns a character vector of fields that do not directly depend on the tabular content, for clonage.
</p>
</dd>
<dt><code>rowOrder(newOrder, na.last = , decreasing = )</code>:</dt><dd>
<p>Reorder the rows of the tables (duplication / subsetting are handled)<br />
- <strong>newOrder</strong>     : new order to apply, as an integer vector of row indexes or a character vector of column names.<br />
- <strong>na.last</strong>      : to be passed to order(), if 'newOrder' is a column name vector.<br />
- <strong>decreasing</strong>   : to be passed to order(), if 'newOrder' is a column name vector.
</p>
</dd>
<dt><code>setColNames(j, value)</code>:</dt><dd>
<p>Replaces one or many column names.<br />
- <strong>j</strong>       : subset of columns to rename.<br />
- <strong>value</strong>   : new column names to use, as a character vector.
</p>
</dd>
<dt><code>setLevels(j = , newLevels)</code>:</dt><dd>
<p>Get or replace levels of a factor column<br />
- <strong>j</strong>           : column index (numeric) or name (character).<br />
- <strong>newLevels</strong>   : new levels to use, as a character vector.
</p>
</dd>
<dt><code>setRowNames(value)</code>:</dt><dd>
<p>Replaces the entire row names set.<br />
- <strong>value</strong>   : new row names to use in the table, as a character vector. NULL will disable row naming.
</p>
</dd>
<dt><code>types(j = )</code>:</dt><dd>
<p>Returns classes of selected columns<br />
- <strong>j</strong>   : column indexes or names (NULL for all columns)
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>, overloaded)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> show (<a href="methods.html#topic+envRefClass-class">envRefClass</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # New empty refTable
  tab &lt;- new("refTable")
  tab$addColumn(1:5, "C1")
  tab$addColumn(letters[1:5], "C2")
  tab$setRowNames(LETTERS[11:15])
  
  # New filled refTable (same content)
  tab &lt;- refTable(C1=1:5, C2=letters[1:5], row.names=LETTERS[11:15])
  
  # Whole table print
  print(tab$extract())
  
  # Data update
  tab$fill(c(2,4), 2, c("B","D"))
  
  # Data extraction
  print(tab$extract(1:3))
  print(tab$extract(c(TRUE, FALSE)))
  print(tab$extract("K", "C1"))
  
  # Expression-based extraction
  expr &lt;- expression(C1 %% 2 == 1)
  print(tab$extract(expr))
  
  # Table extension
  tab$addEmptyRows(5L, LETTERS[1:5])
  tab$fill(6:10, "C1", 6:10)
  print(tab$extract())
  
  # Filling from R objects
  tab &lt;- new("refTable")
  print(tab$extract())
  tab$addVectors(C1=1:5, C2=letters[1:5])
  print(tab$extract())
  tab$addList(list(C1=6:8, C3=LETTERS[6:8]))
  print(tab$extract())
  
  # Beware of recycling !
  tab$addVectors(C1=9:15, C3=LETTERS[9:10])
  print(tab$extract())
</code></pre>

<hr>
<h2 id='refTable-constructor'>
refTable class constructor
</h2><span id='topic+refTable'></span>

<h3>Description</h3>

<p>This function returns a new <code><a href="#topic+refTable-class">refTable</a></code> object from various arguments.
</p>
<p>Notice the <code>new()</code> alternative can be used to produce an empty object, setting only the fields not the content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  refTable(..., row.names, warn = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refTable-constructor_+3A_...">...</code></td>
<td>

<p>A <code>data.frame</code>, a <code>list</code> or a set of named vectors to use as columns. For list and vectors, <code>row.names</code> will be used as row names if provided.
</p>
</td></tr>
<tr><td><code id="refTable-constructor_+3A_row.names">row.names</code></td>
<td>

<p>Character vector, the names of the rows for list or vector input.
</p>
</td></tr>
<tr><td><code id="refTable-constructor_+3A_warn">warn</code></td>
<td>

<p>Single logical value, to be passed to the <code><a href="#topic+refTable-class">refTable</a></code> <code>check</code> method.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+refTable-class">refTable</a></code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refTable-class">refTable-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # From vectors
  tab &lt;- refTable(colA=1:5, colB=letters[1:5])
  print(tab$extract(3,))
  
  # From list (recycling)
  columns &lt;- list(number=1, letters=LETTERS)
  tab &lt;- refTable(columns)
  print(tab$extract())
  
  # data.frame conversion
  dataFrame &lt;- data.frame(colA=1:5, colB=letters[1:5])
  tab &lt;- refTable(dataFrame)
  print(tab$extract())
</code></pre>

<hr>
<h2 id='segMerge'>
Merges consecutive segments
</h2><span id='topic+segMerge'></span>

<h3>Description</h3>

<p>Given a set of segments defined by &quot;chrom&quot;, &quot;start&quot;, &quot;end&quot; and various data, it merges consecutive rows (sorted by &quot;chrom&quot; then &quot;start&quot;) that share same data. As an example, it is useful to merge consecutive regions of the genome sharing same copy numbers after modelization, or filling small gaps. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  segMerge(segTable, on = names(segTable), fun = list(unique, start=min, end=max),
    group = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="segMerge_+3A_segtable">segTable</code></td>
<td>

<p>A <code>data.frame</code> of segments, with at least &quot;chrom&quot;, &quot;start&quot; and &quot;end&quot; columns. Standard behavior (default <code>fun</code> value) assumes &quot;start&quot; and &quot;end&quot; to be at least <code>numeric</code>, preferably <code>integer</code>.
</p>
</td></tr>
<tr><td><code id="segMerge_+3A_on">on</code></td>
<td>

<p>Character vector, <code>segTable</code> columns that must all be identical for the consecutive rows to be merged. For convenience, &quot;chrom&quot; is forced in and &quot;start&quot; / &quot;end&quot; are forced out.
</p>
</td></tr>
<tr><td><code id="segMerge_+3A_fun">fun</code></td>
<td>

<p>A <code>list</code> of <code>function</code>s, defining how to merge values when merging rows. It should contain an unamed element for the default function, and named elements to deal with specific columns.
</p>
</td></tr>
<tr><td><code id="segMerge_+3A_group">group</code></td>
<td>

<p>A vector with as many values as <code>segTable</code> rows. If provided, <code>on</code> is ignored and <code>group</code> is directly used as a grouping factor to merge consecutive rows.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>data.frame</code> similar to <code>segTable</code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+segOverlap">segOverlap</a></code>
</p>

<hr>
<h2 id='segOverlap'>
Merges overlapping segments
</h2><span id='topic+segOverlap'></span>

<h3>Description</h3>

<p>Given a set of segments defined by &quot;chrom&quot;, &quot;start&quot;, &quot;end&quot; and various data, it merges overlapping or jointive rows. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  segOverlap(segTable, fun = list(unique.default, start=min, end=max),
    factorsAsIntegers = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="segOverlap_+3A_segtable">segTable</code></td>
<td>

<p>A <code>data.frame</code> of segments, with at least &quot;chrom&quot;, &quot;start&quot; and &quot;end&quot; columns. Standard behavior (default <code>fun</code> value) assumes &quot;start&quot; and &quot;end&quot; to be at least <code>numeric</code>, preferably <code>integer</code>.
</p>
</td></tr>
<tr><td><code id="segOverlap_+3A_fun">fun</code></td>
<td>

<p>A <code>list</code> of <code>function</code>s, defining how to merge values when merging rows. It should contain an unamed element for the default function, and named elements to deal with specific columns.
</p>
</td></tr>
<tr><td><code id="segOverlap_+3A_factorsasintegers">factorsAsIntegers</code></td>
<td>

<p>Single logical value, whether to handle columns of class factor as integers or as is. Using <code>TRUE</code> dramatically shortens computation time in default user case (&quot;chrom&quot; column of class factor, &quot;unique&quot; function applied), but <code>FALSE</code> may be required to allow non-standard merging functions to access encoded values rather than factor indexes.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>data.frame</code> similar to <code>segTable</code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+segMerge">segMerge</a></code>
</p>

<hr>
<h2 id='singlePlot'>
Whole genome plot of a drawable list
</h2><span id='topic+singlePlot'></span>

<h3>Description</h3>

<p>This function plots a drawable list on all the chromosomes side by side, with a constant X axis scale. It mainly defines a custom layout for a series of <code><a href="#topic+browsePlot">browsePlot</a></code> calls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  singlePlot(drawables, columns = 4, exclude = c("X", "Y"), add = c(5e6, 15e6),
    vertical = FALSE, capWidth = "1 cm", spacer = "1 cm", finalize = TRUE,
    cap.border = "black", cap.font.col = "black", cap.bg.col = NA, cap.adj = c(0.5, 0.5),
    cap.cex = 2, cap.font = 2, mar = c(0,0,0,0), bty = "n", xaxt = "n", xgrid = FALSE,
    yaxt = "n", ylab = "", ysub = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="singlePlot_+3A_drawables">drawables</code></td>
<td>

<p>A <code><a href="#topic+drawable.list-class">drawable.list</a></code> object containing the tracks to plot, as returned by <code><a href="#topic+drawable.list">drawable.list</a></code> or <code><a href="#topic+tk.tracks">tk.tracks</a></code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_columns">columns</code></td>
<td>

<p>Single integer value, column count in the chromosome layout.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_exclude">exclude</code></td>
<td>

<p>Character vector, the names of the chromosomes to not plot.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_add">add</code></td>
<td>

<p>Numeric vector of length 2, the margins to add before and after each chromosome, in base pairs.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_vertical">vertical</code></td>
<td>

<p>Single logical value, whether to produce a plot showing chromosomes horizontally or vertically. Actually the figure will need to be manually rotated in the resulting file.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_capwidth">capWidth</code></td>
<td>

<p>Single value defining the width of the chromosome name caps, in a suitable format for <code>layout</code> <code>widths</code> argument.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_spacer">spacer</code></td>
<td>

<p>Single value defining the height of the inter-chromosome gaps, in a suitable format for <code>layout</code> <code>heights</code> argument.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_finalize">finalize</code></td>
<td>

<p>Single logical value, whether to fill unused space with a blank plot and return to default layout or allow further manual addition. If <code>FALSE</code>, the next plot will fill the whole blank space left.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_cap.border">cap.border</code></td>
<td>

<p>Single value defining the color of chromosome cap borders. See the <code>border</code> argument of <code>rect</code> for further details.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_cap.font.col">cap.font.col</code></td>
<td>

<p>Single value defining the color of chromosome names in the caps. See the <code>font</code> argument of <code>par</code> for further details.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_cap.bg.col">cap.bg.col</code></td>
<td>

<p>Single value defining the color of chromosome cap background. See the <code>col</code> argument of <code>rect</code> for further details.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_cap.adj">cap.adj</code></td>
<td>

<p>Numeric vector of length 2 defining the X and Y adjustment of the chromosome names in the caps, ranging between 0 (left / bottom) to 1 (right / top).
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_cap.cex">cap.cex</code></td>
<td>

<p>Single value defining the character expansion factor for chromosome names in the caps. See the <code>cex</code> argument of <code>par</code> for further details.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_cap.font">cap.font</code></td>
<td>

<p>Single value defining the font type of chromosome names in the caps. See the <code>font</code> argument of <code>par</code> for further details.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_mar">mar</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_bty">bty</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_xaxt">xaxt</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_xgrid">xgrid</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_yaxt">yaxt</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_ylab">ylab</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_ysub">ysub</code></td>
<td>

<p>Refer to ... described here-after. To disable the overriding and let each track define its own value as usual, set this parameter to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="singlePlot_+3A_...">...</code></td>
<td>

<p>Further arguments will be passed to <code><a href="#topic+browsePlot">browsePlot</a></code> for each chromosome, which itself will pass it to the <code>draw</code> method of each track, overriding the class default and object setting.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+browsePlot">browsePlot</a></code>, <code><a href="#topic+drawable.list-class">drawable.list</a></code>, <code><a href="#topic+drawable-class">drawable</a></code>
</p>

<hr>
<h2 id='sliceable-class'>Class <code>"sliceable"</code></h2><span id='topic+sliceable-class'></span>

<h3>Description</h3>

<p>Reference classes extending this virtual class must have a <code>slice</code> method, as a generic <code>draw</code> method based on it is provided.
</p>


<h3>Extends</h3>

<p>Class <code><a href="#topic+drawable-class">drawable</a></code>, directly.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li></ul>



<h3>Methods</h3>


<dl>
<dt><code>slice(chrom, start, end)</code>:</dt><dd>
<p>[Virtual method]<br />
Extract elements in the specified window, in a format suitable to draw().<br />
- <strong>chrom</strong>   : single integer, numeric or character value, the chromosomal location. NA is not handled.<br />
- <strong>start</strong>   : single integer or numeric value, inferior boundary of the window. NA is not handled.<br />
- <strong>end</strong>     : single integer or numeric value, superior boundary of the window. NA is not handled.
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> chromosomes (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> draw (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> show (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+drawable-class">drawable</a></code>, <code><a href="#topic+crossable-class">crossable</a></code>, <code>cghRA.array</code>
</p>

<hr>
<h2 id='subtrack'>
Extract elements within a genomic window
</h2><span id='topic+subtrack'></span><span id='topic+sizetrack'></span><span id='topic+istrack'></span>

<h3>Description</h3>

<p><code>subtrack</code> extracts lines from a <code>data.frame</code>, <code>list</code> or <code>vector</code> collection within a single genomic window, defined by a chromosome name, a starting and an ending positions. As this is a common task in genome-wide analysis, this function relies on an optimized C code in order to achieve good performances.
</p>
<p><code>sizetrack</code> is very similar to <code>subtrack</code>, but only count lines without extracting the data.
</p>
<p><code>istrack</code> checks if a collection of data is suitable for <code>subtrack</code> and <code>sizetrack</code> (See 'Track definition' for further details). As this operation is quite expensive and should be performed once, it is up to the user to check its data before subtracking.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  istrack(...)
  subtrack(...)
  sizetrack(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtrack_+3A_...">...</code></td>
<td>

<p>A collection of data to be considered as a single track. Named vectors are considered as single columns, <code>data.frame</code> and <code>list</code> as collections of columns, all parallelized in a single bidimensional table (assuming they all have same lengths / row counts). See 'Track definition' for further details.
</p>
<p>For <code>subtrack</code> and <code>sizetrack</code>, the first arguments must be the following (preferably unnamed) :
</p>

<ul>
<li><p>chromosome location (<code>integer</code> or <code>character</code>, according to 'chrom' type)
</p>
</li>
<li><p>starting position on the chromosome (<code>integer</code>, considered within)
</p>
</li>
<li><p>ending position on the chromosome (<code>integer</code>, considered within)
</p>
</li>
<li><p>chromosome index (<code>integer</code> vector, see below)
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>The C code relies heavily on the ordering to fastly retrieve the elements that overlap the queried window. Elements entirely comprised in the window are returned, as well as elements that only partially overlap it.
</p>


<h3>Value</h3>

<p><code>subtrack</code> returns a single <code>data.frame</code> merging all columns provided, with the subset of rows corresponding to elements in the queried window. This <code>data.frame</code> has no row name, and is a valid track (See 'Track definition' for further details).
</p>
<p><code>sizetrack</code> returns a single <code>integer</code> value corresponding to the count of rows in the queried window.
</p>
<p><code>istrack</code> returns a single <code>TRUE</code> value if the data collection provided is a valid track. Otherwise it returns a single <code>FALSE</code> value, with a &quot;why&quot; attribute containing a single character string explaining the (first) condition that is not fulfilled.
</p>


<h3>Track definition</h3>

<p>A track is defined as a <code>data.frame</code> with a variable amount of data (in columns) about a variable amount of features (in rows).
</p>
<p>3 columns are mandatory, with restricted names and types :
</p>

<dl>
<dt>chrom</dt><dd><p>The chromosomal location of the feature, as <code>integer</code> or <code>factor</code>.</p>
</dd>
<dt>start</dt><dd><p>The starting position of the feature on the chromosome, as <code>integer</code>.</p>
</dd>
<dt>end</dt><dd><p>The ending position of the feature on the chromosome, as <code>integer</code>.</p>
</dd>
</dl>

<p>The track is supposed to be ordered by chromosome, then by starting position. When chromosomes are stored as <code>factors</code>, they need to be numerically ordered by their internal codes (as the <code>order</code> function does), not alphabetically by their labels.
</p>


<h3>Chromosome index</h3>

<p>In order to guarantee good performances, chromosomes are to be indexed. As the rows are supposed to be ordered by chromosome, then by starting position (see 'Track definition'), reminding starting or ending rows of each chromosome can save huge amounts of computation time in large tracks.
</p>
<p>The following specifications must be fulfilled :
</p>

<ul>
<li><p>It must be an <code>integer</code> vector, with the last row index of each chromosome in the track indexed.
</p>
</li>
<li><p>Values are to be ordered by chromosome, in the same way than the 'chrom' column.
</p>
</li>
<li><p>For <code>integer</code> 'chrom', values are extracted by position (chromosome '1' is the first value ...).
</p>
</li>
<li><p>For <code>factor</code> 'chrom', values are extracted by names (named with 'chrom' levels).
</p>
</li>
<li><p>Chromosomes without data in the track must be described, with <code>NA integer</code> values.
</p>
</li></ul>

<p>See the 'Example' section below for index computation.
</p>


<h3>Note</h3>

<p>These three functions are proposed for generic usage on <code>data.frame</code>, <code>list</code> or vectors. The <code><a href="#topic+track.table-class">track.table</a></code> class implements more suitable <code>slice</code>, <code>size</code> and <code>check</code> methods, and handles autonomously the indexing.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  # Exemplar data : subset of human genes
  data(hsGenes)
  
  # Track validity
  print(istrack(hsGenes))
  hsGenes &lt;- hsGenes[ order(hsGenes$chrom, hsGenes$start) ,]
  print(istrack(hsGenes))
  
  # Chromosome index (factorial 'chrom')
  index &lt;- tapply(1:nrow(hsGenes), hsGenes$chrom, max)
  
  # Factor chrom query
  print(class(hsGenes$chrom))
  subtrack("1", 10e6, 15e6, index, hsGenes)
  
  # Row count
  a &lt;- nrow(subtrack("1", 10e6, 15e6, index, hsGenes))
  b &lt;- sizetrack("1", 10e6, 15e6, index, hsGenes)
  if(a != b) stop("Inconsistency")
  
  # Multiple sources
  length &lt;- hsGenes$end - hsGenes$start
  subtrack("1", 10e6, 15e6, index, hsGenes, length)
  subtrack("1", 10e6, 15e6, index, hsGenes, length=length)
  
  # Speed comparison (x200 here)
  system.time(
    for(i in 1:40000) {
      subtrack("1", 10e6, 15e6, index, hsGenes)
    }
  )
  system.time(
    for(i in 1:200) {
      hsGenes[ hsGenes$chrom == "1" &amp; hsGenes$start &lt;= 15e6 &amp; hsGenes$end &gt;= 10e6 ,]
    }
  )
  
  # Convert chrom from factor to integer
  hsGenes$chrom &lt;- as.integer(as.character(hsGenes$chrom))
  
  # Chromosome index (integer 'chrom')
  index &lt;- rep(NA_integer_, 24)
  tmpIndex &lt;- tapply(1:nrow(hsGenes), hsGenes$chrom, max)
  index[ as.integer(names(tmpIndex)) ] &lt;- tmpIndex
  
  # Integer chrom query
  print(class(hsGenes$chrom))
  subtrack(1, 10e6, 15e6, index, hsGenes)
</code></pre>

<hr>
<h2 id='tk.browse'>
R Genome browser
</h2><span id='topic+tk.browse'></span><span id='topic+browsePlot'></span>

<h3>Description</h3>

<p>The <code>browsePlot</code> function produces an usual R plot from a <code><a href="#topic+drawable-class">drawable</a></code> inheriting object list, at specific coordinates in the genome.
</p>
<p>The <code>tk.browse</code> function summons a TCL-TK interface to navigate through the whole genome, relying on <code>browsePlot</code> for the plotting.
</p>
<p>The former may be called directly to automatically export views from the genome browser, the latter is more suited to an interactive browsing with frequent coordinate jumps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  tk.browse(drawables = drawable.list(), blocking = FALSE, updateLimit = 0.4,
    png.height = NA, png.res = 100, png.file = tempfile(fileext=".png"),
    panelWidth = "5 cm", panel = NA)
  browsePlot(drawables, chrom = NA, start = NA, end = NA, customLayout = FALSE,
    xaxt = "s", xaxm = 1.5, panelWidth = "5 cm", panelSide = "left", panel = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tk.browse_+3A_drawables">drawables</code></td>
<td>

<p>A <code><a href="#topic+drawable.list-class">drawable.list</a></code> object containing the tracks to plot, as returned by <code><a href="#topic+drawable.list">drawable.list</a></code> or <code><a href="#topic+tk.tracks">tk.tracks</a></code>.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_blocking">blocking</code></td>
<td>

<p>Single logical value, whether to wait for the interface window to be closed before unfreezing the R console. The <code>FALSE</code> default let you use R and the interface in parallel (see the 'Typing R commands while browsing' section below), the codeTRUE value is used essentially in the stand alone version.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_updatelimit">updateLimit</code></td>
<td>

<p>Single numeric value, minimal time (in seconds) between two image updates when move or zoom key are continuously pressed. This is used to limit zoom and move speed on fast computers.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_png.height">png.height</code></td>
<td>

<p>Single integer value, the height of the display in pixels. Default value is to adapt the display to the size of the window, taller displays will require the user to use the scrollbar on the right of the display.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_png.res">png.res</code></td>
<td>

<p>Single integer value, the resolution of the plot in Pixels Per Inches. Passed to <code><a href="grDevices.html#topic+png">png</a></code>, see the corresponding manual for further details.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_png.file">png.file</code></td>
<td>

<p>Single character value, the path to the PNG file that is displayed in the main window. The default behavior is to hide it in a temporary location, however you can define this argument to have an easier access to the images displayed in Rgb (the image will be replaced each time Rgb refresh its display).
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_panelwidth">panelWidth</code></td>
<td>

<p>Single value, the width of the panel displays on the left of the tracks, if any is to be plotted. This is handled by the <code>width</code> argument of the <code>layout</code> function, and thus can be defined as a fixed size in centimeters or a relative size considering that the main display has a weight of 1.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_panelside">panelSide</code></td>
<td>

<p>Single character value (&quot;left&quot; or &quot;right&quot;), the side on which to plot the panels, if any. Note that <code>tk.browse</code> currently only implements left panels.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_panel">panel</code></td>
<td>

<p>Single logical value, wheter to force a panel to be displayed or to not be displayed. Default value of <code>NA</code> let the function decide according to the current track settings.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_chrom">chrom</code></td>
<td>

<p>Single character value, the chromosome to plot.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window to plot. <code>NA</code> will use 0.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window to plot. <code>NA</code> will use the maximal drawable$getChromEnd() value throughout <code>trackList</code>.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_customlayout">customLayout</code></td>
<td>

<p>Single logical value, whether to organize the various plot or not. If <code>FALSE</code>, <code><a href="graphics.html#topic+layout">layout</a></code> will be called and you will not be able to add custom plots. If <code>TRUE</code>, it is up to the user to define its <code><a href="graphics.html#topic+layout">layout</a></code> call to handle the track plots that will be issued by the function. If <code>NA</code>, no plotting is actually done but the layout that would be used is returned instead, to help building a custom one.
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_xaxt">xaxt</code></td>
<td>

<p>X axis showing (see <code>par</code>). Only applied to the last track, other ones never show their X axis (&quot;n&quot;).
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_xaxm">xaxm</code></td>
<td>

<p>Minimal bottom margin for the last track (see <code>par</code>). It assures enough margin is reserved for X axis showing (see <code>xaxt</code>).
</p>
</td></tr>
<tr><td><code id="tk.browse_+3A_...">...</code></td>
<td>

<p>Further arguments are passed through to the <code>draw</code> method of each track in <code>trackList</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>tk.browse</code> invisibly returns the <code>drawables</code> argument (See the 'Typing R commands while browsing' section above).
</p>
<p><code>browsePlot</code> invisibly returns the <code>par</code> function output for the last track plot, as it is used by <code>tk.browse</code>.
</p>


<h3>Interactive browsing</h3>



<h4>Jump to specific location</h4>

<p>The left upper panel can be used to jump to specific coordinates, defined by a chromosome name, a starting position and an ending position (as floating point numerics in millions of base pairs.</p>



<h4>Move along a chromosome</h4>

<p>The left and right arrow keys may be used to shift the window to the corresponding side. The page-up and page-down keys can be used to switch chromosome, without changing genomic numeric position.</p>



<h4>Generic zoom</h4>

<p>The up and down arrow keys, as well as the vertical mouse wheel, may be used to zoom in or out on the current location.</p>



<h4>Localized zoom in</h4>

<p>A zoom can also be achieved with a mouse drag on the region to investigate : maintain a left click on the position to use as the new left boundary, and release the click at the position of the new right boundary.</p>



<h4>Resize plot area</h4>

<p>The plot area size is defined by <code>hscale</code> and <code>vscale</code>. During interactive browsing, resize the browser window and use the &quot;r&quot; key to adjust the plot area to the window size.</p>



<h3>Typing R commands while browsing</h3>

<p><code>tk.browse</code> returns the <code><a href="#topic+drawable.list-class">drawable.list</a></code> objects it uses to store currently browsed data. As it is a <a href="methods.html#topic+setRefClass">reference class</a> object, the same memory location is shared by <code>tk.browse</code> and the returned object, so updates (like track addition or edition) made by <code>tk.browse</code> will impact the object in the R Command Line Interface (CLI), and updates made via R commands will impact the current <code>tk.browse</code> session.
</p>
<p>Some sub-interfaces (like information pop-ups and track selection panels) may freeze the R command prompt while opened, make sure to have only the <code>tk.browse</code> main window opened when typing R commands.
</p>
<p>Notice some operating systems (including Windows) restrain users to type R commands while a tcl-tk window is opened, or seems to be instable while doing so. Setting <code>blocking</code> to <code>TRUE</code> will enforce this behavior, keeping users from typing commands while <code>tk.browse</code> is running.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+singlePlot">singlePlot</a></code>, <code><a href="#topic+drawable.list-class">drawable.list</a></code>, <code><a href="#topic+drawable-class">drawable</a></code>
</p>

<hr>
<h2 id='tk.convert'>
RDT file conversion
</h2><span id='topic+tk.convert'></span>

<h3>Description</h3>

<p>This function provides a tcl-tk interface to convert RDT table files into CSV-like files, and to produce basic <code><a href="#topic+track.table-class">track.table</a></code> objects from such files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  tk.convert(blocking = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tk.convert_+3A_blocking">blocking</code></td>
<td>

<p>Single logical value, whether to wait for the interface window to be closed before unfreezing the R console. The <code>FALSE</code> default let you use R and the interface in parallel, the codeTRUE is used essentially in the stand alone version.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table-class">track.table-class</a></code>
</p>

<hr>
<h2 id='tk.tracks'>
Track management for Genome browser
</h2><span id='topic+tk.tracks'></span>

<h3>Description</h3>

<p>This function allows to load and edit the <code><a href="#topic+drawable.list-class">drawable.list</a></code> object that is to be passed to <code><a href="#topic+tk.browse">tk.browse</a></code> and code<a href="#topic+browsePlot">browsePlot</a>, using a Tcl-tk interface
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  tk.tracks(drawables = drawable.list(), parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tk.tracks_+3A_drawables">drawables</code></td>
<td>

<p>A previously built <code><a href="#topic+drawable.list-class">drawable.list</a></code> to edit. Default value builds an empty one.
</p>
</td></tr>
<tr><td><code id="tk.tracks_+3A_parent">parent</code></td>
<td>

<p>An optional <code><a href="tcltk.html#topic+tktoplevel">tktoplevel</a></code> to bind message boxes to. Impact on the Tcl-tk interface behaviour is slight, so only advanced users should take care of it.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+drawable.list-class">drawable.list</a></code> object. Notice that if 'drawables' was provided, it will also be updated &quot;in-place&quot; (standard reference class behavior).
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tk.browse">tk.browse</a></code>, <code><a href="#topic+browsePlot">browsePlot</a></code>, <code><a href="#topic+drawable.list-class">drawable.list-class</a></code>, <code><a href="#topic+drawable-class">drawable-class</a></code>, <code><a href="#topic+findDrawables">findDrawables</a></code>
</p>

<hr>
<h2 id='tk.utils'>
Tk interface utilities
</h2><span id='topic+tk.file'></span><span id='topic+tk.file'></span><span id='topic+tk.files'></span><span id='topic+tk.folder'></span><span id='topic+handle'></span>

<h3>Description</h3>

<p>These functions are components used to build interactive interfaces.
</p>
<p><code>tk.file</code> is a wrapper to tcltk functions <code>tkgetOpenFile</code> and <code>tkgetOpenFile</code> with several enhancements.
</p>
<p><code>tk.files</code> proposes to build and order a file list from multiple calls to <code>tk.file</code>.
</p>
<p><code>tk.folder</code> is a wrapper to tcltk function <code>tkchooseDirectory</code> with small enhancements.
</p>
<p><code>handle</code> is a wrapper to <code>withRestarts</code>, which allows to catch errors, warnings and messages while executing an R expression and handle them with custom functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  tk.file(title = "Choose a file", typeNames = "All files", typeExt = "*",
    multiple = FALSE, mandatory = TRUE, type = c("open", "save"), initialdir = NULL,
    parent = NULL)
  tk.files(preselection = character(0), multiple = TRUE, parent = NULL, ...)
  tk.folder(title = "Choose a directory", mustexist = TRUE, mandatory = TRUE)
  handle(expr, messageHandler, warningHandler, errorHandler)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tk.utils_+3A_title">title</code></td>
<td>

<p>Single character value, the displayed name of the summoned window.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_typenames">typeNames</code></td>
<td>

<p>Character vector defining the displayed names of the filtered file extensions. Parallel to <code>typeExt</code>.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_typeext">typeExt</code></td>
<td>

<p>Character vector defining the filtered file extensions (use &quot;*&quot; as wildcard). Parallel to <code>typeNames</code>.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_multiple">multiple</code></td>
<td>

<p>Single logical value, whether to allow multiple file selection or not.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_mandatory">mandatory</code></td>
<td>

<p>Single logical value, whether to throw an error if no file is selected or not.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_type">type</code></td>
<td>

<p>Single character value defining the label of the button.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_initialdir">initialdir</code></td>
<td>

<p>Single character value, the initially selected directory when the window is summoned. <code>NULL</code> corresponds to the last directory browsed via one of these functions.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_parent">parent</code></td>
<td>

<p>A Tcl-Tk object to consider as the parent of the new frame.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_preselection">preselection</code></td>
<td>

<p>Character vector, the files already selected when the window is summoned.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code>tk.file</code>.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_mustexist">mustexist</code></td>
<td>

<p>Single logical value, whether to throw an error if a non-existing directory is selected or not.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_expr">expr</code></td>
<td>

<p>An R expression in which errors, warnings and messages are to be caught. Use <code>{}</code> to monitor a multi-line R code.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_messagehandler">messageHandler</code></td>
<td>

<p>A function taking as single argument the <a href="base.html#topic+conditions">condition</a> object caught. If missing, <a href="base.html#topic+message">messages</a> will pass through.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_warninghandler">warningHandler</code></td>
<td>

<p>A function taking as single argument the <a href="base.html#topic+conditions">condition</a> object caught. If missing, <a href="base.html#topic+warning">warnings</a> will pass through.
</p>
</td></tr>
<tr><td><code id="tk.utils_+3A_errorhandler">errorHandler</code></td>
<td>

<p>A function taking as single argument the <a href="base.html#topic+conditions">condition</a> object caught. If missing, <a href="base.html#topic+stop">errors</a> will pass through.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>tk.file</code>, <code>tk.files</code> and <code>tk.folder</code> return the selection as a character vector, possibly empty.
</p>
<p><code>handle</code> returns nothing.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tk.browse">tk.browse</a></code>, <code><a href="#topic+tk.convert">tk.convert</a></code>, <code><a href="#topic+tk.tracks">tk.tracks</a></code>
</p>

<hr>
<h2 id='track-constructors'>
Track constructors
</h2><span id='topic+track-constructors'></span><span id='topic+track.table'></span><span id='topic+track.bam'></span><span id='topic+track.genes'></span><span id='topic+track.bands'></span><span id='topic+track.exons'></span><span id='topic+track.CNV'></span>

<h3>Description</h3>

<p>Produces <code><a href="#topic+track.table-class">track.table</a></code>-inheriting objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  track.table(..., .name, .parameters, .organism, .assembly, .chromosomes,
    .makeNames = FALSE, .orderCols = TRUE, warn = TRUE)
  track.bam(bamPath, baiPath, addChr, quiet = FALSE, .name, .organism,
    .assembly, .parameters, warn = TRUE)
  track.genes(...)
  track.bands(...)
  track.exons(...)
  track.CNV(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track-constructors_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to the inherited constructor (<code><a href="#topic+refTable">refTable</a></code> for <code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+track.table">track.table</a></code> for the others). <code>track.table</code> has few more restrictions, see <code><a href="#topic+track.table-class">track.table-class</a></code> for further details.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.name">.name</code></td>
<td>

<p>Single character value, to fill the <code>name</code> field inherited from <code><a href="#topic+drawable-class">drawable</a></code>.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_warn">warn</code></td>
<td>

<p>Single logical value, to be passed to the appropriate <code>check</code> method. Other conversion warnings rely on this value too in <code>track.table</code>.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.parameters">.parameters</code></td>
<td>

<p>A <code>list</code> of drawing parameters, to fill the <code>parameters</code> field of the object.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.organism">.organism</code></td>
<td>

<p>Single character value, to fill the <code>organism</code> field of the object.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.assembly">.assembly</code></td>
<td>

<p>Single character value, to fill the <code>assembly</code> field of the object.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.chromosomes">.chromosomes</code></td>
<td>

<p>Single character value, levels to use for the 'chrom' column if conversion to factor is needed.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.makenames">.makeNames</code></td>
<td>

<p>Single logical value, whether to compute the 'name' column with unique values or not. If <code>TRUE</code>, any existing 'name' column will be replaced.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_.ordercols">.orderCols</code></td>
<td>

<p>Single logical value, whether to reorder the columns for more consistency between tracks or not.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_bampath">bamPath</code></td>
<td>

<p>Single character value, the file name and path to a BAM file (.bam) to build a track around.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_baipath">baiPath</code></td>
<td>

<p>Single character value, the file name and path to the corresponding BAI file (.bai) to build a track around. If missing, a guess will be tried (adding '.bai' to <code>bamPath</code> or remplacing '.bam' by '.bai').
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_addchr">addChr</code></td>
<td>

<p>Single logical value, whether to automatically add 'chr' ahead chromosome names when querying or not. If missing, a guess will be tried looking for chromosome names beginning by 'chr' in the BAM header declaration.
</p>
</td></tr>
<tr><td><code id="track-constructors_+3A_quiet">quiet</code></td>
<td>

<p>Single logical value, whether to throw diagnostic messages during BAI parsing or not.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+track.table-class">track.table</a></code> and <code><a href="#topic+track.bam-class">track.bam</a></code> inheriting objects.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table-class">track.table-class</a></code>, <code><a href="#topic+track.bam-class">track.bam-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # track.table from a data.frame
  df &lt;- data.frame(
    chrom=1, strand="+", start=1:5, end=2:6, name=letters[1:5],
    stringsAsFactors=FALSE
  )
  track.table(df)
  
  # track.table from vectors
  track.table(chrom=1, strand="+", start=1:5, end=2:6, name=letters[1:5])
  
  # track.bam
  track.bam(system.file("extdata/ATM.bam", package="Rgb"))
</code></pre>

<hr>
<h2 id='track.bam-class'>Class <code>"track.bam"</code></h2><span id='topic+track.bam-class'></span>

<h3>Description</h3>

<p><code>"track.bam"</code> is a drawing wraper for Binary Alignment Map files (SAMtools).
</p>
<p>Notice the data are not stored directly in the object, but stay in the original BAM file, thus exported <code>track.bam</code> objects may be broken (the <code>check</code> method can confirm this).
</p>
<p>Objects are produced by the <code><a href="#topic+track.bam">track.bam</a></code> constructor.
</p>


<h3>Extends</h3>

<p>Class <code><a href="#topic+sliceable-class">sliceable</a></code>, directly.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+sliceable-class">sliceable</a></code>, distance 2.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>


<dl>
<dt><code>addChr</code>:</dt><dd><p>Single <code>logical</code> value, whether to automatically add 'chr' ahead chromosome names when querying or not..</p>
</dd>
<dt><code>assembly</code>:</dt><dd><p>Single <code>character</code> value, the assembly version for the coordinates stored in the object. Must have length 1, should not be <code>NA</code>.</p>
</dd>
<dt><code>baiPath</code>:</dt><dd><p>Single <code>character</code> value, the full path to the BAI index file in use.</p>
</dd>
<dt><code>bamPath</code>:</dt><dd><p>Single <code>character</code> value, the full path to the BAM file in use.</p>
</dd>
<dt><code>compression</code>:</dt><dd><p>Single <code>numeric</code> value, an estimation of the BAM file compression ratio.</p>
</dd>
<dt><code>header</code>:</dt><dd><p>A <code>data.frame</code> describing the @SQ elements of the BAM header (one per row).</p>
</dd>
<dt><code>index</code>:</dt><dd><p>The parsed content of the BAI index, as a unamed <code>list</code> with one element by reference sequence, itself a <code>list</code> with 'bins' and 'intervals' elements. 'bins' is a named <code>list</code> of two-column matrices ('start' and 'end'), giving virtual BGZF coordinates of the described bin (as <code>double</code>). 'intervals' is a <code>double</code> vector of virtual BGZF coordinates, used for linear filtering (see SAM specification for further details).</p>
</dd>
<dt><code>organism</code>:</dt><dd><p>Single <code>character</code> value, the name of the organism whose data is stored in the object. Must have length 1, should not be <code>NA</code>.</p>
</dd>
</dl>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li></ul>



<h3>Methods</h3>


<dl>
<dt><code>coverage(chrom, start = , end = , tracks = , binLevel = , rawSize = )</code>:</dt><dd>
<p>Fast estimation of depth coverage in a genomic window, from indexing data. Values are normalized into [0:1] over the genomic window.<br />
- <strong>chrom</strong>      : single integer, numeric or character value, the chromosomal location.<br />
- <strong>start</strong>      : single integer or numeric value, inferior boundary of the window. If NA, the whole chromosome is considered.<br />
- <strong>end</strong>        : single integer or numeric value, superior boundary of the window. If NA, the whole chromosome is considered.<br />
- <strong>tracks</strong>     : single logical value, whether to return a data.frame or a track.table.<br />
- <strong>binLevel</strong>   : single integer value, the higher bin order to allow<br />
0 = 537Mb, 1 = 67Mb, 2 = 8Mb, 3 = 1Mb, 4 = 130kb, 5 = 16kb<br />
incrementing this value enhances boundary precision but discards reads located at bin junctions.<br />
- <strong>rawSize</strong>    : single logical value, whether to output raw size or normalize by the maximum encountered.
</p>
</dd>
<dt><code>crawl(chrom, start, end, addChr = , maxRange = , maxRangeWarn = , verbosity = , ..., init, loop, final)</code>:</dt><dd>
<p>Apply a custom processing to reads in a genomic window (used by 'depth', 'extract' and 'pileup' methods).<br />
- <strong>chrom</strong>          : single integer, numeric or character value, the chromosomal location. NA is not handled.<br />
- <strong>start</strong>          : single integer or numeric value, inferior boundary of the window. NA is not handled.<br />
- <strong>end</strong>            : single integer or numeric value, superior boundary of the window. NA is not handled.<br />
- <strong>addChr</strong>         : single logical value, whether to systematically add 'chr' in front of the 'chrom' value or not.<br />
- <strong>maxRange</strong>       : single integer value, no extraction will be attempted if end and start are more than this value away (returns NULL).<br />
- <strong>maxRangeWarn</strong>   : single logical value, whether to throw a warning when 'maxRange' is exceeded and NULL is returned or not.<br />
- <strong>verbosity</strong>      : single integer value, the level of verbosity during processing (0, 1 or 2).<br />
- <strong>...</strong>            : arguments to be passed to 'init', 'loop' or 'final'.<br />
- <strong>init</strong>           : a function taking a single storage environment as argument, to be evaluated before looping on reads for initialization.<br />
This environment has R 'base' environment as parent and contains :<br />
* all arguments passed to crawl()<br />
* a 'self' reference to the current object.<br />
* 'earlyBreak', a single logical value forcing crawl() to return immediately if set to TRUE.<br />
* 'output', a place-holder for the variable to be returned by crawl().<br />
* 'totalReads', the number of matching reads seen since the beginning of the whole looping process.<br />
* 'blockReads', the number of matching reads seen since the beginning of the current BGZF block.<br />
The 'init', 'loop' and 'final' functions defined by the user can freely store additionnal variables in this environment to share them.                   <br />
- <strong>loop</strong>           : a function taking a list-shapped read and the storage environment, to be evaluated for each read with matching coordinates.<br />
- <strong>final</strong>          : a function taking the storage environment as argument, to be evaluated once all reads were processed for finalization.
</p>
</dd>
<dt><code>depth(..., qBase = , qMap = )</code>:</dt><dd>
<p>Counts covering bases for each genomic position, similarly to SAMtools' depth.<br />
- <strong>...</strong>     : arguments to be passed to the crawl() method.<br />
- <strong>qBase</strong>   : single integer value, minimal base quality for a base to be counted.<br />
- <strong>qMap</strong>    : single integer value, minimal mapping quality for a base to be counted.
</p>
</dd>
<dt><code>extract(...)</code>:</dt><dd>
<p>Extract reads as a list, similarly to SAMtools' view.<br />
- <strong>...</strong>     : arguments to be passed to the crawl() method.
</p>
</dd>
<dt><code>getBlocks(limit = , quiet = )</code>:</dt><dd>
<p>Jump from BGZF blocks to blocks, recording compressed (bsize) and uncompressed (isize) block sizes<br />
- <strong>limit</strong>   : single integer value, the amount of blocks to evaluate (NA for the whole BAM file, may be very time consuming).<br />
- <strong>quiet</strong>   : single logical value, whether to throw diagnostic messages or not.
</p>
</dd>
<dt><code>getCompression(sample = )</code>:</dt><dd>
<p>Estimate BGZF block compression level from a sample of blocks<br />
- <strong>sample</strong>   : single integer value, the amount of blocks to use for estimation (the first block is ignored).
</p>
</dd>
<dt><code>pileup(..., qBase = , qMap = )</code>:</dt><dd>
<p>Counts each nucleotide type for each genomic position, similarly to SAMtools' mpileup.<br />
- <strong>...</strong>     : arguments to be passed to the crawl() method.<br />
- <strong>qBase</strong>   : single integer value, minimal base quality for a base to be counted.<br />
- <strong>qMap</strong>    : single integer value, minimal mapping quality for a base to be counted.
</p>
</dd>
<dt><code>summary(chrom = , tracks = , binLevel = , rawSize = )</code>:</dt><dd>
<p>Fast estimation of depth coverage for the whole genome, from indexing data. Values are normalized into [0:1] over the whole genome.<br />
- <strong>chrom</strong>      : character vector, the names of the chromosome to query. If NA, all chromosomes will be queried.<br />
- <strong>tracks</strong>     : single logical value, whether to return a data.frame or a track.table.<br />
- <strong>binLevel</strong>   : single integer value, the higher bin order to allow<br />
0 = 537Mb, 1 = 67Mb, 2 = 8Mb, 3 = 1Mb, 4 = 130kb, 5 = 16kb<br />
incrementing this value enhances boundary precision but discards reads located at bin junctions<br />
- <strong>rawSize</strong>    : single logical value, whether to output raw size or normalize by the maximum encountered.
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> chromosomes (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> show (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> slice (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p><a href="http://samtools.github.io/hts-specs/SAMv1.pdf">http://samtools.github.io/hts-specs/SAMv1.pdf</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+sliceable-class">sliceable-class</a></code>, <code><a href="#topic+drawable-class">drawable-class</a></code>
</p>

<hr>
<h2 id='track.bands-class'>Class <code>"track.bands"</code></h2><span id='topic+track.bands-class'></span>

<h3>Description</h3>

<p>This class is a variation of the <code><a href="#topic+track.table-class">track.table</a></code> class dedicated to cytogenetic banding, enforcing new drawing parameter defaults and a few specialized methods.
</p>
<p>Objects can be created by two distincts means :
</p>

<ul>
<li><p>Using the corresponding constructors, which work like the <code><a href="#topic+track.table">track.table</a></code> constructor.
</p>
</li>
<li><p>Importing a <code><a href="#topic+track.table-class">track.table</a></code> object in an empty object created by a call to <code>new</code>.
</p>
</li></ul>



<h3>Extends</h3>

<p>Class <code><a href="#topic+track.table-class">track.table</a></code>, directly.<br />
Class <code><a href="#topic+refTable-class">refTable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+crossable-class">crossable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+sliceable-class">sliceable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 3.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 4.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> assembly (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> checktrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colIterator (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colReferences (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> index (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> organism (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> rowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNamed (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> sizetrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> subtrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> values (<a href="#topic+refTable-class">refTable</a>)
</p>
</li></ul>



<h3>Methods</h3>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> addArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addColumn (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addDataFrame (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addEmptyRows (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addList (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addVectors (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> buildCalls (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupPosition (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupSize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildIndex (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> chromosomes (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> coerce (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colOrder (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> copy (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> cross (<a href="#topic+crossable-class">crossable</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> delColumns (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> erase (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> eraseArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> extract (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fill (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getColCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getColNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getLevels (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> indexes (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> isArmed (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> metaFields (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> rowOrder (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segMerge (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segOverlap (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setColNames (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setLevels (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> show (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> size (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> slice (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> types (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table-class">track.table-class</a></code>, <code><a href="#topic+track.table">track.table</a></code>
</p>
<p><code><a href="#topic+track.bands">track.bands</a></code>, <code><a href="#topic+track.bands.UCSC">track.bands.UCSC</a></code>
</p>

<hr>
<h2 id='track.CNV-class'>Class <code>"track.CNV"</code></h2><span id='topic+track.CNV-class'></span>

<h3>Description</h3>

<p>This class is a variation of the <code><a href="#topic+track.table-class">track.table</a></code> class dedicated to constitutive Copy Number Variations, enforcing new drawing parameter defaults and a few specialized methods.
</p>
<p>Objects can be created by two distincts means :
</p>

<ul>
<li><p>Using the corresponding constructors, which work like the <code><a href="#topic+track.table">track.table</a></code> constructor.
</p>
</li>
<li><p>Importing a <code><a href="#topic+track.table-class">track.table</a></code> object in an empty object created by a call to <code>new</code>.
</p>
</li></ul>



<h3>Extends</h3>

<p>Class <code><a href="#topic+track.table-class">track.table</a></code>, directly.<br />
Class <code><a href="#topic+refTable-class">refTable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+crossable-class">crossable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+sliceable-class">sliceable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 3.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 4.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> assembly (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> checktrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colIterator (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colReferences (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> index (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> organism (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> rowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNamed (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> sizetrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> subtrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> values (<a href="#topic+refTable-class">refTable</a>)
</p>
</li></ul>



<h3>Methods</h3>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> addArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addColumn (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addDataFrame (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addEmptyRows (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addList (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addVectors (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> buildCalls (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupPosition (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupSize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildIndex (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> chromosomes (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> coerce (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colOrder (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> copy (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> cross (<a href="#topic+crossable-class">crossable</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> delColumns (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> erase (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> eraseArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> extract (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fill (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getColCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getColNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getLevels (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> indexes (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> isArmed (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> metaFields (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> rowOrder (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segMerge (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segOverlap (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setColNames (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setLevels (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> show (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> size (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> slice (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> types (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table-class">track.table-class</a></code>, <code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+track.table.GTF">track.table.GTF</a></code>
</p>
<p><code><a href="#topic+track.CNV">track.CNV</a></code>, <code><a href="#topic+track.CNV.DGV">track.CNV.DGV</a></code>
</p>

<hr>
<h2 id='track.exons-class'>Class <code>"track.exons"</code></h2><span id='topic+track.exons-class'></span>

<h3>Description</h3>

<p>This class is a variation of the <code><a href="#topic+track.table-class">track.table</a></code> class dedicated to exons, enforcing new drawing parameter defaults and a few specialized methods.
</p>
<p>Objects can be created by two distincts means :
</p>

<ul>
<li><p>Using the corresponding constructors, which work like the <code><a href="#topic+track.table">track.table</a></code> constructor.
</p>
</li>
<li><p>Importing a <code><a href="#topic+track.table-class">track.table</a></code> object in an empty object created by a call to new (see Examples).
</p>
</li></ul>



<h3>Extends</h3>

<p>Class <code><a href="#topic+track.table-class">track.table</a></code>, directly.<br />
Class <code><a href="#topic+refTable-class">refTable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+crossable-class">crossable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+sliceable-class">sliceable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 3.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 4.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> assembly (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> checktrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colIterator (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colReferences (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> index (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> organism (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> rowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNamed (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> sizetrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> subtrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> values (<a href="#topic+refTable-class">refTable</a>)
</p>
</li></ul>



<h3>Methods</h3>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> addArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addColumn (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addDataFrame (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addEmptyRows (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addList (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addVectors (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> buildCalls (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupPosition (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupSize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildIndex (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> chromosomes (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> coerce (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colOrder (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> copy (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> cross (<a href="#topic+crossable-class">crossable</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> delColumns (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> erase (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> eraseArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> extract (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fill (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getColCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getColNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getLevels (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> indexes (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> isArmed (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> metaFields (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> rowOrder (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segMerge (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segOverlap (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setColNames (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setLevels (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> show (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> size (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> slice (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> types (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table-class">track.table-class</a></code>, <code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+track.table.GTF">track.table.GTF</a></code>
</p>
<p><code><a href="#topic+track.exons">track.exons</a></code>, <code><a href="#topic+track.exons.CCDS">track.exons.CCDS</a></code>
</p>

<hr>
<h2 id='track.fasta-class'>Class <code>"track.fasta"</code></h2><span id='topic+track.fasta-class'></span>

<h3>Description</h3>

<p><code>"track.fasta"</code> is a drawing wraper for FASTA files.
</p>
<p>Notice the data are not stored directly in the object, but stay in the original FASTA file(s), thus exported <code>track.fasta</code> objects may be broken (the <code>check</code> method can confirm this).
</p>
<p>Objects are produced by the <code><a href="#topic+track.fasta.multi">track.fasta.multi</a></code> and <code><a href="#topic+track.fasta.collection">track.fasta.collection</a></code> constructors.
</p>


<h3>Extends</h3>

<p>Class <code><a href="#topic+sliceable-class">sliceable</a></code>, directly.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+sliceable-class">sliceable</a></code>, distance 2.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>


<dl>
<dt><code>assembly</code>:</dt><dd><p>Single <code>character</code> value, the assembly version for the coordinates stored in the object. Must have length 1, should not be <code>NA</code>.</p>
</dd>
<dt><code>files</code>:</dt><dd><p>A <code>data.frame</code>, with 6 columns : file (character), header (character), startOffset (numeric), lineLength (integer), breakSize (integer) and contentSize (integer). Each row refers to a distinct chromosome, whose name is stored as row name.</p>
</dd>
<dt><code>organism</code>:</dt><dd><p>Single <code>character</code> value, the name of the organism whose data is stored in the object. Must have length 1, should not be <code>NA</code>.</p>
</dd>
</dl>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li></ul>



<h3>Methods</h3>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> chromosomes (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> copy (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> show (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> slice (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+sliceable-class">sliceable-class</a></code>, <code><a href="#topic+drawable-class">drawable-class</a></code>
</p>

<hr>
<h2 id='track.fasta-constructors'>
Track constructors for FASTA files
</h2><span id='topic+track.fasta-constructors'></span><span id='topic+track.fasta.multi'></span><span id='topic+track.fasta.collection'></span>

<h3>Description</h3>

<p>Produces <code><a href="#topic+sliceable-class">sliceable</a></code>-inheriting objects to query &quot;in situ&quot; FASTA files.
</p>
<p><code>track.fasta.multi</code> is designed to handle a single multi-FASTA file aggregating all the chromosomes of an organism. An index as generated by the HTSlib (formerly &quot;SAMtools&quot;) <code>faidx</code> command is required.
</p>
<p><code>track.fasta.collection</code> is designed to handle a collection of standard FASTA files, one per chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  track.fasta.multi(fastaFile, indexFile, .name, .organism, .assembly,
    .parameters, warn = TRUE)
  track.fasta.collection(files, chromosomes, .name, .organism, .assembly,
    .parameters, warn = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track.fasta-constructors_+3A_fastafile">fastaFile</code></td>
<td>

<p>Single character value, the file name and path to a multi-FASTA file (.fa) to build a track upon.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_indexfile">indexFile</code></td>
<td>

<p>Single character value, the file name and path to a multi-FASTA index file (.fai) corresponding to <code>fastaFile</code>. This index file is supposd to be produced by the <code>faidx</code> command from SAMtools.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_.name">.name</code></td>
<td>

<p>Single character value, to fill the <code>name</code> field inherited from <code><a href="#topic+drawable-class">drawable</a></code>.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_.organism">.organism</code></td>
<td>

<p>Single character value, to fill the <code>organism</code> field of the object.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_.assembly">.assembly</code></td>
<td>

<p>Single character value, to fill the <code>assembly</code> field of the object.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_.parameters">.parameters</code></td>
<td>

<p>A <code>list</code> of drawing parameters, to fill the <code>parameters</code> field of the object.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_warn">warn</code></td>
<td>

<p>Single logical value, to be passed to the appropriate <code>check</code> method. Other conversion warnings rely on this value too in <code>track.table</code>.
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_files">files</code></td>
<td>

<p>Character vector, file names and paths to multiple single-FASTA file (.fa) to build a track upon (one per chromosome).
</p>
</td></tr>
<tr><td><code id="track.fasta-constructors_+3A_chromosomes">chromosomes</code></td>
<td>

<p>Character vector, chromosomes names to attribute to each file in <code>files</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both functions suppose the FASTA files to respect the following :
</p>

<ul>
<li><p>They begin with a single line of comment, after a '&gt;' sign.
</p>
</li>
<li><p>All sequence lines have the same length (whatever it is).
</p>
</li>
<li><p>The line separator (\n, \r\n) is always the same in a file.
</p>
</li>
<li><p>No empty line.
</p>
</li></ul>

<p>Returned sequences may be wrong (without error or notice !) if any of these is not fullfilled. Standard sources (see below) usually enforce these conditions.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+track.fasta-class">track.fasta</a></code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Example of FASTA file collection (human assembly 'hg19'), from UCSC : <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz">http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz</a>
</p>
<p>Example of single multi-FASTA file (human assembly 'hg19'), from 1000 genomes : <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz</a>
</p>
<p>The <a href="http://www.htslib.org/doc/faidx.html">faidx</a> documentation, from the HTSlib project.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track-constructors">track-constructors</a></code>, <code><a href="#topic+Annotation">Annotation</a></code>
</p>
<p><code><a href="#topic+tk.browse">tk.browse</a></code>, <code><a href="#topic+browsePlot">browsePlot</a></code>
</p>

<hr>
<h2 id='track.genes-class'>Class <code>"track.genes"</code></h2><span id='topic+track.genes-class'></span>

<h3>Description</h3>

<p>This class is a variation of the <code><a href="#topic+track.table-class">track.table</a></code> class dedicated to genes, enforcing new drawing parameter defaults and a few specialized methods.
</p>
<p>Objects can be created by two distincts means :
</p>

<ul>
<li><p>Using the corresponding constructors, which work like the <code><a href="#topic+track.table">track.table</a></code> constructor.
</p>
</li>
<li><p>Importing a <code><a href="#topic+track.table-class">track.table</a></code> object in an empty object created by a call to new (see Examples).
</p>
</li></ul>



<h3>Extends</h3>

<p>Class <code><a href="#topic+track.table-class">track.table</a></code>, directly.<br />
Class <code><a href="#topic+refTable-class">refTable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+crossable-class">crossable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 2.<br />
Class <code><a href="#topic+sliceable-class">sliceable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 3.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+track.table-class">track.table</a></code>, distance 4.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> assembly (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> checktrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colIterator (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colReferences (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> index (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> organism (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> rowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNamed (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> sizetrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> subtrack (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> values (<a href="#topic+refTable-class">refTable</a>)
</p>
</li></ul>



<h3>Methods</h3>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> addArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addColumn (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addDataFrame (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addEmptyRows (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addList (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> addVectors (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> buildCalls (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupPosition (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildGroupSize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> buildIndex (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> chromosomes (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> coerce (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> colOrder (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> copy (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> cross (<a href="#topic+crossable-class">crossable</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> delColumns (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> erase (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> eraseArms (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> extract (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fill (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getColCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getColNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getLevels (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> indexes (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> isArmed (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> metaFields (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> rowOrder (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segMerge (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> segOverlap (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setColNames (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setLevels (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> show (<a href="#topic+track.table-class">track.table</a>, overloaded)
</p>
</li>
<li><p> size (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> slice (<a href="#topic+track.table-class">track.table</a>)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> types (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track.table-class">track.table-class</a></code>, <code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+track.table.GTF">track.table.GTF</a></code>
</p>
<p><code><a href="#topic+track.genes">track.genes</a></code>, <code><a href="#topic+track.genes.NCBI">track.genes.NCBI</a></code>
</p>

<hr>
<h2 id='track.table-class'>Class <code>"track.table"</code></h2><span id='topic+track.table-class'></span>

<h3>Description</h3>

<p><code>"track.table"</code> describes a collection of features localized on a genome, defined by a chromosomal location (&quot;chrom&quot;), two boundaries (&quot;start&quot; and &quot;end&quot;) and various data (other columns).
</p>
<p>Objects can be created by two distincts means :
</p>

<ul>
<li><p>The <code><a href="#topic+track.table">track.table</a></code> constructor, similar to the <code>data.frame</code> constructor. It imports a single <code>data.frame</code> or a collection of vectors into the object, reorder it to meet the <code>track.table</code> class restrictions (see below) and check immediatly for validity.
</p>
</li>
<li><p>The <code>new</code> function (standard behavior for S4 and reference classes), which produces an empty object and do NOT check for validity. It takes as named arguments the values to store in the fields of the various classes the new object will inherit from.
</p>
</li></ul>

<p>3 columns are mandatory in <code>"track.table"</code>, with restricted names and types :
</p>

<dl>
<dt>name</dt><dd><p>The unique name of the feature, as <code>character</code>.</p>
</dd>
<dt>chrom</dt><dd><p>The chromosomal location of the feature, as <code>integer</code> or <code>factor</code>.</p>
</dd>
<dt>start</dt><dd><p>The starting position of the feature on the chromosome, as <code>integer</code>.</p>
</dd>
<dt>end</dt><dd><p>The ending position of the feature on the chromosome, as <code>integer</code>.</p>
</dd>
</dl>

<p>The track is supposed to be ordered by chromosome, then by starting position. When chromosomes are stored as <code>factors</code>, they need to be numerically ordered by their internal codes (as the <code>order</code> function does), not alphabetically by their labels.
</p>
<p>As the <code>slice</code> method relies on a row index, each update in the feature coordinates must be followed by a call to the <code>buildIndex</code> method, a behavior that is enforced by overloading most of <code><a href="#topic+refTable-class">refTable</a></code>-inherited methods.
</p>
<p>As the <code>slice</code> method relies on a R call object using <code><a href="#topic+refTable-class">refTable</a></code> column references, each update in the column names must be followed by a call to the <code>buildCalls</code> method, a behavior that is enforced by overloading most of <code><a href="#topic+refTable-class">refTable</a></code>-inherited methods. 
</p>


<h3>Extends</h3>

<p>Class <code><a href="#topic+refTable-class">refTable</a></code>, directly.<br />
Class <code><a href="#topic+crossable-class">crossable</a></code>, directly.<br />
Class <code><a href="#topic+sliceable-class">sliceable</a></code>, by class <code><a href="#topic+crossable-class">crossable</a></code>, distance 2.<br />
Class <code><a href="#topic+drawable-class">drawable</a></code>, by class <code><a href="#topic+crossable-class">crossable</a></code>, distance 3.
</p>
<p>All reference classes extend and inherit methods from <code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>.
</p>


<h3>Fields</h3>


<dl>
<dt><code>assembly</code>:</dt><dd><p>Single <code>character</code> value, the assembly version for the coordinates stored in the object. Must have length 1, should not be <code>NA</code>.</p>
</dd>
<dt><code>checktrack</code>:</dt><dd><p>A <code>call</code> to the C external &quot;checktrack&quot;, for faster object check (see the <code>check</code> method).</p>
</dd>
<dt><code>index</code>:</dt><dd><p><code>integer</code> vector, giving the index of the first row in each chromosome. See the <code><a href="#topic+subtrack">subtrack</a></code> function for further details.</p>
</dd>
<dt><code>organism</code>:</dt><dd><p>Single <code>character</code> value, the name of the organism whose data is stored in the object. Must have length 1, should not be <code>NA</code>.</p>
</dd>
<dt><code>sizetrack</code>:</dt><dd><p>A <code>call</code> to the C external &quot;track&quot;, for faster row counting (see the <code>size</code> method).</p>
</dd>
<dt><code>subtrack</code>:</dt><dd><p>A <code>call</code> to the C external &quot;track&quot;, for faster slicing (see the <code>slice</code> method).</p>
</dd>
</dl>

<p>The following fields are inherited (from the corresponding class):
</p>

<ul>
<li><p> colCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colIterator (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> colReferences (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> name (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> parameters (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> rowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNamed (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> rowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> values (<a href="#topic+refTable-class">refTable</a>)
</p>
</li></ul>



<h3>Methods</h3>


<dl>
<dt><code>addArms(centromeres, temp = )</code>:</dt><dd>
<p>Adds an arm localization ('p' or 'q') to the 'chrom' column.<br />
- <strong>centromeres</strong>   : named numeric vector, providing the centromere position of each chromosome. Can also be a band track, as returned by track.UCSC_bands().<br />
- <strong>temp</strong>          : single logical value, whether to alter the object or return an altered copy.
</p>
</dd>
<dt><code>buildCalls()</code>:</dt><dd>
<p>Updates 'checktrack' and 'subtrack' fields. To be performed after each modification of colNames and colReferences (concerned methods are overloaded to enforce this).
</p>
</dd>
<dt><code>buildGroupPosition(groupBy, colName = , reverse = )</code>:</dt><dd>
<p>Adds a column to be used as 'groupPosition' by draw.boxes()<br />
- <strong>groupBy</strong>   : single character value, the name of a column to group rows on.<br />
- <strong>colName</strong>   : single character value, the name of the column to buid.<br />
- <strong>reverse</strong>   : single logical value, whether to reverse numbering on reverse strand or not.
</p>
</dd>
<dt><code>buildGroupSize(groupBy, colName = )</code>:</dt><dd>
<p>Adds a column to be used as 'groupSize' by draw.boxes()<br />
- <strong>groupBy</strong>   : single character value, the name of a column to group rows on.<br />
- <strong>colName</strong>   : single character value, the name of the column to buid.
</p>
</dd>
<dt><code>buildIndex()</code>:</dt><dd>
<p>Updates the 'index' parameter, should be done after any change made on the 'chrom' column (concerned methods are overloaded to enforce this).
</p>
</dd>
<dt><code>eraseArms(temp = )</code>:</dt><dd>
<p>Removes 'p' and 'q' added by the addArms() method from the 'chrom' column.<br />
- <strong>temp</strong>   : single logical value, whether to alter the object or return an altered copy.
</p>
</dd>
<dt><code>isArmed()</code>:</dt><dd>
<p>Detects whether the 'chrom' column refers to whole chromosomes or chromosome arms.
</p>
</dd>
<dt><code>segMerge(...)</code>:</dt><dd>
<p>Apply the segMerge() function to the track content.<br />
- <strong>...</strong>   : arguments to be passed to segMerge().
</p>
</dd>
<dt><code>segOverlap(...)</code>:</dt><dd>
<p>Apply the segOverlap() function to the track content.<br />
- <strong>...</strong>   : arguments to be passed to segOverlap().
</p>
</dd>
<dt><code>size(chrom, start, end)</code>:</dt><dd>
<p>Count elements in the specified window.<br />
- <strong>chrom</strong>   : single integer, numeric or character value, the chromosomal location.<br />
- <strong>start</strong>   : single integer or numeric value, inferior boundary of the window.<br />
- <strong>end</strong>     : single integer or numeric value, superior boundary of the window.
</p>
</dd>
</dl>

<p>The following methods are inherited (from the corresponding class):
</p>

<ul>
<li><p> addColumn (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> addDataFrame (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addEmptyRows (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> addList (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> addVectors (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> callParams (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> callSuper (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> check (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> chromosomes (<a href="#topic+drawable-class">drawable</a>, overloaded)
</p>
</li>
<li><p> coerce (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> colOrder (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> copy (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> cross (<a href="#topic+crossable-class">crossable</a>)
</p>
</li>
<li><p> defaultParams (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> delColumns (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> draw (<a href="#topic+sliceable-class">sliceable</a>)
</p>
</li>
<li><p> erase (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> export (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> extract (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> field (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> fill (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> fix.param (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getChromEnd (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> getClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getColCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getColNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getLevels (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> getRefClass (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> getRowCount (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> getRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> import (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> indexes (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> initFields (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> initialize (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> metaFields (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> rowOrder (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> setColNames (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> setLevels (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> setName (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setParam (<a href="#topic+drawable-class">drawable</a>)
</p>
</li>
<li><p> setRowNames (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> show (<a href="#topic+refTable-class">refTable</a>, overloaded)
</p>
</li>
<li><p> slice (<a href="#topic+sliceable-class">sliceable</a>, overloaded)
</p>
</li>
<li><p> trace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> types (<a href="#topic+refTable-class">refTable</a>)
</p>
</li>
<li><p> untrace (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li>
<li><p> usingMethods (<a href="methods.html#topic+envRefClass-class">envRefClass</a>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+refTable-class">refTable-class</a></code>
</p>
<p><code><a href="#topic+track.table">track.table</a></code>, <code><a href="#topic+subtrack">subtrack</a></code>, <code><a href="#topic+istrack">istrack</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Exemplar data : subset of human genes
  data(hsGenes)
  
  # Construction
  trackTable &lt;- track.table(
    hsGenes,
    .name = "NCBI Genes",
    .organism = "Homo sapiens",
    .assembly = "GRCh37"
  )
  
  # Slicing
  print(trackTable$slice(chrom="1", as.integer(15e6), as.integer(20e6)))
</code></pre>

<hr>
<h2 id='yline'>
Compute Y line to avoid collision
</h2><span id='topic+yline'></span>

<h3>Description</h3>

<p>This function handles the collision between objects to be drawn, also accounting for labels. It is provided as a convenient component of custom drawing functions, and is currently in use in <code><a href="#topic+draw.boxes">draw.boxes</a></code> and <code><a href="#topic+draw.steps">draw.steps</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  yline(boxes, start, end, label, labelStrand, labelCex, labelSrt, labelAdj,
    labelOverflow, maxDepth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yline_+3A_boxes">boxes</code></td>
<td>

<p>A <code>data.frame</code> with &quot;start.plot&quot; and &quot;end.plot&quot; integer columns defining the boundaries of the graphical elements to be drawn and a &quot;label&quot; character column. In a typical use case, they correspond to the &quot;start&quot;, &quot;end&quot; and &quot;name&quot; columns from the <code>slice</code> to be drawn.
</p>
</td></tr>
<tr><td><code id="yline_+3A_start">start</code></td>
<td>

<p>Single integer value, the left boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="yline_+3A_end">end</code></td>
<td>

<p>Single integer value, the right boundary of the window, in base pairs.
</p>
</td></tr>
<tr><td><code id="yline_+3A_label">label</code></td>
<td>

<p>Single logical value, whether to print labels on boxes or not.
</p>
</td></tr>
<tr><td><code id="yline_+3A_labelstrand">labelStrand</code></td>
<td>

<p>Single logical value, whether to add the strand at the end of labels or not.
</p>
</td></tr>
<tr><td><code id="yline_+3A_labelcex">labelCex</code></td>
<td>

<p>Single numeric value, character expansion factor for labels.
</p>
</td></tr>
<tr><td><code id="yline_+3A_labelsrt">labelSrt</code></td>
<td>

<p>Single numeric value, string rotation angle for labels.
</p>
</td></tr>
<tr><td><code id="yline_+3A_labeladj">labelAdj</code></td>
<td>

<p>'left', 'right' or 'center', the horizontal adjustment of the labels on the boxes.
</p>
</td></tr>
<tr><td><code id="yline_+3A_labeloverflow">labelOverflow</code></td>
<td>

<p>Single logical value, whether to write labels on boxes too narrow to host them or not.
</p>
</td></tr>
<tr><td><code id="yline_+3A_maxdepth">maxDepth</code></td>
<td>

<p>Single integer value, the maximum amount of box heights allowed on the plot to avoid overlaps (if exhausted an error message will be ploted, turning <code>label</code> to <code>FALSE</code> might help).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>boxes</code> with an additional &quot;yline&quot; integer column defining the y coordinate at which the box should be drawn to avoid collision. If an error occurs, a simpleError object will be returned instead and the drawing should be aborted (see <code><a href="#topic+draw.boxes">draw.boxes</a></code> code for a functional example).
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw.boxes">draw.boxes</a></code>, <code><a href="#topic+draw.steps">draw.steps</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
