<!DOCTYPE html><html><head><title>Help for package Rgff</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Rgff}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#check_gff'><p>Test consistency and order of a GFF file</p></a></li>
<li><a href='#check_gff3'><p>Test consistency and order of a GFF3 file</p></a></li>
<li><a href='#check_gtf'><p>Test consistency and order of a GTF file</p></a></li>
<li><a href='#get_features'><p>Analyses the feature type hierarchy of a GFF file</p></a></li>
<li><a href='#get_pairs_from_gff'><p>Creates a features pair file from a GFF file</p></a></li>
<li><a href='#get_pairs_from_gff3'><p>Creates a features pair file from a GFF file</p></a></li>
<li><a href='#get_pairs_from_gtf'><p>Creates a features pair file from a GTF file</p></a></li>
<li><a href='#gff_stats'><p>Summarizes the number of features of each type in a GFF file</p></a></li>
<li><a href='#gff_stats_by_chr'><p>Summarizes the number of elements of each type in each chromosome of a GFF file</p></a></li>
<li><a href='#gff3_to_paths'><p>Creates a features path file from a GFF file</p></a></li>
<li><a href='#gtf_to_gff3'><p>Converts a GTF file into a GFF3 file</p></a></li>
<li><a href='#gtf_to_paths'><p>Creates a features path file from a GTF file</p></a></li>
<li><a href='#plot_features'><p>Plots or exports an image of the feature tree from a GFF file</p></a></li>
<li><a href='#saf_from_gff'><p>Creates a SAF file from a GTF/GFF3 features for the given pairs of blocks and features</p></a></li>
<li><a href='#saf_from_gff3'><p>Creates a SAF file from a GFF3 for the given pairs of blocks and features,</p>
note that groupBy and block vectors must have the same length</a></li>
<li><a href='#saf_from_paths'><p>Generates SAF formatted data from a features path file for the given blocks and features</p></a></li>
<li><a href='#sort_gff'><p>Sorts a GTF/GFF3 file</p></a></li>
<li><a href='#sort_gff3'><p>Sorts a GFF3 file</p></a></li>
<li><a href='#sort_gtf'><p>Sorts a GTF file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>R Utilities for GFF Files</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.6</td>
</tr>
<tr>
<td>Author:</td>
<td>Juan Antonio Garcia-Martin
    <a href="https://orcid.org/0000-0003-0993-4064"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut],
  Juan Carlos Oliveros
    <a href="https://orcid.org/0000-0002-4520-0853"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ctb],
  Rafael Torres-Perez
    <a href="https://orcid.org/0000-0002-3696-4720"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juan Antonio Garcia-Martin &lt;ja.garcia@cnb.csic.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>R utilities for gff files, either general feature format (GFF3) or gene transfer format (GTF) formatted files. This package includes functions for producing summary stats, check for consistency and sorting errors, conversion from GTF to GFF3 format, file sorting, visualization and plotting of feature hierarchy, and exporting user defined feature subsets to SAF format. This tool was developed by the BioinfoGP core facility at CNB-CSIC. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>withr (&ge; 2.4.3), rlang (&ge; 0.4.12), stringi (&ge; 1.7.6),
data.tree (&ge; 1.0.0), tidyr (&ge; 1.1.4), tibble (&ge; 3.1.6),
dplyr (&ge; 1.0.7), RJSONIO (&ge; 1.3-1.6), magrittr (&ge; 2.0.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DiagrammeR (&ge; 1.0.6.1), DiagrammeRsvg (&ge; 0.1), rsvg (&ge;
2.2.0), rmarkdown (&ge; 2.11), knitr (&ge; 1.36), GenomicRanges (&ge;
1.46.1), rtracklayer (&ge; 1.54.0), S4Vectors (&ge; 0.32.3)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-30 08:40:28 UTC; tono</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-30 10:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='check_gff'>Test consistency and order of a GFF file</h2><span id='topic+check_gff'></span>

<h3>Description</h3>

<p>This function tests the consistency and order of a GFF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gff(inFile, fileType = c("AUTO", "GFF3", "GTF"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gff_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="check_gff_+3A_filetype">fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following list indicates the code and description of the issues detected in GFF3 files
</p>

<dl>
<dt>NCOLUMNS_EXCEEDED</dt><dd><p>Input file contains lines with more than 9 fields</p>
</dd>
<dt>NCOLUMNS_INFERIOR</dt><dd><p>Input file contains lines with less than 9 fields</p>
</dd>
<dt>TOO_MANY_FEATURE_TYPES</dt><dd><p>Input file contains too many (more than 100) different feature types</p>
</dd>
<dt>NO_IDs</dt><dd><p>ID attribute not found in any feature</p>
</dd>
<dt>DUPLICATED_IDs</dt><dd><p>There are duplicated IDs</p>
</dd>
<dt>ID_IN_MULTIPLE_CHR</dt><dd><p>The same ID has been found in more than one chromosome</p>
</dd>
<dt>NO_PARENTs</dt><dd><p>Parent attribute not found in any feature</p>
</dd>
<dt>MISSING_PARENT_IDs</dt><dd><p>There are  missing Parent IDs</p>
</dd>
<dt>PARENT_IN DIFFERENT CHR</dt><dd><p>There are features whose Parent is located in a different chromosome</p>
</dd>
<dt>PARENT_DEFINED_BEFORE_ID</dt><dd><p>Feature ids referenced in Parent attribute before being defined as ID</p>
</dd>
<dt>NOT_GROUPED_BY_CHR</dt><dd><p>Features are not grouped by chromosome</p>
</dd>
<dt>NOT_SORTED_BY_COORDINATE</dt><dd><p>Features are not sorted by start coordinate</p>
</dd>
<dt>NOT_VALID_WARNING</dt><dd><p>File cannot be recognized as valid  GFF3. Parsing warnings.</p>
</dd>
<dt>NOT_VALID_ERROR</dt><dd><p>File cannot be recognized as valid GFF3. Parsing errors.</p>
</dd>
</dl>

<p>The following list indicates the code and description of the issues detected in GTF files
</p>

<dl>
<dt>NCOLUMNS_EXCEEDED</dt><dd><p>Input file contains lines with more than 9 fields</p>
</dd>
<dt>NCOLUMNS_INFERIOR</dt><dd><p>Input file contains lines with less than 9 fields</p>
</dd>
<dt>TOO_MANY_FEATURE_TYPES</dt><dd><p>Input file contains too many (more than 100) different feature types</p>
</dd>
<dt>NO_GENE_ID_ATTRIBUTE</dt><dd><p>gene_id attribute not found in any feature</p>
</dd>
<dt>MISSING_GENE_IDs</dt><dd><p>There are features without gene_id attribute</p>
</dd>
<dt>NO_GENE_FEATURES</dt><dd><p>Gene features are not included in this GTF file</p>
</dd>
<dt>DUPLICATED_GENE_IDs</dt><dd><p>There are duplicated gene_ids </p>
</dd>
<dt>GENE_ID_IN_MULTIPLE_CHR</dt><dd><p>The same gene_id has been found in more than one chromosome</p>
</dd>
<dt>NO_TRANSCRIPT_ID_ATTRIBUTE</dt><dd><p>transcript_id attribute not found in any feature There are no elements with transcript_id attribute</p>
</dd>
<dt>MISSING_TRANSCRIPT_IDs</dt><dd><p>There are  features without transcript_id attribute</p>
</dd>
<dt>NO_TRANSCRIPT_FEATURES</dt><dd><p>Transcript features are not included in this GTF file</p>
</dd>
<dt>DUPLICATED_TRANSCRIPT_IDs</dt><dd><p>There are  duplicated transcript_ids </p>
</dd>
<dt>TRANSCRIPT_ID_IN_MULTIPLE_CHR</dt><dd><p>The same transcript_id has been found in more than one chromosome</p>
</dd>
<dt>DUPLICATED_GENE_AND_TRANSCRIPT_IDs</dt><dd><p>Same id has been defined as gene_id and transcript_id</p>
</dd>
<dt>NOT_GROUPED_BY_CHR</dt><dd><p>Features are not grouped by chromosome</p>
</dd>
<dt>NOT_SORTED_BY_COORDINATE</dt><dd><p>Features are not sorted by start coordinate</p>
</dd>
<dt>NOT_VALID_WARNING</dt><dd><p>File cannot be recognized as valid GTF. Parsing warnings.</p>
</dd>
<dt>NOT_VALID_ERROR</dt><dd><p>File cannot be recognized as valid GTF. Parsing errors.</p>
</dd>
</dl>



<h3>Value</h3>

<p>A data frame of detected issues, including a short code name, a description and estimated severity each. In no issues are detected the function will return an empty data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "eden.gff3", package="Rgff")
check_gff(test_gff3)
</code></pre>

<hr>
<h2 id='check_gff3'>Test consistency and order of a GFF3 file</h2><span id='topic+check_gff3'></span>

<h3>Description</h3>

<p>This function test consistency and order of a GFF3 file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gff3(gffFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gff3_+3A_gfffile">gffFile</code></td>
<td>
<p>Path to the input GFF3 file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the errors detected
</p>

<hr>
<h2 id='check_gtf'>Test consistency and order of a GTF file</h2><span id='topic+check_gtf'></span>

<h3>Description</h3>

<p>This function tests consistency and order of a GTF file.
Optionally, can fix some of the consistency errors and sort the GTF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gtf(gtfFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gtf_+3A_gtffile">gtfFile</code></td>
<td>
<p>Path to the input GTF file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the errors detected
</p>

<hr>
<h2 id='get_features'>Analyses the feature type hierarchy of a GFF file</h2><span id='topic+get_features'></span>

<h3>Description</h3>

<p>Based on the feature type hierarchy a GFF file, this function creates and returns
a feature tree or a feature dependency table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_features(
  inFile,
  includeCounts = FALSE,
  outFormat = c("tree", "data.frame", "JSON"),
  fileType = c("AUTO", "GFF3", "GTF")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_features_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GTF/GFF3 features  file</p>
</td></tr>
<tr><td><code id="get_features_+3A_includecounts">includeCounts</code></td>
<td>
<p>Include number of occurrences of each feature and subfeature</p>
</td></tr>
<tr><td><code id="get_features_+3A_outformat">outFormat</code></td>
<td>
<p>Output format of the function. Available formats are: tree (DEFAULT), data.frame and JSON.</p>
</td></tr>
<tr><td><code id="get_features_+3A_filetype">fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the outFormat selected returns a feature tree (tree), a feature dependency table as data.frame (data.frame) or a feature dependency table as JSON object (JSON)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "AthSmall.gff3", package="Rgff")
get_features(test_gff3)
</code></pre>

<hr>
<h2 id='get_pairs_from_gff'>Creates a features pair file from a GFF file</h2><span id='topic+get_pairs_from_gff'></span>

<h3>Description</h3>

<p>This function creates a features pair file from a GFF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pairs_from_gff(
  inFile,
  outFile,
  fileType = c("AUTO", "GFF3", "GTF"),
  forceOverwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pairs_from_gff_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="get_pairs_from_gff_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output features pair file, if not provided the output will be inFile.pairs</p>
</td></tr>
<tr><td><code id="get_pairs_from_gff_+3A_filetype">fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td></tr>
<tr><td><code id="get_pairs_from_gff_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated features pair file
</p>

<hr>
<h2 id='get_pairs_from_gff3'>Creates a features pair file from a GFF file</h2><span id='topic+get_pairs_from_gff3'></span>

<h3>Description</h3>

<p>This function creates a features pair file from a GFF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pairs_from_gff3(gffFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pairs_from_gff3_+3A_gfffile">gffFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="get_pairs_from_gff3_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output features pair file, if not provided the output will be gffFile.pairs</p>
</td></tr>
<tr><td><code id="get_pairs_from_gff3_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated features pair file
</p>

<hr>
<h2 id='get_pairs_from_gtf'>Creates a features pair file from a GTF file</h2><span id='topic+get_pairs_from_gtf'></span>

<h3>Description</h3>

<p>This function creates a features pair file from a GTF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pairs_from_gtf(gtfFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pairs_from_gtf_+3A_gtffile">gtfFile</code></td>
<td>
<p>Path to the input GTF file</p>
</td></tr>
<tr><td><code id="get_pairs_from_gtf_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output features pair file, if not provided the output will be gtfFile.pairs</p>
</td></tr>
<tr><td><code id="get_pairs_from_gtf_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated features pair file
</p>

<hr>
<h2 id='gff_stats'>Summarizes the number of features of each type in a GFF file</h2><span id='topic+gff_stats'></span>

<h3>Description</h3>

<p>This function summarizes the number of features of each type in
a GFF file and returns the statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gff_stats(inFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gff_stats_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with the summary data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "AthSmall.gff3", package="Rgff")
gff_stats(test_gff3)
</code></pre>

<hr>
<h2 id='gff_stats_by_chr'>Summarizes the number of elements of each type in each chromosome of a GFF file</h2><span id='topic+gff_stats_by_chr'></span>

<h3>Description</h3>

<p>This function summarizes the number of features of each type in each chromosome
of a GFF file and returns the statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gff_stats_by_chr(inFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gff_stats_by_chr_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with the summary data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "AthSmall.gff3", package="Rgff")
gff_stats_by_chr(test_gff3)
</code></pre>

<hr>
<h2 id='gff3_to_paths'>Creates a features path file from a GFF file</h2><span id='topic+gff3_to_paths'></span>

<h3>Description</h3>

<p>This function creates a features path file from a GFF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gff3_to_paths(gffFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gff3_to_paths_+3A_gfffile">gffFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="gff3_to_paths_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output features path file, if not provided the output will be gffFile.paths</p>
</td></tr>
<tr><td><code id="gff3_to_paths_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated features path file
</p>

<hr>
<h2 id='gtf_to_gff3'>Converts a GTF file into a GFF3 file</h2><span id='topic+gtf_to_gff3'></span>

<h3>Description</h3>

<p>This function converts a GTF file into a GFF3 file mantaining the feature hierarchy
defined by the gene_id and transcript_id attributes. The remaining attributes of each feature
will be kept with the same name and value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_to_gff3(gtfFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtf_to_gff3_+3A_gtffile">gtfFile</code></td>
<td>
<p>Path to the input GTF file</p>
</td></tr>
<tr><td><code id="gtf_to_gff3_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output GFF3 file, inf not provided the output will be gtfFile.gff3</p>
</td></tr>
<tr><td><code id="gtf_to_gff3_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated GFF3 file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
test_gtf&lt;-system.file("extdata", "AthSmall.gtf", package="Rgff")
gtf_to_gff3(test_gtf)

## End(Not run)
</code></pre>

<hr>
<h2 id='gtf_to_paths'>Creates a features path file from a GTF file</h2><span id='topic+gtf_to_paths'></span>

<h3>Description</h3>

<p>This function creates a features path file from a GTF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_to_paths(gtfFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtf_to_paths_+3A_gtffile">gtfFile</code></td>
<td>
<p>Path to the input GTF file</p>
</td></tr>
<tr><td><code id="gtf_to_paths_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output features path file, if not provided the output will be gtfFile.paths</p>
</td></tr>
<tr><td><code id="gtf_to_paths_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated features path file
</p>

<hr>
<h2 id='plot_features'>Plots or exports an image of the feature tree from a GFF file</h2><span id='topic+plot_features'></span>

<h3>Description</h3>

<p>This function plots the feature tree from a GFF file or, if an output file name is provided,
exports an image of in the desired format (&quot;png&quot;, &quot;pdf&quot; or &quot;svg&quot;).
Packages &quot;DiagrammeR&quot;, &quot;DiagrammeRsvg&quot; and &quot;rsvg&quot; must be installed to use this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_features(
  inFile,
  outFile,
  includeCounts = FALSE,
  fileType = c("AUTO", "GFF3", "GTF"),
  exportFormat = c("png", "pdf", "svg")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_features_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="plot_features_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output features image file, if not provided the tree will be plotted</p>
</td></tr>
<tr><td><code id="plot_features_+3A_includecounts">includeCounts</code></td>
<td>
<p>Include number of occurrences of each subfeature</p>
</td></tr>
<tr><td><code id="plot_features_+3A_filetype">fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). If not provided it will be determined from the file name.</p>
</td></tr>
<tr><td><code id="plot_features_+3A_exportformat">exportFormat</code></td>
<td>
<p>Output image format when it is not possible to deduce it from the extension of outFile (&quot;png&quot;, &quot;pdf&quot; or &quot;svg&quot;). Default, &quot;png&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path of the output features image file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "AthSmall.gff3", package="Rgff")
plot_features(test_gff3)
</code></pre>

<hr>
<h2 id='saf_from_gff'>Creates a SAF file from a GTF/GFF3 features for the given pairs of blocks and features</h2><span id='topic+saf_from_gff'></span>

<h3>Description</h3>

<p>This function creates a SAF file from a GTF/GFF3 features for the given blocks
and features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saf_from_gff(
  inFile,
  outFile,
  fileType = c("AUTO", "GFF3", "GTF"),
  forceOverwrite = FALSE,
  features = c("gene &gt; exon"),
  sep = "&gt;"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saf_from_gff_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="saf_from_gff_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output SAF file, if not provided the output path will be the input path with the suffix &quot;.feature1-block1.feature2-block2(...).saf&quot;</p>
</td></tr>
<tr><td><code id="saf_from_gff_+3A_filetype">fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td></tr>
<tr><td><code id="saf_from_gff_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
<tr><td><code id="saf_from_gff_+3A_features">features</code></td>
<td>
<p>Vector of pairs of features/blocks, separated by '&gt;' (see sep argument). In the case of features without defined blocks, only the feature is needed (see example)</p>
</td></tr>
<tr><td><code id="saf_from_gff_+3A_sep">sep</code></td>
<td>
<p>Separator of each &quot;feature&quot; and &quot;block&quot; provided in the feature argument (default '&gt;')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated SAF file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "AthSmall.gff3", package="Rgff")
## Default usage, extract gene features by exon blocks 
saf_from_gff(test_gff3)
## Define only feature without block to count reads within the whole genomic locus 
saf_from_gff(test_gff3, features=c("gene"))
## Define multiple features for counting readsoverlapping only in exonic regions 
saf_from_gff(test_gff3, features=c("gene &gt; exon", "ncRNA_gene &gt; exon"))
</code></pre>

<hr>
<h2 id='saf_from_gff3'>Creates a SAF file from a GFF3 for the given pairs of blocks and features,
note that groupBy and block vectors must have the same length</h2><span id='topic+saf_from_gff3'></span>

<h3>Description</h3>

<p>This function creates a SAF file from a GFF3 for the given blocks
and features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saf_from_gff3(
  gffFile,
  outFile,
  forceOverwrite = FALSE,
  features = c("gene &gt; exon"),
  sep = "&gt;"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saf_from_gff3_+3A_gfffile">gffFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="saf_from_gff3_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output SAF file,  if not provided the output path will be the input path (without extension) with the suffix &quot;.groupBy1-block1.groupBy2-block2(...).saf&quot;</p>
</td></tr>
<tr><td><code id="saf_from_gff3_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
<tr><td><code id="saf_from_gff3_+3A_features">features</code></td>
<td>
<p>Vector of pairs of features, separated by '&gt;' (see sep argument), to be used respectively as &quot;group by&quot; and &quot;block&quot;</p>
</td></tr>
<tr><td><code id="saf_from_gff3_+3A_sep">sep</code></td>
<td>
<p>Separator of each &quot;group by&quot; and &quot;block&quot; provided in the feature argument (default '&gt;')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the generated SAF file
</p>

<hr>
<h2 id='saf_from_paths'>Generates SAF formatted data from a features path file for the given blocks and features</h2><span id='topic+saf_from_paths'></span>

<h3>Description</h3>

<p>This function creates a SAF file from a features path file for the given blocks
and features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saf_from_paths(
  pathsFile,
  groupBy = c("mRNA", "gene"),
  block = c("exon", "CDS")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saf_from_paths_+3A_pathsfile">pathsFile</code></td>
<td>
<p>Path to the input features path file</p>
</td></tr>
<tr><td><code id="saf_from_paths_+3A_groupby">groupBy</code></td>
<td>
<p>Vector of features to group by in feature count</p>
</td></tr>
<tr><td><code id="saf_from_paths_+3A_block">block</code></td>
<td>
<p>Vector of features to be used as block in feature count</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SAF formatted data frame for the given blocks and features
</p>

<hr>
<h2 id='sort_gff'>Sorts a GTF/GFF3 file</h2><span id='topic+sort_gff'></span>

<h3>Description</h3>

<p>This function produces a sorted GFF file from an unsorted GFF file.
The default order is by Chromosome, Start, End (reverse) and feature (based on the precedency in feature tree)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_gff(
  inFile,
  outFile,
  fileType = c("AUTO", "GFF3", "GTF"),
  forceOverwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sort_gff_+3A_infile">inFile</code></td>
<td>
<p>Path to the input GFF file</p>
</td></tr>
<tr><td><code id="sort_gff_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output sorted file, if not provided the output will be the input path (without extension) with the suffix sorted.gtf/gff3</p>
</td></tr>
<tr><td><code id="sort_gff_+3A_filetype">fileType</code></td>
<td>
<p>Version of the input file (GTF/GFF3). Default AUTO: determined from the file name.</p>
</td></tr>
<tr><td><code id="sort_gff_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the sorted feature file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_gff3&lt;-system.file("extdata", "eden.gff3", package="Rgff")
sort_gff(test_gff3)
</code></pre>

<hr>
<h2 id='sort_gff3'>Sorts a GFF3 file</h2><span id='topic+sort_gff3'></span>

<h3>Description</h3>

<p>This function produces a sorted GFF3 file from an unsorted GFF3 file.
The output file will be sorted by Chromosome, Start, End (reverse) and feature (based on the precedency in feature tree)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_gff3(gffFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sort_gff3_+3A_gfffile">gffFile</code></td>
<td>
<p>Path to the input GFF3 file</p>
</td></tr>
<tr><td><code id="sort_gff3_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output GFF3 file, if not provided the output will be  the input path (without extension) with the suffix sorted.gff3</p>
</td></tr>
<tr><td><code id="sort_gff3_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the sorted GFF3 file
</p>

<hr>
<h2 id='sort_gtf'>Sorts a GTF file</h2><span id='topic+sort_gtf'></span>

<h3>Description</h3>

<p>This function produces a sorted GTF file from an unsorted GTF file.
The default order is by Chromosome, Start, End (reverse) and feature (gene -&gt; transcript -&gt; OTHERS)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_gtf(gtfFile, outFile, forceOverwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sort_gtf_+3A_gtffile">gtfFile</code></td>
<td>
<p>Path to the input GTF file</p>
</td></tr>
<tr><td><code id="sort_gtf_+3A_outfile">outFile</code></td>
<td>
<p>Path to the output GTF file, if not provided the output will be  the input path (without extension) with the suffix sorted.gtf</p>
</td></tr>
<tr><td><code id="sort_gtf_+3A_forceoverwrite">forceOverwrite</code></td>
<td>
<p>If output file exists, overwrite the existing file. (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the sorted GTF file
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
