<!DOCTYPE html><html><head><title>Help for package rhcoclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rhcoclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FCGE_Data_GMP'>
<p>A real glutathione metabolism pathway (GMP) dataset for 'rhcoclust' package</p></a></li>
<li><a href='#FCGE_Data_PPARs'>
<p>A real PPAR signaling pathways (PPAR-SP) dataset for 'rhcoclust' package</p></a></li>
<li><a href='#plot_rhcoclust'>
<p>Plot of the 'rhcoclust' objects</p></a></li>
<li><a href='#reversestring'>
<p>Reverse given string</p></a></li>
<li><a href='#rhcoclust'><p>The function for co-clustering sample and feature to explore significant samples and their regulatory features</p></a></li>
<li><a href='#rhcoclust_internet'>
<p>Interaction network (internet) of the 'rhcoclust' objects</p></a></li>
<li><a href='#rhcoclust_network'>
<p>Visualization of clustering network plot</p></a></li>
<li><a href='#simu_data'>
<p>A predefined simulated data for 'rhcoclust' package</p></a></li>
<li><a href='#simulate_data'>
<p>Simulate data for robust hirarachical clustering to identify significant co-cluster</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Robust Hierarchical Co-Clustering to Identify Significant
Co-Cluster</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Here we performs robust hierarchical co-clustering between row and column entities of a data matrix in 
             absence and presence of outlying observations. It can be used to explore important co-clusters consisting of  
             important samples and their regulatory significant features. Please see Hasan, Badsha and Mollah (2020) 
             &lt;<a href="https://doi.org/10.1101%2F2020.05.13.094946">doi:10.1101/2020.05.13.094946</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Imports:</td>
<td>fields, grDevices, graphics, igraph, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-28 03:48:19 UTC; mdbadsha</td>
</tr>
<tr>
<td>Author:</td>
<td>Md. Bahadur Badsha [aut, cre],
  Mohammad Nazmol Hasan [aut],
  Md. Nurul Haque Mollah [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-29 03:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='FCGE_Data_GMP'>
A real glutathione metabolism pathway (GMP) dataset for 'rhcoclust' package
</h2><span id='topic+FCGE_Data_GMP'></span>

<h3>Description</h3>

<p>This is the real data matrix that used as an example in 'rhcoclust' package.
</p>


<h3>Details</h3>

<p>The real data matrix collected from Nyström-Persson et al., 2013 (https://toxygates.nibiohn.go.jp/toxygates/#columns).The column of the data matrix represents doses of chemical compounds (DCCs) and row represents genes. Each element in the dataset represents fold change gene expression data.
</p>


<h3>Value</h3>

<p>Matrix
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>References</h3>

<p>1. Nyström-Persson et al., (2013). Toxygates: interactive toxicity analysis on a hybrid microarray and linked data platform. Bioinformatics, Volume 29, Issue 23,Pages 3080–3086.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load library
library(rhcoclust)

# Load real data
data("FCGE_Data_GMP")
</code></pre>

<hr>
<h2 id='FCGE_Data_PPARs'>
A real PPAR signaling pathways (PPAR-SP) dataset for 'rhcoclust' package
</h2><span id='topic+FCGE_Data_PPARs'></span>

<h3>Description</h3>

<p>This is the real data matrix that used as an example in 'rhcoclust' package.
</p>


<h3>Details</h3>

<p>The real data matrix collected from Nyström-Persson et al., 2013 (https://toxygates.nibiohn.go.jp/toxygates/#columns).The column of the data matrix represents doses of chemical compounds (DCCs) and row represents genes. Each element in the dataset represents fold change gene expression data.
</p>


<h3>Value</h3>

<p>Matrix
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>References</h3>

<p>1. Nyström-Persson et al., (2013). Toxygates: interactive toxicity analysis on a hybrid microarray and linked data platform. Bioinformatics, Volume 29, Issue 23,Pages 3080–3086.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load library
library(rhcoclust)

# Load real data
data("FCGE_Data_PPARs")
</code></pre>

<hr>
<h2 id='plot_rhcoclust'>
Plot of the 'rhcoclust' objects
</h2><span id='topic+plot_rhcoclust'></span>

<h3>Description</h3>

<p>This function used for two plots from output of rhcoclust (i) plot results for gene (row) and compound (column) co-cluster graph, and (ii) plot graph of QCC for identification of biomarker co-cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_rhcoclust(CoClustObj, plot.cocluster = FALSE, plot.SCC = FALSE,
cex.xaxis = 0.7, cex.yaxis = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_rhcoclust_+3A_coclustobj">CoClustObj</code></td>
<td>

<p>Output objects from rhcoclust
</p>
</td></tr>
<tr><td><code id="plot_rhcoclust_+3A_plot.cocluster">plot.cocluster</code></td>
<td>

<p>To set no plotting as the default for cocluster.
</p>
</td></tr>
<tr><td><code id="plot_rhcoclust_+3A_plot.scc">plot.SCC</code></td>
<td>

<p>To set no plotting as the default for SCC.
</p>
</td></tr>
<tr><td><code id="plot_rhcoclust_+3A_cex.xaxis">cex.xaxis</code></td>
<td>

<p>A numerical value giving to control/annotation text size in x-axis. Default is 0.7.
</p>
</td></tr>
<tr><td><code id="plot_rhcoclust_+3A_cex.yaxis">cex.yaxis</code></td>
<td>

<p>A numerical value giving to control/annotation text size in y-axis. Default is 0.5.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+rhcoclust">rhcoclust</a> for generating a graph objects for clustering network
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load necessary library
library(rhcoclust)
library(fields)

# Load real data
data("FCGE_Data_GMP")
data("FCGE_Data_PPARs")
# Load predefined real data
# Real data use: data &lt;- FCGE_Data_PPARs
# Real data use: data &lt;- FCGE_Data_GMP

# Load predefined simulated data
data("simu_data")

# simulated data
data &lt;- simu_data

# Apply rhcoclust to identify significant co-cluster of samples and their regulatory features
CoClustObj &lt;- rhcoclust(data, rk=4, ck=3, method.dist = "manhattan", method.hclust = "ward.D")
# For real data either FCGE_Data_PPARs or FCGE_Data_GMP
#CoClustObj &lt;- rhcoclust(data, rk=3, ck=3, method.dist = "manhattan", method.hclust = "ward.D")

# Plot co-cluster
# Please use par(mar=c(6, 10, 3, 6))  or modify if needed for best fit of the graph
# mar order: bottom, left, top, and right
plot_rhcoclust (CoClustObj, plot.coclust = TRUE, plot.SCC = FALSE,
cex.xaxis = 0.7, cex.yaxis = 0.5)

# Plot SCC
# Please use dev.off() to avoid the figure margin from previous plot
plot_rhcoclust (CoClustObj, plot.coclust = FALSE, plot.SCC = TRUE)
# Please add legend with change or add any parameters if needed.
legend("topleft",
       legend = c("Upper-significant", "Insignificant","Down-significant"),
       col = c("red","black","blue"),
       bty = "n",
       pch = c(20,20,20),
       pt.cex = 2,
       cex = 1.2,
       x.intersp = 0.2,
       y.intersp = 0.4,
       text.col = "black",
       horiz = FALSE ,
       inset = c(0.3, -0.08))
</code></pre>

<hr>
<h2 id='reversestring'>
Reverse given string
</h2><span id='topic+reversestring'></span>

<h3>Description</h3>

<p>This is the function for reverse string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reversestring(string, n = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reversestring_+3A_string">string</code></td>
<td>

<p>Given string or seq
</p>
</td></tr>
<tr><td><code id="reversestring_+3A_n">n</code></td>
<td>

<p>By which n-plets we should reverse the given string
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is uesd to reverse given string or seq
</p>


<h3>Value</h3>

<p>Reversed string or seq
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 &lt;- c("R1C1", "R2C2", "R3C3")
reversestring(x1, 2)
</code></pre>

<hr>
<h2 id='rhcoclust'>The function for co-clustering sample and feature to explore significant samples and their regulatory features</h2><span id='topic+rhcoclust'></span>

<h3>Description</h3>

<p>Toxicogenomic studies require co-clustering to identify biomarker genes for the assessment of chemical toxicity from gene expression levels. It is also essential in the drug discovery experiments. However, gene expression datasets are often contaminated by outliers due to several steps involve in the data generating process. This package performs robust hierarchical co-clustering between row and column entities of a data matrix by reducing the influence of outlying observations. It can be used to explore biomarker genes those are divided into upregulatory and downregulatory groups by the influence of different chemical compounds groups more accurately. It can also provide the statistical significance of the identified co-clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rhcoclust(data, rk, ck, method.dist = "manhattan", method.hclust= "ward.D")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rhcoclust_+3A_data">data</code></td>
<td>
<p>A data matrix containing data having the characteristics of interval and ratio level of measurement or continuous data</p>
</td></tr>
<tr><td><code id="rhcoclust_+3A_rk">rk</code></td>
<td>
<p>Number of clusters in the row entities of the data matrix</p>
</td></tr>
<tr><td><code id="rhcoclust_+3A_ck">ck</code></td>
<td>
<p>Number of clusters in the column entities of the data matrix</p>
</td></tr>
<tr><td><code id="rhcoclust_+3A_method.dist">method.dist</code></td>
<td>
<p>The distance measure to be used. The default is &quot;manhattan&quot;. The other options are &quot;euclidean&quot;, &quot;maximum&quot;, &quot;canberra&quot;, &quot;binary&quot; or &quot;minkowski&quot;. Any unambiguous substring can be given.</p>
</td></tr>
<tr><td><code id="rhcoclust_+3A_method.hclust">method.hclust</code></td>
<td>
<p>The agglomeration method to be used. The default is &quot;ward.D&quot;. The other options are &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot; (= UPGMA), &quot;mcquitty&quot; (= WPGMA), &quot;median&quot; (= WPGMC) or &quot;centroid&quot; (= UPGMC).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="base.html#topic+list">list</a> of object that containing the following:
</p>
<p>Coclust_MeanMat : A data frame containing combination of row and column cluster number in the first column and their ranked co-cluster mean in the second column. In the first column first number indicates row cluster index and second number indicates column cluster index, respectively.
</p>
<p>CoClsDtMat : The reorganized transformed data matrix to generate co-cluster graph.
</p>
<p>NG_Cocls : The index of gene/row names.
</p>
<p>NC_Cocls : The index of column names.
</p>
<p>rowclust :The gene/row entity clusters.
</p>
<p>colclust : The column entity clusters.
</p>
<p>colorsG: Colors of genes/row entity clusters to generate co-cluster graph.
</p>
<p>colorsC: Colors of DCCs/column entity clusters to generate co-cluster graph.
</p>
<p>CentralLine: Central Line of individual control chart to generate graph of control chart and to identify significant co-clusters.
</p>
<p>UpContLimit: Upper Control Limit to generate graph of control chart and to identify significant co-clusters.
</p>
<p>LowrContLimit: Lower Control Limit to generate graph of control chart and to identify significant co-clusters.
</p>
<p>color: Colors to generate individual control chart.
</p>
<p>pchmark: Shape of points to generate individual control chart.
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load necessary library
library(rhcoclust)
library(fields)

# Load real data
data("FCGE_Data_GMP")
data("FCGE_Data_PPARs")
# Load predefined real data
# Real data use: data &lt;- FCGE_Data_PPARs
# Real data use: data &lt;- FCGE_Data_GMP

# Load predefined simulated data
data("simu_data")

# simulated data
data &lt;- simu_data
# Apply rhcoclust to identify significant co-cluster of samples and their regulatory features
CoClustObj &lt;- rhcoclust(data, rk=4, ck=3, method.dist = "manhattan", method.hclust = "ward.D")
# For real data either FCGE_Data_PPARs or FCGE_Data_GMP
#CoClustObj &lt;- rhcoclust(data, rk=3, ck=3, method.dist = "manhattan", method.hclust = "ward.D")

# A data frame containing combination of row and column cluster number in the first
# column and their ranked co-cluster mean in the second cluster.
#GC_cls_MeanMat &lt;- CoClustObj$Coclust_MeanMat

# The reorganized transformed data matrix to generate co-cluster graph.
CoClsDtMat &lt;- CoClustObj$CoClsDtMat

# The gene/row entity clusters.
rowclust &lt;- CoClustObj$rowclust

# The column entity clusters.
colclust &lt;- CoClustObj$colclust

# Colors of genes/row entity clusters to generate co-cluster graph
colorsG &lt;- CoClustObj$colorsG

# Colors of DCCs/column entity clusters to generate co-cluster graph
colorsC &lt;- CoClustObj$colorsC

# Central Line of individual control chart to generate graph of control chart and to
# identify significant co-clusters.
CntrLine_QC &lt;- CoClustObj$CentralLine

# Upper Control Limit to generate graph of control chart and to identify significant
# co-clusters.
UCL_QC &lt;- CoClustObj$UpContLimit

# Lower Control Limit to generate graph of control chart and to identify significant
# co-clusters.
LCL_QC &lt;- CoClustObj$LowrContLimit

# Colors to generate individual control chart.
ColorQC &lt;- CoClustObj$color

# Shape of points to generate individual control chart.
PcmQC &lt;- CoClustObj$pchmark

# Plot co-cluster
# par(mar=c(6,10,3,6)) # Modify if needed
# mar order: bottom, left, top, and right
# please use different values if needed for cex.xaxis and cex.yaxis
# to adjust xaxis and yaxis text
plot_rhcoclust (CoClustObj, plot.coclust = TRUE, plot.SCC = FALSE)

# Plot SCC
# use dev.off() to avoid the figure margin from previous plot
plot_rhcoclust (CoClustObj, plot.coclust = FALSE, plot.SCC = TRUE)
</code></pre>

<hr>
<h2 id='rhcoclust_internet'>
Interaction network (internet) of the 'rhcoclust' objects
</h2><span id='topic+rhcoclust_internet'></span>

<h3>Description</h3>

<p>This function is used for visualization of clustering interaction network plot for the objects that are generated by rhcoclust and list of up-regulated and down-regulated variables list. There are three layers in the network plot, (i) genes (rows) are shown in the first layer, (ii) co-cluster are in the second layer and (iii) columns (compounds) are in the last layer. Red and blue color indicates up and down regulated respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rhcoclust_internet(data, CoClustObj, CoClust.sig =FALSE,
cex.nodes = 0.7, edge.width = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rhcoclust_internet_+3A_data">data</code></td>
<td>
<p>A data matrix containing data having the characteristics of interval and ratio level of measurement or continuous data</p>
</td></tr>
<tr><td><code id="rhcoclust_internet_+3A_coclustobj">CoClustObj</code></td>
<td>
<p>A list of output objects from rhcoclust</p>
</td></tr>
<tr><td><code id="rhcoclust_internet_+3A_coclust.sig">CoClust.sig</code></td>
<td>
<p>To set no plotting as the default for up and down regulated group. Default is FALSE</p>
</td></tr>
<tr><td><code id="rhcoclust_internet_+3A_cex.nodes">cex.nodes</code></td>
<td>

<p>A numerical value giving to control/annotation node size in the network. Default is 0.7.
</p>
</td></tr>
<tr><td><code id="rhcoclust_internet_+3A_edge.width">edge.width</code></td>
<td>

<p>A numerical value giving to control/annotation edge width in the network. Default is 1.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot
A <a href="base.html#topic+list">list</a> up-regulated and down-regulated variables.
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+rhcoclust">rhcoclust</a> for generating a graph objects for clustering network
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load necessary library
library(rhcoclust)
library(fields)

# Load real data
data("FCGE_Data_GMP")
data("FCGE_Data_PPARs")
# Load predefined real data
# Real data use: data &lt;- FCGE_Data_PPARs
# Real data use: data &lt;- FCGE_Data_GMP

# Load predefined simulated data
data("simu_data")

# simulated data
data &lt;- simu_data

# Apply rhcoclust to identify significant co-cluster of samples and their regulatory features
CoClustObj &lt;- rhcoclust(data, rk=4, ck=3, method.dist = "manhattan", method.hclust = "ward.D")
# For real data either FCGE_Data_PPARs or FCGE_Data_GMP
#CoClustObj &lt;- rhcoclust(data, rk=3, ck=3, method.dist = "manhattan", method.hclust = "ward.D")

# Plot interaction network (internet)
# Please use dev.off() to avoid the figure margin from previous plot
# mar order: bottom, left, top, and right
# please use par(mar=c(5,2,5,2)) or modify when necessary to best fit for the plot
Netrhcoclust &lt;- rhcoclust_internet(data, CoClustObj = CoClustObj,
CoClust.sig = FALSE, cex.nodes = 0.7, edge.width = 1)
# Please change or add any parameter if needed.
text(x = -1, y = 1.1, "Row Cluster", cex = 0.7)
# Please change or add any parameter if needed.
text(x = 0, y = 1.1, "Co-Cluster", cex = 0.7)
# Please change or add any parameter if needed.
text(x = 1, y = 1.1, "Column Cluster", cex = 0.7)
</code></pre>

<hr>
<h2 id='rhcoclust_network'>
Visualization of clustering network plot
</h2><span id='topic+rhcoclust_network'></span>

<h3>Description</h3>

<p>This function is used for visualization of clustering network plot, the plot objects are generated by rhcoclust.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rhcoclust_network(CoClustObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rhcoclust_network_+3A_coclustobj">CoClustObj</code></td>
<td>

<p>Output objects from rhcoclust
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+rhcoclust">rhcoclust</a> for generating a graph objects for clustering network
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load necessary library
library(rhcoclust)
library(fields)
library(igraph)

# Load real data
data("FCGE_Data_GMP")
data("FCGE_Data_PPARs")
# Load predefined real data
# Real data use: data &lt;- FCGE_Data_PPARs
# Real data use: data &lt;- FCGE_Data_GMP

# Load predefined simulated data
data("simu_data")

# simulated data
data &lt;- simu_data


# Apply rhcoclust to identify significant co-cluster of samples and their regulatory features
CoClustObj &lt;- rhcoclust(data, rk=4, ck=3, method.dist = "manhattan", method.hclust = "ward.D")
# For real data either FCGE_Data_PPARs or FCGE_Data_GMP
#CoClustObj &lt;- rhcoclust(data, rk=3, ck=3, method.dist = "manhattan", method.hclust = "ward.D")

# Visualization of clustering network plot
rhcoclust_network(CoClustObj)
</code></pre>

<hr>
<h2 id='simu_data'>
A predefined simulated data for 'rhcoclust' package
</h2><span id='topic+simu_data'></span>

<h3>Description</h3>

<p>This is the  predefined simulated data matrix that used as an example in 'rhcoclust' package.
</p>


<h3>Details</h3>

<p>The column of the simulated data matrix represents doses of chemical compounds (DCCs) and row represents genes. Each element in the dataset represents fold change gene expression data.
</p>


<h3>Value</h3>

<p>Matrix
</p>


<h3>Author(s)</h3>

<p>Md Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+simulate_data">simulate_data</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(rhcoclust)

# Load data
data("simu_data")
</code></pre>

<hr>
<h2 id='simulate_data'>
Simulate data for robust hirarachical clustering to identify significant co-cluster
</h2><span id='topic+simulate_data'></span>

<h3>Description</h3>

<p>We generate fold change gene expression (FCGE) data according to the characteristics of toxicogenomic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_data(no.gene, no.dcc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_data_+3A_no.gene">no.gene</code></td>
<td>

<p>Number of genes in the simulated data.
</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_no.dcc">no.dcc</code></td>
<td>

<p>Numbe of doses of chemical compounds (dcc) in the simulated data.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are four gene groups and three DCCs groups in the simulated dataset. The gene group 1, 2, 3 and 4 consists the genes G1-G10, G11-G20, G21-G30 and G31-G50 respectively and DCCs group 1, 2 and 3 consists the DCCs C1_High-C5_High- C1_Middle-C5_Middle, C6_High-C10_High- C6_Middle-C10_Middle and C1_Low-C12_Low- C11_Middle- C12_Middle- C11_High- C12_High respectively. Where, G stands for gene and C stands for chemical compound arranged in the row and column of the simulated data matrix respectively. The error term N(0,0.35) from normal distribution with mean 0 and variance 0.35 is added to each element of the simulated dataset. In the simulated dataset the gene group-1 is up-regulated by the DCCs group-1, gene goup-2 is up and down-regulated by the DCCs group-2 and 1 respectively. The gene group-3 is down-regulated by the DCs group-2. The gene group-4 is not regulated by any of the DCCs groups and DCCs group-3 does not influence any of the genes in the dataset.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+list">list</a> of object that containing the following:
</p>
<p>SimData: A simulated data matrix as generated.
</p>
<p>SimDataRnd: Randomly distributed row and column entity of simulated data.
</p>
<p>GCmat: Transformed simulated data.
</p>


<h3>Author(s)</h3>

<p>Md. Bahadur Badsha &lt;mbbadshar@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load library
library(rhcoclust)

# Number of genes in the simulated data.
no.gene &lt;- 50

# Numbe of doses of chemical compounds (dcc) in the simulated data.
no.dcc &lt;- 12

SimulteData &lt;- simulate_data(no.gene,no.dcc)

# A simulated data matrix as generated.
SimulteData$SimData

# A randomly distributed row and column entity of simulated data.
SimulteData$SimDataRnd

# A transformed simulated data.
SimulteData$GCmat
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
