<!DOCTYPE html><html><head><title>Help for package rhierbaps</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rhierbaps}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_change_in_ml'><p>calc_change_in_ml</p></a></li>
<li><a href='#calc_log_ml'><p>calc_log_ml</p></a></li>
<li><a href='#hierBAPS'><p>hierBAPS</p></a></li>
<li><a href='#join_units_2'><p>join_units_2</p></a></li>
<li><a href='#load_fasta'><p>load_fasta</p></a></li>
<li><a href='#log_stirling2'><p>log_stirling2</p></a></li>
<li><a href='#model_search_parallel'><p>model_search_parallel</p></a></li>
<li><a href='#move_units_1'><p>move_units_1</p></a></li>
<li><a href='#plot_sub_cluster'><p>plot_sub_cluster</p></a></li>
<li><a href='#preproc_alignment'><p>preproc_alignment</p></a></li>
<li><a href='#reallocate_units_4'><p>reallocate_units_4</p></a></li>
<li><a href='#save_lml_logs'><p>save_lml_logs</p></a></li>
<li><a href='#split_clusters_3'><p>split_clusters_3</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Clustering Genetic Sequence Data Using the HierBAPS Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the hierarchical Bayesian analysis of populations structure (hierBAPS) 
  algorithm of Cheng et al. (2013) &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmst028">doi:10.1093/molbev/mst028</a>&gt; for clustering DNA sequences 
  from multiple sequence alignments in FASTA format. 
  The implementation includes improved defaults and plotting capabilities 
  and unlike the original 'MATLAB' version removes singleton SNPs by default.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, purrr, utils, ggplot2, matrixStats, patchwork, methods</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, ggtree, phytools, testthat, formatR</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/gtonkinhill/rhierbaps">https://github.com/gtonkinhill/rhierbaps</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/gtonkinhill/rhierbaps/issues">https://github.com/gtonkinhill/rhierbaps/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-18 14:26:42 UTC; gt4</td>
</tr>
<tr>
<td>Author:</td>
<td>Gerry Tonkin-Hill [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gerry Tonkin-Hill &lt;g.tonkinhill@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-18 14:50:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_change_in_ml'>calc_change_in_ml</h2><span id='topic+calc_change_in_ml'></span>

<h3>Description</h3>

<p>Calculate the change in the log marginal likelihood after moving index to each possible cluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_change_in_ml(snp.object, partition, indexes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_change_in_ml_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="calc_change_in_ml_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating a partition of the isolates.</p>
</td></tr>
<tr><td><code id="calc_change_in_ml_+3A_indexes">indexes</code></td>
<td>
<p>Indexes of the isolates to be moved (must come from one cluster.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the best cluster to move indexes to.
</p>

<hr>
<h2 id='calc_log_ml'>calc_log_ml</h2><span id='topic+calc_log_ml'></span>

<h3>Description</h3>

<p>Calculate the log marginal likelihood assuming a Multinomial-Dirichlet distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_log_ml(snp.object, partition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_log_ml_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="calc_log_ml_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating a partition of the isolates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log marginal likelihood of the given partition.
</p>

<hr>
<h2 id='hierBAPS'>hierBAPS</h2><span id='topic+hierBAPS'></span>

<h3>Description</h3>

<p>Runs the hierBAPS algorithm of Cheng et al. 2013
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hierBAPS(
  snp.matrix,
  max.depth = 2,
  n.pops = floor(nrow(snp.matrix)/5),
  quiet = FALSE,
  n.extra.rounds = 0,
  assignment.probs = FALSE,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hierBAPS_+3A_snp.matrix">snp.matrix</code></td>
<td>
<p>Character matrix of aligned sequences produced by <a href="#topic+load_fasta">load_fasta</a>.</p>
</td></tr>
<tr><td><code id="hierBAPS_+3A_max.depth">max.depth</code></td>
<td>
<p>Maximum depth of hierarchical search (default = 2).</p>
</td></tr>
<tr><td><code id="hierBAPS_+3A_n.pops">n.pops</code></td>
<td>
<p>Maximum number of populations in the data (default = number of isolates/5)</p>
</td></tr>
<tr><td><code id="hierBAPS_+3A_quiet">quiet</code></td>
<td>
<p>Whether to suppress progress information (default=FALSE).</p>
</td></tr>
<tr><td><code id="hierBAPS_+3A_n.extra.rounds">n.extra.rounds</code></td>
<td>
<p>The number of additional rounds to perform after the default hierBAPS
settings (default=0). If set to Inf it will run until a local optimum is reached
(this might take a long time).</p>
</td></tr>
<tr><td><code id="hierBAPS_+3A_assignment.probs">assignment.probs</code></td>
<td>
<p>whether or not to calculate the assignment probabilities to each cluster (default=FALSE)</p>
</td></tr>
<tr><td><code id="hierBAPS_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing a dataframe indicating an assignment of each sequence
to hierarchical clusters as well as the log marginal likelihoods for each level.
</p>


<h3>Author(s)</h3>

<p>Gerry Tonkin-Hill
</p>


<h3>References</h3>

<p>Cheng, Lu, Thomas R. Connor, Jukka Sirén, David M. Aanensen, and Jukka Corander. 2013. “Hierarchical and Spatially Explicit Clustering of DNA Sequences with BAPS Software.” Molecular Biology and Evolution 30 (5): 1224–28.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>snp.matrix &lt;- load_fasta(system.file("extdata", "small_seqs.fa", package = "rhierbaps"))
hb &lt;- hierBAPS(snp.matrix, max.depth=2, n.pops=20, quiet=FALSE)


snp.matrix &lt;- load_fasta(system.file("extdata", "seqs.fa", package = "rhierbaps"))
system.time({hb &lt;- hierBAPS(snp.matrix, max.depth=2, n.pops=20, quiet=FALSE)})


</code></pre>

<hr>
<h2 id='join_units_2'>join_units_2</h2><span id='topic+join_units_2'></span>

<h3>Description</h3>

<p>Peform an iteration of the second move in the algorithm. That is combine two clusters
to improve the marginal likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>join_units_2(
  snp.object,
  partition,
  threshold = 1e-05,
  n.cores = 1,
  comb.chache = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="join_units_2_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="join_units_2_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating an initial partition of the isolates.</p>
</td></tr>
<tr><td><code id="join_units_2_+3A_threshold">threshold</code></td>
<td>
<p>The increase in marginal log likelihood required to accept a move.</p>
</td></tr>
<tr><td><code id="join_units_2_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td></tr>
<tr><td><code id="join_units_2_+3A_comb.chache">comb.chache</code></td>
<td>
<p>a matrix recording previous marginal llks of combining clusters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The best partition after combining two clusters as well as
a boolean value indicating whether a move increased the marginal likelihood.
</p>

<hr>
<h2 id='load_fasta'>load_fasta</h2><span id='topic+load_fasta'></span>

<h3>Description</h3>

<p>Loads a fasta file into matrix format ready for
running the hierBAPS algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_fasta(msa, keep.singletons = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_fasta_+3A_msa">msa</code></td>
<td>
<p>Either the location of a fasta file or ape DNAbin object containing the multiple sequence alignment data to be clustered</p>
</td></tr>
<tr><td><code id="load_fasta_+3A_keep.singletons">keep.singletons</code></td>
<td>
<p>A logical indicating whether to consider singleton mutations in calculating the clusters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character matrix with filtered SNP data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>msa &lt;- system.file("extdata", "seqs.fa", package = "rhierbaps")
snp.matrix &lt;- load_fasta(msa)
</code></pre>

<hr>
<h2 id='log_stirling2'>log_stirling2</h2><span id='topic+log_stirling2'></span>

<h3>Description</h3>

<p>log_stirling2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_stirling2(n, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log_stirling2_+3A_n">n</code></td>
<td>
<p>number of objects</p>
</td></tr>
<tr><td><code id="log_stirling2_+3A_k">k</code></td>
<td>
<p>number of partitions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log of the Stirling number of the second kind
</p>

<hr>
<h2 id='model_search_parallel'>model_search_parallel</h2><span id='topic+model_search_parallel'></span>

<h3>Description</h3>

<p>Clusters DNA alignment using independent loci model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model_search_parallel(
  snp.object,
  partition,
  round.types,
  quiet = FALSE,
  n.extra.rounds = 0,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model_search_parallel_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="model_search_parallel_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating an initial starting partition.</p>
</td></tr>
<tr><td><code id="model_search_parallel_+3A_round.types">round.types</code></td>
<td>
<p>A vector indicating which series of moves to make.</p>
</td></tr>
<tr><td><code id="model_search_parallel_+3A_quiet">quiet</code></td>
<td>
<p>Whether to suppress progress information (default=FALSE).</p>
</td></tr>
<tr><td><code id="model_search_parallel_+3A_n.extra.rounds">n.extra.rounds</code></td>
<td>
<p>The number of additional rounds to perform after the default hierBAPS
settings (default=0). If set to Inf it will run until a local optimum is reached
(this might take a long time).</p>
</td></tr>
<tr><td><code id="model_search_parallel_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an optimised partition and marginal llk
</p>

<hr>
<h2 id='move_units_1'>move_units_1</h2><span id='topic+move_units_1'></span>

<h3>Description</h3>

<p>Peform an iteration of the first move in the algorithm. That is move units from one cluster to
another to improve the marginal likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>move_units_1(
  snp.object,
  partition,
  threshold = 1e-05,
  frac.clust.searched = 0.3,
  min.clust.size = 20,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="move_units_1_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="move_units_1_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating an initial partition of the isolates.</p>
</td></tr>
<tr><td><code id="move_units_1_+3A_threshold">threshold</code></td>
<td>
<p>The increase in marginal log likelihood required to accept a move.</p>
</td></tr>
<tr><td><code id="move_units_1_+3A_frac.clust.searched">frac.clust.searched</code></td>
<td>
<p>The percentage of a large cluster that will be moved.</p>
</td></tr>
<tr><td><code id="move_units_1_+3A_min.clust.size">min.clust.size</code></td>
<td>
<p>All isolates in clusters less than or equal to min.clus.size will be searched.</p>
</td></tr>
<tr><td><code id="move_units_1_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The best partition after moving units from one cluster to another as well as
a boolean value indicating whether a move increased the marginal likelihood.
</p>

<hr>
<h2 id='plot_sub_cluster'>plot_sub_cluster</h2><span id='topic+plot_sub_cluster'></span>

<h3>Description</h3>

<p>Creates a zoom plot using ggtree focusing on a cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_sub_cluster(hb.object, tree, level, sub.cluster)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_sub_cluster_+3A_hb.object">hb.object</code></td>
<td>
<p>The resulting object from running hierBAPS</p>
</td></tr>
<tr><td><code id="plot_sub_cluster_+3A_tree">tree</code></td>
<td>
<p>A phylo tree object to plot</p>
</td></tr>
<tr><td><code id="plot_sub_cluster_+3A_level">level</code></td>
<td>
<p>The level of the subcluster to be considered.</p>
</td></tr>
<tr><td><code id="plot_sub_cluster_+3A_sub.cluster">sub.cluster</code></td>
<td>
<p>An integer representing the subcluster to be considered.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
snp.matrix &lt;- load_fasta(system.file("extdata", "seqs.fa", package = "rhierbaps"))
newick.file.name &lt;- system.file("extdata", "seqs.fa.treefile", package = "rhierbaps")
tree &lt;- phytools::read.newick(newick.file.name)
hb.result &lt;- hierBAPS(snp.matrix, max.depth=2, n.pops=20)
plot_sub_cluster(hb.result, tree, level = 1, sub.cluster = 9)

</code></pre>

<hr>
<h2 id='preproc_alignment'>preproc_alignment</h2><span id='topic+preproc_alignment'></span>

<h3>Description</h3>

<p>Preprocessed the snp matrix for hierBAPS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preproc_alignment(snp.matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preproc_alignment_+3A_snp.matrix">snp.matrix</code></td>
<td>
<p>A matrix containing SNP data. Rows indicate isolates and columns loci.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an snp.object
</p>

<hr>
<h2 id='reallocate_units_4'>reallocate_units_4</h2><span id='topic+reallocate_units_4'></span>

<h3>Description</h3>

<p>Peform an iteration of the fourth move in the algorithm. That is split cluster into n
subclusters and re-allocate one sub-cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reallocate_units_4(
  snp.object,
  partition,
  threshold = 1e-05,
  min.clust.size = 20,
  split = FALSE,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reallocate_units_4_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="reallocate_units_4_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating an initial partition of the isolates.</p>
</td></tr>
<tr><td><code id="reallocate_units_4_+3A_threshold">threshold</code></td>
<td>
<p>The increase in marginal log likelihood required to accept a move.</p>
</td></tr>
<tr><td><code id="reallocate_units_4_+3A_min.clust.size">min.clust.size</code></td>
<td>
<p>Clusters smaller than min.clust.size will not be split.</p>
</td></tr>
<tr><td><code id="reallocate_units_4_+3A_split">split</code></td>
<td>
<p>Whether to split only into two clusters (for move type 3).</p>
</td></tr>
<tr><td><code id="reallocate_units_4_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The best partition after splitting a cluster and re-allocating as well as
a boolean value indicating whether a move increased the marginal likelihood.
</p>

<hr>
<h2 id='save_lml_logs'>save_lml_logs</h2><span id='topic+save_lml_logs'></span>

<h3>Description</h3>

<p>Saves the log marginal likelihoods to a text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_lml_logs(hb.object, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save_lml_logs_+3A_hb.object">hb.object</code></td>
<td>
<p>The resulting object from runnign hierBAPS</p>
</td></tr>
<tr><td><code id="save_lml_logs_+3A_file">file</code></td>
<td>
<p>The file you would like to save the log output to.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>snp.matrix &lt;- load_fasta(system.file("extdata", "small_seqs.fa", package = "rhierbaps"))
hb.result &lt;- hierBAPS(snp.matrix, max.depth=2, n.pops=20)
save_lml_logs(hb.result,  file.path(tempdir(), "output_file.txt"))

</code></pre>

<hr>
<h2 id='split_clusters_3'>split_clusters_3</h2><span id='topic+split_clusters_3'></span>

<h3>Description</h3>

<p>Peform an iteration of the third move in the algorithm. That is split cluster in two
and re-allocate one sub-cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_clusters_3(
  snp.object,
  partition,
  threshold = 1e-05,
  min.clust.size = 20,
  n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split_clusters_3_+3A_snp.object">snp.object</code></td>
<td>
<p>A snp.object containing the processed SNP data.</p>
</td></tr>
<tr><td><code id="split_clusters_3_+3A_partition">partition</code></td>
<td>
<p>An integer vector indicating an initial partition of the isolates.</p>
</td></tr>
<tr><td><code id="split_clusters_3_+3A_threshold">threshold</code></td>
<td>
<p>The increase in marginal log likelihood required to accept a move.</p>
</td></tr>
<tr><td><code id="split_clusters_3_+3A_min.clust.size">min.clust.size</code></td>
<td>
<p>Clusters smaller than min.clust.size will not be split.</p>
</td></tr>
<tr><td><code id="split_clusters_3_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The best partition after splitting a cluster and re-allocating as well as
a boolean value indicating whether a move increased the marginal likelihood.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
