<!DOCTYPE html><html lang="en"><head><title>Help for package RHybridFinder</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RHybridFinder}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checknetMHCpan'><p>checknetMHCpan</p></a></li>
<li><a href='#db_Human_Liver_AUTD17'><p>db_Human_Liver_AUTD17</p></a></li>
<li><a href='#db_rerun_Human_Liver_AUTD17'><p>db_rerun_Human_Liver_AUTD17</p></a></li>
<li><a href='#denovo_Human_Liver_AUTD17'><p>denovo_Human_Liver_AUTD17</p></a></li>
<li><a href='#export_checknetmhcpan_results'><p>export_checknetmhcpan_results</p></a></li>
<li><a href='#export_HybridFinder_results'><p>export_HybridFinder_results</p></a></li>
<li><a href='#export_step2_results'><p>export_step2_results</p></a></li>
<li><a href='#HybridFinder'><p>HybridFinder</p></a></li>
<li><a href='#mhc_check'><p>mhc_check</p></a></li>
<li><a href='#netmhcpan_list_alleles'><p>netmhcpan_list_alleles</p></a></li>
<li><a href='#step2_wo_netMHCpan'><p>step2_wo_netMHCpan</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of Hybrid Peptides in Immunopeptidomic Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Frederic Saab &lt;frederic.saab@umontreal.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tool for the analysis Mass Spectrometry (MS) data in the context of 
             immunopeptidomic analysis for the identification of hybrid 
             peptides and the predictions of binding affinity of all peptides 
             using 'netMHCpan' &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkaa379">doi:10.1093/nar/gkaa379</a>&gt; while providing a summary 
             of the netMHCpan output. 'RHybridFinder' (RHF) is destined for 
             researchers who are looking to analyze their MS data for the purpose 
             of identification of potential spliced peptides. This package, 
             developed mainly in base R, is based on the workflow published by 
             Faridi et al. in 2018 &lt;<a href="https://doi.org/10.1126%2Fsciimmunol.aar3947">doi:10.1126/sciimmunol.aar3947</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>doParallel, foreach, seqinr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>LazyData:</td>
<td>True</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>bzip2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-17 01:48:50 UTC; caron-fs</td>
</tr>
<tr>
<td>Author:</td>
<td>Frederic Saab [aut, cre],
  Peter Kubiniok [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-17 16:30:24 UTC</td>
</tr>
</table>
<hr>
<h2 id='checknetMHCpan'>checknetMHCpan</h2><span id='topic+checknetMHCpan'></span>

<h3>Description</h3>

<p>checknetMHCpan, utilizes the file from the second (PEAKS) run
and analyzes the data with netMHCpan in order to provide the peptide binding
affinity to different HLA/MHC alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checknetMHCpan(
  netmhcpan_directory,
  netmhcpan_alleles,
  peptide_rerun,
  HF_step1_output,
  export_files = FALSE,
  export_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checknetMHCpan_+3A_netmhcpan_directory">netmhcpan_directory</code></td>
<td>
<p>the directory in which the netMHCpan file is.</p>
</td></tr>
<tr><td><code id="checknetMHCpan_+3A_netmhcpan_alleles">netmhcpan_alleles</code></td>
<td>
<p>vector of comma-separated alleles for which these peptides
should be analyzed (i.e HLA_alleles_Exp1&lt;- c(&quot;HLA-A*02:01&quot;, &quot;HLA-A*03:01&quot;,
&quot;HLA-A24:02&quot;))</p>
</td></tr>
<tr><td><code id="checknetMHCpan_+3A_peptide_rerun">peptide_rerun</code></td>
<td>
<p>dataframe containing the results of the second run</p>
</td></tr>
<tr><td><code id="checknetMHCpan_+3A_hf_step1_output">HF_step1_output</code></td>
<td>
<p>the HybridFinder output containing the potential
splicing categorizations obtained with the HybridFinder function (HybridFinder)
based on the matching of fragment pairs of peptides in 1 or 2 proteins. This
parameter can be provided either by loading the .csv exported file, or if the
results #' object still is in the global environment (i.e results_HF_Exp1),
then it can be accessed by simply writing &quot;results_HF_Exp1[[1]]&quot;.</p>
</td></tr>
<tr><td><code id="checknetMHCpan_+3A_export_files">export_files</code></td>
<td>
<p>a boolean parameter for exporting the dataframes into
files in the next parameter for the output directory, Default: FALSE</p>
</td></tr>
<tr><td><code id="checknetMHCpan_+3A_export_dir">export_dir</code></td>
<td>
<p>export_dir the output directory for the results files
if export_files=TRUE, Default: NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ability to check the peptide binding affinity to the different
MHC/HLA molecules is essential for assessing the antigenicity of all peptides.
This function thus uses netMHCpan (Reynisson et al., 2020) for the generation
of binding affinty results.
</p>


<h3>Value</h3>


<ol>
<li><p> netMHCpan results pertaining to the binding affinity of all peptides
in the database search results (in long- and wide- format, with data tidying
in the wide format in order to compute the amount of HLA molecules to which
a peptide is strong/weak/non-binder binder)(dataframe)
</p>
</li>
<li><p> netMHCpan results pertaining to the binding affinity of the hybrid
peptides to the MHC molecules (in long- and wide- format, with data tidying
in the wide format in order to compute the amount of HLA molecules to which
a peptide is strong/weak/non-binder binder) (dataframe)
</p>
</li>
<li><p> the database search rerun with the categorizations already determined
in step1 (HybridFinder Function)</p>
</li></ol>
<p> (datafrane)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  results_checknetmhcpan_Exp1&lt;- checknetMHCpan('/usr/local/bin', alleles,
  peptide_rerun, Exp1_HF_results[[1]])
  results_checknetmhcpan_Exp1 &lt;- checknetMHCpan('/usr/local/bin', alleles,
  peptide_rerun, Exp1_HF_results_denovo_w_spliced)

## End(Not run)
</code></pre>

<hr>
<h2 id='db_Human_Liver_AUTD17'>db_Human_Liver_AUTD17</h2><span id='topic+db_Human_Liver_AUTD17'></span>

<h3>Description</h3>

<p>database search results from the first run using PEAKS software,
on the raw dataset from the HLA ligand Atlas Human Liver sample of AutDonor17
</p>


<h3>Usage</h3>

<pre><code class='language-R'>db_Human_Liver_AUTD17
</code></pre>


<h3>Format</h3>

<p>A data frame with 6108 rows and 18 variables:
</p>

<dl>
<dt><code>Peptide</code></dt><dd><p>character Peptide</p>
</dd>
<dt><code>X.10lgP</code></dt><dd><p>double X.10lgP</p>
</dd>
<dt><code>Mass</code></dt><dd><p>double Mass</p>
</dd>
<dt><code>Length</code></dt><dd><p>integer Peptide length</p>
</dd>
<dt><code>ppm</code></dt><dd><p>double ppm</p>
</dd>
<dt><code>m.z</code></dt><dd><p>double mass-to-charge</p>
</dd>
<dt><code>Z</code></dt><dd><p>integer charge</p>
</dd>
<dt><code>RT</code></dt><dd><p>double Retention Time</p>
</dd>
<dt><code>Area</code></dt><dd><p>double Area</p>
</dd>
<dt><code>Fraction</code></dt><dd><p>integer Fraction</p>
</dd>
<dt><code>Id</code></dt><dd><p>integer Id</p>
</dd>
<dt><code>Scan</code></dt><dd><p>character Scan</p>
</dd>
<dt><code>from.Chimera</code></dt><dd><p>character from.Chimera</p>
</dd>
<dt><code>Source.File</code></dt><dd><p>character Source.File</p>
</dd>
<dt><code>Accession</code></dt><dd><p>character Accession</p>
</dd>
<dt><code>PTM</code></dt><dd><p>character Post Translational Modification</p>
</dd>
<dt><code>AScore</code></dt><dd><p>character AScore</p>
</dd>
<dt><code>Found.By</code></dt><dd><p>character Found.By</p>
</dd>
</dl>



<h3>Details</h3>

<p>database search results from the first run using PEAKS software,
on the raw dataset from the HLA ligand Atlas Human Liver sample of AutDonor17
</p>

<hr>
<h2 id='db_rerun_Human_Liver_AUTD17'>db_rerun_Human_Liver_AUTD17</h2><span id='topic+db_rerun_Human_Liver_AUTD17'></span>

<h3>Description</h3>

<p>database search results from the second run using PEAKS software,
using the raw file from HLA ligand ATLAS, Human Liver AutDonor 17
</p>


<h3>Usage</h3>

<pre><code class='language-R'>db_rerun_Human_Liver_AUTD17
</code></pre>


<h3>Format</h3>

<p>A data frame with 6315 rows and 18 variables:
</p>

<dl>
<dt><code>Peptide</code></dt><dd><p>character Peptide</p>
</dd>
<dt><code>X.10lgP</code></dt><dd><p>double X.10lgP</p>
</dd>
<dt><code>Mass</code></dt><dd><p>double Mass</p>
</dd>
<dt><code>Length</code></dt><dd><p>integer Peptide sequence</p>
</dd>
<dt><code>ppm</code></dt><dd><p>double ppm</p>
</dd>
<dt><code>m.z</code></dt><dd><p>double mass-to-charge</p>
</dd>
<dt><code>Z</code></dt><dd><p>integer charge</p>
</dd>
<dt><code>RT</code></dt><dd><p>double Retention Time</p>
</dd>
<dt><code>Area</code></dt><dd><p>double Area</p>
</dd>
<dt><code>Fraction</code></dt><dd><p>integer Fraction</p>
</dd>
<dt><code>Id</code></dt><dd><p>integer Id</p>
</dd>
<dt><code>Scan</code></dt><dd><p>character Scan</p>
</dd>
<dt><code>from.Chimera</code></dt><dd><p>character from.Chimera</p>
</dd>
<dt><code>Source.File</code></dt><dd><p>character Source.File</p>
</dd>
<dt><code>Accession</code></dt><dd><p>character Accession</p>
</dd>
<dt><code>PTM</code></dt><dd><p>character Post-Translational Modification</p>
</dd>
<dt><code>AScore</code></dt><dd><p>character AScore</p>
</dd>
<dt><code>Found.By</code></dt><dd><p>character Found.By</p>
</dd>
</dl>



<h3>Details</h3>

<p>database search results from the second run using PEAKS software,
using the raw file from HLA ligand ATLAS, Human Liver AutDonor 17
</p>

<hr>
<h2 id='denovo_Human_Liver_AUTD17'>denovo_Human_Liver_AUTD17</h2><span id='topic+denovo_Human_Liver_AUTD17'></span>

<h3>Description</h3>

<p>denovo sequencing results obtained using PEAKS software, and the
raw file of Human liver from autDonor 17
</p>


<h3>Usage</h3>

<pre><code class='language-R'>denovo_Human_Liver_AUTD17
</code></pre>


<h3>Format</h3>

<p>A data frame with 50114 rows and 18 variables:
</p>

<dl>
<dt><code>Fraction</code></dt><dd><p>integer Fraction</p>
</dd>
<dt><code>Source.File</code></dt><dd><p>character Source.File</p>
</dd>
<dt><code>Feature</code></dt><dd><p>character Feature</p>
</dd>
<dt><code>Peptide</code></dt><dd><p>character Peptide</p>
</dd>
<dt><code>Scan</code></dt><dd><p>character Scan</p>
</dd>
<dt><code>Tag.length</code></dt><dd><p>integer Tag.length</p>
</dd>
<dt><code>Denovo.score</code></dt><dd><p>integer Denovo.score</p>
</dd>
<dt><code>ALC....</code></dt><dd><p>integer Average Local Confidence</p>
</dd>
<dt><code>Length</code></dt><dd><p>integer peptide length</p>
</dd>
<dt><code>m.z</code></dt><dd><p>double mass-to-charge(m/z)</p>
</dd>
<dt><code>z</code></dt><dd><p>integer charge</p>
</dd>
<dt><code>RT</code></dt><dd><p>double Retention Time</p>
</dd>
<dt><code>Predict.RT</code></dt><dd><p>character Predict.RT</p>
</dd>
<dt><code>Mass</code></dt><dd><p>double Mass</p>
</dd>
<dt><code>ppm</code></dt><dd><p>double ppm</p>
</dd>
<dt><code>local.confidence....</code></dt><dd><p>character confidence score per residue</p>
</dd>
<dt><code>tag....0..</code></dt><dd><p>character tag</p>
</dd>
<dt><code>mode</code></dt><dd><p>character mode</p>
</dd>
</dl>



<h3>Details</h3>

<p>denovo sequencing results obtained using PEAKS software, and the
raw file of Human liver from autDonor 17
</p>

<hr>
<h2 id='export_checknetmhcpan_results'>export_checknetmhcpan_results</h2><span id='topic+export_checknetmhcpan_results'></span>

<h3>Description</h3>

<p>this function allows to export the results generated from
checknetMHCpan()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_checknetmhcpan_results(list_checknetMHCpan_results, export_dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_checknetmhcpan_results_+3A_list_checknetmhcpan_results">list_checknetMHCpan_results</code></td>
<td>
<p>the results generated from running
checknetMHCpan()</p>
</td></tr>
<tr><td><code id="export_checknetmhcpan_results_+3A_export_dir">export_dir</code></td>
<td>
<p>the export directory where the results .csv files should
be exported.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to be able to have the checknetMHCpan() function results
exported, this function will come in handy. Please note that this function is
also part of the checknetMHCpan() function (if export_files is set to TRUE
and a valid export directory is indicated)
</p>


<h3>Value</h3>

<p>exports a folder containing three files
</p>

<ol>
<li><p> netMHCpan results in long format (the original output)(.csv file)
</p>
</li>
<li><p> netMHCpan results tidied (in wide format) so as to summarize
the information per peptide (.tsv tab-separated file)
</p>
</li>
<li><p> the updated database search results which contain the categorizatiosn
of the peptides found in common between the 2nd database search and
the HybridFinder function (.csv file)</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 export_checknetmhcpan_results(results_checknetMHCpan_Human_Liver_AUTD17, folder_Human_Liver_AUTD17)

## End(Not run)
</code></pre>

<hr>
<h2 id='export_HybridFinder_results'>export_HybridFinder_results</h2><span id='topic+export_HybridFinder_results'></span>

<h3>Description</h3>

<p>this function allows to export the results list obtained in the
HybridFinder() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_HybridFinder_results(results_list, export_dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_HybridFinder_results_+3A_results_list">results_list</code></td>
<td>
<p>the results list obtained with the HybridFinder() function.</p>
</td></tr>
<tr><td><code id="export_HybridFinder_results_+3A_export_dir">export_dir</code></td>
<td>
<p>the export directory</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to be able to have the HybridFinder() results list exported,
this function will come in handy. Please note that this function is also part
of the HybridFinder() function, therefore if you set export_files=TRUE and you
indicate the export directory in export_dir in the HybridFinder() function,
you would have the exact same outcome.
</p>


<h3>Value</h3>

<p>exports a folder containing three files
</p>

<ol>
<li><p> the HybridFinder output - the spectra that made it to the end
with their respective columns (ALC, m/z, RT, Fraction, Scan) and a
categorization column which denotes their potential splice type
(-cis, -trans) or whether they are linear (the entire sequence was
matched in proteins in the proteome database). Potential cis- &amp;
trans-spliced peptide are peptides whose fragments were matched with
fragments within one protein, or two proteins, respectively.
</p>
</li>
<li><p> list of potential hybrid peptides (excluding the linear peptides)
(.csv file)
</p>
</li>
<li><p> the merged proteome consisting of the reference proteome along
with the hybrid proteome added at the end of the file with the sequence
names following the pattern &quot;sp|denovo_HF_fake_protein&quot; along with
a digit at the end (1,2,3,4,4,etc.) (.fasta file)</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="seqinr.html#topic+write.fasta">write.fasta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 export_results(results_HybridFinder_Human_Liver_AUTD17,folder_Human_Liver_AUTD17)

## End(Not run)
</code></pre>

<hr>
<h2 id='export_step2_results'>export_step2_results</h2><span id='topic+export_step2_results'></span>

<h3>Description</h3>

<p>this function allows to export the results generated from
step2_wo_netmhcpan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_step2_results(step2_RHF_results_Exp1, export_dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_step2_results_+3A_step2_rhf_results_exp1">step2_RHF_results_Exp1</code></td>
<td>
<p>the results generated from running
step2_wo_netMHCpan()</p>
</td></tr>
<tr><td><code id="export_step2_results_+3A_export_dir">export_dir</code></td>
<td>
<p>the export directory where you would like to have the .csv
file saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since netMHCpan is not compatible with Windows OS, the package offers
an alternative by outputting the input for netMHCpan and as well the database
results with their respective categorizations (cis, trans) established in step1.
</p>


<h3>Value</h3>

<p>exports a folder containing 2 files
</p>

<ol>
<li><p> the peptide list to be entered in a netMHCpan-ready format,(.csv)
</p>
</li>
<li><p> the updated database search results which contain the categorizatiosn
of the peptides found in common between the 2nd database search and
the HybridFinder function (.csv file)</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 export_step2_results(results_step2_Human_Liver_AUTD17, folder_Human_Liver_AUTD17)

## End(Not run)
</code></pre>

<hr>
<h2 id='HybridFinder'>HybridFinder</h2><span id='topic+HybridFinder'></span>

<h3>Description</h3>

<p>This function takes in three mandatory inputs: (1) all denovo
candidates (2) database search results and (3) the corresponding proteome
fasta file. The function's role is to extract high confidence de novo peptides
and to search for their existence in the proteome, whether the entire peptide
sequence or its pair fragments (in one or two proteins).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HybridFinder(
  denovo_candidates,
  db_search,
  proteome_db,
  customALCcutoff = NULL,
  with_parallel = TRUE,
  customCores = 6,
  export_files = FALSE,
  export_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HybridFinder_+3A_denovo_candidates">denovo_candidates</code></td>
<td>
<p>dataframe containing all denovo candidate peptides</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_db_search">db_search</code></td>
<td>
<p>dataframe containing the database search peptides</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_proteome_db">proteome_db</code></td>
<td>
<p>path to the proteome FASTA file</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_customalccutoff">customALCcutoff</code></td>
<td>
<p>the default is calculated based on the median ALC of the
assigned spectrum groups (spectrum groups that match in the database search
results and in the denovo sequencing results) where also the peptide sequence
matches, Default: NULL</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_with_parallel">with_parallel</code></td>
<td>
<p>for faster results, this function also utilizes
parallel computing (please read more on parallel computing in order
to be sure that your computer does support this), Default: TRUE</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_customcores">customCores</code></td>
<td>
<p>custom amount of cores strictly higher than 5, Default: 6</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_export_files">export_files</code></td>
<td>
<p>a boolean parameter for exporting the dataframes into
files in the next parameter for the output directory, Default: FALSE,
Default: FALSE</p>
</td></tr>
<tr><td><code id="HybridFinder_+3A_export_dir">export_dir</code></td>
<td>
<p>the output directory for the results files if
export_files=TRUE, Default: NULL, Default: NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is based on the published algorithm by Faridi et al.
(2018) for the identification and categorization of hybrid peptides. The function
described here adopts a slightly modified version of the algorithm for
computational efficiency. The function starts by extracting unassigned denovo
spectra where the Average Local Confidence (assigned by PEAKS software), is
equivalent to the ALC cutoff which is based on the median of the assigned spectra
(between denovo and database search). The sequences of all peptides are searched
against the reference proteome. If there is a hit then, then, the peptide sequence
within a spectrum group considered as being linear and each spectrum group is
is then filtered so as to keep the highest ALC-ranking spectra. Then, the rest
of the spectra (spectra that did not contain any sequence that had an entire
match in the proteome database) then undergo a &quot;cutting&quot; procedure where each
sequence yields n-2 sequences (with n being the length of the peptide. That is
if the peptide contains 9 amino acids i.e NTYASPRFK, then the sequence is cut
into a combination of 7 sequences of 2 fragment pairs each i.e fragment 1: NTY
and fragment 2: ASPRFK, etc).These are then searched in the proteome for hits
of both peptide fragments within a same protein, spectra in which sequences
have fragment pairs that match within a same protein, these are considerent
to be potentially cis-spliced. Potentially cis-spliced spectrum groups are then
filtered based on the highest ranking ALC. Spectrum groups not considered to be
potentially cis-spliced are further checked for potential trans-splicing. The
peptide sequences are cut again in the same fashion, however, this time peptide
fragment pairs are searched for matches in two proteins. Peptide sequences whose
fragment pairs match in 2 proteins are considerend to be potentially trans-spliced.
The same filtering for the highest ranking ALC within each peptide spectrum group.
The remaining spectra that were neither assigned as linear nor potentially spliced
(neither cis- nor trans-) are then discarded. The result is a list of spectra
along with their categorizations (Linear, potentially cis- and potentially trans-)
Potentially cis- and trans-spliced peptides are then concatenated and then broken into
several &quot;fake&quot; proteins and added to the bottom of the reference proteome. The
point of this last step is to create a merged proteome (consisting of the reference
proteome and the hybrid proteome) which  would be used for a second database
search. After the second database search the checknetmhcpan function or the
step2_wo_netMHCpan function can be used in order to obtain the final list
of potentially spliced peptides.
Article: Faridi P, Li C, Ramarathinam SH, Vivian JP, Illing PT, Mifsud NA,
Ayala R, Song J, Gearing LJ, Hertzog PJ, Ternette N, Rossjohn J, Croft NP,
Purcell AW. A subset of HLA-I peptides are not genomically templated: Evidence
for cis- and trans-spliced peptide ligands. Sci Immunol. 2018 Oct 12;3(28):eaar3947.
&lt;doi: 10.1126/sciimmunol.aar3947&gt;. PMID: 30315122.
</p>


<h3>Value</h3>

<p>The output is a list of 3 dataframes containing:
</p>

<ol>
<li><p> the HybridFinder output (dataframe) - the spectra that made
it to the end with their respective columns (ALC, m/z, RT, Fraction,
Scan) and a categorization column which denotes their potential splice
type (-cis, -trans) or whether they are linear (the entire sequence
was matched in proteins in the proteome database). Potential cis- &amp;
trans-spliced peptide are peptides whose fragments were matched with
fragments within one protein, or two proteins, respectively.
</p>
</li>
<li><p> character vector containing potentially hybrid peptides (cis-
and trans-)
</p>
</li>
<li><p> list containing the reference proteome and the &quot;fake&quot; proteins
added at the end with a patterned naming convention (sp|denovo_HF_fake_protein)
made up of the concatenated potential hybrid peptides.</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="seqinr.html#topic+read.fasta">read.fasta</a></code>,<code><a href="seqinr.html#topic+s2c">s2c</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 hybridFinderResult_list &lt;- HybridFinder(denovo_candidates, db_search,
 proteome, export = TRUE, output_dir)
 hybridFinderResult_list &lt;- HybridFinder(denovo_candidates, db_search,
 proteome)
 hybridFinderResult_list &lt;- HybridFinder(denovo_candidates, db_search,
 proteome, export = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='mhc_check'>mhc_check</h2><span id='topic+mhc_check'></span>

<h3>Description</h3>

<p>this function only contains the alleles list, read by netMHCpan,
the list was retrieved by reading the file exported from netMHCpan, using the
following command line &quot;netMHCpan -listMHC&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mhc_check(netmhcpan_alleles)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mhc_check_+3A_netmhcpan_alleles">netmhcpan_alleles</code></td>
<td>
<p>the netmhcpan alleles to be used for the netmhcpan call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>a custom error is printed in case the allele is not written correctly
</p>


<h3>Value</h3>


<ul>
<li><p> returns a custom error message if MHC/HLA allele(s) are not
written correctly
</p>
</li>
<li><p> returns nothing if there are no issues. If HLA alleles are not
written correctly</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
 mhc_check("HLA-A02:01")
 mhc_check("HLA-A0201")
}
</code></pre>

<hr>
<h2 id='netmhcpan_list_alleles'>netmhcpan_list_alleles</h2><span id='topic+netmhcpan_list_alleles'></span>

<h3>Description</h3>

<p>the list of alleles in the acceptable format for netMHCpan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netmhcpan_list_alleles
</code></pre>


<h3>Format</h3>

<p>A data frame with 1024 rows and 1 variables:
</p>

<dl>
<dt><code>V1</code></dt><dd><p>character the alleles</p>
</dd>
</dl>



<h3>Details</h3>

<p>the list of alleles in the acceptable format for netMHCpan
</p>

<hr>
<h2 id='step2_wo_netMHCpan'>step2_wo_netMHCpan</h2><span id='topic+step2_wo_netMHCpan'></span>

<h3>Description</h3>

<p>This function helps retrieve the categorizations for the peptides
from step 1 and apply them to those that are matched in the second database
search.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step2_wo_netMHCpan(
  peptide_rerun,
  HF_step1_output,
  export_files = FALSE,
  export_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="step2_wo_netMHCpan_+3A_peptide_rerun">peptide_rerun</code></td>
<td>
<p>dataframe containing the results of the second database
PEAKS search.</p>
</td></tr>
<tr><td><code id="step2_wo_netMHCpan_+3A_hf_step1_output">HF_step1_output</code></td>
<td>
<p>the HybridFinder output containing the potential
splicing categorizations obtained with the HybridFinder function (HybridFinder)
based on the matching of fragment pairs of peptides in 1 or 2 proteins. This
parameter can be provided either by loading the .csv exported file, or if the
results #' object still is in the global environment (i.e results_HF_Exp1),
then it can be accessed by simply writing &quot;results_HF_Exp1[[1]]&quot;.</p>
</td></tr>
<tr><td><code id="step2_wo_netMHCpan_+3A_export_files">export_files</code></td>
<td>
<p>a boolean parameter for exporting the dataframes into
files in the next parameter for the output directory, Default: FALSE</p>
</td></tr>
<tr><td><code id="step2_wo_netMHCpan_+3A_export_dir">export_dir</code></td>
<td>
<p>export_dir the output directory for the results files
if export_files=TRUE, Default: NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In special cases where the PC runs on windows OS, since it would only
be possible to use the web version of netMHCpan, this function returns the
peptide input file for the webversion of netMHCpan. Also, this function outputs
the database search rerun results with their categorizations (into potentially
cis and potentially trans) obtained from the first step (HybridFinder).
</p>


<h3>Value</h3>


<ol>
<li><p> the input file for the web version of netMHCpan (dataframe)
</p>
</li>
<li><p> the database search rerun with the categorizations already determined
in the previous step. (character vector)</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
data(package="RHybridFinder", "db_rerun_Human_Liver_AUTD17")
results_checknetmhcpan_Human_Liver_AUTD17&lt;- step2_wo_netMHCpan(db_rerun_Human_Liver_AUTD17,
    results_HybridFinder_Human_Liver_AUTD17[[1]])
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
