<!DOCTYPE html><html><head><title>Help for package RIdeogram</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RIdeogram}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#convertSVG'><p>convertSVG</p></a></li>
<li><a href='#Fst_between_CE_and_CW'><p>Fst between two Liriodendron groups</p></a></li>
<li><a href='#gene_density'><p>Gene distribution across the human genome</p></a></li>
<li><a href='#GFFex'><p>GFFex</p></a></li>
<li><a href='#human_karyotype'><p>Karyotype information of the human genome</p></a></li>
<li><a href='#ideogram'><p>ideogram</p></a></li>
<li><a href='#karyotype_dual_comparison'><p>Karyotype for two genome comparison</p></a></li>
<li><a href='#karyotype_ternary_comparison'><p>Karyotype for three genome comparison</p></a></li>
<li><a href='#liriodendron_karyotype'><p>Karyotype information of the Liriodendron genome</p></a></li>
<li><a href='#LTR_density'><p>LTR distribution across the human genome</p></a></li>
<li><a href='#Pi_for_CE'><p>Pi of one Liriodendron group</p></a></li>
<li><a href='#Pi_for_CE_and_CW'><p>Pi of two Liriodendron groups</p></a></li>
<li><a href='#Random_RNAs_500'><p>500 RNAs' position</p></a></li>
<li><a href='#synteny_dual_comparison'><p>Synteny for two genome comparison</p></a></li>
<li><a href='#synteny_ternary_comparison'><p>Synteny for three genome comparison</p></a></li>
<li><a href='#synteny_ternary_comparison_graident'><p>Synteny for three genome comparison with gradient fill</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Drawing SVG Graphics to Visualize and Map Genome-Wide Data on
Idiograms</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhaodong Hao &lt;haozd1992@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>For whole-genome analysis, idiograms are virtually the most intuitive and effective way to map and visualize the genome-wide information. RIdeogram was developed to visualize and map whole-genome data on idiograms with no restriction of species.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, grDevices, grImport2, rsvg, scales, tools, tidyr,
utils</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-20 10:43:48 UTC; haozhaodong</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhaodong Hao [aut, cre],
  Dekang Lv [aut],
  Ying Ge [aut],
  Jisen Shi [aut],
  Weijers Dolf [aut],
  Guangchuang Yu [aut],
  Jinhui Chen [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-20 11:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='convertSVG'>convertSVG</h2><span id='topic+convertSVG'></span><span id='topic+svg2pdf'></span><span id='topic+svg2png'></span><span id='topic+svg2tiff'></span><span id='topic+svg2jpg'></span>

<h3>Description</h3>

<p>convert svg to png or other format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertSVG(svg, file = "chromosome", device = NULL, width = 8.2677,
  height = 11.6929, dpi = 300)

svg2pdf(svg, file = "chromosome", width = 8.2677, height = 11.6929,
  dpi = 300)

svg2png(svg, file = "chromosome", width = 8.2677, height = 11.6929,
  dpi = 300)

svg2tiff(svg, file = "chromosome", width = 8.2677, height = 11.6929,
  dpi = 300)

svg2jpg(svg, file = "chromosome", width = 8.2677, height = 11.6929,
  dpi = 300)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convertSVG_+3A_svg">svg</code></td>
<td>
<p>svg file</p>
</td></tr>
<tr><td><code id="convertSVG_+3A_file">file</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="convertSVG_+3A_device">device</code></td>
<td>
<p>target format</p>
</td></tr>
<tr><td><code id="convertSVG_+3A_width">width</code></td>
<td>
<p>output width</p>
</td></tr>
<tr><td><code id="convertSVG_+3A_height">height</code></td>
<td>
<p>output height</p>
</td></tr>
<tr><td><code id="convertSVG_+3A_dpi">dpi</code></td>
<td>
<p>output dpi</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisible grob object
</p>


<h3>Author(s)</h3>

<p>Zhaodong Hao, Dekang Lv, Ying Ge, Jisen Shi, Dolf Weijers, Guangchuang Yu, Jinhui Chen
</p>

<hr>
<h2 id='Fst_between_CE_and_CW'>Fst between two Liriodendron groups</h2><span id='topic+Fst_between_CE_and_CW'></span>

<h3>Description</h3>

<p>Fst values between China east and west groups of Liriodendron chinense and being calculated in a 2-Mb sliding window with a
1-Mb step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fst_between_CE_and_CW)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Nature Plants, doi: 10.1038/s41477-018-0323-6</p>

<hr>
<h2 id='gene_density'>Gene distribution across the human genome</h2><span id='topic+gene_density'></span>

<h3>Description</h3>

<p>Gene numbers was counted with a window of 1 Mb
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_density)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Gencode (https://www.gencodegenes.org/)</p>

<hr>
<h2 id='GFFex'>GFFex</h2><span id='topic+GFFex'></span>

<h3>Description</h3>

<p>extract some specific feature information from a gff file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GFFex(input, karyotype, feature = "gene", window = 1e+06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GFFex_+3A_input">input</code></td>
<td>
<p>gff file</p>
</td></tr>
<tr><td><code id="GFFex_+3A_karyotype">karyotype</code></td>
<td>
<p>karyotype file</p>
</td></tr>
<tr><td><code id="GFFex_+3A_feature">feature</code></td>
<td>
<p>feature name</p>
</td></tr>
<tr><td><code id="GFFex_+3A_window">window</code></td>
<td>
<p>window size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe
</p>


<h3>Author(s)</h3>

<p>Zhaodong Hao, Dekang Lv, Ying Ge, Jisen Shi, Dolf Weijers, Guangchuang Yu, Jinhui Chen
</p>

<hr>
<h2 id='human_karyotype'>Karyotype information of the human genome</h2><span id='topic+human_karyotype'></span>

<h3>Description</h3>

<p>The version of this geome is gencode.v29.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(human_karyotype)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Gencode (https://www.gencodegenes.org/)</p>

<hr>
<h2 id='ideogram'>ideogram</h2><span id='topic+ideogram'></span>

<h3>Description</h3>

<p>ideogram with overlaid heatmap annotation and optional track label
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ideogram(karyotype, overlaid = NULL, label = NULL, label_type = NULL,
  synteny = NULL, colorset1 = c("#4575b4", "#ffffbf", "#d73027"),
  colorset2 = c("#b35806", "#f7f7f7", "#542788"), width = 170,
  Lx = 160, Ly = 35, output = "chromosome.svg")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ideogram_+3A_karyotype">karyotype</code></td>
<td>
<p>karyotype data</p>
</td></tr>
<tr><td><code id="ideogram_+3A_overlaid">overlaid</code></td>
<td>
<p>overlaid data</p>
</td></tr>
<tr><td><code id="ideogram_+3A_label">label</code></td>
<td>
<p>track label data</p>
</td></tr>
<tr><td><code id="ideogram_+3A_label_type">label_type</code></td>
<td>
<p>track label type, only support four types: marker, heatmap, line and polygon</p>
</td></tr>
<tr><td><code id="ideogram_+3A_synteny">synteny</code></td>
<td>
<p>synteny data</p>
</td></tr>
<tr><td><code id="ideogram_+3A_colorset1">colorset1</code></td>
<td>
<p>overlaid heatmap color</p>
</td></tr>
<tr><td><code id="ideogram_+3A_colorset2">colorset2</code></td>
<td>
<p>label heatmap color</p>
</td></tr>
<tr><td><code id="ideogram_+3A_width">width</code></td>
<td>
<p>width of plot region</p>
</td></tr>
<tr><td><code id="ideogram_+3A_lx">Lx</code></td>
<td>
<p>position of legend (x)</p>
</td></tr>
<tr><td><code id="ideogram_+3A_ly">Ly</code></td>
<td>
<p>position of legend (y)</p>
</td></tr>
<tr><td><code id="ideogram_+3A_output">output</code></td>
<td>
<p>output file, only svg is supported</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output file
</p>


<h3>Author(s)</h3>

<p>Zhaodong Hao, Dekang Lv, Ying Ge, Jisen Shi, Dolf Weijers, Guangchuang Yu, Jinhui Chen
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Loading the package
require(RIdeogram)

# Loading the testing data
data(human_karyotype, package="RIdeogram")
data(gene_density, package="RIdeogram")
data(Random_RNAs_500, package="RIdeogram")

# Checking the data format
head(human_karyotype)
head(gene_density)
head(Random_RNAs_500)

# Running the function (Remove "#" before you run the following codes)
# ideogram(karyotype = human_karyotype)
# convertSVG("chromosome.svg", device = "png")

# Then, you will find a SVG file and a PNG file in your Working Directory.

</code></pre>

<hr>
<h2 id='karyotype_dual_comparison'>Karyotype for two genome comparison</h2><span id='topic+karyotype_dual_comparison'></span>

<h3>Description</h3>

<p>Grape and Populus genomes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(karyotype_dual_comparison)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>MCscan</p>

<hr>
<h2 id='karyotype_ternary_comparison'>Karyotype for three genome comparison</h2><span id='topic+karyotype_ternary_comparison'></span>

<h3>Description</h3>

<p>Amborella, Grape and Liriodendron genomes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(karyotype_ternary_comparison)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>MCscan</p>

<hr>
<h2 id='liriodendron_karyotype'>Karyotype information of the Liriodendron genome</h2><span id='topic+liriodendron_karyotype'></span>

<h3>Description</h3>

<p>Liriodendron chinense genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(liriodendron_karyotype)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Nature Plants, doi: 10.1038/s41477-018-0323-6</p>

<hr>
<h2 id='LTR_density'>LTR distribution across the human genome</h2><span id='topic+LTR_density'></span>

<h3>Description</h3>

<p>LTR numbers was counted with a window of 1 Mb
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LTR_density)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>UCSC (http://genome.ucsc.edu/index.html)</p>

<hr>
<h2 id='Pi_for_CE'>Pi of one Liriodendron group</h2><span id='topic+Pi_for_CE'></span>

<h3>Description</h3>

<p>Pi values of the China east group of Liriodendron chinense and being calculated in a 2-Mb sliding window with a
1-Mb step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Pi_for_CE)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Nature Plants, doi: 10.1038/s41477-018-0323-6</p>

<hr>
<h2 id='Pi_for_CE_and_CW'>Pi of two Liriodendron groups</h2><span id='topic+Pi_for_CE_and_CW'></span>

<h3>Description</h3>

<p>Pi values of the China east and west groups of Liriodendron chinense and being calculated in a 2-Mb sliding window with a
1-Mb step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Pi_for_CE_and_CW)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Nature Plants, doi: 10.1038/s41477-018-0323-6</p>

<hr>
<h2 id='Random_RNAs_500'>500 RNAs' position</h2><span id='topic+Random_RNAs_500'></span>

<h3>Description</h3>

<p>500 RNAs randomly selected from all tRNAs, rRNAs and miRNA in the human genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Random_RNAs_500)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>Gencode (https://www.gencodegenes.org/)</p>

<hr>
<h2 id='synteny_dual_comparison'>Synteny for two genome comparison</h2><span id='topic+synteny_dual_comparison'></span>

<h3>Description</h3>

<p>Genome Synteny between Grape and Populus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(synteny_dual_comparison)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>MCscan</p>

<hr>
<h2 id='synteny_ternary_comparison'>Synteny for three genome comparison</h2><span id='topic+synteny_ternary_comparison'></span>

<h3>Description</h3>

<p>Genome Synteny among Amborella, Grape and Liriodendron
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(synteny_ternary_comparison)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>MCscan</p>

<hr>
<h2 id='synteny_ternary_comparison_graident'>Synteny for three genome comparison with gradient fill</h2><span id='topic+synteny_ternary_comparison_graident'></span>

<h3>Description</h3>

<p>Genome Synteny among Amborella, Grape and Liriodendron with gradient fill
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(synteny_ternary_comparison_graident)</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>MCscan</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
