<!DOCTYPE html><html><head><title>Help for package ridge</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ridge}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#computeRidgeLogistic'>
<p>Internal functions for logistic ridge regression.</p></a></li>
<li><a href='#GenBin'>
<p>Simulated genetic data with a binary phenotypes</p></a></li>
<li><a href='#GenCont'>
<p>Simulated genetic data with continuous outcomes</p></a></li>
<li><a href='#Gorman'>
<p>The Ten-Factor data first described by Gorman and Toman (1966).</p></a></li>
<li><a href='#Hald'>
<p>Hald data</p></a></li>
<li><a href='#linearRidge'>
<p>Linear ridge regression.</p></a></li>
<li><a href='#linearRidgeGenotypes'>
<p>Fits linear ridge regression models for genome-wide SNP data.</p></a></li>
<li><a href='#linearRidgeGenotypesPredict'>
<p>Predict phenotypes from genome-wide SNP data based on a file of coefficients</p></a></li>
<li><a href='#logisticRidge'>
<p>Logistic ridge regression.</p></a></li>
<li><a href='#logisticRidgeGenotypes'>
<p>Fits logistic ridge regression models for genomoe-wide SNP data.</p></a></li>
<li><a href='#logisticRidgeGenotypesPredict'>
<p>Predict fitted probabilities from genome-wide SNP data based on a file of coefficients</p></a></li>
<li><a href='#pvals'>
<p>Compute p-values for ridgeLinear and ridgeLogistic models</p></a></li>
<li><a href='#ridge'><p>ridge: Linear and logistic ridge regression functions.</p></a></li>
<li><a href='#ridge-package'><p>ridge-package description</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Ridge Regression with Automatic Selection of the Penalty
Parameter</td>
</tr>
<tr>
<td>Description:</td>
<td>Linear and logistic ridge regression functions. Additionally includes special functions for 
            genome-wide single-nucleotide polymorphism (SNP) data. More details can be found in
            &lt;<a href="https://doi.org/10.1002%2Fgepi.21750">doi:10.1002/gepi.21750</a>&gt; and &lt;<a href="https://doi.org/10.1186%2F1471-2105-12-372">doi:10.1186/1471-2105-12-372</a>&gt;.</td>
</tr>
<tr>
<td>Version:</td>
<td>3.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Steffen Moritz <a href="https://orcid.org/0000-0002-0085-1804"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Erika Cule [aut],
  Dan Frankowski [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Steffen Moritz &lt;steffen.moritz10@gmail.com&gt;</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/SteffenMoritz/ridge/issues">https://github.com/SteffenMoritz/ridge/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SteffenMoritz/ridge">https://github.com/SteffenMoritz/ridge</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, graphics, grDevices, utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Gnu Scientific Library version &gt;= 1.14</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, datasets, covr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-11 10:44:36 UTC; Steve</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-11 14:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='computeRidgeLogistic'>
Internal functions for logistic ridge regression.
</h2><span id='topic+computeRidgeLogistic'></span><span id='topic+updateBeta'></span><span id='topic+objectiveFunction'></span>

<h3>Description</h3>

<p>Internal functions for logisitc ridge regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeRidgeLogistic(X, y, k, intercept = TRUE, doff = FALSE)
updateBeta(B, X, y, k, intercept = TRUE, doff = FALSE) 
objectiveFunction(B, X, y, k, intercept = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computeRidgeLogistic_+3A_x">X</code></td>
<td>

<p>Matrix of predictors.
</p>
</td></tr>
<tr><td><code id="computeRidgeLogistic_+3A_y">y</code></td>
<td>

<p>vector of outcomes.
</p>
</td></tr>
<tr><td><code id="computeRidgeLogistic_+3A_k">k</code></td>
<td>

<p>ridge regression parameter.
</p>
</td></tr>
<tr><td><code id="computeRidgeLogistic_+3A_intercept">intercept</code></td>
<td>

<p>does the model have an intercept?
</p>
</td></tr>
<tr><td><code id="computeRidgeLogistic_+3A_doff">doff</code></td>
<td>

<p>should degrees of freedom of the model be computed?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are called in the function <code>logisticRidge</code>.
They are not for calling directly by the user.
</p>


<h3>Value</h3>

<p><code>computeRidgeLogistic</code> returns the fitted logistic ridge regression coefficients. If <code>doff = TRUE</code> 
it also returns the degrees of freedom of the model and the degrees of freedom for variance.
<code>updateBeta</code> returns the fitted coefficients after one iteration of the Newton-Raphson algorithm. If 
<code>doff = TRUE</code>, it also returns the penalty matrix and weights matrix used to compute the degrees of 
freedom. 
<code>objectiveFunction</code> returns the objective function for the current iteration of the Newton-Raphson 
algorithm. 
</p>


<h3>Note</h3>

<p>These functions are not to be called directly by the user. They should be called via <code>logisticRidge</code>.
</p>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP] 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+logisticRidge">logisticRidge</a></code>
</p>

<hr>
<h2 id='GenBin'>
Simulated genetic data with a binary phenotypes
</h2><span id='topic+GenBin'></span><span id='topic+GenBin_genotypes'></span><span id='topic+GenBin_phenotypes'></span>

<h3>Description</h3>

<p>Simulated genetic data at 15 SNPs, together with simulated binary phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GenBin)</code></pre>


<h3>Format</h3>

<p>GenBin is a saved R matrix with 500 rows and 15 columns. The first
column is the pheotypes and columns 2-15 contain the genotypes. Each
row represents an indiviaul.
The same data are stored in flat text files in GenBin_genotypes and GenBin_phenotypes
(in the directory extdata (in the installed package) or inst/extdata (in the source)).
</p>


<h3>Source</h3>

<p>Simulated using FREGENE
</p>


<h3>References</h3>

<p>Fregene: Simulation of realistic sequence-level data in populations and ascertained samples Chadeau-Hyam, M. et al, 
2008, BMC Bioinformatics 9:364
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GenBin)
</code></pre>

<hr>
<h2 id='GenCont'>
Simulated genetic data with continuous outcomes
</h2><span id='topic+GenCont'></span><span id='topic+GenCont_genotypes'></span><span id='topic+GenCont_phenotypes'></span>

<h3>Description</h3>

<p>Simulated genetic data with continuous outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GenCont)</code></pre>


<h3>Format</h3>

<p>GenCont is a saved R matrix with 500 rows and 13 columns. The first
column is the pheotypes and columns 2-13 contain the genotypes. Each
row represents an indiviaul.
The same data are stored in flat text files in GenCont_genotypes and GenCont_phenotypes
(in the directory extdata (in the installed package) or inst/extdata (in the source)).
</p>


<h3>Details</h3>

<p>Genotypes were simulated using FREGENE.
</p>


<h3>References</h3>

<p>Fregene: Simulation of realistic sequence-level data in populations and ascertained samples Chadeau-Hyam, M. et al, 
2008, BMC Bioinformatics 9:364
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GenCont)
</code></pre>

<hr>
<h2 id='Gorman'>
The Ten-Factor data first described by Gorman and Toman (1966).
</h2><span id='topic+Gorman'></span>

<h3>Description</h3>

<p>A Ten-Factor data set first described by Gornam and Toman (1966) and used by Hoerl and Kennard (1970) (and others) to 
investigate regression problems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Gorman)</code></pre>


<h3>Format</h3>

<p>Numeric matrix.
</p>


<h3>Details</h3>

<p>The first column is the response on the log scale, the remaining columns are the predictors.
</p>


<h3>Source</h3>

<p>Selection of variables for fitting equations to data. Gorman, J. W. and Toman, R. J. (1966) Technometrics, 8:27.
</p>


<h3>References</h3>

<p>Selection of variables for fitting equations to data. Gorman, J. W. and Toman, R. J. (1966) Technometrics, 8:27.
Ridge Regression: Biased estimators for nonorthogonal problems. Hoerl, A. E. and Kennard, R. W. (1970) Technometrics, 
12:55.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Gorman)
</code></pre>

<hr>
<h2 id='Hald'>
Hald data
</h2><span id='topic+Hald'></span>

<h3>Description</h3>

<p>The Hald data as used by Hoerl, Kennard and Baldwin (1975). 
These data are also in package <code>wle</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Hald)</code></pre>


<h3>Format</h3>

<p>Numeric matrix.
</p>


<h3>Details</h3>

<p>The first column is the response and the remaining four columns are the predictors.
</p>


<h3>References</h3>

<p>Ridge Regression: some simulations, Hoerl, A. E. <em>et al</em>, 1975, Comm Stat Theor Method 4:105
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hald)
</code></pre>

<hr>
<h2 id='linearRidge'>
Linear ridge regression.
</h2><span id='topic+linearRidge'></span><span id='topic+coef.ridgeLinear'></span><span id='topic+nobs.ridgeLinear'></span><span id='topic+plot.ridgeLinear'></span><span id='topic+predict.ridgeLinear'></span><span id='topic+print.ridgeLinear'></span><span id='topic+summary.ridgeLinear'></span><span id='topic+print.summary.ridgeLinear'></span>

<h3>Description</h3>

<p>Fits a linear ridge regression model. Optionally, the ridge regression parameter is chosen automatically using 
the method proposed by Cule et al (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linearRidge(formula, data, lambda = "automatic", nPCs = NULL,
scaling = c("corrForm", "scale", "none"), ...)

## S3 method for class 'ridgeLinear'
coef(object, all.coef = FALSE, ...)

## S3 method for class 'ridgeLinear'
plot(x, y = NULL, ...)

## S3 method for class 'ridgeLinear'
predict(object, newdata, na.action = na.pass, all.coef = FALSE, ...)

## S3 method for class 'ridgeLinear'
print(x, all.coef = FALSE, ...)

## S3 method for class 'ridgeLinear'
summary(object, all.coef = FALSE, ...)

## S3 method for class 'summary.ridgeLinear'
print(x, digits = max(3,
getOption("digits") - 3),
signif.stars = getOption("show.signif.stars"), ...) 

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linearRidge_+3A_formula">formula</code></td>
<td>

<p>a formula expression as for regression models, of the form <code>response ~ predictors</code>. See the 
documentation of <code>formula</code> for other details.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_data">data</code></td>
<td>

<p>an optional data frame in which to interpret the variables occuring in <code>formula</code>.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_lambda">lambda</code></td>
<td>

<p>A ridge regression parameter. May be a vector. If <code>lambda</code> is <code>"automatic"</code> (the default), then the ridge parameter 
is chosen automatically using the method of Cule et al (2012).
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_npcs">nPCs</code></td>
<td>

<p>The number of principal components to use to choose the ridge regression parameter, following the method of 
Cule et al (2012). It is not possible to specify both <code>lambda</code> and <code>nPCs</code>.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_scaling">scaling</code></td>
<td>

<p>The method to be used to scale the predictors. One of
<code>"corrform"</code>(the default) scales the predictors to correlation form, such that the correlation matrix 
has unit diagonal.
<code>"scale"</code>Standardizes the predictors to have mean zero and unit variance.
<code>"none"</code>No scaling.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_object">object</code></td>
<td>

<p>A ridgeLinear object, typically generated by a call to <code>linearRidge</code>.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_newdata">newdata</code></td>
<td>

<p>An optional data frame in which to look for variables with
which to predict.  If omitted, the fitted values are used.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_na.action">na.action</code></td>
<td>

<p>function determining what should be done with missing values
in <code>newdata</code>.  The default is to predict <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_all.coef">all.coef</code></td>
<td>

<p>Logical. Should results be returned for all ridge regression penalty
parameters (<code>all.coef = TRUE</code>), or only for the ridge parameter chosen automatically using the method of Cule et al?
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_x">x</code></td>
<td>
<p>An object of class <code>ridgeLinear</code> (for the
<code>print.ridgeLinear</code> and <code>plot.ridgeLinear</code> functions) or an object of class
<code>summary.ridgeLinear</code> (for the <code>print.summary.ridgeLinear</code>
function)</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_y">y</code></td>
<td>
<p>Dummy argument for compatibility with the default <code>plot</code>
method. Ignored.</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_digits">digits</code></td>
<td>
<p>minimum number of significant digits to be used for most numbers</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_signif.stars">signif.stars</code></td>
<td>
<p>logical; if <code>TRUE</code>, P-values are additionally encoded
visually as <code>significance stars</code> in order to help scanning of
long coefficient tables.  It defaults to the
<code>show.signif.stars</code> slot of <code>options</code>.
</p>
</td></tr>
<tr><td><code id="linearRidge_+3A_...">...</code></td>
<td>

<p>Additional arguments to be passed to or from other methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If an intercept is present in the model, its coefficient is not penalised. If you want to penalise an 
intercept, put in your own constant term and remove the intercept.
</p>


<h3>Value</h3>

<p>An object of class <code>"ridgeLinear"</code>, with components:
</p>
<table>
<tr><td><code>automatic</code></td>
<td>
<p>Logical. Was <code>lambda</code> chosen automatically?</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>A named vector of fitted coefficients.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>A vector of degrees of freedom of the model fit, degrees of freedom for variance, and residual 
degrees of freedom of the fitted model.</p>
</td></tr>
<tr><td><code>Inter</code></td>
<td>
<p>Was an intercept included?</p>
</td></tr>
<tr><td><code>isScaled</code></td>
<td>
<p>Were the predictors scaled before the model was fitted?</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The ridge regression parameter(s).</p>
</td></tr>
<tr><td><code>scales</code></td>
<td>
<p>The scales used to standardize the predictors.</p>
</td></tr>
<tr><td><code>terms</code></td>
<td>
<p>The <code><a href="stats.html#topic+terms">terms</a></code> object used.</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>The scaled predictor matrix.</p>
</td></tr>
<tr><td><code>xm</code></td>
<td>
<p>A vector of means of the predictors.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>The response.</p>
</td></tr>
<tr><td><code>ym</code></td>
<td>
<p>The mean of the response.</p>
</td></tr>
</table>
<p>And optionally the components
</p>
<table>
<tr><td><code>max.nPCs</code></td>
<td>
<p>The maximum number of principal components for which a ridge regression parameter was computed.</p>
</td></tr>
<tr><td><code>chosen.nPCs</code></td>
<td>
<p>The number of principal components used to compute the ridge parameter.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP] 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+logisticRidge">logisticRidge</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GenCont)
mod &lt;- linearRidge(Phenotypes ~ ., data = as.data.frame(GenCont))
summary(mod)
</code></pre>

<hr>
<h2 id='linearRidgeGenotypes'>
Fits linear ridge regression models for genome-wide SNP data.
</h2><span id='topic+linearRidgeGenotypes'></span>

<h3>Description</h3>

<p>Fits linear ridge regression models for genome-wide SNP data. The SNP
genotypes are not read into R but file names are passed the code
directly, enabling the analysis of genome-wide scale SNP data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linearRidgeGenotypes(genotypesfilename, phenotypesfilename, lambda = -1, 
thinfilename = NULL, betafilename = NULL, approxfilename = NULL,
permfilename = NULL, intercept =
TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linearRidgeGenotypes_+3A_genotypesfilename">genotypesfilename</code></td>
<td>

<p>character string: path to file containing SNP genotypes coded 0, 1,
2. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_phenotypesfilename">phenotypesfilename</code></td>
<td>

<p>character string: path to file containing phenotypes. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_lambda">lambda</code></td>
<td>

<p>(optional) shrinkage parameter. If not provided, the default denotes
automatic choice of the shrinkage parameter using the method of Cule &amp;
De Iorio (2012). 
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_thinfilename">thinfilename</code></td>
<td>

<p>(optional) character string: path to file containing three columns: SNP name, chromosme and SNP psotion. See <code>Input file formats</code>. (See <code>details</code>.)
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_betafilename">betafilename</code></td>
<td>

<p>(optional) character string: path to file where the output will be written. See <code>Output file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_approxfilename">approxfilename</code></td>
<td>

<p>(optional) character string: path to fine where the approximate test p-values will be written. 
Approximate p-values are not computed unless this argument is given. Approximate p-values
are computed using the method of Cule et al (2011). See <code>Output file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_permfilename">permfilename</code></td>
<td>

<p>(optional) character string: path to file where the permutation test
p-values will be written.
Permutation test p-values are not computed unless this argument is
given. (See warning). See <code>Output file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_intercept">intercept</code></td>
<td>

<p>Logical: Should the ridge regression model be fitted with an
intercept? (Defaults to <code>TRUE</code>)
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypes_+3A_verbose">verbose</code></td>
<td>

<p>Logical: If <code>TRUE</code>, additional information is printed to the R
output as the code runs. Defaults to <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a file <code>thin</code> is supplied, and the shrinkage parameter
<code>lambda</code> is being computed automatically based on the data, then
this file is used to thin the SNP data by SNP position. If this file
is not supplied, SNPs are thinned automatically based on number of SNPs.
</p>


<h3>Value</h3>

<p>The vector of fitted ridge regression coefficients.
If <code>betafilename</code> is given, the fitted coefficients are written to this
file as well as being returned.
If <code>approxfilename</code> and/or <code>permfilename</code> are given, results of approximate
test p-values and/or permutation test p-values are written to the files
given in their arguments.
</p>


<h3>Input file formats</h3>


<dl>
<dt>genotypesfilename:</dt><dd><p>A header row, plus one row for each
individual, one SNP per column. The header row contains SNP
names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
values are not accommodated. Invariant SNPs in the data cause an
error, please remove these from the file before calling the function.</p>
</dd>
<dt>phenofilename:</dt><dd><p>A single column of phenotypes with the individuals in the same order as those in the file <code>genotypesfilename</code>.</p>
</dd>
<dt>thin:</dt><dd><p>(optional) Three columns and the same number of rows as there are SNPs in the file <code>genotypesfilename</code>, one row per SNP. First column: SNP names (must match names in <code>genotypesfilename</code>); second column: chromosome;  third column: SNP position in BP.</p>
</dd>
</dl>



<h3>Output file formats</h3>

<p>All output files are optional. Whether or not <code>betafilename</code> is provided, fitted coefficients are returned to the R workshpace. If <code>betafilename</code> is provided, fitted coefficients are written to the file specified (in addition).
</p>

<dl>
<dt>betafilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If <code>intercept = TRUE</code> (the default) then the first row is the fitted intercept (with the name Intercept in the first column).</p>
</dd>
<dt>approxfilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is approximate p-values.</p>
</dd>
<dt>permfilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is permutation p-values.</p>
</dd>
</dl>



<h3>Warning </h3>

<p>When data are large, the permutation test p-values
may take a very long time to compute. It is recommended not to request
permutation test p-values (using the argument <code>permfilename</code>)
when data are large.</p>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>Significance testing in ridge regression for genetic data. Cule, E. et
al (2011) BMC Bioinformatics, 12:372
A semi-automatic method to guide the choice of ridge parameter in
ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP]
</p>


<h3>See Also</h3>

<p><code><a href="#topic+linearRidge">linearRidge</a></code> for fitting linear ridge regression models
when the data are small enough to be read into R.
<code><a href="#topic+logisticRidge">logisticRidge</a></code> and <code><a href="#topic+logisticRidgeGenotypes">logisticRidgeGenotypes</a></code> for fitting logistic ridge
regression models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
    genotypesfile &lt;- system.file("extdata","GenCont_genotypes.txt",package = "ridge")
    phenotypesfile &lt;- system.file("extdata","GenCont_phenotypes.txt",package = "ridge")
    beta_linearRidgeGenotypes &lt;- linearRidgeGenotypes(genotypesfilename = genotypesfile,
phenotypesfilename = phenotypesfile)
    ## compare to output of linearRidge
    data(GenCont) ## Same data as in GenCont_genotypes.txt and GenCont_phenotypes.txt
    beta_linearRidge &lt;- linearRidge(Phenotypes ~ ., data = as.data.frame(GenCont))
    cbind(round(coef(beta_linearRidge), 6), beta_linearRidgeGenotypes)

## End(Not run)
  </code></pre>

<hr>
<h2 id='linearRidgeGenotypesPredict'>
Predict phenotypes from genome-wide SNP data based on a file of coefficients
</h2><span id='topic+linearRidgeGenotypesPredict'></span>

<h3>Description</h3>

<p>Predict phenotypes from genome-wide SNP data based on a file of
coefficients. Genotypes and fitted coefficients are provided as
filenames, allowing the computation of fitted probabilities when SNP
data are too large to be read into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linearRidgeGenotypesPredict(genotypesfilename, betafilename, phenotypesfilename = NULL,
verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linearRidgeGenotypesPredict_+3A_genotypesfilename">genotypesfilename</code></td>
<td>

<p>character string: path to file containing SNP genotypes coded 0, 1,
2. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypesPredict_+3A_betafilename">betafilename</code></td>
<td>

<p>character string: path to file containing fitted coefficients. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypesPredict_+3A_phenotypesfilename">phenotypesfilename</code></td>
<td>

<p>(optional) character string: path to file in which to write out the
predicted phenotypes. See <code>Output file formats</code>. Whether or not this argument
is supplied, the fitted coefficients are also returned by the function.
</p>
</td></tr>
<tr><td><code id="linearRidgeGenotypesPredict_+3A_verbose">verbose</code></td>
<td>

<p>Logical: If <code>TRUE</code>, additional information is printed to the R
outupt as the code runs. Defaults to <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of fitted values, the same length as the number of
individuals whose data are in <code>genotypesfilename</code>. If
<code>phenotypesfilename</code> is supplied, the fitted values are also
written there.
</p>


<h3>Input file formats</h3>


<dl>
<dt>genotypesfilename:</dt><dd><p>A header row, plus one row for each
individual, one SNP per column. The header row contains SNP
names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
values are not accommodated. </p>
</dd>
<dt>betafilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If the coefficients include an intercept then the first row of <code>betafilename</code> should contain it with the name Intercept in the first column. An Intercept thus labelled will be used appropriately in predicting the phenotypes. SNP names must match those in <code>genotypesfilename</code>. 
The format of <code>betafilename</code> is
that of the output of <code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code>, meaning
<code>linearRidgeGenotypesPredict</code> can be used to predict using
coefficients fitted using <code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code> (see the example).
</p>
</dd>
</dl>



<h3>Output file format</h3>

<p>Whether or not <code>phenotypesfilename</code> is provided, predicted phenotypes are returned to the R workshpace. If <code>phenotypesfilename</code> is provided, predicted phenotypes are written to the file specified (in addition).
</p>

<dl>
<dt>phenotypesfilename:</dt><dd><p>One column, containing predicted phenotypes, one individual per row.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP] 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code> for model
fitting. <code><a href="#topic+logisticRidgeGenotypes">logisticRidgeGenotypes</a></code> and
<code><a href="#topic+logisticRidgeGenotypesPredict">logisticRidgeGenotypesPredict</a></code> for corresponding functions
to fit and predict on SNP data with binary outcomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
genotypesfile &lt;- system.file("extdata","GenCont_genotypes.txt",package = "ridge")
phenotypesfile &lt;- system.file("extdata","GenCont_phenotypes.txt",package = "ridge")
betafile &lt;- tempfile(pattern = "beta", fileext = ".dat")
beta_linearRidgeGenotypes &lt;- linearRidgeGenotypes(genotypesfilename = genotypesfile,
                                                      phenotypesfilename = phenotypesfile,
                                                      betafilename = betafile)
pred_phen_geno &lt;- linearRidgeGenotypesPredict(genotypesfilename = genotypesfile,
                                                    betafilename = betafile)
## compare to output of linearRidge
data(GenCont) ## Same data as in GenCont_genotypes.txt and GenCont_phenotypes.txt
beta_linearRidge &lt;- linearRidge(Phenotypes ~ ., data = as.data.frame(GenCont))
pred_phen &lt;- predict(beta_linearRidge)
print(cbind(pred_phen_geno, pred_phen))
## Delete the temporary betafile
unlink(betafile)

## End(Not run)
</code></pre>

<hr>
<h2 id='logisticRidge'>
Logistic ridge regression.
</h2><span id='topic+logisticRidge'></span><span id='topic+coef.ridgeLogistic'></span><span id='topic+plot.ridgeLogistic'></span><span id='topic+predict.ridgeLogistic'></span><span id='topic+print.ridgeLogistic'></span><span id='topic+summary.ridgeLogistic'></span><span id='topic+print.summary.ridgeLogistic'></span>

<h3>Description</h3>

<p>Fits a logistic ridge regression model. Optionally, the ridge regression parameter is chosen automatically using 
the method proposed by Cule et al (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logisticRidge(formula, data, lambda = "automatic", nPCs = NULL,
scaling = c("corrForm", "scale", "none"), ...)

## S3 method for class 'ridgeLogistic'
coef(object, all.coef = FALSE, ...)

## S3 method for class 'ridgeLogistic'
plot(x, y = NULL, ...)

## S3 method for class 'ridgeLogistic'
predict(object, newdata = NULL, type = c("link", "response"), 
    na.action = na.pass, all.coef = FALSE, ...)

## S3 method for class 'ridgeLogistic'
print(x, all.coef = FALSE, ...)

## S3 method for class 'ridgeLogistic'
summary(object, all.coef = FALSE, ...)

## S3 method for class 'summary.ridgeLogistic'
print(x, digits = max(3, getOption("digits") - 3),
signif.stars = getOption("show.signif.stars"), ...) 

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logisticRidge_+3A_formula">formula</code></td>
<td>

<p>a formula expression as for regression models, of the form <code>response ~ predictors</code>. See the 
documentation of <code>formula</code> for other details.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_data">data</code></td>
<td>

<p>an optional data frame in which to interpret the variables occuring in <code>formula</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_lambda">lambda</code></td>
<td>

<p>A ridge regression parameter. If <code>lambda</code> is <code>"automatic"</code> (the default), then the ridge parameter 
is chosen automatically using the method of Cule et al (2012).
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_npcs">nPCs</code></td>
<td>

<p>The number of principal components to use to choose the ridge regression parameter, following the method of 
Cule et al (2012). It is not possible to specify both <code>lambda</code> and <code>nPCs</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_scaling">scaling</code></td>
<td>

<p>The method to be used to scale the predictors. One of
<code>"corrform"</code>(the default) scales the predictors to correlation form, such that the correlation matrix 
has unit diagonal.
<code>"scale"</code>Standardizes the predictors to have mean zero and unit variance.
<code>"none"</code>No scaling.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_object">object</code></td>
<td>

<p>A ridgeLogistic object, typically generated by a call to <code>linearRidge</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_newdata">newdata</code></td>
<td>

<p>An optional data frame in which to look for variables with
which to predict.  If omitted, the fitted values are used.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_type">type</code></td>
<td>
<p>the type of prediction required.  The default predictions are of log-odds
(probabilities on logit scale) and <code>type = "response"</code> gives
the predicted probabilities.</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_na.action">na.action</code></td>
<td>

<p>function determining what should be done with missing values
in <code>newdata</code>.  The default is to predict <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_all.coef">all.coef</code></td>
<td>

<p>Logical. Should results be returned for all ridge regression penalty
parameters (<code>all.coef = TRUE</code>), or only for the ridge parameter chosen automatically using the method of Cule et al?
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_x">x</code></td>
<td>
<p>An object of class <code>ridgeLogistic</code> (for the
<code>print.ridgeLogistic</code> and <code>plot.ridgeLogistic</code> functions) or an object of class
<code>summary.ridgeLogistic</code> (for the <code>print.summary.ridgeLogistic</code>
function)</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_y">y</code></td>
<td>
<p>Dummy argument for compatibility with the default <code>plot</code> method. Ignored.</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_digits">digits</code></td>
<td>
<p>minimum number of significant digits to be used for most numbers</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_signif.stars">signif.stars</code></td>
<td>
<p>logical; if <code>TRUE</code>, P-values are additionally encoded
visually as <code>significance stars</code> in order to help scanning of
long coefficient tables.  It defaults to the
<code>show.signif.stars</code> slot of <code>options</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidge_+3A_...">...</code></td>
<td>

<p>Additional arguments to be passed to or from other methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If an intercept is present in the model, its coefficient is not penalised. If you want to penalise an 
intercept, put in your own constant term and remove the intercept.
</p>


<h3>Value</h3>

<p>An object of class <code>"ridgeLogistic"</code>, with components:
</p>
<table>
<tr><td><code>automatic</code></td>
<td>
<p>Was <code>lambda</code> chosen automatically?</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>A named vector of fitted coefficients.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>A vector of degrees of freedom of the model fit and degrees of freedom for variance.</p>
</td></tr>
<tr><td><code>Inter</code></td>
<td>
<p>Was in antercept included?</p>
</td></tr>
<tr><td><code>isScaled</code></td>
<td>
<p>Were the predictors scaled before the model was fitted?</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The ridge regression parameter.</p>
</td></tr>
<tr><td><code>scales</code></td>
<td>
<p>The scales used to standardize the predictors.</p>
</td></tr>
<tr><td><code>terms</code></td>
<td>
<p>The <code><a href="stats.html#topic+terms">terms</a></code> object used.</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>The scaled predictor matrix.</p>
</td></tr>
<tr><td><code>xm</code></td>
<td>
<p>A vector of means of the predictors.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>The response.</p>
</td></tr>
</table>
<p>And optionally the component
</p>
<table>
<tr><td><code>nPCs</code></td>
<td>
<p>The number of principal components used to compute the ridge regression parameter.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP] 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+linearRidge">linearRidge</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GenBin)
mod &lt;- logisticRidge(Phenotypes ~ ., data = as.data.frame(GenBin))
summary(mod)
</code></pre>

<hr>
<h2 id='logisticRidgeGenotypes'>
Fits logistic ridge regression models for genomoe-wide SNP data.
</h2><span id='topic+logisticRidgeGenotypes'></span>

<h3>Description</h3>

<p>Fits logistic ridge regression models for genome-wide SNP data. The SNP
genotypes are not read into R but file names are passed to the code
directly, enabling the analysis of genome-wide SNP data sets which are
too big to be read into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logisticRidgeGenotypes(genotypesfilename, phenotypesfilename, lambda = -1, 
thinfilename = NULL, betafilename = NULL, approxfilename = NULL,
permfilename = NULL, intercept =
TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logisticRidgeGenotypes_+3A_genotypesfilename">genotypesfilename</code></td>
<td>

<p>character string: path to file containing SNP genotypes coded 0, 1,
2. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_phenotypesfilename">phenotypesfilename</code></td>
<td>

<p>character string: path to file containing phenotypes. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_lambda">lambda</code></td>
<td>

<p>(optional) shrinkage parameter. If not provided, the default denotes
automatic choice of the shrinkage parameter using the method of Cule &amp;
De Iorio (2012). 
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_thinfilename">thinfilename</code></td>
<td>

<p>(optional) character string: path to file containing three columns: SNP name, chromosme and SNP psotion. See <code>Input file formats</code>. (See <code>details</code>.)
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_betafilename">betafilename</code></td>
<td>

<p>(optional) character string: path to file where the output will be written. See <code>Output file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_approxfilename">approxfilename</code></td>
<td>

<p>(optional) character string: path to fine where the approximate test p-values will be written. 
Approximate p-values are not computed unless this argument is given. Approximate p-values
are computed using the method of Cule et al (2011). See <code>Output file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_permfilename">permfilename</code></td>
<td>

<p>(optional) character string: path to file where the permutation test
p-values will be written.
Permutation test p-values are not computed unless this argument is
given. (See warning). See <code>Output file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_intercept">intercept</code></td>
<td>

<p>Logical: Should the ridge regression model be fitted with an
intercept? Defaults to <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypes_+3A_verbose">verbose</code></td>
<td>

<p>Logical: If <code>TRUE</code>, additional information is printed to the R
output as the code runs. Defaults to <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a file <code>thin</code> is supplied, and the shrinkage parameter
<code>lambda</code> is being computed automatically based on the data, then
this file is used to thin the SNP data by SNP position. If this file
is not supplied, SNPs are thinned automatically based on number of SNPs.
</p>


<h3>Value</h3>

<p>The vector of fitted ridge regression coefficients.
If <code>betafilename</code> is given, the fitted coefficients are written to this
file as well as being returned.
If <code>approxfilename</code> and/or <code>permfilename</code> are given, results of approximate
test p-values and/or permutation test p-values are written to the files
given in their arguments.
</p>


<h3>Input file formats</h3>


<dl>
<dt>genotypesfilename:</dt><dd><p>A header row, plus one row for each
individual, one SNP per column. The header row contains SNP
names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
values are not accommodated. Invariant SNPs in the data cause an
error, please remove these from the file before calling the function.</p>
</dd>
<dt>phenofilename:</dt><dd><p>A single column of phenotypes with the individuals in the same order as those in the file <code>genotypesfilename</code>. Phenotypes must be coded as 0 or 1.</p>
</dd>
<dt>thin:</dt><dd><p>(optional) Three columns and the same number of rows as there are SNPs in the file <code>genotypesfilename</code>, one row per SNP. First column: SNP names (must match names in <code>genotypesfilename</code>); second column: chromosome;  third column: SNP position in BP.</p>
</dd>
</dl>



<h3>Output file formats</h3>

<p>All output files are optional. Whether or not <code>betafilename</code> is provided, fitted coefficients are returned to the R workshpace. If <code>betafilename</code> is provided, fitted coefficients are written to the file specified (in addition).
</p>

<dl>
<dt>betafilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If <code>intercept = TRUE</code> (the default) then the first row is the fitted intercept (with the name Intercept in the first column).</p>
</dd>
<dt>approxfilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is approximate p-values.</p>
</dd>
<dt>permfilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is permutation p-values.</p>
</dd>
</dl>



<h3>Warning </h3>

<p>When data are large, the permutation test p-values
may take a very long time to compute. It is recommended not to request
permutation test p-values (using the argument <code>permfilename</code>)
when data are large.
</p>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>Significance testing in ridge regression for genetic data. Cule, E. et
al (2011) BMC Bioinformatics, 12:372
A semi-automatic method to guide the choice of ridge parameter in
ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP]
</p>


<h3>See Also</h3>

<p><code><a href="#topic+logisticRidge">logisticRidge</a></code> for fitting logistic ridge regression models
when the data are small enough to be read into R.
<code><a href="#topic+linearRidge">linearRidge</a></code> and <code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code> for fitting linear ridge
regression models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
    genotypesfile &lt;- system.file("extdata","GenBin_genotypes.txt",package = "ridge")
    phenotypesfile &lt;- system.file("extdata","GenBin_phenotypes.txt",package = "ridge")
    beta_logisticRidgeGenotypes &lt;-
logisticRidgeGenotypes(genotypesfilename = genotypesfile, phenotypesfilename = phenotypesfile)
    ## compare to output of logisticRidge
    data(GenBin) ## Same data as in GenBin_genotypes.txt and GenBin_phenotypes.txt
    beta_logisticRidge &lt;- logisticRidge(Phenotypes ~ ., data = as.data.frame(GenBin))
    cbind(round(coef(beta_logisticRidge), 6), beta_logisticRidgeGenotypes)

## End(Not run)
  </code></pre>

<hr>
<h2 id='logisticRidgeGenotypesPredict'>
Predict fitted probabilities from genome-wide SNP data based on a file of coefficients
</h2><span id='topic+logisticRidgeGenotypesPredict'></span>

<h3>Description</h3>

<p>Predict fitted probabilities from genome-wide SNP data based on a file of
coefficients. Genotypes and fitted coefficients are provided as
filenames, allowing the computation of fitted probabilities when SNP
data are too large to be read into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logisticRidgeGenotypesPredict(genotypesfilename, betafilename,
phenotypesfilename = NULL, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logisticRidgeGenotypesPredict_+3A_genotypesfilename">genotypesfilename</code></td>
<td>

<p>character string: path to file containing SNP genotypes coded 0, 1,
2. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypesPredict_+3A_betafilename">betafilename</code></td>
<td>

<p>character string: path to file containing fitted coefficients. See <code>Input file formats</code>.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypesPredict_+3A_phenotypesfilename">phenotypesfilename</code></td>
<td>

<p>(optional) character string: path to file in which to write out the
fitted probabilities. See <code>Output file formats</code>. Whether or not this argument
is supplied, the fitted coefficients are also returned by the function.
</p>
</td></tr>
<tr><td><code id="logisticRidgeGenotypesPredict_+3A_verbose">verbose</code></td>
<td>

<p>Logical: If <code>TRUE</code>, additional information is printed to the R
outupt as the code runs. Defaults to <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of fitted probabilities, the same length as the number of
individuals whose data are in <code>genotypesfilename</code>. If
<code>phenotypesfilename</code> is supplied, the fitted probabilities are also
written there.
</p>


<h3>Input file formats</h3>


<dl>
<dt>genotypesfilename:</dt><dd><p>A header row, plus one row for each
individual, one SNP per column. The header row contains SNP
names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
values are not accommodated. </p>
</dd>
<dt>betafilename:</dt><dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If the coefficients include an intercept then the first row of <code>betafilename</code> should contain it with the name Intercept in the first column. An Intercept thus labelled will be used appropriately in predicting the phenotypes. SNP names must match those in <code>genotypesfilename</code>. 
The format of <code>betafilename</code> is
that of the output of <code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code>, meaning
<code>linearRidgeGenotypesPredict</code> can be used to predict using
coefficients fitted using <code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code> (see the example).
</p>
</dd>
</dl>



<h3>Output file format</h3>

<p>Whether or not <code>phenotypesfilename</code> is provided, fitted probabilities are returned to the R workshpace. If <code>phenotypesfilename</code> is provided, fitted probabilities are written to the file specified (in addition).
</p>

<dl>
<dt>phenotypesfilename:</dt><dd><p>One column, containing fitted probabilities, one individual per row.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP]
</p>


<h3>See Also</h3>

<p><code><a href="#topic+logisticRidgeGenotypes">logisticRidgeGenotypes</a></code> for model
fitting. <code><a href="#topic+linearRidgeGenotypes">linearRidgeGenotypes</a></code> and
<code><a href="#topic+linearRidgeGenotypesPredict">linearRidgeGenotypesPredict</a></code> for corresponding functions
to fit and predict on SNP data with continuous outcomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
genotypesfile &lt;- system.file("extdata","GenBin_genotypes.txt",package = "ridge")
phenotypesfile &lt;- system.file("extdata","GenBin_phenotypes.txt",package = "ridge")
betafile &lt;- tempfile(pattern = "beta", fileext = ".dat")
beta_logisticRidgeGenotypes &lt;- logisticRidgeGenotypes(genotypesfilename = genotypesfile,
                                                      phenotypesfilename = phenotypesfile,
                                                      betafilename = betafile)
pred_phen_geno &lt;- logisticRidgeGenotypesPredict(genotypesfilename = genotypesfile,
                                                    betafilename = betafile)
## compare to output of logisticRidge
data(GenBin) ## Same data as in GenBin_genotypes.txt and GenBin_phenotypes.txt
beta_logisticRidge &lt;- logisticRidge(Phenotypes ~ ., data = as.data.frame(GenBin))
pred_phen &lt;- predict(beta_logisticRidge, type="response")
print(cbind(pred_phen_geno, pred_phen))
## Delete the temporary betafile
unlink(betafile)

## End(Not run)
</code></pre>

<hr>
<h2 id='pvals'>
Compute p-values for ridgeLinear and ridgeLogistic models
</h2><span id='topic+pvals'></span><span id='topic+pvals.ridgeLinear'></span><span id='topic+pvals.ridgeLogistic'></span><span id='topic+plot.pvalsRidgeLinear'></span><span id='topic+plot.pvalsRidgeLogistic'></span><span id='topic+print.pvalsRidgeLinear'></span><span id='topic+print.pvalsRidgeLogistic'></span>

<h3>Description</h3>

<p>Functions for computing, printing and plotting p-values for ridgeLinear
and ridgeLogistic models. The p-values are computed using the
significance test of Cule et al (2011). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvals(x, ...)

## S3 method for class 'ridgeLinear'
pvals(x, ...)

## S3 method for class 'ridgeLogistic'
pvals(x, ...)

## S3 method for class 'pvalsRidgeLinear'
print(x, digits = max(3, getOption("digits") - 3),
signif.stars = getOption("show.signif.stars"), all.coef = FALSE, ...) 

## S3 method for class 'pvalsRidgeLogistic'
print(x, digits = max(3, getOption("digits") - 3),
signif.stars = getOption("show.signif.stars"), all.coef = FALSE, ...)

## S3 method for class 'pvalsRidgeLinear'
plot(x, y = NULL, ...)

## S3 method for class 'pvalsRidgeLogistic'
plot(x, y = NULL, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvals_+3A_x">x</code></td>
<td>
<p>For the pvals methods, an object of class &quot;ridgeLinear&quot; or &quot;ridgeLogistic&quot;, typically from a call to &quot;linearRidge&quot; or 
&quot;logisticRidge&quot;. For the print and plot methods, an object of class
&quot;pvalsRidgeLinear&quot; or &quot;pvalsRidgeLogistic&quot;, typically from a
call to &quot;pvals&quot;.</p>
</td></tr>
<tr><td><code id="pvals_+3A_digits">digits</code></td>
<td>
<p>minimum number of significant digits to be used for most numbers</p>
</td></tr>
<tr><td><code id="pvals_+3A_signif.stars">signif.stars</code></td>
<td>
<p>logical; if <code>TRUE</code>, P-values are additionally encoded
visually as <code>significance stars</code> in order to help scanning of
long coefficient tables.  It defaults to the
<code>show.signif.stars</code> slot of <code>options</code>.
</p>
</td></tr>
<tr><td><code id="pvals_+3A_all.coef">all.coef</code></td>
<td>
<p>Logical. Should p-values for all the ridge
regression parameters be printed, or only the one from the ridge
parameter chosen using the method of Cule et al (2012)</p>
</td></tr>
<tr><td><code id="pvals_+3A_y">y</code></td>
<td>
<p>Dummy argument for compatibility with the default <code>plot</code> method. Ignored.</p>
</td></tr>
<tr><td><code id="pvals_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard errors, test statistics and p-values are computed using coefficients and data on the scale that was 
used to fit them. If the coefficients were standardized before the model was fitted, then the p-values 
relate to the scaled data.
</p>


<h3>Value</h3>

<p>For the pvals methods, an object of class &quot;pvalsRidgeLinear&quot; or &quot;pvalsRidgeLogistic&quot; 
which is a list with elements
</p>
<table>
<tr><td><code>coef</code></td>
<td>
<p>The (scaled) regression coefficients</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>The standard errors of the regression coefficients</p>
</td></tr>
<tr><td><code>tstat</code></td>
<td>
<p>The test statistic of the regression coefficients</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>The p-values of the regression coefficients</p>
</td></tr>
<tr><td><code>isScaled</code></td>
<td>
<p>Were the data scaled before the regression
coefficients were fitted?</p>
</td></tr>
</table>
<p>For the print methods, the argument <code>x</code> is returned invisibly.
</p>


<h3>Author(s)</h3>

<p>Erika Cule
</p>


<h3>References</h3>

<p>Significance testing in ridge regression for genetic data. Cule, E. et al (2011) BMC Bioinformatics, 12:372
</p>


<h3>See Also</h3>

<p><code>linearRidge</code>, <code>logisticRidge</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GenBin)
mod &lt;- logisticRidge(Phenotypes ~ ., data = as.data.frame(GenBin))
pvalsMod &lt;- pvals(mod)
print(pvalsMod)
print(pvalsMod, all.coef = TRUE)
plot(pvalsMod)
</code></pre>

<hr>
<h2 id='ridge'>ridge: Linear and logistic ridge regression functions.</h2><span id='topic+ridge'></span>

<h3>Description</h3>

<p>Additionally includes special functions for 
genome-wide single-nucleotide polymorphism (SNP) data.
</p>

<hr>
<h2 id='ridge-package'>ridge-package description</h2><span id='topic+ridge-package'></span>

<h3>Description</h3>

<p>R package for fitting linear and logistic ridge regression models.
</p>


<h3>Details</h3>

<p>This package contains functions for fitting linear and logistic ridge
regression models, including functions for fitting linear and logistic
ridge regression models for genome-wide SNP data supplied as file names
when the data are too big to read into R.
</p>
<p>For a complete
list of functions, use <code>help(package="ridge")</code>.
</p>


<h3>Author(s)</h3>

<p>Steffen Moritz, Erika Cule
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
