<!DOCTYPE html><html><head><title>Help for package RJSplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RJSplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RJSplot'><p>Interactive graphs with R</p></a></li>
<li><a href='#barplot_rjs'><p>Produce interactive bar plot(s).</p></a></li>
<li><a href='#boxplot_rjs'><p>Produce interactive box-and-whisker plot(s).</p></a></li>
<li><a href='#bubbles_rjs'><p>Produce interactive bubble plots.</p></a></li>
<li><a href='#circularfv_rjs'><p>Pruduce a circular sequence viewer.</p></a></li>
<li><a href='#createAssembly'><p>Creates a genome assembly for <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.</p></a></li>
<li><a href='#dendrogram_rjs'><p>Draw a Dendrogram.</p></a></li>
<li><a href='#densityplot_rjs'><p>Draw an interactive density plot.</p></a></li>
<li><a href='#genomemap_rjs'><p>Create an interative genome map.</p></a></li>
<li><a href='#getAssemblyFromFasta'><p>Create an assembly from a FASTA file for its use as parameter of <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.</p></a></li>
<li><a href='#GRCh37'><p>Length of human chromosomes based on GRCh37 assembly</p></a></li>
<li><a href='#GRCh37.bands'><p>Cytoband information based on GRCh37</p></a></li>
<li><a href='#GRCh38'><p>Length of human chromosomes based on GRCh38 assembly</p></a></li>
<li><a href='#GRCh38.bands'><p>Cytoband information based on GRCh38</p></a></li>
<li><a href='#heatmap_rjs'><p>Produce an interactive heatmap.</p></a></li>
<li><a href='#hiveplot_rjs'><p>Produce interactive hive plot</p></a></li>
<li><a href='#manhattan_rjs'><p>Create an interactive manhattan plot</p></a></li>
<li><a href='#NCBI36'><p>Length of human chromosomes based on NCBI36 assembly</p></a></li>
<li><a href='#network_rjs'><p>Produce interactive network graph</p></a></li>
<li><a href='#newInput'><p>Produce a <code>newInput</code> object for <code>R4web</code>.</p></a></li>
<li><a href='#piechart_rjs'><p>Produce interactive pie charts.</p></a></li>
<li><a href='#R4web'><p>Generate a PHP website which runs an R function.</p></a></li>
<li><a href='#report_rjs'><p>Create a report web page.</p></a></li>
<li><a href='#scatter3d_rjs'><p>Produce and interactive 3D Scatter plot (XYZ plot).</p></a></li>
<li><a href='#scatterplot_rjs'><p>Produce and interactive Scatter plot (XY plot).</p></a></li>
<li><a href='#surface3d_rjs'><p>Produce and interactive 3D surface.</p></a></li>
<li><a href='#symheatmap_rjs'><p>Creates a Symetric Heatmap representation.</p></a></li>
<li><a href='#tables_rjs'><p>Produce interactive html tables.</p></a></li>
<li><a href='#wordcloud_rjs'><p>Draw a wordcloud.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>2.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-30</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactive Graphs with R</td>
</tr>
<tr>
<td>Description:</td>
<td>Creates interactive graphs with 'R'. It joins the data analysis power of R and the visualization libraries of JavaScript in one package.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>IRdisplay</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Carlos Prieto &lt;cprietos@usal.es&gt;,
  David Barrios &lt;metal@usal.es&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Carlos Prieto &lt;cprietos@usal.es&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-30 15:31:42 UTC; dbarrios</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-01 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='RJSplot'>Interactive graphs with R</h2><span id='topic+RJSplot'></span><span id='topic+RJSplot-package'></span>

<h3>Description</h3>

<p>Create interactive graphs with 'R'.</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>

<hr>
<h2 id='barplot_rjs'>Produce interactive bar plot(s).</h2><span id='topic+barplot_rjs'></span>

<h3>Description</h3>

<p><code>barplot_rjs</code> produce interactive bar plot(s) of the given values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barplot_rjs(height, xlab = "", ylab = "", ylim = NULL, cex = 1,
            plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="barplot_rjs_+3A_height">height</code></td>
<td>
<p>either a vector or matrix of values describing the bars which make up the plot.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_ylim">ylim</code></td>
<td>
<p>a vector with two values which specifies the Y minimun and maximun values in the representation.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="barplot_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive barplot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
barplot_rjs(USArrests, "states", "arrests")
}
</code></pre>

<hr>
<h2 id='boxplot_rjs'>Produce interactive box-and-whisker plot(s).</h2><span id='topic+boxplot_rjs'></span>

<h3>Description</h3>

<p><code>boxplot_rjs</code> produce interactive box-and-whisker plot(s) of the given values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxplot_rjs(data, xlab = "", ylab = "",
            outline = TRUE, col = NULL, ylim = NULL, cex = 1,
            plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boxplot_rjs_+3A_data">data</code></td>
<td>
<p>a Data Frame or Matrix from which the data for each box will be taken.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_outline">outline</code></td>
<td>
<p>if <code>outline</code> is not true, the outliers are not drawn.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_col">col</code></td>
<td>
<p>if 'col' is non-null it is assumed to contain colors to be used to colour the bodies of the box plots.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_ylim">ylim</code></td>
<td>
<p>a vector with two values which specifies the Y minimun and maximun values in the representation.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="boxplot_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive boxplot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
boxplot_rjs(attitude, "attitude", "favourable responses")
}
</code></pre>

<hr>
<h2 id='bubbles_rjs'>Produce interactive bubble plots.</h2><span id='topic+bubbles_rjs'></span>

<h3>Description</h3>

<p><code>bubbles_rjs</code> produce interactive bubble plot(s) of the given values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bubbles_rjs(data, size = NULL, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bubbles_rjs_+3A_data">data</code></td>
<td>
<p>a numeric Data Frame or Matrix with two columns and one row per bubble which contains represented values for each bubble. Row names will be used as bubble names.</p>
</td></tr>
<tr><td><code id="bubbles_rjs_+3A_size">size</code></td>
<td>
<p>a vector giving each bubble size.</p>
</td></tr>
<tr><td><code id="bubbles_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="bubbles_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="bubbles_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive barplot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
bubbles_rjs(scale(mtcars[,c("mpg","hp")],FALSE),mtcars$wt)
}
</code></pre>

<hr>
<h2 id='circularfv_rjs'>Pruduce a circular sequence viewer.</h2><span id='topic+circularfv_rjs'></span>

<h3>Description</h3>

<p><code>circularfv_rjs</code> creates  a circular sequence viewer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circularfv_rjs(sequence, features = FALSE,
               plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circularfv_rjs_+3A_sequence">sequence</code></td>
<td>
<p>a &quot;character&quot; string giving the DNA sequence.</p>
</td></tr>
<tr><td><code id="circularfv_rjs_+3A_features">features</code></td>
<td>
<p>Data Frame with the segments that will be seen in the viewer. It cotains 5 columns which describes the identifier, start coordinate, stop coordinate, segment type and segment color.</p>
</td></tr>
<tr><td><code id="circularfv_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="circularfv_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="circularfv_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()){
## Creates a circular sequence viewer in a temporal directory of your local machine
## Generate test input data
sequence &lt;- paste(sample(c("c","a","t","g"), 10000, replace = TRUE), collapse = "")
data &lt;- matrix(round(runif(12,0,10000)),6,2)
features &lt;- data.frame(id = 0:5, start = data[,1], stop = data[,2],
type = c("voluptate","non","voluptate","sit","et","proident"),
color = c("SandyBrown","Tan","Wheat","SteelBlue","LightSkyBlue","LightBlue"))

## Create the circular sequence viewver
circularfv_rjs(sequence, features)
}
</code></pre>

<hr>
<h2 id='createAssembly'>Creates a genome assembly for <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.</h2><span id='topic+createAssembly'></span>

<h3>Description</h3>

<p><code>createAssembly</code> create assemblies for their use as parameter of <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createAssembly(name, size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createAssembly_+3A_name">name</code></td>
<td>
<p>a vector with the chomosome or scaffold names.</p>
</td></tr>
<tr><td><code id="createAssembly_+3A_size">size</code></td>
<td>
<p>a vector with the chomosome or scaffold sizes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame in BED format to be used as an assembly in <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>.
</p>
<p><code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Creates a fake genome assembly for its posterior use as parameter of genomemap or manhattan plot
## Live examples and full tutorial on https://rjsplot.usal.es

## Generate a test assembly with three scaffolds
assembly &lt;- createAssembly(c("scaffold1","scaffold2","scaffold3"), c(1000000,800000,650000))
</code></pre>

<hr>
<h2 id='dendrogram_rjs'>Draw a Dendrogram.</h2><span id='topic+dendrogram_rjs'></span>

<h3>Description</h3>

<p><code>dendrogram_rjs</code> creates a dendrogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dendrogram_rjs(data, metadata = NULL, method = "complete", cex = 1,
               plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dendrogram_rjs_+3A_data">data</code></td>
<td>
<p>a dissimilarity object of class dist (p.e. produced by function dist).</p>
</td></tr>
<tr><td><code id="dendrogram_rjs_+3A_metadata">metadata</code></td>
<td>
<p>a data frame with metadata information for leaf features (p.e. sample phenotypes). Corresponding Leaf/Sample should be provided as row name.</p>
</td></tr>
<tr><td><code id="dendrogram_rjs_+3A_method">method</code></td>
<td>
<p>Name of the agglomeration method to be used. It should be (an unambiguous abbreviation of) one of &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot; (= UPGMA), &quot;mcquitty&quot; (= WPGMA), &quot;median&quot; (= WPGMC) or &quot;centroid&quot; (= UPGMC).</p>
</td></tr>
<tr><td><code id="dendrogram_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="dendrogram_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="dendrogram_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="dendrogram_rjs_+3A_dir">dir</code></td>
<td>
<p>A &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create a dendrogram in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
dendrogram_rjs(dist(USArrests),USArrests)
}
</code></pre>

<hr>
<h2 id='densityplot_rjs'>Draw an interactive density plot.</h2><span id='topic+densityplot_rjs'></span>

<h3>Description</h3>

<p><code>densityplot_rjs</code> creates density plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>densityplot_rjs(data, xlab = "", ylab = "", cex = 1,
                plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="densityplot_rjs_+3A_data">data</code></td>
<td>
<p>a data frame from which distribution will be represented for each column.</p>
</td></tr>
<tr><td><code id="densityplot_rjs_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="densityplot_rjs_+3A_ylab">ylab</code></td>
<td>
<p> a title for the y axis.</p>
</td></tr>
<tr><td><code id="densityplot_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="densityplot_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="densityplot_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="densityplot_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Creates a density plot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
## Generate test input data
data &lt;- data.frame(Uni05 = (1:100)/21, Norm = rnorm(100), `5T` = rt(100, df = 5),
Gam2 = rgamma(100, shape = 2))

# Create the density plot
densityplot_rjs(data, "x", "y")
}
</code></pre>

<hr>
<h2 id='genomemap_rjs'>Create an interative genome map.</h2><span id='topic+genomemap_rjs'></span>

<h3>Description</h3>

<p><code>genomemap_rjs</code> creates an interactive genome map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomemap_rjs(assembly, track = NULL, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genomemap_rjs_+3A_assembly">assembly</code></td>
<td>
<p>a genome assembly data frame. RJSplot provides human assemblies (NCBI36, GRCh37, GRCh38), human assemblies with cytobands (GRCh37.bands or GRCh38.bands), or methods for creating your own assembly from a FASTA file <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>.</p>
</td></tr>
<tr><td><code id="genomemap_rjs_+3A_track">track</code></td>
<td>
<p>a data frame with values to represent on the genome map (chromosome, position, value).</p>
</td></tr>
<tr><td><code id="genomemap_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="genomemap_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="genomemap_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with Firefox. If you want to open this file stored in your local machine with other Web browser, please visit the help section on the RJSplot Web site (<a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>).
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive Genome map in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Create test data
chr &lt;- character()
pos &lt;- numeric()

for(i in 1:nrow(GRCh38)){
  chr &lt;- c(chr,as.character(rep(GRCh38[i,"chr"],100)))
  pos &lt;- c(pos,sample(GRCh38[i,"start"]:GRCh38[i,"end"],100))
}

value &lt;- round(rexp(length(pos)),2)

# Create a genome map
track &lt;- data.frame(chr,pos,pos+1,NA,value)
genomemap_rjs(GRCh38.bands, track)
}
</code></pre>

<hr>
<h2 id='getAssemblyFromFasta'>Create an assembly from a FASTA file for its use as parameter of <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.</h2><span id='topic+getAssemblyFromFasta'></span>

<h3>Description</h3>

<p><code>getAssemblyFromFasta</code> create assemblies from a FASTA file for their use as parameter of <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAssemblyFromFasta(fasta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAssemblyFromFasta_+3A_fasta">fasta</code></td>
<td>
<p>a &quot;character&quot; string representing the input Fasta file to be added in the genome browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame in BED format ready to use as an assembly in <code>genomemap_rjs</code> or <code>manhattan_rjs</code>.
</p>


<h3>Note</h3>

<p>FASTA format is a text-based format for representing either nucleotide sequences or peptide sequences, in which nucleotides or amino acids are represented using single-letter codes.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>.
</p>
<p><code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Live examples and full tutorial on https://rjsplot.usal.es
</code></pre>

<hr>
<h2 id='GRCh37'>Length of human chromosomes based on GRCh37 assembly</h2><span id='topic+GRCh37'></span>

<h3>Description</h3>

<p>Length of human chromosomes based on GRCh37 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/">https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/</a>.</p>

<hr>
<h2 id='GRCh37.bands'>Cytoband information based on GRCh37</h2><span id='topic+GRCh37.bands'></span>

<h3>Description</h3>

<p>A Data Frame containing cytobands of GRCh37 assembly.</p>


<h3>Source</h3>

<p>See <a href="http://grch37.ensembl.org/Homo_sapiens/Location/Genome">http://grch37.ensembl.org/Homo_sapiens/Location/Genome</a>.</p>

<hr>
<h2 id='GRCh38'>Length of human chromosomes based on GRCh38 assembly</h2><span id='topic+GRCh38'></span>

<h3>Description</h3>

<p>Length of human chromosomes based on GRCh38 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ncbi.nlm.nih.gov/assembly/883148/">https://www.ncbi.nlm.nih.gov/assembly/883148/</a>.</p>

<hr>
<h2 id='GRCh38.bands'>Cytoband information based on GRCh38</h2><span id='topic+GRCh38.bands'></span>

<h3>Description</h3>

<p>A Data Frame containing cytobands of GRCh38 assembly.</p>


<h3>Source</h3>

<p>See <a href="http://www.ensembl.org/Homo_sapiens/Location/Genome">http://www.ensembl.org/Homo_sapiens/Location/Genome</a>.</p>

<hr>
<h2 id='heatmap_rjs'>Produce an interactive heatmap.</h2><span id='topic+heatmap_rjs'></span>

<h3>Description</h3>

<p><code>heatmap_rjs</code> creates interactive heatmaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_rjs(data, metadata = NULL, scale = c("row", "column", "none"),
  cluster = TRUE, color = c("Reds", "Greens", "Blues", "RdBkGr", "RdWhBu"),
  na.color = "transparent", cex = 1, plot = TRUE, jupyter = FALSE, 
  dir = tempdir(), distfun = dist, hclustfun = hclust)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_rjs_+3A_data">data</code></td>
<td>
<p>numeric matrix of the values to be plotted.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_metadata">metadata</code></td>
<td>
<p>a data frame with metadata information for column features (p.e. sample phenotypes). Samples are given in rows and features in columns.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_scale">scale</code></td>
<td>
<p>character indicating if the values should be centered and scaled in either the row direction or the column direction, or none. The default value is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_cluster">cluster</code></td>
<td>
<p>logical value to enable (<code>TRUE</code>) or disable (<code>FALSE</code>) data clustering.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_color">color</code></td>
<td>
<p>character indicating the color scale to paint values. The default value is &quot;Blues&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_na.color">na.color</code></td>
<td>
<p>character indicating the color to use for missing values. Transparent by default.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_distfun">distfun</code></td>
<td>
<p>function used to compute the distance between rows and columns.</p>
</td></tr>
<tr><td><code id="heatmap_rjs_+3A_hclustfun">hclustfun</code></td>
<td>
<p>function used to compute the hierarchical clustering.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive heatmap in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
metadata &lt;- data.frame(phenotype1 = sample(c("yes","no"),ncol(mtcars),TRUE),
                       phenotype2 = sample(1:5,ncol(mtcars),TRUE))

heatmap_rjs(data.matrix(mtcars), metadata, scale="column")
}
</code></pre>

<hr>
<h2 id='hiveplot_rjs'>Produce interactive hive plot</h2><span id='topic+hiveplot_rjs'></span>

<h3>Description</h3>

<p><code>hiveplot_rjs</code> creates an interactive hive plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hiveplot_rjs(links, nodes = NULL, group = NULL, size = NULL, color = NULL,
             cex = 1, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hiveplot_rjs_+3A_links">links</code></td>
<td>
<p>an edge list table stored in a data frame with three possible columns (source node id, target node id and link value).</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_nodes">nodes</code></td>
<td>
<p>node attributes stored in a data frame with rownames as node id.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_group">group</code></td>
<td>
<p>column name of <code>nodes</code> data frame which describes visualized axes.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_size">size</code></td>
<td>
<p>column name of <code>nodes</code> data frame which describes each node size.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_color">color</code></td>
<td>
<p>column name of <code>nodes</code> data frame which describes each node color.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="hiveplot_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive hive plot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Prepare data
x &lt;- 1-cor(t(mtcars))

source &lt;- rep(rownames(x),nrow(x))
target &lt;- rep(rownames(x),rep(ncol(x),nrow(x)))
links &lt;- data.frame(source=source,target=target,value=as.vector(x))

# Generate the hive plot
hiveplot_rjs(links, mtcars, group = "cyl", size = "wt", color = "carb")
}
</code></pre>

<hr>
<h2 id='manhattan_rjs'>Create an interactive manhattan plot</h2><span id='topic+manhattan_rjs'></span>

<h3>Description</h3>

<p><code>manhattan_rjs</code> creates an interactive Manhattan plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan_rjs(data, assembly, axisv, distv, cell = 1e+06,
  outliers = TRUE, ylab = "score", yscale = NULL,
  cex = 1, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan_rjs_+3A_data">data</code></td>
<td>
<p>a data frame with four columns (identifier, chromosome/scaffold, position, score).</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_assembly">assembly</code></td>
<td>
<p>a data frame with an assembly to be used. It should be one of the following human assemblies data frames provided by RJSplot: <code>NCBI36</code>, <code>GRCh37</code>, <code>GRCh38</code>.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_axisv">axisv</code></td>
<td>
<p>value for the coordinate of the main horizontal axis.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_distv">distv</code></td>
<td>
<p>distance between the main horizontal axis and the secondary axes (which stablishes the limit for the outlier representation).</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_cell">cell</code></td>
<td>
<p>segment size which is represented.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_outliers">outliers</code></td>
<td>
<p>logical value to enable (<code>TRUE</code>) or disable (<code>FALSE</code>) outliers.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_yscale">yscale</code></td>
<td>
<p>a vector with two values which specifies the minimun and maximun limits in the representation. By default maximun and minimun scores are taken as the limits.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="manhattan_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create an interactive Manhattan plot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Create test data
chr &lt;- character()
pos &lt;- numeric()

for(i in 1:nrow(GRCh38)){
  chr &lt;- c(chr,as.character(rep(GRCh38[i,"chr"],100)))
  pos &lt;- c(pos,sample(GRCh38[i,"start"]:GRCh38[i,"end"],100))
}

value &lt;- round(rexp(length(pos)),2)

# Create a manhattan plot
data &lt;- data.frame(paste0("ProbeSet_",seq_along(pos)),chr,pos,value)
manhattan_rjs(data, GRCh38, 0, 1, 0, TRUE, "log2Ratio")
}

</code></pre>

<hr>
<h2 id='NCBI36'>Length of human chromosomes based on NCBI36 assembly</h2><span id='topic+NCBI36'></span>

<h3>Description</h3>

<p>Length of human chromosomes based on NCBI36 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.12/">https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.12/</a>.</p>

<hr>
<h2 id='network_rjs'>Produce interactive network graph</h2><span id='topic+network_rjs'></span>

<h3>Description</h3>

<p><code>network_rjs</code> creates an interactive network graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_rjs(links, nodes = NULL,
            weight = NULL, linkLabel = NULL, linkColor = NULL,
            group = NULL, size = NULL, color = NULL, label = NULL,
            nodeColorScale = c("RdBk","RdBkGr"), cex = 1, 
            plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="network_rjs_+3A_links">links</code></td>
<td>
<p>an edge list table stored in a data frame with at least two columns (source node id and target node id).</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_nodes">nodes</code></td>
<td>
<p>a node list stored in a data frame object with rownames as node id.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_weight">weight</code></td>
<td>
<p>a vector giving the column of <code>links</code> to show as weight. If <code>NULL</code> but third <code>links</code> column is numeric, this one will be used as weight.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_linklabel">linkLabel</code></td>
<td>
<p>a vector giving the column of <code>links</code> to show as link text.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_linkcolor">linkColor</code></td>
<td>
<p>a vector giving the column of <code>links</code> to compute link color or directly a vector of colors.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_group">group</code></td>
<td>
<p>a vector giving the column of <code>nodes</code> to show as group.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_size">size</code></td>
<td>
<p>a vector giving the column of <code>nodes</code> to show as size.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_color">color</code></td>
<td>
<p>a vector giving the column of <code>nodes</code> to compute node color or directly a vector of colors.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_label">label</code></td>
<td>
<p>a vector giving the column of <code>nodes</code> to show as label.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_nodecolorscale">nodeColorScale</code></td>
<td>
<p>a &quot;character&quot; string representing the color scale applied to nodes by default.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="network_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive network graph in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Prepare data
x &lt;- 1-cor(t(mtcars))

source &lt;- rep(rownames(x),nrow(x))
target &lt;- rep(rownames(x),rep(ncol(x),nrow(x)))
links &lt;- data.frame(source=source,target=target,value=as.vector(x))

# Generate the network graph
network_rjs(links[links[,3]&gt;0.1,], mtcars, group = "cyl", size = "hp", color = "mpg")
}
</code></pre>

<hr>
<h2 id='newInput'>Produce a <code>newInput</code> object for <code>R4web</code>.</h2><span id='topic+newInput'></span>

<h3>Description</h3>

<p><code>newInput</code> produce <code>newInput</code> object for <code>R4web</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newInput(name, type = c("character","numeric","matrix","data.frame","factor"),
         default = "", levels = NULL, multi = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newInput_+3A_name">name</code></td>
<td>
<p>character string with the variable name.</p>
</td></tr>
<tr><td><code id="newInput_+3A_type">type</code></td>
<td>
<p>character string which describes the variable type. Possible values are &quot;character&quot;, &quot;numeric&quot;, &quot;matrix&quot;, &quot;data.frame&quot; or &quot;factor&quot;.</p>
</td></tr>
<tr><td><code id="newInput_+3A_default">default</code></td>
<td>
<p>default value for this input.</p>
</td></tr>
<tr><td><code id="newInput_+3A_levels">levels</code></td>
<td>
<p>a vector with the possible input values. It is described only for factor data <code>type</code>.</p>
</td></tr>
<tr><td><code id="newInput_+3A_multi">multi</code></td>
<td>
<p>boolean value which indicates if multiple selection are enabled for this input. It is described only for factor data <code>type</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a <code>newInput</code> object.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>

<hr>
<h2 id='piechart_rjs'>Produce interactive pie charts.</h2><span id='topic+piechart_rjs'></span>

<h3>Description</h3>

<p><code>piechart_rjs</code> produce interactive pie charts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>piechart_rjs(data, cex = 1, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="piechart_rjs_+3A_data">data</code></td>
<td>
<p>a Data Frame or Matrix from which the data for each pie will be taken.</p>
</td></tr>
<tr><td><code id="piechart_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="piechart_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="piechart_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="piechart_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive pie chart in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
piechart_rjs(VADeaths)
}
</code></pre>

<hr>
<h2 id='R4web'>Generate a PHP website which runs an R function.</h2><span id='topic+R4web'></span>

<h3>Description</h3>

<p><code>R4web</code> generate a PHP website which runs an R function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>R4web(...,fnfile,dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="R4web_+3A_...">...</code></td>
<td>
<p>one or more <code>newInput</code> objects which will be included as form fields. Form values will be used as parameters for the R script.</p>
</td></tr>
<tr><td><code id="R4web_+3A_fnfile">fnfile</code></td>
<td>
<p>a character string representing the R script file to be executed with the website.</p>
</td></tr>
<tr><td><code id="R4web_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the PHP code will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in the working directory of your computer with some PHP files. Resulting folder should be added to the Apache applications directory and enable writting permissions.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>

<hr>
<h2 id='report_rjs'>Create a report web page.</h2><span id='topic+report_rjs'></span><span id='topic+addCSS2report'></span><span id='topic+addHTML2report'></span><span id='topic+addParagraph2report'></span><span id='topic+addPlot2report'></span><span id='topic+addImage2report'></span>

<h3>Description</h3>

<p><code>report_rjs</code> creates a report web page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_rjs(dir = "RJSreport")
addCSS2report(report,css)
addHTML2report(report,HTML)
addParagraph2report(report,p)
addPlot2report(report,plot,height=600)
addImage2report(report,img)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the report will be saved.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_report">report</code></td>
<td>
<p>a RJSreport object produced by the function <code>report_rjs</code>.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_css">css</code></td>
<td>
<p>a character string representing the CSS path to be added.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_html">HTML</code></td>
<td>
<p>a character string giving the HTML code to be added.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_p">p</code></td>
<td>
<p>a character string giving the new paragraph to be added.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_plot">plot</code></td>
<td>
<p>a character string representing the plot directory to be added.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_height">height</code></td>
<td>
<p>an integer vector giving a default height of the plot.</p>
</td></tr>
<tr><td><code id="report_rjs_+3A_img">img</code></td>
<td>
<p>a character string representing the image path to be added.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the report. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>

<hr>
<h2 id='scatter3d_rjs'>Produce and interactive 3D Scatter plot (XYZ plot).</h2><span id='topic+scatter3d_rjs'></span>

<h3>Description</h3>

<p><code>scatter3d_rjs</code> generates an interactive 3D Scatter Plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatter3d_rjs(x, y, z, color="#000", xlab = "x", ylab = "y", zlab = "z",
              plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatter3d_rjs_+3A_x">x</code></td>
<td>
<p>a vector with the x data of each point.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_y">y</code></td>
<td>
<p>a vector with the y data of each point.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_z">z</code></td>
<td>
<p>a vector with the z data of each point.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_zlab">zlab</code></td>
<td>
<p>a title for the z axis.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_color">color</code></td>
<td>
<p>a vector with the colour of each represented point.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="scatter3d_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive 3D Scatter plot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

# 3D Scatter plot creation
if(interactive()){
scatter3d_rjs(iris$Sepal.Width, iris$Sepal.Length, iris$Petal.Width,
  color = iris$Species, xlab = "Sepal Width (cm)", ylab = "Sepal Length (cm)",
  zlab = "Petal Width (cm)")
}
</code></pre>

<hr>
<h2 id='scatterplot_rjs'>Produce and interactive Scatter plot (XY plot).</h2><span id='topic+scatterplot_rjs'></span>

<h3>Description</h3>

<p><code>scatterplot_rjs</code> generates an interactive Scatter Plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatterplot_rjs(x, y, id = NULL, xlab = "", ylab = "",
                xlim = NULL, ylim = NULL, col = NULL,
                pch = NULL, abline.x = NULL, abline.y = NULL, cex = 1,
                plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatterplot_rjs_+3A_x">x</code></td>
<td>
<p>a vector with the x data of each point.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_y">y</code></td>
<td>
<p>a vector with the y data of each point.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_id">id</code></td>
<td>
<p>a vector giving the id of each point.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_xlim">xlim</code></td>
<td>
<p>a vector with two values which specifies the X minimun and maximun values in the representation.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_ylim">ylim</code></td>
<td>
<p>a vector with two values which specifies the Y minimun and maximun values in the representation.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_col">col</code></td>
<td>
<p>a vector with the colour of each represented point.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_pch">pch</code></td>
<td>
<p>a character vector with the shape of each point. It should be one of: 1-&quot;circle&quot;, 2-&quot;triangle-up&quot;, 3-&quot;cross&quot;, 4-&quot;square&quot;, 5-&quot;diamond&quot;, 6-&quot;triangle-down&quot;.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_abline.x">abline.x</code></td>
<td>
<p>a numeric vector with x coordenates where a vertical line will be placed.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_abline.y">abline.y</code></td>
<td>
<p>a numeric vector with y coordenates where a horizontal line will be placed.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="scatterplot_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive Scatter plot in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# Scatter plot creation
scatterplot_rjs(iris$Sepal.Width, iris$Sepal.Length,
  abline.x = c(3.4,3.8), abline.y = c(5.8,7),
  col = iris$Species, pch = as.numeric(iris$Species), id = iris$Species,
  xlab = "Sepal Width (cm)", ylab = "Sepal Length (cm)")
}
</code></pre>

<hr>
<h2 id='surface3d_rjs'>Produce and interactive 3D surface.</h2><span id='topic+surface3d_rjs'></span>

<h3>Description</h3>

<p><code>surface3d_rjs</code> generates an interactive 3D surface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surface3d_rjs(x, color = "#fff", xlab = "x", ylab = "y", zlab = "z",
              plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surface3d_rjs_+3A_x">x</code></td>
<td>
<p>numeric matrix of the values to be plotted.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_color">color</code></td>
<td>
<p>a vector with colors to construct the color scale.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_zlab">zlab</code></td>
<td>
<p>a title for the z axis.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="surface3d_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive 3D Surface in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# 3D surface creation
surface3d_rjs(volcano,color=c("red","green"))
}
</code></pre>

<hr>
<h2 id='symheatmap_rjs'>Creates a Symetric Heatmap representation.</h2><span id='topic+symheatmap_rjs'></span>

<h3>Description</h3>

<p><code>symheatmap_rjs</code> creates an interactive symetric heatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symheatmap_rjs(links, nodes = NULL, group = NULL, label = NULL, cex = 1,
               plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="symheatmap_rjs_+3A_links">links</code></td>
<td>
<p>an edge list table stored in a data frame with three possible columns (source node id, target node id and a link value).</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_nodes">nodes</code></td>
<td>
<p>a node list stored in a data frame object with rownames as node id.</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_group">group</code></td>
<td>
<p>a vector giving the column of <code>nodes</code> to show as group.</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_label">label</code></td>
<td>
<p>a vector giving the column of <code>nodes</code> to show as label.</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_cex">cex</code></td>
<td>
<p>number indicating the amount by which plotting text and symbols should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="symheatmap_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive symetric heatmap in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

if(interactive()){
# prepare data
x &lt;- 1-cor(t(mtcars))

source &lt;- rep(rownames(x),nrow(x))
target &lt;- rep(rownames(x),rep(ncol(x),nrow(x)))
links &lt;- data.frame(source=source,target=target,value=as.vector(x))

# Create symetric heatmap
symheatmap_rjs(links, mtcars, group = "cyl")
}
</code></pre>

<hr>
<h2 id='tables_rjs'>Produce interactive html tables.</h2><span id='topic+tables_rjs'></span>

<h3>Description</h3>

<p><code>tables_rjs</code> produce interactive html tables of the given values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tables_rjs(data, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tables_rjs_+3A_data">data</code></td>
<td>
<p>a Data Frame or Matrix to display.</p>
</td></tr>
<tr><td><code id="tables_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting table in your browser.</p>
</td></tr>
<tr><td><code id="tables_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="tables_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the table. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive html table in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es
if(interactive()){
tables_rjs(swiss)
}
</code></pre>

<hr>
<h2 id='wordcloud_rjs'>Draw a wordcloud.</h2><span id='topic+wordcloud_rjs'></span>

<h3>Description</h3>

<p><code>wordcloud_rjs</code> creates a wordcloud.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wordcloud_rjs(data, plot = TRUE, jupyter = FALSE, dir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wordcloud_rjs_+3A_data">data</code></td>
<td>
<p>a two column data Frame with the words to display and their frequency.</p>
</td></tr>
<tr><td><code id="wordcloud_rjs_+3A_plot">plot</code></td>
<td>
<p>open resulting graph in your browser.</p>
</td></tr>
<tr><td><code id="wordcloud_rjs_+3A_jupyter">jupyter</code></td>
<td>
<p>embed the graph as an iframe into a Jupyter Notebook.</p>
</td></tr>
<tr><td><code id="wordcloud_rjs_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in your computer with an HTML document named index.html which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics Service of Nucleus, University of Salamanca. See <a href="https://bioinfo.usal.es/">https://bioinfo.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;RJSplot&rsquo; Website: <a href="https://rjsplot.usal.es">https://rjsplot.usal.es</a>
</p>
<p><code><a href="#topic+dendrogram_rjs">dendrogram_rjs</a></code>, <code><a href="#topic+densityplot_rjs">densityplot_rjs</a></code>, <code><a href="#topic+genomemap_rjs">genomemap_rjs</a></code>, <code><a href="#topic+heatmap_rjs">heatmap_rjs</a></code>, <code><a href="#topic+manhattan_rjs">manhattan_rjs</a></code>, <code><a href="#topic+network_rjs">network_rjs</a></code>, <code><a href="#topic+scatterplot_rjs">scatterplot_rjs</a></code>, <code><a href="#topic+symheatmap_rjs">symheatmap_rjs</a></code>, <code><a href="#topic+wordcloud_rjs">wordcloud_rjs</a></code>, <code><a href="#topic+boxplot_rjs">boxplot_rjs</a></code>, <code><a href="#topic+bubbles_rjs">bubbles_rjs</a></code>, <code><a href="#topic+hiveplot_rjs">hiveplot_rjs</a></code>, <code><a href="#topic+piechart_rjs">piechart_rjs</a></code>, <code><a href="#topic+barplot_rjs">barplot_rjs</a></code>, <code><a href="#topic+tables_rjs">tables_rjs</a></code>, <code><a href="#topic+surface3d_rjs">surface3d_rjs</a></code>, <code><a href="#topic+scatter3d_rjs">scatter3d_rjs</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create an interactive heatmap in a temporal directory of your local machine
## Live examples on https://rjsplot.usal.es

## Format test data
words &lt;- data.frame(word = rownames(USArrests), freq = USArrests[,4])

# Create WordCloud
if(interactive()){
wordcloud_rjs(words)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
