<!DOCTYPE html><html lang="en"><head><title>Help for package rKOMICS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rKOMICS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#exData'><p>Example dataset</p></a></li>
<li><a href='#matrices'><p>Example cluster matrices</p></a></li>
<li><a href='#msc.depth'><p>Check the read depth of assembled minicircles</p></a></li>
<li><a href='#msc.heatmap'><p>Visualization of cluster matrices</p></a></li>
<li><a href='#msc.length'><p>Length of minicircles</p></a></li>
<li><a href='#msc.matrix'><p>Build cluster matrix</p></a></li>
<li><a href='#msc.pca'><p>Prinicple Component Analysis based on MSC</p></a></li>
<li><a href='#msc.quality'><p>Check the quality of the assembly</p></a></li>
<li><a href='#msc.richness'><p>Minicircle Sequence Cluster richness</p></a></li>
<li><a href='#msc.seqs'><p>Retrieve sequences</p></a></li>
<li><a href='#msc.similarity'><p>Minicircle Sequence Classes similarity</p></a></li>
<li><a href='#msc.subset'><p>Specific Minicircle Sequence Classes</p></a></li>
<li><a href='#msc.uc'><p>Cluster Analyses</p></a></li>
<li><a href='#preprocess'><p>Filtering of minicircle sequences</p></a></li>
<li><a href='#read.uc'><p>Read in uc files</p></a></li>
<li><a href='#rKOMICS'>
<p>Minicircle Sequence Classes (MSC) Analyses</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Minicircle Sequence Classes (MSC) Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-06-28</td>
</tr>
<tr>
<td>Description:</td>
<td>This is a analysis toolkit to streamline the analyses of minicircle sequence diversity in population-scale genome projects. rKOMICS is a user-friendly R package that has simple installation requirements and that is applicable to all 27 trypanosomatid genera. Once minicircle sequence alignments are generated, rKOMICS allows to examine, summarize and visualize minicircle sequence diversity within and between samples through the analyses of minicircle sequence clusters. We showcase the functionalities of the (r)KOMICS tool suite using a whole-genome sequencing dataset from a recently published study on the history of diversification of the Leishmania braziliensis species complex in Peru. Analyses of population diversity and structure highlighted differences in minicircle sequence richness and composition between Leishmania subspecies, and between subpopulations within subspecies. The rKOMICS package establishes a critical framework to manipulate, explore and extract biologically relevant information from mitochondrial minicircle assemblies in tens to hundreds of samples simultaneously and efficiently. This should facilitate research that aims to develop new molecular markers for identifying species-specific minicircles, or to study the ancestry of parasites for complementary insights into their evolutionary history. ***** !! WARNING: this package relies on dependencies from Bioconductor. For Mac users, this can generate errors when installing rKOMICS. Install Bioconductor and ComplexHeatmap at advance: install.packages("BiocManager"); BiocManager::install("ComplexHeatmap") *****.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, ape, circlize, ComplexHeatmap, reshape2, utils,
stats, dplyr, factoextra, FactoMineR, ggpubr, magrittr, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>viridis</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-28 11:25:36 UTC; mgeerts</td>
</tr>
<tr>
<td>Author:</td>
<td>Frederik Van den Broeck [aut],
  Manon Geerts [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Manon Geerts &lt;geertsmanon@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-29 22:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effect
</p>

<hr>
<h2 id='exData'>Example dataset</h2><span id='topic+exData'></span>

<h3>Description</h3>

<p>A dataset containing example inputs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exData
</code></pre>


<h3>Format</h3>

<p>A list with seven objects
</p>

<dl>
<dt>strain</dt><dd><p>a character vector containing the strain names.</p>
</dd>
<dt>subspecies</dt><dd><p>a factor specifying to which subspecie the strains belong to.</p>
</dd>
<dt>HCN</dt><dd><p>a numerical vector with their corresponding median genome wide coverage.</p>
</dd>
<dt>fastafiles</dt><dd><p>a character vector containing the file names of the minicircle sequences in fasta format.</p>
</dd>
<dt>ucs</dt><dd><p>a character vector containing the file names of the cluster information in uc format.</p>
</dd>
<dt>mapstats</dt><dd><p>a character vector containing the files names of the mapping statistics in text format.</p>
</dd>
<dt>depthstats</dt><dd><p>a character vector containing the files names of the depth statistics in text format.</p>
</dd>
</dl>


<hr>
<h2 id='matrices'>Example cluster matrices</h2><span id='topic+matrices'></span>

<h3>Description</h3>

<p>A list containing 15 example cluster matrices with percent identities of 80, 85 and 88:100.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrices
</code></pre>


<h3>Format</h3>

<p>a list of different cluster matrices.
</p>

<hr>
<h2 id='msc.depth'>Check the read depth of assembled minicircles</h2><span id='topic+msc.depth'></span>

<h3>Description</h3>

<p>The msc.depth function allows you to analyze the read depth of assembled minicircles using depth statistics generated by KOMICS. These statistics provide information such as the average, median, minimum, and maximum read depth per site for each minicircle contig. By standardizing the median read depths per minicircle contig to the median genome-wide read depths, you can estimate minicircle copy numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.depth(depthstats, groups, HCN = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.depth_+3A_depthstats">depthstats</code></td>
<td>
<p>character vector containing the file names of the depth statistics generated by KOMICS. Each file should be in the format &quot;sample_name.depthstats.txt&quot; (e.g., sampleA.depthstats.txt, sampleB.depthstats.txt,...).</p>
</td></tr>
<tr><td><code id="msc.depth_+3A_groups">groups</code></td>
<td>
<p>a vector specifying the groups (e.g., species) to which the samples belong.</p>
</td></tr>
<tr><td><code id="msc.depth_+3A_hcn">HCN</code></td>
<td>
<p>an optional numeric vector containing haploid copy numbers of the corresponding samples. This argument is set to null by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>all</code></td>
<td>
<p>a table that merges the depth statistics of all samples. The table includes the average, median, minimum, and maximum per site read depth.</p>
</td></tr>
<tr><td><code>plots</code></td>
<td>
<p>a plot per sample that visualizes the median read depth distribution.</p>
</td></tr>
<tr><td><code>medianRD</code></td>
<td>
<p>a graph summarizing the median read depth distribution of all samples.</p>
</td></tr>
<tr><td><code>CN</code></td>
<td>
<p>a graph summarizing the copy number (if HCN is not null) of all samples.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>require(ggpubr)
data(exData)

### run function

depth &lt;- msc.depth(depthstats = system.file("extdata", 
                  exData$depthstats, package = "rKOMICS"), groups = exData$species,
                  HCN = exData$medGWD/2)

### visualize results 
hist(depth$all[,"MEDIAN.DEPTH"], breaks=100, 
    main="Global median depth distribution",xlab = (''))

### alter plot 
annotate_figure(depth$plots$CUM29A1, fig.lab = "CUM29A1", 
               fig.lab.pos = "bottom.right", fig.lab.face = 'italic')

</code></pre>

<hr>
<h2 id='msc.heatmap'>Visualization of cluster matrices</h2><span id='topic+msc.heatmap'></span>

<h3>Description</h3>

<p>The msc.heatmap function generates a heatmap that summarizes the presence or absence of Minicircle Sequence Classes (MCSs) between groups of samples. It takes an input cluster matrix, generated using the msc.matrix function, and visualizes the clustering patterns of MCSs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.heatmap(clustmatrix, samples, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.heatmap_+3A_clustmatrix">clustmatrix</code></td>
<td>
<p>a cluster matrix generated with the msc.matrix function. This matrix represents the presence or absence of MCSs in each sample.</p>
</td></tr>
<tr><td><code id="msc.heatmap_+3A_samples">samples</code></td>
<td>
<p>a vector containing the sample names.</p>
</td></tr>
<tr><td><code id="msc.heatmap_+3A_groups">groups</code></td>
<td>
<p>a vector specifying the groups (e.g., species) to which the samples belong.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a heatmap that visualizes the clustering patterns of MCSs between sample groups. The heatmap provides an overview of the presence or absence of MCSs and helps identify shared or distinct MCS patterns among the groups.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exData)
data(matrices)

### run function 
msc.heatmap(matrices[["id80"]], groups = exData$species,
            samples = exData$samples )
   
### run function on every cluster matrix with subset of samples
### you will be asked to confirm
table(exData$species)
hybrid &lt;- which(exData$species=="hybrid")
# msc.heatmap(matrices[["id97"]], groups = exData$species[hybrid], 
#             samples = exData$samples[hybrid])


</code></pre>

<hr>
<h2 id='msc.length'>Length of minicircles</h2><span id='topic+msc.length'></span>

<h3>Description</h3>

<p>The msc.length function allows you to check the length of minicircle sequences based on a single FASTA file. This function helps determine the size distribution of minicircle sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.length(file, samples, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.length_+3A_file">file</code></td>
<td>
<p>the name of the FASTA file that contains all the minicircle sequences. The file should be in the format &quot;all.minicircles.circ.fasta&quot;.</p>
</td></tr>
<tr><td><code id="msc.length_+3A_samples">samples</code></td>
<td>
<p>a character vector containing the sample names.</p>
</td></tr>
<tr><td><code id="msc.length_+3A_groups">groups</code></td>
<td>
<p>a vector of the same length as the samples, specifying the groups (e.g., subspecies) to which the samples belong.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>length</code></td>
<td>
<p>a numerical vector containing the lengths of the minicircle sequences. Each element corresponds to the length of a specific minicircle sequence.</p>
</td></tr>
<tr><td><code>plot</code></td>
<td>
<p>a histogram that visualizes the frequency distribution of minicircle sequence lengths. The histogram provides an overview of the length distribution of the minicircles.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>require(ggplot2)
require(ggpubr)

### run function
bf &lt;- msc.length(file = system.file("extdata", "all.minicircles.fasta", package="rKOMICS"),
                 samples = exData$samples, groups = exData$subspecies)
af &lt;- msc.length(file = system.file("extdata", "all.minicircles.circ.fasta", package="rKOMICS"),
                 samples = exData$samples, groups = exData$subspecies)

length(which(bf$length&lt;800)) 
length(which(bf$length&gt;1400)) 

### visualize results
hist(af$length, breaks=50)

### alter plot
ggarrange(bf$plot + labs(caption = "Before filtering"), 
          af$plot + labs(caption = "After filtering"), nrow=2)


</code></pre>

<hr>
<h2 id='msc.matrix'>Build cluster matrix</h2><span id='topic+msc.matrix'></span>

<h3>Description</h3>

<p>The msc.matrix function reads the output of clustering analyses (UC file) for specified minimum percent identity (MPI) values and organizes the data into a matrix format. This matrix represents the presence or absence of Minicircle Sequence Classes (MSCs) in each sample. The resulting matrix simplifies downstream analyses and visualizations by eliminating the need for manual data manipulation and reformatting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.matrix(files, samples, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.matrix_+3A_files">files</code></td>
<td>
<p>a character vector containing the names of the UC files generated by the VSEARCH tool. Each file represents the output of clustering analysis for a specific minimum percent identity (MPI), such as all.minicircles.circ.id70.uc, all.minicircles.circ.id80.uc, and so on. Please ensure that your file names end with 'idxx.uc' for this function to work properly.</p>
</td></tr>
<tr><td><code id="msc.matrix_+3A_samples">samples</code></td>
<td>
<p>a character vector containing the sample names.</p>
</td></tr>
<tr><td><code id="msc.matrix_+3A_groups">groups</code></td>
<td>
<p>a vector of the same length as the samples, specifying the groups (e.g., species) to which the samples belong.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a a list that contains one cluster matrix per percent identity. Each matrix represents the presence or absence of MSCs in each sample. In the cluster matrix, a value of 0 indicates that the MSC is not present in the sample, while a value higher than 0 indicates that the MSC is found at least once in the sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exData)

### run function

matrices &lt;- msc.matrix(files = system.file("extdata", exData$ucs, package="rKOMICS"), 
                      samples = exData$samples, 
                      groups = exData$species)


### or: 
data(matrices)

### show matrix with id 95%
matrices[["id95"]]
rowSums(matrices[["id95"]]) # --&gt; frequency of MSC across all samples
colSums(matrices[["id95"]]) # --&gt; number of MSC per sample

</code></pre>

<hr>
<h2 id='msc.pca'>Prinicple Component Analysis based on MSC</h2><span id='topic+msc.pca'></span>

<h3>Description</h3>

<p>The msc.pca function allows you to perform Principle Component Analysis (PCA) to summarize the variation of Minicircle Sequence Classes (MSCs) in all samples or in a subset of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.pca(clustmatrix, samples, groups, n = 20, labels = TRUE, title = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.pca_+3A_clustmatrix">clustmatrix</code></td>
<td>
<p>a cluster matrix obtained from the msc.matrix function. The cluster matrix represents the presence or absence of MSCs in each sample, where rows represent MSCs and columns represent samples.</p>
</td></tr>
<tr><td><code id="msc.pca_+3A_samples">samples</code></td>
<td>
<p>a vector containing the names of the samples. This can include all samples or a subset of samples that you want to analyze.</p>
</td></tr>
<tr><td><code id="msc.pca_+3A_groups">groups</code></td>
<td>
<p>a vector specifying the groups (e.g., species) to which the samples belong.</p>
</td></tr>
<tr><td><code id="msc.pca_+3A_n">n</code></td>
<td>
<p>the number of clusters to select with the highest contribution to PCA. By default, it is set to 20.</p>
</td></tr>
<tr><td><code id="msc.pca_+3A_labels">labels</code></td>
<td>
<p>a  logical parameter indicating whether to use labels on the PCA plot or not. If set to TRUE (default), the plot will display sample labels.</p>
</td></tr>
<tr><td><code id="msc.pca_+3A_title">title</code></td>
<td>
<p>the  title of the graph. You can provide a title for the PCA plot if desired.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>plot</code></td>
<td>
<p>a PCA plot that visualizes the clustering of samples based on the presence/absence of MSCs. The plot helps identify clusters and patterns of similarity or dissimilarity between samples.</p>
</td></tr>
<tr><td><code>eigenvalues</code></td>
<td>
<p>a barplot showing the percentage of explained variances by each principal component. This plot provides insights into the contribution of each principal component to the overall variation in the data.</p>
</td></tr>
<tr><td><code>clustnames</code></td>
<td>
<p>a  A list of cluster names with the highest contribution to PCA. This list helps identify the MSC clusters that have the most influence on the PCA results.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(matrices)
data(exData)

### run function with all samples
res.pca &lt;- lapply(matrices, function(x) msc.pca(x, samples = exData$samples, 
                  groups = exData$species, n=30, labels=FALSE, title=NULL))

res.pca$id95$eigenvalues
res.pca$id95$plot

### use clusters with highest contribution to visualize in a heatmap
msc.heatmap(matrices[["id95"]][res.pca$id95$clustnames,], samples = exData$samples,
            groups = exData$species)

### run function with a subset of samples
### you will be asked to confirm
table(exData$species)
hybrid &lt;- which(exData$species=="hybrid")
# pca.subset &lt;- msc.pca(clustmatrix = matrices[["id97"]], 
#                       samples = exData$samples[hybrid], 
#                       groups = exData$species[hybrid], labels = TRUE, 
#                       title = "PCA only with hybrids")

</code></pre>

<hr>
<h2 id='msc.quality'>Check the quality of the assembly</h2><span id='topic+msc.quality'></span>

<h3>Description</h3>

<p>The msc.quality function allows you to summarize mapping statistics generated by KOMICS to assess the quality of minicircle assembly and mapping. It focuses on examining the frequencies of various mapped read categories, including read frequency, mapped read frequency, high-quality mapped read frequency, and the proportion of (near-)perfect alignments of CSB3-containing reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.quality(mapstats, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.quality_+3A_mapstats">mapstats</code></td>
<td>
<p>a character vector containing the file names of mapping statistics generated by KOMICS. These files provide information about the mapping and quality of the assembled minicircles.</p>
</td></tr>
<tr><td><code id="msc.quality_+3A_groups">groups</code></td>
<td>
<p>a vector specifying the groups (e.g., species) to which the samples belong.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>all</code></td>
<td>
<p>a table merging the mapping statistics of all samples. The mapping statistics include the number of mapped reads (MR), mapped reads with high quality (MR_HQ), CSB3-containing mapped reads (MR_CSB3), and CSB3-containing mapped reads with high quality (MR_CSB3_HQ).</p>
</td></tr>
<tr><td><code>proportions</code></td>
<td>
<p>a list of tables containing the proportions of the mentioned mapping statistics. It provides insights into the relative frequencies and proportions of each category.</p>
</td></tr>
<tr><td><code>plots</code></td>
<td>
<p>barplots visualizing the mapping statistics and proportions. These plots help visualize and compare the mapping quality across samples.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(exData)

### run function
map &lt;- msc.quality(mapstats = system.file("extdata", exData$mapstats, package = "rKOMICS"),
                   exData$species)

lapply(map$proportions, mean)$MR_HQ 
lapply(map$proportions, mean)$MR_CSB3_HQ

### visualize results
barplot(map$proportions$MR)

</code></pre>

<hr>
<h2 id='msc.richness'>Minicircle Sequence Cluster richness</h2><span id='topic+msc.richness'></span>

<h3>Description</h3>

<p>The msc.richness function calculates the measure of minicircle richness per sample by estimating the number of Minicircle Sequence Classes (MSCs) in each sample. It takes into account different minimum percent identities (MPIs) and returns a table of richness estimates per sample and per MPI. Additionally, it generates a boxplot that illustrates the minicircle richness across samples based on the estimated MSCs over a range of MPIs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.richness(clustmatrices, samples, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.richness_+3A_clustmatrices">clustmatrices</code></td>
<td>
<p>a list of cluster matrices obtained from the msc.matrix function. Each cluster matrix represents the presence or absence of MSCs in each sample.</p>
</td></tr>
<tr><td><code id="msc.richness_+3A_samples">samples</code></td>
<td>
<p>a vector containing the names of the samples.</p>
</td></tr>
<tr><td><code id="msc.richness_+3A_groups">groups</code></td>
<td>
<p>a vector specifying the group (e.g., species) to which each sample belongs.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>table</code></td>
<td>
<p>a table summarizing the number of MSCs per sample at different percent identities. The table provides an overview of the estimated minicircle richness in each sample across the MPIs.</p>
</td></tr>
<tr><td><code>plot</code></td>
<td>
<p>a boxplot visualizing the minicircle richness across samples. The boxplot represents the distribution of richness estimates in each sample over the range of MPIs considered.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>require(ggplot2)
data(matrices)
data(exData)

#### run function
richness &lt;- msc.richness(matrices, samples = exData$samples, groups = exData$species)

apply(richness$table[which(richness$table$group=="L. peruviana"),-(1:2)], 2, mean)
apply(richness$table[which(richness$table$group=="L. braziliensis"),-(1:2)], 2, mean)
apply(richness$table[which(richness$table$group=="hybrid"),-(1:2)], 2, mean)  

#### visualize results
barplot(richness$table[,"id93"], names.arg = richness$table[,1],
        las=2, cex.names=0.4, main="N of MSC at id 93")

#### adjust plot
richness$plot + ggtitle("MSC richness across % id") + 
                theme(axis.text.x = element_text(angle=45, hjust=1))

### show results of subset 
table(exData$species)
hybrid &lt;- which(exData$species=="hybrid")
# richness.subset &lt;- msc.richness(matrices, samples = exData$samples[hybrid], 
#                                 groups = exData$species[hybrid])

</code></pre>

<hr>
<h2 id='msc.seqs'>Retrieve sequences</h2><span id='topic+msc.seqs'></span>

<h3>Description</h3>

<p>The msc.seqs function retrieves the DNA sequence of a Minicircle Sequence Classes (MSC) together with all its hit sequences from a FASTA file and a corresponding UC file. This function is useful for extracting and analyzing specific MSCs and their associated hit sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.seqs(fastafile, ucfile, clustnumbers, writeDNA = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.seqs_+3A_fastafile">fastafile</code></td>
<td>
<p>the name of the FASTA file containing all minicircle sequences.</p>
</td></tr>
<tr><td><code id="msc.seqs_+3A_ucfile">ucfile</code></td>
<td>
<p>the name of the UC file.</p>
</td></tr>
<tr><td><code id="msc.seqs_+3A_clustnumbers">clustnumbers</code></td>
<td>
<p>a character vector containing the cluster numbers (in the format &quot;C0&quot;, &quot;C1&quot;, etc.) of the MSCs for which you want to retrieve the sequences. These cluster numbers specify the MSCs and their associated hit sequences that need to be extracted from the FASTA file and UC file.</p>
</td></tr>
<tr><td><code id="msc.seqs_+3A_writedna">writeDNA</code></td>
<td>
<p>a logical parameter that is set to TRUE by default. When set to TRUE, this parameter will write the extracted sequences into separate FASTA files in the current directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table that summarizes the number of hit sequences found in each MSC, the MSC names, and the samples where the MSCs are present. This table provides an overview of the extracted sequences and their distribution across samples.
</p>
<p>one FASTA file per MSC with all its hit sequences. These FASTA files can be further used for downstream analyses or sequence comparisons.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exData)

### select a subset of MSC
Lpe &lt;- which(exData$species == "L. peruviana")
specific &lt;- msc.subset(matrices[[7]], subset = Lpe)

### run function
seq &lt;- msc.seqs(fastafile = system.file("extdata", "all.minicircles.circ.fasta", package="rKOMICS"),
                ucfile = system.file("extdata", exData$ucs, package="rKOMICS")[7], 
                clustnumbers = specific$clustnumbers, writeDNA = FALSE)

</code></pre>

<hr>
<h2 id='msc.similarity'>Minicircle Sequence Classes similarity</h2><span id='topic+msc.similarity'></span>

<h3>Description</h3>

<p>The function msc.similarity returns a measure of minicircle sequence composition within and between groups of samples. Specifically, it estimates the absolute and relative number of Minicircle Sequence Classes (MSCs) that are unique to each group or shared between two or more groups. The function returns tables and barplots that summarize the number of unique or shared MSCs for each minimum percent identity (MPI) separately or combined over all MPIs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.similarity(clustmatrices, samples, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.similarity_+3A_clustmatrices">clustmatrices</code></td>
<td>
<p>a list of cluster matrices.</p>
</td></tr>
<tr><td><code id="msc.similarity_+3A_samples">samples</code></td>
<td>
<p>a vector containing the names of the samples. This can include all samples or it can be a subset.</p>
</td></tr>
<tr><td><code id="msc.similarity_+3A_groups">groups</code></td>
<td>
<p>a vector, of equal length as samples,  specifying to which group (e.g. species) the samples belong to.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>absfreq</code></td>
<td>
<p>a list per percent identity containing absolute frequency values of shared and unique MSCs.</p>
</td></tr>
<tr><td><code>absfreq.plot</code></td>
<td>
<p>a list of barplots visualizing previous results.</p>
</td></tr>
<tr><td><code>relfreq</code></td>
<td>
<p>a list per percent identity containing relative frequency values of shared and unique MSCs.</p>
</td></tr>
<tr><td><code>relfreq.plot</code></td>
<td>
<p>one barplot visualizing previous results.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>require(viridis)
data(matrices)
data(exData)

### run function
sim &lt;- msc.similarity(matrices, samples = exData$samples, 
                      groups = exData$species)

### visualize results (absolute frequencies)
barplot(sim$absfreq$id93)

### adjust plot (relative frequencies)
sim$relfreq.plot + scale_fill_viridis(discrete = TRUE)

sim$relfreq$id97["2"]*100 
sim$relfreq$id97["3"]*100 

### reduce number of groups 
groups &lt;- exData$species
levels(groups)[levels(groups)!='hybrid'] &lt;- "non-hybrid"
sim.red &lt;- msc.similarity(matrices, samples = exData$samples, groups = groups)
sim.red$relfreq.plot + scale_fill_viridis(discrete = TRUE)

</code></pre>

<hr>
<h2 id='msc.subset'>Specific Minicircle Sequence Classes</h2><span id='topic+msc.subset'></span>

<h3>Description</h3>

<p>The msc.subset function allows you to identify specific Minicircle Sequence Classes (MSCs) for a subset of samples based on the output of the msc.matrix function. It helps in extracting and analyzing MSCs that are present in a particular subset of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.subset(clustmatrix, subset)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.subset_+3A_clustmatrix">clustmatrix</code></td>
<td>
<p>a cluster matrix obtained from the msc.matrix function. The cluster matrix represents the presence or absence of MSCs in each sample.</p>
</td></tr>
<tr><td><code id="msc.subset_+3A_subset">subset</code></td>
<td>
<p>a numerical vector indicating the subset of samples for which you want to identify specific MSCs. The values in the subset vector correspond to the indices of the samples to be included.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>clustnumbers</code></td>
<td>
<p>a vector containing the names of the specific MSCs present in the subset of samples. These are the MSCs that are found in the indicated subset.</p>
</td></tr>
<tr><td><code>freq</code></td>
<td>
<p>frequency values indicating the occurrence of the specific MSCs in the subset of samples. These values represent the number of times each MSC appears in the subset.</p>
</td></tr>
<tr><td><code>matrix</code></td>
<td>
<p>a subset of the cluster matrix containing only the specific MSCs found in the subset of samples. For samples not included in the subset, the values in the matrix should have the value 0, indicating the absence of the MSC.</p>
</td></tr>
<tr><td><code>sum</code></td>
<td>
<p>the total number of MSC found in the indicated subset of samples.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(matrices)
data(exData)

### selecting a group of samples e.g. all L. peruviana species
Lpe &lt;- which(exData$species == "L. peruviana")

### run function
specific &lt;- msc.subset(matrices[["id97"]], subset = Lpe)

### visualize results (check if it is indeed specific)
heatmap(specific$matrix) # or:
msc.heatmap(specific$matrix, samples = exData$samples, groups = exData$species)

### find specific MSC with highest frequency
which.max(specific$freq)

</code></pre>

<hr>
<h2 id='msc.uc'>Cluster Analyses</h2><span id='topic+msc.uc'></span>

<h3>Description</h3>

<p>The function msc.uc reads the output of the clustering analyses (UC file) for each specified minimum percent identity (MPI) into a single list, which will be analyzed automatically to calculate and visualize, per MPI, the number of minicircle sequence classes (MSCs), the proportion of perfect alignments (i.e. alignments without any insertion/deletion, but allowing point mutations) and the number of alignment gaps. Gaps are defined by i) the number of insertions/deletions and ii) the length in base pairs of each individual insertion/deletion. It also issues a warning when large gaps (&gt;500 bp) are found, which points the user to anomalous alignments due to e.g. artificial dimers introduced by the assembly process. This allows the user to make an informed decision about the MPI (or MPI's) that best captures minicircle sequence richness within a (group of) sample(s) while minimizing the number and length of alignment gaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msc.uc(files)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msc.uc_+3A_files">files</code></td>
<td>
<p>a character vector that includes the file names of UC files (produced by USEARCH or VSEARCH), such as all.minicircles.circ.id70.uc, all.minicircles.circ.id80.uc, and so on. Please ensure that your file names end with 'idxx.uc' for this function to work properly.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>MSCs</code></td>
<td>
<p>a numerical vector containing the number of MSC per MPI.</p>
</td></tr>
<tr><td><code>perfect aligments</code></td>
<td>
<p>a numerical vector containing the proportions of perfect alignments per MPI.</p>
</td></tr>
<tr><td><code>insertions</code></td>
<td>
<p>a list showing the insertion lengths per MPI. Each element in the list corresponds to a specific MPI, and it provides the lengths of identified insertions.</p>
</td></tr>
<tr><td><code>deletions</code></td>
<td>
<p>a list showing the deletions lengths per MPI. Each element in the list corresponds to a specific MPI, and it provides the lengths of identified deletions.</p>
</td></tr>
<tr><td><code>insertions summary</code></td>
<td>
<p>a table showing the length and the number of insertions across different MPIs.</p>
</td></tr>
<tr><td><code>deletions summary</code></td>
<td>
<p>a table showing the length and the number of deletions across different MPIs.</p>
</td></tr>
<tr><td><code>plots</code></td>
<td>
<p>various plots showing previous results.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(exData)

### run function

ucs &lt;- msc.uc(files = system.file("extdata", exData$ucs, package="rKOMICS"))

ucs$MSCs["100"] 
ucs$MSCs["97"] 

### results
ucs$plots


</code></pre>

<hr>
<h2 id='preprocess'>Filtering of minicircle sequences</h2><span id='topic+preprocess'></span>

<h3>Description</h3>

<p>The preprocess function is used to filter minicircle sequences based on sequence length and circularization success. When minicircle sequences are assembled with KOMICS, individual fasta files are generated for each sample. This function allows you to filter these sequences based on their length and whether they are circularized or not. The filtered sequences are then written into individual FASTA files in the current working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocess(files, groups, circ = TRUE, min = 500, max = 1500, writeDNA = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="preprocess_+3A_files">files</code></td>
<td>
<p>a character vector containing the names of the fasta files. Each file corresponds to the minicircle sequences of a specific sample. The file names should be in the format sampleA.minicircles.fasta, sampleB.minicircles.fasta, and so on (output of KOMICS).</p>
</td></tr>
<tr><td><code id="preprocess_+3A_groups">groups</code></td>
<td>
<p>a factor specifying the group (e.g., species) to which each sample belongs. It should have the same length as the list of files, indicating the group assignment for each sample.</p>
</td></tr>
<tr><td><code id="preprocess_+3A_circ">circ</code></td>
<td>
<p>a logical parameter that determines whether non-circularized minicircle sequences should be included or excluded from the filtering process. By default, non-circularized sequences are excluded (circ = TRUE). If you are interested in including non-circularized sequences, you can set the parameter to FALSE.</p>
</td></tr>
<tr><td><code id="preprocess_+3A_min">min</code></td>
<td>
<p>the minimum length threshold for filtering minicircle sequences. Sequences with a length below this threshold will be excluded. The default value is set to 500.</p>
</td></tr>
<tr><td><code id="preprocess_+3A_max">max</code></td>
<td>
<p>the maximum length threshold for filtering minicircle sequences. Sequences with a length above this threshold will be excluded. The default value is set to 1500.</p>
</td></tr>
<tr><td><code id="preprocess_+3A_writedna">writeDNA</code></td>
<td>
<p>a logical parameter that determines whether the filtered minicircle sequences should be written in FASTA format to the current working directory. By default, the filtered sequences are written (writeDNA = TRUE). If you are only interested in other output values like plots and summary, you can set this parameter to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>samples</code></td>
<td>
<p>the sample names based on the input files.</p>
</td></tr>
<tr><td><code>N_MC</code></td>
<td>
<p>a table containing the sample names, the corresponding group assignment, and the number of minicircle sequences (N_MC) before and after filtering.</p>
</td></tr>
<tr><td><code>plot</code></td>
<td>
<p>a barplot visualizing the number of minicircle sequences per sample before and after filtering.</p>
</td></tr>
<tr><td><code>summary</code></td>
<td>
<p>the total number of minicircle sequences before and after filtering.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>require(ggplot2)
data(exData)

### setwd("")

### run function
table(exData$species)
pre &lt;- preprocess(files = system.file("extdata", exData$fastafiles, package="rKOMICS"),
                  groups = exData$species,
                  circ = TRUE, min = 500, max = 1200, writeDNA = FALSE)

pre$summary 

### visualize results
barplot(pre$N_MC[,"beforefiltering"], 
        names.arg = pre$N_MC[,1], las=2, cex.names=0.4)

### alter plot
pre$plot + labs(caption = paste0('N of MC sequences before and after filtering, ', Sys.Date()))

</code></pre>

<hr>
<h2 id='read.uc'>Read in uc files</h2><span id='topic+read.uc'></span>

<h3>Description</h3>

<p>The read.uc function is used to read the output of clustering analyses from a UC file. The function stores the information in two tables: one for hit records (H) and one for cluster records (C).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.uc(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.uc_+3A_file">file</code></td>
<td>
<p>the name of the UC file that contains the clustering analysis results. The file should be specified with its full name, including the extension (e.g., all.minicircles.circ.id70.uc).</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>hits</code></td>
<td>
<p>a table containing all the hit records from the UC file. Each row of the table represents a hit record, providing information about the alignment between a query sequence and a target sequence.</p>
</td></tr>
<tr><td><code>clusters</code></td>
<td>
<p>a table containing all the cluster records from the UC file. Each row of the table represents a cluster record, providing information about the clustering of sequences into clusters.</p>
</td></tr>
<tr><td><code>clustnumbers</code></td>
<td>
<p>a vector containing the cluster numbers (0-based). Each element of the vector represents a cluster identified in the clustering analysis.</p>
</td></tr>
</table>

<hr>
<h2 id='rKOMICS'>
Minicircle Sequence Classes (MSC) Analyses
</h2><span id='topic+rKOMICS'></span>

<h3>Description</h3>

<p>This is a analysis toolkit to streamline the analyses of minicircle sequence diversity in population-scale genome projects. rKOMICS is a user-friendly R package that has simple installation requirements and that is applicable to all 27 trypanosomatid genera. Once minicircle sequence alignments are generated, rKOMICS allows to examine, summarize and visualize minicircle sequence diversity within and between samples through the analyses of minicircle sequence clusters. We showcase the functionalities of the (r)KOMICS tool suite using a whole-genome sequencing dataset from a recently published study on the history of diversification of the Leishmania braziliensis species complex in Peru. Analyses of population diversity and structure highlighted differences in minicircle sequence richness and composition between Leishmania subspecies, and between subpopulations within subspecies. The rKOMICS package establishes a critical framework to manipulate, explore and extract biologically relevant information from mitochondrial minicircle assemblies in tens to hundreds of samples simultaneously and efficiently. This should facilitate research that aims to develop new molecular markers for identifying species-specific minicircles, or to study the ancestry of parasites for complementary insights into their evolutionary history. ***** !! WARNING: this package relies on dependencies from Bioconductor. For Mac users, this can generate errors when installing rKOMICS. Install Bioconductor and ComplexHeatmap at advance: install.packages(&quot;BiocManager&quot;); BiocManager::install(&quot;ComplexHeatmap&quot;) *****.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
This package was not yet installed at build time.<br />

Index of help topics:
</p>
<pre>
exData                  Example dataset
matrices                Example cluster matrices
msc.depth               Check the read depth of assembled minicircles
msc.heatmap             Visualization of cluster matrices
msc.length              Length of minicircles
msc.matrix              Build cluster matrix
msc.pca                 Prinicple Component Analysis based on MSC
msc.quality             Check the quality of the assembly
msc.richness            Minicircle Sequence Cluster richness
msc.seqs                Retrieve sequences
msc.similarity          Minicircle Sequence Classes similarity
msc.subset              Specific Minicircle Sequence Classes
msc.uc                  Cluster Analyses
preprocess              Filtering of minicircle sequences
rKOMICS                 Minicircle Sequence Classes (MSC) Analyses
read.uc                 Read in uc files
</pre>


<h3>Author(s)</h3>

<p>Frederik Van den Broeck <a href="mailto:frederik.vandenbroeck@kuleuven.be">frederik.vandenbroeck@kuleuven.be</a>
</p>
<p>Manon Geerts <a href="mailto:mgeerts@itg.be">mgeerts@itg.be</a>
</p>


<h3>References</h3>

<p>Geerts, M., Schnaufer, A. &amp; Van den Broeck, F. rKOMICS: an R package for processing mitochondrial minicircle assemblies in population-scale genome projects. BMC Bioinformatics 22, 468 (2021). doi: <a href="https://doi.org/10.1186/s12859-021-04384-1">10.1186/s12859-021-04384-1</a>
</p>
<p>Van den Broeck F, Savill NJ, Imamura H, Sanders M, Maes I, Cooper S, et al. Ecological divergence and hybridization of Neotropical Leishmania parasites. Proc Natl Acad Sci U S A. 2020;117. doi: <a href="https://doi.org/10.1073/pnas.1920136117">10.1073/pnas.1920136117</a>.
</p>


<h3>See Also</h3>

<p>Github:   <a href="https://frebio.github.io/">https://frebio.github.io/</a>
</p>
<p>komics-suite:   <a href="https://frebio.github.io/komics/">https://frebio.github.io/komics/</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
