<!DOCTYPE html><html><head><title>Help for package rMVP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rMVP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MVP'><p>MVP, Memory-efficient, Visualization-enhanced, Parallel-accelerated</p></a></li>
<li><a href='#MVP.BRENT.Vg.Ve'><p>MVP.BRENT.Vg.Ve</p>
variance component estimation using the BRENT method</a></li>
<li><a href='#MVP.calk'><p>Calculate Kinship matrix by Blocking strategy</p></a></li>
<li><a href='#MVP.Data'><p>MVP.Data: To prepare data for MVP package</p>
Author: Xiaolei Liu, Lilin Yin and Haohao Zhang
Build date: Aug 30, 2016
Last update: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.Bfile2MVP'><p>MVP.Data.Bfile2MVP: To transform plink binary data to MVP package</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.Hapmap2MVP'><p>MVP.Data.Hapmap2MVP: To transform Hapmap data to MVP package</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.impute'><p>MVP.Data.impute: To impute the missing genotype</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.Kin'><p>Kinship</p></a></li>
<li><a href='#MVP.Data.Map'><p>MVP.Data.Map: To check map file</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.MVP2Bfile'><p>MVP.Data.MVP2Bfile: To transform MVP data to binary format</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.Numeric2MVP'><p>MVP.Data.Numeric2MVP: To transform Numeric data to MVP package</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.PC'><p>Principal component analysis</p></a></li>
<li><a href='#MVP.Data.Pheno'><p>MVP.Data.Pheno: To clean up phenotype file</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.Data.VCF2MVP'><p>MVP.Data.VCF2MVP: To transform vcf data to MVP package</p>
Author: Haohao Zhang
Build date: Sep 12, 2018</a></li>
<li><a href='#MVP.EMMA.Vg.Ve'><p>Estimate variance components using EMMA</p></a></li>
<li><a href='#MVP.FarmCPU'><p>Perform GWAS using FarmCPU method</p></a></li>
<li><a href='#MVP.FaSTLMM.LL'><p>Evaluation of the maximum likelihood using FaST-LMM method</p></a></li>
<li><a href='#MVP.GLM'><p>To perform GWAS with GLM and MLM model and get the P value of SNPs</p></a></li>
<li><a href='#MVP.HE.Vg.Ve'><p>To estimate variance component using HE regression</p></a></li>
<li><a href='#MVP.Hist'><p>Phenotype distribution histogram</p></a></li>
<li><a href='#MVP.K.VanRaden'><p>Calculate Kinship matrix by VanRaden method</p></a></li>
<li><a href='#MVP.MLM'><p>To perform GWAS with GLM and MLM model and get the P value of SNPs</p></a></li>
<li><a href='#MVP.PCA'><p>Principal Component Analysis</p></a></li>
<li><a href='#MVP.PCAplot'><p>PCA Plot</p></a></li>
<li><a href='#MVP.Report'><p>MVP.Report</p></a></li>
<li><a href='#MVP.Report.Density'><p>SNP Density</p></a></li>
<li><a href='#MVP.Report.QQplot'><p>QQ Plot</p></a></li>
<li><a href='#MVP.Version'><p>Print MVP Banner</p></a></li>
<li><a href='#pig60K'><p>Genotyped by pig 60k chip</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Memory-Efficient, Visualize-Enhanced, Parallel-Accelerated GWAS
Tool</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-27</td>
</tr>
<tr>
<td>Description:</td>
<td>A memory-efficient, visualize-enhanced, parallel-accelerated Genome-Wide Association Study (GWAS) tool. It can
    (1) effectively process large data, 
    (2) rapidly evaluate population structure, 
    (3) efficiently estimate variance components several algorithms, 
    (4) implement parallel-accelerated association tests of markers three methods, 
    (5) globally efficient design on GWAS process computing, 
    (6) enhance visualization of related information. 
    'rMVP' contains three models GLM (Alkes Price (2006) &lt;<a href="https://doi.org/10.1038%2Fng1847">doi:10.1038/ng1847</a>&gt;), MLM (Jianming Yu (2006) &lt;<a href="https://doi.org/10.1038%2Fng1702">doi:10.1038/ng1702</a>&gt;) 
    and FarmCPU (Xiaolei Liu (2016) &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1005767">doi:10.1371/journal.pgen.1005767</a>&gt;); variance components estimation methods EMMAX 
    (Hyunmin Kang (2008) &lt;<a href="https://doi.org/10.1534%2Fgenetics.107.080101">doi:10.1534/genetics.107.080101</a>&gt;;), FaSTLMM (method: Christoph Lippert (2011) &lt;<a href="https://doi.org/10.1038%2Fnmeth.1681">doi:10.1038/nmeth.1681</a>&gt;, 
    R implementation from 'GAPIT2': You Tang and Xiaolei Liu (2016) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0107684">doi:10.1371/journal.pone.0107684</a>&gt; and 
    'SUPER': Qishan Wang and Feng Tian (2014) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0107684">doi:10.1371/journal.pone.0107684</a>&gt;), and HE regression 
    (Xiang Zhou (2017) &lt;<a href="https://doi.org/10.1214%2F17-AOAS1052">doi:10.1214/17-AOAS1052</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/xiaolei-lab/rMVP">https://github.com/xiaolei-lab/rMVP</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/xiaolei-lab/rMVP/issues">https://github.com/xiaolei-lab/rMVP/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, methods, graphics, grDevices, MASS, bigmemory</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppEigen, RcppProgress, BH, bigmemory</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaolei Liu &lt;xll198708@gmail.com&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Lilin Yin [aut],
  Haohao Zhang [aut],
  Zhenshuang Tang [aut],
  Jingya Xu [aut],
  Dong Yin [aut],
  Zhiwu Zhang [aut],
  Xiaohui Yuan [aut],
  Mengjin Zhu [aut],
  Shuhong Zhao [aut],
  Xinyun Li [aut],
  Qishan Wang [ctb],
  Feng Tian [ctb],
  Hyunmin Kang [ctb],
  Xiang Zhou [ctb],
  Xiaolei Liu [cre, aut, cph]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-27 08:51:59 UTC; hyacz</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-27 09:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='MVP'>MVP, Memory-efficient, Visualization-enhanced, Parallel-accelerated</h2><span id='topic+MVP'></span>

<h3>Description</h3>

<p>Object 1: To perform GWAS using General Linear Model (GLM), Mixed Linear Model (MLM), and FarmCPU model
Object 2: To calculate kinship among individuals using Varaden method
Object 3: Estimate variance components using EMMA, FaST-LMM, and HE regression
Object 4: Generate high-quality figures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP(
  phe,
  geno,
  map,
  K = NULL,
  nPC.GLM = NULL,
  nPC.MLM = NULL,
  nPC.FarmCPU = NULL,
  CV.GLM = NULL,
  CV.MLM = NULL,
  CV.FarmCPU = NULL,
  REML = NULL,
  priority = "speed",
  ncpus = detectCores(logical = FALSE),
  vc.method = c("BRENT", "EMMA", "HE"),
  method = c("GLM", "MLM", "FarmCPU"),
  p.threshold = NA,
  QTN.threshold = 0.01,
  method.bin = "static",
  bin.size = c(5e+05, 5e+06, 5e+07),
  bin.selection = seq(10, 100, 10),
  maxLoop = 10,
  permutation.threshold = FALSE,
  permutation.rep = 100,
  memo = NULL,
  outpath = getwd(),
  col = c("#4197d8", "#f8c120", "#413496", "#495226", "#d60b6f", "#e66519", "#d581b7",
    "#83d3ad", "#7c162c", "#26755d"),
  file.output = TRUE,
  file.type = "jpg",
  dpi = 300,
  threshold = 0.05,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP_+3A_phe">phe</code></td>
<td>
<p>phenotype, n * 2 matrix, n is sample size</p>
</td></tr>
<tr><td><code id="MVP_+3A_geno">geno</code></td>
<td>
<p>Genotype in bigmatrix format; m * n, m is marker size, n is sample size</p>
</td></tr>
<tr><td><code id="MVP_+3A_map">map</code></td>
<td>
<p>SNP map information, SNP name, Chr, Pos</p>
</td></tr>
<tr><td><code id="MVP_+3A_k">K</code></td>
<td>
<p>Kinship, Covariance matrix(n * n) for random effects, must be positive semi-definite</p>
</td></tr>
<tr><td><code id="MVP_+3A_npc.glm">nPC.GLM</code></td>
<td>
<p>number of PCs added as fixed effects in GLM</p>
</td></tr>
<tr><td><code id="MVP_+3A_npc.mlm">nPC.MLM</code></td>
<td>
<p>number of PCs added as fixed effects in MLM</p>
</td></tr>
<tr><td><code id="MVP_+3A_npc.farmcpu">nPC.FarmCPU</code></td>
<td>
<p>number of PCs added as fixed effects in FarmCPU</p>
</td></tr>
<tr><td><code id="MVP_+3A_cv.glm">CV.GLM</code></td>
<td>
<p>covariates added in GLM</p>
</td></tr>
<tr><td><code id="MVP_+3A_cv.mlm">CV.MLM</code></td>
<td>
<p>covariates added in MLM</p>
</td></tr>
<tr><td><code id="MVP_+3A_cv.farmcpu">CV.FarmCPU</code></td>
<td>
<p>covariates added in FarmCPU</p>
</td></tr>
<tr><td><code id="MVP_+3A_reml">REML</code></td>
<td>
<p>a list contains ve and vg</p>
</td></tr>
<tr><td><code id="MVP_+3A_priority">priority</code></td>
<td>
<p>speed or memory</p>
</td></tr>
<tr><td><code id="MVP_+3A_ncpus">ncpus</code></td>
<td>
<p>number of cpus used for parallel</p>
</td></tr>
<tr><td><code id="MVP_+3A_vc.method">vc.method</code></td>
<td>
<p>methods for estimating variance component(&quot;EMMA&quot; or &quot;HE&quot; or &quot;BRENT&quot;)</p>
</td></tr>
<tr><td><code id="MVP_+3A_method">method</code></td>
<td>
<p>the GWAS model, &quot;GLM&quot;, &quot;MLM&quot;, and &quot;FarmCPU&quot;, models can be selected simutaneously, i.e. c(&quot;GLM&quot;, &quot;MLM&quot;, &quot;FarmCPU&quot;)</p>
</td></tr>
<tr><td><code id="MVP_+3A_p.threshold">p.threshold</code></td>
<td>
<p>if all p values generated in the first iteration are bigger than p.threshold, FarmCPU stops</p>
</td></tr>
<tr><td><code id="MVP_+3A_qtn.threshold">QTN.threshold</code></td>
<td>
<p>in second and later iterations, only SNPs with lower p-values than QTN.threshold have chances to be selected as pseudo QTNs</p>
</td></tr>
<tr><td><code id="MVP_+3A_method.bin">method.bin</code></td>
<td>
<p>'static' or 'FaST-LMM'</p>
</td></tr>
<tr><td><code id="MVP_+3A_bin.size">bin.size</code></td>
<td>
<p>window size in genome</p>
</td></tr>
<tr><td><code id="MVP_+3A_bin.selection">bin.selection</code></td>
<td>
<p>a vector, how many windows selected</p>
</td></tr>
<tr><td><code id="MVP_+3A_maxloop">maxLoop</code></td>
<td>
<p>maximum number of iterations</p>
</td></tr>
<tr><td><code id="MVP_+3A_permutation.threshold">permutation.threshold</code></td>
<td>
<p>if use a permutation cutoff or not (bonferroni cutoff)</p>
</td></tr>
<tr><td><code id="MVP_+3A_permutation.rep">permutation.rep</code></td>
<td>
<p>number of permutation replicates</p>
</td></tr>
<tr><td><code id="MVP_+3A_memo">memo</code></td>
<td>
<p>Character. A text marker on output files</p>
</td></tr>
<tr><td><code id="MVP_+3A_outpath">outpath</code></td>
<td>
<p>Effective only when file.output = TRUE, determines the path of the output file</p>
</td></tr>
<tr><td><code id="MVP_+3A_col">col</code></td>
<td>
<p>for color of points in each chromosome on manhattan plot</p>
</td></tr>
<tr><td><code id="MVP_+3A_file.output">file.output</code></td>
<td>
<p>whether to output files or not</p>
</td></tr>
<tr><td><code id="MVP_+3A_file.type">file.type</code></td>
<td>
<p>figure formats, &quot;jpg&quot;, &quot;tiff&quot;</p>
</td></tr>
<tr><td><code id="MVP_+3A_dpi">dpi</code></td>
<td>
<p>resolution for output figures</p>
</td></tr>
<tr><td><code id="MVP_+3A_threshold">threshold</code></td>
<td>
<p>a cutoff line on manhattan plot, 0.05/marker size</p>
</td></tr>
<tr><td><code id="MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Aug 30, 2017
Last update: Dec 14, 2018
</p>


<h3>Value</h3>

<p>a m * 2 matrix, the first column is the SNP effect, the second column is the P values
Output: MVP.return$map - SNP map information, SNP name, Chr, Pos
Output: MVP.return$glm.results - p-values obtained by GLM method
Output: MVP.return$mlm.results - p-values obtained by MLM method
Output: MVP.return$farmcpu.results - p-values obtained by FarmCPU method
</p>


<h3>Author(s)</h3>

<p>Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))
mapPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.map", package = "rMVP")
map &lt;- read.table(mapPath , head = TRUE)

opts &lt;- options(rMVP.OutputLog2File = FALSE)

mvp &lt;- MVP(phe=phenotype, geno=genotype, map=map, maxLoop=3,
  method=c("GLM", "MLM", "FarmCPU"), file.output=FALSE, ncpus=1)
str(mvp)

options(opts)

</code></pre>

<hr>
<h2 id='MVP.BRENT.Vg.Ve'>MVP.BRENT.Vg.Ve
variance component estimation using the BRENT method</h2><span id='topic+MVP.BRENT.Vg.Ve'></span>

<h3>Description</h3>

<p>MVP.BRENT.Vg.Ve
variance component estimation using the BRENT method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.BRENT.Vg.Ve(y, X, eigenK, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.BRENT.Vg.Ve_+3A_y">y</code></td>
<td>
<p>phenotype</p>
</td></tr>
<tr><td><code id="MVP.BRENT.Vg.Ve_+3A_x">X</code></td>
<td>
<p>covariate matrix, the first column is 1s</p>
</td></tr>
<tr><td><code id="MVP.BRENT.Vg.Ve_+3A_eigenk">eigenK</code></td>
<td>
<p>eigen of Kinship matrix</p>
</td></tr>
<tr><td><code id="MVP.BRENT.Vg.Ve_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vg, ve, and delta
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))

eigenK &lt;- eigen(MVP.K.VanRaden(genotype, cpu=1))
vc &lt;- MVP.BRENT.Vg.Ve(y=phenotype[,2], X=matrix(1, nrow(phenotype)), eigenK=eigenK)
print(vc)


</code></pre>

<hr>
<h2 id='MVP.calk'>Calculate Kinship matrix by Blocking strategy</h2><span id='topic+MVP.calk'></span>

<h3>Description</h3>

<p>Build date: Apr 14, 2021
Last update: Apr 14, 2021
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.calk(M, step = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.calk_+3A_m">M</code></td>
<td>
<p>Genotype, m * n, m is marker size, n is population size</p>
</td></tr>
<tr><td><code id="MVP.calk_+3A_step">step</code></td>
<td>
<p>Number of markers processed at one time</p>
</td></tr>
</table>


<h3>Value</h3>

<p>K, n * n matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))

K &lt;- MVP.calk(genotype)

</code></pre>

<hr>
<h2 id='MVP.Data'>MVP.Data: To prepare data for MVP package
Author: Xiaolei Liu, Lilin Yin and Haohao Zhang
Build date: Aug 30, 2016
Last update: Sep 12, 2018</h2><span id='topic+MVP.Data'></span>

<h3>Description</h3>

<p>MVP.Data: To prepare data for MVP package
Author: Xiaolei Liu, Lilin Yin and Haohao Zhang
Build date: Aug 30, 2016
Last update: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data(
  fileMVP = NULL,
  fileVCF = NULL,
  fileHMP = NULL,
  fileBed = NULL,
  fileNum = NULL,
  fileMap = NULL,
  filePhe = NULL,
  fileInd = NULL,
  fileKin = NULL,
  filePC = NULL,
  out = "mvp",
  sep.num = "\t",
  auto_transpose = TRUE,
  sep.map = "\t",
  sep.phe = "\t",
  sep.kin = "\t",
  sep.pc = "\t",
  type.geno = "char",
  pheno_cols = NULL,
  SNP.impute = "Major",
  maxLine = 10000,
  priority = "speed",
  pcs.keep = 5,
  verbose = TRUE,
  ncpus = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data_+3A_filemvp">fileMVP</code></td>
<td>
<p>Genotype in MVP format</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filevcf">fileVCF</code></td>
<td>
<p>Genotype in VCF format</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filehmp">fileHMP</code></td>
<td>
<p>Genotype in hapmap format</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filebed">fileBed</code></td>
<td>
<p>Genotype in PLINK binary format</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filenum">fileNum</code></td>
<td>
<p>Genotype in numeric format; pure 0, 1, 2 matrix; m * n, m is marker size, n is sample size</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filemap">fileMap</code></td>
<td>
<p>SNP map information, there are three columns, including SNP_ID, Chromosome, and Position</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filephe">filePhe</code></td>
<td>
<p>Phenotype, the first column is taxa name, the subsequent columns are traits</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_fileind">fileInd</code></td>
<td>
<p>Individual name file</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filekin">fileKin</code></td>
<td>
<p>Kinship that represents relationship among individuals, n * n matrix, n is sample size</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_filepc">filePC</code></td>
<td>
<p>Principal components, n*npc, n is sample size, npc is number of top columns of principal components</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_out">out</code></td>
<td>
<p>prefix of output file name</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_sep.num">sep.num</code></td>
<td>
<p>seperator for numeric file.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_auto_transpose">auto_transpose</code></td>
<td>
<p>Whether to automatically transpose numeric genotypes,
the default is True, which will identify the most one of the rows or 
columns as a marker, If set to False, the row represents the marker and
the column represents the individual.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_sep.map">sep.map</code></td>
<td>
<p>seperator for map file.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_sep.phe">sep.phe</code></td>
<td>
<p>seperator for phenotype file.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_sep.kin">sep.kin</code></td>
<td>
<p>seperator for Kinship file.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_sep.pc">sep.pc</code></td>
<td>
<p>seperator for PC file.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_type.geno">type.geno</code></td>
<td>
<p>type parameter in bigmemory, genotype data. The default is char, it is highly recommended *NOT* to modify this parameter.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_pheno_cols">pheno_cols</code></td>
<td>
<p>Extract which columns of the phenotype file (including individual IDs)</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_snp.impute">SNP.impute</code></td>
<td>
<p>&quot;Left&quot;, &quot;Middle&quot;, &quot;Right&quot;, or NULL for skip impute.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_maxline">maxLine</code></td>
<td>
<p>number of SNPs, only used for saving memory when calculate kinship matrix</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_priority">priority</code></td>
<td>
<p>&quot;speed&quot; or &quot;memory&quot;</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_pcs.keep">pcs.keep</code></td>
<td>
<p>how many PCs to keep</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_ncpus">ncpus</code></td>
<td>
<p>The number of threads used, if NULL, (logical core number - 1) is automatically used</p>
</td></tr>
<tr><td><code id="MVP.Data_+3A_...">...</code></td>
<td>
<p>Compatible with DEPRECATED parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
k.desc, k.bin: Kinship matrix in bigmemory format
pc.desc, pc.bin: PC matrix in bigmemory format
Requirement: fileHMP, fileBed, and fileNum can not input at the same time
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
bfilePath &lt;- file.path(system.file("extdata", "02_bfile", package = "rMVP"), "mvp")
opts &lt;- options(rMVP.OutputLog2File = FALSE)

MVP.Data(fileBed=bfilePath, out=tempfile("outfile"), ncpus=1)

options(opts)

</code></pre>

<hr>
<h2 id='MVP.Data.Bfile2MVP'>MVP.Data.Bfile2MVP: To transform plink binary data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.Bfile2MVP'></span>

<h3>Description</h3>

<p>MVP.Data.Bfile2MVP: To transform plink binary data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.Bfile2MVP(
  bfile,
  out = "mvp",
  maxLine = 10000,
  priority = "speed",
  type.geno = "char",
  threads = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_bfile">bfile</code></td>
<td>
<p>Genotype in binary format (.bed, .bim, .fam)</p>
</td></tr>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_maxline">maxLine</code></td>
<td>
<p>the max number of line to write to big matrix for each loop</p>
</td></tr>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_priority">priority</code></td>
<td>
<p>'memory' or 'speed'</p>
</td></tr>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_type.geno">type.geno</code></td>
<td>
<p>the type of genotype elements</p>
</td></tr>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_threads">threads</code></td>
<td>
<p>number of thread for transforming</p>
</td></tr>
<tr><td><code id="MVP.Data.Bfile2MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>number of individuals and markers.
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
bfilePath &lt;- file.path(system.file("extdata", "02_bfile", package = "rMVP"), "mvp")

MVP.Data.Bfile2MVP(bfilePath, tempfile("outfile"), threads=1)


</code></pre>

<hr>
<h2 id='MVP.Data.Hapmap2MVP'>MVP.Data.Hapmap2MVP: To transform Hapmap data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.Hapmap2MVP'></span>

<h3>Description</h3>

<p>MVP.Data.Hapmap2MVP: To transform Hapmap data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.Hapmap2MVP(
  hmp_file,
  out = "mvp",
  maxLine = 10000,
  type.geno = "char",
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.Hapmap2MVP_+3A_hmp_file">hmp_file</code></td>
<td>
<p>Genotype in Hapmap format</p>
</td></tr>
<tr><td><code id="MVP.Data.Hapmap2MVP_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.Hapmap2MVP_+3A_maxline">maxLine</code></td>
<td>
<p>the max number of line to write to big matrix for each loop</p>
</td></tr>
<tr><td><code id="MVP.Data.Hapmap2MVP_+3A_type.geno">type.geno</code></td>
<td>
<p>the type of genotype elements</p>
</td></tr>
<tr><td><code id="MVP.Data.Hapmap2MVP_+3A_threads">threads</code></td>
<td>
<p>number of thread for transforming</p>
</td></tr>
<tr><td><code id="MVP.Data.Hapmap2MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>number of individuals and markers.
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
hapmapPath &lt;- system.file("extdata", "03_hapmap", "mvp.hmp.txt", package = "rMVP")

MVP.Data.Hapmap2MVP(hapmapPath, tempfile("outfile"), threads=1)


</code></pre>

<hr>
<h2 id='MVP.Data.impute'>MVP.Data.impute: To impute the missing genotype
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.impute'></span>

<h3>Description</h3>

<p>MVP.Data.impute: To impute the missing genotype
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.impute(
  mvp_prefix,
  out = NULL,
  method = "Major",
  ncpus = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.impute_+3A_mvp_prefix">mvp_prefix</code></td>
<td>
<p>the prefix of mvp file</p>
</td></tr>
<tr><td><code id="MVP.Data.impute_+3A_out">out</code></td>
<td>
<p>the prefix of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.impute_+3A_method">method</code></td>
<td>
<p>'Major', 'Minor', &quot;Middle&quot;</p>
</td></tr>
<tr><td><code id="MVP.Data.impute_+3A_ncpus">ncpus</code></td>
<td>
<p>number of threads for imputing</p>
</td></tr>
<tr><td><code id="MVP.Data.impute_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
Output files:
imputed genotype file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mvpPath &lt;- file.path(system.file("extdata", "05_mvp", package = "rMVP"), "mvp")

MVP.Data.impute(mvpPath, tempfile("outfile"), ncpus=1)

</code></pre>

<hr>
<h2 id='MVP.Data.Kin'>Kinship</h2><span id='topic+MVP.Data.Kin'></span>

<h3>Description</h3>

<p>Kinship
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.Kin(
  fileKin = TRUE,
  mvp_prefix = "mvp",
  out = NULL,
  priority = "speed",
  sep = "\t",
  cpus = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.Kin_+3A_filekin">fileKin</code></td>
<td>
<p>Kinship that represents relationship among individuals, n * n matrix, n is sample size</p>
</td></tr>
<tr><td><code id="MVP.Data.Kin_+3A_mvp_prefix">mvp_prefix</code></td>
<td>
<p>Prefix for mvp format files</p>
</td></tr>
<tr><td><code id="MVP.Data.Kin_+3A_out">out</code></td>
<td>
<p>prefix of output file name</p>
</td></tr>
<tr><td><code id="MVP.Data.Kin_+3A_priority">priority</code></td>
<td>
<p>&quot;speed&quot; or &quot;memory&quot;</p>
</td></tr>
<tr><td><code id="MVP.Data.Kin_+3A_sep">sep</code></td>
<td>
<p>seperator for Kinship file.</p>
</td></tr>
<tr><td><code id="MVP.Data.Kin_+3A_cpus">cpus</code></td>
<td>
<p>the number of cpu</p>
</td></tr>
<tr><td><code id="MVP.Data.Kin_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
&lt;out&gt;.kin.bin
&lt;out&gt;.kin.desc
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
geno &lt;- file.path(system.file("extdata", "06_mvp-impute", package = "rMVP"), "mvp.imp")

MVP.Data.Kin(TRUE, mvp_prefix=geno, out=tempfile("outfile"), cpus=1)


</code></pre>

<hr>
<h2 id='MVP.Data.Map'>MVP.Data.Map: To check map file
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.Map'></span>

<h3>Description</h3>

<p>MVP.Data.Map: To check map file
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.Map(
  map,
  out = "mvp",
  cols = 1:5,
  header = TRUE,
  sep = "\t",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.Map_+3A_map">map</code></td>
<td>
<p>the name of map file or map object(data.frame or matrix)</p>
</td></tr>
<tr><td><code id="MVP.Data.Map_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.Map_+3A_cols">cols</code></td>
<td>
<p>selected columns</p>
</td></tr>
<tr><td><code id="MVP.Data.Map_+3A_header">header</code></td>
<td>
<p>whether the file contains header</p>
</td></tr>
<tr><td><code id="MVP.Data.Map_+3A_sep">sep</code></td>
<td>
<p>seperator of the file</p>
</td></tr>
<tr><td><code id="MVP.Data.Map_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
&lt;out&gt;.map
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mapPath &lt;- system.file("extdata", "05_mvp", "mvp.geno.map", package = "rMVP")

MVP.Data.Map(mapPath, tempfile("outfile"))

</code></pre>

<hr>
<h2 id='MVP.Data.MVP2Bfile'>MVP.Data.MVP2Bfile: To transform MVP data to binary format
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.MVP2Bfile'></span>

<h3>Description</h3>

<p>MVP.Data.MVP2Bfile: To transform MVP data to binary format
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.MVP2Bfile(
  bigmat,
  map,
  pheno = NULL,
  out = "mvp.plink",
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.MVP2Bfile_+3A_bigmat">bigmat</code></td>
<td>
<p>Genotype in bigmatrix format (0,1,2)</p>
</td></tr>
<tr><td><code id="MVP.Data.MVP2Bfile_+3A_map">map</code></td>
<td>
<p>the map file</p>
</td></tr>
<tr><td><code id="MVP.Data.MVP2Bfile_+3A_pheno">pheno</code></td>
<td>
<p>the phenotype file</p>
</td></tr>
<tr><td><code id="MVP.Data.MVP2Bfile_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.MVP2Bfile_+3A_threads">threads</code></td>
<td>
<p>number of thread for transforming</p>
</td></tr>
<tr><td><code id="MVP.Data.MVP2Bfile_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
Output files:
.bed, .bim, .fam
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
bigmat &lt;- as.big.matrix(matrix(1:6, 3, 2))
map &lt;- matrix(c("rs1", "rs2", "rs3", 1, 1, 1, 10, 20, 30), 3, 3)

MVP.Data.MVP2Bfile(bigmat, map, out=tempfile("outfile"), threads=1)


</code></pre>

<hr>
<h2 id='MVP.Data.Numeric2MVP'>MVP.Data.Numeric2MVP: To transform Numeric data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.Numeric2MVP'></span>

<h3>Description</h3>

<p>MVP.Data.Numeric2MVP: To transform Numeric data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.Numeric2MVP(
  num_file,
  map_file,
  out = "mvp",
  maxLine = 10000,
  priority = "speed",
  row_names = FALSE,
  col_names = FALSE,
  type.geno = "char",
  auto_transpose = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_num_file">num_file</code></td>
<td>
<p>Genotype in Numeric format (0,1,2)</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_map_file">map_file</code></td>
<td>
<p>Genotype map file, SNP_name, Chr, Pos</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_maxline">maxLine</code></td>
<td>
<p>the max number of line to write to big matrix for each loop</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_priority">priority</code></td>
<td>
<p>'memory' or 'speed'</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_row_names">row_names</code></td>
<td>
<p>whether the numeric genotype has row names</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_col_names">col_names</code></td>
<td>
<p>whether the numeric genotype has column names</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_type.geno">type.geno</code></td>
<td>
<p>the type of genotype elements</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_auto_transpose">auto_transpose</code></td>
<td>
<p>whether to detecte the row and column</p>
</td></tr>
<tr><td><code id="MVP.Data.Numeric2MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>number of individuals and markers.
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
numericPath &lt;- system.file("extdata", "04_numeric", "mvp.num", package = "rMVP")
mapPath &lt;- system.file("extdata", "04_numeric", "mvp.map", package = "rMVP")
MVP.Data.Numeric2MVP(numericPath, mapPath, tempfile("outfile"))


</code></pre>

<hr>
<h2 id='MVP.Data.PC'>Principal component analysis</h2><span id='topic+MVP.Data.PC'></span>

<h3>Description</h3>

<p>Principal component analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.PC(
  filePC = TRUE,
  mvp_prefix = "mvp",
  K = NULL,
  out = NULL,
  pcs.keep = 5,
  priority = "speed",
  sep = "\t",
  cpus = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.PC_+3A_filepc">filePC</code></td>
<td>
<p>Principal components, n*npc, n is sample size, npc is number of top columns of principal components</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_mvp_prefix">mvp_prefix</code></td>
<td>
<p>Prefix for mvp format files</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_k">K</code></td>
<td>
<p>Kinship matrix</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_out">out</code></td>
<td>
<p>prefix of output file name</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_pcs.keep">pcs.keep</code></td>
<td>
<p>how many PCs to keep</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_priority">priority</code></td>
<td>
<p>speed or memory</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_sep">sep</code></td>
<td>
<p>seperator for PC file.</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_cpus">cpus</code></td>
<td>
<p>the number of cpu</p>
</td></tr>
<tr><td><code id="MVP.Data.PC_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
&lt;out&gt;.pc.bin
&lt;out&gt;.pc.desc
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
geno &lt;- file.path(system.file("extdata", "06_mvp-impute", package = "rMVP"), "mvp.imp")

MVP.Data.PC(TRUE, mvp_prefix=geno, out=tempfile("outfile"), cpus=1)


</code></pre>

<hr>
<h2 id='MVP.Data.Pheno'>MVP.Data.Pheno: To clean up phenotype file
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.Pheno'></span>

<h3>Description</h3>

<p>MVP.Data.Pheno: To clean up phenotype file
Author: Haohao Zhang
Build date: Sep 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.Pheno(
  pheno_file,
  out = "mvp",
  cols = NULL,
  header = TRUE,
  sep = "\t",
  missing = c(NA, "NA", "-9", 9999),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.Pheno_+3A_pheno_file">pheno_file</code></td>
<td>
<p>the name of phenotype file</p>
</td></tr>
<tr><td><code id="MVP.Data.Pheno_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.Pheno_+3A_cols">cols</code></td>
<td>
<p>selected columns</p>
</td></tr>
<tr><td><code id="MVP.Data.Pheno_+3A_header">header</code></td>
<td>
<p>whether the file contains header</p>
</td></tr>
<tr><td><code id="MVP.Data.Pheno_+3A_sep">sep</code></td>
<td>
<p>seperator of the file</p>
</td></tr>
<tr><td><code id="MVP.Data.Pheno_+3A_missing">missing</code></td>
<td>
<p>the missing value</p>
</td></tr>
<tr><td><code id="MVP.Data.Pheno_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
Output files:
cleaned phenotype file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")

MVP.Data.Pheno(phePath, out=tempfile("outfile"))

</code></pre>

<hr>
<h2 id='MVP.Data.VCF2MVP'>MVP.Data.VCF2MVP: To transform vcf data to MVP package
Author: Haohao Zhang
Build date: Sep 12, 2018</h2><span id='topic+MVP.Data.VCF2MVP'></span>

<h3>Description</h3>

<p>Accept the | or / separated markers, any variant sites that are not 0 or 1 will be considered NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Data.VCF2MVP(
  vcf_file,
  out = "mvp",
  maxLine = 10000,
  type.geno = "char",
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Data.VCF2MVP_+3A_vcf_file">vcf_file</code></td>
<td>
<p>Genotype in VCF format</p>
</td></tr>
<tr><td><code id="MVP.Data.VCF2MVP_+3A_out">out</code></td>
<td>
<p>the name of output file</p>
</td></tr>
<tr><td><code id="MVP.Data.VCF2MVP_+3A_maxline">maxLine</code></td>
<td>
<p>the max number of line to write to big matrix for each loop</p>
</td></tr>
<tr><td><code id="MVP.Data.VCF2MVP_+3A_type.geno">type.geno</code></td>
<td>
<p>the type of genotype elements</p>
</td></tr>
<tr><td><code id="MVP.Data.VCF2MVP_+3A_threads">threads</code></td>
<td>
<p>number of thread for transforming</p>
</td></tr>
<tr><td><code id="MVP.Data.VCF2MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>number of individuals and markers.
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
vcfPath &lt;- system.file("extdata", "01_vcf", "mvp.vcf", package = "rMVP")

MVP.Data.VCF2MVP(vcfPath, tempfile("outfile"), threads=1)


</code></pre>

<hr>
<h2 id='MVP.EMMA.Vg.Ve'>Estimate variance components using EMMA</h2><span id='topic+MVP.EMMA.Vg.Ve'></span>

<h3>Description</h3>

<p>Build date: August 30, 2016
Last update: January 27, 2017
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.EMMA.Vg.Ve(y, X, K, ngrids = 100, llim = -10, ulim = 10, esp = 1e-10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_y">y</code></td>
<td>
<p>phenotype, n * 2</p>
</td></tr>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_x">X</code></td>
<td>
<p>covariate matrix, the first column is 1s</p>
</td></tr>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_k">K</code></td>
<td>
<p>Kinship matrix</p>
</td></tr>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_ngrids">ngrids</code></td>
<td>
<p>parameters for estimating vg and ve</p>
</td></tr>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_llim">llim</code></td>
<td>
<p>parameters for estimating vg and ve</p>
</td></tr>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_ulim">ulim</code></td>
<td>
<p>parameters for estimating vg and ve</p>
</td></tr>
<tr><td><code id="MVP.EMMA.Vg.Ve_+3A_esp">esp</code></td>
<td>
<p>parameters for estimating vg and ve</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output: REML - maximum log likelihood
Output: delta - exp(root)
Output: ve - residual variance
Output: vg - genetic variance
</p>


<h3>Author(s)</h3>

<p>EMMA (Kang et. al. Genetics, 2008), Modified only for speed up by Xiaolei Liu and Lilin Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))

K &lt;- MVP.K.VanRaden(genotype, cpu=1)
vc &lt;- MVP.EMMA.Vg.Ve(y=phenotype[,2], X=matrix(1, nrow(phenotype)), K=K)
print(vc)


</code></pre>

<hr>
<h2 id='MVP.FarmCPU'>Perform GWAS using FarmCPU method</h2><span id='topic+MVP.FarmCPU'></span>

<h3>Description</h3>

<p>Date build: Febuary 24, 2013
Last update: May 25, 2017
Requirement: Y, GD, and CV should have same taxa order. GD and GM should have the same order on SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.FarmCPU(
  phe,
  geno,
  map,
  CV = NULL,
  P = NULL,
  method.sub = "reward",
  method.sub.final = "reward",
  method.bin = c("EMMA", "static", "FaST-LMM"),
  bin.size = c(5e+05, 5e+06, 5e+07),
  bin.selection = seq(10, 100, 10),
  memo = "MVP.FarmCPU",
  Prior = NULL,
  ncpus = 2,
  maxLoop = 10,
  threshold.output = 0.01,
  converge = 1,
  iteration.output = FALSE,
  p.threshold = NA,
  QTN.threshold = 0.01,
  bound = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.FarmCPU_+3A_phe">phe</code></td>
<td>
<p>phenotype, n by t matrix, n is sample size, t is number of phenotypes</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_geno">geno</code></td>
<td>
<p>genotype, m by n matrix, m is marker size, n is sample size. This is Pure Genotype Data Matrix(GD). THERE IS NO COLUMN FOR TAXA.</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_map">map</code></td>
<td>
<p>SNP map information, m by 3 matrix, m is marker size, the three columns are SNP_ID, Chr, and Pos</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_cv">CV</code></td>
<td>
<p>covariates, n by c matrix, n is sample size, c is number of covariates</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_p">P</code></td>
<td>
<p>start p values for all SNPs</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_method.sub">method.sub</code></td>
<td>
<p>method used in substitution process, five options: 'penalty', 'reward', 'mean', 'median', or 'onsite'</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_method.sub.final">method.sub.final</code></td>
<td>
<p>method used in substitution process, five options: 'penalty', 'reward', 'mean', 'median', or 'onsite'</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_method.bin">method.bin</code></td>
<td>
<p>method for selecting the most appropriate bins, three options: 'static', 'EMMA' or 'FaST-LMM'</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_bin.size">bin.size</code></td>
<td>
<p>bin sizes for all iterations, a vector, the bin size is always from large to small</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_bin.selection">bin.selection</code></td>
<td>
<p>number of selected bins in each iteration, a vector</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_memo">memo</code></td>
<td>
<p>a marker on output file name</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_prior">Prior</code></td>
<td>
<p>prior information, four columns, which are SNP_ID, Chr, Pos, P-value</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_ncpus">ncpus</code></td>
<td>
<p>number of threads used for parallele computation</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_maxloop">maxLoop</code></td>
<td>
<p>maximum number of iterations</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_threshold.output">threshold.output</code></td>
<td>
<p>only the GWAS results with p-values lower than threshold.output will be output</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_converge">converge</code></td>
<td>
<p>a number, 0 to 1, if selected pseudo QTNs in the last and the second last iterations have a certain probality (the probability is converge) of overlap, the loop will stop</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_iteration.output">iteration.output</code></td>
<td>
<p>whether to output results of all iterations</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_p.threshold">p.threshold</code></td>
<td>
<p>if all p values generated in the first iteration are bigger than p.threshold, FarmCPU stops</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_qtn.threshold">QTN.threshold</code></td>
<td>
<p>in second and later iterations, only SNPs with lower p-values than QTN.threshold have chances to be selected as pseudo QTNs</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_bound">bound</code></td>
<td>
<p>maximum number of SNPs selected as pseudo QTNs in each iteration</p>
</td></tr>
<tr><td><code id="MVP.FarmCPU_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a m by 4 results matrix, m is marker size, the four columns are SNP_ID, Chr, Pos, and p-value
</p>


<h3>Author(s)</h3>

<p>Xiaolei Liu and Zhiwu Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
idx &lt;- !is.na(phenotype[, 2])
phenotype &lt;- phenotype[idx, ]
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
genotype &lt;- deepcopy(genotype, cols=idx)
print(dim(genotype))
mapPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.map", package = "rMVP")
map &lt;- read.table(mapPath , head = TRUE)

farmcpu &lt;- MVP.FarmCPU(phe=phenotype,geno=genotype,map=map,maxLoop=2,method.bin="static")
str(farmcpu)


</code></pre>

<hr>
<h2 id='MVP.FaSTLMM.LL'>Evaluation of the maximum likelihood using FaST-LMM method</h2><span id='topic+MVP.FaSTLMM.LL'></span>

<h3>Description</h3>

<p>Last update: January 11, 2017
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.FaSTLMM.LL(pheno, snp.pool, X0 = NULL, ncpus = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.FaSTLMM.LL_+3A_pheno">pheno</code></td>
<td>
<p>a two-column phenotype matrix</p>
</td></tr>
<tr><td><code id="MVP.FaSTLMM.LL_+3A_snp.pool">snp.pool</code></td>
<td>
<p>matrix for pseudo QTNs</p>
</td></tr>
<tr><td><code id="MVP.FaSTLMM.LL_+3A_x0">X0</code></td>
<td>
<p>covariates matrix</p>
</td></tr>
<tr><td><code id="MVP.FaSTLMM.LL_+3A_ncpus">ncpus</code></td>
<td>
<p>number of threads used for parallel computation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output: beta - beta effect
Output: delta - delta value
Output: LL - log-likelihood
Output: vg - genetic variance
Output: ve - residual variance
</p>


<h3>Author(s)</h3>

<p>Xiaolei Liu (modified)
</p>

<hr>
<h2 id='MVP.GLM'>To perform GWAS with GLM and MLM model and get the P value of SNPs</h2><span id='topic+MVP.GLM'></span>

<h3>Description</h3>

<p>Build date: Aug 30, 2016
Last update: May 25, 2017
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.GLM(phe, geno, CV = NULL, cpu = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.GLM_+3A_phe">phe</code></td>
<td>
<p>phenotype, n * 2 matrix</p>
</td></tr>
<tr><td><code id="MVP.GLM_+3A_geno">geno</code></td>
<td>
<p>Genotype in numeric format, pure 0, 1, 2 matrix; m * n, m is marker size, n is population size</p>
</td></tr>
<tr><td><code id="MVP.GLM_+3A_cv">CV</code></td>
<td>
<p>Covariance, design matrix(n * x) for the fixed effects</p>
</td></tr>
<tr><td><code id="MVP.GLM_+3A_cpu">cpu</code></td>
<td>
<p>number of cpus used for parallel computation</p>
</td></tr>
<tr><td><code id="MVP.GLM_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>m * 2 matrix, the first column is the SNP effect, the second column is the P values
</p>


<h3>Author(s)</h3>

<p>Lilin Yin and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
idx &lt;- !is.na(phenotype[, 2])
phenotype &lt;- phenotype[idx, ]
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
genotype &lt;- deepcopy(genotype, cols=idx)
print(dim(genotype))

glm &lt;- MVP.GLM(phe=phenotype, geno=genotype, cpu=1)
str(glm)

</code></pre>

<hr>
<h2 id='MVP.HE.Vg.Ve'>To estimate variance component using HE regression</h2><span id='topic+MVP.HE.Vg.Ve'></span>

<h3>Description</h3>

<p>Build date: Feb 2, 2017
Last update: Feb 2, 2019
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.HE.Vg.Ve(y, X, K)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.HE.Vg.Ve_+3A_y">y</code></td>
<td>
<p>phenotype</p>
</td></tr>
<tr><td><code id="MVP.HE.Vg.Ve_+3A_x">X</code></td>
<td>
<p>genotype</p>
</td></tr>
<tr><td><code id="MVP.HE.Vg.Ve_+3A_k">K</code></td>
<td>
<p>kinship matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vg, ve, and delta
</p>


<h3>Author(s)</h3>

<p>Translated from C++(GEMMA, Xiang Zhou) to R by: Haohao Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))

K &lt;- MVP.K.VanRaden(genotype, cpu=1)
vc &lt;- MVP.HE.Vg.Ve(y=phenotype[,2], X=matrix(1, nrow(phenotype)), K=K)
print(vc)


</code></pre>

<hr>
<h2 id='MVP.Hist'>Phenotype distribution histogram</h2><span id='topic+MVP.Hist'></span>

<h3>Description</h3>

<p>Phenotype distribution histogram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Hist(
  phe,
  col = c("dodgerblue4", "olivedrab4", "violetred", "darkgoldenrod1", "purple4"),
  breakNum = 15,
  memo = NULL,
  outpath = getwd(),
  test.method = "auto",
  file.type = "pdf",
  file.output = TRUE,
  dpi = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Hist_+3A_phe">phe</code></td>
<td>
<p>phenotype data</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_col">col</code></td>
<td>
<p>The color vector of the histogram. If the number of colors is 
less than break.n, the color will be reused. If the number of colors is 
greater than break.n, only the previous break.n colors will be used.</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_breaknum">breakNum</code></td>
<td>
<p>the number of cells for the histogram. The default value 
is 15.</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_memo">memo</code></td>
<td>
<p>Character. A text marker on output files</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_outpath">outpath</code></td>
<td>
<p>Effective only when file.output = TRUE, determines the path of the output file</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_test.method">test.method</code></td>
<td>
<p>The method used to test the normal distribution. The options are &quot;auto&quot;, 
&quot;Shapiro-Wilk&quot;, &quot;Kolmogorov-Smirnov&quot;, and NULL. When set to &quot;auto&quot;, &quot;Shapiro- Wilk&quot; method,
&quot;Kolmogorov-Smirnov&quot; method will be used when it is greater than 5000, and it will not be 
tested when set to NULL.</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_file.type">file.type</code></td>
<td>
<p>A string or NULL is used to determine the type of output 
file. Can be &quot;jpg&quot;, &quot;pdf&quot;, &quot;tiff&quot;. If it is NULL, it will use 
<code><a href="grDevices.html#topic+dev">dev.new()</a></code> to create a new graphics device 
in the current environment, which may be RStudioGD or the default 
device of the system.</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_file.output">file.output</code></td>
<td>
<p>Logical value. If TRUE, the figures will be generated.</p>
</td></tr>
<tr><td><code id="MVP.Hist_+3A_dpi">dpi</code></td>
<td>
<p>The resolution of the image, specifying how many pixels 
per inch.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
MVP.Phe_Distribution.&lt;trait&gt;.&lt;type&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phe &lt;- read.table(phePath, header=TRUE)

MVP.Hist(phe, file.output = FALSE)

</code></pre>

<hr>
<h2 id='MVP.K.VanRaden'>Calculate Kinship matrix by VanRaden method</h2><span id='topic+MVP.K.VanRaden'></span>

<h3>Description</h3>

<p>Build date: Dec 12, 2016
Last update: Dec 12, 2019
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.K.VanRaden(M, priority = c("speed", "memory"), cpu = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.K.VanRaden_+3A_m">M</code></td>
<td>
<p>Genotype, m * n, m is marker size, n is population size</p>
</td></tr>
<tr><td><code id="MVP.K.VanRaden_+3A_priority">priority</code></td>
<td>
<p>speed or memory</p>
</td></tr>
<tr><td><code id="MVP.K.VanRaden_+3A_cpu">cpu</code></td>
<td>
<p>the number of cpu</p>
</td></tr>
<tr><td><code id="MVP.K.VanRaden_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>K, n * n matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))

K &lt;- MVP.K.VanRaden(genotype, cpu=1)


</code></pre>

<hr>
<h2 id='MVP.MLM'>To perform GWAS with GLM and MLM model and get the P value of SNPs</h2><span id='topic+MVP.MLM'></span>

<h3>Description</h3>

<p>Build date: Aug 30, 2016
Last update: Aug 30, 2016
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.MLM(
  phe,
  geno,
  K = NULL,
  eigenK = NULL,
  CV = NULL,
  REML = NULL,
  cpu = 1,
  vc.method = c("BRENT", "EMMA", "HE"),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.MLM_+3A_phe">phe</code></td>
<td>
<p>phenotype, n * 2 matrix</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_geno">geno</code></td>
<td>
<p>genotype, m * n, m is marker size, n is population size</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_k">K</code></td>
<td>
<p>Kinship, Covariance matrix(n * n) for random effects; must be positive semi-definite</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_eigenk">eigenK</code></td>
<td>
<p>list of eigen Kinship</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_cv">CV</code></td>
<td>
<p>covariates</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_reml">REML</code></td>
<td>
<p>a list that contains ve and vg</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_cpu">cpu</code></td>
<td>
<p>number of cpus used for parallel computation</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_vc.method">vc.method</code></td>
<td>
<p>the methods for estimating variance component(&quot;emma&quot; or &quot;he&quot; or &quot;brent&quot;)</p>
</td></tr>
<tr><td><code id="MVP.MLM_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>results: a m * 2 matrix, the first column is the SNP effect, the second column is the P values
</p>


<h3>Author(s)</h3>

<p>Lilin Yin and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phePath &lt;- system.file("extdata", "07_other", "mvp.phe", package = "rMVP")
phenotype &lt;- read.table(phePath, header=TRUE)
idx &lt;- !is.na(phenotype[, 2])
phenotype &lt;- phenotype[idx, ]
print(dim(phenotype))
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
genotype &lt;- deepcopy(genotype, cols=idx)
print(dim(genotype))
K &lt;- MVP.K.VanRaden(genotype, cpu=1)

mlm &lt;- MVP.MLM(phe=phenotype, geno=genotype, K=K, cpu=1)
str(mlm)


</code></pre>

<hr>
<h2 id='MVP.PCA'>Principal Component Analysis</h2><span id='topic+MVP.PCA'></span>

<h3>Description</h3>

<p>Build date: Dec 14, 2016
Last update: Oct 29, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.PCA(
  M = NULL,
  K = NULL,
  priority = c("speed", "memory"),
  pcs.keep = 5,
  cpu = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.PCA_+3A_m">M</code></td>
<td>
<p>Genotype in numeric format, pure 0, 1, 2 matrix; m * n, m is marker size, n is population size</p>
</td></tr>
<tr><td><code id="MVP.PCA_+3A_k">K</code></td>
<td>
<p>kinship matrix</p>
</td></tr>
<tr><td><code id="MVP.PCA_+3A_priority">priority</code></td>
<td>
<p>speed or memory</p>
</td></tr>
<tr><td><code id="MVP.PCA_+3A_pcs.keep">pcs.keep</code></td>
<td>
<p>maximum number of PCs for output</p>
</td></tr>
<tr><td><code id="MVP.PCA_+3A_cpu">cpu</code></td>
<td>
<p>the number of cpu</p>
</td></tr>
<tr><td><code id="MVP.PCA_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output: PCs - a n * npc matrix of top number of PCs, n is population size and npc is @param pcs.keep
</p>


<h3>Author(s)</h3>

<p>Xiaolei Liu, Lilin Yin and Haohao Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
genotype &lt;- attach.big.matrix(genoPath)
print(dim(genotype))

pca &lt;- MVP.PCA(M=genotype, cpu=1)
str(pca)


</code></pre>

<hr>
<h2 id='MVP.PCAplot'>PCA Plot</h2><span id='topic+MVP.PCAplot'></span>

<h3>Description</h3>

<p>PCA Plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.PCAplot(
  PCA,
  memo = "MVP",
  col = NULL,
  pch = NULL,
  class = NULL,
  legend.pos = "topright",
  Ncluster = 1,
  plot3D = FALSE,
  file.type = "pdf",
  dpi = 300,
  box = FALSE,
  file.output = TRUE,
  outpath = getwd(),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.PCAplot_+3A_pca">PCA</code></td>
<td>
<p>Principal component analysis result, 2-column matrix</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_memo">memo</code></td>
<td>
<p>the prefix of the output image file.</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_col">col</code></td>
<td>
<p>colors for each cluster</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_pch">pch</code></td>
<td>
<p>Either an integer specifying a symbol or a single character to be 
used as the default in plotting points. See <code><a href="graphics.html#topic+points">points</a></code> 
for possible values and their interpretation. Note that only integers and
single-character strings can be set as a graphics parameter (and not NA 
nor NULL).</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_class">class</code></td>
<td>
<p>the class of all individuals, for example: &quot;breed&quot;, &quot;location&quot;</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_legend.pos">legend.pos</code></td>
<td>
<p>position of legend. default is &quot;topright&quot;</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_ncluster">Ncluster</code></td>
<td>
<p>cluster number</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_plot3d">plot3D</code></td>
<td>
<p>(DEPRECATED)if TRUE, plot PC figure in 3D format, it can be only used in windows and mac operation system, &quot;rgl&quot; package should be installed beforehead</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_file.type">file.type</code></td>
<td>
<p>Character. Options are jpg, pdf, and tiff</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_dpi">dpi</code></td>
<td>
<p>Number. Dots per inch for .jpg and .tiff files</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_box">box</code></td>
<td>
<p>Logical value. If TRUE, the border line of Manhattan plot will be added</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_file.output">file.output</code></td>
<td>
<p>Logical value. If TRUE, the figures will be generated.</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_outpath">outpath</code></td>
<td>
<p>Effective only when file.output = TRUE, determines the path of the output file</p>
</td></tr>
<tr><td><code id="MVP.PCAplot_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
MVP.PCA_2D.&lt;type&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genoPath &lt;- system.file("extdata", "06_mvp-impute", "mvp.imp.geno.desc", package = "rMVP")
geno &lt;- attach.big.matrix(genoPath)
pca &lt;- MVP.PCA(M=geno, cpu=1)

MVP.PCAplot(PCA=pca, Ncluster=3, class=NULL, 
    col=c("red", "green", "yellow"), file.output=FALSE, pch=19)


</code></pre>

<hr>
<h2 id='MVP.Report'>MVP.Report</h2><span id='topic+MVP.Report'></span>

<h3>Description</h3>

<p>MVP.Report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Report(
  MVP,
  col = c("#4197d8", "#f8c120", "#413496", "#495226", "#d60b6f", "#e66519", "#d581b7",
    "#83d3ad", "#7c162c", "#26755d"),
  bin.size = 1e+06,
  bin.range = NULL,
  pch = 19,
  band = 1,
  H = 1.5,
  ylim = NULL,
  cex.axis = 1,
  lwd.axis = 1.5,
  cex.lab = 1.5,
  plot.type = "b",
  multracks = FALSE,
  cex = c(0.5, 1, 1),
  r = 0.3,
  xlab = "Chromosome",
  ylab = expression(-log[10](italic(p))),
  xaxs = "i",
  yaxs = "r",
  outward = FALSE,
  threshold = NULL,
  threshold.col = "red",
  threshold.lwd = 1,
  threshold.lty = 2,
  amplify = FALSE,
  signal.cex = 1.5,
  signal.pch = 19,
  signal.col = "red",
  signal.line = 1,
  highlight = NULL,
  highlight.cex = 1.5,
  highlight.pch = 19,
  highlight.col = "green",
  chr.labels = NULL,
  chr.den.col = "black",
  cir.band = 1,
  cir.chr = TRUE,
  cir.chr.h = 1.5,
  cir.legend = TRUE,
  cir.legend.cex = 0.6,
  cir.legend.col = "black",
  LOG10 = TRUE,
  box = FALSE,
  conf.int = TRUE,
  file.output = TRUE,
  outpath = getwd(),
  file.type = "jpg",
  dpi = 300,
  height = NULL,
  width = NULL,
  memo = "",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Report_+3A_mvp">MVP</code></td>
<td>
<p>a dataframe or list, at least four columns. The first column is the name of SNP, the second column is the chromosome of SNP, the third column is the position of SNP, and the remaining columns are the P-value of each trait(Note:each trait a column).</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_col">col</code></td>
<td>
<p>a vector or a matrix, if &quot;col&quot; is a vector, each circle use the same colors, it means that the same chromosome is drewed in the same color, the colors are not fixed, one, two, three or more colors can be used, if the length of the &quot;col&quot; is shorter than the length the chromosome, then colors will be applied circularly. If &quot;col&quot; is a matrix, the row is the number of circles(traits), the columns are the colors that users want to use for different circles, each circle can be plotted in different number of colors, the missing value can be replaced by NA. For example: col=matrix(c(&quot;grey30&quot;,&quot;grey60&quot;,NA,&quot;red&quot;,&quot;blue&quot;,&quot;green&quot;,&quot;orange&quot;,NA,NA),3,3,byrow=T).</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_bin.size">bin.size</code></td>
<td>
<p>the size of bin for SNP_density plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_bin.range">bin.range</code></td>
<td>
<p>a vector, c(min, max). The min/max value of legend of SNP_density plot, the bin whose SNP number is smaller/bigger than 'bin.range' will be use the same color.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_pch">pch</code></td>
<td>
<p>a number, the type for the points or for traits of multi-traits Manhattan plot, is the same with &quot;pch&quot; in &lt;plot&gt;.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_band">band</code></td>
<td>
<p>a number, the space between chromosomes, the default is 1(if the band equals to 0, then there would be no space between chromosomes).</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_h">H</code></td>
<td>
<p>a number, the height for each circle, each circle represents a trait, the default is 1.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_ylim">ylim</code></td>
<td>
<p>a vector, the range of Y-axis when plotting the two type of Manhattan plots, is the same with &quot;ylim&quot; in &lt;plot&gt;.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cex.axis">cex.axis</code></td>
<td>
<p>a number, controls the size of ticks' numbers of X/Y-axis and the size of labels of circle plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_lwd.axis">lwd.axis</code></td>
<td>
<p>a number, controls the width of X/Y-axis lines.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cex.lab">cex.lab</code></td>
<td>
<p>a number, controls the size of labels of X/Y-axis.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_plot.type">plot.type</code></td>
<td>
<p>a character or vector, only &quot;d&quot;, &quot;c&quot;, &quot;m&quot;, &quot;q&quot; or &quot;b&quot; can be used. if plot.type=&quot;d&quot;, SNP density will be plotted; if plot.type=&quot;c&quot;, only circle-Manhattan plot will be plotted; if plot.type=&quot;m&quot;,only Manhattan plot will be plotted; if plot.type=&quot;q&quot;,only Q-Q plot will be plotted;if plot.type=&quot;b&quot;, both circle-Manhattan, Manhattan and Q-Q plots will be plotted; if plot.type=c(&quot;m&quot;,&quot;q&quot;), Both Manhattan and Q-Q plots will be plotted.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_multracks">multracks</code></td>
<td>
<p>a logical,if multracks=FALSE, all Manhattan plots will be drew in separated files, if it is TRUE, all Manhattan plots will be plotted in only one file.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cex">cex</code></td>
<td>
<p>a number or a vector, the size for the points, is the same with &quot;size&quot; in &lt;plot&gt;, and if it is a vector, the first number controls the size of points in circle plot(the default is 0.5), the second number controls the size of points in Manhattan plot(the default is 1), the third number controls the size of points in Q-Q plot(the default is 1)</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_r">r</code></td>
<td>
<p>a number, the radius for the circle(the inside radius), the default is 1.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_xlab">xlab</code></td>
<td>
<p>a character, the labels for x axis.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_ylab">ylab</code></td>
<td>
<p>a character, the labels for y axis.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_xaxs">xaxs</code></td>
<td>
<p>a character, The style of axis interval calculation to be used for the x-axis. Possible values are &quot;r&quot;, &quot;i&quot;, &quot;e&quot;, &quot;s&quot;, &quot;d&quot;. The styles are generally controlled by the range of data or xlim, if given.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_yaxs">yaxs</code></td>
<td>
<p>a character, The style of axis interval calculation to be used for the y-axis. See xaxs above..</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_outward">outward</code></td>
<td>
<p>logical, if outward=TRUE,then all points will be plotted from inside to outside for circular Manhattan plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_threshold">threshold</code></td>
<td>
<p>a number or vector, the significant threshold. For example, Bonfferoni adjustment method: threshold=0.01/nrow(Pmap). More than one significant line can be added on the plots, if threshold=0 or NULL, then the threshold line will not be added.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_threshold.col">threshold.col</code></td>
<td>
<p>a character or vector, the colour for the line of threshold levels.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_threshold.lwd">threshold.lwd</code></td>
<td>
<p>a number or vector, the width for the line of threshold levels.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_threshold.lty">threshold.lty</code></td>
<td>
<p>a number or vector, the type for the line of threshold levels.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_amplify">amplify</code></td>
<td>
<p>logical, CMplot can amplify the significant points, if amplify=T, then the points bigger than the minimal significant level will be amplified, the default: amplify=TRUE.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_signal.cex">signal.cex</code></td>
<td>
<p>a number, if amplify=TRUE, users can set the size of significant points.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_signal.pch">signal.pch</code></td>
<td>
<p>a number, if amplify=TRUE, users can set the shape of significant points.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_signal.col">signal.col</code></td>
<td>
<p>a character, if amplify=TRUE, users can set the colour of significant points, if signal.col=NULL, then the colors of significant points will not be changed.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_signal.line">signal.line</code></td>
<td>
<p>a number, the width of the lines of significant SNPs cross the circle.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_highlight">highlight</code></td>
<td>
<p>a vector, names of SNPs which need to be highlighted.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_highlight.cex">highlight.cex</code></td>
<td>
<p>a number or vector, the size of points for SNPs which need to be highlighted.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_highlight.pch">highlight.pch</code></td>
<td>
<p>a number or vector, the pch of points for SNPs which need to be highlighted.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_highlight.col">highlight.col</code></td>
<td>
<p>a number or vector, the col of points for SNPs which need to be highlighted.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_chr.labels">chr.labels</code></td>
<td>
<p>a vector, the labels for the chromosomes of density plot and circle-Manhattan plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_chr.den.col">chr.den.col</code></td>
<td>
<p>a character or vector or NULL, the colour for the SNP density. If the length of parameter 'chr.den.col' is bigger than 1, SNP density that counts the number of SNP within given size('bin.size') will be plotted around the circle. If chr.den.col=NULL, the density bar will not be attached on the bottom of manhattan plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cir.band">cir.band</code></td>
<td>
<p>a number, the space between circles, the default is 1.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cir.chr">cir.chr</code></td>
<td>
<p>logical, a boundary that represents chromosomes will be plotted on the periphery of a circle, the default is TRUE.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cir.chr.h">cir.chr.h</code></td>
<td>
<p>a number, the width for the boundary, if cir.chr=FALSE, then this parameter will be useless.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cir.legend">cir.legend</code></td>
<td>
<p>logical, whether to add the legend of each circle.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cir.legend.cex">cir.legend.cex</code></td>
<td>
<p>a number, the size of the number of legend.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_cir.legend.col">cir.legend.col</code></td>
<td>
<p>a character, the color of the axis of legend.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_log10">LOG10</code></td>
<td>
<p>logical, whether to change the p-value into log10(p-value).</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_box">box</code></td>
<td>
<p>logical, this function draws a box around the current Manhattan plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_conf.int">conf.int</code></td>
<td>
<p>logical, whether to plot confidence interval on QQ-plot.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_file.output">file.output</code></td>
<td>
<p>a logical, users can choose whether to output the plot results.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_outpath">outpath</code></td>
<td>
<p>Only when file.output = TRUE, determines the path of the output file</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_file.type">file.type</code></td>
<td>
<p>a character, users can choose the different output formats of plot, so for, &quot;jpg&quot;, &quot;pdf&quot;, &quot;tiff&quot; can be selected by users.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_dpi">dpi</code></td>
<td>
<p>a number, the picture resolution for .jpg and .tiff files. The default is 300.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_height">height</code></td>
<td>
<p>the height of output files.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_width">width</code></td>
<td>
<p>the width of output files.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_memo">memo</code></td>
<td>
<p>add a character to the output file name.</p>
</td></tr>
<tr><td><code id="MVP.Report_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output files
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pig60K, package = "rMVP")

MVP.Report(pig60K[,c(1:3, 5)], plot.type="m",
    threshold=0.05/nrow(pig60K), file.output=FALSE)


</code></pre>

<hr>
<h2 id='MVP.Report.Density'>SNP Density</h2><span id='topic+MVP.Report.Density'></span>

<h3>Description</h3>

<p>SNP Density
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Report.Density(
  Pmap,
  col = c("darkgreen", "yellow", "red"),
  dpi = 300,
  outpath = getwd(),
  memo = "MVP",
  bin.size = 1e+06,
  bin.max = NULL,
  file.type = "jpg",
  file.output = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Report.Density_+3A_pmap">Pmap</code></td>
<td>
<p>P value Map</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_col">col</code></td>
<td>
<p>The color vector</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_dpi">dpi</code></td>
<td>
<p>Number. Dots per inch for .jpg and .tiff files</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_outpath">outpath</code></td>
<td>
<p>Only when file.output = TRUE, determines the path of the output file</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_memo">memo</code></td>
<td>
<p>Character. A text marker on output files</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_bin.size">bin.size</code></td>
<td>
<p>the window size for counting SNP number</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_bin.max">bin.max</code></td>
<td>
<p>maximum SNP number, for winows, which has more SNPs than bin.max, will be painted in same color</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_file.type">file.type</code></td>
<td>
<p>format of output figure</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_file.output">file.output</code></td>
<td>
<p>Whether to output the file</p>
</td></tr>
<tr><td><code id="MVP.Report.Density_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
&lt;memo&gt;.SNP_Density.&lt;type&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pig60K, package = "rMVP")

MVP.Report.Density(pig60K, file.output=FALSE)

</code></pre>

<hr>
<h2 id='MVP.Report.QQplot'>QQ Plot</h2><span id='topic+MVP.Report.QQplot'></span>

<h3>Description</h3>

<p>QQ Plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Report.QQplot(
  P.values,
  taxa_name,
  col = c("blue"),
  cex = 0.5,
  threshold = NULL,
  amplify = TRUE,
  signal.col = "red",
  signal.pch = 19,
  signal.cex = 0.8,
  conf.int = TRUE,
  cex.axis = 1,
  conf.int.col = "grey",
  threshold.col = "red",
  outpath = getwd(),
  file.type = "jpg",
  memo = "MVP",
  box = TRUE,
  dpi = 300,
  file.output = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Report.QQplot_+3A_p.values">P.values</code></td>
<td>
<p>P values</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_taxa_name">taxa_name</code></td>
<td>
<p>The identifier of the phenotype will be used to generate a
portion of the image file name. If the title parameter is NULL, it will 
also be part of the title.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_col">col</code></td>
<td>
<p>default color is &quot;blue&quot;</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_cex">cex</code></td>
<td>
<p>A numerical value giving the amount by which plotting text and
symbols should be magnified relative to the default. This starts as 1 
when a device is opened, and is reset when the layout is changed, e.g. 
by setting mfrow. see <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_threshold">threshold</code></td>
<td>
<p>Number or Vector. The cutoff line on Manhattan plot, 
e.g. Bonfferoni correction. More than one significant line can be added
onto one figure. If threshold=0 or NULL, the threshold line will not be
added.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_amplify">amplify</code></td>
<td>
<p>Logical value. If TRUE, the points that passed the threshold line will be highlighted</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_signal.col">signal.col</code></td>
<td>
<p>Character. If &quot;amplify&quot; is TRUE, &quot;signal.col&quot; is used to set the color of significant points, if &quot;signal.col&quot; is NULL, the colors of significant points will not be changed</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_signal.pch">signal.pch</code></td>
<td>
<p>Number. If &quot;amplify&quot; is TRUE, users can set the type of significant points</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_signal.cex">signal.cex</code></td>
<td>
<p>Number. If &quot;amplify&quot; is TRUE, &quot;signal.cex&quot; is used to set the size of significant points</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_conf.int">conf.int</code></td>
<td>
<p>Whether to draw a confidence interval</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>a number, controls the size of numbers of X-axis and the size of labels of circle plot.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_conf.int.col">conf.int.col</code></td>
<td>
<p>a character, the color of the confidence interval on QQ-plot.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_threshold.col">threshold.col</code></td>
<td>
<p>Character or Vector. The colors of threshold lines</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_outpath">outpath</code></td>
<td>
<p>Only when file.output = TRUE, determines the path of the output file</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_file.type">file.type</code></td>
<td>
<p>A string or NULL is used to determine the type of output 
file. Can be &quot;jpg&quot;, &quot;pdf&quot;, &quot;tiff&quot;. If it is NULL, it will use 
<code><a href="grDevices.html#topic+dev">dev.new()</a></code> to create a new graphics device 
in the current environment, which may be RStudioGD or the default 
device of the system.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_memo">memo</code></td>
<td>
<p>the prefix of the output image file.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_box">box</code></td>
<td>
<p>A Boolean value that controls whether to draw a box around
QQplot.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_dpi">dpi</code></td>
<td>
<p>a number, the picture element for .jpg and .tiff files. The default is 300.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_file.output">file.output</code></td>
<td>
<p>Logical value. If TRUE, the figures will be generated.</p>
</td></tr>
<tr><td><code id="MVP.Report.QQplot_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output file:
&lt;memo&gt;.QQplot.&lt;taxa_name&gt;.&lt;type&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pig60K, package = "rMVP")

MVP.Report(pig60K[1:10000,], plot.type="q", file.output=FALSE)

</code></pre>

<hr>
<h2 id='MVP.Version'>Print MVP Banner</h2><span id='topic+MVP.Version'></span>

<h3>Description</h3>

<p>Build date: Aug 30, 2017
Last update: Dec 12, 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVP.Version(width = 60, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVP.Version_+3A_width">width</code></td>
<td>
<p>the width of the message</p>
</td></tr>
<tr><td><code id="MVP.Version_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>version number.
</p>


<h3>Author(s)</h3>

<p>Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MVP.Version()
</code></pre>

<hr>
<h2 id='pig60K'>Genotyped by pig 60k chip</h2><span id='topic+pig60K'></span>

<h3>Description</h3>

<p>This dataset gives the results of Genome-wide association study of 3 traits, individuals were genotyped by pig 60K chip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pig60K)</code></pre>


<h3>Format</h3>

<p>A dataframe containing 3 traits' Pvalue</p>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+is.big.matrix'></span><span id='topic+as.big.matrix'></span><span id='topic+attach.big.matrix'></span><span id='topic+deepcopy'></span><span id='topic+new'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>bigmemory</dt><dd><p><code><a href="bigmemory.html#topic+big.matrix">as.big.matrix</a></code>, <code><a href="bigmemory.html#topic+attach.big.matrix">attach.big.matrix</a></code>, <code><a href="bigmemory.html#topic+deepcopy">deepcopy</a></code>, <code><a href="bigmemory.html#topic+big.matrix">is.big.matrix</a></code></p>
</dd>
<dt>methods</dt><dd><p><code><a href="methods.html#topic+new">new</a></code></p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
