<!DOCTYPE html><html lang="en"><head><title>Help for package RNAseqQC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RNAseqQC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RNAseqQC-package'><p>RNAseqQC: Quality Control for RNA-Seq Data</p></a></li>
<li><a href='#all_numeric'><p>for a vector x,</p>
check if all non-NA elements of x
can be converted to numeric</a></li>
<li><a href='#filter_genes'><p>Filter genes with low counts</p></a></li>
<li><a href='#get_gene_id'><p>Get all gene IDs in a DESeqDataSet for a given gene name.</p></a></li>
<li><a href='#make_dds'><p>Make DESeqDataSet from counts matrix and metadata</p></a></li>
<li><a href='#mean_sd_plot'><p>Create a mean-sd plot</p>
Make a scatterplot that shows for each gene its standard deviation versus mean.</a></li>
<li><a href='#plot_biotypes'><p>Plot number of counts per sample and biotype</p></a></li>
<li><a href='#plot_chromosome'><p>Plot gene expression along a chromosome</p></a></li>
<li><a href='#plot_gene'><p>Plot a gene</p></a></li>
<li><a href='#plot_gene_detection'><p>Plot number of detected genes for each sample</p></a></li>
<li><a href='#plot_library_complexity'><p>Plot the library complexity</p></a></li>
<li><a href='#plot_loadings'><p>Plot loadings of a principal component</p></a></li>
<li><a href='#plot_ma'><p>MA-plot of a differential testing result</p></a></li>
<li><a href='#plot_pca'><p>Plot results of a principal component analysis</p></a></li>
<li><a href='#plot_pca_scatters'><p>Plot matrix of PCA scatter plots</p></a></li>
<li><a href='#plot_sample_clustering'><p>Plot clustering of samples in a distance heatmap</p></a></li>
<li><a href='#plot_sample_MAs'><p>MA plots of samples</p></a></li>
<li><a href='#plot_total_counts'><p>Plot total counts per sample</p></a></li>
<li><a href='#plot_within_level_sample_MAs'><p>Plot correlations of samples within a level of a group</p></a></li>
<li><a href='#save_plots_to_pdf'><p>Save list of plots to PDF</p></a></li>
<li><a href='#T47D'><p>The T47D cell line data of RNA-seq experiment GSE89888</p></a></li>
<li><a href='#T47D_diff_testing'><p>Differential expression results corresponding to the T47D data set.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Quality Control for RNA-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for semi-automated quality control of bulk RNA-seq data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>AnnotationHub, AnnotationFilter, BiocGenerics, ensembldb,
circlize, DESeq2, SummarizedExperiment, ComplexHeatmap,
matrixStats, grid, ggpointdensity, ggrepel, ggplot2, cowplot,
patchwork, purrr, dplyr, stringr, tidyr, tibble, tidyselect,
magrittr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, recount3, apeglm, ggrastr, gghighlight</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/frederikziebell/RNAseqQC">https://github.com/frederikziebell/RNAseqQC</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/frederikziebell/RNAseqQC/issues">https://github.com/frederikziebell/RNAseqQC/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-15 14:16:36 UTC; fxz40010</td>
</tr>
<tr>
<td>Author:</td>
<td>Frederik Ziebell [cre],
  Frederik Ziebell <a href="https://orcid.org/0000-0003-3673-1721"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  GlaxoSmithKline Research &amp; Development Limited [cph] (GlaxoSmithKline
    Research &amp; Development Limited; registered address: 980 Great West
    Road, Brentford, Middlesex TW8 9GS, United Kingdom)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Frederik Ziebell &lt;f_ziebell@web.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-15 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='RNAseqQC-package'>RNAseqQC: Quality Control for RNA-Seq Data</h2><span id='topic+RNAseqQC'></span><span id='topic+RNAseqQC-package'></span>

<h3>Description</h3>

<p>Functions for semi-automated quality control of bulk RNA-seq data.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Frederik Ziebell <a href="mailto:f_ziebell@web.de">f_ziebell@web.de</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Frederik Ziebell <a href="mailto:frederik.x.ziebell@gsk.com">frederik.x.ziebell@gsk.com</a> (<a href="https://orcid.org/0000-0003-3673-1721">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> GlaxoSmithKline Research &amp; Development Limited (GlaxoSmithKline Research &amp; Development Limited; registered address: 980 Great West Road, Brentford, Middlesex TW8 9GS, United Kingdom) [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/frederikziebell/RNAseqQC">https://github.com/frederikziebell/RNAseqQC</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/frederikziebell/RNAseqQC/issues">https://github.com/frederikziebell/RNAseqQC/issues</a>
</p>
</li></ul>


<hr>
<h2 id='all_numeric'>for a vector x,
check if all non-NA elements of x
can be converted to numeric</h2><span id='topic+all_numeric'></span>

<h3>Description</h3>

<p>for a vector x,
check if all non-NA elements of x
can be converted to numeric
</p>


<h3>Usage</h3>

<pre><code class='language-R'>all_numeric(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="all_numeric_+3A_x">x</code></td>
<td>
<p>A non-numeric vector</p>
</td></tr>
</table>

<hr>
<h2 id='filter_genes'>Filter genes with low counts</h2><span id='topic+filter_genes'></span>

<h3>Description</h3>

<p>Filter genes with low counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_genes(dds, min_count = 5, min_rep = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_genes_+3A_dds">dds</code></td>
<td>
<p>A DESeqDataSet</p>
</td></tr>
<tr><td><code id="filter_genes_+3A_min_count">min_count</code>, <code id="filter_genes_+3A_min_rep">min_rep</code></td>
<td>
<p>keep genes with at least min_count counts in at least min_rep replicates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>DESeq2::DESeqDataSet</code> object with only those genes that meet the filter criteria.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
dds &lt;- makeExampleDESeqDataSet()
filter_genes(dds)

</code></pre>

<hr>
<h2 id='get_gene_id'>Get all gene IDs in a DESeqDataSet for a given gene name.</h2><span id='topic+get_gene_id'></span>

<h3>Description</h3>

<p>Get all gene IDs in a DESeqDataSet for a given gene name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_id(gene_name, dds)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gene_id_+3A_gene_name">gene_name</code></td>
<td>
<p>A gene name</p>
</td></tr>
<tr><td><code id="get_gene_id_+3A_dds">dds</code></td>
<td>
<p>A DESeqDataSet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_gene_id("HBA1", T47D)
</code></pre>

<hr>
<h2 id='make_dds'>Make DESeqDataSet from counts matrix and metadata</h2><span id='topic+make_dds'></span>

<h3>Description</h3>

<p>Make DESeqDataSet from counts matrix and metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_dds(counts, metadata, ah_record, design = ~1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_dds_+3A_counts">counts</code></td>
<td>
<p>The genes x samples counts matrix with row names. At least one row name must be an ENSEMBL gene ID,
since gene annotation is done via the ENSEMBL database.</p>
</td></tr>
<tr><td><code id="make_dds_+3A_metadata">metadata</code></td>
<td>
<p>data.frame of sample information. Order of rows corresponds to the order of columns in the counts matrix.</p>
</td></tr>
<tr><td><code id="make_dds_+3A_ah_record">ah_record</code></td>
<td>
<p>ID of AnnotationHub record used to retrieve an <code>EnsDb</code> object.</p>
</td></tr>
<tr><td><code id="make_dds_+3A_design">design</code></td>
<td>
<p>The design formula specified in <code>DESeqDataSet()</code>
To view all valid record IDs, run
</p>
<div class="sourceCode"><pre>library(AnnotationHub)
mcols(AnnotationHub()) %&gt;%
 as_tibble(rownames="ah_record") %&gt;%
 filter(rdataclass=="EnsDb")
</pre></div></td></tr>
</table>


<h3>Value</h3>

<p>A <code>DESeq2::DESeqDataSet</code> object containing the counts matrix and metadata.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("DESeq2")
count_mat &lt;- counts(T47D)
meta &lt;- data.frame(colData(T47D))
dds &lt;- make_dds(counts = count_mat, metadata = meta, ah_record = "AH89426")


</code></pre>

<hr>
<h2 id='mean_sd_plot'>Create a mean-sd plot
Make a scatterplot that shows for each gene its standard deviation versus mean.</h2><span id='topic+mean_sd_plot'></span>

<h3>Description</h3>

<p>Create a mean-sd plot
Make a scatterplot that shows for each gene its standard deviation versus mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean_sd_plot(vsd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mean_sd_plot_+3A_vsd">vsd</code></td>
<td>
<p>A DESeqTransform object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the ggplot2 package that contains the mean-sd plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("DESeq2")
dds &lt;- makeExampleDESeqDataSet(interceptMean=10, n=5000)
vsd &lt;- vst(dds)
mean_sd_plot(vsd)

</code></pre>

<hr>
<h2 id='plot_biotypes'>Plot number of counts per sample and biotype</h2><span id='topic+plot_biotypes'></span>

<h3>Description</h3>

<p>Plot the total number of counts for each sample and the major classes of
ENSEMBL gene biotypes (protein coding, lncRNA, etc.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_biotypes(dds)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_biotypes_+3A_dds">dds</code></td>
<td>
<p>A DESeqDataSet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the ggplot2 package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_biotypes(T47D)

</code></pre>

<hr>
<h2 id='plot_chromosome'>Plot gene expression along a chromosome</h2><span id='topic+plot_chromosome'></span>

<h3>Description</h3>

<p>Plot gene expression along a chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_chromosome(vsd, chr, scale = FALSE, trunc_val = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_chromosome_+3A_vsd">vsd</code></td>
<td>
<p>An object generated by <code>DESeq2::vst()</code></p>
</td></tr>
<tr><td><code id="plot_chromosome_+3A_chr">chr</code></td>
<td>
<p>A string denoting a chromosome as annotated by ENSEMBL, e.g.
'1', '2', 'X', 'Y', 'MT'</p>
</td></tr>
<tr><td><code id="plot_chromosome_+3A_scale">scale</code></td>
<td>
<p>Whether to scale the columns of the heatmap</p>
</td></tr>
<tr><td><code id="plot_chromosome_+3A_trunc_val">trunc_val</code></td>
<td>
<p>Truncate the expression matrix to this value prior to plotting. This is useful
if some very high expression values dominate the heatmap. By default, the heatmap is truncated
to expression values at most 3 standard deviations from the mean.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Heatmap-class object of the <code>ComplexHeatmap</code> package that contains the heatmap of expression values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
chr1 &lt;- T47D[which(mcols(T47D)$chromosome=="1"),]
vsd &lt;- vst(chr1)
plot_chromosome(vsd, chr="1")

</code></pre>

<hr>
<h2 id='plot_gene'>Plot a gene</h2><span id='topic+plot_gene'></span>

<h3>Description</h3>

<p>Plot a gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene(
  gene,
  dds,
  x_var = NULL,
  color_by = NULL,
  point_alpha = 0.7,
  point_rel_size = 2,
  show_plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_gene_+3A_gene">gene</code></td>
<td>
<p>A gene ID or gene name, i.e. an element of <code>rownames(dds)</code> or of <code>rowData(dds)$gene_name</code></p>
</td></tr>
<tr><td><code id="plot_gene_+3A_dds">dds</code></td>
<td>
<p>a DESeqDataSet</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_x_var">x_var</code></td>
<td>
<p>Variable to plot on the x-axis. If NULL, then each sample is plotted separately.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_color_by">color_by</code></td>
<td>
<p>Variable (column in <code>colData(dds)</code>) to color points by.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_point_alpha">point_alpha</code></td>
<td>
<p>alpha value of <code>geom_point()</code></p>
</td></tr>
<tr><td><code id="plot_gene_+3A_point_rel_size">point_rel_size</code></td>
<td>
<p>relative size of <code>geom_point()</code></p>
</td></tr>
<tr><td><code id="plot_gene_+3A_show_plot">show_plot</code></td>
<td>
<p>Whether to show the plot or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function displays the plot and returns invisible the data frame of
expression values and colData annotation for the gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet()
colData(dds)$type &lt;- c("A","A","A","B","B","B")
colData(dds)$patient &lt;- c("1","1","2","2","3","3")
dds &lt;- estimateSizeFactors(dds)
plot_gene("gene1", dds)
plot_gene("gene1", dds, x_var="patient", color_by="type")

</code></pre>

<hr>
<h2 id='plot_gene_detection'>Plot number of detected genes for each sample</h2><span id='topic+plot_gene_detection'></span>

<h3>Description</h3>

<p>For specified thresholds, the number of detected genes is shown for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene_detection(dds, thresholds = c(3, 10, 20, 50))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_gene_detection_+3A_dds">dds</code></td>
<td>
<p>A DESeqDataSet</p>
</td></tr>
<tr><td><code id="plot_gene_detection_+3A_thresholds">thresholds</code></td>
<td>
<p>Vector of thresholds for which the number of genes with
counts greater or equal than the thresholds is plotted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the ggplot2 package that contains the gene detection plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet()
plot_gene_detection(dds)

</code></pre>

<hr>
<h2 id='plot_library_complexity'>Plot the library complexity</h2><span id='topic+plot_library_complexity'></span>

<h3>Description</h3>

<p>Plot per sample the fraction of genes, versus the fraction of total counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_library_complexity(dds, show_progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_library_complexity_+3A_dds">dds</code></td>
<td>
<p>A DESeqDataSet</p>
</td></tr>
<tr><td><code id="plot_library_complexity_+3A_show_progress">show_progress</code></td>
<td>
<p>Whether to show a progress bar of the computation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the ggplot2 package that contains the library complexity plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet()
plot_library_complexity(dds)

</code></pre>

<hr>
<h2 id='plot_loadings'>Plot loadings of a principal component</h2><span id='topic+plot_loadings'></span>

<h3>Description</h3>

<p>Plot loadings of a principal component
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_loadings(
  pca_res,
  PC = 1,
  square = FALSE,
  color_by = NULL,
  annotate_top_n = 0,
  highlight_genes = NULL,
  show_plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_loadings_+3A_pca_res">pca_res</code></td>
<td>
<p>A result returned from <code>plot_pca()</code></p>
</td></tr>
<tr><td><code id="plot_loadings_+3A_pc">PC</code></td>
<td>
<p>Number of the principal component to plot</p>
</td></tr>
<tr><td><code id="plot_loadings_+3A_square">square</code></td>
<td>
<p>Whether to plot squared loadings. The squared loading is equal to the fraction of variance explained by the respective feature in the given principal component.</p>
</td></tr>
<tr><td><code id="plot_loadings_+3A_color_by">color_by</code></td>
<td>
<p>Variable (column in <code>pca_res$loadings</code>) to color points by. Can also be 'pc_sign' to color by the sign of the loading (useful in combination with the square = TRUE parameter).</p>
</td></tr>
<tr><td><code id="plot_loadings_+3A_annotate_top_n">annotate_top_n</code></td>
<td>
<p>Annotate the top n features with positive or negative loading</p>
</td></tr>
<tr><td><code id="plot_loadings_+3A_highlight_genes">highlight_genes</code></td>
<td>
<p>Vector of gene names or gene IDs to highlight on the plot (overwrites top_n annotation)</p>
</td></tr>
<tr><td><code id="plot_loadings_+3A_show_plot">show_plot</code></td>
<td>
<p>Whether to show the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function displays the loadings plot and returns invisible a list of the plot,
the data.frame of the PCA loadings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
data &lt;- matrix(rnorm(100*6), ncol=6)
data &lt;- t(t(data)+c(-1, -1.1, -1.2, 1, 1.1, 1.2))
pca_res &lt;- plot_pca(data)
plot_loadings(pca_res)
</code></pre>

<hr>
<h2 id='plot_ma'>MA-plot of a differential testing result</h2><span id='topic+plot_ma'></span>

<h3>Description</h3>

<p>MA-plot of a differential testing result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ma(de_res, dds, annotate_top_n = 5, highlight_genes = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ma_+3A_de_res">de_res</code></td>
<td>
<p>An object returned by <code>DESeq2::results()</code> or <code>DESeq2::lfcShrink()</code></p>
</td></tr>
<tr><td><code id="plot_ma_+3A_dds">dds</code></td>
<td>
<p>The DESeqDataSet that was used to build the ´de_res´ object. This is needed for gene name annotation.</p>
</td></tr>
<tr><td><code id="plot_ma_+3A_annotate_top_n">annotate_top_n</code></td>
<td>
<p>Annotate the top n significant genes by fold change (up- and down-regulated)</p>
</td></tr>
<tr><td><code id="plot_ma_+3A_highlight_genes">highlight_genes</code></td>
<td>
<p>Vector of gene names or gene IDs to highlight on the plot (overwrites top_n annotation)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the ggplot2 package that contains the MA-plot. The plot shows three classes of points:
Light gray points are genes with low counts that are removed from the analysis by independent filtering. Darker gray points
are not significant genes that show a density map to visualize where the majority of non-significant points are located. Finally,
red point show significant genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet(n=1500, m=6, betaSD=.3, interceptMean=6)
rowData(dds)$gene_name &lt;- rownames(dds)
dds &lt;- DESeq(dds)
de_res &lt;- results(dds)
plot_ma(de_res, dds)

</code></pre>

<hr>
<h2 id='plot_pca'>Plot results of a principal component analysis</h2><span id='topic+plot_pca'></span>

<h3>Description</h3>

<p>Plot results of a principal component analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_pca(
  obj,
  PC_x = 1,
  PC_y = 2,
  n_feats = 500,
  scale_feats = FALSE,
  na_frac = 0.3,
  metadata = NULL,
  color_by = NULL,
  shape_by = NULL,
  point_alpha = 0.7,
  point_rel_size = 2,
  show_plot = TRUE,
  rasterise = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_pca_+3A_obj">obj</code></td>
<td>
<p>A (features x samples) matrix or SummarizedExperiment object</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_pc_x">PC_x</code></td>
<td>
<p>The PC to show on the x-axis.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_pc_y">PC_y</code></td>
<td>
<p>The PC to show on the y-axis.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_n_feats">n_feats</code></td>
<td>
<p>Number of top-variable features to include.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_scale_feats">scale_feats</code></td>
<td>
<p>Whether to scale the features.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_na_frac">na_frac</code></td>
<td>
<p>Only consider features with the stated maximum fraction of NAs or NaNs. NA/NaNs will be mean-imputed for PCA.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_metadata">metadata</code></td>
<td>
<p>A data.frame used for annotating samples. <code>rownames(metadata)</code> must match <code>colnames(obj)</code>.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_color_by">color_by</code></td>
<td>
<p>Variable by which to color points. Must be a column in metadata or in <code>colData(obj)</code>.
Alternatively, it can be the name of a feature (a rowname of obj) or a gene name (an element of rowData(obj)$gene_name).</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_shape_by">shape_by</code></td>
<td>
<p>Variable by which to color points. Must be a column in metadata or in <code>colData(obj)</code>.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_point_alpha">point_alpha</code></td>
<td>
<p>alpha value of <code>geom_point()</code></p>
</td></tr>
<tr><td><code id="plot_pca_+3A_point_rel_size">point_rel_size</code></td>
<td>
<p>relative size of <code>geom_point()</code></p>
</td></tr>
<tr><td><code id="plot_pca_+3A_show_plot">show_plot</code></td>
<td>
<p>Whether to show the plot or not</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_rasterise">rasterise</code></td>
<td>
<p>Whether to rasterise the point, using ggrastr.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_...">...</code></td>
<td>
<p>Other parameters passed on to ggrastr::rasterise</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the <code>metadata</code> or <code>colData</code> of <code>obj</code> contain a column <code>colname</code>, this colum will be removed in the <code style="white-space: pre;">&#8288;$pca_data&#8288;</code> slot,
because this column contains the colnames of the data matrix. Similarly, for the <code style="white-space: pre;">&#8288;$loadings&#8288;</code> slot, the column <code>rowname</code> is
reserved for the rownames of the data matrix.
</p>


<h3>Value</h3>

<p>The function displays the plot and returns invisible a list of the plot,
the data.frame to make the plot, the vector of percentages of variance explained and the loadings matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
data &lt;- matrix(rnorm(100*6), ncol=6)
data &lt;- t(t(data)+c(-1, -1.1, -1.2, 1, 1.1, 1.2))
plot_pca(data)

</code></pre>

<hr>
<h2 id='plot_pca_scatters'>Plot matrix of PCA scatter plots</h2><span id='topic+plot_pca_scatters'></span>

<h3>Description</h3>

<p>Plot matrix of PCA scatter plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_pca_scatters(
  obj,
  n_PCs = min(10, nrow(obj), ncol(obj)),
  show_var_exp = T,
  n_feats = 500,
  scale_feats = FALSE,
  na_frac = 0.3,
  metadata = NULL,
  color_by = NULL,
  shape_by = NULL,
  point_alpha = 0.7,
  point_rel_size = 2,
  transpose = FALSE,
  rasterise = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_pca_scatters_+3A_obj">obj</code></td>
<td>
<p>A (features x samples) matrix or SummarizedExperiment object</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_n_pcs">n_PCs</code></td>
<td>
<p>Number of principal components to plot</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_show_var_exp">show_var_exp</code></td>
<td>
<p>Whether to show a plot of the percentage of variance explained by each PC in the bottom left corner.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_n_feats">n_feats</code></td>
<td>
<p>Number of top-variable features to include.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_scale_feats">scale_feats</code></td>
<td>
<p>Whether to scale the features.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_na_frac">na_frac</code></td>
<td>
<p>Only consider features with the stated maximum fraction of NAs or NaNs. NA/NaNs will be mean-imputed for PCA.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_metadata">metadata</code></td>
<td>
<p>A data.frame used for annotating samples. <code>rownames(metadata)</code> must match <code>colnames(obj)</code>.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_color_by">color_by</code></td>
<td>
<p>Variable by which to color points. Must be a column in metadata or in <code>colData(obj)</code>.
Alternatively, it can be the name of a feature (a rowname of obj) or a gene name (an element of rowData(obj)$gene_name).</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_shape_by">shape_by</code></td>
<td>
<p>Variable by which to color points. Must be a column in metadata or in <code>colData(obj)</code>.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_point_alpha">point_alpha</code></td>
<td>
<p>alpha value of <code>geom_point()</code></p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_point_rel_size">point_rel_size</code></td>
<td>
<p>relative size of <code>geom_point()</code></p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_transpose">transpose</code></td>
<td>
<p>Wheter to transpose the whole matrix of scatter plots</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_rasterise">rasterise</code></td>
<td>
<p>Whether to rasterise the points using ggrastr.</p>
</td></tr>
<tr><td><code id="plot_pca_scatters_+3A_...">...</code></td>
<td>
<p>Other parameters passed on to ggrastr::rasterise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function displays the scatter plots of the PCs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)
data &lt;- matrix(rnorm(100*6), ncol=6)
data &lt;- t(t(data)+c(-1, -1.1, -1.2, 1, 1.1, 1.2))
plot_pca_scatters(data)

</code></pre>

<hr>
<h2 id='plot_sample_clustering'>Plot clustering of samples in a distance heatmap</h2><span id='topic+plot_sample_clustering'></span>

<h3>Description</h3>

<p>Plot clustering of samples in a distance heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_sample_clustering(
  se,
  n_feats = 500,
  anno_vars = NULL,
  anno_title = "group",
  distance = "euclidean",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_sample_clustering_+3A_se">se</code></td>
<td>
<p>A SummarizedExperiment object.</p>
</td></tr>
<tr><td><code id="plot_sample_clustering_+3A_n_feats">n_feats</code></td>
<td>
<p>Number of top-variable features (genes) to consider</p>
</td></tr>
<tr><td><code id="plot_sample_clustering_+3A_anno_vars">anno_vars</code></td>
<td>
<p>Character vector of columns in <code>colData(se)</code> to annotate samples</p>
</td></tr>
<tr><td><code id="plot_sample_clustering_+3A_anno_title">anno_title</code></td>
<td>
<p>The title of the color legend for <code>anno_vars</code></p>
</td></tr>
<tr><td><code id="plot_sample_clustering_+3A_distance">distance</code></td>
<td>
<p>The type of distance metric to consider. Either 'euclidean', 'pearson' or 'spearman'</p>
</td></tr>
<tr><td><code id="plot_sample_clustering_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to ComplexHeatmap::Heatmap()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Heatmap-class object of the <code>ComplexHeatmap</code> package that contains the heatmap of pairwise sample distances.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("DESeq2")
dds &lt;- makeExampleDESeqDataSet(m=8, interceptMean=10)
vsd &lt;- vst(dds)
plot_sample_clustering(vsd)

</code></pre>

<hr>
<h2 id='plot_sample_MAs'>MA plots of samples</h2><span id='topic+plot_sample_MAs'></span>

<h3>Description</h3>

<p>For each level of the grouping variable, the gene-wise median over all samples is computed
to obtain a reference sample. Then, each sample is plotted against the reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_sample_MAs(vsd, group, y_lim = 3, rasterise = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_sample_MAs_+3A_vsd">vsd</code></td>
<td>
<p>An object generated by <code>DESeq2::vst()</code></p>
</td></tr>
<tr><td><code id="plot_sample_MAs_+3A_group">group</code></td>
<td>
<p>A grouping variable, must be a column of <code>colData(vsd)</code></p>
</td></tr>
<tr><td><code id="plot_sample_MAs_+3A_y_lim">y_lim</code></td>
<td>
<p>Y-axis limits, the axis will run from <code>-y_lim</code> to <code>y_lim</code></p>
</td></tr>
<tr><td><code id="plot_sample_MAs_+3A_rasterise">rasterise</code></td>
<td>
<p>Whether to rasterise the points using ggrastr.</p>
</td></tr>
<tr><td><code id="plot_sample_MAs_+3A_...">...</code></td>
<td>
<p>Other parameters passed on to ggrastr::rasterise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot objects of the ggplot2 package, with each element corresponding to one MA-plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet(n=1000, m=4, interceptMean=10)
colData(dds)$type &lt;- c("A","A","B","B")
vsd &lt;- vst(dds)
plot_sample_MAs(vsd, group="type")

</code></pre>

<hr>
<h2 id='plot_total_counts'>Plot total counts per sample</h2><span id='topic+plot_total_counts'></span>

<h3>Description</h3>

<p>Plot the distribution of the total number of counts per sample as histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_total_counts(dds, n_bins = 50)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_total_counts_+3A_dds">dds</code></td>
<td>
<p>A DESeqDataSet</p>
</td></tr>
<tr><td><code id="plot_total_counts_+3A_n_bins">n_bins</code></td>
<td>
<p>Number of histogram bins</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the ggplot2 package that contains the histogram of total counts per sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet(m=30)
plot_total_counts(dds)

</code></pre>

<hr>
<h2 id='plot_within_level_sample_MAs'>Plot correlations of samples within a level of a group</h2><span id='topic+plot_within_level_sample_MAs'></span>

<h3>Description</h3>

<p>For the given level, the gene-wise median over all samples is computed
to obtain a reference sample. Then, each sample is plotted against the
reference as MA-plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_within_level_sample_MAs(
  vsd,
  group,
  level,
  y_lim = 4,
  rasterise = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_within_level_sample_MAs_+3A_vsd">vsd</code></td>
<td>
<p>An object generated by <code>DESeq2::vst()</code></p>
</td></tr>
<tr><td><code id="plot_within_level_sample_MAs_+3A_group">group</code></td>
<td>
<p>A grouping variable, must be a column of <code>colData(vsd)</code></p>
</td></tr>
<tr><td><code id="plot_within_level_sample_MAs_+3A_level">level</code></td>
<td>
<p>A level of the grouping variable</p>
</td></tr>
<tr><td><code id="plot_within_level_sample_MAs_+3A_y_lim">y_lim</code></td>
<td>
<p>Y-axis limits, the axis will run from <code>-y_lim</code> to <code>y_lim</code></p>
</td></tr>
<tr><td><code id="plot_within_level_sample_MAs_+3A_rasterise">rasterise</code></td>
<td>
<p>Whether to rasterise the points using ggrastr.</p>
</td></tr>
<tr><td><code id="plot_within_level_sample_MAs_+3A_...">...</code></td>
<td>
<p>Other parameters passed on to ggrastr::rasterise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot objects of the ggplot2 package that contains for
each sample of the specified level the the sample vs reference MA-plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("DESeq2")
set.seed(1)
dds &lt;- makeExampleDESeqDataSet(n=1000, m=4, interceptMean=10)
colData(dds)$type &lt;- c("A","A","B","B")
vsd &lt;- vst(dds)
plot_within_level_sample_MAs(vsd, group="type", level="A")

</code></pre>

<hr>
<h2 id='save_plots_to_pdf'>Save list of plots to PDF</h2><span id='topic+save_plots_to_pdf'></span>

<h3>Description</h3>

<p>This function takes a list of plots as input and makes a pdf with <code>ncol</code> x <code>nrow</code> plots per page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_plots_to_pdf(
  plots,
  file = "plots.pdf",
  ncol,
  nrow,
  subfig_width = subfig_height * 16/9,
  subfig_height = 2.5,
  legend_position = "original"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="save_plots_to_pdf_+3A_plots">plots</code></td>
<td>
<p>List of plots that is passed to the <code>plotlist</code> argument of <code>cowplot::plot_grid</code></p>
</td></tr>
<tr><td><code id="save_plots_to_pdf_+3A_file">file</code></td>
<td>
<p>file where the plots are saved</p>
</td></tr>
<tr><td><code id="save_plots_to_pdf_+3A_ncol">ncol</code></td>
<td>
<p>number of columns per page for the grid of plots</p>
</td></tr>
<tr><td><code id="save_plots_to_pdf_+3A_nrow">nrow</code></td>
<td>
<p>number of rows per page for the grid of plots</p>
</td></tr>
<tr><td><code id="save_plots_to_pdf_+3A_subfig_width">subfig_width</code></td>
<td>
<p>width of a plot of the grid in inches</p>
</td></tr>
<tr><td><code id="save_plots_to_pdf_+3A_subfig_height">subfig_height</code></td>
<td>
<p>height of a plot of the grid in inches</p>
</td></tr>
<tr><td><code id="save_plots_to_pdf_+3A_legend_position">legend_position</code></td>
<td>
<p>either 'original' if the original legend of each sub-plot is shown, 'none', if no legend should be shown in any of the sub-plots, 'bottom',
if no legend should be shown in the sub plots and one shared legend at the bottom or 'right', which is same as 'bottom', but shown on the right</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns nothing but is called for it's side effect, which is to save a pdf of plots to the filesystem.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("ggplot2")
manuf &lt;- unique(mpg$manufacturer)
plots &lt;- lapply(manuf, function(x){
  df &lt;- mpg[mpg$manufacturer==x,]
  ggplot(df, aes(cty, hwy)) +
    geom_point() +
    labs(title=x)
})
save_plots_to_pdf(plots, ncol=3, nrow=2)


</code></pre>

<hr>
<h2 id='T47D'>The T47D cell line data of RNA-seq experiment GSE89888</h2><span id='topic+T47D'></span>

<h3>Description</h3>

<p>The dataset contains the read counts of experiment GSE89888 in which T47D
cells with different mutation statuses were treated with E2 (estradiol) or
vehicle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>T47D
</code></pre>


<h3>Format</h3>

<p>A DESeqDataSet with 43576 rows (of genes) and 24 columns (of samples).
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1101/2021.05.21.445138">doi:10.1101/2021.05.21.445138</a>
</p>

<hr>
<h2 id='T47D_diff_testing'>Differential expression results corresponding to the T47D data set.</h2><span id='topic+T47D_diff_testing'></span>

<h3>Description</h3>

<p>Differential expression results corresponding to the T47D data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>T47D_diff_testing
</code></pre>


<h3>Format</h3>

<p>A DESeqResults object with 36562 rows and 3 columns.
</p>


<h3>Source</h3>

<p>See the 'data' vignette on how to reproduce this object.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
