<!DOCTYPE html><html><head><title>Help for package RNAseqNet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RNAseqNet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chooseSigma'><p>Select the threshold sigma for hd-MI.</p></a></li>
<li><a href='#GLMnetToGraph'><p>Convert the result of imputedGLMnetwork or a matrix into a network.</p></a></li>
<li><a href='#GLMnetwork'><p>Infer a network from RNA-seq expression.</p></a></li>
<li><a href='#GLMpath'><p>Methods for 'GLMpath' objects.</p></a></li>
<li><a href='#HDImputed'><p>Methods for 'HDImputed' objects.</p></a></li>
<li><a href='#HDpath'><p>Methods for 'HDpath' objects.</p></a></li>
<li><a href='#imputedGLMnetwork'><p>Multiple hot-deck imputation and network inference from RNA-seq data.</p></a></li>
<li><a href='#imputeHD'><p>Impute missing row datasets with multiple hot deck.</p></a></li>
<li><a href='#lung'><p>RNA-seq expression from lung tissue (GTEx).</p></a></li>
<li><a href='#RNAseqNetUsersGuide'><p>View RNAseqNet User's Guide</p></a></li>
<li><a href='#stabilitySelection'><p>Selection of the regularization parameter by StARS (Liu et al., 2010).</p></a></li>
<li><a href='#stars'><p>Methods for 'stars' objects.</p></a></li>
<li><a href='#thyroid'><p>RNA-seq expression from thyroid tissue (GTEx).</p></a></li>
<li><a href='#varIntra'><p>Average intra-donor pool variance.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Log-Linear Poisson Graphical Model with Hot-Deck Multiple
Imputation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-20</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nathalie Vialaneix &lt;nathalie.vialaneix@inrae.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Infer log-linear Poisson Graphical Model with an auxiliary data
    set. Hot-deck multiple imputation method is used to improve the reliability
    of the inference with an auxiliary dataset. Standard log-linear Poisson 
    graphical model can also be used for the inference and the Stability 
    Approach for Regularization Selection (StARS) is implemented to drive the 
    selection of the regularization parameter. The method is fully described in
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx819">doi:10.1093/bioinformatics/btx819</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), ggplot2</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph (&ge; 1.0), hot.deck, PoiClaClu, glmnet, methods, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-20 16:31:31 UTC; nathalie</td>
</tr>
<tr>
<td>Author:</td>
<td>Alyssa Imbert [aut],
  Nathalie Vialaneix [aut, cre]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-20 16:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='chooseSigma'>Select the threshold sigma for hd-MI.</h2><span id='topic+chooseSigma'></span>

<h3>Description</h3>

<p><code>chooseSigma</code> computes the average intra-donor pool variance for
different values of sigma. It helps choosing a sigma that makes a good
trade-off between homogeneity within the pool of donors and variety (large
enough number of donors in every pool).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chooseSigma(X, Y, sigma_list, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chooseSigma_+3A_x">X</code></td>
<td>
<p>n x p numeric matrix containing RNA-seq expression with missing rows
(numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="chooseSigma_+3A_y">Y</code></td>
<td>
<p>auxiliary dataset (n' x q numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="chooseSigma_+3A_sigma_list">sigma_list</code></td>
<td>
<p>a sequence of increasing positive values for sigma (numeric
vector)</p>
</td></tr>
<tr><td><code id="chooseSigma_+3A_seed">seed</code></td>
<td>
<p>single value, interpreted as an in integer, used to initialize
the random number generation state</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The average intra-donor pool variance is described in (Imbert
<em>et al.</em>, 2018).
</p>


<h3>Value</h3>

<p>a data frame with the values of sigma and the corresponding
intra-donor pool variances
</p>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Imbert, A., Valsesia, A., Le Gall, C., Armenise, C., Lefebvre,
G. Gourraud, P.A., Viguerie, N. and Villa-Vialaneix, N. (2018) Multiple
hot-deck imputation for network inference from RNA sequencing data.
<em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btx819">doi:10.1093/bioinformatics/btx819</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+varIntra">varIntra</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
data(thyroid)
nobs &lt;- nrow(lung)
miss_ind &lt;- sample(1:nobs, round(0.2 * nobs), replace = FALSE)
lung[miss_ind, ] &lt;- NA
lung &lt;- na.omit(lung)
sigma_stats &lt;- chooseSigma(lung, thyroid, 1:5)
## Not run: plot(sigma_stats, type = "b")

</code></pre>

<hr>
<h2 id='GLMnetToGraph'>Convert the result of imputedGLMnetwork or a matrix into a network.</h2><span id='topic+GLMnetToGraph'></span>

<h3>Description</h3>

<p><code>GLMnetToGraph</code> combines the m inferred networks, obtained from m
imputed datasets, into a single stable network or convert a matrix of
coefficients of a GLM model into a network (non zero coefficients are
converted to edges)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLMnetToGraph(object, threshold = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GLMnetToGraph_+3A_object">object</code></td>
<td>
<p>an object of class <code>HDpath</code> as obtained from the function
<code><a href="#topic+imputedGLMnetwork">imputedGLMnetwork</a></code> or a squared matrix with zero and non zero
values</p>
</td></tr>
<tr><td><code id="GLMnetToGraph_+3A_threshold">threshold</code></td>
<td>
<p>the percentage of times, among the m imputed networks, that
an edge has to be predicted to be in the final network. Used only for objects
of class <code>HDpath</code>. Default to 0.9</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an 'igraph' object. See <code><a href="igraph.html#topic+igraph">igraph</a></code>
</p>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Imbert, A., Valsesia, A., Le Gall, C., Armenise, C., Lefebvre,
G. Gourraud, P.A., Viguerie, N. and Villa-Vialaneix, N. (2018) Multiple
hot-deck imputation for network inference from RNA sequencing data.
<em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btx819">doi:10.1093/bioinformatics/btx819</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+imputedGLMnetwork">imputedGLMnetwork</a></code>, <code><a href="igraph.html#topic+igraph">igraph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
data(thyroid)
nobs &lt;- nrow(lung)
miss_ind &lt;- sample(1:nobs, round(0.2 * nobs), replace = FALSE)
lung[miss_ind, ] &lt;- NA
lung &lt;- na.omit(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 10))
## Not run: 
lung_hdmi &lt;- imputedGLMnetwork(lung, thyroid, sigma = 2, lambdas = lambdas,
                               m = 10, B = 5)
lung_net &lt;- GLMnetToGraph(lung_hdmi, 0.75)
lung_net
plot(lung_net)

## End(Not run)

</code></pre>

<hr>
<h2 id='GLMnetwork'>Infer a network from RNA-seq expression.</h2><span id='topic+GLMnetwork'></span>

<h3>Description</h3>

<p><code>GLMnetwork</code> infers a network from RNA-seq expression with the
log-linear Poisson graphical model of (Allen and Liu, 2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLMnetwork(counts, lambdas = NULL, normalize = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GLMnetwork_+3A_counts">counts</code></td>
<td>
<p>a n x p matrix of RNA-seq expression (numeric matrix or data
frame)</p>
</td></tr>
<tr><td><code id="GLMnetwork_+3A_lambdas">lambdas</code></td>
<td>
<p>a sequence of decreasing positive numbers to control the
regularization (numeric vector). Default to <code>NULL</code></p>
</td></tr>
<tr><td><code id="GLMnetwork_+3A_normalize">normalize</code></td>
<td>
<p>logical value to normalize predictors in the log-linear
Poisson graphical model. If <code>TRUE</code>, log normalization and scaling are
performed prior the model is fit. Default to <code>TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>When input <code>lambdas</code> are null the default sequence of
<code><a href="glmnet.html#topic+glmnet">glmnet</a></code> for the first model (the one with the first
column of <code>count</code> as the target) is used.
</p>


<h3>Value</h3>

<p>S3 object of class <code>GLMnetwork</code>: a list consisting of
</p>
<table>
<tr><td><code>lambda</code></td>
<td>
<p> regularization parameters used for LLGM path(vector)</p>
</td></tr>
<tr><td><code>path</code></td>
<td>
<p> a list having the same length than <code>lambda</code>. It
contains the estimated coefficients (in a matrix) along the path</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Allen, G. and Liu, Z. (2012) A log-linear model for inferring
genetic networks from high-throughput sequencing data. In <em>Proceedings
of IEEE International Conference on Bioinformatics and Biomedecine (BIBM)</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+stabilitySelection">stabilitySelection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 10))
ref_lung &lt;- GLMnetwork(lung, lambdas = lambdas)

</code></pre>

<hr>
<h2 id='GLMpath'>Methods for 'GLMpath' objects.</h2><span id='topic+GLMpath'></span><span id='topic+summary.GLMpath'></span><span id='topic+print.GLMpath'></span><span id='topic+GLMpath-class'></span>

<h3>Description</h3>

<p>Methods for the result of <code><a href="#topic+GLMnetwork">GLMnetwork</a></code>
(<code>GLMpath</code> object)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GLMpath'
summary(object, ...)

## S3 method for class 'GLMpath'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GLMpath_+3A_object">object</code></td>
<td>
<p><code>GLMpath</code> object</p>
</td></tr>
<tr><td><code id="GLMpath_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="GLMpath_+3A_x">x</code></td>
<td>
<p><code>GLMpath</code> object</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GLMnetwork">GLMnetwork</a></code>
</p>

<hr>
<h2 id='HDImputed'>Methods for 'HDImputed' objects.</h2><span id='topic+HDImputed'></span><span id='topic+summary.HDimputed'></span><span id='topic+summary.HDImputed'></span><span id='topic+print.HDImputed'></span><span id='topic+HDImputed-class'></span>

<h3>Description</h3>

<p>Methods for the result of <code><a href="#topic+imputeHD">imputeHD</a></code>
(<code>HDImputed</code> object)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HDimputed'
summary(object, ...)

## S3 method for class 'HDImputed'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HDImputed_+3A_object">object</code></td>
<td>
<p><code>HDImputed</code> object</p>
</td></tr>
<tr><td><code id="HDImputed_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="HDImputed_+3A_x">x</code></td>
<td>
<p><code>HDImputed</code> object</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+imputeHD">imputeHD</a></code>
</p>

<hr>
<h2 id='HDpath'>Methods for 'HDpath' objects.</h2><span id='topic+HDpath'></span><span id='topic+summary.HDpath'></span><span id='topic+print.HDpath'></span><span id='topic+HDpath-class'></span><span id='topic+plot.HDpath'></span>

<h3>Description</h3>

<p>Methods for the result of <code><a href="#topic+imputedGLMnetwork">imputedGLMnetwork</a></code>
(<code>HDpath</code> object)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HDpath'
summary(object, ...)

## S3 method for class 'HDpath'
print(x, ...)

## S3 method for class 'HDpath'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HDpath_+3A_object">object</code></td>
<td>
<p><code>HDpath</code> object</p>
</td></tr>
<tr><td><code id="HDpath_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="HDpath_+3A_x">x</code></td>
<td>
<p><code>HDpath</code> object</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+imputedGLMnetwork">imputedGLMnetwork</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
data(thyroid)
nobs &lt;- nrow(lung)
miss_ind &lt;- sample(1:nobs, round(0.2 * nobs), replace = FALSE)
lung[miss_ind, ] &lt;- NA
lung &lt;- na.omit(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 10))
## Not run: 
lung_hdmi &lt;- imputedGLMnetwork(lung, thyroid, sigma = 2, lambdas = lambdas,
                               m = 10, B = 5)
plot(lung_hdmi)

## End(Not run)

</code></pre>

<hr>
<h2 id='imputedGLMnetwork'>Multiple hot-deck imputation and network inference from RNA-seq data.</h2><span id='topic+imputedGLMnetwork'></span>

<h3>Description</h3>

<p><code>imputedGLMnetwork</code> performs a multiple hot-deck imputation and infers a
network for each imputed dataset with a log-linear Poisson graphical model
(LLGM).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imputedGLMnetwork(X, Y, sigma, m = 50, lambdas = NULL, B = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imputedGLMnetwork_+3A_x">X</code></td>
<td>
<p>n x p numeric matrix containing RNA-seq expression with missing rows
(numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="imputedGLMnetwork_+3A_y">Y</code></td>
<td>
<p>auxiliary dataset (n' x q numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="imputedGLMnetwork_+3A_sigma">sigma</code></td>
<td>
<p>affinity threshold for donor pool</p>
</td></tr>
<tr><td><code id="imputedGLMnetwork_+3A_m">m</code></td>
<td>
<p>number of replicates in multiple imputation (integer). Default to 50</p>
</td></tr>
<tr><td><code id="imputedGLMnetwork_+3A_lambdas">lambdas</code></td>
<td>
<p>a sequence of decreasing positive numbers to control the
regularization (numeric vector). Default to <code>NULL</code></p>
</td></tr>
<tr><td><code id="imputedGLMnetwork_+3A_b">B</code></td>
<td>
<p>number of iterations for stability selection. Default to 20</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When input <code>lambdas</code> are null the default sequence of
<code><a href="glmnet.html#topic+glmnet">glmnet</a></code> for the first model (the one with the first
column of <code>count</code> as the target) is used. A common default sequence is
generated for all imputed datasets using this method.
</p>


<h3>Value</h3>

<p>S3 object of class <code>HDpath</code>: a list consisting of
</p>
<table>
<tr><td><code>path</code></td>
<td>
<p> a list of <code>m</code> data frames, each containing the
adjacency matrix of the inferred network obtained from the corresonding
imputed dataset. The regularization parameter is selected by StARS</p>
</td></tr>
<tr><td><code>efreq</code></td>
<td>
<p> a numeric matrix of size p x p,  which indicates the
number of times an edge has been predicted among the <code>m</code> inferred
networks</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Imbert, A., Valsesia, A., Le Gall, C., Armenise, C., Lefebvre,
G. Gourraud, P.A., Viguerie, N. and Villa-Vialaneix, N. (2018) Multiple
hot-deck imputation for network inference from RNA sequencing data.
<em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btx819">doi:10.1093/bioinformatics/btx819</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
data(thyroid)
nobs &lt;- nrow(lung)
miss_ind &lt;- sample(1:nobs, round(0.2 * nobs), replace = FALSE)
lung[miss_ind, ] &lt;- NA
lung &lt;- na.omit(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 10))
## Not run: 
lung_hdmi &lt;- imputedGLMnetwork(lung, thyroid, sigma = 2, lambdas = lambdas,
                               m = 10, B = 5)

## End(Not run)

</code></pre>

<hr>
<h2 id='imputeHD'>Impute missing row datasets with multiple hot deck.</h2><span id='topic+imputeHD'></span>

<h3>Description</h3>

<p><code>imputeHD</code> performs multiple hot-deck imputation on an input data frame
with missing rows. Each missing row is imputed with a unique donor. This
method requires an auxiliary dataset to compute similaritities between
individuals and create the pool of donors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imputeHD(X, Y, sigma, m = 50, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imputeHD_+3A_x">X</code></td>
<td>
<p>n x p numeric matrix containing RNA-seq expression with missing rows
(numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="imputeHD_+3A_y">Y</code></td>
<td>
<p>auxiliary dataset (n' x q numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="imputeHD_+3A_sigma">sigma</code></td>
<td>
<p>threshold for hot-deck imputation (numeric, positive)</p>
</td></tr>
<tr><td><code id="imputeHD_+3A_m">m</code></td>
<td>
<p>number of replicates in multiple imputation (integer). Default to 50</p>
</td></tr>
<tr><td><code id="imputeHD_+3A_seed">seed</code></td>
<td>
<p>single value, interpreted as an in integer, used to initialize
the random number generation state. Default to <code>NULL</code> (not used in this
case)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Missing values are identified by matching rownames in <code>X</code> and
<code>Y</code>. If rownames are not provided the missing rows in <code>X</code> are
supposed to correspond to the last rows of <code>Y</code>.
</p>


<h3>Value</h3>

<p>S3 object of class <code>HDImputed</code>: a list consisting of
</p>
<table>
<tr><td><code>donors</code></td>
<td>
<p> a list. Each element of this list contains the donor
pool for every missing observations</p>
</td></tr>
<tr><td><code>draws</code></td>
<td>
<p> a data frame which indicates which donor was chosen
for each missing samples</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p> a list of <code>m</code> imputed datasets</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a> 
</p>


<h3>References</h3>

<p>Imbert, A., Valsesia, A., Le Gall, C., Armenise, C., Lefebvre,
G. Gourraud, P.A., Viguerie, N. and Villa-Vialaneix, N. (2018) Multiple
hot-deck imputation for network inference from RNA sequencing data.
<em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btx819">doi:10.1093/bioinformatics/btx819</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chooseSigma">chooseSigma</a></code>, <code><a href="#topic+imputedGLMnetwork">imputedGLMnetwork</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
data(thyroid)
nobs &lt;- nrow(lung)
miss_ind &lt;- sample(1:nobs, round(0.2 * nobs), replace = FALSE)
lung[miss_ind, ] &lt;- NA
lung &lt;- na.omit(lung)
imputed_lung &lt;- imputeHD(lung, thyroid, sigma = 2)

</code></pre>

<hr>
<h2 id='lung'>RNA-seq expression from lung tissue (GTEx).</h2><span id='topic+lung'></span>

<h3>Description</h3>

<p>This data set is a small subset of the full data set from GTEx.  It contains
RNA-seq expressions measured from lung tissue. The RNA-seq expressions have
been normalized with the TMM method.
</p>


<h3>Format</h3>

<p>a data frame with 221 rows and 100 variables (genes). Row names are
identifiers for individuals.
</p>


<h3>Author(s)</h3>

<p>Alyssa Imbert &lt;alyssa.imbert@gmail.com&gt;
</p>


<h3>Source</h3>

<p>The raw data were download from
<a href="https://gtexportal.org/home/index.html">https://gtexportal.org/home/index.html</a>. The TMM normalization of
RNA-seq expression was performed with the R package <code>edgeR</code>.
</p>

<hr>
<h2 id='RNAseqNetUsersGuide'>View RNAseqNet User's Guide</h2><span id='topic+RNAseqNetUsersGuide'></span>

<h3>Description</h3>

<p>Find the location of the RNAseqNet User's Guide and optionnaly
opens it
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNAseqNetUsersGuide(html = TRUE, view = html)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RNAseqNetUsersGuide_+3A_html">html</code></td>
<td>
<p>logical. Should the document returned by the function be the
compiled PDF or the Rmd source. Default to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="RNAseqNetUsersGuide_+3A_view">view</code></td>
<td>
<p>logical. Should the document be opened using the default HTML
viewer? Default to <code>html</code>. It has no effect if <code>html = FALSE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>vignette("RNAseqNet")</code> will find the short
RNAseqNet vignette that describes how to obtain the RNAseqNet User's Guide.
The User's Guide is not itself a true vignette because it is not
automatically generated during the package build process. However, the
location of the Rmarkdown source is returned by the function if
<code>html = FALSE</code>.
If the operating system is not Windows, then the HTML viewer used is that
given by <code>Sys.getenv("R_BROWSER")</code>. The HTML viewer can be changed using
<code>Sys.setenv(R_BROWSER = )</code>.
</p>


<h3>Value</h3>

<p>Character string giving the file location. If <code>html = TRUE</code> and
<code>view = TRUE</code>, the HTML document reader is started and the User's Guide
is opened in it.
</p>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>RNAseqNetUsersGuide(view = FALSE)
RNAseqNetUsersGuide(html = FALSE)
## Not run: RNAseqNetUsersGuide()
</code></pre>

<hr>
<h2 id='stabilitySelection'>Selection of the regularization parameter by StARS (Liu et al., 2010).</h2><span id='topic+stabilitySelection'></span>

<h3>Description</h3>

<p><code>stabilitySelection</code> implements the regularization parameter selection
of (Liu et al., 2010) called 'Stability Approach to Regularization Selection'
(StARS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stabilitySelection(counts, lambdas = NULL, B = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stabilitySelection_+3A_counts">counts</code></td>
<td>
<p>a n x p matrix of RNA-seq expression (numeric matrix or data
frame)</p>
</td></tr>
<tr><td><code id="stabilitySelection_+3A_lambdas">lambdas</code></td>
<td>
<p>a sequence of decreasing positive numbers to control the
regularization (numeric vector). Default to <code>NULL</code></p>
</td></tr>
<tr><td><code id="stabilitySelection_+3A_b">B</code></td>
<td>
<p>number of iterations for stability selection. Default to 20</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When input <code>lambdas</code> are null the default sequence of
<code><a href="glmnet.html#topic+glmnet">glmnet</a></code> (see <code><a href="#topic+GLMnetwork">GLMnetwork</a></code> for details).
</p>


<h3>Value</h3>

<p>S3 object of class <code>stabilitySelection</code> : a list consisting of
</p>
<table>
<tr><td><code>lambdas</code></td>
<td>
<p> numeric regularization parameters used for
regularization path</p>
</td></tr>
<tr><td><code>B</code></td>
<td>
<p> number of iterations for stability selection</p>
</td></tr>
<tr><td><code>best</code></td>
<td>
<p> index of the regularization parameter selected by StARS
in <code>lambdas</code></p>
</td></tr>
<tr><td><code>variabilities</code></td>
<td>
<p> numeric vector having same length than lambdas
and providing the variability value as defined by StARS along the path</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Liu, H., Roeber, K. and Wasserman, L. (2010) Stability approach
to regularization selection (StARS) for high dimensional graphical models. In
<em>Proceedings of Neural Information Processing Systems (NIPS 2010)</em>,
<strong>23</strong>, 1432-1440, Vancouver, Canada.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GLMnetwork">GLMnetwork</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung)
lambdas &lt;- 4 * 10^(seq(0, -2, length = 5))
stability_lung &lt;- stabilitySelection(lung, lambdas = lambdas, B = 4)
## Not run: plot(stability_lung)

</code></pre>

<hr>
<h2 id='stars'>Methods for 'stars' objects.</h2><span id='topic+stars'></span><span id='topic+summary.stars'></span><span id='topic+print.stars'></span><span id='topic+plot.stars'></span><span id='topic+stars-class'></span>

<h3>Description</h3>

<p>Methods for the result of <code><a href="#topic+stabilitySelection">stabilitySelection</a></code>
(<code>stars</code> object)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'stars'
summary(object, ...)

## S3 method for class 'stars'
print(x, ...)

## S3 method for class 'stars'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stars_+3A_object">object</code></td>
<td>
<p><code>stars</code> object</p>
</td></tr>
<tr><td><code id="stars_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="stars_+3A_x">x</code></td>
<td>
<p><code>stars</code> object</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+stabilitySelection">stabilitySelection</a></code>
</p>

<hr>
<h2 id='thyroid'>RNA-seq expression from thyroid tissue (GTEx).</h2><span id='topic+thyroid'></span>

<h3>Description</h3>

<p>This data set is a small subset of the full data set from GTEx. It contains
RNA-seq expressions measured from thyroid tissue. The RNA-seq expressions
have been normalized with the TMM method.
</p>


<h3>Format</h3>

<p>a data frame with 221 rows and 50 variables (genes). Rown ames are
identifiers for individuals.
</p>


<h3>Author(s)</h3>

<p>Alyssa Imbert &lt;alyssa.imbert@gmail.com&gt;
</p>


<h3>Source</h3>

<p>The raw data were downloaded from
<a href="https://gtexportal.org/home/index.html">https://gtexportal.org/home/index.html</a>. The TMM normalisation of
RNA-seq expression was performed with the R package <code>edgeR</code>.
</p>

<hr>
<h2 id='varIntra'>Average intra-donor pool variance.</h2><span id='topic+varIntra'></span>

<h3>Description</h3>

<p><code>varIntra</code> computes the average intra-donor pool variance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varIntra(X, Y, donors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varIntra_+3A_x">X</code></td>
<td>
<p>n x p numeric matrix containing RNA-seq expression with missing rows
(numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="varIntra_+3A_y">Y</code></td>
<td>
<p>auxiliary dataset (n' x q numeric matrix or data frame)</p>
</td></tr>
<tr><td><code id="varIntra_+3A_donors">donors</code></td>
<td>
<p>donor pool (a list, as given <code>$donors</code> obtained from the
function <code><a href="#topic+imputeHD">imputeHD</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>varIntra</code> returns a numeric value which is the average
intra-donor pool variance, as described in (Imbert <em>et al.</em>, 2018).
</p>


<h3>Author(s)</h3>

<p>Alyssa Imbert, <a href="mailto:alyssa.imbert@gmail.com">alyssa.imbert@gmail.com</a>
</p>
<p>Nathalie Vialaneix, <a href="mailto:nathalie.vialaneix@inrae.fr">nathalie.vialaneix@inrae.fr</a>
</p>


<h3>References</h3>

<p>Imbert, A., Valsesia, A., Le Gall, C., Armenise, C., Lefebvre,
G. Gourraud, P.A., Viguerie, N. and Villa-Vialaneix, N. (2018) Multiple
hot-deck imputation for network inference from RNA sequencing data.
<em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btx819">doi:10.1093/bioinformatics/btx819</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+imputeHD">imputeHD</a></code>, <code><a href="#topic+chooseSigma">chooseSigma</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
