<!DOCTYPE html><html><head><title>Help for package rNeighborGWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rNeighborGWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_PVEnei'><p>Calculating phenotypic variation explained by neighbor effects</p></a></li>
<li><a href='#delta_PVE'><p>Estimating the effective scale of neighbor effects</p></a></li>
<li><a href='#gaston2neiGWAS'><p>Convert gaston's bed.matrix data to rNeighborGWAS genotype data.</p></a></li>
<li><a href='#min_dist'><p>Calculating the minimum distance</p></a></li>
<li><a href='#nei_coval'><p>Calculating neighbor genotypic identity</p></a></li>
<li><a href='#nei_lm'><p>Standard linear models for testing self and neighbor effects</p></a></li>
<li><a href='#nei_lmm'><p>Mixed models for testing self and neighbor effects</p></a></li>
<li><a href='#nei_simu'><p>Simulating phenotypes with self and neighbor effects</p></a></li>
<li><a href='#neiGWAS'><p>Genome-wide association mapping of neighbor effects</p></a></li>
<li><a href='#qtl_pheno_simu'><p>Simulating phenotype values with neighbor effects.</p></a></li>
<li><a href='#rNeighborGWAS-package'><p>rNeigborGWAS: Testing Neighbor Effects in Marker-based Regressions</p></a></li>
<li><a href='#w'><p>Calculating a distance decay weight</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Testing Neighbor Effects in Marker-Based Regressions</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>Description:</td>
<td>To incorporate neighbor genotypic identity into genome-wide association studies, the package provides a set of functions for variation partitioning and association mapping. The theoretical background of the method is described in Sato et al. (2021) &lt;<a href="https://doi.org/10.1038%2Fs41437-020-00401-w">doi:10.1038/s41437-020-00401-w</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yasuhiro Sato &lt;sato.yasuhiro.36c@kyoto-u.jp&gt;</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>gaston, Matrix, RcppParallel, stats, graphics</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-03 13:11:52 UTC; yassato</td>
</tr>
<tr>
<td>Author:</td>
<td>Yasuhiro Sato <a href="https://orcid.org/0000-0002-6466-723X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Eiji Yamamoto [aut],
  Kentaro K. Shimizu
    <a href="https://orcid.org/0000-0002-6483-1781"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Atsushi J. Nagano <a href="https://orcid.org/0000-0001-7891-5049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-10 16:10:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_PVEnei'>Calculating phenotypic variation explained by neighbor effects</h2><span id='topic+calc_PVEnei'></span>

<h3>Description</h3>

<p>A function to calculate PVE by neighbor effects for a series of neighbor distance using a mixed model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_PVEnei(
  pheno,
  geno,
  smap,
  scale_seq,
  addcovar = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  n_core = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_PVEnei_+3A_pheno">pheno</code></td>
<td>
<p>A numeric vector including phenotypes for individuals</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_geno">geno</code></td>
<td>
<p>An individual x marker matrix. Bialleles (i.e., A or a) must be converted into -1 or 1 digit.</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial points along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_scale_seq">scale_seq</code></td>
<td>
<p>A numeric vector including a set of the maximum spatial distance between a focal individual and neighbors to define neighbor effects. A scalar is also allowed.</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_grouping">grouping</code></td>
<td>
<p>A positive integer vector assigning each individual to a group. This argument can be useful when a &quot;smap&quot; contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_response">response</code></td>
<td>
<p>An option to select if the phenotype is a <code>"quantitative"</code> trait subject to linear models, or a <code>"binary"</code> trait subject to logistic models.</p>
</td></tr>
<tr><td><code id="calc_PVEnei_+3A_n_core">n_core</code></td>
<td>
<p>No. of cores for a multi-core computation. This does not work for Windows OS. Default is a single-core computation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses mixed models via the <code>gaston</code> package (Perdry &amp; Dandine-Roulland 2020).
If <code>"binary"</code> is selected, <code>logistic.mm.aireml()</code> is called via the <code>gaston</code> package.
In such a case, <code>PVEnei</code> below is replaced by the ratio of phenotypic variation explained (RVE) by neighbor effects as RVE_nei = <code class="reqn">\sigma_2^2</code>/<code class="reqn">\sigma_1^2</code> and p-values are not provided.
</p>


<h3>Value</h3>

<p>A numeric matrix including a given spatial scale, PVE by neighbor effects, and p-values.
</p>

<ul>
<li><p><code>scale</code> Maximum neighbor distance given as an argument
</p>
</li>
<li><p><code>PVEself</code> Proportion of phenotypic variation explained (PVE) by self effects. RVE is returned when <code>response = "binary"</code>
</p>
</li>
<li><p><code>PVEnei</code> Proportion of phenotypic variation explained (PVE) by neighbor effects. RVE is returned when <code>response = "binary"</code>
</p>
</li>
<li><p><code>p-value</code> p-value by a likelihood ratio test between models with or without neighbor effects (when s is not zero); or between a null model and model with self effects alone (when s = 0). NA when <code>response = "binary"</code>
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>

<p>Perdry H, Dandine-Roulland C. (2020) gaston: Genetic Data Handling (QC, GRM, LD, PCA) &amp; Linear Mixed Models. https://CRAN.R-project.org/package=gaston
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
g &lt;- matrix(sample(c(-1,1),100*1000,replace = TRUE),100,1000)
gmap &lt;- cbind(c(rep(1,nrow(g)/2),rep(2,nrow(g)/2)),c(1:ncol(g)))
x &lt;- runif(nrow(g),1,100)
y &lt;- runif(nrow(g),1,100)
smap &lt;- cbind(x,y)
grouping &lt;- c(rep(1,nrow(g)/2),rep(2,nrow(g)/2))
pheno &lt;- nei_simu(geno=g,smap=smap,scale=44,grouping=grouping,n_causal=50,pveB=0.4,pve=0.8)

fake_nei &lt;- list()
fake_nei[[1]] &lt;- g
fake_nei[[2]] &lt;- gmap
fake_nei[[3]] &lt;- smap
fake_nei[[4]] &lt;- data.frame(pheno,grouping)
names(fake_nei) &lt;- c("geno","gmap","smap","pheno")

min_s &lt;- min_dist(fake_nei$smap, fake_nei$pheno$grouping)
scale_seq &lt;- c(min_s, quantile(dist(fake_nei$smap),c(0.2*rep(1:5))))

pve_out &lt;- calc_PVEnei(geno=fake_nei$geno, pheno=fake_nei$pheno[,1],
                       smap=fake_nei$smap, scale_seq=scale_seq,
                       addcovar=as.matrix(fake_nei$pheno$grouping),
                       grouping=fake_nei$pheno$grouping)
delta_PVE(pve_out)
</code></pre>

<hr>
<h2 id='delta_PVE'>Estimating the effective scale of neighbor effects</h2><span id='topic+delta_PVE'></span>

<h3>Description</h3>

<p>A function to calculate <code class="reqn">\Delta</code>PVE that estimates the effective scale of neighbor effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delta_PVE(res, fig = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delta_PVE_+3A_res">res</code></td>
<td>
<p>Output results of <code>calc_PVEnei()</code>.</p>
</td></tr>
<tr><td><code id="delta_PVE_+3A_fig">fig</code></td>
<td>
<p>TRUE/FALSE to plot the results (or not). Default is TRUE.</p>
</td></tr>
<tr><td><code id="delta_PVE_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to <code>plot()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated effective scale and proportion of phenotypic variation explained by neighbor effects at that scale.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_PVEnei">calc_PVEnei</a></code>
</p>

<hr>
<h2 id='gaston2neiGWAS'>Convert <a href="gaston.html#topic+gaston">gaston</a>'s bed.matrix data to rNeighborGWAS genotype data.</h2><span id='topic+gaston2neiGWAS'></span>

<h3>Description</h3>

<p>A function convert a bed.matrix dataset to rNeighborGWAS genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaston2neiGWAS(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gaston2neiGWAS_+3A_x">x</code></td>
<td>
<p>A <a href="gaston.html#topic+bed.matrix">bed.matrix</a> created using the <code>gaston</code> package (Perdry &amp; Dandine-Roulland 2020).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function converts genotype data into -1, 0, or 1 digit as the rNeighborGWAS format. Zero indicates heterozygotes.
</p>


<h3>Value</h3>

<p>A list including an individual x marker matrix, a data.frame including chromosome numbers in the first column, and SNP positions in the second column, and a numeric vector including phenotypes for individuals.
</p>

<ul>
<li><p><code>geno</code> Individual x marker matrix
</p>
</li>
<li><p><code>gmap</code> Data.frame including chromosome numbers in the first column, and SNP positions in the second column
</p>
</li>
<li><p><code>pheno</code> Numeric vector including phenotypes for individuals
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>

<p>Perdry H, Dandine-Roulland C. (2020) gaston: Genetic Data Handling (QC, GRM, LD, PCA) &amp; Linear Mixed Models. https://CRAN.R-project.org/package=gaston
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("TTN", package="gaston")
x &lt;- gaston::as.bed.matrix(TTN.gen, TTN.fam, TTN.bim)
g &lt;- gaston2neiGWAS(x)
</code></pre>

<hr>
<h2 id='min_dist'>Calculating the minimum distance</h2><span id='topic+min_dist'></span>

<h3>Description</h3>

<p>A function to calculate a Euclidian distance including at least one neighbor for all individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>min_dist(smap, grouping = rep(1, nrow(smap)))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="min_dist_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial points along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="min_dist_+3A_grouping">grouping</code></td>
<td>
<p>A positive integer vector assigning each individual to a group. This argument can be useful when a &quot;smap&quot; contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a scalar of the minimum Euclidian distance that allows all individuals to have at least one neighbor.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
g &lt;- matrix(sample(c(-1,1),100*1000,replace = TRUE),100,1000)
gmap = cbind(c(rep(1,nrow(g)/2),rep(2,nrow(g)/2)),c(1:ncol(g)))
x &lt;- runif(nrow(g),1,100)
y &lt;- runif(nrow(g),1,100)
smap &lt;- cbind(x,y)
grouping &lt;- c(rep(1,nrow(g)/2), rep(2,nrow(g)/2))
pheno &lt;- nei_simu(geno=g,smap=smap,scale=44,grouping=grouping,n_causal=50,pveB=0.4,pve=0.8)

fake_nei &lt;- list()
fake_nei[[1]] &lt;- g
fake_nei[[2]] &lt;- gmap
fake_nei[[3]] &lt;- smap
fake_nei[[4]] &lt;- data.frame(pheno,grouping)
names(fake_nei) &lt;- c("geno","gmap","smap","pheno")

min_s &lt;- min_dist(fake_nei$smap, fake_nei$pheno$grouping)
</code></pre>

<hr>
<h2 id='nei_coval'>Calculating neighbor genotypic identity</h2><span id='topic+nei_coval'></span>

<h3>Description</h3>

<p>A function to calculate neighbor genotypic identity, with a given reference scale and a degree of distance decay.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nei_coval(
  geno,
  smap,
  scale,
  alpha = Inf,
  kernel = c("exp", "gaussian"),
  grouping = rep(1, nrow(smap)),
  n_core = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nei_coval_+3A_geno">geno</code></td>
<td>
<p>An individual x marker matrix. Bialleles (i.e., A or a) must be converted into -1 or 1 digit.</p>
</td></tr>
<tr><td><code id="nei_coval_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial points along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="nei_coval_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="nei_coval_+3A_alpha">alpha</code></td>
<td>
<p>An option to set a distance decay coefficient <code class="reqn">\alpha</code> in a dispersal kernel. Default is set at Inf, meaning no distance decay.</p>
</td></tr>
<tr><td><code id="nei_coval_+3A_kernel">kernel</code></td>
<td>
<p>An option to select either <code>"exp"</code> or <code>"gaussian"</code> for a negative exponential kernel or Gaussian kernel, respectively.</p>
</td></tr>
<tr><td><code id="nei_coval_+3A_grouping">grouping</code></td>
<td>
<p>A positive integer vector assigning each individual to a group. This argument can be useful when a &quot;smap&quot; contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="nei_coval_+3A_n_core">n_core</code></td>
<td>
<p>No. of cores for a multi-core computation. This does not work for Windows OS. Default is a single-core computation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default setting is recommended for <code>alpha</code> and <code>kernel</code> arguments unless spatial distance decay of neighbor effects needs to be modeled.
If <code>alpha</code> is not <code>Inf</code>, output variables are weighted by a distance decay from a focal individual to <code>scale</code>.
For the type of dispersal kernel in the distance decay, we can choose a negative exponential or Gaussian kernel as a fat-tailed or thin-tailed distribution, respectively.
See Nathan et al. (2012) for detailed characteristics of the two dispersal kernels.
</p>


<h3>Value</h3>

<p>A numeric matrix for neighbor covariates, with no. of individuals x markers.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>

<p>Nathan R, Klein E, Robledo-Arnuncio JJ, Revilla E. (2012) Dispersal kernels: review. In: Clobert J, Baguette M, Benton TG, Bullock JM (Eds.), <em>Dispersal Ecology and Evolution</em>. Oxford University Press, pp.186-210.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
g &lt;- matrix(sample(c(-1,1),100*1000,replace = TRUE),100,1000)
gmap &lt;- cbind(c(rep(1,nrow(g)/2),rep(2,nrow(g)/2)),c(1:ncol(g)))
x &lt;- runif(nrow(g),1,100)
y &lt;- runif(nrow(g),1,100)
smap &lt;- cbind(x,y)
grouping &lt;- c(rep(1,nrow(g)/2), rep(2,nrow(g)/2))

g_nei &lt;- nei_coval(g,smap,44,grouping = grouping)
</code></pre>

<hr>
<h2 id='nei_lm'>Standard linear models for testing self and neighbor effects</h2><span id='topic+nei_lm'></span>

<h3>Description</h3>

<p>A function to provide coefficients and p-values of self and neighbor effects for each marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nei_lm(
  geno,
  g_nei,
  pheno,
  addcovar = NULL,
  response = c("quantitative", "binary"),
  n_core = 1L,
  asym = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nei_lm_+3A_geno">geno</code></td>
<td>
<p>An individual x marker matrix. Bialleles (i.e., A or a) must be converted into -1 or 1 digit.</p>
</td></tr>
<tr><td><code id="nei_lm_+3A_g_nei">g_nei</code></td>
<td>
<p>An output of <code>nei_coval()</code> object, namely an individual x marker matrix including neighbor genotypic identity.</p>
</td></tr>
<tr><td><code id="nei_lm_+3A_pheno">pheno</code></td>
<td>
<p>A numeric vector including phenotypes for individuals</p>
</td></tr>
<tr><td><code id="nei_lm_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="nei_lm_+3A_response">response</code></td>
<td>
<p>An option to select if the phenotype is a <code>"quantitative"</code> trait subject to linear models, or a <code>"binary"</code> trait subject to logistic models.</p>
</td></tr>
<tr><td><code id="nei_lm_+3A_n_core">n_core</code></td>
<td>
<p>No. of cores for a multi-core computation. This does not work for Windows OS. Default is a single-core computation.</p>
</td></tr>
<tr><td><code id="nei_lm_+3A_asym">asym</code></td>
<td>
<p>If TRUE, asymmetric neighbor effects are also tested and returned as &quot;beta_sxn&quot; and &quot;p_sxn&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a subset of <code>neiGWAS()</code>. <code>nei_lm()</code> gives detailed results when the option <code>model="lm"</code> is selected in <code>neiGWAS()</code>.
</p>


<h3>Value</h3>

<p>A data.frame including coefficients and p-values of self and neighbor effects, without the chromosome numbers and marker position.
</p>

<ul>
<li><p><code>beta_self</code> coefficient for self effects
</p>
</li>
<li><p><code>beta_self</code> coefficient for neighbor effects
</p>
</li>
<li><p><code>p_self</code> p-value for self effects by a likelihood ratio test between a null and standard linear model
</p>
</li>
<li><p><code>p_nei</code> p-value for neighbor effects by a likelihood ratio test between models with or without neighbor effects
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neiGWAS">neiGWAS</a></code>
</p>

<hr>
<h2 id='nei_lmm'>Mixed models for testing self and neighbor effects</h2><span id='topic+nei_lmm'></span>

<h3>Description</h3>

<p>A function to provide coefficients and p-values of self and neighbor effects for each marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nei_lmm(
  geno,
  g_nei,
  pheno,
  addcovar = NULL,
  response = c("quantitative", "binary"),
  n_core = 1L,
  asym = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nei_lmm_+3A_geno">geno</code></td>
<td>
<p>An individual x marker matrix. Bialleles (i.e., A or a) must be converted into -1 or 1 digit.</p>
</td></tr>
<tr><td><code id="nei_lmm_+3A_g_nei">g_nei</code></td>
<td>
<p>An output of <code>nei_coval()</code> object, namely an individual x marker matrix including neighbor genotypic identity.</p>
</td></tr>
<tr><td><code id="nei_lmm_+3A_pheno">pheno</code></td>
<td>
<p>A numeric vector including phenotypes for individuals</p>
</td></tr>
<tr><td><code id="nei_lmm_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="nei_lmm_+3A_response">response</code></td>
<td>
<p>An option to select if the phenotype is a <code>"quantitative"</code> trait subject to linear models, or a <code>"binary"</code> trait subject to logistic models.</p>
</td></tr>
<tr><td><code id="nei_lmm_+3A_n_core">n_core</code></td>
<td>
<p>No. of cores for a multi-core computation. This does not work for Windows OS. Default is a single-core computation.</p>
</td></tr>
<tr><td><code id="nei_lmm_+3A_asym">asym</code></td>
<td>
<p>If TRUE, asymmetric neighbor effects are also tested and returned as &quot;beta_sxn&quot; and &quot;p_sxn&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a subset of <code>neiGWAS()</code>. <code>nei_lmm()</code> gives detailed results but requires more computational time.
</p>


<h3>Value</h3>

<p>A data.frame including coefficients and p-values of self and neighbor effects, without the chromosome numbers and marker position.
</p>

<ul>
<li><p><code>beta_self</code> coefficient for self effects
</p>
</li>
<li><p><code>beta_self</code> coefficient for neighbor effects
</p>
</li>
<li><p><code>p_self</code> p-value for self effects by a likelihood ratio test between a null and standard GWAS model
</p>
</li>
<li><p><code>p_nei</code> p-value for neighbor effects by a likelihood ratio test between models with or without neighbor effects
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neiGWAS">neiGWAS</a></code>
</p>

<hr>
<h2 id='nei_simu'>Simulating phenotypes with self and neighbor effects</h2><span id='topic+nei_simu'></span>

<h3>Description</h3>

<p>A function to simulate phenotypes caused by self and neighbor effects, with the proportion of phenotypic variation explained (PVE) by fixed and random effects controlled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nei_simu(
  geno,
  smap,
  scale,
  alpha = Inf,
  grouping = rep(1, nrow(smap)),
  kernel = c("exp", "gaussian"),
  n_causal,
  pveB,
  pve,
  b_ratio = c(1, 1)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nei_simu_+3A_geno">geno</code></td>
<td>
<p>An individual x marker matrix. Bialleles (i.e., A or a) must be converted into -1 or 1 digit.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial points along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_alpha">alpha</code></td>
<td>
<p>Distance decay coefficient <code class="reqn">\alpha</code> in a dispersal kernel. Default is set at Inf, meaning no distance decay.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_grouping">grouping</code></td>
<td>
<p>A positive integer vector assigning each individual to a group. This argument can be useful when a &quot;smap&quot; contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_kernel">kernel</code></td>
<td>
<p>An option to select a negative exponential <code>"exp"</code> or Gaussian <code>"gaussian"</code> for a dispersal kernel of neighbor effects.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_n_causal">n_causal</code></td>
<td>
<p>No. of causal markers in a simulated phenotype</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_pveb">pveB</code></td>
<td>
<p>Proportion of phenotypic variation explained by fixed effects.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_pve">pve</code></td>
<td>
<p>Proportion of phenotypic variation explained by fixed and random effects.</p>
</td></tr>
<tr><td><code id="nei_simu_+3A_b_ratio">b_ratio</code></td>
<td>
<p>A vector composed of two numeric scalars that control the ratio of contributions of self or neighbor effects to a phenotype. The first and second element are for self and neighbor effects, respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of simulated phenotype values for all individuals
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
g &lt;- matrix(sample(c(-1,1),100*1000,replace = TRUE),100,1000)
gmap &lt;- cbind(c(rep(1,nrow(g)/2),rep(2,nrow(g)/2)),c(1:ncol(g)))
x &lt;- runif(nrow(g),1,100)
y &lt;- runif(nrow(g),1,100)
smap &lt;- cbind(x,y)
grouping &lt;- c(rep(1,nrow(g)/2),rep(2,nrow(g)/2))
pheno &lt;- nei_simu(geno=g,smap=smap,scale=44,grouping=grouping,n_causal=50,pveB=0.4,pve=0.8)

fake_nei &lt;- list()
fake_nei[[1]] &lt;- g
fake_nei[[2]] &lt;- gmap
fake_nei[[3]] &lt;- smap
fake_nei[[4]] &lt;- data.frame(pheno,grouping)
names(fake_nei) &lt;- c("geno","gmap","smap","pheno")
</code></pre>

<hr>
<h2 id='neiGWAS'>Genome-wide association mapping of neighbor effects</h2><span id='topic+neiGWAS'></span>

<h3>Description</h3>

<p>A function to test neighbor effects for each marker and to calculate p-values at a given reference scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neiGWAS(
  geno,
  pheno,
  gmap,
  smap,
  scale,
  addcovar = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  model = c("lmm", "lm"),
  n_core = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neiGWAS_+3A_geno">geno</code></td>
<td>
<p>An individual x marker matrix. Bialleles (i.e., A or a) must be converted into -1 or 1 digit.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_pheno">pheno</code></td>
<td>
<p>A numeric vector including phenotypes for individuals</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_gmap">gmap</code></td>
<td>
<p>A matrix or data.frame including chromosome numbers in the first column, and SNP positions in the second column.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial points along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_grouping">grouping</code></td>
<td>
<p>A positive integer vector assigning each individual to a group. This argument can be useful when a &quot;smap&quot; contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_response">response</code></td>
<td>
<p>An option to select if the phenotype is a <code>"quantitative"</code> trait subject to linear models, or a <code>"binary"</code> trait subject to logistic models.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_model">model</code></td>
<td>
<p>An option to select linear mixed model <code>"lmm"</code> or linear model <code>"lm"</code>. Default setting is to use a mixed model.</p>
</td></tr>
<tr><td><code id="neiGWAS_+3A_n_core">n_core</code></td>
<td>
<p>No. of cores for a multi-core computation. This does not work for Windows OS. Default is a single-core computation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calls a mixed model via the <code>gaston</code> package. If <code>"lmm"</code> with <code>"binary"</code> is selected, p-values are based on Wald tests.
This is because the logistic mixed model is based on a pseudo-likelihood and thus likelihood ratio tests are not applicable. See Chen et al. (2016) for the theory.
</p>


<h3>Value</h3>

<p>A data.frame including the chromosome number, marker position, and p-values.
</p>

<ul>
<li><p><code>chr</code> Chromosome number
</p>
</li>
<li><p><code>pos</code> Marker position
</p>
</li>
<li><p><code>p</code> p-value by a likelihood ratio test between models with or without neighbor effects
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>

<p>Chen H, Wang C, Conomos M. et al. (2016) Control for population structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98: 653-666.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
g &lt;- matrix(sample(c(-1,1),100*1000,replace = TRUE),100,1000)
gmap &lt;- cbind(c(rep(1,nrow(g)/2),rep(2,nrow(g)/2)),c(1:ncol(g)))
x &lt;- runif(nrow(g),1,100)
y &lt;- runif(nrow(g),1,100)
smap &lt;- cbind(x,y)
grouping &lt;- c(rep(1,nrow(g)/2),rep(2,nrow(g)/2))
pheno &lt;- nei_simu(geno=g,smap=smap,scale=44,grouping=grouping,n_causal=50,pveB=0.4,pve=0.8)

fake_nei &lt;- list()
fake_nei[[1]] &lt;- g
fake_nei[[2]] &lt;- gmap
fake_nei[[3]] &lt;- smap
fake_nei[[4]] &lt;- data.frame(pheno,grouping)
names(fake_nei) &lt;- c("geno","gmap","smap","pheno")

scale &lt;- 43
gwas_out &lt;- neiGWAS(geno=fake_nei$geno, pheno=fake_nei$pheno[,1],
                    gmap=fake_nei$gmap, smap=fake_nei$smap,
                    scale=scale, addcovar=as.matrix(fake_nei$pheno$grouping),
                    grouping=fake_nei$pheno$grouping)

gaston::manhattan(gwas_out)
gaston::qqplot.pvalues(gwas_out$p)
</code></pre>

<hr>
<h2 id='qtl_pheno_simu'>Simulating phenotype values with neighbor effects.</h2><span id='topic+qtl_pheno_simu'></span>

<h3>Description</h3>

<p>A function to simulate phenotype values with multiple sources of variation controlled
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtl_pheno_simu(
  b_self,
  b_nei,
  eigenK_self,
  eigenK_nei,
  b_ratio = c(1, 1),
  pveB,
  pve
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtl_pheno_simu_+3A_b_self">b_self</code></td>
<td>
<p>A n x 1 genotype vector to design major additive genetic effect.</p>
</td></tr>
<tr><td><code id="qtl_pheno_simu_+3A_b_nei">b_nei</code></td>
<td>
<p>A vector of an explanatory variable for neighbor effects</p>
</td></tr>
<tr><td><code id="qtl_pheno_simu_+3A_eigenk_self">eigenK_self</code></td>
<td>
<p>Products of <code>eigen()</code> with self covariance matrices that are used as explanatory variables for the phenotype.</p>
</td></tr>
<tr><td><code id="qtl_pheno_simu_+3A_eigenk_nei">eigenK_nei</code></td>
<td>
<p>Products of <code>eigen()</code> with neighbor covariance matrices that are used as explanatory variables for the phenotype.</p>
</td></tr>
<tr><td><code id="qtl_pheno_simu_+3A_b_ratio">b_ratio</code></td>
<td>
<p>Ratio for contributions of <code>eigenK_self</code> and <code>eigenK_nei</code> to the phenotype.</p>
</td></tr>
<tr><td><code id="qtl_pheno_simu_+3A_pveb">pveB</code></td>
<td>
<p>Proportion of variance explained by genetic effects attributable to the fixed effects (i.e., b_.. vector).</p>
</td></tr>
<tr><td><code id="qtl_pheno_simu_+3A_pve">pve</code></td>
<td>
<p>Proportion of variance explained by all genetic effects (i.e., b_.. and eigenK_..)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of simulated phenotypes
</p>

<ul>
<li><p><code>y</code> Simulated phenotype values
</p>
</li>
<li><p><code>beta_self</code> major self-genetic effects
</p>
</li>
<li><p><code>beta_nei</code> major neighbor effects
</p>
</li>
<li><p><code>sigma_self</code> self polygenic effects
</p>
</li>
<li><p><code>sigma_nei</code> neighbor polygenic effects
</p>
</li>
<li><p><code>epsilon</code> residuals
</p>
</li>
<li><p><code>res_pveB</code> realized proportion of variation explained by major-effect genes
</p>
</li>
<li><p><code>res_pve</code> realized proportion of variation explained by major-effect genes and polygenic effects
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Eiji Yamamoto, and Yasuhiro Sato
</p>

<hr>
<h2 id='rNeighborGWAS-package'>rNeigborGWAS: Testing Neighbor Effects in Marker-based Regressions</h2><span id='topic+rNeighborGWAS'></span><span id='topic+rNeighborGWAS-package'></span>

<h3>Description</h3>

<p>This package provides a set of functions to test neighbor effects in genome-wide association studies.
The neighbor effects are estimated using the Ising model of ferromagnetism.
See Sato et al. (2021) for motivation and modeling.
</p>


<h3>Details</h3>

<p>The flow of neighbor GWAS consists of two steps, (i) variation partitioning and (ii) association mapping.
In the first step, we compute proportion of phenotypic variation explained by neighbor effects, and estimate their effective area.
In the second step, we test neighbor effects, and map their association score on a genome.
In addition to standard GWAS inputs, spatial information of individuals is required to run these analyses.
See <code>vignette("rNeighborGWAS")</code> for how to use this package.
</p>


<h3>Note</h3>

<p>A developer version of rNeighborGWAS is available at the GitHub reporsitory (<a href="https://github.com/yassato/rNeighborGWAS">https://github.com/yassato/rNeighborGWAS</a>).
If the CRAN version is out of work or you want to use the latest methods, you may install the package via GitHub.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>

<p>Sato Y, Yamamoto E, Shimizu KK, Nagano AJ (2021) Neighbor GWAS: incorporating neighbor genotypic identity into genome-wide association studies of field herbivory. Heredity 126(4):597-614. https://doi.org/10.1038/s41437-020-00401-w
</p>

<hr>
<h2 id='w'>Calculating a distance decay weight</h2><span id='topic+w'></span>

<h3>Description</h3>

<p>A function to calculate, with a negative exponential or Gaussian dispersal kernel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>w(s, a, kernel = c("exp", "gaussian"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="w_+3A_s">s</code></td>
<td>
<p>A numeric scalar indicating spatial distance at which the distance decay is referred</p>
</td></tr>
<tr><td><code id="w_+3A_a">a</code></td>
<td>
<p>A numeric scalar indicating a decay coefficient</p>
</td></tr>
<tr><td><code id="w_+3A_kernel">kernel</code></td>
<td>
<p>An option to select a negative exponential <code>"exp"</code> or Gaussian <code>"gaussian"</code> for a dispersal kernel of neighbor effects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric scalar for a distance decay weight.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
