<!DOCTYPE html><html><head><title>Help for package rNeighborQTL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rNeighborQTL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rNeighborQTL-package'><p>rNeighbor QTL: quantitative trait loci mapping for neighbor effects</p></a></li>
<li><a href='#calc_neiprob'><p>Calculating a set of neighbor QTL effects from conditional genotype probabilities</p></a></li>
<li><a href='#calc_pve'><p>Calculating phenotypic variation explained by neighbor effects</p></a></li>
<li><a href='#decompose_genoprobs'><p>Decomposition of conditional genotype probabilities</p></a></li>
<li><a href='#eff_neighbor'><p>Estimation of self and neighbor QTL effects across a genome</p></a></li>
<li><a href='#genoprobs2selfprobs'><p>Calculating a set of self QTL effects from conditional genotype probabilities</p></a></li>
<li><a href='#get_markers'><p>Reshaping marker information</p></a></li>
<li><a href='#int_neighbor'><p>Testing marker-by-marker epistasis in neighbor QTL effects</p></a></li>
<li><a href='#logLik_glm.fit'><p>Calculating log-likelihood in generalized linear models</p></a></li>
<li><a href='#min_dist'><p>Calculating the minimum distance</p></a></li>
<li><a href='#neiprob'><p>Calculating neighbor QTL effects</p></a></li>
<li><a href='#perm_neighbor'><p>Permutation tests for neighbor effects with a QTL model</p></a></li>
<li><a href='#plot_eff'><p>Plot self and neighbor QTL effects across a genome</p></a></li>
<li><a href='#plot_nei'><p>Plot LOD score for self or neighbor QTL effects</p></a></li>
<li><a href='#scan_neighbor'><p>Genome scan for neighbor effects with a QTL model</p></a></li>
<li><a href='#selfprob'><p>Calculating self QTL effects</p></a></li>
<li><a href='#sim_nei_qtl'><p>Phenotype simulation for neighbor QTL effects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Interval Mapping for Quantitative Trait Loci Underlying Neighbor
Effects</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>To enable quantitative trait loci mapping of neighbor effects, this package extends a single-marker regression to interval mapping. The theoretical background of the method is described in Sato et al. (2021) &lt;<a href="https://doi.org/10.1093%2Fg3journal%2Fjkab017">doi:10.1093/g3journal/jkab017</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yasuhiro Sato &lt;sato.yasuhiro.36c@kyoto-u.jp&gt;</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>gaston, Matrix, qtl, parallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-04-09 04:32:45 UTC; yassato</td>
</tr>
<tr>
<td>Author:</td>
<td>Yasuhiro Sato <a href="https://orcid.org/0000-0002-6466-723X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Kazuya Takeda [aut],
  Atsushi J. Nagano <a href="https://orcid.org/0000-0001-7891-5049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-04-09 05:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='rNeighborQTL-package'>rNeighbor QTL: quantitative trait loci mapping for neighbor effects</h2><span id='topic+rNeighborQTL'></span><span id='topic+rNeighborQTL-package'></span>

<h3>Description</h3>

<p>The &quot;rNeighborQTL&quot; package provides a set of functions to perform a genome scan of neighbor QTL effects.
Theoretical background is described in Sato, Takeda &amp; Nagano (2021).
See <code>vignette("rNeighborQTL")</code> for how to use this package.
</p>


<h3>Details</h3>

<p>Neighbor QTL is an extended method of single-marker regression.
Taking conditional genotype probabilities from <code>r/qtl</code> package, the <code>rNeighborQTL</code> package performs an interval mapping of neighbor genotypic identity.
Alternative to QTL mapping, <code>rNeighborGWAS</code> package is also available for genome-wide association studies (GWAS) via CRAN at <a href="https://cran.r-project.org/package=rNeighborGWAS">https://cran.r-project.org/package=rNeighborGWAS</a>.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Sato Y, Yamamoto E, Shimizu KK, Nagano AJ (2021) Neighbor GWAS: incorporating neighbor genotypic identity into genome-wide association studies of field herbivory. Heredity 126:597â€“614.
</p>
</li>
<li><p> Sato Y, Takeda K, Nagano AJ (2021) Neighbor QTL: an interval mapping method for quantitative trait loci underlying plant neighborhood effects. G3; Genes|Genomes|Genetics 11:jkab017.
</p>
</li></ul>


<hr>
<h2 id='calc_neiprob'>Calculating a set of neighbor QTL effects from conditional genotype probabilities</h2><span id='topic+calc_neiprob'></span>

<h3>Description</h3>

<p>A function to calculate self QTL effects for all individuals, with given deviation coefficients and conditional genotype probabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_neiprob(
  genoprobs,
  a2,
  d2,
  contrasts = NULL,
  smap,
  scale,
  grouping = rep(1, nrow(smap)),
  d2sq0 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_neiprob_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_a2">a2</code></td>
<td>
<p>A numeric scalar indicating additive deviation.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_d2">d2</code></td>
<td>
<p>A numeric scalar indicating dominance deviation.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_contrasts">contrasts</code></td>
<td>
<p>A vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_grouping">grouping</code></td>
<td>
<p>An integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="calc_neiprob_+3A_d2sq0">d2sq0</code></td>
<td>
<p>An option to make AB/AB interaction effects zero.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix containing individuals x marker elements for neighbor QTL effects.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='calc_pve'>Calculating phenotypic variation explained by neighbor effects</h2><span id='topic+calc_pve'></span>

<h3>Description</h3>

<p>A function to calculate the proportion or ratio of phenotypic variation explained (PVE or RVE) by neighbor effects for a series of neighbor distance (<code>s_seq</code>) using mixed models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_pve(
  genoprobs,
  pheno,
  smap,
  s_seq,
  addcovar = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  fig = TRUE,
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_pve_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_pheno">pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_s_seq">s_seq</code></td>
<td>
<p>A numeric vector including a set of the maximum spatial distance between a focal individual and neighbors to define neighbor effects. A scalar is also allowed.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_grouping">grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_response">response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> applies the <code>"lmm.aireml()"</code> or <code>"logistic.mm.aireml()"</code> for a mixed model, respectively.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_fig">fig</code></td>
<td>
<p>TRUE/FALSE to add a figure of Delta PVE or not.</p>
</td></tr>
<tr><td><code id="calc_pve_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calls linear or logistic mixed models via the <code>gaston</code> package (Perdry &amp; Dandine-Roulland 2020).
If <code>"quantitative"</code> is selected, <code>Var_self</code> or <code>Var_nei</code> in the output is given by the proportion of phenotypic variation explained (PVE) by neighbor effects as PVEnei =<code class="reqn">\sigma^2_2/(\sigma^2_1+\sigma^2_2+\sigma^2_e)</code>.
If <code>"binary"</code> is selected, <code>Var_self</code> or <code>Var_nei</code> is given by the ratio of phenotypic variation explained (RVE) by neighbor effects as RVEnei =<code class="reqn">\sigma^2_2/\sigma^2_1</code> and p-values are not available.
This is because a logistic mixed model <code>logistic.mm.aireml()</code> called via the <code>gaston</code> package does not provide <code class="reqn">\sigma^2_e</code> and log-likelihood (see Chen et al. 2016 for the theory).
</p>


<h3>Value</h3>

<p>A matrix containing the maximum neighbor distance, phenotypic variation explained by neighbor effects, and p-value by a likelihood ratio test.
</p>

<ul>
<li><p><code>scale</code> Maximum neighbor distance given as an argument
</p>
</li>
<li><p><code>Var_self</code> Proportion or ratio of phenotypic variation explained (PVE or RVE) by self-genotype effects for linear or logistic mixed models, respectively
</p>
</li>
<li><p><code>Var_nei</code> Proportion or ratio of phenotypic variation explained (PVE or RVE) by neighbor effects for linear or logistic mixed models, respectively
</p>
</li>
<li><p><code>p-value</code> p-value by a likelihood ratio test between models with or without neighbor effects. Self effects are tested when the scale is zero
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Perdry H, Dandine-Roulland C (2020) gaston: Genetic Data Handling (QC, GRM, LD, PCA) &amp; Linear Mixed Models. R package version 1.5.6. https://CRAN.R-project.org/package=gaston
</p>
</li>
<li><p> Chen H, Wang C, Conomos M. et al. (2016) Control for population structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98: 653-666.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)
s_seq &lt;- quantile(dist(test_smap),c(0.1*(1:10)))

test_pve &lt;- calc_pve(genoprobs=test_genoprobs,
                     pheno=test_cross$pheno$phenotype,
                     smap=test_smap, s_seq=s_seq,
                     )
</code></pre>

<hr>
<h2 id='decompose_genoprobs'>Decomposition of conditional genotype probabilities</h2><span id='topic+decompose_genoprobs'></span>

<h3>Description</h3>

<p>A function to decompose <code>qtl</code>'s object of conditional genotype probabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decompose_genoprobs(genoprobs, contrasts = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decompose_genoprobs_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="decompose_genoprobs_+3A_contrasts">contrasts</code></td>
<td>
<p>A vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three numeric matrices for genotype probabilities AA, AB, and BB. Each contains elements of individuals x markers.
</p>

<ul>
<li><p><code>AA</code> Homozygote AA probabilities.
</p>
</li>
<li><p><code>AB</code> Heterozygote AB probabilities for. NA if inbred lines
</p>
</li>
<li><p><code>BB</code> Homozygote BB probabilities. NA if backcross lines
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='eff_neighbor'>Estimation of self and neighbor QTL effects across a genome</h2><span id='topic+eff_neighbor'></span>

<h3>Description</h3>

<p>A function to estimate additive and dominance deviation for self and neighbor QTL effects by a simple regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eff_neighbor(
  genoprobs,
  pheno,
  smap,
  scale,
  addcovar = NULL,
  addQTL = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  fig = TRUE,
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eff_neighbor_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_pheno">pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial position along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_addqtl">addQTL</code></td>
<td>
<p>An optional vector containing marker names that are considered covariates. Namely, this option allows composite interval mapping (Jansen 1993).</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_grouping">grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_response">response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> calls the <code>"gaussian"</code> or <code>"binomial"</code> family in <code>glm()</code>, respectively.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_fig">fig</code></td>
<td>
<p>TRUE/FALSE to plot the effects or not.</p>
</td></tr>
<tr><td><code id="eff_neighbor_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Similar to Haley-Knott regression (Haley &amp; Knott 1992), the additive and dominance deviations are approximated by a regression of trait values on conditional genotype probabilities.
The self QTL effects <code>a1</code> and <code>d1</code> are estimated in the same way as the <code>qtl</code> package performs the Haley-Knott regression.
If <code>contrasts = c(TRUE, TRUE, TRUE)</code>, neighbor QTL effects <code>a1</code> and <code>d1</code> are estimated using a quadratic regression; otherwise, the additive neighbor effects are estimated using a linear regression.
See also Sato, Takeda &amp; Nagano (2021) for the rationale behind the approximation.
</p>


<h3>Value</h3>

<p>A matrix of estimated additive and dominance deviation for self and neighbor effects, with the chromosome numbers and positions. The row names correspond to marker names.
</p>

<ul>
<li><p><code>chr</code> Chromosome number
</p>
</li>
<li><p><code>pos</code> Marker position
</p>
</li>
<li><p><code>a1</code> Additive deviation for self effects
</p>
</li>
<li><p><code>d1</code> Dominance deviation for self effects
</p>
</li>
<li><p><code>a2</code> Additive deviation for neighbor effects
</p>
</li>
<li><p><code>d2</code> Dominance deviation for neighbor effects
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Haley CS, Knott SA (1992) A simple regression method for mapping quantitative trait loci in line crosses using flanking markers. Heredity 69:315-324.
</p>
</li>
<li><p> Jansen RC (1993) Interval mapping of multiple quantitative trait loci. Genetics 135:205-211.
</p>
</li>
<li><p> Sato Y, Takeda K, Nagano AJ (2021) Neighbor QTL: an interval mapping method for quantitative trait loci underlying plant neighborhood effects. G3; Genes|Genomes|Genetics 11:jkab017.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

test_eff &lt;- eff_neighbor(genoprobs=test_genoprobs,
                         pheno=test_cross$pheno$phenotype,
                         smap=test_smap, scale=20, fig=TRUE
                         )
</code></pre>

<hr>
<h2 id='genoprobs2selfprobs'>Calculating a set of self QTL effects from conditional genotype probabilities</h2><span id='topic+genoprobs2selfprobs'></span>

<h3>Description</h3>

<p>A function to reshape <code>qtl</code>'s object of conditional genotype probabilities, and to calculate self QTL effects for all individuals with given deviation coefficients and conditional genotype probabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genoprobs2selfprobs(genoprobs, a1, d1, contrasts = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genoprobs2selfprobs_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="genoprobs2selfprobs_+3A_a1">a1</code></td>
<td>
<p>A numeric scalar indicating additive deviation.</p>
</td></tr>
<tr><td><code id="genoprobs2selfprobs_+3A_d1">d1</code></td>
<td>
<p>A numeric scalar indicating dominance deviation.</p>
</td></tr>
<tr><td><code id="genoprobs2selfprobs_+3A_contrasts">contrasts</code></td>
<td>
<p>A vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix containing individuals x marker elements for self QTL effects.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='get_markers'>Reshaping marker information</h2><span id='topic+get_markers'></span>

<h3>Description</h3>

<p>A function to get marker information from a genetic map including observed and pseudo markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_markers(genoprobs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_markers_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix showing the chromosome numbers (the first column) and positions (the second column) for all markers (row names).
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='int_neighbor'>Testing marker-by-marker epistasis in neighbor QTL effects</h2><span id='topic+int_neighbor'></span>

<h3>Description</h3>

<p>A function to test interaction terms between one focal marker and the other markers across a genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>int_neighbor(
  genoprobs,
  pheno,
  smap,
  scale,
  addcovar = NULL,
  addQTL,
  intQTL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="int_neighbor_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_pheno">pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_addqtl">addQTL</code></td>
<td>
<p>A vector containing marker names that are considered covariates. This argument is necessary for <code>int_neighbor()</code>, and must match the marker names of <code>gmap</code>.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_intqtl">intQTL</code></td>
<td>
<p>A name of a focal marker to be tested for its epistasis with the other markers in neighbor effects. The marker name must be included by <code>addQTL</code>.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_grouping">grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_response">response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> calls the <code>"gaussian"</code> or <code>"binomial"</code> family in <code>glm()</code>, respectively.</p>
</td></tr>
<tr><td><code id="int_neighbor_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is an optimal function to test two-way interactions between the main neighbor effect of a focal marker given by <code>intQTL</code> and the others.
All the main neighbor effects are first estimated using <code>eff_neighbor()</code>, and then a two-way interaction term between the focal marker effect and its counterpart was considered an additional explanatory variable.
LOD score was compared between models with or without the two-way interaction.
</p>


<h3>Value</h3>

<p>A matrix of LOD scores for neighbor epistasis effects, with the chromosome numbers and positions. The row names correspond to marker names.
</p>

<ul>
<li><p><code>chr</code> Chromosome number
</p>
</li>
<li><p><code>pos</code> Marker position
</p>
</li>
<li><p><code>LOD_int</code> LOD score for epistasis in neighbor effects between a focal and the other markers
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+scan_neighbor">scan_neighbor</a></code> <code><a href="#topic+eff_neighbor">eff_neighbor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

test_int &lt;- int_neighbor(genoprobs=test_genoprobs,
                         pheno=test_cross$pheno$phenotype,
                         smap=test_smap,scale=20,
                         addQTL=c("c1_D1M1","c1_D1M2"),intQTL="c1_D1M1"
                         )
plot_nei(test_int, type="int")
</code></pre>

<hr>
<h2 id='logLik_glm.fit'>Calculating log-likelihood in generalized linear models</h2><span id='topic+logLik_glm.fit'></span>

<h3>Description</h3>

<p>An utility function to extract log-likelihood based on AIC of glm.fit()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logLik_glm.fit(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logLik_glm.fit_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to <code>glm.fit()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Log-likelihood
</p>

<hr>
<h2 id='min_dist'>Calculating the minimum distance</h2><span id='topic+min_dist'></span>

<h3>Description</h3>

<p>A function to calculate a Euclidian distance including at least one neighbor for all individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>min_dist(smap, grouping = rep(1, nrow(smap)))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="min_dist_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map. The first and second column include spatial points along a x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="min_dist_+3A_grouping">grouping</code></td>
<td>
<p>A integer vector assigning each individual to a group. This argument can be useful when a &quot;smap&quot; contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a scalar of the minimum Euclidian distance that allows all individuals to have at least one neighbor.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='neiprob'>Calculating neighbor QTL effects</h2><span id='topic+neiprob'></span>

<h3>Description</h3>

<p>A function to calculate neighbor QTL effects between two individuals, with given deviation coefficients and conditional genotype probabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neiprob(i, j, a2, d2, AA, AB, BB, d2sq0 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neiprob_+3A_i">i</code></td>
<td>
<p>ID of a target individual.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_j">j</code></td>
<td>
<p>ID of an interacting neighbor.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_a2">a2</code></td>
<td>
<p>A numeric scalar indicating additive deviation.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_d2">d2</code></td>
<td>
<p>A numeric scalar indicating dominance deviation.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_aa">AA</code></td>
<td>
<p>An individual x marker matrix of conditional probabilities for AA genotype.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_ab">AB</code></td>
<td>
<p>An individual x marker matrix of conditional probabilities for AB genotype. Input NA if heterozygotes are absent.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_bb">BB</code></td>
<td>
<p>An individual x marker matrix of conditional probabilities for BB genotype. Input NA for backcross lines.</p>
</td></tr>
<tr><td><code id="neiprob_+3A_d2sq0">d2sq0</code></td>
<td>
<p>An option to make AB/AB interaction effects zero.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing each marker effect for individual i.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='perm_neighbor'>Permutation tests for neighbor effects with a QTL model</h2><span id='topic+perm_neighbor'></span>

<h3>Description</h3>

<p>A function to calculate a genome-wide LOD threshold using permutation tests for self or neighbor effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perm_neighbor(
  genoprobs,
  pheno,
  smap,
  scale,
  addcovar = NULL,
  addQTL = NULL,
  intQTL = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  type = c("neighbor", "self", "int"),
  times = 99,
  p_val = 0.05,
  n_core = 1L,
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perm_neighbor_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_pheno">pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_addqtl">addQTL</code></td>
<td>
<p>An optional vector containing marker names that are considered covariates. Namely, this option allows composite interval mapping (Jansen 1993).</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_intqtl">intQTL</code></td>
<td>
<p>An option when using <code>int_neighbor()</code>. A name of a focal marker to be tested for its epistasis with the other markers in neighbor effects. The marker name must be included by <code>addQTL</code>.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_grouping">grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_response">response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> calls the <code>"gaussian"</code> or <code>"binomial"</code> family in <code>glm()</code>, respectively.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_type">type</code></td>
<td>
<p>Select <code>"self"</code>, <code>"neighbor"</code>, or <code>"int"</code> to perform permutation tests for self effects, neighbor effects, or neighbor epistasis, respectively.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_times">times</code></td>
<td>
<p>No. of permutation iterations. Default at 99 times</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_p_val">p_val</code></td>
<td>
<p>A vector indicating upper quantiles for permutation LOD scores</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_n_core">n_core</code></td>
<td>
<p>No. of cores for a parallel computation. This does not work for Windows OS. Default is a single-core computation.</p>
</td></tr>
<tr><td><code id="perm_neighbor_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>LOD thresholds at given quantiles by <code>p-val</code>
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_nei">plot_nei</a></code> <code><a href="#topic+scan_neighbor">scan_neighbor</a></code> <code><a href="#topic+int_neighbor">int_neighbor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

test_perm &lt;- perm_neighbor(genoprobs=test_genoprobs,
                           pheno=test_cross$pheno$phenotype,
                           smap=test_smap,scale=20,
                           times=3, p_val=c(1.0,0.5)
                           )
</code></pre>

<hr>
<h2 id='plot_eff'>Plot self and neighbor QTL effects across a genome</h2><span id='topic+plot_eff'></span>

<h3>Description</h3>

<p>Plot estimated additive and dominance deviation for self or neighbor effects across a genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_eff(res, type = c("neighbor", "self"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_eff_+3A_res">res</code></td>
<td>
<p>Output results of <code>eff_neighbor()</code>.</p>
</td></tr>
<tr><td><code id="plot_eff_+3A_type">type</code></td>
<td>
<p>An option to select <code>"self"</code> or <code>"neighbor"</code> effects to be shown. Default is <code>"neighbor"</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p>eff_neighbor
</p>

<hr>
<h2 id='plot_nei'>Plot LOD score for self or neighbor QTL effects</h2><span id='topic+plot_nei'></span>

<h3>Description</h3>

<p>Plot LOD curves for a genome scan of self and neighbor QTL effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_nei(res, type = c("neighbor", "self", "int"), chr = NULL, th = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_nei_+3A_res">res</code></td>
<td>
<p>Output results of <code>scan_neighbor()</code>.</p>
</td></tr>
<tr><td><code id="plot_nei_+3A_type">type</code></td>
<td>
<p>Plot <code>"self"</code>, <code>"neighbor"</code> or <code>"int"</code> effects. Default is <code>"neighbor"</code> effects.</p>
</td></tr>
<tr><td><code id="plot_nei_+3A_chr">chr</code></td>
<td>
<p>An optional vector to select chromosome numbers to be plotted. If <code>NULL</code>, shown are all chromosomes.</p>
</td></tr>
<tr><td><code id="plot_nei_+3A_th">th</code></td>
<td>
<p>Add genome-wide threshold by user-defined vectors or Bonferroni correction. Default is no thresholds added.</p>
</td></tr>
<tr><td><code id="plot_nei_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to <code>plot()</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the <code>type</code> argument, <code>"int"</code> can be selected to draw the results of <code>int_neighbor()</code>.
In this case, the <code>res</code> object and <code>type</code> must match, otherwise it returns an error message.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+scan_neighbor">scan_neighbor</a></code> <code><a href="#topic+int_neighbor">int_neighbor</a></code> <code><a href="#topic+perm_neighbor">perm_neighbor</a></code>
</p>

<hr>
<h2 id='scan_neighbor'>Genome scan for neighbor effects with a QTL model</h2><span id='topic+scan_neighbor'></span>

<h3>Description</h3>

<p>Genome scan using a QTL model for self and neighbor effects, with possible allowance for additional covariates and non-normal traits. Theoretical background is described in Sato, Takeda &amp; Nagano (2021).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scan_neighbor(
  genoprobs,
  pheno,
  smap,
  scale,
  addcovar = NULL,
  addQTL = NULL,
  grouping = rep(1, nrow(smap)),
  response = c("quantitative", "binary"),
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scan_neighbor_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_pheno">pheno</code></td>
<td>
<p>A vector of individual phenotypes.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_addcovar">addcovar</code></td>
<td>
<p>An optional matrix including additional non-genetic covariates. It contains no. of individuals x no. of covariates.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_addqtl">addQTL</code></td>
<td>
<p>An optional vector containing marker names that are considered covariates. Namely, this option allows composite interval mapping (Jansen 1993).</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_grouping">grouping</code></td>
<td>
<p>An optional integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_response">response</code></td>
<td>
<p>An optional argument to select trait types. The <code>"quantitative"</code> or <code>"binary"</code> calls the <code>"gaussian"</code> or <code>"binomial"</code> family in <code>glm()</code>, respectively.</p>
</td></tr>
<tr><td><code id="scan_neighbor_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates LOD score after the additive and dominance deviation are estimated using <code>eff_neighbor()</code>.
As it adopts a stepwise testing from self to neighbor effects, <code>LOD_self</code> are the same as standard QTL mapping.
Note that the results return 0 LOD scores for covariate markers when using <code>addQTL</code> option.
</p>


<h3>Value</h3>

<p>A matrix of LOD scores for self and neighbor effects, with the chromosome numbers and positions. The row names correspond to marker names.
</p>

<ul>
<li><p><code>chr</code> Chromosome number
</p>
</li>
<li><p><code>pos</code> Marker position
</p>
</li>
<li><p><code>LOD_self</code> LOD score for self effects
</p>
</li>
<li><p><code>LOD_nei</code> LOD score for neighbor effects
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Jansen RC (1993) Interval mapping of multiple quantitative trait loci. Genetics 135:205-211.
</p>
</li>
<li><p> Sato Y, Takeda K, Nagano AJ (2021) Neighbor QTL: an interval mapping method for quantitative trait loci underlying plant neighborhood effects. G3; Genes|Genomes|Genetics 11:jkab017.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+eff_neighbor">eff_neighbor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

test_scan &lt;- scan_neighbor(genoprobs=test_genoprobs,
                           pheno=test_cross$pheno$phenotype,
                           smap=test_smap, scale=20
                           )
plot_nei(test_scan)
</code></pre>

<hr>
<h2 id='selfprob'>Calculating self QTL effects</h2><span id='topic+selfprob'></span>

<h3>Description</h3>

<p>A function to calculate self QTL effects for an individual, with given deviation coefficients and conditional genotype probabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selfprob(i, a1, d1, AA, AB, BB)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selfprob_+3A_i">i</code></td>
<td>
<p>ID of a target individual.</p>
</td></tr>
<tr><td><code id="selfprob_+3A_a1">a1</code></td>
<td>
<p>A numeric scalar indicating additive deviation.</p>
</td></tr>
<tr><td><code id="selfprob_+3A_d1">d1</code></td>
<td>
<p>A numeric scalar indicating dominance deviation.</p>
</td></tr>
<tr><td><code id="selfprob_+3A_aa">AA</code></td>
<td>
<p>An individual x marker matrix of conditional probabilities for AA genotype.</p>
</td></tr>
<tr><td><code id="selfprob_+3A_ab">AB</code></td>
<td>
<p>An individual x marker matrix of conditional probabilities for AB genotype. Input NA if heterozygotes are absent.</p>
</td></tr>
<tr><td><code id="selfprob_+3A_bb">BB</code></td>
<td>
<p>An individual x marker matrix of conditional probabilities for BB genotype. Input NA for backcross lines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing each marker effect for individual i.
</p>


<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>

<hr>
<h2 id='sim_nei_qtl'>Phenotype simulation for neighbor QTL effects</h2><span id='topic+sim_nei_qtl'></span>

<h3>Description</h3>

<p>A function to simulate neighbor effects with given QTL effects, distance scale, and causal markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_nei_qtl(
  genoprobs,
  a2,
  d2,
  smap,
  scale,
  grouping = rep(1, nrow(smap)),
  n_QTL = 1,
  contrasts = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_nei_qtl_+3A_genoprobs">genoprobs</code></td>
<td>
<p>Conditional genotype probabilities as taken from <code>qtl::calc.genoprob()</code>.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_a2">a2</code></td>
<td>
<p>A numeric scalar indicating additive deviation.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_d2">d2</code></td>
<td>
<p>A numeric scalar indicating dominance deviation.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_smap">smap</code></td>
<td>
<p>A matrix showing a spatial map for individuals. The first and second column include spatial positions along an x-axis and y-axis, respectively.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_scale">scale</code></td>
<td>
<p>A numeric scalar indicating the maximum spatial distance between a focal individual and neighbors to define neighbor effects.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_grouping">grouping</code></td>
<td>
<p>An integer vector assigning each individual to a group. This argument can be used when <code>smap</code> contains different experimental replicates. Default setting means that all individuals are belong to a single group.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_n_qtl">n_QTL</code></td>
<td>
<p>A positive integer indicating the number of causal markers.</p>
</td></tr>
<tr><td><code id="sim_nei_qtl_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional vector composed of three TRUE/FALSE values, which represents the presence/absence of specific genotypes as c(TRUE/FALSE, TRUE/FALSE, TRUE/FALSE) = AA, AB, BB. If <code>NULL</code>, it is compiled from <code>genoprobs</code> automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Major genetic effects, <code>a2</code> and <code>d2</code>, are allocated to causal loci randomly selected by <code>n_QTL</code>, while minor polygenic effects (i.e., 1% of <code>a2</code>) are allocated to the other loci.
</p>


<h3>Value</h3>

<p>A numeric matrix containing individuals x marker elements for neighbor QTL effects.
</p>

<ul>
<li><p><code>true_scale</code> True distance scale of simulated neighbor effects
</p>
</li>
<li><p><code>true_marker</code> The name(s) of causal markers
</p>
</li>
<li><p><code>nei_y</code> Simulated neighbor effects standardized to have zero mean and one variance
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yasuhiro Sato (<a href="mailto:sato.yasuhiro.36c@kyoto-u.jp">sato.yasuhiro.36c@kyoto-u.jp</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
test_map &lt;- qtl::sim.map(len=rep(20,5),n.mar=3,include.x=FALSE)
test_cross &lt;- qtl::sim.cross(test_map,n.ind=50)
test_smap &lt;- cbind(runif(50,1,100),runif(50,1,100))
test_genoprobs &lt;- qtl::calc.genoprob(test_cross,step=2)

nei_eff &lt;- sim_nei_qtl(genoprobs=test_genoprobs, a2=0.5, d2=0.5,
                       smap=test_smap,
                       scale=20, n_QTL=1)

test_scan &lt;- scan_neighbor(genoprobs=test_genoprobs,
                           pheno=nei_eff$nei_y,
                           smap=test_smap, scale=20
                           )
plot_nei(test_scan)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
