<!DOCTYPE html><html><head><title>Help for package RNentropy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RNentropy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RN_BarresLab_design'>

<p>RN_BarresLab_design</p></a></li>
<li><a href='#RN_BarresLab_FPKM'>

<p>RN_BarresLab_FPKM</p></a></li>
<li><a href='#RN_Brain_Example_design'>

<p>RN_Brain_Example_design</p></a></li>
<li><a href='#RN_Brain_Example_tpm'>

<p>RN_Brain_Example_tpm</p></a></li>
<li><a href='#RN_calc'>

<p>RN_calc</p></a></li>
<li><a href='#RN_calc_GPV'>

<p>RN_calc_GPV</p></a></li>
<li><a href='#RN_calc_LPV'>

<p>RN_calc_LPV</p></a></li>
<li><a href='#RN_pmi'>

<p>Compute point mutual information matrix for the experimental conditions.</p></a></li>
<li><a href='#RN_select'>

<p>Select transcripts/genes with significant p-values.</p></a></li>
<li><a href='#RNentropy'>
<p>RNentropy</p></a></li>
<li><a href='#RNentropy-package'>
<p>Entropy Based Method for the Detection of Significant Variation in Gene Expression Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Entropy Based Method for the Detection of Significant Variation
in Gene Expression Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-05</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Federico Zambelli &lt;federico.zambelli@unimi.it&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of a method based on information theory devised 
    for the identification of genes showing a significant variation of expression
    across multiple conditions. Given expression estimates from any number of 
    RNA-Seq samples and conditions it identifies genes or transcripts with a 
    significant variation of expression across all the conditions studied, 
    together with the samples in which they are over- or under-expressed. 
    Zambelli et al. (2018) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgky055">doi:10.1093/nar/gky055</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-13 08:21:36 UTC; federicozambelli</td>
</tr>
<tr>
<td>Author:</td>
<td>Federico Zambelli <a href="https://orcid.org/0000-0003-3487-4331"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre],
  Giulio Pavesi <a href="https://orcid.org/0000-0001-5705-6249"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-13 09:52:35 UTC</td>
</tr>
</table>
<hr>
<h2 id='RN_BarresLab_design'>

RN_BarresLab_design
</h2><span id='topic+RN_BarresLab_design'></span>

<h3>Description</h3>


<p>The experiment design matrix for RN_BarresLab_design
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("RN_Brain_Example_design")</code></pre>


<h3>Format</h3>

<p>The format is: num[1:9,1:3]
</p>


<h3>Details</h3>


<p>A binary matrix where conditions correspond to columns and samples to rows. In this example there are seven conditions (cell types) and seven samples with the expression measured by mean FPKM.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RN_Brain_Example_design)
</code></pre>

<hr>
<h2 id='RN_BarresLab_FPKM'>

RN_BarresLab_FPKM
</h2><span id='topic+RN_BarresLab_FPKM'></span>

<h3>Description</h3>


<p>Data expression values (mean FPKM) for different cerebral cortex cell from Zhang et al: &quot;An RNA-Seq transcriptome and splicing database of glia, neurons, and vascular cells of the cerebral cortex&quot;, J Neurosci 2014 Sep. 
RN_BarresLab_design is the corresponding design matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("RN_BarresLab_FPKM")</code></pre>


<h3>Format</h3>

<p>A data frame with 12978 observations on the following 7 variables.
</p>

<dl>
<dt><code>Astrocyte</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Neuron</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>OPC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NFO </code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MO</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Microglia</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Endothelial</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>


<p>Zhang Y, Chen K, Sloan SA, Bennett ML, Scholze AR, O'Keeffe S, Phatnani HP,
Guarnieri P, Caneda C, Ruderisch N, Deng S, Liddelow SA, Zhang C, Daneman R,
Maniatis T, Barres BA, Wu JQ. An RNA-sequencing transcriptome and splicing
database of glia, neurons, and vascular cells of the cerebral cortex. J Neurosci. 2014 Sep 3;34(36):11929-47. doi: 10.1523/JNEUROSCI.1860-14.2014. Erratum in: J
Neurosci. 2015 Jan 14;35(2):846-6. PubMed PMID: 25186741; PubMed Central PMCID:
PMC4152602.
</p>
<p>Data were downloaded from https://web.stanford.edu/group/barres_lab/brain_rnaseq.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RN_BarresLab_FPKM)
</code></pre>

<hr>
<h2 id='RN_Brain_Example_design'>

RN_Brain_Example_design
</h2><span id='topic+RN_Brain_Example_design'></span>

<h3>Description</h3>


<p>The example design matrix for RN_Brain_Example_tpm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("RN_Brain_Example_design")</code></pre>


<h3>Format</h3>

<p>The format is: num[1:9,1:3]
</p>


<h3>Details</h3>


<p>A binary matrix where conditions correspond to the columns and samples to the rows. In this example there are three conditions (individuals) and nine samples, with three replicates for each condition.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RN_Brain_Example_design)
</code></pre>

<hr>
<h2 id='RN_Brain_Example_tpm'>

RN_Brain_Example_tpm
</h2><span id='topic+RN_Brain_Example_tpm'></span>

<h3>Description</h3>


<p>Example data expression values from brain of three different individuals, each with three replicates. 
RN_Brain_Example_design is the corresponding design matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("RN_Brain_Example_tpm")</code></pre>


<h3>Format</h3>

<p>A data frame with 78699 observations on the following 9 variables.
</p>

<dl>
<dt><code>GENE</code></dt><dd><p>a factor vector</p>
</dd>
<dt><code>BRAIN_2_1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_2_2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_2_3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_3_1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_3_2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_3_3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_1_1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_1_2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRAIN_1_3</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(RN_Brain_Example_tpm)
</code></pre>

<hr>
<h2 id='RN_calc'>

RN_calc
</h2><span id='topic+RN_calc'></span>

<h3>Description</h3>


<p>Computes both global and local p-values, and returns the results in a list containing for each gene the original expression values and the associated global and local p-values (as  -log10(p-value)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RN_calc(X, design)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RN_calc_+3A_x">X</code></td>
<td>


<p>data.frame with expression values. It may contain additional non numeric columns (eg. a column with gene names).
</p>
</td></tr>
<tr><td><code id="RN_calc_+3A_design">design</code></td>
<td>


<p>The RxC design matrix where R (rows) corresponds to the number of numeric columns (samples) in 'file' and C (columns) to the number of conditions. It must be a binary matrix with one and only one '1' for every row, corresponding to the condition (column) for which the sample corresponding to the row has to be considered a biological ot technical replicate. See the example 'RN_Brain_Example_design' for the design matrix of 'RN_Brain_Example_tpm' which has three replicates for three conditions (three rows) for a total of nine samples (nine rows). design defaults to a square matrix of independent samples (diagonal = 1, everything else = 0)
</p>
</td></tr>
</table>


<h3>Value</h3>






<table>
<tr><td><code>gpv</code></td>
<td>
<p>-log10 of the global p-values</p>
</td></tr>
<tr><td><code>lpv</code></td>
<td>
<p>-log10 of the local p-values</p>
</td></tr>
<tr><td><code>c_like</code></td>
<td>
<p>results formatted as in the output of the C++ implementation of RNentropy.</p>
</td></tr>
<tr><td><code>res</code></td>
<td>
<p>The results data.frame with the original expression values and the associated -log10 of global and local p-values.</p>
</td></tr>
<tr><td><code>design</code></td>
<td>
<p>the experimental design matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>


<p>Giulio Pavesi - Dep. of Biosciences, University of Milan
</p>
<p>Federico Zambelli - Dep. of Biosciences, University of Milan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN)
Results &lt;- RN_calc(RN_Brain_Example_tpm[1:10000,], RN_Brain_Example_design)

## The function is currently defined as

function(X, design = NULL)
{
  if(is.null(design))
  {
    design &lt;- .RN_default_design(sum(sapply(X, is.numeric)))
  }
  
	Results &lt;- list(expr = X, design = design)

	GPV &lt;- RN_calc_GPV(X, bind = FALSE)
	LPV &lt;- RN_calc_LPV(X, design = design, bind = FALSE)	

	TABLE = cbind(X,'---',GPV,'---',LPV)	

	Results$gpv &lt;- GPV
	Results$lpv &lt;- LPV
	Results$c_like &lt;- TABLE
	Results$res &lt;- cbind(X, GPV, LPV)
	
	return(Results)
}
</code></pre>

<hr>
<h2 id='RN_calc_GPV'>

RN_calc_GPV 
</h2><span id='topic+RN_calc_GPV'></span>

<h3>Description</h3>


<p>This function calculates global p-values from expression data, represented as -log10(p-values). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RN_calc_GPV(X, bind = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RN_calc_GPV_+3A_x">X</code></td>
<td>


<p>data.frame with expression values. It may contain additional non numeric columns (eg. a column with gene names)
</p>
</td></tr>
<tr><td><code id="RN_calc_GPV_+3A_bind">bind</code></td>
<td>


<p>See Value (Default: TRUE)
</p>
</td></tr>
</table>


<h3>Value</h3>






<p>If bind is TRUE the function returns a data.frame with the original expression values from 'X' and an attached column with the -log10() of the global p-value, otherwise only the numeric vector of -log10(p-values) is returned.
</p>


<h3>Author(s)</h3>


<p>Giulio Pavesi - Dep. of Biosciences, University of Milan
</p>
<p>Federico Zambelli - Dep. of Biosciences, University of Milan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("RN_Brain_Example_tpm")
GPV &lt;- RN_calc_GPV(RN_Brain_Example_tpm)

## The function is currently defined as
function (X, bind = TRUE) 
{
    rnums &lt;- sapply(X, is.numeric)
    GL_LPV &lt;- apply(X[rnums], 1, ".RN_calc_GPV_row")
    if (bind) {
        GPV &lt;- cbind(X, GL_LPV)
        return(GPV)
    }
    else {
        return(GL_LPV)
    }
  }
</code></pre>

<hr>
<h2 id='RN_calc_LPV'>

RN_calc_LPV
</h2><span id='topic+RN_calc_LPV'></span>

<h3>Description</h3>


<p>This function calculates local p-values from expression data, output as -log10(p-value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RN_calc_LPV(X, design, bind = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RN_calc_LPV_+3A_x">X</code></td>
<td>


<p>data.frame with expression values. It may contain additional non numeric columns (eg. a column with gene names)
</p>
</td></tr>
<tr><td><code id="RN_calc_LPV_+3A_design">design</code></td>
<td>


<p>The design matrix. Refer to the help for RN_calc for further info.
</p>
</td></tr>
<tr><td><code id="RN_calc_LPV_+3A_bind">bind</code></td>
<td>


<p>See Value (Default = TRUE)
</p>
</td></tr>
</table>


<h3>Value</h3>






<p>If bind is TRUE the function returns a data.frame with the original expression values from data.frame X and attached columns with the -log10(p-values) computed for each sample, otherwise only the matrix of -log10(p-values) is returned.
</p>


<h3>Author(s)</h3>


<p>Giulio Pavesi - Dep. of Biosciences, University of Milan
</p>
<p>Federico Zambelli - Dep. of Biosciences, University of Milan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
LPV &lt;- RN_calc_LPV(RN_Brain_Example_tpm, RN_Brain_Example_design)
## The function is currently defined as

function(X, design = NULL, bind = TRUE)
{
  if(is.null(design))
  {
    design &lt;- .RN_default_design(sum(sapply(X, is.numeric)))
  }
  
	rnums &lt;- sapply(X, is.numeric)

	.RN_design_check(X, design, rnums)
	RL &lt;- .RN_get_replicate_list(design)
	
	l1 &lt;- rep("LOC_LPV", length(X[rnums]))
	l2 &lt;- colnames(X[rnums])
	PV &lt;- apply(X[rnums],  1, '.RN_calc_LPV_row', RL = RL)
	PV &lt;- t(PV)
	colnames(PV) &lt;- paste(l1, l2, sep="_")

	if(bind)
	{
		return (cbind(X, PV))
	}
	else
		return (PV)
}
</code></pre>

<hr>
<h2 id='RN_pmi'>

Compute point mutual information matrix for the experimental conditions. 
</h2><span id='topic+RN_pmi'></span>

<h3>Description</h3>


<p>Compute point mutual information for experimental conditions from the overxexpressed genes identified by RN_select.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RN_pmi(Results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RN_pmi_+3A_results">Results</code></td>
<td>


<p>The output of RNentropy, RN_calc or RN_select. If RN_select has not already run on the results it will be invoked by RN_pmi using default arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The original input containing
</p>
<table>
<tr><td><code>gpv</code></td>
<td>
<p>-log10 of the Global p-values</p>
</td></tr>
<tr><td><code>lpv</code></td>
<td>
<p>-log10 of the Local p-values</p>
</td></tr>
<tr><td><code>c_like</code></td>
<td>
<p>a table similar to the one you obtain running the C++ implementation of RNentropy.</p>
</td></tr>
<tr><td><code>res</code></td>
<td>
<p>The results data.frame containing the original expression values together with the -log10 of Global and Local p-values.</p>
</td></tr>
<tr><td><code>design</code></td>
<td>
<p>The experimental design matrix.</p>
</td></tr>
<tr><td><code>selected</code></td>
<td>
<p>Transcripts/genes with a corrected Global p-value lower than gpv_t. Each condition N gets a condition_N column which values can be -1,0,1 or NA. 1 means that all the replicates of this condition seems to be consistently over-expressed w.r.t the overall expression of the transcript in all the conditions (that is, all the replicates of condition N have a positive Local p-value &lt;= lpv_t).  -1 means that all the replicates of this condition seems to be consistently under-expressed w.r.t the overall expression of the transcript in all the conditions (that is, all the replicates of condition N have a negative Local p-value and abs(Local p-values) &lt;= lpv_t). 0 means that one or more replicates have an abs(Local p-value) &gt; lpv_t. NA means that the Local p-values of the replicates are not consistent for this condition.</p>
</td></tr>
</table>
<p>And two new matrices:
</p>
<table>
<tr><td><code>pmi</code></td>
<td>
<p>Point mutual information matrix of the conditions.</p>
</td></tr>
<tr><td><code>npmi</code></td>
<td>
<p>Normalized point mutual information matrix of the conditions.</p>
</td></tr>





</table>


<h3>Author(s)</h3>


<p>Giulio Pavesi - Dep. of Biosciences, University of Milan
</p>
<p>Federico Zambelli - Dep. of Biosciences, University of Milan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN)
Results &lt;- RN_calc(RN_Brain_Example_tpm[1:10000,], RN_Brain_Example_design)
Results &lt;- RN_select(Results)
Results &lt;- RN_pmi(Results)

## The function is currently defined as
RN_pmi &lt;- function(Results)
{
  if(is.null(Results$selected))
    Results &lt;- RN_select(Results)
  
  Results$pmi &lt;- matrix(nrow = ncol(Results$design), 
                        ncol = ncol(Results$design))
  
  colnames(Results$pmi) &lt;- colnames(Results$design)
  rownames(Results$pmi) &lt;- colnames(Results$design)
  
  Results$npmi &lt;- Results$pmi
  
  colshift &lt;- ncol(Results$selected) - ncol(Results$design) 
  
  for(x in 1:nrow(Results$pmi))
  {
    for(y in 1:nrow(Results$pmi))
    {
      if(x &gt; y)
      {
        Results$pmi[x,y] &lt;- Results$pmi[y,x]
        Results$npmi[x,y] &lt;- Results$npmi[y,x]
        next
      } else
      {
        sum_x &lt;- sum(Results$selected[,x+colshift] == 1, na.rm = TRUE)
        sum_y &lt;- sum(Results$selected[,y+colshift] == 1, na.rm = TRUE)
        sum_xy &lt;- sum(Results$selected[,x+colshift] == 1 &amp; 
                        Results$selected[,y+colshift] == 1, na.rm = TRUE)
        freq_x &lt;- sum_x / nrow(Results$selected)
        freq_y &lt;- sum_y / nrow(Results$selected)
        freq_xy &lt;- sum_xy / nrow(Results$selected)
        
        h_xy &lt;- log2(1/freq_xy)
        
        Results$pmi[x,y] &lt;- log2(freq_xy / (freq_x * freq_y))
        Results$npmi[x,y] &lt;- Results$pmi[x,y] / h_xy
      }
    }
  }
  
  return (Results)
  }
</code></pre>

<hr>
<h2 id='RN_select'>

Select transcripts/genes with significant p-values.
</h2><span id='topic+RN_select'></span>

<h3>Description</h3>


<p>Select transcripts with global p-value lower than an user defined threshold and provide a summary of over- or under-expression according to local p-values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RN_select(Results, gpv_t = 0.01, lpv_t = 0.01, method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RN_select_+3A_results">Results</code></td>
<td>


<p>The output of RNentropy or RN_calc.
</p>
</td></tr>
<tr><td><code id="RN_select_+3A_gpv_t">gpv_t</code></td>
<td>


<p>Threshold for global p-value. (Default: 0.01)
</p>
</td></tr>
<tr><td><code id="RN_select_+3A_lpv_t">lpv_t</code></td>
<td>


<p>Threshold for local p-value. (Default: 0.01)
</p>
</td></tr>
<tr><td><code id="RN_select_+3A_method">method</code></td>
<td>


<p>Multiple test correction method. Available methods are the ones of p.adjust. Type p.adjust.methods to see the list. Default: BH (Benjamini &amp; Hochberg)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The original input containing
</p>
<table>
<tr><td><code>gpv</code></td>
<td>
<p>-log10 of the global p-values</p>
</td></tr>
<tr><td><code>lpv</code></td>
<td>
<p>-log10 of the local p-values</p>
</td></tr>
<tr><td><code>c_like</code></td>
<td>
<p>results formatted as in the output of the C++ implementation of RNentropy.</p>
</td></tr>
<tr><td><code>res</code></td>
<td>
<p>The results data.frame containing the original expression values together with the -log10 of global and local p-values.</p>
</td></tr>
<tr><td><code>design</code></td>
<td>
<p>The experimental design matrix.</p>
</td></tr>
</table>
<p>and a new dataframe
</p>
<table>
<tr><td><code>selected</code></td>
<td>
<p>Transcripts/genes with a corrected global p-value lower than gpv_t. For each condition it will contain a column where values can be -1,0,1 or NA. 1 means that all the replicates of this condition have expression value higher than the average and local p-value &lt;= lpv_t (thus the corresponding gene will be over-expressed in this condition).  -1 means that all the replicates of this condition  have expression value lower than the average and local p-value &lt;= lpv_t (thus the corresponding gene will be under-expressed in this condition). 0 means that at least one of the replicates has a local p-value &gt; lpv_t. NA means that the local p-values of the replicates are not consistent for this condition, that is, at least one replicate results to be over-expressed and at least one results to be under-expressed.</p>
</td></tr>





</table>


<h3>Author(s)</h3>


<p>Giulio Pavesi - Dep. of Biosciences, University of Milan
</p>
<p>Federico Zambelli - Dep. of Biosciences, University of Milan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN)
Results &lt;- RN_calc(RN_Brain_Example_tpm[1:10000,], RN_Brain_Example_design)
Results &lt;- RN_select(Results)

## The function is currently defined as
function (Results, gpv_t = 0.01, lpv_t = 0.01, method = "BH") 
{
    lpv_t &lt;- -log10(lpv_t)
    gpv_t &lt;- -log10(gpv_t)
    Results$gpv_bh &lt;- -log10(p.adjust(10^-Results$gpv, method = method))
    true_rows &lt;- (Results$gpv_bh &gt;= gpv_t)
    design_b &lt;- t(Results$design &gt; 0)
    Results$lpv_sel &lt;- data.frame(row.names = rownames(Results$lpv)[true_rows])
    for (d in seq_along(design_b[, 1])) {
        col &lt;- apply(Results$lpv[true_rows, ], 1, ".RN_select_lpv_row", 
            design_b[d, ], lpv_t)
        Results$lpv_sel &lt;- cbind(Results$lpv_sel, col)
        colnames(Results$lpv_sel)[length(Results$lpv_sel)] &lt;- paste("condition", 
            d, sep = "_")
    }
    lbl &lt;- Results$res[, !sapply(Results$res, is.numeric)]
    Results$selected &lt;- cbind(lbl[true_rows], Results$gpv[true_rows], 
	Results$gpv_bh[true_rows], Results$lpv_sel)
    colnames(Results$selected) &lt;- c(names(which(!sapply(Results$res, 
	is.numeric))), "GL_LPV", "Corr. GL_LPV", colnames(Results$lpv_sel))
    Results$selected &lt;- Results$selected[order(Results$selected[,3], decreasing=TRUE),]
    Results$lpv_sel &lt;- NULL
    return(Results)
  }
</code></pre>

<hr>
<h2 id='RNentropy'>
RNentropy 
</h2><span id='topic+RNentropy'></span>

<h3>Description</h3>

<p>This function runs RNentropy on a file containing normalized expression values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNentropy(file, tr.col, design, header = TRUE, skip = 0, skip.col = NULL, 
col.names = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RNentropy_+3A_file">file</code></td>
<td>


<p>Expression data file. It must contain a column with univocal identifiers for each row (typically transcript or gene ids).
</p>
</td></tr>
<tr><td><code id="RNentropy_+3A_tr.col">tr.col</code></td>
<td>


<p>The column # (1 based) in the file containing the univocal identifiers (usually transcript or gene ids).
</p>
</td></tr>
<tr><td><code id="RNentropy_+3A_design">design</code></td>
<td>


<p>The RxC design matrix where R (rows) corresponds to the number of numeric columns (samples) in 'file' and C (columns) to the number of conditions. It must be a binary matrix with one and only one '1' for every row, corresponding to the condition (column) for which the sample corresponding to the row has to be considered a biological ot technical replicate. See the example 'RN_Brain_Example_design' for the design matrix of 'RN_Brain_Example_tpm' which has three replicates for three conditions (three rows) for a total of nine samples (nine rows).
</p>
</td></tr>
<tr><td><code id="RNentropy_+3A_header">header</code></td>
<td>


<p>Set this to FALSE if 'file' does not have an header row. If header is TRUE the read.table function used by RNentropy will try to assign names to each column using the header. Default = TRUE
</p>
</td></tr>
<tr><td><code id="RNentropy_+3A_skip">skip</code></td>
<td>


<p>Number of rows to skip at the beginning of 'file'. Rows beginning with a '#' token will be skipped even if skip is set to 0. 
</p>
</td></tr>
<tr><td><code id="RNentropy_+3A_skip.col">skip.col</code></td>
<td>


<p>Columns that will not be imported from 'file' (1 based) and not included in the subsequent analysis. Useful if you have more than annotation column, e.g. gene name in the first, transcript ID in the second columns.
</p>
</td></tr>
<tr><td><code id="RNentropy_+3A_col.names">col.names</code></td>
<td>


<p>Assign names to the imported columns. The number of names must correspond to the columns effectively imported (thus, no name for the skipped ones). Also, tr.col is not considered an imported column. Useful to assign sample names to expression columns.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Refer to the help for RN_calc.





</p>


<h3>Author(s)</h3>


<p>Giulio Pavesi - Dep. of Biosciences, University of Milan
</p>
<p>Federico Zambelli - Dep. of Biosciences, University of Milan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load expression values and experiment design
data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN) 
Results &lt;- RN_calc(RN_Brain_Example_tpm[1:10000,], RN_Brain_Example_design)
#select only genes with significant changes of expression
Results &lt;- RN_select(Results)


## The function is currently defined as
function (file, tr.col, design, header = TRUE, skip = 0, skip.col = NULL, 
    col.names = NULL) 
{
    TABLE &lt;- read.table(file, row.names = tr.col, header = header, 
        skip = skip, blank.lines.skip = TRUE, comment.char = "#")
    if (!is.null(skip.col)) {
        TABLE &lt;- .RN_delete_col(TABLE, tr.col, skip.col)
    }
    if (!is.null(col.names)) {
        colnames(TABLE) &lt;- col.names
    }
    return(RN_calc(TABLE, design))
  }
</code></pre>

<hr>
<h2 id='RNentropy-package'>
Entropy Based Method for the Detection of Significant Variation in Gene Expression Data
</h2><span id='topic+RNentropy-package'></span>

<h3>Description</h3>

<p>An implementation of a method based on information theory devised 
    for the identification of genes showing a significant variation of expression
    across multiple conditions. Given expression estimates from any number of 
    RNA-Seq samples and conditions it identifies genes or transcripts with a 
    significant variation of expression across all the conditions studied, 
    together with the samples in which they are over- or under-expressed. 
    Zambelli et al. (2018) &lt;doi:10.1093/nar/gky055&gt;.
</p>


<h3>Author(s)</h3>

<p>NA
</p>
<p>Maintainer: Federico Zambelli &lt;federico.zambelli@unimi.it&gt;
</p>


<h3>References</h3>

<p>doi = 10.1093/nar/gky055
doi = 10.1007/978-1-0716-1307-8_6
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load expression values and experiment design
data("RN_Brain_Example_tpm", "RN_Brain_Example_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN). 
Results &lt;- RN_calc(RN_Brain_Example_tpm[1:10000,], RN_Brain_Example_design)
#select only genes with significant changes of expression
Results &lt;- RN_select(Results)
#Compute the Point Mutual information Matrix
Results &lt;- RN_pmi(Results)

#load expression values and experiment design
data("RN_BarresLab_FPKM", "RN_BarresLab_design")
#compute statistics and p-values (considering only a subset of genes due to
#examples running time limit of CRAN)
Results_B &lt;- RN_calc(RN_BarresLab_FPKM[1:10000,], RN_BarresLab_design)
#select only genes with significant changes of expression
Results_B &lt;- RN_select(Results_B)
#Compute the Point Mutual information matrix
Results_B &lt;- RN_pmi(Results_B)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
