<!DOCTYPE html><html><head><title>Help for package rnmamod</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rnmamod}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rnmamod-package'><p>rnmamod: Bayesian Network Meta-analysis with Missing Participants</p></a></li>
<li><a href='#balloon_plot'><p>Enhanced balloon plot</p></a></li>
<li><a href='#baseline_model'><p>The baseline model for binary outcome</p></a></li>
<li><a href='#bland_altman_plot'><p>The Bland-Altman plot</p></a></li>
<li><a href='#comp_clustering'><p>End-user-ready results for comparison dissimilarity and hierarchical clustering</p>
(Comparisons' comparability for transitivity evaluation)</a></li>
<li><a href='#data_preparation'><p>Prepare the dataset in the proper format for R2jags</p></a></li>
<li><a href='#dendro_heatmap'><p>Dendrogram with amalgamated heatmap</p>
(Comparisons' comparability for transitivity evaluation)</a></li>
<li><a href='#describe_network'><p>A function to describe the evidence base</p></a></li>
<li><a href='#distr_characteristics'><p>Visualising the distribution of characteristics</p>
(Comparisons' comparability for transitivity evaluation)</a></li>
<li><a href='#forestplot'><p>Comparator-specific forest plot for network meta-analysis</p></a></li>
<li><a href='#forestplot_metareg'><p>Comparator-specific forest plot for network meta-regression</p></a></li>
<li><a href='#gower_distance'><p>Gower's dissimilarity measure</p>
(Trials' comparability for transitivity evaluation)</a></li>
<li><a href='#heatmap_missing_dataset'><p>Heatmap of proportion of missing participants in the dataset</p></a></li>
<li><a href='#heatmap_missing_network'><p>Heatmap of proportion of missing participants in the network</p></a></li>
<li><a href='#heatmap_robustness'><p>Heatmap of robustness</p></a></li>
<li><a href='#heterogeneity_param_prior'><p>Determine the prior distribution for the heterogeneity parameter</p></a></li>
<li><a href='#improved_ume'><p>Detect the frail comparisons in multi-arm trials</p></a></li>
<li><a href='#internal_measures_plot'><p>Internal measures for cluster validation</p>
(Comparisons' comparability for transitivity evaluation)</a></li>
<li><a href='#intervalplot_panel_ume'><p>A panel of interval plots for the unrelated mean effects model</p></a></li>
<li><a href='#kld_barplot'><p>Barplot for the Kullback-Leibler divergence measure</p></a></li>
<li><a href='#league_heatmap'><p>League heatmap for estimation</p></a></li>
<li><a href='#league_heatmap_pred'><p>League heatmap for prediction</p></a></li>
<li><a href='#league_table_absolute'><p>League table for relative and absolute effects</p></a></li>
<li><a href='#league_table_absolute_user'><p>League table for relative and absolute effects (user defined)</p></a></li>
<li><a href='#leverage_plot'><p>Leverage plot</p></a></li>
<li><a href='#mcmc_diagnostics'><p>Markov Chain Monte Carlo diagnostics</p></a></li>
<li><a href='#metareg_plot'><p>End-user-ready results for network meta-regression</p></a></li>
<li><a href='#miss_characteristics'><p>Visualising missing data in characteristics</p>
(Comparisons' comparability for transitivity evaluation)</a></li>
<li><a href='#missingness_param_prior'><p>Define the mean value of the normal distribution of the missingness parameter</p></a></li>
<li><a href='#netplot'><p>Network plot</p></a></li>
<li><a href='#nma.baker2009'><p>Pharmacological interventions for chronic obstructive pulmonary disease</p></a></li>
<li><a href='#nma.bottomley2011'><p>Pharmacological interventions for moderately severe scalp psoriasis</p></a></li>
<li><a href='#nma.dogliotti2014'><p>Oral antithrombotics for stroke episode</p></a></li>
<li><a href='#nma.liu2013'><p>Antidepressants in Parkinson's disease</p></a></li>
<li><a href='#nma.schwingshackl2014'><p>Training modalities for patients with type 2 diabetes</p></a></li>
<li><a href='#nma.stowe2011'><p>Antiparkinsonian interventions for later Parkinson's disease</p></a></li>
<li><a href='#nodesplit_plot'><p>End-user-ready results for the node-splitting approach</p></a></li>
<li><a href='#pma.hetrick2012'><p>Paroxetine versus placebo for depressive disorders</p></a></li>
<li><a href='#pma.taylor2004'><p>Inositol versus glucose for depressive episode</p></a></li>
<li><a href='#prepare_model'><p>WinBUGS code for Bayesian pairwise or network meta-analysis and</p>
meta-regression</a></li>
<li><a href='#prepare_nodesplit'><p>WinBUGS code for the node-splitting approach</p></a></li>
<li><a href='#prepare_ume'><p>WinBUGS code for the unrelated mean effects model</p></a></li>
<li><a href='#rankosucra_plot'><p>Rankograms and SUCRA curves</p></a></li>
<li><a href='#robustness_index'><p>Robustness index</p></a></li>
<li><a href='#robustness_index_user'><p>Robustness index when 'metafor' or 'netmeta' are used</p></a></li>
<li><a href='#run_metareg'><p>Perform Bayesian pairwise or network meta-regression</p></a></li>
<li><a href='#run_model'><p>Perform Bayesian pairwise or network meta-analysis</p></a></li>
<li><a href='#run_nodesplit'><p>Perform the node-splitting approach</p></a></li>
<li><a href='#run_sensitivity'><p>Perform sensitivity analysis for missing participant outcome data</p></a></li>
<li><a href='#run_series_meta'><p>Perform a series of Bayesian pairwise meta-analyses</p></a></li>
<li><a href='#run_ume'><p>Perform the unrelated mean effects model</p></a></li>
<li><a href='#scatterplot_sucra'><p>Scatterplot of SUCRA values</p></a></li>
<li><a href='#scatterplots_dev'><p>Deviance scatterplots</p></a></li>
<li><a href='#series_meta_plot'><p>End-user-ready results for a series of pairwise meta-analyses</p></a></li>
<li><a href='#taylor_continuous'><p>Pattern-mixture model with Taylor series for continuous outcome</p></a></li>
<li><a href='#taylor_imor'><p>Pattern-mixture model with Taylor series for a binary outcome</p></a></li>
<li><a href='#trans_quality'><p>Rating the quality of transitivity evaluation</p></a></li>
<li><a href='#ume_plot'><p>End-user-ready results for the unrelated mean effects model</p></a></li>
<li><a href='#unrelated_effects_plot'><p>End-user-ready results for unrelated trial effects model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Network Meta-Analysis with Missing Participants</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-24</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Loukia Spineli &lt;Spineli.Loukia@mh-hannover.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A comprehensive suite of functions to perform and visualise 
    pairwise and network meta-analysis with aggregate binary or continuous
    missing participant outcome data. The package covers core Bayesian one-stage
    models implemented in a systematic review with multiple interventions, 
    including fixed-effect and random-effects network meta-analysis, 
    meta-regression, evaluation of the consistency assumption via the 
    node-splitting approach and the unrelated mean effects model, and 
    sensitivity analysis. Missing participant outcome data are addressed in all 
    models of the package. The robustness to primary analysis results can 
    also be investigated using a novel intuitive index. Methods to evaluate the 
    transitivity assumption quantitatively are provided. The package also offers 
    a rich, user-friendly visualisation toolkit that aids in appraising and 
    interpreting the results thoroughly and preparing the manuscript for journal 
    submission. The visualisation tools comprise the network plot, forest plots, 
    panel of diagnostic plots, heatmaps on the extent of missing participant 
    outcome data in the network, league heatmaps on estimation and prediction, 
    rankograms, Bland-Altman plot, leverage plot, deviance scatterplot, heatmap 
    of robustness, barplot of Kullback-Leibler divergence, heatmap of comparison
    dissimilarities and dendrogram of comparison clustering. The package also 
    allows the user to export the results to an Excel file at the working 
    directory.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://CRAN.R-project.org/package=rnmamod">https://CRAN.R-project.org/package=rnmamod</a>,
<a href="https://github.com/LoukiaSpin/rnmamod">https://github.com/LoukiaSpin/rnmamod</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/LoukiaSpin/rnmamod/issues">https://github.com/LoukiaSpin/rnmamod/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cluster, dendextend, fdrtool, gemtc, ggfittext, ggplot2,
ggpubr, ggrepel, heatmaply, igraph, knitr, MASS, mcmcplots,
R2jags, reshape2, scales, stringr, writexl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-24 14:07:44 UTC; Loukia</td>
</tr>
<tr>
<td>Author:</td>
<td>Loukia Spineli [aut, cre],
  Chrysostomos Kalyvas [ctb],
  Katerina Papadimitropoulou [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-24 21:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='rnmamod-package'>rnmamod: Bayesian Network Meta-analysis with Missing Participants</h2><span id='topic+rnmamod-package'></span><span id='topic+rnmamod'></span>

<h3>Description</h3>

<p>An R package for performing Bayesian network meta-analysis while handling
missing participant outcome data properly, assessing the robustness of the
primary analysis results, and exploring the transitivity assumption.
</p>


<h3>Details</h3>

<p>R-package <b>rnmamod</b> is built upon the WinBUGS program code found in
the series of tutorial papers on evidence synthesis methods for decision
making (Dias et al., 2013a; Dias et al., 2013b; Dias et al., 2013c) and
Dias et al. (2010) that introduces the node-splitting approach.
All models comprise Bayesian hierarchical models for one-stage network
meta-analysis and they are implemented in JAGS through the R-package
<b>R2jags</b>.
</p>
<p><b>rnmamod</b> comprises a suite of core models implemented in a
systematic review with multiple interventions:
</p>

<ul>
<li><p> fixed-effect and random-effects network meta-analysis
(<code><a href="#topic+run_model">run_model</a></code>) based on Dias et al. (2013c);
</p>
</li>
<li><p> fixed-effect and random-effects network meta-regression
(<code><a href="#topic+run_metareg">run_metareg</a></code>) based on Cooper et al. (2009), and Dias et
al. (2013b);
</p>
</li>
<li><p> fixed-effect and random-effects separate pairwise meta-analyses for
comparisons with at least two trials (<code><a href="#topic+run_series_meta">run_series_meta</a></code>);
</p>
</li>
<li><p> local evaluation of the consistency assumption using the
fixed-effect or random-effects node-splitting approach
(<code><a href="#topic+run_nodesplit">run_nodesplit</a></code>) based on Dias et al. (2010), and
van Valkenhoef et al. (2016);
</p>
</li>
<li><p> global evaluation of the consistency assumption using the
fixed-effect or random-effects unrelated mean effects model
(<code><a href="#topic+run_ume">run_ume</a></code>) based on Dias et al. (2013a) and Spineli (2021);
</p>
</li>
<li><p> comprehensive sensitivity analysis for the impact of aggregate
binary and continuous missing participant outcome data
(<code><a href="#topic+run_sensitivity">run_sensitivity</a></code>) based on Spineli et al. (2021a);
</p>
</li>
<li><p> hierarchical baseline model for the selected reference intervention
(<code><a href="#topic+baseline_model">baseline_model</a></code>) based in Dias et al. (2013d).
</p>
</li></ul>

<p><b>rnmamod</b> also includes a rich suite of visualisation tools to aid in
the interpretation of the results and preparation of the manuscript for
submission:
</p>

<ul>
<li><p> network plot and description of the evidence base
(<code><a href="#topic+netplot">netplot</a></code> and <code><a href="#topic+describe_network">describe_network</a></code>,
respectively) following the  PRISMA statement for systematic reviews with
network meta-analysis (Hutton et al., 2015);
</p>
</li>
<li><p> illustration of the R-hat (Gelman and Rubin, 1992) and MCMC error
for all monitored nodes and creation of an HTML file with a panel of
diagnostic plots for each monitored parameter
(<code><a href="#topic+mcmc_diagnostics">mcmc_diagnostics</a></code>);
</p>
</li>
<li><p> heatmap on the proportion of missing participants across the network
(<code><a href="#topic+heatmap_missing_network">heatmap_missing_network</a></code>) and across the intervention arms
of each trial in the dataset (<code><a href="#topic+heatmap_missing_dataset">heatmap_missing_dataset</a></code>);
</p>
</li>
<li><p> league heatmap with the estimated and predicted summary effects of
all possible pairwise comparisons in the network and integrated SUCRA
(Salanti et al., 2011) or P-scores (Ruecker and Schwarzer, 2015)
(<code><a href="#topic+league_heatmap">league_heatmap</a></code> and
<code><a href="#topic+league_heatmap_pred">league_heatmap_pred</a></code>, respectively) after performing network
meta-analysis or network meta-regression;
</p>
</li>
<li><p> league table for relative and absolute effects for all pairwise
comparisons and interventions when conducting network meta-analysis anew
via the package (<code><a href="#topic+league_table_absolute">league_table_absolute</a></code>) or using the results
of a published systematic review with network meta-analysis
(<code><a href="#topic+league_table_absolute_user">league_table_absolute_user</a></code>);
</p>
</li>
<li><p> forest plot with the trial-specific and summary absolute risks when
employing the hierarchical baseline model for the selected reference
intervention (<code><a href="#topic+baseline_model">baseline_model</a></code>) as described in
Dias et al. (2013d);
</p>
</li>
<li><p> rankograms with integrated SUCRA values for each intervention in
the network (<code><a href="#topic+rankosucra_plot">rankosucra_plot</a></code>) after performing network
meta-analysis (Salanti et al., 2011);
</p>
</li>
<li><p> forest plot with the estimated and predicted summary effects of all
comparisons with a selected intervention (<code><a href="#topic+forestplot">forestplot</a></code>) as
obtained from the network meta-analysis model, and a forest plot with the
corresponding SUCRA values (Salanti et al., 2011);
</p>
</li>
<li><p> tabulation of the estimated regression coefficient(s), the estimated
and predicted summary effects, measures of model fit and estimated
between-trial standard deviation before and after adjusting for a
trial-specific covariate (<code><a href="#topic+metareg_plot">metareg_plot</a></code>), and visualisation
of the summary effects and SUCRA values from both models
(<code><a href="#topic+forestplot_metareg">forestplot_metareg</a></code>, and <code><a href="#topic+scatterplot_sucra">scatterplot_sucra</a></code>,
respectively&ndash;both found in <code><a href="#topic+metareg_plot">metareg_plot</a></code>);
</p>
</li>
<li><p> tabulation of the estimated direct and indirect effects of the split
nodes and corresponding inconsistency factors, measures of model fit and
estimated between-trial standard deviation after each split node, and
visualisation of these results (<code><a href="#topic+nodesplit_plot">nodesplit_plot</a></code>);
</p>
</li>
<li><p> tabulation of the estimated summary effects of all comparisons
observed in the network, measures of model fit and estimated between-trial
standard deviation under the unrelated mean effects model and network
meta-analysis, as well as visualisation of the summary effects from both
models (<code><a href="#topic+intervalplot_panel_ume">intervalplot_panel_ume</a></code>) and the goodness of fit of
each model using a series of complementary plots
(<code><a href="#topic+scatterplots_dev">scatterplots_dev</a></code> (Dias et al., 2013a),
<code><a href="#topic+bland_altman_plot">bland_altman_plot</a></code> (Bland and Altman, 1999), and
<code><a href="#topic+leverage_plot">leverage_plot</a></code> (Dias et al., 2010)&ndash;all found in
<code><a href="#topic+ume_plot">ume_plot</a></code>);
</p>
</li>
<li><p> tabulation of the estimated summary effects and corresponding
between-trial standard deviation for comparisons with at least two trials
under pairwise and network meta-analysis, as well as visualisation of
these results (<code><a href="#topic+series_meta_plot">series_meta_plot</a></code>);
</p>
</li>
<li><p> calculation and visualisation of the robustness index for all
possible comparisons in the network (<code><a href="#topic+robustness_index">robustness_index</a></code>,
<code><a href="#topic+robustness_index_user">robustness_index_user</a></code> and <code><a href="#topic+heatmap_robustness">heatmap_robustness</a></code>)
(Spineli et al., 2021a);
</p>
</li>
<li><p> enhanced balloon plot with the summary effects and between-trial
standard deviation for a selected pairwise comparison under several
scenarios about the missingness parameter (<code><a href="#topic+balloon_plot">balloon_plot</a></code>)
(Spineli et al., 2021a);
</p>
</li>
<li><p> barplot with the Kullback-Leibler divergence measure from each
informative scenario to the missing-at-random assumption about the
missingness parameter for a selected pairwise comparison
(<code><a href="#topic+kld_barplot">kld_barplot</a></code>) (Spineli et al., 2021a).
</p>
</li></ul>

<p><b>rnmamod</b> also assists the researcher in assessing the transitivity
assumption quantitatively based on trial dissimilarities for various
trial-level aggregate participant and methodological characteristics
calculated using the Gower's dissimilarity coefficient
(<code><a href="#topic+gower_distance">gower_distance</a></code> and <code><a href="#topic+comp_clustering">comp_clustering</a></code>) (Gower,
1971). Results on the clustered comparisons based on hierarchical
agglomerative clustering are illustrated using a dendrogram with integrated
heatmap (<code><a href="#topic+dendro_heatmap">dendro_heatmap</a></code>). The distribution of the
characteristics is presented using violin plots with integrated box
plots and dots, and stacked bar plots across the observed treatment
comparisons (<code><a href="#topic+distr_characteristics">distr_characteristics</a></code>). Missing data in the
characteristics across the trials and observed comparisons are visualised
using bar plots and tile plot (<code><a href="#topic+miss_characteristics">miss_characteristics</a></code>).
</p>
<p>Missing participant outcome data are addressed in all models of the package
after extending the code to incorporate the pattern-mixture model
(Spineli et al., 2021b; Spineli, 2019).
</p>
<p>Type <code>citation("rnmamod")</code> on how to cite <b>rnmamod</b>.
</p>
<p>To report possible bugs and errors, send an email to Loukia Spineli
(<a href="mailto:Spineli.Loukia@mh-hannove.de">Spineli.Loukia@mh-hannove.de</a>).
</p>
<p>The development version of <b>rnmamod</b> is available on
<a href="https://github.com/LoukiaSpin/rnmamod">GitHub</a> under the
GPL-3.0 License.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Bland JM, Altman DG. Measuring agreement in method comparison studies.
<em>Stat Methods Med Res</em> 1999;<b>8</b>(2):135&ndash;60.
doi: 10.1177/096228029900800204
</p>
<p>Cooper NJ, Sutton AJ, Morris D, Ades AE, Welton NJ. Addressing between-study
heterogeneity and inconsistency in mixed treatment comparisons: Application
to stroke prevention treatments in individuals with non-rheumatic atrial
fibrillation. <em>Stat Med</em> 2009;<b>28</b>(14):1861&ndash;81.
doi: 10.1002/sim.3594
</p>
<p>Dias S, Welton NJ, Sutton AJ, Caldwell DM, Lu G, Ades AE. Evidence synthesis
for decision making 4: inconsistency in networks of evidence based on
randomized controlled trials.
<em>Med Decis Making</em> 2013a;<b>33</b>(5):641&ndash;56.
doi: 10.1177/0272989X12455847
</p>
<p>Dias S, Sutton AJ, Welton NJ, Ades AE. Evidence synthesis for decision making
3: heterogeneity&ndash;subgroups, meta-regression, bias, and bias-adjustment.
<em>Med Decis Making</em> 2013b;<b>33</b>(5):618&ndash;40.
doi: 10.1177/0272989X13485157
</p>
<p>Dias S, Sutton AJ, Ades AE, Welton NJ. Evidence synthesis for decision
making 2: a generalized linear modeling framework for pairwise and network
meta-analysis of randomized controlled trials. <em>Med Decis Making</em>
2013c;<b>33</b>(5):607&ndash;17. doi: 10.1177/0272989X12458724
</p>
<p>Dias S, Welton NJ, Sutton AJ, Ades AE. Evidence synthesis for decision
making 5: the baseline natural history model. <em>Med Decis Making</em>
2013d;<b>33</b>(5):657&ndash;70. doi: 10.1177/0272989X13485155
</p>
<p>Dias S, Welton NJ, Caldwell DM, Ades AE. Checking consistency in mixed
treatment comparison meta-analysis.
<em>Stat Med</em> 2010;<b>29</b>(7-8):932&ndash;44.
doi: 10.1002/sim.3767
</p>
<p>Gelman, A, Rubin, DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>
<p>Gower JC. A General Coefficient of Similarity and Some of Its Properties.
<em>Biometrics</em> 1971;<b>27</b>(4):857&ndash;71.
http://dx.doi.org/10.2307/2528823
</p>
<p>Hutton B, Salanti G, Caldwell DM, Chaimani A, Schmid CH, Cameron C, et al.
The PRISMA extension statement for reporting of systematic reviews
incorporating network meta-analyses of health care interventions: checklist
and explanations. <em>Ann Intern Med</em> 2015;<b>162</b>(11):777&ndash;84.
doi: 10.7326/M14-2385
</p>
<p>Ruecker G, Schwarzer G. Ranking treatments in frequentist network
meta-analysis works without resampling methods.
<em>BMC Med Res Methodol</em> 2015;<b>15</b>:58.
doi: 10.1186/s12874-015-0060-8
</p>
<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>
<p>Spineli LM. A revised framework to evaluate the consistency assumption
globally in a network of interventions. <em>Med Decis Making</em> 2021.
doi: 10.1177/0272989X211068005
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis.
<em>Res Synth Methods</em> 2021a;<b>12</b>(4):475&ndash;90. doi: 10.1002/jrsm.1478
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021b;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>
<p>van Valkenhoef G, Dias S, Ades AE, Welton NJ. Automated generation of
node-splitting models for assessment of inconsistency in network
meta-analysis. <em>Res Synth Methods</em> 2016;<b>7</b>(1):80&ndash;93.
doi: 10.1002/jrsm.1167
</p>

<hr>
<h2 id='balloon_plot'>Enhanced balloon plot</h2><span id='topic+balloon_plot'></span>

<h3>Description</h3>

<p>Creates the enhanced balloon plot for the summary effect size and
between-trial standard deviation, <em>tau</em>, under different scenarios
about the missingness parameter for a pair of interventions.
<code>balloon_plot</code> uses the scenarios considered in
<code><a href="#topic+run_sensitivity">run_sensitivity</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>balloon_plot(sens, compar, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="balloon_plot_+3A_sens">sens</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>.
See 'Value' in <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>.</p>
</td></tr>
<tr><td><code id="balloon_plot_+3A_compar">compar</code></td>
<td>
<p>A character vector with two elements indicating the pairwise
comparison of interest. The first element refers to the 'experimental'
and the second element to the 'control' intervention of the comparison.</p>
</td></tr>
<tr><td><code id="balloon_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions
in the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the <code>plot_effect_size</code> of the selected pairwise comparison,
the different colours and sizes of the bubbles reflect the
posterior standard deviation and the posterior mean, respectively.
A colour key appears below the plot. The size of the bubble is proportional
to the corresponding posterior mean.
Crossed bubbles indicate scenarios with conclusive evidence (the
95% credible interval excludes the null value), and filled bubbles
indicate scenarios with inconclusive evidence (the 95% credible interval
includes the null value). The missing-at-random assumption (primary
analysis) is labeled in a white frame.
Both axes illustrate the scenarios as specified in the argument
<code>mean_scenarios</code> of the <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>:
the x-axis refers to the 'experimental' intervention, and the y-axis refers
to the 'control' intervention.
</p>
<p>The same enhanced balloon plot is created for <em>tau</em> (<code>plot_tau</code>).
However, filled bubbles indicate low statistical heterogeneity
(the posterior median of <em>tau</em> is lower than the median of the
prior distribution for the heterogeneity parameter),
and crossed bubbles indicate considerable statistical heterogeneity
(the posterior median of <em>tau</em> exceeds the median of the prior
distribution for the heterogeneity parameter).
</p>
<p><code>balloon_plot</code> can be used only when missing participant
outcome data have been extracted for at least one trial.
Otherwise, the execution of the function will be stopped and an error
message will be printed on the R console.
</p>


<h3>Value</h3>

<p><code>balloon_plot</code> returns two enhanced balloon plots for one
comparison (see 'Details'):
</p>
<table>
<tr><td><code>plot_effect_size</code></td>
<td>
<p>The enhanced balloon plot for the
summary effect size (according to the argument <code>measure</code> inherited
by <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>) for one pairwise comparison.</p>
</td></tr>
<tr><td><code>plot_tau</code></td>
<td>
<p>The enhanced balloon plot for <em>tau</em>. When the
fixed-effect model has been performed in <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>,
the function will not return the <code>plot_tau</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis. <em>Res Synth Methods</em>
2021;<b>12</b>(4):475&ndash;490. doi: https://doi.org/10.1002/jrsm.1478
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pma.taylor2004")

# Read results from 'run_sensitivity' (using the default arguments)
res_sens &lt;- readRDS(system.file('extdata/res_sens_taylor.rds',
                    package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "inositol")

# Create the enhanced balloon plot for 'inositol versus placebo'
balloon_plot(sens = res_sens,
             compar = c("inositol", "placebo"),
             drug_names = interv_names)

</code></pre>

<hr>
<h2 id='baseline_model'>The baseline model for binary outcome</h2><span id='topic+baseline_model'></span>

<h3>Description</h3>

<p>To process the elements in the argument <code>base_risk</code> of the
<code><a href="#topic+run_model">run_model</a></code> function. It also runs the hierarchical baseline
model, separately from the relative effects model as described in
Dias et al. (2018) and Dias et al. (2013b). The output is to be passed to
<code><a href="#topic+run_model">run_model</a></code> and <code><a href="#topic+run_metareg">run_metareg</a></code> to obtain the
(unadjusted and adjusted, respectively) absolute risks for each
intervention in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseline_model(base_risk, n_chains, n_iter, n_burnin, n_thin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="baseline_model_+3A_base_risk">base_risk</code></td>
<td>
<p>A scalar, a vector of length three with elements sorted in
ascending order, or a matrix with two columns and number of rows equal to
the number of relevant trials. In the case of a scalar or vector, the
elements should be in the interval (0, 1). For the matrix, the first column
refers to the number of events and the second column to the sample size of
the trials comprising the dataset for the baseline model. See 'Details' in
<code><a href="#topic+run_model">run_model</a></code>.
This argument is only relevant for a binary outcome.</p>
</td></tr>
<tr><td><code id="baseline_model_+3A_n_chains">n_chains</code></td>
<td>
<p>Positive integer specifying the number of chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="baseline_model_+3A_n_iter">n_iter</code></td>
<td>
<p>Positive integer specifying the number of Markov chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="baseline_model_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Positive integer specifying the number of iterations to
discard at the beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="baseline_model_+3A_n_thin">n_thin</code></td>
<td>
<p>Positive integer specifying the thinning rate for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>base_risk</code> is a matrix, <code>baseline_model</code> creates the
hierarchical baseline model in the JAGS dialect of the BUGS language.
The output of this function (see 'Value') constitutes the
posterior mean and precision of the predicted logit of an event for the
selected reference intervention and it is plugged in the WinBUGS code for
the relative effects model (Dias et al., 2013a) via the
<code><a href="#topic+prepare_model">prepare_model</a></code> function. Following (Dias et al., 2013a), a
uniform prior distribution is assigned on the between-trial standard
deviation with upper and lower limit equal to 0 and 5, respectively.
</p>
<p>When <code>base_risk</code> is a matrix, the function also returns a forest plot
with the estimated trial-specific probability of an event and 95% credible
intervals (the random effects) alongside the corresponding observed
probability of an event for the selected reference intervention. A grey
rectangular illustrates the summary mean and 95% credible interval of the
random effects.
</p>
<p>When <code>base_risk</code> is a matrix (predicted baseline), the model is
updated until convergence using the <code><a href="R2jags.html#topic+autojags">autojags</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>


<h3>Value</h3>

<p>When <code>base_risk</code> is scalar (fixed baseline), the function
returns the user-defined baseline for the selected reference intervention
in the logit scale. When <code>base_risk</code> is a vector (random baseline),
the function returns a vector with the calculated logit of an event for the
selected reference intervention and its precision. Finally, when
<code>base_risk</code> is a matrix (predicted baseline), the function returns the
following elements:
</p>
<table>
<tr><td><code>ref_base</code></td>
<td>
<p>A vector with the posterior mean and precision of the
<b>predicted</b> logit of an event for the selected reference intervention.
This vector is be passed to <code><a href="#topic+run_model">run_model</a></code> and
<code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code>figure</code></td>
<td>
<p>A forest plot on the trial-specific observed and estimated
baseline risk. See 'Details'.</p>
</td></tr>
<tr><td><code>table_baseline</code></td>
<td>
<p>A table with the posterior and predictive
distribution of the summary baseline mean and the posterior distribution of
the between-trial standard deviation in baseline. All results are in the
logit scale.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Ades AE, Welton NJ, Jansen JP, Sutton AJ. Network Meta-Analysis for
Decision Making. Chichester (UK): Wiley; 2018.
</p>
<p>Dias S, Sutton AJ, Ades AE, Welton NJ. Evidence synthesis for decision
making 2: a generalized linear modeling framework for pairwise and network
meta-analysis of randomized controlled trials. <em>Med Decis Making</em>
2013a;<b>33</b>(5):607&ndash;17. doi: 10.1177/0272989X12458724
</p>
<p>Dias S, Welton NJ, Sutton AJ, Ades AE. Evidence synthesis for decision
making 5: the baseline natural history model. <em>Med Decis Making</em>
2013b;<b>33</b>(5):657&ndash;70. doi: 10.1177/0272989X13485155
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prepare_model">prepare_model</a></code>,
<code><a href="R2jags.html#topic+autojags">autojags</a></code>, <code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>
</p>

<hr>
<h2 id='bland_altman_plot'>The Bland-Altman plot</h2><span id='topic+bland_altman_plot'></span>

<h3>Description</h3>

<p>This function facilitates creating the Bland-Altman plot on the
posterior mean deviance contribution for two models using only three
arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bland_altman_plot(model1, model2, colour)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bland_altman_plot_+3A_model1">model1</code></td>
<td>
<p>A vector with the numeric values of the target model (for
instance, the consistency model).</p>
</td></tr>
<tr><td><code id="bland_altman_plot_+3A_model2">model2</code></td>
<td>
<p>A vector with the numeric values of the reference model (for
instance, the unrelated mean effects model).</p>
</td></tr>
<tr><td><code id="bland_altman_plot_+3A_colour">colour</code></td>
<td>
<p>A string to define the colour of the data points in the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>bland_altman_plot</code> is integrated in <code><a href="#topic+ume_plot">ume_plot</a></code>
to create the Bland-Altman plot on the posterior mean of deviance
under the consistency model (via <code><a href="#topic+run_model">run_model</a></code>) and the
unrelated mean effects model (via <code><a href="#topic+run_ume">run_ume</a></code>).
</p>
<p>A uniform scattering of the data points within the 95% limits of agreement
and average bias close to 0 indicate that the compared models have a good
agreement. Data points positioned above or below the 95% limits of
agreement correspond to trials that contribute to the poor fit of the
consistency model or unrelated mean effects model, respectively.
</p>
<p><code>bland_altman_plot</code> can be used to compare the following models
regarding deviance contribution:
</p>

<ul>
<li><p> the consistency model (via <code><a href="#topic+run_model">run_model</a></code>) with the
unrelated effect means model (via <code><a href="#topic+run_ume">run_ume</a></code>);
</p>
</li>
<li><p> the network meta-analysis model (via <code><a href="#topic+run_model">run_model</a></code>) with
the network meta-analysis model (via <code><a href="#topic+run_metareg">run_metareg</a></code>).
</p>
</li></ul>



<h3>Value</h3>

<p>Bland-Altman plot on the posterior mean deviance contribution of the
individual data points under model 1 and model 2.
Each data point corresponds to a trial-arm indicated by a pair of numbers.
The first number refers to the position of the trial in the dataset,
and the second arm refers to the corresponding trial-arm (see 'Arguments'
and 'Value' in <code><a href="#topic+data_preparation">data_preparation</a></code>).
The plot also displays the average bias and the 95% limits of agreement
with horizontal solid black lines.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Bland JM, Altman DG. Measuring agreement in method comparison studies.
<em>Stat Methods Med Res</em> 1999;<b>8</b>:135&ndash;60.
doi: 10.1177/096228029900800204
</p>


<h3>See Also</h3>

<p><code><a href="#topic+data_preparation">data_preparation</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_ume">run_ume</a></code>, <code><a href="#topic+ume_plot">ume_plot</a></code>
</p>

<hr>
<h2 id='comp_clustering'>End-user-ready results for comparison dissimilarity and hierarchical clustering
(Comparisons' comparability for transitivity evaluation)</h2><span id='topic+comp_clustering'></span>

<h3>Description</h3>

<p><code>comp_clustering</code> hosts a toolkit of functions that facilitates
conducting, visualising and evaluating hierarchical agglomerative of
observed comparisons of interventions for a specific network and set of
characteristics that act as effect modifiers. It also calculates the
non-statistical heterogeneity within-comparisons and between-comparisons
using the dissimilarities among all trials of the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comp_clustering(
  input,
  drug_names,
  threshold,
  informative = TRUE,
  ranged_values = FALSE,
  optimal_clusters,
  get_plots = "none",
  label_size = 4,
  title_size = 14,
  axis_title_size = 14,
  axis_text_size = 14,
  axis_x_text_angle = 0,
  legend_text_size = 13,
  str_wrap_width = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comp_clustering_+3A_input">input</code></td>
<td>
<p>A data-frame in the long arm-based format. Two-arm trials occupy
one row in the data-frame. Multi-arm trials occupy as many rows as the
number of possible comparisons among the interventions. The first three
columns refer to the trial name, first and second arm of the comparison,
respectively. The remaining columns refer to summary characteristics. See
'Details' for the specification of the columns.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions
in the order they have been defined in the argument <code>input</code>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_threshold">threshold</code></td>
<td>
<p>A positive scalar to indicate the cut-off of low
dissimilarity of two comparisons. The value must be low.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_informative">informative</code></td>
<td>
<p>Logical with <code>TRUE</code> for evaluating only the
comparison dissimilarity and <code>FALSE</code> for performing hierarchical
agglomerative clustering, thus, allowing the user to define the number of
clusters via the argument <code>optimal_clusters</code>.
The default argument is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_ranged_values">ranged_values</code></td>
<td>
<p>Whether to use a colour scale when creating the
heatmap of within-comparison and between-comparison dissimilarities
(<code>TRUE</code>) or colour the cells with green and orange, when below or
exceeding the specified <code>threshold</code>. Relevant only when
<code>informative = TRUE</code>. The default argument is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_optimal_clusters">optimal_clusters</code></td>
<td>
<p>A positive integer for the optimal number of
clusters, ideally, decided after inspecting the profile plot with average
silhouette widths for a range of clusters, and the dendrogram. The user
<b>must</b> define the value. It takes values from two to the number of
trials minus one.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_get_plots">get_plots</code></td>
<td>
<p>Logical with values <code>TRUE</code> for returning all plots and
<code>FALSE</code> for concealing the plots. The default argument is
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_label_size">label_size</code></td>
<td>
<p>A positive integer for the font size of labels in the
violin plot for the study dissimilarities per comparison and comparison
between comparisons. <code>label_size</code> determines the size argument found
in the geom's aesthetic properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_title_size">title_size</code></td>
<td>
<p>A positive integer for the font size of legend title in
the stacked barplot on the percentage studies of each comparison found in
the clusters. <code>title_size</code> determines the title argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A positive integer for the font size of axis title in
the violin plot for the study dissimilarities per comparison and comparison
between comparisons, and the barplot of percentage trials per comparison
and cluster. <code>axis_title_size</code> determines the axis.title
argument found in the theme's properties in the
R-package <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A positive integer for the font size of axis text in
the violin plot for the study dissimilarities per comparison and comparison
between comparisons, the heatmap of comparison dissimilarity, and the
barplot of percentage trials per comparison and cluster.
<code>axis_text_size</code> determines the axis.text argument found in the
theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_axis_x_text_angle">axis_x_text_angle</code></td>
<td>
<p>A positive integer for the angle of axis text in
the violin plot for the study dissimilarities per comparison and comparison
between comparisons. <code>axis_x_text_angle</code> determines the axis.text.x
argument found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A positive integer for the font size of legend text
in the barplot of percentage trials per comparison and cluster.
<code>legend_text_size</code> determines the legend.text argument found in the
theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="comp_clustering_+3A_str_wrap_width">str_wrap_width</code></td>
<td>
<p>A positive integer for wrapping the axis labels in the
the violin plot for the study dissimilarities per comparison between
comparisons. <code>str_wrap_width</code> determines the
<code><a href="stringr.html#topic+str_wrap">str_wrap</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=stringr">stringr</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The correct type mode of columns in <code>input</code> must be ensured to use
the function <code>comp_clustering</code>. The first three columns referring to
the trial name, first and second arm of the comparison, respectively, must
be <strong>character</strong>. The remaining columns referring to the
characteristics must be <strong>double</strong> or <strong>integer</strong> depending on
whether the corresponding characteristic refers to a quantitative or
qualitative variable. The type mode of each column is assessed by
<code>comp_clustering</code> using the base function <code>typeof</code>. Note that
<code>comp_clustering</code> invites unordered and ordered variables; for the
latter, add the argument <code>ordered = TRUE</code> in the base function
<b>factor()</b>.
</p>
<p>The interventions should be sorted in an ascending order of their
identifier number within the trials so that the first intervention column
(second column in <code>input</code>) is the control arm for every pairwise
comparison. This is important to ensure consistency in the intervention
order within the comparisons obtained from the other related functions.
</p>
<p><code>comp_clustering</code> excludes from the dataset the following type of
characteristics: (i) completely missing characteristics and
(ii) characteristics with missing values in all but one studies for at
least one non-single-stufy comparison. Then it proceeds with the clustering
process.
</p>
<p>The cophenetic correlation coefficient is calculated using the
<code><a href="stats.html#topic+cophenetic">cophenetic</a></code> function alongside the
<code><a href="stats.html#topic+hclust">hclust</a></code> function for selected linkage methods.
</p>
<p><code>comp_clustering</code> can be used only for a network with at least three
comparisons. Otherwise, the execution of the function will be stopped and
an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p>Initially, <code>comp_clustering</code> prints on the console the following
messages: the number of observed comparisons (and number of single-study
comparisons, if any); the number of dropped characteristics due to many
missing data; the maximum value of the cophenetic correlation coefficient;
and the optimal linkage method selected based on the cophenetic correlation
coefficient. Then, the function returns the following list of elements:
</p>
<table>
<tr><td><code>Trials_diss_table</code></td>
<td>
<p>A lower off-diagonal matrix of 'dist' class
with the Gower dissimilarities of all pairs of studies in the network.</p>
</td></tr>
<tr><td><code>Comparisons_diss_table</code></td>
<td>
<p>A lower off-diagonal matrix of 'dist' class
with the within-comparison dissimilarities at the main diagonal and the
between-comparison dissimilarities of all pairs of observed
intervention comparisons at the off-diagonal elements.</p>
</td></tr>
<tr><td><code>Total_dissimilarity</code></td>
<td>
<p>A data-frame on the observed comparisons and
comparisons between comparisons, alongside the corresponding
within-comparison and between-comparisons dissimilarity. The data-frame has
been sorted in decreasing within each dissimilarity 'type'.</p>
</td></tr>
<tr><td><code>Types_used</code></td>
<td>
<p>A data-frame with type mode (i.e., double or integer) of
each characteristic.</p>
</td></tr>
<tr><td><code>Total_missing</code></td>
<td>
<p>The percentage of missing cases in the dataset,
calculated as the ratio of total missing cases to the product of the number
of studies with the number of characteristics.</p>
</td></tr>
<tr><td><code>Cluster_comp</code></td>
<td>
<p>A data-frame on the studies and the cluster they
belong (based on the argument <code>optimal_clusters</code>.</p>
</td></tr>
<tr><td><code>Table_average_silhouette_width</code></td>
<td>
<p>A data-frame with the average
silhouette width for a range of 2 to P-1 trials, with P being the number
trials.</p>
</td></tr>
<tr><td><code>Table_cophenetic_coefficient</code></td>
<td>
<p>A data-frame on the cophenetic
correlation coefficient for eight linkage methods (Ward's two
versions, single, complete, average, Mcquitty, median and centroid). The
data-frame has been sorted in decreasing order of the cophenetic correlation
coefficient.</p>
</td></tr>
<tr><td><code>Optimal_link</code></td>
<td>
<p>The optimal linkage method (ward.D, ward.D2, single,
complete, average, mcquitty, median, or centroid) based on the cophenetic
correlation coefficient.</p>
</td></tr>
</table>
<p>If <code>get_plots = FALSE</code> only the list of elements mentioned above is
returned. If <code>get_plots = TRUE</code>, <code>comp_clustering</code> returns a
series of plots in addition to the list of elements mentioned above:
</p>
<table>
<tr><td><code>Within_comparison_dissimilarity</code></td>
<td>
<p>A violin plot with integrated box
plots and dots on the study dissimilarities per observed comparison
(x-axis). Violins are sorted in descending order of the within-comparison
dissimilarities (blue point).</p>
</td></tr>
<tr><td><code>Between_comparison_dissimilarity</code></td>
<td>
<p>A violin plot with integrated box
plots and dots on the study dissimilarities per comparison between
comparisons (x-axis). Violins are sorted in descending order of the
between-comparison dissimilarities (blue point).</p>
</td></tr>
<tr><td><code>Dissimilarity_heatmap</code></td>
<td>
<p>A heatmap on within-comparison and
between-comparison dissimilarities when (<code>informative = TRUE</code>).
Diagonal elements refer to within-comparison dissimilarity, and
off-diagonal elements refer to between-comparisons dissimilarity. Using a
threshold of high similarity (specified using the argument
<code>threshold</code>), cells equal or above this threshold are highlighted in
orange; otherwise, in green. This heatmap aids in finding 'hot spots' of
comparisons that may violate the plausibility of transitivity in the
network. Single-study comparisons are indicated with white numbers.</p>
</td></tr>
<tr><td><code>Profile_plot</code></td>
<td>
<p>A profile plot on the average silhouette width for a
range of 2 to P-1 clusters, with P being the number of trials. The
candidate optimal number of  clusters is indicated with a red point
directly on the line.</p>
</td></tr>
<tr><td><code>Silhouette_width_plot</code></td>
<td>
<p>A silhouette plot illustrating the silhouette
width for each trial, with the trials sorted in decreasing order within the
cluster they belong. This output is obtained by calling the
<code><a href="cluster.html#topic+silhouette">silhouette</a></code> function in the R-package
<a href="https://CRAN.R-project.org/package=cluster">cluster</a>.</p>
</td></tr>
<tr><td><code>Barplot_comparisons_cluster</code></td>
<td>
<p>As stacked barplot on the percentage
trials of each comparison found in the clusters (based on the argument
<code>optimal_clusters</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Gower J. General Coefficient of Similarity and Some of Its Properties.
<em>Biometrics</em> 1971;<b>27</b>(4):857&ndash;71.
doi: 10.2307/2528823
</p>
<p>Sokal R, Rohlf F. The Comparison of Dendrograms by Objective Methods.
<em>Int Assoc Plant Taxon</em> 1962;<b>11</b>(2):33&ndash;40.
doi: 10.2307/1217208
</p>
<p>Handl J, Knowles J, Kell DB. Computational cluster validation in post-genomic
data analysis. <em>Biometrics</em> 2005;<b>21</b>(15):3201&ndash;120.
doi: 10.1093/bioinformatics/bti517
</p>
<p>Rousseeuw PJ. Silhouettes: A graphical aid to the interpretation and
validation of cluster analysis.
<em>J Comput Appl Math</em> 1987;<b>20</b>:53&ndash;65.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cophenetic">cophenetic</a></code>,
<code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="#topic+internal_measures_plot">internal_measures_plot</a></code>,
<code><a href="cluster.html#topic+silhouette">silhouette</a></code>,
<code><a href="stringr.html#topic+str_wrap">str_wrap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fictional dataset
data_set &lt;- data.frame(Trial_name = as.character(1:7),
                      arm1 = c("1", "1", "1", "1", "1", "2", "2"),
                      arm2 = c("2", "2", "2", "3", "3", "3", "3"),
                      sample = c(140, 145, 150, 40, 45, 75, 80),
                      age = c(18, 18, 18, 48, 48, 35, 35),
                      blinding = factor(c("yes", "yes", "yes", "no", "no", "no", "no")))

# Obtain comparison dissimilarities (informative = TRUE)
comp_clustering(input = data_set,
                drug_names = c("A", "B", "C"),
                threshold = 0.13,  # General research setting
                informative = TRUE,
                get_plots = TRUE)


</code></pre>

<hr>
<h2 id='data_preparation'>Prepare the dataset in the proper format for R2jags</h2><span id='topic+data_preparation'></span>

<h3>Description</h3>

<p><code>data_preparation</code> prepares the dataset in the proper format for
R2jags and returns a list of elements that <code><a href="#topic+run_model">run_model</a></code> inherits
via the argument <code>data</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_preparation(data, measure)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data_preparation_+3A_data">data</code></td>
<td>
<p>A data-frame of the one-trial-per-row format with arm-level data.
See 'Format' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="data_preparation_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure. For a binary
outcome, the following can be considered: <code>"OR"</code>, <code>"RR"</code> or
<code>"RD"</code> for the odds ratio, relative risk, and risk difference,
respectively. For a continuous outcome, the following can be considered:
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for mean difference,
standardised mean difference and ratio of means, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>data_preparation</code> prepares the data for the Bayesian analysis
(See 'Format' in <code><a href="#topic+run_model">run_model</a></code>). <code>data_preparation</code>
creates the pseudo-data-frames <code>m_new</code>, <code>I</code>, and <code>m_pseudo</code>
that have the same dimensions with the element <code>N</code>. <code>m_new</code> takes
the zero value for the observed trial-arms with unreported missing
participant outcome data (i.e., <strong>m</strong> equals <code>NA</code> for the
corresponding trial-arms), the same value with <strong>m</strong> for the observed
trial-arms with reported missing participant outcome data, and <code>NA</code>
for the unobserved trial-arms. <code>I</code> is a dummy data-frame and takes the
value one for the observed trial-arms with reported missing participant
outcome data, the zero value for the observed trial-arms with unreported
missing participant outcome data (i.e., <code>m_new</code> equals zero for the
corresponding trial-arms), and <code>NA</code> for the unobserved trial-arms.
Thus, <code>I</code> indicates whether missing participant outcome data have been
collected for the observed trial-arms. If the user has not defined the
element <strong>m</strong> in <code>data_preparation</code>, <code>m_new</code> and <code>I</code>
take the zero value for all observed trial-arms to indicate that no missing
participant outcome data have been collected for the analysed outcome.
<code>I</code> and <code>m_new</code> are used from the following functions of the
package: <code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+prepare_model">prepare_model</a></code>, <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>,
<code><a href="#topic+prepare_nodesplit">prepare_nodesplit</a></code>, <code><a href="#topic+run_ume">run_ume</a></code>,
<code><a href="#topic+prepare_ume">prepare_ume</a></code>, and  <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>.
Lastly, <code>m_pseudo</code> is a variant of <code>m_new</code>: it takes the value -1
for the observed trial-arms with unreported missing participant outcome
data (i.e., <strong>m</strong> equals <code>NA</code> for the corresponding trial-arms),
the same value with <strong>m</strong> for the observed trial-arms with reported
missing participant outcome data, and <code>NA</code> for the unobserved
trial-arms. It is used in function <code><a href="#topic+heatmap_missing_network">heatmap_missing_network</a></code> to
calculate and illustrate the percentage of missing participant outcome data
across the observed comparisons and interventions of the network and the
function <code><a href="#topic+heatmap_missing_dataset">heatmap_missing_dataset</a></code> to illustrate the trial-arms
with unreported missing participant outcome data. All pseudo-data-frames
aim to retain the trials without information on missing participant outcome
data.
</p>
<p>Furthermore, <code>data_preparation</code> sorts the interventions across
the arms of each trial in an ascending order and correspondingly the
remaining elements in <code>data</code> (See 'Format' in
<code><a href="#topic+run_model">run_model</a></code>). <code>data_preparation</code> considers the first
column in <strong>t</strong> as being the control arm for every trial. Thus,
this sorting ensures that interventions with a lower identifier are
consistently treated as the control arm in each trial. This case is
relevant in non-star-shaped networks.
</p>


<h3>Value</h3>

<p>A list of data-frames on the following elements to be passed
to <code><a href="#topic+run_model">run_model</a></code>:
</p>
<table>
<tr><td><code>pseudo_m</code></td>
<td>
<p>A pseudo-data-frame with values -1 and <strong>m</strong> for the
corresponding trial-arms with unreported and reported missing participant
outcome data, respectively (see 'Details').</p>
</td></tr>
<tr><td><code>m</code></td>
<td>
<p>The number of missing participant outcome data in each
trial-arm (see 'Details').</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>The number of randomised participants in each trial-arm.</p>
</td></tr>
<tr><td><code>t</code></td>
<td>
<p>The intervention identifier in each trial-arm.</p>
</td></tr>
<tr><td><code>I</code></td>
<td>
<p>A pseudo-data-frame that indicates whether missing participant
outcome data have been reported or not for each observed trial-arm
(see 'Details').</p>
</td></tr>
<tr><td><code>measure</code></td>
<td>
<p>The effect measure for the analysed outcome.</p>
</td></tr>
<tr><td><code>y0</code></td>
<td>
<p>The observed mean value of the outcome in each trial-arm,
when the outcome is continuous.</p>
</td></tr>
<tr><td><code>se0</code></td>
<td>
<p>The observed standard deviation of the outcome in each
trial-arm, when the outcome is continuous.</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>The number of observed events of the outcome in each
trial-arm, when the outcome is binary.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>See Also</h3>

<p><code><a href="#topic+heatmap_missing_dataset">heatmap_missing_dataset</a></code>,
<code><a href="#topic+heatmap_missing_network">heatmap_missing_network</a></code>,
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>,
<code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_nodesplit">run_nodesplit</a></code>, <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>,
<code><a href="#topic+run_ume">run_ume</a></code>, <code><a href="#topic+prepare_model">prepare_model</a></code>,
<code><a href="#topic+prepare_nodesplit">prepare_nodesplit</a></code>, <code><a href="#topic+prepare_ume">prepare_ume</a></code>
</p>

<hr>
<h2 id='dendro_heatmap'>Dendrogram with amalgamated heatmap
(Comparisons' comparability for transitivity evaluation)</h2><span id='topic+dendro_heatmap'></span>

<h3>Description</h3>

<p><code>dendro_heatmap</code> creates a dendrogram alongside the heatmap of
Gower dissimilarities among the trials in the network for a specific
linkage method and number of clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dendro_heatmap(input, label_size = 12, axis_text_size = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dendro_heatmap_+3A_input">input</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+comp_clustering">comp_clustering</a></code>. See 'Value'
in <code><a href="#topic+comp_clustering">comp_clustering</a></code>.</p>
</td></tr>
<tr><td><code id="dendro_heatmap_+3A_label_size">label_size</code></td>
<td>
<p>A positive integer for the font size of the heatmap
elements. <code>label_size</code> determines the size argument found in the
geom's aesthetic properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="dendro_heatmap_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A positive integer for the font size of row and column
names of the heatmap. <code>axis_text_size</code> determines the axis.text
argument found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dendro_heatmap</code> uses the <code><a href="heatmaply.html#topic+heatmaply">heatmaply</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=heatmaply">heatmaply</a> to create a
cluster heatmap for a selected linkage method and number of clusters. The
function uses different colours to indicate the clusters directly on the
dendrogram, specified using the R-package
<a href="https://CRAN.R-project.org/package=dendextend">dendextend</a>. The names
of the leaves refer to the trials and corresponding pairwise comparison.
</p>
<p>@details
The function inherits the linkage method and number of optimal clusters by
the <code><a href="#topic+comp_clustering">comp_clustering</a></code> function.
</p>
<p>Remember: when using the <code><a href="#topic+comp_clustering">comp_clustering</a></code> function, inspect
the average silhouette width for a wide range of clusters to decide on the
optimal number of clusters.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>See Also</h3>

<p><code><a href="#topic+comp_clustering">comp_clustering</a></code>, <code><a href="heatmaply.html#topic+heatmaply">heatmaply</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fictional dataset
data_set &lt;- data.frame(Trial_name = as.character(1:7),
                      arm1 = c("1", "1", "1", "1", "1", "2", "2"),
                      arm2 = c("2", "2", "2", "3", "3", "3", "3"),
                      sample = c(140, 145, 150, 40, 45, 75, 80),
                      age = c(18, 18, 18, 48, 48, 35, 35),
                      blinding = factor(c("yes", "yes", "yes", "no", "no", "no", "no")))

# Apply hierarchical clustering (informative = FALSE)
hier &lt;- comp_clustering(input = data_set,
                        drug_names = c("A", "B", "C"),
                        threshold = 0.13,  # General research setting
                        informative = FALSE,
                        optimal_clusters = 3,
                        get_plots = TRUE)

# Create the dendrogram with integrated heatmap
dendro_heatmap(hier)


</code></pre>

<hr>
<h2 id='describe_network'>A function to describe the evidence base</h2><span id='topic+describe_network'></span>

<h3>Description</h3>

<p>Calculates the necessary elements to describe the evidence base
for an outcome across the network, the interventions, and observed
comparisons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>describe_network(data, drug_names, measure, save_xls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="describe_network_+3A_data">data</code></td>
<td>
<p>A data-frame of a one-trial-per-row format containing arm-level
data of each trial. See 'Format' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="describe_network_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code>.</p>
</td></tr>
<tr><td><code id="describe_network_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure. For a binary
outcome, the following can be considered: <code>"OR"</code>, <code>"RR"</code> or
<code>"RD"</code> for the odds ratio, relative risk, and risk difference,
respectively. For a continuous outcome, the following can be considered:
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for mean difference,
standardised mean difference and ratio of means, respectively.</p>
</td></tr>
<tr><td><code id="describe_network_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) at the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>describe_network</code> calls <code><a href="#topic+data_preparation">data_preparation</a></code> to
facilitate the calculations.
</p>
<p>Furthermore, <code>describe_network</code> exports the data-frames to separate
'xlsx' files (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code> function
of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) at the working
directory of the user.
</p>


<h3>Value</h3>

<p><code>describe_network</code> returns the following data-frames that describe the
evidence base:
</p>
<table>
<tr><td><code>network_description</code></td>
<td>
<p>The number of: interventions, possible
comparisons, direct and indirect comparisons, number of trials in total,
number of two-arm and multi-arm trials, number of randomised participants,
and proportion of participants completing the trial (completers).
When the outcome is binary, the number of trials with at least one zero
event, and the number of trials with all zero events are also
presented.</p>
</td></tr>
<tr><td><code>table_interventions</code></td>
<td>
<p>For each intervention, the number of trials,
number of randomised participants, and proportion of completers. When the
outcome is binary, the data-frame presents also the corresponding
proportion of total observed events, the minimum, median and maximum
proportion of observed events across the corresponding trials.</p>
</td></tr>
<tr><td><code>table_comparisons</code></td>
<td>
<p>Identical structure to <code>table_interventions</code>
but for each observed comparison in the network.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>See Also</h3>

<p><code><a href="#topic+data_preparation">data_preparation</a></code>, <code><a href="#topic+run_model">run_model</a></code>
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>

<hr>
<h2 id='distr_characteristics'>Visualising the distribution of characteristics
(Comparisons' comparability for transitivity evaluation)</h2><span id='topic+distr_characteristics'></span>

<h3>Description</h3>

<p><code>distr_characteristics</code> uses violin and bar plots to visualise the
distribution of each characteristic in the dataset either per comparison
or cluster of comparisons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_characteristics(
  input,
  drug_names,
  rename_char = NULL,
  cluster = NULL,
  label_size = 4,
  title_size = 14,
  axis_title_size = 14,
  axis_text_size = 14,
  axis_x_text_angle = 0,
  legend_text_size = 13
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_characteristics_+3A_input">input</code></td>
<td>
<p>A data-frame in the long arm-based format. Two-arm trials occupy
one row in the data-frame. Multi-arm trials occupy as many rows as the
number of possible comparisons among the interventions. The first three
columns refer to the trial name, first and second arm of the comparison
(their identifier number), respectively. The remaining columns refer to
summary characteristics. See 'Details' for specifying the columns.</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions
in the order they appear in the argument <code>input</code>.</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_rename_char">rename_char</code></td>
<td>
<p>A list of two elements: (i) a numeric vector with the
position of the characteristics in <code>input</code>, and (ii) a character
vector with the names of the characteristics, as they are wished to
appear in the title of the plots. This argument is optional, in case the
user wants to control the appearance of the titles.</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_cluster">cluster</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+comp_clustering">comp_clustering</a></code> that has
information on the cluster of each comparison. See 'Value' in
<code><a href="#topic+comp_clustering">comp_clustering</a></code>. If <code>cluster</code> is not provided, the
function presents the distribution of characteristics per comparison;
otherwise per cluster. In the latter, the function prints a table with the
comparisons and the corresponding cluster.</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_label_size">label_size</code></td>
<td>
<p>A positive integer for the font size of labels in the
plots. <code>label_size</code> determines the size argument found in the geom's
aesthetic properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_title_size">title_size</code></td>
<td>
<p>A positive integer for the font size of legend title in
the plots. <code>title_size</code> determines the title argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A positive integer for the font size of axis title in
the plots. <code>axis_title_size</code> determines the axis.title argument found
in the theme's properties in the
R-package <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A positive integer for the font size of axis text in
the plots. <code>axis_text_size</code> determines the axis.text argument found in
the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_axis_x_text_angle">axis_x_text_angle</code></td>
<td>
<p>A positive integer for the angle of axis text in
the plots. <code>axis_text_angle</code> determines the axis.text.x argument found
in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="distr_characteristics_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A positive integer for the font size of legend text
in the plots. <code>legend_text_size</code> determines the legend.text argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The correct type mode of columns in <code>input</code> must be ensured to use
the function <code>distr_characteristics</code>. The first three columns
referring to the trial name, first and second arm of the comparison,
respectively, must be <strong>character</strong>. The remaining columns referring
to the characteristics must be <strong>double</strong> or <strong>integer</strong>
depending on whether the corresponding characteristic refers to a
quantitative or qualitative variable. The type mode of each column is
assessed by <code>distr_characteristics</code> using the base function
<code>typeof</code>.
</p>
<p>The interventions should be sorted in an ascending order of their
identifier number within the trials so that the first treatment column
(second column in <code>input</code>) is the control arm for every pairwise
comparison. This is important to ensure consistency in the order of
interventions within the comparisons obtained from the other related
functions.
</p>


<h3>Value</h3>

<p><code>distr_characteristics</code> returns a list of plots using the proper plot
(violin or bar plot) for each characteristic. The size of the dots in the
violin plot (with amalgamated box plots and dots) are proportional to the
total sample size of the study: the large the sample size of the study, the
larger the size of the corresponding point.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>See Also</h3>

<p><code><a href="#topic+comp_clustering">comp_clustering</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fictional dataset
set.seed(13022024)
data_set &lt;- data.frame(Trial_name = as.character(1:(5 + 7 +2)),
                       arm1 = rep(c("1", "2"), c(5 + 7, 2)),
                       arm2 = rep(c("2", "3"), c(5, 7 + 2)),
                       sample = as.numeric(sample(50:300, 5 + 7 + 2)),
                       age = as.numeric(sample(18:50, 5 + 7 + 2)),
                       blinding = factor(rep(c("yes", "no", "yes"), c(5, 7, 2))))

distr_characteristics(input = data_set,
                      drug_names = c("A", "B", "C"))


</code></pre>

<hr>
<h2 id='forestplot'>Comparator-specific forest plot for network meta-analysis</h2><span id='topic+forestplot'></span>

<h3>Description</h3>

<p>Provides a forest plot with the posterior median and 95% credible
and prediction intervals for comparisons with the selected intervention
(comparator) in the network, and a forest plot with the corresponding
SUCRA values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forestplot(full, compar, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forestplot_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="forestplot_+3A_compar">compar</code></td>
<td>
<p>A character to indicate the comparator intervention. It must be
any name found in <code>drug_names</code>.</p>
</td></tr>
<tr><td><code id="forestplot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The y-axis of the forest plot on <b>effect sizes</b> displays the
labels of the interventions in the network; the selected intervention that
comprises the <code>compar</code> argument is annotated in the plot with the
label 'Comparator intervention'.
For each comparison with the selected intervention, the 95% credible and
prediction intervals are displayed as overlapping lines in different
colours. The corresponding numerical results are displayed above each line:
95% credible intervals are found in parentheses, and 95% predictive
intervals are found in brackets. Odds ratios, relative risks, and ratio of
means are reported in the original scale after exponentiation of the
logarithmic scale.
</p>
<p>The y-axis for the forest plot on <b>SUCRA</b> values displays the
labels of the interventions in the network.
The corresponding numerical results are displayed above each line.
Three coloured rectangles appear in the forest plot: a red rectangle for
SUCRA values up to 50%, a yellow rectangular for SUCRA values between
50% and 80%, and a green rectangle for SUCRA values over 80%.
Interventions falling at the green area are considered as the highest
ranked interventions, whilst interventions falling at the red area are
considered as the lowest ranked interventions.
</p>
<p>In both plots, the interventions are sorted in descending order of their
SUCRA values.
</p>
<p><code>forestplot</code> can be used only for a network of interventions.
Otherwise, the execution of the function will be stopped and an error
message will be printed on the R console.
</p>


<h3>Value</h3>

<p>A panel of two forest plots: (1) a forest plot on the effect
estimates and predictions of comparisons with the selected intervention in
the network, and (2) a forest plot on the posterior mean and 95% credible
interval of SUCRA values of the interventions (Salanti et al., 2011).
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.liu2013")

# Show the first six trials of the dataset (one-trial-per-row format)
head(nma.liu2013)

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_liu.rds', package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "pramipexole", "serotonin-norepinephrine
reuptake inhibitor", "serotonin reuptake inhibitor", "tricyclic
antidepressant", "pergolide")

# Create the forest plot
forestplot(full = res,
           compar = "placebo",
           drug_names = interv_names)

</code></pre>

<hr>
<h2 id='forestplot_metareg'>Comparator-specific forest plot for network meta-regression</h2><span id='topic+forestplot_metareg'></span>

<h3>Description</h3>

<p>Provides a forest plot with the posterior median and 95% credible
and prediction intervals for comparisons with the selected intervention
(comparator) in the network under the network meta-analysis <em>and</em>
network meta-regression for a specified level or value of the investigated
covariate, and a forest plot with the corresponding SUCRA values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forestplot_metareg(full, reg, compar, cov_value, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forestplot_metareg_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="forestplot_metareg_+3A_reg">reg</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_metareg">run_metareg</a></code>. See 'Value' in
<code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code id="forestplot_metareg_+3A_compar">compar</code></td>
<td>
<p>A character to indicate the comparator intervention. It must
be any name found in <code>drug_names</code>.</p>
</td></tr>
<tr><td><code id="forestplot_metareg_+3A_cov_value">cov_value</code></td>
<td>
<p>A list of two elements in the following order: a number
for the covariate value of interest (see 'Arguments' in
<code><a href="#topic+run_metareg">run_metareg</a></code>), and a character to indicate the name of
the covariate. See also 'Details'.</p>
</td></tr>
<tr><td><code id="forestplot_metareg_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The y-axis of the forest plot on <b>effect sizes</b> displays the
labels of the interventions in the network; the selected intervention that
comprises the <code>compar</code> argument is annotated in the plot with the
label 'Comparator intervention'.
For each comparison with the selected intervention, the 95% credible and
prediction intervals are displayed as overlapping lines. Black lines refer
to estimation under both analyses. Green and red lines refer to prediction
under network meta-analysis and network meta-regression, respectively. The
corresponding numerical results are displayed above each line:
95% credible intervals are found in parentheses, and 95% predictive
intervals are found in brackets. Odds ratios, relative risks, and ratio of
means are reported in the original scale after exponentiation of the
logarithmic scale.
</p>
<p>The y-axis for the forest plot on <b>SUCRA</b> values displays the
labels of the interventions in the network.
The corresponding numerical results are displayed above each line.
Three coloured rectangles appear in the forest plot: a red rectangle for
SUCRA values up to 50%, a yellow rectangular for SUCRA values between
50% and 80%, and a green rectangle for SUCRA values over 80%.
Interventions falling at the green area are considered as the highest
ranked interventions, whilst interventions falling at the red area are
considered as the lowest ranked interventions.
</p>
<p>In both plots, the interventions are sorted in the descending order of
their SUCRA values based on the network meta-analysis.
</p>
<p>To obtain the posterior distribution of SUCRAs under the network
meta-regression for a specified level or value of the investigated
covariate, a two-step procedured is followed. First, the posterior median and
standard deviation of the basic parameters and corresponding beta
coefficients are considered to calculate the meadin and standard deviation of
the basic parameters at the selected level or value of the investigated
covariate: d + beta * cov_value. Then, these calculated values are fed into
the BUGS code to get the posterior distribution of SUCRA. The progress of
the simulation appears on the R console.
</p>
<p><code>forestplot_metareg</code> is integrated in <code><a href="#topic+metareg_plot">metareg_plot</a></code>.
</p>
<p><code>forestplot_metareg</code> can be used only for a network of interventions.
In the case of two interventions, the execution of the function will be
stopped and an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p>A panel of two forest plots: (1) a forest plot on the effect
estimates and predictions of comparisons with the selected intervention in
the network under the network meta-analysis and network meta-regression for
a specified level or value of the investigated covariate, and (2) a forest
plot on the posterior mean and 95% credible interval of SUCRA values of
the interventions (Salanti et al., 2011).
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
<a href="https://doi.org/10.1016/j.jclinepi.2010.03.016">doi:10.1016/j.jclinepi.2010.03.016</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metareg_plot">metareg_plot</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>
</p>

<hr>
<h2 id='gower_distance'>Gower's dissimilarity measure
(Trials' comparability for transitivity evaluation)</h2><span id='topic+gower_distance'></span>

<h3>Description</h3>

<p><code>gower_distance</code> calculate the Gower's dissimilarity coefficient for
all pairs of trials included in a network of interventions, considering
several characteristics measured at trial level.
It takes values from 0 to 1, with 0 implying complete similarity and 1
complete dissimilarity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gower_distance(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gower_distance_+3A_input">input</code></td>
<td>
<p>A data-frame in the long arm-based format. Two-arm trials occupy
one row in the data-frame. Multi-arm trials occupy as many rows as the
number of possible comparisons among the interventions. The first two
columns refer to the trial name, and the pairwise comparison,
respectively. The remaining columns refer to summary characteristics. See
'Details' for the specification of the columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The correct type mode of columns in <code>input</code> must be ensured to use
the function <code>gower_distance</code>. The first two columns referring to
the trial name, and pairwise comparison, respectively, must
be <strong>character</strong>. The remaining columns referring to the
characteristics must be <strong>double</strong> or <strong>integer</strong> depending on
whether the corresponding characteristic refers to a quantitative or
qualitative variable. The type mode of each column is assessed by
<code>gower_distance</code> using the base function <code>typeof</code>. Note that
<code>gower_distance</code> invites unordered and ordered variables; for the
latter, add the argument <code>ordered = TRUE</code> in the base function
<b>factor()</b>.
</p>
<p><code>gower_distance</code> is integrated in the function
<code><a href="#topic+comp_clustering">comp_clustering</a></code>.
</p>


<h3>Value</h3>

<p><code>gower_distance</code> returns the following list of elements:
</p>
<table>
<tr><td><code>Dissimilarity_table</code></td>
<td>
<p>A lower off-diagonal matrix of 'dist' class
with the dissimilarities of all pairs of trials.</p>
</td></tr>
<tr><td><code>Types_used</code></td>
<td>
<p>A data-frame with type mode (i.e., double or integer) of
each characteristic.</p>
</td></tr>
<tr><td><code>Total_missing</code></td>
<td>
<p>The percentage of missing cases in the comparison,
calculated as the ratio of total missing cases to the product of the number
of studies with the number of characteristics.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Gower J. General Coefficient of Similarity and Some of Its Properties.
<em>Biometrics</em> 1971;<b>27</b>(4):857&ndash;71.
doi: 10.2307/2528823
</p>


<h3>See Also</h3>

<p><code><a href="#topic+comp_clustering">comp_clustering</a></code>
</p>

<hr>
<h2 id='heatmap_missing_dataset'>Heatmap of proportion of missing participants in the dataset</h2><span id='topic+heatmap_missing_dataset'></span>

<h3>Description</h3>

<p>Illustrates the proportion of missing participants and the
associated risk of bias in each arm of every trial in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_missing_dataset(data, trial_names, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_missing_dataset_+3A_data">data</code></td>
<td>
<p>A data-frame of a one-trial-per-row format containing arm-level
data of each trial. See 'Format' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="heatmap_missing_dataset_+3A_trial_names">trial_names</code></td>
<td>
<p>A vector of labels with the name of the trials in the
order they appear in the argument <code>data</code>.</p>
</td></tr>
<tr><td><code id="heatmap_missing_dataset_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the number of missing participants has not been extracted for
any arm of the trials, the execution of the function will be stopped and
an error message will be printed on the R console.
</p>
<p>When there are more than 80 trials, the value on the proportion of missing
participants will not appear on the heatmap. This is because the number on
the cells will not be distinguishable.
</p>


<h3>Value</h3>

<p>A heatmap presenting the proportion of missing participants in each
trial-arm of the dataset. The columns and the rows of the heatmap
correspond to the interventions and trials, respectively.
The 'five-and-twenty' rule of Sackett and colleagues (1997) is used to
characterise the proportion of missing participants as being associated
with low (up to 5%), moderate (more than 5% and up to 20%), and high
risk of bias (more than 20%). Low, moderate, and high risk of bias due to
missing participants are indicated using green, orange, and red colour,
respectively. The function is also applicable for a pairwise meta-analysis.
If missing participants have not been reported for a trial-arm, the
corresponding cell is indicated in grey.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Sackett DL, Richardson WS, Rosenberg WM, Haynes RB. Evidence-based medicine:
how to practice and teach EBM. New York: Churchill Livingstone 1997.
ISBN: 0-443-05686-2.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.schwingshackl2014")

# Return the first six trials of the dataset
head(nma.schwingshackl2014)

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("aerobic", "resistance", "combined training")

# Create the heatmap
heatmap_missing_dataset(data = nma.schwingshackl2014,
                        trial_names = nma.schwingshackl2014$study,
                        drug_names = interv_names)

</code></pre>

<hr>
<h2 id='heatmap_missing_network'>Heatmap of proportion of missing participants in the network</h2><span id='topic+heatmap_missing_network'></span>

<h3>Description</h3>

<p>Illustrates the distribution of missing participants and the
associated risk of bias for each intervention and observed comparison in
the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_missing_network(data, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_missing_network_+3A_data">data</code></td>
<td>
<p>A data-frame of a one-trial-per-row format containing arm-level
data of each trial. See 'Format' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="heatmap_missing_network_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap with the proportion of missing participants in each
intervention and observed comparison in the network. Each cell annotates
the median, minimum and maximum (the latter two in parenthesis) proportion
of missing participants across the corresponding trials.
The proportion of missing participants in each intervention and observed
comparison are depicted in the main diagonal and lower off-diagonal with
white and black colour, respectively. The pairwise comparisons are read
from left to right.
</p>
<p>The 'five-and-twenty' rule of Sackett and colleagues (1997) is used to
characterise the <b>median</b> proportion of missing participants as being
associated with low (up to 5%), moderate (more than 5% and up to 20%),
and high risk of bias (more than 20%). Low, moderate, and high risk of
bias associated with missing participants is indicated using green, orange,
and red colour, respectively. If missing participants have not been
reported for an intervention or comparison, the corresponding cell is
indicated in grey.
</p>
<p>The summary statistics (median, minimum and maximum) for each intervention
(main diagonal; white font) result from calculating the proportion of
missing participants in each arm of every trial and then summarising across
the corresponding trial-arms. Similarly, the summary statistics for each
observed comparison (lower off-diagonal; black font) result from
calculating the proportion of total missing participants in each trial and
then summarising across the corresponding trials.
</p>
<p><code>heatmap_missing_network</code> can be used only for a network of
interventions. Otherwise, the execution of the function will be stopped and
an error message will be printed on the R console. Likewise, when the
number of missing participants has not been extracted for any arm of the
trials.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Sackett DL, Richardson WS, Rosenberg WM, Haynes RB. Evidence-based medicine:
how to practice and teach EBM. New York: Churchill Livingstone 1997.
ISBN: 0-443-05686-2.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.stowe2011")

# Return the first six trials of the dataset
head(nma.stowe2011)

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("PBO+LD", "DA+LD", "COMTI+LD", "MAOBI+LD")

# Create the heatmap
heatmap_missing_network(data = nma.stowe2011,
                        drug_names = interv_names)

</code></pre>

<hr>
<h2 id='heatmap_robustness'>Heatmap of robustness</h2><span id='topic+heatmap_robustness'></span>

<h3>Description</h3>

<p>Facilitates the detection of comparisons that are associated
with a lack of robustness in the context of a sensitivity analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_robustness(robust, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_robustness_+3A_robust">robust</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+robustness_index">robustness_index</a></code> and
<code><a href="#topic+robustness_index_user">robustness_index_user</a></code>.
See 'Value' in <code><a href="#topic+robustness_index">robustness_index</a></code> and
<code><a href="#topic+robustness_index_user">robustness_index_user</a></code>.</p>
</td></tr>
<tr><td><code id="heatmap_robustness_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The heatmap illustrates the robustness index for each possible
pairwise comparison in the network. The pairwise comparisons are read from
left to right. Comparisons highlighted with green or red colour imply
robust or frail conclusions for the primary analysis, respectively.
This corresponds to robustness index below or at least the selected
threshold of robustness. <code>heatmap_robustness</code> inherits the threshold
of robustness selected in the <code><a href="#topic+robustness_index">robustness_index</a></code> or
<code><a href="#topic+robustness_index_user">robustness_index_user</a></code> function.
The robustness index of each pairwise comparison also appears in the
corresponding cell.
When there is at least one comparison with frail conclusions, the primary
analysis results may be questionable for the whole network
(Spineli et al., 2021).
</p>
<p><code>heatmap_robustness</code> is <em>not</em> restricted to the sensitivity
analysis concerning the impact of missing participant outcome data.
</p>
<p><code>heatmap_robustness</code> can be used only for a network of interventions.
Otherwise, the execution of the function will be stopped and an
error message will be printed on the R console.
</p>


<h3>Value</h3>

<p><code>heatmap_robustness</code> first prints on the R console a message on
the threshold of robustness determined by the user in
<code><a href="#topic+robustness_index">robustness_index</a></code> and <code><a href="#topic+robustness_index_user">robustness_index_user</a></code>.
Then, it returns a lower triangular heatmap matrix with the robustness
index value of all possible pairwise comparisons.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis.
<em>Res Synth Methods</em> 2021;<b>12</b>(4):475&ndash;90. doi: 10.1002/jrsm.1478
</p>


<h3>See Also</h3>

<p><code><a href="#topic+robustness_index">robustness_index</a></code>, <code><a href="#topic+robustness_index_user">robustness_index_user</a></code>,
<code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Read results from 'run_sensitivity' (using the default arguments)
res_sens &lt;- readRDS(system.file('extdata/res_sens_baker.rds',
                    package = 'rnmamod'))

# Calculate the robustness index
robust &lt;- robustness_index(sens = res_sens,
                           threshold = 0.28)

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "budesonide", "budesonide plus formoterol",
                  "fluticasone", "fluticasone plus salmeterol",
                  "formoterol", "salmeterol", "tiotropium")

# Create the heatmap of robustness
heatmap_robustness(robust = robust,
                   drug_names = interv_names)

</code></pre>

<hr>
<h2 id='heterogeneity_param_prior'>Determine the prior distribution for the heterogeneity parameter</h2><span id='topic+heterogeneity_param_prior'></span>

<h3>Description</h3>

<p>Generates the prior distribution (weakly informative or empirically-based)
for the heterogeneity parameter.
<code><a href="#topic+run_model">run_model</a></code> inherits <code>heterogeneity_param_prior</code> via the
argument <code>heter_prior</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heterogeneity_param_prior(measure, model, heter_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heterogeneity_param_prior_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure. For a binary
outcome, the following can be considered: <code>"OR"</code>, <code>"RR"</code> or
<code>"RD"</code> for the odds ratio, relative risk, and risk difference,
respectively. For a continuous outcome, the following can be considered:
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for mean difference,
standardised mean difference and ratio of means, respectively.</p>
</td></tr>
<tr><td><code id="heterogeneity_param_prior_+3A_model">model</code></td>
<td>
<p>Character string indicating the analysis model with values
<code>"RE"</code>, or <code>"FE"</code> for the random-effects and fixed-effect model,
respectively. The default argument is <code>"RE"</code>.</p>
</td></tr>
<tr><td><code id="heterogeneity_param_prior_+3A_heter_prior">heter_prior</code></td>
<td>
<p>A list of three elements with the following order:
1) a character string indicating the distribution with
(currently available) values <code>"halfnormal"</code>, <code>"uniform"</code>,
<code>"lognormal"</code>, or <code>"logt"</code>; 2) two numeric values that refer to
the parameters of the selected distribution.  For <code>"lognormal"</code>, and
<code>"logt"</code> these numbers refer to the mean and precision, respectively.
For <code>"halfnormal"</code>, these numbers refer to zero and the scale
parameter (equal to 4 or 1 being the corresponding precision of the scale
parameter 0.5 or 1). For <code>"uniform"</code>, these numbers refer to the
minimum and maximum value of the distribution.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The names of the (current) prior distributions follow the JAGS syntax.
The mean and precision of <code>"lognormal"</code> and <code>"logt"</code> should align
with the values proposed by Turner et al. (2015) and Rhodes et al. (2015)
for the corresponding empirically-based prior distributions when
<code>measure</code> is <code>"OR"</code> or <code>"SMD"</code>, respectively.
The users may refer to Dias et al. (2013) to determine the minimum and
maximum value of the uniform distribution, and to Friede et al. (2017)
to determine the mean and precision of the half-normal distribution.
When <code>model</code> is <code>"FE"</code>, <code>heterogeneity_param_prior</code>
is ignored in <code><a href="#topic+run_model">run_model</a></code>.
</p>


<h3>Value</h3>

<p>A value to be passed to <code><a href="#topic+run_model">run_model</a></code>.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Sutton AJ, Ades AE, Welton NJ. Evidence synthesis for decision
making 2: a generalized linear modeling framework for pairwise and
network meta-analysis of randomized controlled trials.
<em>Med Decis Making</em> 2013;<b>33</b>(5):607&ndash;17.
doi: 10.1177/0272989X12458724
</p>
<p>Friede T, Roever C, Wandel S, Neuenschwander B. Meta-analysis of two studies
in the presence of heterogeneity with applications in rare diseases.
<em>Biom J</em> 2017;<b>59</b>(4):658&ndash;71. doi: 10.1002/bimj.201500236
</p>
<p>Rhodes KM, Turner RM, Higgins JP. Predictive distributions were developed
for the extent of heterogeneity in meta-analyses of continuous outcome data.
<em>J Clin Epidemiol</em> 2015;<b>68</b>(1):52&ndash;60.
doi: 10.1016/j.jclinepi.2014.08.012
</p>
<p>Turner RM, Jackson D, Wei Y, Thompson SG, Higgins JP. Predictive
distributions for between-study heterogeneity and simple methods for their
application in Bayesian meta-analysis.
<em>Stat Med</em> 2015;<b>34</b>(6):984&ndash;98. doi: 10.1002/sim.6381
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>

<hr>
<h2 id='improved_ume'>Detect the frail comparisons in multi-arm trials</h2><span id='topic+improved_ume'></span>

<h3>Description</h3>

<p>Detects the frail comparisons in multi-arm trials, that is,
comparisons between non-baseline interventions not investigated in any
two-arm trial in the network (Spineli, 2021). The 'original' model of
Dias et al. (2013) omits the frail comparisons from the estimation process
of the unrelated mean effects model. Consequently, their posterior
distribution coincides with the prior distribution yielding implausible
posterior standard deviations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>improved_ume(t, N, ns, na)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="improved_ume_+3A_t">t</code></td>
<td>
<p>A data-frame of the one-trial-per-row format containing the
intervention identifier in each arm of every trial (see 'Details' below,
and 'Format' in <code><a href="#topic+run_model">run_model</a></code>).</p>
</td></tr>
<tr><td><code id="improved_ume_+3A_n">N</code></td>
<td>
<p>A data-frame of the one-trial-per-row format containing the number
of participants randomised to the assigned intervention in each arm of
every trial (see 'Details' below, and 'Format' in <code><a href="#topic+run_model">run_model</a></code>).</p>
</td></tr>
<tr><td><code id="improved_ume_+3A_ns">ns</code></td>
<td>
<p>A scale parameter on the number trials.</p>
</td></tr>
<tr><td><code id="improved_ume_+3A_na">na</code></td>
<td>
<p>A vector of length equal to <code>ns</code> with the number of arms in
each trial.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>improved_ume</code> is integrated in <code><a href="#topic+run_ume">run_ume</a></code> and
calls the output of <code><a href="#topic+data_preparation">data_preparation</a></code> after sorting the
rows so that multi-arm trials appear at the bottom of the dataset.
When there are no multi-arm trials or no frail comparisons in the network,
<code>improved_ume</code> returns only the element <code>obs_comp</code>
(see, 'Value').
</p>


<h3>Value</h3>

<p>The output of <code>improved_ume</code> is a list of elements that are
inherited by <code><a href="#topic+run_ume">run_ume</a></code>:
</p>
<table>
<tr><td><code>nbase_multi</code></td>
<td>
<p>A scalar parameter on the number of frail comparisons.</p>
</td></tr>
<tr><td><code>t1_bn</code></td>
<td>
<p>A vector with numeric values referring to the first arm of
each frail comparison.</p>
</td></tr>
<tr><td><code>t2_bn</code></td>
<td>
<p>A vector with numeric values referring to the second arm of
each frail comparison.</p>
</td></tr>
<tr><td><code>ref_base</code></td>
<td>
<p>A scalar referring to the reference intervention for the
subnetwork of interventions in frail comparisons.</p>
</td></tr>
<tr><td><code>base</code></td>
<td>
<p>A vector with numeric values referring to the baseline
intervention of the multi-arm trials that contain the frail comparisons.</p>
</td></tr>
<tr><td><code>obs_comp</code></td>
<td>
<p>A data-frame that indicates how many two-arm and
multi-arm trials have included each pairwise comparison observed in the
network.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Sutton AJ, Caldwell DM, Lu G, Ades AE. Evidence synthesis
for decision making 4: inconsistency in networks of evidence based on
randomized controlled trials.
<em>Med Decis Making</em> 2013;<b>33</b>(5):641&ndash;56.
doi: 10.1177/0272989X12455847
</p>
<p>Spineli LM. A revised framework to evaluate the consistency assumption
globally in a network of interventions. <em>Med Decis Making</em> 2021.
doi: 10.1177/0272989X211068005
</p>


<h3>See Also</h3>

<p><code><a href="#topic+data_preparation">data_preparation</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_ume">run_ume</a></code>
</p>

<hr>
<h2 id='internal_measures_plot'>Internal measures for cluster validation
(Comparisons' comparability for transitivity evaluation)</h2><span id='topic+internal_measures_plot'></span>

<h3>Description</h3>

<p><code>internal_measures_plot</code> currently prepares the table with the results 
of the average silhouette width for a range of clusters, and visualises the 
results using a profile plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>internal_measures_plot(
  input,
  optimal_link,
  label_size = 4,
  axis_title_size = 14,
  axis_text_size = 14
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="internal_measures_plot_+3A_input">input</code></td>
<td>
<p>An object of 'dist' class. It is a lower off-diagonal matrix
with the dissimilarities of all pairs of comparisons.</p>
</td></tr>
<tr><td><code id="internal_measures_plot_+3A_optimal_link">optimal_link</code></td>
<td>
<p>A character string with values <code>"ward.D"</code>,
<code>"ward.D2"</code>, <code>"single"</code>, <code>"complete"</code>, <code>"average"</code>,
<code>"mcquitty"</code>, <code>"median"</code>, or <code>"centroid"</code> for the optimal
linkage method, corresponding to the highest cophenetic correlation
coefficient value.</p>
</td></tr>
<tr><td><code id="internal_measures_plot_+3A_label_size">label_size</code></td>
<td>
<p>A positive integer for the font size of labels in the 
profile plot with the average silhouette width per candidate cluster. 
<code>label_size</code> determines the size argument found in the geom's 
aesthetic properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="internal_measures_plot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A positive integer for the font size of axis title in
the profile plot with the average silhouette width per candidate cluster. 
<code>axis_title_size</code> determines the axis.title argument found in the 
theme's properties in the
R-package <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
<tr><td><code id="internal_measures_plot_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A positive integer for the font size of axis text in
the profile plot with the average silhouette width per candidate cluster. 
<code>axis_text_size</code> determines the axis.text argument found in the 
theme's properties in the R-package 
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>internal_measures_plot</code> also calls the function
<code><a href="#topic+comp_clustering">comp_clustering</a></code> to define the argument <code>optimal_link</code> to
create the silhouette plot for the selected number of clusters.
</p>
<p><code>internal_measures_plot</code> calls the  
<code><a href="cluster.html#topic+silhouette">silhouette</a></code> function in the R-package 
<a href="https://CRAN.R-project.org/package=cluster">cluster</a> to obtain the
results on average silhouette for each candidate cluster.
</p>
<p><code>internal_measures_plot</code> is integrated in the function 
<code><a href="#topic+comp_clustering">comp_clustering</a></code>.
</p>


<h3>Value</h3>

<p><code>internal_measures_plot</code> currently returns the following list of 
elements:
</p>
<table>
<tr><td><code>Table_internal_measures</code></td>
<td>
<p>A data-frame of the average silhouette width 
for a range of 2 to P-1 clusters, with P being the number of trials</p>
</td></tr>
<tr><td><code>Internal_measures_panel</code></td>
<td>
<p>A profile plot on the average silhouette 
width for a range of 2 to P-1 clusters, with P being the number of 
trials The candidate optimal number of clusters is indicated with a 
red point directly on the line.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Handl J, Knowles J, Kell DB. Computational cluster validation in post-genomic
data analysis. <em>Biometrics</em> 2005;<b>21</b>(15):3201&ndash;120.
doi: 10.1093/bioinformatics/bti517
</p>
<p>Rousseeuw PJ. Silhouettes: A graphical aid to the interpretation and
validation of cluster analysis.
<em>J Comput Appl Math</em> 1987;<b>20</b>:53&ndash;65.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+comp_clustering">comp_clustering</a></code>, <code><a href="cluster.html#topic+silhouette">silhouette</a></code>
</p>

<hr>
<h2 id='intervalplot_panel_ume'>A panel of interval plots for the unrelated mean effects model</h2><span id='topic+intervalplot_panel_ume'></span>

<h3>Description</h3>

<p>Creates a panel of interval plots on the summary effect sizes
under the consistency model and the unrelated mean effects model.
The number of interval plots equals the number of pairwise comparisons
observed in the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intervalplot_panel_ume(full, ume, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intervalplot_panel_ume_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="intervalplot_panel_ume_+3A_ume">ume</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_ume">run_ume</a></code>. See 'Value' in
<code><a href="#topic+run_ume">run_ume</a></code>.</p>
</td></tr>
<tr><td><code id="intervalplot_panel_ume_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If the argument <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>intervalplot_panel_ume</code> is integrated in the
<code><a href="#topic+ume_plot">ume_plot</a></code> function. The consistency model and the unrelated
mean effects model are abbreviated in the y-axis as 'NMA model' and
'UME model', respectively. The intervals are highlighted with green, when
the corresponding summary effect sizes do not cross the vertical line of no
difference, and red otherwise. Grey panels refer to the frail comparisons
as detected by the <code><a href="#topic+improved_ume">improved_ume</a></code> function (see 'Details' in
<code><a href="#topic+improved_ume">improved_ume</a></code>).
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>intervalplot_panel_ume</code>
currently presents the results in the odds ratio scale.
</p>


<h3>Value</h3>

<p>A panel of interval plots on the posterior mean and 95% credible
interval of the summary effect size under the consistency model and the
improved unrelated mean effects model (Spineli, 2021) of all pairwise
comparisons observed in the network.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Spineli LM. A revised framework to evaluate the consistency assumption
globally in a network of interventions. <em>Med Decis Making</em> 2021.
doi: 10.1177/0272989X211068005
</p>


<h3>See Also</h3>

<p><code><a href="#topic+improved_ume">improved_ume</a></code> <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_ume">run_ume</a></code>, <code><a href="#topic+ume_plot">ume_plot</a></code>
</p>

<hr>
<h2 id='kld_barplot'>Barplot for the Kullback-Leibler divergence measure</h2><span id='topic+kld_barplot'></span>

<h3>Description</h3>

<p>Produces a barplot with the Kullback-Leibler divergence measure
from each re-analysis to the primary analysis for a pairwise
comparison. Currently, <code>kld_barplot</code> is used concerning the impact of
missing participant outcome data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kld_barplot(robust, compar, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kld_barplot_+3A_robust">robust</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+robustness_index">robustness_index</a></code>.
See 'Value' in <code><a href="#topic+robustness_index">robustness_index</a></code>.</p>
</td></tr>
<tr><td><code id="kld_barplot_+3A_compar">compar</code></td>
<td>
<p>A character vector with two elements that indicates the
pairwise comparison of interest. The first element refers to the
'experimental' intervention and the second element refers to the
'control' intervention of the comparison.</p>
</td></tr>
<tr><td><code id="kld_barplot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>kld_barplot</code> uses the scenarios inherited by
<code><a href="#topic+robustness_index">robustness_index</a></code> via the <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>
function. The scenarios for the missingness parameter (see 'Details' in
<code><a href="#topic+run_sensitivity">run_sensitivity</a></code>) in the compared interventions are split to
<em>Extreme</em>, <em>Sceptical</em>, and <em>Optimistic</em> following the
classification of Spineli et al. (2021). In each class, bars will green,
orange, and red colour refer to scenarios without distance, less distant,
and more distant from the primary analysis
(the missing-at-random assumption).
</p>
<p><code>kld_barplot</code> can be used only when missing participant outcome
data have been extracted for at least one trial. Otherwise, the execution
of the function will be stopped and an error message will be printed on
the R console.
</p>


<h3>Value</h3>

<p><code>kld_barplot</code> returns a panel of barplots on the
Kullback-Leibler divergence measure for each re-analysis.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Kullback S, Leibler RA. On information and sufficiency.
<em>Ann Math Stat</em> 1951;<b>22</b>(1):79&ndash;86. doi: 10.1214/aoms/1177729694
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis.
<em>Res Synth Methods</em> 2021;<b>12</b>(4):475&ndash;90. doi: 10.1002/jrsm.1478
</p>


<h3>See Also</h3>

<p><code><a href="#topic+robustness_index">robustness_index</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_sensitivity">run_sensitivity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pma.taylor2004")

# Read results from 'run_sensitivity' (using the default arguments)
res_sens &lt;- readRDS(system.file('extdata/res_sens_taylor.rds',
                    package = 'rnmamod'))

# Calculate the robustness index
robust &lt;- robustness_index(sens = res_sens,
                           threshold = 0.17)

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "inositol")

# Create the barplot for the comparison 'inositol versus placebo'
kld_barplot(robust = robust,
            compar = c("inositol", "placebo"),
            drug_names = interv_names)

</code></pre>

<hr>
<h2 id='league_heatmap'>League heatmap for estimation</h2><span id='topic+league_heatmap'></span>

<h3>Description</h3>

<p>For one outcome, it creates a heatmap of the estimated effect measure for
all possible comparisons of interventions in the network.
For two outcomes, the heatmap illustrates these two outcomes for the same
effect measure in the upper and lower off-diagonals for all
possible comparisons of interventions in the network.
The function can also be used to illustrate the results of two different
models on the same outcome and effect measure.
<code>league_heatmap</code> can be used for a random-effects or fixed-effect
network meta-analysis, network meta-regression, and series of pairwise
meta-analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>league_heatmap(
  full1,
  full2 = NULL,
  cov_value = NULL,
  drug_names1,
  drug_names2 = NULL,
  name1 = NULL,
  name2 = NULL,
  show = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="league_heatmap_+3A_full1">full1</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code> for network
meta-analysis, or <code><a href="#topic+run_metareg">run_metareg</a></code> for network meta-regression.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code> and <code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_full2">full2</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code> for network
meta-analysis, <code><a href="#topic+run_metareg">run_metareg</a></code> for network meta-regression, or
<code><a href="#topic+run_series_meta">run_series_meta</a></code> for a series of pairwise meta-analyses.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>, and
<code><a href="#topic+run_series_meta">run_series_meta</a></code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_cov_value">cov_value</code></td>
<td>
<p>A list of two elements in the following order: a number
for the covariate value of interest and a character for the name of the
covariate. See also 'Details'.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_drug_names1">drug_names1</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code> for <code>full1</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_drug_names2">drug_names2</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code> for <code>full2</code>. The elements must be a subset of
<code>drug_names1</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_name1">name1</code></td>
<td>
<p>The text for the title of the results that refer to
the outcome or model under <code>full1</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_name2">name2</code></td>
<td>
<p>The text for the title of the results that refer to
the outcome or model under <code>full2</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_+3A_show">show</code></td>
<td>
<p>A vector of at least three character strings that refer to the
names of the interventions <em>exactly</em> as defined in <code>drug_names1</code>.
Then, the league table will be created for these interventions only.
If <code>show</code> is not defined, the league table will present all
interventions as defined in <code>drug_names1</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>heatmap_league</code> offers the following options to display
<b>one</b> estimated effect measure for all (or some) pairwise comparisons:
</p>

<ul>
<li><p> one outcome, with results in the lower triangle referring to
comparisons in the opposite direction after converting negative values
into positive values (in absolute or logarithmic scale), and vice versa.
Comparisons between interventions should be read from left to right.
Therefore, each cell refers to the corresponding row-defining intervention
against the column-defining intervention.
Results that indicate strong evidence in favour of the
row-defining intervention (i.e. the respective 95% credible interval does
not include the null value) are indicated in bold. A message is printed on
the R console on how to read the heatmap;
</p>
</li>
<li><p> two outcomes for the same model, namely, network meta-analysis (via
<code><a href="#topic+run_model">run_model</a></code>) or network meta-regression (via
<code><a href="#topic+run_metareg">run_metareg</a></code>).
When one of the outcomes includes more interventions, the argument
<code>full1</code> should be considered for that outcome.
Comparisons between interventions should be read as follows: for the upper
diagonal, each cell refers to the corresponding row-defining intervention
against the column-defining intervention, and for the lower diagonal, each
cell refers to the corresponding column-defining intervention against the
row-defining intervention. Results that indicate strong evidence (i.e. the
respective 95% credible interval does not include the null value) are
indicated in bold. A message is printed on the R console on how to read
the heatmap;
</p>
</li>
<li><p> two models for the same outcome, namely, network meta-analysis
versus network meta-regression, or network meta-analysis versus series of
pairwise meta-analyses.
The instructions to read the heatmap are in line with the previous point.
A message is printed on the R console on how to read the heatmap.
</p>
</li></ul>

<p>For a beneficial outcome, red favours the first intervention of the
comparison, and blue favours the second intervention. For a harmful
outcome, blue favours the first intervention of the comparison, and red
favours the second intervention. The larger the treatment effect, the
darker the colour shade.
</p>
<p>The function displays the effect measure as inherited by the argument
<code>full1</code>. For binary outcome, it can display the odds ratio,
relative risk, and risk difference. See 'Details' in
<code><a href="#topic+run_model">run_model</a></code> for the relative risk, and risk difference.
For continuous outcome, it can display the mean difference, standardised
mean difference, and ratio of means. Odds ratios, relative risk and ratio
of means are reported in the original scale after exponentiation of the
logarithmic scale.
</p>
<p>The rows and columns of the heatmap display the names of
interventions  sorted by decreasing order from the best to the worst
based on their SUCRA value (Salanti et al., 2011) for the outcome or model
under the argument <code>full1</code>. The off-diagonals contain the posterior
median and 95% credible interval of the effect measure (according to the
argument <code>measure</code> as inherited in the argument <code>full1</code>) of the
corresponding comparisons.
</p>
<p>The main diagonal contains the posterior mean of SUCRA of the corresponding
interventions when the arguments <code>full1</code> refers to the
<code><a href="#topic+run_model">run_model</a></code> function. When the arguments <code>full1</code>
refers to the <code><a href="#topic+run_metareg">run_metareg</a></code> function, the p-score
(Ruecker and Schwarzer, 2015) is calculated for each intervention while
taking into account the covariate value in the argument <code>cov_value</code>.
P-score is the 'frequentist analogue to SUCRA'
(Ruecker and Schwarzer, 2015).
</p>
<p>In the case of network meta-regression, when the covariate is binary,
specify in the second element of <code>cov_value</code> the name of the level
for which the heatmap will be created.
</p>
<p><code>league_heatmap</code> can be used only for a network of interventions.
In the case of two interventions, the execution of the function will be
stopped and an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p>A heatmap of the league table showing the posterior median and 95%
credible interval of the comparisons in the off-diagonals, and
the posterior mean of the SUCRA values in the diagonal.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli, Chrysostomos Kalyvas, Katerina Papadimitropoulou
</p>


<h3>References</h3>

<p>Ruecker G, Schwarzer G. Ranking treatments in frequentist network
meta-analysis works without resampling methods.
<em>BMC Med Res Methodol</em> 2015;<b>15</b>:58.
doi: 10.1186/s12874-015-0060-8
</p>
<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_series_meta">run_series_meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.liu2013")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_liu.rds', package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "pramipexole", "serotonin-norepinephrine
                  reuptake inhibitor", "serotonin reuptake inhibitor",
                  "tricyclic antidepressant", "pergolide")

# Create the league heatmap
league_heatmap(full1 = res,
               drug_names1 = interv_names)

</code></pre>

<hr>
<h2 id='league_heatmap_pred'>League heatmap for prediction</h2><span id='topic+league_heatmap_pred'></span>

<h3>Description</h3>

<p>For one outcome, it creates a heatmap with the predicted effect measure for
all possible comparisons of interventions in the network.
For two outcomes, the heatmap illustrates these two outcomes for the same
effect measure in the upper and lower off-diagonals for all possible
comparisons of interventions in the network.
<code>league_heatmap_pred</code> can be used <em>only</em> for a random-effects
network meta-analysis and network meta-regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>league_heatmap_pred(
  full1,
  full2 = NULL,
  cov_value = NULL,
  drug_names1,
  drug_names2 = NULL,
  name1 = NULL,
  name2 = NULL,
  show = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="league_heatmap_pred_+3A_full1">full1</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code> for network
meta-analysis, or <code><a href="#topic+run_metareg">run_metareg</a></code> for network meta-regression.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code> and <code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_full2">full2</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code> for network
meta-analysis, or <code><a href="#topic+run_metareg">run_metareg</a></code> for network meta-regression.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code> and <code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_cov_value">cov_value</code></td>
<td>
<p>A list of two elements in the following order: a number
for the covariate value of interest and a character for the name of the
covariate. See also 'Details'.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_drug_names1">drug_names1</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code> for <code>full1</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_drug_names2">drug_names2</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code> for <code>full2</code>. The elements must be a subset of
<code>drug_names1</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_name1">name1</code></td>
<td>
<p>The text for the title of the results that refer to
the outcome or model under <code>full1</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_name2">name2</code></td>
<td>
<p>The text for the title of the results that refer to
the outcome or model under <code>full2</code>.</p>
</td></tr>
<tr><td><code id="league_heatmap_pred_+3A_show">show</code></td>
<td>
<p>A vector of at least three character strings that refer to the
names of the interventions <em>exactly</em> as defined in <code>drug_names1</code>.
Then, the league table will be created for these interventions only.
If <code>show</code> is not defined, the league table will present all
interventions as defined in <code>drug_names1</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>heatmap_league</code> offers the following options to display
<b>one</b> estimated effect measure for all (or some) pairwise comparisons:
</p>

<ul>
<li><p> one outcome, with results in the lower triangle referring to
comparisons in the opposite direction after converting negative values
into positive values (in absolute or logarithmic scale), and vice versa.
Darker shades of red and green correspond to larger treatment effects in
the upper and lower triangle, respectively, for a beneficial outcome, and
vice versa for a harmful outcome. Comparisons between interventions should
be read from left to right. Therefore, each cell refers to the
corresponding row-defining intervention against the column-defining
intervention. Results that indicate strong evidence in favour of the
row-defining intervention (i.e. the respective 95% prediction interval does
not include the null value) are indicated in bold. A message is printed on
the R console on how to read the heatmap;
</p>
</li>
<li><p> two outcomes for the same model, namely, network meta-analysis (via
<code><a href="#topic+run_model">run_model</a></code>) or network meta-regression (via
<code><a href="#topic+run_metareg">run_metareg</a></code>).
When one of the outcomes includes more interventions, the argument
<code>full1</code> should be considered for that outcome.
Comparisons between interventions should be read as follows: for the upper
diagonal, each cell refers to the corresponding row-defining intervention
against the column-defining intervention, and for the lower diagonal, each
cell refers to the corresponding column-defining intervention against the
row-defining intervention. Results that indicate strong evidence (i.e. the
respective 95% prediction interval does not include the null value) are
indicated in bold. A message is printed on the R console on how to read
the heatmap;
</p>
</li>
<li><p> two models for the same outcome, namely, network meta-analysis
versus network meta-regression. The instructions to read the heatmap are
in line with the previous point. A message is printed on the R console on
how to read the heatmap.
</p>
</li></ul>

<p>The function displays the effect measure as inherited by the argument
<code>full1</code>. For binary outcome, it can display the odds ratio,
relative risk, and risk difference. See 'Details' in
<code><a href="#topic+run_model">run_model</a></code> for the relative risk, and risk difference.
For continuous outcome, it can display the mean difference, standardised
mean difference, and ratio of means. Odds ratios, relative risk and ratio
of means are reported in the original scale after exponentiation of the
logarithmic scale.
</p>
<p>The rows and columns of the heatmap display the names of interventions
which are sorted by decreasing order from the best to the worst based on
their SUCRA value (Salanti et al., 2011) for the outcome or model under the
argument <code>full1</code>. The off-diagonals contain the posterior median and
95% prediction interval of the effect measure (according to the argument
<code>measure</code> as inherited in the argument <code>full1</code>) of the
corresponding comparisons.
</p>
<p>The main diagonal contains the SUCRA values of the corresponding
interventions when the argument <code>full1</code> refers to the
<code><a href="#topic+run_model">run_model</a></code> function.
When the argument <code>full1</code> refers to the <code><a href="#topic+run_metareg">run_metareg</a></code>
function, the p-score (Ruecker and Schwarzer, 2015) is calculated for each
intervention while taking into account the covariate value in
the argument <code>cov_value</code>. P-score is the 'frequentist analogue to
SUCRA' (Ruecker and Schwarzer, 2015).
</p>
<p>In the case of network meta-regression, when the covariate is binary,
specify in the second element of <code>cov_value</code> the name of the level for
which the heatmap will be created.
</p>
<p><code>league_heatmap_pred</code> can be used only for a network of interventions.
In the case of two interventions, the execution of the function will be
stopped and an error message will be printed on the R console. Similarly,
when the function is executed for a fixed-effect network meta-analysis or
network meta-regression.
</p>


<h3>Value</h3>

<p>A league heatmap of the posterior median and 95% prediction interval
of the effect measure (according to the argument <code>measure</code> defined in
<code><a href="#topic+run_model">run_model</a></code>) for all possible comparisons in the off-diagonals,
and the posterior mean of the SUCRA values in the diagonal.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli, Chrysostomos Kalyvas, Katerina Papadimitropoulou
</p>


<h3>References</h3>

<p>Ruecker G, Schwarzer G. Ranking treatments in frequentist network
meta-analysis works without resampling methods.
<em>BMC Med Res Methodol</em> 2015;<b>15</b>:58.
doi: 10.1186/s12874-015-0060-8
</p>
<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.liu2013")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_liu.rds', package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "pramipexole", "serotonin-norepinephrine
                  reuptake inhibitor", "serotonin reuptake inhibitor",
                  "tricyclic antidepressant", "pergolide")

# Create the league heatmap
league_heatmap_pred(full1 = res,
                    drug_names1 = interv_names)

</code></pre>

<hr>
<h2 id='league_table_absolute'>League table for relative and absolute effects</h2><span id='topic+league_table_absolute'></span>

<h3>Description</h3>

<p>Provides a league table of the estimated odds ratio, and risk difference
per 1000 participants for all possible comparisons of interventions in the
network. The main diagonal of the table presents the absolute risk for each
intervention in the network. <code>league_table_absolute</code> can be used for a
random-effects or fixed-effect network meta-analysis. This function should
be used when the user has access to the raw trial-level data
(one-trial-per-row format with arm-level data).
<code>league_table_absolute</code> is applied for one binary outcome only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>league_table_absolute(full, drug_names, show = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="league_table_absolute_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="league_table_absolute_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="league_table_absolute_+3A_show">show</code></td>
<td>
<p>A vector of at least three character strings that refer to the
names of the interventions <em>exactly</em> as defined in <code>drug_names</code>.
Then, the league table will be created for these interventions only.
If <code>show</code> is not defined, the league table will present all
interventions as defined in <code>drug_names</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user must define the argument <code>measure = "RD"</code> in
<code><a href="#topic+run_model">run_model</a></code>; otherwise, the function will be stopped and an
error message will be printed in the R console.
</p>
<p>The rows and columns of the league table display the names of the
interventions  sorted by decreasing order from the best to the worst
based on their SUCRA value (Salanti et al., 2011) for the odds ratio. The
upper off-diagonals contain the posterior median and 95% credible interval
of the odds ratio, the lower off-diagonals contain the posterior median and
95% credible interval of the risk difference (per 1000 participants), and
the main diagonal comprises the posterior median and 95% credible interval
of the absolute risks (per 1000 participants) of the  corresponding
interventions. The reference intervention of the network (which the
baseline risk has been selected for) is indicated in the main diagonal with
a black, thick frame.
</p>
<p>Comparisons between interventions should be read from left to right.
Results that indicate strong evidence in favor of the row-defining
intervention (i.e. the respective 95% credible interval does not include
the null value) are indicated in bold.
</p>
<p>To obtain unique absolute risks for each intervention, the network
meta-analysis model has been extended to incorporate the transitive risks
framework, namely, an intervention has the same absolute risk regardless of
the comparator intervention(s) in a trial (Spineli et al., 2017).
The absolute risks are a function of the odds ratio (the <strong>base-case</strong>
effect measure for a binary outcome) and the selected baseline risk for the
reference intervention (Appendix in Dias et al., 2013). See 'Arguments' in
<code><a href="#topic+run_model">run_model</a></code>. We advocate using the odds ratio as an effect
measure for its desired mathematical properties. Then, the risk difference
can be obtained as a function of the absolute risks of the corresponding
interventions in the comparison of interest.
</p>
<p><code>league_table_absolute</code> can be used only for a network of
interventions. In the case of two interventions, the execution of the
function will be stopped and an error message will be printed in the R
console.
</p>


<h3>Value</h3>

<p>A league table showing the posterior estimate and 95% credible
interval of the odds ratio (upper off-diagonals), risk difference per 1000
participants (lower off-diagonals), and absolute risks per 1000
participants (main diagonal).
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>
<p>Spineli LM, Brignardello-Petersen R, Heen AF, Achille F, Brandt L,
Guyatt GH, et al. Obtaining absolute effect estimates to facilitate shared
decision making in the context of multiple-treatment comparisons.
Abstracts of the Global Evidence Summit, Cape Town, South Africa.
<em>Cochrane Database of Systematic Reviews</em> 2017;<b>9</b>(Suppl 1):1891.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>

<hr>
<h2 id='league_table_absolute_user'>League table for relative and absolute effects (user defined)</h2><span id='topic+league_table_absolute_user'></span>

<h3>Description</h3>

<p>In line with <code><a href="#topic+league_table_absolute">league_table_absolute</a></code>, provides a league table
of the estimated odds ratio, and risk difference per 1000 participants for
all possible comparisons of interventions in the network.
The main diagonal of the table presents the absolute risk for each
intervention in the network. <code>league_table_absolute_user</code> requires
users to input the summary effect and 95% credible or confidence interval
of the basic parameters in the reported effect measure. This function
should be used when the user has access to the results of a published
systematic review rather than the raw trial-level data. In the latter case,
the user should consider the function <code><a href="#topic+league_table_absolute">league_table_absolute</a></code>.
<code>league_table_absolute_user</code> is applied for one binary outcome only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>league_table_absolute_user(
  data,
  measure,
  base_risk,
  drug_names,
  show = NULL,
  save_xls
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="league_table_absolute_user_+3A_data">data</code></td>
<td>
<p>A data-frame with the summary effects of comparisons with the
reference intervention of the network, known as basic parameters. The
data-frame has <code>T</code> rows (<code>T</code> is the number of interventions in
the network) and four columns that contain the point estimate, the lower
and upper bound of the 95% (confidence or credible) interval of the
corresponding basic parameters, and a ranking measure to indicate the order
of the interventions in the hierarchy from the best to the worst with
possible choices a non-zero positive integer for the rank, the SUCRA value
(Salanti et al., 2011) or p-score value (Ruecker and Schwarzer, 2015).
The first row of the data-frame refers to the selected reference
intervention and should include (1) the null value three times at the
investigated effect measure (i.e. 1 for odds ratio and relative risk, and 0
for risk difference), and (2) the value of the ranking measure.</p>
</td></tr>
<tr><td><code id="league_table_absolute_user_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure of <code>data</code>.
For a binary outcome, the following can be considered: <code>"OR"</code>,
<code>"RR"</code> or <code>"RD"</code> for the odds ratio, relative risk, and risk
difference, respectively.</p>
</td></tr>
<tr><td><code id="league_table_absolute_user_+3A_base_risk">base_risk</code></td>
<td>
<p>A number in the interval (0, 1) that indicates the baseline
risk for the selected reference intervention.</p>
</td></tr>
<tr><td><code id="league_table_absolute_user_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code>.
The first intervention should be the selected reference intervention.</p>
</td></tr>
<tr><td><code id="league_table_absolute_user_+3A_show">show</code></td>
<td>
<p>A vector of at least three character strings that refer to the
names of the interventions <em>exactly</em> as defined in <code>drug_names</code>.
Then, the league table will be created for these interventions only.
If <code>show</code> is not defined, the league table will present all
interventions as defined in <code>drug_names</code>.</p>
</td></tr>
<tr><td><code id="league_table_absolute_user_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) to the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the published results are reported in the relative risk scale
(i.e., <code>measure = "RR"</code>), the function calculates odds ratios and risk
differences (point estimate and 95% confidence interval) for all possible
pairwise comparisons in the network based on the obtained absolute risks
and the selected baseline risk. Likewise, when the published results are in
the odds ratio or risk difference scale (i.e., <code>measure = "OR"</code> or
<code>measure = "RD"</code>, respectively), the function calculates risk
differences or odds ratios (point estimate and 95% confidence interval),
respectively, for all possible pairwise comparisons in the network based on
the obtained absolute risks and the selected baseline risk.
</p>
<p>The rows and columns of the league table display the names of the
interventions  sorted by decreasing order from the best to the worst
based on the ranking measure in the fourth column of the argument
<code>data</code>. The upper off-diagonals contain the estimate and 95%
confidence interval of the odds ratio, the lower off-diagonals contain the
estimate and 95% confidence interval of the risk difference (per 1000
participants), and the main diagonal comprises the absolute risks and their
95% confidence interval (per 1000 participants) of the corresponding
non-reference interventions. The reference intervention of the network
(which the baseline risk has been selected for) is indicated in the main
diagonal with a black, thick frame.
</p>
<p>Comparisons between interventions should be read from left to right.
Results that indicate strong evidence in favour of the row-defining
intervention (i.e. the respective 95% confidence interval does not include
the null value) are indicated in bold.
</p>
<p>Furthermore, <code>league_table_absolute_user</code> exports
<code>table_relative_absolute_effect</code>, a table with the relative and
absolute effects of the basic parameters, as an 'xlsx' file (via the
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code> function) to the working
directory of the user.
</p>
<p>To obtain unique absolute risks for each intervention, we have considered
the transitive risks framework, namely, an intervention has the same
absolute risk regardless of the comparator intervention(s) in a trial
(Spineli et al., 2017).
</p>
<p><code>league_table_absolute_user</code> can be used only for a network of
interventions. In the case of two interventions, the execution of the
function will be stopped and an error message will be printed in the R
console.
</p>


<h3>Value</h3>

<p>A league table showing the estimate and 95% confidence interval of
the odds ratio (upper off-diagonals), risk difference per 1000
participants (lower off-diagonals), and absolute risks per 1000
participants (main diagonal).
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Ruecker G, Schwarzer G. Ranking treatments in frequentist network
meta-analysis works without resampling methods.
<em>BMC Med Res Methodol</em> 2015;<b>15</b>:58.
doi: 10.1186/s12874-015-0060-8
</p>
<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>
<p>Spineli LM, Brignardello-Petersen R, Heen AF, Achille F, Brandt L,
Guyatt GH, et al. Obtaining absolute effect estimates to facilitate shared
decision making in the context of multiple-treatment comparisons.
Abstracts of the Global Evidence Summit, Cape Town, South Africa.
<em>Cochrane Database of Systematic Reviews</em> 2017;<b>9</b>(Suppl 1):1891.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+league_table_absolute">league_table_absolute</a></code>,
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>

<hr>
<h2 id='leverage_plot'>Leverage plot</h2><span id='topic+leverage_plot'></span>

<h3>Description</h3>

<p>Plots the leverage against the square root of the
posterior mean of residual deviance of the trial-arms under the model of
interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leverage_plot(net, drug_names, title)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="leverage_plot_+3A_net">net</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, or <code><a href="#topic+run_ume">run_ume</a></code>.
See 'Value' in <code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>, or
<code><a href="#topic+run_ume">run_ume</a></code>.</p>
</td></tr>
<tr><td><code id="leverage_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>.
If <code>drug_names</code> is not defined, the order of the
interventions as they appear in <code>data</code> is used, instead.</p>
</td></tr>
<tr><td><code id="leverage_plot_+3A_title">title</code></td>
<td>
<p>A title to indicate the model (consistency model, network
meta-regression or unrelated mean effects model).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>leverage_plot</code> is integrated in the <code><a href="#topic+ume_plot">ume_plot</a></code>
function to create the leverage plot for the consistency model and the
unrelated mean effects model. These plots appear side-by-side in the output
of <code><a href="#topic+ume_plot">ume_plot</a></code>. Dias et al. (2010) used leverage plots to
investigate the fit of the consistency and inconsistency models&ndash;the
latter through the node-splitting approach.
</p>


<h3>Value</h3>

<p>A scatterplot of the leverage against the square root of the
posterior mean of residual deviance of the trial-arms under the model of
interest. The green, yellow, and red curves correspond to the parabola
<code class="reqn">x^2 + y = k</code> with <code class="reqn">k</code> = 1, 2, and 3, respectively. The data points
correspond to trial-arms. Data points found outside the yellow parabola are
linked with a pair of numbers. The first number refers to the position
of the trial in the dataset, and the second number refers to the
corresponding trial-arm (see 'Arguments' and 'Value' in
<code><a href="#topic+data_preparation">data_preparation</a></code>). These trial-arms contribute more than
1 to the deviance information criterion and, hence, the model's poor fit.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Caldwell DM, Ades AE. Checking consistency in mixed
treatment comparison meta-analysis.
<em>Stat Med</em> 2010;<b>29</b>(7-8):932&ndash;44. doi: 10.1002/sim.3767
</p>


<h3>See Also</h3>

<p><code><a href="#topic+data_preparation">data_preparation</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_ume">run_ume</a></code>, <code><a href="#topic+ume_plot">ume_plot</a></code>
</p>

<hr>
<h2 id='mcmc_diagnostics'>Markov Chain Monte Carlo diagnostics</h2><span id='topic+mcmc_diagnostics'></span>

<h3>Description</h3>

<p>Evaluates whether convergence has been achieved for the
monitored parameters of the Bayesian models. The Gelman-Rubin convergence
diagnostic, the Markov Chain Monte Carl (MCMC) error and relevant
diagnostic plots are applied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmc_diagnostics(net, par = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcmc_diagnostics_+3A_net">net</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>,
<code><a href="#topic+run_sensitivity">run_sensitivity</a></code>, <code><a href="#topic+run_series_meta">run_series_meta</a></code>, and
<code><a href="#topic+run_ume">run_ume</a></code>. See 'Value' in the functions above.</p>
</td></tr>
<tr><td><code id="mcmc_diagnostics_+3A_par">par</code></td>
<td>
<p>A vector of at least one character string that refers to the
monitored parameters in <code>jagsfit</code> which is an object of S3 class
<code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>, and
<code><a href="#topic+run_ume">run_ume</a></code>. The selected parameters will be considered in the
diagnostic plots (see 'Value'). This argument will be ignored for objects
of S3 class <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>, <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>,
and <code><a href="#topic+run_series_meta">run_series_meta</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each monitored parameter, <code>mcmc_diagnostics</code> considers the
R-hat and MCMC error and compares them with the thresholds 1.1 and 5% of
the posterior standard deviation (the rule of thumb), respectively.
Convergence is achieved for the monitored parameter, when the R-hat is
below the corresponding threshold. Visual inspection of the trace plots
and posterior density of the monitored parameters should also be considered
when drawing conclusions about convergence.
</p>


<h3>Value</h3>

<p><code>mcmc_diagnostics</code> considers the following monitored parameters:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The estimated summary effect measure.</p>
</td></tr>
<tr><td><code>EM_pred</code></td>
<td>
<p>The predicted summary effect measure.</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>The estimated trial-specific effect measure.</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation.</p>
</td></tr>
<tr><td><code>direct</code></td>
<td>
<p>The direct estimate of the split node (see 'Value' in
<code><a href="#topic+run_nodesplit">run_nodesplit</a></code>).</p>
</td></tr>
<tr><td><code>indirect</code></td>
<td>
<p>The indirect estimate of the split node
(see 'Value' in <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>).</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>The inconsistency factor of the split node (see 'Value' in
<code><a href="#topic+run_nodesplit">run_nodesplit</a></code>).</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>The informative missingness parameter.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The regression coefficient.</p>
</td></tr>
</table>
<p>For each monitored parameter mentioned above, <code>mcmc_diagnostics</code> also
returns a barplot on the ratio of MCMC error to the posterior standard
deviation and a barplot on the Gelman-Rubin R diagnostic. Bars that
correspond to a ratio less than 5% are indicated in green (the
corresponding parameters have been estimated accurately); otherwise, the
bars are indicated in red (inaccurate estimation). Furthermore, bars that
correspond to an R value less than 1.10 are indicated in green (the
corresponding parameters have been converged); otherwise, the bars are
indicated in red (convergence is not achieved).
<code>mcmc_diagnostics</code> returns histograms than barplots for <code>EM</code> when
<code><a href="#topic+run_sensitivity">run_sensitivity</a></code> is considered.
</p>
<p><code>mcmc_diagnostics</code> also uses the
<code><a href="mcmcplots.html#topic+mcmcplot">mcmcplot</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=mcmcplots">mcmcplots</a> to create an
HTML file with a panel of diagnostic plots (trace, density, and
autocorrelation) for each monitored parameter.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Gelman, A, Rubin, DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>


<h3>See Also</h3>

<p><code><a href="mcmcplots.html#topic+mcmcplot">mcmcplot</a></code>,
<code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_nodesplit">run_nodesplit</a></code>, <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>,
<code><a href="#topic+run_series_meta">run_series_meta</a></code>, <code><a href="#topic+run_ume">run_ume</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Read results from 'run_nodesplit' (using the default arguments)
res &lt;- readRDS(system.file('extdata/node_baker.rds', package = 'rnmamod'))

# Check convergence based on R-hat
mcmc_diagnostics(net = res,
                 par = c("tau", "EM[2,1]", "EM.pred[2,1]"))

</code></pre>

<hr>
<h2 id='metareg_plot'>End-user-ready results for network meta-regression</h2><span id='topic+metareg_plot'></span>

<h3>Description</h3>

<p>Illustrates the effect estimates, predictions and regression
coefficients of comparisons with a specified comparator intervention for a
selected covariate value and also exports these results to an Excel file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metareg_plot(full, reg, compar, cov_value, drug_names, save_xls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metareg_plot_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="metareg_plot_+3A_reg">reg</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_metareg">run_metareg</a></code>. See 'Value' in
<code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code id="metareg_plot_+3A_compar">compar</code></td>
<td>
<p>A character to indicate the comparator intervention. It must be
any name found in <code>drug_names</code>.</p>
</td></tr>
<tr><td><code id="metareg_plot_+3A_cov_value">cov_value</code></td>
<td>
<p>A list of two elements in the following order: a number
for the covariate value of interest (see 'Arguments' in
<code><a href="#topic+run_metareg">run_metareg</a></code>), and a character to indicate the name of
the covariate. See also 'Details'.</p>
</td></tr>
<tr><td><code id="metareg_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
<tr><td><code id="metareg_plot_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) at the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The deviance information criterion (DIC) of the network
meta-analysis model is compared with the DIC of the network meta-regression
model. If the difference in DIC exceeds 5, the network meta-regression
model is preferred; if the difference in DIC is less than -5, the network
meta-analysis model is preferred; otherwise, there is little to choose
between the compared models.
</p>
<p>When the covariate is binary, specify in the second element of
<code>cov_value</code> the name of the level for which the output will be
created.
</p>
<p>Furthermore, <code>metareg_plot</code> exports all tabulated results to separate
'xlsx' files (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code> function
of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) to the working
directory of the user.
</p>
<p><code>metareg_plot</code> can be used only for a network of interventions. In the
case of two interventions, the execution of the function will be stopped
and an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p><code>metareg_plot</code> prints on the R console a message on the most
parsimonious model (if any) based on the DIC (in red text). Furthermore,
the function returns the following list of elements:
</p>
<table>
<tr><td><code>table_estimates</code></td>
<td>
<p>The posterior median, and 95% credible interval
of the summary effect measure (according to the argument <code>measure</code>
defined in <code><a href="#topic+run_model">run_model</a></code>) for each comparison with the selected
intervention under network meta-analysis and network meta-regression
based on the specified <code>cov_value</code>.</p>
</td></tr>
<tr><td><code>table_predictions</code></td>
<td>
<p>The posterior median, and 95% prediction
interval of the summary effect measure (according to the argument
<code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>) for each comparison
with the selected intervention under network meta-analysis and network
meta-regression based on the specified <code>cov_value</code>.</p>
</td></tr>
<tr><td><code>table_model_assessment</code></td>
<td>
<p>The DIC, total residual deviance,
number of effective parameters, and the posterior median and 95% credible
interval of between-trial standard deviation (<em>tau</em>) under each model
(Spiegelhalter et al., 2002). When a fixed-effect model has been
performed, <code>metareg_plot</code> does not return results on <em>tau</em>. For a
binary outcome, the results refer to the odds ratio scale.</p>
</td></tr>
<tr><td><code>table_regression_coeffients</code></td>
<td>
<p>The posterior median and 95%
credible interval of the regression coefficient(s) (according to the
argument <code>covar_assumption</code> defined in <code><a href="#topic+run_metareg">run_metareg</a></code>).
For a binary outcome, the results refer to the odds ratio scale.</p>
</td></tr>
<tr><td><code>interval_plot</code></td>
<td>
<p>A forest plot on the estimated and predicted effect
sizes of comparisons with the selected comparator intervention under
network meta-analysis and network meta-regression based on the specified
<code>cov_value</code> alongside a forest plot with the corresponding SUCRA
values. See 'Details' and 'Value' in <code><a href="#topic+forestplot_metareg">forestplot_metareg</a></code>.</p>
</td></tr>
<tr><td><code>sucra_scatterplot</code></td>
<td>
<p>A scatterplot of the SUCRA values from the
network meta-analysis against the SUCRA values from the network
meta-regression based on the specified <code>cov_value</code>. See 'Details'
and 'Value' in <code><a href="#topic+scatterplot_sucra">scatterplot_sucra</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>
<p>Spiegelhalter DJ, Best NG, Carlin BP, van der Linde A. Bayesian measures of
model complexity and fit. <em>J R Stat Soc B</em> 2002;<b>64</b>(4):583&ndash;616.
doi: 10.1111/1467-9868.00353
</p>


<h3>See Also</h3>

<p><code><a href="#topic+forestplot_metareg">forestplot_metareg</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+scatterplot_sucra">scatterplot_sucra</a></code>,
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")


# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_baker.rds', package = 'rnmamod'))

# Read results from 'run_metareg' (exchangeable structure)
reg &lt;- readRDS(system.file('extdata/reg_baker.rds', package = 'rnmamod'))

# Publication year as the covariate
pub_year &lt;- c(1996, 1998, 1999, 2000, 2000, 2001, rep(2002, 5), 2003, 2003,
              rep(2005, 4), 2006, 2006, 2007, 2007)

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "budesonide", "budesonide plus formoterol",
                  "fluticasone", "fluticasone plus salmeterol",
                  "formoterol", "salmeterol", "tiotropium")

# Plot the results from both models for all comparisons with salmeterol and
# publication year 2000
metareg_plot(full = res,
             reg = reg,
             compar = "salmeterol",
             cov_value = list(2000, "publication year"),
             drug_names = interv_names)


</code></pre>

<hr>
<h2 id='miss_characteristics'>Visualising missing data in characteristics
(Comparisons' comparability for transitivity evaluation)</h2><span id='topic+miss_characteristics'></span>

<h3>Description</h3>

<p><code>miss_characteristics</code> hosts a set of visualisation tools to assess
the size and pattern of missing characteristics values in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>miss_characteristics(
  input,
  drug_names,
  rename_char = NULL,
  label_size = 4,
  axis_title_size = 14,
  axis_text_size = 14,
  axis_x_text_angle = 0,
  legend_text_size = 14,
  legend_title_size = 14,
  strip_text_size = 14,
  strip_text_angle = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="miss_characteristics_+3A_input">input</code></td>
<td>
<p>A data-frame in the long arm-based format. Two-arm trials occupy
one row in the data-frame. Multi-arm trials occupy as many rows as the
number of possible comparisons among the interventions. The first two
columns refer to the trial name, and the pairwise comparison,
respectively. The remaining columns refer to summary characteristics. See
'Details' for the specification of the columns.</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions
in the order they appear in the argument <code>input</code>.</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_rename_char">rename_char</code></td>
<td>
<p>A list of two elements: (i) a numeric vector with the
position of the characteristics in <code>input</code>, and (ii) a character
vector with the names of the characteristics, as they are wished to
appear in the title of the plots. This argument is optional, in case the
user wants to control the appearance of the titles.</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_label_size">label_size</code></td>
<td>
<p>A positive integer for the font size of labels in the
plots. <code>label_size</code> determines the size argument found in the geom's
aesthetic properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A positive integer for the font size of axis titles in
the plots. <code>axis_title_size</code> determines the axis.title argument found
in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A positive integer for the font size of axis text in
the plots. <code>axis_text_size</code> determines the axis.text argument found in
the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_axis_x_text_angle">axis_x_text_angle</code></td>
<td>
<p>A positive integer for the angle of axis text in
plots related to missing data. <code>axis_text_angle</code> determines the
axis.text.x argument found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A positive integer for the font size of legend text
in the plots. <code>legend_text_size</code> determines the legend.text argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>A positive integer for the font size of legend title
in the plots. <code>legend_title_size</code> determines the legend.title argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_strip_text_size">strip_text_size</code></td>
<td>
<p>A positive integer for the font size of strip text
in the plots. <code>strip_text_size</code> determines the strip.text argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
<tr><td><code id="miss_characteristics_+3A_strip_text_angle">strip_text_angle</code></td>
<td>
<p>A positive integer for the angle of strip text
in the plots. <code>strip_text_angle</code> determines the strip.text argument
found in the theme's properties in the R-package
<a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The correct type mode of columns in <code>input</code> must be ensured to use
the function <code>miss_characteristics</code>. The first two columns referring
to the trial name, and pairwise comparison, respectively, must be
<strong>character</strong>. The remaining columns referring to the characteristics
must be <strong>double</strong> or <strong>integer</strong> depending on whether the
corresponding characteristic refers to a quantitative or qualitative
variable. The type mode of each column is assessed by
<code>miss_characteristics</code> using the base function <code>typeof</code>.
</p>
<p>The interventions should be sorted in an ascending order of their
identifier number within the trials so that the first treatment column
(second column in <code>input</code>) is the control arm for every pairwise
comparison. This is important to ensure consistency in the order of
interventions within the comparisons obtained from the other related
functions.
</p>


<h3>Value</h3>

<p><code>miss_characteristics</code> returns the following list of elements:
</p>
<table>
<tr><td><code>Barplot_missing_combined</code></td>
<td>
<p>A panel of barplots on the percentage of
missing and observed cases for each comparison and characteristic.</p>
</td></tr>
<tr><td><code>Barplot_missing_characteristics</code></td>
<td>
<p>A barplot on the percentage of
missing and observed cases for each comparison.</p>
</td></tr>
<tr><td><code>Tileplot_missing</code></td>
<td>
<p>A plot that illustrates the position of missing
cases for each trial, comparison and characteristic.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Fictional dataset
set.seed(13022024)
data_set &lt;- data.frame(Trial_name = as.character(1:(5 + 7 +2)),
                       arm1 = rep(c("1", "2"), c(5 + 7, 2)),
                       arm2 = rep(c("2", "3"), c(5, 7 + 2)),
                       sample = as.numeric(sample(c(50:300), 5 + 7 + 2)),
                       age = as.numeric(sample(c(18:50), 5 + 7 + 2)),
                       blinding = factor(rep(c("yes", "no", "yes"), c(5, 7, 2))))
data_set$age[data_set$age &lt; 30] &lt;- NA

miss_characteristics(input = data_set,
                      drug_names = c("A", "B", "C"))


</code></pre>

<hr>
<h2 id='missingness_param_prior'>Define the mean value of the normal distribution of the missingness parameter</h2><span id='topic+missingness_param_prior'></span>

<h3>Description</h3>

<p>Generates the mean value of the normal distribution of the missingness
parameter in the proper format depending on the assumed structure of the
missingness parameter. <code><a href="#topic+run_model">run_model</a></code> inherits
<code>missingness_param_prior</code> through the argument <code>mean_misspar</code>
(see 'Argument' in <code><a href="#topic+run_model">run_model</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>missingness_param_prior(assumption, mean_misspar)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="missingness_param_prior_+3A_assumption">assumption</code></td>
<td>
<p>Character string indicating the structure of the
informative missingness parameter.
Set <code>assumption</code> equal to one of the following: <code>"HIE-COMMON"</code>,
<code>"HIE-TRIAL"</code>, <code>"HIE-ARM"</code>, <code>"IDE-COMMON"</code>,
<code>"IDE-TRIAL"</code>, <code>"IDE-ARM"</code>, <code>"IND-CORR"</code>, or
<code>"IND-UNCORR"</code>.
The default argument is <code>"IDE-ARM"</code>. The abbreviations <code>"IDE"</code>,
<code>"HIE"</code>, and <code>"IND"</code> stand for identical, hierarchical and
independent, respectively. <code>"CORR"</code> and <code>"UNCORR"</code> stand for
correlated and uncorrelated, respectively.</p>
</td></tr>
<tr><td><code id="missingness_param_prior_+3A_mean_misspar">mean_misspar</code></td>
<td>
<p>A numeric value or a vector of two numeric values for the
mean of the normal distribution of the informative missingness parameter
(see 'Details'). The default argument is 0 and corresponds to the
missing-at-random assumption for <code>assumption = "IDE-ARM"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+run_model">run_model</a></code> considers the informative missingness odds
ratio in the logarithmic scale for binary outcome data (Spineli, 2019a;
Turner et al., 2015; White et al., 2008), the informative missingness
difference of means when <code>measure</code> is <code>"MD"</code> or <code>"SMD"</code>,
and the informative missingness ratio of means in the logarithmic scale
when <code>measure</code> is <code>"ROM"</code> (Spineli et al., 2021;
Mavridis et al., 2015).
</p>
<p>When <code>assumption</code> is trial-specific (i.e., <code>"IDE-TRIAL"</code> or
<code>"HIE-TRIAL"</code>), or independent (i.e., <code>"IND-CORR"</code> or
<code>"IND-UNCORR"</code>), only one numeric value can be assigned to
<code>mean_misspar</code> because the same missingness scenario is applied to all
trials and trial-arms of the dataset, respectively. When <code>assumption</code>
is <code>"IDE-ARM"</code> or <code>"HIE-ARM"</code>, a maximum of two
<em>different or identical</em> numeric values can be assigned as a vector to
<code>mean_misspars</code>: the first value refers to the experimental arm,
and the second value refers to the control arm of a trial.
In the case of a network, the first value is considered for all
non-reference interventions and the second value is considered for the
reference intervention of the network (see 'Argument' <code>ref</code> in
<code><a href="#topic+run_model">run_model</a></code>). This is necessary to ensure transitivity in the
assumptions for the missingness parameter across the comparisons in the
network (Spineli, 2019b).
</p>
<p>Currently, there are no empirically-based prior distributions for the
informative missingness parameters. The users may refer to
Mavridis et al. (2015) and Spineli (2019) to determine <code>mean_misspar</code>
for an informative missingness parameter.
</p>


<h3>Value</h3>

<p>A scalar or numeric vector to be passed to <code><a href="#topic+run_model">run_model</a></code>.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Mavridis D, White IR, Higgins JP, Cipriani A, Salanti G. Allowing for
uncertainty due to missing continuous outcome data in pairwise and
network meta-analysis. <em>Stat Med</em> 2015;<b>34</b>(5):721&ndash;41.
doi: 10.1002/sim.6365
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019a;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>
<p>Spineli LM. Modeling missing binary outcome data while preserving
transitivity assumption yielded more credible network meta-analysis
results. <em>J Clin Epidemiol</em> 2019b;<b>105</b>:19&ndash;26.
doi: 10.1016/j.jclinepi.2018.09.002
</p>
<p>Turner NL, Dias S, Ades AE, Welton NJ. A Bayesian framework to account
for uncertainty due to missing binary outcome data in pairwise meta-analysis.
<em>Stat Med</em> 2015;<b>34</b>(12):2062&ndash;80. doi: 10.1002/sim.6475
</p>
<p>White IR, Higgins JP, Wood AM. Allowing for uncertainty due to missing data
in meta-analysis&ndash;part 1: two-stage methods. <em>Stat Med</em>
2008;<b>27</b>(5):711&ndash;27. doi: 10.1002/sim.3008
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>

<hr>
<h2 id='netplot'>Network plot</h2><span id='topic+netplot'></span>

<h3>Description</h3>

<p>Illustrates the network plot for one outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netplot(
  data,
  drug_names,
  show_multi = FALSE,
  multi_frame = -16,
  alpha_multi_color = 0.1,
  layout = igraph::layout_in_circle,
  node_color = "tomato",
  node_frame_color = "black",
  node_frame_width = 1,
  node_shape = "circle",
  node_label_color = "black",
  node_label_font = 1,
  node_label_cex = 1,
  node_label_dist = 0,
  edge_color = "grey50",
  edge_arrow_size = 0.5,
  edge_lty = 1,
  edge_label = NULL,
  edge_label_color = "black",
  edge_label_font = 1,
  edge_label_cex = 2,
  edge_curved = 0,
  direction = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netplot_+3A_data">data</code></td>
<td>
<p>A data-frame of a one-trial-per-row format containing arm-level
data of each trial. See 'Format' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="netplot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions
(nodes) in the order they appear in the argument <code>data</code>.</p>
</td></tr>
<tr><td><code id="netplot_+3A_show_multi">show_multi</code></td>
<td>
<p>Logical to indicate whether to colour the closed-loops
informed by multi-arm trials. The default is <code>show_multi = FALSE</code>.</p>
</td></tr>
<tr><td><code id="netplot_+3A_multi_frame">multi_frame</code></td>
<td>
<p>A numeric scalar to determine the size of the border
around the closed-loops formed by multi-arm trials. The default is -16.
<code>multi_frame</code> determines the mark.expand argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_alpha_multi_color">alpha_multi_color</code></td>
<td>
<p>A numerical scalar with range from 0 to 1 to
determine the opacity of <code>multi_frame</code> coloured using the
<code>rainbow</code> colour palette. The default is 0.1.</p>
</td></tr>
<tr><td><code id="netplot_+3A_layout">layout</code></td>
<td>
<p>The layout specification. The default is
<code>layout = layout_in_circle</code> to plot the nodes in a circular layout.
For more information refer to the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>,</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_color">node_color</code></td>
<td>
<p>A character or vector of characters (with length equal to
the number of nodes) to indicate the colour of the nodes. The default is
<code>node_color = "tomato"</code>. <code>node_color</code> determines the vertex.color
argument found in the <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_frame_color">node_frame_color</code></td>
<td>
<p>A character to indicate the colour of the frame
around the nodes. The default is <code>node_frame_color = "black"</code>.
<code>node_frame_color</code> determines the vertex.frame.color argument found in
the <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_frame_width">node_frame_width</code></td>
<td>
<p>A numerical scalar to indicate the width of the frame
around the nodes. The default is 1. <code>node_frame_width</code> determines the
vertex.frame.width argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_shape">node_shape</code></td>
<td>
<p>A character to indicate the shape of the nodes. The default
is <code>node_shape = "circle"</code>. <code>node_shape</code> determines the
vertex.shape argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_label_color">node_label_color</code></td>
<td>
<p>A character to indicate the color of the node labels.
The default is <code>node_label_color = "black"</code>. <code>node_label_color</code>
determines the vertex.label.color argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_label_font">node_label_font</code></td>
<td>
<p>A numerical scalar to indicate the font of the node
labels. The default is 1. <code>node_label_font</code> determines the
vertex.label.font argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_label_cex">node_label_cex</code></td>
<td>
<p>A numerical scalar to indicate the font size of the
node labels. The default is 1. <code>node_label_cex</code> determines the
vertex.label.cex argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_node_label_dist">node_label_dist</code></td>
<td>
<p>A numerical scale between 0 and 1 to indicate the
position of the node labels relative to the node center. The default is 0,
where the label is centered. <code>node_label_dist</code> determines the
vertex.label.dist argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_color">edge_color</code></td>
<td>
<p>A character or vector of characters (with length equal to
the number of edges) to indicate the colour of the edges. The default is
<code>edge_color = "grey50"</code>. <code>edge_color</code> determines the edge.color
argument found in the <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_arrow_size">edge_arrow_size</code></td>
<td>
<p>A numerical scalar between 0 and 1 to indicate the
arrow size. The default is 0.5. <code>edge_arrow_size</code> determines the
edge.arrow.size argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>). This argument
work only when <code>direction = FALSE</code>.</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_lty">edge_lty</code></td>
<td>
<p>A numerical scalar, discrete with values from 0 to 6 to
indicate the line type of the edges. The default is 1 (solid).
<code>edge_lty</code> determines the edge.lty argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_label">edge_label</code></td>
<td>
<p>A vector of number of characters with length equal to the
number of edges to present the edge label. The default is
<code>edge_label = NULL</code> and refers to the number of studies investigating
the corresponding comparisons. <code>edge_label</code> determines the edge.label
argument found in the <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_label_color">edge_label_color</code></td>
<td>
<p>A character to indicate the color of the edge labels.
The default is <code>edge_label_color = "black"</code>. <code>edge_label_color</code>
determines the edge.label.color argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_label_font">edge_label_font</code></td>
<td>
<p>A numerical scalar to indicate the font of the edge
labels. The default is 1. <code>edge_label_font</code> determines the
edge.label.font argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_label_cex">edge_label_cex</code></td>
<td>
<p>A numerical scalar to indicate the font size of the
edge labels. The default is 2. <code>edge_label_cex</code> determines the
edge.label.cex argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_edge_curved">edge_curved</code></td>
<td>
<p>A numerical scalar with range from 0 to 1 that indicates
the edge curvature. The default is 0 (no curvature). <code>edge_curved</code>
determines the edge.curved argument found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_direction">direction</code></td>
<td>
<p>Logical to draw (TRUE) or not (FALSE) arrow for each edge
according to each direction. The default is <code>direction = FALSE</code>. For
more information refer to the R-package
<a href="https://CRAN.R-project.org/package=igraph">igraph</a>).</p>
</td></tr>
<tr><td><code id="netplot_+3A_...">...</code></td>
<td>
<p>Further graphical arguments of the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The edge thickness is proportional to the number of
trials investigating the corresponding comparison. The node size is
weighted by the total sample size of the corresponding intervention.
</p>
<p>The user can control many of the arguments found in the
<code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> function of the
R-package <a href="https://CRAN.R-project.org/package=igraph">igraph</a>), such
as the colour of the nodes and edges, the node and edge label size, and so
on.
</p>


<h3>Value</h3>

<p>A network plot with coloured closed-loops informed by multi-arm
trials. Each node indicates an intervention and each edge an observed
pairwise comparison.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>,
<code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.bottomley2011")

# Return the first six trials of the dataset
head(nma.bottomley2011)

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("betamethasone dipropionate", "betamethasone valerate",
                  "calcipotriol", "calcipotriol plus polytar", "capasal",
                  "two-compound formulation gel", "placebo")

# Create the network plot
netplot(data = nma.bottomley2011,
        drug_names = interv_names,
        edge_label_cex = 1)

</code></pre>

<hr>
<h2 id='nma.baker2009'>Pharmacological interventions for chronic obstructive pulmonary disease</h2><span id='topic+nma.baker2009'></span>

<h3>Description</h3>

<p>A dataset of 21 trials comparing seven pharmacologic
interventions with each other and placebo in patients with chronic
obstructive pulmonary disease (COPD). The exacerbation of COPD is the
analysed binary outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nma.baker2009)
</code></pre>


<h3>Format</h3>

<p>A data-frame with 21 rows of arm-based data and 17 columns (maximum
number of 4 arms).
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows: 1, placebo; 2,
budesonide; 3, budesonide plus formoterol; 4, fluticasone; 5, fluticasone
plus salmeterol; 6, formoterol; 7, salmeterol; and 8, tiotropium.
</p>


<h3>Source</h3>

<p>Baker WL, Baker EL, Coleman CI. Pharmacologic treatments for chronic
obstructive pulmonary disease: a mixed-treatment comparison meta-analysis.
<em>Pharmacotherapy</em> 2009;<b>29</b>(8):891&ndash;905. doi: 10.1592/phco.29.8.891
</p>

<hr>
<h2 id='nma.bottomley2011'>Pharmacological interventions for moderately severe scalp psoriasis</h2><span id='topic+nma.bottomley2011'></span>

<h3>Description</h3>

<p>A dataset of 9 trials comparing six pharmacologic interventions
with each other and placebo for moderately severe scalp psoriasis.
The analysed binary outcome is the investigator's global assessment
response at 4 weeks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nma.bottomley2011)
</code></pre>


<h3>Format</h3>

<p>A data frame with 9 rows of arm-based data and 17 columns
(maximum number of 4 arms).
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows: 1, betamethasone
dipropionate; 2, betamethasone valerate; 3, calcipotriol; 4, calcipotriol
plus polytar; 5, capasal; 6, two-compound formulation gel; and 7, placebo.
</p>


<h3>Source</h3>

<p>Bottomley JM, Taylor RS, Ryttov J. The effectiveness of two-compound
formulation calcipotriol and betamethasone dipropionate gel in the
treatment of moderately severe scalp psoriasis: a systematic review of direct
and indirect evidence. <em>Curr Med Res Opin</em> 2011;<b>27</b>(1):251&ndash;68.
doi: 10.1185/03007995.2010.541022
</p>

<hr>
<h2 id='nma.dogliotti2014'>Oral antithrombotics for stroke episode</h2><span id='topic+nma.dogliotti2014'></span>

<h3>Description</h3>

<p>A dataset of 16 trials comparing seven oral antithrombotics with
each other and placebo in patients with atrial fibrillation. The analysed
binary outcome is prevention of a stroke episode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nma.dogliotti2014)
</code></pre>


<h3>Format</h3>

<p>A data frame with 16 rows of arm-based data and 13 columns (maximum
number of 3 arms).
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows: 1, placebo;
2, aspirin; 3, aspirin plus clopidogrel; 4, dabigatran 110 mg;
5, dabigatran 150 mg; 6, rivaroxaban; 7, vitamin K antagonist; and
8, apixaban
</p>


<h3>Source</h3>

<p>Dogliotti A, Paolasso E, Giugliano RP. Current and new oral antithrombotics
in non-valvular atrial fibrillation: a network meta-analysis
of 79808 patients. <em>Heart</em> 2014;<b>100</b>(5):396&ndash;405.
doi: 10.1136/heartjnl-2013-304347
</p>

<hr>
<h2 id='nma.liu2013'>Antidepressants in Parkinson's disease</h2><span id='topic+nma.liu2013'></span>

<h3>Description</h3>

<p>A dataset of 11 trials comparing the effectiveness of five
antidepressants and placebo in Parkinson's disease. The analysed binary
outcome is the number of patients with a reduction of at least 50/
the baseline score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nma.liu2013)
</code></pre>


<h3>Format</h3>

<p>A data frame with 11 rows of arm-based data and 13 columns (maximum
number of 3 arms).
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows: 1, placebo;
2, pramipexole; 3, serotonin-norepinephrine reuptake inhibitor;
4, selective serotonin reuptake inhibitor; 5, tricyclic antidepressant; and
6, pergolide.
</p>


<h3>Source</h3>

<p>Liu J, Dong J, Wang L, Yan P, Sun S. Comparative efficacy and acceptability
of antidepressants in Parkinson's disease: a network meta-analysis.
<em>PLoS One</em> 2013;<b>8</b>(10):e76651.
doi: 10.1371/journal.pone.0076651
</p>

<hr>
<h2 id='nma.schwingshackl2014'>Training modalities for patients with type 2 diabetes</h2><span id='topic+nma.schwingshackl2014'></span>

<h3>Description</h3>

<p>A dataset of 14 trials comparing three different training
modalities (a triangle network) for patients with type 2 diabetes.
The analysed continuous outcome is change from baseline in HbA1c levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nma.schwingshackl2014)
</code></pre>


<h3>Format</h3>

<p>A data frame with 14 rows of arm-based data and 16 columns (maximum
number of 3 arms).
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows: 1, aerobic;
2, resistance; and 3, combined training;
</p>


<h3>Source</h3>

<p>Schwingshackl L, Missbach B, Dias S, et al. M Impact of different training
modalities on glycaemic control and blood lipids in patients with type 2
diabetes: a systematic review and network meta-analysis.
<em>Diabetologia</em> 2014;<b>57</b>(9):1789&ndash;97.
doi: 10.1007/s00125-014-3303-z
</p>

<hr>
<h2 id='nma.stowe2011'>Antiparkinsonian interventions for later Parkinson's disease</h2><span id='topic+nma.stowe2011'></span>

<h3>Description</h3>

<p>A dataset of 29 trials comparing three antiparkinsonian
interventions with placebo (a star-shaped network) in patients with later
Parkinson's disease. The analysed continuous outcome is the change from
baseline of patient off-time reduction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nma.stowe2011)
</code></pre>


<h3>Format</h3>

<p>A data frame with 29 rows of arm-based data and 11 columns (maximum
number of 2 arms).
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows: 1, placebo plus
levodopa (PBO+LD); 2, dopamine agonist plus levodopa (DA+LD);
3, catechol-O-methyl transferase inhibitors plus levodopa (COMBI+LD); and
4, monoamine oxidase type B inhibitors plus levodopa (MAOBI+LD).
</p>


<h3>Source</h3>

<p>Stowe R, Ives N, Clarke CE, et al. Meta-analysis of the comparative efficacy
and safety of adjuvant treatment to levodopa in later Parkinson's disease.
<em>Mov Disord</em> 2011;<b>26</b>(4):587&ndash;98. doi: 10.1002/mds.23517
</p>

<hr>
<h2 id='nodesplit_plot'>End-user-ready results for the node-splitting approach</h2><span id='topic+nodesplit_plot'></span>

<h3>Description</h3>

<p><code>nodesplit_plot</code> hosts a toolkit of functions that
facilitates the comparison of the consistency model
(via <code><a href="#topic+run_model">run_model</a></code>) with the node-splitting approach
(via <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>) regarding the posterior summaries of the
direct and indirect effects and inconsistency factor of the split
nodes, the between-trial standard deviation and model assessment
parameters (Spiegelhalter et al., 2002) after each split node in the
network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nodesplit_plot(full, node, drug_names, save_xls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nodesplit_plot_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="nodesplit_plot_+3A_node">node</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>. See 'Value'
in <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>.</p>
</td></tr>
<tr><td><code id="nodesplit_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
<tr><td><code id="nodesplit_plot_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) at the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>intervalplot_inconsistency_factor</code> includes as many interval
plots as the number of split nodes in the network. Each interval plot
illustrates the posterior median and 95% credible interval of the direct and
indirect effect of the split nodes and the corresponding inconsistency
factor.
The line that corresponds to the inconsistency factor is highlighted with
green, when it does not cross the vertical line of no difference (between
the direct and indirect effect), and red otherwise. If there are more than
30 split nodes, the function presents the interval plots on split nodes
with conclusive inconsistency factor (green intervals) or those with
an opposite sign in the direct and indirect effects.
</p>
<p><code>intervalplot_tau</code> is an interval plot on the median and 95% credible
interval of <em>tau</em> after each split node. The lines that correspond to
the split nodes are sorted in ascending order of the deviance information
criterion (DIC) which appears at the top of each line.
The estimated median and 95% credible intervals of <em>tau</em> under the
consistency model appear in the interval plot as a solid and two dotted
parallel blue lines, respectively. The different levels of heterogeneity
appear as green, yellow, orange, and red rectangulars to indicate a low,
reasonable, fairly high, and fairly extreme heterogeneity, respectively,
following the classification of Spiegelhalter et al. (2004).
When a fixed-effect model has been performed, <code>nodesplit_plot</code> does
not return the <code>intervalplot_tau</code>.
</p>
<p><code>table_model_assessment</code> also includes the column
<em>DIC-based better fit</em> that indicates the preferred model in terms of
parsimony for each split node. Therefore, the DIC of the model after each
split node is compared with the DIC of the consistency model
(Dias et al., 2010). If the difference in DIC exceeds 5, the consistency
model is preferred; if the difference in DIC is less than -5, the model
after the split node is preferred; otherwise, there is little to choose
between the compared models.
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>nodesplit_plot</code>
currently presents the results in the odds ratio scale. This is because,
the odds ratio is used as the 'best-case' effect measure in
<code><a href="#topic+run_model">run_model</a></code>. Then, relative risk, and risk difference are
obtained as a function of the odds ratio and the selected baseline risk
(See 'Details' in <code><a href="#topic+run_model">run_model</a></code>).
</p>
<p>The split nodes have been automatically selected via the
<code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=gemtc">gemtc</a>.
See 'Details' in <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>.
</p>
<p>Furthermore, <code>nodesplit_plot</code> exports both data-frames to separate
'xlsx' files (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code> function
of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) to the working
directory of the user.
</p>
<p><code>nodesplit_plot</code> can be used only for a network of interventions and
when there is at least one split node. Otherwise, the execution of the
function will be stopped and an error message will be printed on the R
console.
</p>


<h3>Value</h3>

<p><code>nodesplit_plot</code> returns the following list of elements:
</p>
<table>
<tr><td><code>table_effect_size</code></td>
<td>
<p>A data-frame with the posterior median,
posterior standard deviation and 95% credible interval of the direct and
indirect effect and the inconsistency factor of each split node.</p>
</td></tr>
<tr><td><code>table_model_assessment</code></td>
<td>
<p>A data-frame with the model assessment
parameters (DIC, posterior mean of total residual deviance, and number of
effective parameters), the posterior median, posterior standard deviation
and 95% credible interval of <em>tau</em> under the consistency model and
after each split node. See 'Details'.</p>
</td></tr>
<tr><td><code>intervalplot_inconsistency_factor</code></td>
<td>
<p>A panel of interval plots on
the direct and indirect effect of the split nodes and the corresponding
inconsistency factor. See 'Details'.</p>
</td></tr>
<tr><td><code>intervalplot_tau</code></td>
<td>
<p>An interval plot on <em>tau</em> after each
split node. See 'Details'.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Caldwell DM, Ades AE. Checking consistency in mixed
treatment comparison meta-analysis.
<em>Stat Med</em> 2010;<b>29</b>(7-8):932&ndash;44. doi: 10.1002/sim.3767
</p>
<p>Spiegelhalter DJ, Abrams KR, Myles JP. Bayesian approaches to clinical trials
and health-care evaluation. John Wiley and Sons, Chichester, 2004.
</p>
<p>Spiegelhalter DJ, Best NG, Carlin BP, van der Linde A. Bayesian measures of
model complexity and fit. <em>J R Stat Soc B</em> 2002;<b>64</b>(4):583&ndash;616.
doi: 10.1111/1467-9868.00353
</p>


<h3>See Also</h3>

<p><code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>,
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_baker.rds', package = 'rnmamod'))

# Read results from 'run_nodesplit' (using the default arguments)
node &lt;- readRDS(system.file('extdata/node_baker.rds', package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "budesonide", "budesonide plus formoterol",
                  "fluticasone", "fluticasone plus salmeterol",
                  "formoterol", "salmeterol", "tiotropium")

# Plot the results from both models
nodesplit_plot(full = res,
               node = node,
               drug_names = interv_names)

</code></pre>

<hr>
<h2 id='pma.hetrick2012'>Paroxetine versus placebo for depressive disorders</h2><span id='topic+pma.hetrick2012'></span>

<h3>Description</h3>

<p>A pairwise meta-analysis of 4 trials comparing paroxetine with
placebo for depressive disorders in children and adolescents.
The analysed binary outcome is remission or response as defined in the
trials.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pma.hetrick2012)
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 rows of arm-based data and 9 columns.
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows:
1, placebo; 2, paroxetine.
</p>


<h3>Source</h3>

<p>Hetrick SE, McKenzie JE, Cox GR, Simmons MB, Merry SN. Newer generation
antidepressants for depressive disorders in children and adolescents.
<em>Cochrane Database Syst Rev</em> 2012;<b>11</b>(11):CD004851.
doi: 10.1002/14651858.CD004851.pub3
</p>

<hr>
<h2 id='pma.taylor2004'>Inositol versus glucose for depressive episode</h2><span id='topic+pma.taylor2004'></span>

<h3>Description</h3>

<p>A pairwise meta-analysis of 4 trials comparing inositol with
glucose (placebo) for depressive episode. The analysed continuous outcome
is the resolution of a depressive episode using the Hamilton Depression
Rating Scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pma.taylor2004)
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 rows of arm-based data and 11 columns.
</p>


<h3>Details</h3>

<p>The interventions have been coded as follows:
1, placebo; 2, inositol.
</p>


<h3>Source</h3>

<p>Taylor MJ, Wilder H, Bhagwagar Z, Geddes J. Inositol for depressive
disorders. <em>Cochrane Database Syst Rev</em> 2004;<b>2004</b>(2):CD004049.
doi: 10.1002/14651858.CD004049.pub2
</p>

<hr>
<h2 id='prepare_model'>WinBUGS code for Bayesian pairwise or network meta-analysis and
meta-regression</h2><span id='topic+prepare_model'></span>

<h3>Description</h3>

<p>The WinBUGS code, as written by Dias et al. (2013) to run a one-stage
Bayesian network meta-analysis, extended to incorporate the pattern-mixture
model for binary or continuous missing participant outcome data (Spineli
et al., 2021; Spineli, 2019). The model has been also extended to
incorporate a trial-level covariate to apply meta-regression
(Cooper et al., 2009). In the case of two interventions, the code boils
down to a one-stage Bayesian pairwise meta-analysis with pattern-mixture
model (Turner et al., 2015; Spineli et al, 2021).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_model(measure, model, covar_assumption, assumption)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_model_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure. For a binary
outcome, the following can be considered: <code>"OR"</code>, <code>"RR"</code> or
<code>"RD"</code> for the odds ratio, relative risk, and risk difference,
respectively. For a continuous outcome, the following can be considered:
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for mean difference,
standardised mean difference and ratio of means, respectively.</p>
</td></tr>
<tr><td><code id="prepare_model_+3A_model">model</code></td>
<td>
<p>Character string indicating the analysis model with values
<code>"RE"</code>, or <code>"FE"</code> for the random-effects and fixed-effect model,
respectively. The default argument is <code>"RE"</code>.</p>
</td></tr>
<tr><td><code id="prepare_model_+3A_covar_assumption">covar_assumption</code></td>
<td>
<p>Character string indicating the structure of the
intervention-by-covariate interaction, as described in
Cooper et al., (2009). Set <code>covar_assumption</code> equal to one of the
following, when meta-regression is performed: <code>"exchangeable"</code>,
<code>"independent"</code>, and <code>"common"</code>. Assign <code>"NO"</code> to perform
pairwise or network meta-analysis.</p>
</td></tr>
<tr><td><code id="prepare_model_+3A_assumption">assumption</code></td>
<td>
<p>Character string indicating the structure of the
informative missingness parameter. Set <code>assumption</code> equal to one of
the following: <code>"HIE-COMMON"</code>, <code>"HIE-TRIAL"</code>, <code>"HIE-ARM"</code>,
<code>"IDE-COMMON"</code>, <code>"IDE-TRIAL"</code>, <code>"IDE-ARM"</code>,
<code>"IND-CORR"</code>, or <code>"IND-UNCORR"</code>. The default argument is
<code>"IDE-ARM"</code>. The abbreviations <code>"IDE"</code>, <code>"HIE"</code>, and
<code>"IND"</code> stand for identical, hierarchical and independent,
respectively. <code>"CORR"</code> and <code>"UNCORR"</code> stand for correlated and
uncorrelated, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>prepare_model</code> creates the model in the JAGS dialect
of the BUGS language. The output of this function constitutes the argument
<code>model.file</code> of the <code><a href="R2jags.html#topic+jags">jags</a></code> function (in the
R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>) via the
<code><a href="base.html#topic+textConnection">textConnection</a></code> function.
</p>


<h3>Value</h3>

<p>An R character vector object to be passed to <code><a href="#topic+run_model">run_model</a></code>
and <code><a href="#topic+run_metareg">run_metareg</a></code> through the
<code><a href="base.html#topic+textConnection">textConnection</a></code> function as the argument
<code>object</code>.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Cooper NJ, Sutton AJ, Morris D, Ades AE, Welton NJ. Addressing between-study
heterogeneity and inconsistency in mixed treatment comparisons: Application
to stroke prevention treatments in individuals with non-rheumatic atrial
fibrillation. <em>Stat Med</em> 2009;<b>28</b>(14):1861&ndash;81.
doi: 10.1002/sim.3594
</p>
<p>Dias S, Sutton AJ, Ades AE, Welton NJ. Evidence synthesis for decision
making 2: a generalized linear modeling framework for pairwise and network
meta-analysis of randomized controlled trials. <em>Med Decis Making</em>
2013;<b>33</b>(5):607&ndash;17. doi: 10.1177/0272989X12458724
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>
<p>Turner NL, Dias S, Ades AE, Welton NJ. A Bayesian framework to account
for uncertainty due to missing binary outcome data in pairwise meta-analysis.
<em>Stat Med</em> 2015;<b>34</b>(12):2062&ndash;80. doi: 10.1002/sim.6475
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_metareg">run_metareg</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="base.html#topic+textConnection">textConnection</a></code>
</p>

<hr>
<h2 id='prepare_nodesplit'>WinBUGS code for the node-splitting approach</h2><span id='topic+prepare_nodesplit'></span>

<h3>Description</h3>

<p>The WinBUGS code, as written by Dias et al. (2010) to run a
one-stage Bayesian node-splitting model, extended to incorporate the
pattern-mixture model for binary or continuous missing participant outcome
data (Spineli et al., 2021; Spineli, 2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_nodesplit(measure, model, assumption)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_nodesplit_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure. For a binary
outcome, the following can be considered: <code>"OR"</code>, <code>"RR"</code> or
<code>"RD"</code> for the odds ratio, relative risk, and risk difference,
respectively. For a continuous outcome, the following can be considered:
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for mean difference,
standardised mean difference and ratio of means, respectively.</p>
</td></tr>
<tr><td><code id="prepare_nodesplit_+3A_model">model</code></td>
<td>
<p>Character string indicating the analysis model with values
<code>"RE"</code>, or <code>"FE"</code> for the random-effects and fixed-effect model,
respectively. The default argument is <code>"RE"</code>.</p>
</td></tr>
<tr><td><code id="prepare_nodesplit_+3A_assumption">assumption</code></td>
<td>
<p>Character string indicating the structure of the
informative missingness parameter. Set <code>assumption</code> equal to one of
the following: <code>"HIE-COMMON"</code>, <code>"HIE-TRIAL"</code>, <code>"HIE-ARM"</code>,
<code>"IDE-COMMON"</code>, <code>"IDE-TRIAL"</code>, <code>"IDE-ARM"</code>,
<code>"IND-CORR"</code>, or <code>"IND-UNCORR"</code>. The default argument is
<code>"IDE-ARM"</code>. The abbreviations <code>"IDE"</code>, <code>"HIE"</code>, and
<code>"IND"</code> stand for identical, hierarchical and independent,
respectively. <code>"CORR"</code> and <code>"UNCORR"</code> stand for correlated and
uncorrelated, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions creates the model in the JAGS dialect of the BUGS
language. The output of this function constitutes the argument
<code>model.file</code> of <code><a href="R2jags.html#topic+jags">jags</a></code> (in the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>) via the
<code><a href="base.html#topic+textConnection">textConnection</a></code> function.
</p>
<p><code>prepare_nodesplit</code> inherits <code>measure</code>, <code>model</code>, and
<code>assumption</code> from the <code><a href="#topic+run_model">run_model</a></code> function. For a binary
outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>prepare_nodesplit</code>
currently considers the WinBUGS code for the odds ratio.
</p>
<p>The split nodes have been automatically selected via the
<code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=gemtc">gemtc</a>.
See 'Details' in <code><a href="#topic+run_nodesplit">run_nodesplit</a></code>.
</p>


<h3>Value</h3>

<p>An R character vector object to be passed to
<code><a href="#topic+run_nodesplit">run_nodesplit</a></code> through the
<code><a href="base.html#topic+textConnection">textConnection</a></code> function as the argument
<code>object</code>.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Caldwell DM, Ades AE. Checking consistency in mixed
treatment comparison meta-analysis.
<em>Stat Med</em> 2010;<b>29</b>(7-8):932&ndash;44. doi: 10.1002/sim.3767
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="#topic+run_model">run_model</a></code>,
<code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>,
<code><a href="#topic+run_nodesplit">run_nodesplit</a></code>,
<code><a href="base.html#topic+textConnection">textConnection</a></code>
</p>

<hr>
<h2 id='prepare_ume'>WinBUGS code for the unrelated mean effects model</h2><span id='topic+prepare_ume'></span>

<h3>Description</h3>

<p>The WinBUGS code, as proposed by Dias et al. (2013) to run a
one-stage Bayesian unrelated mean effects model, refined (Spineli, 2021),
and extended to incorporate the pattern-mixture model for binary or
continuous missing participant outcome data (Spineli et al., 2021; Spineli,
2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_ume(measure, model, assumption, connected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_ume_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure with values
<code>"OR"</code>, <code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for the odds ratio,
mean difference, standardised mean difference and ratio of means,
respectively.</p>
</td></tr>
<tr><td><code id="prepare_ume_+3A_model">model</code></td>
<td>
<p>Character string indicating the analysis model with values
<code>"RE"</code>, or <code>"FE"</code> for the random-effects and fixed-effect model,
respectively. The default argument is <code>"RE"</code>.</p>
</td></tr>
<tr><td><code id="prepare_ume_+3A_assumption">assumption</code></td>
<td>
<p>Character string indicating the structure of the
informative missingness parameter. Set <code>assumption</code> equal to one of
the following: <code>"HIE-COMMON"</code>, <code>"HIE-TRIAL"</code>, <code>"HIE-ARM"</code>,
<code>"IDE-COMMON"</code>, <code>"IDE-TRIAL"</code>, <code>"IDE-ARM"</code>,
<code>"IND-CORR"</code>, or <code>"IND-UNCORR"</code>. The default argument is
<code>"IDE-ARM"</code>. The abbreviations <code>"IDE"</code>, <code>"HIE"</code>, and
<code>"IND"</code> stand for identical, hierarchical and independent,
respectively. <code>"CORR"</code> and <code>"UNCORR"</code> stand for correlated and
uncorrelated, respectively.</p>
</td></tr>
<tr><td><code id="prepare_ume_+3A_connected">connected</code></td>
<td>
<p>An integer equal to one or larger that indicates the number
of subnetworks.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions creates the model in the JAGS dialect of the BUGS
language. The output of this function constitutes the argument
<code>model.file</code> of <code><a href="R2jags.html#topic+jags">jags</a></code> (in the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>) via the
<code><a href="base.html#topic+textConnection">textConnection</a></code> function.
</p>
<p><code>prepare_ume</code> inherits <code>measure</code>, <code>model</code>, and
<code>assumption</code> from the <code><a href="#topic+run_model">run_model</a></code> function. For a binary
outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>prepare_ume</code>
currently considers the WinBUGS code for the odds ratio.
</p>


<h3>Value</h3>

<p>An R character vector object to be passed to <code><a href="#topic+run_ume">run_ume</a></code>
through the <code><a href="base.html#topic+textConnection">textConnection</a></code> function as
the argument <code>object</code>.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Sutton AJ, Caldwell DM, Lu G, Ades AE. Evidence synthesis
for decision making 4: inconsistency in networks of evidence based on
randomized controlled trials.
<em>Med Decis Making</em> 2013;<b>33</b>(5):641&ndash;56.
doi: 10.1177/0272989X12455847
</p>
<p>Spineli LM. A revised framework to evaluate the consistency assumption
globally in a network of interventions. <em>Med Decis Making</em> 2021.
doi: 10.1177/0272989X211068005
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+jags">jags</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_ume">run_ume</a></code>, <code><a href="base.html#topic+textConnection">textConnection</a></code>
</p>

<hr>
<h2 id='rankosucra_plot'>Rankograms and SUCRA curves</h2><span id='topic+rankosucra_plot'></span>

<h3>Description</h3>

<p>It returns a panel of rankograms with integrated SUCRA curves
for each intervention in the network. The function can illustrate the
results of a single or two outcomes simultaneously.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankosucra_plot(
  full1,
  full2 = NULL,
  drug_names1,
  drug_names2 = NULL,
  name1 = NULL,
  name2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rankosucra_plot_+3A_full1">full1</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code> for network
meta-analysis. See 'Value' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="rankosucra_plot_+3A_full2">full2</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code> for network
meta-analysis of a second outcome. See 'Value' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="rankosucra_plot_+3A_drug_names1">drug_names1</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code> for <code>full1</code>.</p>
</td></tr>
<tr><td><code id="rankosucra_plot_+3A_drug_names2">drug_names2</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code> for <code>full2</code>. The elements must be a subset of
<code>drug_names1</code>.</p>
</td></tr>
<tr><td><code id="rankosucra_plot_+3A_name1">name1</code></td>
<td>
<p>The text for the title of the results that refer to
the outcome under <code>full1</code>.</p>
</td></tr>
<tr><td><code id="rankosucra_plot_+3A_name2">name2</code></td>
<td>
<p>The text for the title of the results that refer to
the outcome under <code>full2</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Interventions are sorted in the descending order of their SUCRA
value. The SUCRA value expressed in percentage appears on the top left
corner of each panel. In the case of two outcomes, the SUCRA values of the
outcome under the argument <code>full1</code> are considered to sort the
interventions from the best to the worst.
</p>
<p>When a second outcome is also considered, different colours are used to
draw the corresponding SUCRA curves and the rankograms: green for the
outcome under <code>full1</code>, and red for the outcome under <code>full2</code>.
</p>
<p><code>rankosucra_plot</code> can be used only for a network of interventions.
Otherwise, the execution of the function will be stopped and an error
message will be printed on the R console.
</p>


<h3>Value</h3>

<p>A panel of rankograms (yellowish bars) with integrated reddish SUCRA
curves for each intervention in the network (Salanti et al., 2011). The
x-axis of each panel refers to the ranking, and the y-axis refers to the
ranking probability expressed in percentage.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli, Chrysostomos Kalyvas, Katerina Papadimitropoulou
</p>


<h3>References</h3>

<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.liu2013")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_liu.rds', package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "pramipexole",
                  "serotonin-norepinephrine reuptake inhibitor",
                  "serotonin reuptake inhibitor",
                  "tricyclic antidepressant", "pergolide")

# Create the integrated rankograms and SUCRA curves
rankosucra_plot(full1 = res,
                drug_names1 = interv_names)

</code></pre>

<hr>
<h2 id='robustness_index'>Robustness index</h2><span id='topic+robustness_index'></span>

<h3>Description</h3>

<p>Calculates the robustness index, a novel index that quantifies the overall
divergence of the sensitivity analysis results from the primary analysis
results. The robustness index considers objective decision rules to infer
the presence or lack of robustness of the primary analysis results when
conducting a sensitivity analysis (Spineli et al., 2021).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robustness_index(sens, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="robustness_index_+3A_sens">sens</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_sensitivity">run_sensitivity</a></code> when
sensitivity analysis refers to different scenarios about the average
missingness parameter. See 'Value' in <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>. For a
<b>general</b> sensitivity analysis, insert a list of at least two objects
of S3 class <code><a href="#topic+run_model">run_model</a></code> indicating different re-analyses: the
first object (of class <code><a href="#topic+run_model">run_model</a></code>) in the list should refer to
the primary analysis.</p>
</td></tr>
<tr><td><code id="robustness_index_+3A_threshold">threshold</code></td>
<td>
<p>A number indicating the threshold of robustness, that is,
the minimally allowed deviation between the primary analysis and
re-analysis results. See 'Details' below.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Thresholds of robustness have been proposed only for the odds ratio
and standardised mean difference (Spineli et al., 2021).
The user may consider the values 0.28 and 0.17 in the argument
<code>threshold</code> for the odds ratio and standardised mean difference effect
measures (the default values), respectively, or consider other plausible
values. When the argument <code>threshold</code> has not been defined,
<code>robustness_index</code> considers the default values 0.28 and 0.17 as
threshold for robustness for binary and continuous outcome, respectively,
regardless of the effect measure (the default thresholds may not be proper
choices for other effect measures; hence, use these threshold with great
caution in this case). Spineli et al. (2021) offers a discussion on
specifying the <code>threshold</code> of robustness.
</p>
<p>In the case of binary outcome, <code>robustness_index</code> considers the
results in the odds ratio scale to calculate the robustness index.
This is because, the odds ratio is used as the 'best-case' effect measure
in <code><a href="#topic+run_model">run_model</a></code>. Then, relative risk, and risk difference are
functions of the odds ratio and the selected baseline risk (See 'Details'
in <code><a href="#topic+run_model">run_model</a></code>).
</p>
<p>In the case of missing participant outcome data, the primary analysis is
considered to be the middle of the numbers in the argument
<code>mean_scenarios</code> of <code><a href="#topic+run_sensitivity">run_sensitivity</a></code> (see 'Arguments'
and 'Details' in <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>).
</p>
<p>In <code>robust</code>, the value <code>"robust"</code> appears when
<code>robust_index</code> is less than <code>threshold</code>; otherwise, the value
<code>"frail"</code> appears.
</p>
<p>In the case of missing participant outcome data, <code>robustness_index</code>
can be used only when missing participant outcome data have been
extracted for at least one trial. Otherwise, the execution of the function
will be stopped and an error message will be printed in the R console.
</p>


<h3>Value</h3>

<p><code>robustness_index</code> prints on the R console a message in green
text on the threshold of robustness determined by the user.
Then, the function returns the following list of elements:
</p>
<table>
<tr><td><code>robust_index</code></td>
<td>
<p>A numeric scalar or vector on the robustness
index values. In the case of a pairwise meta-analysis,
<code>robust_index</code> is scalar as only one summary effect size is obtained.
In the case of network meta-analysis, <code>robust_index</code> is a vector with
length equal to the number of possible pairwise comparisons;
one robustness index per possible comparison.</p>
</td></tr>
<tr><td><code>robust</code></td>
<td>
<p>A character or character vector (of same length with
<code>robust_index</code>) on whether the primary analysis results are
<em>robust</em> or <em>frail</em> to the different re-analyses.</p>
</td></tr>
<tr><td><code>kld</code></td>
<td>
<p>A vector or matrix on the Kullback-Leibler divergence
(KLD) measure in the summary effect size from a subsequent re-analysis to
the primary analysis. In the case of a pairwise meta-analysis, <code>kld</code>
is a vector with length equal to the number of total analyses (one KLD
value is obtained per analysis). The number of total analyses equals the
square of the number of scenarios indicated in the argument
<code>mean_scenarios</code> of <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>, in the case of
missing participant outcome data; otherwise, the length of the character
vector in argument <code>sens</code>.
In the case of network meta-analysis, <code>robust_index</code> is a matrix with
number of rows equal to the number of total analyses and number of columns
equal to the number of  possible pairwise comparisons; one KLD value per
analysis and possible comparison.</p>
</td></tr>
<tr><td><code>threshold</code></td>
<td>
<p>The threshold used to be inherited by the
<code><a href="#topic+heatmap_robustness">heatmap_robustness</a></code> function.</p>
</td></tr>
<tr><td><code>scenarios</code></td>
<td>
<p>The scenarios considered to be inherited by the
<code><a href="#topic+kld_barplot">kld_barplot</a></code> function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Kullback S, Leibler RA. On information and sufficiency.
<em>Ann Math Stat</em> 1951;<b>22</b>(1):79&ndash;86. doi: 10.1214/aoms/1177729694
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis.
<em>Res Synth Methods</em> 2021;<b>12</b>(4):475&ndash;90. doi: 10.1002/jrsm.1478
</p>


<h3>See Also</h3>

<p><code><a href="#topic+heatmap_robustness">heatmap_robustness</a></code>, <code><a href="#topic+kld_barplot">kld_barplot</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_sensitivity">run_sensitivity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Read results from 'run_sensitivity' (using the default arguments)
res_sens &lt;- readRDS(system.file('extdata/res_sens_baker.rds',
                    package = 'rnmamod'))

# Calculate the robustness index
robustness_index(sens = res_sens,
                 threshold = 0.28)

</code></pre>

<hr>
<h2 id='robustness_index_user'>Robustness index when 'metafor' or 'netmeta' are used</h2><span id='topic+robustness_index_user'></span>

<h3>Description</h3>

<p>Calculates the robustness index for a sensitivity analysis
(Spineli et al., 2021) performed using the results of the analysis
performed via the R-package
<a href="https://CRAN.R-project.org/package=netmeta">netmeta</a> or
<a href="https://CRAN.R-project.org/package=metafor">metafor</a>.
The user defines the input and the function returns the robustness index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robustness_index_user(sens, pkg, attribute, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="robustness_index_user_+3A_sens">sens</code></td>
<td>
<p>A list of R objects of class
<code><a href="netmeta.html#topic+netmeta">netmeta</a></code>,
<code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code> (see
<a href="https://CRAN.R-project.org/package=netmeta">netmeta</a>) or
<code><a href="metafor.html#topic+rma">rma</a></code>, <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>,
<code><a href="metafor.html#topic+rma.mh">rma.mh</a></code>, <code><a href="metafor.html#topic+rma.mv">rma.mv</a></code>,
<code><a href="metafor.html#topic+rma.peto">rma.peto</a></code>, and
<code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>
(see <a href="https://CRAN.R-project.org/package=metafor">metafor</a>).
The number of elements equals the number of analyses using the same dataset
and the same R-package. The first element should refer to the primary
analysis. Hence, the list should include at least two elements
(see 'Details').</p>
</td></tr>
<tr><td><code id="robustness_index_user_+3A_pkg">pkg</code></td>
<td>
<p>Character string indicating the R-package with values
<code>"netmeta"</code>, or <code>"metafor"</code>.</p>
</td></tr>
<tr><td><code id="robustness_index_user_+3A_attribute">attribute</code></td>
<td>
<p>This is relevant only for
<a href="https://CRAN.R-project.org/package=netmeta">netmeta</a>. A vector of
at least two characters with values <code>"TE.common"</code> or
<code>"TE.random"</code>. See 'Values' in <code><a href="netmeta.html#topic+netmeta">netmeta</a></code>
or <code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code>.</p>
</td></tr>
<tr><td><code id="robustness_index_user_+3A_threshold">threshold</code></td>
<td>
<p>A number indicating the threshold of robustness, that is,
the minimally allowed deviation between the primary analysis (the first
element in <code>sens</code>) and re-analysis results. See 'Details' below.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Thresholds of robustness have been proposed only for the odds ratio
and standardised mean difference (Spineli et al., 2021).
The user may consider the values 0.28 and 0.17 in the argument
<code>threshold</code> for the odds ratio and standardised mean difference effect
measures (the default values), respectively, or consider other plausible
values. When the argument <code>threshold</code> has not been defined,
<code>robustness_index</code> considers the default values 0.28 and 0.17 as
threshold for robustness for binary and continuous outcome, respectively,
regardless of the effect measure (the default thresholds may not be proper
choices for other effect measures; hence, use these threshold with great
caution in this case). Spineli et al. (2021) offers a discussion on
specifying the <code>threshold</code> of robustness.
</p>
<p>When other effect measure is used (other than odds ratio or standardised
mean difference) or the elements in <code>sens</code> refer to different effect
measures, the execution of the function will be stopped and an error
message will be printed in the R console.
</p>
<p>In <code>robust</code>, the value <code>"robust"</code> appears when the calculated
<code>robust_index</code> is less than <code>threshold</code>; otherwise, the value
<code>"frail"</code> appears.
</p>


<h3>Value</h3>

<p><code>robustness_index_user</code> prints on the R console a message in
red text on the threshold of robustness determined by the user.
Then, the function returns the following list of elements:
</p>
<table>
<tr><td><code>robust_index</code></td>
<td>
<p>A numeric scalar or vector on the robustness
index values. In the case of a pairwise meta-analysis,
<code>robust_index</code> is scalar as only one summary effect size is obtained.
In the case of network meta-analysis, <code>robust_index</code> is a vector with
length equal to the number of possible pairwise comparisons;
one robustness index per pairwise comparison.</p>
</td></tr>
<tr><td><code>robust</code></td>
<td>
<p>A character or character vector (of same length with
<code>robust_index</code>) on whether the primary analysis results are
<em>robust</em> or <em>frail</em> to the different re-analyses.</p>
</td></tr>
<tr><td><code>kld</code></td>
<td>
<p>A vector or matrix on the Kullback-Leibler divergence
(KLD) measure in the summary effect size from a subsequent re-analysis to
the primary analysis. In the case of a pairwise meta-analysis, <code>kld</code>
is a vector with length equal to the number of total analyses (one KLD
value is obtained per analysis). The number of total analyses equals the
length of <code>sens</code>.
In the case of network meta-analysis, <code>robust_index</code> is a matrix with
number of rows equal to the number of total analyses and number of columns
equal to the number of  possible pairwise comparisons; one KLD
value per analysis and possible comparison.</p>
</td></tr>
<tr><td><code>attribute</code></td>
<td>
<p>The attributes considered.</p>
</td></tr>
<tr><td><code>threshold</code></td>
<td>
<p>The threshold used to be inherited by the
<code><a href="#topic+heatmap_robustness">heatmap_robustness</a></code> function. See 'Details'.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Kullback S, Leibler RA. On information and sufficiency.
<em>Ann Math Stat</em> 1951;<b>22</b>(1):79&ndash;86. doi: 10.1214/aoms/1177729694
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis.
<em>Res Synth Methods</em> 2021;<b>12</b>(4):475&ndash;90. doi: 10.1002/jrsm.1478
</p>


<h3>See Also</h3>

<p><code><a href="metafor.html#topic+rma">rma</a></code>,
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>,
<code><a href="metafor.html#topic+rma.mh">rma.mh</a></code>, <code><a href="metafor.html#topic+rma.mv">rma.mv</a></code>,
<code><a href="metafor.html#topic+rma.peto">rma.peto</a></code>,
<code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>,
<code><a href="netmeta.html#topic+netmeta">netmeta</a></code>,
<code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code>,
<code><a href="#topic+heatmap_robustness">heatmap_robustness</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library(netmeta)

data(Baker2009)

# Transform from arm-based to contrast-based format
p1 &lt;- pairwise(treatment, exac, total, studlab = paste(study, year),
data = Baker2009, sm = "OR")

# Conduct standard network meta-analysis
net1 &lt;- netmeta(p1, ref = "Placebo")

# Calculate the robustness index (random-effects versus fixed-effect)
robustness_index_user(sens = list(net1, net1),
                      pkg = "netmeta",
                      attribute = c("TE.random", "TE.common"),
                      threshold = 0.28)

## End(Not run)

</code></pre>

<hr>
<h2 id='run_metareg'>Perform Bayesian pairwise or network meta-regression</h2><span id='topic+run_metareg'></span>

<h3>Description</h3>

<p>Performs a one-stage pairwise or network meta-regression while addressing
aggregate binary or continuous missing participant outcome data via the
pattern-mixture model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_metareg(
  full,
  covariate,
  covar_assumption,
  n_chains,
  n_iter,
  n_burnin,
  n_thin,
  inits = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_metareg_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_covariate">covariate</code></td>
<td>
<p>A numeric vector or matrix for a trial-specific covariate
that is a potential effect modifier. See 'Details'.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_covar_assumption">covar_assumption</code></td>
<td>
<p>Character string indicating the structure of the
intervention-by-covariate interaction, as described in
Cooper et al. (2009). Set <code>covar_assumption</code> equal to
<code>"exchangeable"</code>, <code>"independent"</code>, or <code>"common"</code>.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_n_chains">n_chains</code></td>
<td>
<p>Positive integer specifying the number of chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_n_iter">n_iter</code></td>
<td>
<p>Positive integer specifying the number of Markov chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Positive integer specifying the number of iterations to
discard at the beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_n_thin">n_thin</code></td>
<td>
<p>Positive integer specifying the thinning rate for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
<tr><td><code id="run_metareg_+3A_inits">inits</code></td>
<td>
<p>A list with the initial values for the parameters; an argument
of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is <code>NULL</code>, and JAGS generates the initial values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>run_metareg</code> inherits the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>assumption</code>, <code>heter_prior</code>,
<code>mean_misspar</code>, <code>var_misspar</code>, <code>D</code>, <code>ref</code>,
<code>indic</code>, and <code>base_risk</code> from <code><a href="#topic+run_model">run_model</a></code>
(now contained in the argument <code>full</code>). This prevents specifying a
different Bayesian model from that considered in <code><a href="#topic+run_model">run_model</a></code>.
Therefore, the user needs first to apply <code><a href="#topic+run_model">run_model</a></code>, and then
use <code>run_metareg</code> (see 'Examples').
</p>
<p>The model runs in <code>JAGS</code> and the progress of the simulation appears on
the R console. The output of <code>run_metareg</code> is used as an S3 object by
other functions of the package to be processed further and provide an
end-user-ready output. The model is updated until convergence using the
<code><a href="R2jags.html#topic+autojags">autojags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>
<p>The models described in Spineli et al. (2021), and Spineli (2019) have
been extended to incorporate one <em>study-level covariate</em> variable
following the assumptions of Cooper et al. (2009) for the structure of the
intervention-by-covariate interaction. The covariate can be either a
numeric vector or matrix with columns equal to the maximum number of arms
in the dataset.
</p>


<h3>Value</h3>

<p>A list of R2jags outputs on the summaries of the posterior
distribution, and the Gelman-Rubin convergence diagnostic
(Gelman et al., 1992) for the following monitored parameters for a
fixed-effect pairwise meta-analysis:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The estimated summary effect measure (according to the argument
<code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>).</p>
</td></tr>
<tr><td><code>beta_all</code></td>
<td>
<p>The estimated regression coefficient for all possible
pairwise comparisons according to the argument <code>covar_assumption</code>.</p>
</td></tr>
<tr><td><code>dev_o</code></td>
<td>
<p>The deviance contribution of each trial-arm based on the
observed outcome.</p>
</td></tr>
<tr><td><code>hat_par</code></td>
<td>
<p>The fitted outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>The informative missingness parameter.</p>
</td></tr>
</table>
<p>For a fixed-effect network meta-analysis, the output additionally
includes:
</p>
<table>
<tr><td><code>SUCRA</code></td>
<td>
<p>The surface under the cumulative ranking (SUCRA) curve for
each intervention.</p>
</td></tr>
<tr><td><code>effectiveneness</code></td>
<td>
<p>The ranking probability of each intervention for
every rank.</p>
</td></tr>
</table>
<p>For a random-effects pairwise meta-analysis, the output additionally
includes the following elements:
</p>
<table>
<tr><td><code>EM_pred</code></td>
<td>
<p>The predicted summary effect measure (according to the
argument <code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>).</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>The estimated trial-specific effect measure (according to the
argument <code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>).
For a multi-arm trial, we estimate <em>T-1</em> effects, where <em>T</em>
is the number of interventions in the trial.</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation.</p>
</td></tr>
</table>
<p>In network meta-analysis, <code>EM</code> and <code>EM_pred</code> refer to all
possible pairwise comparisons of interventions in the network. Furthermore,
<code>tau</code> is typically assumed to be common for all observed comparisons
in the network.
For a multi-arm trial, we estimate a total <em>T-1</em> of <code>delta</code> for
comparisons with the baseline intervention of the trial (found in the first
column of the element <b>t</b>), with <em>T</em> being the number of
interventions in the trial.
</p>
<p>Furthermore, the output includes the following elements:
</p>
<table>
<tr><td><code>abs_risk</code></td>
<td>
<p>The adjusted absolute risks for each intervention. This
appears only when <code>measure = "OR"</code>, <code>measure = "RR"</code>, or
<code>measure = "RD"</code>.</p>
</td></tr>
<tr><td><code>leverage_o</code></td>
<td>
<p>The leverage for the observed outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>sign_dev_o</code></td>
<td>
<p>The sign of the difference between observed and fitted
outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>model_assessment</code></td>
<td>
<p>A data-frame on the measures of model assessment:
deviance information criterion, number of effective parameters, and total
residual deviance.</p>
</td></tr>
<tr><td><code>jagsfit</code></td>
<td>
<p>An object of S3 class <code><a href="R2jags.html#topic+jags">jags</a></code> with
the posterior results on all monitored parameters to be used in the
<code><a href="#topic+mcmc_diagnostics">mcmc_diagnostics</a></code> function.</p>
</td></tr>
</table>
<p>The <code>run_metareg</code> function also returns the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>assumption</code>, <code>covariate</code>,
<code>covar_assumption</code>, <code>n_chains</code>, <code>n_iter</code>, <code>n_burnin</code>,
and <code>n_thin</code> to be inherited by other relevant functions of the
package.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Cooper NJ, Sutton AJ, Morris D, Ades AE, Welton NJ. Addressing between-study
heterogeneity and inconsistency in mixed treatment comparisons: Application
to stroke prevention treatments in individuals with non-rheumatic atrial
fibrillation. <em>Stat Med</em> 2009;<b>28</b>(14):1861&ndash;81.
doi: 10.1002/sim.3594
</p>
<p>Gelman A, Rubin DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+autojags">autojags</a></code>, <code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_baker.rds', package = 'rnmamod'))

# Publication year
pub_year &lt;- c(1996, 1998, 1999, 2000, 2000, 2001, rep(2002, 5), 2003, 2003,
              rep(2005, 4), 2006, 2006, 2007, 2007)


# Perform a random-effects network meta-regression (exchangeable structure)
# Note: Ideally, set 'n_iter' to 10000 and 'n_burnin' to 1000
run_metareg(full = res,
            covariate = pub_year,
            covar_assumption = "exchangeable",
            n_chains = 3,
            n_iter = 1000,
            n_burnin = 100,
            n_thin = 1)


</code></pre>

<hr>
<h2 id='run_model'>Perform Bayesian pairwise or network meta-analysis</h2><span id='topic+run_model'></span>

<h3>Description</h3>

<p>Performs a one-stage pairwise or network meta-analysis while addressing
aggregate binary or continuous missing participant outcome data via the
pattern-mixture model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_model(
  data,
  measure,
  model,
  assumption,
  heter_prior,
  mean_misspar,
  var_misspar,
  D,
  ref,
  base_risk,
  n_chains,
  n_iter,
  n_burnin,
  n_thin,
  inits = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_model_+3A_data">data</code></td>
<td>
<p>A data-frame of the one-trial-per-row format with arm-level data.
See 'Format' for the specification of the columns.</p>
</td></tr>
<tr><td><code id="run_model_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure. For a binary
outcome, the following can be considered: <code>"OR"</code>, <code>"RR"</code> or
<code>"RD"</code> for the odds ratio, relative risk, and risk difference,
respectively. For a continuous outcome, the following can be considered:
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for mean difference,
standardised mean difference and ratio of means, respectively.</p>
</td></tr>
<tr><td><code id="run_model_+3A_model">model</code></td>
<td>
<p>Character string indicating the analysis model with values
<code>"RE"</code>, or <code>"FE"</code> for the random-effects and fixed-effect model,
respectively. The default argument is <code>"RE"</code>.</p>
</td></tr>
<tr><td><code id="run_model_+3A_assumption">assumption</code></td>
<td>
<p>Character string indicating the structure of the
informative missingness parameter.
Set <code>assumption</code> equal to one of the following: <code>"HIE-COMMON"</code>,
<code>"HIE-TRIAL"</code>, <code>"HIE-ARM"</code>, <code>"IDE-COMMON"</code>,
<code>"IDE-TRIAL"</code>, <code>"IDE-ARM"</code>, <code>"IND-CORR"</code>,
or <code>"IND-UNCORR"</code>.
The default argument is <code>"IDE-ARM"</code>. The abbreviations <code>"IDE"</code>,
<code>"HIE"</code>, and <code>"IND"</code> stand for identical, hierarchical and
independent, respectively. <code>"CORR"</code> and <code>"UNCORR"</code> stand for
correlated and uncorrelated, respectively.</p>
</td></tr>
<tr><td><code id="run_model_+3A_heter_prior">heter_prior</code></td>
<td>
<p>A list of three elements with the following order:
1) a character string indicating the distribution with
(currently available) values <code>"halfnormal"</code>, <code>"uniform"</code>,
<code>"lognormal"</code>, or <code>"logt"</code>; 2) two numeric values that refer to
the parameters of the selected distribution. For <code>"lognormal"</code>, and
<code>"logt"</code> these numbers refer to the mean and precision, respectively.
For <code>"halfnormal"</code>, these numbers refer to zero and the scale
parameter (equal to 4 or 1 being the corresponding precision of the scale
parameter 0.5 or 1). For <code>"uniform"</code>, these numbers refer to the
minimum and maximum value of the distribution.
See 'Details' in <code><a href="#topic+heterogeneity_param_prior">heterogeneity_param_prior</a></code>.</p>
</td></tr>
<tr><td><code id="run_model_+3A_mean_misspar">mean_misspar</code></td>
<td>
<p>A scalar or numeric vector of two numeric values for the
mean of the normal distribution of the informative missingness parameter
(see 'Details'). The default argument is 0 and corresponds to the
missing-at-random assumption.
See also 'Details' in <code><a href="#topic+missingness_param_prior">missingness_param_prior</a></code>.</p>
</td></tr>
<tr><td><code id="run_model_+3A_var_misspar">var_misspar</code></td>
<td>
<p>A positive non-zero number for the variance of the
normal distribution of the informative missingness parameter.
When the <code>measure</code> is <code>"OR"</code>, <code>"MD"</code>, or <code>"SMD"</code>
the default argument is 1. When the <code>measure</code> is <code>"ROM"</code>
the default argument is 0.04.</p>
</td></tr>
<tr><td><code id="run_model_+3A_d">D</code></td>
<td>
<p>A binary number for the direction of the outcome.
Set <code>D = 1</code> for beneficial outcome and <code>D = 0</code> for harmful
outcome.</p>
</td></tr>
<tr><td><code id="run_model_+3A_ref">ref</code></td>
<td>
<p>An integer specifying the reference intervention. The number
should match the intervention identifier under element <strong>t</strong> in
<code>data</code> (See 'Format').</p>
</td></tr>
<tr><td><code id="run_model_+3A_base_risk">base_risk</code></td>
<td>
<p>A scalar, a vector of length three with elements sorted in
ascending order, or a matrix with two columns and number of rows equal to
the number of relevant trials. In the case of a scalar or vector, the
elements should be in the interval (0, 1) (see 'Details'). If
<code>base_risk</code> has not been defined, the function uses the median event
risk for the reference intervention from the corresponding trials in
<code>data</code>. This argument is only relevant for a binary outcome.</p>
</td></tr>
<tr><td><code id="run_model_+3A_n_chains">n_chains</code></td>
<td>
<p>Positive integer specifying the number of chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="run_model_+3A_n_iter">n_iter</code></td>
<td>
<p>Positive integer specifying the number of Markov chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="run_model_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Positive integer specifying the number of iterations to
discard at the beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="run_model_+3A_n_thin">n_thin</code></td>
<td>
<p>Positive integer specifying the thinning rate for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
<tr><td><code id="run_model_+3A_inits">inits</code></td>
<td>
<p>A list with the initial values for the parameters; an argument
of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is <code>NULL</code>, and JAGS generates the initial values.</p>
</td></tr>
</table>


<h3>Format</h3>

<p>The columns of the data-frame in the argument <code>data</code> refer
to the following elements for a continuous outcome:
</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>t</strong> </td><td style="text-align: left;"> An intervention identifier in each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>y</strong> </td><td style="text-align: left;"> The observed mean value of the outcome in each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>sd</strong> </td><td style="text-align: left;"> The observed standard deviation of the outcome in
   each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>m</strong> </td><td style="text-align: left;"> The number of missing participant outcome data in
   each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>n</strong> </td><td style="text-align: left;"> The number of randomised participants in each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>For a binary outcome, the columns of the data-frame in the argument
<code>data</code> refer to the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>t</strong> </td><td style="text-align: left;"> An intervention identifier in each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>r</strong> </td><td style="text-align: left;"> The observed number of events of the outcome in
   each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>m</strong> </td><td style="text-align: left;"> The number of missing participant outcome data in
   each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
   </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>n</strong> </td><td style="text-align: left;"> The number of randomised participants in each arm.</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>The number of rows in <code>data</code> equals the number of collected trials.
Each element appears in <code>data</code> as many times as the maximum number of
interventions compared in a trial of the dataset.
In pairwise meta-analysis, the maximum number of arms is inherently two.
The same holds for a network meta-analysis without multi-arm trials.
In the case of network meta-analysis with multi-arm trials, the maximum
number of arms exceeds two. See 'Examples' that illustrates the structure
of <code>data</code> for a network with a maximum number of four arms.
It is not a prerequisite of <code>run_model</code> that the multi-arm trials
appear at the bottom of the dataset.
</p>


<h3>Details</h3>

<p>The model runs in <code>JAGS</code> and the progress of the simulation
appears on the R console. The output of <code>run_model</code> is used as an S3
object by other functions of the package to be processed further and
provide an end-user-ready output.
</p>
<p>The <code><a href="#topic+data_preparation">data_preparation</a></code> function is called to prepare the data
for the Bayesian analysis. <code><a href="#topic+data_preparation">data_preparation</a></code> creates the
pseudo-data-frames <code>m_new</code>, and <code>I</code>, that have the same
dimensions with the element <code>N</code>. <code>m_new</code> takes the zero
value for the observed trial-arms with unreported missing participant
outcome data (i.e., <code>m</code> equals <code>NA</code> for the corresponding
trial-arms), the same value with <code>m</code> for the observed trial-arms with
reported missing participant outcome data, and <code>NA</code> for the unobserved
trial-arms. <code>I</code> is a dummy pseudo-data-frame and takes the value one
for the observed trial-arms with reported missing participant outcome data,
the zero value for the observed trial-arms with unreported missing
participant outcome data (i.e., <code>m_new</code> equals zero for the
corresponding trial-arms), and <code>NA</code> for the unobserved trial-arms.
Thus, <code>I</code> indicates whether missing participant outcome data have been
collected for the observed trial-arms. If the user has not defined the
element <strong>m</strong> in <code>data</code>, <code>m_new</code> and <code>I</code> take the zero
value for all observed trial-arms to indicate that no missing participant
outcome data have been collected for the analysed outcome. See 'Details' in
<code><a href="#topic+data_preparation">data_preparation</a></code>.
</p>
<p>Furthermore, <code><a href="#topic+data_preparation">data_preparation</a></code> sorts the interventions across
the arms of each trial in an ascending order and correspondingly the
remaining elements in <code>data</code> (see 'Format').
<code><a href="#topic+data_preparation">data_preparation</a></code> considers the first column in <strong>t</strong> as
being the control arm for every trial. Thus, this sorting ensures that
interventions with a lower identifier are consistently treated as the
control arm in each trial. This case is relevant in non-star-shaped
networks.
</p>
<p>The model is updated until convergence using the
<code><a href="R2jags.html#topic+autojags">autojags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>
<p>To perform a Bayesian pairwise or network meta-analysis, the
<code><a href="#topic+prepare_model">prepare_model</a></code> function is called which contains the WinBUGS
code as written by Dias et al. (2013a) for binomial and normal likelihood to
analyse aggregate binary and continuous outcome data, respectively.
<code><a href="#topic+prepare_model">prepare_model</a></code> uses the consistency model (as described in
Lu and Ades (2006)) to estimate all possible comparisons in the network.
It also accounts for the multi-arm trials by assigning conditional
univariate normal distributions on the underlying trial-specific effect
size of comparisons with the baseline arm of the multi-arm trial
(Dias et al., 2013a).
</p>
<p>The code of Dias et al. (2013a) has been extended to incorporate the
pattern-mixture model to adjust the underlying outcome in each arm of
every trial for missing participant outcome data (Spineli et al., 2021;
Spineli, 2019a; Turner et al., 2015). The assumptions about the
missingness parameter are specified using the arguments <code>mean_misspar</code>
and <code>var_misspar</code>. Specifically, <code>run_model</code> considers the
informative missingness odds ratio in the logarithmic scale for binary
outcome data (Spineli, 2019a; Turner et al., 2015; White et al., 2008), the
informative missingness difference of means when <code>measure</code> is
<code>"MD"</code> or <code>"SMD"</code>, and the informative missingness ratio of means
in the logarithmic scale when <code>measure</code> is <code>"ROM"</code>
(Spineli et al., 2021; Mavridis et al., 2015).
</p>
<p>When <code>assumption</code> is trial-specific (i.e., <code>"IDE-TRIAL"</code> or
<code>"HIE-TRIAL"</code>), or independent (i.e., <code>"IND-CORR"</code> or
<code>"IND-UNCORR"</code>), only one numeric value can be assigned to
<code>mean_misspar</code> because the same missingness scenario is applied to all
trials and trial-arms of the dataset, respectively. When <code>assumption</code>
is <code>"IDE-ARM"</code> or <code>"HIE-ARM"</code>, a maximum of two
<em>different</em> or <em>identical</em> numeric values can be assigned as a
vector to <code>mean_misspars</code>: the first value refers to the experimental
arm, and the second value refers to the control arm of a trial.
In the case of a network, the first value is considered for all
non-reference interventions and the second value is considered for the
reference intervention of the network (i.e., the intervention with
identifier equal to <code>ref</code>). This is necessary to ensure transitivity
in the assumptions for the missingness parameter across the network
(Spineli, 2019b).
</p>
<p>When there is at least one trial-arm with unreported missing participant
outcome data (i.e., <code>m</code> equals <code>NA</code> for the corresponding
trial-arms) or when missing participant outcome data have not been
collected for the analysed outcome (i.e., <code>m</code> is missing in
<code>data</code>), <code>run_model</code> assigns the assumption <code>"IND-UNCORR"</code>
to <code>assumption</code>.
</p>
<p>Currently, there are no empirically-based prior distributions for the
informative missingness parameters. The user may refer to Spineli (2019),
Turner et al. (2015), Mavridis et al. (2015), and White et al. (2008) to
determine <code>mean_misspar</code> and select a proper value for
<code>var_misspar</code>.
</p>
<p>The scalar <code>base_risk</code> refers to a fixed baseline risk for the
selected reference intervention (as specified with <code>ref</code>).
When <code>base_risk</code> is a three-element vector, it refers to a random
baseline risk and the elements should be sorted in ascending order as they
refer to the lower bound, mean value, and upper bound of the 95%
confidence interval for the baseline risk for the selected reference
intervention. The <code><a href="#topic+baseline_model">baseline_model</a></code> function is called to
calculate the mean and variance of the approximately normal distribution of
the logit of an event for <code>ref</code> using these three elements
(Dias et al., 2018). When <code>base_risk</code> is a matrix, it refers to the
predicted baseline risk with first column being the number of events, and
second column being the sample size of the corresponding trials on the
selected reference intervention. Then the <code><a href="#topic+baseline_model">baseline_model</a></code>
function is called that contains the WinBUGS code as written by Dias et al.
(2013b) for the hierarchical baseline model. The posterior mean and
precision of the predictive distribution of the logit of an event
for the selected reference intervention are plugged in the WinBUGS code for
the relative effects model (via the <code><a href="#topic+prepare_model">prepare_model</a></code> function).
The matrix <code>base_risk</code> should not comprise the trials in <code>data</code>
that include the <code>ref</code>, unless justified (Dias et al., 2018).
</p>
<p>To obtain unique absolute risks for each intervention, the network
meta-analysis model has been extended to incorporate the transitive risks
framework, namely, an intervention has the same absolute risk regardless of
the comparator intervention(s) in a trial (Spineli et al., 2017).
The absolute risks are a function of the odds ratio (the <strong>base-case</strong>
effect measure for a binary outcome) and the selected baseline risk for the
reference intervention (<code>ref</code>) (Appendix in Dias et al., 2013a).
We advocate using the odds ratio as an effect measure for its desired
mathematical properties. Then, the relative risk and risk difference can be
obtained as a function of the absolute risks of the corresponding
interventions in the comparison of interest. Hence, regardless of the
selected <code>measure</code> for a binary outcome, <code>run_model</code> performs
pairwise or network meta-analysis based on the odds ratio.
</p>


<h3>Value</h3>

<p>A list of R2jags output on the summaries of the posterior
distribution, and the Gelman-Rubin convergence diagnostic
(Gelman et al., 1992) of the following monitored parameters for a
fixed-effect pairwise meta-analysis:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The estimated summary effect measure (according to the argument
<code>measure</code>).</p>
</td></tr>
<tr><td><code>EM_LOR</code></td>
<td>
<p>The estimated summary odd ratio in the logarithmic scale when
<code>measure = "RR"</code> or <code>measure = "RD"</code>.</p>
</td></tr>
<tr><td><code>dev_o</code></td>
<td>
<p>The deviance contribution of each trial-arm based on the
observed outcome.</p>
</td></tr>
<tr><td><code>hat_par</code></td>
<td>
<p>The fitted outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>The informative missingness parameter.</p>
</td></tr>
</table>
<p>For a fixed-effect network meta-analysis, the output additionally includes:
</p>
<table>
<tr><td><code>SUCRA</code></td>
<td>
<p>The surface under the cumulative ranking curve for each
intervention.</p>
</td></tr>
<tr><td><code>SUCRA_LOR</code></td>
<td>
<p>The surface under the cumulative ranking curve for each
intervention under the odds ratio effect measure when <code>measure = "RR"</code>
or <code>measure = "RD"</code>.</p>
</td></tr>
<tr><td><code>effectiveneness</code></td>
<td>
<p>The ranking probability of each intervention for
every rank.</p>
</td></tr>
</table>
<p>For a random-effects pairwise meta-analysis, the output additionally
includes the following elements:
</p>
<table>
<tr><td><code>EM_pred</code></td>
<td>
<p>The predicted summary effect measure (according to the
argument <code>measure</code>).</p>
</td></tr>
<tr><td><code>EM_pred_LOR</code></td>
<td>
<p>The predicted summary odds ratio in the logarithmic
scale when <code>measure = "RR"</code> or <code>measure = "RD"</code>.</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>The estimated trial-specific effect measure (according to the
argument <code>measure</code>).</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation.</p>
</td></tr>
</table>
<p>In network meta-analysis, <code>EM</code> and <code>EM_pred</code> refer to all
possible pairwise comparisons of interventions in the network. Furthermore,
<code>tau</code> is typically assumed to be common for all observed comparisons
in the network. For a multi-arm trial, we estimate a total of <em>T-1</em>
<code>delta</code> for comparisons with the baseline intervention of the trial
(found in the first column of the element <b>t</b>), with <em>T</em> being
the number of interventions in the trial.
</p>
<p>Furthermore, the output includes the following elements:
</p>
<table>
<tr><td><code>leverage_o</code></td>
<td>
<p>The leverage for the observed outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>sign_dev_o</code></td>
<td>
<p>The sign of the difference between observed and fitted
outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>model_assessment</code></td>
<td>
<p>A data-frame on the measures of model assessment:
deviance information criterion, number of effective parameters, and total
residual deviance.</p>
</td></tr>
<tr><td><code>indic</code></td>
<td>
<p>The sign of basic parameters in relation to the reference
intervention as specified in argument <code>reg</code></p>
</td></tr>
<tr><td><code>jagsfit</code></td>
<td>
<p>An object of S3 class <code><a href="R2jags.html#topic+jags">jags</a></code> with
the posterior results on all monitored parameters to be used in the
<code><a href="#topic+mcmc_diagnostics">mcmc_diagnostics</a></code> function.</p>
</td></tr>
</table>
<p>The <code>run_model</code> function also returns the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>assumption</code>, <code>heter_prior</code>,
<code>mean_misspar</code>, <code>var_misspar</code>, <code>D</code>, <code>ref</code>,
<code>base_risk</code>, <code>n_chains</code>, <code>n_iter</code>, <code>n_burnin</code>,
and <code>n_thin</code> as specified by the user to be inherited by other
functions of the package.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Cooper NJ, Sutton AJ, Morris D, Ades AE, Welton NJ. Addressing
between-study heterogeneity and inconsistency in mixed treatment
comparisons: Application to stroke prevention treatments in individuals
with non-rheumatic atrial fibrillation.
<em>Stat Med</em> 2009;<b>28</b>(14):1861&ndash;81. doi: 10.1002/sim.3594
</p>
<p>Dias S, Ades AE, Welton NJ, Jansen JP, Sutton AJ. Network Meta-Analysis for
Decision Making. Chichester (UK): Wiley; 2018.
</p>
<p>Dias S, Sutton AJ, Ades AE, Welton NJ. Evidence synthesis for decision
making 2: a generalized linear modeling framework for pairwise and network
meta-analysis of randomized controlled trials. <em>Med Decis Making</em>
2013a;<b>33</b>(5):607&ndash;17. doi: 10.1177/0272989X12458724
</p>
<p>Dias S, Welton NJ, Sutton AJ, Ades AE. Evidence synthesis for decision
making 5: the baseline natural history model. <em>Med Decis Making</em>
2013b;<b>33</b>(5):657&ndash;70. doi: 10.1177/0272989X13485155
</p>
<p>Gelman A, Rubin DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>
<p>Lu G, Ades AE. Assessing evidence inconsistency in mixed treatment
comparisons. <em>J Am Stat Assoc</em> 2006;<b>101</b>:447&ndash;59.
doi: 10.1198/016214505000001302
</p>
<p>Mavridis D, White IR, Higgins JP, Cipriani A, Salanti G. Allowing for
uncertainty due to missing continuous outcome data in pairwise and
network meta-analysis. <em>Stat Med</em> 2015;<b>34</b>(5):721&ndash;41.
doi: 10.1002/sim.6365
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019a;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>
<p>Spineli LM. Modeling missing binary outcome data while preserving
transitivity assumption yielded more credible network meta-analysis
results. <em>J Clin Epidemiol</em> 2019b;<b>105</b>:19&ndash;26.
doi: 10.1016/j.jclinepi.2018.09.002
</p>
<p>Spineli LM, Brignardello-Petersen R, Heen AF, Achille F, Brandt L,
Guyatt GH, et al. Obtaining absolute effect estimates to facilitate shared
decision making in the context of multiple-treatment comparisons.
Abstracts of the Global Evidence Summit, Cape Town, South Africa.
<em>Cochrane Database of Systematic Reviews</em> 2017;<b>9</b>(Suppl 1):18911.
</p>
<p>Turner NL, Dias S, Ades AE, Welton NJ. A Bayesian framework to account
for uncertainty due to missing binary outcome data in pairwise
meta-analysis. <em>Stat Med</em> 2015;<b>34</b>(12):2062&ndash;80.
doi: 10.1002/sim.6475
</p>
<p>White IR, Higgins JP, Wood AM. Allowing for uncertainty due to missing data
in meta-analysis&ndash;part 1: two-stage methods. <em>Stat Med</em>
2008;<b>27</b>(5):711&ndash;27. doi: 10.1002/sim.3008
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+autojags">autojags</a></code>,
<code><a href="#topic+baseline_model">baseline_model</a></code>, <code><a href="#topic+data_preparation">data_preparation</a></code>,
<code><a href="#topic+heterogeneity_param_prior">heterogeneity_param_prior</a></code>,
<code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="#topic+missingness_param_prior">missingness_param_prior</a></code>, <code><a href="#topic+prepare_model">prepare_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Show the first six trials of the dataset
head(nma.baker2009)


# Perform a random-effects network meta-analysis
# Note: Ideally, set 'n_iter' to 10000 and 'n_burnin' to 1000
run_model(data = nma.baker2009,
          measure = "OR",
          model = "RE",
          assumption = "IDE-ARM",
          heter_prior = list("halfnormal", 0, 1),
          mean_misspar = c(0, 0),
          var_misspar = 1,
          D = 0,
          ref = 1,
          n_chains = 3,
          n_iter = 1000,
          n_burnin = 100,
          n_thin = 1)


</code></pre>

<hr>
<h2 id='run_nodesplit'>Perform the node-splitting approach</h2><span id='topic+run_nodesplit'></span>

<h3>Description</h3>

<p>Performs the Bayesian node-splitting approach of Dias et al. (2010)
extended to address aggregate binary and continuous missing participant
outcome data via the pattern-mixture model (Spineli et al., 2021;
Spineli, 2019). This model offers a local evaluation of the
plausibility of the consistency assumption in the network
(Dias et al., 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_nodesplit(full, n_chains, n_iter, n_burnin, n_thin, inits = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_nodesplit_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="run_nodesplit_+3A_n_chains">n_chains</code></td>
<td>
<p>Positive integer specifying the number of chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="run_nodesplit_+3A_n_iter">n_iter</code></td>
<td>
<p>Positive integer specifying the number of Markov chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="run_nodesplit_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Positive integer specifying the number of iterations to
discard at the beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="run_nodesplit_+3A_n_thin">n_thin</code></td>
<td>
<p>Positive integer specifying the thinning rate for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
<tr><td><code id="run_nodesplit_+3A_inits">inits</code></td>
<td>
<p>A list with the initial values for the parameters; an argument
of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is <code>NULL</code>, and JAGS generates the initial values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>run_nodesplit</code> inherits the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>assumption</code>, <code>heter_prior</code>,
<code>mean_misspar</code>, <code>var_misspar</code>, <code>ref</code>, and <code>indic</code> from
<code><a href="#topic+run_model">run_model</a></code> (now contained in the argument <code>full</code>).
This prevents specifying a different Bayesian model from that considered
in <code><a href="#topic+run_model">run_model</a></code>. Therefore, the user needs first to apply
<code><a href="#topic+run_model">run_model</a></code>, and then use <code>run_nodesplit</code>
(see 'Examples').
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>run_nodesplit</code>
currently performs node-splitting using the odds ratio as effect measure
for being the <strong>base-case</strong> effect measure in <code><a href="#topic+run_model">run_model</a></code>
for a binary outcome (see also 'Details' in <code><a href="#topic+run_model">run_model</a></code>).
</p>
<p>To perform the Bayesian node-splitting approach, the
<code><a href="#topic+prepare_nodesplit">prepare_nodesplit</a></code> function is called which contains the
WinBUGS code as written by Dias et al. (2010) for binomial and normal
likelihood to analyse binary and continuous outcome data, respectively.
<code><a href="#topic+prepare_nodesplit">prepare_nodesplit</a></code> has been extended to incorporate the
pattern-mixture model with informative missingness parameters for binary
and continuous outcome data (see 'Details' in <code><a href="#topic+run_model">run_model</a></code>).
</p>
<p><code>run_nodesplit</code> runs the Bayesian node-splitting approach in
<code>JAGS</code>. The progress of the simulation appears on the R console. The
number of times <code>run_nodesplit</code> is used appears on the R console as a
text in red and it equals the number of split nodes (see 'Examples').
If there are no split nodes in the network, the execution of the function
will be stopped and an error message will be printed on the R console.
The model is updated until convergence using the
<code><a href="R2jags.html#topic+autojags">autojags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>
<p><code>run_nodesplit</code> uses the
<code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=gemtc">gemtc</a>
to obtain automatically the nodes to split based on the decision rule of
van Valkenhoef et al. (2016).
<code>run_nodesplit</code> uses the option (1) in van Valkenhoef et al. (2016)
to parameterise multi-arm trials that contain the node-to-split.
In contrast,
<code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>
uses the option (3) in van Valkenhoef et al. (2016).
Option (1) keeps the baseline arm of the node-to-split in the corresponding
multi-arms. Option (3) excludes both arms of the node-to-split from the
corresponding multi-arm trials.
</p>
<p>The output of <code>run_nodesplit</code> is not end-user-ready.
The <code><a href="#topic+nodesplit_plot">nodesplit_plot</a></code> function inherits the output of
<code>run_nodesplit</code> as an S3 object and processes it further to provide an
end-user-ready output.
</p>
<p><code>run_nodesplit</code> can be used only for a network of interventions.
In the case of two interventions, the execution of the function will
be stopped and an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p>An R2jags output on the summaries of the posterior distribution,
and the Gelman-Rubin convergence diagnostic of the
following monitored parameters:
</p>
<table>
<tr><td><code>direct</code></td>
<td>
<p>The summary effect measure (according to the argument
<code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>) of each split node based
on the corresponding trials.</p>
</td></tr>
<tr><td><code>indirect</code></td>
<td>
<p>The indirect summary effect measure (according to the
argument <code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>) of each split
node based on the remaining network after removing (splitting) the
corresponding node.</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>The inconsistency parameter for each split node defined as the
difference between the direct and indirect effect of the corresponding
split node.</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation after each split node, when
the random-effects model has been specified.</p>
</td></tr>
</table>
<p>Furthermore, the output includes the following element:
</p>
<table>
<tr><td><code>model_assessment</code></td>
<td>
<p>A data-frame on the measures of model assessment
after each split node: deviance information criterion,
total residual deviance, and number of effective parameters.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Caldwell DM, Ades AE. Checking consistency in mixed
treatment comparison meta-analysis.
<em>Stat Med</em> 2010;<b>29</b>(7-8):932&ndash;44.
doi: 10.1002/sim.3767
</p>
<p>Gelman A, Rubin DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>
<p>van Valkenhoef G, Dias S, Ades AE, Welton NJ. Automated generation of
node-splitting models for assessment of inconsistency in network
meta-analysis. <em>Res Synth Methods</em> 2016;<b>7</b>(1):80&ndash;93.
doi: 10.1002/jrsm.1167
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+autojags">autojags</a></code>,
<code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="gemtc.html#topic+mtc.nodesplit.comparisons">mtc.nodesplit.comparisons</a></code>,
<code><a href="#topic+nodesplit_plot">nodesplit_plot</a></code>, <code><a href="#topic+prepare_nodesplit">prepare_nodesplit</a></code>,
<code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.baker2009")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_baker.rds', package = 'rnmamod'))


# Run random-effects node-splitting approach
# Note: Ideally, set 'n_iter' to 10000 and 'n_burnin' to 1000
run_nodesplit(full = res,
              n_chains = 3,
              n_iter = 1000,
              n_burnin = 100,
              n_thin = 1)


</code></pre>

<hr>
<h2 id='run_sensitivity'>Perform sensitivity analysis for missing participant outcome data</h2><span id='topic+run_sensitivity'></span>

<h3>Description</h3>

<p>Performs a sensitivity analysis by applying pairwise
meta-analysis or network meta-analysis for a series of different scenarios
about the informative missingness parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_sensitivity(
  full,
  assumption,
  mean_scenarios,
  var_misspar,
  n_chains,
  n_iter,
  n_burnin,
  n_thin,
  inits = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_sensitivity_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_assumption">assumption</code></td>
<td>
<p>Character string indicating the structure of the
informative missingness parameter. Set <code>assumption</code> equal to one of
the following two: <code>"HIE-ARM"</code>, or <code>"IDE-ARM"</code> (see 'Details').
The default argument is <code>"IDE-ARM"</code>. The abbreviations <code>"IDE"</code>,
and <code>"HIE"</code> stand for identical, and hierarchical, respectively.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_mean_scenarios">mean_scenarios</code></td>
<td>
<p>A vector with numeric values for the mean of the normal
distribution of the informative missingness parameter (see 'Details').
The vector should have a length equal to 5 or larger.
The missing-at-random (MAR) assumption should be the median of the vector,
so that the same number of informative scenarios appear before and after
the MAR. The default scenarios are c(-log(3), -log(2), log(0.9999), log(2),
log(3)) and c(-2, -1, 0, 1, 2) for binary and continuous outcome data,
respectively.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_var_misspar">var_misspar</code></td>
<td>
<p>A positive non-zero number for the variance of the normal
distribution of the informative missingness parameter. When the
<code>measure</code> (defined in <code><a href="#topic+run_model">run_model</a></code>) is <code>"OR"</code>,
<code>"MD"</code>, or <code>"SMD"</code> the default argument is 1. When the
<code>measure</code> is <code>"ROM"</code>, the default argument is 0.04.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_n_chains">n_chains</code></td>
<td>
<p>Integer specifying the number of chains for the MCMC
sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of
the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_n_iter">n_iter</code></td>
<td>
<p>Integer specifying the number of Markov chains for the MCMC
sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of
the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Integer specifying the number of iterations to discard at the
beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_n_thin">n_thin</code></td>
<td>
<p>Integer specifying the thinning rate for the MCMC sampling; an
argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
<tr><td><code id="run_sensitivity_+3A_inits">inits</code></td>
<td>
<p>A list with the initial values for the parameters; an argument
of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is <code>NULL</code>, and JAGS generates the initial values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model runs in <code>JAGS</code> and the progress of the simulation
appears on the R console. The number of times <code>run_sensitivity</code> is
used appears on the R console as a text in red and it equals the
<b>number of scenarios</b> defined as <em>the square of the length of the
vector specified in <code>mean_scenarios</code></em> (see 'Examples').
The output of <code>run_sensitivity</code> is used as an S3 object by other
functions of the package to be processed further and provide an
end-user-ready output.
</p>
<p>In the case of pairwise meta-analysis, <code>EM</code> and <code>tau</code> are
estimated as many times as the number of scenarios considered.
In the case of network meta-analysis, each possible pairwise comparison is
estimated as many times as the number of scenarios considered.
</p>
<p>The informative missingness parameter is assumed to differ only across the
interventions of the dataset. Therefore, the user can specify the
informative missingness parameter to be arm-specific <em>and</em> identical
(<code>assumption = "IDE-ARM"</code>), or arm-specific <em>and</em>
hierarchical (<code>assumption = "HIE-ARM"</code>) (Spineli et al., 2021).
</p>
<p>The length of the vector specified in argument <code>mean_scenarios</code> should
be equal to or more than 5 (a positive odd integer) to allow for an
adequate number of scenarios. It is important that the number
corresponding to the MAR assumption is the middle of the numbers in the
vector specified in argument <code>mean_scenarios</code>. The <b>MAR
assumption</b> constitutes the <b>primary analysis</b>.
Under the informative missingness difference of means parameter (relevant
for the raw and standardised mean diffenre), the MAR assumption equals 0.
Under the informative missingness odds ratio parameter (IMOR; relevant for
the odds ratio) and the informative missingness ratio of means (IMRoM;
relevant for the ratio of means) parameter, the MAR assumption equals 1;
however, both parameters are analysed in the logarithmic scale. We advise
using the value <code>0.999</code> rather than <code>1</code> in <code>mean_scenarios</code>
for the IMOR and IMRoM parameters; otherwise, the execution of the function
will be stopped and the error 'Invalid parent values' will be printed on
the R console.
</p>
<p>Currently, there are no empirically-based prior distributions for the
informative missingness parameters. The users may refer to Spineli (2019),
Mavridis et al. (2015), Turner et al. (2015), and White et al. (2008) to
determine <code>mean_scenarios</code> for an informative missingness mechanism
and select a proper value for <code>var_misspar</code>.
</p>
<p><code>run_sensitivity</code> inherits the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>heter_prior</code>, <code>D</code>, <code>indic</code>,
<code>base_risk</code>, and <code>ref</code> from <code><a href="#topic+run_model">run_model</a></code>
(now contained in the argument <code>full</code>). This prevents specifying a
different Bayesian model from that considered in the primary analysis
(via <code><a href="#topic+run_model">run_model</a></code>)&ndash;an exception in the <code>assumption</code>
argument as it is restricted to only two character strings. Therefore, the
user needs first to apply <code><a href="#topic+run_model">run_model</a></code>, and then use
<code>run_sensitivity</code> (see 'Examples').
</p>
<p>The <code>run_sensitivity</code> function also returns the arguments
<code>measure</code>, <code>scenarios</code>, <code>D</code>, <code>heter</code>, <code>n_chains</code>,
<code>n_iter</code>, <code>n_burnin</code>, and <code>n_thin</code> as specified by the user
to be inherited by other relevant functions of the package.
</p>
<p>The model is updated until convergence using the
<code><a href="R2jags.html#topic+autojags">autojags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>
<p><code>run_sensitivity</code> can be used only when missing participant
outcome data have been extracted for at least one trial. Otherwise, the
execution of the function will be stopped and an error message will be
printed on the R console.
</p>


<h3>Value</h3>

<p>A list of R2jags outputs on the summaries of the posterior
distribution, and the Gelman-Rubin convergence diagnostic
(Gelman et al., 1992) of the following monitored parameters for a
random-effects pairwise meta-analysis:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The estimated summary effect measure (according to the
argument <code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>).</p>
</td></tr>
<tr><td><code>EM_LOR</code></td>
<td>
<p>The estimated summary odd ratio in the logarithmic scale when
<code>measure = "RR"</code> or <code>measure = "RD"</code>.</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation. This element does
not appear in the case of a fixed-effect pairwise meta-analysis.</p>
</td></tr>
</table>
<p>In a random-effects network meta-analysis, <code>EM</code> refer to all possible
pairwise comparisons of interventions in the network. Furthermore,
<code>tau</code> is typically assumed to be common for all observed comparisons
in the network.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Gelman, A, Rubin, DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>
<p>Mavridis D, White IR, Higgins JP, Cipriani A, Salanti G. Allowing for
uncertainty due to missing continuous outcome data in pairwise and network
meta-analysis. <em>Stat Med</em> 2015;<b>34</b>(5):721&ndash;41.
doi: 10.1002/sim.6365
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Quantifying the robustness of
primary analysis results: A case study on missing outcome data in pairwise
and network meta-analysis.
<em>Res Synth Methods</em> 2021;<b>12</b>(4):475&ndash;90. doi: 10.1002/jrsm.1478
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>
<p>Turner NL, Dias S, Ades AE, Welton NJ. A Bayesian framework to account for
uncertainty due to missing binary outcome data in pairwise meta-analysis.
<em>Stat Med</em> 2015;<b>34</b>(12):2062&ndash;80. doi: 10.1002/sim.6475
</p>
<p>White IR, Higgins JP, Wood AM. Allowing for uncertainty due to missing data
in meta-analysis&ndash;part 1: two-stage methods.
<em>Stat Med</em> 2008;<b>27</b>(5):711&ndash;27. doi: 10.1002/sim.3008
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+autojags">autojags</a></code>,
<code><a href="R2jags.html#topic+jags">jags</a></code>, <code><a href="#topic+run_model">run_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pma.taylor2004")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_taylor.rds', package = 'rnmamod'))


# Perform the sensitivity analysis (default arguments)
# Note: Ideally, set 'n_iter' to 10000 and 'n_burnin' to 1000
run_sensitivity(full = res,
                assumption = "IDE-ARM",
                var_misspar = 1,
                n_chains = 3,
                n_iter = 1000,
                n_burnin = 100,
                n_thin = 5)


</code></pre>

<hr>
<h2 id='run_series_meta'>Perform a series of Bayesian pairwise meta-analyses</h2><span id='topic+run_series_meta'></span>

<h3>Description</h3>

<p>Performs a Bayesian pairwise meta-analysis for each pairwise
comparison with at least two trials in the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_series_meta(full, n_chains, n_iter, n_burnin, n_thin, inits = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_series_meta_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="run_series_meta_+3A_n_chains">n_chains</code></td>
<td>
<p>Integer specifying the number of chains for the MCMC
sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of
the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="run_series_meta_+3A_n_iter">n_iter</code></td>
<td>
<p>Positive integer specifying the number of Markov chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="run_series_meta_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Positive integer specifying the number of iterations to
discard at the beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="run_series_meta_+3A_n_thin">n_thin</code></td>
<td>
<p>Positive integer specifying the thinning rate for the MCMC
sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of
the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
<tr><td><code id="run_series_meta_+3A_inits">inits</code></td>
<td>
<p>A list with the initial values for the parameters; an argument
of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is <code>NULL</code>, and JAGS generates the initial values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>run_series_meta</code> inherits the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>assumption</code>, <code>heter_prior</code>,
<code>mean_misspar</code>, and <code>var_misspar</code> from <code><a href="#topic+run_model">run_model</a></code>
(now contained in the argument <code>full</code>). This prevents specifying
a different Bayesian model from that considered in <code><a href="#topic+run_model">run_model</a></code>.
Therefore, the user needs first to apply <code><a href="#topic+run_model">run_model</a></code>, and then
use <code>run_series_meta</code> (see 'Examples').
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>run_series_meta</code>
currently performs a series of pairwise meta-analysis using the odds ratio
as effect measure for being the <strong>base-case</strong> effect measure in
<code><a href="#topic+run_model">run_model</a></code> for a binary outcome (see also 'Details' in
<code><a href="#topic+run_model">run_model</a></code>).
</p>
<p><code>run_series_meta</code> runs a series of Bayesian pairwise meta-analyses
in <code>JAGS</code>. The progress of the simulation appears on the R console.
The number of times the function is used is also printed on the console
(in red) and is equal to the number of  observed pairwise comparisons
in the network (see 'Examples').
The model is updated until convergence using the
<code><a href="R2jags.html#topic+autojags">autojags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>
<p>The output of <code>run_series_meta</code> is not end-user-ready. The
<code><a href="#topic+series_meta_plot">series_meta_plot</a></code> function inherits the output of
<code>run_series_meta</code> as an S3 object and processes it further to provide
an end-user-ready output.
</p>
<p><code>run_series_meta</code> can be used only for a network of interventions.
In the case of two interventions, the execution of the function will
be stopped and an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p>An R2jags output on the summaries of the posterior distribution, and
the Gelman-Rubin convergence diagnostic (Gelman et al., 1992) of the
following monitored parameters:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The summary effect estimate (according to the argument
<code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>) of each observed
pairwise comparison with at least two trials in the network.</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation for pairwise comparisons
with at least two trials, when the random-effects model has been
specified.</p>
</td></tr>
<tr><td><code>single</code></td>
<td>
<p>A binary vector that indicates the comparisons in <code>EM</code>
with one trial.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Gelman A, Rubin DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+series_meta_plot">series_meta_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.dogliotti2014")

# Show the first six trials of the dataset (one-trial-per-row format)
head(nma.dogliotti2014)

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_dogliotti.rds', package = 'rnmamod'))


# Run separate random-effects pairwise meta-analyses
# Note: Ideally, set 'n_iter' to 10000 and 'n_burnin' to 1000
run_series_meta(full = res,
                n_chains = 3,
                n_iter = 1000,
                n_burnin = 100,
                n_thin = 1)


</code></pre>

<hr>
<h2 id='run_ume'>Perform the unrelated mean effects model</h2><span id='topic+run_ume'></span>

<h3>Description</h3>

<p>Performs the unrelated mean effects model of Dias et al. (2013)
that has been refined (Spineli, 2021) and extended to address aggregate
binary and continuous missing participant outcome data via the
pattern-mixture model (Spineli et al. 2021; Spineli, 2019). This model
offers a global evaluation of the plausibility of the consistency
assumption in the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_ume(full, n_iter, n_burnin, n_chains, n_thin, inits = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_ume_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="run_ume_+3A_n_iter">n_iter</code></td>
<td>
<p>Positive integer specifying the number of Markov chains for the
MCMC sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function
of the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 10000.</p>
</td></tr>
<tr><td><code id="run_ume_+3A_n_burnin">n_burnin</code></td>
<td>
<p>Positive integer specifying the number of iterations to
discard at the beginning of the MCMC sampling; an argument of the
<code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1000.</p>
</td></tr>
<tr><td><code id="run_ume_+3A_n_chains">n_chains</code></td>
<td>
<p>Positive integer specifying the number of chains for the MCMC
sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of
the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 2.</p>
</td></tr>
<tr><td><code id="run_ume_+3A_n_thin">n_thin</code></td>
<td>
<p>Positive integer specifying the thinning rate for the MCMC
sampling; an argument of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of
the R-package <a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is 1.</p>
</td></tr>
<tr><td><code id="run_ume_+3A_inits">inits</code></td>
<td>
<p>A list with the initial values for the parameters; an argument
of the <code><a href="R2jags.html#topic+jags">jags</a></code> function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a>.
The default argument is <code>NULL</code>, and JAGS generates the initial values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>run_ume</code> inherits the arguments <code>data</code>,
<code>measure</code>, <code>model</code>, <code>assumption</code>, <code>heter_prior</code>,
<code>mean_misspar</code>, <code>var_misspar</code>, and <code>ref</code> from
<code><a href="#topic+run_model">run_model</a></code>.
This prevents specifying a different Bayesian model from that considered in
<code><a href="#topic+run_model">run_model</a></code>.Therefore, the user needs first to apply
<code><a href="#topic+run_model">run_model</a></code>, and then use <code>run_ume</code> (see 'Examples').
</p>
<p>The <code>run_ume</code> function also returns the arguments <code>data</code>,
<code>model</code>, <code>measure</code>, <code>assumption</code>, <code>n_chains</code>,
<code>n_iter</code>, <code>n_burnin</code>, and <code>n_thin</code> as specified by the user
to be inherited by other relevant functions of the package.
</p>
<p>Initially, <code>run_ume</code> calls the <code><a href="#topic+improved_ume">improved_ume</a></code> function to
identify the <em>frail comparisons</em>, that is, comparisons between
non-baseline interventions in multi-arm trials not investigated in any
two-arm or multi-arm trial of the network (Spineli, 2021). The 'original'
model of Dias et al. (2013) omits the frail comparisons from the estimation
process. Consequently, the number of estimated summary effects is less
than those obtained by performing separate pairwise meta-analyses
(see <code><a href="#topic+run_series_meta">run_series_meta</a></code>).
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>run_ume</code> currently
considers the odds ratio as effect measure for being the <strong>base-case</strong>
effect measure in <code><a href="#topic+run_model">run_model</a></code> for a binary outcome
(see also 'Details' in <code><a href="#topic+run_model">run_model</a></code>).
</p>
<p><code>run_ume</code> calls the <code><a href="#topic+prepare_ume">prepare_ume</a></code> function which contains
the WinBUGS code as written by Dias et al. (2013) for binomial and normal
likelihood to analyse binary and continuous outcome data, respectively.
<code><a href="#topic+prepare_ume">prepare_ume</a></code> has been extended to incorporate the
pattern-mixture model with informative missingness parameters for binary
and continuous outcome data (see 'Details' in <code><a href="#topic+run_model">run_model</a></code>).
<code><a href="#topic+prepare_ume">prepare_ume</a></code> has also been refined to account for the
multi-arm trials by assigning conditional univariate normal distributions
on the underlying trial-specific effect size of comparisons with the
baseline arm of the multi-arm trial (Spineli, 2021).
</p>
<p><code>run_ume</code> runs Bayesian unrelated mean effects model in <code>JAGS</code>.
The progress of the simulation appears on the R console. The model is
updated until convergence using the <code><a href="R2jags.html#topic+autojags">autojags</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=R2jags">R2jags</a> with 2 updates and
number of iterations and thinning equal to <code>n_iter</code> and <code>n_thin</code>,
respectively.
</p>
<p>The output of <code>run_ume</code> is not end-user-ready. The
<code><a href="#topic+ume_plot">ume_plot</a></code> function uses the output of <code>run_ume</code> as an S3
object and processes it further to provide an end-user-ready output.
</p>
<p><code>run_ume</code> can be used only for a network of interventions. In the case
of two interventions, the execution of the function will be stopped and an
error message will be printed on the R console.
</p>


<h3>Value</h3>

<p>An R2jags output on the summaries of the posterior distribution, and
the Gelman-Rubin convergence diagnostic (Gelman et al., 1992) of the
following monitored parameters:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The summary effect estimate (according to the argument
<code>measure</code> defined in <code><a href="#topic+run_model">run_model</a></code>) for each pairwise
comparison observed in the network.</p>
</td></tr>
<tr><td><code>dev_o</code></td>
<td>
<p>The deviance contribution of each trial-arm based on the
observed outcome.</p>
</td></tr>
<tr><td><code>hat_par</code></td>
<td>
<p>The fitted outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>The between-trial standard deviation (assumed common across the
observed pairwise comparisons) for the whole network, when a random-effects
model has been specified.</p>
</td></tr>
<tr><td><code>m_tau</code></td>
<td>
<p>The between-trial standard deviation (assumed common
across the observed pairwise comparisons) for the subset of multi-arm
trials, when a random-effects model has been specified.</p>
</td></tr>
</table>
<p>The output also includes the following elements:
</p>
<table>
<tr><td><code>leverage_o</code></td>
<td>
<p>The leverage for the observed outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>sign_dev_o</code></td>
<td>
<p>The sign of the difference between observed and fitted
outcome at each trial-arm.</p>
</td></tr>
<tr><td><code>model_assessment</code></td>
<td>
<p>A data-frame on the measures of model assessment:
deviance information criterion, number of effective parameters, and total
residual deviance.</p>
</td></tr>
<tr><td><code>jagsfit</code></td>
<td>
<p>An object of S3 class <code><a href="R2jags.html#topic+jags">jags</a></code> with
the posterior results on all monitored parameters to be used in the
<code><a href="#topic+mcmc_diagnostics">mcmc_diagnostics</a></code> function.</p>
</td></tr>
</table>
<p>Furthermore, <code>run_ume</code> returns a character vector with the pairwise
comparisons observed in the network, <code>obs_comp</code>, and a character
vector with comparisons between the non-baseline interventions observed in
multi-arm trials only, <code>frail_comp</code>. Both vectors are used in
<code><a href="#topic+ume_plot">ume_plot</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Sutton AJ, Caldwell DM, Lu G, Ades AE. Evidence synthesis
for decision making 4: inconsistency in networks of evidence based on
randomized controlled trials.
<em>Med Decis Making</em> 2013;<b>33</b>(5):641&ndash;56.
doi: 10.1177/0272989X12455847
</p>
<p>Gelman A, Rubin DB. Inference from iterative simulation using multiple
sequences. <em>Stat Sci</em> 1992;<b>7</b>(4):457&ndash;72.
doi: 10.1214/ss/1177011136
</p>
<p>Spineli LM. A Revised Framework to Evaluate the Consistency Assumption
Globally in a Network of Interventions. <em>Med Decis Making</em> 2021.
doi: 10.1177/0272989X211068005
</p>
<p>Spineli LM, Kalyvas C, Papadimitropoulou K. Continuous(ly) missing outcome
data in network meta-analysis: a one-stage pattern-mixture model approach.
<em>Stat Methods Med Res</em> 2021;<b>30</b>(4):958&ndash;75.
doi: 10.1177/0962280220983544
</p>
<p>Spineli LM. An empirical comparison of Bayesian modelling strategies for
missing binary outcome data in network meta-analysis.
<em>BMC Med Res Methodol</em> 2019;<b>19</b>(1):86.
doi: 10.1186/s12874-019-0731-y
</p>


<h3>See Also</h3>

<p><code><a href="R2jags.html#topic+autojags">autojags</a></code>,
<code><a href="R2jags.html#topic+jags">jags</a></code>,
<code><a href="#topic+prepare_ume">prepare_ume</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_series_meta">run_series_meta</a></code>, <code><a href="#topic+ume_plot">ume_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.liu2013")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_liu.rds', package = 'rnmamod'))


# Run random-effects unrelated mean effects model
# Note: Ideally, set 'n_iter' to 10000 and 'n_burnin' to 1000
run_ume(full = res,
        n_chains = 3,
        n_iter = 1000,
        n_burnin = 100,
        n_thin = 1)


</code></pre>

<hr>
<h2 id='scatterplot_sucra'>Scatterplot of SUCRA values</h2><span id='topic+scatterplot_sucra'></span>

<h3>Description</h3>

<p>Creates a scatterplot of the SUCRA values from the
network meta-analysis and the network meta-regression for a specified level
or value of the investigated covariate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatterplot_sucra(full, reg, cov_value, drug_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatterplot_sucra_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>.
See 'Value' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="scatterplot_sucra_+3A_reg">reg</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_metareg">run_metareg</a></code>. See 'Value' in
<code><a href="#topic+run_metareg">run_metareg</a></code>.</p>
</td></tr>
<tr><td><code id="scatterplot_sucra_+3A_cov_value">cov_value</code></td>
<td>
<p>A list of two elements in the following order: 1) a number
for the covariate value of interest (see 'Arguments' in
<code><a href="#topic+run_metareg">run_metareg</a></code>), and 2) a character to indicate the name of
the covariate. See also 'Details'.</p>
</td></tr>
<tr><td><code id="scatterplot_sucra_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The names of the interventions appear above each point in the plot.
Three coloured rectangles are drawn in the scatterplot: a red rectangle for
SUCRA values up to 50%, a yellow rectangular for SUCRA values between
50% and 80%, and a green rectangle for SUCRA values over 80%.
Interventions falling at the green area are considered as the highest
ranked interventions, whilst interventions falling at the red area are
considered as the lowest ranked interventions.
</p>
<p>When the covariate is binary, specify in the second element of
<code>cov_value</code> the name of the level for which the scatterplot will be
created.
</p>
<p><code>scatterplot_sucra</code> is integrated in <code><a href="#topic+metareg_plot">metareg_plot</a></code>.
</p>
<p><code>scatterplot_sucra</code> can be used only for a network of interventions.
Otherwise, the execution of the function will be stopped and an error
message will be printed on the R console.
</p>


<h3>Value</h3>

<p>A scatterplot of the SUCRA values under the network meta-analysis
(y-axis) against the SUCRA values under the network meta-regression
(x-axis) for a specified level or value of the investigated covariate.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Salanti G, Ades AE, Ioannidis JP. Graphical methods and numerical summaries
for presenting results from multiple-treatment meta-analysis: an overview and
tutorial. <em>J Clin Epidemiol</em> 2011;<b>64</b>(2):163&ndash;71.
doi: 10.1016/j.jclinepi.2010.03.016
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metareg_plot">metareg_plot</a></code>, <code><a href="#topic+run_metareg">run_metareg</a></code>,
<code><a href="#topic+run_model">run_model</a></code>
</p>

<hr>
<h2 id='scatterplots_dev'>Deviance scatterplots</h2><span id='topic+scatterplots_dev'></span>

<h3>Description</h3>

<p>Illustrates the posterior mean of deviance contribution of the
individual data points under the unrelated mean effects model (via
<code><a href="#topic+run_ume">run_ume</a></code>) against the posterior mean of deviance contribution
under the consistency model (via <code><a href="#topic+run_model">run_model</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatterplots_dev(full, ume, colour)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatterplots_dev_+3A_full">full</code></td>
<td>
<p>A numeric vector with the posterior mean of deviance obtained
using the consistency model (see 'Value' in <code><a href="#topic+run_model">run_model</a></code>).</p>
</td></tr>
<tr><td><code id="scatterplots_dev_+3A_ume">ume</code></td>
<td>
<p>A numeric vector with the posterior mean of deviance obtained
using the unrelated mean effects model (see 'Value' in
<code><a href="#topic+run_ume">run_ume</a></code>).</p>
</td></tr>
<tr><td><code id="scatterplots_dev_+3A_colour">colour</code></td>
<td>
<p>A string to define the colour of the points in the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>scatterplots_dev</code> is integrated in the <code><a href="#topic+ume_plot">ume_plot</a></code>
function to compare the models regarding the posterior mean of deviance.
This scatterplot has also been considered by Dias et al. (2013).
When the majority of data points are scattered across the diagonal line,
we may conclude that the compared models have a good agreement. Data points
systematically scattered above or below the diagonal line may
contribute more to the poor fit of the unrelated mean effects model and the
consistency model, respectively.
</p>


<h3>Value</h3>

<p>A scatterplot of the posterior mean deviance contribution of the
individual data points from the unrelated mean effects model against those
from the consistency model. Each data point corresponds to a trial-arm
indicated by a pair of numbers. The first number refers to the position
of the trial in the dataset, and the second number refers to the
corresponding trial-arm (see 'Arguments' and 'Value' in
<code><a href="#topic+data_preparation">data_preparation</a></code>).
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Sutton AJ, Caldwell DM, Lu G, Ades AE. Evidence synthesis
for decision making 4: inconsistency in networks of evidence based on
randomized controlled trials.
<em>Med Decis Making</em> 2013a;<b>33</b>(5):641&ndash;56.
doi: 10.1177/0272989X12455847
</p>


<h3>See Also</h3>

<p><code><a href="#topic+data_preparation">data_preparation</a></code>, <code><a href="#topic+run_model">run_model</a></code>,
<code><a href="#topic+run_ume">run_ume</a></code>, <code><a href="#topic+ume_plot">ume_plot</a></code>
</p>

<hr>
<h2 id='series_meta_plot'>End-user-ready results for a series of pairwise meta-analyses</h2><span id='topic+series_meta_plot'></span>

<h3>Description</h3>

<p>Facilitates the comparison of the consistency model
(via <code><a href="#topic+run_model">run_model</a></code>) with a series of pairwise meta-analyses
(via <code><a href="#topic+run_series_meta">run_series_meta</a></code>) regarding the estimated summary effect
sizes and between-trial standard deviation for comparisons with at
least two trials.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>series_meta_plot(full, meta, drug_names, save_xls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="series_meta_plot_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="series_meta_plot_+3A_meta">meta</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_series_meta">run_series_meta</a></code>. See 'Value'
in <code><a href="#topic+run_series_meta">run_series_meta</a></code>.</p>
</td></tr>
<tr><td><code id="series_meta_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
<tr><td><code id="series_meta_plot_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) at the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>series_meta_plot</code> can be used only for a network of
interventions. Otherwise, the execution of the function will be stopped and
an error message will be printed on the R console.
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>series_meta_plot</code>
currently presents the results in the odds ratio for being the
<strong>base-case</strong> effect measure in <code><a href="#topic+run_model">run_model</a></code> for a binary
outcome (see also 'Details' in <code><a href="#topic+run_model">run_model</a></code>).
</p>
<p>The user can detect any inconsistencies in the estimated
effects from the compared models and explore the gains in precision
stemming from applying network meta-analysis. Furthermore, the user can
investigate the plausibility of the common between-trial heterogeneity
assumption which is typically considered in network meta-analysis.
</p>


<h3>Value</h3>

<p>The R console prints the data-frame with the estimated summary effect
sizes and between-trial standard deviation of comparisons under both
models. The comparisons have at least two trials. In the case of a
fixed-effect model, the data-frame is printed without the results on the
between-trial standard deviation.
</p>
<p>Furthermore, <code>series_meta_plot</code> exports the data-frame to an 'xlsx'
file at the working  directory of the user.
</p>
<p><code>series_meta_plot</code> returns a panel of two forest plots: (1) a
forest plot on the posterior median and 95% credible interval of the summary
effect size for the observed comparisons from network meta-analysis and the
corresponding pairwise meta-analyses, and (2) a forest plot on the
posterior median and 95% credible interval of the between-trial standard
deviation for these observed comparisons. The estimated median and 95%
credible intervals of the between-trial standard deviation from network
meta-analysis appear in the forest plot as a solid and two dotted parallel
blue lines, respectively. The different levels of heterogeneity appear as
green, yellow, orange, and red rectangles to indicate a low, reasonable,
fairly high, and fairly extreme heterogeneity, respectively, following the
classification of Spiegelhalter et al. (2004).
When a fixed-effect model has been fitted, only the forest plot on the
estimated summary effect sizes is shown.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Spiegelhalter DJ, Abrams KR, Myles JP. Bayesian approaches to clinical trials
and health-care evaluation. John Wiley and Sons, Chichester, 2004.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_series_meta">run_series_meta</a></code>,
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.dogliotti2014")

# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_dogliotti.rds', package = 'rnmamod'))

# Read results from 'run_series_meta' (using the default arguments)
meta &lt;- readRDS(system.file('extdata/meta_dogliotti.rds',
                package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "aspirin", "aspirin plus clopidogrel",
                  "dabigatran 110 mg", "dabigatran 150 mg", "rivaroxaban",
                  "vitamin K antagonist", "apixaban")

# Plot the results from both models
series_meta_plot(full = res,
                 meta = meta,
                 drug_names = interv_names)

</code></pre>

<hr>
<h2 id='taylor_continuous'>Pattern-mixture model with Taylor series for continuous outcome</h2><span id='topic+taylor_continuous'></span>

<h3>Description</h3>

<p>Applies the pattern-mixture model under a specific assumption
about the informative missingness parameter in trial-arms with
<b>continuous</b> missing participant outcome data and uses the Taylor
series to obtain the effect size and standard error for each trial
(Mavridis et al., 2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taylor_continuous(data, measure, mean_value, var_value, rho)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taylor_continuous_+3A_data">data</code></td>
<td>
<p>A data-frame in the long arm-based format. Two-arm trials occupy
one row in the data-frame. Multi-arm trials occupy as many rows as the
number of possible comparisons among the interventions. See 'Format' for
the specification of the columns.</p>
</td></tr>
<tr><td><code id="taylor_continuous_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure with values
<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for the mean difference,
standardised mean difference, and ratio of means, respectively.</p>
</td></tr>
<tr><td><code id="taylor_continuous_+3A_mean_value">mean_value</code></td>
<td>
<p>A numeric value for the mean of the normal distribution of
the informative missingness parameter. The same value is considered for all
trial-arms of the dataset. The default argument is 0 and corresponds to the
missing-at-random assumption. For the informative missingness ratio of
means, the mean value is defined in the logarithmic scale.</p>
</td></tr>
<tr><td><code id="taylor_continuous_+3A_var_value">var_value</code></td>
<td>
<p>A positive non-zero number for the variance of the normal
distribution of the informative missingness parameter. When the
<code>measure</code> is <code>"MD"</code>, or <code>"SMD"</code> the default argument is 1;
when the <code>measure</code> is <code>"ROM"</code> the default argument is 0.04. The
same value is considered for all trial-arms of the dataset.</p>
</td></tr>
<tr><td><code id="taylor_continuous_+3A_rho">rho</code></td>
<td>
<p>A numeric value in the interval [-1, 1] that indicates the
correlation coefficient between two informative missingness parameters in
a trial. The same value is considered across all trials of the dataset.
The default argument is 0 and corresponds to uncorrelated missingness
parameters.</p>
</td></tr>
</table>


<h3>Format</h3>

<p>The columns of the data-frame in the argument <code>data</code> refer to
the following ordered elements for a continuous outcome:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <b>id</b> </td><td style="text-align: left;"> A unique identifier for each trial.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>y1</b> </td><td style="text-align: left;"> The observed mean outcome in the first arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>y2</b> </td><td style="text-align: left;"> The observed mean outcome in the second arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>sd1</b> </td><td style="text-align: left;"> The observed standard deviation of the outcome in the
  first arm of the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>sd2</b> </td><td style="text-align: left;"> The observed standard deviation of the outcome in the
  second arm of the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>m1</b> </td><td style="text-align: left;"> The number of missing participants in the first arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>m2</b> </td><td style="text-align: left;"> The number of missing participants in the second arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>n1</b> </td><td style="text-align: left;"> The number randomised in the first arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>n2</b> </td><td style="text-align: left;"> The number randomised in the second arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>t1</b> </td><td style="text-align: left;"> An identifier for the intervention in the first arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>t2</b> </td><td style="text-align: left;"> An identifier for the intervention in the second arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>



<h3>Details</h3>

<p>The <code>taylor_continuous</code> function is integrated in the
<code><a href="#topic+unrelated_effects_plot">unrelated_effects_plot</a></code> function.
</p>


<h3>Value</h3>

<p>A data-frame that additionally includes the following elements:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The effect size adjusted for the missing participants and
obtained using the Taylor series.</p>
</td></tr>
<tr><td><code>se.EM</code></td>
<td>
<p>The standard error of the effect size adjusted for the missing
participants and obtained using the Taylor series.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Mavridis D, White IR, Higgins JP, Cipriani A, Salanti G. Allowing for
uncertainty due to missing continuous outcome data in pairwise and network
meta-analysis. <em>Stat Med</em> 2015;<b>34</b>(5):721&ndash;41.
doi: 10.1002/sim.6365
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+unrelated_effects_plot">unrelated_effects_plot</a></code>
</p>

<hr>
<h2 id='taylor_imor'>Pattern-mixture model with Taylor series for a binary outcome</h2><span id='topic+taylor_imor'></span>

<h3>Description</h3>

<p>Applies the pattern-mixture model under a specific assumption
about the informative missingness odds ratio in trial-arms with
<b>binary</b> missing participant outcome data and uses the Taylor
series to obtain the odds ratio (in the logarithmic scale) and standard
error for each trial (White et al., 2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taylor_imor(data, mean_value, var_value, rho)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taylor_imor_+3A_data">data</code></td>
<td>
<p>A data-frame in the long arm-based format. Two-arm trials occupy
one row in the data-frame. Multi-arm trials occupy as many rows as the
number of possible comparisons among the interventions. See 'Format' for
the specification of the columns.</p>
</td></tr>
<tr><td><code id="taylor_imor_+3A_mean_value">mean_value</code></td>
<td>
<p>A numeric value for the mean of the normal distribution of
the informative missingness odds ratio in the logarithmic scale. The same
value is considered for all trial-arms of the dataset. The default argument
is 0 and corresponds to the missing-at-random assumption.</p>
</td></tr>
<tr><td><code id="taylor_imor_+3A_var_value">var_value</code></td>
<td>
<p>A positive non-zero number for the variance of the normal
distribution of the informative missingness odds ratio in the logarithmic
scale. The default argument is 1.</p>
</td></tr>
<tr><td><code id="taylor_imor_+3A_rho">rho</code></td>
<td>
<p>A numeric value in the interval [-1, 1] that indicates the
correlation coefficient between two missingness parameters in a trial. The
same value is considered across all trials of the dataset. The default
argument is 0 and corresponds to uncorrelated missingness parameters.</p>
</td></tr>
</table>


<h3>Format</h3>

<p>The columns of the data-frame in the argument <code>data</code> refer to
the following ordered elements for a binary outcome:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <b>id</b> </td><td style="text-align: left;"> A unique identifier for each trial.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>r1</b> </td><td style="text-align: left;"> The observed  number of events in the first arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>r2</b> </td><td style="text-align: left;"> The observed  number of events in the second arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>m1</b> </td><td style="text-align: left;"> The number of missing participants in the first arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>m2</b> </td><td style="text-align: left;"> The number of missing participants in the second arm of the
  comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>n1</b> </td><td style="text-align: left;"> The number of participants randomised in the first arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>n2</b> </td><td style="text-align: left;"> The number of participants randomised in the second arm
  of the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>t1</b> </td><td style="text-align: left;"> An identifier for the intervention in the first arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  <b>t2</b> </td><td style="text-align: left;"> An identifier for the intervention in the second arm of
  the comparison.</td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>



<h3>Details</h3>

<p>The <code>taylor_imor</code> function is integrated in the
<code><a href="#topic+unrelated_effects_plot">unrelated_effects_plot</a></code> function.
</p>


<h3>Value</h3>

<p>A data-frame that additionally includes the following elements:
</p>
<table>
<tr><td><code>EM</code></td>
<td>
<p>The odds ratio in the logarithmic scale (log OR) adjusted
for missing participants and obtained using the Taylor series.</p>
</td></tr>
<tr><td><code>se.EM</code></td>
<td>
<p>The standard error of the log OR adjusted for missing
participants and obtained using the Taylor series.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>White IR, Higgins JP, Wood AM. Allowing for uncertainty due to missing data
in meta-analysis&ndash;part 1: two-stage methods.
<em>Stat Med</em> 2008;<b>27</b>(5):711&ndash;27. doi: 10.1002/sim.3008
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+unrelated_effects_plot">unrelated_effects_plot</a></code>
</p>

<hr>
<h2 id='trans_quality'>Rating the quality of transitivity evaluation</h2><span id='topic+trans_quality'></span>

<h3>Description</h3>

<p>Classifies a systematic review with multiple interventions as
having low, unclear or high quality regarding the transitivity evaluation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trans_quality(
  plan_protocol,
  plan_methods,
  report_results,
  discuss_trans,
  proper_table
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trans_quality_+3A_plan_protocol">plan_protocol</code></td>
<td>
<p>Character string that indicates how the systematic
review planned in the protocol to assess the transitivity assumption. The
following values can be considered: <code>"Both"</code>, <code>"No"</code>,
<code>"No protocol"</code>, <code>"Only direct methods"</code>, and
<code>"Only indirect methods"</code>. <code>"Both"</code> implies that at least one
direct and one indirect method has been used.</p>
</td></tr>
<tr><td><code id="trans_quality_+3A_plan_methods">plan_methods</code></td>
<td>
<p>Character string that indicates whether the systematic
review described in the methods section a strategy to assess the
transitivity assumption. The following values can be considered:
<code>"Yes"</code>, and <code>"No"</code>.</p>
</td></tr>
<tr><td><code id="trans_quality_+3A_report_results">report_results</code></td>
<td>
<p>Character string that indicates whether the systematic
review reported in the results section the transitivity evaluation and
which strategy was employed. The following values can be considered:
<code>"Both"</code>, <code>"No"</code>, <code>"Only direct methods"</code>, and
<code>"Only indirect methods"</code>. <code>"Both"</code> implies that at least one
direct and one indirect method has been used.</p>
</td></tr>
<tr><td><code id="trans_quality_+3A_discuss_trans">discuss_trans</code></td>
<td>
<p>Character string that indicates whether the systematic
review discussed the transitivity assumption and which model parameters
where considered. The following values can be considered: <code>"Both"</code>,
<code>"No"</code>, <code>"Only treatment effects"</code>, <code>"Other parameter"</code>,
and <code>"NMA not conducted"</code>. <code>"Both"</code> implies that the treatment
effect and other model parameters were considered.</p>
</td></tr>
<tr><td><code id="trans_quality_+3A_proper_table">proper_table</code></td>
<td>
<p>Character string that indicates whether the systematic
review reported a proper table of characteristics. The following values can
be considered: <code>"No"</code>, <code>"No table"</code>, and <code>"Yes"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A systematic review with <code>"Low"</code> quality of transitivity
evaluation does not provide a protocol, nor describe the evaluation
strategy in the methods section, does not report the evaluation results,
nor discusses the transitivity evaluation and does not provide a table of
characteristics. On the contrary, a systematic review with <code>"High"</code>
quality of transitivity evaluation provides an evaluation plan in the
protocol (including at least one direct method), describes the evaluation
strategy in the methods section (including at least one direct method),
reports the evaluation results in the results section, discusses the
transitivity evaluation while considering at least one model parameter
(if NMA has been conducted), and provides a proper table of
characteristics. Otherwise, the systematic review is judged to have an
<code>"Unclear"</code> quality of transitivity evaluation.
</p>


<h3>Value</h3>

<p>A character with value <code>"Low"</code>, <code>"Unclear"</code>, or
<code>"High"</code> to indicate low, unclear, or high-quality of transitivity
evaluation.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Spineli LM, Kalyvas C, Seide SE, Papadimitropoulou K. Low awareness of the
transitivity assumption in complex networks of interventions: empirical
evidence from 356 network meta-analyses. 2022 <em>submitted</em>
</p>

<hr>
<h2 id='ume_plot'>End-user-ready results for the unrelated mean effects model</h2><span id='topic+ume_plot'></span>

<h3>Description</h3>

<p><code>ume_plot</code> hosts a toolkit of functions that facilitates
the comparison of the consistency model (via <code><a href="#topic+run_model">run_model</a></code>) with
the unrelated mean effects model (via <code><a href="#topic+run_ume">run_ume</a></code>) regarding the
posterior summaries of the summary effect size for the pairwise comparisons
observed in the network, the between-trial standard deviation (<em>tau</em>)
and model assessment parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ume_plot(full, ume, drug_names, save_xls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ume_plot_+3A_full">full</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_model">run_model</a></code>. See 'Value' in
<code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="ume_plot_+3A_ume">ume</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+run_ume">run_ume</a></code>. See 'Value' in
<code><a href="#topic+run_ume">run_ume</a></code>.</p>
</td></tr>
<tr><td><code id="ume_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code> of
<code><a href="#topic+run_model">run_model</a></code>. If <code>drug_names</code> is not defined,
the order of the interventions as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
<tr><td><code id="ume_plot_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) to the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The deviance information criterion (DIC) of the consistency model is
compared with the DIC of the unrelated mean effects model
(Dias et al., 2013). If the difference in DIC exceeds 5, the unrelated mean
effects model is preferred. If the difference in DIC is less than -5, the
consistency is preferred; otherwise, there is little to choose between the
compared models.
</p>
<p>For a binary outcome, when <code>measure</code> is &quot;RR&quot; (relative risk) or &quot;RD&quot;
(risk difference) in <code><a href="#topic+run_model">run_model</a></code>, <code>ume_plot</code> currently
presents the results from network meta-analysis and unrelated mean effects
in the odds ratio for being the <strong>base-case</strong> effect measure in
<code><a href="#topic+run_model">run_model</a></code> for a binary outcome (see also 'Details' in
<code><a href="#topic+run_model">run_model</a></code>).
</p>
<p>Furthermore, <code>ume_plot</code> exports <code>table_effect_size</code> and
<code>table_model_assessment</code> to separate 'xlsx' files (via the
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code> function) to the working
directory of the user.
</p>
<p><code>ume_plot</code> can be used only for a network of interventions. In the
case of two interventions, the execution of the function will be stopped
and an error message will be printed on the R console.
</p>


<h3>Value</h3>

<p><code>ume_plot</code> prints on the R console a message on the most
parsimonious model (if any) based on the DIC (red text). Then, the function
returns the following list of elements:
</p>
<table>
<tr><td><code>table_effect_size</code></td>
<td>
<p>The posterior median, posterior standard
deviation, and 95% credible interval of the summary effect size for each
pairwise comparison observed in the network under the consistency model
and the unrelated mean effects model.</p>
</td></tr>
<tr><td><code>table_model_assessment</code></td>
<td>
<p>The DIC, number of effective
parameters, and total residual deviance under the consistency model and
the unrelated mean effects model (Spiegelhalter et al., 2002).</p>
</td></tr>
<tr><td><code>table_tau</code></td>
<td>
<p>The posterior median and 95% credible interval of
<em>tau</em> under the consistency model and the unrelated mean effects
model. When a fixed-effect model has been performed, <code>ume_plot</code> does
not return this element.</p>
</td></tr>
<tr><td><code>scatterplots</code></td>
<td>
<p>The scatterplot and the Bland-Altman plot on the
posterior mean deviance contribution of the individual data points under
the consistency model and the unrelated mean effects model. See 'Details'
and 'Value' in <code><a href="#topic+scatterplots_dev">scatterplots_dev</a></code> and
<code><a href="#topic+bland_altman_plot">bland_altman_plot</a></code>, respectively.</p>
</td></tr>
<tr><td><code>levarage_plots</code></td>
<td>
<p>The leverage plot under the consistency model
and the unrelated mean effects model, separately. See 'Details' and
'Value' in <code><a href="#topic+leverage_plot">leverage_plot</a></code>.</p>
</td></tr>
<tr><td><code>intervalplots</code></td>
<td>
<p>A panel of interval plots on the summary effect
size under the consistency model and the unrelated mean effects model for
each pairwise comparison observed in the network. See 'Details' and
'Value' in <code><a href="#topic+intervalplot_panel_ume">intervalplot_panel_ume</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Dias S, Welton NJ, Sutton AJ, Caldwell DM, Lu G, Ades AE. Evidence synthesis
for decision making 4: inconsistency in networks of evidence based on
randomized controlled trials.
<em>Med Decis Making</em> 2013;<b>33</b>(5):641&ndash;56.
doi: 10.1177/0272989X12455847
</p>
<p>Spiegelhalter DJ, Best NG, Carlin BP, van der Linde A. Bayesian measures of
model complexity and fit. <em>J R Stat Soc B</em> 2002;<b>64</b>(4):583&ndash;396.
doi: 10.1111/1467-9868.00353
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bland_altman_plot">bland_altman_plot</a></code>,
<code><a href="#topic+intervalplot_panel_ume">intervalplot_panel_ume</a></code>, <code><a href="#topic+leverage_plot">leverage_plot</a></code>,
<code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+run_ume">run_ume</a></code>,
<code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nma.liu2013")


# Read results from 'run_model' (using the default arguments)
res &lt;- readRDS(system.file('extdata/res_liu.rds', package = 'rnmamod'))

# Read results from 'run_ume' (using the default arguments)
ume &lt;- readRDS(system.file('extdata/ume_liu.rds', package = 'rnmamod'))

# The names of the interventions in the order they appear in the dataset
interv_names &lt;- c("placebo", "pramipexole", "serotonin-norepinephrine
                  reuptake inhibitor", "serotonin reuptake inhibitor",
                  "tricyclic antidepressant", "pergolide")

# Plot the results from both models
ume_plot(full = res,
         ume = ume,
         drug_names = interv_names)


</code></pre>

<hr>
<h2 id='unrelated_effects_plot'>End-user-ready results for unrelated trial effects model</h2><span id='topic+unrelated_effects_plot'></span>

<h3>Description</h3>

<p>Performs the unrelated trial effects model (also known as fixed
effects model) and illustrates the results of each trial and corresponding
pairwise comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unrelated_effects_plot(
  data,
  measure,
  char,
  drug_names,
  trial_names,
  mean_misspar,
  var_misspar,
  rho,
  save_xls
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unrelated_effects_plot_+3A_data">data</code></td>
<td>
<p>A data-frame of a one-trial-per-row format containing arm-level
data of each trial. See 'Format' in <code><a href="#topic+run_model">run_model</a></code>.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_measure">measure</code></td>
<td>
<p>Character string indicating the effect measure with values
<code>"OR"</code>, <code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code> for the odds ratio,
mean difference, standardised mean difference and ratio of means,
respectively.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_char">char</code></td>
<td>
<p>A data-frame of three columns and number of rows equal to the
number of trials in <code>data</code>. Each column refers to a
trial-characteristic with <strong>nominal</strong> elements.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_drug_names">drug_names</code></td>
<td>
<p>A vector of labels with the name of the interventions in
the order they appear in the argument <code>data</code>. If <code>drug_names</code> is
not defined, the order of the interventions as they appear in <code>data</code>
is used, instead.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_trial_names">trial_names</code></td>
<td>
<p>A vector of labels with the name of the trials in the
order they appear in the argument <code>data</code>. If <code>trial_names</code> is not
defined, the order of the trials as they appear in <code>data</code> is used,
instead.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_mean_misspar">mean_misspar</code></td>
<td>
<p>A numeric value for the mean of the normal distribution
of the informative missingness parameter (see 'Details'). The default
argument is 0 and corresponds to the missing-at-random assumption. The same
value is considered across all trials of the dataset.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_var_misspar">var_misspar</code></td>
<td>
<p>A positive non-zero number for the variance of the
normal distribution of the informative missingness parameter.
When the <code>measure</code> is <code>"OR"</code>, <code>"MD"</code>, or <code>"SMD"</code>
the default argument is 1. When the <code>measure</code> is <code>"ROM"</code>
the default argument is 0.04. The same value is considered across all
trials of the dataset.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_rho">rho</code></td>
<td>
<p>A numeric value in the interval [-1, 1] that indicates the
correlation coefficient between two informative missingness parameters in
a trial. The same value is considered across all trials of the dataset.
The default argument is 0 and corresponds to uncorrelated missingness
parameters.</p>
</td></tr>
<tr><td><code id="unrelated_effects_plot_+3A_save_xls">save_xls</code></td>
<td>
<p>Logical to indicate whether to export the tabulated results
to an 'xlsx' file (via the <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
function of the R-package
<a href="https://CRAN.R-project.org/package=writexl">writexl</a>) to the working
directory of the user. The default is <code>FALSE</code> (do not export).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The unrelated trial effects model may be an alternative to network
meta-analysis, when the latter is not deemed appropriate (e.g., there is
considerable statistical heterogeneity, or substantial intransitivity). In
the presence of missing participant outcome data, the effect size and
standard error are adjusted by applying the pattern-mixture model with
Taylor series in trial-arms with reported missing participants (Mavridis et
al., 2015; White et al., 2008). The <code>unrelated_effects_plot</code> function
calls the <code><a href="#topic+taylor_imor">taylor_imor</a></code> and <code><a href="#topic+taylor_continuous">taylor_continuous</a></code>
functions (for a binary and continuous outcome, respectively) to employ
pattern-mixture model with Taylor series. The <code>unrelated_effects_plot</code>
function considers the informative missingness odds ratio in the
logarithmic scale for binary outcome data (White et al., 2008), the
informative missingness difference of means when <code>measure</code> is
<code>"MD"</code> or <code>"SMD"</code>, and the informative missingness ratio of means
in the logarithmic scale when <code>measure</code> is <code>"ROM"</code>
(Mavridis et al., 2015).
</p>
<p>The number of interval plots equals the number of observed comparisons in
the network. In each interval plot, the y-axis refers to all trials of the
network and x-axis refers to the selected effect measure. The odds ratio
and ratio of means are calculated in the logarithmic scale but they are
reported in their original scale after exponentiation.
</p>
<p><code>unrelated_effects_plot</code> depicts all three characteristics for each
trial using different colours, line-types and point-shapes for the
corresponding 95% confidence interval and point estimate. Ideally, each
characteristic should have no more than three categories; otherwise, the
plot becomes cluttered. For now, the <code>unrelated_effects_plot</code> function
uses the default colour palette, line-types and point-shapes.
</p>


<h3>Value</h3>

<p>A panel of interval plots for each observed comparison in the
network, when there are up to 15 trials in the <code>data</code>. Otherwise,
<code>unrelated_effects_plot</code> exports a data-frame to an 'xlsx' file at
the working  directory of the user. This data-frame includes the
<code>data</code> in the long format, the within-trial effect measure and
95% confidence interval of the corresponding comparisons, the
interventions compared, and the three characteristics (as defined in
<code>char</code>).
For datasets with more than 15 trials, the plot becomes cluttered and it is
difficult to identify the trial-names. Hence, exporting the results in an
Excel file is a viable alternative.
</p>


<h3>Author(s)</h3>

<p>Loukia M. Spineli
</p>


<h3>References</h3>

<p>Mavridis D, White IR, Higgins JP, Cipriani A, Salanti G. Allowing for
uncertainty due to missing continuous outcome data in pairwise and network
meta-analysis. <em>Stat Med</em> 2015;<b>34</b>(5):721&ndash;41.
doi: 10.1002/sim.6365
</p>
<p>White IR, Higgins JP, Wood AM. Allowing for uncertainty due to missing data
in meta-analysis&ndash;part 1: two-stage methods.
<em>Stat Med</em> 2008;<b>27</b>(5):711&ndash;27. doi: 10.1002/sim.3008
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_model">run_model</a></code>, <code><a href="#topic+taylor_continuous">taylor_continuous</a></code>,
<code><a href="#topic+taylor_imor">taylor_imor</a></code>, <code><a href="writexl.html#topic+write_xlsx">write_xlsx</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
