<!DOCTYPE html><html><head><title>Help for package RNOmni</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RNOmni}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BasicInputChecks'><p>Basic Input Checks</p></a></li>
<li><a href='#BAT'><p>Basic Association Test</p></a></li>
<li><a href='#CauchyToP'><p>Convert Cauchy Random Variable to P</p></a></li>
<li><a href='#DINT'><p>Direct-INT</p></a></li>
<li><a href='#FitOLS'><p>Ordinary Least Squares</p></a></li>
<li><a href='#IINT'><p>Indirect-INT</p></a></li>
<li><a href='#OINT'><p>Omnibus-INT</p></a></li>
<li><a href='#OmniP'><p>Omnibus P-value.</p></a></li>
<li><a href='#PartitionData'><p>Partition Data</p></a></li>
<li><a href='#PtoCauchy'><p>Convert P-value to Cauchy Random</p></a></li>
<li><a href='#RankNorm'><p>Rank-Normalize</p></a></li>
<li><a href='#RNOmni-package'><p>RNOmni: Rank Normal Transformation Omnibus Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Rank Normal Transformation Omnibus Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-10</td>
</tr>
<tr>
<td>Description:</td>
<td>Inverse normal transformation (INT) based genetic association testing. These tests are recommend for continuous traits with non-normally distributed residuals. INT-based tests robustly control the type I error in settings where standard linear regression does not, as when the residual distribution exhibits excess skew or kurtosis. Moreover, INT-based tests outperform standard linear regression in terms of power. These tests may be classified into two types. In direct INT (D-INT), the phenotype is itself transformed. In indirect INT (I-INT), phenotypic residuals are transformed. The omnibus test (O-INT) adaptively combines D-INT and I-INT into a single robust and statistically powerful approach. See McCaw ZR, Lane JM, Saxena R, Redline S, Lin X. "Operating characteristics of the rank-based inverse normal transformation for quantitative trait analysis in genome-wide association studies" &lt;<a href="https://doi.org/10.1111%2Fbiom.13214">doi:10.1111/biom.13214</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>plyr, Rcpp, stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), withr, R.rsp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-11 00:26:43 UTC; zmccaw</td>
</tr>
<tr>
<td>Author:</td>
<td>Zachary McCaw <a href="https://orcid.org/0000-0002-2006-9828"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zachary McCaw &lt;zmccaw@alumni.harvard.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-11 03:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BasicInputChecks'>Basic Input Checks</h2><span id='topic+BasicInputChecks'></span>

<h3>Description</h3>

<p>Stops evaluation if inputs are improperly formatted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BasicInputChecks(y, G, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BasicInputChecks_+3A_y">y</code></td>
<td>
<p>Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="BasicInputChecks_+3A_g">G</code></td>
<td>
<p>Genotype matrix with observations as rows, SNPs as columns.</p>
</td></tr>
<tr><td><code id="BasicInputChecks_+3A_x">X</code></td>
<td>
<p>Covariate matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>

<hr>
<h2 id='BAT'>Basic Association Test</h2><span id='topic+BAT'></span>

<h3>Description</h3>

<p>Conducts tests of association between the loci in <code>G</code> and the
untransformed phenotype <code>y</code>, adjusting for the model matrix <code>X</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BAT(y, G, X = NULL, test = "Score", simple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BAT_+3A_y">y</code></td>
<td>
<p>Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="BAT_+3A_g">G</code></td>
<td>
<p>Genotype matrix with observations as rows, SNPs as columns.</p>
</td></tr>
<tr><td><code id="BAT_+3A_x">X</code></td>
<td>
<p>Model matrix of covariates and structure adjustments. Should include
an intercept. Omit to perform marginal tests of association.</p>
</td></tr>
<tr><td><code id="BAT_+3A_test">test</code></td>
<td>
<p>Either Score or Wald.</p>
</td></tr>
<tr><td><code id="BAT_+3A_simple">simple</code></td>
<td>
<p>Return the p-values only?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>simple = TRUE</code>, returns a vector of p-values, one for each column
of <code>G</code>. If <code>simple = FALSE</code>, returns a numeric matrix, including the
Wald or Score statistic, its standard error, the Z-score, and the p-value.
</p>


<h3>See Also</h3>


<ul>
<li><p> Direct INT <code><a href="#topic+DINT">DINT</a></code>
</p>
</li>
<li><p> Indirect INT <code><a href="#topic+IINT">IINT</a></code>
</p>
</li>
<li><p> Omnibus INT <code><a href="#topic+OINT">OINT</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
# Design matrix
X &lt;- cbind(1, stats::rnorm(1e3))
# Genotypes
G &lt;- replicate(1e3, stats::rbinom(n = 1e3, size = 2, prob = 0.25))
storage.mode(G) &lt;- "numeric"
# Phenotype
y &lt;- as.numeric(X %*% c(1, 1)) + stats::rnorm(1e3)
# Association test
p &lt;- BAT(y = y, G = G, X = X)
</code></pre>

<hr>
<h2 id='CauchyToP'>Convert Cauchy Random Variable to P</h2><span id='topic+CauchyToP'></span>

<h3>Description</h3>

<p>Convert Cauchy Random Variable to P
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CauchyToP(z)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CauchyToP_+3A_z">z</code></td>
<td>
<p>Numeric Cauchy random variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric p-value.
</p>

<hr>
<h2 id='DINT'>Direct-INT</h2><span id='topic+DINT'></span>

<h3>Description</h3>

<p>Applies the rank-based inverse normal transformation (<code><a href="#topic+RankNorm">RankNorm</a></code>)
to the phenotype <code>y</code>. Conducts tests of association between the loci in
<code>G</code> and transformed phenotype, adjusting for the model matrix <code>X</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DINT(
  y,
  G,
  X = NULL,
  k = 0.375,
  test = "Score",
  ties.method = "average",
  simple = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DINT_+3A_y">y</code></td>
<td>
<p>Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="DINT_+3A_g">G</code></td>
<td>
<p>Genotype matrix with observations as rows, SNPs as columns.</p>
</td></tr>
<tr><td><code id="DINT_+3A_x">X</code></td>
<td>
<p>Model matrix of covariates and structure adjustments. Should include
an intercept. Omit to perform marginal tests of association.</p>
</td></tr>
<tr><td><code id="DINT_+3A_k">k</code></td>
<td>
<p>Offset applied during rank-normalization. See
<code><a href="#topic+RankNorm">RankNorm</a></code>.</p>
</td></tr>
<tr><td><code id="DINT_+3A_test">test</code></td>
<td>
<p>Either Score or Wald.</p>
</td></tr>
<tr><td><code id="DINT_+3A_ties.method">ties.method</code></td>
<td>
<p>Method of breaking ties, passed to <code>base::rank</code>.</p>
</td></tr>
<tr><td><code id="DINT_+3A_simple">simple</code></td>
<td>
<p>Return the p-values only?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>simple = TRUE</code>, returns a vector of p-values, one for each column
of <code>G</code>. If <code>simple = FALSE</code>, returns a numeric matrix, including the
Wald or Score statistic, its standard error, the Z-score, and the p-value.
</p>


<h3>See Also</h3>


<ul>
<li><p> Basic association test <code><a href="#topic+BAT">BAT</a></code>.
</p>
</li>
<li><p> Indirect INT test <code><a href="#topic+IINT">IINT</a></code>.
</p>
</li>
<li><p> Omnibus INT test <code><a href="#topic+OINT">OINT</a></code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
# Design matrix
X &lt;- cbind(1, stats::rnorm(1e3))
# Genotypes
G &lt;- replicate(1e3, stats::rbinom(n = 1e3, size = 2, prob = 0.25))
storage.mode(G) &lt;- "numeric"
# Phenotype
y &lt;- exp(as.numeric(X %*% c(1, 1)) + stats::rnorm(1e3))
# Association test
p &lt;- DINT(y = y, G = G, X = X)
</code></pre>

<hr>
<h2 id='FitOLS'>Ordinary Least Squares</h2><span id='topic+FitOLS'></span>

<h3>Description</h3>

<p>Fits the standard OLS model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FitOLS(y, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FitOLS_+3A_y">y</code></td>
<td>
<p>Nx1 Numeric vector.</p>
</td></tr>
<tr><td><code id="FitOLS_+3A_x">X</code></td>
<td>
<p>NxP Numeric matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the following:
</p>
<table>
<tr><td><code>Beta</code></td>
<td>
<p>Regression coefficient.</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>Outcome variance.</p>
</td></tr>
<tr><td><code>Ibb</code></td>
<td>
<p>Information matrix for beta.</p>
</td></tr>
<tr><td><code>Resid</code></td>
<td>
<p>Outcome residuals.</p>
</td></tr>
</table>

<hr>
<h2 id='IINT'>Indirect-INT</h2><span id='topic+IINT'></span>

<h3>Description</h3>

<p>Two-stage association testing procedure. In the first stage, phenotype 
<code>y</code> and genotype <code>G</code> are each regressed on the model matrix
<code>X</code> to obtain residuals. The phenotypic residuals are transformed
using <code><a href="#topic+RankNorm">RankNorm</a></code>. In the next stage, the INT-transformed
residuals are regressed on the genotypic residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IINT(y, G, X = NULL, k = 0.375, ties.method = "average", simple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IINT_+3A_y">y</code></td>
<td>
<p>Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="IINT_+3A_g">G</code></td>
<td>
<p>Genotype matrix with observations as rows, SNPs as columns.</p>
</td></tr>
<tr><td><code id="IINT_+3A_x">X</code></td>
<td>
<p>Model matrix of covariates and structure adjustments. Should include
an intercept. Omit to perform marginal tests of association.</p>
</td></tr>
<tr><td><code id="IINT_+3A_k">k</code></td>
<td>
<p>Offset applied during rank-normalization. See
<code><a href="#topic+RankNorm">RankNorm</a></code>.</p>
</td></tr>
<tr><td><code id="IINT_+3A_ties.method">ties.method</code></td>
<td>
<p>Method of breaking ties, passed to <code>base::rank</code>.</p>
</td></tr>
<tr><td><code id="IINT_+3A_simple">simple</code></td>
<td>
<p>Return the p-values only?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>simple = TRUE</code>, returns a vector of p-values, one for each column
of <code>G</code>. If <code>simple = FALSE</code>, returns a numeric matrix, including the
Wald or Score statistic, its standard error, the Z-score, and the p-value.
</p>


<h3>See Also</h3>


<ul>
<li><p> Basic association test <code><a href="#topic+BAT">BAT</a></code>.
</p>
</li>
<li><p> Direct INT test <code><a href="#topic+DINT">DINT</a></code>.
</p>
</li>
<li><p> Omnibus INT test <code><a href="#topic+OINT">OINT</a></code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
# Design matrix
X &lt;- cbind(1, stats::rnorm(1e3))
# Genotypes
G &lt;- replicate(1e3, stats::rbinom(n = 1e3, size = 2, prob = 0.25))
storage.mode(G) &lt;- "numeric"
# Phenotype
y &lt;- exp(as.numeric(X %*% c(1,1)) + stats::rnorm(1e3))
# Association test
p &lt;- IINT(y = y, G = G, X = X)
</code></pre>

<hr>
<h2 id='OINT'>Omnibus-INT</h2><span id='topic+OINT'></span>

<h3>Description</h3>

<p>Association test that synthesizes the <code><a href="#topic+DINT">DINT</a></code> and
<code><a href="#topic+IINT">IINT</a></code> tests. The first approach is most powerful for traits that
could have arisen from a rank-preserving transformation of a latent normal
trait. The second approach is most powerful for traits that are linear in
covariates, yet have skewed or kurtotic residual distributions. During the
omnibus test, the direct and indirect tests are separately applied, then the
p-values are combined via the Cauchy combination method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OINT(
  y,
  G,
  X = NULL,
  k = 0.375,
  ties.method = "average",
  weights = c(1, 1),
  simple = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OINT_+3A_y">y</code></td>
<td>
<p>Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="OINT_+3A_g">G</code></td>
<td>
<p>Genotype matrix with observations as rows, SNPs as columns.</p>
</td></tr>
<tr><td><code id="OINT_+3A_x">X</code></td>
<td>
<p>Model matrix of covariates and structure adjustments. Should include
an intercept. Omit to perform marginal tests of association.</p>
</td></tr>
<tr><td><code id="OINT_+3A_k">k</code></td>
<td>
<p>Offset applied during rank-normalization. See
<code><a href="#topic+RankNorm">RankNorm</a></code>.</p>
</td></tr>
<tr><td><code id="OINT_+3A_ties.method">ties.method</code></td>
<td>
<p>Method of breaking ties, passed to <code>base::rank</code>.</p>
</td></tr>
<tr><td><code id="OINT_+3A_weights">weights</code></td>
<td>
<p>Respective weights to allocate the DINT and IINT tests.</p>
</td></tr>
<tr><td><code id="OINT_+3A_simple">simple</code></td>
<td>
<p>Return the OINT p-values only?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix of p-values, three for each column of <code>G</code>.
</p>


<h3>See Also</h3>


<ul>
<li><p> Basic association test <code><a href="#topic+BAT">BAT</a></code>.
</p>
</li>
<li><p> Direct INT test <code><a href="#topic+DINT">DINT</a></code>.
</p>
</li>
<li><p> Indirect INT test <code><a href="#topic+IINT">IINT</a></code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
# Design matrix
X &lt;- cbind(1, rnorm(1e3))
# Genotypes
G &lt;- replicate(1e3, rbinom(n = 1e3, size = 2, prob = 0.25))
storage.mode(G) &lt;- "numeric"
# Phenotype
y &lt;- exp(as.numeric(X %*% c(1, 1)) + rnorm(1e3))
# Omnibus
p &lt;- OINT(y = y, G = G, X = X, simple = TRUE)
</code></pre>

<hr>
<h2 id='OmniP'>Omnibus P-value.</h2><span id='topic+OmniP'></span>

<h3>Description</h3>

<p>Obtains an omnibus p-value from a vector of potentially dependent p-values 
using the method of Cauchy combination. The p-values are converted to Cauchy
random deviates then averaged. The distribution of the average of these 
deviates is well-approximated by a Cauchy distribution in the tails. See
&lt;https://doi.org/10.1080/01621459.2018.1554485&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OmniP(p, w = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OmniP_+3A_p">p</code></td>
<td>
<p>Numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="OmniP_+3A_w">w</code></td>
<td>
<p>Numeric weight vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>OINT p-value.
</p>

<hr>
<h2 id='PartitionData'>Partition Data</h2><span id='topic+PartitionData'></span>

<h3>Description</h3>

<p>Partition y and X according to the missingness pattern of g.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PartitionData(e, g, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PartitionData_+3A_e">e</code></td>
<td>
<p>Numeric residual vector.</p>
</td></tr>
<tr><td><code id="PartitionData_+3A_g">g</code></td>
<td>
<p>Genotype vector.</p>
</td></tr>
<tr><td><code id="PartitionData_+3A_x">X</code></td>
<td>
<p>Model matrix of covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing:
</p>

<ul>
<li><p> &quot;g_obs&quot;, observed genotype vector.
</p>
</li>
<li><p> &quot;X_obs&quot;, covariates for subjects with observed genotypes.
</p>
</li>
<li><p> &quot;X_mis&quot;, covariates for subjects with missing genotypes. 
</p>
</li>
<li><p> &quot;e_obs&quot;, residuals for subjects with observed genotypes.
</p>
</li></ul>


<hr>
<h2 id='PtoCauchy'>Convert P-value to Cauchy Random</h2><span id='topic+PtoCauchy'></span>

<h3>Description</h3>

<p>Convert P-value to Cauchy Random
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PtoCauchy(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PtoCauchy_+3A_p">p</code></td>
<td>
<p>Numeric p-value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric Cauchy random variable.
</p>

<hr>
<h2 id='RankNorm'>Rank-Normalize</h2><span id='topic+RankNorm'></span>

<h3>Description</h3>

<p>Applies the rank-based inverse normal transform (INT) to a numeric vector.
The INT can be broken down into a two-step procedure. In the first, the
observations are transformed onto the probability scale using the empirical
cumulative distribution function (ECDF). In the second, the observations are
transformed onto the real line, as Z-scores, using the probit function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RankNorm(u, k = 0.375, ties.method = "average")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RankNorm_+3A_u">u</code></td>
<td>
<p>Numeric vector.</p>
</td></tr>
<tr><td><code id="RankNorm_+3A_k">k</code></td>
<td>
<p>Offset. Defaults to (3/8), corresponding to the Blom transform.</p>
</td></tr>
<tr><td><code id="RankNorm_+3A_ties.method">ties.method</code></td>
<td>
<p>Method of breaking ties, passed to <code>base::rank</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of rank normalized values.
</p>


<h3>See Also</h3>


<ul>
<li><p> Direct INT test <code><a href="#topic+DINT">DINT</a></code>.
</p>
</li>
<li><p> Indirect INT test <code><a href="#topic+IINT">IINT</a></code>.
</p>
</li>
<li><p> Omnibus INT test <code><a href="#topic+OINT">OINT</a></code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Draw from chi-1 distribution
y &lt;- stats::rchisq(n = 1e3, df = 1)
# Rank normalize
z &lt;- RankNorm(y)
# Plot density of transformed measurement
plot(stats::density(z))
</code></pre>

<hr>
<h2 id='RNOmni-package'>RNOmni: Rank Normal Transformation Omnibus Test</h2><span id='topic+RNOmni'></span><span id='topic+RNOmni-package'></span>

<h3>Description</h3>

<p>Inverse normal transformation (INT) based genetic association testing. These tests are recommend for continuous traits with non-normally distributed residuals. INT-based tests robustly control the type I error in settings where standard linear regression does not, as when the residual distribution exhibits excess skew or kurtosis. Moreover, INT-based tests dominate standard linear regression in terms of power. These tests may be classified into two types. In direct INT (D-INT), the phenotype is itself transformed. In indirect INT (I-INT), phenotypic residuals are transformed. The omnibus test (O-INT) adaptively combines D-INT and I-INT into a single robust and statistically powerful approach. See McCaw ZR, Lane JM, Saxena R, Redline S, Lin X. &quot;Operating characteristics of the rank-based inverse normal transformation for quantitative trait analysis in genome-wide association studies&quot; &lt;doi:10.1111/biom.13214&gt;.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Zachary McCaw <a href="mailto:zmccaw@alumni.harvard.edu">zmccaw@alumni.harvard.edu</a> (<a href="https://orcid.org/0000-0002-2006-9828">ORCID</a>)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
