<!DOCTYPE html><html><head><title>Help for package roben</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {roben}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data'><p>simulated data for demonstrating the features of roben</p></a></li>
<li><a href='#GxESelection'><p>Variable selection for a roben object</p></a></li>
<li><a href='#predict.roben'><p>make predictions from a roben object</p></a></li>
<li><a href='#print.GxESelection'><p>print a GxESelection object</p></a></li>
<li><a href='#print.roben'><p>print a roben object</p></a></li>
<li><a href='#print.roben.pred'><p>print a roben.pred object</p></a></li>
<li><a href='#roben'><p>fit a robust Bayesian variable selection</p></a></li>
<li><a href='#roben-package'><p>roben: Robust Bayesian Variable Selection for Gene-Environment Interactions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Bayesian Variable Selection for Gene-Environment
Interactions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Jie Ren, Fei Zhou, Xiaoxi Li, Cen Wu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jie Ren &lt;jieren@ksu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene-environment (G×E) interactions have important implications to elucidate the 
    etiology of complex diseases beyond the main genetic and environmental effects. 
    Outliers and data contamination in disease phenotypes of G×E studies have been commonly 
    encountered, leading to the development of a broad spectrum of robust penalization methods. 
    Nevertheless, within the Bayesian framework, the issue has not been taken care of in existing 
    studies. We develop a robust Bayesian variable selection method for G×E interaction 
    studies. The proposed Bayesian method can effectively accommodate heavy-tailed errors and 
    outliers in the response variable while conducting variable selection by accounting for 
    structural sparsity. In particular, the spike-and-slab priors have been imposed on both 
    individual and group levels to identify important main and interaction effects. An efficient 
    Gibbs sampler has been developed to facilitate fast computation. The Markov chain Monte Carlo 
    algorithms of the proposed and alternative methods are efficiently implemented in C++.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, glmnet, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), covr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jrhub/roben">https://github.com/jrhub/roben</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jrhub/roben/issues">https://github.com/jrhub/roben/issues</a></td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-12 17:49:25 UTC; 17854</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-12 21:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='data'>simulated data for demonstrating the features of roben</h2><span id='topic+data'></span><span id='topic+GxE_small'></span><span id='topic+GxE_large'></span><span id='topic+X'></span><span id='topic+Y'></span><span id='topic+E'></span><span id='topic+clin'></span><span id='topic+coeff'></span><span id='topic+X2'></span><span id='topic+Y2'></span><span id='topic+E2'></span><span id='topic+clin2'></span><span id='topic+coeff2'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of roben.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("GxE_small")
data("GxE_large")
</code></pre>


<h3>Format</h3>

<p>GxE_small consists of five components: X, Y, E, clin and coeff. coeff contains the true values of parameters used for generating Y.
</p>
<p>GxE_large contains larger datasets: X2, Y2, E2 and clin2
</p>


<h3>Details</h3>

<p><strong>The data model for generating Y</strong>
</p>
<p>Use subscript <code class="reqn">i</code> to denote the <code class="reqn">i</code>th subject. Let <code class="reqn">(X_{i}, Y_{i}, E_{i}, Clin_{i})</code>, (<code class="reqn">i=1,\ldots,n</code>) be
independent and identically distributed random vectors. <code class="reqn">Y_{i}</code> is a continuous response variable representing the
disease phenotype. <code class="reqn">X_{i}</code> is the <code class="reqn">p</code>&ndash;dimensional vector of G factors. The environmental factors and clinical covariates
are denoted as the <code class="reqn">k</code>-dimensional vector <code class="reqn">E_{i}</code> and the <code class="reqn">q</code>-dimensional vector <code class="reqn">Clin_{i}</code>, respectively.
The <code class="reqn">\epsilon</code> follows some heavy-tailed distribution.
Considering the following model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{t=1}^{q}\alpha_{t}Clin_{it} + \sum_{m=1}^{k}\theta_{m}E_{im}  + \sum_{j=1}^{p}\gamma_{j}X_{ij} + \sum_{j=1}^{p}\sum_{m=1}^{k}\zeta_{jm}E_{im}X_{ij} +\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept; <code class="reqn">\alpha_{t}</code>'s, <code class="reqn">\theta_{m}</code>'s, <code class="reqn">\gamma_{j}</code>'s and <code class="reqn">\zeta_{jm}</code>'s are
the regression coefficients for the clinical covariates, environmental factors, genetic factors and G<code class="reqn">\times</code>E interactions, respectively.
</p>
<p>Define <code class="reqn">\beta_{j}=(\gamma_{j}, \zeta_{j1},\ldots,\zeta_{jk})^\top \equiv (\beta_{j1},\ldots,\beta_{jL})^\top</code> and
<code class="reqn">U_{ij}=(X_{ij},X_{ij}E_{i1}\ldots,X_{ij}E_{ik})^\top \equiv (U_{ij1},\dots,U_{ijL})^\top</code>, where <code class="reqn">L=k+1</code>.
The model can be written as
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{t=1}^{q}\alpha_{t}Clin_{it} + \sum_{m=1}^{k}\theta_{m}E_{im}  + \sum_{j=1}^{p} \big(U_{ij}^\top\beta_{j}\big) +\epsilon_{i},</code>
</p>

<p>where the coefficient vector <code class="reqn">\beta_{j}</code> represents all the main and interaction effects corresponding to the <code class="reqn">j</code>th genetic measurement.
</p>
<p>The object <strong>coeff</strong> in GxE_small is a list of four components, corresponding to <code class="reqn">\alpha_{0}</code>, <code class="reqn">\alpha_{t}</code>'s, <code class="reqn">\theta_{m}</code>'s and <code class="reqn">\beta_{j}</code>'s.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+roben">roben</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GxE_small)
dim(X)
print(coeff)

data(GxE_large)
dim(X)
print(coeff)

</code></pre>

<hr>
<h2 id='GxESelection'>Variable selection for a roben object</h2><span id='topic+GxESelection'></span><span id='topic+GxESelection.Sparse'></span><span id='topic+GxESelection.NonSparse'></span>

<h3>Description</h3>

<p>Variable selection for a roben object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GxESelection(obj, ...)

## S3 method for class 'Sparse'
GxESelection(obj, burn.in = obj$burn.in, ...)

## S3 method for class 'NonSparse'
GxESelection(obj, burn.in = obj$burn.in, prob = 0.95, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GxESelection_+3A_obj">obj</code></td>
<td>
<p>roben object.</p>
</td></tr>
<tr><td><code id="GxESelection_+3A_...">...</code></td>
<td>
<p>other GxESelection arguments.</p>
</td></tr>
<tr><td><code id="GxESelection_+3A_burn.in">burn.in</code></td>
<td>
<p>MCMC burn-in.</p>
</td></tr>
<tr><td><code id="GxESelection_+3A_prob">prob</code></td>
<td>
<p>probability for credible interval, between 0 and 1. e.g. prob=0.95 leads to 95% credible interval.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For class &lsquo;Sparse&rsquo;, the median probability model (MPM) (Barbieri and Berger, 2004) is used to identify predictors that are significantly associated
with the response variable. For class &lsquo;NonSparse&rsquo;, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;GxESelection&rsquo; is returned, which is a list with components:
</p>

<ul>
<li><p> method: method used for identifying important effects.
</p>
</li>
<li><p> effects: a list of names of selected effects.
</p>
</li>
<li><p> summary: a summary of selected effects.
</p>
</li>
<li><p> indicator: a matrix of indicators of selected effects.
</p>
</li></ul>



<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023) Robust Bayesian variable selection for
gene-environment interactions. <em>Biometrics</em>, 79(2):684-694. <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>. PMID: 35394058.
</p>
<p>Barbieri, M.M. and Berger, J.O. (2004). Optimal predictive model selection. <em>Ann. Statist</em>, 32(3):870–897
</p>


<h3>See Also</h3>

<p><code><a href="#topic+roben">roben</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GxE_small)
iter=5000
## sparse
fit=roben(X, Y, E, clin, iterations=iter)
selected=GxESelection(fit)
selected


## non-sparse
fit=roben(X, Y, E, clin, iterations=iter, sparse=FALSE)
selected=GxESelection(fit)
selected


</code></pre>

<hr>
<h2 id='predict.roben'>make predictions from a roben object</h2><span id='topic+predict.roben'></span>

<h3>Description</h3>

<p>make predictions from a roben object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'roben'
predict(object, X.new, E.new, clin.new = NULL, Y.new = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.roben_+3A_object">object</code></td>
<td>
<p>roben object.</p>
</td></tr>
<tr><td><code id="predict.roben_+3A_x.new">X.new</code></td>
<td>
<p>a matrix of new values for X at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.roben_+3A_e.new">E.new</code></td>
<td>
<p>a vector of new values for E at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.roben_+3A_clin.new">clin.new</code></td>
<td>
<p>a vector or matrix of new values for clin at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.roben_+3A_y.new">Y.new</code></td>
<td>
<p>a vector of the response of new observations. If provided, the prediction error will be computed based on Y.new.</p>
</td></tr>
<tr><td><code id="predict.roben_+3A_...">...</code></td>
<td>
<p>other predict arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>X.new (E.new) must have the same number of columns as X (E) used for fitting the model. If clin was provided when fit the model, clin.new
must not be NULL, and vice versa. The predictions are made based on the posterior estimates of coefficients in the roben object.
Note that the main effects of environmental exposures E are not subject to selection.
</p>
<p>If Y.new is provided, the prediction error will be computed. For robust methods, the prediction mean absolute deviations (PMAD) will be computed.
For non-robust methods, the prediction mean squared error (PMSE) will be computed.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;roben.pred&rsquo; is returned, which is a list with components:
</p>
<table>
<tr><td><code>error</code></td>
<td>
<p>prediction error. error is NULL is Y.new=NULL.</p>
</td></tr>
<tr><td><code>y.pred</code></td>
<td>
<p>predicted values of the new observations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+roben">roben</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GxE_small)
test=sample((1:nrow(X)), floor(nrow(X)/5))
fit=roben(X[-test,], Y[-test,], E[-test,], clin[-test,], iterations=5000)
predict(fit, X[test,], E[test,], clin[test,], Y[test,])

</code></pre>

<hr>
<h2 id='print.GxESelection'>print a GxESelection object</h2><span id='topic+print.GxESelection'></span>

<h3>Description</h3>

<p>Print a summary of a GxESelection object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GxESelection'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.GxESelection_+3A_x">x</code></td>
<td>
<p>GxESelection object.</p>
</td></tr>
<tr><td><code id="print.GxESelection_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.GxESelection_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+GxESelection">GxESelection</a></code>
</p>

<hr>
<h2 id='print.roben'>print a roben object</h2><span id='topic+print.roben'></span>

<h3>Description</h3>

<p>Print a summary of a roben object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'roben'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.roben_+3A_x">x</code></td>
<td>
<p>roben object.</p>
</td></tr>
<tr><td><code id="print.roben_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.roben_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+roben">roben</a></code>
</p>

<hr>
<h2 id='print.roben.pred'>print a roben.pred object</h2><span id='topic+print.roben.pred'></span>

<h3>Description</h3>

<p>Print a summary of a roben.pred object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'roben.pred'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.roben.pred_+3A_x">x</code></td>
<td>
<p>roben.pred object.</p>
</td></tr>
<tr><td><code id="print.roben.pred_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.roben.pred_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+predict.roben">predict.roben</a></code>
</p>

<hr>
<h2 id='roben'>fit a robust Bayesian variable selection</h2><span id='topic+roben'></span>

<h3>Description</h3>

<p>fit a robust Bayesian variable selection model for G×E interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roben(
  X,
  Y,
  E,
  clin = NULL,
  iterations = 10000,
  burn.in = NULL,
  robust = TRUE,
  sparse = TRUE,
  structure = c("sparsegroup", "group", "individual"),
  hyper = NULL,
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roben_+3A_x">X</code></td>
<td>
<p>the matrix of predictors (genetic factors) without intercept. Each row should be an observation vector. X will be centered and a column of 1 will be added to the X matrix
as the intercept.</p>
</td></tr>
<tr><td><code id="roben_+3A_y">Y</code></td>
<td>
<p>the response variable. The current version of roben only supports continuous response.</p>
</td></tr>
<tr><td><code id="roben_+3A_e">E</code></td>
<td>
<p>a matrix of environmental factors. E will be centered. The interaction terms between X (G factors) and E will be automatically created and included in the model.</p>
</td></tr>
<tr><td><code id="roben_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical variables. Clinical variables are not subject to penalty.</p>
</td></tr>
<tr><td><code id="roben_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="roben_+3A_burn.in">burn.in</code></td>
<td>
<p>the number of iterations for burn-in.</p>
</td></tr>
<tr><td><code id="roben_+3A_robust">robust</code></td>
<td>
<p>logical flag. If TRUE, robust methods will be used.</p>
</td></tr>
<tr><td><code id="roben_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
<tr><td><code id="roben_+3A_structure">structure</code></td>
<td>
<p>three choices are available. &quot;sparsegroup&quot; for sparse-group selection, which is a bi-level selection on both group-level and individual-level. &quot;group&quot; for selection on group-level only. &quot;individual&quot; for selection on individual-level only.</p>
</td></tr>
<tr><td><code id="roben_+3A_hyper">hyper</code></td>
<td>
<p>a named list of hyperparameters.</p>
</td></tr>
<tr><td><code id="roben_+3A_debugging">debugging</code></td>
<td>
<p>logical flag. If TRUE, progress will be output to the console and extra information will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Consider the data model described in &quot;<code><a href="#topic+data">data</a></code>&quot;:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{t=1}^{q}\alpha_{t}Clin_{it} + \sum_{m=1}^{k}\theta_{m}E_{im}  + \sum_{j=1}^{p} \big(U_{ij}^\top\beta_{j}\big) +\epsilon_{i},</code>
</p>

<p>where the main and interaction effects of the <code class="reqn">j</code>th genetic variant is corresponding to the coefficient vector <code class="reqn">\beta_{j}=(\beta_{j1}, \beta_{j2},\ldots,\beta_{jL})^\top</code>.
</p>
<p>When structure=&quot;sparsegroup&quot; (default setting), selection will be conducted on both individual and group levels (bi-level selection):
</p>

<ul>
<li> <p><strong>Group-level selection:</strong> by determining whether <code class="reqn">||\beta_{j}||_{2}=0</code>, we can know if the <code class="reqn">j</code>th genetic variant has any effect at all.
</p>
</li>
<li> <p><strong>Individual-level selection:</strong> investigate whether the <code class="reqn">j</code>th genetic variant has main effect, G<code class="reqn">\times</code>E interaction or both, by determining which components in <code class="reqn">\beta_{j}</code> has non-zero values.
</p>
</li></ul>

<p>If structure=&quot;group&quot;, only group-level selection will be conducted on <code class="reqn">||\beta_{j}||_{2}</code>. If structure=&quot;individual&quot;, only individual-level selection will be conducted on each <code class="reqn">\beta_{jl}</code>, (<code class="reqn">l=1,\ldots,L</code>).
</p>
<p>When sparse=TRUE (default), spike&ndash;and&ndash;slab priors are imposed on individual and/or group levels to identify important main and interaction effects. Otherwise, Laplacian shrinkage will be used.
</p>
<p>When robust=TRUE (default), the distribution of <code class="reqn">\epsilon_{i}</code> is defined as a Laplace distribution with density
<code class="reqn">
f(\epsilon_{i}|\nu) = \frac{\nu}{2}\exp\left\{-\nu |\epsilon_{i}|\right\}
</code>, (<code class="reqn">i=1,\dots,n</code>), which leads to a Bayesian formulation of LAD regression. If robust=FALSE, <code class="reqn">\epsilon_{i}</code> follows a normal distribution.
<br />
</p>
<p>Both <code class="reqn">X</code> and <code class="reqn">E</code> will be centered before the generation of interaction terms, in order to prevent the multicollinearity between main effects and interaction terms.
<br />
</p>
<p>Users can modify the hyper-parameters by providing a named list of hyper-parameters via the argument &lsquo;hyper&rsquo;.
The list can have the following named components
</p>

<ul>
<li><p> a0, b0: shape parameters of the Beta priors (<code class="reqn">\pi^{a_{0}-1}(1-\pi)^{b_{0}-1}</code>) on <code class="reqn">\pi_{0}</code>.
</p>
</li>
<li><p> a1, b1: shape parameters of the Beta priors (<code class="reqn">\pi^{a_{1}-1}(1-\pi)^{b_{1}-1}</code>) on <code class="reqn">\pi_{1}</code>.
</p>
</li>
<li><p> c1, c2: the shape parameter and the rate parameter of the Gamma prior on <code class="reqn">\nu</code>.
</p>
</li>
<li><p> d1, d2: the shape parameter and the rate parameter of the Gamma priors on <code class="reqn">\eta</code>.
</p>
</li></ul>

<p>Please check the references for more details about the prior distributions.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GxESelection">GxESelection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GxE_small)

## default method
iter=5000
fit=roben(X, Y, E, clin, iterations = iter)
fit$coefficient

## Ture values of parameters of main G effects and interactions
coeff$GE

## Compute TP and FP
sel=GxESelection(fit)
pos=which(sel$indicator != 0)
tp=length(intersect(which(coeff$GE != 0), pos))
fp=length(pos)-tp
list(tp=tp, fp=fp)


## alternative: robust group selection
fit=roben(X, Y, E, clin, iterations=iter, structure="g")
fit$coefficient

## alternative: non-robust sparse group selection
fit=roben(X, Y, E, clin, iterations=iter, robust=FALSE)
fit$coefficient


</code></pre>

<hr>
<h2 id='roben-package'>roben: Robust Bayesian Variable Selection for Gene-Environment Interactions</h2><span id='topic+roben-package'></span>

<h3>Description</h3>

<p>Gene-environment (G×E) interactions have important implications to elucidate the etiology of complex diseases beyond the main genetic and environmental effects. Outliers and data contamination in disease phenotypes of G×E studies have been commonly encountered, leading to the development of a broad spectrum of robust penalization methods. Nevertheless, within the Bayesian framework, the issue has not been taken care of in existing studies. We develop a robust Bayesian variable selection method for G×E interaction studies. The proposed Bayesian method can effectively accommodate heavy-tailed errors and outliers in the response variable while conducting variable selection by accounting for structural sparsity. In particular, the spike-and-slab priors have been imposed on both individual and group levels to identify important main and interaction effects. An efficient Gibbs sampler has been developed to facilitate fast computation. The Markov chain Monte Carlo algorithms of the proposed and alternative methods are efficiently implemented in C++.
</p>
<p>In this package, we provide a set of robust Bayesian variable selection methods tailored for interaction analysis. A Bayesian formulation of the least absolute deviation (LAD) regression has been adopted to accommodate data contamination and long-tailed distributions in the response/ phenotype. The default method (the proposed method) conducts variable selection by accounting for structural sparsity. In particular, the spike&ndash;and&ndash;slab priors are imposed on both individual and group levels to identify important main and interaction effects (bi-level/ sparse-group selection).
</p>
<p>In addition to the default method, users can also choose different selection structures (group-level-only or individual-level-only), methods without spike&ndash;and&ndash;slab priors and non-robust methods. In total, <em>roben</em> provides 12 different methods (6 robust and 6 non-robust). Among them, robust methods with spike&ndash;and&ndash;slab priors and the robust method for bi-level selection have been developed for the first time. Please read the Details below for how to configure the method used.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface <strong>roben()</strong> allows users to flexibly choose the fitting methods they prefer. There are three arguments in roben() that control the fitting method:
</p>

<table>
<tr>
 <td style="text-align: right;">
robust: </td><td style="text-align: left;"> whether to use robust methods. </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
sparse: </td><td style="text-align: left;"> whether to use the spike-and-slab priors to create sparsity. </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
structure: </td><td style="text-align: left;"> structural identification. Three choices are available: </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> "sparsegroup", "group" and “individual”.
</td>
</tr>

</table>

<p>The function roben() returns a roben object that contains the posterior estimates of each coefficients.
S3 generic functions GxESelection(), predict() and print() are implemented for roben objects.
GxESelection() takes a roben object and returns the variable selection results.
predict() takes a roben object and returns the predicted values for new observations.
</p>


<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y., and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions.
<em>Biometrics</em>, 79(2):684-694. <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>. PMID: 35394058.
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883. <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>.
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S., and Wu, C. (2021). Gene-Environment Interaction: A Variable Selection Perspective.
<em>Methods in Molecular Biology</em>, 2212:191-223. <a href="https://doi.org/10.1007/978-1-0716-0947-7_13">doi:10.1007/978-1-0716-0947-7_13</a>. PMID: 33733358.
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y., and Wu, C. (2020). Semi-parametric Bayesian variable selection for
gene-environment interactions. <em>Statistics in Medicine</em>, 39: 617–638. <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>.
</p>
<p>Ren, J., Zhou, F., Li, X., Wu, C., and Jiang, Y. (2019). spinBayes: Semi-Parametric Gene-Environment Interaction via
Bayesian Variable Selection. R package version 0.1.0. <a href="https://CRAN.R-project.org/package=spinBayes">https://CRAN.R-project.org/package=spinBayes</a>.
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y., and Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach
accounting for hierarchical structures. <em>Statistics in Medicine</em>, 37:437–456. <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>.
</p>
<p>Wu, C., Shi, X., Cui, Y., and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34(30): 4016–4030. <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>.
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear
varying coefficient model. <em>Statistics in Medicine</em>, 33(28): 4988–4998. <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>.
</p>
<p>Wu, C., Zhong, P.S., and Cui, Y. (2018). Additive varying–coefficient model for nonlinear gene–environment interactions.
<em>Statistical Applications in Genetics and Molecular Biology</em>, 17(2). <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>.
</p>
<p>Wu, C., Zhong, P.S., and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University.</em>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/jrhub/roben">https://github.com/jrhub/roben</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/jrhub/roben/issues">https://github.com/jrhub/roben/issues</a>
</p>
</li></ul>

<p><code><a href="#topic+roben">roben</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
