<!DOCTYPE html><html><head><title>Help for package RobMixReg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RobMixReg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biscalew'><p>biscalew :Robust M-estimates for scale.</p></a></li>
<li><a href='#bisquare'><p>bisquare : Robust estimates for mean.</p></a></li>
<li><a href='#blockMap'><p>Plot the coefficient matrix.</p></a></li>
<li><a href='#CCLE_data'><p>RobMixReg package built-in CCLE data.</p></a></li>
<li><a href='#colon_data'><p>RobMixReg package built-in Colon cancer data.</p></a></li>
<li><a href='#compPlot'><p>The plot wrapper function.</p></a></li>
<li><a href='#Compute_Rbase_SVD'><p>Compute the row space using SVD.</p></a></li>
<li><a href='#CSMR'><p>The main function of the RBSL algorithm.</p></a></li>
<li><a href='#CSMR_one'><p>Perform the RBSL algorithm one times.</p></a></li>
<li><a href='#CSMR_predict'><p>The predict function of the CSMR algorithm.</p></a></li>
<li><a href='#CSMR_train'><p>The train function of the CSMR algorithm.</p></a></li>
<li><a href='#CTLERob'><p>CTLERob: Robust mixture regression based on component-wise adaptive trimming likelihood estimation.</p></a></li>
<li><a href='#denLp'><p>denLp : Density function for Laplace distribution.</p></a></li>
<li><a href='#DeOut'><p>DeOut : Detect outlier observations.</p></a></li>
<li><a href='#flexmix_2'><p>flexmix_2: Multiple runs of MLE based mixture regression to stabilize the output.</p></a></li>
<li><a href='#gaussData'><p>RobMixReg package built-in gaussian example data.</p></a></li>
<li><a href='#lars.lsa'><p>lars variant for LSA.</p></a></li>
<li><a href='#logLik_mixtureReg'><p>Obtain Log-likelihood from a mixtureReg Object</p></a></li>
<li><a href='#lsa'><p>Least square approximation. This version Oct 19, 2006.</p></a></li>
<li><a href='#mixlinrb_bi'><p>mixlinrb_bi: mixlinrb_bione estimates the mixture regression parameters robustly using bisquare function based on multiply initial value.</p></a></li>
<li><a href='#mixlinrb_bione'><p>mixlinrb_bione : mixlinrb_bione estimates the mixture regression parameters robustly using bisquare function based on one initial value.</p></a></li>
<li><a href='#mixLp'><p>mixLp : mixLp_one estimates the mixture regression parameters robustly using Laplace distribution based on multiply initial value..</p></a></li>
<li><a href='#mixLp_one'><p>mixLp_one : mixLp_one estimates the mixture regression parameters robustly using Laplace distribution based on one initial value.</p></a></li>
<li><a href='#mixtureReg'><p>Function to Fit Mixture of Regressions</p></a></li>
<li><a href='#MLM'><p>The main function of mining the latent relationship among variables.</p></a></li>
<li><a href='#MLM_bic'><p>Model selection function for low dimension data.</p></a></li>
<li><a href='#MLM_cv'><p>Cross validation (fold-5) function for high dimension data.</p></a></li>
<li><a href='#orderedLines'><p>Sort by X Coordinates and Add Line to a Plot</p></a></li>
<li><a href='#plot_CTLE'><p>plot_CTLE: Plot the mixture/single regression line(s) in a simply function.</p></a></li>
<li><a href='#plot_mixtureReg'><p>Plot Fit and Mixing Probability of a mixtureReg Object</p></a></li>
<li><a href='#plot_mixtureRegList'><p>Plot a List of mixtureReg Objects</p></a></li>
<li><a href='#Rec_Lm'><p>Adaptive lasso.</p></a></li>
<li><a href='#rmr'><p>The main function of Robust Mixture Regression using five methods.</p></a></li>
<li><a href='#RobMixReg-class'><p>Class RobMixReg.</p></a></li>
<li><a href='#simu_data_sparse'><p>Simulate high dimension data for RBSL algorithm validation.</p></a></li>
<li><a href='#simu_func'><p>The simulation function for low/high dimensional space.</p></a></li>
<li><a href='#simu_low'><p>The simulation function for low dimensional space.</p></a></li>
<li><a href='#simuData'><p>RobMixReg package built-in simulated example data.</p></a></li>
<li><a href='#TLE'><p>TLE: robust mixture regression based on trimmed likelihood estimation.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Mixture Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-08-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Sha Cao [aut, cph, ths],
  Wennan Chang [aut, cre],
  Chi Zhang [aut, ctb, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wennan Chang &lt;wnchang@iu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Finite mixture models are a popular technique for modelling unobserved heterogeneity or to approximate general distribution functions in a semi-parametric way. They are used in a lot of different areas such as astronomy, biology, economics, marketing or medicine.
             This package is the implementation of popular robust mixture regression methods based on different algorithms including: fleximix, finite mixture models and latent class regression; CTLERob, component-wise adaptive trimming likelihood estimation; mixbi, bi-square estimation; mixL, Laplacian distribution; mixt, t-distribution; TLE, trimmed likelihood estimation.
             The implemented algorithms includes:  CTLERob stands for Component-wise adaptive Trimming Likelihood Estimation based mixture regression; mixbi stands for mixture regression based on bi-square estimation; mixLstands for mixture regression based on Laplacian distribution; TLE stands for Trimmed Likelihood Estimation based mixture regression. For more detail of the algorithms, please refer to below references. 
             Reference: Chun Yu, Weixin Yao, Kun Chen (2017) &lt;<a href="https://doi.org/10.1002%2Fcjs.11310">doi:10.1002/cjs.11310</a>&gt;.
             NeyKov N, Filzmoser P, Dimova R et al. (2007) &lt;<a href="https://doi.org/10.1016%2Fj.csda.2006.12.024">doi:10.1016/j.csda.2006.12.024</a>&gt;.
             Bai X, Yao W. Boyer JE (2012) &lt;<a href="https://doi.org/10.1016%2Fj.csda.2012.01.016">doi:10.1016/j.csda.2012.01.016</a>&gt;.
             Wennan Chang, Xinyu Zhou, Yong Zang, Chi Zhang, Sha Cao (2020) &lt;<a href="https://arxiv.org/abs/2005.11599">arXiv:2005.11599</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>flexmix, robustbase, gtools, MASS,
methods,robust,lars,dplyr,rlang,
scales,gplots,grDevices,graphics,RColorBrewer,stats,glmnet</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://changwn.github.io/RobMixReg/">https://changwn.github.io/RobMixReg/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/changwn/RobMixReg/issues">https://github.com/changwn/RobMixReg/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-04 02:28:29 UTC; wnchang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-05 12:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='biscalew'>biscalew :Robust M-estimates for scale.</h2><span id='topic+biscalew'></span>

<h3>Description</h3>

<p>Tukey's bisquare family of functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biscalew(t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="biscalew_+3A_t">t</code></td>
<td>
<p>Numerical input, usually residuals.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>bisquare weight for scale.
</p>

<hr>
<h2 id='bisquare'>bisquare : Robust estimates for mean.</h2><span id='topic+bisquare'></span>

<h3>Description</h3>

<p>Tukey's bisquare family of functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bisquare(t, k = 4.685)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bisquare_+3A_t">t</code></td>
<td>
<p>Numerical input, usually residuals.</p>
</td></tr>
<tr><td><code id="bisquare_+3A_k">k</code></td>
<td>
<p>A constant tuning parameter, default is 4.685.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A bi-square weight for mean.
</p>

<hr>
<h2 id='blockMap'>Plot the coefficient matrix.</h2><span id='topic+blockMap'></span>

<h3>Description</h3>

<p>Plot the coefficient matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blockMap(rrr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blockMap_+3A_rrr">rrr</code></td>
<td>
<p>The result from CSMR function</p>
</td></tr>
</table>

<hr>
<h2 id='CCLE_data'>RobMixReg package built-in CCLE data.</h2><span id='topic+CCLE_data'></span>

<h3>Description</h3>

<p>The list which contain all the information to generate variables used in the real application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CCLE_data
</code></pre>


<h3>Format</h3>

<p>A list whose length is 2:
</p>

<dl>
<dt>X</dt><dd><p>Gene expression dataset.</p>
</dd>
<dt>Y</dt><dd><p>AUCC score.</p>
</dd>
</dl>

<hr>
<h2 id='colon_data'>RobMixReg package built-in Colon cancer data.</h2><span id='topic+colon_data'></span>

<h3>Description</h3>

<p>The list which contain all the information to generate variables used in the real application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colon_data
</code></pre>


<h3>Format</h3>

<p>A list whose length is 3:
</p>

<dl>
<dt>rnames</dt><dd><p>A string contains the name of binding protein and epigenetic regulator.</p>
</dd>
<dt>x3</dt><dd><p>The gene expression profile of CREB3L1.</p>
</dd>
<dt>y3</dt><dd><p>The methylation profile of cg16012690 on 299 colon adenocarcinoma patients.</p>
</dd>
<dt>x2</dt><dd><p>x2</p>
</dd>
<dt>y2</dt><dd><p>y2</p>
</dd>
<dt>x1</dt><dd><p>x1</p>
</dd>
<dt>y1</dt><dd><p>y1</p>
</dd>
</dl>

<hr>
<h2 id='compPlot'>The plot wrapper function.</h2><span id='topic+compPlot'></span>

<h3>Description</h3>

<p>The plot wrapper function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compPlot(type = "rlr", x, y, nc, inds_in, res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compPlot_+3A_type">type</code></td>
<td>
<p>The character to choose which type of plot to generate.</p>
</td></tr>
<tr><td><code id="compPlot_+3A_x">x</code></td>
<td>
<p>The independent variables</p>
</td></tr>
<tr><td><code id="compPlot_+3A_y">y</code></td>
<td>
<p>The external variable</p>
</td></tr>
<tr><td><code id="compPlot_+3A_nc">nc</code></td>
<td>
<p>The number of components</p>
</td></tr>
<tr><td><code id="compPlot_+3A_inds_in">inds_in</code></td>
<td>
<p>A vector indicate the outlier samples.</p>
</td></tr>
<tr><td><code id="compPlot_+3A_res">res</code></td>
<td>
<p>The result object returned by MLM function.</p>
</td></tr>
</table>

<hr>
<h2 id='Compute_Rbase_SVD'>Compute the row space using SVD.</h2><span id='topic+Compute_Rbase_SVD'></span>

<h3>Description</h3>

<p>Compute the row space using SVD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Rbase_SVD(bulk_data, tg_R1_lists_selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Rbase_SVD_+3A_bulk_data">bulk_data</code></td>
<td>
<p>The bulk data..</p>
</td></tr>
<tr><td><code id="Compute_Rbase_SVD_+3A_tg_r1_lists_selected">tg_R1_lists_selected</code></td>
<td>
<p>A list of the marker genes for several cell types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix which each row span the row space using cell type specific marker genes.
</p>

<hr>
<h2 id='CSMR'>The main function of the RBSL algorithm.</h2><span id='topic+CSMR'></span>

<h3>Description</h3>

<p>The main function of the RBSL algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSMR(x, y, nit, nc, max_iter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CSMR_+3A_x">x</code></td>
<td>
<p>The matrix</p>
</td></tr>
<tr><td><code id="CSMR_+3A_y">y</code></td>
<td>
<p>The external supervised variable.</p>
</td></tr>
<tr><td><code id="CSMR_+3A_nit">nit</code></td>
<td>
<p>xxx?</p>
</td></tr>
<tr><td><code id="CSMR_+3A_nc">nc</code></td>
<td>
<p>The component number in the mixture model.</p>
</td></tr>
<tr><td><code id="CSMR_+3A_max_iter">max_iter</code></td>
<td>
<p>The maximum iteration number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consist of coefficient, clustering membership, data x, external variable y, predicted y based on regression model.
</p>

<hr>
<h2 id='CSMR_one'>Perform the RBSL algorithm one times.</h2><span id='topic+CSMR_one'></span>

<h3>Description</h3>

<p>Perform the RBSL algorithm one times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSMR_one(x, y, nit = 1, nc, max_iter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CSMR_one_+3A_x">x</code></td>
<td>
<p>The matrix</p>
</td></tr>
<tr><td><code id="CSMR_one_+3A_y">y</code></td>
<td>
<p>The external supervised variable.</p>
</td></tr>
<tr><td><code id="CSMR_one_+3A_nit">nit</code></td>
<td>
<p>xxx?</p>
</td></tr>
<tr><td><code id="CSMR_one_+3A_nc">nc</code></td>
<td>
<p>The component number in the mixture model.</p>
</td></tr>
<tr><td><code id="CSMR_one_+3A_max_iter">max_iter</code></td>
<td>
<p>The maximum iteration number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consist of coefficient, clustering membership, data x, external variable y, predicted y based on regression model.
</p>

<hr>
<h2 id='CSMR_predict'>The predict function of the CSMR algorithm.</h2><span id='topic+CSMR_predict'></span>

<h3>Description</h3>

<p>The predict function of the CSMR algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSMR_predict(CSMR_coffs, CSMR.model, xnew, ynew, singleMode = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CSMR_predict_+3A_csmr_coffs">CSMR_coffs</code></td>
<td>
<p>The coefficient matrix.</p>
</td></tr>
<tr><td><code id="CSMR_predict_+3A_csmr.model">CSMR.model</code></td>
<td>
<p>The trained model.</p>
</td></tr>
<tr><td><code id="CSMR_predict_+3A_xnew">xnew</code></td>
<td>
<p>x variable.</p>
</td></tr>
<tr><td><code id="CSMR_predict_+3A_ynew">ynew</code></td>
<td>
<p>y variable.</p>
</td></tr>
<tr><td><code id="CSMR_predict_+3A_singlemode">singleMode</code></td>
<td>
<p>A parameter to set the component to one.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consist of coefficient, clustering membership, data x, external variable y, predicted y based on regression model.
</p>

<hr>
<h2 id='CSMR_train'>The train function of the CSMR algorithm.</h2><span id='topic+CSMR_train'></span>

<h3>Description</h3>

<p>The train function of the CSMR algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSMR_train(x, y, nit, nc, max_iter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CSMR_train_+3A_x">x</code></td>
<td>
<p>The matrix</p>
</td></tr>
<tr><td><code id="CSMR_train_+3A_y">y</code></td>
<td>
<p>The external supervised variable.</p>
</td></tr>
<tr><td><code id="CSMR_train_+3A_nit">nit</code></td>
<td>
<p>xxx</p>
</td></tr>
<tr><td><code id="CSMR_train_+3A_nc">nc</code></td>
<td>
<p>The component number in the mixture model.</p>
</td></tr>
<tr><td><code id="CSMR_train_+3A_max_iter">max_iter</code></td>
<td>
<p>The maximum iteration number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consist of coefficient, clustering membership, data x, external variable y, predicted y based on regression model.
</p>

<hr>
<h2 id='CTLERob'>CTLERob: Robust mixture regression based on component-wise adaptive trimming likelihood estimation.</h2><span id='topic+CTLERob'></span><span id='topic+CTLERob+2Cformula+2CANY+2CANY+2Cnumeric-method'></span><span id='topic+CTLERob+2Cformula+2CANY+2CANY+2Cnumeric+2CANY-method'></span>

<h3>Description</h3>

<p>CTLERob performes robust linear regression with high breakdown point and high efficiency in each mixing components and adaptively remove the outlier samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLERob(formula, data, nit = 20, nc = 2, rlr_method = "ltsReg")

## S4 method for signature 'formula,ANY,ANY,numeric'
CTLERob(formula, data, nit = 20,
  nc = 2, rlr_method = "ltsReg")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLERob_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="CTLERob_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="CTLERob_+3A_nit">nit</code></td>
<td>
<p>Number of iterations.</p>
</td></tr>
<tr><td><code id="CTLERob_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="CTLERob_+3A_rlr_method">rlr_method</code></td>
<td>
<p>The regression methods, default is 'ltsReg'.</p>
</td></tr>
</table>

<hr>
<h2 id='denLp'>denLp : Density function for Laplace distribution.</h2><span id='topic+denLp'></span>

<h3>Description</h3>

<p>Laplace distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>denLp(rr, sig)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="denLp_+3A_rr">rr</code></td>
<td>
<p>Shift from the location parameter</p>
</td></tr>
<tr><td><code id="denLp_+3A_sig">sig</code></td>
<td>
<p>Scale parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Laplace density.
</p>

<hr>
<h2 id='DeOut'>DeOut : Detect outlier observations.</h2><span id='topic+DeOut'></span>

<h3>Description</h3>

<p>Detect outlier observations from a vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DeOut(daData, method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DeOut_+3A_dadata">daData</code></td>
<td>
<p>A numerical vector.</p>
</td></tr>
<tr><td><code id="DeOut_+3A_method">method</code></td>
<td>
<p>Choose from '3sigma','hampel' and 'boxplot'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>indices of outlier observations.
</p>

<hr>
<h2 id='flexmix_2'>flexmix_2: Multiple runs of MLE based mixture regression to stabilize the output.</h2><span id='topic+flexmix_2'></span>

<h3>Description</h3>

<p>Mixture regression based on MLE could be unstable when assuming unequal variance. Multiple runs of flexmix is performed to stabilize the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flexmix_2(formula, data1, k, mprior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flexmix_2_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="flexmix_2_+3A_data1">data1</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="flexmix_2_+3A_k">k</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="flexmix_2_+3A_mprior">mprior</code></td>
<td>
<p>A numeric number in (0,1) that specifies the minimum proportion of samples in each mixing components.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A S4 object of flexmix class. xxx
</p>

<hr>
<h2 id='gaussData'>RobMixReg package built-in gaussian example data.</h2><span id='topic+gaussData'></span>

<h3>Description</h3>

<p>A dataset generated from gaussian distribution in RobMixReg package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaussData
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 rows and 3 variables:
</p>

<dl>
<dt>x</dt><dd><p>x variable</p>
</dd>
<dt>y</dt><dd><p>y variable</p>
</dd>
<dt>c</dt><dd><p>cluster information</p>
</dd>
</dl>

<hr>
<h2 id='lars.lsa'>lars variant for LSA.</h2><span id='topic+lars.lsa'></span>

<h3>Description</h3>

<p>lars variant for LSA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lars.lsa(Sigma0, b0, intercept, n, type = c("lasso", "lar"),
  eps = .Machine$double.eps, max.steps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lars.lsa_+3A_sigma0">Sigma0</code></td>
<td>
<p>The parameter.</p>
</td></tr>
<tr><td><code id="lars.lsa_+3A_b0">b0</code></td>
<td>
<p>The intercept of the regression line.</p>
</td></tr>
<tr><td><code id="lars.lsa_+3A_intercept">intercept</code></td>
<td>
<p>The bool variable of whether consider the intercept situation</p>
</td></tr>
<tr><td><code id="lars.lsa_+3A_n">n</code></td>
<td>
<p>The number of data point.</p>
</td></tr>
<tr><td><code id="lars.lsa_+3A_type">type</code></td>
<td>
<p>Regression options, choose form &quot;lasso&quot; or &quot;lar&quot;.</p>
</td></tr>
<tr><td><code id="lars.lsa_+3A_eps">eps</code></td>
<td>
<p>The converge threshold defined by the machine.</p>
</td></tr>
<tr><td><code id="lars.lsa_+3A_max.steps">max.steps</code></td>
<td>
<p>The maximum iteration times to stop.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object.
</p>


<h3>Author(s)</h3>

<p>Reference Wang, H. and Leng, C. (2006) and Efron et al. (2004).
</p>

<hr>
<h2 id='logLik_mixtureReg'>Obtain Log-likelihood from a mixtureReg Object</h2><span id='topic+logLik_mixtureReg'></span>

<h3>Description</h3>

<p>S3 method for class 'mixtureReg'.
However, it doesn't return a 'logLik' object.
For simlicity, it returns a 'numeric' value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logLik_mixtureReg(mixtureModel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logLik_mixtureReg_+3A_mixturemodel">mixtureModel</code></td>
<td>
<p>mixtureReg object, typically result from 'mixtureReg()'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a numeric value of log likelihood.
</p>

<hr>
<h2 id='lsa'>Least square approximation. This version Oct 19, 2006.</h2><span id='topic+lsa'></span>

<h3>Description</h3>

<p>Least square approximation. This version Oct 19, 2006.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsa(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lsa_+3A_obj">obj</code></td>
<td>
<p>lm/glm/coxph or other object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>beta.ols: the MLE estimate ; beta.bic: the LSA-BIC estimate ; beta.aic: the LSA-AIC estimate.
</p>


<h3>Author(s)</h3>

<p>Reference Wang, H. and Leng, C. (2006) and Efron et al. (2004).
</p>

<hr>
<h2 id='mixlinrb_bi'>mixlinrb_bi: mixlinrb_bione estimates the mixture regression parameters robustly using bisquare function based on multiply initial value.</h2><span id='topic+mixlinrb_bi'></span><span id='topic+mixlinrb_bi+2Cformula+2CANY+2Cnumeric+2Cnumeric-method'></span>

<h3>Description</h3>

<p>An EM-type of parameter estimation by replacing the least square estimation in the M-step with a robust criterior.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixlinrb_bi(formula, data, nc = 2, nit = 200)

## S4 method for signature 'formula,ANY,numeric,numeric'
mixlinrb_bi(formula, data,
  nc = 2, nit = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixlinrb_bi_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="mixlinrb_bi_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="mixlinrb_bi_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="mixlinrb_bi_+3A_nit">nit</code></td>
<td>
<p>Number of iterations for biSauqre method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated coefficients of all components.
</p>

<hr>
<h2 id='mixlinrb_bione'>mixlinrb_bione : mixlinrb_bione estimates the mixture regression parameters robustly using bisquare function based on one initial value.</h2><span id='topic+mixlinrb_bione'></span>

<h3>Description</h3>

<p>An EM-type of parameter estimation by replacing the least square estimation in the M-step with a robust criterior.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixlinrb_bione(formula, data, nc = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixlinrb_bione_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="mixlinrb_bione_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="mixlinrb_bione_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated coefficients of all components.
</p>

<hr>
<h2 id='mixLp'>mixLp : mixLp_one estimates the mixture regression parameters robustly using Laplace distribution based on multiply initial value..</h2><span id='topic+mixLp'></span><span id='topic+mixLp+2Cformula+2CANY+2Cnumeric+2Cnumeric-method'></span><span id='topic+CTLE+2Cformula+2CANY+2Cnumeric+2Cnumeric-method'></span>

<h3>Description</h3>

<p>mixLp estimates the mixture regression parameters robustly using bisquare function based on multiple initial values. The solution is found by the modal solution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixLp(formula, data, nc=2, nit=200)

## S4 method for signature 'formula,ANY,numeric,numeric'
mixLp(formula, data, nc = 2,
  nit = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixLp_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="mixLp_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="mixLp_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="mixLp_+3A_nit">nit</code></td>
<td>
<p>Number of iterations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated coefficients of all components.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("RobMixReg")
formula01=as.formula("y~x")
x=(gaussData$x);y=as.numeric(gaussData$y);
example_data01=data.frame(x,y)
res = mixLp(formula01, example_data01, nc=2, nit=20)
</code></pre>

<hr>
<h2 id='mixLp_one'>mixLp_one : mixLp_one estimates the mixture regression parameters robustly using Laplace distribution based on one initial value.</h2><span id='topic+mixLp_one'></span>

<h3>Description</h3>

<p>Robust mixture regression assuming that the error terms follow a Laplace distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixLp_one(formula, data, nc = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixLp_one_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="mixLp_one_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="mixLp_one_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated coefficients of all components.
</p>

<hr>
<h2 id='mixtureReg'>Function to Fit Mixture of Regressions</h2><span id='topic+mixtureReg'></span>

<h3>Description</h3>

<p>The main function in this package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixtureReg(regData, formulaList, xName = NULL, yName = NULL,
  mixingProb = c("Constant", "loess"), initialWList = NULL,
  epsilon = 1e-08, max_iter = 10000, max_restart = 15,
  min_lambda = 0.01, min_sigmaRatio = 0.1, silently = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixtureReg_+3A_regdata">regData</code></td>
<td>
<p>data frame used in fitting model.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_formulalist">formulaList</code></td>
<td>
<p>a list of the regression components that need to be estimated.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_xname">xName</code></td>
<td>
<p>character; Name used to pick x variable from data.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_yname">yName</code></td>
<td>
<p>character; Name used to pick y variable from data.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_mixingprob">mixingProb</code></td>
<td>
<p>character;
Specify how the mixing probabilities are estimated in the M step.
&quot;Constant&quot; specifies a constant mixing probabilities;
&quot;loess&quot; specifies predictor dependent mixing probabilities obtained by loess smoothing.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_initialwlist">initialWList</code></td>
<td>
<p>a list of weights guesses (provided by user).
Typically this is not used, unless the user has a good initial guess.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_epsilon">epsilon</code></td>
<td>
<p>a small value that the function consider as zero.
The value is used in determine matrix sigularity and in determine convergence.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_max_iter">max_iter</code></td>
<td>
<p>the maximum number of iterations.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_max_restart">max_restart</code></td>
<td>
<p>the maximum number of restart before giving up.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_min_lambda">min_lambda</code></td>
<td>
<p>a value used to ensure estimated mixing probabilities (lambda's) are not too close to zero.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_min_sigmaratio">min_sigmaRatio</code></td>
<td>
<p>a value used to prevent estimated variaces of any regression component from collapsing to zero.</p>
</td></tr>
<tr><td><code id="mixtureReg_+3A_silently">silently</code></td>
<td>
<p>a switch to turn off the screen printout.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A class 'mixtureReg' object.
</p>


<h3>Author(s)</h3>

<p>The mixtureReg package is developed by Tianxia Zhou on github. All right reserved by Tianxia Zhou.
</p>

<hr>
<h2 id='MLM'>The main function of mining the latent relationship among variables.</h2><span id='topic+MLM'></span>

<h3>Description</h3>

<p>The main function of mining the latent relationship among variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MLM(ml.method = "rlr", rmr.method = "cat",
  b.formulaList = list(formula(y ~ x), formula(y ~ 1)), formula = y ~
  x, nit = 1, nc = 2, x = NULL, y = NULL, max_iter = 50,
  tRatio = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MLM_+3A_ml.method">ml.method</code></td>
<td>
<p>The option to select the four methods in vignette.</p>
</td></tr>
<tr><td><code id="MLM_+3A_rmr.method">rmr.method</code></td>
<td>
<p>The option to select the robust mixture regression method.</p>
</td></tr>
<tr><td><code id="MLM_+3A_b.formulalist">b.formulaList</code></td>
<td>
<p>The case b require the user provide the formula list. This enable the flexible mixture regression.</p>
</td></tr>
<tr><td><code id="MLM_+3A_formula">formula</code></td>
<td>
<p>The linear relationship between two variables.</p>
</td></tr>
<tr><td><code id="MLM_+3A_nit">nit</code></td>
<td>
<p>Number of iterations for CTLE, mixbi, mixLp.</p>
</td></tr>
<tr><td><code id="MLM_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="MLM_+3A_x">x</code></td>
<td>
<p>The matrix x of the high dimension situation.</p>
</td></tr>
<tr><td><code id="MLM_+3A_y">y</code></td>
<td>
<p>The external outcome variable.</p>
</td></tr>
<tr><td><code id="MLM_+3A_max_iter">max_iter</code></td>
<td>
<p>Maximum iteration for TLE method.</p>
</td></tr>
<tr><td><code id="MLM_+3A_tratio">tRatio</code></td>
<td>
<p>The ratio of the outliers in the TLE robust mixture regression method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Main result object.
</p>

<hr>
<h2 id='MLM_bic'>Model selection function for low dimension data.</h2><span id='topic+MLM_bic'></span>

<h3>Description</h3>

<p>Model selection function for low dimension data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MLM_bic(ml.method = "rlr", x, y, nc = 1, formulaList = NULL, K = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MLM_bic_+3A_ml.method">ml.method</code></td>
<td>
<p>The parameter to choose the fitted model for calculating the BIC</p>
</td></tr>
<tr><td><code id="MLM_bic_+3A_x">x</code></td>
<td>
<p>x variable.</p>
</td></tr>
<tr><td><code id="MLM_bic_+3A_y">y</code></td>
<td>
<p>y variable.</p>
</td></tr>
<tr><td><code id="MLM_bic_+3A_nc">nc</code></td>
<td>
<p>The component number for low dimensional feature</p>
</td></tr>
<tr><td><code id="MLM_bic_+3A_formulalist">formulaList</code></td>
<td>
<p>The list of target formular</p>
</td></tr>
<tr><td><code id="MLM_bic_+3A_k">K</code></td>
<td>
<p>The component number for high dimensional feature</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BIC value.
</p>

<hr>
<h2 id='MLM_cv'>Cross validation (fold-5) function for high dimension data.</h2><span id='topic+MLM_cv'></span>

<h3>Description</h3>

<p>Cross validation (fold-5) function for high dimension data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MLM_cv(x = NULL, y = NULL, nit = 1, nc = 2, max_iter = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MLM_cv_+3A_x">x</code></td>
<td>
<p>x variable.</p>
</td></tr>
<tr><td><code id="MLM_cv_+3A_y">y</code></td>
<td>
<p>y variable.</p>
</td></tr>
<tr><td><code id="MLM_cv_+3A_nit">nit</code></td>
<td>
<p>Iteration number.</p>
</td></tr>
<tr><td><code id="MLM_cv_+3A_nc">nc</code></td>
<td>
<p>The number of component.</p>
</td></tr>
<tr><td><code id="MLM_cv_+3A_max_iter">max_iter</code></td>
<td>
<p>Maximum iteration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The correlation between y and y_hat based on five fold cross validation.
</p>

<hr>
<h2 id='orderedLines'>Sort by X Coordinates and Add Line to a Plot</h2><span id='topic+orderedLines'></span>

<h3>Description</h3>

<p>Rearrange X and Y coordinates before calling &quot;lines()&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orderedLines(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orderedLines_+3A_x">x</code></td>
<td>
<p>X coordinate vectors of points to join.</p>
</td></tr>
<tr><td><code id="orderedLines_+3A_y">y</code></td>
<td>
<p>Y coordinate vectors of points to join.</p>
</td></tr>
<tr><td><code id="orderedLines_+3A_...">...</code></td>
<td>
<p>Further graphical parameters.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_CTLE'>plot_CTLE: Plot the mixture/single regression line(s) in a simply function.</h2><span id='topic+plot_CTLE'></span><span id='topic+plot_CTLE+2Cformula+2CANY+2Cnumeric-method'></span><span id='topic+CTLERob+2Cformula+2CANY+2Cnumeric+2CANY'></span>

<h3>Description</h3>

<p>CTLERob performes robust linear regression with high breakdown point and high efficiency in each mixing components and adaptively remove the outlier samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_CTLE(formula, data, nc = 2, inds_in)

## S4 method for signature 'formula,ANY,numeric'
plot_CTLE(formula, data, nc = 2, inds_in)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_CTLE_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="plot_CTLE_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="plot_CTLE_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="plot_CTLE_+3A_inds_in">inds_in</code></td>
<td>
<p>The index of the point which belongs to the current regression line.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_mixtureReg'>Plot Fit and Mixing Probability of a mixtureReg Object</h2><span id='topic+plot_mixtureReg'></span>

<h3>Description</h3>

<p>S3 plot method for class 'mixtureReg'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_mixtureReg(mixtureModel, which = 1:2, xName = NULL,
  yName = NULL, xlab = NULL, ylab = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_mixtureReg_+3A_mixturemodel">mixtureModel</code></td>
<td>
<p>mixtureReg object, typically result from 'mixtureReg()'.</p>
</td></tr>
<tr><td><code id="plot_mixtureReg_+3A_which">which</code></td>
<td>
<p>numeric; choose which plot to display.
'1' gives a plot of fit; '2' gives a plot of mixing probability.</p>
</td></tr>
<tr><td><code id="plot_mixtureReg_+3A_xname">xName</code></td>
<td>
<p>character; Name used to pick x variable from data.</p>
</td></tr>
<tr><td><code id="plot_mixtureReg_+3A_yname">yName</code></td>
<td>
<p>character; Name used to pick y variable from data.</p>
</td></tr>
<tr><td><code id="plot_mixtureReg_+3A_xlab">xlab</code></td>
<td>
<p>character; label that should be put on the x axis.</p>
</td></tr>
<tr><td><code id="plot_mixtureReg_+3A_ylab">ylab</code></td>
<td>
<p>character; label that should be put on the y axis.</p>
</td></tr>
<tr><td><code id="plot_mixtureReg_+3A_...">...</code></td>
<td>
<p>Further graphical parameters.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_mixtureRegList'>Plot a List of mixtureReg Objects</h2><span id='topic+plot_mixtureRegList'></span>

<h3>Description</h3>

<p>Feed in a list of mixtureReg models and get an overlayed plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_mixtureRegList(mixtureRegList, xName = NULL, yName = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_mixtureRegList_+3A_mixturereglist">mixtureRegList</code></td>
<td>
<p>a list of multiple mixtureReg objects.</p>
</td></tr>
<tr><td><code id="plot_mixtureRegList_+3A_xname">xName</code></td>
<td>
<p>character; Name used to pick x variable from data.</p>
</td></tr>
<tr><td><code id="plot_mixtureRegList_+3A_yname">yName</code></td>
<td>
<p>character; Name used to pick y variable from data.</p>
</td></tr>
<tr><td><code id="plot_mixtureRegList_+3A_...">...</code></td>
<td>
<p>Further graphical parameters.</p>
</td></tr>
</table>

<hr>
<h2 id='Rec_Lm'>Adaptive lasso.</h2><span id='topic+Rec_Lm'></span>

<h3>Description</h3>

<p>Adaptive lasso.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rec_Lm(XX, yy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Rec_Lm_+3A_xx">XX</code></td>
<td>
<p>The independent variable.</p>
</td></tr>
<tr><td><code id="Rec_Lm_+3A_yy">yy</code></td>
<td>
<p>The dependent variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consist of index of selected variable and coefficient for all variables.
</p>

<hr>
<h2 id='rmr'>The main function of Robust Mixture Regression using five methods.</h2><span id='topic+rmr'></span>

<h3>Description</h3>

<p>The main function of Robust Mixture Regression using five methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmr(lr.method = "flexmix", formula = NULL, data = NULL, nc = 2,
  nit = 20, tRatio = 0.05, MaxIt = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmr_+3A_lr.method">lr.method</code></td>
<td>
<p>A robust mixture regression method to be used. Should be one of &quot;flexmix&quot;, &quot;TLE&quot;, &quot;CTLERob&quot;, &quot;mixbi&quot;,&quot;mixLp&quot;.</p>
</td></tr>
<tr><td><code id="rmr_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="rmr_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="rmr_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="rmr_+3A_nit">nit</code></td>
<td>
<p>Number of iterations for CTLE, mixbi, mixLp.</p>
</td></tr>
<tr><td><code id="rmr_+3A_tratio">tRatio</code></td>
<td>
<p>Trimming proportion for TLE method.</p>
</td></tr>
<tr><td><code id="rmr_+3A_maxit">MaxIt</code></td>
<td>
<p>Maximum iteration for TLE method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An S4 object about the regression result.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(RobMixReg)
#library(robust)
library(flexmix)
library(robustbase)
library(MASS)
library(gtools)
# gaussData
x=(gaussData$x);y=as.numeric(gaussData$y);
formula01=as.formula("y~x")
example_data01=data.frame(x,y)
res_rmr = rmr(lr.method='flexmix', formula=formula01, data=example_data01)
res_rmr = rmr(lr.method='CTLERob', formula=formula01, data=example_data01)

</code></pre>

<hr>
<h2 id='RobMixReg-class'>Class RobMixReg.</h2><span id='topic+RobMixReg-class'></span>

<h3>Description</h3>

<p>Class <code>RobMixReg</code> defines a robust mixture regression class as a S4 object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>inds_in</code></dt><dd><p>The indices of observations used in the parameter estimation.</p>
</dd>
<dt><code>indout</code></dt><dd><p>The indices of outlier samples, not used in the parameter estimation.</p>
</dd>
<dt><code>ctleclusters</code></dt><dd><p>The cluster membership of each observation.</p>
</dd>
<dt><code>compcoef</code></dt><dd><p>Regression coefficients for each component.</p>
</dd>
<dt><code>comppvals</code></dt><dd><p>Component p values.</p>
</dd>
<dt><code>compwww</code></dt><dd><p>The posterior of the clustering.</p>
</dd>
<dt><code>call</code></dt><dd><p>Call function.</p>
</dd>
</dl>

<hr>
<h2 id='simu_data_sparse'>Simulate high dimension data for RBSL algorithm validation.</h2><span id='topic+simu_data_sparse'></span>

<h3>Description</h3>

<p>Simulate high dimension data for RBSL algorithm validation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu_data_sparse(n, bet, pr, sigma)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu_data_sparse_+3A_n">n</code></td>
<td>
<p>Patient number.</p>
</td></tr>
<tr><td><code id="simu_data_sparse_+3A_bet">bet</code></td>
<td>
<p>The coefficient matrix.</p>
</td></tr>
<tr><td><code id="simu_data_sparse_+3A_pr">pr</code></td>
<td>
<p>A vector of probability threshold which simulate the sampling based on uniform distribution.</p>
</td></tr>
<tr><td><code id="simu_data_sparse_+3A_sigma">sigma</code></td>
<td>
<p>A vector of noise level. The length should be equal to the component number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consist of x, y, true cluster label.
</p>

<hr>
<h2 id='simu_func'>The simulation function for low/high dimensional space.</h2><span id='topic+simu_func'></span>

<h3>Description</h3>

<p>The simulation function for low/high dimensional space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu_func(beta, sigma, alpha = NULL, n = 400)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu_func_+3A_beta">beta</code></td>
<td>
<p>The slope vector for low dimensional space or matrix for high dimensional space.</p>
</td></tr>
<tr><td><code id="simu_func_+3A_sigma">sigma</code></td>
<td>
<p>A vector whose k-th element is the standard deviation for the k-th regression component.</p>
</td></tr>
<tr><td><code id="simu_func_+3A_alpha">alpha</code></td>
<td>
<p>The parameter to control the number of outliers for low dimensional space.</p>
</td></tr>
<tr><td><code id="simu_func_+3A_n">n</code></td>
<td>
<p>The sample number for high dimensional data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object.
</p>

<hr>
<h2 id='simu_low'>The simulation function for low dimensional space.</h2><span id='topic+simu_low'></span>

<h3>Description</h3>

<p>The simulation function for low dimensional space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu_low(beta, inter, alpha = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu_low_+3A_beta">beta</code></td>
<td>
<p>The slope vector.</p>
</td></tr>
<tr><td><code id="simu_low_+3A_inter">inter</code></td>
<td>
<p>The intercept vector.</p>
</td></tr>
<tr><td><code id="simu_low_+3A_alpha">alpha</code></td>
<td>
<p>The parameter to control the number of outliers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object consists of the x variable in low dimensional space and the external y variable.
</p>

<hr>
<h2 id='simuData'>RobMixReg package built-in simulated example data.</h2><span id='topic+simuData'></span>

<h3>Description</h3>

<p>A simulation dataset from RobMixReg package. This simulation dataset is in dimension 2 and ground truth (include outliers label) of the cluster information also generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simuData
</code></pre>


<h3>Format</h3>

<p>A data frame with 500 rows and 5 variables:
</p>

<dl>
<dt>X1</dt><dd><p>X1 variable</p>
</dd>
<dt>X2</dt><dd><p>X2 variable</p>
</dd>
<dt>y</dt><dd><p>y variable</p>
</dd>
<dt>c</dt><dd><p>cluster information</p>
</dd>
<dt>outlier</dt><dd><p>outlier indicator</p>
</dd>
</dl>

<hr>
<h2 id='TLE'>TLE: robust mixture regression based on trimmed likelihood estimation.</h2><span id='topic+TLE'></span><span id='topic+TLE+2Cformula+2CANY+2Cnumeric+2Cnumeric+2Cnumeric-method'></span><span id='topic+CTLE+2Cformula+2CANY+2Cnumeric+2Cnumeric+2Cnumeric-method'></span>

<h3>Description</h3>

<p>The algorithm fits a mixture regression model after trimming a proportion of the observations, given by tRatio.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TLE(formula, data, nc = 2, tRatio, MaxIt = 200)

## S4 method for signature 'formula,ANY,numeric,numeric,numeric'
TLE(formula, data,
  nc = 2, tRatio, MaxIt = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TLE_+3A_formula">formula</code></td>
<td>
<p>A symbolic description of the model to be fit.</p>
</td></tr>
<tr><td><code id="TLE_+3A_data">data</code></td>
<td>
<p>A data frame containing the predictor and response variables, where the last column is the response varible.</p>
</td></tr>
<tr><td><code id="TLE_+3A_nc">nc</code></td>
<td>
<p>Number of mixture components.</p>
</td></tr>
<tr><td><code id="TLE_+3A_tratio">tRatio</code></td>
<td>
<p>Trimming proportion.</p>
</td></tr>
<tr><td><code id="TLE_+3A_maxit">MaxIt</code></td>
<td>
<p>Maximum iteration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A S4 object of RobMixReg class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("RobMixReg")
formula01=as.formula("y~x")
x=(gaussData$x);y=as.numeric(gaussData$y);
example_data01=data.frame(x,y)

res = TLE(formula01,example_data01, nc=2,tRatio=0.05,MaxIt=200)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
