<!DOCTYPE html><html lang="en"><head><title>Help for package romic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {romic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#romic-package'><p>romic: R for High-Dimensional Omic Data</p></a></li>
<li><a href='#add_pcs'><p>Add PCA Loadings</p></a></li>
<li><a href='#app_flow'><p>Flow</p></a></li>
<li><a href='#app_heatmap'><p>Interactive Heatmap</p></a></li>
<li><a href='#app_pcs'><p>PC Plot</p></a></li>
<li><a href='#brauer_2008'><p>Brauer 2008</p></a></li>
<li><a href='#center_tomic'><p>Center T* Omic</p></a></li>
<li><a href='#check_tidy_omic'><p>Check Tidy Omic</p></a></li>
<li><a href='#check_tomic'><p>Check T*Omic</p></a></li>
<li><a href='#check_triple_omic'><p>Check Triple Omic</p></a></li>
<li><a href='#convert_wide_to_tidy_omic'><p>Convert Wide to Tidy Omic</p></a></li>
<li><a href='#create_tidy_omic'><p>Create Tidy Omic</p></a></li>
<li><a href='#create_triple_omic'><p>Create Triple Omic</p></a></li>
<li><a href='#downsample_heatmap'><p>Downsample Heatmap</p></a></li>
<li><a href='#export_tomic_as_tidy'><p>Export T*Omic in Tidy Format</p></a></li>
<li><a href='#export_tomic_as_triple'><p>Export T*Omic as Triple</p></a></li>
<li><a href='#export_tomic_as_wide'><p>Export T*Omic as Wide Data</p></a></li>
<li><a href='#filter_tomic'><p>Filter T* Omics</p></a></li>
<li><a href='#filterInput'><p>Filter Input</p></a></li>
<li><a href='#filterServer'><p>Filter Server</p></a></li>
<li><a href='#format_names_for_plotting'><p>Format Names for Plotting</p></a></li>
<li><a href='#get_design_tbl'><p>Get Design Table</p></a></li>
<li><a href='#get_tomic_table'><p>Get Tomic Table</p></a></li>
<li><a href='#ggBivOutput'><p>ggBivariate Output</p></a></li>
<li><a href='#ggBivServer'><p>ggBivariate Server</p></a></li>
<li><a href='#ggplotOutput'><p>ggplot Output</p></a></li>
<li><a href='#ggplotServer'><p>ggplot Server</p></a></li>
<li><a href='#ggUnivOutput'><p>ggUnivariate Output</p></a></li>
<li><a href='#ggUnivServer'><p>ggUnivariate Server</p></a></li>
<li><a href='#hclust_order'><p>Hierarchical clustering order</p></a></li>
<li><a href='#impute_missing_values'><p>Impute Missing Values</p></a></li>
<li><a href='#infer_tomic_table_type'><p>Infer Tomic Table Type</p></a></li>
<li><a href='#lassoInput'><p>Lasso Input</p></a></li>
<li><a href='#lassoServer'><p>Lasso Server</p></a></li>
<li><a href='#organizeInput'><p>Organize Input</p></a></li>
<li><a href='#organizeServer'><p>Organize Servers</p></a></li>
<li><a href='#plot_bivariate'><p>Bivariate Plot</p></a></li>
<li><a href='#plot_heatmap'><p>Plot Heatmap</p></a></li>
<li><a href='#plot_univariate'><p>Univariate Plot</p></a></li>
<li><a href='#plotsaverInput'><p>Plot Saver Input</p></a></li>
<li><a href='#plotsaverServer'><p>Plot Saver Server</p></a></li>
<li><a href='#prepare_example_datasets'><p>Prepare Example Datasets</p></a></li>
<li><a href='#reconcile_triple_omic'><p>Reconcile Triple Omic</p></a></li>
<li><a href='#remove_missing_values'><p>Remove Missing Values</p></a></li>
<li><a href='#shiny_filter_test'><p>Shiny Filter Test</p></a></li>
<li><a href='#shiny_ggbiv_test'><p>Shiny ggBivariate Test</p></a></li>
<li><a href='#shiny_ggplot_test'><p>Shiny ggplot Test</p></a></li>
<li><a href='#shiny_gguniv_test'><p>Shiny ggUnivariate Test</p></a></li>
<li><a href='#shiny_lasso_test'><p>Shiny Lasso Test</p></a></li>
<li><a href='#shiny_lasso_test_reactval'><p>Shiny Lasso Test w/ Reactive Values</p></a></li>
<li><a href='#shiny_organize_test'><p>Shiny Organize Test</p></a></li>
<li><a href='#shiny_plotsaver_test'><p>Shiny Plot Saver Test</p></a></li>
<li><a href='#shiny_sort_test'><p>Shiny Sort Test</p></a></li>
<li><a href='#sort_tomic'><p>Sort Triple Omic</p></a></li>
<li><a href='#sort_triple_arrange'><p>Sort Triple Arrange</p></a></li>
<li><a href='#sort_triple_hclust'><p>Sort Triple Hclust</p></a></li>
<li><a href='#sortInput'><p>Sort Input</p></a></li>
<li><a href='#sortServer'><p>Sort Server</p></a></li>
<li><a href='#tidy_to_triple'><p>Tidy omic to triple omic</p></a></li>
<li><a href='#tomic_sort_status'><p>T* Omic Sort Status</p></a></li>
<li><a href='#tomic_to'><p>T* Omic To</p></a></li>
<li><a href='#triple_to_tidy'><p>Triple Omic to Tidy Omic</p></a></li>
<li><a href='#try_brushedPoints'><p>Try brushedPoints</p></a></li>
<li><a href='#update_tidy_omic'><p>Update Tidy Omic</p></a></li>
<li><a href='#update_tomic'><p>Update T* Omic</p></a></li>
<li><a href='#var_partial_match'><p>Var Partial Match</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>R for High-Dimensional Omic Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sean Hackett &lt;sean@calicolabs.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Represents high-dimensional data as tables of features, samples and measurements, and a design list for tracking the meaning of individual variables. Using this format, filtering, normalization, and other transformations of a dataset can be carried out in a flexible manner. 'romic' takes advantage of these transformations to create interactive 'shiny' apps for exploratory data analysis such as an interactive heatmap. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>checkmate, cli, dplyr, ggplot2, glue, purrr, readr, reshape2,
rlang, shiny (&ge; 1.5.0), stringr, tibble, tidyr (&ge; 1.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, impute, lazyeval, plotly, rmarkdown, usethis, testthat
(&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-21 04:48:49 UTC; sean</td>
</tr>
<tr>
<td>Author:</td>
<td>Sean Hackett <a href="https://orcid.org/0000-0002-9553-4341"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Calico Life Sciences LLC [cph, fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-21 05:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='romic-package'>romic: R for High-Dimensional Omic Data</h2><span id='topic+romic'></span><span id='topic+romic-package'></span>

<h3>Description</h3>

<p>'romic' represents high-dimensional data as tables of features,
samples and measurements, and a design list for tracking the meaning of
individual variables. Using this format, filtering, normalization, and
other transformations of a dataset can be carried out in a flexible
manner. 'romic' takes advantage of these transformations to create
interactive shiny apps for exploratory data analysis such as an
interactive heatmap.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Sean Hackett <a href="mailto:sean@calicolabs.com">sean@calicolabs.com</a> (<a href="https://orcid.org/0000-0002-9553-4341">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Calico Life Sciences LLC [copyright holder, funder]
</p>
</li></ul>


<hr>
<h2 id='add_pcs'>Add PCA Loadings</h2><span id='topic+add_pcs'></span>

<h3>Description</h3>

<p>Add Principal Components Analysis Loadings to a tidy or triple omics
dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_pcs(
  tomic,
  value_var = NULL,
  center_rows = TRUE,
  npcs = NULL,
  missing_val_method = "drop_samples",
  label_percent_varex = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_pcs_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="add_pcs_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
<tr><td><code id="add_pcs_+3A_center_rows">center_rows</code></td>
<td>
<p>center rows before performing PCA</p>
</td></tr>
<tr><td><code id="add_pcs_+3A_npcs">npcs</code></td>
<td>
<p>number of principal component loadings to add to samples
(default is number of samples)</p>
</td></tr>
<tr><td><code id="add_pcs_+3A_missing_val_method">missing_val_method</code></td>
<td>
<p>Approach to remove missing values:
</p>

<dl>
<dt>drop_features</dt><dd><p>Drop features with missing values</p>
</dd>
<dt>drop_samples</dt><dd><p>Drop samples which are missing all features,
then drop features</p>
</dd>
<dt>impute</dt><dd><p>Impute missing values</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="add_pcs_+3A_label_percent_varex">label_percent_varex</code></td>
<td>
<p>If true then PCs will be labelled by the percent
of variability they explain.</p>
</td></tr>
<tr><td><code id="add_pcs_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object with principal components added to samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>add_pcs(brauer_2008_triple, npcs = 5)

</code></pre>

<hr>
<h2 id='app_flow'>Flow</h2><span id='topic+app_flow'></span>

<h3>Description</h3>

<p>Using <code>shiny</code> comb through datasets by iterating between
plotting steps, and lassoing steps to select points of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>app_flow(tomic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="app_flow_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  # library(reactlog)
  # reactlog_enable()
  app_flow(brauer_2008_triple)
  # shiny::reactlogShow()
}

</code></pre>

<hr>
<h2 id='app_heatmap'>Interactive Heatmap</h2><span id='topic+app_heatmap'></span>

<h3>Description</h3>

<p>Generate a <code>shiny</code> interactive heatmap that allows for on demand
filtering, ordering and faceting by variables of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>app_heatmap(tomic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="app_heatmap_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  app_heatmap(brauer_2008_tidy)
}
</code></pre>

<hr>
<h2 id='app_pcs'>PC Plot</h2><span id='topic+app_pcs'></span>

<h3>Description</h3>

<p>Generate a Shiny interactive scatter plot which allows visualization of
features, measurements, and samples (with principal components added).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>app_pcs(tomic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="app_pcs_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  app_pcs(brauer_2008_tidy)
}
</code></pre>

<hr>
<h2 id='brauer_2008'>Brauer 2008</h2><span id='topic+brauer_2008'></span><span id='topic+brauer_2008_tidy'></span><span id='topic+brauer_2008_triple'></span>

<h3>Description</h3>

<p>An RNA expression (microarray) dataset looking at how yeast gene expression
changes as nutrient sources and nutrient richness changes.
</p>
<p><code><a href="#topic+brauer_2008">brauer_2008</a></code> formatted as a tidy_omic object
</p>
<p><code><a href="#topic+brauer_2008">brauer_2008</a></code> formatted as a triple_omic object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brauer_2008

brauer_2008_tidy

brauer_2008_triple
</code></pre>


<h3>Format</h3>

<p>A tibble with 18,000 rows and 8 columns:
</p>

<dl>
<dt>name</dt><dd><p>Common gene name</p>
</dd>
<dt>BP</dt><dd><p>Gene ontology biological process of the gene</p>
</dd>
<dt>MF</dt><dd><p>Gene ontology molecular function of the gene</p>
</dd>
<dt>sample</dt><dd><p>Sample name</p>
</dd>
<dt>nutrient</dt><dd><p>Which nutrient limits growth (Glucose, Nitrogen,
Phosphorous, Sulfur, Uracil, Leucine</p>
</dd>
<dt>DR</dt><dd><p>Dilution rate of the culture - basically how fast the cells
are growing</p>
</dd>
<dt>expression</dt><dd><p>Expression level of the gene, log2 observation relative
to a replicate of G0.3</p>
</dd>
</dl>

<p>An object of class <code>tidy_omic</code> (inherits from <code>tomic</code>, <code>general</code>) of length 2.
</p>
<p>An object of class <code>triple_omic</code> (inherits from <code>tomic</code>, <code>general</code>) of length 4.
</p>


<h3>Details</h3>

<p>This version of the dataset contains only 500 genes randomly
selected from the ~6K genes in the complete dataset.
</p>


<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/17959824/">https://pubmed.ncbi.nlm.nih.gov/17959824/</a>
</p>

<hr>
<h2 id='center_tomic'>Center T* Omic</h2><span id='topic+center_tomic'></span>

<h3>Description</h3>

<p>Center each measurement by subtracting the mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>center_tomic(tomic, measurement_vars = "all")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="center_tomic_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="center_tomic_+3A_measurement_vars">measurement_vars</code></td>
<td>
<p>measurement variables to center</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object where one or more measurements have
been centered on a feature-by-feature basis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>center_tomic(brauer_2008_tidy)
</code></pre>

<hr>
<h2 id='check_tidy_omic'>Check Tidy Omic</h2><span id='topic+check_tidy_omic'></span>

<h3>Description</h3>

<p>Check a tidy omic dataset for consistency between the data and design and
validate that the dataset follows the <code>tidy_omic</code>/<code>tomic</code>
specification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tidy_omic(tidy_omic, fast_check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_tidy_omic_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
<tr><td><code id="check_tidy_omic_+3A_fast_check">fast_check</code></td>
<td>
<p>if TRUE then skip some checks which are slow and that are
generally only needed when a <code>tomic</code> object is first created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0 invisibly
</p>

<hr>
<h2 id='check_tomic'>Check T*Omic</h2><span id='topic+check_tomic'></span>

<h3>Description</h3>

<p>Check a tidy or triple 'omic object for common pathologies, such as a
mismatch between data and schema and non-uniqueness of primary keys.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tomic(tomic, fast_check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_tomic_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="check_tomic_+3A_fast_check">fast_check</code></td>
<td>
<p>if TRUE then skip some checks which are slow and that are
generally only needed when a <code>tomic</code> object is first created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0 invisibly
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_tomic(brauer_2008_triple)
</code></pre>

<hr>
<h2 id='check_triple_omic'>Check Triple Omic</h2><span id='topic+check_triple_omic'></span>

<h3>Description</h3>

<p>Check a triple omic dataset for consistency between the data and design and
validate that the dataset follows the <code>triple_omic</code>/<code>tomic</code>
specification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_triple_omic(triple_omic, fast_check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_triple_omic_+3A_triple_omic">triple_omic</code></td>
<td>
<p>an object of class triple_omic produced by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code></p>
</td></tr>
<tr><td><code id="check_triple_omic_+3A_fast_check">fast_check</code></td>
<td>
<p>if TRUE then skip some checks which are slow and that are
generally only needed when a <code>tomic</code> object is first created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0 invisibly
</p>

<hr>
<h2 id='convert_wide_to_tidy_omic'>Convert Wide to Tidy Omic</h2><span id='topic+convert_wide_to_tidy_omic'></span>

<h3>Description</h3>

<p>Convert a wide dataset of species' abundances (gene product, metabolites,
lipids, ...) into a triple_omic dataset (one observation per row)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_wide_to_tidy_omic(
  wide_df,
  feature_pk,
  feature_vars = NULL,
  sample_var = "sample",
  measurement_var = "abundance",
  omic_type_tag = "general",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_wide_to_tidy_omic_+3A_wide_df">wide_df</code></td>
<td>
<p>a data.frame (or tibble) containing 1+ columns of feature
attributes and many columns of samples</p>
</td></tr>
<tr><td><code id="convert_wide_to_tidy_omic_+3A_feature_pk">feature_pk</code></td>
<td>
<p>A unique identifier for features</p>
</td></tr>
<tr><td><code id="convert_wide_to_tidy_omic_+3A_feature_vars">feature_vars</code></td>
<td>
<p>a character vector of additional feature-level
variables (or NULL if there are no additional variables)</p>
</td></tr>
<tr><td><code id="convert_wide_to_tidy_omic_+3A_sample_var">sample_var</code></td>
<td>
<p>variable name to use for samples</p>
</td></tr>
<tr><td><code id="convert_wide_to_tidy_omic_+3A_measurement_var">measurement_var</code></td>
<td>
<p>variable name to use for measurements</p>
</td></tr>
<tr><td><code id="convert_wide_to_tidy_omic_+3A_omic_type_tag">omic_type_tag</code></td>
<td>
<p>an optional subtype of omic data: metabolomics,
lipidomics, proteomics, genomics, general</p>
</td></tr>
<tr><td><code id="convert_wide_to_tidy_omic_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tidy_omic</code> object as produced by <code>create_tidy_omic</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

wide_measurements &lt;- brauer_2008_triple[["measurements"]] %&gt;%
  tidyr::spread(sample, expression)

wide_df &lt;- brauer_2008_triple[["features"]] %&gt;%
  left_join(wide_measurements, by = "name")

convert_wide_to_tidy_omic(wide_df,
  feature_pk = "name",
  feature_vars = c("BP", "MF", "systematic_name")
)
</code></pre>

<hr>
<h2 id='create_tidy_omic'>Create Tidy Omic</h2><span id='topic+create_tidy_omic'></span>

<h3>Description</h3>

<p>A tidy omics object contains a formatted dataset and a summary of the
experimental design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tidy_omic(
  df,
  feature_pk,
  feature_vars = NULL,
  sample_pk,
  sample_vars = NULL,
  omic_type_tag = "general",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_tidy_omic_+3A_df">df</code></td>
<td>
<p>a data.frame (or tibble) containing some combination of feature,
sample and observation-level variables</p>
</td></tr>
<tr><td><code id="create_tidy_omic_+3A_feature_pk">feature_pk</code></td>
<td>
<p>A unique identifier for features</p>
</td></tr>
<tr><td><code id="create_tidy_omic_+3A_feature_vars">feature_vars</code></td>
<td>
<p>a character vector of additional feature-level
variables (or NULL if there are no additional variables)</p>
</td></tr>
<tr><td><code id="create_tidy_omic_+3A_sample_pk">sample_pk</code></td>
<td>
<p>A unique identifier for samples</p>
</td></tr>
<tr><td><code id="create_tidy_omic_+3A_sample_vars">sample_vars</code></td>
<td>
<p>a character vector of additional sample-level variables
(or NULL if there are no additional variables)</p>
</td></tr>
<tr><td><code id="create_tidy_omic_+3A_omic_type_tag">omic_type_tag</code></td>
<td>
<p>an optional subtype of omic data: metabolomics,
lipidomics, proteomics, genomics, general</p>
</td></tr>
<tr><td><code id="create_tidy_omic_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An S3 <code>tidy_omic</code>/<code>tomic</code> object built on a <code>list</code>:
</p>

<dl>
<dt>data</dt><dd><p>A tibble with one row per measurement (i.e., features x
samples)</p>
</dd>
<dt>design</dt><dd><p>A list which organized the dataset's meta-data:
</p>

<dl>
<dt>feature_pk</dt><dd><p>variable specifying a unique feature</p>
</dd>
<dt>sample_pk</dt><dd><p>variable specifying a unique sample</p>
</dd>
<dt>features</dt><dd><p>tibble of feature attributes</p>
</dd>
<dt>samples</dt><dd><p>tibble of sample attributes</p>
</dd>
<dt>measurements</dt><dd><p>tibble of measurement attributes</p>
</dd>
</dl>

</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

measurement_df &lt;- tidyr::expand_grid(
  feature_id = 1:10,
  sample_id = LETTERS[1:5]
) %&gt;%
  dplyr::mutate(value = rnorm(n()))

feature_df &lt;- tibble(
  feature_id = 1:10,
  feature_group = rep(c("a", "b"), each = 5)
)
sample_df &lt;- tibble(
  sample_id = LETTERS[1:5],
  sample_group = c("a", "a", "b", "b", "b")
)

triple_omic &lt;- create_triple_omic(
  measurement_df, feature_df, sample_df,
  "feature_id", "sample_id"
)
raw_tidy_omic &lt;- triple_to_tidy(triple_omic)$data

create_tidy_omic(raw_tidy_omic,
  feature_pk = "feature_id",
  feature_vars = "feature_group", sample_pk = "sample_id",
  sample_vars = "sample_group"
)
</code></pre>

<hr>
<h2 id='create_triple_omic'>Create Triple Omic</h2><span id='topic+create_triple_omic'></span>

<h3>Description</h3>

<p>A triple omics class contains three data.frames, one for features, one for
samples, and one for abundances. This is a good format when there is a
large amount of meta data associated with features or samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_triple_omic(
  measurement_df,
  feature_df = NULL,
  sample_df = NULL,
  feature_pk,
  sample_pk,
  omic_type_tag = "general"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_triple_omic_+3A_measurement_df">measurement_df</code></td>
<td>
<p>A data.frame (or tibble) of measurements - one row
for each combination of feature and sample</p>
</td></tr>
<tr><td><code id="create_triple_omic_+3A_feature_df">feature_df</code></td>
<td>
<p>A data.frame (or tibble) of features - one row per feature</p>
</td></tr>
<tr><td><code id="create_triple_omic_+3A_sample_df">sample_df</code></td>
<td>
<p>A data.frame (or tibble) of samples - one row per sample</p>
</td></tr>
<tr><td><code id="create_triple_omic_+3A_feature_pk">feature_pk</code></td>
<td>
<p>A unique identifier for features</p>
</td></tr>
<tr><td><code id="create_triple_omic_+3A_sample_pk">sample_pk</code></td>
<td>
<p>A unique identifier for samples</p>
</td></tr>
<tr><td><code id="create_triple_omic_+3A_omic_type_tag">omic_type_tag</code></td>
<td>
<p>an optional subtype of omic data: metabolomics,
lipidomics, proteomics, genomics, general</p>
</td></tr>
</table>


<h3>Details</h3>

<p>for now primary keys are unique (rather than allowing for a
multi-index)
</p>


<h3>Value</h3>

<p>An S3 <code>triple_omic</code>/<code>tomic</code> object built on a <code>list</code>:
</p>

<dl>
<dt>features</dt><dd><p>A tibble of feature meta-data (one row per feature)</p>
</dd>
<dt>samples</dt><dd><p>A tibble of sample meta-data (one row per sample)</p>
</dd>
<dt>measurements</dt><dd><p>A tibble with one row per measurement
(i.e., features x samples)</p>
</dd>
<dt>design</dt><dd><p>A list which organized the dataset's meta-data:
</p>

<dl>
<dt>feature_pk</dt><dd><p>variable specifying a unique feature</p>
</dd>
<dt>sample_pk</dt><dd><p>variable specifying a unique sample</p>
</dd>
<dt>features</dt><dd><p>tibble of feature attributes</p>
</dd>
<dt>samples</dt><dd><p>tibble of sample attributes</p>
</dd>
<dt>measurements</dt><dd><p>tibble of measurement attributes</p>
</dd>
</dl>

</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

measurement_df &lt;- tidyr::expand_grid(
  feature_id = 1:10,
  sample_id = LETTERS[1:5]
) %&gt;%
  dplyr::mutate(value = rnorm(n()))

feature_df &lt;- tibble(
  feature_id = 1:10,
  feature_group = rep(c("a", "b"), each = 5)
)
sample_df &lt;- tibble(
  sample_id = LETTERS[1:5],
  sample_group = c("a", "a", "b", "b", "b")
)

triple_omic &lt;- create_triple_omic(
  measurement_df, feature_df, sample_df,
  "feature_id", "sample_id"
)
</code></pre>

<hr>
<h2 id='downsample_heatmap'>Downsample Heatmap</h2><span id='topic+downsample_heatmap'></span>

<h3>Description</h3>

<p>Combine rows to speed up rendering of large heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downsample_heatmap(
  tidy_data,
  value_var,
  design,
  max_display_features = 1000,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downsample_heatmap_+3A_tidy_data">tidy_data</code></td>
<td>
<p>The data frame from a <code>tidy_omic</code> object containing
ordered feature and sample primary keys defined by ordered_featureId
and ordered_sampleId.</p>
</td></tr>
<tr><td><code id="downsample_heatmap_+3A_value_var">value_var</code></td>
<td>
<p>which variable in &quot;measurements&quot; to use for quantification.</p>
</td></tr>
<tr><td><code id="downsample_heatmap_+3A_design">design</code></td>
<td>
<p>a list summarizing the design of the tidy dataset</p>
</td></tr>
<tr><td><code id="downsample_heatmap_+3A_max_display_features">max_display_features</code></td>
<td>
<p>aggregate and downsample distinct feature to
this number to speed to up heatmap rendering.</p>
</td></tr>
<tr><td><code id="downsample_heatmap_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tidy_data with rows collapsed if the number of distinct features is
greater than <code>max_display_features</code>
</p>

<hr>
<h2 id='export_tomic_as_tidy'>Export T*Omic in Tidy Format</h2><span id='topic+export_tomic_as_tidy'></span>

<h3>Description</h3>

<p>Export a data table including all fields from features, samples and
measurements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_tomic_as_tidy(tomic, dir_path, name_preamble, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_tomic_as_tidy_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="export_tomic_as_tidy_+3A_dir_path">dir_path</code></td>
<td>
<p>path to save outputs</p>
</td></tr>
<tr><td><code id="export_tomic_as_tidy_+3A_name_preamble">name_preamble</code></td>
<td>
<p>start of output file name</p>
</td></tr>
<tr><td><code id="export_tomic_as_tidy_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Export one table which is one row per peak, which includes
all feature and sample attributes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  export_tomic_as_tidy(brauer_2008_triple, "/tmp", "brauer")
}
</code></pre>

<hr>
<h2 id='export_tomic_as_triple'>Export T*Omic as Triple</h2><span id='topic+export_tomic_as_triple'></span>

<h3>Description</h3>

<p>Export features, samples and measurements tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_tomic_as_triple(tomic, dir_path, name_preamble, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_tomic_as_triple_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="export_tomic_as_triple_+3A_dir_path">dir_path</code></td>
<td>
<p>path to save outputs</p>
</td></tr>
<tr><td><code id="export_tomic_as_triple_+3A_name_preamble">name_preamble</code></td>
<td>
<p>start of output file name</p>
</td></tr>
<tr><td><code id="export_tomic_as_triple_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Export three tables:
</p>

<ul>
<li><p>features: one row per features measured (i.e., a metabolite)
</p>
</li>
<li><p>sample: one row per sample
</p>
</li>
<li><p>measurements: one row per measurement (i.e., one metabolite in
one sample)
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  export_tomic_as_triple(brauer_2008_triple, "/tmp", "brauer")
}
</code></pre>

<hr>
<h2 id='export_tomic_as_wide'>Export T*Omic as Wide Data</h2><span id='topic+export_tomic_as_wide'></span>

<h3>Description</h3>

<p>abundances form a matrix with metabolites as rows and samples as columns.
Use transpose to treat samples as rows
filename
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_tomic_as_wide(
  tomic,
  dir_path,
  name_preamble,
  value_var = NULL,
  transpose = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_tomic_as_wide_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="export_tomic_as_wide_+3A_dir_path">dir_path</code></td>
<td>
<p>path to save outputs</p>
</td></tr>
<tr><td><code id="export_tomic_as_wide_+3A_name_preamble">name_preamble</code></td>
<td>
<p>start of output file name</p>
</td></tr>
<tr><td><code id="export_tomic_as_wide_+3A_value_var">value_var</code></td>
<td>
<p>measurement variable to use for the matrix</p>
</td></tr>
<tr><td><code id="export_tomic_as_wide_+3A_transpose">transpose</code></td>
<td>
<p>if TRUE then samples will be stored as rows</p>
</td></tr>
<tr><td><code id="export_tomic_as_wide_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Export one table which contains metabolites as rows and samples
as columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  export_tomic_as_wide(brauer_2008_triple, "/tmp", "brauer")
}
</code></pre>

<hr>
<h2 id='filter_tomic'>Filter T* Omics</h2><span id='topic+filter_tomic'></span>

<h3>Description</h3>

<p>Filter a tidy or triple omic to entries of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_tomic(
  tomic,
  filter_type,
  filter_table,
  filter_value,
  filter_variable = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_tomic_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="filter_tomic_+3A_filter_type">filter_type</code></td>
<td>

<dl>
<dt>category</dt><dd><p>filter filter_variable to categories specified in
filter_value</p>
</dd>
<dt>range</dt><dd><p>filter filter_variable to using the range (i.e., lower and
upper limit) provided in filter_value</p>
</dd>
<dt>apply</dt><dd><p>a quosure as a <code>filter_value</code> to a table of interest</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="filter_tomic_+3A_filter_table">filter_table</code></td>
<td>
<p>table where the filter should be applied</p>
</td></tr>
<tr><td><code id="filter_tomic_+3A_filter_value">filter_value</code></td>
<td>
<p>values to filter based on</p>
</td></tr>
<tr><td><code id="filter_tomic_+3A_filter_variable">filter_variable</code></td>
<td>
<p>variable to apply the filter to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object where a subset of features, samples or
measurmenets have been filtered.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filter_tomic(
  brauer_2008_triple,
  filter_type = "category",
  filter_table = "features",
  filter_variable = "BP",
  filter_value = c("biological process unknown", "vacuolar acidification")
)

filter_tomic(
  brauer_2008_triple,
  filter_type = "category",
  filter_table = "samples",
  filter_variable = "DR",
  filter_value = 0.05
)

filter_tomic(
  brauer_2008_tidy,
  filter_type = "range",
  filter_table = "samples",
  filter_variable = "DR",
  filter_value = c(0, 0.2)
)

filter_tomic(
  brauer_2008_triple,
  filter_type = "quo",
  filter_table = "features",
  filter_value = rlang::quo(BP == "biological process unknown")
)
</code></pre>

<hr>
<h2 id='filterInput'>Filter Input</h2><span id='topic+filterInput'></span>

<h3>Description</h3>

<p>UI components for the filter module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterInput(id, filter_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterInput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="filterInput_+3A_filter_table">filter_table</code></td>
<td>
<p>table to filter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='filterServer'>Filter Server</h2><span id='topic+filterServer'></span>

<h3>Description</h3>

<p>Server components for the filter module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterServer(id, tidy_omic, filter_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="filterServer_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
<tr><td><code id="filterServer_+3A_filter_table">filter_table</code></td>
<td>
<p>table to filter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tidy_omic</code> with some features and/or samples filtered.
</p>

<hr>
<h2 id='format_names_for_plotting'>Format Names for Plotting</h2><span id='topic+format_names_for_plotting'></span>

<h3>Description</h3>

<p>Wrap long names over multiple lines so that they will look better on plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_names_for_plotting(chars, width = 40, truncate_at = 80)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="format_names_for_plotting_+3A_chars">chars</code></td>
<td>
<p>a character vector (or a variable that can be converted to one)</p>
</td></tr>
<tr><td><code id="format_names_for_plotting_+3A_width">width</code></td>
<td>
<p>Positive integer giving target line width (in number of
characters). A width less than or equal to 1 will put each word on its
own line.</p>
</td></tr>
<tr><td><code id="format_names_for_plotting_+3A_truncate_at">truncate_at</code></td>
<td>
<p>max character length</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a reformatted character vector of the same length as the input.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chars &lt;- "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
  ac arcu semper erat porttitor egestas. Etiam sagittis, sapien at mattis."

format_names_for_plotting(chars)
</code></pre>

<hr>
<h2 id='get_design_tbl'>Get Design Table</h2><span id='topic+get_design_tbl'></span>

<h3>Description</h3>

<p>Get a tabular summary of all variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_design_tbl(tomic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_design_tbl_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble reflecting the <code>tomic</code> object's design.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_design_tbl(brauer_2008_triple)
</code></pre>

<hr>
<h2 id='get_tomic_table'>Get Tomic Table</h2><span id='topic+get_tomic_table'></span>

<h3>Description</h3>

<p>Extract one of the specific tables from a tomic object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tomic_table(tomic, table_type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tomic_table_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="get_tomic_table_+3A_table_type">table_type</code></td>
<td>
<p>The type of table to extract from the <code>tomic</code> object.
</p>

<dl>
<dt>tidy</dt><dd><p>one row per measurements with feature and sample attributes added. Equivalent to the $data field of a tidy omic object</p>
</dd>
<dt>measurements</dt><dd><p>one row per measurements defined a feature and sample foreign key. Equivalent to the $measurements field of a triple omic object</p>
</dd>
<dt>features</dt><dd><p>one row per feature defined by a feature primary key. Equivalent to the $features field of a triple omic object</p>
</dd>
<dt>samples</dt><dd><p>one row per sample defined by a sample primary key. Equivalent to the $samples field of a triple omic object</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble matching the <code>table_type</code> of the <code>tomic</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_tomic_table(brauer_2008_triple, "samples")
get_tomic_table(brauer_2008_tidy, "features")
</code></pre>

<hr>
<h2 id='ggBivOutput'>ggBivariate Output</h2><span id='topic+ggBivOutput'></span>

<h3>Description</h3>

<p>UI components for the ggBivariate module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggBivOutput(id, return_brushed_points = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggBivOutput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="ggBivOutput_+3A_return_brushed_points">return_brushed_points</code></td>
<td>
<p>Return values selected on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='ggBivServer'>ggBivariate Server</h2><span id='topic+ggBivServer'></span>

<h3>Description</h3>

<p>Server components for the ggBivariate module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggBivServer(id, tomic, plot_table, return_brushed_points = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggBivServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="ggBivServer_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="ggBivServer_+3A_plot_table">plot_table</code></td>
<td>
<p>table containing the data to be plotted</p>
</td></tr>
<tr><td><code id="ggBivServer_+3A_return_brushed_points">return_brushed_points</code></td>
<td>
<p>Return values selected on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tomic_table if return_brushed_points is TRUE, and 0 otherwise
</p>

<hr>
<h2 id='ggplotOutput'>ggplot Output</h2><span id='topic+ggplotOutput'></span>

<h3>Description</h3>

<p>UI components for the ggplot module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplotOutput(
  id,
  default_data_type = "samples",
  default_plot_type = "univariate"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplotOutput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="ggplotOutput_+3A_default_data_type">default_data_type</code></td>
<td>
<p>Default data type selection</p>
</td></tr>
<tr><td><code id="ggplotOutput_+3A_default_plot_type">default_plot_type</code></td>
<td>
<p>Default plot type selection</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='ggplotServer'>ggplot Server</h2><span id='topic+ggplotServer'></span>

<h3>Description</h3>

<p>Server components for the ggplot module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplotServer(id, tomic, return_brushed_points = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplotServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="ggplotServer_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="ggplotServer_+3A_return_brushed_points">return_brushed_points</code></td>
<td>
<p>Return values selected on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>tibble</code> of selected observations if
<code>return_brushed_points</code> is TRUE. Otherwise, returns NULL.
</p>

<hr>
<h2 id='ggUnivOutput'>ggUnivariate Output</h2><span id='topic+ggUnivOutput'></span>

<h3>Description</h3>

<p>UI components for the ggUnivariate module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggUnivOutput(id, return_brushed_points = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggUnivOutput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="ggUnivOutput_+3A_return_brushed_points">return_brushed_points</code></td>
<td>
<p>Return values selected on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='ggUnivServer'>ggUnivariate Server</h2><span id='topic+ggUnivServer'></span>

<h3>Description</h3>

<p>Server components for the ggUnivariate module
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggUnivServer(id, tomic, plot_table, return_brushed_points = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggUnivServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="ggUnivServer_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="ggUnivServer_+3A_plot_table">plot_table</code></td>
<td>
<p>table containing the data to be plotted</p>
</td></tr>
<tr><td><code id="ggUnivServer_+3A_return_brushed_points">return_brushed_points</code></td>
<td>
<p>Return values selected on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tomic_table if return_brushed_points is TRUE, and 0 otherwise.
</p>

<hr>
<h2 id='hclust_order'>Hierarchical clustering order</h2><span id='topic+hclust_order'></span>

<h3>Description</h3>

<p>Format and hierarchically cluster a data.frame. If hclust could not normally
be produced (usually because no samples are in common for a feature) pad
the matrix with zeros and still calculate the distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hclust_order(
  df,
  feature_pk,
  sample_pk,
  value_var,
  cluster_dim,
  distance_measure = "dist",
  hclust_method = "ward.D2"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hclust_order_+3A_df">df</code></td>
<td>
<p>data.frame to cluster</p>
</td></tr>
<tr><td><code id="hclust_order_+3A_feature_pk">feature_pk</code></td>
<td>
<p>variable uniquely defining a row</p>
</td></tr>
<tr><td><code id="hclust_order_+3A_sample_pk">sample_pk</code></td>
<td>
<p>variable uniquely defining a sample</p>
</td></tr>
<tr><td><code id="hclust_order_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
<tr><td><code id="hclust_order_+3A_cluster_dim">cluster_dim</code></td>
<td>
<p>rows, columns, or both</p>
</td></tr>
<tr><td><code id="hclust_order_+3A_distance_measure">distance_measure</code></td>
<td>
<p>variable to use for computing dis-similarity
</p>

<dl>
<dt>corr</dt><dd><p>pearson correlation</p>
</dd>
<dt>dist</dt><dd><p>euclidean distance</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="hclust_order_+3A_hclust_method">hclust_method</code></td>
<td>
<p>method from stats::hclust to use for clustering</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing a hierarchically clustered set of rows and/or
columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

df &lt;- tidyr::crossing(letters = LETTERS, numbers = 1:10) %&gt;%
  mutate(noise = rnorm(n()))
hclust_order(df, "letters", "numbers", "noise", "rows")
</code></pre>

<hr>
<h2 id='impute_missing_values'>Impute Missing Values</h2><span id='topic+impute_missing_values'></span>

<h3>Description</h3>

<p>Impute missing values using K-nearest neighbors imputation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute_missing_values(
  tomic,
  impute_var_name = "imputed",
  value_var = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="impute_missing_values_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="impute_missing_values_+3A_impute_var_name">impute_var_name</code></td>
<td>
<p>variable to create for imputed measurements</p>
</td></tr>
<tr><td><code id="impute_missing_values_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
<tr><td><code id="impute_missing_values_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to <a href="impute.html#topic+impute.knn">impute.knn</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object with imputed measurements.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>impute_missing_values(brauer_2008_triple)

</code></pre>

<hr>
<h2 id='infer_tomic_table_type'>Infer Tomic Table Type</h2><span id='topic+infer_tomic_table_type'></span>

<h3>Description</h3>

<p>From a tomic_table, choose whether it reflects features, samples or
measurements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_tomic_table_type(tomic, tomic_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="infer_tomic_table_type_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="infer_tomic_table_type_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>features, samples or measurements
</p>

<hr>
<h2 id='lassoInput'>Lasso Input</h2><span id='topic+lassoInput'></span>

<h3>Description</h3>

<p>UI components for the lasso module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lassoInput(id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lassoInput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='lassoServer'>Lasso Server</h2><span id='topic+lassoServer'></span>

<h3>Description</h3>

<p>Take a subset of entries from a tomic table (generally selected using the
lasso function) and then either filter a tomic object to these entries
or tag the entries of interest with a user-specified variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lassoServer(id, tomic, tomic_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lassoServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="lassoServer_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="lassoServer_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object amended based on the lasso selection.
</p>

<hr>
<h2 id='organizeInput'>Organize Input</h2><span id='topic+organizeInput'></span>

<h3>Description</h3>

<p>UI components for the organize input module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>organizeInput(id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="organizeInput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='organizeServer'>Organize Servers</h2><span id='topic+organizeServer'></span>

<h3>Description</h3>

<p>Server components for the organize input module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>organizeServer(id, tidy_omic, feature_vars, sample_vars, value_var)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="organizeServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="organizeServer_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
<tr><td><code id="organizeServer_+3A_feature_vars">feature_vars</code></td>
<td>
<p>variables available for arranging features</p>
</td></tr>
<tr><td><code id="organizeServer_+3A_sample_vars">sample_vars</code></td>
<td>
<p>variables available for arrange samples</p>
</td></tr>
<tr><td><code id="organizeServer_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> with sorted features and/or samples.
</p>

<hr>
<h2 id='plot_bivariate'>Bivariate Plot</h2><span id='topic+plot_bivariate'></span>

<h3>Description</h3>

<p>Create a scatter or boxplot from a tomic dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_bivariate(
  tomic_table,
  x_var,
  y_var,
  color_var = NULL,
  shape_var = NULL,
  alpha_var = NULL,
  size_var = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_bivariate_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
<tr><td><code id="plot_bivariate_+3A_x_var">x_var</code></td>
<td>
<p>x-axis variable</p>
</td></tr>
<tr><td><code id="plot_bivariate_+3A_y_var">y_var</code></td>
<td>
<p>y-axis variable</p>
</td></tr>
<tr><td><code id="plot_bivariate_+3A_color_var">color_var</code></td>
<td>
<p>coloring variable (NULL to suppress coloring)</p>
</td></tr>
<tr><td><code id="plot_bivariate_+3A_shape_var">shape_var</code></td>
<td>
<p>shape variable (NULL to suppress shape)</p>
</td></tr>
<tr><td><code id="plot_bivariate_+3A_alpha_var">alpha_var</code></td>
<td>
<p>alpha variable or numeric for constant alpha (NULL to suppress alpha)</p>
</td></tr>
<tr><td><code id="plot_bivariate_+3A_size_var">size_var</code></td>
<td>
<p>size variable or integer/numeric for constant size (NULL to suppress size)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 grob
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)

brauer_augmented &lt;- brauer_2008_tidy %&gt;%
  add_pcs(npcs = 5) %&gt;%
  tomic_to("triple_omic")

tomic_table &lt;- brauer_augmented$samples
plot_bivariate(tomic_table, "PC1", "PC2", "nutrient", "nutrient", 0.5, 10)
plot_bivariate(tomic_table, "PC1", "PC2", NULL)
plot_bivariate(tomic_table, "nutrient", "PC2", "nutrient")
</code></pre>

<hr>
<h2 id='plot_heatmap'>Plot Heatmap</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>Generate a heatmap visualization of a features x samples matrix of
measurements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  tomic,
  feature_var = NULL,
  sample_var = NULL,
  value_var = NULL,
  cluster_dim = "both",
  distance_measure = "dist",
  hclust_method = "ward.D2",
  change_threshold = Inf,
  plot_type = "grob",
  max_display_features = 800,
  x_label = NULL,
  y_label = NULL,
  colorbar_label = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_feature_var">feature_var</code></td>
<td>
<p>variable from &quot;features&quot; to use as a unique feature
label.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_sample_var">sample_var</code></td>
<td>
<p>variable from &quot;samples&quot; to use as a unique sample label.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_value_var">value_var</code></td>
<td>
<p>which variable in &quot;measurements&quot; to use for quantification.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_cluster_dim">cluster_dim</code></td>
<td>
<p>rows, columns, or both</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_distance_measure">distance_measure</code></td>
<td>
<p>variable to use for computing dis-similarity
</p>

<dl>
<dt>corr</dt><dd><p>pearson correlation</p>
</dd>
<dt>dist</dt><dd><p>euclidean distance</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_hclust_method">hclust_method</code></td>
<td>
<p>method from stats::hclust to use for clustering</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_change_threshold">change_threshold</code></td>
<td>
<p>values with a more extreme absolute change will be
thresholded to this value.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_plot_type">plot_type</code></td>
<td>
<p>plotly (for interactivity) or grob (for a static ggplot)</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_max_display_features">max_display_features</code></td>
<td>
<p>aggregate and downsample distinct feature to
this number to speed to up heatmap rendering.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_x_label">x_label</code></td>
<td>
<p>label for x-axis (if NULL then use <code>feature_var</code>)</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_y_label">y_label</code></td>
<td>
<p>label for y-axis (if NULL then use <code>sample_var</code>)</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_colorbar_label">colorbar_label</code></td>
<td>
<p>label for color-bar; default is log2 abundance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 grob
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

tomic &lt;- brauer_2008_triple %&gt;%
  filter_tomic(
    filter_type = "category",
    filter_table = "features",
    filter_variable = "BP",
    filter_value = c(
      "protein biosynthesis",
      "rRNA processing", "response to stress"
    )
  )

plot_heatmap(
  tomic = tomic,
  value_var = "expression",
  change_threshold = 5,
  cluster_dim = "rows",
  plot_type = "grob",
  distance_measure = "corr"
)
</code></pre>

<hr>
<h2 id='plot_univariate'>Univariate Plot</h2><span id='topic+plot_univariate'></span>

<h3>Description</h3>

<p>Create a histogram from a tomic dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_univariate(tomic_table, x_var, color_var = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_univariate_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
<tr><td><code id="plot_univariate_+3A_x_var">x_var</code></td>
<td>
<p>x-axis variable</p>
</td></tr>
<tr><td><code id="plot_univariate_+3A_color_var">color_var</code></td>
<td>
<p>coloring variable (NULL to suppress coloring)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 grob
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)

brauer_augmented &lt;- brauer_2008_tidy %&gt;%
  add_pcs(npcs = 5) %&gt;%
  tomic_to("triple_omic")

plot_univariate(brauer_augmented$samples, "PC1", "nutrient")
plot_univariate(brauer_augmented$measurements, "expression", NULL)
</code></pre>

<hr>
<h2 id='plotsaverInput'>Plot Saver Input</h2><span id='topic+plotsaverInput'></span>

<h3>Description</h3>

<p>UI components for the plot saver module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotsaverInput(id, ui_format = "tall")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotsaverInput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="plotsaverInput_+3A_ui_format">ui_format</code></td>
<td>
<p>Set UI appearance
</p>

<dl>
<dt>tall</dt><dd><p>stack all UI elements</p>
</dd>
<dt>wide</dt><dd><p>UI elements are side-by-side</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>shiny</code> UI
</p>

<hr>
<h2 id='plotsaverServer'>Plot Saver Server</h2><span id='topic+plotsaverServer'></span>

<h3>Description</h3>

<p>Server components for the plot saver module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotsaverServer(id, grob, filename = "grob.png")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotsaverServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="plotsaverServer_+3A_grob">grob</code></td>
<td>
<p>a ggplot2 plot</p>
</td></tr>
<tr><td><code id="plotsaverServer_+3A_filename">filename</code></td>
<td>
<p>filename for saving plot. The extension will be respected
by <a href="ggplot2.html#topic+ggsave">ggsave</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='prepare_example_datasets'>Prepare Example Datasets</h2><span id='topic+prepare_example_datasets'></span>

<h3>Description</h3>

<p>Format example datasets and add them to the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_example_datasets(seed = 1234)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepare_example_datasets_+3A_seed">seed</code></td>
<td>
<p>a seed value used to reproducibly sample random genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None; used for side-effects.
</p>

<hr>
<h2 id='reconcile_triple_omic'>Reconcile Triple Omic</h2><span id='topic+reconcile_triple_omic'></span>

<h3>Description</h3>

<p>If some samples, feature or measurements have been dropped; update other
tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconcile_triple_omic(triple_omic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reconcile_triple_omic_+3A_triple_omic">triple_omic</code></td>
<td>
<p>an object of class triple_omic produced by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a triple_omic object
</p>

<hr>
<h2 id='remove_missing_values'>Remove Missing Values</h2><span id='topic+remove_missing_values'></span>

<h3>Description</h3>

<p>Account for missing values by dropping features, samples or using
imputation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_missing_values(
  tomic,
  value_var = NULL,
  missing_val_method = "drop_samples",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_missing_values_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="remove_missing_values_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
<tr><td><code id="remove_missing_values_+3A_missing_val_method">missing_val_method</code></td>
<td>
<p>Approach to remove missing values:
</p>

<dl>
<dt>drop_features</dt><dd><p>Drop features with missing values</p>
</dd>
<dt>drop_samples</dt><dd><p>Drop samples which are missing all features,
then drop features</p>
</dd>
<dt>impute</dt><dd><p>Impute missing values</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="remove_missing_values_+3A_verbose">verbose</code></td>
<td>
<p>extra reporting messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object where missing values have been accounted
for.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>remove_missing_values(brauer_2008_triple)

</code></pre>

<hr>
<h2 id='shiny_filter_test'>Shiny Filter Test</h2><span id='topic+shiny_filter_test'></span>

<h3>Description</h3>

<p>Tests the shiny filter module as a stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_filter_test(tidy_omic, filter_table = "features")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_filter_test_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
<tr><td><code id="shiny_filter_test_+3A_filter_table">filter_table</code></td>
<td>
<p>table to filter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  shiny_filter_test(brauer_2008_tidy)
}
</code></pre>

<hr>
<h2 id='shiny_ggbiv_test'>Shiny ggBivariate Test</h2><span id='topic+shiny_ggbiv_test'></span>

<h3>Description</h3>

<p>Test the shiny ggBivariate module as a stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_ggbiv_test(tomic, plot_table = "samples")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_ggbiv_test_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="shiny_ggbiv_test_+3A_plot_table">plot_table</code></td>
<td>
<p>table containing the data to be plotted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  shiny_ggbiv_test(
    add_pcs(brauer_2008_triple, npcs = 5),
    plot_table = "samples"
  )
  shiny_ggbiv_test(
    brauer_2008_triple,
    plot_table = "measurements"
  )
}
</code></pre>

<hr>
<h2 id='shiny_ggplot_test'>Shiny ggplot Test</h2><span id='topic+shiny_ggplot_test'></span>

<h3>Description</h3>

<p>Test the shiny ggplot module as a stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_ggplot_test(tomic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_ggplot_test_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  shiny_ggplot_test(add_pcs(brauer_2008_triple, npcs = 5))
  shiny_ggplot_test(brauer_2008_triple)
}
</code></pre>

<hr>
<h2 id='shiny_gguniv_test'>Shiny ggUnivariate Test</h2><span id='topic+shiny_gguniv_test'></span>

<h3>Description</h3>

<p>Test the shiny ggUnivariate module as a stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_gguniv_test(tomic, plot_table = "samples")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_gguniv_test_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="shiny_gguniv_test_+3A_plot_table">plot_table</code></td>
<td>
<p>table containing the data to be plotted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  shiny_gguniv_test(
    add_pcs(brauer_2008_triple, npcs = 5),
    plot_table = "samples"
  )
  shiny_gguniv_test(brauer_2008_triple, plot_table = "measurements")
  shiny_gguniv_test(brauer_2008_triple, plot_table = "features")
}
</code></pre>

<hr>
<h2 id='shiny_lasso_test'>Shiny Lasso Test</h2><span id='topic+shiny_lasso_test'></span>

<h3>Description</h3>

<p>Tests the shiny lasso module as a stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_lasso_test(tomic, tomic_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_lasso_test_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="shiny_lasso_test_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  tomic &lt;- brauer_2008_triple
  tomic_table &lt;- tomic[["samples"]] %&gt;% dplyr::filter(nutrient == "G")
  shiny_lasso_test(tomic, tomic_table)
}
</code></pre>

<hr>
<h2 id='shiny_lasso_test_reactval'>Shiny Lasso Test w/ Reactive Values</h2><span id='topic+shiny_lasso_test_reactval'></span>

<h3>Description</h3>

<p>Tests the shiny lasso module as a stand-alone application when the
<code>tomic</code> is a <code>reativeVal</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_lasso_test_reactval(tomic, tomic_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_lasso_test_reactval_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="shiny_lasso_test_reactval_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  tomic &lt;- brauer_2008_triple
  tomic_table &lt;- tomic[["samples"]] %&gt;% dplyr::filter(nutrient == "G")
  shiny_lasso_test_reactval(tomic, tomic_table)

  tomic_table &lt;- tomic[["measurements"]] %&gt;% dplyr::filter(expression &lt; -3)
  shiny_lasso_test_reactval(tomic, tomic_table)
}
</code></pre>

<hr>
<h2 id='shiny_organize_test'>Shiny Organize Test</h2><span id='topic+shiny_organize_test'></span>

<h3>Description</h3>

<p>Tests the shiny organization module as stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_organize_test(tidy_omic, feature_vars, sample_vars, value_var)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_organize_test_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
<tr><td><code id="shiny_organize_test_+3A_feature_vars">feature_vars</code></td>
<td>
<p>variables available for arranging features</p>
</td></tr>
<tr><td><code id="shiny_organize_test_+3A_sample_vars">sample_vars</code></td>
<td>
<p>variables available for arrange samples</p>
</td></tr>
<tr><td><code id="shiny_organize_test_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  shiny_organize_test(
    brauer_2008_tidy,
    feature_vars = c("BP", "MF"),
    sample_vars = c("sample", "nutrient", "DR"),
    value_var = "expression"
  )
}
</code></pre>

<hr>
<h2 id='shiny_plotsaver_test'>Shiny Plot Saver Test</h2><span id='topic+shiny_plotsaver_test'></span>

<h3>Description</h3>

<p>Test the shiny plotsaver module as a stand-alone application.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_plotsaver_test()
</code></pre>


<h3>Value</h3>

<p>a <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  shiny_plotsaver_test()
}
</code></pre>

<hr>
<h2 id='shiny_sort_test'>Shiny Sort Test</h2><span id='topic+shiny_sort_test'></span>

<h3>Description</h3>

<p>Test the shiny sorting module as a stand-alone app.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny_sort_test(triple_omic, valid_sort_vars, value_var)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny_sort_test_+3A_triple_omic">triple_omic</code></td>
<td>
<p>an object of class triple_omic produced by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code></p>
</td></tr>
<tr><td><code id="shiny_sort_test_+3A_valid_sort_vars">valid_sort_vars</code></td>
<td>
<p>variables available for categorical arranging</p>
</td></tr>
<tr><td><code id="shiny_sort_test_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>shiny</code> app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  shiny_sort_test(brauer_2008_triple,
    valid_sort_vars = c("sample", "nutrient", "DR"),
    value_var = "expression"
  )
}
</code></pre>

<hr>
<h2 id='sort_tomic'>Sort Triple Omic</h2><span id='topic+sort_tomic'></span>

<h3>Description</h3>

<p>Sort a dataset's features or samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_tomic(
  tomic,
  sort_type,
  sort_table,
  sort_variables = NULL,
  value_var = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sort_tomic_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="sort_tomic_+3A_sort_type">sort_type</code></td>
<td>

<dl>
<dt>hclust</dt><dd><p>Arrange samples by hierarchical clustering of a provided
<code>value_var</code></p>
</dd>
<dt>arrange</dt><dd><p>Arrange samples by the factor or alphanumeric ordering of
a set of <code>sort_variables</code></p>
</dd>
</dl>
</td></tr>
<tr><td><code id="sort_tomic_+3A_sort_table">sort_table</code></td>
<td>
<p>samples or features</p>
</td></tr>
<tr><td><code id="sort_tomic_+3A_sort_variables">sort_variables</code></td>
<td>
<p>A set of attributes in sort_table to sort with in
<code>arrange</code>.</p>
</td></tr>
<tr><td><code id="sort_tomic_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>sort_tomic</code> supports the reordering of features or samples using
either hierarchical clustering or based on the levels of other variables.
Sorting occurs by turning either the feature or sample primary key
into a factor whose levels reflect the sort.
</p>


<h3>Value</h3>

<p>A <code>tomic</code> object where feature or sample primary keys have
been turned into a factor reflecting how they are sorted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

sort_tomic(brauer_2008_triple,
  sort_type = "arrange", sort_table = "samples",
  sort_variables = c("nutrient", "DR")
) %&gt;%
  sort_tomic(
    sort_type = "hclust",
    sort_table = "features",
    value_var = "expression"
  )
</code></pre>

<hr>
<h2 id='sort_triple_arrange'>Sort Triple Arrange</h2><span id='topic+sort_triple_arrange'></span>

<h3>Description</h3>

<p>Sort a <code>triple_omic</code> object based on the values of one or more
variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_triple_arrange(triple_omic, sort_table, sort_variables)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sort_triple_arrange_+3A_triple_omic">triple_omic</code></td>
<td>
<p>an object of class triple_omic produced by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code></p>
</td></tr>
<tr><td><code id="sort_triple_arrange_+3A_sort_table">sort_table</code></td>
<td>
<p>samples or features</p>
</td></tr>
<tr><td><code id="sort_triple_arrange_+3A_sort_variables">sort_variables</code></td>
<td>
<p>A set of attributes in sort_table to sort with in
<code>arrange</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>triple_omic</code> with sorted features or samples.
</p>

<hr>
<h2 id='sort_triple_hclust'>Sort Triple Hclust</h2><span id='topic+sort_triple_hclust'></span>

<h3>Description</h3>

<p>Sort a <code>triple_omic</code> object using hierarchical clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_triple_hclust(triple_omic, sort_table, value_var)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sort_triple_hclust_+3A_triple_omic">triple_omic</code></td>
<td>
<p>an object of class triple_omic produced by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code></p>
</td></tr>
<tr><td><code id="sort_triple_hclust_+3A_sort_table">sort_table</code></td>
<td>
<p>samples or features</p>
</td></tr>
<tr><td><code id="sort_triple_hclust_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>triple_omic</code> with clustered features or samples.
</p>

<hr>
<h2 id='sortInput'>Sort Input</h2><span id='topic+sortInput'></span>

<h3>Description</h3>

<p>UI components for the sort module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortInput(id, sort_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sortInput_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="sortInput_+3A_sort_table">sort_table</code></td>
<td>
<p>table to sort</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>shiny</code> UI
</p>

<hr>
<h2 id='sortServer'>Sort Server</h2><span id='topic+sortServer'></span>

<h3>Description</h3>

<p>Server components for the sort module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortServer(id, tomic, sort_table, valid_sort_vars = NULL, value_var = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sortServer_+3A_id">id</code></td>
<td>
<p>An ID string that corresponds with the ID used to call the module's
UI function.</p>
</td></tr>
<tr><td><code id="sortServer_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="sortServer_+3A_sort_table">sort_table</code></td>
<td>
<p>samples or features</p>
</td></tr>
<tr><td><code id="sortServer_+3A_valid_sort_vars">valid_sort_vars</code></td>
<td>
<p>variables available for categorical arranging</p>
</td></tr>
<tr><td><code id="sortServer_+3A_value_var">value_var</code></td>
<td>
<p>An abundance value to use with <code>hclust</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sorted <code>tomic</code> object.
</p>

<hr>
<h2 id='tidy_to_triple'>Tidy omic to triple omic</h2><span id='topic+tidy_to_triple'></span>

<h3>Description</h3>

<p>Convert a <code>tidy_omic</code> object into a <code>triple_omic</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tidy_to_triple(tidy_omic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tidy_to_triple_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>data</code> table will be converted into <code>features</code>,
<code>samples</code>, and <code>measurements</code> tables using the <code>design</code>
to determine which variables belong in each table. The <code>design</code>
will be preserved as-is.
</p>


<h3>Value</h3>

<p>A <code>triple_omic</code> object as created by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tidy_to_triple(brauer_2008_tidy)
</code></pre>

<hr>
<h2 id='tomic_sort_status'>T* Omic Sort Status</h2><span id='topic+tomic_sort_status'></span>

<h3>Description</h3>

<p>Determine whether features &amp;/or samples have been sorted and stored as
ordered_featureId and ordered_sampleId.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tomic_sort_status(tomic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tomic_sort_status_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>length 1 character string indicating whether the <code>tomic</code>
is sorted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
tomic_sort_status(brauer_2008_tidy)
</code></pre>

<hr>
<h2 id='tomic_to'>T* Omic To</h2><span id='topic+tomic_to'></span>

<h3>Description</h3>

<p>Takes in any <code>romic</code> reprsentation of a dataset and returns a specific
representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tomic_to(tomic, to_class)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tomic_to_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="tomic_to_+3A_to_class">to_class</code></td>
<td>
<p>The class to return, either <code>tidy_omic</code> or
<code>triple_omic</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>tomic transformed to <code>to_class</code> class (or un-transformed if
it started that way).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
tomic_to(brauer_2008_tidy, "triple_omic")
</code></pre>

<hr>
<h2 id='triple_to_tidy'>Triple Omic to Tidy Omic</h2><span id='topic+triple_to_tidy'></span>

<h3>Description</h3>

<p>Convert a <code>triple_omic</code> object into a <code>tidy_omic</code> oobject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>triple_to_tidy(triple_omic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="triple_to_tidy_+3A_triple_omic">triple_omic</code></td>
<td>
<p>an object of class triple_omic produced by
<code><a href="#topic+create_triple_omic">create_triple_omic</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Features, samples and measurements will be merged into a single <code>data</code>
table, and the <code>design</code> will be preserved as-is.
</p>


<h3>Value</h3>

<p>A <code>tidy_omic</code> object as created by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

measurement_df &lt;- tidyr::expand_grid(
  feature_id = 1:10,
  sample_id = LETTERS[1:5]
) %&gt;%
  dplyr::mutate(value = rnorm(n()))

feature_df &lt;- tibble(
  feature_id = 1:10,
  feature_group = rep(c("a", "b"), each = 5)
)
sample_df &lt;- tibble(
  sample_id = LETTERS[1:5],
  sample_group = c("a", "a", "b", "b", "b")
)

triple_omic &lt;- create_triple_omic(
  measurement_df, feature_df, sample_df,
  "feature_id", "sample_id"
)
triple_to_tidy(triple_omic)
</code></pre>

<hr>
<h2 id='try_brushedPoints'>Try brushedPoints</h2><span id='topic+try_brushedPoints'></span>

<h3>Description</h3>

<p>This function wraps brushedPoints in a try statement to catch cases where
the brushing is out-of-sync with the df that is selected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>try_brushedPoints(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="try_brushedPoints_+3A_...">...</code></td>
<td>
<p>args to pass to <a href="shiny.html#topic+brushedPoints">brushedPoints</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a df of brushed points
</p>

<hr>
<h2 id='update_tidy_omic'>Update Tidy Omic</h2><span id='topic+update_tidy_omic'></span>

<h3>Description</h3>

<p>Update a Tidy 'Omics data and schema to reflect newly added fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_tidy_omic(tidy_omic, updated_tidy_data, new_variable_tables = c())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="update_tidy_omic_+3A_tidy_omic">tidy_omic</code></td>
<td>
<p>an object of class tidy_omic produced by
<code><a href="#topic+create_tidy_omic">create_tidy_omic</a></code></p>
</td></tr>
<tr><td><code id="update_tidy_omic_+3A_updated_tidy_data">updated_tidy_data</code></td>
<td>
<p>a tibble of data to use to update <code>tidy_omic</code>.</p>
</td></tr>
<tr><td><code id="update_tidy_omic_+3A_new_variable_tables">new_variable_tables</code></td>
<td>
<p>a named character vector of newly added variables
in <code>updated_tidy_data</code> (names) and the table features, samples,
measurements they apply to (values).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>tidy_omic</code> object with an updated schema and/or data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

tidy_omic &lt;- brauer_2008_tidy
updated_tidy_data &lt;- tidy_omic$data %&gt;%
  mutate(new_sample_var = "foo") %&gt;%
  select(-DR)
new_variable_tables &lt;- c("new_sample_var" = "samples")
</code></pre>

<hr>
<h2 id='update_tomic'>Update T* Omic</h2><span id='topic+update_tomic'></span>

<h3>Description</h3>

<p>Provide an updated features, samples or measurements table to a
<code>tomic</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_tomic(tomic, tomic_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="update_tomic_+3A_tomic">tomic</code></td>
<td>
<p>Either a <code>tidy_omic</code> or <code>triple_omic</code> object</p>
</td></tr>
<tr><td><code id="update_tomic_+3A_tomic_table">tomic_table</code></td>
<td>
<p>A table taken from a tidy (i.e., augmented measurements)
or triple omic dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>tomic</code> object with updated features, samples or measurements.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)
updated_features &lt;- brauer_2008_triple$features %&gt;%
  dplyr::filter(BP == "biological process unknown") %&gt;%
  dplyr::mutate(chromosome = purrr::map_int(systematic_name, function(x) {
    which(LETTERS == stringr::str_match(x, "Y([A-Z])")[2])
  }))

update_tomic(brauer_2008_triple, updated_features)
</code></pre>

<hr>
<h2 id='var_partial_match'>Var Partial Match</h2><span id='topic+var_partial_match'></span>

<h3>Description</h3>

<p>Partial string matching of a provided variable to the variables available
in a table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>var_partial_match(x, df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="var_partial_match_+3A_x">x</code></td>
<td>
<p>a variable name or regex match to a variable name</p>
</td></tr>
<tr><td><code id="var_partial_match_+3A_df">df</code></td>
<td>
<p>a data.frame or tibble</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single variable from df
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
