<!DOCTYPE html><html><head><title>Help for package ROptimus</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ROptimus}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ex.m.fun'><p>Tutorial 5 m() function</p></a></li>
<li><a href='#ex.r.fun'><p>Tutorial 5 r() function</p></a></li>
<li><a href='#ex.u.fun'><p>Tutorial 5 u() function</p></a></li>
<li><a href='#IJ_ORIG'><p>Tutorial 5 Genomic Contact Data</p></a></li>
<li><a href='#Optimus'><p>Acceptance Ratio Simulated Annealing and Acceptance Ratio Replica Exchange Monte Carlo Optimisation Engine</p></a></li>
<li><a href='#OptimusExamples'><p>Generate script for reproducing a tutorial</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Parallel General-Purpose Adaptive Optimisation Engine</td>
</tr>
<tr>
<td>Version:</td>
<td>3.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A general-purpose optimisation engine that supports 
	i) Monte Carlo optimisation with Metropolis criterion [Metropolis et al. (1953) &lt;<a href="https://doi.org/10.1063%2F1.1699114">doi:10.1063/1.1699114</a>&gt;, Hastings (1970) &lt;<a href="https://doi.org/10.1093%2Fbiomet%2F57.1.97">doi:10.1093/biomet/57.1.97</a>&gt;]
    and Acceptance Ratio Simulated Annealing [Kirkpatrick et al. (1983) &lt;<a href="https://doi.org/10.1126%2Fscience.220.4598.671">doi:10.1126/science.220.4598.671</a>&gt;, Černý (1985) &lt;<a href="https://doi.org/10.1007%2FBF00940812">doi:10.1007/BF00940812</a>&gt;] 
	on multiple cores, and ii) Acceptance Ratio Replica Exchange Monte 
	Carlo Optimisation. In each case, the system pseudo-temperature is 
	dynamically adjusted such that the observed acceptance ratio is kept 
	near to the desired (fixed or changing) acceptance ratio.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>doParallel (&ge; 1.0.11), foreach (&ge; 1.4.4), iterators (&ge;
1.0.9), parallel (&ge; 3.4.2)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SahakyanLab/ROptimus">https://github.com/SahakyanLab/ROptimus</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-30 17:17:25 UTC; ltamon</td>
</tr>
<tr>
<td>Author:</td>
<td>Nicholas Andre G. Johnson [aut],
  Liezel Tamon [aut],
  Xin Liu [aut],
  Aleksandr B. Sahakyan [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aleksandr B. Sahakyan &lt;sahakyanlab@cantab.net&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-10 13:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='ex.m.fun'>Tutorial 5 m() function</h2><span id='topic+ex.m.fun'></span>

<h3>Description</h3>

<p>The m() function supplied to Optimus() for reproducing Tutorial 5. This is
for the small executable example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ex.m.fun
</code></pre>


<h3>Format</h3>

<p>Function
</p>

<hr>
<h2 id='ex.r.fun'>Tutorial 5 r() function</h2><span id='topic+ex.r.fun'></span>

<h3>Description</h3>

<p>The r() function supplied to Optimus() for reproducing Tutorial 5. This is
for the small executable example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ex.r.fun
</code></pre>


<h3>Format</h3>

<p>Function
</p>

<hr>
<h2 id='ex.u.fun'>Tutorial 5 u() function</h2><span id='topic+ex.u.fun'></span>

<h3>Description</h3>

<p>The u() function supplied to Optimus() for reproducing Tutorial 5. This is
for the small executable example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ex.u.fun
</code></pre>


<h3>Format</h3>

<p>Function
</p>

<hr>
<h2 id='IJ_ORIG'>Tutorial 5 Genomic Contact Data</h2><span id='topic+IJ_ORIG'></span>

<h3>Description</h3>

<p>A dataset containing 734 genomic contact pairs for reproducing Tutorial 5.
Here, we take a set of 734 pairs of i and j 40-kilobase DNA regions that are
known to be in contact inside a cell. Binning the DNA (e.g., a single chromosome)
into 40-kb regions, each region is represented as a single integer that is equal
to its end position divided by the length of the region, which is 40 kb. For
instance, the 1st region, with start and end positions at the 1st and 40000th
nucleotides, respectively, is denoted as 1 (40000th base / 40000 bases = 1).
This simplifies the notation for a contact between two regions to a pair of
positive integers as in used in this dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IJ_ORIG
</code></pre>


<h3>Format</h3>

<p>A data frame with 734 rows and 2 variables:
</p>

<dl>
<dt>i</dt><dd><p>Integer representing 40-kb contact region partner of j</p>
</dd>
<dt>j</dt><dd><p>Integer representing 40-kb contact region partner of i</p>
</dd>
</dl>


<hr>
<h2 id='Optimus'>Acceptance Ratio Simulated Annealing and Acceptance Ratio Replica Exchange Monte Carlo Optimisation Engine</h2><span id='topic+Optimus'></span>

<h3>Description</h3>

<p>Acceptance Ratio Simulated Annealing and Acceptance Ratio Replica Exchange Monte Carlo Optimisation Engine
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Optimus(
  NUMITER = 1e+06,
  STATWINDOW = 70,
  T.INI = 1e-05,
  T.ADJSTEP = 5e-09,
  TSCLnum = 2,
  T.SCALING = 3,
  T.MIN = 5e-09,
  T.DELTA = 2,
  DUMP.FREQ = 10000,
  LIVEPLOT = TRUE,
  LIVEPLOT.FREQ = 1e+05,
  PDFheight = 29,
  PDFwidth = 20,
  NCPU = 4,
  LONG = TRUE,
  SEED = 840,
  OPTNAME = "",
  DATA = NULL,
  K.INITIAL = 0,
  rDEF,
  mDEF,
  uDEF,
  EXCHANGE.FREQ = 1000,
  ACCRATIO = c(90, 50, 5, 1),
  CYCLES = 10,
  ACCRATIO.IN = 90,
  ACCRATIO.FIN = 0.5,
  OPT.TYPE = "SA",
  DIR,
  starcore = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Optimus_+3A_numiter">NUMITER</code></td>
<td>
<p>Number of model optimisation steps.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_statwindow">STATWINDOW</code></td>
<td>
<p>Number of last ongoing iterations to calculate
acceptance ratio for temperature auto-adjustment.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_t.ini">T.INI</code></td>
<td>
<p>Initial temperature (K) for Metropolis criterion.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_t.adjstep">T.ADJSTEP</code></td>
<td>
<p>Temperature change step-size for temperature
auto-adjustment based on the actual acceptance ratio.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_tsclnum">TSCLnum</code></td>
<td>
<p>Cutoff for one of the NumofAccRatSMIdeal and
NumofAccRatGRIdeal numbers after which the adjustment
step is multiplied by T.SCALING.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_t.scaling">T.SCALING</code></td>
<td>
<p>See above.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_t.min">T.MIN</code></td>
<td>
<p>Value to which the pseudo-temperature is set when the</p>
</td></tr>
<tr><td><code id="Optimus_+3A_t.delta">T.DELTA</code></td>
<td>
<p>Minimum value by which acceptance ratio in a STATWINDOW
must differ from the ideal acceptance ratio for the
temperature control unit to make a temperature
adjustment.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_dump.freq">DUMP.FREQ</code></td>
<td>
<p>The frequency (in steps) of writing the found ongoing
best model.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_liveplot">LIVEPLOT</code></td>
<td>
<p>Plotting the optimisation process in a pdf file.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_liveplot.freq">LIVEPLOT.FREQ</code></td>
<td>
<p>Frequency (in steps) of plotting the results.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_pdfheight">PDFheight</code></td>
<td>
<p>Plot height in inches.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_pdfwidth">PDFwidth</code></td>
<td>
<p>Plot width in inches.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_ncpu">NCPU</code></td>
<td>
<p>Number of CPU cores to use, by running more replicas
of the optimisation. The usage of more than 1 cores
will attempt to load the foreach and doParallel
libraries in the case of SA Optimus</p>
</td></tr>
<tr><td><code id="Optimus_+3A_long">LONG</code></td>
<td>
<p>If TRUE, it means that a long simulation is expected to
be done, hence the memory-friendly mode will be
activated.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_seed">SEED</code></td>
<td>
<p>Setting the seed for the random number generator.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_optname">OPTNAME</code></td>
<td>
<p>The name of the optimisation process.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_data">DATA</code></td>
<td>
<p>A list that holds any supplementary data that
functions mDEF or uDEF need to access.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_k.initial">K.INITIAL</code></td>
<td>
<p>The initial parameter configuration from which the
optimisation process will begin.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_rdef">rDEF</code></td>
<td>
<p>Function that defines a rule by which the
parameters(K) are randomly altered.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_mdef">mDEF</code></td>
<td>
<p>Model function that operates on the parameters to be
optimized (K) and returns an observable object O.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_udef">uDEF</code></td>
<td>
<p>Function that evaluates the performance of a given set
of parameters K.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_exchange.freq">EXCHANGE.FREQ</code></td>
<td>
<p>Frequency of exchanges (NUMITER should be divisible by
this number, for RE Optimus).</p>
</td></tr>
<tr><td><code id="Optimus_+3A_accratio">ACCRATIO</code></td>
<td>
<p>Vector of Acceptance Ratios for each replica (length of
ACCRATIO must be equal to NCPU, for RE Optimus).</p>
</td></tr>
<tr><td><code id="Optimus_+3A_cycles">CYCLES</code></td>
<td>
<p>Number of annealing cycles (NUMITER should be divisible
by this number, for SA Optimus).</p>
</td></tr>
<tr><td><code id="Optimus_+3A_accratio.in">ACCRATIO.IN</code></td>
<td>
<p>Initial acceptance ratio (%) at the beginning of each
constituent annealing cycles (for SA Optimus).</p>
</td></tr>
<tr><td><code id="Optimus_+3A_accratio.fin">ACCRATIO.FIN</code></td>
<td>
<p>Final acceptance ratio (%) at the end of each
constituent annealing cycles (for SA Optimus).</p>
</td></tr>
<tr><td><code id="Optimus_+3A_opt.type">OPT.TYPE</code></td>
<td>
<p>String specifying which optimisation protocol to use.
Enter &quot;SA&quot; for Simulated Annealing or &quot;RE&quot; for Replica
Exchange (default value is &quot;SA&quot;)</p>
</td></tr>
<tr><td><code id="Optimus_+3A_dir">DIR</code></td>
<td>
<p>String specifying which optimisation protocol to use.</p>
</td></tr>
<tr><td><code id="Optimus_+3A_starcore">starcore</code></td>
<td>
<p>Experimental variable of type list, holding some
parameters for in-lab starcore use only.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A probabilistic optimal parameter configuration K.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
K &lt;- IJ_ORIG
K$j &lt;- sample(x=K$j, size=nrow(K), replace=FALSE)
out.dir &lt;- tempdir()

Optimus(NCPU=1, OPTNAME="IJ.NEW.OPTI.SA", NUMITER=500, CYCLES=2, DIR=out.dir,
        DUMP.FREQ=10, LONG=FALSE, OPT.TYPE="SA", K.INITIAL=K,
        rDEF=ex.r.fun, mDEF=ex.m.fun, uDEF=ex.u.fun,
        DATA=list(IJ_ORIG=IJ_ORIG, gaplimit=50, numContacts=nrow(IJ_ORIG)))
</code></pre>

<hr>
<h2 id='OptimusExamples'>Generate script for reproducing a tutorial</h2><span id='topic+OptimusExamples'></span>

<h3>Description</h3>

<p>Generate script for reproducing a tutorial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OptimusExamples(
  dir,
  example = 1,
  method = "SA",
  file_name = "example.R",
  mopac = NULL,
  vignette = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OptimusExamples_+3A_dir">dir</code></td>
<td>
<p>String specifying directory where to save the script
and vignette.</p>
</td></tr>
<tr><td><code id="OptimusExamples_+3A_example">example</code></td>
<td>
<p>Integer specifying tutorial or example number based on
manual (default value is 1).</p>
</td></tr>
<tr><td><code id="OptimusExamples_+3A_method">method</code></td>
<td>
<p>String specifying which optimisation protocol to use.
Enter &quot;SA&quot; for Simulated Annealing or &quot;RE&quot; for Replica
Exchange (default value is &quot;SA&quot;).</p>
</td></tr>
<tr><td><code id="OptimusExamples_+3A_file_name">file_name</code></td>
<td>
<p>String specifying file name of script (default value is
example.R).</p>
</td></tr>
<tr><td><code id="OptimusExamples_+3A_mopac">mopac</code></td>
<td>
<p>String specifying the path of the MOPAC (Molecular Orbital
PACkage) executable for tutorial 3 (Geometry Optimisation
of Vitamin C Molecule) (default value is NULL).</p>
</td></tr>
<tr><td><code id="OptimusExamples_+3A_vignette">vignette</code></td>
<td>
<p>If TRUE, add tutorial vignette to directory (default
value is TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A script (and optionally a vignette) for reproducing a tutorial.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
out.dir &lt;- tempdir()
OptimusExamples(dir=out.dir, example=1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
