<!DOCTYPE html><html lang="en"><head><title>Help for package rotl</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rotl}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#get_study'><p>Get all the trees associated with a particular study</p></a></li>
<li><a href='#get_study_subtree'><p>Study Subtree</p></a></li>
<li><a href='#get_study_tree'><p>Study Tree</p></a></li>
<li><a href='#get_tree_ids'><p>Study Metadata</p></a></li>
<li><a href='#inspect.match_names'><p>Inspect and Update alternative matches for a name returned</p>
by tnrs_match_names</a></li>
<li><a href='#is_in_tree'><p>Check that OTT ids occur in the Synthetic Tree</p></a></li>
<li><a href='#list_trees'><p>List trees ids in objects returned by</p>
<code>studies_find_studies</code> and
<code>studies_find_trees</code>.</a></li>
<li><a href='#ott_id.match_names'><p><code>ott_id</code> and <code>flags</code> for taxonomic names matched</p>
by <code>tnrs_match_names</code></a></li>
<li><a href='#rotl'><p>An Interface to the Open Tree of Life API</p></a></li>
<li><a href='#source_list'><p>List of studies used in the Tree of Life</p></a></li>
<li><a href='#strip_ott_ids'><p>Strip OTT ids from tip labels</p></a></li>
<li><a href='#studies_find_studies'><p>Find a Study</p></a></li>
<li><a href='#studies_find_trees'><p>Find Trees</p></a></li>
<li><a href='#studies_properties'><p>Properties of the Studies</p></a></li>
<li><a href='#study_external_IDs'><p>Get external identifiers for data associated with an Open Tree study</p></a></li>
<li><a href='#synonyms.match_names'><p>List the synonyms for a given name</p></a></li>
<li><a href='#tax_lineage'><p>Lineage of a taxon</p></a></li>
<li><a href='#tax_rank'><p>Methods for Taxonomy</p></a></li>
<li><a href='#taxon_external_IDs'><p>Get external identifiers for data associated with an Open Tree taxon</p></a></li>
<li><a href='#taxonomy_about'><p>Information about the Open Tree Taxonomy</p></a></li>
<li><a href='#taxonomy_mrca'><p>Taxonomic MRCA</p></a></li>
<li><a href='#taxonomy_subtree'><p>Taxonomy subtree</p></a></li>
<li><a href='#taxonomy_taxon_info'><p>Taxon information</p></a></li>
<li><a href='#tnrs_contexts'><p>TNRS contexts</p></a></li>
<li><a href='#tnrs_infer_context'><p>Infer the taxonomic context from a list of names</p></a></li>
<li><a href='#tnrs_match_names'><p>Match names to the Open Tree Taxonomy</p></a></li>
<li><a href='#tol_about'><p>Information about the Tree of Life</p></a></li>
<li><a href='#tol_induced_subtree'><p>Subtree from the Open Tree of Life</p></a></li>
<li><a href='#tol_lineage'><p>Node info</p></a></li>
<li><a href='#tol_mrca'><p>MRCA of taxa from the synthetic tree</p></a></li>
<li><a href='#tol_subtree'><p>Extract a subtree from the synthetic tree</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Interface to the 'Open Tree of Life' API</td>
</tr>
<tr>
<td>Version:</td>
<td>3.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>An interface to the 'Open Tree of Life' API to retrieve
    phylogenetic trees, information about studies used to assemble the
    synthetic tree, and utilities to match taxonomic names to 'Open Tree
    identifiers'. The 'Open Tree of Life' aims at assembling a
    comprehensive phylogenetic tree for all named species.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-2-Clause">BSD_2_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/rotl/">https://docs.ropensci.org/rotl/</a>, <a href="https://github.com/ropensci/rotl">https://github.com/ropensci/rotl</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/rotl/issues">https://github.com/ropensci/rotl/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, curl (&ge; 3.0.0), httr, jsonlite, rentrez, rlang, rncl (&ge;
0.6.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr (&ge; 1.12), MCMCglmm, phylobase, readxl, rmarkdown (&ge;
0.7), RNeXML, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>X-schema.org-isPartOf:</td>
<td>https://ropensci.org</td>
</tr>
<tr>
<td>X-schema.org-keywords:</td>
<td>metadata, ropensci, phylogenetics,
independant-contrasts, biodiversity</td>
</tr>
<tr>
<td>X-schema.org-relatedLink:</td>
<td>https://codemeta.github.io/codemetar</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-15 09:34:25 UTC; francois</td>
</tr>
<tr>
<td>Author:</td>
<td>Francois Michonneau
    <a href="https://orcid.org/0000-0002-9092-966X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Joseph Brown <a href="https://orcid.org/0000-0002-3835-8062"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  David Winter <a href="https://orcid.org/0000-0002-6165-0029"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Scott Chamberlain <a href="https://orcid.org/0000-0003-1444-9135"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [rev]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francois Michonneau &lt;francois.michonneau@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-15 13:50:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='get_study'>Get all the trees associated with a particular study</h2><span id='topic+get_study'></span>

<h3>Description</h3>

<p>Returns the trees associated with a given study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study(
  study_id = NULL,
  object_format = c("phylo", "nexml"),
  file_format,
  file,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_study_+3A_study_id">study_id</code></td>
<td>
<p>the study ID for the study of interest (character)</p>
</td></tr>
<tr><td><code id="get_study_+3A_object_format">object_format</code></td>
<td>
<p>the class of the object the query should
return (either <code>phylo</code> or <code>nexml</code>). Ignored if
<code>file_format</code> is specified.</p>
</td></tr>
<tr><td><code id="get_study_+3A_file_format">file_format</code></td>
<td>
<p>the format of the file to be generated
(<code>newick</code>, <code>nexus</code>, <code>nexml</code> or <code>json</code>).</p>
</td></tr>
<tr><td><code id="get_study_+3A_file">file</code></td>
<td>
<p>the file name where the output of the function will be
saved.</p>
</td></tr>
<tr><td><code id="get_study_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>file_format</code> is missing, the function returns an object
of the class <code>phylo</code> from the <code>ape</code> package
(default), or an object of the class <code>nexml</code> from the
<code>RNeXML</code> package.
</p>
<p>Otherwise <code>file_format</code> can be either <code>newick</code>,
<code>nexus</code>, <code>nexml</code> or <code>json</code>, and the function will
generate a file of the selected format. In this case, a file name
needs to be provided using the argument <code>file</code>. If a file
with the same name already exists, it will be silently
overwritten.
</p>


<h3>Value</h3>

<p>if <code>file_format</code> is missing, an object of class
<code>phylo</code> or <code>nexml</code>, otherwise a logical indicating
whether the file was successfully created.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_study_meta">get_study_meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
that_one_study &lt;- get_study(study_id="pg_719", object_format="phylo")
if (require(RNeXML)) { ## if RNeXML is installed get the object directly
   nexml_study &lt;- get_study(study_id="pg_719", object_format="nexml")
} else { ## otherwise write it to a file
   get_study(study_id="pg_719", file_format="nexml", file=tempfile(fileext=".nexml"))
}

## End(Not run)
</code></pre>

<hr>
<h2 id='get_study_subtree'>Study Subtree</h2><span id='topic+get_study_subtree'></span>

<h3>Description</h3>

<p>Retrieve subtree from a specific tree in the Open Tree of Life data store
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_subtree(
  study_id,
  tree_id,
  subtree_id,
  object_format = c("phylo"),
  tip_label = c("original_label", "ott_id", "ott_taxon_name"),
  file_format,
  file,
  deduplicate = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_study_subtree_+3A_study_id">study_id</code></td>
<td>
<p>the study identifier (character)</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_tree_id">tree_id</code></td>
<td>
<p>the tree identifier (character)</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_subtree_id">subtree_id</code></td>
<td>
<p>either a node id that specifies a subtree or
&ldquo;ingroup&rdquo; which returns the ingroup for this subtree.</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_object_format">object_format</code></td>
<td>
<p>the class of the object returned by the
function (default, and currently only possibility <code>phylo</code>
from the <code>ape</code> package)</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_tip_label">tip_label</code></td>
<td>
<p>the format of the tip
labels. &ldquo;<code>original_label</code>&rdquo; (default) returns the
original labels as provided in the study,
&ldquo;<code>ott_id</code>&rdquo; labels are replaced by their ott IDs,
&ldquo;<code>ott_taxon_name</code>&rdquo; labels are replaced by their
Open Tree Taxonomy taxon name.</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_file_format">file_format</code></td>
<td>
<p>character, the file format to use to save the
results of the query (possible values, &lsquo;newick&rsquo; or
&lsquo;nexus&rsquo;).</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_file">file</code></td>
<td>
<p>character, the path and file name where the output
should be written.</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_deduplicate">deduplicate</code></td>
<td>
<p>logical (default <code>TRUE</code>). If the tree
returned by the study contains duplicated taxon names, should
they be made unique? It is normally illegal for NEXUS/Newick
tree strings to contain duplicated tip names. This is a
workaround to circumvent this requirement. If <code>TRUE</code>,
duplicated tip labels will be appended <code>_1</code>, <code>_2</code>,
etc.</p>
</td></tr>
<tr><td><code id="get_study_subtree_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
small_tr &lt;- get_study_subtree(study_id="pg_1144", tree_id="tree5800", subtree_id="node991044")
ingroup  &lt;- get_study_subtree(study_id="pg_1144", tree_id="tree5800", subtree_id="ingroup")
nexus_file &lt;- tempfile(fileext=".nex")
get_study_subtree(study_id="pg_1144", tree_id="tree5800", subtree_id="ingroup", file=nexus_file,
                  file_format="nexus")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_study_tree'>Study Tree</h2><span id='topic+get_study_tree'></span>

<h3>Description</h3>

<p>Returns a specific tree from within a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_tree(
  study_id = NULL,
  tree_id = NULL,
  object_format = c("phylo"),
  tip_label = c("original_label", "ott_id", "ott_taxon_name"),
  file_format,
  file,
  deduplicate = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_study_tree_+3A_study_id">study_id</code></td>
<td>
<p>the identifier of a study (character)</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_tree_id">tree_id</code></td>
<td>
<p>the identifier of a tree within the study</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_object_format">object_format</code></td>
<td>
<p>the class of the object to be returned
(default and currently only possible value <code>phylo</code> from
the <code>ape</code> package).</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_tip_label">tip_label</code></td>
<td>
<p>the format of the tip
labels. &ldquo;<code>original_label</code>&rdquo; (default) returns the
original labels as provided in the study,
&ldquo;<code>ott_id</code>&rdquo; labels are replaced by their ott IDs,
&ldquo;<code>ott_taxon_name</code>&rdquo; labels are replaced by their
Open Tree Taxonomy taxon name.</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_file_format">file_format</code></td>
<td>
<p>the format of the file to be generated
(<code>newick</code> default, <code>nexus</code>, or <code>json</code>).</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_file">file</code></td>
<td>
<p>the file name where the output of the function will be
saved.</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_deduplicate">deduplicate</code></td>
<td>
<p>logical (default <code>TRUE</code>). If the tree
returned by the study contains duplicated taxon names, should they
be made unique? It is normally illegal for NEXUS/Newick tree
strings to contain duplicated tip names. This is a workaround to
circumvent this requirement. If <code>TRUE</code>, duplicated tip labels
will be appended <code>_1</code>, <code>_2</code>, etc.</p>
</td></tr>
<tr><td><code id="get_study_tree_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if <code>file_format</code> is missing, an object of class
<code>phylo</code>, otherwise a logical indicating whether the file
was successfully created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 tree &lt;- get_study_tree(study_id="pg_1144", tree_id="tree2324")

 ## comparison of the first few tip labels depending on the options used
 head(get_study_tree(study_id="pg_1144", tree_id="tree2324", tip_label="original_label")$tip.label)
 head(get_study_tree(study_id="pg_1144", tree_id="tree2324", tip_label="ott_id")$tip.label)
 head(get_study_tree(study_id="pg_1144", tree_id="tree2324", tip_label="ott_taxon_name")$tip.label)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_tree_ids'>Study Metadata</h2><span id='topic+get_tree_ids'></span><span id='topic+get_publication'></span><span id='topic+candidate_for_synth'></span><span id='topic+get_study_year'></span><span id='topic+get_tree_ids.study_meta'></span><span id='topic+get_publication.study_meta'></span><span id='topic+candidate_for_synth.study_meta'></span><span id='topic+get_study_year.study_meta'></span><span id='topic+get_study_meta'></span>

<h3>Description</h3>

<p>Retrieve metadata about a study in the Open Tree of Life datastore.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_ids(sm)

get_publication(sm)

candidate_for_synth(sm)

get_study_year(sm)

## S3 method for class 'study_meta'
get_tree_ids(sm)

## S3 method for class 'study_meta'
get_publication(sm)

## S3 method for class 'study_meta'
candidate_for_synth(sm)

## S3 method for class 'study_meta'
get_study_year(sm)

get_study_meta(study_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tree_ids_+3A_sm">sm</code></td>
<td>
<p>an object created by <code>get_study_meta</code></p>
</td></tr>
<tr><td><code id="get_tree_ids_+3A_study_id">study_id</code></td>
<td>
<p>the study identifier (character)</p>
</td></tr>
<tr><td><code id="get_tree_ids_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>get_study_meta</code> returns a long list of attributes for the
studies that are contributing to the synthetic tree. To help with
the extraction of relevant information from this list, several
helper functions exists: </p>

<ul>
<li> <p>get_tree_ids  The identifiers of the trees
associated with the study 
</p>
</li>
<li> <p>get_publication  The citation information of the
publication for the study. The DOI (or URL) for the study is
available as an attribute to the returned object (i.e.,
<code>attr(object, "DOI")</code> ) .
</p>
</li>
<li> <p>candidate_for_synth  The identifier of the tree(s) from
the study used in the synthetic tree. This is a subset of the
result of <code>get_tree_ids</code>.
</p>
 <p>get_study_year  The year of publication of the study. 
</p>

</li></ul>



<h3>Value</h3>

<p>named-list containing the metadata associated with the
study requested
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
req &lt;- get_study_meta("pg_719")
get_tree_ids(req)
candidate_for_synth(req)
get_publication(req)
get_study_year(req)

## End(Not run)
</code></pre>

<hr>
<h2 id='inspect.match_names'>Inspect and Update alternative matches for a name returned
by tnrs_match_names</h2><span id='topic+inspect.match_names'></span><span id='topic+inspect'></span><span id='topic+update.match_names'></span>

<h3>Description</h3>

<p>Taxonomic names may have different meanings in different taxonomic
contexts, as the same genus name can be applied to animals and
plants for instance. Additionally, the meaning of a taxonomic name
may have change throughout its history, and may have referred to a
different taxon in the past. In such cases, a given names might
have multiple matches in the Open Tree Taxonomy. These functions
allow users to inspect (and update) alternative meaning of a given
name and its current taxonomic status according to the Open Tree
Taxonomy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'match_names'
inspect(response, row_number, taxon_name, ott_id, ...)

inspect(response, ...)

## S3 method for class 'match_names'
update(object, row_number, taxon_name, ott_id, new_row_number, new_ott_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inspect.match_names_+3A_response">response</code></td>
<td>
<p>an object generated by the
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> function</p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_row_number">row_number</code></td>
<td>
<p>the row number corresponding to the name to
inspect</p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_taxon_name">taxon_name</code></td>
<td>
<p>the taxon name corresponding to the name to
inspect</p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_ott_id">ott_id</code></td>
<td>
<p>the ott id corresponding to the name to inspect</p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_...">...</code></td>
<td>
<p>currently ignored</p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_object">object</code></td>
<td>
<p>an object created by <code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code></p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_new_row_number">new_row_number</code></td>
<td>
<p>the row number in the output of
<code><a href="#topic+inspect">inspect</a></code> to replace the taxa specified by
<code>row_number</code>, <code>taxon_name</code>, or <code>ott_id</code>.</p>
</td></tr>
<tr><td><code id="inspect.match_names_+3A_new_ott_id">new_ott_id</code></td>
<td>
<p>the ott id of the taxon to replace the taxa
specified by <code>row_number</code>, <code>taxon_name</code>, or
<code>ott_id</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To inspect alternative taxonomic meanings of a given name, you
need to provide the object resulting from a call to the
tnrs_match_names function, as well as one of either the row number
corresponding to the name in this object, the name itself (as used
in the original query), or the ott_id listed for this name.
</p>
<p>To update one of the name, you also need to provide the row number
in which the name to be replaced appear or its ott id.
</p>


<h3>Value</h3>

<p>a data frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
   matched_names &lt;- tnrs_match_names(c("holothuria", "diadema", "boletus"))
   inspect(matched_names, taxon_name="diadema")
   new_matched_names &lt;- update(matched_names, taxon_name="diadema",
                               new_ott_id = 631176)
   new_matched_names
   
## End(Not run)
</code></pre>

<hr>
<h2 id='is_in_tree'>Check that OTT ids occur in the Synthetic Tree</h2><span id='topic+is_in_tree'></span>

<h3>Description</h3>

<p>Some valid taxonomic names do not occur in the Synthetic
Tree. This convenience function allows you to check whether a
given Open Tree Taxonomy identifier (OTT id) is in the tree. A taxonomic
name may not occur in the synthetic tree because (1) it is an
extinct or invalid taxon, or (2) it is part of a group that is not
monophyletic in the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_in_tree(ott_ids, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_in_tree_+3A_ott_ids">ott_ids</code></td>
<td>
<p>a vector of Open Tree Taxonomy identifiers</p>
</td></tr>
<tr><td><code id="is_in_tree_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named logical vector. <code>TRUE</code> indicates that the OTT
id is in the synthetic tree, and <code>FALSE</code> that it is not.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  plant_families &lt;- c("Asteraceae", "Solanaceae", "Poaceae", "Amaranthaceae",
                      "Zamiaceae", "Araceae", "Juncaceae")
  matched_names &lt;- tnrs_match_names(plant_families)
  ## This fails because some ott ids are not in the tree
  ## plant_tree &lt;- tol_induced_subtree(ott_id(matched_names))
  ## So let's check which ones are actually in the tree first:
  in_tree &lt;- is_in_tree(ott_id(matched_names))
  ## This now works:
  plant_tree &lt;- tol_induced_subtree(ott_id(matched_names)[in_tree])

## End(Not run)

</code></pre>

<hr>
<h2 id='list_trees'>List trees ids in objects returned by
<code><a href="#topic+studies_find_studies">studies_find_studies</a></code> and
<code><a href="#topic+studies_find_trees">studies_find_trees</a></code>.</h2><span id='topic+list_trees'></span><span id='topic+list_trees.matched_studies'></span>

<h3>Description</h3>

<p><code>list_trees</code> returns all trees associated with a particular
study when used on an object returned by
<code><a href="#topic+studies_find_studies">studies_find_studies</a></code>, but only the trees that match
the search criteria when used on objects returned by
<code><a href="#topic+studies_find_trees">studies_find_trees</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_trees(matched_studies, ...)

## S3 method for class 'matched_studies'
list_trees(matched_studies, study_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="list_trees_+3A_matched_studies">matched_studies</code></td>
<td>
<p>an object created by
<code>studies_find_trees</code> or <code>studies_find_studies</code>.</p>
</td></tr>
<tr><td><code id="list_trees_+3A_...">...</code></td>
<td>
<p>Currently unused</p>
</td></tr>
<tr><td><code id="list_trees_+3A_study_id">study_id</code></td>
<td>
<p>a <code>study_id</code> listed in the object returned by
<code>studies_find_trees</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>list_trees</code> returns a list of the tree_ids for each
study that match the requested criteria. If a <code>study_id</code>
is provided, then only the trees for this study are returned
as a vector.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+studies_find_studies">studies_find_studies</a></code> and
<code><a href="#topic+studies_find_trees">studies_find_trees</a></code>. The help for these functions
have examples demonstrating the use of <code>list_trees</code>.
</p>

<hr>
<h2 id='ott_id.match_names'><code>ott_id</code> and <code>flags</code> for taxonomic names matched
by <code>tnrs_match_names</code></h2><span id='topic+ott_id.match_names'></span><span id='topic+flags.match_names'></span><span id='topic+flags'></span>

<h3>Description</h3>

<p><code>rotl</code> provides a collection of functions that allows users
to extract relevant information from an object generated by
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'match_names'
ott_id(tax, row_number, taxon_name, ott_id, ...)

## S3 method for class 'match_names'
flags(tax, row_number, taxon_name, ott_id, ...)

flags(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ott_id.match_names_+3A_tax">tax</code></td>
<td>
<p>an object returned by <code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code></p>
</td></tr>
<tr><td><code id="ott_id.match_names_+3A_row_number">row_number</code></td>
<td>
<p>the row number corresponding to the name for
which to list the synonyms</p>
</td></tr>
<tr><td><code id="ott_id.match_names_+3A_taxon_name">taxon_name</code></td>
<td>
<p>the taxon name corresponding to the name for
which to list the synonyms</p>
</td></tr>
<tr><td><code id="ott_id.match_names_+3A_ott_id">ott_id</code></td>
<td>
<p>the ott id corresponding to the name for which to
list the synonyms</p>
</td></tr>
<tr><td><code id="ott_id.match_names_+3A_...">...</code></td>
<td>
<p>currently ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These methods optionally accept one of the arguments
<code>row_number</code>, <code>taxon_name</code> or <code>ott_id</code> to retrieve
the corresponding information for one of the matches in the object
returned by the <code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> function.
</p>
<p>If these arguments are not provided, these methods can return
information for the matches currently listed in the object
returned by <code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code>.
</p>


<h3>Value</h3>

<p>A list of the ott ids or flags for the taxonomic names
matched with <code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code>, for either one or
all the names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  rsp &lt;- tnrs_match_names(c("Diadema", "Tyrannosaurus"))
  rsp$ott_id    # ott id for match currently in use
  ott_id(rsp)   # similar as above but elements are named

  ## flags() is useful for instance to determine if a taxon is extinct
  flags(rsp, taxon_name="Tyrannosaurus")

## End(Not run)
</code></pre>

<hr>
<h2 id='rotl'>An Interface to the Open Tree of Life API</h2><span id='topic+rotl'></span>

<h3>Description</h3>

<p>The Open Tree of Life is an NSF funded project that is generating
an online, comprehensive phylogenetic tree for 1.8 million
species. <code>rotl</code> provides an interface that allows you to
query and retrieve the parts of the tree of life that is of
interest to you.
</p>


<h3>Details</h3>

<p><code>rotl</code> provides function to most of the end points the API
provides. The documentation of the API is available at:
<a href="https://github.com/OpenTreeOfLife/opentree/wiki/Open-Tree-of-Life-APIs">https://github.com/OpenTreeOfLife/opentree/wiki/Open-Tree-of-Life-APIs</a>
</p>


<h3>Customizing API calls</h3>

<p>All functions that use API end points can take 2 arguments to
customize the API call and are passed as <code>...</code> arguments.
</p>

<ul>
<li> <p><code>otl_v</code>   This argument controls which version
of the API your call is using. The default value for this
argument is a call to the non-exported function
<code>otl_version()</code> which returns the current version of the
Open Tree of Life APIs (v2).
</p>
</li>
<li> <p><code>dev_url</code>   This argument controls whether to use
the development version of the API. By default, <code>dev_url</code>
is set to <code>FALSE</code>, using <code>dev_url = TRUE</code> in your
function calls will use the development version.
</p>
</li></ul>

<p>For example, to use the development version of the API, you
could use: <code>tnrs_match_names("anas", dev_url=TRUE)</code>
</p>
<p>Additional arguments can also be passed to the
<code><a href="httr.html#topic+GET">GET</a></code> and <code><a href="httr.html#topic+POST">POST</a></code> methods.
</p>


<h3>Acknowledgments</h3>

<p>This package was started during the Open Tree of Life Hackathon
organized by the OpenTree of Life, the NESCent Hackathon
Interoperability Phylogenetic group, and Arbor.
</p>

<hr>
<h2 id='source_list'>List of studies used in the Tree of Life</h2><span id='topic+source_list'></span><span id='topic+source_list.tol_summary'></span>

<h3>Description</h3>

<p>Retrieve the detailed information for the list of studies used in
the Tree of Life.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>source_list(tax, ...)

## S3 method for class 'tol_summary'
source_list(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="source_list_+3A_tax">tax</code></td>
<td>
<p>a list containing a <code>source_id_map</code> slot.</p>
</td></tr>
<tr><td><code id="source_list_+3A_...">...</code></td>
<td>
<p>additional arguments (currently unused)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes the object resulting from
<code>tol_about(study_list = TRUE)</code>, <code>tol_mrca()</code>,
<code>tol_node_info()</code>, and returns a data frame listing the
<code>tree_id</code>, <code>study_id</code> and <code>git_sha</code> for the
studies currently included in the Tree of Life.
</p>


<h3>Value</h3>

<p>a data frame
</p>

<hr>
<h2 id='strip_ott_ids'>Strip OTT ids from tip labels</h2><span id='topic+strip_ott_ids'></span>

<h3>Description</h3>

<p>Strip OTT ids from tip labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strip_ott_ids(tip_labels, remove_underscores = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strip_ott_ids_+3A_tip_labels">tip_labels</code></td>
<td>
<p>a character vector containing tip labels (most
likely the <code>tip.label</code> element from a tree returned by
<code><a href="#topic+tol_induced_subtree">tol_induced_subtree</a></code></p>
</td></tr>
<tr><td><code id="strip_ott_ids_+3A_remove_underscores">remove_underscores</code></td>
<td>
<p>logical (defaults to FALSE). If set to
TRUE underscores in tip labels are converted to spaces</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the contents of
<code>tip_labels</code> with any OTT ids removed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
genera &lt;- c("Perdix", "Setophaga", "Cinclus", "Struthio")
tr &lt;- tol_induced_subtree(ott_ids=c(102710, 285198, 267845, 292466))
tr$tip.label %in% genera
tr$tip.label &lt;- strip_ott_ids(tr$tip.label)
tr$tip.label %in% genera

## End(Not run)
</code></pre>

<hr>
<h2 id='studies_find_studies'>Find a Study</h2><span id='topic+studies_find_studies'></span>

<h3>Description</h3>

<p>Return the identifiers of studies that match given properties
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies_find_studies(
  property = NULL,
  value = NULL,
  verbose = FALSE,
  exact = FALSE,
  detailed = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="studies_find_studies_+3A_property">property</code></td>
<td>
<p>The property to be searched on (character)</p>
</td></tr>
<tr><td><code id="studies_find_studies_+3A_value">value</code></td>
<td>
<p>The property value to be searched on (character)</p>
</td></tr>
<tr><td><code id="studies_find_studies_+3A_verbose">verbose</code></td>
<td>
<p>Should the output include all metadata (logical
default <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="studies_find_studies_+3A_exact">exact</code></td>
<td>
<p>Should exact matching be used? (logical, default
<code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="studies_find_studies_+3A_detailed">detailed</code></td>
<td>
<p>If <code>TRUE</code> (default), the function will return
a data frame that summarizes information about the study (see
&lsquo;Value&rsquo;). Otherwise, it only returns the study
identifiers.</p>
</td></tr>
<tr><td><code id="studies_find_studies_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>detailed=TRUE</code>, the function returns a data frame
listing the study id (<code>study_ids</code>), the number of trees
associated with this study (<code>n_trees</code>), the tree ids (at
most 5) associated with the studies (<code>tree_ids</code>), the
tree id that is a candidate for the synthetic tree if any
(<code>candidate</code>), the year of publication of the study
(<code>study_year</code>), the title of the publication for the
study (<code>title</code>), and the DOI (Digital Object Identifier)
for the study (<code>study_doi</code>).
</p>
<p>If <code>detailed=FALSE</code>, the function returns a data frame
with a single column containing the study identifiers.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+studies_properties">studies_properties</a></code> which lists properties
against which the studies can be
searched. <code><a href="#topic+list_trees">list_trees</a></code> that returns a list for all
tree ids associated with a study.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## To match a study for which the identifier is already known
one_study &lt;- studies_find_studies(property="ot:studyId", value="pg_719")
list_trees(one_study)

## To find studies pertaining to Mammals
mammals &lt;- studies_find_studies(property="ot:focalCladeOTTTaxonName",
                                value="mammalia")
## To extract the tree identifiers for each of the studies
list_trees(mammals)
## ... or for a given study
list_trees(mammals, "ot_308")

## Just the identifiers without other information about the studies
mammals &lt;- studies_find_studies(property="ot:focalCladeOTTTaxonName",
                                value="mammalia", detailed=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='studies_find_trees'>Find Trees</h2><span id='topic+studies_find_trees'></span>

<h3>Description</h3>

<p>Return a list of studies for which trees match a given set of
properties
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies_find_trees(
  property = NULL,
  value = NULL,
  verbose = FALSE,
  exact = FALSE,
  detailed = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="studies_find_trees_+3A_property">property</code></td>
<td>
<p>The property to be searched on (character)</p>
</td></tr>
<tr><td><code id="studies_find_trees_+3A_value">value</code></td>
<td>
<p>The property-value to be searched on (character)</p>
</td></tr>
<tr><td><code id="studies_find_trees_+3A_verbose">verbose</code></td>
<td>
<p>Should the output include all metadata? (logical,
default <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="studies_find_trees_+3A_exact">exact</code></td>
<td>
<p>Should exact matching be used for the value?
(logical, default <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="studies_find_trees_+3A_detailed">detailed</code></td>
<td>
<p>Should a detailed report be provided? If
<code>TRUE</code> (default), the output will include metadata about
the study that include trees matching the property. Otherwise,
only information about the trees will be provided.</p>
</td></tr>
<tr><td><code id="studies_find_trees_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The list of possible values to be used as values for the argument
<code>property</code> can be found using the function
<code><a href="#topic+studies_properties">studies_properties</a></code>.
</p>


<h3>Value</h3>

<p>A data frame that summarizes the trees found (and their
associated studies) for the requested criteria. If a study has
more than 5 trees, the <code>tree_ids</code> of the first ones will
be shown, followed by <code>...</code> to indicate that more are
present.
</p>
<p>If <code>detailed=FALSE</code>, the data frame will include the
study ids of the study (<code>study_ids</code>), the number of trees
in this study that match the search criteria
(<code>n_matched_trees</code>), the tree ids that match the search
criteria (<code>match_tree_ids</code>).
</p>
<p>If <code>detailed=TRUE</code>, in addition of the fields listed
above, the data frame will also contain the total number of
trees associated with the study (<code>n_trees</code>), all the tree
ids associated with the study (<code>tree_ids</code>), the tree id
that is a potential candidate for inclusion in the synthetic
tree (if any) (<code>candidate</code>), the year the study was
published (<code>study_year</code>), the title of the study
(<code>title</code>), the DOI for the study (<code>study_doi</code>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+studies_properties">studies_properties</a></code> which lists properties
the studies can be searched on. <code><a href="#topic+list_trees">list_trees</a></code> for
listing the trees that match the query.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- studies_find_trees(property="ot:ottTaxonName", value="Drosophila",
                          detailed=FALSE)
## summary of the trees and associated studies that match this criterion
res
## With metadata about the studies (default)
res &lt;- studies_find_trees(property="ot:ottTaxonName", value="Drosophila",
                          detailed=TRUE)
## The list of trees for each study that match the search criteria
list_trees(res)
## the trees for a given study
list_trees(res, study_id = "pg_2769")

## End(Not run)
</code></pre>

<hr>
<h2 id='studies_properties'>Properties of the Studies</h2><span id='topic+studies_properties'></span>

<h3>Description</h3>

<p>Return the list of study properties that can be used to search
studies and trees used in the synthetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies_properties(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="studies_properties_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The list returned has 2 elements <code>tree_properties</code> and
<code>studies_properties</code>. Each of these elements lists additional
arguments to customize the API request properties that can be used
to search for trees and studies that are contributing to the
synthetic tree. The definitions of these properties are available
from
<a href="https://github.com/OpenTreeOfLife/phylesystem-api/wiki/NexSON">https://github.com/OpenTreeOfLife/phylesystem-api/wiki/NexSON</a>
</p>


<h3>Value</h3>

<p>A list of the study properties that can be used to find
studies and trees that are contributing to the synthetic tree.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+studies_find_trees">studies_find_trees</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 all_the_properties &lt;- studies_properties()
 unlist(all_the_properties$tree_properties)

## End(Not run)
</code></pre>

<hr>
<h2 id='study_external_IDs'>Get external identifiers for data associated with an Open Tree study</h2><span id='topic+study_external_IDs'></span>

<h3>Description</h3>

<p>Data associated with studies contributing to the Open Tree synthesis may
be available from other databases. In particular, trees and alignments
may be available from treebase and DNA sequences and bibliographic
information associated with a given study may be available from the NCBI.
This function retrieves that information for a given study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>study_external_IDs(study_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="study_external_IDs_+3A_study_id">study_id</code></td>
<td>
<p>An open tree study ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A study_external_data object (which inherits from a list) which
contains some of the following.
</p>
<p>doi, character, the DOI for the paper describing this study
</p>
<p>external_data_url, character, a URL to an external data repository
(e.g. a treebase entry) if one exists.
</p>
<p>pubmed_id character, the unique ID for this study in the NCBI's pubmed database
</p>
<p>popset_ids character, vector of IDs for the NCBI's popset database
</p>
<p>nucleotide_ids character, vector of IDs for the NCBI's nucleotide database
</p>


<h3>See Also</h3>

<p>studies_find_studies (used to discover study IDs)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
flies &lt;- studies_find_studies(property="ot:focalCladeOTTTaxonName", value="Drosophilidae")
study_external_IDs(flies[2,]$study_ids)

## End(Not run)
</code></pre>

<hr>
<h2 id='synonyms.match_names'>List the synonyms for a given name</h2><span id='topic+synonyms.match_names'></span>

<h3>Description</h3>

<p>When querying the Taxonomic Name Resolution Services for a
particular taxonomic name, the API returns as possible matches all
names that include the queried name as a possible synonym. This
function allows you to explore other synonyms for an accepted
name, and allows you to determine why the name you queried is
returning an accepted synonym.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'match_names'
synonyms(tax, row_number, taxon_name, ott_id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="synonyms.match_names_+3A_tax">tax</code></td>
<td>
<p>a data frame generated by the
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> function</p>
</td></tr>
<tr><td><code id="synonyms.match_names_+3A_row_number">row_number</code></td>
<td>
<p>the row number corresponding to the name for
which to list the synonyms</p>
</td></tr>
<tr><td><code id="synonyms.match_names_+3A_taxon_name">taxon_name</code></td>
<td>
<p>the taxon name corresponding to the name for
which to list the synonyms</p>
</td></tr>
<tr><td><code id="synonyms.match_names_+3A_ott_id">ott_id</code></td>
<td>
<p>the ott id corresponding to the name for which to
list the synonyms</p>
</td></tr>
<tr><td><code id="synonyms.match_names_+3A_...">...</code></td>
<td>
<p>currently ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To list synonyms for a given taxonomic name, you need to provide
the object resulting from a call to the
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> function, as well as one of either
the row number corresponding to the name in this object, the name
itself (as used in the original query), or the ott_id listed for
this name. Otherwise, the synonyms for all the currently matched
names are returned.
</p>


<h3>Value</h3>

<p>a list whose elements are all synonym names (as vectors of
character) for the taxonomic names that match the query (the
names of the elements of the list).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   echino &lt;- tnrs_match_names(c("Diadema", "Acanthaster", "Fromia"))
   ## These 3 calls are identical
   synonyms(echino, taxon_name="Acanthaster")
   synonyms(echino, row_number=2)
   synonyms(echino, ott_id=337928)

## End(Not run)
</code></pre>

<hr>
<h2 id='tax_lineage'>Lineage of a taxon</h2><span id='topic+tax_lineage'></span><span id='topic+tax_lineage.taxon_info'></span>

<h3>Description</h3>

<p>Extract the lineage information (higher taxonomy) from an object
returned by <code><a href="#topic+taxonomy_taxon_info">taxonomy_taxon_info</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_lineage(tax, ...)

## S3 method for class 'taxon_info'
tax_lineage(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tax_lineage_+3A_tax">tax</code></td>
<td>
<p>an object created by <code><a href="#topic+taxonomy_taxon_info">taxonomy_taxon_info</a></code>
using the argument <code>include_lineage=TRUE</code>.</p>
</td></tr>
<tr><td><code id="tax_lineage_+3A_...">...</code></td>
<td>
<p>additional arguments (currently unused).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The object passed to this function must have been created using
the argument <code>include_lineage=TRUE</code>.
</p>


<h3>Value</h3>

<p>A list with one slot per taxon that contains a data frame
with 3 columns: the taxonomy rank, the name, and unique name
for all taxa included in the lineage of the taxon up to the
root of the tree.
</p>

<hr>
<h2 id='tax_rank'>Methods for Taxonomy</h2><span id='topic+tax_rank'></span><span id='topic+ott_id'></span><span id='topic+synonyms'></span><span id='topic+tax_sources'></span><span id='topic+is_suppressed'></span><span id='topic+unique_name'></span><span id='topic+tax_name'></span>

<h3>Description</h3>

<p>Methods for dealing with objects containing taxonomic information
(Taxonomy, TNRS endpoints)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_rank(tax, ...)

ott_id(tax, ...)

synonyms(tax, ...)

tax_sources(tax, ...)

is_suppressed(tax, ...)

unique_name(tax, ...)

tax_name(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tax_rank_+3A_tax">tax</code></td>
<td>
<p>an object returned by <code><a href="#topic+taxonomy_taxon_info">taxonomy_taxon_info</a></code>,
<code><a href="#topic+taxonomy_mrca">taxonomy_mrca</a></code>, or <code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code></p>
</td></tr>
<tr><td><code id="tax_rank_+3A_...">...</code></td>
<td>
<p>additional arguments (see
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the page for the generic methods. See the help pages for
<code><a href="#topic+taxonomy_taxon_info">taxonomy_taxon_info</a></code>, <code><a href="#topic+taxonomy_mrca">taxonomy_mrca</a></code>, and
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> for more information.
</p>

<hr>
<h2 id='taxon_external_IDs'>Get external identifiers for data associated with an Open Tree taxon</h2><span id='topic+taxon_external_IDs'></span>

<h3>Description</h3>

<p>The Open Tree taxonomy is a synthesis of multiple reference taxonomies. This
function retrieves identifiers to external taxonomic records that have
contributed the rank, position and definition of a given Open Tree taxon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxon_external_IDs(taxon_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxon_external_IDs_+3A_taxon_id">taxon_id</code></td>
<td>
<p>An open tree study ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame in which each row represents a unique record in an
external database. The column &quot;source&quot; provides and abbreviated name for the
database, and &quot;id&quot; the unique ID for the record.
</p>


<h3>See Also</h3>

<p>tnrs_matchnames, which can be used to search for taxa by name.
</p>
<p>taxonomy_taxon, for more information about a given taxon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   gibbon_IDs &lt;- taxon_external_IDs(712902)

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonomy_about'>Information about the Open Tree Taxonomy</h2><span id='topic+taxonomy_about'></span>

<h3>Description</h3>

<p>Summary information about the Open Tree Taxonomy (OTT)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy_about(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxonomy_about_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Return metadata and information about the taxonomy
itself. Currently, the available metadata is fairly sparse, but
includes (at least) the version, and the location from which the
complete taxonomy source files can be downloaded.
</p>


<h3>Value</h3>

<p>A list with the following properties:
</p>

<ul>
<li> <p>weburl String. The release page for this version
of the taxonomy.
</p>
</li>
<li> <p>author String. The author string.
</p>
</li>
<li> <p>name String. The name of the taxonomy.
</p>
</li>
<li> <p>source String. The full identifying information for
this version of the taxonomy.
</p>
</li>
<li> <p>version String. The version number of the taxonomy.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxonomy_about()

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonomy_mrca'>Taxonomic MRCA</h2><span id='topic+taxonomy_mrca'></span><span id='topic+tax_rank.taxon_mrca'></span><span id='topic+tax_name.taxon_mrca'></span><span id='topic+ott_id.taxon_mrca'></span><span id='topic+unique_name.taxon_mrca'></span><span id='topic+tax_sources.taxon_mrca'></span><span id='topic+flags.taxon_mrca'></span><span id='topic+is_suppressed.taxon_mrca'></span>

<h3>Description</h3>

<p>Taxonomic Least Inclusive Common Ancestor (MRCA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy_mrca(ott_ids = NULL, ...)

## S3 method for class 'taxon_mrca'
tax_rank(tax, ...)

## S3 method for class 'taxon_mrca'
tax_name(tax, ...)

## S3 method for class 'taxon_mrca'
ott_id(tax, ...)

## S3 method for class 'taxon_mrca'
unique_name(tax, ...)

## S3 method for class 'taxon_mrca'
tax_sources(tax, ...)

## S3 method for class 'taxon_mrca'
flags(tax, ...)

## S3 method for class 'taxon_mrca'
is_suppressed(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxonomy_mrca_+3A_ott_ids">ott_ids</code></td>
<td>
<p>a vector of ott ids for the taxa whose MRCA is to
be found (numeric).</p>
</td></tr>
<tr><td><code id="taxonomy_mrca_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
<tr><td><code id="taxonomy_mrca_+3A_tax">tax</code></td>
<td>
<p>an object generated by the <code>taxonomy_mrca</code>
function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a set of OTT ids, get the taxon that is the most recent common
ancestor (the MRCA) of all the identified taxa.
</p>


<h3>Value</h3>


<ul>
<li><p><code>taxonomy_mrca</code>  returns a list about the
taxonomic information relating to the MRCA for the ott_ids
provided. 
</p>
</li>
<li><p><code>tax_rank</code>  returns a character vector of the
taxonomic rank for the MRCA. 
</p>
</li>
<li><p><code>tax_name</code>  returns a character vector the
Open Tree Taxonomy name for the MRCA. 
</p>
</li>
<li><p><code>ott_id</code>  returns a numeric vector of the ott id
for the MRCA. 
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
req &lt;- taxonomy_mrca(ott_ids=c(515698,590452,643717))
tax_rank(req)
tax_name(req)
ott_id(req)

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonomy_subtree'>Taxonomy subtree</h2><span id='topic+taxonomy_subtree'></span>

<h3>Description</h3>

<p>Given an ott id, return the inclusive taxonomic subtree descended
from the specified taxon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy_subtree(
  ott_id = NULL,
  output_format = c("taxa", "newick", "phylo", "raw"),
  label_format = NULL,
  file,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxonomy_subtree_+3A_ott_id">ott_id</code></td>
<td>
<p>The ott id of the taxon of interest.</p>
</td></tr>
<tr><td><code id="taxonomy_subtree_+3A_output_format">output_format</code></td>
<td>
<p>the format of the object to be returned. See
the &lsquo;Return&rsquo; section.</p>
</td></tr>
<tr><td><code id="taxonomy_subtree_+3A_label_format">label_format</code></td>
<td>
<p>Character. Defines the label type; one of
&ldquo;<code>name</code>&rdquo;, &ldquo;<code>id</code>&rdquo;, or
&ldquo;<code>name_and_id</code>&rdquo; (the default).</p>
</td></tr>
<tr><td><code id="taxonomy_subtree_+3A_file">file</code></td>
<td>
<p>the file name where to save the output of the
function. Ignored unless <code>output_format</code> is set to
&ldquo;<code>phylo</code>&rdquo;.</p>
</td></tr>
<tr><td><code id="taxonomy_subtree_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the output of this function is exported to a file, the only
possible value for the <code>output_format</code> argument is
&ldquo;<code>newick</code>&rdquo;. If the file provided already exists, it
will be silently overwritten.
</p>


<h3>Value</h3>

<p>If the <code>file</code> argument is missing: </p>

<ul>
<li><p>&ldquo;<code>taxa</code>&rdquo;  a list of the taxa names
(species) in slot <code>tip_label</code>, and higher-level taxonomy
(e.g., families, genera) in slot <code>edge_label</code>, descending
from the taxa corresponding to the <code>ott_id</code> provided. 
</p>
</li>
<li><p>&ldquo;<code>newick</code>&rdquo;  a character vector containing
the newick formatted string corresponding to the taxonomic
subtree for the <code>ott_id</code> provided. 
</p>
</li>
<li><p>&ldquo;<code>phylo</code>&rdquo;  an object of the class
<code>phylo</code> from the <code>ape</code> package. 
</p>
</li>
<li><p>&ldquo;<code>raw</code>&rdquo;  the direct output from the API,
i.e., a list with an element named &lsquo;newick&rsquo; that
contains the subtree as a newick formatted string. 
</p>
</li></ul>

<p>If a <code>file</code> argument is provided (and
<code>output_format</code> is set to &ldquo;<code>phylo</code>&rdquo;), a
logical indicating whether the file was successfully created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
req &lt;- taxonomy_subtree(ott_id=515698)
plot(taxonomy_subtree(ott_id=515698, output_format="phylo"))

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonomy_taxon_info'>Taxon information</h2><span id='topic+taxonomy_taxon_info'></span><span id='topic+tax_rank.taxon_info'></span><span id='topic+tax_name.taxon_info'></span><span id='topic+unique_name.taxon_info'></span><span id='topic+synonyms.taxon_info'></span><span id='topic+ott_id.taxon_info'></span><span id='topic+tax_sources.taxon_info'></span><span id='topic+is_suppressed.taxon_info'></span><span id='topic+flags.taxon_info'></span>

<h3>Description</h3>

<p>Information about taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy_taxon_info(
  ott_ids,
  include_children = FALSE,
  include_lineage = FALSE,
  include_terminal_descendants = FALSE,
  ...
)

## S3 method for class 'taxon_info'
tax_rank(tax, ...)

## S3 method for class 'taxon_info'
tax_name(tax, ...)

## S3 method for class 'taxon_info'
unique_name(tax, ...)

## S3 method for class 'taxon_info'
synonyms(tax, ...)

## S3 method for class 'taxon_info'
ott_id(tax, ...)

## S3 method for class 'taxon_info'
tax_sources(tax, ...)

## S3 method for class 'taxon_info'
is_suppressed(tax, ...)

## S3 method for class 'taxon_info'
flags(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxonomy_taxon_info_+3A_ott_ids">ott_ids</code></td>
<td>
<p>the ott ids of the taxon of interest (numeric or
character containing only numbers)</p>
</td></tr>
<tr><td><code id="taxonomy_taxon_info_+3A_include_children">include_children</code></td>
<td>
<p>whether to include information about all
the children of this taxon. Default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="taxonomy_taxon_info_+3A_include_lineage">include_lineage</code></td>
<td>
<p>whether to include information about all
the higher level taxa that include the <code>ott_ids</code>.
Default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="taxonomy_taxon_info_+3A_include_terminal_descendants">include_terminal_descendants</code></td>
<td>
<p>whether to include the list of
terminal <code>ott_ids</code> contained in the <code>ott_ids</code>
provided.</p>
</td></tr>
<tr><td><code id="taxonomy_taxon_info_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
<tr><td><code id="taxonomy_taxon_info_+3A_tax">tax</code></td>
<td>
<p>an object generated by the <code>taxonomy_taxon_info</code>
function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a vector of ott ids, <code>taxonomy_taxon_info</code> returns
information about the specified taxa.
</p>
<p>The functions <code>tax_rank</code>, <code>tax_name</code>, and
<code>synonyms</code> can extract this information from an object
created by the <code>taxonomy_taxon_info()</code>.
</p>


<h3>Value</h3>

<p><code>taxonomy_taxon_info</code> returns a list detailing
information about the taxa. <code>tax_rank</code> and
<code>tax_name</code> return a vector. <code>synonyms</code> returns a
list whose elements are the synonyms for each of the
<code>ott_id</code> requested.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code> to obtain <code>ott_id</code>
from a taxonomic name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
req &lt;- taxonomy_taxon_info(ott_id=515698)
tax_rank(req)
tax_name(req)
synonyms(req)

## End(Not run)
</code></pre>

<hr>
<h2 id='tnrs_contexts'>TNRS contexts</h2><span id='topic+tnrs_contexts'></span>

<h3>Description</h3>

<p>This function returns a list of pre-defined taxonomic contexts
(i.e. clades) which can be used to limit the scope of tnrs
queries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tnrs_contexts(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tnrs_contexts_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Taxonomic contexts are available to limit the scope of TNRS
searches. These contexts correspond to uncontested higher taxa
such as 'Animals' or 'Land plants'. This service returns a list
containing all available taxonomic context names, which may be
used as input (via the <code>context_name</code> argument in other
functions) to limit the search scope of other services including
<code><a href="#topic+tnrs_match_names">tnrs_match_names</a></code>.
</p>


<h3>Value</h3>

<p>Returns invisibly a list for each major clades (e.g.,
animals, microbes, plants, fungi, life) whose elements
contains the possible contexts.
</p>

<hr>
<h2 id='tnrs_infer_context'>Infer the taxonomic context from a list of names</h2><span id='topic+tnrs_infer_context'></span>

<h3>Description</h3>

<p>Return a taxonomic context given a list of taxonomic names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tnrs_infer_context(names = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tnrs_infer_context_+3A_names">names</code></td>
<td>
<p>Vector of taxon names.</p>
</td></tr>
<tr><td><code id="tnrs_infer_context_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Find the least inclusive taxonomic context that includes all the
unambiguous names in the input set. Unambiguous names are names
with exact matches to non-homonym taxa. Ambiguous names (those
without exact matches to non-homonym taxa) are indicated in
results.
</p>


<h3>Value</h3>

<p>A list including the context name, the context ott id and
possibly the names in the query that have an ambiguous
taxonomic meaning in the query.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- tnrs_infer_context(names=c("Stellula calliope", "Struthio camelus"))

## End(Not run)
</code></pre>

<hr>
<h2 id='tnrs_match_names'>Match names to the Open Tree Taxonomy</h2><span id='topic+tnrs_match_names'></span>

<h3>Description</h3>

<p>Match taxonomic names to the Open Tree Taxonomy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tnrs_match_names(
  names = NULL,
  context_name = "All life",
  do_approximate_matching = TRUE,
  ids = NULL,
  include_suppressed = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tnrs_match_names_+3A_names">names</code></td>
<td>
<p>taxon names to be queried. Currently limited to 10,000 names
for exact matches and 2,500 names for approximate matches (character
vector)</p>
</td></tr>
<tr><td><code id="tnrs_match_names_+3A_context_name">context_name</code></td>
<td>
<p>name of the taxonomic context to be searched (length-one
character vector or <code>NULL</code>). Must match (case sensitive) one of the
values returned by <code><a href="#topic+tnrs_contexts">tnrs_contexts</a></code>. Default to &quot;All life&quot;.</p>
</td></tr>
<tr><td><code id="tnrs_match_names_+3A_do_approximate_matching">do_approximate_matching</code></td>
<td>
<p>A logical indicating whether or not to
perform approximate string (a.k.a. &ldquo;fuzzy&rdquo;) matching. Using
<code>FALSE</code> will greatly improve speed. Default, however, is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="tnrs_match_names_+3A_ids">ids</code></td>
<td>
<p>A vector of ids to use for identifying names. These will be
assigned to each name in the names array. If ids is provided, then ids and
names must be identical in length.</p>
</td></tr>
<tr><td><code id="tnrs_match_names_+3A_include_suppressed">include_suppressed</code></td>
<td>
<p>Ordinarily, some quasi-taxa, such as incertae
sedis buckets and other non-OTUs, are suppressed from TNRS results. If
this parameter is true, these quasi-taxa are allowed as possible TNRS
results.</p>
</td></tr>
<tr><td><code id="tnrs_match_names_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API request (see
<code><a href="#topic+rotl">rotl</a></code> package documentation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Accepts one or more taxonomic names and returns information about
potential matches for these names to known taxa in the Open Tree
Taxonomy.
</p>
<p>This service uses taxonomic contexts to disambiguate homonyms and misspelled
names; a context may be specified using the <code>context_name</code> argument.
The default value for <code>context_name</code> is &quot;All life&quot;. If no context is
specified (i.e., <code>context_name</code> is set to <code>NULL</code>), then the
context will be inferred (i.e., the shallowest taxonomic context that
contains all unambiguous names in the input). Taxonomic contexts are
uncontested higher taxa that have been selected to allow limits to be
applied to the scope of TNRS searches (e.g. 'match names only within
flowering plants'). Once a context has been identified (either
user-specified or inferred), all taxon name matches will performed only
against taxa within that context. For a list of available taxonomic
contexts, see <code><a href="#topic+tnrs_contexts">tnrs_contexts</a></code>.
</p>
<p>A name is considered unambiguous if it is not a synonym and has
only one exact match to any taxon name in the entire taxonomy.
</p>
<p>When the name search returns multiple matches, the taxon with the highest
match score is returned. If the name returned is not the one you intended,
you can use the <code>inspect</code> function to check the other taxa returned by
your search. The
<a href="https://docs.ropensci.org/rotl/articles/rotl.html#how-to-change-the-ott-ids-assigned-to-my-taxa">Getting
Started vignette</a> has more information on how to do this.
</p>
<p>Several functions listed in the &lsquo;See also&rsquo; section can be
used to inspect and manipulate the object generated by this
function.
</p>


<h3>Value</h3>

<p>A data frame summarizing the results of the query. The original
query output is appended as an attribute to the returned object (and can
be obtained using <code>attr(object, "original_response")</code>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+inspect.match_names">inspect.match_names</a></code>,
<code><a href="#topic+update.match_names">update.match_names</a></code>, <code><a href="#topic+synonyms.match_names">synonyms.match_names</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 deuterostomes &lt;- tnrs_match_names(names=c("echinodermata", "xenacoelomorpha",
                                            "chordata", "hemichordata"))

## End(Not run)
</code></pre>

<hr>
<h2 id='tol_about'>Information about the Tree of Life</h2><span id='topic+tol_about'></span><span id='topic+tax_rank.tol_summary'></span><span id='topic+tax_sources.tol_summary'></span><span id='topic+unique_name.tol_summary'></span><span id='topic+tax_name.tol_summary'></span><span id='topic+ott_id.tol_summary'></span>

<h3>Description</h3>

<p>Basic information about the Open Tree of Life (the synthetic tree)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tol_about(include_source_list = FALSE, ...)

## S3 method for class 'tol_summary'
tax_rank(tax, ...)

## S3 method for class 'tol_summary'
tax_sources(tax, ...)

## S3 method for class 'tol_summary'
unique_name(tax, ...)

## S3 method for class 'tol_summary'
tax_name(tax, ...)

## S3 method for class 'tol_summary'
ott_id(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tol_about_+3A_include_source_list">include_source_list</code></td>
<td>
<p>Logical (default =
<code>FALSE</code>). Return an ordered list of source trees.</p>
</td></tr>
<tr><td><code id="tol_about_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API call (see
<code><a href="#topic+rotl">rotl</a></code> for more information).</p>
</td></tr>
<tr><td><code id="tol_about_+3A_tax">tax</code></td>
<td>
<p>an object created with a call to <code>tol_about</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary information about the current draft tree of life,
including information about the list of trees and the taxonomy
used to build it. The object returned by <code>tol_about</code> can
be passed to the taxonomy methods (<code>tax_name()</code>,
<code>tax_rank()</code>, <code>tax_sources()</code>, <code>ott_id</code>), to
extract relevant taxonomic information for the root of the
synthetic tree.
</p>


<h3>Value</h3>

<p>An invisible list of synthetic tree summary statistics:
</p>

<ul>
<li> <p>date_created String. The creation date of the tree.
</p>
</li>
<li> <p>num_source_studies Integer. The number of studies
(publications)used as sources.
</p>
</li>
<li> <p>num_source_trees The number of trees used as sources
(may be &gt;1 tree per study).
</p>
</li>
<li> <p>taxonomy_version The Open Tree Taxonomy version used
as a source.
</p>
</li>
<li> <p>filtered_flags List. Taxa with these taxonomy flags were
not used in construction of the tree.
</p>
</li>
<li> <p>root List. Describes the root node:
</p>

<ul>
<li> <p>node_id String. The canonical identifier of the node.
</p>
</li>
<li> <p>num_tips Numeric. The number of descendant tips.
</p>
</li>
<li> <p>taxon A list of taxonomic properties:
</p>

<ul>
<li> <p>ott_id Numeric. The OpenTree Taxonomy ID (ott_id).
</p>
</li>
<li> <p>name String. The taxonomic name of the queried node.
</p>
</li>
<li> <p>unique_name String. The string that uniquely
identifies the taxon in OTT.
</p>
</li>
<li> <p>rank String. The taxonomic rank of the taxon in OTT.
</p>
</li>
<li> <p>tax_sources List. A list of identifiers for taxonomic
sources, such as other taxonomies, that define taxa judged
equivalent to this taxon.
</p>
</li></ul>

</li></ul>

</li>
<li> <p>source_list List. Present only if
<code>include_source_list</code> is <code>TRUE</code>. The sourceid
ordering is the precedence order for synthesis, with
relationships from earlier trees in the list having priority
over those from later trees in the list. See
<code>source_id_map</code> below for study details.
</p>
</li>
<li> <p>source_id_map Named list of lists. Present only if
<code>include_source_list</code> is <code>TRUE</code>. Names correspond to
the &lsquo;sourceids&rsquo; used in <code>source_list</code>
above. Source trees will have the following properties:
</p>

<ul>
<li> <p>git_sha String. The git SHA identifying a particular source
version.
</p>
</li>
<li> <p>tree_id String. The tree id associated with the study id used.
</p>
</li>
<li> <p>study_id String. The study identifier. Will typically include
a prefix (&quot;pg_&quot; or &quot;ot_&quot;).
</p>
</li></ul>

</li>
<li> <p>synth_id The unique string for this version of the tree.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+source_list">source_list</a></code> to explore the list of studies
used in the synthetic tree (see example).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- tol_about()
tax_sources(res)
ott_id(res)
studies &lt;- source_list(tol_about(include_source_list=TRUE))
## End(Not run)
</code></pre>

<hr>
<h2 id='tol_induced_subtree'>Subtree from the Open Tree of Life</h2><span id='topic+tol_induced_subtree'></span>

<h3>Description</h3>

<p>Return the induced subtree on the synthetic tree that relates a list of nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tol_induced_subtree(
  ott_ids = NULL,
  node_ids = NULL,
  label_format = NULL,
  file,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tol_induced_subtree_+3A_ott_ids">ott_ids</code></td>
<td>
<p>Numeric vector. OTT ids indicating nodes to be used as tips
in the induced tree.</p>
</td></tr>
<tr><td><code id="tol_induced_subtree_+3A_node_ids">node_ids</code></td>
<td>
<p>Character vector. Node ids indicating nodes to be used as
tips in the induced tree.</p>
</td></tr>
<tr><td><code id="tol_induced_subtree_+3A_label_format">label_format</code></td>
<td>
<p>Character. Defines the label type; one of
&ldquo;<code>name</code>&rdquo;, &ldquo;<code>id</code>&rdquo;, or &ldquo;<code>name_and_id</code>&rdquo;
(the default).</p>
</td></tr>
<tr><td><code id="tol_induced_subtree_+3A_file">file</code></td>
<td>
<p>If specified, the function will write the subtree to a file in
newick format.</p>
</td></tr>
<tr><td><code id="tol_induced_subtree_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API call (see
<code><a href="#topic+rotl">rotl</a></code> for more information).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Return a tree with tips corresponding to the nodes identified in
the input set that is consistent with the topology of the current
synthetic tree. This tree is equivalent to the minimal subtree
induced on the draft tree by the set of identified nodes.
</p>


<h3>Value</h3>

<p>If no value is specified to the <code>file</code> argument
(default), a phylogenetic tree of class <code>phylo</code>.
</p>
<p>Otherwise, the function returns invisibly a logical indicating
whether the file was successfully created.
</p>
<p>Note that the tree returned when specifying a file name with the
<code>file</code> argument is the &ldquo;raw&rdquo; Newick string returned by Open
Tree of Life. This string contains singleton nodes, and therefore will
be different from the tree returned as a <code>phylo</code> object which will
not contain these singleton nodes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Result as a `phylo` object
res &lt;- tol_induced_subtree(ott_ids = c(292466, 267845, 316878, 102710))

## Raw Newick string from Open Tree of Life
tree_file &lt;- tempfile(fileext = ".tre")
tol_induced_subtree(ott_ids = c(292466, 267845, 316878, 102710),
                    file=tree_file)


## End(Not run)
</code></pre>

<hr>
<h2 id='tol_lineage'>Node info</h2><span id='topic+tol_lineage'></span><span id='topic+tol_node_info'></span><span id='topic+tax_rank.tol_node'></span><span id='topic+tax_sources.tol_node'></span><span id='topic+unique_name.tol_node'></span><span id='topic+tax_name.tol_node'></span><span id='topic+ott_id.tol_node'></span><span id='topic+source_list.tol_node'></span><span id='topic+tax_lineage.tol_node'></span><span id='topic+tol_lineage.tol_node'></span>

<h3>Description</h3>

<p>Get summary information about a node in the synthetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tol_lineage(tax, ...)

tol_node_info(ott_id = NULL, node_id = NULL, include_lineage = FALSE, ...)

## S3 method for class 'tol_node'
tax_rank(tax, ...)

## S3 method for class 'tol_node'
tax_sources(tax, ...)

## S3 method for class 'tol_node'
unique_name(tax, ...)

## S3 method for class 'tol_node'
tax_name(tax, ...)

## S3 method for class 'tol_node'
ott_id(tax, ...)

## S3 method for class 'tol_node'
source_list(tax, ...)

## S3 method for class 'tol_node'
tax_lineage(tax, ...)

## S3 method for class 'tol_node'
tol_lineage(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tol_lineage_+3A_tax">tax</code></td>
<td>
<p>an object returned by <code>tol_node_info</code>.</p>
</td></tr>
<tr><td><code id="tol_lineage_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API call (see
?rotl for more information)</p>
</td></tr>
<tr><td><code id="tol_lineage_+3A_ott_id">ott_id</code></td>
<td>
<p>Numeric. The OpenTree taxonomic identifier.</p>
</td></tr>
<tr><td><code id="tol_lineage_+3A_node_id">node_id</code></td>
<td>
<p>Character. The OpenTree node identifier.</p>
</td></tr>
<tr><td><code id="tol_lineage_+3A_include_lineage">include_lineage</code></td>
<td>
<p>Logical (default = FALSE). Whether to return the
lineage of the node from the synthetic tree.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns summary information about a node in the graph. The
node of interest may be specified using either a node id or an
taxon id, but not both. If the specified node or OTT id is not
in the graph, an error will be returned.
</p>
<p>If the argument <code>include_lineage=TRUE</code> is used, you can
use <code>tax_lineage()</code> or <code>tol_lineage</code> to return the
taxonomic information or the node information for all the
ancestors to this node, down to the root of the tree.
</p>


<h3>Value</h3>

<p><code>tol_node_info</code> returns an invisible list of summary
information about the queried node:
</p>

<ul>
<li> <p>node_id String. The canonical identifier of the node.
</p>
</li>
<li> <p>num_tips Numeric. The number of descendant tips.
</p>
</li>
<li> <p>partial_path_of List. The edge below this synthetic tree node
is compatible with the edge below each of these input tree nodes (one
per tree). Each returned element is reported as sourceid:nodeid.
</p>
</li>
<li> <p>query  The node id that resolved to this node. This can differ
from the node_id field if the query id is not canonical. 
</p>
</li>
<li> <p>taxon A list of taxonomic properties. Only returned if
the queried node is a taxon. Each source has:
</p>

<ul>
<li> <p>ott_id Numeric. The OpenTree Taxonomy ID (ottID).
</p>
</li>
<li> <p>name String. The taxonomic name of the queried node.
</p>
</li>
<li> <p>unique_name String. The string that uniquely
identifies the taxon in OTT.
</p>
</li>
<li> <p>rank String. The taxonomic rank of the taxon in OTT.
</p>
</li>
<li> <p>tax_sources List. A list of identifiers for taxonomic
sources, such as other taxonomies, that define taxa judged
equivalent to this taxon.
</p>
</li></ul>

<p>The following properties list support/conflict for the node across
synthesis source trees. All properties involve sourceid keys and
nodeid values (see <code>source_id_map</code> below).
</p>
</li>
<li> <p>supported_by List. Input tree nodes (one per tree) that support
this synthetic tree node. Each returned element is reported as
sourceid:nodeid.
</p>
</li>
<li> <p>terminal List. Input tree nodes (one per tree) that are
equivalent to this synthetic tree node (via an exact mapping, or the
input tree terminal may be the only terminal descended from this
synthetic tree node. Each returned element is reported as
sourceid:nodeid.
</p>
</li>
<li> <p>conflicts_with Named list of lists. Names correspond to
sourceid keys. Each list contains input tree node ids (one or more per
tree) that conflict with this synthetic node.
</p>
</li></ul>

<p><code>tol_lineage</code> and <code>tax_lineage</code> return data
frames. <code>tol_lineage</code> indicate for each ancestor its
node identifier, the number of tips descending from that
node, and whether it corresponds to a taxonomic level.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
birds &lt;- tol_node_info(ott_id=81461, include_lineage=TRUE)
source_list(birds)
tax_rank(birds)
ott_id(birds)
tax_lineage(birds)
tol_lineage(birds)
## End(Not run)
</code></pre>

<hr>
<h2 id='tol_mrca'>MRCA of taxa from the synthetic tree</h2><span id='topic+tol_mrca'></span><span id='topic+tax_sources.tol_mrca'></span><span id='topic+unique_name.tol_mrca'></span><span id='topic+tax_name.tol_mrca'></span><span id='topic+tax_rank.tol_mrca'></span><span id='topic+ott_id.tol_mrca'></span><span id='topic+source_list.tol_mrca'></span>

<h3>Description</h3>

<p>Most Recent Common Ancestor for a set of nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tol_mrca(ott_ids = NULL, node_ids = NULL, ...)

## S3 method for class 'tol_mrca'
tax_sources(tax, ...)

## S3 method for class 'tol_mrca'
unique_name(tax, ...)

## S3 method for class 'tol_mrca'
tax_name(tax, ...)

## S3 method for class 'tol_mrca'
tax_rank(tax, ...)

## S3 method for class 'tol_mrca'
ott_id(tax, ...)

## S3 method for class 'tol_mrca'
source_list(tax, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tol_mrca_+3A_ott_ids">ott_ids</code></td>
<td>
<p>Numeric vector. The ott ids for which the MRCA is desired.</p>
</td></tr>
<tr><td><code id="tol_mrca_+3A_node_ids">node_ids</code></td>
<td>
<p>Character vector. The node ids for which the MRCA is desired.</p>
</td></tr>
<tr><td><code id="tol_mrca_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API call (see
<code><a href="#topic+rotl">rotl</a></code> for more information).</p>
</td></tr>
<tr><td><code id="tol_mrca_+3A_tax">tax</code></td>
<td>
<p>an object returned by <code>tol_mrca()</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Get the MRCA of a set of nodes on the current synthetic
tree. Accepts any combination of node ids and ott ids as
input. Returns information about the most recent common
ancestor (MRCA) node as well as the most recent taxonomic
ancestor (MRTA) node (the closest taxonomic node to the MRCA
node in the synthetic tree; the MRCA and MRTA may be the same
node). If they are the same, the taxonomic information will be
in the <code>mrca</code> slot, otherwise they will be in the
<code>nearest_taxon</code> slot of the list. If any of the specified
nodes is not in the synthetic tree an error will be returned.
</p>
<p>Taxonomic methods (<code>tax_sources()</code>, <code>ott_id()</code>,
<code>unique_name()</code>, ...) are available on the objects
returned by <code>tol_mrca()</code>. If the MRCA node is MRTA, the
name of the object returned by these methods will start with
&lsquo;ott&rsquo;, otherwise it will start with &lsquo;mrca&rsquo;.
</p>


<h3>Value</h3>

<p>An invisible list of the MRCA node properties:
</p>

<ul>
<li> <p>mrca List of node properties.
</p>

<ul>
<li> <p>node_id String. The canonical identifier of the node.
</p>
</li>
<li> <p>num_tips Numeric. The number of descendant tips.
</p>
</li>
<li> <p>taxon A list of taxonomic properties. Only returned if
the queried node is a taxon. (If the node is not a taxon, a
<code>nearest_taxon</code> list is returned (see below)).
</p>

<ul>
<li> <p>ott_id Numeric. The OpenTree Taxonomy ID (ottID).
</p>
</li>
<li> <p>name String. The taxonomic name of the queried node.
</p>
</li>
<li> <p>unique_name String. The string that uniquely
identifies the taxon in OTT.
</p>
</li>
<li> <p>rank String. The taxonomic rank of the taxon in OTT.
</p>
</li>
<li> <p>tax_sources List. A list of identifiers for taxonomic
sources, such as other taxonomies, that define taxa judged
equivalent to this taxon.
</p>
</li></ul>

<p>The following properties list support/conflict for the node across
synthesis source trees. All properties involve sourceid keys and
nodeid values (see <code>source_id_map</code> below) Not all properties are
are present for every node.
</p>
</li>
<li> <p>partial_path_of List. The edge below this synthetic tree node
is compatible with the edge below each of these input tree nodes (one
per tree). Each returned element is reported as sourceid:nodeid.
</p>
</li>
<li> <p>supported_by List. Input tree nodes (one per tree) that support
this synthetic tree node. Each returned element is reported as
sourceid:nodeid.
</p>
</li>
<li> <p>terminal List. Input tree nodes (one per tree) that are equivalent
to this synthetic tree node (via an exact mapping, or the input tree
terminal may be the only terminal descended from this synthetic tree node.
Each returned element is reported as sourceid:nodeid.
</p>
</li>
<li> <p>conflicts_with Named list of lists. Names correspond to
sourceid keys. Each list contains input tree node ids (one or more per tree)
that conflict with this synthetic node.
</p>
</li></ul>

</li>
<li> <p>nearest_taxon A list of taxonomic properties of the nearest rootward
taxon node to the MRCA node. Only returned if the MRCA node is a not taxon
(otherwise the <code>taxon</code> list above is returned).
</p>

<ul>
<li> <p>ott_id Numeric. The OpenTree Taxonomy ID (ottID).
</p>
</li>
<li> <p>name String. The taxonomic name of the queried node.
</p>
</li>
<li> <p>unique_name String. The string that uniquely
identifies the taxon in OTT.
</p>
</li>
<li> <p>rank String. The taxonomic rank of the taxon in OTT.
</p>
</li>
<li> <p>tax_sources List. A list of identifiers for taxonomic
sources, such as other taxonomies, that define taxa judged
equivalent to this taxon.
</p>
</li></ul>

</li>
<li> <p>source_id_map Named list of lists. Names correspond to the
sourceid keys used in the support/conflict properties of the <code>mrca</code>
list above. Source trees will have the following properties:
</p>

<ul>
<li> <p>git_sha The git SHA identifying a particular source
version.
</p>
</li>
<li> <p>tree_id The tree id associated with the study id used.
</p>
</li>
<li> <p>study_id The study identifier. Will typically include
a prefix (&quot;pg_&quot; or &quot;ot_&quot;).
</p>
</li></ul>

<p>The only sourceid that does not correspond to a source tree is the taxonomy,
which will have the name &quot;ott&quot;+'taxonomy_version', and the value is the
ott_id of the taxon in that taxonomy version. &quot;Taxonomy&quot; will only ever
appear in <code>supported_by</code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
birds_mrca &lt;- tol_mrca(ott_ids=c(412129, 119214))
ott_id(birds_mrca)
tax_sources(birds_mrca)

## End(Not run)
</code></pre>

<hr>
<h2 id='tol_subtree'>Extract a subtree from the synthetic tree</h2><span id='topic+tol_subtree'></span>

<h3>Description</h3>

<p>Extract a subtree from the synthetic tree from an Open Tree node id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tol_subtree(ott_id = NULL, node_id = NULL, label_format = NULL, file, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tol_subtree_+3A_ott_id">ott_id</code></td>
<td>
<p>Numeric. The ott id of the node in the tree that should
serve as the root of the tree returned.</p>
</td></tr>
<tr><td><code id="tol_subtree_+3A_node_id">node_id</code></td>
<td>
<p>Character. The node id of the node in the tree that should
serve as the root of the tree returned.</p>
</td></tr>
<tr><td><code id="tol_subtree_+3A_label_format">label_format</code></td>
<td>
<p>Character. Defines the label type; one of
&ldquo;<code>name</code>&rdquo;, &ldquo;<code>id</code>&rdquo;, or
&ldquo;<code>name_and_id</code>&rdquo; (the default).</p>
</td></tr>
<tr><td><code id="tol_subtree_+3A_file">file</code></td>
<td>
<p>If specified, the function will write the subtree to a
file in newick format.</p>
</td></tr>
<tr><td><code id="tol_subtree_+3A_...">...</code></td>
<td>
<p>additional arguments to customize the API call (see
<code><a href="#topic+rotl">rotl</a></code> for more information).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a node, return the subtree of the synthetic tree descended
from that node. The start node may be specified using either a node id
or an ott id, but not both. If the specified node is not in the
synthetic tree an error will be returned. There is a size limit of
25000 tips for this method.
</p>


<h3>Value</h3>

<p>If no value is specified to the <code>file</code> argument
(default), a phylogenetic tree of class <code>phylo</code>.
Otherwise, the function returns invisibly a logical indicating
whether the file was successfully created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- tol_subtree(ott_id=241841)
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
